<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.Graphics.GenomeDiagram._Diagram</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.Graphics-module.html">Package&nbsp;Graphics</a> ::
        <a href="Bio.Graphics.GenomeDiagram-module.html">Package&nbsp;GenomeDiagram</a> ::
        Module&nbsp;_Diagram
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.Graphics.GenomeDiagram._Diagram-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.Graphics.GenomeDiagram._Diagram-module.html" onclick="show_private();">Module Bio.Graphics.GenomeDiagram._Diagram</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2003-2008 by Leighton Pritchard.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># This file is part of the Biopython distribution and governed by your</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># choice of the "Biopython License Agreement" or the "BSD 3-Clause License".</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-comment"># Please see the LICENSE file that should have been included as part of this</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-comment"># package.</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-comment"># Contact:       Leighton Pritchard, The James Hutton Institute,</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-comment">#                Invergowrie, Dundee, Scotland, DD2 5DA, UK</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-comment">#                Leighton.Pritchard@hutton.ac.uk</tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-docstring">"""Provides a container for information concerning the tracks to be drawn in a diagram.</tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-docstring">It also provides the interface for defining the diagram (possibly split these</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-docstring">functions in later version?).</tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-docstring">For drawing capabilities, this module uses reportlab to draw and write the</tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-docstring">diagram:</tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-docstring">http://www.reportlab.com</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-docstring">For dealing with biological information, the package expects BioPython</tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-docstring">objects - namely SeqRecord objects containing SeqFeature objects.</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt class="py-name">reportlab</tt><tt class="py-op">.</tt><tt class="py-name">graphics</tt> <tt class="py-keyword">import</tt> <tt class="py-name">renderPM</tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"><tt class="py-keyword">except</tt> <tt class="py-name">ImportError</tt><tt class="py-op">:</tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line">    <tt class="py-comment"># This is an optional part of ReportLab, so may not be installed.</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line">    <tt class="py-name">renderPM</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt><tt id="link-0" class="py-name" targets="Module Bio.Graphics.GenomeDiagram._LinearDrawer=Bio.Graphics.GenomeDiagram._LinearDrawer-module.html"><a title="Bio.Graphics.GenomeDiagram._LinearDrawer" class="py-name" href="#" onclick="return doclink('link-0', '_LinearDrawer', 'link-0');">_LinearDrawer</a></tt> <tt class="py-keyword">import</tt> <tt id="link-1" class="py-name" targets="Class Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html"><a title="Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer" class="py-name" href="#" onclick="return doclink('link-1', 'LinearDrawer', 'link-1');">LinearDrawer</a></tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt><tt id="link-2" class="py-name" targets="Module Bio.Graphics.GenomeDiagram._CircularDrawer=Bio.Graphics.GenomeDiagram._CircularDrawer-module.html"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer" class="py-name" href="#" onclick="return doclink('link-2', '_CircularDrawer', 'link-2');">_CircularDrawer</a></tt> <tt class="py-keyword">import</tt> <tt id="link-3" class="py-name" targets="Class Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer" class="py-name" href="#" onclick="return doclink('link-3', 'CircularDrawer', 'link-3');">CircularDrawer</a></tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt><tt id="link-4" class="py-name" targets="Module Bio.Graphics.GenomeDiagram._Track=Bio.Graphics.GenomeDiagram._Track-module.html"><a title="Bio.Graphics.GenomeDiagram._Track" class="py-name" href="#" onclick="return doclink('link-4', '_Track', 'link-4');">_Track</a></tt> <tt class="py-keyword">import</tt> <tt id="link-5" class="py-name" targets="Class Bio.Graphics.GenomeDiagram._Track.Track=Bio.Graphics.GenomeDiagram._Track.Track-class.html"><a title="Bio.Graphics.GenomeDiagram._Track.Track" class="py-name" href="#" onclick="return doclink('link-5', 'Track', 'link-5');">Track</a></tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-6" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-6', 'Bio', 'link-6');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-7" class="py-name" targets="Package Bio.Graphics=Bio.Graphics-module.html,Class Bio.KEGG.KGML.KGML_pathway.Graphics=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html"><a title="Bio.Graphics
Bio.KEGG.KGML.KGML_pathway.Graphics" class="py-name" href="#" onclick="return doclink('link-7', 'Graphics', 'link-7');">Graphics</a></tt> <tt class="py-keyword">import</tt> <tt id="link-8" class="py-name" targets="Function Bio.Graphics._write()=Bio.Graphics-module.html#_write"><a title="Bio.Graphics._write" class="py-name" href="#" onclick="return doclink('link-8', '_write', 'link-8');">_write</a></tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"> </tt>
<a name="_first_defined"></a><div id="_first_defined-def"><a name="L38"></a><tt class="py-lineno"> 38</tt> <a class="py-toggle" href="#" id="_first_defined-toggle" onclick="return toggle('_first_defined');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Diagram-module.html#_first_defined">_first_defined</a><tt class="py-op">(</tt><tt class="py-op">*</tt><tt class="py-param">args</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_first_defined-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_first_defined-expanded"><a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return the first non-null argument (PRIVATE)."""</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">arg</tt> <tt class="py-keyword">in</tt> <tt class="py-name">args</tt><tt class="py-op">:</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">arg</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">arg</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line"> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line"> </tt>
<a name="Diagram"></a><div id="Diagram-def"><a name="L46"></a><tt class="py-lineno"> 46</tt> <a class="py-toggle" href="#" id="Diagram-toggle" onclick="return toggle('Diagram');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html">Diagram</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Diagram-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Diagram-expanded"><a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line">    <tt class="py-docstring">"""Diagram container.</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"><tt class="py-docstring">    Arguments:</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line"><tt class="py-docstring">    - name           - a string, identifier for the diagram.</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line"><tt class="py-docstring">    - tracks         - a list of Track objects comprising the diagram.</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line"><tt class="py-docstring">    - format         - a string, format of the diagram 'circular' or</tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line"><tt class="py-docstring">      'linear', depending on the sort of diagram required.</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line"><tt class="py-docstring">    - pagesize       - a string, the pagesize of output describing the ISO</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line"><tt class="py-docstring">      size of the image, or a tuple of pixels.</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"><tt class="py-docstring">    - orientation    - a string describing the required orientation of the</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line"><tt class="py-docstring">      final drawing ('landscape' or 'portrait').</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line"><tt class="py-docstring">    - x              - a float (0-&gt;1), the proportion of the page to take</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line"><tt class="py-docstring">      up with even X margins t the page.</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line"><tt class="py-docstring">    - y              - a float (0-&gt;1), the proportion of the page to take</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line"><tt class="py-docstring">      up with even Y margins to the page.</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line"><tt class="py-docstring">    - xl             - a float (0-&gt;1), the proportion of the page to take</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line"><tt class="py-docstring">      up with the left X margin to the page (overrides x).</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line"><tt class="py-docstring">    - xr             - a float (0-&gt;1), the proportion of the page to take</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line"><tt class="py-docstring">      up with the right X margin to the page (overrides x).</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line"><tt class="py-docstring">    - yt             - a float (0-&gt;1), the proportion of the page to take</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line"><tt class="py-docstring">      up with the top Y margin to the page (overrides y).</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line"><tt class="py-docstring">    - yb             - a float (0-&gt;1), the proportion of the page to take</tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line"><tt class="py-docstring">      up with the bottom Y margin to the page (overrides y).</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line"><tt class="py-docstring">    - circle_core    - a float, the proportion of the available radius to</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line"><tt class="py-docstring">      leave empty at the center of a circular diagram (0 to 1).</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line"><tt class="py-docstring">    - start          - an integer, the base/aa position to start the diagram at.</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line"><tt class="py-docstring">    - end            - an integer, the base/aa position to end the diagram at.</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line"><tt class="py-docstring">    - tracklines     - a boolean, True if track guidelines are to be drawn.</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line"><tt class="py-docstring">    - fragments      - and integer, for a linear diagram, the number of equal</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line"><tt class="py-docstring">      divisions into which the sequence is divided.</tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line"><tt class="py-docstring">    - fragment_size  - a float (0-&gt;1), the proportion of the space</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line"><tt class="py-docstring">      available to each fragment that should be used in drawing.</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line"><tt class="py-docstring">    - track_size     - a float (0-&gt;1), the proportion of the space</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line"><tt class="py-docstring">      available to each track that should be used in drawing with sigils.</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line"><tt class="py-docstring">    - circular       - a boolean, True if the genome/sequence to be drawn</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line"><tt class="py-docstring">      is, in reality, circular.</tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line"> </tt>
<a name="Diagram.__init__"></a><div id="Diagram.__init__-def"><a name="L87"></a><tt class="py-lineno"> 87</tt> <a class="py-toggle" href="#" id="Diagram.__init__-toggle" onclick="return toggle('Diagram.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">format</tt><tt class="py-op">=</tt><tt class="py-string">'circular'</tt><tt class="py-op">,</tt> <tt class="py-param">pagesize</tt><tt class="py-op">=</tt><tt class="py-string">'A3'</tt><tt class="py-op">,</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line">                 <tt class="py-param">orientation</tt><tt class="py-op">=</tt><tt class="py-string">'landscape'</tt><tt class="py-op">,</tt> <tt class="py-param">x</tt><tt class="py-op">=</tt><tt class="py-number">0.05</tt><tt class="py-op">,</tt> <tt class="py-param">y</tt><tt class="py-op">=</tt><tt class="py-number">0.05</tt><tt class="py-op">,</tt> <tt class="py-param">xl</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line">                 <tt class="py-param">xr</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">yt</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">yb</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line">                 <tt class="py-param">tracklines</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> <tt class="py-param">fragments</tt><tt class="py-op">=</tt><tt class="py-number">10</tt><tt class="py-op">,</tt> <tt class="py-param">fragment_size</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">                 <tt class="py-param">track_size</tt><tt class="py-op">=</tt><tt class="py-number">0.75</tt><tt class="py-op">,</tt> <tt class="py-param">circular</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">,</tt> <tt class="py-param">circle_core</tt><tt class="py-op">=</tt><tt class="py-number">0.0</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Diagram.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Diagram.__init__-expanded"><a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize.</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line"><tt class="py-docstring">        gdd = Diagram(name=None)</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt>   <tt class="py-comment"># Holds all Track objects, keyed by level</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-9" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-9', 'name', 'link-9');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-10" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-10', 'name', 'link-9');">name</a></tt>    <tt class="py-comment"># Description of the diagram</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line">        <tt class="py-comment"># Diagram page setup attributes</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-11" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.format()=Bio.Align.MultipleSeqAlignment-class.html#format,Method Bio.Phylo.BaseTree.Tree.format()=Bio.Phylo.BaseTree.Tree-class.html#format,Method Bio.Restriction.Restriction.RestrictionBatch.format()=Bio.Restriction.Restriction.RestrictionBatch-class.html#format,Method Bio.SeqRecord.SeqRecord.format()=Bio.SeqRecord.SeqRecord-class.html#format,Method Bio.motifs.Motif.format()=Bio.motifs.Motif-class.html#format"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-11', 'format', 'link-11');">format</a></tt> <tt class="py-op">=</tt> <tt id="link-12" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-12', 'format', 'link-11');">format</a></tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pagesize</tt> <tt class="py-op">=</tt> <tt class="py-name">pagesize</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">orientation</tt> <tt class="py-op">=</tt> <tt class="py-name">orientation</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-13" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.x=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#x,Variable Bio.Statistics.lowess.x=Bio.Statistics.lowess-module.html#x"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-13', 'x', 'link-13');">x</a></tt> <tt class="py-op">=</tt> <tt id="link-14" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-14', 'x', 'link-13');">x</a></tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-15" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.y=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#y"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-15', 'y', 'link-15');">y</a></tt> <tt class="py-op">=</tt> <tt id="link-16" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-16', 'y', 'link-15');">y</a></tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xl</tt> <tt class="py-op">=</tt> <tt class="py-name">xl</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xr</tt> <tt class="py-op">=</tt> <tt class="py-name">xr</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">yt</tt> <tt class="py-op">=</tt> <tt class="py-name">yt</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">yb</tt> <tt class="py-op">=</tt> <tt class="py-name">yb</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-17" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.start=Bio.SeqFeature.CompoundLocation-class.html#start,Variable Bio.SeqFeature.FeatureLocation.start=Bio.SeqFeature.FeatureLocation-class.html#start"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-17', 'start', 'link-17');">start</a></tt> <tt class="py-op">=</tt> <tt id="link-18" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-18', 'start', 'link-17');">start</a></tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-19" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.end=Bio.SeqFeature.CompoundLocation-class.html#end,Variable Bio.SeqFeature.FeatureLocation.end=Bio.SeqFeature.FeatureLocation-class.html#end"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-19', 'end', 'link-19');">end</a></tt> <tt class="py-op">=</tt> <tt id="link-20" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-20', 'end', 'link-19');">end</a></tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracklines</tt> <tt class="py-op">=</tt> <tt class="py-name">tracklines</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-21" class="py-name" targets="Variable Bio.SearchIO._model.hit.Hit.fragments=Bio.SearchIO._model.hit.Hit-class.html#fragments,Variable Bio.SearchIO._model.hsp.HSP.fragments=Bio.SearchIO._model.hsp.HSP-class.html#fragments,Variable Bio.SearchIO._model.query.QueryResult.fragments=Bio.SearchIO._model.query.QueryResult-class.html#fragments"><a title="Bio.SearchIO._model.hit.Hit.fragments
Bio.SearchIO._model.hsp.HSP.fragments
Bio.SearchIO._model.query.QueryResult.fragments" class="py-name" href="#" onclick="return doclink('link-21', 'fragments', 'link-21');">fragments</a></tt> <tt class="py-op">=</tt> <tt id="link-22" class="py-name"><a title="Bio.SearchIO._model.hit.Hit.fragments
Bio.SearchIO._model.hsp.HSP.fragments
Bio.SearchIO._model.query.QueryResult.fragments" class="py-name" href="#" onclick="return doclink('link-22', 'fragments', 'link-21');">fragments</a></tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">fragment_size</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">fragment_size</tt> <tt class="py-op">=</tt> <tt class="py-name">fragment_size</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-23" class="py-name"><a title="Bio.SearchIO._model.hit.Hit.fragments
Bio.SearchIO._model.hsp.HSP.fragments
Bio.SearchIO._model.query.QueryResult.fragments" class="py-name" href="#" onclick="return doclink('link-23', 'fragments', 'link-21');">fragments</a></tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">                <tt class="py-comment"># For single fragments, default to full height</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">fragment_size</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line">                <tt class="py-comment"># Otherwise keep a 10% gap between fragments</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">fragment_size</tt> <tt class="py-op">=</tt> <tt class="py-number">.9</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">track_size</tt> <tt class="py-op">=</tt> <tt class="py-name">track_size</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">circular</tt> <tt class="py-op">=</tt> <tt class="py-name">circular</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">circle_core</tt> <tt class="py-op">=</tt> <tt class="py-name">circle_core</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">cross_track_links</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">drawing</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line"> </tt>
<a name="Diagram.set_all_tracks"></a><div id="Diagram.set_all_tracks-def"><a name="L127"></a><tt class="py-lineno">127</tt> <a class="py-toggle" href="#" id="Diagram.set_all_tracks-toggle" onclick="return toggle('Diagram.set_all_tracks');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#set_all_tracks">set_all_tracks</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">attr</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Diagram.set_all_tracks-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Diagram.set_all_tracks-expanded"><a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line">        <tt class="py-docstring">"""Set the passed attribute of all tracks in the set to the passed value.</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line"><tt class="py-docstring">            - attr    - An attribute of the Track class.</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line"><tt class="py-docstring">            - value   - The value to set that attribute.</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line"><tt class="py-docstring">        set_all_tracks(self, attr, value)</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">track</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">.</tt><tt id="link-24" class="py-name" targets="Method Bio.Crystal.Crystal.values()=Bio.Crystal.Crystal-class.html#values,Method Bio.File._IndexedSeqFileDict.values()=Bio.File._IndexedSeqFileDict-class.html#values,Method Bio.Phylo.PhyloXML.Events.values()=Bio.Phylo.PhyloXML.Events-class.html#values,Method BioSQL.BioSeqDatabase.BioSeqDatabase.values()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#values,Method BioSQL.BioSeqDatabase.DBServer.values()=BioSQL.BioSeqDatabase.DBServer-class.html#values"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-24', 'values', 'link-24');">values</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">track</tt><tt class="py-op">,</tt> <tt class="py-name">attr</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line">                <tt class="py-comment"># If the feature has the attribute set it to the passed value</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">track</tt><tt class="py-op">,</tt> <tt class="py-name">attr</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt id="link-25" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.value()=Bio.Phylo.PhyloXMLIO.Writer-class.html#value"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-25', 'value', 'link-25');">value</a></tt><tt class="py-op">:</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line">                    <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt class="py-name">track</tt><tt class="py-op">,</tt> <tt class="py-name">attr</tt><tt class="py-op">,</tt> <tt id="link-26" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-26', 'value', 'link-25');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line"> </tt>
<a name="Diagram.draw"></a><div id="Diagram.draw-def"><a name="L142"></a><tt class="py-lineno">142</tt> <a class="py-toggle" href="#" id="Diagram.draw-toggle" onclick="return toggle('Diagram.draw');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#draw">draw</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">format</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">pagesize</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">orientation</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line">             <tt class="py-param">x</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">y</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">xl</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">xr</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">yt</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">yb</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line">             <tt class="py-param">start</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">tracklines</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">fragments</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">             <tt class="py-param">fragment_size</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">track_size</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">circular</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">             <tt class="py-param">circle_core</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">cross_track_links</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Diagram.draw-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Diagram.draw-expanded"><a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line">        <tt class="py-docstring">"""Draw the diagram, with passed parameters overriding existing attributes.</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line"><tt class="py-docstring">        gdd.draw(format='circular')</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">        <tt class="py-comment"># Pass the parameters to the drawer objects that will build the</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">        <tt class="py-comment"># diagrams.  At the moment, we detect overrides with an or in the</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">        <tt class="py-comment"># Instantiation arguments, but I suspect there's a neater way to do</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line">        <tt class="py-comment"># this.</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-27" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-27', 'format', 'link-11');">format</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'linear'</tt><tt class="py-op">:</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">            <tt class="py-name">drawer</tt> <tt class="py-op">=</tt> <tt id="link-28" class="py-name"><a title="Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer" class="py-name" href="#" onclick="return doclink('link-28', 'LinearDrawer', 'link-1');">LinearDrawer</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt id="link-29" class="py-name" targets="Function Bio.Graphics.GenomeDiagram._Diagram._first_defined()=Bio.Graphics.GenomeDiagram._Diagram-module.html#_first_defined"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-29', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt class="py-name">pagesize</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pagesize</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">                                  <tt id="link-30" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-30', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt class="py-name">orientation</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">orientation</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">                                  <tt id="link-31" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-31', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt id="link-32" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-32', 'x', 'link-13');">x</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-33" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-33', 'x', 'link-13');">x</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">                                  <tt id="link-34" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-34', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt id="link-35" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-35', 'y', 'link-15');">y</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-36" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-36', 'y', 'link-15');">y</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">                                  <tt id="link-37" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-37', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt class="py-name">xl</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xl</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">                                  <tt id="link-38" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-38', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt class="py-name">xr</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xr</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">                                  <tt id="link-39" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-39', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt class="py-name">yt</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">yt</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">                                  <tt id="link-40" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-40', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt class="py-name">yb</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">yb</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">                                  <tt id="link-41" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-41', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt id="link-42" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-42', 'start', 'link-17');">start</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-43" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-43', 'start', 'link-17');">start</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">                                  <tt id="link-44" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-44', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt id="link-45" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-45', 'end', 'link-19');">end</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-46" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-46', 'end', 'link-19');">end</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">                                  <tt id="link-47" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-47', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt class="py-name">tracklines</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracklines</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">                                  <tt id="link-48" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-48', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt id="link-49" class="py-name"><a title="Bio.SearchIO._model.hit.Hit.fragments
Bio.SearchIO._model.hsp.HSP.fragments
Bio.SearchIO._model.query.QueryResult.fragments" class="py-name" href="#" onclick="return doclink('link-49', 'fragments', 'link-21');">fragments</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-50" class="py-name"><a title="Bio.SearchIO._model.hit.Hit.fragments
Bio.SearchIO._model.hsp.HSP.fragments
Bio.SearchIO._model.query.QueryResult.fragments" class="py-name" href="#" onclick="return doclink('link-50', 'fragments', 'link-21');">fragments</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">                                  <tt id="link-51" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-51', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt class="py-name">fragment_size</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">fragment_size</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line">                                  <tt id="link-52" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-52', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt class="py-name">track_size</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">track_size</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line">                                  <tt id="link-53" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-53', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt class="py-name">cross_track_links</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">cross_track_links</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line">            <tt class="py-name">drawer</tt> <tt class="py-op">=</tt> <tt id="link-54" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer" class="py-name" href="#" onclick="return doclink('link-54', 'CircularDrawer', 'link-3');">CircularDrawer</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt id="link-55" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-55', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt class="py-name">pagesize</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pagesize</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line">                                    <tt id="link-56" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-56', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt class="py-name">orientation</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">orientation</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line">                                    <tt id="link-57" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-57', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt id="link-58" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-58', 'x', 'link-13');">x</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-59" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-59', 'x', 'link-13');">x</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line">                                    <tt id="link-60" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-60', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt id="link-61" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-61', 'y', 'link-15');">y</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-62" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-62', 'y', 'link-15');">y</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">                                    <tt id="link-63" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-63', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt class="py-name">xl</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xl</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">                                    <tt id="link-64" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-64', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt class="py-name">xr</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xr</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">                                    <tt id="link-65" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-65', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt class="py-name">yt</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">yt</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">                                    <tt id="link-66" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-66', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt class="py-name">yb</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">yb</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">                                    <tt id="link-67" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-67', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt id="link-68" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-68', 'start', 'link-17');">start</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-69" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-69', 'start', 'link-17');">start</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">                                    <tt id="link-70" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-70', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt id="link-71" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-71', 'end', 'link-19');">end</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-72" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-72', 'end', 'link-19');">end</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">                                    <tt id="link-73" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-73', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt class="py-name">tracklines</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracklines</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">                                    <tt id="link-74" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-74', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt class="py-name">track_size</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">track_size</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">                                    <tt id="link-75" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-75', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt class="py-name">circular</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">circular</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">                                    <tt id="link-76" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-76', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt class="py-name">circle_core</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">circle_core</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line">                                    <tt id="link-77" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram._first_defined" class="py-name" href="#" onclick="return doclink('link-77', '_first_defined', 'link-29');">_first_defined</a></tt><tt class="py-op">(</tt><tt class="py-name">cross_track_links</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">cross_track_links</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">        <tt class="py-name">drawer</tt><tt class="py-op">.</tt><tt id="link-78" class="py-name" targets="Method Bio.Graphics.BasicChromosome.Chromosome.draw()=Bio.Graphics.BasicChromosome.Chromosome-class.html#draw,Method Bio.Graphics.BasicChromosome.ChromosomeSegment.draw()=Bio.Graphics.BasicChromosome.ChromosomeSegment-class.html#draw,Method Bio.Graphics.BasicChromosome.Organism.draw()=Bio.Graphics.BasicChromosome.Organism-class.html#draw,Method Bio.Graphics.BasicChromosome.SpacerSegment.draw()=Bio.Graphics.BasicChromosome.SpacerSegment-class.html#draw,Method Bio.Graphics.BasicChromosome._ChromosomeComponent.draw()=Bio.Graphics.BasicChromosome._ChromosomeComponent-class.html#draw,Method Bio.Graphics.Distribution.BarChartDistribution.draw()=Bio.Graphics.Distribution.BarChartDistribution-class.html#draw,Method Bio.Graphics.Distribution.DistributionPage.draw()=Bio.Graphics.Distribution.DistributionPage-class.html#draw,Method Bio.Graphics.Distribution.LineDistribution.draw()=Bio.Graphics.Distribution.LineDistribution-class.html#draw,Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.draw()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#draw,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#draw,Method Bio.Graphics.KGML_vis.KGMLCanvas.draw()=Bio.Graphics.KGML_vis.KGMLCanvas-class.html#draw,Function Bio.Phylo._utils.draw()=Bio.Phylo._utils-module.html#draw"><a title="Bio.Graphics.BasicChromosome.Chromosome.draw
Bio.Graphics.BasicChromosome.ChromosomeSegment.draw
Bio.Graphics.BasicChromosome.Organism.draw
Bio.Graphics.BasicChromosome.SpacerSegment.draw
Bio.Graphics.BasicChromosome._ChromosomeComponent.draw
Bio.Graphics.Distribution.BarChartDistribution.draw
Bio.Graphics.Distribution.DistributionPage.draw
Bio.Graphics.Distribution.LineDistribution.draw
Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw
Bio.Graphics.GenomeDiagram._Diagram.Diagram.draw
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw
Bio.Graphics.KGML_vis.KGMLCanvas.draw
Bio.Phylo._utils.draw" class="py-name" href="#" onclick="return doclink('link-78', 'draw', 'link-78');">draw</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt>   <tt class="py-comment"># Tell the drawer to complete the drawing</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">drawing</tt> <tt class="py-op">=</tt> <tt class="py-name">drawer</tt><tt class="py-op">.</tt><tt class="py-name">drawing</tt>  <tt class="py-comment"># Get the completed drawing</tt> </tt>
</div><a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line"> </tt>
<a name="Diagram.write"></a><div id="Diagram.write-def"><a name="L190"></a><tt class="py-lineno">190</tt> <a class="py-toggle" href="#" id="Diagram.write-toggle" onclick="return toggle('Diagram.write');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#write">write</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">filename</tt><tt class="py-op">=</tt><tt class="py-string">'test1.ps'</tt><tt class="py-op">,</tt> <tt class="py-param">output</tt><tt class="py-op">=</tt><tt class="py-string">'PS'</tt><tt class="py-op">,</tt> <tt class="py-param">dpi</tt><tt class="py-op">=</tt><tt class="py-number">72</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Diagram.write-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Diagram.write-expanded"><a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">        <tt class="py-docstring">"""Write the drawn diagram to a specified file, in a specified format.</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line"><tt class="py-docstring">            - filename   - a string indicating the name of the output file,</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line"><tt class="py-docstring">              or a handle to write to.</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line"><tt class="py-docstring">            - output     - a string indicating output format, one of PS, PDF,</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line"><tt class="py-docstring">              SVG, or provided the ReportLab renderPM module is installed, one</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line"><tt class="py-docstring">              of the bitmap formats JPG, BMP, GIF, PNG, TIFF or TIFF.  The</tt> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line"><tt class="py-docstring">              format can be given in upper or lower case.</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line"><tt class="py-docstring">            - dpi        - an integer. Resolution (dots per inch) for bitmap formats.</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line"><tt class="py-docstring">        Returns:</tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line"><tt class="py-docstring">            No return value.</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line"><tt class="py-docstring">        write(self, filename='test1.ps', output='PS', dpi=72)</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-79" class="py-name"><a title="Bio.Graphics._write" class="py-name" href="#" onclick="return doclink('link-79', '_write', 'link-8');">_write</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">drawing</tt><tt class="py-op">,</tt> <tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-name">output</tt><tt class="py-op">,</tt> <tt class="py-name">dpi</tt><tt class="py-op">=</tt><tt class="py-name">dpi</tt><tt class="py-op">)</tt> </tt>
</div><a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line"> </tt>
<a name="Diagram.write_to_string"></a><div id="Diagram.write_to_string-def"><a name="L210"></a><tt class="py-lineno">210</tt> <a class="py-toggle" href="#" id="Diagram.write_to_string-toggle" onclick="return toggle('Diagram.write_to_string');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#write_to_string">write_to_string</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">output</tt><tt class="py-op">=</tt><tt class="py-string">'PS'</tt><tt class="py-op">,</tt> <tt class="py-param">dpi</tt><tt class="py-op">=</tt><tt class="py-number">72</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Diagram.write_to_string-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Diagram.write_to_string-expanded"><a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a byte string containing the diagram in the requested format.</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line"><tt class="py-docstring">            - output    - a string indicating output format, one of PS, PDF,</tt> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line"><tt class="py-docstring">              SVG, JPG, BMP, GIF, PNG, TIFF or TIFF (as specified for the write</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line"><tt class="py-docstring">              method).</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line"><tt class="py-docstring">            - dpi       - Resolution (dots per inch) for bitmap formats.</tt> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line"><tt class="py-docstring">        Returns:</tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line"><tt class="py-docstring">            Return the completed drawing as a bytes string in a prescribed</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line"><tt class="py-docstring">            format.</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line">        <tt class="py-comment"># The ReportLab drawToString method, which this function used to call,</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line">        <tt class="py-comment"># just used a cStringIO or StringIO handle with the drawToFile method.</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line">        <tt class="py-comment"># In order to put all our complicated file format specific code in one</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line">        <tt class="py-comment"># place we just used a StringIO handle here, later a BytesIO handle</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line">        <tt class="py-comment"># for Python 3 compatibility.</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Rename this method to include keyword bytes?</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt class="py-name">io</tt> <tt class="py-keyword">import</tt> <tt class="py-name">BytesIO</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">BytesIO</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-80" class="py-name" targets="Function Bio.AlignIO.write()=Bio.AlignIO-module.html#write,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.write()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#write,Method Bio.Phylo.CDAOIO.Writer.write()=Bio.Phylo.CDAOIO.Writer-class.html#write,Function Bio.Phylo.CDAOIO.write()=Bio.Phylo.CDAOIO-module.html#write,Method Bio.Phylo.NeXMLIO.Writer.write()=Bio.Phylo.NeXMLIO.Writer-class.html#write,Function Bio.Phylo.NeXMLIO.write()=Bio.Phylo.NeXMLIO-module.html#write,Method Bio.Phylo.NewickIO.Writer.write()=Bio.Phylo.NewickIO.Writer-class.html#write,Function Bio.Phylo.NewickIO.write()=Bio.Phylo.NewickIO-module.html#write,Function Bio.Phylo.NexusIO.write()=Bio.Phylo.NexusIO-module.html#write,Method Bio.Phylo.PhyloXMLIO.Writer.write()=Bio.Phylo.PhyloXMLIO.Writer-class.html#write,Function Bio.Phylo.PhyloXMLIO.write()=Bio.Phylo.PhyloXMLIO-module.html#write,Function Bio.Phylo._io.write()=Bio.Phylo._io-module.html#write,Function Bio.SearchIO.write()=Bio.SearchIO-module.html#write,Function Bio.SeqIO.write()=Bio.SeqIO-module.html#write,Method Bio.bgzf.BgzfWriter.write()=Bio.bgzf.BgzfWriter-class.html#write,Function Bio.motifs.clusterbuster.write()=Bio.motifs.clusterbuster-module.html#write,Function Bio.motifs.jaspar.write()=Bio.motifs.jaspar-module.html#write,Function Bio.motifs.pfm.write()=Bio.motifs.pfm-module.html#write,Function Bio.motifs.transfac.write()=Bio.motifs.transfac-module.html#write,Function Bio.motifs.write()=Bio.motifs-module.html#write,Method Bio.phenotype.phen_micro.JsonWriter.write()=Bio.phenotype.phen_micro.JsonWriter-class.html#write,Function Bio.phenotype.write()=Bio.phenotype-module.html#write"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.clusterbuster.write
Bio.motifs.jaspar.write
Bio.motifs.pfm.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-80', 'write', 'link-80');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-name">output</tt><tt class="py-op">,</tt> <tt class="py-name">dpi</tt><tt class="py-op">)</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt class="py-name">getvalue</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line"> </tt>
<a name="Diagram.add_track"></a><div id="Diagram.add_track-def"><a name="L236"></a><tt class="py-lineno">236</tt> <a class="py-toggle" href="#" id="Diagram.add_track-toggle" onclick="return toggle('Diagram.add_track');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#add_track">add_track</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">track</tt><tt class="py-op">,</tt> <tt class="py-param">track_level</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Diagram.add_track-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Diagram.add_track-expanded"><a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add a Track object to the diagram.</tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line"><tt class="py-docstring">        It also accepts instructions to place it at a particular level on the</tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line"><tt class="py-docstring">        diagram.</tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line"><tt class="py-docstring">            - track          - Track object to draw.</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line"><tt class="py-docstring">            - track_level    - an integer. The level at which the track will be</tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line"><tt class="py-docstring">              drawn (above an arbitrary baseline).</tt> </tt>
<a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line"><tt class="py-docstring">        add_track(self, track, track_level)</tt> </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">track</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Must specify track"</tt><tt class="py-op">)</tt> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">track_level</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">:</tt>     <tt class="py-comment"># No track at that level</tt> </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">[</tt><tt class="py-name">track_level</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">track</tt>   <tt class="py-comment"># so just add it</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt>       <tt class="py-comment"># Already a track there, so shunt all higher tracks up one</tt> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line">            <tt class="py-name">occupied_levels</tt> <tt class="py-op">=</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-81" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.get_levels()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#get_levels"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.get_levels" class="py-name" href="#" onclick="return doclink('link-81', 'get_levels', 'link-81');">get_levels</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt>  <tt class="py-comment"># Get list of occupied levels...</tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line">            <tt class="py-name">occupied_levels</tt><tt class="py-op">.</tt><tt id="link-82" class="py-name" targets="Method Bio.Pathway.Reaction.reverse()=Bio.Pathway.Reaction-class.html#reverse,Method Bio.Seq.MutableSeq.reverse()=Bio.Seq.MutableSeq-class.html#reverse"><a title="Bio.Pathway.Reaction.reverse
Bio.Seq.MutableSeq.reverse" class="py-name" href="#" onclick="return doclink('link-82', 'reverse', 'link-82');">reverse</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt>           <tt class="py-comment"># ...reverse it (highest first)</tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">val</tt> <tt class="py-keyword">in</tt> <tt class="py-name">occupied_levels</tt><tt class="py-op">:</tt> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line">                <tt class="py-comment"># If track value &gt;= that to be added</tt> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">val</tt> <tt class="py-op">&gt;=</tt> <tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">track_level</tt><tt class="py-op">:</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">[</tt><tt class="py-name">val</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">[</tt><tt class="py-name">val</tt><tt class="py-op">]</tt>  <tt class="py-comment"># ...increment by 1</tt> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">[</tt><tt class="py-name">track_level</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">track</tt>   <tt class="py-comment"># And put the new track in</tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">[</tt><tt class="py-name">track_level</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">track_level</tt> <tt class="py-op">=</tt> <tt class="py-name">track_level</tt> </tt>
</div><a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line"> </tt>
<a name="Diagram.new_track"></a><div id="Diagram.new_track-def"><a name="L263"></a><tt class="py-lineno">263</tt> <a class="py-toggle" href="#" id="Diagram.new_track-toggle" onclick="return toggle('Diagram.new_track');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#new_track">new_track</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">track_level</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">args</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Diagram.new_track-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Diagram.new_track-expanded"><a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add a new Track to the diagram at a given level.</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line"><tt class="py-docstring">        The track is returned for further user manipulation.</tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line"><tt class="py-docstring">            - track_level   - an integer. The level at which the track will be</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line"><tt class="py-docstring">              drawn (above an arbitrary baseline).</tt> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line"><tt class="py-docstring">        new_track(self, track_level)</tt> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line">        <tt class="py-name">newtrack</tt> <tt class="py-op">=</tt> <tt id="link-83" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Track.Track" class="py-name" href="#" onclick="return doclink('link-83', 'Track', 'link-5');">Track</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">args</tt><tt class="py-op">:</tt> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line">            <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt class="py-name">newtrack</tt><tt class="py-op">,</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">args</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">track_level</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">:</tt>        <tt class="py-comment"># No track at that level</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">[</tt><tt class="py-name">track_level</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">newtrack</tt>   <tt class="py-comment"># so just add it</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt>       <tt class="py-comment"># Already a track there, so shunt all higher tracks up one</tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line">            <tt class="py-name">occupied_levels</tt> <tt class="py-op">=</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-84" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.get_levels" class="py-name" href="#" onclick="return doclink('link-84', 'get_levels', 'link-81');">get_levels</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt>  <tt class="py-comment"># Get list of occupied levels...</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line">            <tt class="py-name">occupied_levels</tt><tt class="py-op">.</tt><tt id="link-85" class="py-name"><a title="Bio.Pathway.Reaction.reverse
Bio.Seq.MutableSeq.reverse" class="py-name" href="#" onclick="return doclink('link-85', 'reverse', 'link-82');">reverse</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt>           <tt class="py-comment"># ...reverse (highest first)...</tt> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">val</tt> <tt class="py-keyword">in</tt> <tt class="py-name">occupied_levels</tt><tt class="py-op">:</tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">val</tt> <tt class="py-op">&gt;=</tt> <tt class="py-name">track_level</tt><tt class="py-op">:</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">                    <tt class="py-comment"># Track value &gt;= that to be added, increment by 1</tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">[</tt><tt class="py-name">val</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">[</tt><tt class="py-name">val</tt><tt class="py-op">]</tt> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">[</tt><tt class="py-name">track_level</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">newtrack</tt>  <tt class="py-comment"># And put the new track in</tt> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">[</tt><tt class="py-name">track_level</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">track_level</tt> <tt class="py-op">=</tt> <tt class="py-name">track_level</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">newtrack</tt> </tt>
</div><a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line"> </tt>
<a name="Diagram.del_track"></a><div id="Diagram.del_track-def"><a name="L290"></a><tt class="py-lineno">290</tt> <a class="py-toggle" href="#" id="Diagram.del_track-toggle" onclick="return toggle('Diagram.del_track');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#del_track">del_track</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">track_level</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Diagram.del_track-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Diagram.del_track-expanded"><a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line">        <tt class="py-docstring">"""Remove the track to be drawn at a particular level on the diagram.</tt> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line"><tt class="py-docstring">            - track_level   - an integer. The level of the track on the diagram</tt> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line"><tt class="py-docstring">              to delete.</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line"><tt class="py-docstring">        del_track(self, track_level)</tt> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">[</tt><tt class="py-name">track_level</tt><tt class="py-op">]</tt> </tt>
</div><a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line"> </tt>
<a name="Diagram.get_tracks"></a><div id="Diagram.get_tracks-def"><a name="L301"></a><tt class="py-lineno">301</tt> <a class="py-toggle" href="#" id="Diagram.get_tracks-toggle" onclick="return toggle('Diagram.get_tracks');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#get_tracks">get_tracks</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Diagram.get_tracks-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Diagram.get_tracks-expanded"><a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a list of the tracks contained in the diagram."""</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">.</tt><tt id="link-86" class="py-name"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-86', 'values', 'link-24');">values</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line"> </tt>
<a name="Diagram.move_track"></a><div id="Diagram.move_track-def"><a name="L305"></a><tt class="py-lineno">305</tt> <a class="py-toggle" href="#" id="Diagram.move_track-toggle" onclick="return toggle('Diagram.move_track');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#move_track">move_track</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">from_level</tt><tt class="py-op">,</tt> <tt class="py-param">to_level</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Diagram.move_track-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Diagram.move_track-expanded"><a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line">        <tt class="py-docstring">"""Move a track from one level on the diagram to another.</tt> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line"><tt class="py-docstring">        - from_level   - an integer. The level at which the track to be</tt> </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line"><tt class="py-docstring">          moved is found.</tt> </tt>
<a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line"><tt class="py-docstring">        - to_level     - an integer. The level to move the track to.</tt> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line">        <tt class="py-name">aux</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">[</tt><tt class="py-name">from_level</tt><tt class="py-op">]</tt> </tt>
<a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">[</tt><tt class="py-name">from_level</tt><tt class="py-op">]</tt> </tt>
<a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-87" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.add_track()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#add_track"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.add_track" class="py-name" href="#" onclick="return doclink('link-87', 'add_track', 'link-87');">add_track</a></tt><tt class="py-op">(</tt><tt class="py-name">aux</tt><tt class="py-op">,</tt> <tt class="py-name">to_level</tt><tt class="py-op">)</tt> </tt>
</div><a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line"> </tt>
<a name="Diagram.renumber_tracks"></a><div id="Diagram.renumber_tracks-def"><a name="L319"></a><tt class="py-lineno">319</tt> <a class="py-toggle" href="#" id="Diagram.renumber_tracks-toggle" onclick="return toggle('Diagram.renumber_tracks');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#renumber_tracks">renumber_tracks</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">low</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-param">step</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Diagram.renumber_tracks-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Diagram.renumber_tracks-expanded"><a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line">        <tt class="py-docstring">"""Renumber all tracks consecutively.</tt> </tt>
<a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line"><tt class="py-docstring">        Optionally from a passed lowest number.</tt> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line"><tt class="py-docstring">        - low     - an integer. The track number to start from.</tt> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line"><tt class="py-docstring">        - step    - an integer. The track interval for separation of</tt> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line"><tt class="py-docstring">          tracks.</tt> </tt>
<a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line">        <tt class="py-name">track</tt> <tt class="py-op">=</tt> <tt class="py-name">low</tt>                 <tt class="py-comment"># Start numbering from here</tt> </tt>
<a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line">        <tt class="py-name">levels</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-88" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.get_levels" class="py-name" href="#" onclick="return doclink('link-88', 'get_levels', 'link-81');">get_levels</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line"> </tt>
<a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line">        <tt class="py-name">conversion</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt>             <tt class="py-comment"># Holds new set of levels</tt> </tt>
<a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">level</tt> <tt class="py-keyword">in</tt> <tt class="py-name">levels</tt><tt class="py-op">:</tt>        <tt class="py-comment"># Starting at low...</tt> </tt>
<a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line">            <tt class="py-name">conversion</tt><tt class="py-op">[</tt><tt class="py-name">track</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">[</tt><tt class="py-name">level</tt><tt class="py-op">]</tt>  <tt class="py-comment"># Add old tracks to new set</tt> </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line">            <tt class="py-name">conversion</tt><tt class="py-op">[</tt><tt class="py-name">track</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">track_level</tt> <tt class="py-op">=</tt> <tt class="py-name">track</tt> </tt>
<a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line">            <tt class="py-name">track</tt> <tt class="py-op">+=</tt> <tt class="py-name">step</tt>                           <tt class="py-comment"># step interval</tt> </tt>
<a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt> <tt class="py-op">=</tt> <tt class="py-name">conversion</tt>   <tt class="py-comment"># Replace old set of levels with new set</tt> </tt>
</div><a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line"> </tt>
<a name="Diagram.get_levels"></a><div id="Diagram.get_levels-def"><a name="L341"></a><tt class="py-lineno">341</tt> <a class="py-toggle" href="#" id="Diagram.get_levels-toggle" onclick="return toggle('Diagram.get_levels');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#get_levels">get_levels</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Diagram.get_levels-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Diagram.get_levels-expanded"><a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a sorted list of levels occupied by tracks in the diagram."""</tt> </tt>
<a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">)</tt> </tt>
</div><a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line"> </tt>
<a name="Diagram.get_drawn_levels"></a><div id="Diagram.get_drawn_levels-def"><a name="L345"></a><tt class="py-lineno">345</tt> <a class="py-toggle" href="#" id="Diagram.get_drawn_levels-toggle" onclick="return toggle('Diagram.get_drawn_levels');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#get_drawn_levels">get_drawn_levels</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Diagram.get_drawn_levels-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Diagram.get_drawn_levels-expanded"><a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a sorted list of levels occupied by tracks.</tt> </tt>
<a name="L347"></a><tt class="py-lineno">347</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line"><tt class="py-docstring">        These tracks are not explicitly hidden.</tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt class="py-name">key</tt> <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt> <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">hide</tt><tt class="py-op">)</tt> </tt>
</div><a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line"> </tt>
<a name="Diagram.range"></a><div id="Diagram.range-def"><a name="L352"></a><tt class="py-lineno">352</tt> <a class="py-toggle" href="#" id="Diagram.range-toggle" onclick="return toggle('Diagram.range');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range">range</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Diagram.range-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Diagram.range-expanded"><a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return lowest and highest base numbers from track features.</tt> </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line"><tt class="py-docstring">        Returned type is a tuple.</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line">        <tt class="py-name">lows</tt><tt class="py-op">,</tt> <tt class="py-name">highs</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L358"></a><tt class="py-lineno">358</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">track</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">.</tt><tt id="link-89" class="py-name"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-89', 'values', 'link-24');">values</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># Get ranges for each track</tt> </tt>
<a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line">            <tt class="py-name">low</tt><tt class="py-op">,</tt> <tt class="py-name">high</tt> <tt class="py-op">=</tt> <tt class="py-name">track</tt><tt class="py-op">.</tt><tt id="link-90" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-90', 'range', 'link-90');">range</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line">            <tt class="py-name">lows</tt><tt class="py-op">.</tt><tt id="link-91" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-91', 'append', 'link-91');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">low</tt><tt class="py-op">)</tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line">            <tt class="py-name">highs</tt><tt class="py-op">.</tt><tt id="link-92" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-92', 'append', 'link-91');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">high</tt><tt class="py-op">)</tt> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-93" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.min=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#min"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-93', 'min', 'link-93');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">lows</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-94" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.max=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#max"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-94', 'max', 'link-94');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">highs</tt><tt class="py-op">)</tt>      <tt class="py-comment"># Return extremes from all tracks</tt> </tt>
</div><a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line"> </tt>
<a name="Diagram.__getitem__"></a><div id="Diagram.__getitem__-def"><a name="L364"></a><tt class="py-lineno">364</tt> <a class="py-toggle" href="#" id="Diagram.__getitem__-toggle" onclick="return toggle('Diagram.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">key</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Diagram.__getitem__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Diagram.__getitem__-expanded"><a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the track contained at the level of the passed key."""</tt> </tt>
<a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> </tt>
</div><a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line"> </tt>
<a name="Diagram.__str__"></a><div id="Diagram.__str__-def"><a name="L368"></a><tt class="py-lineno">368</tt> <a class="py-toggle" href="#" id="Diagram.__str__-toggle" onclick="return toggle('Diagram.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Diagram.__str__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Diagram.__str__-expanded"><a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a formatted string describing the diagram."""</tt> </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line">        <tt class="py-name">outstr</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">"\n&lt;%s: %s&gt;"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-95" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-95', 'name', 'link-9');">name</a></tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line">        <tt class="py-name">outstr</tt><tt class="py-op">.</tt><tt id="link-96" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-96', 'append', 'link-91');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"%d tracks"</tt> <tt class="py-op">%</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">level</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-97" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.get_levels" class="py-name" href="#" onclick="return doclink('link-97', 'get_levels', 'link-81');">get_levels</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line">            <tt class="py-name">outstr</tt><tt class="py-op">.</tt><tt id="link-98" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-98', 'append', 'link-91');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"Track %d: %s\n"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">level</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">[</tt><tt class="py-name">level</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line">        <tt class="py-name">outstr</tt> <tt class="py-op">=</tt> <tt class="py-string">'\n'</tt><tt class="py-op">.</tt><tt id="link-99" class="py-name" targets="Method Bio.Seq.MutableSeq.join()=Bio.Seq.MutableSeq-class.html#join,Method Bio.Seq.Seq.join()=Bio.Seq.Seq-class.html#join,Method Bio.Seq.UnknownSeq.join()=Bio.Seq.UnknownSeq-class.html#join"><a title="Bio.Seq.MutableSeq.join
Bio.Seq.Seq.join
Bio.Seq.UnknownSeq.join" class="py-name" href="#" onclick="return doclink('link-99', 'join', 'link-99');">join</a></tt><tt class="py-op">(</tt><tt class="py-name">outstr</tt><tt class="py-op">)</tt> </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">outstr</tt> </tt>
</div></div><a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Tue Jul 16 17:05:44 2019
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
