<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.SeqFeature.FeatureLocation</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.SeqFeature-module.html">Module&nbsp;SeqFeature</a> ::
        Class&nbsp;FeatureLocation
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.SeqFeature.FeatureLocation-class.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== CLASS DESCRIPTION ==================== -->
<h1 class="epydoc">Class FeatureLocation</h1><p class="nomargin-top"><span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation">source&nbsp;code</a></span></p>
<pre class="base-tree">
object --+
         |
        <strong class="uidshort">FeatureLocation</strong>
</pre>

<hr />
<p>Specify the location of a feature along a sequence.</p>
<p>The FeatureLocation is used for simple continuous features, which can
be described as running from a start position to and end position
(optionally with a strand and reference information).  More complex
locations made up from several non-continuous parts (e.g. a coding
sequence made up of several exons) are described using a SeqFeature
with a CompoundLocation.</p>
<p>Note that the start and end location numbering follow Python's scheme,
thus a GenBank entry of 123..150 (one based counting) becomes a location
of [122:150] (zero based counting).</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqFeature <span class="py-keyword">import</span> FeatureLocation
<span class="py-prompt">&gt;&gt;&gt; </span>f = FeatureLocation(122, 150)
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(f)
<span class="py-output">[122:150]</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(f.start)
<span class="py-output">122</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(f.end)
<span class="py-output">150</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(f.strand)
<span class="py-output">None</span></pre>
<p>Note the strand defaults to None. If you are working with nucleotide
sequences you'd want to be explicit if it is the forward strand:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqFeature <span class="py-keyword">import</span> FeatureLocation
<span class="py-prompt">&gt;&gt;&gt; </span>f = FeatureLocation(122, 150, strand=+1)
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(f)
<span class="py-output">[122:150](+)</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(f.strand)
<span class="py-output">1</span></pre>
<p>Note that for a parent sequence of length n, the FeatureLocation
start and end must satisfy the inequality 0 &lt;= start &lt;= end &lt;= n.
This means even for features on the reverse strand of a nucleotide
sequence, we expect the 'start' coordinate to be less than the
'end'.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqFeature <span class="py-keyword">import</span> FeatureLocation
<span class="py-prompt">&gt;&gt;&gt; </span>r = FeatureLocation(122, 150, strand=-1)
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(r)
<span class="py-output">[122:150](-)</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(r.start)
<span class="py-output">122</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(r.end)
<span class="py-output">150</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(r.strand)
<span class="py-output">-1</span></pre>
<p>i.e. Rather than thinking of the 'start' and 'end' biologically in a
strand aware manner, think of them as the 'left most' or 'minimum'
boundary, and the 'right most' or 'maximum' boundary of the region
being described. This is particularly important with compound
locations describing non-continuous regions.</p>
<p>In the example above we have used standard exact positions, but there
are also specialised position objects used to represent fuzzy positions
as well, for example a GenBank location like complement(&lt;123..150)
would use a BeforePosition object for the start.</p>

<!-- ==================== INSTANCE METHODS ==================== -->
<a name="section-InstanceMethods"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Instance Methods</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-InstanceMethods"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqFeature.FeatureLocation-class.html#__init__" class="summary-sig-name">__init__</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">start</span>,
        <span class="summary-sig-arg">end</span>,
        <span class="summary-sig-arg">strand</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">ref</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">ref_db</span>=<span class="summary-sig-default">None</span>)</span><br />
      Initialize the class.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation.__init__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_get_strand"></a><span class="summary-sig-name">_get_strand</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Get function for the strand property (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation._get_strand">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_set_strand"></a><span class="summary-sig-name">_set_strand</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">value</span>)</span><br />
      Set function for the strand property (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation._set_strand">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqFeature.FeatureLocation-class.html#__str__" class="summary-sig-name">__str__</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Return a representation of the FeatureLocation object (with python counting).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation.__str__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqFeature.FeatureLocation-class.html#__repr__" class="summary-sig-name">__repr__</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Represent the FeatureLocation object as a string for debugging.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation.__repr__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqFeature.FeatureLocation-class.html#__add__" class="summary-sig-name">__add__</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">other</span>)</span><br />
      Combine location with another FeatureLocation object, or shift it.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation.__add__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__radd__"></a><span class="summary-sig-name">__radd__</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">other</span>)</span><br />
      Add a feature locationanother FeatureLocation object to the left.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation.__radd__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqFeature.FeatureLocation-class.html#__nonzero__" class="summary-sig-name">__nonzero__</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Return True regardless of the length of the feature.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation.__nonzero__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqFeature.FeatureLocation-class.html#__len__" class="summary-sig-name">__len__</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Return the length of the region described by the FeatureLocation object.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation.__len__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqFeature.FeatureLocation-class.html#__contains__" class="summary-sig-name">__contains__</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">value</span>)</span><br />
      Check if an integer position is within the FeatureLocation object.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation.__contains__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqFeature.FeatureLocation-class.html#__iter__" class="summary-sig-name">__iter__</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Iterate over the parent positions within the FeatureLocation object.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation.__iter__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__eq__"></a><span class="summary-sig-name">__eq__</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">other</span>)</span><br />
      Implement equality by comparing all the location attributes.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation.__eq__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__ne__"></a><span class="summary-sig-name">__ne__</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">other</span>)</span><br />
      Implement the not-equal operand.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation.__ne__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_shift"></a><span class="summary-sig-name">_shift</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">offset</span>)</span><br />
      Return a copy of the FeatureLocation shifted by an offset (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation._shift">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_flip"></a><span class="summary-sig-name">_flip</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">length</span>)</span><br />
      Return a copy of the location after the parent is reversed (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation._flip">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqFeature.FeatureLocation-class.html#extract" class="summary-sig-name">extract</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">parent_sequence</span>)</span><br />
      Extract the sequence from supplied parent sequence using the FeatureLocation object.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation.extract">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__delattr__</code>,
      <code>__format__</code>,
      <code>__getattribute__</code>,
      <code>__hash__</code>,
      <code>__new__</code>,
      <code>__reduce__</code>,
      <code>__reduce_ex__</code>,
      <code>__setattr__</code>,
      <code>__sizeof__</code>,
      <code>__subclasshook__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== PROPERTIES ==================== -->
<a name="section-Properties"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Properties</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Properties"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.SeqFeature.FeatureLocation-class.html#strand" class="summary-name">strand</a><br />
      Strand of the location (+1, -1, 0 or None).
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.SeqFeature.FeatureLocation-class.html#parts" class="summary-name">parts</a><br />
      Read only list of sections (always one, the FeatureLocation object).
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.SeqFeature.FeatureLocation-class.html#start" class="summary-name">start</a><br />
      Start location - left most (minimum) value, regardless of strand.
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.SeqFeature.FeatureLocation-class.html#end" class="summary-name">end</a><br />
      End location - right most (maximum) value, regardless of strand.
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.SeqFeature.FeatureLocation-class.html#nofuzzy_start" class="summary-name">nofuzzy_start</a><br />
      Start position (integer, approximated if fuzzy, read only) (OBSOLETE).
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.SeqFeature.FeatureLocation-class.html#nofuzzy_end" class="summary-name">nofuzzy_end</a><br />
      End position (integer, approximated if fuzzy, read only) (OBSOLETE).
    </td>
  </tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__class__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== METHOD DETAILS ==================== -->
<a name="section-MethodDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Method Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-MethodDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="__init__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__init__</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">start</span>,
        <span class="sig-arg">end</span>,
        <span class="sig-arg">strand</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">ref</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">ref_db</span>=<span class="sig-default">None</span>)</span>
    <br /><em class="fname">(Constructor)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation.__init__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Initialize the class.</p>
<p>start and end arguments specify the values where the feature begins
and ends. These can either by any of the <tt class="rst-docutils literal">*Position</tt> objects that
inherit from AbstractPosition, or can just be integers specifying the
position. In the case of integers, the values are assumed to be
exact and are converted in ExactPosition arguments. This is meant
to make it easy to deal with non-fuzzy ends.</p>
<p>i.e. Short form:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqFeature <span class="py-keyword">import</span> FeatureLocation
<span class="py-prompt">&gt;&gt;&gt; </span>loc = FeatureLocation(5, 10, strand=-1)
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(loc)
<span class="py-output">[5:10](-)</span></pre>
<p>Explicit form:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqFeature <span class="py-keyword">import</span> FeatureLocation, ExactPosition
<span class="py-prompt">&gt;&gt;&gt; </span>loc = FeatureLocation(ExactPosition(5), ExactPosition(10), strand=-1)
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(loc)
<span class="py-output">[5:10](-)</span></pre>
<p>Other fuzzy positions are used similarly,</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqFeature <span class="py-keyword">import</span> FeatureLocation
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqFeature <span class="py-keyword">import</span> BeforePosition, AfterPosition
<span class="py-prompt">&gt;&gt;&gt; </span>loc2 = FeatureLocation(BeforePosition(5), AfterPosition(10), strand=-1)
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(loc2)
<span class="py-output">[&lt;5:&gt;10](-)</span></pre>
<p>For nucleotide features you will also want to specify the strand,
use 1 for the forward (plus) strand, -1 for the reverse (negative)
strand, 0 for stranded but strand unknown (? in GFF3), or None for
when the strand does not apply (dot in GFF3), e.g. features on
proteins.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>loc = FeatureLocation(5, 10, strand=+1)
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(loc)
<span class="py-output">[5:10](+)</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(loc.strand)
<span class="py-output">1</span></pre>
<p>Normally feature locations are given relative to the parent
sequence you are working with, but an explicit accession can
be given with the optional ref and db_ref strings:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>loc = FeatureLocation(105172, 108462, ref=<span class="py-string">&quot;AL391218.9&quot;</span>, strand=1)
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(loc)
<span class="py-output">AL391218.9[105172:108462](+)</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(loc.ref)
<span class="py-output">AL391218.9</span></pre>
  <dl class="fields">
    <dt>Overrides:
        object.__init__
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="__str__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__str__</span>(<span class="sig-arg">self</span>)</span>
    <br /><em class="fname">(Informal representation operator)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation.__str__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Return a representation of the FeatureLocation object (with python counting).</p>
<p>For the simple case this uses the python splicing syntax, [122:150]
(zero based counting) which GenBank would call 123..150 (one based
counting).</p>
  <dl class="fields">
    <dt>Overrides:
        object.__str__
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="__repr__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__repr__</span>(<span class="sig-arg">self</span>)</span>
    <br /><em class="fname">(Representation operator)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation.__repr__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  Represent the FeatureLocation object as a string for debugging.
  <dl class="fields">
    <dt>Overrides:
        object.__repr__
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="__add__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__add__</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">other</span>)</span>
    <br /><em class="fname">(Addition operator)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation.__add__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Combine location with another FeatureLocation object, or shift it.</p>
<p>You can add two feature locations to make a join CompoundLocation:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqFeature <span class="py-keyword">import</span> FeatureLocation
<span class="py-prompt">&gt;&gt;&gt; </span>f1 = FeatureLocation(5, 10)
<span class="py-prompt">&gt;&gt;&gt; </span>f2 = FeatureLocation(20, 30)
<span class="py-prompt">&gt;&gt;&gt; </span>combined = f1 + f2
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(combined)
<span class="py-output">join{[5:10], [20:30]}</span></pre>
<p>This is thus equivalent to:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqFeature <span class="py-keyword">import</span> CompoundLocation
<span class="py-prompt">&gt;&gt;&gt; </span>join = CompoundLocation([f1, f2])
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(join)
<span class="py-output">join{[5:10], [20:30]}</span></pre>
<p>You can also use sum(...) in this way:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>join = sum([f1, f2])
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(join)
<span class="py-output">join{[5:10], [20:30]}</span></pre>
<p>Furthermore, you can combine a FeatureLocation with a CompoundLocation
in this way.</p>
<p>Separately, adding an integer will give a new FeatureLocation with
its start and end offset by that amount. For example:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(f1)
<span class="py-output">[5:10]</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(f1 + 100)
<span class="py-output">[105:110]</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(200 + f1)
<span class="py-output">[205:210]</span></pre>
<p>This can be useful when editing annotation.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="__nonzero__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__nonzero__</span>(<span class="sig-arg">self</span>)</span>
    <br /><em class="fname">(Boolean test operator)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation.__nonzero__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Return True regardless of the length of the feature.</p>
<p>This behaviour is for backwards compatibility, since until the
__len__ method was added, a FeatureLocation always evaluated as True.</p>
<p>Note that in comparison, Seq objects, strings, lists, etc, will all
evaluate to False if they have length zero.</p>
<p>WARNING: The FeatureLocation may in future evaluate to False when its
length is zero (in order to better match normal python behaviour)!</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="__len__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__len__</span>(<span class="sig-arg">self</span>)</span>
    <br /><em class="fname">(Length operator)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation.__len__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Return the length of the region described by the FeatureLocation object.</p>
<p>Note that extra care may be needed for fuzzy locations, e.g.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqFeature <span class="py-keyword">import</span> FeatureLocation
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqFeature <span class="py-keyword">import</span> BeforePosition, AfterPosition
<span class="py-prompt">&gt;&gt;&gt; </span>loc = FeatureLocation(BeforePosition(5), AfterPosition(10))
<span class="py-prompt">&gt;&gt;&gt; </span>len(loc)
<span class="py-output">5</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="__contains__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__contains__</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">value</span>)</span>
    <br /><em class="fname">(In operator)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation.__contains__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Check if an integer position is within the FeatureLocation object.</p>
<p>Note that extra care may be needed for fuzzy locations, e.g.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqFeature <span class="py-keyword">import</span> FeatureLocation
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqFeature <span class="py-keyword">import</span> BeforePosition, AfterPosition
<span class="py-prompt">&gt;&gt;&gt; </span>loc = FeatureLocation(BeforePosition(5), AfterPosition(10))
<span class="py-prompt">&gt;&gt;&gt; </span>len(loc)
<span class="py-output">5</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>[i <span class="py-keyword">for</span> i <span class="py-keyword">in</span> range(15) <span class="py-keyword">if</span> i <span class="py-keyword">in</span> loc]
<span class="py-output">[5, 6, 7, 8, 9]</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="__iter__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__iter__</span>(<span class="sig-arg">self</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation.__iter__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Iterate over the parent positions within the FeatureLocation object.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqFeature <span class="py-keyword">import</span> FeatureLocation
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqFeature <span class="py-keyword">import</span> BeforePosition, AfterPosition
<span class="py-prompt">&gt;&gt;&gt; </span>loc = FeatureLocation(BeforePosition(5), AfterPosition(10))
<span class="py-prompt">&gt;&gt;&gt; </span>len(loc)
<span class="py-output">5</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">for</span> i <span class="py-keyword">in</span> loc: <span class="py-keyword">print</span>(i)
<span class="py-output">5</span>
<span class="py-output">6</span>
<span class="py-output">7</span>
<span class="py-output">8</span>
<span class="py-output">9</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>list(loc)
<span class="py-output">[5, 6, 7, 8, 9]</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>[i <span class="py-keyword">for</span> i <span class="py-keyword">in</span> range(15) <span class="py-keyword">if</span> i <span class="py-keyword">in</span> loc]
<span class="py-output">[5, 6, 7, 8, 9]</span></pre>
<p>Note this is strand aware:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>loc = FeatureLocation(BeforePosition(5), AfterPosition(10), strand = -1)
<span class="py-prompt">&gt;&gt;&gt; </span>list(loc)
<span class="py-output">[9, 8, 7, 6, 5]</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="extract"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">extract</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">parent_sequence</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqFeature-pysrc.html#FeatureLocation.extract">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Extract the sequence from supplied parent sequence using the FeatureLocation object.</p>
<p>The parent_sequence can be a Seq like object or a string, and will
generally return an object of the same type. The exception to this is
a MutableSeq as the parent sequence will return a Seq object.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> generic_protein
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqFeature <span class="py-keyword">import</span> FeatureLocation
<span class="py-prompt">&gt;&gt;&gt; </span>seq = Seq(<span class="py-string">&quot;MKQHKAMIVALIVICITAVVAAL&quot;</span>, generic_protein)
<span class="py-prompt">&gt;&gt;&gt; </span>feature_loc = FeatureLocation(8, 15)
<span class="py-prompt">&gt;&gt;&gt; </span>feature_loc.extract(seq)
<span class="py-output">Seq('VALIVIC', ProteinAlphabet())</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== PROPERTY DETAILS ==================== -->
<a name="section-PropertyDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Property Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-PropertyDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="strand"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">strand</h3>
  Strand of the location (+1, -1, 0 or None).
  <dl class="fields">
    <dt>Get Method:</dt>
    <dd class="value"><span class="summary-sig"><a href="Bio.SeqFeature.FeatureLocation-class.html#_get_strand" class="summary-sig-name" onclick="show_private();">_get_strand</a>(<span class="summary-sig-arg">self</span>)</span>
        - Get function for the strand property (PRIVATE).
    </dd>
    <dt>Set Method:</dt>
    <dd class="value"><span class="summary-sig"><a href="Bio.SeqFeature.FeatureLocation-class.html#_set_strand" class="summary-sig-name" onclick="show_private();">_set_strand</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">value</span>)</span>
        - Set function for the strand property (PRIVATE).
    </dd>
  </dl>
</td></tr></table>
</div>
<a name="parts"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">parts</h3>
  <p>Read only list of sections (always one, the FeatureLocation object).</p>
<p>This is a convenience property allowing you to write code handling
both simple FeatureLocation objects (with one part) and more complex
CompoundLocation objects (with multiple parts) interchangeably.</p>
  <dl class="fields">
    <dt>Get Method:</dt>
    <dd class="value"><span class="summary-sig"><i>unreachable</i>.parts(<span class="summary-sig-arg">self</span>)</span>
        - Read only list of sections (always one, the FeatureLocation object).
    </dd>
  </dl>
</td></tr></table>
</div>
<a name="start"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">start</h3>
  <p>Start location - left most (minimum) value, regardless of strand.</p>
<p>Read only, returns an integer like position object, possibly a fuzzy
position.</p>
  <dl class="fields">
    <dt>Get Method:</dt>
    <dd class="value"><span class="summary-sig"><i>unreachable</i>.start(<span class="summary-sig-arg">self</span>)</span>
        - Start location - left most (minimum) value, regardless of strand.
    </dd>
  </dl>
</td></tr></table>
</div>
<a name="end"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">end</h3>
  <p>End location - right most (maximum) value, regardless of strand.</p>
<p>Read only, returns an integer like position object, possibly a fuzzy
position.</p>
  <dl class="fields">
    <dt>Get Method:</dt>
    <dd class="value"><span class="summary-sig"><i>unreachable</i>.end(<span class="summary-sig-arg">self</span>)</span>
        - End location - right most (maximum) value, regardless of strand.
    </dd>
  </dl>
</td></tr></table>
</div>
<a name="nofuzzy_start"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">nofuzzy_start</h3>
  <p>Start position (integer, approximated if fuzzy, read only) (OBSOLETE).</p>
<p>This is now an alias for int(feature.start), which should be
used in preference -- unless you are trying to support old
versions of Biopython.</p>
  <dl class="fields">
    <dt>Get Method:</dt>
    <dd class="value"><span class="summary-sig"><i>unreachable</i>.nofuzzy_start(<span class="summary-sig-arg">self</span>)</span>
        - Start position (integer, approximated if fuzzy, read only) (OBSOLETE).
    </dd>
  </dl>
</td></tr></table>
</div>
<a name="nofuzzy_end"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">nofuzzy_end</h3>
  <p>End position (integer, approximated if fuzzy, read only) (OBSOLETE).</p>
<p>This is now an alias for int(feature.end), which should be
used in preference -- unless you are trying to support old
versions of Biopython.</p>
  <dl class="fields">
    <dt>Get Method:</dt>
    <dd class="value"><span class="summary-sig"><i>unreachable</i>.nofuzzy_end(<span class="summary-sig-arg">self</span>)</span>
        - End position (integer, approximated if fuzzy, read only) (OBSOLETE).
    </dd>
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Nov  6 13:33:47 2019
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
