<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.SeqIO</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        Package&nbsp;SeqIO
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.SeqIO-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.SeqIO-module.html">Package Bio.SeqIO</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">   1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2006-2018 by Peter Cock.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">   2</tt>  <tt class="py-line"><tt class="py-comment"># This file is part of the Biopython distribution and governed by your</tt> </tt>
<a name="L3"></a><tt class="py-lineno">   3</tt>  <tt class="py-line"><tt class="py-comment"># choice of the "Biopython License Agreement" or the "BSD 3-Clause License".</tt> </tt>
<a name="L4"></a><tt class="py-lineno">   4</tt>  <tt class="py-line"><tt class="py-comment"># Please see the LICENSE file that should have been included as part of this</tt> </tt>
<a name="L5"></a><tt class="py-lineno">   5</tt>  <tt class="py-line"><tt class="py-comment"># package.</tt> </tt>
<a name="L6"></a><tt class="py-lineno">   6</tt>  <tt class="py-line"> </tt>
<a name="L7"></a><tt class="py-lineno">   7</tt>  <tt class="py-line"><tt class="py-comment"># Nice link:</tt> </tt>
<a name="L8"></a><tt class="py-lineno">   8</tt>  <tt class="py-line"><tt class="py-comment"># http://www.ebi.ac.uk/help/formats_frame.html</tt> </tt>
<a name="L9"></a><tt class="py-lineno">   9</tt>  <tt class="py-line"> </tt>
<a name="L10"></a><tt class="py-lineno">  10</tt>  <tt class="py-line"><tt class="py-docstring">r"""Sequence input/output as SeqRecord objects.</tt> </tt>
<a name="L11"></a><tt class="py-lineno">  11</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L12"></a><tt class="py-lineno">  12</tt>  <tt class="py-line"><tt class="py-docstring">Bio.SeqIO is also documented at SeqIO_ and by a whole chapter in our tutorial:</tt> </tt>
<a name="L13"></a><tt class="py-lineno">  13</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L14"></a><tt class="py-lineno">  14</tt>  <tt class="py-line"><tt class="py-docstring">  - `HTML Tutorial`_</tt> </tt>
<a name="L15"></a><tt class="py-lineno">  15</tt>  <tt class="py-line"><tt class="py-docstring">  - `PDF Tutorial`_</tt> </tt>
<a name="L16"></a><tt class="py-lineno">  16</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L17"></a><tt class="py-lineno">  17</tt>  <tt class="py-line"><tt class="py-docstring">.. _SeqIO: http://biopython.org/wiki/SeqIO</tt> </tt>
<a name="L18"></a><tt class="py-lineno">  18</tt>  <tt class="py-line"><tt class="py-docstring">.. _`HTML Tutorial`: http://biopython.org/DIST/docs/tutorial/Tutorial.html</tt> </tt>
<a name="L19"></a><tt class="py-lineno">  19</tt>  <tt class="py-line"><tt class="py-docstring">.. _`PDF Tutorial`: http://biopython.org/DIST/docs/tutorial/Tutorial.pdf</tt> </tt>
<a name="L20"></a><tt class="py-lineno">  20</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L21"></a><tt class="py-lineno">  21</tt>  <tt class="py-line"><tt class="py-docstring">Input</tt> </tt>
<a name="L22"></a><tt class="py-lineno">  22</tt>  <tt class="py-line"><tt class="py-docstring">-----</tt> </tt>
<a name="L23"></a><tt class="py-lineno">  23</tt>  <tt class="py-line"><tt class="py-docstring">The main function is Bio.SeqIO.parse(...) which takes an input file handle</tt> </tt>
<a name="L24"></a><tt class="py-lineno">  24</tt>  <tt class="py-line"><tt class="py-docstring">(or in recent versions of Biopython alternatively a filename as a string),</tt> </tt>
<a name="L25"></a><tt class="py-lineno">  25</tt>  <tt class="py-line"><tt class="py-docstring">and format string.  This returns an iterator giving SeqRecord objects:</tt> </tt>
<a name="L26"></a><tt class="py-lineno">  26</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L27"></a><tt class="py-lineno">  27</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L28"></a><tt class="py-lineno">  28</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; for record in SeqIO.parse("Fasta/f002", "fasta"):</tt> </tt>
<a name="L29"></a><tt class="py-lineno">  29</tt>  <tt class="py-line"><tt class="py-docstring">...     print("%s %i" % (record.id, len(record)))</tt> </tt>
<a name="L30"></a><tt class="py-lineno">  30</tt>  <tt class="py-line"><tt class="py-docstring">gi|1348912|gb|G26680|G26680 633</tt> </tt>
<a name="L31"></a><tt class="py-lineno">  31</tt>  <tt class="py-line"><tt class="py-docstring">gi|1348917|gb|G26685|G26685 413</tt> </tt>
<a name="L32"></a><tt class="py-lineno">  32</tt>  <tt class="py-line"><tt class="py-docstring">gi|1592936|gb|G29385|G29385 471</tt> </tt>
<a name="L33"></a><tt class="py-lineno">  33</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L34"></a><tt class="py-lineno">  34</tt>  <tt class="py-line"><tt class="py-docstring">Note that the parse() function will invoke the relevant parser for the</tt> </tt>
<a name="L35"></a><tt class="py-lineno">  35</tt>  <tt class="py-line"><tt class="py-docstring">format with its default settings.  You may want more control, in which case</tt> </tt>
<a name="L36"></a><tt class="py-lineno">  36</tt>  <tt class="py-line"><tt class="py-docstring">you need to create a format specific sequence iterator directly.</tt> </tt>
<a name="L37"></a><tt class="py-lineno">  37</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L38"></a><tt class="py-lineno">  38</tt>  <tt class="py-line"><tt class="py-docstring">Some of these parsers are wrappers around low-level parsers which build up</tt> </tt>
<a name="L39"></a><tt class="py-lineno">  39</tt>  <tt class="py-line"><tt class="py-docstring">SeqRecord objects for the consistent SeqIO interface. In cases where the</tt> </tt>
<a name="L40"></a><tt class="py-lineno">  40</tt>  <tt class="py-line"><tt class="py-docstring">run-time is critical, such as large FASTA or FASTQ files, calling these</tt> </tt>
<a name="L41"></a><tt class="py-lineno">  41</tt>  <tt class="py-line"><tt class="py-docstring">underlying parsers will be much faster - in this case these generator</tt> </tt>
<a name="L42"></a><tt class="py-lineno">  42</tt>  <tt class="py-line"><tt class="py-docstring">functions which return tuples of strings:</tt> </tt>
<a name="L43"></a><tt class="py-lineno">  43</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L44"></a><tt class="py-lineno">  44</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; from Bio.SeqIO.FastaIO import SimpleFastaParser</tt> </tt>
<a name="L45"></a><tt class="py-lineno">  45</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; from Bio.SeqIO.QualityIO import FastqGeneralIterator</tt> </tt>
<a name="L46"></a><tt class="py-lineno">  46</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L47"></a><tt class="py-lineno">  47</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L48"></a><tt class="py-lineno">  48</tt>  <tt class="py-line"><tt class="py-docstring">Input - Single Records</tt> </tt>
<a name="L49"></a><tt class="py-lineno">  49</tt>  <tt class="py-line"><tt class="py-docstring">----------------------</tt> </tt>
<a name="L50"></a><tt class="py-lineno">  50</tt>  <tt class="py-line"><tt class="py-docstring">If you expect your file to contain one-and-only-one record, then we provide</tt> </tt>
<a name="L51"></a><tt class="py-lineno">  51</tt>  <tt class="py-line"><tt class="py-docstring">the following 'helper' function which will return a single SeqRecord, or</tt> </tt>
<a name="L52"></a><tt class="py-lineno">  52</tt>  <tt class="py-line"><tt class="py-docstring">raise an exception if there are no records or more than one record:</tt> </tt>
<a name="L53"></a><tt class="py-lineno">  53</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L54"></a><tt class="py-lineno">  54</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L55"></a><tt class="py-lineno">  55</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; record = SeqIO.read("Fasta/f001", "fasta")</tt> </tt>
<a name="L56"></a><tt class="py-lineno">  56</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print("%s %i" % (record.id, len(record)))</tt> </tt>
<a name="L57"></a><tt class="py-lineno">  57</tt>  <tt class="py-line"><tt class="py-docstring">gi|3318709|pdb|1A91| 79</tt> </tt>
<a name="L58"></a><tt class="py-lineno">  58</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L59"></a><tt class="py-lineno">  59</tt>  <tt class="py-line"><tt class="py-docstring">This style is useful when you expect a single record only (and would</tt> </tt>
<a name="L60"></a><tt class="py-lineno">  60</tt>  <tt class="py-line"><tt class="py-docstring">consider multiple records an error).  For example, when dealing with GenBank</tt> </tt>
<a name="L61"></a><tt class="py-lineno">  61</tt>  <tt class="py-line"><tt class="py-docstring">files for bacterial genomes or chromosomes, there is normally only a single</tt> </tt>
<a name="L62"></a><tt class="py-lineno">  62</tt>  <tt class="py-line"><tt class="py-docstring">record.  Alternatively, use this with a handle when downloading a single</tt> </tt>
<a name="L63"></a><tt class="py-lineno">  63</tt>  <tt class="py-line"><tt class="py-docstring">record from the internet.</tt> </tt>
<a name="L64"></a><tt class="py-lineno">  64</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L65"></a><tt class="py-lineno">  65</tt>  <tt class="py-line"><tt class="py-docstring">However, if you just want the first record from a file containing multiple</tt> </tt>
<a name="L66"></a><tt class="py-lineno">  66</tt>  <tt class="py-line"><tt class="py-docstring">record, use the next() function on the iterator (or under Python 2, the</tt> </tt>
<a name="L67"></a><tt class="py-lineno">  67</tt>  <tt class="py-line"><tt class="py-docstring">iterator's next() method):</tt> </tt>
<a name="L68"></a><tt class="py-lineno">  68</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L69"></a><tt class="py-lineno">  69</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L70"></a><tt class="py-lineno">  70</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; record = next(SeqIO.parse("Fasta/f002", "fasta"))</tt> </tt>
<a name="L71"></a><tt class="py-lineno">  71</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print("%s %i" % (record.id, len(record)))</tt> </tt>
<a name="L72"></a><tt class="py-lineno">  72</tt>  <tt class="py-line"><tt class="py-docstring">gi|1348912|gb|G26680|G26680 633</tt> </tt>
<a name="L73"></a><tt class="py-lineno">  73</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L74"></a><tt class="py-lineno">  74</tt>  <tt class="py-line"><tt class="py-docstring">The above code will work as long as the file contains at least one record.</tt> </tt>
<a name="L75"></a><tt class="py-lineno">  75</tt>  <tt class="py-line"><tt class="py-docstring">Note that if there is more than one record, the remaining records will be</tt> </tt>
<a name="L76"></a><tt class="py-lineno">  76</tt>  <tt class="py-line"><tt class="py-docstring">silently ignored.</tt> </tt>
<a name="L77"></a><tt class="py-lineno">  77</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L78"></a><tt class="py-lineno">  78</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L79"></a><tt class="py-lineno">  79</tt>  <tt class="py-line"><tt class="py-docstring">Input - Multiple Records</tt> </tt>
<a name="L80"></a><tt class="py-lineno">  80</tt>  <tt class="py-line"><tt class="py-docstring">------------------------</tt> </tt>
<a name="L81"></a><tt class="py-lineno">  81</tt>  <tt class="py-line"><tt class="py-docstring">For non-interlaced files (e.g. Fasta, GenBank, EMBL) with multiple records</tt> </tt>
<a name="L82"></a><tt class="py-lineno">  82</tt>  <tt class="py-line"><tt class="py-docstring">using a sequence iterator can save you a lot of memory (RAM).  There is</tt> </tt>
<a name="L83"></a><tt class="py-lineno">  83</tt>  <tt class="py-line"><tt class="py-docstring">less benefit for interlaced file formats (e.g. most multiple alignment file</tt> </tt>
<a name="L84"></a><tt class="py-lineno">  84</tt>  <tt class="py-line"><tt class="py-docstring">formats).  However, an iterator only lets you access the records one by one.</tt> </tt>
<a name="L85"></a><tt class="py-lineno">  85</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L86"></a><tt class="py-lineno">  86</tt>  <tt class="py-line"><tt class="py-docstring">If you want random access to the records by number, turn this into a list:</tt> </tt>
<a name="L87"></a><tt class="py-lineno">  87</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L88"></a><tt class="py-lineno">  88</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L89"></a><tt class="py-lineno">  89</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; records = list(SeqIO.parse("Fasta/f002", "fasta"))</tt> </tt>
<a name="L90"></a><tt class="py-lineno">  90</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; len(records)</tt> </tt>
<a name="L91"></a><tt class="py-lineno">  91</tt>  <tt class="py-line"><tt class="py-docstring">3</tt> </tt>
<a name="L92"></a><tt class="py-lineno">  92</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print(records[1].id)</tt> </tt>
<a name="L93"></a><tt class="py-lineno">  93</tt>  <tt class="py-line"><tt class="py-docstring">gi|1348917|gb|G26685|G26685</tt> </tt>
<a name="L94"></a><tt class="py-lineno">  94</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L95"></a><tt class="py-lineno">  95</tt>  <tt class="py-line"><tt class="py-docstring">If you want random access to the records by a key such as the record id,</tt> </tt>
<a name="L96"></a><tt class="py-lineno">  96</tt>  <tt class="py-line"><tt class="py-docstring">turn the iterator into a dictionary:</tt> </tt>
<a name="L97"></a><tt class="py-lineno">  97</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L98"></a><tt class="py-lineno">  98</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L99"></a><tt class="py-lineno">  99</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; record_dict = SeqIO.to_dict(SeqIO.parse("Fasta/f002", "fasta"))</tt> </tt>
<a name="L100"></a><tt class="py-lineno"> 100</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; len(record_dict)</tt> </tt>
<a name="L101"></a><tt class="py-lineno"> 101</tt>  <tt class="py-line"><tt class="py-docstring">3</tt> </tt>
<a name="L102"></a><tt class="py-lineno"> 102</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print(len(record_dict["gi|1348917|gb|G26685|G26685"]))</tt> </tt>
<a name="L103"></a><tt class="py-lineno"> 103</tt>  <tt class="py-line"><tt class="py-docstring">413</tt> </tt>
<a name="L104"></a><tt class="py-lineno"> 104</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L105"></a><tt class="py-lineno"> 105</tt>  <tt class="py-line"><tt class="py-docstring">However, using list() or the to_dict() function will load all the records</tt> </tt>
<a name="L106"></a><tt class="py-lineno"> 106</tt>  <tt class="py-line"><tt class="py-docstring">into memory at once, and therefore is not possible on very large files.</tt> </tt>
<a name="L107"></a><tt class="py-lineno"> 107</tt>  <tt class="py-line"><tt class="py-docstring">Instead, for *some* file formats Bio.SeqIO provides an indexing approach</tt> </tt>
<a name="L108"></a><tt class="py-lineno"> 108</tt>  <tt class="py-line"><tt class="py-docstring">providing dictionary like access to any record. For example,</tt> </tt>
<a name="L109"></a><tt class="py-lineno"> 109</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L110"></a><tt class="py-lineno"> 110</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L111"></a><tt class="py-lineno"> 111</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; record_dict = SeqIO.index("Fasta/f002", "fasta")</tt> </tt>
<a name="L112"></a><tt class="py-lineno"> 112</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; len(record_dict)</tt> </tt>
<a name="L113"></a><tt class="py-lineno"> 113</tt>  <tt class="py-line"><tt class="py-docstring">3</tt> </tt>
<a name="L114"></a><tt class="py-lineno"> 114</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print(len(record_dict["gi|1348917|gb|G26685|G26685"]))</tt> </tt>
<a name="L115"></a><tt class="py-lineno"> 115</tt>  <tt class="py-line"><tt class="py-docstring">413</tt> </tt>
<a name="L116"></a><tt class="py-lineno"> 116</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; record_dict.close()</tt> </tt>
<a name="L117"></a><tt class="py-lineno"> 117</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L118"></a><tt class="py-lineno"> 118</tt>  <tt class="py-line"><tt class="py-docstring">Many but not all of the supported input file formats can be indexed like</tt> </tt>
<a name="L119"></a><tt class="py-lineno"> 119</tt>  <tt class="py-line"><tt class="py-docstring">this. For example "fasta", "fastq", "qual" and even the binary format "sff"</tt> </tt>
<a name="L120"></a><tt class="py-lineno"> 120</tt>  <tt class="py-line"><tt class="py-docstring">work, but alignment formats like "phylip", "clustalw" and "nexus" will not.</tt> </tt>
<a name="L121"></a><tt class="py-lineno"> 121</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L122"></a><tt class="py-lineno"> 122</tt>  <tt class="py-line"><tt class="py-docstring">In most cases you can also use SeqIO.index to get the record from the file</tt> </tt>
<a name="L123"></a><tt class="py-lineno"> 123</tt>  <tt class="py-line"><tt class="py-docstring">as a raw string (not a SeqRecord). This can be useful for example to extract</tt> </tt>
<a name="L124"></a><tt class="py-lineno"> 124</tt>  <tt class="py-line"><tt class="py-docstring">a sub-set of records from a file where SeqIO cannot output the file format</tt> </tt>
<a name="L125"></a><tt class="py-lineno"> 125</tt>  <tt class="py-line"><tt class="py-docstring">(e.g. the plain text SwissProt format, "swiss") or where it is important to</tt> </tt>
<a name="L126"></a><tt class="py-lineno"> 126</tt>  <tt class="py-line"><tt class="py-docstring">keep the output 100% identical to the input). For example,</tt> </tt>
<a name="L127"></a><tt class="py-lineno"> 127</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L128"></a><tt class="py-lineno"> 128</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L129"></a><tt class="py-lineno"> 129</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; record_dict = SeqIO.index("Fasta/f002", "fasta")</tt> </tt>
<a name="L130"></a><tt class="py-lineno"> 130</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; len(record_dict)</tt> </tt>
<a name="L131"></a><tt class="py-lineno"> 131</tt>  <tt class="py-line"><tt class="py-docstring">3</tt> </tt>
<a name="L132"></a><tt class="py-lineno"> 132</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print(record_dict.get_raw("gi|1348917|gb|G26685|G26685").decode())</tt> </tt>
<a name="L133"></a><tt class="py-lineno"> 133</tt>  <tt class="py-line"><tt class="py-docstring">&gt;gi|1348917|gb|G26685|G26685 human STS STS_D11734.</tt> </tt>
<a name="L134"></a><tt class="py-lineno"> 134</tt>  <tt class="py-line"><tt class="py-docstring">CGGAGCCAGCGAGCATATGCTGCATGAGGACCTTTCTATCTTACATTATGGCTGGGAATCTTACTCTTTC</tt> </tt>
<a name="L135"></a><tt class="py-lineno"> 135</tt>  <tt class="py-line"><tt class="py-docstring">ATCTGATACCTTGTTCAGATTTCAAAATAGTTGTAGCCTTATCCTGGTTTTACAGATGTGAAACTTTCAA</tt> </tt>
<a name="L136"></a><tt class="py-lineno"> 136</tt>  <tt class="py-line"><tt class="py-docstring">GAGATTTACTGACTTTCCTAGAATAGTTTCTCTACTGGAAACCTGATGCTTTTATAAGCCATTGTGATTA</tt> </tt>
<a name="L137"></a><tt class="py-lineno"> 137</tt>  <tt class="py-line"><tt class="py-docstring">GGATGACTGTTACAGGCTTAGCTTTGTGTGAAANCCAGTCACCTTTCTCCTAGGTAATGAGTAGTGCTGT</tt> </tt>
<a name="L138"></a><tt class="py-lineno"> 138</tt>  <tt class="py-line"><tt class="py-docstring">TCATATTACTNTAAGTTCTATAGCATACTTGCNATCCTTTANCCATGCTTATCATANGTACCATTTGAGG</tt> </tt>
<a name="L139"></a><tt class="py-lineno"> 139</tt>  <tt class="py-line"><tt class="py-docstring">AATTGNTTTGCCCTTTTGGGTTTNTTNTTGGTAAANNNTTCCCGGGTGGGGGNGGTNNNGAAA</tt> </tt>
<a name="L140"></a><tt class="py-lineno"> 140</tt>  <tt class="py-line"><tt class="py-docstring">&lt;BLANKLINE&gt;</tt> </tt>
<a name="L141"></a><tt class="py-lineno"> 141</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print(record_dict["gi|1348917|gb|G26685|G26685"].format("fasta"))</tt> </tt>
<a name="L142"></a><tt class="py-lineno"> 142</tt>  <tt class="py-line"><tt class="py-docstring">&gt;gi|1348917|gb|G26685|G26685 human STS STS_D11734.</tt> </tt>
<a name="L143"></a><tt class="py-lineno"> 143</tt>  <tt class="py-line"><tt class="py-docstring">CGGAGCCAGCGAGCATATGCTGCATGAGGACCTTTCTATCTTACATTATGGCTGGGAATC</tt> </tt>
<a name="L144"></a><tt class="py-lineno"> 144</tt>  <tt class="py-line"><tt class="py-docstring">TTACTCTTTCATCTGATACCTTGTTCAGATTTCAAAATAGTTGTAGCCTTATCCTGGTTT</tt> </tt>
<a name="L145"></a><tt class="py-lineno"> 145</tt>  <tt class="py-line"><tt class="py-docstring">TACAGATGTGAAACTTTCAAGAGATTTACTGACTTTCCTAGAATAGTTTCTCTACTGGAA</tt> </tt>
<a name="L146"></a><tt class="py-lineno"> 146</tt>  <tt class="py-line"><tt class="py-docstring">ACCTGATGCTTTTATAAGCCATTGTGATTAGGATGACTGTTACAGGCTTAGCTTTGTGTG</tt> </tt>
<a name="L147"></a><tt class="py-lineno"> 147</tt>  <tt class="py-line"><tt class="py-docstring">AAANCCAGTCACCTTTCTCCTAGGTAATGAGTAGTGCTGTTCATATTACTNTAAGTTCTA</tt> </tt>
<a name="L148"></a><tt class="py-lineno"> 148</tt>  <tt class="py-line"><tt class="py-docstring">TAGCATACTTGCNATCCTTTANCCATGCTTATCATANGTACCATTTGAGGAATTGNTTTG</tt> </tt>
<a name="L149"></a><tt class="py-lineno"> 149</tt>  <tt class="py-line"><tt class="py-docstring">CCCTTTTGGGTTTNTTNTTGGTAAANNNTTCCCGGGTGGGGGNGGTNNNGAAA</tt> </tt>
<a name="L150"></a><tt class="py-lineno"> 150</tt>  <tt class="py-line"><tt class="py-docstring">&lt;BLANKLINE&gt;</tt> </tt>
<a name="L151"></a><tt class="py-lineno"> 151</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; record_dict.close()</tt> </tt>
<a name="L152"></a><tt class="py-lineno"> 152</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L153"></a><tt class="py-lineno"> 153</tt>  <tt class="py-line"><tt class="py-docstring">Here the original file and what Biopython would output differ in the line</tt> </tt>
<a name="L154"></a><tt class="py-lineno"> 154</tt>  <tt class="py-line"><tt class="py-docstring">wrapping. Also note that under Python 3, the get_raw method will return a</tt> </tt>
<a name="L155"></a><tt class="py-lineno"> 155</tt>  <tt class="py-line"><tt class="py-docstring">bytes string, hence the use of decode to turn it into a (unicode) string.</tt> </tt>
<a name="L156"></a><tt class="py-lineno"> 156</tt>  <tt class="py-line"><tt class="py-docstring">This is unnecessary on Python 2.</tt> </tt>
<a name="L157"></a><tt class="py-lineno"> 157</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L158"></a><tt class="py-lineno"> 158</tt>  <tt class="py-line"><tt class="py-docstring">Also note that the get_raw method will preserve the newline endings. This</tt> </tt>
<a name="L159"></a><tt class="py-lineno"> 159</tt>  <tt class="py-line"><tt class="py-docstring">example FASTQ file uses Unix style endings (b"\n" only),</tt> </tt>
<a name="L160"></a><tt class="py-lineno"> 160</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L161"></a><tt class="py-lineno"> 161</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L162"></a><tt class="py-lineno"> 162</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; fastq_dict = SeqIO.index("Quality/example.fastq", "fastq")</tt> </tt>
<a name="L163"></a><tt class="py-lineno"> 163</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; len(fastq_dict)</tt> </tt>
<a name="L164"></a><tt class="py-lineno"> 164</tt>  <tt class="py-line"><tt class="py-docstring">3</tt> </tt>
<a name="L165"></a><tt class="py-lineno"> 165</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; raw = fastq_dict.get_raw("EAS54_6_R1_2_1_540_792")</tt> </tt>
<a name="L166"></a><tt class="py-lineno"> 166</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; raw.count(b"\n")</tt> </tt>
<a name="L167"></a><tt class="py-lineno"> 167</tt>  <tt class="py-line"><tt class="py-docstring">4</tt> </tt>
<a name="L168"></a><tt class="py-lineno"> 168</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; raw.count(b"\r\n")</tt> </tt>
<a name="L169"></a><tt class="py-lineno"> 169</tt>  <tt class="py-line"><tt class="py-docstring">0</tt> </tt>
<a name="L170"></a><tt class="py-lineno"> 170</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; b"\r" in raw</tt> </tt>
<a name="L171"></a><tt class="py-lineno"> 171</tt>  <tt class="py-line"><tt class="py-docstring">False</tt> </tt>
<a name="L172"></a><tt class="py-lineno"> 172</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; len(raw)</tt> </tt>
<a name="L173"></a><tt class="py-lineno"> 173</tt>  <tt class="py-line"><tt class="py-docstring">78</tt> </tt>
<a name="L174"></a><tt class="py-lineno"> 174</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; fastq_dict.close()</tt> </tt>
<a name="L175"></a><tt class="py-lineno"> 175</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L176"></a><tt class="py-lineno"> 176</tt>  <tt class="py-line"><tt class="py-docstring">Here is the same file but using DOS/Windows new lines (b"\r\n" instead),</tt> </tt>
<a name="L177"></a><tt class="py-lineno"> 177</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L178"></a><tt class="py-lineno"> 178</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L179"></a><tt class="py-lineno"> 179</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; fastq_dict = SeqIO.index("Quality/example_dos.fastq", "fastq")</tt> </tt>
<a name="L180"></a><tt class="py-lineno"> 180</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; len(fastq_dict)</tt> </tt>
<a name="L181"></a><tt class="py-lineno"> 181</tt>  <tt class="py-line"><tt class="py-docstring">3</tt> </tt>
<a name="L182"></a><tt class="py-lineno"> 182</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; raw = fastq_dict.get_raw("EAS54_6_R1_2_1_540_792")</tt> </tt>
<a name="L183"></a><tt class="py-lineno"> 183</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; raw.count(b"\n")</tt> </tt>
<a name="L184"></a><tt class="py-lineno"> 184</tt>  <tt class="py-line"><tt class="py-docstring">4</tt> </tt>
<a name="L185"></a><tt class="py-lineno"> 185</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; raw.count(b"\r\n")</tt> </tt>
<a name="L186"></a><tt class="py-lineno"> 186</tt>  <tt class="py-line"><tt class="py-docstring">4</tt> </tt>
<a name="L187"></a><tt class="py-lineno"> 187</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; b"\r\n" in raw</tt> </tt>
<a name="L188"></a><tt class="py-lineno"> 188</tt>  <tt class="py-line"><tt class="py-docstring">True</tt> </tt>
<a name="L189"></a><tt class="py-lineno"> 189</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; len(raw)</tt> </tt>
<a name="L190"></a><tt class="py-lineno"> 190</tt>  <tt class="py-line"><tt class="py-docstring">82</tt> </tt>
<a name="L191"></a><tt class="py-lineno"> 191</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; fastq_dict.close()</tt> </tt>
<a name="L192"></a><tt class="py-lineno"> 192</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L193"></a><tt class="py-lineno"> 193</tt>  <tt class="py-line"><tt class="py-docstring">Because this uses two bytes for each new line, the file is longer than</tt> </tt>
<a name="L194"></a><tt class="py-lineno"> 194</tt>  <tt class="py-line"><tt class="py-docstring">the Unix equivalent with only one byte.</tt> </tt>
<a name="L195"></a><tt class="py-lineno"> 195</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L196"></a><tt class="py-lineno"> 196</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L197"></a><tt class="py-lineno"> 197</tt>  <tt class="py-line"><tt class="py-docstring">Input - Alignments</tt> </tt>
<a name="L198"></a><tt class="py-lineno"> 198</tt>  <tt class="py-line"><tt class="py-docstring">------------------</tt> </tt>
<a name="L199"></a><tt class="py-lineno"> 199</tt>  <tt class="py-line"><tt class="py-docstring">You can read in alignment files as alignment objects using Bio.AlignIO.</tt> </tt>
<a name="L200"></a><tt class="py-lineno"> 200</tt>  <tt class="py-line"><tt class="py-docstring">Alternatively, reading in an alignment file format via Bio.SeqIO will give</tt> </tt>
<a name="L201"></a><tt class="py-lineno"> 201</tt>  <tt class="py-line"><tt class="py-docstring">you a SeqRecord for each row of each alignment:</tt> </tt>
<a name="L202"></a><tt class="py-lineno"> 202</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L203"></a><tt class="py-lineno"> 203</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L204"></a><tt class="py-lineno"> 204</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; for record in SeqIO.parse("Clustalw/hedgehog.aln", "clustal"):</tt> </tt>
<a name="L205"></a><tt class="py-lineno"> 205</tt>  <tt class="py-line"><tt class="py-docstring">...     print("%s %i" % (record.id, len(record)))</tt> </tt>
<a name="L206"></a><tt class="py-lineno"> 206</tt>  <tt class="py-line"><tt class="py-docstring">gi|167877390|gb|EDS40773.1| 447</tt> </tt>
<a name="L207"></a><tt class="py-lineno"> 207</tt>  <tt class="py-line"><tt class="py-docstring">gi|167234445|ref|NP_001107837. 447</tt> </tt>
<a name="L208"></a><tt class="py-lineno"> 208</tt>  <tt class="py-line"><tt class="py-docstring">gi|74100009|gb|AAZ99217.1| 447</tt> </tt>
<a name="L209"></a><tt class="py-lineno"> 209</tt>  <tt class="py-line"><tt class="py-docstring">gi|13990994|dbj|BAA33523.2| 447</tt> </tt>
<a name="L210"></a><tt class="py-lineno"> 210</tt>  <tt class="py-line"><tt class="py-docstring">gi|56122354|gb|AAV74328.1| 447</tt> </tt>
<a name="L211"></a><tt class="py-lineno"> 211</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L212"></a><tt class="py-lineno"> 212</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L213"></a><tt class="py-lineno"> 213</tt>  <tt class="py-line"><tt class="py-docstring">Output</tt> </tt>
<a name="L214"></a><tt class="py-lineno"> 214</tt>  <tt class="py-line"><tt class="py-docstring">------</tt> </tt>
<a name="L215"></a><tt class="py-lineno"> 215</tt>  <tt class="py-line"><tt class="py-docstring">Use the function Bio.SeqIO.write(...), which takes a complete set of</tt> </tt>
<a name="L216"></a><tt class="py-lineno"> 216</tt>  <tt class="py-line"><tt class="py-docstring">SeqRecord objects (either as a list, or an iterator), an output file handle</tt> </tt>
<a name="L217"></a><tt class="py-lineno"> 217</tt>  <tt class="py-line"><tt class="py-docstring">(or in recent versions of Biopython an output filename as a string) and of</tt> </tt>
<a name="L218"></a><tt class="py-lineno"> 218</tt>  <tt class="py-line"><tt class="py-docstring">course the file format::</tt> </tt>
<a name="L219"></a><tt class="py-lineno"> 219</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L220"></a><tt class="py-lineno"> 220</tt>  <tt class="py-line"><tt class="py-docstring">  from Bio import SeqIO</tt> </tt>
<a name="L221"></a><tt class="py-lineno"> 221</tt>  <tt class="py-line"><tt class="py-docstring">  records = ...</tt> </tt>
<a name="L222"></a><tt class="py-lineno"> 222</tt>  <tt class="py-line"><tt class="py-docstring">  SeqIO.write(records, "example.faa", "fasta")</tt> </tt>
<a name="L223"></a><tt class="py-lineno"> 223</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L224"></a><tt class="py-lineno"> 224</tt>  <tt class="py-line"><tt class="py-docstring">Or, using a handle::</tt> </tt>
<a name="L225"></a><tt class="py-lineno"> 225</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L226"></a><tt class="py-lineno"> 226</tt>  <tt class="py-line"><tt class="py-docstring">    from Bio import SeqIO</tt> </tt>
<a name="L227"></a><tt class="py-lineno"> 227</tt>  <tt class="py-line"><tt class="py-docstring">    records = ...</tt> </tt>
<a name="L228"></a><tt class="py-lineno"> 228</tt>  <tt class="py-line"><tt class="py-docstring">    with open("example.faa", "w") as handle:</tt> </tt>
<a name="L229"></a><tt class="py-lineno"> 229</tt>  <tt class="py-line"><tt class="py-docstring">      SeqIO.write(records, handle, "fasta")</tt> </tt>
<a name="L230"></a><tt class="py-lineno"> 230</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L231"></a><tt class="py-lineno"> 231</tt>  <tt class="py-line"><tt class="py-docstring">You are expected to call this function once (with all your records) and if</tt> </tt>
<a name="L232"></a><tt class="py-lineno"> 232</tt>  <tt class="py-line"><tt class="py-docstring">using a handle, make sure you close it to flush the data to the hard disk.</tt> </tt>
<a name="L233"></a><tt class="py-lineno"> 233</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L234"></a><tt class="py-lineno"> 234</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L235"></a><tt class="py-lineno"> 235</tt>  <tt class="py-line"><tt class="py-docstring">Output - Advanced</tt> </tt>
<a name="L236"></a><tt class="py-lineno"> 236</tt>  <tt class="py-line"><tt class="py-docstring">-----------------</tt> </tt>
<a name="L237"></a><tt class="py-lineno"> 237</tt>  <tt class="py-line"><tt class="py-docstring">The effect of calling write() multiple times on a single file will vary</tt> </tt>
<a name="L238"></a><tt class="py-lineno"> 238</tt>  <tt class="py-line"><tt class="py-docstring">depending on the file format, and is best avoided unless you have a strong</tt> </tt>
<a name="L239"></a><tt class="py-lineno"> 239</tt>  <tt class="py-line"><tt class="py-docstring">reason to do so.</tt> </tt>
<a name="L240"></a><tt class="py-lineno"> 240</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L241"></a><tt class="py-lineno"> 241</tt>  <tt class="py-line"><tt class="py-docstring">If you give a filename, then each time you call write() the existing file</tt> </tt>
<a name="L242"></a><tt class="py-lineno"> 242</tt>  <tt class="py-line"><tt class="py-docstring">will be overwritten. For sequential files formats (e.g. fasta, genbank) each</tt> </tt>
<a name="L243"></a><tt class="py-lineno"> 243</tt>  <tt class="py-line"><tt class="py-docstring">"record block" holds a single sequence.  For these files it would probably</tt> </tt>
<a name="L244"></a><tt class="py-lineno"> 244</tt>  <tt class="py-line"><tt class="py-docstring">be safe to call write() multiple times by re-using the same handle.</tt> </tt>
<a name="L245"></a><tt class="py-lineno"> 245</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L246"></a><tt class="py-lineno"> 246</tt>  <tt class="py-line"><tt class="py-docstring">However, trying this for certain alignment formats (e.g. phylip, clustal,</tt> </tt>
<a name="L247"></a><tt class="py-lineno"> 247</tt>  <tt class="py-line"><tt class="py-docstring">stockholm) would have the effect of concatenating several multiple sequence</tt> </tt>
<a name="L248"></a><tt class="py-lineno"> 248</tt>  <tt class="py-line"><tt class="py-docstring">alignments together.  Such files are created by the PHYLIP suite of programs</tt> </tt>
<a name="L249"></a><tt class="py-lineno"> 249</tt>  <tt class="py-line"><tt class="py-docstring">for bootstrap analysis, but it is clearer to do this via Bio.AlignIO instead.</tt> </tt>
<a name="L250"></a><tt class="py-lineno"> 250</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L251"></a><tt class="py-lineno"> 251</tt>  <tt class="py-line"><tt class="py-docstring">Worse, many fileformats have an explicit header and/or footer structure</tt> </tt>
<a name="L252"></a><tt class="py-lineno"> 252</tt>  <tt class="py-line"><tt class="py-docstring">(e.g. any XMl format, and most binary file formats like SFF). Here making</tt> </tt>
<a name="L253"></a><tt class="py-lineno"> 253</tt>  <tt class="py-line"><tt class="py-docstring">multiple calls to write() will result in an invalid file.</tt> </tt>
<a name="L254"></a><tt class="py-lineno"> 254</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L255"></a><tt class="py-lineno"> 255</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L256"></a><tt class="py-lineno"> 256</tt>  <tt class="py-line"><tt class="py-docstring">Conversion</tt> </tt>
<a name="L257"></a><tt class="py-lineno"> 257</tt>  <tt class="py-line"><tt class="py-docstring">----------</tt> </tt>
<a name="L258"></a><tt class="py-lineno"> 258</tt>  <tt class="py-line"><tt class="py-docstring">The Bio.SeqIO.convert(...) function allows an easy interface for simple</tt> </tt>
<a name="L259"></a><tt class="py-lineno"> 259</tt>  <tt class="py-line"><tt class="py-docstring">file format conversions. Additionally, it may use file format specific</tt> </tt>
<a name="L260"></a><tt class="py-lineno"> 260</tt>  <tt class="py-line"><tt class="py-docstring">optimisations so this should be the fastest way too.</tt> </tt>
<a name="L261"></a><tt class="py-lineno"> 261</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L262"></a><tt class="py-lineno"> 262</tt>  <tt class="py-line"><tt class="py-docstring">In general however, you can combine the Bio.SeqIO.parse(...) function with</tt> </tt>
<a name="L263"></a><tt class="py-lineno"> 263</tt>  <tt class="py-line"><tt class="py-docstring">the Bio.SeqIO.write(...) function for sequence file conversion. Using</tt> </tt>
<a name="L264"></a><tt class="py-lineno"> 264</tt>  <tt class="py-line"><tt class="py-docstring">generator expressions or generator functions provides a memory efficient way</tt> </tt>
<a name="L265"></a><tt class="py-lineno"> 265</tt>  <tt class="py-line"><tt class="py-docstring">to perform filtering or other extra operations as part of the process.</tt> </tt>
<a name="L266"></a><tt class="py-lineno"> 266</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L267"></a><tt class="py-lineno"> 267</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L268"></a><tt class="py-lineno"> 268</tt>  <tt class="py-line"><tt class="py-docstring">File Formats</tt> </tt>
<a name="L269"></a><tt class="py-lineno"> 269</tt>  <tt class="py-line"><tt class="py-docstring">------------</tt> </tt>
<a name="L270"></a><tt class="py-lineno"> 270</tt>  <tt class="py-line"><tt class="py-docstring">When specifying the file format, use lowercase strings.  The same format</tt> </tt>
<a name="L271"></a><tt class="py-lineno"> 271</tt>  <tt class="py-line"><tt class="py-docstring">names are also used in Bio.AlignIO and include the following:</tt> </tt>
<a name="L272"></a><tt class="py-lineno"> 272</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L273"></a><tt class="py-lineno"> 273</tt>  <tt class="py-line"><tt class="py-docstring">    - abi     - Applied Biosystem's sequencing trace format</tt> </tt>
<a name="L274"></a><tt class="py-lineno"> 274</tt>  <tt class="py-line"><tt class="py-docstring">    - abi-trim - Same as "abi" but with quality trimming with Mott's algorithm</tt> </tt>
<a name="L275"></a><tt class="py-lineno"> 275</tt>  <tt class="py-line"><tt class="py-docstring">    - ace     - Reads the contig sequences from an ACE assembly file.</tt> </tt>
<a name="L276"></a><tt class="py-lineno"> 276</tt>  <tt class="py-line"><tt class="py-docstring">    - cif-atom - Uses Bio.PDB.MMCIFParser to determine the (partial) protein</tt> </tt>
<a name="L277"></a><tt class="py-lineno"> 277</tt>  <tt class="py-line"><tt class="py-docstring">      sequence as it appears in the structure based on the atomic coordinates.</tt> </tt>
<a name="L278"></a><tt class="py-lineno"> 278</tt>  <tt class="py-line"><tt class="py-docstring">    - cif-seqres - Reads a macromolecular Crystallographic Information File</tt> </tt>
<a name="L279"></a><tt class="py-lineno"> 279</tt>  <tt class="py-line"><tt class="py-docstring">      (mmCIF) file to determine the complete protein sequence as defined by the</tt> </tt>
<a name="L280"></a><tt class="py-lineno"> 280</tt>  <tt class="py-line"><tt class="py-docstring">      _pdbx_poly_seq_scheme records.</tt> </tt>
<a name="L281"></a><tt class="py-lineno"> 281</tt>  <tt class="py-line"><tt class="py-docstring">    - embl    - The EMBL flat file format. Uses Bio.GenBank internally.</tt> </tt>
<a name="L282"></a><tt class="py-lineno"> 282</tt>  <tt class="py-line"><tt class="py-docstring">    - fasta   - The generic sequence file format where each record starts with</tt> </tt>
<a name="L283"></a><tt class="py-lineno"> 283</tt>  <tt class="py-line"><tt class="py-docstring">      an identifer line starting with a "&gt;" character, followed by</tt> </tt>
<a name="L284"></a><tt class="py-lineno"> 284</tt>  <tt class="py-line"><tt class="py-docstring">      lines of sequence.</tt> </tt>
<a name="L285"></a><tt class="py-lineno"> 285</tt>  <tt class="py-line"><tt class="py-docstring">    - fasta-2line - Stricter interpretation of the FASTA format using exactly</tt> </tt>
<a name="L286"></a><tt class="py-lineno"> 286</tt>  <tt class="py-line"><tt class="py-docstring">      two lines per record (no line wrapping).</tt> </tt>
<a name="L287"></a><tt class="py-lineno"> 287</tt>  <tt class="py-line"><tt class="py-docstring">    - fastq   - A "FASTA like" format used by Sanger which also stores PHRED</tt> </tt>
<a name="L288"></a><tt class="py-lineno"> 288</tt>  <tt class="py-line"><tt class="py-docstring">      sequence quality values (with an ASCII offset of 33).</tt> </tt>
<a name="L289"></a><tt class="py-lineno"> 289</tt>  <tt class="py-line"><tt class="py-docstring">    - fastq-sanger - An alias for "fastq" for consistency with BioPerl and EMBOSS</tt> </tt>
<a name="L290"></a><tt class="py-lineno"> 290</tt>  <tt class="py-line"><tt class="py-docstring">    - fastq-solexa - Original Solexa/Illumnia variant of the FASTQ format which</tt> </tt>
<a name="L291"></a><tt class="py-lineno"> 291</tt>  <tt class="py-line"><tt class="py-docstring">      encodes Solexa quality scores (not PHRED quality scores) with an</tt> </tt>
<a name="L292"></a><tt class="py-lineno"> 292</tt>  <tt class="py-line"><tt class="py-docstring">      ASCII offset of 64.</tt> </tt>
<a name="L293"></a><tt class="py-lineno"> 293</tt>  <tt class="py-line"><tt class="py-docstring">    - fastq-illumina - Solexa/Illumina 1.3 to 1.7 variant of the FASTQ format</tt> </tt>
<a name="L294"></a><tt class="py-lineno"> 294</tt>  <tt class="py-line"><tt class="py-docstring">      which encodes PHRED quality scores with an ASCII offset of 64</tt> </tt>
<a name="L295"></a><tt class="py-lineno"> 295</tt>  <tt class="py-line"><tt class="py-docstring">      (not 33). Note as of version 1.8 of the CASAVA pipeline Illumina</tt> </tt>
<a name="L296"></a><tt class="py-lineno"> 296</tt>  <tt class="py-line"><tt class="py-docstring">      will produce FASTQ files using the standard Sanger encoding.</tt> </tt>
<a name="L297"></a><tt class="py-lineno"> 297</tt>  <tt class="py-line"><tt class="py-docstring">    - genbank - The GenBank or GenPept flat file format.</tt> </tt>
<a name="L298"></a><tt class="py-lineno"> 298</tt>  <tt class="py-line"><tt class="py-docstring">    - gb      - An alias for "genbank", for consistency with NCBI Entrez Utilities</tt> </tt>
<a name="L299"></a><tt class="py-lineno"> 299</tt>  <tt class="py-line"><tt class="py-docstring">    - ig      - The IntelliGenetics file format, apparently the same as the</tt> </tt>
<a name="L300"></a><tt class="py-lineno"> 300</tt>  <tt class="py-line"><tt class="py-docstring">      MASE alignment format.</tt> </tt>
<a name="L301"></a><tt class="py-lineno"> 301</tt>  <tt class="py-line"><tt class="py-docstring">    - imgt    - An EMBL like format from IMGT where the feature tables are more</tt> </tt>
<a name="L302"></a><tt class="py-lineno"> 302</tt>  <tt class="py-line"><tt class="py-docstring">      indented to allow for longer feature types.</tt> </tt>
<a name="L303"></a><tt class="py-lineno"> 303</tt>  <tt class="py-line"><tt class="py-docstring">    - pdb-seqres -  Reads a Protein Data Bank (PDB) file to determine the</tt> </tt>
<a name="L304"></a><tt class="py-lineno"> 304</tt>  <tt class="py-line"><tt class="py-docstring">      complete protein sequence as it appears in the header (no dependencies).</tt> </tt>
<a name="L305"></a><tt class="py-lineno"> 305</tt>  <tt class="py-line"><tt class="py-docstring">    - pdb-atom - Uses Bio.PDB to determine the (partial) protein sequence as</tt> </tt>
<a name="L306"></a><tt class="py-lineno"> 306</tt>  <tt class="py-line"><tt class="py-docstring">      it appears in the structure based on the atom coordinate section of the</tt> </tt>
<a name="L307"></a><tt class="py-lineno"> 307</tt>  <tt class="py-line"><tt class="py-docstring">      file (requires NumPy for Bio.PDB).</tt> </tt>
<a name="L308"></a><tt class="py-lineno"> 308</tt>  <tt class="py-line"><tt class="py-docstring">    - phd     - Output from PHRED, used by PHRAP and CONSED for input.</tt> </tt>
<a name="L309"></a><tt class="py-lineno"> 309</tt>  <tt class="py-line"><tt class="py-docstring">    - pir     - A "FASTA like" format introduced by the National Biomedical</tt> </tt>
<a name="L310"></a><tt class="py-lineno"> 310</tt>  <tt class="py-line"><tt class="py-docstring">      Research Foundation (NBRF) for the Protein Information Resource</tt> </tt>
<a name="L311"></a><tt class="py-lineno"> 311</tt>  <tt class="py-line"><tt class="py-docstring">      (PIR) database, now part of UniProt.</tt> </tt>
<a name="L312"></a><tt class="py-lineno"> 312</tt>  <tt class="py-line"><tt class="py-docstring">    - seqxml  - SeqXML, simple XML format described in Schmitt et al (2011).</tt> </tt>
<a name="L313"></a><tt class="py-lineno"> 313</tt>  <tt class="py-line"><tt class="py-docstring">    - sff     - Standard Flowgram Format (SFF), typical output from Roche 454.</tt> </tt>
<a name="L314"></a><tt class="py-lineno"> 314</tt>  <tt class="py-line"><tt class="py-docstring">    - sff-trim - Standard Flowgram Format (SFF) with given trimming applied.</tt> </tt>
<a name="L315"></a><tt class="py-lineno"> 315</tt>  <tt class="py-line"><tt class="py-docstring">    - swiss   - Plain text Swiss-Prot aka UniProt format.</tt> </tt>
<a name="L316"></a><tt class="py-lineno"> 316</tt>  <tt class="py-line"><tt class="py-docstring">    - tab     - Simple two column tab separated sequence files, where each</tt> </tt>
<a name="L317"></a><tt class="py-lineno"> 317</tt>  <tt class="py-line"><tt class="py-docstring">      line holds a record's identifier and sequence. For example,</tt> </tt>
<a name="L318"></a><tt class="py-lineno"> 318</tt>  <tt class="py-line"><tt class="py-docstring">      this is used as by Aligent's eArray software when saving</tt> </tt>
<a name="L319"></a><tt class="py-lineno"> 319</tt>  <tt class="py-line"><tt class="py-docstring">      microarray probes in a minimal tab delimited text file.</tt> </tt>
<a name="L320"></a><tt class="py-lineno"> 320</tt>  <tt class="py-line"><tt class="py-docstring">    - qual    - A "FASTA like" format holding PHRED quality values from</tt> </tt>
<a name="L321"></a><tt class="py-lineno"> 321</tt>  <tt class="py-line"><tt class="py-docstring">      sequencing DNA, but no actual sequences (usually provided</tt> </tt>
<a name="L322"></a><tt class="py-lineno"> 322</tt>  <tt class="py-line"><tt class="py-docstring">      in separate FASTA files).</tt> </tt>
<a name="L323"></a><tt class="py-lineno"> 323</tt>  <tt class="py-line"><tt class="py-docstring">    - uniprot-xml - The UniProt XML format (replacement for the SwissProt plain</tt> </tt>
<a name="L324"></a><tt class="py-lineno"> 324</tt>  <tt class="py-line"><tt class="py-docstring">      text format which we call "swiss")</tt> </tt>
<a name="L325"></a><tt class="py-lineno"> 325</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L326"></a><tt class="py-lineno"> 326</tt>  <tt class="py-line"><tt class="py-docstring">Note that while Bio.SeqIO can read all the above file formats, it cannot</tt> </tt>
<a name="L327"></a><tt class="py-lineno"> 327</tt>  <tt class="py-line"><tt class="py-docstring">write to all of them.</tt> </tt>
<a name="L328"></a><tt class="py-lineno"> 328</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L329"></a><tt class="py-lineno"> 329</tt>  <tt class="py-line"><tt class="py-docstring">You can also use any file format supported by Bio.AlignIO, such as "nexus",</tt> </tt>
<a name="L330"></a><tt class="py-lineno"> 330</tt>  <tt class="py-line"><tt class="py-docstring">"phylip" and "stockholm", which gives you access to the individual sequences</tt> </tt>
<a name="L331"></a><tt class="py-lineno"> 331</tt>  <tt class="py-line"><tt class="py-docstring">making up each alignment as SeqRecords.</tt> </tt>
<a name="L332"></a><tt class="py-lineno"> 332</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L333"></a><tt class="py-lineno"> 333</tt>  <tt class="py-line"> </tt>
<a name="L334"></a><tt class="py-lineno"> 334</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">__future__</tt> <tt class="py-keyword">import</tt> <tt class="py-name">print_function</tt> </tt>
<a name="L335"></a><tt class="py-lineno"> 335</tt>  <tt class="py-line"> </tt>
<a name="L336"></a><tt class="py-lineno"> 336</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">sys</tt> </tt>
<a name="L337"></a><tt class="py-lineno"> 337</tt>  <tt class="py-line"> </tt>
<a name="L338"></a><tt class="py-lineno"> 338</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-1', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">basestring</tt> </tt>
<a name="L339"></a><tt class="py-lineno"> 339</tt>  <tt class="py-line"> </tt>
<a name="L340"></a><tt class="py-lineno"> 340</tt>  <tt class="py-line"><tt class="py-comment"># TODO</tt> </tt>
<a name="L341"></a><tt class="py-lineno"> 341</tt>  <tt class="py-line"><tt class="py-comment"># - define policy on reading aligned sequences with gaps in</tt> </tt>
<a name="L342"></a><tt class="py-lineno"> 342</tt>  <tt class="py-line"><tt class="py-comment">#   (e.g. - and . characters) including how the alphabet interacts</tt> </tt>
<a name="L343"></a><tt class="py-lineno"> 343</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L344"></a><tt class="py-lineno"> 344</tt>  <tt class="py-line"><tt class="py-comment"># - How best to handle unique/non unique record.id when writing.</tt> </tt>
<a name="L345"></a><tt class="py-lineno"> 345</tt>  <tt class="py-line"><tt class="py-comment">#   For most file formats reading such files is fine; The stockholm</tt> </tt>
<a name="L346"></a><tt class="py-lineno"> 346</tt>  <tt class="py-line"><tt class="py-comment">#   parser would fail.</tt> </tt>
<a name="L347"></a><tt class="py-lineno"> 347</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L348"></a><tt class="py-lineno"> 348</tt>  <tt class="py-line"><tt class="py-comment"># - MSF multiple alignment format, aka GCG, aka PileUp format (*.msf)</tt> </tt>
<a name="L349"></a><tt class="py-lineno"> 349</tt>  <tt class="py-line"><tt class="py-comment">#   http://www.bioperl.org/wiki/MSF_multiple_alignment_format</tt> </tt>
<a name="L350"></a><tt class="py-lineno"> 350</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L351"></a><tt class="py-lineno"> 351</tt>  <tt class="py-line"><tt class="py-comment"># FAO BioPython Developers</tt> </tt>
<a name="L352"></a><tt class="py-lineno"> 352</tt>  <tt class="py-line"><tt class="py-comment"># ------------------------</tt> </tt>
<a name="L353"></a><tt class="py-lineno"> 353</tt>  <tt class="py-line"><tt class="py-comment"># The way I envision this SeqIO system working as that for any sequence file</tt> </tt>
<a name="L354"></a><tt class="py-lineno"> 354</tt>  <tt class="py-line"><tt class="py-comment"># format we have an iterator that returns SeqRecord objects.</tt> </tt>
<a name="L355"></a><tt class="py-lineno"> 355</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L356"></a><tt class="py-lineno"> 356</tt>  <tt class="py-line"><tt class="py-comment"># This also applies to interlaced file formats (like clustal - although that</tt> </tt>
<a name="L357"></a><tt class="py-lineno"> 357</tt>  <tt class="py-line"><tt class="py-comment"># is now handled via Bio.AlignIO instead) where the file cannot be read record</tt> </tt>
<a name="L358"></a><tt class="py-lineno"> 358</tt>  <tt class="py-line"><tt class="py-comment"># by record.  You should still return an iterator, even if the implementation</tt> </tt>
<a name="L359"></a><tt class="py-lineno"> 359</tt>  <tt class="py-line"><tt class="py-comment"># could just as easily return a list.</tt> </tt>
<a name="L360"></a><tt class="py-lineno"> 360</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L361"></a><tt class="py-lineno"> 361</tt>  <tt class="py-line"><tt class="py-comment"># These file format specific sequence iterators may be implemented as:</tt> </tt>
<a name="L362"></a><tt class="py-lineno"> 362</tt>  <tt class="py-line"><tt class="py-comment">#    - Classes which take a handle for __init__ and provide the __iter__ method</tt> </tt>
<a name="L363"></a><tt class="py-lineno"> 363</tt>  <tt class="py-line"><tt class="py-comment">#    - Functions that take a handle, and return an iterator object</tt> </tt>
<a name="L364"></a><tt class="py-lineno"> 364</tt>  <tt class="py-line"><tt class="py-comment">#    - Generator functions that take a handle, and yield SeqRecord objects</tt> </tt>
<a name="L365"></a><tt class="py-lineno"> 365</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L366"></a><tt class="py-lineno"> 366</tt>  <tt class="py-line"><tt class="py-comment"># It is then trivial to turn this iterator into a list of SeqRecord objects,</tt> </tt>
<a name="L367"></a><tt class="py-lineno"> 367</tt>  <tt class="py-line"><tt class="py-comment"># an in memory dictionary, or a multiple sequence alignment object.</tt> </tt>
<a name="L368"></a><tt class="py-lineno"> 368</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L369"></a><tt class="py-lineno"> 369</tt>  <tt class="py-line"><tt class="py-comment"># For building the dictionary by default the id property of each SeqRecord is</tt> </tt>
<a name="L370"></a><tt class="py-lineno"> 370</tt>  <tt class="py-line"><tt class="py-comment"># used as the key.  You should always populate the id property, and it should</tt> </tt>
<a name="L371"></a><tt class="py-lineno"> 371</tt>  <tt class="py-line"><tt class="py-comment"># be unique in most cases. For some file formats the accession number is a good</tt> </tt>
<a name="L372"></a><tt class="py-lineno"> 372</tt>  <tt class="py-line"><tt class="py-comment"># choice.  If the file itself contains ambiguous identifiers, don't try and</tt> </tt>
<a name="L373"></a><tt class="py-lineno"> 373</tt>  <tt class="py-line"><tt class="py-comment"># dis-ambiguate them - return them as is.</tt> </tt>
<a name="L374"></a><tt class="py-lineno"> 374</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L375"></a><tt class="py-lineno"> 375</tt>  <tt class="py-line"><tt class="py-comment"># When adding a new file format, please use the same lower case format name</tt> </tt>
<a name="L376"></a><tt class="py-lineno"> 376</tt>  <tt class="py-line"><tt class="py-comment"># as BioPerl, or if they have not defined one, try the names used by EMBOSS.</tt> </tt>
<a name="L377"></a><tt class="py-lineno"> 377</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L378"></a><tt class="py-lineno"> 378</tt>  <tt class="py-line"><tt class="py-comment"># See also http://biopython.org/wiki/SeqIO_dev</tt> </tt>
<a name="L379"></a><tt class="py-lineno"> 379</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L380"></a><tt class="py-lineno"> 380</tt>  <tt class="py-line"><tt class="py-comment"># --Peter</tt> </tt>
<a name="L381"></a><tt class="py-lineno"> 381</tt>  <tt class="py-line"> </tt>
<a name="L382"></a><tt class="py-lineno"> 382</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-2" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-2', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-3" class="py-name" targets="Module Bio.File=Bio.File-module.html"><a title="Bio.File" class="py-name" href="#" onclick="return doclink('link-3', 'File', 'link-3');">File</a></tt> <tt class="py-keyword">import</tt> <tt id="link-4" class="py-name" targets="Function Bio.File.as_handle()=Bio.File-module.html#as_handle"><a title="Bio.File.as_handle" class="py-name" href="#" onclick="return doclink('link-4', 'as_handle', 'link-4');">as_handle</a></tt> </tt>
<a name="L383"></a><tt class="py-lineno"> 383</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-5" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-5', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-6" class="py-name" targets="Module Bio.SeqRecord=Bio.SeqRecord-module.html,Class Bio.SeqRecord.SeqRecord=Bio.SeqRecord.SeqRecord-class.html"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-6', 'SeqRecord', 'link-6');">SeqRecord</a></tt> <tt class="py-keyword">import</tt> <tt id="link-7" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-7', 'SeqRecord', 'link-6');">SeqRecord</a></tt> </tt>
<a name="L384"></a><tt class="py-lineno"> 384</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-8" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-8', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-9" class="py-name" targets="Package Bio.Align=Bio.Align-module.html"><a title="Bio.Align" class="py-name" href="#" onclick="return doclink('link-9', 'Align', 'link-9');">Align</a></tt> <tt class="py-keyword">import</tt> <tt id="link-10" class="py-name" targets="Class Bio.Align.MultipleSeqAlignment=Bio.Align.MultipleSeqAlignment-class.html"><a title="Bio.Align.MultipleSeqAlignment" class="py-name" href="#" onclick="return doclink('link-10', 'MultipleSeqAlignment', 'link-10');">MultipleSeqAlignment</a></tt> </tt>
<a name="L385"></a><tt class="py-lineno"> 385</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-11" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-11', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-12" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-12', 'Alphabet', 'link-12');">Alphabet</a></tt> <tt class="py-keyword">import</tt> <tt id="link-13" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-13', 'Alphabet', 'link-12');">Alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-14" class="py-name" targets="Class Bio.Alphabet.AlphabetEncoder=Bio.Alphabet.AlphabetEncoder-class.html"><a title="Bio.Alphabet.AlphabetEncoder" class="py-name" href="#" onclick="return doclink('link-14', 'AlphabetEncoder', 'link-14');">AlphabetEncoder</a></tt><tt class="py-op">,</tt> <tt id="link-15" class="py-name" targets="Function Bio.Alphabet._get_base_alphabet()=Bio.Alphabet-module.html#_get_base_alphabet"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-15', '_get_base_alphabet', 'link-15');">_get_base_alphabet</a></tt> </tt>
<a name="L386"></a><tt class="py-lineno"> 386</tt>  <tt class="py-line"> </tt>
<a name="L387"></a><tt class="py-lineno"> 387</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt> <tt class="py-keyword">import</tt> <tt id="link-16" class="py-name" targets="Module Bio.SeqIO.AbiIO=Bio.SeqIO.AbiIO-module.html"><a title="Bio.SeqIO.AbiIO" class="py-name" href="#" onclick="return doclink('link-16', 'AbiIO', 'link-16');">AbiIO</a></tt> </tt>
<a name="L388"></a><tt class="py-lineno"> 388</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt> <tt class="py-keyword">import</tt> <tt id="link-17" class="py-name" targets="Module Bio.SeqIO.AceIO=Bio.SeqIO.AceIO-module.html"><a title="Bio.SeqIO.AceIO" class="py-name" href="#" onclick="return doclink('link-17', 'AceIO', 'link-17');">AceIO</a></tt> </tt>
<a name="L389"></a><tt class="py-lineno"> 389</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt> <tt class="py-keyword">import</tt> <tt id="link-18" class="py-name" targets="Module Bio.AlignIO.FastaIO=Bio.AlignIO.FastaIO-module.html,Module Bio.SearchIO.FastaIO=Bio.SearchIO.FastaIO-module.html,Module Bio.SeqIO.FastaIO=Bio.SeqIO.FastaIO-module.html"><a title="Bio.AlignIO.FastaIO
Bio.SearchIO.FastaIO
Bio.SeqIO.FastaIO" class="py-name" href="#" onclick="return doclink('link-18', 'FastaIO', 'link-18');">FastaIO</a></tt> </tt>
<a name="L390"></a><tt class="py-lineno"> 390</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt> <tt class="py-keyword">import</tt> <tt id="link-19" class="py-name" targets="Module Bio.SeqIO.IgIO=Bio.SeqIO.IgIO-module.html"><a title="Bio.SeqIO.IgIO" class="py-name" href="#" onclick="return doclink('link-19', 'IgIO', 'link-19');">IgIO</a></tt>  <tt class="py-comment"># IntelliGenetics or MASE format</tt> </tt>
<a name="L391"></a><tt class="py-lineno"> 391</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt> <tt class="py-keyword">import</tt> <tt id="link-20" class="py-name" targets="Module Bio.SeqIO.InsdcIO=Bio.SeqIO.InsdcIO-module.html"><a title="Bio.SeqIO.InsdcIO" class="py-name" href="#" onclick="return doclink('link-20', 'InsdcIO', 'link-20');">InsdcIO</a></tt>  <tt class="py-comment"># EMBL and GenBank</tt> </tt>
<a name="L392"></a><tt class="py-lineno"> 392</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt> <tt class="py-keyword">import</tt> <tt id="link-21" class="py-name" targets="Module Bio.SeqIO.PdbIO=Bio.SeqIO.PdbIO-module.html"><a title="Bio.SeqIO.PdbIO" class="py-name" href="#" onclick="return doclink('link-21', 'PdbIO', 'link-21');">PdbIO</a></tt> </tt>
<a name="L393"></a><tt class="py-lineno"> 393</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt> <tt class="py-keyword">import</tt> <tt id="link-22" class="py-name" targets="Module Bio.SeqIO.PhdIO=Bio.SeqIO.PhdIO-module.html"><a title="Bio.SeqIO.PhdIO" class="py-name" href="#" onclick="return doclink('link-22', 'PhdIO', 'link-22');">PhdIO</a></tt> </tt>
<a name="L394"></a><tt class="py-lineno"> 394</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt> <tt class="py-keyword">import</tt> <tt id="link-23" class="py-name" targets="Module Bio.SeqIO.PirIO=Bio.SeqIO.PirIO-module.html"><a title="Bio.SeqIO.PirIO" class="py-name" href="#" onclick="return doclink('link-23', 'PirIO', 'link-23');">PirIO</a></tt> </tt>
<a name="L395"></a><tt class="py-lineno"> 395</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt> <tt class="py-keyword">import</tt> <tt id="link-24" class="py-name" targets="Module Bio.SeqIO.SeqXmlIO=Bio.SeqIO.SeqXmlIO-module.html"><a title="Bio.SeqIO.SeqXmlIO" class="py-name" href="#" onclick="return doclink('link-24', 'SeqXmlIO', 'link-24');">SeqXmlIO</a></tt> </tt>
<a name="L396"></a><tt class="py-lineno"> 396</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt> <tt class="py-keyword">import</tt> <tt id="link-25" class="py-name" targets="Module Bio.SeqIO.SffIO=Bio.SeqIO.SffIO-module.html"><a title="Bio.SeqIO.SffIO" class="py-name" href="#" onclick="return doclink('link-25', 'SffIO', 'link-25');">SffIO</a></tt> </tt>
<a name="L397"></a><tt class="py-lineno"> 397</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt> <tt class="py-keyword">import</tt> <tt id="link-26" class="py-name" targets="Module Bio.SeqIO.SwissIO=Bio.SeqIO.SwissIO-module.html"><a title="Bio.SeqIO.SwissIO" class="py-name" href="#" onclick="return doclink('link-26', 'SwissIO', 'link-26');">SwissIO</a></tt> </tt>
<a name="L398"></a><tt class="py-lineno"> 398</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt> <tt class="py-keyword">import</tt> <tt id="link-27" class="py-name" targets="Module Bio.SeqIO.TabIO=Bio.SeqIO.TabIO-module.html"><a title="Bio.SeqIO.TabIO" class="py-name" href="#" onclick="return doclink('link-27', 'TabIO', 'link-27');">TabIO</a></tt> </tt>
<a name="L399"></a><tt class="py-lineno"> 399</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt> <tt class="py-keyword">import</tt> <tt id="link-28" class="py-name" targets="Module Bio.SeqIO.QualityIO=Bio.SeqIO.QualityIO-module.html"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-28', 'QualityIO', 'link-28');">QualityIO</a></tt>  <tt class="py-comment"># FastQ and qual files</tt> </tt>
<a name="L400"></a><tt class="py-lineno"> 400</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt> <tt class="py-keyword">import</tt> <tt id="link-29" class="py-name" targets="Module Bio.SeqIO.UniprotIO=Bio.SeqIO.UniprotIO-module.html"><a title="Bio.SeqIO.UniprotIO" class="py-name" href="#" onclick="return doclink('link-29', 'UniprotIO', 'link-29');">UniprotIO</a></tt> </tt>
<a name="L401"></a><tt class="py-lineno"> 401</tt>  <tt class="py-line"> </tt>
<a name="L402"></a><tt class="py-lineno"> 402</tt>  <tt class="py-line"><tt class="py-keyword">if</tt> <tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">version_info</tt> <tt class="py-op">&lt;</tt> <tt class="py-op">(</tt><tt class="py-number">3</tt><tt class="py-op">,</tt> <tt class="py-number">6</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L403"></a><tt class="py-lineno"> 403</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt class="py-name">collections</tt> <tt class="py-keyword">import</tt> <tt class="py-name">OrderedDict</tt> <tt class="py-keyword">as</tt> <tt class="py-name">_dict</tt> </tt>
<a name="L404"></a><tt class="py-lineno"> 404</tt>  <tt class="py-line"><tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L405"></a><tt class="py-lineno"> 405</tt>  <tt class="py-line">    <tt class="py-comment"># Default dict is sorted in Python 3.6 onwards</tt> </tt>
<a name="L406"></a><tt class="py-lineno"> 406</tt>  <tt class="py-line">    <tt class="py-name">_dict</tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt> </tt>
<a name="L407"></a><tt class="py-lineno"> 407</tt>  <tt class="py-line"> </tt>
<a name="L408"></a><tt class="py-lineno"> 408</tt>  <tt class="py-line"><tt class="py-comment"># Convention for format names is "mainname-subtype" in lower case.</tt> </tt>
<a name="L409"></a><tt class="py-lineno"> 409</tt>  <tt class="py-line"><tt class="py-comment"># Please use the same names as BioPerl or EMBOSS where possible.</tt> </tt>
<a name="L410"></a><tt class="py-lineno"> 410</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L411"></a><tt class="py-lineno"> 411</tt>  <tt class="py-line"><tt class="py-comment"># Note that this simple system copes with defining</tt> </tt>
<a name="L412"></a><tt class="py-lineno"> 412</tt>  <tt class="py-line"><tt class="py-comment"># multiple possible iterators for a given format/extension</tt> </tt>
<a name="L413"></a><tt class="py-lineno"> 413</tt>  <tt class="py-line"><tt class="py-comment"># with the -subtype suffix</tt> </tt>
<a name="L414"></a><tt class="py-lineno"> 414</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L415"></a><tt class="py-lineno"> 415</tt>  <tt class="py-line"><tt class="py-comment"># Most alignment file formats will be handled via Bio.AlignIO</tt> </tt>
<a name="L416"></a><tt class="py-lineno"> 416</tt>  <tt class="py-line"> </tt>
<a name="L417"></a><tt class="py-lineno"> 417</tt>  <tt class="py-line"><tt id="link-30" class="py-name" targets="Variable Bio.AlignIO._FormatToIterator=Bio.AlignIO-module.html#_FormatToIterator,Variable Bio.SeqIO._FormatToIterator=Bio.SeqIO-module.html#_FormatToIterator,Variable Bio.phenotype._FormatToIterator=Bio.phenotype-module.html#_FormatToIterator"><a title="Bio.AlignIO._FormatToIterator
Bio.SeqIO._FormatToIterator
Bio.phenotype._FormatToIterator" class="py-name" href="#" onclick="return doclink('link-30', '_FormatToIterator', 'link-30');">_FormatToIterator</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">"fasta"</tt><tt class="py-op">:</tt> <tt id="link-31" class="py-name"><a title="Bio.AlignIO.FastaIO
Bio.SearchIO.FastaIO
Bio.SeqIO.FastaIO" class="py-name" href="#" onclick="return doclink('link-31', 'FastaIO', 'link-18');">FastaIO</a></tt><tt class="py-op">.</tt><tt id="link-32" class="py-name" targets="Function Bio.SeqIO.FastaIO.FastaIterator()=Bio.SeqIO.FastaIO-module.html#FastaIterator"><a title="Bio.SeqIO.FastaIO.FastaIterator" class="py-name" href="#" onclick="return doclink('link-32', 'FastaIterator', 'link-32');">FastaIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L418"></a><tt class="py-lineno"> 418</tt>  <tt class="py-line">                     <tt class="py-string">"fasta-2line"</tt><tt class="py-op">:</tt> <tt id="link-33" class="py-name"><a title="Bio.AlignIO.FastaIO
Bio.SearchIO.FastaIO
Bio.SeqIO.FastaIO" class="py-name" href="#" onclick="return doclink('link-33', 'FastaIO', 'link-18');">FastaIO</a></tt><tt class="py-op">.</tt><tt id="link-34" class="py-name" targets="Function Bio.SeqIO.FastaIO.FastaTwoLineIterator()=Bio.SeqIO.FastaIO-module.html#FastaTwoLineIterator"><a title="Bio.SeqIO.FastaIO.FastaTwoLineIterator" class="py-name" href="#" onclick="return doclink('link-34', 'FastaTwoLineIterator', 'link-34');">FastaTwoLineIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L419"></a><tt class="py-lineno"> 419</tt>  <tt class="py-line">                     <tt class="py-string">"gb"</tt><tt class="py-op">:</tt> <tt id="link-35" class="py-name"><a title="Bio.SeqIO.InsdcIO" class="py-name" href="#" onclick="return doclink('link-35', 'InsdcIO', 'link-20');">InsdcIO</a></tt><tt class="py-op">.</tt><tt id="link-36" class="py-name" targets="Function Bio.SeqIO.InsdcIO.GenBankIterator()=Bio.SeqIO.InsdcIO-module.html#GenBankIterator"><a title="Bio.SeqIO.InsdcIO.GenBankIterator" class="py-name" href="#" onclick="return doclink('link-36', 'GenBankIterator', 'link-36');">GenBankIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L420"></a><tt class="py-lineno"> 420</tt>  <tt class="py-line">                     <tt class="py-string">"genbank"</tt><tt class="py-op">:</tt> <tt id="link-37" class="py-name"><a title="Bio.SeqIO.InsdcIO" class="py-name" href="#" onclick="return doclink('link-37', 'InsdcIO', 'link-20');">InsdcIO</a></tt><tt class="py-op">.</tt><tt id="link-38" class="py-name"><a title="Bio.SeqIO.InsdcIO.GenBankIterator" class="py-name" href="#" onclick="return doclink('link-38', 'GenBankIterator', 'link-36');">GenBankIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L421"></a><tt class="py-lineno"> 421</tt>  <tt class="py-line">                     <tt class="py-string">"genbank-cds"</tt><tt class="py-op">:</tt> <tt id="link-39" class="py-name"><a title="Bio.SeqIO.InsdcIO" class="py-name" href="#" onclick="return doclink('link-39', 'InsdcIO', 'link-20');">InsdcIO</a></tt><tt class="py-op">.</tt><tt id="link-40" class="py-name" targets="Function Bio.SeqIO.InsdcIO.GenBankCdsFeatureIterator()=Bio.SeqIO.InsdcIO-module.html#GenBankCdsFeatureIterator"><a title="Bio.SeqIO.InsdcIO.GenBankCdsFeatureIterator" class="py-name" href="#" onclick="return doclink('link-40', 'GenBankCdsFeatureIterator', 'link-40');">GenBankCdsFeatureIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L422"></a><tt class="py-lineno"> 422</tt>  <tt class="py-line">                     <tt class="py-string">"embl"</tt><tt class="py-op">:</tt> <tt id="link-41" class="py-name"><a title="Bio.SeqIO.InsdcIO" class="py-name" href="#" onclick="return doclink('link-41', 'InsdcIO', 'link-20');">InsdcIO</a></tt><tt class="py-op">.</tt><tt id="link-42" class="py-name" targets="Function Bio.SeqIO.InsdcIO.EmblIterator()=Bio.SeqIO.InsdcIO-module.html#EmblIterator"><a title="Bio.SeqIO.InsdcIO.EmblIterator" class="py-name" href="#" onclick="return doclink('link-42', 'EmblIterator', 'link-42');">EmblIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L423"></a><tt class="py-lineno"> 423</tt>  <tt class="py-line">                     <tt class="py-string">"embl-cds"</tt><tt class="py-op">:</tt> <tt id="link-43" class="py-name"><a title="Bio.SeqIO.InsdcIO" class="py-name" href="#" onclick="return doclink('link-43', 'InsdcIO', 'link-20');">InsdcIO</a></tt><tt class="py-op">.</tt><tt id="link-44" class="py-name" targets="Function Bio.SeqIO.InsdcIO.EmblCdsFeatureIterator()=Bio.SeqIO.InsdcIO-module.html#EmblCdsFeatureIterator"><a title="Bio.SeqIO.InsdcIO.EmblCdsFeatureIterator" class="py-name" href="#" onclick="return doclink('link-44', 'EmblCdsFeatureIterator', 'link-44');">EmblCdsFeatureIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L424"></a><tt class="py-lineno"> 424</tt>  <tt class="py-line">                     <tt class="py-string">"imgt"</tt><tt class="py-op">:</tt> <tt id="link-45" class="py-name"><a title="Bio.SeqIO.InsdcIO" class="py-name" href="#" onclick="return doclink('link-45', 'InsdcIO', 'link-20');">InsdcIO</a></tt><tt class="py-op">.</tt><tt id="link-46" class="py-name" targets="Function Bio.SeqIO.InsdcIO.ImgtIterator()=Bio.SeqIO.InsdcIO-module.html#ImgtIterator"><a title="Bio.SeqIO.InsdcIO.ImgtIterator" class="py-name" href="#" onclick="return doclink('link-46', 'ImgtIterator', 'link-46');">ImgtIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L425"></a><tt class="py-lineno"> 425</tt>  <tt class="py-line">                     <tt class="py-string">"ig"</tt><tt class="py-op">:</tt> <tt id="link-47" class="py-name"><a title="Bio.SeqIO.IgIO" class="py-name" href="#" onclick="return doclink('link-47', 'IgIO', 'link-19');">IgIO</a></tt><tt class="py-op">.</tt><tt id="link-48" class="py-name" targets="Function Bio.SeqIO.IgIO.IgIterator()=Bio.SeqIO.IgIO-module.html#IgIterator"><a title="Bio.SeqIO.IgIO.IgIterator" class="py-name" href="#" onclick="return doclink('link-48', 'IgIterator', 'link-48');">IgIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L426"></a><tt class="py-lineno"> 426</tt>  <tt class="py-line">                     <tt class="py-string">"swiss"</tt><tt class="py-op">:</tt> <tt id="link-49" class="py-name"><a title="Bio.SeqIO.SwissIO" class="py-name" href="#" onclick="return doclink('link-49', 'SwissIO', 'link-26');">SwissIO</a></tt><tt class="py-op">.</tt><tt id="link-50" class="py-name" targets="Function Bio.SeqIO.SwissIO.SwissIterator()=Bio.SeqIO.SwissIO-module.html#SwissIterator"><a title="Bio.SeqIO.SwissIO.SwissIterator" class="py-name" href="#" onclick="return doclink('link-50', 'SwissIterator', 'link-50');">SwissIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L427"></a><tt class="py-lineno"> 427</tt>  <tt class="py-line">                     <tt class="py-string">"pdb-atom"</tt><tt class="py-op">:</tt> <tt id="link-51" class="py-name"><a title="Bio.SeqIO.PdbIO" class="py-name" href="#" onclick="return doclink('link-51', 'PdbIO', 'link-21');">PdbIO</a></tt><tt class="py-op">.</tt><tt id="link-52" class="py-name" targets="Function Bio.SeqIO.PdbIO.PdbAtomIterator()=Bio.SeqIO.PdbIO-module.html#PdbAtomIterator"><a title="Bio.SeqIO.PdbIO.PdbAtomIterator" class="py-name" href="#" onclick="return doclink('link-52', 'PdbAtomIterator', 'link-52');">PdbAtomIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L428"></a><tt class="py-lineno"> 428</tt>  <tt class="py-line">                     <tt class="py-string">"pdb-seqres"</tt><tt class="py-op">:</tt> <tt id="link-53" class="py-name"><a title="Bio.SeqIO.PdbIO" class="py-name" href="#" onclick="return doclink('link-53', 'PdbIO', 'link-21');">PdbIO</a></tt><tt class="py-op">.</tt><tt id="link-54" class="py-name" targets="Function Bio.SeqIO.PdbIO.PdbSeqresIterator()=Bio.SeqIO.PdbIO-module.html#PdbSeqresIterator"><a title="Bio.SeqIO.PdbIO.PdbSeqresIterator" class="py-name" href="#" onclick="return doclink('link-54', 'PdbSeqresIterator', 'link-54');">PdbSeqresIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L429"></a><tt class="py-lineno"> 429</tt>  <tt class="py-line">                     <tt class="py-string">"cif-seqres"</tt><tt class="py-op">:</tt> <tt id="link-55" class="py-name"><a title="Bio.SeqIO.PdbIO" class="py-name" href="#" onclick="return doclink('link-55', 'PdbIO', 'link-21');">PdbIO</a></tt><tt class="py-op">.</tt><tt id="link-56" class="py-name" targets="Function Bio.SeqIO.PdbIO.CifSeqresIterator()=Bio.SeqIO.PdbIO-module.html#CifSeqresIterator"><a title="Bio.SeqIO.PdbIO.CifSeqresIterator" class="py-name" href="#" onclick="return doclink('link-56', 'CifSeqresIterator', 'link-56');">CifSeqresIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L430"></a><tt class="py-lineno"> 430</tt>  <tt class="py-line">                     <tt class="py-string">"cif-atom"</tt><tt class="py-op">:</tt> <tt id="link-57" class="py-name"><a title="Bio.SeqIO.PdbIO" class="py-name" href="#" onclick="return doclink('link-57', 'PdbIO', 'link-21');">PdbIO</a></tt><tt class="py-op">.</tt><tt id="link-58" class="py-name" targets="Function Bio.SeqIO.PdbIO.CifAtomIterator()=Bio.SeqIO.PdbIO-module.html#CifAtomIterator"><a title="Bio.SeqIO.PdbIO.CifAtomIterator" class="py-name" href="#" onclick="return doclink('link-58', 'CifAtomIterator', 'link-58');">CifAtomIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L431"></a><tt class="py-lineno"> 431</tt>  <tt class="py-line">                     <tt class="py-string">"phd"</tt><tt class="py-op">:</tt> <tt id="link-59" class="py-name"><a title="Bio.SeqIO.PhdIO" class="py-name" href="#" onclick="return doclink('link-59', 'PhdIO', 'link-22');">PhdIO</a></tt><tt class="py-op">.</tt><tt id="link-60" class="py-name" targets="Function Bio.SeqIO.PhdIO.PhdIterator()=Bio.SeqIO.PhdIO-module.html#PhdIterator"><a title="Bio.SeqIO.PhdIO.PhdIterator" class="py-name" href="#" onclick="return doclink('link-60', 'PhdIterator', 'link-60');">PhdIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L432"></a><tt class="py-lineno"> 432</tt>  <tt class="py-line">                     <tt class="py-string">"ace"</tt><tt class="py-op">:</tt> <tt id="link-61" class="py-name"><a title="Bio.SeqIO.AceIO" class="py-name" href="#" onclick="return doclink('link-61', 'AceIO', 'link-17');">AceIO</a></tt><tt class="py-op">.</tt><tt id="link-62" class="py-name" targets="Function Bio.SeqIO.AceIO.AceIterator()=Bio.SeqIO.AceIO-module.html#AceIterator"><a title="Bio.SeqIO.AceIO.AceIterator" class="py-name" href="#" onclick="return doclink('link-62', 'AceIterator', 'link-62');">AceIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L433"></a><tt class="py-lineno"> 433</tt>  <tt class="py-line">                     <tt class="py-string">"tab"</tt><tt class="py-op">:</tt> <tt id="link-63" class="py-name"><a title="Bio.SeqIO.TabIO" class="py-name" href="#" onclick="return doclink('link-63', 'TabIO', 'link-27');">TabIO</a></tt><tt class="py-op">.</tt><tt id="link-64" class="py-name" targets="Function Bio.SeqIO.TabIO.TabIterator()=Bio.SeqIO.TabIO-module.html#TabIterator"><a title="Bio.SeqIO.TabIO.TabIterator" class="py-name" href="#" onclick="return doclink('link-64', 'TabIterator', 'link-64');">TabIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L434"></a><tt class="py-lineno"> 434</tt>  <tt class="py-line">                     <tt class="py-string">"pir"</tt><tt class="py-op">:</tt> <tt id="link-65" class="py-name"><a title="Bio.SeqIO.PirIO" class="py-name" href="#" onclick="return doclink('link-65', 'PirIO', 'link-23');">PirIO</a></tt><tt class="py-op">.</tt><tt id="link-66" class="py-name" targets="Function Bio.SeqIO.PirIO.PirIterator()=Bio.SeqIO.PirIO-module.html#PirIterator"><a title="Bio.SeqIO.PirIO.PirIterator" class="py-name" href="#" onclick="return doclink('link-66', 'PirIterator', 'link-66');">PirIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L435"></a><tt class="py-lineno"> 435</tt>  <tt class="py-line">                     <tt class="py-string">"fastq"</tt><tt class="py-op">:</tt> <tt id="link-67" class="py-name"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-67', 'QualityIO', 'link-28');">QualityIO</a></tt><tt class="py-op">.</tt><tt id="link-68" class="py-name" targets="Function Bio.SeqIO.QualityIO.FastqPhredIterator()=Bio.SeqIO.QualityIO-module.html#FastqPhredIterator"><a title="Bio.SeqIO.QualityIO.FastqPhredIterator" class="py-name" href="#" onclick="return doclink('link-68', 'FastqPhredIterator', 'link-68');">FastqPhredIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L436"></a><tt class="py-lineno"> 436</tt>  <tt class="py-line">                     <tt class="py-string">"fastq-sanger"</tt><tt class="py-op">:</tt> <tt id="link-69" class="py-name"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-69', 'QualityIO', 'link-28');">QualityIO</a></tt><tt class="py-op">.</tt><tt id="link-70" class="py-name"><a title="Bio.SeqIO.QualityIO.FastqPhredIterator" class="py-name" href="#" onclick="return doclink('link-70', 'FastqPhredIterator', 'link-68');">FastqPhredIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L437"></a><tt class="py-lineno"> 437</tt>  <tt class="py-line">                     <tt class="py-string">"fastq-solexa"</tt><tt class="py-op">:</tt> <tt id="link-71" class="py-name"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-71', 'QualityIO', 'link-28');">QualityIO</a></tt><tt class="py-op">.</tt><tt id="link-72" class="py-name" targets="Function Bio.SeqIO.QualityIO.FastqSolexaIterator()=Bio.SeqIO.QualityIO-module.html#FastqSolexaIterator"><a title="Bio.SeqIO.QualityIO.FastqSolexaIterator" class="py-name" href="#" onclick="return doclink('link-72', 'FastqSolexaIterator', 'link-72');">FastqSolexaIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L438"></a><tt class="py-lineno"> 438</tt>  <tt class="py-line">                     <tt class="py-string">"fastq-illumina"</tt><tt class="py-op">:</tt> <tt id="link-73" class="py-name"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-73', 'QualityIO', 'link-28');">QualityIO</a></tt><tt class="py-op">.</tt><tt id="link-74" class="py-name" targets="Function Bio.SeqIO.QualityIO.FastqIlluminaIterator()=Bio.SeqIO.QualityIO-module.html#FastqIlluminaIterator"><a title="Bio.SeqIO.QualityIO.FastqIlluminaIterator" class="py-name" href="#" onclick="return doclink('link-74', 'FastqIlluminaIterator', 'link-74');">FastqIlluminaIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L439"></a><tt class="py-lineno"> 439</tt>  <tt class="py-line">                     <tt class="py-string">"qual"</tt><tt class="py-op">:</tt> <tt id="link-75" class="py-name"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-75', 'QualityIO', 'link-28');">QualityIO</a></tt><tt class="py-op">.</tt><tt id="link-76" class="py-name" targets="Function Bio.SeqIO.QualityIO.QualPhredIterator()=Bio.SeqIO.QualityIO-module.html#QualPhredIterator"><a title="Bio.SeqIO.QualityIO.QualPhredIterator" class="py-name" href="#" onclick="return doclink('link-76', 'QualPhredIterator', 'link-76');">QualPhredIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L440"></a><tt class="py-lineno"> 440</tt>  <tt class="py-line">                     <tt class="py-string">"sff"</tt><tt class="py-op">:</tt> <tt id="link-77" class="py-name"><a title="Bio.SeqIO.SffIO" class="py-name" href="#" onclick="return doclink('link-77', 'SffIO', 'link-25');">SffIO</a></tt><tt class="py-op">.</tt><tt id="link-78" class="py-name" targets="Function Bio.SeqIO.SffIO.SffIterator()=Bio.SeqIO.SffIO-module.html#SffIterator"><a title="Bio.SeqIO.SffIO.SffIterator" class="py-name" href="#" onclick="return doclink('link-78', 'SffIterator', 'link-78');">SffIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L441"></a><tt class="py-lineno"> 441</tt>  <tt class="py-line">                     <tt class="py-comment"># Not sure about this in the long run:</tt> </tt>
<a name="L442"></a><tt class="py-lineno"> 442</tt>  <tt class="py-line">                     <tt class="py-string">"sff-trim"</tt><tt class="py-op">:</tt> <tt id="link-79" class="py-name"><a title="Bio.SeqIO.SffIO" class="py-name" href="#" onclick="return doclink('link-79', 'SffIO', 'link-25');">SffIO</a></tt><tt class="py-op">.</tt><tt id="link-80" class="py-name" targets="Function Bio.SeqIO.SffIO._SffTrimIterator()=Bio.SeqIO.SffIO-module.html#_SffTrimIterator"><a title="Bio.SeqIO.SffIO._SffTrimIterator" class="py-name" href="#" onclick="return doclink('link-80', '_SffTrimIterator', 'link-80');">_SffTrimIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L443"></a><tt class="py-lineno"> 443</tt>  <tt class="py-line">                     <tt class="py-string">"uniprot-xml"</tt><tt class="py-op">:</tt> <tt id="link-81" class="py-name"><a title="Bio.SeqIO.UniprotIO" class="py-name" href="#" onclick="return doclink('link-81', 'UniprotIO', 'link-29');">UniprotIO</a></tt><tt class="py-op">.</tt><tt id="link-82" class="py-name" targets="Function Bio.SeqIO.UniprotIO.UniprotIterator()=Bio.SeqIO.UniprotIO-module.html#UniprotIterator"><a title="Bio.SeqIO.UniprotIO.UniprotIterator" class="py-name" href="#" onclick="return doclink('link-82', 'UniprotIterator', 'link-82');">UniprotIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L444"></a><tt class="py-lineno"> 444</tt>  <tt class="py-line">                     <tt class="py-string">"seqxml"</tt><tt class="py-op">:</tt> <tt id="link-83" class="py-name"><a title="Bio.SeqIO.SeqXmlIO" class="py-name" href="#" onclick="return doclink('link-83', 'SeqXmlIO', 'link-24');">SeqXmlIO</a></tt><tt class="py-op">.</tt><tt id="link-84" class="py-name" targets="Class Bio.SeqIO.SeqXmlIO.SeqXmlIterator=Bio.SeqIO.SeqXmlIO.SeqXmlIterator-class.html"><a title="Bio.SeqIO.SeqXmlIO.SeqXmlIterator" class="py-name" href="#" onclick="return doclink('link-84', 'SeqXmlIterator', 'link-84');">SeqXmlIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L445"></a><tt class="py-lineno"> 445</tt>  <tt class="py-line">                     <tt class="py-string">"abi"</tt><tt class="py-op">:</tt> <tt id="link-85" class="py-name"><a title="Bio.SeqIO.AbiIO" class="py-name" href="#" onclick="return doclink('link-85', 'AbiIO', 'link-16');">AbiIO</a></tt><tt class="py-op">.</tt><tt id="link-86" class="py-name" targets="Function Bio.SeqIO.AbiIO.AbiIterator()=Bio.SeqIO.AbiIO-module.html#AbiIterator"><a title="Bio.SeqIO.AbiIO.AbiIterator" class="py-name" href="#" onclick="return doclink('link-86', 'AbiIterator', 'link-86');">AbiIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L446"></a><tt class="py-lineno"> 446</tt>  <tt class="py-line">                     <tt class="py-string">"abi-trim"</tt><tt class="py-op">:</tt> <tt id="link-87" class="py-name"><a title="Bio.SeqIO.AbiIO" class="py-name" href="#" onclick="return doclink('link-87', 'AbiIO', 'link-16');">AbiIO</a></tt><tt class="py-op">.</tt><tt id="link-88" class="py-name" targets="Function Bio.SeqIO.AbiIO._AbiTrimIterator()=Bio.SeqIO.AbiIO-module.html#_AbiTrimIterator"><a title="Bio.SeqIO.AbiIO._AbiTrimIterator" class="py-name" href="#" onclick="return doclink('link-88', '_AbiTrimIterator', 'link-88');">_AbiTrimIterator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L447"></a><tt class="py-lineno"> 447</tt>  <tt class="py-line">                     <tt class="py-op">}</tt> </tt>
<a name="L448"></a><tt class="py-lineno"> 448</tt>  <tt class="py-line"> </tt>
<a name="L449"></a><tt class="py-lineno"> 449</tt>  <tt class="py-line"><tt id="link-89" class="py-name" targets="Variable Bio.SeqIO._FormatToString=Bio.SeqIO-module.html#_FormatToString"><a title="Bio.SeqIO._FormatToString" class="py-name" href="#" onclick="return doclink('link-89', '_FormatToString', 'link-89');">_FormatToString</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L450"></a><tt class="py-lineno"> 450</tt>  <tt class="py-line">    <tt class="py-string">"fasta"</tt><tt class="py-op">:</tt> <tt id="link-90" class="py-name"><a title="Bio.AlignIO.FastaIO
Bio.SearchIO.FastaIO
Bio.SeqIO.FastaIO" class="py-name" href="#" onclick="return doclink('link-90', 'FastaIO', 'link-18');">FastaIO</a></tt><tt class="py-op">.</tt><tt id="link-91" class="py-name" targets="Function Bio.SeqIO.FastaIO.as_fasta()=Bio.SeqIO.FastaIO-module.html#as_fasta"><a title="Bio.SeqIO.FastaIO.as_fasta" class="py-name" href="#" onclick="return doclink('link-91', 'as_fasta', 'link-91');">as_fasta</a></tt><tt class="py-op">,</tt> </tt>
<a name="L451"></a><tt class="py-lineno"> 451</tt>  <tt class="py-line">    <tt class="py-string">"fasta-2line"</tt><tt class="py-op">:</tt> <tt id="link-92" class="py-name"><a title="Bio.AlignIO.FastaIO
Bio.SearchIO.FastaIO
Bio.SeqIO.FastaIO" class="py-name" href="#" onclick="return doclink('link-92', 'FastaIO', 'link-18');">FastaIO</a></tt><tt class="py-op">.</tt><tt id="link-93" class="py-name" targets="Function Bio.SeqIO.FastaIO.as_fasta_2line()=Bio.SeqIO.FastaIO-module.html#as_fasta_2line"><a title="Bio.SeqIO.FastaIO.as_fasta_2line" class="py-name" href="#" onclick="return doclink('link-93', 'as_fasta_2line', 'link-93');">as_fasta_2line</a></tt><tt class="py-op">,</tt> </tt>
<a name="L452"></a><tt class="py-lineno"> 452</tt>  <tt class="py-line">    <tt class="py-string">"tab"</tt><tt class="py-op">:</tt> <tt id="link-94" class="py-name"><a title="Bio.SeqIO.TabIO" class="py-name" href="#" onclick="return doclink('link-94', 'TabIO', 'link-27');">TabIO</a></tt><tt class="py-op">.</tt><tt id="link-95" class="py-name" targets="Function Bio.SeqIO.TabIO.as_tab()=Bio.SeqIO.TabIO-module.html#as_tab"><a title="Bio.SeqIO.TabIO.as_tab" class="py-name" href="#" onclick="return doclink('link-95', 'as_tab', 'link-95');">as_tab</a></tt><tt class="py-op">,</tt> </tt>
<a name="L453"></a><tt class="py-lineno"> 453</tt>  <tt class="py-line">    <tt class="py-string">"fastq"</tt><tt class="py-op">:</tt> <tt id="link-96" class="py-name"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-96', 'QualityIO', 'link-28');">QualityIO</a></tt><tt class="py-op">.</tt><tt id="link-97" class="py-name" targets="Function Bio.SeqIO.QualityIO.as_fastq()=Bio.SeqIO.QualityIO-module.html#as_fastq"><a title="Bio.SeqIO.QualityIO.as_fastq" class="py-name" href="#" onclick="return doclink('link-97', 'as_fastq', 'link-97');">as_fastq</a></tt><tt class="py-op">,</tt> </tt>
<a name="L454"></a><tt class="py-lineno"> 454</tt>  <tt class="py-line">    <tt class="py-string">"fastq-sanger"</tt><tt class="py-op">:</tt> <tt id="link-98" class="py-name"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-98', 'QualityIO', 'link-28');">QualityIO</a></tt><tt class="py-op">.</tt><tt id="link-99" class="py-name"><a title="Bio.SeqIO.QualityIO.as_fastq" class="py-name" href="#" onclick="return doclink('link-99', 'as_fastq', 'link-97');">as_fastq</a></tt><tt class="py-op">,</tt> </tt>
<a name="L455"></a><tt class="py-lineno"> 455</tt>  <tt class="py-line">    <tt class="py-string">"fastq-solexa"</tt><tt class="py-op">:</tt> <tt id="link-100" class="py-name"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-100', 'QualityIO', 'link-28');">QualityIO</a></tt><tt class="py-op">.</tt><tt id="link-101" class="py-name" targets="Function Bio.SeqIO.QualityIO.as_fastq_solexa()=Bio.SeqIO.QualityIO-module.html#as_fastq_solexa"><a title="Bio.SeqIO.QualityIO.as_fastq_solexa" class="py-name" href="#" onclick="return doclink('link-101', 'as_fastq_solexa', 'link-101');">as_fastq_solexa</a></tt><tt class="py-op">,</tt> </tt>
<a name="L456"></a><tt class="py-lineno"> 456</tt>  <tt class="py-line">    <tt class="py-string">"fastq-illumina"</tt><tt class="py-op">:</tt> <tt id="link-102" class="py-name"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-102', 'QualityIO', 'link-28');">QualityIO</a></tt><tt class="py-op">.</tt><tt id="link-103" class="py-name" targets="Function Bio.SeqIO.QualityIO.as_fastq_illumina()=Bio.SeqIO.QualityIO-module.html#as_fastq_illumina"><a title="Bio.SeqIO.QualityIO.as_fastq_illumina" class="py-name" href="#" onclick="return doclink('link-103', 'as_fastq_illumina', 'link-103');">as_fastq_illumina</a></tt><tt class="py-op">,</tt> </tt>
<a name="L457"></a><tt class="py-lineno"> 457</tt>  <tt class="py-line">    <tt class="py-string">"qual"</tt><tt class="py-op">:</tt> <tt id="link-104" class="py-name"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-104', 'QualityIO', 'link-28');">QualityIO</a></tt><tt class="py-op">.</tt><tt id="link-105" class="py-name" targets="Function Bio.SeqIO.QualityIO.as_qual()=Bio.SeqIO.QualityIO-module.html#as_qual"><a title="Bio.SeqIO.QualityIO.as_qual" class="py-name" href="#" onclick="return doclink('link-105', 'as_qual', 'link-105');">as_qual</a></tt><tt class="py-op">,</tt> </tt>
<a name="L458"></a><tt class="py-lineno"> 458</tt>  <tt class="py-line"><tt class="py-op">}</tt> </tt>
<a name="L459"></a><tt class="py-lineno"> 459</tt>  <tt class="py-line"> </tt>
<a name="L460"></a><tt class="py-lineno"> 460</tt>  <tt class="py-line"><tt class="py-comment"># This could exclude file formats covered by _FormatToString?</tt> </tt>
<a name="L461"></a><tt class="py-lineno"> 461</tt>  <tt class="py-line"><tt class="py-comment"># Right now used in the unit tests as proxy for all supported outputs...</tt> </tt>
<a name="L462"></a><tt class="py-lineno"> 462</tt>  <tt class="py-line"><tt id="link-106" class="py-name" targets="Variable Bio.AlignIO._FormatToWriter=Bio.AlignIO-module.html#_FormatToWriter,Variable Bio.SeqIO._FormatToWriter=Bio.SeqIO-module.html#_FormatToWriter,Variable Bio.phenotype._FormatToWriter=Bio.phenotype-module.html#_FormatToWriter"><a title="Bio.AlignIO._FormatToWriter
Bio.SeqIO._FormatToWriter
Bio.phenotype._FormatToWriter" class="py-name" href="#" onclick="return doclink('link-106', '_FormatToWriter', 'link-106');">_FormatToWriter</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">"fasta"</tt><tt class="py-op">:</tt> <tt id="link-107" class="py-name"><a title="Bio.AlignIO.FastaIO
Bio.SearchIO.FastaIO
Bio.SeqIO.FastaIO" class="py-name" href="#" onclick="return doclink('link-107', 'FastaIO', 'link-18');">FastaIO</a></tt><tt class="py-op">.</tt><tt id="link-108" class="py-name" targets="Class Bio.SeqIO.FastaIO.FastaWriter=Bio.SeqIO.FastaIO.FastaWriter-class.html"><a title="Bio.SeqIO.FastaIO.FastaWriter" class="py-name" href="#" onclick="return doclink('link-108', 'FastaWriter', 'link-108');">FastaWriter</a></tt><tt class="py-op">,</tt> </tt>
<a name="L463"></a><tt class="py-lineno"> 463</tt>  <tt class="py-line">                   <tt class="py-string">"fasta-2line"</tt><tt class="py-op">:</tt> <tt id="link-109" class="py-name"><a title="Bio.AlignIO.FastaIO
Bio.SearchIO.FastaIO
Bio.SeqIO.FastaIO" class="py-name" href="#" onclick="return doclink('link-109', 'FastaIO', 'link-18');">FastaIO</a></tt><tt class="py-op">.</tt><tt id="link-110" class="py-name" targets="Class Bio.SeqIO.FastaIO.FastaTwoLineWriter=Bio.SeqIO.FastaIO.FastaTwoLineWriter-class.html"><a title="Bio.SeqIO.FastaIO.FastaTwoLineWriter" class="py-name" href="#" onclick="return doclink('link-110', 'FastaTwoLineWriter', 'link-110');">FastaTwoLineWriter</a></tt><tt class="py-op">,</tt> </tt>
<a name="L464"></a><tt class="py-lineno"> 464</tt>  <tt class="py-line">                   <tt class="py-string">"gb"</tt><tt class="py-op">:</tt> <tt id="link-111" class="py-name"><a title="Bio.SeqIO.InsdcIO" class="py-name" href="#" onclick="return doclink('link-111', 'InsdcIO', 'link-20');">InsdcIO</a></tt><tt class="py-op">.</tt><tt id="link-112" class="py-name" targets="Class Bio.SeqIO.InsdcIO.GenBankWriter=Bio.SeqIO.InsdcIO.GenBankWriter-class.html"><a title="Bio.SeqIO.InsdcIO.GenBankWriter" class="py-name" href="#" onclick="return doclink('link-112', 'GenBankWriter', 'link-112');">GenBankWriter</a></tt><tt class="py-op">,</tt> </tt>
<a name="L465"></a><tt class="py-lineno"> 465</tt>  <tt class="py-line">                   <tt class="py-string">"genbank"</tt><tt class="py-op">:</tt> <tt id="link-113" class="py-name"><a title="Bio.SeqIO.InsdcIO" class="py-name" href="#" onclick="return doclink('link-113', 'InsdcIO', 'link-20');">InsdcIO</a></tt><tt class="py-op">.</tt><tt id="link-114" class="py-name"><a title="Bio.SeqIO.InsdcIO.GenBankWriter" class="py-name" href="#" onclick="return doclink('link-114', 'GenBankWriter', 'link-112');">GenBankWriter</a></tt><tt class="py-op">,</tt> </tt>
<a name="L466"></a><tt class="py-lineno"> 466</tt>  <tt class="py-line">                   <tt class="py-string">"embl"</tt><tt class="py-op">:</tt> <tt id="link-115" class="py-name"><a title="Bio.SeqIO.InsdcIO" class="py-name" href="#" onclick="return doclink('link-115', 'InsdcIO', 'link-20');">InsdcIO</a></tt><tt class="py-op">.</tt><tt id="link-116" class="py-name" targets="Class Bio.SeqIO.InsdcIO.EmblWriter=Bio.SeqIO.InsdcIO.EmblWriter-class.html"><a title="Bio.SeqIO.InsdcIO.EmblWriter" class="py-name" href="#" onclick="return doclink('link-116', 'EmblWriter', 'link-116');">EmblWriter</a></tt><tt class="py-op">,</tt> </tt>
<a name="L467"></a><tt class="py-lineno"> 467</tt>  <tt class="py-line">                   <tt class="py-string">"imgt"</tt><tt class="py-op">:</tt> <tt id="link-117" class="py-name"><a title="Bio.SeqIO.InsdcIO" class="py-name" href="#" onclick="return doclink('link-117', 'InsdcIO', 'link-20');">InsdcIO</a></tt><tt class="py-op">.</tt><tt id="link-118" class="py-name" targets="Class Bio.SeqIO.InsdcIO.ImgtWriter=Bio.SeqIO.InsdcIO.ImgtWriter-class.html"><a title="Bio.SeqIO.InsdcIO.ImgtWriter" class="py-name" href="#" onclick="return doclink('link-118', 'ImgtWriter', 'link-118');">ImgtWriter</a></tt><tt class="py-op">,</tt> </tt>
<a name="L468"></a><tt class="py-lineno"> 468</tt>  <tt class="py-line">                   <tt class="py-string">"tab"</tt><tt class="py-op">:</tt> <tt id="link-119" class="py-name"><a title="Bio.SeqIO.TabIO" class="py-name" href="#" onclick="return doclink('link-119', 'TabIO', 'link-27');">TabIO</a></tt><tt class="py-op">.</tt><tt id="link-120" class="py-name" targets="Class Bio.SeqIO.TabIO.TabWriter=Bio.SeqIO.TabIO.TabWriter-class.html"><a title="Bio.SeqIO.TabIO.TabWriter" class="py-name" href="#" onclick="return doclink('link-120', 'TabWriter', 'link-120');">TabWriter</a></tt><tt class="py-op">,</tt> </tt>
<a name="L469"></a><tt class="py-lineno"> 469</tt>  <tt class="py-line">                   <tt class="py-string">"fastq"</tt><tt class="py-op">:</tt> <tt id="link-121" class="py-name"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-121', 'QualityIO', 'link-28');">QualityIO</a></tt><tt class="py-op">.</tt><tt id="link-122" class="py-name" targets="Class Bio.SeqIO.QualityIO.FastqPhredWriter=Bio.SeqIO.QualityIO.FastqPhredWriter-class.html"><a title="Bio.SeqIO.QualityIO.FastqPhredWriter" class="py-name" href="#" onclick="return doclink('link-122', 'FastqPhredWriter', 'link-122');">FastqPhredWriter</a></tt><tt class="py-op">,</tt> </tt>
<a name="L470"></a><tt class="py-lineno"> 470</tt>  <tt class="py-line">                   <tt class="py-string">"fastq-sanger"</tt><tt class="py-op">:</tt> <tt id="link-123" class="py-name"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-123', 'QualityIO', 'link-28');">QualityIO</a></tt><tt class="py-op">.</tt><tt id="link-124" class="py-name"><a title="Bio.SeqIO.QualityIO.FastqPhredWriter" class="py-name" href="#" onclick="return doclink('link-124', 'FastqPhredWriter', 'link-122');">FastqPhredWriter</a></tt><tt class="py-op">,</tt> </tt>
<a name="L471"></a><tt class="py-lineno"> 471</tt>  <tt class="py-line">                   <tt class="py-string">"fastq-solexa"</tt><tt class="py-op">:</tt> <tt id="link-125" class="py-name"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-125', 'QualityIO', 'link-28');">QualityIO</a></tt><tt class="py-op">.</tt><tt id="link-126" class="py-name" targets="Class Bio.SeqIO.QualityIO.FastqSolexaWriter=Bio.SeqIO.QualityIO.FastqSolexaWriter-class.html"><a title="Bio.SeqIO.QualityIO.FastqSolexaWriter" class="py-name" href="#" onclick="return doclink('link-126', 'FastqSolexaWriter', 'link-126');">FastqSolexaWriter</a></tt><tt class="py-op">,</tt> </tt>
<a name="L472"></a><tt class="py-lineno"> 472</tt>  <tt class="py-line">                   <tt class="py-string">"fastq-illumina"</tt><tt class="py-op">:</tt> <tt id="link-127" class="py-name"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-127', 'QualityIO', 'link-28');">QualityIO</a></tt><tt class="py-op">.</tt><tt id="link-128" class="py-name" targets="Class Bio.SeqIO.QualityIO.FastqIlluminaWriter=Bio.SeqIO.QualityIO.FastqIlluminaWriter-class.html"><a title="Bio.SeqIO.QualityIO.FastqIlluminaWriter" class="py-name" href="#" onclick="return doclink('link-128', 'FastqIlluminaWriter', 'link-128');">FastqIlluminaWriter</a></tt><tt class="py-op">,</tt> </tt>
<a name="L473"></a><tt class="py-lineno"> 473</tt>  <tt class="py-line">                   <tt class="py-string">"phd"</tt><tt class="py-op">:</tt> <tt id="link-129" class="py-name"><a title="Bio.SeqIO.PhdIO" class="py-name" href="#" onclick="return doclink('link-129', 'PhdIO', 'link-22');">PhdIO</a></tt><tt class="py-op">.</tt><tt id="link-130" class="py-name" targets="Class Bio.SeqIO.PhdIO.PhdWriter=Bio.SeqIO.PhdIO.PhdWriter-class.html"><a title="Bio.SeqIO.PhdIO.PhdWriter" class="py-name" href="#" onclick="return doclink('link-130', 'PhdWriter', 'link-130');">PhdWriter</a></tt><tt class="py-op">,</tt> </tt>
<a name="L474"></a><tt class="py-lineno"> 474</tt>  <tt class="py-line">                   <tt class="py-string">"qual"</tt><tt class="py-op">:</tt> <tt id="link-131" class="py-name"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-131', 'QualityIO', 'link-28');">QualityIO</a></tt><tt class="py-op">.</tt><tt id="link-132" class="py-name" targets="Class Bio.SeqIO.QualityIO.QualPhredWriter=Bio.SeqIO.QualityIO.QualPhredWriter-class.html"><a title="Bio.SeqIO.QualityIO.QualPhredWriter" class="py-name" href="#" onclick="return doclink('link-132', 'QualPhredWriter', 'link-132');">QualPhredWriter</a></tt><tt class="py-op">,</tt> </tt>
<a name="L475"></a><tt class="py-lineno"> 475</tt>  <tt class="py-line">                   <tt class="py-string">"sff"</tt><tt class="py-op">:</tt> <tt id="link-133" class="py-name"><a title="Bio.SeqIO.SffIO" class="py-name" href="#" onclick="return doclink('link-133', 'SffIO', 'link-25');">SffIO</a></tt><tt class="py-op">.</tt><tt id="link-134" class="py-name" targets="Class Bio.SeqIO.SffIO.SffWriter=Bio.SeqIO.SffIO.SffWriter-class.html"><a title="Bio.SeqIO.SffIO.SffWriter" class="py-name" href="#" onclick="return doclink('link-134', 'SffWriter', 'link-134');">SffWriter</a></tt><tt class="py-op">,</tt> </tt>
<a name="L476"></a><tt class="py-lineno"> 476</tt>  <tt class="py-line">                   <tt class="py-string">"seqxml"</tt><tt class="py-op">:</tt> <tt id="link-135" class="py-name"><a title="Bio.SeqIO.SeqXmlIO" class="py-name" href="#" onclick="return doclink('link-135', 'SeqXmlIO', 'link-24');">SeqXmlIO</a></tt><tt class="py-op">.</tt><tt id="link-136" class="py-name" targets="Class Bio.SeqIO.SeqXmlIO.SeqXmlWriter=Bio.SeqIO.SeqXmlIO.SeqXmlWriter-class.html"><a title="Bio.SeqIO.SeqXmlIO.SeqXmlWriter" class="py-name" href="#" onclick="return doclink('link-136', 'SeqXmlWriter', 'link-136');">SeqXmlWriter</a></tt><tt class="py-op">,</tt> </tt>
<a name="L477"></a><tt class="py-lineno"> 477</tt>  <tt class="py-line">                   <tt class="py-string">"pir"</tt><tt class="py-op">:</tt> <tt id="link-137" class="py-name"><a title="Bio.SeqIO.PirIO" class="py-name" href="#" onclick="return doclink('link-137', 'PirIO', 'link-23');">PirIO</a></tt><tt class="py-op">.</tt><tt id="link-138" class="py-name" targets="Class Bio.SeqIO.PirIO.PirWriter=Bio.SeqIO.PirIO.PirWriter-class.html"><a title="Bio.SeqIO.PirIO.PirWriter" class="py-name" href="#" onclick="return doclink('link-138', 'PirWriter', 'link-138');">PirWriter</a></tt><tt class="py-op">,</tt> </tt>
<a name="L478"></a><tt class="py-lineno"> 478</tt>  <tt class="py-line">                   <tt class="py-op">}</tt> </tt>
<a name="L479"></a><tt class="py-lineno"> 479</tt>  <tt class="py-line"> </tt>
<a name="L480"></a><tt class="py-lineno"> 480</tt>  <tt class="py-line"><tt id="link-139" class="py-name" targets="Variable Bio.SeqIO._BinaryFormats=Bio.SeqIO-module.html#_BinaryFormats"><a title="Bio.SeqIO._BinaryFormats" class="py-name" href="#" onclick="return doclink('link-139', '_BinaryFormats', 'link-139');">_BinaryFormats</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">"sff"</tt><tt class="py-op">,</tt> <tt class="py-string">"sff-trim"</tt><tt class="py-op">,</tt> <tt class="py-string">"abi"</tt><tt class="py-op">,</tt> <tt class="py-string">"abi-trim"</tt><tt class="py-op">,</tt> <tt class="py-string">"seqxml"</tt><tt class="py-op">]</tt> </tt>
<a name="L481"></a><tt class="py-lineno"> 481</tt>  <tt class="py-line"> </tt>
<a name="L482"></a><tt class="py-lineno"> 482</tt>  <tt class="py-line"> </tt>
<a name="write"></a><div id="write-def"><a name="L483"></a><tt class="py-lineno"> 483</tt> <a class="py-toggle" href="#" id="write-toggle" onclick="return toggle('write');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO-module.html#write">write</a><tt class="py-op">(</tt><tt class="py-param">sequences</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">format</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="write-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="write-expanded"><a name="L484"></a><tt class="py-lineno"> 484</tt>  <tt class="py-line">    <tt class="py-docstring">"""Write complete set of sequences to a file.</tt> </tt>
<a name="L485"></a><tt class="py-lineno"> 485</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L486"></a><tt class="py-lineno"> 486</tt>  <tt class="py-line"><tt class="py-docstring">    Arguments:</tt> </tt>
<a name="L487"></a><tt class="py-lineno"> 487</tt>  <tt class="py-line"><tt class="py-docstring">     - sequences - A list (or iterator) of SeqRecord objects, or (if using</tt> </tt>
<a name="L488"></a><tt class="py-lineno"> 488</tt>  <tt class="py-line"><tt class="py-docstring">       Biopython 1.54 or later) a single SeqRecord.</tt> </tt>
<a name="L489"></a><tt class="py-lineno"> 489</tt>  <tt class="py-line"><tt class="py-docstring">     - handle    - File handle object to write to, or filename as string</tt> </tt>
<a name="L490"></a><tt class="py-lineno"> 490</tt>  <tt class="py-line"><tt class="py-docstring">       (note older versions of Biopython only took a handle).</tt> </tt>
<a name="L491"></a><tt class="py-lineno"> 491</tt>  <tt class="py-line"><tt class="py-docstring">     - format    - lower case string describing the file format to write.</tt> </tt>
<a name="L492"></a><tt class="py-lineno"> 492</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L493"></a><tt class="py-lineno"> 493</tt>  <tt class="py-line"><tt class="py-docstring">    Note if providing a file handle, your code should close the handle</tt> </tt>
<a name="L494"></a><tt class="py-lineno"> 494</tt>  <tt class="py-line"><tt class="py-docstring">    after calling this function (to ensure the data gets flushed to disk).</tt> </tt>
<a name="L495"></a><tt class="py-lineno"> 495</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L496"></a><tt class="py-lineno"> 496</tt>  <tt class="py-line"><tt class="py-docstring">    Returns the number of records written (as an integer).</tt> </tt>
<a name="L497"></a><tt class="py-lineno"> 497</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L498"></a><tt class="py-lineno"> 498</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-140" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-140', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-141" class="py-name" targets="Package Bio.AlignIO=Bio.AlignIO-module.html"><a title="Bio.AlignIO" class="py-name" href="#" onclick="return doclink('link-141', 'AlignIO', 'link-141');">AlignIO</a></tt> </tt>
<a name="L499"></a><tt class="py-lineno"> 499</tt>  <tt class="py-line"> </tt>
<a name="L500"></a><tt class="py-lineno"> 500</tt>  <tt class="py-line">    <tt class="py-comment"># Try and give helpful error messages:</tt> </tt>
<a name="L501"></a><tt class="py-lineno"> 501</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-142" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.format()=Bio.Align.MultipleSeqAlignment-class.html#format,Method Bio.Phylo.BaseTree.Tree.format()=Bio.Phylo.BaseTree.Tree-class.html#format,Method Bio.Restriction.Restriction.RestrictionBatch.format()=Bio.Restriction.Restriction.RestrictionBatch-class.html#format,Method Bio.SeqRecord.SeqRecord.format()=Bio.SeqRecord.SeqRecord-class.html#format,Method Bio.motifs.Motif.format()=Bio.motifs.Motif-class.html#format"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-142', 'format', 'link-142');">format</a></tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L502"></a><tt class="py-lineno"> 502</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"Need a string for the file format (lower case)"</tt><tt class="py-op">)</tt> </tt>
<a name="L503"></a><tt class="py-lineno"> 503</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-143" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-143', 'format', 'link-142');">format</a></tt><tt class="py-op">:</tt> </tt>
<a name="L504"></a><tt class="py-lineno"> 504</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Format required (lower case string)"</tt><tt class="py-op">)</tt> </tt>
<a name="L505"></a><tt class="py-lineno"> 505</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt id="link-144" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-144', 'format', 'link-142');">format</a></tt> <tt class="py-op">!=</tt> <tt id="link-145" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-145', 'format', 'link-142');">format</a></tt><tt class="py-op">.</tt><tt id="link-146" class="py-name" targets="Method Bio.Seq.Seq.lower()=Bio.Seq.Seq-class.html#lower,Method Bio.Seq.UnknownSeq.lower()=Bio.Seq.UnknownSeq-class.html#lower,Method Bio.SeqRecord.SeqRecord.lower()=Bio.SeqRecord.SeqRecord-class.html#lower"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-146', 'lower', 'link-146');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L506"></a><tt class="py-lineno"> 506</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Format string '%s' should be lower case"</tt> <tt class="py-op">%</tt> <tt id="link-147" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-147', 'format', 'link-142');">format</a></tt><tt class="py-op">)</tt> </tt>
<a name="L507"></a><tt class="py-lineno"> 507</tt>  <tt class="py-line"> </tt>
<a name="L508"></a><tt class="py-lineno"> 508</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt id="link-148" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-148', 'SeqRecord', 'link-6');">SeqRecord</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L509"></a><tt class="py-lineno"> 509</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"Check arguments, handle should NOT be a SeqRecord"</tt><tt class="py-op">)</tt> </tt>
<a name="L510"></a><tt class="py-lineno"> 510</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L511"></a><tt class="py-lineno"> 511</tt>  <tt class="py-line">        <tt class="py-comment"># e.g. list of SeqRecord objects</tt> </tt>
<a name="L512"></a><tt class="py-lineno"> 512</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"Check arguments, handle should NOT be a list"</tt><tt class="py-op">)</tt> </tt>
<a name="L513"></a><tt class="py-lineno"> 513</tt>  <tt class="py-line"> </tt>
<a name="L514"></a><tt class="py-lineno"> 514</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">sequences</tt><tt class="py-op">,</tt> <tt id="link-149" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-149', 'SeqRecord', 'link-6');">SeqRecord</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L515"></a><tt class="py-lineno"> 515</tt>  <tt class="py-line">        <tt class="py-comment"># This raised an exception in older versions of Biopython</tt> </tt>
<a name="L516"></a><tt class="py-lineno"> 516</tt>  <tt class="py-line">        <tt class="py-name">sequences</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">sequences</tt><tt class="py-op">]</tt> </tt>
<a name="L517"></a><tt class="py-lineno"> 517</tt>  <tt class="py-line"> </tt>
<a name="L518"></a><tt class="py-lineno"> 518</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt id="link-150" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-150', 'format', 'link-142');">format</a></tt> <tt class="py-keyword">in</tt> <tt id="link-151" class="py-name"><a title="Bio.SeqIO._BinaryFormats" class="py-name" href="#" onclick="return doclink('link-151', '_BinaryFormats', 'link-139');">_BinaryFormats</a></tt><tt class="py-op">:</tt> </tt>
<a name="L519"></a><tt class="py-lineno"> 519</tt>  <tt class="py-line">        <tt class="py-name">mode</tt> <tt class="py-op">=</tt> <tt class="py-string">'wb'</tt> </tt>
<a name="L520"></a><tt class="py-lineno"> 520</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L521"></a><tt class="py-lineno"> 521</tt>  <tt class="py-line">        <tt class="py-name">mode</tt> <tt class="py-op">=</tt> <tt class="py-string">'w'</tt> </tt>
<a name="L522"></a><tt class="py-lineno"> 522</tt>  <tt class="py-line"> </tt>
<a name="L523"></a><tt class="py-lineno"> 523</tt>  <tt class="py-line">    <tt class="py-keyword">with</tt> <tt id="link-152" class="py-name"><a title="Bio.File.as_handle" class="py-name" href="#" onclick="return doclink('link-152', 'as_handle', 'link-4');">as_handle</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-name">mode</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">fp</tt><tt class="py-op">:</tt> </tt>
<a name="L524"></a><tt class="py-lineno"> 524</tt>  <tt class="py-line">        <tt class="py-comment"># Map the file format to a writer function/class</tt> </tt>
<a name="L525"></a><tt class="py-lineno"> 525</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-153" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-153', 'format', 'link-142');">format</a></tt> <tt class="py-keyword">in</tt> <tt id="link-154" class="py-name"><a title="Bio.SeqIO._FormatToString" class="py-name" href="#" onclick="return doclink('link-154', '_FormatToString', 'link-89');">_FormatToString</a></tt><tt class="py-op">:</tt> </tt>
<a name="L526"></a><tt class="py-lineno"> 526</tt>  <tt class="py-line">            <tt class="py-name">format_function</tt> <tt class="py-op">=</tt> <tt id="link-155" class="py-name"><a title="Bio.SeqIO._FormatToString" class="py-name" href="#" onclick="return doclink('link-155', '_FormatToString', 'link-89');">_FormatToString</a></tt><tt class="py-op">[</tt><tt id="link-156" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-156', 'format', 'link-142');">format</a></tt><tt class="py-op">]</tt> </tt>
<a name="L527"></a><tt class="py-lineno"> 527</tt>  <tt class="py-line">            <tt id="link-157" class="py-name" targets="Method Bio.Crystal.Chain.count()=Bio.Crystal.Chain-class.html#count,Method Bio.Seq.MutableSeq.count()=Bio.Seq.MutableSeq-class.html#count,Method Bio.Seq.Seq.count()=Bio.Seq.Seq-class.html#count,Method Bio.Seq.UnknownSeq.count()=Bio.Seq.UnknownSeq-class.html#count,Method Bio.motifs.Instances.count()=Bio.motifs.Instances-class.html#count"><a title="Bio.Crystal.Chain.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-157', 'count', 'link-157');">count</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L528"></a><tt class="py-lineno"> 528</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">sequences</tt><tt class="py-op">:</tt> </tt>
<a name="L529"></a><tt class="py-lineno"> 529</tt>  <tt class="py-line">                <tt class="py-name">fp</tt><tt class="py-op">.</tt><tt id="link-158" class="py-name" targets="Function Bio.AlignIO.write()=Bio.AlignIO-module.html#write,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.write()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#write,Method Bio.Phylo.CDAOIO.Writer.write()=Bio.Phylo.CDAOIO.Writer-class.html#write,Function Bio.Phylo.CDAOIO.write()=Bio.Phylo.CDAOIO-module.html#write,Method Bio.Phylo.NeXMLIO.Writer.write()=Bio.Phylo.NeXMLIO.Writer-class.html#write,Function Bio.Phylo.NeXMLIO.write()=Bio.Phylo.NeXMLIO-module.html#write,Method Bio.Phylo.NewickIO.Writer.write()=Bio.Phylo.NewickIO.Writer-class.html#write,Function Bio.Phylo.NewickIO.write()=Bio.Phylo.NewickIO-module.html#write,Function Bio.Phylo.NexusIO.write()=Bio.Phylo.NexusIO-module.html#write,Method Bio.Phylo.PhyloXMLIO.Writer.write()=Bio.Phylo.PhyloXMLIO.Writer-class.html#write,Function Bio.Phylo.PhyloXMLIO.write()=Bio.Phylo.PhyloXMLIO-module.html#write,Function Bio.Phylo._io.write()=Bio.Phylo._io-module.html#write,Function Bio.SearchIO.write()=Bio.SearchIO-module.html#write,Function Bio.SeqIO.write()=Bio.SeqIO-module.html#write,Method Bio.bgzf.BgzfWriter.write()=Bio.bgzf.BgzfWriter-class.html#write,Function Bio.motifs.jaspar.write()=Bio.motifs.jaspar-module.html#write,Function Bio.motifs.transfac.write()=Bio.motifs.transfac-module.html#write,Function Bio.motifs.write()=Bio.motifs-module.html#write,Method Bio.phenotype.phen_micro.JsonWriter.write()=Bio.phenotype.phen_micro.JsonWriter-class.html#write,Function Bio.phenotype.write()=Bio.phenotype-module.html#write"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-158', 'write', 'link-158');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">format_function</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L530"></a><tt class="py-lineno"> 530</tt>  <tt class="py-line">                <tt id="link-159" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-159', 'count', 'link-157');">count</a></tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L531"></a><tt class="py-lineno"> 531</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-160" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-160', 'format', 'link-142');">format</a></tt> <tt class="py-keyword">in</tt> <tt id="link-161" class="py-name"><a title="Bio.AlignIO._FormatToWriter
Bio.SeqIO._FormatToWriter
Bio.phenotype._FormatToWriter" class="py-name" href="#" onclick="return doclink('link-161', '_FormatToWriter', 'link-106');">_FormatToWriter</a></tt><tt class="py-op">:</tt> </tt>
<a name="L532"></a><tt class="py-lineno"> 532</tt>  <tt class="py-line">            <tt class="py-name">writer_class</tt> <tt class="py-op">=</tt> <tt id="link-162" class="py-name"><a title="Bio.AlignIO._FormatToWriter
Bio.SeqIO._FormatToWriter
Bio.phenotype._FormatToWriter" class="py-name" href="#" onclick="return doclink('link-162', '_FormatToWriter', 'link-106');">_FormatToWriter</a></tt><tt class="py-op">[</tt><tt id="link-163" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-163', 'format', 'link-142');">format</a></tt><tt class="py-op">]</tt> </tt>
<a name="L533"></a><tt class="py-lineno"> 533</tt>  <tt class="py-line">            <tt id="link-164" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-164', 'count', 'link-157');">count</a></tt> <tt class="py-op">=</tt> <tt class="py-name">writer_class</tt><tt class="py-op">(</tt><tt class="py-name">fp</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-165" class="py-name" targets="Method Bio.AlignIO.Interfaces.AlignmentWriter.write_file()=Bio.AlignIO.Interfaces.AlignmentWriter-class.html#write_file,Method Bio.AlignIO.Interfaces.SequentialAlignmentWriter.write_file()=Bio.AlignIO.Interfaces.SequentialAlignmentWriter-class.html#write_file,Method Bio.AlignIO.NexusIO.NexusWriter.write_file()=Bio.AlignIO.NexusIO.NexusWriter-class.html#write_file,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.write_file()=Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter-class.html#write_file,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.write_file()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter-class.html#write_file,Method Bio.SearchIO.BlatIO.BlatPslWriter.write_file()=Bio.SearchIO.BlatIO.BlatPslWriter-class.html#write_file,Method Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.write_file()=Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter-class.html#write_file,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.write_file()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter-class.html#write_file,Method Bio.SeqIO.Interfaces.SequenceWriter.write_file()=Bio.SeqIO.Interfaces.SequenceWriter-class.html#write_file,Method Bio.SeqIO.Interfaces.SequentialSequenceWriter.write_file()=Bio.SeqIO.Interfaces.SequentialSequenceWriter-class.html#write_file,Method Bio.SeqIO.SffIO.SffWriter.write_file()=Bio.SeqIO.SffIO.SffWriter-class.html#write_file"><a title="Bio.AlignIO.Interfaces.AlignmentWriter.write_file
Bio.AlignIO.Interfaces.SequentialAlignmentWriter.write_file
Bio.AlignIO.NexusIO.NexusWriter.write_file
Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.write_file
Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.write_file
Bio.SearchIO.BlatIO.BlatPslWriter.write_file
Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.write_file
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.write_file
Bio.SeqIO.Interfaces.SequenceWriter.write_file
Bio.SeqIO.Interfaces.SequentialSequenceWriter.write_file
Bio.SeqIO.SffIO.SffWriter.write_file" class="py-name" href="#" onclick="return doclink('link-165', 'write_file', 'link-165');">write_file</a></tt><tt class="py-op">(</tt><tt class="py-name">sequences</tt><tt class="py-op">)</tt> </tt>
<a name="L534"></a><tt class="py-lineno"> 534</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-166" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-166', 'format', 'link-142');">format</a></tt> <tt class="py-keyword">in</tt> <tt id="link-167" class="py-name"><a title="Bio.AlignIO" class="py-name" href="#" onclick="return doclink('link-167', 'AlignIO', 'link-141');">AlignIO</a></tt><tt class="py-op">.</tt><tt id="link-168" class="py-name"><a title="Bio.AlignIO._FormatToWriter
Bio.SeqIO._FormatToWriter
Bio.phenotype._FormatToWriter" class="py-name" href="#" onclick="return doclink('link-168', '_FormatToWriter', 'link-106');">_FormatToWriter</a></tt><tt class="py-op">:</tt> </tt>
<a name="L535"></a><tt class="py-lineno"> 535</tt>  <tt class="py-line">            <tt class="py-comment"># Try and turn all the records into a single alignment,</tt> </tt>
<a name="L536"></a><tt class="py-lineno"> 536</tt>  <tt class="py-line">            <tt class="py-comment"># and write that using Bio.AlignIO</tt> </tt>
<a name="L537"></a><tt class="py-lineno"> 537</tt>  <tt class="py-line">            <tt class="py-name">alignment</tt> <tt class="py-op">=</tt> <tt id="link-169" class="py-name"><a title="Bio.Align.MultipleSeqAlignment" class="py-name" href="#" onclick="return doclink('link-169', 'MultipleSeqAlignment', 'link-10');">MultipleSeqAlignment</a></tt><tt class="py-op">(</tt><tt class="py-name">sequences</tt><tt class="py-op">)</tt> </tt>
<a name="L538"></a><tt class="py-lineno"> 538</tt>  <tt class="py-line">            <tt class="py-name">alignment_count</tt> <tt class="py-op">=</tt> <tt id="link-170" class="py-name"><a title="Bio.AlignIO" class="py-name" href="#" onclick="return doclink('link-170', 'AlignIO', 'link-141');">AlignIO</a></tt><tt class="py-op">.</tt><tt id="link-171" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-171', 'write', 'link-158');">write</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">alignment</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">fp</tt><tt class="py-op">,</tt> <tt id="link-172" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-172', 'format', 'link-142');">format</a></tt><tt class="py-op">)</tt> </tt>
<a name="L539"></a><tt class="py-lineno"> 539</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">alignment_count</tt> <tt class="py-op">!=</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L540"></a><tt class="py-lineno"> 540</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">RuntimeError</tt><tt class="py-op">(</tt><tt class="py-string">"Internal error - the underlying writer "</tt> </tt>
<a name="L541"></a><tt class="py-lineno"> 541</tt>  <tt class="py-line">                                   <tt class="py-string">"should have returned 1, not %r"</tt> </tt>
<a name="L542"></a><tt class="py-lineno"> 542</tt>  <tt class="py-line">                                   <tt class="py-op">%</tt> <tt class="py-name">alignment_count</tt><tt class="py-op">)</tt> </tt>
<a name="L543"></a><tt class="py-lineno"> 543</tt>  <tt class="py-line">            <tt id="link-173" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-173', 'count', 'link-157');">count</a></tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">alignment</tt><tt class="py-op">)</tt> </tt>
<a name="L544"></a><tt class="py-lineno"> 544</tt>  <tt class="py-line">            <tt class="py-keyword">del</tt> <tt class="py-name">alignment_count</tt><tt class="py-op">,</tt> <tt class="py-name">alignment</tt> </tt>
<a name="L545"></a><tt class="py-lineno"> 545</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-174" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-174', 'format', 'link-142');">format</a></tt> <tt class="py-keyword">in</tt> <tt id="link-175" class="py-name"><a title="Bio.AlignIO._FormatToIterator
Bio.SeqIO._FormatToIterator
Bio.phenotype._FormatToIterator" class="py-name" href="#" onclick="return doclink('link-175', '_FormatToIterator', 'link-30');">_FormatToIterator</a></tt> <tt class="py-keyword">or</tt> <tt id="link-176" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-176', 'format', 'link-142');">format</a></tt> <tt class="py-keyword">in</tt> <tt id="link-177" class="py-name"><a title="Bio.AlignIO" class="py-name" href="#" onclick="return doclink('link-177', 'AlignIO', 'link-141');">AlignIO</a></tt><tt class="py-op">.</tt><tt id="link-178" class="py-name"><a title="Bio.AlignIO._FormatToIterator
Bio.SeqIO._FormatToIterator
Bio.phenotype._FormatToIterator" class="py-name" href="#" onclick="return doclink('link-178', '_FormatToIterator', 'link-30');">_FormatToIterator</a></tt><tt class="py-op">:</tt> </tt>
<a name="L546"></a><tt class="py-lineno"> 546</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Reading format '%s' is supported, but not writing"</tt> </tt>
<a name="L547"></a><tt class="py-lineno"> 547</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt id="link-179" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-179', 'format', 'link-142');">format</a></tt><tt class="py-op">)</tt> </tt>
<a name="L548"></a><tt class="py-lineno"> 548</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L549"></a><tt class="py-lineno"> 549</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unknown format '%s'"</tt> <tt class="py-op">%</tt> <tt id="link-180" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-180', 'format', 'link-142');">format</a></tt><tt class="py-op">)</tt> </tt>
<a name="L550"></a><tt class="py-lineno"> 550</tt>  <tt class="py-line"> </tt>
<a name="L551"></a><tt class="py-lineno"> 551</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-181" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-181', 'count', 'link-157');">count</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L552"></a><tt class="py-lineno"> 552</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">RuntimeError</tt><tt class="py-op">(</tt><tt class="py-string">"Internal error - the underlying %s writer "</tt> </tt>
<a name="L553"></a><tt class="py-lineno"> 553</tt>  <tt class="py-line">                               <tt class="py-string">"should have returned the record count, not %r"</tt> </tt>
<a name="L554"></a><tt class="py-lineno"> 554</tt>  <tt class="py-line">                               <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-182" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-182', 'format', 'link-142');">format</a></tt><tt class="py-op">,</tt> <tt id="link-183" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-183', 'count', 'link-157');">count</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L555"></a><tt class="py-lineno"> 555</tt>  <tt class="py-line"> </tt>
<a name="L556"></a><tt class="py-lineno"> 556</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-184" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-184', 'count', 'link-157');">count</a></tt> </tt>
</div><a name="L557"></a><tt class="py-lineno"> 557</tt>  <tt class="py-line"> </tt>
<a name="L558"></a><tt class="py-lineno"> 558</tt>  <tt class="py-line"> </tt>
<a name="parse"></a><div id="parse-def"><a name="L559"></a><tt class="py-lineno"> 559</tt> <a class="py-toggle" href="#" id="parse-toggle" onclick="return toggle('parse');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO-module.html#parse">parse</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">format</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="parse-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="parse-expanded"><a name="L560"></a><tt class="py-lineno"> 560</tt>  <tt class="py-line">    <tt class="py-docstring">r"""Turn a sequence file into an iterator returning SeqRecords.</tt> </tt>
<a name="L561"></a><tt class="py-lineno"> 561</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L562"></a><tt class="py-lineno"> 562</tt>  <tt class="py-line"><tt class="py-docstring">    Arguments:</tt> </tt>
<a name="L563"></a><tt class="py-lineno"> 563</tt>  <tt class="py-line"><tt class="py-docstring">     - handle   - handle to the file, or the filename as a string</tt> </tt>
<a name="L564"></a><tt class="py-lineno"> 564</tt>  <tt class="py-line"><tt class="py-docstring">       (note older versions of Biopython only took a handle).</tt> </tt>
<a name="L565"></a><tt class="py-lineno"> 565</tt>  <tt class="py-line"><tt class="py-docstring">     - format   - lower case string describing the file format.</tt> </tt>
<a name="L566"></a><tt class="py-lineno"> 566</tt>  <tt class="py-line"><tt class="py-docstring">     - alphabet - optional Alphabet object, useful when the sequence type</tt> </tt>
<a name="L567"></a><tt class="py-lineno"> 567</tt>  <tt class="py-line"><tt class="py-docstring">       cannot be automatically inferred from the file itself</tt> </tt>
<a name="L568"></a><tt class="py-lineno"> 568</tt>  <tt class="py-line"><tt class="py-docstring">       (e.g. format="fasta" or "tab")</tt> </tt>
<a name="L569"></a><tt class="py-lineno"> 569</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L570"></a><tt class="py-lineno"> 570</tt>  <tt class="py-line"><tt class="py-docstring">    Typical usage, opening a file to read in, and looping over the record(s):</tt> </tt>
<a name="L571"></a><tt class="py-lineno"> 571</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L572"></a><tt class="py-lineno"> 572</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L573"></a><tt class="py-lineno"> 573</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; filename = "Fasta/sweetpea.nu"</tt> </tt>
<a name="L574"></a><tt class="py-lineno"> 574</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; for record in SeqIO.parse(filename, "fasta"):</tt> </tt>
<a name="L575"></a><tt class="py-lineno"> 575</tt>  <tt class="py-line"><tt class="py-docstring">    ...    print("ID %s" % record.id)</tt> </tt>
<a name="L576"></a><tt class="py-lineno"> 576</tt>  <tt class="py-line"><tt class="py-docstring">    ...    print("Sequence length %i" % len(record))</tt> </tt>
<a name="L577"></a><tt class="py-lineno"> 577</tt>  <tt class="py-line"><tt class="py-docstring">    ...    print("Sequence alphabet %s" % record.seq.alphabet)</tt> </tt>
<a name="L578"></a><tt class="py-lineno"> 578</tt>  <tt class="py-line"><tt class="py-docstring">    ID gi|3176602|gb|U78617.1|LOU78617</tt> </tt>
<a name="L579"></a><tt class="py-lineno"> 579</tt>  <tt class="py-line"><tt class="py-docstring">    Sequence length 309</tt> </tt>
<a name="L580"></a><tt class="py-lineno"> 580</tt>  <tt class="py-line"><tt class="py-docstring">    Sequence alphabet SingleLetterAlphabet()</tt> </tt>
<a name="L581"></a><tt class="py-lineno"> 581</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L582"></a><tt class="py-lineno"> 582</tt>  <tt class="py-line"><tt class="py-docstring">    For file formats like FASTA where the alphabet cannot be determined, it</tt> </tt>
<a name="L583"></a><tt class="py-lineno"> 583</tt>  <tt class="py-line"><tt class="py-docstring">    may be useful to specify the alphabet explicitly:</tt> </tt>
<a name="L584"></a><tt class="py-lineno"> 584</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L585"></a><tt class="py-lineno"> 585</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L586"></a><tt class="py-lineno"> 586</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio.Alphabet import generic_dna</tt> </tt>
<a name="L587"></a><tt class="py-lineno"> 587</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; filename = "Fasta/sweetpea.nu"</tt> </tt>
<a name="L588"></a><tt class="py-lineno"> 588</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; for record in SeqIO.parse(filename, "fasta", generic_dna):</tt> </tt>
<a name="L589"></a><tt class="py-lineno"> 589</tt>  <tt class="py-line"><tt class="py-docstring">    ...    print("ID %s" % record.id)</tt> </tt>
<a name="L590"></a><tt class="py-lineno"> 590</tt>  <tt class="py-line"><tt class="py-docstring">    ...    print("Sequence length %i" % len(record))</tt> </tt>
<a name="L591"></a><tt class="py-lineno"> 591</tt>  <tt class="py-line"><tt class="py-docstring">    ...    print("Sequence alphabet %s" % record.seq.alphabet)</tt> </tt>
<a name="L592"></a><tt class="py-lineno"> 592</tt>  <tt class="py-line"><tt class="py-docstring">    ID gi|3176602|gb|U78617.1|LOU78617</tt> </tt>
<a name="L593"></a><tt class="py-lineno"> 593</tt>  <tt class="py-line"><tt class="py-docstring">    Sequence length 309</tt> </tt>
<a name="L594"></a><tt class="py-lineno"> 594</tt>  <tt class="py-line"><tt class="py-docstring">    Sequence alphabet DNAAlphabet()</tt> </tt>
<a name="L595"></a><tt class="py-lineno"> 595</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L596"></a><tt class="py-lineno"> 596</tt>  <tt class="py-line"><tt class="py-docstring">    If you have a string 'data' containing the file contents, you must</tt> </tt>
<a name="L597"></a><tt class="py-lineno"> 597</tt>  <tt class="py-line"><tt class="py-docstring">    first turn this into a handle in order to parse it:</tt> </tt>
<a name="L598"></a><tt class="py-lineno"> 598</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L599"></a><tt class="py-lineno"> 599</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; data = "&gt;Alpha\nACCGGATGTA\n&gt;Beta\nAGGCTCGGTTA\n"</tt> </tt>
<a name="L600"></a><tt class="py-lineno"> 600</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L601"></a><tt class="py-lineno"> 601</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; try:</tt> </tt>
<a name="L602"></a><tt class="py-lineno"> 602</tt>  <tt class="py-line"><tt class="py-docstring">    ...     from StringIO import StringIO # Python 2</tt> </tt>
<a name="L603"></a><tt class="py-lineno"> 603</tt>  <tt class="py-line"><tt class="py-docstring">    ... except ImportError:</tt> </tt>
<a name="L604"></a><tt class="py-lineno"> 604</tt>  <tt class="py-line"><tt class="py-docstring">    ...     from io import StringIO # Python 3</tt> </tt>
<a name="L605"></a><tt class="py-lineno"> 605</tt>  <tt class="py-line"><tt class="py-docstring">    ...</tt> </tt>
<a name="L606"></a><tt class="py-lineno"> 606</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; for record in SeqIO.parse(StringIO(data), "fasta"):</tt> </tt>
<a name="L607"></a><tt class="py-lineno"> 607</tt>  <tt class="py-line"><tt class="py-docstring">    ...     print("%s %s" % (record.id, record.seq))</tt> </tt>
<a name="L608"></a><tt class="py-lineno"> 608</tt>  <tt class="py-line"><tt class="py-docstring">    Alpha ACCGGATGTA</tt> </tt>
<a name="L609"></a><tt class="py-lineno"> 609</tt>  <tt class="py-line"><tt class="py-docstring">    Beta AGGCTCGGTTA</tt> </tt>
<a name="L610"></a><tt class="py-lineno"> 610</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L611"></a><tt class="py-lineno"> 611</tt>  <tt class="py-line"><tt class="py-docstring">    Use the Bio.SeqIO.read(...) function when you expect a single record</tt> </tt>
<a name="L612"></a><tt class="py-lineno"> 612</tt>  <tt class="py-line"><tt class="py-docstring">    only.</tt> </tt>
<a name="L613"></a><tt class="py-lineno"> 613</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L614"></a><tt class="py-lineno"> 614</tt>  <tt class="py-line">    <tt class="py-comment"># NOTE - The above docstring has some raw \n characters needed</tt> </tt>
<a name="L615"></a><tt class="py-lineno"> 615</tt>  <tt class="py-line">    <tt class="py-comment"># for the StringIO example, hence the whole docstring is in raw</tt> </tt>
<a name="L616"></a><tt class="py-lineno"> 616</tt>  <tt class="py-line">    <tt class="py-comment"># string mode (see the leading r before the opening quote).</tt> </tt>
<a name="L617"></a><tt class="py-lineno"> 617</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-185" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-185', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-186" class="py-name"><a title="Bio.AlignIO" class="py-name" href="#" onclick="return doclink('link-186', 'AlignIO', 'link-141');">AlignIO</a></tt> </tt>
<a name="L618"></a><tt class="py-lineno"> 618</tt>  <tt class="py-line"> </tt>
<a name="L619"></a><tt class="py-lineno"> 619</tt>  <tt class="py-line">    <tt class="py-comment"># Hack for SFF, will need to make this more general in future</tt> </tt>
<a name="L620"></a><tt class="py-lineno"> 620</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt id="link-187" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-187', 'format', 'link-142');">format</a></tt> <tt class="py-keyword">in</tt> <tt id="link-188" class="py-name"><a title="Bio.SeqIO._BinaryFormats" class="py-name" href="#" onclick="return doclink('link-188', '_BinaryFormats', 'link-139');">_BinaryFormats</a></tt><tt class="py-op">:</tt> </tt>
<a name="L621"></a><tt class="py-lineno"> 621</tt>  <tt class="py-line">        <tt class="py-name">mode</tt> <tt class="py-op">=</tt> <tt class="py-string">'rb'</tt> </tt>
<a name="L622"></a><tt class="py-lineno"> 622</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L623"></a><tt class="py-lineno"> 623</tt>  <tt class="py-line">        <tt class="py-name">mode</tt> <tt class="py-op">=</tt> <tt class="py-string">'rU'</tt> </tt>
<a name="L624"></a><tt class="py-lineno"> 624</tt>  <tt class="py-line"> </tt>
<a name="L625"></a><tt class="py-lineno"> 625</tt>  <tt class="py-line">    <tt class="py-comment"># Try and give helpful error messages:</tt> </tt>
<a name="L626"></a><tt class="py-lineno"> 626</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-189" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-189', 'format', 'link-142');">format</a></tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L627"></a><tt class="py-lineno"> 627</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"Need a string for the file format (lower case)"</tt><tt class="py-op">)</tt> </tt>
<a name="L628"></a><tt class="py-lineno"> 628</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-190" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-190', 'format', 'link-142');">format</a></tt><tt class="py-op">:</tt> </tt>
<a name="L629"></a><tt class="py-lineno"> 629</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Format required (lower case string)"</tt><tt class="py-op">)</tt> </tt>
<a name="L630"></a><tt class="py-lineno"> 630</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt id="link-191" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-191', 'format', 'link-142');">format</a></tt> <tt class="py-op">!=</tt> <tt id="link-192" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-192', 'format', 'link-142');">format</a></tt><tt class="py-op">.</tt><tt id="link-193" class="py-name"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-193', 'lower', 'link-146');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L631"></a><tt class="py-lineno"> 631</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Format string '%s' should be lower case"</tt> <tt class="py-op">%</tt> <tt id="link-194" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-194', 'format', 'link-142');">format</a></tt><tt class="py-op">)</tt> </tt>
<a name="L632"></a><tt class="py-lineno"> 632</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt id="link-195" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.alphabet=Bio.SearchIO._model.hsp.HSP-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSPFragment.alphabet=Bio.SearchIO._model.hsp.HSPFragment-class.html#alphabet"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-195', 'alphabet', 'link-195');">alphabet</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt class="py-op">(</tt><tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-196" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-196', 'alphabet', 'link-195');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-197" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-197', 'Alphabet', 'link-12');">Alphabet</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> </tt>
<a name="L633"></a><tt class="py-lineno"> 633</tt>  <tt class="py-line">                                     <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-198" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-198', 'alphabet', 'link-195');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-199" class="py-name"><a title="Bio.Alphabet.AlphabetEncoder" class="py-name" href="#" onclick="return doclink('link-199', 'AlphabetEncoder', 'link-14');">AlphabetEncoder</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L634"></a><tt class="py-lineno"> 634</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid alphabet, %r"</tt> <tt class="py-op">%</tt> <tt id="link-200" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-200', 'alphabet', 'link-195');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L635"></a><tt class="py-lineno"> 635</tt>  <tt class="py-line"> </tt>
<a name="L636"></a><tt class="py-lineno"> 636</tt>  <tt class="py-line">    <tt class="py-keyword">with</tt> <tt id="link-201" class="py-name"><a title="Bio.File.as_handle" class="py-name" href="#" onclick="return doclink('link-201', 'as_handle', 'link-4');">as_handle</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-name">mode</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">fp</tt><tt class="py-op">:</tt> </tt>
<a name="L637"></a><tt class="py-lineno"> 637</tt>  <tt class="py-line">        <tt class="py-comment"># Map the file format to a sequence iterator:</tt> </tt>
<a name="L638"></a><tt class="py-lineno"> 638</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-202" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-202', 'format', 'link-142');">format</a></tt> <tt class="py-keyword">in</tt> <tt id="link-203" class="py-name"><a title="Bio.AlignIO._FormatToIterator
Bio.SeqIO._FormatToIterator
Bio.phenotype._FormatToIterator" class="py-name" href="#" onclick="return doclink('link-203', '_FormatToIterator', 'link-30');">_FormatToIterator</a></tt><tt class="py-op">:</tt> </tt>
<a name="L639"></a><tt class="py-lineno"> 639</tt>  <tt class="py-line">            <tt class="py-name">iterator_generator</tt> <tt class="py-op">=</tt> <tt id="link-204" class="py-name"><a title="Bio.AlignIO._FormatToIterator
Bio.SeqIO._FormatToIterator
Bio.phenotype._FormatToIterator" class="py-name" href="#" onclick="return doclink('link-204', '_FormatToIterator', 'link-30');">_FormatToIterator</a></tt><tt class="py-op">[</tt><tt id="link-205" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-205', 'format', 'link-142');">format</a></tt><tt class="py-op">]</tt> </tt>
<a name="L640"></a><tt class="py-lineno"> 640</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-206" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-206', 'alphabet', 'link-195');">alphabet</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L641"></a><tt class="py-lineno"> 641</tt>  <tt class="py-line">                <tt id="link-207" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-207', 'i', 'link-207');">i</a></tt> <tt class="py-op">=</tt> <tt class="py-name">iterator_generator</tt><tt class="py-op">(</tt><tt class="py-name">fp</tt><tt class="py-op">)</tt> </tt>
<a name="L642"></a><tt class="py-lineno"> 642</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L643"></a><tt class="py-lineno"> 643</tt>  <tt class="py-line">                <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L644"></a><tt class="py-lineno"> 644</tt>  <tt class="py-line">                    <tt id="link-208" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-208', 'i', 'link-207');">i</a></tt> <tt class="py-op">=</tt> <tt class="py-name">iterator_generator</tt><tt class="py-op">(</tt><tt class="py-name">fp</tt><tt class="py-op">,</tt> <tt id="link-209" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-209', 'alphabet', 'link-195');">alphabet</a></tt><tt class="py-op">=</tt><tt id="link-210" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-210', 'alphabet', 'link-195');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L645"></a><tt class="py-lineno"> 645</tt>  <tt class="py-line">                <tt class="py-keyword">except</tt> <tt class="py-name">TypeError</tt><tt class="py-op">:</tt> </tt>
<a name="L646"></a><tt class="py-lineno"> 646</tt>  <tt class="py-line">                    <tt id="link-211" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-211', 'i', 'link-207');">i</a></tt> <tt class="py-op">=</tt> <tt id="link-212" class="py-name" targets="Function Bio.AlignIO._force_alphabet()=Bio.AlignIO-module.html#_force_alphabet,Function Bio.SeqIO._force_alphabet()=Bio.SeqIO-module.html#_force_alphabet"><a title="Bio.AlignIO._force_alphabet
Bio.SeqIO._force_alphabet" class="py-name" href="#" onclick="return doclink('link-212', '_force_alphabet', 'link-212');">_force_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">iterator_generator</tt><tt class="py-op">(</tt><tt class="py-name">fp</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-213" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-213', 'alphabet', 'link-195');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L647"></a><tt class="py-lineno"> 647</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-214" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-214', 'format', 'link-142');">format</a></tt> <tt class="py-keyword">in</tt> <tt id="link-215" class="py-name"><a title="Bio.AlignIO" class="py-name" href="#" onclick="return doclink('link-215', 'AlignIO', 'link-141');">AlignIO</a></tt><tt class="py-op">.</tt><tt id="link-216" class="py-name"><a title="Bio.AlignIO._FormatToIterator
Bio.SeqIO._FormatToIterator
Bio.phenotype._FormatToIterator" class="py-name" href="#" onclick="return doclink('link-216', '_FormatToIterator', 'link-30');">_FormatToIterator</a></tt><tt class="py-op">:</tt> </tt>
<a name="L648"></a><tt class="py-lineno"> 648</tt>  <tt class="py-line">            <tt class="py-comment"># Use Bio.AlignIO to read in the alignments</tt> </tt>
<a name="L649"></a><tt class="py-lineno"> 649</tt>  <tt class="py-line">            <tt id="link-217" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-217', 'i', 'link-207');">i</a></tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">r</tt> <tt class="py-keyword">for</tt> <tt class="py-name">alignment</tt> <tt class="py-keyword">in</tt> <tt id="link-218" class="py-name"><a title="Bio.AlignIO" class="py-name" href="#" onclick="return doclink('link-218', 'AlignIO', 'link-141');">AlignIO</a></tt><tt class="py-op">.</tt><tt id="link-219" class="py-name" targets="Function Bio.AlignIO.parse()=Bio.AlignIO-module.html#parse,Function Bio.Blast.NCBIXML.parse()=Bio.Blast.NCBIXML-module.html#parse,Function Bio.Compass.parse()=Bio.Compass-module.html#parse,Function Bio.Emboss.Primer3.parse()=Bio.Emboss.Primer3-module.html#parse,Method Bio.Entrez.Parser.DataHandler.parse()=Bio.Entrez.Parser.DataHandler-class.html#parse,Function Bio.Entrez.parse()=Bio.Entrez-module.html#parse,Function Bio.ExPASy.Enzyme.parse()=Bio.ExPASy.Enzyme-module.html#parse,Function Bio.ExPASy.Prodoc.parse()=Bio.ExPASy.Prodoc-module.html#parse,Function Bio.ExPASy.Prosite.parse()=Bio.ExPASy.Prosite-module.html#parse,Function Bio.ExPASy.cellosaurus.parse()=Bio.ExPASy.cellosaurus-module.html#parse,Method Bio.GenBank.FeatureParser.parse()=Bio.GenBank.FeatureParser-class.html#parse,Method Bio.GenBank.RecordParser.parse()=Bio.GenBank.RecordParser-class.html#parse,Method Bio.GenBank.Scanner.InsdcScanner.parse()=Bio.GenBank.Scanner.InsdcScanner-class.html#parse,Function Bio.Geo.parse()=Bio.Geo-module.html#parse,Function Bio.KEGG.Compound.parse()=Bio.KEGG.Compound-module.html#parse,Function Bio.KEGG.Enzyme.parse()=Bio.KEGG.Enzyme-module.html#parse,Function Bio.KEGG.Gene.parse()=Bio.KEGG.Gene-module.html#parse,Method Bio.KEGG.KGML.KGML_parser.KGMLParser.parse()=Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html#parse,Function Bio.KEGG.KGML.KGML_parser.parse()=Bio.KEGG.KGML.KGML_parser-module.html#parse,Function Bio.KEGG.Map.parse()=Bio.KEGG.Map-module.html#parse,Function Bio.Medline.parse()=Bio.Medline-module.html#parse,Method Bio.Phylo.CDAOIO.Parser.parse()=Bio.Phylo.CDAOIO.Parser-class.html#parse,Function Bio.Phylo.CDAOIO.parse()=Bio.Phylo.CDAOIO-module.html#parse,Method Bio.Phylo.NeXMLIO.Parser.parse()=Bio.Phylo.NeXMLIO.Parser-class.html#parse,Function Bio.Phylo.NeXMLIO.parse()=Bio.Phylo.NeXMLIO-module.html#parse,Method Bio.Phylo.NewickIO.Parser.parse()=Bio.Phylo.NewickIO.Parser-class.html#parse,Function Bio.Phylo.NewickIO.parse()=Bio.Phylo.NewickIO-module.html#parse,Function Bio.Phylo.NexusIO.parse()=Bio.Phylo.NexusIO-module.html#parse,Method Bio.Phylo.PhyloXMLIO.Parser.parse()=Bio.Phylo.PhyloXMLIO.Parser-class.html#parse,Function Bio.Phylo.PhyloXMLIO.parse()=Bio.Phylo.PhyloXMLIO-module.html#parse,Function Bio.Phylo._io.parse()=Bio.Phylo._io-module.html#parse,Function Bio.SCOP.Cla.parse()=Bio.SCOP.Cla-module.html#parse,Function Bio.SCOP.Des.parse()=Bio.SCOP.Des-module.html#parse,Function Bio.SCOP.Dom.parse()=Bio.SCOP.Dom-module.html#parse,Function Bio.SCOP.Hie.parse()=Bio.SCOP.Hie-module.html#parse,Function Bio.SCOP.Raf.parse()=Bio.SCOP.Raf-module.html#parse,Method Bio.SearchIO._legacy.NCBIStandalone.BlastErrorParser.parse()=Bio.SearchIO._legacy.NCBIStandalone.BlastErrorParser-class.html#parse,Method Bio.SearchIO._legacy.NCBIStandalone.BlastParser.parse()=Bio.SearchIO._legacy.NCBIStandalone.BlastParser-class.html#parse,Method Bio.SearchIO._legacy.NCBIStandalone.PSIBlastParser.parse()=Bio.SearchIO._legacy.NCBIStandalone.PSIBlastParser-class.html#parse,Method Bio.SearchIO._legacy.ParserSupport.AbstractParser.parse()=Bio.SearchIO._legacy.ParserSupport.AbstractParser-class.html#parse,Function Bio.SearchIO.parse()=Bio.SearchIO-module.html#parse,Method Bio.SeqIO.UniprotIO.Parser.parse()=Bio.SeqIO.UniprotIO.Parser-class.html#parse,Function Bio.SeqIO.parse()=Bio.SeqIO-module.html#parse,Function Bio.Sequencing.Ace.parse()=Bio.Sequencing.Ace-module.html#parse,Function Bio.Sequencing.Phd.parse()=Bio.Sequencing.Phd-module.html#parse,Function Bio.SwissProt.KeyWList.parse()=Bio.SwissProt.KeyWList-module.html#parse,Function Bio.SwissProt.parse()=Bio.SwissProt-module.html#parse,Function Bio.UniGene.parse()=Bio.UniGene-module.html#parse,Function Bio.Wise.psw.parse()=Bio.Wise.psw-module.html#parse,Function Bio.motifs.parse()=Bio.motifs-module.html#parse,Function Bio.phenotype.parse()=Bio.phenotype-module.html#parse"><a title="Bio.AlignIO.parse
Bio.Blast.NCBIXML.parse
Bio.Compass.parse
Bio.Emboss.Primer3.parse
Bio.Entrez.Parser.DataHandler.parse
Bio.Entrez.parse
Bio.ExPASy.Enzyme.parse
Bio.ExPASy.Prodoc.parse
Bio.ExPASy.Prosite.parse
Bio.ExPASy.cellosaurus.parse
Bio.GenBank.FeatureParser.parse
Bio.GenBank.RecordParser.parse
Bio.GenBank.Scanner.InsdcScanner.parse
Bio.Geo.parse
Bio.KEGG.Compound.parse
Bio.KEGG.Enzyme.parse
Bio.KEGG.Gene.parse
Bio.KEGG.KGML.KGML_parser.KGMLParser.parse
Bio.KEGG.KGML.KGML_parser.parse
Bio.KEGG.Map.parse
Bio.Medline.parse
Bio.Phylo.CDAOIO.Parser.parse
Bio.Phylo.CDAOIO.parse
Bio.Phylo.NeXMLIO.Parser.parse
Bio.Phylo.NeXMLIO.parse
Bio.Phylo.NewickIO.Parser.parse
Bio.Phylo.NewickIO.parse
Bio.Phylo.NexusIO.parse
Bio.Phylo.PhyloXMLIO.Parser.parse
Bio.Phylo.PhyloXMLIO.parse
Bio.Phylo._io.parse
Bio.SCOP.Cla.parse
Bio.SCOP.Des.parse
Bio.SCOP.Dom.parse
Bio.SCOP.Hie.parse
Bio.SCOP.Raf.parse
Bio.SearchIO._legacy.NCBIStandalone.BlastErrorParser.parse
Bio.SearchIO._legacy.NCBIStandalone.BlastParser.parse
Bio.SearchIO._legacy.NCBIStandalone.PSIBlastParser.parse
Bio.SearchIO._legacy.ParserSupport.AbstractParser.parse
Bio.SearchIO.parse
Bio.SeqIO.UniprotIO.Parser.parse
Bio.SeqIO.parse
Bio.Sequencing.Ace.parse
Bio.Sequencing.Phd.parse
Bio.SwissProt.KeyWList.parse
Bio.SwissProt.parse
Bio.UniGene.parse
Bio.Wise.psw.parse
Bio.motifs.parse
Bio.phenotype.parse" class="py-name" href="#" onclick="return doclink('link-219', 'parse', 'link-219');">parse</a></tt><tt class="py-op">(</tt><tt class="py-name">fp</tt><tt class="py-op">,</tt> <tt id="link-220" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-220', 'format', 'link-142');">format</a></tt><tt class="py-op">,</tt> </tt>
<a name="L650"></a><tt class="py-lineno"> 650</tt>  <tt class="py-line">                                                  <tt id="link-221" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-221', 'alphabet', 'link-195');">alphabet</a></tt><tt class="py-op">=</tt><tt id="link-222" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-222', 'alphabet', 'link-195');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L651"></a><tt class="py-lineno"> 651</tt>  <tt class="py-line">                 <tt class="py-keyword">for</tt> <tt class="py-name">r</tt> <tt class="py-keyword">in</tt> <tt class="py-name">alignment</tt><tt class="py-op">)</tt> </tt>
<a name="L652"></a><tt class="py-lineno"> 652</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L653"></a><tt class="py-lineno"> 653</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unknown format '%s'"</tt> <tt class="py-op">%</tt> <tt id="link-223" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-223', 'format', 'link-142');">format</a></tt><tt class="py-op">)</tt> </tt>
<a name="L654"></a><tt class="py-lineno"> 654</tt>  <tt class="py-line">        <tt class="py-comment"># This imposes some overhead... wait until we drop Python 2.4 to fix it</tt> </tt>
<a name="L655"></a><tt class="py-lineno"> 655</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">r</tt> <tt class="py-keyword">in</tt> <tt id="link-224" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-224', 'i', 'link-207');">i</a></tt><tt class="py-op">:</tt> </tt>
<a name="L656"></a><tt class="py-lineno"> 656</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt class="py-name">r</tt> </tt>
</div><a name="L657"></a><tt class="py-lineno"> 657</tt>  <tt class="py-line"> </tt>
<a name="L658"></a><tt class="py-lineno"> 658</tt>  <tt class="py-line"> </tt>
<a name="_force_alphabet"></a><div id="_force_alphabet-def"><a name="L659"></a><tt class="py-lineno"> 659</tt> <a class="py-toggle" href="#" id="_force_alphabet-toggle" onclick="return toggle('_force_alphabet');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO-module.html#_force_alphabet">_force_alphabet</a><tt class="py-op">(</tt><tt class="py-param">record_iterator</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_force_alphabet-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_force_alphabet-expanded"><a name="L660"></a><tt class="py-lineno"> 660</tt>  <tt class="py-line">    <tt class="py-docstring">"""Iterate over records, over-riding the alphabet (PRIVATE)."""</tt> </tt>
<a name="L661"></a><tt class="py-lineno"> 661</tt>  <tt class="py-line">    <tt class="py-comment"># Assume the alphabet argument has been pre-validated</tt> </tt>
<a name="L662"></a><tt class="py-lineno"> 662</tt>  <tt class="py-line">    <tt class="py-name">given_base_class</tt> <tt class="py-op">=</tt> <tt id="link-225" class="py-name"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-225', '_get_base_alphabet', 'link-15');">_get_base_alphabet</a></tt><tt class="py-op">(</tt><tt id="link-226" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-226', 'alphabet', 'link-195');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt> </tt>
<a name="L663"></a><tt class="py-lineno"> 663</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record_iterator</tt><tt class="py-op">:</tt> </tt>
<a name="L664"></a><tt class="py-lineno"> 664</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-227" class="py-name"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-227', '_get_base_alphabet', 'link-15');">_get_base_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-228" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.seq=Bio.SeqRecord.SeqRecord-class.html#seq,Variable BioSQL.BioSeq.DBSeqRecord.seq=BioSQL.BioSeq.DBSeqRecord-class.html#seq"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-228', 'seq', 'link-228');">seq</a></tt><tt class="py-op">.</tt><tt id="link-229" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-229', 'alphabet', 'link-195');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L665"></a><tt class="py-lineno"> 665</tt>  <tt class="py-line">                      <tt class="py-name">given_base_class</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L666"></a><tt class="py-lineno"> 666</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-230" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-230', 'seq', 'link-228');">seq</a></tt><tt class="py-op">.</tt><tt id="link-231" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-231', 'alphabet', 'link-195');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-232" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-232', 'alphabet', 'link-195');">alphabet</a></tt> </tt>
<a name="L667"></a><tt class="py-lineno"> 667</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt class="py-name">record</tt> </tt>
<a name="L668"></a><tt class="py-lineno"> 668</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L669"></a><tt class="py-lineno"> 669</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Specified alphabet %r clashes with "</tt> </tt>
<a name="L670"></a><tt class="py-lineno"> 670</tt>  <tt class="py-line">                             <tt class="py-string">"that determined from the file, %r"</tt> </tt>
<a name="L671"></a><tt class="py-lineno"> 671</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-233" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-233', 'alphabet', 'link-195');">alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-234" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-234', 'seq', 'link-228');">seq</a></tt><tt class="py-op">.</tt><tt id="link-235" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-235', 'alphabet', 'link-195');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L672"></a><tt class="py-lineno"> 672</tt>  <tt class="py-line"> </tt>
<a name="L673"></a><tt class="py-lineno"> 673</tt>  <tt class="py-line"> </tt>
<a name="read"></a><div id="read-def"><a name="L674"></a><tt class="py-lineno"> 674</tt> <a class="py-toggle" href="#" id="read-toggle" onclick="return toggle('read');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO-module.html#read">read</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">format</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="read-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="read-expanded"><a name="L675"></a><tt class="py-lineno"> 675</tt>  <tt class="py-line">    <tt class="py-docstring">"""Turn a sequence file into a single SeqRecord.</tt> </tt>
<a name="L676"></a><tt class="py-lineno"> 676</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L677"></a><tt class="py-lineno"> 677</tt>  <tt class="py-line"><tt class="py-docstring">    Arguments:</tt> </tt>
<a name="L678"></a><tt class="py-lineno"> 678</tt>  <tt class="py-line"><tt class="py-docstring">     - handle   - handle to the file, or the filename as a string</tt> </tt>
<a name="L679"></a><tt class="py-lineno"> 679</tt>  <tt class="py-line"><tt class="py-docstring">       (note older versions of Biopython only took a handle).</tt> </tt>
<a name="L680"></a><tt class="py-lineno"> 680</tt>  <tt class="py-line"><tt class="py-docstring">     - format   - string describing the file format.</tt> </tt>
<a name="L681"></a><tt class="py-lineno"> 681</tt>  <tt class="py-line"><tt class="py-docstring">     - alphabet - optional Alphabet object, useful when the sequence type</tt> </tt>
<a name="L682"></a><tt class="py-lineno"> 682</tt>  <tt class="py-line"><tt class="py-docstring">       cannot be automatically inferred from the file itself</tt> </tt>
<a name="L683"></a><tt class="py-lineno"> 683</tt>  <tt class="py-line"><tt class="py-docstring">       (e.g. format="fasta" or "tab")</tt> </tt>
<a name="L684"></a><tt class="py-lineno"> 684</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L685"></a><tt class="py-lineno"> 685</tt>  <tt class="py-line"><tt class="py-docstring">    This function is for use parsing sequence files containing</tt> </tt>
<a name="L686"></a><tt class="py-lineno"> 686</tt>  <tt class="py-line"><tt class="py-docstring">    exactly one record.  For example, reading a GenBank file:</tt> </tt>
<a name="L687"></a><tt class="py-lineno"> 687</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L688"></a><tt class="py-lineno"> 688</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L689"></a><tt class="py-lineno"> 689</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; record = SeqIO.read("GenBank/arab1.gb", "genbank")</tt> </tt>
<a name="L690"></a><tt class="py-lineno"> 690</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("ID %s" % record.id)</tt> </tt>
<a name="L691"></a><tt class="py-lineno"> 691</tt>  <tt class="py-line"><tt class="py-docstring">    ID AC007323.5</tt> </tt>
<a name="L692"></a><tt class="py-lineno"> 692</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("Sequence length %i" % len(record))</tt> </tt>
<a name="L693"></a><tt class="py-lineno"> 693</tt>  <tt class="py-line"><tt class="py-docstring">    Sequence length 86436</tt> </tt>
<a name="L694"></a><tt class="py-lineno"> 694</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("Sequence alphabet %s" % record.seq.alphabet)</tt> </tt>
<a name="L695"></a><tt class="py-lineno"> 695</tt>  <tt class="py-line"><tt class="py-docstring">    Sequence alphabet IUPACAmbiguousDNA()</tt> </tt>
<a name="L696"></a><tt class="py-lineno"> 696</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L697"></a><tt class="py-lineno"> 697</tt>  <tt class="py-line"><tt class="py-docstring">    If the handle contains no records, or more than one record,</tt> </tt>
<a name="L698"></a><tt class="py-lineno"> 698</tt>  <tt class="py-line"><tt class="py-docstring">    an exception is raised.  For example:</tt> </tt>
<a name="L699"></a><tt class="py-lineno"> 699</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L700"></a><tt class="py-lineno"> 700</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L701"></a><tt class="py-lineno"> 701</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; record = SeqIO.read("GenBank/cor6_6.gb", "genbank")</tt> </tt>
<a name="L702"></a><tt class="py-lineno"> 702</tt>  <tt class="py-line"><tt class="py-docstring">    Traceback (most recent call last):</tt> </tt>
<a name="L703"></a><tt class="py-lineno"> 703</tt>  <tt class="py-line"><tt class="py-docstring">        ...</tt> </tt>
<a name="L704"></a><tt class="py-lineno"> 704</tt>  <tt class="py-line"><tt class="py-docstring">    ValueError: More than one record found in handle</tt> </tt>
<a name="L705"></a><tt class="py-lineno"> 705</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L706"></a><tt class="py-lineno"> 706</tt>  <tt class="py-line"><tt class="py-docstring">    If however you want the first record from a file containing</tt> </tt>
<a name="L707"></a><tt class="py-lineno"> 707</tt>  <tt class="py-line"><tt class="py-docstring">    multiple records this function would raise an exception (as</tt> </tt>
<a name="L708"></a><tt class="py-lineno"> 708</tt>  <tt class="py-line"><tt class="py-docstring">    shown in the example above).  Instead use:</tt> </tt>
<a name="L709"></a><tt class="py-lineno"> 709</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L710"></a><tt class="py-lineno"> 710</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L711"></a><tt class="py-lineno"> 711</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; record = next(SeqIO.parse("GenBank/cor6_6.gb", "genbank"))</tt> </tt>
<a name="L712"></a><tt class="py-lineno"> 712</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("First record's ID %s" % record.id)</tt> </tt>
<a name="L713"></a><tt class="py-lineno"> 713</tt>  <tt class="py-line"><tt class="py-docstring">    First record's ID X55053.1</tt> </tt>
<a name="L714"></a><tt class="py-lineno"> 714</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L715"></a><tt class="py-lineno"> 715</tt>  <tt class="py-line"><tt class="py-docstring">    Use the Bio.SeqIO.parse(handle, format) function if you want</tt> </tt>
<a name="L716"></a><tt class="py-lineno"> 716</tt>  <tt class="py-line"><tt class="py-docstring">    to read multiple records from the handle.</tt> </tt>
<a name="L717"></a><tt class="py-lineno"> 717</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L718"></a><tt class="py-lineno"> 718</tt>  <tt class="py-line">    <tt class="py-name">iterator</tt> <tt class="py-op">=</tt> <tt id="link-236" class="py-name"><a title="Bio.AlignIO.parse
Bio.Blast.NCBIXML.parse
Bio.Compass.parse
Bio.Emboss.Primer3.parse
Bio.Entrez.Parser.DataHandler.parse
Bio.Entrez.parse
Bio.ExPASy.Enzyme.parse
Bio.ExPASy.Prodoc.parse
Bio.ExPASy.Prosite.parse
Bio.ExPASy.cellosaurus.parse
Bio.GenBank.FeatureParser.parse
Bio.GenBank.RecordParser.parse
Bio.GenBank.Scanner.InsdcScanner.parse
Bio.Geo.parse
Bio.KEGG.Compound.parse
Bio.KEGG.Enzyme.parse
Bio.KEGG.Gene.parse
Bio.KEGG.KGML.KGML_parser.KGMLParser.parse
Bio.KEGG.KGML.KGML_parser.parse
Bio.KEGG.Map.parse
Bio.Medline.parse
Bio.Phylo.CDAOIO.Parser.parse
Bio.Phylo.CDAOIO.parse
Bio.Phylo.NeXMLIO.Parser.parse
Bio.Phylo.NeXMLIO.parse
Bio.Phylo.NewickIO.Parser.parse
Bio.Phylo.NewickIO.parse
Bio.Phylo.NexusIO.parse
Bio.Phylo.PhyloXMLIO.Parser.parse
Bio.Phylo.PhyloXMLIO.parse
Bio.Phylo._io.parse
Bio.SCOP.Cla.parse
Bio.SCOP.Des.parse
Bio.SCOP.Dom.parse
Bio.SCOP.Hie.parse
Bio.SCOP.Raf.parse
Bio.SearchIO._legacy.NCBIStandalone.BlastErrorParser.parse
Bio.SearchIO._legacy.NCBIStandalone.BlastParser.parse
Bio.SearchIO._legacy.NCBIStandalone.PSIBlastParser.parse
Bio.SearchIO._legacy.ParserSupport.AbstractParser.parse
Bio.SearchIO.parse
Bio.SeqIO.UniprotIO.Parser.parse
Bio.SeqIO.parse
Bio.Sequencing.Ace.parse
Bio.Sequencing.Phd.parse
Bio.SwissProt.KeyWList.parse
Bio.SwissProt.parse
Bio.UniGene.parse
Bio.Wise.psw.parse
Bio.motifs.parse
Bio.phenotype.parse" class="py-name" href="#" onclick="return doclink('link-236', 'parse', 'link-219');">parse</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt id="link-237" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-237', 'format', 'link-142');">format</a></tt><tt class="py-op">,</tt> <tt id="link-238" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-238', 'alphabet', 'link-195');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L719"></a><tt class="py-lineno"> 719</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L720"></a><tt class="py-lineno"> 720</tt>  <tt class="py-line">        <tt class="py-name">first</tt> <tt class="py-op">=</tt> <tt id="link-239" class="py-name" targets="Method Bio.AlignIO.Interfaces.AlignmentIterator.next()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#next,Method Bio.Blast.ParseBlastTable.BlastTableReader.next()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#next,Method Bio.File.UndoHandle.next()=Bio.File.UndoHandle-class.html#next,Method Bio.GenBank.Iterator.next()=Bio.GenBank.Iterator-class.html#next,Method Bio.Nexus.Nexus.CharBuffer.next()=Bio.Nexus.Nexus.CharBuffer-class.html#next,Method Bio.Nexus.StandardData.StandardData.next()=Bio.Nexus.StandardData.StandardData-class.html#next,Method Bio.PopGen.GenePop.Controller._FileIterator.next()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#next,Method Bio.SearchIO._legacy.NCBIStandalone.Iterator.next()=Bio.SearchIO._legacy.NCBIStandalone.Iterator-class.html#next,Method Bio.SeqIO.Interfaces.SequenceIterator.next()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#next,Method Bio.bgzf.BgzfReader.next()=Bio.bgzf.BgzfReader-class.html#next"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.Nexus.StandardData.StandardData.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SearchIO._legacy.NCBIStandalone.Iterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-239', 'next', 'link-239');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">iterator</tt><tt class="py-op">)</tt> </tt>
<a name="L721"></a><tt class="py-lineno"> 721</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L722"></a><tt class="py-lineno"> 722</tt>  <tt class="py-line">        <tt class="py-name">first</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L723"></a><tt class="py-lineno"> 723</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">first</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L724"></a><tt class="py-lineno"> 724</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"No records found in handle"</tt><tt class="py-op">)</tt> </tt>
<a name="L725"></a><tt class="py-lineno"> 725</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L726"></a><tt class="py-lineno"> 726</tt>  <tt class="py-line">        <tt class="py-name">second</tt> <tt class="py-op">=</tt> <tt id="link-240" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.Nexus.StandardData.StandardData.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SearchIO._legacy.NCBIStandalone.Iterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-240', 'next', 'link-239');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">iterator</tt><tt class="py-op">)</tt> </tt>
<a name="L727"></a><tt class="py-lineno"> 727</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L728"></a><tt class="py-lineno"> 728</tt>  <tt class="py-line">        <tt class="py-name">second</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L729"></a><tt class="py-lineno"> 729</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">second</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L730"></a><tt class="py-lineno"> 730</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"More than one record found in handle"</tt><tt class="py-op">)</tt> </tt>
<a name="L731"></a><tt class="py-lineno"> 731</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">first</tt> </tt>
</div><a name="L732"></a><tt class="py-lineno"> 732</tt>  <tt class="py-line"> </tt>
<a name="L733"></a><tt class="py-lineno"> 733</tt>  <tt class="py-line"> </tt>
<a name="to_dict"></a><div id="to_dict-def"><a name="L734"></a><tt class="py-lineno"> 734</tt> <a class="py-toggle" href="#" id="to_dict-toggle" onclick="return toggle('to_dict');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO-module.html#to_dict">to_dict</a><tt class="py-op">(</tt><tt class="py-param">sequences</tt><tt class="py-op">,</tt> <tt class="py-param">key_function</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="to_dict-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="to_dict-expanded"><a name="L735"></a><tt class="py-lineno"> 735</tt>  <tt class="py-line">    <tt class="py-docstring">"""Turn a sequence iterator or list into a dictionary.</tt> </tt>
<a name="L736"></a><tt class="py-lineno"> 736</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L737"></a><tt class="py-lineno"> 737</tt>  <tt class="py-line"><tt class="py-docstring">    Arguments:</tt> </tt>
<a name="L738"></a><tt class="py-lineno"> 738</tt>  <tt class="py-line"><tt class="py-docstring">     - sequences  - An iterator that returns SeqRecord objects,</tt> </tt>
<a name="L739"></a><tt class="py-lineno"> 739</tt>  <tt class="py-line"><tt class="py-docstring">       or simply a list of SeqRecord objects.</tt> </tt>
<a name="L740"></a><tt class="py-lineno"> 740</tt>  <tt class="py-line"><tt class="py-docstring">     - key_function - Optional callback function which when given a</tt> </tt>
<a name="L741"></a><tt class="py-lineno"> 741</tt>  <tt class="py-line"><tt class="py-docstring">       SeqRecord should return a unique key for the dictionary.</tt> </tt>
<a name="L742"></a><tt class="py-lineno"> 742</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L743"></a><tt class="py-lineno"> 743</tt>  <tt class="py-line"><tt class="py-docstring">    e.g. key_function = lambda rec : rec.name</tt> </tt>
<a name="L744"></a><tt class="py-lineno"> 744</tt>  <tt class="py-line"><tt class="py-docstring">    or,  key_function = lambda rec : rec.description.split()[0]</tt> </tt>
<a name="L745"></a><tt class="py-lineno"> 745</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L746"></a><tt class="py-lineno"> 746</tt>  <tt class="py-line"><tt class="py-docstring">    If key_function is omitted then record.id is used, on the assumption</tt> </tt>
<a name="L747"></a><tt class="py-lineno"> 747</tt>  <tt class="py-line"><tt class="py-docstring">    that the records objects returned are SeqRecords with a unique id.</tt> </tt>
<a name="L748"></a><tt class="py-lineno"> 748</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L749"></a><tt class="py-lineno"> 749</tt>  <tt class="py-line"><tt class="py-docstring">    If there are duplicate keys, an error is raised.</tt> </tt>
<a name="L750"></a><tt class="py-lineno"> 750</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L751"></a><tt class="py-lineno"> 751</tt>  <tt class="py-line"><tt class="py-docstring">    Since Python 3.7, the default dict class maintains key order, meaning</tt> </tt>
<a name="L752"></a><tt class="py-lineno"> 752</tt>  <tt class="py-line"><tt class="py-docstring">    this dictionary will reflect the order of records given to it. For</tt> </tt>
<a name="L753"></a><tt class="py-lineno"> 753</tt>  <tt class="py-line"><tt class="py-docstring">    CPython, this was already implemented in 3.6.</tt> </tt>
<a name="L754"></a><tt class="py-lineno"> 754</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L755"></a><tt class="py-lineno"> 755</tt>  <tt class="py-line"><tt class="py-docstring">    As of Biopython 1.73, we explicitly use OrderedDict for CPython older</tt> </tt>
<a name="L756"></a><tt class="py-lineno"> 756</tt>  <tt class="py-line"><tt class="py-docstring">    than 3.6 (and for other Python older than 3.7) so that you can always</tt> </tt>
<a name="L757"></a><tt class="py-lineno"> 757</tt>  <tt class="py-line"><tt class="py-docstring">    assume the record order is preserved.</tt> </tt>
<a name="L758"></a><tt class="py-lineno"> 758</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L759"></a><tt class="py-lineno"> 759</tt>  <tt class="py-line"><tt class="py-docstring">    Example usage, defaulting to using the record.id as key:</tt> </tt>
<a name="L760"></a><tt class="py-lineno"> 760</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L761"></a><tt class="py-lineno"> 761</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L762"></a><tt class="py-lineno"> 762</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; filename = "GenBank/cor6_6.gb"</tt> </tt>
<a name="L763"></a><tt class="py-lineno"> 763</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; format = "genbank"</tt> </tt>
<a name="L764"></a><tt class="py-lineno"> 764</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; id_dict = SeqIO.to_dict(SeqIO.parse(filename, format))</tt> </tt>
<a name="L765"></a><tt class="py-lineno"> 765</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(list(id_dict))</tt> </tt>
<a name="L766"></a><tt class="py-lineno"> 766</tt>  <tt class="py-line"><tt class="py-docstring">    ['X55053.1', 'X62281.1', 'M81224.1', 'AJ237582.1', 'L31939.1', 'AF297471.1']</tt> </tt>
<a name="L767"></a><tt class="py-lineno"> 767</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(id_dict["L31939.1"].description)</tt> </tt>
<a name="L768"></a><tt class="py-lineno"> 768</tt>  <tt class="py-line"><tt class="py-docstring">    Brassica rapa (clone bif72) kin mRNA, complete cds</tt> </tt>
<a name="L769"></a><tt class="py-lineno"> 769</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L770"></a><tt class="py-lineno"> 770</tt>  <tt class="py-line"><tt class="py-docstring">    A more complex example, using the key_function argument in order to</tt> </tt>
<a name="L771"></a><tt class="py-lineno"> 771</tt>  <tt class="py-line"><tt class="py-docstring">    use a sequence checksum as the dictionary key:</tt> </tt>
<a name="L772"></a><tt class="py-lineno"> 772</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L773"></a><tt class="py-lineno"> 773</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L774"></a><tt class="py-lineno"> 774</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio.SeqUtils.CheckSum import seguid</tt> </tt>
<a name="L775"></a><tt class="py-lineno"> 775</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; filename = "GenBank/cor6_6.gb"</tt> </tt>
<a name="L776"></a><tt class="py-lineno"> 776</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; format = "genbank"</tt> </tt>
<a name="L777"></a><tt class="py-lineno"> 777</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; seguid_dict = SeqIO.to_dict(SeqIO.parse(filename, format),</tt> </tt>
<a name="L778"></a><tt class="py-lineno"> 778</tt>  <tt class="py-line"><tt class="py-docstring">    ...               key_function = lambda rec : seguid(rec.seq))</tt> </tt>
<a name="L779"></a><tt class="py-lineno"> 779</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; for key, record in sorted(seguid_dict.items()):</tt> </tt>
<a name="L780"></a><tt class="py-lineno"> 780</tt>  <tt class="py-line"><tt class="py-docstring">    ...     print("%s %s" % (key, record.id))</tt> </tt>
<a name="L781"></a><tt class="py-lineno"> 781</tt>  <tt class="py-line"><tt class="py-docstring">    /wQvmrl87QWcm9llO4/efg23Vgg AJ237582.1</tt> </tt>
<a name="L782"></a><tt class="py-lineno"> 782</tt>  <tt class="py-line"><tt class="py-docstring">    BUg6YxXSKWEcFFH0L08JzaLGhQs L31939.1</tt> </tt>
<a name="L783"></a><tt class="py-lineno"> 783</tt>  <tt class="py-line"><tt class="py-docstring">    SabZaA4V2eLE9/2Fm5FnyYy07J4 X55053.1</tt> </tt>
<a name="L784"></a><tt class="py-lineno"> 784</tt>  <tt class="py-line"><tt class="py-docstring">    TtWsXo45S3ZclIBy4X/WJc39+CY M81224.1</tt> </tt>
<a name="L785"></a><tt class="py-lineno"> 785</tt>  <tt class="py-line"><tt class="py-docstring">    l7gjJFE6W/S1jJn5+1ASrUKW/FA X62281.1</tt> </tt>
<a name="L786"></a><tt class="py-lineno"> 786</tt>  <tt class="py-line"><tt class="py-docstring">    uVEYeAQSV5EDQOnFoeMmVea+Oow AF297471.1</tt> </tt>
<a name="L787"></a><tt class="py-lineno"> 787</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L788"></a><tt class="py-lineno"> 788</tt>  <tt class="py-line"><tt class="py-docstring">    This approach is not suitable for very large sets of sequences, as all</tt> </tt>
<a name="L789"></a><tt class="py-lineno"> 789</tt>  <tt class="py-line"><tt class="py-docstring">    the SeqRecord objects are held in memory. Instead, consider using the</tt> </tt>
<a name="L790"></a><tt class="py-lineno"> 790</tt>  <tt class="py-line"><tt class="py-docstring">    Bio.SeqIO.index() function (if it supports your particular file format).</tt> </tt>
<a name="L791"></a><tt class="py-lineno"> 791</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L792"></a><tt class="py-lineno"> 792</tt>  <tt class="py-line"><tt class="py-docstring">    Since Python 3.6, the default dict class maintains key order, meaning</tt> </tt>
<a name="L793"></a><tt class="py-lineno"> 793</tt>  <tt class="py-line"><tt class="py-docstring">    this dictionary will reflect the order of records given to it. As of</tt> </tt>
<a name="L794"></a><tt class="py-lineno"> 794</tt>  <tt class="py-line"><tt class="py-docstring">    Biopython 1.72, on older versions of Python we explicitly use an</tt> </tt>
<a name="L795"></a><tt class="py-lineno"> 795</tt>  <tt class="py-line"><tt class="py-docstring">    OrderedDict so that you can always assume the record order is preserved.</tt> </tt>
<a name="L796"></a><tt class="py-lineno"> 796</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L797"></a><tt class="py-lineno"> 797</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">key_function</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L798"></a><tt class="py-lineno"> 798</tt>  <tt class="py-line">        <tt class="py-name">key_function</tt> <tt class="py-op">=</tt> <tt class="py-keyword">lambda</tt> <tt class="py-name">rec</tt><tt class="py-op">:</tt> <tt class="py-name">rec</tt><tt class="py-op">.</tt><tt id="link-241" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Variable Bio.PDB.Entity.Entity.id=Bio.PDB.Entity.Entity-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-241', 'id', 'link-241');">id</a></tt> </tt>
<a name="L799"></a><tt class="py-lineno"> 799</tt>  <tt class="py-line"> </tt>
<a name="L800"></a><tt class="py-lineno"> 800</tt>  <tt class="py-line">    <tt class="py-name">d</tt> <tt class="py-op">=</tt> <tt class="py-name">_dict</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L801"></a><tt class="py-lineno"> 801</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">sequences</tt><tt class="py-op">:</tt> </tt>
<a name="L802"></a><tt class="py-lineno"> 802</tt>  <tt class="py-line">        <tt class="py-name">key</tt> <tt class="py-op">=</tt> <tt class="py-name">key_function</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L803"></a><tt class="py-lineno"> 803</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">d</tt><tt class="py-op">:</tt> </tt>
<a name="L804"></a><tt class="py-lineno"> 804</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Duplicate key '%s'"</tt> <tt class="py-op">%</tt> <tt class="py-name">key</tt><tt class="py-op">)</tt> </tt>
<a name="L805"></a><tt class="py-lineno"> 805</tt>  <tt class="py-line">        <tt class="py-name">d</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt> </tt>
<a name="L806"></a><tt class="py-lineno"> 806</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">d</tt> </tt>
</div><a name="L807"></a><tt class="py-lineno"> 807</tt>  <tt class="py-line"> </tt>
<a name="L808"></a><tt class="py-lineno"> 808</tt>  <tt class="py-line"> </tt>
<a name="index"></a><div id="index-def"><a name="L809"></a><tt class="py-lineno"> 809</tt> <a class="py-toggle" href="#" id="index-toggle" onclick="return toggle('index');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO-module.html#index">index</a><tt class="py-op">(</tt><tt class="py-param">filename</tt><tt class="py-op">,</tt> <tt class="py-param">format</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">key_function</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="index-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="index-expanded"><a name="L810"></a><tt class="py-lineno"> 810</tt>  <tt class="py-line">    <tt class="py-docstring">"""Indexes a sequence file and returns a dictionary like object.</tt> </tt>
<a name="L811"></a><tt class="py-lineno"> 811</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L812"></a><tt class="py-lineno"> 812</tt>  <tt class="py-line"><tt class="py-docstring">    Arguments:</tt> </tt>
<a name="L813"></a><tt class="py-lineno"> 813</tt>  <tt class="py-line"><tt class="py-docstring">     - filename - string giving name of file to be indexed</tt> </tt>
<a name="L814"></a><tt class="py-lineno"> 814</tt>  <tt class="py-line"><tt class="py-docstring">     - format   - lower case string describing the file format</tt> </tt>
<a name="L815"></a><tt class="py-lineno"> 815</tt>  <tt class="py-line"><tt class="py-docstring">     - alphabet - optional Alphabet object, useful when the sequence type</tt> </tt>
<a name="L816"></a><tt class="py-lineno"> 816</tt>  <tt class="py-line"><tt class="py-docstring">       cannot be automatically inferred from the file itself</tt> </tt>
<a name="L817"></a><tt class="py-lineno"> 817</tt>  <tt class="py-line"><tt class="py-docstring">       (e.g. format="fasta" or "tab")</tt> </tt>
<a name="L818"></a><tt class="py-lineno"> 818</tt>  <tt class="py-line"><tt class="py-docstring">     - key_function - Optional callback function which when given a</tt> </tt>
<a name="L819"></a><tt class="py-lineno"> 819</tt>  <tt class="py-line"><tt class="py-docstring">       SeqRecord identifier string should return a unique key for the</tt> </tt>
<a name="L820"></a><tt class="py-lineno"> 820</tt>  <tt class="py-line"><tt class="py-docstring">       dictionary.</tt> </tt>
<a name="L821"></a><tt class="py-lineno"> 821</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L822"></a><tt class="py-lineno"> 822</tt>  <tt class="py-line"><tt class="py-docstring">    This indexing function will return a dictionary like object, giving the</tt> </tt>
<a name="L823"></a><tt class="py-lineno"> 823</tt>  <tt class="py-line"><tt class="py-docstring">    SeqRecord objects as values.</tt> </tt>
<a name="L824"></a><tt class="py-lineno"> 824</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L825"></a><tt class="py-lineno"> 825</tt>  <tt class="py-line"><tt class="py-docstring">    As of Biopython 1.69, this will preserve the ordering of the records in</tt> </tt>
<a name="L826"></a><tt class="py-lineno"> 826</tt>  <tt class="py-line"><tt class="py-docstring">    file when iterating over the entries.</tt> </tt>
<a name="L827"></a><tt class="py-lineno"> 827</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L828"></a><tt class="py-lineno"> 828</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L829"></a><tt class="py-lineno"> 829</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; records = SeqIO.index("Quality/example.fastq", "fastq")</tt> </tt>
<a name="L830"></a><tt class="py-lineno"> 830</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; len(records)</tt> </tt>
<a name="L831"></a><tt class="py-lineno"> 831</tt>  <tt class="py-line"><tt class="py-docstring">    3</tt> </tt>
<a name="L832"></a><tt class="py-lineno"> 832</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; list(records)  # make a list of the keys</tt> </tt>
<a name="L833"></a><tt class="py-lineno"> 833</tt>  <tt class="py-line"><tt class="py-docstring">    ['EAS54_6_R1_2_1_413_324', 'EAS54_6_R1_2_1_540_792', 'EAS54_6_R1_2_1_443_348']</tt> </tt>
<a name="L834"></a><tt class="py-lineno"> 834</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(records["EAS54_6_R1_2_1_540_792"].format("fasta"))</tt> </tt>
<a name="L835"></a><tt class="py-lineno"> 835</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;EAS54_6_R1_2_1_540_792</tt> </tt>
<a name="L836"></a><tt class="py-lineno"> 836</tt>  <tt class="py-line"><tt class="py-docstring">    TTGGCAGGCCAAGGCCGATGGATCA</tt> </tt>
<a name="L837"></a><tt class="py-lineno"> 837</tt>  <tt class="py-line"><tt class="py-docstring">    &lt;BLANKLINE&gt;</tt> </tt>
<a name="L838"></a><tt class="py-lineno"> 838</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; "EAS54_6_R1_2_1_540_792" in records</tt> </tt>
<a name="L839"></a><tt class="py-lineno"> 839</tt>  <tt class="py-line"><tt class="py-docstring">    True</tt> </tt>
<a name="L840"></a><tt class="py-lineno"> 840</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(records.get("Missing", None))</tt> </tt>
<a name="L841"></a><tt class="py-lineno"> 841</tt>  <tt class="py-line"><tt class="py-docstring">    None</tt> </tt>
<a name="L842"></a><tt class="py-lineno"> 842</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; records.close()</tt> </tt>
<a name="L843"></a><tt class="py-lineno"> 843</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L844"></a><tt class="py-lineno"> 844</tt>  <tt class="py-line"><tt class="py-docstring">    If the file is BGZF compressed, this is detected automatically. Ordinary</tt> </tt>
<a name="L845"></a><tt class="py-lineno"> 845</tt>  <tt class="py-line"><tt class="py-docstring">    GZIP files are not supported:</tt> </tt>
<a name="L846"></a><tt class="py-lineno"> 846</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L847"></a><tt class="py-lineno"> 847</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L848"></a><tt class="py-lineno"> 848</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; records = SeqIO.index("Quality/example.fastq.bgz", "fastq")</tt> </tt>
<a name="L849"></a><tt class="py-lineno"> 849</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; len(records)</tt> </tt>
<a name="L850"></a><tt class="py-lineno"> 850</tt>  <tt class="py-line"><tt class="py-docstring">    3</tt> </tt>
<a name="L851"></a><tt class="py-lineno"> 851</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(records["EAS54_6_R1_2_1_540_792"].seq)</tt> </tt>
<a name="L852"></a><tt class="py-lineno"> 852</tt>  <tt class="py-line"><tt class="py-docstring">    TTGGCAGGCCAAGGCCGATGGATCA</tt> </tt>
<a name="L853"></a><tt class="py-lineno"> 853</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; records.close()</tt> </tt>
<a name="L854"></a><tt class="py-lineno"> 854</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L855"></a><tt class="py-lineno"> 855</tt>  <tt class="py-line"><tt class="py-docstring">    Note that this pseudo dictionary will not support all the methods of a</tt> </tt>
<a name="L856"></a><tt class="py-lineno"> 856</tt>  <tt class="py-line"><tt class="py-docstring">    true Python dictionary, for example values() is not defined as in Python 2</tt> </tt>
<a name="L857"></a><tt class="py-lineno"> 857</tt>  <tt class="py-line"><tt class="py-docstring">    since this would require loading all of the records into memory at once.</tt> </tt>
<a name="L858"></a><tt class="py-lineno"> 858</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L859"></a><tt class="py-lineno"> 859</tt>  <tt class="py-line"><tt class="py-docstring">    When you call the index function, it will scan through the file, noting</tt> </tt>
<a name="L860"></a><tt class="py-lineno"> 860</tt>  <tt class="py-line"><tt class="py-docstring">    the location of each record. When you access a particular record via the</tt> </tt>
<a name="L861"></a><tt class="py-lineno"> 861</tt>  <tt class="py-line"><tt class="py-docstring">    dictionary methods, the code will jump to the appropriate part of the</tt> </tt>
<a name="L862"></a><tt class="py-lineno"> 862</tt>  <tt class="py-line"><tt class="py-docstring">    file and then parse that section into a SeqRecord.</tt> </tt>
<a name="L863"></a><tt class="py-lineno"> 863</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L864"></a><tt class="py-lineno"> 864</tt>  <tt class="py-line"><tt class="py-docstring">    Note that not all the input formats supported by Bio.SeqIO can be used</tt> </tt>
<a name="L865"></a><tt class="py-lineno"> 865</tt>  <tt class="py-line"><tt class="py-docstring">    with this index function. It is designed to work only with sequential</tt> </tt>
<a name="L866"></a><tt class="py-lineno"> 866</tt>  <tt class="py-line"><tt class="py-docstring">    file formats (e.g. "fasta", "gb", "fastq") and is not suitable for any</tt> </tt>
<a name="L867"></a><tt class="py-lineno"> 867</tt>  <tt class="py-line"><tt class="py-docstring">    interlaced file format (e.g. alignment formats such as "clustal").</tt> </tt>
<a name="L868"></a><tt class="py-lineno"> 868</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L869"></a><tt class="py-lineno"> 869</tt>  <tt class="py-line"><tt class="py-docstring">    For small files, it may be more efficient to use an in memory Python</tt> </tt>
<a name="L870"></a><tt class="py-lineno"> 870</tt>  <tt class="py-line"><tt class="py-docstring">    dictionary, e.g.</tt> </tt>
<a name="L871"></a><tt class="py-lineno"> 871</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L872"></a><tt class="py-lineno"> 872</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L873"></a><tt class="py-lineno"> 873</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; records = SeqIO.to_dict(SeqIO.parse("Quality/example.fastq", "fastq"))</tt> </tt>
<a name="L874"></a><tt class="py-lineno"> 874</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; len(records)</tt> </tt>
<a name="L875"></a><tt class="py-lineno"> 875</tt>  <tt class="py-line"><tt class="py-docstring">    3</tt> </tt>
<a name="L876"></a><tt class="py-lineno"> 876</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; list(records)  # make a list of the keys</tt> </tt>
<a name="L877"></a><tt class="py-lineno"> 877</tt>  <tt class="py-line"><tt class="py-docstring">    ['EAS54_6_R1_2_1_413_324', 'EAS54_6_R1_2_1_540_792', 'EAS54_6_R1_2_1_443_348']</tt> </tt>
<a name="L878"></a><tt class="py-lineno"> 878</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(records["EAS54_6_R1_2_1_540_792"].format("fasta"))</tt> </tt>
<a name="L879"></a><tt class="py-lineno"> 879</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;EAS54_6_R1_2_1_540_792</tt> </tt>
<a name="L880"></a><tt class="py-lineno"> 880</tt>  <tt class="py-line"><tt class="py-docstring">    TTGGCAGGCCAAGGCCGATGGATCA</tt> </tt>
<a name="L881"></a><tt class="py-lineno"> 881</tt>  <tt class="py-line"><tt class="py-docstring">    &lt;BLANKLINE&gt;</tt> </tt>
<a name="L882"></a><tt class="py-lineno"> 882</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L883"></a><tt class="py-lineno"> 883</tt>  <tt class="py-line"><tt class="py-docstring">    As with the to_dict() function, by default the id string of each record</tt> </tt>
<a name="L884"></a><tt class="py-lineno"> 884</tt>  <tt class="py-line"><tt class="py-docstring">    is used as the key. You can specify a callback function to transform</tt> </tt>
<a name="L885"></a><tt class="py-lineno"> 885</tt>  <tt class="py-line"><tt class="py-docstring">    this (the record identifier string) into your preferred key. For example:</tt> </tt>
<a name="L886"></a><tt class="py-lineno"> 886</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L887"></a><tt class="py-lineno"> 887</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L888"></a><tt class="py-lineno"> 888</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; def make_tuple(identifier):</tt> </tt>
<a name="L889"></a><tt class="py-lineno"> 889</tt>  <tt class="py-line"><tt class="py-docstring">    ...     parts = identifier.split("_")</tt> </tt>
<a name="L890"></a><tt class="py-lineno"> 890</tt>  <tt class="py-line"><tt class="py-docstring">    ...     return int(parts[-2]), int(parts[-1])</tt> </tt>
<a name="L891"></a><tt class="py-lineno"> 891</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; records = SeqIO.index("Quality/example.fastq", "fastq",</tt> </tt>
<a name="L892"></a><tt class="py-lineno"> 892</tt>  <tt class="py-line"><tt class="py-docstring">    ...                       key_function=make_tuple)</tt> </tt>
<a name="L893"></a><tt class="py-lineno"> 893</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; len(records)</tt> </tt>
<a name="L894"></a><tt class="py-lineno"> 894</tt>  <tt class="py-line"><tt class="py-docstring">    3</tt> </tt>
<a name="L895"></a><tt class="py-lineno"> 895</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; list(records)  # make a list of the keys</tt> </tt>
<a name="L896"></a><tt class="py-lineno"> 896</tt>  <tt class="py-line"><tt class="py-docstring">    [(413, 324), (540, 792), (443, 348)]</tt> </tt>
<a name="L897"></a><tt class="py-lineno"> 897</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(records[(540, 792)].format("fasta"))</tt> </tt>
<a name="L898"></a><tt class="py-lineno"> 898</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;EAS54_6_R1_2_1_540_792</tt> </tt>
<a name="L899"></a><tt class="py-lineno"> 899</tt>  <tt class="py-line"><tt class="py-docstring">    TTGGCAGGCCAAGGCCGATGGATCA</tt> </tt>
<a name="L900"></a><tt class="py-lineno"> 900</tt>  <tt class="py-line"><tt class="py-docstring">    &lt;BLANKLINE&gt;</tt> </tt>
<a name="L901"></a><tt class="py-lineno"> 901</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; (540, 792) in records</tt> </tt>
<a name="L902"></a><tt class="py-lineno"> 902</tt>  <tt class="py-line"><tt class="py-docstring">    True</tt> </tt>
<a name="L903"></a><tt class="py-lineno"> 903</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; "EAS54_6_R1_2_1_540_792" in records</tt> </tt>
<a name="L904"></a><tt class="py-lineno"> 904</tt>  <tt class="py-line"><tt class="py-docstring">    False</tt> </tt>
<a name="L905"></a><tt class="py-lineno"> 905</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(records.get("Missing", None))</tt> </tt>
<a name="L906"></a><tt class="py-lineno"> 906</tt>  <tt class="py-line"><tt class="py-docstring">    None</tt> </tt>
<a name="L907"></a><tt class="py-lineno"> 907</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; records.close()</tt> </tt>
<a name="L908"></a><tt class="py-lineno"> 908</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L909"></a><tt class="py-lineno"> 909</tt>  <tt class="py-line"><tt class="py-docstring">    Another common use case would be indexing an NCBI style FASTA file,</tt> </tt>
<a name="L910"></a><tt class="py-lineno"> 910</tt>  <tt class="py-line"><tt class="py-docstring">    where you might want to extract the GI number from the FASTA identifier</tt> </tt>
<a name="L911"></a><tt class="py-lineno"> 911</tt>  <tt class="py-line"><tt class="py-docstring">    to use as the dictionary key.</tt> </tt>
<a name="L912"></a><tt class="py-lineno"> 912</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L913"></a><tt class="py-lineno"> 913</tt>  <tt class="py-line"><tt class="py-docstring">    Notice that unlike the to_dict() function, here the key_function does</tt> </tt>
<a name="L914"></a><tt class="py-lineno"> 914</tt>  <tt class="py-line"><tt class="py-docstring">    not get given the full SeqRecord to use to generate the key. Doing so</tt> </tt>
<a name="L915"></a><tt class="py-lineno"> 915</tt>  <tt class="py-line"><tt class="py-docstring">    would impose a severe performance penalty as it would require the file</tt> </tt>
<a name="L916"></a><tt class="py-lineno"> 916</tt>  <tt class="py-line"><tt class="py-docstring">    to be completely parsed while building the index. Right now this is</tt> </tt>
<a name="L917"></a><tt class="py-lineno"> 917</tt>  <tt class="py-line"><tt class="py-docstring">    usually avoided.</tt> </tt>
<a name="L918"></a><tt class="py-lineno"> 918</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L919"></a><tt class="py-lineno"> 919</tt>  <tt class="py-line"><tt class="py-docstring">    See Also: Bio.SeqIO.index_db() and Bio.SeqIO.to_dict()</tt> </tt>
<a name="L920"></a><tt class="py-lineno"> 920</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L921"></a><tt class="py-lineno"> 921</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L922"></a><tt class="py-lineno"> 922</tt>  <tt class="py-line">    <tt class="py-comment"># Try and give helpful error messages:</tt> </tt>
<a name="L923"></a><tt class="py-lineno"> 923</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L924"></a><tt class="py-lineno"> 924</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"Need a filename (not a handle)"</tt><tt class="py-op">)</tt> </tt>
<a name="L925"></a><tt class="py-lineno"> 925</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-242" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-242', 'format', 'link-142');">format</a></tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L926"></a><tt class="py-lineno"> 926</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"Need a string for the file format (lower case)"</tt><tt class="py-op">)</tt> </tt>
<a name="L927"></a><tt class="py-lineno"> 927</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-243" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-243', 'format', 'link-142');">format</a></tt><tt class="py-op">:</tt> </tt>
<a name="L928"></a><tt class="py-lineno"> 928</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Format required (lower case string)"</tt><tt class="py-op">)</tt> </tt>
<a name="L929"></a><tt class="py-lineno"> 929</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt id="link-244" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-244', 'format', 'link-142');">format</a></tt> <tt class="py-op">!=</tt> <tt id="link-245" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-245', 'format', 'link-142');">format</a></tt><tt class="py-op">.</tt><tt id="link-246" class="py-name"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-246', 'lower', 'link-146');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L930"></a><tt class="py-lineno"> 930</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Format string '%s' should be lower case"</tt> <tt class="py-op">%</tt> <tt id="link-247" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-247', 'format', 'link-142');">format</a></tt><tt class="py-op">)</tt> </tt>
<a name="L931"></a><tt class="py-lineno"> 931</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt id="link-248" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-248', 'alphabet', 'link-195');">alphabet</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt class="py-op">(</tt><tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-249" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-249', 'alphabet', 'link-195');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-250" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-250', 'Alphabet', 'link-12');">Alphabet</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> </tt>
<a name="L932"></a><tt class="py-lineno"> 932</tt>  <tt class="py-line">                                     <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-251" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-251', 'alphabet', 'link-195');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-252" class="py-name"><a title="Bio.Alphabet.AlphabetEncoder" class="py-name" href="#" onclick="return doclink('link-252', 'AlphabetEncoder', 'link-14');">AlphabetEncoder</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L933"></a><tt class="py-lineno"> 933</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid alphabet, %r"</tt> <tt class="py-op">%</tt> <tt id="link-253" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-253', 'alphabet', 'link-195');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L934"></a><tt class="py-lineno"> 934</tt>  <tt class="py-line"> </tt>
<a name="L935"></a><tt class="py-lineno"> 935</tt>  <tt class="py-line">    <tt class="py-comment"># Map the file format to a sequence iterator:</tt> </tt>
<a name="L936"></a><tt class="py-lineno"> 936</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt class="py-op">.</tt><tt id="link-254" class="py-name" targets="Module Bio.SearchIO._index=Bio.SearchIO._index-module.html,Module Bio.SeqIO._index=Bio.SeqIO._index-module.html"><a title="Bio.SearchIO._index
Bio.SeqIO._index" class="py-name" href="#" onclick="return doclink('link-254', '_index', 'link-254');">_index</a></tt> <tt class="py-keyword">import</tt> <tt id="link-255" class="py-name" targets="Variable Bio.SeqIO._index._FormatToRandomAccess=Bio.SeqIO._index-module.html#_FormatToRandomAccess"><a title="Bio.SeqIO._index._FormatToRandomAccess" class="py-name" href="#" onclick="return doclink('link-255', '_FormatToRandomAccess', 'link-255');">_FormatToRandomAccess</a></tt>  <tt class="py-comment"># Lazy import</tt> </tt>
<a name="L937"></a><tt class="py-lineno"> 937</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-256" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-256', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-257" class="py-name"><a title="Bio.File" class="py-name" href="#" onclick="return doclink('link-257', 'File', 'link-3');">File</a></tt> <tt class="py-keyword">import</tt> <tt id="link-258" class="py-name" targets="Class Bio.File._IndexedSeqFileDict=Bio.File._IndexedSeqFileDict-class.html"><a title="Bio.File._IndexedSeqFileDict" class="py-name" href="#" onclick="return doclink('link-258', '_IndexedSeqFileDict', 'link-258');">_IndexedSeqFileDict</a></tt> </tt>
<a name="L938"></a><tt class="py-lineno"> 938</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L939"></a><tt class="py-lineno"> 939</tt>  <tt class="py-line">        <tt class="py-name">proxy_class</tt> <tt class="py-op">=</tt> <tt id="link-259" class="py-name"><a title="Bio.SeqIO._index._FormatToRandomAccess" class="py-name" href="#" onclick="return doclink('link-259', '_FormatToRandomAccess', 'link-255');">_FormatToRandomAccess</a></tt><tt class="py-op">[</tt><tt id="link-260" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-260', 'format', 'link-142');">format</a></tt><tt class="py-op">]</tt> </tt>
<a name="L940"></a><tt class="py-lineno"> 940</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L941"></a><tt class="py-lineno"> 941</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unsupported format %r"</tt> <tt class="py-op">%</tt> <tt id="link-261" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-261', 'format', 'link-142');">format</a></tt><tt class="py-op">)</tt> </tt>
<a name="L942"></a><tt class="py-lineno"> 942</tt>  <tt class="py-line">    <tt class="py-name">repr</tt> <tt class="py-op">=</tt> <tt class="py-string">"SeqIO.index(%r, %r, alphabet=%r, key_function=%r)"</tt> \ </tt>
<a name="L943"></a><tt class="py-lineno"> 943</tt>  <tt class="py-line">        <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt id="link-262" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-262', 'format', 'link-142');">format</a></tt><tt class="py-op">,</tt> <tt id="link-263" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-263', 'alphabet', 'link-195');">alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-name">key_function</tt><tt class="py-op">)</tt> </tt>
<a name="L944"></a><tt class="py-lineno"> 944</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-264" class="py-name"><a title="Bio.File._IndexedSeqFileDict" class="py-name" href="#" onclick="return doclink('link-264', '_IndexedSeqFileDict', 'link-258');">_IndexedSeqFileDict</a></tt><tt class="py-op">(</tt><tt class="py-name">proxy_class</tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt id="link-265" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-265', 'format', 'link-142');">format</a></tt><tt class="py-op">,</tt> <tt id="link-266" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-266', 'alphabet', 'link-195');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L945"></a><tt class="py-lineno"> 945</tt>  <tt class="py-line">                               <tt class="py-name">key_function</tt><tt class="py-op">,</tt> <tt class="py-name">repr</tt><tt class="py-op">,</tt> <tt class="py-string">"SeqRecord"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L946"></a><tt class="py-lineno"> 946</tt>  <tt class="py-line"> </tt>
<a name="L947"></a><tt class="py-lineno"> 947</tt>  <tt class="py-line"> </tt>
<a name="index_db"></a><div id="index_db-def"><a name="L948"></a><tt class="py-lineno"> 948</tt> <a class="py-toggle" href="#" id="index_db-toggle" onclick="return toggle('index_db');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO-module.html#index_db">index_db</a><tt class="py-op">(</tt><tt class="py-param">index_filename</tt><tt class="py-op">,</tt> <tt class="py-param">filenames</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">format</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L949"></a><tt class="py-lineno"> 949</tt>  <tt class="py-line">             <tt class="py-param">key_function</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="index_db-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="index_db-expanded"><a name="L950"></a><tt class="py-lineno"> 950</tt>  <tt class="py-line">    <tt class="py-docstring">"""Index several sequence files and return a dictionary like object.</tt> </tt>
<a name="L951"></a><tt class="py-lineno"> 951</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L952"></a><tt class="py-lineno"> 952</tt>  <tt class="py-line"><tt class="py-docstring">    The index is stored in an SQLite database rather than in memory (as in the</tt> </tt>
<a name="L953"></a><tt class="py-lineno"> 953</tt>  <tt class="py-line"><tt class="py-docstring">    Bio.SeqIO.index(...) function).</tt> </tt>
<a name="L954"></a><tt class="py-lineno"> 954</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L955"></a><tt class="py-lineno"> 955</tt>  <tt class="py-line"><tt class="py-docstring">    Arguments:</tt> </tt>
<a name="L956"></a><tt class="py-lineno"> 956</tt>  <tt class="py-line"><tt class="py-docstring">     - index_filename - Where to store the SQLite index</tt> </tt>
<a name="L957"></a><tt class="py-lineno"> 957</tt>  <tt class="py-line"><tt class="py-docstring">     - filenames - list of strings specifying file(s) to be indexed, or when</tt> </tt>
<a name="L958"></a><tt class="py-lineno"> 958</tt>  <tt class="py-line"><tt class="py-docstring">       indexing a single file this can be given as a string.</tt> </tt>
<a name="L959"></a><tt class="py-lineno"> 959</tt>  <tt class="py-line"><tt class="py-docstring">       (optional if reloading an existing index, but must match)</tt> </tt>
<a name="L960"></a><tt class="py-lineno"> 960</tt>  <tt class="py-line"><tt class="py-docstring">     - format   - lower case string describing the file format</tt> </tt>
<a name="L961"></a><tt class="py-lineno"> 961</tt>  <tt class="py-line"><tt class="py-docstring">       (optional if reloading an existing index, but must match)</tt> </tt>
<a name="L962"></a><tt class="py-lineno"> 962</tt>  <tt class="py-line"><tt class="py-docstring">     - alphabet - optional Alphabet object, useful when the sequence type</tt> </tt>
<a name="L963"></a><tt class="py-lineno"> 963</tt>  <tt class="py-line"><tt class="py-docstring">       cannot be automatically inferred from the file itself</tt> </tt>
<a name="L964"></a><tt class="py-lineno"> 964</tt>  <tt class="py-line"><tt class="py-docstring">       (e.g. format="fasta" or "tab")</tt> </tt>
<a name="L965"></a><tt class="py-lineno"> 965</tt>  <tt class="py-line"><tt class="py-docstring">     - key_function - Optional callback function which when given a</tt> </tt>
<a name="L966"></a><tt class="py-lineno"> 966</tt>  <tt class="py-line"><tt class="py-docstring">       SeqRecord identifier string should return a unique</tt> </tt>
<a name="L967"></a><tt class="py-lineno"> 967</tt>  <tt class="py-line"><tt class="py-docstring">       key for the dictionary.</tt> </tt>
<a name="L968"></a><tt class="py-lineno"> 968</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L969"></a><tt class="py-lineno"> 969</tt>  <tt class="py-line"><tt class="py-docstring">    This indexing function will return a dictionary like object, giving the</tt> </tt>
<a name="L970"></a><tt class="py-lineno"> 970</tt>  <tt class="py-line"><tt class="py-docstring">    SeqRecord objects as values:</tt> </tt>
<a name="L971"></a><tt class="py-lineno"> 971</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L972"></a><tt class="py-lineno"> 972</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio.Alphabet import generic_protein</tt> </tt>
<a name="L973"></a><tt class="py-lineno"> 973</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L974"></a><tt class="py-lineno"> 974</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; files = ["GenBank/NC_000932.faa", "GenBank/NC_005816.faa"]</tt> </tt>
<a name="L975"></a><tt class="py-lineno"> 975</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; def get_gi(name):</tt> </tt>
<a name="L976"></a><tt class="py-lineno"> 976</tt>  <tt class="py-line"><tt class="py-docstring">    ...     parts = name.split("|")</tt> </tt>
<a name="L977"></a><tt class="py-lineno"> 977</tt>  <tt class="py-line"><tt class="py-docstring">    ...     i = parts.index("gi")</tt> </tt>
<a name="L978"></a><tt class="py-lineno"> 978</tt>  <tt class="py-line"><tt class="py-docstring">    ...     assert i != -1</tt> </tt>
<a name="L979"></a><tt class="py-lineno"> 979</tt>  <tt class="py-line"><tt class="py-docstring">    ...     return parts[i+1]</tt> </tt>
<a name="L980"></a><tt class="py-lineno"> 980</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; idx_name = ":memory:" #use an in memory SQLite DB for this test</tt> </tt>
<a name="L981"></a><tt class="py-lineno"> 981</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; records = SeqIO.index_db(idx_name, files, "fasta", generic_protein, get_gi)</tt> </tt>
<a name="L982"></a><tt class="py-lineno"> 982</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; len(records)</tt> </tt>
<a name="L983"></a><tt class="py-lineno"> 983</tt>  <tt class="py-line"><tt class="py-docstring">    95</tt> </tt>
<a name="L984"></a><tt class="py-lineno"> 984</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; records["7525076"].description</tt> </tt>
<a name="L985"></a><tt class="py-lineno"> 985</tt>  <tt class="py-line"><tt class="py-docstring">    'gi|7525076|ref|NP_051101.1| Ycf2 [Arabidopsis thaliana]'</tt> </tt>
<a name="L986"></a><tt class="py-lineno"> 986</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; records["45478717"].description</tt> </tt>
<a name="L987"></a><tt class="py-lineno"> 987</tt>  <tt class="py-line"><tt class="py-docstring">    'gi|45478717|ref|NP_995572.1| pesticin [Yersinia pestis biovar Microtus str. 91001]'</tt> </tt>
<a name="L988"></a><tt class="py-lineno"> 988</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; records.close()</tt> </tt>
<a name="L989"></a><tt class="py-lineno"> 989</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L990"></a><tt class="py-lineno"> 990</tt>  <tt class="py-line"><tt class="py-docstring">    In this example the two files contain 85 and 10 records respectively.</tt> </tt>
<a name="L991"></a><tt class="py-lineno"> 991</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L992"></a><tt class="py-lineno"> 992</tt>  <tt class="py-line"><tt class="py-docstring">    BGZF compressed files are supported, and detected automatically. Ordinary</tt> </tt>
<a name="L993"></a><tt class="py-lineno"> 993</tt>  <tt class="py-line"><tt class="py-docstring">    GZIP compressed files are not supported.</tt> </tt>
<a name="L994"></a><tt class="py-lineno"> 994</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L995"></a><tt class="py-lineno"> 995</tt>  <tt class="py-line"><tt class="py-docstring">    See Also: Bio.SeqIO.index() and Bio.SeqIO.to_dict(), and the Python module</tt> </tt>
<a name="L996"></a><tt class="py-lineno"> 996</tt>  <tt class="py-line"><tt class="py-docstring">    glob which is useful for building lists of files.</tt> </tt>
<a name="L997"></a><tt class="py-lineno"> 997</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L998"></a><tt class="py-lineno"> 998</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L999"></a><tt class="py-lineno"> 999</tt>  <tt class="py-line">    <tt class="py-comment"># Try and give helpful error messages:</tt> </tt>
<a name="L1000"></a><tt class="py-lineno">1000</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">index_filename</tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1001"></a><tt class="py-lineno">1001</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"Need a string for the index filename"</tt><tt class="py-op">)</tt> </tt>
<a name="L1002"></a><tt class="py-lineno">1002</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">filenames</tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1003"></a><tt class="py-lineno">1003</tt>  <tt class="py-line">        <tt class="py-comment"># Make the API a little more friendly, and more similar</tt> </tt>
<a name="L1004"></a><tt class="py-lineno">1004</tt>  <tt class="py-line">        <tt class="py-comment"># to Bio.SeqIO.index(...) for indexing just one file.</tt> </tt>
<a name="L1005"></a><tt class="py-lineno">1005</tt>  <tt class="py-line">        <tt class="py-name">filenames</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">filenames</tt><tt class="py-op">]</tt> </tt>
<a name="L1006"></a><tt class="py-lineno">1006</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">filenames</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">filenames</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1007"></a><tt class="py-lineno">1007</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt> </tt>
<a name="L1008"></a><tt class="py-lineno">1008</tt>  <tt class="py-line">            <tt class="py-string">"Need a list of filenames (as strings), or one filename"</tt><tt class="py-op">)</tt> </tt>
<a name="L1009"></a><tt class="py-lineno">1009</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt id="link-267" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-267', 'format', 'link-142');">format</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-268" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-268', 'format', 'link-142');">format</a></tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1010"></a><tt class="py-lineno">1010</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"Need a string for the file format (lower case)"</tt><tt class="py-op">)</tt> </tt>
<a name="L1011"></a><tt class="py-lineno">1011</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt id="link-269" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-269', 'format', 'link-142');">format</a></tt> <tt class="py-keyword">and</tt> <tt id="link-270" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-270', 'format', 'link-142');">format</a></tt> <tt class="py-op">!=</tt> <tt id="link-271" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-271', 'format', 'link-142');">format</a></tt><tt class="py-op">.</tt><tt id="link-272" class="py-name"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-272', 'lower', 'link-146');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1012"></a><tt class="py-lineno">1012</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Format string '%s' should be lower case"</tt> <tt class="py-op">%</tt> <tt id="link-273" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-273', 'format', 'link-142');">format</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1013"></a><tt class="py-lineno">1013</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt id="link-274" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-274', 'alphabet', 'link-195');">alphabet</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt class="py-op">(</tt><tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-275" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-275', 'alphabet', 'link-195');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-276" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-276', 'Alphabet', 'link-12');">Alphabet</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> </tt>
<a name="L1014"></a><tt class="py-lineno">1014</tt>  <tt class="py-line">                                     <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-277" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-277', 'alphabet', 'link-195');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-278" class="py-name"><a title="Bio.Alphabet.AlphabetEncoder" class="py-name" href="#" onclick="return doclink('link-278', 'AlphabetEncoder', 'link-14');">AlphabetEncoder</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1015"></a><tt class="py-lineno">1015</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid alphabet, %r"</tt> <tt class="py-op">%</tt> <tt id="link-279" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-279', 'alphabet', 'link-195');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1016"></a><tt class="py-lineno">1016</tt>  <tt class="py-line"> </tt>
<a name="L1017"></a><tt class="py-lineno">1017</tt>  <tt class="py-line">    <tt class="py-comment"># Map the file format to a sequence iterator:</tt> </tt>
<a name="L1018"></a><tt class="py-lineno">1018</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt class="py-op">.</tt><tt id="link-280" class="py-name"><a title="Bio.SearchIO._index
Bio.SeqIO._index" class="py-name" href="#" onclick="return doclink('link-280', '_index', 'link-254');">_index</a></tt> <tt class="py-keyword">import</tt> <tt id="link-281" class="py-name"><a title="Bio.SeqIO._index._FormatToRandomAccess" class="py-name" href="#" onclick="return doclink('link-281', '_FormatToRandomAccess', 'link-255');">_FormatToRandomAccess</a></tt>  <tt class="py-comment"># Lazy import</tt> </tt>
<a name="L1019"></a><tt class="py-lineno">1019</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-282" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-282', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-283" class="py-name"><a title="Bio.File" class="py-name" href="#" onclick="return doclink('link-283', 'File', 'link-3');">File</a></tt> <tt class="py-keyword">import</tt> <tt id="link-284" class="py-name" targets="Class Bio.File._SQLiteManySeqFilesDict=Bio.File._SQLiteManySeqFilesDict-class.html"><a title="Bio.File._SQLiteManySeqFilesDict" class="py-name" href="#" onclick="return doclink('link-284', '_SQLiteManySeqFilesDict', 'link-284');">_SQLiteManySeqFilesDict</a></tt> </tt>
<a name="L1020"></a><tt class="py-lineno">1020</tt>  <tt class="py-line">    <tt class="py-name">repr</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-string">"SeqIO.index_db(%r, filenames=%r, format=%r, alphabet=%r, key_function=%r)"</tt> </tt>
<a name="L1021"></a><tt class="py-lineno">1021</tt>  <tt class="py-line">            <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">index_filename</tt><tt class="py-op">,</tt> <tt class="py-name">filenames</tt><tt class="py-op">,</tt> <tt id="link-285" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-285', 'format', 'link-142');">format</a></tt><tt class="py-op">,</tt> <tt id="link-286" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-286', 'alphabet', 'link-195');">alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-name">key_function</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1022"></a><tt class="py-lineno">1022</tt>  <tt class="py-line"> </tt>
<a name="L1023"></a><tt class="py-lineno">1023</tt>  <tt class="py-line">    <tt class="py-keyword">def</tt> <tt class="py-def-name">proxy_factory</tt><tt class="py-op">(</tt><tt class="py-param">format</tt><tt class="py-op">,</tt> <tt class="py-param">filename</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1024"></a><tt class="py-lineno">1024</tt>  <tt class="py-line">        <tt class="py-docstring">"""Given a filename returns proxy object, else boolean if format OK."""</tt> </tt>
<a name="L1025"></a><tt class="py-lineno">1025</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">filename</tt><tt class="py-op">:</tt> </tt>
<a name="L1026"></a><tt class="py-lineno">1026</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-287" class="py-name"><a title="Bio.SeqIO._index._FormatToRandomAccess" class="py-name" href="#" onclick="return doclink('link-287', '_FormatToRandomAccess', 'link-255');">_FormatToRandomAccess</a></tt><tt class="py-op">[</tt><tt id="link-288" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-288', 'format', 'link-142');">format</a></tt><tt class="py-op">]</tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt id="link-289" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-289', 'format', 'link-142');">format</a></tt><tt class="py-op">,</tt> <tt id="link-290" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-290', 'alphabet', 'link-195');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1027"></a><tt class="py-lineno">1027</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1028"></a><tt class="py-lineno">1028</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-291" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-291', 'format', 'link-142');">format</a></tt> <tt class="py-keyword">in</tt> <tt id="link-292" class="py-name"><a title="Bio.SeqIO._index._FormatToRandomAccess" class="py-name" href="#" onclick="return doclink('link-292', '_FormatToRandomAccess', 'link-255');">_FormatToRandomAccess</a></tt> </tt>
</div><a name="L1029"></a><tt class="py-lineno">1029</tt>  <tt class="py-line"> </tt>
<a name="L1030"></a><tt class="py-lineno">1030</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-293" class="py-name"><a title="Bio.File._SQLiteManySeqFilesDict" class="py-name" href="#" onclick="return doclink('link-293', '_SQLiteManySeqFilesDict', 'link-284');">_SQLiteManySeqFilesDict</a></tt><tt class="py-op">(</tt><tt class="py-name">index_filename</tt><tt class="py-op">,</tt> <tt class="py-name">filenames</tt><tt class="py-op">,</tt> </tt>
<a name="L1031"></a><tt class="py-lineno">1031</tt>  <tt class="py-line">                                   <tt class="py-name">proxy_factory</tt><tt class="py-op">,</tt> <tt id="link-294" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-294', 'format', 'link-142');">format</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1032"></a><tt class="py-lineno">1032</tt>  <tt class="py-line">                                   <tt class="py-name">key_function</tt><tt class="py-op">,</tt> <tt class="py-name">repr</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1033"></a><tt class="py-lineno">1033</tt>  <tt class="py-line"> </tt>
<a name="L1034"></a><tt class="py-lineno">1034</tt>  <tt class="py-line"> </tt>
<a name="convert"></a><div id="convert-def"><a name="L1035"></a><tt class="py-lineno">1035</tt> <a class="py-toggle" href="#" id="convert-toggle" onclick="return toggle('convert');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO-module.html#convert">convert</a><tt class="py-op">(</tt><tt class="py-param">in_file</tt><tt class="py-op">,</tt> <tt class="py-param">in_format</tt><tt class="py-op">,</tt> <tt class="py-param">out_file</tt><tt class="py-op">,</tt> <tt class="py-param">out_format</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="convert-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="convert-expanded"><a name="L1036"></a><tt class="py-lineno">1036</tt>  <tt class="py-line">    <tt class="py-docstring">"""Convert between two sequence file formats, return number of records.</tt> </tt>
<a name="L1037"></a><tt class="py-lineno">1037</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1038"></a><tt class="py-lineno">1038</tt>  <tt class="py-line"><tt class="py-docstring">    Arguments:</tt> </tt>
<a name="L1039"></a><tt class="py-lineno">1039</tt>  <tt class="py-line"><tt class="py-docstring">     - in_file - an input handle or filename</tt> </tt>
<a name="L1040"></a><tt class="py-lineno">1040</tt>  <tt class="py-line"><tt class="py-docstring">     - in_format - input file format, lower case string</tt> </tt>
<a name="L1041"></a><tt class="py-lineno">1041</tt>  <tt class="py-line"><tt class="py-docstring">     - out_file - an output handle or filename</tt> </tt>
<a name="L1042"></a><tt class="py-lineno">1042</tt>  <tt class="py-line"><tt class="py-docstring">     - out_format - output file format, lower case string</tt> </tt>
<a name="L1043"></a><tt class="py-lineno">1043</tt>  <tt class="py-line"><tt class="py-docstring">     - alphabet - optional alphabet to assume</tt> </tt>
<a name="L1044"></a><tt class="py-lineno">1044</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1045"></a><tt class="py-lineno">1045</tt>  <tt class="py-line"><tt class="py-docstring">    **NOTE** - If you provide an output filename, it will be opened which will</tt> </tt>
<a name="L1046"></a><tt class="py-lineno">1046</tt>  <tt class="py-line"><tt class="py-docstring">    overwrite any existing file without warning. This may happen if even</tt> </tt>
<a name="L1047"></a><tt class="py-lineno">1047</tt>  <tt class="py-line"><tt class="py-docstring">    the conversion is aborted (e.g. an invalid out_format name is given).</tt> </tt>
<a name="L1048"></a><tt class="py-lineno">1048</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1049"></a><tt class="py-lineno">1049</tt>  <tt class="py-line"><tt class="py-docstring">    For example, going from a filename to a handle:</tt> </tt>
<a name="L1050"></a><tt class="py-lineno">1050</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1051"></a><tt class="py-lineno">1051</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L1052"></a><tt class="py-lineno">1052</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; try:</tt> </tt>
<a name="L1053"></a><tt class="py-lineno">1053</tt>  <tt class="py-line"><tt class="py-docstring">    ...     from StringIO import StringIO # Python 2</tt> </tt>
<a name="L1054"></a><tt class="py-lineno">1054</tt>  <tt class="py-line"><tt class="py-docstring">    ... except ImportError:</tt> </tt>
<a name="L1055"></a><tt class="py-lineno">1055</tt>  <tt class="py-line"><tt class="py-docstring">    ...     from io import StringIO # Python 3</tt> </tt>
<a name="L1056"></a><tt class="py-lineno">1056</tt>  <tt class="py-line"><tt class="py-docstring">    ...</tt> </tt>
<a name="L1057"></a><tt class="py-lineno">1057</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; handle = StringIO("")</tt> </tt>
<a name="L1058"></a><tt class="py-lineno">1058</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; SeqIO.convert("Quality/example.fastq", "fastq", handle, "fasta")</tt> </tt>
<a name="L1059"></a><tt class="py-lineno">1059</tt>  <tt class="py-line"><tt class="py-docstring">    3</tt> </tt>
<a name="L1060"></a><tt class="py-lineno">1060</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(handle.getvalue())</tt> </tt>
<a name="L1061"></a><tt class="py-lineno">1061</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;EAS54_6_R1_2_1_413_324</tt> </tt>
<a name="L1062"></a><tt class="py-lineno">1062</tt>  <tt class="py-line"><tt class="py-docstring">    CCCTTCTTGTCTTCAGCGTTTCTCC</tt> </tt>
<a name="L1063"></a><tt class="py-lineno">1063</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;EAS54_6_R1_2_1_540_792</tt> </tt>
<a name="L1064"></a><tt class="py-lineno">1064</tt>  <tt class="py-line"><tt class="py-docstring">    TTGGCAGGCCAAGGCCGATGGATCA</tt> </tt>
<a name="L1065"></a><tt class="py-lineno">1065</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;EAS54_6_R1_2_1_443_348</tt> </tt>
<a name="L1066"></a><tt class="py-lineno">1066</tt>  <tt class="py-line"><tt class="py-docstring">    GTTGCTTCTGGCGTGGGTGGGGGGG</tt> </tt>
<a name="L1067"></a><tt class="py-lineno">1067</tt>  <tt class="py-line"><tt class="py-docstring">    &lt;BLANKLINE&gt;</tt> </tt>
<a name="L1068"></a><tt class="py-lineno">1068</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1069"></a><tt class="py-lineno">1069</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">in_format</tt> <tt class="py-keyword">in</tt> <tt id="link-295" class="py-name"><a title="Bio.SeqIO._BinaryFormats" class="py-name" href="#" onclick="return doclink('link-295', '_BinaryFormats', 'link-139');">_BinaryFormats</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1070"></a><tt class="py-lineno">1070</tt>  <tt class="py-line">        <tt class="py-name">in_mode</tt> <tt class="py-op">=</tt> <tt class="py-string">'rb'</tt> </tt>
<a name="L1071"></a><tt class="py-lineno">1071</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1072"></a><tt class="py-lineno">1072</tt>  <tt class="py-line">        <tt class="py-name">in_mode</tt> <tt class="py-op">=</tt> <tt class="py-string">'rU'</tt> </tt>
<a name="L1073"></a><tt class="py-lineno">1073</tt>  <tt class="py-line"> </tt>
<a name="L1074"></a><tt class="py-lineno">1074</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">out_format</tt> <tt class="py-keyword">in</tt> <tt id="link-296" class="py-name"><a title="Bio.SeqIO._BinaryFormats" class="py-name" href="#" onclick="return doclink('link-296', '_BinaryFormats', 'link-139');">_BinaryFormats</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1075"></a><tt class="py-lineno">1075</tt>  <tt class="py-line">        <tt class="py-name">out_mode</tt> <tt class="py-op">=</tt> <tt class="py-string">'wb'</tt> </tt>
<a name="L1076"></a><tt class="py-lineno">1076</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1077"></a><tt class="py-lineno">1077</tt>  <tt class="py-line">        <tt class="py-name">out_mode</tt> <tt class="py-op">=</tt> <tt class="py-string">'w'</tt> </tt>
<a name="L1078"></a><tt class="py-lineno">1078</tt>  <tt class="py-line"> </tt>
<a name="L1079"></a><tt class="py-lineno">1079</tt>  <tt class="py-line">    <tt class="py-comment"># This will check the arguments and issue error messages,</tt> </tt>
<a name="L1080"></a><tt class="py-lineno">1080</tt>  <tt class="py-line">    <tt class="py-comment"># after we have opened the file which is a shame.</tt> </tt>
<a name="L1081"></a><tt class="py-lineno">1081</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt class="py-op">.</tt><tt id="link-297" class="py-name" targets="Module Bio.SeqIO._convert=Bio.SeqIO._convert-module.html"><a title="Bio.SeqIO._convert" class="py-name" href="#" onclick="return doclink('link-297', '_convert', 'link-297');">_convert</a></tt> <tt class="py-keyword">import</tt> <tt id="link-298" class="py-name" targets="Function Bio.SeqIO._convert._handle_convert()=Bio.SeqIO._convert-module.html#_handle_convert"><a title="Bio.SeqIO._convert._handle_convert" class="py-name" href="#" onclick="return doclink('link-298', '_handle_convert', 'link-298');">_handle_convert</a></tt>  <tt class="py-comment"># Lazy import</tt> </tt>
<a name="L1082"></a><tt class="py-lineno">1082</tt>  <tt class="py-line">    <tt class="py-keyword">with</tt> <tt id="link-299" class="py-name"><a title="Bio.File.as_handle" class="py-name" href="#" onclick="return doclink('link-299', 'as_handle', 'link-4');">as_handle</a></tt><tt class="py-op">(</tt><tt class="py-name">in_file</tt><tt class="py-op">,</tt> <tt class="py-name">in_mode</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">in_handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1083"></a><tt class="py-lineno">1083</tt>  <tt class="py-line">        <tt class="py-keyword">with</tt> <tt id="link-300" class="py-name"><a title="Bio.File.as_handle" class="py-name" href="#" onclick="return doclink('link-300', 'as_handle', 'link-4');">as_handle</a></tt><tt class="py-op">(</tt><tt class="py-name">out_file</tt><tt class="py-op">,</tt> <tt class="py-name">out_mode</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">out_handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1084"></a><tt class="py-lineno">1084</tt>  <tt class="py-line">            <tt id="link-301" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-301', 'count', 'link-157');">count</a></tt> <tt class="py-op">=</tt> <tt id="link-302" class="py-name"><a title="Bio.SeqIO._convert._handle_convert" class="py-name" href="#" onclick="return doclink('link-302', '_handle_convert', 'link-298');">_handle_convert</a></tt><tt class="py-op">(</tt><tt class="py-name">in_handle</tt><tt class="py-op">,</tt> <tt class="py-name">in_format</tt><tt class="py-op">,</tt> </tt>
<a name="L1085"></a><tt class="py-lineno">1085</tt>  <tt class="py-line">                                    <tt class="py-name">out_handle</tt><tt class="py-op">,</tt> <tt class="py-name">out_format</tt><tt class="py-op">,</tt> </tt>
<a name="L1086"></a><tt class="py-lineno">1086</tt>  <tt class="py-line">                                    <tt id="link-303" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-303', 'alphabet', 'link-195');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1087"></a><tt class="py-lineno">1087</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-304" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-304', 'count', 'link-157');">count</a></tt> </tt>
</div><a name="L1088"></a><tt class="py-lineno">1088</tt>  <tt class="py-line"> </tt>
<a name="L1089"></a><tt class="py-lineno">1089</tt>  <tt class="py-line"> </tt>
<a name="L1090"></a><tt class="py-lineno">1090</tt>  <tt class="py-line"><tt class="py-comment"># This helpful trick for testing no longer works with the</tt> </tt>
<a name="L1091"></a><tt class="py-lineno">1091</tt>  <tt class="py-line"><tt class="py-comment"># local imports :(</tt> </tt>
<a name="L1092"></a><tt class="py-lineno">1092</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L1093"></a><tt class="py-lineno">1093</tt>  <tt class="py-line"><tt class="py-comment"># if __name__ == "__main__":</tt> </tt>
<a name="L1094"></a><tt class="py-lineno">1094</tt>  <tt class="py-line"><tt class="py-comment">#    from Bio._utils import run_doctest</tt> </tt>
<a name="L1095"></a><tt class="py-lineno">1095</tt>  <tt class="py-line"><tt class="py-comment">#    run_doctest()</tt> </tt>
<a name="L1096"></a><tt class="py-lineno">1096</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Tue Dec 18 15:58:16 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
