<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.GenBank.Scanner</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.GenBank-module.html">Package&nbsp;GenBank</a> ::
        Module&nbsp;Scanner
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.GenBank.Scanner-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.GenBank.Scanner-module.html">Module Bio.GenBank.Scanner</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">   1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2007-2010 by Peter Cock.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">   2</tt>  <tt class="py-line"><tt class="py-comment"># Revisions copyright 2010 by Uri Laserson.  All rights reserved.</tt> </tt>
<a name="L3"></a><tt class="py-lineno">   3</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L4"></a><tt class="py-lineno">   4</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L5"></a><tt class="py-lineno">   5</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L6"></a><tt class="py-lineno">   6</tt>  <tt class="py-line"><tt class="py-docstring">"""Internal code for parsing GenBank and EMBL files (PRIVATE).</tt> </tt>
<a name="L7"></a><tt class="py-lineno">   7</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L8"></a><tt class="py-lineno">   8</tt>  <tt class="py-line"><tt class="py-docstring">This code is NOT intended for direct use.  It provides a basic scanner</tt> </tt>
<a name="L9"></a><tt class="py-lineno">   9</tt>  <tt class="py-line"><tt class="py-docstring">(for use with a event consumer such as Bio.GenBank._FeatureConsumer)</tt> </tt>
<a name="L10"></a><tt class="py-lineno">  10</tt>  <tt class="py-line"><tt class="py-docstring">to parse a GenBank or EMBL file (with their shared INSDC feature table).</tt> </tt>
<a name="L11"></a><tt class="py-lineno">  11</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L12"></a><tt class="py-lineno">  12</tt>  <tt class="py-line"><tt class="py-docstring">It is used by Bio.GenBank to parse GenBank files</tt> </tt>
<a name="L13"></a><tt class="py-lineno">  13</tt>  <tt class="py-line"><tt class="py-docstring">It is also used by Bio.SeqIO to parse GenBank and EMBL files</tt> </tt>
<a name="L14"></a><tt class="py-lineno">  14</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L15"></a><tt class="py-lineno">  15</tt>  <tt class="py-line"><tt class="py-docstring">Feature Table Documentation:</tt> </tt>
<a name="L16"></a><tt class="py-lineno">  16</tt>  <tt class="py-line"><tt class="py-docstring">http://www.insdc.org/files/feature_table.html</tt> </tt>
<a name="L17"></a><tt class="py-lineno">  17</tt>  <tt class="py-line"><tt class="py-docstring">http://www.ncbi.nlm.nih.gov/projects/collab/FT/index.html</tt> </tt>
<a name="L18"></a><tt class="py-lineno">  18</tt>  <tt class="py-line"><tt class="py-docstring">ftp://ftp.ncbi.nih.gov/genbank/docs/</tt> </tt>
<a name="L19"></a><tt class="py-lineno">  19</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L20"></a><tt class="py-lineno">  20</tt>  <tt class="py-line"><tt class="py-comment"># 17-MAR-2009: added wgs, wgs_scafld for GenBank whole genome shotgun master records.</tt> </tt>
<a name="L21"></a><tt class="py-lineno">  21</tt>  <tt class="py-line"><tt class="py-comment"># These are GenBank files that summarize the content of a project, and provide lists of</tt> </tt>
<a name="L22"></a><tt class="py-lineno">  22</tt>  <tt class="py-line"><tt class="py-comment"># scaffold and contig files in the project. These will be in annotations['wgs'] and</tt> </tt>
<a name="L23"></a><tt class="py-lineno">  23</tt>  <tt class="py-line"><tt class="py-comment"># annotations['wgs_scafld']. These GenBank files do not have sequences. See</tt> </tt>
<a name="L24"></a><tt class="py-lineno">  24</tt>  <tt class="py-line"><tt class="py-comment"># http://groups.google.com/group/bionet.molbio.genbank/browse_thread/thread/51fb88bf39e7dc36</tt> </tt>
<a name="L25"></a><tt class="py-lineno">  25</tt>  <tt class="py-line"><tt class="py-comment"># http://is.gd/nNgk</tt> </tt>
<a name="L26"></a><tt class="py-lineno">  26</tt>  <tt class="py-line"><tt class="py-comment"># for more details of this format, and an example.</tt> </tt>
<a name="L27"></a><tt class="py-lineno">  27</tt>  <tt class="py-line"><tt class="py-comment"># Added by Ying Huang &amp; Iddo Friedberg</tt> </tt>
<a name="L28"></a><tt class="py-lineno">  28</tt>  <tt class="py-line"> </tt>
<a name="L29"></a><tt class="py-lineno">  29</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">__future__</tt> <tt class="py-keyword">import</tt> <tt class="py-name">print_function</tt> </tt>
<a name="L30"></a><tt class="py-lineno">  30</tt>  <tt class="py-line"> </tt>
<a name="L31"></a><tt class="py-lineno">  31</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L32"></a><tt class="py-lineno">  32</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">re</tt> </tt>
<a name="L33"></a><tt class="py-lineno">  33</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Module Bio.Seq=Bio.Seq-module.html,Class Bio.Seq.Seq=Bio.Seq.Seq-class.html"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-1', 'Seq', 'link-1');">Seq</a></tt> <tt class="py-keyword">import</tt> <tt id="link-2" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-2', 'Seq', 'link-1');">Seq</a></tt> </tt>
<a name="L34"></a><tt class="py-lineno">  34</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-3" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-3', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-4" class="py-name" targets="Module Bio.SeqRecord=Bio.SeqRecord-module.html,Class Bio.SeqRecord.SeqRecord=Bio.SeqRecord.SeqRecord-class.html"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-4', 'SeqRecord', 'link-4');">SeqRecord</a></tt> <tt class="py-keyword">import</tt> <tt id="link-5" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-5', 'SeqRecord', 'link-4');">SeqRecord</a></tt> </tt>
<a name="L35"></a><tt class="py-lineno">  35</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-6" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-6', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-7" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-7', 'Alphabet', 'link-7');">Alphabet</a></tt> <tt class="py-keyword">import</tt> <tt id="link-8" class="py-name" targets="Variable Bio.Alphabet.generic_protein=Bio.Alphabet-module.html#generic_protein"><a title="Bio.Alphabet.generic_protein" class="py-name" href="#" onclick="return doclink('link-8', 'generic_protein', 'link-8');">generic_protein</a></tt> </tt>
<a name="L36"></a><tt class="py-lineno">  36</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-9" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-9', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-10" class="py-name" targets="Class Bio.BiopythonParserWarning=Bio.BiopythonParserWarning-class.html"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-10', 'BiopythonParserWarning', 'link-10');">BiopythonParserWarning</a></tt> </tt>
<a name="L37"></a><tt class="py-lineno">  37</tt>  <tt class="py-line"> </tt>
<a name="L38"></a><tt class="py-lineno">  38</tt>  <tt class="py-line"><tt class="py-name">__docformat__</tt> <tt class="py-op">=</tt> <tt class="py-string">"restructuredtext en"</tt> </tt>
<a name="L39"></a><tt class="py-lineno">  39</tt>  <tt class="py-line"> </tt>
<a name="L40"></a><tt class="py-lineno">  40</tt>  <tt class="py-line"> </tt>
<a name="InsdcScanner"></a><div id="InsdcScanner-def"><a name="L41"></a><tt class="py-lineno">  41</tt> <a class="py-toggle" href="#" id="InsdcScanner-toggle" onclick="return toggle('InsdcScanner');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.InsdcScanner-class.html">InsdcScanner</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="InsdcScanner-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="InsdcScanner-expanded"><a name="L42"></a><tt class="py-lineno">  42</tt>  <tt class="py-line">    <tt class="py-docstring">"""Basic functions for breaking up a GenBank/EMBL file into sub sections.</tt> </tt>
<a name="L43"></a><tt class="py-lineno">  43</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L44"></a><tt class="py-lineno">  44</tt>  <tt class="py-line"><tt class="py-docstring">    The International Nucleotide Sequence Database Collaboration (INSDC)</tt> </tt>
<a name="L45"></a><tt class="py-lineno">  45</tt>  <tt class="py-line"><tt class="py-docstring">    between the DDBJ, EMBL, and GenBank.  These organisations all use the</tt> </tt>
<a name="L46"></a><tt class="py-lineno">  46</tt>  <tt class="py-line"><tt class="py-docstring">    same "Feature Table" layout in their plain text flat file formats.</tt> </tt>
<a name="L47"></a><tt class="py-lineno">  47</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L48"></a><tt class="py-lineno">  48</tt>  <tt class="py-line"><tt class="py-docstring">    However, the header and sequence sections of an EMBL file are very</tt> </tt>
<a name="L49"></a><tt class="py-lineno">  49</tt>  <tt class="py-line"><tt class="py-docstring">    different in layout to those produced by GenBank/DDBJ."""</tt> </tt>
<a name="L50"></a><tt class="py-lineno">  50</tt>  <tt class="py-line"> </tt>
<a name="L51"></a><tt class="py-lineno">  51</tt>  <tt class="py-line">    <tt class="py-comment"># These constants get redefined with sensible values in the sub classes:</tt> </tt>
<a name="L52"></a><tt class="py-lineno">  52</tt>  <tt class="py-line">    <tt id="link-11" class="py-name" targets="Variable Bio.GenBank.Scanner.EmblScanner.RECORD_START=Bio.GenBank.Scanner.EmblScanner-class.html#RECORD_START,Variable Bio.GenBank.Scanner.GenBankScanner.RECORD_START=Bio.GenBank.Scanner.GenBankScanner-class.html#RECORD_START,Variable Bio.GenBank.Scanner.InsdcScanner.RECORD_START=Bio.GenBank.Scanner.InsdcScanner-class.html#RECORD_START"><a title="Bio.GenBank.Scanner.EmblScanner.RECORD_START
Bio.GenBank.Scanner.GenBankScanner.RECORD_START
Bio.GenBank.Scanner.InsdcScanner.RECORD_START" class="py-name" href="#" onclick="return doclink('link-11', 'RECORD_START', 'link-11');">RECORD_START</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"XXX"</tt>  <tt class="py-comment"># "LOCUS       " or "ID   "</tt> </tt>
<a name="L53"></a><tt class="py-lineno">  53</tt>  <tt class="py-line">    <tt id="link-12" class="py-name" targets="Variable Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH=Bio.GenBank.Scanner.EmblScanner-class.html#HEADER_WIDTH,Variable Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH=Bio.GenBank.Scanner.GenBankScanner-class.html#HEADER_WIDTH,Variable Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH=Bio.GenBank.Scanner.InsdcScanner-class.html#HEADER_WIDTH,Variable Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH=Bio.SeqIO.InsdcIO.EmblWriter-class.html#HEADER_WIDTH,Variable Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH=Bio.SeqIO.InsdcIO.GenBankWriter-class.html#HEADER_WIDTH,Variable Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH=Bio.SeqIO.InsdcIO.ImgtWriter-class.html#HEADER_WIDTH"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-12', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt> <tt class="py-op">=</tt> <tt class="py-number">3</tt>   <tt class="py-comment"># 12 or 5</tt> </tt>
<a name="L54"></a><tt class="py-lineno">  54</tt>  <tt class="py-line">    <tt id="link-13" class="py-name" targets="Variable Bio.GenBank.Scanner.EmblScanner.FEATURE_START_MARKERS=Bio.GenBank.Scanner.EmblScanner-class.html#FEATURE_START_MARKERS,Variable Bio.GenBank.Scanner.GenBankScanner.FEATURE_START_MARKERS=Bio.GenBank.Scanner.GenBankScanner-class.html#FEATURE_START_MARKERS,Variable Bio.GenBank.Scanner.InsdcScanner.FEATURE_START_MARKERS=Bio.GenBank.Scanner.InsdcScanner-class.html#FEATURE_START_MARKERS,Variable Bio.GenBank.Scanner._ImgtScanner.FEATURE_START_MARKERS=Bio.GenBank.Scanner._ImgtScanner-class.html#FEATURE_START_MARKERS"><a title="Bio.GenBank.Scanner.EmblScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner.GenBankScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner.InsdcScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner._ImgtScanner.FEATURE_START_MARKERS" class="py-name" href="#" onclick="return doclink('link-13', 'FEATURE_START_MARKERS', 'link-13');">FEATURE_START_MARKERS</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">"XXX***FEATURES***XXX"</tt><tt class="py-op">]</tt> </tt>
<a name="L55"></a><tt class="py-lineno">  55</tt>  <tt class="py-line">    <tt id="link-14" class="py-name" targets="Variable Bio.GenBank.Scanner.EmblScanner.FEATURE_END_MARKERS=Bio.GenBank.Scanner.EmblScanner-class.html#FEATURE_END_MARKERS,Variable Bio.GenBank.Scanner.GenBankScanner.FEATURE_END_MARKERS=Bio.GenBank.Scanner.GenBankScanner-class.html#FEATURE_END_MARKERS,Variable Bio.GenBank.Scanner.InsdcScanner.FEATURE_END_MARKERS=Bio.GenBank.Scanner.InsdcScanner-class.html#FEATURE_END_MARKERS"><a title="Bio.GenBank.Scanner.EmblScanner.FEATURE_END_MARKERS
Bio.GenBank.Scanner.GenBankScanner.FEATURE_END_MARKERS
Bio.GenBank.Scanner.InsdcScanner.FEATURE_END_MARKERS" class="py-name" href="#" onclick="return doclink('link-14', 'FEATURE_END_MARKERS', 'link-14');">FEATURE_END_MARKERS</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">"XXX***END FEATURES***XXX"</tt><tt class="py-op">]</tt> </tt>
<a name="L56"></a><tt class="py-lineno">  56</tt>  <tt class="py-line">    <tt id="link-15" class="py-name" targets="Variable Bio.GenBank.FEATURE_QUALIFIER_INDENT=Bio.GenBank-module.html#FEATURE_QUALIFIER_INDENT,Variable Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_INDENT=Bio.GenBank.Scanner.EmblScanner-class.html#FEATURE_QUALIFIER_INDENT,Variable Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_INDENT=Bio.GenBank.Scanner.GenBankScanner-class.html#FEATURE_QUALIFIER_INDENT,Variable Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_INDENT=Bio.GenBank.Scanner.InsdcScanner-class.html#FEATURE_QUALIFIER_INDENT"><a title="Bio.GenBank.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-15', 'FEATURE_QUALIFIER_INDENT', 'link-15');">FEATURE_QUALIFIER_INDENT</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L57"></a><tt class="py-lineno">  57</tt>  <tt class="py-line">    <tt id="link-16" class="py-name" targets="Variable Bio.GenBank.FEATURE_QUALIFIER_SPACER=Bio.GenBank-module.html#FEATURE_QUALIFIER_SPACER,Variable Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_SPACER=Bio.GenBank.Scanner.EmblScanner-class.html#FEATURE_QUALIFIER_SPACER,Variable Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_SPACER=Bio.GenBank.Scanner.GenBankScanner-class.html#FEATURE_QUALIFIER_SPACER,Variable Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_SPACER=Bio.GenBank.Scanner.InsdcScanner-class.html#FEATURE_QUALIFIER_SPACER"><a title="Bio.GenBank.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_SPACER" class="py-name" href="#" onclick="return doclink('link-16', 'FEATURE_QUALIFIER_SPACER', 'link-16');">FEATURE_QUALIFIER_SPACER</a></tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L58"></a><tt class="py-lineno">  58</tt>  <tt class="py-line">    <tt id="link-17" class="py-name" targets="Variable Bio.GenBank.Scanner.EmblScanner.SEQUENCE_HEADERS=Bio.GenBank.Scanner.EmblScanner-class.html#SEQUENCE_HEADERS,Variable Bio.GenBank.Scanner.GenBankScanner.SEQUENCE_HEADERS=Bio.GenBank.Scanner.GenBankScanner-class.html#SEQUENCE_HEADERS,Variable Bio.GenBank.Scanner.InsdcScanner.SEQUENCE_HEADERS=Bio.GenBank.Scanner.InsdcScanner-class.html#SEQUENCE_HEADERS"><a title="Bio.GenBank.Scanner.EmblScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.GenBankScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.InsdcScanner.SEQUENCE_HEADERS" class="py-name" href="#" onclick="return doclink('link-17', 'SEQUENCE_HEADERS', 'link-17');">SEQUENCE_HEADERS</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">"XXX"</tt><tt class="py-op">]</tt>  <tt class="py-comment"># with right hand side spaces removed</tt> </tt>
<a name="L59"></a><tt class="py-lineno">  59</tt>  <tt class="py-line"> </tt>
<a name="InsdcScanner.__init__"></a><div id="InsdcScanner.__init__-def"><a name="L60"></a><tt class="py-lineno">  60</tt> <a class="py-toggle" href="#" id="InsdcScanner.__init__-toggle" onclick="return toggle('InsdcScanner.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.InsdcScanner-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">debug</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="InsdcScanner.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="InsdcScanner.__init__-expanded"><a name="L61"></a><tt class="py-lineno">  61</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-18" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.RECORD_START
Bio.GenBank.Scanner.GenBankScanner.RECORD_START
Bio.GenBank.Scanner.InsdcScanner.RECORD_START" class="py-name" href="#" onclick="return doclink('link-18', 'RECORD_START', 'link-11');">RECORD_START</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-19" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-19', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt> </tt>
<a name="L62"></a><tt class="py-lineno">  62</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">marker</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-20" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.GenBankScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.InsdcScanner.SEQUENCE_HEADERS" class="py-name" href="#" onclick="return doclink('link-20', 'SEQUENCE_HEADERS', 'link-17');">SEQUENCE_HEADERS</a></tt><tt class="py-op">:</tt> </tt>
<a name="L63"></a><tt class="py-lineno">  63</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">marker</tt> <tt class="py-op">==</tt> <tt class="py-name">marker</tt><tt class="py-op">.</tt><tt id="link-21" class="py-name" targets="Method Bio.Seq.Seq.rstrip()=Bio.Seq.Seq-class.html#rstrip"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-21', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L64"></a><tt class="py-lineno">  64</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-22" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_SPACER" class="py-name" href="#" onclick="return doclink('link-22', 'FEATURE_QUALIFIER_SPACER', 'link-16');">FEATURE_QUALIFIER_SPACER</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-23" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-23', 'FEATURE_QUALIFIER_INDENT', 'link-15');">FEATURE_QUALIFIER_INDENT</a></tt> </tt>
<a name="L65"></a><tt class="py-lineno">  65</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt> <tt class="py-op">=</tt> <tt class="py-name">debug</tt> </tt>
<a name="L66"></a><tt class="py-lineno">  66</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L67"></a><tt class="py-lineno">  67</tt>  <tt class="py-line"> </tt>
<a name="InsdcScanner.set_handle"></a><div id="InsdcScanner.set_handle-def"><a name="L68"></a><tt class="py-lineno">  68</tt> <a class="py-toggle" href="#" id="InsdcScanner.set_handle-toggle" onclick="return toggle('InsdcScanner.set_handle');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.InsdcScanner-class.html#set_handle">set_handle</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="InsdcScanner.set_handle-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="InsdcScanner.set_handle-expanded"><a name="L69"></a><tt class="py-lineno">  69</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt> </tt>
<a name="L70"></a><tt class="py-lineno">  70</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
</div><a name="L71"></a><tt class="py-lineno">  71</tt>  <tt class="py-line"> </tt>
<a name="InsdcScanner.find_start"></a><div id="InsdcScanner.find_start-def"><a name="L72"></a><tt class="py-lineno">  72</tt> <a class="py-toggle" href="#" id="InsdcScanner.find_start-toggle" onclick="return toggle('InsdcScanner.find_start');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.InsdcScanner-class.html#find_start">find_start</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="InsdcScanner.find_start-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="InsdcScanner.find_start-expanded"><a name="L73"></a><tt class="py-lineno">  73</tt>  <tt class="py-line">        <tt class="py-docstring">"""Read in lines until find the ID/LOCUS line, which is returned.</tt> </tt>
<a name="L74"></a><tt class="py-lineno">  74</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L75"></a><tt class="py-lineno">  75</tt>  <tt class="py-line"><tt class="py-docstring">        Any preamble (such as the header used by the NCBI on ``*.seq.gz`` archives)</tt> </tt>
<a name="L76"></a><tt class="py-lineno">  76</tt>  <tt class="py-line"><tt class="py-docstring">        will we ignored."""</tt> </tt>
<a name="L77"></a><tt class="py-lineno">  77</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L78"></a><tt class="py-lineno">  78</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L79"></a><tt class="py-lineno">  79</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> </tt>
<a name="L80"></a><tt class="py-lineno">  80</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L81"></a><tt class="py-lineno">  81</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L82"></a><tt class="py-lineno">  82</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-24" class="py-name" targets="Method Bio.File.UndoHandle.readline()=Bio.File.UndoHandle-class.html#readline,Method Bio.bgzf.BgzfReader.readline()=Bio.bgzf.BgzfReader-class.html#readline"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-24', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L83"></a><tt class="py-lineno">  83</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L84"></a><tt class="py-lineno">  84</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt><tt class="py-op">:</tt> </tt>
<a name="L85"></a><tt class="py-lineno">  85</tt>  <tt class="py-line">                    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"End of file"</tt><tt class="py-op">)</tt> </tt>
<a name="L86"></a><tt class="py-lineno">  86</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
<a name="L87"></a><tt class="py-lineno">  87</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-25" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-25', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-26" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.RECORD_START
Bio.GenBank.Scanner.GenBankScanner.RECORD_START
Bio.GenBank.Scanner.InsdcScanner.RECORD_START" class="py-name" href="#" onclick="return doclink('link-26', 'RECORD_START', 'link-11');">RECORD_START</a></tt><tt class="py-op">:</tt> </tt>
<a name="L88"></a><tt class="py-lineno">  88</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L89"></a><tt class="py-lineno">  89</tt>  <tt class="py-line">                    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Found the start of a record:\n"</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L90"></a><tt class="py-lineno">  90</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L91"></a><tt class="py-lineno">  91</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-27" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-27', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L92"></a><tt class="py-lineno">  92</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt> <tt class="py-op">==</tt> <tt class="py-string">"//"</tt><tt class="py-op">:</tt> </tt>
<a name="L93"></a><tt class="py-lineno">  93</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L94"></a><tt class="py-lineno">  94</tt>  <tt class="py-line">                    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Skipping // marking end of last record"</tt><tt class="py-op">)</tt> </tt>
<a name="L95"></a><tt class="py-lineno">  95</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt> <tt class="py-op">==</tt> <tt class="py-string">""</tt><tt class="py-op">:</tt> </tt>
<a name="L96"></a><tt class="py-lineno">  96</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L97"></a><tt class="py-lineno">  97</tt>  <tt class="py-line">                    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Skipping blank line before record"</tt><tt class="py-op">)</tt> </tt>
<a name="L98"></a><tt class="py-lineno">  98</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L99"></a><tt class="py-lineno">  99</tt>  <tt class="py-line">                <tt class="py-comment"># Ignore any header before the first ID/LOCUS line.</tt> </tt>
<a name="L100"></a><tt class="py-lineno"> 100</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L101"></a><tt class="py-lineno"> 101</tt>  <tt class="py-line">                        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Skipping header line before record:\n"</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L102"></a><tt class="py-lineno"> 102</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> </tt>
<a name="L103"></a><tt class="py-lineno"> 103</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">line</tt> </tt>
</div><a name="L104"></a><tt class="py-lineno"> 104</tt>  <tt class="py-line"> </tt>
<a name="InsdcScanner.parse_header"></a><div id="InsdcScanner.parse_header-def"><a name="L105"></a><tt class="py-lineno"> 105</tt> <a class="py-toggle" href="#" id="InsdcScanner.parse_header-toggle" onclick="return toggle('InsdcScanner.parse_header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.InsdcScanner-class.html#parse_header">parse_header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="InsdcScanner.parse_header-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="InsdcScanner.parse_header-expanded"><a name="L106"></a><tt class="py-lineno"> 106</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return list of strings making up the header</tt> </tt>
<a name="L107"></a><tt class="py-lineno"> 107</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L108"></a><tt class="py-lineno"> 108</tt>  <tt class="py-line"><tt class="py-docstring">        New line characters are removed.</tt> </tt>
<a name="L109"></a><tt class="py-lineno"> 109</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L110"></a><tt class="py-lineno"> 110</tt>  <tt class="py-line"><tt class="py-docstring">        Assumes you have just read in the ID/LOCUS line.</tt> </tt>
<a name="L111"></a><tt class="py-lineno"> 111</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L112"></a><tt class="py-lineno"> 112</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-28" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-28', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-29" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.RECORD_START
Bio.GenBank.Scanner.GenBankScanner.RECORD_START
Bio.GenBank.Scanner.InsdcScanner.RECORD_START" class="py-name" href="#" onclick="return doclink('link-29', 'RECORD_START', 'link-11');">RECORD_START</a></tt><tt class="py-op">,</tt> \ </tt>
<a name="L113"></a><tt class="py-lineno"> 113</tt>  <tt class="py-line">            <tt class="py-string">"Not at start of record"</tt> </tt>
<a name="L114"></a><tt class="py-lineno"> 114</tt>  <tt class="py-line"> </tt>
<a name="L115"></a><tt class="py-lineno"> 115</tt>  <tt class="py-line">        <tt class="py-name">header_lines</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L116"></a><tt class="py-lineno"> 116</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L117"></a><tt class="py-lineno"> 117</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-30" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-30', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L118"></a><tt class="py-lineno"> 118</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L119"></a><tt class="py-lineno"> 119</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Premature end of line during sequence data"</tt><tt class="py-op">)</tt> </tt>
<a name="L120"></a><tt class="py-lineno"> 120</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-31" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-31', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L121"></a><tt class="py-lineno"> 121</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-32" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner.GenBankScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner.InsdcScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner._ImgtScanner.FEATURE_START_MARKERS" class="py-name" href="#" onclick="return doclink('link-32', 'FEATURE_START_MARKERS', 'link-13');">FEATURE_START_MARKERS</a></tt><tt class="py-op">:</tt> </tt>
<a name="L122"></a><tt class="py-lineno"> 122</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt><tt class="py-op">:</tt> </tt>
<a name="L123"></a><tt class="py-lineno"> 123</tt>  <tt class="py-line">                    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Found feature table"</tt><tt class="py-op">)</tt> </tt>
<a name="L124"></a><tt class="py-lineno"> 124</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L125"></a><tt class="py-lineno"> 125</tt>  <tt class="py-line">            <tt class="py-comment"># if line[:self.HEADER_WIDTH]==self.FEATURE_START_MARKER[:self.HEADER_WIDTH]:</tt> </tt>
<a name="L126"></a><tt class="py-lineno"> 126</tt>  <tt class="py-line">            <tt class="py-comment">#    if self.debug : print("Found header table (?)")</tt> </tt>
<a name="L127"></a><tt class="py-lineno"> 127</tt>  <tt class="py-line">            <tt class="py-comment">#    break</tt> </tt>
<a name="L128"></a><tt class="py-lineno"> 128</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-33" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-33', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-34" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-34', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-35" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.GenBankScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.InsdcScanner.SEQUENCE_HEADERS" class="py-name" href="#" onclick="return doclink('link-35', 'SEQUENCE_HEADERS', 'link-17');">SEQUENCE_HEADERS</a></tt><tt class="py-op">:</tt> </tt>
<a name="L129"></a><tt class="py-lineno"> 129</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt><tt class="py-op">:</tt> </tt>
<a name="L130"></a><tt class="py-lineno"> 130</tt>  <tt class="py-line">                    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Found start of sequence"</tt><tt class="py-op">)</tt> </tt>
<a name="L131"></a><tt class="py-lineno"> 131</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L132"></a><tt class="py-lineno"> 132</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt> <tt class="py-op">==</tt> <tt class="py-string">"//"</tt><tt class="py-op">:</tt> </tt>
<a name="L133"></a><tt class="py-lineno"> 133</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Premature end of sequence data marker '//' found"</tt><tt class="py-op">)</tt> </tt>
<a name="L134"></a><tt class="py-lineno"> 134</tt>  <tt class="py-line">            <tt class="py-name">header_lines</tt><tt class="py-op">.</tt><tt id="link-36" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-36', 'append', 'link-36');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L135"></a><tt class="py-lineno"> 135</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> </tt>
<a name="L136"></a><tt class="py-lineno"> 136</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">header_lines</tt> </tt>
</div><a name="L137"></a><tt class="py-lineno"> 137</tt>  <tt class="py-line"> </tt>
<a name="InsdcScanner.parse_features"></a><div id="InsdcScanner.parse_features-def"><a name="L138"></a><tt class="py-lineno"> 138</tt> <a class="py-toggle" href="#" id="InsdcScanner.parse_features-toggle" onclick="return toggle('InsdcScanner.parse_features');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.InsdcScanner-class.html#parse_features">parse_features</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">skip</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="InsdcScanner.parse_features-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="InsdcScanner.parse_features-expanded"><a name="L139"></a><tt class="py-lineno"> 139</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return list of tuples for the features (if present)</tt> </tt>
<a name="L140"></a><tt class="py-lineno"> 140</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L141"></a><tt class="py-lineno"> 141</tt>  <tt class="py-line"><tt class="py-docstring">        Each feature is returned as a tuple (key, location, qualifiers)</tt> </tt>
<a name="L142"></a><tt class="py-lineno"> 142</tt>  <tt class="py-line"><tt class="py-docstring">        where key and location are strings (e.g. "CDS" and</tt> </tt>
<a name="L143"></a><tt class="py-lineno"> 143</tt>  <tt class="py-line"><tt class="py-docstring">        "complement(join(490883..490885,1..879))") while qualifiers</tt> </tt>
<a name="L144"></a><tt class="py-lineno"> 144</tt>  <tt class="py-line"><tt class="py-docstring">        is a list of two string tuples (feature qualifier keys and values).</tt> </tt>
<a name="L145"></a><tt class="py-lineno"> 145</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L146"></a><tt class="py-lineno"> 146</tt>  <tt class="py-line"><tt class="py-docstring">        Assumes you have already read to the start of the features table.</tt> </tt>
<a name="L147"></a><tt class="py-lineno"> 147</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L148"></a><tt class="py-lineno"> 148</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-37" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-37', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-38" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner.GenBankScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner.InsdcScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner._ImgtScanner.FEATURE_START_MARKERS" class="py-name" href="#" onclick="return doclink('link-38', 'FEATURE_START_MARKERS', 'link-13');">FEATURE_START_MARKERS</a></tt><tt class="py-op">:</tt> </tt>
<a name="L149"></a><tt class="py-lineno"> 149</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt><tt class="py-op">:</tt> </tt>
<a name="L150"></a><tt class="py-lineno"> 150</tt>  <tt class="py-line">                <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Didn't find any feature table"</tt><tt class="py-op">)</tt> </tt>
<a name="L151"></a><tt class="py-lineno"> 151</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L152"></a><tt class="py-lineno"> 152</tt>  <tt class="py-line"> </tt>
<a name="L153"></a><tt class="py-lineno"> 153</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-39" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-39', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-40" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner.GenBankScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner.InsdcScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner._ImgtScanner.FEATURE_START_MARKERS" class="py-name" href="#" onclick="return doclink('link-40', 'FEATURE_START_MARKERS', 'link-13');">FEATURE_START_MARKERS</a></tt><tt class="py-op">:</tt> </tt>
<a name="L154"></a><tt class="py-lineno"> 154</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-41" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-41', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L155"></a><tt class="py-lineno"> 155</tt>  <tt class="py-line"> </tt>
<a name="L156"></a><tt class="py-lineno"> 156</tt>  <tt class="py-line">        <tt id="link-42" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.features=BioSQL.BioSeq.DBSeqRecord-class.html#features"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-42', 'features', 'link-42');">features</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L157"></a><tt class="py-lineno"> 157</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> </tt>
<a name="L158"></a><tt class="py-lineno"> 158</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L159"></a><tt class="py-lineno"> 159</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L160"></a><tt class="py-lineno"> 160</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Premature end of line during features table"</tt><tt class="py-op">)</tt> </tt>
<a name="L161"></a><tt class="py-lineno"> 161</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-43" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-43', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-44" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-44', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-45" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.GenBankScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.InsdcScanner.SEQUENCE_HEADERS" class="py-name" href="#" onclick="return doclink('link-45', 'SEQUENCE_HEADERS', 'link-17');">SEQUENCE_HEADERS</a></tt><tt class="py-op">:</tt> </tt>
<a name="L162"></a><tt class="py-lineno"> 162</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt><tt class="py-op">:</tt> </tt>
<a name="L163"></a><tt class="py-lineno"> 163</tt>  <tt class="py-line">                    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Found start of sequence"</tt><tt class="py-op">)</tt> </tt>
<a name="L164"></a><tt class="py-lineno"> 164</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L165"></a><tt class="py-lineno"> 165</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-46" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-46', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L166"></a><tt class="py-lineno"> 166</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt> <tt class="py-op">==</tt> <tt class="py-string">"//"</tt><tt class="py-op">:</tt> </tt>
<a name="L167"></a><tt class="py-lineno"> 167</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Premature end of features table, marker '//' found"</tt><tt class="py-op">)</tt> </tt>
<a name="L168"></a><tt class="py-lineno"> 168</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-47" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.FEATURE_END_MARKERS
Bio.GenBank.Scanner.GenBankScanner.FEATURE_END_MARKERS
Bio.GenBank.Scanner.InsdcScanner.FEATURE_END_MARKERS" class="py-name" href="#" onclick="return doclink('link-47', 'FEATURE_END_MARKERS', 'link-14');">FEATURE_END_MARKERS</a></tt><tt class="py-op">:</tt> </tt>
<a name="L169"></a><tt class="py-lineno"> 169</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt><tt class="py-op">:</tt> </tt>
<a name="L170"></a><tt class="py-lineno"> 170</tt>  <tt class="py-line">                    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Found end of features"</tt><tt class="py-op">)</tt> </tt>
<a name="L171"></a><tt class="py-lineno"> 171</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-48" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-48', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L172"></a><tt class="py-lineno"> 172</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L173"></a><tt class="py-lineno"> 173</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-49" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-49', 'FEATURE_QUALIFIER_INDENT', 'link-15');">FEATURE_QUALIFIER_INDENT</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-50" class="py-name" targets="Method Bio.Seq.Seq.strip()=Bio.Seq.Seq-class.html#strip"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-50', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">""</tt><tt class="py-op">:</tt> </tt>
<a name="L174"></a><tt class="py-lineno"> 174</tt>  <tt class="py-line">                <tt class="py-comment"># This is an empty feature line between qualifiers. Empty</tt> </tt>
<a name="L175"></a><tt class="py-lineno"> 175</tt>  <tt class="py-line">                <tt class="py-comment"># feature lines within qualifiers are handled below (ignored).</tt> </tt>
<a name="L176"></a><tt class="py-lineno"> 176</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-51" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-51', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L177"></a><tt class="py-lineno"> 177</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L178"></a><tt class="py-lineno"> 178</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-52" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-52', 'FEATURE_QUALIFIER_INDENT', 'link-15');">FEATURE_QUALIFIER_INDENT</a></tt><tt class="py-op">:</tt> </tt>
<a name="L179"></a><tt class="py-lineno"> 179</tt>  <tt class="py-line">                <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"line too short to contain a feature: %r"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">,</tt> </tt>
<a name="L180"></a><tt class="py-lineno"> 180</tt>  <tt class="py-line">                              <tt id="link-53" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-53', 'BiopythonParserWarning', 'link-10');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L181"></a><tt class="py-lineno"> 181</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-54" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-54', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L182"></a><tt class="py-lineno"> 182</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L183"></a><tt class="py-lineno"> 183</tt>  <tt class="py-line"> </tt>
<a name="L184"></a><tt class="py-lineno"> 184</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">skip</tt><tt class="py-op">:</tt> </tt>
<a name="L185"></a><tt class="py-lineno"> 185</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-55" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-55', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L186"></a><tt class="py-lineno"> 186</tt>  <tt class="py-line">                <tt class="py-keyword">while</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-56" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-56', 'FEATURE_QUALIFIER_INDENT', 'link-15');">FEATURE_QUALIFIER_INDENT</a></tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-57" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_SPACER" class="py-name" href="#" onclick="return doclink('link-57', 'FEATURE_QUALIFIER_SPACER', 'link-16');">FEATURE_QUALIFIER_SPACER</a></tt><tt class="py-op">:</tt> </tt>
<a name="L187"></a><tt class="py-lineno"> 187</tt>  <tt class="py-line">                    <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-58" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-58', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L188"></a><tt class="py-lineno"> 188</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L189"></a><tt class="py-lineno"> 189</tt>  <tt class="py-line">                <tt class="py-comment"># Build up a list of the lines making up this feature:</tt> </tt>
<a name="L190"></a><tt class="py-lineno"> 190</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-59" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-59', 'FEATURE_QUALIFIER_INDENT', 'link-15');">FEATURE_QUALIFIER_INDENT</a></tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-string">" "</tt> \ </tt>
<a name="L191"></a><tt class="py-lineno"> 191</tt>  <tt class="py-line">                        <tt class="py-keyword">and</tt> <tt class="py-string">" "</tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-60" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-60', 'FEATURE_QUALIFIER_INDENT', 'link-15');">FEATURE_QUALIFIER_INDENT</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L192"></a><tt class="py-lineno"> 192</tt>  <tt class="py-line">                    <tt class="py-comment"># The feature table design enforces a length limit on the feature keys.</tt> </tt>
<a name="L193"></a><tt class="py-lineno"> 193</tt>  <tt class="py-line">                    <tt class="py-comment"># Some third party files (e.g. IGMT's EMBL like files) solve this by</tt> </tt>
<a name="L194"></a><tt class="py-lineno"> 194</tt>  <tt class="py-line">                    <tt class="py-comment"># over indenting the location and qualifiers.</tt> </tt>
<a name="L195"></a><tt class="py-lineno"> 195</tt>  <tt class="py-line">                    <tt id="link-61" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.feature_key()=Bio.GenBank._FeatureConsumer-class.html#feature_key,Method Bio.GenBank._RecordConsumer.feature_key()=Bio.GenBank._RecordConsumer-class.html#feature_key"><a title="Bio.GenBank._FeatureConsumer.feature_key
Bio.GenBank._RecordConsumer.feature_key" class="py-name" href="#" onclick="return doclink('link-61', 'feature_key', 'link-61');">feature_key</a></tt><tt class="py-op">,</tt> <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-62" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-62', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-63" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-63', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L196"></a><tt class="py-lineno"> 196</tt>  <tt class="py-line">                    <tt class="py-name">feature_lines</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">line</tt><tt class="py-op">]</tt> </tt>
<a name="L197"></a><tt class="py-lineno"> 197</tt>  <tt class="py-line">                    <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Overindented %s feature?"</tt> <tt class="py-op">%</tt> <tt id="link-64" class="py-name"><a title="Bio.GenBank._FeatureConsumer.feature_key
Bio.GenBank._RecordConsumer.feature_key" class="py-name" href="#" onclick="return doclink('link-64', 'feature_key', 'link-61');">feature_key</a></tt><tt class="py-op">,</tt> </tt>
<a name="L198"></a><tt class="py-lineno"> 198</tt>  <tt class="py-line">                                  <tt id="link-65" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-65', 'BiopythonParserWarning', 'link-10');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L199"></a><tt class="py-lineno"> 199</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L200"></a><tt class="py-lineno"> 200</tt>  <tt class="py-line">                    <tt id="link-66" class="py-name"><a title="Bio.GenBank._FeatureConsumer.feature_key
Bio.GenBank._RecordConsumer.feature_key" class="py-name" href="#" onclick="return doclink('link-66', 'feature_key', 'link-61');">feature_key</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-67" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-67', 'FEATURE_QUALIFIER_INDENT', 'link-15');">FEATURE_QUALIFIER_INDENT</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-68" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-68', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L201"></a><tt class="py-lineno"> 201</tt>  <tt class="py-line">                    <tt class="py-name">feature_lines</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-69" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-69', 'FEATURE_QUALIFIER_INDENT', 'link-15');">FEATURE_QUALIFIER_INDENT</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L202"></a><tt class="py-lineno"> 202</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-70" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-70', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L203"></a><tt class="py-lineno"> 203</tt>  <tt class="py-line">                <tt class="py-keyword">while</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-71" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-71', 'FEATURE_QUALIFIER_INDENT', 'link-15');">FEATURE_QUALIFIER_INDENT</a></tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-72" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_SPACER" class="py-name" href="#" onclick="return doclink('link-72', 'FEATURE_QUALIFIER_SPACER', 'link-16');">FEATURE_QUALIFIER_SPACER</a></tt> \ </tt>
<a name="L204"></a><tt class="py-lineno"> 204</tt>  <tt class="py-line">                        <tt class="py-keyword">or</tt> <tt class="py-op">(</tt><tt class="py-name">line</tt> <tt class="py-op">!=</tt> <tt class="py-string">''</tt> <tt class="py-keyword">and</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-73" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-73', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># cope with blank lines in the midst of a feature</tt> </tt>
<a name="L205"></a><tt class="py-lineno"> 205</tt>  <tt class="py-line">                    <tt class="py-comment"># Use strip to remove any harmless trailing white space AND and leading</tt> </tt>
<a name="L206"></a><tt class="py-lineno"> 206</tt>  <tt class="py-line">                    <tt class="py-comment"># white space (e.g. out of spec files with too much indentation)</tt> </tt>
<a name="L207"></a><tt class="py-lineno"> 207</tt>  <tt class="py-line">                    <tt class="py-name">feature_lines</tt><tt class="py-op">.</tt><tt id="link-74" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-74', 'append', 'link-36');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-75" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-75', 'FEATURE_QUALIFIER_INDENT', 'link-15');">FEATURE_QUALIFIER_INDENT</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-76" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-76', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L208"></a><tt class="py-lineno"> 208</tt>  <tt class="py-line">                    <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-77" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-77', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L209"></a><tt class="py-lineno"> 209</tt>  <tt class="py-line">                <tt id="link-78" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-78', 'features', 'link-42');">features</a></tt><tt class="py-op">.</tt><tt id="link-79" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-79', 'append', 'link-36');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-80" class="py-name" targets="Method Bio.GenBank.Scanner.InsdcScanner.parse_feature()=Bio.GenBank.Scanner.InsdcScanner-class.html#parse_feature"><a title="Bio.GenBank.Scanner.InsdcScanner.parse_feature" class="py-name" href="#" onclick="return doclink('link-80', 'parse_feature', 'link-80');">parse_feature</a></tt><tt class="py-op">(</tt><tt id="link-81" class="py-name"><a title="Bio.GenBank._FeatureConsumer.feature_key
Bio.GenBank._RecordConsumer.feature_key" class="py-name" href="#" onclick="return doclink('link-81', 'feature_key', 'link-61');">feature_key</a></tt><tt class="py-op">,</tt> <tt class="py-name">feature_lines</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L210"></a><tt class="py-lineno"> 210</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> </tt>
<a name="L211"></a><tt class="py-lineno"> 211</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-82" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-82', 'features', 'link-42');">features</a></tt> </tt>
</div><a name="L212"></a><tt class="py-lineno"> 212</tt>  <tt class="py-line"> </tt>
<a name="InsdcScanner.parse_feature"></a><div id="InsdcScanner.parse_feature-def"><a name="L213"></a><tt class="py-lineno"> 213</tt> <a class="py-toggle" href="#" id="InsdcScanner.parse_feature-toggle" onclick="return toggle('InsdcScanner.parse_feature');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.InsdcScanner-class.html#parse_feature">parse_feature</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">feature_key</tt><tt class="py-op">,</tt> <tt class="py-param">lines</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="InsdcScanner.parse_feature-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="InsdcScanner.parse_feature-expanded"><a name="L214"></a><tt class="py-lineno"> 214</tt>  <tt class="py-line">        <tt class="py-docstring">r"""Expects a feature as a list of strings, returns a tuple (key, location, qualifiers)</tt> </tt>
<a name="L215"></a><tt class="py-lineno"> 215</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L216"></a><tt class="py-lineno"> 216</tt>  <tt class="py-line"><tt class="py-docstring">        For example given this GenBank feature::</tt> </tt>
<a name="L217"></a><tt class="py-lineno"> 217</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L218"></a><tt class="py-lineno"> 218</tt>  <tt class="py-line"><tt class="py-docstring">             CDS             complement(join(490883..490885,1..879))</tt> </tt>
<a name="L219"></a><tt class="py-lineno"> 219</tt>  <tt class="py-line"><tt class="py-docstring">                             /locus_tag="NEQ001"</tt> </tt>
<a name="L220"></a><tt class="py-lineno"> 220</tt>  <tt class="py-line"><tt class="py-docstring">                             /note="conserved hypothetical [Methanococcus jannaschii];</tt> </tt>
<a name="L221"></a><tt class="py-lineno"> 221</tt>  <tt class="py-line"><tt class="py-docstring">                             COG1583:Uncharacterized ACR; IPR001472:Bipartite nuclear</tt> </tt>
<a name="L222"></a><tt class="py-lineno"> 222</tt>  <tt class="py-line"><tt class="py-docstring">                             localization signal; IPR002743: Protein of unknown</tt> </tt>
<a name="L223"></a><tt class="py-lineno"> 223</tt>  <tt class="py-line"><tt class="py-docstring">                             function DUF57"</tt> </tt>
<a name="L224"></a><tt class="py-lineno"> 224</tt>  <tt class="py-line"><tt class="py-docstring">                             /codon_start=1</tt> </tt>
<a name="L225"></a><tt class="py-lineno"> 225</tt>  <tt class="py-line"><tt class="py-docstring">                             /transl_table=11</tt> </tt>
<a name="L226"></a><tt class="py-lineno"> 226</tt>  <tt class="py-line"><tt class="py-docstring">                             /product="hypothetical protein"</tt> </tt>
<a name="L227"></a><tt class="py-lineno"> 227</tt>  <tt class="py-line"><tt class="py-docstring">                             /protein_id="NP_963295.1"</tt> </tt>
<a name="L228"></a><tt class="py-lineno"> 228</tt>  <tt class="py-line"><tt class="py-docstring">                             /db_xref="GI:41614797"</tt> </tt>
<a name="L229"></a><tt class="py-lineno"> 229</tt>  <tt class="py-line"><tt class="py-docstring">                             /db_xref="GeneID:2732620"</tt> </tt>
<a name="L230"></a><tt class="py-lineno"> 230</tt>  <tt class="py-line"><tt class="py-docstring">                             /translation="MRLLLELKALNSIDKKQLSNYLIQGFIYNILKNTEYSWLHNWKK</tt> </tt>
<a name="L231"></a><tt class="py-lineno"> 231</tt>  <tt class="py-line"><tt class="py-docstring">                             EKYFNFTLIPKKDIIENKRYYLIISSPDKRFIEVLHNKIKDLDIITIGLAQFQLRKTK</tt> </tt>
<a name="L232"></a><tt class="py-lineno"> 232</tt>  <tt class="py-line"><tt class="py-docstring">                             KFDPKLRFPWVTITPIVLREGKIVILKGDKYYKVFVKRLEELKKYNLIKKKEPILEEP</tt> </tt>
<a name="L233"></a><tt class="py-lineno"> 233</tt>  <tt class="py-line"><tt class="py-docstring">                             IEISLNQIKDGWKIIDVKDRYYDFRNKSFSAFSNWLRDLKEQSLRKYNNFCGKNFYFE</tt> </tt>
<a name="L234"></a><tt class="py-lineno"> 234</tt>  <tt class="py-line"><tt class="py-docstring">                             EAIFEGFTFYKTVSIRIRINRGEAVYIGTLWKELNVYRKLDKEEREFYKFLYDCGLGS</tt> </tt>
<a name="L235"></a><tt class="py-lineno"> 235</tt>  <tt class="py-line"><tt class="py-docstring">                             LNSMGFGFVNTKKNSAR"</tt> </tt>
<a name="L236"></a><tt class="py-lineno"> 236</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L237"></a><tt class="py-lineno"> 237</tt>  <tt class="py-line"><tt class="py-docstring">        Then should give input key="CDS" and the rest of the data as a list of strings</tt> </tt>
<a name="L238"></a><tt class="py-lineno"> 238</tt>  <tt class="py-line"><tt class="py-docstring">        lines=["complement(join(490883..490885,1..879))", ..., "LNSMGFGFVNTKKNSAR"]</tt> </tt>
<a name="L239"></a><tt class="py-lineno"> 239</tt>  <tt class="py-line"><tt class="py-docstring">        where the leading spaces and trailing newlines have been removed.</tt> </tt>
<a name="L240"></a><tt class="py-lineno"> 240</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L241"></a><tt class="py-lineno"> 241</tt>  <tt class="py-line"><tt class="py-docstring">        Returns tuple containing: (key as string, location string, qualifiers as list)</tt> </tt>
<a name="L242"></a><tt class="py-lineno"> 242</tt>  <tt class="py-line"><tt class="py-docstring">        as follows for this example:</tt> </tt>
<a name="L243"></a><tt class="py-lineno"> 243</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L244"></a><tt class="py-lineno"> 244</tt>  <tt class="py-line"><tt class="py-docstring">        key = "CDS", string</tt> </tt>
<a name="L245"></a><tt class="py-lineno"> 245</tt>  <tt class="py-line"><tt class="py-docstring">        location = "complement(join(490883..490885,1..879))", string</tt> </tt>
<a name="L246"></a><tt class="py-lineno"> 246</tt>  <tt class="py-line"><tt class="py-docstring">        qualifiers = list of string tuples:</tt> </tt>
<a name="L247"></a><tt class="py-lineno"> 247</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L248"></a><tt class="py-lineno"> 248</tt>  <tt class="py-line"><tt class="py-docstring">        [('locus_tag', '"NEQ001"'),</tt> </tt>
<a name="L249"></a><tt class="py-lineno"> 249</tt>  <tt class="py-line"><tt class="py-docstring">         ('note', '"conserved hypothetical [Methanococcus jannaschii];\nCOG1583:..."'),</tt> </tt>
<a name="L250"></a><tt class="py-lineno"> 250</tt>  <tt class="py-line"><tt class="py-docstring">         ('codon_start', '1'),</tt> </tt>
<a name="L251"></a><tt class="py-lineno"> 251</tt>  <tt class="py-line"><tt class="py-docstring">         ('transl_table', '11'),</tt> </tt>
<a name="L252"></a><tt class="py-lineno"> 252</tt>  <tt class="py-line"><tt class="py-docstring">         ('product', '"hypothetical protein"'),</tt> </tt>
<a name="L253"></a><tt class="py-lineno"> 253</tt>  <tt class="py-line"><tt class="py-docstring">         ('protein_id', '"NP_963295.1"'),</tt> </tt>
<a name="L254"></a><tt class="py-lineno"> 254</tt>  <tt class="py-line"><tt class="py-docstring">         ('db_xref', '"GI:41614797"'),</tt> </tt>
<a name="L255"></a><tt class="py-lineno"> 255</tt>  <tt class="py-line"><tt class="py-docstring">         ('db_xref', '"GeneID:2732620"'),</tt> </tt>
<a name="L256"></a><tt class="py-lineno"> 256</tt>  <tt class="py-line"><tt class="py-docstring">         ('translation', '"MRLLLELKALNSIDKKQLSNYLIQGFIYNILKNTEYSWLHNWKK\nEKYFNFT..."')]</tt> </tt>
<a name="L257"></a><tt class="py-lineno"> 257</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L258"></a><tt class="py-lineno"> 258</tt>  <tt class="py-line"><tt class="py-docstring">        In the above example, the "note" and "translation" were edited for compactness,</tt> </tt>
<a name="L259"></a><tt class="py-lineno"> 259</tt>  <tt class="py-line"><tt class="py-docstring">        and they would contain multiple new line characters (displayed above as \n)</tt> </tt>
<a name="L260"></a><tt class="py-lineno"> 260</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L261"></a><tt class="py-lineno"> 261</tt>  <tt class="py-line"><tt class="py-docstring">        If a qualifier is quoted (in this case, everything except codon_start and</tt> </tt>
<a name="L262"></a><tt class="py-lineno"> 262</tt>  <tt class="py-line"><tt class="py-docstring">        transl_table) then the quotes are NOT removed.</tt> </tt>
<a name="L263"></a><tt class="py-lineno"> 263</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L264"></a><tt class="py-lineno"> 264</tt>  <tt class="py-line"><tt class="py-docstring">        Note that no whitespace is removed.</tt> </tt>
<a name="L265"></a><tt class="py-lineno"> 265</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L266"></a><tt class="py-lineno"> 266</tt>  <tt class="py-line">        <tt class="py-comment"># Skip any blank lines</tt> </tt>
<a name="L267"></a><tt class="py-lineno"> 267</tt>  <tt class="py-line">        <tt class="py-name">iterator</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt id="link-83" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.x=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#x"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-83', 'x', 'link-83');">x</a></tt> <tt class="py-keyword">for</tt> <tt id="link-84" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-84', 'x', 'link-83');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">lines</tt> <tt class="py-keyword">if</tt> <tt id="link-85" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-85', 'x', 'link-83');">x</a></tt><tt class="py-op">)</tt> </tt>
<a name="L268"></a><tt class="py-lineno"> 268</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L269"></a><tt class="py-lineno"> 269</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-86" class="py-name" targets="Method Bio.AlignIO.Interfaces.AlignmentIterator.next()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#next,Method Bio.Blast.NCBIStandalone.Iterator.next()=Bio.Blast.NCBIStandalone.Iterator-class.html#next,Method Bio.Blast.ParseBlastTable.BlastTableReader.next()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#next,Method Bio.DocSQL.IterationCursor.next()=Bio.DocSQL.IterationCursor-class.html#next,Method Bio.File.UndoHandle.next()=Bio.File.UndoHandle-class.html#next,Method Bio.GenBank.Iterator.next()=Bio.GenBank.Iterator-class.html#next,Method Bio.Nexus.Nexus.CharBuffer.next()=Bio.Nexus.Nexus.CharBuffer-class.html#next,Method Bio.PopGen.GenePop.Controller._FileIterator.next()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#next,Method Bio.SeqIO.Interfaces.SequenceIterator.next()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#next,Method Bio.bgzf.BgzfReader.next()=Bio.bgzf.BgzfReader-class.html#next"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-86', 'next', 'link-86');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">iterator</tt><tt class="py-op">)</tt> </tt>
<a name="L270"></a><tt class="py-lineno"> 270</tt>  <tt class="py-line"> </tt>
<a name="L271"></a><tt class="py-lineno"> 271</tt>  <tt class="py-line">            <tt class="py-name">feature_location</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-87" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-87', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L272"></a><tt class="py-lineno"> 272</tt>  <tt class="py-line">            <tt class="py-keyword">while</tt> <tt class="py-name">feature_location</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">","</tt><tt class="py-op">:</tt> </tt>
<a name="L273"></a><tt class="py-lineno"> 273</tt>  <tt class="py-line">                <tt class="py-comment"># Multiline location, still more to come!</tt> </tt>
<a name="L274"></a><tt class="py-lineno"> 274</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-88" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-88', 'next', 'link-86');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">iterator</tt><tt class="py-op">)</tt> </tt>
<a name="L275"></a><tt class="py-lineno"> 275</tt>  <tt class="py-line">                <tt class="py-name">feature_location</tt> <tt class="py-op">+=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-89" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-89', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L276"></a><tt class="py-lineno"> 276</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">feature_location</tt><tt class="py-op">.</tt><tt id="link-90" class="py-name" targets="Method Bio.Crystal.Chain.count()=Bio.Crystal.Chain-class.html#count,Method Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count()=Bio.NeuralNetwork.Gene.Pattern.PatternRepository-class.html#count,Method Bio.Seq.MutableSeq.count()=Bio.Seq.MutableSeq-class.html#count,Method Bio.Seq.Seq.count()=Bio.Seq.Seq-class.html#count,Method Bio.Seq.UnknownSeq.count()=Bio.Seq.UnknownSeq-class.html#count,Method Bio.motifs.Instances.count()=Bio.motifs.Instances-class.html#count"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-90', 'count', 'link-90');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">"("</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-name">feature_location</tt><tt class="py-op">.</tt><tt id="link-91" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-91', 'count', 'link-90');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">")"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L277"></a><tt class="py-lineno"> 277</tt>  <tt class="py-line">                <tt class="py-comment"># Including the prev line in warning would be more explicit,</tt> </tt>
<a name="L278"></a><tt class="py-lineno"> 278</tt>  <tt class="py-line">                <tt class="py-comment"># but this way get one-and-only-one warning shown by default:</tt> </tt>
<a name="L279"></a><tt class="py-lineno"> 279</tt>  <tt class="py-line">                <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Non-standard feature line wrapping (didn't break on comma)?"</tt><tt class="py-op">,</tt> </tt>
<a name="L280"></a><tt class="py-lineno"> 280</tt>  <tt class="py-line">                              <tt id="link-92" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-92', 'BiopythonParserWarning', 'link-10');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L281"></a><tt class="py-lineno"> 281</tt>  <tt class="py-line">                <tt class="py-keyword">while</tt> <tt class="py-name">feature_location</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">","</tt> <tt class="py-keyword">or</tt> <tt class="py-name">feature_location</tt><tt class="py-op">.</tt><tt id="link-93" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-93', 'count', 'link-90');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">"("</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-name">feature_location</tt><tt class="py-op">.</tt><tt id="link-94" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-94', 'count', 'link-90');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">")"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L282"></a><tt class="py-lineno"> 282</tt>  <tt class="py-line">                    <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-95" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-95', 'next', 'link-86');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">iterator</tt><tt class="py-op">)</tt> </tt>
<a name="L283"></a><tt class="py-lineno"> 283</tt>  <tt class="py-line">                    <tt class="py-name">feature_location</tt> <tt class="py-op">+=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-96" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-96', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L284"></a><tt class="py-lineno"> 284</tt>  <tt class="py-line"> </tt>
<a name="L285"></a><tt class="py-lineno"> 285</tt>  <tt class="py-line">            <tt class="py-name">qualifiers</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L286"></a><tt class="py-lineno"> 286</tt>  <tt class="py-line"> </tt>
<a name="L287"></a><tt class="py-lineno"> 287</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">line_number</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt class="py-name">iterator</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L288"></a><tt class="py-lineno"> 288</tt>  <tt class="py-line">                <tt class="py-comment"># check for extra wrapping of the location closing parentheses</tt> </tt>
<a name="L289"></a><tt class="py-lineno"> 289</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">line_number</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt> <tt class="py-keyword">and</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-97" class="py-name" targets="Method Bio.Seq.Seq.startswith()=Bio.Seq.Seq-class.html#startswith"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-97', 'startswith', 'link-97');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">")"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L290"></a><tt class="py-lineno"> 290</tt>  <tt class="py-line">                    <tt class="py-name">feature_location</tt> <tt class="py-op">+=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-98" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-98', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L291"></a><tt class="py-lineno"> 291</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"/"</tt><tt class="py-op">:</tt> </tt>
<a name="L292"></a><tt class="py-lineno"> 292</tt>  <tt class="py-line">                    <tt class="py-comment"># New qualifier</tt> </tt>
<a name="L293"></a><tt class="py-lineno"> 293</tt>  <tt class="py-line">                    <tt id="link-99" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-99', 'i', 'link-99');">i</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-100" class="py-name" targets="Method Bio.NeuralNetwork.Gene.Motif.MotifFinder.find()=Bio.NeuralNetwork.Gene.Motif.MotifFinder-class.html#find,Method Bio.NeuralNetwork.Gene.Schema.SchemaFinder.find()=Bio.NeuralNetwork.Gene.Schema.SchemaFinder-class.html#find,Method Bio.NeuralNetwork.Gene.Signature.SignatureFinder.find()=Bio.NeuralNetwork.Gene.Signature.SignatureFinder-class.html#find,Method Bio.Seq.Seq.find()=Bio.Seq.Seq-class.html#find,Function Bio.triefind.find()=Bio.triefind-module.html#find"><a title="Bio.NeuralNetwork.Gene.Motif.MotifFinder.find
Bio.NeuralNetwork.Gene.Schema.SchemaFinder.find
Bio.NeuralNetwork.Gene.Signature.SignatureFinder.find
Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-100', 'find', 'link-100');">find</a></tt><tt class="py-op">(</tt><tt class="py-string">"="</tt><tt class="py-op">)</tt> </tt>
<a name="L294"></a><tt class="py-lineno"> 294</tt>  <tt class="py-line">                    <tt class="py-name">key</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt id="link-101" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-101', 'i', 'link-99');">i</a></tt><tt class="py-op">]</tt>  <tt class="py-comment"># does not work if i==-1</tt> </tt>
<a name="L295"></a><tt class="py-lineno"> 295</tt>  <tt class="py-line">                    <tt id="link-102" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.value()=Bio.Phylo.PhyloXMLIO.Writer-class.html#value"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-102', 'value', 'link-102');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt id="link-103" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-103', 'i', 'link-99');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt>  <tt class="py-comment"># we ignore 'value' if i==-1</tt> </tt>
<a name="L296"></a><tt class="py-lineno"> 296</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt id="link-104" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-104', 'i', 'link-99');">i</a></tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L297"></a><tt class="py-lineno"> 297</tt>  <tt class="py-line">                        <tt class="py-comment"># Qualifier with no key, e.g. /pseudo</tt> </tt>
<a name="L298"></a><tt class="py-lineno"> 298</tt>  <tt class="py-line">                        <tt class="py-name">key</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L299"></a><tt class="py-lineno"> 299</tt>  <tt class="py-line">                        <tt class="py-name">qualifiers</tt><tt class="py-op">.</tt><tt id="link-105" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-105', 'append', 'link-36');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L300"></a><tt class="py-lineno"> 300</tt>  <tt class="py-line">                    <tt class="py-keyword">elif</tt> <tt class="py-keyword">not</tt> <tt id="link-106" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-106', 'value', 'link-102');">value</a></tt><tt class="py-op">:</tt> </tt>
<a name="L301"></a><tt class="py-lineno"> 301</tt>  <tt class="py-line">                        <tt class="py-comment"># ApE can output /note=</tt> </tt>
<a name="L302"></a><tt class="py-lineno"> 302</tt>  <tt class="py-line">                        <tt class="py-name">qualifiers</tt><tt class="py-op">.</tt><tt id="link-107" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-107', 'append', 'link-36');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L303"></a><tt class="py-lineno"> 303</tt>  <tt class="py-line">                    <tt class="py-keyword">elif</tt> <tt id="link-108" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-108', 'value', 'link-102');">value</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'"'</tt><tt class="py-op">:</tt> </tt>
<a name="L304"></a><tt class="py-lineno"> 304</tt>  <tt class="py-line">                        <tt class="py-comment"># One single quote</tt> </tt>
<a name="L305"></a><tt class="py-lineno"> 305</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt><tt class="py-op">:</tt> </tt>
<a name="L306"></a><tt class="py-lineno"> 306</tt>  <tt class="py-line">                            <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Single quote %s:%s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt id="link-109" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-109', 'value', 'link-102');">value</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L307"></a><tt class="py-lineno"> 307</tt>  <tt class="py-line">                        <tt class="py-comment"># DO NOT remove the quote...</tt> </tt>
<a name="L308"></a><tt class="py-lineno"> 308</tt>  <tt class="py-line">                        <tt class="py-name">qualifiers</tt><tt class="py-op">.</tt><tt id="link-110" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-110', 'append', 'link-36');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt id="link-111" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-111', 'value', 'link-102');">value</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L309"></a><tt class="py-lineno"> 309</tt>  <tt class="py-line">                    <tt class="py-keyword">elif</tt> <tt id="link-112" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-112', 'value', 'link-102');">value</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'"'</tt><tt class="py-op">:</tt> </tt>
<a name="L310"></a><tt class="py-lineno"> 310</tt>  <tt class="py-line">                        <tt class="py-comment"># Quoted...</tt> </tt>
<a name="L311"></a><tt class="py-lineno"> 311</tt>  <tt class="py-line">                        <tt class="py-name">value_list</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-113" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-113', 'value', 'link-102');">value</a></tt><tt class="py-op">]</tt> </tt>
<a name="L312"></a><tt class="py-lineno"> 312</tt>  <tt class="py-line">                        <tt class="py-keyword">while</tt> <tt class="py-name">value_list</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-string">'"'</tt><tt class="py-op">:</tt> </tt>
<a name="L313"></a><tt class="py-lineno"> 313</tt>  <tt class="py-line">                            <tt class="py-name">value_list</tt><tt class="py-op">.</tt><tt id="link-114" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-114', 'append', 'link-36');">append</a></tt><tt class="py-op">(</tt><tt id="link-115" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-115', 'next', 'link-86');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">iterator</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L314"></a><tt class="py-lineno"> 314</tt>  <tt class="py-line">                        <tt id="link-116" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-116', 'value', 'link-102');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'\n'</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">value_list</tt><tt class="py-op">)</tt> </tt>
<a name="L315"></a><tt class="py-lineno"> 315</tt>  <tt class="py-line">                        <tt class="py-comment"># DO NOT remove the quotes...</tt> </tt>
<a name="L316"></a><tt class="py-lineno"> 316</tt>  <tt class="py-line">                        <tt class="py-name">qualifiers</tt><tt class="py-op">.</tt><tt id="link-117" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-117', 'append', 'link-36');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt id="link-118" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-118', 'value', 'link-102');">value</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L317"></a><tt class="py-lineno"> 317</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L318"></a><tt class="py-lineno"> 318</tt>  <tt class="py-line">                        <tt class="py-comment"># Unquoted</tt> </tt>
<a name="L319"></a><tt class="py-lineno"> 319</tt>  <tt class="py-line">                        <tt class="py-comment"># if debug : print("Unquoted line %s:%s" % (key,value))</tt> </tt>
<a name="L320"></a><tt class="py-lineno"> 320</tt>  <tt class="py-line">                        <tt class="py-name">qualifiers</tt><tt class="py-op">.</tt><tt id="link-119" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-119', 'append', 'link-36');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt id="link-120" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-120', 'value', 'link-102');">value</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L321"></a><tt class="py-lineno"> 321</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L322"></a><tt class="py-lineno"> 322</tt>  <tt class="py-line">                    <tt class="py-comment"># Unquoted continuation</tt> </tt>
<a name="L323"></a><tt class="py-lineno"> 323</tt>  <tt class="py-line">                    <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">qualifiers</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt> </tt>
<a name="L324"></a><tt class="py-lineno"> 324</tt>  <tt class="py-line">                    <tt class="py-keyword">assert</tt> <tt class="py-name">key</tt> <tt class="py-op">==</tt> <tt class="py-name">qualifiers</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L325"></a><tt class="py-lineno"> 325</tt>  <tt class="py-line">                    <tt class="py-comment"># if debug : print("Unquoted Cont %s:%s" % (key, line))</tt> </tt>
<a name="L326"></a><tt class="py-lineno"> 326</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">qualifiers</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L327"></a><tt class="py-lineno"> 327</tt>  <tt class="py-line">                        <tt class="py-keyword">raise</tt> <tt class="py-name">StopIteration</tt> </tt>
<a name="L328"></a><tt class="py-lineno"> 328</tt>  <tt class="py-line">                    <tt class="py-name">qualifiers</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">qualifiers</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L329"></a><tt class="py-lineno"> 329</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt id="link-121" class="py-name"><a title="Bio.GenBank._FeatureConsumer.feature_key
Bio.GenBank._RecordConsumer.feature_key" class="py-name" href="#" onclick="return doclink('link-121', 'feature_key', 'link-61');">feature_key</a></tt><tt class="py-op">,</tt> <tt class="py-name">feature_location</tt><tt class="py-op">,</tt> <tt class="py-name">qualifiers</tt><tt class="py-op">)</tt> </tt>
<a name="L330"></a><tt class="py-lineno"> 330</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L331"></a><tt class="py-lineno"> 331</tt>  <tt class="py-line">            <tt class="py-comment"># Bummer</tt> </tt>
<a name="L332"></a><tt class="py-lineno"> 332</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Problem with '%s' feature:\n%s"</tt> </tt>
<a name="L333"></a><tt class="py-lineno"> 333</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-122" class="py-name"><a title="Bio.GenBank._FeatureConsumer.feature_key
Bio.GenBank._RecordConsumer.feature_key" class="py-name" href="#" onclick="return doclink('link-122', 'feature_key', 'link-61');">feature_key</a></tt><tt class="py-op">,</tt> <tt class="py-string">"\n"</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">lines</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L334"></a><tt class="py-lineno"> 334</tt>  <tt class="py-line"> </tt>
<a name="InsdcScanner.parse_footer"></a><div id="InsdcScanner.parse_footer-def"><a name="L335"></a><tt class="py-lineno"> 335</tt> <a class="py-toggle" href="#" id="InsdcScanner.parse_footer-toggle" onclick="return toggle('InsdcScanner.parse_footer');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.InsdcScanner-class.html#parse_footer">parse_footer</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="InsdcScanner.parse_footer-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="InsdcScanner.parse_footer-expanded"><a name="L336"></a><tt class="py-lineno"> 336</tt>  <tt class="py-line">        <tt class="py-docstring">"""returns a tuple containing a list of any misc strings, and the sequence"""</tt> </tt>
<a name="L337"></a><tt class="py-lineno"> 337</tt>  <tt class="py-line">        <tt class="py-comment"># This is a basic bit of code to scan and discard the sequence,</tt> </tt>
<a name="L338"></a><tt class="py-lineno"> 338</tt>  <tt class="py-line">        <tt class="py-comment"># which was useful when developing the sub classes.</tt> </tt>
<a name="L339"></a><tt class="py-lineno"> 339</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-123" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.FEATURE_END_MARKERS
Bio.GenBank.Scanner.GenBankScanner.FEATURE_END_MARKERS
Bio.GenBank.Scanner.InsdcScanner.FEATURE_END_MARKERS" class="py-name" href="#" onclick="return doclink('link-123', 'FEATURE_END_MARKERS', 'link-14');">FEATURE_END_MARKERS</a></tt><tt class="py-op">:</tt> </tt>
<a name="L340"></a><tt class="py-lineno"> 340</tt>  <tt class="py-line">            <tt class="py-keyword">while</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-124" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-124', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-125" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-125', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-126" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.GenBankScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.InsdcScanner.SEQUENCE_HEADERS" class="py-name" href="#" onclick="return doclink('link-126', 'SEQUENCE_HEADERS', 'link-17');">SEQUENCE_HEADERS</a></tt><tt class="py-op">:</tt> </tt>
<a name="L341"></a><tt class="py-lineno"> 341</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-127" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-127', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L342"></a><tt class="py-lineno"> 342</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L343"></a><tt class="py-lineno"> 343</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Premature end of file"</tt><tt class="py-op">)</tt> </tt>
<a name="L344"></a><tt class="py-lineno"> 344</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-128" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-128', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L345"></a><tt class="py-lineno"> 345</tt>  <tt class="py-line"> </tt>
<a name="L346"></a><tt class="py-lineno"> 346</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-129" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-129', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-130" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-130', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-131" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.GenBankScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.InsdcScanner.SEQUENCE_HEADERS" class="py-name" href="#" onclick="return doclink('link-131', 'SEQUENCE_HEADERS', 'link-17');">SEQUENCE_HEADERS</a></tt><tt class="py-op">,</tt> \ </tt>
<a name="L347"></a><tt class="py-lineno"> 347</tt>  <tt class="py-line">            <tt class="py-string">"Not at start of sequence"</tt> </tt>
<a name="L348"></a><tt class="py-lineno"> 348</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L349"></a><tt class="py-lineno"> 349</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-132" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-132', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L350"></a><tt class="py-lineno"> 350</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L351"></a><tt class="py-lineno"> 351</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Premature end of line during sequence data"</tt><tt class="py-op">)</tt> </tt>
<a name="L352"></a><tt class="py-lineno"> 352</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-133" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-133', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L353"></a><tt class="py-lineno"> 353</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt> <tt class="py-op">==</tt> <tt class="py-string">"//"</tt><tt class="py-op">:</tt> </tt>
<a name="L354"></a><tt class="py-lineno"> 354</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L355"></a><tt class="py-lineno"> 355</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> </tt>
<a name="L356"></a><tt class="py-lineno"> 356</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt>  <tt class="py-comment"># Dummy values!</tt> </tt>
</div><a name="L357"></a><tt class="py-lineno"> 357</tt>  <tt class="py-line"> </tt>
<a name="InsdcScanner._feed_first_line"></a><div id="InsdcScanner._feed_first_line-def"><a name="L358"></a><tt class="py-lineno"> 358</tt> <a class="py-toggle" href="#" id="InsdcScanner._feed_first_line-toggle" onclick="return toggle('InsdcScanner._feed_first_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.InsdcScanner-class.html#_feed_first_line">_feed_first_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="InsdcScanner._feed_first_line-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="InsdcScanner._feed_first_line-expanded"><a name="L359"></a><tt class="py-lineno"> 359</tt>  <tt class="py-line">        <tt class="py-docstring">"""Handle the LOCUS/ID line, passing data to the comsumer</tt> </tt>
<a name="L360"></a><tt class="py-lineno"> 360</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L361"></a><tt class="py-lineno"> 361</tt>  <tt class="py-line"><tt class="py-docstring">        This should be implemented by the EMBL / GenBank specific subclass</tt> </tt>
<a name="L362"></a><tt class="py-lineno"> 362</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L363"></a><tt class="py-lineno"> 363</tt>  <tt class="py-line"><tt class="py-docstring">        Used by the parse_records() and parse() methods.</tt> </tt>
<a name="L364"></a><tt class="py-lineno"> 364</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L365"></a><tt class="py-lineno"> 365</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L366"></a><tt class="py-lineno"> 366</tt>  <tt class="py-line"> </tt>
<a name="InsdcScanner._feed_header_lines"></a><div id="InsdcScanner._feed_header_lines-def"><a name="L367"></a><tt class="py-lineno"> 367</tt> <a class="py-toggle" href="#" id="InsdcScanner._feed_header_lines-toggle" onclick="return toggle('InsdcScanner._feed_header_lines');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.InsdcScanner-class.html#_feed_header_lines">_feed_header_lines</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">,</tt> <tt class="py-param">lines</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="InsdcScanner._feed_header_lines-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="InsdcScanner._feed_header_lines-expanded"><a name="L368"></a><tt class="py-lineno"> 368</tt>  <tt class="py-line">        <tt class="py-docstring">"""Handle the header lines (list of strings), passing data to the comsumer</tt> </tt>
<a name="L369"></a><tt class="py-lineno"> 369</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L370"></a><tt class="py-lineno"> 370</tt>  <tt class="py-line"><tt class="py-docstring">        This should be implemented by the EMBL / GenBank specific subclass</tt> </tt>
<a name="L371"></a><tt class="py-lineno"> 371</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L372"></a><tt class="py-lineno"> 372</tt>  <tt class="py-line"><tt class="py-docstring">        Used by the parse_records() and parse() methods.</tt> </tt>
<a name="L373"></a><tt class="py-lineno"> 373</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L374"></a><tt class="py-lineno"> 374</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L375"></a><tt class="py-lineno"> 375</tt>  <tt class="py-line"> </tt>
<a name="InsdcScanner._feed_feature_table"></a><div id="InsdcScanner._feed_feature_table-def"><a name="L376"></a><tt class="py-lineno"> 376</tt> <a class="py-toggle" href="#" id="InsdcScanner._feed_feature_table-toggle" onclick="return toggle('InsdcScanner._feed_feature_table');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.InsdcScanner-class.html#_feed_feature_table">_feed_feature_table</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">,</tt> <tt class="py-param">feature_tuples</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="InsdcScanner._feed_feature_table-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="InsdcScanner._feed_feature_table-expanded"><a name="L377"></a><tt class="py-lineno"> 377</tt>  <tt class="py-line">        <tt class="py-docstring">"""Handle the feature table (list of tuples), passing data to the comsumer</tt> </tt>
<a name="L378"></a><tt class="py-lineno"> 378</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L379"></a><tt class="py-lineno"> 379</tt>  <tt class="py-line"><tt class="py-docstring">        Used by the parse_records() and parse() methods.</tt> </tt>
<a name="L380"></a><tt class="py-lineno"> 380</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L381"></a><tt class="py-lineno"> 381</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-134" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.start_feature_table()=Bio.GenBank._FeatureConsumer-class.html#start_feature_table,Method Bio.GenBank._RecordConsumer.start_feature_table()=Bio.GenBank._RecordConsumer-class.html#start_feature_table"><a title="Bio.GenBank._FeatureConsumer.start_feature_table
Bio.GenBank._RecordConsumer.start_feature_table" class="py-name" href="#" onclick="return doclink('link-134', 'start_feature_table', 'link-134');">start_feature_table</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L382"></a><tt class="py-lineno"> 382</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-135" class="py-name"><a title="Bio.GenBank._FeatureConsumer.feature_key
Bio.GenBank._RecordConsumer.feature_key" class="py-name" href="#" onclick="return doclink('link-135', 'feature_key', 'link-61');">feature_key</a></tt><tt class="py-op">,</tt> <tt class="py-name">location_string</tt><tt class="py-op">,</tt> <tt class="py-name">qualifiers</tt> <tt class="py-keyword">in</tt> <tt class="py-name">feature_tuples</tt><tt class="py-op">:</tt> </tt>
<a name="L383"></a><tt class="py-lineno"> 383</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-136" class="py-name"><a title="Bio.GenBank._FeatureConsumer.feature_key
Bio.GenBank._RecordConsumer.feature_key" class="py-name" href="#" onclick="return doclink('link-136', 'feature_key', 'link-61');">feature_key</a></tt><tt class="py-op">(</tt><tt id="link-137" class="py-name"><a title="Bio.GenBank._FeatureConsumer.feature_key
Bio.GenBank._RecordConsumer.feature_key" class="py-name" href="#" onclick="return doclink('link-137', 'feature_key', 'link-61');">feature_key</a></tt><tt class="py-op">)</tt> </tt>
<a name="L384"></a><tt class="py-lineno"> 384</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-138" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.location()=Bio.GenBank._FeatureConsumer-class.html#location,Method Bio.GenBank._RecordConsumer.location()=Bio.GenBank._RecordConsumer-class.html#location,Method Bio.Phylo.PhyloXMLIO.Writer.location()=Bio.Phylo.PhyloXMLIO.Writer-class.html#location"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-138', 'location', 'link-138');">location</a></tt><tt class="py-op">(</tt><tt class="py-name">location_string</tt><tt class="py-op">)</tt> </tt>
<a name="L385"></a><tt class="py-lineno"> 385</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">q_key</tt><tt class="py-op">,</tt> <tt class="py-name">q_value</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualifiers</tt><tt class="py-op">:</tt> </tt>
<a name="L386"></a><tt class="py-lineno"> 386</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">q_value</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L387"></a><tt class="py-lineno"> 387</tt>  <tt class="py-line">                    <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-139" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.feature_qualifier()=Bio.GenBank._FeatureConsumer-class.html#feature_qualifier,Method Bio.GenBank._RecordConsumer.feature_qualifier()=Bio.GenBank._RecordConsumer-class.html#feature_qualifier"><a title="Bio.GenBank._FeatureConsumer.feature_qualifier
Bio.GenBank._RecordConsumer.feature_qualifier" class="py-name" href="#" onclick="return doclink('link-139', 'feature_qualifier', 'link-139');">feature_qualifier</a></tt><tt class="py-op">(</tt><tt class="py-name">q_key</tt><tt class="py-op">,</tt> <tt class="py-name">q_value</tt><tt class="py-op">)</tt> </tt>
<a name="L388"></a><tt class="py-lineno"> 388</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L389"></a><tt class="py-lineno"> 389</tt>  <tt class="py-line">                    <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-140" class="py-name"><a title="Bio.GenBank._FeatureConsumer.feature_qualifier
Bio.GenBank._RecordConsumer.feature_qualifier" class="py-name" href="#" onclick="return doclink('link-140', 'feature_qualifier', 'link-139');">feature_qualifier</a></tt><tt class="py-op">(</tt><tt class="py-name">q_key</tt><tt class="py-op">,</tt> <tt class="py-name">q_value</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"\n"</tt><tt class="py-op">,</tt> <tt class="py-string">" "</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L390"></a><tt class="py-lineno"> 390</tt>  <tt class="py-line"> </tt>
<a name="InsdcScanner._feed_misc_lines"></a><div id="InsdcScanner._feed_misc_lines-def"><a name="L391"></a><tt class="py-lineno"> 391</tt> <a class="py-toggle" href="#" id="InsdcScanner._feed_misc_lines-toggle" onclick="return toggle('InsdcScanner._feed_misc_lines');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.InsdcScanner-class.html#_feed_misc_lines">_feed_misc_lines</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">,</tt> <tt class="py-param">lines</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="InsdcScanner._feed_misc_lines-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="InsdcScanner._feed_misc_lines-expanded"><a name="L392"></a><tt class="py-lineno"> 392</tt>  <tt class="py-line">        <tt class="py-docstring">"""Handle any lines between features and sequence (list of strings), passing data to the consumer</tt> </tt>
<a name="L393"></a><tt class="py-lineno"> 393</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L394"></a><tt class="py-lineno"> 394</tt>  <tt class="py-line"><tt class="py-docstring">        This should be implemented by the EMBL / GenBank specific subclass</tt> </tt>
<a name="L395"></a><tt class="py-lineno"> 395</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L396"></a><tt class="py-lineno"> 396</tt>  <tt class="py-line"><tt class="py-docstring">        Used by the parse_records() and parse() methods.</tt> </tt>
<a name="L397"></a><tt class="py-lineno"> 397</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L398"></a><tt class="py-lineno"> 398</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L399"></a><tt class="py-lineno"> 399</tt>  <tt class="py-line"> </tt>
<a name="InsdcScanner.feed"></a><div id="InsdcScanner.feed-def"><a name="L400"></a><tt class="py-lineno"> 400</tt> <a class="py-toggle" href="#" id="InsdcScanner.feed-toggle" onclick="return toggle('InsdcScanner.feed');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.InsdcScanner-class.html#feed">feed</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">,</tt> <tt class="py-param">do_features</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="InsdcScanner.feed-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="InsdcScanner.feed-expanded"><a name="L401"></a><tt class="py-lineno"> 401</tt>  <tt class="py-line">        <tt class="py-docstring">"""Feed a set of data into the consumer.</tt> </tt>
<a name="L402"></a><tt class="py-lineno"> 402</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L403"></a><tt class="py-lineno"> 403</tt>  <tt class="py-line"><tt class="py-docstring">        This method is intended for use with the "old" code in Bio.GenBank</tt> </tt>
<a name="L404"></a><tt class="py-lineno"> 404</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L405"></a><tt class="py-lineno"> 405</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L406"></a><tt class="py-lineno"> 406</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L407"></a><tt class="py-lineno"> 407</tt>  <tt class="py-line"><tt class="py-docstring">            - handle - A handle with the information to parse.</tt> </tt>
<a name="L408"></a><tt class="py-lineno"> 408</tt>  <tt class="py-line"><tt class="py-docstring">            - consumer - The consumer that should be informed of events.</tt> </tt>
<a name="L409"></a><tt class="py-lineno"> 409</tt>  <tt class="py-line"><tt class="py-docstring">            - do_features - Boolean, should the features be parsed?</tt> </tt>
<a name="L410"></a><tt class="py-lineno"> 410</tt>  <tt class="py-line"><tt class="py-docstring">                          Skipping the features can be much faster.</tt> </tt>
<a name="L411"></a><tt class="py-lineno"> 411</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L412"></a><tt class="py-lineno"> 412</tt>  <tt class="py-line"><tt class="py-docstring">        Return values:</tt> </tt>
<a name="L413"></a><tt class="py-lineno"> 413</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L414"></a><tt class="py-lineno"> 414</tt>  <tt class="py-line"><tt class="py-docstring">            - true  - Passed a record</tt> </tt>
<a name="L415"></a><tt class="py-lineno"> 415</tt>  <tt class="py-line"><tt class="py-docstring">            - false - Did not find a record</tt> </tt>
<a name="L416"></a><tt class="py-lineno"> 416</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L417"></a><tt class="py-lineno"> 417</tt>  <tt class="py-line">        <tt class="py-comment"># Should work with both EMBL and GenBank files provided the</tt> </tt>
<a name="L418"></a><tt class="py-lineno"> 418</tt>  <tt class="py-line">        <tt class="py-comment"># equivalent Bio.GenBank._FeatureConsumer methods are called...</tt> </tt>
<a name="L419"></a><tt class="py-lineno"> 419</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-141" class="py-name" targets="Method Bio.GenBank.Scanner.InsdcScanner.set_handle()=Bio.GenBank.Scanner.InsdcScanner-class.html#set_handle"><a title="Bio.GenBank.Scanner.InsdcScanner.set_handle" class="py-name" href="#" onclick="return doclink('link-141', 'set_handle', 'link-141');">set_handle</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L420"></a><tt class="py-lineno"> 420</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-142" class="py-name" targets="Method Bio.GenBank.Scanner.InsdcScanner.find_start()=Bio.GenBank.Scanner.InsdcScanner-class.html#find_start"><a title="Bio.GenBank.Scanner.InsdcScanner.find_start" class="py-name" href="#" onclick="return doclink('link-142', 'find_start', 'link-142');">find_start</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L421"></a><tt class="py-lineno"> 421</tt>  <tt class="py-line">            <tt class="py-comment"># Could not find (another) record</tt> </tt>
<a name="L422"></a><tt class="py-lineno"> 422</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-143" class="py-name" targets="Variable BioSQL.BioSeq.DBSeq.data=BioSQL.BioSeq.DBSeq-class.html#data"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-143', 'data', 'link-143');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L423"></a><tt class="py-lineno"> 423</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
<a name="L424"></a><tt class="py-lineno"> 424</tt>  <tt class="py-line"> </tt>
<a name="L425"></a><tt class="py-lineno"> 425</tt>  <tt class="py-line">        <tt class="py-comment"># We use the above class methods to parse the file into a simplified format.</tt> </tt>
<a name="L426"></a><tt class="py-lineno"> 426</tt>  <tt class="py-line">        <tt class="py-comment"># The first line, header lines and any misc lines after the features will be</tt> </tt>
<a name="L427"></a><tt class="py-lineno"> 427</tt>  <tt class="py-line">        <tt class="py-comment"># dealt with by GenBank / EMBL specific derived classes.</tt> </tt>
<a name="L428"></a><tt class="py-lineno"> 428</tt>  <tt class="py-line"> </tt>
<a name="L429"></a><tt class="py-lineno"> 429</tt>  <tt class="py-line">        <tt class="py-comment"># First line and header:</tt> </tt>
<a name="L430"></a><tt class="py-lineno"> 430</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-144" class="py-name" targets="Method Bio.GenBank.Scanner.EmblScanner._feed_first_line()=Bio.GenBank.Scanner.EmblScanner-class.html#_feed_first_line,Method Bio.GenBank.Scanner.GenBankScanner._feed_first_line()=Bio.GenBank.Scanner.GenBankScanner-class.html#_feed_first_line,Method Bio.GenBank.Scanner.InsdcScanner._feed_first_line()=Bio.GenBank.Scanner.InsdcScanner-class.html#_feed_first_line"><a title="Bio.GenBank.Scanner.EmblScanner._feed_first_line
Bio.GenBank.Scanner.GenBankScanner._feed_first_line
Bio.GenBank.Scanner.InsdcScanner._feed_first_line" class="py-name" href="#" onclick="return doclink('link-144', '_feed_first_line', 'link-144');">_feed_first_line</a></tt><tt class="py-op">(</tt><tt class="py-name">consumer</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L431"></a><tt class="py-lineno"> 431</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-145" class="py-name" targets="Method Bio.GenBank.Scanner.EmblScanner._feed_header_lines()=Bio.GenBank.Scanner.EmblScanner-class.html#_feed_header_lines,Method Bio.GenBank.Scanner.GenBankScanner._feed_header_lines()=Bio.GenBank.Scanner.GenBankScanner-class.html#_feed_header_lines,Method Bio.GenBank.Scanner.InsdcScanner._feed_header_lines()=Bio.GenBank.Scanner.InsdcScanner-class.html#_feed_header_lines"><a title="Bio.GenBank.Scanner.EmblScanner._feed_header_lines
Bio.GenBank.Scanner.GenBankScanner._feed_header_lines
Bio.GenBank.Scanner.InsdcScanner._feed_header_lines" class="py-name" href="#" onclick="return doclink('link-145', '_feed_header_lines', 'link-145');">_feed_header_lines</a></tt><tt class="py-op">(</tt><tt class="py-name">consumer</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-146" class="py-name" targets="Method Bio.GenBank.Scanner.InsdcScanner.parse_header()=Bio.GenBank.Scanner.InsdcScanner-class.html#parse_header"><a title="Bio.GenBank.Scanner.InsdcScanner.parse_header" class="py-name" href="#" onclick="return doclink('link-146', 'parse_header', 'link-146');">parse_header</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L432"></a><tt class="py-lineno"> 432</tt>  <tt class="py-line"> </tt>
<a name="L433"></a><tt class="py-lineno"> 433</tt>  <tt class="py-line">        <tt class="py-comment"># Features (common to both EMBL and GenBank):</tt> </tt>
<a name="L434"></a><tt class="py-lineno"> 434</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">do_features</tt><tt class="py-op">:</tt> </tt>
<a name="L435"></a><tt class="py-lineno"> 435</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-147" class="py-name" targets="Method Bio.GenBank.Scanner.InsdcScanner._feed_feature_table()=Bio.GenBank.Scanner.InsdcScanner-class.html#_feed_feature_table"><a title="Bio.GenBank.Scanner.InsdcScanner._feed_feature_table" class="py-name" href="#" onclick="return doclink('link-147', '_feed_feature_table', 'link-147');">_feed_feature_table</a></tt><tt class="py-op">(</tt><tt class="py-name">consumer</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-148" class="py-name" targets="Method Bio.GenBank.Scanner.InsdcScanner.parse_features()=Bio.GenBank.Scanner.InsdcScanner-class.html#parse_features,Method Bio.GenBank.Scanner._ImgtScanner.parse_features()=Bio.GenBank.Scanner._ImgtScanner-class.html#parse_features"><a title="Bio.GenBank.Scanner.InsdcScanner.parse_features
Bio.GenBank.Scanner._ImgtScanner.parse_features" class="py-name" href="#" onclick="return doclink('link-148', 'parse_features', 'link-148');">parse_features</a></tt><tt class="py-op">(</tt><tt class="py-name">skip</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L436"></a><tt class="py-lineno"> 436</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L437"></a><tt class="py-lineno"> 437</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-149" class="py-name"><a title="Bio.GenBank.Scanner.InsdcScanner.parse_features
Bio.GenBank.Scanner._ImgtScanner.parse_features" class="py-name" href="#" onclick="return doclink('link-149', 'parse_features', 'link-148');">parse_features</a></tt><tt class="py-op">(</tt><tt class="py-name">skip</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt>  <tt class="py-comment"># ignore the data</tt> </tt>
<a name="L438"></a><tt class="py-lineno"> 438</tt>  <tt class="py-line"> </tt>
<a name="L439"></a><tt class="py-lineno"> 439</tt>  <tt class="py-line">        <tt class="py-comment"># Footer and sequence</tt> </tt>
<a name="L440"></a><tt class="py-lineno"> 440</tt>  <tt class="py-line">        <tt class="py-name">misc_lines</tt><tt class="py-op">,</tt> <tt class="py-name">sequence_string</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-150" class="py-name" targets="Method Bio.GenBank.Scanner.EmblScanner.parse_footer()=Bio.GenBank.Scanner.EmblScanner-class.html#parse_footer,Method Bio.GenBank.Scanner.GenBankScanner.parse_footer()=Bio.GenBank.Scanner.GenBankScanner-class.html#parse_footer,Method Bio.GenBank.Scanner.InsdcScanner.parse_footer()=Bio.GenBank.Scanner.InsdcScanner-class.html#parse_footer"><a title="Bio.GenBank.Scanner.EmblScanner.parse_footer
Bio.GenBank.Scanner.GenBankScanner.parse_footer
Bio.GenBank.Scanner.InsdcScanner.parse_footer" class="py-name" href="#" onclick="return doclink('link-150', 'parse_footer', 'link-150');">parse_footer</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L441"></a><tt class="py-lineno"> 441</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-151" class="py-name" targets="Method Bio.GenBank.Scanner.EmblScanner._feed_misc_lines()=Bio.GenBank.Scanner.EmblScanner-class.html#_feed_misc_lines,Method Bio.GenBank.Scanner.GenBankScanner._feed_misc_lines()=Bio.GenBank.Scanner.GenBankScanner-class.html#_feed_misc_lines,Method Bio.GenBank.Scanner.InsdcScanner._feed_misc_lines()=Bio.GenBank.Scanner.InsdcScanner-class.html#_feed_misc_lines"><a title="Bio.GenBank.Scanner.EmblScanner._feed_misc_lines
Bio.GenBank.Scanner.GenBankScanner._feed_misc_lines
Bio.GenBank.Scanner.InsdcScanner._feed_misc_lines" class="py-name" href="#" onclick="return doclink('link-151', '_feed_misc_lines', 'link-151');">_feed_misc_lines</a></tt><tt class="py-op">(</tt><tt class="py-name">consumer</tt><tt class="py-op">,</tt> <tt class="py-name">misc_lines</tt><tt class="py-op">)</tt> </tt>
<a name="L442"></a><tt class="py-lineno"> 442</tt>  <tt class="py-line"> </tt>
<a name="L443"></a><tt class="py-lineno"> 443</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-152" class="py-name" targets="Method Bio.FSSP.FSSPAlignDict.sequence()=Bio.FSSP.FSSPAlignDict-class.html#sequence,Method Bio.GenBank._FeatureConsumer.sequence()=Bio.GenBank._FeatureConsumer-class.html#sequence,Method Bio.GenBank._RecordConsumer.sequence()=Bio.GenBank._RecordConsumer-class.html#sequence,Method Bio.Phylo.PhyloXMLIO.Writer.sequence()=Bio.Phylo.PhyloXMLIO.Writer-class.html#sequence"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-152', 'sequence', 'link-152');">sequence</a></tt><tt class="py-op">(</tt><tt class="py-name">sequence_string</tt><tt class="py-op">)</tt> </tt>
<a name="L444"></a><tt class="py-lineno"> 444</tt>  <tt class="py-line">        <tt class="py-comment"># Calls to consumer.base_number() do nothing anyway</tt> </tt>
<a name="L445"></a><tt class="py-lineno"> 445</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-153" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.record_end()=Bio.GenBank._FeatureConsumer-class.html#record_end,Method Bio.GenBank._RecordConsumer.record_end()=Bio.GenBank._RecordConsumer-class.html#record_end"><a title="Bio.GenBank._FeatureConsumer.record_end
Bio.GenBank._RecordConsumer.record_end" class="py-name" href="#" onclick="return doclink('link-153', 'record_end', 'link-153');">record_end</a></tt><tt class="py-op">(</tt><tt class="py-string">"//"</tt><tt class="py-op">)</tt> </tt>
<a name="L446"></a><tt class="py-lineno"> 446</tt>  <tt class="py-line"> </tt>
<a name="L447"></a><tt class="py-lineno"> 447</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">==</tt> <tt class="py-string">"//"</tt> </tt>
<a name="L448"></a><tt class="py-lineno"> 448</tt>  <tt class="py-line"> </tt>
<a name="L449"></a><tt class="py-lineno"> 449</tt>  <tt class="py-line">        <tt class="py-comment"># And we are done</tt> </tt>
<a name="L450"></a><tt class="py-lineno"> 450</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">True</tt> </tt>
</div><a name="L451"></a><tt class="py-lineno"> 451</tt>  <tt class="py-line"> </tt>
<a name="InsdcScanner.parse"></a><div id="InsdcScanner.parse-def"><a name="L452"></a><tt class="py-lineno"> 452</tt> <a class="py-toggle" href="#" id="InsdcScanner.parse-toggle" onclick="return toggle('InsdcScanner.parse');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.InsdcScanner-class.html#parse">parse</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">do_features</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="InsdcScanner.parse-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="InsdcScanner.parse-expanded"><a name="L453"></a><tt class="py-lineno"> 453</tt>  <tt class="py-line">        <tt class="py-docstring">"""Returns a SeqRecord (with SeqFeatures if do_features=True)</tt> </tt>
<a name="L454"></a><tt class="py-lineno"> 454</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L455"></a><tt class="py-lineno"> 455</tt>  <tt class="py-line"><tt class="py-docstring">        See also the method parse_records() for use on multi-record files.</tt> </tt>
<a name="L456"></a><tt class="py-lineno"> 456</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L457"></a><tt class="py-lineno"> 457</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt id="link-154" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-154', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-155" class="py-name" targets="Package Bio.GenBank=Bio.GenBank-module.html"><a title="Bio.GenBank" class="py-name" href="#" onclick="return doclink('link-155', 'GenBank', 'link-155');">GenBank</a></tt> <tt class="py-keyword">import</tt> <tt id="link-156" class="py-name" targets="Class Bio.GenBank._FeatureConsumer=Bio.GenBank._FeatureConsumer-class.html"><a title="Bio.GenBank._FeatureConsumer" class="py-name" href="#" onclick="return doclink('link-156', '_FeatureConsumer', 'link-156');">_FeatureConsumer</a></tt> </tt>
<a name="L458"></a><tt class="py-lineno"> 458</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt id="link-157" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-157', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-158" class="py-name"><a title="Bio.GenBank" class="py-name" href="#" onclick="return doclink('link-158', 'GenBank', 'link-155');">GenBank</a></tt><tt class="py-op">.</tt><tt id="link-159" class="py-name" targets="Module Bio.GenBank.utils=Bio.GenBank.utils-module.html"><a title="Bio.GenBank.utils" class="py-name" href="#" onclick="return doclink('link-159', 'utils', 'link-159');">utils</a></tt> <tt class="py-keyword">import</tt> <tt id="link-160" class="py-name" targets="Class Bio.GenBank.utils.FeatureValueCleaner=Bio.GenBank.utils.FeatureValueCleaner-class.html"><a title="Bio.GenBank.utils.FeatureValueCleaner" class="py-name" href="#" onclick="return doclink('link-160', 'FeatureValueCleaner', 'link-160');">FeatureValueCleaner</a></tt> </tt>
<a name="L459"></a><tt class="py-lineno"> 459</tt>  <tt class="py-line"> </tt>
<a name="L460"></a><tt class="py-lineno"> 460</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt> <tt class="py-op">=</tt> <tt id="link-161" class="py-name"><a title="Bio.GenBank._FeatureConsumer" class="py-name" href="#" onclick="return doclink('link-161', '_FeatureConsumer', 'link-156');">_FeatureConsumer</a></tt><tt class="py-op">(</tt><tt class="py-name">use_fuzziness</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> </tt>
<a name="L461"></a><tt class="py-lineno"> 461</tt>  <tt class="py-line">                                    <tt class="py-name">feature_cleaner</tt><tt class="py-op">=</tt><tt id="link-162" class="py-name"><a title="Bio.GenBank.utils.FeatureValueCleaner" class="py-name" href="#" onclick="return doclink('link-162', 'FeatureValueCleaner', 'link-160');">FeatureValueCleaner</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L462"></a><tt class="py-lineno"> 462</tt>  <tt class="py-line"> </tt>
<a name="L463"></a><tt class="py-lineno"> 463</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-163" class="py-name" targets="Method Bio.Blast.NCBIStandalone._Scanner.feed()=Bio.Blast.NCBIStandalone._Scanner-class.html#feed,Method Bio.ExPASy.ScanProsite.Parser.feed()=Bio.ExPASy.ScanProsite.Parser-class.html#feed,Method Bio.GenBank.Scanner.InsdcScanner.feed()=Bio.GenBank.Scanner.InsdcScanner-class.html#feed"><a title="Bio.Blast.NCBIStandalone._Scanner.feed
Bio.ExPASy.ScanProsite.Parser.feed
Bio.GenBank.Scanner.InsdcScanner.feed" class="py-name" href="#" onclick="return doclink('link-163', 'feed', 'link-163');">feed</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">,</tt> <tt class="py-name">do_features</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L464"></a><tt class="py-lineno"> 464</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-164" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-164', 'data', 'link-143');">data</a></tt> </tt>
<a name="L465"></a><tt class="py-lineno"> 465</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L466"></a><tt class="py-lineno"> 466</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L467"></a><tt class="py-lineno"> 467</tt>  <tt class="py-line"> </tt>
<a name="InsdcScanner.parse_records"></a><div id="InsdcScanner.parse_records-def"><a name="L468"></a><tt class="py-lineno"> 468</tt> <a class="py-toggle" href="#" id="InsdcScanner.parse_records-toggle" onclick="return toggle('InsdcScanner.parse_records');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.InsdcScanner-class.html#parse_records">parse_records</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">do_features</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="InsdcScanner.parse_records-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="InsdcScanner.parse_records-expanded"><a name="L469"></a><tt class="py-lineno"> 469</tt>  <tt class="py-line">        <tt class="py-docstring">"""Returns a SeqRecord object iterator</tt> </tt>
<a name="L470"></a><tt class="py-lineno"> 470</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L471"></a><tt class="py-lineno"> 471</tt>  <tt class="py-line"><tt class="py-docstring">        Each record (from the ID/LOCUS line to the // line) becomes a SeqRecord</tt> </tt>
<a name="L472"></a><tt class="py-lineno"> 472</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L473"></a><tt class="py-lineno"> 473</tt>  <tt class="py-line"><tt class="py-docstring">        The SeqRecord objects include SeqFeatures if do_features=True</tt> </tt>
<a name="L474"></a><tt class="py-lineno"> 474</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L475"></a><tt class="py-lineno"> 475</tt>  <tt class="py-line"><tt class="py-docstring">        This method is intended for use in Bio.SeqIO</tt> </tt>
<a name="L476"></a><tt class="py-lineno"> 476</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L477"></a><tt class="py-lineno"> 477</tt>  <tt class="py-line">        <tt class="py-comment"># This is a generator function</tt> </tt>
<a name="L478"></a><tt class="py-lineno"> 478</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L479"></a><tt class="py-lineno"> 479</tt>  <tt class="py-line">            <tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-165" class="py-name" targets="Function Bio.AlignIO.parse()=Bio.AlignIO-module.html#parse,Method Bio.Blast.NCBIStandalone.BlastErrorParser.parse()=Bio.Blast.NCBIStandalone.BlastErrorParser-class.html#parse,Method Bio.Blast.NCBIStandalone.BlastParser.parse()=Bio.Blast.NCBIStandalone.BlastParser-class.html#parse,Method Bio.Blast.NCBIStandalone.PSIBlastParser.parse()=Bio.Blast.NCBIStandalone.PSIBlastParser-class.html#parse,Function Bio.Blast.NCBIXML.parse()=Bio.Blast.NCBIXML-module.html#parse,Function Bio.Compass.parse()=Bio.Compass-module.html#parse,Function Bio.Emboss.Primer3.parse()=Bio.Emboss.Primer3-module.html#parse,Method Bio.Entrez.Parser.DataHandler.parse()=Bio.Entrez.Parser.DataHandler-class.html#parse,Function Bio.Entrez.parse()=Bio.Entrez-module.html#parse,Function Bio.ExPASy.Enzyme.parse()=Bio.ExPASy.Enzyme-module.html#parse,Function Bio.ExPASy.Prodoc.parse()=Bio.ExPASy.Prodoc-module.html#parse,Function Bio.ExPASy.Prosite.parse()=Bio.ExPASy.Prosite-module.html#parse,Method Bio.GenBank.FeatureParser.parse()=Bio.GenBank.FeatureParser-class.html#parse,Method Bio.GenBank.RecordParser.parse()=Bio.GenBank.RecordParser-class.html#parse,Method Bio.GenBank.Scanner.InsdcScanner.parse()=Bio.GenBank.Scanner.InsdcScanner-class.html#parse,Function Bio.Geo.parse()=Bio.Geo-module.html#parse,Function Bio.KEGG.Compound.parse()=Bio.KEGG.Compound-module.html#parse,Function Bio.KEGG.Enzyme.parse()=Bio.KEGG.Enzyme-module.html#parse,Method Bio.KEGG.KGML.KGML_parser.KGMLParser.parse()=Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html#parse,Function Bio.KEGG.KGML.KGML_parser.parse()=Bio.KEGG.KGML.KGML_parser-module.html#parse,Function Bio.KEGG.Map.parse()=Bio.KEGG.Map-module.html#parse,Function Bio.Medline.parse()=Bio.Medline-module.html#parse,Function Bio.Motif.parse()=Bio.Motif-module.html#parse,Method Bio.ParserSupport.AbstractParser.parse()=Bio.ParserSupport.AbstractParser-class.html#parse,Method Bio.Phylo.CDAOIO.Parser.parse()=Bio.Phylo.CDAOIO.Parser-class.html#parse,Function Bio.Phylo.CDAOIO.parse()=Bio.Phylo.CDAOIO-module.html#parse,Method Bio.Phylo.NeXMLIO.Parser.parse()=Bio.Phylo.NeXMLIO.Parser-class.html#parse,Function Bio.Phylo.NeXMLIO.parse()=Bio.Phylo.NeXMLIO-module.html#parse,Method Bio.Phylo.NewickIO.Parser.parse()=Bio.Phylo.NewickIO.Parser-class.html#parse,Function Bio.Phylo.NewickIO.parse()=Bio.Phylo.NewickIO-module.html#parse,Function Bio.Phylo.NexusIO.parse()=Bio.Phylo.NexusIO-module.html#parse,Method Bio.Phylo.PhyloXMLIO.Parser.parse()=Bio.Phylo.PhyloXMLIO.Parser-class.html#parse,Function Bio.Phylo.PhyloXMLIO.parse()=Bio.Phylo.PhyloXMLIO-module.html#parse,Function Bio.Phylo._io.parse()=Bio.Phylo._io-module.html#parse,Function Bio.SCOP.Cla.parse()=Bio.SCOP.Cla-module.html#parse,Function Bio.SCOP.Des.parse()=Bio.SCOP.Des-module.html#parse,Function Bio.SCOP.Dom.parse()=Bio.SCOP.Dom-module.html#parse,Function Bio.SCOP.Hie.parse()=Bio.SCOP.Hie-module.html#parse,Function Bio.SCOP.Raf.parse()=Bio.SCOP.Raf-module.html#parse,Function Bio.SearchIO.parse()=Bio.SearchIO-module.html#parse,Method Bio.SeqIO.UniprotIO.Parser.parse()=Bio.SeqIO.UniprotIO.Parser-class.html#parse,Function Bio.SeqIO.parse()=Bio.SeqIO-module.html#parse,Function Bio.Sequencing.Ace.parse()=Bio.Sequencing.Ace-module.html#parse,Function Bio.Sequencing.Phd.parse()=Bio.Sequencing.Phd-module.html#parse,Function Bio.SwissProt.KeyWList.parse()=Bio.SwissProt.KeyWList-module.html#parse,Function Bio.SwissProt.parse()=Bio.SwissProt-module.html#parse,Function Bio.UniGene.parse()=Bio.UniGene-module.html#parse,Function Bio.Wise.psw.parse()=Bio.Wise.psw-module.html#parse,Function Bio.motifs.parse()=Bio.motifs-module.html#parse"><a title="Bio.AlignIO.parse
Bio.Blast.NCBIStandalone.BlastErrorParser.parse
Bio.Blast.NCBIStandalone.BlastParser.parse
Bio.Blast.NCBIStandalone.PSIBlastParser.parse
Bio.Blast.NCBIXML.parse
Bio.Compass.parse
Bio.Emboss.Primer3.parse
Bio.Entrez.Parser.DataHandler.parse
Bio.Entrez.parse
Bio.ExPASy.Enzyme.parse
Bio.ExPASy.Prodoc.parse
Bio.ExPASy.Prosite.parse
Bio.GenBank.FeatureParser.parse
Bio.GenBank.RecordParser.parse
Bio.GenBank.Scanner.InsdcScanner.parse
Bio.Geo.parse
Bio.KEGG.Compound.parse
Bio.KEGG.Enzyme.parse
Bio.KEGG.KGML.KGML_parser.KGMLParser.parse
Bio.KEGG.KGML.KGML_parser.parse
Bio.KEGG.Map.parse
Bio.Medline.parse
Bio.Motif.parse
Bio.ParserSupport.AbstractParser.parse
Bio.Phylo.CDAOIO.Parser.parse
Bio.Phylo.CDAOIO.parse
Bio.Phylo.NeXMLIO.Parser.parse
Bio.Phylo.NeXMLIO.parse
Bio.Phylo.NewickIO.Parser.parse
Bio.Phylo.NewickIO.parse
Bio.Phylo.NexusIO.parse
Bio.Phylo.PhyloXMLIO.Parser.parse
Bio.Phylo.PhyloXMLIO.parse
Bio.Phylo._io.parse
Bio.SCOP.Cla.parse
Bio.SCOP.Des.parse
Bio.SCOP.Dom.parse
Bio.SCOP.Hie.parse
Bio.SCOP.Raf.parse
Bio.SearchIO.parse
Bio.SeqIO.UniprotIO.Parser.parse
Bio.SeqIO.parse
Bio.Sequencing.Ace.parse
Bio.Sequencing.Phd.parse
Bio.SwissProt.KeyWList.parse
Bio.SwissProt.parse
Bio.UniGene.parse
Bio.Wise.psw.parse
Bio.motifs.parse" class="py-name" href="#" onclick="return doclink('link-165', 'parse', 'link-165');">parse</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-name">do_features</tt><tt class="py-op">)</tt> </tt>
<a name="L480"></a><tt class="py-lineno"> 480</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">record</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L481"></a><tt class="py-lineno"> 481</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L482"></a><tt class="py-lineno"> 482</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-166" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-166', 'id', 'link-166');">id</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L483"></a><tt class="py-lineno"> 483</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Failed to parse the record's ID. Invalid ID line?"</tt><tt class="py-op">)</tt> </tt>
<a name="L484"></a><tt class="py-lineno"> 484</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-167" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-167', 'name', 'link-167');">name</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"&lt;unknown name&gt;"</tt><tt class="py-op">:</tt> </tt>
<a name="L485"></a><tt class="py-lineno"> 485</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Failed to parse the record's name. Invalid ID line?"</tt><tt class="py-op">)</tt> </tt>
<a name="L486"></a><tt class="py-lineno"> 486</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-168" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DescriptionConsumer.description()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#description,Method Bio.Phylo.PhyloXMLIO.Writer.description()=Bio.Phylo.PhyloXMLIO.Writer-class.html#description,Variable Bio.SearchIO._model.hit.Hit.description=Bio.SearchIO._model.hit.Hit-class.html#description,Variable Bio.SearchIO._model.query.QueryResult.description=Bio.SearchIO._model.query.QueryResult-class.html#description"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-168', 'description', 'link-168');">description</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"&lt;unknown description&gt;"</tt><tt class="py-op">:</tt> </tt>
<a name="L487"></a><tt class="py-lineno"> 487</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Failed to parse the record's description"</tt><tt class="py-op">)</tt> </tt>
<a name="L488"></a><tt class="py-lineno"> 488</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt class="py-name">record</tt> </tt>
</div><a name="L489"></a><tt class="py-lineno"> 489</tt>  <tt class="py-line"> </tt>
<a name="InsdcScanner.parse_cds_features"></a><div id="InsdcScanner.parse_cds_features-def"><a name="L490"></a><tt class="py-lineno"> 490</tt> <a class="py-toggle" href="#" id="InsdcScanner.parse_cds_features-toggle" onclick="return toggle('InsdcScanner.parse_cds_features');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.InsdcScanner-class.html#parse_cds_features">parse_cds_features</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">,</tt> </tt>
<a name="L491"></a><tt class="py-lineno"> 491</tt>  <tt class="py-line">                           <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt id="link-169" class="py-name"><a title="Bio.Alphabet.generic_protein" class="py-name" href="#" onclick="return doclink('link-169', 'generic_protein', 'link-8');">generic_protein</a></tt><tt class="py-op">,</tt> </tt>
<a name="L492"></a><tt class="py-lineno"> 492</tt>  <tt class="py-line">                           <tt class="py-param">tags2id</tt><tt class="py-op">=</tt><tt class="py-op">(</tt><tt class="py-string">'protein_id'</tt><tt class="py-op">,</tt> <tt class="py-string">'locus_tag'</tt><tt class="py-op">,</tt> <tt class="py-string">'product'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="InsdcScanner.parse_cds_features-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="InsdcScanner.parse_cds_features-expanded"><a name="L493"></a><tt class="py-lineno"> 493</tt>  <tt class="py-line">        <tt class="py-docstring">"""Returns SeqRecord object iterator</tt> </tt>
<a name="L494"></a><tt class="py-lineno"> 494</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L495"></a><tt class="py-lineno"> 495</tt>  <tt class="py-line"><tt class="py-docstring">        Each CDS feature becomes a SeqRecord.</tt> </tt>
<a name="L496"></a><tt class="py-lineno"> 496</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L497"></a><tt class="py-lineno"> 497</tt>  <tt class="py-line"><tt class="py-docstring">            - alphabet - Used for any sequence found in a translation field.</tt> </tt>
<a name="L498"></a><tt class="py-lineno"> 498</tt>  <tt class="py-line"><tt class="py-docstring">            - tags2id  - Tupple of three strings, the feature keys to use</tt> </tt>
<a name="L499"></a><tt class="py-lineno"> 499</tt>  <tt class="py-line"><tt class="py-docstring">                   for the record id, name and description,</tt> </tt>
<a name="L500"></a><tt class="py-lineno"> 500</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L501"></a><tt class="py-lineno"> 501</tt>  <tt class="py-line"><tt class="py-docstring">        This method is intended for use in Bio.SeqIO</tt> </tt>
<a name="L502"></a><tt class="py-lineno"> 502</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L503"></a><tt class="py-lineno"> 503</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-170" class="py-name"><a title="Bio.GenBank.Scanner.InsdcScanner.set_handle" class="py-name" href="#" onclick="return doclink('link-170', 'set_handle', 'link-141');">set_handle</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L504"></a><tt class="py-lineno"> 504</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-171" class="py-name"><a title="Bio.GenBank.Scanner.InsdcScanner.find_start" class="py-name" href="#" onclick="return doclink('link-171', 'find_start', 'link-142');">find_start</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L505"></a><tt class="py-lineno"> 505</tt>  <tt class="py-line">            <tt class="py-comment"># Got an EMBL or GenBank record...</tt> </tt>
<a name="L506"></a><tt class="py-lineno"> 506</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-172" class="py-name"><a title="Bio.GenBank.Scanner.InsdcScanner.parse_header" class="py-name" href="#" onclick="return doclink('link-172', 'parse_header', 'link-146');">parse_header</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt>  <tt class="py-comment"># ignore header lines!</tt> </tt>
<a name="L507"></a><tt class="py-lineno"> 507</tt>  <tt class="py-line">            <tt class="py-name">feature_tuples</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-173" class="py-name"><a title="Bio.GenBank.Scanner.InsdcScanner.parse_features
Bio.GenBank.Scanner._ImgtScanner.parse_features" class="py-name" href="#" onclick="return doclink('link-173', 'parse_features', 'link-148');">parse_features</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L508"></a><tt class="py-lineno"> 508</tt>  <tt class="py-line">            <tt class="py-comment"># self.parse_footer() # ignore footer lines!</tt> </tt>
<a name="L509"></a><tt class="py-lineno"> 509</tt>  <tt class="py-line">            <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L510"></a><tt class="py-lineno"> 510</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-174" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-174', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L511"></a><tt class="py-lineno"> 511</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L512"></a><tt class="py-lineno"> 512</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L513"></a><tt class="py-lineno"> 513</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"//"</tt><tt class="py-op">:</tt> </tt>
<a name="L514"></a><tt class="py-lineno"> 514</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L515"></a><tt class="py-lineno"> 515</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-175" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-175', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L516"></a><tt class="py-lineno"> 516</tt>  <tt class="py-line"> </tt>
<a name="L517"></a><tt class="py-lineno"> 517</tt>  <tt class="py-line">            <tt class="py-comment"># Now go though those features...</tt> </tt>
<a name="L518"></a><tt class="py-lineno"> 518</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">location_string</tt><tt class="py-op">,</tt> <tt class="py-name">qualifiers</tt> <tt class="py-keyword">in</tt> <tt class="py-name">feature_tuples</tt><tt class="py-op">:</tt> </tt>
<a name="L519"></a><tt class="py-lineno"> 519</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-op">==</tt> <tt class="py-string">"CDS"</tt><tt class="py-op">:</tt> </tt>
<a name="L520"></a><tt class="py-lineno"> 520</tt>  <tt class="py-line">                    <tt class="py-comment"># Create SeqRecord</tt> </tt>
<a name="L521"></a><tt class="py-lineno"> 521</tt>  <tt class="py-line">                    <tt class="py-comment"># ================</tt> </tt>
<a name="L522"></a><tt class="py-lineno"> 522</tt>  <tt class="py-line">                    <tt class="py-comment"># SeqRecord objects cannot be created with annotations, they</tt> </tt>
<a name="L523"></a><tt class="py-lineno"> 523</tt>  <tt class="py-line">                    <tt class="py-comment"># must be added afterwards.  So create an empty record and</tt> </tt>
<a name="L524"></a><tt class="py-lineno"> 524</tt>  <tt class="py-line">                    <tt class="py-comment"># then populate it:</tt> </tt>
<a name="L525"></a><tt class="py-lineno"> 525</tt>  <tt class="py-line">                    <tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt id="link-176" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-176', 'SeqRecord', 'link-4');">SeqRecord</a></tt><tt class="py-op">(</tt><tt id="link-177" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.seq=Bio.SeqRecord.SeqRecord-class.html#seq,Variable BioSQL.BioSeq.DBSeqRecord.seq=BioSQL.BioSeq.DBSeqRecord-class.html#seq"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-177', 'seq', 'link-177');">seq</a></tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
<a name="L526"></a><tt class="py-lineno"> 526</tt>  <tt class="py-line">                    <tt id="link-178" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.annotations=BioSQL.BioSeq.DBSeqRecord-class.html#annotations"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-178', 'annotations', 'link-178');">annotations</a></tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-179" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-179', 'annotations', 'link-178');">annotations</a></tt> </tt>
<a name="L527"></a><tt class="py-lineno"> 527</tt>  <tt class="py-line"> </tt>
<a name="L528"></a><tt class="py-lineno"> 528</tt>  <tt class="py-line">                    <tt class="py-comment"># Should we add a location object to the annotations?</tt> </tt>
<a name="L529"></a><tt class="py-lineno"> 529</tt>  <tt class="py-line">                    <tt class="py-comment"># I *think* that only makes sense for SeqFeatures with their</tt> </tt>
<a name="L530"></a><tt class="py-lineno"> 530</tt>  <tt class="py-line">                    <tt class="py-comment"># sub features...</tt> </tt>
<a name="L531"></a><tt class="py-lineno"> 531</tt>  <tt class="py-line">                    <tt id="link-180" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-180', 'annotations', 'link-178');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'raw_location'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">location_string</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">' '</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L532"></a><tt class="py-lineno"> 532</tt>  <tt class="py-line"> </tt>
<a name="L533"></a><tt class="py-lineno"> 533</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-op">(</tt><tt class="py-name">qualifier_name</tt><tt class="py-op">,</tt> <tt class="py-name">qualifier_data</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualifiers</tt><tt class="py-op">:</tt> </tt>
<a name="L534"></a><tt class="py-lineno"> 534</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">qualifier_data</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> \ </tt>
<a name="L535"></a><tt class="py-lineno"> 535</tt>  <tt class="py-line">                                <tt class="py-keyword">and</tt> <tt class="py-name">qualifier_data</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'"'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">qualifier_data</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'"'</tt><tt class="py-op">:</tt> </tt>
<a name="L536"></a><tt class="py-lineno"> 536</tt>  <tt class="py-line">                            <tt class="py-comment"># Remove quotes</tt> </tt>
<a name="L537"></a><tt class="py-lineno"> 537</tt>  <tt class="py-line">                            <tt class="py-name">qualifier_data</tt> <tt class="py-op">=</tt> <tt class="py-name">qualifier_data</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L538"></a><tt class="py-lineno"> 538</tt>  <tt class="py-line">                        <tt class="py-comment"># Append the data to the annotation qualifier...</tt> </tt>
<a name="L539"></a><tt class="py-lineno"> 539</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">qualifier_name</tt> <tt class="py-op">==</tt> <tt class="py-string">"translation"</tt><tt class="py-op">:</tt> </tt>
<a name="L540"></a><tt class="py-lineno"> 540</tt>  <tt class="py-line">                            <tt class="py-keyword">assert</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-181" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-181', 'seq', 'link-177');">seq</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-string">"Multiple translations!"</tt> </tt>
<a name="L541"></a><tt class="py-lineno"> 541</tt>  <tt class="py-line">                            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-182" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-182', 'seq', 'link-177');">seq</a></tt> <tt class="py-op">=</tt> <tt id="link-183" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-183', 'Seq', 'link-1');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">qualifier_data</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"\n"</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-184" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.alphabet=Bio.SearchIO._model.hsp.HSP-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSPFragment.alphabet=Bio.SearchIO._model.hsp.HSPFragment-class.html#alphabet"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-184', 'alphabet', 'link-184');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L542"></a><tt class="py-lineno"> 542</tt>  <tt class="py-line">                        <tt class="py-keyword">elif</tt> <tt class="py-name">qualifier_name</tt> <tt class="py-op">==</tt> <tt class="py-string">"db_xref"</tt><tt class="py-op">:</tt> </tt>
<a name="L543"></a><tt class="py-lineno"> 543</tt>  <tt class="py-line">                            <tt class="py-comment"># its a list, possibly empty.  Its safe to extend</tt> </tt>
<a name="L544"></a><tt class="py-lineno"> 544</tt>  <tt class="py-line">                            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-185" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.dbxrefs=BioSQL.BioSeq.DBSeqRecord-class.html#dbxrefs"><a title="BioSQL.BioSeq.DBSeqRecord.dbxrefs" class="py-name" href="#" onclick="return doclink('link-185', 'dbxrefs', 'link-185');">dbxrefs</a></tt><tt class="py-op">.</tt><tt id="link-186" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-186', 'append', 'link-36');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">qualifier_data</tt><tt class="py-op">)</tt> </tt>
<a name="L545"></a><tt class="py-lineno"> 545</tt>  <tt class="py-line">                        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L546"></a><tt class="py-lineno"> 546</tt>  <tt class="py-line">                            <tt class="py-keyword">if</tt> <tt class="py-name">qualifier_data</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L547"></a><tt class="py-lineno"> 547</tt>  <tt class="py-line">                                <tt class="py-name">qualifier_data</tt> <tt class="py-op">=</tt> <tt class="py-name">qualifier_data</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"\n"</tt><tt class="py-op">,</tt> <tt class="py-string">" "</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"  "</tt><tt class="py-op">,</tt> <tt class="py-string">" "</tt><tt class="py-op">)</tt> </tt>
<a name="L548"></a><tt class="py-lineno"> 548</tt>  <tt class="py-line">                            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L549"></a><tt class="py-lineno"> 549</tt>  <tt class="py-line">                                <tt id="link-187" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-187', 'annotations', 'link-178');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-name">qualifier_name</tt><tt class="py-op">]</tt> <tt class="py-op">+=</tt> <tt class="py-string">" "</tt> <tt class="py-op">+</tt> <tt class="py-name">qualifier_data</tt> </tt>
<a name="L550"></a><tt class="py-lineno"> 550</tt>  <tt class="py-line">                            <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L551"></a><tt class="py-lineno"> 551</tt>  <tt class="py-line">                                <tt class="py-comment"># Not an addition to existing data, its the first bit</tt> </tt>
<a name="L552"></a><tt class="py-lineno"> 552</tt>  <tt class="py-line">                                <tt id="link-188" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-188', 'annotations', 'link-178');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-name">qualifier_name</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">qualifier_data</tt> </tt>
<a name="L553"></a><tt class="py-lineno"> 553</tt>  <tt class="py-line"> </tt>
<a name="L554"></a><tt class="py-lineno"> 554</tt>  <tt class="py-line">                    <tt class="py-comment"># Fill in the ID, Name, Description</tt> </tt>
<a name="L555"></a><tt class="py-lineno"> 555</tt>  <tt class="py-line">                    <tt class="py-comment"># =================================</tt> </tt>
<a name="L556"></a><tt class="py-lineno"> 556</tt>  <tt class="py-line">                    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L557"></a><tt class="py-lineno"> 557</tt>  <tt class="py-line">                        <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-189" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-189', 'id', 'link-166');">id</a></tt> <tt class="py-op">=</tt> <tt id="link-190" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-190', 'annotations', 'link-178');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-name">tags2id</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L558"></a><tt class="py-lineno"> 558</tt>  <tt class="py-line">                    <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L559"></a><tt class="py-lineno"> 559</tt>  <tt class="py-line">                        <tt class="py-keyword">pass</tt> </tt>
<a name="L560"></a><tt class="py-lineno"> 560</tt>  <tt class="py-line">                    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L561"></a><tt class="py-lineno"> 561</tt>  <tt class="py-line">                        <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-191" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-191', 'name', 'link-167');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-192" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-192', 'annotations', 'link-178');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-name">tags2id</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L562"></a><tt class="py-lineno"> 562</tt>  <tt class="py-line">                    <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L563"></a><tt class="py-lineno"> 563</tt>  <tt class="py-line">                        <tt class="py-keyword">pass</tt> </tt>
<a name="L564"></a><tt class="py-lineno"> 564</tt>  <tt class="py-line">                    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L565"></a><tt class="py-lineno"> 565</tt>  <tt class="py-line">                        <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-193" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-193', 'description', 'link-168');">description</a></tt> <tt class="py-op">=</tt> <tt id="link-194" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-194', 'annotations', 'link-178');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-name">tags2id</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L566"></a><tt class="py-lineno"> 566</tt>  <tt class="py-line">                    <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L567"></a><tt class="py-lineno"> 567</tt>  <tt class="py-line">                        <tt class="py-keyword">pass</tt> </tt>
<a name="L568"></a><tt class="py-lineno"> 568</tt>  <tt class="py-line"> </tt>
<a name="L569"></a><tt class="py-lineno"> 569</tt>  <tt class="py-line">                    <tt class="py-keyword">yield</tt> <tt class="py-name">record</tt> </tt>
</div></div><a name="L570"></a><tt class="py-lineno"> 570</tt>  <tt class="py-line"> </tt>
<a name="L571"></a><tt class="py-lineno"> 571</tt>  <tt class="py-line"> </tt>
<a name="EmblScanner"></a><div id="EmblScanner-def"><a name="L572"></a><tt class="py-lineno"> 572</tt> <a class="py-toggle" href="#" id="EmblScanner-toggle" onclick="return toggle('EmblScanner');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.EmblScanner-class.html">EmblScanner</a><tt class="py-op">(</tt><tt class="py-base-class">InsdcScanner</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EmblScanner-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="EmblScanner-expanded"><a name="L573"></a><tt class="py-lineno"> 573</tt>  <tt class="py-line">    <tt class="py-docstring">"""For extracting chunks of information in EMBL files"""</tt> </tt>
<a name="L574"></a><tt class="py-lineno"> 574</tt>  <tt class="py-line"> </tt>
<a name="L575"></a><tt class="py-lineno"> 575</tt>  <tt class="py-line">    <tt id="link-195" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.RECORD_START
Bio.GenBank.Scanner.GenBankScanner.RECORD_START
Bio.GenBank.Scanner.InsdcScanner.RECORD_START" class="py-name" href="#" onclick="return doclink('link-195', 'RECORD_START', 'link-11');">RECORD_START</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"ID   "</tt> </tt>
<a name="L576"></a><tt class="py-lineno"> 576</tt>  <tt class="py-line">    <tt id="link-196" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-196', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt> <tt class="py-op">=</tt> <tt class="py-number">5</tt> </tt>
<a name="L577"></a><tt class="py-lineno"> 577</tt>  <tt class="py-line">    <tt id="link-197" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner.GenBankScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner.InsdcScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner._ImgtScanner.FEATURE_START_MARKERS" class="py-name" href="#" onclick="return doclink('link-197', 'FEATURE_START_MARKERS', 'link-13');">FEATURE_START_MARKERS</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">"FH   Key             Location/Qualifiers"</tt><tt class="py-op">,</tt> <tt class="py-string">"FH"</tt><tt class="py-op">]</tt> </tt>
<a name="L578"></a><tt class="py-lineno"> 578</tt>  <tt class="py-line">    <tt id="link-198" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.FEATURE_END_MARKERS
Bio.GenBank.Scanner.GenBankScanner.FEATURE_END_MARKERS
Bio.GenBank.Scanner.InsdcScanner.FEATURE_END_MARKERS" class="py-name" href="#" onclick="return doclink('link-198', 'FEATURE_END_MARKERS', 'link-14');">FEATURE_END_MARKERS</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">"XX"</tt><tt class="py-op">]</tt>  <tt class="py-comment"># XX can also mark the end of many things!</tt> </tt>
<a name="L579"></a><tt class="py-lineno"> 579</tt>  <tt class="py-line">    <tt id="link-199" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-199', 'FEATURE_QUALIFIER_INDENT', 'link-15');">FEATURE_QUALIFIER_INDENT</a></tt> <tt class="py-op">=</tt> <tt class="py-number">21</tt> </tt>
<a name="L580"></a><tt class="py-lineno"> 580</tt>  <tt class="py-line">    <tt id="link-200" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_SPACER" class="py-name" href="#" onclick="return doclink('link-200', 'FEATURE_QUALIFIER_SPACER', 'link-16');">FEATURE_QUALIFIER_SPACER</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"FT"</tt> <tt class="py-op">+</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt id="link-201" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-201', 'FEATURE_QUALIFIER_INDENT', 'link-15');">FEATURE_QUALIFIER_INDENT</a></tt> <tt class="py-op">-</tt> <tt class="py-number">2</tt><tt class="py-op">)</tt> </tt>
<a name="L581"></a><tt class="py-lineno"> 581</tt>  <tt class="py-line">    <tt id="link-202" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.GenBankScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.InsdcScanner.SEQUENCE_HEADERS" class="py-name" href="#" onclick="return doclink('link-202', 'SEQUENCE_HEADERS', 'link-17');">SEQUENCE_HEADERS</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">"SQ"</tt><tt class="py-op">,</tt> <tt class="py-string">"CO"</tt><tt class="py-op">]</tt>  <tt class="py-comment"># Remove trailing spaces</tt> </tt>
<a name="L582"></a><tt class="py-lineno"> 582</tt>  <tt class="py-line"> </tt>
<a name="EmblScanner.parse_footer"></a><div id="EmblScanner.parse_footer-def"><a name="L583"></a><tt class="py-lineno"> 583</tt> <a class="py-toggle" href="#" id="EmblScanner.parse_footer-toggle" onclick="return toggle('EmblScanner.parse_footer');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.EmblScanner-class.html#parse_footer">parse_footer</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EmblScanner.parse_footer-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EmblScanner.parse_footer-expanded"><a name="L584"></a><tt class="py-lineno"> 584</tt>  <tt class="py-line">        <tt class="py-docstring">"""returns a tuple containing a list of any misc strings, and the sequence"""</tt> </tt>
<a name="L585"></a><tt class="py-lineno"> 585</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-203" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-203', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-204" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-204', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-205" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.GenBankScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.InsdcScanner.SEQUENCE_HEADERS" class="py-name" href="#" onclick="return doclink('link-205', 'SEQUENCE_HEADERS', 'link-17');">SEQUENCE_HEADERS</a></tt><tt class="py-op">,</tt> \ </tt>
<a name="L586"></a><tt class="py-lineno"> 586</tt>  <tt class="py-line">            <tt class="py-string">"Eh? '%s'"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> </tt>
<a name="L587"></a><tt class="py-lineno"> 587</tt>  <tt class="py-line"> </tt>
<a name="L588"></a><tt class="py-lineno"> 588</tt>  <tt class="py-line">        <tt class="py-comment"># Note that the SQ line can be split into several lines...</tt> </tt>
<a name="L589"></a><tt class="py-lineno"> 589</tt>  <tt class="py-line">        <tt class="py-name">misc_lines</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L590"></a><tt class="py-lineno"> 590</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-206" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-206', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-207" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-207', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-208" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.GenBankScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.InsdcScanner.SEQUENCE_HEADERS" class="py-name" href="#" onclick="return doclink('link-208', 'SEQUENCE_HEADERS', 'link-17');">SEQUENCE_HEADERS</a></tt><tt class="py-op">:</tt> </tt>
<a name="L591"></a><tt class="py-lineno"> 591</tt>  <tt class="py-line">            <tt class="py-name">misc_lines</tt><tt class="py-op">.</tt><tt id="link-209" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-209', 'append', 'link-36');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L592"></a><tt class="py-lineno"> 592</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-210" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-210', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L593"></a><tt class="py-lineno"> 593</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L594"></a><tt class="py-lineno"> 594</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Premature end of file"</tt><tt class="py-op">)</tt> </tt>
<a name="L595"></a><tt class="py-lineno"> 595</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-211" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-211', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L596"></a><tt class="py-lineno"> 596</tt>  <tt class="py-line"> </tt>
<a name="L597"></a><tt class="py-lineno"> 597</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-212" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-212', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-213" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-213', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt> \ </tt>
<a name="L598"></a><tt class="py-lineno"> 598</tt>  <tt class="py-line">            <tt class="py-keyword">or</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-214" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-214', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">'//'</tt><tt class="py-op">,</tt> <tt class="py-string">"Unexpected content after SQ or CO line: %r"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> </tt>
<a name="L599"></a><tt class="py-lineno"> 599</tt>  <tt class="py-line"> </tt>
<a name="L600"></a><tt class="py-lineno"> 600</tt>  <tt class="py-line">        <tt class="py-name">seq_lines</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L601"></a><tt class="py-lineno"> 601</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> </tt>
<a name="L602"></a><tt class="py-lineno"> 602</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L603"></a><tt class="py-lineno"> 603</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L604"></a><tt class="py-lineno"> 604</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Premature end of file in sequence data"</tt><tt class="py-op">)</tt> </tt>
<a name="L605"></a><tt class="py-lineno"> 605</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-215" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-215', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L606"></a><tt class="py-lineno"> 606</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L607"></a><tt class="py-lineno"> 607</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Blank line in sequence data"</tt><tt class="py-op">)</tt> </tt>
<a name="L608"></a><tt class="py-lineno"> 608</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt> <tt class="py-op">==</tt> <tt class="py-string">'//'</tt><tt class="py-op">:</tt> </tt>
<a name="L609"></a><tt class="py-lineno"> 609</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L610"></a><tt class="py-lineno"> 610</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-216" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-216', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-217" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-217', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">,</tt> \ </tt>
<a name="L611"></a><tt class="py-lineno"> 611</tt>  <tt class="py-line">                <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L612"></a><tt class="py-lineno"> 612</tt>  <tt class="py-line">            <tt class="py-comment"># Remove tailing number now, remove spaces later</tt> </tt>
<a name="L613"></a><tt class="py-lineno"> 613</tt>  <tt class="py-line">            <tt class="py-name">seq_lines</tt><tt class="py-op">.</tt><tt id="link-218" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-218', 'append', 'link-36');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-219" class="py-name" targets="Method Bio.Seq.Seq.rsplit()=Bio.Seq.Seq-class.html#rsplit"><a title="Bio.Seq.Seq.rsplit" class="py-name" href="#" onclick="return doclink('link-219', 'rsplit', 'link-219');">rsplit</a></tt><tt class="py-op">(</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L614"></a><tt class="py-lineno"> 614</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-220" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-220', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L615"></a><tt class="py-lineno"> 615</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> </tt>
<a name="L616"></a><tt class="py-lineno"> 616</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-name">misc_lines</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">seq_lines</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">" "</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L617"></a><tt class="py-lineno"> 617</tt>  <tt class="py-line"> </tt>
<a name="EmblScanner._feed_first_line"></a><div id="EmblScanner._feed_first_line-def"><a name="L618"></a><tt class="py-lineno"> 618</tt> <a class="py-toggle" href="#" id="EmblScanner._feed_first_line-toggle" onclick="return toggle('EmblScanner._feed_first_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.EmblScanner-class.html#_feed_first_line">_feed_first_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EmblScanner._feed_first_line-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EmblScanner._feed_first_line-expanded"><a name="L619"></a><tt class="py-lineno"> 619</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-221" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-221', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-222" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-222', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">"ID"</tt> </tt>
<a name="L620"></a><tt class="py-lineno"> 620</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-223" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-223', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-224" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-224', 'count', 'link-90');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">";"</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">6</tt><tt class="py-op">:</tt> </tt>
<a name="L621"></a><tt class="py-lineno"> 621</tt>  <tt class="py-line">            <tt class="py-comment"># Looks like the semi colon separated style introduced in 2006</tt> </tt>
<a name="L622"></a><tt class="py-lineno"> 622</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-225" class="py-name" targets="Method Bio.GenBank.Scanner.EmblScanner._feed_first_line_new()=Bio.GenBank.Scanner.EmblScanner-class.html#_feed_first_line_new"><a title="Bio.GenBank.Scanner.EmblScanner._feed_first_line_new" class="py-name" href="#" onclick="return doclink('link-225', '_feed_first_line_new', 'link-225');">_feed_first_line_new</a></tt><tt class="py-op">(</tt><tt class="py-name">consumer</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L623"></a><tt class="py-lineno"> 623</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-226" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-226', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-227" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-227', 'count', 'link-90');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">";"</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">3</tt><tt class="py-op">:</tt> </tt>
<a name="L624"></a><tt class="py-lineno"> 624</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-228" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-228', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-229" class="py-name" targets="Method Bio.Seq.Seq.endswith()=Bio.Seq.Seq-class.html#endswith"><a title="Bio.Seq.Seq.endswith" class="py-name" href="#" onclick="return doclink('link-229', 'endswith', 'link-229');">endswith</a></tt><tt class="py-op">(</tt><tt class="py-string">" SQ"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L625"></a><tt class="py-lineno"> 625</tt>  <tt class="py-line">                <tt class="py-comment"># EMBL-bank patent data</tt> </tt>
<a name="L626"></a><tt class="py-lineno"> 626</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-230" class="py-name" targets="Method Bio.GenBank.Scanner.EmblScanner._feed_first_line_patents()=Bio.GenBank.Scanner.EmblScanner-class.html#_feed_first_line_patents"><a title="Bio.GenBank.Scanner.EmblScanner._feed_first_line_patents" class="py-name" href="#" onclick="return doclink('link-230', '_feed_first_line_patents', 'link-230');">_feed_first_line_patents</a></tt><tt class="py-op">(</tt><tt class="py-name">consumer</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L627"></a><tt class="py-lineno"> 627</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L628"></a><tt class="py-lineno"> 628</tt>  <tt class="py-line">                <tt class="py-comment"># Looks like the pre 2006 style</tt> </tt>
<a name="L629"></a><tt class="py-lineno"> 629</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-231" class="py-name" targets="Method Bio.GenBank.Scanner.EmblScanner._feed_first_line_old()=Bio.GenBank.Scanner.EmblScanner-class.html#_feed_first_line_old"><a title="Bio.GenBank.Scanner.EmblScanner._feed_first_line_old" class="py-name" href="#" onclick="return doclink('link-231', '_feed_first_line_old', 'link-231');">_feed_first_line_old</a></tt><tt class="py-op">(</tt><tt class="py-name">consumer</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L630"></a><tt class="py-lineno"> 630</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L631"></a><tt class="py-lineno"> 631</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">'Did not recognise the ID line layout:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
</div><a name="L632"></a><tt class="py-lineno"> 632</tt>  <tt class="py-line"> </tt>
<a name="EmblScanner._feed_first_line_patents"></a><div id="EmblScanner._feed_first_line_patents-def"><a name="L633"></a><tt class="py-lineno"> 633</tt> <a class="py-toggle" href="#" id="EmblScanner._feed_first_line_patents-toggle" onclick="return toggle('EmblScanner._feed_first_line_patents');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.EmblScanner-class.html#_feed_first_line_patents">_feed_first_line_patents</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EmblScanner._feed_first_line_patents-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EmblScanner._feed_first_line_patents-expanded"><a name="L634"></a><tt class="py-lineno"> 634</tt>  <tt class="py-line">        <tt class="py-comment"># Either Non-Redundant Level 1 database records,</tt> </tt>
<a name="L635"></a><tt class="py-lineno"> 635</tt>  <tt class="py-line">        <tt class="py-comment"># ID &lt;accession&gt;; &lt;molecule type&gt;; &lt;non-redundant level 1&gt;; &lt;cluster size L1&gt;</tt> </tt>
<a name="L636"></a><tt class="py-lineno"> 636</tt>  <tt class="py-line">        <tt class="py-comment"># e.g. ID   NRP_AX000635; PRT; NR1; 15 SQ</tt> </tt>
<a name="L637"></a><tt class="py-lineno"> 637</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L638"></a><tt class="py-lineno"> 638</tt>  <tt class="py-line">        <tt class="py-comment"># Or, Non-Redundant Level 2 database records:</tt> </tt>
<a name="L639"></a><tt class="py-lineno"> 639</tt>  <tt class="py-line">        <tt class="py-comment"># ID &lt;L2-accession&gt;; &lt;molecule type&gt;; &lt;non-redundant level 2&gt;; &lt;cluster size L2&gt;</tt> </tt>
<a name="L640"></a><tt class="py-lineno"> 640</tt>  <tt class="py-line">        <tt class="py-comment"># e.g. ID   NRP0000016E; PRT; NR2; 5 SQ</tt> </tt>
<a name="L641"></a><tt class="py-lineno"> 641</tt>  <tt class="py-line">        <tt class="py-name">fields</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-232" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-232', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-233" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-233', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-234" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-234', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">";"</tt><tt class="py-op">)</tt> </tt>
<a name="L642"></a><tt class="py-lineno"> 642</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">fields</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">4</tt> </tt>
<a name="L643"></a><tt class="py-lineno"> 643</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-235" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.locus()=Bio.GenBank._FeatureConsumer-class.html#locus,Method Bio.GenBank._RecordConsumer.locus()=Bio.GenBank._RecordConsumer-class.html#locus"><a title="Bio.GenBank._FeatureConsumer.locus
Bio.GenBank._RecordConsumer.locus" class="py-name" href="#" onclick="return doclink('link-235', 'locus', 'link-235');">locus</a></tt><tt class="py-op">(</tt><tt class="py-name">fields</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L644"></a><tt class="py-lineno"> 644</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-236" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.residue_type()=Bio.GenBank._FeatureConsumer-class.html#residue_type,Method Bio.GenBank._RecordConsumer.residue_type()=Bio.GenBank._RecordConsumer-class.html#residue_type"><a title="Bio.GenBank._FeatureConsumer.residue_type
Bio.GenBank._RecordConsumer.residue_type" class="py-name" href="#" onclick="return doclink('link-236', 'residue_type', 'link-236');">residue_type</a></tt><tt class="py-op">(</tt><tt class="py-name">fields</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L645"></a><tt class="py-lineno"> 645</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-237" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.data_file_division()=Bio.GenBank._FeatureConsumer-class.html#data_file_division,Method Bio.GenBank._RecordConsumer.data_file_division()=Bio.GenBank._RecordConsumer-class.html#data_file_division"><a title="Bio.GenBank._FeatureConsumer.data_file_division
Bio.GenBank._RecordConsumer.data_file_division" class="py-name" href="#" onclick="return doclink('link-237', 'data_file_division', 'link-237');">data_file_division</a></tt><tt class="py-op">(</tt><tt class="py-name">fields</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L646"></a><tt class="py-lineno"> 646</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Record cluster size?</tt> </tt>
<a name="L647"></a><tt class="py-lineno"> 647</tt>  <tt class="py-line"> </tt>
<a name="EmblScanner._feed_first_line_old"></a><div id="EmblScanner._feed_first_line_old-def"><a name="L648"></a><tt class="py-lineno"> 648</tt> <a class="py-toggle" href="#" id="EmblScanner._feed_first_line_old-toggle" onclick="return toggle('EmblScanner._feed_first_line_old');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.EmblScanner-class.html#_feed_first_line_old">_feed_first_line_old</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EmblScanner._feed_first_line_old-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EmblScanner._feed_first_line_old-expanded"><a name="L649"></a><tt class="py-lineno"> 649</tt>  <tt class="py-line">        <tt class="py-comment"># Expects an ID line in the style before 2006, e.g.</tt> </tt>
<a name="L650"></a><tt class="py-lineno"> 650</tt>  <tt class="py-line">        <tt class="py-comment"># ID   SC10H5 standard; DNA; PRO; 4870 BP.</tt> </tt>
<a name="L651"></a><tt class="py-lineno"> 651</tt>  <tt class="py-line">        <tt class="py-comment"># ID   BSUB9999   standard; circular DNA; PRO; 4214630 BP.</tt> </tt>
<a name="L652"></a><tt class="py-lineno"> 652</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-238" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-238', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-239" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-239', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">"ID"</tt> </tt>
<a name="L653"></a><tt class="py-lineno"> 653</tt>  <tt class="py-line">        <tt class="py-name">fields</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-240" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-240', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-241" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-241', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L654"></a><tt class="py-lineno"> 654</tt>  <tt class="py-line">        <tt class="py-name">fields</tt><tt class="py-op">.</tt><tt id="link-242" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.extend()=Bio.Align.MultipleSeqAlignment-class.html#extend,Method Bio.SCOP.Raf.SeqMap.extend()=Bio.SCOP.Raf.SeqMap-class.html#extend,Method Bio.Seq.MutableSeq.extend()=Bio.Seq.MutableSeq-class.html#extend"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-242', 'extend', 'link-242');">extend</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-243" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-243', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-244" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-244', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-245" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-245', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">";"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L655"></a><tt class="py-lineno"> 655</tt>  <tt class="py-line">        <tt class="py-name">fields</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-246" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Reaction.entry=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#entry,Function Bio.TogoWS.entry()=Bio.TogoWS-module.html#entry"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-246', 'entry', 'link-246');">entry</a></tt><tt class="py-op">.</tt><tt id="link-247" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-247', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-248" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-248', 'entry', 'link-246');">entry</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">fields</tt><tt class="py-op">]</tt> </tt>
<a name="L656"></a><tt class="py-lineno"> 656</tt>  <tt class="py-line">        <tt class="py-string">"""</tt> </tt>
<a name="L657"></a><tt class="py-lineno"> 657</tt>  <tt class="py-line"><tt class="py-string">        The tokens represent:</tt> </tt>
<a name="L658"></a><tt class="py-lineno"> 658</tt>  <tt class="py-line"><tt class="py-string"></tt> </tt>
<a name="L659"></a><tt class="py-lineno"> 659</tt>  <tt class="py-line"><tt class="py-string">           0. Primary accession number</tt> </tt>
<a name="L660"></a><tt class="py-lineno"> 660</tt>  <tt class="py-line"><tt class="py-string">           (space sep)</tt> </tt>
<a name="L661"></a><tt class="py-lineno"> 661</tt>  <tt class="py-line"><tt class="py-string">           1. ??? (e.g. standard)</tt> </tt>
<a name="L662"></a><tt class="py-lineno"> 662</tt>  <tt class="py-line"><tt class="py-string">           (semi-colon)</tt> </tt>
<a name="L663"></a><tt class="py-lineno"> 663</tt>  <tt class="py-line"><tt class="py-string">           2. Topology and/or Molecule type (e.g. 'circular DNA' or 'DNA')</tt> </tt>
<a name="L664"></a><tt class="py-lineno"> 664</tt>  <tt class="py-line"><tt class="py-string">           3. Taxonomic division (e.g. 'PRO')</tt> </tt>
<a name="L665"></a><tt class="py-lineno"> 665</tt>  <tt class="py-line"><tt class="py-string">           4. Sequence length (e.g. '4639675 BP.')</tt> </tt>
<a name="L666"></a><tt class="py-lineno"> 666</tt>  <tt class="py-line"><tt class="py-string">        """</tt> </tt>
<a name="L667"></a><tt class="py-lineno"> 667</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-249" class="py-name"><a title="Bio.GenBank._FeatureConsumer.locus
Bio.GenBank._RecordConsumer.locus" class="py-name" href="#" onclick="return doclink('link-249', 'locus', 'link-235');">locus</a></tt><tt class="py-op">(</tt><tt class="py-name">fields</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>  <tt class="py-comment"># Should we also call the accession consumer?</tt> </tt>
<a name="L668"></a><tt class="py-lineno"> 668</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-250" class="py-name"><a title="Bio.GenBank._FeatureConsumer.residue_type
Bio.GenBank._RecordConsumer.residue_type" class="py-name" href="#" onclick="return doclink('link-250', 'residue_type', 'link-236');">residue_type</a></tt><tt class="py-op">(</tt><tt class="py-name">fields</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L669"></a><tt class="py-lineno"> 669</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-251" class="py-name"><a title="Bio.GenBank._FeatureConsumer.data_file_division
Bio.GenBank._RecordConsumer.data_file_division" class="py-name" href="#" onclick="return doclink('link-251', 'data_file_division', 'link-237');">data_file_division</a></tt><tt class="py-op">(</tt><tt class="py-name">fields</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L670"></a><tt class="py-lineno"> 670</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-252" class="py-name" targets="Method Bio.GenBank.Scanner.EmblScanner._feed_seq_length()=Bio.GenBank.Scanner.EmblScanner-class.html#_feed_seq_length"><a title="Bio.GenBank.Scanner.EmblScanner._feed_seq_length" class="py-name" href="#" onclick="return doclink('link-252', '_feed_seq_length', 'link-252');">_feed_seq_length</a></tt><tt class="py-op">(</tt><tt class="py-name">consumer</tt><tt class="py-op">,</tt> <tt class="py-name">fields</tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L671"></a><tt class="py-lineno"> 671</tt>  <tt class="py-line"> </tt>
<a name="EmblScanner._feed_first_line_new"></a><div id="EmblScanner._feed_first_line_new-def"><a name="L672"></a><tt class="py-lineno"> 672</tt> <a class="py-toggle" href="#" id="EmblScanner._feed_first_line_new-toggle" onclick="return toggle('EmblScanner._feed_first_line_new');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.EmblScanner-class.html#_feed_first_line_new">_feed_first_line_new</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EmblScanner._feed_first_line_new-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EmblScanner._feed_first_line_new-expanded"><a name="L673"></a><tt class="py-lineno"> 673</tt>  <tt class="py-line">        <tt class="py-comment"># Expects an ID line in the style introduced in 2006, e.g.</tt> </tt>
<a name="L674"></a><tt class="py-lineno"> 674</tt>  <tt class="py-line">        <tt class="py-comment"># ID   X56734; SV 1; linear; mRNA; STD; PLN; 1859 BP.</tt> </tt>
<a name="L675"></a><tt class="py-lineno"> 675</tt>  <tt class="py-line">        <tt class="py-comment"># ID   CD789012; SV 4; linear; genomic DNA; HTG; MAM; 500 BP.</tt> </tt>
<a name="L676"></a><tt class="py-lineno"> 676</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-253" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-253', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-254" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-254', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">"ID"</tt> </tt>
<a name="L677"></a><tt class="py-lineno"> 677</tt>  <tt class="py-line">        <tt class="py-name">fields</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-255" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-255', 'data', 'link-143');">data</a></tt><tt class="py-op">.</tt><tt id="link-256" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-256', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-257" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-257', 'data', 'link-143');">data</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-258" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-258', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-259" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-259', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-260" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-260', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">";"</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L678"></a><tt class="py-lineno"> 678</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">fields</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">7</tt> </tt>
<a name="L679"></a><tt class="py-lineno"> 679</tt>  <tt class="py-line">        <tt class="py-string">"""</tt> </tt>
<a name="L680"></a><tt class="py-lineno"> 680</tt>  <tt class="py-line"><tt class="py-string">        The tokens represent:</tt> </tt>
<a name="L681"></a><tt class="py-lineno"> 681</tt>  <tt class="py-line"><tt class="py-string"></tt> </tt>
<a name="L682"></a><tt class="py-lineno"> 682</tt>  <tt class="py-line"><tt class="py-string">           0. Primary accession number</tt> </tt>
<a name="L683"></a><tt class="py-lineno"> 683</tt>  <tt class="py-line"><tt class="py-string">           1. Sequence version number</tt> </tt>
<a name="L684"></a><tt class="py-lineno"> 684</tt>  <tt class="py-line"><tt class="py-string">           2. Topology: 'circular' or 'linear'</tt> </tt>
<a name="L685"></a><tt class="py-lineno"> 685</tt>  <tt class="py-line"><tt class="py-string">           3. Molecule type (e.g. 'genomic DNA')</tt> </tt>
<a name="L686"></a><tt class="py-lineno"> 686</tt>  <tt class="py-line"><tt class="py-string">           4. Data class (e.g. 'STD')</tt> </tt>
<a name="L687"></a><tt class="py-lineno"> 687</tt>  <tt class="py-line"><tt class="py-string">           5. Taxonomic division (e.g. 'PRO')</tt> </tt>
<a name="L688"></a><tt class="py-lineno"> 688</tt>  <tt class="py-line"><tt class="py-string">           6. Sequence length (e.g. '4639675 BP.')</tt> </tt>
<a name="L689"></a><tt class="py-lineno"> 689</tt>  <tt class="py-line"><tt class="py-string">        """</tt> </tt>
<a name="L690"></a><tt class="py-lineno"> 690</tt>  <tt class="py-line"> </tt>
<a name="L691"></a><tt class="py-lineno"> 691</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-261" class="py-name"><a title="Bio.GenBank._FeatureConsumer.locus
Bio.GenBank._RecordConsumer.locus" class="py-name" href="#" onclick="return doclink('link-261', 'locus', 'link-235');">locus</a></tt><tt class="py-op">(</tt><tt class="py-name">fields</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L692"></a><tt class="py-lineno"> 692</tt>  <tt class="py-line"> </tt>
<a name="L693"></a><tt class="py-lineno"> 693</tt>  <tt class="py-line">        <tt class="py-comment"># Call the accession consumer now, to make sure we record</tt> </tt>
<a name="L694"></a><tt class="py-lineno"> 694</tt>  <tt class="py-line">        <tt class="py-comment"># something as the record.id, in case there is no AC line</tt> </tt>
<a name="L695"></a><tt class="py-lineno"> 695</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-262" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.accession()=Bio.GenBank._FeatureConsumer-class.html#accession,Method Bio.GenBank._RecordConsumer.accession()=Bio.GenBank._RecordConsumer-class.html#accession,Method Bio.Phylo.PhyloXMLIO.Parser.accession()=Bio.Phylo.PhyloXMLIO.Parser-class.html#accession,Method Bio.Phylo.PhyloXMLIO.Writer.accession()=Bio.Phylo.PhyloXMLIO.Writer-class.html#accession"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-262', 'accession', 'link-262');">accession</a></tt><tt class="py-op">(</tt><tt class="py-name">fields</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L696"></a><tt class="py-lineno"> 696</tt>  <tt class="py-line"> </tt>
<a name="L697"></a><tt class="py-lineno"> 697</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - How to deal with the version field?  At the moment the consumer</tt> </tt>
<a name="L698"></a><tt class="py-lineno"> 698</tt>  <tt class="py-line">        <tt class="py-comment"># will try and use this for the ID which isn't ideal for EMBL files.</tt> </tt>
<a name="L699"></a><tt class="py-lineno"> 699</tt>  <tt class="py-line">        <tt class="py-name">version_parts</tt> <tt class="py-op">=</tt> <tt class="py-name">fields</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-263" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-263', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L700"></a><tt class="py-lineno"> 700</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">version_parts</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt> \ </tt>
<a name="L701"></a><tt class="py-lineno"> 701</tt>  <tt class="py-line">            <tt class="py-keyword">and</tt> <tt class="py-name">version_parts</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"SV"</tt> \ </tt>
<a name="L702"></a><tt class="py-lineno"> 702</tt>  <tt class="py-line">                <tt class="py-keyword">and</tt> <tt class="py-name">version_parts</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">isdigit</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L703"></a><tt class="py-lineno"> 703</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-264" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.version_suffix()=Bio.GenBank._FeatureConsumer-class.html#version_suffix"><a title="Bio.GenBank._FeatureConsumer.version_suffix" class="py-name" href="#" onclick="return doclink('link-264', 'version_suffix', 'link-264');">version_suffix</a></tt><tt class="py-op">(</tt><tt class="py-name">version_parts</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L704"></a><tt class="py-lineno"> 704</tt>  <tt class="py-line"> </tt>
<a name="L705"></a><tt class="py-lineno"> 705</tt>  <tt class="py-line">        <tt class="py-comment"># Based on how the old GenBank parser worked, merge these two:</tt> </tt>
<a name="L706"></a><tt class="py-lineno"> 706</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-265" class="py-name"><a title="Bio.GenBank._FeatureConsumer.residue_type
Bio.GenBank._RecordConsumer.residue_type" class="py-name" href="#" onclick="return doclink('link-265', 'residue_type', 'link-236');">residue_type</a></tt><tt class="py-op">(</tt><tt class="py-string">" "</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">fields</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">:</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt>  <tt class="py-comment"># TODO - Store as two fields?</tt> </tt>
<a name="L707"></a><tt class="py-lineno"> 707</tt>  <tt class="py-line"> </tt>
<a name="L708"></a><tt class="py-lineno"> 708</tt>  <tt class="py-line">        <tt class="py-comment"># consumer.xxx(fields[4]) # TODO - What should we do with the data class?</tt> </tt>
<a name="L709"></a><tt class="py-lineno"> 709</tt>  <tt class="py-line"> </tt>
<a name="L710"></a><tt class="py-lineno"> 710</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-266" class="py-name"><a title="Bio.GenBank._FeatureConsumer.data_file_division
Bio.GenBank._RecordConsumer.data_file_division" class="py-name" href="#" onclick="return doclink('link-266', 'data_file_division', 'link-237');">data_file_division</a></tt><tt class="py-op">(</tt><tt class="py-name">fields</tt><tt class="py-op">[</tt><tt class="py-number">5</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L711"></a><tt class="py-lineno"> 711</tt>  <tt class="py-line"> </tt>
<a name="L712"></a><tt class="py-lineno"> 712</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-267" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner._feed_seq_length" class="py-name" href="#" onclick="return doclink('link-267', '_feed_seq_length', 'link-252');">_feed_seq_length</a></tt><tt class="py-op">(</tt><tt class="py-name">consumer</tt><tt class="py-op">,</tt> <tt class="py-name">fields</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L713"></a><tt class="py-lineno"> 713</tt>  <tt class="py-line"> </tt>
<a name="EmblScanner._feed_seq_length"></a><div id="EmblScanner._feed_seq_length-def"><a name="L714"></a><tt class="py-lineno"> 714</tt> <a class="py-toggle" href="#" id="EmblScanner._feed_seq_length-toggle" onclick="return toggle('EmblScanner._feed_seq_length');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.EmblScanner-class.html#_feed_seq_length">_feed_seq_length</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EmblScanner._feed_seq_length-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EmblScanner._feed_seq_length-expanded"><a name="L715"></a><tt class="py-lineno"> 715</tt>  <tt class="py-line">        <tt class="py-name">length_parts</tt> <tt class="py-op">=</tt> <tt class="py-name">text</tt><tt class="py-op">.</tt><tt id="link-268" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-268', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L716"></a><tt class="py-lineno"> 716</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">length_parts</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt><tt class="py-op">,</tt> <tt class="py-string">"Invalid sequence length string %r"</tt> <tt class="py-op">%</tt> <tt class="py-name">text</tt> </tt>
<a name="L717"></a><tt class="py-lineno"> 717</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">length_parts</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-269" class="py-name" targets="Method Bio.Seq.Seq.upper()=Bio.Seq.Seq-class.html#upper,Method Bio.Seq.UnknownSeq.upper()=Bio.Seq.UnknownSeq-class.html#upper,Method Bio.SeqRecord.SeqRecord.upper()=Bio.SeqRecord.SeqRecord-class.html#upper"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-269', 'upper', 'link-269');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"BP"</tt><tt class="py-op">,</tt> <tt class="py-string">"BP."</tt><tt class="py-op">,</tt> <tt class="py-string">"AA"</tt><tt class="py-op">,</tt> <tt class="py-string">"AA."</tt><tt class="py-op">]</tt> </tt>
<a name="L718"></a><tt class="py-lineno"> 718</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-270" class="py-name" targets="Variable Bio.Alphabet.Alphabet.size=Bio.Alphabet.Alphabet-class.html#size,Variable Bio.Alphabet.Reduced.HPModel.size=Bio.Alphabet.Reduced.HPModel-class.html#size,Variable Bio.Alphabet.Reduced.Murphy10.size=Bio.Alphabet.Reduced.Murphy10-class.html#size,Variable Bio.Alphabet.Reduced.Murphy15.size=Bio.Alphabet.Reduced.Murphy15-class.html#size,Variable Bio.Alphabet.Reduced.Murphy4.size=Bio.Alphabet.Reduced.Murphy4-class.html#size,Variable Bio.Alphabet.Reduced.Murphy8.size=Bio.Alphabet.Reduced.Murphy8-class.html#size,Variable Bio.Alphabet.Reduced.PC5.size=Bio.Alphabet.Reduced.PC5-class.html#size,Variable Bio.Alphabet.SingleLetterAlphabet.size=Bio.Alphabet.SingleLetterAlphabet-class.html#size,Variable Bio.Alphabet.ThreeLetterProtein.size=Bio.Alphabet.ThreeLetterProtein-class.html#size,Method Bio.GenBank._FeatureConsumer.size()=Bio.GenBank._FeatureConsumer-class.html#size,Method Bio.GenBank._RecordConsumer.size()=Bio.GenBank._RecordConsumer-class.html#size,Variable Bio.Restriction.Restriction.AanI.size=Bio.Restriction.Restriction.AanI-class.html#size,Variable Bio.Restriction.Restriction.AarI.size=Bio.Restriction.Restriction.AarI-class.html#size,Variable Bio.Restriction.Restriction.AasI.size=Bio.Restriction.Restriction.AasI-class.html#size,Variable Bio.Restriction.Restriction.AatII.size=Bio.Restriction.Restriction.AatII-class.html#size,Variable Bio.Restriction.Restriction.AbaSI.size=Bio.Restriction.Restriction.AbaSI-class.html#size,Variable Bio.Restriction.Restriction.AbsI.size=Bio.Restriction.Restriction.AbsI-class.html#size,Variable Bio.Restriction.Restriction.Acc16I.size=Bio.Restriction.Restriction.Acc16I-class.html#size,Variable Bio.Restriction.Restriction.Acc36I.size=Bio.Restriction.Restriction.Acc36I-class.html#size,Variable Bio.Restriction.Restriction.Acc65I.size=Bio.Restriction.Restriction.Acc65I-class.html#size,Variable Bio.Restriction.Restriction.AccB1I.size=Bio.Restriction.Restriction.AccB1I-class.html#size,Variable Bio.Restriction.Restriction.AccB7I.size=Bio.Restriction.Restriction.AccB7I-class.html#size,Variable Bio.Restriction.Restriction.AccBSI.size=Bio.Restriction.Restriction.AccBSI-class.html#size,Variable Bio.Restriction.Restriction.AccI.size=Bio.Restriction.Restriction.AccI-class.html#size,Variable Bio.Restriction.Restriction.AccII.size=Bio.Restriction.Restriction.AccII-class.html#size,Variable Bio.Restriction.Restriction.AccIII.size=Bio.Restriction.Restriction.AccIII-class.html#size,Variable Bio.Restriction.Restriction.AceIII.size=Bio.Restriction.Restriction.AceIII-class.html#size,Variable Bio.Restriction.Restriction.AciI.size=Bio.Restriction.Restriction.AciI-class.html#size,Variable Bio.Restriction.Restriction.AclI.size=Bio.Restriction.Restriction.AclI-class.html#size,Variable Bio.Restriction.Restriction.AclWI.size=Bio.Restriction.Restriction.AclWI-class.html#size,Variable Bio.Restriction.Restriction.AcoI.size=Bio.Restriction.Restriction.AcoI-class.html#size,Variable Bio.Restriction.Restriction.AcsI.size=Bio.Restriction.Restriction.AcsI-class.html#size,Variable Bio.Restriction.Restriction.AcuI.size=Bio.Restriction.Restriction.AcuI-class.html#size,Variable Bio.Restriction.Restriction.AcvI.size=Bio.Restriction.Restriction.AcvI-class.html#size,Variable Bio.Restriction.Restriction.AcyI.size=Bio.Restriction.Restriction.AcyI-class.html#size,Variable Bio.Restriction.Restriction.AdeI.size=Bio.Restriction.Restriction.AdeI-class.html#size,Variable Bio.Restriction.Restriction.AfaI.size=Bio.Restriction.Restriction.AfaI-class.html#size,Variable Bio.Restriction.Restriction.AfeI.size=Bio.Restriction.Restriction.AfeI-class.html#size,Variable Bio.Restriction.Restriction.AfiI.size=Bio.Restriction.Restriction.AfiI-class.html#size,Variable Bio.Restriction.Restriction.AflII.size=Bio.Restriction.Restriction.AflII-class.html#size,Variable Bio.Restriction.Restriction.AflIII.size=Bio.Restriction.Restriction.AflIII-class.html#size,Variable Bio.Restriction.Restriction.AgeI.size=Bio.Restriction.Restriction.AgeI-class.html#size,Variable Bio.Restriction.Restriction.AgsI.size=Bio.Restriction.Restriction.AgsI-class.html#size,Variable Bio.Restriction.Restriction.AhaIII.size=Bio.Restriction.Restriction.AhaIII-class.html#size,Variable Bio.Restriction.Restriction.AhdI.size=Bio.Restriction.Restriction.AhdI-class.html#size,Variable Bio.Restriction.Restriction.AhlI.size=Bio.Restriction.Restriction.AhlI-class.html#size,Variable Bio.Restriction.Restriction.AjiI.size=Bio.Restriction.Restriction.AjiI-class.html#size,Variable Bio.Restriction.Restriction.AjnI.size=Bio.Restriction.Restriction.AjnI-class.html#size,Variable Bio.Restriction.Restriction.AjuI.size=Bio.Restriction.Restriction.AjuI-class.html#size,Variable Bio.Restriction.Restriction.AleI.size=Bio.Restriction.Restriction.AleI-class.html#size,Variable Bio.Restriction.Restriction.AlfI.size=Bio.Restriction.Restriction.AlfI-class.html#size,Variable Bio.Restriction.Restriction.AloI.size=Bio.Restriction.Restriction.AloI-class.html#size,Variable Bio.Restriction.Restriction.AluBI.size=Bio.Restriction.Restriction.AluBI-class.html#size,Variable Bio.Restriction.Restriction.AluI.size=Bio.Restriction.Restriction.AluI-class.html#size,Variable Bio.Restriction.Restriction.Alw21I.size=Bio.Restriction.Restriction.Alw21I-class.html#size,Variable Bio.Restriction.Restriction.Alw26I.size=Bio.Restriction.Restriction.Alw26I-class.html#size,Variable Bio.Restriction.Restriction.Alw44I.size=Bio.Restriction.Restriction.Alw44I-class.html#size,Variable Bio.Restriction.Restriction.AlwFI.size=Bio.Restriction.Restriction.AlwFI-class.html#size,Variable Bio.Restriction.Restriction.AlwI.size=Bio.Restriction.Restriction.AlwI-class.html#size,Variable Bio.Restriction.Restriction.AlwNI.size=Bio.Restriction.Restriction.AlwNI-class.html#size,Variable Bio.Restriction.Restriction.Ama87I.size=Bio.Restriction.Restriction.Ama87I-class.html#size,Variable Bio.Restriction.Restriction.Aor13HI.size=Bio.Restriction.Restriction.Aor13HI-class.html#size,Variable Bio.Restriction.Restriction.Aor51HI.size=Bio.Restriction.Restriction.Aor51HI-class.html#size,Variable Bio.Restriction.Restriction.AoxI.size=Bio.Restriction.Restriction.AoxI-class.html#size,Variable Bio.Restriction.Restriction.ApaBI.size=Bio.Restriction.Restriction.ApaBI-class.html#size,Variable Bio.Restriction.Restriction.ApaI.size=Bio.Restriction.Restriction.ApaI-class.html#size,Variable Bio.Restriction.Restriction.ApaLI.size=Bio.Restriction.Restriction.ApaLI-class.html#size,Variable Bio.Restriction.Restriction.ApeKI.size=Bio.Restriction.Restriction.ApeKI-class.html#size,Variable Bio.Restriction.Restriction.ApoI.size=Bio.Restriction.Restriction.ApoI-class.html#size,Variable Bio.Restriction.Restriction.ApyPI.size=Bio.Restriction.Restriction.ApyPI-class.html#size,Variable Bio.Restriction.Restriction.AquII.size=Bio.Restriction.Restriction.AquII-class.html#size,Variable Bio.Restriction.Restriction.AquIII.size=Bio.Restriction.Restriction.AquIII-class.html#size,Variable Bio.Restriction.Restriction.AquIV.size=Bio.Restriction.Restriction.AquIV-class.html#size,Variable Bio.Restriction.Restriction.ArsI.size=Bio.Restriction.Restriction.ArsI-class.html#size,Variable Bio.Restriction.Restriction.AscI.size=Bio.Restriction.Restriction.AscI-class.html#size,Variable Bio.Restriction.Restriction.AseI.size=Bio.Restriction.Restriction.AseI-class.html#size,Variable Bio.Restriction.Restriction.Asi256I.size=Bio.Restriction.Restriction.Asi256I-class.html#size,Variable Bio.Restriction.Restriction.AsiGI.size=Bio.Restriction.Restriction.AsiGI-class.html#size,Variable Bio.Restriction.Restriction.AsiSI.size=Bio.Restriction.Restriction.AsiSI-class.html#size,Variable Bio.Restriction.Restriction.Asp700I.size=Bio.Restriction.Restriction.Asp700I-class.html#size,Variable Bio.Restriction.Restriction.Asp718I.size=Bio.Restriction.Restriction.Asp718I-class.html#size,Variable Bio.Restriction.Restriction.AspA2I.size=Bio.Restriction.Restriction.AspA2I-class.html#size,Variable Bio.Restriction.Restriction.AspBHI.size=Bio.Restriction.Restriction.AspBHI-class.html#size,Variable Bio.Restriction.Restriction.AspLEI.size=Bio.Restriction.Restriction.AspLEI-class.html#size,Variable Bio.Restriction.Restriction.AspS9I.size=Bio.Restriction.Restriction.AspS9I-class.html#size,Variable Bio.Restriction.Restriction.AssI.size=Bio.Restriction.Restriction.AssI-class.html#size,Variable Bio.Restriction.Restriction.AsuC2I.size=Bio.Restriction.Restriction.AsuC2I-class.html#size,Variable Bio.Restriction.Restriction.AsuHPI.size=Bio.Restriction.Restriction.AsuHPI-class.html#size,Variable Bio.Restriction.Restriction.AsuI.size=Bio.Restriction.Restriction.AsuI-class.html#size,Variable Bio.Restriction.Restriction.AsuII.size=Bio.Restriction.Restriction.AsuII-class.html#size,Variable Bio.Restriction.Restriction.AsuNHI.size=Bio.Restriction.Restriction.AsuNHI-class.html#size,Variable Bio.Restriction.Restriction.AvaI.size=Bio.Restriction.Restriction.AvaI-class.html#size,Variable Bio.Restriction.Restriction.AvaII.size=Bio.Restriction.Restriction.AvaII-class.html#size,Variable Bio.Restriction.Restriction.AvaIII.size=Bio.Restriction.Restriction.AvaIII-class.html#size,Variable Bio.Restriction.Restriction.AvrII.size=Bio.Restriction.Restriction.AvrII-class.html#size,Variable Bio.Restriction.Restriction.AxyI.size=Bio.Restriction.Restriction.AxyI-class.html#size,Variable Bio.Restriction.Restriction.BaeGI.size=Bio.Restriction.Restriction.BaeGI-class.html#size,Variable Bio.Restriction.Restriction.BaeI.size=Bio.Restriction.Restriction.BaeI-class.html#size,Variable Bio.Restriction.Restriction.BalI.size=Bio.Restriction.Restriction.BalI-class.html#size,Variable Bio.Restriction.Restriction.BamHI.size=Bio.Restriction.Restriction.BamHI-class.html#size,Variable Bio.Restriction.Restriction.BanI.size=Bio.Restriction.Restriction.BanI-class.html#size,Variable Bio.Restriction.Restriction.BanII.size=Bio.Restriction.Restriction.BanII-class.html#size,Variable Bio.Restriction.Restriction.BarI.size=Bio.Restriction.Restriction.BarI-class.html#size,Variable Bio.Restriction.Restriction.BasI.size=Bio.Restriction.Restriction.BasI-class.html#size,Variable Bio.Restriction.Restriction.BauI.size=Bio.Restriction.Restriction.BauI-class.html#size,Variable Bio.Restriction.Restriction.Bbr7I.size=Bio.Restriction.Restriction.Bbr7I-class.html#size,Variable Bio.Restriction.Restriction.BbrPI.size=Bio.Restriction.Restriction.BbrPI-class.html#size,Variable Bio.Restriction.Restriction.BbsI.size=Bio.Restriction.Restriction.BbsI-class.html#size,Variable Bio.Restriction.Restriction.Bbv12I.size=Bio.Restriction.Restriction.Bbv12I-class.html#size,Variable Bio.Restriction.Restriction.BbvCI.size=Bio.Restriction.Restriction.BbvCI-class.html#size,Variable Bio.Restriction.Restriction.BbvI.size=Bio.Restriction.Restriction.BbvI-class.html#size,Variable Bio.Restriction.Restriction.BbvII.size=Bio.Restriction.Restriction.BbvII-class.html#size,Variable Bio.Restriction.Restriction.BccI.size=Bio.Restriction.Restriction.BccI-class.html#size,Variable Bio.Restriction.Restriction.Bce83I.size=Bio.Restriction.Restriction.Bce83I-class.html#size,Variable Bio.Restriction.Restriction.BceAI.size=Bio.Restriction.Restriction.BceAI-class.html#size,Variable Bio.Restriction.Restriction.BcefI.size=Bio.Restriction.Restriction.BcefI-class.html#size,Variable Bio.Restriction.Restriction.BcgI.size=Bio.Restriction.Restriction.BcgI-class.html#size,Variable Bio.Restriction.Restriction.BciT130I.size=Bio.Restriction.Restriction.BciT130I-class.html#size,Variable Bio.Restriction.Restriction.BciVI.size=Bio.Restriction.Restriction.BciVI-class.html#size,Variable Bio.Restriction.Restriction.BclI.size=Bio.Restriction.Restriction.BclI-class.html#size,Variable Bio.Restriction.Restriction.BcnI.size=Bio.Restriction.Restriction.BcnI-class.html#size,Variable Bio.Restriction.Restriction.BcoDI.size=Bio.Restriction.Restriction.BcoDI-class.html#size,Variable Bio.Restriction.Restriction.BcuI.size=Bio.Restriction.Restriction.BcuI-class.html#size,Variable Bio.Restriction.Restriction.BdaI.size=Bio.Restriction.Restriction.BdaI-class.html#size,Variable Bio.Restriction.Restriction.BetI.size=Bio.Restriction.Restriction.BetI-class.html#size,Variable Bio.Restriction.Restriction.BfaI.size=Bio.Restriction.Restriction.BfaI-class.html#size,Variable Bio.Restriction.Restriction.BfiI.size=Bio.Restriction.Restriction.BfiI-class.html#size,Variable Bio.Restriction.Restriction.BfmI.size=Bio.Restriction.Restriction.BfmI-class.html#size,Variable Bio.Restriction.Restriction.BfoI.size=Bio.Restriction.Restriction.BfoI-class.html#size,Variable Bio.Restriction.Restriction.BfrI.size=Bio.Restriction.Restriction.BfrI-class.html#size,Variable Bio.Restriction.Restriction.BfuAI.size=Bio.Restriction.Restriction.BfuAI-class.html#size,Variable Bio.Restriction.Restriction.BfuCI.size=Bio.Restriction.Restriction.BfuCI-class.html#size,Variable Bio.Restriction.Restriction.BfuI.size=Bio.Restriction.Restriction.BfuI-class.html#size,Variable Bio.Restriction.Restriction.BglI.size=Bio.Restriction.Restriction.BglI-class.html#size,Variable Bio.Restriction.Restriction.BglII.size=Bio.Restriction.Restriction.BglII-class.html#size,Variable Bio.Restriction.Restriction.BinI.size=Bio.Restriction.Restriction.BinI-class.html#size,Variable Bio.Restriction.Restriction.BisI.size=Bio.Restriction.Restriction.BisI-class.html#size,Variable Bio.Restriction.Restriction.BlnI.size=Bio.Restriction.Restriction.BlnI-class.html#size,Variable Bio.Restriction.Restriction.BlpI.size=Bio.Restriction.Restriction.BlpI-class.html#size,Variable Bio.Restriction.Restriction.BlsI.size=Bio.Restriction.Restriction.BlsI-class.html#size,Variable Bio.Restriction.Restriction.BmcAI.size=Bio.Restriction.Restriction.BmcAI-class.html#size,Variable Bio.Restriction.Restriction.Bme1390I.size=Bio.Restriction.Restriction.Bme1390I-class.html#size,Variable Bio.Restriction.Restriction.Bme18I.size=Bio.Restriction.Restriction.Bme18I-class.html#size,Variable Bio.Restriction.Restriction.BmeDI.size=Bio.Restriction.Restriction.BmeDI-class.html#size,Variable Bio.Restriction.Restriction.BmeRI.size=Bio.Restriction.Restriction.BmeRI-class.html#size,Variable Bio.Restriction.Restriction.BmeT110I.size=Bio.Restriction.Restriction.BmeT110I-class.html#size,Variable Bio.Restriction.Restriction.BmgBI.size=Bio.Restriction.Restriction.BmgBI-class.html#size,Variable Bio.Restriction.Restriction.BmgI.size=Bio.Restriction.Restriction.BmgI-class.html#size,Variable Bio.Restriction.Restriction.BmgT120I.size=Bio.Restriction.Restriction.BmgT120I-class.html#size,Variable Bio.Restriction.Restriction.BmiI.size=Bio.Restriction.Restriction.BmiI-class.html#size,Variable Bio.Restriction.Restriction.BmrFI.size=Bio.Restriction.Restriction.BmrFI-class.html#size,Variable Bio.Restriction.Restriction.BmrI.size=Bio.Restriction.Restriction.BmrI-class.html#size,Variable Bio.Restriction.Restriction.BmsI.size=Bio.Restriction.Restriction.BmsI-class.html#size,Variable Bio.Restriction.Restriction.BmtI.size=Bio.Restriction.Restriction.BmtI-class.html#size,Variable Bio.Restriction.Restriction.BmuI.size=Bio.Restriction.Restriction.BmuI-class.html#size,Variable Bio.Restriction.Restriction.BoxI.size=Bio.Restriction.Restriction.BoxI-class.html#size,Variable Bio.Restriction.Restriction.BpiI.size=Bio.Restriction.Restriction.BpiI-class.html#size,Variable Bio.Restriction.Restriction.BplI.size=Bio.Restriction.Restriction.BplI-class.html#size,Variable Bio.Restriction.Restriction.BpmI.size=Bio.Restriction.Restriction.BpmI-class.html#size,Variable Bio.Restriction.Restriction.Bpu10I.size=Bio.Restriction.Restriction.Bpu10I-class.html#size,Variable Bio.Restriction.Restriction.Bpu1102I.size=Bio.Restriction.Restriction.Bpu1102I-class.html#size,Variable Bio.Restriction.Restriction.Bpu14I.size=Bio.Restriction.Restriction.Bpu14I-class.html#size,Variable Bio.Restriction.Restriction.BpuEI.size=Bio.Restriction.Restriction.BpuEI-class.html#size,Variable Bio.Restriction.Restriction.BpuMI.size=Bio.Restriction.Restriction.BpuMI-class.html#size,Variable Bio.Restriction.Restriction.BpvUI.size=Bio.Restriction.Restriction.BpvUI-class.html#size,Variable Bio.Restriction.Restriction.Bsa29I.size=Bio.Restriction.Restriction.Bsa29I-class.html#size,Variable Bio.Restriction.Restriction.BsaAI.size=Bio.Restriction.Restriction.BsaAI-class.html#size,Variable Bio.Restriction.Restriction.BsaBI.size=Bio.Restriction.Restriction.BsaBI-class.html#size,Variable Bio.Restriction.Restriction.BsaHI.size=Bio.Restriction.Restriction.BsaHI-class.html#size,Variable Bio.Restriction.Restriction.BsaI.size=Bio.Restriction.Restriction.BsaI-class.html#size,Variable Bio.Restriction.Restriction.BsaJI.size=Bio.Restriction.Restriction.BsaJI-class.html#size,Variable Bio.Restriction.Restriction.BsaWI.size=Bio.Restriction.Restriction.BsaWI-class.html#size,Variable Bio.Restriction.Restriction.BsaXI.size=Bio.Restriction.Restriction.BsaXI-class.html#size,Variable Bio.Restriction.Restriction.BsbI.size=Bio.Restriction.Restriction.BsbI-class.html#size,Variable Bio.Restriction.Restriction.Bsc4I.size=Bio.Restriction.Restriction.Bsc4I-class.html#size,Variable Bio.Restriction.Restriction.BscAI.size=Bio.Restriction.Restriction.BscAI-class.html#size,Variable Bio.Restriction.Restriction.BscGI.size=Bio.Restriction.Restriction.BscGI-class.html#size,Variable Bio.Restriction.Restriction.Bse118I.size=Bio.Restriction.Restriction.Bse118I-class.html#size,Variable Bio.Restriction.Restriction.Bse1I.size=Bio.Restriction.Restriction.Bse1I-class.html#size,Variable Bio.Restriction.Restriction.Bse21I.size=Bio.Restriction.Restriction.Bse21I-class.html#size,Variable Bio.Restriction.Restriction.Bse3DI.size=Bio.Restriction.Restriction.Bse3DI-class.html#size,Variable Bio.Restriction.Restriction.Bse8I.size=Bio.Restriction.Restriction.Bse8I-class.html#size,Variable Bio.Restriction.Restriction.BseAI.size=Bio.Restriction.Restriction.BseAI-class.html#size,Variable Bio.Restriction.Restriction.BseBI.size=Bio.Restriction.Restriction.BseBI-class.html#size,Variable Bio.Restriction.Restriction.BseCI.size=Bio.Restriction.Restriction.BseCI-class.html#size,Variable Bio.Restriction.Restriction.BseDI.size=Bio.Restriction.Restriction.BseDI-class.html#size,Variable Bio.Restriction.Restriction.BseGI.size=Bio.Restriction.Restriction.BseGI-class.html#size,Variable Bio.Restriction.Restriction.BseJI.size=Bio.Restriction.Restriction.BseJI-class.html#size,Variable Bio.Restriction.Restriction.BseLI.size=Bio.Restriction.Restriction.BseLI-class.html#size,Variable Bio.Restriction.Restriction.BseMI.size=Bio.Restriction.Restriction.BseMI-class.html#size,Variable Bio.Restriction.Restriction.BseMII.size=Bio.Restriction.Restriction.BseMII-class.html#size,Variable Bio.Restriction.Restriction.BseNI.size=Bio.Restriction.Restriction.BseNI-class.html#size,Variable Bio.Restriction.Restriction.BsePI.size=Bio.Restriction.Restriction.BsePI-class.html#size,Variable Bio.Restriction.Restriction.BseRI.size=Bio.Restriction.Restriction.BseRI-class.html#size,Variable Bio.Restriction.Restriction.BseSI.size=Bio.Restriction.Restriction.BseSI-class.html#size,Variable Bio.Restriction.Restriction.BseX3I.size=Bio.Restriction.Restriction.BseX3I-class.html#size,Variable Bio.Restriction.Restriction.BseXI.size=Bio.Restriction.Restriction.BseXI-class.html#size,Variable Bio.Restriction.Restriction.BseYI.size=Bio.Restriction.Restriction.BseYI-class.html#size,Variable Bio.Restriction.Restriction.BsgI.size=Bio.Restriction.Restriction.BsgI-class.html#size,Variable Bio.Restriction.Restriction.Bsh1236I.size=Bio.Restriction.Restriction.Bsh1236I-class.html#size,Variable Bio.Restriction.Restriction.Bsh1285I.size=Bio.Restriction.Restriction.Bsh1285I-class.html#size,Variable Bio.Restriction.Restriction.BshFI.size=Bio.Restriction.Restriction.BshFI-class.html#size,Variable Bio.Restriction.Restriction.BshNI.size=Bio.Restriction.Restriction.BshNI-class.html#size,Variable Bio.Restriction.Restriction.BshTI.size=Bio.Restriction.Restriction.BshTI-class.html#size,Variable Bio.Restriction.Restriction.BshVI.size=Bio.Restriction.Restriction.BshVI-class.html#size,Variable Bio.Restriction.Restriction.BsiEI.size=Bio.Restriction.Restriction.BsiEI-class.html#size,Variable Bio.Restriction.Restriction.BsiHKAI.size=Bio.Restriction.Restriction.BsiHKAI-class.html#size,Variable Bio.Restriction.Restriction.BsiHKCI.size=Bio.Restriction.Restriction.BsiHKCI-class.html#size,Variable Bio.Restriction.Restriction.BsiI.size=Bio.Restriction.Restriction.BsiI-class.html#size,Variable Bio.Restriction.Restriction.BsiSI.size=Bio.Restriction.Restriction.BsiSI-class.html#size,Variable Bio.Restriction.Restriction.BsiWI.size=Bio.Restriction.Restriction.BsiWI-class.html#size,Variable Bio.Restriction.Restriction.BsiYI.size=Bio.Restriction.Restriction.BsiYI-class.html#size,Variable Bio.Restriction.Restriction.BslFI.size=Bio.Restriction.Restriction.BslFI-class.html#size,Variable Bio.Restriction.Restriction.BslI.size=Bio.Restriction.Restriction.BslI-class.html#size,Variable Bio.Restriction.Restriction.BsmAI.size=Bio.Restriction.Restriction.BsmAI-class.html#size,Variable Bio.Restriction.Restriction.BsmBI.size=Bio.Restriction.Restriction.BsmBI-class.html#size,Variable Bio.Restriction.Restriction.BsmFI.size=Bio.Restriction.Restriction.BsmFI-class.html#size,Variable Bio.Restriction.Restriction.BsmI.size=Bio.Restriction.Restriction.BsmI-class.html#size,Variable Bio.Restriction.Restriction.BsnI.size=Bio.Restriction.Restriction.BsnI-class.html#size,Variable Bio.Restriction.Restriction.Bso31I.size=Bio.Restriction.Restriction.Bso31I-class.html#size,Variable Bio.Restriction.Restriction.BsoBI.size=Bio.Restriction.Restriction.BsoBI-class.html#size,Variable Bio.Restriction.Restriction.Bsp119I.size=Bio.Restriction.Restriction.Bsp119I-class.html#size,Variable Bio.Restriction.Restriction.Bsp120I.size=Bio.Restriction.Restriction.Bsp120I-class.html#size,Variable Bio.Restriction.Restriction.Bsp1286I.size=Bio.Restriction.Restriction.Bsp1286I-class.html#size,Variable Bio.Restriction.Restriction.Bsp13I.size=Bio.Restriction.Restriction.Bsp13I-class.html#size,Variable Bio.Restriction.Restriction.Bsp1407I.size=Bio.Restriction.Restriction.Bsp1407I-class.html#size,Variable Bio.Restriction.Restriction.Bsp143I.size=Bio.Restriction.Restriction.Bsp143I-class.html#size,Variable Bio.Restriction.Restriction.Bsp1720I.size=Bio.Restriction.Restriction.Bsp1720I-class.html#size,Variable Bio.Restriction.Restriction.Bsp19I.size=Bio.Restriction.Restriction.Bsp19I-class.html#size,Variable Bio.Restriction.Restriction.Bsp24I.size=Bio.Restriction.Restriction.Bsp24I-class.html#size,Variable Bio.Restriction.Restriction.Bsp68I.size=Bio.Restriction.Restriction.Bsp68I-class.html#size,Variable Bio.Restriction.Restriction.BspACI.size=Bio.Restriction.Restriction.BspACI-class.html#size,Variable Bio.Restriction.Restriction.BspCNI.size=Bio.Restriction.Restriction.BspCNI-class.html#size,Variable Bio.Restriction.Restriction.BspD6I.size=Bio.Restriction.Restriction.BspD6I-class.html#size,Variable Bio.Restriction.Restriction.BspDI.size=Bio.Restriction.Restriction.BspDI-class.html#size,Variable Bio.Restriction.Restriction.BspEI.size=Bio.Restriction.Restriction.BspEI-class.html#size,Variable Bio.Restriction.Restriction.BspFNI.size=Bio.Restriction.Restriction.BspFNI-class.html#size,Variable Bio.Restriction.Restriction.BspGI.size=Bio.Restriction.Restriction.BspGI-class.html#size,Variable Bio.Restriction.Restriction.BspHI.size=Bio.Restriction.Restriction.BspHI-class.html#size,Variable Bio.Restriction.Restriction.BspLI.size=Bio.Restriction.Restriction.BspLI-class.html#size,Variable Bio.Restriction.Restriction.BspLU11I.size=Bio.Restriction.Restriction.BspLU11I-class.html#size,Variable Bio.Restriction.Restriction.BspMI.size=Bio.Restriction.Restriction.BspMI-class.html#size,Variable Bio.Restriction.Restriction.BspMII.size=Bio.Restriction.Restriction.BspMII-class.html#size,Variable Bio.Restriction.Restriction.BspNCI.size=Bio.Restriction.Restriction.BspNCI-class.html#size,Variable Bio.Restriction.Restriction.BspOI.size=Bio.Restriction.Restriction.BspOI-class.html#size,Variable Bio.Restriction.Restriction.BspPI.size=Bio.Restriction.Restriction.BspPI-class.html#size,Variable Bio.Restriction.Restriction.BspQI.size=Bio.Restriction.Restriction.BspQI-class.html#size,Variable Bio.Restriction.Restriction.BspT104I.size=Bio.Restriction.Restriction.BspT104I-class.html#size,Variable Bio.Restriction.Restriction.BspT107I.size=Bio.Restriction.Restriction.BspT107I-class.html#size,Variable Bio.Restriction.Restriction.BspTI.size=Bio.Restriction.Restriction.BspTI-class.html#size,Variable Bio.Restriction.Restriction.BsrBI.size=Bio.Restriction.Restriction.BsrBI-class.html#size,Variable Bio.Restriction.Restriction.BsrDI.size=Bio.Restriction.Restriction.BsrDI-class.html#size,Variable Bio.Restriction.Restriction.BsrFI.size=Bio.Restriction.Restriction.BsrFI-class.html#size,Variable Bio.Restriction.Restriction.BsrGI.size=Bio.Restriction.Restriction.BsrGI-class.html#size,Variable Bio.Restriction.Restriction.BsrI.size=Bio.Restriction.Restriction.BsrI-class.html#size,Variable Bio.Restriction.Restriction.BsrSI.size=Bio.Restriction.Restriction.BsrSI-class.html#size,Variable Bio.Restriction.Restriction.BssAI.size=Bio.Restriction.Restriction.BssAI-class.html#size,Variable Bio.Restriction.Restriction.BssECI.size=Bio.Restriction.Restriction.BssECI-class.html#size,Variable Bio.Restriction.Restriction.BssHII.size=Bio.Restriction.Restriction.BssHII-class.html#size,Variable Bio.Restriction.Restriction.BssKI.size=Bio.Restriction.Restriction.BssKI-class.html#size,Variable Bio.Restriction.Restriction.BssMI.size=Bio.Restriction.Restriction.BssMI-class.html#size,Variable Bio.Restriction.Restriction.BssNAI.size=Bio.Restriction.Restriction.BssNAI-class.html#size,Variable Bio.Restriction.Restriction.BssNI.size=Bio.Restriction.Restriction.BssNI-class.html#size,Variable Bio.Restriction.Restriction.BssSI.size=Bio.Restriction.Restriction.BssSI-class.html#size,Variable Bio.Restriction.Restriction.BssT1I.size=Bio.Restriction.Restriction.BssT1I-class.html#size,Variable Bio.Restriction.Restriction.Bst1107I.size=Bio.Restriction.Restriction.Bst1107I-class.html#size,Variable Bio.Restriction.Restriction.Bst2BI.size=Bio.Restriction.Restriction.Bst2BI-class.html#size,Variable Bio.Restriction.Restriction.Bst2UI.size=Bio.Restriction.Restriction.Bst2UI-class.html#size,Variable Bio.Restriction.Restriction.Bst4CI.size=Bio.Restriction.Restriction.Bst4CI-class.html#size,Variable Bio.Restriction.Restriction.Bst6I.size=Bio.Restriction.Restriction.Bst6I-class.html#size,Variable Bio.Restriction.Restriction.BstACI.size=Bio.Restriction.Restriction.BstACI-class.html#size,Variable Bio.Restriction.Restriction.BstAFI.size=Bio.Restriction.Restriction.BstAFI-class.html#size,Variable Bio.Restriction.Restriction.BstAPI.size=Bio.Restriction.Restriction.BstAPI-class.html#size,Variable Bio.Restriction.Restriction.BstAUI.size=Bio.Restriction.Restriction.BstAUI-class.html#size,Variable Bio.Restriction.Restriction.BstBAI.size=Bio.Restriction.Restriction.BstBAI-class.html#size,Variable Bio.Restriction.Restriction.BstBI.size=Bio.Restriction.Restriction.BstBI-class.html#size,Variable Bio.Restriction.Restriction.BstC8I.size=Bio.Restriction.Restriction.BstC8I-class.html#size,Variable Bio.Restriction.Restriction.BstDEI.size=Bio.Restriction.Restriction.BstDEI-class.html#size,Variable Bio.Restriction.Restriction.BstDSI.size=Bio.Restriction.Restriction.BstDSI-class.html#size,Variable Bio.Restriction.Restriction.BstEII.size=Bio.Restriction.Restriction.BstEII-class.html#size,Variable Bio.Restriction.Restriction.BstENI.size=Bio.Restriction.Restriction.BstENI-class.html#size,Variable Bio.Restriction.Restriction.BstF5I.size=Bio.Restriction.Restriction.BstF5I-class.html#size,Variable Bio.Restriction.Restriction.BstFNI.size=Bio.Restriction.Restriction.BstFNI-class.html#size,Variable Bio.Restriction.Restriction.BstH2I.size=Bio.Restriction.Restriction.BstH2I-class.html#size,Variable Bio.Restriction.Restriction.BstHHI.size=Bio.Restriction.Restriction.BstHHI-class.html#size,Variable Bio.Restriction.Restriction.BstKTI.size=Bio.Restriction.Restriction.BstKTI-class.html#size,Variable Bio.Restriction.Restriction.BstMAI.size=Bio.Restriction.Restriction.BstMAI-class.html#size,Variable Bio.Restriction.Restriction.BstMBI.size=Bio.Restriction.Restriction.BstMBI-class.html#size,Variable Bio.Restriction.Restriction.BstMCI.size=Bio.Restriction.Restriction.BstMCI-class.html#size,Variable Bio.Restriction.Restriction.BstMWI.size=Bio.Restriction.Restriction.BstMWI-class.html#size,Variable Bio.Restriction.Restriction.BstNI.size=Bio.Restriction.Restriction.BstNI-class.html#size,Variable Bio.Restriction.Restriction.BstNSI.size=Bio.Restriction.Restriction.BstNSI-class.html#size,Variable Bio.Restriction.Restriction.BstOI.size=Bio.Restriction.Restriction.BstOI-class.html#size,Variable Bio.Restriction.Restriction.BstPAI.size=Bio.Restriction.Restriction.BstPAI-class.html#size,Variable Bio.Restriction.Restriction.BstPI.size=Bio.Restriction.Restriction.BstPI-class.html#size,Variable Bio.Restriction.Restriction.BstSCI.size=Bio.Restriction.Restriction.BstSCI-class.html#size,Variable Bio.Restriction.Restriction.BstSFI.size=Bio.Restriction.Restriction.BstSFI-class.html#size,Variable Bio.Restriction.Restriction.BstSLI.size=Bio.Restriction.Restriction.BstSLI-class.html#size,Variable Bio.Restriction.Restriction.BstSNI.size=Bio.Restriction.Restriction.BstSNI-class.html#size,Variable Bio.Restriction.Restriction.BstUI.size=Bio.Restriction.Restriction.BstUI-class.html#size,Variable Bio.Restriction.Restriction.BstV1I.size=Bio.Restriction.Restriction.BstV1I-class.html#size,Variable Bio.Restriction.Restriction.BstV2I.size=Bio.Restriction.Restriction.BstV2I-class.html#size,Variable Bio.Restriction.Restriction.BstX2I.size=Bio.Restriction.Restriction.BstX2I-class.html#size,Variable Bio.Restriction.Restriction.BstXI.size=Bio.Restriction.Restriction.BstXI-class.html#size,Variable Bio.Restriction.Restriction.BstYI.size=Bio.Restriction.Restriction.BstYI-class.html#size,Variable Bio.Restriction.Restriction.BstZ17I.size=Bio.Restriction.Restriction.BstZ17I-class.html#size,Variable Bio.Restriction.Restriction.BstZI.size=Bio.Restriction.Restriction.BstZI-class.html#size,Variable Bio.Restriction.Restriction.Bsu15I.size=Bio.Restriction.Restriction.Bsu15I-class.html#size,Variable Bio.Restriction.Restriction.Bsu36I.size=Bio.Restriction.Restriction.Bsu36I-class.html#size,Variable Bio.Restriction.Restriction.BsuI.size=Bio.Restriction.Restriction.BsuI-class.html#size,Variable Bio.Restriction.Restriction.BsuRI.size=Bio.Restriction.Restriction.BsuRI-class.html#size,Variable Bio.Restriction.Restriction.BtgI.size=Bio.Restriction.Restriction.BtgI-class.html#size,Variable Bio.Restriction.Restriction.BtgZI.size=Bio.Restriction.Restriction.BtgZI-class.html#size,Variable Bio.Restriction.Restriction.BthCI.size=Bio.Restriction.Restriction.BthCI-class.html#size,Variable Bio.Restriction.Restriction.BtrI.size=Bio.Restriction.Restriction.BtrI-class.html#size,Variable Bio.Restriction.Restriction.BtsCI.size=Bio.Restriction.Restriction.BtsCI-class.html#size,Variable Bio.Restriction.Restriction.BtsI.size=Bio.Restriction.Restriction.BtsI-class.html#size,Variable Bio.Restriction.Restriction.BtsIMutI.size=Bio.Restriction.Restriction.BtsIMutI-class.html#size,Variable Bio.Restriction.Restriction.BtuMI.size=Bio.Restriction.Restriction.BtuMI-class.html#size,Variable Bio.Restriction.Restriction.BveI.size=Bio.Restriction.Restriction.BveI-class.html#size,Variable Bio.Restriction.Restriction.Cac8I.size=Bio.Restriction.Restriction.Cac8I-class.html#size,Variable Bio.Restriction.Restriction.CaiI.size=Bio.Restriction.Restriction.CaiI-class.html#size,Variable Bio.Restriction.Restriction.CauII.size=Bio.Restriction.Restriction.CauII-class.html#size,Variable Bio.Restriction.Restriction.CchII.size=Bio.Restriction.Restriction.CchII-class.html#size,Variable Bio.Restriction.Restriction.CchIII.size=Bio.Restriction.Restriction.CchIII-class.html#size,Variable Bio.Restriction.Restriction.CciI.size=Bio.Restriction.Restriction.CciI-class.html#size,Variable Bio.Restriction.Restriction.CciNI.size=Bio.Restriction.Restriction.CciNI-class.html#size,Variable Bio.Restriction.Restriction.Cdi630V.size=Bio.Restriction.Restriction.Cdi630V-class.html#size,Variable Bio.Restriction.Restriction.CdiI.size=Bio.Restriction.Restriction.CdiI-class.html#size,Variable Bio.Restriction.Restriction.CdpI.size=Bio.Restriction.Restriction.CdpI-class.html#size,Variable Bio.Restriction.Restriction.CfoI.size=Bio.Restriction.Restriction.CfoI-class.html#size,Variable Bio.Restriction.Restriction.Cfr10I.size=Bio.Restriction.Restriction.Cfr10I-class.html#size,Variable Bio.Restriction.Restriction.Cfr13I.size=Bio.Restriction.Restriction.Cfr13I-class.html#size,Variable Bio.Restriction.Restriction.Cfr42I.size=Bio.Restriction.Restriction.Cfr42I-class.html#size,Variable Bio.Restriction.Restriction.Cfr9I.size=Bio.Restriction.Restriction.Cfr9I-class.html#size,Variable Bio.Restriction.Restriction.CfrI.size=Bio.Restriction.Restriction.CfrI-class.html#size,Variable Bio.Restriction.Restriction.Cgl13032I.size=Bio.Restriction.Restriction.Cgl13032I-class.html#size,Variable Bio.Restriction.Restriction.Cgl13032II.size=Bio.Restriction.Restriction.Cgl13032II-class.html#size,Variable Bio.Restriction.Restriction.ChaI.size=Bio.Restriction.Restriction.ChaI-class.html#size,Variable Bio.Restriction.Restriction.CjeFIII.size=Bio.Restriction.Restriction.CjeFIII-class.html#size,Variable Bio.Restriction.Restriction.CjeFV.size=Bio.Restriction.Restriction.CjeFV-class.html#size,Variable Bio.Restriction.Restriction.CjeI.size=Bio.Restriction.Restriction.CjeI-class.html#size,Variable Bio.Restriction.Restriction.CjeNII.size=Bio.Restriction.Restriction.CjeNII-class.html#size,Variable Bio.Restriction.Restriction.CjeNIII.size=Bio.Restriction.Restriction.CjeNIII-class.html#size,Variable Bio.Restriction.Restriction.CjeP659IV.size=Bio.Restriction.Restriction.CjeP659IV-class.html#size,Variable Bio.Restriction.Restriction.CjePI.size=Bio.Restriction.Restriction.CjePI-class.html#size,Variable Bio.Restriction.Restriction.CjuI.size=Bio.Restriction.Restriction.CjuI-class.html#size,Variable Bio.Restriction.Restriction.CjuII.size=Bio.Restriction.Restriction.CjuII-class.html#size,Variable Bio.Restriction.Restriction.ClaI.size=Bio.Restriction.Restriction.ClaI-class.html#size,Variable Bio.Restriction.Restriction.CpoI.size=Bio.Restriction.Restriction.CpoI-class.html#size,Variable Bio.Restriction.Restriction.CseI.size=Bio.Restriction.Restriction.CseI-class.html#size,Variable Bio.Restriction.Restriction.CsiI.size=Bio.Restriction.Restriction.CsiI-class.html#size,Variable Bio.Restriction.Restriction.Csp6I.size=Bio.Restriction.Restriction.Csp6I-class.html#size,Variable Bio.Restriction.Restriction.CspAI.size=Bio.Restriction.Restriction.CspAI-class.html#size,Variable Bio.Restriction.Restriction.CspCI.size=Bio.Restriction.Restriction.CspCI-class.html#size,Variable Bio.Restriction.Restriction.CspI.size=Bio.Restriction.Restriction.CspI-class.html#size,Variable Bio.Restriction.Restriction.CstMI.size=Bio.Restriction.Restriction.CstMI-class.html#size,Variable Bio.Restriction.Restriction.CviAII.size=Bio.Restriction.Restriction.CviAII-class.html#size,Variable Bio.Restriction.Restriction.CviJI.size=Bio.Restriction.Restriction.CviJI-class.html#size,Variable Bio.Restriction.Restriction.CviKI_1.size=Bio.Restriction.Restriction.CviKI_1-class.html#size,Variable Bio.Restriction.Restriction.CviQI.size=Bio.Restriction.Restriction.CviQI-class.html#size,Variable Bio.Restriction.Restriction.CviRI.size=Bio.Restriction.Restriction.CviRI-class.html#size,Variable Bio.Restriction.Restriction.DdeI.size=Bio.Restriction.Restriction.DdeI-class.html#size,Variable Bio.Restriction.Restriction.DinI.size=Bio.Restriction.Restriction.DinI-class.html#size,Variable Bio.Restriction.Restriction.DpnI.size=Bio.Restriction.Restriction.DpnI-class.html#size,Variable Bio.Restriction.Restriction.DpnII.size=Bio.Restriction.Restriction.DpnII-class.html#size,Variable Bio.Restriction.Restriction.DraI.size=Bio.Restriction.Restriction.DraI-class.html#size,Variable Bio.Restriction.Restriction.DraII.size=Bio.Restriction.Restriction.DraII-class.html#size,Variable Bio.Restriction.Restriction.DraIII.size=Bio.Restriction.Restriction.DraIII-class.html#size,Variable Bio.Restriction.Restriction.DraRI.size=Bio.Restriction.Restriction.DraRI-class.html#size,Variable Bio.Restriction.Restriction.DrdI.size=Bio.Restriction.Restriction.DrdI-class.html#size,Variable Bio.Restriction.Restriction.DrdII.size=Bio.Restriction.Restriction.DrdII-class.html#size,Variable Bio.Restriction.Restriction.DriI.size=Bio.Restriction.Restriction.DriI-class.html#size,Variable Bio.Restriction.Restriction.DsaI.size=Bio.Restriction.Restriction.DsaI-class.html#size,Variable Bio.Restriction.Restriction.DseDI.size=Bio.Restriction.Restriction.DseDI-class.html#size,Variable Bio.Restriction.Restriction.EaeI.size=Bio.Restriction.Restriction.EaeI-class.html#size,Variable Bio.Restriction.Restriction.EagI.size=Bio.Restriction.Restriction.EagI-class.html#size,Variable Bio.Restriction.Restriction.Eam1104I.size=Bio.Restriction.Restriction.Eam1104I-class.html#size,Variable Bio.Restriction.Restriction.Eam1105I.size=Bio.Restriction.Restriction.Eam1105I-class.html#size,Variable Bio.Restriction.Restriction.EarI.size=Bio.Restriction.Restriction.EarI-class.html#size,Variable Bio.Restriction.Restriction.EciI.size=Bio.Restriction.Restriction.EciI-class.html#size,Variable Bio.Restriction.Restriction.Ecl136II.size=Bio.Restriction.Restriction.Ecl136II-class.html#size,Variable Bio.Restriction.Restriction.EclXI.size=Bio.Restriction.Restriction.EclXI-class.html#size,Variable Bio.Restriction.Restriction.Eco105I.size=Bio.Restriction.Restriction.Eco105I-class.html#size,Variable Bio.Restriction.Restriction.Eco130I.size=Bio.Restriction.Restriction.Eco130I-class.html#size,Variable Bio.Restriction.Restriction.Eco147I.size=Bio.Restriction.Restriction.Eco147I-class.html#size,Variable Bio.Restriction.Restriction.Eco24I.size=Bio.Restriction.Restriction.Eco24I-class.html#size,Variable Bio.Restriction.Restriction.Eco31I.size=Bio.Restriction.Restriction.Eco31I-class.html#size,Variable Bio.Restriction.Restriction.Eco32I.size=Bio.Restriction.Restriction.Eco32I-class.html#size,Variable Bio.Restriction.Restriction.Eco47I.size=Bio.Restriction.Restriction.Eco47I-class.html#size,Variable Bio.Restriction.Restriction.Eco47III.size=Bio.Restriction.Restriction.Eco47III-class.html#size,Variable Bio.Restriction.Restriction.Eco52I.size=Bio.Restriction.Restriction.Eco52I-class.html#size,Variable Bio.Restriction.Restriction.Eco53kI.size=Bio.Restriction.Restriction.Eco53kI-class.html#size,Variable Bio.Restriction.Restriction.Eco57I.size=Bio.Restriction.Restriction.Eco57I-class.html#size,Variable Bio.Restriction.Restriction.Eco57MI.size=Bio.Restriction.Restriction.Eco57MI-class.html#size,Variable Bio.Restriction.Restriction.Eco72I.size=Bio.Restriction.Restriction.Eco72I-class.html#size,Variable Bio.Restriction.Restriction.Eco81I.size=Bio.Restriction.Restriction.Eco81I-class.html#size,Variable Bio.Restriction.Restriction.Eco88I.size=Bio.Restriction.Restriction.Eco88I-class.html#size,Variable Bio.Restriction.Restriction.Eco91I.size=Bio.Restriction.Restriction.Eco91I-class.html#size,Variable Bio.Restriction.Restriction.EcoHI.size=Bio.Restriction.Restriction.EcoHI-class.html#size,Variable Bio.Restriction.Restriction.EcoICRI.size=Bio.Restriction.Restriction.EcoICRI-class.html#size,Variable Bio.Restriction.Restriction.EcoNI.size=Bio.Restriction.Restriction.EcoNI-class.html#size,Variable Bio.Restriction.Restriction.EcoO109I.size=Bio.Restriction.Restriction.EcoO109I-class.html#size,Variable Bio.Restriction.Restriction.EcoO65I.size=Bio.Restriction.Restriction.EcoO65I-class.html#size,Variable Bio.Restriction.Restriction.EcoRI.size=Bio.Restriction.Restriction.EcoRI-class.html#size,Variable Bio.Restriction.Restriction.EcoRII.size=Bio.Restriction.Restriction.EcoRII-class.html#size,Variable Bio.Restriction.Restriction.EcoRV.size=Bio.Restriction.Restriction.EcoRV-class.html#size,Variable Bio.Restriction.Restriction.EcoT14I.size=Bio.Restriction.Restriction.EcoT14I-class.html#size,Variable Bio.Restriction.Restriction.EcoT22I.size=Bio.Restriction.Restriction.EcoT22I-class.html#size,Variable Bio.Restriction.Restriction.EcoT38I.size=Bio.Restriction.Restriction.EcoT38I-class.html#size,Variable Bio.Restriction.Restriction.EgeI.size=Bio.Restriction.Restriction.EgeI-class.html#size,Variable Bio.Restriction.Restriction.EheI.size=Bio.Restriction.Restriction.EheI-class.html#size,Variable Bio.Restriction.Restriction.ErhI.size=Bio.Restriction.Restriction.ErhI-class.html#size,Variable Bio.Restriction.Restriction.EsaBC3I.size=Bio.Restriction.Restriction.EsaBC3I-class.html#size,Variable Bio.Restriction.Restriction.EsaSSI.size=Bio.Restriction.Restriction.EsaSSI-class.html#size,Variable Bio.Restriction.Restriction.Esp3I.size=Bio.Restriction.Restriction.Esp3I-class.html#size,Variable Bio.Restriction.Restriction.EspI.size=Bio.Restriction.Restriction.EspI-class.html#size,Variable Bio.Restriction.Restriction.FaeI.size=Bio.Restriction.Restriction.FaeI-class.html#size,Variable Bio.Restriction.Restriction.FaiI.size=Bio.Restriction.Restriction.FaiI-class.html#size,Variable Bio.Restriction.Restriction.FalI.size=Bio.Restriction.Restriction.FalI-class.html#size,Variable Bio.Restriction.Restriction.FaqI.size=Bio.Restriction.Restriction.FaqI-class.html#size,Variable Bio.Restriction.Restriction.FatI.size=Bio.Restriction.Restriction.FatI-class.html#size,Variable Bio.Restriction.Restriction.FauI.size=Bio.Restriction.Restriction.FauI-class.html#size,Variable Bio.Restriction.Restriction.FauNDI.size=Bio.Restriction.Restriction.FauNDI-class.html#size,Variable Bio.Restriction.Restriction.FbaI.size=Bio.Restriction.Restriction.FbaI-class.html#size,Variable Bio.Restriction.Restriction.FblI.size=Bio.Restriction.Restriction.FblI-class.html#size,Variable Bio.Restriction.Restriction.FinI.size=Bio.Restriction.Restriction.FinI-class.html#size,Variable Bio.Restriction.Restriction.FmuI.size=Bio.Restriction.Restriction.FmuI-class.html#size,Variable Bio.Restriction.Restriction.Fnu4HI.size=Bio.Restriction.Restriction.Fnu4HI-class.html#size,Variable Bio.Restriction.Restriction.FnuDII.size=Bio.Restriction.Restriction.FnuDII-class.html#size,Variable Bio.Restriction.Restriction.FokI.size=Bio.Restriction.Restriction.FokI-class.html#size,Variable Bio.Restriction.Restriction.FriOI.size=Bio.Restriction.Restriction.FriOI-class.html#size,Variable Bio.Restriction.Restriction.FseI.size=Bio.Restriction.Restriction.FseI-class.html#size,Variable Bio.Restriction.Restriction.Fsp4HI.size=Bio.Restriction.Restriction.Fsp4HI-class.html#size,Variable Bio.Restriction.Restriction.FspAI.size=Bio.Restriction.Restriction.FspAI-class.html#size,Variable Bio.Restriction.Restriction.FspBI.size=Bio.Restriction.Restriction.FspBI-class.html#size,Variable Bio.Restriction.Restriction.FspEI.size=Bio.Restriction.Restriction.FspEI-class.html#size,Variable Bio.Restriction.Restriction.FspI.size=Bio.Restriction.Restriction.FspI-class.html#size,Variable Bio.Restriction.Restriction.GauT27I.size=Bio.Restriction.Restriction.GauT27I-class.html#size,Variable Bio.Restriction.Restriction.GdiII.size=Bio.Restriction.Restriction.GdiII-class.html#size,Variable Bio.Restriction.Restriction.GlaI.size=Bio.Restriction.Restriction.GlaI-class.html#size,Variable Bio.Restriction.Restriction.GluI.size=Bio.Restriction.Restriction.GluI-class.html#size,Variable Bio.Restriction.Restriction.GsaI.size=Bio.Restriction.Restriction.GsaI-class.html#size,Variable Bio.Restriction.Restriction.GsuI.size=Bio.Restriction.Restriction.GsuI-class.html#size,Variable Bio.Restriction.Restriction.HaeI.size=Bio.Restriction.Restriction.HaeI-class.html#size,Variable Bio.Restriction.Restriction.HaeII.size=Bio.Restriction.Restriction.HaeII-class.html#size,Variable Bio.Restriction.Restriction.HaeIII.size=Bio.Restriction.Restriction.HaeIII-class.html#size,Variable Bio.Restriction.Restriction.HapII.size=Bio.Restriction.Restriction.HapII-class.html#size,Variable Bio.Restriction.Restriction.HauII.size=Bio.Restriction.Restriction.HauII-class.html#size,Variable Bio.Restriction.Restriction.HgaI.size=Bio.Restriction.Restriction.HgaI-class.html#size,Variable Bio.Restriction.Restriction.HgiAI.size=Bio.Restriction.Restriction.HgiAI-class.html#size,Variable Bio.Restriction.Restriction.HgiCI.size=Bio.Restriction.Restriction.HgiCI-class.html#size,Variable Bio.Restriction.Restriction.HgiEII.size=Bio.Restriction.Restriction.HgiEII-class.html#size,Variable Bio.Restriction.Restriction.HgiJII.size=Bio.Restriction.Restriction.HgiJII-class.html#size,Variable Bio.Restriction.Restriction.HhaI.size=Bio.Restriction.Restriction.HhaI-class.html#size,Variable Bio.Restriction.Restriction.Hin1I.size=Bio.Restriction.Restriction.Hin1I-class.html#size,Variable Bio.Restriction.Restriction.Hin1II.size=Bio.Restriction.Restriction.Hin1II-class.html#size,Variable Bio.Restriction.Restriction.Hin4I.size=Bio.Restriction.Restriction.Hin4I-class.html#size,Variable Bio.Restriction.Restriction.Hin4II.size=Bio.Restriction.Restriction.Hin4II-class.html#size,Variable Bio.Restriction.Restriction.Hin6I.size=Bio.Restriction.Restriction.Hin6I-class.html#size,Variable Bio.Restriction.Restriction.HinP1I.size=Bio.Restriction.Restriction.HinP1I-class.html#size,Variable Bio.Restriction.Restriction.HincII.size=Bio.Restriction.Restriction.HincII-class.html#size,Variable Bio.Restriction.Restriction.HindII.size=Bio.Restriction.Restriction.HindII-class.html#size,Variable Bio.Restriction.Restriction.HindIII.size=Bio.Restriction.Restriction.HindIII-class.html#size,Variable Bio.Restriction.Restriction.HinfI.size=Bio.Restriction.Restriction.HinfI-class.html#size,Variable Bio.Restriction.Restriction.HpaI.size=Bio.Restriction.Restriction.HpaI-class.html#size,Variable Bio.Restriction.Restriction.HpaII.size=Bio.Restriction.Restriction.HpaII-class.html#size,Variable Bio.Restriction.Restriction.HphI.size=Bio.Restriction.Restriction.HphI-class.html#size,Variable Bio.Restriction.Restriction.Hpy166II.size=Bio.Restriction.Restriction.Hpy166II-class.html#size,Variable Bio.Restriction.Restriction.Hpy178III.size=Bio.Restriction.Restriction.Hpy178III-class.html#size,Variable Bio.Restriction.Restriction.Hpy188I.size=Bio.Restriction.Restriction.Hpy188I-class.html#size,Variable Bio.Restriction.Restriction.Hpy188III.size=Bio.Restriction.Restriction.Hpy188III-class.html#size,Variable Bio.Restriction.Restriction.Hpy8I.size=Bio.Restriction.Restriction.Hpy8I-class.html#size,Variable Bio.Restriction.Restriction.Hpy99I.size=Bio.Restriction.Restriction.Hpy99I-class.html#size,Variable Bio.Restriction.Restriction.Hpy99XIII.size=Bio.Restriction.Restriction.Hpy99XIII-class.html#size,Variable Bio.Restriction.Restriction.Hpy99XIV.size=Bio.Restriction.Restriction.Hpy99XIV-class.html#size,Variable Bio.Restriction.Restriction.HpyAV.size=Bio.Restriction.Restriction.HpyAV-class.html#size,Variable Bio.Restriction.Restriction.HpyCH4III.size=Bio.Restriction.Restriction.HpyCH4III-class.html#size,Variable Bio.Restriction.Restriction.HpyCH4IV.size=Bio.Restriction.Restriction.HpyCH4IV-class.html#size,Variable Bio.Restriction.Restriction.HpyCH4V.size=Bio.Restriction.Restriction.HpyCH4V-class.html#size,Variable Bio.Restriction.Restriction.HpyF10VI.size=Bio.Restriction.Restriction.HpyF10VI-class.html#size,Variable Bio.Restriction.Restriction.HpyF3I.size=Bio.Restriction.Restriction.HpyF3I-class.html#size,Variable Bio.Restriction.Restriction.HpySE526I.size=Bio.Restriction.Restriction.HpySE526I-class.html#size,Variable Bio.Restriction.Restriction.Hsp92I.size=Bio.Restriction.Restriction.Hsp92I-class.html#size,Variable Bio.Restriction.Restriction.Hsp92II.size=Bio.Restriction.Restriction.Hsp92II-class.html#size,Variable Bio.Restriction.Restriction.HspAI.size=Bio.Restriction.Restriction.HspAI-class.html#size,Variable Bio.Restriction.Restriction.Jma19592I.size=Bio.Restriction.Restriction.Jma19592I-class.html#size,Variable Bio.Restriction.Restriction.KasI.size=Bio.Restriction.Restriction.KasI-class.html#size,Variable Bio.Restriction.Restriction.KflI.size=Bio.Restriction.Restriction.KflI-class.html#size,Variable Bio.Restriction.Restriction.Kpn2I.size=Bio.Restriction.Restriction.Kpn2I-class.html#size,Variable Bio.Restriction.Restriction.KpnI.size=Bio.Restriction.Restriction.KpnI-class.html#size,Variable Bio.Restriction.Restriction.KroI.size=Bio.Restriction.Restriction.KroI-class.html#size,Variable Bio.Restriction.Restriction.Ksp22I.size=Bio.Restriction.Restriction.Ksp22I-class.html#size,Variable Bio.Restriction.Restriction.Ksp632I.size=Bio.Restriction.Restriction.Ksp632I-class.html#size,Variable Bio.Restriction.Restriction.KspAI.size=Bio.Restriction.Restriction.KspAI-class.html#size,Variable Bio.Restriction.Restriction.KspI.size=Bio.Restriction.Restriction.KspI-class.html#size,Variable Bio.Restriction.Restriction.Kzo9I.size=Bio.Restriction.Restriction.Kzo9I-class.html#size,Variable Bio.Restriction.Restriction.LguI.size=Bio.Restriction.Restriction.LguI-class.html#size,Variable Bio.Restriction.Restriction.LpnI.size=Bio.Restriction.Restriction.LpnI-class.html#size,Variable Bio.Restriction.Restriction.LpnPI.size=Bio.Restriction.Restriction.LpnPI-class.html#size,Variable Bio.Restriction.Restriction.Lsp1109I.size=Bio.Restriction.Restriction.Lsp1109I-class.html#size,Variable Bio.Restriction.Restriction.LweI.size=Bio.Restriction.Restriction.LweI-class.html#size,Variable Bio.Restriction.Restriction.MabI.size=Bio.Restriction.Restriction.MabI-class.html#size,Variable Bio.Restriction.Restriction.MaeI.size=Bio.Restriction.Restriction.MaeI-class.html#size,Variable Bio.Restriction.Restriction.MaeII.size=Bio.Restriction.Restriction.MaeII-class.html#size,Variable Bio.Restriction.Restriction.MaeIII.size=Bio.Restriction.Restriction.MaeIII-class.html#size,Variable Bio.Restriction.Restriction.MalI.size=Bio.Restriction.Restriction.MalI-class.html#size,Variable Bio.Restriction.Restriction.MaqI.size=Bio.Restriction.Restriction.MaqI-class.html#size,Variable Bio.Restriction.Restriction.MauBI.size=Bio.Restriction.Restriction.MauBI-class.html#size,Variable Bio.Restriction.Restriction.MbiI.size=Bio.Restriction.Restriction.MbiI-class.html#size,Variable Bio.Restriction.Restriction.MboI.size=Bio.Restriction.Restriction.MboI-class.html#size,Variable Bio.Restriction.Restriction.MboII.size=Bio.Restriction.Restriction.MboII-class.html#size,Variable Bio.Restriction.Restriction.McaTI.size=Bio.Restriction.Restriction.McaTI-class.html#size,Variable Bio.Restriction.Restriction.McrI.size=Bio.Restriction.Restriction.McrI-class.html#size,Variable Bio.Restriction.Restriction.MfeI.size=Bio.Restriction.Restriction.MfeI-class.html#size,Variable Bio.Restriction.Restriction.MflI.size=Bio.Restriction.Restriction.MflI-class.html#size,Variable Bio.Restriction.Restriction.MhlI.size=Bio.Restriction.Restriction.MhlI-class.html#size,Variable Bio.Restriction.Restriction.MjaIV.size=Bio.Restriction.Restriction.MjaIV-class.html#size,Variable Bio.Restriction.Restriction.MkaDII.size=Bio.Restriction.Restriction.MkaDII-class.html#size,Variable Bio.Restriction.Restriction.MlsI.size=Bio.Restriction.Restriction.MlsI-class.html#size,Variable Bio.Restriction.Restriction.MluCI.size=Bio.Restriction.Restriction.MluCI-class.html#size,Variable Bio.Restriction.Restriction.MluI.size=Bio.Restriction.Restriction.MluI-class.html#size,Variable Bio.Restriction.Restriction.MluNI.size=Bio.Restriction.Restriction.MluNI-class.html#size,Variable Bio.Restriction.Restriction.Mly113I.size=Bio.Restriction.Restriction.Mly113I-class.html#size,Variable Bio.Restriction.Restriction.MlyI.size=Bio.Restriction.Restriction.MlyI-class.html#size,Variable Bio.Restriction.Restriction.MmeI.size=Bio.Restriction.Restriction.MmeI-class.html#size,Variable Bio.Restriction.Restriction.MnlI.size=Bio.Restriction.Restriction.MnlI-class.html#size,Variable Bio.Restriction.Restriction.Mph1103I.size=Bio.Restriction.Restriction.Mph1103I-class.html#size,Variable Bio.Restriction.Restriction.MreI.size=Bio.Restriction.Restriction.MreI-class.html#size,Variable Bio.Restriction.Restriction.MroI.size=Bio.Restriction.Restriction.MroI-class.html#size,Variable Bio.Restriction.Restriction.MroNI.size=Bio.Restriction.Restriction.MroNI-class.html#size,Variable Bio.Restriction.Restriction.MroXI.size=Bio.Restriction.Restriction.MroXI-class.html#size,Variable Bio.Restriction.Restriction.MscI.size=Bio.Restriction.Restriction.MscI-class.html#size,Variable Bio.Restriction.Restriction.MseI.size=Bio.Restriction.Restriction.MseI-class.html#size,Variable Bio.Restriction.Restriction.MslI.size=Bio.Restriction.Restriction.MslI-class.html#size,Variable Bio.Restriction.Restriction.Msp20I.size=Bio.Restriction.Restriction.Msp20I-class.html#size,Variable Bio.Restriction.Restriction.MspA1I.size=Bio.Restriction.Restriction.MspA1I-class.html#size,Variable Bio.Restriction.Restriction.MspCI.size=Bio.Restriction.Restriction.MspCI-class.html#size,Variable Bio.Restriction.Restriction.MspI.size=Bio.Restriction.Restriction.MspI-class.html#size,Variable Bio.Restriction.Restriction.MspJI.size=Bio.Restriction.Restriction.MspJI-class.html#size,Variable Bio.Restriction.Restriction.MspR9I.size=Bio.Restriction.Restriction.MspR9I-class.html#size,Variable Bio.Restriction.Restriction.MssI.size=Bio.Restriction.Restriction.MssI-class.html#size,Variable Bio.Restriction.Restriction.MstI.size=Bio.Restriction.Restriction.MstI-class.html#size,Variable Bio.Restriction.Restriction.MunI.size=Bio.Restriction.Restriction.MunI-class.html#size,Variable Bio.Restriction.Restriction.Mva1269I.size=Bio.Restriction.Restriction.Mva1269I-class.html#size,Variable Bio.Restriction.Restriction.MvaI.size=Bio.Restriction.Restriction.MvaI-class.html#size,Variable Bio.Restriction.Restriction.MvnI.size=Bio.Restriction.Restriction.MvnI-class.html#size,Variable Bio.Restriction.Restriction.MvrI.size=Bio.Restriction.Restriction.MvrI-class.html#size,Variable Bio.Restriction.Restriction.MwoI.size=Bio.Restriction.Restriction.MwoI-class.html#size,Variable Bio.Restriction.Restriction.NaeI.size=Bio.Restriction.Restriction.NaeI-class.html#size,Variable Bio.Restriction.Restriction.NarI.size=Bio.Restriction.Restriction.NarI-class.html#size,Variable Bio.Restriction.Restriction.NciI.size=Bio.Restriction.Restriction.NciI-class.html#size,Variable Bio.Restriction.Restriction.NcoI.size=Bio.Restriction.Restriction.NcoI-class.html#size,Variable Bio.Restriction.Restriction.NdeI.size=Bio.Restriction.Restriction.NdeI-class.html#size,Variable Bio.Restriction.Restriction.NdeII.size=Bio.Restriction.Restriction.NdeII-class.html#size,Variable Bio.Restriction.Restriction.NgoAVIII.size=Bio.Restriction.Restriction.NgoAVIII-class.html#size,Variable Bio.Restriction.Restriction.NgoMIV.size=Bio.Restriction.Restriction.NgoMIV-class.html#size,Variable Bio.Restriction.Restriction.NhaXI.size=Bio.Restriction.Restriction.NhaXI-class.html#size,Variable Bio.Restriction.Restriction.NheI.size=Bio.Restriction.Restriction.NheI-class.html#size,Variable Bio.Restriction.Restriction.NlaCI.size=Bio.Restriction.Restriction.NlaCI-class.html#size,Variable Bio.Restriction.Restriction.NlaIII.size=Bio.Restriction.Restriction.NlaIII-class.html#size,Variable Bio.Restriction.Restriction.NlaIV.size=Bio.Restriction.Restriction.NlaIV-class.html#size,Variable Bio.Restriction.Restriction.Nli3877I.size=Bio.Restriction.Restriction.Nli3877I-class.html#size,Variable Bio.Restriction.Restriction.NmeAIII.size=Bio.Restriction.Restriction.NmeAIII-class.html#size,Variable Bio.Restriction.Restriction.NmeDI.size=Bio.Restriction.Restriction.NmeDI-class.html#size,Variable Bio.Restriction.Restriction.NmuCI.size=Bio.Restriction.Restriction.NmuCI-class.html#size,Variable Bio.Restriction.Restriction.NotI.size=Bio.Restriction.Restriction.NotI-class.html#size,Variable Bio.Restriction.Restriction.NruI.size=Bio.Restriction.Restriction.NruI-class.html#size,Variable Bio.Restriction.Restriction.NsbI.size=Bio.Restriction.Restriction.NsbI-class.html#size,Variable Bio.Restriction.Restriction.NsiI.size=Bio.Restriction.Restriction.NsiI-class.html#size,Variable Bio.Restriction.Restriction.NspBII.size=Bio.Restriction.Restriction.NspBII-class.html#size,Variable Bio.Restriction.Restriction.NspI.size=Bio.Restriction.Restriction.NspI-class.html#size,Variable Bio.Restriction.Restriction.NspV.size=Bio.Restriction.Restriction.NspV-class.html#size,Variable Bio.Restriction.Restriction.OliI.size=Bio.Restriction.Restriction.OliI-class.html#size,Variable Bio.Restriction.Restriction.PabI.size=Bio.Restriction.Restriction.PabI-class.html#size,Variable Bio.Restriction.Restriction.PacI.size=Bio.Restriction.Restriction.PacI-class.html#size,Variable Bio.Restriction.Restriction.PaeI.size=Bio.Restriction.Restriction.PaeI-class.html#size,Variable Bio.Restriction.Restriction.PaeR7I.size=Bio.Restriction.Restriction.PaeR7I-class.html#size,Variable Bio.Restriction.Restriction.PagI.size=Bio.Restriction.Restriction.PagI-class.html#size,Variable Bio.Restriction.Restriction.PalAI.size=Bio.Restriction.Restriction.PalAI-class.html#size,Variable Bio.Restriction.Restriction.PasI.size=Bio.Restriction.Restriction.PasI-class.html#size,Variable Bio.Restriction.Restriction.PauI.size=Bio.Restriction.Restriction.PauI-class.html#size,Variable Bio.Restriction.Restriction.PceI.size=Bio.Restriction.Restriction.PceI-class.html#size,Variable Bio.Restriction.Restriction.PciI.size=Bio.Restriction.Restriction.PciI-class.html#size,Variable Bio.Restriction.Restriction.PciSI.size=Bio.Restriction.Restriction.PciSI-class.html#size,Variable Bio.Restriction.Restriction.PcsI.size=Bio.Restriction.Restriction.PcsI-class.html#size,Variable Bio.Restriction.Restriction.PctI.size=Bio.Restriction.Restriction.PctI-class.html#size,Variable Bio.Restriction.Restriction.PdiI.size=Bio.Restriction.Restriction.PdiI-class.html#size,Variable Bio.Restriction.Restriction.PdmI.size=Bio.Restriction.Restriction.PdmI-class.html#size,Variable Bio.Restriction.Restriction.PenI.size=Bio.Restriction.Restriction.PenI-class.html#size,Variable Bio.Restriction.Restriction.PfeI.size=Bio.Restriction.Restriction.PfeI-class.html#size,Variable Bio.Restriction.Restriction.Pfl1108I.size=Bio.Restriction.Restriction.Pfl1108I-class.html#size,Variable Bio.Restriction.Restriction.Pfl23II.size=Bio.Restriction.Restriction.Pfl23II-class.html#size,Variable Bio.Restriction.Restriction.PflFI.size=Bio.Restriction.Restriction.PflFI-class.html#size,Variable Bio.Restriction.Restriction.PflMI.size=Bio.Restriction.Restriction.PflMI-class.html#size,Variable Bio.Restriction.Restriction.PfoI.size=Bio.Restriction.Restriction.PfoI-class.html#size,Variable Bio.Restriction.Restriction.PinAI.size=Bio.Restriction.Restriction.PinAI-class.html#size,Variable Bio.Restriction.Restriction.PlaDI.size=Bio.Restriction.Restriction.PlaDI-class.html#size,Variable Bio.Restriction.Restriction.Ple19I.size=Bio.Restriction.Restriction.Ple19I-class.html#size,Variable Bio.Restriction.Restriction.PleI.size=Bio.Restriction.Restriction.PleI-class.html#size,Variable Bio.Restriction.Restriction.PluTI.size=Bio.Restriction.Restriction.PluTI-class.html#size,Variable Bio.Restriction.Restriction.PmaCI.size=Bio.Restriction.Restriction.PmaCI-class.html#size,Variable Bio.Restriction.Restriction.PmeI.size=Bio.Restriction.Restriction.PmeI-class.html#size,Variable Bio.Restriction.Restriction.PmlI.size=Bio.Restriction.Restriction.PmlI-class.html#size,Variable Bio.Restriction.Restriction.PpiI.size=Bio.Restriction.Restriction.PpiI-class.html#size,Variable Bio.Restriction.Restriction.PpsI.size=Bio.Restriction.Restriction.PpsI-class.html#size,Variable Bio.Restriction.Restriction.Ppu10I.size=Bio.Restriction.Restriction.Ppu10I-class.html#size,Variable Bio.Restriction.Restriction.Ppu21I.size=Bio.Restriction.Restriction.Ppu21I-class.html#size,Variable Bio.Restriction.Restriction.PpuMI.size=Bio.Restriction.Restriction.PpuMI-class.html#size,Variable Bio.Restriction.Restriction.PscI.size=Bio.Restriction.Restriction.PscI-class.html#size,Variable Bio.Restriction.Restriction.PshAI.size=Bio.Restriction.Restriction.PshAI-class.html#size,Variable Bio.Restriction.Restriction.PshBI.size=Bio.Restriction.Restriction.PshBI-class.html#size,Variable Bio.Restriction.Restriction.PsiI.size=Bio.Restriction.Restriction.PsiI-class.html#size,Variable Bio.Restriction.Restriction.Psp03I.size=Bio.Restriction.Restriction.Psp03I-class.html#size,Variable Bio.Restriction.Restriction.Psp124BI.size=Bio.Restriction.Restriction.Psp124BI-class.html#size,Variable Bio.Restriction.Restriction.Psp1406I.size=Bio.Restriction.Restriction.Psp1406I-class.html#size,Variable Bio.Restriction.Restriction.Psp5II.size=Bio.Restriction.Restriction.Psp5II-class.html#size,Variable Bio.Restriction.Restriction.Psp6I.size=Bio.Restriction.Restriction.Psp6I-class.html#size,Variable Bio.Restriction.Restriction.PspCI.size=Bio.Restriction.Restriction.PspCI-class.html#size,Variable Bio.Restriction.Restriction.PspEI.size=Bio.Restriction.Restriction.PspEI-class.html#size,Variable Bio.Restriction.Restriction.PspGI.size=Bio.Restriction.Restriction.PspGI-class.html#size,Variable Bio.Restriction.Restriction.PspLI.size=Bio.Restriction.Restriction.PspLI-class.html#size,Variable Bio.Restriction.Restriction.PspN4I.size=Bio.Restriction.Restriction.PspN4I-class.html#size,Variable Bio.Restriction.Restriction.PspOMI.size=Bio.Restriction.Restriction.PspOMI-class.html#size,Variable Bio.Restriction.Restriction.PspOMII.size=Bio.Restriction.Restriction.PspOMII-class.html#size,Variable Bio.Restriction.Restriction.PspPI.size=Bio.Restriction.Restriction.PspPI-class.html#size,Variable Bio.Restriction.Restriction.PspPPI.size=Bio.Restriction.Restriction.PspPPI-class.html#size,Variable Bio.Restriction.Restriction.PspPRI.size=Bio.Restriction.Restriction.PspPRI-class.html#size,Variable Bio.Restriction.Restriction.PspXI.size=Bio.Restriction.Restriction.PspXI-class.html#size,Variable Bio.Restriction.Restriction.PsrI.size=Bio.Restriction.Restriction.PsrI-class.html#size,Variable Bio.Restriction.Restriction.PssI.size=Bio.Restriction.Restriction.PssI-class.html#size,Variable Bio.Restriction.Restriction.PstI.size=Bio.Restriction.Restriction.PstI-class.html#size,Variable Bio.Restriction.Restriction.PstNI.size=Bio.Restriction.Restriction.PstNI-class.html#size,Variable Bio.Restriction.Restriction.PsuI.size=Bio.Restriction.Restriction.PsuI-class.html#size,Variable Bio.Restriction.Restriction.PsyI.size=Bio.Restriction.Restriction.PsyI-class.html#size,Variable Bio.Restriction.Restriction.PteI.size=Bio.Restriction.Restriction.PteI-class.html#size,Variable Bio.Restriction.Restriction.PvuI.size=Bio.Restriction.Restriction.PvuI-class.html#size,Variable Bio.Restriction.Restriction.PvuII.size=Bio.Restriction.Restriction.PvuII-class.html#size,Variable Bio.Restriction.Restriction.R2_BceSIV.size=Bio.Restriction.Restriction.R2_BceSIV-class.html#size,Variable Bio.Restriction.Restriction.RceI.size=Bio.Restriction.Restriction.RceI-class.html#size,Variable Bio.Restriction.Restriction.RdeGBI.size=Bio.Restriction.Restriction.RdeGBI-class.html#size,Variable Bio.Restriction.Restriction.RdeGBII.size=Bio.Restriction.Restriction.RdeGBII-class.html#size,Variable Bio.Restriction.Restriction.RdeGBIII.size=Bio.Restriction.Restriction.RdeGBIII-class.html#size,Variable Bio.Restriction.Restriction.RflFIII.size=Bio.Restriction.Restriction.RflFIII-class.html#size,Variable Bio.Restriction.Restriction.RgaI.size=Bio.Restriction.Restriction.RgaI-class.html#size,Variable Bio.Restriction.Restriction.RigI.size=Bio.Restriction.Restriction.RigI-class.html#size,Variable Bio.Restriction.Restriction.RlaI.size=Bio.Restriction.Restriction.RlaI-class.html#size,Variable Bio.Restriction.Restriction.RleAI.size=Bio.Restriction.Restriction.RleAI-class.html#size,Variable Bio.Restriction.Restriction.RpaB5I.size=Bio.Restriction.Restriction.RpaB5I-class.html#size,Variable Bio.Restriction.Restriction.RpaBI.size=Bio.Restriction.Restriction.RpaBI-class.html#size,Variable Bio.Restriction.Restriction.RpaI.size=Bio.Restriction.Restriction.RpaI-class.html#size,Variable Bio.Restriction.Restriction.RpaTI.size=Bio.Restriction.Restriction.RpaTI-class.html#size,Variable Bio.Restriction.Restriction.RruI.size=Bio.Restriction.Restriction.RruI-class.html#size,Variable Bio.Restriction.Restriction.RsaI.size=Bio.Restriction.Restriction.RsaI-class.html#size,Variable Bio.Restriction.Restriction.RsaNI.size=Bio.Restriction.Restriction.RsaNI-class.html#size,Variable Bio.Restriction.Restriction.RseI.size=Bio.Restriction.Restriction.RseI-class.html#size,Variable Bio.Restriction.Restriction.Rsr2I.size=Bio.Restriction.Restriction.Rsr2I-class.html#size,Variable Bio.Restriction.Restriction.RsrII.size=Bio.Restriction.Restriction.RsrII-class.html#size,Variable Bio.Restriction.Restriction.SacI.size=Bio.Restriction.Restriction.SacI-class.html#size,Variable Bio.Restriction.Restriction.SacII.size=Bio.Restriction.Restriction.SacII-class.html#size,Variable Bio.Restriction.Restriction.SalI.size=Bio.Restriction.Restriction.SalI-class.html#size,Variable Bio.Restriction.Restriction.SanDI.size=Bio.Restriction.Restriction.SanDI-class.html#size,Variable Bio.Restriction.Restriction.SapI.size=Bio.Restriction.Restriction.SapI-class.html#size,Variable Bio.Restriction.Restriction.SaqAI.size=Bio.Restriction.Restriction.SaqAI-class.html#size,Variable Bio.Restriction.Restriction.SatI.size=Bio.Restriction.Restriction.SatI-class.html#size,Variable Bio.Restriction.Restriction.Sau3AI.size=Bio.Restriction.Restriction.Sau3AI-class.html#size,Variable Bio.Restriction.Restriction.Sau96I.size=Bio.Restriction.Restriction.Sau96I-class.html#size,Variable Bio.Restriction.Restriction.SauI.size=Bio.Restriction.Restriction.SauI-class.html#size,Variable Bio.Restriction.Restriction.SbfI.size=Bio.Restriction.Restriction.SbfI-class.html#size,Variable Bio.Restriction.Restriction.ScaI.size=Bio.Restriction.Restriction.ScaI-class.html#size,Variable Bio.Restriction.Restriction.SchI.size=Bio.Restriction.Restriction.SchI-class.html#size,Variable Bio.Restriction.Restriction.SciI.size=Bio.Restriction.Restriction.SciI-class.html#size,Variable Bio.Restriction.Restriction.ScrFI.size=Bio.Restriction.Restriction.ScrFI-class.html#size,Variable Bio.Restriction.Restriction.SdaI.size=Bio.Restriction.Restriction.SdaI-class.html#size,Variable Bio.Restriction.Restriction.SdeAI.size=Bio.Restriction.Restriction.SdeAI-class.html#size,Variable Bio.Restriction.Restriction.SdeOSI.size=Bio.Restriction.Restriction.SdeOSI-class.html#size,Variable Bio.Restriction.Restriction.SduI.size=Bio.Restriction.Restriction.SduI-class.html#size,Variable Bio.Restriction.Restriction.SecI.size=Bio.Restriction.Restriction.SecI-class.html#size,Variable Bio.Restriction.Restriction.SelI.size=Bio.Restriction.Restriction.SelI-class.html#size,Variable Bio.Restriction.Restriction.SetI.size=Bio.Restriction.Restriction.SetI-class.html#size,Variable Bio.Restriction.Restriction.SexAI.size=Bio.Restriction.Restriction.SexAI-class.html#size,Variable Bio.Restriction.Restriction.SfaAI.size=Bio.Restriction.Restriction.SfaAI-class.html#size,Variable Bio.Restriction.Restriction.SfaNI.size=Bio.Restriction.Restriction.SfaNI-class.html#size,Variable Bio.Restriction.Restriction.SfcI.size=Bio.Restriction.Restriction.SfcI-class.html#size,Variable Bio.Restriction.Restriction.SfeI.size=Bio.Restriction.Restriction.SfeI-class.html#size,Variable Bio.Restriction.Restriction.SfiI.size=Bio.Restriction.Restriction.SfiI-class.html#size,Variable Bio.Restriction.Restriction.SfoI.size=Bio.Restriction.Restriction.SfoI-class.html#size,Variable Bio.Restriction.Restriction.Sfr274I.size=Bio.Restriction.Restriction.Sfr274I-class.html#size,Variable Bio.Restriction.Restriction.Sfr303I.size=Bio.Restriction.Restriction.Sfr303I-class.html#size,Variable Bio.Restriction.Restriction.SfuI.size=Bio.Restriction.Restriction.SfuI-class.html#size,Variable Bio.Restriction.Restriction.SgeI.size=Bio.Restriction.Restriction.SgeI-class.html#size,Variable Bio.Restriction.Restriction.SgfI.size=Bio.Restriction.Restriction.SgfI-class.html#size,Variable Bio.Restriction.Restriction.SgrAI.size=Bio.Restriction.Restriction.SgrAI-class.html#size,Variable Bio.Restriction.Restriction.SgrBI.size=Bio.Restriction.Restriction.SgrBI-class.html#size,Variable Bio.Restriction.Restriction.SgrDI.size=Bio.Restriction.Restriction.SgrDI-class.html#size,Variable Bio.Restriction.Restriction.SgrTI.size=Bio.Restriction.Restriction.SgrTI-class.html#size,Variable Bio.Restriction.Restriction.SgsI.size=Bio.Restriction.Restriction.SgsI-class.html#size,Variable Bio.Restriction.Restriction.SimI.size=Bio.Restriction.Restriction.SimI-class.html#size,Variable Bio.Restriction.Restriction.SlaI.size=Bio.Restriction.Restriction.SlaI-class.html#size,Variable Bio.Restriction.Restriction.SmaI.size=Bio.Restriction.Restriction.SmaI-class.html#size,Variable Bio.Restriction.Restriction.SmiI.size=Bio.Restriction.Restriction.SmiI-class.html#size,Variable Bio.Restriction.Restriction.SmiMI.size=Bio.Restriction.Restriction.SmiMI-class.html#size,Variable Bio.Restriction.Restriction.SmlI.size=Bio.Restriction.Restriction.SmlI-class.html#size,Variable Bio.Restriction.Restriction.SmoI.size=Bio.Restriction.Restriction.SmoI-class.html#size,Variable Bio.Restriction.Restriction.SnaBI.size=Bio.Restriction.Restriction.SnaBI-class.html#size,Variable Bio.Restriction.Restriction.SnaI.size=Bio.Restriction.Restriction.SnaI-class.html#size,Variable Bio.Restriction.Restriction.Sno506I.size=Bio.Restriction.Restriction.Sno506I-class.html#size,Variable Bio.Restriction.Restriction.SpeI.size=Bio.Restriction.Restriction.SpeI-class.html#size,Variable Bio.Restriction.Restriction.SphI.size=Bio.Restriction.Restriction.SphI-class.html#size,Variable Bio.Restriction.Restriction.SplI.size=Bio.Restriction.Restriction.SplI-class.html#size,Variable Bio.Restriction.Restriction.SpoDI.size=Bio.Restriction.Restriction.SpoDI-class.html#size,Variable Bio.Restriction.Restriction.SrfI.size=Bio.Restriction.Restriction.SrfI-class.html#size,Variable Bio.Restriction.Restriction.Sse232I.size=Bio.Restriction.Restriction.Sse232I-class.html#size,Variable Bio.Restriction.Restriction.Sse8387I.size=Bio.Restriction.Restriction.Sse8387I-class.html#size,Variable Bio.Restriction.Restriction.Sse8647I.size=Bio.Restriction.Restriction.Sse8647I-class.html#size,Variable Bio.Restriction.Restriction.Sse9I.size=Bio.Restriction.Restriction.Sse9I-class.html#size,Variable Bio.Restriction.Restriction.SseBI.size=Bio.Restriction.Restriction.SseBI-class.html#size,Variable Bio.Restriction.Restriction.SsiI.size=Bio.Restriction.Restriction.SsiI-class.html#size,Variable Bio.Restriction.Restriction.SspD5I.size=Bio.Restriction.Restriction.SspD5I-class.html#size,Variable Bio.Restriction.Restriction.SspDI.size=Bio.Restriction.Restriction.SspDI-class.html#size,Variable Bio.Restriction.Restriction.SspI.size=Bio.Restriction.Restriction.SspI-class.html#size,Variable Bio.Restriction.Restriction.SstE37I.size=Bio.Restriction.Restriction.SstE37I-class.html#size,Variable Bio.Restriction.Restriction.SstI.size=Bio.Restriction.Restriction.SstI-class.html#size,Variable Bio.Restriction.Restriction.Sth132I.size=Bio.Restriction.Restriction.Sth132I-class.html#size,Variable Bio.Restriction.Restriction.Sth302II.size=Bio.Restriction.Restriction.Sth302II-class.html#size,Variable Bio.Restriction.Restriction.StrI.size=Bio.Restriction.Restriction.StrI-class.html#size,Variable Bio.Restriction.Restriction.StsI.size=Bio.Restriction.Restriction.StsI-class.html#size,Variable Bio.Restriction.Restriction.StuI.size=Bio.Restriction.Restriction.StuI-class.html#size,Variable Bio.Restriction.Restriction.StyD4I.size=Bio.Restriction.Restriction.StyD4I-class.html#size,Variable Bio.Restriction.Restriction.StyI.size=Bio.Restriction.Restriction.StyI-class.html#size,Variable Bio.Restriction.Restriction.SwaI.size=Bio.Restriction.Restriction.SwaI-class.html#size,Variable Bio.Restriction.Restriction.TaaI.size=Bio.Restriction.Restriction.TaaI-class.html#size,Variable Bio.Restriction.Restriction.TaiI.size=Bio.Restriction.Restriction.TaiI-class.html#size,Variable Bio.Restriction.Restriction.TaqI.size=Bio.Restriction.Restriction.TaqI-class.html#size,Variable Bio.Restriction.Restriction.TaqII.size=Bio.Restriction.Restriction.TaqII-class.html#size,Variable Bio.Restriction.Restriction.TasI.size=Bio.Restriction.Restriction.TasI-class.html#size,Variable Bio.Restriction.Restriction.TatI.size=Bio.Restriction.Restriction.TatI-class.html#size,Variable Bio.Restriction.Restriction.TauI.size=Bio.Restriction.Restriction.TauI-class.html#size,Variable Bio.Restriction.Restriction.TfiI.size=Bio.Restriction.Restriction.TfiI-class.html#size,Variable Bio.Restriction.Restriction.Tru1I.size=Bio.Restriction.Restriction.Tru1I-class.html#size,Variable Bio.Restriction.Restriction.Tru9I.size=Bio.Restriction.Restriction.Tru9I-class.html#size,Variable Bio.Restriction.Restriction.TscAI.size=Bio.Restriction.Restriction.TscAI-class.html#size,Variable Bio.Restriction.Restriction.TseFI.size=Bio.Restriction.Restriction.TseFI-class.html#size,Variable Bio.Restriction.Restriction.TseI.size=Bio.Restriction.Restriction.TseI-class.html#size,Variable Bio.Restriction.Restriction.TsoI.size=Bio.Restriction.Restriction.TsoI-class.html#size,Variable Bio.Restriction.Restriction.Tsp45I.size=Bio.Restriction.Restriction.Tsp45I-class.html#size,Variable Bio.Restriction.Restriction.Tsp4CI.size=Bio.Restriction.Restriction.Tsp4CI-class.html#size,Variable Bio.Restriction.Restriction.TspDTI.size=Bio.Restriction.Restriction.TspDTI-class.html#size,Variable Bio.Restriction.Restriction.TspEI.size=Bio.Restriction.Restriction.TspEI-class.html#size,Variable Bio.Restriction.Restriction.TspGWI.size=Bio.Restriction.Restriction.TspGWI-class.html#size,Variable Bio.Restriction.Restriction.TspMI.size=Bio.Restriction.Restriction.TspMI-class.html#size,Variable Bio.Restriction.Restriction.TspRI.size=Bio.Restriction.Restriction.TspRI-class.html#size,Variable Bio.Restriction.Restriction.TssI.size=Bio.Restriction.Restriction.TssI-class.html#size,Variable Bio.Restriction.Restriction.TstI.size=Bio.Restriction.Restriction.TstI-class.html#size,Variable Bio.Restriction.Restriction.TsuI.size=Bio.Restriction.Restriction.TsuI-class.html#size,Variable Bio.Restriction.Restriction.Tth111I.size=Bio.Restriction.Restriction.Tth111I-class.html#size,Variable Bio.Restriction.Restriction.Tth111II.size=Bio.Restriction.Restriction.Tth111II-class.html#size,Variable Bio.Restriction.Restriction.UbaF11I.size=Bio.Restriction.Restriction.UbaF11I-class.html#size,Variable Bio.Restriction.Restriction.UbaF12I.size=Bio.Restriction.Restriction.UbaF12I-class.html#size,Variable Bio.Restriction.Restriction.UbaF13I.size=Bio.Restriction.Restriction.UbaF13I-class.html#size,Variable Bio.Restriction.Restriction.UbaF14I.size=Bio.Restriction.Restriction.UbaF14I-class.html#size,Variable Bio.Restriction.Restriction.UbaF9I.size=Bio.Restriction.Restriction.UbaF9I-class.html#size,Variable Bio.Restriction.Restriction.UbaPI.size=Bio.Restriction.Restriction.UbaPI-class.html#size,Variable Bio.Restriction.Restriction.UcoMSI.size=Bio.Restriction.Restriction.UcoMSI-class.html#size,Variable Bio.Restriction.Restriction.UnbI.size=Bio.Restriction.Restriction.UnbI-class.html#size,Variable Bio.Restriction.Restriction.Van91I.size=Bio.Restriction.Restriction.Van91I-class.html#size,Variable Bio.Restriction.Restriction.Vha464I.size=Bio.Restriction.Restriction.Vha464I-class.html#size,Variable Bio.Restriction.Restriction.VneI.size=Bio.Restriction.Restriction.VneI-class.html#size,Variable Bio.Restriction.Restriction.VpaK11AI.size=Bio.Restriction.Restriction.VpaK11AI-class.html#size,Variable Bio.Restriction.Restriction.VpaK11BI.size=Bio.Restriction.Restriction.VpaK11BI-class.html#size,Variable Bio.Restriction.Restriction.VspI.size=Bio.Restriction.Restriction.VspI-class.html#size,Variable Bio.Restriction.Restriction.WviI.size=Bio.Restriction.Restriction.WviI-class.html#size,Variable Bio.Restriction.Restriction.XagI.size=Bio.Restriction.Restriction.XagI-class.html#size,Variable Bio.Restriction.Restriction.XapI.size=Bio.Restriction.Restriction.XapI-class.html#size,Variable Bio.Restriction.Restriction.XbaI.size=Bio.Restriction.Restriction.XbaI-class.html#size,Variable Bio.Restriction.Restriction.XceI.size=Bio.Restriction.Restriction.XceI-class.html#size,Variable Bio.Restriction.Restriction.XcmI.size=Bio.Restriction.Restriction.XcmI-class.html#size,Variable Bio.Restriction.Restriction.XhoI.size=Bio.Restriction.Restriction.XhoI-class.html#size,Variable Bio.Restriction.Restriction.XhoII.size=Bio.Restriction.Restriction.XhoII-class.html#size,Variable Bio.Restriction.Restriction.XmaI.size=Bio.Restriction.Restriction.XmaI-class.html#size,Variable Bio.Restriction.Restriction.XmaIII.size=Bio.Restriction.Restriction.XmaIII-class.html#size,Variable Bio.Restriction.Restriction.XmaJI.size=Bio.Restriction.Restriction.XmaJI-class.html#size,Variable Bio.Restriction.Restriction.XmiI.size=Bio.Restriction.Restriction.XmiI-class.html#size,Variable Bio.Restriction.Restriction.XmnI.size=Bio.Restriction.Restriction.XmnI-class.html#size,Variable Bio.Restriction.Restriction.XspI.size=Bio.Restriction.Restriction.XspI-class.html#size,Variable Bio.Restriction.Restriction.YkrI.size=Bio.Restriction.Restriction.YkrI-class.html#size,Variable Bio.Restriction.Restriction.ZraI.size=Bio.Restriction.Restriction.ZraI-class.html#size,Variable Bio.Restriction.Restriction.ZrmI.size=Bio.Restriction.Restriction.ZrmI-class.html#size,Variable Bio.Restriction.Restriction.Zsp2I.size=Bio.Restriction.Restriction.Zsp2I-class.html#size,Variable Bio.codonalign.codonalphabet.CodonAlphabet.size=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#size"><a title="Bio.Alphabet.Alphabet.size
Bio.Alphabet.Reduced.HPModel.size
Bio.Alphabet.Reduced.Murphy10.size
Bio.Alphabet.Reduced.Murphy15.size
Bio.Alphabet.Reduced.Murphy4.size
Bio.Alphabet.Reduced.Murphy8.size
Bio.Alphabet.Reduced.PC5.size
Bio.Alphabet.SingleLetterAlphabet.size
Bio.Alphabet.ThreeLetterProtein.size
Bio.GenBank._FeatureConsumer.size
Bio.GenBank._RecordConsumer.size
Bio.Restriction.Restriction.AanI.size
Bio.Restriction.Restriction.AarI.size
Bio.Restriction.Restriction.AasI.size
Bio.Restriction.Restriction.AatII.size
Bio.Restriction.Restriction.AbaSI.size
Bio.Restriction.Restriction.AbsI.size
Bio.Restriction.Restriction.Acc16I.size
Bio.Restriction.Restriction.Acc36I.size
Bio.Restriction.Restriction.Acc65I.size
Bio.Restriction.Restriction.AccB1I.size
Bio.Restriction.Restriction.AccB7I.size
Bio.Restriction.Restriction.AccBSI.size
Bio.Restriction.Restriction.AccI.size
Bio.Restriction.Restriction.AccII.size
Bio.Restriction.Restriction.AccIII.size
Bio.Restriction.Restriction.AceIII.size
Bio.Restriction.Restriction.AciI.size
Bio.Restriction.Restriction.AclI.size
Bio.Restriction.Restriction.AclWI.size
Bio.Restriction.Restriction.AcoI.size
Bio.Restriction.Restriction.AcsI.size
Bio.Restriction.Restriction.AcuI.size
Bio.Restriction.Restriction.AcvI.size
Bio.Restriction.Restriction.AcyI.size
Bio.Restriction.Restriction.AdeI.size
Bio.Restriction.Restriction.AfaI.size
Bio.Restriction.Restriction.AfeI.size
Bio.Restriction.Restriction.AfiI.size
Bio.Restriction.Restriction.AflII.size
Bio.Restriction.Restriction.AflIII.size
Bio.Restriction.Restriction.AgeI.size
Bio.Restriction.Restriction.AgsI.size
Bio.Restriction.Restriction.AhaIII.size
Bio.Restriction.Restriction.AhdI.size
Bio.Restriction.Restriction.AhlI.size
Bio.Restriction.Restriction.AjiI.size
Bio.Restriction.Restriction.AjnI.size
Bio.Restriction.Restriction.AjuI.size
Bio.Restriction.Restriction.AleI.size
Bio.Restriction.Restriction.AlfI.size
Bio.Restriction.Restriction.AloI.size
Bio.Restriction.Restriction.AluBI.size
Bio.Restriction.Restriction.AluI.size
Bio.Restriction.Restriction.Alw21I.size
Bio.Restriction.Restriction.Alw26I.size
Bio.Restriction.Restriction.Alw44I.size
Bio.Restriction.Restriction.AlwFI.size
Bio.Restriction.Restriction.AlwI.size
Bio.Restriction.Restriction.AlwNI.size
Bio.Restriction.Restriction.Ama87I.size
Bio.Restriction.Restriction.Aor13HI.size
Bio.Restriction.Restriction.Aor51HI.size
Bio.Restriction.Restriction.AoxI.size
Bio.Restriction.Restriction.ApaBI.size
Bio.Restriction.Restriction.ApaI.size
Bio.Restriction.Restriction.ApaLI.size
Bio.Restriction.Restriction.ApeKI.size
Bio.Restriction.Restriction.ApoI.size
Bio.Restriction.Restriction.ApyPI.size
Bio.Restriction.Restriction.AquII.size
Bio.Restriction.Restriction.AquIII.size
Bio.Restriction.Restriction.AquIV.size
Bio.Restriction.Restriction.ArsI.size
Bio.Restriction.Restriction.AscI.size
Bio.Restriction.Restriction.AseI.size
Bio.Restriction.Restriction.Asi256I.size
Bio.Restriction.Restriction.AsiGI.size
Bio.Restriction.Restriction.AsiSI.size
Bio.Restriction.Restriction.Asp700I.size
Bio.Restriction.Restriction.Asp718I.size
Bio.Restriction.Restriction.AspA2I.size
Bio.Restriction.Restriction.AspBHI.size
Bio.Restriction.Restriction.AspLEI.size
Bio.Restriction.Restriction.AspS9I.size
Bio.Restriction.Restriction.AssI.size
Bio.Restriction.Restriction.AsuC2I.size
Bio.Restriction.Restriction.AsuHPI.size
Bio.Restriction.Restriction.AsuI.size
Bio.Restriction.Restriction.AsuII.size
Bio.Restriction.Restriction.AsuNHI.size
Bio.Restriction.Restriction.AvaI.size
Bio.Restriction.Restriction.AvaII.size
Bio.Restriction.Restriction.AvaIII.size
Bio.Restriction.Restriction.AvrII.size
Bio.Restriction.Restriction.AxyI.size
Bio.Restriction.Restriction.BaeGI.size
Bio.Restriction.Restriction.BaeI.size
Bio.Restriction.Restriction.BalI.size
Bio.Restriction.Restriction.BamHI.size
Bio.Restriction.Restriction.BanI.size
Bio.Restriction.Restriction.BanII.size
Bio.Restriction.Restriction.BarI.size
Bio.Restriction.Restriction.BasI.size
Bio.Restriction.Restriction.BauI.size
Bio.Restriction.Restriction.Bbr7I.size
Bio.Restriction.Restriction.BbrPI.size
Bio.Restriction.Restriction.BbsI.size
Bio.Restriction.Restriction.Bbv12I.size
Bio.Restriction.Restriction.BbvCI.size
Bio.Restriction.Restriction.BbvI.size
Bio.Restriction.Restriction.BbvII.size
Bio.Restriction.Restriction.BccI.size
Bio.Restriction.Restriction.Bce83I.size
Bio.Restriction.Restriction.BceAI.size
Bio.Restriction.Restriction.BcefI.size
Bio.Restriction.Restriction.BcgI.size
Bio.Restriction.Restriction.BciT130I.size
Bio.Restriction.Restriction.BciVI.size
Bio.Restriction.Restriction.BclI.size
Bio.Restriction.Restriction.BcnI.size
Bio.Restriction.Restriction.BcoDI.size
Bio.Restriction.Restriction.BcuI.size
Bio.Restriction.Restriction.BdaI.size
Bio.Restriction.Restriction.BetI.size
Bio.Restriction.Restriction.BfaI.size
Bio.Restriction.Restriction.BfiI.size
Bio.Restriction.Restriction.BfmI.size
Bio.Restriction.Restriction.BfoI.size
Bio.Restriction.Restriction.BfrI.size
Bio.Restriction.Restriction.BfuAI.size
Bio.Restriction.Restriction.BfuCI.size
Bio.Restriction.Restriction.BfuI.size
Bio.Restriction.Restriction.BglI.size
Bio.Restriction.Restriction.BglII.size
Bio.Restriction.Restriction.BinI.size
Bio.Restriction.Restriction.BisI.size
Bio.Restriction.Restriction.BlnI.size
Bio.Restriction.Restriction.BlpI.size
Bio.Restriction.Restriction.BlsI.size
Bio.Restriction.Restriction.BmcAI.size
Bio.Restriction.Restriction.Bme1390I.size
Bio.Restriction.Restriction.Bme18I.size
Bio.Restriction.Restriction.BmeDI.size
Bio.Restriction.Restriction.BmeRI.size
Bio.Restriction.Restriction.BmeT110I.size
Bio.Restriction.Restriction.BmgBI.size
Bio.Restriction.Restriction.BmgI.size
Bio.Restriction.Restriction.BmgT120I.size
Bio.Restriction.Restriction.BmiI.size
Bio.Restriction.Restriction.BmrFI.size
Bio.Restriction.Restriction.BmrI.size
Bio.Restriction.Restriction.BmsI.size
Bio.Restriction.Restriction.BmtI.size
Bio.Restriction.Restriction.BmuI.size
Bio.Restriction.Restriction.BoxI.size
Bio.Restriction.Restriction.BpiI.size
Bio.Restriction.Restriction.BplI.size
Bio.Restriction.Restriction.BpmI.size
Bio.Restriction.Restriction.Bpu10I.size
Bio.Restriction.Restriction.Bpu1102I.size
Bio.Restriction.Restriction.Bpu14I.size
Bio.Restriction.Restriction.BpuEI.size
Bio.Restriction.Restriction.BpuMI.size
Bio.Restriction.Restriction.BpvUI.size
Bio.Restriction.Restriction.Bsa29I.size
Bio.Restriction.Restriction.BsaAI.size
Bio.Restriction.Restriction.BsaBI.size
Bio.Restriction.Restriction.BsaHI.size
Bio.Restriction.Restriction.BsaI.size
Bio.Restriction.Restriction.BsaJI.size
Bio.Restriction.Restriction.BsaWI.size
Bio.Restriction.Restriction.BsaXI.size
Bio.Restriction.Restriction.BsbI.size
Bio.Restriction.Restriction.Bsc4I.size
Bio.Restriction.Restriction.BscAI.size
Bio.Restriction.Restriction.BscGI.size
Bio.Restriction.Restriction.Bse118I.size
Bio.Restriction.Restriction.Bse1I.size
Bio.Restriction.Restriction.Bse21I.size
Bio.Restriction.Restriction.Bse3DI.size
Bio.Restriction.Restriction.Bse8I.size
Bio.Restriction.Restriction.BseAI.size
Bio.Restriction.Restriction.BseBI.size
Bio.Restriction.Restriction.BseCI.size
Bio.Restriction.Restriction.BseDI.size
Bio.Restriction.Restriction.BseGI.size
Bio.Restriction.Restriction.BseJI.size
Bio.Restriction.Restriction.BseLI.size
Bio.Restriction.Restriction.BseMI.size
Bio.Restriction.Restriction.BseMII.size
Bio.Restriction.Restriction.BseNI.size
Bio.Restriction.Restriction.BsePI.size
Bio.Restriction.Restriction.BseRI.size
Bio.Restriction.Restriction.BseSI.size
Bio.Restriction.Restriction.BseX3I.size
Bio.Restriction.Restriction.BseXI.size
Bio.Restriction.Restriction.BseYI.size
Bio.Restriction.Restriction.BsgI.size
Bio.Restriction.Restriction.Bsh1236I.size
Bio.Restriction.Restriction.Bsh1285I.size
Bio.Restriction.Restriction.BshFI.size
Bio.Restriction.Restriction.BshNI.size
Bio.Restriction.Restriction.BshTI.size
Bio.Restriction.Restriction.BshVI.size
Bio.Restriction.Restriction.BsiEI.size
Bio.Restriction.Restriction.BsiHKAI.size
Bio.Restriction.Restriction.BsiHKCI.size
Bio.Restriction.Restriction.BsiI.size
Bio.Restriction.Restriction.BsiSI.size
Bio.Restriction.Restriction.BsiWI.size
Bio.Restriction.Restriction.BsiYI.size
Bio.Restriction.Restriction.BslFI.size
Bio.Restriction.Restriction.BslI.size
Bio.Restriction.Restriction.BsmAI.size
Bio.Restriction.Restriction.BsmBI.size
Bio.Restriction.Restriction.BsmFI.size
Bio.Restriction.Restriction.BsmI.size
Bio.Restriction.Restriction.BsnI.size
Bio.Restriction.Restriction.Bso31I.size
Bio.Restriction.Restriction.BsoBI.size
Bio.Restriction.Restriction.Bsp119I.size
Bio.Restriction.Restriction.Bsp120I.size
Bio.Restriction.Restriction.Bsp1286I.size
Bio.Restriction.Restriction.Bsp13I.size
Bio.Restriction.Restriction.Bsp1407I.size
Bio.Restriction.Restriction.Bsp143I.size
Bio.Restriction.Restriction.Bsp1720I.size
Bio.Restriction.Restriction.Bsp19I.size
Bio.Restriction.Restriction.Bsp24I.size
Bio.Restriction.Restriction.Bsp68I.size
Bio.Restriction.Restriction.BspACI.size
Bio.Restriction.Restriction.BspCNI.size
Bio.Restriction.Restriction.BspD6I.size
Bio.Restriction.Restriction.BspDI.size
Bio.Restriction.Restriction.BspEI.size
Bio.Restriction.Restriction.BspFNI.size
Bio.Restriction.Restriction.BspGI.size
Bio.Restriction.Restriction.BspHI.size
Bio.Restriction.Restriction.BspLI.size
Bio.Restriction.Restriction.BspLU11I.size
Bio.Restriction.Restriction.BspMI.size
Bio.Restriction.Restriction.BspMII.size
Bio.Restriction.Restriction.BspNCI.size
Bio.Restriction.Restriction.BspOI.size
Bio.Restriction.Restriction.BspPI.size
Bio.Restriction.Restriction.BspQI.size
Bio.Restriction.Restriction.BspT104I.size
Bio.Restriction.Restriction.BspT107I.size
Bio.Restriction.Restriction.BspTI.size
Bio.Restriction.Restriction.BsrBI.size
Bio.Restriction.Restriction.BsrDI.size
Bio.Restriction.Restriction.BsrFI.size
Bio.Restriction.Restriction.BsrGI.size
Bio.Restriction.Restriction.BsrI.size
Bio.Restriction.Restriction.BsrSI.size
Bio.Restriction.Restriction.BssAI.size
Bio.Restriction.Restriction.BssECI.size
Bio.Restriction.Restriction.BssHII.size
Bio.Restriction.Restriction.BssKI.size
Bio.Restriction.Restriction.BssMI.size
Bio.Restriction.Restriction.BssNAI.size
Bio.Restriction.Restriction.BssNI.size
Bio.Restriction.Restriction.BssSI.size
Bio.Restriction.Restriction.BssT1I.size
Bio.Restriction.Restriction.Bst1107I.size
Bio.Restriction.Restriction.Bst2BI.size
Bio.Restriction.Restriction.Bst2UI.size
Bio.Restriction.Restriction.Bst4CI.size
Bio.Restriction.Restriction.Bst6I.size
Bio.Restriction.Restriction.BstACI.size
Bio.Restriction.Restriction.BstAFI.size
Bio.Restriction.Restriction.BstAPI.size
Bio.Restriction.Restriction.BstAUI.size
Bio.Restriction.Restriction.BstBAI.size
Bio.Restriction.Restriction.BstBI.size
Bio.Restriction.Restriction.BstC8I.size
Bio.Restriction.Restriction.BstDEI.size
Bio.Restriction.Restriction.BstDSI.size
Bio.Restriction.Restriction.BstEII.size
Bio.Restriction.Restriction.BstENI.size
Bio.Restriction.Restriction.BstF5I.size
Bio.Restriction.Restriction.BstFNI.size
Bio.Restriction.Restriction.BstH2I.size
Bio.Restriction.Restriction.BstHHI.size
Bio.Restriction.Restriction.BstKTI.size
Bio.Restriction.Restriction.BstMAI.size
Bio.Restriction.Restriction.BstMBI.size
Bio.Restriction.Restriction.BstMCI.size
Bio.Restriction.Restriction.BstMWI.size
Bio.Restriction.Restriction.BstNI.size
Bio.Restriction.Restriction.BstNSI.size
Bio.Restriction.Restriction.BstOI.size
Bio.Restriction.Restriction.BstPAI.size
Bio.Restriction.Restriction.BstPI.size
Bio.Restriction.Restriction.BstSCI.size
Bio.Restriction.Restriction.BstSFI.size
Bio.Restriction.Restriction.BstSLI.size
Bio.Restriction.Restriction.BstSNI.size
Bio.Restriction.Restriction.BstUI.size
Bio.Restriction.Restriction.BstV1I.size
Bio.Restriction.Restriction.BstV2I.size
Bio.Restriction.Restriction.BstX2I.size
Bio.Restriction.Restriction.BstXI.size
Bio.Restriction.Restriction.BstYI.size
Bio.Restriction.Restriction.BstZ17I.size
Bio.Restriction.Restriction.BstZI.size
Bio.Restriction.Restriction.Bsu15I.size
Bio.Restriction.Restriction.Bsu36I.size
Bio.Restriction.Restriction.BsuI.size
Bio.Restriction.Restriction.BsuRI.size
Bio.Restriction.Restriction.BtgI.size
Bio.Restriction.Restriction.BtgZI.size
Bio.Restriction.Restriction.BthCI.size
Bio.Restriction.Restriction.BtrI.size
Bio.Restriction.Restriction.BtsCI.size
Bio.Restriction.Restriction.BtsI.size
Bio.Restriction.Restriction.BtsIMutI.size
Bio.Restriction.Restriction.BtuMI.size
Bio.Restriction.Restriction.BveI.size
Bio.Restriction.Restriction.Cac8I.size
Bio.Restriction.Restriction.CaiI.size
Bio.Restriction.Restriction.CauII.size
Bio.Restriction.Restriction.CchII.size
Bio.Restriction.Restriction.CchIII.size
Bio.Restriction.Restriction.CciI.size
Bio.Restriction.Restriction.CciNI.size
Bio.Restriction.Restriction.Cdi630V.size
Bio.Restriction.Restriction.CdiI.size
Bio.Restriction.Restriction.CdpI.size
Bio.Restriction.Restriction.CfoI.size
Bio.Restriction.Restriction.Cfr10I.size
Bio.Restriction.Restriction.Cfr13I.size
Bio.Restriction.Restriction.Cfr42I.size
Bio.Restriction.Restriction.Cfr9I.size
Bio.Restriction.Restriction.CfrI.size
Bio.Restriction.Restriction.Cgl13032I.size
Bio.Restriction.Restriction.Cgl13032II.size
Bio.Restriction.Restriction.ChaI.size
Bio.Restriction.Restriction.CjeFIII.size
Bio.Restriction.Restriction.CjeFV.size
Bio.Restriction.Restriction.CjeI.size
Bio.Restriction.Restriction.CjeNII.size
Bio.Restriction.Restriction.CjeNIII.size
Bio.Restriction.Restriction.CjeP659IV.size
Bio.Restriction.Restriction.CjePI.size
Bio.Restriction.Restriction.CjuI.size
Bio.Restriction.Restriction.CjuII.size
Bio.Restriction.Restriction.ClaI.size
Bio.Restriction.Restriction.CpoI.size
Bio.Restriction.Restriction.CseI.size
Bio.Restriction.Restriction.CsiI.size
Bio.Restriction.Restriction.Csp6I.size
Bio.Restriction.Restriction.CspAI.size
Bio.Restriction.Restriction.CspCI.size
Bio.Restriction.Restriction.CspI.size
Bio.Restriction.Restriction.CstMI.size
Bio.Restriction.Restriction.CviAII.size
Bio.Restriction.Restriction.CviJI.size
Bio.Restriction.Restriction.CviKI_1.size
Bio.Restriction.Restriction.CviQI.size
Bio.Restriction.Restriction.CviRI.size
Bio.Restriction.Restriction.DdeI.size
Bio.Restriction.Restriction.DinI.size
Bio.Restriction.Restriction.DpnI.size
Bio.Restriction.Restriction.DpnII.size
Bio.Restriction.Restriction.DraI.size
Bio.Restriction.Restriction.DraII.size
Bio.Restriction.Restriction.DraIII.size
Bio.Restriction.Restriction.DraRI.size
Bio.Restriction.Restriction.DrdI.size
Bio.Restriction.Restriction.DrdII.size
Bio.Restriction.Restriction.DriI.size
Bio.Restriction.Restriction.DsaI.size
Bio.Restriction.Restriction.DseDI.size
Bio.Restriction.Restriction.EaeI.size
Bio.Restriction.Restriction.EagI.size
Bio.Restriction.Restriction.Eam1104I.size
Bio.Restriction.Restriction.Eam1105I.size
Bio.Restriction.Restriction.EarI.size
Bio.Restriction.Restriction.EciI.size
Bio.Restriction.Restriction.Ecl136II.size
Bio.Restriction.Restriction.EclXI.size
Bio.Restriction.Restriction.Eco105I.size
Bio.Restriction.Restriction.Eco130I.size
Bio.Restriction.Restriction.Eco147I.size
Bio.Restriction.Restriction.Eco24I.size
Bio.Restriction.Restriction.Eco31I.size
Bio.Restriction.Restriction.Eco32I.size
Bio.Restriction.Restriction.Eco47I.size
Bio.Restriction.Restriction.Eco47III.size
Bio.Restriction.Restriction.Eco52I.size
Bio.Restriction.Restriction.Eco53kI.size
Bio.Restriction.Restriction.Eco57I.size
Bio.Restriction.Restriction.Eco57MI.size
Bio.Restriction.Restriction.Eco72I.size
Bio.Restriction.Restriction.Eco81I.size
Bio.Restriction.Restriction.Eco88I.size
Bio.Restriction.Restriction.Eco91I.size
Bio.Restriction.Restriction.EcoHI.size
Bio.Restriction.Restriction.EcoICRI.size
Bio.Restriction.Restriction.EcoNI.size
Bio.Restriction.Restriction.EcoO109I.size
Bio.Restriction.Restriction.EcoO65I.size
Bio.Restriction.Restriction.EcoRI.size
Bio.Restriction.Restriction.EcoRII.size
Bio.Restriction.Restriction.EcoRV.size
Bio.Restriction.Restriction.EcoT14I.size
Bio.Restriction.Restriction.EcoT22I.size
Bio.Restriction.Restriction.EcoT38I.size
Bio.Restriction.Restriction.EgeI.size
Bio.Restriction.Restriction.EheI.size
Bio.Restriction.Restriction.ErhI.size
Bio.Restriction.Restriction.EsaBC3I.size
Bio.Restriction.Restriction.EsaSSI.size
Bio.Restriction.Restriction.Esp3I.size
Bio.Restriction.Restriction.EspI.size
Bio.Restriction.Restriction.FaeI.size
Bio.Restriction.Restriction.FaiI.size
Bio.Restriction.Restriction.FalI.size
Bio.Restriction.Restriction.FaqI.size
Bio.Restriction.Restriction.FatI.size
Bio.Restriction.Restriction.FauI.size
Bio.Restriction.Restriction.FauNDI.size
Bio.Restriction.Restriction.FbaI.size
Bio.Restriction.Restriction.FblI.size
Bio.Restriction.Restriction.FinI.size
Bio.Restriction.Restriction.FmuI.size
Bio.Restriction.Restriction.Fnu4HI.size
Bio.Restriction.Restriction.FnuDII.size
Bio.Restriction.Restriction.FokI.size
Bio.Restriction.Restriction.FriOI.size
Bio.Restriction.Restriction.FseI.size
Bio.Restriction.Restriction.Fsp4HI.size
Bio.Restriction.Restriction.FspAI.size
Bio.Restriction.Restriction.FspBI.size
Bio.Restriction.Restriction.FspEI.size
Bio.Restriction.Restriction.FspI.size
Bio.Restriction.Restriction.GauT27I.size
Bio.Restriction.Restriction.GdiII.size
Bio.Restriction.Restriction.GlaI.size
Bio.Restriction.Restriction.GluI.size
Bio.Restriction.Restriction.GsaI.size
Bio.Restriction.Restriction.GsuI.size
Bio.Restriction.Restriction.HaeI.size
Bio.Restriction.Restriction.HaeII.size
Bio.Restriction.Restriction.HaeIII.size
Bio.Restriction.Restriction.HapII.size
Bio.Restriction.Restriction.HauII.size
Bio.Restriction.Restriction.HgaI.size
Bio.Restriction.Restriction.HgiAI.size
Bio.Restriction.Restriction.HgiCI.size
Bio.Restriction.Restriction.HgiEII.size
Bio.Restriction.Restriction.HgiJII.size
Bio.Restriction.Restriction.HhaI.size
Bio.Restriction.Restriction.Hin1I.size
Bio.Restriction.Restriction.Hin1II.size
Bio.Restriction.Restriction.Hin4I.size
Bio.Restriction.Restriction.Hin4II.size
Bio.Restriction.Restriction.Hin6I.size
Bio.Restriction.Restriction.HinP1I.size
Bio.Restriction.Restriction.HincII.size
Bio.Restriction.Restriction.HindII.size
Bio.Restriction.Restriction.HindIII.size
Bio.Restriction.Restriction.HinfI.size
Bio.Restriction.Restriction.HpaI.size
Bio.Restriction.Restriction.HpaII.size
Bio.Restriction.Restriction.HphI.size
Bio.Restriction.Restriction.Hpy166II.size
Bio.Restriction.Restriction.Hpy178III.size
Bio.Restriction.Restriction.Hpy188I.size
Bio.Restriction.Restriction.Hpy188III.size
Bio.Restriction.Restriction.Hpy8I.size
Bio.Restriction.Restriction.Hpy99I.size
Bio.Restriction.Restriction.Hpy99XIII.size
Bio.Restriction.Restriction.Hpy99XIV.size
Bio.Restriction.Restriction.HpyAV.size
Bio.Restriction.Restriction.HpyCH4III.size
Bio.Restriction.Restriction.HpyCH4IV.size
Bio.Restriction.Restriction.HpyCH4V.size
Bio.Restriction.Restriction.HpyF10VI.size
Bio.Restriction.Restriction.HpyF3I.size
Bio.Restriction.Restriction.HpySE526I.size
Bio.Restriction.Restriction.Hsp92I.size
Bio.Restriction.Restriction.Hsp92II.size
Bio.Restriction.Restriction.HspAI.size
Bio.Restriction.Restriction.Jma19592I.size
Bio.Restriction.Restriction.KasI.size
Bio.Restriction.Restriction.KflI.size
Bio.Restriction.Restriction.Kpn2I.size
Bio.Restriction.Restriction.KpnI.size
Bio.Restriction.Restriction.KroI.size
Bio.Restriction.Restriction.Ksp22I.size
Bio.Restriction.Restriction.Ksp632I.size
Bio.Restriction.Restriction.KspAI.size
Bio.Restriction.Restriction.KspI.size
Bio.Restriction.Restriction.Kzo9I.size
Bio.Restriction.Restriction.LguI.size
Bio.Restriction.Restriction.LpnI.size
Bio.Restriction.Restriction.LpnPI.size
Bio.Restriction.Restriction.Lsp1109I.size
Bio.Restriction.Restriction.LweI.size
Bio.Restriction.Restriction.MabI.size
Bio.Restriction.Restriction.MaeI.size
Bio.Restriction.Restriction.MaeII.size
Bio.Restriction.Restriction.MaeIII.size
Bio.Restriction.Restriction.MalI.size
Bio.Restriction.Restriction.MaqI.size
Bio.Restriction.Restriction.MauBI.size
Bio.Restriction.Restriction.MbiI.size
Bio.Restriction.Restriction.MboI.size
Bio.Restriction.Restriction.MboII.size
Bio.Restriction.Restriction.McaTI.size
Bio.Restriction.Restriction.McrI.size
Bio.Restriction.Restriction.MfeI.size
Bio.Restriction.Restriction.MflI.size
Bio.Restriction.Restriction.MhlI.size
Bio.Restriction.Restriction.MjaIV.size
Bio.Restriction.Restriction.MkaDII.size
Bio.Restriction.Restriction.MlsI.size
Bio.Restriction.Restriction.MluCI.size
Bio.Restriction.Restriction.MluI.size
Bio.Restriction.Restriction.MluNI.size
Bio.Restriction.Restriction.Mly113I.size
Bio.Restriction.Restriction.MlyI.size
Bio.Restriction.Restriction.MmeI.size
Bio.Restriction.Restriction.MnlI.size
Bio.Restriction.Restriction.Mph1103I.size
Bio.Restriction.Restriction.MreI.size
Bio.Restriction.Restriction.MroI.size
Bio.Restriction.Restriction.MroNI.size
Bio.Restriction.Restriction.MroXI.size
Bio.Restriction.Restriction.MscI.size
Bio.Restriction.Restriction.MseI.size
Bio.Restriction.Restriction.MslI.size
Bio.Restriction.Restriction.Msp20I.size
Bio.Restriction.Restriction.MspA1I.size
Bio.Restriction.Restriction.MspCI.size
Bio.Restriction.Restriction.MspI.size
Bio.Restriction.Restriction.MspJI.size
Bio.Restriction.Restriction.MspR9I.size
Bio.Restriction.Restriction.MssI.size
Bio.Restriction.Restriction.MstI.size
Bio.Restriction.Restriction.MunI.size
Bio.Restriction.Restriction.Mva1269I.size
Bio.Restriction.Restriction.MvaI.size
Bio.Restriction.Restriction.MvnI.size
Bio.Restriction.Restriction.MvrI.size
Bio.Restriction.Restriction.MwoI.size
Bio.Restriction.Restriction.NaeI.size
Bio.Restriction.Restriction.NarI.size
Bio.Restriction.Restriction.NciI.size
Bio.Restriction.Restriction.NcoI.size
Bio.Restriction.Restriction.NdeI.size
Bio.Restriction.Restriction.NdeII.size
Bio.Restriction.Restriction.NgoAVIII.size
Bio.Restriction.Restriction.NgoMIV.size
Bio.Restriction.Restriction.NhaXI.size
Bio.Restriction.Restriction.NheI.size
Bio.Restriction.Restriction.NlaCI.size
Bio.Restriction.Restriction.NlaIII.size
Bio.Restriction.Restriction.NlaIV.size
Bio.Restriction.Restriction.Nli3877I.size
Bio.Restriction.Restriction.NmeAIII.size
Bio.Restriction.Restriction.NmeDI.size
Bio.Restriction.Restriction.NmuCI.size
Bio.Restriction.Restriction.NotI.size
Bio.Restriction.Restriction.NruI.size
Bio.Restriction.Restriction.NsbI.size
Bio.Restriction.Restriction.NsiI.size
Bio.Restriction.Restriction.NspBII.size
Bio.Restriction.Restriction.NspI.size
Bio.Restriction.Restriction.NspV.size
Bio.Restriction.Restriction.OliI.size
Bio.Restriction.Restriction.PabI.size
Bio.Restriction.Restriction.PacI.size
Bio.Restriction.Restriction.PaeI.size
Bio.Restriction.Restriction.PaeR7I.size
Bio.Restriction.Restriction.PagI.size
Bio.Restriction.Restriction.PalAI.size
Bio.Restriction.Restriction.PasI.size
Bio.Restriction.Restriction.PauI.size
Bio.Restriction.Restriction.PceI.size
Bio.Restriction.Restriction.PciI.size
Bio.Restriction.Restriction.PciSI.size
Bio.Restriction.Restriction.PcsI.size
Bio.Restriction.Restriction.PctI.size
Bio.Restriction.Restriction.PdiI.size
Bio.Restriction.Restriction.PdmI.size
Bio.Restriction.Restriction.PenI.size
Bio.Restriction.Restriction.PfeI.size
Bio.Restriction.Restriction.Pfl1108I.size
Bio.Restriction.Restriction.Pfl23II.size
Bio.Restriction.Restriction.PflFI.size
Bio.Restriction.Restriction.PflMI.size
Bio.Restriction.Restriction.PfoI.size
Bio.Restriction.Restriction.PinAI.size
Bio.Restriction.Restriction.PlaDI.size
Bio.Restriction.Restriction.Ple19I.size
Bio.Restriction.Restriction.PleI.size
Bio.Restriction.Restriction.PluTI.size
Bio.Restriction.Restriction.PmaCI.size
Bio.Restriction.Restriction.PmeI.size
Bio.Restriction.Restriction.PmlI.size
Bio.Restriction.Restriction.PpiI.size
Bio.Restriction.Restriction.PpsI.size
Bio.Restriction.Restriction.Ppu10I.size
Bio.Restriction.Restriction.Ppu21I.size
Bio.Restriction.Restriction.PpuMI.size
Bio.Restriction.Restriction.PscI.size
Bio.Restriction.Restriction.PshAI.size
Bio.Restriction.Restriction.PshBI.size
Bio.Restriction.Restriction.PsiI.size
Bio.Restriction.Restriction.Psp03I.size
Bio.Restriction.Restriction.Psp124BI.size
Bio.Restriction.Restriction.Psp1406I.size
Bio.Restriction.Restriction.Psp5II.size
Bio.Restriction.Restriction.Psp6I.size
Bio.Restriction.Restriction.PspCI.size
Bio.Restriction.Restriction.PspEI.size
Bio.Restriction.Restriction.PspGI.size
Bio.Restriction.Restriction.PspLI.size
Bio.Restriction.Restriction.PspN4I.size
Bio.Restriction.Restriction.PspOMI.size
Bio.Restriction.Restriction.PspOMII.size
Bio.Restriction.Restriction.PspPI.size
Bio.Restriction.Restriction.PspPPI.size
Bio.Restriction.Restriction.PspPRI.size
Bio.Restriction.Restriction.PspXI.size
Bio.Restriction.Restriction.PsrI.size
Bio.Restriction.Restriction.PssI.size
Bio.Restriction.Restriction.PstI.size
Bio.Restriction.Restriction.PstNI.size
Bio.Restriction.Restriction.PsuI.size
Bio.Restriction.Restriction.PsyI.size
Bio.Restriction.Restriction.PteI.size
Bio.Restriction.Restriction.PvuI.size
Bio.Restriction.Restriction.PvuII.size
Bio.Restriction.Restriction.R2_BceSIV.size
Bio.Restriction.Restriction.RceI.size
Bio.Restriction.Restriction.RdeGBI.size
Bio.Restriction.Restriction.RdeGBII.size
Bio.Restriction.Restriction.RdeGBIII.size
Bio.Restriction.Restriction.RflFIII.size
Bio.Restriction.Restriction.RgaI.size
Bio.Restriction.Restriction.RigI.size
Bio.Restriction.Restriction.RlaI.size
Bio.Restriction.Restriction.RleAI.size
Bio.Restriction.Restriction.RpaB5I.size
Bio.Restriction.Restriction.RpaBI.size
Bio.Restriction.Restriction.RpaI.size
Bio.Restriction.Restriction.RpaTI.size
Bio.Restriction.Restriction.RruI.size
Bio.Restriction.Restriction.RsaI.size
Bio.Restriction.Restriction.RsaNI.size
Bio.Restriction.Restriction.RseI.size
Bio.Restriction.Restriction.Rsr2I.size
Bio.Restriction.Restriction.RsrII.size
Bio.Restriction.Restriction.SacI.size
Bio.Restriction.Restriction.SacII.size
Bio.Restriction.Restriction.SalI.size
Bio.Restriction.Restriction.SanDI.size
Bio.Restriction.Restriction.SapI.size
Bio.Restriction.Restriction.SaqAI.size
Bio.Restriction.Restriction.SatI.size
Bio.Restriction.Restriction.Sau3AI.size
Bio.Restriction.Restriction.Sau96I.size
Bio.Restriction.Restriction.SauI.size
Bio.Restriction.Restriction.SbfI.size
Bio.Restriction.Restriction.ScaI.size
Bio.Restriction.Restriction.SchI.size
Bio.Restriction.Restriction.SciI.size
Bio.Restriction.Restriction.ScrFI.size
Bio.Restriction.Restriction.SdaI.size
Bio.Restriction.Restriction.SdeAI.size
Bio.Restriction.Restriction.SdeOSI.size
Bio.Restriction.Restriction.SduI.size
Bio.Restriction.Restriction.SecI.size
Bio.Restriction.Restriction.SelI.size
Bio.Restriction.Restriction.SetI.size
Bio.Restriction.Restriction.SexAI.size
Bio.Restriction.Restriction.SfaAI.size
Bio.Restriction.Restriction.SfaNI.size
Bio.Restriction.Restriction.SfcI.size
Bio.Restriction.Restriction.SfeI.size
Bio.Restriction.Restriction.SfiI.size
Bio.Restriction.Restriction.SfoI.size
Bio.Restriction.Restriction.Sfr274I.size
Bio.Restriction.Restriction.Sfr303I.size
Bio.Restriction.Restriction.SfuI.size
Bio.Restriction.Restriction.SgeI.size
Bio.Restriction.Restriction.SgfI.size
Bio.Restriction.Restriction.SgrAI.size
Bio.Restriction.Restriction.SgrBI.size
Bio.Restriction.Restriction.SgrDI.size
Bio.Restriction.Restriction.SgrTI.size
Bio.Restriction.Restriction.SgsI.size
Bio.Restriction.Restriction.SimI.size
Bio.Restriction.Restriction.SlaI.size
Bio.Restriction.Restriction.SmaI.size
Bio.Restriction.Restriction.SmiI.size
Bio.Restriction.Restriction.SmiMI.size
Bio.Restriction.Restriction.SmlI.size
Bio.Restriction.Restriction.SmoI.size
Bio.Restriction.Restriction.SnaBI.size
Bio.Restriction.Restriction.SnaI.size
Bio.Restriction.Restriction.Sno506I.size
Bio.Restriction.Restriction.SpeI.size
Bio.Restriction.Restriction.SphI.size
Bio.Restriction.Restriction.SplI.size
Bio.Restriction.Restriction.SpoDI.size
Bio.Restriction.Restriction.SrfI.size
Bio.Restriction.Restriction.Sse232I.size
Bio.Restriction.Restriction.Sse8387I.size
Bio.Restriction.Restriction.Sse8647I.size
Bio.Restriction.Restriction.Sse9I.size
Bio.Restriction.Restriction.SseBI.size
Bio.Restriction.Restriction.SsiI.size
Bio.Restriction.Restriction.SspD5I.size
Bio.Restriction.Restriction.SspDI.size
Bio.Restriction.Restriction.SspI.size
Bio.Restriction.Restriction.SstE37I.size
Bio.Restriction.Restriction.SstI.size
Bio.Restriction.Restriction.Sth132I.size
Bio.Restriction.Restriction.Sth302II.size
Bio.Restriction.Restriction.StrI.size
Bio.Restriction.Restriction.StsI.size
Bio.Restriction.Restriction.StuI.size
Bio.Restriction.Restriction.StyD4I.size
Bio.Restriction.Restriction.StyI.size
Bio.Restriction.Restriction.SwaI.size
Bio.Restriction.Restriction.TaaI.size
Bio.Restriction.Restriction.TaiI.size
Bio.Restriction.Restriction.TaqI.size
Bio.Restriction.Restriction.TaqII.size
Bio.Restriction.Restriction.TasI.size
Bio.Restriction.Restriction.TatI.size
Bio.Restriction.Restriction.TauI.size
Bio.Restriction.Restriction.TfiI.size
Bio.Restriction.Restriction.Tru1I.size
Bio.Restriction.Restriction.Tru9I.size
Bio.Restriction.Restriction.TscAI.size
Bio.Restriction.Restriction.TseFI.size
Bio.Restriction.Restriction.TseI.size
Bio.Restriction.Restriction.TsoI.size
Bio.Restriction.Restriction.Tsp45I.size
Bio.Restriction.Restriction.Tsp4CI.size
Bio.Restriction.Restriction.TspDTI.size
Bio.Restriction.Restriction.TspEI.size
Bio.Restriction.Restriction.TspGWI.size
Bio.Restriction.Restriction.TspMI.size
Bio.Restriction.Restriction.TspRI.size
Bio.Restriction.Restriction.TssI.size
Bio.Restriction.Restriction.TstI.size
Bio.Restriction.Restriction.TsuI.size
Bio.Restriction.Restriction.Tth111I.size
Bio.Restriction.Restriction.Tth111II.size
Bio.Restriction.Restriction.UbaF11I.size
Bio.Restriction.Restriction.UbaF12I.size
Bio.Restriction.Restriction.UbaF13I.size
Bio.Restriction.Restriction.UbaF14I.size
Bio.Restriction.Restriction.UbaF9I.size
Bio.Restriction.Restriction.UbaPI.size
Bio.Restriction.Restriction.UcoMSI.size
Bio.Restriction.Restriction.UnbI.size
Bio.Restriction.Restriction.Van91I.size
Bio.Restriction.Restriction.Vha464I.size
Bio.Restriction.Restriction.VneI.size
Bio.Restriction.Restriction.VpaK11AI.size
Bio.Restriction.Restriction.VpaK11BI.size
Bio.Restriction.Restriction.VspI.size
Bio.Restriction.Restriction.WviI.size
Bio.Restriction.Restriction.XagI.size
Bio.Restriction.Restriction.XapI.size
Bio.Restriction.Restriction.XbaI.size
Bio.Restriction.Restriction.XceI.size
Bio.Restriction.Restriction.XcmI.size
Bio.Restriction.Restriction.XhoI.size
Bio.Restriction.Restriction.XhoII.size
Bio.Restriction.Restriction.XmaI.size
Bio.Restriction.Restriction.XmaIII.size
Bio.Restriction.Restriction.XmaJI.size
Bio.Restriction.Restriction.XmiI.size
Bio.Restriction.Restriction.XmnI.size
Bio.Restriction.Restriction.XspI.size
Bio.Restriction.Restriction.YkrI.size
Bio.Restriction.Restriction.ZraI.size
Bio.Restriction.Restriction.ZrmI.size
Bio.Restriction.Restriction.Zsp2I.size
Bio.codonalign.codonalphabet.CodonAlphabet.size" class="py-name" href="#" onclick="return doclink('link-270', 'size', 'link-270');">size</a></tt><tt class="py-op">(</tt><tt class="py-name">length_parts</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L719"></a><tt class="py-lineno"> 719</tt>  <tt class="py-line"> </tt>
<a name="EmblScanner._feed_header_lines"></a><div id="EmblScanner._feed_header_lines-def"><a name="L720"></a><tt class="py-lineno"> 720</tt> <a class="py-toggle" href="#" id="EmblScanner._feed_header_lines-toggle" onclick="return toggle('EmblScanner._feed_header_lines');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.EmblScanner-class.html#_feed_header_lines">_feed_header_lines</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">,</tt> <tt class="py-param">lines</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EmblScanner._feed_header_lines-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EmblScanner._feed_header_lines-expanded"><a name="L721"></a><tt class="py-lineno"> 721</tt>  <tt class="py-line">        <tt class="py-name">EMBL_INDENT</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-271" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-271', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt> </tt>
<a name="L722"></a><tt class="py-lineno"> 722</tt>  <tt class="py-line">        <tt class="py-name">EMBL_SPACER</tt> <tt class="py-op">=</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt class="py-name">EMBL_INDENT</tt> </tt>
<a name="L723"></a><tt class="py-lineno"> 723</tt>  <tt class="py-line">        <tt class="py-name">consumer_dict</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L724"></a><tt class="py-lineno"> 724</tt>  <tt class="py-line">            <tt class="py-string">'AC'</tt><tt class="py-op">:</tt> <tt class="py-string">'accession'</tt><tt class="py-op">,</tt> </tt>
<a name="L725"></a><tt class="py-lineno"> 725</tt>  <tt class="py-line">            <tt class="py-string">'SV'</tt><tt class="py-op">:</tt> <tt class="py-string">'version'</tt><tt class="py-op">,</tt>  <tt class="py-comment"># SV line removed in June 2006, now part of ID line</tt> </tt>
<a name="L726"></a><tt class="py-lineno"> 726</tt>  <tt class="py-line">            <tt class="py-string">'DE'</tt><tt class="py-op">:</tt> <tt class="py-string">'definition'</tt><tt class="py-op">,</tt> </tt>
<a name="L727"></a><tt class="py-lineno"> 727</tt>  <tt class="py-line">            <tt class="py-comment"># 'RN' : 'reference_num',</tt> </tt>
<a name="L728"></a><tt class="py-lineno"> 728</tt>  <tt class="py-line">            <tt class="py-comment"># 'RC' : reference comment... TODO</tt> </tt>
<a name="L729"></a><tt class="py-lineno"> 729</tt>  <tt class="py-line">            <tt class="py-comment"># 'RP' : 'reference_bases',</tt> </tt>
<a name="L730"></a><tt class="py-lineno"> 730</tt>  <tt class="py-line">            <tt class="py-comment"># 'RX' : reference cross reference... DOI or Pubmed</tt> </tt>
<a name="L731"></a><tt class="py-lineno"> 731</tt>  <tt class="py-line">            <tt class="py-string">'RG'</tt><tt class="py-op">:</tt> <tt class="py-string">'consrtm'</tt><tt class="py-op">,</tt>  <tt class="py-comment"># optional consortium</tt> </tt>
<a name="L732"></a><tt class="py-lineno"> 732</tt>  <tt class="py-line">            <tt class="py-comment"># 'RA' : 'authors',</tt> </tt>
<a name="L733"></a><tt class="py-lineno"> 733</tt>  <tt class="py-line">            <tt class="py-comment"># 'RT' : 'title',</tt> </tt>
<a name="L734"></a><tt class="py-lineno"> 734</tt>  <tt class="py-line">            <tt class="py-string">'RL'</tt><tt class="py-op">:</tt> <tt class="py-string">'journal'</tt><tt class="py-op">,</tt> </tt>
<a name="L735"></a><tt class="py-lineno"> 735</tt>  <tt class="py-line">            <tt class="py-string">'OS'</tt><tt class="py-op">:</tt> <tt class="py-string">'organism'</tt><tt class="py-op">,</tt> </tt>
<a name="L736"></a><tt class="py-lineno"> 736</tt>  <tt class="py-line">            <tt class="py-string">'OC'</tt><tt class="py-op">:</tt> <tt class="py-string">'taxonomy'</tt><tt class="py-op">,</tt> </tt>
<a name="L737"></a><tt class="py-lineno"> 737</tt>  <tt class="py-line">            <tt class="py-comment"># 'DR' : data reference</tt> </tt>
<a name="L738"></a><tt class="py-lineno"> 738</tt>  <tt class="py-line">            <tt class="py-string">'CC'</tt><tt class="py-op">:</tt> <tt class="py-string">'comment'</tt><tt class="py-op">,</tt> </tt>
<a name="L739"></a><tt class="py-lineno"> 739</tt>  <tt class="py-line">            <tt class="py-comment"># 'XX' : splitter</tt> </tt>
<a name="L740"></a><tt class="py-lineno"> 740</tt>  <tt class="py-line">        <tt class="py-op">}</tt> </tt>
<a name="L741"></a><tt class="py-lineno"> 741</tt>  <tt class="py-line">        <tt class="py-comment"># We have to handle the following specially:</tt> </tt>
<a name="L742"></a><tt class="py-lineno"> 742</tt>  <tt class="py-line">        <tt class="py-comment"># RX (depending on reference type...)</tt> </tt>
<a name="L743"></a><tt class="py-lineno"> 743</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">lines</tt><tt class="py-op">:</tt> </tt>
<a name="L744"></a><tt class="py-lineno"> 744</tt>  <tt class="py-line">            <tt class="py-name">line_type</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">EMBL_INDENT</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-272" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-272', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L745"></a><tt class="py-lineno"> 745</tt>  <tt class="py-line">            <tt id="link-273" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-273', 'data', 'link-143');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">EMBL_INDENT</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-274" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-274', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L746"></a><tt class="py-lineno"> 746</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line_type</tt> <tt class="py-op">==</tt> <tt class="py-string">'XX'</tt><tt class="py-op">:</tt> </tt>
<a name="L747"></a><tt class="py-lineno"> 747</tt>  <tt class="py-line">                <tt class="py-keyword">pass</tt> </tt>
<a name="L748"></a><tt class="py-lineno"> 748</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line_type</tt> <tt class="py-op">==</tt> <tt class="py-string">'RN'</tt><tt class="py-op">:</tt> </tt>
<a name="L749"></a><tt class="py-lineno"> 749</tt>  <tt class="py-line">                <tt class="py-comment"># Reformat reference numbers for the GenBank based consumer</tt> </tt>
<a name="L750"></a><tt class="py-lineno"> 750</tt>  <tt class="py-line">                <tt class="py-comment"># e.g. '[1]' becomes '1'</tt> </tt>
<a name="L751"></a><tt class="py-lineno"> 751</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-275" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-275', 'data', 'link-143');">data</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"["</tt> <tt class="py-keyword">and</tt> <tt id="link-276" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-276', 'data', 'link-143');">data</a></tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"]"</tt><tt class="py-op">:</tt> </tt>
<a name="L752"></a><tt class="py-lineno"> 752</tt>  <tt class="py-line">                    <tt id="link-277" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-277', 'data', 'link-143');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-278" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-278', 'data', 'link-143');">data</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L753"></a><tt class="py-lineno"> 753</tt>  <tt class="py-line">                <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-279" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.reference_num()=Bio.GenBank._FeatureConsumer-class.html#reference_num,Method Bio.GenBank._RecordConsumer.reference_num()=Bio.GenBank._RecordConsumer-class.html#reference_num"><a title="Bio.GenBank._FeatureConsumer.reference_num
Bio.GenBank._RecordConsumer.reference_num" class="py-name" href="#" onclick="return doclink('link-279', 'reference_num', 'link-279');">reference_num</a></tt><tt class="py-op">(</tt><tt id="link-280" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-280', 'data', 'link-143');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L754"></a><tt class="py-lineno"> 754</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line_type</tt> <tt class="py-op">==</tt> <tt class="py-string">'RP'</tt><tt class="py-op">:</tt> </tt>
<a name="L755"></a><tt class="py-lineno"> 755</tt>  <tt class="py-line">                <tt class="py-comment"># Reformat reference numbers for the GenBank based consumer</tt> </tt>
<a name="L756"></a><tt class="py-lineno"> 756</tt>  <tt class="py-line">                <tt class="py-comment"># e.g. '1-4639675' becomes '(bases 1 to 4639675)'</tt> </tt>
<a name="L757"></a><tt class="py-lineno"> 757</tt>  <tt class="py-line">                <tt class="py-comment"># and '160-550, 904-1055' becomes '(bases 160 to 550; 904 to 1055)'</tt> </tt>
<a name="L758"></a><tt class="py-lineno"> 758</tt>  <tt class="py-line">                <tt class="py-comment"># Note could be multi-line, and end with a comma</tt> </tt>
<a name="L759"></a><tt class="py-lineno"> 759</tt>  <tt class="py-line">                <tt id="link-281" class="py-name" targets="Variable Bio.SeqFeature.FeatureLocation.parts=Bio.SeqFeature.FeatureLocation-class.html#parts"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-281', 'parts', 'link-281');">parts</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">bases</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"-"</tt><tt class="py-op">,</tt> <tt class="py-string">" to "</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-282" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-282', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">bases</tt> <tt class="py-keyword">in</tt> <tt id="link-283" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-283', 'data', 'link-143');">data</a></tt><tt class="py-op">.</tt><tt id="link-284" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-284', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">","</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt class="py-name">bases</tt><tt class="py-op">.</tt><tt id="link-285" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-285', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L760"></a><tt class="py-lineno"> 760</tt>  <tt class="py-line">                <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-286" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.reference_bases()=Bio.GenBank._FeatureConsumer-class.html#reference_bases,Method Bio.GenBank._RecordConsumer.reference_bases()=Bio.GenBank._RecordConsumer-class.html#reference_bases"><a title="Bio.GenBank._FeatureConsumer.reference_bases
Bio.GenBank._RecordConsumer.reference_bases" class="py-name" href="#" onclick="return doclink('link-286', 'reference_bases', 'link-286');">reference_bases</a></tt><tt class="py-op">(</tt><tt class="py-string">"(bases %s)"</tt> <tt class="py-op">%</tt> <tt class="py-string">"; "</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt id="link-287" class="py-name"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-287', 'parts', 'link-281');">parts</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L761"></a><tt class="py-lineno"> 761</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line_type</tt> <tt class="py-op">==</tt> <tt class="py-string">'RT'</tt><tt class="py-op">:</tt> </tt>
<a name="L762"></a><tt class="py-lineno"> 762</tt>  <tt class="py-line">                <tt class="py-comment"># Remove the enclosing quotes and trailing semi colon.</tt> </tt>
<a name="L763"></a><tt class="py-lineno"> 763</tt>  <tt class="py-line">                <tt class="py-comment"># Note the title can be split over multiple lines.</tt> </tt>
<a name="L764"></a><tt class="py-lineno"> 764</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-288" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-288', 'data', 'link-143');">data</a></tt><tt class="py-op">.</tt><tt id="link-289" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-289', 'startswith', 'link-97');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'"'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L765"></a><tt class="py-lineno"> 765</tt>  <tt class="py-line">                    <tt id="link-290" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-290', 'data', 'link-143');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-291" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-291', 'data', 'link-143');">data</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L766"></a><tt class="py-lineno"> 766</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-292" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-292', 'data', 'link-143');">data</a></tt><tt class="py-op">.</tt><tt id="link-293" class="py-name"><a title="Bio.Seq.Seq.endswith" class="py-name" href="#" onclick="return doclink('link-293', 'endswith', 'link-229');">endswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'";'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L767"></a><tt class="py-lineno"> 767</tt>  <tt class="py-line">                    <tt id="link-294" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-294', 'data', 'link-143');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-295" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-295', 'data', 'link-143');">data</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> </tt>
<a name="L768"></a><tt class="py-lineno"> 768</tt>  <tt class="py-line">                <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-296" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.title()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#title,Method Bio.GenBank._FeatureConsumer.title()=Bio.GenBank._FeatureConsumer-class.html#title,Method Bio.GenBank._RecordConsumer.title()=Bio.GenBank._RecordConsumer-class.html#title"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-296', 'title', 'link-296');">title</a></tt><tt class="py-op">(</tt><tt id="link-297" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-297', 'data', 'link-143');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L769"></a><tt class="py-lineno"> 769</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line_type</tt> <tt class="py-op">==</tt> <tt class="py-string">'RX'</tt><tt class="py-op">:</tt> </tt>
<a name="L770"></a><tt class="py-lineno"> 770</tt>  <tt class="py-line">                <tt class="py-comment"># EMBL support three reference types at the moment:</tt> </tt>
<a name="L771"></a><tt class="py-lineno"> 771</tt>  <tt class="py-line">                <tt class="py-comment"># - PUBMED    PUBMED bibliographic database (NLM)</tt> </tt>
<a name="L772"></a><tt class="py-lineno"> 772</tt>  <tt class="py-line">                <tt class="py-comment"># - DOI       Digital Object Identifier (International DOI Foundation)</tt> </tt>
<a name="L773"></a><tt class="py-lineno"> 773</tt>  <tt class="py-line">                <tt class="py-comment"># - AGRICOLA  US National Agriculture Library (NAL) of the US Department</tt> </tt>
<a name="L774"></a><tt class="py-lineno"> 774</tt>  <tt class="py-line">                <tt class="py-comment">#             of Agriculture (USDA)</tt> </tt>
<a name="L775"></a><tt class="py-lineno"> 775</tt>  <tt class="py-line">                <tt class="py-comment">#</tt> </tt>
<a name="L776"></a><tt class="py-lineno"> 776</tt>  <tt class="py-line">                <tt class="py-comment"># Format:</tt> </tt>
<a name="L777"></a><tt class="py-lineno"> 777</tt>  <tt class="py-line">                <tt class="py-comment"># RX  resource_identifier; identifier.</tt> </tt>
<a name="L778"></a><tt class="py-lineno"> 778</tt>  <tt class="py-line">                <tt class="py-comment">#</tt> </tt>
<a name="L779"></a><tt class="py-lineno"> 779</tt>  <tt class="py-line">                <tt class="py-comment"># e.g.</tt> </tt>
<a name="L780"></a><tt class="py-lineno"> 780</tt>  <tt class="py-line">                <tt class="py-comment"># RX   DOI; 10.1016/0024-3205(83)90010-3.</tt> </tt>
<a name="L781"></a><tt class="py-lineno"> 781</tt>  <tt class="py-line">                <tt class="py-comment"># RX   PUBMED; 264242.</tt> </tt>
<a name="L782"></a><tt class="py-lineno"> 782</tt>  <tt class="py-line">                <tt class="py-comment">#</tt> </tt>
<a name="L783"></a><tt class="py-lineno"> 783</tt>  <tt class="py-line">                <tt class="py-comment"># Currently our reference object only supports PUBMED and MEDLINE</tt> </tt>
<a name="L784"></a><tt class="py-lineno"> 784</tt>  <tt class="py-line">                <tt class="py-comment"># (as these were in GenBank files?).</tt> </tt>
<a name="L785"></a><tt class="py-lineno"> 785</tt>  <tt class="py-line">                <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt id="link-298" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-298', 'value', 'link-102');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-299" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-299', 'data', 'link-143');">data</a></tt><tt class="py-op">.</tt><tt id="link-300" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-300', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">";"</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L786"></a><tt class="py-lineno"> 786</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-301" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-301', 'value', 'link-102');">value</a></tt><tt class="py-op">.</tt><tt id="link-302" class="py-name"><a title="Bio.Seq.Seq.endswith" class="py-name" href="#" onclick="return doclink('link-302', 'endswith', 'link-229');">endswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"."</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L787"></a><tt class="py-lineno"> 787</tt>  <tt class="py-line">                    <tt id="link-303" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-303', 'value', 'link-102');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-304" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-304', 'value', 'link-102');">value</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L788"></a><tt class="py-lineno"> 788</tt>  <tt class="py-line">                <tt id="link-305" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-305', 'value', 'link-102');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-306" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-306', 'value', 'link-102');">value</a></tt><tt class="py-op">.</tt><tt id="link-307" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-307', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L789"></a><tt class="py-lineno"> 789</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-op">==</tt> <tt class="py-string">"PUBMED"</tt><tt class="py-op">:</tt> </tt>
<a name="L790"></a><tt class="py-lineno"> 790</tt>  <tt class="py-line">                    <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-308" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.pubmed_id()=Bio.GenBank._FeatureConsumer-class.html#pubmed_id,Method Bio.GenBank._RecordConsumer.pubmed_id()=Bio.GenBank._RecordConsumer-class.html#pubmed_id"><a title="Bio.GenBank._FeatureConsumer.pubmed_id
Bio.GenBank._RecordConsumer.pubmed_id" class="py-name" href="#" onclick="return doclink('link-308', 'pubmed_id', 'link-308');">pubmed_id</a></tt><tt class="py-op">(</tt><tt id="link-309" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-309', 'value', 'link-102');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L791"></a><tt class="py-lineno"> 791</tt>  <tt class="py-line">                <tt class="py-comment"># TODO - Handle other reference types (here and in BioSQL bindings)</tt> </tt>
<a name="L792"></a><tt class="py-lineno"> 792</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line_type</tt> <tt class="py-op">==</tt> <tt class="py-string">'CC'</tt><tt class="py-op">:</tt> </tt>
<a name="L793"></a><tt class="py-lineno"> 793</tt>  <tt class="py-line">                <tt class="py-comment"># Have to pass a list of strings for this one (not just a string)</tt> </tt>
<a name="L794"></a><tt class="py-lineno"> 794</tt>  <tt class="py-line">                <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-310" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.comment()=Bio.GenBank._FeatureConsumer-class.html#comment,Method Bio.GenBank._RecordConsumer.comment()=Bio.GenBank._RecordConsumer-class.html#comment"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-310', 'comment', 'link-310');">comment</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt id="link-311" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-311', 'data', 'link-143');">data</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L795"></a><tt class="py-lineno"> 795</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line_type</tt> <tt class="py-op">==</tt> <tt class="py-string">'DR'</tt><tt class="py-op">:</tt> </tt>
<a name="L796"></a><tt class="py-lineno"> 796</tt>  <tt class="py-line">                <tt class="py-comment"># Database Cross-reference, format:</tt> </tt>
<a name="L797"></a><tt class="py-lineno"> 797</tt>  <tt class="py-line">                <tt class="py-comment"># DR   database_identifier; primary_identifier; secondary_identifier.</tt> </tt>
<a name="L798"></a><tt class="py-lineno"> 798</tt>  <tt class="py-line">                <tt class="py-comment">#</tt> </tt>
<a name="L799"></a><tt class="py-lineno"> 799</tt>  <tt class="py-line">                <tt class="py-comment"># e.g.</tt> </tt>
<a name="L800"></a><tt class="py-lineno"> 800</tt>  <tt class="py-line">                <tt class="py-comment"># DR   MGI; 98599; Tcrb-V4.</tt> </tt>
<a name="L801"></a><tt class="py-lineno"> 801</tt>  <tt class="py-line">                <tt class="py-comment">#</tt> </tt>
<a name="L802"></a><tt class="py-lineno"> 802</tt>  <tt class="py-line">                <tt class="py-comment"># TODO - How should we store any secondary identifier?</tt> </tt>
<a name="L803"></a><tt class="py-lineno"> 803</tt>  <tt class="py-line">                <tt id="link-312" class="py-name"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-312', 'parts', 'link-281');">parts</a></tt> <tt class="py-op">=</tt> <tt id="link-313" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-313', 'data', 'link-143');">data</a></tt><tt class="py-op">.</tt><tt id="link-314" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-314', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-string">"."</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-315" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-315', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">";"</tt><tt class="py-op">)</tt> </tt>
<a name="L804"></a><tt class="py-lineno"> 804</tt>  <tt class="py-line">                <tt class="py-comment"># Turn it into "database_identifier:primary_identifier" to</tt> </tt>
<a name="L805"></a><tt class="py-lineno"> 805</tt>  <tt class="py-line">                <tt class="py-comment"># mimic the GenBank parser. e.g. "MGI:98599"</tt> </tt>
<a name="L806"></a><tt class="py-lineno"> 806</tt>  <tt class="py-line">                <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-316" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.dblink()=Bio.GenBank._FeatureConsumer-class.html#dblink,Method Bio.GenBank._RecordConsumer.dblink()=Bio.GenBank._RecordConsumer-class.html#dblink"><a title="Bio.GenBank._FeatureConsumer.dblink
Bio.GenBank._RecordConsumer.dblink" class="py-name" href="#" onclick="return doclink('link-316', 'dblink', 'link-316');">dblink</a></tt><tt class="py-op">(</tt><tt class="py-string">"%s:%s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-317" class="py-name"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-317', 'parts', 'link-281');">parts</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-318" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-318', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L807"></a><tt class="py-lineno"> 807</tt>  <tt class="py-line">                                           <tt id="link-319" class="py-name"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-319', 'parts', 'link-281');">parts</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-320" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-320', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L808"></a><tt class="py-lineno"> 808</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line_type</tt> <tt class="py-op">==</tt> <tt class="py-string">'RA'</tt><tt class="py-op">:</tt> </tt>
<a name="L809"></a><tt class="py-lineno"> 809</tt>  <tt class="py-line">                <tt class="py-comment"># Remove trailing ; at end of authors list</tt> </tt>
<a name="L810"></a><tt class="py-lineno"> 810</tt>  <tt class="py-line">                <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-321" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.authors()=Bio.GenBank._FeatureConsumer-class.html#authors,Method Bio.GenBank._RecordConsumer.authors()=Bio.GenBank._RecordConsumer-class.html#authors"><a title="Bio.GenBank._FeatureConsumer.authors
Bio.GenBank._RecordConsumer.authors" class="py-name" href="#" onclick="return doclink('link-321', 'authors', 'link-321');">authors</a></tt><tt class="py-op">(</tt><tt id="link-322" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-322', 'data', 'link-143');">data</a></tt><tt class="py-op">.</tt><tt id="link-323" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-323', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-string">";"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L811"></a><tt class="py-lineno"> 811</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line_type</tt> <tt class="py-op">==</tt> <tt class="py-string">'PR'</tt><tt class="py-op">:</tt> </tt>
<a name="L812"></a><tt class="py-lineno"> 812</tt>  <tt class="py-line">                <tt class="py-comment"># Remove trailing ; at end of the project reference</tt> </tt>
<a name="L813"></a><tt class="py-lineno"> 813</tt>  <tt class="py-line">                <tt class="py-comment"># In GenBank files this corresponds to the old PROJECT</tt> </tt>
<a name="L814"></a><tt class="py-lineno"> 814</tt>  <tt class="py-line">                <tt class="py-comment"># line which is being replaced with the DBLINK line.</tt> </tt>
<a name="L815"></a><tt class="py-lineno"> 815</tt>  <tt class="py-line">                <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-324" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.project()=Bio.GenBank._FeatureConsumer-class.html#project,Method Bio.GenBank._RecordConsumer.project()=Bio.GenBank._RecordConsumer-class.html#project"><a title="Bio.GenBank._FeatureConsumer.project
Bio.GenBank._RecordConsumer.project" class="py-name" href="#" onclick="return doclink('link-324', 'project', 'link-324');">project</a></tt><tt class="py-op">(</tt><tt id="link-325" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-325', 'data', 'link-143');">data</a></tt><tt class="py-op">.</tt><tt id="link-326" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-326', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-string">";"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L816"></a><tt class="py-lineno"> 816</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line_type</tt> <tt class="py-op">==</tt> <tt class="py-string">'KW'</tt><tt class="py-op">:</tt> </tt>
<a name="L817"></a><tt class="py-lineno"> 817</tt>  <tt class="py-line">                <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-327" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.keywords()=Bio.GenBank._FeatureConsumer-class.html#keywords,Method Bio.GenBank._RecordConsumer.keywords()=Bio.GenBank._RecordConsumer-class.html#keywords"><a title="Bio.GenBank._FeatureConsumer.keywords
Bio.GenBank._RecordConsumer.keywords" class="py-name" href="#" onclick="return doclink('link-327', 'keywords', 'link-327');">keywords</a></tt><tt class="py-op">(</tt><tt id="link-328" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-328', 'data', 'link-143');">data</a></tt><tt class="py-op">.</tt><tt id="link-329" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-329', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-string">";"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L818"></a><tt class="py-lineno"> 818</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line_type</tt> <tt class="py-keyword">in</tt> <tt class="py-name">consumer_dict</tt><tt class="py-op">:</tt> </tt>
<a name="L819"></a><tt class="py-lineno"> 819</tt>  <tt class="py-line">                <tt class="py-comment"># Its a semi-automatic entry!</tt> </tt>
<a name="L820"></a><tt class="py-lineno"> 820</tt>  <tt class="py-line">                <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">consumer</tt><tt class="py-op">,</tt> <tt class="py-name">consumer_dict</tt><tt class="py-op">[</tt><tt class="py-name">line_type</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">(</tt><tt id="link-330" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-330', 'data', 'link-143');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L821"></a><tt class="py-lineno"> 821</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L822"></a><tt class="py-lineno"> 822</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt><tt class="py-op">:</tt> </tt>
<a name="L823"></a><tt class="py-lineno"> 823</tt>  <tt class="py-line">                    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Ignoring EMBL header line:\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
</div><a name="L824"></a><tt class="py-lineno"> 824</tt>  <tt class="py-line"> </tt>
<a name="EmblScanner._feed_misc_lines"></a><div id="EmblScanner._feed_misc_lines-def"><a name="L825"></a><tt class="py-lineno"> 825</tt> <a class="py-toggle" href="#" id="EmblScanner._feed_misc_lines-toggle" onclick="return toggle('EmblScanner._feed_misc_lines');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.EmblScanner-class.html#_feed_misc_lines">_feed_misc_lines</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">,</tt> <tt class="py-param">lines</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EmblScanner._feed_misc_lines-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EmblScanner._feed_misc_lines-expanded"><a name="L826"></a><tt class="py-lineno"> 826</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Should we do something with the information on the SQ line(s)?</tt> </tt>
<a name="L827"></a><tt class="py-lineno"> 827</tt>  <tt class="py-line">        <tt class="py-name">lines</tt><tt class="py-op">.</tt><tt id="link-331" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-331', 'append', 'link-36');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">)</tt> </tt>
<a name="L828"></a><tt class="py-lineno"> 828</tt>  <tt class="py-line">        <tt class="py-name">line_iter</tt> <tt class="py-op">=</tt> <tt class="py-name">iter</tt><tt class="py-op">(</tt><tt class="py-name">lines</tt><tt class="py-op">)</tt> </tt>
<a name="L829"></a><tt class="py-lineno"> 829</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L830"></a><tt class="py-lineno"> 830</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">line_iter</tt><tt class="py-op">:</tt> </tt>
<a name="L831"></a><tt class="py-lineno"> 831</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-332" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-332', 'startswith', 'link-97');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"CO   "</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L832"></a><tt class="py-lineno"> 832</tt>  <tt class="py-line">                    <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">5</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-333" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-333', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L833"></a><tt class="py-lineno"> 833</tt>  <tt class="py-line">                    <tt id="link-334" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.contig_location()=Bio.GenBank._FeatureConsumer-class.html#contig_location,Method Bio.GenBank._RecordConsumer.contig_location()=Bio.GenBank._RecordConsumer-class.html#contig_location"><a title="Bio.GenBank._FeatureConsumer.contig_location
Bio.GenBank._RecordConsumer.contig_location" class="py-name" href="#" onclick="return doclink('link-334', 'contig_location', 'link-334');">contig_location</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> </tt>
<a name="L834"></a><tt class="py-lineno"> 834</tt>  <tt class="py-line">                    <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L835"></a><tt class="py-lineno"> 835</tt>  <tt class="py-line">                        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-335" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-335', 'next', 'link-86');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">line_iter</tt><tt class="py-op">)</tt> </tt>
<a name="L836"></a><tt class="py-lineno"> 836</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L837"></a><tt class="py-lineno"> 837</tt>  <tt class="py-line">                            <tt class="py-keyword">break</tt> </tt>
<a name="L838"></a><tt class="py-lineno"> 838</tt>  <tt class="py-line">                        <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-336" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-336', 'startswith', 'link-97');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"CO   "</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L839"></a><tt class="py-lineno"> 839</tt>  <tt class="py-line">                            <tt class="py-comment"># Don't need to preseve the whitespace here.</tt> </tt>
<a name="L840"></a><tt class="py-lineno"> 840</tt>  <tt class="py-line">                            <tt id="link-337" class="py-name"><a title="Bio.GenBank._FeatureConsumer.contig_location
Bio.GenBank._RecordConsumer.contig_location" class="py-name" href="#" onclick="return doclink('link-337', 'contig_location', 'link-334');">contig_location</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">5</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-338" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-338', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L841"></a><tt class="py-lineno"> 841</tt>  <tt class="py-line">                        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L842"></a><tt class="py-lineno"> 842</tt>  <tt class="py-line">                            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">'Expected CO (contig) continuation line, got:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L843"></a><tt class="py-lineno"> 843</tt>  <tt class="py-line">                    <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-339" class="py-name"><a title="Bio.GenBank._FeatureConsumer.contig_location
Bio.GenBank._RecordConsumer.contig_location" class="py-name" href="#" onclick="return doclink('link-339', 'contig_location', 'link-334');">contig_location</a></tt><tt class="py-op">(</tt><tt id="link-340" class="py-name"><a title="Bio.GenBank._FeatureConsumer.contig_location
Bio.GenBank._RecordConsumer.contig_location" class="py-name" href="#" onclick="return doclink('link-340', 'contig_location', 'link-334');">contig_location</a></tt><tt class="py-op">)</tt> </tt>
<a name="L844"></a><tt class="py-lineno"> 844</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-341" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-341', 'startswith', 'link-97');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"SQ   Sequence "</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L845"></a><tt class="py-lineno"> 845</tt>  <tt class="py-line">                    <tt class="py-comment"># e.g.</tt> </tt>
<a name="L846"></a><tt class="py-lineno"> 846</tt>  <tt class="py-line">                    <tt class="py-comment"># SQ   Sequence 219 BP; 82 A; 48 C; 33 G; 45 T; 11 other;</tt> </tt>
<a name="L847"></a><tt class="py-lineno"> 847</tt>  <tt class="py-line">                    <tt class="py-comment">#</tt> </tt>
<a name="L848"></a><tt class="py-lineno"> 848</tt>  <tt class="py-line">                    <tt class="py-comment"># Or, EMBL-bank patent, e.g.</tt> </tt>
<a name="L849"></a><tt class="py-lineno"> 849</tt>  <tt class="py-line">                    <tt class="py-comment"># SQ   Sequence 465 AA; 3963407aa91d3a0d622fec679a4524e0; MD5;</tt> </tt>
<a name="L850"></a><tt class="py-lineno"> 850</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-342" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner._feed_seq_length" class="py-name" href="#" onclick="return doclink('link-342', '_feed_seq_length', 'link-252');">_feed_seq_length</a></tt><tt class="py-op">(</tt><tt class="py-name">consumer</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">14</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-343" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-343', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-344" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-344', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-string">";"</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-345" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-345', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">";"</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L851"></a><tt class="py-lineno"> 851</tt>  <tt class="py-line">                    <tt class="py-comment"># TODO - Record the checksum etc?</tt> </tt>
<a name="L852"></a><tt class="py-lineno"> 852</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L853"></a><tt class="py-lineno"> 853</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L854"></a><tt class="py-lineno"> 854</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Problem in misc lines before sequence"</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L855"></a><tt class="py-lineno"> 855</tt>  <tt class="py-line"> </tt>
<a name="L856"></a><tt class="py-lineno"> 856</tt>  <tt class="py-line"> </tt>
<a name="_ImgtScanner"></a><div id="_ImgtScanner-def"><a name="L857"></a><tt class="py-lineno"> 857</tt> <a class="py-toggle" href="#" id="_ImgtScanner-toggle" onclick="return toggle('_ImgtScanner');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.GenBank.Scanner._ImgtScanner-class.html">_ImgtScanner</a><tt class="py-op">(</tt><tt class="py-base-class">EmblScanner</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ImgtScanner-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_ImgtScanner-expanded"><a name="L858"></a><tt class="py-lineno"> 858</tt>  <tt class="py-line">    <tt class="py-docstring">"""For extracting chunks of information in IMGT (EMBL like) files (PRIVATE).</tt> </tt>
<a name="L859"></a><tt class="py-lineno"> 859</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L860"></a><tt class="py-lineno"> 860</tt>  <tt class="py-line"><tt class="py-docstring">    IMGT files are like EMBL files but in order to allow longer feature types</tt> </tt>
<a name="L861"></a><tt class="py-lineno"> 861</tt>  <tt class="py-line"><tt class="py-docstring">    the features should be indented by 25 characters not 21 characters. In</tt> </tt>
<a name="L862"></a><tt class="py-lineno"> 862</tt>  <tt class="py-line"><tt class="py-docstring">    practice the IMGT flat files tend to use either 21 or 25 characters, so we</tt> </tt>
<a name="L863"></a><tt class="py-lineno"> 863</tt>  <tt class="py-line"><tt class="py-docstring">    must cope with both.</tt> </tt>
<a name="L864"></a><tt class="py-lineno"> 864</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L865"></a><tt class="py-lineno"> 865</tt>  <tt class="py-line"><tt class="py-docstring">    This is private to encourage use of Bio.SeqIO rather than Bio.GenBank.</tt> </tt>
<a name="L866"></a><tt class="py-lineno"> 866</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L867"></a><tt class="py-lineno"> 867</tt>  <tt class="py-line"> </tt>
<a name="L868"></a><tt class="py-lineno"> 868</tt>  <tt class="py-line">    <tt id="link-346" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner.GenBankScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner.InsdcScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner._ImgtScanner.FEATURE_START_MARKERS" class="py-name" href="#" onclick="return doclink('link-346', 'FEATURE_START_MARKERS', 'link-13');">FEATURE_START_MARKERS</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">"FH   Key             Location/Qualifiers"</tt><tt class="py-op">,</tt> </tt>
<a name="L869"></a><tt class="py-lineno"> 869</tt>  <tt class="py-line">                             <tt class="py-string">"FH   Key             Location/Qualifiers (from EMBL)"</tt><tt class="py-op">,</tt> </tt>
<a name="L870"></a><tt class="py-lineno"> 870</tt>  <tt class="py-line">                             <tt class="py-string">"FH   Key                 Location/Qualifiers"</tt><tt class="py-op">,</tt> </tt>
<a name="L871"></a><tt class="py-lineno"> 871</tt>  <tt class="py-line">                             <tt class="py-string">"FH"</tt><tt class="py-op">]</tt> </tt>
<a name="L872"></a><tt class="py-lineno"> 872</tt>  <tt class="py-line"> </tt>
<a name="_ImgtScanner.parse_features"></a><div id="_ImgtScanner.parse_features-def"><a name="L873"></a><tt class="py-lineno"> 873</tt> <a class="py-toggle" href="#" id="_ImgtScanner.parse_features-toggle" onclick="return toggle('_ImgtScanner.parse_features');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner._ImgtScanner-class.html#parse_features">parse_features</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">skip</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ImgtScanner.parse_features-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ImgtScanner.parse_features-expanded"><a name="L874"></a><tt class="py-lineno"> 874</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return list of tuples for the features (if present)</tt> </tt>
<a name="L875"></a><tt class="py-lineno"> 875</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L876"></a><tt class="py-lineno"> 876</tt>  <tt class="py-line"><tt class="py-docstring">        Each feature is returned as a tuple (key, location, qualifiers)</tt> </tt>
<a name="L877"></a><tt class="py-lineno"> 877</tt>  <tt class="py-line"><tt class="py-docstring">        where key and location are strings (e.g. "CDS" and</tt> </tt>
<a name="L878"></a><tt class="py-lineno"> 878</tt>  <tt class="py-line"><tt class="py-docstring">        "complement(join(490883..490885,1..879))") while qualifiers</tt> </tt>
<a name="L879"></a><tt class="py-lineno"> 879</tt>  <tt class="py-line"><tt class="py-docstring">        is a list of two string tuples (feature qualifier keys and values).</tt> </tt>
<a name="L880"></a><tt class="py-lineno"> 880</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L881"></a><tt class="py-lineno"> 881</tt>  <tt class="py-line"><tt class="py-docstring">        Assumes you have already read to the start of the features table.</tt> </tt>
<a name="L882"></a><tt class="py-lineno"> 882</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L883"></a><tt class="py-lineno"> 883</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-347" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-347', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-348" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner.GenBankScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner.InsdcScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner._ImgtScanner.FEATURE_START_MARKERS" class="py-name" href="#" onclick="return doclink('link-348', 'FEATURE_START_MARKERS', 'link-13');">FEATURE_START_MARKERS</a></tt><tt class="py-op">:</tt> </tt>
<a name="L884"></a><tt class="py-lineno"> 884</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt><tt class="py-op">:</tt> </tt>
<a name="L885"></a><tt class="py-lineno"> 885</tt>  <tt class="py-line">                <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Didn't find any feature table"</tt><tt class="py-op">)</tt> </tt>
<a name="L886"></a><tt class="py-lineno"> 886</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L887"></a><tt class="py-lineno"> 887</tt>  <tt class="py-line"> </tt>
<a name="L888"></a><tt class="py-lineno"> 888</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-349" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-349', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-350" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner.GenBankScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner.InsdcScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner._ImgtScanner.FEATURE_START_MARKERS" class="py-name" href="#" onclick="return doclink('link-350', 'FEATURE_START_MARKERS', 'link-13');">FEATURE_START_MARKERS</a></tt><tt class="py-op">:</tt> </tt>
<a name="L889"></a><tt class="py-lineno"> 889</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-351" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-351', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L890"></a><tt class="py-lineno"> 890</tt>  <tt class="py-line"> </tt>
<a name="L891"></a><tt class="py-lineno"> 891</tt>  <tt class="py-line">        <tt class="py-name">bad_position_re</tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'([0-9]+)&gt;{1}'</tt><tt class="py-op">)</tt> </tt>
<a name="L892"></a><tt class="py-lineno"> 892</tt>  <tt class="py-line"> </tt>
<a name="L893"></a><tt class="py-lineno"> 893</tt>  <tt class="py-line">        <tt id="link-352" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-352', 'features', 'link-42');">features</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L894"></a><tt class="py-lineno"> 894</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> </tt>
<a name="L895"></a><tt class="py-lineno"> 895</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L896"></a><tt class="py-lineno"> 896</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L897"></a><tt class="py-lineno"> 897</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Premature end of line during features table"</tt><tt class="py-op">)</tt> </tt>
<a name="L898"></a><tt class="py-lineno"> 898</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-353" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-353', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-354" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-354', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-355" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.GenBankScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.InsdcScanner.SEQUENCE_HEADERS" class="py-name" href="#" onclick="return doclink('link-355', 'SEQUENCE_HEADERS', 'link-17');">SEQUENCE_HEADERS</a></tt><tt class="py-op">:</tt> </tt>
<a name="L899"></a><tt class="py-lineno"> 899</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt><tt class="py-op">:</tt> </tt>
<a name="L900"></a><tt class="py-lineno"> 900</tt>  <tt class="py-line">                    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Found start of sequence"</tt><tt class="py-op">)</tt> </tt>
<a name="L901"></a><tt class="py-lineno"> 901</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L902"></a><tt class="py-lineno"> 902</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-356" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-356', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L903"></a><tt class="py-lineno"> 903</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt> <tt class="py-op">==</tt> <tt class="py-string">"//"</tt><tt class="py-op">:</tt> </tt>
<a name="L904"></a><tt class="py-lineno"> 904</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Premature end of features table, marker '//' found"</tt><tt class="py-op">)</tt> </tt>
<a name="L905"></a><tt class="py-lineno"> 905</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-357" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.FEATURE_END_MARKERS
Bio.GenBank.Scanner.GenBankScanner.FEATURE_END_MARKERS
Bio.GenBank.Scanner.InsdcScanner.FEATURE_END_MARKERS" class="py-name" href="#" onclick="return doclink('link-357', 'FEATURE_END_MARKERS', 'link-14');">FEATURE_END_MARKERS</a></tt><tt class="py-op">:</tt> </tt>
<a name="L906"></a><tt class="py-lineno"> 906</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt><tt class="py-op">:</tt> </tt>
<a name="L907"></a><tt class="py-lineno"> 907</tt>  <tt class="py-line">                    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Found end of features"</tt><tt class="py-op">)</tt> </tt>
<a name="L908"></a><tt class="py-lineno"> 908</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-358" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-358', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L909"></a><tt class="py-lineno"> 909</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L910"></a><tt class="py-lineno"> 910</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-359" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-359', 'FEATURE_QUALIFIER_INDENT', 'link-15');">FEATURE_QUALIFIER_INDENT</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-360" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-360', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">""</tt><tt class="py-op">:</tt> </tt>
<a name="L911"></a><tt class="py-lineno"> 911</tt>  <tt class="py-line">                <tt class="py-comment"># This is an empty feature line between qualifiers. Empty</tt> </tt>
<a name="L912"></a><tt class="py-lineno"> 912</tt>  <tt class="py-line">                <tt class="py-comment"># feature lines within qualifiers are handled below (ignored).</tt> </tt>
<a name="L913"></a><tt class="py-lineno"> 913</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-361" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-361', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L914"></a><tt class="py-lineno"> 914</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L915"></a><tt class="py-lineno"> 915</tt>  <tt class="py-line"> </tt>
<a name="L916"></a><tt class="py-lineno"> 916</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">skip</tt><tt class="py-op">:</tt> </tt>
<a name="L917"></a><tt class="py-lineno"> 917</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-362" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-362', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L918"></a><tt class="py-lineno"> 918</tt>  <tt class="py-line">                <tt class="py-keyword">while</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-363" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-363', 'FEATURE_QUALIFIER_INDENT', 'link-15');">FEATURE_QUALIFIER_INDENT</a></tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-364" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_SPACER" class="py-name" href="#" onclick="return doclink('link-364', 'FEATURE_QUALIFIER_SPACER', 'link-16');">FEATURE_QUALIFIER_SPACER</a></tt><tt class="py-op">:</tt> </tt>
<a name="L919"></a><tt class="py-lineno"> 919</tt>  <tt class="py-line">                    <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-365" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-365', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L920"></a><tt class="py-lineno"> 920</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L921"></a><tt class="py-lineno"> 921</tt>  <tt class="py-line">                <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"FT"</tt> </tt>
<a name="L922"></a><tt class="py-lineno"> 922</tt>  <tt class="py-line">                <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L923"></a><tt class="py-lineno"> 923</tt>  <tt class="py-line">                    <tt id="link-366" class="py-name"><a title="Bio.GenBank._FeatureConsumer.feature_key
Bio.GenBank._RecordConsumer.feature_key" class="py-name" href="#" onclick="return doclink('link-366', 'feature_key', 'link-61');">feature_key</a></tt><tt class="py-op">,</tt> <tt class="py-name">location_start</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-367" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-367', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-368" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-368', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L924"></a><tt class="py-lineno"> 924</tt>  <tt class="py-line">                <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L925"></a><tt class="py-lineno"> 925</tt>  <tt class="py-line">                    <tt class="py-comment"># e.g. "FT   TRANSMEMBRANE-REGION2163..2240\n"</tt> </tt>
<a name="L926"></a><tt class="py-lineno"> 926</tt>  <tt class="py-line">                    <tt class="py-comment"># Assume indent of 25 as per IMGT spec, with the location</tt> </tt>
<a name="L927"></a><tt class="py-lineno"> 927</tt>  <tt class="py-line">                    <tt class="py-comment"># start in column 26 (one-based).</tt> </tt>
<a name="L928"></a><tt class="py-lineno"> 928</tt>  <tt class="py-line">                    <tt id="link-369" class="py-name"><a title="Bio.GenBank._FeatureConsumer.feature_key
Bio.GenBank._RecordConsumer.feature_key" class="py-name" href="#" onclick="return doclink('link-369', 'feature_key', 'link-61');">feature_key</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">:</tt><tt class="py-number">25</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-370" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-370', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L929"></a><tt class="py-lineno"> 929</tt>  <tt class="py-line">                    <tt class="py-name">location_start</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">25</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-371" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-371', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L930"></a><tt class="py-lineno"> 930</tt>  <tt class="py-line">                <tt class="py-name">feature_lines</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">location_start</tt><tt class="py-op">]</tt> </tt>
<a name="L931"></a><tt class="py-lineno"> 931</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-372" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-372', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L932"></a><tt class="py-lineno"> 932</tt>  <tt class="py-line">                <tt class="py-keyword">while</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-373" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-373', 'FEATURE_QUALIFIER_INDENT', 'link-15');">FEATURE_QUALIFIER_INDENT</a></tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-374" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_SPACER" class="py-name" href="#" onclick="return doclink('link-374', 'FEATURE_QUALIFIER_SPACER', 'link-16');">FEATURE_QUALIFIER_SPACER</a></tt> \ </tt>
<a name="L933"></a><tt class="py-lineno"> 933</tt>  <tt class="py-line">                        <tt class="py-keyword">or</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-375" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-375', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">""</tt><tt class="py-op">:</tt>  <tt class="py-comment"># cope with blank lines in the midst of a feature</tt> </tt>
<a name="L934"></a><tt class="py-lineno"> 934</tt>  <tt class="py-line">                    <tt class="py-comment"># Use strip to remove any harmless trailing white space AND and leading</tt> </tt>
<a name="L935"></a><tt class="py-lineno"> 935</tt>  <tt class="py-line">                    <tt class="py-comment"># white space (copes with 21 or 26 indents and orther variants)</tt> </tt>
<a name="L936"></a><tt class="py-lineno"> 936</tt>  <tt class="py-line">                    <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"FT"</tt> </tt>
<a name="L937"></a><tt class="py-lineno"> 937</tt>  <tt class="py-line">                    <tt class="py-name">feature_lines</tt><tt class="py-op">.</tt><tt id="link-376" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-376', 'append', 'link-36');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-377" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-377', 'FEATURE_QUALIFIER_INDENT', 'link-15');">FEATURE_QUALIFIER_INDENT</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-378" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-378', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L938"></a><tt class="py-lineno"> 938</tt>  <tt class="py-line">                    <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-379" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-379', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L939"></a><tt class="py-lineno"> 939</tt>  <tt class="py-line">                <tt id="link-380" class="py-name"><a title="Bio.GenBank._FeatureConsumer.feature_key
Bio.GenBank._RecordConsumer.feature_key" class="py-name" href="#" onclick="return doclink('link-380', 'feature_key', 'link-61');">feature_key</a></tt><tt class="py-op">,</tt> <tt id="link-381" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-381', 'location', 'link-138');">location</a></tt><tt class="py-op">,</tt> <tt class="py-name">qualifiers</tt> <tt class="py-op">=</tt> \ </tt>
<a name="L940"></a><tt class="py-lineno"> 940</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-382" class="py-name"><a title="Bio.GenBank.Scanner.InsdcScanner.parse_feature" class="py-name" href="#" onclick="return doclink('link-382', 'parse_feature', 'link-80');">parse_feature</a></tt><tt class="py-op">(</tt><tt id="link-383" class="py-name"><a title="Bio.GenBank._FeatureConsumer.feature_key
Bio.GenBank._RecordConsumer.feature_key" class="py-name" href="#" onclick="return doclink('link-383', 'feature_key', 'link-61');">feature_key</a></tt><tt class="py-op">,</tt> <tt class="py-name">feature_lines</tt><tt class="py-op">)</tt> </tt>
<a name="L941"></a><tt class="py-lineno"> 941</tt>  <tt class="py-line">                <tt class="py-comment"># Try to handle known problems with IMGT locations here:</tt> </tt>
<a name="L942"></a><tt class="py-lineno"> 942</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-string">"&gt;"</tt> <tt class="py-keyword">in</tt> <tt id="link-384" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-384', 'location', 'link-138');">location</a></tt><tt class="py-op">:</tt> </tt>
<a name="L943"></a><tt class="py-lineno"> 943</tt>  <tt class="py-line">                    <tt class="py-comment"># Nasty hack for common IMGT bug, should be &gt;123 not 123&gt;</tt> </tt>
<a name="L944"></a><tt class="py-lineno"> 944</tt>  <tt class="py-line">                    <tt class="py-comment"># in a location string. At least here the meaning is clear,</tt> </tt>
<a name="L945"></a><tt class="py-lineno"> 945</tt>  <tt class="py-line">                    <tt class="py-comment"># and since it is so common I don't want to issue a warning</tt> </tt>
<a name="L946"></a><tt class="py-lineno"> 946</tt>  <tt class="py-line">                    <tt class="py-comment"># warnings.warn("Feature location %s is invalid, "</tt> </tt>
<a name="L947"></a><tt class="py-lineno"> 947</tt>  <tt class="py-line">                    <tt class="py-comment">#              "moving greater than sign before position"</tt> </tt>
<a name="L948"></a><tt class="py-lineno"> 948</tt>  <tt class="py-line">                    <tt class="py-comment">#              % location, BiopythonParserWarning)</tt> </tt>
<a name="L949"></a><tt class="py-lineno"> 949</tt>  <tt class="py-line">                    <tt id="link-385" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-385', 'location', 'link-138');">location</a></tt> <tt class="py-op">=</tt> <tt class="py-name">bad_position_re</tt><tt class="py-op">.</tt><tt class="py-name">sub</tt><tt class="py-op">(</tt><tt class="py-string">r'&gt;\1'</tt><tt class="py-op">,</tt> <tt id="link-386" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-386', 'location', 'link-138');">location</a></tt><tt class="py-op">)</tt> </tt>
<a name="L950"></a><tt class="py-lineno"> 950</tt>  <tt class="py-line">                <tt id="link-387" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-387', 'features', 'link-42');">features</a></tt><tt class="py-op">.</tt><tt id="link-388" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-388', 'append', 'link-36');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt id="link-389" class="py-name"><a title="Bio.GenBank._FeatureConsumer.feature_key
Bio.GenBank._RecordConsumer.feature_key" class="py-name" href="#" onclick="return doclink('link-389', 'feature_key', 'link-61');">feature_key</a></tt><tt class="py-op">,</tt> <tt id="link-390" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-390', 'location', 'link-138');">location</a></tt><tt class="py-op">,</tt> <tt class="py-name">qualifiers</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L951"></a><tt class="py-lineno"> 951</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> </tt>
<a name="L952"></a><tt class="py-lineno"> 952</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-391" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-391', 'features', 'link-42');">features</a></tt> </tt>
</div></div><a name="L953"></a><tt class="py-lineno"> 953</tt>  <tt class="py-line"> </tt>
<a name="L954"></a><tt class="py-lineno"> 954</tt>  <tt class="py-line"> </tt>
<a name="GenBankScanner"></a><div id="GenBankScanner-def"><a name="L955"></a><tt class="py-lineno"> 955</tt> <a class="py-toggle" href="#" id="GenBankScanner-toggle" onclick="return toggle('GenBankScanner');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.GenBankScanner-class.html">GenBankScanner</a><tt class="py-op">(</tt><tt class="py-base-class">InsdcScanner</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GenBankScanner-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="GenBankScanner-expanded"><a name="L956"></a><tt class="py-lineno"> 956</tt>  <tt class="py-line">    <tt class="py-docstring">"""For extracting chunks of information in GenBank files"""</tt> </tt>
<a name="L957"></a><tt class="py-lineno"> 957</tt>  <tt class="py-line"> </tt>
<a name="L958"></a><tt class="py-lineno"> 958</tt>  <tt class="py-line">    <tt id="link-392" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.RECORD_START
Bio.GenBank.Scanner.GenBankScanner.RECORD_START
Bio.GenBank.Scanner.InsdcScanner.RECORD_START" class="py-name" href="#" onclick="return doclink('link-392', 'RECORD_START', 'link-11');">RECORD_START</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"LOCUS       "</tt> </tt>
<a name="L959"></a><tt class="py-lineno"> 959</tt>  <tt class="py-line">    <tt id="link-393" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-393', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt> <tt class="py-op">=</tt> <tt class="py-number">12</tt> </tt>
<a name="L960"></a><tt class="py-lineno"> 960</tt>  <tt class="py-line">    <tt id="link-394" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner.GenBankScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner.InsdcScanner.FEATURE_START_MARKERS
Bio.GenBank.Scanner._ImgtScanner.FEATURE_START_MARKERS" class="py-name" href="#" onclick="return doclink('link-394', 'FEATURE_START_MARKERS', 'link-13');">FEATURE_START_MARKERS</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">"FEATURES             Location/Qualifiers"</tt><tt class="py-op">,</tt> <tt class="py-string">"FEATURES"</tt><tt class="py-op">]</tt> </tt>
<a name="L961"></a><tt class="py-lineno"> 961</tt>  <tt class="py-line">    <tt id="link-395" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.FEATURE_END_MARKERS
Bio.GenBank.Scanner.GenBankScanner.FEATURE_END_MARKERS
Bio.GenBank.Scanner.InsdcScanner.FEATURE_END_MARKERS" class="py-name" href="#" onclick="return doclink('link-395', 'FEATURE_END_MARKERS', 'link-14');">FEATURE_END_MARKERS</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L962"></a><tt class="py-lineno"> 962</tt>  <tt class="py-line">    <tt id="link-396" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-396', 'FEATURE_QUALIFIER_INDENT', 'link-15');">FEATURE_QUALIFIER_INDENT</a></tt> <tt class="py-op">=</tt> <tt class="py-number">21</tt> </tt>
<a name="L963"></a><tt class="py-lineno"> 963</tt>  <tt class="py-line">    <tt id="link-397" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_SPACER" class="py-name" href="#" onclick="return doclink('link-397', 'FEATURE_QUALIFIER_SPACER', 'link-16');">FEATURE_QUALIFIER_SPACER</a></tt> <tt class="py-op">=</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt id="link-398" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-398', 'FEATURE_QUALIFIER_INDENT', 'link-15');">FEATURE_QUALIFIER_INDENT</a></tt> </tt>
<a name="L964"></a><tt class="py-lineno"> 964</tt>  <tt class="py-line">    <tt id="link-399" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.GenBankScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.InsdcScanner.SEQUENCE_HEADERS" class="py-name" href="#" onclick="return doclink('link-399', 'SEQUENCE_HEADERS', 'link-17');">SEQUENCE_HEADERS</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">"CONTIG"</tt><tt class="py-op">,</tt> <tt class="py-string">"ORIGIN"</tt><tt class="py-op">,</tt> <tt class="py-string">"BASE COUNT"</tt><tt class="py-op">,</tt> <tt class="py-string">"WGS"</tt><tt class="py-op">]</tt>  <tt class="py-comment"># trailing spaces removed</tt> </tt>
<a name="L965"></a><tt class="py-lineno"> 965</tt>  <tt class="py-line"> </tt>
<a name="GenBankScanner.parse_footer"></a><div id="GenBankScanner.parse_footer-def"><a name="L966"></a><tt class="py-lineno"> 966</tt> <a class="py-toggle" href="#" id="GenBankScanner.parse_footer-toggle" onclick="return toggle('GenBankScanner.parse_footer');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.GenBankScanner-class.html#parse_footer">parse_footer</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GenBankScanner.parse_footer-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="GenBankScanner.parse_footer-expanded"><a name="L967"></a><tt class="py-lineno"> 967</tt>  <tt class="py-line">        <tt class="py-docstring">"""returns a tuple containing a list of any misc strings, and the sequence"""</tt> </tt>
<a name="L968"></a><tt class="py-lineno"> 968</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-400" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-400', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-401" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-401', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-402" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.GenBankScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.InsdcScanner.SEQUENCE_HEADERS" class="py-name" href="#" onclick="return doclink('link-402', 'SEQUENCE_HEADERS', 'link-17');">SEQUENCE_HEADERS</a></tt><tt class="py-op">,</tt> \ </tt>
<a name="L969"></a><tt class="py-lineno"> 969</tt>  <tt class="py-line">            <tt class="py-string">"Eh? '%s'"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> </tt>
<a name="L970"></a><tt class="py-lineno"> 970</tt>  <tt class="py-line"> </tt>
<a name="L971"></a><tt class="py-lineno"> 971</tt>  <tt class="py-line">        <tt class="py-name">misc_lines</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L972"></a><tt class="py-lineno"> 972</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-403" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-403', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-404" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-404', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-405" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.GenBankScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.InsdcScanner.SEQUENCE_HEADERS" class="py-name" href="#" onclick="return doclink('link-405', 'SEQUENCE_HEADERS', 'link-17');">SEQUENCE_HEADERS</a></tt> \ </tt>
<a name="L973"></a><tt class="py-lineno"> 973</tt>  <tt class="py-line">            <tt class="py-keyword">or</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-406" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-406', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-407" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-407', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt> \ </tt>
<a name="L974"></a><tt class="py-lineno"> 974</tt>  <tt class="py-line">                <tt class="py-keyword">or</tt> <tt class="py-string">"WGS"</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L975"></a><tt class="py-lineno"> 975</tt>  <tt class="py-line">            <tt class="py-name">misc_lines</tt><tt class="py-op">.</tt><tt id="link-408" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-408', 'append', 'link-36');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-409" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-409', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L976"></a><tt class="py-lineno"> 976</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-410" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-410', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L977"></a><tt class="py-lineno"> 977</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L978"></a><tt class="py-lineno"> 978</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Premature end of file"</tt><tt class="py-op">)</tt> </tt>
<a name="L979"></a><tt class="py-lineno"> 979</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> </tt>
<a name="L980"></a><tt class="py-lineno"> 980</tt>  <tt class="py-line"> </tt>
<a name="L981"></a><tt class="py-lineno"> 981</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-411" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-411', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-412" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-412', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-413" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.GenBankScanner.SEQUENCE_HEADERS
Bio.GenBank.Scanner.InsdcScanner.SEQUENCE_HEADERS" class="py-name" href="#" onclick="return doclink('link-413', 'SEQUENCE_HEADERS', 'link-17');">SEQUENCE_HEADERS</a></tt><tt class="py-op">,</tt> \ </tt>
<a name="L982"></a><tt class="py-lineno"> 982</tt>  <tt class="py-line">            <tt class="py-string">"Eh? '%s'"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> </tt>
<a name="L983"></a><tt class="py-lineno"> 983</tt>  <tt class="py-line"> </tt>
<a name="L984"></a><tt class="py-lineno"> 984</tt>  <tt class="py-line">        <tt class="py-comment"># Now just consume the sequence lines until reach the // marker</tt> </tt>
<a name="L985"></a><tt class="py-lineno"> 985</tt>  <tt class="py-line">        <tt class="py-comment"># or a CONTIG line</tt> </tt>
<a name="L986"></a><tt class="py-lineno"> 986</tt>  <tt class="py-line">        <tt class="py-name">seq_lines</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L987"></a><tt class="py-lineno"> 987</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> </tt>
<a name="L988"></a><tt class="py-lineno"> 988</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L989"></a><tt class="py-lineno"> 989</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L990"></a><tt class="py-lineno"> 990</tt>  <tt class="py-line">                <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Premature end of file in sequence data"</tt><tt class="py-op">,</tt> </tt>
<a name="L991"></a><tt class="py-lineno"> 991</tt>  <tt class="py-line">                              <tt id="link-414" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-414', 'BiopythonParserWarning', 'link-10');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L992"></a><tt class="py-lineno"> 992</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-string">'//'</tt> </tt>
<a name="L993"></a><tt class="py-lineno"> 993</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L994"></a><tt class="py-lineno"> 994</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-415" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-415', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L995"></a><tt class="py-lineno"> 995</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L996"></a><tt class="py-lineno"> 996</tt>  <tt class="py-line">                <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Blank line in sequence data"</tt><tt class="py-op">,</tt> </tt>
<a name="L997"></a><tt class="py-lineno"> 997</tt>  <tt class="py-line">                              <tt id="link-416" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-416', 'BiopythonParserWarning', 'link-10');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L998"></a><tt class="py-lineno"> 998</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-417" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-417', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L999"></a><tt class="py-lineno"> 999</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L1000"></a><tt class="py-lineno">1000</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt> <tt class="py-op">==</tt> <tt class="py-string">'//'</tt><tt class="py-op">:</tt> </tt>
<a name="L1001"></a><tt class="py-lineno">1001</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L1002"></a><tt class="py-lineno">1002</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-418" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-418', 'startswith', 'link-97');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'CONTIG'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1003"></a><tt class="py-lineno">1003</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L1004"></a><tt class="py-lineno">1004</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">9</tt> <tt class="py-keyword">and</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">:</tt><tt class="py-number">10</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-string">' '</tt><tt class="py-op">:</tt> </tt>
<a name="L1005"></a><tt class="py-lineno">1005</tt>  <tt class="py-line">                <tt class="py-comment"># Some broken programs indent the sequence by one space too many</tt> </tt>
<a name="L1006"></a><tt class="py-lineno">1006</tt>  <tt class="py-line">                <tt class="py-comment"># so try to get rid of that and test again.</tt> </tt>
<a name="L1007"></a><tt class="py-lineno">1007</tt>  <tt class="py-line">                <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid indentation for sequence line"</tt><tt class="py-op">,</tt> </tt>
<a name="L1008"></a><tt class="py-lineno">1008</tt>  <tt class="py-line">                              <tt id="link-419" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-419', 'BiopythonParserWarning', 'link-10');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1009"></a><tt class="py-lineno">1009</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L1010"></a><tt class="py-lineno">1010</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">9</tt> <tt class="py-keyword">and</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">:</tt><tt class="py-number">10</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-string">' '</tt><tt class="py-op">:</tt> </tt>
<a name="L1011"></a><tt class="py-lineno">1011</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Sequence line mal-formed, '%s'"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1012"></a><tt class="py-lineno">1012</tt>  <tt class="py-line">            <tt class="py-name">seq_lines</tt><tt class="py-op">.</tt><tt id="link-420" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-420', 'append', 'link-36');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">10</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>  <tt class="py-comment"># remove spaces later</tt> </tt>
<a name="L1013"></a><tt class="py-lineno">1013</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-421" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-421', 'readline', 'link-24');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1014"></a><tt class="py-lineno">1014</tt>  <tt class="py-line"> </tt>
<a name="L1015"></a><tt class="py-lineno">1015</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> </tt>
<a name="L1016"></a><tt class="py-lineno">1016</tt>  <tt class="py-line">        <tt class="py-comment"># Seq("".join(seq_lines), self.alphabet)</tt> </tt>
<a name="L1017"></a><tt class="py-lineno">1017</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-name">misc_lines</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">seq_lines</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">" "</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1018"></a><tt class="py-lineno">1018</tt>  <tt class="py-line"> </tt>
<a name="GenBankScanner._feed_first_line"></a><div id="GenBankScanner._feed_first_line-def"><a name="L1019"></a><tt class="py-lineno">1019</tt> <a class="py-toggle" href="#" id="GenBankScanner._feed_first_line-toggle" onclick="return toggle('GenBankScanner._feed_first_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.GenBankScanner-class.html#_feed_first_line">_feed_first_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GenBankScanner._feed_first_line-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="GenBankScanner._feed_first_line-expanded"><a name="L1020"></a><tt class="py-lineno">1020</tt>  <tt class="py-line">        <tt class="py-docstring">"""Scan over and parse GenBank LOCUS line (PRIVATE).</tt> </tt>
<a name="L1021"></a><tt class="py-lineno">1021</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1022"></a><tt class="py-lineno">1022</tt>  <tt class="py-line"><tt class="py-docstring">        This must cope with several variants, primarily the old and new column</tt> </tt>
<a name="L1023"></a><tt class="py-lineno">1023</tt>  <tt class="py-line"><tt class="py-docstring">        based standards from GenBank. Additionally EnsEMBL produces GenBank</tt> </tt>
<a name="L1024"></a><tt class="py-lineno">1024</tt>  <tt class="py-line"><tt class="py-docstring">        files where the LOCUS line is space separated rather that following</tt> </tt>
<a name="L1025"></a><tt class="py-lineno">1025</tt>  <tt class="py-line"><tt class="py-docstring">        the column based layout.</tt> </tt>
<a name="L1026"></a><tt class="py-lineno">1026</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1027"></a><tt class="py-lineno">1027</tt>  <tt class="py-line"><tt class="py-docstring">        We also try to cope with GenBank like files with partial LOCUS lines.</tt> </tt>
<a name="L1028"></a><tt class="py-lineno">1028</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1029"></a><tt class="py-lineno">1029</tt>  <tt class="py-line">        <tt class="py-comment">#####################################</tt> </tt>
<a name="L1030"></a><tt class="py-lineno">1030</tt>  <tt class="py-line">        <tt class="py-comment"># LOCUS line                        #</tt> </tt>
<a name="L1031"></a><tt class="py-lineno">1031</tt>  <tt class="py-line">        <tt class="py-comment">#####################################</tt> </tt>
<a name="L1032"></a><tt class="py-lineno">1032</tt>  <tt class="py-line">        <tt id="link-422" class="py-name" targets="Variable Bio.GenBank.GENBANK_INDENT=Bio.GenBank-module.html#GENBANK_INDENT"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-422', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-423" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-423', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt> </tt>
<a name="L1033"></a><tt class="py-lineno">1033</tt>  <tt class="py-line">        <tt id="link-424" class="py-name" targets="Variable Bio.GenBank.GENBANK_SPACER=Bio.GenBank-module.html#GENBANK_SPACER"><a title="Bio.GenBank.GENBANK_SPACER" class="py-name" href="#" onclick="return doclink('link-424', 'GENBANK_SPACER', 'link-424');">GENBANK_SPACER</a></tt> <tt class="py-op">=</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt id="link-425" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-425', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt> </tt>
<a name="L1034"></a><tt class="py-lineno">1034</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">:</tt><tt id="link-426" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-426', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'LOCUS       '</tt><tt class="py-op">,</tt> \ </tt>
<a name="L1035"></a><tt class="py-lineno">1035</tt>  <tt class="py-line">            <tt class="py-string">'LOCUS line does not start correctly:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L1036"></a><tt class="py-lineno">1036</tt>  <tt class="py-line"> </tt>
<a name="L1037"></a><tt class="py-lineno">1037</tt>  <tt class="py-line">        <tt class="py-comment"># Have to break up the locus line, and handle the different bits of it.</tt> </tt>
<a name="L1038"></a><tt class="py-lineno">1038</tt>  <tt class="py-line">        <tt class="py-comment"># There are at least two different versions of the locus line...</tt> </tt>
<a name="L1039"></a><tt class="py-lineno">1039</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">29</tt><tt class="py-op">:</tt><tt class="py-number">33</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">' bp '</tt><tt class="py-op">,</tt> <tt class="py-string">' aa '</tt><tt class="py-op">,</tt> <tt class="py-string">' rc '</tt><tt class="py-op">]</tt> <tt class="py-keyword">and</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">55</tt><tt class="py-op">:</tt><tt class="py-number">62</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'       '</tt><tt class="py-op">:</tt> </tt>
<a name="L1040"></a><tt class="py-lineno">1040</tt>  <tt class="py-line">            <tt class="py-comment"># Old... note we insist on the 55:62 being empty to avoid trying</tt> </tt>
<a name="L1041"></a><tt class="py-lineno">1041</tt>  <tt class="py-line">            <tt class="py-comment"># to parse space separated LOCUS lines from Ensembl etc, see below.</tt> </tt>
<a name="L1042"></a><tt class="py-lineno">1042</tt>  <tt class="py-line">            <tt class="py-comment">#</tt> </tt>
<a name="L1043"></a><tt class="py-lineno">1043</tt>  <tt class="py-line">            <tt class="py-comment">#    Positions  Contents</tt> </tt>
<a name="L1044"></a><tt class="py-lineno">1044</tt>  <tt class="py-line">            <tt class="py-comment">#    ---------  --------</tt> </tt>
<a name="L1045"></a><tt class="py-lineno">1045</tt>  <tt class="py-line">            <tt class="py-comment">#    00:06      LOCUS</tt> </tt>
<a name="L1046"></a><tt class="py-lineno">1046</tt>  <tt class="py-line">            <tt class="py-comment">#    06:12      spaces</tt> </tt>
<a name="L1047"></a><tt class="py-lineno">1047</tt>  <tt class="py-line">            <tt class="py-comment">#    12:??      Locus name</tt> </tt>
<a name="L1048"></a><tt class="py-lineno">1048</tt>  <tt class="py-line">            <tt class="py-comment">#    ??:??      space</tt> </tt>
<a name="L1049"></a><tt class="py-lineno">1049</tt>  <tt class="py-line">            <tt class="py-comment">#    ??:29      Length of sequence, right-justified</tt> </tt>
<a name="L1050"></a><tt class="py-lineno">1050</tt>  <tt class="py-line">            <tt class="py-comment">#    29:33      space, bp, space</tt> </tt>
<a name="L1051"></a><tt class="py-lineno">1051</tt>  <tt class="py-line">            <tt class="py-comment">#    33:41      strand type</tt> </tt>
<a name="L1052"></a><tt class="py-lineno">1052</tt>  <tt class="py-line">            <tt class="py-comment">#    41:42      space</tt> </tt>
<a name="L1053"></a><tt class="py-lineno">1053</tt>  <tt class="py-line">            <tt class="py-comment">#    42:51      Blank (implies linear), linear or circular</tt> </tt>
<a name="L1054"></a><tt class="py-lineno">1054</tt>  <tt class="py-line">            <tt class="py-comment">#    51:52      space</tt> </tt>
<a name="L1055"></a><tt class="py-lineno">1055</tt>  <tt class="py-line">            <tt class="py-comment">#    52:55      The division code (e.g. BCT, VRL, INV)</tt> </tt>
<a name="L1056"></a><tt class="py-lineno">1056</tt>  <tt class="py-line">            <tt class="py-comment">#    55:62      space</tt> </tt>
<a name="L1057"></a><tt class="py-lineno">1057</tt>  <tt class="py-line">            <tt class="py-comment">#    62:73      Date, in the form dd-MMM-yyyy (e.g., 15-MAR-1991)</tt> </tt>
<a name="L1058"></a><tt class="py-lineno">1058</tt>  <tt class="py-line">            <tt class="py-comment">#</tt> </tt>
<a name="L1059"></a><tt class="py-lineno">1059</tt>  <tt class="py-line">            <tt class="py-comment"># assert line[29:33] in [' bp ', ' aa ',' rc '] , \</tt> </tt>
<a name="L1060"></a><tt class="py-lineno">1060</tt>  <tt class="py-line">            <tt class="py-comment">#       'LOCUS line does not contain size units at expected position:\n' + line</tt> </tt>
<a name="L1061"></a><tt class="py-lineno">1061</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">41</tt><tt class="py-op">:</tt><tt class="py-number">42</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">' '</tt><tt class="py-op">,</tt> \ </tt>
<a name="L1062"></a><tt class="py-lineno">1062</tt>  <tt class="py-line">                <tt class="py-string">'LOCUS line does not contain space at position 42:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L1063"></a><tt class="py-lineno">1063</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">42</tt><tt class="py-op">:</tt><tt class="py-number">51</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-427" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-427', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">''</tt><tt class="py-op">,</tt> <tt class="py-string">'linear'</tt><tt class="py-op">,</tt> <tt class="py-string">'circular'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> \ </tt>
<a name="L1064"></a><tt class="py-lineno">1064</tt>  <tt class="py-line">                <tt class="py-string">'LOCUS line does not contain valid entry (linear, circular, ...):\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L1065"></a><tt class="py-lineno">1065</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">51</tt><tt class="py-op">:</tt><tt class="py-number">52</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">' '</tt><tt class="py-op">,</tt> \ </tt>
<a name="L1066"></a><tt class="py-lineno">1066</tt>  <tt class="py-line">                <tt class="py-string">'LOCUS line does not contain space at position 52:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L1067"></a><tt class="py-lineno">1067</tt>  <tt class="py-line">            <tt class="py-comment"># assert line[55:62] == '       ', \</tt> </tt>
<a name="L1068"></a><tt class="py-lineno">1068</tt>  <tt class="py-line">            <tt class="py-comment">#      'LOCUS line does not contain spaces from position 56 to 62:\n' + line</tt> </tt>
<a name="L1069"></a><tt class="py-lineno">1069</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">62</tt><tt class="py-op">:</tt><tt class="py-number">73</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-428" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-428', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1070"></a><tt class="py-lineno">1070</tt>  <tt class="py-line">                <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">64</tt><tt class="py-op">:</tt><tt class="py-number">65</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'-'</tt><tt class="py-op">,</tt> \ </tt>
<a name="L1071"></a><tt class="py-lineno">1071</tt>  <tt class="py-line">                    <tt class="py-string">'LOCUS line does not contain - at position 65 in date:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L1072"></a><tt class="py-lineno">1072</tt>  <tt class="py-line">                <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">68</tt><tt class="py-op">:</tt><tt class="py-number">69</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'-'</tt><tt class="py-op">,</tt> \ </tt>
<a name="L1073"></a><tt class="py-lineno">1073</tt>  <tt class="py-line">                    <tt class="py-string">'LOCUS line does not contain - at position 69 in date:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L1074"></a><tt class="py-lineno">1074</tt>  <tt class="py-line"> </tt>
<a name="L1075"></a><tt class="py-lineno">1075</tt>  <tt class="py-line">            <tt class="py-name">name_and_length_str</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt id="link-429" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-429', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt><tt class="py-op">:</tt><tt class="py-number">29</tt><tt class="py-op">]</tt> </tt>
<a name="L1076"></a><tt class="py-lineno">1076</tt>  <tt class="py-line">            <tt class="py-keyword">while</tt> <tt class="py-string">'  '</tt> <tt class="py-keyword">in</tt> <tt class="py-name">name_and_length_str</tt><tt class="py-op">:</tt> </tt>
<a name="L1077"></a><tt class="py-lineno">1077</tt>  <tt class="py-line">                <tt class="py-name">name_and_length_str</tt> <tt class="py-op">=</tt> <tt class="py-name">name_and_length_str</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'  '</tt><tt class="py-op">,</tt> <tt class="py-string">' '</tt><tt class="py-op">)</tt> </tt>
<a name="L1078"></a><tt class="py-lineno">1078</tt>  <tt class="py-line">            <tt class="py-name">name_and_length</tt> <tt class="py-op">=</tt> <tt class="py-name">name_and_length_str</tt><tt class="py-op">.</tt><tt id="link-430" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-430', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">' '</tt><tt class="py-op">)</tt> </tt>
<a name="L1079"></a><tt class="py-lineno">1079</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">name_and_length</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;=</tt> <tt class="py-number">2</tt><tt class="py-op">,</tt> \ </tt>
<a name="L1080"></a><tt class="py-lineno">1080</tt>  <tt class="py-line">                <tt class="py-string">'Cannot parse the name and length in the LOCUS line:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L1081"></a><tt class="py-lineno">1081</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">name_and_length</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> \ </tt>
<a name="L1082"></a><tt class="py-lineno">1082</tt>  <tt class="py-line">                <tt class="py-string">'Name and length collide in the LOCUS line:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L1083"></a><tt class="py-lineno">1083</tt>  <tt class="py-line">            <tt class="py-comment"># Should be possible to split them based on position, if</tt> </tt>
<a name="L1084"></a><tt class="py-lineno">1084</tt>  <tt class="py-line">            <tt class="py-comment"># a clear definition of the standard exists THAT AGREES with</tt> </tt>
<a name="L1085"></a><tt class="py-lineno">1085</tt>  <tt class="py-line">            <tt class="py-comment"># existing files.</tt> </tt>
<a name="L1086"></a><tt class="py-lineno">1086</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-431" class="py-name"><a title="Bio.GenBank._FeatureConsumer.locus
Bio.GenBank._RecordConsumer.locus" class="py-name" href="#" onclick="return doclink('link-431', 'locus', 'link-235');">locus</a></tt><tt class="py-op">(</tt><tt class="py-name">name_and_length</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1087"></a><tt class="py-lineno">1087</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-432" class="py-name"><a title="Bio.Alphabet.Alphabet.size
Bio.Alphabet.Reduced.HPModel.size
Bio.Alphabet.Reduced.Murphy10.size
Bio.Alphabet.Reduced.Murphy15.size
Bio.Alphabet.Reduced.Murphy4.size
Bio.Alphabet.Reduced.Murphy8.size
Bio.Alphabet.Reduced.PC5.size
Bio.Alphabet.SingleLetterAlphabet.size
Bio.Alphabet.ThreeLetterProtein.size
Bio.GenBank._FeatureConsumer.size
Bio.GenBank._RecordConsumer.size
Bio.Restriction.Restriction.AanI.size
Bio.Restriction.Restriction.AarI.size
Bio.Restriction.Restriction.AasI.size
Bio.Restriction.Restriction.AatII.size
Bio.Restriction.Restriction.AbaSI.size
Bio.Restriction.Restriction.AbsI.size
Bio.Restriction.Restriction.Acc16I.size
Bio.Restriction.Restriction.Acc36I.size
Bio.Restriction.Restriction.Acc65I.size
Bio.Restriction.Restriction.AccB1I.size
Bio.Restriction.Restriction.AccB7I.size
Bio.Restriction.Restriction.AccBSI.size
Bio.Restriction.Restriction.AccI.size
Bio.Restriction.Restriction.AccII.size
Bio.Restriction.Restriction.AccIII.size
Bio.Restriction.Restriction.AceIII.size
Bio.Restriction.Restriction.AciI.size
Bio.Restriction.Restriction.AclI.size
Bio.Restriction.Restriction.AclWI.size
Bio.Restriction.Restriction.AcoI.size
Bio.Restriction.Restriction.AcsI.size
Bio.Restriction.Restriction.AcuI.size
Bio.Restriction.Restriction.AcvI.size
Bio.Restriction.Restriction.AcyI.size
Bio.Restriction.Restriction.AdeI.size
Bio.Restriction.Restriction.AfaI.size
Bio.Restriction.Restriction.AfeI.size
Bio.Restriction.Restriction.AfiI.size
Bio.Restriction.Restriction.AflII.size
Bio.Restriction.Restriction.AflIII.size
Bio.Restriction.Restriction.AgeI.size
Bio.Restriction.Restriction.AgsI.size
Bio.Restriction.Restriction.AhaIII.size
Bio.Restriction.Restriction.AhdI.size
Bio.Restriction.Restriction.AhlI.size
Bio.Restriction.Restriction.AjiI.size
Bio.Restriction.Restriction.AjnI.size
Bio.Restriction.Restriction.AjuI.size
Bio.Restriction.Restriction.AleI.size
Bio.Restriction.Restriction.AlfI.size
Bio.Restriction.Restriction.AloI.size
Bio.Restriction.Restriction.AluBI.size
Bio.Restriction.Restriction.AluI.size
Bio.Restriction.Restriction.Alw21I.size
Bio.Restriction.Restriction.Alw26I.size
Bio.Restriction.Restriction.Alw44I.size
Bio.Restriction.Restriction.AlwFI.size
Bio.Restriction.Restriction.AlwI.size
Bio.Restriction.Restriction.AlwNI.size
Bio.Restriction.Restriction.Ama87I.size
Bio.Restriction.Restriction.Aor13HI.size
Bio.Restriction.Restriction.Aor51HI.size
Bio.Restriction.Restriction.AoxI.size
Bio.Restriction.Restriction.ApaBI.size
Bio.Restriction.Restriction.ApaI.size
Bio.Restriction.Restriction.ApaLI.size
Bio.Restriction.Restriction.ApeKI.size
Bio.Restriction.Restriction.ApoI.size
Bio.Restriction.Restriction.ApyPI.size
Bio.Restriction.Restriction.AquII.size
Bio.Restriction.Restriction.AquIII.size
Bio.Restriction.Restriction.AquIV.size
Bio.Restriction.Restriction.ArsI.size
Bio.Restriction.Restriction.AscI.size
Bio.Restriction.Restriction.AseI.size
Bio.Restriction.Restriction.Asi256I.size
Bio.Restriction.Restriction.AsiGI.size
Bio.Restriction.Restriction.AsiSI.size
Bio.Restriction.Restriction.Asp700I.size
Bio.Restriction.Restriction.Asp718I.size
Bio.Restriction.Restriction.AspA2I.size
Bio.Restriction.Restriction.AspBHI.size
Bio.Restriction.Restriction.AspLEI.size
Bio.Restriction.Restriction.AspS9I.size
Bio.Restriction.Restriction.AssI.size
Bio.Restriction.Restriction.AsuC2I.size
Bio.Restriction.Restriction.AsuHPI.size
Bio.Restriction.Restriction.AsuI.size
Bio.Restriction.Restriction.AsuII.size
Bio.Restriction.Restriction.AsuNHI.size
Bio.Restriction.Restriction.AvaI.size
Bio.Restriction.Restriction.AvaII.size
Bio.Restriction.Restriction.AvaIII.size
Bio.Restriction.Restriction.AvrII.size
Bio.Restriction.Restriction.AxyI.size
Bio.Restriction.Restriction.BaeGI.size
Bio.Restriction.Restriction.BaeI.size
Bio.Restriction.Restriction.BalI.size
Bio.Restriction.Restriction.BamHI.size
Bio.Restriction.Restriction.BanI.size
Bio.Restriction.Restriction.BanII.size
Bio.Restriction.Restriction.BarI.size
Bio.Restriction.Restriction.BasI.size
Bio.Restriction.Restriction.BauI.size
Bio.Restriction.Restriction.Bbr7I.size
Bio.Restriction.Restriction.BbrPI.size
Bio.Restriction.Restriction.BbsI.size
Bio.Restriction.Restriction.Bbv12I.size
Bio.Restriction.Restriction.BbvCI.size
Bio.Restriction.Restriction.BbvI.size
Bio.Restriction.Restriction.BbvII.size
Bio.Restriction.Restriction.BccI.size
Bio.Restriction.Restriction.Bce83I.size
Bio.Restriction.Restriction.BceAI.size
Bio.Restriction.Restriction.BcefI.size
Bio.Restriction.Restriction.BcgI.size
Bio.Restriction.Restriction.BciT130I.size
Bio.Restriction.Restriction.BciVI.size
Bio.Restriction.Restriction.BclI.size
Bio.Restriction.Restriction.BcnI.size
Bio.Restriction.Restriction.BcoDI.size
Bio.Restriction.Restriction.BcuI.size
Bio.Restriction.Restriction.BdaI.size
Bio.Restriction.Restriction.BetI.size
Bio.Restriction.Restriction.BfaI.size
Bio.Restriction.Restriction.BfiI.size
Bio.Restriction.Restriction.BfmI.size
Bio.Restriction.Restriction.BfoI.size
Bio.Restriction.Restriction.BfrI.size
Bio.Restriction.Restriction.BfuAI.size
Bio.Restriction.Restriction.BfuCI.size
Bio.Restriction.Restriction.BfuI.size
Bio.Restriction.Restriction.BglI.size
Bio.Restriction.Restriction.BglII.size
Bio.Restriction.Restriction.BinI.size
Bio.Restriction.Restriction.BisI.size
Bio.Restriction.Restriction.BlnI.size
Bio.Restriction.Restriction.BlpI.size
Bio.Restriction.Restriction.BlsI.size
Bio.Restriction.Restriction.BmcAI.size
Bio.Restriction.Restriction.Bme1390I.size
Bio.Restriction.Restriction.Bme18I.size
Bio.Restriction.Restriction.BmeDI.size
Bio.Restriction.Restriction.BmeRI.size
Bio.Restriction.Restriction.BmeT110I.size
Bio.Restriction.Restriction.BmgBI.size
Bio.Restriction.Restriction.BmgI.size
Bio.Restriction.Restriction.BmgT120I.size
Bio.Restriction.Restriction.BmiI.size
Bio.Restriction.Restriction.BmrFI.size
Bio.Restriction.Restriction.BmrI.size
Bio.Restriction.Restriction.BmsI.size
Bio.Restriction.Restriction.BmtI.size
Bio.Restriction.Restriction.BmuI.size
Bio.Restriction.Restriction.BoxI.size
Bio.Restriction.Restriction.BpiI.size
Bio.Restriction.Restriction.BplI.size
Bio.Restriction.Restriction.BpmI.size
Bio.Restriction.Restriction.Bpu10I.size
Bio.Restriction.Restriction.Bpu1102I.size
Bio.Restriction.Restriction.Bpu14I.size
Bio.Restriction.Restriction.BpuEI.size
Bio.Restriction.Restriction.BpuMI.size
Bio.Restriction.Restriction.BpvUI.size
Bio.Restriction.Restriction.Bsa29I.size
Bio.Restriction.Restriction.BsaAI.size
Bio.Restriction.Restriction.BsaBI.size
Bio.Restriction.Restriction.BsaHI.size
Bio.Restriction.Restriction.BsaI.size
Bio.Restriction.Restriction.BsaJI.size
Bio.Restriction.Restriction.BsaWI.size
Bio.Restriction.Restriction.BsaXI.size
Bio.Restriction.Restriction.BsbI.size
Bio.Restriction.Restriction.Bsc4I.size
Bio.Restriction.Restriction.BscAI.size
Bio.Restriction.Restriction.BscGI.size
Bio.Restriction.Restriction.Bse118I.size
Bio.Restriction.Restriction.Bse1I.size
Bio.Restriction.Restriction.Bse21I.size
Bio.Restriction.Restriction.Bse3DI.size
Bio.Restriction.Restriction.Bse8I.size
Bio.Restriction.Restriction.BseAI.size
Bio.Restriction.Restriction.BseBI.size
Bio.Restriction.Restriction.BseCI.size
Bio.Restriction.Restriction.BseDI.size
Bio.Restriction.Restriction.BseGI.size
Bio.Restriction.Restriction.BseJI.size
Bio.Restriction.Restriction.BseLI.size
Bio.Restriction.Restriction.BseMI.size
Bio.Restriction.Restriction.BseMII.size
Bio.Restriction.Restriction.BseNI.size
Bio.Restriction.Restriction.BsePI.size
Bio.Restriction.Restriction.BseRI.size
Bio.Restriction.Restriction.BseSI.size
Bio.Restriction.Restriction.BseX3I.size
Bio.Restriction.Restriction.BseXI.size
Bio.Restriction.Restriction.BseYI.size
Bio.Restriction.Restriction.BsgI.size
Bio.Restriction.Restriction.Bsh1236I.size
Bio.Restriction.Restriction.Bsh1285I.size
Bio.Restriction.Restriction.BshFI.size
Bio.Restriction.Restriction.BshNI.size
Bio.Restriction.Restriction.BshTI.size
Bio.Restriction.Restriction.BshVI.size
Bio.Restriction.Restriction.BsiEI.size
Bio.Restriction.Restriction.BsiHKAI.size
Bio.Restriction.Restriction.BsiHKCI.size
Bio.Restriction.Restriction.BsiI.size
Bio.Restriction.Restriction.BsiSI.size
Bio.Restriction.Restriction.BsiWI.size
Bio.Restriction.Restriction.BsiYI.size
Bio.Restriction.Restriction.BslFI.size
Bio.Restriction.Restriction.BslI.size
Bio.Restriction.Restriction.BsmAI.size
Bio.Restriction.Restriction.BsmBI.size
Bio.Restriction.Restriction.BsmFI.size
Bio.Restriction.Restriction.BsmI.size
Bio.Restriction.Restriction.BsnI.size
Bio.Restriction.Restriction.Bso31I.size
Bio.Restriction.Restriction.BsoBI.size
Bio.Restriction.Restriction.Bsp119I.size
Bio.Restriction.Restriction.Bsp120I.size
Bio.Restriction.Restriction.Bsp1286I.size
Bio.Restriction.Restriction.Bsp13I.size
Bio.Restriction.Restriction.Bsp1407I.size
Bio.Restriction.Restriction.Bsp143I.size
Bio.Restriction.Restriction.Bsp1720I.size
Bio.Restriction.Restriction.Bsp19I.size
Bio.Restriction.Restriction.Bsp24I.size
Bio.Restriction.Restriction.Bsp68I.size
Bio.Restriction.Restriction.BspACI.size
Bio.Restriction.Restriction.BspCNI.size
Bio.Restriction.Restriction.BspD6I.size
Bio.Restriction.Restriction.BspDI.size
Bio.Restriction.Restriction.BspEI.size
Bio.Restriction.Restriction.BspFNI.size
Bio.Restriction.Restriction.BspGI.size
Bio.Restriction.Restriction.BspHI.size
Bio.Restriction.Restriction.BspLI.size
Bio.Restriction.Restriction.BspLU11I.size
Bio.Restriction.Restriction.BspMI.size
Bio.Restriction.Restriction.BspMII.size
Bio.Restriction.Restriction.BspNCI.size
Bio.Restriction.Restriction.BspOI.size
Bio.Restriction.Restriction.BspPI.size
Bio.Restriction.Restriction.BspQI.size
Bio.Restriction.Restriction.BspT104I.size
Bio.Restriction.Restriction.BspT107I.size
Bio.Restriction.Restriction.BspTI.size
Bio.Restriction.Restriction.BsrBI.size
Bio.Restriction.Restriction.BsrDI.size
Bio.Restriction.Restriction.BsrFI.size
Bio.Restriction.Restriction.BsrGI.size
Bio.Restriction.Restriction.BsrI.size
Bio.Restriction.Restriction.BsrSI.size
Bio.Restriction.Restriction.BssAI.size
Bio.Restriction.Restriction.BssECI.size
Bio.Restriction.Restriction.BssHII.size
Bio.Restriction.Restriction.BssKI.size
Bio.Restriction.Restriction.BssMI.size
Bio.Restriction.Restriction.BssNAI.size
Bio.Restriction.Restriction.BssNI.size
Bio.Restriction.Restriction.BssSI.size
Bio.Restriction.Restriction.BssT1I.size
Bio.Restriction.Restriction.Bst1107I.size
Bio.Restriction.Restriction.Bst2BI.size
Bio.Restriction.Restriction.Bst2UI.size
Bio.Restriction.Restriction.Bst4CI.size
Bio.Restriction.Restriction.Bst6I.size
Bio.Restriction.Restriction.BstACI.size
Bio.Restriction.Restriction.BstAFI.size
Bio.Restriction.Restriction.BstAPI.size
Bio.Restriction.Restriction.BstAUI.size
Bio.Restriction.Restriction.BstBAI.size
Bio.Restriction.Restriction.BstBI.size
Bio.Restriction.Restriction.BstC8I.size
Bio.Restriction.Restriction.BstDEI.size
Bio.Restriction.Restriction.BstDSI.size
Bio.Restriction.Restriction.BstEII.size
Bio.Restriction.Restriction.BstENI.size
Bio.Restriction.Restriction.BstF5I.size
Bio.Restriction.Restriction.BstFNI.size
Bio.Restriction.Restriction.BstH2I.size
Bio.Restriction.Restriction.BstHHI.size
Bio.Restriction.Restriction.BstKTI.size
Bio.Restriction.Restriction.BstMAI.size
Bio.Restriction.Restriction.BstMBI.size
Bio.Restriction.Restriction.BstMCI.size
Bio.Restriction.Restriction.BstMWI.size
Bio.Restriction.Restriction.BstNI.size
Bio.Restriction.Restriction.BstNSI.size
Bio.Restriction.Restriction.BstOI.size
Bio.Restriction.Restriction.BstPAI.size
Bio.Restriction.Restriction.BstPI.size
Bio.Restriction.Restriction.BstSCI.size
Bio.Restriction.Restriction.BstSFI.size
Bio.Restriction.Restriction.BstSLI.size
Bio.Restriction.Restriction.BstSNI.size
Bio.Restriction.Restriction.BstUI.size
Bio.Restriction.Restriction.BstV1I.size
Bio.Restriction.Restriction.BstV2I.size
Bio.Restriction.Restriction.BstX2I.size
Bio.Restriction.Restriction.BstXI.size
Bio.Restriction.Restriction.BstYI.size
Bio.Restriction.Restriction.BstZ17I.size
Bio.Restriction.Restriction.BstZI.size
Bio.Restriction.Restriction.Bsu15I.size
Bio.Restriction.Restriction.Bsu36I.size
Bio.Restriction.Restriction.BsuI.size
Bio.Restriction.Restriction.BsuRI.size
Bio.Restriction.Restriction.BtgI.size
Bio.Restriction.Restriction.BtgZI.size
Bio.Restriction.Restriction.BthCI.size
Bio.Restriction.Restriction.BtrI.size
Bio.Restriction.Restriction.BtsCI.size
Bio.Restriction.Restriction.BtsI.size
Bio.Restriction.Restriction.BtsIMutI.size
Bio.Restriction.Restriction.BtuMI.size
Bio.Restriction.Restriction.BveI.size
Bio.Restriction.Restriction.Cac8I.size
Bio.Restriction.Restriction.CaiI.size
Bio.Restriction.Restriction.CauII.size
Bio.Restriction.Restriction.CchII.size
Bio.Restriction.Restriction.CchIII.size
Bio.Restriction.Restriction.CciI.size
Bio.Restriction.Restriction.CciNI.size
Bio.Restriction.Restriction.Cdi630V.size
Bio.Restriction.Restriction.CdiI.size
Bio.Restriction.Restriction.CdpI.size
Bio.Restriction.Restriction.CfoI.size
Bio.Restriction.Restriction.Cfr10I.size
Bio.Restriction.Restriction.Cfr13I.size
Bio.Restriction.Restriction.Cfr42I.size
Bio.Restriction.Restriction.Cfr9I.size
Bio.Restriction.Restriction.CfrI.size
Bio.Restriction.Restriction.Cgl13032I.size
Bio.Restriction.Restriction.Cgl13032II.size
Bio.Restriction.Restriction.ChaI.size
Bio.Restriction.Restriction.CjeFIII.size
Bio.Restriction.Restriction.CjeFV.size
Bio.Restriction.Restriction.CjeI.size
Bio.Restriction.Restriction.CjeNII.size
Bio.Restriction.Restriction.CjeNIII.size
Bio.Restriction.Restriction.CjeP659IV.size
Bio.Restriction.Restriction.CjePI.size
Bio.Restriction.Restriction.CjuI.size
Bio.Restriction.Restriction.CjuII.size
Bio.Restriction.Restriction.ClaI.size
Bio.Restriction.Restriction.CpoI.size
Bio.Restriction.Restriction.CseI.size
Bio.Restriction.Restriction.CsiI.size
Bio.Restriction.Restriction.Csp6I.size
Bio.Restriction.Restriction.CspAI.size
Bio.Restriction.Restriction.CspCI.size
Bio.Restriction.Restriction.CspI.size
Bio.Restriction.Restriction.CstMI.size
Bio.Restriction.Restriction.CviAII.size
Bio.Restriction.Restriction.CviJI.size
Bio.Restriction.Restriction.CviKI_1.size
Bio.Restriction.Restriction.CviQI.size
Bio.Restriction.Restriction.CviRI.size
Bio.Restriction.Restriction.DdeI.size
Bio.Restriction.Restriction.DinI.size
Bio.Restriction.Restriction.DpnI.size
Bio.Restriction.Restriction.DpnII.size
Bio.Restriction.Restriction.DraI.size
Bio.Restriction.Restriction.DraII.size
Bio.Restriction.Restriction.DraIII.size
Bio.Restriction.Restriction.DraRI.size
Bio.Restriction.Restriction.DrdI.size
Bio.Restriction.Restriction.DrdII.size
Bio.Restriction.Restriction.DriI.size
Bio.Restriction.Restriction.DsaI.size
Bio.Restriction.Restriction.DseDI.size
Bio.Restriction.Restriction.EaeI.size
Bio.Restriction.Restriction.EagI.size
Bio.Restriction.Restriction.Eam1104I.size
Bio.Restriction.Restriction.Eam1105I.size
Bio.Restriction.Restriction.EarI.size
Bio.Restriction.Restriction.EciI.size
Bio.Restriction.Restriction.Ecl136II.size
Bio.Restriction.Restriction.EclXI.size
Bio.Restriction.Restriction.Eco105I.size
Bio.Restriction.Restriction.Eco130I.size
Bio.Restriction.Restriction.Eco147I.size
Bio.Restriction.Restriction.Eco24I.size
Bio.Restriction.Restriction.Eco31I.size
Bio.Restriction.Restriction.Eco32I.size
Bio.Restriction.Restriction.Eco47I.size
Bio.Restriction.Restriction.Eco47III.size
Bio.Restriction.Restriction.Eco52I.size
Bio.Restriction.Restriction.Eco53kI.size
Bio.Restriction.Restriction.Eco57I.size
Bio.Restriction.Restriction.Eco57MI.size
Bio.Restriction.Restriction.Eco72I.size
Bio.Restriction.Restriction.Eco81I.size
Bio.Restriction.Restriction.Eco88I.size
Bio.Restriction.Restriction.Eco91I.size
Bio.Restriction.Restriction.EcoHI.size
Bio.Restriction.Restriction.EcoICRI.size
Bio.Restriction.Restriction.EcoNI.size
Bio.Restriction.Restriction.EcoO109I.size
Bio.Restriction.Restriction.EcoO65I.size
Bio.Restriction.Restriction.EcoRI.size
Bio.Restriction.Restriction.EcoRII.size
Bio.Restriction.Restriction.EcoRV.size
Bio.Restriction.Restriction.EcoT14I.size
Bio.Restriction.Restriction.EcoT22I.size
Bio.Restriction.Restriction.EcoT38I.size
Bio.Restriction.Restriction.EgeI.size
Bio.Restriction.Restriction.EheI.size
Bio.Restriction.Restriction.ErhI.size
Bio.Restriction.Restriction.EsaBC3I.size
Bio.Restriction.Restriction.EsaSSI.size
Bio.Restriction.Restriction.Esp3I.size
Bio.Restriction.Restriction.EspI.size
Bio.Restriction.Restriction.FaeI.size
Bio.Restriction.Restriction.FaiI.size
Bio.Restriction.Restriction.FalI.size
Bio.Restriction.Restriction.FaqI.size
Bio.Restriction.Restriction.FatI.size
Bio.Restriction.Restriction.FauI.size
Bio.Restriction.Restriction.FauNDI.size
Bio.Restriction.Restriction.FbaI.size
Bio.Restriction.Restriction.FblI.size
Bio.Restriction.Restriction.FinI.size
Bio.Restriction.Restriction.FmuI.size
Bio.Restriction.Restriction.Fnu4HI.size
Bio.Restriction.Restriction.FnuDII.size
Bio.Restriction.Restriction.FokI.size
Bio.Restriction.Restriction.FriOI.size
Bio.Restriction.Restriction.FseI.size
Bio.Restriction.Restriction.Fsp4HI.size
Bio.Restriction.Restriction.FspAI.size
Bio.Restriction.Restriction.FspBI.size
Bio.Restriction.Restriction.FspEI.size
Bio.Restriction.Restriction.FspI.size
Bio.Restriction.Restriction.GauT27I.size
Bio.Restriction.Restriction.GdiII.size
Bio.Restriction.Restriction.GlaI.size
Bio.Restriction.Restriction.GluI.size
Bio.Restriction.Restriction.GsaI.size
Bio.Restriction.Restriction.GsuI.size
Bio.Restriction.Restriction.HaeI.size
Bio.Restriction.Restriction.HaeII.size
Bio.Restriction.Restriction.HaeIII.size
Bio.Restriction.Restriction.HapII.size
Bio.Restriction.Restriction.HauII.size
Bio.Restriction.Restriction.HgaI.size
Bio.Restriction.Restriction.HgiAI.size
Bio.Restriction.Restriction.HgiCI.size
Bio.Restriction.Restriction.HgiEII.size
Bio.Restriction.Restriction.HgiJII.size
Bio.Restriction.Restriction.HhaI.size
Bio.Restriction.Restriction.Hin1I.size
Bio.Restriction.Restriction.Hin1II.size
Bio.Restriction.Restriction.Hin4I.size
Bio.Restriction.Restriction.Hin4II.size
Bio.Restriction.Restriction.Hin6I.size
Bio.Restriction.Restriction.HinP1I.size
Bio.Restriction.Restriction.HincII.size
Bio.Restriction.Restriction.HindII.size
Bio.Restriction.Restriction.HindIII.size
Bio.Restriction.Restriction.HinfI.size
Bio.Restriction.Restriction.HpaI.size
Bio.Restriction.Restriction.HpaII.size
Bio.Restriction.Restriction.HphI.size
Bio.Restriction.Restriction.Hpy166II.size
Bio.Restriction.Restriction.Hpy178III.size
Bio.Restriction.Restriction.Hpy188I.size
Bio.Restriction.Restriction.Hpy188III.size
Bio.Restriction.Restriction.Hpy8I.size
Bio.Restriction.Restriction.Hpy99I.size
Bio.Restriction.Restriction.Hpy99XIII.size
Bio.Restriction.Restriction.Hpy99XIV.size
Bio.Restriction.Restriction.HpyAV.size
Bio.Restriction.Restriction.HpyCH4III.size
Bio.Restriction.Restriction.HpyCH4IV.size
Bio.Restriction.Restriction.HpyCH4V.size
Bio.Restriction.Restriction.HpyF10VI.size
Bio.Restriction.Restriction.HpyF3I.size
Bio.Restriction.Restriction.HpySE526I.size
Bio.Restriction.Restriction.Hsp92I.size
Bio.Restriction.Restriction.Hsp92II.size
Bio.Restriction.Restriction.HspAI.size
Bio.Restriction.Restriction.Jma19592I.size
Bio.Restriction.Restriction.KasI.size
Bio.Restriction.Restriction.KflI.size
Bio.Restriction.Restriction.Kpn2I.size
Bio.Restriction.Restriction.KpnI.size
Bio.Restriction.Restriction.KroI.size
Bio.Restriction.Restriction.Ksp22I.size
Bio.Restriction.Restriction.Ksp632I.size
Bio.Restriction.Restriction.KspAI.size
Bio.Restriction.Restriction.KspI.size
Bio.Restriction.Restriction.Kzo9I.size
Bio.Restriction.Restriction.LguI.size
Bio.Restriction.Restriction.LpnI.size
Bio.Restriction.Restriction.LpnPI.size
Bio.Restriction.Restriction.Lsp1109I.size
Bio.Restriction.Restriction.LweI.size
Bio.Restriction.Restriction.MabI.size
Bio.Restriction.Restriction.MaeI.size
Bio.Restriction.Restriction.MaeII.size
Bio.Restriction.Restriction.MaeIII.size
Bio.Restriction.Restriction.MalI.size
Bio.Restriction.Restriction.MaqI.size
Bio.Restriction.Restriction.MauBI.size
Bio.Restriction.Restriction.MbiI.size
Bio.Restriction.Restriction.MboI.size
Bio.Restriction.Restriction.MboII.size
Bio.Restriction.Restriction.McaTI.size
Bio.Restriction.Restriction.McrI.size
Bio.Restriction.Restriction.MfeI.size
Bio.Restriction.Restriction.MflI.size
Bio.Restriction.Restriction.MhlI.size
Bio.Restriction.Restriction.MjaIV.size
Bio.Restriction.Restriction.MkaDII.size
Bio.Restriction.Restriction.MlsI.size
Bio.Restriction.Restriction.MluCI.size
Bio.Restriction.Restriction.MluI.size
Bio.Restriction.Restriction.MluNI.size
Bio.Restriction.Restriction.Mly113I.size
Bio.Restriction.Restriction.MlyI.size
Bio.Restriction.Restriction.MmeI.size
Bio.Restriction.Restriction.MnlI.size
Bio.Restriction.Restriction.Mph1103I.size
Bio.Restriction.Restriction.MreI.size
Bio.Restriction.Restriction.MroI.size
Bio.Restriction.Restriction.MroNI.size
Bio.Restriction.Restriction.MroXI.size
Bio.Restriction.Restriction.MscI.size
Bio.Restriction.Restriction.MseI.size
Bio.Restriction.Restriction.MslI.size
Bio.Restriction.Restriction.Msp20I.size
Bio.Restriction.Restriction.MspA1I.size
Bio.Restriction.Restriction.MspCI.size
Bio.Restriction.Restriction.MspI.size
Bio.Restriction.Restriction.MspJI.size
Bio.Restriction.Restriction.MspR9I.size
Bio.Restriction.Restriction.MssI.size
Bio.Restriction.Restriction.MstI.size
Bio.Restriction.Restriction.MunI.size
Bio.Restriction.Restriction.Mva1269I.size
Bio.Restriction.Restriction.MvaI.size
Bio.Restriction.Restriction.MvnI.size
Bio.Restriction.Restriction.MvrI.size
Bio.Restriction.Restriction.MwoI.size
Bio.Restriction.Restriction.NaeI.size
Bio.Restriction.Restriction.NarI.size
Bio.Restriction.Restriction.NciI.size
Bio.Restriction.Restriction.NcoI.size
Bio.Restriction.Restriction.NdeI.size
Bio.Restriction.Restriction.NdeII.size
Bio.Restriction.Restriction.NgoAVIII.size
Bio.Restriction.Restriction.NgoMIV.size
Bio.Restriction.Restriction.NhaXI.size
Bio.Restriction.Restriction.NheI.size
Bio.Restriction.Restriction.NlaCI.size
Bio.Restriction.Restriction.NlaIII.size
Bio.Restriction.Restriction.NlaIV.size
Bio.Restriction.Restriction.Nli3877I.size
Bio.Restriction.Restriction.NmeAIII.size
Bio.Restriction.Restriction.NmeDI.size
Bio.Restriction.Restriction.NmuCI.size
Bio.Restriction.Restriction.NotI.size
Bio.Restriction.Restriction.NruI.size
Bio.Restriction.Restriction.NsbI.size
Bio.Restriction.Restriction.NsiI.size
Bio.Restriction.Restriction.NspBII.size
Bio.Restriction.Restriction.NspI.size
Bio.Restriction.Restriction.NspV.size
Bio.Restriction.Restriction.OliI.size
Bio.Restriction.Restriction.PabI.size
Bio.Restriction.Restriction.PacI.size
Bio.Restriction.Restriction.PaeI.size
Bio.Restriction.Restriction.PaeR7I.size
Bio.Restriction.Restriction.PagI.size
Bio.Restriction.Restriction.PalAI.size
Bio.Restriction.Restriction.PasI.size
Bio.Restriction.Restriction.PauI.size
Bio.Restriction.Restriction.PceI.size
Bio.Restriction.Restriction.PciI.size
Bio.Restriction.Restriction.PciSI.size
Bio.Restriction.Restriction.PcsI.size
Bio.Restriction.Restriction.PctI.size
Bio.Restriction.Restriction.PdiI.size
Bio.Restriction.Restriction.PdmI.size
Bio.Restriction.Restriction.PenI.size
Bio.Restriction.Restriction.PfeI.size
Bio.Restriction.Restriction.Pfl1108I.size
Bio.Restriction.Restriction.Pfl23II.size
Bio.Restriction.Restriction.PflFI.size
Bio.Restriction.Restriction.PflMI.size
Bio.Restriction.Restriction.PfoI.size
Bio.Restriction.Restriction.PinAI.size
Bio.Restriction.Restriction.PlaDI.size
Bio.Restriction.Restriction.Ple19I.size
Bio.Restriction.Restriction.PleI.size
Bio.Restriction.Restriction.PluTI.size
Bio.Restriction.Restriction.PmaCI.size
Bio.Restriction.Restriction.PmeI.size
Bio.Restriction.Restriction.PmlI.size
Bio.Restriction.Restriction.PpiI.size
Bio.Restriction.Restriction.PpsI.size
Bio.Restriction.Restriction.Ppu10I.size
Bio.Restriction.Restriction.Ppu21I.size
Bio.Restriction.Restriction.PpuMI.size
Bio.Restriction.Restriction.PscI.size
Bio.Restriction.Restriction.PshAI.size
Bio.Restriction.Restriction.PshBI.size
Bio.Restriction.Restriction.PsiI.size
Bio.Restriction.Restriction.Psp03I.size
Bio.Restriction.Restriction.Psp124BI.size
Bio.Restriction.Restriction.Psp1406I.size
Bio.Restriction.Restriction.Psp5II.size
Bio.Restriction.Restriction.Psp6I.size
Bio.Restriction.Restriction.PspCI.size
Bio.Restriction.Restriction.PspEI.size
Bio.Restriction.Restriction.PspGI.size
Bio.Restriction.Restriction.PspLI.size
Bio.Restriction.Restriction.PspN4I.size
Bio.Restriction.Restriction.PspOMI.size
Bio.Restriction.Restriction.PspOMII.size
Bio.Restriction.Restriction.PspPI.size
Bio.Restriction.Restriction.PspPPI.size
Bio.Restriction.Restriction.PspPRI.size
Bio.Restriction.Restriction.PspXI.size
Bio.Restriction.Restriction.PsrI.size
Bio.Restriction.Restriction.PssI.size
Bio.Restriction.Restriction.PstI.size
Bio.Restriction.Restriction.PstNI.size
Bio.Restriction.Restriction.PsuI.size
Bio.Restriction.Restriction.PsyI.size
Bio.Restriction.Restriction.PteI.size
Bio.Restriction.Restriction.PvuI.size
Bio.Restriction.Restriction.PvuII.size
Bio.Restriction.Restriction.R2_BceSIV.size
Bio.Restriction.Restriction.RceI.size
Bio.Restriction.Restriction.RdeGBI.size
Bio.Restriction.Restriction.RdeGBII.size
Bio.Restriction.Restriction.RdeGBIII.size
Bio.Restriction.Restriction.RflFIII.size
Bio.Restriction.Restriction.RgaI.size
Bio.Restriction.Restriction.RigI.size
Bio.Restriction.Restriction.RlaI.size
Bio.Restriction.Restriction.RleAI.size
Bio.Restriction.Restriction.RpaB5I.size
Bio.Restriction.Restriction.RpaBI.size
Bio.Restriction.Restriction.RpaI.size
Bio.Restriction.Restriction.RpaTI.size
Bio.Restriction.Restriction.RruI.size
Bio.Restriction.Restriction.RsaI.size
Bio.Restriction.Restriction.RsaNI.size
Bio.Restriction.Restriction.RseI.size
Bio.Restriction.Restriction.Rsr2I.size
Bio.Restriction.Restriction.RsrII.size
Bio.Restriction.Restriction.SacI.size
Bio.Restriction.Restriction.SacII.size
Bio.Restriction.Restriction.SalI.size
Bio.Restriction.Restriction.SanDI.size
Bio.Restriction.Restriction.SapI.size
Bio.Restriction.Restriction.SaqAI.size
Bio.Restriction.Restriction.SatI.size
Bio.Restriction.Restriction.Sau3AI.size
Bio.Restriction.Restriction.Sau96I.size
Bio.Restriction.Restriction.SauI.size
Bio.Restriction.Restriction.SbfI.size
Bio.Restriction.Restriction.ScaI.size
Bio.Restriction.Restriction.SchI.size
Bio.Restriction.Restriction.SciI.size
Bio.Restriction.Restriction.ScrFI.size
Bio.Restriction.Restriction.SdaI.size
Bio.Restriction.Restriction.SdeAI.size
Bio.Restriction.Restriction.SdeOSI.size
Bio.Restriction.Restriction.SduI.size
Bio.Restriction.Restriction.SecI.size
Bio.Restriction.Restriction.SelI.size
Bio.Restriction.Restriction.SetI.size
Bio.Restriction.Restriction.SexAI.size
Bio.Restriction.Restriction.SfaAI.size
Bio.Restriction.Restriction.SfaNI.size
Bio.Restriction.Restriction.SfcI.size
Bio.Restriction.Restriction.SfeI.size
Bio.Restriction.Restriction.SfiI.size
Bio.Restriction.Restriction.SfoI.size
Bio.Restriction.Restriction.Sfr274I.size
Bio.Restriction.Restriction.Sfr303I.size
Bio.Restriction.Restriction.SfuI.size
Bio.Restriction.Restriction.SgeI.size
Bio.Restriction.Restriction.SgfI.size
Bio.Restriction.Restriction.SgrAI.size
Bio.Restriction.Restriction.SgrBI.size
Bio.Restriction.Restriction.SgrDI.size
Bio.Restriction.Restriction.SgrTI.size
Bio.Restriction.Restriction.SgsI.size
Bio.Restriction.Restriction.SimI.size
Bio.Restriction.Restriction.SlaI.size
Bio.Restriction.Restriction.SmaI.size
Bio.Restriction.Restriction.SmiI.size
Bio.Restriction.Restriction.SmiMI.size
Bio.Restriction.Restriction.SmlI.size
Bio.Restriction.Restriction.SmoI.size
Bio.Restriction.Restriction.SnaBI.size
Bio.Restriction.Restriction.SnaI.size
Bio.Restriction.Restriction.Sno506I.size
Bio.Restriction.Restriction.SpeI.size
Bio.Restriction.Restriction.SphI.size
Bio.Restriction.Restriction.SplI.size
Bio.Restriction.Restriction.SpoDI.size
Bio.Restriction.Restriction.SrfI.size
Bio.Restriction.Restriction.Sse232I.size
Bio.Restriction.Restriction.Sse8387I.size
Bio.Restriction.Restriction.Sse8647I.size
Bio.Restriction.Restriction.Sse9I.size
Bio.Restriction.Restriction.SseBI.size
Bio.Restriction.Restriction.SsiI.size
Bio.Restriction.Restriction.SspD5I.size
Bio.Restriction.Restriction.SspDI.size
Bio.Restriction.Restriction.SspI.size
Bio.Restriction.Restriction.SstE37I.size
Bio.Restriction.Restriction.SstI.size
Bio.Restriction.Restriction.Sth132I.size
Bio.Restriction.Restriction.Sth302II.size
Bio.Restriction.Restriction.StrI.size
Bio.Restriction.Restriction.StsI.size
Bio.Restriction.Restriction.StuI.size
Bio.Restriction.Restriction.StyD4I.size
Bio.Restriction.Restriction.StyI.size
Bio.Restriction.Restriction.SwaI.size
Bio.Restriction.Restriction.TaaI.size
Bio.Restriction.Restriction.TaiI.size
Bio.Restriction.Restriction.TaqI.size
Bio.Restriction.Restriction.TaqII.size
Bio.Restriction.Restriction.TasI.size
Bio.Restriction.Restriction.TatI.size
Bio.Restriction.Restriction.TauI.size
Bio.Restriction.Restriction.TfiI.size
Bio.Restriction.Restriction.Tru1I.size
Bio.Restriction.Restriction.Tru9I.size
Bio.Restriction.Restriction.TscAI.size
Bio.Restriction.Restriction.TseFI.size
Bio.Restriction.Restriction.TseI.size
Bio.Restriction.Restriction.TsoI.size
Bio.Restriction.Restriction.Tsp45I.size
Bio.Restriction.Restriction.Tsp4CI.size
Bio.Restriction.Restriction.TspDTI.size
Bio.Restriction.Restriction.TspEI.size
Bio.Restriction.Restriction.TspGWI.size
Bio.Restriction.Restriction.TspMI.size
Bio.Restriction.Restriction.TspRI.size
Bio.Restriction.Restriction.TssI.size
Bio.Restriction.Restriction.TstI.size
Bio.Restriction.Restriction.TsuI.size
Bio.Restriction.Restriction.Tth111I.size
Bio.Restriction.Restriction.Tth111II.size
Bio.Restriction.Restriction.UbaF11I.size
Bio.Restriction.Restriction.UbaF12I.size
Bio.Restriction.Restriction.UbaF13I.size
Bio.Restriction.Restriction.UbaF14I.size
Bio.Restriction.Restriction.UbaF9I.size
Bio.Restriction.Restriction.UbaPI.size
Bio.Restriction.Restriction.UcoMSI.size
Bio.Restriction.Restriction.UnbI.size
Bio.Restriction.Restriction.Van91I.size
Bio.Restriction.Restriction.Vha464I.size
Bio.Restriction.Restriction.VneI.size
Bio.Restriction.Restriction.VpaK11AI.size
Bio.Restriction.Restriction.VpaK11BI.size
Bio.Restriction.Restriction.VspI.size
Bio.Restriction.Restriction.WviI.size
Bio.Restriction.Restriction.XagI.size
Bio.Restriction.Restriction.XapI.size
Bio.Restriction.Restriction.XbaI.size
Bio.Restriction.Restriction.XceI.size
Bio.Restriction.Restriction.XcmI.size
Bio.Restriction.Restriction.XhoI.size
Bio.Restriction.Restriction.XhoII.size
Bio.Restriction.Restriction.XmaI.size
Bio.Restriction.Restriction.XmaIII.size
Bio.Restriction.Restriction.XmaJI.size
Bio.Restriction.Restriction.XmiI.size
Bio.Restriction.Restriction.XmnI.size
Bio.Restriction.Restriction.XspI.size
Bio.Restriction.Restriction.YkrI.size
Bio.Restriction.Restriction.ZraI.size
Bio.Restriction.Restriction.ZrmI.size
Bio.Restriction.Restriction.Zsp2I.size
Bio.codonalign.codonalphabet.CodonAlphabet.size" class="py-name" href="#" onclick="return doclink('link-432', 'size', 'link-270');">size</a></tt><tt class="py-op">(</tt><tt class="py-name">name_and_length</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1088"></a><tt class="py-lineno">1088</tt>  <tt class="py-line">            <tt class="py-comment"># consumer.residue_type(line[33:41].strip())</tt> </tt>
<a name="L1089"></a><tt class="py-lineno">1089</tt>  <tt class="py-line"> </tt>
<a name="L1090"></a><tt class="py-lineno">1090</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">33</tt><tt class="py-op">:</tt><tt class="py-number">51</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-433" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-433', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">""</tt> <tt class="py-keyword">and</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">29</tt><tt class="py-op">:</tt><tt class="py-number">33</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">' aa '</tt><tt class="py-op">:</tt> </tt>
<a name="L1091"></a><tt class="py-lineno">1091</tt>  <tt class="py-line">                <tt class="py-comment"># Amino acids -&gt; protein (even if there is no residue type given)</tt> </tt>
<a name="L1092"></a><tt class="py-lineno">1092</tt>  <tt class="py-line">                <tt class="py-comment"># We want to use a protein alphabet in this case, rather than a</tt> </tt>
<a name="L1093"></a><tt class="py-lineno">1093</tt>  <tt class="py-line">                <tt class="py-comment"># generic one. Not sure if this is the best way to achieve this,</tt> </tt>
<a name="L1094"></a><tt class="py-lineno">1094</tt>  <tt class="py-line">                <tt class="py-comment"># but it works because the scanner checks for this:</tt> </tt>
<a name="L1095"></a><tt class="py-lineno">1095</tt>  <tt class="py-line">                <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-434" class="py-name"><a title="Bio.GenBank._FeatureConsumer.residue_type
Bio.GenBank._RecordConsumer.residue_type" class="py-name" href="#" onclick="return doclink('link-434', 'residue_type', 'link-236');">residue_type</a></tt><tt class="py-op">(</tt><tt class="py-string">"PROTEIN"</tt><tt class="py-op">)</tt> </tt>
<a name="L1096"></a><tt class="py-lineno">1096</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1097"></a><tt class="py-lineno">1097</tt>  <tt class="py-line">                <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-435" class="py-name"><a title="Bio.GenBank._FeatureConsumer.residue_type
Bio.GenBank._RecordConsumer.residue_type" class="py-name" href="#" onclick="return doclink('link-435', 'residue_type', 'link-236');">residue_type</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">33</tt><tt class="py-op">:</tt><tt class="py-number">51</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-436" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-436', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1098"></a><tt class="py-lineno">1098</tt>  <tt class="py-line"> </tt>
<a name="L1099"></a><tt class="py-lineno">1099</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-437" class="py-name"><a title="Bio.GenBank._FeatureConsumer.data_file_division
Bio.GenBank._RecordConsumer.data_file_division" class="py-name" href="#" onclick="return doclink('link-437', 'data_file_division', 'link-237');">data_file_division</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">52</tt><tt class="py-op">:</tt><tt class="py-number">55</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1100"></a><tt class="py-lineno">1100</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">62</tt><tt class="py-op">:</tt><tt class="py-number">73</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-438" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-438', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1101"></a><tt class="py-lineno">1101</tt>  <tt class="py-line">                <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-439" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.date()=Bio.GenBank._FeatureConsumer-class.html#date,Method Bio.GenBank._RecordConsumer.date()=Bio.GenBank._RecordConsumer-class.html#date,Method Bio.Phylo.PhyloXMLIO.Parser.date()=Bio.Phylo.PhyloXMLIO.Parser-class.html#date,Method Bio.Phylo.PhyloXMLIO.Writer.date()=Bio.Phylo.PhyloXMLIO.Writer-class.html#date"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-439', 'date', 'link-439');">date</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">62</tt><tt class="py-op">:</tt><tt class="py-number">73</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1102"></a><tt class="py-lineno">1102</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">40</tt><tt class="py-op">:</tt><tt class="py-number">44</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">' bp '</tt><tt class="py-op">,</tt> <tt class="py-string">' aa '</tt><tt class="py-op">,</tt> <tt class="py-string">' rc '</tt><tt class="py-op">]</tt> \ </tt>
<a name="L1103"></a><tt class="py-lineno">1103</tt>  <tt class="py-line">                <tt class="py-keyword">and</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">54</tt><tt class="py-op">:</tt><tt class="py-number">64</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-440" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-440', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">''</tt><tt class="py-op">,</tt> <tt class="py-string">'linear'</tt><tt class="py-op">,</tt> <tt class="py-string">'circular'</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L1104"></a><tt class="py-lineno">1104</tt>  <tt class="py-line">            <tt class="py-comment"># New... linear/circular/big blank test should avoid EnsEMBL style</tt> </tt>
<a name="L1105"></a><tt class="py-lineno">1105</tt>  <tt class="py-line">            <tt class="py-comment"># LOCUS line being treated like a proper column based LOCUS line.</tt> </tt>
<a name="L1106"></a><tt class="py-lineno">1106</tt>  <tt class="py-line">            <tt class="py-comment">#</tt> </tt>
<a name="L1107"></a><tt class="py-lineno">1107</tt>  <tt class="py-line">            <tt class="py-comment">#    Positions  Contents</tt> </tt>
<a name="L1108"></a><tt class="py-lineno">1108</tt>  <tt class="py-line">            <tt class="py-comment">#    ---------  --------</tt> </tt>
<a name="L1109"></a><tt class="py-lineno">1109</tt>  <tt class="py-line">            <tt class="py-comment">#    00:06      LOCUS</tt> </tt>
<a name="L1110"></a><tt class="py-lineno">1110</tt>  <tt class="py-line">            <tt class="py-comment">#    06:12      spaces</tt> </tt>
<a name="L1111"></a><tt class="py-lineno">1111</tt>  <tt class="py-line">            <tt class="py-comment">#    12:??      Locus name</tt> </tt>
<a name="L1112"></a><tt class="py-lineno">1112</tt>  <tt class="py-line">            <tt class="py-comment">#    ??:??      space</tt> </tt>
<a name="L1113"></a><tt class="py-lineno">1113</tt>  <tt class="py-line">            <tt class="py-comment">#    ??:40      Length of sequence, right-justified</tt> </tt>
<a name="L1114"></a><tt class="py-lineno">1114</tt>  <tt class="py-line">            <tt class="py-comment">#    40:44      space, bp, space</tt> </tt>
<a name="L1115"></a><tt class="py-lineno">1115</tt>  <tt class="py-line">            <tt class="py-comment">#    44:47      Blank, ss-, ds-, ms-</tt> </tt>
<a name="L1116"></a><tt class="py-lineno">1116</tt>  <tt class="py-line">            <tt class="py-comment">#    47:54      Blank, DNA, RNA, tRNA, mRNA, uRNA, snRNA, cDNA</tt> </tt>
<a name="L1117"></a><tt class="py-lineno">1117</tt>  <tt class="py-line">            <tt class="py-comment">#    54:55      space</tt> </tt>
<a name="L1118"></a><tt class="py-lineno">1118</tt>  <tt class="py-line">            <tt class="py-comment">#    55:63      Blank (implies linear), linear or circular</tt> </tt>
<a name="L1119"></a><tt class="py-lineno">1119</tt>  <tt class="py-line">            <tt class="py-comment">#    63:64      space</tt> </tt>
<a name="L1120"></a><tt class="py-lineno">1120</tt>  <tt class="py-line">            <tt class="py-comment">#    64:67      The division code (e.g. BCT, VRL, INV)</tt> </tt>
<a name="L1121"></a><tt class="py-lineno">1121</tt>  <tt class="py-line">            <tt class="py-comment">#    67:68      space</tt> </tt>
<a name="L1122"></a><tt class="py-lineno">1122</tt>  <tt class="py-line">            <tt class="py-comment">#    68:79      Date, in the form dd-MMM-yyyy (e.g., 15-MAR-1991)</tt> </tt>
<a name="L1123"></a><tt class="py-lineno">1123</tt>  <tt class="py-line">            <tt class="py-comment">#</tt> </tt>
<a name="L1124"></a><tt class="py-lineno">1124</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">40</tt><tt class="py-op">:</tt><tt class="py-number">44</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">' bp '</tt><tt class="py-op">,</tt> <tt class="py-string">' aa '</tt><tt class="py-op">,</tt> <tt class="py-string">' rc '</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> \ </tt>
<a name="L1125"></a><tt class="py-lineno">1125</tt>  <tt class="py-line">                <tt class="py-string">'LOCUS line does not contain size units at expected position:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L1126"></a><tt class="py-lineno">1126</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">44</tt><tt class="py-op">:</tt><tt class="py-number">47</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">'   '</tt><tt class="py-op">,</tt> <tt class="py-string">'ss-'</tt><tt class="py-op">,</tt> <tt class="py-string">'ds-'</tt><tt class="py-op">,</tt> <tt class="py-string">'ms-'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> \ </tt>
<a name="L1127"></a><tt class="py-lineno">1127</tt>  <tt class="py-line">                <tt class="py-string">'LOCUS line does not have valid strand type (Single stranded, ...):\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L1128"></a><tt class="py-lineno">1128</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">47</tt><tt class="py-op">:</tt><tt class="py-number">54</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-441" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-441', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">""</tt> \ </tt>
<a name="L1129"></a><tt class="py-lineno">1129</tt>  <tt class="py-line">                <tt class="py-keyword">or</tt> <tt class="py-string">'DNA'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">47</tt><tt class="py-op">:</tt><tt class="py-number">54</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-442" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-442', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-443" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-443', 'upper', 'link-269');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> \ </tt>
<a name="L1130"></a><tt class="py-lineno">1130</tt>  <tt class="py-line">                <tt class="py-keyword">or</tt> <tt class="py-string">'RNA'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">47</tt><tt class="py-op">:</tt><tt class="py-number">54</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-444" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-444', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-445" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-445', 'upper', 'link-269');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> \ </tt>
<a name="L1131"></a><tt class="py-lineno">1131</tt>  <tt class="py-line">                   <tt class="py-string">'LOCUS line does not contain valid sequence type (DNA, RNA, ...):\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L1132"></a><tt class="py-lineno">1132</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">54</tt><tt class="py-op">:</tt><tt class="py-number">55</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">' '</tt><tt class="py-op">,</tt> \ </tt>
<a name="L1133"></a><tt class="py-lineno">1133</tt>  <tt class="py-line">                <tt class="py-string">'LOCUS line does not contain space at position 55:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L1134"></a><tt class="py-lineno">1134</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">55</tt><tt class="py-op">:</tt><tt class="py-number">63</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-446" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-446', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">''</tt><tt class="py-op">,</tt> <tt class="py-string">'linear'</tt><tt class="py-op">,</tt> <tt class="py-string">'circular'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> \ </tt>
<a name="L1135"></a><tt class="py-lineno">1135</tt>  <tt class="py-line">                <tt class="py-string">'LOCUS line does not contain valid entry (linear, circular, ...):\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L1136"></a><tt class="py-lineno">1136</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">63</tt><tt class="py-op">:</tt><tt class="py-number">64</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">' '</tt><tt class="py-op">,</tt> \ </tt>
<a name="L1137"></a><tt class="py-lineno">1137</tt>  <tt class="py-line">                <tt class="py-string">'LOCUS line does not contain space at position 64:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L1138"></a><tt class="py-lineno">1138</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">67</tt><tt class="py-op">:</tt><tt class="py-number">68</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">' '</tt><tt class="py-op">,</tt> \ </tt>
<a name="L1139"></a><tt class="py-lineno">1139</tt>  <tt class="py-line">                <tt class="py-string">'LOCUS line does not contain space at position 68:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L1140"></a><tt class="py-lineno">1140</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">68</tt><tt class="py-op">:</tt><tt class="py-number">79</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-447" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-447', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1141"></a><tt class="py-lineno">1141</tt>  <tt class="py-line">                <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">70</tt><tt class="py-op">:</tt><tt class="py-number">71</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'-'</tt><tt class="py-op">,</tt> \ </tt>
<a name="L1142"></a><tt class="py-lineno">1142</tt>  <tt class="py-line">                    <tt class="py-string">'LOCUS line does not contain - at position 71 in date:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L1143"></a><tt class="py-lineno">1143</tt>  <tt class="py-line">                <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">74</tt><tt class="py-op">:</tt><tt class="py-number">75</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'-'</tt><tt class="py-op">,</tt> \ </tt>
<a name="L1144"></a><tt class="py-lineno">1144</tt>  <tt class="py-line">                    <tt class="py-string">'LOCUS line does not contain - at position 75 in date:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L1145"></a><tt class="py-lineno">1145</tt>  <tt class="py-line"> </tt>
<a name="L1146"></a><tt class="py-lineno">1146</tt>  <tt class="py-line">            <tt class="py-name">name_and_length_str</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt id="link-448" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-448', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt><tt class="py-op">:</tt><tt class="py-number">40</tt><tt class="py-op">]</tt> </tt>
<a name="L1147"></a><tt class="py-lineno">1147</tt>  <tt class="py-line">            <tt class="py-keyword">while</tt> <tt class="py-string">'  '</tt> <tt class="py-keyword">in</tt> <tt class="py-name">name_and_length_str</tt><tt class="py-op">:</tt> </tt>
<a name="L1148"></a><tt class="py-lineno">1148</tt>  <tt class="py-line">                <tt class="py-name">name_and_length_str</tt> <tt class="py-op">=</tt> <tt class="py-name">name_and_length_str</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'  '</tt><tt class="py-op">,</tt> <tt class="py-string">' '</tt><tt class="py-op">)</tt> </tt>
<a name="L1149"></a><tt class="py-lineno">1149</tt>  <tt class="py-line">            <tt class="py-name">name_and_length</tt> <tt class="py-op">=</tt> <tt class="py-name">name_and_length_str</tt><tt class="py-op">.</tt><tt id="link-449" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-449', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">' '</tt><tt class="py-op">)</tt> </tt>
<a name="L1150"></a><tt class="py-lineno">1150</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">name_and_length</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;=</tt> <tt class="py-number">2</tt><tt class="py-op">,</tt> \ </tt>
<a name="L1151"></a><tt class="py-lineno">1151</tt>  <tt class="py-line">                <tt class="py-string">'Cannot parse the name and length in the LOCUS line:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L1152"></a><tt class="py-lineno">1152</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">name_and_length</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> \ </tt>
<a name="L1153"></a><tt class="py-lineno">1153</tt>  <tt class="py-line">                <tt class="py-string">'Name and length collide in the LOCUS line:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L1154"></a><tt class="py-lineno">1154</tt>  <tt class="py-line">            <tt class="py-comment"># Should be possible to split them based on position, if</tt> </tt>
<a name="L1155"></a><tt class="py-lineno">1155</tt>  <tt class="py-line">            <tt class="py-comment"># a clear definition of the stand exists THAT AGREES with</tt> </tt>
<a name="L1156"></a><tt class="py-lineno">1156</tt>  <tt class="py-line">            <tt class="py-comment"># existing files.</tt> </tt>
<a name="L1157"></a><tt class="py-lineno">1157</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-450" class="py-name"><a title="Bio.GenBank._FeatureConsumer.locus
Bio.GenBank._RecordConsumer.locus" class="py-name" href="#" onclick="return doclink('link-450', 'locus', 'link-235');">locus</a></tt><tt class="py-op">(</tt><tt class="py-name">name_and_length</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1158"></a><tt class="py-lineno">1158</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-451" class="py-name"><a title="Bio.Alphabet.Alphabet.size
Bio.Alphabet.Reduced.HPModel.size
Bio.Alphabet.Reduced.Murphy10.size
Bio.Alphabet.Reduced.Murphy15.size
Bio.Alphabet.Reduced.Murphy4.size
Bio.Alphabet.Reduced.Murphy8.size
Bio.Alphabet.Reduced.PC5.size
Bio.Alphabet.SingleLetterAlphabet.size
Bio.Alphabet.ThreeLetterProtein.size
Bio.GenBank._FeatureConsumer.size
Bio.GenBank._RecordConsumer.size
Bio.Restriction.Restriction.AanI.size
Bio.Restriction.Restriction.AarI.size
Bio.Restriction.Restriction.AasI.size
Bio.Restriction.Restriction.AatII.size
Bio.Restriction.Restriction.AbaSI.size
Bio.Restriction.Restriction.AbsI.size
Bio.Restriction.Restriction.Acc16I.size
Bio.Restriction.Restriction.Acc36I.size
Bio.Restriction.Restriction.Acc65I.size
Bio.Restriction.Restriction.AccB1I.size
Bio.Restriction.Restriction.AccB7I.size
Bio.Restriction.Restriction.AccBSI.size
Bio.Restriction.Restriction.AccI.size
Bio.Restriction.Restriction.AccII.size
Bio.Restriction.Restriction.AccIII.size
Bio.Restriction.Restriction.AceIII.size
Bio.Restriction.Restriction.AciI.size
Bio.Restriction.Restriction.AclI.size
Bio.Restriction.Restriction.AclWI.size
Bio.Restriction.Restriction.AcoI.size
Bio.Restriction.Restriction.AcsI.size
Bio.Restriction.Restriction.AcuI.size
Bio.Restriction.Restriction.AcvI.size
Bio.Restriction.Restriction.AcyI.size
Bio.Restriction.Restriction.AdeI.size
Bio.Restriction.Restriction.AfaI.size
Bio.Restriction.Restriction.AfeI.size
Bio.Restriction.Restriction.AfiI.size
Bio.Restriction.Restriction.AflII.size
Bio.Restriction.Restriction.AflIII.size
Bio.Restriction.Restriction.AgeI.size
Bio.Restriction.Restriction.AgsI.size
Bio.Restriction.Restriction.AhaIII.size
Bio.Restriction.Restriction.AhdI.size
Bio.Restriction.Restriction.AhlI.size
Bio.Restriction.Restriction.AjiI.size
Bio.Restriction.Restriction.AjnI.size
Bio.Restriction.Restriction.AjuI.size
Bio.Restriction.Restriction.AleI.size
Bio.Restriction.Restriction.AlfI.size
Bio.Restriction.Restriction.AloI.size
Bio.Restriction.Restriction.AluBI.size
Bio.Restriction.Restriction.AluI.size
Bio.Restriction.Restriction.Alw21I.size
Bio.Restriction.Restriction.Alw26I.size
Bio.Restriction.Restriction.Alw44I.size
Bio.Restriction.Restriction.AlwFI.size
Bio.Restriction.Restriction.AlwI.size
Bio.Restriction.Restriction.AlwNI.size
Bio.Restriction.Restriction.Ama87I.size
Bio.Restriction.Restriction.Aor13HI.size
Bio.Restriction.Restriction.Aor51HI.size
Bio.Restriction.Restriction.AoxI.size
Bio.Restriction.Restriction.ApaBI.size
Bio.Restriction.Restriction.ApaI.size
Bio.Restriction.Restriction.ApaLI.size
Bio.Restriction.Restriction.ApeKI.size
Bio.Restriction.Restriction.ApoI.size
Bio.Restriction.Restriction.ApyPI.size
Bio.Restriction.Restriction.AquII.size
Bio.Restriction.Restriction.AquIII.size
Bio.Restriction.Restriction.AquIV.size
Bio.Restriction.Restriction.ArsI.size
Bio.Restriction.Restriction.AscI.size
Bio.Restriction.Restriction.AseI.size
Bio.Restriction.Restriction.Asi256I.size
Bio.Restriction.Restriction.AsiGI.size
Bio.Restriction.Restriction.AsiSI.size
Bio.Restriction.Restriction.Asp700I.size
Bio.Restriction.Restriction.Asp718I.size
Bio.Restriction.Restriction.AspA2I.size
Bio.Restriction.Restriction.AspBHI.size
Bio.Restriction.Restriction.AspLEI.size
Bio.Restriction.Restriction.AspS9I.size
Bio.Restriction.Restriction.AssI.size
Bio.Restriction.Restriction.AsuC2I.size
Bio.Restriction.Restriction.AsuHPI.size
Bio.Restriction.Restriction.AsuI.size
Bio.Restriction.Restriction.AsuII.size
Bio.Restriction.Restriction.AsuNHI.size
Bio.Restriction.Restriction.AvaI.size
Bio.Restriction.Restriction.AvaII.size
Bio.Restriction.Restriction.AvaIII.size
Bio.Restriction.Restriction.AvrII.size
Bio.Restriction.Restriction.AxyI.size
Bio.Restriction.Restriction.BaeGI.size
Bio.Restriction.Restriction.BaeI.size
Bio.Restriction.Restriction.BalI.size
Bio.Restriction.Restriction.BamHI.size
Bio.Restriction.Restriction.BanI.size
Bio.Restriction.Restriction.BanII.size
Bio.Restriction.Restriction.BarI.size
Bio.Restriction.Restriction.BasI.size
Bio.Restriction.Restriction.BauI.size
Bio.Restriction.Restriction.Bbr7I.size
Bio.Restriction.Restriction.BbrPI.size
Bio.Restriction.Restriction.BbsI.size
Bio.Restriction.Restriction.Bbv12I.size
Bio.Restriction.Restriction.BbvCI.size
Bio.Restriction.Restriction.BbvI.size
Bio.Restriction.Restriction.BbvII.size
Bio.Restriction.Restriction.BccI.size
Bio.Restriction.Restriction.Bce83I.size
Bio.Restriction.Restriction.BceAI.size
Bio.Restriction.Restriction.BcefI.size
Bio.Restriction.Restriction.BcgI.size
Bio.Restriction.Restriction.BciT130I.size
Bio.Restriction.Restriction.BciVI.size
Bio.Restriction.Restriction.BclI.size
Bio.Restriction.Restriction.BcnI.size
Bio.Restriction.Restriction.BcoDI.size
Bio.Restriction.Restriction.BcuI.size
Bio.Restriction.Restriction.BdaI.size
Bio.Restriction.Restriction.BetI.size
Bio.Restriction.Restriction.BfaI.size
Bio.Restriction.Restriction.BfiI.size
Bio.Restriction.Restriction.BfmI.size
Bio.Restriction.Restriction.BfoI.size
Bio.Restriction.Restriction.BfrI.size
Bio.Restriction.Restriction.BfuAI.size
Bio.Restriction.Restriction.BfuCI.size
Bio.Restriction.Restriction.BfuI.size
Bio.Restriction.Restriction.BglI.size
Bio.Restriction.Restriction.BglII.size
Bio.Restriction.Restriction.BinI.size
Bio.Restriction.Restriction.BisI.size
Bio.Restriction.Restriction.BlnI.size
Bio.Restriction.Restriction.BlpI.size
Bio.Restriction.Restriction.BlsI.size
Bio.Restriction.Restriction.BmcAI.size
Bio.Restriction.Restriction.Bme1390I.size
Bio.Restriction.Restriction.Bme18I.size
Bio.Restriction.Restriction.BmeDI.size
Bio.Restriction.Restriction.BmeRI.size
Bio.Restriction.Restriction.BmeT110I.size
Bio.Restriction.Restriction.BmgBI.size
Bio.Restriction.Restriction.BmgI.size
Bio.Restriction.Restriction.BmgT120I.size
Bio.Restriction.Restriction.BmiI.size
Bio.Restriction.Restriction.BmrFI.size
Bio.Restriction.Restriction.BmrI.size
Bio.Restriction.Restriction.BmsI.size
Bio.Restriction.Restriction.BmtI.size
Bio.Restriction.Restriction.BmuI.size
Bio.Restriction.Restriction.BoxI.size
Bio.Restriction.Restriction.BpiI.size
Bio.Restriction.Restriction.BplI.size
Bio.Restriction.Restriction.BpmI.size
Bio.Restriction.Restriction.Bpu10I.size
Bio.Restriction.Restriction.Bpu1102I.size
Bio.Restriction.Restriction.Bpu14I.size
Bio.Restriction.Restriction.BpuEI.size
Bio.Restriction.Restriction.BpuMI.size
Bio.Restriction.Restriction.BpvUI.size
Bio.Restriction.Restriction.Bsa29I.size
Bio.Restriction.Restriction.BsaAI.size
Bio.Restriction.Restriction.BsaBI.size
Bio.Restriction.Restriction.BsaHI.size
Bio.Restriction.Restriction.BsaI.size
Bio.Restriction.Restriction.BsaJI.size
Bio.Restriction.Restriction.BsaWI.size
Bio.Restriction.Restriction.BsaXI.size
Bio.Restriction.Restriction.BsbI.size
Bio.Restriction.Restriction.Bsc4I.size
Bio.Restriction.Restriction.BscAI.size
Bio.Restriction.Restriction.BscGI.size
Bio.Restriction.Restriction.Bse118I.size
Bio.Restriction.Restriction.Bse1I.size
Bio.Restriction.Restriction.Bse21I.size
Bio.Restriction.Restriction.Bse3DI.size
Bio.Restriction.Restriction.Bse8I.size
Bio.Restriction.Restriction.BseAI.size
Bio.Restriction.Restriction.BseBI.size
Bio.Restriction.Restriction.BseCI.size
Bio.Restriction.Restriction.BseDI.size
Bio.Restriction.Restriction.BseGI.size
Bio.Restriction.Restriction.BseJI.size
Bio.Restriction.Restriction.BseLI.size
Bio.Restriction.Restriction.BseMI.size
Bio.Restriction.Restriction.BseMII.size
Bio.Restriction.Restriction.BseNI.size
Bio.Restriction.Restriction.BsePI.size
Bio.Restriction.Restriction.BseRI.size
Bio.Restriction.Restriction.BseSI.size
Bio.Restriction.Restriction.BseX3I.size
Bio.Restriction.Restriction.BseXI.size
Bio.Restriction.Restriction.BseYI.size
Bio.Restriction.Restriction.BsgI.size
Bio.Restriction.Restriction.Bsh1236I.size
Bio.Restriction.Restriction.Bsh1285I.size
Bio.Restriction.Restriction.BshFI.size
Bio.Restriction.Restriction.BshNI.size
Bio.Restriction.Restriction.BshTI.size
Bio.Restriction.Restriction.BshVI.size
Bio.Restriction.Restriction.BsiEI.size
Bio.Restriction.Restriction.BsiHKAI.size
Bio.Restriction.Restriction.BsiHKCI.size
Bio.Restriction.Restriction.BsiI.size
Bio.Restriction.Restriction.BsiSI.size
Bio.Restriction.Restriction.BsiWI.size
Bio.Restriction.Restriction.BsiYI.size
Bio.Restriction.Restriction.BslFI.size
Bio.Restriction.Restriction.BslI.size
Bio.Restriction.Restriction.BsmAI.size
Bio.Restriction.Restriction.BsmBI.size
Bio.Restriction.Restriction.BsmFI.size
Bio.Restriction.Restriction.BsmI.size
Bio.Restriction.Restriction.BsnI.size
Bio.Restriction.Restriction.Bso31I.size
Bio.Restriction.Restriction.BsoBI.size
Bio.Restriction.Restriction.Bsp119I.size
Bio.Restriction.Restriction.Bsp120I.size
Bio.Restriction.Restriction.Bsp1286I.size
Bio.Restriction.Restriction.Bsp13I.size
Bio.Restriction.Restriction.Bsp1407I.size
Bio.Restriction.Restriction.Bsp143I.size
Bio.Restriction.Restriction.Bsp1720I.size
Bio.Restriction.Restriction.Bsp19I.size
Bio.Restriction.Restriction.Bsp24I.size
Bio.Restriction.Restriction.Bsp68I.size
Bio.Restriction.Restriction.BspACI.size
Bio.Restriction.Restriction.BspCNI.size
Bio.Restriction.Restriction.BspD6I.size
Bio.Restriction.Restriction.BspDI.size
Bio.Restriction.Restriction.BspEI.size
Bio.Restriction.Restriction.BspFNI.size
Bio.Restriction.Restriction.BspGI.size
Bio.Restriction.Restriction.BspHI.size
Bio.Restriction.Restriction.BspLI.size
Bio.Restriction.Restriction.BspLU11I.size
Bio.Restriction.Restriction.BspMI.size
Bio.Restriction.Restriction.BspMII.size
Bio.Restriction.Restriction.BspNCI.size
Bio.Restriction.Restriction.BspOI.size
Bio.Restriction.Restriction.BspPI.size
Bio.Restriction.Restriction.BspQI.size
Bio.Restriction.Restriction.BspT104I.size
Bio.Restriction.Restriction.BspT107I.size
Bio.Restriction.Restriction.BspTI.size
Bio.Restriction.Restriction.BsrBI.size
Bio.Restriction.Restriction.BsrDI.size
Bio.Restriction.Restriction.BsrFI.size
Bio.Restriction.Restriction.BsrGI.size
Bio.Restriction.Restriction.BsrI.size
Bio.Restriction.Restriction.BsrSI.size
Bio.Restriction.Restriction.BssAI.size
Bio.Restriction.Restriction.BssECI.size
Bio.Restriction.Restriction.BssHII.size
Bio.Restriction.Restriction.BssKI.size
Bio.Restriction.Restriction.BssMI.size
Bio.Restriction.Restriction.BssNAI.size
Bio.Restriction.Restriction.BssNI.size
Bio.Restriction.Restriction.BssSI.size
Bio.Restriction.Restriction.BssT1I.size
Bio.Restriction.Restriction.Bst1107I.size
Bio.Restriction.Restriction.Bst2BI.size
Bio.Restriction.Restriction.Bst2UI.size
Bio.Restriction.Restriction.Bst4CI.size
Bio.Restriction.Restriction.Bst6I.size
Bio.Restriction.Restriction.BstACI.size
Bio.Restriction.Restriction.BstAFI.size
Bio.Restriction.Restriction.BstAPI.size
Bio.Restriction.Restriction.BstAUI.size
Bio.Restriction.Restriction.BstBAI.size
Bio.Restriction.Restriction.BstBI.size
Bio.Restriction.Restriction.BstC8I.size
Bio.Restriction.Restriction.BstDEI.size
Bio.Restriction.Restriction.BstDSI.size
Bio.Restriction.Restriction.BstEII.size
Bio.Restriction.Restriction.BstENI.size
Bio.Restriction.Restriction.BstF5I.size
Bio.Restriction.Restriction.BstFNI.size
Bio.Restriction.Restriction.BstH2I.size
Bio.Restriction.Restriction.BstHHI.size
Bio.Restriction.Restriction.BstKTI.size
Bio.Restriction.Restriction.BstMAI.size
Bio.Restriction.Restriction.BstMBI.size
Bio.Restriction.Restriction.BstMCI.size
Bio.Restriction.Restriction.BstMWI.size
Bio.Restriction.Restriction.BstNI.size
Bio.Restriction.Restriction.BstNSI.size
Bio.Restriction.Restriction.BstOI.size
Bio.Restriction.Restriction.BstPAI.size
Bio.Restriction.Restriction.BstPI.size
Bio.Restriction.Restriction.BstSCI.size
Bio.Restriction.Restriction.BstSFI.size
Bio.Restriction.Restriction.BstSLI.size
Bio.Restriction.Restriction.BstSNI.size
Bio.Restriction.Restriction.BstUI.size
Bio.Restriction.Restriction.BstV1I.size
Bio.Restriction.Restriction.BstV2I.size
Bio.Restriction.Restriction.BstX2I.size
Bio.Restriction.Restriction.BstXI.size
Bio.Restriction.Restriction.BstYI.size
Bio.Restriction.Restriction.BstZ17I.size
Bio.Restriction.Restriction.BstZI.size
Bio.Restriction.Restriction.Bsu15I.size
Bio.Restriction.Restriction.Bsu36I.size
Bio.Restriction.Restriction.BsuI.size
Bio.Restriction.Restriction.BsuRI.size
Bio.Restriction.Restriction.BtgI.size
Bio.Restriction.Restriction.BtgZI.size
Bio.Restriction.Restriction.BthCI.size
Bio.Restriction.Restriction.BtrI.size
Bio.Restriction.Restriction.BtsCI.size
Bio.Restriction.Restriction.BtsI.size
Bio.Restriction.Restriction.BtsIMutI.size
Bio.Restriction.Restriction.BtuMI.size
Bio.Restriction.Restriction.BveI.size
Bio.Restriction.Restriction.Cac8I.size
Bio.Restriction.Restriction.CaiI.size
Bio.Restriction.Restriction.CauII.size
Bio.Restriction.Restriction.CchII.size
Bio.Restriction.Restriction.CchIII.size
Bio.Restriction.Restriction.CciI.size
Bio.Restriction.Restriction.CciNI.size
Bio.Restriction.Restriction.Cdi630V.size
Bio.Restriction.Restriction.CdiI.size
Bio.Restriction.Restriction.CdpI.size
Bio.Restriction.Restriction.CfoI.size
Bio.Restriction.Restriction.Cfr10I.size
Bio.Restriction.Restriction.Cfr13I.size
Bio.Restriction.Restriction.Cfr42I.size
Bio.Restriction.Restriction.Cfr9I.size
Bio.Restriction.Restriction.CfrI.size
Bio.Restriction.Restriction.Cgl13032I.size
Bio.Restriction.Restriction.Cgl13032II.size
Bio.Restriction.Restriction.ChaI.size
Bio.Restriction.Restriction.CjeFIII.size
Bio.Restriction.Restriction.CjeFV.size
Bio.Restriction.Restriction.CjeI.size
Bio.Restriction.Restriction.CjeNII.size
Bio.Restriction.Restriction.CjeNIII.size
Bio.Restriction.Restriction.CjeP659IV.size
Bio.Restriction.Restriction.CjePI.size
Bio.Restriction.Restriction.CjuI.size
Bio.Restriction.Restriction.CjuII.size
Bio.Restriction.Restriction.ClaI.size
Bio.Restriction.Restriction.CpoI.size
Bio.Restriction.Restriction.CseI.size
Bio.Restriction.Restriction.CsiI.size
Bio.Restriction.Restriction.Csp6I.size
Bio.Restriction.Restriction.CspAI.size
Bio.Restriction.Restriction.CspCI.size
Bio.Restriction.Restriction.CspI.size
Bio.Restriction.Restriction.CstMI.size
Bio.Restriction.Restriction.CviAII.size
Bio.Restriction.Restriction.CviJI.size
Bio.Restriction.Restriction.CviKI_1.size
Bio.Restriction.Restriction.CviQI.size
Bio.Restriction.Restriction.CviRI.size
Bio.Restriction.Restriction.DdeI.size
Bio.Restriction.Restriction.DinI.size
Bio.Restriction.Restriction.DpnI.size
Bio.Restriction.Restriction.DpnII.size
Bio.Restriction.Restriction.DraI.size
Bio.Restriction.Restriction.DraII.size
Bio.Restriction.Restriction.DraIII.size
Bio.Restriction.Restriction.DraRI.size
Bio.Restriction.Restriction.DrdI.size
Bio.Restriction.Restriction.DrdII.size
Bio.Restriction.Restriction.DriI.size
Bio.Restriction.Restriction.DsaI.size
Bio.Restriction.Restriction.DseDI.size
Bio.Restriction.Restriction.EaeI.size
Bio.Restriction.Restriction.EagI.size
Bio.Restriction.Restriction.Eam1104I.size
Bio.Restriction.Restriction.Eam1105I.size
Bio.Restriction.Restriction.EarI.size
Bio.Restriction.Restriction.EciI.size
Bio.Restriction.Restriction.Ecl136II.size
Bio.Restriction.Restriction.EclXI.size
Bio.Restriction.Restriction.Eco105I.size
Bio.Restriction.Restriction.Eco130I.size
Bio.Restriction.Restriction.Eco147I.size
Bio.Restriction.Restriction.Eco24I.size
Bio.Restriction.Restriction.Eco31I.size
Bio.Restriction.Restriction.Eco32I.size
Bio.Restriction.Restriction.Eco47I.size
Bio.Restriction.Restriction.Eco47III.size
Bio.Restriction.Restriction.Eco52I.size
Bio.Restriction.Restriction.Eco53kI.size
Bio.Restriction.Restriction.Eco57I.size
Bio.Restriction.Restriction.Eco57MI.size
Bio.Restriction.Restriction.Eco72I.size
Bio.Restriction.Restriction.Eco81I.size
Bio.Restriction.Restriction.Eco88I.size
Bio.Restriction.Restriction.Eco91I.size
Bio.Restriction.Restriction.EcoHI.size
Bio.Restriction.Restriction.EcoICRI.size
Bio.Restriction.Restriction.EcoNI.size
Bio.Restriction.Restriction.EcoO109I.size
Bio.Restriction.Restriction.EcoO65I.size
Bio.Restriction.Restriction.EcoRI.size
Bio.Restriction.Restriction.EcoRII.size
Bio.Restriction.Restriction.EcoRV.size
Bio.Restriction.Restriction.EcoT14I.size
Bio.Restriction.Restriction.EcoT22I.size
Bio.Restriction.Restriction.EcoT38I.size
Bio.Restriction.Restriction.EgeI.size
Bio.Restriction.Restriction.EheI.size
Bio.Restriction.Restriction.ErhI.size
Bio.Restriction.Restriction.EsaBC3I.size
Bio.Restriction.Restriction.EsaSSI.size
Bio.Restriction.Restriction.Esp3I.size
Bio.Restriction.Restriction.EspI.size
Bio.Restriction.Restriction.FaeI.size
Bio.Restriction.Restriction.FaiI.size
Bio.Restriction.Restriction.FalI.size
Bio.Restriction.Restriction.FaqI.size
Bio.Restriction.Restriction.FatI.size
Bio.Restriction.Restriction.FauI.size
Bio.Restriction.Restriction.FauNDI.size
Bio.Restriction.Restriction.FbaI.size
Bio.Restriction.Restriction.FblI.size
Bio.Restriction.Restriction.FinI.size
Bio.Restriction.Restriction.FmuI.size
Bio.Restriction.Restriction.Fnu4HI.size
Bio.Restriction.Restriction.FnuDII.size
Bio.Restriction.Restriction.FokI.size
Bio.Restriction.Restriction.FriOI.size
Bio.Restriction.Restriction.FseI.size
Bio.Restriction.Restriction.Fsp4HI.size
Bio.Restriction.Restriction.FspAI.size
Bio.Restriction.Restriction.FspBI.size
Bio.Restriction.Restriction.FspEI.size
Bio.Restriction.Restriction.FspI.size
Bio.Restriction.Restriction.GauT27I.size
Bio.Restriction.Restriction.GdiII.size
Bio.Restriction.Restriction.GlaI.size
Bio.Restriction.Restriction.GluI.size
Bio.Restriction.Restriction.GsaI.size
Bio.Restriction.Restriction.GsuI.size
Bio.Restriction.Restriction.HaeI.size
Bio.Restriction.Restriction.HaeII.size
Bio.Restriction.Restriction.HaeIII.size
Bio.Restriction.Restriction.HapII.size
Bio.Restriction.Restriction.HauII.size
Bio.Restriction.Restriction.HgaI.size
Bio.Restriction.Restriction.HgiAI.size
Bio.Restriction.Restriction.HgiCI.size
Bio.Restriction.Restriction.HgiEII.size
Bio.Restriction.Restriction.HgiJII.size
Bio.Restriction.Restriction.HhaI.size
Bio.Restriction.Restriction.Hin1I.size
Bio.Restriction.Restriction.Hin1II.size
Bio.Restriction.Restriction.Hin4I.size
Bio.Restriction.Restriction.Hin4II.size
Bio.Restriction.Restriction.Hin6I.size
Bio.Restriction.Restriction.HinP1I.size
Bio.Restriction.Restriction.HincII.size
Bio.Restriction.Restriction.HindII.size
Bio.Restriction.Restriction.HindIII.size
Bio.Restriction.Restriction.HinfI.size
Bio.Restriction.Restriction.HpaI.size
Bio.Restriction.Restriction.HpaII.size
Bio.Restriction.Restriction.HphI.size
Bio.Restriction.Restriction.Hpy166II.size
Bio.Restriction.Restriction.Hpy178III.size
Bio.Restriction.Restriction.Hpy188I.size
Bio.Restriction.Restriction.Hpy188III.size
Bio.Restriction.Restriction.Hpy8I.size
Bio.Restriction.Restriction.Hpy99I.size
Bio.Restriction.Restriction.Hpy99XIII.size
Bio.Restriction.Restriction.Hpy99XIV.size
Bio.Restriction.Restriction.HpyAV.size
Bio.Restriction.Restriction.HpyCH4III.size
Bio.Restriction.Restriction.HpyCH4IV.size
Bio.Restriction.Restriction.HpyCH4V.size
Bio.Restriction.Restriction.HpyF10VI.size
Bio.Restriction.Restriction.HpyF3I.size
Bio.Restriction.Restriction.HpySE526I.size
Bio.Restriction.Restriction.Hsp92I.size
Bio.Restriction.Restriction.Hsp92II.size
Bio.Restriction.Restriction.HspAI.size
Bio.Restriction.Restriction.Jma19592I.size
Bio.Restriction.Restriction.KasI.size
Bio.Restriction.Restriction.KflI.size
Bio.Restriction.Restriction.Kpn2I.size
Bio.Restriction.Restriction.KpnI.size
Bio.Restriction.Restriction.KroI.size
Bio.Restriction.Restriction.Ksp22I.size
Bio.Restriction.Restriction.Ksp632I.size
Bio.Restriction.Restriction.KspAI.size
Bio.Restriction.Restriction.KspI.size
Bio.Restriction.Restriction.Kzo9I.size
Bio.Restriction.Restriction.LguI.size
Bio.Restriction.Restriction.LpnI.size
Bio.Restriction.Restriction.LpnPI.size
Bio.Restriction.Restriction.Lsp1109I.size
Bio.Restriction.Restriction.LweI.size
Bio.Restriction.Restriction.MabI.size
Bio.Restriction.Restriction.MaeI.size
Bio.Restriction.Restriction.MaeII.size
Bio.Restriction.Restriction.MaeIII.size
Bio.Restriction.Restriction.MalI.size
Bio.Restriction.Restriction.MaqI.size
Bio.Restriction.Restriction.MauBI.size
Bio.Restriction.Restriction.MbiI.size
Bio.Restriction.Restriction.MboI.size
Bio.Restriction.Restriction.MboII.size
Bio.Restriction.Restriction.McaTI.size
Bio.Restriction.Restriction.McrI.size
Bio.Restriction.Restriction.MfeI.size
Bio.Restriction.Restriction.MflI.size
Bio.Restriction.Restriction.MhlI.size
Bio.Restriction.Restriction.MjaIV.size
Bio.Restriction.Restriction.MkaDII.size
Bio.Restriction.Restriction.MlsI.size
Bio.Restriction.Restriction.MluCI.size
Bio.Restriction.Restriction.MluI.size
Bio.Restriction.Restriction.MluNI.size
Bio.Restriction.Restriction.Mly113I.size
Bio.Restriction.Restriction.MlyI.size
Bio.Restriction.Restriction.MmeI.size
Bio.Restriction.Restriction.MnlI.size
Bio.Restriction.Restriction.Mph1103I.size
Bio.Restriction.Restriction.MreI.size
Bio.Restriction.Restriction.MroI.size
Bio.Restriction.Restriction.MroNI.size
Bio.Restriction.Restriction.MroXI.size
Bio.Restriction.Restriction.MscI.size
Bio.Restriction.Restriction.MseI.size
Bio.Restriction.Restriction.MslI.size
Bio.Restriction.Restriction.Msp20I.size
Bio.Restriction.Restriction.MspA1I.size
Bio.Restriction.Restriction.MspCI.size
Bio.Restriction.Restriction.MspI.size
Bio.Restriction.Restriction.MspJI.size
Bio.Restriction.Restriction.MspR9I.size
Bio.Restriction.Restriction.MssI.size
Bio.Restriction.Restriction.MstI.size
Bio.Restriction.Restriction.MunI.size
Bio.Restriction.Restriction.Mva1269I.size
Bio.Restriction.Restriction.MvaI.size
Bio.Restriction.Restriction.MvnI.size
Bio.Restriction.Restriction.MvrI.size
Bio.Restriction.Restriction.MwoI.size
Bio.Restriction.Restriction.NaeI.size
Bio.Restriction.Restriction.NarI.size
Bio.Restriction.Restriction.NciI.size
Bio.Restriction.Restriction.NcoI.size
Bio.Restriction.Restriction.NdeI.size
Bio.Restriction.Restriction.NdeII.size
Bio.Restriction.Restriction.NgoAVIII.size
Bio.Restriction.Restriction.NgoMIV.size
Bio.Restriction.Restriction.NhaXI.size
Bio.Restriction.Restriction.NheI.size
Bio.Restriction.Restriction.NlaCI.size
Bio.Restriction.Restriction.NlaIII.size
Bio.Restriction.Restriction.NlaIV.size
Bio.Restriction.Restriction.Nli3877I.size
Bio.Restriction.Restriction.NmeAIII.size
Bio.Restriction.Restriction.NmeDI.size
Bio.Restriction.Restriction.NmuCI.size
Bio.Restriction.Restriction.NotI.size
Bio.Restriction.Restriction.NruI.size
Bio.Restriction.Restriction.NsbI.size
Bio.Restriction.Restriction.NsiI.size
Bio.Restriction.Restriction.NspBII.size
Bio.Restriction.Restriction.NspI.size
Bio.Restriction.Restriction.NspV.size
Bio.Restriction.Restriction.OliI.size
Bio.Restriction.Restriction.PabI.size
Bio.Restriction.Restriction.PacI.size
Bio.Restriction.Restriction.PaeI.size
Bio.Restriction.Restriction.PaeR7I.size
Bio.Restriction.Restriction.PagI.size
Bio.Restriction.Restriction.PalAI.size
Bio.Restriction.Restriction.PasI.size
Bio.Restriction.Restriction.PauI.size
Bio.Restriction.Restriction.PceI.size
Bio.Restriction.Restriction.PciI.size
Bio.Restriction.Restriction.PciSI.size
Bio.Restriction.Restriction.PcsI.size
Bio.Restriction.Restriction.PctI.size
Bio.Restriction.Restriction.PdiI.size
Bio.Restriction.Restriction.PdmI.size
Bio.Restriction.Restriction.PenI.size
Bio.Restriction.Restriction.PfeI.size
Bio.Restriction.Restriction.Pfl1108I.size
Bio.Restriction.Restriction.Pfl23II.size
Bio.Restriction.Restriction.PflFI.size
Bio.Restriction.Restriction.PflMI.size
Bio.Restriction.Restriction.PfoI.size
Bio.Restriction.Restriction.PinAI.size
Bio.Restriction.Restriction.PlaDI.size
Bio.Restriction.Restriction.Ple19I.size
Bio.Restriction.Restriction.PleI.size
Bio.Restriction.Restriction.PluTI.size
Bio.Restriction.Restriction.PmaCI.size
Bio.Restriction.Restriction.PmeI.size
Bio.Restriction.Restriction.PmlI.size
Bio.Restriction.Restriction.PpiI.size
Bio.Restriction.Restriction.PpsI.size
Bio.Restriction.Restriction.Ppu10I.size
Bio.Restriction.Restriction.Ppu21I.size
Bio.Restriction.Restriction.PpuMI.size
Bio.Restriction.Restriction.PscI.size
Bio.Restriction.Restriction.PshAI.size
Bio.Restriction.Restriction.PshBI.size
Bio.Restriction.Restriction.PsiI.size
Bio.Restriction.Restriction.Psp03I.size
Bio.Restriction.Restriction.Psp124BI.size
Bio.Restriction.Restriction.Psp1406I.size
Bio.Restriction.Restriction.Psp5II.size
Bio.Restriction.Restriction.Psp6I.size
Bio.Restriction.Restriction.PspCI.size
Bio.Restriction.Restriction.PspEI.size
Bio.Restriction.Restriction.PspGI.size
Bio.Restriction.Restriction.PspLI.size
Bio.Restriction.Restriction.PspN4I.size
Bio.Restriction.Restriction.PspOMI.size
Bio.Restriction.Restriction.PspOMII.size
Bio.Restriction.Restriction.PspPI.size
Bio.Restriction.Restriction.PspPPI.size
Bio.Restriction.Restriction.PspPRI.size
Bio.Restriction.Restriction.PspXI.size
Bio.Restriction.Restriction.PsrI.size
Bio.Restriction.Restriction.PssI.size
Bio.Restriction.Restriction.PstI.size
Bio.Restriction.Restriction.PstNI.size
Bio.Restriction.Restriction.PsuI.size
Bio.Restriction.Restriction.PsyI.size
Bio.Restriction.Restriction.PteI.size
Bio.Restriction.Restriction.PvuI.size
Bio.Restriction.Restriction.PvuII.size
Bio.Restriction.Restriction.R2_BceSIV.size
Bio.Restriction.Restriction.RceI.size
Bio.Restriction.Restriction.RdeGBI.size
Bio.Restriction.Restriction.RdeGBII.size
Bio.Restriction.Restriction.RdeGBIII.size
Bio.Restriction.Restriction.RflFIII.size
Bio.Restriction.Restriction.RgaI.size
Bio.Restriction.Restriction.RigI.size
Bio.Restriction.Restriction.RlaI.size
Bio.Restriction.Restriction.RleAI.size
Bio.Restriction.Restriction.RpaB5I.size
Bio.Restriction.Restriction.RpaBI.size
Bio.Restriction.Restriction.RpaI.size
Bio.Restriction.Restriction.RpaTI.size
Bio.Restriction.Restriction.RruI.size
Bio.Restriction.Restriction.RsaI.size
Bio.Restriction.Restriction.RsaNI.size
Bio.Restriction.Restriction.RseI.size
Bio.Restriction.Restriction.Rsr2I.size
Bio.Restriction.Restriction.RsrII.size
Bio.Restriction.Restriction.SacI.size
Bio.Restriction.Restriction.SacII.size
Bio.Restriction.Restriction.SalI.size
Bio.Restriction.Restriction.SanDI.size
Bio.Restriction.Restriction.SapI.size
Bio.Restriction.Restriction.SaqAI.size
Bio.Restriction.Restriction.SatI.size
Bio.Restriction.Restriction.Sau3AI.size
Bio.Restriction.Restriction.Sau96I.size
Bio.Restriction.Restriction.SauI.size
Bio.Restriction.Restriction.SbfI.size
Bio.Restriction.Restriction.ScaI.size
Bio.Restriction.Restriction.SchI.size
Bio.Restriction.Restriction.SciI.size
Bio.Restriction.Restriction.ScrFI.size
Bio.Restriction.Restriction.SdaI.size
Bio.Restriction.Restriction.SdeAI.size
Bio.Restriction.Restriction.SdeOSI.size
Bio.Restriction.Restriction.SduI.size
Bio.Restriction.Restriction.SecI.size
Bio.Restriction.Restriction.SelI.size
Bio.Restriction.Restriction.SetI.size
Bio.Restriction.Restriction.SexAI.size
Bio.Restriction.Restriction.SfaAI.size
Bio.Restriction.Restriction.SfaNI.size
Bio.Restriction.Restriction.SfcI.size
Bio.Restriction.Restriction.SfeI.size
Bio.Restriction.Restriction.SfiI.size
Bio.Restriction.Restriction.SfoI.size
Bio.Restriction.Restriction.Sfr274I.size
Bio.Restriction.Restriction.Sfr303I.size
Bio.Restriction.Restriction.SfuI.size
Bio.Restriction.Restriction.SgeI.size
Bio.Restriction.Restriction.SgfI.size
Bio.Restriction.Restriction.SgrAI.size
Bio.Restriction.Restriction.SgrBI.size
Bio.Restriction.Restriction.SgrDI.size
Bio.Restriction.Restriction.SgrTI.size
Bio.Restriction.Restriction.SgsI.size
Bio.Restriction.Restriction.SimI.size
Bio.Restriction.Restriction.SlaI.size
Bio.Restriction.Restriction.SmaI.size
Bio.Restriction.Restriction.SmiI.size
Bio.Restriction.Restriction.SmiMI.size
Bio.Restriction.Restriction.SmlI.size
Bio.Restriction.Restriction.SmoI.size
Bio.Restriction.Restriction.SnaBI.size
Bio.Restriction.Restriction.SnaI.size
Bio.Restriction.Restriction.Sno506I.size
Bio.Restriction.Restriction.SpeI.size
Bio.Restriction.Restriction.SphI.size
Bio.Restriction.Restriction.SplI.size
Bio.Restriction.Restriction.SpoDI.size
Bio.Restriction.Restriction.SrfI.size
Bio.Restriction.Restriction.Sse232I.size
Bio.Restriction.Restriction.Sse8387I.size
Bio.Restriction.Restriction.Sse8647I.size
Bio.Restriction.Restriction.Sse9I.size
Bio.Restriction.Restriction.SseBI.size
Bio.Restriction.Restriction.SsiI.size
Bio.Restriction.Restriction.SspD5I.size
Bio.Restriction.Restriction.SspDI.size
Bio.Restriction.Restriction.SspI.size
Bio.Restriction.Restriction.SstE37I.size
Bio.Restriction.Restriction.SstI.size
Bio.Restriction.Restriction.Sth132I.size
Bio.Restriction.Restriction.Sth302II.size
Bio.Restriction.Restriction.StrI.size
Bio.Restriction.Restriction.StsI.size
Bio.Restriction.Restriction.StuI.size
Bio.Restriction.Restriction.StyD4I.size
Bio.Restriction.Restriction.StyI.size
Bio.Restriction.Restriction.SwaI.size
Bio.Restriction.Restriction.TaaI.size
Bio.Restriction.Restriction.TaiI.size
Bio.Restriction.Restriction.TaqI.size
Bio.Restriction.Restriction.TaqII.size
Bio.Restriction.Restriction.TasI.size
Bio.Restriction.Restriction.TatI.size
Bio.Restriction.Restriction.TauI.size
Bio.Restriction.Restriction.TfiI.size
Bio.Restriction.Restriction.Tru1I.size
Bio.Restriction.Restriction.Tru9I.size
Bio.Restriction.Restriction.TscAI.size
Bio.Restriction.Restriction.TseFI.size
Bio.Restriction.Restriction.TseI.size
Bio.Restriction.Restriction.TsoI.size
Bio.Restriction.Restriction.Tsp45I.size
Bio.Restriction.Restriction.Tsp4CI.size
Bio.Restriction.Restriction.TspDTI.size
Bio.Restriction.Restriction.TspEI.size
Bio.Restriction.Restriction.TspGWI.size
Bio.Restriction.Restriction.TspMI.size
Bio.Restriction.Restriction.TspRI.size
Bio.Restriction.Restriction.TssI.size
Bio.Restriction.Restriction.TstI.size
Bio.Restriction.Restriction.TsuI.size
Bio.Restriction.Restriction.Tth111I.size
Bio.Restriction.Restriction.Tth111II.size
Bio.Restriction.Restriction.UbaF11I.size
Bio.Restriction.Restriction.UbaF12I.size
Bio.Restriction.Restriction.UbaF13I.size
Bio.Restriction.Restriction.UbaF14I.size
Bio.Restriction.Restriction.UbaF9I.size
Bio.Restriction.Restriction.UbaPI.size
Bio.Restriction.Restriction.UcoMSI.size
Bio.Restriction.Restriction.UnbI.size
Bio.Restriction.Restriction.Van91I.size
Bio.Restriction.Restriction.Vha464I.size
Bio.Restriction.Restriction.VneI.size
Bio.Restriction.Restriction.VpaK11AI.size
Bio.Restriction.Restriction.VpaK11BI.size
Bio.Restriction.Restriction.VspI.size
Bio.Restriction.Restriction.WviI.size
Bio.Restriction.Restriction.XagI.size
Bio.Restriction.Restriction.XapI.size
Bio.Restriction.Restriction.XbaI.size
Bio.Restriction.Restriction.XceI.size
Bio.Restriction.Restriction.XcmI.size
Bio.Restriction.Restriction.XhoI.size
Bio.Restriction.Restriction.XhoII.size
Bio.Restriction.Restriction.XmaI.size
Bio.Restriction.Restriction.XmaIII.size
Bio.Restriction.Restriction.XmaJI.size
Bio.Restriction.Restriction.XmiI.size
Bio.Restriction.Restriction.XmnI.size
Bio.Restriction.Restriction.XspI.size
Bio.Restriction.Restriction.YkrI.size
Bio.Restriction.Restriction.ZraI.size
Bio.Restriction.Restriction.ZrmI.size
Bio.Restriction.Restriction.Zsp2I.size
Bio.codonalign.codonalphabet.CodonAlphabet.size" class="py-name" href="#" onclick="return doclink('link-451', 'size', 'link-270');">size</a></tt><tt class="py-op">(</tt><tt class="py-name">name_and_length</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1159"></a><tt class="py-lineno">1159</tt>  <tt class="py-line"> </tt>
<a name="L1160"></a><tt class="py-lineno">1160</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">44</tt><tt class="py-op">:</tt><tt class="py-number">54</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-452" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-452', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">""</tt> <tt class="py-keyword">and</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">40</tt><tt class="py-op">:</tt><tt class="py-number">44</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">' aa '</tt><tt class="py-op">:</tt> </tt>
<a name="L1161"></a><tt class="py-lineno">1161</tt>  <tt class="py-line">                <tt class="py-comment"># Amino acids -&gt; protein (even if there is no residue type given)</tt> </tt>
<a name="L1162"></a><tt class="py-lineno">1162</tt>  <tt class="py-line">                <tt class="py-comment"># We want to use a protein alphabet in this case, rather than a</tt> </tt>
<a name="L1163"></a><tt class="py-lineno">1163</tt>  <tt class="py-line">                <tt class="py-comment"># generic one. Not sure if this is the best way to achieve this,</tt> </tt>
<a name="L1164"></a><tt class="py-lineno">1164</tt>  <tt class="py-line">                <tt class="py-comment"># but it works because the scanner checks for this:</tt> </tt>
<a name="L1165"></a><tt class="py-lineno">1165</tt>  <tt class="py-line">                <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-453" class="py-name"><a title="Bio.GenBank._FeatureConsumer.residue_type
Bio.GenBank._RecordConsumer.residue_type" class="py-name" href="#" onclick="return doclink('link-453', 'residue_type', 'link-236');">residue_type</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">"PROTEIN "</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">54</tt><tt class="py-op">:</tt><tt class="py-number">63</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-454" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-454', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1166"></a><tt class="py-lineno">1166</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1167"></a><tt class="py-lineno">1167</tt>  <tt class="py-line">                <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-455" class="py-name"><a title="Bio.GenBank._FeatureConsumer.residue_type
Bio.GenBank._RecordConsumer.residue_type" class="py-name" href="#" onclick="return doclink('link-455', 'residue_type', 'link-236');">residue_type</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">44</tt><tt class="py-op">:</tt><tt class="py-number">63</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-456" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-456', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1168"></a><tt class="py-lineno">1168</tt>  <tt class="py-line"> </tt>
<a name="L1169"></a><tt class="py-lineno">1169</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-457" class="py-name"><a title="Bio.GenBank._FeatureConsumer.data_file_division
Bio.GenBank._RecordConsumer.data_file_division" class="py-name" href="#" onclick="return doclink('link-457', 'data_file_division', 'link-237');">data_file_division</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">64</tt><tt class="py-op">:</tt><tt class="py-number">67</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1170"></a><tt class="py-lineno">1170</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">68</tt><tt class="py-op">:</tt><tt class="py-number">79</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-458" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-458', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1171"></a><tt class="py-lineno">1171</tt>  <tt class="py-line">                <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-459" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-459', 'date', 'link-439');">date</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">68</tt><tt class="py-op">:</tt><tt class="py-number">79</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1172"></a><tt class="py-lineno">1172</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt id="link-460" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-460', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-461" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-461', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-462" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-462', 'count', 'link-90');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">" "</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1173"></a><tt class="py-lineno">1173</tt>  <tt class="py-line">            <tt class="py-comment"># Truncated LOCUS line, as produced by some EMBOSS tools - see bug 1762</tt> </tt>
<a name="L1174"></a><tt class="py-lineno">1174</tt>  <tt class="py-line">            <tt class="py-comment">#</tt> </tt>
<a name="L1175"></a><tt class="py-lineno">1175</tt>  <tt class="py-line">            <tt class="py-comment"># e.g.</tt> </tt>
<a name="L1176"></a><tt class="py-lineno">1176</tt>  <tt class="py-line">            <tt class="py-comment">#</tt> </tt>
<a name="L1177"></a><tt class="py-lineno">1177</tt>  <tt class="py-line">            <tt class="py-comment">#    "LOCUS       U00096"</tt> </tt>
<a name="L1178"></a><tt class="py-lineno">1178</tt>  <tt class="py-line">            <tt class="py-comment">#</tt> </tt>
<a name="L1179"></a><tt class="py-lineno">1179</tt>  <tt class="py-line">            <tt class="py-comment"># rather than:</tt> </tt>
<a name="L1180"></a><tt class="py-lineno">1180</tt>  <tt class="py-line">            <tt class="py-comment">#</tt> </tt>
<a name="L1181"></a><tt class="py-lineno">1181</tt>  <tt class="py-line">            <tt class="py-comment">#    "LOCUS       U00096               4639675 bp    DNA     circular BCT"</tt> </tt>
<a name="L1182"></a><tt class="py-lineno">1182</tt>  <tt class="py-line">            <tt class="py-comment">#</tt> </tt>
<a name="L1183"></a><tt class="py-lineno">1183</tt>  <tt class="py-line">            <tt class="py-comment">#    Positions  Contents</tt> </tt>
<a name="L1184"></a><tt class="py-lineno">1184</tt>  <tt class="py-line">            <tt class="py-comment">#    ---------  --------</tt> </tt>
<a name="L1185"></a><tt class="py-lineno">1185</tt>  <tt class="py-line">            <tt class="py-comment">#    00:06      LOCUS</tt> </tt>
<a name="L1186"></a><tt class="py-lineno">1186</tt>  <tt class="py-line">            <tt class="py-comment">#    06:12      spaces</tt> </tt>
<a name="L1187"></a><tt class="py-lineno">1187</tt>  <tt class="py-line">            <tt class="py-comment">#    12:??      Locus name</tt> </tt>
<a name="L1188"></a><tt class="py-lineno">1188</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt id="link-463" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-463', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-464" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-464', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-string">""</tt><tt class="py-op">:</tt> </tt>
<a name="L1189"></a><tt class="py-lineno">1189</tt>  <tt class="py-line">                <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-465" class="py-name"><a title="Bio.GenBank._FeatureConsumer.locus
Bio.GenBank._RecordConsumer.locus" class="py-name" href="#" onclick="return doclink('link-465', 'locus', 'link-235');">locus</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt id="link-466" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-466', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-467" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-467', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1190"></a><tt class="py-lineno">1190</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1191"></a><tt class="py-lineno">1191</tt>  <tt class="py-line">                <tt class="py-comment"># Must just have just "LOCUS       ", is this even legitimate?</tt> </tt>
<a name="L1192"></a><tt class="py-lineno">1192</tt>  <tt class="py-line">                <tt class="py-comment"># We should be able to continue parsing... we need real world testcases!</tt> </tt>
<a name="L1193"></a><tt class="py-lineno">1193</tt>  <tt class="py-line">                <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Minimal LOCUS line found - is this "</tt> </tt>
<a name="L1194"></a><tt class="py-lineno">1194</tt>  <tt class="py-line">                              <tt class="py-string">"correct?\n:%r"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">,</tt> <tt id="link-468" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-468', 'BiopythonParserWarning', 'link-10');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1195"></a><tt class="py-lineno">1195</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-469" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-469', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">8</tt> <tt class="py-keyword">and</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-470" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-470', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">"aa"</tt><tt class="py-op">,</tt> <tt class="py-string">"bp"</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L1196"></a><tt class="py-lineno">1196</tt>  <tt class="py-line">             <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-471" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-471', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">5</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'linear'</tt><tt class="py-op">,</tt> <tt class="py-string">'circular'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1197"></a><tt class="py-lineno">1197</tt>  <tt class="py-line">            <tt class="py-comment"># Cope with invalidly spaced GenBank LOCUS lines like</tt> </tt>
<a name="L1198"></a><tt class="py-lineno">1198</tt>  <tt class="py-line">            <tt class="py-comment"># LOCUS       AB070938          6497 bp    DNA     linear   BCT 11-OCT-2001</tt> </tt>
<a name="L1199"></a><tt class="py-lineno">1199</tt>  <tt class="py-line">            <tt class="py-name">splitline</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-472" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-472', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1200"></a><tt class="py-lineno">1200</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-473" class="py-name"><a title="Bio.GenBank._FeatureConsumer.locus
Bio.GenBank._RecordConsumer.locus" class="py-name" href="#" onclick="return doclink('link-473', 'locus', 'link-235');">locus</a></tt><tt class="py-op">(</tt><tt class="py-name">splitline</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1201"></a><tt class="py-lineno">1201</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-474" class="py-name"><a title="Bio.Alphabet.Alphabet.size
Bio.Alphabet.Reduced.HPModel.size
Bio.Alphabet.Reduced.Murphy10.size
Bio.Alphabet.Reduced.Murphy15.size
Bio.Alphabet.Reduced.Murphy4.size
Bio.Alphabet.Reduced.Murphy8.size
Bio.Alphabet.Reduced.PC5.size
Bio.Alphabet.SingleLetterAlphabet.size
Bio.Alphabet.ThreeLetterProtein.size
Bio.GenBank._FeatureConsumer.size
Bio.GenBank._RecordConsumer.size
Bio.Restriction.Restriction.AanI.size
Bio.Restriction.Restriction.AarI.size
Bio.Restriction.Restriction.AasI.size
Bio.Restriction.Restriction.AatII.size
Bio.Restriction.Restriction.AbaSI.size
Bio.Restriction.Restriction.AbsI.size
Bio.Restriction.Restriction.Acc16I.size
Bio.Restriction.Restriction.Acc36I.size
Bio.Restriction.Restriction.Acc65I.size
Bio.Restriction.Restriction.AccB1I.size
Bio.Restriction.Restriction.AccB7I.size
Bio.Restriction.Restriction.AccBSI.size
Bio.Restriction.Restriction.AccI.size
Bio.Restriction.Restriction.AccII.size
Bio.Restriction.Restriction.AccIII.size
Bio.Restriction.Restriction.AceIII.size
Bio.Restriction.Restriction.AciI.size
Bio.Restriction.Restriction.AclI.size
Bio.Restriction.Restriction.AclWI.size
Bio.Restriction.Restriction.AcoI.size
Bio.Restriction.Restriction.AcsI.size
Bio.Restriction.Restriction.AcuI.size
Bio.Restriction.Restriction.AcvI.size
Bio.Restriction.Restriction.AcyI.size
Bio.Restriction.Restriction.AdeI.size
Bio.Restriction.Restriction.AfaI.size
Bio.Restriction.Restriction.AfeI.size
Bio.Restriction.Restriction.AfiI.size
Bio.Restriction.Restriction.AflII.size
Bio.Restriction.Restriction.AflIII.size
Bio.Restriction.Restriction.AgeI.size
Bio.Restriction.Restriction.AgsI.size
Bio.Restriction.Restriction.AhaIII.size
Bio.Restriction.Restriction.AhdI.size
Bio.Restriction.Restriction.AhlI.size
Bio.Restriction.Restriction.AjiI.size
Bio.Restriction.Restriction.AjnI.size
Bio.Restriction.Restriction.AjuI.size
Bio.Restriction.Restriction.AleI.size
Bio.Restriction.Restriction.AlfI.size
Bio.Restriction.Restriction.AloI.size
Bio.Restriction.Restriction.AluBI.size
Bio.Restriction.Restriction.AluI.size
Bio.Restriction.Restriction.Alw21I.size
Bio.Restriction.Restriction.Alw26I.size
Bio.Restriction.Restriction.Alw44I.size
Bio.Restriction.Restriction.AlwFI.size
Bio.Restriction.Restriction.AlwI.size
Bio.Restriction.Restriction.AlwNI.size
Bio.Restriction.Restriction.Ama87I.size
Bio.Restriction.Restriction.Aor13HI.size
Bio.Restriction.Restriction.Aor51HI.size
Bio.Restriction.Restriction.AoxI.size
Bio.Restriction.Restriction.ApaBI.size
Bio.Restriction.Restriction.ApaI.size
Bio.Restriction.Restriction.ApaLI.size
Bio.Restriction.Restriction.ApeKI.size
Bio.Restriction.Restriction.ApoI.size
Bio.Restriction.Restriction.ApyPI.size
Bio.Restriction.Restriction.AquII.size
Bio.Restriction.Restriction.AquIII.size
Bio.Restriction.Restriction.AquIV.size
Bio.Restriction.Restriction.ArsI.size
Bio.Restriction.Restriction.AscI.size
Bio.Restriction.Restriction.AseI.size
Bio.Restriction.Restriction.Asi256I.size
Bio.Restriction.Restriction.AsiGI.size
Bio.Restriction.Restriction.AsiSI.size
Bio.Restriction.Restriction.Asp700I.size
Bio.Restriction.Restriction.Asp718I.size
Bio.Restriction.Restriction.AspA2I.size
Bio.Restriction.Restriction.AspBHI.size
Bio.Restriction.Restriction.AspLEI.size
Bio.Restriction.Restriction.AspS9I.size
Bio.Restriction.Restriction.AssI.size
Bio.Restriction.Restriction.AsuC2I.size
Bio.Restriction.Restriction.AsuHPI.size
Bio.Restriction.Restriction.AsuI.size
Bio.Restriction.Restriction.AsuII.size
Bio.Restriction.Restriction.AsuNHI.size
Bio.Restriction.Restriction.AvaI.size
Bio.Restriction.Restriction.AvaII.size
Bio.Restriction.Restriction.AvaIII.size
Bio.Restriction.Restriction.AvrII.size
Bio.Restriction.Restriction.AxyI.size
Bio.Restriction.Restriction.BaeGI.size
Bio.Restriction.Restriction.BaeI.size
Bio.Restriction.Restriction.BalI.size
Bio.Restriction.Restriction.BamHI.size
Bio.Restriction.Restriction.BanI.size
Bio.Restriction.Restriction.BanII.size
Bio.Restriction.Restriction.BarI.size
Bio.Restriction.Restriction.BasI.size
Bio.Restriction.Restriction.BauI.size
Bio.Restriction.Restriction.Bbr7I.size
Bio.Restriction.Restriction.BbrPI.size
Bio.Restriction.Restriction.BbsI.size
Bio.Restriction.Restriction.Bbv12I.size
Bio.Restriction.Restriction.BbvCI.size
Bio.Restriction.Restriction.BbvI.size
Bio.Restriction.Restriction.BbvII.size
Bio.Restriction.Restriction.BccI.size
Bio.Restriction.Restriction.Bce83I.size
Bio.Restriction.Restriction.BceAI.size
Bio.Restriction.Restriction.BcefI.size
Bio.Restriction.Restriction.BcgI.size
Bio.Restriction.Restriction.BciT130I.size
Bio.Restriction.Restriction.BciVI.size
Bio.Restriction.Restriction.BclI.size
Bio.Restriction.Restriction.BcnI.size
Bio.Restriction.Restriction.BcoDI.size
Bio.Restriction.Restriction.BcuI.size
Bio.Restriction.Restriction.BdaI.size
Bio.Restriction.Restriction.BetI.size
Bio.Restriction.Restriction.BfaI.size
Bio.Restriction.Restriction.BfiI.size
Bio.Restriction.Restriction.BfmI.size
Bio.Restriction.Restriction.BfoI.size
Bio.Restriction.Restriction.BfrI.size
Bio.Restriction.Restriction.BfuAI.size
Bio.Restriction.Restriction.BfuCI.size
Bio.Restriction.Restriction.BfuI.size
Bio.Restriction.Restriction.BglI.size
Bio.Restriction.Restriction.BglII.size
Bio.Restriction.Restriction.BinI.size
Bio.Restriction.Restriction.BisI.size
Bio.Restriction.Restriction.BlnI.size
Bio.Restriction.Restriction.BlpI.size
Bio.Restriction.Restriction.BlsI.size
Bio.Restriction.Restriction.BmcAI.size
Bio.Restriction.Restriction.Bme1390I.size
Bio.Restriction.Restriction.Bme18I.size
Bio.Restriction.Restriction.BmeDI.size
Bio.Restriction.Restriction.BmeRI.size
Bio.Restriction.Restriction.BmeT110I.size
Bio.Restriction.Restriction.BmgBI.size
Bio.Restriction.Restriction.BmgI.size
Bio.Restriction.Restriction.BmgT120I.size
Bio.Restriction.Restriction.BmiI.size
Bio.Restriction.Restriction.BmrFI.size
Bio.Restriction.Restriction.BmrI.size
Bio.Restriction.Restriction.BmsI.size
Bio.Restriction.Restriction.BmtI.size
Bio.Restriction.Restriction.BmuI.size
Bio.Restriction.Restriction.BoxI.size
Bio.Restriction.Restriction.BpiI.size
Bio.Restriction.Restriction.BplI.size
Bio.Restriction.Restriction.BpmI.size
Bio.Restriction.Restriction.Bpu10I.size
Bio.Restriction.Restriction.Bpu1102I.size
Bio.Restriction.Restriction.Bpu14I.size
Bio.Restriction.Restriction.BpuEI.size
Bio.Restriction.Restriction.BpuMI.size
Bio.Restriction.Restriction.BpvUI.size
Bio.Restriction.Restriction.Bsa29I.size
Bio.Restriction.Restriction.BsaAI.size
Bio.Restriction.Restriction.BsaBI.size
Bio.Restriction.Restriction.BsaHI.size
Bio.Restriction.Restriction.BsaI.size
Bio.Restriction.Restriction.BsaJI.size
Bio.Restriction.Restriction.BsaWI.size
Bio.Restriction.Restriction.BsaXI.size
Bio.Restriction.Restriction.BsbI.size
Bio.Restriction.Restriction.Bsc4I.size
Bio.Restriction.Restriction.BscAI.size
Bio.Restriction.Restriction.BscGI.size
Bio.Restriction.Restriction.Bse118I.size
Bio.Restriction.Restriction.Bse1I.size
Bio.Restriction.Restriction.Bse21I.size
Bio.Restriction.Restriction.Bse3DI.size
Bio.Restriction.Restriction.Bse8I.size
Bio.Restriction.Restriction.BseAI.size
Bio.Restriction.Restriction.BseBI.size
Bio.Restriction.Restriction.BseCI.size
Bio.Restriction.Restriction.BseDI.size
Bio.Restriction.Restriction.BseGI.size
Bio.Restriction.Restriction.BseJI.size
Bio.Restriction.Restriction.BseLI.size
Bio.Restriction.Restriction.BseMI.size
Bio.Restriction.Restriction.BseMII.size
Bio.Restriction.Restriction.BseNI.size
Bio.Restriction.Restriction.BsePI.size
Bio.Restriction.Restriction.BseRI.size
Bio.Restriction.Restriction.BseSI.size
Bio.Restriction.Restriction.BseX3I.size
Bio.Restriction.Restriction.BseXI.size
Bio.Restriction.Restriction.BseYI.size
Bio.Restriction.Restriction.BsgI.size
Bio.Restriction.Restriction.Bsh1236I.size
Bio.Restriction.Restriction.Bsh1285I.size
Bio.Restriction.Restriction.BshFI.size
Bio.Restriction.Restriction.BshNI.size
Bio.Restriction.Restriction.BshTI.size
Bio.Restriction.Restriction.BshVI.size
Bio.Restriction.Restriction.BsiEI.size
Bio.Restriction.Restriction.BsiHKAI.size
Bio.Restriction.Restriction.BsiHKCI.size
Bio.Restriction.Restriction.BsiI.size
Bio.Restriction.Restriction.BsiSI.size
Bio.Restriction.Restriction.BsiWI.size
Bio.Restriction.Restriction.BsiYI.size
Bio.Restriction.Restriction.BslFI.size
Bio.Restriction.Restriction.BslI.size
Bio.Restriction.Restriction.BsmAI.size
Bio.Restriction.Restriction.BsmBI.size
Bio.Restriction.Restriction.BsmFI.size
Bio.Restriction.Restriction.BsmI.size
Bio.Restriction.Restriction.BsnI.size
Bio.Restriction.Restriction.Bso31I.size
Bio.Restriction.Restriction.BsoBI.size
Bio.Restriction.Restriction.Bsp119I.size
Bio.Restriction.Restriction.Bsp120I.size
Bio.Restriction.Restriction.Bsp1286I.size
Bio.Restriction.Restriction.Bsp13I.size
Bio.Restriction.Restriction.Bsp1407I.size
Bio.Restriction.Restriction.Bsp143I.size
Bio.Restriction.Restriction.Bsp1720I.size
Bio.Restriction.Restriction.Bsp19I.size
Bio.Restriction.Restriction.Bsp24I.size
Bio.Restriction.Restriction.Bsp68I.size
Bio.Restriction.Restriction.BspACI.size
Bio.Restriction.Restriction.BspCNI.size
Bio.Restriction.Restriction.BspD6I.size
Bio.Restriction.Restriction.BspDI.size
Bio.Restriction.Restriction.BspEI.size
Bio.Restriction.Restriction.BspFNI.size
Bio.Restriction.Restriction.BspGI.size
Bio.Restriction.Restriction.BspHI.size
Bio.Restriction.Restriction.BspLI.size
Bio.Restriction.Restriction.BspLU11I.size
Bio.Restriction.Restriction.BspMI.size
Bio.Restriction.Restriction.BspMII.size
Bio.Restriction.Restriction.BspNCI.size
Bio.Restriction.Restriction.BspOI.size
Bio.Restriction.Restriction.BspPI.size
Bio.Restriction.Restriction.BspQI.size
Bio.Restriction.Restriction.BspT104I.size
Bio.Restriction.Restriction.BspT107I.size
Bio.Restriction.Restriction.BspTI.size
Bio.Restriction.Restriction.BsrBI.size
Bio.Restriction.Restriction.BsrDI.size
Bio.Restriction.Restriction.BsrFI.size
Bio.Restriction.Restriction.BsrGI.size
Bio.Restriction.Restriction.BsrI.size
Bio.Restriction.Restriction.BsrSI.size
Bio.Restriction.Restriction.BssAI.size
Bio.Restriction.Restriction.BssECI.size
Bio.Restriction.Restriction.BssHII.size
Bio.Restriction.Restriction.BssKI.size
Bio.Restriction.Restriction.BssMI.size
Bio.Restriction.Restriction.BssNAI.size
Bio.Restriction.Restriction.BssNI.size
Bio.Restriction.Restriction.BssSI.size
Bio.Restriction.Restriction.BssT1I.size
Bio.Restriction.Restriction.Bst1107I.size
Bio.Restriction.Restriction.Bst2BI.size
Bio.Restriction.Restriction.Bst2UI.size
Bio.Restriction.Restriction.Bst4CI.size
Bio.Restriction.Restriction.Bst6I.size
Bio.Restriction.Restriction.BstACI.size
Bio.Restriction.Restriction.BstAFI.size
Bio.Restriction.Restriction.BstAPI.size
Bio.Restriction.Restriction.BstAUI.size
Bio.Restriction.Restriction.BstBAI.size
Bio.Restriction.Restriction.BstBI.size
Bio.Restriction.Restriction.BstC8I.size
Bio.Restriction.Restriction.BstDEI.size
Bio.Restriction.Restriction.BstDSI.size
Bio.Restriction.Restriction.BstEII.size
Bio.Restriction.Restriction.BstENI.size
Bio.Restriction.Restriction.BstF5I.size
Bio.Restriction.Restriction.BstFNI.size
Bio.Restriction.Restriction.BstH2I.size
Bio.Restriction.Restriction.BstHHI.size
Bio.Restriction.Restriction.BstKTI.size
Bio.Restriction.Restriction.BstMAI.size
Bio.Restriction.Restriction.BstMBI.size
Bio.Restriction.Restriction.BstMCI.size
Bio.Restriction.Restriction.BstMWI.size
Bio.Restriction.Restriction.BstNI.size
Bio.Restriction.Restriction.BstNSI.size
Bio.Restriction.Restriction.BstOI.size
Bio.Restriction.Restriction.BstPAI.size
Bio.Restriction.Restriction.BstPI.size
Bio.Restriction.Restriction.BstSCI.size
Bio.Restriction.Restriction.BstSFI.size
Bio.Restriction.Restriction.BstSLI.size
Bio.Restriction.Restriction.BstSNI.size
Bio.Restriction.Restriction.BstUI.size
Bio.Restriction.Restriction.BstV1I.size
Bio.Restriction.Restriction.BstV2I.size
Bio.Restriction.Restriction.BstX2I.size
Bio.Restriction.Restriction.BstXI.size
Bio.Restriction.Restriction.BstYI.size
Bio.Restriction.Restriction.BstZ17I.size
Bio.Restriction.Restriction.BstZI.size
Bio.Restriction.Restriction.Bsu15I.size
Bio.Restriction.Restriction.Bsu36I.size
Bio.Restriction.Restriction.BsuI.size
Bio.Restriction.Restriction.BsuRI.size
Bio.Restriction.Restriction.BtgI.size
Bio.Restriction.Restriction.BtgZI.size
Bio.Restriction.Restriction.BthCI.size
Bio.Restriction.Restriction.BtrI.size
Bio.Restriction.Restriction.BtsCI.size
Bio.Restriction.Restriction.BtsI.size
Bio.Restriction.Restriction.BtsIMutI.size
Bio.Restriction.Restriction.BtuMI.size
Bio.Restriction.Restriction.BveI.size
Bio.Restriction.Restriction.Cac8I.size
Bio.Restriction.Restriction.CaiI.size
Bio.Restriction.Restriction.CauII.size
Bio.Restriction.Restriction.CchII.size
Bio.Restriction.Restriction.CchIII.size
Bio.Restriction.Restriction.CciI.size
Bio.Restriction.Restriction.CciNI.size
Bio.Restriction.Restriction.Cdi630V.size
Bio.Restriction.Restriction.CdiI.size
Bio.Restriction.Restriction.CdpI.size
Bio.Restriction.Restriction.CfoI.size
Bio.Restriction.Restriction.Cfr10I.size
Bio.Restriction.Restriction.Cfr13I.size
Bio.Restriction.Restriction.Cfr42I.size
Bio.Restriction.Restriction.Cfr9I.size
Bio.Restriction.Restriction.CfrI.size
Bio.Restriction.Restriction.Cgl13032I.size
Bio.Restriction.Restriction.Cgl13032II.size
Bio.Restriction.Restriction.ChaI.size
Bio.Restriction.Restriction.CjeFIII.size
Bio.Restriction.Restriction.CjeFV.size
Bio.Restriction.Restriction.CjeI.size
Bio.Restriction.Restriction.CjeNII.size
Bio.Restriction.Restriction.CjeNIII.size
Bio.Restriction.Restriction.CjeP659IV.size
Bio.Restriction.Restriction.CjePI.size
Bio.Restriction.Restriction.CjuI.size
Bio.Restriction.Restriction.CjuII.size
Bio.Restriction.Restriction.ClaI.size
Bio.Restriction.Restriction.CpoI.size
Bio.Restriction.Restriction.CseI.size
Bio.Restriction.Restriction.CsiI.size
Bio.Restriction.Restriction.Csp6I.size
Bio.Restriction.Restriction.CspAI.size
Bio.Restriction.Restriction.CspCI.size
Bio.Restriction.Restriction.CspI.size
Bio.Restriction.Restriction.CstMI.size
Bio.Restriction.Restriction.CviAII.size
Bio.Restriction.Restriction.CviJI.size
Bio.Restriction.Restriction.CviKI_1.size
Bio.Restriction.Restriction.CviQI.size
Bio.Restriction.Restriction.CviRI.size
Bio.Restriction.Restriction.DdeI.size
Bio.Restriction.Restriction.DinI.size
Bio.Restriction.Restriction.DpnI.size
Bio.Restriction.Restriction.DpnII.size
Bio.Restriction.Restriction.DraI.size
Bio.Restriction.Restriction.DraII.size
Bio.Restriction.Restriction.DraIII.size
Bio.Restriction.Restriction.DraRI.size
Bio.Restriction.Restriction.DrdI.size
Bio.Restriction.Restriction.DrdII.size
Bio.Restriction.Restriction.DriI.size
Bio.Restriction.Restriction.DsaI.size
Bio.Restriction.Restriction.DseDI.size
Bio.Restriction.Restriction.EaeI.size
Bio.Restriction.Restriction.EagI.size
Bio.Restriction.Restriction.Eam1104I.size
Bio.Restriction.Restriction.Eam1105I.size
Bio.Restriction.Restriction.EarI.size
Bio.Restriction.Restriction.EciI.size
Bio.Restriction.Restriction.Ecl136II.size
Bio.Restriction.Restriction.EclXI.size
Bio.Restriction.Restriction.Eco105I.size
Bio.Restriction.Restriction.Eco130I.size
Bio.Restriction.Restriction.Eco147I.size
Bio.Restriction.Restriction.Eco24I.size
Bio.Restriction.Restriction.Eco31I.size
Bio.Restriction.Restriction.Eco32I.size
Bio.Restriction.Restriction.Eco47I.size
Bio.Restriction.Restriction.Eco47III.size
Bio.Restriction.Restriction.Eco52I.size
Bio.Restriction.Restriction.Eco53kI.size
Bio.Restriction.Restriction.Eco57I.size
Bio.Restriction.Restriction.Eco57MI.size
Bio.Restriction.Restriction.Eco72I.size
Bio.Restriction.Restriction.Eco81I.size
Bio.Restriction.Restriction.Eco88I.size
Bio.Restriction.Restriction.Eco91I.size
Bio.Restriction.Restriction.EcoHI.size
Bio.Restriction.Restriction.EcoICRI.size
Bio.Restriction.Restriction.EcoNI.size
Bio.Restriction.Restriction.EcoO109I.size
Bio.Restriction.Restriction.EcoO65I.size
Bio.Restriction.Restriction.EcoRI.size
Bio.Restriction.Restriction.EcoRII.size
Bio.Restriction.Restriction.EcoRV.size
Bio.Restriction.Restriction.EcoT14I.size
Bio.Restriction.Restriction.EcoT22I.size
Bio.Restriction.Restriction.EcoT38I.size
Bio.Restriction.Restriction.EgeI.size
Bio.Restriction.Restriction.EheI.size
Bio.Restriction.Restriction.ErhI.size
Bio.Restriction.Restriction.EsaBC3I.size
Bio.Restriction.Restriction.EsaSSI.size
Bio.Restriction.Restriction.Esp3I.size
Bio.Restriction.Restriction.EspI.size
Bio.Restriction.Restriction.FaeI.size
Bio.Restriction.Restriction.FaiI.size
Bio.Restriction.Restriction.FalI.size
Bio.Restriction.Restriction.FaqI.size
Bio.Restriction.Restriction.FatI.size
Bio.Restriction.Restriction.FauI.size
Bio.Restriction.Restriction.FauNDI.size
Bio.Restriction.Restriction.FbaI.size
Bio.Restriction.Restriction.FblI.size
Bio.Restriction.Restriction.FinI.size
Bio.Restriction.Restriction.FmuI.size
Bio.Restriction.Restriction.Fnu4HI.size
Bio.Restriction.Restriction.FnuDII.size
Bio.Restriction.Restriction.FokI.size
Bio.Restriction.Restriction.FriOI.size
Bio.Restriction.Restriction.FseI.size
Bio.Restriction.Restriction.Fsp4HI.size
Bio.Restriction.Restriction.FspAI.size
Bio.Restriction.Restriction.FspBI.size
Bio.Restriction.Restriction.FspEI.size
Bio.Restriction.Restriction.FspI.size
Bio.Restriction.Restriction.GauT27I.size
Bio.Restriction.Restriction.GdiII.size
Bio.Restriction.Restriction.GlaI.size
Bio.Restriction.Restriction.GluI.size
Bio.Restriction.Restriction.GsaI.size
Bio.Restriction.Restriction.GsuI.size
Bio.Restriction.Restriction.HaeI.size
Bio.Restriction.Restriction.HaeII.size
Bio.Restriction.Restriction.HaeIII.size
Bio.Restriction.Restriction.HapII.size
Bio.Restriction.Restriction.HauII.size
Bio.Restriction.Restriction.HgaI.size
Bio.Restriction.Restriction.HgiAI.size
Bio.Restriction.Restriction.HgiCI.size
Bio.Restriction.Restriction.HgiEII.size
Bio.Restriction.Restriction.HgiJII.size
Bio.Restriction.Restriction.HhaI.size
Bio.Restriction.Restriction.Hin1I.size
Bio.Restriction.Restriction.Hin1II.size
Bio.Restriction.Restriction.Hin4I.size
Bio.Restriction.Restriction.Hin4II.size
Bio.Restriction.Restriction.Hin6I.size
Bio.Restriction.Restriction.HinP1I.size
Bio.Restriction.Restriction.HincII.size
Bio.Restriction.Restriction.HindII.size
Bio.Restriction.Restriction.HindIII.size
Bio.Restriction.Restriction.HinfI.size
Bio.Restriction.Restriction.HpaI.size
Bio.Restriction.Restriction.HpaII.size
Bio.Restriction.Restriction.HphI.size
Bio.Restriction.Restriction.Hpy166II.size
Bio.Restriction.Restriction.Hpy178III.size
Bio.Restriction.Restriction.Hpy188I.size
Bio.Restriction.Restriction.Hpy188III.size
Bio.Restriction.Restriction.Hpy8I.size
Bio.Restriction.Restriction.Hpy99I.size
Bio.Restriction.Restriction.Hpy99XIII.size
Bio.Restriction.Restriction.Hpy99XIV.size
Bio.Restriction.Restriction.HpyAV.size
Bio.Restriction.Restriction.HpyCH4III.size
Bio.Restriction.Restriction.HpyCH4IV.size
Bio.Restriction.Restriction.HpyCH4V.size
Bio.Restriction.Restriction.HpyF10VI.size
Bio.Restriction.Restriction.HpyF3I.size
Bio.Restriction.Restriction.HpySE526I.size
Bio.Restriction.Restriction.Hsp92I.size
Bio.Restriction.Restriction.Hsp92II.size
Bio.Restriction.Restriction.HspAI.size
Bio.Restriction.Restriction.Jma19592I.size
Bio.Restriction.Restriction.KasI.size
Bio.Restriction.Restriction.KflI.size
Bio.Restriction.Restriction.Kpn2I.size
Bio.Restriction.Restriction.KpnI.size
Bio.Restriction.Restriction.KroI.size
Bio.Restriction.Restriction.Ksp22I.size
Bio.Restriction.Restriction.Ksp632I.size
Bio.Restriction.Restriction.KspAI.size
Bio.Restriction.Restriction.KspI.size
Bio.Restriction.Restriction.Kzo9I.size
Bio.Restriction.Restriction.LguI.size
Bio.Restriction.Restriction.LpnI.size
Bio.Restriction.Restriction.LpnPI.size
Bio.Restriction.Restriction.Lsp1109I.size
Bio.Restriction.Restriction.LweI.size
Bio.Restriction.Restriction.MabI.size
Bio.Restriction.Restriction.MaeI.size
Bio.Restriction.Restriction.MaeII.size
Bio.Restriction.Restriction.MaeIII.size
Bio.Restriction.Restriction.MalI.size
Bio.Restriction.Restriction.MaqI.size
Bio.Restriction.Restriction.MauBI.size
Bio.Restriction.Restriction.MbiI.size
Bio.Restriction.Restriction.MboI.size
Bio.Restriction.Restriction.MboII.size
Bio.Restriction.Restriction.McaTI.size
Bio.Restriction.Restriction.McrI.size
Bio.Restriction.Restriction.MfeI.size
Bio.Restriction.Restriction.MflI.size
Bio.Restriction.Restriction.MhlI.size
Bio.Restriction.Restriction.MjaIV.size
Bio.Restriction.Restriction.MkaDII.size
Bio.Restriction.Restriction.MlsI.size
Bio.Restriction.Restriction.MluCI.size
Bio.Restriction.Restriction.MluI.size
Bio.Restriction.Restriction.MluNI.size
Bio.Restriction.Restriction.Mly113I.size
Bio.Restriction.Restriction.MlyI.size
Bio.Restriction.Restriction.MmeI.size
Bio.Restriction.Restriction.MnlI.size
Bio.Restriction.Restriction.Mph1103I.size
Bio.Restriction.Restriction.MreI.size
Bio.Restriction.Restriction.MroI.size
Bio.Restriction.Restriction.MroNI.size
Bio.Restriction.Restriction.MroXI.size
Bio.Restriction.Restriction.MscI.size
Bio.Restriction.Restriction.MseI.size
Bio.Restriction.Restriction.MslI.size
Bio.Restriction.Restriction.Msp20I.size
Bio.Restriction.Restriction.MspA1I.size
Bio.Restriction.Restriction.MspCI.size
Bio.Restriction.Restriction.MspI.size
Bio.Restriction.Restriction.MspJI.size
Bio.Restriction.Restriction.MspR9I.size
Bio.Restriction.Restriction.MssI.size
Bio.Restriction.Restriction.MstI.size
Bio.Restriction.Restriction.MunI.size
Bio.Restriction.Restriction.Mva1269I.size
Bio.Restriction.Restriction.MvaI.size
Bio.Restriction.Restriction.MvnI.size
Bio.Restriction.Restriction.MvrI.size
Bio.Restriction.Restriction.MwoI.size
Bio.Restriction.Restriction.NaeI.size
Bio.Restriction.Restriction.NarI.size
Bio.Restriction.Restriction.NciI.size
Bio.Restriction.Restriction.NcoI.size
Bio.Restriction.Restriction.NdeI.size
Bio.Restriction.Restriction.NdeII.size
Bio.Restriction.Restriction.NgoAVIII.size
Bio.Restriction.Restriction.NgoMIV.size
Bio.Restriction.Restriction.NhaXI.size
Bio.Restriction.Restriction.NheI.size
Bio.Restriction.Restriction.NlaCI.size
Bio.Restriction.Restriction.NlaIII.size
Bio.Restriction.Restriction.NlaIV.size
Bio.Restriction.Restriction.Nli3877I.size
Bio.Restriction.Restriction.NmeAIII.size
Bio.Restriction.Restriction.NmeDI.size
Bio.Restriction.Restriction.NmuCI.size
Bio.Restriction.Restriction.NotI.size
Bio.Restriction.Restriction.NruI.size
Bio.Restriction.Restriction.NsbI.size
Bio.Restriction.Restriction.NsiI.size
Bio.Restriction.Restriction.NspBII.size
Bio.Restriction.Restriction.NspI.size
Bio.Restriction.Restriction.NspV.size
Bio.Restriction.Restriction.OliI.size
Bio.Restriction.Restriction.PabI.size
Bio.Restriction.Restriction.PacI.size
Bio.Restriction.Restriction.PaeI.size
Bio.Restriction.Restriction.PaeR7I.size
Bio.Restriction.Restriction.PagI.size
Bio.Restriction.Restriction.PalAI.size
Bio.Restriction.Restriction.PasI.size
Bio.Restriction.Restriction.PauI.size
Bio.Restriction.Restriction.PceI.size
Bio.Restriction.Restriction.PciI.size
Bio.Restriction.Restriction.PciSI.size
Bio.Restriction.Restriction.PcsI.size
Bio.Restriction.Restriction.PctI.size
Bio.Restriction.Restriction.PdiI.size
Bio.Restriction.Restriction.PdmI.size
Bio.Restriction.Restriction.PenI.size
Bio.Restriction.Restriction.PfeI.size
Bio.Restriction.Restriction.Pfl1108I.size
Bio.Restriction.Restriction.Pfl23II.size
Bio.Restriction.Restriction.PflFI.size
Bio.Restriction.Restriction.PflMI.size
Bio.Restriction.Restriction.PfoI.size
Bio.Restriction.Restriction.PinAI.size
Bio.Restriction.Restriction.PlaDI.size
Bio.Restriction.Restriction.Ple19I.size
Bio.Restriction.Restriction.PleI.size
Bio.Restriction.Restriction.PluTI.size
Bio.Restriction.Restriction.PmaCI.size
Bio.Restriction.Restriction.PmeI.size
Bio.Restriction.Restriction.PmlI.size
Bio.Restriction.Restriction.PpiI.size
Bio.Restriction.Restriction.PpsI.size
Bio.Restriction.Restriction.Ppu10I.size
Bio.Restriction.Restriction.Ppu21I.size
Bio.Restriction.Restriction.PpuMI.size
Bio.Restriction.Restriction.PscI.size
Bio.Restriction.Restriction.PshAI.size
Bio.Restriction.Restriction.PshBI.size
Bio.Restriction.Restriction.PsiI.size
Bio.Restriction.Restriction.Psp03I.size
Bio.Restriction.Restriction.Psp124BI.size
Bio.Restriction.Restriction.Psp1406I.size
Bio.Restriction.Restriction.Psp5II.size
Bio.Restriction.Restriction.Psp6I.size
Bio.Restriction.Restriction.PspCI.size
Bio.Restriction.Restriction.PspEI.size
Bio.Restriction.Restriction.PspGI.size
Bio.Restriction.Restriction.PspLI.size
Bio.Restriction.Restriction.PspN4I.size
Bio.Restriction.Restriction.PspOMI.size
Bio.Restriction.Restriction.PspOMII.size
Bio.Restriction.Restriction.PspPI.size
Bio.Restriction.Restriction.PspPPI.size
Bio.Restriction.Restriction.PspPRI.size
Bio.Restriction.Restriction.PspXI.size
Bio.Restriction.Restriction.PsrI.size
Bio.Restriction.Restriction.PssI.size
Bio.Restriction.Restriction.PstI.size
Bio.Restriction.Restriction.PstNI.size
Bio.Restriction.Restriction.PsuI.size
Bio.Restriction.Restriction.PsyI.size
Bio.Restriction.Restriction.PteI.size
Bio.Restriction.Restriction.PvuI.size
Bio.Restriction.Restriction.PvuII.size
Bio.Restriction.Restriction.R2_BceSIV.size
Bio.Restriction.Restriction.RceI.size
Bio.Restriction.Restriction.RdeGBI.size
Bio.Restriction.Restriction.RdeGBII.size
Bio.Restriction.Restriction.RdeGBIII.size
Bio.Restriction.Restriction.RflFIII.size
Bio.Restriction.Restriction.RgaI.size
Bio.Restriction.Restriction.RigI.size
Bio.Restriction.Restriction.RlaI.size
Bio.Restriction.Restriction.RleAI.size
Bio.Restriction.Restriction.RpaB5I.size
Bio.Restriction.Restriction.RpaBI.size
Bio.Restriction.Restriction.RpaI.size
Bio.Restriction.Restriction.RpaTI.size
Bio.Restriction.Restriction.RruI.size
Bio.Restriction.Restriction.RsaI.size
Bio.Restriction.Restriction.RsaNI.size
Bio.Restriction.Restriction.RseI.size
Bio.Restriction.Restriction.Rsr2I.size
Bio.Restriction.Restriction.RsrII.size
Bio.Restriction.Restriction.SacI.size
Bio.Restriction.Restriction.SacII.size
Bio.Restriction.Restriction.SalI.size
Bio.Restriction.Restriction.SanDI.size
Bio.Restriction.Restriction.SapI.size
Bio.Restriction.Restriction.SaqAI.size
Bio.Restriction.Restriction.SatI.size
Bio.Restriction.Restriction.Sau3AI.size
Bio.Restriction.Restriction.Sau96I.size
Bio.Restriction.Restriction.SauI.size
Bio.Restriction.Restriction.SbfI.size
Bio.Restriction.Restriction.ScaI.size
Bio.Restriction.Restriction.SchI.size
Bio.Restriction.Restriction.SciI.size
Bio.Restriction.Restriction.ScrFI.size
Bio.Restriction.Restriction.SdaI.size
Bio.Restriction.Restriction.SdeAI.size
Bio.Restriction.Restriction.SdeOSI.size
Bio.Restriction.Restriction.SduI.size
Bio.Restriction.Restriction.SecI.size
Bio.Restriction.Restriction.SelI.size
Bio.Restriction.Restriction.SetI.size
Bio.Restriction.Restriction.SexAI.size
Bio.Restriction.Restriction.SfaAI.size
Bio.Restriction.Restriction.SfaNI.size
Bio.Restriction.Restriction.SfcI.size
Bio.Restriction.Restriction.SfeI.size
Bio.Restriction.Restriction.SfiI.size
Bio.Restriction.Restriction.SfoI.size
Bio.Restriction.Restriction.Sfr274I.size
Bio.Restriction.Restriction.Sfr303I.size
Bio.Restriction.Restriction.SfuI.size
Bio.Restriction.Restriction.SgeI.size
Bio.Restriction.Restriction.SgfI.size
Bio.Restriction.Restriction.SgrAI.size
Bio.Restriction.Restriction.SgrBI.size
Bio.Restriction.Restriction.SgrDI.size
Bio.Restriction.Restriction.SgrTI.size
Bio.Restriction.Restriction.SgsI.size
Bio.Restriction.Restriction.SimI.size
Bio.Restriction.Restriction.SlaI.size
Bio.Restriction.Restriction.SmaI.size
Bio.Restriction.Restriction.SmiI.size
Bio.Restriction.Restriction.SmiMI.size
Bio.Restriction.Restriction.SmlI.size
Bio.Restriction.Restriction.SmoI.size
Bio.Restriction.Restriction.SnaBI.size
Bio.Restriction.Restriction.SnaI.size
Bio.Restriction.Restriction.Sno506I.size
Bio.Restriction.Restriction.SpeI.size
Bio.Restriction.Restriction.SphI.size
Bio.Restriction.Restriction.SplI.size
Bio.Restriction.Restriction.SpoDI.size
Bio.Restriction.Restriction.SrfI.size
Bio.Restriction.Restriction.Sse232I.size
Bio.Restriction.Restriction.Sse8387I.size
Bio.Restriction.Restriction.Sse8647I.size
Bio.Restriction.Restriction.Sse9I.size
Bio.Restriction.Restriction.SseBI.size
Bio.Restriction.Restriction.SsiI.size
Bio.Restriction.Restriction.SspD5I.size
Bio.Restriction.Restriction.SspDI.size
Bio.Restriction.Restriction.SspI.size
Bio.Restriction.Restriction.SstE37I.size
Bio.Restriction.Restriction.SstI.size
Bio.Restriction.Restriction.Sth132I.size
Bio.Restriction.Restriction.Sth302II.size
Bio.Restriction.Restriction.StrI.size
Bio.Restriction.Restriction.StsI.size
Bio.Restriction.Restriction.StuI.size
Bio.Restriction.Restriction.StyD4I.size
Bio.Restriction.Restriction.StyI.size
Bio.Restriction.Restriction.SwaI.size
Bio.Restriction.Restriction.TaaI.size
Bio.Restriction.Restriction.TaiI.size
Bio.Restriction.Restriction.TaqI.size
Bio.Restriction.Restriction.TaqII.size
Bio.Restriction.Restriction.TasI.size
Bio.Restriction.Restriction.TatI.size
Bio.Restriction.Restriction.TauI.size
Bio.Restriction.Restriction.TfiI.size
Bio.Restriction.Restriction.Tru1I.size
Bio.Restriction.Restriction.Tru9I.size
Bio.Restriction.Restriction.TscAI.size
Bio.Restriction.Restriction.TseFI.size
Bio.Restriction.Restriction.TseI.size
Bio.Restriction.Restriction.TsoI.size
Bio.Restriction.Restriction.Tsp45I.size
Bio.Restriction.Restriction.Tsp4CI.size
Bio.Restriction.Restriction.TspDTI.size
Bio.Restriction.Restriction.TspEI.size
Bio.Restriction.Restriction.TspGWI.size
Bio.Restriction.Restriction.TspMI.size
Bio.Restriction.Restriction.TspRI.size
Bio.Restriction.Restriction.TssI.size
Bio.Restriction.Restriction.TstI.size
Bio.Restriction.Restriction.TsuI.size
Bio.Restriction.Restriction.Tth111I.size
Bio.Restriction.Restriction.Tth111II.size
Bio.Restriction.Restriction.UbaF11I.size
Bio.Restriction.Restriction.UbaF12I.size
Bio.Restriction.Restriction.UbaF13I.size
Bio.Restriction.Restriction.UbaF14I.size
Bio.Restriction.Restriction.UbaF9I.size
Bio.Restriction.Restriction.UbaPI.size
Bio.Restriction.Restriction.UcoMSI.size
Bio.Restriction.Restriction.UnbI.size
Bio.Restriction.Restriction.Van91I.size
Bio.Restriction.Restriction.Vha464I.size
Bio.Restriction.Restriction.VneI.size
Bio.Restriction.Restriction.VpaK11AI.size
Bio.Restriction.Restriction.VpaK11BI.size
Bio.Restriction.Restriction.VspI.size
Bio.Restriction.Restriction.WviI.size
Bio.Restriction.Restriction.XagI.size
Bio.Restriction.Restriction.XapI.size
Bio.Restriction.Restriction.XbaI.size
Bio.Restriction.Restriction.XceI.size
Bio.Restriction.Restriction.XcmI.size
Bio.Restriction.Restriction.XhoI.size
Bio.Restriction.Restriction.XhoII.size
Bio.Restriction.Restriction.XmaI.size
Bio.Restriction.Restriction.XmaIII.size
Bio.Restriction.Restriction.XmaJI.size
Bio.Restriction.Restriction.XmiI.size
Bio.Restriction.Restriction.XmnI.size
Bio.Restriction.Restriction.XspI.size
Bio.Restriction.Restriction.YkrI.size
Bio.Restriction.Restriction.ZraI.size
Bio.Restriction.Restriction.ZrmI.size
Bio.Restriction.Restriction.Zsp2I.size
Bio.codonalign.codonalphabet.CodonAlphabet.size" class="py-name" href="#" onclick="return doclink('link-474', 'size', 'link-270');">size</a></tt><tt class="py-op">(</tt><tt class="py-name">splitline</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1202"></a><tt class="py-lineno">1202</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-475" class="py-name"><a title="Bio.GenBank._FeatureConsumer.residue_type
Bio.GenBank._RecordConsumer.residue_type" class="py-name" href="#" onclick="return doclink('link-475', 'residue_type', 'link-236');">residue_type</a></tt><tt class="py-op">(</tt><tt class="py-name">splitline</tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1203"></a><tt class="py-lineno">1203</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-476" class="py-name"><a title="Bio.GenBank._FeatureConsumer.data_file_division
Bio.GenBank._RecordConsumer.data_file_division" class="py-name" href="#" onclick="return doclink('link-476', 'data_file_division', 'link-237');">data_file_division</a></tt><tt class="py-op">(</tt><tt class="py-name">splitline</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1204"></a><tt class="py-lineno">1204</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-477" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-477', 'date', 'link-439');">date</a></tt><tt class="py-op">(</tt><tt class="py-name">splitline</tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1205"></a><tt class="py-lineno">1205</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Attempting to parse malformed locus line:\n%r\n"</tt> </tt>
<a name="L1206"></a><tt class="py-lineno">1206</tt>  <tt class="py-line">                          <tt class="py-string">"Found locus %r size %r residue_type %r\n"</tt> </tt>
<a name="L1207"></a><tt class="py-lineno">1207</tt>  <tt class="py-line">                          <tt class="py-string">"Some fields may be wrong."</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-name">splitline</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L1208"></a><tt class="py-lineno">1208</tt>  <tt class="py-line">                          <tt class="py-name">splitline</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">splitline</tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-478" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-478', 'BiopythonParserWarning', 'link-10');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1209"></a><tt class="py-lineno">1209</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-479" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-479', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">7</tt> <tt class="py-keyword">and</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-480" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-480', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"aa"</tt><tt class="py-op">,</tt> <tt class="py-string">"bp"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L1210"></a><tt class="py-lineno">1210</tt>  <tt class="py-line">            <tt class="py-comment"># Cope with EnsEMBL genbank files which use space separation rather</tt> </tt>
<a name="L1211"></a><tt class="py-lineno">1211</tt>  <tt class="py-line">            <tt class="py-comment"># than the expected column based layout. e.g.</tt> </tt>
<a name="L1212"></a><tt class="py-lineno">1212</tt>  <tt class="py-line">            <tt class="py-comment"># LOCUS       HG531_PATCH 1000000 bp DNA HTG 18-JUN-2011</tt> </tt>
<a name="L1213"></a><tt class="py-lineno">1213</tt>  <tt class="py-line">            <tt class="py-comment"># LOCUS       HG531_PATCH 759984 bp DNA HTG 18-JUN-2011</tt> </tt>
<a name="L1214"></a><tt class="py-lineno">1214</tt>  <tt class="py-line">            <tt class="py-comment"># LOCUS       HG506_HG1000_1_PATCH 814959 bp DNA HTG 18-JUN-2011</tt> </tt>
<a name="L1215"></a><tt class="py-lineno">1215</tt>  <tt class="py-line">            <tt class="py-comment"># LOCUS       HG506_HG1000_1_PATCH 1219964 bp DNA HTG 18-JUN-2011</tt> </tt>
<a name="L1216"></a><tt class="py-lineno">1216</tt>  <tt class="py-line">            <tt class="py-comment"># Notice that the 'bp' can occur in the position expected by either</tt> </tt>
<a name="L1217"></a><tt class="py-lineno">1217</tt>  <tt class="py-line">            <tt class="py-comment"># the old or the new fixed column standards (parsed above).</tt> </tt>
<a name="L1218"></a><tt class="py-lineno">1218</tt>  <tt class="py-line">            <tt class="py-name">splitline</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-481" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-481', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1219"></a><tt class="py-lineno">1219</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-482" class="py-name"><a title="Bio.GenBank._FeatureConsumer.locus
Bio.GenBank._RecordConsumer.locus" class="py-name" href="#" onclick="return doclink('link-482', 'locus', 'link-235');">locus</a></tt><tt class="py-op">(</tt><tt class="py-name">splitline</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1220"></a><tt class="py-lineno">1220</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-483" class="py-name"><a title="Bio.Alphabet.Alphabet.size
Bio.Alphabet.Reduced.HPModel.size
Bio.Alphabet.Reduced.Murphy10.size
Bio.Alphabet.Reduced.Murphy15.size
Bio.Alphabet.Reduced.Murphy4.size
Bio.Alphabet.Reduced.Murphy8.size
Bio.Alphabet.Reduced.PC5.size
Bio.Alphabet.SingleLetterAlphabet.size
Bio.Alphabet.ThreeLetterProtein.size
Bio.GenBank._FeatureConsumer.size
Bio.GenBank._RecordConsumer.size
Bio.Restriction.Restriction.AanI.size
Bio.Restriction.Restriction.AarI.size
Bio.Restriction.Restriction.AasI.size
Bio.Restriction.Restriction.AatII.size
Bio.Restriction.Restriction.AbaSI.size
Bio.Restriction.Restriction.AbsI.size
Bio.Restriction.Restriction.Acc16I.size
Bio.Restriction.Restriction.Acc36I.size
Bio.Restriction.Restriction.Acc65I.size
Bio.Restriction.Restriction.AccB1I.size
Bio.Restriction.Restriction.AccB7I.size
Bio.Restriction.Restriction.AccBSI.size
Bio.Restriction.Restriction.AccI.size
Bio.Restriction.Restriction.AccII.size
Bio.Restriction.Restriction.AccIII.size
Bio.Restriction.Restriction.AceIII.size
Bio.Restriction.Restriction.AciI.size
Bio.Restriction.Restriction.AclI.size
Bio.Restriction.Restriction.AclWI.size
Bio.Restriction.Restriction.AcoI.size
Bio.Restriction.Restriction.AcsI.size
Bio.Restriction.Restriction.AcuI.size
Bio.Restriction.Restriction.AcvI.size
Bio.Restriction.Restriction.AcyI.size
Bio.Restriction.Restriction.AdeI.size
Bio.Restriction.Restriction.AfaI.size
Bio.Restriction.Restriction.AfeI.size
Bio.Restriction.Restriction.AfiI.size
Bio.Restriction.Restriction.AflII.size
Bio.Restriction.Restriction.AflIII.size
Bio.Restriction.Restriction.AgeI.size
Bio.Restriction.Restriction.AgsI.size
Bio.Restriction.Restriction.AhaIII.size
Bio.Restriction.Restriction.AhdI.size
Bio.Restriction.Restriction.AhlI.size
Bio.Restriction.Restriction.AjiI.size
Bio.Restriction.Restriction.AjnI.size
Bio.Restriction.Restriction.AjuI.size
Bio.Restriction.Restriction.AleI.size
Bio.Restriction.Restriction.AlfI.size
Bio.Restriction.Restriction.AloI.size
Bio.Restriction.Restriction.AluBI.size
Bio.Restriction.Restriction.AluI.size
Bio.Restriction.Restriction.Alw21I.size
Bio.Restriction.Restriction.Alw26I.size
Bio.Restriction.Restriction.Alw44I.size
Bio.Restriction.Restriction.AlwFI.size
Bio.Restriction.Restriction.AlwI.size
Bio.Restriction.Restriction.AlwNI.size
Bio.Restriction.Restriction.Ama87I.size
Bio.Restriction.Restriction.Aor13HI.size
Bio.Restriction.Restriction.Aor51HI.size
Bio.Restriction.Restriction.AoxI.size
Bio.Restriction.Restriction.ApaBI.size
Bio.Restriction.Restriction.ApaI.size
Bio.Restriction.Restriction.ApaLI.size
Bio.Restriction.Restriction.ApeKI.size
Bio.Restriction.Restriction.ApoI.size
Bio.Restriction.Restriction.ApyPI.size
Bio.Restriction.Restriction.AquII.size
Bio.Restriction.Restriction.AquIII.size
Bio.Restriction.Restriction.AquIV.size
Bio.Restriction.Restriction.ArsI.size
Bio.Restriction.Restriction.AscI.size
Bio.Restriction.Restriction.AseI.size
Bio.Restriction.Restriction.Asi256I.size
Bio.Restriction.Restriction.AsiGI.size
Bio.Restriction.Restriction.AsiSI.size
Bio.Restriction.Restriction.Asp700I.size
Bio.Restriction.Restriction.Asp718I.size
Bio.Restriction.Restriction.AspA2I.size
Bio.Restriction.Restriction.AspBHI.size
Bio.Restriction.Restriction.AspLEI.size
Bio.Restriction.Restriction.AspS9I.size
Bio.Restriction.Restriction.AssI.size
Bio.Restriction.Restriction.AsuC2I.size
Bio.Restriction.Restriction.AsuHPI.size
Bio.Restriction.Restriction.AsuI.size
Bio.Restriction.Restriction.AsuII.size
Bio.Restriction.Restriction.AsuNHI.size
Bio.Restriction.Restriction.AvaI.size
Bio.Restriction.Restriction.AvaII.size
Bio.Restriction.Restriction.AvaIII.size
Bio.Restriction.Restriction.AvrII.size
Bio.Restriction.Restriction.AxyI.size
Bio.Restriction.Restriction.BaeGI.size
Bio.Restriction.Restriction.BaeI.size
Bio.Restriction.Restriction.BalI.size
Bio.Restriction.Restriction.BamHI.size
Bio.Restriction.Restriction.BanI.size
Bio.Restriction.Restriction.BanII.size
Bio.Restriction.Restriction.BarI.size
Bio.Restriction.Restriction.BasI.size
Bio.Restriction.Restriction.BauI.size
Bio.Restriction.Restriction.Bbr7I.size
Bio.Restriction.Restriction.BbrPI.size
Bio.Restriction.Restriction.BbsI.size
Bio.Restriction.Restriction.Bbv12I.size
Bio.Restriction.Restriction.BbvCI.size
Bio.Restriction.Restriction.BbvI.size
Bio.Restriction.Restriction.BbvII.size
Bio.Restriction.Restriction.BccI.size
Bio.Restriction.Restriction.Bce83I.size
Bio.Restriction.Restriction.BceAI.size
Bio.Restriction.Restriction.BcefI.size
Bio.Restriction.Restriction.BcgI.size
Bio.Restriction.Restriction.BciT130I.size
Bio.Restriction.Restriction.BciVI.size
Bio.Restriction.Restriction.BclI.size
Bio.Restriction.Restriction.BcnI.size
Bio.Restriction.Restriction.BcoDI.size
Bio.Restriction.Restriction.BcuI.size
Bio.Restriction.Restriction.BdaI.size
Bio.Restriction.Restriction.BetI.size
Bio.Restriction.Restriction.BfaI.size
Bio.Restriction.Restriction.BfiI.size
Bio.Restriction.Restriction.BfmI.size
Bio.Restriction.Restriction.BfoI.size
Bio.Restriction.Restriction.BfrI.size
Bio.Restriction.Restriction.BfuAI.size
Bio.Restriction.Restriction.BfuCI.size
Bio.Restriction.Restriction.BfuI.size
Bio.Restriction.Restriction.BglI.size
Bio.Restriction.Restriction.BglII.size
Bio.Restriction.Restriction.BinI.size
Bio.Restriction.Restriction.BisI.size
Bio.Restriction.Restriction.BlnI.size
Bio.Restriction.Restriction.BlpI.size
Bio.Restriction.Restriction.BlsI.size
Bio.Restriction.Restriction.BmcAI.size
Bio.Restriction.Restriction.Bme1390I.size
Bio.Restriction.Restriction.Bme18I.size
Bio.Restriction.Restriction.BmeDI.size
Bio.Restriction.Restriction.BmeRI.size
Bio.Restriction.Restriction.BmeT110I.size
Bio.Restriction.Restriction.BmgBI.size
Bio.Restriction.Restriction.BmgI.size
Bio.Restriction.Restriction.BmgT120I.size
Bio.Restriction.Restriction.BmiI.size
Bio.Restriction.Restriction.BmrFI.size
Bio.Restriction.Restriction.BmrI.size
Bio.Restriction.Restriction.BmsI.size
Bio.Restriction.Restriction.BmtI.size
Bio.Restriction.Restriction.BmuI.size
Bio.Restriction.Restriction.BoxI.size
Bio.Restriction.Restriction.BpiI.size
Bio.Restriction.Restriction.BplI.size
Bio.Restriction.Restriction.BpmI.size
Bio.Restriction.Restriction.Bpu10I.size
Bio.Restriction.Restriction.Bpu1102I.size
Bio.Restriction.Restriction.Bpu14I.size
Bio.Restriction.Restriction.BpuEI.size
Bio.Restriction.Restriction.BpuMI.size
Bio.Restriction.Restriction.BpvUI.size
Bio.Restriction.Restriction.Bsa29I.size
Bio.Restriction.Restriction.BsaAI.size
Bio.Restriction.Restriction.BsaBI.size
Bio.Restriction.Restriction.BsaHI.size
Bio.Restriction.Restriction.BsaI.size
Bio.Restriction.Restriction.BsaJI.size
Bio.Restriction.Restriction.BsaWI.size
Bio.Restriction.Restriction.BsaXI.size
Bio.Restriction.Restriction.BsbI.size
Bio.Restriction.Restriction.Bsc4I.size
Bio.Restriction.Restriction.BscAI.size
Bio.Restriction.Restriction.BscGI.size
Bio.Restriction.Restriction.Bse118I.size
Bio.Restriction.Restriction.Bse1I.size
Bio.Restriction.Restriction.Bse21I.size
Bio.Restriction.Restriction.Bse3DI.size
Bio.Restriction.Restriction.Bse8I.size
Bio.Restriction.Restriction.BseAI.size
Bio.Restriction.Restriction.BseBI.size
Bio.Restriction.Restriction.BseCI.size
Bio.Restriction.Restriction.BseDI.size
Bio.Restriction.Restriction.BseGI.size
Bio.Restriction.Restriction.BseJI.size
Bio.Restriction.Restriction.BseLI.size
Bio.Restriction.Restriction.BseMI.size
Bio.Restriction.Restriction.BseMII.size
Bio.Restriction.Restriction.BseNI.size
Bio.Restriction.Restriction.BsePI.size
Bio.Restriction.Restriction.BseRI.size
Bio.Restriction.Restriction.BseSI.size
Bio.Restriction.Restriction.BseX3I.size
Bio.Restriction.Restriction.BseXI.size
Bio.Restriction.Restriction.BseYI.size
Bio.Restriction.Restriction.BsgI.size
Bio.Restriction.Restriction.Bsh1236I.size
Bio.Restriction.Restriction.Bsh1285I.size
Bio.Restriction.Restriction.BshFI.size
Bio.Restriction.Restriction.BshNI.size
Bio.Restriction.Restriction.BshTI.size
Bio.Restriction.Restriction.BshVI.size
Bio.Restriction.Restriction.BsiEI.size
Bio.Restriction.Restriction.BsiHKAI.size
Bio.Restriction.Restriction.BsiHKCI.size
Bio.Restriction.Restriction.BsiI.size
Bio.Restriction.Restriction.BsiSI.size
Bio.Restriction.Restriction.BsiWI.size
Bio.Restriction.Restriction.BsiYI.size
Bio.Restriction.Restriction.BslFI.size
Bio.Restriction.Restriction.BslI.size
Bio.Restriction.Restriction.BsmAI.size
Bio.Restriction.Restriction.BsmBI.size
Bio.Restriction.Restriction.BsmFI.size
Bio.Restriction.Restriction.BsmI.size
Bio.Restriction.Restriction.BsnI.size
Bio.Restriction.Restriction.Bso31I.size
Bio.Restriction.Restriction.BsoBI.size
Bio.Restriction.Restriction.Bsp119I.size
Bio.Restriction.Restriction.Bsp120I.size
Bio.Restriction.Restriction.Bsp1286I.size
Bio.Restriction.Restriction.Bsp13I.size
Bio.Restriction.Restriction.Bsp1407I.size
Bio.Restriction.Restriction.Bsp143I.size
Bio.Restriction.Restriction.Bsp1720I.size
Bio.Restriction.Restriction.Bsp19I.size
Bio.Restriction.Restriction.Bsp24I.size
Bio.Restriction.Restriction.Bsp68I.size
Bio.Restriction.Restriction.BspACI.size
Bio.Restriction.Restriction.BspCNI.size
Bio.Restriction.Restriction.BspD6I.size
Bio.Restriction.Restriction.BspDI.size
Bio.Restriction.Restriction.BspEI.size
Bio.Restriction.Restriction.BspFNI.size
Bio.Restriction.Restriction.BspGI.size
Bio.Restriction.Restriction.BspHI.size
Bio.Restriction.Restriction.BspLI.size
Bio.Restriction.Restriction.BspLU11I.size
Bio.Restriction.Restriction.BspMI.size
Bio.Restriction.Restriction.BspMII.size
Bio.Restriction.Restriction.BspNCI.size
Bio.Restriction.Restriction.BspOI.size
Bio.Restriction.Restriction.BspPI.size
Bio.Restriction.Restriction.BspQI.size
Bio.Restriction.Restriction.BspT104I.size
Bio.Restriction.Restriction.BspT107I.size
Bio.Restriction.Restriction.BspTI.size
Bio.Restriction.Restriction.BsrBI.size
Bio.Restriction.Restriction.BsrDI.size
Bio.Restriction.Restriction.BsrFI.size
Bio.Restriction.Restriction.BsrGI.size
Bio.Restriction.Restriction.BsrI.size
Bio.Restriction.Restriction.BsrSI.size
Bio.Restriction.Restriction.BssAI.size
Bio.Restriction.Restriction.BssECI.size
Bio.Restriction.Restriction.BssHII.size
Bio.Restriction.Restriction.BssKI.size
Bio.Restriction.Restriction.BssMI.size
Bio.Restriction.Restriction.BssNAI.size
Bio.Restriction.Restriction.BssNI.size
Bio.Restriction.Restriction.BssSI.size
Bio.Restriction.Restriction.BssT1I.size
Bio.Restriction.Restriction.Bst1107I.size
Bio.Restriction.Restriction.Bst2BI.size
Bio.Restriction.Restriction.Bst2UI.size
Bio.Restriction.Restriction.Bst4CI.size
Bio.Restriction.Restriction.Bst6I.size
Bio.Restriction.Restriction.BstACI.size
Bio.Restriction.Restriction.BstAFI.size
Bio.Restriction.Restriction.BstAPI.size
Bio.Restriction.Restriction.BstAUI.size
Bio.Restriction.Restriction.BstBAI.size
Bio.Restriction.Restriction.BstBI.size
Bio.Restriction.Restriction.BstC8I.size
Bio.Restriction.Restriction.BstDEI.size
Bio.Restriction.Restriction.BstDSI.size
Bio.Restriction.Restriction.BstEII.size
Bio.Restriction.Restriction.BstENI.size
Bio.Restriction.Restriction.BstF5I.size
Bio.Restriction.Restriction.BstFNI.size
Bio.Restriction.Restriction.BstH2I.size
Bio.Restriction.Restriction.BstHHI.size
Bio.Restriction.Restriction.BstKTI.size
Bio.Restriction.Restriction.BstMAI.size
Bio.Restriction.Restriction.BstMBI.size
Bio.Restriction.Restriction.BstMCI.size
Bio.Restriction.Restriction.BstMWI.size
Bio.Restriction.Restriction.BstNI.size
Bio.Restriction.Restriction.BstNSI.size
Bio.Restriction.Restriction.BstOI.size
Bio.Restriction.Restriction.BstPAI.size
Bio.Restriction.Restriction.BstPI.size
Bio.Restriction.Restriction.BstSCI.size
Bio.Restriction.Restriction.BstSFI.size
Bio.Restriction.Restriction.BstSLI.size
Bio.Restriction.Restriction.BstSNI.size
Bio.Restriction.Restriction.BstUI.size
Bio.Restriction.Restriction.BstV1I.size
Bio.Restriction.Restriction.BstV2I.size
Bio.Restriction.Restriction.BstX2I.size
Bio.Restriction.Restriction.BstXI.size
Bio.Restriction.Restriction.BstYI.size
Bio.Restriction.Restriction.BstZ17I.size
Bio.Restriction.Restriction.BstZI.size
Bio.Restriction.Restriction.Bsu15I.size
Bio.Restriction.Restriction.Bsu36I.size
Bio.Restriction.Restriction.BsuI.size
Bio.Restriction.Restriction.BsuRI.size
Bio.Restriction.Restriction.BtgI.size
Bio.Restriction.Restriction.BtgZI.size
Bio.Restriction.Restriction.BthCI.size
Bio.Restriction.Restriction.BtrI.size
Bio.Restriction.Restriction.BtsCI.size
Bio.Restriction.Restriction.BtsI.size
Bio.Restriction.Restriction.BtsIMutI.size
Bio.Restriction.Restriction.BtuMI.size
Bio.Restriction.Restriction.BveI.size
Bio.Restriction.Restriction.Cac8I.size
Bio.Restriction.Restriction.CaiI.size
Bio.Restriction.Restriction.CauII.size
Bio.Restriction.Restriction.CchII.size
Bio.Restriction.Restriction.CchIII.size
Bio.Restriction.Restriction.CciI.size
Bio.Restriction.Restriction.CciNI.size
Bio.Restriction.Restriction.Cdi630V.size
Bio.Restriction.Restriction.CdiI.size
Bio.Restriction.Restriction.CdpI.size
Bio.Restriction.Restriction.CfoI.size
Bio.Restriction.Restriction.Cfr10I.size
Bio.Restriction.Restriction.Cfr13I.size
Bio.Restriction.Restriction.Cfr42I.size
Bio.Restriction.Restriction.Cfr9I.size
Bio.Restriction.Restriction.CfrI.size
Bio.Restriction.Restriction.Cgl13032I.size
Bio.Restriction.Restriction.Cgl13032II.size
Bio.Restriction.Restriction.ChaI.size
Bio.Restriction.Restriction.CjeFIII.size
Bio.Restriction.Restriction.CjeFV.size
Bio.Restriction.Restriction.CjeI.size
Bio.Restriction.Restriction.CjeNII.size
Bio.Restriction.Restriction.CjeNIII.size
Bio.Restriction.Restriction.CjeP659IV.size
Bio.Restriction.Restriction.CjePI.size
Bio.Restriction.Restriction.CjuI.size
Bio.Restriction.Restriction.CjuII.size
Bio.Restriction.Restriction.ClaI.size
Bio.Restriction.Restriction.CpoI.size
Bio.Restriction.Restriction.CseI.size
Bio.Restriction.Restriction.CsiI.size
Bio.Restriction.Restriction.Csp6I.size
Bio.Restriction.Restriction.CspAI.size
Bio.Restriction.Restriction.CspCI.size
Bio.Restriction.Restriction.CspI.size
Bio.Restriction.Restriction.CstMI.size
Bio.Restriction.Restriction.CviAII.size
Bio.Restriction.Restriction.CviJI.size
Bio.Restriction.Restriction.CviKI_1.size
Bio.Restriction.Restriction.CviQI.size
Bio.Restriction.Restriction.CviRI.size
Bio.Restriction.Restriction.DdeI.size
Bio.Restriction.Restriction.DinI.size
Bio.Restriction.Restriction.DpnI.size
Bio.Restriction.Restriction.DpnII.size
Bio.Restriction.Restriction.DraI.size
Bio.Restriction.Restriction.DraII.size
Bio.Restriction.Restriction.DraIII.size
Bio.Restriction.Restriction.DraRI.size
Bio.Restriction.Restriction.DrdI.size
Bio.Restriction.Restriction.DrdII.size
Bio.Restriction.Restriction.DriI.size
Bio.Restriction.Restriction.DsaI.size
Bio.Restriction.Restriction.DseDI.size
Bio.Restriction.Restriction.EaeI.size
Bio.Restriction.Restriction.EagI.size
Bio.Restriction.Restriction.Eam1104I.size
Bio.Restriction.Restriction.Eam1105I.size
Bio.Restriction.Restriction.EarI.size
Bio.Restriction.Restriction.EciI.size
Bio.Restriction.Restriction.Ecl136II.size
Bio.Restriction.Restriction.EclXI.size
Bio.Restriction.Restriction.Eco105I.size
Bio.Restriction.Restriction.Eco130I.size
Bio.Restriction.Restriction.Eco147I.size
Bio.Restriction.Restriction.Eco24I.size
Bio.Restriction.Restriction.Eco31I.size
Bio.Restriction.Restriction.Eco32I.size
Bio.Restriction.Restriction.Eco47I.size
Bio.Restriction.Restriction.Eco47III.size
Bio.Restriction.Restriction.Eco52I.size
Bio.Restriction.Restriction.Eco53kI.size
Bio.Restriction.Restriction.Eco57I.size
Bio.Restriction.Restriction.Eco57MI.size
Bio.Restriction.Restriction.Eco72I.size
Bio.Restriction.Restriction.Eco81I.size
Bio.Restriction.Restriction.Eco88I.size
Bio.Restriction.Restriction.Eco91I.size
Bio.Restriction.Restriction.EcoHI.size
Bio.Restriction.Restriction.EcoICRI.size
Bio.Restriction.Restriction.EcoNI.size
Bio.Restriction.Restriction.EcoO109I.size
Bio.Restriction.Restriction.EcoO65I.size
Bio.Restriction.Restriction.EcoRI.size
Bio.Restriction.Restriction.EcoRII.size
Bio.Restriction.Restriction.EcoRV.size
Bio.Restriction.Restriction.EcoT14I.size
Bio.Restriction.Restriction.EcoT22I.size
Bio.Restriction.Restriction.EcoT38I.size
Bio.Restriction.Restriction.EgeI.size
Bio.Restriction.Restriction.EheI.size
Bio.Restriction.Restriction.ErhI.size
Bio.Restriction.Restriction.EsaBC3I.size
Bio.Restriction.Restriction.EsaSSI.size
Bio.Restriction.Restriction.Esp3I.size
Bio.Restriction.Restriction.EspI.size
Bio.Restriction.Restriction.FaeI.size
Bio.Restriction.Restriction.FaiI.size
Bio.Restriction.Restriction.FalI.size
Bio.Restriction.Restriction.FaqI.size
Bio.Restriction.Restriction.FatI.size
Bio.Restriction.Restriction.FauI.size
Bio.Restriction.Restriction.FauNDI.size
Bio.Restriction.Restriction.FbaI.size
Bio.Restriction.Restriction.FblI.size
Bio.Restriction.Restriction.FinI.size
Bio.Restriction.Restriction.FmuI.size
Bio.Restriction.Restriction.Fnu4HI.size
Bio.Restriction.Restriction.FnuDII.size
Bio.Restriction.Restriction.FokI.size
Bio.Restriction.Restriction.FriOI.size
Bio.Restriction.Restriction.FseI.size
Bio.Restriction.Restriction.Fsp4HI.size
Bio.Restriction.Restriction.FspAI.size
Bio.Restriction.Restriction.FspBI.size
Bio.Restriction.Restriction.FspEI.size
Bio.Restriction.Restriction.FspI.size
Bio.Restriction.Restriction.GauT27I.size
Bio.Restriction.Restriction.GdiII.size
Bio.Restriction.Restriction.GlaI.size
Bio.Restriction.Restriction.GluI.size
Bio.Restriction.Restriction.GsaI.size
Bio.Restriction.Restriction.GsuI.size
Bio.Restriction.Restriction.HaeI.size
Bio.Restriction.Restriction.HaeII.size
Bio.Restriction.Restriction.HaeIII.size
Bio.Restriction.Restriction.HapII.size
Bio.Restriction.Restriction.HauII.size
Bio.Restriction.Restriction.HgaI.size
Bio.Restriction.Restriction.HgiAI.size
Bio.Restriction.Restriction.HgiCI.size
Bio.Restriction.Restriction.HgiEII.size
Bio.Restriction.Restriction.HgiJII.size
Bio.Restriction.Restriction.HhaI.size
Bio.Restriction.Restriction.Hin1I.size
Bio.Restriction.Restriction.Hin1II.size
Bio.Restriction.Restriction.Hin4I.size
Bio.Restriction.Restriction.Hin4II.size
Bio.Restriction.Restriction.Hin6I.size
Bio.Restriction.Restriction.HinP1I.size
Bio.Restriction.Restriction.HincII.size
Bio.Restriction.Restriction.HindII.size
Bio.Restriction.Restriction.HindIII.size
Bio.Restriction.Restriction.HinfI.size
Bio.Restriction.Restriction.HpaI.size
Bio.Restriction.Restriction.HpaII.size
Bio.Restriction.Restriction.HphI.size
Bio.Restriction.Restriction.Hpy166II.size
Bio.Restriction.Restriction.Hpy178III.size
Bio.Restriction.Restriction.Hpy188I.size
Bio.Restriction.Restriction.Hpy188III.size
Bio.Restriction.Restriction.Hpy8I.size
Bio.Restriction.Restriction.Hpy99I.size
Bio.Restriction.Restriction.Hpy99XIII.size
Bio.Restriction.Restriction.Hpy99XIV.size
Bio.Restriction.Restriction.HpyAV.size
Bio.Restriction.Restriction.HpyCH4III.size
Bio.Restriction.Restriction.HpyCH4IV.size
Bio.Restriction.Restriction.HpyCH4V.size
Bio.Restriction.Restriction.HpyF10VI.size
Bio.Restriction.Restriction.HpyF3I.size
Bio.Restriction.Restriction.HpySE526I.size
Bio.Restriction.Restriction.Hsp92I.size
Bio.Restriction.Restriction.Hsp92II.size
Bio.Restriction.Restriction.HspAI.size
Bio.Restriction.Restriction.Jma19592I.size
Bio.Restriction.Restriction.KasI.size
Bio.Restriction.Restriction.KflI.size
Bio.Restriction.Restriction.Kpn2I.size
Bio.Restriction.Restriction.KpnI.size
Bio.Restriction.Restriction.KroI.size
Bio.Restriction.Restriction.Ksp22I.size
Bio.Restriction.Restriction.Ksp632I.size
Bio.Restriction.Restriction.KspAI.size
Bio.Restriction.Restriction.KspI.size
Bio.Restriction.Restriction.Kzo9I.size
Bio.Restriction.Restriction.LguI.size
Bio.Restriction.Restriction.LpnI.size
Bio.Restriction.Restriction.LpnPI.size
Bio.Restriction.Restriction.Lsp1109I.size
Bio.Restriction.Restriction.LweI.size
Bio.Restriction.Restriction.MabI.size
Bio.Restriction.Restriction.MaeI.size
Bio.Restriction.Restriction.MaeII.size
Bio.Restriction.Restriction.MaeIII.size
Bio.Restriction.Restriction.MalI.size
Bio.Restriction.Restriction.MaqI.size
Bio.Restriction.Restriction.MauBI.size
Bio.Restriction.Restriction.MbiI.size
Bio.Restriction.Restriction.MboI.size
Bio.Restriction.Restriction.MboII.size
Bio.Restriction.Restriction.McaTI.size
Bio.Restriction.Restriction.McrI.size
Bio.Restriction.Restriction.MfeI.size
Bio.Restriction.Restriction.MflI.size
Bio.Restriction.Restriction.MhlI.size
Bio.Restriction.Restriction.MjaIV.size
Bio.Restriction.Restriction.MkaDII.size
Bio.Restriction.Restriction.MlsI.size
Bio.Restriction.Restriction.MluCI.size
Bio.Restriction.Restriction.MluI.size
Bio.Restriction.Restriction.MluNI.size
Bio.Restriction.Restriction.Mly113I.size
Bio.Restriction.Restriction.MlyI.size
Bio.Restriction.Restriction.MmeI.size
Bio.Restriction.Restriction.MnlI.size
Bio.Restriction.Restriction.Mph1103I.size
Bio.Restriction.Restriction.MreI.size
Bio.Restriction.Restriction.MroI.size
Bio.Restriction.Restriction.MroNI.size
Bio.Restriction.Restriction.MroXI.size
Bio.Restriction.Restriction.MscI.size
Bio.Restriction.Restriction.MseI.size
Bio.Restriction.Restriction.MslI.size
Bio.Restriction.Restriction.Msp20I.size
Bio.Restriction.Restriction.MspA1I.size
Bio.Restriction.Restriction.MspCI.size
Bio.Restriction.Restriction.MspI.size
Bio.Restriction.Restriction.MspJI.size
Bio.Restriction.Restriction.MspR9I.size
Bio.Restriction.Restriction.MssI.size
Bio.Restriction.Restriction.MstI.size
Bio.Restriction.Restriction.MunI.size
Bio.Restriction.Restriction.Mva1269I.size
Bio.Restriction.Restriction.MvaI.size
Bio.Restriction.Restriction.MvnI.size
Bio.Restriction.Restriction.MvrI.size
Bio.Restriction.Restriction.MwoI.size
Bio.Restriction.Restriction.NaeI.size
Bio.Restriction.Restriction.NarI.size
Bio.Restriction.Restriction.NciI.size
Bio.Restriction.Restriction.NcoI.size
Bio.Restriction.Restriction.NdeI.size
Bio.Restriction.Restriction.NdeII.size
Bio.Restriction.Restriction.NgoAVIII.size
Bio.Restriction.Restriction.NgoMIV.size
Bio.Restriction.Restriction.NhaXI.size
Bio.Restriction.Restriction.NheI.size
Bio.Restriction.Restriction.NlaCI.size
Bio.Restriction.Restriction.NlaIII.size
Bio.Restriction.Restriction.NlaIV.size
Bio.Restriction.Restriction.Nli3877I.size
Bio.Restriction.Restriction.NmeAIII.size
Bio.Restriction.Restriction.NmeDI.size
Bio.Restriction.Restriction.NmuCI.size
Bio.Restriction.Restriction.NotI.size
Bio.Restriction.Restriction.NruI.size
Bio.Restriction.Restriction.NsbI.size
Bio.Restriction.Restriction.NsiI.size
Bio.Restriction.Restriction.NspBII.size
Bio.Restriction.Restriction.NspI.size
Bio.Restriction.Restriction.NspV.size
Bio.Restriction.Restriction.OliI.size
Bio.Restriction.Restriction.PabI.size
Bio.Restriction.Restriction.PacI.size
Bio.Restriction.Restriction.PaeI.size
Bio.Restriction.Restriction.PaeR7I.size
Bio.Restriction.Restriction.PagI.size
Bio.Restriction.Restriction.PalAI.size
Bio.Restriction.Restriction.PasI.size
Bio.Restriction.Restriction.PauI.size
Bio.Restriction.Restriction.PceI.size
Bio.Restriction.Restriction.PciI.size
Bio.Restriction.Restriction.PciSI.size
Bio.Restriction.Restriction.PcsI.size
Bio.Restriction.Restriction.PctI.size
Bio.Restriction.Restriction.PdiI.size
Bio.Restriction.Restriction.PdmI.size
Bio.Restriction.Restriction.PenI.size
Bio.Restriction.Restriction.PfeI.size
Bio.Restriction.Restriction.Pfl1108I.size
Bio.Restriction.Restriction.Pfl23II.size
Bio.Restriction.Restriction.PflFI.size
Bio.Restriction.Restriction.PflMI.size
Bio.Restriction.Restriction.PfoI.size
Bio.Restriction.Restriction.PinAI.size
Bio.Restriction.Restriction.PlaDI.size
Bio.Restriction.Restriction.Ple19I.size
Bio.Restriction.Restriction.PleI.size
Bio.Restriction.Restriction.PluTI.size
Bio.Restriction.Restriction.PmaCI.size
Bio.Restriction.Restriction.PmeI.size
Bio.Restriction.Restriction.PmlI.size
Bio.Restriction.Restriction.PpiI.size
Bio.Restriction.Restriction.PpsI.size
Bio.Restriction.Restriction.Ppu10I.size
Bio.Restriction.Restriction.Ppu21I.size
Bio.Restriction.Restriction.PpuMI.size
Bio.Restriction.Restriction.PscI.size
Bio.Restriction.Restriction.PshAI.size
Bio.Restriction.Restriction.PshBI.size
Bio.Restriction.Restriction.PsiI.size
Bio.Restriction.Restriction.Psp03I.size
Bio.Restriction.Restriction.Psp124BI.size
Bio.Restriction.Restriction.Psp1406I.size
Bio.Restriction.Restriction.Psp5II.size
Bio.Restriction.Restriction.Psp6I.size
Bio.Restriction.Restriction.PspCI.size
Bio.Restriction.Restriction.PspEI.size
Bio.Restriction.Restriction.PspGI.size
Bio.Restriction.Restriction.PspLI.size
Bio.Restriction.Restriction.PspN4I.size
Bio.Restriction.Restriction.PspOMI.size
Bio.Restriction.Restriction.PspOMII.size
Bio.Restriction.Restriction.PspPI.size
Bio.Restriction.Restriction.PspPPI.size
Bio.Restriction.Restriction.PspPRI.size
Bio.Restriction.Restriction.PspXI.size
Bio.Restriction.Restriction.PsrI.size
Bio.Restriction.Restriction.PssI.size
Bio.Restriction.Restriction.PstI.size
Bio.Restriction.Restriction.PstNI.size
Bio.Restriction.Restriction.PsuI.size
Bio.Restriction.Restriction.PsyI.size
Bio.Restriction.Restriction.PteI.size
Bio.Restriction.Restriction.PvuI.size
Bio.Restriction.Restriction.PvuII.size
Bio.Restriction.Restriction.R2_BceSIV.size
Bio.Restriction.Restriction.RceI.size
Bio.Restriction.Restriction.RdeGBI.size
Bio.Restriction.Restriction.RdeGBII.size
Bio.Restriction.Restriction.RdeGBIII.size
Bio.Restriction.Restriction.RflFIII.size
Bio.Restriction.Restriction.RgaI.size
Bio.Restriction.Restriction.RigI.size
Bio.Restriction.Restriction.RlaI.size
Bio.Restriction.Restriction.RleAI.size
Bio.Restriction.Restriction.RpaB5I.size
Bio.Restriction.Restriction.RpaBI.size
Bio.Restriction.Restriction.RpaI.size
Bio.Restriction.Restriction.RpaTI.size
Bio.Restriction.Restriction.RruI.size
Bio.Restriction.Restriction.RsaI.size
Bio.Restriction.Restriction.RsaNI.size
Bio.Restriction.Restriction.RseI.size
Bio.Restriction.Restriction.Rsr2I.size
Bio.Restriction.Restriction.RsrII.size
Bio.Restriction.Restriction.SacI.size
Bio.Restriction.Restriction.SacII.size
Bio.Restriction.Restriction.SalI.size
Bio.Restriction.Restriction.SanDI.size
Bio.Restriction.Restriction.SapI.size
Bio.Restriction.Restriction.SaqAI.size
Bio.Restriction.Restriction.SatI.size
Bio.Restriction.Restriction.Sau3AI.size
Bio.Restriction.Restriction.Sau96I.size
Bio.Restriction.Restriction.SauI.size
Bio.Restriction.Restriction.SbfI.size
Bio.Restriction.Restriction.ScaI.size
Bio.Restriction.Restriction.SchI.size
Bio.Restriction.Restriction.SciI.size
Bio.Restriction.Restriction.ScrFI.size
Bio.Restriction.Restriction.SdaI.size
Bio.Restriction.Restriction.SdeAI.size
Bio.Restriction.Restriction.SdeOSI.size
Bio.Restriction.Restriction.SduI.size
Bio.Restriction.Restriction.SecI.size
Bio.Restriction.Restriction.SelI.size
Bio.Restriction.Restriction.SetI.size
Bio.Restriction.Restriction.SexAI.size
Bio.Restriction.Restriction.SfaAI.size
Bio.Restriction.Restriction.SfaNI.size
Bio.Restriction.Restriction.SfcI.size
Bio.Restriction.Restriction.SfeI.size
Bio.Restriction.Restriction.SfiI.size
Bio.Restriction.Restriction.SfoI.size
Bio.Restriction.Restriction.Sfr274I.size
Bio.Restriction.Restriction.Sfr303I.size
Bio.Restriction.Restriction.SfuI.size
Bio.Restriction.Restriction.SgeI.size
Bio.Restriction.Restriction.SgfI.size
Bio.Restriction.Restriction.SgrAI.size
Bio.Restriction.Restriction.SgrBI.size
Bio.Restriction.Restriction.SgrDI.size
Bio.Restriction.Restriction.SgrTI.size
Bio.Restriction.Restriction.SgsI.size
Bio.Restriction.Restriction.SimI.size
Bio.Restriction.Restriction.SlaI.size
Bio.Restriction.Restriction.SmaI.size
Bio.Restriction.Restriction.SmiI.size
Bio.Restriction.Restriction.SmiMI.size
Bio.Restriction.Restriction.SmlI.size
Bio.Restriction.Restriction.SmoI.size
Bio.Restriction.Restriction.SnaBI.size
Bio.Restriction.Restriction.SnaI.size
Bio.Restriction.Restriction.Sno506I.size
Bio.Restriction.Restriction.SpeI.size
Bio.Restriction.Restriction.SphI.size
Bio.Restriction.Restriction.SplI.size
Bio.Restriction.Restriction.SpoDI.size
Bio.Restriction.Restriction.SrfI.size
Bio.Restriction.Restriction.Sse232I.size
Bio.Restriction.Restriction.Sse8387I.size
Bio.Restriction.Restriction.Sse8647I.size
Bio.Restriction.Restriction.Sse9I.size
Bio.Restriction.Restriction.SseBI.size
Bio.Restriction.Restriction.SsiI.size
Bio.Restriction.Restriction.SspD5I.size
Bio.Restriction.Restriction.SspDI.size
Bio.Restriction.Restriction.SspI.size
Bio.Restriction.Restriction.SstE37I.size
Bio.Restriction.Restriction.SstI.size
Bio.Restriction.Restriction.Sth132I.size
Bio.Restriction.Restriction.Sth302II.size
Bio.Restriction.Restriction.StrI.size
Bio.Restriction.Restriction.StsI.size
Bio.Restriction.Restriction.StuI.size
Bio.Restriction.Restriction.StyD4I.size
Bio.Restriction.Restriction.StyI.size
Bio.Restriction.Restriction.SwaI.size
Bio.Restriction.Restriction.TaaI.size
Bio.Restriction.Restriction.TaiI.size
Bio.Restriction.Restriction.TaqI.size
Bio.Restriction.Restriction.TaqII.size
Bio.Restriction.Restriction.TasI.size
Bio.Restriction.Restriction.TatI.size
Bio.Restriction.Restriction.TauI.size
Bio.Restriction.Restriction.TfiI.size
Bio.Restriction.Restriction.Tru1I.size
Bio.Restriction.Restriction.Tru9I.size
Bio.Restriction.Restriction.TscAI.size
Bio.Restriction.Restriction.TseFI.size
Bio.Restriction.Restriction.TseI.size
Bio.Restriction.Restriction.TsoI.size
Bio.Restriction.Restriction.Tsp45I.size
Bio.Restriction.Restriction.Tsp4CI.size
Bio.Restriction.Restriction.TspDTI.size
Bio.Restriction.Restriction.TspEI.size
Bio.Restriction.Restriction.TspGWI.size
Bio.Restriction.Restriction.TspMI.size
Bio.Restriction.Restriction.TspRI.size
Bio.Restriction.Restriction.TssI.size
Bio.Restriction.Restriction.TstI.size
Bio.Restriction.Restriction.TsuI.size
Bio.Restriction.Restriction.Tth111I.size
Bio.Restriction.Restriction.Tth111II.size
Bio.Restriction.Restriction.UbaF11I.size
Bio.Restriction.Restriction.UbaF12I.size
Bio.Restriction.Restriction.UbaF13I.size
Bio.Restriction.Restriction.UbaF14I.size
Bio.Restriction.Restriction.UbaF9I.size
Bio.Restriction.Restriction.UbaPI.size
Bio.Restriction.Restriction.UcoMSI.size
Bio.Restriction.Restriction.UnbI.size
Bio.Restriction.Restriction.Van91I.size
Bio.Restriction.Restriction.Vha464I.size
Bio.Restriction.Restriction.VneI.size
Bio.Restriction.Restriction.VpaK11AI.size
Bio.Restriction.Restriction.VpaK11BI.size
Bio.Restriction.Restriction.VspI.size
Bio.Restriction.Restriction.WviI.size
Bio.Restriction.Restriction.XagI.size
Bio.Restriction.Restriction.XapI.size
Bio.Restriction.Restriction.XbaI.size
Bio.Restriction.Restriction.XceI.size
Bio.Restriction.Restriction.XcmI.size
Bio.Restriction.Restriction.XhoI.size
Bio.Restriction.Restriction.XhoII.size
Bio.Restriction.Restriction.XmaI.size
Bio.Restriction.Restriction.XmaIII.size
Bio.Restriction.Restriction.XmaJI.size
Bio.Restriction.Restriction.XmiI.size
Bio.Restriction.Restriction.XmnI.size
Bio.Restriction.Restriction.XspI.size
Bio.Restriction.Restriction.YkrI.size
Bio.Restriction.Restriction.ZraI.size
Bio.Restriction.Restriction.ZrmI.size
Bio.Restriction.Restriction.Zsp2I.size
Bio.codonalign.codonalphabet.CodonAlphabet.size" class="py-name" href="#" onclick="return doclink('link-483', 'size', 'link-270');">size</a></tt><tt class="py-op">(</tt><tt class="py-name">splitline</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1221"></a><tt class="py-lineno">1221</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-484" class="py-name"><a title="Bio.GenBank._FeatureConsumer.residue_type
Bio.GenBank._RecordConsumer.residue_type" class="py-name" href="#" onclick="return doclink('link-484', 'residue_type', 'link-236');">residue_type</a></tt><tt class="py-op">(</tt><tt class="py-name">splitline</tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1222"></a><tt class="py-lineno">1222</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-485" class="py-name"><a title="Bio.GenBank._FeatureConsumer.data_file_division
Bio.GenBank._RecordConsumer.data_file_division" class="py-name" href="#" onclick="return doclink('link-485', 'data_file_division', 'link-237');">data_file_division</a></tt><tt class="py-op">(</tt><tt class="py-name">splitline</tt><tt class="py-op">[</tt><tt class="py-number">5</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1223"></a><tt class="py-lineno">1223</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-486" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-486', 'date', 'link-439');">date</a></tt><tt class="py-op">(</tt><tt class="py-name">splitline</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1224"></a><tt class="py-lineno">1224</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-487" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-487', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">4</tt> <tt class="py-keyword">and</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-488" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-488', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"aa"</tt><tt class="py-op">,</tt> <tt class="py-string">"bp"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L1225"></a><tt class="py-lineno">1225</tt>  <tt class="py-line">            <tt class="py-comment"># Cope with EMBOSS seqret output where it seems the locus id can cause</tt> </tt>
<a name="L1226"></a><tt class="py-lineno">1226</tt>  <tt class="py-line">            <tt class="py-comment"># the other fields to overflow.  We just IGNORE the other fields!</tt> </tt>
<a name="L1227"></a><tt class="py-lineno">1227</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Malformed LOCUS line found - is this "</tt> </tt>
<a name="L1228"></a><tt class="py-lineno">1228</tt>  <tt class="py-line">                          <tt class="py-string">"correct?\n:%r"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">,</tt> <tt id="link-489" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-489', 'BiopythonParserWarning', 'link-10');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1229"></a><tt class="py-lineno">1229</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-490" class="py-name"><a title="Bio.GenBank._FeatureConsumer.locus
Bio.GenBank._RecordConsumer.locus" class="py-name" href="#" onclick="return doclink('link-490', 'locus', 'link-235');">locus</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-491" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-491', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1230"></a><tt class="py-lineno">1230</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-492" class="py-name"><a title="Bio.Alphabet.Alphabet.size
Bio.Alphabet.Reduced.HPModel.size
Bio.Alphabet.Reduced.Murphy10.size
Bio.Alphabet.Reduced.Murphy15.size
Bio.Alphabet.Reduced.Murphy4.size
Bio.Alphabet.Reduced.Murphy8.size
Bio.Alphabet.Reduced.PC5.size
Bio.Alphabet.SingleLetterAlphabet.size
Bio.Alphabet.ThreeLetterProtein.size
Bio.GenBank._FeatureConsumer.size
Bio.GenBank._RecordConsumer.size
Bio.Restriction.Restriction.AanI.size
Bio.Restriction.Restriction.AarI.size
Bio.Restriction.Restriction.AasI.size
Bio.Restriction.Restriction.AatII.size
Bio.Restriction.Restriction.AbaSI.size
Bio.Restriction.Restriction.AbsI.size
Bio.Restriction.Restriction.Acc16I.size
Bio.Restriction.Restriction.Acc36I.size
Bio.Restriction.Restriction.Acc65I.size
Bio.Restriction.Restriction.AccB1I.size
Bio.Restriction.Restriction.AccB7I.size
Bio.Restriction.Restriction.AccBSI.size
Bio.Restriction.Restriction.AccI.size
Bio.Restriction.Restriction.AccII.size
Bio.Restriction.Restriction.AccIII.size
Bio.Restriction.Restriction.AceIII.size
Bio.Restriction.Restriction.AciI.size
Bio.Restriction.Restriction.AclI.size
Bio.Restriction.Restriction.AclWI.size
Bio.Restriction.Restriction.AcoI.size
Bio.Restriction.Restriction.AcsI.size
Bio.Restriction.Restriction.AcuI.size
Bio.Restriction.Restriction.AcvI.size
Bio.Restriction.Restriction.AcyI.size
Bio.Restriction.Restriction.AdeI.size
Bio.Restriction.Restriction.AfaI.size
Bio.Restriction.Restriction.AfeI.size
Bio.Restriction.Restriction.AfiI.size
Bio.Restriction.Restriction.AflII.size
Bio.Restriction.Restriction.AflIII.size
Bio.Restriction.Restriction.AgeI.size
Bio.Restriction.Restriction.AgsI.size
Bio.Restriction.Restriction.AhaIII.size
Bio.Restriction.Restriction.AhdI.size
Bio.Restriction.Restriction.AhlI.size
Bio.Restriction.Restriction.AjiI.size
Bio.Restriction.Restriction.AjnI.size
Bio.Restriction.Restriction.AjuI.size
Bio.Restriction.Restriction.AleI.size
Bio.Restriction.Restriction.AlfI.size
Bio.Restriction.Restriction.AloI.size
Bio.Restriction.Restriction.AluBI.size
Bio.Restriction.Restriction.AluI.size
Bio.Restriction.Restriction.Alw21I.size
Bio.Restriction.Restriction.Alw26I.size
Bio.Restriction.Restriction.Alw44I.size
Bio.Restriction.Restriction.AlwFI.size
Bio.Restriction.Restriction.AlwI.size
Bio.Restriction.Restriction.AlwNI.size
Bio.Restriction.Restriction.Ama87I.size
Bio.Restriction.Restriction.Aor13HI.size
Bio.Restriction.Restriction.Aor51HI.size
Bio.Restriction.Restriction.AoxI.size
Bio.Restriction.Restriction.ApaBI.size
Bio.Restriction.Restriction.ApaI.size
Bio.Restriction.Restriction.ApaLI.size
Bio.Restriction.Restriction.ApeKI.size
Bio.Restriction.Restriction.ApoI.size
Bio.Restriction.Restriction.ApyPI.size
Bio.Restriction.Restriction.AquII.size
Bio.Restriction.Restriction.AquIII.size
Bio.Restriction.Restriction.AquIV.size
Bio.Restriction.Restriction.ArsI.size
Bio.Restriction.Restriction.AscI.size
Bio.Restriction.Restriction.AseI.size
Bio.Restriction.Restriction.Asi256I.size
Bio.Restriction.Restriction.AsiGI.size
Bio.Restriction.Restriction.AsiSI.size
Bio.Restriction.Restriction.Asp700I.size
Bio.Restriction.Restriction.Asp718I.size
Bio.Restriction.Restriction.AspA2I.size
Bio.Restriction.Restriction.AspBHI.size
Bio.Restriction.Restriction.AspLEI.size
Bio.Restriction.Restriction.AspS9I.size
Bio.Restriction.Restriction.AssI.size
Bio.Restriction.Restriction.AsuC2I.size
Bio.Restriction.Restriction.AsuHPI.size
Bio.Restriction.Restriction.AsuI.size
Bio.Restriction.Restriction.AsuII.size
Bio.Restriction.Restriction.AsuNHI.size
Bio.Restriction.Restriction.AvaI.size
Bio.Restriction.Restriction.AvaII.size
Bio.Restriction.Restriction.AvaIII.size
Bio.Restriction.Restriction.AvrII.size
Bio.Restriction.Restriction.AxyI.size
Bio.Restriction.Restriction.BaeGI.size
Bio.Restriction.Restriction.BaeI.size
Bio.Restriction.Restriction.BalI.size
Bio.Restriction.Restriction.BamHI.size
Bio.Restriction.Restriction.BanI.size
Bio.Restriction.Restriction.BanII.size
Bio.Restriction.Restriction.BarI.size
Bio.Restriction.Restriction.BasI.size
Bio.Restriction.Restriction.BauI.size
Bio.Restriction.Restriction.Bbr7I.size
Bio.Restriction.Restriction.BbrPI.size
Bio.Restriction.Restriction.BbsI.size
Bio.Restriction.Restriction.Bbv12I.size
Bio.Restriction.Restriction.BbvCI.size
Bio.Restriction.Restriction.BbvI.size
Bio.Restriction.Restriction.BbvII.size
Bio.Restriction.Restriction.BccI.size
Bio.Restriction.Restriction.Bce83I.size
Bio.Restriction.Restriction.BceAI.size
Bio.Restriction.Restriction.BcefI.size
Bio.Restriction.Restriction.BcgI.size
Bio.Restriction.Restriction.BciT130I.size
Bio.Restriction.Restriction.BciVI.size
Bio.Restriction.Restriction.BclI.size
Bio.Restriction.Restriction.BcnI.size
Bio.Restriction.Restriction.BcoDI.size
Bio.Restriction.Restriction.BcuI.size
Bio.Restriction.Restriction.BdaI.size
Bio.Restriction.Restriction.BetI.size
Bio.Restriction.Restriction.BfaI.size
Bio.Restriction.Restriction.BfiI.size
Bio.Restriction.Restriction.BfmI.size
Bio.Restriction.Restriction.BfoI.size
Bio.Restriction.Restriction.BfrI.size
Bio.Restriction.Restriction.BfuAI.size
Bio.Restriction.Restriction.BfuCI.size
Bio.Restriction.Restriction.BfuI.size
Bio.Restriction.Restriction.BglI.size
Bio.Restriction.Restriction.BglII.size
Bio.Restriction.Restriction.BinI.size
Bio.Restriction.Restriction.BisI.size
Bio.Restriction.Restriction.BlnI.size
Bio.Restriction.Restriction.BlpI.size
Bio.Restriction.Restriction.BlsI.size
Bio.Restriction.Restriction.BmcAI.size
Bio.Restriction.Restriction.Bme1390I.size
Bio.Restriction.Restriction.Bme18I.size
Bio.Restriction.Restriction.BmeDI.size
Bio.Restriction.Restriction.BmeRI.size
Bio.Restriction.Restriction.BmeT110I.size
Bio.Restriction.Restriction.BmgBI.size
Bio.Restriction.Restriction.BmgI.size
Bio.Restriction.Restriction.BmgT120I.size
Bio.Restriction.Restriction.BmiI.size
Bio.Restriction.Restriction.BmrFI.size
Bio.Restriction.Restriction.BmrI.size
Bio.Restriction.Restriction.BmsI.size
Bio.Restriction.Restriction.BmtI.size
Bio.Restriction.Restriction.BmuI.size
Bio.Restriction.Restriction.BoxI.size
Bio.Restriction.Restriction.BpiI.size
Bio.Restriction.Restriction.BplI.size
Bio.Restriction.Restriction.BpmI.size
Bio.Restriction.Restriction.Bpu10I.size
Bio.Restriction.Restriction.Bpu1102I.size
Bio.Restriction.Restriction.Bpu14I.size
Bio.Restriction.Restriction.BpuEI.size
Bio.Restriction.Restriction.BpuMI.size
Bio.Restriction.Restriction.BpvUI.size
Bio.Restriction.Restriction.Bsa29I.size
Bio.Restriction.Restriction.BsaAI.size
Bio.Restriction.Restriction.BsaBI.size
Bio.Restriction.Restriction.BsaHI.size
Bio.Restriction.Restriction.BsaI.size
Bio.Restriction.Restriction.BsaJI.size
Bio.Restriction.Restriction.BsaWI.size
Bio.Restriction.Restriction.BsaXI.size
Bio.Restriction.Restriction.BsbI.size
Bio.Restriction.Restriction.Bsc4I.size
Bio.Restriction.Restriction.BscAI.size
Bio.Restriction.Restriction.BscGI.size
Bio.Restriction.Restriction.Bse118I.size
Bio.Restriction.Restriction.Bse1I.size
Bio.Restriction.Restriction.Bse21I.size
Bio.Restriction.Restriction.Bse3DI.size
Bio.Restriction.Restriction.Bse8I.size
Bio.Restriction.Restriction.BseAI.size
Bio.Restriction.Restriction.BseBI.size
Bio.Restriction.Restriction.BseCI.size
Bio.Restriction.Restriction.BseDI.size
Bio.Restriction.Restriction.BseGI.size
Bio.Restriction.Restriction.BseJI.size
Bio.Restriction.Restriction.BseLI.size
Bio.Restriction.Restriction.BseMI.size
Bio.Restriction.Restriction.BseMII.size
Bio.Restriction.Restriction.BseNI.size
Bio.Restriction.Restriction.BsePI.size
Bio.Restriction.Restriction.BseRI.size
Bio.Restriction.Restriction.BseSI.size
Bio.Restriction.Restriction.BseX3I.size
Bio.Restriction.Restriction.BseXI.size
Bio.Restriction.Restriction.BseYI.size
Bio.Restriction.Restriction.BsgI.size
Bio.Restriction.Restriction.Bsh1236I.size
Bio.Restriction.Restriction.Bsh1285I.size
Bio.Restriction.Restriction.BshFI.size
Bio.Restriction.Restriction.BshNI.size
Bio.Restriction.Restriction.BshTI.size
Bio.Restriction.Restriction.BshVI.size
Bio.Restriction.Restriction.BsiEI.size
Bio.Restriction.Restriction.BsiHKAI.size
Bio.Restriction.Restriction.BsiHKCI.size
Bio.Restriction.Restriction.BsiI.size
Bio.Restriction.Restriction.BsiSI.size
Bio.Restriction.Restriction.BsiWI.size
Bio.Restriction.Restriction.BsiYI.size
Bio.Restriction.Restriction.BslFI.size
Bio.Restriction.Restriction.BslI.size
Bio.Restriction.Restriction.BsmAI.size
Bio.Restriction.Restriction.BsmBI.size
Bio.Restriction.Restriction.BsmFI.size
Bio.Restriction.Restriction.BsmI.size
Bio.Restriction.Restriction.BsnI.size
Bio.Restriction.Restriction.Bso31I.size
Bio.Restriction.Restriction.BsoBI.size
Bio.Restriction.Restriction.Bsp119I.size
Bio.Restriction.Restriction.Bsp120I.size
Bio.Restriction.Restriction.Bsp1286I.size
Bio.Restriction.Restriction.Bsp13I.size
Bio.Restriction.Restriction.Bsp1407I.size
Bio.Restriction.Restriction.Bsp143I.size
Bio.Restriction.Restriction.Bsp1720I.size
Bio.Restriction.Restriction.Bsp19I.size
Bio.Restriction.Restriction.Bsp24I.size
Bio.Restriction.Restriction.Bsp68I.size
Bio.Restriction.Restriction.BspACI.size
Bio.Restriction.Restriction.BspCNI.size
Bio.Restriction.Restriction.BspD6I.size
Bio.Restriction.Restriction.BspDI.size
Bio.Restriction.Restriction.BspEI.size
Bio.Restriction.Restriction.BspFNI.size
Bio.Restriction.Restriction.BspGI.size
Bio.Restriction.Restriction.BspHI.size
Bio.Restriction.Restriction.BspLI.size
Bio.Restriction.Restriction.BspLU11I.size
Bio.Restriction.Restriction.BspMI.size
Bio.Restriction.Restriction.BspMII.size
Bio.Restriction.Restriction.BspNCI.size
Bio.Restriction.Restriction.BspOI.size
Bio.Restriction.Restriction.BspPI.size
Bio.Restriction.Restriction.BspQI.size
Bio.Restriction.Restriction.BspT104I.size
Bio.Restriction.Restriction.BspT107I.size
Bio.Restriction.Restriction.BspTI.size
Bio.Restriction.Restriction.BsrBI.size
Bio.Restriction.Restriction.BsrDI.size
Bio.Restriction.Restriction.BsrFI.size
Bio.Restriction.Restriction.BsrGI.size
Bio.Restriction.Restriction.BsrI.size
Bio.Restriction.Restriction.BsrSI.size
Bio.Restriction.Restriction.BssAI.size
Bio.Restriction.Restriction.BssECI.size
Bio.Restriction.Restriction.BssHII.size
Bio.Restriction.Restriction.BssKI.size
Bio.Restriction.Restriction.BssMI.size
Bio.Restriction.Restriction.BssNAI.size
Bio.Restriction.Restriction.BssNI.size
Bio.Restriction.Restriction.BssSI.size
Bio.Restriction.Restriction.BssT1I.size
Bio.Restriction.Restriction.Bst1107I.size
Bio.Restriction.Restriction.Bst2BI.size
Bio.Restriction.Restriction.Bst2UI.size
Bio.Restriction.Restriction.Bst4CI.size
Bio.Restriction.Restriction.Bst6I.size
Bio.Restriction.Restriction.BstACI.size
Bio.Restriction.Restriction.BstAFI.size
Bio.Restriction.Restriction.BstAPI.size
Bio.Restriction.Restriction.BstAUI.size
Bio.Restriction.Restriction.BstBAI.size
Bio.Restriction.Restriction.BstBI.size
Bio.Restriction.Restriction.BstC8I.size
Bio.Restriction.Restriction.BstDEI.size
Bio.Restriction.Restriction.BstDSI.size
Bio.Restriction.Restriction.BstEII.size
Bio.Restriction.Restriction.BstENI.size
Bio.Restriction.Restriction.BstF5I.size
Bio.Restriction.Restriction.BstFNI.size
Bio.Restriction.Restriction.BstH2I.size
Bio.Restriction.Restriction.BstHHI.size
Bio.Restriction.Restriction.BstKTI.size
Bio.Restriction.Restriction.BstMAI.size
Bio.Restriction.Restriction.BstMBI.size
Bio.Restriction.Restriction.BstMCI.size
Bio.Restriction.Restriction.BstMWI.size
Bio.Restriction.Restriction.BstNI.size
Bio.Restriction.Restriction.BstNSI.size
Bio.Restriction.Restriction.BstOI.size
Bio.Restriction.Restriction.BstPAI.size
Bio.Restriction.Restriction.BstPI.size
Bio.Restriction.Restriction.BstSCI.size
Bio.Restriction.Restriction.BstSFI.size
Bio.Restriction.Restriction.BstSLI.size
Bio.Restriction.Restriction.BstSNI.size
Bio.Restriction.Restriction.BstUI.size
Bio.Restriction.Restriction.BstV1I.size
Bio.Restriction.Restriction.BstV2I.size
Bio.Restriction.Restriction.BstX2I.size
Bio.Restriction.Restriction.BstXI.size
Bio.Restriction.Restriction.BstYI.size
Bio.Restriction.Restriction.BstZ17I.size
Bio.Restriction.Restriction.BstZI.size
Bio.Restriction.Restriction.Bsu15I.size
Bio.Restriction.Restriction.Bsu36I.size
Bio.Restriction.Restriction.BsuI.size
Bio.Restriction.Restriction.BsuRI.size
Bio.Restriction.Restriction.BtgI.size
Bio.Restriction.Restriction.BtgZI.size
Bio.Restriction.Restriction.BthCI.size
Bio.Restriction.Restriction.BtrI.size
Bio.Restriction.Restriction.BtsCI.size
Bio.Restriction.Restriction.BtsI.size
Bio.Restriction.Restriction.BtsIMutI.size
Bio.Restriction.Restriction.BtuMI.size
Bio.Restriction.Restriction.BveI.size
Bio.Restriction.Restriction.Cac8I.size
Bio.Restriction.Restriction.CaiI.size
Bio.Restriction.Restriction.CauII.size
Bio.Restriction.Restriction.CchII.size
Bio.Restriction.Restriction.CchIII.size
Bio.Restriction.Restriction.CciI.size
Bio.Restriction.Restriction.CciNI.size
Bio.Restriction.Restriction.Cdi630V.size
Bio.Restriction.Restriction.CdiI.size
Bio.Restriction.Restriction.CdpI.size
Bio.Restriction.Restriction.CfoI.size
Bio.Restriction.Restriction.Cfr10I.size
Bio.Restriction.Restriction.Cfr13I.size
Bio.Restriction.Restriction.Cfr42I.size
Bio.Restriction.Restriction.Cfr9I.size
Bio.Restriction.Restriction.CfrI.size
Bio.Restriction.Restriction.Cgl13032I.size
Bio.Restriction.Restriction.Cgl13032II.size
Bio.Restriction.Restriction.ChaI.size
Bio.Restriction.Restriction.CjeFIII.size
Bio.Restriction.Restriction.CjeFV.size
Bio.Restriction.Restriction.CjeI.size
Bio.Restriction.Restriction.CjeNII.size
Bio.Restriction.Restriction.CjeNIII.size
Bio.Restriction.Restriction.CjeP659IV.size
Bio.Restriction.Restriction.CjePI.size
Bio.Restriction.Restriction.CjuI.size
Bio.Restriction.Restriction.CjuII.size
Bio.Restriction.Restriction.ClaI.size
Bio.Restriction.Restriction.CpoI.size
Bio.Restriction.Restriction.CseI.size
Bio.Restriction.Restriction.CsiI.size
Bio.Restriction.Restriction.Csp6I.size
Bio.Restriction.Restriction.CspAI.size
Bio.Restriction.Restriction.CspCI.size
Bio.Restriction.Restriction.CspI.size
Bio.Restriction.Restriction.CstMI.size
Bio.Restriction.Restriction.CviAII.size
Bio.Restriction.Restriction.CviJI.size
Bio.Restriction.Restriction.CviKI_1.size
Bio.Restriction.Restriction.CviQI.size
Bio.Restriction.Restriction.CviRI.size
Bio.Restriction.Restriction.DdeI.size
Bio.Restriction.Restriction.DinI.size
Bio.Restriction.Restriction.DpnI.size
Bio.Restriction.Restriction.DpnII.size
Bio.Restriction.Restriction.DraI.size
Bio.Restriction.Restriction.DraII.size
Bio.Restriction.Restriction.DraIII.size
Bio.Restriction.Restriction.DraRI.size
Bio.Restriction.Restriction.DrdI.size
Bio.Restriction.Restriction.DrdII.size
Bio.Restriction.Restriction.DriI.size
Bio.Restriction.Restriction.DsaI.size
Bio.Restriction.Restriction.DseDI.size
Bio.Restriction.Restriction.EaeI.size
Bio.Restriction.Restriction.EagI.size
Bio.Restriction.Restriction.Eam1104I.size
Bio.Restriction.Restriction.Eam1105I.size
Bio.Restriction.Restriction.EarI.size
Bio.Restriction.Restriction.EciI.size
Bio.Restriction.Restriction.Ecl136II.size
Bio.Restriction.Restriction.EclXI.size
Bio.Restriction.Restriction.Eco105I.size
Bio.Restriction.Restriction.Eco130I.size
Bio.Restriction.Restriction.Eco147I.size
Bio.Restriction.Restriction.Eco24I.size
Bio.Restriction.Restriction.Eco31I.size
Bio.Restriction.Restriction.Eco32I.size
Bio.Restriction.Restriction.Eco47I.size
Bio.Restriction.Restriction.Eco47III.size
Bio.Restriction.Restriction.Eco52I.size
Bio.Restriction.Restriction.Eco53kI.size
Bio.Restriction.Restriction.Eco57I.size
Bio.Restriction.Restriction.Eco57MI.size
Bio.Restriction.Restriction.Eco72I.size
Bio.Restriction.Restriction.Eco81I.size
Bio.Restriction.Restriction.Eco88I.size
Bio.Restriction.Restriction.Eco91I.size
Bio.Restriction.Restriction.EcoHI.size
Bio.Restriction.Restriction.EcoICRI.size
Bio.Restriction.Restriction.EcoNI.size
Bio.Restriction.Restriction.EcoO109I.size
Bio.Restriction.Restriction.EcoO65I.size
Bio.Restriction.Restriction.EcoRI.size
Bio.Restriction.Restriction.EcoRII.size
Bio.Restriction.Restriction.EcoRV.size
Bio.Restriction.Restriction.EcoT14I.size
Bio.Restriction.Restriction.EcoT22I.size
Bio.Restriction.Restriction.EcoT38I.size
Bio.Restriction.Restriction.EgeI.size
Bio.Restriction.Restriction.EheI.size
Bio.Restriction.Restriction.ErhI.size
Bio.Restriction.Restriction.EsaBC3I.size
Bio.Restriction.Restriction.EsaSSI.size
Bio.Restriction.Restriction.Esp3I.size
Bio.Restriction.Restriction.EspI.size
Bio.Restriction.Restriction.FaeI.size
Bio.Restriction.Restriction.FaiI.size
Bio.Restriction.Restriction.FalI.size
Bio.Restriction.Restriction.FaqI.size
Bio.Restriction.Restriction.FatI.size
Bio.Restriction.Restriction.FauI.size
Bio.Restriction.Restriction.FauNDI.size
Bio.Restriction.Restriction.FbaI.size
Bio.Restriction.Restriction.FblI.size
Bio.Restriction.Restriction.FinI.size
Bio.Restriction.Restriction.FmuI.size
Bio.Restriction.Restriction.Fnu4HI.size
Bio.Restriction.Restriction.FnuDII.size
Bio.Restriction.Restriction.FokI.size
Bio.Restriction.Restriction.FriOI.size
Bio.Restriction.Restriction.FseI.size
Bio.Restriction.Restriction.Fsp4HI.size
Bio.Restriction.Restriction.FspAI.size
Bio.Restriction.Restriction.FspBI.size
Bio.Restriction.Restriction.FspEI.size
Bio.Restriction.Restriction.FspI.size
Bio.Restriction.Restriction.GauT27I.size
Bio.Restriction.Restriction.GdiII.size
Bio.Restriction.Restriction.GlaI.size
Bio.Restriction.Restriction.GluI.size
Bio.Restriction.Restriction.GsaI.size
Bio.Restriction.Restriction.GsuI.size
Bio.Restriction.Restriction.HaeI.size
Bio.Restriction.Restriction.HaeII.size
Bio.Restriction.Restriction.HaeIII.size
Bio.Restriction.Restriction.HapII.size
Bio.Restriction.Restriction.HauII.size
Bio.Restriction.Restriction.HgaI.size
Bio.Restriction.Restriction.HgiAI.size
Bio.Restriction.Restriction.HgiCI.size
Bio.Restriction.Restriction.HgiEII.size
Bio.Restriction.Restriction.HgiJII.size
Bio.Restriction.Restriction.HhaI.size
Bio.Restriction.Restriction.Hin1I.size
Bio.Restriction.Restriction.Hin1II.size
Bio.Restriction.Restriction.Hin4I.size
Bio.Restriction.Restriction.Hin4II.size
Bio.Restriction.Restriction.Hin6I.size
Bio.Restriction.Restriction.HinP1I.size
Bio.Restriction.Restriction.HincII.size
Bio.Restriction.Restriction.HindII.size
Bio.Restriction.Restriction.HindIII.size
Bio.Restriction.Restriction.HinfI.size
Bio.Restriction.Restriction.HpaI.size
Bio.Restriction.Restriction.HpaII.size
Bio.Restriction.Restriction.HphI.size
Bio.Restriction.Restriction.Hpy166II.size
Bio.Restriction.Restriction.Hpy178III.size
Bio.Restriction.Restriction.Hpy188I.size
Bio.Restriction.Restriction.Hpy188III.size
Bio.Restriction.Restriction.Hpy8I.size
Bio.Restriction.Restriction.Hpy99I.size
Bio.Restriction.Restriction.Hpy99XIII.size
Bio.Restriction.Restriction.Hpy99XIV.size
Bio.Restriction.Restriction.HpyAV.size
Bio.Restriction.Restriction.HpyCH4III.size
Bio.Restriction.Restriction.HpyCH4IV.size
Bio.Restriction.Restriction.HpyCH4V.size
Bio.Restriction.Restriction.HpyF10VI.size
Bio.Restriction.Restriction.HpyF3I.size
Bio.Restriction.Restriction.HpySE526I.size
Bio.Restriction.Restriction.Hsp92I.size
Bio.Restriction.Restriction.Hsp92II.size
Bio.Restriction.Restriction.HspAI.size
Bio.Restriction.Restriction.Jma19592I.size
Bio.Restriction.Restriction.KasI.size
Bio.Restriction.Restriction.KflI.size
Bio.Restriction.Restriction.Kpn2I.size
Bio.Restriction.Restriction.KpnI.size
Bio.Restriction.Restriction.KroI.size
Bio.Restriction.Restriction.Ksp22I.size
Bio.Restriction.Restriction.Ksp632I.size
Bio.Restriction.Restriction.KspAI.size
Bio.Restriction.Restriction.KspI.size
Bio.Restriction.Restriction.Kzo9I.size
Bio.Restriction.Restriction.LguI.size
Bio.Restriction.Restriction.LpnI.size
Bio.Restriction.Restriction.LpnPI.size
Bio.Restriction.Restriction.Lsp1109I.size
Bio.Restriction.Restriction.LweI.size
Bio.Restriction.Restriction.MabI.size
Bio.Restriction.Restriction.MaeI.size
Bio.Restriction.Restriction.MaeII.size
Bio.Restriction.Restriction.MaeIII.size
Bio.Restriction.Restriction.MalI.size
Bio.Restriction.Restriction.MaqI.size
Bio.Restriction.Restriction.MauBI.size
Bio.Restriction.Restriction.MbiI.size
Bio.Restriction.Restriction.MboI.size
Bio.Restriction.Restriction.MboII.size
Bio.Restriction.Restriction.McaTI.size
Bio.Restriction.Restriction.McrI.size
Bio.Restriction.Restriction.MfeI.size
Bio.Restriction.Restriction.MflI.size
Bio.Restriction.Restriction.MhlI.size
Bio.Restriction.Restriction.MjaIV.size
Bio.Restriction.Restriction.MkaDII.size
Bio.Restriction.Restriction.MlsI.size
Bio.Restriction.Restriction.MluCI.size
Bio.Restriction.Restriction.MluI.size
Bio.Restriction.Restriction.MluNI.size
Bio.Restriction.Restriction.Mly113I.size
Bio.Restriction.Restriction.MlyI.size
Bio.Restriction.Restriction.MmeI.size
Bio.Restriction.Restriction.MnlI.size
Bio.Restriction.Restriction.Mph1103I.size
Bio.Restriction.Restriction.MreI.size
Bio.Restriction.Restriction.MroI.size
Bio.Restriction.Restriction.MroNI.size
Bio.Restriction.Restriction.MroXI.size
Bio.Restriction.Restriction.MscI.size
Bio.Restriction.Restriction.MseI.size
Bio.Restriction.Restriction.MslI.size
Bio.Restriction.Restriction.Msp20I.size
Bio.Restriction.Restriction.MspA1I.size
Bio.Restriction.Restriction.MspCI.size
Bio.Restriction.Restriction.MspI.size
Bio.Restriction.Restriction.MspJI.size
Bio.Restriction.Restriction.MspR9I.size
Bio.Restriction.Restriction.MssI.size
Bio.Restriction.Restriction.MstI.size
Bio.Restriction.Restriction.MunI.size
Bio.Restriction.Restriction.Mva1269I.size
Bio.Restriction.Restriction.MvaI.size
Bio.Restriction.Restriction.MvnI.size
Bio.Restriction.Restriction.MvrI.size
Bio.Restriction.Restriction.MwoI.size
Bio.Restriction.Restriction.NaeI.size
Bio.Restriction.Restriction.NarI.size
Bio.Restriction.Restriction.NciI.size
Bio.Restriction.Restriction.NcoI.size
Bio.Restriction.Restriction.NdeI.size
Bio.Restriction.Restriction.NdeII.size
Bio.Restriction.Restriction.NgoAVIII.size
Bio.Restriction.Restriction.NgoMIV.size
Bio.Restriction.Restriction.NhaXI.size
Bio.Restriction.Restriction.NheI.size
Bio.Restriction.Restriction.NlaCI.size
Bio.Restriction.Restriction.NlaIII.size
Bio.Restriction.Restriction.NlaIV.size
Bio.Restriction.Restriction.Nli3877I.size
Bio.Restriction.Restriction.NmeAIII.size
Bio.Restriction.Restriction.NmeDI.size
Bio.Restriction.Restriction.NmuCI.size
Bio.Restriction.Restriction.NotI.size
Bio.Restriction.Restriction.NruI.size
Bio.Restriction.Restriction.NsbI.size
Bio.Restriction.Restriction.NsiI.size
Bio.Restriction.Restriction.NspBII.size
Bio.Restriction.Restriction.NspI.size
Bio.Restriction.Restriction.NspV.size
Bio.Restriction.Restriction.OliI.size
Bio.Restriction.Restriction.PabI.size
Bio.Restriction.Restriction.PacI.size
Bio.Restriction.Restriction.PaeI.size
Bio.Restriction.Restriction.PaeR7I.size
Bio.Restriction.Restriction.PagI.size
Bio.Restriction.Restriction.PalAI.size
Bio.Restriction.Restriction.PasI.size
Bio.Restriction.Restriction.PauI.size
Bio.Restriction.Restriction.PceI.size
Bio.Restriction.Restriction.PciI.size
Bio.Restriction.Restriction.PciSI.size
Bio.Restriction.Restriction.PcsI.size
Bio.Restriction.Restriction.PctI.size
Bio.Restriction.Restriction.PdiI.size
Bio.Restriction.Restriction.PdmI.size
Bio.Restriction.Restriction.PenI.size
Bio.Restriction.Restriction.PfeI.size
Bio.Restriction.Restriction.Pfl1108I.size
Bio.Restriction.Restriction.Pfl23II.size
Bio.Restriction.Restriction.PflFI.size
Bio.Restriction.Restriction.PflMI.size
Bio.Restriction.Restriction.PfoI.size
Bio.Restriction.Restriction.PinAI.size
Bio.Restriction.Restriction.PlaDI.size
Bio.Restriction.Restriction.Ple19I.size
Bio.Restriction.Restriction.PleI.size
Bio.Restriction.Restriction.PluTI.size
Bio.Restriction.Restriction.PmaCI.size
Bio.Restriction.Restriction.PmeI.size
Bio.Restriction.Restriction.PmlI.size
Bio.Restriction.Restriction.PpiI.size
Bio.Restriction.Restriction.PpsI.size
Bio.Restriction.Restriction.Ppu10I.size
Bio.Restriction.Restriction.Ppu21I.size
Bio.Restriction.Restriction.PpuMI.size
Bio.Restriction.Restriction.PscI.size
Bio.Restriction.Restriction.PshAI.size
Bio.Restriction.Restriction.PshBI.size
Bio.Restriction.Restriction.PsiI.size
Bio.Restriction.Restriction.Psp03I.size
Bio.Restriction.Restriction.Psp124BI.size
Bio.Restriction.Restriction.Psp1406I.size
Bio.Restriction.Restriction.Psp5II.size
Bio.Restriction.Restriction.Psp6I.size
Bio.Restriction.Restriction.PspCI.size
Bio.Restriction.Restriction.PspEI.size
Bio.Restriction.Restriction.PspGI.size
Bio.Restriction.Restriction.PspLI.size
Bio.Restriction.Restriction.PspN4I.size
Bio.Restriction.Restriction.PspOMI.size
Bio.Restriction.Restriction.PspOMII.size
Bio.Restriction.Restriction.PspPI.size
Bio.Restriction.Restriction.PspPPI.size
Bio.Restriction.Restriction.PspPRI.size
Bio.Restriction.Restriction.PspXI.size
Bio.Restriction.Restriction.PsrI.size
Bio.Restriction.Restriction.PssI.size
Bio.Restriction.Restriction.PstI.size
Bio.Restriction.Restriction.PstNI.size
Bio.Restriction.Restriction.PsuI.size
Bio.Restriction.Restriction.PsyI.size
Bio.Restriction.Restriction.PteI.size
Bio.Restriction.Restriction.PvuI.size
Bio.Restriction.Restriction.PvuII.size
Bio.Restriction.Restriction.R2_BceSIV.size
Bio.Restriction.Restriction.RceI.size
Bio.Restriction.Restriction.RdeGBI.size
Bio.Restriction.Restriction.RdeGBII.size
Bio.Restriction.Restriction.RdeGBIII.size
Bio.Restriction.Restriction.RflFIII.size
Bio.Restriction.Restriction.RgaI.size
Bio.Restriction.Restriction.RigI.size
Bio.Restriction.Restriction.RlaI.size
Bio.Restriction.Restriction.RleAI.size
Bio.Restriction.Restriction.RpaB5I.size
Bio.Restriction.Restriction.RpaBI.size
Bio.Restriction.Restriction.RpaI.size
Bio.Restriction.Restriction.RpaTI.size
Bio.Restriction.Restriction.RruI.size
Bio.Restriction.Restriction.RsaI.size
Bio.Restriction.Restriction.RsaNI.size
Bio.Restriction.Restriction.RseI.size
Bio.Restriction.Restriction.Rsr2I.size
Bio.Restriction.Restriction.RsrII.size
Bio.Restriction.Restriction.SacI.size
Bio.Restriction.Restriction.SacII.size
Bio.Restriction.Restriction.SalI.size
Bio.Restriction.Restriction.SanDI.size
Bio.Restriction.Restriction.SapI.size
Bio.Restriction.Restriction.SaqAI.size
Bio.Restriction.Restriction.SatI.size
Bio.Restriction.Restriction.Sau3AI.size
Bio.Restriction.Restriction.Sau96I.size
Bio.Restriction.Restriction.SauI.size
Bio.Restriction.Restriction.SbfI.size
Bio.Restriction.Restriction.ScaI.size
Bio.Restriction.Restriction.SchI.size
Bio.Restriction.Restriction.SciI.size
Bio.Restriction.Restriction.ScrFI.size
Bio.Restriction.Restriction.SdaI.size
Bio.Restriction.Restriction.SdeAI.size
Bio.Restriction.Restriction.SdeOSI.size
Bio.Restriction.Restriction.SduI.size
Bio.Restriction.Restriction.SecI.size
Bio.Restriction.Restriction.SelI.size
Bio.Restriction.Restriction.SetI.size
Bio.Restriction.Restriction.SexAI.size
Bio.Restriction.Restriction.SfaAI.size
Bio.Restriction.Restriction.SfaNI.size
Bio.Restriction.Restriction.SfcI.size
Bio.Restriction.Restriction.SfeI.size
Bio.Restriction.Restriction.SfiI.size
Bio.Restriction.Restriction.SfoI.size
Bio.Restriction.Restriction.Sfr274I.size
Bio.Restriction.Restriction.Sfr303I.size
Bio.Restriction.Restriction.SfuI.size
Bio.Restriction.Restriction.SgeI.size
Bio.Restriction.Restriction.SgfI.size
Bio.Restriction.Restriction.SgrAI.size
Bio.Restriction.Restriction.SgrBI.size
Bio.Restriction.Restriction.SgrDI.size
Bio.Restriction.Restriction.SgrTI.size
Bio.Restriction.Restriction.SgsI.size
Bio.Restriction.Restriction.SimI.size
Bio.Restriction.Restriction.SlaI.size
Bio.Restriction.Restriction.SmaI.size
Bio.Restriction.Restriction.SmiI.size
Bio.Restriction.Restriction.SmiMI.size
Bio.Restriction.Restriction.SmlI.size
Bio.Restriction.Restriction.SmoI.size
Bio.Restriction.Restriction.SnaBI.size
Bio.Restriction.Restriction.SnaI.size
Bio.Restriction.Restriction.Sno506I.size
Bio.Restriction.Restriction.SpeI.size
Bio.Restriction.Restriction.SphI.size
Bio.Restriction.Restriction.SplI.size
Bio.Restriction.Restriction.SpoDI.size
Bio.Restriction.Restriction.SrfI.size
Bio.Restriction.Restriction.Sse232I.size
Bio.Restriction.Restriction.Sse8387I.size
Bio.Restriction.Restriction.Sse8647I.size
Bio.Restriction.Restriction.Sse9I.size
Bio.Restriction.Restriction.SseBI.size
Bio.Restriction.Restriction.SsiI.size
Bio.Restriction.Restriction.SspD5I.size
Bio.Restriction.Restriction.SspDI.size
Bio.Restriction.Restriction.SspI.size
Bio.Restriction.Restriction.SstE37I.size
Bio.Restriction.Restriction.SstI.size
Bio.Restriction.Restriction.Sth132I.size
Bio.Restriction.Restriction.Sth302II.size
Bio.Restriction.Restriction.StrI.size
Bio.Restriction.Restriction.StsI.size
Bio.Restriction.Restriction.StuI.size
Bio.Restriction.Restriction.StyD4I.size
Bio.Restriction.Restriction.StyI.size
Bio.Restriction.Restriction.SwaI.size
Bio.Restriction.Restriction.TaaI.size
Bio.Restriction.Restriction.TaiI.size
Bio.Restriction.Restriction.TaqI.size
Bio.Restriction.Restriction.TaqII.size
Bio.Restriction.Restriction.TasI.size
Bio.Restriction.Restriction.TatI.size
Bio.Restriction.Restriction.TauI.size
Bio.Restriction.Restriction.TfiI.size
Bio.Restriction.Restriction.Tru1I.size
Bio.Restriction.Restriction.Tru9I.size
Bio.Restriction.Restriction.TscAI.size
Bio.Restriction.Restriction.TseFI.size
Bio.Restriction.Restriction.TseI.size
Bio.Restriction.Restriction.TsoI.size
Bio.Restriction.Restriction.Tsp45I.size
Bio.Restriction.Restriction.Tsp4CI.size
Bio.Restriction.Restriction.TspDTI.size
Bio.Restriction.Restriction.TspEI.size
Bio.Restriction.Restriction.TspGWI.size
Bio.Restriction.Restriction.TspMI.size
Bio.Restriction.Restriction.TspRI.size
Bio.Restriction.Restriction.TssI.size
Bio.Restriction.Restriction.TstI.size
Bio.Restriction.Restriction.TsuI.size
Bio.Restriction.Restriction.Tth111I.size
Bio.Restriction.Restriction.Tth111II.size
Bio.Restriction.Restriction.UbaF11I.size
Bio.Restriction.Restriction.UbaF12I.size
Bio.Restriction.Restriction.UbaF13I.size
Bio.Restriction.Restriction.UbaF14I.size
Bio.Restriction.Restriction.UbaF9I.size
Bio.Restriction.Restriction.UbaPI.size
Bio.Restriction.Restriction.UcoMSI.size
Bio.Restriction.Restriction.UnbI.size
Bio.Restriction.Restriction.Van91I.size
Bio.Restriction.Restriction.Vha464I.size
Bio.Restriction.Restriction.VneI.size
Bio.Restriction.Restriction.VpaK11AI.size
Bio.Restriction.Restriction.VpaK11BI.size
Bio.Restriction.Restriction.VspI.size
Bio.Restriction.Restriction.WviI.size
Bio.Restriction.Restriction.XagI.size
Bio.Restriction.Restriction.XapI.size
Bio.Restriction.Restriction.XbaI.size
Bio.Restriction.Restriction.XceI.size
Bio.Restriction.Restriction.XcmI.size
Bio.Restriction.Restriction.XhoI.size
Bio.Restriction.Restriction.XhoII.size
Bio.Restriction.Restriction.XmaI.size
Bio.Restriction.Restriction.XmaIII.size
Bio.Restriction.Restriction.XmaJI.size
Bio.Restriction.Restriction.XmiI.size
Bio.Restriction.Restriction.XmnI.size
Bio.Restriction.Restriction.XspI.size
Bio.Restriction.Restriction.YkrI.size
Bio.Restriction.Restriction.ZraI.size
Bio.Restriction.Restriction.ZrmI.size
Bio.Restriction.Restriction.Zsp2I.size
Bio.codonalign.codonalphabet.CodonAlphabet.size" class="py-name" href="#" onclick="return doclink('link-492', 'size', 'link-270');">size</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-493" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-493', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1231"></a><tt class="py-lineno">1231</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-494" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-494', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">4</tt> <tt class="py-keyword">and</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-495" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-495', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"aa"</tt><tt class="py-op">,</tt> <tt class="py-string">"bp"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L1232"></a><tt class="py-lineno">1232</tt>  <tt class="py-line">            <tt class="py-comment"># Cope with pseudo-GenBank files like this:</tt> </tt>
<a name="L1233"></a><tt class="py-lineno">1233</tt>  <tt class="py-line">            <tt class="py-comment">#   "LOCUS       RNA5 complete       1718 bp"</tt> </tt>
<a name="L1234"></a><tt class="py-lineno">1234</tt>  <tt class="py-line">            <tt class="py-comment"># Treat everything between LOCUS and the size as the identifier.</tt> </tt>
<a name="L1235"></a><tt class="py-lineno">1235</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Malformed LOCUS line found - is this "</tt> </tt>
<a name="L1236"></a><tt class="py-lineno">1236</tt>  <tt class="py-line">                          <tt class="py-string">"correct?\n:%r"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">,</tt> <tt id="link-496" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-496', 'BiopythonParserWarning', 'link-10');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1237"></a><tt class="py-lineno">1237</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-497" class="py-name"><a title="Bio.GenBank._FeatureConsumer.locus
Bio.GenBank._RecordConsumer.locus" class="py-name" href="#" onclick="return doclink('link-497', 'locus', 'link-235');">locus</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">5</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-498" class="py-name"><a title="Bio.Seq.Seq.rsplit" class="py-name" href="#" onclick="return doclink('link-498', 'rsplit', 'link-219');">rsplit</a></tt><tt class="py-op">(</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-number">2</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-499" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-499', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1238"></a><tt class="py-lineno">1238</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-500" class="py-name"><a title="Bio.Alphabet.Alphabet.size
Bio.Alphabet.Reduced.HPModel.size
Bio.Alphabet.Reduced.Murphy10.size
Bio.Alphabet.Reduced.Murphy15.size
Bio.Alphabet.Reduced.Murphy4.size
Bio.Alphabet.Reduced.Murphy8.size
Bio.Alphabet.Reduced.PC5.size
Bio.Alphabet.SingleLetterAlphabet.size
Bio.Alphabet.ThreeLetterProtein.size
Bio.GenBank._FeatureConsumer.size
Bio.GenBank._RecordConsumer.size
Bio.Restriction.Restriction.AanI.size
Bio.Restriction.Restriction.AarI.size
Bio.Restriction.Restriction.AasI.size
Bio.Restriction.Restriction.AatII.size
Bio.Restriction.Restriction.AbaSI.size
Bio.Restriction.Restriction.AbsI.size
Bio.Restriction.Restriction.Acc16I.size
Bio.Restriction.Restriction.Acc36I.size
Bio.Restriction.Restriction.Acc65I.size
Bio.Restriction.Restriction.AccB1I.size
Bio.Restriction.Restriction.AccB7I.size
Bio.Restriction.Restriction.AccBSI.size
Bio.Restriction.Restriction.AccI.size
Bio.Restriction.Restriction.AccII.size
Bio.Restriction.Restriction.AccIII.size
Bio.Restriction.Restriction.AceIII.size
Bio.Restriction.Restriction.AciI.size
Bio.Restriction.Restriction.AclI.size
Bio.Restriction.Restriction.AclWI.size
Bio.Restriction.Restriction.AcoI.size
Bio.Restriction.Restriction.AcsI.size
Bio.Restriction.Restriction.AcuI.size
Bio.Restriction.Restriction.AcvI.size
Bio.Restriction.Restriction.AcyI.size
Bio.Restriction.Restriction.AdeI.size
Bio.Restriction.Restriction.AfaI.size
Bio.Restriction.Restriction.AfeI.size
Bio.Restriction.Restriction.AfiI.size
Bio.Restriction.Restriction.AflII.size
Bio.Restriction.Restriction.AflIII.size
Bio.Restriction.Restriction.AgeI.size
Bio.Restriction.Restriction.AgsI.size
Bio.Restriction.Restriction.AhaIII.size
Bio.Restriction.Restriction.AhdI.size
Bio.Restriction.Restriction.AhlI.size
Bio.Restriction.Restriction.AjiI.size
Bio.Restriction.Restriction.AjnI.size
Bio.Restriction.Restriction.AjuI.size
Bio.Restriction.Restriction.AleI.size
Bio.Restriction.Restriction.AlfI.size
Bio.Restriction.Restriction.AloI.size
Bio.Restriction.Restriction.AluBI.size
Bio.Restriction.Restriction.AluI.size
Bio.Restriction.Restriction.Alw21I.size
Bio.Restriction.Restriction.Alw26I.size
Bio.Restriction.Restriction.Alw44I.size
Bio.Restriction.Restriction.AlwFI.size
Bio.Restriction.Restriction.AlwI.size
Bio.Restriction.Restriction.AlwNI.size
Bio.Restriction.Restriction.Ama87I.size
Bio.Restriction.Restriction.Aor13HI.size
Bio.Restriction.Restriction.Aor51HI.size
Bio.Restriction.Restriction.AoxI.size
Bio.Restriction.Restriction.ApaBI.size
Bio.Restriction.Restriction.ApaI.size
Bio.Restriction.Restriction.ApaLI.size
Bio.Restriction.Restriction.ApeKI.size
Bio.Restriction.Restriction.ApoI.size
Bio.Restriction.Restriction.ApyPI.size
Bio.Restriction.Restriction.AquII.size
Bio.Restriction.Restriction.AquIII.size
Bio.Restriction.Restriction.AquIV.size
Bio.Restriction.Restriction.ArsI.size
Bio.Restriction.Restriction.AscI.size
Bio.Restriction.Restriction.AseI.size
Bio.Restriction.Restriction.Asi256I.size
Bio.Restriction.Restriction.AsiGI.size
Bio.Restriction.Restriction.AsiSI.size
Bio.Restriction.Restriction.Asp700I.size
Bio.Restriction.Restriction.Asp718I.size
Bio.Restriction.Restriction.AspA2I.size
Bio.Restriction.Restriction.AspBHI.size
Bio.Restriction.Restriction.AspLEI.size
Bio.Restriction.Restriction.AspS9I.size
Bio.Restriction.Restriction.AssI.size
Bio.Restriction.Restriction.AsuC2I.size
Bio.Restriction.Restriction.AsuHPI.size
Bio.Restriction.Restriction.AsuI.size
Bio.Restriction.Restriction.AsuII.size
Bio.Restriction.Restriction.AsuNHI.size
Bio.Restriction.Restriction.AvaI.size
Bio.Restriction.Restriction.AvaII.size
Bio.Restriction.Restriction.AvaIII.size
Bio.Restriction.Restriction.AvrII.size
Bio.Restriction.Restriction.AxyI.size
Bio.Restriction.Restriction.BaeGI.size
Bio.Restriction.Restriction.BaeI.size
Bio.Restriction.Restriction.BalI.size
Bio.Restriction.Restriction.BamHI.size
Bio.Restriction.Restriction.BanI.size
Bio.Restriction.Restriction.BanII.size
Bio.Restriction.Restriction.BarI.size
Bio.Restriction.Restriction.BasI.size
Bio.Restriction.Restriction.BauI.size
Bio.Restriction.Restriction.Bbr7I.size
Bio.Restriction.Restriction.BbrPI.size
Bio.Restriction.Restriction.BbsI.size
Bio.Restriction.Restriction.Bbv12I.size
Bio.Restriction.Restriction.BbvCI.size
Bio.Restriction.Restriction.BbvI.size
Bio.Restriction.Restriction.BbvII.size
Bio.Restriction.Restriction.BccI.size
Bio.Restriction.Restriction.Bce83I.size
Bio.Restriction.Restriction.BceAI.size
Bio.Restriction.Restriction.BcefI.size
Bio.Restriction.Restriction.BcgI.size
Bio.Restriction.Restriction.BciT130I.size
Bio.Restriction.Restriction.BciVI.size
Bio.Restriction.Restriction.BclI.size
Bio.Restriction.Restriction.BcnI.size
Bio.Restriction.Restriction.BcoDI.size
Bio.Restriction.Restriction.BcuI.size
Bio.Restriction.Restriction.BdaI.size
Bio.Restriction.Restriction.BetI.size
Bio.Restriction.Restriction.BfaI.size
Bio.Restriction.Restriction.BfiI.size
Bio.Restriction.Restriction.BfmI.size
Bio.Restriction.Restriction.BfoI.size
Bio.Restriction.Restriction.BfrI.size
Bio.Restriction.Restriction.BfuAI.size
Bio.Restriction.Restriction.BfuCI.size
Bio.Restriction.Restriction.BfuI.size
Bio.Restriction.Restriction.BglI.size
Bio.Restriction.Restriction.BglII.size
Bio.Restriction.Restriction.BinI.size
Bio.Restriction.Restriction.BisI.size
Bio.Restriction.Restriction.BlnI.size
Bio.Restriction.Restriction.BlpI.size
Bio.Restriction.Restriction.BlsI.size
Bio.Restriction.Restriction.BmcAI.size
Bio.Restriction.Restriction.Bme1390I.size
Bio.Restriction.Restriction.Bme18I.size
Bio.Restriction.Restriction.BmeDI.size
Bio.Restriction.Restriction.BmeRI.size
Bio.Restriction.Restriction.BmeT110I.size
Bio.Restriction.Restriction.BmgBI.size
Bio.Restriction.Restriction.BmgI.size
Bio.Restriction.Restriction.BmgT120I.size
Bio.Restriction.Restriction.BmiI.size
Bio.Restriction.Restriction.BmrFI.size
Bio.Restriction.Restriction.BmrI.size
Bio.Restriction.Restriction.BmsI.size
Bio.Restriction.Restriction.BmtI.size
Bio.Restriction.Restriction.BmuI.size
Bio.Restriction.Restriction.BoxI.size
Bio.Restriction.Restriction.BpiI.size
Bio.Restriction.Restriction.BplI.size
Bio.Restriction.Restriction.BpmI.size
Bio.Restriction.Restriction.Bpu10I.size
Bio.Restriction.Restriction.Bpu1102I.size
Bio.Restriction.Restriction.Bpu14I.size
Bio.Restriction.Restriction.BpuEI.size
Bio.Restriction.Restriction.BpuMI.size
Bio.Restriction.Restriction.BpvUI.size
Bio.Restriction.Restriction.Bsa29I.size
Bio.Restriction.Restriction.BsaAI.size
Bio.Restriction.Restriction.BsaBI.size
Bio.Restriction.Restriction.BsaHI.size
Bio.Restriction.Restriction.BsaI.size
Bio.Restriction.Restriction.BsaJI.size
Bio.Restriction.Restriction.BsaWI.size
Bio.Restriction.Restriction.BsaXI.size
Bio.Restriction.Restriction.BsbI.size
Bio.Restriction.Restriction.Bsc4I.size
Bio.Restriction.Restriction.BscAI.size
Bio.Restriction.Restriction.BscGI.size
Bio.Restriction.Restriction.Bse118I.size
Bio.Restriction.Restriction.Bse1I.size
Bio.Restriction.Restriction.Bse21I.size
Bio.Restriction.Restriction.Bse3DI.size
Bio.Restriction.Restriction.Bse8I.size
Bio.Restriction.Restriction.BseAI.size
Bio.Restriction.Restriction.BseBI.size
Bio.Restriction.Restriction.BseCI.size
Bio.Restriction.Restriction.BseDI.size
Bio.Restriction.Restriction.BseGI.size
Bio.Restriction.Restriction.BseJI.size
Bio.Restriction.Restriction.BseLI.size
Bio.Restriction.Restriction.BseMI.size
Bio.Restriction.Restriction.BseMII.size
Bio.Restriction.Restriction.BseNI.size
Bio.Restriction.Restriction.BsePI.size
Bio.Restriction.Restriction.BseRI.size
Bio.Restriction.Restriction.BseSI.size
Bio.Restriction.Restriction.BseX3I.size
Bio.Restriction.Restriction.BseXI.size
Bio.Restriction.Restriction.BseYI.size
Bio.Restriction.Restriction.BsgI.size
Bio.Restriction.Restriction.Bsh1236I.size
Bio.Restriction.Restriction.Bsh1285I.size
Bio.Restriction.Restriction.BshFI.size
Bio.Restriction.Restriction.BshNI.size
Bio.Restriction.Restriction.BshTI.size
Bio.Restriction.Restriction.BshVI.size
Bio.Restriction.Restriction.BsiEI.size
Bio.Restriction.Restriction.BsiHKAI.size
Bio.Restriction.Restriction.BsiHKCI.size
Bio.Restriction.Restriction.BsiI.size
Bio.Restriction.Restriction.BsiSI.size
Bio.Restriction.Restriction.BsiWI.size
Bio.Restriction.Restriction.BsiYI.size
Bio.Restriction.Restriction.BslFI.size
Bio.Restriction.Restriction.BslI.size
Bio.Restriction.Restriction.BsmAI.size
Bio.Restriction.Restriction.BsmBI.size
Bio.Restriction.Restriction.BsmFI.size
Bio.Restriction.Restriction.BsmI.size
Bio.Restriction.Restriction.BsnI.size
Bio.Restriction.Restriction.Bso31I.size
Bio.Restriction.Restriction.BsoBI.size
Bio.Restriction.Restriction.Bsp119I.size
Bio.Restriction.Restriction.Bsp120I.size
Bio.Restriction.Restriction.Bsp1286I.size
Bio.Restriction.Restriction.Bsp13I.size
Bio.Restriction.Restriction.Bsp1407I.size
Bio.Restriction.Restriction.Bsp143I.size
Bio.Restriction.Restriction.Bsp1720I.size
Bio.Restriction.Restriction.Bsp19I.size
Bio.Restriction.Restriction.Bsp24I.size
Bio.Restriction.Restriction.Bsp68I.size
Bio.Restriction.Restriction.BspACI.size
Bio.Restriction.Restriction.BspCNI.size
Bio.Restriction.Restriction.BspD6I.size
Bio.Restriction.Restriction.BspDI.size
Bio.Restriction.Restriction.BspEI.size
Bio.Restriction.Restriction.BspFNI.size
Bio.Restriction.Restriction.BspGI.size
Bio.Restriction.Restriction.BspHI.size
Bio.Restriction.Restriction.BspLI.size
Bio.Restriction.Restriction.BspLU11I.size
Bio.Restriction.Restriction.BspMI.size
Bio.Restriction.Restriction.BspMII.size
Bio.Restriction.Restriction.BspNCI.size
Bio.Restriction.Restriction.BspOI.size
Bio.Restriction.Restriction.BspPI.size
Bio.Restriction.Restriction.BspQI.size
Bio.Restriction.Restriction.BspT104I.size
Bio.Restriction.Restriction.BspT107I.size
Bio.Restriction.Restriction.BspTI.size
Bio.Restriction.Restriction.BsrBI.size
Bio.Restriction.Restriction.BsrDI.size
Bio.Restriction.Restriction.BsrFI.size
Bio.Restriction.Restriction.BsrGI.size
Bio.Restriction.Restriction.BsrI.size
Bio.Restriction.Restriction.BsrSI.size
Bio.Restriction.Restriction.BssAI.size
Bio.Restriction.Restriction.BssECI.size
Bio.Restriction.Restriction.BssHII.size
Bio.Restriction.Restriction.BssKI.size
Bio.Restriction.Restriction.BssMI.size
Bio.Restriction.Restriction.BssNAI.size
Bio.Restriction.Restriction.BssNI.size
Bio.Restriction.Restriction.BssSI.size
Bio.Restriction.Restriction.BssT1I.size
Bio.Restriction.Restriction.Bst1107I.size
Bio.Restriction.Restriction.Bst2BI.size
Bio.Restriction.Restriction.Bst2UI.size
Bio.Restriction.Restriction.Bst4CI.size
Bio.Restriction.Restriction.Bst6I.size
Bio.Restriction.Restriction.BstACI.size
Bio.Restriction.Restriction.BstAFI.size
Bio.Restriction.Restriction.BstAPI.size
Bio.Restriction.Restriction.BstAUI.size
Bio.Restriction.Restriction.BstBAI.size
Bio.Restriction.Restriction.BstBI.size
Bio.Restriction.Restriction.BstC8I.size
Bio.Restriction.Restriction.BstDEI.size
Bio.Restriction.Restriction.BstDSI.size
Bio.Restriction.Restriction.BstEII.size
Bio.Restriction.Restriction.BstENI.size
Bio.Restriction.Restriction.BstF5I.size
Bio.Restriction.Restriction.BstFNI.size
Bio.Restriction.Restriction.BstH2I.size
Bio.Restriction.Restriction.BstHHI.size
Bio.Restriction.Restriction.BstKTI.size
Bio.Restriction.Restriction.BstMAI.size
Bio.Restriction.Restriction.BstMBI.size
Bio.Restriction.Restriction.BstMCI.size
Bio.Restriction.Restriction.BstMWI.size
Bio.Restriction.Restriction.BstNI.size
Bio.Restriction.Restriction.BstNSI.size
Bio.Restriction.Restriction.BstOI.size
Bio.Restriction.Restriction.BstPAI.size
Bio.Restriction.Restriction.BstPI.size
Bio.Restriction.Restriction.BstSCI.size
Bio.Restriction.Restriction.BstSFI.size
Bio.Restriction.Restriction.BstSLI.size
Bio.Restriction.Restriction.BstSNI.size
Bio.Restriction.Restriction.BstUI.size
Bio.Restriction.Restriction.BstV1I.size
Bio.Restriction.Restriction.BstV2I.size
Bio.Restriction.Restriction.BstX2I.size
Bio.Restriction.Restriction.BstXI.size
Bio.Restriction.Restriction.BstYI.size
Bio.Restriction.Restriction.BstZ17I.size
Bio.Restriction.Restriction.BstZI.size
Bio.Restriction.Restriction.Bsu15I.size
Bio.Restriction.Restriction.Bsu36I.size
Bio.Restriction.Restriction.BsuI.size
Bio.Restriction.Restriction.BsuRI.size
Bio.Restriction.Restriction.BtgI.size
Bio.Restriction.Restriction.BtgZI.size
Bio.Restriction.Restriction.BthCI.size
Bio.Restriction.Restriction.BtrI.size
Bio.Restriction.Restriction.BtsCI.size
Bio.Restriction.Restriction.BtsI.size
Bio.Restriction.Restriction.BtsIMutI.size
Bio.Restriction.Restriction.BtuMI.size
Bio.Restriction.Restriction.BveI.size
Bio.Restriction.Restriction.Cac8I.size
Bio.Restriction.Restriction.CaiI.size
Bio.Restriction.Restriction.CauII.size
Bio.Restriction.Restriction.CchII.size
Bio.Restriction.Restriction.CchIII.size
Bio.Restriction.Restriction.CciI.size
Bio.Restriction.Restriction.CciNI.size
Bio.Restriction.Restriction.Cdi630V.size
Bio.Restriction.Restriction.CdiI.size
Bio.Restriction.Restriction.CdpI.size
Bio.Restriction.Restriction.CfoI.size
Bio.Restriction.Restriction.Cfr10I.size
Bio.Restriction.Restriction.Cfr13I.size
Bio.Restriction.Restriction.Cfr42I.size
Bio.Restriction.Restriction.Cfr9I.size
Bio.Restriction.Restriction.CfrI.size
Bio.Restriction.Restriction.Cgl13032I.size
Bio.Restriction.Restriction.Cgl13032II.size
Bio.Restriction.Restriction.ChaI.size
Bio.Restriction.Restriction.CjeFIII.size
Bio.Restriction.Restriction.CjeFV.size
Bio.Restriction.Restriction.CjeI.size
Bio.Restriction.Restriction.CjeNII.size
Bio.Restriction.Restriction.CjeNIII.size
Bio.Restriction.Restriction.CjeP659IV.size
Bio.Restriction.Restriction.CjePI.size
Bio.Restriction.Restriction.CjuI.size
Bio.Restriction.Restriction.CjuII.size
Bio.Restriction.Restriction.ClaI.size
Bio.Restriction.Restriction.CpoI.size
Bio.Restriction.Restriction.CseI.size
Bio.Restriction.Restriction.CsiI.size
Bio.Restriction.Restriction.Csp6I.size
Bio.Restriction.Restriction.CspAI.size
Bio.Restriction.Restriction.CspCI.size
Bio.Restriction.Restriction.CspI.size
Bio.Restriction.Restriction.CstMI.size
Bio.Restriction.Restriction.CviAII.size
Bio.Restriction.Restriction.CviJI.size
Bio.Restriction.Restriction.CviKI_1.size
Bio.Restriction.Restriction.CviQI.size
Bio.Restriction.Restriction.CviRI.size
Bio.Restriction.Restriction.DdeI.size
Bio.Restriction.Restriction.DinI.size
Bio.Restriction.Restriction.DpnI.size
Bio.Restriction.Restriction.DpnII.size
Bio.Restriction.Restriction.DraI.size
Bio.Restriction.Restriction.DraII.size
Bio.Restriction.Restriction.DraIII.size
Bio.Restriction.Restriction.DraRI.size
Bio.Restriction.Restriction.DrdI.size
Bio.Restriction.Restriction.DrdII.size
Bio.Restriction.Restriction.DriI.size
Bio.Restriction.Restriction.DsaI.size
Bio.Restriction.Restriction.DseDI.size
Bio.Restriction.Restriction.EaeI.size
Bio.Restriction.Restriction.EagI.size
Bio.Restriction.Restriction.Eam1104I.size
Bio.Restriction.Restriction.Eam1105I.size
Bio.Restriction.Restriction.EarI.size
Bio.Restriction.Restriction.EciI.size
Bio.Restriction.Restriction.Ecl136II.size
Bio.Restriction.Restriction.EclXI.size
Bio.Restriction.Restriction.Eco105I.size
Bio.Restriction.Restriction.Eco130I.size
Bio.Restriction.Restriction.Eco147I.size
Bio.Restriction.Restriction.Eco24I.size
Bio.Restriction.Restriction.Eco31I.size
Bio.Restriction.Restriction.Eco32I.size
Bio.Restriction.Restriction.Eco47I.size
Bio.Restriction.Restriction.Eco47III.size
Bio.Restriction.Restriction.Eco52I.size
Bio.Restriction.Restriction.Eco53kI.size
Bio.Restriction.Restriction.Eco57I.size
Bio.Restriction.Restriction.Eco57MI.size
Bio.Restriction.Restriction.Eco72I.size
Bio.Restriction.Restriction.Eco81I.size
Bio.Restriction.Restriction.Eco88I.size
Bio.Restriction.Restriction.Eco91I.size
Bio.Restriction.Restriction.EcoHI.size
Bio.Restriction.Restriction.EcoICRI.size
Bio.Restriction.Restriction.EcoNI.size
Bio.Restriction.Restriction.EcoO109I.size
Bio.Restriction.Restriction.EcoO65I.size
Bio.Restriction.Restriction.EcoRI.size
Bio.Restriction.Restriction.EcoRII.size
Bio.Restriction.Restriction.EcoRV.size
Bio.Restriction.Restriction.EcoT14I.size
Bio.Restriction.Restriction.EcoT22I.size
Bio.Restriction.Restriction.EcoT38I.size
Bio.Restriction.Restriction.EgeI.size
Bio.Restriction.Restriction.EheI.size
Bio.Restriction.Restriction.ErhI.size
Bio.Restriction.Restriction.EsaBC3I.size
Bio.Restriction.Restriction.EsaSSI.size
Bio.Restriction.Restriction.Esp3I.size
Bio.Restriction.Restriction.EspI.size
Bio.Restriction.Restriction.FaeI.size
Bio.Restriction.Restriction.FaiI.size
Bio.Restriction.Restriction.FalI.size
Bio.Restriction.Restriction.FaqI.size
Bio.Restriction.Restriction.FatI.size
Bio.Restriction.Restriction.FauI.size
Bio.Restriction.Restriction.FauNDI.size
Bio.Restriction.Restriction.FbaI.size
Bio.Restriction.Restriction.FblI.size
Bio.Restriction.Restriction.FinI.size
Bio.Restriction.Restriction.FmuI.size
Bio.Restriction.Restriction.Fnu4HI.size
Bio.Restriction.Restriction.FnuDII.size
Bio.Restriction.Restriction.FokI.size
Bio.Restriction.Restriction.FriOI.size
Bio.Restriction.Restriction.FseI.size
Bio.Restriction.Restriction.Fsp4HI.size
Bio.Restriction.Restriction.FspAI.size
Bio.Restriction.Restriction.FspBI.size
Bio.Restriction.Restriction.FspEI.size
Bio.Restriction.Restriction.FspI.size
Bio.Restriction.Restriction.GauT27I.size
Bio.Restriction.Restriction.GdiII.size
Bio.Restriction.Restriction.GlaI.size
Bio.Restriction.Restriction.GluI.size
Bio.Restriction.Restriction.GsaI.size
Bio.Restriction.Restriction.GsuI.size
Bio.Restriction.Restriction.HaeI.size
Bio.Restriction.Restriction.HaeII.size
Bio.Restriction.Restriction.HaeIII.size
Bio.Restriction.Restriction.HapII.size
Bio.Restriction.Restriction.HauII.size
Bio.Restriction.Restriction.HgaI.size
Bio.Restriction.Restriction.HgiAI.size
Bio.Restriction.Restriction.HgiCI.size
Bio.Restriction.Restriction.HgiEII.size
Bio.Restriction.Restriction.HgiJII.size
Bio.Restriction.Restriction.HhaI.size
Bio.Restriction.Restriction.Hin1I.size
Bio.Restriction.Restriction.Hin1II.size
Bio.Restriction.Restriction.Hin4I.size
Bio.Restriction.Restriction.Hin4II.size
Bio.Restriction.Restriction.Hin6I.size
Bio.Restriction.Restriction.HinP1I.size
Bio.Restriction.Restriction.HincII.size
Bio.Restriction.Restriction.HindII.size
Bio.Restriction.Restriction.HindIII.size
Bio.Restriction.Restriction.HinfI.size
Bio.Restriction.Restriction.HpaI.size
Bio.Restriction.Restriction.HpaII.size
Bio.Restriction.Restriction.HphI.size
Bio.Restriction.Restriction.Hpy166II.size
Bio.Restriction.Restriction.Hpy178III.size
Bio.Restriction.Restriction.Hpy188I.size
Bio.Restriction.Restriction.Hpy188III.size
Bio.Restriction.Restriction.Hpy8I.size
Bio.Restriction.Restriction.Hpy99I.size
Bio.Restriction.Restriction.Hpy99XIII.size
Bio.Restriction.Restriction.Hpy99XIV.size
Bio.Restriction.Restriction.HpyAV.size
Bio.Restriction.Restriction.HpyCH4III.size
Bio.Restriction.Restriction.HpyCH4IV.size
Bio.Restriction.Restriction.HpyCH4V.size
Bio.Restriction.Restriction.HpyF10VI.size
Bio.Restriction.Restriction.HpyF3I.size
Bio.Restriction.Restriction.HpySE526I.size
Bio.Restriction.Restriction.Hsp92I.size
Bio.Restriction.Restriction.Hsp92II.size
Bio.Restriction.Restriction.HspAI.size
Bio.Restriction.Restriction.Jma19592I.size
Bio.Restriction.Restriction.KasI.size
Bio.Restriction.Restriction.KflI.size
Bio.Restriction.Restriction.Kpn2I.size
Bio.Restriction.Restriction.KpnI.size
Bio.Restriction.Restriction.KroI.size
Bio.Restriction.Restriction.Ksp22I.size
Bio.Restriction.Restriction.Ksp632I.size
Bio.Restriction.Restriction.KspAI.size
Bio.Restriction.Restriction.KspI.size
Bio.Restriction.Restriction.Kzo9I.size
Bio.Restriction.Restriction.LguI.size
Bio.Restriction.Restriction.LpnI.size
Bio.Restriction.Restriction.LpnPI.size
Bio.Restriction.Restriction.Lsp1109I.size
Bio.Restriction.Restriction.LweI.size
Bio.Restriction.Restriction.MabI.size
Bio.Restriction.Restriction.MaeI.size
Bio.Restriction.Restriction.MaeII.size
Bio.Restriction.Restriction.MaeIII.size
Bio.Restriction.Restriction.MalI.size
Bio.Restriction.Restriction.MaqI.size
Bio.Restriction.Restriction.MauBI.size
Bio.Restriction.Restriction.MbiI.size
Bio.Restriction.Restriction.MboI.size
Bio.Restriction.Restriction.MboII.size
Bio.Restriction.Restriction.McaTI.size
Bio.Restriction.Restriction.McrI.size
Bio.Restriction.Restriction.MfeI.size
Bio.Restriction.Restriction.MflI.size
Bio.Restriction.Restriction.MhlI.size
Bio.Restriction.Restriction.MjaIV.size
Bio.Restriction.Restriction.MkaDII.size
Bio.Restriction.Restriction.MlsI.size
Bio.Restriction.Restriction.MluCI.size
Bio.Restriction.Restriction.MluI.size
Bio.Restriction.Restriction.MluNI.size
Bio.Restriction.Restriction.Mly113I.size
Bio.Restriction.Restriction.MlyI.size
Bio.Restriction.Restriction.MmeI.size
Bio.Restriction.Restriction.MnlI.size
Bio.Restriction.Restriction.Mph1103I.size
Bio.Restriction.Restriction.MreI.size
Bio.Restriction.Restriction.MroI.size
Bio.Restriction.Restriction.MroNI.size
Bio.Restriction.Restriction.MroXI.size
Bio.Restriction.Restriction.MscI.size
Bio.Restriction.Restriction.MseI.size
Bio.Restriction.Restriction.MslI.size
Bio.Restriction.Restriction.Msp20I.size
Bio.Restriction.Restriction.MspA1I.size
Bio.Restriction.Restriction.MspCI.size
Bio.Restriction.Restriction.MspI.size
Bio.Restriction.Restriction.MspJI.size
Bio.Restriction.Restriction.MspR9I.size
Bio.Restriction.Restriction.MssI.size
Bio.Restriction.Restriction.MstI.size
Bio.Restriction.Restriction.MunI.size
Bio.Restriction.Restriction.Mva1269I.size
Bio.Restriction.Restriction.MvaI.size
Bio.Restriction.Restriction.MvnI.size
Bio.Restriction.Restriction.MvrI.size
Bio.Restriction.Restriction.MwoI.size
Bio.Restriction.Restriction.NaeI.size
Bio.Restriction.Restriction.NarI.size
Bio.Restriction.Restriction.NciI.size
Bio.Restriction.Restriction.NcoI.size
Bio.Restriction.Restriction.NdeI.size
Bio.Restriction.Restriction.NdeII.size
Bio.Restriction.Restriction.NgoAVIII.size
Bio.Restriction.Restriction.NgoMIV.size
Bio.Restriction.Restriction.NhaXI.size
Bio.Restriction.Restriction.NheI.size
Bio.Restriction.Restriction.NlaCI.size
Bio.Restriction.Restriction.NlaIII.size
Bio.Restriction.Restriction.NlaIV.size
Bio.Restriction.Restriction.Nli3877I.size
Bio.Restriction.Restriction.NmeAIII.size
Bio.Restriction.Restriction.NmeDI.size
Bio.Restriction.Restriction.NmuCI.size
Bio.Restriction.Restriction.NotI.size
Bio.Restriction.Restriction.NruI.size
Bio.Restriction.Restriction.NsbI.size
Bio.Restriction.Restriction.NsiI.size
Bio.Restriction.Restriction.NspBII.size
Bio.Restriction.Restriction.NspI.size
Bio.Restriction.Restriction.NspV.size
Bio.Restriction.Restriction.OliI.size
Bio.Restriction.Restriction.PabI.size
Bio.Restriction.Restriction.PacI.size
Bio.Restriction.Restriction.PaeI.size
Bio.Restriction.Restriction.PaeR7I.size
Bio.Restriction.Restriction.PagI.size
Bio.Restriction.Restriction.PalAI.size
Bio.Restriction.Restriction.PasI.size
Bio.Restriction.Restriction.PauI.size
Bio.Restriction.Restriction.PceI.size
Bio.Restriction.Restriction.PciI.size
Bio.Restriction.Restriction.PciSI.size
Bio.Restriction.Restriction.PcsI.size
Bio.Restriction.Restriction.PctI.size
Bio.Restriction.Restriction.PdiI.size
Bio.Restriction.Restriction.PdmI.size
Bio.Restriction.Restriction.PenI.size
Bio.Restriction.Restriction.PfeI.size
Bio.Restriction.Restriction.Pfl1108I.size
Bio.Restriction.Restriction.Pfl23II.size
Bio.Restriction.Restriction.PflFI.size
Bio.Restriction.Restriction.PflMI.size
Bio.Restriction.Restriction.PfoI.size
Bio.Restriction.Restriction.PinAI.size
Bio.Restriction.Restriction.PlaDI.size
Bio.Restriction.Restriction.Ple19I.size
Bio.Restriction.Restriction.PleI.size
Bio.Restriction.Restriction.PluTI.size
Bio.Restriction.Restriction.PmaCI.size
Bio.Restriction.Restriction.PmeI.size
Bio.Restriction.Restriction.PmlI.size
Bio.Restriction.Restriction.PpiI.size
Bio.Restriction.Restriction.PpsI.size
Bio.Restriction.Restriction.Ppu10I.size
Bio.Restriction.Restriction.Ppu21I.size
Bio.Restriction.Restriction.PpuMI.size
Bio.Restriction.Restriction.PscI.size
Bio.Restriction.Restriction.PshAI.size
Bio.Restriction.Restriction.PshBI.size
Bio.Restriction.Restriction.PsiI.size
Bio.Restriction.Restriction.Psp03I.size
Bio.Restriction.Restriction.Psp124BI.size
Bio.Restriction.Restriction.Psp1406I.size
Bio.Restriction.Restriction.Psp5II.size
Bio.Restriction.Restriction.Psp6I.size
Bio.Restriction.Restriction.PspCI.size
Bio.Restriction.Restriction.PspEI.size
Bio.Restriction.Restriction.PspGI.size
Bio.Restriction.Restriction.PspLI.size
Bio.Restriction.Restriction.PspN4I.size
Bio.Restriction.Restriction.PspOMI.size
Bio.Restriction.Restriction.PspOMII.size
Bio.Restriction.Restriction.PspPI.size
Bio.Restriction.Restriction.PspPPI.size
Bio.Restriction.Restriction.PspPRI.size
Bio.Restriction.Restriction.PspXI.size
Bio.Restriction.Restriction.PsrI.size
Bio.Restriction.Restriction.PssI.size
Bio.Restriction.Restriction.PstI.size
Bio.Restriction.Restriction.PstNI.size
Bio.Restriction.Restriction.PsuI.size
Bio.Restriction.Restriction.PsyI.size
Bio.Restriction.Restriction.PteI.size
Bio.Restriction.Restriction.PvuI.size
Bio.Restriction.Restriction.PvuII.size
Bio.Restriction.Restriction.R2_BceSIV.size
Bio.Restriction.Restriction.RceI.size
Bio.Restriction.Restriction.RdeGBI.size
Bio.Restriction.Restriction.RdeGBII.size
Bio.Restriction.Restriction.RdeGBIII.size
Bio.Restriction.Restriction.RflFIII.size
Bio.Restriction.Restriction.RgaI.size
Bio.Restriction.Restriction.RigI.size
Bio.Restriction.Restriction.RlaI.size
Bio.Restriction.Restriction.RleAI.size
Bio.Restriction.Restriction.RpaB5I.size
Bio.Restriction.Restriction.RpaBI.size
Bio.Restriction.Restriction.RpaI.size
Bio.Restriction.Restriction.RpaTI.size
Bio.Restriction.Restriction.RruI.size
Bio.Restriction.Restriction.RsaI.size
Bio.Restriction.Restriction.RsaNI.size
Bio.Restriction.Restriction.RseI.size
Bio.Restriction.Restriction.Rsr2I.size
Bio.Restriction.Restriction.RsrII.size
Bio.Restriction.Restriction.SacI.size
Bio.Restriction.Restriction.SacII.size
Bio.Restriction.Restriction.SalI.size
Bio.Restriction.Restriction.SanDI.size
Bio.Restriction.Restriction.SapI.size
Bio.Restriction.Restriction.SaqAI.size
Bio.Restriction.Restriction.SatI.size
Bio.Restriction.Restriction.Sau3AI.size
Bio.Restriction.Restriction.Sau96I.size
Bio.Restriction.Restriction.SauI.size
Bio.Restriction.Restriction.SbfI.size
Bio.Restriction.Restriction.ScaI.size
Bio.Restriction.Restriction.SchI.size
Bio.Restriction.Restriction.SciI.size
Bio.Restriction.Restriction.ScrFI.size
Bio.Restriction.Restriction.SdaI.size
Bio.Restriction.Restriction.SdeAI.size
Bio.Restriction.Restriction.SdeOSI.size
Bio.Restriction.Restriction.SduI.size
Bio.Restriction.Restriction.SecI.size
Bio.Restriction.Restriction.SelI.size
Bio.Restriction.Restriction.SetI.size
Bio.Restriction.Restriction.SexAI.size
Bio.Restriction.Restriction.SfaAI.size
Bio.Restriction.Restriction.SfaNI.size
Bio.Restriction.Restriction.SfcI.size
Bio.Restriction.Restriction.SfeI.size
Bio.Restriction.Restriction.SfiI.size
Bio.Restriction.Restriction.SfoI.size
Bio.Restriction.Restriction.Sfr274I.size
Bio.Restriction.Restriction.Sfr303I.size
Bio.Restriction.Restriction.SfuI.size
Bio.Restriction.Restriction.SgeI.size
Bio.Restriction.Restriction.SgfI.size
Bio.Restriction.Restriction.SgrAI.size
Bio.Restriction.Restriction.SgrBI.size
Bio.Restriction.Restriction.SgrDI.size
Bio.Restriction.Restriction.SgrTI.size
Bio.Restriction.Restriction.SgsI.size
Bio.Restriction.Restriction.SimI.size
Bio.Restriction.Restriction.SlaI.size
Bio.Restriction.Restriction.SmaI.size
Bio.Restriction.Restriction.SmiI.size
Bio.Restriction.Restriction.SmiMI.size
Bio.Restriction.Restriction.SmlI.size
Bio.Restriction.Restriction.SmoI.size
Bio.Restriction.Restriction.SnaBI.size
Bio.Restriction.Restriction.SnaI.size
Bio.Restriction.Restriction.Sno506I.size
Bio.Restriction.Restriction.SpeI.size
Bio.Restriction.Restriction.SphI.size
Bio.Restriction.Restriction.SplI.size
Bio.Restriction.Restriction.SpoDI.size
Bio.Restriction.Restriction.SrfI.size
Bio.Restriction.Restriction.Sse232I.size
Bio.Restriction.Restriction.Sse8387I.size
Bio.Restriction.Restriction.Sse8647I.size
Bio.Restriction.Restriction.Sse9I.size
Bio.Restriction.Restriction.SseBI.size
Bio.Restriction.Restriction.SsiI.size
Bio.Restriction.Restriction.SspD5I.size
Bio.Restriction.Restriction.SspDI.size
Bio.Restriction.Restriction.SspI.size
Bio.Restriction.Restriction.SstE37I.size
Bio.Restriction.Restriction.SstI.size
Bio.Restriction.Restriction.Sth132I.size
Bio.Restriction.Restriction.Sth302II.size
Bio.Restriction.Restriction.StrI.size
Bio.Restriction.Restriction.StsI.size
Bio.Restriction.Restriction.StuI.size
Bio.Restriction.Restriction.StyD4I.size
Bio.Restriction.Restriction.StyI.size
Bio.Restriction.Restriction.SwaI.size
Bio.Restriction.Restriction.TaaI.size
Bio.Restriction.Restriction.TaiI.size
Bio.Restriction.Restriction.TaqI.size
Bio.Restriction.Restriction.TaqII.size
Bio.Restriction.Restriction.TasI.size
Bio.Restriction.Restriction.TatI.size
Bio.Restriction.Restriction.TauI.size
Bio.Restriction.Restriction.TfiI.size
Bio.Restriction.Restriction.Tru1I.size
Bio.Restriction.Restriction.Tru9I.size
Bio.Restriction.Restriction.TscAI.size
Bio.Restriction.Restriction.TseFI.size
Bio.Restriction.Restriction.TseI.size
Bio.Restriction.Restriction.TsoI.size
Bio.Restriction.Restriction.Tsp45I.size
Bio.Restriction.Restriction.Tsp4CI.size
Bio.Restriction.Restriction.TspDTI.size
Bio.Restriction.Restriction.TspEI.size
Bio.Restriction.Restriction.TspGWI.size
Bio.Restriction.Restriction.TspMI.size
Bio.Restriction.Restriction.TspRI.size
Bio.Restriction.Restriction.TssI.size
Bio.Restriction.Restriction.TstI.size
Bio.Restriction.Restriction.TsuI.size
Bio.Restriction.Restriction.Tth111I.size
Bio.Restriction.Restriction.Tth111II.size
Bio.Restriction.Restriction.UbaF11I.size
Bio.Restriction.Restriction.UbaF12I.size
Bio.Restriction.Restriction.UbaF13I.size
Bio.Restriction.Restriction.UbaF14I.size
Bio.Restriction.Restriction.UbaF9I.size
Bio.Restriction.Restriction.UbaPI.size
Bio.Restriction.Restriction.UcoMSI.size
Bio.Restriction.Restriction.UnbI.size
Bio.Restriction.Restriction.Van91I.size
Bio.Restriction.Restriction.Vha464I.size
Bio.Restriction.Restriction.VneI.size
Bio.Restriction.Restriction.VpaK11AI.size
Bio.Restriction.Restriction.VpaK11BI.size
Bio.Restriction.Restriction.VspI.size
Bio.Restriction.Restriction.WviI.size
Bio.Restriction.Restriction.XagI.size
Bio.Restriction.Restriction.XapI.size
Bio.Restriction.Restriction.XbaI.size
Bio.Restriction.Restriction.XceI.size
Bio.Restriction.Restriction.XcmI.size
Bio.Restriction.Restriction.XhoI.size
Bio.Restriction.Restriction.XhoII.size
Bio.Restriction.Restriction.XmaI.size
Bio.Restriction.Restriction.XmaIII.size
Bio.Restriction.Restriction.XmaJI.size
Bio.Restriction.Restriction.XmiI.size
Bio.Restriction.Restriction.XmnI.size
Bio.Restriction.Restriction.XspI.size
Bio.Restriction.Restriction.YkrI.size
Bio.Restriction.Restriction.ZraI.size
Bio.Restriction.Restriction.ZrmI.size
Bio.Restriction.Restriction.Zsp2I.size
Bio.codonalign.codonalphabet.CodonAlphabet.size" class="py-name" href="#" onclick="return doclink('link-500', 'size', 'link-270');">size</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-501" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-501', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1239"></a><tt class="py-lineno">1239</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1240"></a><tt class="py-lineno">1240</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">'Did not recognise the LOCUS line layout:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1241"></a><tt class="py-lineno">1241</tt>  <tt class="py-line"> </tt>
<a name="GenBankScanner._feed_header_lines"></a><div id="GenBankScanner._feed_header_lines-def"><a name="L1242"></a><tt class="py-lineno">1242</tt> <a class="py-toggle" href="#" id="GenBankScanner._feed_header_lines-toggle" onclick="return toggle('GenBankScanner._feed_header_lines');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.GenBankScanner-class.html#_feed_header_lines">_feed_header_lines</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">,</tt> <tt class="py-param">lines</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GenBankScanner._feed_header_lines-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="GenBankScanner._feed_header_lines-expanded"><a name="L1243"></a><tt class="py-lineno">1243</tt>  <tt class="py-line">        <tt class="py-comment"># Following dictionary maps GenBank lines to the associated</tt> </tt>
<a name="L1244"></a><tt class="py-lineno">1244</tt>  <tt class="py-line">        <tt class="py-comment"># consumer methods - the special cases like LOCUS where one</tt> </tt>
<a name="L1245"></a><tt class="py-lineno">1245</tt>  <tt class="py-line">        <tt class="py-comment"># genbank line triggers several consumer calls have to be</tt> </tt>
<a name="L1246"></a><tt class="py-lineno">1246</tt>  <tt class="py-line">        <tt class="py-comment"># handled individually.</tt> </tt>
<a name="L1247"></a><tt class="py-lineno">1247</tt>  <tt class="py-line">        <tt id="link-502" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-502', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-503" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-503', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt> </tt>
<a name="L1248"></a><tt class="py-lineno">1248</tt>  <tt class="py-line">        <tt id="link-504" class="py-name"><a title="Bio.GenBank.GENBANK_SPACER" class="py-name" href="#" onclick="return doclink('link-504', 'GENBANK_SPACER', 'link-424');">GENBANK_SPACER</a></tt> <tt class="py-op">=</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt id="link-505" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-505', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt> </tt>
<a name="L1249"></a><tt class="py-lineno">1249</tt>  <tt class="py-line">        <tt class="py-name">consumer_dict</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L1250"></a><tt class="py-lineno">1250</tt>  <tt class="py-line">            <tt class="py-string">'DEFINITION'</tt><tt class="py-op">:</tt> <tt class="py-string">'definition'</tt><tt class="py-op">,</tt> </tt>
<a name="L1251"></a><tt class="py-lineno">1251</tt>  <tt class="py-line">            <tt class="py-string">'ACCESSION'</tt><tt class="py-op">:</tt> <tt class="py-string">'accession'</tt><tt class="py-op">,</tt> </tt>
<a name="L1252"></a><tt class="py-lineno">1252</tt>  <tt class="py-line">            <tt class="py-string">'NID'</tt><tt class="py-op">:</tt> <tt class="py-string">'nid'</tt><tt class="py-op">,</tt> </tt>
<a name="L1253"></a><tt class="py-lineno">1253</tt>  <tt class="py-line">            <tt class="py-string">'PID'</tt><tt class="py-op">:</tt> <tt class="py-string">'pid'</tt><tt class="py-op">,</tt> </tt>
<a name="L1254"></a><tt class="py-lineno">1254</tt>  <tt class="py-line">            <tt class="py-string">'DBSOURCE'</tt><tt class="py-op">:</tt> <tt class="py-string">'db_source'</tt><tt class="py-op">,</tt> </tt>
<a name="L1255"></a><tt class="py-lineno">1255</tt>  <tt class="py-line">            <tt class="py-string">'KEYWORDS'</tt><tt class="py-op">:</tt> <tt class="py-string">'keywords'</tt><tt class="py-op">,</tt> </tt>
<a name="L1256"></a><tt class="py-lineno">1256</tt>  <tt class="py-line">            <tt class="py-string">'SEGMENT'</tt><tt class="py-op">:</tt> <tt class="py-string">'segment'</tt><tt class="py-op">,</tt> </tt>
<a name="L1257"></a><tt class="py-lineno">1257</tt>  <tt class="py-line">            <tt class="py-string">'SOURCE'</tt><tt class="py-op">:</tt> <tt class="py-string">'source'</tt><tt class="py-op">,</tt> </tt>
<a name="L1258"></a><tt class="py-lineno">1258</tt>  <tt class="py-line">            <tt class="py-string">'AUTHORS'</tt><tt class="py-op">:</tt> <tt class="py-string">'authors'</tt><tt class="py-op">,</tt> </tt>
<a name="L1259"></a><tt class="py-lineno">1259</tt>  <tt class="py-line">            <tt class="py-string">'CONSRTM'</tt><tt class="py-op">:</tt> <tt class="py-string">'consrtm'</tt><tt class="py-op">,</tt> </tt>
<a name="L1260"></a><tt class="py-lineno">1260</tt>  <tt class="py-line">            <tt class="py-string">'PROJECT'</tt><tt class="py-op">:</tt> <tt class="py-string">'project'</tt><tt class="py-op">,</tt> </tt>
<a name="L1261"></a><tt class="py-lineno">1261</tt>  <tt class="py-line">            <tt class="py-string">'DBLINK'</tt><tt class="py-op">:</tt> <tt class="py-string">'dblink'</tt><tt class="py-op">,</tt> </tt>
<a name="L1262"></a><tt class="py-lineno">1262</tt>  <tt class="py-line">            <tt class="py-string">'TITLE'</tt><tt class="py-op">:</tt> <tt class="py-string">'title'</tt><tt class="py-op">,</tt> </tt>
<a name="L1263"></a><tt class="py-lineno">1263</tt>  <tt class="py-line">            <tt class="py-string">'JOURNAL'</tt><tt class="py-op">:</tt> <tt class="py-string">'journal'</tt><tt class="py-op">,</tt> </tt>
<a name="L1264"></a><tt class="py-lineno">1264</tt>  <tt class="py-line">            <tt class="py-string">'MEDLINE'</tt><tt class="py-op">:</tt> <tt class="py-string">'medline_id'</tt><tt class="py-op">,</tt> </tt>
<a name="L1265"></a><tt class="py-lineno">1265</tt>  <tt class="py-line">            <tt class="py-string">'PUBMED'</tt><tt class="py-op">:</tt> <tt class="py-string">'pubmed_id'</tt><tt class="py-op">,</tt> </tt>
<a name="L1266"></a><tt class="py-lineno">1266</tt>  <tt class="py-line">            <tt class="py-string">'REMARK'</tt><tt class="py-op">:</tt> <tt class="py-string">'remark'</tt><tt class="py-op">}</tt> </tt>
<a name="L1267"></a><tt class="py-lineno">1267</tt>  <tt class="py-line">        <tt class="py-comment"># We have to handle the following specially:</tt> </tt>
<a name="L1268"></a><tt class="py-lineno">1268</tt>  <tt class="py-line">        <tt class="py-comment"># ORIGIN (locus, size, residue_type, data_file_division and date)</tt> </tt>
<a name="L1269"></a><tt class="py-lineno">1269</tt>  <tt class="py-line">        <tt class="py-comment"># COMMENT (comment)</tt> </tt>
<a name="L1270"></a><tt class="py-lineno">1270</tt>  <tt class="py-line">        <tt class="py-comment"># VERSION (version and gi)</tt> </tt>
<a name="L1271"></a><tt class="py-lineno">1271</tt>  <tt class="py-line">        <tt class="py-comment"># REFERENCE (eference_num and reference_bases)</tt> </tt>
<a name="L1272"></a><tt class="py-lineno">1272</tt>  <tt class="py-line">        <tt class="py-comment"># ORGANISM (organism and taxonomy)</tt> </tt>
<a name="L1273"></a><tt class="py-lineno">1273</tt>  <tt class="py-line">        <tt class="py-name">lines</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">_f</tt> <tt class="py-keyword">for</tt> <tt class="py-name">_f</tt> <tt class="py-keyword">in</tt> <tt class="py-name">lines</tt> <tt class="py-keyword">if</tt> <tt class="py-name">_f</tt><tt class="py-op">]</tt> </tt>
<a name="L1274"></a><tt class="py-lineno">1274</tt>  <tt class="py-line">        <tt class="py-name">lines</tt><tt class="py-op">.</tt><tt id="link-506" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-506', 'append', 'link-36');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">)</tt>  <tt class="py-comment"># helps avoid getting StopIteration all the time</tt> </tt>
<a name="L1275"></a><tt class="py-lineno">1275</tt>  <tt class="py-line">        <tt class="py-name">line_iter</tt> <tt class="py-op">=</tt> <tt class="py-name">iter</tt><tt class="py-op">(</tt><tt class="py-name">lines</tt><tt class="py-op">)</tt> </tt>
<a name="L1276"></a><tt class="py-lineno">1276</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1277"></a><tt class="py-lineno">1277</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-507" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-507', 'next', 'link-86');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">line_iter</tt><tt class="py-op">)</tt> </tt>
<a name="L1278"></a><tt class="py-lineno">1278</tt>  <tt class="py-line">            <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L1279"></a><tt class="py-lineno">1279</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L1280"></a><tt class="py-lineno">1280</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L1281"></a><tt class="py-lineno">1281</tt>  <tt class="py-line">                <tt class="py-name">line_type</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt id="link-508" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-508', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-509" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-509', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1282"></a><tt class="py-lineno">1282</tt>  <tt class="py-line">                <tt id="link-510" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-510', 'data', 'link-143');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt id="link-511" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-511', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-512" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-512', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1283"></a><tt class="py-lineno">1283</tt>  <tt class="py-line"> </tt>
<a name="L1284"></a><tt class="py-lineno">1284</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">line_type</tt> <tt class="py-op">==</tt> <tt class="py-string">'VERSION'</tt><tt class="py-op">:</tt> </tt>
<a name="L1285"></a><tt class="py-lineno">1285</tt>  <tt class="py-line">                    <tt class="py-comment"># Need to call consumer.version(), and maybe also consumer.gi() as well.</tt> </tt>
<a name="L1286"></a><tt class="py-lineno">1286</tt>  <tt class="py-line">                    <tt class="py-comment"># e.g.</tt> </tt>
<a name="L1287"></a><tt class="py-lineno">1287</tt>  <tt class="py-line">                    <tt class="py-comment"># VERSION     AC007323.5  GI:6587720</tt> </tt>
<a name="L1288"></a><tt class="py-lineno">1288</tt>  <tt class="py-line">                    <tt class="py-keyword">while</tt> <tt class="py-string">'  '</tt> <tt class="py-keyword">in</tt> <tt id="link-513" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-513', 'data', 'link-143');">data</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1289"></a><tt class="py-lineno">1289</tt>  <tt class="py-line">                        <tt id="link-514" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-514', 'data', 'link-143');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-515" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-515', 'data', 'link-143');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'  '</tt><tt class="py-op">,</tt> <tt class="py-string">' '</tt><tt class="py-op">)</tt> </tt>
<a name="L1290"></a><tt class="py-lineno">1290</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-string">' GI:'</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt id="link-516" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-516', 'data', 'link-143');">data</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1291"></a><tt class="py-lineno">1291</tt>  <tt class="py-line">                        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-517" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HeaderConsumer.version()=Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#version,Function Bio.Cluster.cluster.version()=Bio.Cluster.cluster-module.html#version,Method Bio.GenBank._FeatureConsumer.version()=Bio.GenBank._FeatureConsumer-class.html#version,Method Bio.GenBank._RecordConsumer.version()=Bio.GenBank._RecordConsumer-class.html#version,Variable Bio.motifs.jaspar.Motif.version=Bio.motifs.jaspar.Motif-class.html#version"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-517', 'version', 'link-517');">version</a></tt><tt class="py-op">(</tt><tt id="link-518" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-518', 'data', 'link-143');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1292"></a><tt class="py-lineno">1292</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1293"></a><tt class="py-lineno">1293</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt><tt class="py-op">:</tt> </tt>
<a name="L1294"></a><tt class="py-lineno">1294</tt>  <tt class="py-line">                            <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Version ["</tt> <tt class="py-op">+</tt> <tt id="link-519" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-519', 'data', 'link-143');">data</a></tt><tt class="py-op">.</tt><tt id="link-520" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-520', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">' GI:'</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">"], gi ["</tt> <tt class="py-op">+</tt> <tt id="link-521" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-521', 'data', 'link-143');">data</a></tt><tt class="py-op">.</tt><tt id="link-522" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-522', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">' GI:'</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">"]"</tt><tt class="py-op">)</tt> </tt>
<a name="L1295"></a><tt class="py-lineno">1295</tt>  <tt class="py-line">                        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-523" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-523', 'version', 'link-517');">version</a></tt><tt class="py-op">(</tt><tt id="link-524" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-524', 'data', 'link-143');">data</a></tt><tt class="py-op">.</tt><tt id="link-525" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-525', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">' GI:'</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1296"></a><tt class="py-lineno">1296</tt>  <tt class="py-line">                        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-526" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.gi()=Bio.GenBank._FeatureConsumer-class.html#gi,Method Bio.GenBank._RecordConsumer.gi()=Bio.GenBank._RecordConsumer-class.html#gi"><a title="Bio.GenBank._FeatureConsumer.gi
Bio.GenBank._RecordConsumer.gi" class="py-name" href="#" onclick="return doclink('link-526', 'gi', 'link-526');">gi</a></tt><tt class="py-op">(</tt><tt id="link-527" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-527', 'data', 'link-143');">data</a></tt><tt class="py-op">.</tt><tt id="link-528" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-528', 'split', 'link-63');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">' GI:'</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1297"></a><tt class="py-lineno">1297</tt>  <tt class="py-line">                    <tt class="py-comment"># Read in the next line!</tt> </tt>
<a name="L1298"></a><tt class="py-lineno">1298</tt>  <tt class="py-line">                    <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-529" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-529', 'next', 'link-86');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">line_iter</tt><tt class="py-op">)</tt> </tt>
<a name="L1299"></a><tt class="py-lineno">1299</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">line_type</tt> <tt class="py-op">==</tt> <tt class="py-string">'REFERENCE'</tt><tt class="py-op">:</tt> </tt>
<a name="L1300"></a><tt class="py-lineno">1300</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L1301"></a><tt class="py-lineno">1301</tt>  <tt class="py-line">                        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Found reference ["</tt> <tt class="py-op">+</tt> <tt id="link-530" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-530', 'data', 'link-143');">data</a></tt> <tt class="py-op">+</tt> <tt class="py-string">"]"</tt><tt class="py-op">)</tt> </tt>
<a name="L1302"></a><tt class="py-lineno">1302</tt>  <tt class="py-line">                    <tt class="py-comment"># Need to call consumer.reference_num() and consumer.reference_bases()</tt> </tt>
<a name="L1303"></a><tt class="py-lineno">1303</tt>  <tt class="py-line">                    <tt class="py-comment"># e.g.</tt> </tt>
<a name="L1304"></a><tt class="py-lineno">1304</tt>  <tt class="py-line">                    <tt class="py-comment"># REFERENCE   1  (bases 1 to 86436)</tt> </tt>
<a name="L1305"></a><tt class="py-lineno">1305</tt>  <tt class="py-line">                    <tt class="py-comment">#</tt> </tt>
<a name="L1306"></a><tt class="py-lineno">1306</tt>  <tt class="py-line">                    <tt class="py-comment"># Note that this can be multiline, see Bug 1968, e.g.</tt> </tt>
<a name="L1307"></a><tt class="py-lineno">1307</tt>  <tt class="py-line">                    <tt class="py-comment">#</tt> </tt>
<a name="L1308"></a><tt class="py-lineno">1308</tt>  <tt class="py-line">                    <tt class="py-comment"># REFERENCE   42 (bases 1517 to 1696; 3932 to 4112; 17880 to 17975; 21142 to</tt> </tt>
<a name="L1309"></a><tt class="py-lineno">1309</tt>  <tt class="py-line">                    <tt class="py-comment">#             28259)</tt> </tt>
<a name="L1310"></a><tt class="py-lineno">1310</tt>  <tt class="py-line">                    <tt class="py-comment">#</tt> </tt>
<a name="L1311"></a><tt class="py-lineno">1311</tt>  <tt class="py-line">                    <tt class="py-comment"># For such cases we will call the consumer once only.</tt> </tt>
<a name="L1312"></a><tt class="py-lineno">1312</tt>  <tt class="py-line">                    <tt id="link-531" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-531', 'data', 'link-143');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-532" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-532', 'data', 'link-143');">data</a></tt><tt class="py-op">.</tt><tt id="link-533" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-533', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1313"></a><tt class="py-lineno">1313</tt>  <tt class="py-line"> </tt>
<a name="L1314"></a><tt class="py-lineno">1314</tt>  <tt class="py-line">                    <tt class="py-comment"># Read in the next line, and see if its more of the reference:</tt> </tt>
<a name="L1315"></a><tt class="py-lineno">1315</tt>  <tt class="py-line">                    <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L1316"></a><tt class="py-lineno">1316</tt>  <tt class="py-line">                        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-534" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-534', 'next', 'link-86');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">line_iter</tt><tt class="py-op">)</tt> </tt>
<a name="L1317"></a><tt class="py-lineno">1317</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt id="link-535" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-535', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt id="link-536" class="py-name"><a title="Bio.GenBank.GENBANK_SPACER" class="py-name" href="#" onclick="return doclink('link-536', 'GENBANK_SPACER', 'link-424');">GENBANK_SPACER</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1318"></a><tt class="py-lineno">1318</tt>  <tt class="py-line">                            <tt class="py-comment"># Add this continuation to the data string</tt> </tt>
<a name="L1319"></a><tt class="py-lineno">1319</tt>  <tt class="py-line">                            <tt id="link-537" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-537', 'data', 'link-143');">data</a></tt> <tt class="py-op">+=</tt> <tt class="py-string">" "</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt id="link-538" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-538', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L1320"></a><tt class="py-lineno">1320</tt>  <tt class="py-line">                            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L1321"></a><tt class="py-lineno">1321</tt>  <tt class="py-line">                                <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Extended reference text ["</tt> <tt class="py-op">+</tt> <tt id="link-539" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-539', 'data', 'link-143');">data</a></tt> <tt class="py-op">+</tt> <tt class="py-string">"]"</tt><tt class="py-op">)</tt> </tt>
<a name="L1322"></a><tt class="py-lineno">1322</tt>  <tt class="py-line">                        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1323"></a><tt class="py-lineno">1323</tt>  <tt class="py-line">                            <tt class="py-comment"># End of the reference, leave this text in the variable "line"</tt> </tt>
<a name="L1324"></a><tt class="py-lineno">1324</tt>  <tt class="py-line">                            <tt class="py-keyword">break</tt> </tt>
<a name="L1325"></a><tt class="py-lineno">1325</tt>  <tt class="py-line"> </tt>
<a name="L1326"></a><tt class="py-lineno">1326</tt>  <tt class="py-line">                    <tt class="py-comment"># We now have all the reference line(s) stored in a string, data,</tt> </tt>
<a name="L1327"></a><tt class="py-lineno">1327</tt>  <tt class="py-line">                    <tt class="py-comment"># which we pass to the consumer</tt> </tt>
<a name="L1328"></a><tt class="py-lineno">1328</tt>  <tt class="py-line">                    <tt class="py-keyword">while</tt> <tt class="py-string">'  '</tt> <tt class="py-keyword">in</tt> <tt id="link-540" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-540', 'data', 'link-143');">data</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1329"></a><tt class="py-lineno">1329</tt>  <tt class="py-line">                        <tt id="link-541" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-541', 'data', 'link-143');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-542" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-542', 'data', 'link-143');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'  '</tt><tt class="py-op">,</tt> <tt class="py-string">' '</tt><tt class="py-op">)</tt> </tt>
<a name="L1330"></a><tt class="py-lineno">1330</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-string">' '</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt id="link-543" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-543', 'data', 'link-143');">data</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1331"></a><tt class="py-lineno">1331</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L1332"></a><tt class="py-lineno">1332</tt>  <tt class="py-line">                            <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">'Reference number \"'</tt> <tt class="py-op">+</tt> <tt id="link-544" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-544', 'data', 'link-143');">data</a></tt> <tt class="py-op">+</tt> <tt class="py-string">'\"'</tt><tt class="py-op">)</tt> </tt>
<a name="L1333"></a><tt class="py-lineno">1333</tt>  <tt class="py-line">                        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-545" class="py-name"><a title="Bio.GenBank._FeatureConsumer.reference_num
Bio.GenBank._RecordConsumer.reference_num" class="py-name" href="#" onclick="return doclink('link-545', 'reference_num', 'link-279');">reference_num</a></tt><tt class="py-op">(</tt><tt id="link-546" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-546', 'data', 'link-143');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1334"></a><tt class="py-lineno">1334</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1335"></a><tt class="py-lineno">1335</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L1336"></a><tt class="py-lineno">1336</tt>  <tt class="py-line">                            <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">'Reference number \"'</tt> <tt class="py-op">+</tt> <tt id="link-547" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-547', 'data', 'link-143');">data</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt id="link-548" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-548', 'data', 'link-143');">data</a></tt><tt class="py-op">.</tt><tt id="link-549" class="py-name"><a title="Bio.NeuralNetwork.Gene.Motif.MotifFinder.find
Bio.NeuralNetwork.Gene.Schema.SchemaFinder.find
Bio.NeuralNetwork.Gene.Signature.SignatureFinder.find
Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-549', 'find', 'link-100');">find</a></tt><tt class="py-op">(</tt><tt class="py-string">' '</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">'\", \"'</tt> <tt class="py-op">+</tt> <tt id="link-550" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-550', 'data', 'link-143');">data</a></tt><tt class="py-op">[</tt><tt id="link-551" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-551', 'data', 'link-143');">data</a></tt><tt class="py-op">.</tt><tt id="link-552" class="py-name"><a title="Bio.NeuralNetwork.Gene.Motif.MotifFinder.find
Bio.NeuralNetwork.Gene.Schema.SchemaFinder.find
Bio.NeuralNetwork.Gene.Signature.SignatureFinder.find
Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-552', 'find', 'link-100');">find</a></tt><tt class="py-op">(</tt><tt class="py-string">' '</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">'\"'</tt><tt class="py-op">)</tt> </tt>
<a name="L1337"></a><tt class="py-lineno">1337</tt>  <tt class="py-line">                        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-553" class="py-name"><a title="Bio.GenBank._FeatureConsumer.reference_num
Bio.GenBank._RecordConsumer.reference_num" class="py-name" href="#" onclick="return doclink('link-553', 'reference_num', 'link-279');">reference_num</a></tt><tt class="py-op">(</tt><tt id="link-554" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-554', 'data', 'link-143');">data</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt id="link-555" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-555', 'data', 'link-143');">data</a></tt><tt class="py-op">.</tt><tt id="link-556" class="py-name"><a title="Bio.NeuralNetwork.Gene.Motif.MotifFinder.find
Bio.NeuralNetwork.Gene.Schema.SchemaFinder.find
Bio.NeuralNetwork.Gene.Signature.SignatureFinder.find
Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-556', 'find', 'link-100');">find</a></tt><tt class="py-op">(</tt><tt class="py-string">' '</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1338"></a><tt class="py-lineno">1338</tt>  <tt class="py-line">                        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-557" class="py-name"><a title="Bio.GenBank._FeatureConsumer.reference_bases
Bio.GenBank._RecordConsumer.reference_bases" class="py-name" href="#" onclick="return doclink('link-557', 'reference_bases', 'link-286');">reference_bases</a></tt><tt class="py-op">(</tt><tt id="link-558" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-558', 'data', 'link-143');">data</a></tt><tt class="py-op">[</tt><tt id="link-559" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-559', 'data', 'link-143');">data</a></tt><tt class="py-op">.</tt><tt id="link-560" class="py-name"><a title="Bio.NeuralNetwork.Gene.Motif.MotifFinder.find
Bio.NeuralNetwork.Gene.Schema.SchemaFinder.find
Bio.NeuralNetwork.Gene.Signature.SignatureFinder.find
Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-560', 'find', 'link-100');">find</a></tt><tt class="py-op">(</tt><tt class="py-string">' '</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1339"></a><tt class="py-lineno">1339</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">line_type</tt> <tt class="py-op">==</tt> <tt class="py-string">'ORGANISM'</tt><tt class="py-op">:</tt> </tt>
<a name="L1340"></a><tt class="py-lineno">1340</tt>  <tt class="py-line">                    <tt class="py-comment"># Typically the first line is the organism, and subsequent lines</tt> </tt>
<a name="L1341"></a><tt class="py-lineno">1341</tt>  <tt class="py-line">                    <tt class="py-comment"># are the taxonomy lineage.  However, given longer and longer</tt> </tt>
<a name="L1342"></a><tt class="py-lineno">1342</tt>  <tt class="py-line">                    <tt class="py-comment"># species names (as more and more strains and sub strains get</tt> </tt>
<a name="L1343"></a><tt class="py-lineno">1343</tt>  <tt class="py-line">                    <tt class="py-comment"># sequenced) the oragnism name can now get wrapped onto multiple</tt> </tt>
<a name="L1344"></a><tt class="py-lineno">1344</tt>  <tt class="py-line">                    <tt class="py-comment"># lines.  The NCBI say we have to recognise the lineage line by</tt> </tt>
<a name="L1345"></a><tt class="py-lineno">1345</tt>  <tt class="py-line">                    <tt class="py-comment"># the presence of semi-colon delimited entries.  In the long term,</tt> </tt>
<a name="L1346"></a><tt class="py-lineno">1346</tt>  <tt class="py-line">                    <tt class="py-comment"># they are considering adding a new keyword (e.g. LINEAGE).</tt> </tt>
<a name="L1347"></a><tt class="py-lineno">1347</tt>  <tt class="py-line">                    <tt class="py-comment"># See Bug 2591 for details.</tt> </tt>
<a name="L1348"></a><tt class="py-lineno">1348</tt>  <tt class="py-line">                    <tt class="py-name">organism_data</tt> <tt class="py-op">=</tt> <tt id="link-561" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-561', 'data', 'link-143');">data</a></tt> </tt>
<a name="L1349"></a><tt class="py-lineno">1349</tt>  <tt class="py-line">                    <tt class="py-name">lineage_data</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L1350"></a><tt class="py-lineno">1350</tt>  <tt class="py-line">                    <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L1351"></a><tt class="py-lineno">1351</tt>  <tt class="py-line">                        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-562" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-562', 'next', 'link-86');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">line_iter</tt><tt class="py-op">)</tt> </tt>
<a name="L1352"></a><tt class="py-lineno">1352</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">:</tt><tt id="link-563" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-563', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt id="link-564" class="py-name"><a title="Bio.GenBank.GENBANK_SPACER" class="py-name" href="#" onclick="return doclink('link-564', 'GENBANK_SPACER', 'link-424');">GENBANK_SPACER</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1353"></a><tt class="py-lineno">1353</tt>  <tt class="py-line">                            <tt class="py-keyword">if</tt> <tt class="py-name">lineage_data</tt> <tt class="py-keyword">or</tt> <tt class="py-string">";"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L1354"></a><tt class="py-lineno">1354</tt>  <tt class="py-line">                                <tt class="py-name">lineage_data</tt> <tt class="py-op">+=</tt> <tt class="py-string">" "</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt id="link-565" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-565', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L1355"></a><tt class="py-lineno">1355</tt>  <tt class="py-line">                            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt id="link-566" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-566', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-567" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-567', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">"."</tt><tt class="py-op">:</tt> </tt>
<a name="L1356"></a><tt class="py-lineno">1356</tt>  <tt class="py-line">                                <tt class="py-comment"># No lineage data, just . place holder</tt> </tt>
<a name="L1357"></a><tt class="py-lineno">1357</tt>  <tt class="py-line">                                <tt class="py-keyword">pass</tt> </tt>
<a name="L1358"></a><tt class="py-lineno">1358</tt>  <tt class="py-line">                            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1359"></a><tt class="py-lineno">1359</tt>  <tt class="py-line">                                <tt class="py-name">organism_data</tt> <tt class="py-op">+=</tt> <tt class="py-string">" "</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt id="link-568" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-568', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-569" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-569', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1360"></a><tt class="py-lineno">1360</tt>  <tt class="py-line">                        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1361"></a><tt class="py-lineno">1361</tt>  <tt class="py-line">                            <tt class="py-comment"># End of organism and taxonomy</tt> </tt>
<a name="L1362"></a><tt class="py-lineno">1362</tt>  <tt class="py-line">                            <tt class="py-keyword">break</tt> </tt>
<a name="L1363"></a><tt class="py-lineno">1363</tt>  <tt class="py-line">                    <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-570" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.organism()=Bio.GenBank._FeatureConsumer-class.html#organism,Method Bio.GenBank._RecordConsumer.organism()=Bio.GenBank._RecordConsumer-class.html#organism"><a title="Bio.GenBank._FeatureConsumer.organism
Bio.GenBank._RecordConsumer.organism" class="py-name" href="#" onclick="return doclink('link-570', 'organism', 'link-570');">organism</a></tt><tt class="py-op">(</tt><tt class="py-name">organism_data</tt><tt class="py-op">)</tt> </tt>
<a name="L1364"></a><tt class="py-lineno">1364</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">lineage_data</tt><tt class="py-op">.</tt><tt id="link-571" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-571', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">""</tt> <tt class="py-keyword">and</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L1365"></a><tt class="py-lineno">1365</tt>  <tt class="py-line">                        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Taxonomy line(s) missing or blank"</tt><tt class="py-op">)</tt> </tt>
<a name="L1366"></a><tt class="py-lineno">1366</tt>  <tt class="py-line">                    <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-572" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.taxonomy()=Bio.GenBank._FeatureConsumer-class.html#taxonomy,Method Bio.GenBank._RecordConsumer.taxonomy()=Bio.GenBank._RecordConsumer-class.html#taxonomy,Variable Bio.Phylo.PhyloXML.Clade.taxonomy=Bio.Phylo.PhyloXML.Clade-class.html#taxonomy,Method Bio.Phylo.PhyloXMLIO.Writer.taxonomy()=Bio.Phylo.PhyloXMLIO.Writer-class.html#taxonomy"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-572', 'taxonomy', 'link-572');">taxonomy</a></tt><tt class="py-op">(</tt><tt class="py-name">lineage_data</tt><tt class="py-op">.</tt><tt id="link-573" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-573', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1367"></a><tt class="py-lineno">1367</tt>  <tt class="py-line">                    <tt class="py-keyword">del</tt> <tt class="py-name">organism_data</tt><tt class="py-op">,</tt> <tt class="py-name">lineage_data</tt> </tt>
<a name="L1368"></a><tt class="py-lineno">1368</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">line_type</tt> <tt class="py-op">==</tt> <tt class="py-string">'COMMENT'</tt><tt class="py-op">:</tt> </tt>
<a name="L1369"></a><tt class="py-lineno">1369</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L1370"></a><tt class="py-lineno">1370</tt>  <tt class="py-line">                        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Found comment"</tt><tt class="py-op">)</tt> </tt>
<a name="L1371"></a><tt class="py-lineno">1371</tt>  <tt class="py-line">                    <tt class="py-comment"># This can be multiline, and should call consumer.comment() once</tt> </tt>
<a name="L1372"></a><tt class="py-lineno">1372</tt>  <tt class="py-line">                    <tt class="py-comment"># with a list where each entry is a line.</tt> </tt>
<a name="L1373"></a><tt class="py-lineno">1373</tt>  <tt class="py-line">                    <tt class="py-name">comment_list</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1374"></a><tt class="py-lineno">1374</tt>  <tt class="py-line">                    <tt class="py-name">comment_list</tt><tt class="py-op">.</tt><tt id="link-574" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-574', 'append', 'link-36');">append</a></tt><tt class="py-op">(</tt><tt id="link-575" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-575', 'data', 'link-143');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1375"></a><tt class="py-lineno">1375</tt>  <tt class="py-line">                    <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L1376"></a><tt class="py-lineno">1376</tt>  <tt class="py-line">                        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-576" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-576', 'next', 'link-86');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">line_iter</tt><tt class="py-op">)</tt> </tt>
<a name="L1377"></a><tt class="py-lineno">1377</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">:</tt><tt id="link-577" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-577', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt id="link-578" class="py-name"><a title="Bio.GenBank.GENBANK_SPACER" class="py-name" href="#" onclick="return doclink('link-578', 'GENBANK_SPACER', 'link-424');">GENBANK_SPACER</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1378"></a><tt class="py-lineno">1378</tt>  <tt class="py-line">                            <tt id="link-579" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-579', 'data', 'link-143');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt id="link-580" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-580', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L1379"></a><tt class="py-lineno">1379</tt>  <tt class="py-line">                            <tt class="py-name">comment_list</tt><tt class="py-op">.</tt><tt id="link-581" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-581', 'append', 'link-36');">append</a></tt><tt class="py-op">(</tt><tt id="link-582" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-582', 'data', 'link-143');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1380"></a><tt class="py-lineno">1380</tt>  <tt class="py-line">                            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L1381"></a><tt class="py-lineno">1381</tt>  <tt class="py-line">                                <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Comment continuation ["</tt> <tt class="py-op">+</tt> <tt id="link-583" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-583', 'data', 'link-143');">data</a></tt> <tt class="py-op">+</tt> <tt class="py-string">"]"</tt><tt class="py-op">)</tt> </tt>
<a name="L1382"></a><tt class="py-lineno">1382</tt>  <tt class="py-line">                        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1383"></a><tt class="py-lineno">1383</tt>  <tt class="py-line">                            <tt class="py-comment"># End of the comment</tt> </tt>
<a name="L1384"></a><tt class="py-lineno">1384</tt>  <tt class="py-line">                            <tt class="py-keyword">break</tt> </tt>
<a name="L1385"></a><tt class="py-lineno">1385</tt>  <tt class="py-line">                    <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-584" class="py-name"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-584', 'comment', 'link-310');">comment</a></tt><tt class="py-op">(</tt><tt class="py-name">comment_list</tt><tt class="py-op">)</tt> </tt>
<a name="L1386"></a><tt class="py-lineno">1386</tt>  <tt class="py-line">                    <tt class="py-keyword">del</tt> <tt class="py-name">comment_list</tt> </tt>
<a name="L1387"></a><tt class="py-lineno">1387</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">line_type</tt> <tt class="py-keyword">in</tt> <tt class="py-name">consumer_dict</tt><tt class="py-op">:</tt> </tt>
<a name="L1388"></a><tt class="py-lineno">1388</tt>  <tt class="py-line">                    <tt class="py-comment"># Its a semi-automatic entry!</tt> </tt>
<a name="L1389"></a><tt class="py-lineno">1389</tt>  <tt class="py-line">                    <tt class="py-comment"># Now, this may be a multi line entry...</tt> </tt>
<a name="L1390"></a><tt class="py-lineno">1390</tt>  <tt class="py-line">                    <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L1391"></a><tt class="py-lineno">1391</tt>  <tt class="py-line">                        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-585" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-585', 'next', 'link-86');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">line_iter</tt><tt class="py-op">)</tt> </tt>
<a name="L1392"></a><tt class="py-lineno">1392</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">:</tt><tt id="link-586" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-586', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt id="link-587" class="py-name"><a title="Bio.GenBank.GENBANK_SPACER" class="py-name" href="#" onclick="return doclink('link-587', 'GENBANK_SPACER', 'link-424');">GENBANK_SPACER</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1393"></a><tt class="py-lineno">1393</tt>  <tt class="py-line">                            <tt id="link-588" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-588', 'data', 'link-143');">data</a></tt> <tt class="py-op">+=</tt> <tt class="py-string">' '</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt id="link-589" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-589', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L1394"></a><tt class="py-lineno">1394</tt>  <tt class="py-line">                        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1395"></a><tt class="py-lineno">1395</tt>  <tt class="py-line">                            <tt class="py-comment"># We now have all the data for this entry:</tt> </tt>
<a name="L1396"></a><tt class="py-lineno">1396</tt>  <tt class="py-line">                            <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">consumer</tt><tt class="py-op">,</tt> <tt class="py-name">consumer_dict</tt><tt class="py-op">[</tt><tt class="py-name">line_type</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">(</tt><tt id="link-590" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-590', 'data', 'link-143');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1397"></a><tt class="py-lineno">1397</tt>  <tt class="py-line">                            <tt class="py-comment"># End of continuation - return to top of loop!</tt> </tt>
<a name="L1398"></a><tt class="py-lineno">1398</tt>  <tt class="py-line">                            <tt class="py-keyword">break</tt> </tt>
<a name="L1399"></a><tt class="py-lineno">1399</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1400"></a><tt class="py-lineno">1400</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt><tt class="py-op">:</tt> </tt>
<a name="L1401"></a><tt class="py-lineno">1401</tt>  <tt class="py-line">                        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Ignoring GenBank header line:\n"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1402"></a><tt class="py-lineno">1402</tt>  <tt class="py-line">                    <tt class="py-comment"># Read in next line</tt> </tt>
<a name="L1403"></a><tt class="py-lineno">1403</tt>  <tt class="py-line">                    <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-591" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-591', 'next', 'link-86');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">line_iter</tt><tt class="py-op">)</tt> </tt>
<a name="L1404"></a><tt class="py-lineno">1404</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L1405"></a><tt class="py-lineno">1405</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Problem in header"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1406"></a><tt class="py-lineno">1406</tt>  <tt class="py-line"> </tt>
<a name="GenBankScanner._feed_misc_lines"></a><div id="GenBankScanner._feed_misc_lines-def"><a name="L1407"></a><tt class="py-lineno">1407</tt> <a class="py-toggle" href="#" id="GenBankScanner._feed_misc_lines-toggle" onclick="return toggle('GenBankScanner._feed_misc_lines');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Scanner.GenBankScanner-class.html#_feed_misc_lines">_feed_misc_lines</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">,</tt> <tt class="py-param">lines</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GenBankScanner._feed_misc_lines-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="GenBankScanner._feed_misc_lines-expanded"><a name="L1408"></a><tt class="py-lineno">1408</tt>  <tt class="py-line">        <tt class="py-comment"># Deals with a few misc lines between the features and the sequence</tt> </tt>
<a name="L1409"></a><tt class="py-lineno">1409</tt>  <tt class="py-line">        <tt id="link-592" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-592', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-593" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-593', 'HEADER_WIDTH', 'link-12');">HEADER_WIDTH</a></tt> </tt>
<a name="L1410"></a><tt class="py-lineno">1410</tt>  <tt class="py-line">        <tt id="link-594" class="py-name"><a title="Bio.GenBank.GENBANK_SPACER" class="py-name" href="#" onclick="return doclink('link-594', 'GENBANK_SPACER', 'link-424');">GENBANK_SPACER</a></tt> <tt class="py-op">=</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt id="link-595" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-595', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt> </tt>
<a name="L1411"></a><tt class="py-lineno">1411</tt>  <tt class="py-line">        <tt class="py-name">lines</tt><tt class="py-op">.</tt><tt id="link-596" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-596', 'append', 'link-36');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">)</tt> </tt>
<a name="L1412"></a><tt class="py-lineno">1412</tt>  <tt class="py-line">        <tt class="py-name">line_iter</tt> <tt class="py-op">=</tt> <tt class="py-name">iter</tt><tt class="py-op">(</tt><tt class="py-name">lines</tt><tt class="py-op">)</tt> </tt>
<a name="L1413"></a><tt class="py-lineno">1413</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1414"></a><tt class="py-lineno">1414</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">line_iter</tt><tt class="py-op">:</tt> </tt>
<a name="L1415"></a><tt class="py-lineno">1415</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-597" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-597', 'startswith', 'link-97');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'BASE COUNT'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1416"></a><tt class="py-lineno">1416</tt>  <tt class="py-line">                    <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">10</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-598" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-598', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1417"></a><tt class="py-lineno">1417</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L1418"></a><tt class="py-lineno">1418</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt><tt class="py-op">:</tt> </tt>
<a name="L1419"></a><tt class="py-lineno">1419</tt>  <tt class="py-line">                            <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"base_count = "</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1420"></a><tt class="py-lineno">1420</tt>  <tt class="py-line">                        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-599" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.base_count()=Bio.GenBank._FeatureConsumer-class.html#base_count,Method Bio.GenBank._RecordConsumer.base_count()=Bio.GenBank._RecordConsumer-class.html#base_count"><a title="Bio.GenBank._FeatureConsumer.base_count
Bio.GenBank._RecordConsumer.base_count" class="py-name" href="#" onclick="return doclink('link-599', 'base_count', 'link-599');">base_count</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1421"></a><tt class="py-lineno">1421</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-600" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-600', 'startswith', 'link-97');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'ORIGIN'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1422"></a><tt class="py-lineno">1422</tt>  <tt class="py-line">                    <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-601" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-601', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1423"></a><tt class="py-lineno">1423</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L1424"></a><tt class="py-lineno">1424</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">debug</tt><tt class="py-op">:</tt> </tt>
<a name="L1425"></a><tt class="py-lineno">1425</tt>  <tt class="py-line">                            <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"origin_name = "</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1426"></a><tt class="py-lineno">1426</tt>  <tt class="py-line">                        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-602" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.origin_name()=Bio.GenBank._FeatureConsumer-class.html#origin_name,Method Bio.GenBank._RecordConsumer.origin_name()=Bio.GenBank._RecordConsumer-class.html#origin_name"><a title="Bio.GenBank._FeatureConsumer.origin_name
Bio.GenBank._RecordConsumer.origin_name" class="py-name" href="#" onclick="return doclink('link-602', 'origin_name', 'link-602');">origin_name</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1427"></a><tt class="py-lineno">1427</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-603" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-603', 'startswith', 'link-97');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'WGS '</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1428"></a><tt class="py-lineno">1428</tt>  <tt class="py-line">                    <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-604" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-604', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1429"></a><tt class="py-lineno">1429</tt>  <tt class="py-line">                    <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-605" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.wgs()=Bio.GenBank._FeatureConsumer-class.html#wgs,Method Bio.GenBank._RecordConsumer.wgs()=Bio.GenBank._RecordConsumer-class.html#wgs"><a title="Bio.GenBank._FeatureConsumer.wgs
Bio.GenBank._RecordConsumer.wgs" class="py-name" href="#" onclick="return doclink('link-605', 'wgs', 'link-605');">wgs</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1430"></a><tt class="py-lineno">1430</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-606" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-606', 'startswith', 'link-97');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'WGS_SCAFLD'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1431"></a><tt class="py-lineno">1431</tt>  <tt class="py-line">                    <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">10</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-607" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-607', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1432"></a><tt class="py-lineno">1432</tt>  <tt class="py-line">                    <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-608" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.add_wgs_scafld()=Bio.GenBank._FeatureConsumer-class.html#add_wgs_scafld,Method Bio.GenBank._RecordConsumer.add_wgs_scafld()=Bio.GenBank._RecordConsumer-class.html#add_wgs_scafld"><a title="Bio.GenBank._FeatureConsumer.add_wgs_scafld
Bio.GenBank._RecordConsumer.add_wgs_scafld" class="py-name" href="#" onclick="return doclink('link-608', 'add_wgs_scafld', 'link-608');">add_wgs_scafld</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1433"></a><tt class="py-lineno">1433</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-609" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-609', 'startswith', 'link-97');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'CONTIG'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1434"></a><tt class="py-lineno">1434</tt>  <tt class="py-line">                    <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-610" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-610', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1435"></a><tt class="py-lineno">1435</tt>  <tt class="py-line">                    <tt id="link-611" class="py-name"><a title="Bio.GenBank._FeatureConsumer.contig_location
Bio.GenBank._RecordConsumer.contig_location" class="py-name" href="#" onclick="return doclink('link-611', 'contig_location', 'link-334');">contig_location</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> </tt>
<a name="L1436"></a><tt class="py-lineno">1436</tt>  <tt class="py-line">                    <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L1437"></a><tt class="py-lineno">1437</tt>  <tt class="py-line">                        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-612" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-612', 'next', 'link-86');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">line_iter</tt><tt class="py-op">)</tt> </tt>
<a name="L1438"></a><tt class="py-lineno">1438</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L1439"></a><tt class="py-lineno">1439</tt>  <tt class="py-line">                            <tt class="py-keyword">break</tt> </tt>
<a name="L1440"></a><tt class="py-lineno">1440</tt>  <tt class="py-line">                        <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt id="link-613" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-613', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt id="link-614" class="py-name"><a title="Bio.GenBank.GENBANK_SPACER" class="py-name" href="#" onclick="return doclink('link-614', 'GENBANK_SPACER', 'link-424');">GENBANK_SPACER</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1441"></a><tt class="py-lineno">1441</tt>  <tt class="py-line">                            <tt class="py-comment"># Don't need to preseve the whitespace here.</tt> </tt>
<a name="L1442"></a><tt class="py-lineno">1442</tt>  <tt class="py-line">                            <tt id="link-615" class="py-name"><a title="Bio.GenBank._FeatureConsumer.contig_location
Bio.GenBank._RecordConsumer.contig_location" class="py-name" href="#" onclick="return doclink('link-615', 'contig_location', 'link-334');">contig_location</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt id="link-616" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-616', 'GENBANK_INDENT', 'link-422');">GENBANK_INDENT</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-617" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-617', 'rstrip', 'link-21');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1443"></a><tt class="py-lineno">1443</tt>  <tt class="py-line">                        <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-618" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-618', 'startswith', 'link-97');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'ORIGIN'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1444"></a><tt class="py-lineno">1444</tt>  <tt class="py-line">                            <tt class="py-comment"># Strange, seen this in GenPept files via Entrez gbwithparts</tt> </tt>
<a name="L1445"></a><tt class="py-lineno">1445</tt>  <tt class="py-line">                            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-619" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-619', 'strip', 'link-50');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1446"></a><tt class="py-lineno">1446</tt>  <tt class="py-line">                            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L1447"></a><tt class="py-lineno">1447</tt>  <tt class="py-line">                                <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-620" class="py-name"><a title="Bio.GenBank._FeatureConsumer.origin_name
Bio.GenBank._RecordConsumer.origin_name" class="py-name" href="#" onclick="return doclink('link-620', 'origin_name', 'link-602');">origin_name</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1448"></a><tt class="py-lineno">1448</tt>  <tt class="py-line">                            <tt class="py-keyword">break</tt> </tt>
<a name="L1449"></a><tt class="py-lineno">1449</tt>  <tt class="py-line">                        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1450"></a><tt class="py-lineno">1450</tt>  <tt class="py-line">                            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">'Expected CONTIG continuation line, got:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1451"></a><tt class="py-lineno">1451</tt>  <tt class="py-line">                    <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-621" class="py-name"><a title="Bio.GenBank._FeatureConsumer.contig_location
Bio.GenBank._RecordConsumer.contig_location" class="py-name" href="#" onclick="return doclink('link-621', 'contig_location', 'link-334');">contig_location</a></tt><tt class="py-op">(</tt><tt id="link-622" class="py-name"><a title="Bio.GenBank._FeatureConsumer.contig_location
Bio.GenBank._RecordConsumer.contig_location" class="py-name" href="#" onclick="return doclink('link-622', 'contig_location', 'link-334');">contig_location</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1452"></a><tt class="py-lineno">1452</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L1453"></a><tt class="py-lineno">1453</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L1454"></a><tt class="py-lineno">1454</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Problem in misc lines before sequence"</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1455"></a><tt class="py-lineno">1455</tt>  <tt class="py-line"> </tt>
<a name="L1456"></a><tt class="py-lineno">1456</tt>  <tt class="py-line"><tt class="py-keyword">if</tt> <tt class="py-name">__name__</tt> <tt class="py-op">==</tt> <tt class="py-string">"__main__"</tt><tt class="py-op">:</tt> </tt>
<a name="L1457"></a><tt class="py-lineno">1457</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-623" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-623', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-624" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-624', '_py3k', 'link-624');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">StringIO</tt> </tt>
<a name="L1458"></a><tt class="py-lineno">1458</tt>  <tt class="py-line"> </tt>
<a name="L1459"></a><tt class="py-lineno">1459</tt>  <tt class="py-line">    <tt class="py-name">gbk_example</tt> <tt class="py-op">=</tt> \ </tt>
<a name="L1460"></a><tt class="py-lineno">1460</tt>  <tt class="py-line">        <tt class="py-string">"""LOCUS       SCU49845     5028 bp    DNA             PLN       21-JUN-1999</tt> </tt>
<a name="L1461"></a><tt class="py-lineno">1461</tt>  <tt class="py-line"><tt class="py-string">DEFINITION  Saccharomyces cerevisiae TCP1-beta gene, partial cds, and Axl2p</tt> </tt>
<a name="L1462"></a><tt class="py-lineno">1462</tt>  <tt class="py-line"><tt class="py-string">            (AXL2) and Rev7p (REV7) genes, complete cds.</tt> </tt>
<a name="L1463"></a><tt class="py-lineno">1463</tt>  <tt class="py-line"><tt class="py-string">ACCESSION   U49845</tt> </tt>
<a name="L1464"></a><tt class="py-lineno">1464</tt>  <tt class="py-line"><tt class="py-string">VERSION     U49845.1  GI:1293613</tt> </tt>
<a name="L1465"></a><tt class="py-lineno">1465</tt>  <tt class="py-line"><tt class="py-string">KEYWORDS    .</tt> </tt>
<a name="L1466"></a><tt class="py-lineno">1466</tt>  <tt class="py-line"><tt class="py-string">SOURCE      Saccharomyces cerevisiae (baker's yeast)</tt> </tt>
<a name="L1467"></a><tt class="py-lineno">1467</tt>  <tt class="py-line"><tt class="py-string">  ORGANISM  Saccharomyces cerevisiae</tt> </tt>
<a name="L1468"></a><tt class="py-lineno">1468</tt>  <tt class="py-line"><tt class="py-string">            Eukaryota; Fungi; Ascomycota; Saccharomycotina; Saccharomycetes;</tt> </tt>
<a name="L1469"></a><tt class="py-lineno">1469</tt>  <tt class="py-line"><tt class="py-string">            Saccharomycetales; Saccharomycetaceae; Saccharomyces.</tt> </tt>
<a name="L1470"></a><tt class="py-lineno">1470</tt>  <tt class="py-line"><tt class="py-string">REFERENCE   1  (bases 1 to 5028)</tt> </tt>
<a name="L1471"></a><tt class="py-lineno">1471</tt>  <tt class="py-line"><tt class="py-string">  AUTHORS   Torpey,L.E., Gibbs,P.E., Nelson,J. and Lawrence,C.W.</tt> </tt>
<a name="L1472"></a><tt class="py-lineno">1472</tt>  <tt class="py-line"><tt class="py-string">  TITLE     Cloning and sequence of REV7, a gene whose function is required for</tt> </tt>
<a name="L1473"></a><tt class="py-lineno">1473</tt>  <tt class="py-line"><tt class="py-string">            DNA damage-induced mutagenesis in Saccharomyces cerevisiae</tt> </tt>
<a name="L1474"></a><tt class="py-lineno">1474</tt>  <tt class="py-line"><tt class="py-string">  JOURNAL   Yeast 10 (11), 1503-1509 (1994)</tt> </tt>
<a name="L1475"></a><tt class="py-lineno">1475</tt>  <tt class="py-line"><tt class="py-string">  PUBMED    7871890</tt> </tt>
<a name="L1476"></a><tt class="py-lineno">1476</tt>  <tt class="py-line"><tt class="py-string">REFERENCE   2  (bases 1 to 5028)</tt> </tt>
<a name="L1477"></a><tt class="py-lineno">1477</tt>  <tt class="py-line"><tt class="py-string">  AUTHORS   Roemer,T., Madden,K., Chang,J. and Snyder,M.</tt> </tt>
<a name="L1478"></a><tt class="py-lineno">1478</tt>  <tt class="py-line"><tt class="py-string">  TITLE     Selection of axial growth sites in yeast requires Axl2p, a novel</tt> </tt>
<a name="L1479"></a><tt class="py-lineno">1479</tt>  <tt class="py-line"><tt class="py-string">            plasma membrane glycoprotein</tt> </tt>
<a name="L1480"></a><tt class="py-lineno">1480</tt>  <tt class="py-line"><tt class="py-string">  JOURNAL   Genes Dev. 10 (7), 777-793 (1996)</tt> </tt>
<a name="L1481"></a><tt class="py-lineno">1481</tt>  <tt class="py-line"><tt class="py-string">  PUBMED    8846915</tt> </tt>
<a name="L1482"></a><tt class="py-lineno">1482</tt>  <tt class="py-line"><tt class="py-string">REFERENCE   3  (bases 1 to 5028)</tt> </tt>
<a name="L1483"></a><tt class="py-lineno">1483</tt>  <tt class="py-line"><tt class="py-string">  AUTHORS   Roemer,T.</tt> </tt>
<a name="L1484"></a><tt class="py-lineno">1484</tt>  <tt class="py-line"><tt class="py-string">  TITLE     Direct Submission</tt> </tt>
<a name="L1485"></a><tt class="py-lineno">1485</tt>  <tt class="py-line"><tt class="py-string">  JOURNAL   Submitted (22-FEB-1996) Terry Roemer, Biology, Yale University, New</tt> </tt>
<a name="L1486"></a><tt class="py-lineno">1486</tt>  <tt class="py-line"><tt class="py-string">            Haven, CT, USA</tt> </tt>
<a name="L1487"></a><tt class="py-lineno">1487</tt>  <tt class="py-line"><tt class="py-string">FEATURES             Location/Qualifiers</tt> </tt>
<a name="L1488"></a><tt class="py-lineno">1488</tt>  <tt class="py-line"><tt class="py-string">     source          1..5028</tt> </tt>
<a name="L1489"></a><tt class="py-lineno">1489</tt>  <tt class="py-line"><tt class="py-string">                     /organism="Saccharomyces cerevisiae"</tt> </tt>
<a name="L1490"></a><tt class="py-lineno">1490</tt>  <tt class="py-line"><tt class="py-string">                     /db_xref="taxon:4932"</tt> </tt>
<a name="L1491"></a><tt class="py-lineno">1491</tt>  <tt class="py-line"><tt class="py-string">                     /chromosome="IX"</tt> </tt>
<a name="L1492"></a><tt class="py-lineno">1492</tt>  <tt class="py-line"><tt class="py-string">                     /map="9"</tt> </tt>
<a name="L1493"></a><tt class="py-lineno">1493</tt>  <tt class="py-line"><tt class="py-string">     CDS             &lt;1..206</tt> </tt>
<a name="L1494"></a><tt class="py-lineno">1494</tt>  <tt class="py-line"><tt class="py-string">                     /codon_start=3</tt> </tt>
<a name="L1495"></a><tt class="py-lineno">1495</tt>  <tt class="py-line"><tt class="py-string">                     /product="TCP1-beta"</tt> </tt>
<a name="L1496"></a><tt class="py-lineno">1496</tt>  <tt class="py-line"><tt class="py-string">                     /protein_id="AAA98665.1"</tt> </tt>
<a name="L1497"></a><tt class="py-lineno">1497</tt>  <tt class="py-line"><tt class="py-string">                     /db_xref="GI:1293614"</tt> </tt>
<a name="L1498"></a><tt class="py-lineno">1498</tt>  <tt class="py-line"><tt class="py-string">                     /translation="SSIYNGISTSGLDLNNGTIADMRQLGIVESYKLKRAVVSSASEA</tt> </tt>
<a name="L1499"></a><tt class="py-lineno">1499</tt>  <tt class="py-line"><tt class="py-string">                     AEVLLRVDNIIRARPRTANRQHM"</tt> </tt>
<a name="L1500"></a><tt class="py-lineno">1500</tt>  <tt class="py-line"><tt class="py-string">     gene            687..3158</tt> </tt>
<a name="L1501"></a><tt class="py-lineno">1501</tt>  <tt class="py-line"><tt class="py-string">                     /gene="AXL2"</tt> </tt>
<a name="L1502"></a><tt class="py-lineno">1502</tt>  <tt class="py-line"><tt class="py-string">     CDS             687..3158</tt> </tt>
<a name="L1503"></a><tt class="py-lineno">1503</tt>  <tt class="py-line"><tt class="py-string">                     /gene="AXL2"</tt> </tt>
<a name="L1504"></a><tt class="py-lineno">1504</tt>  <tt class="py-line"><tt class="py-string">                     /note="plasma membrane glycoprotein"</tt> </tt>
<a name="L1505"></a><tt class="py-lineno">1505</tt>  <tt class="py-line"><tt class="py-string">                     /codon_start=1</tt> </tt>
<a name="L1506"></a><tt class="py-lineno">1506</tt>  <tt class="py-line"><tt class="py-string">                     /function="required for axial budding pattern of S.</tt> </tt>
<a name="L1507"></a><tt class="py-lineno">1507</tt>  <tt class="py-line"><tt class="py-string">                     cerevisiae"</tt> </tt>
<a name="L1508"></a><tt class="py-lineno">1508</tt>  <tt class="py-line"><tt class="py-string">                     /product="Axl2p"</tt> </tt>
<a name="L1509"></a><tt class="py-lineno">1509</tt>  <tt class="py-line"><tt class="py-string">                     /protein_id="AAA98666.1"</tt> </tt>
<a name="L1510"></a><tt class="py-lineno">1510</tt>  <tt class="py-line"><tt class="py-string">                     /db_xref="GI:1293615"</tt> </tt>
<a name="L1511"></a><tt class="py-lineno">1511</tt>  <tt class="py-line"><tt class="py-string">                     /translation="MTQLQISLLLTATISLLHLVVATPYEAYPIGKQYPPVARVNESF</tt> </tt>
<a name="L1512"></a><tt class="py-lineno">1512</tt>  <tt class="py-line"><tt class="py-string">                     TFQISNDTYKSSVDKTAQITYNCFDLPSWLSFDSSSRTFSGEPSSDLLSDANTTLYFN</tt> </tt>
<a name="L1513"></a><tt class="py-lineno">1513</tt>  <tt class="py-line"><tt class="py-string">                     VILEGTDSADSTSLNNTYQFVVTNRPSISLSSDFNLLALLKNYGYTNGKNALKLDPNE</tt> </tt>
<a name="L1514"></a><tt class="py-lineno">1514</tt>  <tt class="py-line"><tt class="py-string">                     VFNVTFDRSMFTNEESIVSYYGRSQLYNAPLPNWLFFDSGELKFTGTAPVINSAIAPE</tt> </tt>
<a name="L1515"></a><tt class="py-lineno">1515</tt>  <tt class="py-line"><tt class="py-string">                     TSYSFVIIATDIEGFSAVEVEFELVIGAHQLTTSIQNSLIINVTDTGNVSYDLPLNYV</tt> </tt>
<a name="L1516"></a><tt class="py-lineno">1516</tt>  <tt class="py-line"><tt class="py-string">                     YLDDDPISSDKLGSINLLDAPDWVALDNATISGSVPDELLGKNSNPANFSVSIYDTYG</tt> </tt>
<a name="L1517"></a><tt class="py-lineno">1517</tt>  <tt class="py-line"><tt class="py-string">                     DVIYFNFEVVSTTDLFAISSLPNINATRGEWFSYYFLPSQFTDYVNTNVSLEFTNSSQ</tt> </tt>
<a name="L1518"></a><tt class="py-lineno">1518</tt>  <tt class="py-line"><tt class="py-string">                     DHDWVKFQSSNLTLAGEVPKNFDKLSLGLKANQGSQSQELYFNIIGMDSKITHSNHSA</tt> </tt>
<a name="L1519"></a><tt class="py-lineno">1519</tt>  <tt class="py-line"><tt class="py-string">                     NATSTRSSHHSTSTSSYTSSTYTAKISSTSAAATSSAPAALPAANKTSSHNKKAVAIA</tt> </tt>
<a name="L1520"></a><tt class="py-lineno">1520</tt>  <tt class="py-line"><tt class="py-string">                     CGVAIPLGVILVALICFLIFWRRRRENPDDENLPHAISGPDLNNPANKPNQENATPLN</tt> </tt>
<a name="L1521"></a><tt class="py-lineno">1521</tt>  <tt class="py-line"><tt class="py-string">                     NPFDDDASSYDDTSIARRLAALNTLKLDNHSATESDISSVDEKRDSLSGMNTYNDQFQ</tt> </tt>
<a name="L1522"></a><tt class="py-lineno">1522</tt>  <tt class="py-line"><tt class="py-string">                     SQSKEELLAKPPVQPPESPFFDPQNRSSSVYMDSEPAVNKSWRYTGNLSPVSDIVRDS</tt> </tt>
<a name="L1523"></a><tt class="py-lineno">1523</tt>  <tt class="py-line"><tt class="py-string">                     YGSQKTVDTEKLFDLEAPEKEKRTSRDVTMSSLDPWNSNISPSPVRKSVTPSPYNVTK</tt> </tt>
<a name="L1524"></a><tt class="py-lineno">1524</tt>  <tt class="py-line"><tt class="py-string">                     HRNRHLQNIQDSQSGKNGITPTTMSTSSSDDFVPVKDGENFCWVHSMEPDRRPSKKRL</tt> </tt>
<a name="L1525"></a><tt class="py-lineno">1525</tt>  <tt class="py-line"><tt class="py-string">                     VDFSNKSNVNVGQVKDIHGRIPEML"</tt> </tt>
<a name="L1526"></a><tt class="py-lineno">1526</tt>  <tt class="py-line"><tt class="py-string">     gene            complement(3300..4037)</tt> </tt>
<a name="L1527"></a><tt class="py-lineno">1527</tt>  <tt class="py-line"><tt class="py-string">                     /gene="REV7"</tt> </tt>
<a name="L1528"></a><tt class="py-lineno">1528</tt>  <tt class="py-line"><tt class="py-string">     CDS             complement(3300..4037)</tt> </tt>
<a name="L1529"></a><tt class="py-lineno">1529</tt>  <tt class="py-line"><tt class="py-string">                     /gene="REV7"</tt> </tt>
<a name="L1530"></a><tt class="py-lineno">1530</tt>  <tt class="py-line"><tt class="py-string">                     /codon_start=1</tt> </tt>
<a name="L1531"></a><tt class="py-lineno">1531</tt>  <tt class="py-line"><tt class="py-string">                     /product="Rev7p"</tt> </tt>
<a name="L1532"></a><tt class="py-lineno">1532</tt>  <tt class="py-line"><tt class="py-string">                     /protein_id="AAA98667.1"</tt> </tt>
<a name="L1533"></a><tt class="py-lineno">1533</tt>  <tt class="py-line"><tt class="py-string">                     /db_xref="GI:1293616"</tt> </tt>
<a name="L1534"></a><tt class="py-lineno">1534</tt>  <tt class="py-line"><tt class="py-string">                     /translation="MNRWVEKWLRVYLKCYINLILFYRNVYPPQSFDYTTYQSFNLPQ</tt> </tt>
<a name="L1535"></a><tt class="py-lineno">1535</tt>  <tt class="py-line"><tt class="py-string">                     FVPINRHPALIDYIEELILDVLSKLTHVYRFSICIINKKNDLCIEKYVLDFSELQHVD</tt> </tt>
<a name="L1536"></a><tt class="py-lineno">1536</tt>  <tt class="py-line"><tt class="py-string">                     KDDQIITETEVFDEFRSSLNSLIMHLEKLPKVNDDTITFEAVINAIELELGHKLDRNR</tt> </tt>
<a name="L1537"></a><tt class="py-lineno">1537</tt>  <tt class="py-line"><tt class="py-string">                     RVDSLEEKAEIERDSNWVKCQEDENLPDNNGFQPPKIKLTSLVGSDVGPLIIHQFSEK</tt> </tt>
<a name="L1538"></a><tt class="py-lineno">1538</tt>  <tt class="py-line"><tt class="py-string">                     LISGDDKILNGVYSQYEEGESIFGSLF"</tt> </tt>
<a name="L1539"></a><tt class="py-lineno">1539</tt>  <tt class="py-line"><tt class="py-string">ORIGIN</tt> </tt>
<a name="L1540"></a><tt class="py-lineno">1540</tt>  <tt class="py-line"><tt class="py-string">        1 gatcctccat atacaacggt atctccacct caggtttaga tctcaacaac ggaaccattg</tt> </tt>
<a name="L1541"></a><tt class="py-lineno">1541</tt>  <tt class="py-line"><tt class="py-string">       61 ccgacatgag acagttaggt atcgtcgaga gttacaagct aaaacgagca gtagtcagct</tt> </tt>
<a name="L1542"></a><tt class="py-lineno">1542</tt>  <tt class="py-line"><tt class="py-string">      121 ctgcatctga agccgctgaa gttctactaa gggtggataa catcatccgt gcaagaccaa</tt> </tt>
<a name="L1543"></a><tt class="py-lineno">1543</tt>  <tt class="py-line"><tt class="py-string">      181 gaaccgccaa tagacaacat atgtaacata tttaggatat acctcgaaaa taataaaccg</tt> </tt>
<a name="L1544"></a><tt class="py-lineno">1544</tt>  <tt class="py-line"><tt class="py-string">      241 ccacactgtc attattataa ttagaaacag aacgcaaaaa ttatccacta tataattcaa</tt> </tt>
<a name="L1545"></a><tt class="py-lineno">1545</tt>  <tt class="py-line"><tt class="py-string">      301 agacgcgaaa aaaaaagaac aacgcgtcat agaacttttg gcaattcgcg tcacaaataa</tt> </tt>
<a name="L1546"></a><tt class="py-lineno">1546</tt>  <tt class="py-line"><tt class="py-string">      361 attttggcaa cttatgtttc ctcttcgagc agtactcgag ccctgtctca agaatgtaat</tt> </tt>
<a name="L1547"></a><tt class="py-lineno">1547</tt>  <tt class="py-line"><tt class="py-string">      421 aatacccatc gtaggtatgg ttaaagatag catctccaca acctcaaagc tccttgccga</tt> </tt>
<a name="L1548"></a><tt class="py-lineno">1548</tt>  <tt class="py-line"><tt class="py-string">      481 gagtcgccct cctttgtcga gtaattttca cttttcatat gagaacttat tttcttattc</tt> </tt>
<a name="L1549"></a><tt class="py-lineno">1549</tt>  <tt class="py-line"><tt class="py-string">      541 tttactctca catcctgtag tgattgacac tgcaacagcc accatcacta gaagaacaga</tt> </tt>
<a name="L1550"></a><tt class="py-lineno">1550</tt>  <tt class="py-line"><tt class="py-string">      601 acaattactt aatagaaaaa ttatatcttc ctcgaaacga tttcctgctt ccaacatcta</tt> </tt>
<a name="L1551"></a><tt class="py-lineno">1551</tt>  <tt class="py-line"><tt class="py-string">      661 cgtatatcaa gaagcattca cttaccatga cacagcttca gatttcatta ttgctgacag</tt> </tt>
<a name="L1552"></a><tt class="py-lineno">1552</tt>  <tt class="py-line"><tt class="py-string">      721 ctactatatc actactccat ctagtagtgg ccacgcccta tgaggcatat cctatcggaa</tt> </tt>
<a name="L1553"></a><tt class="py-lineno">1553</tt>  <tt class="py-line"><tt class="py-string">      781 aacaataccc cccagtggca agagtcaatg aatcgtttac atttcaaatt tccaatgata</tt> </tt>
<a name="L1554"></a><tt class="py-lineno">1554</tt>  <tt class="py-line"><tt class="py-string">      841 cctataaatc gtctgtagac aagacagctc aaataacata caattgcttc gacttaccga</tt> </tt>
<a name="L1555"></a><tt class="py-lineno">1555</tt>  <tt class="py-line"><tt class="py-string">      901 gctggctttc gtttgactct agttctagaa cgttctcagg tgaaccttct tctgacttac</tt> </tt>
<a name="L1556"></a><tt class="py-lineno">1556</tt>  <tt class="py-line"><tt class="py-string">      961 tatctgatgc gaacaccacg ttgtatttca atgtaatact cgagggtacg gactctgccg</tt> </tt>
<a name="L1557"></a><tt class="py-lineno">1557</tt>  <tt class="py-line"><tt class="py-string">     1021 acagcacgtc tttgaacaat acataccaat ttgttgttac aaaccgtcca tccatctcgc</tt> </tt>
<a name="L1558"></a><tt class="py-lineno">1558</tt>  <tt class="py-line"><tt class="py-string">     1081 tatcgtcaga tttcaatcta ttggcgttgt taaaaaacta tggttatact aacggcaaaa</tt> </tt>
<a name="L1559"></a><tt class="py-lineno">1559</tt>  <tt class="py-line"><tt class="py-string">     1141 acgctctgaa actagatcct aatgaagtct tcaacgtgac ttttgaccgt tcaatgttca</tt> </tt>
<a name="L1560"></a><tt class="py-lineno">1560</tt>  <tt class="py-line"><tt class="py-string">     1201 ctaacgaaga atccattgtg tcgtattacg gacgttctca gttgtataat gcgccgttac</tt> </tt>
<a name="L1561"></a><tt class="py-lineno">1561</tt>  <tt class="py-line"><tt class="py-string">     1261 ccaattggct gttcttcgat tctggcgagt tgaagtttac tgggacggca ccggtgataa</tt> </tt>
<a name="L1562"></a><tt class="py-lineno">1562</tt>  <tt class="py-line"><tt class="py-string">     1321 actcggcgat tgctccagaa acaagctaca gttttgtcat catcgctaca gacattgaag</tt> </tt>
<a name="L1563"></a><tt class="py-lineno">1563</tt>  <tt class="py-line"><tt class="py-string">     1381 gattttctgc cgttgaggta gaattcgaat tagtcatcgg ggctcaccag ttaactacct</tt> </tt>
<a name="L1564"></a><tt class="py-lineno">1564</tt>  <tt class="py-line"><tt class="py-string">     1441 ctattcaaaa tagtttgata atcaacgtta ctgacacagg taacgtttca tatgacttac</tt> </tt>
<a name="L1565"></a><tt class="py-lineno">1565</tt>  <tt class="py-line"><tt class="py-string">     1501 ctctaaacta tgtttatctc gatgacgatc ctatttcttc tgataaattg ggttctataa</tt> </tt>
<a name="L1566"></a><tt class="py-lineno">1566</tt>  <tt class="py-line"><tt class="py-string">     1561 acttattgga tgctccagac tgggtggcat tagataatgc taccatttcc gggtctgtcc</tt> </tt>
<a name="L1567"></a><tt class="py-lineno">1567</tt>  <tt class="py-line"><tt class="py-string">     1621 cagatgaatt actcggtaag aactccaatc ctgccaattt ttctgtgtcc atttatgata</tt> </tt>
<a name="L1568"></a><tt class="py-lineno">1568</tt>  <tt class="py-line"><tt class="py-string">     1681 cttatggtga tgtgatttat ttcaacttcg aagttgtctc cacaacggat ttgtttgcca</tt> </tt>
<a name="L1569"></a><tt class="py-lineno">1569</tt>  <tt class="py-line"><tt class="py-string">     1741 ttagttctct tcccaatatt aacgctacaa ggggtgaatg gttctcctac tattttttgc</tt> </tt>
<a name="L1570"></a><tt class="py-lineno">1570</tt>  <tt class="py-line"><tt class="py-string">     1801 cttctcagtt tacagactac gtgaatacaa acgtttcatt agagtttact aattcaagcc</tt> </tt>
<a name="L1571"></a><tt class="py-lineno">1571</tt>  <tt class="py-line"><tt class="py-string">     1861 aagaccatga ctgggtgaaa ttccaatcat ctaatttaac attagctgga gaagtgccca</tt> </tt>
<a name="L1572"></a><tt class="py-lineno">1572</tt>  <tt class="py-line"><tt class="py-string">     1921 agaatttcga caagctttca ttaggtttga aagcgaacca aggttcacaa tctcaagagc</tt> </tt>
<a name="L1573"></a><tt class="py-lineno">1573</tt>  <tt class="py-line"><tt class="py-string">     1981 tatattttaa catcattggc atggattcaa agataactca ctcaaaccac agtgcgaatg</tt> </tt>
<a name="L1574"></a><tt class="py-lineno">1574</tt>  <tt class="py-line"><tt class="py-string">     2041 caacgtccac aagaagttct caccactcca cctcaacaag ttcttacaca tcttctactt</tt> </tt>
<a name="L1575"></a><tt class="py-lineno">1575</tt>  <tt class="py-line"><tt class="py-string">     2101 acactgcaaa aatttcttct acctccgctg ctgctacttc ttctgctcca gcagcgctgc</tt> </tt>
<a name="L1576"></a><tt class="py-lineno">1576</tt>  <tt class="py-line"><tt class="py-string">     2161 cagcagccaa taaaacttca tctcacaata aaaaagcagt agcaattgcg tgcggtgttg</tt> </tt>
<a name="L1577"></a><tt class="py-lineno">1577</tt>  <tt class="py-line"><tt class="py-string">     2221 ctatcccatt aggcgttatc ctagtagctc tcatttgctt cctaatattc tggagacgca</tt> </tt>
<a name="L1578"></a><tt class="py-lineno">1578</tt>  <tt class="py-line"><tt class="py-string">     2281 gaagggaaaa tccagacgat gaaaacttac cgcatgctat tagtggacct gatttgaata</tt> </tt>
<a name="L1579"></a><tt class="py-lineno">1579</tt>  <tt class="py-line"><tt class="py-string">     2341 atcctgcaaa taaaccaaat caagaaaacg ctacaccttt gaacaacccc tttgatgatg</tt> </tt>
<a name="L1580"></a><tt class="py-lineno">1580</tt>  <tt class="py-line"><tt class="py-string">     2401 atgcttcctc gtacgatgat acttcaatag caagaagatt ggctgctttg aacactttga</tt> </tt>
<a name="L1581"></a><tt class="py-lineno">1581</tt>  <tt class="py-line"><tt class="py-string">     2461 aattggataa ccactctgcc actgaatctg atatttccag cgtggatgaa aagagagatt</tt> </tt>
<a name="L1582"></a><tt class="py-lineno">1582</tt>  <tt class="py-line"><tt class="py-string">     2521 ctctatcagg tatgaataca tacaatgatc agttccaatc ccaaagtaaa gaagaattat</tt> </tt>
<a name="L1583"></a><tt class="py-lineno">1583</tt>  <tt class="py-line"><tt class="py-string">     2581 tagcaaaacc cccagtacag cctccagaga gcccgttctt tgacccacag aataggtctt</tt> </tt>
<a name="L1584"></a><tt class="py-lineno">1584</tt>  <tt class="py-line"><tt class="py-string">     2641 cttctgtgta tatggatagt gaaccagcag taaataaatc ctggcgatat actggcaacc</tt> </tt>
<a name="L1585"></a><tt class="py-lineno">1585</tt>  <tt class="py-line"><tt class="py-string">     2701 tgtcaccagt ctctgatatt gtcagagaca gttacggatc acaaaaaact gttgatacag</tt> </tt>
<a name="L1586"></a><tt class="py-lineno">1586</tt>  <tt class="py-line"><tt class="py-string">     2761 aaaaactttt cgatttagaa gcaccagaga aggaaaaacg tacgtcaagg gatgtcacta</tt> </tt>
<a name="L1587"></a><tt class="py-lineno">1587</tt>  <tt class="py-line"><tt class="py-string">     2821 tgtcttcact ggacccttgg aacagcaata ttagcccttc tcccgtaaga aaatcagtaa</tt> </tt>
<a name="L1588"></a><tt class="py-lineno">1588</tt>  <tt class="py-line"><tt class="py-string">     2881 caccatcacc atataacgta acgaagcatc gtaaccgcca cttacaaaat attcaagact</tt> </tt>
<a name="L1589"></a><tt class="py-lineno">1589</tt>  <tt class="py-line"><tt class="py-string">     2941 ctcaaagcgg taaaaacgga atcactccca caacaatgtc aacttcatct tctgacgatt</tt> </tt>
<a name="L1590"></a><tt class="py-lineno">1590</tt>  <tt class="py-line"><tt class="py-string">     3001 ttgttccggt taaagatggt gaaaattttt gctgggtcca tagcatggaa ccagacagaa</tt> </tt>
<a name="L1591"></a><tt class="py-lineno">1591</tt>  <tt class="py-line"><tt class="py-string">     3061 gaccaagtaa gaaaaggtta gtagattttt caaataagag taatgtcaat gttggtcaag</tt> </tt>
<a name="L1592"></a><tt class="py-lineno">1592</tt>  <tt class="py-line"><tt class="py-string">     3121 ttaaggacat tcacggacgc atcccagaaa tgctgtgatt atacgcaacg atattttgct</tt> </tt>
<a name="L1593"></a><tt class="py-lineno">1593</tt>  <tt class="py-line"><tt class="py-string">     3181 taattttatt ttcctgtttt attttttatt agtggtttac agatacccta tattttattt</tt> </tt>
<a name="L1594"></a><tt class="py-lineno">1594</tt>  <tt class="py-line"><tt class="py-string">     3241 agtttttata cttagagaca tttaatttta attccattct tcaaatttca tttttgcact</tt> </tt>
<a name="L1595"></a><tt class="py-lineno">1595</tt>  <tt class="py-line"><tt class="py-string">     3301 taaaacaaag atccaaaaat gctctcgccc tcttcatatt gagaatacac tccattcaaa</tt> </tt>
<a name="L1596"></a><tt class="py-lineno">1596</tt>  <tt class="py-line"><tt class="py-string">     3361 attttgtcgt caccgctgat taatttttca ctaaactgat gaataatcaa aggccccacg</tt> </tt>
<a name="L1597"></a><tt class="py-lineno">1597</tt>  <tt class="py-line"><tt class="py-string">     3421 tcagaaccga ctaaagaagt gagttttatt ttaggaggtt gaaaaccatt attgtctggt</tt> </tt>
<a name="L1598"></a><tt class="py-lineno">1598</tt>  <tt class="py-line"><tt class="py-string">     3481 aaattttcat cttcttgaca tttaacccag tttgaatccc tttcaatttc tgctttttcc</tt> </tt>
<a name="L1599"></a><tt class="py-lineno">1599</tt>  <tt class="py-line"><tt class="py-string">     3541 tccaaactat cgaccctcct gtttctgtcc aacttatgtc ctagttccaa ttcgatcgca</tt> </tt>
<a name="L1600"></a><tt class="py-lineno">1600</tt>  <tt class="py-line"><tt class="py-string">     3601 ttaataactg cttcaaatgt tattgtgtca tcgttgactt taggtaattt ctccaaatgc</tt> </tt>
<a name="L1601"></a><tt class="py-lineno">1601</tt>  <tt class="py-line"><tt class="py-string">     3661 ataatcaaac tatttaagga agatcggaat tcgtcgaaca cttcagtttc cgtaatgatc</tt> </tt>
<a name="L1602"></a><tt class="py-lineno">1602</tt>  <tt class="py-line"><tt class="py-string">     3721 tgatcgtctt tatccacatg ttgtaattca ctaaaatcta aaacgtattt ttcaatgcat</tt> </tt>
<a name="L1603"></a><tt class="py-lineno">1603</tt>  <tt class="py-line"><tt class="py-string">     3781 aaatcgttct ttttattaat aatgcagatg gaaaatctgt aaacgtgcgt taatttagaa</tt> </tt>
<a name="L1604"></a><tt class="py-lineno">1604</tt>  <tt class="py-line"><tt class="py-string">     3841 agaacatcca gtataagttc ttctatatag tcaattaaag caggatgcct attaatggga</tt> </tt>
<a name="L1605"></a><tt class="py-lineno">1605</tt>  <tt class="py-line"><tt class="py-string">     3901 acgaactgcg gcaagttgaa tgactggtaa gtagtgtagt cgaatgactg aggtgggtat</tt> </tt>
<a name="L1606"></a><tt class="py-lineno">1606</tt>  <tt class="py-line"><tt class="py-string">     3961 acatttctat aaaataaaat caaattaatg tagcatttta agtataccct cagccacttc</tt> </tt>
<a name="L1607"></a><tt class="py-lineno">1607</tt>  <tt class="py-line"><tt class="py-string">     4021 tctacccatc tattcataaa gctgacgcaa cgattactat tttttttttc ttcttggatc</tt> </tt>
<a name="L1608"></a><tt class="py-lineno">1608</tt>  <tt class="py-line"><tt class="py-string">     4081 tcagtcgtcg caaaaacgta taccttcttt ttccgacctt ttttttagct ttctggaaaa</tt> </tt>
<a name="L1609"></a><tt class="py-lineno">1609</tt>  <tt class="py-line"><tt class="py-string">     4141 gtttatatta gttaaacagg gtctagtctt agtgtgaaag ctagtggttt cgattgactg</tt> </tt>
<a name="L1610"></a><tt class="py-lineno">1610</tt>  <tt class="py-line"><tt class="py-string">     4201 atattaagaa agtggaaatt aaattagtag tgtagacgta tatgcatatg tatttctcgc</tt> </tt>
<a name="L1611"></a><tt class="py-lineno">1611</tt>  <tt class="py-line"><tt class="py-string">     4261 ctgtttatgt ttctacgtac ttttgattta tagcaagggg aaaagaaata catactattt</tt> </tt>
<a name="L1612"></a><tt class="py-lineno">1612</tt>  <tt class="py-line"><tt class="py-string">     4321 tttggtaaag gtgaaagcat aatgtaaaag ctagaataaa atggacgaaa taaagagagg</tt> </tt>
<a name="L1613"></a><tt class="py-lineno">1613</tt>  <tt class="py-line"><tt class="py-string">     4381 cttagttcat cttttttcca aaaagcaccc aatgataata actaaaatga aaaggatttg</tt> </tt>
<a name="L1614"></a><tt class="py-lineno">1614</tt>  <tt class="py-line"><tt class="py-string">     4441 ccatctgtca gcaacatcag ttgtgtgagc aataataaaa tcatcacctc cgttgccttt</tt> </tt>
<a name="L1615"></a><tt class="py-lineno">1615</tt>  <tt class="py-line"><tt class="py-string">     4501 agcgcgtttg tcgtttgtat cttccgtaat tttagtctta tcaatgggaa tcataaattt</tt> </tt>
<a name="L1616"></a><tt class="py-lineno">1616</tt>  <tt class="py-line"><tt class="py-string">     4561 tccaatgaat tagcaatttc gtccaattct ttttgagctt cttcatattt gctttggaat</tt> </tt>
<a name="L1617"></a><tt class="py-lineno">1617</tt>  <tt class="py-line"><tt class="py-string">     4621 tcttcgcact tcttttccca ttcatctctt tcttcttcca aagcaacgat ccttctaccc</tt> </tt>
<a name="L1618"></a><tt class="py-lineno">1618</tt>  <tt class="py-line"><tt class="py-string">     4681 atttgctcag agttcaaatc ggcctctttc agtttatcca ttgcttcctt cagtttggct</tt> </tt>
<a name="L1619"></a><tt class="py-lineno">1619</tt>  <tt class="py-line"><tt class="py-string">     4741 tcactgtctt ctagctgttg ttctagatcc tggtttttct tggtgtagtt ctcattatta</tt> </tt>
<a name="L1620"></a><tt class="py-lineno">1620</tt>  <tt class="py-line"><tt class="py-string">     4801 gatctcaagt tattggagtc ttcagccaat tgctttgtat cagacaattg actctctaac</tt> </tt>
<a name="L1621"></a><tt class="py-lineno">1621</tt>  <tt class="py-line"><tt class="py-string">     4861 ttctccactt cactgtcgag ttgctcgttt ttagcggaca aagatttaat ctcgttttct</tt> </tt>
<a name="L1622"></a><tt class="py-lineno">1622</tt>  <tt class="py-line"><tt class="py-string">     4921 ttttcagtgt tagattgctc taattctttg agctgttctc tcagctcctc atatttttct</tt> </tt>
<a name="L1623"></a><tt class="py-lineno">1623</tt>  <tt class="py-line"><tt class="py-string">     4981 tgccatgact cagattctaa ttttaagcta ttcaatttct ctttgatc</tt> </tt>
<a name="L1624"></a><tt class="py-lineno">1624</tt>  <tt class="py-line"><tt class="py-string">//"""</tt> </tt>
<a name="L1625"></a><tt class="py-lineno">1625</tt>  <tt class="py-line"> </tt>
<a name="L1626"></a><tt class="py-lineno">1626</tt>  <tt class="py-line">    <tt class="py-comment"># GenBank format protein (aka GenPept) file from:</tt> </tt>
<a name="L1627"></a><tt class="py-lineno">1627</tt>  <tt class="py-line">    <tt class="py-comment"># http://www.molecularevolution.org/resources/fileformats/</tt> </tt>
<a name="L1628"></a><tt class="py-lineno">1628</tt>  <tt class="py-line">    <tt class="py-name">gbk_example2</tt> <tt class="py-op">=</tt> \ </tt>
<a name="L1629"></a><tt class="py-lineno">1629</tt>  <tt class="py-line">        <tt class="py-string">"""LOCUS       AAD51968                 143 aa            linear   BCT 21-AUG-2001</tt> </tt>
<a name="L1630"></a><tt class="py-lineno">1630</tt>  <tt class="py-line"><tt class="py-string">DEFINITION  transcriptional regulator RovA [Yersinia enterocolitica].</tt> </tt>
<a name="L1631"></a><tt class="py-lineno">1631</tt>  <tt class="py-line"><tt class="py-string">ACCESSION   AAD51968</tt> </tt>
<a name="L1632"></a><tt class="py-lineno">1632</tt>  <tt class="py-line"><tt class="py-string">VERSION     AAD51968.1  GI:5805369</tt> </tt>
<a name="L1633"></a><tt class="py-lineno">1633</tt>  <tt class="py-line"><tt class="py-string">DBSOURCE    locus AF171097 accession AF171097.1</tt> </tt>
<a name="L1634"></a><tt class="py-lineno">1634</tt>  <tt class="py-line"><tt class="py-string">KEYWORDS    .</tt> </tt>
<a name="L1635"></a><tt class="py-lineno">1635</tt>  <tt class="py-line"><tt class="py-string">SOURCE      Yersinia enterocolitica</tt> </tt>
<a name="L1636"></a><tt class="py-lineno">1636</tt>  <tt class="py-line"><tt class="py-string">  ORGANISM  Yersinia enterocolitica</tt> </tt>
<a name="L1637"></a><tt class="py-lineno">1637</tt>  <tt class="py-line"><tt class="py-string">            Bacteria; Proteobacteria; Gammaproteobacteria; Enterobacteriales;</tt> </tt>
<a name="L1638"></a><tt class="py-lineno">1638</tt>  <tt class="py-line"><tt class="py-string">            Enterobacteriaceae; Yersinia.</tt> </tt>
<a name="L1639"></a><tt class="py-lineno">1639</tt>  <tt class="py-line"><tt class="py-string">REFERENCE   1  (residues 1 to 143)</tt> </tt>
<a name="L1640"></a><tt class="py-lineno">1640</tt>  <tt class="py-line"><tt class="py-string">  AUTHORS   Revell,P.A. and Miller,V.L.</tt> </tt>
<a name="L1641"></a><tt class="py-lineno">1641</tt>  <tt class="py-line"><tt class="py-string">  TITLE     A chromosomally encoded regulator is required for expression of the</tt> </tt>
<a name="L1642"></a><tt class="py-lineno">1642</tt>  <tt class="py-line"><tt class="py-string">            Yersinia enterocolitica inv gene and for virulence</tt> </tt>
<a name="L1643"></a><tt class="py-lineno">1643</tt>  <tt class="py-line"><tt class="py-string">  JOURNAL   Mol. Microbiol. 35 (3), 677-685 (2000)</tt> </tt>
<a name="L1644"></a><tt class="py-lineno">1644</tt>  <tt class="py-line"><tt class="py-string">  MEDLINE   20138369</tt> </tt>
<a name="L1645"></a><tt class="py-lineno">1645</tt>  <tt class="py-line"><tt class="py-string">   PUBMED   10672189</tt> </tt>
<a name="L1646"></a><tt class="py-lineno">1646</tt>  <tt class="py-line"><tt class="py-string">REFERENCE   2  (residues 1 to 143)</tt> </tt>
<a name="L1647"></a><tt class="py-lineno">1647</tt>  <tt class="py-line"><tt class="py-string">  AUTHORS   Revell,P.A. and Miller,V.L.</tt> </tt>
<a name="L1648"></a><tt class="py-lineno">1648</tt>  <tt class="py-line"><tt class="py-string">  TITLE     Direct Submission</tt> </tt>
<a name="L1649"></a><tt class="py-lineno">1649</tt>  <tt class="py-line"><tt class="py-string">  JOURNAL   Submitted (22-JUL-1999) Molecular Microbiology, Washington</tt> </tt>
<a name="L1650"></a><tt class="py-lineno">1650</tt>  <tt class="py-line"><tt class="py-string">            University School of Medicine, Campus Box 8230, 660 South Euclid,</tt> </tt>
<a name="L1651"></a><tt class="py-lineno">1651</tt>  <tt class="py-line"><tt class="py-string">            St. Louis, MO 63110, USA</tt> </tt>
<a name="L1652"></a><tt class="py-lineno">1652</tt>  <tt class="py-line"><tt class="py-string">COMMENT     Method: conceptual translation.</tt> </tt>
<a name="L1653"></a><tt class="py-lineno">1653</tt>  <tt class="py-line"><tt class="py-string">FEATURES             Location/Qualifiers</tt> </tt>
<a name="L1654"></a><tt class="py-lineno">1654</tt>  <tt class="py-line"><tt class="py-string">     source          1..143</tt> </tt>
<a name="L1655"></a><tt class="py-lineno">1655</tt>  <tt class="py-line"><tt class="py-string">                     /organism="Yersinia enterocolitica"</tt> </tt>
<a name="L1656"></a><tt class="py-lineno">1656</tt>  <tt class="py-line"><tt class="py-string">                     /mol_type="unassigned DNA"</tt> </tt>
<a name="L1657"></a><tt class="py-lineno">1657</tt>  <tt class="py-line"><tt class="py-string">                     /strain="JB580v"</tt> </tt>
<a name="L1658"></a><tt class="py-lineno">1658</tt>  <tt class="py-line"><tt class="py-string">                     /serotype="O:8"</tt> </tt>
<a name="L1659"></a><tt class="py-lineno">1659</tt>  <tt class="py-line"><tt class="py-string">                     /db_xref="taxon:630"</tt> </tt>
<a name="L1660"></a><tt class="py-lineno">1660</tt>  <tt class="py-line"><tt class="py-string">     Protein         1..143</tt> </tt>
<a name="L1661"></a><tt class="py-lineno">1661</tt>  <tt class="py-line"><tt class="py-string">                     /product="transcriptional regulator RovA"</tt> </tt>
<a name="L1662"></a><tt class="py-lineno">1662</tt>  <tt class="py-line"><tt class="py-string">                     /name="regulates inv expression"</tt> </tt>
<a name="L1663"></a><tt class="py-lineno">1663</tt>  <tt class="py-line"><tt class="py-string">     CDS             1..143</tt> </tt>
<a name="L1664"></a><tt class="py-lineno">1664</tt>  <tt class="py-line"><tt class="py-string">                     /gene="rovA"</tt> </tt>
<a name="L1665"></a><tt class="py-lineno">1665</tt>  <tt class="py-line"><tt class="py-string">                     /coded_by="AF171097.1:380..811"</tt> </tt>
<a name="L1666"></a><tt class="py-lineno">1666</tt>  <tt class="py-line"><tt class="py-string">                     /note="regulator of virulence"</tt> </tt>
<a name="L1667"></a><tt class="py-lineno">1667</tt>  <tt class="py-line"><tt class="py-string">                     /transl_table=11</tt> </tt>
<a name="L1668"></a><tt class="py-lineno">1668</tt>  <tt class="py-line"><tt class="py-string">ORIGIN</tt> </tt>
<a name="L1669"></a><tt class="py-lineno">1669</tt>  <tt class="py-line"><tt class="py-string">        1 mestlgsdla rlvrvwrali dhrlkplelt qthwvtlhni nrlppeqsqi qlakaigieq</tt> </tt>
<a name="L1670"></a><tt class="py-lineno">1670</tt>  <tt class="py-line"><tt class="py-string">       61 pslvrtldql eekglitrht candrrakri klteqsspii eqvdgvicst rkeilggisp</tt> </tt>
<a name="L1671"></a><tt class="py-lineno">1671</tt>  <tt class="py-line"><tt class="py-string">      121 deiellsgli dklerniiql qsk</tt> </tt>
<a name="L1672"></a><tt class="py-lineno">1672</tt>  <tt class="py-line"><tt class="py-string">//</tt> </tt>
<a name="L1673"></a><tt class="py-lineno">1673</tt>  <tt class="py-line"><tt class="py-string">"""</tt> </tt>
<a name="L1674"></a><tt class="py-lineno">1674</tt>  <tt class="py-line"> </tt>
<a name="L1675"></a><tt class="py-lineno">1675</tt>  <tt class="py-line">    <tt class="py-name">embl_example</tt> <tt class="py-op">=</tt> <tt class="py-string">"""ID   X56734; SV 1; linear; mRNA; STD; PLN; 1859 BP.</tt> </tt>
<a name="L1676"></a><tt class="py-lineno">1676</tt>  <tt class="py-line"><tt class="py-string">XX</tt> </tt>
<a name="L1677"></a><tt class="py-lineno">1677</tt>  <tt class="py-line"><tt class="py-string">AC   X56734; S46826;</tt> </tt>
<a name="L1678"></a><tt class="py-lineno">1678</tt>  <tt class="py-line"><tt class="py-string">XX</tt> </tt>
<a name="L1679"></a><tt class="py-lineno">1679</tt>  <tt class="py-line"><tt class="py-string">DT   12-SEP-1991 (Rel. 29, Created)</tt> </tt>
<a name="L1680"></a><tt class="py-lineno">1680</tt>  <tt class="py-line"><tt class="py-string">DT   25-NOV-2005 (Rel. 85, Last updated, Version 11)</tt> </tt>
<a name="L1681"></a><tt class="py-lineno">1681</tt>  <tt class="py-line"><tt class="py-string">XX</tt> </tt>
<a name="L1682"></a><tt class="py-lineno">1682</tt>  <tt class="py-line"><tt class="py-string">DE   Trifolium repens mRNA for non-cyanogenic beta-glucosidase</tt> </tt>
<a name="L1683"></a><tt class="py-lineno">1683</tt>  <tt class="py-line"><tt class="py-string">XX</tt> </tt>
<a name="L1684"></a><tt class="py-lineno">1684</tt>  <tt class="py-line"><tt class="py-string">KW   beta-glucosidase.</tt> </tt>
<a name="L1685"></a><tt class="py-lineno">1685</tt>  <tt class="py-line"><tt class="py-string">XX</tt> </tt>
<a name="L1686"></a><tt class="py-lineno">1686</tt>  <tt class="py-line"><tt class="py-string">OS   Trifolium repens (white clover)</tt> </tt>
<a name="L1687"></a><tt class="py-lineno">1687</tt>  <tt class="py-line"><tt class="py-string">OC   Eukaryota; Viridiplantae; Streptophyta; Embryophyta; Tracheophyta;</tt> </tt>
<a name="L1688"></a><tt class="py-lineno">1688</tt>  <tt class="py-line"><tt class="py-string">OC   Spermatophyta; Magnoliophyta; eudicotyledons; core eudicotyledons; rosids;</tt> </tt>
<a name="L1689"></a><tt class="py-lineno">1689</tt>  <tt class="py-line"><tt class="py-string">OC   eurosids I; Fabales; Fabaceae; Papilionoideae; Trifolieae; Trifolium.</tt> </tt>
<a name="L1690"></a><tt class="py-lineno">1690</tt>  <tt class="py-line"><tt class="py-string">XX</tt> </tt>
<a name="L1691"></a><tt class="py-lineno">1691</tt>  <tt class="py-line"><tt class="py-string">RN   [5]</tt> </tt>
<a name="L1692"></a><tt class="py-lineno">1692</tt>  <tt class="py-line"><tt class="py-string">RP   1-1859</tt> </tt>
<a name="L1693"></a><tt class="py-lineno">1693</tt>  <tt class="py-line"><tt class="py-string">RX   PUBMED; 1907511.</tt> </tt>
<a name="L1694"></a><tt class="py-lineno">1694</tt>  <tt class="py-line"><tt class="py-string">RA   Oxtoby E., Dunn M.A., Pancoro A., Hughes M.A.;</tt> </tt>
<a name="L1695"></a><tt class="py-lineno">1695</tt>  <tt class="py-line"><tt class="py-string">RT   "Nucleotide and derived amino acid sequence of the cyanogenic</tt> </tt>
<a name="L1696"></a><tt class="py-lineno">1696</tt>  <tt class="py-line"><tt class="py-string">RT   beta-glucosidase (linamarase) from white clover (Trifolium repens L.)";</tt> </tt>
<a name="L1697"></a><tt class="py-lineno">1697</tt>  <tt class="py-line"><tt class="py-string">RL   Plant Mol. Biol. 17(2):209-219(1991).</tt> </tt>
<a name="L1698"></a><tt class="py-lineno">1698</tt>  <tt class="py-line"><tt class="py-string">XX</tt> </tt>
<a name="L1699"></a><tt class="py-lineno">1699</tt>  <tt class="py-line"><tt class="py-string">RN   [6]</tt> </tt>
<a name="L1700"></a><tt class="py-lineno">1700</tt>  <tt class="py-line"><tt class="py-string">RP   1-1859</tt> </tt>
<a name="L1701"></a><tt class="py-lineno">1701</tt>  <tt class="py-line"><tt class="py-string">RA   Hughes M.A.;</tt> </tt>
<a name="L1702"></a><tt class="py-lineno">1702</tt>  <tt class="py-line"><tt class="py-string">RT   ;</tt> </tt>
<a name="L1703"></a><tt class="py-lineno">1703</tt>  <tt class="py-line"><tt class="py-string">RL   Submitted (19-NOV-1990) to the EMBL/GenBank/DDBJ databases.</tt> </tt>
<a name="L1704"></a><tt class="py-lineno">1704</tt>  <tt class="py-line"><tt class="py-string">RL   Hughes M.A., University of Newcastle Upon Tyne, Medical School, Newcastle</tt> </tt>
<a name="L1705"></a><tt class="py-lineno">1705</tt>  <tt class="py-line"><tt class="py-string">RL   Upon Tyne, NE2 4HH, UK</tt> </tt>
<a name="L1706"></a><tt class="py-lineno">1706</tt>  <tt class="py-line"><tt class="py-string">XX</tt> </tt>
<a name="L1707"></a><tt class="py-lineno">1707</tt>  <tt class="py-line"><tt class="py-string">FH   Key             Location/Qualifiers</tt> </tt>
<a name="L1708"></a><tt class="py-lineno">1708</tt>  <tt class="py-line"><tt class="py-string">FH</tt> </tt>
<a name="L1709"></a><tt class="py-lineno">1709</tt>  <tt class="py-line"><tt class="py-string">FT   source          1..1859</tt> </tt>
<a name="L1710"></a><tt class="py-lineno">1710</tt>  <tt class="py-line"><tt class="py-string">FT                   /organism="Trifolium repens"</tt> </tt>
<a name="L1711"></a><tt class="py-lineno">1711</tt>  <tt class="py-line"><tt class="py-string">FT                   /mol_type="mRNA"</tt> </tt>
<a name="L1712"></a><tt class="py-lineno">1712</tt>  <tt class="py-line"><tt class="py-string">FT                   /clone_lib="lambda gt10"</tt> </tt>
<a name="L1713"></a><tt class="py-lineno">1713</tt>  <tt class="py-line"><tt class="py-string">FT                   /clone="TRE361"</tt> </tt>
<a name="L1714"></a><tt class="py-lineno">1714</tt>  <tt class="py-line"><tt class="py-string">FT                   /tissue_type="leaves"</tt> </tt>
<a name="L1715"></a><tt class="py-lineno">1715</tt>  <tt class="py-line"><tt class="py-string">FT                   /db_xref="taxon:3899"</tt> </tt>
<a name="L1716"></a><tt class="py-lineno">1716</tt>  <tt class="py-line"><tt class="py-string">FT   CDS             14..1495</tt> </tt>
<a name="L1717"></a><tt class="py-lineno">1717</tt>  <tt class="py-line"><tt class="py-string">FT                   /product="beta-glucosidase"</tt> </tt>
<a name="L1718"></a><tt class="py-lineno">1718</tt>  <tt class="py-line"><tt class="py-string">FT                   /EC_number="3.2.1.21"</tt> </tt>
<a name="L1719"></a><tt class="py-lineno">1719</tt>  <tt class="py-line"><tt class="py-string">FT                   /note="non-cyanogenic"</tt> </tt>
<a name="L1720"></a><tt class="py-lineno">1720</tt>  <tt class="py-line"><tt class="py-string">FT                   /db_xref="GOA:P26204"</tt> </tt>
<a name="L1721"></a><tt class="py-lineno">1721</tt>  <tt class="py-line"><tt class="py-string">FT                   /db_xref="InterPro:IPR001360"</tt> </tt>
<a name="L1722"></a><tt class="py-lineno">1722</tt>  <tt class="py-line"><tt class="py-string">FT                   /db_xref="InterPro:IPR013781"</tt> </tt>
<a name="L1723"></a><tt class="py-lineno">1723</tt>  <tt class="py-line"><tt class="py-string">FT                   /db_xref="UniProtKB/Swiss-Prot:P26204"</tt> </tt>
<a name="L1724"></a><tt class="py-lineno">1724</tt>  <tt class="py-line"><tt class="py-string">FT                   /protein_id="CAA40058.1"</tt> </tt>
<a name="L1725"></a><tt class="py-lineno">1725</tt>  <tt class="py-line"><tt class="py-string">FT                   /translation="MDFIVAIFALFVISSFTITSTNAVEASTLLDIGNLSRSSFPRGFI</tt> </tt>
<a name="L1726"></a><tt class="py-lineno">1726</tt>  <tt class="py-line"><tt class="py-string">FT                   FGAGSSAYQFEGAVNEGGRGPSIWDTFTHKYPEKIRDGSNADITVDQYHRYKEDVGIMK</tt> </tt>
<a name="L1727"></a><tt class="py-lineno">1727</tt>  <tt class="py-line"><tt class="py-string">FT                   DQNMDSYRFSISWPRILPKGKLSGGINHEGIKYYNNLINELLANGIQPFVTLFHWDLPQ</tt> </tt>
<a name="L1728"></a><tt class="py-lineno">1728</tt>  <tt class="py-line"><tt class="py-string">FT                   VLEDEYGGFLNSGVINDFRDYTDLCFKEFGDRVRYWSTLNEPWVFSNSGYALGTNAPGR</tt> </tt>
<a name="L1729"></a><tt class="py-lineno">1729</tt>  <tt class="py-line"><tt class="py-string">FT                   CSASNVAKPGDSGTGPYIVTHNQILAHAEAVHVYKTKYQAYQKGKIGITLVSNWLMPLD</tt> </tt>
<a name="L1730"></a><tt class="py-lineno">1730</tt>  <tt class="py-line"><tt class="py-string">FT                   DNSIPDIKAAERSLDFQFGLFMEQLTTGDYSKSMRRIVKNRLPKFSKFESSLVNGSFDF</tt> </tt>
<a name="L1731"></a><tt class="py-lineno">1731</tt>  <tt class="py-line"><tt class="py-string">FT                   IGINYYSSSYISNAPSHGNAKPSYSTNPMTNISFEKHGIPLGPRAASIWIYVYPYMFIQ</tt> </tt>
<a name="L1732"></a><tt class="py-lineno">1732</tt>  <tt class="py-line"><tt class="py-string">FT                   EDFEIFCYILKINITILQFSITENGMNEFNDATLPVEEALLNTYRIDYYYRHLYYIRSA</tt> </tt>
<a name="L1733"></a><tt class="py-lineno">1733</tt>  <tt class="py-line"><tt class="py-string">FT                   IRAGSNVKGFYAWSFLDCNEWFAGFTVRFGLNFVD"</tt> </tt>
<a name="L1734"></a><tt class="py-lineno">1734</tt>  <tt class="py-line"><tt class="py-string">FT   mRNA            1..1859</tt> </tt>
<a name="L1735"></a><tt class="py-lineno">1735</tt>  <tt class="py-line"><tt class="py-string">FT                   /experiment="experimental evidence, no additional details</tt> </tt>
<a name="L1736"></a><tt class="py-lineno">1736</tt>  <tt class="py-line"><tt class="py-string">FT                   recorded"</tt> </tt>
<a name="L1737"></a><tt class="py-lineno">1737</tt>  <tt class="py-line"><tt class="py-string">XX</tt> </tt>
<a name="L1738"></a><tt class="py-lineno">1738</tt>  <tt class="py-line"><tt class="py-string">SQ   Sequence 1859 BP; 609 A; 314 C; 355 G; 581 T; 0 other;</tt> </tt>
<a name="L1739"></a><tt class="py-lineno">1739</tt>  <tt class="py-line"><tt class="py-string">     aaacaaacca aatatggatt ttattgtagc catatttgct ctgtttgtta ttagctcatt        60</tt> </tt>
<a name="L1740"></a><tt class="py-lineno">1740</tt>  <tt class="py-line"><tt class="py-string">     cacaattact tccacaaatg cagttgaagc ttctactctt cttgacatag gtaacctgag       120</tt> </tt>
<a name="L1741"></a><tt class="py-lineno">1741</tt>  <tt class="py-line"><tt class="py-string">     tcggagcagt tttcctcgtg gcttcatctt tggtgctgga tcttcagcat accaatttga       180</tt> </tt>
<a name="L1742"></a><tt class="py-lineno">1742</tt>  <tt class="py-line"><tt class="py-string">     aggtgcagta aacgaaggcg gtagaggacc aagtatttgg gataccttca cccataaata       240</tt> </tt>
<a name="L1743"></a><tt class="py-lineno">1743</tt>  <tt class="py-line"><tt class="py-string">     tccagaaaaa ataagggatg gaagcaatgc agacatcacg gttgaccaat atcaccgcta       300</tt> </tt>
<a name="L1744"></a><tt class="py-lineno">1744</tt>  <tt class="py-line"><tt class="py-string">     caaggaagat gttgggatta tgaaggatca aaatatggat tcgtatagat tctcaatctc       360</tt> </tt>
<a name="L1745"></a><tt class="py-lineno">1745</tt>  <tt class="py-line"><tt class="py-string">     ttggccaaga atactcccaa agggaaagtt gagcggaggc ataaatcacg aaggaatcaa       420</tt> </tt>
<a name="L1746"></a><tt class="py-lineno">1746</tt>  <tt class="py-line"><tt class="py-string">     atattacaac aaccttatca acgaactatt ggctaacggt atacaaccat ttgtaactct       480</tt> </tt>
<a name="L1747"></a><tt class="py-lineno">1747</tt>  <tt class="py-line"><tt class="py-string">     ttttcattgg gatcttcccc aagtcttaga agatgagtat ggtggtttct taaactccgg       540</tt> </tt>
<a name="L1748"></a><tt class="py-lineno">1748</tt>  <tt class="py-line"><tt class="py-string">     tgtaataaat gattttcgag actatacgga tctttgcttc aaggaatttg gagatagagt       600</tt> </tt>
<a name="L1749"></a><tt class="py-lineno">1749</tt>  <tt class="py-line"><tt class="py-string">     gaggtattgg agtactctaa atgagccatg ggtgtttagc aattctggat atgcactagg       660</tt> </tt>
<a name="L1750"></a><tt class="py-lineno">1750</tt>  <tt class="py-line"><tt class="py-string">     aacaaatgca ccaggtcgat gttcggcctc caacgtggcc aagcctggtg attctggaac       720</tt> </tt>
<a name="L1751"></a><tt class="py-lineno">1751</tt>  <tt class="py-line"><tt class="py-string">     aggaccttat atagttacac acaatcaaat tcttgctcat gcagaagctg tacatgtgta       780</tt> </tt>
<a name="L1752"></a><tt class="py-lineno">1752</tt>  <tt class="py-line"><tt class="py-string">     taagactaaa taccaggcat atcaaaaggg aaagataggc ataacgttgg tatctaactg       840</tt> </tt>
<a name="L1753"></a><tt class="py-lineno">1753</tt>  <tt class="py-line"><tt class="py-string">     gttaatgcca cttgatgata atagcatacc agatataaag gctgccgaga gatcacttga       900</tt> </tt>
<a name="L1754"></a><tt class="py-lineno">1754</tt>  <tt class="py-line"><tt class="py-string">     cttccaattt ggattgttta tggaacaatt aacaacagga gattattcta agagcatgcg       960</tt> </tt>
<a name="L1755"></a><tt class="py-lineno">1755</tt>  <tt class="py-line"><tt class="py-string">     gcgtatagtt aaaaaccgat tacctaagtt ctcaaaattc gaatcaagcc tagtgaatgg      1020</tt> </tt>
<a name="L1756"></a><tt class="py-lineno">1756</tt>  <tt class="py-line"><tt class="py-string">     ttcatttgat tttattggta taaactatta ctcttctagt tatattagca atgccccttc      1080</tt> </tt>
<a name="L1757"></a><tt class="py-lineno">1757</tt>  <tt class="py-line"><tt class="py-string">     acatggcaat gccaaaccca gttactcaac aaatcctatg accaatattt catttgaaaa      1140</tt> </tt>
<a name="L1758"></a><tt class="py-lineno">1758</tt>  <tt class="py-line"><tt class="py-string">     acatgggata cccttaggtc caagggctgc ttcaatttgg atatatgttt atccatatat      1200</tt> </tt>
<a name="L1759"></a><tt class="py-lineno">1759</tt>  <tt class="py-line"><tt class="py-string">     gtttatccaa gaggacttcg agatcttttg ttacatatta aaaataaata taacaatcct      1260</tt> </tt>
<a name="L1760"></a><tt class="py-lineno">1760</tt>  <tt class="py-line"><tt class="py-string">     gcaattttca atcactgaaa atggtatgaa tgaattcaac gatgcaacac ttccagtaga      1320</tt> </tt>
<a name="L1761"></a><tt class="py-lineno">1761</tt>  <tt class="py-line"><tt class="py-string">     agaagctctt ttgaatactt acagaattga ttactattac cgtcacttat actacattcg      1380</tt> </tt>
<a name="L1762"></a><tt class="py-lineno">1762</tt>  <tt class="py-line"><tt class="py-string">     ttctgcaatc agggctggct caaatgtgaa gggtttttac gcatggtcat ttttggactg      1440</tt> </tt>
<a name="L1763"></a><tt class="py-lineno">1763</tt>  <tt class="py-line"><tt class="py-string">     taatgaatgg tttgcaggct ttactgttcg ttttggatta aactttgtag attagaaaga      1500</tt> </tt>
<a name="L1764"></a><tt class="py-lineno">1764</tt>  <tt class="py-line"><tt class="py-string">     tggattaaaa aggtacccta agctttctgc ccaatggtac aagaactttc tcaaaagaaa      1560</tt> </tt>
<a name="L1765"></a><tt class="py-lineno">1765</tt>  <tt class="py-line"><tt class="py-string">     ctagctagta ttattaaaag aactttgtag tagattacag tacatcgttt gaagttgagt      1620</tt> </tt>
<a name="L1766"></a><tt class="py-lineno">1766</tt>  <tt class="py-line"><tt class="py-string">     tggtgcacct aattaaataa aagaggttac tcttaacata tttttaggcc attcgttgtg      1680</tt> </tt>
<a name="L1767"></a><tt class="py-lineno">1767</tt>  <tt class="py-line"><tt class="py-string">     aagttgttag gctgttattt ctattatact atgttgtagt aataagtgca ttgttgtacc      1740</tt> </tt>
<a name="L1768"></a><tt class="py-lineno">1768</tt>  <tt class="py-line"><tt class="py-string">     agaagctatg atcataacta taggttgatc cttcatgtat cagtttgatg ttgagaatac      1800</tt> </tt>
<a name="L1769"></a><tt class="py-lineno">1769</tt>  <tt class="py-line"><tt class="py-string">     tttgaattaa aagtcttttt ttattttttt aaaaaaaaaa aaaaaaaaaa aaaaaaaaa       1859</tt> </tt>
<a name="L1770"></a><tt class="py-lineno">1770</tt>  <tt class="py-line"><tt class="py-string">//</tt> </tt>
<a name="L1771"></a><tt class="py-lineno">1771</tt>  <tt class="py-line"><tt class="py-string">"""</tt> </tt>
<a name="L1772"></a><tt class="py-lineno">1772</tt>  <tt class="py-line"> </tt>
<a name="L1773"></a><tt class="py-lineno">1773</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"GenBank CDS Iteration"</tt><tt class="py-op">)</tt> </tt>
<a name="L1774"></a><tt class="py-lineno">1774</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"====================="</tt><tt class="py-op">)</tt> </tt>
<a name="L1775"></a><tt class="py-lineno">1775</tt>  <tt class="py-line"> </tt>
<a name="L1776"></a><tt class="py-lineno">1776</tt>  <tt class="py-line">    <tt class="py-name">g</tt> <tt class="py-op">=</tt> <tt id="link-625" class="py-name" targets="Class Bio.GenBank.Scanner.GenBankScanner=Bio.GenBank.Scanner.GenBankScanner-class.html"><a title="Bio.GenBank.Scanner.GenBankScanner" class="py-name" href="#" onclick="return doclink('link-625', 'GenBankScanner', 'link-625');">GenBankScanner</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1777"></a><tt class="py-lineno">1777</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">g</tt><tt class="py-op">.</tt><tt id="link-626" class="py-name" targets="Method Bio.GenBank.Scanner.InsdcScanner.parse_cds_features()=Bio.GenBank.Scanner.InsdcScanner-class.html#parse_cds_features"><a title="Bio.GenBank.Scanner.InsdcScanner.parse_cds_features" class="py-name" href="#" onclick="return doclink('link-626', 'parse_cds_features', 'link-626');">parse_cds_features</a></tt><tt class="py-op">(</tt><tt class="py-name">StringIO</tt><tt class="py-op">(</tt><tt class="py-name">gbk_example</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1778"></a><tt class="py-lineno">1778</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L1779"></a><tt class="py-lineno">1779</tt>  <tt class="py-line"> </tt>
<a name="L1780"></a><tt class="py-lineno">1780</tt>  <tt class="py-line">    <tt class="py-name">g</tt> <tt class="py-op">=</tt> <tt id="link-627" class="py-name"><a title="Bio.GenBank.Scanner.GenBankScanner" class="py-name" href="#" onclick="return doclink('link-627', 'GenBankScanner', 'link-625');">GenBankScanner</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1781"></a><tt class="py-lineno">1781</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">g</tt><tt class="py-op">.</tt><tt id="link-628" class="py-name"><a title="Bio.GenBank.Scanner.InsdcScanner.parse_cds_features" class="py-name" href="#" onclick="return doclink('link-628', 'parse_cds_features', 'link-626');">parse_cds_features</a></tt><tt class="py-op">(</tt><tt class="py-name">StringIO</tt><tt class="py-op">(</tt><tt class="py-name">gbk_example2</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1782"></a><tt class="py-lineno">1782</tt>  <tt class="py-line">                                       <tt class="py-name">tags2id</tt><tt class="py-op">=</tt><tt class="py-op">(</tt><tt class="py-string">'gene'</tt><tt class="py-op">,</tt> <tt class="py-string">'locus_tag'</tt><tt class="py-op">,</tt> <tt class="py-string">'product'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1783"></a><tt class="py-lineno">1783</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L1784"></a><tt class="py-lineno">1784</tt>  <tt class="py-line"> </tt>
<a name="L1785"></a><tt class="py-lineno">1785</tt>  <tt class="py-line">    <tt class="py-name">g</tt> <tt class="py-op">=</tt> <tt id="link-629" class="py-name"><a title="Bio.GenBank.Scanner.GenBankScanner" class="py-name" href="#" onclick="return doclink('link-629', 'GenBankScanner', 'link-625');">GenBankScanner</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1786"></a><tt class="py-lineno">1786</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">g</tt><tt class="py-op">.</tt><tt id="link-630" class="py-name"><a title="Bio.GenBank.Scanner.InsdcScanner.parse_cds_features" class="py-name" href="#" onclick="return doclink('link-630', 'parse_cds_features', 'link-626');">parse_cds_features</a></tt><tt class="py-op">(</tt><tt class="py-name">StringIO</tt><tt class="py-op">(</tt><tt class="py-name">gbk_example</tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt> <tt class="py-op">+</tt> <tt class="py-name">gbk_example2</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1787"></a><tt class="py-lineno">1787</tt>  <tt class="py-line">                                       <tt class="py-name">tags2id</tt><tt class="py-op">=</tt><tt class="py-op">(</tt><tt class="py-string">'gene'</tt><tt class="py-op">,</tt> <tt class="py-string">'locus_tag'</tt><tt class="py-op">,</tt> <tt class="py-string">'product'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1788"></a><tt class="py-lineno">1788</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L1789"></a><tt class="py-lineno">1789</tt>  <tt class="py-line"> </tt>
<a name="L1790"></a><tt class="py-lineno">1790</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">)</tt> </tt>
<a name="L1791"></a><tt class="py-lineno">1791</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"GenBank Iteration"</tt><tt class="py-op">)</tt> </tt>
<a name="L1792"></a><tt class="py-lineno">1792</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"================="</tt><tt class="py-op">)</tt> </tt>
<a name="L1793"></a><tt class="py-lineno">1793</tt>  <tt class="py-line">    <tt class="py-name">g</tt> <tt class="py-op">=</tt> <tt id="link-631" class="py-name"><a title="Bio.GenBank.Scanner.GenBankScanner" class="py-name" href="#" onclick="return doclink('link-631', 'GenBankScanner', 'link-625');">GenBankScanner</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1794"></a><tt class="py-lineno">1794</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">g</tt><tt class="py-op">.</tt><tt id="link-632" class="py-name" targets="Method Bio.GenBank.Scanner.InsdcScanner.parse_records()=Bio.GenBank.Scanner.InsdcScanner-class.html#parse_records"><a title="Bio.GenBank.Scanner.InsdcScanner.parse_records" class="py-name" href="#" onclick="return doclink('link-632', 'parse_records', 'link-632');">parse_records</a></tt><tt class="py-op">(</tt><tt class="py-name">StringIO</tt><tt class="py-op">(</tt><tt class="py-name">gbk_example</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">do_features</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1795"></a><tt class="py-lineno">1795</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"%s %s %s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-633" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-633', 'id', 'link-166');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-634" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-634', 'name', 'link-167');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-635" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-635', 'description', 'link-168');">description</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1796"></a><tt class="py-lineno">1796</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-636" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-636', 'seq', 'link-177');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1797"></a><tt class="py-lineno">1797</tt>  <tt class="py-line"> </tt>
<a name="L1798"></a><tt class="py-lineno">1798</tt>  <tt class="py-line">    <tt class="py-name">g</tt> <tt class="py-op">=</tt> <tt id="link-637" class="py-name"><a title="Bio.GenBank.Scanner.GenBankScanner" class="py-name" href="#" onclick="return doclink('link-637', 'GenBankScanner', 'link-625');">GenBankScanner</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1799"></a><tt class="py-lineno">1799</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">g</tt><tt class="py-op">.</tt><tt id="link-638" class="py-name"><a title="Bio.GenBank.Scanner.InsdcScanner.parse_records" class="py-name" href="#" onclick="return doclink('link-638', 'parse_records', 'link-632');">parse_records</a></tt><tt class="py-op">(</tt><tt class="py-name">StringIO</tt><tt class="py-op">(</tt><tt class="py-name">gbk_example</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">do_features</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1800"></a><tt class="py-lineno">1800</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"%s %s %s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-639" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-639', 'id', 'link-166');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-640" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-640', 'name', 'link-167');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-641" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-641', 'description', 'link-168');">description</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1801"></a><tt class="py-lineno">1801</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-642" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-642', 'seq', 'link-177');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1802"></a><tt class="py-lineno">1802</tt>  <tt class="py-line"> </tt>
<a name="L1803"></a><tt class="py-lineno">1803</tt>  <tt class="py-line">    <tt class="py-name">g</tt> <tt class="py-op">=</tt> <tt id="link-643" class="py-name"><a title="Bio.GenBank.Scanner.GenBankScanner" class="py-name" href="#" onclick="return doclink('link-643', 'GenBankScanner', 'link-625');">GenBankScanner</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1804"></a><tt class="py-lineno">1804</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">g</tt><tt class="py-op">.</tt><tt id="link-644" class="py-name"><a title="Bio.GenBank.Scanner.InsdcScanner.parse_records" class="py-name" href="#" onclick="return doclink('link-644', 'parse_records', 'link-632');">parse_records</a></tt><tt class="py-op">(</tt><tt class="py-name">StringIO</tt><tt class="py-op">(</tt><tt class="py-name">gbk_example2</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">do_features</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1805"></a><tt class="py-lineno">1805</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"%s %s %s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-645" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-645', 'id', 'link-166');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-646" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-646', 'name', 'link-167');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-647" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-647', 'description', 'link-168');">description</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1806"></a><tt class="py-lineno">1806</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-648" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-648', 'seq', 'link-177');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1807"></a><tt class="py-lineno">1807</tt>  <tt class="py-line"> </tt>
<a name="L1808"></a><tt class="py-lineno">1808</tt>  <tt class="py-line">    <tt class="py-name">g</tt> <tt class="py-op">=</tt> <tt id="link-649" class="py-name"><a title="Bio.GenBank.Scanner.GenBankScanner" class="py-name" href="#" onclick="return doclink('link-649', 'GenBankScanner', 'link-625');">GenBankScanner</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1809"></a><tt class="py-lineno">1809</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">g</tt><tt class="py-op">.</tt><tt id="link-650" class="py-name"><a title="Bio.GenBank.Scanner.InsdcScanner.parse_records" class="py-name" href="#" onclick="return doclink('link-650', 'parse_records', 'link-632');">parse_records</a></tt><tt class="py-op">(</tt><tt class="py-name">StringIO</tt><tt class="py-op">(</tt><tt class="py-name">gbk_example2</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">do_features</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1810"></a><tt class="py-lineno">1810</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"%s %s %s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-651" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-651', 'id', 'link-166');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-652" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-652', 'name', 'link-167');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-653" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-653', 'description', 'link-168');">description</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1811"></a><tt class="py-lineno">1811</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-654" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-654', 'seq', 'link-177');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1812"></a><tt class="py-lineno">1812</tt>  <tt class="py-line"> </tt>
<a name="L1813"></a><tt class="py-lineno">1813</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">)</tt> </tt>
<a name="L1814"></a><tt class="py-lineno">1814</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"EMBL CDS Iteration"</tt><tt class="py-op">)</tt> </tt>
<a name="L1815"></a><tt class="py-lineno">1815</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"=================="</tt><tt class="py-op">)</tt> </tt>
<a name="L1816"></a><tt class="py-lineno">1816</tt>  <tt class="py-line"> </tt>
<a name="L1817"></a><tt class="py-lineno">1817</tt>  <tt class="py-line">    <tt class="py-name">e</tt> <tt class="py-op">=</tt> <tt id="link-655" class="py-name" targets="Class Bio.GenBank.Scanner.EmblScanner=Bio.GenBank.Scanner.EmblScanner-class.html"><a title="Bio.GenBank.Scanner.EmblScanner" class="py-name" href="#" onclick="return doclink('link-655', 'EmblScanner', 'link-655');">EmblScanner</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1818"></a><tt class="py-lineno">1818</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">e</tt><tt class="py-op">.</tt><tt id="link-656" class="py-name"><a title="Bio.GenBank.Scanner.InsdcScanner.parse_cds_features" class="py-name" href="#" onclick="return doclink('link-656', 'parse_cds_features', 'link-626');">parse_cds_features</a></tt><tt class="py-op">(</tt><tt class="py-name">StringIO</tt><tt class="py-op">(</tt><tt class="py-name">embl_example</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1819"></a><tt class="py-lineno">1819</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L1820"></a><tt class="py-lineno">1820</tt>  <tt class="py-line"> </tt>
<a name="L1821"></a><tt class="py-lineno">1821</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">)</tt> </tt>
<a name="L1822"></a><tt class="py-lineno">1822</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"EMBL Iteration"</tt><tt class="py-op">)</tt> </tt>
<a name="L1823"></a><tt class="py-lineno">1823</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"=============="</tt><tt class="py-op">)</tt> </tt>
<a name="L1824"></a><tt class="py-lineno">1824</tt>  <tt class="py-line">    <tt class="py-name">e</tt> <tt class="py-op">=</tt> <tt id="link-657" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner" class="py-name" href="#" onclick="return doclink('link-657', 'EmblScanner', 'link-655');">EmblScanner</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1825"></a><tt class="py-lineno">1825</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">e</tt><tt class="py-op">.</tt><tt id="link-658" class="py-name"><a title="Bio.GenBank.Scanner.InsdcScanner.parse_records" class="py-name" href="#" onclick="return doclink('link-658', 'parse_records', 'link-632');">parse_records</a></tt><tt class="py-op">(</tt><tt class="py-name">StringIO</tt><tt class="py-op">(</tt><tt class="py-name">embl_example</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">do_features</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1826"></a><tt class="py-lineno">1826</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"%s %s %s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-659" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-659', 'id', 'link-166');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-660" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-660', 'name', 'link-167');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-661" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-661', 'description', 'link-168');">description</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1827"></a><tt class="py-lineno">1827</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-662" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-662', 'seq', 'link-177');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1828"></a><tt class="py-lineno">1828</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:04:46 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
