<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.motifs.transfac</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.motifs-module.html">Package&nbsp;motifs</a> ::
        Module&nbsp;transfac
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.motifs.transfac-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.motifs.transfac-module.html">Module Bio.motifs.transfac</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2003 by Bartek Wilczynski.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-docstring">"""Parsing TRANSFAC files."""</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-1" class="py-name" targets="Class Bio.BiopythonParserWarning=Bio.BiopythonParserWarning-class.html"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-1', 'BiopythonParserWarning', 'link-1');">BiopythonParserWarning</a></tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-2" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-2', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-3" class="py-name" targets="Package Bio.motifs=Bio.motifs-module.html"><a title="Bio.motifs" class="py-name" href="#" onclick="return doclink('link-3', 'motifs', 'link-3');">motifs</a></tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-4" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-4', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-5" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-5', 'Alphabet', 'link-5');">Alphabet</a></tt> <tt class="py-keyword">import</tt> <tt id="link-6" class="py-name" targets="Module Bio.Alphabet.IUPAC=Bio.Alphabet.IUPAC-module.html"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-6', 'IUPAC', 'link-6');">IUPAC</a></tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"> </tt>
<a name="Motif"></a><div id="Motif-def"><a name="L15"></a><tt class="py-lineno"> 15</tt> <a class="py-toggle" href="#" id="Motif-toggle" onclick="return toggle('Motif');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.motifs.transfac.Motif-class.html">Motif</a><tt class="py-op">(</tt><tt class="py-base-class">motifs</tt><tt class="py-op">.</tt><tt class="py-base-class">Motif</tt><tt class="py-op">,</tt> <tt class="py-base-class">dict</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Motif-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Motif-expanded"><a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line">    <tt class="py-docstring">"""Store the information for one TRANSFAC motif.</tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-docstring">    This class inherits from the Bio.motifs.Motif base class, as well</tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-docstring">    as from a Python dictionary. All motif information found by the parser</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-docstring">    is stored as attributes of the base class when possible; see the</tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-docstring">    Bio.motifs.Motif base class for a description of these attributes. All</tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-docstring">    other information associated with the motif is stored as (key, value)</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-docstring">    pairs in the dictionary, where the key is the two-letter fields as found</tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-docstring">    in the TRANSFAC file. References are an exception: These are stored in</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-docstring">    the .references attribute.</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"><tt class="py-docstring">    These fields are commonly found in TRANSFAC files::</tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"><tt class="py-docstring">        AC:    Accession number</tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"><tt class="py-docstring">        AS:    Accession numbers, secondary</tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"><tt class="py-docstring">        BA:    Statistical basis</tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"><tt class="py-docstring">        BF:    Binding factors</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-docstring">        BS:    Factor binding sites underlying the matrix</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-docstring">               [sequence; SITE accession number; start position for matrix</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-docstring">               sequence; length of sequence used; number of gaps inserted;</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"><tt class="py-docstring">               strand orientation.]</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"><tt class="py-docstring">        CC:    Comments</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"><tt class="py-docstring">        CO:    Copyright notice</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"><tt class="py-docstring">        DE:    Short factor description</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"><tt class="py-docstring">        DR:    External databases</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"><tt class="py-docstring">               [database name: database accession number]</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"><tt class="py-docstring">        DT:    Date created/updated</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line"><tt class="py-docstring">        HC:    Subfamilies</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line"><tt class="py-docstring">        HP:    Superfamilies</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line"><tt class="py-docstring">        ID:    Identifier</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line"><tt class="py-docstring">        NA:    Name of the binding factor</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line"><tt class="py-docstring">        OC:    Taxonomic classification</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"><tt class="py-docstring">        OS:    Species/Taxon</tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"><tt class="py-docstring">        OV:    Older version</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line"><tt class="py-docstring">        PV:    Preferred version</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line"><tt class="py-docstring">        TY:    Type</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line"><tt class="py-docstring">        XX:    Empty line; these are not stored in the Record.</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line"><tt class="py-docstring">    References are stored in an .references attribute, which is a list of</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line"><tt class="py-docstring">    dictionaries with the following keys::</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"><tt class="py-docstring">        RN:    Reference number</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line"><tt class="py-docstring">        RA:    Reference authors</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line"><tt class="py-docstring">        RL:    Reference data</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line"><tt class="py-docstring">        RT:    Reference title</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line"><tt class="py-docstring">        RX:    PubMed ID</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line"><tt class="py-docstring">    For more information, see the TRANSFAC documentation.</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line"> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line">    <tt id="link-7" class="py-name" targets="Variable Bio.motifs.transfac.Motif.multiple_value_keys=Bio.motifs.transfac.Motif-class.html#multiple_value_keys"><a title="Bio.motifs.transfac.Motif.multiple_value_keys" class="py-name" href="#" onclick="return doclink('link-7', 'multiple_value_keys', 'link-7');">multiple_value_keys</a></tt> <tt class="py-op">=</tt> <tt id="link-8" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.set()=Bio.Nexus.Nexus.StepMatrix-class.html#set,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.set()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#set,Method Bio.SVDSuperimposer.SVDSuperimposer.set()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#set"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-8', 'set', 'link-8');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-string">'BF'</tt><tt class="py-op">,</tt> <tt class="py-string">'OV'</tt><tt class="py-op">,</tt> <tt class="py-string">'HP'</tt><tt class="py-op">,</tt> <tt class="py-string">'BS'</tt><tt class="py-op">,</tt> <tt class="py-string">'HC'</tt><tt class="py-op">,</tt> <tt class="py-string">'DT'</tt><tt class="py-op">,</tt> <tt class="py-string">'DR'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line">    <tt class="py-comment"># These keys can occur multiple times for one motif</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line"> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line">    <tt id="link-9" class="py-name" targets="Variable Bio.motifs.transfac.Motif.reference_keys=Bio.motifs.transfac.Motif-class.html#reference_keys"><a title="Bio.motifs.transfac.Motif.reference_keys" class="py-name" href="#" onclick="return doclink('link-9', 'reference_keys', 'link-9');">reference_keys</a></tt> <tt class="py-op">=</tt> <tt id="link-10" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-10', 'set', 'link-8');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-string">'RX'</tt><tt class="py-op">,</tt> <tt class="py-string">'RA'</tt><tt class="py-op">,</tt> <tt class="py-string">'RT'</tt><tt class="py-op">,</tt> <tt class="py-string">'RL'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line">    <tt class="py-comment"># These keys occur for references</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line"> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line"> </tt>
<a name="Record"></a><div id="Record-def"><a name="L73"></a><tt class="py-lineno"> 73</tt> <a class="py-toggle" href="#" id="Record-toggle" onclick="return toggle('Record');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.motifs.transfac.Record-class.html">Record</a><tt class="py-op">(</tt><tt class="py-base-class">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Record-expanded"><a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">    <tt class="py-docstring">"""Store the information in a TRANSFAC matrix table.</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line"><tt class="py-docstring">    The record inherits from a list containing the individual motifs.</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line"><tt class="py-docstring">    Attributes:</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line"><tt class="py-docstring">     - version - The version number, corresponding to the 'VV' field</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line"><tt class="py-docstring">       in the TRANSFAC file;</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line"> </tt>
<a name="Record.__init__"></a><div id="Record.__init__-def"><a name="L84"></a><tt class="py-lineno"> 84</tt> <a class="py-toggle" href="#" id="Record.__init__-toggle" onclick="return toggle('Record.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.transfac.Record-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record.__init__-expanded"><a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize."""</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-11" class="py-name" targets="Function Bio.Cluster._cluster.version()=Bio.Cluster._cluster-module.html#version,Method Bio.GenBank._FeatureConsumer.version()=Bio.GenBank._FeatureConsumer-class.html#version,Method Bio.GenBank._RecordConsumer.version()=Bio.GenBank._RecordConsumer-class.html#version,Method Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.version()=Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer-class.html#version,Variable Bio.motifs.jaspar.Motif.version=Bio.motifs.jaspar.Motif-class.html#version"><a title="Bio.Cluster._cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-11', 'version', 'link-11');">version</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line"> </tt>
<a name="Record.__str__"></a><div id="Record.__str__-def"><a name="L88"></a><tt class="py-lineno"> 88</tt> <a class="py-toggle" href="#" id="Record.__str__-toggle" onclick="return toggle('Record.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.transfac.Record-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record.__str__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record.__str__-expanded"><a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line">        <tt class="py-docstring">"""Turn the TRANSFAC matrix into a string."""</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-12" class="py-name" targets="Function Bio.AlignIO.write()=Bio.AlignIO-module.html#write,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.write()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#write,Method Bio.Phylo.CDAOIO.Writer.write()=Bio.Phylo.CDAOIO.Writer-class.html#write,Function Bio.Phylo.CDAOIO.write()=Bio.Phylo.CDAOIO-module.html#write,Method Bio.Phylo.NeXMLIO.Writer.write()=Bio.Phylo.NeXMLIO.Writer-class.html#write,Function Bio.Phylo.NeXMLIO.write()=Bio.Phylo.NeXMLIO-module.html#write,Method Bio.Phylo.NewickIO.Writer.write()=Bio.Phylo.NewickIO.Writer-class.html#write,Function Bio.Phylo.NewickIO.write()=Bio.Phylo.NewickIO-module.html#write,Function Bio.Phylo.NexusIO.write()=Bio.Phylo.NexusIO-module.html#write,Method Bio.Phylo.PhyloXMLIO.Writer.write()=Bio.Phylo.PhyloXMLIO.Writer-class.html#write,Function Bio.Phylo.PhyloXMLIO.write()=Bio.Phylo.PhyloXMLIO-module.html#write,Function Bio.Phylo._io.write()=Bio.Phylo._io-module.html#write,Function Bio.SearchIO.write()=Bio.SearchIO-module.html#write,Function Bio.SeqIO.write()=Bio.SeqIO-module.html#write,Method Bio.bgzf.BgzfWriter.write()=Bio.bgzf.BgzfWriter-class.html#write,Function Bio.motifs.jaspar.write()=Bio.motifs.jaspar-module.html#write,Function Bio.motifs.transfac.write()=Bio.motifs.transfac-module.html#write,Function Bio.motifs.write()=Bio.motifs-module.html#write,Method Bio.phenotype.phen_micro.JsonWriter.write()=Bio.phenotype.phen_micro.JsonWriter-class.html#write,Function Bio.phenotype.write()=Bio.phenotype-module.html#write"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-12', 'write', 'link-12');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line"> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line"> </tt>
<a name="read"></a><div id="read-def"><a name="L93"></a><tt class="py-lineno"> 93</tt> <a class="py-toggle" href="#" id="read-toggle" onclick="return toggle('read');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.transfac-module.html#read">read</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="read-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="read-expanded"><a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line">    <tt class="py-docstring">"""Parse a transfac format handle into a Record object."""</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line">    <tt id="link-13" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.annotations=BioSQL.BioSeq.DBSeqRecord-class.html#annotations"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-13', 'annotations', 'link-13');">annotations</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line">    <tt class="py-name">references</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line">    <tt class="py-name">counts</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line">    <tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt id="link-14" class="py-name" targets="Class Bio.Affy.CelFile.Record=Bio.Affy.CelFile.Record-class.html,Module Bio.Blast.Record=Bio.Blast.Record-module.html,Class Bio.Cluster.Record=Bio.Cluster.Record-class.html,Class Bio.Compass.Record=Bio.Compass.Record-class.html,Class Bio.Emboss.Primer3.Record=Bio.Emboss.Primer3.Record-class.html,Class Bio.ExPASy.Enzyme.Record=Bio.ExPASy.Enzyme.Record-class.html,Class Bio.ExPASy.Prodoc.Record=Bio.ExPASy.Prodoc.Record-class.html,Class Bio.ExPASy.Prosite.Record=Bio.ExPASy.Prosite.Record-class.html,Class Bio.ExPASy.ScanProsite.Record=Bio.ExPASy.ScanProsite.Record-class.html,Class Bio.ExPASy.cellosaurus.Record=Bio.ExPASy.cellosaurus.Record-class.html,Module Bio.GenBank.Record=Bio.GenBank.Record-module.html,Class Bio.GenBank.Record.Record=Bio.GenBank.Record.Record-class.html,Module Bio.Geo.Record=Bio.Geo.Record-module.html,Class Bio.Geo.Record.Record=Bio.Geo.Record.Record-class.html,Class Bio.KEGG.Compound.Record=Bio.KEGG.Compound.Record-class.html,Class Bio.KEGG.Enzyme.Record=Bio.KEGG.Enzyme.Record-class.html,Class Bio.KEGG.Gene.Record=Bio.KEGG.Gene.Record-class.html,Class Bio.Medline.Record=Bio.Medline.Record-class.html,Class Bio.PopGen.GenePop.LargeFileParser.Record=Bio.PopGen.GenePop.LargeFileParser.Record-class.html,Class Bio.PopGen.GenePop.Record=Bio.PopGen.GenePop.Record-class.html,Class Bio.SCOP.Cla.Record=Bio.SCOP.Cla.Record-class.html,Class Bio.SCOP.Des.Record=Bio.SCOP.Des.Record-class.html,Class Bio.SCOP.Dom.Record=Bio.SCOP.Dom.Record-class.html,Class Bio.SCOP.Hie.Record=Bio.SCOP.Hie.Record-class.html,Class Bio.Sequencing.Phd.Record=Bio.Sequencing.Phd.Record-class.html,Class Bio.SwissProt.KeyWList.Record=Bio.SwissProt.KeyWList.Record-class.html,Class Bio.SwissProt.Record=Bio.SwissProt.Record-class.html,Class Bio.UniGene.Record=Bio.UniGene.Record-class.html,Class Bio.motifs.alignace.Record=Bio.motifs.alignace.Record-class.html,Class Bio.motifs.jaspar.Record=Bio.motifs.jaspar.Record-class.html,Class Bio.motifs.mast.Record=Bio.motifs.mast.Record-class.html,Class Bio.motifs.meme.Record=Bio.motifs.meme.Record-class.html,Class Bio.motifs.minimal.Record=Bio.motifs.minimal.Record-class.html,Class Bio.motifs.transfac.Record=Bio.motifs.transfac.Record-class.html"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.ExPASy.cellosaurus.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.KEGG.Gene.Record
Bio.Medline.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.minimal.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-14', 'Record', 'link-14');">Record</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-15" class="py-name" targets="Method Bio.Seq.Seq.strip()=Bio.Seq.Seq-class.html#strip"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-15', 'strip', 'link-15');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line">            <tt class="py-keyword">continue</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">        <tt class="py-name">key_value</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-16" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-16', 'split', 'link-16');">split</a></tt><tt class="py-op">(</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line">        <tt class="py-name">key</tt> <tt class="py-op">=</tt> <tt class="py-name">key_value</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-17" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-17', 'strip', 'link-15');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt><tt class="py-op">,</tt> <tt class="py-string">'The key value of a TRANSFAC motif line should have 2 characters: "{0:s}"'</tt><tt class="py-op">.</tt><tt id="link-18" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.format()=Bio.Align.MultipleSeqAlignment-class.html#format,Method Bio.Phylo.BaseTree.Tree.format()=Bio.Phylo.BaseTree.Tree-class.html#format,Method Bio.Restriction.Restriction.RestrictionBatch.format()=Bio.Restriction.Restriction.RestrictionBatch-class.html#format,Method Bio.SeqRecord.SeqRecord.format()=Bio.SeqRecord.SeqRecord-class.html#format,Method Bio.motifs.Motif.format()=Bio.motifs.Motif-class.html#format"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-18', 'format', 'link-18');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">key_value</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line">            <tt id="link-19" class="py-name" targets="Variable Bio.Entrez.Parser.Consumer.value=Bio.Entrez.Parser.Consumer-class.html#value,Variable Bio.Entrez.Parser.DictionaryConsumer.value=Bio.Entrez.Parser.DictionaryConsumer-class.html#value,Variable Bio.Entrez.Parser.ErrorConsumer.value=Bio.Entrez.Parser.ErrorConsumer-class.html#value,Variable Bio.Entrez.Parser.IntegerConsumer.value=Bio.Entrez.Parser.IntegerConsumer-class.html#value,Variable Bio.Entrez.Parser.ListConsumer.value=Bio.Entrez.Parser.ListConsumer-class.html#value,Variable Bio.Entrez.Parser.StringConsumer.value=Bio.Entrez.Parser.StringConsumer-class.html#value,Method Bio.Phylo.PhyloXMLIO.Writer.value()=Bio.Phylo.PhyloXMLIO.Writer-class.html#value"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-19', 'value', 'link-19');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">key_value</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-20" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-20', 'strip', 'link-15');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">partition</tt><tt class="py-op">(</tt><tt class="py-string">'  '</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">                <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">                    <tt class="py-string">'A TRANSFAC motif line should have 2 spaces between key and value columns: "{0:s}"'</tt><tt class="py-op">.</tt><tt id="link-21" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-21', 'format', 'link-18');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">                    <tt id="link-22" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-22', 'BiopythonParserWarning', 'link-1');">BiopythonParserWarning</a></tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">                <tt class="py-op">)</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-op">==</tt> <tt class="py-string">'VV'</tt><tt class="py-op">:</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-23" class="py-name"><a title="Bio.Cluster._cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-23', 'version', 'link-11');">version</a></tt> <tt class="py-op">=</tt> <tt id="link-24" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-24', 'value', 'link-19');">value</a></tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'P0'</tt><tt class="py-op">,</tt> <tt class="py-string">'PO'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># Old TRANSFAC files use PO instead of P0</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">            <tt class="py-name">counts</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-25" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-25', 'value', 'link-19');">value</a></tt><tt class="py-op">.</tt><tt id="link-26" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-26', 'split', 'link-16');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">4</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-op">[</tt><tt class="py-string">'A'</tt><tt class="py-op">,</tt> <tt class="py-string">'C'</tt><tt class="py-op">,</tt> <tt class="py-string">'G'</tt><tt class="py-op">,</tt> <tt class="py-string">'T'</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">'A TRANSFAC matrix "{0:s}" line should be '</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line">                                 <tt class="py-string">'followed by "A C G T": '</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">                                 <tt class="py-string">'"{0:s}"'</tt><tt class="py-op">.</tt><tt id="link-27" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-27', 'format', 'link-18');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">            <tt id="link-28" class="py-name" targets="Method Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length()=Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer-class.html#length"><a title="Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-28', 'length', 'link-28');">length</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">c</tt> <tt class="py-keyword">in</tt> <tt class="py-string">"ACGT"</tt><tt class="py-op">:</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">                <tt class="py-name">counts</tt><tt class="py-op">[</tt><tt class="py-name">c</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-29" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-29', 'strip', 'link-15');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line">                <tt class="py-name">key_value</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-30" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-30', 'split', 'link-16');">split</a></tt><tt class="py-op">(</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line">                <tt class="py-name">key</tt> <tt class="py-op">=</tt> <tt class="py-name">key_value</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-31" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-31', 'strip', 'link-15');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">key_value</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line">                    <tt id="link-32" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-32', 'value', 'link-19');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">key_value</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-33" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-33', 'strip', 'link-15');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">partition</tt><tt class="py-op">(</tt><tt class="py-string">'  '</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line">                        <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line">                            <tt class="py-string">'A TRANSFAC motif line should have 2 spaces between key and value columns: "{0:s}"'</tt><tt class="py-op">.</tt><tt id="link-34" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-34', 'format', 'link-18');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line">                                <tt class="py-name">line</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line">                            <tt id="link-35" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-35', 'BiopythonParserWarning', 'link-1');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line">                <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">                    <tt id="link-36" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-36', 'i', 'link-36');">i</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">)</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line">                <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-37" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-37', 'length', 'link-28');">length</a></tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt> <tt class="py-keyword">and</tt> <tt id="link-38" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-38', 'i', 'link-36');">i</a></tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line">                    <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line">                        <tt class="py-string">'A TRANSFAC matrix should start with "01" as first row of the matrix, '</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line">                        <tt class="py-string">'but this matrix uses "00": "{0:s}"'</tt><tt class="py-op">.</tt><tt id="link-39" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-39', 'format', 'link-18');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line">                        <tt id="link-40" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-40', 'BiopythonParserWarning', 'link-1');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">                    <tt id="link-41" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-41', 'length', 'link-28');">length</a></tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-42" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-42', 'i', 'link-36');">i</a></tt> <tt class="py-op">!=</tt> <tt id="link-43" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-43', 'length', 'link-28');">length</a></tt><tt class="py-op">:</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">'The TRANSFAC matrix row number does not '</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">                                     <tt class="py-string">'match the position in the matrix: '</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line">                                     <tt class="py-string">'"{0:s}"'</tt><tt class="py-op">.</tt><tt id="link-44" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-44', 'format', 'link-18');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">                    <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">                        <tt class="py-string">'A TRANSFAC matrix line should have a 2 digit key at the start of the lin ("{0:02d}"), '</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">                        <tt class="py-string">'but this matrix uses "{0:d}": "{1:s}".'</tt><tt class="py-op">.</tt><tt id="link-45" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-45', 'format', 'link-18');">format</a></tt><tt class="py-op">(</tt><tt id="link-46" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-46', 'i', 'link-36');">i</a></tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line">                        <tt id="link-47" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-47', 'BiopythonParserWarning', 'link-1');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">                <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">key_value</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt><tt class="py-op">,</tt> <tt class="py-string">'A TRANSFAC matrix line should have a key and a value: "{0:s}"'</tt><tt class="py-op">.</tt><tt id="link-48" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-48', 'format', 'link-18');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">                <tt id="link-49" class="py-name" targets="Method Bio.Crystal.Crystal.values()=Bio.Crystal.Crystal-class.html#values,Method Bio.File._IndexedSeqFileDict.values()=Bio.File._IndexedSeqFileDict-class.html#values,Method Bio.Phylo.PhyloXML.Events.values()=Bio.Phylo.PhyloXML.Events-class.html#values,Method BioSQL.BioSeqDatabase.BioSeqDatabase.values()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#values,Method BioSQL.BioSeqDatabase.DBServer.values()=BioSQL.BioSeqDatabase.DBServer-class.html#values"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-49', 'values', 'link-49');">values</a></tt> <tt class="py-op">=</tt> <tt id="link-50" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-50', 'value', 'link-19');">value</a></tt><tt class="py-op">.</tt><tt id="link-51" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-51', 'split', 'link-16');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">4</tt><tt class="py-op">]</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-52" class="py-name"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-52', 'values', 'link-49');">values</a></tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-number">4</tt><tt class="py-op">:</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">'A TRANSFAC matrix line should have a '</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">                                     <tt class="py-string">'value for each nucleotide '</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">                                     <tt class="py-string">'(A, C, G and T): "{0:s}"'</tt><tt class="py-op">.</tt><tt id="link-53" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-53', 'format', 'link-18');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">c</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-string">"ACGT"</tt><tt class="py-op">,</tt> <tt id="link-54" class="py-name"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-54', 'values', 'link-49');">values</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">                    <tt class="py-name">counts</tt><tt class="py-op">[</tt><tt class="py-name">c</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-55" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-55', 'append', 'link-55');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">v</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt> <tt class="py-op">==</tt> <tt class="py-string">'XX'</tt><tt class="py-op">:</tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">key</tt> <tt class="py-op">==</tt> <tt class="py-string">'RN'</tt><tt class="py-op">:</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">            <tt id="link-56" class="py-name" targets="Method Bio.Crystal.Chain.index()=Bio.Crystal.Chain-class.html#index,Method Bio.SCOP.Raf.SeqMap.index()=Bio.SCOP.Raf.SeqMap-class.html#index,Method Bio.SearchIO._model.hit.Hit.index()=Bio.SearchIO._model.hit.Hit-class.html#index,Method Bio.SearchIO._model.query.QueryResult.index()=Bio.SearchIO._model.query.QueryResult-class.html#index,Function Bio.SearchIO.index()=Bio.SearchIO-module.html#index,Method Bio.Seq.MutableSeq.index()=Bio.Seq.MutableSeq-class.html#index,Function Bio.SeqIO.index()=Bio.SeqIO-module.html#index"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-56', 'index', 'link-56');">index</a></tt><tt class="py-op">,</tt> <tt class="py-name">separator</tt><tt class="py-op">,</tt> <tt id="link-57" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.accession()=Bio.GenBank._FeatureConsumer-class.html#accession,Method Bio.GenBank._RecordConsumer.accession()=Bio.GenBank._RecordConsumer-class.html#accession,Method Bio.Phylo.PhyloXMLIO.Parser.accession()=Bio.Phylo.PhyloXMLIO.Parser-class.html#accession,Method Bio.Phylo.PhyloXMLIO.Writer.accession()=Bio.Phylo.PhyloXMLIO.Writer-class.html#accession"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-57', 'accession', 'link-57');">accession</a></tt> <tt class="py-op">=</tt> <tt id="link-58" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-58', 'value', 'link-19');">value</a></tt><tt class="py-op">.</tt><tt class="py-name">partition</tt><tt class="py-op">(</tt><tt class="py-string">";"</tt><tt class="py-op">)</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-59" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-59', 'index', 'link-56');">index</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-string">'['</tt><tt class="py-op">:</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">'The index "{0:s}" in a TRANSFAC RN line '</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line">                                 <tt class="py-string">'should start with a '</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line">                                 <tt class="py-string">'"[": "{0:s}"'</tt><tt class="py-op">.</tt><tt id="link-60" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-60', 'format', 'link-18');">format</a></tt><tt class="py-op">(</tt><tt id="link-61" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-61', 'index', 'link-56');">index</a></tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-62" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-62', 'index', 'link-56');">index</a></tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-string">']'</tt><tt class="py-op">:</tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">'The index "{0:s}" in a TRANSFAC RN line '</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line">                                 <tt class="py-string">'should end with a '</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line">                                 <tt class="py-string">'"]": "{0:s}"'</tt><tt class="py-op">.</tt><tt id="link-63" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-63', 'format', 'link-18');">format</a></tt><tt class="py-op">(</tt><tt id="link-64" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-64', 'index', 'link-56');">index</a></tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line">            <tt id="link-65" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-65', 'index', 'link-56');">index</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-66" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-66', 'index', 'link-56');">index</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">references</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt id="link-67" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-67', 'index', 'link-56');">index</a></tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">'The index "{0:d}" of the TRANSFAC RN line '</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">                                 <tt class="py-string">'does not match the current number of seen '</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">                                 <tt class="py-string">'references '</tt><tt class="py-string">'"{1:d}": "{2:s}"'</tt><tt class="py-op">.</tt><tt id="link-68" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-68', 'format', 'link-18');">format</a></tt><tt class="py-op">(</tt><tt id="link-69" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-69', 'index', 'link-56');">index</a></tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">references</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">            <tt id="link-70" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.reference()=Bio.Phylo.PhyloXMLIO.Parser-class.html#reference,Method Bio.Phylo.PhyloXMLIO.Writer.reference()=Bio.Phylo.PhyloXMLIO.Writer-class.html#reference,Method Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.reference()=Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer-class.html#reference"><a title="Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference
Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.reference" class="py-name" href="#" onclick="return doclink('link-70', 'reference', 'link-70');">reference</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-name">key</tt><tt class="py-op">:</tt> <tt id="link-71" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-71', 'value', 'link-19');">value</a></tt><tt class="py-op">}</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">            <tt class="py-name">references</tt><tt class="py-op">.</tt><tt id="link-72" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-72', 'append', 'link-55');">append</a></tt><tt class="py-op">(</tt><tt id="link-73" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference
Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.reference" class="py-name" href="#" onclick="return doclink('link-73', 'reference', 'link-70');">reference</a></tt><tt class="py-op">)</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">key</tt> <tt class="py-op">==</tt> <tt class="py-string">'//'</tt><tt class="py-op">:</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">counts</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">                <tt class="py-name">motif</tt> <tt class="py-op">=</tt> <tt id="link-74" class="py-name" targets="Class Bio.motifs.Motif=Bio.motifs.Motif-class.html,Class Bio.motifs.jaspar.Motif=Bio.motifs.jaspar.Motif-class.html,Class Bio.motifs.meme.Motif=Bio.motifs.meme.Motif-class.html,Class Bio.motifs.transfac.Motif=Bio.motifs.transfac.Motif-class.html"><a title="Bio.motifs.Motif
Bio.motifs.jaspar.Motif
Bio.motifs.meme.Motif
Bio.motifs.transfac.Motif" class="py-name" href="#" onclick="return doclink('link-74', 'Motif', 'link-74');">Motif</a></tt><tt class="py-op">(</tt><tt id="link-75" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.alphabet=Bio.SearchIO._model.hsp.HSP-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSPFragment.alphabet=Bio.SearchIO._model.hsp.HSPFragment-class.html#alphabet"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-75', 'alphabet', 'link-75');">alphabet</a></tt><tt class="py-op">=</tt><tt id="link-76" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-76', 'IUPAC', 'link-6');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-77" class="py-name" targets="Variable Bio.Alphabet.IUPAC.unambiguous_dna=Bio.Alphabet.IUPAC-module.html#unambiguous_dna"><a title="Bio.Alphabet.IUPAC.unambiguous_dna" class="py-name" href="#" onclick="return doclink('link-77', 'unambiguous_dna', 'link-77');">unambiguous_dna</a></tt><tt class="py-op">,</tt> <tt class="py-name">counts</tt><tt class="py-op">=</tt><tt class="py-name">counts</tt><tt class="py-op">)</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">                <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt id="link-78" class="py-name" targets="Method Bio.File._IndexedSeqFileDict.update()=Bio.File._IndexedSeqFileDict-class.html#update,Method Bio.Index._InMemoryIndex.update()=Bio.Index._InMemoryIndex-class.html#update,Method Bio.SeqRecord._RestrictedDict.update()=Bio.SeqRecord._RestrictedDict-class.html#update"><a title="Bio.File._IndexedSeqFileDict.update
Bio.Index._InMemoryIndex.update
Bio.SeqRecord._RestrictedDict.update" class="py-name" href="#" onclick="return doclink('link-78', 'update', 'link-78');">update</a></tt><tt class="py-op">(</tt><tt id="link-79" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-79', 'annotations', 'link-13');">annotations</a></tt><tt class="py-op">)</tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line">                <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt class="py-name">references</tt> <tt class="py-op">=</tt> <tt class="py-name">references</tt> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">                <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-80" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-80', 'append', 'link-55');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">motif</tt><tt class="py-op">)</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">            <tt id="link-81" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-81', 'annotations', 'link-13');">annotations</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">            <tt class="py-name">references</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt id="link-82" class="py-name"><a title="Bio.motifs.Motif
Bio.motifs.jaspar.Motif
Bio.motifs.meme.Motif
Bio.motifs.transfac.Motif" class="py-name" href="#" onclick="return doclink('link-82', 'Motif', 'link-74');">Motif</a></tt><tt class="py-op">.</tt><tt id="link-83" class="py-name"><a title="Bio.motifs.transfac.Motif.reference_keys" class="py-name" href="#" onclick="return doclink('link-83', 'reference_keys', 'link-9');">reference_keys</a></tt><tt class="py-op">:</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">            <tt id="link-84" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference
Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.reference" class="py-name" href="#" onclick="return doclink('link-84', 'reference', 'link-70');">reference</a></tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-85" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-85', 'value', 'link-19');">value</a></tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt id="link-86" class="py-name"><a title="Bio.motifs.Motif
Bio.motifs.jaspar.Motif
Bio.motifs.meme.Motif
Bio.motifs.transfac.Motif" class="py-name" href="#" onclick="return doclink('link-86', 'Motif', 'link-74');">Motif</a></tt><tt class="py-op">.</tt><tt id="link-87" class="py-name"><a title="Bio.motifs.transfac.Motif.multiple_value_keys" class="py-name" href="#" onclick="return doclink('link-87', 'multiple_value_keys', 'link-7');">multiple_value_keys</a></tt><tt class="py-op">:</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt id="link-88" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-88', 'annotations', 'link-13');">annotations</a></tt><tt class="py-op">:</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">                <tt id="link-89" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-89', 'annotations', 'link-13');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">            <tt id="link-90" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-90', 'annotations', 'link-13');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-91" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-91', 'append', 'link-55');">append</a></tt><tt class="py-op">(</tt><tt id="link-92" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-92', 'value', 'link-19');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">            <tt id="link-93" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-93', 'annotations', 'link-13');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-94" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-94', 'value', 'link-19');">value</a></tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">record</tt> </tt>
</div><a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line"> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line"> </tt>
<a name="write"></a><div id="write-def"><a name="L201"></a><tt class="py-lineno">201</tt> <a class="py-toggle" href="#" id="write-toggle" onclick="return toggle('write');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.transfac-module.html#write">write</a><tt class="py-op">(</tt><tt class="py-param">motifs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="write-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="write-expanded"><a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line">    <tt class="py-docstring">"""Write the representation of a motif in TRANSFAC format."""</tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line">    <tt class="py-name">blocks</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line">        <tt id="link-95" class="py-name"><a title="Bio.Cluster._cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-95', 'version', 'link-11');">version</a></tt> <tt class="py-op">=</tt> <tt id="link-96" class="py-name"><a title="Bio.motifs" class="py-name" href="#" onclick="return doclink('link-96', 'motifs', 'link-3');">motifs</a></tt><tt class="py-op">.</tt><tt id="link-97" class="py-name"><a title="Bio.Cluster._cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-97', 'version', 'link-11');">version</a></tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-98" class="py-name"><a title="Bio.Cluster._cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-98', 'version', 'link-11');">version</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line">            <tt class="py-name">block</tt> <tt class="py-op">=</tt> <tt class="py-string">"""\</tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line"><tt class="py-string">VV  %s</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line"><tt class="py-string">XX</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line"><tt class="py-string">//</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line"><tt class="py-string">"""</tt> <tt class="py-op">%</tt> <tt id="link-99" class="py-name"><a title="Bio.Cluster._cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-99', 'version', 'link-11');">version</a></tt> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">            <tt class="py-name">blocks</tt><tt class="py-op">.</tt><tt id="link-100" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-100', 'append', 'link-55');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">block</tt><tt class="py-op">)</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line">    <tt id="link-101" class="py-name"><a title="Bio.motifs.transfac.Motif.multiple_value_keys" class="py-name" href="#" onclick="return doclink('link-101', 'multiple_value_keys', 'link-7');">multiple_value_keys</a></tt> <tt class="py-op">=</tt> <tt id="link-102" class="py-name"><a title="Bio.motifs.Motif
Bio.motifs.jaspar.Motif
Bio.motifs.meme.Motif
Bio.motifs.transfac.Motif" class="py-name" href="#" onclick="return doclink('link-102', 'Motif', 'link-74');">Motif</a></tt><tt class="py-op">.</tt><tt id="link-103" class="py-name"><a title="Bio.motifs.transfac.Motif.multiple_value_keys" class="py-name" href="#" onclick="return doclink('link-103', 'multiple_value_keys', 'link-7');">multiple_value_keys</a></tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line">    <tt class="py-name">sections</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">'AC'</tt><tt class="py-op">,</tt> <tt class="py-string">'AS'</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  <tt class="py-comment"># Accession</tt> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-string">'ID'</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>        <tt class="py-comment"># ID</tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-string">'DT'</tt><tt class="py-op">,</tt> <tt class="py-string">'CO'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>   <tt class="py-comment"># Date, copyright</tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-string">'NA'</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>        <tt class="py-comment"># Name</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-string">'DE'</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>        <tt class="py-comment"># Short factor description</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-string">'TY'</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>        <tt class="py-comment"># Type</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-string">'OS'</tt><tt class="py-op">,</tt> <tt class="py-string">'OC'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>   <tt class="py-comment"># Organism</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-string">'HP'</tt><tt class="py-op">,</tt> <tt class="py-string">'HC'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>   <tt class="py-comment"># Superfamilies, subfamilies</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-string">'BF'</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>        <tt class="py-comment"># Binding factors</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-string">'P0'</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>        <tt class="py-comment"># Frequency matrix</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-string">'BA'</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>        <tt class="py-comment"># Statistical basis</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-string">'BS'</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>        <tt class="py-comment"># Factor binding sites</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-string">'CC'</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>        <tt class="py-comment"># Comments</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-string">'DR'</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>        <tt class="py-comment"># External databases</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-string">'OV'</tt><tt class="py-op">,</tt> <tt class="py-string">'PV'</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  <tt class="py-comment"># Versions</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">                <tt class="py-op">)</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">motif</tt> <tt class="py-keyword">in</tt> <tt id="link-104" class="py-name"><a title="Bio.motifs" class="py-name" href="#" onclick="return doclink('link-104', 'motifs', 'link-3');">motifs</a></tt><tt class="py-op">:</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line">        <tt class="py-name">lines</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">section</tt> <tt class="py-keyword">in</tt> <tt class="py-name">sections</tt><tt class="py-op">:</tt> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line">            <tt class="py-name">blank</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">section</tt><tt class="py-op">:</tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-op">==</tt> <tt class="py-string">'P0'</tt><tt class="py-op">:</tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line">                    <tt class="py-comment"># Frequency matrix</tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line">                    <tt id="link-105" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-105', 'length', 'link-28');">length</a></tt> <tt class="py-op">=</tt> <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt id="link-106" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-106', 'length', 'link-28');">length</a></tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt id="link-107" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-107', 'length', 'link-28');">length</a></tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line">                        <tt class="py-keyword">continue</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line">                    <tt id="link-108" class="py-name" targets="Method Bio.FSSP.FSSPAlignDict.sequence()=Bio.FSSP.FSSPAlignDict-class.html#sequence,Method Bio.GenBank._FeatureConsumer.sequence()=Bio.GenBank._FeatureConsumer-class.html#sequence,Method Bio.GenBank._RecordConsumer.sequence()=Bio.GenBank._RecordConsumer-class.html#sequence,Method Bio.Phylo.PhyloXMLIO.Writer.sequence()=Bio.Phylo.PhyloXMLIO.Writer-class.html#sequence"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-108', 'sequence', 'link-108');">sequence</a></tt> <tt class="py-op">=</tt> <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt id="link-109" class="py-name" targets="Variable Bio.motifs.Motif.degenerate_consensus=Bio.motifs.Motif-class.html#degenerate_consensus,Variable Bio.motifs.matrix.GenericPositionMatrix.degenerate_consensus=Bio.motifs.matrix.GenericPositionMatrix-class.html#degenerate_consensus"><a title="Bio.motifs.Motif.degenerate_consensus
Bio.motifs.matrix.GenericPositionMatrix.degenerate_consensus" class="py-name" href="#" onclick="return doclink('link-109', 'degenerate_consensus', 'link-109');">degenerate_consensus</a></tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line">                    <tt id="link-110" class="py-name" targets="Variable Bio.Alphabet.Alphabet.letters=Bio.Alphabet.Alphabet-class.html#letters,Variable Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters=Bio.Alphabet.IUPAC.ExtendedIUPACDNA-class.html#letters,Variable Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters=Bio.Alphabet.IUPAC.ExtendedIUPACProtein-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters=Bio.Alphabet.IUPAC.IUPACAmbiguousDNA-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters=Bio.Alphabet.IUPAC.IUPACAmbiguousRNA-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACProtein.letters=Bio.Alphabet.IUPAC.IUPACProtein-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters=Bio.Alphabet.IUPAC.IUPACUnambiguousDNA-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters=Bio.Alphabet.IUPAC.IUPACUnambiguousRNA-class.html#letters,Variable Bio.Alphabet.Reduced.HPModel.letters=Bio.Alphabet.Reduced.HPModel-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy10.letters=Bio.Alphabet.Reduced.Murphy10-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy15.letters=Bio.Alphabet.Reduced.Murphy15-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy4.letters=Bio.Alphabet.Reduced.Murphy4-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy8.letters=Bio.Alphabet.Reduced.Murphy8-class.html#letters,Variable Bio.Alphabet.Reduced.PC5.letters=Bio.Alphabet.Reduced.PC5-class.html#letters,Variable Bio.Alphabet.SecondaryStructure.letters=Bio.Alphabet.SecondaryStructure-class.html#letters,Variable Bio.Alphabet.SingleLetterAlphabet.letters=Bio.Alphabet.SingleLetterAlphabet-class.html#letters,Variable Bio.Alphabet.ThreeLetterProtein.letters=Bio.Alphabet.ThreeLetterProtein-class.html#letters,Variable Bio.codonalign.codonalphabet.CodonAlphabet.letters=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#letters"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-110', 'letters', 'link-110');">letters</a></tt> <tt class="py-op">=</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt class="py-name">motif</tt><tt class="py-op">.</tt><tt id="link-111" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-111', 'alphabet', 'link-75');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-112" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-112', 'letters', 'link-110');">letters</a></tt><tt class="py-op">)</tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line">                    <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-string">"      "</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-string">"P0"</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt id="link-113" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-113', 'letters', 'link-110');">letters</a></tt><tt class="py-op">)</tt> </tt>
<a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line"> </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line">                    <tt class="py-name">lines</tt><tt class="py-op">.</tt><tt id="link-114" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-114', 'append', 'link-55');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt id="link-115" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-115', 'i', 'link-36');">i</a></tt> <tt class="py-keyword">in</tt> <tt id="link-116" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-116', 'range', 'link-116');">range</a></tt><tt class="py-op">(</tt><tt id="link-117" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-117', 'length', 'link-28');">length</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line">                        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-string">" "</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line">                            <tt class="py-op">[</tt><tt class="py-string">"%02.d"</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line">                            <tt class="py-op">[</tt><tt class="py-string">"%6.20g"</tt> <tt class="py-keyword">for</tt> <tt class="py-name">l</tt> <tt class="py-keyword">in</tt> <tt id="link-118" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-118', 'letters', 'link-110');">letters</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line">                            <tt class="py-string">"      %s"</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line">                        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> <tt class="py-op">%</tt> <tt class="py-name">tuple</tt><tt class="py-op">(</tt> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line">                            <tt class="py-op">[</tt><tt id="link-119" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-119', 'i', 'link-36');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line">                            <tt class="py-op">[</tt><tt class="py-name">motif</tt><tt class="py-op">.</tt><tt class="py-name">counts</tt><tt class="py-op">[</tt><tt class="py-name">l</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt id="link-120" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-120', 'i', 'link-36');">i</a></tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt class="py-name">l</tt> <tt class="py-keyword">in</tt> <tt id="link-121" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-121', 'letters', 'link-110');">letters</a></tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line">                            <tt class="py-op">[</tt><tt id="link-122" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-122', 'sequence', 'link-108');">sequence</a></tt><tt class="py-op">[</tt><tt id="link-123" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-123', 'i', 'link-36');">i</a></tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line">                        <tt class="py-op">)</tt> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line">                        <tt class="py-name">lines</tt><tt class="py-op">.</tt><tt id="link-124" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-124', 'append', 'link-55');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line">                    <tt class="py-name">blank</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line">                    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line">                        <tt id="link-125" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-125', 'value', 'link-19');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt id="link-126" class="py-name" targets="Method Bio.Crystal.Crystal.get()=Bio.Crystal.Crystal-class.html#get,Method Bio.Data.CodonTable.AmbiguousForwardTable.get()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#get,Method Bio.File._IndexedSeqFileDict.get()=Bio.File._IndexedSeqFileDict-class.html#get,Method Bio.File._IndexedSeqFileProxy.get()=Bio.File._IndexedSeqFileProxy-class.html#get,Method Bio.File._SQLiteManySeqFilesDict.get()=Bio.File._SQLiteManySeqFilesDict-class.html#get,Method Bio.Restriction.Restriction.RestrictionBatch.get()=Bio.Restriction.Restriction.RestrictionBatch-class.html#get,Method Bio.SearchIO._index.SearchIndexer.get()=Bio.SearchIO._index.SearchIndexer-class.html#get,Method Bio.SeqIO._index.SeqFileRandomAccess.get()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#get,Method Bio.SeqIO._index.SffRandomAccess.get()=Bio.SeqIO._index.SffRandomAccess-class.html#get,Method Bio.SeqIO._index.SffTrimedRandomAccess.get()=Bio.SeqIO._index.SffTrimedRandomAccess-class.html#get,Method Bio.SeqIO._index.UniprotRandomAccess.get()=Bio.SeqIO._index.UniprotRandomAccess-class.html#get"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-126', 'get', 'link-126');">get</a></tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">)</tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line">                    <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line">                        <tt id="link-127" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-127', 'value', 'link-19');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt id="link-128" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-128', 'value', 'link-19');">value</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt id="link-129" class="py-name"><a title="Bio.motifs.transfac.Motif.multiple_value_keys" class="py-name" href="#" onclick="return doclink('link-129', 'multiple_value_keys', 'link-7');">multiple_value_keys</a></tt><tt class="py-op">:</tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">                            <tt class="py-keyword">for</tt> <tt class="py-name">v</tt> <tt class="py-keyword">in</tt> <tt id="link-130" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-130', 'value', 'link-19');">value</a></tt><tt class="py-op">:</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line">                                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-string">"%s  %s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt><tt class="py-op">)</tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">                                <tt class="py-name">lines</tt><tt class="py-op">.</tt><tt id="link-131" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-131', 'append', 'link-55');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">                        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line">                            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-string">"%s  %s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt id="link-132" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-132', 'value', 'link-19');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line">                            <tt class="py-name">lines</tt><tt class="py-op">.</tt><tt id="link-133" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-133', 'append', 'link-55');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line">                        <tt class="py-name">blank</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-op">==</tt> <tt class="py-string">'PV'</tt><tt class="py-op">:</tt> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line">                    <tt class="py-comment"># References</tt> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line">                    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line">                        <tt class="py-name">references</tt> <tt class="py-op">=</tt> <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt class="py-name">references</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line">                    <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line">                        <tt class="py-keyword">pass</tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line">                        <tt id="link-134" class="py-name" targets="Method Bio.Crystal.Crystal.keys()=Bio.Crystal.Crystal-class.html#keys,Variable Bio.Entrez.Parser.ListConsumer.keys=Bio.Entrez.Parser.ListConsumer-class.html#keys,Method Bio.File._IndexedSeqFileDict.keys()=Bio.File._IndexedSeqFileDict-class.html#keys,Method Bio.File._SQLiteManySeqFilesDict.keys()=Bio.File._SQLiteManySeqFilesDict-class.html#keys,Method Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys()=Bio.PDB.AbstractPropertyMap.AbstractPropertyMap-class.html#keys,Method Bio.Phylo.PhyloXML.Events.keys()=Bio.Phylo.PhyloXML.Events-class.html#keys,Method BioSQL.BioSeqDatabase.BioSeqDatabase.keys()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#keys,Method BioSQL.BioSeqDatabase.DBServer.keys()=BioSQL.BioSeqDatabase.DBServer-class.html#keys"><a title="Bio.Crystal.Crystal.keys
Bio.Entrez.Parser.ListConsumer.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-134', 'keys', 'link-134');">keys</a></tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-string">"RN"</tt><tt class="py-op">,</tt> <tt class="py-string">"RX"</tt><tt class="py-op">,</tt> <tt class="py-string">"RA"</tt><tt class="py-op">,</tt> <tt class="py-string">"RT"</tt><tt class="py-op">,</tt> <tt class="py-string">"RL"</tt><tt class="py-op">)</tt> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line">                        <tt class="py-keyword">for</tt> <tt id="link-135" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference
Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.reference" class="py-name" href="#" onclick="return doclink('link-135', 'reference', 'link-70');">reference</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">references</tt><tt class="py-op">:</tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line">                            <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt id="link-136" class="py-name"><a title="Bio.Crystal.Crystal.keys
Bio.Entrez.Parser.ListConsumer.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-136', 'keys', 'link-134');">keys</a></tt><tt class="py-op">:</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">                                <tt id="link-137" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-137', 'value', 'link-19');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-138" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference
Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.reference" class="py-name" href="#" onclick="return doclink('link-138', 'reference', 'link-70');">reference</a></tt><tt class="py-op">.</tt><tt id="link-139" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-139', 'get', 'link-126');">get</a></tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">)</tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line">                                <tt class="py-keyword">if</tt> <tt id="link-140" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-140', 'value', 'link-19');">value</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line">                                    <tt class="py-keyword">continue</tt> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">                                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-string">"%s  %s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt id="link-141" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-141', 'value', 'link-19');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">                                <tt class="py-name">lines</tt><tt class="py-op">.</tt><tt id="link-142" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-142', 'append', 'link-55');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">                                <tt class="py-name">blank</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">blank</tt><tt class="py-op">:</tt> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-string">'XX'</tt> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line">                <tt class="py-name">lines</tt><tt class="py-op">.</tt><tt id="link-143" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-143', 'append', 'link-55');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line">        <tt class="py-comment"># Finished this motif; glue the lines together</tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-string">"//"</tt> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line">        <tt class="py-name">lines</tt><tt class="py-op">.</tt><tt id="link-144" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-144', 'append', 'link-55');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line">        <tt class="py-name">block</tt> <tt class="py-op">=</tt> <tt class="py-string">"\n"</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">lines</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line">        <tt class="py-name">blocks</tt><tt class="py-op">.</tt><tt id="link-145" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-145', 'append', 'link-55');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">block</tt><tt class="py-op">)</tt> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line">    <tt class="py-comment"># Finished all motifs; glue the blocks together</tt> </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line">    <tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">blocks</tt><tt class="py-op">)</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">text</tt> </tt>
</div><a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Tue Dec 18 15:58:15 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
