<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.SeqIO.InsdcIO</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.SeqIO-module.html">Package&nbsp;SeqIO</a> ::
        Module&nbsp;InsdcIO
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.SeqIO.InsdcIO-module.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== MODULE DESCRIPTION ==================== -->
<h1 class="epydoc">Module InsdcIO</h1><p class="nomargin-top"><span class="codelink"><a href="Bio.SeqIO.InsdcIO-pysrc.html">source&nbsp;code</a></span></p>
<p>Bio.SeqIO support for the &quot;genbank&quot; and &quot;embl&quot; file formats.</p>
<p>You are expected to use this module via the Bio.SeqIO functions.
Note that internally this module calls Bio.GenBank to do the actual
parsing of GenBank, EMBL and IMGT files.</p>
<p>See Also:
International Nucleotide Sequence Database Collaboration
<a class="rst-reference external" href="http://www.insdc.org/" target="_top">http://www.insdc.org/</a></p>
<p>GenBank
<a class="rst-reference external" href="http://www.ncbi.nlm.nih.gov/Genbank/" target="_top">http://www.ncbi.nlm.nih.gov/Genbank/</a></p>
<p>EMBL Nucleotide Sequence Database
<a class="rst-reference external" href="http://www.ebi.ac.uk/embl/" target="_top">http://www.ebi.ac.uk/embl/</a></p>
<p>DDBJ (DNA Data Bank of Japan)
<a class="rst-reference external" href="http://www.ddbj.nig.ac.jp/" target="_top">http://www.ddbj.nig.ac.jp/</a></p>
<p>IMGT (use a variant of EMBL format with longer feature indents)
<a class="rst-reference external" href="http://imgt.cines.fr/download/LIGM-DB/userman_doc.html" target="_top">http://imgt.cines.fr/download/LIGM-DB/userman_doc.html</a>
<a class="rst-reference external" href="http://imgt.cines.fr/download/LIGM-DB/ftable_doc.html" target="_top">http://imgt.cines.fr/download/LIGM-DB/ftable_doc.html</a>
<a class="rst-reference external" href="http://www.ebi.ac.uk/imgt/hla/docs/manual.html" target="_top">http://www.ebi.ac.uk/imgt/hla/docs/manual.html</a></p>

<!-- ==================== CLASSES ==================== -->
<a name="section-Classes"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Classes</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Classes"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.SeqIO.InsdcIO._InsdcWriter-class.html" class="summary-name" onclick="show_private();">_InsdcWriter</a><br />
      Base class for GenBank and EMBL writers (PRIVATE).
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.SeqIO.InsdcIO.GenBankWriter-class.html" class="summary-name">GenBankWriter</a><br />
      GenBank writer.
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.SeqIO.InsdcIO.EmblWriter-class.html" class="summary-name">EmblWriter</a><br />
      EMBL writer.
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.SeqIO.InsdcIO.ImgtWriter-class.html" class="summary-name">ImgtWriter</a><br />
      IMGT writer (EMBL format variant).
    </td>
  </tr>
</table>
<!-- ==================== FUNCTIONS ==================== -->
<a name="section-Functions"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Functions</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Functions"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqIO.InsdcIO-module.html#GenBankIterator" class="summary-sig-name">GenBankIterator</a>(<span class="summary-sig-arg">handle</span>)</span><br />
      Break up a Genbank file into SeqRecord objects.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqIO.InsdcIO-pysrc.html#GenBankIterator">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqIO.InsdcIO-module.html#EmblIterator" class="summary-sig-name">EmblIterator</a>(<span class="summary-sig-arg">handle</span>)</span><br />
      Break up an EMBL file into SeqRecord objects.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqIO.InsdcIO-pysrc.html#EmblIterator">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqIO.InsdcIO-module.html#ImgtIterator" class="summary-sig-name">ImgtIterator</a>(<span class="summary-sig-arg">handle</span>)</span><br />
      Break up an IMGT file into SeqRecord objects.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqIO.InsdcIO-pysrc.html#ImgtIterator">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqIO.InsdcIO-module.html#GenBankCdsFeatureIterator" class="summary-sig-name">GenBankCdsFeatureIterator</a>(<span class="summary-sig-arg">handle</span>,
        <span class="summary-sig-arg">alphabet</span>=<span class="summary-sig-default">ProteinAlphabet()</span>)</span><br />
      Break up a Genbank file into SeqRecord objects for each CDS feature.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqIO.InsdcIO-pysrc.html#GenBankCdsFeatureIterator">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqIO.InsdcIO-module.html#EmblCdsFeatureIterator" class="summary-sig-name">EmblCdsFeatureIterator</a>(<span class="summary-sig-arg">handle</span>,
        <span class="summary-sig-arg">alphabet</span>=<span class="summary-sig-default">ProteinAlphabet()</span>)</span><br />
      Break up a EMBL file into SeqRecord objects for each CDS feature.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqIO.InsdcIO-pysrc.html#EmblCdsFeatureIterator">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqIO.InsdcIO-module.html#_insdc_feature_position_string" class="summary-sig-name" onclick="show_private();">_insdc_feature_position_string</a>(<span class="summary-sig-arg">pos</span>,
        <span class="summary-sig-arg">offset</span>=<span class="summary-sig-default">0</span>)</span><br />
      Build a GenBank/EMBL position string (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqIO.InsdcIO-pysrc.html#_insdc_feature_position_string">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_insdc_location_string_ignoring_strand_and_subfeatures"></a><span class="summary-sig-name">_insdc_location_string_ignoring_strand_and_subfeatures</span>(<span class="summary-sig-arg">location</span>,
        <span class="summary-sig-arg">rec_length</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqIO.InsdcIO-pysrc.html#_insdc_location_string_ignoring_strand_and_subfeatures">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqIO.InsdcIO-module.html#_insdc_location_string" class="summary-sig-name" onclick="show_private();">_insdc_location_string</a>(<span class="summary-sig-arg">location</span>,
        <span class="summary-sig-arg">rec_length</span>)</span><br />
      Build a GenBank/EMBL location from a (Compound) FeatureLocation (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqIO.InsdcIO-pysrc.html#_insdc_location_string">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
</table>
<!-- ==================== VARIABLES ==================== -->
<a name="section-Variables"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Variables</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Variables"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="__package__"></a><span class="summary-name">__package__</span> = <code title="'Bio.SeqIO'"><code class="variable-quote">'</code><code class="variable-string">Bio.SeqIO</code><code class="variable-quote">'</code></code>
    </td>
  </tr>
</table>
<!-- ==================== FUNCTION DETAILS ==================== -->
<a name="section-FunctionDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Function Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-FunctionDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="GenBankIterator"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">GenBankIterator</span>(<span class="sig-arg">handle</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqIO.InsdcIO-pysrc.html#GenBankIterator">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Break up a Genbank file into SeqRecord objects.</p>
<p>Every section from the LOCUS line to the terminating // becomes
a single SeqRecord with associated annotation and features.</p>
<p>Note that for genomes or chromosomes, there is typically only
one record.</p>
<p>This gets called internally by Bio.SeqIO for the GenBank file format:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio <span class="py-keyword">import</span> SeqIO
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">for</span> record <span class="py-keyword">in</span> SeqIO.parse(<span class="py-string">&quot;GenBank/cor6_6.gb&quot;</span>, <span class="py-string">&quot;gb&quot;</span>):
<span class="py-more">... </span>    <span class="py-keyword">print</span>(record.id)
<span class="py-more">...</span>
<span class="py-output">X55053.1</span>
<span class="py-output">X62281.1</span>
<span class="py-output">M81224.1</span>
<span class="py-output">AJ237582.1</span>
<span class="py-output">L31939.1</span>
<span class="py-output">AF297471.1</span></pre>
<p>Equivalently,</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>with open(<span class="py-string">&quot;GenBank/cor6_6.gb&quot;</span>) <span class="py-keyword">as</span> handle:
<span class="py-more">... </span>    <span class="py-keyword">for</span> record <span class="py-keyword">in</span> GenBankIterator(handle):
<span class="py-more">... </span>        <span class="py-keyword">print</span>(record.id)
<span class="py-more">...</span>
<span class="py-output">X55053.1</span>
<span class="py-output">X62281.1</span>
<span class="py-output">M81224.1</span>
<span class="py-output">AJ237582.1</span>
<span class="py-output">L31939.1</span>
<span class="py-output">AF297471.1</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="EmblIterator"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">EmblIterator</span>(<span class="sig-arg">handle</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqIO.InsdcIO-pysrc.html#EmblIterator">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Break up an EMBL file into SeqRecord objects.</p>
<p>Every section from the LOCUS line to the terminating // becomes
a single SeqRecord with associated annotation and features.</p>
<p>Note that for genomes or chromosomes, there is typically only
one record.</p>
<p>This gets called internally by Bio.SeqIO for the EMBL file format:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio <span class="py-keyword">import</span> SeqIO
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">for</span> record <span class="py-keyword">in</span> SeqIO.parse(<span class="py-string">&quot;EMBL/epo_prt_selection.embl&quot;</span>, <span class="py-string">&quot;embl&quot;</span>):
<span class="py-more">... </span>    <span class="py-keyword">print</span>(record.id)
<span class="py-more">...</span>
<span class="py-output">A00022.1</span>
<span class="py-output">A00028.1</span>
<span class="py-output">A00031.1</span>
<span class="py-output">A00034.1</span>
<span class="py-output">A00060.1</span>
<span class="py-output">A00071.1</span>
<span class="py-output">A00072.1</span>
<span class="py-output">A00078.1</span>
<span class="py-output">CQ797900.1</span></pre>
<p>Equivalently,</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>with open(<span class="py-string">&quot;EMBL/epo_prt_selection.embl&quot;</span>) <span class="py-keyword">as</span> handle:
<span class="py-more">... </span>    <span class="py-keyword">for</span> record <span class="py-keyword">in</span> EmblIterator(handle):
<span class="py-more">... </span>        <span class="py-keyword">print</span>(record.id)
<span class="py-more">...</span>
<span class="py-output">A00022.1</span>
<span class="py-output">A00028.1</span>
<span class="py-output">A00031.1</span>
<span class="py-output">A00034.1</span>
<span class="py-output">A00060.1</span>
<span class="py-output">A00071.1</span>
<span class="py-output">A00072.1</span>
<span class="py-output">A00078.1</span>
<span class="py-output">CQ797900.1</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="ImgtIterator"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">ImgtIterator</span>(<span class="sig-arg">handle</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqIO.InsdcIO-pysrc.html#ImgtIterator">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Break up an IMGT file into SeqRecord objects.</p>
<p>Every section from the LOCUS line to the terminating // becomes
a single SeqRecord with associated annotation and features.</p>
<p>Note that for genomes or chromosomes, there is typically only
one record.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="GenBankCdsFeatureIterator"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">GenBankCdsFeatureIterator</span>(<span class="sig-arg">handle</span>,
        <span class="sig-arg">alphabet</span>=<span class="sig-default">ProteinAlphabet()</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqIO.InsdcIO-pysrc.html#GenBankCdsFeatureIterator">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Break up a Genbank file into SeqRecord objects for each CDS feature.</p>
<p>Every section from the LOCUS line to the terminating // can contain
many CDS features.  These are returned as with the stated amino acid
translation sequence (if given).</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="EmblCdsFeatureIterator"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">EmblCdsFeatureIterator</span>(<span class="sig-arg">handle</span>,
        <span class="sig-arg">alphabet</span>=<span class="sig-default">ProteinAlphabet()</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqIO.InsdcIO-pysrc.html#EmblCdsFeatureIterator">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Break up a EMBL file into SeqRecord objects for each CDS feature.</p>
<p>Every section from the LOCUS line to the terminating // can contain
many CDS features.  These are returned as with the stated amino acid
translation sequence (if given).</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_insdc_feature_position_string"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_insdc_feature_position_string</span>(<span class="sig-arg">pos</span>,
        <span class="sig-arg">offset</span>=<span class="sig-default">0</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqIO.InsdcIO-pysrc.html#_insdc_feature_position_string">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Build a GenBank/EMBL position string (PRIVATE).</p>
<p>Use offset=1 to add one to convert a start position from python counting.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_insdc_location_string"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_insdc_location_string</span>(<span class="sig-arg">location</span>,
        <span class="sig-arg">rec_length</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqIO.InsdcIO-pysrc.html#_insdc_location_string">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Build a GenBank/EMBL location from a (Compound) FeatureLocation (PRIVATE).</p>
<p>There is a choice of how to show joins on the reverse complement strand,
GenBank used &quot;complement(join(1,10),(20,100))&quot; while EMBL used to use
&quot;join(complement(20,100),complement(1,10))&quot; instead (but appears to have
now adopted the GenBank convention). Notice that the order of the entries
is reversed! This function therefore uses the first form. In this situation
we expect the CompoundFeatureLocation and its parts to all be marked as
strand == -1, and to be in the order 19:100 then 0:10.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Tue Dec 18 15:56:21 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
