<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.SeqUtils.ProtParam</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.SeqUtils-module.html">Package&nbsp;SeqUtils</a> ::
        Module&nbsp;ProtParam
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.SeqUtils.ProtParam-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.SeqUtils.ProtParam-module.html">Module Bio.SeqUtils.ProtParam</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright Yair Benita Y.Benita@pharm.uu.nl</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># Biopython (http://biopython.org) license applies</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-docstring">"""Simple protein analysis.</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-docstring">Example:</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio.SeqUtils.ProtParam import ProteinAnalysis</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; X = ProteinAnalysis("MAEGEITTFTALTEKFNLPPGNYKKPKLLYCSNGGHFLRILPDGTVDGT"</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-docstring">    ...                     "RDRSDQHIQLQLSAESVGEVYIKSTETGQYLAMDTSGLLYGSQTPSEEC"</tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-docstring">    ...                     "LFLERLEENHYNTYTSKKHAEKNWFVGLKKNGSCKRGPRTHYGQKAILF"</tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-docstring">    ...                     "LPLPV")</tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(X.count_amino_acids()['A'])</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-docstring">    6</tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(X.count_amino_acids()['E'])</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-docstring">    12</tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % X.get_amino_acids_percent()['A'])</tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-docstring">    0.04</tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % X.get_amino_acids_percent()['L'])</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-docstring">    0.12</tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % X.molecular_weight())</tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-docstring">    17103.16</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % X.aromaticity())</tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-docstring">    0.10</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % X.instability_index())</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-docstring">    41.98</tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % X.isoelectric_point())</tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt class="py-docstring">    7.72</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; sec_struc = X.secondary_structure_fraction()  # [helix, turn, sheet]</tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % sec_struc[0])  # helix</tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"><tt class="py-docstring">    0.28</tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; epsilon_prot = X.molar_extinction_coefficient()  # [reduced, oxidized]</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(epsilon_prot[0])  # with reduced cysteines</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-docstring">    17420</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(epsilon_prot[1])  # with disulfid bridges</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"><tt class="py-docstring">    17545</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"><tt class="py-docstring">    Other public methods are:</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"><tt class="py-docstring">     - gravy</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"><tt class="py-docstring">     - protein_scale</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"><tt class="py-docstring">     - flexibility</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line"> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">__future__</tt> <tt class="py-keyword">import</tt> <tt class="py-name">print_function</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line"> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">sys</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio.SeqUtils=Bio.SeqUtils-module.html"><a title="Bio.SeqUtils" class="py-name" href="#" onclick="return doclink('link-1', 'SeqUtils', 'link-1');">SeqUtils</a></tt> <tt class="py-keyword">import</tt> <tt id="link-2" class="py-name" targets="Module Bio.SeqUtils.ProtParamData=Bio.SeqUtils.ProtParamData-module.html"><a title="Bio.SeqUtils.ProtParamData" class="py-name" href="#" onclick="return doclink('link-2', 'ProtParamData', 'link-2');">ProtParamData</a></tt>  <tt class="py-comment"># Local</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-3" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-3', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-4" class="py-name"><a title="Bio.SeqUtils" class="py-name" href="#" onclick="return doclink('link-4', 'SeqUtils', 'link-1');">SeqUtils</a></tt> <tt class="py-keyword">import</tt> <tt id="link-5" class="py-name" targets="Module Bio.SeqUtils.IsoelectricPoint=Bio.SeqUtils.IsoelectricPoint-module.html,Class Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint=Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint-class.html"><a title="Bio.SeqUtils.IsoelectricPoint
Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint" class="py-name" href="#" onclick="return doclink('link-5', 'IsoelectricPoint', 'link-5');">IsoelectricPoint</a></tt>  <tt class="py-comment"># Local</tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-6" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-6', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-7" class="py-name" targets="Module Bio.Seq=Bio.Seq-module.html,Class Bio.Seq.Seq=Bio.Seq.Seq-class.html"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-7', 'Seq', 'link-7');">Seq</a></tt> <tt class="py-keyword">import</tt> <tt id="link-8" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-8', 'Seq', 'link-7');">Seq</a></tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-9" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-9', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-10" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-10', 'Alphabet', 'link-10');">Alphabet</a></tt> <tt class="py-keyword">import</tt> <tt id="link-11" class="py-name" targets="Module Bio.Alphabet.IUPAC=Bio.Alphabet.IUPAC-module.html"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-11', 'IUPAC', 'link-11');">IUPAC</a></tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-12" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-12', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-13" class="py-name" targets="Package Bio.Data=Bio.Data-module.html"><a title="Bio.Data" class="py-name" href="#" onclick="return doclink('link-13', 'Data', 'link-13');">Data</a></tt> <tt class="py-keyword">import</tt> <tt id="link-14" class="py-name" targets="Module Bio.Data.IUPACData=Bio.Data.IUPACData-module.html"><a title="Bio.Data.IUPACData" class="py-name" href="#" onclick="return doclink('link-14', 'IUPACData', 'link-14');">IUPACData</a></tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-15" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-15', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-16" class="py-name"><a title="Bio.SeqUtils" class="py-name" href="#" onclick="return doclink('link-16', 'SeqUtils', 'link-1');">SeqUtils</a></tt> <tt class="py-keyword">import</tt> <tt id="link-17" class="py-name" targets="Method Bio.SeqUtils.ProtParam.ProteinAnalysis.molecular_weight()=Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#molecular_weight,Function Bio.SeqUtils.molecular_weight()=Bio.SeqUtils-module.html#molecular_weight"><a title="Bio.SeqUtils.ProtParam.ProteinAnalysis.molecular_weight
Bio.SeqUtils.molecular_weight" class="py-name" href="#" onclick="return doclink('link-17', 'molecular_weight', 'link-17');">molecular_weight</a></tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line"> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line"> </tt>
<a name="ProteinAnalysis"></a><div id="ProteinAnalysis-def"><a name="L55"></a><tt class="py-lineno"> 55</tt> <a class="py-toggle" href="#" id="ProteinAnalysis-toggle" onclick="return toggle('ProteinAnalysis');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html">ProteinAnalysis</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ProteinAnalysis-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="ProteinAnalysis-expanded"><a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line">    <tt class="py-docstring">"""Class containing methods for protein analysis.</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line"><tt class="py-docstring">    The constructor takes two arguments.</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line"><tt class="py-docstring">    The first is the protein sequence as a string, which is then converted to a</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line"><tt class="py-docstring">    sequence object using the Bio.Seq module. This is done just to make sure</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line"><tt class="py-docstring">    the sequence is a protein sequence and not anything else.</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line"><tt class="py-docstring">    The second argument is optional. If set to True, the weight of the amino</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line"><tt class="py-docstring">    acids will be calculated using their monoisotopic mass (the weight of the</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line"><tt class="py-docstring">    most abundant isotopes for each element), instead of the average molecular</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line"><tt class="py-docstring">    mass (the averaged weight of all stable isotopes for each element).</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line"><tt class="py-docstring">    If set to false (the default value) or left out, the IUPAC average</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line"><tt class="py-docstring">    molecular mass will be used for the calculation.</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line"> </tt>
<a name="ProteinAnalysis.__init__"></a><div id="ProteinAnalysis.__init__-def"><a name="L72"></a><tt class="py-lineno"> 72</tt> <a class="py-toggle" href="#" id="ProteinAnalysis.__init__-toggle" onclick="return toggle('ProteinAnalysis.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">prot_sequence</tt><tt class="py-op">,</tt> <tt class="py-param">monoisotopic</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ProteinAnalysis.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ProteinAnalysis.__init__-expanded"><a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the class."""</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">prot_sequence</tt><tt class="py-op">.</tt><tt class="py-name">islower</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-18" class="py-name" targets="Method Bio.FSSP.FSSPAlignDict.sequence()=Bio.FSSP.FSSPAlignDict-class.html#sequence,Method Bio.GenBank._FeatureConsumer.sequence()=Bio.GenBank._FeatureConsumer-class.html#sequence,Method Bio.GenBank._RecordConsumer.sequence()=Bio.GenBank._RecordConsumer-class.html#sequence,Method Bio.Phylo.PhyloXMLIO.Writer.sequence()=Bio.Phylo.PhyloXMLIO.Writer-class.html#sequence"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-18', 'sequence', 'link-18');">sequence</a></tt> <tt class="py-op">=</tt> <tt id="link-19" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-19', 'Seq', 'link-7');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">prot_sequence</tt><tt class="py-op">.</tt><tt id="link-20" class="py-name" targets="Method Bio.Seq.Seq.upper()=Bio.Seq.Seq-class.html#upper,Method Bio.Seq.UnknownSeq.upper()=Bio.Seq.UnknownSeq-class.html#upper,Method Bio.SeqRecord.SeqRecord.upper()=Bio.SeqRecord.SeqRecord-class.html#upper"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-20', 'upper', 'link-20');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-21" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-21', 'IUPAC', 'link-11');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-22" class="py-name" targets="Variable Bio.Alphabet.IUPAC.protein=Bio.Alphabet.IUPAC-module.html#protein"><a title="Bio.Alphabet.IUPAC.protein" class="py-name" href="#" onclick="return doclink('link-22', 'protein', 'link-22');">protein</a></tt><tt class="py-op">)</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-23" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-23', 'sequence', 'link-18');">sequence</a></tt> <tt class="py-op">=</tt> <tt id="link-24" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-24', 'Seq', 'link-7');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">prot_sequence</tt><tt class="py-op">,</tt> <tt id="link-25" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-25', 'IUPAC', 'link-11');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-26" class="py-name"><a title="Bio.Alphabet.IUPAC.protein" class="py-name" href="#" onclick="return doclink('link-26', 'protein', 'link-22');">protein</a></tt><tt class="py-op">)</tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">amino_acids_content</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">amino_acids_percent</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-27" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.length()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#length"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-27', 'length', 'link-27');">length</a></tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-28" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-28', 'sequence', 'link-18');">sequence</a></tt><tt class="py-op">)</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">monoisotopic</tt> <tt class="py-op">=</tt> <tt class="py-name">monoisotopic</tt> </tt>
</div><a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line"> </tt>
<a name="ProteinAnalysis.count_amino_acids"></a><div id="ProteinAnalysis.count_amino_acids-def"><a name="L83"></a><tt class="py-lineno"> 83</tt> <a class="py-toggle" href="#" id="ProteinAnalysis.count_amino_acids-toggle" onclick="return toggle('ProteinAnalysis.count_amino_acids');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#count_amino_acids">count_amino_acids</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ProteinAnalysis.count_amino_acids-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ProteinAnalysis.count_amino_acids-expanded"><a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line">        <tt class="py-docstring">"""Count standard amino acids, return a dict.</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line"><tt class="py-docstring">        Counts the number times each amino acid is in the protein</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line"><tt class="py-docstring">        sequence. Returns a dictionary {AminoAcid:Number}.</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line"><tt class="py-docstring">        The return value is cached in self.amino_acids_content.</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line"><tt class="py-docstring">        It is not recalculated upon subsequent calls.</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">amino_acids_content</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">            <tt class="py-name">prot_dic</tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">k</tt> <tt class="py-keyword">in</tt> <tt id="link-29" class="py-name"><a title="Bio.Data.IUPACData" class="py-name" href="#" onclick="return doclink('link-29', 'IUPACData', 'link-14');">IUPACData</a></tt><tt class="py-op">.</tt><tt id="link-30" class="py-name" targets="Variable Bio.Data.IUPACData.protein_letters=Bio.Data.IUPACData-module.html#protein_letters"><a title="Bio.Data.IUPACData.protein_letters" class="py-name" href="#" onclick="return doclink('link-30', 'protein_letters', 'link-30');">protein_letters</a></tt><tt class="py-op">)</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">aa</tt> <tt class="py-keyword">in</tt> <tt class="py-name">prot_dic</tt><tt class="py-op">:</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line">                <tt class="py-name">prot_dic</tt><tt class="py-op">[</tt><tt class="py-name">aa</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-31" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-31', 'sequence', 'link-18');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-32" class="py-name" targets="Method Bio.Crystal.Chain.count()=Bio.Crystal.Chain-class.html#count,Method Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count()=Bio.NeuralNetwork.Gene.Pattern.PatternRepository-class.html#count,Method Bio.Seq.MutableSeq.count()=Bio.Seq.MutableSeq-class.html#count,Method Bio.Seq.Seq.count()=Bio.Seq.Seq-class.html#count,Method Bio.Seq.UnknownSeq.count()=Bio.Seq.UnknownSeq-class.html#count,Method Bio.motifs.Instances.count()=Bio.motifs.Instances-class.html#count"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-32', 'count', 'link-32');">count</a></tt><tt class="py-op">(</tt><tt class="py-name">aa</tt><tt class="py-op">)</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line"> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">amino_acids_content</tt> <tt class="py-op">=</tt> <tt class="py-name">prot_dic</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line"> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">amino_acids_content</tt> </tt>
</div><a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line"> </tt>
<a name="ProteinAnalysis.get_amino_acids_percent"></a><div id="ProteinAnalysis.get_amino_acids_percent-def"><a name="L101"></a><tt class="py-lineno">101</tt> <a class="py-toggle" href="#" id="ProteinAnalysis.get_amino_acids_percent-toggle" onclick="return toggle('ProteinAnalysis.get_amino_acids_percent');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#get_amino_acids_percent">get_amino_acids_percent</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ProteinAnalysis.get_amino_acids_percent-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ProteinAnalysis.get_amino_acids_percent-expanded"><a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line">        <tt class="py-docstring">"""Calculate the amino acid content in percentages.</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line"><tt class="py-docstring">        The same as count_amino_acids only returns the Number in percentage of</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line"><tt class="py-docstring">        entire sequence. Returns a dictionary of {AminoAcid:percentage}.</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line"><tt class="py-docstring">        The return value is cached in self.amino_acids_percent.</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line"><tt class="py-docstring">        input is the dictionary self.amino_acids_content.</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line"><tt class="py-docstring">        output is a dictionary with amino acids as keys.</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">amino_acids_percent</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">            <tt class="py-name">aa_counts</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-33" class="py-name" targets="Method Bio.SeqUtils.ProtParam.ProteinAnalysis.count_amino_acids()=Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#count_amino_acids"><a title="Bio.SeqUtils.ProtParam.ProteinAnalysis.count_amino_acids" class="py-name" href="#" onclick="return doclink('link-33', 'count_amino_acids', 'link-33');">count_amino_acids</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line"> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">            <tt class="py-name">percentages</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">aa</tt> <tt class="py-keyword">in</tt> <tt class="py-name">aa_counts</tt><tt class="py-op">:</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">                <tt class="py-name">percentages</tt><tt class="py-op">[</tt><tt class="py-name">aa</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">aa_counts</tt><tt class="py-op">[</tt><tt class="py-name">aa</tt><tt class="py-op">]</tt> <tt class="py-op">/</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-34" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-34', 'length', 'link-27');">length</a></tt><tt class="py-op">)</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line"> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">amino_acids_percent</tt> <tt class="py-op">=</tt> <tt class="py-name">percentages</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line"> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">amino_acids_percent</tt> </tt>
</div><a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line"> </tt>
<a name="ProteinAnalysis.molecular_weight"></a><div id="ProteinAnalysis.molecular_weight-def"><a name="L123"></a><tt class="py-lineno">123</tt> <a class="py-toggle" href="#" id="ProteinAnalysis.molecular_weight-toggle" onclick="return toggle('ProteinAnalysis.molecular_weight');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#molecular_weight">molecular_weight</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ProteinAnalysis.molecular_weight-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ProteinAnalysis.molecular_weight-expanded"><a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">        <tt class="py-docstring">"""Calculate MW from Protein sequence."""</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-35" class="py-name"><a title="Bio.SeqUtils.ProtParam.ProteinAnalysis.molecular_weight
Bio.SeqUtils.molecular_weight" class="py-name" href="#" onclick="return doclink('link-35', 'molecular_weight', 'link-17');">molecular_weight</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-36" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-36', 'sequence', 'link-18');">sequence</a></tt><tt class="py-op">,</tt> <tt class="py-name">monoisotopic</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">monoisotopic</tt><tt class="py-op">)</tt> </tt>
</div><a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line"> </tt>
<a name="ProteinAnalysis.aromaticity"></a><div id="ProteinAnalysis.aromaticity-def"><a name="L127"></a><tt class="py-lineno">127</tt> <a class="py-toggle" href="#" id="ProteinAnalysis.aromaticity-toggle" onclick="return toggle('ProteinAnalysis.aromaticity');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#aromaticity">aromaticity</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ProteinAnalysis.aromaticity-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ProteinAnalysis.aromaticity-expanded"><a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line">        <tt class="py-docstring">"""Calculate the aromaticity according to Lobry, 1994.</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line"><tt class="py-docstring">        Calculates the aromaticity value of a protein according to Lobry, 1994.</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line"><tt class="py-docstring">        It is simply the relative frequency of Phe+Trp+Tyr.</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line">        <tt class="py-name">aromatic_aas</tt> <tt class="py-op">=</tt> <tt class="py-string">'YWF'</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line">        <tt class="py-name">aa_percentages</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-37" class="py-name" targets="Method Bio.SeqUtils.ProtParam.ProteinAnalysis.get_amino_acids_percent()=Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#get_amino_acids_percent"><a title="Bio.SeqUtils.ProtParam.ProteinAnalysis.get_amino_acids_percent" class="py-name" href="#" onclick="return doclink('link-37', 'get_amino_acids_percent', 'link-37');">get_amino_acids_percent</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line"> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">        <tt id="link-38" class="py-name" targets="Method Bio.SeqUtils.ProtParam.ProteinAnalysis.aromaticity()=Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#aromaticity"><a title="Bio.SeqUtils.ProtParam.ProteinAnalysis.aromaticity" class="py-name" href="#" onclick="return doclink('link-38', 'aromaticity', 'link-38');">aromaticity</a></tt> <tt class="py-op">=</tt> <tt id="link-39" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.sum()=Bio.Nexus.Nexus.StepMatrix-class.html#sum,Method Bio.SubsMat.SeqMat.sum()=Bio.SubsMat.SeqMat-class.html#sum"><a title="Bio.Nexus.Nexus.StepMatrix.sum
Bio.SubsMat.SeqMat.sum" class="py-name" href="#" onclick="return doclink('link-39', 'sum', 'link-39');">sum</a></tt><tt class="py-op">(</tt><tt class="py-name">aa_percentages</tt><tt class="py-op">[</tt><tt class="py-name">aa</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt class="py-name">aa</tt> <tt class="py-keyword">in</tt> <tt class="py-name">aromatic_aas</tt><tt class="py-op">)</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line"> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-40" class="py-name"><a title="Bio.SeqUtils.ProtParam.ProteinAnalysis.aromaticity" class="py-name" href="#" onclick="return doclink('link-40', 'aromaticity', 'link-38');">aromaticity</a></tt> </tt>
</div><a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line"> </tt>
<a name="ProteinAnalysis.instability_index"></a><div id="ProteinAnalysis.instability_index-def"><a name="L140"></a><tt class="py-lineno">140</tt> <a class="py-toggle" href="#" id="ProteinAnalysis.instability_index-toggle" onclick="return toggle('ProteinAnalysis.instability_index');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#instability_index">instability_index</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ProteinAnalysis.instability_index-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ProteinAnalysis.instability_index-expanded"><a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line">        <tt class="py-docstring">"""Calculate the instability index according to Guruprasad et al 1990.</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line"><tt class="py-docstring">        Implementation of the method of Guruprasad et al. 1990 to test a</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line"><tt class="py-docstring">        protein for stability. Any value above 40 means the protein is unstable</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line"><tt class="py-docstring">        (has a short half life).</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line"><tt class="py-docstring">        See: Guruprasad K., Reddy B.V.B., Pandit M.W.</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line"><tt class="py-docstring">        Protein Engineering 4:155-161(1990).</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">        <tt id="link-41" class="py-name" targets="Method Bio.Crystal.Chain.index()=Bio.Crystal.Chain-class.html#index,Method Bio.SCOP.Raf.SeqMap.index()=Bio.SCOP.Raf.SeqMap-class.html#index,Method Bio.SearchIO._model.hit.Hit.index()=Bio.SearchIO._model.hit.Hit-class.html#index,Method Bio.SearchIO._model.query.QueryResult.index()=Bio.SearchIO._model.query.QueryResult-class.html#index,Function Bio.SearchIO.index()=Bio.SearchIO-module.html#index,Method Bio.Seq.MutableSeq.index()=Bio.Seq.MutableSeq-class.html#index,Function Bio.SeqIO.index()=Bio.SeqIO-module.html#index"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-41', 'index', 'link-41');">index</a></tt> <tt class="py-op">=</tt> <tt id="link-42" class="py-name"><a title="Bio.SeqUtils.ProtParamData" class="py-name" href="#" onclick="return doclink('link-42', 'ProtParamData', 'link-2');">ProtParamData</a></tt><tt class="py-op">.</tt><tt id="link-43" class="py-name" targets="Variable Bio.SeqUtils.ProtParamData.DIWV=Bio.SeqUtils.ProtParamData-module.html#DIWV"><a title="Bio.SeqUtils.ProtParamData.DIWV" class="py-name" href="#" onclick="return doclink('link-43', 'DIWV', 'link-43');">DIWV</a></tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">        <tt id="link-44" class="py-name" targets="Method Bio.Align.PairwiseAligner.score()=Bio.Align.PairwiseAligner-class.html#score,Method Bio.Blast.NCBIStandalone._HSPConsumer.score()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#score"><a title="Bio.Align.PairwiseAligner.score
Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-44', 'score', 'link-44');">score</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0.0</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line"> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-45" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-45', 'i', 'link-45');">i</a></tt> <tt class="py-keyword">in</tt> <tt id="link-46" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-46', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-47" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-47', 'length', 'link-27');">length</a></tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line">            <tt class="py-name">this</tt><tt class="py-op">,</tt> <tt id="link-48" class="py-name" targets="Method Bio.AlignIO.Interfaces.AlignmentIterator.next()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#next,Method Bio.Blast.NCBIStandalone.Iterator.next()=Bio.Blast.NCBIStandalone.Iterator-class.html#next,Method Bio.Blast.ParseBlastTable.BlastTableReader.next()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#next,Method Bio.File.UndoHandle.next()=Bio.File.UndoHandle-class.html#next,Method Bio.GenBank.Iterator.next()=Bio.GenBank.Iterator-class.html#next,Method Bio.Nexus.Nexus.CharBuffer.next()=Bio.Nexus.Nexus.CharBuffer-class.html#next,Method Bio.Nexus.StandardData.StandardData.next()=Bio.Nexus.StandardData.StandardData-class.html#next,Method Bio.PopGen.GenePop.Controller._FileIterator.next()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#next,Method Bio.SeqIO.Interfaces.SequenceIterator.next()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#next,Method Bio.bgzf.BgzfReader.next()=Bio.bgzf.BgzfReader-class.html#next"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.Nexus.StandardData.StandardData.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-48', 'next', 'link-48');">next</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-49" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-49', 'sequence', 'link-18');">sequence</a></tt><tt class="py-op">[</tt><tt id="link-50" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-50', 'i', 'link-45');">i</a></tt><tt class="py-op">:</tt><tt id="link-51" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-51', 'i', 'link-45');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-number">2</tt><tt class="py-op">]</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">            <tt class="py-name">dipeptide_value</tt> <tt class="py-op">=</tt> <tt id="link-52" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-52', 'index', 'link-41');">index</a></tt><tt class="py-op">[</tt><tt class="py-name">this</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt id="link-53" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.Nexus.StandardData.StandardData.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-53', 'next', 'link-48');">next</a></tt><tt class="py-op">]</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">            <tt id="link-54" class="py-name"><a title="Bio.Align.PairwiseAligner.score
Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-54', 'score', 'link-44');">score</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">dipeptide_value</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line"> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-number">10.0</tt> <tt class="py-op">/</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-55" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-55', 'length', 'link-27');">length</a></tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt id="link-56" class="py-name"><a title="Bio.Align.PairwiseAligner.score
Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-56', 'score', 'link-44');">score</a></tt> </tt>
</div><a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line"> </tt>
<a name="ProteinAnalysis.flexibility"></a><div id="ProteinAnalysis.flexibility-def"><a name="L160"></a><tt class="py-lineno">160</tt> <a class="py-toggle" href="#" id="ProteinAnalysis.flexibility-toggle" onclick="return toggle('ProteinAnalysis.flexibility');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#flexibility">flexibility</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ProteinAnalysis.flexibility-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ProteinAnalysis.flexibility-expanded"><a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">        <tt class="py-docstring">"""Calculate the flexibility according to Vihinen, 1994.</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line"><tt class="py-docstring">        No argument to change window size because parameters are specific for</tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line"><tt class="py-docstring">        a window=9. The parameters used are optimized for determining the</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line"><tt class="py-docstring">        flexibility.</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">        <tt class="py-name">flexibilities</tt> <tt class="py-op">=</tt> <tt id="link-57" class="py-name"><a title="Bio.SeqUtils.ProtParamData" class="py-name" href="#" onclick="return doclink('link-57', 'ProtParamData', 'link-2');">ProtParamData</a></tt><tt class="py-op">.</tt><tt id="link-58" class="py-name" targets="Variable Bio.SeqUtils.ProtParamData.Flex=Bio.SeqUtils.ProtParamData-module.html#Flex"><a title="Bio.SeqUtils.ProtParamData.Flex" class="py-name" href="#" onclick="return doclink('link-58', 'Flex', 'link-58');">Flex</a></tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">        <tt id="link-59" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.window_size()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#window_size"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.window_size" class="py-name" href="#" onclick="return doclink('link-59', 'window_size', 'link-59');">window_size</a></tt> <tt class="py-op">=</tt> <tt class="py-number">9</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line">        <tt class="py-name">weights</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-number">0.25</tt><tt class="py-op">,</tt> <tt class="py-number">0.4375</tt><tt class="py-op">,</tt> <tt class="py-number">0.625</tt><tt class="py-op">,</tt> <tt class="py-number">0.8125</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line">        <tt class="py-name">scores</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line"> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-60" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-60', 'i', 'link-45');">i</a></tt> <tt class="py-keyword">in</tt> <tt id="link-61" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-61', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-62" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-62', 'length', 'link-27');">length</a></tt> <tt class="py-op">-</tt> <tt id="link-63" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.window_size" class="py-name" href="#" onclick="return doclink('link-63', 'window_size', 'link-59');">window_size</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line">            <tt class="py-name">subsequence</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-64" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-64', 'sequence', 'link-18');">sequence</a></tt><tt class="py-op">[</tt><tt id="link-65" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-65', 'i', 'link-45');">i</a></tt><tt class="py-op">:</tt><tt id="link-66" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-66', 'i', 'link-45');">i</a></tt> <tt class="py-op">+</tt> <tt id="link-67" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.window_size" class="py-name" href="#" onclick="return doclink('link-67', 'window_size', 'link-59');">window_size</a></tt><tt class="py-op">]</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line">            <tt id="link-68" class="py-name"><a title="Bio.Align.PairwiseAligner.score
Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-68', 'score', 'link-44');">score</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0.0</tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line"> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">j</tt> <tt class="py-keyword">in</tt> <tt id="link-69" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-69', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt id="link-70" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.window_size" class="py-name" href="#" onclick="return doclink('link-70', 'window_size', 'link-59');">window_size</a></tt> <tt class="py-op">//</tt> <tt class="py-number">2</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">                <tt class="py-name">front</tt> <tt class="py-op">=</tt> <tt class="py-name">subsequence</tt><tt class="py-op">[</tt><tt class="py-name">j</tt><tt class="py-op">]</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">                <tt class="py-name">back</tt> <tt class="py-op">=</tt> <tt class="py-name">subsequence</tt><tt class="py-op">[</tt><tt id="link-71" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.window_size" class="py-name" href="#" onclick="return doclink('link-71', 'window_size', 'link-59');">window_size</a></tt> <tt class="py-op">-</tt> <tt class="py-name">j</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">                <tt id="link-72" class="py-name"><a title="Bio.Align.PairwiseAligner.score
Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-72', 'score', 'link-44');">score</a></tt> <tt class="py-op">+=</tt> <tt class="py-op">(</tt><tt class="py-name">flexibilities</tt><tt class="py-op">[</tt><tt class="py-name">front</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">                          <tt class="py-name">flexibilities</tt><tt class="py-op">[</tt><tt class="py-name">back</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-name">weights</tt><tt class="py-op">[</tt><tt class="py-name">j</tt><tt class="py-op">]</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line"> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">            <tt class="py-name">middle</tt> <tt class="py-op">=</tt> <tt class="py-name">subsequence</tt><tt class="py-op">[</tt><tt id="link-73" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.window_size" class="py-name" href="#" onclick="return doclink('link-73', 'window_size', 'link-59');">window_size</a></tt> <tt class="py-op">//</tt> <tt class="py-number">2</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">            <tt id="link-74" class="py-name"><a title="Bio.Align.PairwiseAligner.score
Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-74', 'score', 'link-44');">score</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">flexibilities</tt><tt class="py-op">[</tt><tt class="py-name">middle</tt><tt class="py-op">]</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line"> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">            <tt class="py-name">scores</tt><tt class="py-op">.</tt><tt id="link-75" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-75', 'append', 'link-75');">append</a></tt><tt class="py-op">(</tt><tt id="link-76" class="py-name"><a title="Bio.Align.PairwiseAligner.score
Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-76', 'score', 'link-44');">score</a></tt> <tt class="py-op">/</tt> <tt class="py-number">5.25</tt><tt class="py-op">)</tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line"> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">scores</tt> </tt>
</div><a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line"> </tt>
<a name="ProteinAnalysis.gravy"></a><div id="ProteinAnalysis.gravy-def"><a name="L189"></a><tt class="py-lineno">189</tt> <a class="py-toggle" href="#" id="ProteinAnalysis.gravy-toggle" onclick="return toggle('ProteinAnalysis.gravy');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#gravy">gravy</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ProteinAnalysis.gravy-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ProteinAnalysis.gravy-expanded"><a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">        <tt class="py-docstring">"""Calculate the gravy according to Kyte and Doolittle."""</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">        <tt class="py-name">total_gravy</tt> <tt class="py-op">=</tt> <tt id="link-77" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.sum
Bio.SubsMat.SeqMat.sum" class="py-name" href="#" onclick="return doclink('link-77', 'sum', 'link-39');">sum</a></tt><tt class="py-op">(</tt><tt id="link-78" class="py-name"><a title="Bio.SeqUtils.ProtParamData" class="py-name" href="#" onclick="return doclink('link-78', 'ProtParamData', 'link-2');">ProtParamData</a></tt><tt class="py-op">.</tt><tt id="link-79" class="py-name" targets="Variable Bio.SeqUtils.ProtParamData.kd=Bio.SeqUtils.ProtParamData-module.html#kd"><a title="Bio.SeqUtils.ProtParamData.kd" class="py-name" href="#" onclick="return doclink('link-79', 'kd', 'link-79');">kd</a></tt><tt class="py-op">[</tt><tt class="py-name">aa</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt class="py-name">aa</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-80" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-80', 'sequence', 'link-18');">sequence</a></tt><tt class="py-op">)</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line"> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">total_gravy</tt> <tt class="py-op">/</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-81" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-81', 'length', 'link-27');">length</a></tt> </tt>
</div><a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line"> </tt>
<a name="ProteinAnalysis._weight_list"></a><div id="ProteinAnalysis._weight_list-def"><a name="L195"></a><tt class="py-lineno">195</tt> <a class="py-toggle" href="#" id="ProteinAnalysis._weight_list-toggle" onclick="return toggle('ProteinAnalysis._weight_list');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#_weight_list">_weight_list</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">window</tt><tt class="py-op">,</tt> <tt class="py-param">edge</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ProteinAnalysis._weight_list-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ProteinAnalysis._weight_list-expanded"><a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">        <tt class="py-docstring">"""Make list of relative weight of window edges (PRIVATE).</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line"><tt class="py-docstring">        The relative weight of window edges are compared to the window</tt> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line"><tt class="py-docstring">        center. The weights are linear. It actually generates half a list.</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line"><tt class="py-docstring">        For a window of size 9 and edge 0.4 you get a list of</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line"><tt class="py-docstring">        [0.4, 0.55, 0.7, 0.85].</tt> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line">        <tt class="py-name">unit</tt> <tt class="py-op">=</tt> <tt class="py-number">2</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-number">1.0</tt> <tt class="py-op">-</tt> <tt class="py-name">edge</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-op">(</tt><tt class="py-name">window</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line">        <tt class="py-name">weights</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-number">0.0</tt><tt class="py-op">]</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">window</tt> <tt class="py-op">//</tt> <tt class="py-number">2</tt><tt class="py-op">)</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line"> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-82" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-82', 'i', 'link-45');">i</a></tt> <tt class="py-keyword">in</tt> <tt id="link-83" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-83', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">window</tt> <tt class="py-op">//</tt> <tt class="py-number">2</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line">            <tt class="py-name">weights</tt><tt class="py-op">[</tt><tt id="link-84" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-84', 'i', 'link-45');">i</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">edge</tt> <tt class="py-op">+</tt> <tt class="py-name">unit</tt> <tt class="py-op">*</tt> <tt id="link-85" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-85', 'i', 'link-45');">i</a></tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line"> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">weights</tt> </tt>
</div><a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line"> </tt>
<a name="ProteinAnalysis.protein_scale"></a><div id="ProteinAnalysis.protein_scale-def"><a name="L211"></a><tt class="py-lineno">211</tt> <a class="py-toggle" href="#" id="ProteinAnalysis.protein_scale-toggle" onclick="return toggle('ProteinAnalysis.protein_scale');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#protein_scale">protein_scale</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">param_dict</tt><tt class="py-op">,</tt> <tt class="py-param">window</tt><tt class="py-op">,</tt> <tt class="py-param">edge</tt><tt class="py-op">=</tt><tt class="py-number">1.0</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ProteinAnalysis.protein_scale-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ProteinAnalysis.protein_scale-expanded"><a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line">        <tt class="py-docstring">"""Compute a profile by any amino acid scale.</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line"><tt class="py-docstring">        An amino acid scale is defined by a numerical value assigned to each</tt> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line"><tt class="py-docstring">        type of amino acid. The most frequently used scales are the</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line"><tt class="py-docstring">        hydrophobicity or hydrophilicity scales and the secondary structure</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line"><tt class="py-docstring">        conformational parameters scales, but many other scales exist which</tt> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line"><tt class="py-docstring">        are based on different chemical and physical properties of the</tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line"><tt class="py-docstring">        amino acids.  You can set several parameters that control the</tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line"><tt class="py-docstring">        computation of a scale profile, such as the window size and the window</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line"><tt class="py-docstring">        edge relative weight value.</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line"><tt class="py-docstring">        WindowSize: The window size is the length of the interval to use for</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line"><tt class="py-docstring">        the profile computation. For a window size n, we use the i-(n-1)/2</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line"><tt class="py-docstring">        neighboring residues on each side to compute the score for residue i.</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line"><tt class="py-docstring">        The score for residue i is the sum of the scaled values for these</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line"><tt class="py-docstring">        amino acids, optionally weighted according to their position in the</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line"><tt class="py-docstring">        window.</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line"><tt class="py-docstring">        Edge: The central amino acid of the window always has a weight of 1.</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line"><tt class="py-docstring">        By default, the amino acids at the remaining window positions have the</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line"><tt class="py-docstring">        same weight, but you can make the residue at the center of the window</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line"><tt class="py-docstring">        have a larger weight than the others by setting the edge value for the</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line"><tt class="py-docstring">        residues at the beginning and end of the interval to a value between</tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line"><tt class="py-docstring">        0 and 1. For instance, for Edge=0.4 and a window size of 5 the weights</tt> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line"><tt class="py-docstring">        will be: 0.4, 0.7, 1.0, 0.7, 0.4.</tt> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line"><tt class="py-docstring">        The method returns a list of values which can be plotted to view the</tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line"><tt class="py-docstring">        change along a protein sequence.  Many scales exist. Just add your</tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line"><tt class="py-docstring">        favorites to the ProtParamData modules.</tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line"><tt class="py-docstring">        Similar to expasy's ProtScale:</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line"><tt class="py-docstring">        http://www.expasy.org/cgi-bin/protscale.pl</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line">        <tt class="py-comment"># generate the weights</tt> </tt>
<a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line">        <tt class="py-comment">#   _weight_list returns only one tail. If the list should be</tt> </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line">        <tt class="py-comment">#   [0.4,0.7,1.0,0.7,0.4] what you actually get from _weights_list</tt> </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line">        <tt class="py-comment">#   is [0.4,0.7]. The correct calculation is done in the loop.</tt> </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line">        <tt class="py-name">weights</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-86" class="py-name" targets="Method Bio.SeqUtils.ProtParam.ProteinAnalysis._weight_list()=Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#_weight_list"><a title="Bio.SeqUtils.ProtParam.ProteinAnalysis._weight_list" class="py-name" href="#" onclick="return doclink('link-86', '_weight_list', 'link-86');">_weight_list</a></tt><tt class="py-op">(</tt><tt class="py-name">window</tt><tt class="py-op">,</tt> <tt class="py-name">edge</tt><tt class="py-op">)</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line">        <tt class="py-name">scores</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line"> </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line">        <tt class="py-comment"># the score in each Window is divided by the sum of weights</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line">        <tt class="py-comment"># (* 2 + 1) since the weight list is one sided:</tt> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line">        <tt class="py-name">sum_of_weights</tt> <tt class="py-op">=</tt> <tt id="link-87" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.sum
Bio.SubsMat.SeqMat.sum" class="py-name" href="#" onclick="return doclink('link-87', 'sum', 'link-39');">sum</a></tt><tt class="py-op">(</tt><tt class="py-name">weights</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-number">2</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line"> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-88" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-88', 'i', 'link-45');">i</a></tt> <tt class="py-keyword">in</tt> <tt id="link-89" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-89', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-90" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-90', 'length', 'link-27');">length</a></tt> <tt class="py-op">-</tt> <tt class="py-name">window</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line">            <tt class="py-name">subsequence</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-91" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-91', 'sequence', 'link-18');">sequence</a></tt><tt class="py-op">[</tt><tt id="link-92" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-92', 'i', 'link-45');">i</a></tt><tt class="py-op">:</tt><tt id="link-93" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-93', 'i', 'link-45');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-name">window</tt><tt class="py-op">]</tt> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line">            <tt id="link-94" class="py-name"><a title="Bio.Align.PairwiseAligner.score
Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-94', 'score', 'link-44');">score</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0.0</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line"> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">j</tt> <tt class="py-keyword">in</tt> <tt id="link-95" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-95', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">window</tt> <tt class="py-op">//</tt> <tt class="py-number">2</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line">                <tt class="py-comment"># walk from the outside of the Window towards the middle.</tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line">                <tt class="py-comment"># Iddo: try/except clauses added to avoid raising an exception</tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line">                <tt class="py-comment"># on a non-standard amino acid</tt> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line">                <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line">                    <tt class="py-name">front</tt> <tt class="py-op">=</tt> <tt class="py-name">param_dict</tt><tt class="py-op">[</tt><tt class="py-name">subsequence</tt><tt class="py-op">[</tt><tt class="py-name">j</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">                    <tt class="py-name">back</tt> <tt class="py-op">=</tt> <tt class="py-name">param_dict</tt><tt class="py-op">[</tt><tt class="py-name">subsequence</tt><tt class="py-op">[</tt><tt class="py-name">window</tt> <tt class="py-op">-</tt> <tt class="py-name">j</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">                    <tt id="link-96" class="py-name"><a title="Bio.Align.PairwiseAligner.score
Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-96', 'score', 'link-44');">score</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">weights</tt><tt class="py-op">[</tt><tt class="py-name">j</tt><tt class="py-op">]</tt> <tt class="py-op">*</tt> <tt class="py-name">front</tt> <tt class="py-op">+</tt> <tt class="py-name">weights</tt><tt class="py-op">[</tt><tt class="py-name">j</tt><tt class="py-op">]</tt> <tt class="py-op">*</tt> <tt class="py-name">back</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line">                <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">                    <tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">stderr</tt><tt class="py-op">.</tt><tt id="link-97" class="py-name" targets="Function Bio.AlignIO.write()=Bio.AlignIO-module.html#write,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.write()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#write,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.write()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#write,Method Bio.Phylo.CDAOIO.Writer.write()=Bio.Phylo.CDAOIO.Writer-class.html#write,Function Bio.Phylo.CDAOIO.write()=Bio.Phylo.CDAOIO-module.html#write,Method Bio.Phylo.NeXMLIO.Writer.write()=Bio.Phylo.NeXMLIO.Writer-class.html#write,Function Bio.Phylo.NeXMLIO.write()=Bio.Phylo.NeXMLIO-module.html#write,Method Bio.Phylo.NewickIO.Writer.write()=Bio.Phylo.NewickIO.Writer-class.html#write,Function Bio.Phylo.NewickIO.write()=Bio.Phylo.NewickIO-module.html#write,Function Bio.Phylo.NexusIO.write()=Bio.Phylo.NexusIO-module.html#write,Method Bio.Phylo.PhyloXMLIO.Writer.write()=Bio.Phylo.PhyloXMLIO.Writer-class.html#write,Function Bio.Phylo.PhyloXMLIO.write()=Bio.Phylo.PhyloXMLIO-module.html#write,Function Bio.Phylo._io.write()=Bio.Phylo._io-module.html#write,Function Bio.SearchIO.write()=Bio.SearchIO-module.html#write,Function Bio.SeqIO.write()=Bio.SeqIO-module.html#write,Method Bio.bgzf.BgzfWriter.write()=Bio.bgzf.BgzfWriter-class.html#write,Function Bio.motifs.jaspar.write()=Bio.motifs.jaspar-module.html#write,Function Bio.motifs.transfac.write()=Bio.motifs.transfac-module.html#write,Function Bio.motifs.write()=Bio.motifs-module.html#write,Method Bio.phenotype.phen_micro.JsonWriter.write()=Bio.phenotype.phen_micro.JsonWriter-class.html#write,Function Bio.phenotype.write()=Bio.phenotype-module.html#write"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-97', 'write', 'link-97');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">'warning: %s or %s is not a standard '</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">                                     <tt class="py-string">'amino acid.\n'</tt> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line">                                     <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">subsequence</tt><tt class="py-op">[</tt><tt class="py-name">j</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line">                                        <tt class="py-name">subsequence</tt><tt class="py-op">[</tt><tt class="py-name">window</tt> <tt class="py-op">-</tt> <tt class="py-name">j</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line"> </tt>
<a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line">            <tt class="py-comment"># Now add the middle value, which always has a weight of 1.</tt> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line">            <tt class="py-name">middle</tt> <tt class="py-op">=</tt> <tt class="py-name">subsequence</tt><tt class="py-op">[</tt><tt class="py-name">window</tt> <tt class="py-op">//</tt> <tt class="py-number">2</tt><tt class="py-op">]</tt> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">middle</tt> <tt class="py-keyword">in</tt> <tt class="py-name">param_dict</tt><tt class="py-op">:</tt> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line">                <tt id="link-98" class="py-name"><a title="Bio.Align.PairwiseAligner.score
Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-98', 'score', 'link-44');">score</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">param_dict</tt><tt class="py-op">[</tt><tt class="py-name">middle</tt><tt class="py-op">]</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line">                <tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">stderr</tt><tt class="py-op">.</tt><tt id="link-99" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-99', 'write', 'link-97');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">'warning: %s  is not a standard amino acid.\n'</tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line">                                 <tt class="py-op">%</tt> <tt class="py-name">middle</tt><tt class="py-op">)</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line"> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line">            <tt class="py-name">scores</tt><tt class="py-op">.</tt><tt id="link-100" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-100', 'append', 'link-75');">append</a></tt><tt class="py-op">(</tt><tt id="link-101" class="py-name"><a title="Bio.Align.PairwiseAligner.score
Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-101', 'score', 'link-44');">score</a></tt> <tt class="py-op">/</tt> <tt class="py-name">sum_of_weights</tt><tt class="py-op">)</tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line"> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">scores</tt> </tt>
</div><a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line"> </tt>
<a name="ProteinAnalysis.isoelectric_point"></a><div id="ProteinAnalysis.isoelectric_point-def"><a name="L286"></a><tt class="py-lineno">286</tt> <a class="py-toggle" href="#" id="ProteinAnalysis.isoelectric_point-toggle" onclick="return toggle('ProteinAnalysis.isoelectric_point');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#isoelectric_point">isoelectric_point</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ProteinAnalysis.isoelectric_point-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ProteinAnalysis.isoelectric_point-expanded"><a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">        <tt class="py-docstring">"""Calculate the isoelectric point.</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line"><tt class="py-docstring">        Uses the module IsoelectricPoint to calculate the pI of a protein.</tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line">        <tt class="py-name">aa_content</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-102" class="py-name"><a title="Bio.SeqUtils.ProtParam.ProteinAnalysis.count_amino_acids" class="py-name" href="#" onclick="return doclink('link-102', 'count_amino_acids', 'link-33');">count_amino_acids</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line"> </tt>
<a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line">        <tt class="py-name">ie_point</tt> <tt class="py-op">=</tt> <tt id="link-103" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint
Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint" class="py-name" href="#" onclick="return doclink('link-103', 'IsoelectricPoint', 'link-5');">IsoelectricPoint</a></tt><tt class="py-op">.</tt><tt id="link-104" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint
Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint" class="py-name" href="#" onclick="return doclink('link-104', 'IsoelectricPoint', 'link-5');">IsoelectricPoint</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-105" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-105', 'sequence', 'link-18');">sequence</a></tt><tt class="py-op">,</tt> <tt class="py-name">aa_content</tt><tt class="py-op">)</tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">ie_point</tt><tt class="py-op">.</tt><tt id="link-106" class="py-name" targets="Method Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi()=Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint-class.html#pi"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-106', 'pi', 'link-106');">pi</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line"> </tt>
<a name="ProteinAnalysis.secondary_structure_fraction"></a><div id="ProteinAnalysis.secondary_structure_fraction-def"><a name="L296"></a><tt class="py-lineno">296</tt> <a class="py-toggle" href="#" id="ProteinAnalysis.secondary_structure_fraction-toggle" onclick="return toggle('ProteinAnalysis.secondary_structure_fraction');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#secondary_structure_fraction">secondary_structure_fraction</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ProteinAnalysis.secondary_structure_fraction-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ProteinAnalysis.secondary_structure_fraction-expanded"><a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line">        <tt class="py-docstring">"""Calculate fraction of helix, turn and sheet.</tt> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line"><tt class="py-docstring">        Returns a list of the fraction of amino acids which tend</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line"><tt class="py-docstring">        to be in Helix, Turn or Sheet.</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line"><tt class="py-docstring">        Amino acids in helix: V, I, Y, F, W, L.</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line"><tt class="py-docstring">        Amino acids in Turn: N, P, G, S.</tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line"><tt class="py-docstring">        Amino acids in sheet: E, M, A, L.</tt> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line"><tt class="py-docstring">        Returns a tuple of three floats (Helix, Turn, Sheet).</tt> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line">        <tt class="py-name">aa_percentages</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-107" class="py-name"><a title="Bio.SeqUtils.ProtParam.ProteinAnalysis.get_amino_acids_percent" class="py-name" href="#" onclick="return doclink('link-107', 'get_amino_acids_percent', 'link-37');">get_amino_acids_percent</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line"> </tt>
<a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line">        <tt class="py-name">helix</tt> <tt class="py-op">=</tt> <tt id="link-108" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.sum
Bio.SubsMat.SeqMat.sum" class="py-name" href="#" onclick="return doclink('link-108', 'sum', 'link-39');">sum</a></tt><tt class="py-op">(</tt><tt class="py-name">aa_percentages</tt><tt class="py-op">[</tt><tt class="py-name">r</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt class="py-name">r</tt> <tt class="py-keyword">in</tt> <tt class="py-string">'VIYFWL'</tt><tt class="py-op">)</tt> </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line">        <tt class="py-name">turn</tt> <tt class="py-op">=</tt> <tt id="link-109" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.sum
Bio.SubsMat.SeqMat.sum" class="py-name" href="#" onclick="return doclink('link-109', 'sum', 'link-39');">sum</a></tt><tt class="py-op">(</tt><tt class="py-name">aa_percentages</tt><tt class="py-op">[</tt><tt class="py-name">r</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt class="py-name">r</tt> <tt class="py-keyword">in</tt> <tt class="py-string">'NPGS'</tt><tt class="py-op">)</tt> </tt>
<a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line">        <tt class="py-name">sheet</tt> <tt class="py-op">=</tt> <tt id="link-110" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.sum
Bio.SubsMat.SeqMat.sum" class="py-name" href="#" onclick="return doclink('link-110', 'sum', 'link-39');">sum</a></tt><tt class="py-op">(</tt><tt class="py-name">aa_percentages</tt><tt class="py-op">[</tt><tt class="py-name">r</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt class="py-name">r</tt> <tt class="py-keyword">in</tt> <tt class="py-string">'EMAL'</tt><tt class="py-op">)</tt> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line"> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">helix</tt><tt class="py-op">,</tt> <tt class="py-name">turn</tt><tt class="py-op">,</tt> <tt class="py-name">sheet</tt> </tt>
</div><a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line"> </tt>
<a name="ProteinAnalysis.molar_extinction_coefficient"></a><div id="ProteinAnalysis.molar_extinction_coefficient-def"><a name="L316"></a><tt class="py-lineno">316</tt> <a class="py-toggle" href="#" id="ProteinAnalysis.molar_extinction_coefficient-toggle" onclick="return toggle('ProteinAnalysis.molar_extinction_coefficient');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#molar_extinction_coefficient">molar_extinction_coefficient</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ProteinAnalysis.molar_extinction_coefficient-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ProteinAnalysis.molar_extinction_coefficient-expanded"><a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line">        <tt class="py-docstring">"""Calculate the molar extinction coefficient.</tt> </tt>
<a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line"><tt class="py-docstring">        Calculates the molar extinction coefficient assuming cysteines</tt> </tt>
<a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line"><tt class="py-docstring">        (reduced) and cystines residues (Cys-Cys-bond)</tt> </tt>
<a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">        <tt class="py-name">num_aa</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-111" class="py-name"><a title="Bio.SeqUtils.ProtParam.ProteinAnalysis.count_amino_acids" class="py-name" href="#" onclick="return doclink('link-111', 'count_amino_acids', 'link-33');">count_amino_acids</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">        <tt class="py-name">mec_reduced</tt> <tt class="py-op">=</tt> <tt class="py-name">num_aa</tt><tt class="py-op">[</tt><tt class="py-string">'W'</tt><tt class="py-op">]</tt> <tt class="py-op">*</tt> <tt class="py-number">5500</tt> <tt class="py-op">+</tt> <tt class="py-name">num_aa</tt><tt class="py-op">[</tt><tt class="py-string">'Y'</tt><tt class="py-op">]</tt> <tt class="py-op">*</tt> <tt class="py-number">1490</tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line">        <tt class="py-name">mec_cystines</tt> <tt class="py-op">=</tt> <tt class="py-name">mec_reduced</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt class="py-name">num_aa</tt><tt class="py-op">[</tt><tt class="py-string">'C'</tt><tt class="py-op">]</tt> <tt class="py-op">//</tt> <tt class="py-number">2</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-number">125</tt> </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt><tt class="py-op">(</tt><tt class="py-name">mec_reduced</tt><tt class="py-op">,</tt> <tt class="py-name">mec_cystines</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line"> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line"> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line"><tt class="py-keyword">if</tt> <tt class="py-name">__name__</tt> <tt class="py-op">==</tt> <tt class="py-string">"__main__"</tt><tt class="py-op">:</tt> </tt>
<a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-112" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-112', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-113" class="py-name" targets="Module Bio.Phylo._utils=Bio.Phylo._utils-module.html,Module Bio.SearchIO._utils=Bio.SearchIO._utils-module.html,Module Bio._utils=Bio._utils-module.html"><a title="Bio.Phylo._utils
Bio.SearchIO._utils
Bio._utils" class="py-name" href="#" onclick="return doclink('link-113', '_utils', 'link-113');">_utils</a></tt> <tt class="py-keyword">import</tt> <tt id="link-114" class="py-name" targets="Function Bio._utils.run_doctest()=Bio._utils-module.html#run_doctest"><a title="Bio._utils.run_doctest" class="py-name" href="#" onclick="return doclink('link-114', 'run_doctest', 'link-114');">run_doctest</a></tt> </tt>
<a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line">    <tt id="link-115" class="py-name"><a title="Bio._utils.run_doctest" class="py-name" href="#" onclick="return doclink('link-115', 'run_doctest', 'link-114');">run_doctest</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Fri Jun 22 16:36:18 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
