<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.Seq</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        Module&nbsp;Seq
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.Seq-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.Seq-module.html">Module Bio.Seq</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">   1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2000 Andrew Dalke.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">   2</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2000-2002 Brad Chapman.</tt> </tt>
<a name="L3"></a><tt class="py-lineno">   3</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2004-2005, 2010 by M de Hoon.</tt> </tt>
<a name="L4"></a><tt class="py-lineno">   4</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2007-2018 by Peter Cock.</tt> </tt>
<a name="L5"></a><tt class="py-lineno">   5</tt>  <tt class="py-line"><tt class="py-comment"># All rights reserved.</tt> </tt>
<a name="L6"></a><tt class="py-lineno">   6</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L7"></a><tt class="py-lineno">   7</tt>  <tt class="py-line"><tt class="py-comment"># This file is part of the Biopython distribution and governed by your</tt> </tt>
<a name="L8"></a><tt class="py-lineno">   8</tt>  <tt class="py-line"><tt class="py-comment"># choice of the "Biopython License Agreement" or the "BSD 3-Clause License".</tt> </tt>
<a name="L9"></a><tt class="py-lineno">   9</tt>  <tt class="py-line"><tt class="py-comment"># Please see the LICENSE file that should have been included as part of this</tt> </tt>
<a name="L10"></a><tt class="py-lineno">  10</tt>  <tt class="py-line"><tt class="py-comment"># package.</tt> </tt>
<a name="L11"></a><tt class="py-lineno">  11</tt>  <tt class="py-line"><tt class="py-docstring">"""Provide objects to represent biological sequences with alphabets.</tt> </tt>
<a name="L12"></a><tt class="py-lineno">  12</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L13"></a><tt class="py-lineno">  13</tt>  <tt class="py-line"><tt class="py-docstring">See also the Seq_ wiki and the chapter in our tutorial:</tt> </tt>
<a name="L14"></a><tt class="py-lineno">  14</tt>  <tt class="py-line"><tt class="py-docstring"> - `HTML Tutorial`_</tt> </tt>
<a name="L15"></a><tt class="py-lineno">  15</tt>  <tt class="py-line"><tt class="py-docstring"> - `PDF Tutorial`_</tt> </tt>
<a name="L16"></a><tt class="py-lineno">  16</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L17"></a><tt class="py-lineno">  17</tt>  <tt class="py-line"><tt class="py-docstring">.. _Seq: http://biopython.org/wiki/Seq</tt> </tt>
<a name="L18"></a><tt class="py-lineno">  18</tt>  <tt class="py-line"><tt class="py-docstring">.. _`HTML Tutorial`: http://biopython.org/DIST/docs/tutorial/Tutorial.html</tt> </tt>
<a name="L19"></a><tt class="py-lineno">  19</tt>  <tt class="py-line"><tt class="py-docstring">.. _`PDF Tutorial`: http://biopython.org/DIST/docs/tutorial/Tutorial.pdf</tt> </tt>
<a name="L20"></a><tt class="py-lineno">  20</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L21"></a><tt class="py-lineno">  21</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L22"></a><tt class="py-lineno">  22</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">__future__</tt> <tt class="py-keyword">import</tt> <tt class="py-name">print_function</tt> </tt>
<a name="L23"></a><tt class="py-lineno">  23</tt>  <tt class="py-line"> </tt>
<a name="L24"></a><tt class="py-lineno">  24</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">string</tt>  <tt class="py-comment"># for maketrans only</tt> </tt>
<a name="L25"></a><tt class="py-lineno">  25</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">array</tt> </tt>
<a name="L26"></a><tt class="py-lineno">  26</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">sys</tt> </tt>
<a name="L27"></a><tt class="py-lineno">  27</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L28"></a><tt class="py-lineno">  28</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">collections</tt> </tt>
<a name="L29"></a><tt class="py-lineno">  29</tt>  <tt class="py-line"> </tt>
<a name="L30"></a><tt class="py-lineno">  30</tt>  <tt class="py-line"><tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L31"></a><tt class="py-lineno">  31</tt>  <tt class="py-line">    <tt class="py-comment"># Python 3</tt> </tt>
<a name="L32"></a><tt class="py-lineno">  32</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt class="py-name">collections</tt><tt class="py-op">.</tt><tt class="py-name">abc</tt> <tt class="py-keyword">import</tt> <tt class="py-name">Iterable</tt> <tt class="py-keyword">as</tt> <tt class="py-name">_Iterable</tt> </tt>
<a name="L33"></a><tt class="py-lineno">  33</tt>  <tt class="py-line"><tt class="py-keyword">except</tt> <tt class="py-name">ImportError</tt><tt class="py-op">:</tt> </tt>
<a name="L34"></a><tt class="py-lineno">  34</tt>  <tt class="py-line">    <tt class="py-comment"># Python 2.7</tt> </tt>
<a name="L35"></a><tt class="py-lineno">  35</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt class="py-name">collections</tt> <tt class="py-keyword">import</tt> <tt class="py-name">Iterable</tt> <tt class="py-keyword">as</tt> <tt class="py-name">_Iterable</tt> </tt>
<a name="L36"></a><tt class="py-lineno">  36</tt>  <tt class="py-line"> </tt>
<a name="L37"></a><tt class="py-lineno">  37</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-1', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt id="link-2" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-2', 'range', 'link-2');">range</a></tt> </tt>
<a name="L38"></a><tt class="py-lineno">  38</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-3" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-3', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-4" class="py-name"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-4', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">basestring</tt> </tt>
<a name="L39"></a><tt class="py-lineno">  39</tt>  <tt class="py-line"> </tt>
<a name="L40"></a><tt class="py-lineno">  40</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-5" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-5', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-6" class="py-name" targets="Class Bio.BiopythonWarning=Bio.BiopythonWarning-class.html"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-6', 'BiopythonWarning', 'link-6');">BiopythonWarning</a></tt> </tt>
<a name="L41"></a><tt class="py-lineno">  41</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-7" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-7', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-8" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-8', 'Alphabet', 'link-8');">Alphabet</a></tt> </tt>
<a name="L42"></a><tt class="py-lineno">  42</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-9" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-9', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-10" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-10', 'Alphabet', 'link-8');">Alphabet</a></tt> <tt class="py-keyword">import</tt> <tt id="link-11" class="py-name" targets="Module Bio.Alphabet.IUPAC=Bio.Alphabet.IUPAC-module.html"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-11', 'IUPAC', 'link-11');">IUPAC</a></tt> </tt>
<a name="L43"></a><tt class="py-lineno">  43</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-12" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-12', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-13" class="py-name" targets="Package Bio.Data=Bio.Data-module.html"><a title="Bio.Data" class="py-name" href="#" onclick="return doclink('link-13', 'Data', 'link-13');">Data</a></tt><tt class="py-op">.</tt><tt id="link-14" class="py-name" targets="Module Bio.Data.IUPACData=Bio.Data.IUPACData-module.html"><a title="Bio.Data.IUPACData" class="py-name" href="#" onclick="return doclink('link-14', 'IUPACData', 'link-14');">IUPACData</a></tt> <tt class="py-keyword">import</tt> <tt class="py-op">(</tt><tt id="link-15" class="py-name" targets="Variable Bio.Data.IUPACData.ambiguous_dna_complement=Bio.Data.IUPACData-module.html#ambiguous_dna_complement"><a title="Bio.Data.IUPACData.ambiguous_dna_complement" class="py-name" href="#" onclick="return doclink('link-15', 'ambiguous_dna_complement', 'link-15');">ambiguous_dna_complement</a></tt><tt class="py-op">,</tt> </tt>
<a name="L44"></a><tt class="py-lineno">  44</tt>  <tt class="py-line">                                <tt id="link-16" class="py-name" targets="Variable Bio.Data.IUPACData.ambiguous_rna_complement=Bio.Data.IUPACData-module.html#ambiguous_rna_complement"><a title="Bio.Data.IUPACData.ambiguous_rna_complement" class="py-name" href="#" onclick="return doclink('link-16', 'ambiguous_rna_complement', 'link-16');">ambiguous_rna_complement</a></tt><tt class="py-op">)</tt> </tt>
<a name="L45"></a><tt class="py-lineno">  45</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-17" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-17', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-18" class="py-name"><a title="Bio.Data" class="py-name" href="#" onclick="return doclink('link-18', 'Data', 'link-13');">Data</a></tt><tt class="py-op">.</tt><tt id="link-19" class="py-name"><a title="Bio.Data.IUPACData" class="py-name" href="#" onclick="return doclink('link-19', 'IUPACData', 'link-14');">IUPACData</a></tt> <tt class="py-keyword">import</tt> <tt id="link-20" class="py-name" targets="Variable Bio.Data.IUPACData.ambiguous_dna_letters=Bio.Data.IUPACData-module.html#ambiguous_dna_letters"><a title="Bio.Data.IUPACData.ambiguous_dna_letters" class="py-name" href="#" onclick="return doclink('link-20', 'ambiguous_dna_letters', 'link-20');">ambiguous_dna_letters</a></tt> <tt class="py-keyword">as</tt> <tt class="py-name">_ambiguous_dna_letters</tt> </tt>
<a name="L46"></a><tt class="py-lineno">  46</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-21" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-21', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-22" class="py-name"><a title="Bio.Data" class="py-name" href="#" onclick="return doclink('link-22', 'Data', 'link-13');">Data</a></tt><tt class="py-op">.</tt><tt id="link-23" class="py-name"><a title="Bio.Data.IUPACData" class="py-name" href="#" onclick="return doclink('link-23', 'IUPACData', 'link-14');">IUPACData</a></tt> <tt class="py-keyword">import</tt> <tt id="link-24" class="py-name" targets="Variable Bio.Data.IUPACData.ambiguous_rna_letters=Bio.Data.IUPACData-module.html#ambiguous_rna_letters"><a title="Bio.Data.IUPACData.ambiguous_rna_letters" class="py-name" href="#" onclick="return doclink('link-24', 'ambiguous_rna_letters', 'link-24');">ambiguous_rna_letters</a></tt> <tt class="py-keyword">as</tt> <tt class="py-name">_ambiguous_rna_letters</tt> </tt>
<a name="L47"></a><tt class="py-lineno">  47</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-25" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-25', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-26" class="py-name"><a title="Bio.Data" class="py-name" href="#" onclick="return doclink('link-26', 'Data', 'link-13');">Data</a></tt> <tt class="py-keyword">import</tt> <tt id="link-27" class="py-name" targets="Module Bio.Data.CodonTable=Bio.Data.CodonTable-module.html,Class Bio.Data.CodonTable.CodonTable=Bio.Data.CodonTable.CodonTable-class.html"><a title="Bio.Data.CodonTable
Bio.Data.CodonTable.CodonTable" class="py-name" href="#" onclick="return doclink('link-27', 'CodonTable', 'link-27');">CodonTable</a></tt> </tt>
<a name="L48"></a><tt class="py-lineno">  48</tt>  <tt class="py-line"> </tt>
<a name="L49"></a><tt class="py-lineno">  49</tt>  <tt class="py-line"> </tt>
<a name="_maketrans"></a><div id="_maketrans-def"><a name="L50"></a><tt class="py-lineno">  50</tt> <a class="py-toggle" href="#" id="_maketrans-toggle" onclick="return toggle('_maketrans');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq-module.html#_maketrans">_maketrans</a><tt class="py-op">(</tt><tt class="py-param">complement_mapping</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_maketrans-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_maketrans-expanded"><a name="L51"></a><tt class="py-lineno">  51</tt>  <tt class="py-line">    <tt class="py-docstring">"""Make a python string translation table (PRIVATE).</tt> </tt>
<a name="L52"></a><tt class="py-lineno">  52</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L53"></a><tt class="py-lineno">  53</tt>  <tt class="py-line"><tt class="py-docstring">    Arguments:</tt> </tt>
<a name="L54"></a><tt class="py-lineno">  54</tt>  <tt class="py-line"><tt class="py-docstring">     - complement_mapping - a dictionary such as ambiguous_dna_complement</tt> </tt>
<a name="L55"></a><tt class="py-lineno">  55</tt>  <tt class="py-line"><tt class="py-docstring">       and ambiguous_rna_complement from Data.IUPACData.</tt> </tt>
<a name="L56"></a><tt class="py-lineno">  56</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L57"></a><tt class="py-lineno">  57</tt>  <tt class="py-line"><tt class="py-docstring">    Returns a translation table (a string of length 256) for use with the</tt> </tt>
<a name="L58"></a><tt class="py-lineno">  58</tt>  <tt class="py-line"><tt class="py-docstring">    python string's translate method to use in a (reverse) complement.</tt> </tt>
<a name="L59"></a><tt class="py-lineno">  59</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L60"></a><tt class="py-lineno">  60</tt>  <tt class="py-line"><tt class="py-docstring">    Compatible with lower case and upper case sequences.</tt> </tt>
<a name="L61"></a><tt class="py-lineno">  61</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L62"></a><tt class="py-lineno">  62</tt>  <tt class="py-line"><tt class="py-docstring">    For internal use only.</tt> </tt>
<a name="L63"></a><tt class="py-lineno">  63</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L64"></a><tt class="py-lineno">  64</tt>  <tt class="py-line">    <tt class="py-name">before</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt id="link-28" class="py-name" targets="Method Bio.Seq.MutableSeq.join()=Bio.Seq.MutableSeq-class.html#join,Method Bio.Seq.Seq.join()=Bio.Seq.Seq-class.html#join,Method Bio.Seq.UnknownSeq.join()=Bio.Seq.UnknownSeq-class.html#join"><a title="Bio.Seq.MutableSeq.join
Bio.Seq.Seq.join
Bio.Seq.UnknownSeq.join" class="py-name" href="#" onclick="return doclink('link-28', 'join', 'link-28');">join</a></tt><tt class="py-op">(</tt><tt class="py-name">complement_mapping</tt><tt class="py-op">.</tt><tt id="link-29" class="py-name" targets="Method Bio.Align.substitution_matrices.Array.keys()=Bio.Align.substitution_matrices.Array-class.html#keys,Method Bio.Crystal.Crystal.keys()=Bio.Crystal.Crystal-class.html#keys,Method Bio.File._IndexedSeqFileDict.keys()=Bio.File._IndexedSeqFileDict-class.html#keys,Method Bio.File._SQLiteManySeqFilesDict.keys()=Bio.File._SQLiteManySeqFilesDict-class.html#keys,Method Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys()=Bio.PDB.AbstractPropertyMap.AbstractPropertyMap-class.html#keys,Method Bio.Phylo.PhyloXML.Events.keys()=Bio.Phylo.PhyloXML.Events-class.html#keys,Method BioSQL.BioSeqDatabase.BioSeqDatabase.keys()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#keys,Method BioSQL.BioSeqDatabase.DBServer.keys()=BioSQL.BioSeqDatabase.DBServer-class.html#keys"><a title="Bio.Align.substitution_matrices.Array.keys
Bio.Crystal.Crystal.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-29', 'keys', 'link-29');">keys</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L65"></a><tt class="py-lineno">  65</tt>  <tt class="py-line">    <tt class="py-name">after</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt id="link-30" class="py-name"><a title="Bio.Seq.MutableSeq.join
Bio.Seq.Seq.join
Bio.Seq.UnknownSeq.join" class="py-name" href="#" onclick="return doclink('link-30', 'join', 'link-28');">join</a></tt><tt class="py-op">(</tt><tt class="py-name">complement_mapping</tt><tt class="py-op">.</tt><tt id="link-31" class="py-name" targets="Method Bio.Align.substitution_matrices.Array.values()=Bio.Align.substitution_matrices.Array-class.html#values,Method Bio.Crystal.Crystal.values()=Bio.Crystal.Crystal-class.html#values,Method Bio.File._IndexedSeqFileDict.values()=Bio.File._IndexedSeqFileDict-class.html#values,Method Bio.Phylo.PhyloXML.Events.values()=Bio.Phylo.PhyloXML.Events-class.html#values,Method BioSQL.BioSeqDatabase.BioSeqDatabase.values()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#values,Method BioSQL.BioSeqDatabase.DBServer.values()=BioSQL.BioSeqDatabase.DBServer-class.html#values"><a title="Bio.Align.substitution_matrices.Array.values
Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-31', 'values', 'link-31');">values</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L66"></a><tt class="py-lineno">  66</tt>  <tt class="py-line">    <tt class="py-name">before</tt> <tt class="py-op">+=</tt> <tt class="py-name">before</tt><tt class="py-op">.</tt><tt id="link-32" class="py-name" targets="Method Bio.Seq.Seq.lower()=Bio.Seq.Seq-class.html#lower,Method Bio.Seq.UnknownSeq.lower()=Bio.Seq.UnknownSeq-class.html#lower,Method Bio.SeqRecord.SeqRecord.lower()=Bio.SeqRecord.SeqRecord-class.html#lower"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-32', 'lower', 'link-32');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L67"></a><tt class="py-lineno">  67</tt>  <tt class="py-line">    <tt class="py-name">after</tt> <tt class="py-op">+=</tt> <tt class="py-name">after</tt><tt class="py-op">.</tt><tt id="link-33" class="py-name"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-33', 'lower', 'link-32');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L68"></a><tt class="py-lineno">  68</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">version_info</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-number">3</tt><tt class="py-op">:</tt> </tt>
<a name="L69"></a><tt class="py-lineno">  69</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">.</tt><tt class="py-name">maketrans</tt><tt class="py-op">(</tt><tt class="py-name">before</tt><tt class="py-op">,</tt> <tt class="py-name">after</tt><tt class="py-op">)</tt> </tt>
<a name="L70"></a><tt class="py-lineno">  70</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L71"></a><tt class="py-lineno">  71</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt class="py-name">maketrans</tt><tt class="py-op">(</tt><tt class="py-name">before</tt><tt class="py-op">,</tt> <tt class="py-name">after</tt><tt class="py-op">)</tt> </tt>
</div><a name="L72"></a><tt class="py-lineno">  72</tt>  <tt class="py-line"> </tt>
<a name="L73"></a><tt class="py-lineno">  73</tt>  <tt class="py-line"> </tt>
<a name="L74"></a><tt class="py-lineno">  74</tt>  <tt class="py-line"><tt id="link-34" class="py-name" targets="Variable Bio.Seq._dna_complement_table=Bio.Seq-module.html#_dna_complement_table"><a title="Bio.Seq._dna_complement_table" class="py-name" href="#" onclick="return doclink('link-34', '_dna_complement_table', 'link-34');">_dna_complement_table</a></tt> <tt class="py-op">=</tt> <tt id="link-35" class="py-name" targets="Function Bio.Seq._maketrans()=Bio.Seq-module.html#_maketrans"><a title="Bio.Seq._maketrans" class="py-name" href="#" onclick="return doclink('link-35', '_maketrans', 'link-35');">_maketrans</a></tt><tt class="py-op">(</tt><tt id="link-36" class="py-name"><a title="Bio.Data.IUPACData.ambiguous_dna_complement" class="py-name" href="#" onclick="return doclink('link-36', 'ambiguous_dna_complement', 'link-15');">ambiguous_dna_complement</a></tt><tt class="py-op">)</tt> </tt>
<a name="L75"></a><tt class="py-lineno">  75</tt>  <tt class="py-line"><tt id="link-37" class="py-name" targets="Variable Bio.Seq._rna_complement_table=Bio.Seq-module.html#_rna_complement_table"><a title="Bio.Seq._rna_complement_table" class="py-name" href="#" onclick="return doclink('link-37', '_rna_complement_table', 'link-37');">_rna_complement_table</a></tt> <tt class="py-op">=</tt> <tt id="link-38" class="py-name"><a title="Bio.Seq._maketrans" class="py-name" href="#" onclick="return doclink('link-38', '_maketrans', 'link-35');">_maketrans</a></tt><tt class="py-op">(</tt><tt id="link-39" class="py-name"><a title="Bio.Data.IUPACData.ambiguous_rna_complement" class="py-name" href="#" onclick="return doclink('link-39', 'ambiguous_rna_complement', 'link-16');">ambiguous_rna_complement</a></tt><tt class="py-op">)</tt> </tt>
<a name="L76"></a><tt class="py-lineno">  76</tt>  <tt class="py-line"> </tt>
<a name="L77"></a><tt class="py-lineno">  77</tt>  <tt class="py-line"> </tt>
<a name="Seq"></a><div id="Seq-def"><a name="L78"></a><tt class="py-lineno">  78</tt> <a class="py-toggle" href="#" id="Seq-toggle" onclick="return toggle('Seq');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html">Seq</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Seq-expanded"><a name="L79"></a><tt class="py-lineno">  79</tt>  <tt class="py-line">    <tt class="py-docstring">"""Read-only sequence object (essentially a string with an alphabet).</tt> </tt>
<a name="L80"></a><tt class="py-lineno">  80</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L81"></a><tt class="py-lineno">  81</tt>  <tt class="py-line"><tt class="py-docstring">    Like normal python strings, our basic sequence object is immutable.</tt> </tt>
<a name="L82"></a><tt class="py-lineno">  82</tt>  <tt class="py-line"><tt class="py-docstring">    This prevents you from doing my_seq[5] = "A" for example, but does allow</tt> </tt>
<a name="L83"></a><tt class="py-lineno">  83</tt>  <tt class="py-line"><tt class="py-docstring">    Seq objects to be used as dictionary keys.</tt> </tt>
<a name="L84"></a><tt class="py-lineno">  84</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L85"></a><tt class="py-lineno">  85</tt>  <tt class="py-line"><tt class="py-docstring">    The Seq object provides a number of string like methods (such as count,</tt> </tt>
<a name="L86"></a><tt class="py-lineno">  86</tt>  <tt class="py-line"><tt class="py-docstring">    find, split and strip), which are alphabet aware where appropriate.</tt> </tt>
<a name="L87"></a><tt class="py-lineno">  87</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L88"></a><tt class="py-lineno">  88</tt>  <tt class="py-line"><tt class="py-docstring">    In addition to the string like sequence, the Seq object has an alphabet</tt> </tt>
<a name="L89"></a><tt class="py-lineno">  89</tt>  <tt class="py-line"><tt class="py-docstring">    property. This is an instance of an Alphabet class from Bio.Alphabet,</tt> </tt>
<a name="L90"></a><tt class="py-lineno">  90</tt>  <tt class="py-line"><tt class="py-docstring">    for example generic DNA, or IUPAC DNA. This describes the type of molecule</tt> </tt>
<a name="L91"></a><tt class="py-lineno">  91</tt>  <tt class="py-line"><tt class="py-docstring">    (e.g. RNA, DNA, protein) and may also indicate the expected symbols</tt> </tt>
<a name="L92"></a><tt class="py-lineno">  92</tt>  <tt class="py-line"><tt class="py-docstring">    (letters).</tt> </tt>
<a name="L93"></a><tt class="py-lineno">  93</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L94"></a><tt class="py-lineno">  94</tt>  <tt class="py-line"><tt class="py-docstring">    The Seq object also provides some biological methods, such as complement,</tt> </tt>
<a name="L95"></a><tt class="py-lineno">  95</tt>  <tt class="py-line"><tt class="py-docstring">    reverse_complement, transcribe, back_transcribe and translate (which are</tt> </tt>
<a name="L96"></a><tt class="py-lineno">  96</tt>  <tt class="py-line"><tt class="py-docstring">    not applicable to sequences with a protein alphabet).</tt> </tt>
<a name="L97"></a><tt class="py-lineno">  97</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L98"></a><tt class="py-lineno">  98</tt>  <tt class="py-line"> </tt>
<a name="Seq.__init__"></a><div id="Seq.__init__-def"><a name="L99"></a><tt class="py-lineno">  99</tt> <a class="py-toggle" href="#" id="Seq.__init__-toggle" onclick="return toggle('Seq.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">data</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt id="link-40" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-40', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-41" class="py-name" targets="Variable Bio.Alphabet.generic_alphabet=Bio.Alphabet-module.html#generic_alphabet"><a title="Bio.Alphabet.generic_alphabet" class="py-name" href="#" onclick="return doclink('link-41', 'generic_alphabet', 'link-41');">generic_alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.__init__-expanded"><a name="L100"></a><tt class="py-lineno"> 100</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a Seq object.</tt> </tt>
<a name="L101"></a><tt class="py-lineno"> 101</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L102"></a><tt class="py-lineno"> 102</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L103"></a><tt class="py-lineno"> 103</tt>  <tt class="py-line"><tt class="py-docstring">         - seq - Sequence, required (string)</tt> </tt>
<a name="L104"></a><tt class="py-lineno"> 104</tt>  <tt class="py-line"><tt class="py-docstring">         - alphabet - Optional argument, an Alphabet object from</tt> </tt>
<a name="L105"></a><tt class="py-lineno"> 105</tt>  <tt class="py-line"><tt class="py-docstring">           Bio.Alphabet</tt> </tt>
<a name="L106"></a><tt class="py-lineno"> 106</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L107"></a><tt class="py-lineno"> 107</tt>  <tt class="py-line"><tt class="py-docstring">        You will typically use Bio.SeqIO to read in sequences from files as</tt> </tt>
<a name="L108"></a><tt class="py-lineno"> 108</tt>  <tt class="py-line"><tt class="py-docstring">        SeqRecord objects, whose sequence will be exposed as a Seq object via</tt> </tt>
<a name="L109"></a><tt class="py-lineno"> 109</tt>  <tt class="py-line"><tt class="py-docstring">        the seq property.</tt> </tt>
<a name="L110"></a><tt class="py-lineno"> 110</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L111"></a><tt class="py-lineno"> 111</tt>  <tt class="py-line"><tt class="py-docstring">        However, will often want to create your own Seq objects directly:</tt> </tt>
<a name="L112"></a><tt class="py-lineno"> 112</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L113"></a><tt class="py-lineno"> 113</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L114"></a><tt class="py-lineno"> 114</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import IUPAC</tt> </tt>
<a name="L115"></a><tt class="py-lineno"> 115</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq = Seq("MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF",</tt> </tt>
<a name="L116"></a><tt class="py-lineno"> 116</tt>  <tt class="py-line"><tt class="py-docstring">        ...              IUPAC.protein)</tt> </tt>
<a name="L117"></a><tt class="py-lineno"> 117</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L118"></a><tt class="py-lineno"> 118</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF', IUPACProtein())</tt> </tt>
<a name="L119"></a><tt class="py-lineno"> 119</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_seq)</tt> </tt>
<a name="L120"></a><tt class="py-lineno"> 120</tt>  <tt class="py-line"><tt class="py-docstring">        MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF</tt> </tt>
<a name="L121"></a><tt class="py-lineno"> 121</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.alphabet</tt> </tt>
<a name="L122"></a><tt class="py-lineno"> 122</tt>  <tt class="py-line"><tt class="py-docstring">        IUPACProtein()</tt> </tt>
<a name="L123"></a><tt class="py-lineno"> 123</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L124"></a><tt class="py-lineno"> 124</tt>  <tt class="py-line">        <tt class="py-comment"># Enforce string storage</tt> </tt>
<a name="L125"></a><tt class="py-lineno"> 125</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-42" class="py-name" targets="Variable BioSQL.BioSeq.DBSeq.data=BioSQL.BioSeq.DBSeq-class.html#data"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-42', 'data', 'link-42');">data</a></tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L126"></a><tt class="py-lineno"> 126</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"The sequence data given to a Seq object should "</tt> </tt>
<a name="L127"></a><tt class="py-lineno"> 127</tt>  <tt class="py-line">                            <tt class="py-string">"be a string (not another Seq object etc)"</tt><tt class="py-op">)</tt> </tt>
<a name="L128"></a><tt class="py-lineno"> 128</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_data</tt> <tt class="py-op">=</tt> <tt id="link-43" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-43', 'data', 'link-42');">data</a></tt> </tt>
<a name="L129"></a><tt class="py-lineno"> 129</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-44" class="py-name" targets="Variable Bio.Align.substitution_matrices.Array.alphabet=Bio.Align.substitution_matrices.Array-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSP.alphabet=Bio.SearchIO._model.hsp.HSP-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSPFragment.alphabet=Bio.SearchIO._model.hsp.HSPFragment-class.html#alphabet"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-44', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-45" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-45', 'alphabet', 'link-44');">alphabet</a></tt>  <tt class="py-comment"># Seq API requirement</tt> </tt>
</div><a name="L130"></a><tt class="py-lineno"> 130</tt>  <tt class="py-line"> </tt>
<a name="Seq.__repr__"></a><div id="Seq.__repr__-def"><a name="L131"></a><tt class="py-lineno"> 131</tt> <a class="py-toggle" href="#" id="Seq.__repr__-toggle" onclick="return toggle('Seq.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.__repr__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.__repr__-expanded"><a name="L132"></a><tt class="py-lineno"> 132</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return (truncated) representation of the sequence for debugging."""</tt> </tt>
<a name="L133"></a><tt class="py-lineno"> 133</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-46" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-46', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-keyword">is</tt> <tt id="link-47" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-47', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-48" class="py-name"><a title="Bio.Alphabet.generic_alphabet" class="py-name" href="#" onclick="return doclink('link-48', 'generic_alphabet', 'link-41');">generic_alphabet</a></tt><tt class="py-op">:</tt> </tt>
<a name="L134"></a><tt class="py-lineno"> 134</tt>  <tt class="py-line">            <tt class="py-comment"># Default used, we can omit it and simplify the representation</tt> </tt>
<a name="L135"></a><tt class="py-lineno"> 135</tt>  <tt class="py-line">            <tt id="link-49" class="py-name" targets="Variable Bio.Graphics.ColorSpiral.ColorSpiral.a=Bio.Graphics.ColorSpiral.ColorSpiral-class.html#a"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-49', 'a', 'link-49');">a</a></tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L136"></a><tt class="py-lineno"> 136</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L137"></a><tt class="py-lineno"> 137</tt>  <tt class="py-line">            <tt id="link-50" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-50', 'a', 'link-49');">a</a></tt> <tt class="py-op">=</tt> <tt class="py-string">", %r"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-51" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-51', 'alphabet', 'link-44');">alphabet</a></tt> </tt>
<a name="L138"></a><tt class="py-lineno"> 138</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">60</tt><tt class="py-op">:</tt> </tt>
<a name="L139"></a><tt class="py-lineno"> 139</tt>  <tt class="py-line">            <tt class="py-comment"># Shows the last three letters as it is often useful to see if</tt> </tt>
<a name="L140"></a><tt class="py-lineno"> 140</tt>  <tt class="py-line">            <tt class="py-comment"># there is a stop codon at the end of a sequence.</tt> </tt>
<a name="L141"></a><tt class="py-lineno"> 141</tt>  <tt class="py-line">            <tt class="py-comment"># Note total length is 54+3+3=60</tt> </tt>
<a name="L142"></a><tt class="py-lineno"> 142</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-string">"{0}('{1}...{2}'{3!s})"</tt><tt class="py-op">.</tt><tt id="link-52" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.format()=Bio.Align.MultipleSeqAlignment-class.html#format,Method Bio.Align.PairwiseAlignment.format()=Bio.Align.PairwiseAlignment-class.html#format,Method Bio.Phylo.BaseTree.Tree.format()=Bio.Phylo.BaseTree.Tree-class.html#format,Method Bio.Restriction.Restriction.RestrictionBatch.format()=Bio.Restriction.Restriction.RestrictionBatch-class.html#format,Method Bio.SeqRecord.SeqRecord.format()=Bio.SeqRecord.SeqRecord-class.html#format,Method Bio.SubsMat.SeqMat.format()=Bio.SubsMat.SeqMat-class.html#format,Method Bio.motifs.Motif.format()=Bio.motifs.Motif-class.html#format"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-52', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">,</tt> </tt>
<a name="L143"></a><tt class="py-lineno"> 143</tt>  <tt class="py-line">                                                  <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">54</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L144"></a><tt class="py-lineno"> 144</tt>  <tt class="py-line">                                                  <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">3</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L145"></a><tt class="py-lineno"> 145</tt>  <tt class="py-line">                                                  <tt id="link-53" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-53', 'a', 'link-49');">a</a></tt><tt class="py-op">)</tt> </tt>
<a name="L146"></a><tt class="py-lineno"> 146</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L147"></a><tt class="py-lineno"> 147</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-string">"{0}({1!r}{2!s})"</tt><tt class="py-op">.</tt><tt id="link-54" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-54', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">,</tt> </tt>
<a name="L148"></a><tt class="py-lineno"> 148</tt>  <tt class="py-line">                                            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_data</tt><tt class="py-op">,</tt> </tt>
<a name="L149"></a><tt class="py-lineno"> 149</tt>  <tt class="py-line">                                            <tt id="link-55" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-55', 'a', 'link-49');">a</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L150"></a><tt class="py-lineno"> 150</tt>  <tt class="py-line"> </tt>
<a name="Seq.__str__"></a><div id="Seq.__str__-def"><a name="L151"></a><tt class="py-lineno"> 151</tt> <a class="py-toggle" href="#" id="Seq.__str__-toggle" onclick="return toggle('Seq.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.__str__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.__str__-expanded"><a name="L152"></a><tt class="py-lineno"> 152</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the full sequence as a python string, use str(my_seq).</tt> </tt>
<a name="L153"></a><tt class="py-lineno"> 153</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L154"></a><tt class="py-lineno"> 154</tt>  <tt class="py-line"><tt class="py-docstring">        Note that Biopython 1.44 and earlier would give a truncated</tt> </tt>
<a name="L155"></a><tt class="py-lineno"> 155</tt>  <tt class="py-line"><tt class="py-docstring">        version of repr(my_seq) for str(my_seq).  If you are writing code</tt> </tt>
<a name="L156"></a><tt class="py-lineno"> 156</tt>  <tt class="py-line"><tt class="py-docstring">        which need to be backwards compatible with really old Biopython,</tt> </tt>
<a name="L157"></a><tt class="py-lineno"> 157</tt>  <tt class="py-line"><tt class="py-docstring">        you should continue to use my_seq.tostring() as follows::</tt> </tt>
<a name="L158"></a><tt class="py-lineno"> 158</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L159"></a><tt class="py-lineno"> 159</tt>  <tt class="py-line"><tt class="py-docstring">            try:</tt> </tt>
<a name="L160"></a><tt class="py-lineno"> 160</tt>  <tt class="py-line"><tt class="py-docstring">                # The old way, removed in Biopython 1.73</tt> </tt>
<a name="L161"></a><tt class="py-lineno"> 161</tt>  <tt class="py-line"><tt class="py-docstring">                as_string = seq_obj.tostring()</tt> </tt>
<a name="L162"></a><tt class="py-lineno"> 162</tt>  <tt class="py-line"><tt class="py-docstring">            except AttributeError:</tt> </tt>
<a name="L163"></a><tt class="py-lineno"> 163</tt>  <tt class="py-line"><tt class="py-docstring">                # The new way, needs Biopython 1.45 or later.</tt> </tt>
<a name="L164"></a><tt class="py-lineno"> 164</tt>  <tt class="py-line"><tt class="py-docstring">                # Don't use this on Biopython 1.44 or older as truncates</tt> </tt>
<a name="L165"></a><tt class="py-lineno"> 165</tt>  <tt class="py-line"><tt class="py-docstring">                as_string = str(seq_obj)</tt> </tt>
<a name="L166"></a><tt class="py-lineno"> 166</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L167"></a><tt class="py-lineno"> 167</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L168"></a><tt class="py-lineno"> 168</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_data</tt> </tt>
</div><a name="L169"></a><tt class="py-lineno"> 169</tt>  <tt class="py-line"> </tt>
<a name="Seq.__hash__"></a><div id="Seq.__hash__-def"><a name="L170"></a><tt class="py-lineno"> 170</tt> <a class="py-toggle" href="#" id="Seq.__hash__-toggle" onclick="return toggle('Seq.__hash__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#__hash__">__hash__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.__hash__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.__hash__-expanded"><a name="L171"></a><tt class="py-lineno"> 171</tt>  <tt class="py-line">        <tt class="py-docstring">"""Hash for comparison.</tt> </tt>
<a name="L172"></a><tt class="py-lineno"> 172</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L173"></a><tt class="py-lineno"> 173</tt>  <tt class="py-line"><tt class="py-docstring">        See the __cmp__ documentation - this has changed from past</tt> </tt>
<a name="L174"></a><tt class="py-lineno"> 174</tt>  <tt class="py-line"><tt class="py-docstring">        versions of Biopython!</tt> </tt>
<a name="L175"></a><tt class="py-lineno"> 175</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L176"></a><tt class="py-lineno"> 176</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - remove this warning in a future release</tt> </tt>
<a name="L177"></a><tt class="py-lineno"> 177</tt>  <tt class="py-line">        <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Biopython Seq objects now use string comparison. "</tt> </tt>
<a name="L178"></a><tt class="py-lineno"> 178</tt>  <tt class="py-line">                      <tt class="py-string">"Older versions of Biopython used object comparison. "</tt> </tt>
<a name="L179"></a><tt class="py-lineno"> 179</tt>  <tt class="py-line">                      <tt class="py-string">"During this transition, please use hash(id(my_seq)) "</tt> </tt>
<a name="L180"></a><tt class="py-lineno"> 180</tt>  <tt class="py-line">                      <tt class="py-string">"or my_dict[id(my_seq)] if you want the old behaviour, "</tt> </tt>
<a name="L181"></a><tt class="py-lineno"> 181</tt>  <tt class="py-line">                      <tt class="py-string">"or use hash(str(my_seq)) or my_dict[str(my_seq)] for "</tt> </tt>
<a name="L182"></a><tt class="py-lineno"> 182</tt>  <tt class="py-line">                      <tt class="py-string">"the new string hashing behaviour."</tt><tt class="py-op">,</tt> <tt id="link-56" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-56', 'BiopythonWarning', 'link-6');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L183"></a><tt class="py-lineno"> 183</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">hash</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L184"></a><tt class="py-lineno"> 184</tt>  <tt class="py-line"> </tt>
<a name="Seq.__eq__"></a><div id="Seq.__eq__-def"><a name="L185"></a><tt class="py-lineno"> 185</tt> <a class="py-toggle" href="#" id="Seq.__eq__-toggle" onclick="return toggle('Seq.__eq__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#__eq__">__eq__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.__eq__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.__eq__-expanded"><a name="L186"></a><tt class="py-lineno"> 186</tt>  <tt class="py-line">        <tt class="py-docstring">"""Compare the sequence to another sequence or a string (README).</tt> </tt>
<a name="L187"></a><tt class="py-lineno"> 187</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L188"></a><tt class="py-lineno"> 188</tt>  <tt class="py-line"><tt class="py-docstring">        Historically comparing Seq objects has done Python object comparison.</tt> </tt>
<a name="L189"></a><tt class="py-lineno"> 189</tt>  <tt class="py-line"><tt class="py-docstring">        After considerable discussion (keeping in mind constraints of the</tt> </tt>
<a name="L190"></a><tt class="py-lineno"> 190</tt>  <tt class="py-line"><tt class="py-docstring">        Python language, hashes and dictionary support), Biopython now uses</tt> </tt>
<a name="L191"></a><tt class="py-lineno"> 191</tt>  <tt class="py-line"><tt class="py-docstring">        simple string comparison (with a warning about the change).</tt> </tt>
<a name="L192"></a><tt class="py-lineno"> 192</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L193"></a><tt class="py-lineno"> 193</tt>  <tt class="py-line"><tt class="py-docstring">        Note that incompatible alphabets (e.g. DNA to RNA) will trigger a</tt> </tt>
<a name="L194"></a><tt class="py-lineno"> 194</tt>  <tt class="py-line"><tt class="py-docstring">        warning.</tt> </tt>
<a name="L195"></a><tt class="py-lineno"> 195</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L196"></a><tt class="py-lineno"> 196</tt>  <tt class="py-line"><tt class="py-docstring">        During this transition period, please just do explicit comparisons:</tt> </tt>
<a name="L197"></a><tt class="py-lineno"> 197</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L198"></a><tt class="py-lineno"> 198</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L199"></a><tt class="py-lineno"> 199</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; seq1 = Seq("ACGT")</tt> </tt>
<a name="L200"></a><tt class="py-lineno"> 200</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; seq2 = Seq("ACGT")</tt> </tt>
<a name="L201"></a><tt class="py-lineno"> 201</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; id(seq1) == id(seq2)</tt> </tt>
<a name="L202"></a><tt class="py-lineno"> 202</tt>  <tt class="py-line"><tt class="py-docstring">        False</tt> </tt>
<a name="L203"></a><tt class="py-lineno"> 203</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; str(seq1) == str(seq2)</tt> </tt>
<a name="L204"></a><tt class="py-lineno"> 204</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L205"></a><tt class="py-lineno"> 205</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L206"></a><tt class="py-lineno"> 206</tt>  <tt class="py-line"><tt class="py-docstring">        The new behaviour is to use string-like equality:</tt> </tt>
<a name="L207"></a><tt class="py-lineno"> 207</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L208"></a><tt class="py-lineno"> 208</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L209"></a><tt class="py-lineno"> 209</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_dna</tt> </tt>
<a name="L210"></a><tt class="py-lineno"> 210</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; seq1 == seq2</tt> </tt>
<a name="L211"></a><tt class="py-lineno"> 211</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L212"></a><tt class="py-lineno"> 212</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; seq1 == "ACGT"</tt> </tt>
<a name="L213"></a><tt class="py-lineno"> 213</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L214"></a><tt class="py-lineno"> 214</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; seq1 == Seq("ACGT", generic_dna)</tt> </tt>
<a name="L215"></a><tt class="py-lineno"> 215</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L216"></a><tt class="py-lineno"> 216</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L217"></a><tt class="py-lineno"> 217</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L218"></a><tt class="py-lineno"> 218</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-57" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.other()=Bio.Phylo.PhyloXMLIO.Parser-class.html#other,Method Bio.Phylo.PhyloXMLIO.Writer.other()=Bio.Phylo.PhyloXMLIO.Writer-class.html#other"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-57', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-string">"alphabet"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L219"></a><tt class="py-lineno"> 219</tt>  <tt class="py-line">            <tt class="py-comment"># other could be a Seq or a MutableSeq</tt> </tt>
<a name="L220"></a><tt class="py-lineno"> 220</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-58" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-58', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-59" class="py-name" targets="Function Bio.Alphabet._check_type_compatible()=Bio.Alphabet-module.html#_check_type_compatible"><a title="Bio.Alphabet._check_type_compatible" class="py-name" href="#" onclick="return doclink('link-59', '_check_type_compatible', 'link-59');">_check_type_compatible</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-60" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-60', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L221"></a><tt class="py-lineno"> 221</tt>  <tt class="py-line">                                                    <tt id="link-61" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-61', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-62" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-62', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L222"></a><tt class="py-lineno"> 222</tt>  <tt class="py-line">                <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Incompatible alphabets {0!r} and {1!r}"</tt><tt class="py-op">.</tt><tt id="link-63" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-63', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L223"></a><tt class="py-lineno"> 223</tt>  <tt class="py-line">                              <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-64" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-64', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-65" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-65', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-66" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-66', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L224"></a><tt class="py-lineno"> 224</tt>  <tt class="py-line">                              <tt id="link-67" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-67', 'BiopythonWarning', 'link-6');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L225"></a><tt class="py-lineno"> 225</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-68" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-68', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L226"></a><tt class="py-lineno"> 226</tt>  <tt class="py-line"> </tt>
<a name="Seq.__ne__"></a><div id="Seq.__ne__-def"><a name="L227"></a><tt class="py-lineno"> 227</tt> <a class="py-toggle" href="#" id="Seq.__ne__-toggle" onclick="return toggle('Seq.__ne__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#__ne__">__ne__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.__ne__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.__ne__-expanded"><a name="L228"></a><tt class="py-lineno"> 228</tt>  <tt class="py-line">        <tt class="py-docstring">"""Implement the not-equal operand."""</tt> </tt>
<a name="L229"></a><tt class="py-lineno"> 229</tt>  <tt class="py-line">        <tt class="py-comment"># Require this method for Python 2 but not needed on Python 3</tt> </tt>
<a name="L230"></a><tt class="py-lineno"> 230</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt> <tt class="py-op">==</tt> <tt id="link-69" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-69', 'other', 'link-57');">other</a></tt> </tt>
</div><a name="L231"></a><tt class="py-lineno"> 231</tt>  <tt class="py-line"> </tt>
<a name="Seq.__lt__"></a><div id="Seq.__lt__-def"><a name="L232"></a><tt class="py-lineno"> 232</tt> <a class="py-toggle" href="#" id="Seq.__lt__-toggle" onclick="return toggle('Seq.__lt__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#__lt__">__lt__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.__lt__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.__lt__-expanded"><a name="L233"></a><tt class="py-lineno"> 233</tt>  <tt class="py-line">        <tt class="py-docstring">"""Implement the less-than operand."""</tt> </tt>
<a name="L234"></a><tt class="py-lineno"> 234</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-70" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-70', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-string">"alphabet"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L235"></a><tt class="py-lineno"> 235</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-71" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-71', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-72" class="py-name"><a title="Bio.Alphabet._check_type_compatible" class="py-name" href="#" onclick="return doclink('link-72', '_check_type_compatible', 'link-59');">_check_type_compatible</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-73" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-73', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L236"></a><tt class="py-lineno"> 236</tt>  <tt class="py-line">                                                    <tt id="link-74" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-74', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-75" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-75', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L237"></a><tt class="py-lineno"> 237</tt>  <tt class="py-line">                <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Incompatible alphabets {0!r} and {1!r}"</tt><tt class="py-op">.</tt><tt id="link-76" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-76', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L238"></a><tt class="py-lineno"> 238</tt>  <tt class="py-line">                              <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-77" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-77', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-78" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-78', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-79" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-79', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L239"></a><tt class="py-lineno"> 239</tt>  <tt class="py-line">                              <tt id="link-80" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-80', 'BiopythonWarning', 'link-6');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L240"></a><tt class="py-lineno"> 240</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-81" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-81', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">,</tt> <tt id="link-82" class="py-name" targets="Module Bio.Seq=Bio.Seq-module.html,Class Bio.Seq.Seq=Bio.Seq.Seq-class.html"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-82', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">,</tt> <tt id="link-83" class="py-name" targets="Class Bio.Seq.MutableSeq=Bio.Seq.MutableSeq-class.html"><a title="Bio.Seq.MutableSeq" class="py-name" href="#" onclick="return doclink('link-83', 'MutableSeq', 'link-83');">MutableSeq</a></tt><tt class="py-op">,</tt> <tt id="link-84" class="py-name" targets="Class Bio.Seq.UnknownSeq=Bio.Seq.UnknownSeq-class.html"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-84', 'UnknownSeq', 'link-84');">UnknownSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L241"></a><tt class="py-lineno"> 241</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-85" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-85', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt> </tt>
<a name="L242"></a><tt class="py-lineno"> 242</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"'&lt;' not supported between instances of '{}' and '{}'"</tt> </tt>
<a name="L243"></a><tt class="py-lineno"> 243</tt>  <tt class="py-line">                        <tt class="py-op">.</tt><tt id="link-86" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-86', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt id="link-87" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.type()=Bio.Phylo.PhyloXMLIO.Writer-class.html#type"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-87', 'type', 'link-87');">type</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">,</tt> <tt id="link-88" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-88', 'type', 'link-87');">type</a></tt><tt class="py-op">(</tt><tt id="link-89" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-89', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L244"></a><tt class="py-lineno"> 244</tt>  <tt class="py-line"> </tt>
<a name="Seq.__le__"></a><div id="Seq.__le__-def"><a name="L245"></a><tt class="py-lineno"> 245</tt> <a class="py-toggle" href="#" id="Seq.__le__-toggle" onclick="return toggle('Seq.__le__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#__le__">__le__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.__le__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.__le__-expanded"><a name="L246"></a><tt class="py-lineno"> 246</tt>  <tt class="py-line">        <tt class="py-docstring">"""Implement the less-than or equal operand."""</tt> </tt>
<a name="L247"></a><tt class="py-lineno"> 247</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-90" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-90', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-string">"alphabet"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L248"></a><tt class="py-lineno"> 248</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-91" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-91', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-92" class="py-name"><a title="Bio.Alphabet._check_type_compatible" class="py-name" href="#" onclick="return doclink('link-92', '_check_type_compatible', 'link-59');">_check_type_compatible</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-93" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-93', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L249"></a><tt class="py-lineno"> 249</tt>  <tt class="py-line">                                                    <tt id="link-94" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-94', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-95" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-95', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L250"></a><tt class="py-lineno"> 250</tt>  <tt class="py-line">                <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Incompatible alphabets {0!r} and {1!r}"</tt><tt class="py-op">.</tt><tt id="link-96" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-96', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L251"></a><tt class="py-lineno"> 251</tt>  <tt class="py-line">                              <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-97" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-97', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-98" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-98', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-99" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-99', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L252"></a><tt class="py-lineno"> 252</tt>  <tt class="py-line">                              <tt id="link-100" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-100', 'BiopythonWarning', 'link-6');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L253"></a><tt class="py-lineno"> 253</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-101" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-101', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">,</tt> <tt id="link-102" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-102', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">,</tt> <tt id="link-103" class="py-name"><a title="Bio.Seq.MutableSeq" class="py-name" href="#" onclick="return doclink('link-103', 'MutableSeq', 'link-83');">MutableSeq</a></tt><tt class="py-op">,</tt> <tt id="link-104" class="py-name"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-104', 'UnknownSeq', 'link-84');">UnknownSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L254"></a><tt class="py-lineno"> 254</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-105" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-105', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt> </tt>
<a name="L255"></a><tt class="py-lineno"> 255</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"'&lt;=' not supported between instances of '{}' and '{}'"</tt> </tt>
<a name="L256"></a><tt class="py-lineno"> 256</tt>  <tt class="py-line">                        <tt class="py-op">.</tt><tt id="link-106" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-106', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt id="link-107" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-107', 'type', 'link-87');">type</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">,</tt> <tt id="link-108" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-108', 'type', 'link-87');">type</a></tt><tt class="py-op">(</tt><tt id="link-109" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-109', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L257"></a><tt class="py-lineno"> 257</tt>  <tt class="py-line"> </tt>
<a name="Seq.__gt__"></a><div id="Seq.__gt__-def"><a name="L258"></a><tt class="py-lineno"> 258</tt> <a class="py-toggle" href="#" id="Seq.__gt__-toggle" onclick="return toggle('Seq.__gt__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#__gt__">__gt__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.__gt__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.__gt__-expanded"><a name="L259"></a><tt class="py-lineno"> 259</tt>  <tt class="py-line">        <tt class="py-docstring">"""Implement the greater-than operand."""</tt> </tt>
<a name="L260"></a><tt class="py-lineno"> 260</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-110" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-110', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-string">"alphabet"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L261"></a><tt class="py-lineno"> 261</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-111" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-111', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-112" class="py-name"><a title="Bio.Alphabet._check_type_compatible" class="py-name" href="#" onclick="return doclink('link-112', '_check_type_compatible', 'link-59');">_check_type_compatible</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-113" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-113', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L262"></a><tt class="py-lineno"> 262</tt>  <tt class="py-line">                                                    <tt id="link-114" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-114', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-115" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-115', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L263"></a><tt class="py-lineno"> 263</tt>  <tt class="py-line">                <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Incompatible alphabets {0!r} and {1!r}"</tt><tt class="py-op">.</tt><tt id="link-116" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-116', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L264"></a><tt class="py-lineno"> 264</tt>  <tt class="py-line">                              <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-117" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-117', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-118" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-118', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-119" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-119', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L265"></a><tt class="py-lineno"> 265</tt>  <tt class="py-line">                              <tt id="link-120" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-120', 'BiopythonWarning', 'link-6');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L266"></a><tt class="py-lineno"> 266</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-121" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-121', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">,</tt> <tt id="link-122" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-122', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">,</tt> <tt id="link-123" class="py-name"><a title="Bio.Seq.MutableSeq" class="py-name" href="#" onclick="return doclink('link-123', 'MutableSeq', 'link-83');">MutableSeq</a></tt><tt class="py-op">,</tt> <tt id="link-124" class="py-name"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-124', 'UnknownSeq', 'link-84');">UnknownSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L267"></a><tt class="py-lineno"> 267</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-125" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-125', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt> </tt>
<a name="L268"></a><tt class="py-lineno"> 268</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"'&gt;' not supported between instances of '{}' and '{}'"</tt> </tt>
<a name="L269"></a><tt class="py-lineno"> 269</tt>  <tt class="py-line">                        <tt class="py-op">.</tt><tt id="link-126" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-126', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt id="link-127" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-127', 'type', 'link-87');">type</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">,</tt> <tt id="link-128" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-128', 'type', 'link-87');">type</a></tt><tt class="py-op">(</tt><tt id="link-129" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-129', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L270"></a><tt class="py-lineno"> 270</tt>  <tt class="py-line"> </tt>
<a name="Seq.__ge__"></a><div id="Seq.__ge__-def"><a name="L271"></a><tt class="py-lineno"> 271</tt> <a class="py-toggle" href="#" id="Seq.__ge__-toggle" onclick="return toggle('Seq.__ge__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#__ge__">__ge__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.__ge__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.__ge__-expanded"><a name="L272"></a><tt class="py-lineno"> 272</tt>  <tt class="py-line">        <tt class="py-docstring">"""Implement the greater-than or equal operand."""</tt> </tt>
<a name="L273"></a><tt class="py-lineno"> 273</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-130" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-130', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-string">"alphabet"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L274"></a><tt class="py-lineno"> 274</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-131" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-131', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-132" class="py-name"><a title="Bio.Alphabet._check_type_compatible" class="py-name" href="#" onclick="return doclink('link-132', '_check_type_compatible', 'link-59');">_check_type_compatible</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-133" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-133', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L275"></a><tt class="py-lineno"> 275</tt>  <tt class="py-line">                                                    <tt id="link-134" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-134', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-135" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-135', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L276"></a><tt class="py-lineno"> 276</tt>  <tt class="py-line">                <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Incompatible alphabets {0!r} and {1!r}"</tt><tt class="py-op">.</tt><tt id="link-136" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-136', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L277"></a><tt class="py-lineno"> 277</tt>  <tt class="py-line">                              <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-137" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-137', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-138" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-138', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-139" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-139', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L278"></a><tt class="py-lineno"> 278</tt>  <tt class="py-line">                              <tt id="link-140" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-140', 'BiopythonWarning', 'link-6');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L279"></a><tt class="py-lineno"> 279</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-141" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-141', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">,</tt> <tt id="link-142" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-142', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">,</tt> <tt id="link-143" class="py-name"><a title="Bio.Seq.MutableSeq" class="py-name" href="#" onclick="return doclink('link-143', 'MutableSeq', 'link-83');">MutableSeq</a></tt><tt class="py-op">,</tt> <tt id="link-144" class="py-name"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-144', 'UnknownSeq', 'link-84');">UnknownSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L280"></a><tt class="py-lineno"> 280</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-145" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-145', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt> </tt>
<a name="L281"></a><tt class="py-lineno"> 281</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"'&gt;=' not supported between instances of '{}' and '{}'"</tt> </tt>
<a name="L282"></a><tt class="py-lineno"> 282</tt>  <tt class="py-line">                        <tt class="py-op">.</tt><tt id="link-146" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-146', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt id="link-147" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-147', 'type', 'link-87');">type</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">,</tt> <tt id="link-148" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-148', 'type', 'link-87');">type</a></tt><tt class="py-op">(</tt><tt id="link-149" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-149', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L283"></a><tt class="py-lineno"> 283</tt>  <tt class="py-line"> </tt>
<a name="Seq.__len__"></a><div id="Seq.__len__-def"><a name="L284"></a><tt class="py-lineno"> 284</tt> <a class="py-toggle" href="#" id="Seq.__len__-toggle" onclick="return toggle('Seq.__len__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#__len__">__len__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.__len__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.__len__-expanded"><a name="L285"></a><tt class="py-lineno"> 285</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the length of the sequence, use len(my_seq)."""</tt> </tt>
<a name="L286"></a><tt class="py-lineno"> 286</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_data</tt><tt class="py-op">)</tt>  <tt class="py-comment"># Seq API requirement</tt> </tt>
</div><a name="L287"></a><tt class="py-lineno"> 287</tt>  <tt class="py-line"> </tt>
<a name="Seq.__getitem__"></a><div id="Seq.__getitem__-def"><a name="L288"></a><tt class="py-lineno"> 288</tt> <a class="py-toggle" href="#" id="Seq.__getitem__-toggle" onclick="return toggle('Seq.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">index</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># Seq API requirement</tt> </tt>
</div><a name="L289"></a><tt class="py-lineno"> 289</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a subsequence of single letter, use my_seq[index].</tt> </tt>
<a name="L290"></a><tt class="py-lineno"> 290</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L291"></a><tt class="py-lineno"> 291</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq = Seq('ACTCGACGTCG')</tt> </tt>
<a name="L292"></a><tt class="py-lineno"> 292</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq[5]</tt> </tt>
<a name="L293"></a><tt class="py-lineno"> 293</tt>  <tt class="py-line"><tt class="py-docstring">        'A'</tt> </tt>
<a name="L294"></a><tt class="py-lineno"> 294</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L295"></a><tt class="py-lineno"> 295</tt>  <tt class="py-line">        <tt class="py-comment"># Note since Python 2.0, __getslice__ is deprecated</tt> </tt>
<a name="L296"></a><tt class="py-lineno"> 296</tt>  <tt class="py-line">        <tt class="py-comment"># and __getitem__ is used instead.</tt> </tt>
<a name="L297"></a><tt class="py-lineno"> 297</tt>  <tt class="py-line">        <tt class="py-comment"># See http://docs.python.org/ref/sequence-methods.html</tt> </tt>
<a name="L298"></a><tt class="py-lineno"> 298</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-150" class="py-name" targets="Method Bio.Crystal.Chain.index()=Bio.Crystal.Chain-class.html#index,Method Bio.SCOP.Raf.SeqMap.index()=Bio.SCOP.Raf.SeqMap-class.html#index,Method Bio.SearchIO._model.hit.Hit.index()=Bio.SearchIO._model.hit.Hit-class.html#index,Method Bio.SearchIO._model.query.QueryResult.index()=Bio.SearchIO._model.query.QueryResult-class.html#index,Function Bio.SearchIO.index()=Bio.SearchIO-module.html#index,Method Bio.Seq.MutableSeq.index()=Bio.Seq.MutableSeq-class.html#index,Method Bio.Seq.Seq.index()=Bio.Seq.Seq-class.html#index,Function Bio.SeqIO.index()=Bio.SeqIO-module.html#index"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.Seq.Seq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-150', 'index', 'link-150');">index</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L299"></a><tt class="py-lineno"> 299</tt>  <tt class="py-line">            <tt class="py-comment"># Return a single letter as a string</tt> </tt>
<a name="L300"></a><tt class="py-lineno"> 300</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_data</tt><tt class="py-op">[</tt><tt id="link-151" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.Seq.Seq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-151', 'index', 'link-150');">index</a></tt><tt class="py-op">]</tt> </tt>
<a name="L301"></a><tt class="py-lineno"> 301</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L302"></a><tt class="py-lineno"> 302</tt>  <tt class="py-line">            <tt class="py-comment"># Return the (sub)sequence as another Seq object</tt> </tt>
<a name="L303"></a><tt class="py-lineno"> 303</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-152" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-152', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_data</tt><tt class="py-op">[</tt><tt id="link-153" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.Seq.Seq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-153', 'index', 'link-150');">index</a></tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-154" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-154', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L304"></a><tt class="py-lineno"> 304</tt>  <tt class="py-line"> </tt>
<a name="Seq.__add__"></a><div id="Seq.__add__-def"><a name="L305"></a><tt class="py-lineno"> 305</tt> <a class="py-toggle" href="#" id="Seq.__add__-toggle" onclick="return toggle('Seq.__add__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#__add__">__add__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.__add__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.__add__-expanded"><a name="L306"></a><tt class="py-lineno"> 306</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add another sequence or string to this sequence.</tt> </tt>
<a name="L307"></a><tt class="py-lineno"> 307</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L308"></a><tt class="py-lineno"> 308</tt>  <tt class="py-line"><tt class="py-docstring">        If adding a string to a Seq, the alphabet is preserved:</tt> </tt>
<a name="L309"></a><tt class="py-lineno"> 309</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L310"></a><tt class="py-lineno"> 310</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L311"></a><tt class="py-lineno"> 311</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_protein</tt> </tt>
<a name="L312"></a><tt class="py-lineno"> 312</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; Seq("MELKI", generic_protein) + "LV"</tt> </tt>
<a name="L313"></a><tt class="py-lineno"> 313</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('MELKILV', ProteinAlphabet())</tt> </tt>
<a name="L314"></a><tt class="py-lineno"> 314</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L315"></a><tt class="py-lineno"> 315</tt>  <tt class="py-line"><tt class="py-docstring">        When adding two Seq (like) objects, the alphabets are important.</tt> </tt>
<a name="L316"></a><tt class="py-lineno"> 316</tt>  <tt class="py-line"><tt class="py-docstring">        Consider this example:</tt> </tt>
<a name="L317"></a><tt class="py-lineno"> 317</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L318"></a><tt class="py-lineno"> 318</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L319"></a><tt class="py-lineno"> 319</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet.IUPAC import unambiguous_dna, ambiguous_dna</tt> </tt>
<a name="L320"></a><tt class="py-lineno"> 320</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; unamb_dna_seq = Seq("ACGT", unambiguous_dna)</tt> </tt>
<a name="L321"></a><tt class="py-lineno"> 321</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; ambig_dna_seq = Seq("ACRGT", ambiguous_dna)</tt> </tt>
<a name="L322"></a><tt class="py-lineno"> 322</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; unamb_dna_seq</tt> </tt>
<a name="L323"></a><tt class="py-lineno"> 323</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('ACGT', IUPACUnambiguousDNA())</tt> </tt>
<a name="L324"></a><tt class="py-lineno"> 324</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; ambig_dna_seq</tt> </tt>
<a name="L325"></a><tt class="py-lineno"> 325</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('ACRGT', IUPACAmbiguousDNA())</tt> </tt>
<a name="L326"></a><tt class="py-lineno"> 326</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L327"></a><tt class="py-lineno"> 327</tt>  <tt class="py-line"><tt class="py-docstring">        If we add the ambiguous and unambiguous IUPAC DNA alphabets, we get</tt> </tt>
<a name="L328"></a><tt class="py-lineno"> 328</tt>  <tt class="py-line"><tt class="py-docstring">        the more general ambiguous IUPAC DNA alphabet:</tt> </tt>
<a name="L329"></a><tt class="py-lineno"> 329</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L330"></a><tt class="py-lineno"> 330</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; unamb_dna_seq + ambig_dna_seq</tt> </tt>
<a name="L331"></a><tt class="py-lineno"> 331</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('ACGTACRGT', IUPACAmbiguousDNA())</tt> </tt>
<a name="L332"></a><tt class="py-lineno"> 332</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L333"></a><tt class="py-lineno"> 333</tt>  <tt class="py-line"><tt class="py-docstring">        However, if the default generic alphabet is included, the result is</tt> </tt>
<a name="L334"></a><tt class="py-lineno"> 334</tt>  <tt class="py-line"><tt class="py-docstring">        a generic alphabet:</tt> </tt>
<a name="L335"></a><tt class="py-lineno"> 335</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L336"></a><tt class="py-lineno"> 336</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; Seq("") + ambig_dna_seq</tt> </tt>
<a name="L337"></a><tt class="py-lineno"> 337</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('ACRGT')</tt> </tt>
<a name="L338"></a><tt class="py-lineno"> 338</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L339"></a><tt class="py-lineno"> 339</tt>  <tt class="py-line"><tt class="py-docstring">        You can't add RNA and DNA sequences:</tt> </tt>
<a name="L340"></a><tt class="py-lineno"> 340</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L341"></a><tt class="py-lineno"> 341</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_dna, generic_rna</tt> </tt>
<a name="L342"></a><tt class="py-lineno"> 342</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; Seq("ACGT", generic_dna) + Seq("ACGU", generic_rna)</tt> </tt>
<a name="L343"></a><tt class="py-lineno"> 343</tt>  <tt class="py-line"><tt class="py-docstring">        Traceback (most recent call last):</tt> </tt>
<a name="L344"></a><tt class="py-lineno"> 344</tt>  <tt class="py-line"><tt class="py-docstring">           ...</tt> </tt>
<a name="L345"></a><tt class="py-lineno"> 345</tt>  <tt class="py-line"><tt class="py-docstring">        TypeError: Incompatible alphabets DNAAlphabet() and RNAAlphabet()</tt> </tt>
<a name="L346"></a><tt class="py-lineno"> 346</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L347"></a><tt class="py-lineno"> 347</tt>  <tt class="py-line"><tt class="py-docstring">        You can't add nucleotide and protein sequences:</tt> </tt>
<a name="L348"></a><tt class="py-lineno"> 348</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L349"></a><tt class="py-lineno"> 349</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_dna, generic_protein</tt> </tt>
<a name="L350"></a><tt class="py-lineno"> 350</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; Seq("ACGT", generic_dna) + Seq("MELKI", generic_protein)</tt> </tt>
<a name="L351"></a><tt class="py-lineno"> 351</tt>  <tt class="py-line"><tt class="py-docstring">        Traceback (most recent call last):</tt> </tt>
<a name="L352"></a><tt class="py-lineno"> 352</tt>  <tt class="py-line"><tt class="py-docstring">           ...</tt> </tt>
<a name="L353"></a><tt class="py-lineno"> 353</tt>  <tt class="py-line"><tt class="py-docstring">        TypeError: Incompatible alphabets DNAAlphabet() and ProteinAlphabet()</tt> </tt>
<a name="L354"></a><tt class="py-lineno"> 354</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L355"></a><tt class="py-lineno"> 355</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-155" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-155', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-string">"alphabet"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L356"></a><tt class="py-lineno"> 356</tt>  <tt class="py-line">            <tt class="py-comment"># other should be a Seq or a MutableSeq</tt> </tt>
<a name="L357"></a><tt class="py-lineno"> 357</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-156" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-156', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-157" class="py-name"><a title="Bio.Alphabet._check_type_compatible" class="py-name" href="#" onclick="return doclink('link-157', '_check_type_compatible', 'link-59');">_check_type_compatible</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-158" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-158', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L358"></a><tt class="py-lineno"> 358</tt>  <tt class="py-line">                                                    <tt id="link-159" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-159', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-160" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-160', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L359"></a><tt class="py-lineno"> 359</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt> </tt>
<a name="L360"></a><tt class="py-lineno"> 360</tt>  <tt class="py-line">                    <tt class="py-string">"Incompatible alphabets {0!r} and {1!r}"</tt><tt class="py-op">.</tt><tt id="link-161" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-161', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L361"></a><tt class="py-lineno"> 361</tt>  <tt class="py-line">                        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-162" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-162', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-163" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-163', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-164" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-164', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L362"></a><tt class="py-lineno"> 362</tt>  <tt class="py-line">            <tt class="py-comment"># They should be the same sequence type (or one of them is generic)</tt> </tt>
<a name="L363"></a><tt class="py-lineno"> 363</tt>  <tt class="py-line">            <tt id="link-165" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-165', 'a', 'link-49');">a</a></tt> <tt class="py-op">=</tt> <tt id="link-166" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-166', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-167" class="py-name" targets="Function Bio.Alphabet._consensus_alphabet()=Bio.Alphabet-module.html#_consensus_alphabet"><a title="Bio.Alphabet._consensus_alphabet" class="py-name" href="#" onclick="return doclink('link-167', '_consensus_alphabet', 'link-167');">_consensus_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-168" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-168', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-169" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-169', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-170" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-170', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L364"></a><tt class="py-lineno"> 364</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-171" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-171', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-172" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-172', 'a', 'link-49');">a</a></tt><tt class="py-op">)</tt> </tt>
<a name="L365"></a><tt class="py-lineno"> 365</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-173" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-173', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L366"></a><tt class="py-lineno"> 366</tt>  <tt class="py-line">            <tt class="py-comment"># other is a plain string - use the current alphabet</tt> </tt>
<a name="L367"></a><tt class="py-lineno"> 367</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt id="link-174" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-174', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-175" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-175', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L368"></a><tt class="py-lineno"> 368</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt id="link-176" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-176', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-177" class="py-name" targets="Module Bio.SeqRecord=Bio.SeqRecord-module.html,Class Bio.SeqRecord.SeqRecord=Bio.SeqRecord.SeqRecord-class.html"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-177', 'SeqRecord', 'link-177');">SeqRecord</a></tt> <tt class="py-keyword">import</tt> <tt id="link-178" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-178', 'SeqRecord', 'link-177');">SeqRecord</a></tt>  <tt class="py-comment"># Lazy to avoid circular imports</tt> </tt>
<a name="L369"></a><tt class="py-lineno"> 369</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-179" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-179', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt id="link-180" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-180', 'SeqRecord', 'link-177');">SeqRecord</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L370"></a><tt class="py-lineno"> 370</tt>  <tt class="py-line">            <tt class="py-comment"># Get the SeqRecord's __radd__ to handle this</tt> </tt>
<a name="L371"></a><tt class="py-lineno"> 371</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">NotImplemented</tt> </tt>
<a name="L372"></a><tt class="py-lineno"> 372</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L373"></a><tt class="py-lineno"> 373</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt> </tt>
</div><a name="L374"></a><tt class="py-lineno"> 374</tt>  <tt class="py-line"> </tt>
<a name="Seq.__radd__"></a><div id="Seq.__radd__-def"><a name="L375"></a><tt class="py-lineno"> 375</tt> <a class="py-toggle" href="#" id="Seq.__radd__-toggle" onclick="return toggle('Seq.__radd__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#__radd__">__radd__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.__radd__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.__radd__-expanded"><a name="L376"></a><tt class="py-lineno"> 376</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add a sequence on the left.</tt> </tt>
<a name="L377"></a><tt class="py-lineno"> 377</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L378"></a><tt class="py-lineno"> 378</tt>  <tt class="py-line"><tt class="py-docstring">        If adding a string to a Seq, the alphabet is preserved:</tt> </tt>
<a name="L379"></a><tt class="py-lineno"> 379</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L380"></a><tt class="py-lineno"> 380</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L381"></a><tt class="py-lineno"> 381</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_protein</tt> </tt>
<a name="L382"></a><tt class="py-lineno"> 382</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; "LV" + Seq("MELKI", generic_protein)</tt> </tt>
<a name="L383"></a><tt class="py-lineno"> 383</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('LVMELKI', ProteinAlphabet())</tt> </tt>
<a name="L384"></a><tt class="py-lineno"> 384</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L385"></a><tt class="py-lineno"> 385</tt>  <tt class="py-line"><tt class="py-docstring">        Adding two Seq (like) objects is handled via the __add__ method.</tt> </tt>
<a name="L386"></a><tt class="py-lineno"> 386</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L387"></a><tt class="py-lineno"> 387</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-181" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-181', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-string">"alphabet"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L388"></a><tt class="py-lineno"> 388</tt>  <tt class="py-line">            <tt class="py-comment"># other should be a Seq or a MutableSeq</tt> </tt>
<a name="L389"></a><tt class="py-lineno"> 389</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-182" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-182', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-183" class="py-name"><a title="Bio.Alphabet._check_type_compatible" class="py-name" href="#" onclick="return doclink('link-183', '_check_type_compatible', 'link-59');">_check_type_compatible</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-184" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-184', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L390"></a><tt class="py-lineno"> 390</tt>  <tt class="py-line">                                                    <tt id="link-185" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-185', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-186" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-186', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L391"></a><tt class="py-lineno"> 391</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt> </tt>
<a name="L392"></a><tt class="py-lineno"> 392</tt>  <tt class="py-line">                    <tt class="py-string">"Incompatible alphabets {0!r} and {1!r}"</tt><tt class="py-op">.</tt><tt id="link-187" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-187', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L393"></a><tt class="py-lineno"> 393</tt>  <tt class="py-line">                        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-188" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-188', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-189" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-189', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-190" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-190', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L394"></a><tt class="py-lineno"> 394</tt>  <tt class="py-line">            <tt class="py-comment"># They should be the same sequence type (or one of them is generic)</tt> </tt>
<a name="L395"></a><tt class="py-lineno"> 395</tt>  <tt class="py-line">            <tt id="link-191" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-191', 'a', 'link-49');">a</a></tt> <tt class="py-op">=</tt> <tt id="link-192" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-192', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-193" class="py-name"><a title="Bio.Alphabet._consensus_alphabet" class="py-name" href="#" onclick="return doclink('link-193', '_consensus_alphabet', 'link-167');">_consensus_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-194" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-194', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-195" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-195', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-196" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-196', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L396"></a><tt class="py-lineno"> 396</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-197" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-197', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-198" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-198', 'a', 'link-49');">a</a></tt><tt class="py-op">)</tt> </tt>
<a name="L397"></a><tt class="py-lineno"> 397</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-199" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-199', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L398"></a><tt class="py-lineno"> 398</tt>  <tt class="py-line">            <tt class="py-comment"># other is a plain string - use the current alphabet</tt> </tt>
<a name="L399"></a><tt class="py-lineno"> 399</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt id="link-200" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-200', 'other', 'link-57');">other</a></tt> <tt class="py-op">+</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-201" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-201', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L400"></a><tt class="py-lineno"> 400</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L401"></a><tt class="py-lineno"> 401</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt> </tt>
</div><a name="L402"></a><tt class="py-lineno"> 402</tt>  <tt class="py-line"> </tt>
<a name="Seq.__mul__"></a><div id="Seq.__mul__-def"><a name="L403"></a><tt class="py-lineno"> 403</tt> <a class="py-toggle" href="#" id="Seq.__mul__-toggle" onclick="return toggle('Seq.__mul__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#__mul__">__mul__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.__mul__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.__mul__-expanded"><a name="L404"></a><tt class="py-lineno"> 404</tt>  <tt class="py-line">        <tt class="py-docstring">"""Multiply Seq by integer.</tt> </tt>
<a name="L405"></a><tt class="py-lineno"> 405</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L406"></a><tt class="py-lineno"> 406</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L407"></a><tt class="py-lineno"> 407</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_dna</tt> </tt>
<a name="L408"></a><tt class="py-lineno"> 408</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; Seq('ATG') * 2</tt> </tt>
<a name="L409"></a><tt class="py-lineno"> 409</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('ATGATG')</tt> </tt>
<a name="L410"></a><tt class="py-lineno"> 410</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; Seq('ATG', generic_dna) * 2</tt> </tt>
<a name="L411"></a><tt class="py-lineno"> 411</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('ATGATG', DNAAlphabet())</tt> </tt>
<a name="L412"></a><tt class="py-lineno"> 412</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L413"></a><tt class="py-lineno"> 413</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-202" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-202', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L414"></a><tt class="py-lineno"> 414</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"can't multiply {} by non-int type"</tt><tt class="py-op">.</tt><tt id="link-203" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-203', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L415"></a><tt class="py-lineno"> 415</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt id="link-204" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-204', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-205" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-205', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L416"></a><tt class="py-lineno"> 416</tt>  <tt class="py-line"> </tt>
<a name="Seq.__rmul__"></a><div id="Seq.__rmul__-def"><a name="L417"></a><tt class="py-lineno"> 417</tt> <a class="py-toggle" href="#" id="Seq.__rmul__-toggle" onclick="return toggle('Seq.__rmul__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#__rmul__">__rmul__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.__rmul__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.__rmul__-expanded"><a name="L418"></a><tt class="py-lineno"> 418</tt>  <tt class="py-line">        <tt class="py-docstring">"""Multiply integer by Seq.</tt> </tt>
<a name="L419"></a><tt class="py-lineno"> 419</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L420"></a><tt class="py-lineno"> 420</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L421"></a><tt class="py-lineno"> 421</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_dna</tt> </tt>
<a name="L422"></a><tt class="py-lineno"> 422</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; 2 * Seq('ATG')</tt> </tt>
<a name="L423"></a><tt class="py-lineno"> 423</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('ATGATG')</tt> </tt>
<a name="L424"></a><tt class="py-lineno"> 424</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; 2 * Seq('ATG', generic_dna)</tt> </tt>
<a name="L425"></a><tt class="py-lineno"> 425</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('ATGATG', DNAAlphabet())</tt> </tt>
<a name="L426"></a><tt class="py-lineno"> 426</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L427"></a><tt class="py-lineno"> 427</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-206" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-206', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L428"></a><tt class="py-lineno"> 428</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"can't multiply {} by non-int type"</tt><tt class="py-op">.</tt><tt id="link-207" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-207', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L429"></a><tt class="py-lineno"> 429</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt id="link-208" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-208', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-209" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-209', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L430"></a><tt class="py-lineno"> 430</tt>  <tt class="py-line"> </tt>
<a name="Seq.__imul__"></a><div id="Seq.__imul__-def"><a name="L431"></a><tt class="py-lineno"> 431</tt> <a class="py-toggle" href="#" id="Seq.__imul__-toggle" onclick="return toggle('Seq.__imul__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#__imul__">__imul__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.__imul__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.__imul__-expanded"><a name="L432"></a><tt class="py-lineno"> 432</tt>  <tt class="py-line">        <tt class="py-docstring">"""Multiply Seq in-place.</tt> </tt>
<a name="L433"></a><tt class="py-lineno"> 433</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L434"></a><tt class="py-lineno"> 434</tt>  <tt class="py-line"><tt class="py-docstring">        Note although Seq is immutable, the in-place method is</tt> </tt>
<a name="L435"></a><tt class="py-lineno"> 435</tt>  <tt class="py-line"><tt class="py-docstring">        included to match the behaviour for regular Python strings.</tt> </tt>
<a name="L436"></a><tt class="py-lineno"> 436</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L437"></a><tt class="py-lineno"> 437</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L438"></a><tt class="py-lineno"> 438</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_dna</tt> </tt>
<a name="L439"></a><tt class="py-lineno"> 439</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; seq = Seq('ATG', generic_dna)</tt> </tt>
<a name="L440"></a><tt class="py-lineno"> 440</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; seq *= 2</tt> </tt>
<a name="L441"></a><tt class="py-lineno"> 441</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; seq</tt> </tt>
<a name="L442"></a><tt class="py-lineno"> 442</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('ATGATG', DNAAlphabet())</tt> </tt>
<a name="L443"></a><tt class="py-lineno"> 443</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L444"></a><tt class="py-lineno"> 444</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-210" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-210', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L445"></a><tt class="py-lineno"> 445</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"can't multiply {} by non-int type"</tt><tt class="py-op">.</tt><tt id="link-211" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-211', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L446"></a><tt class="py-lineno"> 446</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt id="link-212" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-212', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-213" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-213', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L447"></a><tt class="py-lineno"> 447</tt>  <tt class="py-line"> </tt>
<a name="Seq.tomutable"></a><div id="Seq.tomutable-def"><a name="L448"></a><tt class="py-lineno"> 448</tt> <a class="py-toggle" href="#" id="Seq.tomutable-toggle" onclick="return toggle('Seq.tomutable');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#tomutable">tomutable</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># Needed?  Or use a function?</tt> </tt>
</div><a name="L449"></a><tt class="py-lineno"> 449</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the full sequence as a MutableSeq object.</tt> </tt>
<a name="L450"></a><tt class="py-lineno"> 450</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L451"></a><tt class="py-lineno"> 451</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L452"></a><tt class="py-lineno"> 452</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import IUPAC</tt> </tt>
<a name="L453"></a><tt class="py-lineno"> 453</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq = Seq("MKQHKAMIVALIVICITAVVAAL",</tt> </tt>
<a name="L454"></a><tt class="py-lineno"> 454</tt>  <tt class="py-line"><tt class="py-docstring">        ...              IUPAC.protein)</tt> </tt>
<a name="L455"></a><tt class="py-lineno"> 455</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L456"></a><tt class="py-lineno"> 456</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('MKQHKAMIVALIVICITAVVAAL', IUPACProtein())</tt> </tt>
<a name="L457"></a><tt class="py-lineno"> 457</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.tomutable()</tt> </tt>
<a name="L458"></a><tt class="py-lineno"> 458</tt>  <tt class="py-line"><tt class="py-docstring">        MutableSeq('MKQHKAMIVALIVICITAVVAAL', IUPACProtein())</tt> </tt>
<a name="L459"></a><tt class="py-lineno"> 459</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L460"></a><tt class="py-lineno"> 460</tt>  <tt class="py-line"><tt class="py-docstring">        Note that the alphabet is preserved.</tt> </tt>
<a name="L461"></a><tt class="py-lineno"> 461</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L462"></a><tt class="py-lineno"> 462</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-214" class="py-name"><a title="Bio.Seq.MutableSeq" class="py-name" href="#" onclick="return doclink('link-214', 'MutableSeq', 'link-83');">MutableSeq</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-215" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-215', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L463"></a><tt class="py-lineno"> 463</tt>  <tt class="py-line"> </tt>
<a name="Seq._get_seq_str_and_check_alphabet"></a><div id="Seq._get_seq_str_and_check_alphabet-def"><a name="L464"></a><tt class="py-lineno"> 464</tt> <a class="py-toggle" href="#" id="Seq._get_seq_str_and_check_alphabet-toggle" onclick="return toggle('Seq._get_seq_str_and_check_alphabet');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#_get_seq_str_and_check_alphabet">_get_seq_str_and_check_alphabet</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other_sequence</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq._get_seq_str_and_check_alphabet-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq._get_seq_str_and_check_alphabet-expanded"><a name="L465"></a><tt class="py-lineno"> 465</tt>  <tt class="py-line">        <tt class="py-docstring">"""Convert string/Seq/MutableSeq to string, checking alphabet (PRIVATE).</tt> </tt>
<a name="L466"></a><tt class="py-lineno"> 466</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L467"></a><tt class="py-lineno"> 467</tt>  <tt class="py-line"><tt class="py-docstring">        For a string argument, returns the string.</tt> </tt>
<a name="L468"></a><tt class="py-lineno"> 468</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L469"></a><tt class="py-lineno"> 469</tt>  <tt class="py-line"><tt class="py-docstring">        For a Seq or MutableSeq, it checks the alphabet is compatible</tt> </tt>
<a name="L470"></a><tt class="py-lineno"> 470</tt>  <tt class="py-line"><tt class="py-docstring">        (raising an exception if it isn't), and then returns a string.</tt> </tt>
<a name="L471"></a><tt class="py-lineno"> 471</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L472"></a><tt class="py-lineno"> 472</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L473"></a><tt class="py-lineno"> 473</tt>  <tt class="py-line">            <tt class="py-name">other_alpha</tt> <tt class="py-op">=</tt> <tt class="py-name">other_sequence</tt><tt class="py-op">.</tt><tt id="link-216" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-216', 'alphabet', 'link-44');">alphabet</a></tt> </tt>
<a name="L474"></a><tt class="py-lineno"> 474</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L475"></a><tt class="py-lineno"> 475</tt>  <tt class="py-line">            <tt class="py-comment"># Assume other_sequence is a string</tt> </tt>
<a name="L476"></a><tt class="py-lineno"> 476</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">other_sequence</tt> </tt>
<a name="L477"></a><tt class="py-lineno"> 477</tt>  <tt class="py-line"> </tt>
<a name="L478"></a><tt class="py-lineno"> 478</tt>  <tt class="py-line">        <tt class="py-comment"># Other should be a Seq or a MutableSeq</tt> </tt>
<a name="L479"></a><tt class="py-lineno"> 479</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-217" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-217', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-218" class="py-name"><a title="Bio.Alphabet._check_type_compatible" class="py-name" href="#" onclick="return doclink('link-218', '_check_type_compatible', 'link-59');">_check_type_compatible</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-219" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-219', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-name">other_alpha</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L480"></a><tt class="py-lineno"> 480</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"Incompatible alphabets {0!r} and {1!r}"</tt><tt class="py-op">.</tt><tt id="link-220" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-220', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L481"></a><tt class="py-lineno"> 481</tt>  <tt class="py-line">                            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-221" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-221', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-name">other_alpha</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L482"></a><tt class="py-lineno"> 482</tt>  <tt class="py-line">        <tt class="py-comment"># Return as a string</tt> </tt>
<a name="L483"></a><tt class="py-lineno"> 483</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">other_sequence</tt><tt class="py-op">)</tt> </tt>
</div><a name="L484"></a><tt class="py-lineno"> 484</tt>  <tt class="py-line"> </tt>
<a name="Seq.count"></a><div id="Seq.count-def"><a name="L485"></a><tt class="py-lineno"> 485</tt> <a class="py-toggle" href="#" id="Seq.count-toggle" onclick="return toggle('Seq.count');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#count">count</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sub</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">=</tt><tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">maxsize</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.count-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.count-expanded"><a name="L486"></a><tt class="py-lineno"> 486</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a non-overlapping count, like that of a python string.</tt> </tt>
<a name="L487"></a><tt class="py-lineno"> 487</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L488"></a><tt class="py-lineno"> 488</tt>  <tt class="py-line"><tt class="py-docstring">        This behaves like the python string method of the same name,</tt> </tt>
<a name="L489"></a><tt class="py-lineno"> 489</tt>  <tt class="py-line"><tt class="py-docstring">        which does a non-overlapping count!</tt> </tt>
<a name="L490"></a><tt class="py-lineno"> 490</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L491"></a><tt class="py-lineno"> 491</tt>  <tt class="py-line"><tt class="py-docstring">        For an overlapping search use the newer count_overlap() method.</tt> </tt>
<a name="L492"></a><tt class="py-lineno"> 492</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L493"></a><tt class="py-lineno"> 493</tt>  <tt class="py-line"><tt class="py-docstring">        Returns an integer, the number of occurrences of substring</tt> </tt>
<a name="L494"></a><tt class="py-lineno"> 494</tt>  <tt class="py-line"><tt class="py-docstring">        argument sub in the (sub)sequence given by [start:end].</tt> </tt>
<a name="L495"></a><tt class="py-lineno"> 495</tt>  <tt class="py-line"><tt class="py-docstring">        Optional arguments start and end are interpreted as in slice</tt> </tt>
<a name="L496"></a><tt class="py-lineno"> 496</tt>  <tt class="py-line"><tt class="py-docstring">        notation.</tt> </tt>
<a name="L497"></a><tt class="py-lineno"> 497</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L498"></a><tt class="py-lineno"> 498</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L499"></a><tt class="py-lineno"> 499</tt>  <tt class="py-line"><tt class="py-docstring">         - sub - a string or another Seq object to look for</tt> </tt>
<a name="L500"></a><tt class="py-lineno"> 500</tt>  <tt class="py-line"><tt class="py-docstring">         - start - optional integer, slice start</tt> </tt>
<a name="L501"></a><tt class="py-lineno"> 501</tt>  <tt class="py-line"><tt class="py-docstring">         - end - optional integer, slice end</tt> </tt>
<a name="L502"></a><tt class="py-lineno"> 502</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L503"></a><tt class="py-lineno"> 503</tt>  <tt class="py-line"><tt class="py-docstring">        e.g.</tt> </tt>
<a name="L504"></a><tt class="py-lineno"> 504</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L505"></a><tt class="py-lineno"> 505</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L506"></a><tt class="py-lineno"> 506</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq = Seq("AAAATGA")</tt> </tt>
<a name="L507"></a><tt class="py-lineno"> 507</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_seq.count("A"))</tt> </tt>
<a name="L508"></a><tt class="py-lineno"> 508</tt>  <tt class="py-line"><tt class="py-docstring">        5</tt> </tt>
<a name="L509"></a><tt class="py-lineno"> 509</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_seq.count("ATG"))</tt> </tt>
<a name="L510"></a><tt class="py-lineno"> 510</tt>  <tt class="py-line"><tt class="py-docstring">        1</tt> </tt>
<a name="L511"></a><tt class="py-lineno"> 511</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_seq.count(Seq("AT")))</tt> </tt>
<a name="L512"></a><tt class="py-lineno"> 512</tt>  <tt class="py-line"><tt class="py-docstring">        1</tt> </tt>
<a name="L513"></a><tt class="py-lineno"> 513</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_seq.count("AT", 2, -1))</tt> </tt>
<a name="L514"></a><tt class="py-lineno"> 514</tt>  <tt class="py-line"><tt class="py-docstring">        1</tt> </tt>
<a name="L515"></a><tt class="py-lineno"> 515</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L516"></a><tt class="py-lineno"> 516</tt>  <tt class="py-line"><tt class="py-docstring">        HOWEVER, please note because python strings and Seq objects (and</tt> </tt>
<a name="L517"></a><tt class="py-lineno"> 517</tt>  <tt class="py-line"><tt class="py-docstring">        MutableSeq objects) do a non-overlapping search, this may not give</tt> </tt>
<a name="L518"></a><tt class="py-lineno"> 518</tt>  <tt class="py-line"><tt class="py-docstring">        the answer you expect:</tt> </tt>
<a name="L519"></a><tt class="py-lineno"> 519</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L520"></a><tt class="py-lineno"> 520</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; "AAAA".count("AA")</tt> </tt>
<a name="L521"></a><tt class="py-lineno"> 521</tt>  <tt class="py-line"><tt class="py-docstring">        2</tt> </tt>
<a name="L522"></a><tt class="py-lineno"> 522</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(Seq("AAAA").count("AA"))</tt> </tt>
<a name="L523"></a><tt class="py-lineno"> 523</tt>  <tt class="py-line"><tt class="py-docstring">        2</tt> </tt>
<a name="L524"></a><tt class="py-lineno"> 524</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L525"></a><tt class="py-lineno"> 525</tt>  <tt class="py-line"><tt class="py-docstring">        An overlapping search, as implemented in .count_overlap(),</tt> </tt>
<a name="L526"></a><tt class="py-lineno"> 526</tt>  <tt class="py-line"><tt class="py-docstring">        would give the answer as three!</tt> </tt>
<a name="L527"></a><tt class="py-lineno"> 527</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L528"></a><tt class="py-lineno"> 528</tt>  <tt class="py-line">        <tt class="py-comment"># If it has one, check the alphabet:</tt> </tt>
<a name="L529"></a><tt class="py-lineno"> 529</tt>  <tt class="py-line">        <tt class="py-name">sub_str</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-222" class="py-name" targets="Method Bio.Seq.Seq._get_seq_str_and_check_alphabet()=Bio.Seq.Seq-class.html#_get_seq_str_and_check_alphabet"><a title="Bio.Seq.Seq._get_seq_str_and_check_alphabet" class="py-name" href="#" onclick="return doclink('link-222', '_get_seq_str_and_check_alphabet', 'link-222');">_get_seq_str_and_check_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">sub</tt><tt class="py-op">)</tt> </tt>
<a name="L530"></a><tt class="py-lineno"> 530</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-223" class="py-name" targets="Method Bio.Crystal.Chain.count()=Bio.Crystal.Chain-class.html#count,Method Bio.Seq.MutableSeq.count()=Bio.Seq.MutableSeq-class.html#count,Method Bio.Seq.Seq.count()=Bio.Seq.Seq-class.html#count,Method Bio.Seq.UnknownSeq.count()=Bio.Seq.UnknownSeq-class.html#count,Method Bio.motifs.Instances.count()=Bio.motifs.Instances-class.html#count"><a title="Bio.Crystal.Chain.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-223', 'count', 'link-223');">count</a></tt><tt class="py-op">(</tt><tt class="py-name">sub_str</tt><tt class="py-op">,</tt> <tt id="link-224" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.start=Bio.SeqFeature.CompoundLocation-class.html#start,Variable Bio.SeqFeature.FeatureLocation.start=Bio.SeqFeature.FeatureLocation-class.html#start"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-224', 'start', 'link-224');">start</a></tt><tt class="py-op">,</tt> <tt id="link-225" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.end=Bio.SeqFeature.CompoundLocation-class.html#end,Variable Bio.SeqFeature.FeatureLocation.end=Bio.SeqFeature.FeatureLocation-class.html#end"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-225', 'end', 'link-225');">end</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L531"></a><tt class="py-lineno"> 531</tt>  <tt class="py-line"> </tt>
<a name="Seq.count_overlap"></a><div id="Seq.count_overlap-def"><a name="L532"></a><tt class="py-lineno"> 532</tt> <a class="py-toggle" href="#" id="Seq.count_overlap-toggle" onclick="return toggle('Seq.count_overlap');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#count_overlap">count_overlap</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sub</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">=</tt><tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">maxsize</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.count_overlap-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.count_overlap-expanded"><a name="L533"></a><tt class="py-lineno"> 533</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return an overlapping count.</tt> </tt>
<a name="L534"></a><tt class="py-lineno"> 534</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L535"></a><tt class="py-lineno"> 535</tt>  <tt class="py-line"><tt class="py-docstring">        For a non-overlapping search use the count() method.</tt> </tt>
<a name="L536"></a><tt class="py-lineno"> 536</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L537"></a><tt class="py-lineno"> 537</tt>  <tt class="py-line"><tt class="py-docstring">        Returns an integer, the number of occurrences of substring</tt> </tt>
<a name="L538"></a><tt class="py-lineno"> 538</tt>  <tt class="py-line"><tt class="py-docstring">        argument sub in the (sub)sequence given by [start:end].</tt> </tt>
<a name="L539"></a><tt class="py-lineno"> 539</tt>  <tt class="py-line"><tt class="py-docstring">        Optional arguments start and end are interpreted as in slice</tt> </tt>
<a name="L540"></a><tt class="py-lineno"> 540</tt>  <tt class="py-line"><tt class="py-docstring">        notation.</tt> </tt>
<a name="L541"></a><tt class="py-lineno"> 541</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L542"></a><tt class="py-lineno"> 542</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L543"></a><tt class="py-lineno"> 543</tt>  <tt class="py-line"><tt class="py-docstring">         - sub - a string or another Seq object to look for</tt> </tt>
<a name="L544"></a><tt class="py-lineno"> 544</tt>  <tt class="py-line"><tt class="py-docstring">         - start - optional integer, slice start</tt> </tt>
<a name="L545"></a><tt class="py-lineno"> 545</tt>  <tt class="py-line"><tt class="py-docstring">         - end - optional integer, slice end</tt> </tt>
<a name="L546"></a><tt class="py-lineno"> 546</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L547"></a><tt class="py-lineno"> 547</tt>  <tt class="py-line"><tt class="py-docstring">        e.g.</tt> </tt>
<a name="L548"></a><tt class="py-lineno"> 548</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L549"></a><tt class="py-lineno"> 549</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L550"></a><tt class="py-lineno"> 550</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(Seq("AAAA").count_overlap("AA"))</tt> </tt>
<a name="L551"></a><tt class="py-lineno"> 551</tt>  <tt class="py-line"><tt class="py-docstring">        3</tt> </tt>
<a name="L552"></a><tt class="py-lineno"> 552</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(Seq("ATATATATA").count_overlap("ATA"))</tt> </tt>
<a name="L553"></a><tt class="py-lineno"> 553</tt>  <tt class="py-line"><tt class="py-docstring">        4</tt> </tt>
<a name="L554"></a><tt class="py-lineno"> 554</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(Seq("ATATATATA").count_overlap("ATA", 3, -1))</tt> </tt>
<a name="L555"></a><tt class="py-lineno"> 555</tt>  <tt class="py-line"><tt class="py-docstring">        1</tt> </tt>
<a name="L556"></a><tt class="py-lineno"> 556</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L557"></a><tt class="py-lineno"> 557</tt>  <tt class="py-line"><tt class="py-docstring">        Where substrings do not overlap, should behave the same as</tt> </tt>
<a name="L558"></a><tt class="py-lineno"> 558</tt>  <tt class="py-line"><tt class="py-docstring">        the count() method:</tt> </tt>
<a name="L559"></a><tt class="py-lineno"> 559</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L560"></a><tt class="py-lineno"> 560</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L561"></a><tt class="py-lineno"> 561</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq = Seq("AAAATGA")</tt> </tt>
<a name="L562"></a><tt class="py-lineno"> 562</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_seq.count_overlap("A"))</tt> </tt>
<a name="L563"></a><tt class="py-lineno"> 563</tt>  <tt class="py-line"><tt class="py-docstring">        5</tt> </tt>
<a name="L564"></a><tt class="py-lineno"> 564</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.count_overlap("A") == my_seq.count("A")</tt> </tt>
<a name="L565"></a><tt class="py-lineno"> 565</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L566"></a><tt class="py-lineno"> 566</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_seq.count_overlap("ATG"))</tt> </tt>
<a name="L567"></a><tt class="py-lineno"> 567</tt>  <tt class="py-line"><tt class="py-docstring">        1</tt> </tt>
<a name="L568"></a><tt class="py-lineno"> 568</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.count_overlap("ATG") == my_seq.count("ATG")</tt> </tt>
<a name="L569"></a><tt class="py-lineno"> 569</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L570"></a><tt class="py-lineno"> 570</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_seq.count_overlap(Seq("AT")))</tt> </tt>
<a name="L571"></a><tt class="py-lineno"> 571</tt>  <tt class="py-line"><tt class="py-docstring">        1</tt> </tt>
<a name="L572"></a><tt class="py-lineno"> 572</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.count_overlap(Seq("AT")) == my_seq.count(Seq("AT"))</tt> </tt>
<a name="L573"></a><tt class="py-lineno"> 573</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L574"></a><tt class="py-lineno"> 574</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_seq.count_overlap("AT", 2, -1))</tt> </tt>
<a name="L575"></a><tt class="py-lineno"> 575</tt>  <tt class="py-line"><tt class="py-docstring">        1</tt> </tt>
<a name="L576"></a><tt class="py-lineno"> 576</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.count_overlap("AT", 2, -1) == my_seq.count("AT", 2, -1)</tt> </tt>
<a name="L577"></a><tt class="py-lineno"> 577</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L578"></a><tt class="py-lineno"> 578</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L579"></a><tt class="py-lineno"> 579</tt>  <tt class="py-line"><tt class="py-docstring">        HOWEVER, do not use this method for such cases because the</tt> </tt>
<a name="L580"></a><tt class="py-lineno"> 580</tt>  <tt class="py-line"><tt class="py-docstring">        count() method is much for efficient.</tt> </tt>
<a name="L581"></a><tt class="py-lineno"> 581</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L582"></a><tt class="py-lineno"> 582</tt>  <tt class="py-line">        <tt class="py-name">sub_str</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-226" class="py-name"><a title="Bio.Seq.Seq._get_seq_str_and_check_alphabet" class="py-name" href="#" onclick="return doclink('link-226', '_get_seq_str_and_check_alphabet', 'link-222');">_get_seq_str_and_check_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">sub</tt><tt class="py-op">)</tt> </tt>
<a name="L583"></a><tt class="py-lineno"> 583</tt>  <tt class="py-line">        <tt class="py-name">self_str</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
<a name="L584"></a><tt class="py-lineno"> 584</tt>  <tt class="py-line">        <tt class="py-name">overlap_count</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L585"></a><tt class="py-lineno"> 585</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L586"></a><tt class="py-lineno"> 586</tt>  <tt class="py-line">            <tt id="link-227" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-227', 'start', 'link-224');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self_str</tt><tt class="py-op">.</tt><tt id="link-228" class="py-name" targets="Method Bio.Seq.Seq.find()=Bio.Seq.Seq-class.html#find,Function Bio.triefind.find()=Bio.triefind-module.html#find"><a title="Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-228', 'find', 'link-228');">find</a></tt><tt class="py-op">(</tt><tt class="py-name">sub_str</tt><tt class="py-op">,</tt> <tt id="link-229" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-229', 'start', 'link-224');">start</a></tt><tt class="py-op">,</tt> <tt id="link-230" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-230', 'end', 'link-225');">end</a></tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt> </tt>
<a name="L587"></a><tt class="py-lineno"> 587</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-231" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-231', 'start', 'link-224');">start</a></tt> <tt class="py-op">!=</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L588"></a><tt class="py-lineno"> 588</tt>  <tt class="py-line">                <tt class="py-name">overlap_count</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L589"></a><tt class="py-lineno"> 589</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L590"></a><tt class="py-lineno"> 590</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">overlap_count</tt> </tt>
</div><a name="L591"></a><tt class="py-lineno"> 591</tt>  <tt class="py-line"> </tt>
<a name="Seq.__contains__"></a><div id="Seq.__contains__-def"><a name="L592"></a><tt class="py-lineno"> 592</tt> <a class="py-toggle" href="#" id="Seq.__contains__-toggle" onclick="return toggle('Seq.__contains__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#__contains__">__contains__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">char</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.__contains__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.__contains__-expanded"><a name="L593"></a><tt class="py-lineno"> 593</tt>  <tt class="py-line">        <tt class="py-docstring">"""Implement the 'in' keyword, like a python string.</tt> </tt>
<a name="L594"></a><tt class="py-lineno"> 594</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L595"></a><tt class="py-lineno"> 595</tt>  <tt class="py-line"><tt class="py-docstring">        e.g.</tt> </tt>
<a name="L596"></a><tt class="py-lineno"> 596</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L597"></a><tt class="py-lineno"> 597</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L598"></a><tt class="py-lineno"> 598</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_dna, generic_rna, generic_protein</tt> </tt>
<a name="L599"></a><tt class="py-lineno"> 599</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna = Seq("ATATGAAATTTGAAAA", generic_dna)</tt> </tt>
<a name="L600"></a><tt class="py-lineno"> 600</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; "AAA" in my_dna</tt> </tt>
<a name="L601"></a><tt class="py-lineno"> 601</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L602"></a><tt class="py-lineno"> 602</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; Seq("AAA") in my_dna</tt> </tt>
<a name="L603"></a><tt class="py-lineno"> 603</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L604"></a><tt class="py-lineno"> 604</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; Seq("AAA", generic_dna) in my_dna</tt> </tt>
<a name="L605"></a><tt class="py-lineno"> 605</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L606"></a><tt class="py-lineno"> 606</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L607"></a><tt class="py-lineno"> 607</tt>  <tt class="py-line"><tt class="py-docstring">        Like other Seq methods, this will raise a type error if another Seq</tt> </tt>
<a name="L608"></a><tt class="py-lineno"> 608</tt>  <tt class="py-line"><tt class="py-docstring">        (or Seq like) object with an incompatible alphabet is used:</tt> </tt>
<a name="L609"></a><tt class="py-lineno"> 609</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L610"></a><tt class="py-lineno"> 610</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; Seq("AAA", generic_rna) in my_dna</tt> </tt>
<a name="L611"></a><tt class="py-lineno"> 611</tt>  <tt class="py-line"><tt class="py-docstring">        Traceback (most recent call last):</tt> </tt>
<a name="L612"></a><tt class="py-lineno"> 612</tt>  <tt class="py-line"><tt class="py-docstring">           ...</tt> </tt>
<a name="L613"></a><tt class="py-lineno"> 613</tt>  <tt class="py-line"><tt class="py-docstring">        TypeError: Incompatible alphabets DNAAlphabet() and RNAAlphabet()</tt> </tt>
<a name="L614"></a><tt class="py-lineno"> 614</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; Seq("AAA", generic_protein) in my_dna</tt> </tt>
<a name="L615"></a><tt class="py-lineno"> 615</tt>  <tt class="py-line"><tt class="py-docstring">        Traceback (most recent call last):</tt> </tt>
<a name="L616"></a><tt class="py-lineno"> 616</tt>  <tt class="py-line"><tt class="py-docstring">           ...</tt> </tt>
<a name="L617"></a><tt class="py-lineno"> 617</tt>  <tt class="py-line"><tt class="py-docstring">        TypeError: Incompatible alphabets DNAAlphabet() and ProteinAlphabet()</tt> </tt>
<a name="L618"></a><tt class="py-lineno"> 618</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L619"></a><tt class="py-lineno"> 619</tt>  <tt class="py-line">        <tt class="py-comment"># If it has one, check the alphabet:</tt> </tt>
<a name="L620"></a><tt class="py-lineno"> 620</tt>  <tt class="py-line">        <tt class="py-name">sub_str</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-232" class="py-name"><a title="Bio.Seq.Seq._get_seq_str_and_check_alphabet" class="py-name" href="#" onclick="return doclink('link-232', '_get_seq_str_and_check_alphabet', 'link-222');">_get_seq_str_and_check_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">char</tt><tt class="py-op">)</tt> </tt>
<a name="L621"></a><tt class="py-lineno"> 621</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">sub_str</tt> <tt class="py-keyword">in</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
</div><a name="L622"></a><tt class="py-lineno"> 622</tt>  <tt class="py-line"> </tt>
<a name="Seq.find"></a><div id="Seq.find-def"><a name="L623"></a><tt class="py-lineno"> 623</tt> <a class="py-toggle" href="#" id="Seq.find-toggle" onclick="return toggle('Seq.find');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#find">find</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sub</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">=</tt><tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">maxsize</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.find-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.find-expanded"><a name="L624"></a><tt class="py-lineno"> 624</tt>  <tt class="py-line">        <tt class="py-docstring">"""Find method, like that of a python string.</tt> </tt>
<a name="L625"></a><tt class="py-lineno"> 625</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L626"></a><tt class="py-lineno"> 626</tt>  <tt class="py-line"><tt class="py-docstring">        This behaves like the python string method of the same name.</tt> </tt>
<a name="L627"></a><tt class="py-lineno"> 627</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L628"></a><tt class="py-lineno"> 628</tt>  <tt class="py-line"><tt class="py-docstring">        Returns an integer, the index of the first occurrence of substring</tt> </tt>
<a name="L629"></a><tt class="py-lineno"> 629</tt>  <tt class="py-line"><tt class="py-docstring">        argument sub in the (sub)sequence given by [start:end].</tt> </tt>
<a name="L630"></a><tt class="py-lineno"> 630</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L631"></a><tt class="py-lineno"> 631</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L632"></a><tt class="py-lineno"> 632</tt>  <tt class="py-line"><tt class="py-docstring">         - sub - a string or another Seq object to look for</tt> </tt>
<a name="L633"></a><tt class="py-lineno"> 633</tt>  <tt class="py-line"><tt class="py-docstring">         - start - optional integer, slice start</tt> </tt>
<a name="L634"></a><tt class="py-lineno"> 634</tt>  <tt class="py-line"><tt class="py-docstring">         - end - optional integer, slice end</tt> </tt>
<a name="L635"></a><tt class="py-lineno"> 635</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L636"></a><tt class="py-lineno"> 636</tt>  <tt class="py-line"><tt class="py-docstring">        Returns -1 if the subsequence is NOT found.</tt> </tt>
<a name="L637"></a><tt class="py-lineno"> 637</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L638"></a><tt class="py-lineno"> 638</tt>  <tt class="py-line"><tt class="py-docstring">        e.g. Locating the first typical start codon, AUG, in an RNA sequence:</tt> </tt>
<a name="L639"></a><tt class="py-lineno"> 639</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L640"></a><tt class="py-lineno"> 640</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L641"></a><tt class="py-lineno"> 641</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_rna = Seq("GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG")</tt> </tt>
<a name="L642"></a><tt class="py-lineno"> 642</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_rna.find("AUG")</tt> </tt>
<a name="L643"></a><tt class="py-lineno"> 643</tt>  <tt class="py-line"><tt class="py-docstring">        3</tt> </tt>
<a name="L644"></a><tt class="py-lineno"> 644</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L645"></a><tt class="py-lineno"> 645</tt>  <tt class="py-line">        <tt class="py-comment"># If it has one, check the alphabet:</tt> </tt>
<a name="L646"></a><tt class="py-lineno"> 646</tt>  <tt class="py-line">        <tt class="py-name">sub_str</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-233" class="py-name"><a title="Bio.Seq.Seq._get_seq_str_and_check_alphabet" class="py-name" href="#" onclick="return doclink('link-233', '_get_seq_str_and_check_alphabet', 'link-222');">_get_seq_str_and_check_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">sub</tt><tt class="py-op">)</tt> </tt>
<a name="L647"></a><tt class="py-lineno"> 647</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-234" class="py-name"><a title="Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-234', 'find', 'link-228');">find</a></tt><tt class="py-op">(</tt><tt class="py-name">sub_str</tt><tt class="py-op">,</tt> <tt id="link-235" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-235', 'start', 'link-224');">start</a></tt><tt class="py-op">,</tt> <tt id="link-236" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-236', 'end', 'link-225');">end</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L648"></a><tt class="py-lineno"> 648</tt>  <tt class="py-line"> </tt>
<a name="Seq.rfind"></a><div id="Seq.rfind-def"><a name="L649"></a><tt class="py-lineno"> 649</tt> <a class="py-toggle" href="#" id="Seq.rfind-toggle" onclick="return toggle('Seq.rfind');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#rfind">rfind</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sub</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">=</tt><tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">maxsize</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.rfind-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.rfind-expanded"><a name="L650"></a><tt class="py-lineno"> 650</tt>  <tt class="py-line">        <tt class="py-docstring">"""Find from right method, like that of a python string.</tt> </tt>
<a name="L651"></a><tt class="py-lineno"> 651</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L652"></a><tt class="py-lineno"> 652</tt>  <tt class="py-line"><tt class="py-docstring">        This behaves like the python string method of the same name.</tt> </tt>
<a name="L653"></a><tt class="py-lineno"> 653</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L654"></a><tt class="py-lineno"> 654</tt>  <tt class="py-line"><tt class="py-docstring">        Returns an integer, the index of the last (right most) occurrence of</tt> </tt>
<a name="L655"></a><tt class="py-lineno"> 655</tt>  <tt class="py-line"><tt class="py-docstring">        substring argument sub in the (sub)sequence given by [start:end].</tt> </tt>
<a name="L656"></a><tt class="py-lineno"> 656</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L657"></a><tt class="py-lineno"> 657</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L658"></a><tt class="py-lineno"> 658</tt>  <tt class="py-line"><tt class="py-docstring">         - sub - a string or another Seq object to look for</tt> </tt>
<a name="L659"></a><tt class="py-lineno"> 659</tt>  <tt class="py-line"><tt class="py-docstring">         - start - optional integer, slice start</tt> </tt>
<a name="L660"></a><tt class="py-lineno"> 660</tt>  <tt class="py-line"><tt class="py-docstring">         - end - optional integer, slice end</tt> </tt>
<a name="L661"></a><tt class="py-lineno"> 661</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L662"></a><tt class="py-lineno"> 662</tt>  <tt class="py-line"><tt class="py-docstring">        Returns -1 if the subsequence is NOT found.</tt> </tt>
<a name="L663"></a><tt class="py-lineno"> 663</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L664"></a><tt class="py-lineno"> 664</tt>  <tt class="py-line"><tt class="py-docstring">        e.g. Locating the last typical start codon, AUG, in an RNA sequence:</tt> </tt>
<a name="L665"></a><tt class="py-lineno"> 665</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L666"></a><tt class="py-lineno"> 666</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L667"></a><tt class="py-lineno"> 667</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_rna = Seq("GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG")</tt> </tt>
<a name="L668"></a><tt class="py-lineno"> 668</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_rna.rfind("AUG")</tt> </tt>
<a name="L669"></a><tt class="py-lineno"> 669</tt>  <tt class="py-line"><tt class="py-docstring">        15</tt> </tt>
<a name="L670"></a><tt class="py-lineno"> 670</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L671"></a><tt class="py-lineno"> 671</tt>  <tt class="py-line">        <tt class="py-comment"># If it has one, check the alphabet:</tt> </tt>
<a name="L672"></a><tt class="py-lineno"> 672</tt>  <tt class="py-line">        <tt class="py-name">sub_str</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-237" class="py-name"><a title="Bio.Seq.Seq._get_seq_str_and_check_alphabet" class="py-name" href="#" onclick="return doclink('link-237', '_get_seq_str_and_check_alphabet', 'link-222');">_get_seq_str_and_check_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">sub</tt><tt class="py-op">)</tt> </tt>
<a name="L673"></a><tt class="py-lineno"> 673</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-238" class="py-name" targets="Method Bio.Seq.Seq.rfind()=Bio.Seq.Seq-class.html#rfind"><a title="Bio.Seq.Seq.rfind" class="py-name" href="#" onclick="return doclink('link-238', 'rfind', 'link-238');">rfind</a></tt><tt class="py-op">(</tt><tt class="py-name">sub_str</tt><tt class="py-op">,</tt> <tt id="link-239" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-239', 'start', 'link-224');">start</a></tt><tt class="py-op">,</tt> <tt id="link-240" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-240', 'end', 'link-225');">end</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L674"></a><tt class="py-lineno"> 674</tt>  <tt class="py-line"> </tt>
<a name="Seq.index"></a><div id="Seq.index-def"><a name="L675"></a><tt class="py-lineno"> 675</tt> <a class="py-toggle" href="#" id="Seq.index-toggle" onclick="return toggle('Seq.index');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#index">index</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sub</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">=</tt><tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">maxsize</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.index-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.index-expanded"><a name="L676"></a><tt class="py-lineno"> 676</tt>  <tt class="py-line">        <tt class="py-docstring">"""Like find() but raise ValueError when the substring is not found.</tt> </tt>
<a name="L677"></a><tt class="py-lineno"> 677</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L678"></a><tt class="py-lineno"> 678</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L679"></a><tt class="py-lineno"> 679</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_rna = Seq("GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG")</tt> </tt>
<a name="L680"></a><tt class="py-lineno"> 680</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_rna.find("T")</tt> </tt>
<a name="L681"></a><tt class="py-lineno"> 681</tt>  <tt class="py-line"><tt class="py-docstring">        -1</tt> </tt>
<a name="L682"></a><tt class="py-lineno"> 682</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_rna.index("T")</tt> </tt>
<a name="L683"></a><tt class="py-lineno"> 683</tt>  <tt class="py-line"><tt class="py-docstring">        Traceback (most recent call last):</tt> </tt>
<a name="L684"></a><tt class="py-lineno"> 684</tt>  <tt class="py-line"><tt class="py-docstring">                   ...</tt> </tt>
<a name="L685"></a><tt class="py-lineno"> 685</tt>  <tt class="py-line"><tt class="py-docstring">        ValueError: substring not found...</tt> </tt>
<a name="L686"></a><tt class="py-lineno"> 686</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L687"></a><tt class="py-lineno"> 687</tt>  <tt class="py-line">        <tt class="py-comment"># If it has one, check the alphabet:</tt> </tt>
<a name="L688"></a><tt class="py-lineno"> 688</tt>  <tt class="py-line">        <tt class="py-name">sub_str</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-241" class="py-name"><a title="Bio.Seq.Seq._get_seq_str_and_check_alphabet" class="py-name" href="#" onclick="return doclink('link-241', '_get_seq_str_and_check_alphabet', 'link-222');">_get_seq_str_and_check_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">sub</tt><tt class="py-op">)</tt> </tt>
<a name="L689"></a><tt class="py-lineno"> 689</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-242" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.Seq.Seq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-242', 'index', 'link-150');">index</a></tt><tt class="py-op">(</tt><tt class="py-name">sub_str</tt><tt class="py-op">,</tt> <tt id="link-243" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-243', 'start', 'link-224');">start</a></tt><tt class="py-op">,</tt> <tt id="link-244" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-244', 'end', 'link-225');">end</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L690"></a><tt class="py-lineno"> 690</tt>  <tt class="py-line"> </tt>
<a name="Seq.rindex"></a><div id="Seq.rindex-def"><a name="L691"></a><tt class="py-lineno"> 691</tt> <a class="py-toggle" href="#" id="Seq.rindex-toggle" onclick="return toggle('Seq.rindex');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#rindex">rindex</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sub</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">=</tt><tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">maxsize</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.rindex-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.rindex-expanded"><a name="L692"></a><tt class="py-lineno"> 692</tt>  <tt class="py-line">        <tt class="py-docstring">"""Like rfind() but raise ValueError when the substring is not found."""</tt> </tt>
<a name="L693"></a><tt class="py-lineno"> 693</tt>  <tt class="py-line">        <tt class="py-comment"># If it has one, check the alphabet:</tt> </tt>
<a name="L694"></a><tt class="py-lineno"> 694</tt>  <tt class="py-line">        <tt class="py-name">sub_str</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-245" class="py-name"><a title="Bio.Seq.Seq._get_seq_str_and_check_alphabet" class="py-name" href="#" onclick="return doclink('link-245', '_get_seq_str_and_check_alphabet', 'link-222');">_get_seq_str_and_check_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">sub</tt><tt class="py-op">)</tt> </tt>
<a name="L695"></a><tt class="py-lineno"> 695</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-246" class="py-name" targets="Method Bio.Seq.Seq.rindex()=Bio.Seq.Seq-class.html#rindex"><a title="Bio.Seq.Seq.rindex" class="py-name" href="#" onclick="return doclink('link-246', 'rindex', 'link-246');">rindex</a></tt><tt class="py-op">(</tt><tt class="py-name">sub_str</tt><tt class="py-op">,</tt> <tt id="link-247" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-247', 'start', 'link-224');">start</a></tt><tt class="py-op">,</tt> <tt id="link-248" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-248', 'end', 'link-225');">end</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L696"></a><tt class="py-lineno"> 696</tt>  <tt class="py-line"> </tt>
<a name="Seq.startswith"></a><div id="Seq.startswith-def"><a name="L697"></a><tt class="py-lineno"> 697</tt> <a class="py-toggle" href="#" id="Seq.startswith-toggle" onclick="return toggle('Seq.startswith');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#startswith">startswith</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">prefix</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">=</tt><tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">maxsize</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.startswith-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.startswith-expanded"><a name="L698"></a><tt class="py-lineno"> 698</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return True if the Seq starts with the given prefix, False otherwise.</tt> </tt>
<a name="L699"></a><tt class="py-lineno"> 699</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L700"></a><tt class="py-lineno"> 700</tt>  <tt class="py-line"><tt class="py-docstring">        This behaves like the python string method of the same name.</tt> </tt>
<a name="L701"></a><tt class="py-lineno"> 701</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L702"></a><tt class="py-lineno"> 702</tt>  <tt class="py-line"><tt class="py-docstring">        Return True if the sequence starts with the specified prefix</tt> </tt>
<a name="L703"></a><tt class="py-lineno"> 703</tt>  <tt class="py-line"><tt class="py-docstring">        (a string or another Seq object), False otherwise.</tt> </tt>
<a name="L704"></a><tt class="py-lineno"> 704</tt>  <tt class="py-line"><tt class="py-docstring">        With optional start, test sequence beginning at that position.</tt> </tt>
<a name="L705"></a><tt class="py-lineno"> 705</tt>  <tt class="py-line"><tt class="py-docstring">        With optional end, stop comparing sequence at that position.</tt> </tt>
<a name="L706"></a><tt class="py-lineno"> 706</tt>  <tt class="py-line"><tt class="py-docstring">        prefix can also be a tuple of strings to try.  e.g.</tt> </tt>
<a name="L707"></a><tt class="py-lineno"> 707</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L708"></a><tt class="py-lineno"> 708</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L709"></a><tt class="py-lineno"> 709</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_rna = Seq("GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG")</tt> </tt>
<a name="L710"></a><tt class="py-lineno"> 710</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_rna.startswith("GUC")</tt> </tt>
<a name="L711"></a><tt class="py-lineno"> 711</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L712"></a><tt class="py-lineno"> 712</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_rna.startswith("AUG")</tt> </tt>
<a name="L713"></a><tt class="py-lineno"> 713</tt>  <tt class="py-line"><tt class="py-docstring">        False</tt> </tt>
<a name="L714"></a><tt class="py-lineno"> 714</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_rna.startswith("AUG", 3)</tt> </tt>
<a name="L715"></a><tt class="py-lineno"> 715</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L716"></a><tt class="py-lineno"> 716</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_rna.startswith(("UCC", "UCA", "UCG"), 1)</tt> </tt>
<a name="L717"></a><tt class="py-lineno"> 717</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L718"></a><tt class="py-lineno"> 718</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L719"></a><tt class="py-lineno"> 719</tt>  <tt class="py-line">        <tt class="py-comment"># If it has one, check the alphabet:</tt> </tt>
<a name="L720"></a><tt class="py-lineno"> 720</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-249" class="py-name" targets="Variable Bio.Phylo.NeXMLIO.prefix=Bio.Phylo.NeXMLIO-module.html#prefix,Variable Bio.Phylo.PhyloXMLIO.prefix=Bio.Phylo.PhyloXMLIO-module.html#prefix"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-249', 'prefix', 'link-249');">prefix</a></tt><tt class="py-op">,</tt> <tt class="py-name">tuple</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L721"></a><tt class="py-lineno"> 721</tt>  <tt class="py-line">            <tt class="py-name">prefix_strs</tt> <tt class="py-op">=</tt> <tt class="py-name">tuple</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-250" class="py-name"><a title="Bio.Seq.Seq._get_seq_str_and_check_alphabet" class="py-name" href="#" onclick="return doclink('link-250', '_get_seq_str_and_check_alphabet', 'link-222');">_get_seq_str_and_check_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">p</tt><tt class="py-op">)</tt> </tt>
<a name="L722"></a><tt class="py-lineno"> 722</tt>  <tt class="py-line">                                <tt class="py-keyword">for</tt> <tt class="py-name">p</tt> <tt class="py-keyword">in</tt> <tt id="link-251" class="py-name"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-251', 'prefix', 'link-249');">prefix</a></tt><tt class="py-op">)</tt> </tt>
<a name="L723"></a><tt class="py-lineno"> 723</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-252" class="py-name" targets="Method Bio.Seq.Seq.startswith()=Bio.Seq.Seq-class.html#startswith"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-252', 'startswith', 'link-252');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-name">prefix_strs</tt><tt class="py-op">,</tt> <tt id="link-253" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-253', 'start', 'link-224');">start</a></tt><tt class="py-op">,</tt> <tt id="link-254" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-254', 'end', 'link-225');">end</a></tt><tt class="py-op">)</tt> </tt>
<a name="L724"></a><tt class="py-lineno"> 724</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L725"></a><tt class="py-lineno"> 725</tt>  <tt class="py-line">            <tt class="py-name">prefix_str</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-255" class="py-name"><a title="Bio.Seq.Seq._get_seq_str_and_check_alphabet" class="py-name" href="#" onclick="return doclink('link-255', '_get_seq_str_and_check_alphabet', 'link-222');">_get_seq_str_and_check_alphabet</a></tt><tt class="py-op">(</tt><tt id="link-256" class="py-name"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-256', 'prefix', 'link-249');">prefix</a></tt><tt class="py-op">)</tt> </tt>
<a name="L726"></a><tt class="py-lineno"> 726</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-257" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-257', 'startswith', 'link-252');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-name">prefix_str</tt><tt class="py-op">,</tt> <tt id="link-258" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-258', 'start', 'link-224');">start</a></tt><tt class="py-op">,</tt> <tt id="link-259" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-259', 'end', 'link-225');">end</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L727"></a><tt class="py-lineno"> 727</tt>  <tt class="py-line"> </tt>
<a name="Seq.endswith"></a><div id="Seq.endswith-def"><a name="L728"></a><tt class="py-lineno"> 728</tt> <a class="py-toggle" href="#" id="Seq.endswith-toggle" onclick="return toggle('Seq.endswith');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#endswith">endswith</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">suffix</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">=</tt><tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">maxsize</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.endswith-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.endswith-expanded"><a name="L729"></a><tt class="py-lineno"> 729</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return True if the Seq ends with the given suffix, False otherwise.</tt> </tt>
<a name="L730"></a><tt class="py-lineno"> 730</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L731"></a><tt class="py-lineno"> 731</tt>  <tt class="py-line"><tt class="py-docstring">        This behaves like the python string method of the same name.</tt> </tt>
<a name="L732"></a><tt class="py-lineno"> 732</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L733"></a><tt class="py-lineno"> 733</tt>  <tt class="py-line"><tt class="py-docstring">        Return True if the sequence ends with the specified suffix</tt> </tt>
<a name="L734"></a><tt class="py-lineno"> 734</tt>  <tt class="py-line"><tt class="py-docstring">        (a string or another Seq object), False otherwise.</tt> </tt>
<a name="L735"></a><tt class="py-lineno"> 735</tt>  <tt class="py-line"><tt class="py-docstring">        With optional start, test sequence beginning at that position.</tt> </tt>
<a name="L736"></a><tt class="py-lineno"> 736</tt>  <tt class="py-line"><tt class="py-docstring">        With optional end, stop comparing sequence at that position.</tt> </tt>
<a name="L737"></a><tt class="py-lineno"> 737</tt>  <tt class="py-line"><tt class="py-docstring">        suffix can also be a tuple of strings to try.  e.g.</tt> </tt>
<a name="L738"></a><tt class="py-lineno"> 738</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L739"></a><tt class="py-lineno"> 739</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L740"></a><tt class="py-lineno"> 740</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_rna = Seq("GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG")</tt> </tt>
<a name="L741"></a><tt class="py-lineno"> 741</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_rna.endswith("UUG")</tt> </tt>
<a name="L742"></a><tt class="py-lineno"> 742</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L743"></a><tt class="py-lineno"> 743</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_rna.endswith("AUG")</tt> </tt>
<a name="L744"></a><tt class="py-lineno"> 744</tt>  <tt class="py-line"><tt class="py-docstring">        False</tt> </tt>
<a name="L745"></a><tt class="py-lineno"> 745</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_rna.endswith("AUG", 0, 18)</tt> </tt>
<a name="L746"></a><tt class="py-lineno"> 746</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L747"></a><tt class="py-lineno"> 747</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_rna.endswith(("UCC", "UCA", "UUG"))</tt> </tt>
<a name="L748"></a><tt class="py-lineno"> 748</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L749"></a><tt class="py-lineno"> 749</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L750"></a><tt class="py-lineno"> 750</tt>  <tt class="py-line">        <tt class="py-comment"># If it has one, check the alphabet:</tt> </tt>
<a name="L751"></a><tt class="py-lineno"> 751</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">suffix</tt><tt class="py-op">,</tt> <tt class="py-name">tuple</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L752"></a><tt class="py-lineno"> 752</tt>  <tt class="py-line">            <tt class="py-name">suffix_strs</tt> <tt class="py-op">=</tt> <tt class="py-name">tuple</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-260" class="py-name"><a title="Bio.Seq.Seq._get_seq_str_and_check_alphabet" class="py-name" href="#" onclick="return doclink('link-260', '_get_seq_str_and_check_alphabet', 'link-222');">_get_seq_str_and_check_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">p</tt><tt class="py-op">)</tt> </tt>
<a name="L753"></a><tt class="py-lineno"> 753</tt>  <tt class="py-line">                                <tt class="py-keyword">for</tt> <tt class="py-name">p</tt> <tt class="py-keyword">in</tt> <tt class="py-name">suffix</tt><tt class="py-op">)</tt> </tt>
<a name="L754"></a><tt class="py-lineno"> 754</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-261" class="py-name" targets="Method Bio.Seq.Seq.endswith()=Bio.Seq.Seq-class.html#endswith"><a title="Bio.Seq.Seq.endswith" class="py-name" href="#" onclick="return doclink('link-261', 'endswith', 'link-261');">endswith</a></tt><tt class="py-op">(</tt><tt class="py-name">suffix_strs</tt><tt class="py-op">,</tt> <tt id="link-262" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-262', 'start', 'link-224');">start</a></tt><tt class="py-op">,</tt> <tt id="link-263" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-263', 'end', 'link-225');">end</a></tt><tt class="py-op">)</tt> </tt>
<a name="L755"></a><tt class="py-lineno"> 755</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L756"></a><tt class="py-lineno"> 756</tt>  <tt class="py-line">            <tt class="py-name">suffix_str</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-264" class="py-name"><a title="Bio.Seq.Seq._get_seq_str_and_check_alphabet" class="py-name" href="#" onclick="return doclink('link-264', '_get_seq_str_and_check_alphabet', 'link-222');">_get_seq_str_and_check_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">suffix</tt><tt class="py-op">)</tt> </tt>
<a name="L757"></a><tt class="py-lineno"> 757</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-265" class="py-name"><a title="Bio.Seq.Seq.endswith" class="py-name" href="#" onclick="return doclink('link-265', 'endswith', 'link-261');">endswith</a></tt><tt class="py-op">(</tt><tt class="py-name">suffix_str</tt><tt class="py-op">,</tt> <tt id="link-266" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-266', 'start', 'link-224');">start</a></tt><tt class="py-op">,</tt> <tt id="link-267" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-267', 'end', 'link-225');">end</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L758"></a><tt class="py-lineno"> 758</tt>  <tt class="py-line"> </tt>
<a name="Seq.split"></a><div id="Seq.split-def"><a name="L759"></a><tt class="py-lineno"> 759</tt> <a class="py-toggle" href="#" id="Seq.split-toggle" onclick="return toggle('Seq.split');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#split">split</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sep</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">maxsplit</tt><tt class="py-op">=</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.split-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.split-expanded"><a name="L760"></a><tt class="py-lineno"> 760</tt>  <tt class="py-line">        <tt class="py-docstring">"""Split method, like that of a python string.</tt> </tt>
<a name="L761"></a><tt class="py-lineno"> 761</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L762"></a><tt class="py-lineno"> 762</tt>  <tt class="py-line"><tt class="py-docstring">        This behaves like the python string method of the same name.</tt> </tt>
<a name="L763"></a><tt class="py-lineno"> 763</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L764"></a><tt class="py-lineno"> 764</tt>  <tt class="py-line"><tt class="py-docstring">        Return a list of the 'words' in the string (as Seq objects),</tt> </tt>
<a name="L765"></a><tt class="py-lineno"> 765</tt>  <tt class="py-line"><tt class="py-docstring">        using sep as the delimiter string.  If maxsplit is given, at</tt> </tt>
<a name="L766"></a><tt class="py-lineno"> 766</tt>  <tt class="py-line"><tt class="py-docstring">        most maxsplit splits are done.  If maxsplit is omitted, all</tt> </tt>
<a name="L767"></a><tt class="py-lineno"> 767</tt>  <tt class="py-line"><tt class="py-docstring">        splits are made.</tt> </tt>
<a name="L768"></a><tt class="py-lineno"> 768</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L769"></a><tt class="py-lineno"> 769</tt>  <tt class="py-line"><tt class="py-docstring">        Following the python string method, sep will by default be any</tt> </tt>
<a name="L770"></a><tt class="py-lineno"> 770</tt>  <tt class="py-line"><tt class="py-docstring">        white space (tabs, spaces, newlines) but this is unlikely to</tt> </tt>
<a name="L771"></a><tt class="py-lineno"> 771</tt>  <tt class="py-line"><tt class="py-docstring">        apply to biological sequences.</tt> </tt>
<a name="L772"></a><tt class="py-lineno"> 772</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L773"></a><tt class="py-lineno"> 773</tt>  <tt class="py-line"><tt class="py-docstring">        e.g.</tt> </tt>
<a name="L774"></a><tt class="py-lineno"> 774</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L775"></a><tt class="py-lineno"> 775</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L776"></a><tt class="py-lineno"> 776</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_rna = Seq("GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG")</tt> </tt>
<a name="L777"></a><tt class="py-lineno"> 777</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_aa = my_rna.translate()</tt> </tt>
<a name="L778"></a><tt class="py-lineno"> 778</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_aa</tt> </tt>
<a name="L779"></a><tt class="py-lineno"> 779</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('VMAIVMGR*KGAR*L', HasStopCodon(ExtendedIUPACProtein(), '*'))</tt> </tt>
<a name="L780"></a><tt class="py-lineno"> 780</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; for pep in my_aa.split("*"):</tt> </tt>
<a name="L781"></a><tt class="py-lineno"> 781</tt>  <tt class="py-line"><tt class="py-docstring">        ...     pep</tt> </tt>
<a name="L782"></a><tt class="py-lineno"> 782</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('VMAIVMGR', HasStopCodon(ExtendedIUPACProtein(), '*'))</tt> </tt>
<a name="L783"></a><tt class="py-lineno"> 783</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('KGAR', HasStopCodon(ExtendedIUPACProtein(), '*'))</tt> </tt>
<a name="L784"></a><tt class="py-lineno"> 784</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('L', HasStopCodon(ExtendedIUPACProtein(), '*'))</tt> </tt>
<a name="L785"></a><tt class="py-lineno"> 785</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; for pep in my_aa.split("*", 1):</tt> </tt>
<a name="L786"></a><tt class="py-lineno"> 786</tt>  <tt class="py-line"><tt class="py-docstring">        ...     pep</tt> </tt>
<a name="L787"></a><tt class="py-lineno"> 787</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('VMAIVMGR', HasStopCodon(ExtendedIUPACProtein(), '*'))</tt> </tt>
<a name="L788"></a><tt class="py-lineno"> 788</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('KGAR*L', HasStopCodon(ExtendedIUPACProtein(), '*'))</tt> </tt>
<a name="L789"></a><tt class="py-lineno"> 789</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L790"></a><tt class="py-lineno"> 790</tt>  <tt class="py-line"><tt class="py-docstring">        See also the rsplit method:</tt> </tt>
<a name="L791"></a><tt class="py-lineno"> 791</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L792"></a><tt class="py-lineno"> 792</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; for pep in my_aa.rsplit("*", 1):</tt> </tt>
<a name="L793"></a><tt class="py-lineno"> 793</tt>  <tt class="py-line"><tt class="py-docstring">        ...     pep</tt> </tt>
<a name="L794"></a><tt class="py-lineno"> 794</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('VMAIVMGR*KGAR', HasStopCodon(ExtendedIUPACProtein(), '*'))</tt> </tt>
<a name="L795"></a><tt class="py-lineno"> 795</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('L', HasStopCodon(ExtendedIUPACProtein(), '*'))</tt> </tt>
<a name="L796"></a><tt class="py-lineno"> 796</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L797"></a><tt class="py-lineno"> 797</tt>  <tt class="py-line">        <tt class="py-comment"># If it has one, check the alphabet:</tt> </tt>
<a name="L798"></a><tt class="py-lineno"> 798</tt>  <tt class="py-line">        <tt class="py-name">sep_str</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-268" class="py-name"><a title="Bio.Seq.Seq._get_seq_str_and_check_alphabet" class="py-name" href="#" onclick="return doclink('link-268', '_get_seq_str_and_check_alphabet', 'link-222');">_get_seq_str_and_check_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">sep</tt><tt class="py-op">)</tt> </tt>
<a name="L799"></a><tt class="py-lineno"> 799</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - If the sep is the defined stop symbol, or gap char,</tt> </tt>
<a name="L800"></a><tt class="py-lineno"> 800</tt>  <tt class="py-line">        <tt class="py-comment"># should we adjust the alphabet?</tt> </tt>
<a name="L801"></a><tt class="py-lineno"> 801</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt id="link-269" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-269', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">part</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-270" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-270', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L802"></a><tt class="py-lineno"> 802</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">part</tt> <tt class="py-keyword">in</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-271" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-271', 'split', 'link-271');">split</a></tt><tt class="py-op">(</tt><tt class="py-name">sep_str</tt><tt class="py-op">,</tt> <tt class="py-name">maxsplit</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
</div><a name="L803"></a><tt class="py-lineno"> 803</tt>  <tt class="py-line"> </tt>
<a name="Seq.rsplit"></a><div id="Seq.rsplit-def"><a name="L804"></a><tt class="py-lineno"> 804</tt> <a class="py-toggle" href="#" id="Seq.rsplit-toggle" onclick="return toggle('Seq.rsplit');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#rsplit">rsplit</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sep</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">maxsplit</tt><tt class="py-op">=</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.rsplit-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.rsplit-expanded"><a name="L805"></a><tt class="py-lineno"> 805</tt>  <tt class="py-line">        <tt class="py-docstring">"""Do a right split method, like that of a python string.</tt> </tt>
<a name="L806"></a><tt class="py-lineno"> 806</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L807"></a><tt class="py-lineno"> 807</tt>  <tt class="py-line"><tt class="py-docstring">        This behaves like the python string method of the same name.</tt> </tt>
<a name="L808"></a><tt class="py-lineno"> 808</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L809"></a><tt class="py-lineno"> 809</tt>  <tt class="py-line"><tt class="py-docstring">        Return a list of the 'words' in the string (as Seq objects),</tt> </tt>
<a name="L810"></a><tt class="py-lineno"> 810</tt>  <tt class="py-line"><tt class="py-docstring">        using sep as the delimiter string.  If maxsplit is given, at</tt> </tt>
<a name="L811"></a><tt class="py-lineno"> 811</tt>  <tt class="py-line"><tt class="py-docstring">        most maxsplit splits are done COUNTING FROM THE RIGHT.</tt> </tt>
<a name="L812"></a><tt class="py-lineno"> 812</tt>  <tt class="py-line"><tt class="py-docstring">        If maxsplit is omitted, all splits are made.</tt> </tt>
<a name="L813"></a><tt class="py-lineno"> 813</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L814"></a><tt class="py-lineno"> 814</tt>  <tt class="py-line"><tt class="py-docstring">        Following the python string method, sep will by default be any</tt> </tt>
<a name="L815"></a><tt class="py-lineno"> 815</tt>  <tt class="py-line"><tt class="py-docstring">        white space (tabs, spaces, newlines) but this is unlikely to</tt> </tt>
<a name="L816"></a><tt class="py-lineno"> 816</tt>  <tt class="py-line"><tt class="py-docstring">        apply to biological sequences.</tt> </tt>
<a name="L817"></a><tt class="py-lineno"> 817</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L818"></a><tt class="py-lineno"> 818</tt>  <tt class="py-line"><tt class="py-docstring">        e.g. print(my_seq.rsplit("*",1))</tt> </tt>
<a name="L819"></a><tt class="py-lineno"> 819</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L820"></a><tt class="py-lineno"> 820</tt>  <tt class="py-line"><tt class="py-docstring">        See also the split method.</tt> </tt>
<a name="L821"></a><tt class="py-lineno"> 821</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L822"></a><tt class="py-lineno"> 822</tt>  <tt class="py-line">        <tt class="py-comment"># If it has one, check the alphabet:</tt> </tt>
<a name="L823"></a><tt class="py-lineno"> 823</tt>  <tt class="py-line">        <tt class="py-name">sep_str</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-272" class="py-name"><a title="Bio.Seq.Seq._get_seq_str_and_check_alphabet" class="py-name" href="#" onclick="return doclink('link-272', '_get_seq_str_and_check_alphabet', 'link-222');">_get_seq_str_and_check_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">sep</tt><tt class="py-op">)</tt> </tt>
<a name="L824"></a><tt class="py-lineno"> 824</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt id="link-273" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-273', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">part</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-274" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-274', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L825"></a><tt class="py-lineno"> 825</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">part</tt> <tt class="py-keyword">in</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-275" class="py-name" targets="Method Bio.Seq.Seq.rsplit()=Bio.Seq.Seq-class.html#rsplit"><a title="Bio.Seq.Seq.rsplit" class="py-name" href="#" onclick="return doclink('link-275', 'rsplit', 'link-275');">rsplit</a></tt><tt class="py-op">(</tt><tt class="py-name">sep_str</tt><tt class="py-op">,</tt> <tt class="py-name">maxsplit</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
</div><a name="L826"></a><tt class="py-lineno"> 826</tt>  <tt class="py-line"> </tt>
<a name="Seq.strip"></a><div id="Seq.strip-def"><a name="L827"></a><tt class="py-lineno"> 827</tt> <a class="py-toggle" href="#" id="Seq.strip-toggle" onclick="return toggle('Seq.strip');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#strip">strip</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">chars</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.strip-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.strip-expanded"><a name="L828"></a><tt class="py-lineno"> 828</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a new Seq object with leading and trailing ends stripped.</tt> </tt>
<a name="L829"></a><tt class="py-lineno"> 829</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L830"></a><tt class="py-lineno"> 830</tt>  <tt class="py-line"><tt class="py-docstring">        This behaves like the python string method of the same name.</tt> </tt>
<a name="L831"></a><tt class="py-lineno"> 831</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L832"></a><tt class="py-lineno"> 832</tt>  <tt class="py-line"><tt class="py-docstring">        Optional argument chars defines which characters to remove.  If</tt> </tt>
<a name="L833"></a><tt class="py-lineno"> 833</tt>  <tt class="py-line"><tt class="py-docstring">        omitted or None (default) then as for the python string method,</tt> </tt>
<a name="L834"></a><tt class="py-lineno"> 834</tt>  <tt class="py-line"><tt class="py-docstring">        this defaults to removing any white space.</tt> </tt>
<a name="L835"></a><tt class="py-lineno"> 835</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L836"></a><tt class="py-lineno"> 836</tt>  <tt class="py-line"><tt class="py-docstring">        e.g. print(my_seq.strip("-"))</tt> </tt>
<a name="L837"></a><tt class="py-lineno"> 837</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L838"></a><tt class="py-lineno"> 838</tt>  <tt class="py-line"><tt class="py-docstring">        See also the lstrip and rstrip methods.</tt> </tt>
<a name="L839"></a><tt class="py-lineno"> 839</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L840"></a><tt class="py-lineno"> 840</tt>  <tt class="py-line">        <tt class="py-comment"># If it has one, check the alphabet:</tt> </tt>
<a name="L841"></a><tt class="py-lineno"> 841</tt>  <tt class="py-line">        <tt class="py-name">strip_str</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-276" class="py-name"><a title="Bio.Seq.Seq._get_seq_str_and_check_alphabet" class="py-name" href="#" onclick="return doclink('link-276', '_get_seq_str_and_check_alphabet', 'link-222');">_get_seq_str_and_check_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">chars</tt><tt class="py-op">)</tt> </tt>
<a name="L842"></a><tt class="py-lineno"> 842</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-277" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-277', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-278" class="py-name" targets="Method Bio.Seq.Seq.strip()=Bio.Seq.Seq-class.html#strip"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-278', 'strip', 'link-278');">strip</a></tt><tt class="py-op">(</tt><tt class="py-name">strip_str</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-279" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-279', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L843"></a><tt class="py-lineno"> 843</tt>  <tt class="py-line"> </tt>
<a name="Seq.lstrip"></a><div id="Seq.lstrip-def"><a name="L844"></a><tt class="py-lineno"> 844</tt> <a class="py-toggle" href="#" id="Seq.lstrip-toggle" onclick="return toggle('Seq.lstrip');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#lstrip">lstrip</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">chars</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.lstrip-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.lstrip-expanded"><a name="L845"></a><tt class="py-lineno"> 845</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a new Seq object with leading (left) end stripped.</tt> </tt>
<a name="L846"></a><tt class="py-lineno"> 846</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L847"></a><tt class="py-lineno"> 847</tt>  <tt class="py-line"><tt class="py-docstring">        This behaves like the python string method of the same name.</tt> </tt>
<a name="L848"></a><tt class="py-lineno"> 848</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L849"></a><tt class="py-lineno"> 849</tt>  <tt class="py-line"><tt class="py-docstring">        Optional argument chars defines which characters to remove.  If</tt> </tt>
<a name="L850"></a><tt class="py-lineno"> 850</tt>  <tt class="py-line"><tt class="py-docstring">        omitted or None (default) then as for the python string method,</tt> </tt>
<a name="L851"></a><tt class="py-lineno"> 851</tt>  <tt class="py-line"><tt class="py-docstring">        this defaults to removing any white space.</tt> </tt>
<a name="L852"></a><tt class="py-lineno"> 852</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L853"></a><tt class="py-lineno"> 853</tt>  <tt class="py-line"><tt class="py-docstring">        e.g. print(my_seq.lstrip("-"))</tt> </tt>
<a name="L854"></a><tt class="py-lineno"> 854</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L855"></a><tt class="py-lineno"> 855</tt>  <tt class="py-line"><tt class="py-docstring">        See also the strip and rstrip methods.</tt> </tt>
<a name="L856"></a><tt class="py-lineno"> 856</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L857"></a><tt class="py-lineno"> 857</tt>  <tt class="py-line">        <tt class="py-comment"># If it has one, check the alphabet:</tt> </tt>
<a name="L858"></a><tt class="py-lineno"> 858</tt>  <tt class="py-line">        <tt class="py-name">strip_str</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-280" class="py-name"><a title="Bio.Seq.Seq._get_seq_str_and_check_alphabet" class="py-name" href="#" onclick="return doclink('link-280', '_get_seq_str_and_check_alphabet', 'link-222');">_get_seq_str_and_check_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">chars</tt><tt class="py-op">)</tt> </tt>
<a name="L859"></a><tt class="py-lineno"> 859</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-281" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-281', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-282" class="py-name" targets="Method Bio.Seq.Seq.lstrip()=Bio.Seq.Seq-class.html#lstrip"><a title="Bio.Seq.Seq.lstrip" class="py-name" href="#" onclick="return doclink('link-282', 'lstrip', 'link-282');">lstrip</a></tt><tt class="py-op">(</tt><tt class="py-name">strip_str</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-283" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-283', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L860"></a><tt class="py-lineno"> 860</tt>  <tt class="py-line"> </tt>
<a name="Seq.rstrip"></a><div id="Seq.rstrip-def"><a name="L861"></a><tt class="py-lineno"> 861</tt> <a class="py-toggle" href="#" id="Seq.rstrip-toggle" onclick="return toggle('Seq.rstrip');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#rstrip">rstrip</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">chars</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.rstrip-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.rstrip-expanded"><a name="L862"></a><tt class="py-lineno"> 862</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a new Seq object with trailing (right) end stripped.</tt> </tt>
<a name="L863"></a><tt class="py-lineno"> 863</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L864"></a><tt class="py-lineno"> 864</tt>  <tt class="py-line"><tt class="py-docstring">        This behaves like the python string method of the same name.</tt> </tt>
<a name="L865"></a><tt class="py-lineno"> 865</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L866"></a><tt class="py-lineno"> 866</tt>  <tt class="py-line"><tt class="py-docstring">        Optional argument chars defines which characters to remove.  If</tt> </tt>
<a name="L867"></a><tt class="py-lineno"> 867</tt>  <tt class="py-line"><tt class="py-docstring">        omitted or None (default) then as for the python string method,</tt> </tt>
<a name="L868"></a><tt class="py-lineno"> 868</tt>  <tt class="py-line"><tt class="py-docstring">        this defaults to removing any white space.</tt> </tt>
<a name="L869"></a><tt class="py-lineno"> 869</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L870"></a><tt class="py-lineno"> 870</tt>  <tt class="py-line"><tt class="py-docstring">        e.g. Removing a nucleotide sequence's polyadenylation (poly-A tail):</tt> </tt>
<a name="L871"></a><tt class="py-lineno"> 871</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L872"></a><tt class="py-lineno"> 872</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import IUPAC</tt> </tt>
<a name="L873"></a><tt class="py-lineno"> 873</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L874"></a><tt class="py-lineno"> 874</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq = Seq("CGGTACGCTTATGTCACGTAGAAAAAA", IUPAC.unambiguous_dna)</tt> </tt>
<a name="L875"></a><tt class="py-lineno"> 875</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L876"></a><tt class="py-lineno"> 876</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('CGGTACGCTTATGTCACGTAGAAAAAA', IUPACUnambiguousDNA())</tt> </tt>
<a name="L877"></a><tt class="py-lineno"> 877</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.rstrip("A")</tt> </tt>
<a name="L878"></a><tt class="py-lineno"> 878</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('CGGTACGCTTATGTCACGTAG', IUPACUnambiguousDNA())</tt> </tt>
<a name="L879"></a><tt class="py-lineno"> 879</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L880"></a><tt class="py-lineno"> 880</tt>  <tt class="py-line"><tt class="py-docstring">        See also the strip and lstrip methods.</tt> </tt>
<a name="L881"></a><tt class="py-lineno"> 881</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L882"></a><tt class="py-lineno"> 882</tt>  <tt class="py-line">        <tt class="py-comment"># If it has one, check the alphabet:</tt> </tt>
<a name="L883"></a><tt class="py-lineno"> 883</tt>  <tt class="py-line">        <tt class="py-name">strip_str</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-284" class="py-name"><a title="Bio.Seq.Seq._get_seq_str_and_check_alphabet" class="py-name" href="#" onclick="return doclink('link-284', '_get_seq_str_and_check_alphabet', 'link-222');">_get_seq_str_and_check_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">chars</tt><tt class="py-op">)</tt> </tt>
<a name="L884"></a><tt class="py-lineno"> 884</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-285" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-285', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-286" class="py-name" targets="Method Bio.Seq.Seq.rstrip()=Bio.Seq.Seq-class.html#rstrip"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-286', 'rstrip', 'link-286');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-name">strip_str</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-287" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-287', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L885"></a><tt class="py-lineno"> 885</tt>  <tt class="py-line"> </tt>
<a name="Seq.upper"></a><div id="Seq.upper-def"><a name="L886"></a><tt class="py-lineno"> 886</tt> <a class="py-toggle" href="#" id="Seq.upper-toggle" onclick="return toggle('Seq.upper');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#upper">upper</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.upper-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.upper-expanded"><a name="L887"></a><tt class="py-lineno"> 887</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return an upper case copy of the sequence.</tt> </tt>
<a name="L888"></a><tt class="py-lineno"> 888</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L889"></a><tt class="py-lineno"> 889</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import HasStopCodon, generic_protein</tt> </tt>
<a name="L890"></a><tt class="py-lineno"> 890</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L891"></a><tt class="py-lineno"> 891</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq = Seq("VHLTPeeK*", HasStopCodon(generic_protein))</tt> </tt>
<a name="L892"></a><tt class="py-lineno"> 892</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L893"></a><tt class="py-lineno"> 893</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('VHLTPeeK*', HasStopCodon(ProteinAlphabet(), '*'))</tt> </tt>
<a name="L894"></a><tt class="py-lineno"> 894</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.lower()</tt> </tt>
<a name="L895"></a><tt class="py-lineno"> 895</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('vhltpeek*', HasStopCodon(ProteinAlphabet(), '*'))</tt> </tt>
<a name="L896"></a><tt class="py-lineno"> 896</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.upper()</tt> </tt>
<a name="L897"></a><tt class="py-lineno"> 897</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('VHLTPEEK*', HasStopCodon(ProteinAlphabet(), '*'))</tt> </tt>
<a name="L898"></a><tt class="py-lineno"> 898</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L899"></a><tt class="py-lineno"> 899</tt>  <tt class="py-line"><tt class="py-docstring">        This will adjust the alphabet if required. See also the lower method.</tt> </tt>
<a name="L900"></a><tt class="py-lineno"> 900</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L901"></a><tt class="py-lineno"> 901</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-288" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-288', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-289" class="py-name" targets="Method Bio.Seq.Seq.upper()=Bio.Seq.Seq-class.html#upper,Method Bio.Seq.UnknownSeq.upper()=Bio.Seq.UnknownSeq-class.html#upper,Method Bio.SeqRecord.SeqRecord.upper()=Bio.SeqRecord.SeqRecord-class.html#upper"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-289', 'upper', 'link-289');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-290" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-290', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-291" class="py-name" targets="Method Bio.Alphabet.Alphabet._upper()=Bio.Alphabet.Alphabet-class.html#_upper,Method Bio.Alphabet.AlphabetEncoder._upper()=Bio.Alphabet.AlphabetEncoder-class.html#_upper,Method Bio.Alphabet.Gapped._upper()=Bio.Alphabet.Gapped-class.html#_upper,Method Bio.Alphabet.HasStopCodon._upper()=Bio.Alphabet.HasStopCodon-class.html#_upper,Method Bio.Alphabet.ThreeLetterProtein._upper()=Bio.Alphabet.ThreeLetterProtein-class.html#_upper"><a title="Bio.Alphabet.Alphabet._upper
Bio.Alphabet.AlphabetEncoder._upper
Bio.Alphabet.Gapped._upper
Bio.Alphabet.HasStopCodon._upper
Bio.Alphabet.ThreeLetterProtein._upper" class="py-name" href="#" onclick="return doclink('link-291', '_upper', 'link-291');">_upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L902"></a><tt class="py-lineno"> 902</tt>  <tt class="py-line"> </tt>
<a name="Seq.lower"></a><div id="Seq.lower-def"><a name="L903"></a><tt class="py-lineno"> 903</tt> <a class="py-toggle" href="#" id="Seq.lower-toggle" onclick="return toggle('Seq.lower');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#lower">lower</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.lower-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.lower-expanded"><a name="L904"></a><tt class="py-lineno"> 904</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a lower case copy of the sequence.</tt> </tt>
<a name="L905"></a><tt class="py-lineno"> 905</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L906"></a><tt class="py-lineno"> 906</tt>  <tt class="py-line"><tt class="py-docstring">        This will adjust the alphabet if required. Note that the IUPAC</tt> </tt>
<a name="L907"></a><tt class="py-lineno"> 907</tt>  <tt class="py-line"><tt class="py-docstring">        alphabets are upper case only, and thus a generic alphabet must be</tt> </tt>
<a name="L908"></a><tt class="py-lineno"> 908</tt>  <tt class="py-line"><tt class="py-docstring">        substituted.</tt> </tt>
<a name="L909"></a><tt class="py-lineno"> 909</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L910"></a><tt class="py-lineno"> 910</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import Gapped, generic_dna</tt> </tt>
<a name="L911"></a><tt class="py-lineno"> 911</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import IUPAC</tt> </tt>
<a name="L912"></a><tt class="py-lineno"> 912</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L913"></a><tt class="py-lineno"> 913</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq = Seq("CGGTACGCTTATGTCACGTAG*AAAAAA",</tt> </tt>
<a name="L914"></a><tt class="py-lineno"> 914</tt>  <tt class="py-line"><tt class="py-docstring">        ...          Gapped(IUPAC.unambiguous_dna, "*"))</tt> </tt>
<a name="L915"></a><tt class="py-lineno"> 915</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L916"></a><tt class="py-lineno"> 916</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('CGGTACGCTTATGTCACGTAG*AAAAAA', Gapped(IUPACUnambiguousDNA(), '*'))</tt> </tt>
<a name="L917"></a><tt class="py-lineno"> 917</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.lower()</tt> </tt>
<a name="L918"></a><tt class="py-lineno"> 918</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('cggtacgcttatgtcacgtag*aaaaaa', Gapped(DNAAlphabet(), '*'))</tt> </tt>
<a name="L919"></a><tt class="py-lineno"> 919</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L920"></a><tt class="py-lineno"> 920</tt>  <tt class="py-line"><tt class="py-docstring">        See also the upper method.</tt> </tt>
<a name="L921"></a><tt class="py-lineno"> 921</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L922"></a><tt class="py-lineno"> 922</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-292" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-292', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-293" class="py-name"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-293', 'lower', 'link-32');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-294" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-294', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-295" class="py-name" targets="Method Bio.Alphabet.Alphabet._lower()=Bio.Alphabet.Alphabet-class.html#_lower,Method Bio.Alphabet.AlphabetEncoder._lower()=Bio.Alphabet.AlphabetEncoder-class.html#_lower,Method Bio.Alphabet.Gapped._lower()=Bio.Alphabet.Gapped-class.html#_lower,Method Bio.Alphabet.HasStopCodon._lower()=Bio.Alphabet.HasStopCodon-class.html#_lower,Method Bio.Alphabet.ThreeLetterProtein._lower()=Bio.Alphabet.ThreeLetterProtein-class.html#_lower"><a title="Bio.Alphabet.Alphabet._lower
Bio.Alphabet.AlphabetEncoder._lower
Bio.Alphabet.Gapped._lower
Bio.Alphabet.HasStopCodon._lower
Bio.Alphabet.ThreeLetterProtein._lower" class="py-name" href="#" onclick="return doclink('link-295', '_lower', 'link-295');">_lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L923"></a><tt class="py-lineno"> 923</tt>  <tt class="py-line"> </tt>
<a name="Seq.complement"></a><div id="Seq.complement-def"><a name="L924"></a><tt class="py-lineno"> 924</tt> <a class="py-toggle" href="#" id="Seq.complement-toggle" onclick="return toggle('Seq.complement');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#complement">complement</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.complement-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.complement-expanded"><a name="L925"></a><tt class="py-lineno"> 925</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the complement sequence by creating a new Seq object.</tt> </tt>
<a name="L926"></a><tt class="py-lineno"> 926</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L927"></a><tt class="py-lineno"> 927</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L928"></a><tt class="py-lineno"> 928</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import IUPAC</tt> </tt>
<a name="L929"></a><tt class="py-lineno"> 929</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna = Seq("CCCCCGATAG", IUPAC.unambiguous_dna)</tt> </tt>
<a name="L930"></a><tt class="py-lineno"> 930</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna</tt> </tt>
<a name="L931"></a><tt class="py-lineno"> 931</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('CCCCCGATAG', IUPACUnambiguousDNA())</tt> </tt>
<a name="L932"></a><tt class="py-lineno"> 932</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna.complement()</tt> </tt>
<a name="L933"></a><tt class="py-lineno"> 933</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('GGGGGCTATC', IUPACUnambiguousDNA())</tt> </tt>
<a name="L934"></a><tt class="py-lineno"> 934</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L935"></a><tt class="py-lineno"> 935</tt>  <tt class="py-line"><tt class="py-docstring">        You can of course used mixed case sequences,</tt> </tt>
<a name="L936"></a><tt class="py-lineno"> 936</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L937"></a><tt class="py-lineno"> 937</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L938"></a><tt class="py-lineno"> 938</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_dna</tt> </tt>
<a name="L939"></a><tt class="py-lineno"> 939</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna = Seq("CCCCCgatA-GD", generic_dna)</tt> </tt>
<a name="L940"></a><tt class="py-lineno"> 940</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna</tt> </tt>
<a name="L941"></a><tt class="py-lineno"> 941</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('CCCCCgatA-GD', DNAAlphabet())</tt> </tt>
<a name="L942"></a><tt class="py-lineno"> 942</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna.complement()</tt> </tt>
<a name="L943"></a><tt class="py-lineno"> 943</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('GGGGGctaT-CH', DNAAlphabet())</tt> </tt>
<a name="L944"></a><tt class="py-lineno"> 944</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L945"></a><tt class="py-lineno"> 945</tt>  <tt class="py-line"><tt class="py-docstring">        Note in the above example, ambiguous character D denotes</tt> </tt>
<a name="L946"></a><tt class="py-lineno"> 946</tt>  <tt class="py-line"><tt class="py-docstring">        G, A or T so its complement is H (for C, T or A).</tt> </tt>
<a name="L947"></a><tt class="py-lineno"> 947</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L948"></a><tt class="py-lineno"> 948</tt>  <tt class="py-line"><tt class="py-docstring">        Trying to complement a protein sequence raises an exception.</tt> </tt>
<a name="L949"></a><tt class="py-lineno"> 949</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L950"></a><tt class="py-lineno"> 950</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_protein = Seq("MAIVMGR", IUPAC.protein)</tt> </tt>
<a name="L951"></a><tt class="py-lineno"> 951</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_protein.complement()</tt> </tt>
<a name="L952"></a><tt class="py-lineno"> 952</tt>  <tt class="py-line"><tt class="py-docstring">        Traceback (most recent call last):</tt> </tt>
<a name="L953"></a><tt class="py-lineno"> 953</tt>  <tt class="py-line"><tt class="py-docstring">           ...</tt> </tt>
<a name="L954"></a><tt class="py-lineno"> 954</tt>  <tt class="py-line"><tt class="py-docstring">        ValueError: Proteins do not have complements!</tt> </tt>
<a name="L955"></a><tt class="py-lineno"> 955</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L956"></a><tt class="py-lineno"> 956</tt>  <tt class="py-line">        <tt class="py-name">base</tt> <tt class="py-op">=</tt> <tt id="link-296" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-296', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-297" class="py-name" targets="Function Bio.Alphabet._get_base_alphabet()=Bio.Alphabet-module.html#_get_base_alphabet"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-297', '_get_base_alphabet', 'link-297');">_get_base_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-298" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-298', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L957"></a><tt class="py-lineno"> 957</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">base</tt><tt class="py-op">,</tt> <tt id="link-299" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-299', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-300" class="py-name" targets="Class Bio.Alphabet.ProteinAlphabet=Bio.Alphabet.ProteinAlphabet-class.html"><a title="Bio.Alphabet.ProteinAlphabet" class="py-name" href="#" onclick="return doclink('link-300', 'ProteinAlphabet', 'link-300');">ProteinAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L958"></a><tt class="py-lineno"> 958</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Proteins do not have complements!"</tt><tt class="py-op">)</tt> </tt>
<a name="L959"></a><tt class="py-lineno"> 959</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">base</tt><tt class="py-op">,</tt> <tt id="link-301" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-301', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-302" class="py-name" targets="Class Bio.Alphabet.DNAAlphabet=Bio.Alphabet.DNAAlphabet-class.html"><a title="Bio.Alphabet.DNAAlphabet" class="py-name" href="#" onclick="return doclink('link-302', 'DNAAlphabet', 'link-302');">DNAAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L960"></a><tt class="py-lineno"> 960</tt>  <tt class="py-line">            <tt class="py-name">ttable</tt> <tt class="py-op">=</tt> <tt id="link-303" class="py-name"><a title="Bio.Seq._dna_complement_table" class="py-name" href="#" onclick="return doclink('link-303', '_dna_complement_table', 'link-34');">_dna_complement_table</a></tt> </tt>
<a name="L961"></a><tt class="py-lineno"> 961</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">base</tt><tt class="py-op">,</tt> <tt id="link-304" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-304', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-305" class="py-name" targets="Class Bio.Alphabet.RNAAlphabet=Bio.Alphabet.RNAAlphabet-class.html"><a title="Bio.Alphabet.RNAAlphabet" class="py-name" href="#" onclick="return doclink('link-305', 'RNAAlphabet', 'link-305');">RNAAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L962"></a><tt class="py-lineno"> 962</tt>  <tt class="py-line">            <tt class="py-name">ttable</tt> <tt class="py-op">=</tt> <tt id="link-306" class="py-name"><a title="Bio.Seq._rna_complement_table" class="py-name" href="#" onclick="return doclink('link-306', '_rna_complement_table', 'link-37');">_rna_complement_table</a></tt> </tt>
<a name="L963"></a><tt class="py-lineno"> 963</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-op">(</tt><tt class="py-string">"U"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_data</tt> <tt class="py-keyword">or</tt> <tt class="py-string">"u"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_data</tt><tt class="py-op">)</tt> \ </tt>
<a name="L964"></a><tt class="py-lineno"> 964</tt>  <tt class="py-line">                <tt class="py-keyword">and</tt> <tt class="py-op">(</tt><tt class="py-string">"T"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_data</tt> <tt class="py-keyword">or</tt> <tt class="py-string">"t"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_data</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L965"></a><tt class="py-lineno"> 965</tt>  <tt class="py-line">            <tt class="py-comment"># TODO - Handle this cleanly?</tt> </tt>
<a name="L966"></a><tt class="py-lineno"> 966</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Mixed RNA/DNA found"</tt><tt class="py-op">)</tt> </tt>
<a name="L967"></a><tt class="py-lineno"> 967</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-string">"U"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_data</tt> <tt class="py-keyword">or</tt> <tt class="py-string">"u"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_data</tt><tt class="py-op">:</tt> </tt>
<a name="L968"></a><tt class="py-lineno"> 968</tt>  <tt class="py-line">            <tt class="py-name">ttable</tt> <tt class="py-op">=</tt> <tt id="link-307" class="py-name"><a title="Bio.Seq._rna_complement_table" class="py-name" href="#" onclick="return doclink('link-307', '_rna_complement_table', 'link-37');">_rna_complement_table</a></tt> </tt>
<a name="L969"></a><tt class="py-lineno"> 969</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L970"></a><tt class="py-lineno"> 970</tt>  <tt class="py-line">            <tt class="py-name">ttable</tt> <tt class="py-op">=</tt> <tt id="link-308" class="py-name"><a title="Bio.Seq._dna_complement_table" class="py-name" href="#" onclick="return doclink('link-308', '_dna_complement_table', 'link-34');">_dna_complement_table</a></tt> </tt>
<a name="L971"></a><tt class="py-lineno"> 971</tt>  <tt class="py-line">        <tt class="py-comment"># Much faster on really long sequences than the previous loop based</tt> </tt>
<a name="L972"></a><tt class="py-lineno"> 972</tt>  <tt class="py-line">        <tt class="py-comment"># one. Thanks to Michael Palmer, University of Waterloo.</tt> </tt>
<a name="L973"></a><tt class="py-lineno"> 973</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-309" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-309', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-310" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.translate()=Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#translate,Method Bio.Seq.Seq.translate()=Bio.Seq.Seq-class.html#translate,Method Bio.Seq.UnknownSeq.translate()=Bio.Seq.UnknownSeq-class.html#translate,Function Bio.Seq.translate()=Bio.Seq-module.html#translate,Method Bio.SeqFeature.SeqFeature.translate()=Bio.SeqFeature.SeqFeature-class.html#translate,Method Bio.SeqRecord.SeqRecord.translate()=Bio.SeqRecord.SeqRecord-class.html#translate,Method Bio.codonalign.codonseq.CodonSeq.translate()=Bio.codonalign.codonseq.CodonSeq-class.html#translate"><a title="Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.translate
Bio.Seq.Seq.translate
Bio.Seq.UnknownSeq.translate
Bio.Seq.translate
Bio.SeqFeature.SeqFeature.translate
Bio.SeqRecord.SeqRecord.translate
Bio.codonalign.codonseq.CodonSeq.translate" class="py-name" href="#" onclick="return doclink('link-310', 'translate', 'link-310');">translate</a></tt><tt class="py-op">(</tt><tt class="py-name">ttable</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-311" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-311', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L974"></a><tt class="py-lineno"> 974</tt>  <tt class="py-line"> </tt>
<a name="Seq.reverse_complement"></a><div id="Seq.reverse_complement-def"><a name="L975"></a><tt class="py-lineno"> 975</tt> <a class="py-toggle" href="#" id="Seq.reverse_complement-toggle" onclick="return toggle('Seq.reverse_complement');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#reverse_complement">reverse_complement</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.reverse_complement-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.reverse_complement-expanded"><a name="L976"></a><tt class="py-lineno"> 976</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the reverse complement sequence by creating a new Seq object.</tt> </tt>
<a name="L977"></a><tt class="py-lineno"> 977</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L978"></a><tt class="py-lineno"> 978</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L979"></a><tt class="py-lineno"> 979</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import IUPAC</tt> </tt>
<a name="L980"></a><tt class="py-lineno"> 980</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna = Seq("CCCCCGATAGNR", IUPAC.ambiguous_dna)</tt> </tt>
<a name="L981"></a><tt class="py-lineno"> 981</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna</tt> </tt>
<a name="L982"></a><tt class="py-lineno"> 982</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('CCCCCGATAGNR', IUPACAmbiguousDNA())</tt> </tt>
<a name="L983"></a><tt class="py-lineno"> 983</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna.reverse_complement()</tt> </tt>
<a name="L984"></a><tt class="py-lineno"> 984</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('YNCTATCGGGGG', IUPACAmbiguousDNA())</tt> </tt>
<a name="L985"></a><tt class="py-lineno"> 985</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L986"></a><tt class="py-lineno"> 986</tt>  <tt class="py-line"><tt class="py-docstring">        Note in the above example, since R = G or A, its complement</tt> </tt>
<a name="L987"></a><tt class="py-lineno"> 987</tt>  <tt class="py-line"><tt class="py-docstring">        is Y (which denotes C or T).</tt> </tt>
<a name="L988"></a><tt class="py-lineno"> 988</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L989"></a><tt class="py-lineno"> 989</tt>  <tt class="py-line"><tt class="py-docstring">        You can of course used mixed case sequences,</tt> </tt>
<a name="L990"></a><tt class="py-lineno"> 990</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L991"></a><tt class="py-lineno"> 991</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L992"></a><tt class="py-lineno"> 992</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_dna</tt> </tt>
<a name="L993"></a><tt class="py-lineno"> 993</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna = Seq("CCCCCgatA-G", generic_dna)</tt> </tt>
<a name="L994"></a><tt class="py-lineno"> 994</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna</tt> </tt>
<a name="L995"></a><tt class="py-lineno"> 995</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('CCCCCgatA-G', DNAAlphabet())</tt> </tt>
<a name="L996"></a><tt class="py-lineno"> 996</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna.reverse_complement()</tt> </tt>
<a name="L997"></a><tt class="py-lineno"> 997</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('C-TatcGGGGG', DNAAlphabet())</tt> </tt>
<a name="L998"></a><tt class="py-lineno"> 998</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L999"></a><tt class="py-lineno"> 999</tt>  <tt class="py-line"><tt class="py-docstring">        Trying to complement a protein sequence raises an exception:</tt> </tt>
<a name="L1000"></a><tt class="py-lineno">1000</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1001"></a><tt class="py-lineno">1001</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_protein = Seq("MAIVMGR", IUPAC.protein)</tt> </tt>
<a name="L1002"></a><tt class="py-lineno">1002</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_protein.reverse_complement()</tt> </tt>
<a name="L1003"></a><tt class="py-lineno">1003</tt>  <tt class="py-line"><tt class="py-docstring">        Traceback (most recent call last):</tt> </tt>
<a name="L1004"></a><tt class="py-lineno">1004</tt>  <tt class="py-line"><tt class="py-docstring">           ...</tt> </tt>
<a name="L1005"></a><tt class="py-lineno">1005</tt>  <tt class="py-line"><tt class="py-docstring">        ValueError: Proteins do not have complements!</tt> </tt>
<a name="L1006"></a><tt class="py-lineno">1006</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1007"></a><tt class="py-lineno">1007</tt>  <tt class="py-line">        <tt class="py-comment"># Use -1 stride/step to reverse the complement</tt> </tt>
<a name="L1008"></a><tt class="py-lineno">1008</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-312" class="py-name" targets="Method Bio.Seq.MutableSeq.complement()=Bio.Seq.MutableSeq-class.html#complement,Method Bio.Seq.Seq.complement()=Bio.Seq.Seq-class.html#complement,Method Bio.Seq.UnknownSeq.complement()=Bio.Seq.UnknownSeq-class.html#complement,Function Bio.Seq.complement()=Bio.Seq-module.html#complement"><a title="Bio.Seq.MutableSeq.complement
Bio.Seq.Seq.complement
Bio.Seq.UnknownSeq.complement
Bio.Seq.complement" class="py-name" href="#" onclick="return doclink('link-312', 'complement', 'link-312');">complement</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
</div><a name="L1009"></a><tt class="py-lineno">1009</tt>  <tt class="py-line"> </tt>
<a name="Seq.transcribe"></a><div id="Seq.transcribe-def"><a name="L1010"></a><tt class="py-lineno">1010</tt> <a class="py-toggle" href="#" id="Seq.transcribe-toggle" onclick="return toggle('Seq.transcribe');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#transcribe">transcribe</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.transcribe-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.transcribe-expanded"><a name="L1011"></a><tt class="py-lineno">1011</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the RNA sequence from a DNA sequence by creating a new Seq object.</tt> </tt>
<a name="L1012"></a><tt class="py-lineno">1012</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1013"></a><tt class="py-lineno">1013</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L1014"></a><tt class="py-lineno">1014</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import IUPAC</tt> </tt>
<a name="L1015"></a><tt class="py-lineno">1015</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; coding_dna = Seq("ATGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG",</tt> </tt>
<a name="L1016"></a><tt class="py-lineno">1016</tt>  <tt class="py-line"><tt class="py-docstring">        ...                  IUPAC.unambiguous_dna)</tt> </tt>
<a name="L1017"></a><tt class="py-lineno">1017</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; coding_dna</tt> </tt>
<a name="L1018"></a><tt class="py-lineno">1018</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('ATGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG', IUPACUnambiguousDNA())</tt> </tt>
<a name="L1019"></a><tt class="py-lineno">1019</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; coding_dna.transcribe()</tt> </tt>
<a name="L1020"></a><tt class="py-lineno">1020</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('AUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAG', IUPACUnambiguousRNA())</tt> </tt>
<a name="L1021"></a><tt class="py-lineno">1021</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1022"></a><tt class="py-lineno">1022</tt>  <tt class="py-line"><tt class="py-docstring">        Trying to transcribe a protein or RNA sequence raises an exception:</tt> </tt>
<a name="L1023"></a><tt class="py-lineno">1023</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1024"></a><tt class="py-lineno">1024</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_protein = Seq("MAIVMGR", IUPAC.protein)</tt> </tt>
<a name="L1025"></a><tt class="py-lineno">1025</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_protein.transcribe()</tt> </tt>
<a name="L1026"></a><tt class="py-lineno">1026</tt>  <tt class="py-line"><tt class="py-docstring">        Traceback (most recent call last):</tt> </tt>
<a name="L1027"></a><tt class="py-lineno">1027</tt>  <tt class="py-line"><tt class="py-docstring">           ...</tt> </tt>
<a name="L1028"></a><tt class="py-lineno">1028</tt>  <tt class="py-line"><tt class="py-docstring">        ValueError: Proteins cannot be transcribed!</tt> </tt>
<a name="L1029"></a><tt class="py-lineno">1029</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1030"></a><tt class="py-lineno">1030</tt>  <tt class="py-line">        <tt class="py-name">base</tt> <tt class="py-op">=</tt> <tt id="link-313" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-313', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-314" class="py-name"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-314', '_get_base_alphabet', 'link-297');">_get_base_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-315" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-315', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1031"></a><tt class="py-lineno">1031</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">base</tt><tt class="py-op">,</tt> <tt id="link-316" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-316', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-317" class="py-name"><a title="Bio.Alphabet.ProteinAlphabet" class="py-name" href="#" onclick="return doclink('link-317', 'ProteinAlphabet', 'link-300');">ProteinAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1032"></a><tt class="py-lineno">1032</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Proteins cannot be transcribed!"</tt><tt class="py-op">)</tt> </tt>
<a name="L1033"></a><tt class="py-lineno">1033</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">base</tt><tt class="py-op">,</tt> <tt id="link-318" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-318', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-319" class="py-name"><a title="Bio.Alphabet.RNAAlphabet" class="py-name" href="#" onclick="return doclink('link-319', 'RNAAlphabet', 'link-305');">RNAAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1034"></a><tt class="py-lineno">1034</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"RNA cannot be transcribed!"</tt><tt class="py-op">)</tt> </tt>
<a name="L1035"></a><tt class="py-lineno">1035</tt>  <tt class="py-line"> </tt>
<a name="L1036"></a><tt class="py-lineno">1036</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-320" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-320', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-op">==</tt> <tt id="link-321" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-321', 'IUPAC', 'link-11');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-322" class="py-name" targets="Variable Bio.Alphabet.IUPAC.unambiguous_dna=Bio.Alphabet.IUPAC-module.html#unambiguous_dna"><a title="Bio.Alphabet.IUPAC.unambiguous_dna" class="py-name" href="#" onclick="return doclink('link-322', 'unambiguous_dna', 'link-322');">unambiguous_dna</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1037"></a><tt class="py-lineno">1037</tt>  <tt class="py-line">            <tt id="link-323" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-323', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-324" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-324', 'IUPAC', 'link-11');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-325" class="py-name" targets="Variable Bio.Alphabet.IUPAC.unambiguous_rna=Bio.Alphabet.IUPAC-module.html#unambiguous_rna"><a title="Bio.Alphabet.IUPAC.unambiguous_rna" class="py-name" href="#" onclick="return doclink('link-325', 'unambiguous_rna', 'link-325');">unambiguous_rna</a></tt> </tt>
<a name="L1038"></a><tt class="py-lineno">1038</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-326" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-326', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-op">==</tt> <tt id="link-327" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-327', 'IUPAC', 'link-11');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-328" class="py-name" targets="Variable Bio.Alphabet.IUPAC.ambiguous_dna=Bio.Alphabet.IUPAC-module.html#ambiguous_dna"><a title="Bio.Alphabet.IUPAC.ambiguous_dna" class="py-name" href="#" onclick="return doclink('link-328', 'ambiguous_dna', 'link-328');">ambiguous_dna</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1039"></a><tt class="py-lineno">1039</tt>  <tt class="py-line">            <tt id="link-329" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-329', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-330" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-330', 'IUPAC', 'link-11');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-331" class="py-name" targets="Variable Bio.Alphabet.IUPAC.ambiguous_rna=Bio.Alphabet.IUPAC-module.html#ambiguous_rna"><a title="Bio.Alphabet.IUPAC.ambiguous_rna" class="py-name" href="#" onclick="return doclink('link-331', 'ambiguous_rna', 'link-331');">ambiguous_rna</a></tt> </tt>
<a name="L1040"></a><tt class="py-lineno">1040</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1041"></a><tt class="py-lineno">1041</tt>  <tt class="py-line">            <tt id="link-332" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-332', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-333" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-333', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-334" class="py-name" targets="Variable Bio.Alphabet.generic_rna=Bio.Alphabet-module.html#generic_rna"><a title="Bio.Alphabet.generic_rna" class="py-name" href="#" onclick="return doclink('link-334', 'generic_rna', 'link-334');">generic_rna</a></tt> </tt>
<a name="L1042"></a><tt class="py-lineno">1042</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-335" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-335', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"T"</tt><tt class="py-op">,</tt> <tt class="py-string">"U"</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"t"</tt><tt class="py-op">,</tt> <tt class="py-string">"u"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-336" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-336', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1043"></a><tt class="py-lineno">1043</tt>  <tt class="py-line"> </tt>
<a name="Seq.back_transcribe"></a><div id="Seq.back_transcribe-def"><a name="L1044"></a><tt class="py-lineno">1044</tt> <a class="py-toggle" href="#" id="Seq.back_transcribe-toggle" onclick="return toggle('Seq.back_transcribe');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#back_transcribe">back_transcribe</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.back_transcribe-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.back_transcribe-expanded"><a name="L1045"></a><tt class="py-lineno">1045</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the DNA sequence from an RNA sequence by creating a new Seq object.</tt> </tt>
<a name="L1046"></a><tt class="py-lineno">1046</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1047"></a><tt class="py-lineno">1047</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L1048"></a><tt class="py-lineno">1048</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import IUPAC</tt> </tt>
<a name="L1049"></a><tt class="py-lineno">1049</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; messenger_rna = Seq("AUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAG",</tt> </tt>
<a name="L1050"></a><tt class="py-lineno">1050</tt>  <tt class="py-line"><tt class="py-docstring">        ...                     IUPAC.unambiguous_rna)</tt> </tt>
<a name="L1051"></a><tt class="py-lineno">1051</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; messenger_rna</tt> </tt>
<a name="L1052"></a><tt class="py-lineno">1052</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('AUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAG', IUPACUnambiguousRNA())</tt> </tt>
<a name="L1053"></a><tt class="py-lineno">1053</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; messenger_rna.back_transcribe()</tt> </tt>
<a name="L1054"></a><tt class="py-lineno">1054</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('ATGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG', IUPACUnambiguousDNA())</tt> </tt>
<a name="L1055"></a><tt class="py-lineno">1055</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1056"></a><tt class="py-lineno">1056</tt>  <tt class="py-line"><tt class="py-docstring">        Trying to back-transcribe a protein or DNA sequence raises an</tt> </tt>
<a name="L1057"></a><tt class="py-lineno">1057</tt>  <tt class="py-line"><tt class="py-docstring">        exception:</tt> </tt>
<a name="L1058"></a><tt class="py-lineno">1058</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1059"></a><tt class="py-lineno">1059</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_protein = Seq("MAIVMGR", IUPAC.protein)</tt> </tt>
<a name="L1060"></a><tt class="py-lineno">1060</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_protein.back_transcribe()</tt> </tt>
<a name="L1061"></a><tt class="py-lineno">1061</tt>  <tt class="py-line"><tt class="py-docstring">        Traceback (most recent call last):</tt> </tt>
<a name="L1062"></a><tt class="py-lineno">1062</tt>  <tt class="py-line"><tt class="py-docstring">           ...</tt> </tt>
<a name="L1063"></a><tt class="py-lineno">1063</tt>  <tt class="py-line"><tt class="py-docstring">        ValueError: Proteins cannot be back transcribed!</tt> </tt>
<a name="L1064"></a><tt class="py-lineno">1064</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1065"></a><tt class="py-lineno">1065</tt>  <tt class="py-line">        <tt class="py-name">base</tt> <tt class="py-op">=</tt> <tt id="link-337" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-337', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-338" class="py-name"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-338', '_get_base_alphabet', 'link-297');">_get_base_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-339" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-339', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1066"></a><tt class="py-lineno">1066</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">base</tt><tt class="py-op">,</tt> <tt id="link-340" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-340', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-341" class="py-name"><a title="Bio.Alphabet.ProteinAlphabet" class="py-name" href="#" onclick="return doclink('link-341', 'ProteinAlphabet', 'link-300');">ProteinAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1067"></a><tt class="py-lineno">1067</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Proteins cannot be back transcribed!"</tt><tt class="py-op">)</tt> </tt>
<a name="L1068"></a><tt class="py-lineno">1068</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">base</tt><tt class="py-op">,</tt> <tt id="link-342" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-342', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-343" class="py-name"><a title="Bio.Alphabet.DNAAlphabet" class="py-name" href="#" onclick="return doclink('link-343', 'DNAAlphabet', 'link-302');">DNAAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1069"></a><tt class="py-lineno">1069</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"DNA cannot be back transcribed!"</tt><tt class="py-op">)</tt> </tt>
<a name="L1070"></a><tt class="py-lineno">1070</tt>  <tt class="py-line"> </tt>
<a name="L1071"></a><tt class="py-lineno">1071</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-344" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-344', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-op">==</tt> <tt id="link-345" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-345', 'IUPAC', 'link-11');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-346" class="py-name"><a title="Bio.Alphabet.IUPAC.unambiguous_rna" class="py-name" href="#" onclick="return doclink('link-346', 'unambiguous_rna', 'link-325');">unambiguous_rna</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1072"></a><tt class="py-lineno">1072</tt>  <tt class="py-line">            <tt id="link-347" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-347', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-348" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-348', 'IUPAC', 'link-11');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-349" class="py-name"><a title="Bio.Alphabet.IUPAC.unambiguous_dna" class="py-name" href="#" onclick="return doclink('link-349', 'unambiguous_dna', 'link-322');">unambiguous_dna</a></tt> </tt>
<a name="L1073"></a><tt class="py-lineno">1073</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-350" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-350', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-op">==</tt> <tt id="link-351" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-351', 'IUPAC', 'link-11');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-352" class="py-name"><a title="Bio.Alphabet.IUPAC.ambiguous_rna" class="py-name" href="#" onclick="return doclink('link-352', 'ambiguous_rna', 'link-331');">ambiguous_rna</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1074"></a><tt class="py-lineno">1074</tt>  <tt class="py-line">            <tt id="link-353" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-353', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-354" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-354', 'IUPAC', 'link-11');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-355" class="py-name"><a title="Bio.Alphabet.IUPAC.ambiguous_dna" class="py-name" href="#" onclick="return doclink('link-355', 'ambiguous_dna', 'link-328');">ambiguous_dna</a></tt> </tt>
<a name="L1075"></a><tt class="py-lineno">1075</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1076"></a><tt class="py-lineno">1076</tt>  <tt class="py-line">            <tt id="link-356" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-356', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-357" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-357', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-358" class="py-name" targets="Variable Bio.Alphabet.generic_dna=Bio.Alphabet-module.html#generic_dna"><a title="Bio.Alphabet.generic_dna" class="py-name" href="#" onclick="return doclink('link-358', 'generic_dna', 'link-358');">generic_dna</a></tt> </tt>
<a name="L1077"></a><tt class="py-lineno">1077</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-359" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-359', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"U"</tt><tt class="py-op">,</tt> <tt class="py-string">"T"</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"u"</tt><tt class="py-op">,</tt> <tt class="py-string">"t"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-360" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-360', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1078"></a><tt class="py-lineno">1078</tt>  <tt class="py-line"> </tt>
<a name="Seq.translate"></a><div id="Seq.translate-def"><a name="L1079"></a><tt class="py-lineno">1079</tt> <a class="py-toggle" href="#" id="Seq.translate-toggle" onclick="return toggle('Seq.translate');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#translate">translate</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">table</tt><tt class="py-op">=</tt><tt class="py-string">"Standard"</tt><tt class="py-op">,</tt> <tt class="py-param">stop_symbol</tt><tt class="py-op">=</tt><tt class="py-string">"*"</tt><tt class="py-op">,</tt> <tt class="py-param">to_stop</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> </tt>
<a name="L1080"></a><tt class="py-lineno">1080</tt>  <tt class="py-line">                  <tt class="py-param">cds</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> <tt class="py-param">gap</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.translate-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.translate-expanded"><a name="L1081"></a><tt class="py-lineno">1081</tt>  <tt class="py-line">        <tt class="py-docstring">"""Turn a nucleotide sequence into a protein sequence by creating a new Seq object.</tt> </tt>
<a name="L1082"></a><tt class="py-lineno">1082</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1083"></a><tt class="py-lineno">1083</tt>  <tt class="py-line"><tt class="py-docstring">        This method will translate DNA or RNA sequences, and those with a</tt> </tt>
<a name="L1084"></a><tt class="py-lineno">1084</tt>  <tt class="py-line"><tt class="py-docstring">        nucleotide or generic alphabet.  Trying to translate a protein</tt> </tt>
<a name="L1085"></a><tt class="py-lineno">1085</tt>  <tt class="py-line"><tt class="py-docstring">        sequence raises an exception.</tt> </tt>
<a name="L1086"></a><tt class="py-lineno">1086</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1087"></a><tt class="py-lineno">1087</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L1088"></a><tt class="py-lineno">1088</tt>  <tt class="py-line"><tt class="py-docstring">         - table - Which codon table to use?  This can be either a name</tt> </tt>
<a name="L1089"></a><tt class="py-lineno">1089</tt>  <tt class="py-line"><tt class="py-docstring">           (string), an NCBI identifier (integer), or a CodonTable</tt> </tt>
<a name="L1090"></a><tt class="py-lineno">1090</tt>  <tt class="py-line"><tt class="py-docstring">           object (useful for non-standard genetic codes).  This</tt> </tt>
<a name="L1091"></a><tt class="py-lineno">1091</tt>  <tt class="py-line"><tt class="py-docstring">           defaults to the "Standard" table.</tt> </tt>
<a name="L1092"></a><tt class="py-lineno">1092</tt>  <tt class="py-line"><tt class="py-docstring">         - stop_symbol - Single character string, what to use for</tt> </tt>
<a name="L1093"></a><tt class="py-lineno">1093</tt>  <tt class="py-line"><tt class="py-docstring">           terminators.  This defaults to the asterisk, "*".</tt> </tt>
<a name="L1094"></a><tt class="py-lineno">1094</tt>  <tt class="py-line"><tt class="py-docstring">         - to_stop - Boolean, defaults to False meaning do a full</tt> </tt>
<a name="L1095"></a><tt class="py-lineno">1095</tt>  <tt class="py-line"><tt class="py-docstring">           translation continuing on past any stop codons (translated as the</tt> </tt>
<a name="L1096"></a><tt class="py-lineno">1096</tt>  <tt class="py-line"><tt class="py-docstring">           specified stop_symbol).  If True, translation is terminated at</tt> </tt>
<a name="L1097"></a><tt class="py-lineno">1097</tt>  <tt class="py-line"><tt class="py-docstring">           the first in frame stop codon (and the stop_symbol is not</tt> </tt>
<a name="L1098"></a><tt class="py-lineno">1098</tt>  <tt class="py-line"><tt class="py-docstring">           appended to the returned protein sequence).</tt> </tt>
<a name="L1099"></a><tt class="py-lineno">1099</tt>  <tt class="py-line"><tt class="py-docstring">         - cds - Boolean, indicates this is a complete CDS.  If True,</tt> </tt>
<a name="L1100"></a><tt class="py-lineno">1100</tt>  <tt class="py-line"><tt class="py-docstring">           this checks the sequence starts with a valid alternative start</tt> </tt>
<a name="L1101"></a><tt class="py-lineno">1101</tt>  <tt class="py-line"><tt class="py-docstring">           codon (which will be translated as methionine, M), that the</tt> </tt>
<a name="L1102"></a><tt class="py-lineno">1102</tt>  <tt class="py-line"><tt class="py-docstring">           sequence length is a multiple of three, and that there is a</tt> </tt>
<a name="L1103"></a><tt class="py-lineno">1103</tt>  <tt class="py-line"><tt class="py-docstring">           single in frame stop codon at the end (this will be excluded</tt> </tt>
<a name="L1104"></a><tt class="py-lineno">1104</tt>  <tt class="py-line"><tt class="py-docstring">           from the protein sequence, regardless of the to_stop option).</tt> </tt>
<a name="L1105"></a><tt class="py-lineno">1105</tt>  <tt class="py-line"><tt class="py-docstring">           If these tests fail, an exception is raised.</tt> </tt>
<a name="L1106"></a><tt class="py-lineno">1106</tt>  <tt class="py-line"><tt class="py-docstring">         - gap - Single character string to denote symbol used for gaps.</tt> </tt>
<a name="L1107"></a><tt class="py-lineno">1107</tt>  <tt class="py-line"><tt class="py-docstring">           It will try to guess the gap character from the alphabet.</tt> </tt>
<a name="L1108"></a><tt class="py-lineno">1108</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1109"></a><tt class="py-lineno">1109</tt>  <tt class="py-line"><tt class="py-docstring">        e.g. Using the standard table:</tt> </tt>
<a name="L1110"></a><tt class="py-lineno">1110</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1111"></a><tt class="py-lineno">1111</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; coding_dna = Seq("GTGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG")</tt> </tt>
<a name="L1112"></a><tt class="py-lineno">1112</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; coding_dna.translate()</tt> </tt>
<a name="L1113"></a><tt class="py-lineno">1113</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('VAIVMGR*KGAR*', HasStopCodon(ExtendedIUPACProtein(), '*'))</tt> </tt>
<a name="L1114"></a><tt class="py-lineno">1114</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; coding_dna.translate(stop_symbol="@")</tt> </tt>
<a name="L1115"></a><tt class="py-lineno">1115</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('VAIVMGR@KGAR@', HasStopCodon(ExtendedIUPACProtein(), '@'))</tt> </tt>
<a name="L1116"></a><tt class="py-lineno">1116</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; coding_dna.translate(to_stop=True)</tt> </tt>
<a name="L1117"></a><tt class="py-lineno">1117</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('VAIVMGR', ExtendedIUPACProtein())</tt> </tt>
<a name="L1118"></a><tt class="py-lineno">1118</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1119"></a><tt class="py-lineno">1119</tt>  <tt class="py-line"><tt class="py-docstring">        Now using NCBI table 2, where TGA is not a stop codon:</tt> </tt>
<a name="L1120"></a><tt class="py-lineno">1120</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1121"></a><tt class="py-lineno">1121</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; coding_dna.translate(table=2)</tt> </tt>
<a name="L1122"></a><tt class="py-lineno">1122</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('VAIVMGRWKGAR*', HasStopCodon(ExtendedIUPACProtein(), '*'))</tt> </tt>
<a name="L1123"></a><tt class="py-lineno">1123</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; coding_dna.translate(table=2, to_stop=True)</tt> </tt>
<a name="L1124"></a><tt class="py-lineno">1124</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('VAIVMGRWKGAR', ExtendedIUPACProtein())</tt> </tt>
<a name="L1125"></a><tt class="py-lineno">1125</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1126"></a><tt class="py-lineno">1126</tt>  <tt class="py-line"><tt class="py-docstring">        In fact, GTG is an alternative start codon under NCBI table 2, meaning</tt> </tt>
<a name="L1127"></a><tt class="py-lineno">1127</tt>  <tt class="py-line"><tt class="py-docstring">        this sequence could be a complete CDS:</tt> </tt>
<a name="L1128"></a><tt class="py-lineno">1128</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1129"></a><tt class="py-lineno">1129</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; coding_dna.translate(table=2, cds=True)</tt> </tt>
<a name="L1130"></a><tt class="py-lineno">1130</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('MAIVMGRWKGAR', ExtendedIUPACProtein())</tt> </tt>
<a name="L1131"></a><tt class="py-lineno">1131</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1132"></a><tt class="py-lineno">1132</tt>  <tt class="py-line"><tt class="py-docstring">        It isn't a valid CDS under NCBI table 1, due to both the start codon</tt> </tt>
<a name="L1133"></a><tt class="py-lineno">1133</tt>  <tt class="py-line"><tt class="py-docstring">        and also the in frame stop codons:</tt> </tt>
<a name="L1134"></a><tt class="py-lineno">1134</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1135"></a><tt class="py-lineno">1135</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; coding_dna.translate(table=1, cds=True)</tt> </tt>
<a name="L1136"></a><tt class="py-lineno">1136</tt>  <tt class="py-line"><tt class="py-docstring">        Traceback (most recent call last):</tt> </tt>
<a name="L1137"></a><tt class="py-lineno">1137</tt>  <tt class="py-line"><tt class="py-docstring">            ...</tt> </tt>
<a name="L1138"></a><tt class="py-lineno">1138</tt>  <tt class="py-line"><tt class="py-docstring">        Bio.Data.CodonTable.TranslationError: First codon 'GTG' is not a start codon</tt> </tt>
<a name="L1139"></a><tt class="py-lineno">1139</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1140"></a><tt class="py-lineno">1140</tt>  <tt class="py-line"><tt class="py-docstring">        If the sequence has no in-frame stop codon, then the to_stop argument</tt> </tt>
<a name="L1141"></a><tt class="py-lineno">1141</tt>  <tt class="py-line"><tt class="py-docstring">        has no effect:</tt> </tt>
<a name="L1142"></a><tt class="py-lineno">1142</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1143"></a><tt class="py-lineno">1143</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; coding_dna2 = Seq("TTGGCCATTGTAATGGGCCGC")</tt> </tt>
<a name="L1144"></a><tt class="py-lineno">1144</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; coding_dna2.translate()</tt> </tt>
<a name="L1145"></a><tt class="py-lineno">1145</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('LAIVMGR', ExtendedIUPACProtein())</tt> </tt>
<a name="L1146"></a><tt class="py-lineno">1146</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; coding_dna2.translate(to_stop=True)</tt> </tt>
<a name="L1147"></a><tt class="py-lineno">1147</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('LAIVMGR', ExtendedIUPACProtein())</tt> </tt>
<a name="L1148"></a><tt class="py-lineno">1148</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1149"></a><tt class="py-lineno">1149</tt>  <tt class="py-line"><tt class="py-docstring">        When translating gapped sequences, the gap character is inferred from</tt> </tt>
<a name="L1150"></a><tt class="py-lineno">1150</tt>  <tt class="py-line"><tt class="py-docstring">        the alphabet:</tt> </tt>
<a name="L1151"></a><tt class="py-lineno">1151</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1152"></a><tt class="py-lineno">1152</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import Gapped</tt> </tt>
<a name="L1153"></a><tt class="py-lineno">1153</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; coding_dna3 = Seq("GTG---GCCATT", Gapped(IUPAC.unambiguous_dna))</tt> </tt>
<a name="L1154"></a><tt class="py-lineno">1154</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; coding_dna3.translate()</tt> </tt>
<a name="L1155"></a><tt class="py-lineno">1155</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('V-AI', Gapped(ExtendedIUPACProtein(), '-'))</tt> </tt>
<a name="L1156"></a><tt class="py-lineno">1156</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1157"></a><tt class="py-lineno">1157</tt>  <tt class="py-line"><tt class="py-docstring">        It is possible to pass the gap character when the alphabet is missing:</tt> </tt>
<a name="L1158"></a><tt class="py-lineno">1158</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1159"></a><tt class="py-lineno">1159</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; coding_dna4 = Seq("GTG---GCCATT")</tt> </tt>
<a name="L1160"></a><tt class="py-lineno">1160</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; coding_dna4.translate(gap='-')</tt> </tt>
<a name="L1161"></a><tt class="py-lineno">1161</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('V-AI', Gapped(ExtendedIUPACProtein(), '-'))</tt> </tt>
<a name="L1162"></a><tt class="py-lineno">1162</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1163"></a><tt class="py-lineno">1163</tt>  <tt class="py-line"><tt class="py-docstring">        NOTE - Ambiguous codons like "TAN" or "NNN" could be an amino acid</tt> </tt>
<a name="L1164"></a><tt class="py-lineno">1164</tt>  <tt class="py-line"><tt class="py-docstring">        or a stop codon.  These are translated as "X".  Any invalid codon</tt> </tt>
<a name="L1165"></a><tt class="py-lineno">1165</tt>  <tt class="py-line"><tt class="py-docstring">        (e.g. "TA?" or "T-A") will throw a TranslationError.</tt> </tt>
<a name="L1166"></a><tt class="py-lineno">1166</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1167"></a><tt class="py-lineno">1167</tt>  <tt class="py-line"><tt class="py-docstring">        NOTE - This does NOT behave like the python string's translate</tt> </tt>
<a name="L1168"></a><tt class="py-lineno">1168</tt>  <tt class="py-line"><tt class="py-docstring">        method.  For that use str(my_seq).translate(...) instead.</tt> </tt>
<a name="L1169"></a><tt class="py-lineno">1169</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1170"></a><tt class="py-lineno">1170</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">table</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">table</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">256</tt><tt class="py-op">:</tt> </tt>
<a name="L1171"></a><tt class="py-lineno">1171</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"The Seq object translate method DOES NOT take "</tt> </tt>
<a name="L1172"></a><tt class="py-lineno">1172</tt>  <tt class="py-line">                             <tt class="py-string">"a 256 character string mapping table like "</tt> </tt>
<a name="L1173"></a><tt class="py-lineno">1173</tt>  <tt class="py-line">                             <tt class="py-string">"the python string object's translate method. "</tt> </tt>
<a name="L1174"></a><tt class="py-lineno">1174</tt>  <tt class="py-line">                             <tt class="py-string">"Use str(my_seq).translate(...) instead."</tt><tt class="py-op">)</tt> </tt>
<a name="L1175"></a><tt class="py-lineno">1175</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-361" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-361', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-362" class="py-name"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-362', '_get_base_alphabet', 'link-297');">_get_base_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-363" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-363', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1176"></a><tt class="py-lineno">1176</tt>  <tt class="py-line">                      <tt id="link-364" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-364', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-365" class="py-name"><a title="Bio.Alphabet.ProteinAlphabet" class="py-name" href="#" onclick="return doclink('link-365', 'ProteinAlphabet', 'link-300');">ProteinAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1177"></a><tt class="py-lineno">1177</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Proteins cannot be translated!"</tt><tt class="py-op">)</tt> </tt>
<a name="L1178"></a><tt class="py-lineno">1178</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1179"></a><tt class="py-lineno">1179</tt>  <tt class="py-line">            <tt class="py-name">table_id</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">table</tt><tt class="py-op">)</tt> </tt>
<a name="L1180"></a><tt class="py-lineno">1180</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L1181"></a><tt class="py-lineno">1181</tt>  <tt class="py-line">            <tt class="py-comment"># Assume its a table name</tt> </tt>
<a name="L1182"></a><tt class="py-lineno">1182</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-366" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-366', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-op">==</tt> <tt id="link-367" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-367', 'IUPAC', 'link-11');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-368" class="py-name"><a title="Bio.Alphabet.IUPAC.unambiguous_dna" class="py-name" href="#" onclick="return doclink('link-368', 'unambiguous_dna', 'link-322');">unambiguous_dna</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1183"></a><tt class="py-lineno">1183</tt>  <tt class="py-line">                <tt class="py-comment"># Will use standard IUPAC protein alphabet, no need for X</tt> </tt>
<a name="L1184"></a><tt class="py-lineno">1184</tt>  <tt class="py-line">                <tt class="py-name">codon_table</tt> <tt class="py-op">=</tt> <tt id="link-369" class="py-name"><a title="Bio.Data.CodonTable
Bio.Data.CodonTable.CodonTable" class="py-name" href="#" onclick="return doclink('link-369', 'CodonTable', 'link-27');">CodonTable</a></tt><tt class="py-op">.</tt><tt id="link-370" class="py-name" targets="Variable Bio.Data.CodonTable.unambiguous_dna_by_name=Bio.Data.CodonTable-module.html#unambiguous_dna_by_name"><a title="Bio.Data.CodonTable.unambiguous_dna_by_name" class="py-name" href="#" onclick="return doclink('link-370', 'unambiguous_dna_by_name', 'link-370');">unambiguous_dna_by_name</a></tt><tt class="py-op">[</tt><tt class="py-name">table</tt><tt class="py-op">]</tt> </tt>
<a name="L1185"></a><tt class="py-lineno">1185</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-371" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-371', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-op">==</tt> <tt id="link-372" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-372', 'IUPAC', 'link-11');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-373" class="py-name"><a title="Bio.Alphabet.IUPAC.unambiguous_rna" class="py-name" href="#" onclick="return doclink('link-373', 'unambiguous_rna', 'link-325');">unambiguous_rna</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1186"></a><tt class="py-lineno">1186</tt>  <tt class="py-line">                <tt class="py-comment"># Will use standard IUPAC protein alphabet, no need for X</tt> </tt>
<a name="L1187"></a><tt class="py-lineno">1187</tt>  <tt class="py-line">                <tt class="py-name">codon_table</tt> <tt class="py-op">=</tt> <tt id="link-374" class="py-name"><a title="Bio.Data.CodonTable
Bio.Data.CodonTable.CodonTable" class="py-name" href="#" onclick="return doclink('link-374', 'CodonTable', 'link-27');">CodonTable</a></tt><tt class="py-op">.</tt><tt id="link-375" class="py-name" targets="Variable Bio.Data.CodonTable.unambiguous_rna_by_name=Bio.Data.CodonTable-module.html#unambiguous_rna_by_name"><a title="Bio.Data.CodonTable.unambiguous_rna_by_name" class="py-name" href="#" onclick="return doclink('link-375', 'unambiguous_rna_by_name', 'link-375');">unambiguous_rna_by_name</a></tt><tt class="py-op">[</tt><tt class="py-name">table</tt><tt class="py-op">]</tt> </tt>
<a name="L1188"></a><tt class="py-lineno">1188</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1189"></a><tt class="py-lineno">1189</tt>  <tt class="py-line">                <tt class="py-comment"># This will use the extended IUPAC protein alphabet with X etc.</tt> </tt>
<a name="L1190"></a><tt class="py-lineno">1190</tt>  <tt class="py-line">                <tt class="py-comment"># The same table can be used for RNA or DNA (we use this for</tt> </tt>
<a name="L1191"></a><tt class="py-lineno">1191</tt>  <tt class="py-line">                <tt class="py-comment"># translating strings).</tt> </tt>
<a name="L1192"></a><tt class="py-lineno">1192</tt>  <tt class="py-line">                <tt class="py-name">codon_table</tt> <tt class="py-op">=</tt> <tt id="link-376" class="py-name"><a title="Bio.Data.CodonTable
Bio.Data.CodonTable.CodonTable" class="py-name" href="#" onclick="return doclink('link-376', 'CodonTable', 'link-27');">CodonTable</a></tt><tt class="py-op">.</tt><tt id="link-377" class="py-name" targets="Variable Bio.Data.CodonTable.ambiguous_generic_by_name=Bio.Data.CodonTable-module.html#ambiguous_generic_by_name"><a title="Bio.Data.CodonTable.ambiguous_generic_by_name" class="py-name" href="#" onclick="return doclink('link-377', 'ambiguous_generic_by_name', 'link-377');">ambiguous_generic_by_name</a></tt><tt class="py-op">[</tt><tt class="py-name">table</tt><tt class="py-op">]</tt> </tt>
<a name="L1193"></a><tt class="py-lineno">1193</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-op">(</tt><tt class="py-name">AttributeError</tt><tt class="py-op">,</tt> <tt class="py-name">TypeError</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1194"></a><tt class="py-lineno">1194</tt>  <tt class="py-line">            <tt class="py-comment"># Assume its a CodonTable object</tt> </tt>
<a name="L1195"></a><tt class="py-lineno">1195</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">table</tt><tt class="py-op">,</tt> <tt id="link-378" class="py-name"><a title="Bio.Data.CodonTable
Bio.Data.CodonTable.CodonTable" class="py-name" href="#" onclick="return doclink('link-378', 'CodonTable', 'link-27');">CodonTable</a></tt><tt class="py-op">.</tt><tt id="link-379" class="py-name"><a title="Bio.Data.CodonTable
Bio.Data.CodonTable.CodonTable" class="py-name" href="#" onclick="return doclink('link-379', 'CodonTable', 'link-27');">CodonTable</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1196"></a><tt class="py-lineno">1196</tt>  <tt class="py-line">                <tt class="py-name">codon_table</tt> <tt class="py-op">=</tt> <tt class="py-name">table</tt> </tt>
<a name="L1197"></a><tt class="py-lineno">1197</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1198"></a><tt class="py-lineno">1198</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Bad table argument"</tt><tt class="py-op">)</tt> </tt>
<a name="L1199"></a><tt class="py-lineno">1199</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1200"></a><tt class="py-lineno">1200</tt>  <tt class="py-line">            <tt class="py-comment"># Assume its a table ID</tt> </tt>
<a name="L1201"></a><tt class="py-lineno">1201</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-380" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-380', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-op">==</tt> <tt id="link-381" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-381', 'IUPAC', 'link-11');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-382" class="py-name"><a title="Bio.Alphabet.IUPAC.unambiguous_dna" class="py-name" href="#" onclick="return doclink('link-382', 'unambiguous_dna', 'link-322');">unambiguous_dna</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1202"></a><tt class="py-lineno">1202</tt>  <tt class="py-line">                <tt class="py-comment"># Will use standard IUPAC protein alphabet, no need for X</tt> </tt>
<a name="L1203"></a><tt class="py-lineno">1203</tt>  <tt class="py-line">                <tt class="py-name">codon_table</tt> <tt class="py-op">=</tt> <tt id="link-383" class="py-name"><a title="Bio.Data.CodonTable
Bio.Data.CodonTable.CodonTable" class="py-name" href="#" onclick="return doclink('link-383', 'CodonTable', 'link-27');">CodonTable</a></tt><tt class="py-op">.</tt><tt id="link-384" class="py-name" targets="Variable Bio.Data.CodonTable.unambiguous_dna_by_id=Bio.Data.CodonTable-module.html#unambiguous_dna_by_id"><a title="Bio.Data.CodonTable.unambiguous_dna_by_id" class="py-name" href="#" onclick="return doclink('link-384', 'unambiguous_dna_by_id', 'link-384');">unambiguous_dna_by_id</a></tt><tt class="py-op">[</tt><tt class="py-name">table_id</tt><tt class="py-op">]</tt> </tt>
<a name="L1204"></a><tt class="py-lineno">1204</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-385" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-385', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-op">==</tt> <tt id="link-386" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-386', 'IUPAC', 'link-11');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-387" class="py-name"><a title="Bio.Alphabet.IUPAC.unambiguous_rna" class="py-name" href="#" onclick="return doclink('link-387', 'unambiguous_rna', 'link-325');">unambiguous_rna</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1205"></a><tt class="py-lineno">1205</tt>  <tt class="py-line">                <tt class="py-comment"># Will use standard IUPAC protein alphabet, no need for X</tt> </tt>
<a name="L1206"></a><tt class="py-lineno">1206</tt>  <tt class="py-line">                <tt class="py-name">codon_table</tt> <tt class="py-op">=</tt> <tt id="link-388" class="py-name"><a title="Bio.Data.CodonTable
Bio.Data.CodonTable.CodonTable" class="py-name" href="#" onclick="return doclink('link-388', 'CodonTable', 'link-27');">CodonTable</a></tt><tt class="py-op">.</tt><tt id="link-389" class="py-name" targets="Variable Bio.Data.CodonTable.unambiguous_rna_by_id=Bio.Data.CodonTable-module.html#unambiguous_rna_by_id"><a title="Bio.Data.CodonTable.unambiguous_rna_by_id" class="py-name" href="#" onclick="return doclink('link-389', 'unambiguous_rna_by_id', 'link-389');">unambiguous_rna_by_id</a></tt><tt class="py-op">[</tt><tt class="py-name">table_id</tt><tt class="py-op">]</tt> </tt>
<a name="L1207"></a><tt class="py-lineno">1207</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1208"></a><tt class="py-lineno">1208</tt>  <tt class="py-line">                <tt class="py-comment"># This will use the extended IUPAC protein alphabet with X etc.</tt> </tt>
<a name="L1209"></a><tt class="py-lineno">1209</tt>  <tt class="py-line">                <tt class="py-comment"># The same table can be used for RNA or DNA (we use this for</tt> </tt>
<a name="L1210"></a><tt class="py-lineno">1210</tt>  <tt class="py-line">                <tt class="py-comment"># translating strings).</tt> </tt>
<a name="L1211"></a><tt class="py-lineno">1211</tt>  <tt class="py-line">                <tt class="py-name">codon_table</tt> <tt class="py-op">=</tt> <tt id="link-390" class="py-name"><a title="Bio.Data.CodonTable
Bio.Data.CodonTable.CodonTable" class="py-name" href="#" onclick="return doclink('link-390', 'CodonTable', 'link-27');">CodonTable</a></tt><tt class="py-op">.</tt><tt id="link-391" class="py-name" targets="Variable Bio.Data.CodonTable.ambiguous_generic_by_id=Bio.Data.CodonTable-module.html#ambiguous_generic_by_id"><a title="Bio.Data.CodonTable.ambiguous_generic_by_id" class="py-name" href="#" onclick="return doclink('link-391', 'ambiguous_generic_by_id', 'link-391');">ambiguous_generic_by_id</a></tt><tt class="py-op">[</tt><tt class="py-name">table_id</tt><tt class="py-op">]</tt> </tt>
<a name="L1212"></a><tt class="py-lineno">1212</tt>  <tt class="py-line"> </tt>
<a name="L1213"></a><tt class="py-lineno">1213</tt>  <tt class="py-line">        <tt class="py-comment"># Deal with gaps for translation</tt> </tt>
<a name="L1214"></a><tt class="py-lineno">1214</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-392" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-392', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-string">"gap_char"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1215"></a><tt class="py-lineno">1215</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">gap</tt><tt class="py-op">:</tt> </tt>
<a name="L1216"></a><tt class="py-lineno">1216</tt>  <tt class="py-line">                <tt class="py-name">gap</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-393" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-393', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">.</tt><tt class="py-name">gap_char</tt> </tt>
<a name="L1217"></a><tt class="py-lineno">1217</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">gap</tt> <tt class="py-op">!=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-394" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-394', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">.</tt><tt class="py-name">gap_char</tt><tt class="py-op">:</tt> </tt>
<a name="L1218"></a><tt class="py-lineno">1218</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt> </tt>
<a name="L1219"></a><tt class="py-lineno">1219</tt>  <tt class="py-line">                    <tt class="py-string">"Gap {0!r} does not match {1!r} from alphabet"</tt><tt class="py-op">.</tt><tt id="link-395" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-395', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1220"></a><tt class="py-lineno">1220</tt>  <tt class="py-line">                        <tt class="py-name">gap</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-396" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-396', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">.</tt><tt class="py-name">gap_char</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1221"></a><tt class="py-lineno">1221</tt>  <tt class="py-line"> </tt>
<a name="L1222"></a><tt class="py-lineno">1222</tt>  <tt class="py-line">        <tt id="link-397" class="py-name" targets="Variable Bio.Alphabet.IUPAC.protein=Bio.Alphabet.IUPAC-module.html#protein"><a title="Bio.Alphabet.IUPAC.protein" class="py-name" href="#" onclick="return doclink('link-397', 'protein', 'link-397');">protein</a></tt> <tt class="py-op">=</tt> <tt id="link-398" class="py-name" targets="Function Bio.Seq._translate_str()=Bio.Seq-module.html#_translate_str"><a title="Bio.Seq._translate_str" class="py-name" href="#" onclick="return doclink('link-398', '_translate_str', 'link-398');">_translate_str</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">codon_table</tt><tt class="py-op">,</tt> <tt class="py-name">stop_symbol</tt><tt class="py-op">,</tt> <tt class="py-name">to_stop</tt><tt class="py-op">,</tt> </tt>
<a name="L1223"></a><tt class="py-lineno">1223</tt>  <tt class="py-line">                                 <tt class="py-name">cds</tt><tt class="py-op">,</tt> <tt class="py-name">gap</tt><tt class="py-op">=</tt><tt class="py-name">gap</tt><tt class="py-op">)</tt> </tt>
<a name="L1224"></a><tt class="py-lineno">1224</tt>  <tt class="py-line"> </tt>
<a name="L1225"></a><tt class="py-lineno">1225</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">gap</tt> <tt class="py-keyword">and</tt> <tt class="py-name">gap</tt> <tt class="py-keyword">in</tt> <tt id="link-399" class="py-name"><a title="Bio.Alphabet.IUPAC.protein" class="py-name" href="#" onclick="return doclink('link-399', 'protein', 'link-397');">protein</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1226"></a><tt class="py-lineno">1226</tt>  <tt class="py-line">            <tt id="link-400" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-400', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-401" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-401', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-402" class="py-name" targets="Class Bio.Alphabet.Gapped=Bio.Alphabet.Gapped-class.html"><a title="Bio.Alphabet.Gapped" class="py-name" href="#" onclick="return doclink('link-402', 'Gapped', 'link-402');">Gapped</a></tt><tt class="py-op">(</tt><tt class="py-name">codon_table</tt><tt class="py-op">.</tt><tt id="link-403" class="py-name" targets="Variable Bio.Data.CodonTable.CodonTable.protein_alphabet=Bio.Data.CodonTable.CodonTable-class.html#protein_alphabet,Variable Bio.Data.CodonTable.NCBICodonTable.protein_alphabet=Bio.Data.CodonTable.NCBICodonTable-class.html#protein_alphabet,Variable Bio.Phylo.TreeConstruction.DistanceCalculator.protein_alphabet=Bio.Phylo.TreeConstruction.DistanceCalculator-class.html#protein_alphabet"><a title="Bio.Data.CodonTable.CodonTable.protein_alphabet
Bio.Data.CodonTable.NCBICodonTable.protein_alphabet
Bio.Phylo.TreeConstruction.DistanceCalculator.protein_alphabet" class="py-name" href="#" onclick="return doclink('link-403', 'protein_alphabet', 'link-403');">protein_alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-name">gap</tt><tt class="py-op">)</tt> </tt>
<a name="L1227"></a><tt class="py-lineno">1227</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1228"></a><tt class="py-lineno">1228</tt>  <tt class="py-line">            <tt id="link-404" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-404', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-op">=</tt> <tt class="py-name">codon_table</tt><tt class="py-op">.</tt><tt id="link-405" class="py-name"><a title="Bio.Data.CodonTable.CodonTable.protein_alphabet
Bio.Data.CodonTable.NCBICodonTable.protein_alphabet
Bio.Phylo.TreeConstruction.DistanceCalculator.protein_alphabet" class="py-name" href="#" onclick="return doclink('link-405', 'protein_alphabet', 'link-403');">protein_alphabet</a></tt> </tt>
<a name="L1229"></a><tt class="py-lineno">1229</tt>  <tt class="py-line"> </tt>
<a name="L1230"></a><tt class="py-lineno">1230</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">stop_symbol</tt> <tt class="py-keyword">in</tt> <tt id="link-406" class="py-name"><a title="Bio.Alphabet.IUPAC.protein" class="py-name" href="#" onclick="return doclink('link-406', 'protein', 'link-397');">protein</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1231"></a><tt class="py-lineno">1231</tt>  <tt class="py-line">            <tt id="link-407" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-407', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-408" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-408', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-409" class="py-name" targets="Class Bio.Alphabet.HasStopCodon=Bio.Alphabet.HasStopCodon-class.html"><a title="Bio.Alphabet.HasStopCodon" class="py-name" href="#" onclick="return doclink('link-409', 'HasStopCodon', 'link-409');">HasStopCodon</a></tt><tt class="py-op">(</tt><tt id="link-410" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-410', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-name">stop_symbol</tt><tt class="py-op">)</tt> </tt>
<a name="L1232"></a><tt class="py-lineno">1232</tt>  <tt class="py-line"> </tt>
<a name="L1233"></a><tt class="py-lineno">1233</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-411" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-411', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">(</tt><tt id="link-412" class="py-name"><a title="Bio.Alphabet.IUPAC.protein" class="py-name" href="#" onclick="return doclink('link-412', 'protein', 'link-397');">protein</a></tt><tt class="py-op">,</tt> <tt id="link-413" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-413', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1234"></a><tt class="py-lineno">1234</tt>  <tt class="py-line"> </tt>
<a name="Seq.ungap"></a><div id="Seq.ungap-def"><a name="L1235"></a><tt class="py-lineno">1235</tt> <a class="py-toggle" href="#" id="Seq.ungap-toggle" onclick="return toggle('Seq.ungap');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#ungap">ungap</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">gap</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.ungap-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.ungap-expanded"><a name="L1236"></a><tt class="py-lineno">1236</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a copy of the sequence without the gap character(s).</tt> </tt>
<a name="L1237"></a><tt class="py-lineno">1237</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1238"></a><tt class="py-lineno">1238</tt>  <tt class="py-line"><tt class="py-docstring">        The gap character can be specified in two ways - either as an explicit</tt> </tt>
<a name="L1239"></a><tt class="py-lineno">1239</tt>  <tt class="py-line"><tt class="py-docstring">        argument, or via the sequence's alphabet. For example:</tt> </tt>
<a name="L1240"></a><tt class="py-lineno">1240</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1241"></a><tt class="py-lineno">1241</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L1242"></a><tt class="py-lineno">1242</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_dna</tt> </tt>
<a name="L1243"></a><tt class="py-lineno">1243</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna = Seq("-ATA--TGAAAT-TTGAAAA", generic_dna)</tt> </tt>
<a name="L1244"></a><tt class="py-lineno">1244</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna</tt> </tt>
<a name="L1245"></a><tt class="py-lineno">1245</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('-ATA--TGAAAT-TTGAAAA', DNAAlphabet())</tt> </tt>
<a name="L1246"></a><tt class="py-lineno">1246</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna.ungap("-")</tt> </tt>
<a name="L1247"></a><tt class="py-lineno">1247</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('ATATGAAATTTGAAAA', DNAAlphabet())</tt> </tt>
<a name="L1248"></a><tt class="py-lineno">1248</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1249"></a><tt class="py-lineno">1249</tt>  <tt class="py-line"><tt class="py-docstring">        If the gap character is not given as an argument, it will be taken from</tt> </tt>
<a name="L1250"></a><tt class="py-lineno">1250</tt>  <tt class="py-line"><tt class="py-docstring">        the sequence's alphabet (if defined). Notice that the returned</tt> </tt>
<a name="L1251"></a><tt class="py-lineno">1251</tt>  <tt class="py-line"><tt class="py-docstring">        sequence's alphabet is adjusted since it no longer requires a gapped</tt> </tt>
<a name="L1252"></a><tt class="py-lineno">1252</tt>  <tt class="py-line"><tt class="py-docstring">        alphabet:</tt> </tt>
<a name="L1253"></a><tt class="py-lineno">1253</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1254"></a><tt class="py-lineno">1254</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L1255"></a><tt class="py-lineno">1255</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import IUPAC, Gapped, HasStopCodon</tt> </tt>
<a name="L1256"></a><tt class="py-lineno">1256</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_pro = Seq("MVVLE=AD*", HasStopCodon(Gapped(IUPAC.protein, "=")))</tt> </tt>
<a name="L1257"></a><tt class="py-lineno">1257</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_pro</tt> </tt>
<a name="L1258"></a><tt class="py-lineno">1258</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('MVVLE=AD*', HasStopCodon(Gapped(IUPACProtein(), '='), '*'))</tt> </tt>
<a name="L1259"></a><tt class="py-lineno">1259</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_pro.ungap()</tt> </tt>
<a name="L1260"></a><tt class="py-lineno">1260</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('MVVLEAD*', HasStopCodon(IUPACProtein(), '*'))</tt> </tt>
<a name="L1261"></a><tt class="py-lineno">1261</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1262"></a><tt class="py-lineno">1262</tt>  <tt class="py-line"><tt class="py-docstring">        Or, with a simpler gapped DNA example:</tt> </tt>
<a name="L1263"></a><tt class="py-lineno">1263</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1264"></a><tt class="py-lineno">1264</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L1265"></a><tt class="py-lineno">1265</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import IUPAC, Gapped</tt> </tt>
<a name="L1266"></a><tt class="py-lineno">1266</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq = Seq("CGGGTAG=AAAAAA", Gapped(IUPAC.unambiguous_dna, "="))</tt> </tt>
<a name="L1267"></a><tt class="py-lineno">1267</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L1268"></a><tt class="py-lineno">1268</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('CGGGTAG=AAAAAA', Gapped(IUPACUnambiguousDNA(), '='))</tt> </tt>
<a name="L1269"></a><tt class="py-lineno">1269</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.ungap()</tt> </tt>
<a name="L1270"></a><tt class="py-lineno">1270</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('CGGGTAGAAAAAA', IUPACUnambiguousDNA())</tt> </tt>
<a name="L1271"></a><tt class="py-lineno">1271</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1272"></a><tt class="py-lineno">1272</tt>  <tt class="py-line"><tt class="py-docstring">        As long as it is consistent with the alphabet, although it is</tt> </tt>
<a name="L1273"></a><tt class="py-lineno">1273</tt>  <tt class="py-line"><tt class="py-docstring">        redundant, you can still supply the gap character as an argument to</tt> </tt>
<a name="L1274"></a><tt class="py-lineno">1274</tt>  <tt class="py-line"><tt class="py-docstring">        this method:</tt> </tt>
<a name="L1275"></a><tt class="py-lineno">1275</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1276"></a><tt class="py-lineno">1276</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L1277"></a><tt class="py-lineno">1277</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('CGGGTAG=AAAAAA', Gapped(IUPACUnambiguousDNA(), '='))</tt> </tt>
<a name="L1278"></a><tt class="py-lineno">1278</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.ungap("=")</tt> </tt>
<a name="L1279"></a><tt class="py-lineno">1279</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('CGGGTAGAAAAAA', IUPACUnambiguousDNA())</tt> </tt>
<a name="L1280"></a><tt class="py-lineno">1280</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1281"></a><tt class="py-lineno">1281</tt>  <tt class="py-line"><tt class="py-docstring">        However, if the gap character given as the argument disagrees with that</tt> </tt>
<a name="L1282"></a><tt class="py-lineno">1282</tt>  <tt class="py-line"><tt class="py-docstring">        declared in the alphabet, an exception is raised:</tt> </tt>
<a name="L1283"></a><tt class="py-lineno">1283</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1284"></a><tt class="py-lineno">1284</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L1285"></a><tt class="py-lineno">1285</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('CGGGTAG=AAAAAA', Gapped(IUPACUnambiguousDNA(), '='))</tt> </tt>
<a name="L1286"></a><tt class="py-lineno">1286</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.ungap("-")</tt> </tt>
<a name="L1287"></a><tt class="py-lineno">1287</tt>  <tt class="py-line"><tt class="py-docstring">        Traceback (most recent call last):</tt> </tt>
<a name="L1288"></a><tt class="py-lineno">1288</tt>  <tt class="py-line"><tt class="py-docstring">           ...</tt> </tt>
<a name="L1289"></a><tt class="py-lineno">1289</tt>  <tt class="py-line"><tt class="py-docstring">        ValueError: Gap '-' does not match '=' from alphabet</tt> </tt>
<a name="L1290"></a><tt class="py-lineno">1290</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1291"></a><tt class="py-lineno">1291</tt>  <tt class="py-line"><tt class="py-docstring">        Finally, if a gap character is not supplied, and the alphabet does not</tt> </tt>
<a name="L1292"></a><tt class="py-lineno">1292</tt>  <tt class="py-line"><tt class="py-docstring">        define one, an exception is raised:</tt> </tt>
<a name="L1293"></a><tt class="py-lineno">1293</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1294"></a><tt class="py-lineno">1294</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L1295"></a><tt class="py-lineno">1295</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_dna</tt> </tt>
<a name="L1296"></a><tt class="py-lineno">1296</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna = Seq("ATA--TGAAAT-TTGAAAA", generic_dna)</tt> </tt>
<a name="L1297"></a><tt class="py-lineno">1297</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna</tt> </tt>
<a name="L1298"></a><tt class="py-lineno">1298</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('ATA--TGAAAT-TTGAAAA', DNAAlphabet())</tt> </tt>
<a name="L1299"></a><tt class="py-lineno">1299</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna.ungap()</tt> </tt>
<a name="L1300"></a><tt class="py-lineno">1300</tt>  <tt class="py-line"><tt class="py-docstring">        Traceback (most recent call last):</tt> </tt>
<a name="L1301"></a><tt class="py-lineno">1301</tt>  <tt class="py-line"><tt class="py-docstring">           ...</tt> </tt>
<a name="L1302"></a><tt class="py-lineno">1302</tt>  <tt class="py-line"><tt class="py-docstring">        ValueError: Gap character not given and not defined in alphabet</tt> </tt>
<a name="L1303"></a><tt class="py-lineno">1303</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1304"></a><tt class="py-lineno">1304</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1305"></a><tt class="py-lineno">1305</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-414" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-414', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-string">"gap_char"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1306"></a><tt class="py-lineno">1306</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">gap</tt><tt class="py-op">:</tt> </tt>
<a name="L1307"></a><tt class="py-lineno">1307</tt>  <tt class="py-line">                <tt class="py-name">gap</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-415" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-415', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">.</tt><tt class="py-name">gap_char</tt> </tt>
<a name="L1308"></a><tt class="py-lineno">1308</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">gap</tt> <tt class="py-op">!=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-416" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-416', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">.</tt><tt class="py-name">gap_char</tt><tt class="py-op">:</tt> </tt>
<a name="L1309"></a><tt class="py-lineno">1309</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt> </tt>
<a name="L1310"></a><tt class="py-lineno">1310</tt>  <tt class="py-line">                    <tt class="py-string">"Gap {0!r} does not match {1!r} from alphabet"</tt><tt class="py-op">.</tt><tt id="link-417" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-417', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1311"></a><tt class="py-lineno">1311</tt>  <tt class="py-line">                        <tt class="py-name">gap</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-418" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-418', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">.</tt><tt class="py-name">gap_char</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1312"></a><tt class="py-lineno">1312</tt>  <tt class="py-line">            <tt class="py-name">alpha</tt> <tt class="py-op">=</tt> <tt id="link-419" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-419', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-420" class="py-name" targets="Function Bio.Alphabet._ungap()=Bio.Alphabet-module.html#_ungap"><a title="Bio.Alphabet._ungap" class="py-name" href="#" onclick="return doclink('link-420', '_ungap', 'link-420');">_ungap</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-421" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-421', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1313"></a><tt class="py-lineno">1313</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-keyword">not</tt> <tt class="py-name">gap</tt><tt class="py-op">:</tt> </tt>
<a name="L1314"></a><tt class="py-lineno">1314</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Gap character not given and not defined in "</tt> </tt>
<a name="L1315"></a><tt class="py-lineno">1315</tt>  <tt class="py-line">                             <tt class="py-string">"alphabet"</tt><tt class="py-op">)</tt> </tt>
<a name="L1316"></a><tt class="py-lineno">1316</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1317"></a><tt class="py-lineno">1317</tt>  <tt class="py-line">            <tt class="py-name">alpha</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-422" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-422', 'alphabet', 'link-44');">alphabet</a></tt>  <tt class="py-comment"># modify!</tt> </tt>
<a name="L1318"></a><tt class="py-lineno">1318</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">gap</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-number">1</tt> <tt class="py-keyword">or</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">gap</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1319"></a><tt class="py-lineno">1319</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected gap character, {0!r}"</tt><tt class="py-op">.</tt><tt id="link-423" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-423', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">gap</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1320"></a><tt class="py-lineno">1320</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-424" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-424', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-name">gap</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">alpha</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1321"></a><tt class="py-lineno">1321</tt>  <tt class="py-line"> </tt>
<a name="Seq.join"></a><div id="Seq.join-def"><a name="L1322"></a><tt class="py-lineno">1322</tt> <a class="py-toggle" href="#" id="Seq.join-toggle" onclick="return toggle('Seq.join');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.Seq-class.html#join">join</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Seq.join-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Seq.join-expanded"><a name="L1323"></a><tt class="py-lineno">1323</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a merge of the sequences in other, spaced by the sequence from self.</tt> </tt>
<a name="L1324"></a><tt class="py-lineno">1324</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1325"></a><tt class="py-lineno">1325</tt>  <tt class="py-line"><tt class="py-docstring">        Accepts all Seq objects and Strings as objects to be concatenated with the spacer</tt> </tt>
<a name="L1326"></a><tt class="py-lineno">1326</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1327"></a><tt class="py-lineno">1327</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; concatenated = Seq('NNNNN').join([Seq("AAA"), Seq("TTT"), Seq("PPP")])</tt> </tt>
<a name="L1328"></a><tt class="py-lineno">1328</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; concatenated</tt> </tt>
<a name="L1329"></a><tt class="py-lineno">1329</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('AAANNNNNTTTNNNNNPPP')</tt> </tt>
<a name="L1330"></a><tt class="py-lineno">1330</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1331"></a><tt class="py-lineno">1331</tt>  <tt class="py-line"><tt class="py-docstring">        Throws error if other is not an iterable and if objects inside of the iterable</tt> </tt>
<a name="L1332"></a><tt class="py-lineno">1332</tt>  <tt class="py-line"><tt class="py-docstring">        are not Seq or String objects</tt> </tt>
<a name="L1333"></a><tt class="py-lineno">1333</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1334"></a><tt class="py-lineno">1334</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-425" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-425', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">_Iterable</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># doesn't detect single strings</tt> </tt>
<a name="L1335"></a><tt class="py-lineno">1335</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Input must be an iterable"</tt><tt class="py-op">)</tt> </tt>
<a name="L1336"></a><tt class="py-lineno">1336</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-426" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-426', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1337"></a><tt class="py-lineno">1337</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Input must be an iterable"</tt><tt class="py-op">)</tt> </tt>
<a name="L1338"></a><tt class="py-lineno">1338</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt id="link-427" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-427', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-428" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-428', 'SeqRecord', 'link-177');">SeqRecord</a></tt> <tt class="py-keyword">import</tt> <tt id="link-429" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-429', 'SeqRecord', 'link-177');">SeqRecord</a></tt>  <tt class="py-comment"># Lazy to avoid circular imports</tt> </tt>
<a name="L1339"></a><tt class="py-lineno">1339</tt>  <tt class="py-line">        <tt id="link-430" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-430', 'a', 'link-49');">a</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-431" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-431', 'alphabet', 'link-44');">alphabet</a></tt> </tt>
<a name="L1340"></a><tt class="py-lineno">1340</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">c</tt> <tt class="py-keyword">in</tt> <tt id="link-432" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-432', 'other', 'link-57');">other</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1341"></a><tt class="py-lineno">1341</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">,</tt> <tt id="link-433" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-433', 'SeqRecord', 'link-177');">SeqRecord</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1342"></a><tt class="py-lineno">1342</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"Iterable cannot contain SeqRecords"</tt><tt class="py-op">)</tt> </tt>
<a name="L1343"></a><tt class="py-lineno">1343</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">,</tt> <tt class="py-string">"alphabet"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1344"></a><tt class="py-lineno">1344</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-434" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-434', 'a', 'link-49');">a</a></tt> <tt class="py-op">!=</tt> <tt class="py-name">c</tt><tt class="py-op">.</tt><tt id="link-435" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-435', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1345"></a><tt class="py-lineno">1345</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-436" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-436', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-437" class="py-name"><a title="Bio.Alphabet._check_type_compatible" class="py-name" href="#" onclick="return doclink('link-437', '_check_type_compatible', 'link-59');">_check_type_compatible</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt id="link-438" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-438', 'a', 'link-49');">a</a></tt><tt class="py-op">,</tt> <tt class="py-name">c</tt><tt class="py-op">.</tt><tt id="link-439" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-439', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1346"></a><tt class="py-lineno">1346</tt>  <tt class="py-line">                        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt> </tt>
<a name="L1347"></a><tt class="py-lineno">1347</tt>  <tt class="py-line">                            <tt class="py-string">"Incompatible alphabets {0!r} and {1!r}"</tt><tt class="py-op">.</tt><tt id="link-440" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-440', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1348"></a><tt class="py-lineno">1348</tt>  <tt class="py-line">                                <tt id="link-441" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-441', 'a', 'link-49');">a</a></tt><tt class="py-op">,</tt> <tt class="py-name">c</tt><tt class="py-op">.</tt><tt id="link-442" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-442', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1349"></a><tt class="py-lineno">1349</tt>  <tt class="py-line">                    <tt id="link-443" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-443', 'a', 'link-49');">a</a></tt> <tt class="py-op">=</tt> <tt id="link-444" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-444', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-445" class="py-name"><a title="Bio.Alphabet._consensus_alphabet" class="py-name" href="#" onclick="return doclink('link-445', '_consensus_alphabet', 'link-167');">_consensus_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt id="link-446" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-446', 'a', 'link-49');">a</a></tt><tt class="py-op">,</tt> <tt class="py-name">c</tt><tt class="py-op">.</tt><tt id="link-447" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-447', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1350"></a><tt class="py-lineno">1350</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1351"></a><tt class="py-lineno">1351</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Input must be an iterable of Seqs or Strings"</tt><tt class="py-op">)</tt> </tt>
<a name="L1352"></a><tt class="py-lineno">1352</tt>  <tt class="py-line">        <tt class="py-name">temp_data</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-448" class="py-name"><a title="Bio.Seq.MutableSeq.join
Bio.Seq.Seq.join
Bio.Seq.UnknownSeq.join" class="py-name" href="#" onclick="return doclink('link-448', 'join', 'link-28');">join</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">z</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">z</tt> <tt class="py-keyword">in</tt> <tt id="link-449" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-449', 'other', 'link-57');">other</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1353"></a><tt class="py-lineno">1353</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt class="py-name">temp_data</tt><tt class="py-op">,</tt> <tt id="link-450" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-450', 'a', 'link-49');">a</a></tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1354"></a><tt class="py-lineno">1354</tt>  <tt class="py-line"> </tt>
<a name="L1355"></a><tt class="py-lineno">1355</tt>  <tt class="py-line"> </tt>
<a name="UnknownSeq"></a><div id="UnknownSeq-def"><a name="L1356"></a><tt class="py-lineno">1356</tt> <a class="py-toggle" href="#" id="UnknownSeq-toggle" onclick="return toggle('UnknownSeq');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Seq.UnknownSeq-class.html">UnknownSeq</a><tt class="py-op">(</tt><tt class="py-base-class">Seq</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnknownSeq-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="UnknownSeq-expanded"><a name="L1357"></a><tt class="py-lineno">1357</tt>  <tt class="py-line">    <tt class="py-docstring">"""Read-only sequence object of known length but unknown contents.</tt> </tt>
<a name="L1358"></a><tt class="py-lineno">1358</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1359"></a><tt class="py-lineno">1359</tt>  <tt class="py-line"><tt class="py-docstring">    If you have an unknown sequence, you can represent this with a normal</tt> </tt>
<a name="L1360"></a><tt class="py-lineno">1360</tt>  <tt class="py-line"><tt class="py-docstring">    Seq object, for example:</tt> </tt>
<a name="L1361"></a><tt class="py-lineno">1361</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1362"></a><tt class="py-lineno">1362</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; my_seq = Seq("N"*5)</tt> </tt>
<a name="L1363"></a><tt class="py-lineno">1363</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L1364"></a><tt class="py-lineno">1364</tt>  <tt class="py-line"><tt class="py-docstring">    Seq('NNNNN')</tt> </tt>
<a name="L1365"></a><tt class="py-lineno">1365</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; len(my_seq)</tt> </tt>
<a name="L1366"></a><tt class="py-lineno">1366</tt>  <tt class="py-line"><tt class="py-docstring">    5</tt> </tt>
<a name="L1367"></a><tt class="py-lineno">1367</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(my_seq)</tt> </tt>
<a name="L1368"></a><tt class="py-lineno">1368</tt>  <tt class="py-line"><tt class="py-docstring">    NNNNN</tt> </tt>
<a name="L1369"></a><tt class="py-lineno">1369</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1370"></a><tt class="py-lineno">1370</tt>  <tt class="py-line"><tt class="py-docstring">    However, this is rather wasteful of memory (especially for large</tt> </tt>
<a name="L1371"></a><tt class="py-lineno">1371</tt>  <tt class="py-line"><tt class="py-docstring">    sequences), which is where this class is most useful:</tt> </tt>
<a name="L1372"></a><tt class="py-lineno">1372</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1373"></a><tt class="py-lineno">1373</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; unk_five = UnknownSeq(5)</tt> </tt>
<a name="L1374"></a><tt class="py-lineno">1374</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; unk_five</tt> </tt>
<a name="L1375"></a><tt class="py-lineno">1375</tt>  <tt class="py-line"><tt class="py-docstring">    UnknownSeq(5, character='?')</tt> </tt>
<a name="L1376"></a><tt class="py-lineno">1376</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; len(unk_five)</tt> </tt>
<a name="L1377"></a><tt class="py-lineno">1377</tt>  <tt class="py-line"><tt class="py-docstring">    5</tt> </tt>
<a name="L1378"></a><tt class="py-lineno">1378</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(unk_five)</tt> </tt>
<a name="L1379"></a><tt class="py-lineno">1379</tt>  <tt class="py-line"><tt class="py-docstring">    ?????</tt> </tt>
<a name="L1380"></a><tt class="py-lineno">1380</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1381"></a><tt class="py-lineno">1381</tt>  <tt class="py-line"><tt class="py-docstring">    You can add unknown sequence together, provided their alphabets and</tt> </tt>
<a name="L1382"></a><tt class="py-lineno">1382</tt>  <tt class="py-line"><tt class="py-docstring">    characters are compatible, and get another memory saving UnknownSeq:</tt> </tt>
<a name="L1383"></a><tt class="py-lineno">1383</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1384"></a><tt class="py-lineno">1384</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; unk_four = UnknownSeq(4)</tt> </tt>
<a name="L1385"></a><tt class="py-lineno">1385</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; unk_four</tt> </tt>
<a name="L1386"></a><tt class="py-lineno">1386</tt>  <tt class="py-line"><tt class="py-docstring">    UnknownSeq(4, character='?')</tt> </tt>
<a name="L1387"></a><tt class="py-lineno">1387</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; unk_four + unk_five</tt> </tt>
<a name="L1388"></a><tt class="py-lineno">1388</tt>  <tt class="py-line"><tt class="py-docstring">    UnknownSeq(9, character='?')</tt> </tt>
<a name="L1389"></a><tt class="py-lineno">1389</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1390"></a><tt class="py-lineno">1390</tt>  <tt class="py-line"><tt class="py-docstring">    If the alphabet or characters don't match up, the addition gives an</tt> </tt>
<a name="L1391"></a><tt class="py-lineno">1391</tt>  <tt class="py-line"><tt class="py-docstring">    ordinary Seq object:</tt> </tt>
<a name="L1392"></a><tt class="py-lineno">1392</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1393"></a><tt class="py-lineno">1393</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; unk_nnnn = UnknownSeq(4, character="N")</tt> </tt>
<a name="L1394"></a><tt class="py-lineno">1394</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; unk_nnnn</tt> </tt>
<a name="L1395"></a><tt class="py-lineno">1395</tt>  <tt class="py-line"><tt class="py-docstring">    UnknownSeq(4, character='N')</tt> </tt>
<a name="L1396"></a><tt class="py-lineno">1396</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; unk_nnnn + unk_four</tt> </tt>
<a name="L1397"></a><tt class="py-lineno">1397</tt>  <tt class="py-line"><tt class="py-docstring">    Seq('NNNN????')</tt> </tt>
<a name="L1398"></a><tt class="py-lineno">1398</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1399"></a><tt class="py-lineno">1399</tt>  <tt class="py-line"><tt class="py-docstring">    Combining with a real Seq gives a new Seq object:</tt> </tt>
<a name="L1400"></a><tt class="py-lineno">1400</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1401"></a><tt class="py-lineno">1401</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; known_seq = Seq("ACGT")</tt> </tt>
<a name="L1402"></a><tt class="py-lineno">1402</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; unk_four + known_seq</tt> </tt>
<a name="L1403"></a><tt class="py-lineno">1403</tt>  <tt class="py-line"><tt class="py-docstring">    Seq('????ACGT')</tt> </tt>
<a name="L1404"></a><tt class="py-lineno">1404</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; known_seq + unk_four</tt> </tt>
<a name="L1405"></a><tt class="py-lineno">1405</tt>  <tt class="py-line"><tt class="py-docstring">    Seq('ACGT????')</tt> </tt>
<a name="L1406"></a><tt class="py-lineno">1406</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1407"></a><tt class="py-lineno">1407</tt>  <tt class="py-line"> </tt>
<a name="UnknownSeq.__init__"></a><div id="UnknownSeq.__init__-def"><a name="L1408"></a><tt class="py-lineno">1408</tt> <a class="py-toggle" href="#" id="UnknownSeq.__init__-toggle" onclick="return toggle('UnknownSeq.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.UnknownSeq-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">length</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt id="link-451" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-451', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-452" class="py-name"><a title="Bio.Alphabet.generic_alphabet" class="py-name" href="#" onclick="return doclink('link-452', 'generic_alphabet', 'link-41');">generic_alphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1409"></a><tt class="py-lineno">1409</tt>  <tt class="py-line">                 <tt class="py-param">character</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnknownSeq.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="UnknownSeq.__init__-expanded"><a name="L1410"></a><tt class="py-lineno">1410</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a new UnknownSeq object.</tt> </tt>
<a name="L1411"></a><tt class="py-lineno">1411</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1412"></a><tt class="py-lineno">1412</tt>  <tt class="py-line"><tt class="py-docstring">        If character is omitted, it is determined from the alphabet, "N" for</tt> </tt>
<a name="L1413"></a><tt class="py-lineno">1413</tt>  <tt class="py-line"><tt class="py-docstring">        nucleotides, "X" for proteins, and "?" otherwise.</tt> </tt>
<a name="L1414"></a><tt class="py-lineno">1414</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1415"></a><tt class="py-lineno">1415</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_length</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-453" class="py-name" targets="Method Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length()=Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer-class.html#length"><a title="Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-453', 'length', 'link-453');">length</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1416"></a><tt class="py-lineno">1416</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_length</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1417"></a><tt class="py-lineno">1417</tt>  <tt class="py-line">            <tt class="py-comment"># TODO - Block zero length UnknownSeq?  You can just use a Seq!</tt> </tt>
<a name="L1418"></a><tt class="py-lineno">1418</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Length must not be negative."</tt><tt class="py-op">)</tt> </tt>
<a name="L1419"></a><tt class="py-lineno">1419</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-454" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-454', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-455" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-455', 'alphabet', 'link-44');">alphabet</a></tt> </tt>
<a name="L1420"></a><tt class="py-lineno">1420</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">character</tt><tt class="py-op">:</tt> </tt>
<a name="L1421"></a><tt class="py-lineno">1421</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">character</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L1422"></a><tt class="py-lineno">1422</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"character argument should be a single "</tt> </tt>
<a name="L1423"></a><tt class="py-lineno">1423</tt>  <tt class="py-line">                                 <tt class="py-string">"letter string."</tt><tt class="py-op">)</tt> </tt>
<a name="L1424"></a><tt class="py-lineno">1424</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_character</tt> <tt class="py-op">=</tt> <tt class="py-name">character</tt> </tt>
<a name="L1425"></a><tt class="py-lineno">1425</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1426"></a><tt class="py-lineno">1426</tt>  <tt class="py-line">            <tt class="py-name">base</tt> <tt class="py-op">=</tt> <tt id="link-456" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-456', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-457" class="py-name"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-457', '_get_base_alphabet', 'link-297');">_get_base_alphabet</a></tt><tt class="py-op">(</tt><tt id="link-458" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-458', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1427"></a><tt class="py-lineno">1427</tt>  <tt class="py-line">            <tt class="py-comment"># TODO? Check the case of the letters in the alphabet?</tt> </tt>
<a name="L1428"></a><tt class="py-lineno">1428</tt>  <tt class="py-line">            <tt class="py-comment"># We may have to use "n" instead of "N" etc.</tt> </tt>
<a name="L1429"></a><tt class="py-lineno">1429</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">base</tt><tt class="py-op">,</tt> <tt id="link-459" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-459', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-460" class="py-name" targets="Class Bio.Alphabet.NucleotideAlphabet=Bio.Alphabet.NucleotideAlphabet-class.html"><a title="Bio.Alphabet.NucleotideAlphabet" class="py-name" href="#" onclick="return doclink('link-460', 'NucleotideAlphabet', 'link-460');">NucleotideAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1430"></a><tt class="py-lineno">1430</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_character</tt> <tt class="py-op">=</tt> <tt class="py-string">"N"</tt> </tt>
<a name="L1431"></a><tt class="py-lineno">1431</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">base</tt><tt class="py-op">,</tt> <tt id="link-461" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-461', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-462" class="py-name"><a title="Bio.Alphabet.ProteinAlphabet" class="py-name" href="#" onclick="return doclink('link-462', 'ProteinAlphabet', 'link-300');">ProteinAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1432"></a><tt class="py-lineno">1432</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_character</tt> <tt class="py-op">=</tt> <tt class="py-string">"X"</tt> </tt>
<a name="L1433"></a><tt class="py-lineno">1433</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1434"></a><tt class="py-lineno">1434</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_character</tt> <tt class="py-op">=</tt> <tt class="py-string">"?"</tt> </tt>
</div><a name="L1435"></a><tt class="py-lineno">1435</tt>  <tt class="py-line"> </tt>
<a name="UnknownSeq.__len__"></a><div id="UnknownSeq.__len__-def"><a name="L1436"></a><tt class="py-lineno">1436</tt> <a class="py-toggle" href="#" id="UnknownSeq.__len__-toggle" onclick="return toggle('UnknownSeq.__len__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.UnknownSeq-class.html#__len__">__len__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnknownSeq.__len__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="UnknownSeq.__len__-expanded"><a name="L1437"></a><tt class="py-lineno">1437</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the stated length of the unknown sequence."""</tt> </tt>
<a name="L1438"></a><tt class="py-lineno">1438</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_length</tt> </tt>
</div><a name="L1439"></a><tt class="py-lineno">1439</tt>  <tt class="py-line"> </tt>
<a name="UnknownSeq.__str__"></a><div id="UnknownSeq.__str__-def"><a name="L1440"></a><tt class="py-lineno">1440</tt> <a class="py-toggle" href="#" id="UnknownSeq.__str__-toggle" onclick="return toggle('UnknownSeq.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.UnknownSeq-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnknownSeq.__str__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="UnknownSeq.__str__-expanded"><a name="L1441"></a><tt class="py-lineno">1441</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the unknown sequence as full string of the given length."""</tt> </tt>
<a name="L1442"></a><tt class="py-lineno">1442</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_character</tt> <tt class="py-op">*</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_length</tt> </tt>
</div><a name="L1443"></a><tt class="py-lineno">1443</tt>  <tt class="py-line"> </tt>
<a name="UnknownSeq.__repr__"></a><div id="UnknownSeq.__repr__-def"><a name="L1444"></a><tt class="py-lineno">1444</tt> <a class="py-toggle" href="#" id="UnknownSeq.__repr__-toggle" onclick="return toggle('UnknownSeq.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.UnknownSeq-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnknownSeq.__repr__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="UnknownSeq.__repr__-expanded"><a name="L1445"></a><tt class="py-lineno">1445</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return (truncated) representation of the sequence for debugging."""</tt> </tt>
<a name="L1446"></a><tt class="py-lineno">1446</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-463" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-463', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-keyword">is</tt> <tt id="link-464" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-464', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-465" class="py-name"><a title="Bio.Alphabet.generic_alphabet" class="py-name" href="#" onclick="return doclink('link-465', 'generic_alphabet', 'link-41');">generic_alphabet</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1447"></a><tt class="py-lineno">1447</tt>  <tt class="py-line">            <tt class="py-comment"># Default used, we can omit it and simplify the representation</tt> </tt>
<a name="L1448"></a><tt class="py-lineno">1448</tt>  <tt class="py-line">            <tt id="link-466" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-466', 'a', 'link-49');">a</a></tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L1449"></a><tt class="py-lineno">1449</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1450"></a><tt class="py-lineno">1450</tt>  <tt class="py-line">            <tt id="link-467" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-467', 'a', 'link-49');">a</a></tt> <tt class="py-op">=</tt> <tt class="py-string">", alphabet=%r"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-468" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-468', 'alphabet', 'link-44');">alphabet</a></tt> </tt>
<a name="L1451"></a><tt class="py-lineno">1451</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">"UnknownSeq({0}{1!s}, character={2!r})"</tt><tt class="py-op">.</tt><tt id="link-469" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-469', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1452"></a><tt class="py-lineno">1452</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_length</tt><tt class="py-op">,</tt> <tt id="link-470" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-470', 'a', 'link-49');">a</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_character</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1453"></a><tt class="py-lineno">1453</tt>  <tt class="py-line"> </tt>
<a name="UnknownSeq.__add__"></a><div id="UnknownSeq.__add__-def"><a name="L1454"></a><tt class="py-lineno">1454</tt> <a class="py-toggle" href="#" id="UnknownSeq.__add__-toggle" onclick="return toggle('UnknownSeq.__add__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.UnknownSeq-class.html#__add__">__add__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnknownSeq.__add__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="UnknownSeq.__add__-expanded"><a name="L1455"></a><tt class="py-lineno">1455</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add another sequence or string to this sequence.</tt> </tt>
<a name="L1456"></a><tt class="py-lineno">1456</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1457"></a><tt class="py-lineno">1457</tt>  <tt class="py-line"><tt class="py-docstring">        Adding two UnknownSeq objects returns another UnknownSeq object</tt> </tt>
<a name="L1458"></a><tt class="py-lineno">1458</tt>  <tt class="py-line"><tt class="py-docstring">        provided the character is the same and the alphabets are compatible.</tt> </tt>
<a name="L1459"></a><tt class="py-lineno">1459</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1460"></a><tt class="py-lineno">1460</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import UnknownSeq</tt> </tt>
<a name="L1461"></a><tt class="py-lineno">1461</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_protein</tt> </tt>
<a name="L1462"></a><tt class="py-lineno">1462</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; UnknownSeq(10, generic_protein) + UnknownSeq(5, generic_protein)</tt> </tt>
<a name="L1463"></a><tt class="py-lineno">1463</tt>  <tt class="py-line"><tt class="py-docstring">        UnknownSeq(15, alphabet=ProteinAlphabet(), character='X')</tt> </tt>
<a name="L1464"></a><tt class="py-lineno">1464</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1465"></a><tt class="py-lineno">1465</tt>  <tt class="py-line"><tt class="py-docstring">        If the characters differ, an UnknownSeq object cannot be used, so a</tt> </tt>
<a name="L1466"></a><tt class="py-lineno">1466</tt>  <tt class="py-line"><tt class="py-docstring">        Seq object is returned:</tt> </tt>
<a name="L1467"></a><tt class="py-lineno">1467</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1468"></a><tt class="py-lineno">1468</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import UnknownSeq</tt> </tt>
<a name="L1469"></a><tt class="py-lineno">1469</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_protein</tt> </tt>
<a name="L1470"></a><tt class="py-lineno">1470</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; UnknownSeq(10, generic_protein) + UnknownSeq(5, generic_protein,</tt> </tt>
<a name="L1471"></a><tt class="py-lineno">1471</tt>  <tt class="py-line"><tt class="py-docstring">        ...                                              character="x")</tt> </tt>
<a name="L1472"></a><tt class="py-lineno">1472</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('XXXXXXXXXXxxxxx', ProteinAlphabet())</tt> </tt>
<a name="L1473"></a><tt class="py-lineno">1473</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1474"></a><tt class="py-lineno">1474</tt>  <tt class="py-line"><tt class="py-docstring">        If adding a string to an UnknownSeq, a new Seq is returned with the</tt> </tt>
<a name="L1475"></a><tt class="py-lineno">1475</tt>  <tt class="py-line"><tt class="py-docstring">        same alphabet:</tt> </tt>
<a name="L1476"></a><tt class="py-lineno">1476</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1477"></a><tt class="py-lineno">1477</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import UnknownSeq</tt> </tt>
<a name="L1478"></a><tt class="py-lineno">1478</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_protein</tt> </tt>
<a name="L1479"></a><tt class="py-lineno">1479</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; UnknownSeq(5, generic_protein) + "LV"</tt> </tt>
<a name="L1480"></a><tt class="py-lineno">1480</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('XXXXXLV', ProteinAlphabet())</tt> </tt>
<a name="L1481"></a><tt class="py-lineno">1481</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1482"></a><tt class="py-lineno">1482</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-471" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-471', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt id="link-472" class="py-name"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-472', 'UnknownSeq', 'link-84');">UnknownSeq</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L1483"></a><tt class="py-lineno">1483</tt>  <tt class="py-line">           <tt id="link-473" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-473', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt class="py-name">_character</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_character</tt><tt class="py-op">:</tt> </tt>
<a name="L1484"></a><tt class="py-lineno">1484</tt>  <tt class="py-line">            <tt class="py-comment"># TODO - Check the alphabets match</tt> </tt>
<a name="L1485"></a><tt class="py-lineno">1485</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-474" class="py-name"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-474', 'UnknownSeq', 'link-84');">UnknownSeq</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-475" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-475', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1486"></a><tt class="py-lineno">1486</tt>  <tt class="py-line">                              <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-476" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-476', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_character</tt><tt class="py-op">)</tt> </tt>
<a name="L1487"></a><tt class="py-lineno">1487</tt>  <tt class="py-line">        <tt class="py-comment"># Offload to the base class...</tt> </tt>
<a name="L1488"></a><tt class="py-lineno">1488</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-477" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-477', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-478" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-478', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt id="link-479" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-479', 'other', 'link-57');">other</a></tt> </tt>
</div><a name="L1489"></a><tt class="py-lineno">1489</tt>  <tt class="py-line"> </tt>
<a name="UnknownSeq.__radd__"></a><div id="UnknownSeq.__radd__-def"><a name="L1490"></a><tt class="py-lineno">1490</tt> <a class="py-toggle" href="#" id="UnknownSeq.__radd__-toggle" onclick="return toggle('UnknownSeq.__radd__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.UnknownSeq-class.html#__radd__">__radd__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnknownSeq.__radd__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="UnknownSeq.__radd__-expanded"><a name="L1491"></a><tt class="py-lineno">1491</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add a sequence on the left."""</tt> </tt>
<a name="L1492"></a><tt class="py-lineno">1492</tt>  <tt class="py-line">        <tt class="py-comment"># If other is an UnknownSeq, then __add__ would be called.</tt> </tt>
<a name="L1493"></a><tt class="py-lineno">1493</tt>  <tt class="py-line">        <tt class="py-comment"># Offload to the base class...</tt> </tt>
<a name="L1494"></a><tt class="py-lineno">1494</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-480" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-480', 'other', 'link-57');">other</a></tt> <tt class="py-op">+</tt> <tt id="link-481" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-481', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-482" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-482', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1495"></a><tt class="py-lineno">1495</tt>  <tt class="py-line"> </tt>
<a name="UnknownSeq.__mul__"></a><div id="UnknownSeq.__mul__-def"><a name="L1496"></a><tt class="py-lineno">1496</tt> <a class="py-toggle" href="#" id="UnknownSeq.__mul__-toggle" onclick="return toggle('UnknownSeq.__mul__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.UnknownSeq-class.html#__mul__">__mul__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnknownSeq.__mul__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="UnknownSeq.__mul__-expanded"><a name="L1497"></a><tt class="py-lineno">1497</tt>  <tt class="py-line">        <tt class="py-docstring">"""Multiply UnknownSeq by integer.</tt> </tt>
<a name="L1498"></a><tt class="py-lineno">1498</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1499"></a><tt class="py-lineno">1499</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import UnknownSeq</tt> </tt>
<a name="L1500"></a><tt class="py-lineno">1500</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_dna</tt> </tt>
<a name="L1501"></a><tt class="py-lineno">1501</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; UnknownSeq(3) * 2</tt> </tt>
<a name="L1502"></a><tt class="py-lineno">1502</tt>  <tt class="py-line"><tt class="py-docstring">        UnknownSeq(6, character='?')</tt> </tt>
<a name="L1503"></a><tt class="py-lineno">1503</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; UnknownSeq(3, generic_dna) * 2</tt> </tt>
<a name="L1504"></a><tt class="py-lineno">1504</tt>  <tt class="py-line"><tt class="py-docstring">        UnknownSeq(6, alphabet=DNAAlphabet(), character='N')</tt> </tt>
<a name="L1505"></a><tt class="py-lineno">1505</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1506"></a><tt class="py-lineno">1506</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-483" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-483', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1507"></a><tt class="py-lineno">1507</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"can't multiply {} by non-int type"</tt><tt class="py-op">.</tt><tt id="link-484" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-484', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1508"></a><tt class="py-lineno">1508</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt id="link-485" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-485', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-486" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-486', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1509"></a><tt class="py-lineno">1509</tt>  <tt class="py-line"> </tt>
<a name="UnknownSeq.__rmul__"></a><div id="UnknownSeq.__rmul__-def"><a name="L1510"></a><tt class="py-lineno">1510</tt> <a class="py-toggle" href="#" id="UnknownSeq.__rmul__-toggle" onclick="return toggle('UnknownSeq.__rmul__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.UnknownSeq-class.html#__rmul__">__rmul__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnknownSeq.__rmul__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="UnknownSeq.__rmul__-expanded"><a name="L1511"></a><tt class="py-lineno">1511</tt>  <tt class="py-line">        <tt class="py-docstring">"""Multiply integer by UnknownSeq.</tt> </tt>
<a name="L1512"></a><tt class="py-lineno">1512</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1513"></a><tt class="py-lineno">1513</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import UnknownSeq</tt> </tt>
<a name="L1514"></a><tt class="py-lineno">1514</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_dna</tt> </tt>
<a name="L1515"></a><tt class="py-lineno">1515</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; 2 * UnknownSeq(3)</tt> </tt>
<a name="L1516"></a><tt class="py-lineno">1516</tt>  <tt class="py-line"><tt class="py-docstring">        UnknownSeq(6, character='?')</tt> </tt>
<a name="L1517"></a><tt class="py-lineno">1517</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; 2 * UnknownSeq(3, generic_dna)</tt> </tt>
<a name="L1518"></a><tt class="py-lineno">1518</tt>  <tt class="py-line"><tt class="py-docstring">        UnknownSeq(6, alphabet=DNAAlphabet(), character='N')</tt> </tt>
<a name="L1519"></a><tt class="py-lineno">1519</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1520"></a><tt class="py-lineno">1520</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-487" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-487', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1521"></a><tt class="py-lineno">1521</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"can't multiply {} by non-int type"</tt><tt class="py-op">.</tt><tt id="link-488" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-488', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1522"></a><tt class="py-lineno">1522</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt id="link-489" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-489', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-490" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-490', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1523"></a><tt class="py-lineno">1523</tt>  <tt class="py-line"> </tt>
<a name="UnknownSeq.__imul__"></a><div id="UnknownSeq.__imul__-def"><a name="L1524"></a><tt class="py-lineno">1524</tt> <a class="py-toggle" href="#" id="UnknownSeq.__imul__-toggle" onclick="return toggle('UnknownSeq.__imul__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.UnknownSeq-class.html#__imul__">__imul__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnknownSeq.__imul__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="UnknownSeq.__imul__-expanded"><a name="L1525"></a><tt class="py-lineno">1525</tt>  <tt class="py-line">        <tt class="py-docstring">"""Multiply UnknownSeq in-place.</tt> </tt>
<a name="L1526"></a><tt class="py-lineno">1526</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1527"></a><tt class="py-lineno">1527</tt>  <tt class="py-line"><tt class="py-docstring">        Note although UnknownSeq is immutable, the in-place method is</tt> </tt>
<a name="L1528"></a><tt class="py-lineno">1528</tt>  <tt class="py-line"><tt class="py-docstring">        included to match the behaviour for regular Python strings.</tt> </tt>
<a name="L1529"></a><tt class="py-lineno">1529</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1530"></a><tt class="py-lineno">1530</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import UnknownSeq</tt> </tt>
<a name="L1531"></a><tt class="py-lineno">1531</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_dna</tt> </tt>
<a name="L1532"></a><tt class="py-lineno">1532</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; seq = UnknownSeq(3, generic_dna)</tt> </tt>
<a name="L1533"></a><tt class="py-lineno">1533</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; seq *= 2</tt> </tt>
<a name="L1534"></a><tt class="py-lineno">1534</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; seq</tt> </tt>
<a name="L1535"></a><tt class="py-lineno">1535</tt>  <tt class="py-line"><tt class="py-docstring">        UnknownSeq(6, alphabet=DNAAlphabet(), character='N')</tt> </tt>
<a name="L1536"></a><tt class="py-lineno">1536</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1537"></a><tt class="py-lineno">1537</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-491" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-491', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1538"></a><tt class="py-lineno">1538</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"can't multiply {} by non-int type"</tt><tt class="py-op">.</tt><tt id="link-492" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-492', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1539"></a><tt class="py-lineno">1539</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt id="link-493" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-493', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-494" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-494', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1540"></a><tt class="py-lineno">1540</tt>  <tt class="py-line"> </tt>
<a name="UnknownSeq.__getitem__"></a><div id="UnknownSeq.__getitem__-def"><a name="L1541"></a><tt class="py-lineno">1541</tt> <a class="py-toggle" href="#" id="UnknownSeq.__getitem__-toggle" onclick="return toggle('UnknownSeq.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.UnknownSeq-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">index</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnknownSeq.__getitem__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="UnknownSeq.__getitem__-expanded"><a name="L1542"></a><tt class="py-lineno">1542</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get a subsequence from the UnknownSeq object.</tt> </tt>
<a name="L1543"></a><tt class="py-lineno">1543</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1544"></a><tt class="py-lineno">1544</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; unk = UnknownSeq(8, character="N")</tt> </tt>
<a name="L1545"></a><tt class="py-lineno">1545</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(unk[:])</tt> </tt>
<a name="L1546"></a><tt class="py-lineno">1546</tt>  <tt class="py-line"><tt class="py-docstring">        NNNNNNNN</tt> </tt>
<a name="L1547"></a><tt class="py-lineno">1547</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(unk[5:3])</tt> </tt>
<a name="L1548"></a><tt class="py-lineno">1548</tt>  <tt class="py-line"><tt class="py-docstring">        &lt;BLANKLINE&gt;</tt> </tt>
<a name="L1549"></a><tt class="py-lineno">1549</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(unk[1:-1])</tt> </tt>
<a name="L1550"></a><tt class="py-lineno">1550</tt>  <tt class="py-line"><tt class="py-docstring">        NNNNNN</tt> </tt>
<a name="L1551"></a><tt class="py-lineno">1551</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(unk[1:-1:2])</tt> </tt>
<a name="L1552"></a><tt class="py-lineno">1552</tt>  <tt class="py-line"><tt class="py-docstring">        NNN</tt> </tt>
<a name="L1553"></a><tt class="py-lineno">1553</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1554"></a><tt class="py-lineno">1554</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-495" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.Seq.Seq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-495', 'index', 'link-150');">index</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1555"></a><tt class="py-lineno">1555</tt>  <tt class="py-line">            <tt class="py-comment"># TODO - Check the bounds without wasting memory</tt> </tt>
<a name="L1556"></a><tt class="py-lineno">1556</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt id="link-496" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.Seq.Seq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-496', 'index', 'link-150');">index</a></tt><tt class="py-op">]</tt> </tt>
<a name="L1557"></a><tt class="py-lineno">1557</tt>  <tt class="py-line">        <tt class="py-name">old_length</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_length</tt> </tt>
<a name="L1558"></a><tt class="py-lineno">1558</tt>  <tt class="py-line">        <tt class="py-name">step</tt> <tt class="py-op">=</tt> <tt id="link-497" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.Seq.Seq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-497', 'index', 'link-150');">index</a></tt><tt class="py-op">.</tt><tt class="py-name">step</tt> </tt>
<a name="L1559"></a><tt class="py-lineno">1559</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">step</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt> <tt class="py-keyword">or</tt> <tt class="py-name">step</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L1560"></a><tt class="py-lineno">1560</tt>  <tt class="py-line">            <tt class="py-comment"># This calculates the length you'd get from ("N"*old_length)[index]</tt> </tt>
<a name="L1561"></a><tt class="py-lineno">1561</tt>  <tt class="py-line">            <tt id="link-498" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-498', 'start', 'link-224');">start</a></tt> <tt class="py-op">=</tt> <tt id="link-499" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.Seq.Seq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-499', 'index', 'link-150');">index</a></tt><tt class="py-op">.</tt><tt id="link-500" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-500', 'start', 'link-224');">start</a></tt> </tt>
<a name="L1562"></a><tt class="py-lineno">1562</tt>  <tt class="py-line">            <tt id="link-501" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-501', 'end', 'link-225');">end</a></tt> <tt class="py-op">=</tt> <tt id="link-502" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.Seq.Seq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-502', 'index', 'link-150');">index</a></tt><tt class="py-op">.</tt><tt class="py-name">stop</tt> </tt>
<a name="L1563"></a><tt class="py-lineno">1563</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-503" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-503', 'start', 'link-224');">start</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1564"></a><tt class="py-lineno">1564</tt>  <tt class="py-line">                <tt id="link-504" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-504', 'start', 'link-224');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L1565"></a><tt class="py-lineno">1565</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-505" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-505', 'start', 'link-224');">start</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1566"></a><tt class="py-lineno">1566</tt>  <tt class="py-line">                <tt id="link-506" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-506', 'start', 'link-224');">start</a></tt> <tt class="py-op">=</tt> <tt id="link-507" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.max=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#max"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-507', 'max', 'link-507');">max</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">old_length</tt> <tt class="py-op">+</tt> <tt id="link-508" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-508', 'start', 'link-224');">start</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1567"></a><tt class="py-lineno">1567</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-509" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-509', 'start', 'link-224');">start</a></tt> <tt class="py-op">&gt;</tt> <tt class="py-name">old_length</tt><tt class="py-op">:</tt> </tt>
<a name="L1568"></a><tt class="py-lineno">1568</tt>  <tt class="py-line">                <tt id="link-510" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-510', 'start', 'link-224');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">old_length</tt> </tt>
<a name="L1569"></a><tt class="py-lineno">1569</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-511" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-511', 'end', 'link-225');">end</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1570"></a><tt class="py-lineno">1570</tt>  <tt class="py-line">                <tt id="link-512" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-512', 'end', 'link-225');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">old_length</tt> </tt>
<a name="L1571"></a><tt class="py-lineno">1571</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-513" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-513', 'end', 'link-225');">end</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1572"></a><tt class="py-lineno">1572</tt>  <tt class="py-line">                <tt id="link-514" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-514', 'end', 'link-225');">end</a></tt> <tt class="py-op">=</tt> <tt id="link-515" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-515', 'max', 'link-507');">max</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">old_length</tt> <tt class="py-op">+</tt> <tt id="link-516" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-516', 'end', 'link-225');">end</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1573"></a><tt class="py-lineno">1573</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-517" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-517', 'end', 'link-225');">end</a></tt> <tt class="py-op">&gt;</tt> <tt class="py-name">old_length</tt><tt class="py-op">:</tt> </tt>
<a name="L1574"></a><tt class="py-lineno">1574</tt>  <tt class="py-line">                <tt id="link-518" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-518', 'end', 'link-225');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">old_length</tt> </tt>
<a name="L1575"></a><tt class="py-lineno">1575</tt>  <tt class="py-line">            <tt class="py-name">new_length</tt> <tt class="py-op">=</tt> <tt id="link-519" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-519', 'max', 'link-507');">max</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt id="link-520" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-520', 'end', 'link-225');">end</a></tt> <tt class="py-op">-</tt> <tt id="link-521" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-521', 'start', 'link-224');">start</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1576"></a><tt class="py-lineno">1576</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">step</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1577"></a><tt class="py-lineno">1577</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"slice step cannot be zero"</tt><tt class="py-op">)</tt> </tt>
<a name="L1578"></a><tt class="py-lineno">1578</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1579"></a><tt class="py-lineno">1579</tt>  <tt class="py-line">            <tt class="py-comment"># TODO - handle step efficiently</tt> </tt>
<a name="L1580"></a><tt class="py-lineno">1580</tt>  <tt class="py-line">            <tt class="py-name">new_length</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">"X"</tt> <tt class="py-op">*</tt> <tt class="py-name">old_length</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt id="link-522" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.Seq.Seq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-522', 'index', 'link-150');">index</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1581"></a><tt class="py-lineno">1581</tt>  <tt class="py-line">        <tt class="py-comment"># assert new_length == len(("X"*old_length)[index]), \</tt> </tt>
<a name="L1582"></a><tt class="py-lineno">1582</tt>  <tt class="py-line">        <tt class="py-comment">#       (index, start, end, step, old_length,</tt> </tt>
<a name="L1583"></a><tt class="py-lineno">1583</tt>  <tt class="py-line">        <tt class="py-comment">#        new_length, len(("X"*old_length)[index]))</tt> </tt>
<a name="L1584"></a><tt class="py-lineno">1584</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-523" class="py-name"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-523', 'UnknownSeq', 'link-84');">UnknownSeq</a></tt><tt class="py-op">(</tt><tt class="py-name">new_length</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-524" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-524', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_character</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1585"></a><tt class="py-lineno">1585</tt>  <tt class="py-line"> </tt>
<a name="UnknownSeq.count"></a><div id="UnknownSeq.count-def"><a name="L1586"></a><tt class="py-lineno">1586</tt> <a class="py-toggle" href="#" id="UnknownSeq.count-toggle" onclick="return toggle('UnknownSeq.count');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.UnknownSeq-class.html#count">count</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sub</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">=</tt><tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">maxsize</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnknownSeq.count-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="UnknownSeq.count-expanded"><a name="L1587"></a><tt class="py-lineno">1587</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a non-overlapping count, like that of a python string.</tt> </tt>
<a name="L1588"></a><tt class="py-lineno">1588</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1589"></a><tt class="py-lineno">1589</tt>  <tt class="py-line"><tt class="py-docstring">        This behaves like the python string (and Seq object) method of the</tt> </tt>
<a name="L1590"></a><tt class="py-lineno">1590</tt>  <tt class="py-line"><tt class="py-docstring">        same name, which does a non-overlapping count!</tt> </tt>
<a name="L1591"></a><tt class="py-lineno">1591</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1592"></a><tt class="py-lineno">1592</tt>  <tt class="py-line"><tt class="py-docstring">        For an overlapping search use the newer count_overlap() method.</tt> </tt>
<a name="L1593"></a><tt class="py-lineno">1593</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1594"></a><tt class="py-lineno">1594</tt>  <tt class="py-line"><tt class="py-docstring">        Returns an integer, the number of occurrences of substring</tt> </tt>
<a name="L1595"></a><tt class="py-lineno">1595</tt>  <tt class="py-line"><tt class="py-docstring">        argument sub in the (sub)sequence given by [start:end].</tt> </tt>
<a name="L1596"></a><tt class="py-lineno">1596</tt>  <tt class="py-line"><tt class="py-docstring">        Optional arguments start and end are interpreted as in slice</tt> </tt>
<a name="L1597"></a><tt class="py-lineno">1597</tt>  <tt class="py-line"><tt class="py-docstring">        notation.</tt> </tt>
<a name="L1598"></a><tt class="py-lineno">1598</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1599"></a><tt class="py-lineno">1599</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L1600"></a><tt class="py-lineno">1600</tt>  <tt class="py-line"><tt class="py-docstring">         - sub - a string or another Seq object to look for</tt> </tt>
<a name="L1601"></a><tt class="py-lineno">1601</tt>  <tt class="py-line"><tt class="py-docstring">         - start - optional integer, slice start</tt> </tt>
<a name="L1602"></a><tt class="py-lineno">1602</tt>  <tt class="py-line"><tt class="py-docstring">         - end - optional integer, slice end</tt> </tt>
<a name="L1603"></a><tt class="py-lineno">1603</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1604"></a><tt class="py-lineno">1604</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; "NNNN".count("N")</tt> </tt>
<a name="L1605"></a><tt class="py-lineno">1605</tt>  <tt class="py-line"><tt class="py-docstring">        4</tt> </tt>
<a name="L1606"></a><tt class="py-lineno">1606</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; Seq("NNNN").count("N")</tt> </tt>
<a name="L1607"></a><tt class="py-lineno">1607</tt>  <tt class="py-line"><tt class="py-docstring">        4</tt> </tt>
<a name="L1608"></a><tt class="py-lineno">1608</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; UnknownSeq(4, character="N").count("N")</tt> </tt>
<a name="L1609"></a><tt class="py-lineno">1609</tt>  <tt class="py-line"><tt class="py-docstring">        4</tt> </tt>
<a name="L1610"></a><tt class="py-lineno">1610</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; UnknownSeq(4, character="N").count("A")</tt> </tt>
<a name="L1611"></a><tt class="py-lineno">1611</tt>  <tt class="py-line"><tt class="py-docstring">        0</tt> </tt>
<a name="L1612"></a><tt class="py-lineno">1612</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; UnknownSeq(4, character="N").count("AA")</tt> </tt>
<a name="L1613"></a><tt class="py-lineno">1613</tt>  <tt class="py-line"><tt class="py-docstring">        0</tt> </tt>
<a name="L1614"></a><tt class="py-lineno">1614</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1615"></a><tt class="py-lineno">1615</tt>  <tt class="py-line"><tt class="py-docstring">        HOWEVER, please note because that python strings and Seq objects (and</tt> </tt>
<a name="L1616"></a><tt class="py-lineno">1616</tt>  <tt class="py-line"><tt class="py-docstring">        MutableSeq objects) do a non-overlapping search, this may not give</tt> </tt>
<a name="L1617"></a><tt class="py-lineno">1617</tt>  <tt class="py-line"><tt class="py-docstring">        the answer you expect:</tt> </tt>
<a name="L1618"></a><tt class="py-lineno">1618</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1619"></a><tt class="py-lineno">1619</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; UnknownSeq(4, character="N").count("NN")</tt> </tt>
<a name="L1620"></a><tt class="py-lineno">1620</tt>  <tt class="py-line"><tt class="py-docstring">        2</tt> </tt>
<a name="L1621"></a><tt class="py-lineno">1621</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; UnknownSeq(4, character="N").count("NNN")</tt> </tt>
<a name="L1622"></a><tt class="py-lineno">1622</tt>  <tt class="py-line"><tt class="py-docstring">        1</tt> </tt>
<a name="L1623"></a><tt class="py-lineno">1623</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1624"></a><tt class="py-lineno">1624</tt>  <tt class="py-line">        <tt class="py-name">sub_str</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-525" class="py-name"><a title="Bio.Seq.Seq._get_seq_str_and_check_alphabet" class="py-name" href="#" onclick="return doclink('link-525', '_get_seq_str_and_check_alphabet', 'link-222');">_get_seq_str_and_check_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">sub</tt><tt class="py-op">)</tt> </tt>
<a name="L1625"></a><tt class="py-lineno">1625</tt>  <tt class="py-line">        <tt class="py-name">len_self</tt><tt class="py-op">,</tt> <tt class="py-name">len_sub_str</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_length</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">sub_str</tt><tt class="py-op">)</tt> </tt>
<a name="L1626"></a><tt class="py-lineno">1626</tt>  <tt class="py-line">        <tt class="py-comment"># Handling case where substring not in self</tt> </tt>
<a name="L1627"></a><tt class="py-lineno">1627</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-526" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.set()=Bio.Nexus.Nexus.StepMatrix-class.html#set,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.set()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#set,Method Bio.SVDSuperimposer.SVDSuperimposer.set()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#set"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-526', 'set', 'link-526');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">sub_str</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt id="link-527" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-527', 'set', 'link-526');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_character</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1628"></a><tt class="py-lineno">1628</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-number">0</tt> </tt>
<a name="L1629"></a><tt class="py-lineno">1629</tt>  <tt class="py-line">        <tt class="py-comment"># Setting None to the default arguments</tt> </tt>
<a name="L1630"></a><tt class="py-lineno">1630</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-528" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-528', 'start', 'link-224');">start</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1631"></a><tt class="py-lineno">1631</tt>  <tt class="py-line">            <tt id="link-529" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-529', 'start', 'link-224');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L1632"></a><tt class="py-lineno">1632</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-530" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-530', 'end', 'link-225');">end</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1633"></a><tt class="py-lineno">1633</tt>  <tt class="py-line">            <tt id="link-531" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-531', 'end', 'link-225');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">maxsize</tt> </tt>
<a name="L1634"></a><tt class="py-lineno">1634</tt>  <tt class="py-line">        <tt class="py-comment"># Truncating start and end to max of self._length and min of -self._length</tt> </tt>
<a name="L1635"></a><tt class="py-lineno">1635</tt>  <tt class="py-line">        <tt id="link-532" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-532', 'start', 'link-224');">start</a></tt> <tt class="py-op">=</tt> <tt id="link-533" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-533', 'max', 'link-507');">max</a></tt><tt class="py-op">(</tt><tt id="link-534" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.min=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#min"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-534', 'min', 'link-534');">min</a></tt><tt class="py-op">(</tt><tt id="link-535" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-535', 'start', 'link-224');">start</a></tt><tt class="py-op">,</tt> <tt class="py-name">len_self</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">-</tt><tt class="py-name">len_self</tt><tt class="py-op">)</tt> </tt>
<a name="L1636"></a><tt class="py-lineno">1636</tt>  <tt class="py-line">        <tt id="link-536" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-536', 'end', 'link-225');">end</a></tt> <tt class="py-op">=</tt> <tt id="link-537" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-537', 'max', 'link-507');">max</a></tt><tt class="py-op">(</tt><tt id="link-538" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-538', 'min', 'link-534');">min</a></tt><tt class="py-op">(</tt><tt id="link-539" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-539', 'end', 'link-225');">end</a></tt><tt class="py-op">,</tt> <tt class="py-name">len_self</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">-</tt><tt class="py-name">len_self</tt><tt class="py-op">)</tt> </tt>
<a name="L1637"></a><tt class="py-lineno">1637</tt>  <tt class="py-line">        <tt class="py-comment"># Convert start and ends to positive indexes</tt> </tt>
<a name="L1638"></a><tt class="py-lineno">1638</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-540" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-540', 'start', 'link-224');">start</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1639"></a><tt class="py-lineno">1639</tt>  <tt class="py-line">            <tt id="link-541" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-541', 'start', 'link-224');">start</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">len_self</tt> </tt>
<a name="L1640"></a><tt class="py-lineno">1640</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-542" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-542', 'end', 'link-225');">end</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1641"></a><tt class="py-lineno">1641</tt>  <tt class="py-line">            <tt id="link-543" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-543', 'end', 'link-225');">end</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">len_self</tt> </tt>
<a name="L1642"></a><tt class="py-lineno">1642</tt>  <tt class="py-line">        <tt class="py-comment"># Handle case where end &lt;= start (no negative step argument here)</tt> </tt>
<a name="L1643"></a><tt class="py-lineno">1643</tt>  <tt class="py-line">        <tt class="py-comment"># and case where len_sub_str is larger than the search space</tt> </tt>
<a name="L1644"></a><tt class="py-lineno">1644</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-544" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-544', 'end', 'link-225');">end</a></tt> <tt class="py-op">&lt;=</tt> <tt id="link-545" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-545', 'start', 'link-224');">start</a></tt> <tt class="py-keyword">or</tt> <tt class="py-op">(</tt><tt id="link-546" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-546', 'end', 'link-225');">end</a></tt> <tt class="py-op">-</tt> <tt id="link-547" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-547', 'start', 'link-224');">start</a></tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">len_sub_str</tt><tt class="py-op">:</tt> </tt>
<a name="L1645"></a><tt class="py-lineno">1645</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-number">0</tt> </tt>
<a name="L1646"></a><tt class="py-lineno">1646</tt>  <tt class="py-line">        <tt class="py-comment"># 'Normal' calculation</tt> </tt>
<a name="L1647"></a><tt class="py-lineno">1647</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt id="link-548" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-548', 'end', 'link-225');">end</a></tt> <tt class="py-op">-</tt> <tt id="link-549" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-549', 'start', 'link-224');">start</a></tt><tt class="py-op">)</tt> <tt class="py-op">//</tt> <tt class="py-name">len_sub_str</tt> </tt>
</div><a name="L1648"></a><tt class="py-lineno">1648</tt>  <tt class="py-line"> </tt>
<a name="UnknownSeq.count_overlap"></a><div id="UnknownSeq.count_overlap-def"><a name="L1649"></a><tt class="py-lineno">1649</tt> <a class="py-toggle" href="#" id="UnknownSeq.count_overlap-toggle" onclick="return toggle('UnknownSeq.count_overlap');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.UnknownSeq-class.html#count_overlap">count_overlap</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sub</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">=</tt><tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">maxsize</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnknownSeq.count_overlap-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="UnknownSeq.count_overlap-expanded"><a name="L1650"></a><tt class="py-lineno">1650</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return an overlapping count.</tt> </tt>
<a name="L1651"></a><tt class="py-lineno">1651</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1652"></a><tt class="py-lineno">1652</tt>  <tt class="py-line"><tt class="py-docstring">        For a non-overlapping search use the count() method.</tt> </tt>
<a name="L1653"></a><tt class="py-lineno">1653</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1654"></a><tt class="py-lineno">1654</tt>  <tt class="py-line"><tt class="py-docstring">        Returns an integer, the number of occurrences of substring</tt> </tt>
<a name="L1655"></a><tt class="py-lineno">1655</tt>  <tt class="py-line"><tt class="py-docstring">        argument sub in the (sub)sequence given by [start:end].</tt> </tt>
<a name="L1656"></a><tt class="py-lineno">1656</tt>  <tt class="py-line"><tt class="py-docstring">        Optional arguments start and end are interpreted as in slice</tt> </tt>
<a name="L1657"></a><tt class="py-lineno">1657</tt>  <tt class="py-line"><tt class="py-docstring">        notation.</tt> </tt>
<a name="L1658"></a><tt class="py-lineno">1658</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1659"></a><tt class="py-lineno">1659</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L1660"></a><tt class="py-lineno">1660</tt>  <tt class="py-line"><tt class="py-docstring">         - sub - a string or another Seq object to look for</tt> </tt>
<a name="L1661"></a><tt class="py-lineno">1661</tt>  <tt class="py-line"><tt class="py-docstring">         - start - optional integer, slice start</tt> </tt>
<a name="L1662"></a><tt class="py-lineno">1662</tt>  <tt class="py-line"><tt class="py-docstring">         - end - optional integer, slice end</tt> </tt>
<a name="L1663"></a><tt class="py-lineno">1663</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1664"></a><tt class="py-lineno">1664</tt>  <tt class="py-line"><tt class="py-docstring">        e.g.</tt> </tt>
<a name="L1665"></a><tt class="py-lineno">1665</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1666"></a><tt class="py-lineno">1666</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import UnknownSeq</tt> </tt>
<a name="L1667"></a><tt class="py-lineno">1667</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; UnknownSeq(4, character="N").count_overlap("NN")</tt> </tt>
<a name="L1668"></a><tt class="py-lineno">1668</tt>  <tt class="py-line"><tt class="py-docstring">        3</tt> </tt>
<a name="L1669"></a><tt class="py-lineno">1669</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; UnknownSeq(4, character="N").count_overlap("NNN")</tt> </tt>
<a name="L1670"></a><tt class="py-lineno">1670</tt>  <tt class="py-line"><tt class="py-docstring">        2</tt> </tt>
<a name="L1671"></a><tt class="py-lineno">1671</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1672"></a><tt class="py-lineno">1672</tt>  <tt class="py-line"><tt class="py-docstring">        Where substrings do not overlap, should behave the same as</tt> </tt>
<a name="L1673"></a><tt class="py-lineno">1673</tt>  <tt class="py-line"><tt class="py-docstring">        the count() method:</tt> </tt>
<a name="L1674"></a><tt class="py-lineno">1674</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1675"></a><tt class="py-lineno">1675</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; UnknownSeq(4, character="N").count_overlap("N")</tt> </tt>
<a name="L1676"></a><tt class="py-lineno">1676</tt>  <tt class="py-line"><tt class="py-docstring">        4</tt> </tt>
<a name="L1677"></a><tt class="py-lineno">1677</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; UnknownSeq(4, character="N").count_overlap("N") == UnknownSeq(4, character="N").count("N")</tt> </tt>
<a name="L1678"></a><tt class="py-lineno">1678</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L1679"></a><tt class="py-lineno">1679</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; UnknownSeq(4, character="N").count_overlap("A")</tt> </tt>
<a name="L1680"></a><tt class="py-lineno">1680</tt>  <tt class="py-line"><tt class="py-docstring">        0</tt> </tt>
<a name="L1681"></a><tt class="py-lineno">1681</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; UnknownSeq(4, character="N").count_overlap("A") == UnknownSeq(4, character="N").count("A")</tt> </tt>
<a name="L1682"></a><tt class="py-lineno">1682</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L1683"></a><tt class="py-lineno">1683</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; UnknownSeq(4, character="N").count_overlap("AA")</tt> </tt>
<a name="L1684"></a><tt class="py-lineno">1684</tt>  <tt class="py-line"><tt class="py-docstring">        0</tt> </tt>
<a name="L1685"></a><tt class="py-lineno">1685</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; UnknownSeq(4, character="N").count_overlap("AA") == UnknownSeq(4, character="N").count("AA")</tt> </tt>
<a name="L1686"></a><tt class="py-lineno">1686</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L1687"></a><tt class="py-lineno">1687</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1688"></a><tt class="py-lineno">1688</tt>  <tt class="py-line">        <tt class="py-name">sub_str</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-550" class="py-name"><a title="Bio.Seq.Seq._get_seq_str_and_check_alphabet" class="py-name" href="#" onclick="return doclink('link-550', '_get_seq_str_and_check_alphabet', 'link-222');">_get_seq_str_and_check_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">sub</tt><tt class="py-op">)</tt> </tt>
<a name="L1689"></a><tt class="py-lineno">1689</tt>  <tt class="py-line">        <tt class="py-name">len_self</tt><tt class="py-op">,</tt> <tt class="py-name">len_sub_str</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_length</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">sub_str</tt><tt class="py-op">)</tt> </tt>
<a name="L1690"></a><tt class="py-lineno">1690</tt>  <tt class="py-line">        <tt class="py-comment"># Handling case where substring not in self</tt> </tt>
<a name="L1691"></a><tt class="py-lineno">1691</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-551" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-551', 'set', 'link-526');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">sub_str</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt id="link-552" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-552', 'set', 'link-526');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_character</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1692"></a><tt class="py-lineno">1692</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-number">0</tt> </tt>
<a name="L1693"></a><tt class="py-lineno">1693</tt>  <tt class="py-line">        <tt class="py-comment"># Setting None to the default arguments</tt> </tt>
<a name="L1694"></a><tt class="py-lineno">1694</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-553" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-553', 'start', 'link-224');">start</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1695"></a><tt class="py-lineno">1695</tt>  <tt class="py-line">            <tt id="link-554" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-554', 'start', 'link-224');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L1696"></a><tt class="py-lineno">1696</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-555" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-555', 'end', 'link-225');">end</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1697"></a><tt class="py-lineno">1697</tt>  <tt class="py-line">            <tt id="link-556" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-556', 'end', 'link-225');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">maxsize</tt> </tt>
<a name="L1698"></a><tt class="py-lineno">1698</tt>  <tt class="py-line">        <tt class="py-comment"># Truncating start and end to max of self._length and min of -self._length</tt> </tt>
<a name="L1699"></a><tt class="py-lineno">1699</tt>  <tt class="py-line">        <tt id="link-557" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-557', 'start', 'link-224');">start</a></tt> <tt class="py-op">=</tt> <tt id="link-558" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-558', 'max', 'link-507');">max</a></tt><tt class="py-op">(</tt><tt id="link-559" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-559', 'min', 'link-534');">min</a></tt><tt class="py-op">(</tt><tt id="link-560" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-560', 'start', 'link-224');">start</a></tt><tt class="py-op">,</tt> <tt class="py-name">len_self</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">-</tt><tt class="py-name">len_self</tt><tt class="py-op">)</tt> </tt>
<a name="L1700"></a><tt class="py-lineno">1700</tt>  <tt class="py-line">        <tt id="link-561" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-561', 'end', 'link-225');">end</a></tt> <tt class="py-op">=</tt> <tt id="link-562" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-562', 'max', 'link-507');">max</a></tt><tt class="py-op">(</tt><tt id="link-563" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-563', 'min', 'link-534');">min</a></tt><tt class="py-op">(</tt><tt id="link-564" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-564', 'end', 'link-225');">end</a></tt><tt class="py-op">,</tt> <tt class="py-name">len_self</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">-</tt><tt class="py-name">len_self</tt><tt class="py-op">)</tt> </tt>
<a name="L1701"></a><tt class="py-lineno">1701</tt>  <tt class="py-line">        <tt class="py-comment"># Convert start and ends to positive indexes</tt> </tt>
<a name="L1702"></a><tt class="py-lineno">1702</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-565" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-565', 'start', 'link-224');">start</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1703"></a><tt class="py-lineno">1703</tt>  <tt class="py-line">            <tt id="link-566" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-566', 'start', 'link-224');">start</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">len_self</tt> </tt>
<a name="L1704"></a><tt class="py-lineno">1704</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-567" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-567', 'end', 'link-225');">end</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1705"></a><tt class="py-lineno">1705</tt>  <tt class="py-line">            <tt id="link-568" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-568', 'end', 'link-225');">end</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">len_self</tt> </tt>
<a name="L1706"></a><tt class="py-lineno">1706</tt>  <tt class="py-line">        <tt class="py-comment"># Handle case where end &lt;= start (no negative step argument here)</tt> </tt>
<a name="L1707"></a><tt class="py-lineno">1707</tt>  <tt class="py-line">        <tt class="py-comment"># and case where len_sub_str is larger than the search space</tt> </tt>
<a name="L1708"></a><tt class="py-lineno">1708</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-569" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-569', 'end', 'link-225');">end</a></tt> <tt class="py-op">&lt;=</tt> <tt id="link-570" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-570', 'start', 'link-224');">start</a></tt> <tt class="py-keyword">or</tt> <tt class="py-op">(</tt><tt id="link-571" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-571', 'end', 'link-225');">end</a></tt> <tt class="py-op">-</tt> <tt id="link-572" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-572', 'start', 'link-224');">start</a></tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">len_sub_str</tt><tt class="py-op">:</tt> </tt>
<a name="L1709"></a><tt class="py-lineno">1709</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-number">0</tt> </tt>
<a name="L1710"></a><tt class="py-lineno">1710</tt>  <tt class="py-line">        <tt class="py-comment"># 'Normal' calculation</tt> </tt>
<a name="L1711"></a><tt class="py-lineno">1711</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-573" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-573', 'end', 'link-225');">end</a></tt> <tt class="py-op">-</tt> <tt id="link-574" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-574', 'start', 'link-224');">start</a></tt> <tt class="py-op">-</tt> <tt class="py-name">len_sub_str</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt> </tt>
</div><a name="L1712"></a><tt class="py-lineno">1712</tt>  <tt class="py-line"> </tt>
<a name="UnknownSeq.complement"></a><div id="UnknownSeq.complement-def"><a name="L1713"></a><tt class="py-lineno">1713</tt> <a class="py-toggle" href="#" id="UnknownSeq.complement-toggle" onclick="return toggle('UnknownSeq.complement');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.UnknownSeq-class.html#complement">complement</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnknownSeq.complement-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="UnknownSeq.complement-expanded"><a name="L1714"></a><tt class="py-lineno">1714</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the complement of an unknown nucleotide equals itself.</tt> </tt>
<a name="L1715"></a><tt class="py-lineno">1715</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1716"></a><tt class="py-lineno">1716</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_nuc = UnknownSeq(8)</tt> </tt>
<a name="L1717"></a><tt class="py-lineno">1717</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_nuc</tt> </tt>
<a name="L1718"></a><tt class="py-lineno">1718</tt>  <tt class="py-line"><tt class="py-docstring">        UnknownSeq(8, character='?')</tt> </tt>
<a name="L1719"></a><tt class="py-lineno">1719</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_nuc)</tt> </tt>
<a name="L1720"></a><tt class="py-lineno">1720</tt>  <tt class="py-line"><tt class="py-docstring">        ????????</tt> </tt>
<a name="L1721"></a><tt class="py-lineno">1721</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_nuc.complement()</tt> </tt>
<a name="L1722"></a><tt class="py-lineno">1722</tt>  <tt class="py-line"><tt class="py-docstring">        UnknownSeq(8, character='?')</tt> </tt>
<a name="L1723"></a><tt class="py-lineno">1723</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_nuc.complement())</tt> </tt>
<a name="L1724"></a><tt class="py-lineno">1724</tt>  <tt class="py-line"><tt class="py-docstring">        ????????</tt> </tt>
<a name="L1725"></a><tt class="py-lineno">1725</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1726"></a><tt class="py-lineno">1726</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-575" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-575', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-576" class="py-name"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-576', '_get_base_alphabet', 'link-297');">_get_base_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-577" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-577', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1727"></a><tt class="py-lineno">1727</tt>  <tt class="py-line">                      <tt id="link-578" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-578', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-579" class="py-name"><a title="Bio.Alphabet.ProteinAlphabet" class="py-name" href="#" onclick="return doclink('link-579', 'ProteinAlphabet', 'link-300');">ProteinAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1728"></a><tt class="py-lineno">1728</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Proteins do not have complements!"</tt><tt class="py-op">)</tt> </tt>
<a name="L1729"></a><tt class="py-lineno">1729</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt> </tt>
</div><a name="L1730"></a><tt class="py-lineno">1730</tt>  <tt class="py-line"> </tt>
<a name="UnknownSeq.reverse_complement"></a><div id="UnknownSeq.reverse_complement-def"><a name="L1731"></a><tt class="py-lineno">1731</tt> <a class="py-toggle" href="#" id="UnknownSeq.reverse_complement-toggle" onclick="return toggle('UnknownSeq.reverse_complement');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.UnknownSeq-class.html#reverse_complement">reverse_complement</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnknownSeq.reverse_complement-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="UnknownSeq.reverse_complement-expanded"><a name="L1732"></a><tt class="py-lineno">1732</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the reverse complement of an unknown sequence.</tt> </tt>
<a name="L1733"></a><tt class="py-lineno">1733</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1734"></a><tt class="py-lineno">1734</tt>  <tt class="py-line"><tt class="py-docstring">        The reverse complement of an unknown nucleotide equals itself:</tt> </tt>
<a name="L1735"></a><tt class="py-lineno">1735</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1736"></a><tt class="py-lineno">1736</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import UnknownSeq</tt> </tt>
<a name="L1737"></a><tt class="py-lineno">1737</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_dna</tt> </tt>
<a name="L1738"></a><tt class="py-lineno">1738</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; example = UnknownSeq(6, generic_dna)</tt> </tt>
<a name="L1739"></a><tt class="py-lineno">1739</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(example)</tt> </tt>
<a name="L1740"></a><tt class="py-lineno">1740</tt>  <tt class="py-line"><tt class="py-docstring">        NNNNNN</tt> </tt>
<a name="L1741"></a><tt class="py-lineno">1741</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(example.reverse_complement())</tt> </tt>
<a name="L1742"></a><tt class="py-lineno">1742</tt>  <tt class="py-line"><tt class="py-docstring">        NNNNNN</tt> </tt>
<a name="L1743"></a><tt class="py-lineno">1743</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1744"></a><tt class="py-lineno">1744</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-580" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-580', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-581" class="py-name"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-581', '_get_base_alphabet', 'link-297');">_get_base_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-582" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-582', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1745"></a><tt class="py-lineno">1745</tt>  <tt class="py-line">                      <tt id="link-583" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-583', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-584" class="py-name"><a title="Bio.Alphabet.ProteinAlphabet" class="py-name" href="#" onclick="return doclink('link-584', 'ProteinAlphabet', 'link-300');">ProteinAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1746"></a><tt class="py-lineno">1746</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Proteins do not have complements!"</tt><tt class="py-op">)</tt> </tt>
<a name="L1747"></a><tt class="py-lineno">1747</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt> </tt>
</div><a name="L1748"></a><tt class="py-lineno">1748</tt>  <tt class="py-line"> </tt>
<a name="UnknownSeq.transcribe"></a><div id="UnknownSeq.transcribe-def"><a name="L1749"></a><tt class="py-lineno">1749</tt> <a class="py-toggle" href="#" id="UnknownSeq.transcribe-toggle" onclick="return toggle('UnknownSeq.transcribe');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.UnknownSeq-class.html#transcribe">transcribe</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnknownSeq.transcribe-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="UnknownSeq.transcribe-expanded"><a name="L1750"></a><tt class="py-lineno">1750</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return an unknown RNA sequence from an unknown DNA sequence.</tt> </tt>
<a name="L1751"></a><tt class="py-lineno">1751</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1752"></a><tt class="py-lineno">1752</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna = UnknownSeq(10, character="N")</tt> </tt>
<a name="L1753"></a><tt class="py-lineno">1753</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna</tt> </tt>
<a name="L1754"></a><tt class="py-lineno">1754</tt>  <tt class="py-line"><tt class="py-docstring">        UnknownSeq(10, character='N')</tt> </tt>
<a name="L1755"></a><tt class="py-lineno">1755</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_dna)</tt> </tt>
<a name="L1756"></a><tt class="py-lineno">1756</tt>  <tt class="py-line"><tt class="py-docstring">        NNNNNNNNNN</tt> </tt>
<a name="L1757"></a><tt class="py-lineno">1757</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_rna = my_dna.transcribe()</tt> </tt>
<a name="L1758"></a><tt class="py-lineno">1758</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_rna</tt> </tt>
<a name="L1759"></a><tt class="py-lineno">1759</tt>  <tt class="py-line"><tt class="py-docstring">        UnknownSeq(10, alphabet=RNAAlphabet(), character='N')</tt> </tt>
<a name="L1760"></a><tt class="py-lineno">1760</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_rna)</tt> </tt>
<a name="L1761"></a><tt class="py-lineno">1761</tt>  <tt class="py-line"><tt class="py-docstring">        NNNNNNNNNN</tt> </tt>
<a name="L1762"></a><tt class="py-lineno">1762</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1763"></a><tt class="py-lineno">1763</tt>  <tt class="py-line">        <tt class="py-comment"># Offload the alphabet stuff</tt> </tt>
<a name="L1764"></a><tt class="py-lineno">1764</tt>  <tt class="py-line">        <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt id="link-585" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-585', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_character</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-586" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-586', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-587" class="py-name" targets="Method Bio.Seq.Seq.transcribe()=Bio.Seq.Seq-class.html#transcribe,Method Bio.Seq.UnknownSeq.transcribe()=Bio.Seq.UnknownSeq-class.html#transcribe,Function Bio.Seq.transcribe()=Bio.Seq-module.html#transcribe"><a title="Bio.Seq.Seq.transcribe
Bio.Seq.UnknownSeq.transcribe
Bio.Seq.transcribe" class="py-name" href="#" onclick="return doclink('link-587', 'transcribe', 'link-587');">transcribe</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1765"></a><tt class="py-lineno">1765</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-588" class="py-name"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-588', 'UnknownSeq', 'link-84');">UnknownSeq</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_length</tt><tt class="py-op">,</tt> <tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-589" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-589', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_character</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1766"></a><tt class="py-lineno">1766</tt>  <tt class="py-line"> </tt>
<a name="UnknownSeq.back_transcribe"></a><div id="UnknownSeq.back_transcribe-def"><a name="L1767"></a><tt class="py-lineno">1767</tt> <a class="py-toggle" href="#" id="UnknownSeq.back_transcribe-toggle" onclick="return toggle('UnknownSeq.back_transcribe');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.UnknownSeq-class.html#back_transcribe">back_transcribe</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnknownSeq.back_transcribe-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="UnknownSeq.back_transcribe-expanded"><a name="L1768"></a><tt class="py-lineno">1768</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return an unknown DNA sequence from an unknown RNA sequence.</tt> </tt>
<a name="L1769"></a><tt class="py-lineno">1769</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1770"></a><tt class="py-lineno">1770</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_rna = UnknownSeq(20, character="N")</tt> </tt>
<a name="L1771"></a><tt class="py-lineno">1771</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_rna</tt> </tt>
<a name="L1772"></a><tt class="py-lineno">1772</tt>  <tt class="py-line"><tt class="py-docstring">        UnknownSeq(20, character='N')</tt> </tt>
<a name="L1773"></a><tt class="py-lineno">1773</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_rna)</tt> </tt>
<a name="L1774"></a><tt class="py-lineno">1774</tt>  <tt class="py-line"><tt class="py-docstring">        NNNNNNNNNNNNNNNNNNNN</tt> </tt>
<a name="L1775"></a><tt class="py-lineno">1775</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna = my_rna.back_transcribe()</tt> </tt>
<a name="L1776"></a><tt class="py-lineno">1776</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna</tt> </tt>
<a name="L1777"></a><tt class="py-lineno">1777</tt>  <tt class="py-line"><tt class="py-docstring">        UnknownSeq(20, alphabet=DNAAlphabet(), character='N')</tt> </tt>
<a name="L1778"></a><tt class="py-lineno">1778</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_dna)</tt> </tt>
<a name="L1779"></a><tt class="py-lineno">1779</tt>  <tt class="py-line"><tt class="py-docstring">        NNNNNNNNNNNNNNNNNNNN</tt> </tt>
<a name="L1780"></a><tt class="py-lineno">1780</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1781"></a><tt class="py-lineno">1781</tt>  <tt class="py-line">        <tt class="py-comment"># Offload the alphabet stuff</tt> </tt>
<a name="L1782"></a><tt class="py-lineno">1782</tt>  <tt class="py-line">        <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt id="link-590" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-590', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_character</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-591" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-591', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-592" class="py-name" targets="Method Bio.Seq.Seq.back_transcribe()=Bio.Seq.Seq-class.html#back_transcribe,Method Bio.Seq.UnknownSeq.back_transcribe()=Bio.Seq.UnknownSeq-class.html#back_transcribe,Function Bio.Seq.back_transcribe()=Bio.Seq-module.html#back_transcribe"><a title="Bio.Seq.Seq.back_transcribe
Bio.Seq.UnknownSeq.back_transcribe
Bio.Seq.back_transcribe" class="py-name" href="#" onclick="return doclink('link-592', 'back_transcribe', 'link-592');">back_transcribe</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1783"></a><tt class="py-lineno">1783</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-593" class="py-name"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-593', 'UnknownSeq', 'link-84');">UnknownSeq</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_length</tt><tt class="py-op">,</tt> <tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-594" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-594', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_character</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1784"></a><tt class="py-lineno">1784</tt>  <tt class="py-line"> </tt>
<a name="UnknownSeq.upper"></a><div id="UnknownSeq.upper-def"><a name="L1785"></a><tt class="py-lineno">1785</tt> <a class="py-toggle" href="#" id="UnknownSeq.upper-toggle" onclick="return toggle('UnknownSeq.upper');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.UnknownSeq-class.html#upper">upper</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnknownSeq.upper-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="UnknownSeq.upper-expanded"><a name="L1786"></a><tt class="py-lineno">1786</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return an upper case copy of the sequence.</tt> </tt>
<a name="L1787"></a><tt class="py-lineno">1787</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1788"></a><tt class="py-lineno">1788</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_dna</tt> </tt>
<a name="L1789"></a><tt class="py-lineno">1789</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import UnknownSeq</tt> </tt>
<a name="L1790"></a><tt class="py-lineno">1790</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq = UnknownSeq(20, generic_dna, character="n")</tt> </tt>
<a name="L1791"></a><tt class="py-lineno">1791</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L1792"></a><tt class="py-lineno">1792</tt>  <tt class="py-line"><tt class="py-docstring">        UnknownSeq(20, alphabet=DNAAlphabet(), character='n')</tt> </tt>
<a name="L1793"></a><tt class="py-lineno">1793</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_seq)</tt> </tt>
<a name="L1794"></a><tt class="py-lineno">1794</tt>  <tt class="py-line"><tt class="py-docstring">        nnnnnnnnnnnnnnnnnnnn</tt> </tt>
<a name="L1795"></a><tt class="py-lineno">1795</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.upper()</tt> </tt>
<a name="L1796"></a><tt class="py-lineno">1796</tt>  <tt class="py-line"><tt class="py-docstring">        UnknownSeq(20, alphabet=DNAAlphabet(), character='N')</tt> </tt>
<a name="L1797"></a><tt class="py-lineno">1797</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_seq.upper())</tt> </tt>
<a name="L1798"></a><tt class="py-lineno">1798</tt>  <tt class="py-line"><tt class="py-docstring">        NNNNNNNNNNNNNNNNNNNN</tt> </tt>
<a name="L1799"></a><tt class="py-lineno">1799</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1800"></a><tt class="py-lineno">1800</tt>  <tt class="py-line"><tt class="py-docstring">        This will adjust the alphabet if required. See also the lower method.</tt> </tt>
<a name="L1801"></a><tt class="py-lineno">1801</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1802"></a><tt class="py-lineno">1802</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-595" class="py-name"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-595', 'UnknownSeq', 'link-84');">UnknownSeq</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_length</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-596" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-596', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-597" class="py-name"><a title="Bio.Alphabet.Alphabet._upper
Bio.Alphabet.AlphabetEncoder._upper
Bio.Alphabet.Gapped._upper
Bio.Alphabet.HasStopCodon._upper
Bio.Alphabet.ThreeLetterProtein._upper" class="py-name" href="#" onclick="return doclink('link-597', '_upper', 'link-291');">_upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1803"></a><tt class="py-lineno">1803</tt>  <tt class="py-line">                          <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_character</tt><tt class="py-op">.</tt><tt id="link-598" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-598', 'upper', 'link-289');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1804"></a><tt class="py-lineno">1804</tt>  <tt class="py-line"> </tt>
<a name="UnknownSeq.lower"></a><div id="UnknownSeq.lower-def"><a name="L1805"></a><tt class="py-lineno">1805</tt> <a class="py-toggle" href="#" id="UnknownSeq.lower-toggle" onclick="return toggle('UnknownSeq.lower');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.UnknownSeq-class.html#lower">lower</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnknownSeq.lower-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="UnknownSeq.lower-expanded"><a name="L1806"></a><tt class="py-lineno">1806</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a lower case copy of the sequence.</tt> </tt>
<a name="L1807"></a><tt class="py-lineno">1807</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1808"></a><tt class="py-lineno">1808</tt>  <tt class="py-line"><tt class="py-docstring">        This will adjust the alphabet if required:</tt> </tt>
<a name="L1809"></a><tt class="py-lineno">1809</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1810"></a><tt class="py-lineno">1810</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import IUPAC</tt> </tt>
<a name="L1811"></a><tt class="py-lineno">1811</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import UnknownSeq</tt> </tt>
<a name="L1812"></a><tt class="py-lineno">1812</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq = UnknownSeq(20, IUPAC.extended_protein)</tt> </tt>
<a name="L1813"></a><tt class="py-lineno">1813</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L1814"></a><tt class="py-lineno">1814</tt>  <tt class="py-line"><tt class="py-docstring">        UnknownSeq(20, alphabet=ExtendedIUPACProtein(), character='X')</tt> </tt>
<a name="L1815"></a><tt class="py-lineno">1815</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_seq)</tt> </tt>
<a name="L1816"></a><tt class="py-lineno">1816</tt>  <tt class="py-line"><tt class="py-docstring">        XXXXXXXXXXXXXXXXXXXX</tt> </tt>
<a name="L1817"></a><tt class="py-lineno">1817</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.lower()</tt> </tt>
<a name="L1818"></a><tt class="py-lineno">1818</tt>  <tt class="py-line"><tt class="py-docstring">        UnknownSeq(20, alphabet=ProteinAlphabet(), character='x')</tt> </tt>
<a name="L1819"></a><tt class="py-lineno">1819</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_seq.lower())</tt> </tt>
<a name="L1820"></a><tt class="py-lineno">1820</tt>  <tt class="py-line"><tt class="py-docstring">        xxxxxxxxxxxxxxxxxxxx</tt> </tt>
<a name="L1821"></a><tt class="py-lineno">1821</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1822"></a><tt class="py-lineno">1822</tt>  <tt class="py-line"><tt class="py-docstring">        See also the upper method.</tt> </tt>
<a name="L1823"></a><tt class="py-lineno">1823</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1824"></a><tt class="py-lineno">1824</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-599" class="py-name"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-599', 'UnknownSeq', 'link-84');">UnknownSeq</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_length</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-600" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-600', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-601" class="py-name"><a title="Bio.Alphabet.Alphabet._lower
Bio.Alphabet.AlphabetEncoder._lower
Bio.Alphabet.Gapped._lower
Bio.Alphabet.HasStopCodon._lower
Bio.Alphabet.ThreeLetterProtein._lower" class="py-name" href="#" onclick="return doclink('link-601', '_lower', 'link-295');">_lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1825"></a><tt class="py-lineno">1825</tt>  <tt class="py-line">                          <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_character</tt><tt class="py-op">.</tt><tt id="link-602" class="py-name"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-602', 'lower', 'link-32');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1826"></a><tt class="py-lineno">1826</tt>  <tt class="py-line"> </tt>
<a name="UnknownSeq.translate"></a><div id="UnknownSeq.translate-def"><a name="L1827"></a><tt class="py-lineno">1827</tt> <a class="py-toggle" href="#" id="UnknownSeq.translate-toggle" onclick="return toggle('UnknownSeq.translate');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.UnknownSeq-class.html#translate">translate</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnknownSeq.translate-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="UnknownSeq.translate-expanded"><a name="L1828"></a><tt class="py-lineno">1828</tt>  <tt class="py-line">        <tt class="py-docstring">"""Translate an unknown nucleotide sequence into an unknown protein.</tt> </tt>
<a name="L1829"></a><tt class="py-lineno">1829</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1830"></a><tt class="py-lineno">1830</tt>  <tt class="py-line"><tt class="py-docstring">        e.g.</tt> </tt>
<a name="L1831"></a><tt class="py-lineno">1831</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1832"></a><tt class="py-lineno">1832</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq = UnknownSeq(9, character="N")</tt> </tt>
<a name="L1833"></a><tt class="py-lineno">1833</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_seq)</tt> </tt>
<a name="L1834"></a><tt class="py-lineno">1834</tt>  <tt class="py-line"><tt class="py-docstring">        NNNNNNNNN</tt> </tt>
<a name="L1835"></a><tt class="py-lineno">1835</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_protein = my_seq.translate()</tt> </tt>
<a name="L1836"></a><tt class="py-lineno">1836</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_protein</tt> </tt>
<a name="L1837"></a><tt class="py-lineno">1837</tt>  <tt class="py-line"><tt class="py-docstring">        UnknownSeq(3, alphabet=ProteinAlphabet(), character='X')</tt> </tt>
<a name="L1838"></a><tt class="py-lineno">1838</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_protein)</tt> </tt>
<a name="L1839"></a><tt class="py-lineno">1839</tt>  <tt class="py-line"><tt class="py-docstring">        XXX</tt> </tt>
<a name="L1840"></a><tt class="py-lineno">1840</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1841"></a><tt class="py-lineno">1841</tt>  <tt class="py-line"><tt class="py-docstring">        In comparison, using a normal Seq object:</tt> </tt>
<a name="L1842"></a><tt class="py-lineno">1842</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1843"></a><tt class="py-lineno">1843</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq = Seq("NNNNNNNNN")</tt> </tt>
<a name="L1844"></a><tt class="py-lineno">1844</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_seq)</tt> </tt>
<a name="L1845"></a><tt class="py-lineno">1845</tt>  <tt class="py-line"><tt class="py-docstring">        NNNNNNNNN</tt> </tt>
<a name="L1846"></a><tt class="py-lineno">1846</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_protein = my_seq.translate()</tt> </tt>
<a name="L1847"></a><tt class="py-lineno">1847</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_protein</tt> </tt>
<a name="L1848"></a><tt class="py-lineno">1848</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('XXX', ExtendedIUPACProtein())</tt> </tt>
<a name="L1849"></a><tt class="py-lineno">1849</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_protein)</tt> </tt>
<a name="L1850"></a><tt class="py-lineno">1850</tt>  <tt class="py-line"><tt class="py-docstring">        XXX</tt> </tt>
<a name="L1851"></a><tt class="py-lineno">1851</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1852"></a><tt class="py-lineno">1852</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1853"></a><tt class="py-lineno">1853</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-603" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-603', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-604" class="py-name"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-604', '_get_base_alphabet', 'link-297');">_get_base_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-605" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-605', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1854"></a><tt class="py-lineno">1854</tt>  <tt class="py-line">                      <tt id="link-606" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-606', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-607" class="py-name"><a title="Bio.Alphabet.ProteinAlphabet" class="py-name" href="#" onclick="return doclink('link-607', 'ProteinAlphabet', 'link-300');">ProteinAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1855"></a><tt class="py-lineno">1855</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Proteins cannot be translated!"</tt><tt class="py-op">)</tt> </tt>
<a name="L1856"></a><tt class="py-lineno">1856</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-608" class="py-name"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-608', 'UnknownSeq', 'link-84');">UnknownSeq</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_length</tt> <tt class="py-op">//</tt> <tt class="py-number">3</tt><tt class="py-op">,</tt> <tt id="link-609" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-609', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-610" class="py-name" targets="Variable Bio.Alphabet.generic_protein=Bio.Alphabet-module.html#generic_protein"><a title="Bio.Alphabet.generic_protein" class="py-name" href="#" onclick="return doclink('link-610', 'generic_protein', 'link-610');">generic_protein</a></tt><tt class="py-op">,</tt> <tt class="py-string">"X"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1857"></a><tt class="py-lineno">1857</tt>  <tt class="py-line"> </tt>
<a name="UnknownSeq.ungap"></a><div id="UnknownSeq.ungap-def"><a name="L1858"></a><tt class="py-lineno">1858</tt> <a class="py-toggle" href="#" id="UnknownSeq.ungap-toggle" onclick="return toggle('UnknownSeq.ungap');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.UnknownSeq-class.html#ungap">ungap</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">gap</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnknownSeq.ungap-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="UnknownSeq.ungap-expanded"><a name="L1859"></a><tt class="py-lineno">1859</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a copy of the sequence without the gap character(s).</tt> </tt>
<a name="L1860"></a><tt class="py-lineno">1860</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1861"></a><tt class="py-lineno">1861</tt>  <tt class="py-line"><tt class="py-docstring">        The gap character can be specified in two ways - either as an explicit</tt> </tt>
<a name="L1862"></a><tt class="py-lineno">1862</tt>  <tt class="py-line"><tt class="py-docstring">        argument, or via the sequence's alphabet. For example:</tt> </tt>
<a name="L1863"></a><tt class="py-lineno">1863</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1864"></a><tt class="py-lineno">1864</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import UnknownSeq</tt> </tt>
<a name="L1865"></a><tt class="py-lineno">1865</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import Gapped, generic_dna</tt> </tt>
<a name="L1866"></a><tt class="py-lineno">1866</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna = UnknownSeq(20, Gapped(generic_dna, "-"))</tt> </tt>
<a name="L1867"></a><tt class="py-lineno">1867</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna</tt> </tt>
<a name="L1868"></a><tt class="py-lineno">1868</tt>  <tt class="py-line"><tt class="py-docstring">        UnknownSeq(20, alphabet=Gapped(DNAAlphabet(), '-'), character='N')</tt> </tt>
<a name="L1869"></a><tt class="py-lineno">1869</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna.ungap()</tt> </tt>
<a name="L1870"></a><tt class="py-lineno">1870</tt>  <tt class="py-line"><tt class="py-docstring">        UnknownSeq(20, alphabet=DNAAlphabet(), character='N')</tt> </tt>
<a name="L1871"></a><tt class="py-lineno">1871</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_dna.ungap("-")</tt> </tt>
<a name="L1872"></a><tt class="py-lineno">1872</tt>  <tt class="py-line"><tt class="py-docstring">        UnknownSeq(20, alphabet=DNAAlphabet(), character='N')</tt> </tt>
<a name="L1873"></a><tt class="py-lineno">1873</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1874"></a><tt class="py-lineno">1874</tt>  <tt class="py-line"><tt class="py-docstring">        If the UnknownSeq is using the gap character, then an empty Seq is</tt> </tt>
<a name="L1875"></a><tt class="py-lineno">1875</tt>  <tt class="py-line"><tt class="py-docstring">        returned:</tt> </tt>
<a name="L1876"></a><tt class="py-lineno">1876</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1877"></a><tt class="py-lineno">1877</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_gap = UnknownSeq(20, Gapped(generic_dna, "-"), character="-")</tt> </tt>
<a name="L1878"></a><tt class="py-lineno">1878</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_gap</tt> </tt>
<a name="L1879"></a><tt class="py-lineno">1879</tt>  <tt class="py-line"><tt class="py-docstring">        UnknownSeq(20, alphabet=Gapped(DNAAlphabet(), '-'), character='-')</tt> </tt>
<a name="L1880"></a><tt class="py-lineno">1880</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_gap.ungap()</tt> </tt>
<a name="L1881"></a><tt class="py-lineno">1881</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('', DNAAlphabet())</tt> </tt>
<a name="L1882"></a><tt class="py-lineno">1882</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_gap.ungap("-")</tt> </tt>
<a name="L1883"></a><tt class="py-lineno">1883</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('', DNAAlphabet())</tt> </tt>
<a name="L1884"></a><tt class="py-lineno">1884</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1885"></a><tt class="py-lineno">1885</tt>  <tt class="py-line"><tt class="py-docstring">        Notice that the returned sequence's alphabet is adjusted to remove any</tt> </tt>
<a name="L1886"></a><tt class="py-lineno">1886</tt>  <tt class="py-line"><tt class="py-docstring">        explicit gap character declaration.</tt> </tt>
<a name="L1887"></a><tt class="py-lineno">1887</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1888"></a><tt class="py-lineno">1888</tt>  <tt class="py-line">        <tt class="py-comment"># Offload the alphabet stuff</tt> </tt>
<a name="L1889"></a><tt class="py-lineno">1889</tt>  <tt class="py-line">        <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt id="link-611" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-611', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_character</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-612" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-612', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-613" class="py-name" targets="Method Bio.Seq.Seq.ungap()=Bio.Seq.Seq-class.html#ungap,Method Bio.Seq.UnknownSeq.ungap()=Bio.Seq.UnknownSeq-class.html#ungap,Method Bio.codonalign.codonseq.CodonSeq.ungap()=Bio.codonalign.codonseq.CodonSeq-class.html#ungap"><a title="Bio.Seq.Seq.ungap
Bio.Seq.UnknownSeq.ungap
Bio.codonalign.codonseq.CodonSeq.ungap" class="py-name" href="#" onclick="return doclink('link-613', 'ungap', 'link-613');">ungap</a></tt><tt class="py-op">(</tt><tt class="py-name">gap</tt><tt class="py-op">)</tt> </tt>
<a name="L1890"></a><tt class="py-lineno">1890</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">s</tt><tt class="py-op">:</tt> </tt>
<a name="L1891"></a><tt class="py-lineno">1891</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-614" class="py-name"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-614', 'UnknownSeq', 'link-84');">UnknownSeq</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_length</tt><tt class="py-op">,</tt> <tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-615" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-615', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_character</tt><tt class="py-op">)</tt> </tt>
<a name="L1892"></a><tt class="py-lineno">1892</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1893"></a><tt class="py-lineno">1893</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-616" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-616', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">,</tt> <tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-617" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-617', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1894"></a><tt class="py-lineno">1894</tt>  <tt class="py-line"> </tt>
<a name="UnknownSeq.join"></a><div id="UnknownSeq.join-def"><a name="L1895"></a><tt class="py-lineno">1895</tt> <a class="py-toggle" href="#" id="UnknownSeq.join-toggle" onclick="return toggle('UnknownSeq.join');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.UnknownSeq-class.html#join">join</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnknownSeq.join-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="UnknownSeq.join-expanded"><a name="L1896"></a><tt class="py-lineno">1896</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a merge of the sequences in other, spaced by the sequence from self.</tt> </tt>
<a name="L1897"></a><tt class="py-lineno">1897</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1898"></a><tt class="py-lineno">1898</tt>  <tt class="py-line"><tt class="py-docstring">        Accepts Seq/UnknownSeq objects and Strings as objects to be concatenated with</tt> </tt>
<a name="L1899"></a><tt class="py-lineno">1899</tt>  <tt class="py-line"><tt class="py-docstring">        the spacer</tt> </tt>
<a name="L1900"></a><tt class="py-lineno">1900</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1901"></a><tt class="py-lineno">1901</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; concatenated = UnknownSeq(5).join([Seq("AAA"), Seq("TTT"), Seq("PPP")])</tt> </tt>
<a name="L1902"></a><tt class="py-lineno">1902</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; concatenated</tt> </tt>
<a name="L1903"></a><tt class="py-lineno">1903</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('AAA?????TTT?????PPP')</tt> </tt>
<a name="L1904"></a><tt class="py-lineno">1904</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1905"></a><tt class="py-lineno">1905</tt>  <tt class="py-line"><tt class="py-docstring">        Throws error if other is not an iterable and if objects inside of the iterable</tt> </tt>
<a name="L1906"></a><tt class="py-lineno">1906</tt>  <tt class="py-line"><tt class="py-docstring">        are not Seq/UnknownSeq or String objects.</tt> </tt>
<a name="L1907"></a><tt class="py-lineno">1907</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1908"></a><tt class="py-lineno">1908</tt>  <tt class="py-line"><tt class="py-docstring">        Will only return an UnknownSeq object of all of the objects to be joined are</tt> </tt>
<a name="L1909"></a><tt class="py-lineno">1909</tt>  <tt class="py-line"><tt class="py-docstring">        also UnknownSeqs with the same character as the spacer, similar to how the</tt> </tt>
<a name="L1910"></a><tt class="py-lineno">1910</tt>  <tt class="py-line"><tt class="py-docstring">        addition of an UnknownSeq and another UnknownSeq would work.</tt> </tt>
<a name="L1911"></a><tt class="py-lineno">1911</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1912"></a><tt class="py-lineno">1912</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-618" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-618', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">collections</tt><tt class="py-op">.</tt><tt class="py-name">Iterable</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># doesn't detect single strings</tt> </tt>
<a name="L1913"></a><tt class="py-lineno">1913</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Input must be an iterable"</tt><tt class="py-op">)</tt> </tt>
<a name="L1914"></a><tt class="py-lineno">1914</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-619" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-619', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1915"></a><tt class="py-lineno">1915</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Input must be an iterable"</tt><tt class="py-op">)</tt> </tt>
<a name="L1916"></a><tt class="py-lineno">1916</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt id="link-620" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-620', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-621" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-621', 'SeqRecord', 'link-177');">SeqRecord</a></tt> <tt class="py-keyword">import</tt> <tt id="link-622" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-622', 'SeqRecord', 'link-177');">SeqRecord</a></tt>  <tt class="py-comment"># Lazy to avoid circular imports</tt> </tt>
<a name="L1917"></a><tt class="py-lineno">1917</tt>  <tt class="py-line">        <tt id="link-623" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-623', 'a', 'link-49');">a</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-624" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-624', 'alphabet', 'link-44');">alphabet</a></tt> </tt>
<a name="L1918"></a><tt class="py-lineno">1918</tt>  <tt class="py-line">        <tt class="py-name">type_is_unknown</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L1919"></a><tt class="py-lineno">1919</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">c</tt> <tt class="py-keyword">in</tt> <tt id="link-625" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-625', 'other', 'link-57');">other</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1920"></a><tt class="py-lineno">1920</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">,</tt> <tt id="link-626" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-626', 'SeqRecord', 'link-177');">SeqRecord</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1921"></a><tt class="py-lineno">1921</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"Iterable cannot contain SeqRecords"</tt><tt class="py-op">)</tt> </tt>
<a name="L1922"></a><tt class="py-lineno">1922</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">,</tt> <tt class="py-string">"alphabet"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1923"></a><tt class="py-lineno">1923</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-627" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-627', 'a', 'link-49');">a</a></tt> <tt class="py-op">!=</tt> <tt class="py-name">c</tt><tt class="py-op">.</tt><tt id="link-628" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-628', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1924"></a><tt class="py-lineno">1924</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-629" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-629', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-630" class="py-name"><a title="Bio.Alphabet._check_type_compatible" class="py-name" href="#" onclick="return doclink('link-630', '_check_type_compatible', 'link-59');">_check_type_compatible</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt id="link-631" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-631', 'a', 'link-49');">a</a></tt><tt class="py-op">,</tt> <tt class="py-name">c</tt><tt class="py-op">.</tt><tt id="link-632" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-632', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1925"></a><tt class="py-lineno">1925</tt>  <tt class="py-line">                        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt> </tt>
<a name="L1926"></a><tt class="py-lineno">1926</tt>  <tt class="py-line">                            <tt class="py-string">"Incompatible alphabets {0!r} and {1!r}"</tt><tt class="py-op">.</tt><tt id="link-633" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-633', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1927"></a><tt class="py-lineno">1927</tt>  <tt class="py-line">                                <tt id="link-634" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-634', 'a', 'link-49');">a</a></tt><tt class="py-op">,</tt> <tt class="py-name">c</tt><tt class="py-op">.</tt><tt id="link-635" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-635', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1928"></a><tt class="py-lineno">1928</tt>  <tt class="py-line">                    <tt id="link-636" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-636', 'a', 'link-49');">a</a></tt> <tt class="py-op">=</tt> <tt id="link-637" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-637', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-638" class="py-name"><a title="Bio.Alphabet._consensus_alphabet" class="py-name" href="#" onclick="return doclink('link-638', '_consensus_alphabet', 'link-167');">_consensus_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt id="link-639" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-639', 'a', 'link-49');">a</a></tt><tt class="py-op">,</tt> <tt class="py-name">c</tt><tt class="py-op">.</tt><tt id="link-640" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-640', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1929"></a><tt class="py-lineno">1929</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">,</tt> <tt id="link-641" class="py-name"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-641', 'UnknownSeq', 'link-84');">UnknownSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1930"></a><tt class="py-lineno">1930</tt>  <tt class="py-line">                    <tt class="py-name">type_is_unknown</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L1931"></a><tt class="py-lineno">1931</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1932"></a><tt class="py-lineno">1932</tt>  <tt class="py-line">                <tt class="py-name">type_is_unknown</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L1933"></a><tt class="py-lineno">1933</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1934"></a><tt class="py-lineno">1934</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Input must be an iterable of Seqs or Strings"</tt><tt class="py-op">)</tt> </tt>
<a name="L1935"></a><tt class="py-lineno">1935</tt>  <tt class="py-line">        <tt class="py-name">temp_data</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-642" class="py-name"><a title="Bio.Seq.MutableSeq.join
Bio.Seq.Seq.join
Bio.Seq.UnknownSeq.join" class="py-name" href="#" onclick="return doclink('link-642', 'join', 'link-28');">join</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">z</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">z</tt> <tt class="py-keyword">in</tt> <tt id="link-643" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-643', 'other', 'link-57');">other</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1936"></a><tt class="py-lineno">1936</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">temp_data</tt><tt class="py-op">.</tt><tt id="link-644" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-644', 'count', 'link-223');">count</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_character</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">temp_data</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-name">type_is_unknown</tt> <tt class="py-keyword">is</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L1937"></a><tt class="py-lineno">1937</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">temp_data</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-645" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-645', 'a', 'link-49');">a</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_character</tt><tt class="py-op">)</tt> </tt>
<a name="L1938"></a><tt class="py-lineno">1938</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-646" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-646', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">temp_data</tt><tt class="py-op">,</tt> <tt id="link-647" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-647', 'a', 'link-49');">a</a></tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1939"></a><tt class="py-lineno">1939</tt>  <tt class="py-line"> </tt>
<a name="L1940"></a><tt class="py-lineno">1940</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq"></a><div id="MutableSeq-def"><a name="L1941"></a><tt class="py-lineno">1941</tt> <a class="py-toggle" href="#" id="MutableSeq-toggle" onclick="return toggle('MutableSeq');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html">MutableSeq</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="MutableSeq-expanded"><a name="L1942"></a><tt class="py-lineno">1942</tt>  <tt class="py-line">    <tt class="py-docstring">"""An editable sequence object (with an alphabet).</tt> </tt>
<a name="L1943"></a><tt class="py-lineno">1943</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1944"></a><tt class="py-lineno">1944</tt>  <tt class="py-line"><tt class="py-docstring">    Unlike normal python strings and our basic sequence object (the Seq class)</tt> </tt>
<a name="L1945"></a><tt class="py-lineno">1945</tt>  <tt class="py-line"><tt class="py-docstring">    which are immutable, the MutableSeq lets you edit the sequence in place.</tt> </tt>
<a name="L1946"></a><tt class="py-lineno">1946</tt>  <tt class="py-line"><tt class="py-docstring">    However, this means you cannot use a MutableSeq object as a dictionary key.</tt> </tt>
<a name="L1947"></a><tt class="py-lineno">1947</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1948"></a><tt class="py-lineno">1948</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio.Seq import MutableSeq</tt> </tt>
<a name="L1949"></a><tt class="py-lineno">1949</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio.Alphabet import generic_dna</tt> </tt>
<a name="L1950"></a><tt class="py-lineno">1950</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; my_seq = MutableSeq("ACTCGTCGTCG", generic_dna)</tt> </tt>
<a name="L1951"></a><tt class="py-lineno">1951</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L1952"></a><tt class="py-lineno">1952</tt>  <tt class="py-line"><tt class="py-docstring">    MutableSeq('ACTCGTCGTCG', DNAAlphabet())</tt> </tt>
<a name="L1953"></a><tt class="py-lineno">1953</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; my_seq[5]</tt> </tt>
<a name="L1954"></a><tt class="py-lineno">1954</tt>  <tt class="py-line"><tt class="py-docstring">    'T'</tt> </tt>
<a name="L1955"></a><tt class="py-lineno">1955</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; my_seq[5] = "A"</tt> </tt>
<a name="L1956"></a><tt class="py-lineno">1956</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L1957"></a><tt class="py-lineno">1957</tt>  <tt class="py-line"><tt class="py-docstring">    MutableSeq('ACTCGACGTCG', DNAAlphabet())</tt> </tt>
<a name="L1958"></a><tt class="py-lineno">1958</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; my_seq[5]</tt> </tt>
<a name="L1959"></a><tt class="py-lineno">1959</tt>  <tt class="py-line"><tt class="py-docstring">    'A'</tt> </tt>
<a name="L1960"></a><tt class="py-lineno">1960</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; my_seq[5:8] = "NNN"</tt> </tt>
<a name="L1961"></a><tt class="py-lineno">1961</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L1962"></a><tt class="py-lineno">1962</tt>  <tt class="py-line"><tt class="py-docstring">    MutableSeq('ACTCGNNNTCG', DNAAlphabet())</tt> </tt>
<a name="L1963"></a><tt class="py-lineno">1963</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; len(my_seq)</tt> </tt>
<a name="L1964"></a><tt class="py-lineno">1964</tt>  <tt class="py-line"><tt class="py-docstring">    11</tt> </tt>
<a name="L1965"></a><tt class="py-lineno">1965</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1966"></a><tt class="py-lineno">1966</tt>  <tt class="py-line"><tt class="py-docstring">    Note that the MutableSeq object does not support as many string-like</tt> </tt>
<a name="L1967"></a><tt class="py-lineno">1967</tt>  <tt class="py-line"><tt class="py-docstring">    or biological methods as the Seq object.</tt> </tt>
<a name="L1968"></a><tt class="py-lineno">1968</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1969"></a><tt class="py-lineno">1969</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.__init__"></a><div id="MutableSeq.__init__-def"><a name="L1970"></a><tt class="py-lineno">1970</tt> <a class="py-toggle" href="#" id="MutableSeq.__init__-toggle" onclick="return toggle('MutableSeq.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">data</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt id="link-648" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-648', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-649" class="py-name"><a title="Bio.Alphabet.generic_alphabet" class="py-name" href="#" onclick="return doclink('link-649', 'generic_alphabet', 'link-41');">generic_alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.__init__-expanded"><a name="L1971"></a><tt class="py-lineno">1971</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the class."""</tt> </tt>
<a name="L1972"></a><tt class="py-lineno">1972</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">version_info</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-number">3</tt><tt class="py-op">:</tt> </tt>
<a name="L1973"></a><tt class="py-lineno">1973</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">array_indicator</tt> <tt class="py-op">=</tt> <tt class="py-string">"u"</tt> </tt>
<a name="L1974"></a><tt class="py-lineno">1974</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1975"></a><tt class="py-lineno">1975</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">array_indicator</tt> <tt class="py-op">=</tt> <tt class="py-string">"c"</tt> </tt>
<a name="L1976"></a><tt class="py-lineno">1976</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-650" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-650', 'data', 'link-42');">data</a></tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># TODO - What about unicode?</tt> </tt>
<a name="L1977"></a><tt class="py-lineno">1977</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-651" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-651', 'data', 'link-42');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">array</tt><tt class="py-op">.</tt><tt class="py-name">array</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">array_indicator</tt><tt class="py-op">,</tt> <tt id="link-652" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-652', 'data', 'link-42');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1978"></a><tt class="py-lineno">1978</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-653" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-653', 'data', 'link-42');">data</a></tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt id="link-654" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-654', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">,</tt> <tt class="py-name">float</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1979"></a><tt class="py-lineno">1979</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"The sequence data given to a MutableSeq object "</tt> </tt>
<a name="L1980"></a><tt class="py-lineno">1980</tt>  <tt class="py-line">                            <tt class="py-string">"should be a string or an array "</tt> </tt>
<a name="L1981"></a><tt class="py-lineno">1981</tt>  <tt class="py-line">                            <tt class="py-string">"(not a Seq object etc)"</tt><tt class="py-op">)</tt> </tt>
<a name="L1982"></a><tt class="py-lineno">1982</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1983"></a><tt class="py-lineno">1983</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-655" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-655', 'data', 'link-42');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-656" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-656', 'data', 'link-42');">data</a></tt>   <tt class="py-comment"># assumes the input is an array</tt> </tt>
<a name="L1984"></a><tt class="py-lineno">1984</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-657" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-657', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-658" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-658', 'alphabet', 'link-44');">alphabet</a></tt> </tt>
</div><a name="L1985"></a><tt class="py-lineno">1985</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.__repr__"></a><div id="MutableSeq.__repr__-def"><a name="L1986"></a><tt class="py-lineno">1986</tt> <a class="py-toggle" href="#" id="MutableSeq.__repr__-toggle" onclick="return toggle('MutableSeq.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.__repr__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.__repr__-expanded"><a name="L1987"></a><tt class="py-lineno">1987</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return (truncated) representation of the sequence for debugging."""</tt> </tt>
<a name="L1988"></a><tt class="py-lineno">1988</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-659" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-659', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-keyword">is</tt> <tt id="link-660" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-660', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-661" class="py-name"><a title="Bio.Alphabet.generic_alphabet" class="py-name" href="#" onclick="return doclink('link-661', 'generic_alphabet', 'link-41');">generic_alphabet</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1989"></a><tt class="py-lineno">1989</tt>  <tt class="py-line">            <tt class="py-comment"># Default used, we can omit it and simplify the representation</tt> </tt>
<a name="L1990"></a><tt class="py-lineno">1990</tt>  <tt class="py-line">            <tt id="link-662" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-662', 'a', 'link-49');">a</a></tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L1991"></a><tt class="py-lineno">1991</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1992"></a><tt class="py-lineno">1992</tt>  <tt class="py-line">            <tt id="link-663" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-663', 'a', 'link-49');">a</a></tt> <tt class="py-op">=</tt> <tt class="py-string">", %r"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-664" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-664', 'alphabet', 'link-44');">alphabet</a></tt> </tt>
<a name="L1993"></a><tt class="py-lineno">1993</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">60</tt><tt class="py-op">:</tt> </tt>
<a name="L1994"></a><tt class="py-lineno">1994</tt>  <tt class="py-line">            <tt class="py-comment"># Shows the last three letters as it is often useful to see if</tt> </tt>
<a name="L1995"></a><tt class="py-lineno">1995</tt>  <tt class="py-line">            <tt class="py-comment"># there is a stop codon at the end of a sequence.</tt> </tt>
<a name="L1996"></a><tt class="py-lineno">1996</tt>  <tt class="py-line">            <tt class="py-comment"># Note total length is 54+3+3=60</tt> </tt>
<a name="L1997"></a><tt class="py-lineno">1997</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-string">"{0}('{1}...{2}'{3!s})"</tt><tt class="py-op">.</tt><tt id="link-665" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-665', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">,</tt> </tt>
<a name="L1998"></a><tt class="py-lineno">1998</tt>  <tt class="py-line">                                                  <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">54</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1999"></a><tt class="py-lineno">1999</tt>  <tt class="py-line">                                                  <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">3</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L2000"></a><tt class="py-lineno">2000</tt>  <tt class="py-line">                                                  <tt id="link-666" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-666', 'a', 'link-49');">a</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2001"></a><tt class="py-lineno">2001</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L2002"></a><tt class="py-lineno">2002</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-string">"{0}('{1}'{2!s})"</tt><tt class="py-op">.</tt><tt id="link-667" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-667', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">,</tt> </tt>
<a name="L2003"></a><tt class="py-lineno">2003</tt>  <tt class="py-line">                                            <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L2004"></a><tt class="py-lineno">2004</tt>  <tt class="py-line">                                            <tt id="link-668" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-668', 'a', 'link-49');">a</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L2005"></a><tt class="py-lineno">2005</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.__str__"></a><div id="MutableSeq.__str__-def"><a name="L2006"></a><tt class="py-lineno">2006</tt> <a class="py-toggle" href="#" id="MutableSeq.__str__-toggle" onclick="return toggle('MutableSeq.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.__str__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.__str__-expanded"><a name="L2007"></a><tt class="py-lineno">2007</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the full sequence as a python string.</tt> </tt>
<a name="L2008"></a><tt class="py-lineno">2008</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2009"></a><tt class="py-lineno">2009</tt>  <tt class="py-line"><tt class="py-docstring">        Note that Biopython 1.44 and earlier would give a truncated</tt> </tt>
<a name="L2010"></a><tt class="py-lineno">2010</tt>  <tt class="py-line"><tt class="py-docstring">        version of repr(my_seq) for str(my_seq).  If you are writing code</tt> </tt>
<a name="L2011"></a><tt class="py-lineno">2011</tt>  <tt class="py-line"><tt class="py-docstring">        which needs to be backwards compatible with old Biopython, you</tt> </tt>
<a name="L2012"></a><tt class="py-lineno">2012</tt>  <tt class="py-line"><tt class="py-docstring">        should continue to use my_seq.tostring() rather than str(my_seq).</tt> </tt>
<a name="L2013"></a><tt class="py-lineno">2013</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2014"></a><tt class="py-lineno">2014</tt>  <tt class="py-line">        <tt class="py-comment"># See test_GAQueens.py for an historic usage of a non-string alphabet!</tt> </tt>
<a name="L2015"></a><tt class="py-lineno">2015</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt id="link-669" class="py-name"><a title="Bio.Seq.MutableSeq.join
Bio.Seq.Seq.join
Bio.Seq.UnknownSeq.join" class="py-name" href="#" onclick="return doclink('link-669', 'join', 'link-28');">join</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-670" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-670', 'data', 'link-42');">data</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L2016"></a><tt class="py-lineno">2016</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.__eq__"></a><div id="MutableSeq.__eq__-def"><a name="L2017"></a><tt class="py-lineno">2017</tt> <a class="py-toggle" href="#" id="MutableSeq.__eq__-toggle" onclick="return toggle('MutableSeq.__eq__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#__eq__">__eq__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.__eq__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.__eq__-expanded"><a name="L2018"></a><tt class="py-lineno">2018</tt>  <tt class="py-line">        <tt class="py-docstring">"""Compare the sequence to another sequence or a string (README).</tt> </tt>
<a name="L2019"></a><tt class="py-lineno">2019</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2020"></a><tt class="py-lineno">2020</tt>  <tt class="py-line"><tt class="py-docstring">        Currently if compared to another sequence the alphabets must be</tt> </tt>
<a name="L2021"></a><tt class="py-lineno">2021</tt>  <tt class="py-line"><tt class="py-docstring">        compatible. Comparing DNA to RNA, or Nucleotide to Protein will raise</tt> </tt>
<a name="L2022"></a><tt class="py-lineno">2022</tt>  <tt class="py-line"><tt class="py-docstring">        an exception. Otherwise only the sequence itself is compared, not the</tt> </tt>
<a name="L2023"></a><tt class="py-lineno">2023</tt>  <tt class="py-line"><tt class="py-docstring">        precise alphabet.</tt> </tt>
<a name="L2024"></a><tt class="py-lineno">2024</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2025"></a><tt class="py-lineno">2025</tt>  <tt class="py-line"><tt class="py-docstring">        A future release of Biopython will change this (and the Seq object etc)</tt> </tt>
<a name="L2026"></a><tt class="py-lineno">2026</tt>  <tt class="py-line"><tt class="py-docstring">        to use simple string comparison. The plan is that comparing sequences</tt> </tt>
<a name="L2027"></a><tt class="py-lineno">2027</tt>  <tt class="py-line"><tt class="py-docstring">        with incompatible alphabets (e.g. DNA to RNA) will trigger a warning</tt> </tt>
<a name="L2028"></a><tt class="py-lineno">2028</tt>  <tt class="py-line"><tt class="py-docstring">        but not an exception.</tt> </tt>
<a name="L2029"></a><tt class="py-lineno">2029</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2030"></a><tt class="py-lineno">2030</tt>  <tt class="py-line"><tt class="py-docstring">        During this transition period, please just do explicit comparisons:</tt> </tt>
<a name="L2031"></a><tt class="py-lineno">2031</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2032"></a><tt class="py-lineno">2032</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; seq1 = MutableSeq("ACGT")</tt> </tt>
<a name="L2033"></a><tt class="py-lineno">2033</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; seq2 = MutableSeq("ACGT")</tt> </tt>
<a name="L2034"></a><tt class="py-lineno">2034</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; id(seq1) == id(seq2)</tt> </tt>
<a name="L2035"></a><tt class="py-lineno">2035</tt>  <tt class="py-line"><tt class="py-docstring">        False</tt> </tt>
<a name="L2036"></a><tt class="py-lineno">2036</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; str(seq1) == str(seq2)</tt> </tt>
<a name="L2037"></a><tt class="py-lineno">2037</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L2038"></a><tt class="py-lineno">2038</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2039"></a><tt class="py-lineno">2039</tt>  <tt class="py-line"><tt class="py-docstring">        Biopython now does:</tt> </tt>
<a name="L2040"></a><tt class="py-lineno">2040</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2041"></a><tt class="py-lineno">2041</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; seq1 == seq2</tt> </tt>
<a name="L2042"></a><tt class="py-lineno">2042</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L2043"></a><tt class="py-lineno">2043</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; seq1 == Seq("ACGT")</tt> </tt>
<a name="L2044"></a><tt class="py-lineno">2044</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L2045"></a><tt class="py-lineno">2045</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; seq1 == "ACGT"</tt> </tt>
<a name="L2046"></a><tt class="py-lineno">2046</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L2047"></a><tt class="py-lineno">2047</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2048"></a><tt class="py-lineno">2048</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2049"></a><tt class="py-lineno">2049</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-671" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-671', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-string">"alphabet"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2050"></a><tt class="py-lineno">2050</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-672" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-672', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-673" class="py-name"><a title="Bio.Alphabet._check_type_compatible" class="py-name" href="#" onclick="return doclink('link-673', '_check_type_compatible', 'link-59');">_check_type_compatible</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-674" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-674', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L2051"></a><tt class="py-lineno">2051</tt>  <tt class="py-line">                                                    <tt id="link-675" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-675', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-676" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-676', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2052"></a><tt class="py-lineno">2052</tt>  <tt class="py-line">                <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Incompatible alphabets {0!r} and {1!r}"</tt><tt class="py-op">.</tt><tt id="link-677" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-677', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L2053"></a><tt class="py-lineno">2053</tt>  <tt class="py-line">                              <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-678" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-678', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-679" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-679', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-680" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-680', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L2054"></a><tt class="py-lineno">2054</tt>  <tt class="py-line">                              <tt id="link-681" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-681', 'BiopythonWarning', 'link-6');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2055"></a><tt class="py-lineno">2055</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-682" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-682', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt id="link-683" class="py-name"><a title="Bio.Seq.MutableSeq" class="py-name" href="#" onclick="return doclink('link-683', 'MutableSeq', 'link-83');">MutableSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2056"></a><tt class="py-lineno">2056</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-684" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-684', 'data', 'link-42');">data</a></tt> <tt class="py-op">==</tt> <tt id="link-685" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-685', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-686" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-686', 'data', 'link-42');">data</a></tt> </tt>
<a name="L2057"></a><tt class="py-lineno">2057</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-687" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-687', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L2058"></a><tt class="py-lineno">2058</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.__ne__"></a><div id="MutableSeq.__ne__-def"><a name="L2059"></a><tt class="py-lineno">2059</tt> <a class="py-toggle" href="#" id="MutableSeq.__ne__-toggle" onclick="return toggle('MutableSeq.__ne__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#__ne__">__ne__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.__ne__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.__ne__-expanded"><a name="L2060"></a><tt class="py-lineno">2060</tt>  <tt class="py-line">        <tt class="py-docstring">"""Implement the not-equal operand."""</tt> </tt>
<a name="L2061"></a><tt class="py-lineno">2061</tt>  <tt class="py-line">        <tt class="py-comment"># Seem to require this method for Python 2 but not needed on Python 3?</tt> </tt>
<a name="L2062"></a><tt class="py-lineno">2062</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-keyword">not</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt> <tt class="py-op">==</tt> <tt id="link-688" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-688', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L2063"></a><tt class="py-lineno">2063</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.__lt__"></a><div id="MutableSeq.__lt__-def"><a name="L2064"></a><tt class="py-lineno">2064</tt> <a class="py-toggle" href="#" id="MutableSeq.__lt__-toggle" onclick="return toggle('MutableSeq.__lt__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#__lt__">__lt__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.__lt__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.__lt__-expanded"><a name="L2065"></a><tt class="py-lineno">2065</tt>  <tt class="py-line">        <tt class="py-docstring">"""Implement the less-than operand."""</tt> </tt>
<a name="L2066"></a><tt class="py-lineno">2066</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-689" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-689', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-string">"alphabet"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2067"></a><tt class="py-lineno">2067</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-690" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-690', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-691" class="py-name"><a title="Bio.Alphabet._check_type_compatible" class="py-name" href="#" onclick="return doclink('link-691', '_check_type_compatible', 'link-59');">_check_type_compatible</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-692" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-692', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L2068"></a><tt class="py-lineno">2068</tt>  <tt class="py-line">                                                    <tt id="link-693" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-693', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-694" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-694', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2069"></a><tt class="py-lineno">2069</tt>  <tt class="py-line">                <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Incompatible alphabets {0!r} and {1!r}"</tt><tt class="py-op">.</tt><tt id="link-695" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-695', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L2070"></a><tt class="py-lineno">2070</tt>  <tt class="py-line">                              <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-696" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-696', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-697" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-697', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-698" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-698', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L2071"></a><tt class="py-lineno">2071</tt>  <tt class="py-line">                              <tt id="link-699" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-699', 'BiopythonWarning', 'link-6');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2072"></a><tt class="py-lineno">2072</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-700" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-700', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt id="link-701" class="py-name"><a title="Bio.Seq.MutableSeq" class="py-name" href="#" onclick="return doclink('link-701', 'MutableSeq', 'link-83');">MutableSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2073"></a><tt class="py-lineno">2073</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-702" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-702', 'data', 'link-42');">data</a></tt> <tt class="py-op">&lt;</tt> <tt id="link-703" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-703', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-704" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-704', 'data', 'link-42');">data</a></tt> </tt>
<a name="L2074"></a><tt class="py-lineno">2074</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-705" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-705', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">,</tt> <tt id="link-706" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-706', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">,</tt> <tt id="link-707" class="py-name"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-707', 'UnknownSeq', 'link-84');">UnknownSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2075"></a><tt class="py-lineno">2075</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-708" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-708', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2076"></a><tt class="py-lineno">2076</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"'&lt;' not supported between instances of '{}' and '{}'"</tt> </tt>
<a name="L2077"></a><tt class="py-lineno">2077</tt>  <tt class="py-line">                        <tt class="py-op">.</tt><tt id="link-709" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-709', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt id="link-710" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-710', 'type', 'link-87');">type</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">,</tt> <tt id="link-711" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-711', 'type', 'link-87');">type</a></tt><tt class="py-op">(</tt><tt id="link-712" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-712', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L2078"></a><tt class="py-lineno">2078</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.__le__"></a><div id="MutableSeq.__le__-def"><a name="L2079"></a><tt class="py-lineno">2079</tt> <a class="py-toggle" href="#" id="MutableSeq.__le__-toggle" onclick="return toggle('MutableSeq.__le__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#__le__">__le__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.__le__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.__le__-expanded"><a name="L2080"></a><tt class="py-lineno">2080</tt>  <tt class="py-line">        <tt class="py-docstring">"""Implement the less-than or equal operand."""</tt> </tt>
<a name="L2081"></a><tt class="py-lineno">2081</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-713" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-713', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-string">"alphabet"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2082"></a><tt class="py-lineno">2082</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-714" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-714', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-715" class="py-name"><a title="Bio.Alphabet._check_type_compatible" class="py-name" href="#" onclick="return doclink('link-715', '_check_type_compatible', 'link-59');">_check_type_compatible</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-716" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-716', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L2083"></a><tt class="py-lineno">2083</tt>  <tt class="py-line">                                                    <tt id="link-717" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-717', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-718" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-718', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2084"></a><tt class="py-lineno">2084</tt>  <tt class="py-line">                <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Incompatible alphabets {0!r} and {1!r}"</tt><tt class="py-op">.</tt><tt id="link-719" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-719', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L2085"></a><tt class="py-lineno">2085</tt>  <tt class="py-line">                              <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-720" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-720', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-721" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-721', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-722" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-722', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L2086"></a><tt class="py-lineno">2086</tt>  <tt class="py-line">                              <tt id="link-723" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-723', 'BiopythonWarning', 'link-6');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2087"></a><tt class="py-lineno">2087</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-724" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-724', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt id="link-725" class="py-name"><a title="Bio.Seq.MutableSeq" class="py-name" href="#" onclick="return doclink('link-725', 'MutableSeq', 'link-83');">MutableSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2088"></a><tt class="py-lineno">2088</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-726" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-726', 'data', 'link-42');">data</a></tt> <tt class="py-op">&lt;=</tt> <tt id="link-727" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-727', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-728" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-728', 'data', 'link-42');">data</a></tt> </tt>
<a name="L2089"></a><tt class="py-lineno">2089</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-729" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-729', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">,</tt> <tt id="link-730" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-730', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">,</tt> <tt id="link-731" class="py-name"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-731', 'UnknownSeq', 'link-84');">UnknownSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2090"></a><tt class="py-lineno">2090</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-732" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-732', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2091"></a><tt class="py-lineno">2091</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"'&lt;=' not supported between instances of '{}' and '{}'"</tt> </tt>
<a name="L2092"></a><tt class="py-lineno">2092</tt>  <tt class="py-line">                        <tt class="py-op">.</tt><tt id="link-733" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-733', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt id="link-734" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-734', 'type', 'link-87');">type</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">,</tt> <tt id="link-735" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-735', 'type', 'link-87');">type</a></tt><tt class="py-op">(</tt><tt id="link-736" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-736', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L2093"></a><tt class="py-lineno">2093</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.__gt__"></a><div id="MutableSeq.__gt__-def"><a name="L2094"></a><tt class="py-lineno">2094</tt> <a class="py-toggle" href="#" id="MutableSeq.__gt__-toggle" onclick="return toggle('MutableSeq.__gt__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#__gt__">__gt__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.__gt__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.__gt__-expanded"><a name="L2095"></a><tt class="py-lineno">2095</tt>  <tt class="py-line">        <tt class="py-docstring">"""Implement the greater-than operand."""</tt> </tt>
<a name="L2096"></a><tt class="py-lineno">2096</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-737" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-737', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-string">"alphabet"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2097"></a><tt class="py-lineno">2097</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-738" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-738', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-739" class="py-name"><a title="Bio.Alphabet._check_type_compatible" class="py-name" href="#" onclick="return doclink('link-739', '_check_type_compatible', 'link-59');">_check_type_compatible</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-740" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-740', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L2098"></a><tt class="py-lineno">2098</tt>  <tt class="py-line">                                                    <tt id="link-741" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-741', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-742" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-742', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2099"></a><tt class="py-lineno">2099</tt>  <tt class="py-line">                <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Incompatible alphabets {0!r} and {1!r}"</tt><tt class="py-op">.</tt><tt id="link-743" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-743', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L2100"></a><tt class="py-lineno">2100</tt>  <tt class="py-line">                              <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-744" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-744', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-745" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-745', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-746" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-746', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L2101"></a><tt class="py-lineno">2101</tt>  <tt class="py-line">                              <tt id="link-747" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-747', 'BiopythonWarning', 'link-6');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2102"></a><tt class="py-lineno">2102</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-748" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-748', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt id="link-749" class="py-name"><a title="Bio.Seq.MutableSeq" class="py-name" href="#" onclick="return doclink('link-749', 'MutableSeq', 'link-83');">MutableSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2103"></a><tt class="py-lineno">2103</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-750" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-750', 'data', 'link-42');">data</a></tt> <tt class="py-op">&gt;</tt> <tt id="link-751" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-751', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-752" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-752', 'data', 'link-42');">data</a></tt> </tt>
<a name="L2104"></a><tt class="py-lineno">2104</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-753" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-753', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">,</tt> <tt id="link-754" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-754', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">,</tt> <tt id="link-755" class="py-name"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-755', 'UnknownSeq', 'link-84');">UnknownSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2105"></a><tt class="py-lineno">2105</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-756" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-756', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2106"></a><tt class="py-lineno">2106</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"'&gt;' not supported between instances of '{}' and '{}'"</tt> </tt>
<a name="L2107"></a><tt class="py-lineno">2107</tt>  <tt class="py-line">                        <tt class="py-op">.</tt><tt id="link-757" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-757', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt id="link-758" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-758', 'type', 'link-87');">type</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">,</tt> <tt id="link-759" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-759', 'type', 'link-87');">type</a></tt><tt class="py-op">(</tt><tt id="link-760" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-760', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L2108"></a><tt class="py-lineno">2108</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.__ge__"></a><div id="MutableSeq.__ge__-def"><a name="L2109"></a><tt class="py-lineno">2109</tt> <a class="py-toggle" href="#" id="MutableSeq.__ge__-toggle" onclick="return toggle('MutableSeq.__ge__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#__ge__">__ge__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.__ge__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.__ge__-expanded"><a name="L2110"></a><tt class="py-lineno">2110</tt>  <tt class="py-line">        <tt class="py-docstring">"""Implement the greater-than or equal operand."""</tt> </tt>
<a name="L2111"></a><tt class="py-lineno">2111</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-761" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-761', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-string">"alphabet"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2112"></a><tt class="py-lineno">2112</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-762" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-762', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-763" class="py-name"><a title="Bio.Alphabet._check_type_compatible" class="py-name" href="#" onclick="return doclink('link-763', '_check_type_compatible', 'link-59');">_check_type_compatible</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-764" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-764', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L2113"></a><tt class="py-lineno">2113</tt>  <tt class="py-line">                                                    <tt id="link-765" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-765', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-766" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-766', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2114"></a><tt class="py-lineno">2114</tt>  <tt class="py-line">                <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Incompatible alphabets {0!r} and {1!r}"</tt><tt class="py-op">.</tt><tt id="link-767" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-767', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L2115"></a><tt class="py-lineno">2115</tt>  <tt class="py-line">                              <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-768" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-768', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-769" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-769', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-770" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-770', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L2116"></a><tt class="py-lineno">2116</tt>  <tt class="py-line">                              <tt id="link-771" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-771', 'BiopythonWarning', 'link-6');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2117"></a><tt class="py-lineno">2117</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-772" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-772', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt id="link-773" class="py-name"><a title="Bio.Seq.MutableSeq" class="py-name" href="#" onclick="return doclink('link-773', 'MutableSeq', 'link-83');">MutableSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2118"></a><tt class="py-lineno">2118</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-774" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-774', 'data', 'link-42');">data</a></tt> <tt class="py-op">&gt;=</tt> <tt id="link-775" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-775', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-776" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-776', 'data', 'link-42');">data</a></tt> </tt>
<a name="L2119"></a><tt class="py-lineno">2119</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-777" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-777', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">,</tt> <tt id="link-778" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-778', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">,</tt> <tt id="link-779" class="py-name"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-779', 'UnknownSeq', 'link-84');">UnknownSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2120"></a><tt class="py-lineno">2120</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-780" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-780', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2121"></a><tt class="py-lineno">2121</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"'&gt;=' not supported between instances of '{}' and '{}'"</tt> </tt>
<a name="L2122"></a><tt class="py-lineno">2122</tt>  <tt class="py-line">                        <tt class="py-op">.</tt><tt id="link-781" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-781', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt id="link-782" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-782', 'type', 'link-87');">type</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">,</tt> <tt id="link-783" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-783', 'type', 'link-87');">type</a></tt><tt class="py-op">(</tt><tt id="link-784" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-784', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L2123"></a><tt class="py-lineno">2123</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.__len__"></a><div id="MutableSeq.__len__-def"><a name="L2124"></a><tt class="py-lineno">2124</tt> <a class="py-toggle" href="#" id="MutableSeq.__len__-toggle" onclick="return toggle('MutableSeq.__len__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#__len__">__len__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.__len__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.__len__-expanded"><a name="L2125"></a><tt class="py-lineno">2125</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the length of the sequence, use len(my_seq)."""</tt> </tt>
<a name="L2126"></a><tt class="py-lineno">2126</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-785" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-785', 'data', 'link-42');">data</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L2127"></a><tt class="py-lineno">2127</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.__getitem__"></a><div id="MutableSeq.__getitem__-def"><a name="L2128"></a><tt class="py-lineno">2128</tt> <a class="py-toggle" href="#" id="MutableSeq.__getitem__-toggle" onclick="return toggle('MutableSeq.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">index</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.__getitem__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.__getitem__-expanded"><a name="L2129"></a><tt class="py-lineno">2129</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a subsequence of single letter, use my_seq[index].</tt> </tt>
<a name="L2130"></a><tt class="py-lineno">2130</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2131"></a><tt class="py-lineno">2131</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq = MutableSeq('ACTCGACGTCG')</tt> </tt>
<a name="L2132"></a><tt class="py-lineno">2132</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq[5]</tt> </tt>
<a name="L2133"></a><tt class="py-lineno">2133</tt>  <tt class="py-line"><tt class="py-docstring">        'A'</tt> </tt>
<a name="L2134"></a><tt class="py-lineno">2134</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2135"></a><tt class="py-lineno">2135</tt>  <tt class="py-line">        <tt class="py-comment"># Note since Python 2.0, __getslice__ is deprecated</tt> </tt>
<a name="L2136"></a><tt class="py-lineno">2136</tt>  <tt class="py-line">        <tt class="py-comment"># and __getitem__ is used instead.</tt> </tt>
<a name="L2137"></a><tt class="py-lineno">2137</tt>  <tt class="py-line">        <tt class="py-comment"># See http://docs.python.org/ref/sequence-methods.html</tt> </tt>
<a name="L2138"></a><tt class="py-lineno">2138</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-786" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.Seq.Seq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-786', 'index', 'link-150');">index</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2139"></a><tt class="py-lineno">2139</tt>  <tt class="py-line">            <tt class="py-comment"># Return a single letter as a string</tt> </tt>
<a name="L2140"></a><tt class="py-lineno">2140</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-787" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-787', 'data', 'link-42');">data</a></tt><tt class="py-op">[</tt><tt id="link-788" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.Seq.Seq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-788', 'index', 'link-150');">index</a></tt><tt class="py-op">]</tt> </tt>
<a name="L2141"></a><tt class="py-lineno">2141</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L2142"></a><tt class="py-lineno">2142</tt>  <tt class="py-line">            <tt class="py-comment"># Return the (sub)sequence as another Seq object</tt> </tt>
<a name="L2143"></a><tt class="py-lineno">2143</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-789" class="py-name"><a title="Bio.Seq.MutableSeq" class="py-name" href="#" onclick="return doclink('link-789', 'MutableSeq', 'link-83');">MutableSeq</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-790" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-790', 'data', 'link-42');">data</a></tt><tt class="py-op">[</tt><tt id="link-791" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.Seq.Seq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-791', 'index', 'link-150');">index</a></tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-792" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-792', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L2144"></a><tt class="py-lineno">2144</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.__setitem__"></a><div id="MutableSeq.__setitem__-def"><a name="L2145"></a><tt class="py-lineno">2145</tt> <a class="py-toggle" href="#" id="MutableSeq.__setitem__-toggle" onclick="return toggle('MutableSeq.__setitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#__setitem__">__setitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">index</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.__setitem__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.__setitem__-expanded"><a name="L2146"></a><tt class="py-lineno">2146</tt>  <tt class="py-line">        <tt class="py-docstring">"""Set a subsequence of single letter via value parameter.</tt> </tt>
<a name="L2147"></a><tt class="py-lineno">2147</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2148"></a><tt class="py-lineno">2148</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq = MutableSeq('ACTCGACGTCG')</tt> </tt>
<a name="L2149"></a><tt class="py-lineno">2149</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq[0] = 'T'</tt> </tt>
<a name="L2150"></a><tt class="py-lineno">2150</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L2151"></a><tt class="py-lineno">2151</tt>  <tt class="py-line"><tt class="py-docstring">        MutableSeq('TCTCGACGTCG')</tt> </tt>
<a name="L2152"></a><tt class="py-lineno">2152</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2153"></a><tt class="py-lineno">2153</tt>  <tt class="py-line">        <tt class="py-comment"># Note since Python 2.0, __setslice__ is deprecated</tt> </tt>
<a name="L2154"></a><tt class="py-lineno">2154</tt>  <tt class="py-line">        <tt class="py-comment"># and __setitem__ is used instead.</tt> </tt>
<a name="L2155"></a><tt class="py-lineno">2155</tt>  <tt class="py-line">        <tt class="py-comment"># See http://docs.python.org/ref/sequence-methods.html</tt> </tt>
<a name="L2156"></a><tt class="py-lineno">2156</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-793" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.Seq.Seq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-793', 'index', 'link-150');">index</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2157"></a><tt class="py-lineno">2157</tt>  <tt class="py-line">            <tt class="py-comment"># Replacing a single letter with a new string</tt> </tt>
<a name="L2158"></a><tt class="py-lineno">2158</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-794" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-794', 'data', 'link-42');">data</a></tt><tt class="py-op">[</tt><tt id="link-795" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.Seq.Seq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-795', 'index', 'link-150');">index</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-796" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.value()=Bio.Phylo.PhyloXMLIO.Writer-class.html#value"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-796', 'value', 'link-796');">value</a></tt> </tt>
<a name="L2159"></a><tt class="py-lineno">2159</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L2160"></a><tt class="py-lineno">2160</tt>  <tt class="py-line">            <tt class="py-comment"># Replacing a sub-sequence</tt> </tt>
<a name="L2161"></a><tt class="py-lineno">2161</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-797" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-797', 'value', 'link-796');">value</a></tt><tt class="py-op">,</tt> <tt id="link-798" class="py-name"><a title="Bio.Seq.MutableSeq" class="py-name" href="#" onclick="return doclink('link-798', 'MutableSeq', 'link-83');">MutableSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2162"></a><tt class="py-lineno">2162</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-799" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-799', 'data', 'link-42');">data</a></tt><tt class="py-op">[</tt><tt id="link-800" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.Seq.Seq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-800', 'index', 'link-150');">index</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-801" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-801', 'value', 'link-796');">value</a></tt><tt class="py-op">.</tt><tt id="link-802" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-802', 'data', 'link-42');">data</a></tt> </tt>
<a name="L2163"></a><tt class="py-lineno">2163</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-803" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-803', 'value', 'link-796');">value</a></tt><tt class="py-op">,</tt> <tt id="link-804" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-804', 'type', 'link-87');">type</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-805" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-805', 'data', 'link-42');">data</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2164"></a><tt class="py-lineno">2164</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-806" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-806', 'data', 'link-42');">data</a></tt><tt class="py-op">[</tt><tt id="link-807" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.Seq.Seq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-807', 'index', 'link-150');">index</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-808" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-808', 'value', 'link-796');">value</a></tt> </tt>
<a name="L2165"></a><tt class="py-lineno">2165</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L2166"></a><tt class="py-lineno">2166</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-809" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-809', 'data', 'link-42');">data</a></tt><tt class="py-op">[</tt><tt id="link-810" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.Seq.Seq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-810', 'index', 'link-150');">index</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">array</tt><tt class="py-op">.</tt><tt class="py-name">array</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">array_indicator</tt><tt class="py-op">,</tt> </tt>
<a name="L2167"></a><tt class="py-lineno">2167</tt>  <tt class="py-line">                                               <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-811" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-811', 'value', 'link-796');">value</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L2168"></a><tt class="py-lineno">2168</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.__delitem__"></a><div id="MutableSeq.__delitem__-def"><a name="L2169"></a><tt class="py-lineno">2169</tt> <a class="py-toggle" href="#" id="MutableSeq.__delitem__-toggle" onclick="return toggle('MutableSeq.__delitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#__delitem__">__delitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">index</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.__delitem__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.__delitem__-expanded"><a name="L2170"></a><tt class="py-lineno">2170</tt>  <tt class="py-line">        <tt class="py-docstring">"""Delete a subsequence of single letter.</tt> </tt>
<a name="L2171"></a><tt class="py-lineno">2171</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2172"></a><tt class="py-lineno">2172</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq = MutableSeq('ACTCGACGTCG')</tt> </tt>
<a name="L2173"></a><tt class="py-lineno">2173</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; del my_seq[0]</tt> </tt>
<a name="L2174"></a><tt class="py-lineno">2174</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L2175"></a><tt class="py-lineno">2175</tt>  <tt class="py-line"><tt class="py-docstring">        MutableSeq('CTCGACGTCG')</tt> </tt>
<a name="L2176"></a><tt class="py-lineno">2176</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2177"></a><tt class="py-lineno">2177</tt>  <tt class="py-line">        <tt class="py-comment"># Note since Python 2.0, __delslice__ is deprecated</tt> </tt>
<a name="L2178"></a><tt class="py-lineno">2178</tt>  <tt class="py-line">        <tt class="py-comment"># and __delitem__ is used instead.</tt> </tt>
<a name="L2179"></a><tt class="py-lineno">2179</tt>  <tt class="py-line">        <tt class="py-comment"># See http://docs.python.org/ref/sequence-methods.html</tt> </tt>
<a name="L2180"></a><tt class="py-lineno">2180</tt>  <tt class="py-line"> </tt>
<a name="L2181"></a><tt class="py-lineno">2181</tt>  <tt class="py-line">        <tt class="py-comment"># Could be deleting a single letter, or a slice</tt> </tt>
<a name="L2182"></a><tt class="py-lineno">2182</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-812" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-812', 'data', 'link-42');">data</a></tt><tt class="py-op">[</tt><tt id="link-813" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.Seq.Seq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-813', 'index', 'link-150');">index</a></tt><tt class="py-op">]</tt> </tt>
</div><a name="L2183"></a><tt class="py-lineno">2183</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.__add__"></a><div id="MutableSeq.__add__-def"><a name="L2184"></a><tt class="py-lineno">2184</tt> <a class="py-toggle" href="#" id="MutableSeq.__add__-toggle" onclick="return toggle('MutableSeq.__add__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#__add__">__add__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.__add__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.__add__-expanded"><a name="L2185"></a><tt class="py-lineno">2185</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add another sequence or string to this sequence.</tt> </tt>
<a name="L2186"></a><tt class="py-lineno">2186</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2187"></a><tt class="py-lineno">2187</tt>  <tt class="py-line"><tt class="py-docstring">        Returns a new MutableSeq object.</tt> </tt>
<a name="L2188"></a><tt class="py-lineno">2188</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2189"></a><tt class="py-lineno">2189</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-814" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-814', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-string">"alphabet"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2190"></a><tt class="py-lineno">2190</tt>  <tt class="py-line">            <tt class="py-comment"># other should be a Seq or a MutableSeq</tt> </tt>
<a name="L2191"></a><tt class="py-lineno">2191</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-815" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-815', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-816" class="py-name"><a title="Bio.Alphabet._check_type_compatible" class="py-name" href="#" onclick="return doclink('link-816', '_check_type_compatible', 'link-59');">_check_type_compatible</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-817" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-817', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L2192"></a><tt class="py-lineno">2192</tt>  <tt class="py-line">                                                    <tt id="link-818" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-818', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-819" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-819', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2193"></a><tt class="py-lineno">2193</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt> </tt>
<a name="L2194"></a><tt class="py-lineno">2194</tt>  <tt class="py-line">                    <tt class="py-string">"Incompatible alphabets {0!r} and {1!r}"</tt><tt class="py-op">.</tt><tt id="link-820" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-820', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L2195"></a><tt class="py-lineno">2195</tt>  <tt class="py-line">                        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-821" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-821', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-822" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-822', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-823" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-823', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L2196"></a><tt class="py-lineno">2196</tt>  <tt class="py-line">            <tt class="py-comment"># They should be the same sequence type (or one of them is generic)</tt> </tt>
<a name="L2197"></a><tt class="py-lineno">2197</tt>  <tt class="py-line">            <tt id="link-824" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-824', 'a', 'link-49');">a</a></tt> <tt class="py-op">=</tt> <tt id="link-825" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-825', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-826" class="py-name"><a title="Bio.Alphabet._consensus_alphabet" class="py-name" href="#" onclick="return doclink('link-826', '_consensus_alphabet', 'link-167');">_consensus_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-827" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-827', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-828" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-828', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-829" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-829', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L2198"></a><tt class="py-lineno">2198</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-830" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-830', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt id="link-831" class="py-name"><a title="Bio.Seq.MutableSeq" class="py-name" href="#" onclick="return doclink('link-831', 'MutableSeq', 'link-83');">MutableSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2199"></a><tt class="py-lineno">2199</tt>  <tt class="py-line">                <tt class="py-comment"># See test_GAQueens.py for an historic usage of a non-string</tt> </tt>
<a name="L2200"></a><tt class="py-lineno">2200</tt>  <tt class="py-line">                <tt class="py-comment"># alphabet!  Adding the arrays should support this.</tt> </tt>
<a name="L2201"></a><tt class="py-lineno">2201</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-832" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-832', 'data', 'link-42');">data</a></tt> <tt class="py-op">+</tt> <tt id="link-833" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-833', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-834" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-834', 'data', 'link-42');">data</a></tt><tt class="py-op">,</tt> <tt id="link-835" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-835', 'a', 'link-49');">a</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2202"></a><tt class="py-lineno">2202</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L2203"></a><tt class="py-lineno">2203</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-836" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-836', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-837" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-837', 'a', 'link-49');">a</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2204"></a><tt class="py-lineno">2204</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-838" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-838', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2205"></a><tt class="py-lineno">2205</tt>  <tt class="py-line">            <tt class="py-comment"># other is a plain string - use the current alphabet</tt> </tt>
<a name="L2206"></a><tt class="py-lineno">2206</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-839" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-839', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-840" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-840', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2207"></a><tt class="py-lineno">2207</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L2208"></a><tt class="py-lineno">2208</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt> </tt>
</div><a name="L2209"></a><tt class="py-lineno">2209</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.__radd__"></a><div id="MutableSeq.__radd__-def"><a name="L2210"></a><tt class="py-lineno">2210</tt> <a class="py-toggle" href="#" id="MutableSeq.__radd__-toggle" onclick="return toggle('MutableSeq.__radd__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#__radd__">__radd__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.__radd__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.__radd__-expanded"><a name="L2211"></a><tt class="py-lineno">2211</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add a sequence on the left.</tt> </tt>
<a name="L2212"></a><tt class="py-lineno">2212</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2213"></a><tt class="py-lineno">2213</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import MutableSeq</tt> </tt>
<a name="L2214"></a><tt class="py-lineno">2214</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_protein</tt> </tt>
<a name="L2215"></a><tt class="py-lineno">2215</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; "LV" + MutableSeq("MELKI", generic_protein)</tt> </tt>
<a name="L2216"></a><tt class="py-lineno">2216</tt>  <tt class="py-line"><tt class="py-docstring">        MutableSeq('LVMELKI', ProteinAlphabet())</tt> </tt>
<a name="L2217"></a><tt class="py-lineno">2217</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2218"></a><tt class="py-lineno">2218</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-841" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-841', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-string">"alphabet"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2219"></a><tt class="py-lineno">2219</tt>  <tt class="py-line">            <tt class="py-comment"># other should be a Seq or a MutableSeq</tt> </tt>
<a name="L2220"></a><tt class="py-lineno">2220</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-842" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-842', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-843" class="py-name"><a title="Bio.Alphabet._check_type_compatible" class="py-name" href="#" onclick="return doclink('link-843', '_check_type_compatible', 'link-59');">_check_type_compatible</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-844" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-844', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L2221"></a><tt class="py-lineno">2221</tt>  <tt class="py-line">                                                    <tt id="link-845" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-845', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-846" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-846', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2222"></a><tt class="py-lineno">2222</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt> </tt>
<a name="L2223"></a><tt class="py-lineno">2223</tt>  <tt class="py-line">                    <tt class="py-string">"Incompatible alphabets {0!r} and {1!r}"</tt><tt class="py-op">.</tt><tt id="link-847" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-847', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L2224"></a><tt class="py-lineno">2224</tt>  <tt class="py-line">                        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-848" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-848', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-849" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-849', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-850" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-850', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L2225"></a><tt class="py-lineno">2225</tt>  <tt class="py-line">            <tt class="py-comment"># They should be the same sequence type (or one of them is generic)</tt> </tt>
<a name="L2226"></a><tt class="py-lineno">2226</tt>  <tt class="py-line">            <tt id="link-851" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-851', 'a', 'link-49');">a</a></tt> <tt class="py-op">=</tt> <tt id="link-852" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-852', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-853" class="py-name"><a title="Bio.Alphabet._consensus_alphabet" class="py-name" href="#" onclick="return doclink('link-853', '_consensus_alphabet', 'link-167');">_consensus_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-854" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-854', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-855" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-855', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-856" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-856', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L2227"></a><tt class="py-lineno">2227</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-857" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-857', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt id="link-858" class="py-name"><a title="Bio.Seq.MutableSeq" class="py-name" href="#" onclick="return doclink('link-858', 'MutableSeq', 'link-83');">MutableSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2228"></a><tt class="py-lineno">2228</tt>  <tt class="py-line">                <tt class="py-comment"># See test_GAQueens.py for an historic usage of a non-string</tt> </tt>
<a name="L2229"></a><tt class="py-lineno">2229</tt>  <tt class="py-line">                <tt class="py-comment"># alphabet!  Adding the arrays should support this.</tt> </tt>
<a name="L2230"></a><tt class="py-lineno">2230</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt id="link-859" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-859', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-860" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-860', 'data', 'link-42');">data</a></tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-861" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-861', 'data', 'link-42');">data</a></tt><tt class="py-op">,</tt> <tt id="link-862" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-862', 'a', 'link-49');">a</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2231"></a><tt class="py-lineno">2231</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L2232"></a><tt class="py-lineno">2232</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-863" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-863', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-864" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-864', 'a', 'link-49');">a</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2233"></a><tt class="py-lineno">2233</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-865" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-865', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2234"></a><tt class="py-lineno">2234</tt>  <tt class="py-line">            <tt class="py-comment"># other is a plain string - use the current alphabet</tt> </tt>
<a name="L2235"></a><tt class="py-lineno">2235</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-866" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-866', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-867" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-867', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2236"></a><tt class="py-lineno">2236</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L2237"></a><tt class="py-lineno">2237</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt> </tt>
</div><a name="L2238"></a><tt class="py-lineno">2238</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.__mul__"></a><div id="MutableSeq.__mul__-def"><a name="L2239"></a><tt class="py-lineno">2239</tt> <a class="py-toggle" href="#" id="MutableSeq.__mul__-toggle" onclick="return toggle('MutableSeq.__mul__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#__mul__">__mul__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.__mul__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.__mul__-expanded"><a name="L2240"></a><tt class="py-lineno">2240</tt>  <tt class="py-line">        <tt class="py-docstring">"""Multiply MutableSeq by integer.</tt> </tt>
<a name="L2241"></a><tt class="py-lineno">2241</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2242"></a><tt class="py-lineno">2242</tt>  <tt class="py-line"><tt class="py-docstring">        Note this is not in-place and returns a new object,</tt> </tt>
<a name="L2243"></a><tt class="py-lineno">2243</tt>  <tt class="py-line"><tt class="py-docstring">        matching native Python list multiplication.</tt> </tt>
<a name="L2244"></a><tt class="py-lineno">2244</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2245"></a><tt class="py-lineno">2245</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import MutableSeq</tt> </tt>
<a name="L2246"></a><tt class="py-lineno">2246</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_dna</tt> </tt>
<a name="L2247"></a><tt class="py-lineno">2247</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; MutableSeq('ATG') * 2</tt> </tt>
<a name="L2248"></a><tt class="py-lineno">2248</tt>  <tt class="py-line"><tt class="py-docstring">        MutableSeq('ATGATG')</tt> </tt>
<a name="L2249"></a><tt class="py-lineno">2249</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; MutableSeq('ATG', generic_dna) * 2</tt> </tt>
<a name="L2250"></a><tt class="py-lineno">2250</tt>  <tt class="py-line"><tt class="py-docstring">        MutableSeq('ATGATG', DNAAlphabet())</tt> </tt>
<a name="L2251"></a><tt class="py-lineno">2251</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2252"></a><tt class="py-lineno">2252</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-868" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-868', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2253"></a><tt class="py-lineno">2253</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"can't multiply {} by non-int type"</tt><tt class="py-op">.</tt><tt id="link-869" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-869', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L2254"></a><tt class="py-lineno">2254</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-870" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-870', 'data', 'link-42');">data</a></tt> <tt class="py-op">*</tt> <tt id="link-871" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-871', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-872" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-872', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L2255"></a><tt class="py-lineno">2255</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.__rmul__"></a><div id="MutableSeq.__rmul__-def"><a name="L2256"></a><tt class="py-lineno">2256</tt> <a class="py-toggle" href="#" id="MutableSeq.__rmul__-toggle" onclick="return toggle('MutableSeq.__rmul__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#__rmul__">__rmul__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.__rmul__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.__rmul__-expanded"><a name="L2257"></a><tt class="py-lineno">2257</tt>  <tt class="py-line">        <tt class="py-docstring">"""Multiply integer by MutableSeq.</tt> </tt>
<a name="L2258"></a><tt class="py-lineno">2258</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2259"></a><tt class="py-lineno">2259</tt>  <tt class="py-line"><tt class="py-docstring">        Note this is not in-place and returns a new object,</tt> </tt>
<a name="L2260"></a><tt class="py-lineno">2260</tt>  <tt class="py-line"><tt class="py-docstring">        matching native Python list multiplication.</tt> </tt>
<a name="L2261"></a><tt class="py-lineno">2261</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2262"></a><tt class="py-lineno">2262</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import MutableSeq</tt> </tt>
<a name="L2263"></a><tt class="py-lineno">2263</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_dna</tt> </tt>
<a name="L2264"></a><tt class="py-lineno">2264</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; 2 * MutableSeq('ATG')</tt> </tt>
<a name="L2265"></a><tt class="py-lineno">2265</tt>  <tt class="py-line"><tt class="py-docstring">        MutableSeq('ATGATG')</tt> </tt>
<a name="L2266"></a><tt class="py-lineno">2266</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; 2 * MutableSeq('ATG', generic_dna)</tt> </tt>
<a name="L2267"></a><tt class="py-lineno">2267</tt>  <tt class="py-line"><tt class="py-docstring">        MutableSeq('ATGATG', DNAAlphabet())</tt> </tt>
<a name="L2268"></a><tt class="py-lineno">2268</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2269"></a><tt class="py-lineno">2269</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-873" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-873', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2270"></a><tt class="py-lineno">2270</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"can't multiply {} by non-int type"</tt><tt class="py-op">.</tt><tt id="link-874" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-874', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L2271"></a><tt class="py-lineno">2271</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-875" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-875', 'data', 'link-42');">data</a></tt> <tt class="py-op">*</tt> <tt id="link-876" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-876', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-877" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-877', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L2272"></a><tt class="py-lineno">2272</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.__imul__"></a><div id="MutableSeq.__imul__-def"><a name="L2273"></a><tt class="py-lineno">2273</tt> <a class="py-toggle" href="#" id="MutableSeq.__imul__-toggle" onclick="return toggle('MutableSeq.__imul__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#__imul__">__imul__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.__imul__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.__imul__-expanded"><a name="L2274"></a><tt class="py-lineno">2274</tt>  <tt class="py-line">        <tt class="py-docstring">"""Multiply MutableSeq in-place.</tt> </tt>
<a name="L2275"></a><tt class="py-lineno">2275</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2276"></a><tt class="py-lineno">2276</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import MutableSeq</tt> </tt>
<a name="L2277"></a><tt class="py-lineno">2277</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import generic_dna</tt> </tt>
<a name="L2278"></a><tt class="py-lineno">2278</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; seq = MutableSeq('ATG', generic_dna)</tt> </tt>
<a name="L2279"></a><tt class="py-lineno">2279</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; seq *= 2</tt> </tt>
<a name="L2280"></a><tt class="py-lineno">2280</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; seq</tt> </tt>
<a name="L2281"></a><tt class="py-lineno">2281</tt>  <tt class="py-line"><tt class="py-docstring">        MutableSeq('ATGATG', DNAAlphabet())</tt> </tt>
<a name="L2282"></a><tt class="py-lineno">2282</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2283"></a><tt class="py-lineno">2283</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-878" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-878', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2284"></a><tt class="py-lineno">2284</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"can't multiply {} by non-int type"</tt><tt class="py-op">.</tt><tt id="link-879" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-879', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L2285"></a><tt class="py-lineno">2285</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-880" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-880', 'data', 'link-42');">data</a></tt> <tt class="py-op">*</tt> <tt id="link-881" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-881', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-882" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-882', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L2286"></a><tt class="py-lineno">2286</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.append"></a><div id="MutableSeq.append-def"><a name="L2287"></a><tt class="py-lineno">2287</tt> <a class="py-toggle" href="#" id="MutableSeq.append-toggle" onclick="return toggle('MutableSeq.append');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#append">append</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">c</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.append-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.append-expanded"><a name="L2288"></a><tt class="py-lineno">2288</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add a subsequence to the mutable sequence object.</tt> </tt>
<a name="L2289"></a><tt class="py-lineno">2289</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2290"></a><tt class="py-lineno">2290</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq = MutableSeq('ACTCGACGTCG')</tt> </tt>
<a name="L2291"></a><tt class="py-lineno">2291</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.append('A')</tt> </tt>
<a name="L2292"></a><tt class="py-lineno">2292</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L2293"></a><tt class="py-lineno">2293</tt>  <tt class="py-line"><tt class="py-docstring">        MutableSeq('ACTCGACGTCGA')</tt> </tt>
<a name="L2294"></a><tt class="py-lineno">2294</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2295"></a><tt class="py-lineno">2295</tt>  <tt class="py-line"><tt class="py-docstring">        No return value.</tt> </tt>
<a name="L2296"></a><tt class="py-lineno">2296</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2297"></a><tt class="py-lineno">2297</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-883" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-883', 'data', 'link-42');">data</a></tt><tt class="py-op">.</tt><tt id="link-884" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-884', 'append', 'link-884');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">)</tt> </tt>
</div><a name="L2298"></a><tt class="py-lineno">2298</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.insert"></a><div id="MutableSeq.insert-def"><a name="L2299"></a><tt class="py-lineno">2299</tt> <a class="py-toggle" href="#" id="MutableSeq.insert-toggle" onclick="return toggle('MutableSeq.insert');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#insert">insert</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">i</tt><tt class="py-op">,</tt> <tt class="py-param">c</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.insert-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.insert-expanded"><a name="L2300"></a><tt class="py-lineno">2300</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add a subsequence to the mutable sequence object at a given index.</tt> </tt>
<a name="L2301"></a><tt class="py-lineno">2301</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2302"></a><tt class="py-lineno">2302</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq = MutableSeq('ACTCGACGTCG')</tt> </tt>
<a name="L2303"></a><tt class="py-lineno">2303</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.insert(0,'A')</tt> </tt>
<a name="L2304"></a><tt class="py-lineno">2304</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L2305"></a><tt class="py-lineno">2305</tt>  <tt class="py-line"><tt class="py-docstring">        MutableSeq('AACTCGACGTCG')</tt> </tt>
<a name="L2306"></a><tt class="py-lineno">2306</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.insert(8,'G')</tt> </tt>
<a name="L2307"></a><tt class="py-lineno">2307</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L2308"></a><tt class="py-lineno">2308</tt>  <tt class="py-line"><tt class="py-docstring">        MutableSeq('AACTCGACGGTCG')</tt> </tt>
<a name="L2309"></a><tt class="py-lineno">2309</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2310"></a><tt class="py-lineno">2310</tt>  <tt class="py-line"><tt class="py-docstring">        No return value.</tt> </tt>
<a name="L2311"></a><tt class="py-lineno">2311</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2312"></a><tt class="py-lineno">2312</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-885" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-885', 'data', 'link-42');">data</a></tt><tt class="py-op">.</tt><tt id="link-886" class="py-name" targets="Method Bio.Crystal.Chain.insert()=Bio.Crystal.Chain-class.html#insert,Method Bio.PDB.Entity.Entity.insert()=Bio.PDB.Entity.Entity-class.html#insert,Method Bio.Phylo.TreeConstruction._Matrix.insert()=Bio.Phylo.TreeConstruction._Matrix-class.html#insert,Method Bio.Seq.MutableSeq.insert()=Bio.Seq.MutableSeq-class.html#insert"><a title="Bio.Crystal.Chain.insert
Bio.PDB.Entity.Entity.insert
Bio.Phylo.TreeConstruction._Matrix.insert
Bio.Seq.MutableSeq.insert" class="py-name" href="#" onclick="return doclink('link-886', 'insert', 'link-886');">insert</a></tt><tt class="py-op">(</tt><tt id="link-887" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-887', 'i', 'link-887');">i</a></tt><tt class="py-op">,</tt> <tt class="py-name">c</tt><tt class="py-op">)</tt> </tt>
</div><a name="L2313"></a><tt class="py-lineno">2313</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.pop"></a><div id="MutableSeq.pop-def"><a name="L2314"></a><tt class="py-lineno">2314</tt> <a class="py-toggle" href="#" id="MutableSeq.pop-toggle" onclick="return toggle('MutableSeq.pop');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#pop">pop</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">i</tt><tt class="py-op">=</tt><tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.pop-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.pop-expanded"><a name="L2315"></a><tt class="py-lineno">2315</tt>  <tt class="py-line">        <tt class="py-docstring">"""Remove a subsequence of a single letter at given index.</tt> </tt>
<a name="L2316"></a><tt class="py-lineno">2316</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2317"></a><tt class="py-lineno">2317</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq = MutableSeq('ACTCGACGTCG')</tt> </tt>
<a name="L2318"></a><tt class="py-lineno">2318</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.pop()</tt> </tt>
<a name="L2319"></a><tt class="py-lineno">2319</tt>  <tt class="py-line"><tt class="py-docstring">        'G'</tt> </tt>
<a name="L2320"></a><tt class="py-lineno">2320</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L2321"></a><tt class="py-lineno">2321</tt>  <tt class="py-line"><tt class="py-docstring">        MutableSeq('ACTCGACGTC')</tt> </tt>
<a name="L2322"></a><tt class="py-lineno">2322</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.pop()</tt> </tt>
<a name="L2323"></a><tt class="py-lineno">2323</tt>  <tt class="py-line"><tt class="py-docstring">        'C'</tt> </tt>
<a name="L2324"></a><tt class="py-lineno">2324</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L2325"></a><tt class="py-lineno">2325</tt>  <tt class="py-line"><tt class="py-docstring">        MutableSeq('ACTCGACGT')</tt> </tt>
<a name="L2326"></a><tt class="py-lineno">2326</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2327"></a><tt class="py-lineno">2327</tt>  <tt class="py-line"><tt class="py-docstring">        Returns the last character of the sequence.</tt> </tt>
<a name="L2328"></a><tt class="py-lineno">2328</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2329"></a><tt class="py-lineno">2329</tt>  <tt class="py-line">        <tt class="py-name">c</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-888" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-888', 'data', 'link-42');">data</a></tt><tt class="py-op">[</tt><tt id="link-889" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-889', 'i', 'link-887');">i</a></tt><tt class="py-op">]</tt> </tt>
<a name="L2330"></a><tt class="py-lineno">2330</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-890" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-890', 'data', 'link-42');">data</a></tt><tt class="py-op">[</tt><tt id="link-891" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-891', 'i', 'link-887');">i</a></tt><tt class="py-op">]</tt> </tt>
<a name="L2331"></a><tt class="py-lineno">2331</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">c</tt> </tt>
</div><a name="L2332"></a><tt class="py-lineno">2332</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.remove"></a><div id="MutableSeq.remove-def"><a name="L2333"></a><tt class="py-lineno">2333</tt> <a class="py-toggle" href="#" id="MutableSeq.remove-toggle" onclick="return toggle('MutableSeq.remove');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#remove">remove</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">item</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.remove-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.remove-expanded"><a name="L2334"></a><tt class="py-lineno">2334</tt>  <tt class="py-line">        <tt class="py-docstring">"""Remove a subsequence of a single letter from mutable sequence.</tt> </tt>
<a name="L2335"></a><tt class="py-lineno">2335</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2336"></a><tt class="py-lineno">2336</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq = MutableSeq('ACTCGACGTCG')</tt> </tt>
<a name="L2337"></a><tt class="py-lineno">2337</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.remove('C')</tt> </tt>
<a name="L2338"></a><tt class="py-lineno">2338</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L2339"></a><tt class="py-lineno">2339</tt>  <tt class="py-line"><tt class="py-docstring">        MutableSeq('ATCGACGTCG')</tt> </tt>
<a name="L2340"></a><tt class="py-lineno">2340</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.remove('A')</tt> </tt>
<a name="L2341"></a><tt class="py-lineno">2341</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L2342"></a><tt class="py-lineno">2342</tt>  <tt class="py-line"><tt class="py-docstring">        MutableSeq('TCGACGTCG')</tt> </tt>
<a name="L2343"></a><tt class="py-lineno">2343</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2344"></a><tt class="py-lineno">2344</tt>  <tt class="py-line"><tt class="py-docstring">        No return value.</tt> </tt>
<a name="L2345"></a><tt class="py-lineno">2345</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2346"></a><tt class="py-lineno">2346</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-892" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-892', 'i', 'link-887');">i</a></tt> <tt class="py-keyword">in</tt> <tt id="link-893" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-893', 'range', 'link-2');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-894" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-894', 'data', 'link-42');">data</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2347"></a><tt class="py-lineno">2347</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-895" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-895', 'data', 'link-42');">data</a></tt><tt class="py-op">[</tt><tt id="link-896" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-896', 'i', 'link-887');">i</a></tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">item</tt><tt class="py-op">:</tt> </tt>
<a name="L2348"></a><tt class="py-lineno">2348</tt>  <tt class="py-line">                <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-897" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-897', 'data', 'link-42');">data</a></tt><tt class="py-op">[</tt><tt id="link-898" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-898', 'i', 'link-887');">i</a></tt><tt class="py-op">]</tt> </tt>
<a name="L2349"></a><tt class="py-lineno">2349</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> </tt>
<a name="L2350"></a><tt class="py-lineno">2350</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"MutableSeq.remove(x): x not in list"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L2351"></a><tt class="py-lineno">2351</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.count"></a><div id="MutableSeq.count-def"><a name="L2352"></a><tt class="py-lineno">2352</tt> <a class="py-toggle" href="#" id="MutableSeq.count-toggle" onclick="return toggle('MutableSeq.count');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#count">count</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sub</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">=</tt><tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">maxsize</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.count-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.count-expanded"><a name="L2353"></a><tt class="py-lineno">2353</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a non-overlapping count, like that of a python string.</tt> </tt>
<a name="L2354"></a><tt class="py-lineno">2354</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2355"></a><tt class="py-lineno">2355</tt>  <tt class="py-line"><tt class="py-docstring">        This behaves like the python string method of the same name,</tt> </tt>
<a name="L2356"></a><tt class="py-lineno">2356</tt>  <tt class="py-line"><tt class="py-docstring">        which does a non-overlapping count!</tt> </tt>
<a name="L2357"></a><tt class="py-lineno">2357</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2358"></a><tt class="py-lineno">2358</tt>  <tt class="py-line"><tt class="py-docstring">        For an overlapping search use the newer count_overlap() method.</tt> </tt>
<a name="L2359"></a><tt class="py-lineno">2359</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2360"></a><tt class="py-lineno">2360</tt>  <tt class="py-line"><tt class="py-docstring">        Returns an integer, the number of occurrences of substring</tt> </tt>
<a name="L2361"></a><tt class="py-lineno">2361</tt>  <tt class="py-line"><tt class="py-docstring">        argument sub in the (sub)sequence given by [start:end].</tt> </tt>
<a name="L2362"></a><tt class="py-lineno">2362</tt>  <tt class="py-line"><tt class="py-docstring">        Optional arguments start and end are interpreted as in slice</tt> </tt>
<a name="L2363"></a><tt class="py-lineno">2363</tt>  <tt class="py-line"><tt class="py-docstring">        notation.</tt> </tt>
<a name="L2364"></a><tt class="py-lineno">2364</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2365"></a><tt class="py-lineno">2365</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L2366"></a><tt class="py-lineno">2366</tt>  <tt class="py-line"><tt class="py-docstring">         - sub - a string or another Seq object to look for</tt> </tt>
<a name="L2367"></a><tt class="py-lineno">2367</tt>  <tt class="py-line"><tt class="py-docstring">         - start - optional integer, slice start</tt> </tt>
<a name="L2368"></a><tt class="py-lineno">2368</tt>  <tt class="py-line"><tt class="py-docstring">         - end - optional integer, slice end</tt> </tt>
<a name="L2369"></a><tt class="py-lineno">2369</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2370"></a><tt class="py-lineno">2370</tt>  <tt class="py-line"><tt class="py-docstring">        e.g.</tt> </tt>
<a name="L2371"></a><tt class="py-lineno">2371</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2372"></a><tt class="py-lineno">2372</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import MutableSeq</tt> </tt>
<a name="L2373"></a><tt class="py-lineno">2373</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_mseq = MutableSeq("AAAATGA")</tt> </tt>
<a name="L2374"></a><tt class="py-lineno">2374</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_mseq.count("A"))</tt> </tt>
<a name="L2375"></a><tt class="py-lineno">2375</tt>  <tt class="py-line"><tt class="py-docstring">        5</tt> </tt>
<a name="L2376"></a><tt class="py-lineno">2376</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_mseq.count("ATG"))</tt> </tt>
<a name="L2377"></a><tt class="py-lineno">2377</tt>  <tt class="py-line"><tt class="py-docstring">        1</tt> </tt>
<a name="L2378"></a><tt class="py-lineno">2378</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_mseq.count(Seq("AT")))</tt> </tt>
<a name="L2379"></a><tt class="py-lineno">2379</tt>  <tt class="py-line"><tt class="py-docstring">        1</tt> </tt>
<a name="L2380"></a><tt class="py-lineno">2380</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_mseq.count("AT", 2, -1))</tt> </tt>
<a name="L2381"></a><tt class="py-lineno">2381</tt>  <tt class="py-line"><tt class="py-docstring">        1</tt> </tt>
<a name="L2382"></a><tt class="py-lineno">2382</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2383"></a><tt class="py-lineno">2383</tt>  <tt class="py-line"><tt class="py-docstring">        HOWEVER, please note because that python strings, Seq objects and</tt> </tt>
<a name="L2384"></a><tt class="py-lineno">2384</tt>  <tt class="py-line"><tt class="py-docstring">        MutableSeq objects do a non-overlapping search, this may not give</tt> </tt>
<a name="L2385"></a><tt class="py-lineno">2385</tt>  <tt class="py-line"><tt class="py-docstring">        the answer you expect:</tt> </tt>
<a name="L2386"></a><tt class="py-lineno">2386</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2387"></a><tt class="py-lineno">2387</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; "AAAA".count("AA")</tt> </tt>
<a name="L2388"></a><tt class="py-lineno">2388</tt>  <tt class="py-line"><tt class="py-docstring">        2</tt> </tt>
<a name="L2389"></a><tt class="py-lineno">2389</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(MutableSeq("AAAA").count("AA"))</tt> </tt>
<a name="L2390"></a><tt class="py-lineno">2390</tt>  <tt class="py-line"><tt class="py-docstring">        2</tt> </tt>
<a name="L2391"></a><tt class="py-lineno">2391</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2392"></a><tt class="py-lineno">2392</tt>  <tt class="py-line"><tt class="py-docstring">        An overlapping search would give the answer as three!</tt> </tt>
<a name="L2393"></a><tt class="py-lineno">2393</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2394"></a><tt class="py-lineno">2394</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L2395"></a><tt class="py-lineno">2395</tt>  <tt class="py-line">            <tt class="py-comment"># TODO - Should we check the alphabet?</tt> </tt>
<a name="L2396"></a><tt class="py-lineno">2396</tt>  <tt class="py-line">            <tt id="link-899" class="py-name" targets="Method Bio.AlignIO.MafIO.MafIndex.search()=Bio.AlignIO.MafIO.MafIndex-class.html#search,Method Bio.KDTree.KDTree'.KDTree.search()=Bio.KDTree.KDTree%27.KDTree-class.html#search,Method Bio.PDB.NeighborSearch'.NeighborSearch.search()=Bio.PDB.NeighborSearch%27.NeighborSearch-class.html#search,Method Bio.Phylo.TreeConstruction.NNITreeSearcher.search()=Bio.Phylo.TreeConstruction.NNITreeSearcher-class.html#search,Method Bio.Phylo.TreeConstruction.TreeSearcher.search()=Bio.Phylo.TreeConstruction.TreeSearcher-class.html#search,Class Method Bio.Restriction.Restriction.AbstractCut.search()=Bio.Restriction.Restriction.AbstractCut-class.html#search,Method Bio.Restriction.Restriction.RestrictionBatch.search()=Bio.Restriction.Restriction.RestrictionBatch-class.html#search,Function Bio.SCOP.search()=Bio.SCOP-module.html#search,Function Bio.TogoWS.search()=Bio.TogoWS-module.html#search,Method Bio.motifs.Instances.search()=Bio.motifs.Instances-class.html#search,Method Bio.motifs.matrix.PositionSpecificScoringMatrix.search()=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#search"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-899', 'search', 'link-899');">search</a></tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">sub</tt><tt class="py-op">)</tt> </tt>
<a name="L2397"></a><tt class="py-lineno">2397</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L2398"></a><tt class="py-lineno">2398</tt>  <tt class="py-line">            <tt id="link-900" class="py-name"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-900', 'search', 'link-899');">search</a></tt> <tt class="py-op">=</tt> <tt class="py-name">sub</tt> </tt>
<a name="L2399"></a><tt class="py-lineno">2399</tt>  <tt class="py-line"> </tt>
<a name="L2400"></a><tt class="py-lineno">2400</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-901" class="py-name"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-901', 'search', 'link-899');">search</a></tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2401"></a><tt class="py-lineno">2401</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"expected a string, Seq or MutableSeq"</tt><tt class="py-op">)</tt> </tt>
<a name="L2402"></a><tt class="py-lineno">2402</tt>  <tt class="py-line"> </tt>
<a name="L2403"></a><tt class="py-lineno">2403</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-902" class="py-name"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-902', 'search', 'link-899');">search</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L2404"></a><tt class="py-lineno">2404</tt>  <tt class="py-line">            <tt class="py-comment"># Try and be efficient and work directly from the array.</tt> </tt>
<a name="L2405"></a><tt class="py-lineno">2405</tt>  <tt class="py-line">            <tt id="link-903" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-903', 'count', 'link-223');">count</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L2406"></a><tt class="py-lineno">2406</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">c</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-904" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-904', 'data', 'link-42');">data</a></tt><tt class="py-op">[</tt><tt id="link-905" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-905', 'start', 'link-224');">start</a></tt><tt class="py-op">:</tt><tt id="link-906" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-906', 'end', 'link-225');">end</a></tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L2407"></a><tt class="py-lineno">2407</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">c</tt> <tt class="py-op">==</tt> <tt id="link-907" class="py-name"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-907', 'search', 'link-899');">search</a></tt><tt class="py-op">:</tt> </tt>
<a name="L2408"></a><tt class="py-lineno">2408</tt>  <tt class="py-line">                    <tt id="link-908" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-908', 'count', 'link-223');">count</a></tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L2409"></a><tt class="py-lineno">2409</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-909" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-909', 'count', 'link-223');">count</a></tt> </tt>
<a name="L2410"></a><tt class="py-lineno">2410</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L2411"></a><tt class="py-lineno">2411</tt>  <tt class="py-line">            <tt class="py-comment"># TODO - Can we do this more efficiently?</tt> </tt>
<a name="L2412"></a><tt class="py-lineno">2412</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-910" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-910', 'count', 'link-223');">count</a></tt><tt class="py-op">(</tt><tt id="link-911" class="py-name"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-911', 'search', 'link-899');">search</a></tt><tt class="py-op">,</tt> <tt id="link-912" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-912', 'start', 'link-224');">start</a></tt><tt class="py-op">,</tt> <tt id="link-913" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-913', 'end', 'link-225');">end</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L2413"></a><tt class="py-lineno">2413</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.count_overlap"></a><div id="MutableSeq.count_overlap-def"><a name="L2414"></a><tt class="py-lineno">2414</tt> <a class="py-toggle" href="#" id="MutableSeq.count_overlap-toggle" onclick="return toggle('MutableSeq.count_overlap');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#count_overlap">count_overlap</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sub</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">=</tt><tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">maxsize</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.count_overlap-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.count_overlap-expanded"><a name="L2415"></a><tt class="py-lineno">2415</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return an overlapping count.</tt> </tt>
<a name="L2416"></a><tt class="py-lineno">2416</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2417"></a><tt class="py-lineno">2417</tt>  <tt class="py-line"><tt class="py-docstring">        For a non-overlapping search use the count() method.</tt> </tt>
<a name="L2418"></a><tt class="py-lineno">2418</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2419"></a><tt class="py-lineno">2419</tt>  <tt class="py-line"><tt class="py-docstring">        Returns an integer, the number of occurrences of substring</tt> </tt>
<a name="L2420"></a><tt class="py-lineno">2420</tt>  <tt class="py-line"><tt class="py-docstring">        argument sub in the (sub)sequence given by [start:end].</tt> </tt>
<a name="L2421"></a><tt class="py-lineno">2421</tt>  <tt class="py-line"><tt class="py-docstring">        Optional arguments start and end are interpreted as in slice</tt> </tt>
<a name="L2422"></a><tt class="py-lineno">2422</tt>  <tt class="py-line"><tt class="py-docstring">        notation.</tt> </tt>
<a name="L2423"></a><tt class="py-lineno">2423</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2424"></a><tt class="py-lineno">2424</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L2425"></a><tt class="py-lineno">2425</tt>  <tt class="py-line"><tt class="py-docstring">         - sub - a string or another Seq object to look for</tt> </tt>
<a name="L2426"></a><tt class="py-lineno">2426</tt>  <tt class="py-line"><tt class="py-docstring">         - start - optional integer, slice start</tt> </tt>
<a name="L2427"></a><tt class="py-lineno">2427</tt>  <tt class="py-line"><tt class="py-docstring">         - end - optional integer, slice end</tt> </tt>
<a name="L2428"></a><tt class="py-lineno">2428</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2429"></a><tt class="py-lineno">2429</tt>  <tt class="py-line"><tt class="py-docstring">        e.g.</tt> </tt>
<a name="L2430"></a><tt class="py-lineno">2430</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2431"></a><tt class="py-lineno">2431</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import MutableSeq</tt> </tt>
<a name="L2432"></a><tt class="py-lineno">2432</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(MutableSeq("AAAA").count_overlap("AA"))</tt> </tt>
<a name="L2433"></a><tt class="py-lineno">2433</tt>  <tt class="py-line"><tt class="py-docstring">        3</tt> </tt>
<a name="L2434"></a><tt class="py-lineno">2434</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(MutableSeq("ATATATATA").count_overlap("ATA"))</tt> </tt>
<a name="L2435"></a><tt class="py-lineno">2435</tt>  <tt class="py-line"><tt class="py-docstring">        4</tt> </tt>
<a name="L2436"></a><tt class="py-lineno">2436</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(MutableSeq("ATATATATA").count_overlap("ATA", 3, -1))</tt> </tt>
<a name="L2437"></a><tt class="py-lineno">2437</tt>  <tt class="py-line"><tt class="py-docstring">        1</tt> </tt>
<a name="L2438"></a><tt class="py-lineno">2438</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2439"></a><tt class="py-lineno">2439</tt>  <tt class="py-line"><tt class="py-docstring">        Where substrings do not overlap, should behave the same as</tt> </tt>
<a name="L2440"></a><tt class="py-lineno">2440</tt>  <tt class="py-line"><tt class="py-docstring">        the count() method:</tt> </tt>
<a name="L2441"></a><tt class="py-lineno">2441</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2442"></a><tt class="py-lineno">2442</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import MutableSeq</tt> </tt>
<a name="L2443"></a><tt class="py-lineno">2443</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_mseq = MutableSeq("AAAATGA")</tt> </tt>
<a name="L2444"></a><tt class="py-lineno">2444</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_mseq.count_overlap("A"))</tt> </tt>
<a name="L2445"></a><tt class="py-lineno">2445</tt>  <tt class="py-line"><tt class="py-docstring">        5</tt> </tt>
<a name="L2446"></a><tt class="py-lineno">2446</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_mseq.count_overlap("A") == my_mseq.count("A")</tt> </tt>
<a name="L2447"></a><tt class="py-lineno">2447</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L2448"></a><tt class="py-lineno">2448</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_mseq.count_overlap("ATG"))</tt> </tt>
<a name="L2449"></a><tt class="py-lineno">2449</tt>  <tt class="py-line"><tt class="py-docstring">        1</tt> </tt>
<a name="L2450"></a><tt class="py-lineno">2450</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_mseq.count_overlap("ATG") == my_mseq.count("ATG")</tt> </tt>
<a name="L2451"></a><tt class="py-lineno">2451</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L2452"></a><tt class="py-lineno">2452</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_mseq.count_overlap(Seq("AT")))</tt> </tt>
<a name="L2453"></a><tt class="py-lineno">2453</tt>  <tt class="py-line"><tt class="py-docstring">        1</tt> </tt>
<a name="L2454"></a><tt class="py-lineno">2454</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_mseq.count_overlap(Seq("AT")) == my_mseq.count(Seq("AT"))</tt> </tt>
<a name="L2455"></a><tt class="py-lineno">2455</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L2456"></a><tt class="py-lineno">2456</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; print(my_mseq.count_overlap("AT", 2, -1))</tt> </tt>
<a name="L2457"></a><tt class="py-lineno">2457</tt>  <tt class="py-line"><tt class="py-docstring">        1</tt> </tt>
<a name="L2458"></a><tt class="py-lineno">2458</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_mseq.count_overlap("AT", 2, -1) == my_mseq.count("AT", 2, -1)</tt> </tt>
<a name="L2459"></a><tt class="py-lineno">2459</tt>  <tt class="py-line"><tt class="py-docstring">        True</tt> </tt>
<a name="L2460"></a><tt class="py-lineno">2460</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2461"></a><tt class="py-lineno">2461</tt>  <tt class="py-line"><tt class="py-docstring">        HOWEVER, do not use this method for such cases because the</tt> </tt>
<a name="L2462"></a><tt class="py-lineno">2462</tt>  <tt class="py-line"><tt class="py-docstring">        count() method is much for efficient.</tt> </tt>
<a name="L2463"></a><tt class="py-lineno">2463</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2464"></a><tt class="py-lineno">2464</tt>  <tt class="py-line">        <tt class="py-comment"># The implementation is currently identical to that of</tt> </tt>
<a name="L2465"></a><tt class="py-lineno">2465</tt>  <tt class="py-line">        <tt class="py-comment"># Seq.count_overlap() apart from the definition of sub_str</tt> </tt>
<a name="L2466"></a><tt class="py-lineno">2466</tt>  <tt class="py-line">        <tt class="py-name">sub_str</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">sub</tt><tt class="py-op">)</tt> </tt>
<a name="L2467"></a><tt class="py-lineno">2467</tt>  <tt class="py-line">        <tt class="py-name">self_str</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
<a name="L2468"></a><tt class="py-lineno">2468</tt>  <tt class="py-line">        <tt class="py-name">overlap_count</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L2469"></a><tt class="py-lineno">2469</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L2470"></a><tt class="py-lineno">2470</tt>  <tt class="py-line">            <tt id="link-914" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-914', 'start', 'link-224');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self_str</tt><tt class="py-op">.</tt><tt id="link-915" class="py-name"><a title="Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-915', 'find', 'link-228');">find</a></tt><tt class="py-op">(</tt><tt class="py-name">sub_str</tt><tt class="py-op">,</tt> <tt id="link-916" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-916', 'start', 'link-224');">start</a></tt><tt class="py-op">,</tt> <tt id="link-917" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-917', 'end', 'link-225');">end</a></tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt> </tt>
<a name="L2471"></a><tt class="py-lineno">2471</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-918" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-918', 'start', 'link-224');">start</a></tt> <tt class="py-op">!=</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L2472"></a><tt class="py-lineno">2472</tt>  <tt class="py-line">                <tt class="py-name">overlap_count</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L2473"></a><tt class="py-lineno">2473</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L2474"></a><tt class="py-lineno">2474</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">overlap_count</tt> </tt>
</div><a name="L2475"></a><tt class="py-lineno">2475</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.index"></a><div id="MutableSeq.index-def"><a name="L2476"></a><tt class="py-lineno">2476</tt> <a class="py-toggle" href="#" id="MutableSeq.index-toggle" onclick="return toggle('MutableSeq.index');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#index">index</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">item</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.index-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.index-expanded"><a name="L2477"></a><tt class="py-lineno">2477</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return first occurrence position of a single entry (i.e. letter).</tt> </tt>
<a name="L2478"></a><tt class="py-lineno">2478</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2479"></a><tt class="py-lineno">2479</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq = MutableSeq("ACTCGACGTCG")</tt> </tt>
<a name="L2480"></a><tt class="py-lineno">2480</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.index("A")</tt> </tt>
<a name="L2481"></a><tt class="py-lineno">2481</tt>  <tt class="py-line"><tt class="py-docstring">        0</tt> </tt>
<a name="L2482"></a><tt class="py-lineno">2482</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.index("T")</tt> </tt>
<a name="L2483"></a><tt class="py-lineno">2483</tt>  <tt class="py-line"><tt class="py-docstring">        2</tt> </tt>
<a name="L2484"></a><tt class="py-lineno">2484</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.index(Seq("T"))</tt> </tt>
<a name="L2485"></a><tt class="py-lineno">2485</tt>  <tt class="py-line"><tt class="py-docstring">        2</tt> </tt>
<a name="L2486"></a><tt class="py-lineno">2486</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2487"></a><tt class="py-lineno">2487</tt>  <tt class="py-line"><tt class="py-docstring">        Note unlike a Biopython Seq object, or Python string, multi-letter</tt> </tt>
<a name="L2488"></a><tt class="py-lineno">2488</tt>  <tt class="py-line"><tt class="py-docstring">        subsequences are not supported.  Instead this acts like an array or</tt> </tt>
<a name="L2489"></a><tt class="py-lineno">2489</tt>  <tt class="py-line"><tt class="py-docstring">        a list of the entries. There is therefore no ``.rindex()`` method.</tt> </tt>
<a name="L2490"></a><tt class="py-lineno">2490</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2491"></a><tt class="py-lineno">2491</tt>  <tt class="py-line">        <tt class="py-comment"># TODO?: return self.data.index(i)</tt> </tt>
<a name="L2492"></a><tt class="py-lineno">2492</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-919" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-919', 'i', 'link-887');">i</a></tt> <tt class="py-keyword">in</tt> <tt id="link-920" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-920', 'range', 'link-2');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-921" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-921', 'data', 'link-42');">data</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2493"></a><tt class="py-lineno">2493</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-922" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-922', 'data', 'link-42');">data</a></tt><tt class="py-op">[</tt><tt id="link-923" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-923', 'i', 'link-887');">i</a></tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">item</tt><tt class="py-op">:</tt> </tt>
<a name="L2494"></a><tt class="py-lineno">2494</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt id="link-924" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-924', 'i', 'link-887');">i</a></tt> </tt>
<a name="L2495"></a><tt class="py-lineno">2495</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"MutableSeq.index(x): x not in list"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L2496"></a><tt class="py-lineno">2496</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.reverse"></a><div id="MutableSeq.reverse-def"><a name="L2497"></a><tt class="py-lineno">2497</tt> <a class="py-toggle" href="#" id="MutableSeq.reverse-toggle" onclick="return toggle('MutableSeq.reverse');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#reverse">reverse</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.reverse-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.reverse-expanded"><a name="L2498"></a><tt class="py-lineno">2498</tt>  <tt class="py-line">        <tt class="py-docstring">"""Modify the mutable sequence to reverse itself.</tt> </tt>
<a name="L2499"></a><tt class="py-lineno">2499</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2500"></a><tt class="py-lineno">2500</tt>  <tt class="py-line"><tt class="py-docstring">        No return value.</tt> </tt>
<a name="L2501"></a><tt class="py-lineno">2501</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2502"></a><tt class="py-lineno">2502</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-925" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-925', 'data', 'link-42');">data</a></tt><tt class="py-op">.</tt><tt id="link-926" class="py-name" targets="Method Bio.Pathway.Reaction.reverse()=Bio.Pathway.Reaction-class.html#reverse,Method Bio.Seq.MutableSeq.reverse()=Bio.Seq.MutableSeq-class.html#reverse"><a title="Bio.Pathway.Reaction.reverse
Bio.Seq.MutableSeq.reverse" class="py-name" href="#" onclick="return doclink('link-926', 'reverse', 'link-926');">reverse</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L2503"></a><tt class="py-lineno">2503</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.complement"></a><div id="MutableSeq.complement-def"><a name="L2504"></a><tt class="py-lineno">2504</tt> <a class="py-toggle" href="#" id="MutableSeq.complement-toggle" onclick="return toggle('MutableSeq.complement');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#complement">complement</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.complement-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.complement-expanded"><a name="L2505"></a><tt class="py-lineno">2505</tt>  <tt class="py-line">        <tt class="py-docstring">"""Modify the mutable sequence to take on its complement.</tt> </tt>
<a name="L2506"></a><tt class="py-lineno">2506</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2507"></a><tt class="py-lineno">2507</tt>  <tt class="py-line"><tt class="py-docstring">        Trying to complement a protein sequence raises an exception.</tt> </tt>
<a name="L2508"></a><tt class="py-lineno">2508</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2509"></a><tt class="py-lineno">2509</tt>  <tt class="py-line"><tt class="py-docstring">        No return value.</tt> </tt>
<a name="L2510"></a><tt class="py-lineno">2510</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2511"></a><tt class="py-lineno">2511</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-927" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-927', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-928" class="py-name"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-928', '_get_base_alphabet', 'link-297');">_get_base_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-929" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-929', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L2512"></a><tt class="py-lineno">2512</tt>  <tt class="py-line">                      <tt id="link-930" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-930', 'Alphabet', 'link-8');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-931" class="py-name"><a title="Bio.Alphabet.ProteinAlphabet" class="py-name" href="#" onclick="return doclink('link-931', 'ProteinAlphabet', 'link-300');">ProteinAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2513"></a><tt class="py-lineno">2513</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Proteins do not have complements!"</tt><tt class="py-op">)</tt> </tt>
<a name="L2514"></a><tt class="py-lineno">2514</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-932" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-932', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt id="link-933" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-933', 'IUPAC', 'link-11');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-934" class="py-name"><a title="Bio.Alphabet.IUPAC.ambiguous_dna" class="py-name" href="#" onclick="return doclink('link-934', 'ambiguous_dna', 'link-328');">ambiguous_dna</a></tt><tt class="py-op">,</tt> <tt id="link-935" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-935', 'IUPAC', 'link-11');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-936" class="py-name"><a title="Bio.Alphabet.IUPAC.unambiguous_dna" class="py-name" href="#" onclick="return doclink('link-936', 'unambiguous_dna', 'link-322');">unambiguous_dna</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2515"></a><tt class="py-lineno">2515</tt>  <tt class="py-line">            <tt class="py-name">d</tt> <tt class="py-op">=</tt> <tt id="link-937" class="py-name"><a title="Bio.Data.IUPACData.ambiguous_dna_complement" class="py-name" href="#" onclick="return doclink('link-937', 'ambiguous_dna_complement', 'link-15');">ambiguous_dna_complement</a></tt> </tt>
<a name="L2516"></a><tt class="py-lineno">2516</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-938" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-938', 'alphabet', 'link-44');">alphabet</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt id="link-939" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-939', 'IUPAC', 'link-11');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-940" class="py-name"><a title="Bio.Alphabet.IUPAC.ambiguous_rna" class="py-name" href="#" onclick="return doclink('link-940', 'ambiguous_rna', 'link-331');">ambiguous_rna</a></tt><tt class="py-op">,</tt> <tt id="link-941" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-941', 'IUPAC', 'link-11');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-942" class="py-name"><a title="Bio.Alphabet.IUPAC.unambiguous_rna" class="py-name" href="#" onclick="return doclink('link-942', 'unambiguous_rna', 'link-325');">unambiguous_rna</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2517"></a><tt class="py-lineno">2517</tt>  <tt class="py-line">            <tt class="py-name">d</tt> <tt class="py-op">=</tt> <tt id="link-943" class="py-name"><a title="Bio.Data.IUPACData.ambiguous_rna_complement" class="py-name" href="#" onclick="return doclink('link-943', 'ambiguous_rna_complement', 'link-16');">ambiguous_rna_complement</a></tt> </tt>
<a name="L2518"></a><tt class="py-lineno">2518</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-string">"U"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-944" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-944', 'data', 'link-42');">data</a></tt> <tt class="py-keyword">and</tt> <tt class="py-string">"T"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-945" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-945', 'data', 'link-42');">data</a></tt><tt class="py-op">:</tt> </tt>
<a name="L2519"></a><tt class="py-lineno">2519</tt>  <tt class="py-line">            <tt class="py-comment"># TODO - Handle this cleanly?</tt> </tt>
<a name="L2520"></a><tt class="py-lineno">2520</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Mixed RNA/DNA found"</tt><tt class="py-op">)</tt> </tt>
<a name="L2521"></a><tt class="py-lineno">2521</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-string">"U"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-946" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-946', 'data', 'link-42');">data</a></tt><tt class="py-op">:</tt> </tt>
<a name="L2522"></a><tt class="py-lineno">2522</tt>  <tt class="py-line">            <tt class="py-name">d</tt> <tt class="py-op">=</tt> <tt id="link-947" class="py-name"><a title="Bio.Data.IUPACData.ambiguous_rna_complement" class="py-name" href="#" onclick="return doclink('link-947', 'ambiguous_rna_complement', 'link-16');">ambiguous_rna_complement</a></tt> </tt>
<a name="L2523"></a><tt class="py-lineno">2523</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L2524"></a><tt class="py-lineno">2524</tt>  <tt class="py-line">            <tt class="py-name">d</tt> <tt class="py-op">=</tt> <tt id="link-948" class="py-name"><a title="Bio.Data.IUPACData.ambiguous_dna_complement" class="py-name" href="#" onclick="return doclink('link-948', 'ambiguous_dna_complement', 'link-15');">ambiguous_dna_complement</a></tt> </tt>
<a name="L2525"></a><tt class="py-lineno">2525</tt>  <tt class="py-line">        <tt class="py-name">mixed</tt> <tt class="py-op">=</tt> <tt class="py-name">d</tt><tt class="py-op">.</tt><tt id="link-949" class="py-name" targets="Method Bio.Align.substitution_matrices.Array.copy()=Bio.Align.substitution_matrices.Array-class.html#copy,Method Bio.Crystal.Crystal.copy()=Bio.Crystal.Crystal-class.html#copy,Method Bio.File._IndexedSeqFileDict.copy()=Bio.File._IndexedSeqFileDict-class.html#copy,Method Bio.PDB.Atom.Atom.copy()=Bio.PDB.Atom.Atom-class.html#copy,Method Bio.PDB.Entity.Entity.copy()=Bio.PDB.Entity.Entity-class.html#copy,Method Bio.PDB.vectors.Vector.copy()=Bio.PDB.vectors.Vector-class.html#copy"><a title="Bio.Align.substitution_matrices.Array.copy
Bio.Crystal.Crystal.copy
Bio.File._IndexedSeqFileDict.copy
Bio.PDB.Atom.Atom.copy
Bio.PDB.Entity.Entity.copy
Bio.PDB.vectors.Vector.copy" class="py-name" href="#" onclick="return doclink('link-949', 'copy', 'link-949');">copy</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt>  <tt class="py-comment"># We're going to edit this to be mixed case!</tt> </tt>
<a name="L2526"></a><tt class="py-lineno">2526</tt>  <tt class="py-line">        <tt class="py-name">mixed</tt><tt class="py-op">.</tt><tt id="link-950" class="py-name" targets="Method Bio.Align.substitution_matrices.Array.update()=Bio.Align.substitution_matrices.Array-class.html#update,Method Bio.File._IndexedSeqFileDict.update()=Bio.File._IndexedSeqFileDict-class.html#update,Method Bio.Index._InMemoryIndex.update()=Bio.Index._InMemoryIndex-class.html#update,Method Bio.SeqRecord._RestrictedDict.update()=Bio.SeqRecord._RestrictedDict-class.html#update"><a title="Bio.Align.substitution_matrices.Array.update
Bio.File._IndexedSeqFileDict.update
Bio.Index._InMemoryIndex.update
Bio.SeqRecord._RestrictedDict.update" class="py-name" href="#" onclick="return doclink('link-950', 'update', 'link-950');">update</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt id="link-951" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.x=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#x,Variable Bio.Statistics.lowess.x=Bio.Statistics.lowess-module.html#x"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-951', 'x', 'link-951');">x</a></tt><tt class="py-op">.</tt><tt id="link-952" class="py-name"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-952', 'lower', 'link-32');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-953" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.y=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#y"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-953', 'y', 'link-953');">y</a></tt><tt class="py-op">.</tt><tt id="link-954" class="py-name"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-954', 'lower', 'link-32');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-955" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-955', 'x', 'link-951');">x</a></tt><tt class="py-op">,</tt> <tt id="link-956" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-956', 'y', 'link-953');">y</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">d</tt><tt class="py-op">.</tt><tt id="link-957" class="py-name" targets="Method Bio.Align.substitution_matrices.Array.items()=Bio.Align.substitution_matrices.Array-class.html#items,Method Bio.Crystal.Crystal.items()=Bio.Crystal.Crystal-class.html#items,Method Bio.File._IndexedSeqFileDict.items()=Bio.File._IndexedSeqFileDict-class.html#items,Method Bio.Phylo.PhyloXML.Events.items()=Bio.Phylo.PhyloXML.Events-class.html#items,Variable Bio.SearchIO._model.query.QueryResult.items=Bio.SearchIO._model.query.QueryResult-class.html#items,Method BioSQL.BioSeqDatabase.BioSeqDatabase.items()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#items,Method BioSQL.BioSeqDatabase.DBServer.items()=BioSQL.BioSeqDatabase.DBServer-class.html#items"><a title="Bio.Align.substitution_matrices.Array.items
Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-957', 'items', 'link-957');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L2527"></a><tt class="py-lineno">2527</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-958" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-958', 'data', 'link-42');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">mixed</tt><tt class="py-op">[</tt><tt class="py-name">_</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt class="py-name">_</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-959" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-959', 'data', 'link-42');">data</a></tt><tt class="py-op">]</tt> </tt>
<a name="L2528"></a><tt class="py-lineno">2528</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-960" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-960', 'data', 'link-42');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">array</tt><tt class="py-op">.</tt><tt class="py-name">array</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">array_indicator</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-961" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-961', 'data', 'link-42');">data</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L2529"></a><tt class="py-lineno">2529</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.reverse_complement"></a><div id="MutableSeq.reverse_complement-def"><a name="L2530"></a><tt class="py-lineno">2530</tt> <a class="py-toggle" href="#" id="MutableSeq.reverse_complement-toggle" onclick="return toggle('MutableSeq.reverse_complement');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#reverse_complement">reverse_complement</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.reverse_complement-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.reverse_complement-expanded"><a name="L2531"></a><tt class="py-lineno">2531</tt>  <tt class="py-line">        <tt class="py-docstring">"""Modify the mutable sequence to take on its reverse complement.</tt> </tt>
<a name="L2532"></a><tt class="py-lineno">2532</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2533"></a><tt class="py-lineno">2533</tt>  <tt class="py-line"><tt class="py-docstring">        Trying to reverse complement a protein sequence raises an exception.</tt> </tt>
<a name="L2534"></a><tt class="py-lineno">2534</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2535"></a><tt class="py-lineno">2535</tt>  <tt class="py-line"><tt class="py-docstring">        No return value.</tt> </tt>
<a name="L2536"></a><tt class="py-lineno">2536</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2537"></a><tt class="py-lineno">2537</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-962" class="py-name"><a title="Bio.Seq.MutableSeq.complement
Bio.Seq.Seq.complement
Bio.Seq.UnknownSeq.complement
Bio.Seq.complement" class="py-name" href="#" onclick="return doclink('link-962', 'complement', 'link-312');">complement</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L2538"></a><tt class="py-lineno">2538</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-963" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-963', 'data', 'link-42');">data</a></tt><tt class="py-op">.</tt><tt id="link-964" class="py-name"><a title="Bio.Pathway.Reaction.reverse
Bio.Seq.MutableSeq.reverse" class="py-name" href="#" onclick="return doclink('link-964', 'reverse', 'link-926');">reverse</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L2539"></a><tt class="py-lineno">2539</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.extend"></a><div id="MutableSeq.extend-def"><a name="L2540"></a><tt class="py-lineno">2540</tt> <a class="py-toggle" href="#" id="MutableSeq.extend-toggle" onclick="return toggle('MutableSeq.extend');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#extend">extend</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.extend-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.extend-expanded"><a name="L2541"></a><tt class="py-lineno">2541</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add a sequence to the original mutable sequence object.</tt> </tt>
<a name="L2542"></a><tt class="py-lineno">2542</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2543"></a><tt class="py-lineno">2543</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq = MutableSeq('ACTCGACGTCG')</tt> </tt>
<a name="L2544"></a><tt class="py-lineno">2544</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.extend('A')</tt> </tt>
<a name="L2545"></a><tt class="py-lineno">2545</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L2546"></a><tt class="py-lineno">2546</tt>  <tt class="py-line"><tt class="py-docstring">        MutableSeq('ACTCGACGTCGA')</tt> </tt>
<a name="L2547"></a><tt class="py-lineno">2547</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq.extend('TTT')</tt> </tt>
<a name="L2548"></a><tt class="py-lineno">2548</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_seq</tt> </tt>
<a name="L2549"></a><tt class="py-lineno">2549</tt>  <tt class="py-line"><tt class="py-docstring">        MutableSeq('ACTCGACGTCGATTT')</tt> </tt>
<a name="L2550"></a><tt class="py-lineno">2550</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2551"></a><tt class="py-lineno">2551</tt>  <tt class="py-line"><tt class="py-docstring">        No return value.</tt> </tt>
<a name="L2552"></a><tt class="py-lineno">2552</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2553"></a><tt class="py-lineno">2553</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-965" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-965', 'other', 'link-57');">other</a></tt><tt class="py-op">,</tt> <tt id="link-966" class="py-name"><a title="Bio.Seq.MutableSeq" class="py-name" href="#" onclick="return doclink('link-966', 'MutableSeq', 'link-83');">MutableSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2554"></a><tt class="py-lineno">2554</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">c</tt> <tt class="py-keyword">in</tt> <tt id="link-967" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-967', 'other', 'link-57');">other</a></tt><tt class="py-op">.</tt><tt id="link-968" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-968', 'data', 'link-42');">data</a></tt><tt class="py-op">:</tt> </tt>
<a name="L2555"></a><tt class="py-lineno">2555</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-969" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-969', 'data', 'link-42');">data</a></tt><tt class="py-op">.</tt><tt id="link-970" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-970', 'append', 'link-884');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">)</tt> </tt>
<a name="L2556"></a><tt class="py-lineno">2556</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L2557"></a><tt class="py-lineno">2557</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">c</tt> <tt class="py-keyword">in</tt> <tt id="link-971" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-971', 'other', 'link-57');">other</a></tt><tt class="py-op">:</tt> </tt>
<a name="L2558"></a><tt class="py-lineno">2558</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-972" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-972', 'data', 'link-42');">data</a></tt><tt class="py-op">.</tt><tt id="link-973" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-973', 'append', 'link-884');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">)</tt> </tt>
</div><a name="L2559"></a><tt class="py-lineno">2559</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.toseq"></a><div id="MutableSeq.toseq-def"><a name="L2560"></a><tt class="py-lineno">2560</tt> <a class="py-toggle" href="#" id="MutableSeq.toseq-toggle" onclick="return toggle('MutableSeq.toseq');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#toseq">toseq</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.toseq-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.toseq-expanded"><a name="L2561"></a><tt class="py-lineno">2561</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the full sequence as a new immutable Seq object.</tt> </tt>
<a name="L2562"></a><tt class="py-lineno">2562</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2563"></a><tt class="py-lineno">2563</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L2564"></a><tt class="py-lineno">2564</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.Alphabet import IUPAC</tt> </tt>
<a name="L2565"></a><tt class="py-lineno">2565</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_mseq = MutableSeq("MKQHKAMIVALIVICITAVVAAL",</tt> </tt>
<a name="L2566"></a><tt class="py-lineno">2566</tt>  <tt class="py-line"><tt class="py-docstring">        ...                      IUPAC.protein)</tt> </tt>
<a name="L2567"></a><tt class="py-lineno">2567</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_mseq</tt> </tt>
<a name="L2568"></a><tt class="py-lineno">2568</tt>  <tt class="py-line"><tt class="py-docstring">        MutableSeq('MKQHKAMIVALIVICITAVVAAL', IUPACProtein())</tt> </tt>
<a name="L2569"></a><tt class="py-lineno">2569</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; my_mseq.toseq()</tt> </tt>
<a name="L2570"></a><tt class="py-lineno">2570</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('MKQHKAMIVALIVICITAVVAAL', IUPACProtein())</tt> </tt>
<a name="L2571"></a><tt class="py-lineno">2571</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2572"></a><tt class="py-lineno">2572</tt>  <tt class="py-line"><tt class="py-docstring">        Note that the alphabet is preserved.</tt> </tt>
<a name="L2573"></a><tt class="py-lineno">2573</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2574"></a><tt class="py-lineno">2574</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-974" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-974', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">.</tt><tt id="link-975" class="py-name"><a title="Bio.Seq.MutableSeq.join
Bio.Seq.Seq.join
Bio.Seq.UnknownSeq.join" class="py-name" href="#" onclick="return doclink('link-975', 'join', 'link-28');">join</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-976" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-976', 'data', 'link-42');">data</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-977" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-977', 'alphabet', 'link-44');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L2575"></a><tt class="py-lineno">2575</tt>  <tt class="py-line"> </tt>
<a name="MutableSeq.join"></a><div id="MutableSeq.join-def"><a name="L2576"></a><tt class="py-lineno">2576</tt> <a class="py-toggle" href="#" id="MutableSeq.join-toggle" onclick="return toggle('MutableSeq.join');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq.MutableSeq-class.html#join">join</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MutableSeq.join-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MutableSeq.join-expanded"><a name="L2577"></a><tt class="py-lineno">2577</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a merge of the sequences in other, spaced by the sequence from self.</tt> </tt>
<a name="L2578"></a><tt class="py-lineno">2578</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2579"></a><tt class="py-lineno">2579</tt>  <tt class="py-line"><tt class="py-docstring">        Accepts all Seq objects and Strings as objects to be concatenated with the spacer</tt> </tt>
<a name="L2580"></a><tt class="py-lineno">2580</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2581"></a><tt class="py-lineno">2581</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; concatenated = MutableSeq('NNNNN').join([Seq("AAA"), Seq("TTT"), Seq("PPP")])</tt> </tt>
<a name="L2582"></a><tt class="py-lineno">2582</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; concatenated</tt> </tt>
<a name="L2583"></a><tt class="py-lineno">2583</tt>  <tt class="py-line"><tt class="py-docstring">        Seq('AAANNNNNTTTNNNNNPPP')</tt> </tt>
<a name="L2584"></a><tt class="py-lineno">2584</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2585"></a><tt class="py-lineno">2585</tt>  <tt class="py-line"><tt class="py-docstring">        Throws error if other is not an iterable and if objects inside of the iterable</tt> </tt>
<a name="L2586"></a><tt class="py-lineno">2586</tt>  <tt class="py-line"><tt class="py-docstring">        are not Seq or String objects</tt> </tt>
<a name="L2587"></a><tt class="py-lineno">2587</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2588"></a><tt class="py-lineno">2588</tt>  <tt class="py-line">        <tt class="py-name">seq_joined</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-978" class="py-name" targets="Method Bio.Seq.MutableSeq.toseq()=Bio.Seq.MutableSeq-class.html#toseq,Method BioSQL.BioSeq.DBSeq.toseq()=BioSQL.BioSeq.DBSeq-class.html#toseq"><a title="Bio.Seq.MutableSeq.toseq
BioSQL.BioSeq.DBSeq.toseq" class="py-name" href="#" onclick="return doclink('link-978', 'toseq', 'link-978');">toseq</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-979" class="py-name"><a title="Bio.Seq.MutableSeq.join
Bio.Seq.Seq.join
Bio.Seq.UnknownSeq.join" class="py-name" href="#" onclick="return doclink('link-979', 'join', 'link-28');">join</a></tt><tt class="py-op">(</tt><tt id="link-980" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-980', 'other', 'link-57');">other</a></tt><tt class="py-op">)</tt>  <tt class="py-comment"># returns Seq object instead of MutableSeq</tt> </tt>
<a name="L2589"></a><tt class="py-lineno">2589</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">seq_joined</tt> </tt>
</div></div><a name="L2590"></a><tt class="py-lineno">2590</tt>  <tt class="py-line"> </tt>
<a name="L2591"></a><tt class="py-lineno">2591</tt>  <tt class="py-line"> </tt>
<a name="L2592"></a><tt class="py-lineno">2592</tt>  <tt class="py-line"><tt class="py-comment"># The transcribe, backward_transcribe, and translate functions are</tt> </tt>
<a name="L2593"></a><tt class="py-lineno">2593</tt>  <tt class="py-line"><tt class="py-comment"># user-friendly versions of the corresponding functions in Bio.Transcribe</tt> </tt>
<a name="L2594"></a><tt class="py-lineno">2594</tt>  <tt class="py-line"><tt class="py-comment"># and Bio.Translate. The functions work both on Seq objects, and on strings.</tt> </tt>
<a name="L2595"></a><tt class="py-lineno">2595</tt>  <tt class="py-line"> </tt>
<a name="transcribe"></a><div id="transcribe-def"><a name="L2596"></a><tt class="py-lineno">2596</tt> <a class="py-toggle" href="#" id="transcribe-toggle" onclick="return toggle('transcribe');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq-module.html#transcribe">transcribe</a><tt class="py-op">(</tt><tt class="py-param">dna</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="transcribe-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="transcribe-expanded"><a name="L2597"></a><tt class="py-lineno">2597</tt>  <tt class="py-line">    <tt class="py-docstring">"""Transcribe a DNA sequence into RNA.</tt> </tt>
<a name="L2598"></a><tt class="py-lineno">2598</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2599"></a><tt class="py-lineno">2599</tt>  <tt class="py-line"><tt class="py-docstring">    If given a string, returns a new string object.</tt> </tt>
<a name="L2600"></a><tt class="py-lineno">2600</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2601"></a><tt class="py-lineno">2601</tt>  <tt class="py-line"><tt class="py-docstring">    Given a Seq or MutableSeq, returns a new Seq object with an RNA alphabet.</tt> </tt>
<a name="L2602"></a><tt class="py-lineno">2602</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2603"></a><tt class="py-lineno">2603</tt>  <tt class="py-line"><tt class="py-docstring">    Trying to transcribe a protein or RNA sequence raises an exception.</tt> </tt>
<a name="L2604"></a><tt class="py-lineno">2604</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2605"></a><tt class="py-lineno">2605</tt>  <tt class="py-line"><tt class="py-docstring">    e.g.</tt> </tt>
<a name="L2606"></a><tt class="py-lineno">2606</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2607"></a><tt class="py-lineno">2607</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; transcribe("ACTGN")</tt> </tt>
<a name="L2608"></a><tt class="py-lineno">2608</tt>  <tt class="py-line"><tt class="py-docstring">    'ACUGN'</tt> </tt>
<a name="L2609"></a><tt class="py-lineno">2609</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L2610"></a><tt class="py-lineno">2610</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-981" class="py-name" targets="Variable Bio.Restriction.Restriction.AanI.dna=Bio.Restriction.Restriction.AanI-class.html#dna,Variable Bio.Restriction.Restriction.AarI.dna=Bio.Restriction.Restriction.AarI-class.html#dna,Variable Bio.Restriction.Restriction.AasI.dna=Bio.Restriction.Restriction.AasI-class.html#dna,Variable Bio.Restriction.Restriction.AatII.dna=Bio.Restriction.Restriction.AatII-class.html#dna,Variable Bio.Restriction.Restriction.Aba6411II.dna=Bio.Restriction.Restriction.Aba6411II-class.html#dna,Variable Bio.Restriction.Restriction.AbaB8342IV.dna=Bio.Restriction.Restriction.AbaB8342IV-class.html#dna,Variable Bio.Restriction.Restriction.AbaCIII.dna=Bio.Restriction.Restriction.AbaCIII-class.html#dna,Variable Bio.Restriction.Restriction.AbaSI.dna=Bio.Restriction.Restriction.AbaSI-class.html#dna,Variable Bio.Restriction.Restriction.AbaUMB2I.dna=Bio.Restriction.Restriction.AbaUMB2I-class.html#dna,Variable Bio.Restriction.Restriction.AbsI.dna=Bio.Restriction.Restriction.AbsI-class.html#dna,Variable Bio.Restriction.Restriction.Acc16I.dna=Bio.Restriction.Restriction.Acc16I-class.html#dna,Variable Bio.Restriction.Restriction.Acc36I.dna=Bio.Restriction.Restriction.Acc36I-class.html#dna,Variable Bio.Restriction.Restriction.Acc65I.dna=Bio.Restriction.Restriction.Acc65I-class.html#dna,Variable Bio.Restriction.Restriction.Acc65V.dna=Bio.Restriction.Restriction.Acc65V-class.html#dna,Variable Bio.Restriction.Restriction.AccB1I.dna=Bio.Restriction.Restriction.AccB1I-class.html#dna,Variable Bio.Restriction.Restriction.AccB7I.dna=Bio.Restriction.Restriction.AccB7I-class.html#dna,Variable Bio.Restriction.Restriction.AccBSI.dna=Bio.Restriction.Restriction.AccBSI-class.html#dna,Variable Bio.Restriction.Restriction.AccI.dna=Bio.Restriction.Restriction.AccI-class.html#dna,Variable Bio.Restriction.Restriction.AccII.dna=Bio.Restriction.Restriction.AccII-class.html#dna,Variable Bio.Restriction.Restriction.AccIII.dna=Bio.Restriction.Restriction.AccIII-class.html#dna,Variable Bio.Restriction.Restriction.AceIII.dna=Bio.Restriction.Restriction.AceIII-class.html#dna,Variable Bio.Restriction.Restriction.AchA6III.dna=Bio.Restriction.Restriction.AchA6III-class.html#dna,Variable Bio.Restriction.Restriction.AciI.dna=Bio.Restriction.Restriction.AciI-class.html#dna,Variable Bio.Restriction.Restriction.AclI.dna=Bio.Restriction.Restriction.AclI-class.html#dna,Variable Bio.Restriction.Restriction.AclWI.dna=Bio.Restriction.Restriction.AclWI-class.html#dna,Variable Bio.Restriction.Restriction.Aco12261II.dna=Bio.Restriction.Restriction.Aco12261II-class.html#dna,Variable Bio.Restriction.Restriction.AcoI.dna=Bio.Restriction.Restriction.AcoI-class.html#dna,Variable Bio.Restriction.Restriction.AcoY31II.dna=Bio.Restriction.Restriction.AcoY31II-class.html#dna,Variable Bio.Restriction.Restriction.AcsI.dna=Bio.Restriction.Restriction.AcsI-class.html#dna,Variable Bio.Restriction.Restriction.AcuI.dna=Bio.Restriction.Restriction.AcuI-class.html#dna,Variable Bio.Restriction.Restriction.AcvI.dna=Bio.Restriction.Restriction.AcvI-class.html#dna,Variable Bio.Restriction.Restriction.AcyI.dna=Bio.Restriction.Restriction.AcyI-class.html#dna,Variable Bio.Restriction.Restriction.AdeI.dna=Bio.Restriction.Restriction.AdeI-class.html#dna,Variable Bio.Restriction.Restriction.Adh6U21I.dna=Bio.Restriction.Restriction.Adh6U21I-class.html#dna,Variable Bio.Restriction.Restriction.AfaI.dna=Bio.Restriction.Restriction.AfaI-class.html#dna,Variable Bio.Restriction.Restriction.AfeI.dna=Bio.Restriction.Restriction.AfeI-class.html#dna,Variable Bio.Restriction.Restriction.AfiI.dna=Bio.Restriction.Restriction.AfiI-class.html#dna,Variable Bio.Restriction.Restriction.AflII.dna=Bio.Restriction.Restriction.AflII-class.html#dna,Variable Bio.Restriction.Restriction.AflIII.dna=Bio.Restriction.Restriction.AflIII-class.html#dna,Variable Bio.Restriction.Restriction.AgeI.dna=Bio.Restriction.Restriction.AgeI-class.html#dna,Variable Bio.Restriction.Restriction.AgsI.dna=Bio.Restriction.Restriction.AgsI-class.html#dna,Variable Bio.Restriction.Restriction.AhaIII.dna=Bio.Restriction.Restriction.AhaIII-class.html#dna,Variable Bio.Restriction.Restriction.AhdI.dna=Bio.Restriction.Restriction.AhdI-class.html#dna,Variable Bio.Restriction.Restriction.AhlI.dna=Bio.Restriction.Restriction.AhlI-class.html#dna,Variable Bio.Restriction.Restriction.AhyRBAHI.dna=Bio.Restriction.Restriction.AhyRBAHI-class.html#dna,Variable Bio.Restriction.Restriction.AhyYL17I.dna=Bio.Restriction.Restriction.AhyYL17I-class.html#dna,Variable Bio.Restriction.Restriction.AjiI.dna=Bio.Restriction.Restriction.AjiI-class.html#dna,Variable Bio.Restriction.Restriction.AjnI.dna=Bio.Restriction.Restriction.AjnI-class.html#dna,Variable Bio.Restriction.Restriction.AjuI.dna=Bio.Restriction.Restriction.AjuI-class.html#dna,Variable Bio.Restriction.Restriction.AleI.dna=Bio.Restriction.Restriction.AleI-class.html#dna,Variable Bio.Restriction.Restriction.AlfI.dna=Bio.Restriction.Restriction.AlfI-class.html#dna,Variable Bio.Restriction.Restriction.AloI.dna=Bio.Restriction.Restriction.AloI-class.html#dna,Variable Bio.Restriction.Restriction.AluBI.dna=Bio.Restriction.Restriction.AluBI-class.html#dna,Variable Bio.Restriction.Restriction.AluI.dna=Bio.Restriction.Restriction.AluI-class.html#dna,Variable Bio.Restriction.Restriction.Alw21I.dna=Bio.Restriction.Restriction.Alw21I-class.html#dna,Variable Bio.Restriction.Restriction.Alw26I.dna=Bio.Restriction.Restriction.Alw26I-class.html#dna,Variable Bio.Restriction.Restriction.Alw44I.dna=Bio.Restriction.Restriction.Alw44I-class.html#dna,Variable Bio.Restriction.Restriction.AlwFI.dna=Bio.Restriction.Restriction.AlwFI-class.html#dna,Variable Bio.Restriction.Restriction.AlwI.dna=Bio.Restriction.Restriction.AlwI-class.html#dna,Variable Bio.Restriction.Restriction.AlwNI.dna=Bio.Restriction.Restriction.AlwNI-class.html#dna,Variable Bio.Restriction.Restriction.Ama87I.dna=Bio.Restriction.Restriction.Ama87I-class.html#dna,Variable Bio.Restriction.Restriction.AmaCSI.dna=Bio.Restriction.Restriction.AmaCSI-class.html#dna,Variable Bio.Restriction.Restriction.Aod1I.dna=Bio.Restriction.Restriction.Aod1I-class.html#dna,Variable Bio.Restriction.Restriction.Aor13HI.dna=Bio.Restriction.Restriction.Aor13HI-class.html#dna,Variable Bio.Restriction.Restriction.Aor51HI.dna=Bio.Restriction.Restriction.Aor51HI-class.html#dna,Variable Bio.Restriction.Restriction.AoxI.dna=Bio.Restriction.Restriction.AoxI-class.html#dna,Variable Bio.Restriction.Restriction.ApaBI.dna=Bio.Restriction.Restriction.ApaBI-class.html#dna,Variable Bio.Restriction.Restriction.ApaI.dna=Bio.Restriction.Restriction.ApaI-class.html#dna,Variable Bio.Restriction.Restriction.ApaLI.dna=Bio.Restriction.Restriction.ApaLI-class.html#dna,Variable Bio.Restriction.Restriction.ApeKI.dna=Bio.Restriction.Restriction.ApeKI-class.html#dna,Variable Bio.Restriction.Restriction.ApoI.dna=Bio.Restriction.Restriction.ApoI-class.html#dna,Variable Bio.Restriction.Restriction.ApyPI.dna=Bio.Restriction.Restriction.ApyPI-class.html#dna,Variable Bio.Restriction.Restriction.AquII.dna=Bio.Restriction.Restriction.AquII-class.html#dna,Variable Bio.Restriction.Restriction.AquIII.dna=Bio.Restriction.Restriction.AquIII-class.html#dna,Variable Bio.Restriction.Restriction.AquIV.dna=Bio.Restriction.Restriction.AquIV-class.html#dna,Variable Bio.Restriction.Restriction.ArsI.dna=Bio.Restriction.Restriction.ArsI-class.html#dna,Variable Bio.Restriction.Restriction.AscI.dna=Bio.Restriction.Restriction.AscI-class.html#dna,Variable Bio.Restriction.Restriction.AseI.dna=Bio.Restriction.Restriction.AseI-class.html#dna,Variable Bio.Restriction.Restriction.Asi256I.dna=Bio.Restriction.Restriction.Asi256I-class.html#dna,Variable Bio.Restriction.Restriction.AsiGI.dna=Bio.Restriction.Restriction.AsiGI-class.html#dna,Variable Bio.Restriction.Restriction.AsiSI.dna=Bio.Restriction.Restriction.AsiSI-class.html#dna,Variable Bio.Restriction.Restriction.Asp103I.dna=Bio.Restriction.Restriction.Asp103I-class.html#dna,Variable Bio.Restriction.Restriction.Asp114pII.dna=Bio.Restriction.Restriction.Asp114pII-class.html#dna,Variable Bio.Restriction.Restriction.Asp700I.dna=Bio.Restriction.Restriction.Asp700I-class.html#dna,Variable Bio.Restriction.Restriction.Asp718I.dna=Bio.Restriction.Restriction.Asp718I-class.html#dna,Variable Bio.Restriction.Restriction.AspA2I.dna=Bio.Restriction.Restriction.AspA2I-class.html#dna,Variable Bio.Restriction.Restriction.AspBHI.dna=Bio.Restriction.Restriction.AspBHI-class.html#dna,Variable Bio.Restriction.Restriction.AspDUT2V.dna=Bio.Restriction.Restriction.AspDUT2V-class.html#dna,Variable Bio.Restriction.Restriction.AspJHL3II.dna=Bio.Restriction.Restriction.AspJHL3II-class.html#dna,Variable Bio.Restriction.Restriction.AspLEI.dna=Bio.Restriction.Restriction.AspLEI-class.html#dna,Variable Bio.Restriction.Restriction.AspNIH4III.dna=Bio.Restriction.Restriction.AspNIH4III-class.html#dna,Variable Bio.Restriction.Restriction.AspS9I.dna=Bio.Restriction.Restriction.AspS9I-class.html#dna,Variable Bio.Restriction.Restriction.AspSLV7III.dna=Bio.Restriction.Restriction.AspSLV7III-class.html#dna,Variable Bio.Restriction.Restriction.Asu14238IV.dna=Bio.Restriction.Restriction.Asu14238IV-class.html#dna,Variable Bio.Restriction.Restriction.AsuC2I.dna=Bio.Restriction.Restriction.AsuC2I-class.html#dna,Variable Bio.Restriction.Restriction.AsuHPI.dna=Bio.Restriction.Restriction.AsuHPI-class.html#dna,Variable Bio.Restriction.Restriction.AsuI.dna=Bio.Restriction.Restriction.AsuI-class.html#dna,Variable Bio.Restriction.Restriction.AsuII.dna=Bio.Restriction.Restriction.AsuII-class.html#dna,Variable Bio.Restriction.Restriction.AsuNHI.dna=Bio.Restriction.Restriction.AsuNHI-class.html#dna,Variable Bio.Restriction.Restriction.AteTI.dna=Bio.Restriction.Restriction.AteTI-class.html#dna,Variable Bio.Restriction.Restriction.AvaI.dna=Bio.Restriction.Restriction.AvaI-class.html#dna,Variable Bio.Restriction.Restriction.AvaII.dna=Bio.Restriction.Restriction.AvaII-class.html#dna,Variable Bio.Restriction.Restriction.AvaIII.dna=Bio.Restriction.Restriction.AvaIII-class.html#dna,Variable Bio.Restriction.Restriction.AvrII.dna=Bio.Restriction.Restriction.AvrII-class.html#dna,Variable Bio.Restriction.Restriction.Awo1030IV.dna=Bio.Restriction.Restriction.Awo1030IV-class.html#dna,Variable Bio.Restriction.Restriction.AxyI.dna=Bio.Restriction.Restriction.AxyI-class.html#dna,Variable Bio.Restriction.Restriction.BaeGI.dna=Bio.Restriction.Restriction.BaeGI-class.html#dna,Variable Bio.Restriction.Restriction.BaeI.dna=Bio.Restriction.Restriction.BaeI-class.html#dna,Variable Bio.Restriction.Restriction.Bag18758I.dna=Bio.Restriction.Restriction.Bag18758I-class.html#dna,Variable Bio.Restriction.Restriction.BalI.dna=Bio.Restriction.Restriction.BalI-class.html#dna,Variable Bio.Restriction.Restriction.BamHI.dna=Bio.Restriction.Restriction.BamHI-class.html#dna,Variable Bio.Restriction.Restriction.BanI.dna=Bio.Restriction.Restriction.BanI-class.html#dna,Variable Bio.Restriction.Restriction.BanII.dna=Bio.Restriction.Restriction.BanII-class.html#dna,Variable Bio.Restriction.Restriction.BanLI.dna=Bio.Restriction.Restriction.BanLI-class.html#dna,Variable Bio.Restriction.Restriction.BarI.dna=Bio.Restriction.Restriction.BarI-class.html#dna,Variable Bio.Restriction.Restriction.Bau1417V.dna=Bio.Restriction.Restriction.Bau1417V-class.html#dna,Variable Bio.Restriction.Restriction.BauI.dna=Bio.Restriction.Restriction.BauI-class.html#dna,Variable Bio.Restriction.Restriction.Bbr52II.dna=Bio.Restriction.Restriction.Bbr52II-class.html#dna,Variable Bio.Restriction.Restriction.Bbr57III.dna=Bio.Restriction.Restriction.Bbr57III-class.html#dna,Variable Bio.Restriction.Restriction.Bbr7017II.dna=Bio.Restriction.Restriction.Bbr7017II-class.html#dna,Variable Bio.Restriction.Restriction.Bbr7017III.dna=Bio.Restriction.Restriction.Bbr7017III-class.html#dna,Variable Bio.Restriction.Restriction.BbrPI.dna=Bio.Restriction.Restriction.BbrPI-class.html#dna,Variable Bio.Restriction.Restriction.BbsI.dna=Bio.Restriction.Restriction.BbsI-class.html#dna,Variable Bio.Restriction.Restriction.BbuB31I.dna=Bio.Restriction.Restriction.BbuB31I-class.html#dna,Variable Bio.Restriction.Restriction.BbuB31II.dna=Bio.Restriction.Restriction.BbuB31II-class.html#dna,Variable Bio.Restriction.Restriction.Bbv12I.dna=Bio.Restriction.Restriction.Bbv12I-class.html#dna,Variable Bio.Restriction.Restriction.BbvCI.dna=Bio.Restriction.Restriction.BbvCI-class.html#dna,Variable Bio.Restriction.Restriction.BbvI.dna=Bio.Restriction.Restriction.BbvI-class.html#dna,Variable Bio.Restriction.Restriction.BbvII.dna=Bio.Restriction.Restriction.BbvII-class.html#dna,Variable Bio.Restriction.Restriction.BccI.dna=Bio.Restriction.Restriction.BccI-class.html#dna,Variable Bio.Restriction.Restriction.Bce3081I.dna=Bio.Restriction.Restriction.Bce3081I-class.html#dna,Variable Bio.Restriction.Restriction.Bce83I.dna=Bio.Restriction.Restriction.Bce83I-class.html#dna,Variable Bio.Restriction.Restriction.BceAI.dna=Bio.Restriction.Restriction.BceAI-class.html#dna,Variable Bio.Restriction.Restriction.BceSIV.dna=Bio.Restriction.Restriction.BceSIV-class.html#dna,Variable Bio.Restriction.Restriction.BcefI.dna=Bio.Restriction.Restriction.BcefI-class.html#dna,Variable Bio.Restriction.Restriction.BcgI.dna=Bio.Restriction.Restriction.BcgI-class.html#dna,Variable Bio.Restriction.Restriction.BciT130I.dna=Bio.Restriction.Restriction.BciT130I-class.html#dna,Variable Bio.Restriction.Restriction.BciVI.dna=Bio.Restriction.Restriction.BciVI-class.html#dna,Variable Bio.Restriction.Restriction.BclI.dna=Bio.Restriction.Restriction.BclI-class.html#dna,Variable Bio.Restriction.Restriction.BcnI.dna=Bio.Restriction.Restriction.BcnI-class.html#dna,Variable Bio.Restriction.Restriction.BcoDI.dna=Bio.Restriction.Restriction.BcoDI-class.html#dna,Variable Bio.Restriction.Restriction.BcuI.dna=Bio.Restriction.Restriction.BcuI-class.html#dna,Variable Bio.Restriction.Restriction.BdaI.dna=Bio.Restriction.Restriction.BdaI-class.html#dna,Variable Bio.Restriction.Restriction.BetI.dna=Bio.Restriction.Restriction.BetI-class.html#dna,Variable Bio.Restriction.Restriction.BfaI.dna=Bio.Restriction.Restriction.BfaI-class.html#dna,Variable Bio.Restriction.Restriction.BfaSII.dna=Bio.Restriction.Restriction.BfaSII-class.html#dna,Variable Bio.Restriction.Restriction.BfiI.dna=Bio.Restriction.Restriction.BfiI-class.html#dna,Variable Bio.Restriction.Restriction.BfmI.dna=Bio.Restriction.Restriction.BfmI-class.html#dna,Variable Bio.Restriction.Restriction.BfoI.dna=Bio.Restriction.Restriction.BfoI-class.html#dna,Variable Bio.Restriction.Restriction.BfrI.dna=Bio.Restriction.Restriction.BfrI-class.html#dna,Variable Bio.Restriction.Restriction.BfuAI.dna=Bio.Restriction.Restriction.BfuAI-class.html#dna,Variable Bio.Restriction.Restriction.BfuI.dna=Bio.Restriction.Restriction.BfuI-class.html#dna,Variable Bio.Restriction.Restriction.Bga514I.dna=Bio.Restriction.Restriction.Bga514I-class.html#dna,Variable Bio.Restriction.Restriction.BglI.dna=Bio.Restriction.Restriction.BglI-class.html#dna,Variable Bio.Restriction.Restriction.BglII.dna=Bio.Restriction.Restriction.BglII-class.html#dna,Variable Bio.Restriction.Restriction.BinI.dna=Bio.Restriction.Restriction.BinI-class.html#dna,Variable Bio.Restriction.Restriction.BisI.dna=Bio.Restriction.Restriction.BisI-class.html#dna,Variable Bio.Restriction.Restriction.BkrAM31DI.dna=Bio.Restriction.Restriction.BkrAM31DI-class.html#dna,Variable Bio.Restriction.Restriction.Ble402II.dna=Bio.Restriction.Restriction.Ble402II-class.html#dna,Variable Bio.Restriction.Restriction.BlnI.dna=Bio.Restriction.Restriction.BlnI-class.html#dna,Variable Bio.Restriction.Restriction.BloAII.dna=Bio.Restriction.Restriction.BloAII-class.html#dna,Variable Bio.Restriction.Restriction.BlpI.dna=Bio.Restriction.Restriction.BlpI-class.html#dna,Variable Bio.Restriction.Restriction.BlsI.dna=Bio.Restriction.Restriction.BlsI-class.html#dna,Variable Bio.Restriction.Restriction.BmcAI.dna=Bio.Restriction.Restriction.BmcAI-class.html#dna,Variable Bio.Restriction.Restriction.Bme1390I.dna=Bio.Restriction.Restriction.Bme1390I-class.html#dna,Variable Bio.Restriction.Restriction.Bme18I.dna=Bio.Restriction.Restriction.Bme18I-class.html#dna,Variable Bio.Restriction.Restriction.BmeDI.dna=Bio.Restriction.Restriction.BmeDI-class.html#dna,Variable Bio.Restriction.Restriction.BmeRI.dna=Bio.Restriction.Restriction.BmeRI-class.html#dna,Variable Bio.Restriction.Restriction.BmeT110I.dna=Bio.Restriction.Restriction.BmeT110I-class.html#dna,Variable Bio.Restriction.Restriction.BmgBI.dna=Bio.Restriction.Restriction.BmgBI-class.html#dna,Variable Bio.Restriction.Restriction.BmgI.dna=Bio.Restriction.Restriction.BmgI-class.html#dna,Variable Bio.Restriction.Restriction.BmgT120I.dna=Bio.Restriction.Restriction.BmgT120I-class.html#dna,Variable Bio.Restriction.Restriction.BmiI.dna=Bio.Restriction.Restriction.BmiI-class.html#dna,Variable Bio.Restriction.Restriction.BmrFI.dna=Bio.Restriction.Restriction.BmrFI-class.html#dna,Variable Bio.Restriction.Restriction.BmrI.dna=Bio.Restriction.Restriction.BmrI-class.html#dna,Variable Bio.Restriction.Restriction.BmsI.dna=Bio.Restriction.Restriction.BmsI-class.html#dna,Variable Bio.Restriction.Restriction.BmtI.dna=Bio.Restriction.Restriction.BmtI-class.html#dna,Variable Bio.Restriction.Restriction.BmuI.dna=Bio.Restriction.Restriction.BmuI-class.html#dna,Variable Bio.Restriction.Restriction.BoxI.dna=Bio.Restriction.Restriction.BoxI-class.html#dna,Variable Bio.Restriction.Restriction.BpiI.dna=Bio.Restriction.Restriction.BpiI-class.html#dna,Variable Bio.Restriction.Restriction.BplI.dna=Bio.Restriction.Restriction.BplI-class.html#dna,Variable Bio.Restriction.Restriction.BpmI.dna=Bio.Restriction.Restriction.BpmI-class.html#dna,Variable Bio.Restriction.Restriction.Bpu10I.dna=Bio.Restriction.Restriction.Bpu10I-class.html#dna,Variable Bio.Restriction.Restriction.Bpu1102I.dna=Bio.Restriction.Restriction.Bpu1102I-class.html#dna,Variable Bio.Restriction.Restriction.Bpu14I.dna=Bio.Restriction.Restriction.Bpu14I-class.html#dna,Variable Bio.Restriction.Restriction.BpuEI.dna=Bio.Restriction.Restriction.BpuEI-class.html#dna,Variable Bio.Restriction.Restriction.BpuMI.dna=Bio.Restriction.Restriction.BpuMI-class.html#dna,Variable Bio.Restriction.Restriction.Bsa29I.dna=Bio.Restriction.Restriction.Bsa29I-class.html#dna,Variable Bio.Restriction.Restriction.BsaAI.dna=Bio.Restriction.Restriction.BsaAI-class.html#dna,Variable Bio.Restriction.Restriction.BsaBI.dna=Bio.Restriction.Restriction.BsaBI-class.html#dna,Variable Bio.Restriction.Restriction.BsaHI.dna=Bio.Restriction.Restriction.BsaHI-class.html#dna,Variable Bio.Restriction.Restriction.BsaI.dna=Bio.Restriction.Restriction.BsaI-class.html#dna,Variable Bio.Restriction.Restriction.BsaJI.dna=Bio.Restriction.Restriction.BsaJI-class.html#dna,Variable Bio.Restriction.Restriction.BsaWI.dna=Bio.Restriction.Restriction.BsaWI-class.html#dna,Variable Bio.Restriction.Restriction.BsaXI.dna=Bio.Restriction.Restriction.BsaXI-class.html#dna,Variable Bio.Restriction.Restriction.BsbI.dna=Bio.Restriction.Restriction.BsbI-class.html#dna,Variable Bio.Restriction.Restriction.Bsc4I.dna=Bio.Restriction.Restriction.Bsc4I-class.html#dna,Variable Bio.Restriction.Restriction.BscAI.dna=Bio.Restriction.Restriction.BscAI-class.html#dna,Variable Bio.Restriction.Restriction.BscGI.dna=Bio.Restriction.Restriction.BscGI-class.html#dna,Variable Bio.Restriction.Restriction.Bse118I.dna=Bio.Restriction.Restriction.Bse118I-class.html#dna,Variable Bio.Restriction.Restriction.Bse1I.dna=Bio.Restriction.Restriction.Bse1I-class.html#dna,Variable Bio.Restriction.Restriction.Bse21I.dna=Bio.Restriction.Restriction.Bse21I-class.html#dna,Variable Bio.Restriction.Restriction.Bse3DI.dna=Bio.Restriction.Restriction.Bse3DI-class.html#dna,Variable Bio.Restriction.Restriction.Bse8I.dna=Bio.Restriction.Restriction.Bse8I-class.html#dna,Variable Bio.Restriction.Restriction.BseAI.dna=Bio.Restriction.Restriction.BseAI-class.html#dna,Variable Bio.Restriction.Restriction.BseBI.dna=Bio.Restriction.Restriction.BseBI-class.html#dna,Variable Bio.Restriction.Restriction.BseCI.dna=Bio.Restriction.Restriction.BseCI-class.html#dna,Variable Bio.Restriction.Restriction.BseDI.dna=Bio.Restriction.Restriction.BseDI-class.html#dna,Variable Bio.Restriction.Restriction.BseGI.dna=Bio.Restriction.Restriction.BseGI-class.html#dna,Variable Bio.Restriction.Restriction.BseJI.dna=Bio.Restriction.Restriction.BseJI-class.html#dna,Variable Bio.Restriction.Restriction.BseLI.dna=Bio.Restriction.Restriction.BseLI-class.html#dna,Variable Bio.Restriction.Restriction.BseMI.dna=Bio.Restriction.Restriction.BseMI-class.html#dna,Variable Bio.Restriction.Restriction.BseMII.dna=Bio.Restriction.Restriction.BseMII-class.html#dna,Variable Bio.Restriction.Restriction.BseNI.dna=Bio.Restriction.Restriction.BseNI-class.html#dna,Variable Bio.Restriction.Restriction.BsePI.dna=Bio.Restriction.Restriction.BsePI-class.html#dna,Variable Bio.Restriction.Restriction.BseRI.dna=Bio.Restriction.Restriction.BseRI-class.html#dna,Variable Bio.Restriction.Restriction.BseSI.dna=Bio.Restriction.Restriction.BseSI-class.html#dna,Variable Bio.Restriction.Restriction.BseX3I.dna=Bio.Restriction.Restriction.BseX3I-class.html#dna,Variable Bio.Restriction.Restriction.BseXI.dna=Bio.Restriction.Restriction.BseXI-class.html#dna,Variable Bio.Restriction.Restriction.BseYI.dna=Bio.Restriction.Restriction.BseYI-class.html#dna,Variable Bio.Restriction.Restriction.BsgI.dna=Bio.Restriction.Restriction.BsgI-class.html#dna,Variable Bio.Restriction.Restriction.Bsh1236I.dna=Bio.Restriction.Restriction.Bsh1236I-class.html#dna,Variable Bio.Restriction.Restriction.Bsh1285I.dna=Bio.Restriction.Restriction.Bsh1285I-class.html#dna,Variable Bio.Restriction.Restriction.BshFI.dna=Bio.Restriction.Restriction.BshFI-class.html#dna,Variable Bio.Restriction.Restriction.BshNI.dna=Bio.Restriction.Restriction.BshNI-class.html#dna,Variable Bio.Restriction.Restriction.BshTI.dna=Bio.Restriction.Restriction.BshTI-class.html#dna,Variable Bio.Restriction.Restriction.BshVI.dna=Bio.Restriction.Restriction.BshVI-class.html#dna,Variable Bio.Restriction.Restriction.BsiEI.dna=Bio.Restriction.Restriction.BsiEI-class.html#dna,Variable Bio.Restriction.Restriction.BsiHKAI.dna=Bio.Restriction.Restriction.BsiHKAI-class.html#dna,Variable Bio.Restriction.Restriction.BsiHKCI.dna=Bio.Restriction.Restriction.BsiHKCI-class.html#dna,Variable Bio.Restriction.Restriction.BsiI.dna=Bio.Restriction.Restriction.BsiI-class.html#dna,Variable Bio.Restriction.Restriction.BsiSI.dna=Bio.Restriction.Restriction.BsiSI-class.html#dna,Variable Bio.Restriction.Restriction.BsiWI.dna=Bio.Restriction.Restriction.BsiWI-class.html#dna,Variable Bio.Restriction.Restriction.BsiYI.dna=Bio.Restriction.Restriction.BsiYI-class.html#dna,Variable Bio.Restriction.Restriction.BslFI.dna=Bio.Restriction.Restriction.BslFI-class.html#dna,Variable Bio.Restriction.Restriction.BslI.dna=Bio.Restriction.Restriction.BslI-class.html#dna,Variable Bio.Restriction.Restriction.BsmAI.dna=Bio.Restriction.Restriction.BsmAI-class.html#dna,Variable Bio.Restriction.Restriction.BsmBI.dna=Bio.Restriction.Restriction.BsmBI-class.html#dna,Variable Bio.Restriction.Restriction.BsmFI.dna=Bio.Restriction.Restriction.BsmFI-class.html#dna,Variable Bio.Restriction.Restriction.BsmI.dna=Bio.Restriction.Restriction.BsmI-class.html#dna,Variable Bio.Restriction.Restriction.BsnI.dna=Bio.Restriction.Restriction.BsnI-class.html#dna,Variable Bio.Restriction.Restriction.Bso31I.dna=Bio.Restriction.Restriction.Bso31I-class.html#dna,Variable Bio.Restriction.Restriction.BsoBI.dna=Bio.Restriction.Restriction.BsoBI-class.html#dna,Variable Bio.Restriction.Restriction.Bsp119I.dna=Bio.Restriction.Restriction.Bsp119I-class.html#dna,Variable Bio.Restriction.Restriction.Bsp120I.dna=Bio.Restriction.Restriction.Bsp120I-class.html#dna,Variable Bio.Restriction.Restriction.Bsp1286I.dna=Bio.Restriction.Restriction.Bsp1286I-class.html#dna,Variable Bio.Restriction.Restriction.Bsp13I.dna=Bio.Restriction.Restriction.Bsp13I-class.html#dna,Variable Bio.Restriction.Restriction.Bsp1407I.dna=Bio.Restriction.Restriction.Bsp1407I-class.html#dna,Variable Bio.Restriction.Restriction.Bsp143I.dna=Bio.Restriction.Restriction.Bsp143I-class.html#dna,Variable Bio.Restriction.Restriction.Bsp1720I.dna=Bio.Restriction.Restriction.Bsp1720I-class.html#dna,Variable Bio.Restriction.Restriction.Bsp19I.dna=Bio.Restriction.Restriction.Bsp19I-class.html#dna,Variable Bio.Restriction.Restriction.Bsp24I.dna=Bio.Restriction.Restriction.Bsp24I-class.html#dna,Variable Bio.Restriction.Restriction.Bsp3004IV.dna=Bio.Restriction.Restriction.Bsp3004IV-class.html#dna,Variable Bio.Restriction.Restriction.Bsp460III.dna=Bio.Restriction.Restriction.Bsp460III-class.html#dna,Variable Bio.Restriction.Restriction.Bsp68I.dna=Bio.Restriction.Restriction.Bsp68I-class.html#dna,Variable Bio.Restriction.Restriction.BspACI.dna=Bio.Restriction.Restriction.BspACI-class.html#dna,Variable Bio.Restriction.Restriction.BspANI.dna=Bio.Restriction.Restriction.BspANI-class.html#dna,Variable Bio.Restriction.Restriction.BspCNI.dna=Bio.Restriction.Restriction.BspCNI-class.html#dna,Variable Bio.Restriction.Restriction.BspD6I.dna=Bio.Restriction.Restriction.BspD6I-class.html#dna,Variable Bio.Restriction.Restriction.BspDI.dna=Bio.Restriction.Restriction.BspDI-class.html#dna,Variable Bio.Restriction.Restriction.BspEI.dna=Bio.Restriction.Restriction.BspEI-class.html#dna,Variable Bio.Restriction.Restriction.BspFNI.dna=Bio.Restriction.Restriction.BspFNI-class.html#dna,Variable Bio.Restriction.Restriction.BspGI.dna=Bio.Restriction.Restriction.BspGI-class.html#dna,Variable Bio.Restriction.Restriction.BspHI.dna=Bio.Restriction.Restriction.BspHI-class.html#dna,Variable Bio.Restriction.Restriction.BspLI.dna=Bio.Restriction.Restriction.BspLI-class.html#dna,Variable Bio.Restriction.Restriction.BspLU11I.dna=Bio.Restriction.Restriction.BspLU11I-class.html#dna,Variable Bio.Restriction.Restriction.BspMAI.dna=Bio.Restriction.Restriction.BspMAI-class.html#dna,Variable Bio.Restriction.Restriction.BspMI.dna=Bio.Restriction.Restriction.BspMI-class.html#dna,Variable Bio.Restriction.Restriction.BspMII.dna=Bio.Restriction.Restriction.BspMII-class.html#dna,Variable Bio.Restriction.Restriction.BspNCI.dna=Bio.Restriction.Restriction.BspNCI-class.html#dna,Variable Bio.Restriction.Restriction.BspOI.dna=Bio.Restriction.Restriction.BspOI-class.html#dna,Variable Bio.Restriction.Restriction.BspPI.dna=Bio.Restriction.Restriction.BspPI-class.html#dna,Variable Bio.Restriction.Restriction.BspQI.dna=Bio.Restriction.Restriction.BspQI-class.html#dna,Variable Bio.Restriction.Restriction.BspT104I.dna=Bio.Restriction.Restriction.BspT104I-class.html#dna,Variable Bio.Restriction.Restriction.BspT107I.dna=Bio.Restriction.Restriction.BspT107I-class.html#dna,Variable Bio.Restriction.Restriction.BspTI.dna=Bio.Restriction.Restriction.BspTI-class.html#dna,Variable Bio.Restriction.Restriction.BspTNI.dna=Bio.Restriction.Restriction.BspTNI-class.html#dna,Variable Bio.Restriction.Restriction.BsrBI.dna=Bio.Restriction.Restriction.BsrBI-class.html#dna,Variable Bio.Restriction.Restriction.BsrDI.dna=Bio.Restriction.Restriction.BsrDI-class.html#dna,Variable Bio.Restriction.Restriction.BsrFI.dna=Bio.Restriction.Restriction.BsrFI-class.html#dna,Variable Bio.Restriction.Restriction.BsrGI.dna=Bio.Restriction.Restriction.BsrGI-class.html#dna,Variable Bio.Restriction.Restriction.BsrI.dna=Bio.Restriction.Restriction.BsrI-class.html#dna,Variable Bio.Restriction.Restriction.BssAI.dna=Bio.Restriction.Restriction.BssAI-class.html#dna,Variable Bio.Restriction.Restriction.BssECI.dna=Bio.Restriction.Restriction.BssECI-class.html#dna,Variable Bio.Restriction.Restriction.BssHII.dna=Bio.Restriction.Restriction.BssHII-class.html#dna,Variable Bio.Restriction.Restriction.BssMI.dna=Bio.Restriction.Restriction.BssMI-class.html#dna,Variable Bio.Restriction.Restriction.BssNAI.dna=Bio.Restriction.Restriction.BssNAI-class.html#dna,Variable Bio.Restriction.Restriction.BssNI.dna=Bio.Restriction.Restriction.BssNI-class.html#dna,Variable Bio.Restriction.Restriction.BssSI.dna=Bio.Restriction.Restriction.BssSI-class.html#dna,Variable Bio.Restriction.Restriction.BssT1I.dna=Bio.Restriction.Restriction.BssT1I-class.html#dna,Variable Bio.Restriction.Restriction.Bst1107I.dna=Bio.Restriction.Restriction.Bst1107I-class.html#dna,Variable Bio.Restriction.Restriction.Bst2BI.dna=Bio.Restriction.Restriction.Bst2BI-class.html#dna,Variable Bio.Restriction.Restriction.Bst2UI.dna=Bio.Restriction.Restriction.Bst2UI-class.html#dna,Variable Bio.Restriction.Restriction.Bst4CI.dna=Bio.Restriction.Restriction.Bst4CI-class.html#dna,Variable Bio.Restriction.Restriction.Bst6I.dna=Bio.Restriction.Restriction.Bst6I-class.html#dna,Variable Bio.Restriction.Restriction.BstACI.dna=Bio.Restriction.Restriction.BstACI-class.html#dna,Variable Bio.Restriction.Restriction.BstAFI.dna=Bio.Restriction.Restriction.BstAFI-class.html#dna,Variable Bio.Restriction.Restriction.BstAPI.dna=Bio.Restriction.Restriction.BstAPI-class.html#dna,Variable Bio.Restriction.Restriction.BstAUI.dna=Bio.Restriction.Restriction.BstAUI-class.html#dna,Variable Bio.Restriction.Restriction.BstBAI.dna=Bio.Restriction.Restriction.BstBAI-class.html#dna,Variable Bio.Restriction.Restriction.BstBI.dna=Bio.Restriction.Restriction.BstBI-class.html#dna,Variable Bio.Restriction.Restriction.BstC8I.dna=Bio.Restriction.Restriction.BstC8I-class.html#dna,Variable Bio.Restriction.Restriction.BstDEI.dna=Bio.Restriction.Restriction.BstDEI-class.html#dna,Variable Bio.Restriction.Restriction.BstDSI.dna=Bio.Restriction.Restriction.BstDSI-class.html#dna,Variable Bio.Restriction.Restriction.BstEII.dna=Bio.Restriction.Restriction.BstEII-class.html#dna,Variable Bio.Restriction.Restriction.BstENI.dna=Bio.Restriction.Restriction.BstENI-class.html#dna,Variable Bio.Restriction.Restriction.BstF5I.dna=Bio.Restriction.Restriction.BstF5I-class.html#dna,Variable Bio.Restriction.Restriction.BstFNI.dna=Bio.Restriction.Restriction.BstFNI-class.html#dna,Variable Bio.Restriction.Restriction.BstH2I.dna=Bio.Restriction.Restriction.BstH2I-class.html#dna,Variable Bio.Restriction.Restriction.BstHHI.dna=Bio.Restriction.Restriction.BstHHI-class.html#dna,Variable Bio.Restriction.Restriction.BstKTI.dna=Bio.Restriction.Restriction.BstKTI-class.html#dna,Variable Bio.Restriction.Restriction.BstMAI.dna=Bio.Restriction.Restriction.BstMAI-class.html#dna,Variable Bio.Restriction.Restriction.BstMBI.dna=Bio.Restriction.Restriction.BstMBI-class.html#dna,Variable Bio.Restriction.Restriction.BstMCI.dna=Bio.Restriction.Restriction.BstMCI-class.html#dna,Variable Bio.Restriction.Restriction.BstMWI.dna=Bio.Restriction.Restriction.BstMWI-class.html#dna,Variable Bio.Restriction.Restriction.BstNI.dna=Bio.Restriction.Restriction.BstNI-class.html#dna,Variable Bio.Restriction.Restriction.BstNSI.dna=Bio.Restriction.Restriction.BstNSI-class.html#dna,Variable Bio.Restriction.Restriction.BstPAI.dna=Bio.Restriction.Restriction.BstPAI-class.html#dna,Variable Bio.Restriction.Restriction.BstPI.dna=Bio.Restriction.Restriction.BstPI-class.html#dna,Variable Bio.Restriction.Restriction.BstSCI.dna=Bio.Restriction.Restriction.BstSCI-class.html#dna,Variable Bio.Restriction.Restriction.BstSFI.dna=Bio.Restriction.Restriction.BstSFI-class.html#dna,Variable Bio.Restriction.Restriction.BstSLI.dna=Bio.Restriction.Restriction.BstSLI-class.html#dna,Variable Bio.Restriction.Restriction.BstSNI.dna=Bio.Restriction.Restriction.BstSNI-class.html#dna,Variable Bio.Restriction.Restriction.BstUI.dna=Bio.Restriction.Restriction.BstUI-class.html#dna,Variable Bio.Restriction.Restriction.BstV1I.dna=Bio.Restriction.Restriction.BstV1I-class.html#dna,Variable Bio.Restriction.Restriction.BstV2I.dna=Bio.Restriction.Restriction.BstV2I-class.html#dna,Variable Bio.Restriction.Restriction.BstX2I.dna=Bio.Restriction.Restriction.BstX2I-class.html#dna,Variable Bio.Restriction.Restriction.BstXI.dna=Bio.Restriction.Restriction.BstXI-class.html#dna,Variable Bio.Restriction.Restriction.BstYI.dna=Bio.Restriction.Restriction.BstYI-class.html#dna,Variable Bio.Restriction.Restriction.BstZ17I.dna=Bio.Restriction.Restriction.BstZ17I-class.html#dna,Variable Bio.Restriction.Restriction.BstZI.dna=Bio.Restriction.Restriction.BstZI-class.html#dna,Variable Bio.Restriction.Restriction.Bsu15I.dna=Bio.Restriction.Restriction.Bsu15I-class.html#dna,Variable Bio.Restriction.Restriction.Bsu36I.dna=Bio.Restriction.Restriction.Bsu36I-class.html#dna,Variable Bio.Restriction.Restriction.BsuI.dna=Bio.Restriction.Restriction.BsuI-class.html#dna,Variable Bio.Restriction.Restriction.BsuRI.dna=Bio.Restriction.Restriction.BsuRI-class.html#dna,Variable Bio.Restriction.Restriction.BsuTUI.dna=Bio.Restriction.Restriction.BsuTUI-class.html#dna,Variable Bio.Restriction.Restriction.BtgI.dna=Bio.Restriction.Restriction.BtgI-class.html#dna,Variable Bio.Restriction.Restriction.BtgZI.dna=Bio.Restriction.Restriction.BtgZI-class.html#dna,Variable Bio.Restriction.Restriction.BthCI.dna=Bio.Restriction.Restriction.BthCI-class.html#dna,Variable Bio.Restriction.Restriction.BtrI.dna=Bio.Restriction.Restriction.BtrI-class.html#dna,Variable Bio.Restriction.Restriction.BtsCI.dna=Bio.Restriction.Restriction.BtsCI-class.html#dna,Variable Bio.Restriction.Restriction.BtsI.dna=Bio.Restriction.Restriction.BtsI-class.html#dna,Variable Bio.Restriction.Restriction.BtsIMutI.dna=Bio.Restriction.Restriction.BtsIMutI-class.html#dna,Variable Bio.Restriction.Restriction.BtuMI.dna=Bio.Restriction.Restriction.BtuMI-class.html#dna,Variable Bio.Restriction.Restriction.Bve1B23I.dna=Bio.Restriction.Restriction.Bve1B23I-class.html#dna,Variable Bio.Restriction.Restriction.BveI.dna=Bio.Restriction.Restriction.BveI-class.html#dna,Variable Bio.Restriction.Restriction.Cac8I.dna=Bio.Restriction.Restriction.Cac8I-class.html#dna,Variable Bio.Restriction.Restriction.CaiI.dna=Bio.Restriction.Restriction.CaiI-class.html#dna,Variable Bio.Restriction.Restriction.Cal14237I.dna=Bio.Restriction.Restriction.Cal14237I-class.html#dna,Variable Bio.Restriction.Restriction.CalB3II.dna=Bio.Restriction.Restriction.CalB3II-class.html#dna,Variable Bio.Restriction.Restriction.Cau10061II.dna=Bio.Restriction.Restriction.Cau10061II-class.html#dna,Variable Bio.Restriction.Restriction.CauII.dna=Bio.Restriction.Restriction.CauII-class.html#dna,Variable Bio.Restriction.Restriction.Cba13II.dna=Bio.Restriction.Restriction.Cba13II-class.html#dna,Variable Bio.Restriction.Restriction.Cba16038I.dna=Bio.Restriction.Restriction.Cba16038I-class.html#dna,Variable Bio.Restriction.Restriction.Cbo67071IV.dna=Bio.Restriction.Restriction.Cbo67071IV-class.html#dna,Variable Bio.Restriction.Restriction.Cch467III.dna=Bio.Restriction.Restriction.Cch467III-class.html#dna,Variable Bio.Restriction.Restriction.CchII.dna=Bio.Restriction.Restriction.CchII-class.html#dna,Variable Bio.Restriction.Restriction.CchIII.dna=Bio.Restriction.Restriction.CchIII-class.html#dna,Variable Bio.Restriction.Restriction.CciI.dna=Bio.Restriction.Restriction.CciI-class.html#dna,Variable Bio.Restriction.Restriction.CciNI.dna=Bio.Restriction.Restriction.CciNI-class.html#dna,Variable Bio.Restriction.Restriction.Cco14983V.dna=Bio.Restriction.Restriction.Cco14983V-class.html#dna,Variable Bio.Restriction.Restriction.Cco14983VI.dna=Bio.Restriction.Restriction.Cco14983VI-class.html#dna,Variable Bio.Restriction.Restriction.CcrNAIII.dna=Bio.Restriction.Restriction.CcrNAIII-class.html#dna,Variable Bio.Restriction.Restriction.Cdi11397I.dna=Bio.Restriction.Restriction.Cdi11397I-class.html#dna,Variable Bio.Restriction.Restriction.CdiI.dna=Bio.Restriction.Restriction.CdiI-class.html#dna,Variable Bio.Restriction.Restriction.CdpI.dna=Bio.Restriction.Restriction.CdpI-class.html#dna,Variable Bio.Restriction.Restriction.Cdu23823II.dna=Bio.Restriction.Restriction.Cdu23823II-class.html#dna,Variable Bio.Restriction.Restriction.CfoI.dna=Bio.Restriction.Restriction.CfoI-class.html#dna,Variable Bio.Restriction.Restriction.Cfr10I.dna=Bio.Restriction.Restriction.Cfr10I-class.html#dna,Variable Bio.Restriction.Restriction.Cfr13I.dna=Bio.Restriction.Restriction.Cfr13I-class.html#dna,Variable Bio.Restriction.Restriction.Cfr42I.dna=Bio.Restriction.Restriction.Cfr42I-class.html#dna,Variable Bio.Restriction.Restriction.Cfr9I.dna=Bio.Restriction.Restriction.Cfr9I-class.html#dna,Variable Bio.Restriction.Restriction.CfrI.dna=Bio.Restriction.Restriction.CfrI-class.html#dna,Variable Bio.Restriction.Restriction.CfrMH13II.dna=Bio.Restriction.Restriction.CfrMH13II-class.html#dna,Variable Bio.Restriction.Restriction.CfrMH16VI.dna=Bio.Restriction.Restriction.CfrMH16VI-class.html#dna,Variable Bio.Restriction.Restriction.Cfupf3II.dna=Bio.Restriction.Restriction.Cfupf3II-class.html#dna,Variable Bio.Restriction.Restriction.Cgl13032I.dna=Bio.Restriction.Restriction.Cgl13032I-class.html#dna,Variable Bio.Restriction.Restriction.Cgl13032II.dna=Bio.Restriction.Restriction.Cgl13032II-class.html#dna,Variable Bio.Restriction.Restriction.ChaI.dna=Bio.Restriction.Restriction.ChaI-class.html#dna,Variable Bio.Restriction.Restriction.Cje265V.dna=Bio.Restriction.Restriction.Cje265V-class.html#dna,Variable Bio.Restriction.Restriction.Cje54107III.dna=Bio.Restriction.Restriction.Cje54107III-class.html#dna,Variable Bio.Restriction.Restriction.CjeFIII.dna=Bio.Restriction.Restriction.CjeFIII-class.html#dna,Variable Bio.Restriction.Restriction.CjeFV.dna=Bio.Restriction.Restriction.CjeFV-class.html#dna,Variable Bio.Restriction.Restriction.CjeI.dna=Bio.Restriction.Restriction.CjeI-class.html#dna,Variable Bio.Restriction.Restriction.CjeNII.dna=Bio.Restriction.Restriction.CjeNII-class.html#dna,Variable Bio.Restriction.Restriction.CjeNIII.dna=Bio.Restriction.Restriction.CjeNIII-class.html#dna,Variable Bio.Restriction.Restriction.CjeNV.dna=Bio.Restriction.Restriction.CjeNV-class.html#dna,Variable Bio.Restriction.Restriction.CjeP659IV.dna=Bio.Restriction.Restriction.CjeP659IV-class.html#dna,Variable Bio.Restriction.Restriction.CjePI.dna=Bio.Restriction.Restriction.CjePI-class.html#dna,Variable Bio.Restriction.Restriction.CjuI.dna=Bio.Restriction.Restriction.CjuI-class.html#dna,Variable Bio.Restriction.Restriction.CjuII.dna=Bio.Restriction.Restriction.CjuII-class.html#dna,Variable Bio.Restriction.Restriction.Cla11845III.dna=Bio.Restriction.Restriction.Cla11845III-class.html#dna,Variable Bio.Restriction.Restriction.ClaI.dna=Bio.Restriction.Restriction.ClaI-class.html#dna,Variable Bio.Restriction.Restriction.Cly7489II.dna=Bio.Restriction.Restriction.Cly7489II-class.html#dna,Variable Bio.Restriction.Restriction.Cma23826I.dna=Bio.Restriction.Restriction.Cma23826I-class.html#dna,Variable Bio.Restriction.Restriction.CpoI.dna=Bio.Restriction.Restriction.CpoI-class.html#dna,Variable Bio.Restriction.Restriction.CseI.dna=Bio.Restriction.Restriction.CseI-class.html#dna,Variable Bio.Restriction.Restriction.CsiI.dna=Bio.Restriction.Restriction.CsiI-class.html#dna,Variable Bio.Restriction.Restriction.Csp2014I.dna=Bio.Restriction.Restriction.Csp2014I-class.html#dna,Variable Bio.Restriction.Restriction.Csp6I.dna=Bio.Restriction.Restriction.Csp6I-class.html#dna,Variable Bio.Restriction.Restriction.CspAI.dna=Bio.Restriction.Restriction.CspAI-class.html#dna,Variable Bio.Restriction.Restriction.CspCI.dna=Bio.Restriction.Restriction.CspCI-class.html#dna,Variable Bio.Restriction.Restriction.CspI.dna=Bio.Restriction.Restriction.CspI-class.html#dna,Variable Bio.Restriction.Restriction.CstMI.dna=Bio.Restriction.Restriction.CstMI-class.html#dna,Variable Bio.Restriction.Restriction.CviAII.dna=Bio.Restriction.Restriction.CviAII-class.html#dna,Variable Bio.Restriction.Restriction.CviJI.dna=Bio.Restriction.Restriction.CviJI-class.html#dna,Variable Bio.Restriction.Restriction.CviKI_1.dna=Bio.Restriction.Restriction.CviKI_1-class.html#dna,Variable Bio.Restriction.Restriction.CviQI.dna=Bio.Restriction.Restriction.CviQI-class.html#dna,Variable Bio.Restriction.Restriction.CviRI.dna=Bio.Restriction.Restriction.CviRI-class.html#dna,Variable Bio.Restriction.Restriction.Dde51507I.dna=Bio.Restriction.Restriction.Dde51507I-class.html#dna,Variable Bio.Restriction.Restriction.DdeI.dna=Bio.Restriction.Restriction.DdeI-class.html#dna,Variable Bio.Restriction.Restriction.DinI.dna=Bio.Restriction.Restriction.DinI-class.html#dna,Variable Bio.Restriction.Restriction.DpnI.dna=Bio.Restriction.Restriction.DpnI-class.html#dna,Variable Bio.Restriction.Restriction.DpnII.dna=Bio.Restriction.Restriction.DpnII-class.html#dna,Variable Bio.Restriction.Restriction.DraI.dna=Bio.Restriction.Restriction.DraI-class.html#dna,Variable Bio.Restriction.Restriction.DraII.dna=Bio.Restriction.Restriction.DraII-class.html#dna,Variable Bio.Restriction.Restriction.DraIII.dna=Bio.Restriction.Restriction.DraIII-class.html#dna,Variable Bio.Restriction.Restriction.DraRI.dna=Bio.Restriction.Restriction.DraRI-class.html#dna,Variable Bio.Restriction.Restriction.DrdI.dna=Bio.Restriction.Restriction.DrdI-class.html#dna,Variable Bio.Restriction.Restriction.DrdII.dna=Bio.Restriction.Restriction.DrdII-class.html#dna,Variable Bio.Restriction.Restriction.DrdIV.dna=Bio.Restriction.Restriction.DrdIV-class.html#dna,Variable Bio.Restriction.Restriction.DriI.dna=Bio.Restriction.Restriction.DriI-class.html#dna,Variable Bio.Restriction.Restriction.DsaI.dna=Bio.Restriction.Restriction.DsaI-class.html#dna,Variable Bio.Restriction.Restriction.DseDI.dna=Bio.Restriction.Restriction.DseDI-class.html#dna,Variable Bio.Restriction.Restriction.DvuIII.dna=Bio.Restriction.Restriction.DvuIII-class.html#dna,Variable Bio.Restriction.Restriction.EaeI.dna=Bio.Restriction.Restriction.EaeI-class.html#dna,Variable Bio.Restriction.Restriction.EagI.dna=Bio.Restriction.Restriction.EagI-class.html#dna,Variable Bio.Restriction.Restriction.Eam1104I.dna=Bio.Restriction.Restriction.Eam1104I-class.html#dna,Variable Bio.Restriction.Restriction.Eam1105I.dna=Bio.Restriction.Restriction.Eam1105I-class.html#dna,Variable Bio.Restriction.Restriction.EarI.dna=Bio.Restriction.Restriction.EarI-class.html#dna,Variable Bio.Restriction.Restriction.EciI.dna=Bio.Restriction.Restriction.EciI-class.html#dna,Variable Bio.Restriction.Restriction.Ecl136II.dna=Bio.Restriction.Restriction.Ecl136II-class.html#dna,Variable Bio.Restriction.Restriction.Ecl234I.dna=Bio.Restriction.Restriction.Ecl234I-class.html#dna,Variable Bio.Restriction.Restriction.Ecl35734I.dna=Bio.Restriction.Restriction.Ecl35734I-class.html#dna,Variable Bio.Restriction.Restriction.EclXI.dna=Bio.Restriction.Restriction.EclXI-class.html#dna,Variable Bio.Restriction.Restriction.Eco105I.dna=Bio.Restriction.Restriction.Eco105I-class.html#dna,Variable Bio.Restriction.Restriction.Eco130I.dna=Bio.Restriction.Restriction.Eco130I-class.html#dna,Variable Bio.Restriction.Restriction.Eco147I.dna=Bio.Restriction.Restriction.Eco147I-class.html#dna,Variable Bio.Restriction.Restriction.Eco24I.dna=Bio.Restriction.Restriction.Eco24I-class.html#dna,Variable Bio.Restriction.Restriction.Eco31I.dna=Bio.Restriction.Restriction.Eco31I-class.html#dna,Variable Bio.Restriction.Restriction.Eco32I.dna=Bio.Restriction.Restriction.Eco32I-class.html#dna,Variable Bio.Restriction.Restriction.Eco43896II.dna=Bio.Restriction.Restriction.Eco43896II-class.html#dna,Variable Bio.Restriction.Restriction.Eco4465II.dna=Bio.Restriction.Restriction.Eco4465II-class.html#dna,Variable Bio.Restriction.Restriction.Eco47I.dna=Bio.Restriction.Restriction.Eco47I-class.html#dna,Variable Bio.Restriction.Restriction.Eco47III.dna=Bio.Restriction.Restriction.Eco47III-class.html#dna,Variable Bio.Restriction.Restriction.Eco52I.dna=Bio.Restriction.Restriction.Eco52I-class.html#dna,Variable Bio.Restriction.Restriction.Eco53kI.dna=Bio.Restriction.Restriction.Eco53kI-class.html#dna,Variable Bio.Restriction.Restriction.Eco57I.dna=Bio.Restriction.Restriction.Eco57I-class.html#dna,Variable Bio.Restriction.Restriction.Eco57MI.dna=Bio.Restriction.Restriction.Eco57MI-class.html#dna,Variable Bio.Restriction.Restriction.Eco72I.dna=Bio.Restriction.Restriction.Eco72I-class.html#dna,Variable Bio.Restriction.Restriction.Eco81I.dna=Bio.Restriction.Restriction.Eco81I-class.html#dna,Variable Bio.Restriction.Restriction.Eco88I.dna=Bio.Restriction.Restriction.Eco88I-class.html#dna,Variable Bio.Restriction.Restriction.Eco91I.dna=Bio.Restriction.Restriction.Eco91I-class.html#dna,Variable Bio.Restriction.Restriction.EcoBLMcrX.dna=Bio.Restriction.Restriction.EcoBLMcrX-class.html#dna,Variable Bio.Restriction.Restriction.EcoE1140I.dna=Bio.Restriction.Restriction.EcoE1140I-class.html#dna,Variable Bio.Restriction.Restriction.EcoHI.dna=Bio.Restriction.Restriction.EcoHI-class.html#dna,Variable Bio.Restriction.Restriction.EcoHSI.dna=Bio.Restriction.Restriction.EcoHSI-class.html#dna,Variable Bio.Restriction.Restriction.EcoICRI.dna=Bio.Restriction.Restriction.EcoICRI-class.html#dna,Variable Bio.Restriction.Restriction.EcoMVII.dna=Bio.Restriction.Restriction.EcoMVII-class.html#dna,Variable Bio.Restriction.Restriction.EcoNI.dna=Bio.Restriction.Restriction.EcoNI-class.html#dna,Variable Bio.Restriction.Restriction.EcoNIH6II.dna=Bio.Restriction.Restriction.EcoNIH6II-class.html#dna,Variable Bio.Restriction.Restriction.EcoO109I.dna=Bio.Restriction.Restriction.EcoO109I-class.html#dna,Variable Bio.Restriction.Restriction.EcoO65I.dna=Bio.Restriction.Restriction.EcoO65I-class.html#dna,Variable Bio.Restriction.Restriction.EcoRI.dna=Bio.Restriction.Restriction.EcoRI-class.html#dna,Variable Bio.Restriction.Restriction.EcoRII.dna=Bio.Restriction.Restriction.EcoRII-class.html#dna,Variable Bio.Restriction.Restriction.EcoRV.dna=Bio.Restriction.Restriction.EcoRV-class.html#dna,Variable Bio.Restriction.Restriction.EcoT14I.dna=Bio.Restriction.Restriction.EcoT14I-class.html#dna,Variable Bio.Restriction.Restriction.EcoT22I.dna=Bio.Restriction.Restriction.EcoT22I-class.html#dna,Variable Bio.Restriction.Restriction.EcoT38I.dna=Bio.Restriction.Restriction.EcoT38I-class.html#dna,Variable Bio.Restriction.Restriction.EgeI.dna=Bio.Restriction.Restriction.EgeI-class.html#dna,Variable Bio.Restriction.Restriction.EheI.dna=Bio.Restriction.Restriction.EheI-class.html#dna,Variable Bio.Restriction.Restriction.Eli8509II.dna=Bio.Restriction.Restriction.Eli8509II-class.html#dna,Variable Bio.Restriction.Restriction.ErhI.dna=Bio.Restriction.Restriction.ErhI-class.html#dna,Variable Bio.Restriction.Restriction.EsaBC3I.dna=Bio.Restriction.Restriction.EsaBC3I-class.html#dna,Variable Bio.Restriction.Restriction.EsaSSI.dna=Bio.Restriction.Restriction.EsaSSI-class.html#dna,Variable Bio.Restriction.Restriction.Esp3007I.dna=Bio.Restriction.Restriction.Esp3007I-class.html#dna,Variable Bio.Restriction.Restriction.Esp3I.dna=Bio.Restriction.Restriction.Esp3I-class.html#dna,Variable Bio.Restriction.Restriction.EspI.dna=Bio.Restriction.Restriction.EspI-class.html#dna,Variable Bio.Restriction.Restriction.FaeI.dna=Bio.Restriction.Restriction.FaeI-class.html#dna,Variable Bio.Restriction.Restriction.FaiI.dna=Bio.Restriction.Restriction.FaiI-class.html#dna,Variable Bio.Restriction.Restriction.FalI.dna=Bio.Restriction.Restriction.FalI-class.html#dna,Variable Bio.Restriction.Restriction.FaqI.dna=Bio.Restriction.Restriction.FaqI-class.html#dna,Variable Bio.Restriction.Restriction.FatI.dna=Bio.Restriction.Restriction.FatI-class.html#dna,Variable Bio.Restriction.Restriction.FauI.dna=Bio.Restriction.Restriction.FauI-class.html#dna,Variable Bio.Restriction.Restriction.FauNDI.dna=Bio.Restriction.Restriction.FauNDI-class.html#dna,Variable Bio.Restriction.Restriction.FbaI.dna=Bio.Restriction.Restriction.FbaI-class.html#dna,Variable Bio.Restriction.Restriction.FblI.dna=Bio.Restriction.Restriction.FblI-class.html#dna,Variable Bio.Restriction.Restriction.Fco1691IV.dna=Bio.Restriction.Restriction.Fco1691IV-class.html#dna,Variable Bio.Restriction.Restriction.FinI.dna=Bio.Restriction.Restriction.FinI-class.html#dna,Variable Bio.Restriction.Restriction.FmuI.dna=Bio.Restriction.Restriction.FmuI-class.html#dna,Variable Bio.Restriction.Restriction.Fnu4HI.dna=Bio.Restriction.Restriction.Fnu4HI-class.html#dna,Variable Bio.Restriction.Restriction.FnuDII.dna=Bio.Restriction.Restriction.FnuDII-class.html#dna,Variable Bio.Restriction.Restriction.FokI.dna=Bio.Restriction.Restriction.FokI-class.html#dna,Variable Bio.Restriction.Restriction.FriOI.dna=Bio.Restriction.Restriction.FriOI-class.html#dna,Variable Bio.Restriction.Restriction.FseI.dna=Bio.Restriction.Restriction.FseI-class.html#dna,Variable Bio.Restriction.Restriction.Fsp4HI.dna=Bio.Restriction.Restriction.Fsp4HI-class.html#dna,Variable Bio.Restriction.Restriction.FspAI.dna=Bio.Restriction.Restriction.FspAI-class.html#dna,Variable Bio.Restriction.Restriction.FspBI.dna=Bio.Restriction.Restriction.FspBI-class.html#dna,Variable Bio.Restriction.Restriction.FspEI.dna=Bio.Restriction.Restriction.FspEI-class.html#dna,Variable Bio.Restriction.Restriction.FspI.dna=Bio.Restriction.Restriction.FspI-class.html#dna,Variable Bio.Restriction.Restriction.FspPK15I.dna=Bio.Restriction.Restriction.FspPK15I-class.html#dna,Variable Bio.Restriction.Restriction.FtnUV.dna=Bio.Restriction.Restriction.FtnUV-class.html#dna,Variable Bio.Restriction.Restriction.GauT27I.dna=Bio.Restriction.Restriction.GauT27I-class.html#dna,Variable Bio.Restriction.Restriction.Gba708II.dna=Bio.Restriction.Restriction.Gba708II-class.html#dna,Variable Bio.Restriction.Restriction.GdiII.dna=Bio.Restriction.Restriction.GdiII-class.html#dna,Variable Bio.Restriction.Restriction.GlaI.dna=Bio.Restriction.Restriction.GlaI-class.html#dna,Variable Bio.Restriction.Restriction.GluI.dna=Bio.Restriction.Restriction.GluI-class.html#dna,Variable Bio.Restriction.Restriction.GsaI.dna=Bio.Restriction.Restriction.GsaI-class.html#dna,Variable Bio.Restriction.Restriction.GsuI.dna=Bio.Restriction.Restriction.GsuI-class.html#dna,Variable Bio.Restriction.Restriction.HaeI.dna=Bio.Restriction.Restriction.HaeI-class.html#dna,Variable Bio.Restriction.Restriction.HaeII.dna=Bio.Restriction.Restriction.HaeII-class.html#dna,Variable Bio.Restriction.Restriction.HaeIII.dna=Bio.Restriction.Restriction.HaeIII-class.html#dna,Variable Bio.Restriction.Restriction.HapII.dna=Bio.Restriction.Restriction.HapII-class.html#dna,Variable Bio.Restriction.Restriction.HauII.dna=Bio.Restriction.Restriction.HauII-class.html#dna,Variable Bio.Restriction.Restriction.HbaII.dna=Bio.Restriction.Restriction.HbaII-class.html#dna,Variable Bio.Restriction.Restriction.HdeNY26I.dna=Bio.Restriction.Restriction.HdeNY26I-class.html#dna,Variable Bio.Restriction.Restriction.HdeZA17I.dna=Bio.Restriction.Restriction.HdeZA17I-class.html#dna,Variable Bio.Restriction.Restriction.HgaI.dna=Bio.Restriction.Restriction.HgaI-class.html#dna,Variable Bio.Restriction.Restriction.HgiAI.dna=Bio.Restriction.Restriction.HgiAI-class.html#dna,Variable Bio.Restriction.Restriction.HgiCI.dna=Bio.Restriction.Restriction.HgiCI-class.html#dna,Variable Bio.Restriction.Restriction.HgiEII.dna=Bio.Restriction.Restriction.HgiEII-class.html#dna,Variable Bio.Restriction.Restriction.HgiJII.dna=Bio.Restriction.Restriction.HgiJII-class.html#dna,Variable Bio.Restriction.Restriction.HhaI.dna=Bio.Restriction.Restriction.HhaI-class.html#dna,Variable Bio.Restriction.Restriction.Hin1I.dna=Bio.Restriction.Restriction.Hin1I-class.html#dna,Variable Bio.Restriction.Restriction.Hin1II.dna=Bio.Restriction.Restriction.Hin1II-class.html#dna,Variable Bio.Restriction.Restriction.Hin4I.dna=Bio.Restriction.Restriction.Hin4I-class.html#dna,Variable Bio.Restriction.Restriction.Hin4II.dna=Bio.Restriction.Restriction.Hin4II-class.html#dna,Variable Bio.Restriction.Restriction.Hin6I.dna=Bio.Restriction.Restriction.Hin6I-class.html#dna,Variable Bio.Restriction.Restriction.HinP1I.dna=Bio.Restriction.Restriction.HinP1I-class.html#dna,Variable Bio.Restriction.Restriction.HincII.dna=Bio.Restriction.Restriction.HincII-class.html#dna,Variable Bio.Restriction.Restriction.HindII.dna=Bio.Restriction.Restriction.HindII-class.html#dna,Variable Bio.Restriction.Restriction.HindIII.dna=Bio.Restriction.Restriction.HindIII-class.html#dna,Variable Bio.Restriction.Restriction.HinfI.dna=Bio.Restriction.Restriction.HinfI-class.html#dna,Variable Bio.Restriction.Restriction.HpaI.dna=Bio.Restriction.Restriction.HpaI-class.html#dna,Variable Bio.Restriction.Restriction.HpaII.dna=Bio.Restriction.Restriction.HpaII-class.html#dna,Variable Bio.Restriction.Restriction.HphI.dna=Bio.Restriction.Restriction.HphI-class.html#dna,Variable Bio.Restriction.Restriction.Hpy166II.dna=Bio.Restriction.Restriction.Hpy166II-class.html#dna,Variable Bio.Restriction.Restriction.Hpy178III.dna=Bio.Restriction.Restriction.Hpy178III-class.html#dna,Variable Bio.Restriction.Restriction.Hpy188I.dna=Bio.Restriction.Restriction.Hpy188I-class.html#dna,Variable Bio.Restriction.Restriction.Hpy188III.dna=Bio.Restriction.Restriction.Hpy188III-class.html#dna,Variable Bio.Restriction.Restriction.Hpy300XI.dna=Bio.Restriction.Restriction.Hpy300XI-class.html#dna,Variable Bio.Restriction.Restriction.Hpy8I.dna=Bio.Restriction.Restriction.Hpy8I-class.html#dna,Variable Bio.Restriction.Restriction.Hpy99I.dna=Bio.Restriction.Restriction.Hpy99I-class.html#dna,Variable Bio.Restriction.Restriction.Hpy99XIII.dna=Bio.Restriction.Restriction.Hpy99XIII-class.html#dna,Variable Bio.Restriction.Restriction.Hpy99XIV.dna=Bio.Restriction.Restriction.Hpy99XIV-class.html#dna,Variable Bio.Restriction.Restriction.Hpy99XIV_mut1.dna=Bio.Restriction.Restriction.Hpy99XIV_mut1-class.html#dna,Variable Bio.Restriction.Restriction.Hpy99XXII.dna=Bio.Restriction.Restriction.Hpy99XXII-class.html#dna,Variable Bio.Restriction.Restriction.HpyAV.dna=Bio.Restriction.Restriction.HpyAV-class.html#dna,Variable Bio.Restriction.Restriction.HpyAXIV.dna=Bio.Restriction.Restriction.HpyAXIV-class.html#dna,Variable Bio.Restriction.Restriction.HpyAXVIII.dna=Bio.Restriction.Restriction.HpyAXVIII-class.html#dna,Variable Bio.Restriction.Restriction.HpyAXVI_mut1.dna=Bio.Restriction.Restriction.HpyAXVI_mut1-class.html#dna,Variable Bio.Restriction.Restriction.HpyAXVI_mut2.dna=Bio.Restriction.Restriction.HpyAXVI_mut2-class.html#dna,Variable Bio.Restriction.Restriction.HpyCH4III.dna=Bio.Restriction.Restriction.HpyCH4III-class.html#dna,Variable Bio.Restriction.Restriction.HpyCH4IV.dna=Bio.Restriction.Restriction.HpyCH4IV-class.html#dna,Variable Bio.Restriction.Restriction.HpyCH4V.dna=Bio.Restriction.Restriction.HpyCH4V-class.html#dna,Variable Bio.Restriction.Restriction.HpyF10VI.dna=Bio.Restriction.Restriction.HpyF10VI-class.html#dna,Variable Bio.Restriction.Restriction.HpyF3I.dna=Bio.Restriction.Restriction.HpyF3I-class.html#dna,Variable Bio.Restriction.Restriction.HpySE526I.dna=Bio.Restriction.Restriction.HpySE526I-class.html#dna,Variable Bio.Restriction.Restriction.HpyUM032XIII.dna=Bio.Restriction.Restriction.HpyUM032XIII-class.html#dna,Variable Bio.Restriction.Restriction.HpyUM032XIII_mut1.dna=Bio.Restriction.Restriction.HpyUM032XIII_mut1-class.html#dna,Variable Bio.Restriction.Restriction.HpyUM032XIV.dna=Bio.Restriction.Restriction.HpyUM032XIV-class.html#dna,Variable Bio.Restriction.Restriction.HpyUM037X.dna=Bio.Restriction.Restriction.HpyUM037X-class.html#dna,Variable Bio.Restriction.Restriction.Hsp92I.dna=Bio.Restriction.Restriction.Hsp92I-class.html#dna,Variable Bio.Restriction.Restriction.Hsp92II.dna=Bio.Restriction.Restriction.Hsp92II-class.html#dna,Variable Bio.Restriction.Restriction.HspAI.dna=Bio.Restriction.Restriction.HspAI-class.html#dna,Variable Bio.Restriction.Restriction.HspMHR1II.dna=Bio.Restriction.Restriction.HspMHR1II-class.html#dna,Variable Bio.Restriction.Restriction.Jma19592I.dna=Bio.Restriction.Restriction.Jma19592I-class.html#dna,Variable Bio.Restriction.Restriction.Jma19592II.dna=Bio.Restriction.Restriction.Jma19592II-class.html#dna,Variable Bio.Restriction.Restriction.Jsp2502II.dna=Bio.Restriction.Restriction.Jsp2502II-class.html#dna,Variable Bio.Restriction.Restriction.KasI.dna=Bio.Restriction.Restriction.KasI-class.html#dna,Variable Bio.Restriction.Restriction.KflI.dna=Bio.Restriction.Restriction.KflI-class.html#dna,Variable Bio.Restriction.Restriction.Kor51II.dna=Bio.Restriction.Restriction.Kor51II-class.html#dna,Variable Bio.Restriction.Restriction.Kpn156V.dna=Bio.Restriction.Restriction.Kpn156V-class.html#dna,Variable Bio.Restriction.Restriction.Kpn2I.dna=Bio.Restriction.Restriction.Kpn2I-class.html#dna,Variable Bio.Restriction.Restriction.Kpn327I.dna=Bio.Restriction.Restriction.Kpn327I-class.html#dna,Variable Bio.Restriction.Restriction.KpnI.dna=Bio.Restriction.Restriction.KpnI-class.html#dna,Variable Bio.Restriction.Restriction.KpnNH25III.dna=Bio.Restriction.Restriction.KpnNH25III-class.html#dna,Variable Bio.Restriction.Restriction.KpnNIH30III.dna=Bio.Restriction.Restriction.KpnNIH30III-class.html#dna,Variable Bio.Restriction.Restriction.KpnNIH50I.dna=Bio.Restriction.Restriction.KpnNIH50I-class.html#dna,Variable Bio.Restriction.Restriction.KroI.dna=Bio.Restriction.Restriction.KroI-class.html#dna,Variable Bio.Restriction.Restriction.Ksp22I.dna=Bio.Restriction.Restriction.Ksp22I-class.html#dna,Variable Bio.Restriction.Restriction.Ksp632I.dna=Bio.Restriction.Restriction.Ksp632I-class.html#dna,Variable Bio.Restriction.Restriction.KspAI.dna=Bio.Restriction.Restriction.KspAI-class.html#dna,Variable Bio.Restriction.Restriction.KspI.dna=Bio.Restriction.Restriction.KspI-class.html#dna,Variable Bio.Restriction.Restriction.Kzo9I.dna=Bio.Restriction.Restriction.Kzo9I-class.html#dna,Variable Bio.Restriction.Restriction.Lba2029III.dna=Bio.Restriction.Restriction.Lba2029III-class.html#dna,Variable Bio.Restriction.Restriction.Lde4408II.dna=Bio.Restriction.Restriction.Lde4408II-class.html#dna,Variable Bio.Restriction.Restriction.LguI.dna=Bio.Restriction.Restriction.LguI-class.html#dna,Variable Bio.Restriction.Restriction.LlaG50I.dna=Bio.Restriction.Restriction.LlaG50I-class.html#dna,Variable Bio.Restriction.Restriction.LmnI.dna=Bio.Restriction.Restriction.LmnI-class.html#dna,Variable Bio.Restriction.Restriction.Lmo370I.dna=Bio.Restriction.Restriction.Lmo370I-class.html#dna,Variable Bio.Restriction.Restriction.Lmo911II.dna=Bio.Restriction.Restriction.Lmo911II-class.html#dna,Variable Bio.Restriction.Restriction.Lpl1004II.dna=Bio.Restriction.Restriction.Lpl1004II-class.html#dna,Variable Bio.Restriction.Restriction.LpnI.dna=Bio.Restriction.Restriction.LpnI-class.html#dna,Variable Bio.Restriction.Restriction.LpnPI.dna=Bio.Restriction.Restriction.LpnPI-class.html#dna,Variable Bio.Restriction.Restriction.Lra68I.dna=Bio.Restriction.Restriction.Lra68I-class.html#dna,Variable Bio.Restriction.Restriction.LsaDS4I.dna=Bio.Restriction.Restriction.LsaDS4I-class.html#dna,Variable Bio.Restriction.Restriction.Lsp1109I.dna=Bio.Restriction.Restriction.Lsp1109I-class.html#dna,Variable Bio.Restriction.Restriction.Lsp48III.dna=Bio.Restriction.Restriction.Lsp48III-class.html#dna,Variable Bio.Restriction.Restriction.Lsp6406VI.dna=Bio.Restriction.Restriction.Lsp6406VI-class.html#dna,Variable Bio.Restriction.Restriction.LweI.dna=Bio.Restriction.Restriction.LweI-class.html#dna,Variable Bio.Restriction.Restriction.MabI.dna=Bio.Restriction.Restriction.MabI-class.html#dna,Variable Bio.Restriction.Restriction.MaeI.dna=Bio.Restriction.Restriction.MaeI-class.html#dna,Variable Bio.Restriction.Restriction.MaeII.dna=Bio.Restriction.Restriction.MaeII-class.html#dna,Variable Bio.Restriction.Restriction.MaeIII.dna=Bio.Restriction.Restriction.MaeIII-class.html#dna,Variable Bio.Restriction.Restriction.MalI.dna=Bio.Restriction.Restriction.MalI-class.html#dna,Variable Bio.Restriction.Restriction.MaqI.dna=Bio.Restriction.Restriction.MaqI-class.html#dna,Variable Bio.Restriction.Restriction.MauBI.dna=Bio.Restriction.Restriction.MauBI-class.html#dna,Variable Bio.Restriction.Restriction.Mba11I.dna=Bio.Restriction.Restriction.Mba11I-class.html#dna,Variable Bio.Restriction.Restriction.MbiI.dna=Bio.Restriction.Restriction.MbiI-class.html#dna,Variable Bio.Restriction.Restriction.MboI.dna=Bio.Restriction.Restriction.MboI-class.html#dna,Variable Bio.Restriction.Restriction.MboII.dna=Bio.Restriction.Restriction.MboII-class.html#dna,Variable Bio.Restriction.Restriction.McaTI.dna=Bio.Restriction.Restriction.McaTI-class.html#dna,Variable Bio.Restriction.Restriction.Mcr10I.dna=Bio.Restriction.Restriction.Mcr10I-class.html#dna,Variable Bio.Restriction.Restriction.McrI.dna=Bio.Restriction.Restriction.McrI-class.html#dna,Variable Bio.Restriction.Restriction.MfeI.dna=Bio.Restriction.Restriction.MfeI-class.html#dna,Variable Bio.Restriction.Restriction.MflI.dna=Bio.Restriction.Restriction.MflI-class.html#dna,Variable Bio.Restriction.Restriction.MhlI.dna=Bio.Restriction.Restriction.MhlI-class.html#dna,Variable Bio.Restriction.Restriction.MjaIV.dna=Bio.Restriction.Restriction.MjaIV-class.html#dna,Variable Bio.Restriction.Restriction.MkaDII.dna=Bio.Restriction.Restriction.MkaDII-class.html#dna,Variable Bio.Restriction.Restriction.MlsI.dna=Bio.Restriction.Restriction.MlsI-class.html#dna,Variable Bio.Restriction.Restriction.Mlu211III.dna=Bio.Restriction.Restriction.Mlu211III-class.html#dna,Variable Bio.Restriction.Restriction.MluCI.dna=Bio.Restriction.Restriction.MluCI-class.html#dna,Variable Bio.Restriction.Restriction.MluI.dna=Bio.Restriction.Restriction.MluI-class.html#dna,Variable Bio.Restriction.Restriction.MluNI.dna=Bio.Restriction.Restriction.MluNI-class.html#dna,Variable Bio.Restriction.Restriction.Mly113I.dna=Bio.Restriction.Restriction.Mly113I-class.html#dna,Variable Bio.Restriction.Restriction.MlyI.dna=Bio.Restriction.Restriction.MlyI-class.html#dna,Variable Bio.Restriction.Restriction.MmeI.dna=Bio.Restriction.Restriction.MmeI-class.html#dna,Variable Bio.Restriction.Restriction.MnlI.dna=Bio.Restriction.Restriction.MnlI-class.html#dna,Variable Bio.Restriction.Restriction.Mox20I.dna=Bio.Restriction.Restriction.Mox20I-class.html#dna,Variable Bio.Restriction.Restriction.Mph1103I.dna=Bio.Restriction.Restriction.Mph1103I-class.html#dna,Variable Bio.Restriction.Restriction.MreI.dna=Bio.Restriction.Restriction.MreI-class.html#dna,Variable Bio.Restriction.Restriction.MroI.dna=Bio.Restriction.Restriction.MroI-class.html#dna,Variable Bio.Restriction.Restriction.MroNI.dna=Bio.Restriction.Restriction.MroNI-class.html#dna,Variable Bio.Restriction.Restriction.MroXI.dna=Bio.Restriction.Restriction.MroXI-class.html#dna,Variable Bio.Restriction.Restriction.MscI.dna=Bio.Restriction.Restriction.MscI-class.html#dna,Variable Bio.Restriction.Restriction.MseI.dna=Bio.Restriction.Restriction.MseI-class.html#dna,Variable Bio.Restriction.Restriction.MslI.dna=Bio.Restriction.Restriction.MslI-class.html#dna,Variable Bio.Restriction.Restriction.Msp20I.dna=Bio.Restriction.Restriction.Msp20I-class.html#dna,Variable Bio.Restriction.Restriction.MspA1I.dna=Bio.Restriction.Restriction.MspA1I-class.html#dna,Variable Bio.Restriction.Restriction.MspCI.dna=Bio.Restriction.Restriction.MspCI-class.html#dna,Variable Bio.Restriction.Restriction.MspGI.dna=Bio.Restriction.Restriction.MspGI-class.html#dna,Variable Bio.Restriction.Restriction.MspI.dna=Bio.Restriction.Restriction.MspI-class.html#dna,Variable Bio.Restriction.Restriction.MspI7II.dna=Bio.Restriction.Restriction.MspI7II-class.html#dna,Variable Bio.Restriction.Restriction.MspI7IV.dna=Bio.Restriction.Restriction.MspI7IV-class.html#dna,Variable Bio.Restriction.Restriction.MspJI.dna=Bio.Restriction.Restriction.MspJI-class.html#dna,Variable Bio.Restriction.Restriction.MspR9I.dna=Bio.Restriction.Restriction.MspR9I-class.html#dna,Variable Bio.Restriction.Restriction.MspSC27II.dna=Bio.Restriction.Restriction.MspSC27II-class.html#dna,Variable Bio.Restriction.Restriction.MssI.dna=Bio.Restriction.Restriction.MssI-class.html#dna,Variable Bio.Restriction.Restriction.MstI.dna=Bio.Restriction.Restriction.MstI-class.html#dna,Variable Bio.Restriction.Restriction.MteI.dna=Bio.Restriction.Restriction.MteI-class.html#dna,Variable Bio.Restriction.Restriction.MtuHN878II.dna=Bio.Restriction.Restriction.MtuHN878II-class.html#dna,Variable Bio.Restriction.Restriction.MunI.dna=Bio.Restriction.Restriction.MunI-class.html#dna,Variable Bio.Restriction.Restriction.Mva1269I.dna=Bio.Restriction.Restriction.Mva1269I-class.html#dna,Variable Bio.Restriction.Restriction.MvaI.dna=Bio.Restriction.Restriction.MvaI-class.html#dna,Variable Bio.Restriction.Restriction.MvnI.dna=Bio.Restriction.Restriction.MvnI-class.html#dna,Variable Bio.Restriction.Restriction.MwoI.dna=Bio.Restriction.Restriction.MwoI-class.html#dna,Variable Bio.Restriction.Restriction.NaeI.dna=Bio.Restriction.Restriction.NaeI-class.html#dna,Variable Bio.Restriction.Restriction.Nal45188II.dna=Bio.Restriction.Restriction.Nal45188II-class.html#dna,Variable Bio.Restriction.Restriction.NarI.dna=Bio.Restriction.Restriction.NarI-class.html#dna,Variable Bio.Restriction.Restriction.Nbr128II.dna=Bio.Restriction.Restriction.Nbr128II-class.html#dna,Variable Bio.Restriction.Restriction.NciI.dna=Bio.Restriction.Restriction.NciI-class.html#dna,Variable Bio.Restriction.Restriction.NcoI.dna=Bio.Restriction.Restriction.NcoI-class.html#dna,Variable Bio.Restriction.Restriction.NdeI.dna=Bio.Restriction.Restriction.NdeI-class.html#dna,Variable Bio.Restriction.Restriction.NdeII.dna=Bio.Restriction.Restriction.NdeII-class.html#dna,Variable Bio.Restriction.Restriction.NgoAVII.dna=Bio.Restriction.Restriction.NgoAVII-class.html#dna,Variable Bio.Restriction.Restriction.NgoAVIII.dna=Bio.Restriction.Restriction.NgoAVIII-class.html#dna,Variable Bio.Restriction.Restriction.NgoMIV.dna=Bio.Restriction.Restriction.NgoMIV-class.html#dna,Variable Bio.Restriction.Restriction.NhaXI.dna=Bio.Restriction.Restriction.NhaXI-class.html#dna,Variable Bio.Restriction.Restriction.NheI.dna=Bio.Restriction.Restriction.NheI-class.html#dna,Variable Bio.Restriction.Restriction.NhoI.dna=Bio.Restriction.Restriction.NhoI-class.html#dna,Variable Bio.Restriction.Restriction.NlaCI.dna=Bio.Restriction.Restriction.NlaCI-class.html#dna,Variable Bio.Restriction.Restriction.NlaIII.dna=Bio.Restriction.Restriction.NlaIII-class.html#dna,Variable Bio.Restriction.Restriction.NlaIV.dna=Bio.Restriction.Restriction.NlaIV-class.html#dna,Variable Bio.Restriction.Restriction.Nli3877I.dna=Bio.Restriction.Restriction.Nli3877I-class.html#dna,Variable Bio.Restriction.Restriction.NmeA6CIII.dna=Bio.Restriction.Restriction.NmeA6CIII-class.html#dna,Variable Bio.Restriction.Restriction.NmeAIII.dna=Bio.Restriction.Restriction.NmeAIII-class.html#dna,Variable Bio.Restriction.Restriction.NmeDI.dna=Bio.Restriction.Restriction.NmeDI-class.html#dna,Variable Bio.Restriction.Restriction.NmuCI.dna=Bio.Restriction.Restriction.NmuCI-class.html#dna,Variable Bio.Restriction.Restriction.NotI.dna=Bio.Restriction.Restriction.NotI-class.html#dna,Variable Bio.Restriction.Restriction.NpeUS61II.dna=Bio.Restriction.Restriction.NpeUS61II-class.html#dna,Variable Bio.Restriction.Restriction.NruI.dna=Bio.Restriction.Restriction.NruI-class.html#dna,Variable Bio.Restriction.Restriction.NsbI.dna=Bio.Restriction.Restriction.NsbI-class.html#dna,Variable Bio.Restriction.Restriction.NsiI.dna=Bio.Restriction.Restriction.NsiI-class.html#dna,Variable Bio.Restriction.Restriction.NspBII.dna=Bio.Restriction.Restriction.NspBII-class.html#dna,Variable Bio.Restriction.Restriction.NspI.dna=Bio.Restriction.Restriction.NspI-class.html#dna,Variable Bio.Restriction.Restriction.NspV.dna=Bio.Restriction.Restriction.NspV-class.html#dna,Variable Bio.Restriction.Restriction.ObaBS10I.dna=Bio.Restriction.Restriction.ObaBS10I-class.html#dna,Variable Bio.Restriction.Restriction.OgrI.dna=Bio.Restriction.Restriction.OgrI-class.html#dna,Variable Bio.Restriction.Restriction.OliI.dna=Bio.Restriction.Restriction.OliI-class.html#dna,Variable Bio.Restriction.Restriction.OspHL35III.dna=Bio.Restriction.Restriction.OspHL35III-class.html#dna,Variable Bio.Restriction.Restriction.PabI.dna=Bio.Restriction.Restriction.PabI-class.html#dna,Variable Bio.Restriction.Restriction.Pac19842II.dna=Bio.Restriction.Restriction.Pac19842II-class.html#dna,Variable Bio.Restriction.Restriction.PacI.dna=Bio.Restriction.Restriction.PacI-class.html#dna,Variable Bio.Restriction.Restriction.PacIII.dna=Bio.Restriction.Restriction.PacIII-class.html#dna,Variable Bio.Restriction.Restriction.PaeI.dna=Bio.Restriction.Restriction.PaeI-class.html#dna,Variable Bio.Restriction.Restriction.PaeR7I.dna=Bio.Restriction.Restriction.PaeR7I-class.html#dna,Variable Bio.Restriction.Restriction.PagI.dna=Bio.Restriction.Restriction.PagI-class.html#dna,Variable Bio.Restriction.Restriction.Pal408I.dna=Bio.Restriction.Restriction.Pal408I-class.html#dna,Variable Bio.Restriction.Restriction.PalAI.dna=Bio.Restriction.Restriction.PalAI-class.html#dna,Variable Bio.Restriction.Restriction.PasI.dna=Bio.Restriction.Restriction.PasI-class.html#dna,Variable Bio.Restriction.Restriction.PauI.dna=Bio.Restriction.Restriction.PauI-class.html#dna,Variable Bio.Restriction.Restriction.Pba2294I.dna=Bio.Restriction.Restriction.Pba2294I-class.html#dna,Variable Bio.Restriction.Restriction.PcaII.dna=Bio.Restriction.Restriction.PcaII-class.html#dna,Variable Bio.Restriction.Restriction.PceI.dna=Bio.Restriction.Restriction.PceI-class.html#dna,Variable Bio.Restriction.Restriction.PciI.dna=Bio.Restriction.Restriction.PciI-class.html#dna,Variable Bio.Restriction.Restriction.PciSI.dna=Bio.Restriction.Restriction.PciSI-class.html#dna,Variable Bio.Restriction.Restriction.Pcr308II.dna=Bio.Restriction.Restriction.Pcr308II-class.html#dna,Variable Bio.Restriction.Restriction.PcsI.dna=Bio.Restriction.Restriction.PcsI-class.html#dna,Variable Bio.Restriction.Restriction.PctI.dna=Bio.Restriction.Restriction.PctI-class.html#dna,Variable Bio.Restriction.Restriction.Pdi8503III.dna=Bio.Restriction.Restriction.Pdi8503III-class.html#dna,Variable Bio.Restriction.Restriction.PdiI.dna=Bio.Restriction.Restriction.PdiI-class.html#dna,Variable Bio.Restriction.Restriction.PdmI.dna=Bio.Restriction.Restriction.PdmI-class.html#dna,Variable Bio.Restriction.Restriction.Pdu1735I.dna=Bio.Restriction.Restriction.Pdu1735I-class.html#dna,Variable Bio.Restriction.Restriction.PenI.dna=Bio.Restriction.Restriction.PenI-class.html#dna,Variable Bio.Restriction.Restriction.PfeI.dna=Bio.Restriction.Restriction.PfeI-class.html#dna,Variable Bio.Restriction.Restriction.Pfl1108I.dna=Bio.Restriction.Restriction.Pfl1108I-class.html#dna,Variable Bio.Restriction.Restriction.Pfl23II.dna=Bio.Restriction.Restriction.Pfl23II-class.html#dna,Variable Bio.Restriction.Restriction.Pfl8569I.dna=Bio.Restriction.Restriction.Pfl8569I-class.html#dna,Variable Bio.Restriction.Restriction.PflFI.dna=Bio.Restriction.Restriction.PflFI-class.html#dna,Variable Bio.Restriction.Restriction.PflMI.dna=Bio.Restriction.Restriction.PflMI-class.html#dna,Variable Bio.Restriction.Restriction.PflPt14I.dna=Bio.Restriction.Restriction.PflPt14I-class.html#dna,Variable Bio.Restriction.Restriction.PfoI.dna=Bio.Restriction.Restriction.PfoI-class.html#dna,Variable Bio.Restriction.Restriction.PfrJS12IV.dna=Bio.Restriction.Restriction.PfrJS12IV-class.html#dna,Variable Bio.Restriction.Restriction.PfrJS12V.dna=Bio.Restriction.Restriction.PfrJS12V-class.html#dna,Variable Bio.Restriction.Restriction.PfrJS15III.dna=Bio.Restriction.Restriction.PfrJS15III-class.html#dna,Variable Bio.Restriction.Restriction.Pin17FIII.dna=Bio.Restriction.Restriction.Pin17FIII-class.html#dna,Variable Bio.Restriction.Restriction.PinAI.dna=Bio.Restriction.Restriction.PinAI-class.html#dna,Variable Bio.Restriction.Restriction.PinP23II.dna=Bio.Restriction.Restriction.PinP23II-class.html#dna,Variable Bio.Restriction.Restriction.PinP59III.dna=Bio.Restriction.Restriction.PinP59III-class.html#dna,Variable Bio.Restriction.Restriction.PkrI.dna=Bio.Restriction.Restriction.PkrI-class.html#dna,Variable Bio.Restriction.Restriction.PlaDI.dna=Bio.Restriction.Restriction.PlaDI-class.html#dna,Variable Bio.Restriction.Restriction.Ple19I.dna=Bio.Restriction.Restriction.Ple19I-class.html#dna,Variable Bio.Restriction.Restriction.PleI.dna=Bio.Restriction.Restriction.PleI-class.html#dna,Variable Bio.Restriction.Restriction.PliMI.dna=Bio.Restriction.Restriction.PliMI-class.html#dna,Variable Bio.Restriction.Restriction.PluTI.dna=Bio.Restriction.Restriction.PluTI-class.html#dna,Variable Bio.Restriction.Restriction.PmaCI.dna=Bio.Restriction.Restriction.PmaCI-class.html#dna,Variable Bio.Restriction.Restriction.PmeI.dna=Bio.Restriction.Restriction.PmeI-class.html#dna,Variable Bio.Restriction.Restriction.PmlI.dna=Bio.Restriction.Restriction.PmlI-class.html#dna,Variable Bio.Restriction.Restriction.PpiI.dna=Bio.Restriction.Restriction.PpiI-class.html#dna,Variable Bio.Restriction.Restriction.PpiP13II.dna=Bio.Restriction.Restriction.PpiP13II-class.html#dna,Variable Bio.Restriction.Restriction.PpsI.dna=Bio.Restriction.Restriction.PpsI-class.html#dna,Variable Bio.Restriction.Restriction.Ppu10I.dna=Bio.Restriction.Restriction.Ppu10I-class.html#dna,Variable Bio.Restriction.Restriction.Ppu21I.dna=Bio.Restriction.Restriction.Ppu21I-class.html#dna,Variable Bio.Restriction.Restriction.PpuMI.dna=Bio.Restriction.Restriction.PpuMI-class.html#dna,Variable Bio.Restriction.Restriction.PscI.dna=Bio.Restriction.Restriction.PscI-class.html#dna,Variable Bio.Restriction.Restriction.Pse18267I.dna=Bio.Restriction.Restriction.Pse18267I-class.html#dna,Variable Bio.Restriction.Restriction.PshAI.dna=Bio.Restriction.Restriction.PshAI-class.html#dna,Variable Bio.Restriction.Restriction.PshBI.dna=Bio.Restriction.Restriction.PshBI-class.html#dna,Variable Bio.Restriction.Restriction.PsiI.dna=Bio.Restriction.Restriction.PsiI-class.html#dna,Variable Bio.Restriction.Restriction.Psp0357II.dna=Bio.Restriction.Restriction.Psp0357II-class.html#dna,Variable Bio.Restriction.Restriction.Psp03I.dna=Bio.Restriction.Restriction.Psp03I-class.html#dna,Variable Bio.Restriction.Restriction.Psp124BI.dna=Bio.Restriction.Restriction.Psp124BI-class.html#dna,Variable Bio.Restriction.Restriction.Psp1406I.dna=Bio.Restriction.Restriction.Psp1406I-class.html#dna,Variable Bio.Restriction.Restriction.Psp5II.dna=Bio.Restriction.Restriction.Psp5II-class.html#dna,Variable Bio.Restriction.Restriction.Psp6I.dna=Bio.Restriction.Restriction.Psp6I-class.html#dna,Variable Bio.Restriction.Restriction.PspCI.dna=Bio.Restriction.Restriction.PspCI-class.html#dna,Variable Bio.Restriction.Restriction.PspEI.dna=Bio.Restriction.Restriction.PspEI-class.html#dna,Variable Bio.Restriction.Restriction.PspFI.dna=Bio.Restriction.Restriction.PspFI-class.html#dna,Variable Bio.Restriction.Restriction.PspGI.dna=Bio.Restriction.Restriction.PspGI-class.html#dna,Variable Bio.Restriction.Restriction.PspLI.dna=Bio.Restriction.Restriction.PspLI-class.html#dna,Variable Bio.Restriction.Restriction.PspN4I.dna=Bio.Restriction.Restriction.PspN4I-class.html#dna,Variable Bio.Restriction.Restriction.PspOMI.dna=Bio.Restriction.Restriction.PspOMI-class.html#dna,Variable Bio.Restriction.Restriction.PspOMII.dna=Bio.Restriction.Restriction.PspOMII-class.html#dna,Variable Bio.Restriction.Restriction.PspPI.dna=Bio.Restriction.Restriction.PspPI-class.html#dna,Variable Bio.Restriction.Restriction.PspPPI.dna=Bio.Restriction.Restriction.PspPPI-class.html#dna,Variable Bio.Restriction.Restriction.PspPRI.dna=Bio.Restriction.Restriction.PspPRI-class.html#dna,Variable Bio.Restriction.Restriction.PspXI.dna=Bio.Restriction.Restriction.PspXI-class.html#dna,Variable Bio.Restriction.Restriction.PsrI.dna=Bio.Restriction.Restriction.PsrI-class.html#dna,Variable Bio.Restriction.Restriction.PssI.dna=Bio.Restriction.Restriction.PssI-class.html#dna,Variable Bio.Restriction.Restriction.Pst14472I.dna=Bio.Restriction.Restriction.Pst14472I-class.html#dna,Variable Bio.Restriction.Restriction.Pst145I.dna=Bio.Restriction.Restriction.Pst145I-class.html#dna,Variable Bio.Restriction.Restriction.Pst273I.dna=Bio.Restriction.Restriction.Pst273I-class.html#dna,Variable Bio.Restriction.Restriction.PstI.dna=Bio.Restriction.Restriction.PstI-class.html#dna,Variable Bio.Restriction.Restriction.PstNI.dna=Bio.Restriction.Restriction.PstNI-class.html#dna,Variable Bio.Restriction.Restriction.PsuGI.dna=Bio.Restriction.Restriction.PsuGI-class.html#dna,Variable Bio.Restriction.Restriction.PsuI.dna=Bio.Restriction.Restriction.PsuI-class.html#dna,Variable Bio.Restriction.Restriction.PsyI.dna=Bio.Restriction.Restriction.PsyI-class.html#dna,Variable Bio.Restriction.Restriction.PteI.dna=Bio.Restriction.Restriction.PteI-class.html#dna,Variable Bio.Restriction.Restriction.PvuI.dna=Bio.Restriction.Restriction.PvuI-class.html#dna,Variable Bio.Restriction.Restriction.PvuII.dna=Bio.Restriction.Restriction.PvuII-class.html#dna,Variable Bio.Restriction.Restriction.Rba2021I.dna=Bio.Restriction.Restriction.Rba2021I-class.html#dna,Variable Bio.Restriction.Restriction.RceI.dna=Bio.Restriction.Restriction.RceI-class.html#dna,Variable Bio.Restriction.Restriction.RdeGBI.dna=Bio.Restriction.Restriction.RdeGBI-class.html#dna,Variable Bio.Restriction.Restriction.RdeGBII.dna=Bio.Restriction.Restriction.RdeGBII-class.html#dna,Variable Bio.Restriction.Restriction.RdeGBIII.dna=Bio.Restriction.Restriction.RdeGBIII-class.html#dna,Variable Bio.Restriction.Restriction.RflFIII.dna=Bio.Restriction.Restriction.RflFIII-class.html#dna,Variable Bio.Restriction.Restriction.RgaI.dna=Bio.Restriction.Restriction.RgaI-class.html#dna,Variable Bio.Restriction.Restriction.RigI.dna=Bio.Restriction.Restriction.RigI-class.html#dna,Variable Bio.Restriction.Restriction.RlaI.dna=Bio.Restriction.Restriction.RlaI-class.html#dna,Variable Bio.Restriction.Restriction.RlaII.dna=Bio.Restriction.Restriction.RlaII-class.html#dna,Variable Bio.Restriction.Restriction.RleAI.dna=Bio.Restriction.Restriction.RleAI-class.html#dna,Variable Bio.Restriction.Restriction.Rmu369III.dna=Bio.Restriction.Restriction.Rmu369III-class.html#dna,Variable Bio.Restriction.Restriction.RpaB5I.dna=Bio.Restriction.Restriction.RpaB5I-class.html#dna,Variable Bio.Restriction.Restriction.RpaBI.dna=Bio.Restriction.Restriction.RpaBI-class.html#dna,Variable Bio.Restriction.Restriction.RpaI.dna=Bio.Restriction.Restriction.RpaI-class.html#dna,Variable Bio.Restriction.Restriction.RpaTI.dna=Bio.Restriction.Restriction.RpaTI-class.html#dna,Variable Bio.Restriction.Restriction.RruI.dna=Bio.Restriction.Restriction.RruI-class.html#dna,Variable Bio.Restriction.Restriction.RsaI.dna=Bio.Restriction.Restriction.RsaI-class.html#dna,Variable Bio.Restriction.Restriction.RsaNI.dna=Bio.Restriction.Restriction.RsaNI-class.html#dna,Variable Bio.Restriction.Restriction.RseI.dna=Bio.Restriction.Restriction.RseI-class.html#dna,Variable Bio.Restriction.Restriction.Rsp008IV.dna=Bio.Restriction.Restriction.Rsp008IV-class.html#dna,Variable Bio.Restriction.Restriction.Rsp008V.dna=Bio.Restriction.Restriction.Rsp008V-class.html#dna,Variable Bio.Restriction.Restriction.Rsp531II.dna=Bio.Restriction.Restriction.Rsp531II-class.html#dna,Variable Bio.Restriction.Restriction.RspPBTS2III.dna=Bio.Restriction.Restriction.RspPBTS2III-class.html#dna,Variable Bio.Restriction.Restriction.Rsr2I.dna=Bio.Restriction.Restriction.Rsr2I-class.html#dna,Variable Bio.Restriction.Restriction.RsrII.dna=Bio.Restriction.Restriction.RsrII-class.html#dna,Variable Bio.Restriction.Restriction.Rtr1953I.dna=Bio.Restriction.Restriction.Rtr1953I-class.html#dna,Variable Bio.Restriction.Restriction.SacI.dna=Bio.Restriction.Restriction.SacI-class.html#dna,Variable Bio.Restriction.Restriction.SacII.dna=Bio.Restriction.Restriction.SacII-class.html#dna,Variable Bio.Restriction.Restriction.Saf8902III.dna=Bio.Restriction.Restriction.Saf8902III-class.html#dna,Variable Bio.Restriction.Restriction.Sag901I.dna=Bio.Restriction.Restriction.Sag901I-class.html#dna,Variable Bio.Restriction.Restriction.SalI.dna=Bio.Restriction.Restriction.SalI-class.html#dna,Variable Bio.Restriction.Restriction.SanDI.dna=Bio.Restriction.Restriction.SanDI-class.html#dna,Variable Bio.Restriction.Restriction.SapI.dna=Bio.Restriction.Restriction.SapI-class.html#dna,Variable Bio.Restriction.Restriction.SaqAI.dna=Bio.Restriction.Restriction.SaqAI-class.html#dna,Variable Bio.Restriction.Restriction.SatI.dna=Bio.Restriction.Restriction.SatI-class.html#dna,Variable Bio.Restriction.Restriction.Sau3AI.dna=Bio.Restriction.Restriction.Sau3AI-class.html#dna,Variable Bio.Restriction.Restriction.Sau96I.dna=Bio.Restriction.Restriction.Sau96I-class.html#dna,Variable Bio.Restriction.Restriction.SauI.dna=Bio.Restriction.Restriction.SauI-class.html#dna,Variable Bio.Restriction.Restriction.Sba460II.dna=Bio.Restriction.Restriction.Sba460II-class.html#dna,Variable Bio.Restriction.Restriction.SbfI.dna=Bio.Restriction.Restriction.SbfI-class.html#dna,Variable Bio.Restriction.Restriction.Sbo46I.dna=Bio.Restriction.Restriction.Sbo46I-class.html#dna,Variable Bio.Restriction.Restriction.ScaI.dna=Bio.Restriction.Restriction.ScaI-class.html#dna,Variable Bio.Restriction.Restriction.SchI.dna=Bio.Restriction.Restriction.SchI-class.html#dna,Variable Bio.Restriction.Restriction.SciI.dna=Bio.Restriction.Restriction.SciI-class.html#dna,Variable Bio.Restriction.Restriction.ScoDS2II.dna=Bio.Restriction.Restriction.ScoDS2II-class.html#dna,Variable Bio.Restriction.Restriction.ScrFI.dna=Bio.Restriction.Restriction.ScrFI-class.html#dna,Variable Bio.Restriction.Restriction.SdaI.dna=Bio.Restriction.Restriction.SdaI-class.html#dna,Variable Bio.Restriction.Restriction.SdeAI.dna=Bio.Restriction.Restriction.SdeAI-class.html#dna,Variable Bio.Restriction.Restriction.SdeOSI.dna=Bio.Restriction.Restriction.SdeOSI-class.html#dna,Variable Bio.Restriction.Restriction.SduI.dna=Bio.Restriction.Restriction.SduI-class.html#dna,Variable Bio.Restriction.Restriction.SecI.dna=Bio.Restriction.Restriction.SecI-class.html#dna,Variable Bio.Restriction.Restriction.SelI.dna=Bio.Restriction.Restriction.SelI-class.html#dna,Variable Bio.Restriction.Restriction.Sen17963III.dna=Bio.Restriction.Restriction.Sen17963III-class.html#dna,Variable Bio.Restriction.Restriction.SenA1673III.dna=Bio.Restriction.Restriction.SenA1673III-class.html#dna,Variable Bio.Restriction.Restriction.SenSARA26III.dna=Bio.Restriction.Restriction.SenSARA26III-class.html#dna,Variable Bio.Restriction.Restriction.SenTFIV.dna=Bio.Restriction.Restriction.SenTFIV-class.html#dna,Variable Bio.Restriction.Restriction.SetI.dna=Bio.Restriction.Restriction.SetI-class.html#dna,Variable Bio.Restriction.Restriction.SexAI.dna=Bio.Restriction.Restriction.SexAI-class.html#dna,Variable Bio.Restriction.Restriction.SfaAI.dna=Bio.Restriction.Restriction.SfaAI-class.html#dna,Variable Bio.Restriction.Restriction.SfaNI.dna=Bio.Restriction.Restriction.SfaNI-class.html#dna,Variable Bio.Restriction.Restriction.SfcI.dna=Bio.Restriction.Restriction.SfcI-class.html#dna,Variable Bio.Restriction.Restriction.SfeI.dna=Bio.Restriction.Restriction.SfeI-class.html#dna,Variable Bio.Restriction.Restriction.SfiI.dna=Bio.Restriction.Restriction.SfiI-class.html#dna,Variable Bio.Restriction.Restriction.SfoI.dna=Bio.Restriction.Restriction.SfoI-class.html#dna,Variable Bio.Restriction.Restriction.Sfr274I.dna=Bio.Restriction.Restriction.Sfr274I-class.html#dna,Variable Bio.Restriction.Restriction.Sfr303I.dna=Bio.Restriction.Restriction.Sfr303I-class.html#dna,Variable Bio.Restriction.Restriction.SfuI.dna=Bio.Restriction.Restriction.SfuI-class.html#dna,Variable Bio.Restriction.Restriction.SgeI.dna=Bio.Restriction.Restriction.SgeI-class.html#dna,Variable Bio.Restriction.Restriction.SgfI.dna=Bio.Restriction.Restriction.SgfI-class.html#dna,Variable Bio.Restriction.Restriction.SgrAI.dna=Bio.Restriction.Restriction.SgrAI-class.html#dna,Variable Bio.Restriction.Restriction.SgrBI.dna=Bio.Restriction.Restriction.SgrBI-class.html#dna,Variable Bio.Restriction.Restriction.SgrDI.dna=Bio.Restriction.Restriction.SgrDI-class.html#dna,Variable Bio.Restriction.Restriction.SgrTI.dna=Bio.Restriction.Restriction.SgrTI-class.html#dna,Variable Bio.Restriction.Restriction.SgsI.dna=Bio.Restriction.Restriction.SgsI-class.html#dna,Variable Bio.Restriction.Restriction.SimI.dna=Bio.Restriction.Restriction.SimI-class.html#dna,Variable Bio.Restriction.Restriction.SinI.dna=Bio.Restriction.Restriction.SinI-class.html#dna,Variable Bio.Restriction.Restriction.SlaI.dna=Bio.Restriction.Restriction.SlaI-class.html#dna,Variable Bio.Restriction.Restriction.SmaI.dna=Bio.Restriction.Restriction.SmaI-class.html#dna,Variable Bio.Restriction.Restriction.SmaUMH5I.dna=Bio.Restriction.Restriction.SmaUMH5I-class.html#dna,Variable Bio.Restriction.Restriction.SmaUMH8I.dna=Bio.Restriction.Restriction.SmaUMH8I-class.html#dna,Variable Bio.Restriction.Restriction.SmiI.dna=Bio.Restriction.Restriction.SmiI-class.html#dna,Variable Bio.Restriction.Restriction.SmiMI.dna=Bio.Restriction.Restriction.SmiMI-class.html#dna,Variable Bio.Restriction.Restriction.SmlI.dna=Bio.Restriction.Restriction.SmlI-class.html#dna,Variable Bio.Restriction.Restriction.SmoI.dna=Bio.Restriction.Restriction.SmoI-class.html#dna,Variable Bio.Restriction.Restriction.SnaBI.dna=Bio.Restriction.Restriction.SnaBI-class.html#dna,Variable Bio.Restriction.Restriction.SnaI.dna=Bio.Restriction.Restriction.SnaI-class.html#dna,Variable Bio.Restriction.Restriction.Sno506I.dna=Bio.Restriction.Restriction.Sno506I-class.html#dna,Variable Bio.Restriction.Restriction.SpeI.dna=Bio.Restriction.Restriction.SpeI-class.html#dna,Variable Bio.Restriction.Restriction.SphI.dna=Bio.Restriction.Restriction.SphI-class.html#dna,Variable Bio.Restriction.Restriction.SplI.dna=Bio.Restriction.Restriction.SplI-class.html#dna,Variable Bio.Restriction.Restriction.SpnRII.dna=Bio.Restriction.Restriction.SpnRII-class.html#dna,Variable Bio.Restriction.Restriction.SpoDI.dna=Bio.Restriction.Restriction.SpoDI-class.html#dna,Variable Bio.Restriction.Restriction.SrfI.dna=Bio.Restriction.Restriction.SrfI-class.html#dna,Variable Bio.Restriction.Restriction.Sse232I.dna=Bio.Restriction.Restriction.Sse232I-class.html#dna,Variable Bio.Restriction.Restriction.Sse8387I.dna=Bio.Restriction.Restriction.Sse8387I-class.html#dna,Variable Bio.Restriction.Restriction.Sse8647I.dna=Bio.Restriction.Restriction.Sse8647I-class.html#dna,Variable Bio.Restriction.Restriction.Sse9I.dna=Bio.Restriction.Restriction.Sse9I-class.html#dna,Variable Bio.Restriction.Restriction.SseBI.dna=Bio.Restriction.Restriction.SseBI-class.html#dna,Variable Bio.Restriction.Restriction.SsiI.dna=Bio.Restriction.Restriction.SsiI-class.html#dna,Variable Bio.Restriction.Restriction.Ssp6803IV.dna=Bio.Restriction.Restriction.Ssp6803IV-class.html#dna,Variable Bio.Restriction.Restriction.Ssp714II.dna=Bio.Restriction.Restriction.Ssp714II-class.html#dna,Variable Bio.Restriction.Restriction.SspD5I.dna=Bio.Restriction.Restriction.SspD5I-class.html#dna,Variable Bio.Restriction.Restriction.SspDI.dna=Bio.Restriction.Restriction.SspDI-class.html#dna,Variable Bio.Restriction.Restriction.SspI.dna=Bio.Restriction.Restriction.SspI-class.html#dna,Variable Bio.Restriction.Restriction.SspMI.dna=Bio.Restriction.Restriction.SspMI-class.html#dna,Variable Bio.Restriction.Restriction.SstE37I.dna=Bio.Restriction.Restriction.SstE37I-class.html#dna,Variable Bio.Restriction.Restriction.SstI.dna=Bio.Restriction.Restriction.SstI-class.html#dna,Variable Bio.Restriction.Restriction.Sth132I.dna=Bio.Restriction.Restriction.Sth132I-class.html#dna,Variable Bio.Restriction.Restriction.Sth20745III.dna=Bio.Restriction.Restriction.Sth20745III-class.html#dna,Variable Bio.Restriction.Restriction.Sth302II.dna=Bio.Restriction.Restriction.Sth302II-class.html#dna,Variable Bio.Restriction.Restriction.SthSt3II.dna=Bio.Restriction.Restriction.SthSt3II-class.html#dna,Variable Bio.Restriction.Restriction.StsI.dna=Bio.Restriction.Restriction.StsI-class.html#dna,Variable Bio.Restriction.Restriction.StuI.dna=Bio.Restriction.Restriction.StuI-class.html#dna,Variable Bio.Restriction.Restriction.StyD4I.dna=Bio.Restriction.Restriction.StyD4I-class.html#dna,Variable Bio.Restriction.Restriction.StyI.dna=Bio.Restriction.Restriction.StyI-class.html#dna,Variable Bio.Restriction.Restriction.SurP32aII.dna=Bio.Restriction.Restriction.SurP32aII-class.html#dna,Variable Bio.Restriction.Restriction.SwaI.dna=Bio.Restriction.Restriction.SwaI-class.html#dna,Variable Bio.Restriction.Restriction.TaaI.dna=Bio.Restriction.Restriction.TaaI-class.html#dna,Variable Bio.Restriction.Restriction.TaiI.dna=Bio.Restriction.Restriction.TaiI-class.html#dna,Variable Bio.Restriction.Restriction.TaqI.dna=Bio.Restriction.Restriction.TaqI-class.html#dna,Variable Bio.Restriction.Restriction.TaqII.dna=Bio.Restriction.Restriction.TaqII-class.html#dna,Variable Bio.Restriction.Restriction.TaqIII.dna=Bio.Restriction.Restriction.TaqIII-class.html#dna,Variable Bio.Restriction.Restriction.TasI.dna=Bio.Restriction.Restriction.TasI-class.html#dna,Variable Bio.Restriction.Restriction.TatI.dna=Bio.Restriction.Restriction.TatI-class.html#dna,Variable Bio.Restriction.Restriction.TauI.dna=Bio.Restriction.Restriction.TauI-class.html#dna,Variable Bio.Restriction.Restriction.TfiI.dna=Bio.Restriction.Restriction.TfiI-class.html#dna,Variable Bio.Restriction.Restriction.TpyTP2I.dna=Bio.Restriction.Restriction.TpyTP2I-class.html#dna,Variable Bio.Restriction.Restriction.Tru1I.dna=Bio.Restriction.Restriction.Tru1I-class.html#dna,Variable Bio.Restriction.Restriction.Tru9I.dna=Bio.Restriction.Restriction.Tru9I-class.html#dna,Variable Bio.Restriction.Restriction.TscAI.dna=Bio.Restriction.Restriction.TscAI-class.html#dna,Variable Bio.Restriction.Restriction.TseFI.dna=Bio.Restriction.Restriction.TseFI-class.html#dna,Variable Bio.Restriction.Restriction.TseI.dna=Bio.Restriction.Restriction.TseI-class.html#dna,Variable Bio.Restriction.Restriction.TsoI.dna=Bio.Restriction.Restriction.TsoI-class.html#dna,Variable Bio.Restriction.Restriction.Tsp45I.dna=Bio.Restriction.Restriction.Tsp45I-class.html#dna,Variable Bio.Restriction.Restriction.Tsp4CI.dna=Bio.Restriction.Restriction.Tsp4CI-class.html#dna,Variable Bio.Restriction.Restriction.TspARh3I.dna=Bio.Restriction.Restriction.TspARh3I-class.html#dna,Variable Bio.Restriction.Restriction.TspDTI.dna=Bio.Restriction.Restriction.TspDTI-class.html#dna,Variable Bio.Restriction.Restriction.TspEI.dna=Bio.Restriction.Restriction.TspEI-class.html#dna,Variable Bio.Restriction.Restriction.TspGWI.dna=Bio.Restriction.Restriction.TspGWI-class.html#dna,Variable Bio.Restriction.Restriction.TspMI.dna=Bio.Restriction.Restriction.TspMI-class.html#dna,Variable Bio.Restriction.Restriction.TspRI.dna=Bio.Restriction.Restriction.TspRI-class.html#dna,Variable Bio.Restriction.Restriction.TssI.dna=Bio.Restriction.Restriction.TssI-class.html#dna,Variable Bio.Restriction.Restriction.TstI.dna=Bio.Restriction.Restriction.TstI-class.html#dna,Variable Bio.Restriction.Restriction.TsuI.dna=Bio.Restriction.Restriction.TsuI-class.html#dna,Variable Bio.Restriction.Restriction.Tth111I.dna=Bio.Restriction.Restriction.Tth111I-class.html#dna,Variable Bio.Restriction.Restriction.Tth111II.dna=Bio.Restriction.Restriction.Tth111II-class.html#dna,Variable Bio.Restriction.Restriction.UbaF11I.dna=Bio.Restriction.Restriction.UbaF11I-class.html#dna,Variable Bio.Restriction.Restriction.UbaF12I.dna=Bio.Restriction.Restriction.UbaF12I-class.html#dna,Variable Bio.Restriction.Restriction.UbaF13I.dna=Bio.Restriction.Restriction.UbaF13I-class.html#dna,Variable Bio.Restriction.Restriction.UbaF14I.dna=Bio.Restriction.Restriction.UbaF14I-class.html#dna,Variable Bio.Restriction.Restriction.UbaF9I.dna=Bio.Restriction.Restriction.UbaF9I-class.html#dna,Variable Bio.Restriction.Restriction.UbaPI.dna=Bio.Restriction.Restriction.UbaPI-class.html#dna,Variable Bio.Restriction.Restriction.UcoMSI.dna=Bio.Restriction.Restriction.UcoMSI-class.html#dna,Variable Bio.Restriction.Restriction.UnbI.dna=Bio.Restriction.Restriction.UnbI-class.html#dna,Variable Bio.Restriction.Restriction.Van9116I.dna=Bio.Restriction.Restriction.Van9116I-class.html#dna,Variable Bio.Restriction.Restriction.Van91I.dna=Bio.Restriction.Restriction.Van91I-class.html#dna,Variable Bio.Restriction.Restriction.VchE4II.dna=Bio.Restriction.Restriction.VchE4II-class.html#dna,Variable Bio.Restriction.Restriction.Vdi96II.dna=Bio.Restriction.Restriction.Vdi96II-class.html#dna,Variable Bio.Restriction.Restriction.Vha464I.dna=Bio.Restriction.Restriction.Vha464I-class.html#dna,Variable Bio.Restriction.Restriction.VneI.dna=Bio.Restriction.Restriction.VneI-class.html#dna,Variable Bio.Restriction.Restriction.VpaK11AI.dna=Bio.Restriction.Restriction.VpaK11AI-class.html#dna,Variable Bio.Restriction.Restriction.VpaK11BI.dna=Bio.Restriction.Restriction.VpaK11BI-class.html#dna,Variable Bio.Restriction.Restriction.VspI.dna=Bio.Restriction.Restriction.VspI-class.html#dna,Variable Bio.Restriction.Restriction.Vtu19109I.dna=Bio.Restriction.Restriction.Vtu19109I-class.html#dna,Variable Bio.Restriction.Restriction.WviI.dna=Bio.Restriction.Restriction.WviI-class.html#dna,Variable Bio.Restriction.Restriction.XagI.dna=Bio.Restriction.Restriction.XagI-class.html#dna,Variable Bio.Restriction.Restriction.XapI.dna=Bio.Restriction.Restriction.XapI-class.html#dna,Variable Bio.Restriction.Restriction.XbaI.dna=Bio.Restriction.Restriction.XbaI-class.html#dna,Variable Bio.Restriction.Restriction.Xca85IV.dna=Bio.Restriction.Restriction.Xca85IV-class.html#dna,Variable Bio.Restriction.Restriction.XceI.dna=Bio.Restriction.Restriction.XceI-class.html#dna,Variable Bio.Restriction.Restriction.XcmI.dna=Bio.Restriction.Restriction.XcmI-class.html#dna,Variable Bio.Restriction.Restriction.XhoI.dna=Bio.Restriction.Restriction.XhoI-class.html#dna,Variable Bio.Restriction.Restriction.XhoII.dna=Bio.Restriction.Restriction.XhoII-class.html#dna,Variable Bio.Restriction.Restriction.XmaI.dna=Bio.Restriction.Restriction.XmaI-class.html#dna,Variable Bio.Restriction.Restriction.XmaIII.dna=Bio.Restriction.Restriction.XmaIII-class.html#dna,Variable Bio.Restriction.Restriction.XmaJI.dna=Bio.Restriction.Restriction.XmaJI-class.html#dna,Variable Bio.Restriction.Restriction.XmiI.dna=Bio.Restriction.Restriction.XmiI-class.html#dna,Variable Bio.Restriction.Restriction.XmnI.dna=Bio.Restriction.Restriction.XmnI-class.html#dna,Variable Bio.Restriction.Restriction.XspI.dna=Bio.Restriction.Restriction.XspI-class.html#dna,Variable Bio.Restriction.Restriction.YkrI.dna=Bio.Restriction.Restriction.YkrI-class.html#dna,Variable Bio.Restriction.Restriction.ZraI.dna=Bio.Restriction.Restriction.ZraI-class.html#dna,Variable Bio.Restriction.Restriction.ZrmI.dna=Bio.Restriction.Restriction.ZrmI-class.html#dna,Variable Bio.Restriction.Restriction.Zsp2I.dna=Bio.Restriction.Restriction.Zsp2I-class.html#dna"><a title="Bio.Restriction.Restriction.AanI.dna
Bio.Restriction.Restriction.AarI.dna
Bio.Restriction.Restriction.AasI.dna
Bio.Restriction.Restriction.AatII.dna
Bio.Restriction.Restriction.Aba6411II.dna
Bio.Restriction.Restriction.AbaB8342IV.dna
Bio.Restriction.Restriction.AbaCIII.dna
Bio.Restriction.Restriction.AbaSI.dna
Bio.Restriction.Restriction.AbaUMB2I.dna
Bio.Restriction.Restriction.AbsI.dna
Bio.Restriction.Restriction.Acc16I.dna
Bio.Restriction.Restriction.Acc36I.dna
Bio.Restriction.Restriction.Acc65I.dna
Bio.Restriction.Restriction.Acc65V.dna
Bio.Restriction.Restriction.AccB1I.dna
Bio.Restriction.Restriction.AccB7I.dna
Bio.Restriction.Restriction.AccBSI.dna
Bio.Restriction.Restriction.AccI.dna
Bio.Restriction.Restriction.AccII.dna
Bio.Restriction.Restriction.AccIII.dna
Bio.Restriction.Restriction.AceIII.dna
Bio.Restriction.Restriction.AchA6III.dna
Bio.Restriction.Restriction.AciI.dna
Bio.Restriction.Restriction.AclI.dna
Bio.Restriction.Restriction.AclWI.dna
Bio.Restriction.Restriction.Aco12261II.dna
Bio.Restriction.Restriction.AcoI.dna
Bio.Restriction.Restriction.AcoY31II.dna
Bio.Restriction.Restriction.AcsI.dna
Bio.Restriction.Restriction.AcuI.dna
Bio.Restriction.Restriction.AcvI.dna
Bio.Restriction.Restriction.AcyI.dna
Bio.Restriction.Restriction.AdeI.dna
Bio.Restriction.Restriction.Adh6U21I.dna
Bio.Restriction.Restriction.AfaI.dna
Bio.Restriction.Restriction.AfeI.dna
Bio.Restriction.Restriction.AfiI.dna
Bio.Restriction.Restriction.AflII.dna
Bio.Restriction.Restriction.AflIII.dna
Bio.Restriction.Restriction.AgeI.dna
Bio.Restriction.Restriction.AgsI.dna
Bio.Restriction.Restriction.AhaIII.dna
Bio.Restriction.Restriction.AhdI.dna
Bio.Restriction.Restriction.AhlI.dna
Bio.Restriction.Restriction.AhyRBAHI.dna
Bio.Restriction.Restriction.AhyYL17I.dna
Bio.Restriction.Restriction.AjiI.dna
Bio.Restriction.Restriction.AjnI.dna
Bio.Restriction.Restriction.AjuI.dna
Bio.Restriction.Restriction.AleI.dna
Bio.Restriction.Restriction.AlfI.dna
Bio.Restriction.Restriction.AloI.dna
Bio.Restriction.Restriction.AluBI.dna
Bio.Restriction.Restriction.AluI.dna
Bio.Restriction.Restriction.Alw21I.dna
Bio.Restriction.Restriction.Alw26I.dna
Bio.Restriction.Restriction.Alw44I.dna
Bio.Restriction.Restriction.AlwFI.dna
Bio.Restriction.Restriction.AlwI.dna
Bio.Restriction.Restriction.AlwNI.dna
Bio.Restriction.Restriction.Ama87I.dna
Bio.Restriction.Restriction.AmaCSI.dna
Bio.Restriction.Restriction.Aod1I.dna
Bio.Restriction.Restriction.Aor13HI.dna
Bio.Restriction.Restriction.Aor51HI.dna
Bio.Restriction.Restriction.AoxI.dna
Bio.Restriction.Restriction.ApaBI.dna
Bio.Restriction.Restriction.ApaI.dna
Bio.Restriction.Restriction.ApaLI.dna
Bio.Restriction.Restriction.ApeKI.dna
Bio.Restriction.Restriction.ApoI.dna
Bio.Restriction.Restriction.ApyPI.dna
Bio.Restriction.Restriction.AquII.dna
Bio.Restriction.Restriction.AquIII.dna
Bio.Restriction.Restriction.AquIV.dna
Bio.Restriction.Restriction.ArsI.dna
Bio.Restriction.Restriction.AscI.dna
Bio.Restriction.Restriction.AseI.dna
Bio.Restriction.Restriction.Asi256I.dna
Bio.Restriction.Restriction.AsiGI.dna
Bio.Restriction.Restriction.AsiSI.dna
Bio.Restriction.Restriction.Asp103I.dna
Bio.Restriction.Restriction.Asp114pII.dna
Bio.Restriction.Restriction.Asp700I.dna
Bio.Restriction.Restriction.Asp718I.dna
Bio.Restriction.Restriction.AspA2I.dna
Bio.Restriction.Restriction.AspBHI.dna
Bio.Restriction.Restriction.AspDUT2V.dna
Bio.Restriction.Restriction.AspJHL3II.dna
Bio.Restriction.Restriction.AspLEI.dna
Bio.Restriction.Restriction.AspNIH4III.dna
Bio.Restriction.Restriction.AspS9I.dna
Bio.Restriction.Restriction.AspSLV7III.dna
Bio.Restriction.Restriction.Asu14238IV.dna
Bio.Restriction.Restriction.AsuC2I.dna
Bio.Restriction.Restriction.AsuHPI.dna
Bio.Restriction.Restriction.AsuI.dna
Bio.Restriction.Restriction.AsuII.dna
Bio.Restriction.Restriction.AsuNHI.dna
Bio.Restriction.Restriction.AteTI.dna
Bio.Restriction.Restriction.AvaI.dna
Bio.Restriction.Restriction.AvaII.dna
Bio.Restriction.Restriction.AvaIII.dna
Bio.Restriction.Restriction.AvrII.dna
Bio.Restriction.Restriction.Awo1030IV.dna
Bio.Restriction.Restriction.AxyI.dna
Bio.Restriction.Restriction.BaeGI.dna
Bio.Restriction.Restriction.BaeI.dna
Bio.Restriction.Restriction.Bag18758I.dna
Bio.Restriction.Restriction.BalI.dna
Bio.Restriction.Restriction.BamHI.dna
Bio.Restriction.Restriction.BanI.dna
Bio.Restriction.Restriction.BanII.dna
Bio.Restriction.Restriction.BanLI.dna
Bio.Restriction.Restriction.BarI.dna
Bio.Restriction.Restriction.Bau1417V.dna
Bio.Restriction.Restriction.BauI.dna
Bio.Restriction.Restriction.Bbr52II.dna
Bio.Restriction.Restriction.Bbr57III.dna
Bio.Restriction.Restriction.Bbr7017II.dna
Bio.Restriction.Restriction.Bbr7017III.dna
Bio.Restriction.Restriction.BbrPI.dna
Bio.Restriction.Restriction.BbsI.dna
Bio.Restriction.Restriction.BbuB31I.dna
Bio.Restriction.Restriction.BbuB31II.dna
Bio.Restriction.Restriction.Bbv12I.dna
Bio.Restriction.Restriction.BbvCI.dna
Bio.Restriction.Restriction.BbvI.dna
Bio.Restriction.Restriction.BbvII.dna
Bio.Restriction.Restriction.BccI.dna
Bio.Restriction.Restriction.Bce3081I.dna
Bio.Restriction.Restriction.Bce83I.dna
Bio.Restriction.Restriction.BceAI.dna
Bio.Restriction.Restriction.BceSIV.dna
Bio.Restriction.Restriction.BcefI.dna
Bio.Restriction.Restriction.BcgI.dna
Bio.Restriction.Restriction.BciT130I.dna
Bio.Restriction.Restriction.BciVI.dna
Bio.Restriction.Restriction.BclI.dna
Bio.Restriction.Restriction.BcnI.dna
Bio.Restriction.Restriction.BcoDI.dna
Bio.Restriction.Restriction.BcuI.dna
Bio.Restriction.Restriction.BdaI.dna
Bio.Restriction.Restriction.BetI.dna
Bio.Restriction.Restriction.BfaI.dna
Bio.Restriction.Restriction.BfaSII.dna
Bio.Restriction.Restriction.BfiI.dna
Bio.Restriction.Restriction.BfmI.dna
Bio.Restriction.Restriction.BfoI.dna
Bio.Restriction.Restriction.BfrI.dna
Bio.Restriction.Restriction.BfuAI.dna
Bio.Restriction.Restriction.BfuI.dna
Bio.Restriction.Restriction.Bga514I.dna
Bio.Restriction.Restriction.BglI.dna
Bio.Restriction.Restriction.BglII.dna
Bio.Restriction.Restriction.BinI.dna
Bio.Restriction.Restriction.BisI.dna
Bio.Restriction.Restriction.BkrAM31DI.dna
Bio.Restriction.Restriction.Ble402II.dna
Bio.Restriction.Restriction.BlnI.dna
Bio.Restriction.Restriction.BloAII.dna
Bio.Restriction.Restriction.BlpI.dna
Bio.Restriction.Restriction.BlsI.dna
Bio.Restriction.Restriction.BmcAI.dna
Bio.Restriction.Restriction.Bme1390I.dna
Bio.Restriction.Restriction.Bme18I.dna
Bio.Restriction.Restriction.BmeDI.dna
Bio.Restriction.Restriction.BmeRI.dna
Bio.Restriction.Restriction.BmeT110I.dna
Bio.Restriction.Restriction.BmgBI.dna
Bio.Restriction.Restriction.BmgI.dna
Bio.Restriction.Restriction.BmgT120I.dna
Bio.Restriction.Restriction.BmiI.dna
Bio.Restriction.Restriction.BmrFI.dna
Bio.Restriction.Restriction.BmrI.dna
Bio.Restriction.Restriction.BmsI.dna
Bio.Restriction.Restriction.BmtI.dna
Bio.Restriction.Restriction.BmuI.dna
Bio.Restriction.Restriction.BoxI.dna
Bio.Restriction.Restriction.BpiI.dna
Bio.Restriction.Restriction.BplI.dna
Bio.Restriction.Restriction.BpmI.dna
Bio.Restriction.Restriction.Bpu10I.dna
Bio.Restriction.Restriction.Bpu1102I.dna
Bio.Restriction.Restriction.Bpu14I.dna
Bio.Restriction.Restriction.BpuEI.dna
Bio.Restriction.Restriction.BpuMI.dna
Bio.Restriction.Restriction.Bsa29I.dna
Bio.Restriction.Restriction.BsaAI.dna
Bio.Restriction.Restriction.BsaBI.dna
Bio.Restriction.Restriction.BsaHI.dna
Bio.Restriction.Restriction.BsaI.dna
Bio.Restriction.Restriction.BsaJI.dna
Bio.Restriction.Restriction.BsaWI.dna
Bio.Restriction.Restriction.BsaXI.dna
Bio.Restriction.Restriction.BsbI.dna
Bio.Restriction.Restriction.Bsc4I.dna
Bio.Restriction.Restriction.BscAI.dna
Bio.Restriction.Restriction.BscGI.dna
Bio.Restriction.Restriction.Bse118I.dna
Bio.Restriction.Restriction.Bse1I.dna
Bio.Restriction.Restriction.Bse21I.dna
Bio.Restriction.Restriction.Bse3DI.dna
Bio.Restriction.Restriction.Bse8I.dna
Bio.Restriction.Restriction.BseAI.dna
Bio.Restriction.Restriction.BseBI.dna
Bio.Restriction.Restriction.BseCI.dna
Bio.Restriction.Restriction.BseDI.dna
Bio.Restriction.Restriction.BseGI.dna
Bio.Restriction.Restriction.BseJI.dna
Bio.Restriction.Restriction.BseLI.dna
Bio.Restriction.Restriction.BseMI.dna
Bio.Restriction.Restriction.BseMII.dna
Bio.Restriction.Restriction.BseNI.dna
Bio.Restriction.Restriction.BsePI.dna
Bio.Restriction.Restriction.BseRI.dna
Bio.Restriction.Restriction.BseSI.dna
Bio.Restriction.Restriction.BseX3I.dna
Bio.Restriction.Restriction.BseXI.dna
Bio.Restriction.Restriction.BseYI.dna
Bio.Restriction.Restriction.BsgI.dna
Bio.Restriction.Restriction.Bsh1236I.dna
Bio.Restriction.Restriction.Bsh1285I.dna
Bio.Restriction.Restriction.BshFI.dna
Bio.Restriction.Restriction.BshNI.dna
Bio.Restriction.Restriction.BshTI.dna
Bio.Restriction.Restriction.BshVI.dna
Bio.Restriction.Restriction.BsiEI.dna
Bio.Restriction.Restriction.BsiHKAI.dna
Bio.Restriction.Restriction.BsiHKCI.dna
Bio.Restriction.Restriction.BsiI.dna
Bio.Restriction.Restriction.BsiSI.dna
Bio.Restriction.Restriction.BsiWI.dna
Bio.Restriction.Restriction.BsiYI.dna
Bio.Restriction.Restriction.BslFI.dna
Bio.Restriction.Restriction.BslI.dna
Bio.Restriction.Restriction.BsmAI.dna
Bio.Restriction.Restriction.BsmBI.dna
Bio.Restriction.Restriction.BsmFI.dna
Bio.Restriction.Restriction.BsmI.dna
Bio.Restriction.Restriction.BsnI.dna
Bio.Restriction.Restriction.Bso31I.dna
Bio.Restriction.Restriction.BsoBI.dna
Bio.Restriction.Restriction.Bsp119I.dna
Bio.Restriction.Restriction.Bsp120I.dna
Bio.Restriction.Restriction.Bsp1286I.dna
Bio.Restriction.Restriction.Bsp13I.dna
Bio.Restriction.Restriction.Bsp1407I.dna
Bio.Restriction.Restriction.Bsp143I.dna
Bio.Restriction.Restriction.Bsp1720I.dna
Bio.Restriction.Restriction.Bsp19I.dna
Bio.Restriction.Restriction.Bsp24I.dna
Bio.Restriction.Restriction.Bsp3004IV.dna
Bio.Restriction.Restriction.Bsp460III.dna
Bio.Restriction.Restriction.Bsp68I.dna
Bio.Restriction.Restriction.BspACI.dna
Bio.Restriction.Restriction.BspANI.dna
Bio.Restriction.Restriction.BspCNI.dna
Bio.Restriction.Restriction.BspD6I.dna
Bio.Restriction.Restriction.BspDI.dna
Bio.Restriction.Restriction.BspEI.dna
Bio.Restriction.Restriction.BspFNI.dna
Bio.Restriction.Restriction.BspGI.dna
Bio.Restriction.Restriction.BspHI.dna
Bio.Restriction.Restriction.BspLI.dna
Bio.Restriction.Restriction.BspLU11I.dna
Bio.Restriction.Restriction.BspMAI.dna
Bio.Restriction.Restriction.BspMI.dna
Bio.Restriction.Restriction.BspMII.dna
Bio.Restriction.Restriction.BspNCI.dna
Bio.Restriction.Restriction.BspOI.dna
Bio.Restriction.Restriction.BspPI.dna
Bio.Restriction.Restriction.BspQI.dna
Bio.Restriction.Restriction.BspT104I.dna
Bio.Restriction.Restriction.BspT107I.dna
Bio.Restriction.Restriction.BspTI.dna
Bio.Restriction.Restriction.BspTNI.dna
Bio.Restriction.Restriction.BsrBI.dna
Bio.Restriction.Restriction.BsrDI.dna
Bio.Restriction.Restriction.BsrFI.dna
Bio.Restriction.Restriction.BsrGI.dna
Bio.Restriction.Restriction.BsrI.dna
Bio.Restriction.Restriction.BssAI.dna
Bio.Restriction.Restriction.BssECI.dna
Bio.Restriction.Restriction.BssHII.dna
Bio.Restriction.Restriction.BssMI.dna
Bio.Restriction.Restriction.BssNAI.dna
Bio.Restriction.Restriction.BssNI.dna
Bio.Restriction.Restriction.BssSI.dna
Bio.Restriction.Restriction.BssT1I.dna
Bio.Restriction.Restriction.Bst1107I.dna
Bio.Restriction.Restriction.Bst2BI.dna
Bio.Restriction.Restriction.Bst2UI.dna
Bio.Restriction.Restriction.Bst4CI.dna
Bio.Restriction.Restriction.Bst6I.dna
Bio.Restriction.Restriction.BstACI.dna
Bio.Restriction.Restriction.BstAFI.dna
Bio.Restriction.Restriction.BstAPI.dna
Bio.Restriction.Restriction.BstAUI.dna
Bio.Restriction.Restriction.BstBAI.dna
Bio.Restriction.Restriction.BstBI.dna
Bio.Restriction.Restriction.BstC8I.dna
Bio.Restriction.Restriction.BstDEI.dna
Bio.Restriction.Restriction.BstDSI.dna
Bio.Restriction.Restriction.BstEII.dna
Bio.Restriction.Restriction.BstENI.dna
Bio.Restriction.Restriction.BstF5I.dna
Bio.Restriction.Restriction.BstFNI.dna
Bio.Restriction.Restriction.BstH2I.dna
Bio.Restriction.Restriction.BstHHI.dna
Bio.Restriction.Restriction.BstKTI.dna
Bio.Restriction.Restriction.BstMAI.dna
Bio.Restriction.Restriction.BstMBI.dna
Bio.Restriction.Restriction.BstMCI.dna
Bio.Restriction.Restriction.BstMWI.dna
Bio.Restriction.Restriction.BstNI.dna
Bio.Restriction.Restriction.BstNSI.dna
Bio.Restriction.Restriction.BstPAI.dna
Bio.Restriction.Restriction.BstPI.dna
Bio.Restriction.Restriction.BstSCI.dna
Bio.Restriction.Restriction.BstSFI.dna
Bio.Restriction.Restriction.BstSLI.dna
Bio.Restriction.Restriction.BstSNI.dna
Bio.Restriction.Restriction.BstUI.dna
Bio.Restriction.Restriction.BstV1I.dna
Bio.Restriction.Restriction.BstV2I.dna
Bio.Restriction.Restriction.BstX2I.dna
Bio.Restriction.Restriction.BstXI.dna
Bio.Restriction.Restriction.BstYI.dna
Bio.Restriction.Restriction.BstZ17I.dna
Bio.Restriction.Restriction.BstZI.dna
Bio.Restriction.Restriction.Bsu15I.dna
Bio.Restriction.Restriction.Bsu36I.dna
Bio.Restriction.Restriction.BsuI.dna
Bio.Restriction.Restriction.BsuRI.dna
Bio.Restriction.Restriction.BsuTUI.dna
Bio.Restriction.Restriction.BtgI.dna
Bio.Restriction.Restriction.BtgZI.dna
Bio.Restriction.Restriction.BthCI.dna
Bio.Restriction.Restriction.BtrI.dna
Bio.Restriction.Restriction.BtsCI.dna
Bio.Restriction.Restriction.BtsI.dna
Bio.Restriction.Restriction.BtsIMutI.dna
Bio.Restriction.Restriction.BtuMI.dna
Bio.Restriction.Restriction.Bve1B23I.dna
Bio.Restriction.Restriction.BveI.dna
Bio.Restriction.Restriction.Cac8I.dna
Bio.Restriction.Restriction.CaiI.dna
Bio.Restriction.Restriction.Cal14237I.dna
Bio.Restriction.Restriction.CalB3II.dna
Bio.Restriction.Restriction.Cau10061II.dna
Bio.Restriction.Restriction.CauII.dna
Bio.Restriction.Restriction.Cba13II.dna
Bio.Restriction.Restriction.Cba16038I.dna
Bio.Restriction.Restriction.Cbo67071IV.dna
Bio.Restriction.Restriction.Cch467III.dna
Bio.Restriction.Restriction.CchII.dna
Bio.Restriction.Restriction.CchIII.dna
Bio.Restriction.Restriction.CciI.dna
Bio.Restriction.Restriction.CciNI.dna
Bio.Restriction.Restriction.Cco14983V.dna
Bio.Restriction.Restriction.Cco14983VI.dna
Bio.Restriction.Restriction.CcrNAIII.dna
Bio.Restriction.Restriction.Cdi11397I.dna
Bio.Restriction.Restriction.CdiI.dna
Bio.Restriction.Restriction.CdpI.dna
Bio.Restriction.Restriction.Cdu23823II.dna
Bio.Restriction.Restriction.CfoI.dna
Bio.Restriction.Restriction.Cfr10I.dna
Bio.Restriction.Restriction.Cfr13I.dna
Bio.Restriction.Restriction.Cfr42I.dna
Bio.Restriction.Restriction.Cfr9I.dna
Bio.Restriction.Restriction.CfrI.dna
Bio.Restriction.Restriction.CfrMH13II.dna
Bio.Restriction.Restriction.CfrMH16VI.dna
Bio.Restriction.Restriction.Cfupf3II.dna
Bio.Restriction.Restriction.Cgl13032I.dna
Bio.Restriction.Restriction.Cgl13032II.dna
Bio.Restriction.Restriction.ChaI.dna
Bio.Restriction.Restriction.Cje265V.dna
Bio.Restriction.Restriction.Cje54107III.dna
Bio.Restriction.Restriction.CjeFIII.dna
Bio.Restriction.Restriction.CjeFV.dna
Bio.Restriction.Restriction.CjeI.dna
Bio.Restriction.Restriction.CjeNII.dna
Bio.Restriction.Restriction.CjeNIII.dna
Bio.Restriction.Restriction.CjeNV.dna
Bio.Restriction.Restriction.CjeP659IV.dna
Bio.Restriction.Restriction.CjePI.dna
Bio.Restriction.Restriction.CjuI.dna
Bio.Restriction.Restriction.CjuII.dna
Bio.Restriction.Restriction.Cla11845III.dna
Bio.Restriction.Restriction.ClaI.dna
Bio.Restriction.Restriction.Cly7489II.dna
Bio.Restriction.Restriction.Cma23826I.dna
Bio.Restriction.Restriction.CpoI.dna
Bio.Restriction.Restriction.CseI.dna
Bio.Restriction.Restriction.CsiI.dna
Bio.Restriction.Restriction.Csp2014I.dna
Bio.Restriction.Restriction.Csp6I.dna
Bio.Restriction.Restriction.CspAI.dna
Bio.Restriction.Restriction.CspCI.dna
Bio.Restriction.Restriction.CspI.dna
Bio.Restriction.Restriction.CstMI.dna
Bio.Restriction.Restriction.CviAII.dna
Bio.Restriction.Restriction.CviJI.dna
Bio.Restriction.Restriction.CviKI_1.dna
Bio.Restriction.Restriction.CviQI.dna
Bio.Restriction.Restriction.CviRI.dna
Bio.Restriction.Restriction.Dde51507I.dna
Bio.Restriction.Restriction.DdeI.dna
Bio.Restriction.Restriction.DinI.dna
Bio.Restriction.Restriction.DpnI.dna
Bio.Restriction.Restriction.DpnII.dna
Bio.Restriction.Restriction.DraI.dna
Bio.Restriction.Restriction.DraII.dna
Bio.Restriction.Restriction.DraIII.dna
Bio.Restriction.Restriction.DraRI.dna
Bio.Restriction.Restriction.DrdI.dna
Bio.Restriction.Restriction.DrdII.dna
Bio.Restriction.Restriction.DrdIV.dna
Bio.Restriction.Restriction.DriI.dna
Bio.Restriction.Restriction.DsaI.dna
Bio.Restriction.Restriction.DseDI.dna
Bio.Restriction.Restriction.DvuIII.dna
Bio.Restriction.Restriction.EaeI.dna
Bio.Restriction.Restriction.EagI.dna
Bio.Restriction.Restriction.Eam1104I.dna
Bio.Restriction.Restriction.Eam1105I.dna
Bio.Restriction.Restriction.EarI.dna
Bio.Restriction.Restriction.EciI.dna
Bio.Restriction.Restriction.Ecl136II.dna
Bio.Restriction.Restriction.Ecl234I.dna
Bio.Restriction.Restriction.Ecl35734I.dna
Bio.Restriction.Restriction.EclXI.dna
Bio.Restriction.Restriction.Eco105I.dna
Bio.Restriction.Restriction.Eco130I.dna
Bio.Restriction.Restriction.Eco147I.dna
Bio.Restriction.Restriction.Eco24I.dna
Bio.Restriction.Restriction.Eco31I.dna
Bio.Restriction.Restriction.Eco32I.dna
Bio.Restriction.Restriction.Eco43896II.dna
Bio.Restriction.Restriction.Eco4465II.dna
Bio.Restriction.Restriction.Eco47I.dna
Bio.Restriction.Restriction.Eco47III.dna
Bio.Restriction.Restriction.Eco52I.dna
Bio.Restriction.Restriction.Eco53kI.dna
Bio.Restriction.Restriction.Eco57I.dna
Bio.Restriction.Restriction.Eco57MI.dna
Bio.Restriction.Restriction.Eco72I.dna
Bio.Restriction.Restriction.Eco81I.dna
Bio.Restriction.Restriction.Eco88I.dna
Bio.Restriction.Restriction.Eco91I.dna
Bio.Restriction.Restriction.EcoBLMcrX.dna
Bio.Restriction.Restriction.EcoE1140I.dna
Bio.Restriction.Restriction.EcoHI.dna
Bio.Restriction.Restriction.EcoHSI.dna
Bio.Restriction.Restriction.EcoICRI.dna
Bio.Restriction.Restriction.EcoMVII.dna
Bio.Restriction.Restriction.EcoNI.dna
Bio.Restriction.Restriction.EcoNIH6II.dna
Bio.Restriction.Restriction.EcoO109I.dna
Bio.Restriction.Restriction.EcoO65I.dna
Bio.Restriction.Restriction.EcoRI.dna
Bio.Restriction.Restriction.EcoRII.dna
Bio.Restriction.Restriction.EcoRV.dna
Bio.Restriction.Restriction.EcoT14I.dna
Bio.Restriction.Restriction.EcoT22I.dna
Bio.Restriction.Restriction.EcoT38I.dna
Bio.Restriction.Restriction.EgeI.dna
Bio.Restriction.Restriction.EheI.dna
Bio.Restriction.Restriction.Eli8509II.dna
Bio.Restriction.Restriction.ErhI.dna
Bio.Restriction.Restriction.EsaBC3I.dna
Bio.Restriction.Restriction.EsaSSI.dna
Bio.Restriction.Restriction.Esp3007I.dna
Bio.Restriction.Restriction.Esp3I.dna
Bio.Restriction.Restriction.EspI.dna
Bio.Restriction.Restriction.FaeI.dna
Bio.Restriction.Restriction.FaiI.dna
Bio.Restriction.Restriction.FalI.dna
Bio.Restriction.Restriction.FaqI.dna
Bio.Restriction.Restriction.FatI.dna
Bio.Restriction.Restriction.FauI.dna
Bio.Restriction.Restriction.FauNDI.dna
Bio.Restriction.Restriction.FbaI.dna
Bio.Restriction.Restriction.FblI.dna
Bio.Restriction.Restriction.Fco1691IV.dna
Bio.Restriction.Restriction.FinI.dna
Bio.Restriction.Restriction.FmuI.dna
Bio.Restriction.Restriction.Fnu4HI.dna
Bio.Restriction.Restriction.FnuDII.dna
Bio.Restriction.Restriction.FokI.dna
Bio.Restriction.Restriction.FriOI.dna
Bio.Restriction.Restriction.FseI.dna
Bio.Restriction.Restriction.Fsp4HI.dna
Bio.Restriction.Restriction.FspAI.dna
Bio.Restriction.Restriction.FspBI.dna
Bio.Restriction.Restriction.FspEI.dna
Bio.Restriction.Restriction.FspI.dna
Bio.Restriction.Restriction.FspPK15I.dna
Bio.Restriction.Restriction.FtnUV.dna
Bio.Restriction.Restriction.GauT27I.dna
Bio.Restriction.Restriction.Gba708II.dna
Bio.Restriction.Restriction.GdiII.dna
Bio.Restriction.Restriction.GlaI.dna
Bio.Restriction.Restriction.GluI.dna
Bio.Restriction.Restriction.GsaI.dna
Bio.Restriction.Restriction.GsuI.dna
Bio.Restriction.Restriction.HaeI.dna
Bio.Restriction.Restriction.HaeII.dna
Bio.Restriction.Restriction.HaeIII.dna
Bio.Restriction.Restriction.HapII.dna
Bio.Restriction.Restriction.HauII.dna
Bio.Restriction.Restriction.HbaII.dna
Bio.Restriction.Restriction.HdeNY26I.dna
Bio.Restriction.Restriction.HdeZA17I.dna
Bio.Restriction.Restriction.HgaI.dna
Bio.Restriction.Restriction.HgiAI.dna
Bio.Restriction.Restriction.HgiCI.dna
Bio.Restriction.Restriction.HgiEII.dna
Bio.Restriction.Restriction.HgiJII.dna
Bio.Restriction.Restriction.HhaI.dna
Bio.Restriction.Restriction.Hin1I.dna
Bio.Restriction.Restriction.Hin1II.dna
Bio.Restriction.Restriction.Hin4I.dna
Bio.Restriction.Restriction.Hin4II.dna
Bio.Restriction.Restriction.Hin6I.dna
Bio.Restriction.Restriction.HinP1I.dna
Bio.Restriction.Restriction.HincII.dna
Bio.Restriction.Restriction.HindII.dna
Bio.Restriction.Restriction.HindIII.dna
Bio.Restriction.Restriction.HinfI.dna
Bio.Restriction.Restriction.HpaI.dna
Bio.Restriction.Restriction.HpaII.dna
Bio.Restriction.Restriction.HphI.dna
Bio.Restriction.Restriction.Hpy166II.dna
Bio.Restriction.Restriction.Hpy178III.dna
Bio.Restriction.Restriction.Hpy188I.dna
Bio.Restriction.Restriction.Hpy188III.dna
Bio.Restriction.Restriction.Hpy300XI.dna
Bio.Restriction.Restriction.Hpy8I.dna
Bio.Restriction.Restriction.Hpy99I.dna
Bio.Restriction.Restriction.Hpy99XIII.dna
Bio.Restriction.Restriction.Hpy99XIV.dna
Bio.Restriction.Restriction.Hpy99XIV_mut1.dna
Bio.Restriction.Restriction.Hpy99XXII.dna
Bio.Restriction.Restriction.HpyAV.dna
Bio.Restriction.Restriction.HpyAXIV.dna
Bio.Restriction.Restriction.HpyAXVIII.dna
Bio.Restriction.Restriction.HpyAXVI_mut1.dna
Bio.Restriction.Restriction.HpyAXVI_mut2.dna
Bio.Restriction.Restriction.HpyCH4III.dna
Bio.Restriction.Restriction.HpyCH4IV.dna
Bio.Restriction.Restriction.HpyCH4V.dna
Bio.Restriction.Restriction.HpyF10VI.dna
Bio.Restriction.Restriction.HpyF3I.dna
Bio.Restriction.Restriction.HpySE526I.dna
Bio.Restriction.Restriction.HpyUM032XIII.dna
Bio.Restriction.Restriction.HpyUM032XIII_mut1.dna
Bio.Restriction.Restriction.HpyUM032XIV.dna
Bio.Restriction.Restriction.HpyUM037X.dna
Bio.Restriction.Restriction.Hsp92I.dna
Bio.Restriction.Restriction.Hsp92II.dna
Bio.Restriction.Restriction.HspAI.dna
Bio.Restriction.Restriction.HspMHR1II.dna
Bio.Restriction.Restriction.Jma19592I.dna
Bio.Restriction.Restriction.Jma19592II.dna
Bio.Restriction.Restriction.Jsp2502II.dna
Bio.Restriction.Restriction.KasI.dna
Bio.Restriction.Restriction.KflI.dna
Bio.Restriction.Restriction.Kor51II.dna
Bio.Restriction.Restriction.Kpn156V.dna
Bio.Restriction.Restriction.Kpn2I.dna
Bio.Restriction.Restriction.Kpn327I.dna
Bio.Restriction.Restriction.KpnI.dna
Bio.Restriction.Restriction.KpnNH25III.dna
Bio.Restriction.Restriction.KpnNIH30III.dna
Bio.Restriction.Restriction.KpnNIH50I.dna
Bio.Restriction.Restriction.KroI.dna
Bio.Restriction.Restriction.Ksp22I.dna
Bio.Restriction.Restriction.Ksp632I.dna
Bio.Restriction.Restriction.KspAI.dna
Bio.Restriction.Restriction.KspI.dna
Bio.Restriction.Restriction.Kzo9I.dna
Bio.Restriction.Restriction.Lba2029III.dna
Bio.Restriction.Restriction.Lde4408II.dna
Bio.Restriction.Restriction.LguI.dna
Bio.Restriction.Restriction.LlaG50I.dna
Bio.Restriction.Restriction.LmnI.dna
Bio.Restriction.Restriction.Lmo370I.dna
Bio.Restriction.Restriction.Lmo911II.dna
Bio.Restriction.Restriction.Lpl1004II.dna
Bio.Restriction.Restriction.LpnI.dna
Bio.Restriction.Restriction.LpnPI.dna
Bio.Restriction.Restriction.Lra68I.dna
Bio.Restriction.Restriction.LsaDS4I.dna
Bio.Restriction.Restriction.Lsp1109I.dna
Bio.Restriction.Restriction.Lsp48III.dna
Bio.Restriction.Restriction.Lsp6406VI.dna
Bio.Restriction.Restriction.LweI.dna
Bio.Restriction.Restriction.MabI.dna
Bio.Restriction.Restriction.MaeI.dna
Bio.Restriction.Restriction.MaeII.dna
Bio.Restriction.Restriction.MaeIII.dna
Bio.Restriction.Restriction.MalI.dna
Bio.Restriction.Restriction.MaqI.dna
Bio.Restriction.Restriction.MauBI.dna
Bio.Restriction.Restriction.Mba11I.dna
Bio.Restriction.Restriction.MbiI.dna
Bio.Restriction.Restriction.MboI.dna
Bio.Restriction.Restriction.MboII.dna
Bio.Restriction.Restriction.McaTI.dna
Bio.Restriction.Restriction.Mcr10I.dna
Bio.Restriction.Restriction.McrI.dna
Bio.Restriction.Restriction.MfeI.dna
Bio.Restriction.Restriction.MflI.dna
Bio.Restriction.Restriction.MhlI.dna
Bio.Restriction.Restriction.MjaIV.dna
Bio.Restriction.Restriction.MkaDII.dna
Bio.Restriction.Restriction.MlsI.dna
Bio.Restriction.Restriction.Mlu211III.dna
Bio.Restriction.Restriction.MluCI.dna
Bio.Restriction.Restriction.MluI.dna
Bio.Restriction.Restriction.MluNI.dna
Bio.Restriction.Restriction.Mly113I.dna
Bio.Restriction.Restriction.MlyI.dna
Bio.Restriction.Restriction.MmeI.dna
Bio.Restriction.Restriction.MnlI.dna
Bio.Restriction.Restriction.Mox20I.dna
Bio.Restriction.Restriction.Mph1103I.dna
Bio.Restriction.Restriction.MreI.dna
Bio.Restriction.Restriction.MroI.dna
Bio.Restriction.Restriction.MroNI.dna
Bio.Restriction.Restriction.MroXI.dna
Bio.Restriction.Restriction.MscI.dna
Bio.Restriction.Restriction.MseI.dna
Bio.Restriction.Restriction.MslI.dna
Bio.Restriction.Restriction.Msp20I.dna
Bio.Restriction.Restriction.MspA1I.dna
Bio.Restriction.Restriction.MspCI.dna
Bio.Restriction.Restriction.MspGI.dna
Bio.Restriction.Restriction.MspI.dna
Bio.Restriction.Restriction.MspI7II.dna
Bio.Restriction.Restriction.MspI7IV.dna
Bio.Restriction.Restriction.MspJI.dna
Bio.Restriction.Restriction.MspR9I.dna
Bio.Restriction.Restriction.MspSC27II.dna
Bio.Restriction.Restriction.MssI.dna
Bio.Restriction.Restriction.MstI.dna
Bio.Restriction.Restriction.MteI.dna
Bio.Restriction.Restriction.MtuHN878II.dna
Bio.Restriction.Restriction.MunI.dna
Bio.Restriction.Restriction.Mva1269I.dna
Bio.Restriction.Restriction.MvaI.dna
Bio.Restriction.Restriction.MvnI.dna
Bio.Restriction.Restriction.MwoI.dna
Bio.Restriction.Restriction.NaeI.dna
Bio.Restriction.Restriction.Nal45188II.dna
Bio.Restriction.Restriction.NarI.dna
Bio.Restriction.Restriction.Nbr128II.dna
Bio.Restriction.Restriction.NciI.dna
Bio.Restriction.Restriction.NcoI.dna
Bio.Restriction.Restriction.NdeI.dna
Bio.Restriction.Restriction.NdeII.dna
Bio.Restriction.Restriction.NgoAVII.dna
Bio.Restriction.Restriction.NgoAVIII.dna
Bio.Restriction.Restriction.NgoMIV.dna
Bio.Restriction.Restriction.NhaXI.dna
Bio.Restriction.Restriction.NheI.dna
Bio.Restriction.Restriction.NhoI.dna
Bio.Restriction.Restriction.NlaCI.dna
Bio.Restriction.Restriction.NlaIII.dna
Bio.Restriction.Restriction.NlaIV.dna
Bio.Restriction.Restriction.Nli3877I.dna
Bio.Restriction.Restriction.NmeA6CIII.dna
Bio.Restriction.Restriction.NmeAIII.dna
Bio.Restriction.Restriction.NmeDI.dna
Bio.Restriction.Restriction.NmuCI.dna
Bio.Restriction.Restriction.NotI.dna
Bio.Restriction.Restriction.NpeUS61II.dna
Bio.Restriction.Restriction.NruI.dna
Bio.Restriction.Restriction.NsbI.dna
Bio.Restriction.Restriction.NsiI.dna
Bio.Restriction.Restriction.NspBII.dna
Bio.Restriction.Restriction.NspI.dna
Bio.Restriction.Restriction.NspV.dna
Bio.Restriction.Restriction.ObaBS10I.dna
Bio.Restriction.Restriction.OgrI.dna
Bio.Restriction.Restriction.OliI.dna
Bio.Restriction.Restriction.OspHL35III.dna
Bio.Restriction.Restriction.PabI.dna
Bio.Restriction.Restriction.Pac19842II.dna
Bio.Restriction.Restriction.PacI.dna
Bio.Restriction.Restriction.PacIII.dna
Bio.Restriction.Restriction.PaeI.dna
Bio.Restriction.Restriction.PaeR7I.dna
Bio.Restriction.Restriction.PagI.dna
Bio.Restriction.Restriction.Pal408I.dna
Bio.Restriction.Restriction.PalAI.dna
Bio.Restriction.Restriction.PasI.dna
Bio.Restriction.Restriction.PauI.dna
Bio.Restriction.Restriction.Pba2294I.dna
Bio.Restriction.Restriction.PcaII.dna
Bio.Restriction.Restriction.PceI.dna
Bio.Restriction.Restriction.PciI.dna
Bio.Restriction.Restriction.PciSI.dna
Bio.Restriction.Restriction.Pcr308II.dna
Bio.Restriction.Restriction.PcsI.dna
Bio.Restriction.Restriction.PctI.dna
Bio.Restriction.Restriction.Pdi8503III.dna
Bio.Restriction.Restriction.PdiI.dna
Bio.Restriction.Restriction.PdmI.dna
Bio.Restriction.Restriction.Pdu1735I.dna
Bio.Restriction.Restriction.PenI.dna
Bio.Restriction.Restriction.PfeI.dna
Bio.Restriction.Restriction.Pfl1108I.dna
Bio.Restriction.Restriction.Pfl23II.dna
Bio.Restriction.Restriction.Pfl8569I.dna
Bio.Restriction.Restriction.PflFI.dna
Bio.Restriction.Restriction.PflMI.dna
Bio.Restriction.Restriction.PflPt14I.dna
Bio.Restriction.Restriction.PfoI.dna
Bio.Restriction.Restriction.PfrJS12IV.dna
Bio.Restriction.Restriction.PfrJS12V.dna
Bio.Restriction.Restriction.PfrJS15III.dna
Bio.Restriction.Restriction.Pin17FIII.dna
Bio.Restriction.Restriction.PinAI.dna
Bio.Restriction.Restriction.PinP23II.dna
Bio.Restriction.Restriction.PinP59III.dna
Bio.Restriction.Restriction.PkrI.dna
Bio.Restriction.Restriction.PlaDI.dna
Bio.Restriction.Restriction.Ple19I.dna
Bio.Restriction.Restriction.PleI.dna
Bio.Restriction.Restriction.PliMI.dna
Bio.Restriction.Restriction.PluTI.dna
Bio.Restriction.Restriction.PmaCI.dna
Bio.Restriction.Restriction.PmeI.dna
Bio.Restriction.Restriction.PmlI.dna
Bio.Restriction.Restriction.PpiI.dna
Bio.Restriction.Restriction.PpiP13II.dna
Bio.Restriction.Restriction.PpsI.dna
Bio.Restriction.Restriction.Ppu10I.dna
Bio.Restriction.Restriction.Ppu21I.dna
Bio.Restriction.Restriction.PpuMI.dna
Bio.Restriction.Restriction.PscI.dna
Bio.Restriction.Restriction.Pse18267I.dna
Bio.Restriction.Restriction.PshAI.dna
Bio.Restriction.Restriction.PshBI.dna
Bio.Restriction.Restriction.PsiI.dna
Bio.Restriction.Restriction.Psp0357II.dna
Bio.Restriction.Restriction.Psp03I.dna
Bio.Restriction.Restriction.Psp124BI.dna
Bio.Restriction.Restriction.Psp1406I.dna
Bio.Restriction.Restriction.Psp5II.dna
Bio.Restriction.Restriction.Psp6I.dna
Bio.Restriction.Restriction.PspCI.dna
Bio.Restriction.Restriction.PspEI.dna
Bio.Restriction.Restriction.PspFI.dna
Bio.Restriction.Restriction.PspGI.dna
Bio.Restriction.Restriction.PspLI.dna
Bio.Restriction.Restriction.PspN4I.dna
Bio.Restriction.Restriction.PspOMI.dna
Bio.Restriction.Restriction.PspOMII.dna
Bio.Restriction.Restriction.PspPI.dna
Bio.Restriction.Restriction.PspPPI.dna
Bio.Restriction.Restriction.PspPRI.dna
Bio.Restriction.Restriction.PspXI.dna
Bio.Restriction.Restriction.PsrI.dna
Bio.Restriction.Restriction.PssI.dna
Bio.Restriction.Restriction.Pst14472I.dna
Bio.Restriction.Restriction.Pst145I.dna
Bio.Restriction.Restriction.Pst273I.dna
Bio.Restriction.Restriction.PstI.dna
Bio.Restriction.Restriction.PstNI.dna
Bio.Restriction.Restriction.PsuGI.dna
Bio.Restriction.Restriction.PsuI.dna
Bio.Restriction.Restriction.PsyI.dna
Bio.Restriction.Restriction.PteI.dna
Bio.Restriction.Restriction.PvuI.dna
Bio.Restriction.Restriction.PvuII.dna
Bio.Restriction.Restriction.Rba2021I.dna
Bio.Restriction.Restriction.RceI.dna
Bio.Restriction.Restriction.RdeGBI.dna
Bio.Restriction.Restriction.RdeGBII.dna
Bio.Restriction.Restriction.RdeGBIII.dna
Bio.Restriction.Restriction.RflFIII.dna
Bio.Restriction.Restriction.RgaI.dna
Bio.Restriction.Restriction.RigI.dna
Bio.Restriction.Restriction.RlaI.dna
Bio.Restriction.Restriction.RlaII.dna
Bio.Restriction.Restriction.RleAI.dna
Bio.Restriction.Restriction.Rmu369III.dna
Bio.Restriction.Restriction.RpaB5I.dna
Bio.Restriction.Restriction.RpaBI.dna
Bio.Restriction.Restriction.RpaI.dna
Bio.Restriction.Restriction.RpaTI.dna
Bio.Restriction.Restriction.RruI.dna
Bio.Restriction.Restriction.RsaI.dna
Bio.Restriction.Restriction.RsaNI.dna
Bio.Restriction.Restriction.RseI.dna
Bio.Restriction.Restriction.Rsp008IV.dna
Bio.Restriction.Restriction.Rsp008V.dna
Bio.Restriction.Restriction.Rsp531II.dna
Bio.Restriction.Restriction.RspPBTS2III.dna
Bio.Restriction.Restriction.Rsr2I.dna
Bio.Restriction.Restriction.RsrII.dna
Bio.Restriction.Restriction.Rtr1953I.dna
Bio.Restriction.Restriction.SacI.dna
Bio.Restriction.Restriction.SacII.dna
Bio.Restriction.Restriction.Saf8902III.dna
Bio.Restriction.Restriction.Sag901I.dna
Bio.Restriction.Restriction.SalI.dna
Bio.Restriction.Restriction.SanDI.dna
Bio.Restriction.Restriction.SapI.dna
Bio.Restriction.Restriction.SaqAI.dna
Bio.Restriction.Restriction.SatI.dna
Bio.Restriction.Restriction.Sau3AI.dna
Bio.Restriction.Restriction.Sau96I.dna
Bio.Restriction.Restriction.SauI.dna
Bio.Restriction.Restriction.Sba460II.dna
Bio.Restriction.Restriction.SbfI.dna
Bio.Restriction.Restriction.Sbo46I.dna
Bio.Restriction.Restriction.ScaI.dna
Bio.Restriction.Restriction.SchI.dna
Bio.Restriction.Restriction.SciI.dna
Bio.Restriction.Restriction.ScoDS2II.dna
Bio.Restriction.Restriction.ScrFI.dna
Bio.Restriction.Restriction.SdaI.dna
Bio.Restriction.Restriction.SdeAI.dna
Bio.Restriction.Restriction.SdeOSI.dna
Bio.Restriction.Restriction.SduI.dna
Bio.Restriction.Restriction.SecI.dna
Bio.Restriction.Restriction.SelI.dna
Bio.Restriction.Restriction.Sen17963III.dna
Bio.Restriction.Restriction.SenA1673III.dna
Bio.Restriction.Restriction.SenSARA26III.dna
Bio.Restriction.Restriction.SenTFIV.dna
Bio.Restriction.Restriction.SetI.dna
Bio.Restriction.Restriction.SexAI.dna
Bio.Restriction.Restriction.SfaAI.dna
Bio.Restriction.Restriction.SfaNI.dna
Bio.Restriction.Restriction.SfcI.dna
Bio.Restriction.Restriction.SfeI.dna
Bio.Restriction.Restriction.SfiI.dna
Bio.Restriction.Restriction.SfoI.dna
Bio.Restriction.Restriction.Sfr274I.dna
Bio.Restriction.Restriction.Sfr303I.dna
Bio.Restriction.Restriction.SfuI.dna
Bio.Restriction.Restriction.SgeI.dna
Bio.Restriction.Restriction.SgfI.dna
Bio.Restriction.Restriction.SgrAI.dna
Bio.Restriction.Restriction.SgrBI.dna
Bio.Restriction.Restriction.SgrDI.dna
Bio.Restriction.Restriction.SgrTI.dna
Bio.Restriction.Restriction.SgsI.dna
Bio.Restriction.Restriction.SimI.dna
Bio.Restriction.Restriction.SinI.dna
Bio.Restriction.Restriction.SlaI.dna
Bio.Restriction.Restriction.SmaI.dna
Bio.Restriction.Restriction.SmaUMH5I.dna
Bio.Restriction.Restriction.SmaUMH8I.dna
Bio.Restriction.Restriction.SmiI.dna
Bio.Restriction.Restriction.SmiMI.dna
Bio.Restriction.Restriction.SmlI.dna
Bio.Restriction.Restriction.SmoI.dna
Bio.Restriction.Restriction.SnaBI.dna
Bio.Restriction.Restriction.SnaI.dna
Bio.Restriction.Restriction.Sno506I.dna
Bio.Restriction.Restriction.SpeI.dna
Bio.Restriction.Restriction.SphI.dna
Bio.Restriction.Restriction.SplI.dna
Bio.Restriction.Restriction.SpnRII.dna
Bio.Restriction.Restriction.SpoDI.dna
Bio.Restriction.Restriction.SrfI.dna
Bio.Restriction.Restriction.Sse232I.dna
Bio.Restriction.Restriction.Sse8387I.dna
Bio.Restriction.Restriction.Sse8647I.dna
Bio.Restriction.Restriction.Sse9I.dna
Bio.Restriction.Restriction.SseBI.dna
Bio.Restriction.Restriction.SsiI.dna
Bio.Restriction.Restriction.Ssp6803IV.dna
Bio.Restriction.Restriction.Ssp714II.dna
Bio.Restriction.Restriction.SspD5I.dna
Bio.Restriction.Restriction.SspDI.dna
Bio.Restriction.Restriction.SspI.dna
Bio.Restriction.Restriction.SspMI.dna
Bio.Restriction.Restriction.SstE37I.dna
Bio.Restriction.Restriction.SstI.dna
Bio.Restriction.Restriction.Sth132I.dna
Bio.Restriction.Restriction.Sth20745III.dna
Bio.Restriction.Restriction.Sth302II.dna
Bio.Restriction.Restriction.SthSt3II.dna
Bio.Restriction.Restriction.StsI.dna
Bio.Restriction.Restriction.StuI.dna
Bio.Restriction.Restriction.StyD4I.dna
Bio.Restriction.Restriction.StyI.dna
Bio.Restriction.Restriction.SurP32aII.dna
Bio.Restriction.Restriction.SwaI.dna
Bio.Restriction.Restriction.TaaI.dna
Bio.Restriction.Restriction.TaiI.dna
Bio.Restriction.Restriction.TaqI.dna
Bio.Restriction.Restriction.TaqII.dna
Bio.Restriction.Restriction.TaqIII.dna
Bio.Restriction.Restriction.TasI.dna
Bio.Restriction.Restriction.TatI.dna
Bio.Restriction.Restriction.TauI.dna
Bio.Restriction.Restriction.TfiI.dna
Bio.Restriction.Restriction.TpyTP2I.dna
Bio.Restriction.Restriction.Tru1I.dna
Bio.Restriction.Restriction.Tru9I.dna
Bio.Restriction.Restriction.TscAI.dna
Bio.Restriction.Restriction.TseFI.dna
Bio.Restriction.Restriction.TseI.dna
Bio.Restriction.Restriction.TsoI.dna
Bio.Restriction.Restriction.Tsp45I.dna
Bio.Restriction.Restriction.Tsp4CI.dna
Bio.Restriction.Restriction.TspARh3I.dna
Bio.Restriction.Restriction.TspDTI.dna
Bio.Restriction.Restriction.TspEI.dna
Bio.Restriction.Restriction.TspGWI.dna
Bio.Restriction.Restriction.TspMI.dna
Bio.Restriction.Restriction.TspRI.dna
Bio.Restriction.Restriction.TssI.dna
Bio.Restriction.Restriction.TstI.dna
Bio.Restriction.Restriction.TsuI.dna
Bio.Restriction.Restriction.Tth111I.dna
Bio.Restriction.Restriction.Tth111II.dna
Bio.Restriction.Restriction.UbaF11I.dna
Bio.Restriction.Restriction.UbaF12I.dna
Bio.Restriction.Restriction.UbaF13I.dna
Bio.Restriction.Restriction.UbaF14I.dna
Bio.Restriction.Restriction.UbaF9I.dna
Bio.Restriction.Restriction.UbaPI.dna
Bio.Restriction.Restriction.UcoMSI.dna
Bio.Restriction.Restriction.UnbI.dna
Bio.Restriction.Restriction.Van9116I.dna
Bio.Restriction.Restriction.Van91I.dna
Bio.Restriction.Restriction.VchE4II.dna
Bio.Restriction.Restriction.Vdi96II.dna
Bio.Restriction.Restriction.Vha464I.dna
Bio.Restriction.Restriction.VneI.dna
Bio.Restriction.Restriction.VpaK11AI.dna
Bio.Restriction.Restriction.VpaK11BI.dna
Bio.Restriction.Restriction.VspI.dna
Bio.Restriction.Restriction.Vtu19109I.dna
Bio.Restriction.Restriction.WviI.dna
Bio.Restriction.Restriction.XagI.dna
Bio.Restriction.Restriction.XapI.dna
Bio.Restriction.Restriction.XbaI.dna
Bio.Restriction.Restriction.Xca85IV.dna
Bio.Restriction.Restriction.XceI.dna
Bio.Restriction.Restriction.XcmI.dna
Bio.Restriction.Restriction.XhoI.dna
Bio.Restriction.Restriction.XhoII.dna
Bio.Restriction.Restriction.XmaI.dna
Bio.Restriction.Restriction.XmaIII.dna
Bio.Restriction.Restriction.XmaJI.dna
Bio.Restriction.Restriction.XmiI.dna
Bio.Restriction.Restriction.XmnI.dna
Bio.Restriction.Restriction.XspI.dna
Bio.Restriction.Restriction.YkrI.dna
Bio.Restriction.Restriction.ZraI.dna
Bio.Restriction.Restriction.ZrmI.dna
Bio.Restriction.Restriction.Zsp2I.dna" class="py-name" href="#" onclick="return doclink('link-981', 'dna', 'link-981');">dna</a></tt><tt class="py-op">,</tt> <tt id="link-982" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-982', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2611"></a><tt class="py-lineno">2611</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-983" class="py-name"><a title="Bio.Restriction.Restriction.AanI.dna
Bio.Restriction.Restriction.AarI.dna
Bio.Restriction.Restriction.AasI.dna
Bio.Restriction.Restriction.AatII.dna
Bio.Restriction.Restriction.Aba6411II.dna
Bio.Restriction.Restriction.AbaB8342IV.dna
Bio.Restriction.Restriction.AbaCIII.dna
Bio.Restriction.Restriction.AbaSI.dna
Bio.Restriction.Restriction.AbaUMB2I.dna
Bio.Restriction.Restriction.AbsI.dna
Bio.Restriction.Restriction.Acc16I.dna
Bio.Restriction.Restriction.Acc36I.dna
Bio.Restriction.Restriction.Acc65I.dna
Bio.Restriction.Restriction.Acc65V.dna
Bio.Restriction.Restriction.AccB1I.dna
Bio.Restriction.Restriction.AccB7I.dna
Bio.Restriction.Restriction.AccBSI.dna
Bio.Restriction.Restriction.AccI.dna
Bio.Restriction.Restriction.AccII.dna
Bio.Restriction.Restriction.AccIII.dna
Bio.Restriction.Restriction.AceIII.dna
Bio.Restriction.Restriction.AchA6III.dna
Bio.Restriction.Restriction.AciI.dna
Bio.Restriction.Restriction.AclI.dna
Bio.Restriction.Restriction.AclWI.dna
Bio.Restriction.Restriction.Aco12261II.dna
Bio.Restriction.Restriction.AcoI.dna
Bio.Restriction.Restriction.AcoY31II.dna
Bio.Restriction.Restriction.AcsI.dna
Bio.Restriction.Restriction.AcuI.dna
Bio.Restriction.Restriction.AcvI.dna
Bio.Restriction.Restriction.AcyI.dna
Bio.Restriction.Restriction.AdeI.dna
Bio.Restriction.Restriction.Adh6U21I.dna
Bio.Restriction.Restriction.AfaI.dna
Bio.Restriction.Restriction.AfeI.dna
Bio.Restriction.Restriction.AfiI.dna
Bio.Restriction.Restriction.AflII.dna
Bio.Restriction.Restriction.AflIII.dna
Bio.Restriction.Restriction.AgeI.dna
Bio.Restriction.Restriction.AgsI.dna
Bio.Restriction.Restriction.AhaIII.dna
Bio.Restriction.Restriction.AhdI.dna
Bio.Restriction.Restriction.AhlI.dna
Bio.Restriction.Restriction.AhyRBAHI.dna
Bio.Restriction.Restriction.AhyYL17I.dna
Bio.Restriction.Restriction.AjiI.dna
Bio.Restriction.Restriction.AjnI.dna
Bio.Restriction.Restriction.AjuI.dna
Bio.Restriction.Restriction.AleI.dna
Bio.Restriction.Restriction.AlfI.dna
Bio.Restriction.Restriction.AloI.dna
Bio.Restriction.Restriction.AluBI.dna
Bio.Restriction.Restriction.AluI.dna
Bio.Restriction.Restriction.Alw21I.dna
Bio.Restriction.Restriction.Alw26I.dna
Bio.Restriction.Restriction.Alw44I.dna
Bio.Restriction.Restriction.AlwFI.dna
Bio.Restriction.Restriction.AlwI.dna
Bio.Restriction.Restriction.AlwNI.dna
Bio.Restriction.Restriction.Ama87I.dna
Bio.Restriction.Restriction.AmaCSI.dna
Bio.Restriction.Restriction.Aod1I.dna
Bio.Restriction.Restriction.Aor13HI.dna
Bio.Restriction.Restriction.Aor51HI.dna
Bio.Restriction.Restriction.AoxI.dna
Bio.Restriction.Restriction.ApaBI.dna
Bio.Restriction.Restriction.ApaI.dna
Bio.Restriction.Restriction.ApaLI.dna
Bio.Restriction.Restriction.ApeKI.dna
Bio.Restriction.Restriction.ApoI.dna
Bio.Restriction.Restriction.ApyPI.dna
Bio.Restriction.Restriction.AquII.dna
Bio.Restriction.Restriction.AquIII.dna
Bio.Restriction.Restriction.AquIV.dna
Bio.Restriction.Restriction.ArsI.dna
Bio.Restriction.Restriction.AscI.dna
Bio.Restriction.Restriction.AseI.dna
Bio.Restriction.Restriction.Asi256I.dna
Bio.Restriction.Restriction.AsiGI.dna
Bio.Restriction.Restriction.AsiSI.dna
Bio.Restriction.Restriction.Asp103I.dna
Bio.Restriction.Restriction.Asp114pII.dna
Bio.Restriction.Restriction.Asp700I.dna
Bio.Restriction.Restriction.Asp718I.dna
Bio.Restriction.Restriction.AspA2I.dna
Bio.Restriction.Restriction.AspBHI.dna
Bio.Restriction.Restriction.AspDUT2V.dna
Bio.Restriction.Restriction.AspJHL3II.dna
Bio.Restriction.Restriction.AspLEI.dna
Bio.Restriction.Restriction.AspNIH4III.dna
Bio.Restriction.Restriction.AspS9I.dna
Bio.Restriction.Restriction.AspSLV7III.dna
Bio.Restriction.Restriction.Asu14238IV.dna
Bio.Restriction.Restriction.AsuC2I.dna
Bio.Restriction.Restriction.AsuHPI.dna
Bio.Restriction.Restriction.AsuI.dna
Bio.Restriction.Restriction.AsuII.dna
Bio.Restriction.Restriction.AsuNHI.dna
Bio.Restriction.Restriction.AteTI.dna
Bio.Restriction.Restriction.AvaI.dna
Bio.Restriction.Restriction.AvaII.dna
Bio.Restriction.Restriction.AvaIII.dna
Bio.Restriction.Restriction.AvrII.dna
Bio.Restriction.Restriction.Awo1030IV.dna
Bio.Restriction.Restriction.AxyI.dna
Bio.Restriction.Restriction.BaeGI.dna
Bio.Restriction.Restriction.BaeI.dna
Bio.Restriction.Restriction.Bag18758I.dna
Bio.Restriction.Restriction.BalI.dna
Bio.Restriction.Restriction.BamHI.dna
Bio.Restriction.Restriction.BanI.dna
Bio.Restriction.Restriction.BanII.dna
Bio.Restriction.Restriction.BanLI.dna
Bio.Restriction.Restriction.BarI.dna
Bio.Restriction.Restriction.Bau1417V.dna
Bio.Restriction.Restriction.BauI.dna
Bio.Restriction.Restriction.Bbr52II.dna
Bio.Restriction.Restriction.Bbr57III.dna
Bio.Restriction.Restriction.Bbr7017II.dna
Bio.Restriction.Restriction.Bbr7017III.dna
Bio.Restriction.Restriction.BbrPI.dna
Bio.Restriction.Restriction.BbsI.dna
Bio.Restriction.Restriction.BbuB31I.dna
Bio.Restriction.Restriction.BbuB31II.dna
Bio.Restriction.Restriction.Bbv12I.dna
Bio.Restriction.Restriction.BbvCI.dna
Bio.Restriction.Restriction.BbvI.dna
Bio.Restriction.Restriction.BbvII.dna
Bio.Restriction.Restriction.BccI.dna
Bio.Restriction.Restriction.Bce3081I.dna
Bio.Restriction.Restriction.Bce83I.dna
Bio.Restriction.Restriction.BceAI.dna
Bio.Restriction.Restriction.BceSIV.dna
Bio.Restriction.Restriction.BcefI.dna
Bio.Restriction.Restriction.BcgI.dna
Bio.Restriction.Restriction.BciT130I.dna
Bio.Restriction.Restriction.BciVI.dna
Bio.Restriction.Restriction.BclI.dna
Bio.Restriction.Restriction.BcnI.dna
Bio.Restriction.Restriction.BcoDI.dna
Bio.Restriction.Restriction.BcuI.dna
Bio.Restriction.Restriction.BdaI.dna
Bio.Restriction.Restriction.BetI.dna
Bio.Restriction.Restriction.BfaI.dna
Bio.Restriction.Restriction.BfaSII.dna
Bio.Restriction.Restriction.BfiI.dna
Bio.Restriction.Restriction.BfmI.dna
Bio.Restriction.Restriction.BfoI.dna
Bio.Restriction.Restriction.BfrI.dna
Bio.Restriction.Restriction.BfuAI.dna
Bio.Restriction.Restriction.BfuI.dna
Bio.Restriction.Restriction.Bga514I.dna
Bio.Restriction.Restriction.BglI.dna
Bio.Restriction.Restriction.BglII.dna
Bio.Restriction.Restriction.BinI.dna
Bio.Restriction.Restriction.BisI.dna
Bio.Restriction.Restriction.BkrAM31DI.dna
Bio.Restriction.Restriction.Ble402II.dna
Bio.Restriction.Restriction.BlnI.dna
Bio.Restriction.Restriction.BloAII.dna
Bio.Restriction.Restriction.BlpI.dna
Bio.Restriction.Restriction.BlsI.dna
Bio.Restriction.Restriction.BmcAI.dna
Bio.Restriction.Restriction.Bme1390I.dna
Bio.Restriction.Restriction.Bme18I.dna
Bio.Restriction.Restriction.BmeDI.dna
Bio.Restriction.Restriction.BmeRI.dna
Bio.Restriction.Restriction.BmeT110I.dna
Bio.Restriction.Restriction.BmgBI.dna
Bio.Restriction.Restriction.BmgI.dna
Bio.Restriction.Restriction.BmgT120I.dna
Bio.Restriction.Restriction.BmiI.dna
Bio.Restriction.Restriction.BmrFI.dna
Bio.Restriction.Restriction.BmrI.dna
Bio.Restriction.Restriction.BmsI.dna
Bio.Restriction.Restriction.BmtI.dna
Bio.Restriction.Restriction.BmuI.dna
Bio.Restriction.Restriction.BoxI.dna
Bio.Restriction.Restriction.BpiI.dna
Bio.Restriction.Restriction.BplI.dna
Bio.Restriction.Restriction.BpmI.dna
Bio.Restriction.Restriction.Bpu10I.dna
Bio.Restriction.Restriction.Bpu1102I.dna
Bio.Restriction.Restriction.Bpu14I.dna
Bio.Restriction.Restriction.BpuEI.dna
Bio.Restriction.Restriction.BpuMI.dna
Bio.Restriction.Restriction.Bsa29I.dna
Bio.Restriction.Restriction.BsaAI.dna
Bio.Restriction.Restriction.BsaBI.dna
Bio.Restriction.Restriction.BsaHI.dna
Bio.Restriction.Restriction.BsaI.dna
Bio.Restriction.Restriction.BsaJI.dna
Bio.Restriction.Restriction.BsaWI.dna
Bio.Restriction.Restriction.BsaXI.dna
Bio.Restriction.Restriction.BsbI.dna
Bio.Restriction.Restriction.Bsc4I.dna
Bio.Restriction.Restriction.BscAI.dna
Bio.Restriction.Restriction.BscGI.dna
Bio.Restriction.Restriction.Bse118I.dna
Bio.Restriction.Restriction.Bse1I.dna
Bio.Restriction.Restriction.Bse21I.dna
Bio.Restriction.Restriction.Bse3DI.dna
Bio.Restriction.Restriction.Bse8I.dna
Bio.Restriction.Restriction.BseAI.dna
Bio.Restriction.Restriction.BseBI.dna
Bio.Restriction.Restriction.BseCI.dna
Bio.Restriction.Restriction.BseDI.dna
Bio.Restriction.Restriction.BseGI.dna
Bio.Restriction.Restriction.BseJI.dna
Bio.Restriction.Restriction.BseLI.dna
Bio.Restriction.Restriction.BseMI.dna
Bio.Restriction.Restriction.BseMII.dna
Bio.Restriction.Restriction.BseNI.dna
Bio.Restriction.Restriction.BsePI.dna
Bio.Restriction.Restriction.BseRI.dna
Bio.Restriction.Restriction.BseSI.dna
Bio.Restriction.Restriction.BseX3I.dna
Bio.Restriction.Restriction.BseXI.dna
Bio.Restriction.Restriction.BseYI.dna
Bio.Restriction.Restriction.BsgI.dna
Bio.Restriction.Restriction.Bsh1236I.dna
Bio.Restriction.Restriction.Bsh1285I.dna
Bio.Restriction.Restriction.BshFI.dna
Bio.Restriction.Restriction.BshNI.dna
Bio.Restriction.Restriction.BshTI.dna
Bio.Restriction.Restriction.BshVI.dna
Bio.Restriction.Restriction.BsiEI.dna
Bio.Restriction.Restriction.BsiHKAI.dna
Bio.Restriction.Restriction.BsiHKCI.dna
Bio.Restriction.Restriction.BsiI.dna
Bio.Restriction.Restriction.BsiSI.dna
Bio.Restriction.Restriction.BsiWI.dna
Bio.Restriction.Restriction.BsiYI.dna
Bio.Restriction.Restriction.BslFI.dna
Bio.Restriction.Restriction.BslI.dna
Bio.Restriction.Restriction.BsmAI.dna
Bio.Restriction.Restriction.BsmBI.dna
Bio.Restriction.Restriction.BsmFI.dna
Bio.Restriction.Restriction.BsmI.dna
Bio.Restriction.Restriction.BsnI.dna
Bio.Restriction.Restriction.Bso31I.dna
Bio.Restriction.Restriction.BsoBI.dna
Bio.Restriction.Restriction.Bsp119I.dna
Bio.Restriction.Restriction.Bsp120I.dna
Bio.Restriction.Restriction.Bsp1286I.dna
Bio.Restriction.Restriction.Bsp13I.dna
Bio.Restriction.Restriction.Bsp1407I.dna
Bio.Restriction.Restriction.Bsp143I.dna
Bio.Restriction.Restriction.Bsp1720I.dna
Bio.Restriction.Restriction.Bsp19I.dna
Bio.Restriction.Restriction.Bsp24I.dna
Bio.Restriction.Restriction.Bsp3004IV.dna
Bio.Restriction.Restriction.Bsp460III.dna
Bio.Restriction.Restriction.Bsp68I.dna
Bio.Restriction.Restriction.BspACI.dna
Bio.Restriction.Restriction.BspANI.dna
Bio.Restriction.Restriction.BspCNI.dna
Bio.Restriction.Restriction.BspD6I.dna
Bio.Restriction.Restriction.BspDI.dna
Bio.Restriction.Restriction.BspEI.dna
Bio.Restriction.Restriction.BspFNI.dna
Bio.Restriction.Restriction.BspGI.dna
Bio.Restriction.Restriction.BspHI.dna
Bio.Restriction.Restriction.BspLI.dna
Bio.Restriction.Restriction.BspLU11I.dna
Bio.Restriction.Restriction.BspMAI.dna
Bio.Restriction.Restriction.BspMI.dna
Bio.Restriction.Restriction.BspMII.dna
Bio.Restriction.Restriction.BspNCI.dna
Bio.Restriction.Restriction.BspOI.dna
Bio.Restriction.Restriction.BspPI.dna
Bio.Restriction.Restriction.BspQI.dna
Bio.Restriction.Restriction.BspT104I.dna
Bio.Restriction.Restriction.BspT107I.dna
Bio.Restriction.Restriction.BspTI.dna
Bio.Restriction.Restriction.BspTNI.dna
Bio.Restriction.Restriction.BsrBI.dna
Bio.Restriction.Restriction.BsrDI.dna
Bio.Restriction.Restriction.BsrFI.dna
Bio.Restriction.Restriction.BsrGI.dna
Bio.Restriction.Restriction.BsrI.dna
Bio.Restriction.Restriction.BssAI.dna
Bio.Restriction.Restriction.BssECI.dna
Bio.Restriction.Restriction.BssHII.dna
Bio.Restriction.Restriction.BssMI.dna
Bio.Restriction.Restriction.BssNAI.dna
Bio.Restriction.Restriction.BssNI.dna
Bio.Restriction.Restriction.BssSI.dna
Bio.Restriction.Restriction.BssT1I.dna
Bio.Restriction.Restriction.Bst1107I.dna
Bio.Restriction.Restriction.Bst2BI.dna
Bio.Restriction.Restriction.Bst2UI.dna
Bio.Restriction.Restriction.Bst4CI.dna
Bio.Restriction.Restriction.Bst6I.dna
Bio.Restriction.Restriction.BstACI.dna
Bio.Restriction.Restriction.BstAFI.dna
Bio.Restriction.Restriction.BstAPI.dna
Bio.Restriction.Restriction.BstAUI.dna
Bio.Restriction.Restriction.BstBAI.dna
Bio.Restriction.Restriction.BstBI.dna
Bio.Restriction.Restriction.BstC8I.dna
Bio.Restriction.Restriction.BstDEI.dna
Bio.Restriction.Restriction.BstDSI.dna
Bio.Restriction.Restriction.BstEII.dna
Bio.Restriction.Restriction.BstENI.dna
Bio.Restriction.Restriction.BstF5I.dna
Bio.Restriction.Restriction.BstFNI.dna
Bio.Restriction.Restriction.BstH2I.dna
Bio.Restriction.Restriction.BstHHI.dna
Bio.Restriction.Restriction.BstKTI.dna
Bio.Restriction.Restriction.BstMAI.dna
Bio.Restriction.Restriction.BstMBI.dna
Bio.Restriction.Restriction.BstMCI.dna
Bio.Restriction.Restriction.BstMWI.dna
Bio.Restriction.Restriction.BstNI.dna
Bio.Restriction.Restriction.BstNSI.dna
Bio.Restriction.Restriction.BstPAI.dna
Bio.Restriction.Restriction.BstPI.dna
Bio.Restriction.Restriction.BstSCI.dna
Bio.Restriction.Restriction.BstSFI.dna
Bio.Restriction.Restriction.BstSLI.dna
Bio.Restriction.Restriction.BstSNI.dna
Bio.Restriction.Restriction.BstUI.dna
Bio.Restriction.Restriction.BstV1I.dna
Bio.Restriction.Restriction.BstV2I.dna
Bio.Restriction.Restriction.BstX2I.dna
Bio.Restriction.Restriction.BstXI.dna
Bio.Restriction.Restriction.BstYI.dna
Bio.Restriction.Restriction.BstZ17I.dna
Bio.Restriction.Restriction.BstZI.dna
Bio.Restriction.Restriction.Bsu15I.dna
Bio.Restriction.Restriction.Bsu36I.dna
Bio.Restriction.Restriction.BsuI.dna
Bio.Restriction.Restriction.BsuRI.dna
Bio.Restriction.Restriction.BsuTUI.dna
Bio.Restriction.Restriction.BtgI.dna
Bio.Restriction.Restriction.BtgZI.dna
Bio.Restriction.Restriction.BthCI.dna
Bio.Restriction.Restriction.BtrI.dna
Bio.Restriction.Restriction.BtsCI.dna
Bio.Restriction.Restriction.BtsI.dna
Bio.Restriction.Restriction.BtsIMutI.dna
Bio.Restriction.Restriction.BtuMI.dna
Bio.Restriction.Restriction.Bve1B23I.dna
Bio.Restriction.Restriction.BveI.dna
Bio.Restriction.Restriction.Cac8I.dna
Bio.Restriction.Restriction.CaiI.dna
Bio.Restriction.Restriction.Cal14237I.dna
Bio.Restriction.Restriction.CalB3II.dna
Bio.Restriction.Restriction.Cau10061II.dna
Bio.Restriction.Restriction.CauII.dna
Bio.Restriction.Restriction.Cba13II.dna
Bio.Restriction.Restriction.Cba16038I.dna
Bio.Restriction.Restriction.Cbo67071IV.dna
Bio.Restriction.Restriction.Cch467III.dna
Bio.Restriction.Restriction.CchII.dna
Bio.Restriction.Restriction.CchIII.dna
Bio.Restriction.Restriction.CciI.dna
Bio.Restriction.Restriction.CciNI.dna
Bio.Restriction.Restriction.Cco14983V.dna
Bio.Restriction.Restriction.Cco14983VI.dna
Bio.Restriction.Restriction.CcrNAIII.dna
Bio.Restriction.Restriction.Cdi11397I.dna
Bio.Restriction.Restriction.CdiI.dna
Bio.Restriction.Restriction.CdpI.dna
Bio.Restriction.Restriction.Cdu23823II.dna
Bio.Restriction.Restriction.CfoI.dna
Bio.Restriction.Restriction.Cfr10I.dna
Bio.Restriction.Restriction.Cfr13I.dna
Bio.Restriction.Restriction.Cfr42I.dna
Bio.Restriction.Restriction.Cfr9I.dna
Bio.Restriction.Restriction.CfrI.dna
Bio.Restriction.Restriction.CfrMH13II.dna
Bio.Restriction.Restriction.CfrMH16VI.dna
Bio.Restriction.Restriction.Cfupf3II.dna
Bio.Restriction.Restriction.Cgl13032I.dna
Bio.Restriction.Restriction.Cgl13032II.dna
Bio.Restriction.Restriction.ChaI.dna
Bio.Restriction.Restriction.Cje265V.dna
Bio.Restriction.Restriction.Cje54107III.dna
Bio.Restriction.Restriction.CjeFIII.dna
Bio.Restriction.Restriction.CjeFV.dna
Bio.Restriction.Restriction.CjeI.dna
Bio.Restriction.Restriction.CjeNII.dna
Bio.Restriction.Restriction.CjeNIII.dna
Bio.Restriction.Restriction.CjeNV.dna
Bio.Restriction.Restriction.CjeP659IV.dna
Bio.Restriction.Restriction.CjePI.dna
Bio.Restriction.Restriction.CjuI.dna
Bio.Restriction.Restriction.CjuII.dna
Bio.Restriction.Restriction.Cla11845III.dna
Bio.Restriction.Restriction.ClaI.dna
Bio.Restriction.Restriction.Cly7489II.dna
Bio.Restriction.Restriction.Cma23826I.dna
Bio.Restriction.Restriction.CpoI.dna
Bio.Restriction.Restriction.CseI.dna
Bio.Restriction.Restriction.CsiI.dna
Bio.Restriction.Restriction.Csp2014I.dna
Bio.Restriction.Restriction.Csp6I.dna
Bio.Restriction.Restriction.CspAI.dna
Bio.Restriction.Restriction.CspCI.dna
Bio.Restriction.Restriction.CspI.dna
Bio.Restriction.Restriction.CstMI.dna
Bio.Restriction.Restriction.CviAII.dna
Bio.Restriction.Restriction.CviJI.dna
Bio.Restriction.Restriction.CviKI_1.dna
Bio.Restriction.Restriction.CviQI.dna
Bio.Restriction.Restriction.CviRI.dna
Bio.Restriction.Restriction.Dde51507I.dna
Bio.Restriction.Restriction.DdeI.dna
Bio.Restriction.Restriction.DinI.dna
Bio.Restriction.Restriction.DpnI.dna
Bio.Restriction.Restriction.DpnII.dna
Bio.Restriction.Restriction.DraI.dna
Bio.Restriction.Restriction.DraII.dna
Bio.Restriction.Restriction.DraIII.dna
Bio.Restriction.Restriction.DraRI.dna
Bio.Restriction.Restriction.DrdI.dna
Bio.Restriction.Restriction.DrdII.dna
Bio.Restriction.Restriction.DrdIV.dna
Bio.Restriction.Restriction.DriI.dna
Bio.Restriction.Restriction.DsaI.dna
Bio.Restriction.Restriction.DseDI.dna
Bio.Restriction.Restriction.DvuIII.dna
Bio.Restriction.Restriction.EaeI.dna
Bio.Restriction.Restriction.EagI.dna
Bio.Restriction.Restriction.Eam1104I.dna
Bio.Restriction.Restriction.Eam1105I.dna
Bio.Restriction.Restriction.EarI.dna
Bio.Restriction.Restriction.EciI.dna
Bio.Restriction.Restriction.Ecl136II.dna
Bio.Restriction.Restriction.Ecl234I.dna
Bio.Restriction.Restriction.Ecl35734I.dna
Bio.Restriction.Restriction.EclXI.dna
Bio.Restriction.Restriction.Eco105I.dna
Bio.Restriction.Restriction.Eco130I.dna
Bio.Restriction.Restriction.Eco147I.dna
Bio.Restriction.Restriction.Eco24I.dna
Bio.Restriction.Restriction.Eco31I.dna
Bio.Restriction.Restriction.Eco32I.dna
Bio.Restriction.Restriction.Eco43896II.dna
Bio.Restriction.Restriction.Eco4465II.dna
Bio.Restriction.Restriction.Eco47I.dna
Bio.Restriction.Restriction.Eco47III.dna
Bio.Restriction.Restriction.Eco52I.dna
Bio.Restriction.Restriction.Eco53kI.dna
Bio.Restriction.Restriction.Eco57I.dna
Bio.Restriction.Restriction.Eco57MI.dna
Bio.Restriction.Restriction.Eco72I.dna
Bio.Restriction.Restriction.Eco81I.dna
Bio.Restriction.Restriction.Eco88I.dna
Bio.Restriction.Restriction.Eco91I.dna
Bio.Restriction.Restriction.EcoBLMcrX.dna
Bio.Restriction.Restriction.EcoE1140I.dna
Bio.Restriction.Restriction.EcoHI.dna
Bio.Restriction.Restriction.EcoHSI.dna
Bio.Restriction.Restriction.EcoICRI.dna
Bio.Restriction.Restriction.EcoMVII.dna
Bio.Restriction.Restriction.EcoNI.dna
Bio.Restriction.Restriction.EcoNIH6II.dna
Bio.Restriction.Restriction.EcoO109I.dna
Bio.Restriction.Restriction.EcoO65I.dna
Bio.Restriction.Restriction.EcoRI.dna
Bio.Restriction.Restriction.EcoRII.dna
Bio.Restriction.Restriction.EcoRV.dna
Bio.Restriction.Restriction.EcoT14I.dna
Bio.Restriction.Restriction.EcoT22I.dna
Bio.Restriction.Restriction.EcoT38I.dna
Bio.Restriction.Restriction.EgeI.dna
Bio.Restriction.Restriction.EheI.dna
Bio.Restriction.Restriction.Eli8509II.dna
Bio.Restriction.Restriction.ErhI.dna
Bio.Restriction.Restriction.EsaBC3I.dna
Bio.Restriction.Restriction.EsaSSI.dna
Bio.Restriction.Restriction.Esp3007I.dna
Bio.Restriction.Restriction.Esp3I.dna
Bio.Restriction.Restriction.EspI.dna
Bio.Restriction.Restriction.FaeI.dna
Bio.Restriction.Restriction.FaiI.dna
Bio.Restriction.Restriction.FalI.dna
Bio.Restriction.Restriction.FaqI.dna
Bio.Restriction.Restriction.FatI.dna
Bio.Restriction.Restriction.FauI.dna
Bio.Restriction.Restriction.FauNDI.dna
Bio.Restriction.Restriction.FbaI.dna
Bio.Restriction.Restriction.FblI.dna
Bio.Restriction.Restriction.Fco1691IV.dna
Bio.Restriction.Restriction.FinI.dna
Bio.Restriction.Restriction.FmuI.dna
Bio.Restriction.Restriction.Fnu4HI.dna
Bio.Restriction.Restriction.FnuDII.dna
Bio.Restriction.Restriction.FokI.dna
Bio.Restriction.Restriction.FriOI.dna
Bio.Restriction.Restriction.FseI.dna
Bio.Restriction.Restriction.Fsp4HI.dna
Bio.Restriction.Restriction.FspAI.dna
Bio.Restriction.Restriction.FspBI.dna
Bio.Restriction.Restriction.FspEI.dna
Bio.Restriction.Restriction.FspI.dna
Bio.Restriction.Restriction.FspPK15I.dna
Bio.Restriction.Restriction.FtnUV.dna
Bio.Restriction.Restriction.GauT27I.dna
Bio.Restriction.Restriction.Gba708II.dna
Bio.Restriction.Restriction.GdiII.dna
Bio.Restriction.Restriction.GlaI.dna
Bio.Restriction.Restriction.GluI.dna
Bio.Restriction.Restriction.GsaI.dna
Bio.Restriction.Restriction.GsuI.dna
Bio.Restriction.Restriction.HaeI.dna
Bio.Restriction.Restriction.HaeII.dna
Bio.Restriction.Restriction.HaeIII.dna
Bio.Restriction.Restriction.HapII.dna
Bio.Restriction.Restriction.HauII.dna
Bio.Restriction.Restriction.HbaII.dna
Bio.Restriction.Restriction.HdeNY26I.dna
Bio.Restriction.Restriction.HdeZA17I.dna
Bio.Restriction.Restriction.HgaI.dna
Bio.Restriction.Restriction.HgiAI.dna
Bio.Restriction.Restriction.HgiCI.dna
Bio.Restriction.Restriction.HgiEII.dna
Bio.Restriction.Restriction.HgiJII.dna
Bio.Restriction.Restriction.HhaI.dna
Bio.Restriction.Restriction.Hin1I.dna
Bio.Restriction.Restriction.Hin1II.dna
Bio.Restriction.Restriction.Hin4I.dna
Bio.Restriction.Restriction.Hin4II.dna
Bio.Restriction.Restriction.Hin6I.dna
Bio.Restriction.Restriction.HinP1I.dna
Bio.Restriction.Restriction.HincII.dna
Bio.Restriction.Restriction.HindII.dna
Bio.Restriction.Restriction.HindIII.dna
Bio.Restriction.Restriction.HinfI.dna
Bio.Restriction.Restriction.HpaI.dna
Bio.Restriction.Restriction.HpaII.dna
Bio.Restriction.Restriction.HphI.dna
Bio.Restriction.Restriction.Hpy166II.dna
Bio.Restriction.Restriction.Hpy178III.dna
Bio.Restriction.Restriction.Hpy188I.dna
Bio.Restriction.Restriction.Hpy188III.dna
Bio.Restriction.Restriction.Hpy300XI.dna
Bio.Restriction.Restriction.Hpy8I.dna
Bio.Restriction.Restriction.Hpy99I.dna
Bio.Restriction.Restriction.Hpy99XIII.dna
Bio.Restriction.Restriction.Hpy99XIV.dna
Bio.Restriction.Restriction.Hpy99XIV_mut1.dna
Bio.Restriction.Restriction.Hpy99XXII.dna
Bio.Restriction.Restriction.HpyAV.dna
Bio.Restriction.Restriction.HpyAXIV.dna
Bio.Restriction.Restriction.HpyAXVIII.dna
Bio.Restriction.Restriction.HpyAXVI_mut1.dna
Bio.Restriction.Restriction.HpyAXVI_mut2.dna
Bio.Restriction.Restriction.HpyCH4III.dna
Bio.Restriction.Restriction.HpyCH4IV.dna
Bio.Restriction.Restriction.HpyCH4V.dna
Bio.Restriction.Restriction.HpyF10VI.dna
Bio.Restriction.Restriction.HpyF3I.dna
Bio.Restriction.Restriction.HpySE526I.dna
Bio.Restriction.Restriction.HpyUM032XIII.dna
Bio.Restriction.Restriction.HpyUM032XIII_mut1.dna
Bio.Restriction.Restriction.HpyUM032XIV.dna
Bio.Restriction.Restriction.HpyUM037X.dna
Bio.Restriction.Restriction.Hsp92I.dna
Bio.Restriction.Restriction.Hsp92II.dna
Bio.Restriction.Restriction.HspAI.dna
Bio.Restriction.Restriction.HspMHR1II.dna
Bio.Restriction.Restriction.Jma19592I.dna
Bio.Restriction.Restriction.Jma19592II.dna
Bio.Restriction.Restriction.Jsp2502II.dna
Bio.Restriction.Restriction.KasI.dna
Bio.Restriction.Restriction.KflI.dna
Bio.Restriction.Restriction.Kor51II.dna
Bio.Restriction.Restriction.Kpn156V.dna
Bio.Restriction.Restriction.Kpn2I.dna
Bio.Restriction.Restriction.Kpn327I.dna
Bio.Restriction.Restriction.KpnI.dna
Bio.Restriction.Restriction.KpnNH25III.dna
Bio.Restriction.Restriction.KpnNIH30III.dna
Bio.Restriction.Restriction.KpnNIH50I.dna
Bio.Restriction.Restriction.KroI.dna
Bio.Restriction.Restriction.Ksp22I.dna
Bio.Restriction.Restriction.Ksp632I.dna
Bio.Restriction.Restriction.KspAI.dna
Bio.Restriction.Restriction.KspI.dna
Bio.Restriction.Restriction.Kzo9I.dna
Bio.Restriction.Restriction.Lba2029III.dna
Bio.Restriction.Restriction.Lde4408II.dna
Bio.Restriction.Restriction.LguI.dna
Bio.Restriction.Restriction.LlaG50I.dna
Bio.Restriction.Restriction.LmnI.dna
Bio.Restriction.Restriction.Lmo370I.dna
Bio.Restriction.Restriction.Lmo911II.dna
Bio.Restriction.Restriction.Lpl1004II.dna
Bio.Restriction.Restriction.LpnI.dna
Bio.Restriction.Restriction.LpnPI.dna
Bio.Restriction.Restriction.Lra68I.dna
Bio.Restriction.Restriction.LsaDS4I.dna
Bio.Restriction.Restriction.Lsp1109I.dna
Bio.Restriction.Restriction.Lsp48III.dna
Bio.Restriction.Restriction.Lsp6406VI.dna
Bio.Restriction.Restriction.LweI.dna
Bio.Restriction.Restriction.MabI.dna
Bio.Restriction.Restriction.MaeI.dna
Bio.Restriction.Restriction.MaeII.dna
Bio.Restriction.Restriction.MaeIII.dna
Bio.Restriction.Restriction.MalI.dna
Bio.Restriction.Restriction.MaqI.dna
Bio.Restriction.Restriction.MauBI.dna
Bio.Restriction.Restriction.Mba11I.dna
Bio.Restriction.Restriction.MbiI.dna
Bio.Restriction.Restriction.MboI.dna
Bio.Restriction.Restriction.MboII.dna
Bio.Restriction.Restriction.McaTI.dna
Bio.Restriction.Restriction.Mcr10I.dna
Bio.Restriction.Restriction.McrI.dna
Bio.Restriction.Restriction.MfeI.dna
Bio.Restriction.Restriction.MflI.dna
Bio.Restriction.Restriction.MhlI.dna
Bio.Restriction.Restriction.MjaIV.dna
Bio.Restriction.Restriction.MkaDII.dna
Bio.Restriction.Restriction.MlsI.dna
Bio.Restriction.Restriction.Mlu211III.dna
Bio.Restriction.Restriction.MluCI.dna
Bio.Restriction.Restriction.MluI.dna
Bio.Restriction.Restriction.MluNI.dna
Bio.Restriction.Restriction.Mly113I.dna
Bio.Restriction.Restriction.MlyI.dna
Bio.Restriction.Restriction.MmeI.dna
Bio.Restriction.Restriction.MnlI.dna
Bio.Restriction.Restriction.Mox20I.dna
Bio.Restriction.Restriction.Mph1103I.dna
Bio.Restriction.Restriction.MreI.dna
Bio.Restriction.Restriction.MroI.dna
Bio.Restriction.Restriction.MroNI.dna
Bio.Restriction.Restriction.MroXI.dna
Bio.Restriction.Restriction.MscI.dna
Bio.Restriction.Restriction.MseI.dna
Bio.Restriction.Restriction.MslI.dna
Bio.Restriction.Restriction.Msp20I.dna
Bio.Restriction.Restriction.MspA1I.dna
Bio.Restriction.Restriction.MspCI.dna
Bio.Restriction.Restriction.MspGI.dna
Bio.Restriction.Restriction.MspI.dna
Bio.Restriction.Restriction.MspI7II.dna
Bio.Restriction.Restriction.MspI7IV.dna
Bio.Restriction.Restriction.MspJI.dna
Bio.Restriction.Restriction.MspR9I.dna
Bio.Restriction.Restriction.MspSC27II.dna
Bio.Restriction.Restriction.MssI.dna
Bio.Restriction.Restriction.MstI.dna
Bio.Restriction.Restriction.MteI.dna
Bio.Restriction.Restriction.MtuHN878II.dna
Bio.Restriction.Restriction.MunI.dna
Bio.Restriction.Restriction.Mva1269I.dna
Bio.Restriction.Restriction.MvaI.dna
Bio.Restriction.Restriction.MvnI.dna
Bio.Restriction.Restriction.MwoI.dna
Bio.Restriction.Restriction.NaeI.dna
Bio.Restriction.Restriction.Nal45188II.dna
Bio.Restriction.Restriction.NarI.dna
Bio.Restriction.Restriction.Nbr128II.dna
Bio.Restriction.Restriction.NciI.dna
Bio.Restriction.Restriction.NcoI.dna
Bio.Restriction.Restriction.NdeI.dna
Bio.Restriction.Restriction.NdeII.dna
Bio.Restriction.Restriction.NgoAVII.dna
Bio.Restriction.Restriction.NgoAVIII.dna
Bio.Restriction.Restriction.NgoMIV.dna
Bio.Restriction.Restriction.NhaXI.dna
Bio.Restriction.Restriction.NheI.dna
Bio.Restriction.Restriction.NhoI.dna
Bio.Restriction.Restriction.NlaCI.dna
Bio.Restriction.Restriction.NlaIII.dna
Bio.Restriction.Restriction.NlaIV.dna
Bio.Restriction.Restriction.Nli3877I.dna
Bio.Restriction.Restriction.NmeA6CIII.dna
Bio.Restriction.Restriction.NmeAIII.dna
Bio.Restriction.Restriction.NmeDI.dna
Bio.Restriction.Restriction.NmuCI.dna
Bio.Restriction.Restriction.NotI.dna
Bio.Restriction.Restriction.NpeUS61II.dna
Bio.Restriction.Restriction.NruI.dna
Bio.Restriction.Restriction.NsbI.dna
Bio.Restriction.Restriction.NsiI.dna
Bio.Restriction.Restriction.NspBII.dna
Bio.Restriction.Restriction.NspI.dna
Bio.Restriction.Restriction.NspV.dna
Bio.Restriction.Restriction.ObaBS10I.dna
Bio.Restriction.Restriction.OgrI.dna
Bio.Restriction.Restriction.OliI.dna
Bio.Restriction.Restriction.OspHL35III.dna
Bio.Restriction.Restriction.PabI.dna
Bio.Restriction.Restriction.Pac19842II.dna
Bio.Restriction.Restriction.PacI.dna
Bio.Restriction.Restriction.PacIII.dna
Bio.Restriction.Restriction.PaeI.dna
Bio.Restriction.Restriction.PaeR7I.dna
Bio.Restriction.Restriction.PagI.dna
Bio.Restriction.Restriction.Pal408I.dna
Bio.Restriction.Restriction.PalAI.dna
Bio.Restriction.Restriction.PasI.dna
Bio.Restriction.Restriction.PauI.dna
Bio.Restriction.Restriction.Pba2294I.dna
Bio.Restriction.Restriction.PcaII.dna
Bio.Restriction.Restriction.PceI.dna
Bio.Restriction.Restriction.PciI.dna
Bio.Restriction.Restriction.PciSI.dna
Bio.Restriction.Restriction.Pcr308II.dna
Bio.Restriction.Restriction.PcsI.dna
Bio.Restriction.Restriction.PctI.dna
Bio.Restriction.Restriction.Pdi8503III.dna
Bio.Restriction.Restriction.PdiI.dna
Bio.Restriction.Restriction.PdmI.dna
Bio.Restriction.Restriction.Pdu1735I.dna
Bio.Restriction.Restriction.PenI.dna
Bio.Restriction.Restriction.PfeI.dna
Bio.Restriction.Restriction.Pfl1108I.dna
Bio.Restriction.Restriction.Pfl23II.dna
Bio.Restriction.Restriction.Pfl8569I.dna
Bio.Restriction.Restriction.PflFI.dna
Bio.Restriction.Restriction.PflMI.dna
Bio.Restriction.Restriction.PflPt14I.dna
Bio.Restriction.Restriction.PfoI.dna
Bio.Restriction.Restriction.PfrJS12IV.dna
Bio.Restriction.Restriction.PfrJS12V.dna
Bio.Restriction.Restriction.PfrJS15III.dna
Bio.Restriction.Restriction.Pin17FIII.dna
Bio.Restriction.Restriction.PinAI.dna
Bio.Restriction.Restriction.PinP23II.dna
Bio.Restriction.Restriction.PinP59III.dna
Bio.Restriction.Restriction.PkrI.dna
Bio.Restriction.Restriction.PlaDI.dna
Bio.Restriction.Restriction.Ple19I.dna
Bio.Restriction.Restriction.PleI.dna
Bio.Restriction.Restriction.PliMI.dna
Bio.Restriction.Restriction.PluTI.dna
Bio.Restriction.Restriction.PmaCI.dna
Bio.Restriction.Restriction.PmeI.dna
Bio.Restriction.Restriction.PmlI.dna
Bio.Restriction.Restriction.PpiI.dna
Bio.Restriction.Restriction.PpiP13II.dna
Bio.Restriction.Restriction.PpsI.dna
Bio.Restriction.Restriction.Ppu10I.dna
Bio.Restriction.Restriction.Ppu21I.dna
Bio.Restriction.Restriction.PpuMI.dna
Bio.Restriction.Restriction.PscI.dna
Bio.Restriction.Restriction.Pse18267I.dna
Bio.Restriction.Restriction.PshAI.dna
Bio.Restriction.Restriction.PshBI.dna
Bio.Restriction.Restriction.PsiI.dna
Bio.Restriction.Restriction.Psp0357II.dna
Bio.Restriction.Restriction.Psp03I.dna
Bio.Restriction.Restriction.Psp124BI.dna
Bio.Restriction.Restriction.Psp1406I.dna
Bio.Restriction.Restriction.Psp5II.dna
Bio.Restriction.Restriction.Psp6I.dna
Bio.Restriction.Restriction.PspCI.dna
Bio.Restriction.Restriction.PspEI.dna
Bio.Restriction.Restriction.PspFI.dna
Bio.Restriction.Restriction.PspGI.dna
Bio.Restriction.Restriction.PspLI.dna
Bio.Restriction.Restriction.PspN4I.dna
Bio.Restriction.Restriction.PspOMI.dna
Bio.Restriction.Restriction.PspOMII.dna
Bio.Restriction.Restriction.PspPI.dna
Bio.Restriction.Restriction.PspPPI.dna
Bio.Restriction.Restriction.PspPRI.dna
Bio.Restriction.Restriction.PspXI.dna
Bio.Restriction.Restriction.PsrI.dna
Bio.Restriction.Restriction.PssI.dna
Bio.Restriction.Restriction.Pst14472I.dna
Bio.Restriction.Restriction.Pst145I.dna
Bio.Restriction.Restriction.Pst273I.dna
Bio.Restriction.Restriction.PstI.dna
Bio.Restriction.Restriction.PstNI.dna
Bio.Restriction.Restriction.PsuGI.dna
Bio.Restriction.Restriction.PsuI.dna
Bio.Restriction.Restriction.PsyI.dna
Bio.Restriction.Restriction.PteI.dna
Bio.Restriction.Restriction.PvuI.dna
Bio.Restriction.Restriction.PvuII.dna
Bio.Restriction.Restriction.Rba2021I.dna
Bio.Restriction.Restriction.RceI.dna
Bio.Restriction.Restriction.RdeGBI.dna
Bio.Restriction.Restriction.RdeGBII.dna
Bio.Restriction.Restriction.RdeGBIII.dna
Bio.Restriction.Restriction.RflFIII.dna
Bio.Restriction.Restriction.RgaI.dna
Bio.Restriction.Restriction.RigI.dna
Bio.Restriction.Restriction.RlaI.dna
Bio.Restriction.Restriction.RlaII.dna
Bio.Restriction.Restriction.RleAI.dna
Bio.Restriction.Restriction.Rmu369III.dna
Bio.Restriction.Restriction.RpaB5I.dna
Bio.Restriction.Restriction.RpaBI.dna
Bio.Restriction.Restriction.RpaI.dna
Bio.Restriction.Restriction.RpaTI.dna
Bio.Restriction.Restriction.RruI.dna
Bio.Restriction.Restriction.RsaI.dna
Bio.Restriction.Restriction.RsaNI.dna
Bio.Restriction.Restriction.RseI.dna
Bio.Restriction.Restriction.Rsp008IV.dna
Bio.Restriction.Restriction.Rsp008V.dna
Bio.Restriction.Restriction.Rsp531II.dna
Bio.Restriction.Restriction.RspPBTS2III.dna
Bio.Restriction.Restriction.Rsr2I.dna
Bio.Restriction.Restriction.RsrII.dna
Bio.Restriction.Restriction.Rtr1953I.dna
Bio.Restriction.Restriction.SacI.dna
Bio.Restriction.Restriction.SacII.dna
Bio.Restriction.Restriction.Saf8902III.dna
Bio.Restriction.Restriction.Sag901I.dna
Bio.Restriction.Restriction.SalI.dna
Bio.Restriction.Restriction.SanDI.dna
Bio.Restriction.Restriction.SapI.dna
Bio.Restriction.Restriction.SaqAI.dna
Bio.Restriction.Restriction.SatI.dna
Bio.Restriction.Restriction.Sau3AI.dna
Bio.Restriction.Restriction.Sau96I.dna
Bio.Restriction.Restriction.SauI.dna
Bio.Restriction.Restriction.Sba460II.dna
Bio.Restriction.Restriction.SbfI.dna
Bio.Restriction.Restriction.Sbo46I.dna
Bio.Restriction.Restriction.ScaI.dna
Bio.Restriction.Restriction.SchI.dna
Bio.Restriction.Restriction.SciI.dna
Bio.Restriction.Restriction.ScoDS2II.dna
Bio.Restriction.Restriction.ScrFI.dna
Bio.Restriction.Restriction.SdaI.dna
Bio.Restriction.Restriction.SdeAI.dna
Bio.Restriction.Restriction.SdeOSI.dna
Bio.Restriction.Restriction.SduI.dna
Bio.Restriction.Restriction.SecI.dna
Bio.Restriction.Restriction.SelI.dna
Bio.Restriction.Restriction.Sen17963III.dna
Bio.Restriction.Restriction.SenA1673III.dna
Bio.Restriction.Restriction.SenSARA26III.dna
Bio.Restriction.Restriction.SenTFIV.dna
Bio.Restriction.Restriction.SetI.dna
Bio.Restriction.Restriction.SexAI.dna
Bio.Restriction.Restriction.SfaAI.dna
Bio.Restriction.Restriction.SfaNI.dna
Bio.Restriction.Restriction.SfcI.dna
Bio.Restriction.Restriction.SfeI.dna
Bio.Restriction.Restriction.SfiI.dna
Bio.Restriction.Restriction.SfoI.dna
Bio.Restriction.Restriction.Sfr274I.dna
Bio.Restriction.Restriction.Sfr303I.dna
Bio.Restriction.Restriction.SfuI.dna
Bio.Restriction.Restriction.SgeI.dna
Bio.Restriction.Restriction.SgfI.dna
Bio.Restriction.Restriction.SgrAI.dna
Bio.Restriction.Restriction.SgrBI.dna
Bio.Restriction.Restriction.SgrDI.dna
Bio.Restriction.Restriction.SgrTI.dna
Bio.Restriction.Restriction.SgsI.dna
Bio.Restriction.Restriction.SimI.dna
Bio.Restriction.Restriction.SinI.dna
Bio.Restriction.Restriction.SlaI.dna
Bio.Restriction.Restriction.SmaI.dna
Bio.Restriction.Restriction.SmaUMH5I.dna
Bio.Restriction.Restriction.SmaUMH8I.dna
Bio.Restriction.Restriction.SmiI.dna
Bio.Restriction.Restriction.SmiMI.dna
Bio.Restriction.Restriction.SmlI.dna
Bio.Restriction.Restriction.SmoI.dna
Bio.Restriction.Restriction.SnaBI.dna
Bio.Restriction.Restriction.SnaI.dna
Bio.Restriction.Restriction.Sno506I.dna
Bio.Restriction.Restriction.SpeI.dna
Bio.Restriction.Restriction.SphI.dna
Bio.Restriction.Restriction.SplI.dna
Bio.Restriction.Restriction.SpnRII.dna
Bio.Restriction.Restriction.SpoDI.dna
Bio.Restriction.Restriction.SrfI.dna
Bio.Restriction.Restriction.Sse232I.dna
Bio.Restriction.Restriction.Sse8387I.dna
Bio.Restriction.Restriction.Sse8647I.dna
Bio.Restriction.Restriction.Sse9I.dna
Bio.Restriction.Restriction.SseBI.dna
Bio.Restriction.Restriction.SsiI.dna
Bio.Restriction.Restriction.Ssp6803IV.dna
Bio.Restriction.Restriction.Ssp714II.dna
Bio.Restriction.Restriction.SspD5I.dna
Bio.Restriction.Restriction.SspDI.dna
Bio.Restriction.Restriction.SspI.dna
Bio.Restriction.Restriction.SspMI.dna
Bio.Restriction.Restriction.SstE37I.dna
Bio.Restriction.Restriction.SstI.dna
Bio.Restriction.Restriction.Sth132I.dna
Bio.Restriction.Restriction.Sth20745III.dna
Bio.Restriction.Restriction.Sth302II.dna
Bio.Restriction.Restriction.SthSt3II.dna
Bio.Restriction.Restriction.StsI.dna
Bio.Restriction.Restriction.StuI.dna
Bio.Restriction.Restriction.StyD4I.dna
Bio.Restriction.Restriction.StyI.dna
Bio.Restriction.Restriction.SurP32aII.dna
Bio.Restriction.Restriction.SwaI.dna
Bio.Restriction.Restriction.TaaI.dna
Bio.Restriction.Restriction.TaiI.dna
Bio.Restriction.Restriction.TaqI.dna
Bio.Restriction.Restriction.TaqII.dna
Bio.Restriction.Restriction.TaqIII.dna
Bio.Restriction.Restriction.TasI.dna
Bio.Restriction.Restriction.TatI.dna
Bio.Restriction.Restriction.TauI.dna
Bio.Restriction.Restriction.TfiI.dna
Bio.Restriction.Restriction.TpyTP2I.dna
Bio.Restriction.Restriction.Tru1I.dna
Bio.Restriction.Restriction.Tru9I.dna
Bio.Restriction.Restriction.TscAI.dna
Bio.Restriction.Restriction.TseFI.dna
Bio.Restriction.Restriction.TseI.dna
Bio.Restriction.Restriction.TsoI.dna
Bio.Restriction.Restriction.Tsp45I.dna
Bio.Restriction.Restriction.Tsp4CI.dna
Bio.Restriction.Restriction.TspARh3I.dna
Bio.Restriction.Restriction.TspDTI.dna
Bio.Restriction.Restriction.TspEI.dna
Bio.Restriction.Restriction.TspGWI.dna
Bio.Restriction.Restriction.TspMI.dna
Bio.Restriction.Restriction.TspRI.dna
Bio.Restriction.Restriction.TssI.dna
Bio.Restriction.Restriction.TstI.dna
Bio.Restriction.Restriction.TsuI.dna
Bio.Restriction.Restriction.Tth111I.dna
Bio.Restriction.Restriction.Tth111II.dna
Bio.Restriction.Restriction.UbaF11I.dna
Bio.Restriction.Restriction.UbaF12I.dna
Bio.Restriction.Restriction.UbaF13I.dna
Bio.Restriction.Restriction.UbaF14I.dna
Bio.Restriction.Restriction.UbaF9I.dna
Bio.Restriction.Restriction.UbaPI.dna
Bio.Restriction.Restriction.UcoMSI.dna
Bio.Restriction.Restriction.UnbI.dna
Bio.Restriction.Restriction.Van9116I.dna
Bio.Restriction.Restriction.Van91I.dna
Bio.Restriction.Restriction.VchE4II.dna
Bio.Restriction.Restriction.Vdi96II.dna
Bio.Restriction.Restriction.Vha464I.dna
Bio.Restriction.Restriction.VneI.dna
Bio.Restriction.Restriction.VpaK11AI.dna
Bio.Restriction.Restriction.VpaK11BI.dna
Bio.Restriction.Restriction.VspI.dna
Bio.Restriction.Restriction.Vtu19109I.dna
Bio.Restriction.Restriction.WviI.dna
Bio.Restriction.Restriction.XagI.dna
Bio.Restriction.Restriction.XapI.dna
Bio.Restriction.Restriction.XbaI.dna
Bio.Restriction.Restriction.Xca85IV.dna
Bio.Restriction.Restriction.XceI.dna
Bio.Restriction.Restriction.XcmI.dna
Bio.Restriction.Restriction.XhoI.dna
Bio.Restriction.Restriction.XhoII.dna
Bio.Restriction.Restriction.XmaI.dna
Bio.Restriction.Restriction.XmaIII.dna
Bio.Restriction.Restriction.XmaJI.dna
Bio.Restriction.Restriction.XmiI.dna
Bio.Restriction.Restriction.XmnI.dna
Bio.Restriction.Restriction.XspI.dna
Bio.Restriction.Restriction.YkrI.dna
Bio.Restriction.Restriction.ZraI.dna
Bio.Restriction.Restriction.ZrmI.dna
Bio.Restriction.Restriction.Zsp2I.dna" class="py-name" href="#" onclick="return doclink('link-983', 'dna', 'link-981');">dna</a></tt><tt class="py-op">.</tt><tt id="link-984" class="py-name"><a title="Bio.Seq.Seq.transcribe
Bio.Seq.UnknownSeq.transcribe
Bio.Seq.transcribe" class="py-name" href="#" onclick="return doclink('link-984', 'transcribe', 'link-587');">transcribe</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L2612"></a><tt class="py-lineno">2612</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-985" class="py-name"><a title="Bio.Restriction.Restriction.AanI.dna
Bio.Restriction.Restriction.AarI.dna
Bio.Restriction.Restriction.AasI.dna
Bio.Restriction.Restriction.AatII.dna
Bio.Restriction.Restriction.Aba6411II.dna
Bio.Restriction.Restriction.AbaB8342IV.dna
Bio.Restriction.Restriction.AbaCIII.dna
Bio.Restriction.Restriction.AbaSI.dna
Bio.Restriction.Restriction.AbaUMB2I.dna
Bio.Restriction.Restriction.AbsI.dna
Bio.Restriction.Restriction.Acc16I.dna
Bio.Restriction.Restriction.Acc36I.dna
Bio.Restriction.Restriction.Acc65I.dna
Bio.Restriction.Restriction.Acc65V.dna
Bio.Restriction.Restriction.AccB1I.dna
Bio.Restriction.Restriction.AccB7I.dna
Bio.Restriction.Restriction.AccBSI.dna
Bio.Restriction.Restriction.AccI.dna
Bio.Restriction.Restriction.AccII.dna
Bio.Restriction.Restriction.AccIII.dna
Bio.Restriction.Restriction.AceIII.dna
Bio.Restriction.Restriction.AchA6III.dna
Bio.Restriction.Restriction.AciI.dna
Bio.Restriction.Restriction.AclI.dna
Bio.Restriction.Restriction.AclWI.dna
Bio.Restriction.Restriction.Aco12261II.dna
Bio.Restriction.Restriction.AcoI.dna
Bio.Restriction.Restriction.AcoY31II.dna
Bio.Restriction.Restriction.AcsI.dna
Bio.Restriction.Restriction.AcuI.dna
Bio.Restriction.Restriction.AcvI.dna
Bio.Restriction.Restriction.AcyI.dna
Bio.Restriction.Restriction.AdeI.dna
Bio.Restriction.Restriction.Adh6U21I.dna
Bio.Restriction.Restriction.AfaI.dna
Bio.Restriction.Restriction.AfeI.dna
Bio.Restriction.Restriction.AfiI.dna
Bio.Restriction.Restriction.AflII.dna
Bio.Restriction.Restriction.AflIII.dna
Bio.Restriction.Restriction.AgeI.dna
Bio.Restriction.Restriction.AgsI.dna
Bio.Restriction.Restriction.AhaIII.dna
Bio.Restriction.Restriction.AhdI.dna
Bio.Restriction.Restriction.AhlI.dna
Bio.Restriction.Restriction.AhyRBAHI.dna
Bio.Restriction.Restriction.AhyYL17I.dna
Bio.Restriction.Restriction.AjiI.dna
Bio.Restriction.Restriction.AjnI.dna
Bio.Restriction.Restriction.AjuI.dna
Bio.Restriction.Restriction.AleI.dna
Bio.Restriction.Restriction.AlfI.dna
Bio.Restriction.Restriction.AloI.dna
Bio.Restriction.Restriction.AluBI.dna
Bio.Restriction.Restriction.AluI.dna
Bio.Restriction.Restriction.Alw21I.dna
Bio.Restriction.Restriction.Alw26I.dna
Bio.Restriction.Restriction.Alw44I.dna
Bio.Restriction.Restriction.AlwFI.dna
Bio.Restriction.Restriction.AlwI.dna
Bio.Restriction.Restriction.AlwNI.dna
Bio.Restriction.Restriction.Ama87I.dna
Bio.Restriction.Restriction.AmaCSI.dna
Bio.Restriction.Restriction.Aod1I.dna
Bio.Restriction.Restriction.Aor13HI.dna
Bio.Restriction.Restriction.Aor51HI.dna
Bio.Restriction.Restriction.AoxI.dna
Bio.Restriction.Restriction.ApaBI.dna
Bio.Restriction.Restriction.ApaI.dna
Bio.Restriction.Restriction.ApaLI.dna
Bio.Restriction.Restriction.ApeKI.dna
Bio.Restriction.Restriction.ApoI.dna
Bio.Restriction.Restriction.ApyPI.dna
Bio.Restriction.Restriction.AquII.dna
Bio.Restriction.Restriction.AquIII.dna
Bio.Restriction.Restriction.AquIV.dna
Bio.Restriction.Restriction.ArsI.dna
Bio.Restriction.Restriction.AscI.dna
Bio.Restriction.Restriction.AseI.dna
Bio.Restriction.Restriction.Asi256I.dna
Bio.Restriction.Restriction.AsiGI.dna
Bio.Restriction.Restriction.AsiSI.dna
Bio.Restriction.Restriction.Asp103I.dna
Bio.Restriction.Restriction.Asp114pII.dna
Bio.Restriction.Restriction.Asp700I.dna
Bio.Restriction.Restriction.Asp718I.dna
Bio.Restriction.Restriction.AspA2I.dna
Bio.Restriction.Restriction.AspBHI.dna
Bio.Restriction.Restriction.AspDUT2V.dna
Bio.Restriction.Restriction.AspJHL3II.dna
Bio.Restriction.Restriction.AspLEI.dna
Bio.Restriction.Restriction.AspNIH4III.dna
Bio.Restriction.Restriction.AspS9I.dna
Bio.Restriction.Restriction.AspSLV7III.dna
Bio.Restriction.Restriction.Asu14238IV.dna
Bio.Restriction.Restriction.AsuC2I.dna
Bio.Restriction.Restriction.AsuHPI.dna
Bio.Restriction.Restriction.AsuI.dna
Bio.Restriction.Restriction.AsuII.dna
Bio.Restriction.Restriction.AsuNHI.dna
Bio.Restriction.Restriction.AteTI.dna
Bio.Restriction.Restriction.AvaI.dna
Bio.Restriction.Restriction.AvaII.dna
Bio.Restriction.Restriction.AvaIII.dna
Bio.Restriction.Restriction.AvrII.dna
Bio.Restriction.Restriction.Awo1030IV.dna
Bio.Restriction.Restriction.AxyI.dna
Bio.Restriction.Restriction.BaeGI.dna
Bio.Restriction.Restriction.BaeI.dna
Bio.Restriction.Restriction.Bag18758I.dna
Bio.Restriction.Restriction.BalI.dna
Bio.Restriction.Restriction.BamHI.dna
Bio.Restriction.Restriction.BanI.dna
Bio.Restriction.Restriction.BanII.dna
Bio.Restriction.Restriction.BanLI.dna
Bio.Restriction.Restriction.BarI.dna
Bio.Restriction.Restriction.Bau1417V.dna
Bio.Restriction.Restriction.BauI.dna
Bio.Restriction.Restriction.Bbr52II.dna
Bio.Restriction.Restriction.Bbr57III.dna
Bio.Restriction.Restriction.Bbr7017II.dna
Bio.Restriction.Restriction.Bbr7017III.dna
Bio.Restriction.Restriction.BbrPI.dna
Bio.Restriction.Restriction.BbsI.dna
Bio.Restriction.Restriction.BbuB31I.dna
Bio.Restriction.Restriction.BbuB31II.dna
Bio.Restriction.Restriction.Bbv12I.dna
Bio.Restriction.Restriction.BbvCI.dna
Bio.Restriction.Restriction.BbvI.dna
Bio.Restriction.Restriction.BbvII.dna
Bio.Restriction.Restriction.BccI.dna
Bio.Restriction.Restriction.Bce3081I.dna
Bio.Restriction.Restriction.Bce83I.dna
Bio.Restriction.Restriction.BceAI.dna
Bio.Restriction.Restriction.BceSIV.dna
Bio.Restriction.Restriction.BcefI.dna
Bio.Restriction.Restriction.BcgI.dna
Bio.Restriction.Restriction.BciT130I.dna
Bio.Restriction.Restriction.BciVI.dna
Bio.Restriction.Restriction.BclI.dna
Bio.Restriction.Restriction.BcnI.dna
Bio.Restriction.Restriction.BcoDI.dna
Bio.Restriction.Restriction.BcuI.dna
Bio.Restriction.Restriction.BdaI.dna
Bio.Restriction.Restriction.BetI.dna
Bio.Restriction.Restriction.BfaI.dna
Bio.Restriction.Restriction.BfaSII.dna
Bio.Restriction.Restriction.BfiI.dna
Bio.Restriction.Restriction.BfmI.dna
Bio.Restriction.Restriction.BfoI.dna
Bio.Restriction.Restriction.BfrI.dna
Bio.Restriction.Restriction.BfuAI.dna
Bio.Restriction.Restriction.BfuI.dna
Bio.Restriction.Restriction.Bga514I.dna
Bio.Restriction.Restriction.BglI.dna
Bio.Restriction.Restriction.BglII.dna
Bio.Restriction.Restriction.BinI.dna
Bio.Restriction.Restriction.BisI.dna
Bio.Restriction.Restriction.BkrAM31DI.dna
Bio.Restriction.Restriction.Ble402II.dna
Bio.Restriction.Restriction.BlnI.dna
Bio.Restriction.Restriction.BloAII.dna
Bio.Restriction.Restriction.BlpI.dna
Bio.Restriction.Restriction.BlsI.dna
Bio.Restriction.Restriction.BmcAI.dna
Bio.Restriction.Restriction.Bme1390I.dna
Bio.Restriction.Restriction.Bme18I.dna
Bio.Restriction.Restriction.BmeDI.dna
Bio.Restriction.Restriction.BmeRI.dna
Bio.Restriction.Restriction.BmeT110I.dna
Bio.Restriction.Restriction.BmgBI.dna
Bio.Restriction.Restriction.BmgI.dna
Bio.Restriction.Restriction.BmgT120I.dna
Bio.Restriction.Restriction.BmiI.dna
Bio.Restriction.Restriction.BmrFI.dna
Bio.Restriction.Restriction.BmrI.dna
Bio.Restriction.Restriction.BmsI.dna
Bio.Restriction.Restriction.BmtI.dna
Bio.Restriction.Restriction.BmuI.dna
Bio.Restriction.Restriction.BoxI.dna
Bio.Restriction.Restriction.BpiI.dna
Bio.Restriction.Restriction.BplI.dna
Bio.Restriction.Restriction.BpmI.dna
Bio.Restriction.Restriction.Bpu10I.dna
Bio.Restriction.Restriction.Bpu1102I.dna
Bio.Restriction.Restriction.Bpu14I.dna
Bio.Restriction.Restriction.BpuEI.dna
Bio.Restriction.Restriction.BpuMI.dna
Bio.Restriction.Restriction.Bsa29I.dna
Bio.Restriction.Restriction.BsaAI.dna
Bio.Restriction.Restriction.BsaBI.dna
Bio.Restriction.Restriction.BsaHI.dna
Bio.Restriction.Restriction.BsaI.dna
Bio.Restriction.Restriction.BsaJI.dna
Bio.Restriction.Restriction.BsaWI.dna
Bio.Restriction.Restriction.BsaXI.dna
Bio.Restriction.Restriction.BsbI.dna
Bio.Restriction.Restriction.Bsc4I.dna
Bio.Restriction.Restriction.BscAI.dna
Bio.Restriction.Restriction.BscGI.dna
Bio.Restriction.Restriction.Bse118I.dna
Bio.Restriction.Restriction.Bse1I.dna
Bio.Restriction.Restriction.Bse21I.dna
Bio.Restriction.Restriction.Bse3DI.dna
Bio.Restriction.Restriction.Bse8I.dna
Bio.Restriction.Restriction.BseAI.dna
Bio.Restriction.Restriction.BseBI.dna
Bio.Restriction.Restriction.BseCI.dna
Bio.Restriction.Restriction.BseDI.dna
Bio.Restriction.Restriction.BseGI.dna
Bio.Restriction.Restriction.BseJI.dna
Bio.Restriction.Restriction.BseLI.dna
Bio.Restriction.Restriction.BseMI.dna
Bio.Restriction.Restriction.BseMII.dna
Bio.Restriction.Restriction.BseNI.dna
Bio.Restriction.Restriction.BsePI.dna
Bio.Restriction.Restriction.BseRI.dna
Bio.Restriction.Restriction.BseSI.dna
Bio.Restriction.Restriction.BseX3I.dna
Bio.Restriction.Restriction.BseXI.dna
Bio.Restriction.Restriction.BseYI.dna
Bio.Restriction.Restriction.BsgI.dna
Bio.Restriction.Restriction.Bsh1236I.dna
Bio.Restriction.Restriction.Bsh1285I.dna
Bio.Restriction.Restriction.BshFI.dna
Bio.Restriction.Restriction.BshNI.dna
Bio.Restriction.Restriction.BshTI.dna
Bio.Restriction.Restriction.BshVI.dna
Bio.Restriction.Restriction.BsiEI.dna
Bio.Restriction.Restriction.BsiHKAI.dna
Bio.Restriction.Restriction.BsiHKCI.dna
Bio.Restriction.Restriction.BsiI.dna
Bio.Restriction.Restriction.BsiSI.dna
Bio.Restriction.Restriction.BsiWI.dna
Bio.Restriction.Restriction.BsiYI.dna
Bio.Restriction.Restriction.BslFI.dna
Bio.Restriction.Restriction.BslI.dna
Bio.Restriction.Restriction.BsmAI.dna
Bio.Restriction.Restriction.BsmBI.dna
Bio.Restriction.Restriction.BsmFI.dna
Bio.Restriction.Restriction.BsmI.dna
Bio.Restriction.Restriction.BsnI.dna
Bio.Restriction.Restriction.Bso31I.dna
Bio.Restriction.Restriction.BsoBI.dna
Bio.Restriction.Restriction.Bsp119I.dna
Bio.Restriction.Restriction.Bsp120I.dna
Bio.Restriction.Restriction.Bsp1286I.dna
Bio.Restriction.Restriction.Bsp13I.dna
Bio.Restriction.Restriction.Bsp1407I.dna
Bio.Restriction.Restriction.Bsp143I.dna
Bio.Restriction.Restriction.Bsp1720I.dna
Bio.Restriction.Restriction.Bsp19I.dna
Bio.Restriction.Restriction.Bsp24I.dna
Bio.Restriction.Restriction.Bsp3004IV.dna
Bio.Restriction.Restriction.Bsp460III.dna
Bio.Restriction.Restriction.Bsp68I.dna
Bio.Restriction.Restriction.BspACI.dna
Bio.Restriction.Restriction.BspANI.dna
Bio.Restriction.Restriction.BspCNI.dna
Bio.Restriction.Restriction.BspD6I.dna
Bio.Restriction.Restriction.BspDI.dna
Bio.Restriction.Restriction.BspEI.dna
Bio.Restriction.Restriction.BspFNI.dna
Bio.Restriction.Restriction.BspGI.dna
Bio.Restriction.Restriction.BspHI.dna
Bio.Restriction.Restriction.BspLI.dna
Bio.Restriction.Restriction.BspLU11I.dna
Bio.Restriction.Restriction.BspMAI.dna
Bio.Restriction.Restriction.BspMI.dna
Bio.Restriction.Restriction.BspMII.dna
Bio.Restriction.Restriction.BspNCI.dna
Bio.Restriction.Restriction.BspOI.dna
Bio.Restriction.Restriction.BspPI.dna
Bio.Restriction.Restriction.BspQI.dna
Bio.Restriction.Restriction.BspT104I.dna
Bio.Restriction.Restriction.BspT107I.dna
Bio.Restriction.Restriction.BspTI.dna
Bio.Restriction.Restriction.BspTNI.dna
Bio.Restriction.Restriction.BsrBI.dna
Bio.Restriction.Restriction.BsrDI.dna
Bio.Restriction.Restriction.BsrFI.dna
Bio.Restriction.Restriction.BsrGI.dna
Bio.Restriction.Restriction.BsrI.dna
Bio.Restriction.Restriction.BssAI.dna
Bio.Restriction.Restriction.BssECI.dna
Bio.Restriction.Restriction.BssHII.dna
Bio.Restriction.Restriction.BssMI.dna
Bio.Restriction.Restriction.BssNAI.dna
Bio.Restriction.Restriction.BssNI.dna
Bio.Restriction.Restriction.BssSI.dna
Bio.Restriction.Restriction.BssT1I.dna
Bio.Restriction.Restriction.Bst1107I.dna
Bio.Restriction.Restriction.Bst2BI.dna
Bio.Restriction.Restriction.Bst2UI.dna
Bio.Restriction.Restriction.Bst4CI.dna
Bio.Restriction.Restriction.Bst6I.dna
Bio.Restriction.Restriction.BstACI.dna
Bio.Restriction.Restriction.BstAFI.dna
Bio.Restriction.Restriction.BstAPI.dna
Bio.Restriction.Restriction.BstAUI.dna
Bio.Restriction.Restriction.BstBAI.dna
Bio.Restriction.Restriction.BstBI.dna
Bio.Restriction.Restriction.BstC8I.dna
Bio.Restriction.Restriction.BstDEI.dna
Bio.Restriction.Restriction.BstDSI.dna
Bio.Restriction.Restriction.BstEII.dna
Bio.Restriction.Restriction.BstENI.dna
Bio.Restriction.Restriction.BstF5I.dna
Bio.Restriction.Restriction.BstFNI.dna
Bio.Restriction.Restriction.BstH2I.dna
Bio.Restriction.Restriction.BstHHI.dna
Bio.Restriction.Restriction.BstKTI.dna
Bio.Restriction.Restriction.BstMAI.dna
Bio.Restriction.Restriction.BstMBI.dna
Bio.Restriction.Restriction.BstMCI.dna
Bio.Restriction.Restriction.BstMWI.dna
Bio.Restriction.Restriction.BstNI.dna
Bio.Restriction.Restriction.BstNSI.dna
Bio.Restriction.Restriction.BstPAI.dna
Bio.Restriction.Restriction.BstPI.dna
Bio.Restriction.Restriction.BstSCI.dna
Bio.Restriction.Restriction.BstSFI.dna
Bio.Restriction.Restriction.BstSLI.dna
Bio.Restriction.Restriction.BstSNI.dna
Bio.Restriction.Restriction.BstUI.dna
Bio.Restriction.Restriction.BstV1I.dna
Bio.Restriction.Restriction.BstV2I.dna
Bio.Restriction.Restriction.BstX2I.dna
Bio.Restriction.Restriction.BstXI.dna
Bio.Restriction.Restriction.BstYI.dna
Bio.Restriction.Restriction.BstZ17I.dna
Bio.Restriction.Restriction.BstZI.dna
Bio.Restriction.Restriction.Bsu15I.dna
Bio.Restriction.Restriction.Bsu36I.dna
Bio.Restriction.Restriction.BsuI.dna
Bio.Restriction.Restriction.BsuRI.dna
Bio.Restriction.Restriction.BsuTUI.dna
Bio.Restriction.Restriction.BtgI.dna
Bio.Restriction.Restriction.BtgZI.dna
Bio.Restriction.Restriction.BthCI.dna
Bio.Restriction.Restriction.BtrI.dna
Bio.Restriction.Restriction.BtsCI.dna
Bio.Restriction.Restriction.BtsI.dna
Bio.Restriction.Restriction.BtsIMutI.dna
Bio.Restriction.Restriction.BtuMI.dna
Bio.Restriction.Restriction.Bve1B23I.dna
Bio.Restriction.Restriction.BveI.dna
Bio.Restriction.Restriction.Cac8I.dna
Bio.Restriction.Restriction.CaiI.dna
Bio.Restriction.Restriction.Cal14237I.dna
Bio.Restriction.Restriction.CalB3II.dna
Bio.Restriction.Restriction.Cau10061II.dna
Bio.Restriction.Restriction.CauII.dna
Bio.Restriction.Restriction.Cba13II.dna
Bio.Restriction.Restriction.Cba16038I.dna
Bio.Restriction.Restriction.Cbo67071IV.dna
Bio.Restriction.Restriction.Cch467III.dna
Bio.Restriction.Restriction.CchII.dna
Bio.Restriction.Restriction.CchIII.dna
Bio.Restriction.Restriction.CciI.dna
Bio.Restriction.Restriction.CciNI.dna
Bio.Restriction.Restriction.Cco14983V.dna
Bio.Restriction.Restriction.Cco14983VI.dna
Bio.Restriction.Restriction.CcrNAIII.dna
Bio.Restriction.Restriction.Cdi11397I.dna
Bio.Restriction.Restriction.CdiI.dna
Bio.Restriction.Restriction.CdpI.dna
Bio.Restriction.Restriction.Cdu23823II.dna
Bio.Restriction.Restriction.CfoI.dna
Bio.Restriction.Restriction.Cfr10I.dna
Bio.Restriction.Restriction.Cfr13I.dna
Bio.Restriction.Restriction.Cfr42I.dna
Bio.Restriction.Restriction.Cfr9I.dna
Bio.Restriction.Restriction.CfrI.dna
Bio.Restriction.Restriction.CfrMH13II.dna
Bio.Restriction.Restriction.CfrMH16VI.dna
Bio.Restriction.Restriction.Cfupf3II.dna
Bio.Restriction.Restriction.Cgl13032I.dna
Bio.Restriction.Restriction.Cgl13032II.dna
Bio.Restriction.Restriction.ChaI.dna
Bio.Restriction.Restriction.Cje265V.dna
Bio.Restriction.Restriction.Cje54107III.dna
Bio.Restriction.Restriction.CjeFIII.dna
Bio.Restriction.Restriction.CjeFV.dna
Bio.Restriction.Restriction.CjeI.dna
Bio.Restriction.Restriction.CjeNII.dna
Bio.Restriction.Restriction.CjeNIII.dna
Bio.Restriction.Restriction.CjeNV.dna
Bio.Restriction.Restriction.CjeP659IV.dna
Bio.Restriction.Restriction.CjePI.dna
Bio.Restriction.Restriction.CjuI.dna
Bio.Restriction.Restriction.CjuII.dna
Bio.Restriction.Restriction.Cla11845III.dna
Bio.Restriction.Restriction.ClaI.dna
Bio.Restriction.Restriction.Cly7489II.dna
Bio.Restriction.Restriction.Cma23826I.dna
Bio.Restriction.Restriction.CpoI.dna
Bio.Restriction.Restriction.CseI.dna
Bio.Restriction.Restriction.CsiI.dna
Bio.Restriction.Restriction.Csp2014I.dna
Bio.Restriction.Restriction.Csp6I.dna
Bio.Restriction.Restriction.CspAI.dna
Bio.Restriction.Restriction.CspCI.dna
Bio.Restriction.Restriction.CspI.dna
Bio.Restriction.Restriction.CstMI.dna
Bio.Restriction.Restriction.CviAII.dna
Bio.Restriction.Restriction.CviJI.dna
Bio.Restriction.Restriction.CviKI_1.dna
Bio.Restriction.Restriction.CviQI.dna
Bio.Restriction.Restriction.CviRI.dna
Bio.Restriction.Restriction.Dde51507I.dna
Bio.Restriction.Restriction.DdeI.dna
Bio.Restriction.Restriction.DinI.dna
Bio.Restriction.Restriction.DpnI.dna
Bio.Restriction.Restriction.DpnII.dna
Bio.Restriction.Restriction.DraI.dna
Bio.Restriction.Restriction.DraII.dna
Bio.Restriction.Restriction.DraIII.dna
Bio.Restriction.Restriction.DraRI.dna
Bio.Restriction.Restriction.DrdI.dna
Bio.Restriction.Restriction.DrdII.dna
Bio.Restriction.Restriction.DrdIV.dna
Bio.Restriction.Restriction.DriI.dna
Bio.Restriction.Restriction.DsaI.dna
Bio.Restriction.Restriction.DseDI.dna
Bio.Restriction.Restriction.DvuIII.dna
Bio.Restriction.Restriction.EaeI.dna
Bio.Restriction.Restriction.EagI.dna
Bio.Restriction.Restriction.Eam1104I.dna
Bio.Restriction.Restriction.Eam1105I.dna
Bio.Restriction.Restriction.EarI.dna
Bio.Restriction.Restriction.EciI.dna
Bio.Restriction.Restriction.Ecl136II.dna
Bio.Restriction.Restriction.Ecl234I.dna
Bio.Restriction.Restriction.Ecl35734I.dna
Bio.Restriction.Restriction.EclXI.dna
Bio.Restriction.Restriction.Eco105I.dna
Bio.Restriction.Restriction.Eco130I.dna
Bio.Restriction.Restriction.Eco147I.dna
Bio.Restriction.Restriction.Eco24I.dna
Bio.Restriction.Restriction.Eco31I.dna
Bio.Restriction.Restriction.Eco32I.dna
Bio.Restriction.Restriction.Eco43896II.dna
Bio.Restriction.Restriction.Eco4465II.dna
Bio.Restriction.Restriction.Eco47I.dna
Bio.Restriction.Restriction.Eco47III.dna
Bio.Restriction.Restriction.Eco52I.dna
Bio.Restriction.Restriction.Eco53kI.dna
Bio.Restriction.Restriction.Eco57I.dna
Bio.Restriction.Restriction.Eco57MI.dna
Bio.Restriction.Restriction.Eco72I.dna
Bio.Restriction.Restriction.Eco81I.dna
Bio.Restriction.Restriction.Eco88I.dna
Bio.Restriction.Restriction.Eco91I.dna
Bio.Restriction.Restriction.EcoBLMcrX.dna
Bio.Restriction.Restriction.EcoE1140I.dna
Bio.Restriction.Restriction.EcoHI.dna
Bio.Restriction.Restriction.EcoHSI.dna
Bio.Restriction.Restriction.EcoICRI.dna
Bio.Restriction.Restriction.EcoMVII.dna
Bio.Restriction.Restriction.EcoNI.dna
Bio.Restriction.Restriction.EcoNIH6II.dna
Bio.Restriction.Restriction.EcoO109I.dna
Bio.Restriction.Restriction.EcoO65I.dna
Bio.Restriction.Restriction.EcoRI.dna
Bio.Restriction.Restriction.EcoRII.dna
Bio.Restriction.Restriction.EcoRV.dna
Bio.Restriction.Restriction.EcoT14I.dna
Bio.Restriction.Restriction.EcoT22I.dna
Bio.Restriction.Restriction.EcoT38I.dna
Bio.Restriction.Restriction.EgeI.dna
Bio.Restriction.Restriction.EheI.dna
Bio.Restriction.Restriction.Eli8509II.dna
Bio.Restriction.Restriction.ErhI.dna
Bio.Restriction.Restriction.EsaBC3I.dna
Bio.Restriction.Restriction.EsaSSI.dna
Bio.Restriction.Restriction.Esp3007I.dna
Bio.Restriction.Restriction.Esp3I.dna
Bio.Restriction.Restriction.EspI.dna
Bio.Restriction.Restriction.FaeI.dna
Bio.Restriction.Restriction.FaiI.dna
Bio.Restriction.Restriction.FalI.dna
Bio.Restriction.Restriction.FaqI.dna
Bio.Restriction.Restriction.FatI.dna
Bio.Restriction.Restriction.FauI.dna
Bio.Restriction.Restriction.FauNDI.dna
Bio.Restriction.Restriction.FbaI.dna
Bio.Restriction.Restriction.FblI.dna
Bio.Restriction.Restriction.Fco1691IV.dna
Bio.Restriction.Restriction.FinI.dna
Bio.Restriction.Restriction.FmuI.dna
Bio.Restriction.Restriction.Fnu4HI.dna
Bio.Restriction.Restriction.FnuDII.dna
Bio.Restriction.Restriction.FokI.dna
Bio.Restriction.Restriction.FriOI.dna
Bio.Restriction.Restriction.FseI.dna
Bio.Restriction.Restriction.Fsp4HI.dna
Bio.Restriction.Restriction.FspAI.dna
Bio.Restriction.Restriction.FspBI.dna
Bio.Restriction.Restriction.FspEI.dna
Bio.Restriction.Restriction.FspI.dna
Bio.Restriction.Restriction.FspPK15I.dna
Bio.Restriction.Restriction.FtnUV.dna
Bio.Restriction.Restriction.GauT27I.dna
Bio.Restriction.Restriction.Gba708II.dna
Bio.Restriction.Restriction.GdiII.dna
Bio.Restriction.Restriction.GlaI.dna
Bio.Restriction.Restriction.GluI.dna
Bio.Restriction.Restriction.GsaI.dna
Bio.Restriction.Restriction.GsuI.dna
Bio.Restriction.Restriction.HaeI.dna
Bio.Restriction.Restriction.HaeII.dna
Bio.Restriction.Restriction.HaeIII.dna
Bio.Restriction.Restriction.HapII.dna
Bio.Restriction.Restriction.HauII.dna
Bio.Restriction.Restriction.HbaII.dna
Bio.Restriction.Restriction.HdeNY26I.dna
Bio.Restriction.Restriction.HdeZA17I.dna
Bio.Restriction.Restriction.HgaI.dna
Bio.Restriction.Restriction.HgiAI.dna
Bio.Restriction.Restriction.HgiCI.dna
Bio.Restriction.Restriction.HgiEII.dna
Bio.Restriction.Restriction.HgiJII.dna
Bio.Restriction.Restriction.HhaI.dna
Bio.Restriction.Restriction.Hin1I.dna
Bio.Restriction.Restriction.Hin1II.dna
Bio.Restriction.Restriction.Hin4I.dna
Bio.Restriction.Restriction.Hin4II.dna
Bio.Restriction.Restriction.Hin6I.dna
Bio.Restriction.Restriction.HinP1I.dna
Bio.Restriction.Restriction.HincII.dna
Bio.Restriction.Restriction.HindII.dna
Bio.Restriction.Restriction.HindIII.dna
Bio.Restriction.Restriction.HinfI.dna
Bio.Restriction.Restriction.HpaI.dna
Bio.Restriction.Restriction.HpaII.dna
Bio.Restriction.Restriction.HphI.dna
Bio.Restriction.Restriction.Hpy166II.dna
Bio.Restriction.Restriction.Hpy178III.dna
Bio.Restriction.Restriction.Hpy188I.dna
Bio.Restriction.Restriction.Hpy188III.dna
Bio.Restriction.Restriction.Hpy300XI.dna
Bio.Restriction.Restriction.Hpy8I.dna
Bio.Restriction.Restriction.Hpy99I.dna
Bio.Restriction.Restriction.Hpy99XIII.dna
Bio.Restriction.Restriction.Hpy99XIV.dna
Bio.Restriction.Restriction.Hpy99XIV_mut1.dna
Bio.Restriction.Restriction.Hpy99XXII.dna
Bio.Restriction.Restriction.HpyAV.dna
Bio.Restriction.Restriction.HpyAXIV.dna
Bio.Restriction.Restriction.HpyAXVIII.dna
Bio.Restriction.Restriction.HpyAXVI_mut1.dna
Bio.Restriction.Restriction.HpyAXVI_mut2.dna
Bio.Restriction.Restriction.HpyCH4III.dna
Bio.Restriction.Restriction.HpyCH4IV.dna
Bio.Restriction.Restriction.HpyCH4V.dna
Bio.Restriction.Restriction.HpyF10VI.dna
Bio.Restriction.Restriction.HpyF3I.dna
Bio.Restriction.Restriction.HpySE526I.dna
Bio.Restriction.Restriction.HpyUM032XIII.dna
Bio.Restriction.Restriction.HpyUM032XIII_mut1.dna
Bio.Restriction.Restriction.HpyUM032XIV.dna
Bio.Restriction.Restriction.HpyUM037X.dna
Bio.Restriction.Restriction.Hsp92I.dna
Bio.Restriction.Restriction.Hsp92II.dna
Bio.Restriction.Restriction.HspAI.dna
Bio.Restriction.Restriction.HspMHR1II.dna
Bio.Restriction.Restriction.Jma19592I.dna
Bio.Restriction.Restriction.Jma19592II.dna
Bio.Restriction.Restriction.Jsp2502II.dna
Bio.Restriction.Restriction.KasI.dna
Bio.Restriction.Restriction.KflI.dna
Bio.Restriction.Restriction.Kor51II.dna
Bio.Restriction.Restriction.Kpn156V.dna
Bio.Restriction.Restriction.Kpn2I.dna
Bio.Restriction.Restriction.Kpn327I.dna
Bio.Restriction.Restriction.KpnI.dna
Bio.Restriction.Restriction.KpnNH25III.dna
Bio.Restriction.Restriction.KpnNIH30III.dna
Bio.Restriction.Restriction.KpnNIH50I.dna
Bio.Restriction.Restriction.KroI.dna
Bio.Restriction.Restriction.Ksp22I.dna
Bio.Restriction.Restriction.Ksp632I.dna
Bio.Restriction.Restriction.KspAI.dna
Bio.Restriction.Restriction.KspI.dna
Bio.Restriction.Restriction.Kzo9I.dna
Bio.Restriction.Restriction.Lba2029III.dna
Bio.Restriction.Restriction.Lde4408II.dna
Bio.Restriction.Restriction.LguI.dna
Bio.Restriction.Restriction.LlaG50I.dna
Bio.Restriction.Restriction.LmnI.dna
Bio.Restriction.Restriction.Lmo370I.dna
Bio.Restriction.Restriction.Lmo911II.dna
Bio.Restriction.Restriction.Lpl1004II.dna
Bio.Restriction.Restriction.LpnI.dna
Bio.Restriction.Restriction.LpnPI.dna
Bio.Restriction.Restriction.Lra68I.dna
Bio.Restriction.Restriction.LsaDS4I.dna
Bio.Restriction.Restriction.Lsp1109I.dna
Bio.Restriction.Restriction.Lsp48III.dna
Bio.Restriction.Restriction.Lsp6406VI.dna
Bio.Restriction.Restriction.LweI.dna
Bio.Restriction.Restriction.MabI.dna
Bio.Restriction.Restriction.MaeI.dna
Bio.Restriction.Restriction.MaeII.dna
Bio.Restriction.Restriction.MaeIII.dna
Bio.Restriction.Restriction.MalI.dna
Bio.Restriction.Restriction.MaqI.dna
Bio.Restriction.Restriction.MauBI.dna
Bio.Restriction.Restriction.Mba11I.dna
Bio.Restriction.Restriction.MbiI.dna
Bio.Restriction.Restriction.MboI.dna
Bio.Restriction.Restriction.MboII.dna
Bio.Restriction.Restriction.McaTI.dna
Bio.Restriction.Restriction.Mcr10I.dna
Bio.Restriction.Restriction.McrI.dna
Bio.Restriction.Restriction.MfeI.dna
Bio.Restriction.Restriction.MflI.dna
Bio.Restriction.Restriction.MhlI.dna
Bio.Restriction.Restriction.MjaIV.dna
Bio.Restriction.Restriction.MkaDII.dna
Bio.Restriction.Restriction.MlsI.dna
Bio.Restriction.Restriction.Mlu211III.dna
Bio.Restriction.Restriction.MluCI.dna
Bio.Restriction.Restriction.MluI.dna
Bio.Restriction.Restriction.MluNI.dna
Bio.Restriction.Restriction.Mly113I.dna
Bio.Restriction.Restriction.MlyI.dna
Bio.Restriction.Restriction.MmeI.dna
Bio.Restriction.Restriction.MnlI.dna
Bio.Restriction.Restriction.Mox20I.dna
Bio.Restriction.Restriction.Mph1103I.dna
Bio.Restriction.Restriction.MreI.dna
Bio.Restriction.Restriction.MroI.dna
Bio.Restriction.Restriction.MroNI.dna
Bio.Restriction.Restriction.MroXI.dna
Bio.Restriction.Restriction.MscI.dna
Bio.Restriction.Restriction.MseI.dna
Bio.Restriction.Restriction.MslI.dna
Bio.Restriction.Restriction.Msp20I.dna
Bio.Restriction.Restriction.MspA1I.dna
Bio.Restriction.Restriction.MspCI.dna
Bio.Restriction.Restriction.MspGI.dna
Bio.Restriction.Restriction.MspI.dna
Bio.Restriction.Restriction.MspI7II.dna
Bio.Restriction.Restriction.MspI7IV.dna
Bio.Restriction.Restriction.MspJI.dna
Bio.Restriction.Restriction.MspR9I.dna
Bio.Restriction.Restriction.MspSC27II.dna
Bio.Restriction.Restriction.MssI.dna
Bio.Restriction.Restriction.MstI.dna
Bio.Restriction.Restriction.MteI.dna
Bio.Restriction.Restriction.MtuHN878II.dna
Bio.Restriction.Restriction.MunI.dna
Bio.Restriction.Restriction.Mva1269I.dna
Bio.Restriction.Restriction.MvaI.dna
Bio.Restriction.Restriction.MvnI.dna
Bio.Restriction.Restriction.MwoI.dna
Bio.Restriction.Restriction.NaeI.dna
Bio.Restriction.Restriction.Nal45188II.dna
Bio.Restriction.Restriction.NarI.dna
Bio.Restriction.Restriction.Nbr128II.dna
Bio.Restriction.Restriction.NciI.dna
Bio.Restriction.Restriction.NcoI.dna
Bio.Restriction.Restriction.NdeI.dna
Bio.Restriction.Restriction.NdeII.dna
Bio.Restriction.Restriction.NgoAVII.dna
Bio.Restriction.Restriction.NgoAVIII.dna
Bio.Restriction.Restriction.NgoMIV.dna
Bio.Restriction.Restriction.NhaXI.dna
Bio.Restriction.Restriction.NheI.dna
Bio.Restriction.Restriction.NhoI.dna
Bio.Restriction.Restriction.NlaCI.dna
Bio.Restriction.Restriction.NlaIII.dna
Bio.Restriction.Restriction.NlaIV.dna
Bio.Restriction.Restriction.Nli3877I.dna
Bio.Restriction.Restriction.NmeA6CIII.dna
Bio.Restriction.Restriction.NmeAIII.dna
Bio.Restriction.Restriction.NmeDI.dna
Bio.Restriction.Restriction.NmuCI.dna
Bio.Restriction.Restriction.NotI.dna
Bio.Restriction.Restriction.NpeUS61II.dna
Bio.Restriction.Restriction.NruI.dna
Bio.Restriction.Restriction.NsbI.dna
Bio.Restriction.Restriction.NsiI.dna
Bio.Restriction.Restriction.NspBII.dna
Bio.Restriction.Restriction.NspI.dna
Bio.Restriction.Restriction.NspV.dna
Bio.Restriction.Restriction.ObaBS10I.dna
Bio.Restriction.Restriction.OgrI.dna
Bio.Restriction.Restriction.OliI.dna
Bio.Restriction.Restriction.OspHL35III.dna
Bio.Restriction.Restriction.PabI.dna
Bio.Restriction.Restriction.Pac19842II.dna
Bio.Restriction.Restriction.PacI.dna
Bio.Restriction.Restriction.PacIII.dna
Bio.Restriction.Restriction.PaeI.dna
Bio.Restriction.Restriction.PaeR7I.dna
Bio.Restriction.Restriction.PagI.dna
Bio.Restriction.Restriction.Pal408I.dna
Bio.Restriction.Restriction.PalAI.dna
Bio.Restriction.Restriction.PasI.dna
Bio.Restriction.Restriction.PauI.dna
Bio.Restriction.Restriction.Pba2294I.dna
Bio.Restriction.Restriction.PcaII.dna
Bio.Restriction.Restriction.PceI.dna
Bio.Restriction.Restriction.PciI.dna
Bio.Restriction.Restriction.PciSI.dna
Bio.Restriction.Restriction.Pcr308II.dna
Bio.Restriction.Restriction.PcsI.dna
Bio.Restriction.Restriction.PctI.dna
Bio.Restriction.Restriction.Pdi8503III.dna
Bio.Restriction.Restriction.PdiI.dna
Bio.Restriction.Restriction.PdmI.dna
Bio.Restriction.Restriction.Pdu1735I.dna
Bio.Restriction.Restriction.PenI.dna
Bio.Restriction.Restriction.PfeI.dna
Bio.Restriction.Restriction.Pfl1108I.dna
Bio.Restriction.Restriction.Pfl23II.dna
Bio.Restriction.Restriction.Pfl8569I.dna
Bio.Restriction.Restriction.PflFI.dna
Bio.Restriction.Restriction.PflMI.dna
Bio.Restriction.Restriction.PflPt14I.dna
Bio.Restriction.Restriction.PfoI.dna
Bio.Restriction.Restriction.PfrJS12IV.dna
Bio.Restriction.Restriction.PfrJS12V.dna
Bio.Restriction.Restriction.PfrJS15III.dna
Bio.Restriction.Restriction.Pin17FIII.dna
Bio.Restriction.Restriction.PinAI.dna
Bio.Restriction.Restriction.PinP23II.dna
Bio.Restriction.Restriction.PinP59III.dna
Bio.Restriction.Restriction.PkrI.dna
Bio.Restriction.Restriction.PlaDI.dna
Bio.Restriction.Restriction.Ple19I.dna
Bio.Restriction.Restriction.PleI.dna
Bio.Restriction.Restriction.PliMI.dna
Bio.Restriction.Restriction.PluTI.dna
Bio.Restriction.Restriction.PmaCI.dna
Bio.Restriction.Restriction.PmeI.dna
Bio.Restriction.Restriction.PmlI.dna
Bio.Restriction.Restriction.PpiI.dna
Bio.Restriction.Restriction.PpiP13II.dna
Bio.Restriction.Restriction.PpsI.dna
Bio.Restriction.Restriction.Ppu10I.dna
Bio.Restriction.Restriction.Ppu21I.dna
Bio.Restriction.Restriction.PpuMI.dna
Bio.Restriction.Restriction.PscI.dna
Bio.Restriction.Restriction.Pse18267I.dna
Bio.Restriction.Restriction.PshAI.dna
Bio.Restriction.Restriction.PshBI.dna
Bio.Restriction.Restriction.PsiI.dna
Bio.Restriction.Restriction.Psp0357II.dna
Bio.Restriction.Restriction.Psp03I.dna
Bio.Restriction.Restriction.Psp124BI.dna
Bio.Restriction.Restriction.Psp1406I.dna
Bio.Restriction.Restriction.Psp5II.dna
Bio.Restriction.Restriction.Psp6I.dna
Bio.Restriction.Restriction.PspCI.dna
Bio.Restriction.Restriction.PspEI.dna
Bio.Restriction.Restriction.PspFI.dna
Bio.Restriction.Restriction.PspGI.dna
Bio.Restriction.Restriction.PspLI.dna
Bio.Restriction.Restriction.PspN4I.dna
Bio.Restriction.Restriction.PspOMI.dna
Bio.Restriction.Restriction.PspOMII.dna
Bio.Restriction.Restriction.PspPI.dna
Bio.Restriction.Restriction.PspPPI.dna
Bio.Restriction.Restriction.PspPRI.dna
Bio.Restriction.Restriction.PspXI.dna
Bio.Restriction.Restriction.PsrI.dna
Bio.Restriction.Restriction.PssI.dna
Bio.Restriction.Restriction.Pst14472I.dna
Bio.Restriction.Restriction.Pst145I.dna
Bio.Restriction.Restriction.Pst273I.dna
Bio.Restriction.Restriction.PstI.dna
Bio.Restriction.Restriction.PstNI.dna
Bio.Restriction.Restriction.PsuGI.dna
Bio.Restriction.Restriction.PsuI.dna
Bio.Restriction.Restriction.PsyI.dna
Bio.Restriction.Restriction.PteI.dna
Bio.Restriction.Restriction.PvuI.dna
Bio.Restriction.Restriction.PvuII.dna
Bio.Restriction.Restriction.Rba2021I.dna
Bio.Restriction.Restriction.RceI.dna
Bio.Restriction.Restriction.RdeGBI.dna
Bio.Restriction.Restriction.RdeGBII.dna
Bio.Restriction.Restriction.RdeGBIII.dna
Bio.Restriction.Restriction.RflFIII.dna
Bio.Restriction.Restriction.RgaI.dna
Bio.Restriction.Restriction.RigI.dna
Bio.Restriction.Restriction.RlaI.dna
Bio.Restriction.Restriction.RlaII.dna
Bio.Restriction.Restriction.RleAI.dna
Bio.Restriction.Restriction.Rmu369III.dna
Bio.Restriction.Restriction.RpaB5I.dna
Bio.Restriction.Restriction.RpaBI.dna
Bio.Restriction.Restriction.RpaI.dna
Bio.Restriction.Restriction.RpaTI.dna
Bio.Restriction.Restriction.RruI.dna
Bio.Restriction.Restriction.RsaI.dna
Bio.Restriction.Restriction.RsaNI.dna
Bio.Restriction.Restriction.RseI.dna
Bio.Restriction.Restriction.Rsp008IV.dna
Bio.Restriction.Restriction.Rsp008V.dna
Bio.Restriction.Restriction.Rsp531II.dna
Bio.Restriction.Restriction.RspPBTS2III.dna
Bio.Restriction.Restriction.Rsr2I.dna
Bio.Restriction.Restriction.RsrII.dna
Bio.Restriction.Restriction.Rtr1953I.dna
Bio.Restriction.Restriction.SacI.dna
Bio.Restriction.Restriction.SacII.dna
Bio.Restriction.Restriction.Saf8902III.dna
Bio.Restriction.Restriction.Sag901I.dna
Bio.Restriction.Restriction.SalI.dna
Bio.Restriction.Restriction.SanDI.dna
Bio.Restriction.Restriction.SapI.dna
Bio.Restriction.Restriction.SaqAI.dna
Bio.Restriction.Restriction.SatI.dna
Bio.Restriction.Restriction.Sau3AI.dna
Bio.Restriction.Restriction.Sau96I.dna
Bio.Restriction.Restriction.SauI.dna
Bio.Restriction.Restriction.Sba460II.dna
Bio.Restriction.Restriction.SbfI.dna
Bio.Restriction.Restriction.Sbo46I.dna
Bio.Restriction.Restriction.ScaI.dna
Bio.Restriction.Restriction.SchI.dna
Bio.Restriction.Restriction.SciI.dna
Bio.Restriction.Restriction.ScoDS2II.dna
Bio.Restriction.Restriction.ScrFI.dna
Bio.Restriction.Restriction.SdaI.dna
Bio.Restriction.Restriction.SdeAI.dna
Bio.Restriction.Restriction.SdeOSI.dna
Bio.Restriction.Restriction.SduI.dna
Bio.Restriction.Restriction.SecI.dna
Bio.Restriction.Restriction.SelI.dna
Bio.Restriction.Restriction.Sen17963III.dna
Bio.Restriction.Restriction.SenA1673III.dna
Bio.Restriction.Restriction.SenSARA26III.dna
Bio.Restriction.Restriction.SenTFIV.dna
Bio.Restriction.Restriction.SetI.dna
Bio.Restriction.Restriction.SexAI.dna
Bio.Restriction.Restriction.SfaAI.dna
Bio.Restriction.Restriction.SfaNI.dna
Bio.Restriction.Restriction.SfcI.dna
Bio.Restriction.Restriction.SfeI.dna
Bio.Restriction.Restriction.SfiI.dna
Bio.Restriction.Restriction.SfoI.dna
Bio.Restriction.Restriction.Sfr274I.dna
Bio.Restriction.Restriction.Sfr303I.dna
Bio.Restriction.Restriction.SfuI.dna
Bio.Restriction.Restriction.SgeI.dna
Bio.Restriction.Restriction.SgfI.dna
Bio.Restriction.Restriction.SgrAI.dna
Bio.Restriction.Restriction.SgrBI.dna
Bio.Restriction.Restriction.SgrDI.dna
Bio.Restriction.Restriction.SgrTI.dna
Bio.Restriction.Restriction.SgsI.dna
Bio.Restriction.Restriction.SimI.dna
Bio.Restriction.Restriction.SinI.dna
Bio.Restriction.Restriction.SlaI.dna
Bio.Restriction.Restriction.SmaI.dna
Bio.Restriction.Restriction.SmaUMH5I.dna
Bio.Restriction.Restriction.SmaUMH8I.dna
Bio.Restriction.Restriction.SmiI.dna
Bio.Restriction.Restriction.SmiMI.dna
Bio.Restriction.Restriction.SmlI.dna
Bio.Restriction.Restriction.SmoI.dna
Bio.Restriction.Restriction.SnaBI.dna
Bio.Restriction.Restriction.SnaI.dna
Bio.Restriction.Restriction.Sno506I.dna
Bio.Restriction.Restriction.SpeI.dna
Bio.Restriction.Restriction.SphI.dna
Bio.Restriction.Restriction.SplI.dna
Bio.Restriction.Restriction.SpnRII.dna
Bio.Restriction.Restriction.SpoDI.dna
Bio.Restriction.Restriction.SrfI.dna
Bio.Restriction.Restriction.Sse232I.dna
Bio.Restriction.Restriction.Sse8387I.dna
Bio.Restriction.Restriction.Sse8647I.dna
Bio.Restriction.Restriction.Sse9I.dna
Bio.Restriction.Restriction.SseBI.dna
Bio.Restriction.Restriction.SsiI.dna
Bio.Restriction.Restriction.Ssp6803IV.dna
Bio.Restriction.Restriction.Ssp714II.dna
Bio.Restriction.Restriction.SspD5I.dna
Bio.Restriction.Restriction.SspDI.dna
Bio.Restriction.Restriction.SspI.dna
Bio.Restriction.Restriction.SspMI.dna
Bio.Restriction.Restriction.SstE37I.dna
Bio.Restriction.Restriction.SstI.dna
Bio.Restriction.Restriction.Sth132I.dna
Bio.Restriction.Restriction.Sth20745III.dna
Bio.Restriction.Restriction.Sth302II.dna
Bio.Restriction.Restriction.SthSt3II.dna
Bio.Restriction.Restriction.StsI.dna
Bio.Restriction.Restriction.StuI.dna
Bio.Restriction.Restriction.StyD4I.dna
Bio.Restriction.Restriction.StyI.dna
Bio.Restriction.Restriction.SurP32aII.dna
Bio.Restriction.Restriction.SwaI.dna
Bio.Restriction.Restriction.TaaI.dna
Bio.Restriction.Restriction.TaiI.dna
Bio.Restriction.Restriction.TaqI.dna
Bio.Restriction.Restriction.TaqII.dna
Bio.Restriction.Restriction.TaqIII.dna
Bio.Restriction.Restriction.TasI.dna
Bio.Restriction.Restriction.TatI.dna
Bio.Restriction.Restriction.TauI.dna
Bio.Restriction.Restriction.TfiI.dna
Bio.Restriction.Restriction.TpyTP2I.dna
Bio.Restriction.Restriction.Tru1I.dna
Bio.Restriction.Restriction.Tru9I.dna
Bio.Restriction.Restriction.TscAI.dna
Bio.Restriction.Restriction.TseFI.dna
Bio.Restriction.Restriction.TseI.dna
Bio.Restriction.Restriction.TsoI.dna
Bio.Restriction.Restriction.Tsp45I.dna
Bio.Restriction.Restriction.Tsp4CI.dna
Bio.Restriction.Restriction.TspARh3I.dna
Bio.Restriction.Restriction.TspDTI.dna
Bio.Restriction.Restriction.TspEI.dna
Bio.Restriction.Restriction.TspGWI.dna
Bio.Restriction.Restriction.TspMI.dna
Bio.Restriction.Restriction.TspRI.dna
Bio.Restriction.Restriction.TssI.dna
Bio.Restriction.Restriction.TstI.dna
Bio.Restriction.Restriction.TsuI.dna
Bio.Restriction.Restriction.Tth111I.dna
Bio.Restriction.Restriction.Tth111II.dna
Bio.Restriction.Restriction.UbaF11I.dna
Bio.Restriction.Restriction.UbaF12I.dna
Bio.Restriction.Restriction.UbaF13I.dna
Bio.Restriction.Restriction.UbaF14I.dna
Bio.Restriction.Restriction.UbaF9I.dna
Bio.Restriction.Restriction.UbaPI.dna
Bio.Restriction.Restriction.UcoMSI.dna
Bio.Restriction.Restriction.UnbI.dna
Bio.Restriction.Restriction.Van9116I.dna
Bio.Restriction.Restriction.Van91I.dna
Bio.Restriction.Restriction.VchE4II.dna
Bio.Restriction.Restriction.Vdi96II.dna
Bio.Restriction.Restriction.Vha464I.dna
Bio.Restriction.Restriction.VneI.dna
Bio.Restriction.Restriction.VpaK11AI.dna
Bio.Restriction.Restriction.VpaK11BI.dna
Bio.Restriction.Restriction.VspI.dna
Bio.Restriction.Restriction.Vtu19109I.dna
Bio.Restriction.Restriction.WviI.dna
Bio.Restriction.Restriction.XagI.dna
Bio.Restriction.Restriction.XapI.dna
Bio.Restriction.Restriction.XbaI.dna
Bio.Restriction.Restriction.Xca85IV.dna
Bio.Restriction.Restriction.XceI.dna
Bio.Restriction.Restriction.XcmI.dna
Bio.Restriction.Restriction.XhoI.dna
Bio.Restriction.Restriction.XhoII.dna
Bio.Restriction.Restriction.XmaI.dna
Bio.Restriction.Restriction.XmaIII.dna
Bio.Restriction.Restriction.XmaJI.dna
Bio.Restriction.Restriction.XmiI.dna
Bio.Restriction.Restriction.XmnI.dna
Bio.Restriction.Restriction.XspI.dna
Bio.Restriction.Restriction.YkrI.dna
Bio.Restriction.Restriction.ZraI.dna
Bio.Restriction.Restriction.ZrmI.dna
Bio.Restriction.Restriction.Zsp2I.dna" class="py-name" href="#" onclick="return doclink('link-985', 'dna', 'link-981');">dna</a></tt><tt class="py-op">,</tt> <tt id="link-986" class="py-name"><a title="Bio.Seq.MutableSeq" class="py-name" href="#" onclick="return doclink('link-986', 'MutableSeq', 'link-83');">MutableSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2613"></a><tt class="py-lineno">2613</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-987" class="py-name"><a title="Bio.Restriction.Restriction.AanI.dna
Bio.Restriction.Restriction.AarI.dna
Bio.Restriction.Restriction.AasI.dna
Bio.Restriction.Restriction.AatII.dna
Bio.Restriction.Restriction.Aba6411II.dna
Bio.Restriction.Restriction.AbaB8342IV.dna
Bio.Restriction.Restriction.AbaCIII.dna
Bio.Restriction.Restriction.AbaSI.dna
Bio.Restriction.Restriction.AbaUMB2I.dna
Bio.Restriction.Restriction.AbsI.dna
Bio.Restriction.Restriction.Acc16I.dna
Bio.Restriction.Restriction.Acc36I.dna
Bio.Restriction.Restriction.Acc65I.dna
Bio.Restriction.Restriction.Acc65V.dna
Bio.Restriction.Restriction.AccB1I.dna
Bio.Restriction.Restriction.AccB7I.dna
Bio.Restriction.Restriction.AccBSI.dna
Bio.Restriction.Restriction.AccI.dna
Bio.Restriction.Restriction.AccII.dna
Bio.Restriction.Restriction.AccIII.dna
Bio.Restriction.Restriction.AceIII.dna
Bio.Restriction.Restriction.AchA6III.dna
Bio.Restriction.Restriction.AciI.dna
Bio.Restriction.Restriction.AclI.dna
Bio.Restriction.Restriction.AclWI.dna
Bio.Restriction.Restriction.Aco12261II.dna
Bio.Restriction.Restriction.AcoI.dna
Bio.Restriction.Restriction.AcoY31II.dna
Bio.Restriction.Restriction.AcsI.dna
Bio.Restriction.Restriction.AcuI.dna
Bio.Restriction.Restriction.AcvI.dna
Bio.Restriction.Restriction.AcyI.dna
Bio.Restriction.Restriction.AdeI.dna
Bio.Restriction.Restriction.Adh6U21I.dna
Bio.Restriction.Restriction.AfaI.dna
Bio.Restriction.Restriction.AfeI.dna
Bio.Restriction.Restriction.AfiI.dna
Bio.Restriction.Restriction.AflII.dna
Bio.Restriction.Restriction.AflIII.dna
Bio.Restriction.Restriction.AgeI.dna
Bio.Restriction.Restriction.AgsI.dna
Bio.Restriction.Restriction.AhaIII.dna
Bio.Restriction.Restriction.AhdI.dna
Bio.Restriction.Restriction.AhlI.dna
Bio.Restriction.Restriction.AhyRBAHI.dna
Bio.Restriction.Restriction.AhyYL17I.dna
Bio.Restriction.Restriction.AjiI.dna
Bio.Restriction.Restriction.AjnI.dna
Bio.Restriction.Restriction.AjuI.dna
Bio.Restriction.Restriction.AleI.dna
Bio.Restriction.Restriction.AlfI.dna
Bio.Restriction.Restriction.AloI.dna
Bio.Restriction.Restriction.AluBI.dna
Bio.Restriction.Restriction.AluI.dna
Bio.Restriction.Restriction.Alw21I.dna
Bio.Restriction.Restriction.Alw26I.dna
Bio.Restriction.Restriction.Alw44I.dna
Bio.Restriction.Restriction.AlwFI.dna
Bio.Restriction.Restriction.AlwI.dna
Bio.Restriction.Restriction.AlwNI.dna
Bio.Restriction.Restriction.Ama87I.dna
Bio.Restriction.Restriction.AmaCSI.dna
Bio.Restriction.Restriction.Aod1I.dna
Bio.Restriction.Restriction.Aor13HI.dna
Bio.Restriction.Restriction.Aor51HI.dna
Bio.Restriction.Restriction.AoxI.dna
Bio.Restriction.Restriction.ApaBI.dna
Bio.Restriction.Restriction.ApaI.dna
Bio.Restriction.Restriction.ApaLI.dna
Bio.Restriction.Restriction.ApeKI.dna
Bio.Restriction.Restriction.ApoI.dna
Bio.Restriction.Restriction.ApyPI.dna
Bio.Restriction.Restriction.AquII.dna
Bio.Restriction.Restriction.AquIII.dna
Bio.Restriction.Restriction.AquIV.dna
Bio.Restriction.Restriction.ArsI.dna
Bio.Restriction.Restriction.AscI.dna
Bio.Restriction.Restriction.AseI.dna
Bio.Restriction.Restriction.Asi256I.dna
Bio.Restriction.Restriction.AsiGI.dna
Bio.Restriction.Restriction.AsiSI.dna
Bio.Restriction.Restriction.Asp103I.dna
Bio.Restriction.Restriction.Asp114pII.dna
Bio.Restriction.Restriction.Asp700I.dna
Bio.Restriction.Restriction.Asp718I.dna
Bio.Restriction.Restriction.AspA2I.dna
Bio.Restriction.Restriction.AspBHI.dna
Bio.Restriction.Restriction.AspDUT2V.dna
Bio.Restriction.Restriction.AspJHL3II.dna
Bio.Restriction.Restriction.AspLEI.dna
Bio.Restriction.Restriction.AspNIH4III.dna
Bio.Restriction.Restriction.AspS9I.dna
Bio.Restriction.Restriction.AspSLV7III.dna
Bio.Restriction.Restriction.Asu14238IV.dna
Bio.Restriction.Restriction.AsuC2I.dna
Bio.Restriction.Restriction.AsuHPI.dna
Bio.Restriction.Restriction.AsuI.dna
Bio.Restriction.Restriction.AsuII.dna
Bio.Restriction.Restriction.AsuNHI.dna
Bio.Restriction.Restriction.AteTI.dna
Bio.Restriction.Restriction.AvaI.dna
Bio.Restriction.Restriction.AvaII.dna
Bio.Restriction.Restriction.AvaIII.dna
Bio.Restriction.Restriction.AvrII.dna
Bio.Restriction.Restriction.Awo1030IV.dna
Bio.Restriction.Restriction.AxyI.dna
Bio.Restriction.Restriction.BaeGI.dna
Bio.Restriction.Restriction.BaeI.dna
Bio.Restriction.Restriction.Bag18758I.dna
Bio.Restriction.Restriction.BalI.dna
Bio.Restriction.Restriction.BamHI.dna
Bio.Restriction.Restriction.BanI.dna
Bio.Restriction.Restriction.BanII.dna
Bio.Restriction.Restriction.BanLI.dna
Bio.Restriction.Restriction.BarI.dna
Bio.Restriction.Restriction.Bau1417V.dna
Bio.Restriction.Restriction.BauI.dna
Bio.Restriction.Restriction.Bbr52II.dna
Bio.Restriction.Restriction.Bbr57III.dna
Bio.Restriction.Restriction.Bbr7017II.dna
Bio.Restriction.Restriction.Bbr7017III.dna
Bio.Restriction.Restriction.BbrPI.dna
Bio.Restriction.Restriction.BbsI.dna
Bio.Restriction.Restriction.BbuB31I.dna
Bio.Restriction.Restriction.BbuB31II.dna
Bio.Restriction.Restriction.Bbv12I.dna
Bio.Restriction.Restriction.BbvCI.dna
Bio.Restriction.Restriction.BbvI.dna
Bio.Restriction.Restriction.BbvII.dna
Bio.Restriction.Restriction.BccI.dna
Bio.Restriction.Restriction.Bce3081I.dna
Bio.Restriction.Restriction.Bce83I.dna
Bio.Restriction.Restriction.BceAI.dna
Bio.Restriction.Restriction.BceSIV.dna
Bio.Restriction.Restriction.BcefI.dna
Bio.Restriction.Restriction.BcgI.dna
Bio.Restriction.Restriction.BciT130I.dna
Bio.Restriction.Restriction.BciVI.dna
Bio.Restriction.Restriction.BclI.dna
Bio.Restriction.Restriction.BcnI.dna
Bio.Restriction.Restriction.BcoDI.dna
Bio.Restriction.Restriction.BcuI.dna
Bio.Restriction.Restriction.BdaI.dna
Bio.Restriction.Restriction.BetI.dna
Bio.Restriction.Restriction.BfaI.dna
Bio.Restriction.Restriction.BfaSII.dna
Bio.Restriction.Restriction.BfiI.dna
Bio.Restriction.Restriction.BfmI.dna
Bio.Restriction.Restriction.BfoI.dna
Bio.Restriction.Restriction.BfrI.dna
Bio.Restriction.Restriction.BfuAI.dna
Bio.Restriction.Restriction.BfuI.dna
Bio.Restriction.Restriction.Bga514I.dna
Bio.Restriction.Restriction.BglI.dna
Bio.Restriction.Restriction.BglII.dna
Bio.Restriction.Restriction.BinI.dna
Bio.Restriction.Restriction.BisI.dna
Bio.Restriction.Restriction.BkrAM31DI.dna
Bio.Restriction.Restriction.Ble402II.dna
Bio.Restriction.Restriction.BlnI.dna
Bio.Restriction.Restriction.BloAII.dna
Bio.Restriction.Restriction.BlpI.dna
Bio.Restriction.Restriction.BlsI.dna
Bio.Restriction.Restriction.BmcAI.dna
Bio.Restriction.Restriction.Bme1390I.dna
Bio.Restriction.Restriction.Bme18I.dna
Bio.Restriction.Restriction.BmeDI.dna
Bio.Restriction.Restriction.BmeRI.dna
Bio.Restriction.Restriction.BmeT110I.dna
Bio.Restriction.Restriction.BmgBI.dna
Bio.Restriction.Restriction.BmgI.dna
Bio.Restriction.Restriction.BmgT120I.dna
Bio.Restriction.Restriction.BmiI.dna
Bio.Restriction.Restriction.BmrFI.dna
Bio.Restriction.Restriction.BmrI.dna
Bio.Restriction.Restriction.BmsI.dna
Bio.Restriction.Restriction.BmtI.dna
Bio.Restriction.Restriction.BmuI.dna
Bio.Restriction.Restriction.BoxI.dna
Bio.Restriction.Restriction.BpiI.dna
Bio.Restriction.Restriction.BplI.dna
Bio.Restriction.Restriction.BpmI.dna
Bio.Restriction.Restriction.Bpu10I.dna
Bio.Restriction.Restriction.Bpu1102I.dna
Bio.Restriction.Restriction.Bpu14I.dna
Bio.Restriction.Restriction.BpuEI.dna
Bio.Restriction.Restriction.BpuMI.dna
Bio.Restriction.Restriction.Bsa29I.dna
Bio.Restriction.Restriction.BsaAI.dna
Bio.Restriction.Restriction.BsaBI.dna
Bio.Restriction.Restriction.BsaHI.dna
Bio.Restriction.Restriction.BsaI.dna
Bio.Restriction.Restriction.BsaJI.dna
Bio.Restriction.Restriction.BsaWI.dna
Bio.Restriction.Restriction.BsaXI.dna
Bio.Restriction.Restriction.BsbI.dna
Bio.Restriction.Restriction.Bsc4I.dna
Bio.Restriction.Restriction.BscAI.dna
Bio.Restriction.Restriction.BscGI.dna
Bio.Restriction.Restriction.Bse118I.dna
Bio.Restriction.Restriction.Bse1I.dna
Bio.Restriction.Restriction.Bse21I.dna
Bio.Restriction.Restriction.Bse3DI.dna
Bio.Restriction.Restriction.Bse8I.dna
Bio.Restriction.Restriction.BseAI.dna
Bio.Restriction.Restriction.BseBI.dna
Bio.Restriction.Restriction.BseCI.dna
Bio.Restriction.Restriction.BseDI.dna
Bio.Restriction.Restriction.BseGI.dna
Bio.Restriction.Restriction.BseJI.dna
Bio.Restriction.Restriction.BseLI.dna
Bio.Restriction.Restriction.BseMI.dna
Bio.Restriction.Restriction.BseMII.dna
Bio.Restriction.Restriction.BseNI.dna
Bio.Restriction.Restriction.BsePI.dna
Bio.Restriction.Restriction.BseRI.dna
Bio.Restriction.Restriction.BseSI.dna
Bio.Restriction.Restriction.BseX3I.dna
Bio.Restriction.Restriction.BseXI.dna
Bio.Restriction.Restriction.BseYI.dna
Bio.Restriction.Restriction.BsgI.dna
Bio.Restriction.Restriction.Bsh1236I.dna
Bio.Restriction.Restriction.Bsh1285I.dna
Bio.Restriction.Restriction.BshFI.dna
Bio.Restriction.Restriction.BshNI.dna
Bio.Restriction.Restriction.BshTI.dna
Bio.Restriction.Restriction.BshVI.dna
Bio.Restriction.Restriction.BsiEI.dna
Bio.Restriction.Restriction.BsiHKAI.dna
Bio.Restriction.Restriction.BsiHKCI.dna
Bio.Restriction.Restriction.BsiI.dna
Bio.Restriction.Restriction.BsiSI.dna
Bio.Restriction.Restriction.BsiWI.dna
Bio.Restriction.Restriction.BsiYI.dna
Bio.Restriction.Restriction.BslFI.dna
Bio.Restriction.Restriction.BslI.dna
Bio.Restriction.Restriction.BsmAI.dna
Bio.Restriction.Restriction.BsmBI.dna
Bio.Restriction.Restriction.BsmFI.dna
Bio.Restriction.Restriction.BsmI.dna
Bio.Restriction.Restriction.BsnI.dna
Bio.Restriction.Restriction.Bso31I.dna
Bio.Restriction.Restriction.BsoBI.dna
Bio.Restriction.Restriction.Bsp119I.dna
Bio.Restriction.Restriction.Bsp120I.dna
Bio.Restriction.Restriction.Bsp1286I.dna
Bio.Restriction.Restriction.Bsp13I.dna
Bio.Restriction.Restriction.Bsp1407I.dna
Bio.Restriction.Restriction.Bsp143I.dna
Bio.Restriction.Restriction.Bsp1720I.dna
Bio.Restriction.Restriction.Bsp19I.dna
Bio.Restriction.Restriction.Bsp24I.dna
Bio.Restriction.Restriction.Bsp3004IV.dna
Bio.Restriction.Restriction.Bsp460III.dna
Bio.Restriction.Restriction.Bsp68I.dna
Bio.Restriction.Restriction.BspACI.dna
Bio.Restriction.Restriction.BspANI.dna
Bio.Restriction.Restriction.BspCNI.dna
Bio.Restriction.Restriction.BspD6I.dna
Bio.Restriction.Restriction.BspDI.dna
Bio.Restriction.Restriction.BspEI.dna
Bio.Restriction.Restriction.BspFNI.dna
Bio.Restriction.Restriction.BspGI.dna
Bio.Restriction.Restriction.BspHI.dna
Bio.Restriction.Restriction.BspLI.dna
Bio.Restriction.Restriction.BspLU11I.dna
Bio.Restriction.Restriction.BspMAI.dna
Bio.Restriction.Restriction.BspMI.dna
Bio.Restriction.Restriction.BspMII.dna
Bio.Restriction.Restriction.BspNCI.dna
Bio.Restriction.Restriction.BspOI.dna
Bio.Restriction.Restriction.BspPI.dna
Bio.Restriction.Restriction.BspQI.dna
Bio.Restriction.Restriction.BspT104I.dna
Bio.Restriction.Restriction.BspT107I.dna
Bio.Restriction.Restriction.BspTI.dna
Bio.Restriction.Restriction.BspTNI.dna
Bio.Restriction.Restriction.BsrBI.dna
Bio.Restriction.Restriction.BsrDI.dna
Bio.Restriction.Restriction.BsrFI.dna
Bio.Restriction.Restriction.BsrGI.dna
Bio.Restriction.Restriction.BsrI.dna
Bio.Restriction.Restriction.BssAI.dna
Bio.Restriction.Restriction.BssECI.dna
Bio.Restriction.Restriction.BssHII.dna
Bio.Restriction.Restriction.BssMI.dna
Bio.Restriction.Restriction.BssNAI.dna
Bio.Restriction.Restriction.BssNI.dna
Bio.Restriction.Restriction.BssSI.dna
Bio.Restriction.Restriction.BssT1I.dna
Bio.Restriction.Restriction.Bst1107I.dna
Bio.Restriction.Restriction.Bst2BI.dna
Bio.Restriction.Restriction.Bst2UI.dna
Bio.Restriction.Restriction.Bst4CI.dna
Bio.Restriction.Restriction.Bst6I.dna
Bio.Restriction.Restriction.BstACI.dna
Bio.Restriction.Restriction.BstAFI.dna
Bio.Restriction.Restriction.BstAPI.dna
Bio.Restriction.Restriction.BstAUI.dna
Bio.Restriction.Restriction.BstBAI.dna
Bio.Restriction.Restriction.BstBI.dna
Bio.Restriction.Restriction.BstC8I.dna
Bio.Restriction.Restriction.BstDEI.dna
Bio.Restriction.Restriction.BstDSI.dna
Bio.Restriction.Restriction.BstEII.dna
Bio.Restriction.Restriction.BstENI.dna
Bio.Restriction.Restriction.BstF5I.dna
Bio.Restriction.Restriction.BstFNI.dna
Bio.Restriction.Restriction.BstH2I.dna
Bio.Restriction.Restriction.BstHHI.dna
Bio.Restriction.Restriction.BstKTI.dna
Bio.Restriction.Restriction.BstMAI.dna
Bio.Restriction.Restriction.BstMBI.dna
Bio.Restriction.Restriction.BstMCI.dna
Bio.Restriction.Restriction.BstMWI.dna
Bio.Restriction.Restriction.BstNI.dna
Bio.Restriction.Restriction.BstNSI.dna
Bio.Restriction.Restriction.BstPAI.dna
Bio.Restriction.Restriction.BstPI.dna
Bio.Restriction.Restriction.BstSCI.dna
Bio.Restriction.Restriction.BstSFI.dna
Bio.Restriction.Restriction.BstSLI.dna
Bio.Restriction.Restriction.BstSNI.dna
Bio.Restriction.Restriction.BstUI.dna
Bio.Restriction.Restriction.BstV1I.dna
Bio.Restriction.Restriction.BstV2I.dna
Bio.Restriction.Restriction.BstX2I.dna
Bio.Restriction.Restriction.BstXI.dna
Bio.Restriction.Restriction.BstYI.dna
Bio.Restriction.Restriction.BstZ17I.dna
Bio.Restriction.Restriction.BstZI.dna
Bio.Restriction.Restriction.Bsu15I.dna
Bio.Restriction.Restriction.Bsu36I.dna
Bio.Restriction.Restriction.BsuI.dna
Bio.Restriction.Restriction.BsuRI.dna
Bio.Restriction.Restriction.BsuTUI.dna
Bio.Restriction.Restriction.BtgI.dna
Bio.Restriction.Restriction.BtgZI.dna
Bio.Restriction.Restriction.BthCI.dna
Bio.Restriction.Restriction.BtrI.dna
Bio.Restriction.Restriction.BtsCI.dna
Bio.Restriction.Restriction.BtsI.dna
Bio.Restriction.Restriction.BtsIMutI.dna
Bio.Restriction.Restriction.BtuMI.dna
Bio.Restriction.Restriction.Bve1B23I.dna
Bio.Restriction.Restriction.BveI.dna
Bio.Restriction.Restriction.Cac8I.dna
Bio.Restriction.Restriction.CaiI.dna
Bio.Restriction.Restriction.Cal14237I.dna
Bio.Restriction.Restriction.CalB3II.dna
Bio.Restriction.Restriction.Cau10061II.dna
Bio.Restriction.Restriction.CauII.dna
Bio.Restriction.Restriction.Cba13II.dna
Bio.Restriction.Restriction.Cba16038I.dna
Bio.Restriction.Restriction.Cbo67071IV.dna
Bio.Restriction.Restriction.Cch467III.dna
Bio.Restriction.Restriction.CchII.dna
Bio.Restriction.Restriction.CchIII.dna
Bio.Restriction.Restriction.CciI.dna
Bio.Restriction.Restriction.CciNI.dna
Bio.Restriction.Restriction.Cco14983V.dna
Bio.Restriction.Restriction.Cco14983VI.dna
Bio.Restriction.Restriction.CcrNAIII.dna
Bio.Restriction.Restriction.Cdi11397I.dna
Bio.Restriction.Restriction.CdiI.dna
Bio.Restriction.Restriction.CdpI.dna
Bio.Restriction.Restriction.Cdu23823II.dna
Bio.Restriction.Restriction.CfoI.dna
Bio.Restriction.Restriction.Cfr10I.dna
Bio.Restriction.Restriction.Cfr13I.dna
Bio.Restriction.Restriction.Cfr42I.dna
Bio.Restriction.Restriction.Cfr9I.dna
Bio.Restriction.Restriction.CfrI.dna
Bio.Restriction.Restriction.CfrMH13II.dna
Bio.Restriction.Restriction.CfrMH16VI.dna
Bio.Restriction.Restriction.Cfupf3II.dna
Bio.Restriction.Restriction.Cgl13032I.dna
Bio.Restriction.Restriction.Cgl13032II.dna
Bio.Restriction.Restriction.ChaI.dna
Bio.Restriction.Restriction.Cje265V.dna
Bio.Restriction.Restriction.Cje54107III.dna
Bio.Restriction.Restriction.CjeFIII.dna
Bio.Restriction.Restriction.CjeFV.dna
Bio.Restriction.Restriction.CjeI.dna
Bio.Restriction.Restriction.CjeNII.dna
Bio.Restriction.Restriction.CjeNIII.dna
Bio.Restriction.Restriction.CjeNV.dna
Bio.Restriction.Restriction.CjeP659IV.dna
Bio.Restriction.Restriction.CjePI.dna
Bio.Restriction.Restriction.CjuI.dna
Bio.Restriction.Restriction.CjuII.dna
Bio.Restriction.Restriction.Cla11845III.dna
Bio.Restriction.Restriction.ClaI.dna
Bio.Restriction.Restriction.Cly7489II.dna
Bio.Restriction.Restriction.Cma23826I.dna
Bio.Restriction.Restriction.CpoI.dna
Bio.Restriction.Restriction.CseI.dna
Bio.Restriction.Restriction.CsiI.dna
Bio.Restriction.Restriction.Csp2014I.dna
Bio.Restriction.Restriction.Csp6I.dna
Bio.Restriction.Restriction.CspAI.dna
Bio.Restriction.Restriction.CspCI.dna
Bio.Restriction.Restriction.CspI.dna
Bio.Restriction.Restriction.CstMI.dna
Bio.Restriction.Restriction.CviAII.dna
Bio.Restriction.Restriction.CviJI.dna
Bio.Restriction.Restriction.CviKI_1.dna
Bio.Restriction.Restriction.CviQI.dna
Bio.Restriction.Restriction.CviRI.dna
Bio.Restriction.Restriction.Dde51507I.dna
Bio.Restriction.Restriction.DdeI.dna
Bio.Restriction.Restriction.DinI.dna
Bio.Restriction.Restriction.DpnI.dna
Bio.Restriction.Restriction.DpnII.dna
Bio.Restriction.Restriction.DraI.dna
Bio.Restriction.Restriction.DraII.dna
Bio.Restriction.Restriction.DraIII.dna
Bio.Restriction.Restriction.DraRI.dna
Bio.Restriction.Restriction.DrdI.dna
Bio.Restriction.Restriction.DrdII.dna
Bio.Restriction.Restriction.DrdIV.dna
Bio.Restriction.Restriction.DriI.dna
Bio.Restriction.Restriction.DsaI.dna
Bio.Restriction.Restriction.DseDI.dna
Bio.Restriction.Restriction.DvuIII.dna
Bio.Restriction.Restriction.EaeI.dna
Bio.Restriction.Restriction.EagI.dna
Bio.Restriction.Restriction.Eam1104I.dna
Bio.Restriction.Restriction.Eam1105I.dna
Bio.Restriction.Restriction.EarI.dna
Bio.Restriction.Restriction.EciI.dna
Bio.Restriction.Restriction.Ecl136II.dna
Bio.Restriction.Restriction.Ecl234I.dna
Bio.Restriction.Restriction.Ecl35734I.dna
Bio.Restriction.Restriction.EclXI.dna
Bio.Restriction.Restriction.Eco105I.dna
Bio.Restriction.Restriction.Eco130I.dna
Bio.Restriction.Restriction.Eco147I.dna
Bio.Restriction.Restriction.Eco24I.dna
Bio.Restriction.Restriction.Eco31I.dna
Bio.Restriction.Restriction.Eco32I.dna
Bio.Restriction.Restriction.Eco43896II.dna
Bio.Restriction.Restriction.Eco4465II.dna
Bio.Restriction.Restriction.Eco47I.dna
Bio.Restriction.Restriction.Eco47III.dna
Bio.Restriction.Restriction.Eco52I.dna
Bio.Restriction.Restriction.Eco53kI.dna
Bio.Restriction.Restriction.Eco57I.dna
Bio.Restriction.Restriction.Eco57MI.dna
Bio.Restriction.Restriction.Eco72I.dna
Bio.Restriction.Restriction.Eco81I.dna
Bio.Restriction.Restriction.Eco88I.dna
Bio.Restriction.Restriction.Eco91I.dna
Bio.Restriction.Restriction.EcoBLMcrX.dna
Bio.Restriction.Restriction.EcoE1140I.dna
Bio.Restriction.Restriction.EcoHI.dna
Bio.Restriction.Restriction.EcoHSI.dna
Bio.Restriction.Restriction.EcoICRI.dna
Bio.Restriction.Restriction.EcoMVII.dna
Bio.Restriction.Restriction.EcoNI.dna
Bio.Restriction.Restriction.EcoNIH6II.dna
Bio.Restriction.Restriction.EcoO109I.dna
Bio.Restriction.Restriction.EcoO65I.dna
Bio.Restriction.Restriction.EcoRI.dna
Bio.Restriction.Restriction.EcoRII.dna
Bio.Restriction.Restriction.EcoRV.dna
Bio.Restriction.Restriction.EcoT14I.dna
Bio.Restriction.Restriction.EcoT22I.dna
Bio.Restriction.Restriction.EcoT38I.dna
Bio.Restriction.Restriction.EgeI.dna
Bio.Restriction.Restriction.EheI.dna
Bio.Restriction.Restriction.Eli8509II.dna
Bio.Restriction.Restriction.ErhI.dna
Bio.Restriction.Restriction.EsaBC3I.dna
Bio.Restriction.Restriction.EsaSSI.dna
Bio.Restriction.Restriction.Esp3007I.dna
Bio.Restriction.Restriction.Esp3I.dna
Bio.Restriction.Restriction.EspI.dna
Bio.Restriction.Restriction.FaeI.dna
Bio.Restriction.Restriction.FaiI.dna
Bio.Restriction.Restriction.FalI.dna
Bio.Restriction.Restriction.FaqI.dna
Bio.Restriction.Restriction.FatI.dna
Bio.Restriction.Restriction.FauI.dna
Bio.Restriction.Restriction.FauNDI.dna
Bio.Restriction.Restriction.FbaI.dna
Bio.Restriction.Restriction.FblI.dna
Bio.Restriction.Restriction.Fco1691IV.dna
Bio.Restriction.Restriction.FinI.dna
Bio.Restriction.Restriction.FmuI.dna
Bio.Restriction.Restriction.Fnu4HI.dna
Bio.Restriction.Restriction.FnuDII.dna
Bio.Restriction.Restriction.FokI.dna
Bio.Restriction.Restriction.FriOI.dna
Bio.Restriction.Restriction.FseI.dna
Bio.Restriction.Restriction.Fsp4HI.dna
Bio.Restriction.Restriction.FspAI.dna
Bio.Restriction.Restriction.FspBI.dna
Bio.Restriction.Restriction.FspEI.dna
Bio.Restriction.Restriction.FspI.dna
Bio.Restriction.Restriction.FspPK15I.dna
Bio.Restriction.Restriction.FtnUV.dna
Bio.Restriction.Restriction.GauT27I.dna
Bio.Restriction.Restriction.Gba708II.dna
Bio.Restriction.Restriction.GdiII.dna
Bio.Restriction.Restriction.GlaI.dna
Bio.Restriction.Restriction.GluI.dna
Bio.Restriction.Restriction.GsaI.dna
Bio.Restriction.Restriction.GsuI.dna
Bio.Restriction.Restriction.HaeI.dna
Bio.Restriction.Restriction.HaeII.dna
Bio.Restriction.Restriction.HaeIII.dna
Bio.Restriction.Restriction.HapII.dna
Bio.Restriction.Restriction.HauII.dna
Bio.Restriction.Restriction.HbaII.dna
Bio.Restriction.Restriction.HdeNY26I.dna
Bio.Restriction.Restriction.HdeZA17I.dna
Bio.Restriction.Restriction.HgaI.dna
Bio.Restriction.Restriction.HgiAI.dna
Bio.Restriction.Restriction.HgiCI.dna
Bio.Restriction.Restriction.HgiEII.dna
Bio.Restriction.Restriction.HgiJII.dna
Bio.Restriction.Restriction.HhaI.dna
Bio.Restriction.Restriction.Hin1I.dna
Bio.Restriction.Restriction.Hin1II.dna
Bio.Restriction.Restriction.Hin4I.dna
Bio.Restriction.Restriction.Hin4II.dna
Bio.Restriction.Restriction.Hin6I.dna
Bio.Restriction.Restriction.HinP1I.dna
Bio.Restriction.Restriction.HincII.dna
Bio.Restriction.Restriction.HindII.dna
Bio.Restriction.Restriction.HindIII.dna
Bio.Restriction.Restriction.HinfI.dna
Bio.Restriction.Restriction.HpaI.dna
Bio.Restriction.Restriction.HpaII.dna
Bio.Restriction.Restriction.HphI.dna
Bio.Restriction.Restriction.Hpy166II.dna
Bio.Restriction.Restriction.Hpy178III.dna
Bio.Restriction.Restriction.Hpy188I.dna
Bio.Restriction.Restriction.Hpy188III.dna
Bio.Restriction.Restriction.Hpy300XI.dna
Bio.Restriction.Restriction.Hpy8I.dna
Bio.Restriction.Restriction.Hpy99I.dna
Bio.Restriction.Restriction.Hpy99XIII.dna
Bio.Restriction.Restriction.Hpy99XIV.dna
Bio.Restriction.Restriction.Hpy99XIV_mut1.dna
Bio.Restriction.Restriction.Hpy99XXII.dna
Bio.Restriction.Restriction.HpyAV.dna
Bio.Restriction.Restriction.HpyAXIV.dna
Bio.Restriction.Restriction.HpyAXVIII.dna
Bio.Restriction.Restriction.HpyAXVI_mut1.dna
Bio.Restriction.Restriction.HpyAXVI_mut2.dna
Bio.Restriction.Restriction.HpyCH4III.dna
Bio.Restriction.Restriction.HpyCH4IV.dna
Bio.Restriction.Restriction.HpyCH4V.dna
Bio.Restriction.Restriction.HpyF10VI.dna
Bio.Restriction.Restriction.HpyF3I.dna
Bio.Restriction.Restriction.HpySE526I.dna
Bio.Restriction.Restriction.HpyUM032XIII.dna
Bio.Restriction.Restriction.HpyUM032XIII_mut1.dna
Bio.Restriction.Restriction.HpyUM032XIV.dna
Bio.Restriction.Restriction.HpyUM037X.dna
Bio.Restriction.Restriction.Hsp92I.dna
Bio.Restriction.Restriction.Hsp92II.dna
Bio.Restriction.Restriction.HspAI.dna
Bio.Restriction.Restriction.HspMHR1II.dna
Bio.Restriction.Restriction.Jma19592I.dna
Bio.Restriction.Restriction.Jma19592II.dna
Bio.Restriction.Restriction.Jsp2502II.dna
Bio.Restriction.Restriction.KasI.dna
Bio.Restriction.Restriction.KflI.dna
Bio.Restriction.Restriction.Kor51II.dna
Bio.Restriction.Restriction.Kpn156V.dna
Bio.Restriction.Restriction.Kpn2I.dna
Bio.Restriction.Restriction.Kpn327I.dna
Bio.Restriction.Restriction.KpnI.dna
Bio.Restriction.Restriction.KpnNH25III.dna
Bio.Restriction.Restriction.KpnNIH30III.dna
Bio.Restriction.Restriction.KpnNIH50I.dna
Bio.Restriction.Restriction.KroI.dna
Bio.Restriction.Restriction.Ksp22I.dna
Bio.Restriction.Restriction.Ksp632I.dna
Bio.Restriction.Restriction.KspAI.dna
Bio.Restriction.Restriction.KspI.dna
Bio.Restriction.Restriction.Kzo9I.dna
Bio.Restriction.Restriction.Lba2029III.dna
Bio.Restriction.Restriction.Lde4408II.dna
Bio.Restriction.Restriction.LguI.dna
Bio.Restriction.Restriction.LlaG50I.dna
Bio.Restriction.Restriction.LmnI.dna
Bio.Restriction.Restriction.Lmo370I.dna
Bio.Restriction.Restriction.Lmo911II.dna
Bio.Restriction.Restriction.Lpl1004II.dna
Bio.Restriction.Restriction.LpnI.dna
Bio.Restriction.Restriction.LpnPI.dna
Bio.Restriction.Restriction.Lra68I.dna
Bio.Restriction.Restriction.LsaDS4I.dna
Bio.Restriction.Restriction.Lsp1109I.dna
Bio.Restriction.Restriction.Lsp48III.dna
Bio.Restriction.Restriction.Lsp6406VI.dna
Bio.Restriction.Restriction.LweI.dna
Bio.Restriction.Restriction.MabI.dna
Bio.Restriction.Restriction.MaeI.dna
Bio.Restriction.Restriction.MaeII.dna
Bio.Restriction.Restriction.MaeIII.dna
Bio.Restriction.Restriction.MalI.dna
Bio.Restriction.Restriction.MaqI.dna
Bio.Restriction.Restriction.MauBI.dna
Bio.Restriction.Restriction.Mba11I.dna
Bio.Restriction.Restriction.MbiI.dna
Bio.Restriction.Restriction.MboI.dna
Bio.Restriction.Restriction.MboII.dna
Bio.Restriction.Restriction.McaTI.dna
Bio.Restriction.Restriction.Mcr10I.dna
Bio.Restriction.Restriction.McrI.dna
Bio.Restriction.Restriction.MfeI.dna
Bio.Restriction.Restriction.MflI.dna
Bio.Restriction.Restriction.MhlI.dna
Bio.Restriction.Restriction.MjaIV.dna
Bio.Restriction.Restriction.MkaDII.dna
Bio.Restriction.Restriction.MlsI.dna
Bio.Restriction.Restriction.Mlu211III.dna
Bio.Restriction.Restriction.MluCI.dna
Bio.Restriction.Restriction.MluI.dna
Bio.Restriction.Restriction.MluNI.dna
Bio.Restriction.Restriction.Mly113I.dna
Bio.Restriction.Restriction.MlyI.dna
Bio.Restriction.Restriction.MmeI.dna
Bio.Restriction.Restriction.MnlI.dna
Bio.Restriction.Restriction.Mox20I.dna
Bio.Restriction.Restriction.Mph1103I.dna
Bio.Restriction.Restriction.MreI.dna
Bio.Restriction.Restriction.MroI.dna
Bio.Restriction.Restriction.MroNI.dna
Bio.Restriction.Restriction.MroXI.dna
Bio.Restriction.Restriction.MscI.dna
Bio.Restriction.Restriction.MseI.dna
Bio.Restriction.Restriction.MslI.dna
Bio.Restriction.Restriction.Msp20I.dna
Bio.Restriction.Restriction.MspA1I.dna
Bio.Restriction.Restriction.MspCI.dna
Bio.Restriction.Restriction.MspGI.dna
Bio.Restriction.Restriction.MspI.dna
Bio.Restriction.Restriction.MspI7II.dna
Bio.Restriction.Restriction.MspI7IV.dna
Bio.Restriction.Restriction.MspJI.dna
Bio.Restriction.Restriction.MspR9I.dna
Bio.Restriction.Restriction.MspSC27II.dna
Bio.Restriction.Restriction.MssI.dna
Bio.Restriction.Restriction.MstI.dna
Bio.Restriction.Restriction.MteI.dna
Bio.Restriction.Restriction.MtuHN878II.dna
Bio.Restriction.Restriction.MunI.dna
Bio.Restriction.Restriction.Mva1269I.dna
Bio.Restriction.Restriction.MvaI.dna
Bio.Restriction.Restriction.MvnI.dna
Bio.Restriction.Restriction.MwoI.dna
Bio.Restriction.Restriction.NaeI.dna
Bio.Restriction.Restriction.Nal45188II.dna
Bio.Restriction.Restriction.NarI.dna
Bio.Restriction.Restriction.Nbr128II.dna
Bio.Restriction.Restriction.NciI.dna
Bio.Restriction.Restriction.NcoI.dna
Bio.Restriction.Restriction.NdeI.dna
Bio.Restriction.Restriction.NdeII.dna
Bio.Restriction.Restriction.NgoAVII.dna
Bio.Restriction.Restriction.NgoAVIII.dna
Bio.Restriction.Restriction.NgoMIV.dna
Bio.Restriction.Restriction.NhaXI.dna
Bio.Restriction.Restriction.NheI.dna
Bio.Restriction.Restriction.NhoI.dna
Bio.Restriction.Restriction.NlaCI.dna
Bio.Restriction.Restriction.NlaIII.dna
Bio.Restriction.Restriction.NlaIV.dna
Bio.Restriction.Restriction.Nli3877I.dna
Bio.Restriction.Restriction.NmeA6CIII.dna
Bio.Restriction.Restriction.NmeAIII.dna
Bio.Restriction.Restriction.NmeDI.dna
Bio.Restriction.Restriction.NmuCI.dna
Bio.Restriction.Restriction.NotI.dna
Bio.Restriction.Restriction.NpeUS61II.dna
Bio.Restriction.Restriction.NruI.dna
Bio.Restriction.Restriction.NsbI.dna
Bio.Restriction.Restriction.NsiI.dna
Bio.Restriction.Restriction.NspBII.dna
Bio.Restriction.Restriction.NspI.dna
Bio.Restriction.Restriction.NspV.dna
Bio.Restriction.Restriction.ObaBS10I.dna
Bio.Restriction.Restriction.OgrI.dna
Bio.Restriction.Restriction.OliI.dna
Bio.Restriction.Restriction.OspHL35III.dna
Bio.Restriction.Restriction.PabI.dna
Bio.Restriction.Restriction.Pac19842II.dna
Bio.Restriction.Restriction.PacI.dna
Bio.Restriction.Restriction.PacIII.dna
Bio.Restriction.Restriction.PaeI.dna
Bio.Restriction.Restriction.PaeR7I.dna
Bio.Restriction.Restriction.PagI.dna
Bio.Restriction.Restriction.Pal408I.dna
Bio.Restriction.Restriction.PalAI.dna
Bio.Restriction.Restriction.PasI.dna
Bio.Restriction.Restriction.PauI.dna
Bio.Restriction.Restriction.Pba2294I.dna
Bio.Restriction.Restriction.PcaII.dna
Bio.Restriction.Restriction.PceI.dna
Bio.Restriction.Restriction.PciI.dna
Bio.Restriction.Restriction.PciSI.dna
Bio.Restriction.Restriction.Pcr308II.dna
Bio.Restriction.Restriction.PcsI.dna
Bio.Restriction.Restriction.PctI.dna
Bio.Restriction.Restriction.Pdi8503III.dna
Bio.Restriction.Restriction.PdiI.dna
Bio.Restriction.Restriction.PdmI.dna
Bio.Restriction.Restriction.Pdu1735I.dna
Bio.Restriction.Restriction.PenI.dna
Bio.Restriction.Restriction.PfeI.dna
Bio.Restriction.Restriction.Pfl1108I.dna
Bio.Restriction.Restriction.Pfl23II.dna
Bio.Restriction.Restriction.Pfl8569I.dna
Bio.Restriction.Restriction.PflFI.dna
Bio.Restriction.Restriction.PflMI.dna
Bio.Restriction.Restriction.PflPt14I.dna
Bio.Restriction.Restriction.PfoI.dna
Bio.Restriction.Restriction.PfrJS12IV.dna
Bio.Restriction.Restriction.PfrJS12V.dna
Bio.Restriction.Restriction.PfrJS15III.dna
Bio.Restriction.Restriction.Pin17FIII.dna
Bio.Restriction.Restriction.PinAI.dna
Bio.Restriction.Restriction.PinP23II.dna
Bio.Restriction.Restriction.PinP59III.dna
Bio.Restriction.Restriction.PkrI.dna
Bio.Restriction.Restriction.PlaDI.dna
Bio.Restriction.Restriction.Ple19I.dna
Bio.Restriction.Restriction.PleI.dna
Bio.Restriction.Restriction.PliMI.dna
Bio.Restriction.Restriction.PluTI.dna
Bio.Restriction.Restriction.PmaCI.dna
Bio.Restriction.Restriction.PmeI.dna
Bio.Restriction.Restriction.PmlI.dna
Bio.Restriction.Restriction.PpiI.dna
Bio.Restriction.Restriction.PpiP13II.dna
Bio.Restriction.Restriction.PpsI.dna
Bio.Restriction.Restriction.Ppu10I.dna
Bio.Restriction.Restriction.Ppu21I.dna
Bio.Restriction.Restriction.PpuMI.dna
Bio.Restriction.Restriction.PscI.dna
Bio.Restriction.Restriction.Pse18267I.dna
Bio.Restriction.Restriction.PshAI.dna
Bio.Restriction.Restriction.PshBI.dna
Bio.Restriction.Restriction.PsiI.dna
Bio.Restriction.Restriction.Psp0357II.dna
Bio.Restriction.Restriction.Psp03I.dna
Bio.Restriction.Restriction.Psp124BI.dna
Bio.Restriction.Restriction.Psp1406I.dna
Bio.Restriction.Restriction.Psp5II.dna
Bio.Restriction.Restriction.Psp6I.dna
Bio.Restriction.Restriction.PspCI.dna
Bio.Restriction.Restriction.PspEI.dna
Bio.Restriction.Restriction.PspFI.dna
Bio.Restriction.Restriction.PspGI.dna
Bio.Restriction.Restriction.PspLI.dna
Bio.Restriction.Restriction.PspN4I.dna
Bio.Restriction.Restriction.PspOMI.dna
Bio.Restriction.Restriction.PspOMII.dna
Bio.Restriction.Restriction.PspPI.dna
Bio.Restriction.Restriction.PspPPI.dna
Bio.Restriction.Restriction.PspPRI.dna
Bio.Restriction.Restriction.PspXI.dna
Bio.Restriction.Restriction.PsrI.dna
Bio.Restriction.Restriction.PssI.dna
Bio.Restriction.Restriction.Pst14472I.dna
Bio.Restriction.Restriction.Pst145I.dna
Bio.Restriction.Restriction.Pst273I.dna
Bio.Restriction.Restriction.PstI.dna
Bio.Restriction.Restriction.PstNI.dna
Bio.Restriction.Restriction.PsuGI.dna
Bio.Restriction.Restriction.PsuI.dna
Bio.Restriction.Restriction.PsyI.dna
Bio.Restriction.Restriction.PteI.dna
Bio.Restriction.Restriction.PvuI.dna
Bio.Restriction.Restriction.PvuII.dna
Bio.Restriction.Restriction.Rba2021I.dna
Bio.Restriction.Restriction.RceI.dna
Bio.Restriction.Restriction.RdeGBI.dna
Bio.Restriction.Restriction.RdeGBII.dna
Bio.Restriction.Restriction.RdeGBIII.dna
Bio.Restriction.Restriction.RflFIII.dna
Bio.Restriction.Restriction.RgaI.dna
Bio.Restriction.Restriction.RigI.dna
Bio.Restriction.Restriction.RlaI.dna
Bio.Restriction.Restriction.RlaII.dna
Bio.Restriction.Restriction.RleAI.dna
Bio.Restriction.Restriction.Rmu369III.dna
Bio.Restriction.Restriction.RpaB5I.dna
Bio.Restriction.Restriction.RpaBI.dna
Bio.Restriction.Restriction.RpaI.dna
Bio.Restriction.Restriction.RpaTI.dna
Bio.Restriction.Restriction.RruI.dna
Bio.Restriction.Restriction.RsaI.dna
Bio.Restriction.Restriction.RsaNI.dna
Bio.Restriction.Restriction.RseI.dna
Bio.Restriction.Restriction.Rsp008IV.dna
Bio.Restriction.Restriction.Rsp008V.dna
Bio.Restriction.Restriction.Rsp531II.dna
Bio.Restriction.Restriction.RspPBTS2III.dna
Bio.Restriction.Restriction.Rsr2I.dna
Bio.Restriction.Restriction.RsrII.dna
Bio.Restriction.Restriction.Rtr1953I.dna
Bio.Restriction.Restriction.SacI.dna
Bio.Restriction.Restriction.SacII.dna
Bio.Restriction.Restriction.Saf8902III.dna
Bio.Restriction.Restriction.Sag901I.dna
Bio.Restriction.Restriction.SalI.dna
Bio.Restriction.Restriction.SanDI.dna
Bio.Restriction.Restriction.SapI.dna
Bio.Restriction.Restriction.SaqAI.dna
Bio.Restriction.Restriction.SatI.dna
Bio.Restriction.Restriction.Sau3AI.dna
Bio.Restriction.Restriction.Sau96I.dna
Bio.Restriction.Restriction.SauI.dna
Bio.Restriction.Restriction.Sba460II.dna
Bio.Restriction.Restriction.SbfI.dna
Bio.Restriction.Restriction.Sbo46I.dna
Bio.Restriction.Restriction.ScaI.dna
Bio.Restriction.Restriction.SchI.dna
Bio.Restriction.Restriction.SciI.dna
Bio.Restriction.Restriction.ScoDS2II.dna
Bio.Restriction.Restriction.ScrFI.dna
Bio.Restriction.Restriction.SdaI.dna
Bio.Restriction.Restriction.SdeAI.dna
Bio.Restriction.Restriction.SdeOSI.dna
Bio.Restriction.Restriction.SduI.dna
Bio.Restriction.Restriction.SecI.dna
Bio.Restriction.Restriction.SelI.dna
Bio.Restriction.Restriction.Sen17963III.dna
Bio.Restriction.Restriction.SenA1673III.dna
Bio.Restriction.Restriction.SenSARA26III.dna
Bio.Restriction.Restriction.SenTFIV.dna
Bio.Restriction.Restriction.SetI.dna
Bio.Restriction.Restriction.SexAI.dna
Bio.Restriction.Restriction.SfaAI.dna
Bio.Restriction.Restriction.SfaNI.dna
Bio.Restriction.Restriction.SfcI.dna
Bio.Restriction.Restriction.SfeI.dna
Bio.Restriction.Restriction.SfiI.dna
Bio.Restriction.Restriction.SfoI.dna
Bio.Restriction.Restriction.Sfr274I.dna
Bio.Restriction.Restriction.Sfr303I.dna
Bio.Restriction.Restriction.SfuI.dna
Bio.Restriction.Restriction.SgeI.dna
Bio.Restriction.Restriction.SgfI.dna
Bio.Restriction.Restriction.SgrAI.dna
Bio.Restriction.Restriction.SgrBI.dna
Bio.Restriction.Restriction.SgrDI.dna
Bio.Restriction.Restriction.SgrTI.dna
Bio.Restriction.Restriction.SgsI.dna
Bio.Restriction.Restriction.SimI.dna
Bio.Restriction.Restriction.SinI.dna
Bio.Restriction.Restriction.SlaI.dna
Bio.Restriction.Restriction.SmaI.dna
Bio.Restriction.Restriction.SmaUMH5I.dna
Bio.Restriction.Restriction.SmaUMH8I.dna
Bio.Restriction.Restriction.SmiI.dna
Bio.Restriction.Restriction.SmiMI.dna
Bio.Restriction.Restriction.SmlI.dna
Bio.Restriction.Restriction.SmoI.dna
Bio.Restriction.Restriction.SnaBI.dna
Bio.Restriction.Restriction.SnaI.dna
Bio.Restriction.Restriction.Sno506I.dna
Bio.Restriction.Restriction.SpeI.dna
Bio.Restriction.Restriction.SphI.dna
Bio.Restriction.Restriction.SplI.dna
Bio.Restriction.Restriction.SpnRII.dna
Bio.Restriction.Restriction.SpoDI.dna
Bio.Restriction.Restriction.SrfI.dna
Bio.Restriction.Restriction.Sse232I.dna
Bio.Restriction.Restriction.Sse8387I.dna
Bio.Restriction.Restriction.Sse8647I.dna
Bio.Restriction.Restriction.Sse9I.dna
Bio.Restriction.Restriction.SseBI.dna
Bio.Restriction.Restriction.SsiI.dna
Bio.Restriction.Restriction.Ssp6803IV.dna
Bio.Restriction.Restriction.Ssp714II.dna
Bio.Restriction.Restriction.SspD5I.dna
Bio.Restriction.Restriction.SspDI.dna
Bio.Restriction.Restriction.SspI.dna
Bio.Restriction.Restriction.SspMI.dna
Bio.Restriction.Restriction.SstE37I.dna
Bio.Restriction.Restriction.SstI.dna
Bio.Restriction.Restriction.Sth132I.dna
Bio.Restriction.Restriction.Sth20745III.dna
Bio.Restriction.Restriction.Sth302II.dna
Bio.Restriction.Restriction.SthSt3II.dna
Bio.Restriction.Restriction.StsI.dna
Bio.Restriction.Restriction.StuI.dna
Bio.Restriction.Restriction.StyD4I.dna
Bio.Restriction.Restriction.StyI.dna
Bio.Restriction.Restriction.SurP32aII.dna
Bio.Restriction.Restriction.SwaI.dna
Bio.Restriction.Restriction.TaaI.dna
Bio.Restriction.Restriction.TaiI.dna
Bio.Restriction.Restriction.TaqI.dna
Bio.Restriction.Restriction.TaqII.dna
Bio.Restriction.Restriction.TaqIII.dna
Bio.Restriction.Restriction.TasI.dna
Bio.Restriction.Restriction.TatI.dna
Bio.Restriction.Restriction.TauI.dna
Bio.Restriction.Restriction.TfiI.dna
Bio.Restriction.Restriction.TpyTP2I.dna
Bio.Restriction.Restriction.Tru1I.dna
Bio.Restriction.Restriction.Tru9I.dna
Bio.Restriction.Restriction.TscAI.dna
Bio.Restriction.Restriction.TseFI.dna
Bio.Restriction.Restriction.TseI.dna
Bio.Restriction.Restriction.TsoI.dna
Bio.Restriction.Restriction.Tsp45I.dna
Bio.Restriction.Restriction.Tsp4CI.dna
Bio.Restriction.Restriction.TspARh3I.dna
Bio.Restriction.Restriction.TspDTI.dna
Bio.Restriction.Restriction.TspEI.dna
Bio.Restriction.Restriction.TspGWI.dna
Bio.Restriction.Restriction.TspMI.dna
Bio.Restriction.Restriction.TspRI.dna
Bio.Restriction.Restriction.TssI.dna
Bio.Restriction.Restriction.TstI.dna
Bio.Restriction.Restriction.TsuI.dna
Bio.Restriction.Restriction.Tth111I.dna
Bio.Restriction.Restriction.Tth111II.dna
Bio.Restriction.Restriction.UbaF11I.dna
Bio.Restriction.Restriction.UbaF12I.dna
Bio.Restriction.Restriction.UbaF13I.dna
Bio.Restriction.Restriction.UbaF14I.dna
Bio.Restriction.Restriction.UbaF9I.dna
Bio.Restriction.Restriction.UbaPI.dna
Bio.Restriction.Restriction.UcoMSI.dna
Bio.Restriction.Restriction.UnbI.dna
Bio.Restriction.Restriction.Van9116I.dna
Bio.Restriction.Restriction.Van91I.dna
Bio.Restriction.Restriction.VchE4II.dna
Bio.Restriction.Restriction.Vdi96II.dna
Bio.Restriction.Restriction.Vha464I.dna
Bio.Restriction.Restriction.VneI.dna
Bio.Restriction.Restriction.VpaK11AI.dna
Bio.Restriction.Restriction.VpaK11BI.dna
Bio.Restriction.Restriction.VspI.dna
Bio.Restriction.Restriction.Vtu19109I.dna
Bio.Restriction.Restriction.WviI.dna
Bio.Restriction.Restriction.XagI.dna
Bio.Restriction.Restriction.XapI.dna
Bio.Restriction.Restriction.XbaI.dna
Bio.Restriction.Restriction.Xca85IV.dna
Bio.Restriction.Restriction.XceI.dna
Bio.Restriction.Restriction.XcmI.dna
Bio.Restriction.Restriction.XhoI.dna
Bio.Restriction.Restriction.XhoII.dna
Bio.Restriction.Restriction.XmaI.dna
Bio.Restriction.Restriction.XmaIII.dna
Bio.Restriction.Restriction.XmaJI.dna
Bio.Restriction.Restriction.XmiI.dna
Bio.Restriction.Restriction.XmnI.dna
Bio.Restriction.Restriction.XspI.dna
Bio.Restriction.Restriction.YkrI.dna
Bio.Restriction.Restriction.ZraI.dna
Bio.Restriction.Restriction.ZrmI.dna
Bio.Restriction.Restriction.Zsp2I.dna" class="py-name" href="#" onclick="return doclink('link-987', 'dna', 'link-981');">dna</a></tt><tt class="py-op">.</tt><tt id="link-988" class="py-name"><a title="Bio.Seq.MutableSeq.toseq
BioSQL.BioSeq.DBSeq.toseq" class="py-name" href="#" onclick="return doclink('link-988', 'toseq', 'link-978');">toseq</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-989" class="py-name"><a title="Bio.Seq.Seq.transcribe
Bio.Seq.UnknownSeq.transcribe
Bio.Seq.transcribe" class="py-name" href="#" onclick="return doclink('link-989', 'transcribe', 'link-587');">transcribe</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L2614"></a><tt class="py-lineno">2614</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L2615"></a><tt class="py-lineno">2615</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-990" class="py-name"><a title="Bio.Restriction.Restriction.AanI.dna
Bio.Restriction.Restriction.AarI.dna
Bio.Restriction.Restriction.AasI.dna
Bio.Restriction.Restriction.AatII.dna
Bio.Restriction.Restriction.Aba6411II.dna
Bio.Restriction.Restriction.AbaB8342IV.dna
Bio.Restriction.Restriction.AbaCIII.dna
Bio.Restriction.Restriction.AbaSI.dna
Bio.Restriction.Restriction.AbaUMB2I.dna
Bio.Restriction.Restriction.AbsI.dna
Bio.Restriction.Restriction.Acc16I.dna
Bio.Restriction.Restriction.Acc36I.dna
Bio.Restriction.Restriction.Acc65I.dna
Bio.Restriction.Restriction.Acc65V.dna
Bio.Restriction.Restriction.AccB1I.dna
Bio.Restriction.Restriction.AccB7I.dna
Bio.Restriction.Restriction.AccBSI.dna
Bio.Restriction.Restriction.AccI.dna
Bio.Restriction.Restriction.AccII.dna
Bio.Restriction.Restriction.AccIII.dna
Bio.Restriction.Restriction.AceIII.dna
Bio.Restriction.Restriction.AchA6III.dna
Bio.Restriction.Restriction.AciI.dna
Bio.Restriction.Restriction.AclI.dna
Bio.Restriction.Restriction.AclWI.dna
Bio.Restriction.Restriction.Aco12261II.dna
Bio.Restriction.Restriction.AcoI.dna
Bio.Restriction.Restriction.AcoY31II.dna
Bio.Restriction.Restriction.AcsI.dna
Bio.Restriction.Restriction.AcuI.dna
Bio.Restriction.Restriction.AcvI.dna
Bio.Restriction.Restriction.AcyI.dna
Bio.Restriction.Restriction.AdeI.dna
Bio.Restriction.Restriction.Adh6U21I.dna
Bio.Restriction.Restriction.AfaI.dna
Bio.Restriction.Restriction.AfeI.dna
Bio.Restriction.Restriction.AfiI.dna
Bio.Restriction.Restriction.AflII.dna
Bio.Restriction.Restriction.AflIII.dna
Bio.Restriction.Restriction.AgeI.dna
Bio.Restriction.Restriction.AgsI.dna
Bio.Restriction.Restriction.AhaIII.dna
Bio.Restriction.Restriction.AhdI.dna
Bio.Restriction.Restriction.AhlI.dna
Bio.Restriction.Restriction.AhyRBAHI.dna
Bio.Restriction.Restriction.AhyYL17I.dna
Bio.Restriction.Restriction.AjiI.dna
Bio.Restriction.Restriction.AjnI.dna
Bio.Restriction.Restriction.AjuI.dna
Bio.Restriction.Restriction.AleI.dna
Bio.Restriction.Restriction.AlfI.dna
Bio.Restriction.Restriction.AloI.dna
Bio.Restriction.Restriction.AluBI.dna
Bio.Restriction.Restriction.AluI.dna
Bio.Restriction.Restriction.Alw21I.dna
Bio.Restriction.Restriction.Alw26I.dna
Bio.Restriction.Restriction.Alw44I.dna
Bio.Restriction.Restriction.AlwFI.dna
Bio.Restriction.Restriction.AlwI.dna
Bio.Restriction.Restriction.AlwNI.dna
Bio.Restriction.Restriction.Ama87I.dna
Bio.Restriction.Restriction.AmaCSI.dna
Bio.Restriction.Restriction.Aod1I.dna
Bio.Restriction.Restriction.Aor13HI.dna
Bio.Restriction.Restriction.Aor51HI.dna
Bio.Restriction.Restriction.AoxI.dna
Bio.Restriction.Restriction.ApaBI.dna
Bio.Restriction.Restriction.ApaI.dna
Bio.Restriction.Restriction.ApaLI.dna
Bio.Restriction.Restriction.ApeKI.dna
Bio.Restriction.Restriction.ApoI.dna
Bio.Restriction.Restriction.ApyPI.dna
Bio.Restriction.Restriction.AquII.dna
Bio.Restriction.Restriction.AquIII.dna
Bio.Restriction.Restriction.AquIV.dna
Bio.Restriction.Restriction.ArsI.dna
Bio.Restriction.Restriction.AscI.dna
Bio.Restriction.Restriction.AseI.dna
Bio.Restriction.Restriction.Asi256I.dna
Bio.Restriction.Restriction.AsiGI.dna
Bio.Restriction.Restriction.AsiSI.dna
Bio.Restriction.Restriction.Asp103I.dna
Bio.Restriction.Restriction.Asp114pII.dna
Bio.Restriction.Restriction.Asp700I.dna
Bio.Restriction.Restriction.Asp718I.dna
Bio.Restriction.Restriction.AspA2I.dna
Bio.Restriction.Restriction.AspBHI.dna
Bio.Restriction.Restriction.AspDUT2V.dna
Bio.Restriction.Restriction.AspJHL3II.dna
Bio.Restriction.Restriction.AspLEI.dna
Bio.Restriction.Restriction.AspNIH4III.dna
Bio.Restriction.Restriction.AspS9I.dna
Bio.Restriction.Restriction.AspSLV7III.dna
Bio.Restriction.Restriction.Asu14238IV.dna
Bio.Restriction.Restriction.AsuC2I.dna
Bio.Restriction.Restriction.AsuHPI.dna
Bio.Restriction.Restriction.AsuI.dna
Bio.Restriction.Restriction.AsuII.dna
Bio.Restriction.Restriction.AsuNHI.dna
Bio.Restriction.Restriction.AteTI.dna
Bio.Restriction.Restriction.AvaI.dna
Bio.Restriction.Restriction.AvaII.dna
Bio.Restriction.Restriction.AvaIII.dna
Bio.Restriction.Restriction.AvrII.dna
Bio.Restriction.Restriction.Awo1030IV.dna
Bio.Restriction.Restriction.AxyI.dna
Bio.Restriction.Restriction.BaeGI.dna
Bio.Restriction.Restriction.BaeI.dna
Bio.Restriction.Restriction.Bag18758I.dna
Bio.Restriction.Restriction.BalI.dna
Bio.Restriction.Restriction.BamHI.dna
Bio.Restriction.Restriction.BanI.dna
Bio.Restriction.Restriction.BanII.dna
Bio.Restriction.Restriction.BanLI.dna
Bio.Restriction.Restriction.BarI.dna
Bio.Restriction.Restriction.Bau1417V.dna
Bio.Restriction.Restriction.BauI.dna
Bio.Restriction.Restriction.Bbr52II.dna
Bio.Restriction.Restriction.Bbr57III.dna
Bio.Restriction.Restriction.Bbr7017II.dna
Bio.Restriction.Restriction.Bbr7017III.dna
Bio.Restriction.Restriction.BbrPI.dna
Bio.Restriction.Restriction.BbsI.dna
Bio.Restriction.Restriction.BbuB31I.dna
Bio.Restriction.Restriction.BbuB31II.dna
Bio.Restriction.Restriction.Bbv12I.dna
Bio.Restriction.Restriction.BbvCI.dna
Bio.Restriction.Restriction.BbvI.dna
Bio.Restriction.Restriction.BbvII.dna
Bio.Restriction.Restriction.BccI.dna
Bio.Restriction.Restriction.Bce3081I.dna
Bio.Restriction.Restriction.Bce83I.dna
Bio.Restriction.Restriction.BceAI.dna
Bio.Restriction.Restriction.BceSIV.dna
Bio.Restriction.Restriction.BcefI.dna
Bio.Restriction.Restriction.BcgI.dna
Bio.Restriction.Restriction.BciT130I.dna
Bio.Restriction.Restriction.BciVI.dna
Bio.Restriction.Restriction.BclI.dna
Bio.Restriction.Restriction.BcnI.dna
Bio.Restriction.Restriction.BcoDI.dna
Bio.Restriction.Restriction.BcuI.dna
Bio.Restriction.Restriction.BdaI.dna
Bio.Restriction.Restriction.BetI.dna
Bio.Restriction.Restriction.BfaI.dna
Bio.Restriction.Restriction.BfaSII.dna
Bio.Restriction.Restriction.BfiI.dna
Bio.Restriction.Restriction.BfmI.dna
Bio.Restriction.Restriction.BfoI.dna
Bio.Restriction.Restriction.BfrI.dna
Bio.Restriction.Restriction.BfuAI.dna
Bio.Restriction.Restriction.BfuI.dna
Bio.Restriction.Restriction.Bga514I.dna
Bio.Restriction.Restriction.BglI.dna
Bio.Restriction.Restriction.BglII.dna
Bio.Restriction.Restriction.BinI.dna
Bio.Restriction.Restriction.BisI.dna
Bio.Restriction.Restriction.BkrAM31DI.dna
Bio.Restriction.Restriction.Ble402II.dna
Bio.Restriction.Restriction.BlnI.dna
Bio.Restriction.Restriction.BloAII.dna
Bio.Restriction.Restriction.BlpI.dna
Bio.Restriction.Restriction.BlsI.dna
Bio.Restriction.Restriction.BmcAI.dna
Bio.Restriction.Restriction.Bme1390I.dna
Bio.Restriction.Restriction.Bme18I.dna
Bio.Restriction.Restriction.BmeDI.dna
Bio.Restriction.Restriction.BmeRI.dna
Bio.Restriction.Restriction.BmeT110I.dna
Bio.Restriction.Restriction.BmgBI.dna
Bio.Restriction.Restriction.BmgI.dna
Bio.Restriction.Restriction.BmgT120I.dna
Bio.Restriction.Restriction.BmiI.dna
Bio.Restriction.Restriction.BmrFI.dna
Bio.Restriction.Restriction.BmrI.dna
Bio.Restriction.Restriction.BmsI.dna
Bio.Restriction.Restriction.BmtI.dna
Bio.Restriction.Restriction.BmuI.dna
Bio.Restriction.Restriction.BoxI.dna
Bio.Restriction.Restriction.BpiI.dna
Bio.Restriction.Restriction.BplI.dna
Bio.Restriction.Restriction.BpmI.dna
Bio.Restriction.Restriction.Bpu10I.dna
Bio.Restriction.Restriction.Bpu1102I.dna
Bio.Restriction.Restriction.Bpu14I.dna
Bio.Restriction.Restriction.BpuEI.dna
Bio.Restriction.Restriction.BpuMI.dna
Bio.Restriction.Restriction.Bsa29I.dna
Bio.Restriction.Restriction.BsaAI.dna
Bio.Restriction.Restriction.BsaBI.dna
Bio.Restriction.Restriction.BsaHI.dna
Bio.Restriction.Restriction.BsaI.dna
Bio.Restriction.Restriction.BsaJI.dna
Bio.Restriction.Restriction.BsaWI.dna
Bio.Restriction.Restriction.BsaXI.dna
Bio.Restriction.Restriction.BsbI.dna
Bio.Restriction.Restriction.Bsc4I.dna
Bio.Restriction.Restriction.BscAI.dna
Bio.Restriction.Restriction.BscGI.dna
Bio.Restriction.Restriction.Bse118I.dna
Bio.Restriction.Restriction.Bse1I.dna
Bio.Restriction.Restriction.Bse21I.dna
Bio.Restriction.Restriction.Bse3DI.dna
Bio.Restriction.Restriction.Bse8I.dna
Bio.Restriction.Restriction.BseAI.dna
Bio.Restriction.Restriction.BseBI.dna
Bio.Restriction.Restriction.BseCI.dna
Bio.Restriction.Restriction.BseDI.dna
Bio.Restriction.Restriction.BseGI.dna
Bio.Restriction.Restriction.BseJI.dna
Bio.Restriction.Restriction.BseLI.dna
Bio.Restriction.Restriction.BseMI.dna
Bio.Restriction.Restriction.BseMII.dna
Bio.Restriction.Restriction.BseNI.dna
Bio.Restriction.Restriction.BsePI.dna
Bio.Restriction.Restriction.BseRI.dna
Bio.Restriction.Restriction.BseSI.dna
Bio.Restriction.Restriction.BseX3I.dna
Bio.Restriction.Restriction.BseXI.dna
Bio.Restriction.Restriction.BseYI.dna
Bio.Restriction.Restriction.BsgI.dna
Bio.Restriction.Restriction.Bsh1236I.dna
Bio.Restriction.Restriction.Bsh1285I.dna
Bio.Restriction.Restriction.BshFI.dna
Bio.Restriction.Restriction.BshNI.dna
Bio.Restriction.Restriction.BshTI.dna
Bio.Restriction.Restriction.BshVI.dna
Bio.Restriction.Restriction.BsiEI.dna
Bio.Restriction.Restriction.BsiHKAI.dna
Bio.Restriction.Restriction.BsiHKCI.dna
Bio.Restriction.Restriction.BsiI.dna
Bio.Restriction.Restriction.BsiSI.dna
Bio.Restriction.Restriction.BsiWI.dna
Bio.Restriction.Restriction.BsiYI.dna
Bio.Restriction.Restriction.BslFI.dna
Bio.Restriction.Restriction.BslI.dna
Bio.Restriction.Restriction.BsmAI.dna
Bio.Restriction.Restriction.BsmBI.dna
Bio.Restriction.Restriction.BsmFI.dna
Bio.Restriction.Restriction.BsmI.dna
Bio.Restriction.Restriction.BsnI.dna
Bio.Restriction.Restriction.Bso31I.dna
Bio.Restriction.Restriction.BsoBI.dna
Bio.Restriction.Restriction.Bsp119I.dna
Bio.Restriction.Restriction.Bsp120I.dna
Bio.Restriction.Restriction.Bsp1286I.dna
Bio.Restriction.Restriction.Bsp13I.dna
Bio.Restriction.Restriction.Bsp1407I.dna
Bio.Restriction.Restriction.Bsp143I.dna
Bio.Restriction.Restriction.Bsp1720I.dna
Bio.Restriction.Restriction.Bsp19I.dna
Bio.Restriction.Restriction.Bsp24I.dna
Bio.Restriction.Restriction.Bsp3004IV.dna
Bio.Restriction.Restriction.Bsp460III.dna
Bio.Restriction.Restriction.Bsp68I.dna
Bio.Restriction.Restriction.BspACI.dna
Bio.Restriction.Restriction.BspANI.dna
Bio.Restriction.Restriction.BspCNI.dna
Bio.Restriction.Restriction.BspD6I.dna
Bio.Restriction.Restriction.BspDI.dna
Bio.Restriction.Restriction.BspEI.dna
Bio.Restriction.Restriction.BspFNI.dna
Bio.Restriction.Restriction.BspGI.dna
Bio.Restriction.Restriction.BspHI.dna
Bio.Restriction.Restriction.BspLI.dna
Bio.Restriction.Restriction.BspLU11I.dna
Bio.Restriction.Restriction.BspMAI.dna
Bio.Restriction.Restriction.BspMI.dna
Bio.Restriction.Restriction.BspMII.dna
Bio.Restriction.Restriction.BspNCI.dna
Bio.Restriction.Restriction.BspOI.dna
Bio.Restriction.Restriction.BspPI.dna
Bio.Restriction.Restriction.BspQI.dna
Bio.Restriction.Restriction.BspT104I.dna
Bio.Restriction.Restriction.BspT107I.dna
Bio.Restriction.Restriction.BspTI.dna
Bio.Restriction.Restriction.BspTNI.dna
Bio.Restriction.Restriction.BsrBI.dna
Bio.Restriction.Restriction.BsrDI.dna
Bio.Restriction.Restriction.BsrFI.dna
Bio.Restriction.Restriction.BsrGI.dna
Bio.Restriction.Restriction.BsrI.dna
Bio.Restriction.Restriction.BssAI.dna
Bio.Restriction.Restriction.BssECI.dna
Bio.Restriction.Restriction.BssHII.dna
Bio.Restriction.Restriction.BssMI.dna
Bio.Restriction.Restriction.BssNAI.dna
Bio.Restriction.Restriction.BssNI.dna
Bio.Restriction.Restriction.BssSI.dna
Bio.Restriction.Restriction.BssT1I.dna
Bio.Restriction.Restriction.Bst1107I.dna
Bio.Restriction.Restriction.Bst2BI.dna
Bio.Restriction.Restriction.Bst2UI.dna
Bio.Restriction.Restriction.Bst4CI.dna
Bio.Restriction.Restriction.Bst6I.dna
Bio.Restriction.Restriction.BstACI.dna
Bio.Restriction.Restriction.BstAFI.dna
Bio.Restriction.Restriction.BstAPI.dna
Bio.Restriction.Restriction.BstAUI.dna
Bio.Restriction.Restriction.BstBAI.dna
Bio.Restriction.Restriction.BstBI.dna
Bio.Restriction.Restriction.BstC8I.dna
Bio.Restriction.Restriction.BstDEI.dna
Bio.Restriction.Restriction.BstDSI.dna
Bio.Restriction.Restriction.BstEII.dna
Bio.Restriction.Restriction.BstENI.dna
Bio.Restriction.Restriction.BstF5I.dna
Bio.Restriction.Restriction.BstFNI.dna
Bio.Restriction.Restriction.BstH2I.dna
Bio.Restriction.Restriction.BstHHI.dna
Bio.Restriction.Restriction.BstKTI.dna
Bio.Restriction.Restriction.BstMAI.dna
Bio.Restriction.Restriction.BstMBI.dna
Bio.Restriction.Restriction.BstMCI.dna
Bio.Restriction.Restriction.BstMWI.dna
Bio.Restriction.Restriction.BstNI.dna
Bio.Restriction.Restriction.BstNSI.dna
Bio.Restriction.Restriction.BstPAI.dna
Bio.Restriction.Restriction.BstPI.dna
Bio.Restriction.Restriction.BstSCI.dna
Bio.Restriction.Restriction.BstSFI.dna
Bio.Restriction.Restriction.BstSLI.dna
Bio.Restriction.Restriction.BstSNI.dna
Bio.Restriction.Restriction.BstUI.dna
Bio.Restriction.Restriction.BstV1I.dna
Bio.Restriction.Restriction.BstV2I.dna
Bio.Restriction.Restriction.BstX2I.dna
Bio.Restriction.Restriction.BstXI.dna
Bio.Restriction.Restriction.BstYI.dna
Bio.Restriction.Restriction.BstZ17I.dna
Bio.Restriction.Restriction.BstZI.dna
Bio.Restriction.Restriction.Bsu15I.dna
Bio.Restriction.Restriction.Bsu36I.dna
Bio.Restriction.Restriction.BsuI.dna
Bio.Restriction.Restriction.BsuRI.dna
Bio.Restriction.Restriction.BsuTUI.dna
Bio.Restriction.Restriction.BtgI.dna
Bio.Restriction.Restriction.BtgZI.dna
Bio.Restriction.Restriction.BthCI.dna
Bio.Restriction.Restriction.BtrI.dna
Bio.Restriction.Restriction.BtsCI.dna
Bio.Restriction.Restriction.BtsI.dna
Bio.Restriction.Restriction.BtsIMutI.dna
Bio.Restriction.Restriction.BtuMI.dna
Bio.Restriction.Restriction.Bve1B23I.dna
Bio.Restriction.Restriction.BveI.dna
Bio.Restriction.Restriction.Cac8I.dna
Bio.Restriction.Restriction.CaiI.dna
Bio.Restriction.Restriction.Cal14237I.dna
Bio.Restriction.Restriction.CalB3II.dna
Bio.Restriction.Restriction.Cau10061II.dna
Bio.Restriction.Restriction.CauII.dna
Bio.Restriction.Restriction.Cba13II.dna
Bio.Restriction.Restriction.Cba16038I.dna
Bio.Restriction.Restriction.Cbo67071IV.dna
Bio.Restriction.Restriction.Cch467III.dna
Bio.Restriction.Restriction.CchII.dna
Bio.Restriction.Restriction.CchIII.dna
Bio.Restriction.Restriction.CciI.dna
Bio.Restriction.Restriction.CciNI.dna
Bio.Restriction.Restriction.Cco14983V.dna
Bio.Restriction.Restriction.Cco14983VI.dna
Bio.Restriction.Restriction.CcrNAIII.dna
Bio.Restriction.Restriction.Cdi11397I.dna
Bio.Restriction.Restriction.CdiI.dna
Bio.Restriction.Restriction.CdpI.dna
Bio.Restriction.Restriction.Cdu23823II.dna
Bio.Restriction.Restriction.CfoI.dna
Bio.Restriction.Restriction.Cfr10I.dna
Bio.Restriction.Restriction.Cfr13I.dna
Bio.Restriction.Restriction.Cfr42I.dna
Bio.Restriction.Restriction.Cfr9I.dna
Bio.Restriction.Restriction.CfrI.dna
Bio.Restriction.Restriction.CfrMH13II.dna
Bio.Restriction.Restriction.CfrMH16VI.dna
Bio.Restriction.Restriction.Cfupf3II.dna
Bio.Restriction.Restriction.Cgl13032I.dna
Bio.Restriction.Restriction.Cgl13032II.dna
Bio.Restriction.Restriction.ChaI.dna
Bio.Restriction.Restriction.Cje265V.dna
Bio.Restriction.Restriction.Cje54107III.dna
Bio.Restriction.Restriction.CjeFIII.dna
Bio.Restriction.Restriction.CjeFV.dna
Bio.Restriction.Restriction.CjeI.dna
Bio.Restriction.Restriction.CjeNII.dna
Bio.Restriction.Restriction.CjeNIII.dna
Bio.Restriction.Restriction.CjeNV.dna
Bio.Restriction.Restriction.CjeP659IV.dna
Bio.Restriction.Restriction.CjePI.dna
Bio.Restriction.Restriction.CjuI.dna
Bio.Restriction.Restriction.CjuII.dna
Bio.Restriction.Restriction.Cla11845III.dna
Bio.Restriction.Restriction.ClaI.dna
Bio.Restriction.Restriction.Cly7489II.dna
Bio.Restriction.Restriction.Cma23826I.dna
Bio.Restriction.Restriction.CpoI.dna
Bio.Restriction.Restriction.CseI.dna
Bio.Restriction.Restriction.CsiI.dna
Bio.Restriction.Restriction.Csp2014I.dna
Bio.Restriction.Restriction.Csp6I.dna
Bio.Restriction.Restriction.CspAI.dna
Bio.Restriction.Restriction.CspCI.dna
Bio.Restriction.Restriction.CspI.dna
Bio.Restriction.Restriction.CstMI.dna
Bio.Restriction.Restriction.CviAII.dna
Bio.Restriction.Restriction.CviJI.dna
Bio.Restriction.Restriction.CviKI_1.dna
Bio.Restriction.Restriction.CviQI.dna
Bio.Restriction.Restriction.CviRI.dna
Bio.Restriction.Restriction.Dde51507I.dna
Bio.Restriction.Restriction.DdeI.dna
Bio.Restriction.Restriction.DinI.dna
Bio.Restriction.Restriction.DpnI.dna
Bio.Restriction.Restriction.DpnII.dna
Bio.Restriction.Restriction.DraI.dna
Bio.Restriction.Restriction.DraII.dna
Bio.Restriction.Restriction.DraIII.dna
Bio.Restriction.Restriction.DraRI.dna
Bio.Restriction.Restriction.DrdI.dna
Bio.Restriction.Restriction.DrdII.dna
Bio.Restriction.Restriction.DrdIV.dna
Bio.Restriction.Restriction.DriI.dna
Bio.Restriction.Restriction.DsaI.dna
Bio.Restriction.Restriction.DseDI.dna
Bio.Restriction.Restriction.DvuIII.dna
Bio.Restriction.Restriction.EaeI.dna
Bio.Restriction.Restriction.EagI.dna
Bio.Restriction.Restriction.Eam1104I.dna
Bio.Restriction.Restriction.Eam1105I.dna
Bio.Restriction.Restriction.EarI.dna
Bio.Restriction.Restriction.EciI.dna
Bio.Restriction.Restriction.Ecl136II.dna
Bio.Restriction.Restriction.Ecl234I.dna
Bio.Restriction.Restriction.Ecl35734I.dna
Bio.Restriction.Restriction.EclXI.dna
Bio.Restriction.Restriction.Eco105I.dna
Bio.Restriction.Restriction.Eco130I.dna
Bio.Restriction.Restriction.Eco147I.dna
Bio.Restriction.Restriction.Eco24I.dna
Bio.Restriction.Restriction.Eco31I.dna
Bio.Restriction.Restriction.Eco32I.dna
Bio.Restriction.Restriction.Eco43896II.dna
Bio.Restriction.Restriction.Eco4465II.dna
Bio.Restriction.Restriction.Eco47I.dna
Bio.Restriction.Restriction.Eco47III.dna
Bio.Restriction.Restriction.Eco52I.dna
Bio.Restriction.Restriction.Eco53kI.dna
Bio.Restriction.Restriction.Eco57I.dna
Bio.Restriction.Restriction.Eco57MI.dna
Bio.Restriction.Restriction.Eco72I.dna
Bio.Restriction.Restriction.Eco81I.dna
Bio.Restriction.Restriction.Eco88I.dna
Bio.Restriction.Restriction.Eco91I.dna
Bio.Restriction.Restriction.EcoBLMcrX.dna
Bio.Restriction.Restriction.EcoE1140I.dna
Bio.Restriction.Restriction.EcoHI.dna
Bio.Restriction.Restriction.EcoHSI.dna
Bio.Restriction.Restriction.EcoICRI.dna
Bio.Restriction.Restriction.EcoMVII.dna
Bio.Restriction.Restriction.EcoNI.dna
Bio.Restriction.Restriction.EcoNIH6II.dna
Bio.Restriction.Restriction.EcoO109I.dna
Bio.Restriction.Restriction.EcoO65I.dna
Bio.Restriction.Restriction.EcoRI.dna
Bio.Restriction.Restriction.EcoRII.dna
Bio.Restriction.Restriction.EcoRV.dna
Bio.Restriction.Restriction.EcoT14I.dna
Bio.Restriction.Restriction.EcoT22I.dna
Bio.Restriction.Restriction.EcoT38I.dna
Bio.Restriction.Restriction.EgeI.dna
Bio.Restriction.Restriction.EheI.dna
Bio.Restriction.Restriction.Eli8509II.dna
Bio.Restriction.Restriction.ErhI.dna
Bio.Restriction.Restriction.EsaBC3I.dna
Bio.Restriction.Restriction.EsaSSI.dna
Bio.Restriction.Restriction.Esp3007I.dna
Bio.Restriction.Restriction.Esp3I.dna
Bio.Restriction.Restriction.EspI.dna
Bio.Restriction.Restriction.FaeI.dna
Bio.Restriction.Restriction.FaiI.dna
Bio.Restriction.Restriction.FalI.dna
Bio.Restriction.Restriction.FaqI.dna
Bio.Restriction.Restriction.FatI.dna
Bio.Restriction.Restriction.FauI.dna
Bio.Restriction.Restriction.FauNDI.dna
Bio.Restriction.Restriction.FbaI.dna
Bio.Restriction.Restriction.FblI.dna
Bio.Restriction.Restriction.Fco1691IV.dna
Bio.Restriction.Restriction.FinI.dna
Bio.Restriction.Restriction.FmuI.dna
Bio.Restriction.Restriction.Fnu4HI.dna
Bio.Restriction.Restriction.FnuDII.dna
Bio.Restriction.Restriction.FokI.dna
Bio.Restriction.Restriction.FriOI.dna
Bio.Restriction.Restriction.FseI.dna
Bio.Restriction.Restriction.Fsp4HI.dna
Bio.Restriction.Restriction.FspAI.dna
Bio.Restriction.Restriction.FspBI.dna
Bio.Restriction.Restriction.FspEI.dna
Bio.Restriction.Restriction.FspI.dna
Bio.Restriction.Restriction.FspPK15I.dna
Bio.Restriction.Restriction.FtnUV.dna
Bio.Restriction.Restriction.GauT27I.dna
Bio.Restriction.Restriction.Gba708II.dna
Bio.Restriction.Restriction.GdiII.dna
Bio.Restriction.Restriction.GlaI.dna
Bio.Restriction.Restriction.GluI.dna
Bio.Restriction.Restriction.GsaI.dna
Bio.Restriction.Restriction.GsuI.dna
Bio.Restriction.Restriction.HaeI.dna
Bio.Restriction.Restriction.HaeII.dna
Bio.Restriction.Restriction.HaeIII.dna
Bio.Restriction.Restriction.HapII.dna
Bio.Restriction.Restriction.HauII.dna
Bio.Restriction.Restriction.HbaII.dna
Bio.Restriction.Restriction.HdeNY26I.dna
Bio.Restriction.Restriction.HdeZA17I.dna
Bio.Restriction.Restriction.HgaI.dna
Bio.Restriction.Restriction.HgiAI.dna
Bio.Restriction.Restriction.HgiCI.dna
Bio.Restriction.Restriction.HgiEII.dna
Bio.Restriction.Restriction.HgiJII.dna
Bio.Restriction.Restriction.HhaI.dna
Bio.Restriction.Restriction.Hin1I.dna
Bio.Restriction.Restriction.Hin1II.dna
Bio.Restriction.Restriction.Hin4I.dna
Bio.Restriction.Restriction.Hin4II.dna
Bio.Restriction.Restriction.Hin6I.dna
Bio.Restriction.Restriction.HinP1I.dna
Bio.Restriction.Restriction.HincII.dna
Bio.Restriction.Restriction.HindII.dna
Bio.Restriction.Restriction.HindIII.dna
Bio.Restriction.Restriction.HinfI.dna
Bio.Restriction.Restriction.HpaI.dna
Bio.Restriction.Restriction.HpaII.dna
Bio.Restriction.Restriction.HphI.dna
Bio.Restriction.Restriction.Hpy166II.dna
Bio.Restriction.Restriction.Hpy178III.dna
Bio.Restriction.Restriction.Hpy188I.dna
Bio.Restriction.Restriction.Hpy188III.dna
Bio.Restriction.Restriction.Hpy300XI.dna
Bio.Restriction.Restriction.Hpy8I.dna
Bio.Restriction.Restriction.Hpy99I.dna
Bio.Restriction.Restriction.Hpy99XIII.dna
Bio.Restriction.Restriction.Hpy99XIV.dna
Bio.Restriction.Restriction.Hpy99XIV_mut1.dna
Bio.Restriction.Restriction.Hpy99XXII.dna
Bio.Restriction.Restriction.HpyAV.dna
Bio.Restriction.Restriction.HpyAXIV.dna
Bio.Restriction.Restriction.HpyAXVIII.dna
Bio.Restriction.Restriction.HpyAXVI_mut1.dna
Bio.Restriction.Restriction.HpyAXVI_mut2.dna
Bio.Restriction.Restriction.HpyCH4III.dna
Bio.Restriction.Restriction.HpyCH4IV.dna
Bio.Restriction.Restriction.HpyCH4V.dna
Bio.Restriction.Restriction.HpyF10VI.dna
Bio.Restriction.Restriction.HpyF3I.dna
Bio.Restriction.Restriction.HpySE526I.dna
Bio.Restriction.Restriction.HpyUM032XIII.dna
Bio.Restriction.Restriction.HpyUM032XIII_mut1.dna
Bio.Restriction.Restriction.HpyUM032XIV.dna
Bio.Restriction.Restriction.HpyUM037X.dna
Bio.Restriction.Restriction.Hsp92I.dna
Bio.Restriction.Restriction.Hsp92II.dna
Bio.Restriction.Restriction.HspAI.dna
Bio.Restriction.Restriction.HspMHR1II.dna
Bio.Restriction.Restriction.Jma19592I.dna
Bio.Restriction.Restriction.Jma19592II.dna
Bio.Restriction.Restriction.Jsp2502II.dna
Bio.Restriction.Restriction.KasI.dna
Bio.Restriction.Restriction.KflI.dna
Bio.Restriction.Restriction.Kor51II.dna
Bio.Restriction.Restriction.Kpn156V.dna
Bio.Restriction.Restriction.Kpn2I.dna
Bio.Restriction.Restriction.Kpn327I.dna
Bio.Restriction.Restriction.KpnI.dna
Bio.Restriction.Restriction.KpnNH25III.dna
Bio.Restriction.Restriction.KpnNIH30III.dna
Bio.Restriction.Restriction.KpnNIH50I.dna
Bio.Restriction.Restriction.KroI.dna
Bio.Restriction.Restriction.Ksp22I.dna
Bio.Restriction.Restriction.Ksp632I.dna
Bio.Restriction.Restriction.KspAI.dna
Bio.Restriction.Restriction.KspI.dna
Bio.Restriction.Restriction.Kzo9I.dna
Bio.Restriction.Restriction.Lba2029III.dna
Bio.Restriction.Restriction.Lde4408II.dna
Bio.Restriction.Restriction.LguI.dna
Bio.Restriction.Restriction.LlaG50I.dna
Bio.Restriction.Restriction.LmnI.dna
Bio.Restriction.Restriction.Lmo370I.dna
Bio.Restriction.Restriction.Lmo911II.dna
Bio.Restriction.Restriction.Lpl1004II.dna
Bio.Restriction.Restriction.LpnI.dna
Bio.Restriction.Restriction.LpnPI.dna
Bio.Restriction.Restriction.Lra68I.dna
Bio.Restriction.Restriction.LsaDS4I.dna
Bio.Restriction.Restriction.Lsp1109I.dna
Bio.Restriction.Restriction.Lsp48III.dna
Bio.Restriction.Restriction.Lsp6406VI.dna
Bio.Restriction.Restriction.LweI.dna
Bio.Restriction.Restriction.MabI.dna
Bio.Restriction.Restriction.MaeI.dna
Bio.Restriction.Restriction.MaeII.dna
Bio.Restriction.Restriction.MaeIII.dna
Bio.Restriction.Restriction.MalI.dna
Bio.Restriction.Restriction.MaqI.dna
Bio.Restriction.Restriction.MauBI.dna
Bio.Restriction.Restriction.Mba11I.dna
Bio.Restriction.Restriction.MbiI.dna
Bio.Restriction.Restriction.MboI.dna
Bio.Restriction.Restriction.MboII.dna
Bio.Restriction.Restriction.McaTI.dna
Bio.Restriction.Restriction.Mcr10I.dna
Bio.Restriction.Restriction.McrI.dna
Bio.Restriction.Restriction.MfeI.dna
Bio.Restriction.Restriction.MflI.dna
Bio.Restriction.Restriction.MhlI.dna
Bio.Restriction.Restriction.MjaIV.dna
Bio.Restriction.Restriction.MkaDII.dna
Bio.Restriction.Restriction.MlsI.dna
Bio.Restriction.Restriction.Mlu211III.dna
Bio.Restriction.Restriction.MluCI.dna
Bio.Restriction.Restriction.MluI.dna
Bio.Restriction.Restriction.MluNI.dna
Bio.Restriction.Restriction.Mly113I.dna
Bio.Restriction.Restriction.MlyI.dna
Bio.Restriction.Restriction.MmeI.dna
Bio.Restriction.Restriction.MnlI.dna
Bio.Restriction.Restriction.Mox20I.dna
Bio.Restriction.Restriction.Mph1103I.dna
Bio.Restriction.Restriction.MreI.dna
Bio.Restriction.Restriction.MroI.dna
Bio.Restriction.Restriction.MroNI.dna
Bio.Restriction.Restriction.MroXI.dna
Bio.Restriction.Restriction.MscI.dna
Bio.Restriction.Restriction.MseI.dna
Bio.Restriction.Restriction.MslI.dna
Bio.Restriction.Restriction.Msp20I.dna
Bio.Restriction.Restriction.MspA1I.dna
Bio.Restriction.Restriction.MspCI.dna
Bio.Restriction.Restriction.MspGI.dna
Bio.Restriction.Restriction.MspI.dna
Bio.Restriction.Restriction.MspI7II.dna
Bio.Restriction.Restriction.MspI7IV.dna
Bio.Restriction.Restriction.MspJI.dna
Bio.Restriction.Restriction.MspR9I.dna
Bio.Restriction.Restriction.MspSC27II.dna
Bio.Restriction.Restriction.MssI.dna
Bio.Restriction.Restriction.MstI.dna
Bio.Restriction.Restriction.MteI.dna
Bio.Restriction.Restriction.MtuHN878II.dna
Bio.Restriction.Restriction.MunI.dna
Bio.Restriction.Restriction.Mva1269I.dna
Bio.Restriction.Restriction.MvaI.dna
Bio.Restriction.Restriction.MvnI.dna
Bio.Restriction.Restriction.MwoI.dna
Bio.Restriction.Restriction.NaeI.dna
Bio.Restriction.Restriction.Nal45188II.dna
Bio.Restriction.Restriction.NarI.dna
Bio.Restriction.Restriction.Nbr128II.dna
Bio.Restriction.Restriction.NciI.dna
Bio.Restriction.Restriction.NcoI.dna
Bio.Restriction.Restriction.NdeI.dna
Bio.Restriction.Restriction.NdeII.dna
Bio.Restriction.Restriction.NgoAVII.dna
Bio.Restriction.Restriction.NgoAVIII.dna
Bio.Restriction.Restriction.NgoMIV.dna
Bio.Restriction.Restriction.NhaXI.dna
Bio.Restriction.Restriction.NheI.dna
Bio.Restriction.Restriction.NhoI.dna
Bio.Restriction.Restriction.NlaCI.dna
Bio.Restriction.Restriction.NlaIII.dna
Bio.Restriction.Restriction.NlaIV.dna
Bio.Restriction.Restriction.Nli3877I.dna
Bio.Restriction.Restriction.NmeA6CIII.dna
Bio.Restriction.Restriction.NmeAIII.dna
Bio.Restriction.Restriction.NmeDI.dna
Bio.Restriction.Restriction.NmuCI.dna
Bio.Restriction.Restriction.NotI.dna
Bio.Restriction.Restriction.NpeUS61II.dna
Bio.Restriction.Restriction.NruI.dna
Bio.Restriction.Restriction.NsbI.dna
Bio.Restriction.Restriction.NsiI.dna
Bio.Restriction.Restriction.NspBII.dna
Bio.Restriction.Restriction.NspI.dna
Bio.Restriction.Restriction.NspV.dna
Bio.Restriction.Restriction.ObaBS10I.dna
Bio.Restriction.Restriction.OgrI.dna
Bio.Restriction.Restriction.OliI.dna
Bio.Restriction.Restriction.OspHL35III.dna
Bio.Restriction.Restriction.PabI.dna
Bio.Restriction.Restriction.Pac19842II.dna
Bio.Restriction.Restriction.PacI.dna
Bio.Restriction.Restriction.PacIII.dna
Bio.Restriction.Restriction.PaeI.dna
Bio.Restriction.Restriction.PaeR7I.dna
Bio.Restriction.Restriction.PagI.dna
Bio.Restriction.Restriction.Pal408I.dna
Bio.Restriction.Restriction.PalAI.dna
Bio.Restriction.Restriction.PasI.dna
Bio.Restriction.Restriction.PauI.dna
Bio.Restriction.Restriction.Pba2294I.dna
Bio.Restriction.Restriction.PcaII.dna
Bio.Restriction.Restriction.PceI.dna
Bio.Restriction.Restriction.PciI.dna
Bio.Restriction.Restriction.PciSI.dna
Bio.Restriction.Restriction.Pcr308II.dna
Bio.Restriction.Restriction.PcsI.dna
Bio.Restriction.Restriction.PctI.dna
Bio.Restriction.Restriction.Pdi8503III.dna
Bio.Restriction.Restriction.PdiI.dna
Bio.Restriction.Restriction.PdmI.dna
Bio.Restriction.Restriction.Pdu1735I.dna
Bio.Restriction.Restriction.PenI.dna
Bio.Restriction.Restriction.PfeI.dna
Bio.Restriction.Restriction.Pfl1108I.dna
Bio.Restriction.Restriction.Pfl23II.dna
Bio.Restriction.Restriction.Pfl8569I.dna
Bio.Restriction.Restriction.PflFI.dna
Bio.Restriction.Restriction.PflMI.dna
Bio.Restriction.Restriction.PflPt14I.dna
Bio.Restriction.Restriction.PfoI.dna
Bio.Restriction.Restriction.PfrJS12IV.dna
Bio.Restriction.Restriction.PfrJS12V.dna
Bio.Restriction.Restriction.PfrJS15III.dna
Bio.Restriction.Restriction.Pin17FIII.dna
Bio.Restriction.Restriction.PinAI.dna
Bio.Restriction.Restriction.PinP23II.dna
Bio.Restriction.Restriction.PinP59III.dna
Bio.Restriction.Restriction.PkrI.dna
Bio.Restriction.Restriction.PlaDI.dna
Bio.Restriction.Restriction.Ple19I.dna
Bio.Restriction.Restriction.PleI.dna
Bio.Restriction.Restriction.PliMI.dna
Bio.Restriction.Restriction.PluTI.dna
Bio.Restriction.Restriction.PmaCI.dna
Bio.Restriction.Restriction.PmeI.dna
Bio.Restriction.Restriction.PmlI.dna
Bio.Restriction.Restriction.PpiI.dna
Bio.Restriction.Restriction.PpiP13II.dna
Bio.Restriction.Restriction.PpsI.dna
Bio.Restriction.Restriction.Ppu10I.dna
Bio.Restriction.Restriction.Ppu21I.dna
Bio.Restriction.Restriction.PpuMI.dna
Bio.Restriction.Restriction.PscI.dna
Bio.Restriction.Restriction.Pse18267I.dna
Bio.Restriction.Restriction.PshAI.dna
Bio.Restriction.Restriction.PshBI.dna
Bio.Restriction.Restriction.PsiI.dna
Bio.Restriction.Restriction.Psp0357II.dna
Bio.Restriction.Restriction.Psp03I.dna
Bio.Restriction.Restriction.Psp124BI.dna
Bio.Restriction.Restriction.Psp1406I.dna
Bio.Restriction.Restriction.Psp5II.dna
Bio.Restriction.Restriction.Psp6I.dna
Bio.Restriction.Restriction.PspCI.dna
Bio.Restriction.Restriction.PspEI.dna
Bio.Restriction.Restriction.PspFI.dna
Bio.Restriction.Restriction.PspGI.dna
Bio.Restriction.Restriction.PspLI.dna
Bio.Restriction.Restriction.PspN4I.dna
Bio.Restriction.Restriction.PspOMI.dna
Bio.Restriction.Restriction.PspOMII.dna
Bio.Restriction.Restriction.PspPI.dna
Bio.Restriction.Restriction.PspPPI.dna
Bio.Restriction.Restriction.PspPRI.dna
Bio.Restriction.Restriction.PspXI.dna
Bio.Restriction.Restriction.PsrI.dna
Bio.Restriction.Restriction.PssI.dna
Bio.Restriction.Restriction.Pst14472I.dna
Bio.Restriction.Restriction.Pst145I.dna
Bio.Restriction.Restriction.Pst273I.dna
Bio.Restriction.Restriction.PstI.dna
Bio.Restriction.Restriction.PstNI.dna
Bio.Restriction.Restriction.PsuGI.dna
Bio.Restriction.Restriction.PsuI.dna
Bio.Restriction.Restriction.PsyI.dna
Bio.Restriction.Restriction.PteI.dna
Bio.Restriction.Restriction.PvuI.dna
Bio.Restriction.Restriction.PvuII.dna
Bio.Restriction.Restriction.Rba2021I.dna
Bio.Restriction.Restriction.RceI.dna
Bio.Restriction.Restriction.RdeGBI.dna
Bio.Restriction.Restriction.RdeGBII.dna
Bio.Restriction.Restriction.RdeGBIII.dna
Bio.Restriction.Restriction.RflFIII.dna
Bio.Restriction.Restriction.RgaI.dna
Bio.Restriction.Restriction.RigI.dna
Bio.Restriction.Restriction.RlaI.dna
Bio.Restriction.Restriction.RlaII.dna
Bio.Restriction.Restriction.RleAI.dna
Bio.Restriction.Restriction.Rmu369III.dna
Bio.Restriction.Restriction.RpaB5I.dna
Bio.Restriction.Restriction.RpaBI.dna
Bio.Restriction.Restriction.RpaI.dna
Bio.Restriction.Restriction.RpaTI.dna
Bio.Restriction.Restriction.RruI.dna
Bio.Restriction.Restriction.RsaI.dna
Bio.Restriction.Restriction.RsaNI.dna
Bio.Restriction.Restriction.RseI.dna
Bio.Restriction.Restriction.Rsp008IV.dna
Bio.Restriction.Restriction.Rsp008V.dna
Bio.Restriction.Restriction.Rsp531II.dna
Bio.Restriction.Restriction.RspPBTS2III.dna
Bio.Restriction.Restriction.Rsr2I.dna
Bio.Restriction.Restriction.RsrII.dna
Bio.Restriction.Restriction.Rtr1953I.dna
Bio.Restriction.Restriction.SacI.dna
Bio.Restriction.Restriction.SacII.dna
Bio.Restriction.Restriction.Saf8902III.dna
Bio.Restriction.Restriction.Sag901I.dna
Bio.Restriction.Restriction.SalI.dna
Bio.Restriction.Restriction.SanDI.dna
Bio.Restriction.Restriction.SapI.dna
Bio.Restriction.Restriction.SaqAI.dna
Bio.Restriction.Restriction.SatI.dna
Bio.Restriction.Restriction.Sau3AI.dna
Bio.Restriction.Restriction.Sau96I.dna
Bio.Restriction.Restriction.SauI.dna
Bio.Restriction.Restriction.Sba460II.dna
Bio.Restriction.Restriction.SbfI.dna
Bio.Restriction.Restriction.Sbo46I.dna
Bio.Restriction.Restriction.ScaI.dna
Bio.Restriction.Restriction.SchI.dna
Bio.Restriction.Restriction.SciI.dna
Bio.Restriction.Restriction.ScoDS2II.dna
Bio.Restriction.Restriction.ScrFI.dna
Bio.Restriction.Restriction.SdaI.dna
Bio.Restriction.Restriction.SdeAI.dna
Bio.Restriction.Restriction.SdeOSI.dna
Bio.Restriction.Restriction.SduI.dna
Bio.Restriction.Restriction.SecI.dna
Bio.Restriction.Restriction.SelI.dna
Bio.Restriction.Restriction.Sen17963III.dna
Bio.Restriction.Restriction.SenA1673III.dna
Bio.Restriction.Restriction.SenSARA26III.dna
Bio.Restriction.Restriction.SenTFIV.dna
Bio.Restriction.Restriction.SetI.dna
Bio.Restriction.Restriction.SexAI.dna
Bio.Restriction.Restriction.SfaAI.dna
Bio.Restriction.Restriction.SfaNI.dna
Bio.Restriction.Restriction.SfcI.dna
Bio.Restriction.Restriction.SfeI.dna
Bio.Restriction.Restriction.SfiI.dna
Bio.Restriction.Restriction.SfoI.dna
Bio.Restriction.Restriction.Sfr274I.dna
Bio.Restriction.Restriction.Sfr303I.dna
Bio.Restriction.Restriction.SfuI.dna
Bio.Restriction.Restriction.SgeI.dna
Bio.Restriction.Restriction.SgfI.dna
Bio.Restriction.Restriction.SgrAI.dna
Bio.Restriction.Restriction.SgrBI.dna
Bio.Restriction.Restriction.SgrDI.dna
Bio.Restriction.Restriction.SgrTI.dna
Bio.Restriction.Restriction.SgsI.dna
Bio.Restriction.Restriction.SimI.dna
Bio.Restriction.Restriction.SinI.dna
Bio.Restriction.Restriction.SlaI.dna
Bio.Restriction.Restriction.SmaI.dna
Bio.Restriction.Restriction.SmaUMH5I.dna
Bio.Restriction.Restriction.SmaUMH8I.dna
Bio.Restriction.Restriction.SmiI.dna
Bio.Restriction.Restriction.SmiMI.dna
Bio.Restriction.Restriction.SmlI.dna
Bio.Restriction.Restriction.SmoI.dna
Bio.Restriction.Restriction.SnaBI.dna
Bio.Restriction.Restriction.SnaI.dna
Bio.Restriction.Restriction.Sno506I.dna
Bio.Restriction.Restriction.SpeI.dna
Bio.Restriction.Restriction.SphI.dna
Bio.Restriction.Restriction.SplI.dna
Bio.Restriction.Restriction.SpnRII.dna
Bio.Restriction.Restriction.SpoDI.dna
Bio.Restriction.Restriction.SrfI.dna
Bio.Restriction.Restriction.Sse232I.dna
Bio.Restriction.Restriction.Sse8387I.dna
Bio.Restriction.Restriction.Sse8647I.dna
Bio.Restriction.Restriction.Sse9I.dna
Bio.Restriction.Restriction.SseBI.dna
Bio.Restriction.Restriction.SsiI.dna
Bio.Restriction.Restriction.Ssp6803IV.dna
Bio.Restriction.Restriction.Ssp714II.dna
Bio.Restriction.Restriction.SspD5I.dna
Bio.Restriction.Restriction.SspDI.dna
Bio.Restriction.Restriction.SspI.dna
Bio.Restriction.Restriction.SspMI.dna
Bio.Restriction.Restriction.SstE37I.dna
Bio.Restriction.Restriction.SstI.dna
Bio.Restriction.Restriction.Sth132I.dna
Bio.Restriction.Restriction.Sth20745III.dna
Bio.Restriction.Restriction.Sth302II.dna
Bio.Restriction.Restriction.SthSt3II.dna
Bio.Restriction.Restriction.StsI.dna
Bio.Restriction.Restriction.StuI.dna
Bio.Restriction.Restriction.StyD4I.dna
Bio.Restriction.Restriction.StyI.dna
Bio.Restriction.Restriction.SurP32aII.dna
Bio.Restriction.Restriction.SwaI.dna
Bio.Restriction.Restriction.TaaI.dna
Bio.Restriction.Restriction.TaiI.dna
Bio.Restriction.Restriction.TaqI.dna
Bio.Restriction.Restriction.TaqII.dna
Bio.Restriction.Restriction.TaqIII.dna
Bio.Restriction.Restriction.TasI.dna
Bio.Restriction.Restriction.TatI.dna
Bio.Restriction.Restriction.TauI.dna
Bio.Restriction.Restriction.TfiI.dna
Bio.Restriction.Restriction.TpyTP2I.dna
Bio.Restriction.Restriction.Tru1I.dna
Bio.Restriction.Restriction.Tru9I.dna
Bio.Restriction.Restriction.TscAI.dna
Bio.Restriction.Restriction.TseFI.dna
Bio.Restriction.Restriction.TseI.dna
Bio.Restriction.Restriction.TsoI.dna
Bio.Restriction.Restriction.Tsp45I.dna
Bio.Restriction.Restriction.Tsp4CI.dna
Bio.Restriction.Restriction.TspARh3I.dna
Bio.Restriction.Restriction.TspDTI.dna
Bio.Restriction.Restriction.TspEI.dna
Bio.Restriction.Restriction.TspGWI.dna
Bio.Restriction.Restriction.TspMI.dna
Bio.Restriction.Restriction.TspRI.dna
Bio.Restriction.Restriction.TssI.dna
Bio.Restriction.Restriction.TstI.dna
Bio.Restriction.Restriction.TsuI.dna
Bio.Restriction.Restriction.Tth111I.dna
Bio.Restriction.Restriction.Tth111II.dna
Bio.Restriction.Restriction.UbaF11I.dna
Bio.Restriction.Restriction.UbaF12I.dna
Bio.Restriction.Restriction.UbaF13I.dna
Bio.Restriction.Restriction.UbaF14I.dna
Bio.Restriction.Restriction.UbaF9I.dna
Bio.Restriction.Restriction.UbaPI.dna
Bio.Restriction.Restriction.UcoMSI.dna
Bio.Restriction.Restriction.UnbI.dna
Bio.Restriction.Restriction.Van9116I.dna
Bio.Restriction.Restriction.Van91I.dna
Bio.Restriction.Restriction.VchE4II.dna
Bio.Restriction.Restriction.Vdi96II.dna
Bio.Restriction.Restriction.Vha464I.dna
Bio.Restriction.Restriction.VneI.dna
Bio.Restriction.Restriction.VpaK11AI.dna
Bio.Restriction.Restriction.VpaK11BI.dna
Bio.Restriction.Restriction.VspI.dna
Bio.Restriction.Restriction.Vtu19109I.dna
Bio.Restriction.Restriction.WviI.dna
Bio.Restriction.Restriction.XagI.dna
Bio.Restriction.Restriction.XapI.dna
Bio.Restriction.Restriction.XbaI.dna
Bio.Restriction.Restriction.Xca85IV.dna
Bio.Restriction.Restriction.XceI.dna
Bio.Restriction.Restriction.XcmI.dna
Bio.Restriction.Restriction.XhoI.dna
Bio.Restriction.Restriction.XhoII.dna
Bio.Restriction.Restriction.XmaI.dna
Bio.Restriction.Restriction.XmaIII.dna
Bio.Restriction.Restriction.XmaJI.dna
Bio.Restriction.Restriction.XmiI.dna
Bio.Restriction.Restriction.XmnI.dna
Bio.Restriction.Restriction.XspI.dna
Bio.Restriction.Restriction.YkrI.dna
Bio.Restriction.Restriction.ZraI.dna
Bio.Restriction.Restriction.ZrmI.dna
Bio.Restriction.Restriction.Zsp2I.dna" class="py-name" href="#" onclick="return doclink('link-990', 'dna', 'link-981');">dna</a></tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"T"</tt><tt class="py-op">,</tt> <tt class="py-string">"U"</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"t"</tt><tt class="py-op">,</tt> <tt class="py-string">"u"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L2616"></a><tt class="py-lineno">2616</tt>  <tt class="py-line"> </tt>
<a name="L2617"></a><tt class="py-lineno">2617</tt>  <tt class="py-line"> </tt>
<a name="back_transcribe"></a><div id="back_transcribe-def"><a name="L2618"></a><tt class="py-lineno">2618</tt> <a class="py-toggle" href="#" id="back_transcribe-toggle" onclick="return toggle('back_transcribe');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq-module.html#back_transcribe">back_transcribe</a><tt class="py-op">(</tt><tt class="py-param">rna</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="back_transcribe-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="back_transcribe-expanded"><a name="L2619"></a><tt class="py-lineno">2619</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return the RNA sequence back-transcribed into DNA.</tt> </tt>
<a name="L2620"></a><tt class="py-lineno">2620</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2621"></a><tt class="py-lineno">2621</tt>  <tt class="py-line"><tt class="py-docstring">    If given a string, returns a new string object.</tt> </tt>
<a name="L2622"></a><tt class="py-lineno">2622</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2623"></a><tt class="py-lineno">2623</tt>  <tt class="py-line"><tt class="py-docstring">    Given a Seq or MutableSeq, returns a new Seq object with an RNA alphabet.</tt> </tt>
<a name="L2624"></a><tt class="py-lineno">2624</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2625"></a><tt class="py-lineno">2625</tt>  <tt class="py-line"><tt class="py-docstring">    Trying to transcribe a protein or DNA sequence raises an exception.</tt> </tt>
<a name="L2626"></a><tt class="py-lineno">2626</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2627"></a><tt class="py-lineno">2627</tt>  <tt class="py-line"><tt class="py-docstring">    e.g.</tt> </tt>
<a name="L2628"></a><tt class="py-lineno">2628</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2629"></a><tt class="py-lineno">2629</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; back_transcribe("ACUGN")</tt> </tt>
<a name="L2630"></a><tt class="py-lineno">2630</tt>  <tt class="py-line"><tt class="py-docstring">    'ACTGN'</tt> </tt>
<a name="L2631"></a><tt class="py-lineno">2631</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L2632"></a><tt class="py-lineno">2632</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">rna</tt><tt class="py-op">,</tt> <tt id="link-991" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-991', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2633"></a><tt class="py-lineno">2633</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">rna</tt><tt class="py-op">.</tt><tt id="link-992" class="py-name"><a title="Bio.Seq.Seq.back_transcribe
Bio.Seq.UnknownSeq.back_transcribe
Bio.Seq.back_transcribe" class="py-name" href="#" onclick="return doclink('link-992', 'back_transcribe', 'link-592');">back_transcribe</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L2634"></a><tt class="py-lineno">2634</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">rna</tt><tt class="py-op">,</tt> <tt id="link-993" class="py-name"><a title="Bio.Seq.MutableSeq" class="py-name" href="#" onclick="return doclink('link-993', 'MutableSeq', 'link-83');">MutableSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2635"></a><tt class="py-lineno">2635</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">rna</tt><tt class="py-op">.</tt><tt id="link-994" class="py-name"><a title="Bio.Seq.MutableSeq.toseq
BioSQL.BioSeq.DBSeq.toseq" class="py-name" href="#" onclick="return doclink('link-994', 'toseq', 'link-978');">toseq</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-995" class="py-name"><a title="Bio.Seq.Seq.back_transcribe
Bio.Seq.UnknownSeq.back_transcribe
Bio.Seq.back_transcribe" class="py-name" href="#" onclick="return doclink('link-995', 'back_transcribe', 'link-592');">back_transcribe</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L2636"></a><tt class="py-lineno">2636</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L2637"></a><tt class="py-lineno">2637</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">rna</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"U"</tt><tt class="py-op">,</tt> <tt class="py-string">"T"</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"u"</tt><tt class="py-op">,</tt> <tt class="py-string">"t"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L2638"></a><tt class="py-lineno">2638</tt>  <tt class="py-line"> </tt>
<a name="L2639"></a><tt class="py-lineno">2639</tt>  <tt class="py-line"> </tt>
<a name="_translate_str"></a><div id="_translate_str-def"><a name="L2640"></a><tt class="py-lineno">2640</tt> <a class="py-toggle" href="#" id="_translate_str-toggle" onclick="return toggle('_translate_str');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq-module.html#_translate_str">_translate_str</a><tt class="py-op">(</tt><tt class="py-param">sequence</tt><tt class="py-op">,</tt> <tt class="py-param">table</tt><tt class="py-op">,</tt> <tt class="py-param">stop_symbol</tt><tt class="py-op">=</tt><tt class="py-string">"*"</tt><tt class="py-op">,</tt> <tt class="py-param">to_stop</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> </tt>
<a name="L2641"></a><tt class="py-lineno">2641</tt>  <tt class="py-line">                   <tt class="py-param">cds</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> <tt class="py-param">pos_stop</tt><tt class="py-op">=</tt><tt class="py-string">"X"</tt><tt class="py-op">,</tt> <tt class="py-param">gap</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_translate_str-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_translate_str-expanded"><a name="L2642"></a><tt class="py-lineno">2642</tt>  <tt class="py-line">    <tt class="py-docstring">"""Translate nucleotide string into a protein string (PRIVATE).</tt> </tt>
<a name="L2643"></a><tt class="py-lineno">2643</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2644"></a><tt class="py-lineno">2644</tt>  <tt class="py-line"><tt class="py-docstring">    Arguments:</tt> </tt>
<a name="L2645"></a><tt class="py-lineno">2645</tt>  <tt class="py-line"><tt class="py-docstring">     - sequence - a string</tt> </tt>
<a name="L2646"></a><tt class="py-lineno">2646</tt>  <tt class="py-line"><tt class="py-docstring">     - table - a CodonTable object (NOT a table name or id number)</tt> </tt>
<a name="L2647"></a><tt class="py-lineno">2647</tt>  <tt class="py-line"><tt class="py-docstring">     - stop_symbol - a single character string, what to use for terminators.</tt> </tt>
<a name="L2648"></a><tt class="py-lineno">2648</tt>  <tt class="py-line"><tt class="py-docstring">     - to_stop - boolean, should translation terminate at the first</tt> </tt>
<a name="L2649"></a><tt class="py-lineno">2649</tt>  <tt class="py-line"><tt class="py-docstring">       in frame stop codon?  If there is no in-frame stop codon</tt> </tt>
<a name="L2650"></a><tt class="py-lineno">2650</tt>  <tt class="py-line"><tt class="py-docstring">       then translation continues to the end.</tt> </tt>
<a name="L2651"></a><tt class="py-lineno">2651</tt>  <tt class="py-line"><tt class="py-docstring">     - pos_stop - a single character string for a possible stop codon</tt> </tt>
<a name="L2652"></a><tt class="py-lineno">2652</tt>  <tt class="py-line"><tt class="py-docstring">       (e.g. TAN or NNN)</tt> </tt>
<a name="L2653"></a><tt class="py-lineno">2653</tt>  <tt class="py-line"><tt class="py-docstring">     - cds - Boolean, indicates this is a complete CDS.  If True, this</tt> </tt>
<a name="L2654"></a><tt class="py-lineno">2654</tt>  <tt class="py-line"><tt class="py-docstring">       checks the sequence starts with a valid alternative start</tt> </tt>
<a name="L2655"></a><tt class="py-lineno">2655</tt>  <tt class="py-line"><tt class="py-docstring">       codon (which will be translated as methionine, M), that the</tt> </tt>
<a name="L2656"></a><tt class="py-lineno">2656</tt>  <tt class="py-line"><tt class="py-docstring">       sequence length is a multiple of three, and that there is a</tt> </tt>
<a name="L2657"></a><tt class="py-lineno">2657</tt>  <tt class="py-line"><tt class="py-docstring">       single in frame stop codon at the end (this will be excluded</tt> </tt>
<a name="L2658"></a><tt class="py-lineno">2658</tt>  <tt class="py-line"><tt class="py-docstring">       from the protein sequence, regardless of the to_stop option).</tt> </tt>
<a name="L2659"></a><tt class="py-lineno">2659</tt>  <tt class="py-line"><tt class="py-docstring">       If these tests fail, an exception is raised.</tt> </tt>
<a name="L2660"></a><tt class="py-lineno">2660</tt>  <tt class="py-line"><tt class="py-docstring">     - gap - Single character string to denote symbol used for gaps.</tt> </tt>
<a name="L2661"></a><tt class="py-lineno">2661</tt>  <tt class="py-line"><tt class="py-docstring">       Defaults to None.</tt> </tt>
<a name="L2662"></a><tt class="py-lineno">2662</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2663"></a><tt class="py-lineno">2663</tt>  <tt class="py-line"><tt class="py-docstring">    Returns a string.</tt> </tt>
<a name="L2664"></a><tt class="py-lineno">2664</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2665"></a><tt class="py-lineno">2665</tt>  <tt class="py-line"><tt class="py-docstring">    e.g.</tt> </tt>
<a name="L2666"></a><tt class="py-lineno">2666</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2667"></a><tt class="py-lineno">2667</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio.Data import CodonTable</tt> </tt>
<a name="L2668"></a><tt class="py-lineno">2668</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; table = CodonTable.ambiguous_dna_by_id[1]</tt> </tt>
<a name="L2669"></a><tt class="py-lineno">2669</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _translate_str("AAA", table)</tt> </tt>
<a name="L2670"></a><tt class="py-lineno">2670</tt>  <tt class="py-line"><tt class="py-docstring">    'K'</tt> </tt>
<a name="L2671"></a><tt class="py-lineno">2671</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _translate_str("TAR", table)</tt> </tt>
<a name="L2672"></a><tt class="py-lineno">2672</tt>  <tt class="py-line"><tt class="py-docstring">    '*'</tt> </tt>
<a name="L2673"></a><tt class="py-lineno">2673</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _translate_str("TAN", table)</tt> </tt>
<a name="L2674"></a><tt class="py-lineno">2674</tt>  <tt class="py-line"><tt class="py-docstring">    'X'</tt> </tt>
<a name="L2675"></a><tt class="py-lineno">2675</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _translate_str("TAN", table, pos_stop="@")</tt> </tt>
<a name="L2676"></a><tt class="py-lineno">2676</tt>  <tt class="py-line"><tt class="py-docstring">    '@'</tt> </tt>
<a name="L2677"></a><tt class="py-lineno">2677</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _translate_str("TA?", table)</tt> </tt>
<a name="L2678"></a><tt class="py-lineno">2678</tt>  <tt class="py-line"><tt class="py-docstring">    Traceback (most recent call last):</tt> </tt>
<a name="L2679"></a><tt class="py-lineno">2679</tt>  <tt class="py-line"><tt class="py-docstring">       ...</tt> </tt>
<a name="L2680"></a><tt class="py-lineno">2680</tt>  <tt class="py-line"><tt class="py-docstring">    Bio.Data.CodonTable.TranslationError: Codon 'TA?' is invalid</tt> </tt>
<a name="L2681"></a><tt class="py-lineno">2681</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2682"></a><tt class="py-lineno">2682</tt>  <tt class="py-line"><tt class="py-docstring">    In a change to older versions of Biopython, partial codons are now</tt> </tt>
<a name="L2683"></a><tt class="py-lineno">2683</tt>  <tt class="py-line"><tt class="py-docstring">    always regarded as an error (previously only checked if cds=True)</tt> </tt>
<a name="L2684"></a><tt class="py-lineno">2684</tt>  <tt class="py-line"><tt class="py-docstring">    and will trigger a warning (likely to become an exception in a</tt> </tt>
<a name="L2685"></a><tt class="py-lineno">2685</tt>  <tt class="py-line"><tt class="py-docstring">    future release).</tt> </tt>
<a name="L2686"></a><tt class="py-lineno">2686</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2687"></a><tt class="py-lineno">2687</tt>  <tt class="py-line"><tt class="py-docstring">    If **cds=True**, the start and stop codons are checked, and the start</tt> </tt>
<a name="L2688"></a><tt class="py-lineno">2688</tt>  <tt class="py-line"><tt class="py-docstring">    codon will be translated at methionine. The sequence must be an</tt> </tt>
<a name="L2689"></a><tt class="py-lineno">2689</tt>  <tt class="py-line"><tt class="py-docstring">    while number of codons.</tt> </tt>
<a name="L2690"></a><tt class="py-lineno">2690</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2691"></a><tt class="py-lineno">2691</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _translate_str("ATGCCCTAG", table, cds=True)</tt> </tt>
<a name="L2692"></a><tt class="py-lineno">2692</tt>  <tt class="py-line"><tt class="py-docstring">    'MP'</tt> </tt>
<a name="L2693"></a><tt class="py-lineno">2693</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _translate_str("AAACCCTAG", table, cds=True)</tt> </tt>
<a name="L2694"></a><tt class="py-lineno">2694</tt>  <tt class="py-line"><tt class="py-docstring">    Traceback (most recent call last):</tt> </tt>
<a name="L2695"></a><tt class="py-lineno">2695</tt>  <tt class="py-line"><tt class="py-docstring">       ...</tt> </tt>
<a name="L2696"></a><tt class="py-lineno">2696</tt>  <tt class="py-line"><tt class="py-docstring">    Bio.Data.CodonTable.TranslationError: First codon 'AAA' is not a start codon</tt> </tt>
<a name="L2697"></a><tt class="py-lineno">2697</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _translate_str("ATGCCCTAGCCCTAG", table, cds=True)</tt> </tt>
<a name="L2698"></a><tt class="py-lineno">2698</tt>  <tt class="py-line"><tt class="py-docstring">    Traceback (most recent call last):</tt> </tt>
<a name="L2699"></a><tt class="py-lineno">2699</tt>  <tt class="py-line"><tt class="py-docstring">       ...</tt> </tt>
<a name="L2700"></a><tt class="py-lineno">2700</tt>  <tt class="py-line"><tt class="py-docstring">    Bio.Data.CodonTable.TranslationError: Extra in frame stop codon found.</tt> </tt>
<a name="L2701"></a><tt class="py-lineno">2701</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L2702"></a><tt class="py-lineno">2702</tt>  <tt class="py-line">    <tt id="link-996" class="py-name" targets="Method Bio.FSSP.FSSPAlignDict.sequence()=Bio.FSSP.FSSPAlignDict-class.html#sequence,Method Bio.GenBank._FeatureConsumer.sequence()=Bio.GenBank._FeatureConsumer-class.html#sequence,Method Bio.GenBank._RecordConsumer.sequence()=Bio.GenBank._RecordConsumer-class.html#sequence,Method Bio.Phylo.PhyloXMLIO.Writer.sequence()=Bio.Phylo.PhyloXMLIO.Writer-class.html#sequence"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-996', 'sequence', 'link-996');">sequence</a></tt> <tt class="py-op">=</tt> <tt id="link-997" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-997', 'sequence', 'link-996');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-998" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-998', 'upper', 'link-289');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L2703"></a><tt class="py-lineno">2703</tt>  <tt class="py-line">    <tt class="py-name">amino_acids</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L2704"></a><tt class="py-lineno">2704</tt>  <tt class="py-line">    <tt id="link-999" class="py-name" targets="Variable Bio.Data.CodonTable.CodonTable.forward_table=Bio.Data.CodonTable.CodonTable-class.html#forward_table"><a title="Bio.Data.CodonTable.CodonTable.forward_table" class="py-name" href="#" onclick="return doclink('link-999', 'forward_table', 'link-999');">forward_table</a></tt> <tt class="py-op">=</tt> <tt class="py-name">table</tt><tt class="py-op">.</tt><tt id="link-1000" class="py-name"><a title="Bio.Data.CodonTable.CodonTable.forward_table" class="py-name" href="#" onclick="return doclink('link-1000', 'forward_table', 'link-999');">forward_table</a></tt> </tt>
<a name="L2705"></a><tt class="py-lineno">2705</tt>  <tt class="py-line">    <tt id="link-1001" class="py-name" targets="Variable Bio.Data.CodonTable.CodonTable.stop_codons=Bio.Data.CodonTable.CodonTable-class.html#stop_codons"><a title="Bio.Data.CodonTable.CodonTable.stop_codons" class="py-name" href="#" onclick="return doclink('link-1001', 'stop_codons', 'link-1001');">stop_codons</a></tt> <tt class="py-op">=</tt> <tt class="py-name">table</tt><tt class="py-op">.</tt><tt id="link-1002" class="py-name"><a title="Bio.Data.CodonTable.CodonTable.stop_codons" class="py-name" href="#" onclick="return doclink('link-1002', 'stop_codons', 'link-1001');">stop_codons</a></tt> </tt>
<a name="L2706"></a><tt class="py-lineno">2706</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">table</tt><tt class="py-op">.</tt><tt id="link-1003" class="py-name" targets="Variable Bio.Data.CodonTable.CodonTable.nucleotide_alphabet=Bio.Data.CodonTable.CodonTable-class.html#nucleotide_alphabet,Variable Bio.Data.CodonTable.NCBICodonTable.nucleotide_alphabet=Bio.Data.CodonTable.NCBICodonTable-class.html#nucleotide_alphabet,Variable Bio.Data.CodonTable.NCBICodonTableDNA.nucleotide_alphabet=Bio.Data.CodonTable.NCBICodonTableDNA-class.html#nucleotide_alphabet,Variable Bio.Data.CodonTable.NCBICodonTableRNA.nucleotide_alphabet=Bio.Data.CodonTable.NCBICodonTableRNA-class.html#nucleotide_alphabet"><a title="Bio.Data.CodonTable.CodonTable.nucleotide_alphabet
Bio.Data.CodonTable.NCBICodonTable.nucleotide_alphabet
Bio.Data.CodonTable.NCBICodonTableDNA.nucleotide_alphabet
Bio.Data.CodonTable.NCBICodonTableRNA.nucleotide_alphabet" class="py-name" href="#" onclick="return doclink('link-1003', 'nucleotide_alphabet', 'link-1003');">nucleotide_alphabet</a></tt><tt class="py-op">.</tt><tt id="link-1004" class="py-name" targets="Variable Bio.Alphabet.Alphabet.letters=Bio.Alphabet.Alphabet-class.html#letters,Variable Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters=Bio.Alphabet.IUPAC.ExtendedIUPACDNA-class.html#letters,Variable Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters=Bio.Alphabet.IUPAC.ExtendedIUPACProtein-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters=Bio.Alphabet.IUPAC.IUPACAmbiguousDNA-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters=Bio.Alphabet.IUPAC.IUPACAmbiguousRNA-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACProtein.letters=Bio.Alphabet.IUPAC.IUPACProtein-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters=Bio.Alphabet.IUPAC.IUPACUnambiguousDNA-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters=Bio.Alphabet.IUPAC.IUPACUnambiguousRNA-class.html#letters,Variable Bio.Alphabet.Reduced.HPModel.letters=Bio.Alphabet.Reduced.HPModel-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy10.letters=Bio.Alphabet.Reduced.Murphy10-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy15.letters=Bio.Alphabet.Reduced.Murphy15-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy4.letters=Bio.Alphabet.Reduced.Murphy4-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy8.letters=Bio.Alphabet.Reduced.Murphy8-class.html#letters,Variable Bio.Alphabet.Reduced.PC5.letters=Bio.Alphabet.Reduced.PC5-class.html#letters,Variable Bio.Alphabet.SecondaryStructure.letters=Bio.Alphabet.SecondaryStructure-class.html#letters,Variable Bio.Alphabet.SingleLetterAlphabet.letters=Bio.Alphabet.SingleLetterAlphabet-class.html#letters,Variable Bio.Alphabet.ThreeLetterProtein.letters=Bio.Alphabet.ThreeLetterProtein-class.html#letters,Variable Bio.codonalign.codonalphabet.CodonAlphabet.letters=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#letters"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-1004', 'letters', 'link-1004');">letters</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L2707"></a><tt class="py-lineno">2707</tt>  <tt class="py-line">        <tt class="py-name">valid_letters</tt> <tt class="py-op">=</tt> <tt id="link-1005" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-1005', 'set', 'link-526');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">table</tt><tt class="py-op">.</tt><tt id="link-1006" class="py-name"><a title="Bio.Data.CodonTable.CodonTable.nucleotide_alphabet
Bio.Data.CodonTable.NCBICodonTable.nucleotide_alphabet
Bio.Data.CodonTable.NCBICodonTableDNA.nucleotide_alphabet
Bio.Data.CodonTable.NCBICodonTableRNA.nucleotide_alphabet" class="py-name" href="#" onclick="return doclink('link-1006', 'nucleotide_alphabet', 'link-1003');">nucleotide_alphabet</a></tt><tt class="py-op">.</tt><tt id="link-1007" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-1007', 'letters', 'link-1004');">letters</a></tt><tt class="py-op">.</tt><tt id="link-1008" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-1008', 'upper', 'link-289');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L2708"></a><tt class="py-lineno">2708</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L2709"></a><tt class="py-lineno">2709</tt>  <tt class="py-line">        <tt class="py-comment"># Assume the worst case, ambiguous DNA or RNA:</tt> </tt>
<a name="L2710"></a><tt class="py-lineno">2710</tt>  <tt class="py-line">        <tt class="py-name">valid_letters</tt> <tt class="py-op">=</tt> <tt id="link-1009" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-1009', 'set', 'link-526');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">_ambiguous_dna_letters</tt><tt class="py-op">.</tt><tt id="link-1010" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-1010', 'upper', 'link-289');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> </tt>
<a name="L2711"></a><tt class="py-lineno">2711</tt>  <tt class="py-line">                            <tt class="py-name">_ambiguous_rna_letters</tt><tt class="py-op">.</tt><tt id="link-1011" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-1011', 'upper', 'link-289');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L2712"></a><tt class="py-lineno">2712</tt>  <tt class="py-line">    <tt class="py-name">n</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-1012" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1012', 'sequence', 'link-996');">sequence</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2713"></a><tt class="py-lineno">2713</tt>  <tt class="py-line"> </tt>
<a name="L2714"></a><tt class="py-lineno">2714</tt>  <tt class="py-line">    <tt class="py-comment"># Check for tables with 'ambiguous' (dual-coding) stop codons:</tt> </tt>
<a name="L2715"></a><tt class="py-lineno">2715</tt>  <tt class="py-line">    <tt class="py-name">dual_coding</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">c</tt> <tt class="py-keyword">for</tt> <tt class="py-name">c</tt> <tt class="py-keyword">in</tt> <tt id="link-1013" class="py-name"><a title="Bio.Data.CodonTable.CodonTable.stop_codons" class="py-name" href="#" onclick="return doclink('link-1013', 'stop_codons', 'link-1001');">stop_codons</a></tt> <tt class="py-keyword">if</tt> <tt class="py-name">c</tt> <tt class="py-keyword">in</tt> <tt id="link-1014" class="py-name"><a title="Bio.Data.CodonTable.CodonTable.forward_table" class="py-name" href="#" onclick="return doclink('link-1014', 'forward_table', 'link-999');">forward_table</a></tt><tt class="py-op">]</tt> </tt>
<a name="L2716"></a><tt class="py-lineno">2716</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">dual_coding</tt><tt class="py-op">:</tt> </tt>
<a name="L2717"></a><tt class="py-lineno">2717</tt>  <tt class="py-line">        <tt class="py-name">c</tt> <tt class="py-op">=</tt> <tt class="py-name">dual_coding</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L2718"></a><tt class="py-lineno">2718</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">to_stop</tt><tt class="py-op">:</tt> </tt>
<a name="L2719"></a><tt class="py-lineno">2719</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"You cannot use 'to_stop=True' with this table "</tt> </tt>
<a name="L2720"></a><tt class="py-lineno">2720</tt>  <tt class="py-line">                             <tt class="py-string">"as it contains {} codon(s) which can be both "</tt> </tt>
<a name="L2721"></a><tt class="py-lineno">2721</tt>  <tt class="py-line">                             <tt class="py-string">" STOP and an  amino acid (e.g. '{}' -&gt; '{}' or "</tt> </tt>
<a name="L2722"></a><tt class="py-lineno">2722</tt>  <tt class="py-line">                             <tt class="py-string">"STOP)."</tt> </tt>
<a name="L2723"></a><tt class="py-lineno">2723</tt>  <tt class="py-line">                             <tt class="py-op">.</tt><tt id="link-1015" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-1015', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">dual_coding</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">c</tt><tt class="py-op">,</tt> <tt id="link-1016" class="py-name"><a title="Bio.Data.CodonTable.CodonTable.forward_table" class="py-name" href="#" onclick="return doclink('link-1016', 'forward_table', 'link-999');">forward_table</a></tt><tt class="py-op">[</tt><tt class="py-name">c</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L2724"></a><tt class="py-lineno">2724</tt>  <tt class="py-line">        <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"This table contains {} codon(s) which code(s) for both "</tt> </tt>
<a name="L2725"></a><tt class="py-lineno">2725</tt>  <tt class="py-line">                      <tt class="py-string">"STOP and an amino acid (e.g. '{}' -&gt; '{}' or STOP). "</tt> </tt>
<a name="L2726"></a><tt class="py-lineno">2726</tt>  <tt class="py-line">                      <tt class="py-string">"Such codons will be translated as amino acid."</tt> </tt>
<a name="L2727"></a><tt class="py-lineno">2727</tt>  <tt class="py-line">                      <tt class="py-op">.</tt><tt id="link-1017" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-1017', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">dual_coding</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">c</tt><tt class="py-op">,</tt> <tt id="link-1018" class="py-name"><a title="Bio.Data.CodonTable.CodonTable.forward_table" class="py-name" href="#" onclick="return doclink('link-1018', 'forward_table', 'link-999');">forward_table</a></tt><tt class="py-op">[</tt><tt class="py-name">c</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L2728"></a><tt class="py-lineno">2728</tt>  <tt class="py-line">                      <tt id="link-1019" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-1019', 'BiopythonWarning', 'link-6');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2729"></a><tt class="py-lineno">2729</tt>  <tt class="py-line"> </tt>
<a name="L2730"></a><tt class="py-lineno">2730</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">cds</tt><tt class="py-op">:</tt> </tt>
<a name="L2731"></a><tt class="py-lineno">2731</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-1020" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1020', 'sequence', 'link-996');">sequence</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-1021" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-1021', 'upper', 'link-289');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">table</tt><tt class="py-op">.</tt><tt id="link-1022" class="py-name" targets="Variable Bio.Data.CodonTable.CodonTable.start_codons=Bio.Data.CodonTable.CodonTable-class.html#start_codons"><a title="Bio.Data.CodonTable.CodonTable.start_codons" class="py-name" href="#" onclick="return doclink('link-1022', 'start_codons', 'link-1022');">start_codons</a></tt><tt class="py-op">:</tt> </tt>
<a name="L2732"></a><tt class="py-lineno">2732</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-1023" class="py-name"><a title="Bio.Data.CodonTable
Bio.Data.CodonTable.CodonTable" class="py-name" href="#" onclick="return doclink('link-1023', 'CodonTable', 'link-27');">CodonTable</a></tt><tt class="py-op">.</tt><tt id="link-1024" class="py-name" targets="Class Bio.Data.CodonTable.TranslationError=Bio.Data.CodonTable.TranslationError-class.html"><a title="Bio.Data.CodonTable.TranslationError" class="py-name" href="#" onclick="return doclink('link-1024', 'TranslationError', 'link-1024');">TranslationError</a></tt><tt class="py-op">(</tt> </tt>
<a name="L2733"></a><tt class="py-lineno">2733</tt>  <tt class="py-line">                <tt class="py-string">"First codon '{0}' is not a start codon"</tt><tt class="py-op">.</tt><tt id="link-1025" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-1025', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt id="link-1026" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1026', 'sequence', 'link-996');">sequence</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L2734"></a><tt class="py-lineno">2734</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">n</tt> <tt class="py-op">%</tt> <tt class="py-number">3</tt> <tt class="py-op">!=</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L2735"></a><tt class="py-lineno">2735</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-1027" class="py-name"><a title="Bio.Data.CodonTable
Bio.Data.CodonTable.CodonTable" class="py-name" href="#" onclick="return doclink('link-1027', 'CodonTable', 'link-27');">CodonTable</a></tt><tt class="py-op">.</tt><tt id="link-1028" class="py-name"><a title="Bio.Data.CodonTable.TranslationError" class="py-name" href="#" onclick="return doclink('link-1028', 'TranslationError', 'link-1024');">TranslationError</a></tt><tt class="py-op">(</tt> </tt>
<a name="L2736"></a><tt class="py-lineno">2736</tt>  <tt class="py-line">                <tt class="py-string">"Sequence length {0} is not a multiple of three"</tt><tt class="py-op">.</tt><tt id="link-1029" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-1029', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L2737"></a><tt class="py-lineno">2737</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-1030" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1030', 'sequence', 'link-996');">sequence</a></tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">3</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-1031" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-1031', 'upper', 'link-289');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt id="link-1032" class="py-name"><a title="Bio.Data.CodonTable.CodonTable.stop_codons" class="py-name" href="#" onclick="return doclink('link-1032', 'stop_codons', 'link-1001');">stop_codons</a></tt><tt class="py-op">:</tt> </tt>
<a name="L2738"></a><tt class="py-lineno">2738</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-1033" class="py-name"><a title="Bio.Data.CodonTable
Bio.Data.CodonTable.CodonTable" class="py-name" href="#" onclick="return doclink('link-1033', 'CodonTable', 'link-27');">CodonTable</a></tt><tt class="py-op">.</tt><tt id="link-1034" class="py-name"><a title="Bio.Data.CodonTable.TranslationError" class="py-name" href="#" onclick="return doclink('link-1034', 'TranslationError', 'link-1024');">TranslationError</a></tt><tt class="py-op">(</tt> </tt>
<a name="L2739"></a><tt class="py-lineno">2739</tt>  <tt class="py-line">                <tt class="py-string">"Final codon '{0}' is not a stop codon"</tt><tt class="py-op">.</tt><tt id="link-1035" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-1035', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt id="link-1036" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1036', 'sequence', 'link-996');">sequence</a></tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">3</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L2740"></a><tt class="py-lineno">2740</tt>  <tt class="py-line">        <tt class="py-comment"># Don't translate the stop symbol, and manually translate the M</tt> </tt>
<a name="L2741"></a><tt class="py-lineno">2741</tt>  <tt class="py-line">        <tt id="link-1037" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1037', 'sequence', 'link-996');">sequence</a></tt> <tt class="py-op">=</tt> <tt id="link-1038" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1038', 'sequence', 'link-996');">sequence</a></tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">3</tt><tt class="py-op">]</tt> </tt>
<a name="L2742"></a><tt class="py-lineno">2742</tt>  <tt class="py-line">        <tt class="py-name">n</tt> <tt class="py-op">-=</tt> <tt class="py-number">6</tt> </tt>
<a name="L2743"></a><tt class="py-lineno">2743</tt>  <tt class="py-line">        <tt class="py-name">amino_acids</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">"M"</tt><tt class="py-op">]</tt> </tt>
<a name="L2744"></a><tt class="py-lineno">2744</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">n</tt> <tt class="py-op">%</tt> <tt class="py-number">3</tt> <tt class="py-op">!=</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L2745"></a><tt class="py-lineno">2745</tt>  <tt class="py-line">        <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Partial codon, len(sequence) not a multiple of three. "</tt> </tt>
<a name="L2746"></a><tt class="py-lineno">2746</tt>  <tt class="py-line">                      <tt class="py-string">"Explicitly trim the sequence or add trailing N before "</tt> </tt>
<a name="L2747"></a><tt class="py-lineno">2747</tt>  <tt class="py-line">                      <tt class="py-string">"translation. This may become an error in future."</tt><tt class="py-op">,</tt> </tt>
<a name="L2748"></a><tt class="py-lineno">2748</tt>  <tt class="py-line">                      <tt id="link-1039" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-1039', 'BiopythonWarning', 'link-6');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2749"></a><tt class="py-lineno">2749</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">gap</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L2750"></a><tt class="py-lineno">2750</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">gap</tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2751"></a><tt class="py-lineno">2751</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"Gap character should be a single character "</tt> </tt>
<a name="L2752"></a><tt class="py-lineno">2752</tt>  <tt class="py-line">                            <tt class="py-string">"string."</tt><tt class="py-op">)</tt> </tt>
<a name="L2753"></a><tt class="py-lineno">2753</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">gap</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L2754"></a><tt class="py-lineno">2754</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Gap character should be a single character "</tt> </tt>
<a name="L2755"></a><tt class="py-lineno">2755</tt>  <tt class="py-line">                             <tt class="py-string">"string."</tt><tt class="py-op">)</tt> </tt>
<a name="L2756"></a><tt class="py-lineno">2756</tt>  <tt class="py-line"> </tt>
<a name="L2757"></a><tt class="py-lineno">2757</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt id="link-1040" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-1040', 'i', 'link-887');">i</a></tt> <tt class="py-keyword">in</tt> <tt id="link-1041" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-1041', 'range', 'link-2');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">n</tt> <tt class="py-op">-</tt> <tt class="py-name">n</tt> <tt class="py-op">%</tt> <tt class="py-number">3</tt><tt class="py-op">,</tt> <tt class="py-number">3</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2758"></a><tt class="py-lineno">2758</tt>  <tt class="py-line">        <tt class="py-name">codon</tt> <tt class="py-op">=</tt> <tt id="link-1042" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1042', 'sequence', 'link-996');">sequence</a></tt><tt class="py-op">[</tt><tt id="link-1043" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-1043', 'i', 'link-887');">i</a></tt><tt class="py-op">:</tt><tt id="link-1044" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-1044', 'i', 'link-887');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-number">3</tt><tt class="py-op">]</tt> </tt>
<a name="L2759"></a><tt class="py-lineno">2759</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L2760"></a><tt class="py-lineno">2760</tt>  <tt class="py-line">            <tt class="py-name">amino_acids</tt><tt class="py-op">.</tt><tt id="link-1045" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-1045', 'append', 'link-884');">append</a></tt><tt class="py-op">(</tt><tt id="link-1046" class="py-name"><a title="Bio.Data.CodonTable.CodonTable.forward_table" class="py-name" href="#" onclick="return doclink('link-1046', 'forward_table', 'link-999');">forward_table</a></tt><tt class="py-op">[</tt><tt class="py-name">codon</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L2761"></a><tt class="py-lineno">2761</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-op">(</tt><tt class="py-name">KeyError</tt><tt class="py-op">,</tt> <tt id="link-1047" class="py-name"><a title="Bio.Data.CodonTable
Bio.Data.CodonTable.CodonTable" class="py-name" href="#" onclick="return doclink('link-1047', 'CodonTable', 'link-27');">CodonTable</a></tt><tt class="py-op">.</tt><tt id="link-1048" class="py-name"><a title="Bio.Data.CodonTable.TranslationError" class="py-name" href="#" onclick="return doclink('link-1048', 'TranslationError', 'link-1024');">TranslationError</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2762"></a><tt class="py-lineno">2762</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">codon</tt> <tt class="py-keyword">in</tt> <tt class="py-name">table</tt><tt class="py-op">.</tt><tt id="link-1049" class="py-name"><a title="Bio.Data.CodonTable.CodonTable.stop_codons" class="py-name" href="#" onclick="return doclink('link-1049', 'stop_codons', 'link-1001');">stop_codons</a></tt><tt class="py-op">:</tt> </tt>
<a name="L2763"></a><tt class="py-lineno">2763</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">cds</tt><tt class="py-op">:</tt> </tt>
<a name="L2764"></a><tt class="py-lineno">2764</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt id="link-1050" class="py-name"><a title="Bio.Data.CodonTable
Bio.Data.CodonTable.CodonTable" class="py-name" href="#" onclick="return doclink('link-1050', 'CodonTable', 'link-27');">CodonTable</a></tt><tt class="py-op">.</tt><tt id="link-1051" class="py-name"><a title="Bio.Data.CodonTable.TranslationError" class="py-name" href="#" onclick="return doclink('link-1051', 'TranslationError', 'link-1024');">TranslationError</a></tt><tt class="py-op">(</tt> </tt>
<a name="L2765"></a><tt class="py-lineno">2765</tt>  <tt class="py-line">                        <tt class="py-string">"Extra in frame stop codon found."</tt><tt class="py-op">)</tt> </tt>
<a name="L2766"></a><tt class="py-lineno">2766</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">to_stop</tt><tt class="py-op">:</tt> </tt>
<a name="L2767"></a><tt class="py-lineno">2767</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L2768"></a><tt class="py-lineno">2768</tt>  <tt class="py-line">                <tt class="py-name">amino_acids</tt><tt class="py-op">.</tt><tt id="link-1052" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-1052', 'append', 'link-884');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">stop_symbol</tt><tt class="py-op">)</tt> </tt>
<a name="L2769"></a><tt class="py-lineno">2769</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">valid_letters</tt><tt class="py-op">.</tt><tt class="py-name">issuperset</tt><tt class="py-op">(</tt><tt id="link-1053" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-1053', 'set', 'link-526');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">codon</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2770"></a><tt class="py-lineno">2770</tt>  <tt class="py-line">                <tt class="py-comment"># Possible stop codon (e.g. NNN or TAN)</tt> </tt>
<a name="L2771"></a><tt class="py-lineno">2771</tt>  <tt class="py-line">                <tt class="py-name">amino_acids</tt><tt class="py-op">.</tt><tt id="link-1054" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-1054', 'append', 'link-884');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">pos_stop</tt><tt class="py-op">)</tt> </tt>
<a name="L2772"></a><tt class="py-lineno">2772</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">gap</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> <tt class="py-name">codon</tt> <tt class="py-op">==</tt> <tt class="py-name">gap</tt> <tt class="py-op">*</tt> <tt class="py-number">3</tt><tt class="py-op">:</tt> </tt>
<a name="L2773"></a><tt class="py-lineno">2773</tt>  <tt class="py-line">                <tt class="py-comment"># Gapped translation</tt> </tt>
<a name="L2774"></a><tt class="py-lineno">2774</tt>  <tt class="py-line">                <tt class="py-name">amino_acids</tt><tt class="py-op">.</tt><tt id="link-1055" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-1055', 'append', 'link-884');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">gap</tt><tt class="py-op">)</tt> </tt>
<a name="L2775"></a><tt class="py-lineno">2775</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L2776"></a><tt class="py-lineno">2776</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt id="link-1056" class="py-name"><a title="Bio.Data.CodonTable
Bio.Data.CodonTable.CodonTable" class="py-name" href="#" onclick="return doclink('link-1056', 'CodonTable', 'link-27');">CodonTable</a></tt><tt class="py-op">.</tt><tt id="link-1057" class="py-name"><a title="Bio.Data.CodonTable.TranslationError" class="py-name" href="#" onclick="return doclink('link-1057', 'TranslationError', 'link-1024');">TranslationError</a></tt><tt class="py-op">(</tt> </tt>
<a name="L2777"></a><tt class="py-lineno">2777</tt>  <tt class="py-line">                    <tt class="py-string">"Codon '{0}' is invalid"</tt><tt class="py-op">.</tt><tt id="link-1058" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-1058', 'format', 'link-52');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">codon</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L2778"></a><tt class="py-lineno">2778</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt id="link-1059" class="py-name"><a title="Bio.Seq.MutableSeq.join
Bio.Seq.Seq.join
Bio.Seq.UnknownSeq.join" class="py-name" href="#" onclick="return doclink('link-1059', 'join', 'link-28');">join</a></tt><tt class="py-op">(</tt><tt class="py-name">amino_acids</tt><tt class="py-op">)</tt> </tt>
</div><a name="L2779"></a><tt class="py-lineno">2779</tt>  <tt class="py-line"> </tt>
<a name="L2780"></a><tt class="py-lineno">2780</tt>  <tt class="py-line"> </tt>
<a name="translate"></a><div id="translate-def"><a name="L2781"></a><tt class="py-lineno">2781</tt> <a class="py-toggle" href="#" id="translate-toggle" onclick="return toggle('translate');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq-module.html#translate">translate</a><tt class="py-op">(</tt><tt class="py-param">sequence</tt><tt class="py-op">,</tt> <tt class="py-param">table</tt><tt class="py-op">=</tt><tt class="py-string">"Standard"</tt><tt class="py-op">,</tt> <tt class="py-param">stop_symbol</tt><tt class="py-op">=</tt><tt class="py-string">"*"</tt><tt class="py-op">,</tt> <tt class="py-param">to_stop</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> </tt>
<a name="L2782"></a><tt class="py-lineno">2782</tt>  <tt class="py-line">              <tt class="py-param">cds</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> <tt class="py-param">gap</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="translate-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="translate-expanded"><a name="L2783"></a><tt class="py-lineno">2783</tt>  <tt class="py-line">    <tt class="py-docstring">"""Translate a nucleotide sequence into amino acids.</tt> </tt>
<a name="L2784"></a><tt class="py-lineno">2784</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2785"></a><tt class="py-lineno">2785</tt>  <tt class="py-line"><tt class="py-docstring">    If given a string, returns a new string object. Given a Seq or</tt> </tt>
<a name="L2786"></a><tt class="py-lineno">2786</tt>  <tt class="py-line"><tt class="py-docstring">    MutableSeq, returns a Seq object with a protein alphabet.</tt> </tt>
<a name="L2787"></a><tt class="py-lineno">2787</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2788"></a><tt class="py-lineno">2788</tt>  <tt class="py-line"><tt class="py-docstring">    Arguments:</tt> </tt>
<a name="L2789"></a><tt class="py-lineno">2789</tt>  <tt class="py-line"><tt class="py-docstring">     - table - Which codon table to use?  This can be either a name</tt> </tt>
<a name="L2790"></a><tt class="py-lineno">2790</tt>  <tt class="py-line"><tt class="py-docstring">       (string), an NCBI identifier (integer), or a CodonTable object</tt> </tt>
<a name="L2791"></a><tt class="py-lineno">2791</tt>  <tt class="py-line"><tt class="py-docstring">       (useful for non-standard genetic codes).  Defaults to the "Standard"</tt> </tt>
<a name="L2792"></a><tt class="py-lineno">2792</tt>  <tt class="py-line"><tt class="py-docstring">       table.</tt> </tt>
<a name="L2793"></a><tt class="py-lineno">2793</tt>  <tt class="py-line"><tt class="py-docstring">     - stop_symbol - Single character string, what to use for any</tt> </tt>
<a name="L2794"></a><tt class="py-lineno">2794</tt>  <tt class="py-line"><tt class="py-docstring">       terminators, defaults to the asterisk, "*".</tt> </tt>
<a name="L2795"></a><tt class="py-lineno">2795</tt>  <tt class="py-line"><tt class="py-docstring">     - to_stop - Boolean, defaults to False meaning do a full</tt> </tt>
<a name="L2796"></a><tt class="py-lineno">2796</tt>  <tt class="py-line"><tt class="py-docstring">       translation continuing on past any stop codons</tt> </tt>
<a name="L2797"></a><tt class="py-lineno">2797</tt>  <tt class="py-line"><tt class="py-docstring">       (translated as the specified stop_symbol).  If</tt> </tt>
<a name="L2798"></a><tt class="py-lineno">2798</tt>  <tt class="py-line"><tt class="py-docstring">       True, translation is terminated at the first in</tt> </tt>
<a name="L2799"></a><tt class="py-lineno">2799</tt>  <tt class="py-line"><tt class="py-docstring">       frame stop codon (and the stop_symbol is not</tt> </tt>
<a name="L2800"></a><tt class="py-lineno">2800</tt>  <tt class="py-line"><tt class="py-docstring">       appended to the returned protein sequence).</tt> </tt>
<a name="L2801"></a><tt class="py-lineno">2801</tt>  <tt class="py-line"><tt class="py-docstring">     - cds - Boolean, indicates this is a complete CDS.  If True, this</tt> </tt>
<a name="L2802"></a><tt class="py-lineno">2802</tt>  <tt class="py-line"><tt class="py-docstring">       checks the sequence starts with a valid alternative start</tt> </tt>
<a name="L2803"></a><tt class="py-lineno">2803</tt>  <tt class="py-line"><tt class="py-docstring">       codon (which will be translated as methionine, M), that the</tt> </tt>
<a name="L2804"></a><tt class="py-lineno">2804</tt>  <tt class="py-line"><tt class="py-docstring">       sequence length is a multiple of three, and that there is a</tt> </tt>
<a name="L2805"></a><tt class="py-lineno">2805</tt>  <tt class="py-line"><tt class="py-docstring">       single in frame stop codon at the end (this will be excluded</tt> </tt>
<a name="L2806"></a><tt class="py-lineno">2806</tt>  <tt class="py-line"><tt class="py-docstring">       from the protein sequence, regardless of the to_stop option).</tt> </tt>
<a name="L2807"></a><tt class="py-lineno">2807</tt>  <tt class="py-line"><tt class="py-docstring">       If these tests fail, an exception is raised.</tt> </tt>
<a name="L2808"></a><tt class="py-lineno">2808</tt>  <tt class="py-line"><tt class="py-docstring">     - gap - Single character string to denote symbol used for gaps.</tt> </tt>
<a name="L2809"></a><tt class="py-lineno">2809</tt>  <tt class="py-line"><tt class="py-docstring">       Defaults to None.</tt> </tt>
<a name="L2810"></a><tt class="py-lineno">2810</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2811"></a><tt class="py-lineno">2811</tt>  <tt class="py-line"><tt class="py-docstring">    A simple string example using the default (standard) genetic code:</tt> </tt>
<a name="L2812"></a><tt class="py-lineno">2812</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2813"></a><tt class="py-lineno">2813</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; coding_dna = "GTGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG"</tt> </tt>
<a name="L2814"></a><tt class="py-lineno">2814</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; translate(coding_dna)</tt> </tt>
<a name="L2815"></a><tt class="py-lineno">2815</tt>  <tt class="py-line"><tt class="py-docstring">    'VAIVMGR*KGAR*'</tt> </tt>
<a name="L2816"></a><tt class="py-lineno">2816</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; translate(coding_dna, stop_symbol="@")</tt> </tt>
<a name="L2817"></a><tt class="py-lineno">2817</tt>  <tt class="py-line"><tt class="py-docstring">    'VAIVMGR@KGAR@'</tt> </tt>
<a name="L2818"></a><tt class="py-lineno">2818</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; translate(coding_dna, to_stop=True)</tt> </tt>
<a name="L2819"></a><tt class="py-lineno">2819</tt>  <tt class="py-line"><tt class="py-docstring">    'VAIVMGR'</tt> </tt>
<a name="L2820"></a><tt class="py-lineno">2820</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2821"></a><tt class="py-lineno">2821</tt>  <tt class="py-line"><tt class="py-docstring">    Now using NCBI table 2, where TGA is not a stop codon:</tt> </tt>
<a name="L2822"></a><tt class="py-lineno">2822</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2823"></a><tt class="py-lineno">2823</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; translate(coding_dna, table=2)</tt> </tt>
<a name="L2824"></a><tt class="py-lineno">2824</tt>  <tt class="py-line"><tt class="py-docstring">    'VAIVMGRWKGAR*'</tt> </tt>
<a name="L2825"></a><tt class="py-lineno">2825</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; translate(coding_dna, table=2, to_stop=True)</tt> </tt>
<a name="L2826"></a><tt class="py-lineno">2826</tt>  <tt class="py-line"><tt class="py-docstring">    'VAIVMGRWKGAR'</tt> </tt>
<a name="L2827"></a><tt class="py-lineno">2827</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2828"></a><tt class="py-lineno">2828</tt>  <tt class="py-line"><tt class="py-docstring">    In fact this example uses an alternative start codon valid under NCBI</tt> </tt>
<a name="L2829"></a><tt class="py-lineno">2829</tt>  <tt class="py-line"><tt class="py-docstring">    table 2, GTG, which means this example is a complete valid CDS which</tt> </tt>
<a name="L2830"></a><tt class="py-lineno">2830</tt>  <tt class="py-line"><tt class="py-docstring">    when translated should really start with methionine (not valine):</tt> </tt>
<a name="L2831"></a><tt class="py-lineno">2831</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2832"></a><tt class="py-lineno">2832</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; translate(coding_dna, table=2, cds=True)</tt> </tt>
<a name="L2833"></a><tt class="py-lineno">2833</tt>  <tt class="py-line"><tt class="py-docstring">    'MAIVMGRWKGAR'</tt> </tt>
<a name="L2834"></a><tt class="py-lineno">2834</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2835"></a><tt class="py-lineno">2835</tt>  <tt class="py-line"><tt class="py-docstring">    Note that if the sequence has no in-frame stop codon, then the to_stop</tt> </tt>
<a name="L2836"></a><tt class="py-lineno">2836</tt>  <tt class="py-line"><tt class="py-docstring">    argument has no effect:</tt> </tt>
<a name="L2837"></a><tt class="py-lineno">2837</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2838"></a><tt class="py-lineno">2838</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; coding_dna2 = "GTGGCCATTGTAATGGGCCGC"</tt> </tt>
<a name="L2839"></a><tt class="py-lineno">2839</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; translate(coding_dna2)</tt> </tt>
<a name="L2840"></a><tt class="py-lineno">2840</tt>  <tt class="py-line"><tt class="py-docstring">    'VAIVMGR'</tt> </tt>
<a name="L2841"></a><tt class="py-lineno">2841</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; translate(coding_dna2, to_stop=True)</tt> </tt>
<a name="L2842"></a><tt class="py-lineno">2842</tt>  <tt class="py-line"><tt class="py-docstring">    'VAIVMGR'</tt> </tt>
<a name="L2843"></a><tt class="py-lineno">2843</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2844"></a><tt class="py-lineno">2844</tt>  <tt class="py-line"><tt class="py-docstring">    NOTE - Ambiguous codons like "TAN" or "NNN" could be an amino acid</tt> </tt>
<a name="L2845"></a><tt class="py-lineno">2845</tt>  <tt class="py-line"><tt class="py-docstring">    or a stop codon.  These are translated as "X".  Any invalid codon</tt> </tt>
<a name="L2846"></a><tt class="py-lineno">2846</tt>  <tt class="py-line"><tt class="py-docstring">    (e.g. "TA?" or "T-A") will throw a TranslationError.</tt> </tt>
<a name="L2847"></a><tt class="py-lineno">2847</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2848"></a><tt class="py-lineno">2848</tt>  <tt class="py-line"><tt class="py-docstring">    It will however translate either DNA or RNA.</tt> </tt>
<a name="L2849"></a><tt class="py-lineno">2849</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2850"></a><tt class="py-lineno">2850</tt>  <tt class="py-line"><tt class="py-docstring">    NOTE - Since version 1.71 Biopython contains codon tables with 'ambiguous</tt> </tt>
<a name="L2851"></a><tt class="py-lineno">2851</tt>  <tt class="py-line"><tt class="py-docstring">    stop codons'. These are stop codons with unambiguous sequence but which</tt> </tt>
<a name="L2852"></a><tt class="py-lineno">2852</tt>  <tt class="py-line"><tt class="py-docstring">    have a context dependent coding as STOP or as amino acid. With these tables</tt> </tt>
<a name="L2853"></a><tt class="py-lineno">2853</tt>  <tt class="py-line"><tt class="py-docstring">    'to_stop' must be False (otherwise a ValueError is raised). The dual</tt> </tt>
<a name="L2854"></a><tt class="py-lineno">2854</tt>  <tt class="py-line"><tt class="py-docstring">    coding codons will always be translated as amino acid, except for</tt> </tt>
<a name="L2855"></a><tt class="py-lineno">2855</tt>  <tt class="py-line"><tt class="py-docstring">    'cds=True', where the last codon will be translated as STOP.</tt> </tt>
<a name="L2856"></a><tt class="py-lineno">2856</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2857"></a><tt class="py-lineno">2857</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; coding_dna3 = "ATGGCACGGAAGTGA"</tt> </tt>
<a name="L2858"></a><tt class="py-lineno">2858</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; translate(coding_dna3)</tt> </tt>
<a name="L2859"></a><tt class="py-lineno">2859</tt>  <tt class="py-line"><tt class="py-docstring">    'MARK*'</tt> </tt>
<a name="L2860"></a><tt class="py-lineno">2860</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2861"></a><tt class="py-lineno">2861</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; translate(coding_dna3, table=27)  # Table 27: TGA -&gt; STOP or W</tt> </tt>
<a name="L2862"></a><tt class="py-lineno">2862</tt>  <tt class="py-line"><tt class="py-docstring">    'MARKW'</tt> </tt>
<a name="L2863"></a><tt class="py-lineno">2863</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2864"></a><tt class="py-lineno">2864</tt>  <tt class="py-line"><tt class="py-docstring">    It will however raise a BiopythonWarning (not shown).</tt> </tt>
<a name="L2865"></a><tt class="py-lineno">2865</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2866"></a><tt class="py-lineno">2866</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; translate(coding_dna3, table=27, cds=True)</tt> </tt>
<a name="L2867"></a><tt class="py-lineno">2867</tt>  <tt class="py-line"><tt class="py-docstring">    'MARK'</tt> </tt>
<a name="L2868"></a><tt class="py-lineno">2868</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2869"></a><tt class="py-lineno">2869</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; translate(coding_dna3, table=27, to_stop=True)</tt> </tt>
<a name="L2870"></a><tt class="py-lineno">2870</tt>  <tt class="py-line"><tt class="py-docstring">    Traceback (most recent call last):</tt> </tt>
<a name="L2871"></a><tt class="py-lineno">2871</tt>  <tt class="py-line"><tt class="py-docstring">       ...</tt> </tt>
<a name="L2872"></a><tt class="py-lineno">2872</tt>  <tt class="py-line"><tt class="py-docstring">    ValueError: You cannot use 'to_stop=True' with this table ...</tt> </tt>
<a name="L2873"></a><tt class="py-lineno">2873</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L2874"></a><tt class="py-lineno">2874</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-1060" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1060', 'sequence', 'link-996');">sequence</a></tt><tt class="py-op">,</tt> <tt id="link-1061" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-1061', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2875"></a><tt class="py-lineno">2875</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-1062" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1062', 'sequence', 'link-996');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-1063" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.translate
Bio.Seq.Seq.translate
Bio.Seq.UnknownSeq.translate
Bio.Seq.translate
Bio.SeqFeature.SeqFeature.translate
Bio.SeqRecord.SeqRecord.translate
Bio.codonalign.codonseq.CodonSeq.translate" class="py-name" href="#" onclick="return doclink('link-1063', 'translate', 'link-310');">translate</a></tt><tt class="py-op">(</tt><tt class="py-name">table</tt><tt class="py-op">,</tt> <tt class="py-name">stop_symbol</tt><tt class="py-op">,</tt> <tt class="py-name">to_stop</tt><tt class="py-op">,</tt> <tt class="py-name">cds</tt><tt class="py-op">)</tt> </tt>
<a name="L2876"></a><tt class="py-lineno">2876</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-1064" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1064', 'sequence', 'link-996');">sequence</a></tt><tt class="py-op">,</tt> <tt id="link-1065" class="py-name"><a title="Bio.Seq.MutableSeq" class="py-name" href="#" onclick="return doclink('link-1065', 'MutableSeq', 'link-83');">MutableSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2877"></a><tt class="py-lineno">2877</tt>  <tt class="py-line">        <tt class="py-comment"># Return a Seq object</tt> </tt>
<a name="L2878"></a><tt class="py-lineno">2878</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-1066" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1066', 'sequence', 'link-996');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-1067" class="py-name"><a title="Bio.Seq.MutableSeq.toseq
BioSQL.BioSeq.DBSeq.toseq" class="py-name" href="#" onclick="return doclink('link-1067', 'toseq', 'link-978');">toseq</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-1068" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.translate
Bio.Seq.Seq.translate
Bio.Seq.UnknownSeq.translate
Bio.Seq.translate
Bio.SeqFeature.SeqFeature.translate
Bio.SeqRecord.SeqRecord.translate
Bio.codonalign.codonseq.CodonSeq.translate" class="py-name" href="#" onclick="return doclink('link-1068', 'translate', 'link-310');">translate</a></tt><tt class="py-op">(</tt><tt class="py-name">table</tt><tt class="py-op">,</tt> <tt class="py-name">stop_symbol</tt><tt class="py-op">,</tt> <tt class="py-name">to_stop</tt><tt class="py-op">,</tt> <tt class="py-name">cds</tt><tt class="py-op">)</tt> </tt>
<a name="L2879"></a><tt class="py-lineno">2879</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L2880"></a><tt class="py-lineno">2880</tt>  <tt class="py-line">        <tt class="py-comment"># Assume its a string, return a string</tt> </tt>
<a name="L2881"></a><tt class="py-lineno">2881</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L2882"></a><tt class="py-lineno">2882</tt>  <tt class="py-line">            <tt class="py-name">codon_table</tt> <tt class="py-op">=</tt> <tt id="link-1069" class="py-name"><a title="Bio.Data.CodonTable
Bio.Data.CodonTable.CodonTable" class="py-name" href="#" onclick="return doclink('link-1069', 'CodonTable', 'link-27');">CodonTable</a></tt><tt class="py-op">.</tt><tt id="link-1070" class="py-name"><a title="Bio.Data.CodonTable.ambiguous_generic_by_id" class="py-name" href="#" onclick="return doclink('link-1070', 'ambiguous_generic_by_id', 'link-391');">ambiguous_generic_by_id</a></tt><tt class="py-op">[</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">table</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L2883"></a><tt class="py-lineno">2883</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L2884"></a><tt class="py-lineno">2884</tt>  <tt class="py-line">            <tt class="py-name">codon_table</tt> <tt class="py-op">=</tt> <tt id="link-1071" class="py-name"><a title="Bio.Data.CodonTable
Bio.Data.CodonTable.CodonTable" class="py-name" href="#" onclick="return doclink('link-1071', 'CodonTable', 'link-27');">CodonTable</a></tt><tt class="py-op">.</tt><tt id="link-1072" class="py-name"><a title="Bio.Data.CodonTable.ambiguous_generic_by_name" class="py-name" href="#" onclick="return doclink('link-1072', 'ambiguous_generic_by_name', 'link-377');">ambiguous_generic_by_name</a></tt><tt class="py-op">[</tt><tt class="py-name">table</tt><tt class="py-op">]</tt> </tt>
<a name="L2885"></a><tt class="py-lineno">2885</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-op">(</tt><tt class="py-name">AttributeError</tt><tt class="py-op">,</tt> <tt class="py-name">TypeError</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2886"></a><tt class="py-lineno">2886</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">table</tt><tt class="py-op">,</tt> <tt id="link-1073" class="py-name"><a title="Bio.Data.CodonTable
Bio.Data.CodonTable.CodonTable" class="py-name" href="#" onclick="return doclink('link-1073', 'CodonTable', 'link-27');">CodonTable</a></tt><tt class="py-op">.</tt><tt id="link-1074" class="py-name"><a title="Bio.Data.CodonTable
Bio.Data.CodonTable.CodonTable" class="py-name" href="#" onclick="return doclink('link-1074', 'CodonTable', 'link-27');">CodonTable</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2887"></a><tt class="py-lineno">2887</tt>  <tt class="py-line">                <tt class="py-name">codon_table</tt> <tt class="py-op">=</tt> <tt class="py-name">table</tt> </tt>
<a name="L2888"></a><tt class="py-lineno">2888</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L2889"></a><tt class="py-lineno">2889</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Bad table argument"</tt><tt class="py-op">)</tt> </tt>
<a name="L2890"></a><tt class="py-lineno">2890</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-1075" class="py-name"><a title="Bio.Seq._translate_str" class="py-name" href="#" onclick="return doclink('link-1075', '_translate_str', 'link-398');">_translate_str</a></tt><tt class="py-op">(</tt><tt id="link-1076" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1076', 'sequence', 'link-996');">sequence</a></tt><tt class="py-op">,</tt> <tt class="py-name">codon_table</tt><tt class="py-op">,</tt> <tt class="py-name">stop_symbol</tt><tt class="py-op">,</tt> <tt class="py-name">to_stop</tt><tt class="py-op">,</tt> <tt class="py-name">cds</tt><tt class="py-op">,</tt> </tt>
<a name="L2891"></a><tt class="py-lineno">2891</tt>  <tt class="py-line">                              <tt class="py-name">gap</tt><tt class="py-op">=</tt><tt class="py-name">gap</tt><tt class="py-op">)</tt> </tt>
</div><a name="L2892"></a><tt class="py-lineno">2892</tt>  <tt class="py-line"> </tt>
<a name="L2893"></a><tt class="py-lineno">2893</tt>  <tt class="py-line"> </tt>
<a name="reverse_complement"></a><div id="reverse_complement-def"><a name="L2894"></a><tt class="py-lineno">2894</tt> <a class="py-toggle" href="#" id="reverse_complement-toggle" onclick="return toggle('reverse_complement');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq-module.html#reverse_complement">reverse_complement</a><tt class="py-op">(</tt><tt class="py-param">sequence</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="reverse_complement-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="reverse_complement-expanded"><a name="L2895"></a><tt class="py-lineno">2895</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return the reverse complement sequence of a nucleotide string.</tt> </tt>
<a name="L2896"></a><tt class="py-lineno">2896</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2897"></a><tt class="py-lineno">2897</tt>  <tt class="py-line"><tt class="py-docstring">    If given a string, returns a new string object.</tt> </tt>
<a name="L2898"></a><tt class="py-lineno">2898</tt>  <tt class="py-line"><tt class="py-docstring">    Given a Seq or a MutableSeq, returns a new Seq object with the same</tt> </tt>
<a name="L2899"></a><tt class="py-lineno">2899</tt>  <tt class="py-line"><tt class="py-docstring">    alphabet.</tt> </tt>
<a name="L2900"></a><tt class="py-lineno">2900</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2901"></a><tt class="py-lineno">2901</tt>  <tt class="py-line"><tt class="py-docstring">    Supports unambiguous and ambiguous nucleotide sequences.</tt> </tt>
<a name="L2902"></a><tt class="py-lineno">2902</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2903"></a><tt class="py-lineno">2903</tt>  <tt class="py-line"><tt class="py-docstring">    e.g.</tt> </tt>
<a name="L2904"></a><tt class="py-lineno">2904</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2905"></a><tt class="py-lineno">2905</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; reverse_complement("ACTG-NH")</tt> </tt>
<a name="L2906"></a><tt class="py-lineno">2906</tt>  <tt class="py-line"><tt class="py-docstring">    'DN-CAGT'</tt> </tt>
<a name="L2907"></a><tt class="py-lineno">2907</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L2908"></a><tt class="py-lineno">2908</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-1077" class="py-name"><a title="Bio.Seq.MutableSeq.complement
Bio.Seq.Seq.complement
Bio.Seq.UnknownSeq.complement
Bio.Seq.complement" class="py-name" href="#" onclick="return doclink('link-1077', 'complement', 'link-312');">complement</a></tt><tt class="py-op">(</tt><tt id="link-1078" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1078', 'sequence', 'link-996');">sequence</a></tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
</div><a name="L2909"></a><tt class="py-lineno">2909</tt>  <tt class="py-line"> </tt>
<a name="L2910"></a><tt class="py-lineno">2910</tt>  <tt class="py-line"> </tt>
<a name="complement"></a><div id="complement-def"><a name="L2911"></a><tt class="py-lineno">2911</tt> <a class="py-toggle" href="#" id="complement-toggle" onclick="return toggle('complement');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq-module.html#complement">complement</a><tt class="py-op">(</tt><tt class="py-param">sequence</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="complement-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="complement-expanded"><a name="L2912"></a><tt class="py-lineno">2912</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return the complement sequence of a nucleotide string.</tt> </tt>
<a name="L2913"></a><tt class="py-lineno">2913</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2914"></a><tt class="py-lineno">2914</tt>  <tt class="py-line"><tt class="py-docstring">    If given a string, returns a new string object.</tt> </tt>
<a name="L2915"></a><tt class="py-lineno">2915</tt>  <tt class="py-line"><tt class="py-docstring">    Given a Seq or a MutableSeq, returns a new Seq object with the same</tt> </tt>
<a name="L2916"></a><tt class="py-lineno">2916</tt>  <tt class="py-line"><tt class="py-docstring">    alphabet.</tt> </tt>
<a name="L2917"></a><tt class="py-lineno">2917</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2918"></a><tt class="py-lineno">2918</tt>  <tt class="py-line"><tt class="py-docstring">    Supports unambiguous and ambiguous nucleotide sequences.</tt> </tt>
<a name="L2919"></a><tt class="py-lineno">2919</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2920"></a><tt class="py-lineno">2920</tt>  <tt class="py-line"><tt class="py-docstring">    e.g.</tt> </tt>
<a name="L2921"></a><tt class="py-lineno">2921</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2922"></a><tt class="py-lineno">2922</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; complement("ACTG-NH")</tt> </tt>
<a name="L2923"></a><tt class="py-lineno">2923</tt>  <tt class="py-line"><tt class="py-docstring">    'TGAC-ND'</tt> </tt>
<a name="L2924"></a><tt class="py-lineno">2924</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L2925"></a><tt class="py-lineno">2925</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-1079" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1079', 'sequence', 'link-996');">sequence</a></tt><tt class="py-op">,</tt> <tt id="link-1080" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-1080', 'Seq', 'link-82');">Seq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2926"></a><tt class="py-lineno">2926</tt>  <tt class="py-line">        <tt class="py-comment"># Return a Seq</tt> </tt>
<a name="L2927"></a><tt class="py-lineno">2927</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-1081" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1081', 'sequence', 'link-996');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-1082" class="py-name"><a title="Bio.Seq.MutableSeq.complement
Bio.Seq.Seq.complement
Bio.Seq.UnknownSeq.complement
Bio.Seq.complement" class="py-name" href="#" onclick="return doclink('link-1082', 'complement', 'link-312');">complement</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L2928"></a><tt class="py-lineno">2928</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-1083" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1083', 'sequence', 'link-996');">sequence</a></tt><tt class="py-op">,</tt> <tt id="link-1084" class="py-name"><a title="Bio.Seq.MutableSeq" class="py-name" href="#" onclick="return doclink('link-1084', 'MutableSeq', 'link-83');">MutableSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2929"></a><tt class="py-lineno">2929</tt>  <tt class="py-line">        <tt class="py-comment"># Return a Seq</tt> </tt>
<a name="L2930"></a><tt class="py-lineno">2930</tt>  <tt class="py-line">        <tt class="py-comment"># Don't use the MutableSeq reverse_complement method as it is</tt> </tt>
<a name="L2931"></a><tt class="py-lineno">2931</tt>  <tt class="py-line">        <tt class="py-comment"># 'in place'.</tt> </tt>
<a name="L2932"></a><tt class="py-lineno">2932</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-1085" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1085', 'sequence', 'link-996');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-1086" class="py-name"><a title="Bio.Seq.MutableSeq.toseq
BioSQL.BioSeq.DBSeq.toseq" class="py-name" href="#" onclick="return doclink('link-1086', 'toseq', 'link-978');">toseq</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-1087" class="py-name"><a title="Bio.Seq.MutableSeq.complement
Bio.Seq.Seq.complement
Bio.Seq.UnknownSeq.complement
Bio.Seq.complement" class="py-name" href="#" onclick="return doclink('link-1087', 'complement', 'link-312');">complement</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L2933"></a><tt class="py-lineno">2933</tt>  <tt class="py-line"> </tt>
<a name="L2934"></a><tt class="py-lineno">2934</tt>  <tt class="py-line">    <tt class="py-comment"># Assume its a string.</tt> </tt>
<a name="L2935"></a><tt class="py-lineno">2935</tt>  <tt class="py-line">    <tt class="py-comment"># In order to avoid some code duplication, the old code would turn the</tt> </tt>
<a name="L2936"></a><tt class="py-lineno">2936</tt>  <tt class="py-line">    <tt class="py-comment"># string into a Seq, use the reverse_complement method, and convert back</tt> </tt>
<a name="L2937"></a><tt class="py-lineno">2937</tt>  <tt class="py-line">    <tt class="py-comment"># to a string.</tt> </tt>
<a name="L2938"></a><tt class="py-lineno">2938</tt>  <tt class="py-line">    <tt class="py-comment"># This worked, but is over five times slower on short sequences!</tt> </tt>
<a name="L2939"></a><tt class="py-lineno">2939</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-op">(</tt><tt class="py-string">"U"</tt> <tt class="py-keyword">in</tt> <tt id="link-1088" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1088', 'sequence', 'link-996');">sequence</a></tt> <tt class="py-keyword">or</tt> <tt class="py-string">"u"</tt> <tt class="py-keyword">in</tt> <tt id="link-1089" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1089', 'sequence', 'link-996');">sequence</a></tt><tt class="py-op">)</tt> \ </tt>
<a name="L2940"></a><tt class="py-lineno">2940</tt>  <tt class="py-line">            <tt class="py-keyword">and</tt> <tt class="py-op">(</tt><tt class="py-string">"T"</tt> <tt class="py-keyword">in</tt> <tt id="link-1090" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1090', 'sequence', 'link-996');">sequence</a></tt> <tt class="py-keyword">or</tt> <tt class="py-string">"t"</tt> <tt class="py-keyword">in</tt> <tt id="link-1091" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1091', 'sequence', 'link-996');">sequence</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2941"></a><tt class="py-lineno">2941</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Mixed RNA/DNA found"</tt><tt class="py-op">)</tt> </tt>
<a name="L2942"></a><tt class="py-lineno">2942</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-string">"U"</tt> <tt class="py-keyword">in</tt> <tt id="link-1092" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1092', 'sequence', 'link-996');">sequence</a></tt> <tt class="py-keyword">or</tt> <tt class="py-string">"u"</tt> <tt class="py-keyword">in</tt> <tt id="link-1093" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1093', 'sequence', 'link-996');">sequence</a></tt><tt class="py-op">:</tt> </tt>
<a name="L2943"></a><tt class="py-lineno">2943</tt>  <tt class="py-line">        <tt class="py-name">ttable</tt> <tt class="py-op">=</tt> <tt id="link-1094" class="py-name"><a title="Bio.Seq._rna_complement_table" class="py-name" href="#" onclick="return doclink('link-1094', '_rna_complement_table', 'link-37');">_rna_complement_table</a></tt> </tt>
<a name="L2944"></a><tt class="py-lineno">2944</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L2945"></a><tt class="py-lineno">2945</tt>  <tt class="py-line">        <tt class="py-name">ttable</tt> <tt class="py-op">=</tt> <tt id="link-1095" class="py-name"><a title="Bio.Seq._dna_complement_table" class="py-name" href="#" onclick="return doclink('link-1095', '_dna_complement_table', 'link-34');">_dna_complement_table</a></tt> </tt>
<a name="L2946"></a><tt class="py-lineno">2946</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-1096" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-1096', 'sequence', 'link-996');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-1097" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.translate
Bio.Seq.Seq.translate
Bio.Seq.UnknownSeq.translate
Bio.Seq.translate
Bio.SeqFeature.SeqFeature.translate
Bio.SeqRecord.SeqRecord.translate
Bio.codonalign.codonseq.CodonSeq.translate" class="py-name" href="#" onclick="return doclink('link-1097', 'translate', 'link-310');">translate</a></tt><tt class="py-op">(</tt><tt class="py-name">ttable</tt><tt class="py-op">)</tt> </tt>
</div><a name="L2947"></a><tt class="py-lineno">2947</tt>  <tt class="py-line"> </tt>
<a name="L2948"></a><tt class="py-lineno">2948</tt>  <tt class="py-line"> </tt>
<a name="_test"></a><div id="_test-def"><a name="L2949"></a><tt class="py-lineno">2949</tt> <a class="py-toggle" href="#" id="_test-toggle" onclick="return toggle('_test');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Seq-module.html#_test">_test</a><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_test-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_test-expanded"><a name="L2950"></a><tt class="py-lineno">2950</tt>  <tt class="py-line">    <tt class="py-docstring">"""Run the Bio.Seq module's doctests (PRIVATE)."""</tt> </tt>
<a name="L2951"></a><tt class="py-lineno">2951</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Running doctests..."</tt><tt class="py-op">)</tt> </tt>
<a name="L2952"></a><tt class="py-lineno">2952</tt>  <tt class="py-line">    <tt class="py-keyword">import</tt> <tt class="py-name">doctest</tt> </tt>
<a name="L2953"></a><tt class="py-lineno">2953</tt>  <tt class="py-line">    <tt class="py-name">doctest</tt><tt class="py-op">.</tt><tt class="py-name">testmod</tt><tt class="py-op">(</tt><tt class="py-name">optionflags</tt><tt class="py-op">=</tt><tt class="py-name">doctest</tt><tt class="py-op">.</tt><tt class="py-name">IGNORE_EXCEPTION_DETAIL</tt><tt class="py-op">)</tt> </tt>
<a name="L2954"></a><tt class="py-lineno">2954</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Done"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L2955"></a><tt class="py-lineno">2955</tt>  <tt class="py-line"> </tt>
<a name="L2956"></a><tt class="py-lineno">2956</tt>  <tt class="py-line"> </tt>
<a name="L2957"></a><tt class="py-lineno">2957</tt>  <tt class="py-line"><tt class="py-keyword">if</tt> <tt class="py-name">__name__</tt> <tt class="py-op">==</tt> <tt class="py-string">"__main__"</tt><tt class="py-op">:</tt> </tt>
<a name="L2958"></a><tt class="py-lineno">2958</tt>  <tt class="py-line">    <tt id="link-1098" class="py-name" targets="Function Bio.Application._test()=Bio.Application-module.html#_test,Function Bio.Blast.Applications._test()=Bio.Blast.Applications-module.html#_test,Function Bio.Seq._test()=Bio.Seq-module.html#_test,Function Bio.Wise._test()=Bio.Wise-module.html#_test,Function Bio.Wise.dnal._test()=Bio.Wise.dnal-module.html#_test,Function Bio.Wise.psw._test()=Bio.Wise.psw-module.html#_test"><a title="Bio.Application._test
Bio.Blast.Applications._test
Bio.Seq._test
Bio.Wise._test
Bio.Wise.dnal._test
Bio.Wise.psw._test" class="py-name" href="#" onclick="return doclink('link-1098', '_test', 'link-1098');">_test</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L2959"></a><tt class="py-lineno">2959</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Nov  6 13:34:08 2019
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
