<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.motifs.meme</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.motifs-module.html">Package&nbsp;motifs</a> ::
        Module&nbsp;meme
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.motifs.meme-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.motifs.meme-module.html">Module Bio.motifs.meme</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2008 by Bartek Wilczynski</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># Adapted from  Bio.MEME.Parser by Jason A. Hackney.  All rights reserved.</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">__future__</tt> <tt class="py-keyword">import</tt> <tt class="py-name">print_function</tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-1', 'Alphabet', 'link-1');">Alphabet</a></tt> <tt class="py-keyword">import</tt> <tt id="link-2" class="py-name" targets="Module Bio.Alphabet.IUPAC=Bio.Alphabet.IUPAC-module.html"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-2', 'IUPAC', 'link-2');">IUPAC</a></tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-3" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-3', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-4" class="py-name" targets="Module Bio.Seq=Bio.Seq-module.html,Class Bio.Seq.Seq=Bio.Seq.Seq-class.html"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-4', 'Seq', 'link-4');">Seq</a></tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-5" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-5', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-6" class="py-name" targets="Package Bio.motifs=Bio.motifs-module.html"><a title="Bio.motifs" class="py-name" href="#" onclick="return doclink('link-6', 'motifs', 'link-6');">motifs</a></tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"> </tt>
<a name="read"></a><div id="read-def"><a name="L14"></a><tt class="py-lineno"> 14</tt> <a class="py-toggle" href="#" id="read-toggle" onclick="return toggle('read');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.meme-module.html#read">read</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="read-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="read-expanded"><a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line">    <tt class="py-docstring">"""Parses the text output of the MEME program into a meme.Record object.</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-docstring">    Example:</tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio.motifs import meme</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; with open("meme.output.txt") as f:</tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-docstring">    ...     record = meme.read(f)</tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; for motif in record:</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-docstring">    ...     for instance in motif.instances:</tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-docstring">    ...         print(instance.motif_name, instance.sequence_name, instance.strand, instance.pvalue)</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line">    <tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt id="link-7" class="py-name" targets="Class Bio.Affy.CelFile.Record=Bio.Affy.CelFile.Record-class.html,Module Bio.Blast.Record=Bio.Blast.Record-module.html,Class Bio.Cluster.Record=Bio.Cluster.Record-class.html,Class Bio.Compass.Record=Bio.Compass.Record-class.html,Class Bio.Emboss.Primer3.Record=Bio.Emboss.Primer3.Record-class.html,Class Bio.ExPASy.Enzyme.Record=Bio.ExPASy.Enzyme.Record-class.html,Class Bio.ExPASy.Prodoc.Record=Bio.ExPASy.Prodoc.Record-class.html,Class Bio.ExPASy.Prosite.Record=Bio.ExPASy.Prosite.Record-class.html,Class Bio.ExPASy.ScanProsite.Record=Bio.ExPASy.ScanProsite.Record-class.html,Module Bio.GenBank.Record=Bio.GenBank.Record-module.html,Class Bio.GenBank.Record.Record=Bio.GenBank.Record.Record-class.html,Module Bio.Geo.Record=Bio.Geo.Record-module.html,Class Bio.Geo.Record.Record=Bio.Geo.Record.Record-class.html,Class Bio.KEGG.Compound.Record=Bio.KEGG.Compound.Record-class.html,Class Bio.KEGG.Enzyme.Record=Bio.KEGG.Enzyme.Record-class.html,Class Bio.Medline.Record=Bio.Medline.Record-class.html,Class Bio.Motif.Parsers.AlignAce.Record=Bio.Motif.Parsers.AlignAce.Record-class.html,Class Bio.Motif.Parsers.MAST.Record=Bio.Motif.Parsers.MAST.Record-class.html,Class Bio.PopGen.FDist.Record=Bio.PopGen.FDist.Record-class.html,Class Bio.PopGen.GenePop.LargeFileParser.Record=Bio.PopGen.GenePop.LargeFileParser.Record-class.html,Class Bio.PopGen.GenePop.Record=Bio.PopGen.GenePop.Record-class.html,Class Bio.SCOP.Cla.Record=Bio.SCOP.Cla.Record-class.html,Class Bio.SCOP.Des.Record=Bio.SCOP.Des.Record-class.html,Class Bio.SCOP.Dom.Record=Bio.SCOP.Dom.Record-class.html,Class Bio.SCOP.Hie.Record=Bio.SCOP.Hie.Record-class.html,Class Bio.Sequencing.Phd.Record=Bio.Sequencing.Phd.Record-class.html,Class Bio.SwissProt.KeyWList.Record=Bio.SwissProt.KeyWList.Record-class.html,Class Bio.SwissProt.Record=Bio.SwissProt.Record-class.html,Class Bio.UniGene.Record=Bio.UniGene.Record-class.html,Class Bio.motifs.alignace.Record=Bio.motifs.alignace.Record-class.html,Class Bio.motifs.jaspar.Record=Bio.motifs.jaspar.Record-class.html,Class Bio.motifs.mast.Record=Bio.motifs.mast.Record-class.html,Class Bio.motifs.meme.Record=Bio.motifs.meme.Record-class.html,Class Bio.motifs.transfac.Record=Bio.motifs.transfac.Record-class.html"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-7', 'Record', 'link-7');">Record</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line">    <tt id="link-8" class="py-name" targets="Function Bio.Motif.Parsers.MAST.__read_version()=Bio.Motif.Parsers.MAST-module.html#__read_version,Function Bio.Motif.Parsers.MEME.__read_version()=Bio.Motif.Parsers.MEME-module.html#__read_version,Function Bio.motifs.mast.__read_version()=Bio.motifs.mast-module.html#__read_version,Function Bio.motifs.meme.__read_version()=Bio.motifs.meme-module.html#__read_version"><a title="Bio.Motif.Parsers.MAST.__read_version
Bio.Motif.Parsers.MEME.__read_version
Bio.motifs.mast.__read_version
Bio.motifs.meme.__read_version" class="py-name" href="#" onclick="return doclink('link-8', '__read_version', 'link-8');">__read_version</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">,</tt> <tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line">    <tt id="link-9" class="py-name" targets="Function Bio.Motif.Parsers.MEME.__read_datafile()=Bio.Motif.Parsers.MEME-module.html#__read_datafile,Function Bio.motifs.meme.__read_datafile()=Bio.motifs.meme-module.html#__read_datafile"><a title="Bio.Motif.Parsers.MEME.__read_datafile
Bio.motifs.meme.__read_datafile" class="py-name" href="#" onclick="return doclink('link-9', '__read_datafile', 'link-9');">__read_datafile</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">,</tt> <tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line">    <tt id="link-10" class="py-name" targets="Function Bio.Motif.Parsers.MEME.__read_alphabet()=Bio.Motif.Parsers.MEME-module.html#__read_alphabet,Function Bio.motifs.meme.__read_alphabet()=Bio.motifs.meme-module.html#__read_alphabet"><a title="Bio.Motif.Parsers.MEME.__read_alphabet
Bio.motifs.meme.__read_alphabet" class="py-name" href="#" onclick="return doclink('link-10', '__read_alphabet', 'link-10');">__read_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">,</tt> <tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line">    <tt id="link-11" class="py-name" targets="Function Bio.motifs.meme.__read_sequences()=Bio.motifs.meme-module.html#__read_sequences"><a title="Bio.motifs.meme.__read_sequences" class="py-name" href="#" onclick="return doclink('link-11', '__read_sequences', 'link-11');">__read_sequences</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">,</tt> <tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line">    <tt id="link-12" class="py-name" targets="Function Bio.Motif.Parsers.MEME.__read_command()=Bio.Motif.Parsers.MEME-module.html#__read_command,Function Bio.motifs.meme.__read_command()=Bio.motifs.meme-module.html#__read_command"><a title="Bio.Motif.Parsers.MEME.__read_command
Bio.motifs.meme.__read_command" class="py-name" href="#" onclick="return doclink('link-12', '__read_command', 'link-12');">__read_command</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">,</tt> <tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-13" class="py-name" targets="Method Bio.Seq.Seq.startswith()=Bio.Seq.Seq-class.html#startswith"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-13', 'startswith', 'link-13');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'MOTIF  1'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line">            <tt class="py-keyword">break</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">'Unexpected end of stream'</tt><tt class="py-op">)</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line">    <tt id="link-14" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.alphabet=Bio.SearchIO._model.hsp.HSP-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSPFragment.alphabet=Bio.SearchIO._model.hsp.HSPFragment-class.html#alphabet"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-14', 'alphabet', 'link-14');">alphabet</a></tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-15" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-15', 'alphabet', 'link-14');">alphabet</a></tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line">    <tt class="py-name">revcomp</tt> <tt class="py-op">=</tt> <tt class="py-string">'revcomp'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">command</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line">    <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line">        <tt class="py-name">motif_number</tt><tt class="py-op">,</tt> <tt id="link-16" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.length()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#length"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-16', 'length', 'link-16');">length</a></tt><tt class="py-op">,</tt> <tt class="py-name">num_occurrences</tt><tt class="py-op">,</tt> <tt class="py-name">evalue</tt> <tt class="py-op">=</tt> <tt id="link-17" class="py-name" targets="Function Bio.motifs.meme.__read_motif_statistics()=Bio.motifs.meme-module.html#__read_motif_statistics"><a title="Bio.motifs.meme.__read_motif_statistics" class="py-name" href="#" onclick="return doclink('link-17', '__read_motif_statistics', 'link-17');">__read_motif_statistics</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line">        <tt id="link-18" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-18', 'name', 'link-18');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-19" class="py-name" targets="Function Bio.Motif.Parsers.MEME.__read_motif_name()=Bio.Motif.Parsers.MEME-module.html#__read_motif_name,Function Bio.motifs.meme.__read_motif_name()=Bio.motifs.meme-module.html#__read_motif_name"><a title="Bio.Motif.Parsers.MEME.__read_motif_name
Bio.motifs.meme.__read_motif_name" class="py-name" href="#" onclick="return doclink('link-19', '__read_motif_name', 'link-19');">__read_motif_name</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line">        <tt class="py-name">instances</tt> <tt class="py-op">=</tt> <tt id="link-20" class="py-name" targets="Function Bio.Motif.Parsers.MEME.__read_motif_sequences()=Bio.Motif.Parsers.MEME-module.html#__read_motif_sequences,Function Bio.motifs.meme.__read_motif_sequences()=Bio.motifs.meme-module.html#__read_motif_sequences"><a title="Bio.Motif.Parsers.MEME.__read_motif_sequences
Bio.motifs.meme.__read_motif_sequences" class="py-name" href="#" onclick="return doclink('link-20', '__read_motif_sequences', 'link-20');">__read_motif_sequences</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt id="link-21" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-21', 'name', 'link-18');">name</a></tt><tt class="py-op">,</tt> <tt id="link-22" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-22', 'alphabet', 'link-14');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-23" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-23', 'length', 'link-16');">length</a></tt><tt class="py-op">,</tt> <tt class="py-name">revcomp</tt><tt class="py-op">)</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line">        <tt class="py-name">motif</tt> <tt class="py-op">=</tt> <tt id="link-24" class="py-name" targets="Package Bio.Motif=Bio.Motif-module.html,Class Bio.Motif._Motif.Motif=Bio.Motif._Motif.Motif-class.html,Module Bio.NeuralNetwork.Gene.Motif=Bio.NeuralNetwork.Gene.Motif-module.html,Class Bio.motifs.Motif=Bio.motifs.Motif-class.html,Class Bio.motifs.jaspar.Motif=Bio.motifs.jaspar.Motif-class.html,Class Bio.motifs.meme.Motif=Bio.motifs.meme.Motif-class.html,Class Bio.motifs.transfac.Motif=Bio.motifs.transfac.Motif-class.html"><a title="Bio.Motif
Bio.Motif._Motif.Motif
Bio.NeuralNetwork.Gene.Motif
Bio.motifs.Motif
Bio.motifs.jaspar.Motif
Bio.motifs.meme.Motif
Bio.motifs.transfac.Motif" class="py-name" href="#" onclick="return doclink('link-24', 'Motif', 'link-24');">Motif</a></tt><tt class="py-op">(</tt><tt id="link-25" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-25', 'alphabet', 'link-14');">alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-name">instances</tt><tt class="py-op">)</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line">        <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt id="link-26" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-26', 'length', 'link-16');">length</a></tt> <tt class="py-op">=</tt> <tt id="link-27" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-27', 'length', 'link-16');">length</a></tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line">        <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt class="py-name">num_occurrences</tt> <tt class="py-op">=</tt> <tt class="py-name">num_occurrences</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line">        <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt class="py-name">evalue</tt> <tt class="py-op">=</tt> <tt class="py-name">evalue</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line">        <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt id="link-28" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-28', 'name', 'link-18');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-29" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-29', 'name', 'link-18');">name</a></tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line">        <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-30" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-30', 'append', 'link-30');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">motif</tt><tt class="py-op">)</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt><tt class="py-op">==</tt><tt class="py-name">motif_number</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line">        <tt id="link-31" class="py-name" targets="Function Bio.Motif.Parsers.MEME.__skip_unused_lines()=Bio.Motif.Parsers.MEME-module.html#__skip_unused_lines,Function Bio.motifs.meme.__skip_unused_lines()=Bio.motifs.meme-module.html#__skip_unused_lines"><a title="Bio.Motif.Parsers.MEME.__skip_unused_lines
Bio.motifs.meme.__skip_unused_lines" class="py-name" href="#" onclick="return doclink('link-31', '__skip_unused_lines', 'link-31');">__skip_unused_lines</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-32" class="py-name" targets="Method Bio.AlignIO.Interfaces.AlignmentIterator.next()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#next,Method Bio.Blast.NCBIStandalone.Iterator.next()=Bio.Blast.NCBIStandalone.Iterator-class.html#next,Method Bio.Blast.ParseBlastTable.BlastTableReader.next()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#next,Method Bio.DocSQL.IterationCursor.next()=Bio.DocSQL.IterationCursor-class.html#next,Method Bio.File.UndoHandle.next()=Bio.File.UndoHandle-class.html#next,Method Bio.GenBank.Iterator.next()=Bio.GenBank.Iterator-class.html#next,Method Bio.Nexus.Nexus.CharBuffer.next()=Bio.Nexus.Nexus.CharBuffer-class.html#next,Method Bio.PopGen.GenePop.Controller._FileIterator.next()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#next,Method Bio.SeqIO.Interfaces.SequenceIterator.next()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#next,Method Bio.bgzf.BgzfReader.next()=Bio.bgzf.BgzfReader-class.html#next"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-32', 'next', 'link-32');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">'Unexpected end of stream: Expected to find new motif, or the summary of motifs'</tt><tt class="py-op">)</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-33" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-33', 'startswith', 'link-13');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"SUMMARY OF MOTIFS"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line">            <tt class="py-keyword">break</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-34" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-34', 'startswith', 'link-13');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'MOTIF'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Line does not start with 'MOTIF':\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">record</tt> </tt>
</div><a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line"> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line"> </tt>
<a name="Motif"></a><div id="Motif-def"><a name="L63"></a><tt class="py-lineno"> 63</tt> <a class="py-toggle" href="#" id="Motif-toggle" onclick="return toggle('Motif');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.motifs.meme.Motif-class.html">Motif</a><tt class="py-op">(</tt><tt class="py-base-class">motifs</tt><tt class="py-op">.</tt><tt class="py-base-class">Motif</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Motif-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Motif-expanded"><a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line">    <tt class="py-docstring">"""A subclass of Motif used in parsing MEME (and MAST) output.</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line"><tt class="py-docstring">    This subclass defines functions and data specific to MEME motifs.</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line"><tt class="py-docstring">    This includes the motif name, the evalue for a motif, and its number</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line"><tt class="py-docstring">    of occurrences.</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="Motif.__init__"></a><div id="Motif.__init__-def"><a name="L70"></a><tt class="py-lineno"> 70</tt> <a class="py-toggle" href="#" id="Motif.__init__-toggle" onclick="return toggle('Motif.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.meme.Motif-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">instances</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Motif.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Motif.__init__-expanded"><a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line">        <tt id="link-35" class="py-name"><a title="Bio.motifs" class="py-name" href="#" onclick="return doclink('link-35', 'motifs', 'link-6');">motifs</a></tt><tt class="py-op">.</tt><tt id="link-36" class="py-name"><a title="Bio.Motif
Bio.Motif._Motif.Motif
Bio.NeuralNetwork.Gene.Motif
Bio.motifs.Motif
Bio.motifs.jaspar.Motif
Bio.motifs.meme.Motif
Bio.motifs.transfac.Motif" class="py-name" href="#" onclick="return doclink('link-36', 'Motif', 'link-24');">Motif</a></tt><tt class="py-op">.</tt><tt id="link-37" class="py-name" targets="Method Bio.Affy.CelFile.Record.__init__()=Bio.Affy.CelFile.Record-class.html#__init__,Method Bio.Align.AlignInfo.PSSM.__init__()=Bio.Align.AlignInfo.PSSM-class.html#__init__,Method Bio.Align.AlignInfo.SummaryInfo.__init__()=Bio.Align.AlignInfo.SummaryInfo-class.html#__init__,Method Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline.__init__()=Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline-class.html#__init__,Method Bio.Align.Applications._Clustalw.ClustalwCommandline.__init__()=Bio.Align.Applications._Clustalw.ClustalwCommandline-class.html#__init__,Method Bio.Align.Applications._Dialign.DialignCommandline.__init__()=Bio.Align.Applications._Dialign.DialignCommandline-class.html#__init__,Method Bio.Align.Applications._MSAProbs.MSAProbsCommandline.__init__()=Bio.Align.Applications._MSAProbs.MSAProbsCommandline-class.html#__init__,Method Bio.Align.Applications._Mafft.MafftCommandline.__init__()=Bio.Align.Applications._Mafft.MafftCommandline-class.html#__init__,Method Bio.Align.Applications._Muscle.MuscleCommandline.__init__()=Bio.Align.Applications._Muscle.MuscleCommandline-class.html#__init__,Method Bio.Align.Applications._Prank.PrankCommandline.__init__()=Bio.Align.Applications._Prank.PrankCommandline-class.html#__init__,Method Bio.Align.Applications._Probcons.ProbconsCommandline.__init__()=Bio.Align.Applications._Probcons.ProbconsCommandline-class.html#__init__,Method Bio.Align.Applications._TCoffee.TCoffeeCommandline.__init__()=Bio.Align.Applications._TCoffee.TCoffeeCommandline-class.html#__init__,Method Bio.Align.Generic.Alignment.__init__()=Bio.Align.Generic.Alignment-class.html#__init__,Method Bio.Align.MultipleSeqAlignment.__init__()=Bio.Align.MultipleSeqAlignment-class.html#__init__,Method Bio.AlignIO.Interfaces.AlignmentIterator.__init__()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#__init__,Method Bio.AlignIO.Interfaces.AlignmentWriter.__init__()=Bio.AlignIO.Interfaces.AlignmentWriter-class.html#__init__,Method Bio.AlignIO.Interfaces.SequentialAlignmentWriter.__init__()=Bio.AlignIO.Interfaces.SequentialAlignmentWriter-class.html#__init__,Method Bio.Alphabet.AlphabetEncoder.__init__()=Bio.Alphabet.AlphabetEncoder-class.html#__init__,Method Bio.Alphabet.Gapped.__init__()=Bio.Alphabet.Gapped-class.html#__init__,Method Bio.Alphabet.HasStopCodon.__init__()=Bio.Alphabet.HasStopCodon-class.html#__init__,Method Bio.Application.AbstractCommandline.__init__()=Bio.Application.AbstractCommandline-class.html#__init__,Method Bio.Application.ApplicationError.__init__()=Bio.Application.ApplicationError-class.html#__init__,Method Bio.Application._AbstractParameter.__init__()=Bio.Application._AbstractParameter-class.html#__init__,Method Bio.Application._Argument.__init__()=Bio.Application._Argument-class.html#__init__,Method Bio.Application._Option.__init__()=Bio.Application._Option-class.html#__init__,Method Bio.Application._StaticArgument.__init__()=Bio.Application._StaticArgument-class.html#__init__,Method Bio.Application._Switch.__init__()=Bio.Application._Switch-class.html#__init__,Method Bio.Blast.Applications.NcbiblastformatterCommandline.__init__()=Bio.Blast.Applications.NcbiblastformatterCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastnCommandline.__init__()=Bio.Blast.Applications.NcbiblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastpCommandline.__init__()=Bio.Blast.Applications.NcbiblastpCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastxCommandline.__init__()=Bio.Blast.Applications.NcbiblastxCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbideltablastCommandline.__init__()=Bio.Blast.Applications.NcbideltablastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbipsiblastCommandline.__init__()=Bio.Blast.Applications.NcbipsiblastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbirpsblastCommandline.__init__()=Bio.Blast.Applications.NcbirpsblastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbirpstblastnCommandline.__init__()=Bio.Blast.Applications.NcbirpstblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbitblastnCommandline.__init__()=Bio.Blast.Applications.NcbitblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbitblastxCommandline.__init__()=Bio.Blast.Applications.NcbitblastxCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbibaseblastCommandline.__init__()=Bio.Blast.Applications._NcbibaseblastCommandline-class.html#__init__,Method Bio.Blast.Applications._Ncbiblast2SeqCommandline.__init__()=Bio.Blast.Applications._Ncbiblast2SeqCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbiblastCommandline.__init__()=Bio.Blast.Applications._NcbiblastCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbiblastMain2SeqCommandline.__init__()=Bio.Blast.Applications._NcbiblastMain2SeqCommandline-class.html#__init__,Method Bio.Blast.NCBIStandalone.BlastErrorParser.__init__()=Bio.Blast.NCBIStandalone.BlastErrorParser-class.html#__init__,Method Bio.Blast.NCBIStandalone.BlastParser.__init__()=Bio.Blast.NCBIStandalone.BlastParser-class.html#__init__,Method Bio.Blast.NCBIStandalone.Iterator.__init__()=Bio.Blast.NCBIStandalone.Iterator-class.html#__init__,Method Bio.Blast.NCBIStandalone.PSIBlastParser.__init__()=Bio.Blast.NCBIStandalone.PSIBlastParser-class.html#__init__,Method Bio.Blast.NCBIStandalone._BlastConsumer.__init__()=Bio.Blast.NCBIStandalone._BlastConsumer-class.html#__init__,Method Bio.Blast.NCBIStandalone._BlastErrorConsumer.__init__()=Bio.Blast.NCBIStandalone._BlastErrorConsumer-class.html#__init__,Method Bio.Blast.NCBIStandalone._PSIBlastConsumer.__init__()=Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html#__init__,Method Bio.Blast.NCBIXML.BlastParser.__init__()=Bio.Blast.NCBIXML.BlastParser-class.html#__init__,Method Bio.Blast.NCBIXML._XMLparser.__init__()=Bio.Blast.NCBIXML._XMLparser-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableEntry.__init__()=Bio.Blast.ParseBlastTable.BlastTableEntry-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableReader.__init__()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableRec.__init__()=Bio.Blast.ParseBlastTable.BlastTableRec-class.html#__init__,Method Bio.Blast.Record.Alignment.__init__()=Bio.Blast.Record.Alignment-class.html#__init__,Method Bio.Blast.Record.Blast.__init__()=Bio.Blast.Record.Blast-class.html#__init__,Method Bio.Blast.Record.DatabaseReport.__init__()=Bio.Blast.Record.DatabaseReport-class.html#__init__,Method Bio.Blast.Record.Description.__init__()=Bio.Blast.Record.Description-class.html#__init__,Method Bio.Blast.Record.HSP.__init__()=Bio.Blast.Record.HSP-class.html#__init__,Method Bio.Blast.Record.Header.__init__()=Bio.Blast.Record.Header-class.html#__init__,Method Bio.Blast.Record.MultipleAlignment.__init__()=Bio.Blast.Record.MultipleAlignment-class.html#__init__,Method Bio.Blast.Record.PSIBlast.__init__()=Bio.Blast.Record.PSIBlast-class.html#__init__,Method Bio.Blast.Record.Parameters.__init__()=Bio.Blast.Record.Parameters-class.html#__init__,Method Bio.Blast.Record.Round.__init__()=Bio.Blast.Record.Round-class.html#__init__,Method Bio.CAPS.CAPSMap.__init__()=Bio.CAPS.CAPSMap-class.html#__init__,Method Bio.CAPS.DifferentialCutsite.__init__()=Bio.CAPS.DifferentialCutsite-class.html#__init__,Method Bio.Cluster.Record.__init__()=Bio.Cluster.Record-class.html#__init__,Method Bio.Compass.Record.__init__()=Bio.Compass.Record-class.html#__init__,Method Bio.Crystal.Chain.__init__()=Bio.Crystal.Chain-class.html#__init__,Method Bio.Crystal.Crystal.__init__()=Bio.Crystal.Crystal-class.html#__init__,Method Bio.Crystal.Hetero.__init__()=Bio.Crystal.Hetero-class.html#__init__,Method Bio.Data.CodonTable.AmbiguousCodonTable.__init__()=Bio.Data.CodonTable.AmbiguousCodonTable-class.html#__init__,Method Bio.Data.CodonTable.AmbiguousForwardTable.__init__()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#__init__,Method Bio.Data.CodonTable.CodonTable.__init__()=Bio.Data.CodonTable.CodonTable-class.html#__init__,Method Bio.Data.CodonTable.NCBICodonTable.__init__()=Bio.Data.CodonTable.NCBICodonTable-class.html#__init__,Method Bio.DocSQL.Create.__init__()=Bio.DocSQL.Create-class.html#__init__,Method Bio.DocSQL.Insert.__init__()=Bio.DocSQL.Insert-class.html#__init__,Method Bio.DocSQL.IterationCursor.__init__()=Bio.DocSQL.IterationCursor-class.html#__init__,Method Bio.DocSQL.Query.__init__()=Bio.DocSQL.Query-class.html#__init__,Method Bio.DocSQL.QueryAll.__init__()=Bio.DocSQL.QueryAll-class.html#__init__,Method Bio.DocSQL.QueryGeneric.__init__()=Bio.DocSQL.QueryGeneric-class.html#__init__,Method Bio.DocSQL.QueryRow.__init__()=Bio.DocSQL.QueryRow-class.html#__init__,Method Bio.DocSQL.QuerySingle.__init__()=Bio.DocSQL.QuerySingle-class.html#__init__,Method Bio.Emboss.Applications.DiffseqCommandline.__init__()=Bio.Emboss.Applications.DiffseqCommandline-class.html#__init__,Method Bio.Emboss.Applications.EInvertedCommandline.__init__()=Bio.Emboss.Applications.EInvertedCommandline-class.html#__init__,Method Bio.Emboss.Applications.ETandemCommandline.__init__()=Bio.Emboss.Applications.ETandemCommandline-class.html#__init__,Method Bio.Emboss.Applications.Est2GenomeCommandline.__init__()=Bio.Emboss.Applications.Est2GenomeCommandline-class.html#__init__,Method Bio.Emboss.Applications.FConsenseCommandline.__init__()=Bio.Emboss.Applications.FConsenseCommandline-class.html#__init__,Method Bio.Emboss.Applications.FDNADistCommandline.__init__()=Bio.Emboss.Applications.FDNADistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FDNAParsCommandline.__init__()=Bio.Emboss.Applications.FDNAParsCommandline-class.html#__init__,Method Bio.Emboss.Applications.FNeighborCommandline.__init__()=Bio.Emboss.Applications.FNeighborCommandline-class.html#__init__,Method Bio.Emboss.Applications.FProtDistCommandline.__init__()=Bio.Emboss.Applications.FProtDistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FProtParsCommandline.__init__()=Bio.Emboss.Applications.FProtParsCommandline-class.html#__init__,Method Bio.Emboss.Applications.FSeqBootCommandline.__init__()=Bio.Emboss.Applications.FSeqBootCommandline-class.html#__init__,Method Bio.Emboss.Applications.FTreeDistCommandline.__init__()=Bio.Emboss.Applications.FTreeDistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FuzznucCommandline.__init__()=Bio.Emboss.Applications.FuzznucCommandline-class.html#__init__,Method Bio.Emboss.Applications.IepCommandline.__init__()=Bio.Emboss.Applications.IepCommandline-class.html#__init__,Method Bio.Emboss.Applications.NeedleCommandline.__init__()=Bio.Emboss.Applications.NeedleCommandline-class.html#__init__,Method Bio.Emboss.Applications.NeedleallCommandline.__init__()=Bio.Emboss.Applications.NeedleallCommandline-class.html#__init__,Method Bio.Emboss.Applications.PalindromeCommandline.__init__()=Bio.Emboss.Applications.PalindromeCommandline-class.html#__init__,Method Bio.Emboss.Applications.Primer3Commandline.__init__()=Bio.Emboss.Applications.Primer3Commandline-class.html#__init__,Method Bio.Emboss.Applications.PrimerSearchCommandline.__init__()=Bio.Emboss.Applications.PrimerSearchCommandline-class.html#__init__,Method Bio.Emboss.Applications.SeqmatchallCommandline.__init__()=Bio.Emboss.Applications.SeqmatchallCommandline-class.html#__init__,Method Bio.Emboss.Applications.SeqretCommandline.__init__()=Bio.Emboss.Applications.SeqretCommandline-class.html#__init__,Method Bio.Emboss.Applications.StretcherCommandline.__init__()=Bio.Emboss.Applications.StretcherCommandline-class.html#__init__,Method Bio.Emboss.Applications.TranalignCommandline.__init__()=Bio.Emboss.Applications.TranalignCommandline-class.html#__init__,Method Bio.Emboss.Applications.WaterCommandline.__init__()=Bio.Emboss.Applications.WaterCommandline-class.html#__init__,Method Bio.Emboss.Applications._EmbossCommandLine.__init__()=Bio.Emboss.Applications._EmbossCommandLine-class.html#__init__,Method Bio.Emboss.Applications._EmbossMinimalCommandLine.__init__()=Bio.Emboss.Applications._EmbossMinimalCommandLine-class.html#__init__,Method Bio.Emboss.Primer3.Primers.__init__()=Bio.Emboss.Primer3.Primers-class.html#__init__,Method Bio.Emboss.Primer3.Record.__init__()=Bio.Emboss.Primer3.Record-class.html#__init__,Method Bio.Emboss.PrimerSearch.Amplifier.__init__()=Bio.Emboss.PrimerSearch.Amplifier-class.html#__init__,Method Bio.Emboss.PrimerSearch.InputRecord.__init__()=Bio.Emboss.PrimerSearch.InputRecord-class.html#__init__,Method Bio.Emboss.PrimerSearch.OutputRecord.__init__()=Bio.Emboss.PrimerSearch.OutputRecord-class.html#__init__,Method Bio.Entrez.Parser.CorruptedXMLError.__init__()=Bio.Entrez.Parser.CorruptedXMLError-class.html#__init__,Method Bio.Entrez.Parser.DataHandler.__init__()=Bio.Entrez.Parser.DataHandler-class.html#__init__,Method Bio.Entrez.Parser.NotXMLError.__init__()=Bio.Entrez.Parser.NotXMLError-class.html#__init__,Method Bio.Entrez.Parser.StructureElement.__init__()=Bio.Entrez.Parser.StructureElement-class.html#__init__,Method Bio.Entrez.Parser.ValidationError.__init__()=Bio.Entrez.Parser.ValidationError-class.html#__init__,Method Bio.ExPASy.Enzyme.Record.__init__()=Bio.ExPASy.Enzyme.Record-class.html#__init__,Method Bio.ExPASy.Prodoc.Record.__init__()=Bio.ExPASy.Prodoc.Record-class.html#__init__,Method Bio.ExPASy.Prodoc.Reference.__init__()=Bio.ExPASy.Prodoc.Reference-class.html#__init__,Method Bio.ExPASy.Prosite.Record.__init__()=Bio.ExPASy.Prosite.Record-class.html#__init__,Method Bio.ExPASy.ScanProsite.ContentHandler.__init__()=Bio.ExPASy.ScanProsite.ContentHandler-class.html#__init__,Method Bio.ExPASy.ScanProsite.Parser.__init__()=Bio.ExPASy.ScanProsite.Parser-class.html#__init__,Method Bio.ExPASy.ScanProsite.Record.__init__()=Bio.ExPASy.ScanProsite.Record-class.html#__init__,Method Bio.FSSP.FSSPAlignDict.__init__()=Bio.FSSP.FSSPAlignDict-class.html#__init__,Method Bio.FSSP.FSSPAlignRec.__init__()=Bio.FSSP.FSSPAlignRec-class.html#__init__,Method Bio.FSSP.FSSPHeader.__init__()=Bio.FSSP.FSSPHeader-class.html#__init__,Method Bio.FSSP.FSSPSumRec.__init__()=Bio.FSSP.FSSPSumRec-class.html#__init__,Method Bio.FSSP.FSSPTools.FSSPMultAlign.__init__()=Bio.FSSP.FSSPTools.FSSPMultAlign-class.html#__init__,Method Bio.FSSP.PosAlign.__init__()=Bio.FSSP.PosAlign-class.html#__init__,Method Bio.FSSP.fssp_rec.fff_rec.__init__()=Bio.FSSP.fssp_rec.fff_rec-class.html#__init__,Method Bio.File.UndoHandle.__init__()=Bio.File.UndoHandle-class.html#__init__,Method Bio.File._IndexedSeqFileDict.__init__()=Bio.File._IndexedSeqFileDict-class.html#__init__,Method Bio.File._SQLiteManySeqFilesDict.__init__()=Bio.File._SQLiteManySeqFilesDict-class.html#__init__,Method Bio.GA.Crossover.General.SafeFitnessCrossover.__init__()=Bio.GA.Crossover.General.SafeFitnessCrossover-class.html#__init__,Method Bio.GA.Crossover.GeneralPoint.GeneralPointCrossover.__init__()=Bio.GA.Crossover.GeneralPoint.GeneralPointCrossover-class.html#__init__,Method Bio.GA.Crossover.GeneralPoint.InterleaveCrossover.__init__()=Bio.GA.Crossover.GeneralPoint.InterleaveCrossover-class.html#__init__,Method Bio.GA.Crossover.Point.SinglePointCrossover.__init__()=Bio.GA.Crossover.Point.SinglePointCrossover-class.html#__init__,Method Bio.GA.Crossover.TwoPoint.TwoPointCrossover.__init__()=Bio.GA.Crossover.TwoPoint.TwoPointCrossover-class.html#__init__,Method Bio.GA.Crossover.Uniform.UniformCrossover.__init__()=Bio.GA.Crossover.Uniform.UniformCrossover-class.html#__init__,Method Bio.GA.Evolver.GenerationEvolver.__init__()=Bio.GA.Evolver.GenerationEvolver-class.html#__init__,Method Bio.GA.Evolver.SteadyStateEvolver.__init__()=Bio.GA.Evolver.SteadyStateEvolver-class.html#__init__,Method Bio.GA.Mutation.General.SafeFitnessMutation.__init__()=Bio.GA.Mutation.General.SafeFitnessMutation-class.html#__init__,Method Bio.GA.Mutation.Simple.ConversionMutation.__init__()=Bio.GA.Mutation.Simple.ConversionMutation-class.html#__init__,Method Bio.GA.Mutation.Simple.SinglePositionMutation.__init__()=Bio.GA.Mutation.Simple.SinglePositionMutation-class.html#__init__,Method Bio.GA.Organism.Organism.__init__()=Bio.GA.Organism.Organism-class.html#__init__,Method Bio.GA.Repair.Stabilizing.AmbiguousRepair.__init__()=Bio.GA.Repair.Stabilizing.AmbiguousRepair-class.html#__init__,Method Bio.GA.Selection.Abstract.AbstractSelection.__init__()=Bio.GA.Selection.Abstract.AbstractSelection-class.html#__init__,Method Bio.GA.Selection.Diversity.DiversitySelection.__init__()=Bio.GA.Selection.Diversity.DiversitySelection-class.html#__init__,Method Bio.GA.Selection.RouletteWheel.RouletteWheelSelection.__init__()=Bio.GA.Selection.RouletteWheel.RouletteWheelSelection-class.html#__init__,Method Bio.GA.Selection.Tournament.TournamentSelection.__init__()=Bio.GA.Selection.Tournament.TournamentSelection-class.html#__init__,Method Bio.GenBank.FeatureParser.__init__()=Bio.GenBank.FeatureParser-class.html#__init__,Method Bio.GenBank.Iterator.__init__()=Bio.GenBank.Iterator-class.html#__init__,Method Bio.GenBank.Record.Feature.__init__()=Bio.GenBank.Record.Feature-class.html#__init__,Method Bio.GenBank.Record.Qualifier.__init__()=Bio.GenBank.Record.Qualifier-class.html#__init__,Method Bio.GenBank.Record.Record.__init__()=Bio.GenBank.Record.Record-class.html#__init__,Method Bio.GenBank.Record.Reference.__init__()=Bio.GenBank.Record.Reference-class.html#__init__,Method Bio.GenBank.RecordParser.__init__()=Bio.GenBank.RecordParser-class.html#__init__,Method Bio.GenBank.Scanner.InsdcScanner.__init__()=Bio.GenBank.Scanner.InsdcScanner-class.html#__init__,Method Bio.GenBank._BaseGenBankConsumer.__init__()=Bio.GenBank._BaseGenBankConsumer-class.html#__init__,Method Bio.GenBank._FeatureConsumer.__init__()=Bio.GenBank._FeatureConsumer-class.html#__init__,Method Bio.GenBank._RecordConsumer.__init__()=Bio.GenBank._RecordConsumer-class.html#__init__,Method Bio.GenBank.utils.FeatureValueCleaner.__init__()=Bio.GenBank.utils.FeatureValueCleaner-class.html#__init__,Method Bio.Geo.Record.Record.__init__()=Bio.Geo.Record.Record-class.html#__init__,Method Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment.__init__()=Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome.Chromosome.__init__()=Bio.Graphics.BasicChromosome.Chromosome-class.html#__init__,Method Bio.Graphics.BasicChromosome.ChromosomeSegment.__init__()=Bio.Graphics.BasicChromosome.ChromosomeSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome.Organism.__init__()=Bio.Graphics.BasicChromosome.Organism-class.html#__init__,Method Bio.Graphics.BasicChromosome.TelomereSegment.__init__()=Bio.Graphics.BasicChromosome.TelomereSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome._ChromosomeComponent.__init__()=Bio.Graphics.BasicChromosome._ChromosomeComponent-class.html#__init__,Method Bio.Graphics.ColorSpiral.ColorSpiral.__init__()=Bio.Graphics.ColorSpiral.ColorSpiral-class.html#__init__,Method Bio.Graphics.Comparative.ComparativeScatterPlot.__init__()=Bio.Graphics.Comparative.ComparativeScatterPlot-class.html#__init__,Method Bio.Graphics.DisplayRepresentation.ChromosomeCounts.__init__()=Bio.Graphics.DisplayRepresentation.ChromosomeCounts-class.html#__init__,Method Bio.Graphics.Distribution.BarChartDistribution.__init__()=Bio.Graphics.Distribution.BarChartDistribution-class.html#__init__,Method Bio.Graphics.Distribution.DistributionPage.__init__()=Bio.Graphics.Distribution.DistributionPage-class.html#__init__,Method Bio.Graphics.Distribution.LineDistribution.__init__()=Bio.Graphics.Distribution.LineDistribution-class.html#__init__,Method Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.__init__()=Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.__init__()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.__init__()=Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#__init__,Method Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.__init__()=Bio.Graphics.GenomeDiagram._CrossLink.CrossLink-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.__init__()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Feature.Feature.__init__()=Bio.Graphics.GenomeDiagram._Feature.Feature-class.html#__init__,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__init__()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.__init__()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#__init__,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__init__()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#__init__,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.__init__()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Track.Track.__init__()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#__init__,Method Bio.Graphics.KGML_vis.KGMLCanvas.__init__()=Bio.Graphics.KGML_vis.KGMLCanvas-class.html#__init__,Method Bio.HMM.DynamicProgramming.AbstractDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.AbstractDPAlgorithms-class.html#__init__,Method Bio.HMM.DynamicProgramming.LogDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.LogDPAlgorithms-class.html#__init__,Method Bio.HMM.DynamicProgramming.ScaledDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.ScaledDPAlgorithms-class.html#__init__,Method Bio.HMM.MarkovModel.HiddenMarkovModel.__init__()=Bio.HMM.MarkovModel.HiddenMarkovModel-class.html#__init__,Method Bio.HMM.MarkovModel.MarkovModelBuilder.__init__()=Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#__init__,Method Bio.HMM.Trainer.AbstractTrainer.__init__()=Bio.HMM.Trainer.AbstractTrainer-class.html#__init__,Method Bio.HMM.Trainer.BaumWelchTrainer.__init__()=Bio.HMM.Trainer.BaumWelchTrainer-class.html#__init__,Method Bio.HMM.Trainer.KnownStateTrainer.__init__()=Bio.HMM.Trainer.KnownStateTrainer-class.html#__init__,Method Bio.HMM.Trainer.TrainingSequence.__init__()=Bio.HMM.Trainer.TrainingSequence-class.html#__init__,Method Bio.Index._InMemoryIndex.__init__()=Bio.Index._InMemoryIndex-class.html#__init__,Method Bio.Index._ShelveIndex.__init__()=Bio.Index._ShelveIndex-class.html#__init__,Method Bio.KDTree.KDTree'.KDTree.__init__()=Bio.KDTree.KDTree%27.KDTree-class.html#__init__,Method Bio.KEGG.Compound.Record.__init__()=Bio.KEGG.Compound.Record-class.html#__init__,Method Bio.KEGG.Enzyme.Record.__init__()=Bio.KEGG.Enzyme.Record-class.html#__init__,Method Bio.KEGG.KGML.KGML_parser.KGMLParser.__init__()=Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Component.__init__()=Bio.KEGG.KGML.KGML_pathway.Component-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Entry.__init__()=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Graphics.__init__()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Pathway.__init__()=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Reaction.__init__()=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Relation.__init__()=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#__init__,Method Bio.LogisticRegression.LogisticRegression.__init__()=Bio.LogisticRegression.LogisticRegression-class.html#__init__,Method Bio.MarkovModel.MarkovModel.__init__()=Bio.MarkovModel.MarkovModel-class.html#__init__,Method Bio.MaxEntropy.MaxEntropy.__init__()=Bio.MaxEntropy.MaxEntropy-class.html#__init__,Method Bio.Motif.Parsers.AlignAce.Record.__init__()=Bio.Motif.Parsers.AlignAce.Record-class.html#__init__,Method Bio.Motif.Parsers.MAST.Record.__init__()=Bio.Motif.Parsers.MAST.Record-class.html#__init__,Method Bio.Motif.Parsers.MEME.MEMEInstance.__init__()=Bio.Motif.Parsers.MEME.MEMEInstance-class.html#__init__,Method Bio.Motif.Parsers.MEME.MEMEMotif.__init__()=Bio.Motif.Parsers.MEME.MEMEMotif-class.html#__init__,Method Bio.Motif.Parsers.MEME.MEMERecord.__init__()=Bio.Motif.Parsers.MEME.MEMERecord-class.html#__init__,Method Bio.Motif.Thresholds.ScoreDistribution.__init__()=Bio.Motif.Thresholds.ScoreDistribution-class.html#__init__,Method Bio.Motif._Motif.Motif.__init__()=Bio.Motif._Motif.Motif-class.html#__init__,Method Bio.NMR.xpktools.Peaklist.__init__()=Bio.NMR.xpktools.Peaklist-class.html#__init__,Method Bio.NMR.xpktools.XpkEntry.__init__()=Bio.NMR.xpktools.XpkEntry-class.html#__init__,Method Bio.NaiveBayes.NaiveBayes.__init__()=Bio.NaiveBayes.NaiveBayes-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.AbstractLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.AbstractLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.InputLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Network.BasicNetwork.__init__()=Bio.NeuralNetwork.BackPropagation.Network.BasicNetwork-class.html#__init__,Method Bio.NeuralNetwork.Gene.Motif.MotifCoder.__init__()=Bio.NeuralNetwork.Gene.Motif.MotifCoder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Motif.MotifFinder.__init__()=Bio.NeuralNetwork.Gene.Motif.MotifFinder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.__init__()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#__init__,Method Bio.NeuralNetwork.Gene.Pattern.PatternRepository.__init__()=Bio.NeuralNetwork.Gene.Pattern.PatternRepository-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness.__init__()=Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder.__init__()=Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness.__init__()=Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator.__init__()=Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.Schema.__init__()=Bio.NeuralNetwork.Gene.Schema.Schema-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SchemaCoder.__init__()=Bio.NeuralNetwork.Gene.Schema.SchemaCoder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SchemaFactory.__init__()=Bio.NeuralNetwork.Gene.Schema.SchemaFactory-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SchemaFinder.__init__()=Bio.NeuralNetwork.Gene.Schema.SchemaFinder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SimpleFinisher.__init__()=Bio.NeuralNetwork.Gene.Schema.SimpleFinisher-class.html#__init__,Method Bio.NeuralNetwork.Gene.Signature.SignatureCoder.__init__()=Bio.NeuralNetwork.Gene.Signature.SignatureCoder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Signature.SignatureFinder.__init__()=Bio.NeuralNetwork.Gene.Signature.SignatureFinder-class.html#__init__,Method Bio.NeuralNetwork.StopTraining.ValidationIncreaseStop.__init__()=Bio.NeuralNetwork.StopTraining.ValidationIncreaseStop-class.html#__init__,Method Bio.NeuralNetwork.Training.ExampleManager.__init__()=Bio.NeuralNetwork.Training.ExampleManager-class.html#__init__,Method Bio.NeuralNetwork.Training.TrainingExample.__init__()=Bio.NeuralNetwork.Training.TrainingExample-class.html#__init__,Method Bio.Nexus.Nexus.Block.__init__()=Bio.Nexus.Nexus.Block-class.html#__init__,Method Bio.Nexus.Nexus.CharBuffer.__init__()=Bio.Nexus.Nexus.CharBuffer-class.html#__init__,Method Bio.Nexus.Nexus.Commandline.__init__()=Bio.Nexus.Nexus.Commandline-class.html#__init__,Method Bio.Nexus.Nexus.Nexus.__init__()=Bio.Nexus.Nexus.Nexus-class.html#__init__,Method Bio.Nexus.Nexus.StepMatrix.__init__()=Bio.Nexus.Nexus.StepMatrix-class.html#__init__,Method Bio.Nexus.Nodes.Chain.__init__()=Bio.Nexus.Nodes.Chain-class.html#__init__,Method Bio.Nexus.Nodes.Node.__init__()=Bio.Nexus.Nodes.Node-class.html#__init__,Method Bio.Nexus.Trees.NodeData.__init__()=Bio.Nexus.Trees.NodeData-class.html#__init__,Method Bio.Nexus.Trees.Tree.__init__()=Bio.Nexus.Trees.Tree-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractPropertyMap-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap-class.html#__init__,Method Bio.PDB.Atom.Atom.__init__()=Bio.PDB.Atom.Atom-class.html#__init__,Method Bio.PDB.Atom.DisorderedAtom.__init__()=Bio.PDB.Atom.DisorderedAtom-class.html#__init__,Method Bio.PDB.Chain.Chain.__init__()=Bio.PDB.Chain.Chain-class.html#__init__,Method Bio.PDB.DSSP'.DSSP.__init__()=Bio.PDB.DSSP%27.DSSP-class.html#__init__,Method Bio.PDB.Dice.ChainSelector.__init__()=Bio.PDB.Dice.ChainSelector-class.html#__init__,Method Bio.PDB.Entity.DisorderedEntityWrapper.__init__()=Bio.PDB.Entity.DisorderedEntityWrapper-class.html#__init__,Method Bio.PDB.Entity.Entity.__init__()=Bio.PDB.Entity.Entity-class.html#__init__,Method Bio.PDB.FragmentMapper'.Fragment.__init__()=Bio.PDB.FragmentMapper%27.Fragment-class.html#__init__,Method Bio.PDB.FragmentMapper'.FragmentMapper.__init__()=Bio.PDB.FragmentMapper%27.FragmentMapper-class.html#__init__,Method Bio.PDB.HSExposure.ExposureCN.__init__()=Bio.PDB.HSExposure.ExposureCN-class.html#__init__,Method Bio.PDB.HSExposure.HSExposureCA.__init__()=Bio.PDB.HSExposure.HSExposureCA-class.html#__init__,Method Bio.PDB.HSExposure.HSExposureCB.__init__()=Bio.PDB.HSExposure.HSExposureCB-class.html#__init__,Method Bio.PDB.HSExposure._AbstractHSExposure.__init__()=Bio.PDB.HSExposure._AbstractHSExposure-class.html#__init__,Method Bio.PDB.MMCIF2Dict.MMCIF2Dict.__init__()=Bio.PDB.MMCIF2Dict.MMCIF2Dict-class.html#__init__,Method Bio.PDB.MMCIFParser'.MMCIFParser.__init__()=Bio.PDB.MMCIFParser%27.MMCIFParser-class.html#__init__,Method Bio.PDB.Model.Model.__init__()=Bio.PDB.Model.Model-class.html#__init__,Method Bio.PDB.NACCESS.NACCESS.__init__()=Bio.PDB.NACCESS.NACCESS-class.html#__init__,Method Bio.PDB.NACCESS.NACCESS_atomic.__init__()=Bio.PDB.NACCESS.NACCESS_atomic-class.html#__init__,Method Bio.PDB.NeighborSearch'.NeighborSearch.__init__()=Bio.PDB.NeighborSearch%27.NeighborSearch-class.html#__init__,Method Bio.PDB.PDBIO'.PDBIO.__init__()=Bio.PDB.PDBIO%27.PDBIO-class.html#__init__,Method Bio.PDB.PDBList'.PDBList.__init__()=Bio.PDB.PDBList%27.PDBList-class.html#__init__,Method Bio.PDB.PDBParser'.PDBParser.__init__()=Bio.PDB.PDBParser%27.PDBParser-class.html#__init__,Method Bio.PDB.PSEA.PSEA.__init__()=Bio.PDB.PSEA.PSEA-class.html#__init__,Method Bio.PDB.Polypeptide.CaPPBuilder.__init__()=Bio.PDB.Polypeptide.CaPPBuilder-class.html#__init__,Method Bio.PDB.Polypeptide.PPBuilder.__init__()=Bio.PDB.Polypeptide.PPBuilder-class.html#__init__,Method Bio.PDB.Polypeptide._PPBuilder.__init__()=Bio.PDB.Polypeptide._PPBuilder-class.html#__init__,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.__init__()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#__init__,Method Bio.PDB.Residue.DisorderedResidue.__init__()=Bio.PDB.Residue.DisorderedResidue-class.html#__init__,Method Bio.PDB.Residue.Residue.__init__()=Bio.PDB.Residue.Residue-class.html#__init__,Method Bio.PDB.ResidueDepth'.ResidueDepth.__init__()=Bio.PDB.ResidueDepth%27.ResidueDepth-class.html#__init__,Method Bio.PDB.Structure.Structure.__init__()=Bio.PDB.Structure.Structure-class.html#__init__,Method Bio.PDB.StructureAlignment'.StructureAlignment.__init__()=Bio.PDB.StructureAlignment%27.StructureAlignment-class.html#__init__,Method Bio.PDB.StructureBuilder.StructureBuilder.__init__()=Bio.PDB.StructureBuilder.StructureBuilder-class.html#__init__,Method Bio.PDB.Superimposer'.Superimposer.__init__()=Bio.PDB.Superimposer%27.Superimposer-class.html#__init__,Method Bio.PDB.Vector'.Vector.__init__()=Bio.PDB.Vector%27.Vector-class.html#__init__,Method Bio.ParserSupport.EventGenerator.__init__()=Bio.ParserSupport.EventGenerator-class.html#__init__,Method Bio.ParserSupport.TaggingConsumer.__init__()=Bio.ParserSupport.TaggingConsumer-class.html#__init__,Method Bio.Pathway.Network.__init__()=Bio.Pathway.Network-class.html#__init__,Method Bio.Pathway.Reaction.__init__()=Bio.Pathway.Reaction-class.html#__init__,Method Bio.Pathway.Rep.Graph.Graph.__init__()=Bio.Pathway.Rep.Graph.Graph-class.html#__init__,Method Bio.Pathway.Rep.MultiGraph.MultiGraph.__init__()=Bio.Pathway.Rep.MultiGraph.MultiGraph-class.html#__init__,Method Bio.Pathway.System.__init__()=Bio.Pathway.System-class.html#__init__,Method Bio.Phylo.Applications._Fasttree.FastTreeCommandline.__init__()=Bio.Phylo.Applications._Fasttree.FastTreeCommandline-class.html#__init__,Method Bio.Phylo.Applications._Phyml.PhymlCommandline.__init__()=Bio.Phylo.Applications._Phyml.PhymlCommandline-class.html#__init__,Method Bio.Phylo.Applications._Raxml.RaxmlCommandline.__init__()=Bio.Phylo.Applications._Raxml.RaxmlCommandline-class.html#__init__,Method Bio.Phylo.BaseTree.BranchColor.__init__()=Bio.Phylo.BaseTree.BranchColor-class.html#__init__,Method Bio.Phylo.BaseTree.Clade.__init__()=Bio.Phylo.BaseTree.Clade-class.html#__init__,Method Bio.Phylo.BaseTree.Tree.__init__()=Bio.Phylo.BaseTree.Tree-class.html#__init__,Method Bio.Phylo.CDAO.Clade.__init__()=Bio.Phylo.CDAO.Clade-class.html#__init__,Method Bio.Phylo.CDAO.Tree.__init__()=Bio.Phylo.CDAO.Tree-class.html#__init__,Method Bio.Phylo.CDAOIO.Parser.__init__()=Bio.Phylo.CDAOIO.Parser-class.html#__init__,Method Bio.Phylo.CDAOIO.Writer.__init__()=Bio.Phylo.CDAOIO.Writer-class.html#__init__,Method Bio.Phylo.NeXML.Clade.__init__()=Bio.Phylo.NeXML.Clade-class.html#__init__,Method Bio.Phylo.NeXML.Tree.__init__()=Bio.Phylo.NeXML.Tree-class.html#__init__,Method Bio.Phylo.NeXMLIO.Parser.__init__()=Bio.Phylo.NeXMLIO.Parser-class.html#__init__,Method Bio.Phylo.NeXMLIO.Writer.__init__()=Bio.Phylo.NeXMLIO.Writer-class.html#__init__,Method Bio.Phylo.Newick.Clade.__init__()=Bio.Phylo.Newick.Clade-class.html#__init__,Method Bio.Phylo.Newick.Tree.__init__()=Bio.Phylo.Newick.Tree-class.html#__init__,Method Bio.Phylo.NewickIO.Parser.__init__()=Bio.Phylo.NewickIO.Parser-class.html#__init__,Method Bio.Phylo.NewickIO.Writer.__init__()=Bio.Phylo.NewickIO.Writer-class.html#__init__,Method Bio.Phylo.PAML._paml.Paml.__init__()=Bio.Phylo.PAML._paml.Paml-class.html#__init__,Method Bio.Phylo.PAML.baseml.Baseml.__init__()=Bio.Phylo.PAML.baseml.Baseml-class.html#__init__,Method Bio.Phylo.PAML.codeml.Codeml.__init__()=Bio.Phylo.PAML.codeml.Codeml-class.html#__init__,Method Bio.Phylo.PAML.yn00.Yn00.__init__()=Bio.Phylo.PAML.yn00.Yn00-class.html#__init__,Method Bio.Phylo.PhyloXML.Accession.__init__()=Bio.Phylo.PhyloXML.Accession-class.html#__init__,Method Bio.Phylo.PhyloXML.Annotation.__init__()=Bio.Phylo.PhyloXML.Annotation-class.html#__init__,Method Bio.Phylo.PhyloXML.BinaryCharacters.__init__()=Bio.Phylo.PhyloXML.BinaryCharacters-class.html#__init__,Method Bio.Phylo.PhyloXML.BranchColor.__init__()=Bio.Phylo.PhyloXML.BranchColor-class.html#__init__,Method Bio.Phylo.PhyloXML.Clade.__init__()=Bio.Phylo.PhyloXML.Clade-class.html#__init__,Method Bio.Phylo.PhyloXML.CladeRelation.__init__()=Bio.Phylo.PhyloXML.CladeRelation-class.html#__init__,Method Bio.Phylo.PhyloXML.Confidence.__init__()=Bio.Phylo.PhyloXML.Confidence-class.html#__init__,Method Bio.Phylo.PhyloXML.Date.__init__()=Bio.Phylo.PhyloXML.Date-class.html#__init__,Method Bio.Phylo.PhyloXML.Distribution.__init__()=Bio.Phylo.PhyloXML.Distribution-class.html#__init__,Method Bio.Phylo.PhyloXML.DomainArchitecture.__init__()=Bio.Phylo.PhyloXML.DomainArchitecture-class.html#__init__,Method Bio.Phylo.PhyloXML.Events.__init__()=Bio.Phylo.PhyloXML.Events-class.html#__init__,Method Bio.Phylo.PhyloXML.Id.__init__()=Bio.Phylo.PhyloXML.Id-class.html#__init__,Method Bio.Phylo.PhyloXML.MolSeq.__init__()=Bio.Phylo.PhyloXML.MolSeq-class.html#__init__,Method Bio.Phylo.PhyloXML.Other.__init__()=Bio.Phylo.PhyloXML.Other-class.html#__init__,Method Bio.Phylo.PhyloXML.Phylogeny.__init__()=Bio.Phylo.PhyloXML.Phylogeny-class.html#__init__,Method Bio.Phylo.PhyloXML.Phyloxml.__init__()=Bio.Phylo.PhyloXML.Phyloxml-class.html#__init__,Method Bio.Phylo.PhyloXML.Point.__init__()=Bio.Phylo.PhyloXML.Point-class.html#__init__,Method Bio.Phylo.PhyloXML.Polygon.__init__()=Bio.Phylo.PhyloXML.Polygon-class.html#__init__,Method Bio.Phylo.PhyloXML.Property.__init__()=Bio.Phylo.PhyloXML.Property-class.html#__init__,Method Bio.Phylo.PhyloXML.ProteinDomain.__init__()=Bio.Phylo.PhyloXML.ProteinDomain-class.html#__init__,Method Bio.Phylo.PhyloXML.Reference.__init__()=Bio.Phylo.PhyloXML.Reference-class.html#__init__,Method Bio.Phylo.PhyloXML.Sequence.__init__()=Bio.Phylo.PhyloXML.Sequence-class.html#__init__,Method Bio.Phylo.PhyloXML.SequenceRelation.__init__()=Bio.Phylo.PhyloXML.SequenceRelation-class.html#__init__,Method Bio.Phylo.PhyloXML.Taxonomy.__init__()=Bio.Phylo.PhyloXML.Taxonomy-class.html#__init__,Method Bio.Phylo.PhyloXML.Uri.__init__()=Bio.Phylo.PhyloXML.Uri-class.html#__init__,Method Bio.Phylo.PhyloXMLIO.Parser.__init__()=Bio.Phylo.PhyloXMLIO.Parser-class.html#__init__,Method Bio.Phylo.PhyloXMLIO.Writer.__init__()=Bio.Phylo.PhyloXMLIO.Writer-class.html#__init__,Method Bio.Phylo.TreeConstruction.DistanceCalculator.__init__()=Bio.Phylo.TreeConstruction.DistanceCalculator-class.html#__init__,Method Bio.Phylo.TreeConstruction.DistanceTreeConstructor.__init__()=Bio.Phylo.TreeConstruction.DistanceTreeConstructor-class.html#__init__,Method Bio.Phylo.TreeConstruction.NNITreeSearcher.__init__()=Bio.Phylo.TreeConstruction.NNITreeSearcher-class.html#__init__,Method Bio.Phylo.TreeConstruction.ParsimonyScorer.__init__()=Bio.Phylo.TreeConstruction.ParsimonyScorer-class.html#__init__,Method Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor.__init__()=Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor-class.html#__init__,Method Bio.Phylo.TreeConstruction._DistanceMatrix.__init__()=Bio.Phylo.TreeConstruction._DistanceMatrix-class.html#__init__,Method Bio.Phylo.TreeConstruction._Matrix.__init__()=Bio.Phylo.TreeConstruction._Matrix-class.html#__init__,Method Bio.PopGen.Async.Async.__init__()=Bio.PopGen.Async.Async-class.html#__init__,Method Bio.PopGen.Async.DirectoryRetriever.__init__()=Bio.PopGen.Async.DirectoryRetriever-class.html#__init__,Method Bio.PopGen.Async.FileRetriever.__init__()=Bio.PopGen.Async.FileRetriever-class.html#__init__,Method Bio.PopGen.Async.Local.Local.__init__()=Bio.PopGen.Async.Local.Local-class.html#__init__,Method Bio.PopGen.FDist.Async.FDistAsync.__init__()=Bio.PopGen.FDist.Async.FDistAsync-class.html#__init__,Method Bio.PopGen.FDist.Async.SplitFDist.__init__()=Bio.PopGen.FDist.Async.SplitFDist-class.html#__init__,Method Bio.PopGen.FDist.Controller.FDistController.__init__()=Bio.PopGen.FDist.Controller.FDistController-class.html#__init__,Method Bio.PopGen.FDist.Record.__init__()=Bio.PopGen.FDist.Record-class.html#__init__,Method Bio.PopGen.GenePop.Controller.GenePopController.__init__()=Bio.PopGen.GenePop.Controller.GenePopController-class.html#__init__,Method Bio.PopGen.GenePop.Controller._FileIterator.__init__()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#__init__,Method Bio.PopGen.GenePop.Controller._GenePopCommandline.__init__()=Bio.PopGen.GenePop.Controller._GenePopCommandline-class.html#__init__,Method Bio.PopGen.GenePop.EasyController.EasyController.__init__()=Bio.PopGen.GenePop.EasyController.EasyController-class.html#__init__,Method Bio.PopGen.GenePop.FileParser.FileRecord.__init__()=Bio.PopGen.GenePop.FileParser.FileRecord-class.html#__init__,Method Bio.PopGen.GenePop.LargeFileParser.Record.__init__()=Bio.PopGen.GenePop.LargeFileParser.Record-class.html#__init__,Method Bio.PopGen.GenePop.Record.__init__()=Bio.PopGen.GenePop.Record-class.html#__init__,Method Bio.PopGen.SimCoal.Async.SimCoalCache.__init__()=Bio.PopGen.SimCoal.Async.SimCoalCache-class.html#__init__,Method Bio.PopGen.SimCoal.Cache.SimCoalCache.__init__()=Bio.PopGen.SimCoal.Cache.SimCoalCache-class.html#__init__,Method Bio.PopGen.SimCoal.Controller.FastSimCoalController.__init__()=Bio.PopGen.SimCoal.Controller.FastSimCoalController-class.html#__init__,Method Bio.PopGen.SimCoal.Controller.SimCoalController.__init__()=Bio.PopGen.SimCoal.Controller.SimCoalController-class.html#__init__,Method Bio.PopGen.SimCoal.Controller._FastSimCoalCommandLine.__init__()=Bio.PopGen.SimCoal.Controller._FastSimCoalCommandLine-class.html#__init__,Method Bio.Restriction.PrintFormat.PrintFormat.__init__()=Bio.Restriction.PrintFormat.PrintFormat-class.html#__init__,Method Bio.Restriction.Restriction.Analysis.__init__()=Bio.Restriction.Restriction.Analysis-class.html#__init__,Method Bio.Restriction.Restriction.FormattedSeq.__init__()=Bio.Restriction.Restriction.FormattedSeq-class.html#__init__,Method Bio.Restriction.Restriction.RestrictionBatch.__init__()=Bio.Restriction.Restriction.RestrictionBatch-class.html#__init__,Method Bio.Restriction.Restriction.RestrictionType.__init__()=Bio.Restriction.Restriction.RestrictionType-class.html#__init__,Method Bio.SCOP.Astral.__init__()=Bio.SCOP.Astral-class.html#__init__,Method Bio.SCOP.Cla.Index.__init__()=Bio.SCOP.Cla.Index-class.html#__init__,Method Bio.SCOP.Cla.Record.__init__()=Bio.SCOP.Cla.Record-class.html#__init__,Method Bio.SCOP.Des.Record.__init__()=Bio.SCOP.Des.Record-class.html#__init__,Method Bio.SCOP.Dom.Record.__init__()=Bio.SCOP.Dom.Record-class.html#__init__,Method Bio.SCOP.Domain.__init__()=Bio.SCOP.Domain-class.html#__init__,Method Bio.SCOP.Hie.Record.__init__()=Bio.SCOP.Hie.Record-class.html#__init__,Method Bio.SCOP.Node.__init__()=Bio.SCOP.Node-class.html#__init__,Method Bio.SCOP.Raf.Res.__init__()=Bio.SCOP.Raf.Res-class.html#__init__,Method Bio.SCOP.Raf.SeqMap.__init__()=Bio.SCOP.Raf.SeqMap-class.html#__init__,Method Bio.SCOP.Raf.SeqMapIndex.__init__()=Bio.SCOP.Raf.SeqMapIndex-class.html#__init__,Method Bio.SCOP.Residues.Residues.__init__()=Bio.SCOP.Residues.Residues-class.html#__init__,Method Bio.SCOP.Scop.__init__()=Bio.SCOP.Scop-class.html#__init__,Method Bio.SVDSuperimposer.SVDSuperimposer.__init__()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabParser.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_text.BlastTextParser.__init__()=Bio.SearchIO.BlastIO.blast_text.BlastTextParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator.__init__()=Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslIndexer.__init__()=Bio.SearchIO.BlatIO.BlatPslIndexer-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslParser.__init__()=Bio.SearchIO.BlatIO.BlatPslParser-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslWriter.__init__()=Bio.SearchIO.BlatIO.BlatPslWriter-class.html#__init__,Method Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.__init__()=Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html#__init__,Method Bio.SearchIO.FastaIO.FastaM10Indexer.__init__()=Bio.SearchIO.FastaIO.FastaM10Indexer-class.html#__init__,Method Bio.SearchIO.FastaIO.FastaM10Parser.__init__()=Bio.SearchIO.FastaIO.FastaM10Parser-class.html#__init__,Method Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer.__init__()=Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser.__init__()=Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.__init__()=Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser.__init__()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.__init__()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser.__init__()=Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#__init__,Method Bio.SearchIO._index.SearchIndexer.__init__()=Bio.SearchIO._index.SearchIndexer-class.html#__init__,Method Bio.SearchIO._model.hit.Hit.__init__()=Bio.SearchIO._model.hit.Hit-class.html#__init__,Method Bio.SearchIO._model.hsp.HSP.__init__()=Bio.SearchIO._model.hsp.HSP-class.html#__init__,Method Bio.SearchIO._model.hsp.HSPFragment.__init__()=Bio.SearchIO._model.hsp.HSPFragment-class.html#__init__,Method Bio.SearchIO._model.query.QueryResult.__init__()=Bio.SearchIO._model.query.QueryResult-class.html#__init__,Method Bio.Seq.MutableSeq.__init__()=Bio.Seq.MutableSeq-class.html#__init__,Method Bio.Seq.Seq.__init__()=Bio.Seq.Seq-class.html#__init__,Method Bio.Seq.UnknownSeq.__init__()=Bio.Seq.UnknownSeq-class.html#__init__,Method Bio.SeqFeature.CompoundLocation.__init__()=Bio.SeqFeature.CompoundLocation-class.html#__init__,Method Bio.SeqFeature.FeatureLocation.__init__()=Bio.SeqFeature.FeatureLocation-class.html#__init__,Method Bio.SeqFeature.PositionGap.__init__()=Bio.SeqFeature.PositionGap-class.html#__init__,Method Bio.SeqFeature.Reference.__init__()=Bio.SeqFeature.Reference-class.html#__init__,Method Bio.SeqFeature.SeqFeature.__init__()=Bio.SeqFeature.SeqFeature-class.html#__init__,Method Bio.SeqIO.FastaIO.FastaWriter.__init__()=Bio.SeqIO.FastaIO.FastaWriter-class.html#__init__,Method Bio.SeqIO.Interfaces.SequenceIterator.__init__()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#__init__,Method Bio.SeqIO.Interfaces.SequenceWriter.__init__()=Bio.SeqIO.Interfaces.SequenceWriter-class.html#__init__,Method Bio.SeqIO.Interfaces.SequentialSequenceWriter.__init__()=Bio.SeqIO.Interfaces.SequentialSequenceWriter-class.html#__init__,Method Bio.SeqIO.PhdIO.PhdWriter.__init__()=Bio.SeqIO.PhdIO.PhdWriter-class.html#__init__,Method Bio.SeqIO.QualityIO.QualPhredWriter.__init__()=Bio.SeqIO.QualityIO.QualPhredWriter-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.SeqXmlIterator.__init__()=Bio.SeqIO.SeqXmlIO.SeqXmlIterator-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.SeqXmlWriter.__init__()=Bio.SeqIO.SeqXmlIO.SeqXmlWriter-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.XMLRecordIterator.__init__()=Bio.SeqIO.SeqXmlIO.XMLRecordIterator-class.html#__init__,Method Bio.SeqIO.SffIO.SffWriter.__init__()=Bio.SeqIO.SffIO.SffWriter-class.html#__init__,Method Bio.SeqIO.SffIO._AddTellHandle.__init__()=Bio.SeqIO.SffIO._AddTellHandle-class.html#__init__,Method Bio.SeqIO.UniprotIO.Parser.__init__()=Bio.SeqIO.UniprotIO.Parser-class.html#__init__,Method Bio.SeqIO._index.IntelliGeneticsRandomAccess.__init__()=Bio.SeqIO._index.IntelliGeneticsRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SeqFileRandomAccess.__init__()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SequentialSeqFileRandomAccess.__init__()=Bio.SeqIO._index.SequentialSeqFileRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SffRandomAccess.__init__()=Bio.SeqIO._index.SffRandomAccess-class.html#__init__,Method Bio.SeqRecord.SeqRecord.__init__()=Bio.SeqRecord.SeqRecord-class.html#__init__,Method Bio.SeqRecord._RestrictedDict.__init__()=Bio.SeqRecord._RestrictedDict-class.html#__init__,Method Bio.SeqUtils.CodonUsage.CodonAdaptationIndex.__init__()=Bio.SeqUtils.CodonUsage.CodonAdaptationIndex-class.html#__init__,Method Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.__init__()=Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint-class.html#__init__,Method Bio.SeqUtils.ProtParam.ProteinAnalysis.__init__()=Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#__init__,Method Bio.Sequencing.Ace.ACEFileRecord.__init__()=Bio.Sequencing.Ace.ACEFileRecord-class.html#__init__,Method Bio.Sequencing.Ace.Contig.__init__()=Bio.Sequencing.Ace.Contig-class.html#__init__,Method Bio.Sequencing.Ace.Reads.__init__()=Bio.Sequencing.Ace.Reads-class.html#__init__,Method Bio.Sequencing.Ace.af.__init__()=Bio.Sequencing.Ace.af-class.html#__init__,Method Bio.Sequencing.Ace.bs.__init__()=Bio.Sequencing.Ace.bs-class.html#__init__,Method Bio.Sequencing.Ace.ct.__init__()=Bio.Sequencing.Ace.ct-class.html#__init__,Method Bio.Sequencing.Ace.ds.__init__()=Bio.Sequencing.Ace.ds-class.html#__init__,Method Bio.Sequencing.Ace.qa.__init__()=Bio.Sequencing.Ace.qa-class.html#__init__,Method Bio.Sequencing.Ace.rd.__init__()=Bio.Sequencing.Ace.rd-class.html#__init__,Method Bio.Sequencing.Ace.rt.__init__()=Bio.Sequencing.Ace.rt-class.html#__init__,Method Bio.Sequencing.Ace.wa.__init__()=Bio.Sequencing.Ace.wa-class.html#__init__,Method Bio.Sequencing.Ace.wr.__init__()=Bio.Sequencing.Ace.wr-class.html#__init__,Method Bio.Sequencing.Applications._Novoalign.NovoalignCommandline.__init__()=Bio.Sequencing.Applications._Novoalign.NovoalignCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaAlignCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaAlignCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaBwaswCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaBwaswCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaIndexCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaIndexCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaSampeCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaSampeCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaSamseCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaSamseCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsSortCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsSortCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline-class.html#__init__,Method Bio.Sequencing.Phd.Record.__init__()=Bio.Sequencing.Phd.Record-class.html#__init__,Method Bio.SubsMat.FreqTable.FreqTable.__init__()=Bio.SubsMat.FreqTable.FreqTable-class.html#__init__,Method Bio.SubsMat.SeqMat.__init__()=Bio.SubsMat.SeqMat-class.html#__init__,Method Bio.SwissProt.KeyWList.Record.__init__()=Bio.SwissProt.KeyWList.Record-class.html#__init__,Method Bio.SwissProt.Record.__init__()=Bio.SwissProt.Record-class.html#__init__,Method Bio.SwissProt.Reference.__init__()=Bio.SwissProt.Reference-class.html#__init__,Method Bio.UniGene.ProtsimLine.__init__()=Bio.UniGene.ProtsimLine-class.html#__init__,Method Bio.UniGene.Record.__init__()=Bio.UniGene.Record-class.html#__init__,Method Bio.UniGene.STSLine.__init__()=Bio.UniGene.STSLine-class.html#__init__,Method Bio.UniGene.SequenceLine.__init__()=Bio.UniGene.SequenceLine-class.html#__init__,Method Bio.Wise.dnal.Statistics.__init__()=Bio.Wise.dnal.Statistics-class.html#__init__,Method Bio.Wise.psw.AlignmentColumn.__init__()=Bio.Wise.psw.AlignmentColumn-class.html#__init__,Method Bio.Wise.psw.ColumnUnit.__init__()=Bio.Wise.psw.ColumnUnit-class.html#__init__,Method Bio._py3k._ordereddict.OrderedDict.__init__()=Bio._py3k._ordereddict.OrderedDict-class.html#__init__,Method Bio.bgzf.BgzfReader.__init__()=Bio.bgzf.BgzfReader-class.html#__init__,Method Bio.bgzf.BgzfWriter.__init__()=Bio.bgzf.BgzfWriter-class.html#__init__,Method Bio.codonalign.codonalignment.CodonAlignment.__init__()=Bio.codonalign.codonalignment.CodonAlignment-class.html#__init__,Method Bio.codonalign.codonseq.CodonSeq.__init__()=Bio.codonalign.codonseq.CodonSeq-class.html#__init__,Method Bio.kNN.kNN.__init__()=Bio.kNN.kNN-class.html#__init__,Method Bio.motifs.Instances.__init__()=Bio.motifs.Instances-class.html#__init__,Method Bio.motifs.Motif.__init__()=Bio.motifs.Motif-class.html#__init__,Method Bio.motifs.alignace.Record.__init__()=Bio.motifs.alignace.Record-class.html#__init__,Method Bio.motifs.applications._alignace.AlignAceCommandline.__init__()=Bio.motifs.applications._alignace.AlignAceCommandline-class.html#__init__,Method Bio.motifs.applications._alignace.CompareAceCommandline.__init__()=Bio.motifs.applications._alignace.CompareAceCommandline-class.html#__init__,Method Bio.motifs.applications._xxmotif.XXmotifCommandline.__init__()=Bio.motifs.applications._xxmotif.XXmotifCommandline-class.html#__init__,Method Bio.motifs.jaspar.Motif.__init__()=Bio.motifs.jaspar.Motif-class.html#__init__,Method Bio.motifs.jaspar.Record.__init__()=Bio.motifs.jaspar.Record-class.html#__init__,Method Bio.motifs.jaspar.db.JASPAR5.__init__()=Bio.motifs.jaspar.db.JASPAR5-class.html#__init__,Method Bio.motifs.mast.Record.__init__()=Bio.motifs.mast.Record-class.html#__init__,Method Bio.motifs.matrix.GenericPositionMatrix.__init__()=Bio.motifs.matrix.GenericPositionMatrix-class.html#__init__,Method Bio.motifs.matrix.PositionWeightMatrix.__init__()=Bio.motifs.matrix.PositionWeightMatrix-class.html#__init__,Method Bio.motifs.meme.Instance.__init__()=Bio.motifs.meme.Instance-class.html#__init__,Method Bio.motifs.meme.Motif.__init__()=Bio.motifs.meme.Motif-class.html#__init__,Method Bio.motifs.meme.Record.__init__()=Bio.motifs.meme.Record-class.html#__init__,Method Bio.motifs.thresholds.ScoreDistribution.__init__()=Bio.motifs.thresholds.ScoreDistribution-class.html#__init__,Method Bio.motifs.transfac.Record.__init__()=Bio.motifs.transfac.Record-class.html#__init__,Method Bio.pairwise2.affine_penalty.__init__()=Bio.pairwise2.affine_penalty-class.html#__init__,Method Bio.pairwise2.dictionary_match.__init__()=Bio.pairwise2.dictionary_match-class.html#__init__,Method Bio.pairwise2.identity_match.__init__()=Bio.pairwise2.identity_match-class.html#__init__,Method BioSQL.BioSeq.DBSeq.__init__()=BioSQL.BioSeq.DBSeq-class.html#__init__,Method BioSQL.BioSeq.DBSeqRecord.__init__()=BioSQL.BioSeq.DBSeqRecord-class.html#__init__,Method BioSQL.BioSeqDatabase.Adaptor.__init__()=BioSQL.BioSeqDatabase.Adaptor-class.html#__init__,Method BioSQL.BioSeqDatabase.BioSeqDatabase.__init__()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#__init__,Method BioSQL.BioSeqDatabase.DBServer.__init__()=BioSQL.BioSeqDatabase.DBServer-class.html#__init__,Method BioSQL.BioSeqDatabase._CursorWrapper.__init__()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#__init__,Method BioSQL.DBUtils.Generic_dbutils.__init__()=BioSQL.DBUtils.Generic_dbutils-class.html#__init__,Method BioSQL.Loader.DatabaseLoader.__init__()=BioSQL.Loader.DatabaseLoader-class.html#__init__,Method BioSQL.Loader.DatabaseRemover.__init__()=BioSQL.Loader.DatabaseRemover-class.html#__init__"><a title="Bio.Affy.CelFile.Record.__init__
Bio.Align.AlignInfo.PSSM.__init__
Bio.Align.AlignInfo.SummaryInfo.__init__
Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline.__init__
Bio.Align.Applications._Clustalw.ClustalwCommandline.__init__
Bio.Align.Applications._Dialign.DialignCommandline.__init__
Bio.Align.Applications._MSAProbs.MSAProbsCommandline.__init__
Bio.Align.Applications._Mafft.MafftCommandline.__init__
Bio.Align.Applications._Muscle.MuscleCommandline.__init__
Bio.Align.Applications._Prank.PrankCommandline.__init__
Bio.Align.Applications._Probcons.ProbconsCommandline.__init__
Bio.Align.Applications._TCoffee.TCoffeeCommandline.__init__
Bio.Align.Generic.Alignment.__init__
Bio.Align.MultipleSeqAlignment.__init__
Bio.AlignIO.Interfaces.AlignmentIterator.__init__
Bio.AlignIO.Interfaces.AlignmentWriter.__init__
Bio.AlignIO.Interfaces.SequentialAlignmentWriter.__init__
Bio.Alphabet.AlphabetEncoder.__init__
Bio.Alphabet.Gapped.__init__
Bio.Alphabet.HasStopCodon.__init__
Bio.Application.AbstractCommandline.__init__
Bio.Application.ApplicationError.__init__
Bio.Application._AbstractParameter.__init__
Bio.Application._Argument.__init__
Bio.Application._Option.__init__
Bio.Application._StaticArgument.__init__
Bio.Application._Switch.__init__
Bio.Blast.Applications.NcbiblastformatterCommandline.__init__
Bio.Blast.Applications.NcbiblastnCommandline.__init__
Bio.Blast.Applications.NcbiblastpCommandline.__init__
Bio.Blast.Applications.NcbiblastxCommandline.__init__
Bio.Blast.Applications.NcbideltablastCommandline.__init__
Bio.Blast.Applications.NcbipsiblastCommandline.__init__
Bio.Blast.Applications.NcbirpsblastCommandline.__init__
Bio.Blast.Applications.NcbirpstblastnCommandline.__init__
Bio.Blast.Applications.NcbitblastnCommandline.__init__
Bio.Blast.Applications.NcbitblastxCommandline.__init__
Bio.Blast.Applications._NcbibaseblastCommandline.__init__
Bio.Blast.Applications._Ncbiblast2SeqCommandline.__init__
Bio.Blast.Applications._NcbiblastCommandline.__init__
Bio.Blast.Applications._NcbiblastMain2SeqCommandline.__init__
Bio.Blast.NCBIStandalone.BlastErrorParser.__init__
Bio.Blast.NCBIStandalone.BlastParser.__init__
Bio.Blast.NCBIStandalone.Iterator.__init__
Bio.Blast.NCBIStandalone.PSIBlastParser.__init__
Bio.Blast.NCBIStandalone._BlastConsumer.__init__
Bio.Blast.NCBIStandalone._BlastErrorConsumer.__init__
Bio.Blast.NCBIStandalone._PSIBlastConsumer.__init__
Bio.Blast.NCBIXML.BlastParser.__init__
Bio.Blast.NCBIXML._XMLparser.__init__
Bio.Blast.ParseBlastTable.BlastTableEntry.__init__
Bio.Blast.ParseBlastTable.BlastTableReader.__init__
Bio.Blast.ParseBlastTable.BlastTableRec.__init__
Bio.Blast.Record.Alignment.__init__
Bio.Blast.Record.Blast.__init__
Bio.Blast.Record.DatabaseReport.__init__
Bio.Blast.Record.Description.__init__
Bio.Blast.Record.HSP.__init__
Bio.Blast.Record.Header.__init__
Bio.Blast.Record.MultipleAlignment.__init__
Bio.Blast.Record.PSIBlast.__init__
Bio.Blast.Record.Parameters.__init__
Bio.Blast.Record.Round.__init__
Bio.CAPS.CAPSMap.__init__
Bio.CAPS.DifferentialCutsite.__init__
Bio.Cluster.Record.__init__
Bio.Compass.Record.__init__
Bio.Crystal.Chain.__init__
Bio.Crystal.Crystal.__init__
Bio.Crystal.Hetero.__init__
Bio.Data.CodonTable.AmbiguousCodonTable.__init__
Bio.Data.CodonTable.AmbiguousForwardTable.__init__
Bio.Data.CodonTable.CodonTable.__init__
Bio.Data.CodonTable.NCBICodonTable.__init__
Bio.DocSQL.Create.__init__
Bio.DocSQL.Insert.__init__
Bio.DocSQL.IterationCursor.__init__
Bio.DocSQL.Query.__init__
Bio.DocSQL.QueryAll.__init__
Bio.DocSQL.QueryGeneric.__init__
Bio.DocSQL.QueryRow.__init__
Bio.DocSQL.QuerySingle.__init__
Bio.Emboss.Applications.DiffseqCommandline.__init__
Bio.Emboss.Applications.EInvertedCommandline.__init__
Bio.Emboss.Applications.ETandemCommandline.__init__
Bio.Emboss.Applications.Est2GenomeCommandline.__init__
Bio.Emboss.Applications.FConsenseCommandline.__init__
Bio.Emboss.Applications.FDNADistCommandline.__init__
Bio.Emboss.Applications.FDNAParsCommandline.__init__
Bio.Emboss.Applications.FNeighborCommandline.__init__
Bio.Emboss.Applications.FProtDistCommandline.__init__
Bio.Emboss.Applications.FProtParsCommandline.__init__
Bio.Emboss.Applications.FSeqBootCommandline.__init__
Bio.Emboss.Applications.FTreeDistCommandline.__init__
Bio.Emboss.Applications.FuzznucCommandline.__init__
Bio.Emboss.Applications.IepCommandline.__init__
Bio.Emboss.Applications.NeedleCommandline.__init__
Bio.Emboss.Applications.NeedleallCommandline.__init__
Bio.Emboss.Applications.PalindromeCommandline.__init__
Bio.Emboss.Applications.Primer3Commandline.__init__
Bio.Emboss.Applications.PrimerSearchCommandline.__init__
Bio.Emboss.Applications.SeqmatchallCommandline.__init__
Bio.Emboss.Applications.SeqretCommandline.__init__
Bio.Emboss.Applications.StretcherCommandline.__init__
Bio.Emboss.Applications.TranalignCommandline.__init__
Bio.Emboss.Applications.WaterCommandline.__init__
Bio.Emboss.Applications._EmbossCommandLine.__init__
Bio.Emboss.Applications._EmbossMinimalCommandLine.__init__
Bio.Emboss.Primer3.Primers.__init__
Bio.Emboss.Primer3.Record.__init__
Bio.Emboss.PrimerSearch.Amplifier.__init__
Bio.Emboss.PrimerSearch.InputRecord.__init__
Bio.Emboss.PrimerSearch.OutputRecord.__init__
Bio.Entrez.Parser.CorruptedXMLError.__init__
Bio.Entrez.Parser.DataHandler.__init__
Bio.Entrez.Parser.NotXMLError.__init__
Bio.Entrez.Parser.StructureElement.__init__
Bio.Entrez.Parser.ValidationError.__init__
Bio.ExPASy.Enzyme.Record.__init__
Bio.ExPASy.Prodoc.Record.__init__
Bio.ExPASy.Prodoc.Reference.__init__
Bio.ExPASy.Prosite.Record.__init__
Bio.ExPASy.ScanProsite.ContentHandler.__init__
Bio.ExPASy.ScanProsite.Parser.__init__
Bio.ExPASy.ScanProsite.Record.__init__
Bio.FSSP.FSSPAlignDict.__init__
Bio.FSSP.FSSPAlignRec.__init__
Bio.FSSP.FSSPHeader.__init__
Bio.FSSP.FSSPSumRec.__init__
Bio.FSSP.FSSPTools.FSSPMultAlign.__init__
Bio.FSSP.PosAlign.__init__
Bio.FSSP.fssp_rec.fff_rec.__init__
Bio.File.UndoHandle.__init__
Bio.File._IndexedSeqFileDict.__init__
Bio.File._SQLiteManySeqFilesDict.__init__
Bio.GA.Crossover.General.SafeFitnessCrossover.__init__
Bio.GA.Crossover.GeneralPoint.GeneralPointCrossover.__init__
Bio.GA.Crossover.GeneralPoint.InterleaveCrossover.__init__
Bio.GA.Crossover.Point.SinglePointCrossover.__init__
Bio.GA.Crossover.TwoPoint.TwoPointCrossover.__init__
Bio.GA.Crossover.Uniform.UniformCrossover.__init__
Bio.GA.Evolver.GenerationEvolver.__init__
Bio.GA.Evolver.SteadyStateEvolver.__init__
Bio.GA.Mutation.General.SafeFitnessMutation.__init__
Bio.GA.Mutation.Simple.ConversionMutation.__init__
Bio.GA.Mutation.Simple.SinglePositionMutation.__init__
Bio.GA.Organism.Organism.__init__
Bio.GA.Repair.Stabilizing.AmbiguousRepair.__init__
Bio.GA.Selection.Abstract.AbstractSelection.__init__
Bio.GA.Selection.Diversity.DiversitySelection.__init__
Bio.GA.Selection.RouletteWheel.RouletteWheelSelection.__init__
Bio.GA.Selection.Tournament.TournamentSelection.__init__
Bio.GenBank.FeatureParser.__init__
Bio.GenBank.Iterator.__init__
Bio.GenBank.Record.Feature.__init__
Bio.GenBank.Record.Qualifier.__init__
Bio.GenBank.Record.Record.__init__
Bio.GenBank.Record.Reference.__init__
Bio.GenBank.RecordParser.__init__
Bio.GenBank.Scanner.InsdcScanner.__init__
Bio.GenBank._BaseGenBankConsumer.__init__
Bio.GenBank._FeatureConsumer.__init__
Bio.GenBank._RecordConsumer.__init__
Bio.GenBank.utils.FeatureValueCleaner.__init__
Bio.Geo.Record.Record.__init__
Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment.__init__
Bio.Graphics.BasicChromosome.Chromosome.__init__
Bio.Graphics.BasicChromosome.ChromosomeSegment.__init__
Bio.Graphics.BasicChromosome.Organism.__init__
Bio.Graphics.BasicChromosome.TelomereSegment.__init__
Bio.Graphics.BasicChromosome._ChromosomeComponent.__init__
Bio.Graphics.ColorSpiral.ColorSpiral.__init__
Bio.Graphics.Comparative.ComparativeScatterPlot.__init__
Bio.Graphics.DisplayRepresentation.ChromosomeCounts.__init__
Bio.Graphics.Distribution.BarChartDistribution.__init__
Bio.Graphics.Distribution.DistributionPage.__init__
Bio.Graphics.Distribution.LineDistribution.__init__
Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.__init__
Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.__init__
Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.__init__
Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.__init__
Bio.Graphics.GenomeDiagram._Diagram.Diagram.__init__
Bio.Graphics.GenomeDiagram._Feature.Feature.__init__
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__init__
Bio.Graphics.GenomeDiagram._Graph.GraphData.__init__
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__init__
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.__init__
Bio.Graphics.GenomeDiagram._Track.Track.__init__
Bio.Graphics.KGML_vis.KGMLCanvas.__init__
Bio.HMM.DynamicProgramming.AbstractDPAlgorithms.__init__
Bio.HMM.DynamicProgramming.LogDPAlgorithms.__init__
Bio.HMM.DynamicProgramming.ScaledDPAlgorithms.__init__
Bio.HMM.MarkovModel.HiddenMarkovModel.__init__
Bio.HMM.MarkovModel.MarkovModelBuilder.__init__
Bio.HMM.Trainer.AbstractTrainer.__init__
Bio.HMM.Trainer.BaumWelchTrainer.__init__
Bio.HMM.Trainer.KnownStateTrainer.__init__
Bio.HMM.Trainer.TrainingSequence.__init__
Bio.Index._InMemoryIndex.__init__
Bio.Index._ShelveIndex.__init__
Bio.KDTree.KDTree'.KDTree.__init__
Bio.KEGG.Compound.Record.__init__
Bio.KEGG.Enzyme.Record.__init__
Bio.KEGG.KGML.KGML_parser.KGMLParser.__init__
Bio.KEGG.KGML.KGML_pathway.Component.__init__
Bio.KEGG.KGML.KGML_pathway.Entry.__init__
Bio.KEGG.KGML.KGML_pathway.Graphics.__init__
Bio.KEGG.KGML.KGML_pathway.Pathway.__init__
Bio.KEGG.KGML.KGML_pathway.Reaction.__init__
Bio.KEGG.KGML.KGML_pathway.Relation.__init__
Bio.LogisticRegression.LogisticRegression.__init__
Bio.MarkovModel.MarkovModel.__init__
Bio.MaxEntropy.MaxEntropy.__init__
Bio.Motif.Parsers.AlignAce.Record.__init__
Bio.Motif.Parsers.MAST.Record.__init__
Bio.Motif.Parsers.MEME.MEMEInstance.__init__
Bio.Motif.Parsers.MEME.MEMEMotif.__init__
Bio.Motif.Parsers.MEME.MEMERecord.__init__
Bio.Motif.Thresholds.ScoreDistribution.__init__
Bio.Motif._Motif.Motif.__init__
Bio.NMR.xpktools.Peaklist.__init__
Bio.NMR.xpktools.XpkEntry.__init__
Bio.NaiveBayes.NaiveBayes.__init__
Bio.NeuralNetwork.BackPropagation.Layer.AbstractLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.__init__
Bio.NeuralNetwork.BackPropagation.Network.BasicNetwork.__init__
Bio.NeuralNetwork.Gene.Motif.MotifCoder.__init__
Bio.NeuralNetwork.Gene.Motif.MotifFinder.__init__
Bio.NeuralNetwork.Gene.Pattern.PatternIO.__init__
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.__init__
Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness.__init__
Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder.__init__
Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness.__init__
Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator.__init__
Bio.NeuralNetwork.Gene.Schema.Schema.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaCoder.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaFactory.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaFinder.__init__
Bio.NeuralNetwork.Gene.Schema.SimpleFinisher.__init__
Bio.NeuralNetwork.Gene.Signature.SignatureCoder.__init__
Bio.NeuralNetwork.Gene.Signature.SignatureFinder.__init__
Bio.NeuralNetwork.StopTraining.ValidationIncreaseStop.__init__
Bio.NeuralNetwork.Training.ExampleManager.__init__
Bio.NeuralNetwork.Training.TrainingExample.__init__
Bio.Nexus.Nexus.Block.__init__
Bio.Nexus.Nexus.CharBuffer.__init__
Bio.Nexus.Nexus.Commandline.__init__
Bio.Nexus.Nexus.Nexus.__init__
Bio.Nexus.Nexus.StepMatrix.__init__
Bio.Nexus.Nodes.Chain.__init__
Bio.Nexus.Nodes.Node.__init__
Bio.Nexus.Trees.NodeData.__init__
Bio.Nexus.Trees.Tree.__init__
Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap.__init__
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__init__
Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap.__init__
Bio.PDB.Atom.Atom.__init__
Bio.PDB.Atom.DisorderedAtom.__init__
Bio.PDB.Chain.Chain.__init__
Bio.PDB.DSSP'.DSSP.__init__
Bio.PDB.Dice.ChainSelector.__init__
Bio.PDB.Entity.DisorderedEntityWrapper.__init__
Bio.PDB.Entity.Entity.__init__
Bio.PDB.FragmentMapper'.Fragment.__init__
Bio.PDB.FragmentMapper'.FragmentMapper.__init__
Bio.PDB.HSExposure.ExposureCN.__init__
Bio.PDB.HSExposure.HSExposureCA.__init__
Bio.PDB.HSExposure.HSExposureCB.__init__
Bio.PDB.HSExposure._AbstractHSExposure.__init__
Bio.PDB.MMCIF2Dict.MMCIF2Dict.__init__
Bio.PDB.MMCIFParser'.MMCIFParser.__init__
Bio.PDB.Model.Model.__init__
Bio.PDB.NACCESS.NACCESS.__init__
Bio.PDB.NACCESS.NACCESS_atomic.__init__
Bio.PDB.NeighborSearch'.NeighborSearch.__init__
Bio.PDB.PDBIO'.PDBIO.__init__
Bio.PDB.PDBList'.PDBList.__init__
Bio.PDB.PDBParser'.PDBParser.__init__
Bio.PDB.PSEA.PSEA.__init__
Bio.PDB.Polypeptide.CaPPBuilder.__init__
Bio.PDB.Polypeptide.PPBuilder.__init__
Bio.PDB.Polypeptide._PPBuilder.__init__
Bio.PDB.QCPSuperimposer.QCPSuperimposer.__init__
Bio.PDB.Residue.DisorderedResidue.__init__
Bio.PDB.Residue.Residue.__init__
Bio.PDB.ResidueDepth'.ResidueDepth.__init__
Bio.PDB.Structure.Structure.__init__
Bio.PDB.StructureAlignment'.StructureAlignment.__init__
Bio.PDB.StructureBuilder.StructureBuilder.__init__
Bio.PDB.Superimposer'.Superimposer.__init__
Bio.PDB.Vector'.Vector.__init__
Bio.ParserSupport.EventGenerator.__init__
Bio.ParserSupport.TaggingConsumer.__init__
Bio.Pathway.Network.__init__
Bio.Pathway.Reaction.__init__
Bio.Pathway.Rep.Graph.Graph.__init__
Bio.Pathway.Rep.MultiGraph.MultiGraph.__init__
Bio.Pathway.System.__init__
Bio.Phylo.Applications._Fasttree.FastTreeCommandline.__init__
Bio.Phylo.Applications._Phyml.PhymlCommandline.__init__
Bio.Phylo.Applications._Raxml.RaxmlCommandline.__init__
Bio.Phylo.BaseTree.BranchColor.__init__
Bio.Phylo.BaseTree.Clade.__init__
Bio.Phylo.BaseTree.Tree.__init__
Bio.Phylo.CDAO.Clade.__init__
Bio.Phylo.CDAO.Tree.__init__
Bio.Phylo.CDAOIO.Parser.__init__
Bio.Phylo.CDAOIO.Writer.__init__
Bio.Phylo.NeXML.Clade.__init__
Bio.Phylo.NeXML.Tree.__init__
Bio.Phylo.NeXMLIO.Parser.__init__
Bio.Phylo.NeXMLIO.Writer.__init__
Bio.Phylo.Newick.Clade.__init__
Bio.Phylo.Newick.Tree.__init__
Bio.Phylo.NewickIO.Parser.__init__
Bio.Phylo.NewickIO.Writer.__init__
Bio.Phylo.PAML._paml.Paml.__init__
Bio.Phylo.PAML.baseml.Baseml.__init__
Bio.Phylo.PAML.codeml.Codeml.__init__
Bio.Phylo.PAML.yn00.Yn00.__init__
Bio.Phylo.PhyloXML.Accession.__init__
Bio.Phylo.PhyloXML.Annotation.__init__
Bio.Phylo.PhyloXML.BinaryCharacters.__init__
Bio.Phylo.PhyloXML.BranchColor.__init__
Bio.Phylo.PhyloXML.Clade.__init__
Bio.Phylo.PhyloXML.CladeRelation.__init__
Bio.Phylo.PhyloXML.Confidence.__init__
Bio.Phylo.PhyloXML.Date.__init__
Bio.Phylo.PhyloXML.Distribution.__init__
Bio.Phylo.PhyloXML.DomainArchitecture.__init__
Bio.Phylo.PhyloXML.Events.__init__
Bio.Phylo.PhyloXML.Id.__init__
Bio.Phylo.PhyloXML.MolSeq.__init__
Bio.Phylo.PhyloXML.Other.__init__
Bio.Phylo.PhyloXML.Phylogeny.__init__
Bio.Phylo.PhyloXML.Phyloxml.__init__
Bio.Phylo.PhyloXML.Point.__init__
Bio.Phylo.PhyloXML.Polygon.__init__
Bio.Phylo.PhyloXML.Property.__init__
Bio.Phylo.PhyloXML.ProteinDomain.__init__
Bio.Phylo.PhyloXML.Reference.__init__
Bio.Phylo.PhyloXML.Sequence.__init__
Bio.Phylo.PhyloXML.SequenceRelation.__init__
Bio.Phylo.PhyloXML.Taxonomy.__init__
Bio.Phylo.PhyloXML.Uri.__init__
Bio.Phylo.PhyloXMLIO.Parser.__init__
Bio.Phylo.PhyloXMLIO.Writer.__init__
Bio.Phylo.TreeConstruction.DistanceCalculator.__init__
Bio.Phylo.TreeConstruction.DistanceTreeConstructor.__init__
Bio.Phylo.TreeConstruction.NNITreeSearcher.__init__
Bio.Phylo.TreeConstruction.ParsimonyScorer.__init__
Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor.__init__
Bio.Phylo.TreeConstruction._DistanceMatrix.__init__
Bio.Phylo.TreeConstruction._Matrix.__init__
Bio.PopGen.Async.Async.__init__
Bio.PopGen.Async.DirectoryRetriever.__init__
Bio.PopGen.Async.FileRetriever.__init__
Bio.PopGen.Async.Local.Local.__init__
Bio.PopGen.FDist.Async.FDistAsync.__init__
Bio.PopGen.FDist.Async.SplitFDist.__init__
Bio.PopGen.FDist.Controller.FDistController.__init__
Bio.PopGen.FDist.Record.__init__
Bio.PopGen.GenePop.Controller.GenePopController.__init__
Bio.PopGen.GenePop.Controller._FileIterator.__init__
Bio.PopGen.GenePop.Controller._GenePopCommandline.__init__
Bio.PopGen.GenePop.EasyController.EasyController.__init__
Bio.PopGen.GenePop.FileParser.FileRecord.__init__
Bio.PopGen.GenePop.LargeFileParser.Record.__init__
Bio.PopGen.GenePop.Record.__init__
Bio.PopGen.SimCoal.Async.SimCoalCache.__init__
Bio.PopGen.SimCoal.Cache.SimCoalCache.__init__
Bio.PopGen.SimCoal.Controller.FastSimCoalController.__init__
Bio.PopGen.SimCoal.Controller.SimCoalController.__init__
Bio.PopGen.SimCoal.Controller._FastSimCoalCommandLine.__init__
Bio.Restriction.PrintFormat.PrintFormat.__init__
Bio.Restriction.Restriction.Analysis.__init__
Bio.Restriction.Restriction.FormattedSeq.__init__
Bio.Restriction.Restriction.RestrictionBatch.__init__
Bio.Restriction.Restriction.RestrictionType.__init__
Bio.SCOP.Astral.__init__
Bio.SCOP.Cla.Index.__init__
Bio.SCOP.Cla.Record.__init__
Bio.SCOP.Des.Record.__init__
Bio.SCOP.Dom.Record.__init__
Bio.SCOP.Domain.__init__
Bio.SCOP.Hie.Record.__init__
Bio.SCOP.Node.__init__
Bio.SCOP.Raf.Res.__init__
Bio.SCOP.Raf.SeqMap.__init__
Bio.SCOP.Raf.SeqMapIndex.__init__
Bio.SCOP.Residues.Residues.__init__
Bio.SCOP.Scop.__init__
Bio.SVDSuperimposer.SVDSuperimposer.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabParser.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.__init__
Bio.SearchIO.BlastIO.blast_text.BlastTextParser.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.__init__
Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator.__init__
Bio.SearchIO.BlatIO.BlatPslIndexer.__init__
Bio.SearchIO.BlatIO.BlatPslParser.__init__
Bio.SearchIO.BlatIO.BlatPslWriter.__init__
Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.__init__
Bio.SearchIO.FastaIO.FastaM10Indexer.__init__
Bio.SearchIO.FastaIO.FastaM10Parser.__init__
Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer.__init__
Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser.__init__
Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.__init__
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser.__init__
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.__init__
Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser.__init__
Bio.SearchIO._index.SearchIndexer.__init__
Bio.SearchIO._model.hit.Hit.__init__
Bio.SearchIO._model.hsp.HSP.__init__
Bio.SearchIO._model.hsp.HSPFragment.__init__
Bio.SearchIO._model.query.QueryResult.__init__
Bio.Seq.MutableSeq.__init__
Bio.Seq.Seq.__init__
Bio.Seq.UnknownSeq.__init__
Bio.SeqFeature.CompoundLocation.__init__
Bio.SeqFeature.FeatureLocation.__init__
Bio.SeqFeature.PositionGap.__init__
Bio.SeqFeature.Reference.__init__
Bio.SeqFeature.SeqFeature.__init__
Bio.SeqIO.FastaIO.FastaWriter.__init__
Bio.SeqIO.Interfaces.SequenceIterator.__init__
Bio.SeqIO.Interfaces.SequenceWriter.__init__
Bio.SeqIO.Interfaces.SequentialSequenceWriter.__init__
Bio.SeqIO.PhdIO.PhdWriter.__init__
Bio.SeqIO.QualityIO.QualPhredWriter.__init__
Bio.SeqIO.SeqXmlIO.SeqXmlIterator.__init__
Bio.SeqIO.SeqXmlIO.SeqXmlWriter.__init__
Bio.SeqIO.SeqXmlIO.XMLRecordIterator.__init__
Bio.SeqIO.SffIO.SffWriter.__init__
Bio.SeqIO.SffIO._AddTellHandle.__init__
Bio.SeqIO.UniprotIO.Parser.__init__
Bio.SeqIO._index.IntelliGeneticsRandomAccess.__init__
Bio.SeqIO._index.SeqFileRandomAccess.__init__
Bio.SeqIO._index.SequentialSeqFileRandomAccess.__init__
Bio.SeqIO._index.SffRandomAccess.__init__
Bio.SeqRecord.SeqRecord.__init__
Bio.SeqRecord._RestrictedDict.__init__
Bio.SeqUtils.CodonUsage.CodonAdaptationIndex.__init__
Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.__init__
Bio.SeqUtils.ProtParam.ProteinAnalysis.__init__
Bio.Sequencing.Ace.ACEFileRecord.__init__
Bio.Sequencing.Ace.Contig.__init__
Bio.Sequencing.Ace.Reads.__init__
Bio.Sequencing.Ace.af.__init__
Bio.Sequencing.Ace.bs.__init__
Bio.Sequencing.Ace.ct.__init__
Bio.Sequencing.Ace.ds.__init__
Bio.Sequencing.Ace.qa.__init__
Bio.Sequencing.Ace.rd.__init__
Bio.Sequencing.Ace.rt.__init__
Bio.Sequencing.Ace.wa.__init__
Bio.Sequencing.Ace.wr.__init__
Bio.Sequencing.Applications._Novoalign.NovoalignCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaAlignCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaBwaswCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaIndexCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaSampeCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaSamseCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsSortCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline.__init__
Bio.Sequencing.Phd.Record.__init__
Bio.SubsMat.FreqTable.FreqTable.__init__
Bio.SubsMat.SeqMat.__init__
Bio.SwissProt.KeyWList.Record.__init__
Bio.SwissProt.Record.__init__
Bio.SwissProt.Reference.__init__
Bio.UniGene.ProtsimLine.__init__
Bio.UniGene.Record.__init__
Bio.UniGene.STSLine.__init__
Bio.UniGene.SequenceLine.__init__
Bio.Wise.dnal.Statistics.__init__
Bio.Wise.psw.AlignmentColumn.__init__
Bio.Wise.psw.ColumnUnit.__init__
Bio._py3k._ordereddict.OrderedDict.__init__
Bio.bgzf.BgzfReader.__init__
Bio.bgzf.BgzfWriter.__init__
Bio.codonalign.codonalignment.CodonAlignment.__init__
Bio.codonalign.codonseq.CodonSeq.__init__
Bio.kNN.kNN.__init__
Bio.motifs.Instances.__init__
Bio.motifs.Motif.__init__
Bio.motifs.alignace.Record.__init__
Bio.motifs.applications._alignace.AlignAceCommandline.__init__
Bio.motifs.applications._alignace.CompareAceCommandline.__init__
Bio.motifs.applications._xxmotif.XXmotifCommandline.__init__
Bio.motifs.jaspar.Motif.__init__
Bio.motifs.jaspar.Record.__init__
Bio.motifs.jaspar.db.JASPAR5.__init__
Bio.motifs.mast.Record.__init__
Bio.motifs.matrix.GenericPositionMatrix.__init__
Bio.motifs.matrix.PositionWeightMatrix.__init__
Bio.motifs.meme.Instance.__init__
Bio.motifs.meme.Motif.__init__
Bio.motifs.meme.Record.__init__
Bio.motifs.thresholds.ScoreDistribution.__init__
Bio.motifs.transfac.Record.__init__
Bio.pairwise2.affine_penalty.__init__
Bio.pairwise2.dictionary_match.__init__
Bio.pairwise2.identity_match.__init__
BioSQL.BioSeq.DBSeq.__init__
BioSQL.BioSeq.DBSeqRecord.__init__
BioSQL.BioSeqDatabase.Adaptor.__init__
BioSQL.BioSeqDatabase.BioSeqDatabase.__init__
BioSQL.BioSeqDatabase.DBServer.__init__
BioSQL.BioSeqDatabase._CursorWrapper.__init__
BioSQL.DBUtils.Generic_dbutils.__init__
BioSQL.Loader.DatabaseLoader.__init__
BioSQL.Loader.DatabaseRemover.__init__" class="py-name" href="#" onclick="return doclink('link-37', '__init__', 'link-37');">__init__</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt id="link-38" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-38', 'alphabet', 'link-14');">alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-name">instances</tt><tt class="py-op">)</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">evalue</tt> <tt class="py-op">=</tt> <tt class="py-number">0.0</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">num_occurrences</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-39" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-39', 'name', 'link-18');">name</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
</div></div><a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line"> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line"> </tt>
<a name="Instance"></a><div id="Instance-def"><a name="L77"></a><tt class="py-lineno"> 77</tt> <a class="py-toggle" href="#" id="Instance-toggle" onclick="return toggle('Instance');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.motifs.meme.Instance-class.html">Instance</a><tt class="py-op">(</tt><tt class="py-base-class">Seq</tt><tt class="py-op">.</tt><tt class="py-base-class">Seq</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Instance-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Instance-expanded"><a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line">    <tt class="py-docstring">"""A class describing the instances of a MEME motif, and the data thereof.</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="Instance.__init__"></a><div id="Instance.__init__-def"><a name="L80"></a><tt class="py-lineno"> 80</tt> <a class="py-toggle" href="#" id="Instance.__init__-toggle" onclick="return toggle('Instance.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.meme.Instance-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-op">*</tt><tt class="py-param">args</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwds</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Instance.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Instance.__init__-expanded"><a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">        <tt id="link-40" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-40', 'Seq', 'link-4');">Seq</a></tt><tt class="py-op">.</tt><tt id="link-41" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-41', 'Seq', 'link-4');">Seq</a></tt><tt class="py-op">.</tt><tt id="link-42" class="py-name"><a title="Bio.Affy.CelFile.Record.__init__
Bio.Align.AlignInfo.PSSM.__init__
Bio.Align.AlignInfo.SummaryInfo.__init__
Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline.__init__
Bio.Align.Applications._Clustalw.ClustalwCommandline.__init__
Bio.Align.Applications._Dialign.DialignCommandline.__init__
Bio.Align.Applications._MSAProbs.MSAProbsCommandline.__init__
Bio.Align.Applications._Mafft.MafftCommandline.__init__
Bio.Align.Applications._Muscle.MuscleCommandline.__init__
Bio.Align.Applications._Prank.PrankCommandline.__init__
Bio.Align.Applications._Probcons.ProbconsCommandline.__init__
Bio.Align.Applications._TCoffee.TCoffeeCommandline.__init__
Bio.Align.Generic.Alignment.__init__
Bio.Align.MultipleSeqAlignment.__init__
Bio.AlignIO.Interfaces.AlignmentIterator.__init__
Bio.AlignIO.Interfaces.AlignmentWriter.__init__
Bio.AlignIO.Interfaces.SequentialAlignmentWriter.__init__
Bio.Alphabet.AlphabetEncoder.__init__
Bio.Alphabet.Gapped.__init__
Bio.Alphabet.HasStopCodon.__init__
Bio.Application.AbstractCommandline.__init__
Bio.Application.ApplicationError.__init__
Bio.Application._AbstractParameter.__init__
Bio.Application._Argument.__init__
Bio.Application._Option.__init__
Bio.Application._StaticArgument.__init__
Bio.Application._Switch.__init__
Bio.Blast.Applications.NcbiblastformatterCommandline.__init__
Bio.Blast.Applications.NcbiblastnCommandline.__init__
Bio.Blast.Applications.NcbiblastpCommandline.__init__
Bio.Blast.Applications.NcbiblastxCommandline.__init__
Bio.Blast.Applications.NcbideltablastCommandline.__init__
Bio.Blast.Applications.NcbipsiblastCommandline.__init__
Bio.Blast.Applications.NcbirpsblastCommandline.__init__
Bio.Blast.Applications.NcbirpstblastnCommandline.__init__
Bio.Blast.Applications.NcbitblastnCommandline.__init__
Bio.Blast.Applications.NcbitblastxCommandline.__init__
Bio.Blast.Applications._NcbibaseblastCommandline.__init__
Bio.Blast.Applications._Ncbiblast2SeqCommandline.__init__
Bio.Blast.Applications._NcbiblastCommandline.__init__
Bio.Blast.Applications._NcbiblastMain2SeqCommandline.__init__
Bio.Blast.NCBIStandalone.BlastErrorParser.__init__
Bio.Blast.NCBIStandalone.BlastParser.__init__
Bio.Blast.NCBIStandalone.Iterator.__init__
Bio.Blast.NCBIStandalone.PSIBlastParser.__init__
Bio.Blast.NCBIStandalone._BlastConsumer.__init__
Bio.Blast.NCBIStandalone._BlastErrorConsumer.__init__
Bio.Blast.NCBIStandalone._PSIBlastConsumer.__init__
Bio.Blast.NCBIXML.BlastParser.__init__
Bio.Blast.NCBIXML._XMLparser.__init__
Bio.Blast.ParseBlastTable.BlastTableEntry.__init__
Bio.Blast.ParseBlastTable.BlastTableReader.__init__
Bio.Blast.ParseBlastTable.BlastTableRec.__init__
Bio.Blast.Record.Alignment.__init__
Bio.Blast.Record.Blast.__init__
Bio.Blast.Record.DatabaseReport.__init__
Bio.Blast.Record.Description.__init__
Bio.Blast.Record.HSP.__init__
Bio.Blast.Record.Header.__init__
Bio.Blast.Record.MultipleAlignment.__init__
Bio.Blast.Record.PSIBlast.__init__
Bio.Blast.Record.Parameters.__init__
Bio.Blast.Record.Round.__init__
Bio.CAPS.CAPSMap.__init__
Bio.CAPS.DifferentialCutsite.__init__
Bio.Cluster.Record.__init__
Bio.Compass.Record.__init__
Bio.Crystal.Chain.__init__
Bio.Crystal.Crystal.__init__
Bio.Crystal.Hetero.__init__
Bio.Data.CodonTable.AmbiguousCodonTable.__init__
Bio.Data.CodonTable.AmbiguousForwardTable.__init__
Bio.Data.CodonTable.CodonTable.__init__
Bio.Data.CodonTable.NCBICodonTable.__init__
Bio.DocSQL.Create.__init__
Bio.DocSQL.Insert.__init__
Bio.DocSQL.IterationCursor.__init__
Bio.DocSQL.Query.__init__
Bio.DocSQL.QueryAll.__init__
Bio.DocSQL.QueryGeneric.__init__
Bio.DocSQL.QueryRow.__init__
Bio.DocSQL.QuerySingle.__init__
Bio.Emboss.Applications.DiffseqCommandline.__init__
Bio.Emboss.Applications.EInvertedCommandline.__init__
Bio.Emboss.Applications.ETandemCommandline.__init__
Bio.Emboss.Applications.Est2GenomeCommandline.__init__
Bio.Emboss.Applications.FConsenseCommandline.__init__
Bio.Emboss.Applications.FDNADistCommandline.__init__
Bio.Emboss.Applications.FDNAParsCommandline.__init__
Bio.Emboss.Applications.FNeighborCommandline.__init__
Bio.Emboss.Applications.FProtDistCommandline.__init__
Bio.Emboss.Applications.FProtParsCommandline.__init__
Bio.Emboss.Applications.FSeqBootCommandline.__init__
Bio.Emboss.Applications.FTreeDistCommandline.__init__
Bio.Emboss.Applications.FuzznucCommandline.__init__
Bio.Emboss.Applications.IepCommandline.__init__
Bio.Emboss.Applications.NeedleCommandline.__init__
Bio.Emboss.Applications.NeedleallCommandline.__init__
Bio.Emboss.Applications.PalindromeCommandline.__init__
Bio.Emboss.Applications.Primer3Commandline.__init__
Bio.Emboss.Applications.PrimerSearchCommandline.__init__
Bio.Emboss.Applications.SeqmatchallCommandline.__init__
Bio.Emboss.Applications.SeqretCommandline.__init__
Bio.Emboss.Applications.StretcherCommandline.__init__
Bio.Emboss.Applications.TranalignCommandline.__init__
Bio.Emboss.Applications.WaterCommandline.__init__
Bio.Emboss.Applications._EmbossCommandLine.__init__
Bio.Emboss.Applications._EmbossMinimalCommandLine.__init__
Bio.Emboss.Primer3.Primers.__init__
Bio.Emboss.Primer3.Record.__init__
Bio.Emboss.PrimerSearch.Amplifier.__init__
Bio.Emboss.PrimerSearch.InputRecord.__init__
Bio.Emboss.PrimerSearch.OutputRecord.__init__
Bio.Entrez.Parser.CorruptedXMLError.__init__
Bio.Entrez.Parser.DataHandler.__init__
Bio.Entrez.Parser.NotXMLError.__init__
Bio.Entrez.Parser.StructureElement.__init__
Bio.Entrez.Parser.ValidationError.__init__
Bio.ExPASy.Enzyme.Record.__init__
Bio.ExPASy.Prodoc.Record.__init__
Bio.ExPASy.Prodoc.Reference.__init__
Bio.ExPASy.Prosite.Record.__init__
Bio.ExPASy.ScanProsite.ContentHandler.__init__
Bio.ExPASy.ScanProsite.Parser.__init__
Bio.ExPASy.ScanProsite.Record.__init__
Bio.FSSP.FSSPAlignDict.__init__
Bio.FSSP.FSSPAlignRec.__init__
Bio.FSSP.FSSPHeader.__init__
Bio.FSSP.FSSPSumRec.__init__
Bio.FSSP.FSSPTools.FSSPMultAlign.__init__
Bio.FSSP.PosAlign.__init__
Bio.FSSP.fssp_rec.fff_rec.__init__
Bio.File.UndoHandle.__init__
Bio.File._IndexedSeqFileDict.__init__
Bio.File._SQLiteManySeqFilesDict.__init__
Bio.GA.Crossover.General.SafeFitnessCrossover.__init__
Bio.GA.Crossover.GeneralPoint.GeneralPointCrossover.__init__
Bio.GA.Crossover.GeneralPoint.InterleaveCrossover.__init__
Bio.GA.Crossover.Point.SinglePointCrossover.__init__
Bio.GA.Crossover.TwoPoint.TwoPointCrossover.__init__
Bio.GA.Crossover.Uniform.UniformCrossover.__init__
Bio.GA.Evolver.GenerationEvolver.__init__
Bio.GA.Evolver.SteadyStateEvolver.__init__
Bio.GA.Mutation.General.SafeFitnessMutation.__init__
Bio.GA.Mutation.Simple.ConversionMutation.__init__
Bio.GA.Mutation.Simple.SinglePositionMutation.__init__
Bio.GA.Organism.Organism.__init__
Bio.GA.Repair.Stabilizing.AmbiguousRepair.__init__
Bio.GA.Selection.Abstract.AbstractSelection.__init__
Bio.GA.Selection.Diversity.DiversitySelection.__init__
Bio.GA.Selection.RouletteWheel.RouletteWheelSelection.__init__
Bio.GA.Selection.Tournament.TournamentSelection.__init__
Bio.GenBank.FeatureParser.__init__
Bio.GenBank.Iterator.__init__
Bio.GenBank.Record.Feature.__init__
Bio.GenBank.Record.Qualifier.__init__
Bio.GenBank.Record.Record.__init__
Bio.GenBank.Record.Reference.__init__
Bio.GenBank.RecordParser.__init__
Bio.GenBank.Scanner.InsdcScanner.__init__
Bio.GenBank._BaseGenBankConsumer.__init__
Bio.GenBank._FeatureConsumer.__init__
Bio.GenBank._RecordConsumer.__init__
Bio.GenBank.utils.FeatureValueCleaner.__init__
Bio.Geo.Record.Record.__init__
Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment.__init__
Bio.Graphics.BasicChromosome.Chromosome.__init__
Bio.Graphics.BasicChromosome.ChromosomeSegment.__init__
Bio.Graphics.BasicChromosome.Organism.__init__
Bio.Graphics.BasicChromosome.TelomereSegment.__init__
Bio.Graphics.BasicChromosome._ChromosomeComponent.__init__
Bio.Graphics.ColorSpiral.ColorSpiral.__init__
Bio.Graphics.Comparative.ComparativeScatterPlot.__init__
Bio.Graphics.DisplayRepresentation.ChromosomeCounts.__init__
Bio.Graphics.Distribution.BarChartDistribution.__init__
Bio.Graphics.Distribution.DistributionPage.__init__
Bio.Graphics.Distribution.LineDistribution.__init__
Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.__init__
Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.__init__
Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.__init__
Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.__init__
Bio.Graphics.GenomeDiagram._Diagram.Diagram.__init__
Bio.Graphics.GenomeDiagram._Feature.Feature.__init__
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__init__
Bio.Graphics.GenomeDiagram._Graph.GraphData.__init__
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__init__
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.__init__
Bio.Graphics.GenomeDiagram._Track.Track.__init__
Bio.Graphics.KGML_vis.KGMLCanvas.__init__
Bio.HMM.DynamicProgramming.AbstractDPAlgorithms.__init__
Bio.HMM.DynamicProgramming.LogDPAlgorithms.__init__
Bio.HMM.DynamicProgramming.ScaledDPAlgorithms.__init__
Bio.HMM.MarkovModel.HiddenMarkovModel.__init__
Bio.HMM.MarkovModel.MarkovModelBuilder.__init__
Bio.HMM.Trainer.AbstractTrainer.__init__
Bio.HMM.Trainer.BaumWelchTrainer.__init__
Bio.HMM.Trainer.KnownStateTrainer.__init__
Bio.HMM.Trainer.TrainingSequence.__init__
Bio.Index._InMemoryIndex.__init__
Bio.Index._ShelveIndex.__init__
Bio.KDTree.KDTree'.KDTree.__init__
Bio.KEGG.Compound.Record.__init__
Bio.KEGG.Enzyme.Record.__init__
Bio.KEGG.KGML.KGML_parser.KGMLParser.__init__
Bio.KEGG.KGML.KGML_pathway.Component.__init__
Bio.KEGG.KGML.KGML_pathway.Entry.__init__
Bio.KEGG.KGML.KGML_pathway.Graphics.__init__
Bio.KEGG.KGML.KGML_pathway.Pathway.__init__
Bio.KEGG.KGML.KGML_pathway.Reaction.__init__
Bio.KEGG.KGML.KGML_pathway.Relation.__init__
Bio.LogisticRegression.LogisticRegression.__init__
Bio.MarkovModel.MarkovModel.__init__
Bio.MaxEntropy.MaxEntropy.__init__
Bio.Motif.Parsers.AlignAce.Record.__init__
Bio.Motif.Parsers.MAST.Record.__init__
Bio.Motif.Parsers.MEME.MEMEInstance.__init__
Bio.Motif.Parsers.MEME.MEMEMotif.__init__
Bio.Motif.Parsers.MEME.MEMERecord.__init__
Bio.Motif.Thresholds.ScoreDistribution.__init__
Bio.Motif._Motif.Motif.__init__
Bio.NMR.xpktools.Peaklist.__init__
Bio.NMR.xpktools.XpkEntry.__init__
Bio.NaiveBayes.NaiveBayes.__init__
Bio.NeuralNetwork.BackPropagation.Layer.AbstractLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.__init__
Bio.NeuralNetwork.BackPropagation.Network.BasicNetwork.__init__
Bio.NeuralNetwork.Gene.Motif.MotifCoder.__init__
Bio.NeuralNetwork.Gene.Motif.MotifFinder.__init__
Bio.NeuralNetwork.Gene.Pattern.PatternIO.__init__
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.__init__
Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness.__init__
Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder.__init__
Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness.__init__
Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator.__init__
Bio.NeuralNetwork.Gene.Schema.Schema.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaCoder.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaFactory.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaFinder.__init__
Bio.NeuralNetwork.Gene.Schema.SimpleFinisher.__init__
Bio.NeuralNetwork.Gene.Signature.SignatureCoder.__init__
Bio.NeuralNetwork.Gene.Signature.SignatureFinder.__init__
Bio.NeuralNetwork.StopTraining.ValidationIncreaseStop.__init__
Bio.NeuralNetwork.Training.ExampleManager.__init__
Bio.NeuralNetwork.Training.TrainingExample.__init__
Bio.Nexus.Nexus.Block.__init__
Bio.Nexus.Nexus.CharBuffer.__init__
Bio.Nexus.Nexus.Commandline.__init__
Bio.Nexus.Nexus.Nexus.__init__
Bio.Nexus.Nexus.StepMatrix.__init__
Bio.Nexus.Nodes.Chain.__init__
Bio.Nexus.Nodes.Node.__init__
Bio.Nexus.Trees.NodeData.__init__
Bio.Nexus.Trees.Tree.__init__
Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap.__init__
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__init__
Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap.__init__
Bio.PDB.Atom.Atom.__init__
Bio.PDB.Atom.DisorderedAtom.__init__
Bio.PDB.Chain.Chain.__init__
Bio.PDB.DSSP'.DSSP.__init__
Bio.PDB.Dice.ChainSelector.__init__
Bio.PDB.Entity.DisorderedEntityWrapper.__init__
Bio.PDB.Entity.Entity.__init__
Bio.PDB.FragmentMapper'.Fragment.__init__
Bio.PDB.FragmentMapper'.FragmentMapper.__init__
Bio.PDB.HSExposure.ExposureCN.__init__
Bio.PDB.HSExposure.HSExposureCA.__init__
Bio.PDB.HSExposure.HSExposureCB.__init__
Bio.PDB.HSExposure._AbstractHSExposure.__init__
Bio.PDB.MMCIF2Dict.MMCIF2Dict.__init__
Bio.PDB.MMCIFParser'.MMCIFParser.__init__
Bio.PDB.Model.Model.__init__
Bio.PDB.NACCESS.NACCESS.__init__
Bio.PDB.NACCESS.NACCESS_atomic.__init__
Bio.PDB.NeighborSearch'.NeighborSearch.__init__
Bio.PDB.PDBIO'.PDBIO.__init__
Bio.PDB.PDBList'.PDBList.__init__
Bio.PDB.PDBParser'.PDBParser.__init__
Bio.PDB.PSEA.PSEA.__init__
Bio.PDB.Polypeptide.CaPPBuilder.__init__
Bio.PDB.Polypeptide.PPBuilder.__init__
Bio.PDB.Polypeptide._PPBuilder.__init__
Bio.PDB.QCPSuperimposer.QCPSuperimposer.__init__
Bio.PDB.Residue.DisorderedResidue.__init__
Bio.PDB.Residue.Residue.__init__
Bio.PDB.ResidueDepth'.ResidueDepth.__init__
Bio.PDB.Structure.Structure.__init__
Bio.PDB.StructureAlignment'.StructureAlignment.__init__
Bio.PDB.StructureBuilder.StructureBuilder.__init__
Bio.PDB.Superimposer'.Superimposer.__init__
Bio.PDB.Vector'.Vector.__init__
Bio.ParserSupport.EventGenerator.__init__
Bio.ParserSupport.TaggingConsumer.__init__
Bio.Pathway.Network.__init__
Bio.Pathway.Reaction.__init__
Bio.Pathway.Rep.Graph.Graph.__init__
Bio.Pathway.Rep.MultiGraph.MultiGraph.__init__
Bio.Pathway.System.__init__
Bio.Phylo.Applications._Fasttree.FastTreeCommandline.__init__
Bio.Phylo.Applications._Phyml.PhymlCommandline.__init__
Bio.Phylo.Applications._Raxml.RaxmlCommandline.__init__
Bio.Phylo.BaseTree.BranchColor.__init__
Bio.Phylo.BaseTree.Clade.__init__
Bio.Phylo.BaseTree.Tree.__init__
Bio.Phylo.CDAO.Clade.__init__
Bio.Phylo.CDAO.Tree.__init__
Bio.Phylo.CDAOIO.Parser.__init__
Bio.Phylo.CDAOIO.Writer.__init__
Bio.Phylo.NeXML.Clade.__init__
Bio.Phylo.NeXML.Tree.__init__
Bio.Phylo.NeXMLIO.Parser.__init__
Bio.Phylo.NeXMLIO.Writer.__init__
Bio.Phylo.Newick.Clade.__init__
Bio.Phylo.Newick.Tree.__init__
Bio.Phylo.NewickIO.Parser.__init__
Bio.Phylo.NewickIO.Writer.__init__
Bio.Phylo.PAML._paml.Paml.__init__
Bio.Phylo.PAML.baseml.Baseml.__init__
Bio.Phylo.PAML.codeml.Codeml.__init__
Bio.Phylo.PAML.yn00.Yn00.__init__
Bio.Phylo.PhyloXML.Accession.__init__
Bio.Phylo.PhyloXML.Annotation.__init__
Bio.Phylo.PhyloXML.BinaryCharacters.__init__
Bio.Phylo.PhyloXML.BranchColor.__init__
Bio.Phylo.PhyloXML.Clade.__init__
Bio.Phylo.PhyloXML.CladeRelation.__init__
Bio.Phylo.PhyloXML.Confidence.__init__
Bio.Phylo.PhyloXML.Date.__init__
Bio.Phylo.PhyloXML.Distribution.__init__
Bio.Phylo.PhyloXML.DomainArchitecture.__init__
Bio.Phylo.PhyloXML.Events.__init__
Bio.Phylo.PhyloXML.Id.__init__
Bio.Phylo.PhyloXML.MolSeq.__init__
Bio.Phylo.PhyloXML.Other.__init__
Bio.Phylo.PhyloXML.Phylogeny.__init__
Bio.Phylo.PhyloXML.Phyloxml.__init__
Bio.Phylo.PhyloXML.Point.__init__
Bio.Phylo.PhyloXML.Polygon.__init__
Bio.Phylo.PhyloXML.Property.__init__
Bio.Phylo.PhyloXML.ProteinDomain.__init__
Bio.Phylo.PhyloXML.Reference.__init__
Bio.Phylo.PhyloXML.Sequence.__init__
Bio.Phylo.PhyloXML.SequenceRelation.__init__
Bio.Phylo.PhyloXML.Taxonomy.__init__
Bio.Phylo.PhyloXML.Uri.__init__
Bio.Phylo.PhyloXMLIO.Parser.__init__
Bio.Phylo.PhyloXMLIO.Writer.__init__
Bio.Phylo.TreeConstruction.DistanceCalculator.__init__
Bio.Phylo.TreeConstruction.DistanceTreeConstructor.__init__
Bio.Phylo.TreeConstruction.NNITreeSearcher.__init__
Bio.Phylo.TreeConstruction.ParsimonyScorer.__init__
Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor.__init__
Bio.Phylo.TreeConstruction._DistanceMatrix.__init__
Bio.Phylo.TreeConstruction._Matrix.__init__
Bio.PopGen.Async.Async.__init__
Bio.PopGen.Async.DirectoryRetriever.__init__
Bio.PopGen.Async.FileRetriever.__init__
Bio.PopGen.Async.Local.Local.__init__
Bio.PopGen.FDist.Async.FDistAsync.__init__
Bio.PopGen.FDist.Async.SplitFDist.__init__
Bio.PopGen.FDist.Controller.FDistController.__init__
Bio.PopGen.FDist.Record.__init__
Bio.PopGen.GenePop.Controller.GenePopController.__init__
Bio.PopGen.GenePop.Controller._FileIterator.__init__
Bio.PopGen.GenePop.Controller._GenePopCommandline.__init__
Bio.PopGen.GenePop.EasyController.EasyController.__init__
Bio.PopGen.GenePop.FileParser.FileRecord.__init__
Bio.PopGen.GenePop.LargeFileParser.Record.__init__
Bio.PopGen.GenePop.Record.__init__
Bio.PopGen.SimCoal.Async.SimCoalCache.__init__
Bio.PopGen.SimCoal.Cache.SimCoalCache.__init__
Bio.PopGen.SimCoal.Controller.FastSimCoalController.__init__
Bio.PopGen.SimCoal.Controller.SimCoalController.__init__
Bio.PopGen.SimCoal.Controller._FastSimCoalCommandLine.__init__
Bio.Restriction.PrintFormat.PrintFormat.__init__
Bio.Restriction.Restriction.Analysis.__init__
Bio.Restriction.Restriction.FormattedSeq.__init__
Bio.Restriction.Restriction.RestrictionBatch.__init__
Bio.Restriction.Restriction.RestrictionType.__init__
Bio.SCOP.Astral.__init__
Bio.SCOP.Cla.Index.__init__
Bio.SCOP.Cla.Record.__init__
Bio.SCOP.Des.Record.__init__
Bio.SCOP.Dom.Record.__init__
Bio.SCOP.Domain.__init__
Bio.SCOP.Hie.Record.__init__
Bio.SCOP.Node.__init__
Bio.SCOP.Raf.Res.__init__
Bio.SCOP.Raf.SeqMap.__init__
Bio.SCOP.Raf.SeqMapIndex.__init__
Bio.SCOP.Residues.Residues.__init__
Bio.SCOP.Scop.__init__
Bio.SVDSuperimposer.SVDSuperimposer.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabParser.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.__init__
Bio.SearchIO.BlastIO.blast_text.BlastTextParser.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.__init__
Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator.__init__
Bio.SearchIO.BlatIO.BlatPslIndexer.__init__
Bio.SearchIO.BlatIO.BlatPslParser.__init__
Bio.SearchIO.BlatIO.BlatPslWriter.__init__
Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.__init__
Bio.SearchIO.FastaIO.FastaM10Indexer.__init__
Bio.SearchIO.FastaIO.FastaM10Parser.__init__
Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer.__init__
Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser.__init__
Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.__init__
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser.__init__
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.__init__
Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser.__init__
Bio.SearchIO._index.SearchIndexer.__init__
Bio.SearchIO._model.hit.Hit.__init__
Bio.SearchIO._model.hsp.HSP.__init__
Bio.SearchIO._model.hsp.HSPFragment.__init__
Bio.SearchIO._model.query.QueryResult.__init__
Bio.Seq.MutableSeq.__init__
Bio.Seq.Seq.__init__
Bio.Seq.UnknownSeq.__init__
Bio.SeqFeature.CompoundLocation.__init__
Bio.SeqFeature.FeatureLocation.__init__
Bio.SeqFeature.PositionGap.__init__
Bio.SeqFeature.Reference.__init__
Bio.SeqFeature.SeqFeature.__init__
Bio.SeqIO.FastaIO.FastaWriter.__init__
Bio.SeqIO.Interfaces.SequenceIterator.__init__
Bio.SeqIO.Interfaces.SequenceWriter.__init__
Bio.SeqIO.Interfaces.SequentialSequenceWriter.__init__
Bio.SeqIO.PhdIO.PhdWriter.__init__
Bio.SeqIO.QualityIO.QualPhredWriter.__init__
Bio.SeqIO.SeqXmlIO.SeqXmlIterator.__init__
Bio.SeqIO.SeqXmlIO.SeqXmlWriter.__init__
Bio.SeqIO.SeqXmlIO.XMLRecordIterator.__init__
Bio.SeqIO.SffIO.SffWriter.__init__
Bio.SeqIO.SffIO._AddTellHandle.__init__
Bio.SeqIO.UniprotIO.Parser.__init__
Bio.SeqIO._index.IntelliGeneticsRandomAccess.__init__
Bio.SeqIO._index.SeqFileRandomAccess.__init__
Bio.SeqIO._index.SequentialSeqFileRandomAccess.__init__
Bio.SeqIO._index.SffRandomAccess.__init__
Bio.SeqRecord.SeqRecord.__init__
Bio.SeqRecord._RestrictedDict.__init__
Bio.SeqUtils.CodonUsage.CodonAdaptationIndex.__init__
Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.__init__
Bio.SeqUtils.ProtParam.ProteinAnalysis.__init__
Bio.Sequencing.Ace.ACEFileRecord.__init__
Bio.Sequencing.Ace.Contig.__init__
Bio.Sequencing.Ace.Reads.__init__
Bio.Sequencing.Ace.af.__init__
Bio.Sequencing.Ace.bs.__init__
Bio.Sequencing.Ace.ct.__init__
Bio.Sequencing.Ace.ds.__init__
Bio.Sequencing.Ace.qa.__init__
Bio.Sequencing.Ace.rd.__init__
Bio.Sequencing.Ace.rt.__init__
Bio.Sequencing.Ace.wa.__init__
Bio.Sequencing.Ace.wr.__init__
Bio.Sequencing.Applications._Novoalign.NovoalignCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaAlignCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaBwaswCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaIndexCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaSampeCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaSamseCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsSortCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline.__init__
Bio.Sequencing.Phd.Record.__init__
Bio.SubsMat.FreqTable.FreqTable.__init__
Bio.SubsMat.SeqMat.__init__
Bio.SwissProt.KeyWList.Record.__init__
Bio.SwissProt.Record.__init__
Bio.SwissProt.Reference.__init__
Bio.UniGene.ProtsimLine.__init__
Bio.UniGene.Record.__init__
Bio.UniGene.STSLine.__init__
Bio.UniGene.SequenceLine.__init__
Bio.Wise.dnal.Statistics.__init__
Bio.Wise.psw.AlignmentColumn.__init__
Bio.Wise.psw.ColumnUnit.__init__
Bio._py3k._ordereddict.OrderedDict.__init__
Bio.bgzf.BgzfReader.__init__
Bio.bgzf.BgzfWriter.__init__
Bio.codonalign.codonalignment.CodonAlignment.__init__
Bio.codonalign.codonseq.CodonSeq.__init__
Bio.kNN.kNN.__init__
Bio.motifs.Instances.__init__
Bio.motifs.Motif.__init__
Bio.motifs.alignace.Record.__init__
Bio.motifs.applications._alignace.AlignAceCommandline.__init__
Bio.motifs.applications._alignace.CompareAceCommandline.__init__
Bio.motifs.applications._xxmotif.XXmotifCommandline.__init__
Bio.motifs.jaspar.Motif.__init__
Bio.motifs.jaspar.Record.__init__
Bio.motifs.jaspar.db.JASPAR5.__init__
Bio.motifs.mast.Record.__init__
Bio.motifs.matrix.GenericPositionMatrix.__init__
Bio.motifs.matrix.PositionWeightMatrix.__init__
Bio.motifs.meme.Instance.__init__
Bio.motifs.meme.Motif.__init__
Bio.motifs.meme.Record.__init__
Bio.motifs.thresholds.ScoreDistribution.__init__
Bio.motifs.transfac.Record.__init__
Bio.pairwise2.affine_penalty.__init__
Bio.pairwise2.dictionary_match.__init__
Bio.pairwise2.identity_match.__init__
BioSQL.BioSeq.DBSeq.__init__
BioSQL.BioSeq.DBSeqRecord.__init__
BioSQL.BioSeqDatabase.Adaptor.__init__
BioSQL.BioSeqDatabase.BioSeqDatabase.__init__
BioSQL.BioSeqDatabase.DBServer.__init__
BioSQL.BioSeqDatabase._CursorWrapper.__init__
BioSQL.DBUtils.Generic_dbutils.__init__
BioSQL.Loader.DatabaseLoader.__init__
BioSQL.Loader.DatabaseRemover.__init__" class="py-name" href="#" onclick="return doclink('link-42', '__init__', 'link-37');">__init__</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-op">*</tt><tt class="py-name">args</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-name">kwds</tt><tt class="py-op">)</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">sequence_name</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-43" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.start=Bio.SeqFeature.CompoundLocation-class.html#start,Variable Bio.SeqFeature.FeatureLocation.start=Bio.SeqFeature.FeatureLocation-class.html#start"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-43', 'start', 'link-43');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pvalue</tt> <tt class="py-op">=</tt> <tt class="py-number">1.0</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-44" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.strand()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#strand,Variable Bio.SeqFeature.CompoundLocation.strand=Bio.SeqFeature.CompoundLocation-class.html#strand,Variable Bio.SeqFeature.FeatureLocation.strand=Bio.SeqFeature.FeatureLocation-class.html#strand,Variable Bio.SeqFeature.SeqFeature.strand=Bio.SeqFeature.SeqFeature-class.html#strand"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-44', 'strand', 'link-44');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-45" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-45', 'length', 'link-16');">length</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">motif_name</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
</div></div><a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line"> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line"> </tt>
<a name="Record"></a><div id="Record-def"><a name="L90"></a><tt class="py-lineno"> 90</tt> <a class="py-toggle" href="#" id="Record-toggle" onclick="return toggle('Record');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.motifs.meme.Record-class.html">Record</a><tt class="py-op">(</tt><tt class="py-base-class">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Record-expanded"><a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">    <tt class="py-docstring">"""A class for holding the results of a MEME run.</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line"><tt class="py-docstring">    A meme.Record is an object that holds the results from running</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line"><tt class="py-docstring">    MEME. It implements no methods of its own.</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line"><tt class="py-docstring">    The meme.Record class inherits from list, so you can access individual</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line"><tt class="py-docstring">    motifs in the record by their index. Alternatively, you can find a motif</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line"><tt class="py-docstring">    by its name:</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import motifs</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; with open("meme.output.txt") as f:</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line"><tt class="py-docstring">    ...     record = motifs.parse(f, 'MEME')</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; motif = record[0]</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(motif.name)</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line"><tt class="py-docstring">    Motif 1</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; motif = record['Motif 1']</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(motif.name)</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line"><tt class="py-docstring">    Motif 1</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line"> </tt>
<a name="Record.__init__"></a><div id="Record.__init__-def"><a name="L111"></a><tt class="py-lineno">111</tt> <a class="py-toggle" href="#" id="Record.__init__-toggle" onclick="return toggle('Record.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.meme.Record-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record.__init__-expanded"><a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">        <tt class="py-docstring">"""__init__ (self)"""</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-46" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HeaderConsumer.version()=Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#version,Function Bio.Cluster.cluster.version()=Bio.Cluster.cluster-module.html#version,Method Bio.GenBank._FeatureConsumer.version()=Bio.GenBank._FeatureConsumer-class.html#version,Method Bio.GenBank._RecordConsumer.version()=Bio.GenBank._RecordConsumer-class.html#version,Variable Bio.motifs.jaspar.Motif.version=Bio.motifs.jaspar.Motif-class.html#version"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-46', 'version', 'link-46');">version</a></tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">datafile</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">command</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-47" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-47', 'alphabet', 'link-14');">alphabet</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">sequences</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div><a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line"> </tt>
<a name="Record.__getitem__"></a><div id="Record.__getitem__-def"><a name="L119"></a><tt class="py-lineno">119</tt> <a class="py-toggle" href="#" id="Record.__getitem__-toggle" onclick="return toggle('Record.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.meme.Record-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">key</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record.__getitem__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record.__getitem__-expanded"><a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">motif</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">:</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt id="link-48" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-48', 'name', 'link-18');">name</a></tt> <tt class="py-op">==</tt> <tt class="py-name">key</tt><tt class="py-op">:</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">                    <tt class="py-keyword">return</tt> <tt class="py-name">motif</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">list</tt><tt class="py-op">.</tt><tt id="link-49" class="py-name" targets="Method Bio.Align.AlignInfo.PSSM.__getitem__()=Bio.Align.AlignInfo.PSSM-class.html#__getitem__,Method Bio.Align.Generic.Alignment.__getitem__()=Bio.Align.Generic.Alignment-class.html#__getitem__,Method Bio.Align.MultipleSeqAlignment.__getitem__()=Bio.Align.MultipleSeqAlignment-class.html#__getitem__,Method Bio.Crystal.Chain.__getitem__()=Bio.Crystal.Chain-class.html#__getitem__,Method Bio.Crystal.Crystal.__getitem__()=Bio.Crystal.Crystal-class.html#__getitem__,Method Bio.Data.CodonTable.AmbiguousForwardTable.__getitem__()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#__getitem__,Method Bio.FSSP.fssp_rec.fff_rec.__getitem__()=Bio.FSSP.fssp_rec.fff_rec-class.html#__getitem__,Method Bio.File._IndexedSeqFileDict.__getitem__()=Bio.File._IndexedSeqFileDict-class.html#__getitem__,Method Bio.File._SQLiteManySeqFilesDict.__getitem__()=Bio.File._SQLiteManySeqFilesDict-class.html#__getitem__,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.__getitem__()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#__getitem__,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__getitem__()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#__getitem__,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.__getitem__()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#__getitem__,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__getitem__()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#__getitem__,Method Bio.Graphics.GenomeDiagram._Track.Track.__getitem__()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#__getitem__,Method Bio.Motif._Motif.Motif.__getitem__()=Bio.Motif._Motif.Motif-class.html#__getitem__,Method Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__getitem__()=Bio.PDB.AbstractPropertyMap.AbstractPropertyMap-class.html#__getitem__,Method Bio.PDB.Chain.Chain.__getitem__()=Bio.PDB.Chain.Chain-class.html#__getitem__,Method Bio.PDB.Entity.DisorderedEntityWrapper.__getitem__()=Bio.PDB.Entity.DisorderedEntityWrapper-class.html#__getitem__,Method Bio.PDB.Entity.Entity.__getitem__()=Bio.PDB.Entity.Entity-class.html#__getitem__,Method Bio.PDB.FragmentMapper'.FragmentMapper.__getitem__()=Bio.PDB.FragmentMapper%27.FragmentMapper-class.html#__getitem__,Method Bio.PDB.Vector'.Vector.__getitem__()=Bio.PDB.Vector%27.Vector-class.html#__getitem__,Method Bio.Phylo.BaseTree.Clade.__getitem__()=Bio.Phylo.BaseTree.Clade-class.html#__getitem__,Method Bio.Phylo.PhyloXML.Events.__getitem__()=Bio.Phylo.PhyloXML.Events-class.html#__getitem__,Method Bio.Phylo.PhyloXML.Phyloxml.__getitem__()=Bio.Phylo.PhyloXML.Phyloxml-class.html#__getitem__,Method Bio.Phylo.TreeConstruction._Matrix.__getitem__()=Bio.Phylo.TreeConstruction._Matrix-class.html#__getitem__,Method Bio.Restriction.Restriction.FormattedSeq.__getitem__()=Bio.Restriction.Restriction.FormattedSeq-class.html#__getitem__,Method Bio.SCOP.Cla.Index.__getitem__()=Bio.SCOP.Cla.Index-class.html#__getitem__,Method Bio.SCOP.Raf.SeqMap.__getitem__()=Bio.SCOP.Raf.SeqMap-class.html#__getitem__,Method Bio.SCOP.Raf.SeqMapIndex.__getitem__()=Bio.SCOP.Raf.SeqMapIndex-class.html#__getitem__,Method Bio.SearchIO._model.hit.Hit.__getitem__()=Bio.SearchIO._model.hit.Hit-class.html#__getitem__,Method Bio.SearchIO._model.hsp.HSP.__getitem__()=Bio.SearchIO._model.hsp.HSP-class.html#__getitem__,Method Bio.SearchIO._model.hsp.HSPFragment.__getitem__()=Bio.SearchIO._model.hsp.HSPFragment-class.html#__getitem__,Method Bio.SearchIO._model.query.QueryResult.__getitem__()=Bio.SearchIO._model.query.QueryResult-class.html#__getitem__,Method Bio.Seq.MutableSeq.__getitem__()=Bio.Seq.MutableSeq-class.html#__getitem__,Method Bio.Seq.Seq.__getitem__()=Bio.Seq.Seq-class.html#__getitem__,Method Bio.Seq.UnknownSeq.__getitem__()=Bio.Seq.UnknownSeq-class.html#__getitem__,Method Bio.SeqRecord.SeqRecord.__getitem__()=Bio.SeqRecord.SeqRecord-class.html#__getitem__,Method Bio.codonalign.codonalignment.CodonAlignment.__getitem__()=Bio.codonalign.codonalignment.CodonAlignment-class.html#__getitem__,Method Bio.codonalign.codonseq.CodonSeq.__getitem__()=Bio.codonalign.codonseq.CodonSeq-class.html#__getitem__,Method Bio.motifs.mast.Record.__getitem__()=Bio.motifs.mast.Record-class.html#__getitem__,Method Bio.motifs.matrix.GenericPositionMatrix.__getitem__()=Bio.motifs.matrix.GenericPositionMatrix-class.html#__getitem__,Method Bio.motifs.meme.Record.__getitem__()=Bio.motifs.meme.Record-class.html#__getitem__,Method BioSQL.BioSeq.DBSeq.__getitem__()=BioSQL.BioSeq.DBSeq-class.html#__getitem__,Method BioSQL.BioSeqDatabase.BioSeqDatabase.__getitem__()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#__getitem__,Method BioSQL.BioSeqDatabase.DBServer.__getitem__()=BioSQL.BioSeqDatabase.DBServer-class.html#__getitem__"><a title="Bio.Align.AlignInfo.PSSM.__getitem__
Bio.Align.Generic.Alignment.__getitem__
Bio.Align.MultipleSeqAlignment.__getitem__
Bio.Crystal.Chain.__getitem__
Bio.Crystal.Crystal.__getitem__
Bio.Data.CodonTable.AmbiguousForwardTable.__getitem__
Bio.FSSP.fssp_rec.fff_rec.__getitem__
Bio.File._IndexedSeqFileDict.__getitem__
Bio.File._SQLiteManySeqFilesDict.__getitem__
Bio.Graphics.GenomeDiagram._Diagram.Diagram.__getitem__
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__getitem__
Bio.Graphics.GenomeDiagram._Graph.GraphData.__getitem__
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__getitem__
Bio.Graphics.GenomeDiagram._Track.Track.__getitem__
Bio.Motif._Motif.Motif.__getitem__
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__getitem__
Bio.PDB.Chain.Chain.__getitem__
Bio.PDB.Entity.DisorderedEntityWrapper.__getitem__
Bio.PDB.Entity.Entity.__getitem__
Bio.PDB.FragmentMapper'.FragmentMapper.__getitem__
Bio.PDB.Vector'.Vector.__getitem__
Bio.Phylo.BaseTree.Clade.__getitem__
Bio.Phylo.PhyloXML.Events.__getitem__
Bio.Phylo.PhyloXML.Phyloxml.__getitem__
Bio.Phylo.TreeConstruction._Matrix.__getitem__
Bio.Restriction.Restriction.FormattedSeq.__getitem__
Bio.SCOP.Cla.Index.__getitem__
Bio.SCOP.Raf.SeqMap.__getitem__
Bio.SCOP.Raf.SeqMapIndex.__getitem__
Bio.SearchIO._model.hit.Hit.__getitem__
Bio.SearchIO._model.hsp.HSP.__getitem__
Bio.SearchIO._model.hsp.HSPFragment.__getitem__
Bio.SearchIO._model.query.QueryResult.__getitem__
Bio.Seq.MutableSeq.__getitem__
Bio.Seq.Seq.__getitem__
Bio.Seq.UnknownSeq.__getitem__
Bio.SeqRecord.SeqRecord.__getitem__
Bio.codonalign.codonalignment.CodonAlignment.__getitem__
Bio.codonalign.codonseq.CodonSeq.__getitem__
Bio.motifs.mast.Record.__getitem__
Bio.motifs.matrix.GenericPositionMatrix.__getitem__
Bio.motifs.meme.Record.__getitem__
BioSQL.BioSeq.DBSeq.__getitem__
BioSQL.BioSeqDatabase.BioSeqDatabase.__getitem__
BioSQL.BioSeqDatabase.DBServer.__getitem__" class="py-name" href="#" onclick="return doclink('link-49', '__getitem__', 'link-49');">__getitem__</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-name">key</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line"> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line"> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line"><tt class="py-comment"># Everything below is private</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line"> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line"> </tt>
<a name="__read_version"></a><div id="__read_version-def"><a name="L131"></a><tt class="py-lineno">131</tt> <a class="py-toggle" href="#" id="__read_version-toggle" onclick="return toggle('__read_version');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.meme-module.html#__read_version">__read_version</a><tt class="py-op">(</tt><tt class="py-param">record</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="__read_version-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="__read_version-expanded"><a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-50" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-50', 'startswith', 'link-13');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'MEME version'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line">            <tt class="py-keyword">break</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Improper input file. File should contain a line starting MEME version."</tt><tt class="py-op">)</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line">    <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-51" class="py-name" targets="Method Bio.Seq.Seq.strip()=Bio.Seq.Seq-class.html#strip"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-51', 'strip', 'link-51');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line">    <tt class="py-name">ls</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-52" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-52', 'split', 'link-52');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line">    <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-53" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-53', 'version', 'link-46');">version</a></tt> <tt class="py-op">=</tt> <tt class="py-name">ls</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> </tt>
</div><a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line"> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line"> </tt>
<a name="__read_datafile"></a><div id="__read_datafile-def"><a name="L142"></a><tt class="py-lineno">142</tt> <a class="py-toggle" href="#" id="__read_datafile-toggle" onclick="return toggle('__read_datafile');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.meme-module.html#__read_datafile">__read_datafile</a><tt class="py-op">(</tt><tt class="py-param">record</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="__read_datafile-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="__read_datafile-expanded"><a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-54" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-54', 'startswith', 'link-13');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'TRAINING SET'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">            <tt class="py-keyword">break</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected end of stream: 'TRAINING SET' not found."</tt><tt class="py-op">)</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-55" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-55', 'next', 'link-32');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected end of stream: Expected to find line starting with '****'"</tt><tt class="py-op">)</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-56" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-56', 'startswith', 'link-13');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'****'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Line does not start with '****':\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-57" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-57', 'next', 'link-32');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected end of stream: Expected to find line starting with 'DATAFILE'"</tt><tt class="py-op">)</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-58" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-58', 'startswith', 'link-13');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'DATAFILE'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Line does not start with 'DATAFILE':\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">    <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-59" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-59', 'strip', 'link-51');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">    <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'DATAFILE= '</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">    <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">datafile</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> </tt>
</div><a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line"> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line"> </tt>
<a name="__read_alphabet"></a><div id="__read_alphabet-def"><a name="L165"></a><tt class="py-lineno">165</tt> <a class="py-toggle" href="#" id="__read_alphabet-toggle" onclick="return toggle('__read_alphabet');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.meme-module.html#__read_alphabet">__read_alphabet</a><tt class="py-op">(</tt><tt class="py-param">record</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="__read_alphabet-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="__read_alphabet-expanded"><a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-60" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-60', 'next', 'link-32');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected end of stream: Expected to find line starting with 'ALPHABET'"</tt><tt class="py-op">)</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-61" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-61', 'startswith', 'link-13');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'ALPHABET'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Line does not start with 'ALPHABET':\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line">    <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-62" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-62', 'strip', 'link-51');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line">    <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'ALPHABET= '</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">line</tt> <tt class="py-op">==</tt> <tt class="py-string">'ACGT'</tt><tt class="py-op">:</tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line">        <tt class="py-name">al</tt> <tt class="py-op">=</tt> <tt id="link-63" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-63', 'IUPAC', 'link-2');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-64" class="py-name" targets="Variable Bio.Alphabet.IUPAC.unambiguous_dna=Bio.Alphabet.IUPAC-module.html#unambiguous_dna"><a title="Bio.Alphabet.IUPAC.unambiguous_dna" class="py-name" href="#" onclick="return doclink('link-64', 'unambiguous_dna', 'link-64');">unambiguous_dna</a></tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">        <tt class="py-name">al</tt> <tt class="py-op">=</tt> <tt id="link-65" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-65', 'IUPAC', 'link-2');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-66" class="py-name" targets="Variable Bio.Alphabet.IUPAC.protein=Bio.Alphabet.IUPAC-module.html#protein"><a title="Bio.Alphabet.IUPAC.protein" class="py-name" href="#" onclick="return doclink('link-66', 'protein', 'link-66');">protein</a></tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">    <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-67" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-67', 'alphabet', 'link-14');">alphabet</a></tt> <tt class="py-op">=</tt> <tt class="py-name">al</tt> </tt>
</div><a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line"> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line"> </tt>
<a name="__read_sequences"></a><div id="__read_sequences-def"><a name="L181"></a><tt class="py-lineno">181</tt> <a class="py-toggle" href="#" id="__read_sequences-toggle" onclick="return toggle('__read_sequences');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.meme-module.html#__read_sequences">__read_sequences</a><tt class="py-op">(</tt><tt class="py-param">record</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="__read_sequences-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="__read_sequences-expanded"><a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-68" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-68', 'next', 'link-32');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected end of stream: Expected to find line starting with 'Sequence name'"</tt><tt class="py-op">)</tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-69" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-69', 'startswith', 'link-13');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Sequence name'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Line does not start with 'Sequence name':\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-70" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-70', 'next', 'link-32');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected end of stream: Expected to find line starting with '----'"</tt><tt class="py-op">)</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-71" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-71', 'startswith', 'link-13');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'----'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Line does not start with '----':\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-72" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-72', 'startswith', 'link-13');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'***'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">            <tt class="py-keyword">break</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-73" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-73', 'strip', 'link-51');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line">        <tt class="py-name">ls</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-74" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-74', 'split', 'link-52');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line">        <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">sequences</tt><tt class="py-op">.</tt><tt id="link-75" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-75', 'append', 'link-30');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">ls</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">ls</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">6</tt><tt class="py-op">:</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">sequences</tt><tt class="py-op">.</tt><tt id="link-76" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-76', 'append', 'link-30');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">ls</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected end of stream: Expected to find line starting with '***'"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line"> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line"> </tt>
<a name="__read_command"></a><div id="__read_command-def"><a name="L206"></a><tt class="py-lineno">206</tt> <a class="py-toggle" href="#" id="__read_command-toggle" onclick="return toggle('__read_command');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.meme-module.html#__read_command">__read_command</a><tt class="py-op">(</tt><tt class="py-param">record</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="__read_command-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="__read_command-expanded"><a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-77" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-77', 'startswith', 'link-13');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'command:'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line">            <tt class="py-keyword">break</tt> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected end of stream: Expected to find line starting with 'command'"</tt><tt class="py-op">)</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line">    <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-78" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-78', 'strip', 'link-51');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line">    <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'command: '</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line">    <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">command</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> </tt>
</div><a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line"> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line"> </tt>
<a name="__read_motif_statistics"></a><div id="__read_motif_statistics-def"><a name="L217"></a><tt class="py-lineno">217</tt> <a class="py-toggle" href="#" id="__read_motif_statistics-toggle" onclick="return toggle('__read_motif_statistics');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.meme-module.html#__read_motif_statistics">__read_motif_statistics</a><tt class="py-op">(</tt><tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="__read_motif_statistics-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="__read_motif_statistics-expanded"><a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line">    <tt class="py-comment"># Depending on the version of MEME, this line either like like</tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line">    <tt class="py-comment">#    MOTIF  1        width =  19  sites =   3  llr = 43  E-value = 6.9e-002</tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">    <tt class="py-comment"># or like</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line">    <tt class="py-comment">#    MOTIF  1 MEME    width =  19  sites =   3  llr = 43  E-value = 6.9e-002</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line">    <tt class="py-name">words</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-79" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-79', 'split', 'link-52');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">words</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">==</tt><tt class="py-string">'MOTIF'</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line">    <tt class="py-name">motif_number</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">words</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">words</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">==</tt><tt class="py-string">'MEME'</tt><tt class="py-op">:</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line">        <tt class="py-name">key_values</tt> <tt class="py-op">=</tt> <tt class="py-name">words</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line">        <tt class="py-name">key_values</tt> <tt class="py-op">=</tt> <tt class="py-name">words</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">    <tt id="link-80" class="py-name" targets="Method Bio.Crystal.Crystal.keys()=Bio.Crystal.Crystal-class.html#keys,Method Bio.File._IndexedSeqFileDict.keys()=Bio.File._IndexedSeqFileDict-class.html#keys,Method Bio.File._SQLiteManySeqFilesDict.keys()=Bio.File._SQLiteManySeqFilesDict-class.html#keys,Method Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys()=Bio.PDB.AbstractPropertyMap.AbstractPropertyMap-class.html#keys,Method Bio.Phylo.PhyloXML.Events.keys()=Bio.Phylo.PhyloXML.Events-class.html#keys,Method Bio._py3k._ordereddict.OrderedDict.keys()=Bio._py3k._ordereddict.OrderedDict-class.html#keys,Method BioSQL.BioSeqDatabase.BioSeqDatabase.keys()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#keys,Method BioSQL.BioSeqDatabase.DBServer.keys()=BioSQL.BioSeqDatabase.DBServer-class.html#keys"><a title="Bio.Crystal.Crystal.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
Bio._py3k._ordereddict.OrderedDict.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-80', 'keys', 'link-80');">keys</a></tt> <tt class="py-op">=</tt> <tt class="py-name">key_values</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">:</tt><tt class="py-number">3</tt><tt class="py-op">]</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line">    <tt class="py-name">equal_signs</tt> <tt class="py-op">=</tt> <tt class="py-name">key_values</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">:</tt><tt class="py-number">3</tt><tt class="py-op">]</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line">    <tt id="link-81" class="py-name" targets="Method Bio.Crystal.Crystal.values()=Bio.Crystal.Crystal-class.html#values,Method Bio.File._IndexedSeqFileDict.values()=Bio.File._IndexedSeqFileDict-class.html#values,Method Bio.Phylo.PhyloXML.Events.values()=Bio.Phylo.PhyloXML.Events-class.html#values,Method Bio._py3k._ordereddict.OrderedDict.values()=Bio._py3k._ordereddict.OrderedDict-class.html#values,Method BioSQL.BioSeqDatabase.BioSeqDatabase.values()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#values,Method BioSQL.BioSeqDatabase.DBServer.values()=BioSQL.BioSeqDatabase.DBServer-class.html#values"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
Bio._py3k._ordereddict.OrderedDict.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-81', 'values', 'link-81');">values</a></tt> <tt class="py-op">=</tt> <tt class="py-name">key_values</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">:</tt><tt class="py-op">:</tt><tt class="py-number">3</tt><tt class="py-op">]</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt id="link-82" class="py-name"><a title="Bio.Crystal.Crystal.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
Bio._py3k._ordereddict.OrderedDict.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-82', 'keys', 'link-80');">keys</a></tt><tt class="py-op">==</tt><tt class="py-op">[</tt><tt class="py-string">'width'</tt><tt class="py-op">,</tt> <tt class="py-string">'sites'</tt><tt class="py-op">,</tt> <tt class="py-string">'llr'</tt><tt class="py-op">,</tt> <tt class="py-string">'E-value'</tt><tt class="py-op">]</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">equal_sign</tt> <tt class="py-keyword">in</tt> <tt class="py-name">equal_signs</tt><tt class="py-op">:</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">equal_sign</tt><tt class="py-op">==</tt><tt class="py-string">'='</tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line">    <tt id="link-83" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-83', 'length', 'link-16');">length</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-84" class="py-name"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
Bio._py3k._ordereddict.OrderedDict.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-84', 'values', 'link-81');">values</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line">    <tt class="py-name">num_occurrences</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-85" class="py-name"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
Bio._py3k._ordereddict.OrderedDict.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-85', 'values', 'link-81');">values</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line">    <tt class="py-name">evalue</tt> <tt class="py-op">=</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt id="link-86" class="py-name"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
Bio._py3k._ordereddict.OrderedDict.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-86', 'values', 'link-81');">values</a></tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">motif_number</tt><tt class="py-op">,</tt> <tt id="link-87" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-87', 'length', 'link-16');">length</a></tt><tt class="py-op">,</tt> <tt class="py-name">num_occurrences</tt><tt class="py-op">,</tt> <tt class="py-name">evalue</tt> </tt>
</div><a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line"> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line"> </tt>
<a name="__read_motif_name"></a><div id="__read_motif_name-def"><a name="L241"></a><tt class="py-lineno">241</tt> <a class="py-toggle" href="#" id="__read_motif_name-toggle" onclick="return toggle('__read_motif_name');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.meme-module.html#__read_motif_name">__read_motif_name</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="__read_motif_name-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="__read_motif_name-expanded"><a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">'sorted by position p-value'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line">            <tt class="py-keyword">break</tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">'Unexpected end of stream: Failed to find motif name'</tt><tt class="py-op">)</tt> </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line">    <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-88" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-88', 'strip', 'link-51');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line">    <tt class="py-name">words</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-89" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-89', 'split', 'link-52');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line">    <tt id="link-90" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-90', 'name', 'link-18');">name</a></tt> <tt class="py-op">=</tt> <tt class="py-string">" "</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">words</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-91" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-91', 'name', 'link-18');">name</a></tt> </tt>
</div><a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line"> </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line"> </tt>
<a name="__read_motif_sequences"></a><div id="__read_motif_sequences-def"><a name="L253"></a><tt class="py-lineno">253</tt> <a class="py-toggle" href="#" id="__read_motif_sequences-toggle" onclick="return toggle('__read_motif_sequences');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.meme-module.html#__read_motif_sequences">__read_motif_sequences</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">motif_name</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">,</tt> <tt class="py-param">length</tt><tt class="py-op">,</tt> <tt class="py-param">revcomp</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="__read_motif_sequences-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="__read_motif_sequences-expanded"><a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-92" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-92', 'next', 'link-32');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">'Unexpected end of stream: Failed to find motif sequences'</tt><tt class="py-op">)</tt> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-93" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-93', 'startswith', 'link-13');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'---'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Line does not start with '---':\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-94" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-94', 'next', 'link-32');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected end of stream: Expected to find line starting with 'Sequence name'"</tt><tt class="py-op">)</tt> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-95" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-95', 'startswith', 'link-13');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Sequence name'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Line does not start with 'Sequence name':\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-96" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-96', 'next', 'link-32');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">'Unexpected end of stream: Failed to find motif sequences'</tt><tt class="py-op">)</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-97" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-97', 'startswith', 'link-13');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'---'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Line does not start with '---':\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line">    <tt class="py-name">instances</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-98" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-98', 'startswith', 'link-13');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'---'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line">            <tt class="py-keyword">break</tt> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-99" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-99', 'strip', 'link-51');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line">        <tt class="py-name">words</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-100" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-100', 'split', 'link-52');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">revcomp</tt><tt class="py-op">:</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line">            <tt id="link-101" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-101', 'strand', 'link-44');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-name">words</tt><tt class="py-op">.</tt><tt id="link-102" class="py-name" targets="Method Bio.File._IndexedSeqFileDict.pop()=Bio.File._IndexedSeqFileDict-class.html#pop,Method Bio.SearchIO._model.hit.Hit.pop()=Bio.SearchIO._model.hit.Hit-class.html#pop,Method Bio.SearchIO._model.query.QueryResult.pop()=Bio.SearchIO._model.query.QueryResult-class.html#pop,Method Bio.Seq.MutableSeq.pop()=Bio.Seq.MutableSeq-class.html#pop,Method Bio._py3k._ordereddict.OrderedDict.pop()=Bio._py3k._ordereddict.OrderedDict-class.html#pop"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop
Bio._py3k._ordereddict.OrderedDict.pop" class="py-name" href="#" onclick="return doclink('link-102', 'pop', 'link-102');">pop</a></tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line">            <tt id="link-103" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-103', 'strand', 'link-44');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'+'</tt> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line">        <tt id="link-104" class="py-name" targets="Method Bio.FSSP.FSSPAlignDict.sequence()=Bio.FSSP.FSSPAlignDict-class.html#sequence,Method Bio.GenBank._FeatureConsumer.sequence()=Bio.GenBank._FeatureConsumer-class.html#sequence,Method Bio.GenBank._RecordConsumer.sequence()=Bio.GenBank._RecordConsumer-class.html#sequence,Method Bio.Phylo.PhyloXMLIO.Writer.sequence()=Bio.Phylo.PhyloXMLIO.Writer-class.html#sequence"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-104', 'sequence', 'link-104');">sequence</a></tt> <tt class="py-op">=</tt> <tt class="py-name">words</tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-105" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-105', 'sequence', 'link-104');">sequence</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt id="link-106" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-106', 'length', 'link-16');">length</a></tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">        <tt id="link-107" class="py-name" targets="Variable Bio.PopGen.SimCoal.instance=Bio.PopGen.SimCoal-module.html#instance"><a title="Bio.PopGen.SimCoal.instance" class="py-name" href="#" onclick="return doclink('link-107', 'instance', 'link-107');">instance</a></tt> <tt class="py-op">=</tt> <tt id="link-108" class="py-name" targets="Class Bio.motifs.meme.Instance=Bio.motifs.meme.Instance-class.html"><a title="Bio.motifs.meme.Instance" class="py-name" href="#" onclick="return doclink('link-108', 'Instance', 'link-108');">Instance</a></tt><tt class="py-op">(</tt><tt id="link-109" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-109', 'sequence', 'link-104');">sequence</a></tt><tt class="py-op">,</tt> <tt id="link-110" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-110', 'alphabet', 'link-14');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line">        <tt id="link-111" class="py-name"><a title="Bio.PopGen.SimCoal.instance" class="py-name" href="#" onclick="return doclink('link-111', 'instance', 'link-107');">instance</a></tt><tt class="py-op">.</tt><tt class="py-name">motif_name</tt> <tt class="py-op">=</tt> <tt class="py-name">motif_name</tt> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line">        <tt id="link-112" class="py-name"><a title="Bio.PopGen.SimCoal.instance" class="py-name" href="#" onclick="return doclink('link-112', 'instance', 'link-107');">instance</a></tt><tt class="py-op">.</tt><tt class="py-name">sequence_name</tt> <tt class="py-op">=</tt> <tt class="py-name">words</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">        <tt id="link-113" class="py-name"><a title="Bio.PopGen.SimCoal.instance" class="py-name" href="#" onclick="return doclink('link-113', 'instance', 'link-107');">instance</a></tt><tt class="py-op">.</tt><tt id="link-114" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-114', 'start', 'link-43');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">words</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">        <tt id="link-115" class="py-name"><a title="Bio.PopGen.SimCoal.instance" class="py-name" href="#" onclick="return doclink('link-115', 'instance', 'link-107');">instance</a></tt><tt class="py-op">.</tt><tt class="py-name">pvalue</tt> <tt class="py-op">=</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">words</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">        <tt id="link-116" class="py-name"><a title="Bio.PopGen.SimCoal.instance" class="py-name" href="#" onclick="return doclink('link-116', 'instance', 'link-107');">instance</a></tt><tt class="py-op">.</tt><tt id="link-117" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-117', 'strand', 'link-44');">strand</a></tt> <tt class="py-op">=</tt> <tt id="link-118" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-118', 'strand', 'link-44');">strand</a></tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">        <tt id="link-119" class="py-name"><a title="Bio.PopGen.SimCoal.instance" class="py-name" href="#" onclick="return doclink('link-119', 'instance', 'link-107');">instance</a></tt><tt class="py-op">.</tt><tt id="link-120" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-120', 'length', 'link-16');">length</a></tt> <tt class="py-op">=</tt> <tt id="link-121" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-121', 'length', 'link-16');">length</a></tt> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line">        <tt class="py-name">instances</tt><tt class="py-op">.</tt><tt id="link-122" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-122', 'append', 'link-30');">append</a></tt><tt class="py-op">(</tt><tt id="link-123" class="py-name"><a title="Bio.PopGen.SimCoal.instance" class="py-name" href="#" onclick="return doclink('link-123', 'instance', 'link-107');">instance</a></tt><tt class="py-op">)</tt> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">'Unexpected end of stream'</tt><tt class="py-op">)</tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-124" class="py-name"><a title="Bio.motifs" class="py-name" href="#" onclick="return doclink('link-124', 'motifs', 'link-6');">motifs</a></tt><tt class="py-op">.</tt><tt id="link-125" class="py-name" targets="Class Bio.motifs.Instances=Bio.motifs.Instances-class.html"><a title="Bio.motifs.Instances" class="py-name" href="#" onclick="return doclink('link-125', 'Instances', 'link-125');">Instances</a></tt><tt class="py-op">(</tt><tt class="py-name">instances</tt><tt class="py-op">,</tt> <tt id="link-126" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-126', 'alphabet', 'link-14');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line"> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line"> </tt>
<a name="__skip_unused_lines"></a><div id="__skip_unused_lines-def"><a name="L297"></a><tt class="py-lineno">297</tt> <a class="py-toggle" href="#" id="__skip_unused_lines-toggle" onclick="return toggle('__skip_unused_lines');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.meme-module.html#__skip_unused_lines">__skip_unused_lines</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="__skip_unused_lines-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="__skip_unused_lines-expanded"><a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-127" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-127', 'startswith', 'link-13');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'log-odds matrix'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line">            <tt class="py-keyword">break</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected end of stream: Expected to find line starting with 'log-odds matrix'"</tt><tt class="py-op">)</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-128" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-128', 'startswith', 'link-13');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'---'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line">            <tt class="py-keyword">break</tt> </tt>
<a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected end of stream: Expected to find line starting with '---'"</tt><tt class="py-op">)</tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-129" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-129', 'startswith', 'link-13');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'letter-probability matrix'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line">            <tt class="py-keyword">break</tt> </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected end of stream: Expected to find line starting with 'letter-probability matrix'"</tt><tt class="py-op">)</tt> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-130" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-130', 'startswith', 'link-13');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'---'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line">            <tt class="py-keyword">break</tt> </tt>
<a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected end of stream: Expected to find line starting with '---'"</tt><tt class="py-op">)</tt> </tt>
<a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-131" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-131', 'startswith', 'link-13');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Time'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line">            <tt class="py-keyword">break</tt> </tt>
<a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected end of stream: Expected to find line starting with 'Time'"</tt><tt class="py-op">)</tt> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-132" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-132', 'next', 'link-32');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">'Unexpected end of stream: Expected to find blank line'</tt><tt class="py-op">)</tt> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-133" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-133', 'strip', 'link-51');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Expected blank line, but got:\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-134" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-134', 'next', 'link-32');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected end of stream: Expected to find line starting with '***'"</tt><tt class="py-op">)</tt> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-135" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-135', 'startswith', 'link-13');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'***'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Line does not start with '***':\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-136" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-136', 'strip', 'link-51');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line">            <tt class="py-keyword">break</tt> </tt>
<a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected end of stream: Expected to find line starting with '***'"</tt><tt class="py-op">)</tt> </tt>
<a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-137" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-137', 'startswith', 'link-13');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'***'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Line does not start with '***':\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
</div><a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:05:00 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
