<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.SeqFeature.CompoundLocation</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.SeqFeature-module.html">Module&nbsp;SeqFeature</a> ::
        Class&nbsp;CompoundLocation
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.SeqFeature.CompoundLocation-class.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== CLASS DESCRIPTION ==================== -->
<h1 class="epydoc">Class CompoundLocation</h1><p class="nomargin-top"><span class="codelink"><a href="Bio.SeqFeature-pysrc.html#CompoundLocation">source&nbsp;code</a></span></p>
<pre class="base-tree">
object --+
         |
        <strong class="uidshort">CompoundLocation</strong>
</pre>

<hr />
For handling joins etc where a feature location has several parts.

<!-- ==================== INSTANCE METHODS ==================== -->
<a name="section-InstanceMethods"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Instance Methods</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-InstanceMethods"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqFeature.CompoundLocation-class.html#__init__" class="summary-sig-name">__init__</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">parts</span>,
        <span class="summary-sig-arg">operator</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">join</code><code class="variable-quote">'</code></span>)</span><br />
      Initialize the class.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#CompoundLocation.__init__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqFeature.CompoundLocation-class.html#__str__" class="summary-sig-name">__str__</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Return a representation of the CompoundLocation object (with python counting).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#CompoundLocation.__str__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqFeature.CompoundLocation-class.html#__repr__" class="summary-sig-name">__repr__</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Represent the CompoundLocation object as string for debugging.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#CompoundLocation.__repr__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_get_strand"></a><span class="summary-sig-name">_get_strand</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Get function for the strand property (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#CompoundLocation._get_strand">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_set_strand"></a><span class="summary-sig-name">_set_strand</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">value</span>)</span><br />
      Set function for the strand property (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#CompoundLocation._set_strand">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqFeature.CompoundLocation-class.html#__add__" class="summary-sig-name">__add__</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">other</span>)</span><br />
      Combine locations, or shift the location by an integer offset.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#CompoundLocation.__add__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__radd__"></a><span class="summary-sig-name">__radd__</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">other</span>)</span><br />
      Add a feature to the left.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#CompoundLocation.__radd__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__contains__"></a><span class="summary-sig-name">__contains__</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">value</span>)</span><br />
      Check if an integer position is within the CompoundLocation object.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#CompoundLocation.__contains__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqFeature.CompoundLocation-class.html#__nonzero__" class="summary-sig-name">__nonzero__</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Return True regardless of the length of the feature.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#CompoundLocation.__nonzero__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__len__"></a><span class="summary-sig-name">__len__</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Return the length of the CompoundLocation object.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#CompoundLocation.__len__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__iter__"></a><span class="summary-sig-name">__iter__</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Iterate over the parent positions within the CompoundLocation object.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#CompoundLocation.__iter__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__eq__"></a><span class="summary-sig-name">__eq__</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">other</span>)</span><br />
      Check if all parts of CompoundLocation are equal to all parts of other CompoundLocation.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#CompoundLocation.__eq__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__ne__"></a><span class="summary-sig-name">__ne__</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">other</span>)</span><br />
      Implement the not-equal operand.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#CompoundLocation.__ne__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_shift"></a><span class="summary-sig-name">_shift</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">offset</span>)</span><br />
      Return a copy of the CompoundLocation shifted by an offset (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#CompoundLocation._shift">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqFeature.CompoundLocation-class.html#_flip" class="summary-sig-name" onclick="show_private();">_flip</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">length</span>)</span><br />
      Return a copy of the locations after the parent is reversed (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#CompoundLocation._flip">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqFeature.CompoundLocation-class.html#extract" class="summary-sig-name">extract</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">parent_sequence</span>)</span><br />
      Extract the sequence from supplied parent sequence using the CompoundLocation object.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqFeature-pysrc.html#CompoundLocation.extract">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__delattr__</code>,
      <code>__format__</code>,
      <code>__getattribute__</code>,
      <code>__hash__</code>,
      <code>__new__</code>,
      <code>__reduce__</code>,
      <code>__reduce_ex__</code>,
      <code>__setattr__</code>,
      <code>__sizeof__</code>,
      <code>__subclasshook__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== PROPERTIES ==================== -->
<a name="section-Properties"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Properties</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Properties"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.SeqFeature.CompoundLocation-class.html#strand" class="summary-name">strand</a><br />
      Overall strand of the compound location.
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.SeqFeature.CompoundLocation-class.html#start" class="summary-name">start</a><br />
      Start location - left most (minimum) value, regardless of strand.
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.SeqFeature.CompoundLocation-class.html#end" class="summary-name">end</a><br />
      End location - right most (maximum) value, regardless of strand.
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.SeqFeature.CompoundLocation-class.html#nofuzzy_start" class="summary-name">nofuzzy_start</a><br />
      Start position (integer, approximated if fuzzy, read only) (OBSOLETE).
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.SeqFeature.CompoundLocation-class.html#nofuzzy_end" class="summary-name">nofuzzy_end</a><br />
      End position (integer, approximated if fuzzy, read only) (OBSOLETE).
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.SeqFeature.CompoundLocation-class.html#ref" class="summary-name">ref</a><br />
      Not present in CompoundLocation, dummy method for API compatibility.
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.SeqFeature.CompoundLocation-class.html#ref_db" class="summary-name">ref_db</a><br />
      Not present in CompoundLocation, dummy method for API compatibility.
    </td>
  </tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__class__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== METHOD DETAILS ==================== -->
<a name="section-MethodDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Method Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-MethodDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="__init__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__init__</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">parts</span>,
        <span class="sig-arg">operator</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">join</code><code class="variable-quote">'</code></span>)</span>
    <br /><em class="fname">(Constructor)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqFeature-pysrc.html#CompoundLocation.__init__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Initialize the class.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqFeature <span class="py-keyword">import</span> FeatureLocation, CompoundLocation
<span class="py-prompt">&gt;&gt;&gt; </span>f1 = FeatureLocation(10, 40, strand=+1)
<span class="py-prompt">&gt;&gt;&gt; </span>f2 = FeatureLocation(50, 59, strand=+1)
<span class="py-prompt">&gt;&gt;&gt; </span>f = CompoundLocation([f1, f2])
<span class="py-prompt">&gt;&gt;&gt; </span>len(f) == len(f1) + len(f2) == 39 == len(list(f))
<span class="py-output">True</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(f.operator)
<span class="py-output">join</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>5 <span class="py-keyword">in</span> f
<span class="py-output">False</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>15 <span class="py-keyword">in</span> f
<span class="py-output">True</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>f.strand
<span class="py-output">1</span></pre>
<p>Notice that the strand of the compound location is computed
automatically - in the case of mixed strands on the sub-locations
the overall strand is set to None.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>f = CompoundLocation([FeatureLocation(3, 6, strand=+1),
<span class="py-more">... </span>                      FeatureLocation(10, 13, strand=-1)])
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(f.strand)
<span class="py-output">None</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>len(f)
<span class="py-output">6</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>list(f)
<span class="py-output">[3, 4, 5, 12, 11, 10]</span></pre>
<p>The example above doing list(f) iterates over the coordinates within the
feature. This allows you to use max and min on the location, to find the
range covered:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>min(f)
<span class="py-output">3</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>max(f)
<span class="py-output">12</span></pre>
<p>More generally, you can use the compound location's start and end which
give the full range covered, 0 &lt;= start &lt;= end &lt;= full sequence length.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>f.start == min(f)
<span class="py-output">True</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>f.end == max(f) + 1
<span class="py-output">True</span></pre>
<p>This is consistent with the behaviour of the simple FeatureLocation for
a single region, where again the 'start' and 'end' do not necessarily
give the biological start and end, but rather the 'minimal' and 'maximal'
coordinate boundaries.</p>
<p>Note that adding locations provides a more intuitive method of
construction:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>f = FeatureLocation(3, 6, strand=+1) + FeatureLocation(10, 13, strand=-1)
<span class="py-prompt">&gt;&gt;&gt; </span>len(f)
<span class="py-output">6</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>list(f)
<span class="py-output">[3, 4, 5, 12, 11, 10]</span></pre>
  <dl class="fields">
    <dt>Overrides:
        object.__init__
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="__str__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__str__</span>(<span class="sig-arg">self</span>)</span>
    <br /><em class="fname">(Informal representation operator)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqFeature-pysrc.html#CompoundLocation.__str__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  Return a representation of the CompoundLocation object (with python counting).
  <dl class="fields">
    <dt>Overrides:
        object.__str__
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="__repr__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__repr__</span>(<span class="sig-arg">self</span>)</span>
    <br /><em class="fname">(Representation operator)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqFeature-pysrc.html#CompoundLocation.__repr__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  Represent the CompoundLocation object as string for debugging.
  <dl class="fields">
    <dt>Overrides:
        object.__repr__
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="__add__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__add__</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">other</span>)</span>
    <br /><em class="fname">(Addition operator)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqFeature-pysrc.html#CompoundLocation.__add__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Combine locations, or shift the location by an integer offset.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqFeature <span class="py-keyword">import</span> FeatureLocation, CompoundLocation
<span class="py-prompt">&gt;&gt;&gt; </span>f1 = FeatureLocation(15, 17) + FeatureLocation(20, 30)
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(f1)
<span class="py-output">join{[15:17], [20:30]}</span></pre>
<p>You can add another FeatureLocation:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(f1 + FeatureLocation(40, 50))
<span class="py-output">join{[15:17], [20:30], [40:50]}</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(FeatureLocation(5, 10) + f1)
<span class="py-output">join{[5:10], [15:17], [20:30]}</span></pre>
<p>You can also add another CompoundLocation:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>f2 = FeatureLocation(40, 50) + FeatureLocation(60, 70)
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(f2)
<span class="py-output">join{[40:50], [60:70]}</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(f1 + f2)
<span class="py-output">join{[15:17], [20:30], [40:50], [60:70]}</span></pre>
<p>Also, as with the FeatureLocation, adding an integer shifts the
location's co-ordinates by that offset:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(f1 + 100)
<span class="py-output">join{[115:117], [120:130]}</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(200 + f1)
<span class="py-output">join{[215:217], [220:230]}</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(f1 + (-5))
<span class="py-output">join{[10:12], [15:25]}</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="__nonzero__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__nonzero__</span>(<span class="sig-arg">self</span>)</span>
    <br /><em class="fname">(Boolean test operator)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqFeature-pysrc.html#CompoundLocation.__nonzero__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Return True regardless of the length of the feature.</p>
<p>This behaviour is for backwards compatibility, since until the
__len__ method was added, a FeatureLocation always evaluated as True.</p>
<p>Note that in comparison, Seq objects, strings, lists, etc, will all
evaluate to False if they have length zero.</p>
<p>WARNING: The FeatureLocation may in future evaluate to False when its
length is zero (in order to better match normal python behaviour)!</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_flip"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_flip</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">length</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqFeature-pysrc.html#CompoundLocation._flip">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Return a copy of the locations after the parent is reversed (PRIVATE).</p>
<p>Note that the order of the parts is NOT reversed too. Consider a CDS
on the forward strand with exons small, medium and large (in length).
Once we change the frame of reference to the reverse complement strand,
the start codon is still part of the small exon, and the stop codon
still part of the large exon - so the part order remains the same!</p>
<p>Here is an artificial example, were the features map to the two upper
case regions and the lower case runs of n are not used:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqFeature <span class="py-keyword">import</span> FeatureLocation
<span class="py-prompt">&gt;&gt;&gt; </span>dna = Seq(<span class="py-string">&quot;nnnnnAGCATCCTGCTGTACnnnnnnnnGAGAMTGCCATGCCCCTGGAGTGAnnnnn&quot;</span>)
<span class="py-prompt">&gt;&gt;&gt; </span>small = FeatureLocation(5, 20, strand=1)
<span class="py-prompt">&gt;&gt;&gt; </span>large = FeatureLocation(28, 52, strand=1)
<span class="py-prompt">&gt;&gt;&gt; </span>location = small + large
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(small)
<span class="py-output">[5:20](+)</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(large)
<span class="py-output">[28:52](+)</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(location)
<span class="py-output">join{[5:20](+), [28:52](+)}</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">for</span> part <span class="py-keyword">in</span> location.parts:
<span class="py-more">... </span>    <span class="py-keyword">print</span>(len(part))
<span class="py-more">...</span>
<span class="py-output">15</span>
<span class="py-output">24</span></pre>
<p>As you can see, this is a silly example where each &quot;exon&quot; is a word:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(small.extract(dna).translate())
<span class="py-output">SILLY</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(large.extract(dna).translate())
<span class="py-output">EXAMPLE*</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(location.extract(dna).translate())
<span class="py-output">SILLYEXAMPLE*</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">for</span> part <span class="py-keyword">in</span> location.parts:
<span class="py-more">... </span>    <span class="py-keyword">print</span>(part.extract(dna).translate())
<span class="py-more">...</span>
<span class="py-output">SILLY</span>
<span class="py-output">EXAMPLE*</span></pre>
<p>Now, let's look at this from the reverse strand frame of reference:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>flipped_dna = dna.reverse_complement()
<span class="py-prompt">&gt;&gt;&gt; </span>flipped_location = location._flip(len(dna))
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(flipped_location.extract(flipped_dna).translate())
<span class="py-output">SILLYEXAMPLE*</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">for</span> part <span class="py-keyword">in</span> flipped_location.parts:
<span class="py-more">... </span>    <span class="py-keyword">print</span>(part.extract(flipped_dna).translate())
<span class="py-more">...</span>
<span class="py-output">SILLY</span>
<span class="py-output">EXAMPLE*</span></pre>
<p>The key point here is the first part of the CompoundFeature is still the
small exon, while the second part is still the large exon:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">for</span> part <span class="py-keyword">in</span> flipped_location.parts:
<span class="py-more">... </span>    <span class="py-keyword">print</span>(len(part))
<span class="py-more">...</span>
<span class="py-output">15</span>
<span class="py-output">24</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(flipped_location)
<span class="py-output">join{[37:52](-), [5:29](-)}</span></pre>
<p>Notice the parts are not reversed. However, there was a bug here in older
versions of Biopython which would have given join{[5:29](-), [37:52](-)}
and the translation would have wrongly been &quot;EXAMPLE*SILLY&quot; instead.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="extract"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">extract</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">parent_sequence</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqFeature-pysrc.html#CompoundLocation.extract">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Extract the sequence from supplied parent sequence using the CompoundLocation object.</p>
<p>The parent_sequence can be a Seq like object or a string, and will
generally return an object of the same type. The exception to this is
a MutableSeq as the parent sequence will return a Seq object.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> generic_protein
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqFeature <span class="py-keyword">import</span> FeatureLocation, CompoundLocation
<span class="py-prompt">&gt;&gt;&gt; </span>seq = Seq(<span class="py-string">&quot;MKQHKAMIVALIVICITAVVAAL&quot;</span>, generic_protein)
<span class="py-prompt">&gt;&gt;&gt; </span>fl1 = FeatureLocation(2, 8)
<span class="py-prompt">&gt;&gt;&gt; </span>fl2 = FeatureLocation(10, 15)
<span class="py-prompt">&gt;&gt;&gt; </span>fl3 = CompoundLocation([fl1,fl2])
<span class="py-prompt">&gt;&gt;&gt; </span>fl3.extract(seq)
<span class="py-output">Seq('QHKAMILIVIC', ProteinAlphabet())</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== PROPERTY DETAILS ==================== -->
<a name="section-PropertyDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Property Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-PropertyDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="strand"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">strand</h3>
  <p>Overall strand of the compound location.</p>
<p>If all the parts have the same strand, that is returned. Otherwise
for mixed strands, this returns None.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqFeature <span class="py-keyword">import</span> FeatureLocation, CompoundLocation
<span class="py-prompt">&gt;&gt;&gt; </span>f1 = FeatureLocation(15, 17, strand=1)
<span class="py-prompt">&gt;&gt;&gt; </span>f2 = FeatureLocation(20, 30, strand=-1)
<span class="py-prompt">&gt;&gt;&gt; </span>f = f1 + f2
<span class="py-prompt">&gt;&gt;&gt; </span>f1.strand
<span class="py-output">1</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>f2.strand
<span class="py-output">-1</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>f.strand
<span class="py-prompt">&gt;&gt;&gt; </span>f.strand <span class="py-keyword">is</span> None
<span class="py-output">True</span></pre>
<p>If you set the strand of a CompoundLocation, this is applied to
all the parts - use with caution:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>f.strand = 1
<span class="py-prompt">&gt;&gt;&gt; </span>f1.strand
<span class="py-output">1</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>f2.strand
<span class="py-output">1</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>f.strand
<span class="py-output">1</span></pre>
  <dl class="fields">
    <dt>Get Method:</dt>
    <dd class="value"><span class="summary-sig"><a href="Bio.SeqFeature.CompoundLocation-class.html#_get_strand" class="summary-sig-name" onclick="show_private();">_get_strand</a>(<span class="summary-sig-arg">self</span>)</span>
        - Get function for the strand property (PRIVATE).
    </dd>
    <dt>Set Method:</dt>
    <dd class="value"><span class="summary-sig"><a href="Bio.SeqFeature.CompoundLocation-class.html#_set_strand" class="summary-sig-name" onclick="show_private();">_set_strand</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">value</span>)</span>
        - Set function for the strand property (PRIVATE).
    </dd>
  </dl>
</td></tr></table>
</div>
<a name="start"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">start</h3>
  <p>Start location - left most (minimum) value, regardless of strand.</p>
<p>Read only, returns an integer like position object, possibly a fuzzy
position.</p>
<p>For the special case of a CompoundLocation wrapping the origin of a
circular genome, this will return zero.</p>
  <dl class="fields">
    <dt>Get Method:</dt>
    <dd class="value"><span class="summary-sig"><i>unreachable</i>.start(<span class="summary-sig-arg">self</span>)</span>
        - Start location - left most (minimum) value, regardless of strand.
    </dd>
  </dl>
</td></tr></table>
</div>
<a name="end"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">end</h3>
  <p>End location - right most (maximum) value, regardless of strand.</p>
<p>Read only, returns an integer like position object, possibly a fuzzy
position.</p>
<p>For the special case of a CompoundLocation wrapping the origin of
a circular genome this will match the genome length (minus one
given how Python counts from zero).</p>
  <dl class="fields">
    <dt>Get Method:</dt>
    <dd class="value"><span class="summary-sig"><i>unreachable</i>.end(<span class="summary-sig-arg">self</span>)</span>
        - End location - right most (maximum) value, regardless of strand.
    </dd>
  </dl>
</td></tr></table>
</div>
<a name="nofuzzy_start"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">nofuzzy_start</h3>
  <p>Start position (integer, approximated if fuzzy, read only) (OBSOLETE).</p>
<p>This is an alias for int(feature.start), which should be used in
preference -- unless you are trying to support old versions of
Biopython.</p>
  <dl class="fields">
    <dt>Get Method:</dt>
    <dd class="value"><span class="summary-sig"><i>unreachable</i>.nofuzzy_start(<span class="summary-sig-arg">self</span>)</span>
        - Start position (integer, approximated if fuzzy, read only) (OBSOLETE).
    </dd>
  </dl>
</td></tr></table>
</div>
<a name="nofuzzy_end"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">nofuzzy_end</h3>
  <p>End position (integer, approximated if fuzzy, read only) (OBSOLETE).</p>
<p>This is an alias for int(feature.end), which should be used in
preference -- unless you are trying to support old versions of
Biopython.</p>
  <dl class="fields">
    <dt>Get Method:</dt>
    <dd class="value"><span class="summary-sig"><i>unreachable</i>.nofuzzy_end(<span class="summary-sig-arg">self</span>)</span>
        - End position (integer, approximated if fuzzy, read only) (OBSOLETE).
    </dd>
  </dl>
</td></tr></table>
</div>
<a name="ref"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">ref</h3>
  Not present in CompoundLocation, dummy method for API compatibility.
  <dl class="fields">
    <dt>Get Method:</dt>
    <dd class="value"><span class="summary-sig"><i>unreachable</i>.ref(<span class="summary-sig-arg">self</span>)</span>
        - Not present in CompoundLocation, dummy method for API compatibility.
    </dd>
  </dl>
</td></tr></table>
</div>
<a name="ref_db"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">ref_db</h3>
  Not present in CompoundLocation, dummy method for API compatibility.
  <dl class="fields">
    <dt>Get Method:</dt>
    <dd class="value"><span class="summary-sig"><i>unreachable</i>.ref_db(<span class="summary-sig-arg">self</span>)</span>
        - Not present in CompoundLocation, dummy method for API compatibility.
    </dd>
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Mon Jul 10 15:16:36 2017
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
