<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.SearchIO.BlastIO.blast_tab</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.SearchIO-module.html">Package&nbsp;SearchIO</a> ::
        <a href="Bio.SearchIO.BlastIO-module.html">Package&nbsp;BlastIO</a> ::
        Module&nbsp;blast_tab
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.SearchIO.BlastIO.blast_tab-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.SearchIO.BlastIO.blast_tab-module.html">Module Bio.SearchIO.BlastIO.blast_tab</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2012 by Wibowo Arindrarto.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># This file is part of the Biopython distribution and governed by your</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># choice of the "Biopython License Agreement" or the "BSD 3-Clause License".</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-comment"># Please see the LICENSE file that should have been included as part of this</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-comment"># package.</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-docstring">"""Bio.SearchIO parser for BLAST+ tab output format, with or without comments."""</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">re</tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-1', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt id="link-2" class="py-name" targets="Function Bio._py3k._as_bytes()=Bio._py3k-module.html#_as_bytes"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-2', '_as_bytes', 'link-2');">_as_bytes</a></tt><tt class="py-op">,</tt> <tt id="link-3" class="py-name" targets="Function Bio._py3k._bytes_to_string()=Bio._py3k-module.html#_bytes_to_string"><a title="Bio._py3k._bytes_to_string" class="py-name" href="#" onclick="return doclink('link-3', '_bytes_to_string', 'link-3');">_bytes_to_string</a></tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-4" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-4', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-5" class="py-name"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-5', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">basestring</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-6" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-6', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-7" class="py-name" targets="Package Bio.SearchIO=Bio.SearchIO-module.html"><a title="Bio.SearchIO" class="py-name" href="#" onclick="return doclink('link-7', 'SearchIO', 'link-7');">SearchIO</a></tt><tt class="py-op">.</tt><tt id="link-8" class="py-name" targets="Module Bio.SearchIO._index=Bio.SearchIO._index-module.html,Module Bio.SeqIO._index=Bio.SeqIO._index-module.html"><a title="Bio.SearchIO._index
Bio.SeqIO._index" class="py-name" href="#" onclick="return doclink('link-8', '_index', 'link-8');">_index</a></tt> <tt class="py-keyword">import</tt> <tt id="link-9" class="py-name" targets="Class Bio.SearchIO._index.SearchIndexer=Bio.SearchIO._index.SearchIndexer-class.html"><a title="Bio.SearchIO._index.SearchIndexer" class="py-name" href="#" onclick="return doclink('link-9', 'SearchIndexer', 'link-9');">SearchIndexer</a></tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-10" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-10', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-11" class="py-name"><a title="Bio.SearchIO" class="py-name" href="#" onclick="return doclink('link-11', 'SearchIO', 'link-7');">SearchIO</a></tt><tt class="py-op">.</tt><tt id="link-12" class="py-name" targets="Package Bio.SearchIO._model=Bio.SearchIO._model-module.html"><a title="Bio.SearchIO._model" class="py-name" href="#" onclick="return doclink('link-12', '_model', 'link-12');">_model</a></tt> <tt class="py-keyword">import</tt> <tt id="link-13" class="py-name" targets="Class Bio.SearchIO._model.query.QueryResult=Bio.SearchIO._model.query.QueryResult-class.html"><a title="Bio.SearchIO._model.query.QueryResult" class="py-name" href="#" onclick="return doclink('link-13', 'QueryResult', 'link-13');">QueryResult</a></tt><tt class="py-op">,</tt> <tt id="link-14" class="py-name" targets="Class Bio.SearchIO._model.hit.Hit=Bio.SearchIO._model.hit.Hit-class.html"><a title="Bio.SearchIO._model.hit.Hit" class="py-name" href="#" onclick="return doclink('link-14', 'Hit', 'link-14');">Hit</a></tt><tt class="py-op">,</tt> <tt id="link-15" class="py-name" targets="Class Bio.Blast.Record.HSP=Bio.Blast.Record.HSP-class.html,Class Bio.SearchIO._model.hsp.HSP=Bio.SearchIO._model.hsp.HSP-class.html"><a title="Bio.Blast.Record.HSP
Bio.SearchIO._model.hsp.HSP" class="py-name" href="#" onclick="return doclink('link-15', 'HSP', 'link-15');">HSP</a></tt><tt class="py-op">,</tt> <tt id="link-16" class="py-name" targets="Class Bio.SearchIO._model.hsp.HSPFragment=Bio.SearchIO._model.hsp.HSPFragment-class.html"><a title="Bio.SearchIO._model.hsp.HSPFragment" class="py-name" href="#" onclick="return doclink('link-16', 'HSPFragment', 'link-16');">HSPFragment</a></tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-name">__all__</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-string">'BlastTabIndexer'</tt><tt class="py-op">,</tt> <tt class="py-string">'BlastTabParser'</tt><tt class="py-op">,</tt> <tt class="py-string">'BlastTabWriter'</tt><tt class="py-op">)</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-comment"># longname-shortname map</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-comment"># maps the column names shown in a commented output to its short name</tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-comment"># (the one used in the command line)</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt id="link-17" class="py-name" targets="Variable Bio.SearchIO.BlastIO.blast_tab._LONG_SHORT_MAP=Bio.SearchIO.BlastIO.blast_tab-module.html#_LONG_SHORT_MAP"><a title="Bio.SearchIO.BlastIO.blast_tab._LONG_SHORT_MAP" class="py-name" href="#" onclick="return doclink('link-17', '_LONG_SHORT_MAP', 'link-17');">_LONG_SHORT_MAP</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line">    <tt class="py-string">'query id'</tt><tt class="py-op">:</tt> <tt class="py-string">'qseqid'</tt><tt class="py-op">,</tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line">    <tt class="py-string">'query acc.'</tt><tt class="py-op">:</tt> <tt class="py-string">'qacc'</tt><tt class="py-op">,</tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line">    <tt class="py-string">'query acc.ver'</tt><tt class="py-op">:</tt> <tt class="py-string">'qaccver'</tt><tt class="py-op">,</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line">    <tt class="py-string">'query length'</tt><tt class="py-op">:</tt> <tt class="py-string">'qlen'</tt><tt class="py-op">,</tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line">    <tt class="py-string">'subject id'</tt><tt class="py-op">:</tt> <tt class="py-string">'sseqid'</tt><tt class="py-op">,</tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line">    <tt class="py-string">'subject acc.'</tt><tt class="py-op">:</tt> <tt class="py-string">'sacc'</tt><tt class="py-op">,</tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line">    <tt class="py-string">'subject acc.ver'</tt><tt class="py-op">:</tt> <tt class="py-string">'saccver'</tt><tt class="py-op">,</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line">    <tt class="py-string">'subject length'</tt><tt class="py-op">:</tt> <tt class="py-string">'slen'</tt><tt class="py-op">,</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line">    <tt class="py-string">'alignment length'</tt><tt class="py-op">:</tt> <tt class="py-string">'length'</tt><tt class="py-op">,</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line">    <tt class="py-string">'bit score'</tt><tt class="py-op">:</tt> <tt class="py-string">'bitscore'</tt><tt class="py-op">,</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line">    <tt class="py-string">'score'</tt><tt class="py-op">:</tt> <tt class="py-string">'score'</tt><tt class="py-op">,</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line">    <tt class="py-string">'evalue'</tt><tt class="py-op">:</tt> <tt class="py-string">'evalue'</tt><tt class="py-op">,</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line">    <tt class="py-string">'identical'</tt><tt class="py-op">:</tt> <tt class="py-string">'nident'</tt><tt class="py-op">,</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line">    <tt class="py-string">'% identity'</tt><tt class="py-op">:</tt> <tt class="py-string">'pident'</tt><tt class="py-op">,</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line">    <tt class="py-string">'positives'</tt><tt class="py-op">:</tt> <tt class="py-string">'positive'</tt><tt class="py-op">,</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line">    <tt class="py-string">'% positives'</tt><tt class="py-op">:</tt> <tt class="py-string">'ppos'</tt><tt class="py-op">,</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line">    <tt class="py-string">'mismatches'</tt><tt class="py-op">:</tt> <tt class="py-string">'mismatch'</tt><tt class="py-op">,</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line">    <tt class="py-string">'gaps'</tt><tt class="py-op">:</tt> <tt class="py-string">'gaps'</tt><tt class="py-op">,</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line">    <tt class="py-string">'q. start'</tt><tt class="py-op">:</tt> <tt class="py-string">'qstart'</tt><tt class="py-op">,</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line">    <tt class="py-string">'q. end'</tt><tt class="py-op">:</tt> <tt class="py-string">'qend'</tt><tt class="py-op">,</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line">    <tt class="py-string">'s. start'</tt><tt class="py-op">:</tt> <tt class="py-string">'sstart'</tt><tt class="py-op">,</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line">    <tt class="py-string">'s. end'</tt><tt class="py-op">:</tt> <tt class="py-string">'send'</tt><tt class="py-op">,</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line">    <tt class="py-string">'query frame'</tt><tt class="py-op">:</tt> <tt class="py-string">'qframe'</tt><tt class="py-op">,</tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line">    <tt class="py-string">'sbjct frame'</tt><tt class="py-op">:</tt> <tt class="py-string">'sframe'</tt><tt class="py-op">,</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line">    <tt class="py-string">'query/sbjct frames'</tt><tt class="py-op">:</tt> <tt class="py-string">'frames'</tt><tt class="py-op">,</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line">    <tt class="py-string">'query seq'</tt><tt class="py-op">:</tt> <tt class="py-string">'qseq'</tt><tt class="py-op">,</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line">    <tt class="py-string">'subject seq'</tt><tt class="py-op">:</tt> <tt class="py-string">'sseq'</tt><tt class="py-op">,</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line">    <tt class="py-string">'gap opens'</tt><tt class="py-op">:</tt> <tt class="py-string">'gapopen'</tt><tt class="py-op">,</tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line">    <tt class="py-string">'query gi'</tt><tt class="py-op">:</tt> <tt class="py-string">'qgi'</tt><tt class="py-op">,</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line">    <tt class="py-string">'subject ids'</tt><tt class="py-op">:</tt> <tt class="py-string">'sallseqid'</tt><tt class="py-op">,</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line">    <tt class="py-string">'subject gi'</tt><tt class="py-op">:</tt> <tt class="py-string">'sgi'</tt><tt class="py-op">,</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line">    <tt class="py-string">'subject gis'</tt><tt class="py-op">:</tt> <tt class="py-string">'sallgi'</tt><tt class="py-op">,</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line">    <tt class="py-string">'BTOP'</tt><tt class="py-op">:</tt> <tt class="py-string">'btop'</tt><tt class="py-op">,</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line">    <tt class="py-string">'subject accs.'</tt><tt class="py-op">:</tt> <tt class="py-string">'sallacc'</tt><tt class="py-op">,</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line">    <tt class="py-string">'subject tax ids'</tt><tt class="py-op">:</tt> <tt class="py-string">'staxids'</tt><tt class="py-op">,</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line">    <tt class="py-string">'subject sci names'</tt><tt class="py-op">:</tt> <tt class="py-string">'sscinames'</tt><tt class="py-op">,</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line">    <tt class="py-string">'subject com names'</tt><tt class="py-op">:</tt> <tt class="py-string">'scomnames'</tt><tt class="py-op">,</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line">    <tt class="py-string">'subject blast names'</tt><tt class="py-op">:</tt> <tt class="py-string">'sblastnames'</tt><tt class="py-op">,</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line">    <tt class="py-string">'subject super kingdoms'</tt><tt class="py-op">:</tt> <tt class="py-string">'sskingdoms'</tt><tt class="py-op">,</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line">    <tt class="py-string">'subject title'</tt><tt class="py-op">:</tt> <tt class="py-string">'stitle'</tt><tt class="py-op">,</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line">    <tt class="py-string">'subject titles'</tt><tt class="py-op">:</tt> <tt class="py-string">'salltitles'</tt><tt class="py-op">,</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line">    <tt class="py-string">'subject strand'</tt><tt class="py-op">:</tt> <tt class="py-string">'sstrand'</tt><tt class="py-op">,</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line">    <tt class="py-string">'% subject coverage'</tt><tt class="py-op">:</tt> <tt class="py-string">'qcovs'</tt><tt class="py-op">,</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line">    <tt class="py-string">'% hsp coverage'</tt><tt class="py-op">:</tt> <tt class="py-string">'qcovhsp'</tt><tt class="py-op">,</tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line"><tt class="py-op">}</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line"> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line"><tt class="py-comment"># function to create a list from semicolon-delimited string</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line"><tt class="py-comment"># used in BlastTabParser._parse_result_row</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line"><tt id="link-18" class="py-name" targets="Function Bio.SearchIO.BlastIO.blast_tab._list_semicol()=Bio.SearchIO.BlastIO.blast_tab-module.html#_list_semicol"><a title="Bio.SearchIO.BlastIO.blast_tab._list_semicol" class="py-name" href="#" onclick="return doclink('link-18', '_list_semicol', 'link-18');">_list_semicol</a></tt> <tt class="py-op">=</tt> <tt class="py-keyword">lambda</tt> <tt id="link-19" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.x=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#x,Variable Bio.Statistics.lowess.x=Bio.Statistics.lowess-module.html#x"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-19', 'x', 'link-19');">x</a></tt><tt class="py-op">:</tt> <tt id="link-20" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-20', 'x', 'link-19');">x</a></tt><tt class="py-op">.</tt><tt id="link-21" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-21', 'split', 'link-21');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">';'</tt><tt class="py-op">)</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line"><tt id="link-22" class="py-name" targets="Function Bio.SearchIO.BlastIO.blast_tab._list_diamond()=Bio.SearchIO.BlastIO.blast_tab-module.html#_list_diamond"><a title="Bio.SearchIO.BlastIO.blast_tab._list_diamond" class="py-name" href="#" onclick="return doclink('link-22', '_list_diamond', 'link-22');">_list_diamond</a></tt> <tt class="py-op">=</tt> <tt class="py-keyword">lambda</tt> <tt id="link-23" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-23', 'x', 'link-19');">x</a></tt><tt class="py-op">:</tt> <tt id="link-24" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-24', 'x', 'link-19');">x</a></tt><tt class="py-op">.</tt><tt id="link-25" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-25', 'split', 'link-21');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">'&lt;&gt;'</tt><tt class="py-op">)</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line"><tt class="py-comment"># column to class attribute map</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line"><tt id="link-26" class="py-name" targets="Variable Bio.SearchIO.BlastIO.blast_tab._COLUMN_QRESULT=Bio.SearchIO.BlastIO.blast_tab-module.html#_COLUMN_QRESULT"><a title="Bio.SearchIO.BlastIO.blast_tab._COLUMN_QRESULT" class="py-name" href="#" onclick="return doclink('link-26', '_COLUMN_QRESULT', 'link-26');">_COLUMN_QRESULT</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line">    <tt class="py-string">'qseqid'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'id'</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line">    <tt class="py-string">'qacc'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'accession'</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line">    <tt class="py-string">'qaccver'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'accession_version'</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">    <tt class="py-string">'qlen'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'seq_len'</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line">    <tt class="py-string">'qgi'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'gi'</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line"><tt class="py-op">}</tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line"><tt id="link-27" class="py-name" targets="Variable Bio.SearchIO.BlastIO.blast_tab._COLUMN_HIT=Bio.SearchIO.BlastIO.blast_tab-module.html#_COLUMN_HIT"><a title="Bio.SearchIO.BlastIO.blast_tab._COLUMN_HIT" class="py-name" href="#" onclick="return doclink('link-27', '_COLUMN_HIT', 'link-27');">_COLUMN_HIT</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line">    <tt class="py-string">'sseqid'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'id'</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line">    <tt class="py-string">'sallseqid'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'id_all'</tt><tt class="py-op">,</tt> <tt id="link-28" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._list_semicol" class="py-name" href="#" onclick="return doclink('link-28', '_list_semicol', 'link-18');">_list_semicol</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line">    <tt class="py-string">'sacc'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'accession'</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line">    <tt class="py-string">'saccver'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'accession_version'</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line">    <tt class="py-string">'sallacc'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'accession_all'</tt><tt class="py-op">,</tt> <tt id="link-29" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._list_semicol" class="py-name" href="#" onclick="return doclink('link-29', '_list_semicol', 'link-18');">_list_semicol</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line">    <tt class="py-string">'sgi'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'gi'</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">    <tt class="py-string">'sallgi'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'gi_all'</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line">    <tt class="py-string">'slen'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'seq_len'</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">    <tt class="py-string">'staxids'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'tax_ids'</tt><tt class="py-op">,</tt> <tt id="link-30" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._list_semicol" class="py-name" href="#" onclick="return doclink('link-30', '_list_semicol', 'link-18');">_list_semicol</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line">    <tt class="py-string">'sscinames'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'sci_names'</tt><tt class="py-op">,</tt> <tt id="link-31" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._list_semicol" class="py-name" href="#" onclick="return doclink('link-31', '_list_semicol', 'link-18');">_list_semicol</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line">    <tt class="py-string">'scomnames'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'com_names'</tt><tt class="py-op">,</tt> <tt id="link-32" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._list_semicol" class="py-name" href="#" onclick="return doclink('link-32', '_list_semicol', 'link-18');">_list_semicol</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line">    <tt class="py-string">'sblastnames'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'blast_names'</tt><tt class="py-op">,</tt> <tt id="link-33" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._list_semicol" class="py-name" href="#" onclick="return doclink('link-33', '_list_semicol', 'link-18');">_list_semicol</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line">    <tt class="py-string">'sskingdoms'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'super_kingdoms'</tt><tt class="py-op">,</tt> <tt id="link-34" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._list_semicol" class="py-name" href="#" onclick="return doclink('link-34', '_list_semicol', 'link-18');">_list_semicol</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line">    <tt class="py-string">'stitle'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'title'</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">    <tt class="py-string">'salltitles'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'title_all'</tt><tt class="py-op">,</tt> <tt id="link-35" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._list_diamond" class="py-name" href="#" onclick="return doclink('link-35', '_list_diamond', 'link-22');">_list_diamond</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line">    <tt class="py-comment"># set strand as HSP property?</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line">    <tt class="py-string">'sstrand'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'strand'</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line">    <tt class="py-string">'qcovs'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'query_coverage'</tt><tt class="py-op">,</tt> <tt class="py-name">float</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line"><tt class="py-op">}</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line"><tt id="link-36" class="py-name" targets="Variable Bio.SearchIO.BlastIO.blast_tab._COLUMN_HSP=Bio.SearchIO.BlastIO.blast_tab-module.html#_COLUMN_HSP"><a title="Bio.SearchIO.BlastIO.blast_tab._COLUMN_HSP" class="py-name" href="#" onclick="return doclink('link-36', '_COLUMN_HSP', 'link-36');">_COLUMN_HSP</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">    <tt class="py-string">'bitscore'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'bitscore'</tt><tt class="py-op">,</tt> <tt class="py-name">float</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">    <tt class="py-string">'score'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'bitscore_raw'</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line">    <tt class="py-string">'evalue'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'evalue'</tt><tt class="py-op">,</tt> <tt class="py-name">float</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line">    <tt class="py-string">'nident'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'ident_num'</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">    <tt class="py-string">'pident'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'ident_pct'</tt><tt class="py-op">,</tt> <tt class="py-name">float</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">    <tt class="py-string">'positive'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'pos_num'</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">    <tt class="py-string">'ppos'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'pos_pct'</tt><tt class="py-op">,</tt> <tt class="py-name">float</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">    <tt class="py-string">'mismatch'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'mismatch_num'</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">    <tt class="py-string">'gaps'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'gap_num'</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">    <tt class="py-string">'gapopen'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'gapopen_num'</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">    <tt class="py-string">'btop'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'btop'</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">    <tt class="py-string">'qcovhsp'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'query_coverage'</tt><tt class="py-op">,</tt> <tt class="py-name">float</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line"><tt class="py-op">}</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line"><tt id="link-37" class="py-name" targets="Variable Bio.SearchIO.BlastIO.blast_tab._COLUMN_FRAG=Bio.SearchIO.BlastIO.blast_tab-module.html#_COLUMN_FRAG"><a title="Bio.SearchIO.BlastIO.blast_tab._COLUMN_FRAG" class="py-name" href="#" onclick="return doclink('link-37', '_COLUMN_FRAG', 'link-37');">_COLUMN_FRAG</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line">    <tt class="py-string">'length'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'aln_span'</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">    <tt class="py-string">'qstart'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'query_start'</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">    <tt class="py-string">'qend'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'query_end'</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">    <tt class="py-string">'sstart'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'hit_start'</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">    <tt class="py-string">'send'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'hit_end'</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">    <tt class="py-string">'qframe'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'query_frame'</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line">    <tt class="py-string">'sframe'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'hit_frame'</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line">    <tt class="py-string">'frames'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'frames'</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line">    <tt class="py-string">'qseq'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'query'</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line">    <tt class="py-string">'sseq'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-string">'hit'</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line"><tt class="py-op">}</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line"><tt id="link-38" class="py-name" targets="Variable Bio.SearchIO.BlastIO.blast_tab._SUPPORTED_FIELDS=Bio.SearchIO.BlastIO.blast_tab-module.html#_SUPPORTED_FIELDS"><a title="Bio.SearchIO.BlastIO.blast_tab._SUPPORTED_FIELDS" class="py-name" href="#" onclick="return doclink('link-38', '_SUPPORTED_FIELDS', 'link-38');">_SUPPORTED_FIELDS</a></tt> <tt class="py-op">=</tt> <tt id="link-39" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.set()=Bio.Nexus.Nexus.StepMatrix-class.html#set,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.set()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#set,Method Bio.SVDSuperimposer.SVDSuperimposer.set()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#set"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-39', 'set', 'link-39');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-40" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._COLUMN_QRESULT" class="py-name" href="#" onclick="return doclink('link-40', '_COLUMN_QRESULT', 'link-26');">_COLUMN_QRESULT</a></tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-41" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._COLUMN_HIT" class="py-name" href="#" onclick="return doclink('link-41', '_COLUMN_HIT', 'link-27');">_COLUMN_HIT</a></tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line">                        <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-42" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._COLUMN_HSP" class="py-name" href="#" onclick="return doclink('link-42', '_COLUMN_HSP', 'link-36');">_COLUMN_HSP</a></tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-43" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._COLUMN_FRAG" class="py-name" href="#" onclick="return doclink('link-43', '_COLUMN_FRAG', 'link-37');">_COLUMN_FRAG</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line"> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line"><tt class="py-comment"># column order in the non-commented tabular output variant</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line"><tt class="py-comment"># values must be keys inside the column-attribute maps above</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line"><tt id="link-44" class="py-name" targets="Variable Bio.SearchIO.BlastIO.blast_tab._DEFAULT_FIELDS=Bio.SearchIO.BlastIO.blast_tab-module.html#_DEFAULT_FIELDS"><a title="Bio.SearchIO.BlastIO.blast_tab._DEFAULT_FIELDS" class="py-name" href="#" onclick="return doclink('link-44', '_DEFAULT_FIELDS', 'link-44');">_DEFAULT_FIELDS</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">'qseqid'</tt><tt class="py-op">,</tt> <tt class="py-string">'sseqid'</tt><tt class="py-op">,</tt> <tt class="py-string">'pident'</tt><tt class="py-op">,</tt> <tt class="py-string">'length'</tt><tt class="py-op">,</tt> <tt class="py-string">'mismatch'</tt><tt class="py-op">,</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">        <tt class="py-string">'gapopen'</tt><tt class="py-op">,</tt> <tt class="py-string">'qstart'</tt><tt class="py-op">,</tt> <tt class="py-string">'qend'</tt><tt class="py-op">,</tt> <tt class="py-string">'sstart'</tt><tt class="py-op">,</tt> <tt class="py-string">'send'</tt><tt class="py-op">,</tt> <tt class="py-string">'evalue'</tt><tt class="py-op">,</tt> <tt class="py-string">'bitscore'</tt><tt class="py-op">]</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line"><tt class="py-comment"># one field from each of the following sets must exist in order for the</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line"><tt class="py-comment"># parser to work</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line"><tt id="link-45" class="py-name" targets="Variable Bio.SearchIO.BlastIO.blast_tab._MIN_QUERY_FIELDS=Bio.SearchIO.BlastIO.blast_tab-module.html#_MIN_QUERY_FIELDS"><a title="Bio.SearchIO.BlastIO.blast_tab._MIN_QUERY_FIELDS" class="py-name" href="#" onclick="return doclink('link-45', '_MIN_QUERY_FIELDS', 'link-45');">_MIN_QUERY_FIELDS</a></tt> <tt class="py-op">=</tt> <tt id="link-46" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-46', 'set', 'link-39');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-string">'qseqid'</tt><tt class="py-op">,</tt> <tt class="py-string">'qacc'</tt><tt class="py-op">,</tt> <tt class="py-string">'qaccver'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line"><tt id="link-47" class="py-name" targets="Variable Bio.SearchIO.BlastIO.blast_tab._MIN_HIT_FIELDS=Bio.SearchIO.BlastIO.blast_tab-module.html#_MIN_HIT_FIELDS"><a title="Bio.SearchIO.BlastIO.blast_tab._MIN_HIT_FIELDS" class="py-name" href="#" onclick="return doclink('link-47', '_MIN_HIT_FIELDS', 'link-47');">_MIN_HIT_FIELDS</a></tt> <tt class="py-op">=</tt> <tt id="link-48" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-48', 'set', 'link-39');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-string">'sseqid'</tt><tt class="py-op">,</tt> <tt class="py-string">'sacc'</tt><tt class="py-op">,</tt> <tt class="py-string">'saccver'</tt><tt class="py-op">,</tt> <tt class="py-string">'sallseqid'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line"> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line"><tt class="py-comment"># simple function to create BLAST HSP attributes that may be computed if</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line"><tt class="py-comment"># other certain attributes are present</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line"><tt class="py-comment"># This was previously implemented in the HSP objects in the old model</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line"> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line"><tt id="link-49" class="py-name" targets="Variable Bio.SearchIO.BlastIO.blast_tab._RE_GAPOPEN=Bio.SearchIO.BlastIO.blast_tab-module.html#_RE_GAPOPEN"><a title="Bio.SearchIO.BlastIO.blast_tab._RE_GAPOPEN" class="py-name" href="#" onclick="return doclink('link-49', '_RE_GAPOPEN', 'link-49');">_RE_GAPOPEN</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'\w-'</tt><tt class="py-op">)</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line"> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line"> </tt>
<a name="_compute_gapopen_num"></a><div id="_compute_gapopen_num-def"><a name="L149"></a><tt class="py-lineno">149</tt> <a class="py-toggle" href="#" id="_compute_gapopen_num-toggle" onclick="return toggle('_compute_gapopen_num');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab-module.html#_compute_gapopen_num">_compute_gapopen_num</a><tt class="py-op">(</tt><tt class="py-param">hsp</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_compute_gapopen_num-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_compute_gapopen_num-expanded"><a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return the number of gap openings in the given HSP (PRIVATE)."""</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">    <tt class="py-name">gapopen</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">seq_type</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'query'</tt><tt class="py-op">,</tt> <tt class="py-string">'hit'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">        <tt id="link-50" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.seq=Bio.SeqRecord.SeqRecord-class.html#seq,Variable BioSQL.BioSeq.DBSeqRecord.seq=BioSQL.BioSeq.DBSeqRecord-class.html#seq"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-50', 'seq', 'link-50');">seq</a></tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt id="link-51" class="py-name" targets="Module Bio.SearchIO._model.hsp=Bio.SearchIO._model.hsp-module.html"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-51', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">,</tt> <tt class="py-name">seq_type</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-52" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-52', 'seq', 'link-50');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line">        <tt class="py-name">gapopen</tt> <tt class="py-op">+=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">findall</tt><tt class="py-op">(</tt><tt id="link-53" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._RE_GAPOPEN" class="py-name" href="#" onclick="return doclink('link-53', '_RE_GAPOPEN', 'link-49');">_RE_GAPOPEN</a></tt><tt class="py-op">,</tt> <tt id="link-54" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-54', 'seq', 'link-50');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">gapopen</tt> </tt>
</div><a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line"> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line"> </tt>
<a name="_augment_blast_hsp"></a><div id="_augment_blast_hsp-def"><a name="L158"></a><tt class="py-lineno">158</tt> <a class="py-toggle" href="#" id="_augment_blast_hsp-toggle" onclick="return toggle('_augment_blast_hsp');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab-module.html#_augment_blast_hsp">_augment_blast_hsp</a><tt class="py-op">(</tt><tt class="py-param">hsp</tt><tt class="py-op">,</tt> <tt class="py-param">attr</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_augment_blast_hsp-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_augment_blast_hsp-expanded"><a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">    <tt class="py-docstring">"""Calculate the given HSP attribute, for writing (PRIVATE)."""</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">attr</tt> <tt class="py-op">==</tt> <tt class="py-string">'aln_span'</tt><tt class="py-op">:</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">        <tt class="py-comment"># aln_span is number of identical matches + mismatches + gaps</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">        <tt class="py-name">func</tt> <tt class="py-op">=</tt> <tt class="py-keyword">lambda</tt> <tt id="link-55" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-55', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">:</tt> <tt id="link-56" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-56', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">ident_num</tt> <tt class="py-op">+</tt> <tt id="link-57" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-57', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">mismatch_num</tt> <tt class="py-op">+</tt> <tt id="link-58" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-58', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">gap_num</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">    <tt class="py-comment"># ident and gap will require the num values to be computed first</tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">attr</tt><tt class="py-op">.</tt><tt id="link-59" class="py-name" targets="Method Bio.Seq.Seq.startswith()=Bio.Seq.Seq-class.html#startswith"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-59', 'startswith', 'link-59');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'ident'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">        <tt class="py-name">func</tt> <tt class="py-op">=</tt> <tt class="py-keyword">lambda</tt> <tt id="link-60" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-60', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">:</tt> <tt id="link-61" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-61', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-62" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.aln_span=Bio.SearchIO._model.hsp.HSP-class.html#aln_span,Variable Bio.SearchIO._model.hsp.HSPFragment.aln_span=Bio.SearchIO._model.hsp.HSPFragment-class.html#aln_span"><a title="Bio.SearchIO._model.hsp.HSP.aln_span
Bio.SearchIO._model.hsp.HSPFragment.aln_span" class="py-name" href="#" onclick="return doclink('link-62', 'aln_span', 'link-62');">aln_span</a></tt> <tt class="py-op">-</tt> <tt id="link-63" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-63', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">mismatch_num</tt> <tt class="py-op">-</tt> <tt id="link-64" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-64', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">gap_num</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">attr</tt><tt class="py-op">.</tt><tt id="link-65" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-65', 'startswith', 'link-59');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'gap'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">        <tt class="py-name">func</tt> <tt class="py-op">=</tt> <tt class="py-keyword">lambda</tt> <tt id="link-66" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-66', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">:</tt> <tt id="link-67" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-67', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-68" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.aln_span
Bio.SearchIO._model.hsp.HSPFragment.aln_span" class="py-name" href="#" onclick="return doclink('link-68', 'aln_span', 'link-62');">aln_span</a></tt> <tt class="py-op">-</tt> <tt id="link-69" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-69', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">ident_num</tt> <tt class="py-op">-</tt> <tt id="link-70" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-70', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">mismatch_num</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">attr</tt> <tt class="py-op">==</tt> <tt class="py-string">'mismatch_num'</tt><tt class="py-op">:</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line">        <tt class="py-name">func</tt> <tt class="py-op">=</tt> <tt class="py-keyword">lambda</tt> <tt id="link-71" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-71', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">:</tt> <tt id="link-72" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-72', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-73" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.aln_span
Bio.SearchIO._model.hsp.HSPFragment.aln_span" class="py-name" href="#" onclick="return doclink('link-73', 'aln_span', 'link-62');">aln_span</a></tt> <tt class="py-op">-</tt> <tt id="link-74" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-74', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">ident_num</tt> <tt class="py-op">-</tt> <tt id="link-75" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-75', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">gap_num</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">attr</tt> <tt class="py-op">==</tt> <tt class="py-string">'gapopen_num'</tt><tt class="py-op">:</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-76" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-76', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">,</tt> <tt class="py-string">'query'</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-keyword">not</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-77" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-77', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">,</tt> <tt class="py-string">'hit'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line">            <tt class="py-comment"># mock function so that the except clause below is triggered</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line">            <tt class="py-comment"># as both the query and hit are required to compute gapopen</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line">            <tt class="py-keyword">def</tt> <tt class="py-def-name">mock</tt><tt class="py-op">(</tt><tt class="py-param">hsp</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">AttributeError</tt> </tt>
</div><a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">            <tt class="py-name">func</tt> <tt class="py-op">=</tt> <tt class="py-name">mock</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">            <tt class="py-name">func</tt> <tt class="py-op">=</tt> <tt id="link-78" class="py-name" targets="Function Bio.SearchIO.BlastIO.blast_tab._compute_gapopen_num()=Bio.SearchIO.BlastIO.blast_tab-module.html#_compute_gapopen_num"><a title="Bio.SearchIO.BlastIO.blast_tab._compute_gapopen_num" class="py-name" href="#" onclick="return doclink('link-78', '_compute_gapopen_num', 'link-78');">_compute_gapopen_num</a></tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line"> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">    <tt class="py-comment"># set the num values</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">    <tt class="py-comment"># requires the endswith check, since we only want to set 'num' or 'span'</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">    <tt class="py-comment"># attributes here</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-79" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-79', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">,</tt> <tt class="py-name">attr</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt class="py-name">attr</tt><tt class="py-op">.</tt><tt id="link-80" class="py-name" targets="Method Bio.Seq.Seq.endswith()=Bio.Seq.Seq-class.html#endswith"><a title="Bio.Seq.Seq.endswith" class="py-name" href="#" onclick="return doclink('link-80', 'endswith', 'link-80');">endswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'_pct'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">        <tt id="link-81" class="py-name" targets="Variable Bio.Entrez.Parser.Consumer.value=Bio.Entrez.Parser.Consumer-class.html#value,Variable Bio.Entrez.Parser.DictionaryConsumer.value=Bio.Entrez.Parser.DictionaryConsumer-class.html#value,Variable Bio.Entrez.Parser.ErrorConsumer.value=Bio.Entrez.Parser.ErrorConsumer-class.html#value,Variable Bio.Entrez.Parser.IntegerConsumer.value=Bio.Entrez.Parser.IntegerConsumer-class.html#value,Variable Bio.Entrez.Parser.ListConsumer.value=Bio.Entrez.Parser.ListConsumer-class.html#value,Variable Bio.Entrez.Parser.StringConsumer.value=Bio.Entrez.Parser.StringConsumer-class.html#value,Method Bio.Phylo.PhyloXMLIO.Writer.value()=Bio.Phylo.PhyloXMLIO.Writer-class.html#value"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-81', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">func</tt><tt class="py-op">(</tt><tt id="link-82" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-82', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">)</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">        <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt id="link-83" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-83', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">,</tt> <tt class="py-name">attr</tt><tt class="py-op">,</tt> <tt id="link-84" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-84', 'value', 'link-81');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line"> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">    <tt class="py-comment"># if the attr is a percent value, calculate it</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">attr</tt> <tt class="py-op">==</tt> <tt class="py-string">'ident_pct'</tt><tt class="py-op">:</tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">        <tt class="py-name">func2</tt> <tt class="py-op">=</tt> <tt class="py-keyword">lambda</tt> <tt id="link-85" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-85', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">:</tt> <tt id="link-86" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-86', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">ident_num</tt> <tt class="py-op">/</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt id="link-87" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-87', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-88" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.aln_span
Bio.SearchIO._model.hsp.HSPFragment.aln_span" class="py-name" href="#" onclick="return doclink('link-88', 'aln_span', 'link-62');">aln_span</a></tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-number">100</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">attr</tt> <tt class="py-op">==</tt> <tt class="py-string">'pos_pct'</tt><tt class="py-op">:</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">        <tt class="py-name">func</tt> <tt class="py-op">=</tt> <tt class="py-keyword">lambda</tt> <tt id="link-89" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-89', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">:</tt> <tt id="link-90" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-90', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">pos_num</tt> <tt class="py-op">/</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt id="link-91" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-91', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-92" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.aln_span
Bio.SearchIO._model.hsp.HSPFragment.aln_span" class="py-name" href="#" onclick="return doclink('link-92', 'aln_span', 'link-62');">aln_span</a></tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-number">100</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">attr</tt> <tt class="py-op">==</tt> <tt class="py-string">'gap_pct'</tt><tt class="py-op">:</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">        <tt class="py-name">func2</tt> <tt class="py-op">=</tt> <tt class="py-keyword">lambda</tt> <tt id="link-93" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-93', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">:</tt> <tt id="link-94" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-94', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">gap_num</tt> <tt class="py-op">/</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt id="link-95" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-95', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-96" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.aln_span
Bio.SearchIO._model.hsp.HSPFragment.aln_span" class="py-name" href="#" onclick="return doclink('link-96', 'aln_span', 'link-62');">aln_span</a></tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-number">100</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">        <tt class="py-name">func2</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line"> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">    <tt class="py-comment"># set the pct values</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">func2</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line">        <tt id="link-97" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-97', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">func2</tt><tt class="py-op">(</tt><tt id="link-98" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-98', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">)</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line">        <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt id="link-99" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-99', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">,</tt> <tt class="py-name">attr</tt><tt class="py-op">,</tt> <tt id="link-100" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-100', 'value', 'link-81');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line"> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line"> </tt>
<a name="BlastTabParser"></a><div id="BlastTabParser-def"><a name="L203"></a><tt class="py-lineno">203</tt> <a class="py-toggle" href="#" id="BlastTabParser-toggle" onclick="return toggle('BlastTabParser');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html">BlastTabParser</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabParser-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="BlastTabParser-expanded"><a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line">    <tt class="py-docstring">"""Parser for the BLAST tabular format."""</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line"> </tt>
<a name="BlastTabParser.__init__"></a><div id="BlastTabParser.__init__-def"><a name="L206"></a><tt class="py-lineno">206</tt> <a class="py-toggle" href="#" id="BlastTabParser.__init__-toggle" onclick="return toggle('BlastTabParser.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">comments</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> <tt class="py-param">fields</tt><tt class="py-op">=</tt><tt id="link-101" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._DEFAULT_FIELDS" class="py-name" href="#" onclick="return doclink('link-101', '_DEFAULT_FIELDS', 'link-44');">_DEFAULT_FIELDS</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabParser.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlastTabParser.__init__-expanded"><a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the class."""</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">has_comments</tt> <tt class="py-op">=</tt> <tt class="py-name">comments</tt> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">fields</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-102" class="py-name" targets="Method Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._prep_fields()=Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#_prep_fields"><a title="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._prep_fields" class="py-name" href="#" onclick="return doclink('link-102', '_prep_fields', 'link-102');">_prep_fields</a></tt><tt class="py-op">(</tt><tt class="py-name">fields</tt><tt class="py-op">)</tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-103" class="py-name" targets="Method Bio.File.UndoHandle.readline()=Bio.File.UndoHandle-class.html#readline,Method Bio.bgzf.BgzfReader.readline()=Bio.bgzf.BgzfReader-class.html#readline"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-103', 'readline', 'link-103');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-104" class="py-name" targets="Method Bio.Seq.Seq.strip()=Bio.Seq.Seq-class.html#strip"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-104', 'strip', 'link-104');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line"> </tt>
<a name="BlastTabParser.__iter__"></a><div id="BlastTabParser.__iter__-def"><a name="L213"></a><tt class="py-lineno">213</tt> <a class="py-toggle" href="#" id="BlastTabParser.__iter__-toggle" onclick="return toggle('BlastTabParser.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabParser.__iter__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlastTabParser.__iter__-expanded"><a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line">        <tt class="py-comment"># stop iteration if file has no lines</tt> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line">        <tt class="py-comment"># determine which iterator to use</tt> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">has_comments</tt><tt class="py-op">:</tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line">            <tt class="py-name">iterfunc</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-105" class="py-name" targets="Method Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._parse_commented_qresult()=Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#_parse_commented_qresult"><a title="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._parse_commented_qresult" class="py-name" href="#" onclick="return doclink('link-105', '_parse_commented_qresult', 'link-105');">_parse_commented_qresult</a></tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-106" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-106', 'startswith', 'link-59');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"#"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Encountered unexpected character '#' at the"</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line">                                 <tt class="py-string">" beginning of a line. Set comments=True if"</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line">                                 <tt class="py-string">" the file is a commented file."</tt><tt class="py-op">)</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line">            <tt class="py-name">iterfunc</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-107" class="py-name" targets="Method Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._parse_qresult()=Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#_parse_qresult,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser._parse_qresult()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser-class.html#_parse_qresult,Method Bio.SearchIO.BlatIO.BlatPslParser._parse_qresult()=Bio.SearchIO.BlatIO.BlatPslParser-class.html#_parse_qresult,Method Bio.SearchIO.ExonerateIO._base._BaseExonerateParser._parse_qresult()=Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html#_parse_qresult,Method Bio.SearchIO.FastaIO.FastaM10Parser._parse_qresult()=Bio.SearchIO.FastaIO.FastaM10Parser-class.html#_parse_qresult,Method Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabParser._parse_qresult()=Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabParser-class.html#_parse_qresult,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser._parse_qresult()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser-class.html#_parse_qresult,Method Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser._parse_qresult()=Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#_parse_qresult,Method Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser._parse_qresult()=Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser-class.html#_parse_qresult"><a title="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._parse_qresult
Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser._parse_qresult
Bio.SearchIO.BlatIO.BlatPslParser._parse_qresult
Bio.SearchIO.ExonerateIO._base._BaseExonerateParser._parse_qresult
Bio.SearchIO.FastaIO.FastaM10Parser._parse_qresult
Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabParser._parse_qresult
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser._parse_qresult
Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser._parse_qresult
Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser._parse_qresult" class="py-name" href="#" onclick="return doclink('link-107', '_parse_qresult', 'link-107');">_parse_qresult</a></tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line"> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">qresult</tt> <tt class="py-keyword">in</tt> <tt class="py-name">iterfunc</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt class="py-name">qresult</tt> </tt>
</div><a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line"> </tt>
<a name="BlastTabParser._prep_fields"></a><div id="BlastTabParser._prep_fields-def"><a name="L230"></a><tt class="py-lineno">230</tt> <a class="py-toggle" href="#" id="BlastTabParser._prep_fields-toggle" onclick="return toggle('BlastTabParser._prep_fields');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#_prep_fields">_prep_fields</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">fields</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabParser._prep_fields-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlastTabParser._prep_fields-expanded"><a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line">        <tt class="py-docstring">"""Validate and format the given fields for use by the parser (PRIVATE)."""</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">        <tt class="py-comment"># cast into list if fields is a space-separated string</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">fields</tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line">            <tt class="py-name">fields</tt> <tt class="py-op">=</tt> <tt class="py-name">fields</tt><tt class="py-op">.</tt><tt id="link-108" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-108', 'strip', 'link-104');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-109" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-109', 'split', 'link-21');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">' '</tt><tt class="py-op">)</tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line">        <tt class="py-comment"># blast allows 'std' as a proxy for the standard default lists</tt> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line">        <tt class="py-comment"># we want to transform 'std' to its proper column names</tt> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">'std'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">fields</tt><tt class="py-op">:</tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line">            <tt class="py-name">idx</tt> <tt class="py-op">=</tt> <tt class="py-name">fields</tt><tt class="py-op">.</tt><tt id="link-110" class="py-name" targets="Method Bio.Crystal.Chain.index()=Bio.Crystal.Chain-class.html#index,Method Bio.SCOP.Raf.SeqMap.index()=Bio.SCOP.Raf.SeqMap-class.html#index,Method Bio.SearchIO._model.hit.Hit.index()=Bio.SearchIO._model.hit.Hit-class.html#index,Method Bio.SearchIO._model.query.QueryResult.index()=Bio.SearchIO._model.query.QueryResult-class.html#index,Function Bio.SearchIO.index()=Bio.SearchIO-module.html#index,Method Bio.Seq.MutableSeq.index()=Bio.Seq.MutableSeq-class.html#index,Function Bio.SeqIO.index()=Bio.SeqIO-module.html#index"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-110', 'index', 'link-110');">index</a></tt><tt class="py-op">(</tt><tt class="py-string">'std'</tt><tt class="py-op">)</tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line">            <tt class="py-name">fields</tt> <tt class="py-op">=</tt> <tt class="py-name">fields</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">idx</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt id="link-111" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._DEFAULT_FIELDS" class="py-name" href="#" onclick="return doclink('link-111', '_DEFAULT_FIELDS', 'link-44');">_DEFAULT_FIELDS</a></tt> <tt class="py-op">+</tt> <tt class="py-name">fields</tt><tt class="py-op">[</tt><tt class="py-name">idx</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line">        <tt class="py-comment"># if set(fields) has a null intersection with minimum required</tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line">        <tt class="py-comment"># fields for hit and query, raise an exception</tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-112" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-112', 'set', 'link-39');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">fields</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">intersection</tt><tt class="py-op">(</tt><tt id="link-113" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._MIN_QUERY_FIELDS" class="py-name" href="#" onclick="return doclink('link-113', '_MIN_QUERY_FIELDS', 'link-45');">_MIN_QUERY_FIELDS</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> \ </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line">                <tt class="py-keyword">not</tt> <tt id="link-114" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-114', 'set', 'link-39');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">fields</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">intersection</tt><tt class="py-op">(</tt><tt id="link-115" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._MIN_HIT_FIELDS" class="py-name" href="#" onclick="return doclink('link-115', '_MIN_HIT_FIELDS', 'link-47');">_MIN_HIT_FIELDS</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Required query and/or hit ID field not found."</tt><tt class="py-op">)</tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line"> </tt>
<a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">fields</tt> </tt>
</div><a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line"> </tt>
<a name="BlastTabParser._parse_commented_qresult"></a><div id="BlastTabParser._parse_commented_qresult-def"><a name="L248"></a><tt class="py-lineno">248</tt> <a class="py-toggle" href="#" id="BlastTabParser._parse_commented_qresult-toggle" onclick="return toggle('BlastTabParser._parse_commented_qresult');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#_parse_commented_qresult">_parse_commented_qresult</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabParser._parse_commented_qresult-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlastTabParser._parse_commented_qresult-expanded"><a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line">        <tt class="py-docstring">"""Yield `QueryResult` objects from a commented file (PRIVATE)."""</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line">            <tt class="py-name">comments</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-116" class="py-name" targets="Method Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._parse_comments()=Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#_parse_comments"><a title="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._parse_comments" class="py-name" href="#" onclick="return doclink('link-116', '_parse_comments', 'link-116');">_parse_comments</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">comments</tt><tt class="py-op">:</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line">                <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">fields</tt> <tt class="py-op">=</tt> <tt class="py-name">comments</tt><tt class="py-op">[</tt><tt class="py-string">'fields'</tt><tt class="py-op">]</tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line">                    <tt class="py-comment"># iterator for the query results</tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line">                    <tt class="py-name">qres_iter</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-117" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._parse_qresult
Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser._parse_qresult
Bio.SearchIO.BlatIO.BlatPslParser._parse_qresult
Bio.SearchIO.ExonerateIO._base._BaseExonerateParser._parse_qresult
Bio.SearchIO.FastaIO.FastaM10Parser._parse_qresult
Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabParser._parse_qresult
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser._parse_qresult
Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser._parse_qresult
Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser._parse_qresult" class="py-name" href="#" onclick="return doclink('link-117', '_parse_qresult', 'link-107');">_parse_qresult</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line">                <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line">                    <tt class="py-comment"># no fields means the query has no results</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line">                    <tt class="py-keyword">assert</tt> <tt class="py-string">'fields'</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">comments</tt> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line">                    <tt class="py-comment"># create an iterator returning one empty qresult</tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line">                    <tt class="py-comment"># if the query has no results</tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line">                    <tt class="py-name">qres_iter</tt> <tt class="py-op">=</tt> <tt class="py-name">iter</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt id="link-118" class="py-name"><a title="Bio.SearchIO._model.query.QueryResult" class="py-name" href="#" onclick="return doclink('link-118', 'QueryResult', 'link-13');">QueryResult</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line"> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">qresult</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qres_iter</tt><tt class="py-op">:</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt id="link-119" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-119', 'value', 'link-81');">value</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">comments</tt><tt class="py-op">.</tt><tt id="link-120" class="py-name" targets="Method Bio.Crystal.Crystal.items()=Bio.Crystal.Crystal-class.html#items,Method Bio.File._IndexedSeqFileDict.items()=Bio.File._IndexedSeqFileDict-class.html#items,Method Bio.Phylo.PhyloXML.Events.items()=Bio.Phylo.PhyloXML.Events-class.html#items,Variable Bio.SearchIO._model.query.QueryResult.items=Bio.SearchIO._model.query.QueryResult-class.html#items,Method BioSQL.BioSeqDatabase.BioSeqDatabase.items()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#items,Method BioSQL.BioSeqDatabase.DBServer.items()=BioSQL.BioSeqDatabase.DBServer-class.html#items"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-120', 'items', 'link-120');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">                        <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt class="py-name">qresult</tt><tt class="py-op">,</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt id="link-121" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-121', 'value', 'link-81');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">                    <tt class="py-keyword">yield</tt> <tt class="py-name">qresult</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line"> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
</div><a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line"> </tt>
<a name="BlastTabParser._parse_comments"></a><div id="BlastTabParser._parse_comments-def"><a name="L272"></a><tt class="py-lineno">272</tt> <a class="py-toggle" href="#" id="BlastTabParser._parse_comments-toggle" onclick="return toggle('BlastTabParser._parse_comments');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#_parse_comments">_parse_comments</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabParser._parse_comments-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlastTabParser._parse_comments-expanded"><a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a dictionary containing tab file comments (PRIVATE)."""</tt> </tt>
<a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line">        <tt class="py-name">comments</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line">            <tt class="py-comment"># parse program and version</tt> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line">            <tt class="py-comment"># example: # BLASTX 2.2.26+</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-string">'BLAST'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-keyword">and</tt> <tt class="py-string">'processed'</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line">                <tt class="py-name">program_line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-string">' #'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-122" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-122', 'split', 'link-21');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">' '</tt><tt class="py-op">)</tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line">                <tt class="py-name">comments</tt><tt class="py-op">[</tt><tt class="py-string">'program'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">program_line</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-123" class="py-name" targets="Method Bio.Seq.Seq.lower()=Bio.Seq.Seq-class.html#lower,Method Bio.Seq.UnknownSeq.lower()=Bio.Seq.UnknownSeq-class.html#lower,Method Bio.SeqRecord.SeqRecord.lower()=Bio.SeqRecord.SeqRecord-class.html#lower"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-123', 'lower', 'link-123');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line">                <tt class="py-name">comments</tt><tt class="py-op">[</tt><tt class="py-string">'version'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">program_line</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line">            <tt class="py-comment"># parse query id and description (if available)</tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line">            <tt class="py-comment"># example: # Query: gi|356995852 Mus musculus POU domain</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-string">'Query'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line">                <tt class="py-name">query_line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-string">'# Query: '</tt><tt class="py-op">)</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-124" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-124', 'split', 'link-21');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">' '</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line">                <tt class="py-name">comments</tt><tt class="py-op">[</tt><tt class="py-string">'id'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">query_line</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">query_line</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">                    <tt class="py-name">comments</tt><tt class="py-op">[</tt><tt class="py-string">'description'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">query_line</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">            <tt class="py-comment"># parse target database</tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">            <tt class="py-comment"># example: # Database: db/minirefseq_protein</tt> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-string">'Database'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line">                <tt class="py-name">comments</tt><tt class="py-op">[</tt><tt class="py-string">'target'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-string">'# Database: '</tt><tt class="py-op">)</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line">            <tt class="py-comment"># parse RID (from remote searches)</tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-string">'RID'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line">                <tt class="py-name">comments</tt><tt class="py-op">[</tt><tt class="py-string">'rid'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-string">'# RID: '</tt><tt class="py-op">)</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line">            <tt class="py-comment"># parse column order, required for parsing the result lines</tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line">            <tt class="py-comment"># example: # Fields: query id, query gi, query acc., query length</tt> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-string">'Fields'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line">                <tt class="py-name">comments</tt><tt class="py-op">[</tt><tt class="py-string">'fields'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-125" class="py-name" targets="Method Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._parse_fields_line()=Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#_parse_fields_line"><a title="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._parse_fields_line" class="py-name" href="#" onclick="return doclink('link-125', '_parse_fields_line', 'link-125');">_parse_fields_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line">            <tt class="py-comment"># if the line has these strings, it's either the end of a comment</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line">            <tt class="py-comment"># or the end of a file, so we return all the comments we've parsed</tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-string">' hits found'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-keyword">or</tt> <tt class="py-string">'processed'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-126" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-126', 'readline', 'link-103');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-127" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-127', 'strip', 'link-104');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">comments</tt> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line"> </tt>
<a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-128" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-128', 'readline', 'link-103');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line"> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">comments</tt> </tt>
<a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-129" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-129', 'strip', 'link-104');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line"> </tt>
<a name="BlastTabParser._parse_fields_line"></a><div id="BlastTabParser._parse_fields_line-def"><a name="L313"></a><tt class="py-lineno">313</tt> <a class="py-toggle" href="#" id="BlastTabParser._parse_fields_line-toggle" onclick="return toggle('BlastTabParser._parse_fields_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#_parse_fields_line">_parse_fields_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabParser._parse_fields_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlastTabParser._parse_fields_line-expanded"><a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return column short names line from 'Fields' comment line (PRIVATE)."""</tt> </tt>
<a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line">        <tt class="py-name">raw_field_str</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-string">'# Fields: '</tt><tt class="py-op">)</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line">        <tt class="py-name">long_fields</tt> <tt class="py-op">=</tt> <tt class="py-name">raw_field_str</tt><tt class="py-op">.</tt><tt id="link-130" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-130', 'split', 'link-21');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">', '</tt><tt class="py-op">)</tt> </tt>
<a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line">        <tt class="py-name">fields</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-131" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._LONG_SHORT_MAP" class="py-name" href="#" onclick="return doclink('link-131', '_LONG_SHORT_MAP', 'link-17');">_LONG_SHORT_MAP</a></tt><tt class="py-op">[</tt><tt class="py-name">long_name</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt class="py-name">long_name</tt> <tt class="py-keyword">in</tt> <tt class="py-name">long_fields</tt><tt class="py-op">]</tt> </tt>
<a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-132" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._prep_fields" class="py-name" href="#" onclick="return doclink('link-132', '_prep_fields', 'link-102');">_prep_fields</a></tt><tt class="py-op">(</tt><tt class="py-name">fields</tt><tt class="py-op">)</tt> </tt>
</div><a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line"> </tt>
<a name="BlastTabParser._parse_result_row"></a><div id="BlastTabParser._parse_result_row-def"><a name="L320"></a><tt class="py-lineno">320</tt> <a class="py-toggle" href="#" id="BlastTabParser._parse_result_row-toggle" onclick="return toggle('BlastTabParser._parse_result_row');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#_parse_result_row">_parse_result_row</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabParser._parse_result_row-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlastTabParser._parse_result_row-expanded"><a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a dictionary of parsed row values (PRIVATE)."""</tt> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">        <tt class="py-name">fields</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">fields</tt> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">        <tt class="py-name">columns</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-133" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-133', 'strip', 'link-104');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-134" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-134', 'split', 'link-21');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">'\t'</tt><tt class="py-op">)</tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">fields</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">columns</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"Expected %i columns, found: "</tt> \ </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line">            <tt class="py-string">"%i"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">fields</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">columns</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line"> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line">        <tt class="py-name">qresult</tt><tt class="py-op">,</tt> <tt id="link-135" class="py-name" targets="Module Bio.SearchIO._model.hit=Bio.SearchIO._model.hit-module.html,Variable Bio.SearchIO._model.hsp.HSP.hit=Bio.SearchIO._model.hsp.HSP-class.html#hit,Variable Bio.SearchIO._model.hsp.HSPFragment.hit=Bio.SearchIO._model.hsp.HSPFragment-class.html#hit"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-135', 'hit', 'link-135');">hit</a></tt><tt class="py-op">,</tt> <tt id="link-136" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-136', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">,</tt> <tt class="py-name">frag</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt><tt class="py-op">,</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt><tt class="py-op">,</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt><tt class="py-op">,</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">idx</tt><tt class="py-op">,</tt> <tt id="link-137" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-137', 'value', 'link-81');">value</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt class="py-name">columns</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line">            <tt class="py-name">sname</tt> <tt class="py-op">=</tt> <tt class="py-name">fields</tt><tt class="py-op">[</tt><tt class="py-name">idx</tt><tt class="py-op">]</tt> </tt>
<a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line">            <tt class="py-comment"># flag to check if any of the _COLUMNs contain sname</tt> </tt>
<a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line">            <tt class="py-name">in_mapping</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line">            <tt class="py-comment"># iterate over each dict, mapping pair to determine</tt> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line">            <tt class="py-comment"># attribute name and value of each column</tt> </tt>
<a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">parsed_dict</tt><tt class="py-op">,</tt> <tt class="py-name">mapping</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt> </tt>
<a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line">                    <tt class="py-op">(</tt><tt class="py-name">qresult</tt><tt class="py-op">,</tt> <tt id="link-138" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._COLUMN_QRESULT" class="py-name" href="#" onclick="return doclink('link-138', '_COLUMN_QRESULT', 'link-26');">_COLUMN_QRESULT</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line">                    <tt class="py-op">(</tt><tt id="link-139" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-139', 'hit', 'link-135');">hit</a></tt><tt class="py-op">,</tt> <tt id="link-140" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._COLUMN_HIT" class="py-name" href="#" onclick="return doclink('link-140', '_COLUMN_HIT', 'link-27');">_COLUMN_HIT</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line">                    <tt class="py-op">(</tt><tt id="link-141" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-141', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">,</tt> <tt id="link-142" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._COLUMN_HSP" class="py-name" href="#" onclick="return doclink('link-142', '_COLUMN_HSP', 'link-36');">_COLUMN_HSP</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line">                    <tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">,</tt> <tt id="link-143" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._COLUMN_FRAG" class="py-name" href="#" onclick="return doclink('link-143', '_COLUMN_FRAG', 'link-37');">_COLUMN_FRAG</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line">                <tt class="py-comment"># process parsed value according to mapping</tt> </tt>
<a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">sname</tt> <tt class="py-keyword">in</tt> <tt class="py-name">mapping</tt><tt class="py-op">:</tt> </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line">                    <tt class="py-name">attr_name</tt><tt class="py-op">,</tt> <tt class="py-name">caster</tt> <tt class="py-op">=</tt> <tt class="py-name">mapping</tt><tt class="py-op">[</tt><tt class="py-name">sname</tt><tt class="py-op">]</tt> </tt>
<a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">caster</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">str</tt><tt class="py-op">:</tt> </tt>
<a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line">                        <tt id="link-144" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-144', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">caster</tt><tt class="py-op">(</tt><tt id="link-145" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-145', 'value', 'link-81');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line">                    <tt class="py-name">parsed_dict</tt><tt class="py-op">[</tt><tt class="py-name">attr_name</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-146" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-146', 'value', 'link-81');">value</a></tt> </tt>
<a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line">                    <tt class="py-name">in_mapping</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line">            <tt class="py-comment"># make sure that any unhandled field is not supported</tt> </tt>
<a name="L347"></a><tt class="py-lineno">347</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">in_mapping</tt><tt class="py-op">:</tt> </tt>
<a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line">                <tt class="py-keyword">assert</tt> <tt class="py-name">sname</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt id="link-147" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._SUPPORTED_FIELDS" class="py-name" href="#" onclick="return doclink('link-147', '_SUPPORTED_FIELDS', 'link-38');">_SUPPORTED_FIELDS</a></tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line"> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">{</tt><tt class="py-string">'qresult'</tt><tt class="py-op">:</tt> <tt class="py-name">qresult</tt><tt class="py-op">,</tt> <tt class="py-string">'hit'</tt><tt class="py-op">:</tt> <tt id="link-148" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-148', 'hit', 'link-135');">hit</a></tt><tt class="py-op">,</tt> <tt class="py-string">'hsp'</tt><tt class="py-op">:</tt> <tt id="link-149" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-149', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">,</tt> <tt class="py-string">'frag'</tt><tt class="py-op">:</tt> <tt class="py-name">frag</tt><tt class="py-op">}</tt> </tt>
</div><a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line"> </tt>
<a name="BlastTabParser._get_id"></a><div id="BlastTabParser._get_id-def"><a name="L352"></a><tt class="py-lineno">352</tt> <a class="py-toggle" href="#" id="BlastTabParser._get_id-toggle" onclick="return toggle('BlastTabParser._get_id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#_get_id">_get_id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">parsed</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabParser._get_id-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlastTabParser._get_id-expanded"><a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the value used for a QueryResult or Hit ID from a parsed row (PRIVATE)."""</tt> </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line">        <tt class="py-comment"># use 'id', with 'id_all', 'accession' and 'accession_version'</tt> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">        <tt class="py-comment"># fallbacks one of these must have a value since we've checked whether</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line">        <tt class="py-comment"># they exist or not when parsing the comments</tt> </tt>
<a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line">        <tt class="py-name">id_cache</tt> <tt class="py-op">=</tt> <tt class="py-name">parsed</tt><tt class="py-op">.</tt><tt id="link-150" class="py-name" targets="Method Bio.Crystal.Crystal.get()=Bio.Crystal.Crystal-class.html#get,Method Bio.Data.CodonTable.AmbiguousForwardTable.get()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#get,Method Bio.File._IndexedSeqFileDict.get()=Bio.File._IndexedSeqFileDict-class.html#get,Method Bio.File._IndexedSeqFileProxy.get()=Bio.File._IndexedSeqFileProxy-class.html#get,Method Bio.File._SQLiteManySeqFilesDict.get()=Bio.File._SQLiteManySeqFilesDict-class.html#get,Method Bio.Restriction.Restriction.RestrictionBatch.get()=Bio.Restriction.Restriction.RestrictionBatch-class.html#get,Method Bio.SearchIO._index.SearchIndexer.get()=Bio.SearchIO._index.SearchIndexer-class.html#get,Method Bio.SeqIO._index.SeqFileRandomAccess.get()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#get,Method Bio.SeqIO._index.SffRandomAccess.get()=Bio.SeqIO._index.SffRandomAccess-class.html#get,Method Bio.SeqIO._index.SffTrimedRandomAccess.get()=Bio.SeqIO._index.SffTrimedRandomAccess-class.html#get,Method Bio.SeqIO._index.UniprotRandomAccess.get()=Bio.SeqIO._index.UniprotRandomAccess-class.html#get"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-150', 'get', 'link-150');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'id'</tt><tt class="py-op">)</tt> </tt>
<a name="L358"></a><tt class="py-lineno">358</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">id_cache</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> <tt class="py-string">'id_all'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">parsed</tt><tt class="py-op">:</tt> </tt>
<a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line">            <tt class="py-name">id_cache</tt> <tt class="py-op">=</tt> <tt class="py-name">parsed</tt><tt class="py-op">.</tt><tt id="link-151" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-151', 'get', 'link-150');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'id_all'</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">id_cache</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line">            <tt class="py-name">id_cache</tt> <tt class="py-op">=</tt> <tt class="py-name">parsed</tt><tt class="py-op">.</tt><tt id="link-152" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-152', 'get', 'link-150');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'accession'</tt><tt class="py-op">)</tt> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">id_cache</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line">            <tt class="py-name">id_cache</tt> <tt class="py-op">=</tt> <tt class="py-name">parsed</tt><tt class="py-op">.</tt><tt id="link-153" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-153', 'get', 'link-150');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'accession_version'</tt><tt class="py-op">)</tt> </tt>
<a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line"> </tt>
<a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">id_cache</tt> </tt>
</div><a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line"> </tt>
<a name="BlastTabParser._parse_qresult"></a><div id="BlastTabParser._parse_qresult-def"><a name="L367"></a><tt class="py-lineno">367</tt> <a class="py-toggle" href="#" id="BlastTabParser._parse_qresult-toggle" onclick="return toggle('BlastTabParser._parse_qresult');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#_parse_qresult">_parse_qresult</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabParser._parse_qresult-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlastTabParser._parse_qresult-expanded"><a name="L368"></a><tt class="py-lineno">368</tt>  <tt class="py-line">        <tt class="py-docstring">"""Yield QueryResult objects (PRIVATE)."""</tt> </tt>
<a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line">        <tt class="py-comment"># state values, used to determine what to do with each line</tt> </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line">        <tt class="py-name">state_EOF</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line">        <tt class="py-name">state_QRES_NEW</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line">        <tt class="py-name">state_QRES_SAME</tt> <tt class="py-op">=</tt> <tt class="py-number">3</tt> </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line">        <tt class="py-name">state_HIT_NEW</tt> <tt class="py-op">=</tt> <tt class="py-number">2</tt> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line">        <tt class="py-name">state_HIT_SAME</tt> <tt class="py-op">=</tt> <tt class="py-number">4</tt> </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line">        <tt class="py-comment"># dummies for initial states</tt> </tt>
<a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line">        <tt class="py-name">qres_state</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line">        <tt class="py-name">hit_state</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L378"></a><tt class="py-lineno">378</tt>  <tt class="py-line">        <tt class="py-name">file_state</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L379"></a><tt class="py-lineno">379</tt>  <tt class="py-line">        <tt class="py-name">cur_qid</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L380"></a><tt class="py-lineno">380</tt>  <tt class="py-line">        <tt class="py-name">cur_hid</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L381"></a><tt class="py-lineno">381</tt>  <tt class="py-line">        <tt class="py-comment"># dummies for initial id caches</tt> </tt>
<a name="L382"></a><tt class="py-lineno">382</tt>  <tt class="py-line">        <tt class="py-name">prev_qid</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L383"></a><tt class="py-lineno">383</tt>  <tt class="py-line">        <tt class="py-name">prev_hid</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L384"></a><tt class="py-lineno">384</tt>  <tt class="py-line">        <tt class="py-comment"># dummies for initial parsed value containers</tt> </tt>
<a name="L385"></a><tt class="py-lineno">385</tt>  <tt class="py-line">        <tt class="py-name">cur</tt><tt class="py-op">,</tt> <tt class="py-name">prev</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt> </tt>
<a name="L386"></a><tt class="py-lineno">386</tt>  <tt class="py-line">        <tt class="py-name">hit_list</tt><tt class="py-op">,</tt> <tt class="py-name">hsp_list</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L387"></a><tt class="py-lineno">387</tt>  <tt class="py-line"> </tt>
<a name="L388"></a><tt class="py-lineno">388</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L389"></a><tt class="py-lineno">389</tt>  <tt class="py-line">            <tt class="py-comment"># store previous line's parsed values if we've past the first line</tt> </tt>
<a name="L390"></a><tt class="py-lineno">390</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">cur</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L391"></a><tt class="py-lineno">391</tt>  <tt class="py-line">                <tt class="py-name">prev</tt> <tt class="py-op">=</tt> <tt class="py-name">cur</tt> </tt>
<a name="L392"></a><tt class="py-lineno">392</tt>  <tt class="py-line">                <tt class="py-name">prev_qid</tt> <tt class="py-op">=</tt> <tt class="py-name">cur_qid</tt> </tt>
<a name="L393"></a><tt class="py-lineno">393</tt>  <tt class="py-line">                <tt class="py-name">prev_hid</tt> <tt class="py-op">=</tt> <tt class="py-name">cur_hid</tt> </tt>
<a name="L394"></a><tt class="py-lineno">394</tt>  <tt class="py-line">            <tt class="py-comment"># only parse the line if it's not EOF or not a comment line</tt> </tt>
<a name="L395"></a><tt class="py-lineno">395</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-154" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-154', 'startswith', 'link-59');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'#'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L396"></a><tt class="py-lineno">396</tt>  <tt class="py-line">                <tt class="py-name">cur</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-155" class="py-name" targets="Method Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._parse_result_row()=Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#_parse_result_row"><a title="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._parse_result_row" class="py-name" href="#" onclick="return doclink('link-155', '_parse_result_row', 'link-155');">_parse_result_row</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L397"></a><tt class="py-lineno">397</tt>  <tt class="py-line">                <tt class="py-name">cur_qid</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-156" class="py-name" targets="Method Bio.Nexus.Nodes.Chain._get_id()=Bio.Nexus.Nodes.Chain-class.html#_get_id,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._get_id()=Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#_get_id"><a title="Bio.Nexus.Nodes.Chain._get_id
Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._get_id" class="py-name" href="#" onclick="return doclink('link-156', '_get_id', 'link-156');">_get_id</a></tt><tt class="py-op">(</tt><tt class="py-name">cur</tt><tt class="py-op">[</tt><tt class="py-string">'qresult'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L398"></a><tt class="py-lineno">398</tt>  <tt class="py-line">                <tt class="py-name">cur_hid</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-157" class="py-name"><a title="Bio.Nexus.Nodes.Chain._get_id
Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._get_id" class="py-name" href="#" onclick="return doclink('link-157', '_get_id', 'link-156');">_get_id</a></tt><tt class="py-op">(</tt><tt class="py-name">cur</tt><tt class="py-op">[</tt><tt class="py-string">'hit'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L399"></a><tt class="py-lineno">399</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L400"></a><tt class="py-lineno">400</tt>  <tt class="py-line">                <tt class="py-name">file_state</tt> <tt class="py-op">=</tt> <tt class="py-name">state_EOF</tt> </tt>
<a name="L401"></a><tt class="py-lineno">401</tt>  <tt class="py-line">                <tt class="py-comment"># mock values for cur_qid and cur_hid since the line is empty</tt> </tt>
<a name="L402"></a><tt class="py-lineno">402</tt>  <tt class="py-line">                <tt class="py-name">cur_qid</tt><tt class="py-op">,</tt> <tt class="py-name">cur_hid</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt> </tt>
<a name="L403"></a><tt class="py-lineno">403</tt>  <tt class="py-line"> </tt>
<a name="L404"></a><tt class="py-lineno">404</tt>  <tt class="py-line">            <tt class="py-comment"># get the state of hit and qresult</tt> </tt>
<a name="L405"></a><tt class="py-lineno">405</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">prev_qid</tt> <tt class="py-op">!=</tt> <tt class="py-name">cur_qid</tt><tt class="py-op">:</tt> </tt>
<a name="L406"></a><tt class="py-lineno">406</tt>  <tt class="py-line">                <tt class="py-name">qres_state</tt> <tt class="py-op">=</tt> <tt class="py-name">state_QRES_NEW</tt> </tt>
<a name="L407"></a><tt class="py-lineno">407</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L408"></a><tt class="py-lineno">408</tt>  <tt class="py-line">                <tt class="py-name">qres_state</tt> <tt class="py-op">=</tt> <tt class="py-name">state_QRES_SAME</tt> </tt>
<a name="L409"></a><tt class="py-lineno">409</tt>  <tt class="py-line">            <tt class="py-comment"># new hits are hits with different id or hits in a new qresult</tt> </tt>
<a name="L410"></a><tt class="py-lineno">410</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">prev_hid</tt> <tt class="py-op">!=</tt> <tt class="py-name">cur_hid</tt> <tt class="py-keyword">or</tt> <tt class="py-name">qres_state</tt> <tt class="py-op">==</tt> <tt class="py-name">state_QRES_NEW</tt><tt class="py-op">:</tt> </tt>
<a name="L411"></a><tt class="py-lineno">411</tt>  <tt class="py-line">                <tt class="py-name">hit_state</tt> <tt class="py-op">=</tt> <tt class="py-name">state_HIT_NEW</tt> </tt>
<a name="L412"></a><tt class="py-lineno">412</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L413"></a><tt class="py-lineno">413</tt>  <tt class="py-line">                <tt class="py-name">hit_state</tt> <tt class="py-op">=</tt> <tt class="py-name">state_HIT_SAME</tt> </tt>
<a name="L414"></a><tt class="py-lineno">414</tt>  <tt class="py-line"> </tt>
<a name="L415"></a><tt class="py-lineno">415</tt>  <tt class="py-line">            <tt class="py-comment"># we're creating objects for the previously parsed line(s),</tt> </tt>
<a name="L416"></a><tt class="py-lineno">416</tt>  <tt class="py-line">            <tt class="py-comment"># so nothing is done in the first parsed line (prev == None)</tt> </tt>
<a name="L417"></a><tt class="py-lineno">417</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">prev</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L418"></a><tt class="py-lineno">418</tt>  <tt class="py-line">                <tt class="py-comment"># every line is essentially an HSP with one fragment, so we</tt> </tt>
<a name="L419"></a><tt class="py-lineno">419</tt>  <tt class="py-line">                <tt class="py-comment"># create both of these for every line</tt> </tt>
<a name="L420"></a><tt class="py-lineno">420</tt>  <tt class="py-line">                <tt class="py-name">frag</tt> <tt class="py-op">=</tt> <tt id="link-158" class="py-name"><a title="Bio.SearchIO._model.hsp.HSPFragment" class="py-name" href="#" onclick="return doclink('link-158', 'HSPFragment', 'link-16');">HSPFragment</a></tt><tt class="py-op">(</tt><tt class="py-name">prev_hid</tt><tt class="py-op">,</tt> <tt class="py-name">prev_qid</tt><tt class="py-op">)</tt> </tt>
<a name="L421"></a><tt class="py-lineno">421</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">attr</tt><tt class="py-op">,</tt> <tt id="link-159" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-159', 'value', 'link-81');">value</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">prev</tt><tt class="py-op">[</tt><tt class="py-string">'frag'</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-160" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-160', 'items', 'link-120');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L422"></a><tt class="py-lineno">422</tt>  <tt class="py-line">                    <tt class="py-comment"># adjust coordinates to Python range</tt> </tt>
<a name="L423"></a><tt class="py-lineno">423</tt>  <tt class="py-line">                    <tt class="py-comment"># NOTE: this requires both start and end coords to be</tt> </tt>
<a name="L424"></a><tt class="py-lineno">424</tt>  <tt class="py-line">                    <tt class="py-comment"># present, otherwise a KeyError will be raised.</tt> </tt>
<a name="L425"></a><tt class="py-lineno">425</tt>  <tt class="py-line">                    <tt class="py-comment"># Without this limitation, we might misleadingly set the</tt> </tt>
<a name="L426"></a><tt class="py-lineno">426</tt>  <tt class="py-line">                    <tt class="py-comment"># start / end coords</tt> </tt>
<a name="L427"></a><tt class="py-lineno">427</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-name">seq_type</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'query'</tt><tt class="py-op">,</tt> <tt class="py-string">'hit'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L428"></a><tt class="py-lineno">428</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">attr</tt> <tt class="py-op">==</tt> <tt class="py-name">seq_type</tt> <tt class="py-op">+</tt> <tt class="py-string">'_start'</tt><tt class="py-op">:</tt> </tt>
<a name="L429"></a><tt class="py-lineno">429</tt>  <tt class="py-line">                            <tt id="link-161" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-161', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-162" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.min=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#min"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-162', 'min', 'link-162');">min</a></tt><tt class="py-op">(</tt><tt id="link-163" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-163', 'value', 'link-81');">value</a></tt><tt class="py-op">,</tt> </tt>
<a name="L430"></a><tt class="py-lineno">430</tt>  <tt class="py-line">                                    <tt class="py-name">prev</tt><tt class="py-op">[</tt><tt class="py-string">'frag'</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">seq_type</tt> <tt class="py-op">+</tt> <tt class="py-string">'_end'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt> </tt>
<a name="L431"></a><tt class="py-lineno">431</tt>  <tt class="py-line">                        <tt class="py-keyword">elif</tt> <tt class="py-name">attr</tt> <tt class="py-op">==</tt> <tt class="py-name">seq_type</tt> <tt class="py-op">+</tt> <tt class="py-string">'_end'</tt><tt class="py-op">:</tt> </tt>
<a name="L432"></a><tt class="py-lineno">432</tt>  <tt class="py-line">                            <tt id="link-164" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-164', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-165" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.max=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#max"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-165', 'max', 'link-165');">max</a></tt><tt class="py-op">(</tt><tt id="link-166" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-166', 'value', 'link-81');">value</a></tt><tt class="py-op">,</tt> </tt>
<a name="L433"></a><tt class="py-lineno">433</tt>  <tt class="py-line">                                    <tt class="py-name">prev</tt><tt class="py-op">[</tt><tt class="py-string">'frag'</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">seq_type</tt> <tt class="py-op">+</tt> <tt class="py-string">'_start'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L434"></a><tt class="py-lineno">434</tt>  <tt class="py-line">                    <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">,</tt> <tt class="py-name">attr</tt><tt class="py-op">,</tt> <tt id="link-167" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-167', 'value', 'link-81');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L435"></a><tt class="py-lineno">435</tt>  <tt class="py-line">                <tt class="py-comment"># strand and frame setattr require the full parsed values</tt> </tt>
<a name="L436"></a><tt class="py-lineno">436</tt>  <tt class="py-line">                <tt class="py-comment"># to be set first</tt> </tt>
<a name="L437"></a><tt class="py-lineno">437</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">seq_type</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'hit'</tt><tt class="py-op">,</tt> <tt class="py-string">'query'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L438"></a><tt class="py-lineno">438</tt>  <tt class="py-line">                    <tt class="py-comment"># try to set hit and query frame</tt> </tt>
<a name="L439"></a><tt class="py-lineno">439</tt>  <tt class="py-line">                    <tt id="link-168" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.frame()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#frame"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.frame" class="py-name" href="#" onclick="return doclink('link-168', 'frame', 'link-168');">frame</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-169" class="py-name" targets="Method Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._get_frag_frame()=Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#_get_frag_frame"><a title="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._get_frag_frame" class="py-name" href="#" onclick="return doclink('link-169', '_get_frag_frame', 'link-169');">_get_frag_frame</a></tt><tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">,</tt> <tt class="py-name">seq_type</tt><tt class="py-op">,</tt> </tt>
<a name="L440"></a><tt class="py-lineno">440</tt>  <tt class="py-line">                            <tt class="py-name">prev</tt><tt class="py-op">[</tt><tt class="py-string">'frag'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L441"></a><tt class="py-lineno">441</tt>  <tt class="py-line">                    <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">,</tt> <tt class="py-string">'%s_frame'</tt> <tt class="py-op">%</tt> <tt class="py-name">seq_type</tt><tt class="py-op">,</tt> <tt id="link-170" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.frame" class="py-name" href="#" onclick="return doclink('link-170', 'frame', 'link-168');">frame</a></tt><tt class="py-op">)</tt> </tt>
<a name="L442"></a><tt class="py-lineno">442</tt>  <tt class="py-line">                    <tt class="py-comment"># try to set hit and query strand</tt> </tt>
<a name="L443"></a><tt class="py-lineno">443</tt>  <tt class="py-line">                    <tt id="link-171" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.strand()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#strand,Variable Bio.SeqFeature.CompoundLocation.strand=Bio.SeqFeature.CompoundLocation-class.html#strand,Variable Bio.SeqFeature.FeatureLocation.strand=Bio.SeqFeature.FeatureLocation-class.html#strand,Variable Bio.SeqFeature.SeqFeature.strand=Bio.SeqFeature.SeqFeature-class.html#strand"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-171', 'strand', 'link-171');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-172" class="py-name" targets="Method Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._get_frag_strand()=Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#_get_frag_strand"><a title="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._get_frag_strand" class="py-name" href="#" onclick="return doclink('link-172', '_get_frag_strand', 'link-172');">_get_frag_strand</a></tt><tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">,</tt> <tt class="py-name">seq_type</tt><tt class="py-op">,</tt> </tt>
<a name="L444"></a><tt class="py-lineno">444</tt>  <tt class="py-line">                            <tt class="py-name">prev</tt><tt class="py-op">[</tt><tt class="py-string">'frag'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L445"></a><tt class="py-lineno">445</tt>  <tt class="py-line">                    <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">,</tt> <tt class="py-string">'%s_strand'</tt> <tt class="py-op">%</tt> <tt class="py-name">seq_type</tt><tt class="py-op">,</tt> <tt id="link-173" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-173', 'strand', 'link-171');">strand</a></tt><tt class="py-op">)</tt> </tt>
<a name="L446"></a><tt class="py-lineno">446</tt>  <tt class="py-line"> </tt>
<a name="L447"></a><tt class="py-lineno">447</tt>  <tt class="py-line">                <tt id="link-174" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-174', 'hsp', 'link-51');">hsp</a></tt> <tt class="py-op">=</tt> <tt id="link-175" class="py-name"><a title="Bio.Blast.Record.HSP
Bio.SearchIO._model.hsp.HSP" class="py-name" href="#" onclick="return doclink('link-175', 'HSP', 'link-15');">HSP</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">frag</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L448"></a><tt class="py-lineno">448</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">attr</tt><tt class="py-op">,</tt> <tt id="link-176" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-176', 'value', 'link-81');">value</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">prev</tt><tt class="py-op">[</tt><tt class="py-string">'hsp'</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-177" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-177', 'items', 'link-120');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L449"></a><tt class="py-lineno">449</tt>  <tt class="py-line">                    <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt id="link-178" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-178', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">,</tt> <tt class="py-name">attr</tt><tt class="py-op">,</tt> <tt id="link-179" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-179', 'value', 'link-81');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L450"></a><tt class="py-lineno">450</tt>  <tt class="py-line">                <tt class="py-name">hsp_list</tt><tt class="py-op">.</tt><tt id="link-180" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-180', 'append', 'link-180');">append</a></tt><tt class="py-op">(</tt><tt id="link-181" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-181', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">)</tt> </tt>
<a name="L451"></a><tt class="py-lineno">451</tt>  <tt class="py-line"> </tt>
<a name="L452"></a><tt class="py-lineno">452</tt>  <tt class="py-line">                <tt class="py-comment"># create hit and append to temp hit container if hit_state</tt> </tt>
<a name="L453"></a><tt class="py-lineno">453</tt>  <tt class="py-line">                <tt class="py-comment"># says we're not at the same hit or at a new query</tt> </tt>
<a name="L454"></a><tt class="py-lineno">454</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">hit_state</tt> <tt class="py-op">==</tt> <tt class="py-name">state_HIT_NEW</tt><tt class="py-op">:</tt> </tt>
<a name="L455"></a><tt class="py-lineno">455</tt>  <tt class="py-line">                    <tt id="link-182" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-182', 'hit', 'link-135');">hit</a></tt> <tt class="py-op">=</tt> <tt id="link-183" class="py-name"><a title="Bio.SearchIO._model.hit.Hit" class="py-name" href="#" onclick="return doclink('link-183', 'Hit', 'link-14');">Hit</a></tt><tt class="py-op">(</tt><tt class="py-name">hsp_list</tt><tt class="py-op">)</tt> </tt>
<a name="L456"></a><tt class="py-lineno">456</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-name">attr</tt><tt class="py-op">,</tt> <tt id="link-184" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-184', 'value', 'link-81');">value</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">prev</tt><tt class="py-op">[</tt><tt class="py-string">'hit'</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-185" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-185', 'items', 'link-120');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L457"></a><tt class="py-lineno">457</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">attr</tt> <tt class="py-op">!=</tt> <tt class="py-string">'id_all'</tt><tt class="py-op">:</tt> </tt>
<a name="L458"></a><tt class="py-lineno">458</tt>  <tt class="py-line">                            <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt id="link-186" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-186', 'hit', 'link-135');">hit</a></tt><tt class="py-op">,</tt> <tt class="py-name">attr</tt><tt class="py-op">,</tt> <tt id="link-187" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-187', 'value', 'link-81');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L459"></a><tt class="py-lineno">459</tt>  <tt class="py-line">                        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L460"></a><tt class="py-lineno">460</tt>  <tt class="py-line">                            <tt class="py-comment"># not setting hit ID since it's already set from the</tt> </tt>
<a name="L461"></a><tt class="py-lineno">461</tt>  <tt class="py-line">                            <tt class="py-comment"># prev_hid above</tt> </tt>
<a name="L462"></a><tt class="py-lineno">462</tt>  <tt class="py-line">                            <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt id="link-188" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-188', 'hit', 'link-135');">hit</a></tt><tt class="py-op">,</tt> <tt class="py-string">'_id_alt'</tt><tt class="py-op">,</tt> <tt id="link-189" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-189', 'value', 'link-81');">value</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L463"></a><tt class="py-lineno">463</tt>  <tt class="py-line">                    <tt class="py-name">hit_list</tt><tt class="py-op">.</tt><tt id="link-190" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-190', 'append', 'link-180');">append</a></tt><tt class="py-op">(</tt><tt id="link-191" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-191', 'hit', 'link-135');">hit</a></tt><tt class="py-op">)</tt> </tt>
<a name="L464"></a><tt class="py-lineno">464</tt>  <tt class="py-line">                    <tt class="py-name">hsp_list</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L465"></a><tt class="py-lineno">465</tt>  <tt class="py-line">                <tt class="py-comment"># create qresult and yield if we're at a new qresult or EOF</tt> </tt>
<a name="L466"></a><tt class="py-lineno">466</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">qres_state</tt> <tt class="py-op">==</tt> <tt class="py-name">state_QRES_NEW</tt> <tt class="py-keyword">or</tt> <tt class="py-name">file_state</tt> <tt class="py-op">==</tt> <tt class="py-name">state_EOF</tt><tt class="py-op">:</tt> </tt>
<a name="L467"></a><tt class="py-lineno">467</tt>  <tt class="py-line">                    <tt class="py-name">qresult</tt> <tt class="py-op">=</tt> <tt id="link-192" class="py-name"><a title="Bio.SearchIO._model.query.QueryResult" class="py-name" href="#" onclick="return doclink('link-192', 'QueryResult', 'link-13');">QueryResult</a></tt><tt class="py-op">(</tt><tt class="py-name">hit_list</tt><tt class="py-op">,</tt> <tt class="py-name">prev_qid</tt><tt class="py-op">)</tt> </tt>
<a name="L468"></a><tt class="py-lineno">468</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-name">attr</tt><tt class="py-op">,</tt> <tt id="link-193" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-193', 'value', 'link-81');">value</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">prev</tt><tt class="py-op">[</tt><tt class="py-string">'qresult'</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-194" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-194', 'items', 'link-120');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L469"></a><tt class="py-lineno">469</tt>  <tt class="py-line">                        <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt class="py-name">qresult</tt><tt class="py-op">,</tt> <tt class="py-name">attr</tt><tt class="py-op">,</tt> <tt id="link-195" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-195', 'value', 'link-81');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L470"></a><tt class="py-lineno">470</tt>  <tt class="py-line">                    <tt class="py-keyword">yield</tt> <tt class="py-name">qresult</tt> </tt>
<a name="L471"></a><tt class="py-lineno">471</tt>  <tt class="py-line">                    <tt class="py-comment"># if current line is EOF, break</tt> </tt>
<a name="L472"></a><tt class="py-lineno">472</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">file_state</tt> <tt class="py-op">==</tt> <tt class="py-name">state_EOF</tt><tt class="py-op">:</tt> </tt>
<a name="L473"></a><tt class="py-lineno">473</tt>  <tt class="py-line">                        <tt class="py-keyword">break</tt> </tt>
<a name="L474"></a><tt class="py-lineno">474</tt>  <tt class="py-line">                    <tt class="py-name">hit_list</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L475"></a><tt class="py-lineno">475</tt>  <tt class="py-line"> </tt>
<a name="L476"></a><tt class="py-lineno">476</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-196" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-196', 'readline', 'link-103');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-197" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-197', 'strip', 'link-104');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L477"></a><tt class="py-lineno">477</tt>  <tt class="py-line"> </tt>
<a name="BlastTabParser._get_frag_frame"></a><div id="BlastTabParser._get_frag_frame-def"><a name="L478"></a><tt class="py-lineno">478</tt> <a class="py-toggle" href="#" id="BlastTabParser._get_frag_frame-toggle" onclick="return toggle('BlastTabParser._get_frag_frame');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#_get_frag_frame">_get_frag_frame</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">frag</tt><tt class="py-op">,</tt> <tt class="py-param">seq_type</tt><tt class="py-op">,</tt> <tt class="py-param">parsedict</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabParser._get_frag_frame-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlastTabParser._get_frag_frame-expanded"><a name="L479"></a><tt class="py-lineno">479</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return fragment frame for given object (PRIVATE).</tt> </tt>
<a name="L480"></a><tt class="py-lineno">480</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L481"></a><tt class="py-lineno">481</tt>  <tt class="py-line"><tt class="py-docstring">        Returns `HSPFragment` frame given the object, its sequence type,</tt> </tt>
<a name="L482"></a><tt class="py-lineno">482</tt>  <tt class="py-line"><tt class="py-docstring">        and its parsed dictionary values.</tt> </tt>
<a name="L483"></a><tt class="py-lineno">483</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L484"></a><tt class="py-lineno">484</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">seq_type</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'query'</tt><tt class="py-op">,</tt> <tt class="py-string">'hit'</tt><tt class="py-op">)</tt> </tt>
<a name="L485"></a><tt class="py-lineno">485</tt>  <tt class="py-line">        <tt id="link-198" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.frame" class="py-name" href="#" onclick="return doclink('link-198', 'frame', 'link-168');">frame</a></tt> <tt class="py-op">=</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">,</tt> <tt class="py-string">'%s_frame'</tt> <tt class="py-op">%</tt> <tt class="py-name">seq_type</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
<a name="L486"></a><tt class="py-lineno">486</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-199" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.frame" class="py-name" href="#" onclick="return doclink('link-199', 'frame', 'link-168');">frame</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L487"></a><tt class="py-lineno">487</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-200" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.frame" class="py-name" href="#" onclick="return doclink('link-200', 'frame', 'link-168');">frame</a></tt> </tt>
<a name="L488"></a><tt class="py-lineno">488</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L489"></a><tt class="py-lineno">489</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-string">'frames'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">parsedict</tt><tt class="py-op">:</tt> </tt>
<a name="L490"></a><tt class="py-lineno">490</tt>  <tt class="py-line">                <tt class="py-comment"># frames is 'x1/x2' string, x1 is query frame, x2 is hit frame</tt> </tt>
<a name="L491"></a><tt class="py-lineno">491</tt>  <tt class="py-line">                <tt class="py-name">idx</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> <tt class="py-keyword">if</tt> <tt class="py-name">seq_type</tt> <tt class="py-op">==</tt> <tt class="py-string">'query'</tt> <tt class="py-keyword">else</tt> <tt class="py-number">1</tt> </tt>
<a name="L492"></a><tt class="py-lineno">492</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">parsedict</tt><tt class="py-op">[</tt><tt class="py-string">'frames'</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-201" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-201', 'split', 'link-21');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">'/'</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-name">idx</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L493"></a><tt class="py-lineno">493</tt>  <tt class="py-line">            <tt class="py-comment"># else implicit None return</tt> </tt>
<a name="L494"></a><tt class="py-lineno">494</tt>  <tt class="py-line"> </tt>
<a name="BlastTabParser._get_frag_strand"></a><div id="BlastTabParser._get_frag_strand-def"><a name="L495"></a><tt class="py-lineno">495</tt> <a class="py-toggle" href="#" id="BlastTabParser._get_frag_strand-toggle" onclick="return toggle('BlastTabParser._get_frag_strand');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#_get_frag_strand">_get_frag_strand</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">frag</tt><tt class="py-op">,</tt> <tt class="py-param">seq_type</tt><tt class="py-op">,</tt> <tt class="py-param">parsedict</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabParser._get_frag_strand-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlastTabParser._get_frag_strand-expanded"><a name="L496"></a><tt class="py-lineno">496</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return fragment strand for given object (PRIVATE).</tt> </tt>
<a name="L497"></a><tt class="py-lineno">497</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L498"></a><tt class="py-lineno">498</tt>  <tt class="py-line"><tt class="py-docstring">        Returns `HSPFragment` strand given the object, its sequence type,</tt> </tt>
<a name="L499"></a><tt class="py-lineno">499</tt>  <tt class="py-line"><tt class="py-docstring">        and its parsed dictionary values.</tt> </tt>
<a name="L500"></a><tt class="py-lineno">500</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L501"></a><tt class="py-lineno">501</tt>  <tt class="py-line">        <tt class="py-comment"># NOTE: this will never set the strands as 0 for protein</tt> </tt>
<a name="L502"></a><tt class="py-lineno">502</tt>  <tt class="py-line">        <tt class="py-comment"># queries / hits, since we can't detect the blast flavors</tt> </tt>
<a name="L503"></a><tt class="py-lineno">503</tt>  <tt class="py-line">        <tt class="py-comment"># from the columns alone.</tt> </tt>
<a name="L504"></a><tt class="py-lineno">504</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">seq_type</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'query'</tt><tt class="py-op">,</tt> <tt class="py-string">'hit'</tt><tt class="py-op">)</tt> </tt>
<a name="L505"></a><tt class="py-lineno">505</tt>  <tt class="py-line">        <tt id="link-202" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-202', 'strand', 'link-171');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">,</tt> <tt class="py-string">'%s_strand'</tt> <tt class="py-op">%</tt> <tt class="py-name">seq_type</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
<a name="L506"></a><tt class="py-lineno">506</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-203" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-203', 'strand', 'link-171');">strand</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L507"></a><tt class="py-lineno">507</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-204" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-204', 'strand', 'link-171');">strand</a></tt> </tt>
<a name="L508"></a><tt class="py-lineno">508</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L509"></a><tt class="py-lineno">509</tt>  <tt class="py-line">            <tt class="py-comment"># using parsedict instead of the fragment object since</tt> </tt>
<a name="L510"></a><tt class="py-lineno">510</tt>  <tt class="py-line">            <tt class="py-comment"># we need the unadjusted coordinated values</tt> </tt>
<a name="L511"></a><tt class="py-lineno">511</tt>  <tt class="py-line">            <tt id="link-205" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.start=Bio.SeqFeature.CompoundLocation-class.html#start,Variable Bio.SeqFeature.FeatureLocation.start=Bio.SeqFeature.FeatureLocation-class.html#start"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-205', 'start', 'link-205');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">parsedict</tt><tt class="py-op">.</tt><tt id="link-206" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-206', 'get', 'link-150');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'%s_start'</tt> <tt class="py-op">%</tt> <tt class="py-name">seq_type</tt><tt class="py-op">)</tt> </tt>
<a name="L512"></a><tt class="py-lineno">512</tt>  <tt class="py-line">            <tt id="link-207" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.end=Bio.SeqFeature.CompoundLocation-class.html#end,Variable Bio.SeqFeature.FeatureLocation.end=Bio.SeqFeature.FeatureLocation-class.html#end"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-207', 'end', 'link-207');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">parsedict</tt><tt class="py-op">.</tt><tt id="link-208" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-208', 'get', 'link-150');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'%s_end'</tt> <tt class="py-op">%</tt> <tt class="py-name">seq_type</tt><tt class="py-op">)</tt> </tt>
<a name="L513"></a><tt class="py-lineno">513</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-209" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-209', 'start', 'link-205');">start</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> <tt id="link-210" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-210', 'end', 'link-207');">end</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L514"></a><tt class="py-lineno">514</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-number">1</tt> <tt class="py-keyword">if</tt> <tt id="link-211" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-211', 'start', 'link-205');">start</a></tt> <tt class="py-op">&lt;=</tt> <tt id="link-212" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-212', 'end', 'link-207');">end</a></tt> <tt class="py-keyword">else</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt> </tt>
</div></div><a name="L515"></a><tt class="py-lineno">515</tt>  <tt class="py-line">            <tt class="py-comment"># else implicit None return</tt> </tt>
<a name="L516"></a><tt class="py-lineno">516</tt>  <tt class="py-line"> </tt>
<a name="L517"></a><tt class="py-lineno">517</tt>  <tt class="py-line"> </tt>
<a name="BlastTabIndexer"></a><div id="BlastTabIndexer-def"><a name="L518"></a><tt class="py-lineno">518</tt> <a class="py-toggle" href="#" id="BlastTabIndexer-toggle" onclick="return toggle('BlastTabIndexer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer-class.html">BlastTabIndexer</a><tt class="py-op">(</tt><tt class="py-base-class">SearchIndexer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabIndexer-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="BlastTabIndexer-expanded"><a name="L519"></a><tt class="py-lineno">519</tt>  <tt class="py-line">    <tt class="py-docstring">"""Indexer class for BLAST+ tab output."""</tt> </tt>
<a name="L520"></a><tt class="py-lineno">520</tt>  <tt class="py-line"> </tt>
<a name="L521"></a><tt class="py-lineno">521</tt>  <tt class="py-line">    <tt id="link-213" class="py-name" targets="Variable Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer._parser=Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer-class.html#_parser"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer._parser" class="py-name" href="#" onclick="return doclink('link-213', '_parser', 'link-213');">_parser</a></tt> <tt class="py-op">=</tt> <tt id="link-214" class="py-name" targets="Class Bio.SearchIO.BlastIO.blast_tab.BlastTabParser=Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html"><a title="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser" class="py-name" href="#" onclick="return doclink('link-214', 'BlastTabParser', 'link-214');">BlastTabParser</a></tt> </tt>
<a name="L522"></a><tt class="py-lineno">522</tt>  <tt class="py-line"> </tt>
<a name="BlastTabIndexer.__init__"></a><div id="BlastTabIndexer.__init__-def"><a name="L523"></a><tt class="py-lineno">523</tt> <a class="py-toggle" href="#" id="BlastTabIndexer.__init__-toggle" onclick="return toggle('BlastTabIndexer.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">filename</tt><tt class="py-op">,</tt> <tt class="py-param">comments</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> <tt class="py-param">fields</tt><tt class="py-op">=</tt><tt id="link-215" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._DEFAULT_FIELDS" class="py-name" href="#" onclick="return doclink('link-215', '_DEFAULT_FIELDS', 'link-44');">_DEFAULT_FIELDS</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabIndexer.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlastTabIndexer.__init__-expanded"><a name="L524"></a><tt class="py-lineno">524</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the class."""</tt> </tt>
<a name="L525"></a><tt class="py-lineno">525</tt>  <tt class="py-line">        <tt id="link-216" class="py-name"><a title="Bio.SearchIO._index.SearchIndexer" class="py-name" href="#" onclick="return doclink('link-216', 'SearchIndexer', 'link-9');">SearchIndexer</a></tt><tt class="py-op">.</tt><tt id="link-217" class="py-name" targets="Method Bio.Affy.CelFile.ParserError.__init__()=Bio.Affy.CelFile.ParserError-class.html#__init__,Method Bio.Affy.CelFile.Record.__init__()=Bio.Affy.CelFile.Record-class.html#__init__,Method Bio.Align.AlignInfo.PSSM.__init__()=Bio.Align.AlignInfo.PSSM-class.html#__init__,Method Bio.Align.AlignInfo.SummaryInfo.__init__()=Bio.Align.AlignInfo.SummaryInfo-class.html#__init__,Method Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline.__init__()=Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline-class.html#__init__,Method Bio.Align.Applications._Clustalw.ClustalwCommandline.__init__()=Bio.Align.Applications._Clustalw.ClustalwCommandline-class.html#__init__,Method Bio.Align.Applications._Dialign.DialignCommandline.__init__()=Bio.Align.Applications._Dialign.DialignCommandline-class.html#__init__,Method Bio.Align.Applications._MSAProbs.MSAProbsCommandline.__init__()=Bio.Align.Applications._MSAProbs.MSAProbsCommandline-class.html#__init__,Method Bio.Align.Applications._Mafft.MafftCommandline.__init__()=Bio.Align.Applications._Mafft.MafftCommandline-class.html#__init__,Method Bio.Align.Applications._Muscle.MuscleCommandline.__init__()=Bio.Align.Applications._Muscle.MuscleCommandline-class.html#__init__,Method Bio.Align.Applications._Prank.PrankCommandline.__init__()=Bio.Align.Applications._Prank.PrankCommandline-class.html#__init__,Method Bio.Align.Applications._Probcons.ProbconsCommandline.__init__()=Bio.Align.Applications._Probcons.ProbconsCommandline-class.html#__init__,Method Bio.Align.Applications._TCoffee.TCoffeeCommandline.__init__()=Bio.Align.Applications._TCoffee.TCoffeeCommandline-class.html#__init__,Method Bio.Align.MultipleSeqAlignment.__init__()=Bio.Align.MultipleSeqAlignment-class.html#__init__,Method Bio.Align.PairwiseAlignment.__init__()=Bio.Align.PairwiseAlignment-class.html#__init__,Method Bio.Align.PairwiseAlignments.__init__()=Bio.Align.PairwiseAlignments-class.html#__init__,Method Bio.AlignIO.Interfaces.AlignmentIterator.__init__()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#__init__,Method Bio.AlignIO.Interfaces.AlignmentWriter.__init__()=Bio.AlignIO.Interfaces.AlignmentWriter-class.html#__init__,Method Bio.AlignIO.Interfaces.SequentialAlignmentWriter.__init__()=Bio.AlignIO.Interfaces.SequentialAlignmentWriter-class.html#__init__,Method Bio.AlignIO.MafIO.MafIndex.__init__()=Bio.AlignIO.MafIO.MafIndex-class.html#__init__,Method Bio.AlignIO.MauveIO.MauveWriter.__init__()=Bio.AlignIO.MauveIO.MauveWriter-class.html#__init__,Method Bio.Alphabet.AlphabetEncoder.__init__()=Bio.Alphabet.AlphabetEncoder-class.html#__init__,Method Bio.Alphabet.Gapped.__init__()=Bio.Alphabet.Gapped-class.html#__init__,Method Bio.Alphabet.HasStopCodon.__init__()=Bio.Alphabet.HasStopCodon-class.html#__init__,Method Bio.Application.AbstractCommandline.__init__()=Bio.Application.AbstractCommandline-class.html#__init__,Method Bio.Application.ApplicationError.__init__()=Bio.Application.ApplicationError-class.html#__init__,Method Bio.Application._AbstractParameter.__init__()=Bio.Application._AbstractParameter-class.html#__init__,Method Bio.Application._Argument.__init__()=Bio.Application._Argument-class.html#__init__,Method Bio.Application._Option.__init__()=Bio.Application._Option-class.html#__init__,Method Bio.Application._StaticArgument.__init__()=Bio.Application._StaticArgument-class.html#__init__,Method Bio.Application._Switch.__init__()=Bio.Application._Switch-class.html#__init__,Method Bio.Blast.Applications.NcbiblastformatterCommandline.__init__()=Bio.Blast.Applications.NcbiblastformatterCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastnCommandline.__init__()=Bio.Blast.Applications.NcbiblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastpCommandline.__init__()=Bio.Blast.Applications.NcbiblastpCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastxCommandline.__init__()=Bio.Blast.Applications.NcbiblastxCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbideltablastCommandline.__init__()=Bio.Blast.Applications.NcbideltablastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbipsiblastCommandline.__init__()=Bio.Blast.Applications.NcbipsiblastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbirpsblastCommandline.__init__()=Bio.Blast.Applications.NcbirpsblastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbirpstblastnCommandline.__init__()=Bio.Blast.Applications.NcbirpstblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbitblastnCommandline.__init__()=Bio.Blast.Applications.NcbitblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbitblastxCommandline.__init__()=Bio.Blast.Applications.NcbitblastxCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbibaseblastCommandline.__init__()=Bio.Blast.Applications._NcbibaseblastCommandline-class.html#__init__,Method Bio.Blast.Applications._Ncbiblast2SeqCommandline.__init__()=Bio.Blast.Applications._Ncbiblast2SeqCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbiblastCommandline.__init__()=Bio.Blast.Applications._NcbiblastCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbiblastMain2SeqCommandline.__init__()=Bio.Blast.Applications._NcbiblastMain2SeqCommandline-class.html#__init__,Method Bio.Blast.NCBIStandalone.BlastErrorParser.__init__()=Bio.Blast.NCBIStandalone.BlastErrorParser-class.html#__init__,Method Bio.Blast.NCBIStandalone.BlastParser.__init__()=Bio.Blast.NCBIStandalone.BlastParser-class.html#__init__,Method Bio.Blast.NCBIStandalone.Iterator.__init__()=Bio.Blast.NCBIStandalone.Iterator-class.html#__init__,Method Bio.Blast.NCBIStandalone.PSIBlastParser.__init__()=Bio.Blast.NCBIStandalone.PSIBlastParser-class.html#__init__,Method Bio.Blast.NCBIStandalone._BlastConsumer.__init__()=Bio.Blast.NCBIStandalone._BlastConsumer-class.html#__init__,Method Bio.Blast.NCBIStandalone._BlastErrorConsumer.__init__()=Bio.Blast.NCBIStandalone._BlastErrorConsumer-class.html#__init__,Method Bio.Blast.NCBIStandalone._PSIBlastConsumer.__init__()=Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html#__init__,Method Bio.Blast.NCBIXML.BlastParser.__init__()=Bio.Blast.NCBIXML.BlastParser-class.html#__init__,Method Bio.Blast.NCBIXML._XMLparser.__init__()=Bio.Blast.NCBIXML._XMLparser-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableEntry.__init__()=Bio.Blast.ParseBlastTable.BlastTableEntry-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableReader.__init__()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableRec.__init__()=Bio.Blast.ParseBlastTable.BlastTableRec-class.html#__init__,Method Bio.Blast.Record.Alignment.__init__()=Bio.Blast.Record.Alignment-class.html#__init__,Method Bio.Blast.Record.Blast.__init__()=Bio.Blast.Record.Blast-class.html#__init__,Method Bio.Blast.Record.DatabaseReport.__init__()=Bio.Blast.Record.DatabaseReport-class.html#__init__,Method Bio.Blast.Record.Description.__init__()=Bio.Blast.Record.Description-class.html#__init__,Method Bio.Blast.Record.HSP.__init__()=Bio.Blast.Record.HSP-class.html#__init__,Method Bio.Blast.Record.Header.__init__()=Bio.Blast.Record.Header-class.html#__init__,Method Bio.Blast.Record.MultipleAlignment.__init__()=Bio.Blast.Record.MultipleAlignment-class.html#__init__,Method Bio.Blast.Record.PSIBlast.__init__()=Bio.Blast.Record.PSIBlast-class.html#__init__,Method Bio.Blast.Record.Parameters.__init__()=Bio.Blast.Record.Parameters-class.html#__init__,Method Bio.Blast.Record.Round.__init__()=Bio.Blast.Record.Round-class.html#__init__,Method Bio.CAPS.CAPSMap.__init__()=Bio.CAPS.CAPSMap-class.html#__init__,Method Bio.CAPS.DifferentialCutsite.__init__()=Bio.CAPS.DifferentialCutsite-class.html#__init__,Method Bio.Cluster.Record.__init__()=Bio.Cluster.Record-class.html#__init__,Method Bio.Compass.Record.__init__()=Bio.Compass.Record-class.html#__init__,Method Bio.Crystal.Chain.__init__()=Bio.Crystal.Chain-class.html#__init__,Method Bio.Crystal.Crystal.__init__()=Bio.Crystal.Crystal-class.html#__init__,Method Bio.Crystal.Hetero.__init__()=Bio.Crystal.Hetero-class.html#__init__,Method Bio.Data.CodonTable.AmbiguousCodonTable.__init__()=Bio.Data.CodonTable.AmbiguousCodonTable-class.html#__init__,Method Bio.Data.CodonTable.AmbiguousForwardTable.__init__()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#__init__,Method Bio.Data.CodonTable.CodonTable.__init__()=Bio.Data.CodonTable.CodonTable-class.html#__init__,Method Bio.Data.CodonTable.NCBICodonTable.__init__()=Bio.Data.CodonTable.NCBICodonTable-class.html#__init__,Method Bio.Emboss.Applications.DiffseqCommandline.__init__()=Bio.Emboss.Applications.DiffseqCommandline-class.html#__init__,Method Bio.Emboss.Applications.EInvertedCommandline.__init__()=Bio.Emboss.Applications.EInvertedCommandline-class.html#__init__,Method Bio.Emboss.Applications.ETandemCommandline.__init__()=Bio.Emboss.Applications.ETandemCommandline-class.html#__init__,Method Bio.Emboss.Applications.Est2GenomeCommandline.__init__()=Bio.Emboss.Applications.Est2GenomeCommandline-class.html#__init__,Method Bio.Emboss.Applications.FConsenseCommandline.__init__()=Bio.Emboss.Applications.FConsenseCommandline-class.html#__init__,Method Bio.Emboss.Applications.FDNADistCommandline.__init__()=Bio.Emboss.Applications.FDNADistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FDNAParsCommandline.__init__()=Bio.Emboss.Applications.FDNAParsCommandline-class.html#__init__,Method Bio.Emboss.Applications.FNeighborCommandline.__init__()=Bio.Emboss.Applications.FNeighborCommandline-class.html#__init__,Method Bio.Emboss.Applications.FProtDistCommandline.__init__()=Bio.Emboss.Applications.FProtDistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FProtParsCommandline.__init__()=Bio.Emboss.Applications.FProtParsCommandline-class.html#__init__,Method Bio.Emboss.Applications.FSeqBootCommandline.__init__()=Bio.Emboss.Applications.FSeqBootCommandline-class.html#__init__,Method Bio.Emboss.Applications.FTreeDistCommandline.__init__()=Bio.Emboss.Applications.FTreeDistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FuzznucCommandline.__init__()=Bio.Emboss.Applications.FuzznucCommandline-class.html#__init__,Method Bio.Emboss.Applications.FuzzproCommandline.__init__()=Bio.Emboss.Applications.FuzzproCommandline-class.html#__init__,Method Bio.Emboss.Applications.IepCommandline.__init__()=Bio.Emboss.Applications.IepCommandline-class.html#__init__,Method Bio.Emboss.Applications.NeedleCommandline.__init__()=Bio.Emboss.Applications.NeedleCommandline-class.html#__init__,Method Bio.Emboss.Applications.NeedleallCommandline.__init__()=Bio.Emboss.Applications.NeedleallCommandline-class.html#__init__,Method Bio.Emboss.Applications.PalindromeCommandline.__init__()=Bio.Emboss.Applications.PalindromeCommandline-class.html#__init__,Method Bio.Emboss.Applications.Primer3Commandline.__init__()=Bio.Emboss.Applications.Primer3Commandline-class.html#__init__,Method Bio.Emboss.Applications.PrimerSearchCommandline.__init__()=Bio.Emboss.Applications.PrimerSearchCommandline-class.html#__init__,Method Bio.Emboss.Applications.SeqmatchallCommandline.__init__()=Bio.Emboss.Applications.SeqmatchallCommandline-class.html#__init__,Method Bio.Emboss.Applications.SeqretCommandline.__init__()=Bio.Emboss.Applications.SeqretCommandline-class.html#__init__,Method Bio.Emboss.Applications.StretcherCommandline.__init__()=Bio.Emboss.Applications.StretcherCommandline-class.html#__init__,Method Bio.Emboss.Applications.TranalignCommandline.__init__()=Bio.Emboss.Applications.TranalignCommandline-class.html#__init__,Method Bio.Emboss.Applications.WaterCommandline.__init__()=Bio.Emboss.Applications.WaterCommandline-class.html#__init__,Method Bio.Emboss.Applications._EmbossCommandLine.__init__()=Bio.Emboss.Applications._EmbossCommandLine-class.html#__init__,Method Bio.Emboss.Applications._EmbossMinimalCommandLine.__init__()=Bio.Emboss.Applications._EmbossMinimalCommandLine-class.html#__init__,Method Bio.Emboss.Primer3.Primers.__init__()=Bio.Emboss.Primer3.Primers-class.html#__init__,Method Bio.Emboss.Primer3.Record.__init__()=Bio.Emboss.Primer3.Record-class.html#__init__,Method Bio.Emboss.PrimerSearch.Amplifier.__init__()=Bio.Emboss.PrimerSearch.Amplifier-class.html#__init__,Method Bio.Emboss.PrimerSearch.InputRecord.__init__()=Bio.Emboss.PrimerSearch.InputRecord-class.html#__init__,Method Bio.Emboss.PrimerSearch.OutputRecord.__init__()=Bio.Emboss.PrimerSearch.OutputRecord-class.html#__init__,Method Bio.Entrez.Parser.Consumer.__init__()=Bio.Entrez.Parser.Consumer-class.html#__init__,Method Bio.Entrez.Parser.CorruptedXMLError.__init__()=Bio.Entrez.Parser.CorruptedXMLError-class.html#__init__,Method Bio.Entrez.Parser.DataHandler.__init__()=Bio.Entrez.Parser.DataHandler-class.html#__init__,Method Bio.Entrez.Parser.DictionaryConsumer.__init__()=Bio.Entrez.Parser.DictionaryConsumer-class.html#__init__,Method Bio.Entrez.Parser.ErrorConsumer.__init__()=Bio.Entrez.Parser.ErrorConsumer-class.html#__init__,Method Bio.Entrez.Parser.IntegerConsumer.__init__()=Bio.Entrez.Parser.IntegerConsumer-class.html#__init__,Method Bio.Entrez.Parser.ListConsumer.__init__()=Bio.Entrez.Parser.ListConsumer-class.html#__init__,Method Bio.Entrez.Parser.NotXMLError.__init__()=Bio.Entrez.Parser.NotXMLError-class.html#__init__,Method Bio.Entrez.Parser.StringConsumer.__init__()=Bio.Entrez.Parser.StringConsumer-class.html#__init__,Method Bio.Entrez.Parser.ValidationError.__init__()=Bio.Entrez.Parser.ValidationError-class.html#__init__,Method Bio.ExPASy.Enzyme.Record.__init__()=Bio.ExPASy.Enzyme.Record-class.html#__init__,Method Bio.ExPASy.Prodoc.Record.__init__()=Bio.ExPASy.Prodoc.Record-class.html#__init__,Method Bio.ExPASy.Prodoc.Reference.__init__()=Bio.ExPASy.Prodoc.Reference-class.html#__init__,Method Bio.ExPASy.Prosite.Record.__init__()=Bio.ExPASy.Prosite.Record-class.html#__init__,Method Bio.ExPASy.ScanProsite.ContentHandler.__init__()=Bio.ExPASy.ScanProsite.ContentHandler-class.html#__init__,Method Bio.ExPASy.ScanProsite.Parser.__init__()=Bio.ExPASy.ScanProsite.Parser-class.html#__init__,Method Bio.ExPASy.ScanProsite.Record.__init__()=Bio.ExPASy.ScanProsite.Record-class.html#__init__,Method Bio.ExPASy.cellosaurus.Record.__init__()=Bio.ExPASy.cellosaurus.Record-class.html#__init__,Method Bio.FSSP.FSSPAlignDict.__init__()=Bio.FSSP.FSSPAlignDict-class.html#__init__,Method Bio.FSSP.FSSPAlignRec.__init__()=Bio.FSSP.FSSPAlignRec-class.html#__init__,Method Bio.FSSP.FSSPHeader.__init__()=Bio.FSSP.FSSPHeader-class.html#__init__,Method Bio.FSSP.FSSPSumRec.__init__()=Bio.FSSP.FSSPSumRec-class.html#__init__,Method Bio.FSSP.FSSPTools.FSSPMultAlign.__init__()=Bio.FSSP.FSSPTools.FSSPMultAlign-class.html#__init__,Method Bio.FSSP.PosAlign.__init__()=Bio.FSSP.PosAlign-class.html#__init__,Method Bio.FSSP.fssp_rec.fff_rec.__init__()=Bio.FSSP.fssp_rec.fff_rec-class.html#__init__,Method Bio.File.UndoHandle.__init__()=Bio.File.UndoHandle-class.html#__init__,Method Bio.File._IndexedSeqFileDict.__init__()=Bio.File._IndexedSeqFileDict-class.html#__init__,Method Bio.File._SQLiteManySeqFilesDict.__init__()=Bio.File._SQLiteManySeqFilesDict-class.html#__init__,Method Bio.GA.Crossover.General.SafeFitnessCrossover.__init__()=Bio.GA.Crossover.General.SafeFitnessCrossover-class.html#__init__,Method Bio.GA.Crossover.GeneralPoint.GeneralPointCrossover.__init__()=Bio.GA.Crossover.GeneralPoint.GeneralPointCrossover-class.html#__init__,Method Bio.GA.Crossover.GeneralPoint.InterleaveCrossover.__init__()=Bio.GA.Crossover.GeneralPoint.InterleaveCrossover-class.html#__init__,Method Bio.GA.Crossover.Point.SinglePointCrossover.__init__()=Bio.GA.Crossover.Point.SinglePointCrossover-class.html#__init__,Method Bio.GA.Crossover.TwoPoint.TwoPointCrossover.__init__()=Bio.GA.Crossover.TwoPoint.TwoPointCrossover-class.html#__init__,Method Bio.GA.Crossover.Uniform.UniformCrossover.__init__()=Bio.GA.Crossover.Uniform.UniformCrossover-class.html#__init__,Method Bio.GA.Evolver.GenerationEvolver.__init__()=Bio.GA.Evolver.GenerationEvolver-class.html#__init__,Method Bio.GA.Evolver.SteadyStateEvolver.__init__()=Bio.GA.Evolver.SteadyStateEvolver-class.html#__init__,Method Bio.GA.Mutation.General.SafeFitnessMutation.__init__()=Bio.GA.Mutation.General.SafeFitnessMutation-class.html#__init__,Method Bio.GA.Mutation.Simple.ConversionMutation.__init__()=Bio.GA.Mutation.Simple.ConversionMutation-class.html#__init__,Method Bio.GA.Mutation.Simple.SinglePositionMutation.__init__()=Bio.GA.Mutation.Simple.SinglePositionMutation-class.html#__init__,Method Bio.GA.Organism.Organism.__init__()=Bio.GA.Organism.Organism-class.html#__init__,Method Bio.GA.Repair.Stabilizing.AmbiguousRepair.__init__()=Bio.GA.Repair.Stabilizing.AmbiguousRepair-class.html#__init__,Method Bio.GA.Selection.Abstract.AbstractSelection.__init__()=Bio.GA.Selection.Abstract.AbstractSelection-class.html#__init__,Method Bio.GA.Selection.Diversity.DiversitySelection.__init__()=Bio.GA.Selection.Diversity.DiversitySelection-class.html#__init__,Method Bio.GA.Selection.RouletteWheel.RouletteWheelSelection.__init__()=Bio.GA.Selection.RouletteWheel.RouletteWheelSelection-class.html#__init__,Method Bio.GA.Selection.Tournament.TournamentSelection.__init__()=Bio.GA.Selection.Tournament.TournamentSelection-class.html#__init__,Method Bio.GenBank.FeatureParser.__init__()=Bio.GenBank.FeatureParser-class.html#__init__,Method Bio.GenBank.Iterator.__init__()=Bio.GenBank.Iterator-class.html#__init__,Method Bio.GenBank.Record.Feature.__init__()=Bio.GenBank.Record.Feature-class.html#__init__,Method Bio.GenBank.Record.Qualifier.__init__()=Bio.GenBank.Record.Qualifier-class.html#__init__,Method Bio.GenBank.Record.Record.__init__()=Bio.GenBank.Record.Record-class.html#__init__,Method Bio.GenBank.Record.Reference.__init__()=Bio.GenBank.Record.Reference-class.html#__init__,Method Bio.GenBank.RecordParser.__init__()=Bio.GenBank.RecordParser-class.html#__init__,Method Bio.GenBank.Scanner.InsdcScanner.__init__()=Bio.GenBank.Scanner.InsdcScanner-class.html#__init__,Method Bio.GenBank._BaseGenBankConsumer.__init__()=Bio.GenBank._BaseGenBankConsumer-class.html#__init__,Method Bio.GenBank._FeatureConsumer.__init__()=Bio.GenBank._FeatureConsumer-class.html#__init__,Method Bio.GenBank._RecordConsumer.__init__()=Bio.GenBank._RecordConsumer-class.html#__init__,Method Bio.GenBank.utils.FeatureValueCleaner.__init__()=Bio.GenBank.utils.FeatureValueCleaner-class.html#__init__,Method Bio.Geo.Record.Record.__init__()=Bio.Geo.Record.Record-class.html#__init__,Method Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment.__init__()=Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome.Chromosome.__init__()=Bio.Graphics.BasicChromosome.Chromosome-class.html#__init__,Method Bio.Graphics.BasicChromosome.ChromosomeSegment.__init__()=Bio.Graphics.BasicChromosome.ChromosomeSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome.Organism.__init__()=Bio.Graphics.BasicChromosome.Organism-class.html#__init__,Method Bio.Graphics.BasicChromosome.TelomereSegment.__init__()=Bio.Graphics.BasicChromosome.TelomereSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome._ChromosomeComponent.__init__()=Bio.Graphics.BasicChromosome._ChromosomeComponent-class.html#__init__,Method Bio.Graphics.ColorSpiral.ColorSpiral.__init__()=Bio.Graphics.ColorSpiral.ColorSpiral-class.html#__init__,Method Bio.Graphics.Comparative.ComparativeScatterPlot.__init__()=Bio.Graphics.Comparative.ComparativeScatterPlot-class.html#__init__,Method Bio.Graphics.DisplayRepresentation.ChromosomeCounts.__init__()=Bio.Graphics.DisplayRepresentation.ChromosomeCounts-class.html#__init__,Method Bio.Graphics.Distribution.BarChartDistribution.__init__()=Bio.Graphics.Distribution.BarChartDistribution-class.html#__init__,Method Bio.Graphics.Distribution.DistributionPage.__init__()=Bio.Graphics.Distribution.DistributionPage-class.html#__init__,Method Bio.Graphics.Distribution.LineDistribution.__init__()=Bio.Graphics.Distribution.LineDistribution-class.html#__init__,Method Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.__init__()=Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.__init__()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.__init__()=Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#__init__,Method Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.__init__()=Bio.Graphics.GenomeDiagram._CrossLink.CrossLink-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.__init__()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Feature.Feature.__init__()=Bio.Graphics.GenomeDiagram._Feature.Feature-class.html#__init__,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__init__()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.__init__()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#__init__,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__init__()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#__init__,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.__init__()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Track.Track.__init__()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#__init__,Method Bio.Graphics.KGML_vis.KGMLCanvas.__init__()=Bio.Graphics.KGML_vis.KGMLCanvas-class.html#__init__,Method Bio.HMM.DynamicProgramming.AbstractDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.AbstractDPAlgorithms-class.html#__init__,Method Bio.HMM.DynamicProgramming.LogDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.LogDPAlgorithms-class.html#__init__,Method Bio.HMM.DynamicProgramming.ScaledDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.ScaledDPAlgorithms-class.html#__init__,Method Bio.HMM.MarkovModel.HiddenMarkovModel.__init__()=Bio.HMM.MarkovModel.HiddenMarkovModel-class.html#__init__,Method Bio.HMM.MarkovModel.MarkovModelBuilder.__init__()=Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#__init__,Method Bio.HMM.Trainer.AbstractTrainer.__init__()=Bio.HMM.Trainer.AbstractTrainer-class.html#__init__,Method Bio.HMM.Trainer.BaumWelchTrainer.__init__()=Bio.HMM.Trainer.BaumWelchTrainer-class.html#__init__,Method Bio.HMM.Trainer.KnownStateTrainer.__init__()=Bio.HMM.Trainer.KnownStateTrainer-class.html#__init__,Method Bio.HMM.Trainer.TrainingSequence.__init__()=Bio.HMM.Trainer.TrainingSequence-class.html#__init__,Method Bio.Index._InMemoryIndex.__init__()=Bio.Index._InMemoryIndex-class.html#__init__,Method Bio.Index._ShelveIndex.__init__()=Bio.Index._ShelveIndex-class.html#__init__,Method Bio.KDTree.KDTree'.KDTree.__init__()=Bio.KDTree.KDTree%27.KDTree-class.html#__init__,Method Bio.KEGG.Compound.Record.__init__()=Bio.KEGG.Compound.Record-class.html#__init__,Method Bio.KEGG.Enzyme.Record.__init__()=Bio.KEGG.Enzyme.Record-class.html#__init__,Method Bio.KEGG.Gene.Record.__init__()=Bio.KEGG.Gene.Record-class.html#__init__,Method Bio.KEGG.KGML.KGML_parser.KGMLParser.__init__()=Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Component.__init__()=Bio.KEGG.KGML.KGML_pathway.Component-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Entry.__init__()=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Graphics.__init__()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Pathway.__init__()=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Reaction.__init__()=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Relation.__init__()=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#__init__,Method Bio.LogisticRegression.LogisticRegression.__init__()=Bio.LogisticRegression.LogisticRegression-class.html#__init__,Method Bio.MarkovModel.MarkovModel.__init__()=Bio.MarkovModel.MarkovModel-class.html#__init__,Method Bio.MaxEntropy.MaxEntropy.__init__()=Bio.MaxEntropy.MaxEntropy-class.html#__init__,Method Bio.NMR.xpktools.Peaklist.__init__()=Bio.NMR.xpktools.Peaklist-class.html#__init__,Method Bio.NMR.xpktools.XpkEntry.__init__()=Bio.NMR.xpktools.XpkEntry-class.html#__init__,Method Bio.NaiveBayes.NaiveBayes.__init__()=Bio.NaiveBayes.NaiveBayes-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.AbstractLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.AbstractLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.InputLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Network.BasicNetwork.__init__()=Bio.NeuralNetwork.BackPropagation.Network.BasicNetwork-class.html#__init__,Method Bio.NeuralNetwork.Gene.Motif.MotifCoder.__init__()=Bio.NeuralNetwork.Gene.Motif.MotifCoder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Motif.MotifFinder.__init__()=Bio.NeuralNetwork.Gene.Motif.MotifFinder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.__init__()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#__init__,Method Bio.NeuralNetwork.Gene.Pattern.PatternRepository.__init__()=Bio.NeuralNetwork.Gene.Pattern.PatternRepository-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness.__init__()=Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder.__init__()=Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness.__init__()=Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator.__init__()=Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.Schema.__init__()=Bio.NeuralNetwork.Gene.Schema.Schema-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SchemaCoder.__init__()=Bio.NeuralNetwork.Gene.Schema.SchemaCoder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SchemaFactory.__init__()=Bio.NeuralNetwork.Gene.Schema.SchemaFactory-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SchemaFinder.__init__()=Bio.NeuralNetwork.Gene.Schema.SchemaFinder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SimpleFinisher.__init__()=Bio.NeuralNetwork.Gene.Schema.SimpleFinisher-class.html#__init__,Method Bio.NeuralNetwork.Gene.Signature.SignatureCoder.__init__()=Bio.NeuralNetwork.Gene.Signature.SignatureCoder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Signature.SignatureFinder.__init__()=Bio.NeuralNetwork.Gene.Signature.SignatureFinder-class.html#__init__,Method Bio.NeuralNetwork.StopTraining.ValidationIncreaseStop.__init__()=Bio.NeuralNetwork.StopTraining.ValidationIncreaseStop-class.html#__init__,Method Bio.NeuralNetwork.Training.ExampleManager.__init__()=Bio.NeuralNetwork.Training.ExampleManager-class.html#__init__,Method Bio.NeuralNetwork.Training.TrainingExample.__init__()=Bio.NeuralNetwork.Training.TrainingExample-class.html#__init__,Method Bio.Nexus.Nexus.Block.__init__()=Bio.Nexus.Nexus.Block-class.html#__init__,Method Bio.Nexus.Nexus.CharBuffer.__init__()=Bio.Nexus.Nexus.CharBuffer-class.html#__init__,Method Bio.Nexus.Nexus.Commandline.__init__()=Bio.Nexus.Nexus.Commandline-class.html#__init__,Method Bio.Nexus.Nexus.Nexus.__init__()=Bio.Nexus.Nexus.Nexus-class.html#__init__,Method Bio.Nexus.Nexus.StepMatrix.__init__()=Bio.Nexus.Nexus.StepMatrix-class.html#__init__,Method Bio.Nexus.Nodes.Chain.__init__()=Bio.Nexus.Nodes.Chain-class.html#__init__,Method Bio.Nexus.Nodes.Node.__init__()=Bio.Nexus.Nodes.Node-class.html#__init__,Method Bio.Nexus.StandardData.StandardData.__init__()=Bio.Nexus.StandardData.StandardData-class.html#__init__,Method Bio.Nexus.Trees.NodeData.__init__()=Bio.Nexus.Trees.NodeData-class.html#__init__,Method Bio.Nexus.Trees.Tree.__init__()=Bio.Nexus.Trees.Tree-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractPropertyMap-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap-class.html#__init__,Method Bio.PDB.Atom.Atom.__init__()=Bio.PDB.Atom.Atom-class.html#__init__,Method Bio.PDB.Atom.DisorderedAtom.__init__()=Bio.PDB.Atom.DisorderedAtom-class.html#__init__,Method Bio.PDB.Chain.Chain.__init__()=Bio.PDB.Chain.Chain-class.html#__init__,Method Bio.PDB.DSSP'.DSSP.__init__()=Bio.PDB.DSSP%27.DSSP-class.html#__init__,Method Bio.PDB.Dice.ChainSelector.__init__()=Bio.PDB.Dice.ChainSelector-class.html#__init__,Method Bio.PDB.Entity.DisorderedEntityWrapper.__init__()=Bio.PDB.Entity.DisorderedEntityWrapper-class.html#__init__,Method Bio.PDB.Entity.Entity.__init__()=Bio.PDB.Entity.Entity-class.html#__init__,Method Bio.PDB.FragmentMapper'.Fragment.__init__()=Bio.PDB.FragmentMapper%27.Fragment-class.html#__init__,Method Bio.PDB.FragmentMapper'.FragmentMapper.__init__()=Bio.PDB.FragmentMapper%27.FragmentMapper-class.html#__init__,Method Bio.PDB.HSExposure.ExposureCN.__init__()=Bio.PDB.HSExposure.ExposureCN-class.html#__init__,Method Bio.PDB.HSExposure.HSExposureCA.__init__()=Bio.PDB.HSExposure.HSExposureCA-class.html#__init__,Method Bio.PDB.HSExposure.HSExposureCB.__init__()=Bio.PDB.HSExposure.HSExposureCB-class.html#__init__,Method Bio.PDB.HSExposure._AbstractHSExposure.__init__()=Bio.PDB.HSExposure._AbstractHSExposure-class.html#__init__,Method Bio.PDB.MMCIF2Dict.MMCIF2Dict.__init__()=Bio.PDB.MMCIF2Dict.MMCIF2Dict-class.html#__init__,Method Bio.PDB.MMCIFParser'.FastMMCIFParser.__init__()=Bio.PDB.MMCIFParser%27.FastMMCIFParser-class.html#__init__,Method Bio.PDB.MMCIFParser'.MMCIFParser.__init__()=Bio.PDB.MMCIFParser%27.MMCIFParser-class.html#__init__,Method Bio.PDB.Model.Model.__init__()=Bio.PDB.Model.Model-class.html#__init__,Method Bio.PDB.NACCESS.NACCESS.__init__()=Bio.PDB.NACCESS.NACCESS-class.html#__init__,Method Bio.PDB.NACCESS.NACCESS_atomic.__init__()=Bio.PDB.NACCESS.NACCESS_atomic-class.html#__init__,Method Bio.PDB.NeighborSearch'.NeighborSearch.__init__()=Bio.PDB.NeighborSearch%27.NeighborSearch-class.html#__init__,Method Bio.PDB.PDBIO'.PDBIO.__init__()=Bio.PDB.PDBIO%27.PDBIO-class.html#__init__,Method Bio.PDB.PDBList'.PDBList.__init__()=Bio.PDB.PDBList%27.PDBList-class.html#__init__,Method Bio.PDB.PDBParser'.PDBParser.__init__()=Bio.PDB.PDBParser%27.PDBParser-class.html#__init__,Method Bio.PDB.PSEA.PSEA.__init__()=Bio.PDB.PSEA.PSEA-class.html#__init__,Method Bio.PDB.Polypeptide.CaPPBuilder.__init__()=Bio.PDB.Polypeptide.CaPPBuilder-class.html#__init__,Method Bio.PDB.Polypeptide.PPBuilder.__init__()=Bio.PDB.Polypeptide.PPBuilder-class.html#__init__,Method Bio.PDB.Polypeptide._PPBuilder.__init__()=Bio.PDB.Polypeptide._PPBuilder-class.html#__init__,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.__init__()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#__init__,Method Bio.PDB.Residue.DisorderedResidue.__init__()=Bio.PDB.Residue.DisorderedResidue-class.html#__init__,Method Bio.PDB.Residue.Residue.__init__()=Bio.PDB.Residue.Residue-class.html#__init__,Method Bio.PDB.ResidueDepth'.ResidueDepth.__init__()=Bio.PDB.ResidueDepth%27.ResidueDepth-class.html#__init__,Method Bio.PDB.Structure.Structure.__init__()=Bio.PDB.Structure.Structure-class.html#__init__,Method Bio.PDB.StructureAlignment'.StructureAlignment.__init__()=Bio.PDB.StructureAlignment%27.StructureAlignment-class.html#__init__,Method Bio.PDB.StructureBuilder.StructureBuilder.__init__()=Bio.PDB.StructureBuilder.StructureBuilder-class.html#__init__,Method Bio.PDB.Superimposer'.Superimposer.__init__()=Bio.PDB.Superimposer%27.Superimposer-class.html#__init__,Method Bio.PDB.mmcifio.MMCIFIO.__init__()=Bio.PDB.mmcifio.MMCIFIO-class.html#__init__,Method Bio.PDB.mmtf.DefaultParser.StructureDecoder.__init__()=Bio.PDB.mmtf.DefaultParser.StructureDecoder-class.html#__init__,Method Bio.PDB.vectors.Vector.__init__()=Bio.PDB.vectors.Vector-class.html#__init__,Method Bio.ParserSupport.TaggingConsumer.__init__()=Bio.ParserSupport.TaggingConsumer-class.html#__init__,Method Bio.Pathway.Network.__init__()=Bio.Pathway.Network-class.html#__init__,Method Bio.Pathway.Reaction.__init__()=Bio.Pathway.Reaction-class.html#__init__,Method Bio.Pathway.Rep.Graph.Graph.__init__()=Bio.Pathway.Rep.Graph.Graph-class.html#__init__,Method Bio.Pathway.Rep.MultiGraph.MultiGraph.__init__()=Bio.Pathway.Rep.MultiGraph.MultiGraph-class.html#__init__,Method Bio.Pathway.System.__init__()=Bio.Pathway.System-class.html#__init__,Method Bio.Phylo.Applications._Fasttree.FastTreeCommandline.__init__()=Bio.Phylo.Applications._Fasttree.FastTreeCommandline-class.html#__init__,Method Bio.Phylo.Applications._Phyml.PhymlCommandline.__init__()=Bio.Phylo.Applications._Phyml.PhymlCommandline-class.html#__init__,Method Bio.Phylo.Applications._Raxml.RaxmlCommandline.__init__()=Bio.Phylo.Applications._Raxml.RaxmlCommandline-class.html#__init__,Method Bio.Phylo.BaseTree.BranchColor.__init__()=Bio.Phylo.BaseTree.BranchColor-class.html#__init__,Method Bio.Phylo.BaseTree.Clade.__init__()=Bio.Phylo.BaseTree.Clade-class.html#__init__,Method Bio.Phylo.BaseTree.Tree.__init__()=Bio.Phylo.BaseTree.Tree-class.html#__init__,Method Bio.Phylo.CDAO.Clade.__init__()=Bio.Phylo.CDAO.Clade-class.html#__init__,Method Bio.Phylo.CDAO.Tree.__init__()=Bio.Phylo.CDAO.Tree-class.html#__init__,Method Bio.Phylo.CDAOIO.Parser.__init__()=Bio.Phylo.CDAOIO.Parser-class.html#__init__,Method Bio.Phylo.CDAOIO.Writer.__init__()=Bio.Phylo.CDAOIO.Writer-class.html#__init__,Method Bio.Phylo.NeXML.Clade.__init__()=Bio.Phylo.NeXML.Clade-class.html#__init__,Method Bio.Phylo.NeXML.Tree.__init__()=Bio.Phylo.NeXML.Tree-class.html#__init__,Method Bio.Phylo.NeXMLIO.Parser.__init__()=Bio.Phylo.NeXMLIO.Parser-class.html#__init__,Method Bio.Phylo.NeXMLIO.Writer.__init__()=Bio.Phylo.NeXMLIO.Writer-class.html#__init__,Method Bio.Phylo.Newick.Clade.__init__()=Bio.Phylo.Newick.Clade-class.html#__init__,Method Bio.Phylo.Newick.Tree.__init__()=Bio.Phylo.Newick.Tree-class.html#__init__,Method Bio.Phylo.NewickIO.Parser.__init__()=Bio.Phylo.NewickIO.Parser-class.html#__init__,Method Bio.Phylo.NewickIO.Writer.__init__()=Bio.Phylo.NewickIO.Writer-class.html#__init__,Method Bio.Phylo.PAML._paml.Paml.__init__()=Bio.Phylo.PAML._paml.Paml-class.html#__init__,Method Bio.Phylo.PAML.baseml.Baseml.__init__()=Bio.Phylo.PAML.baseml.Baseml-class.html#__init__,Method Bio.Phylo.PAML.codeml.Codeml.__init__()=Bio.Phylo.PAML.codeml.Codeml-class.html#__init__,Method Bio.Phylo.PAML.yn00.Yn00.__init__()=Bio.Phylo.PAML.yn00.Yn00-class.html#__init__,Method Bio.Phylo.PhyloXML.Accession.__init__()=Bio.Phylo.PhyloXML.Accession-class.html#__init__,Method Bio.Phylo.PhyloXML.Annotation.__init__()=Bio.Phylo.PhyloXML.Annotation-class.html#__init__,Method Bio.Phylo.PhyloXML.BinaryCharacters.__init__()=Bio.Phylo.PhyloXML.BinaryCharacters-class.html#__init__,Method Bio.Phylo.PhyloXML.BranchColor.__init__()=Bio.Phylo.PhyloXML.BranchColor-class.html#__init__,Method Bio.Phylo.PhyloXML.Clade.__init__()=Bio.Phylo.PhyloXML.Clade-class.html#__init__,Method Bio.Phylo.PhyloXML.CladeRelation.__init__()=Bio.Phylo.PhyloXML.CladeRelation-class.html#__init__,Method Bio.Phylo.PhyloXML.Confidence.__init__()=Bio.Phylo.PhyloXML.Confidence-class.html#__init__,Method Bio.Phylo.PhyloXML.Date.__init__()=Bio.Phylo.PhyloXML.Date-class.html#__init__,Method Bio.Phylo.PhyloXML.Distribution.__init__()=Bio.Phylo.PhyloXML.Distribution-class.html#__init__,Method Bio.Phylo.PhyloXML.DomainArchitecture.__init__()=Bio.Phylo.PhyloXML.DomainArchitecture-class.html#__init__,Method Bio.Phylo.PhyloXML.Events.__init__()=Bio.Phylo.PhyloXML.Events-class.html#__init__,Method Bio.Phylo.PhyloXML.Id.__init__()=Bio.Phylo.PhyloXML.Id-class.html#__init__,Method Bio.Phylo.PhyloXML.MolSeq.__init__()=Bio.Phylo.PhyloXML.MolSeq-class.html#__init__,Method Bio.Phylo.PhyloXML.Other.__init__()=Bio.Phylo.PhyloXML.Other-class.html#__init__,Method Bio.Phylo.PhyloXML.Phylogeny.__init__()=Bio.Phylo.PhyloXML.Phylogeny-class.html#__init__,Method Bio.Phylo.PhyloXML.Phyloxml.__init__()=Bio.Phylo.PhyloXML.Phyloxml-class.html#__init__,Method Bio.Phylo.PhyloXML.Point.__init__()=Bio.Phylo.PhyloXML.Point-class.html#__init__,Method Bio.Phylo.PhyloXML.Polygon.__init__()=Bio.Phylo.PhyloXML.Polygon-class.html#__init__,Method Bio.Phylo.PhyloXML.Property.__init__()=Bio.Phylo.PhyloXML.Property-class.html#__init__,Method Bio.Phylo.PhyloXML.ProteinDomain.__init__()=Bio.Phylo.PhyloXML.ProteinDomain-class.html#__init__,Method Bio.Phylo.PhyloXML.Reference.__init__()=Bio.Phylo.PhyloXML.Reference-class.html#__init__,Method Bio.Phylo.PhyloXML.Sequence.__init__()=Bio.Phylo.PhyloXML.Sequence-class.html#__init__,Method Bio.Phylo.PhyloXML.SequenceRelation.__init__()=Bio.Phylo.PhyloXML.SequenceRelation-class.html#__init__,Method Bio.Phylo.PhyloXML.Taxonomy.__init__()=Bio.Phylo.PhyloXML.Taxonomy-class.html#__init__,Method Bio.Phylo.PhyloXML.Uri.__init__()=Bio.Phylo.PhyloXML.Uri-class.html#__init__,Method Bio.Phylo.PhyloXMLIO.Parser.__init__()=Bio.Phylo.PhyloXMLIO.Parser-class.html#__init__,Method Bio.Phylo.PhyloXMLIO.Writer.__init__()=Bio.Phylo.PhyloXMLIO.Writer-class.html#__init__,Method Bio.Phylo.TreeConstruction.DistanceCalculator.__init__()=Bio.Phylo.TreeConstruction.DistanceCalculator-class.html#__init__,Method Bio.Phylo.TreeConstruction.DistanceMatrix.__init__()=Bio.Phylo.TreeConstruction.DistanceMatrix-class.html#__init__,Method Bio.Phylo.TreeConstruction.DistanceTreeConstructor.__init__()=Bio.Phylo.TreeConstruction.DistanceTreeConstructor-class.html#__init__,Method Bio.Phylo.TreeConstruction.NNITreeSearcher.__init__()=Bio.Phylo.TreeConstruction.NNITreeSearcher-class.html#__init__,Method Bio.Phylo.TreeConstruction.ParsimonyScorer.__init__()=Bio.Phylo.TreeConstruction.ParsimonyScorer-class.html#__init__,Method Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor.__init__()=Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor-class.html#__init__,Method Bio.Phylo.TreeConstruction._Matrix.__init__()=Bio.Phylo.TreeConstruction._Matrix-class.html#__init__,Method Bio.PopGen.GenePop.Controller.GenePopController.__init__()=Bio.PopGen.GenePop.Controller.GenePopController-class.html#__init__,Method Bio.PopGen.GenePop.Controller._FileIterator.__init__()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#__init__,Method Bio.PopGen.GenePop.Controller._GenePopCommandline.__init__()=Bio.PopGen.GenePop.Controller._GenePopCommandline-class.html#__init__,Method Bio.PopGen.GenePop.EasyController.EasyController.__init__()=Bio.PopGen.GenePop.EasyController.EasyController-class.html#__init__,Method Bio.PopGen.GenePop.FileParser.FileRecord.__init__()=Bio.PopGen.GenePop.FileParser.FileRecord-class.html#__init__,Method Bio.PopGen.GenePop.LargeFileParser.Record.__init__()=Bio.PopGen.GenePop.LargeFileParser.Record-class.html#__init__,Method Bio.PopGen.GenePop.Record.__init__()=Bio.PopGen.GenePop.Record-class.html#__init__,Method Bio.Restriction.PrintFormat.PrintFormat.__init__()=Bio.Restriction.PrintFormat.PrintFormat-class.html#__init__,Method Bio.Restriction.Restriction.Analysis.__init__()=Bio.Restriction.Restriction.Analysis-class.html#__init__,Method Bio.Restriction.Restriction.FormattedSeq.__init__()=Bio.Restriction.Restriction.FormattedSeq-class.html#__init__,Method Bio.Restriction.Restriction.RestrictionBatch.__init__()=Bio.Restriction.Restriction.RestrictionBatch-class.html#__init__,Method Bio.Restriction.Restriction.RestrictionType.__init__()=Bio.Restriction.Restriction.RestrictionType-class.html#__init__,Method Bio.SCOP.Astral.__init__()=Bio.SCOP.Astral-class.html#__init__,Method Bio.SCOP.Cla.Index.__init__()=Bio.SCOP.Cla.Index-class.html#__init__,Method Bio.SCOP.Cla.Record.__init__()=Bio.SCOP.Cla.Record-class.html#__init__,Method Bio.SCOP.Des.Record.__init__()=Bio.SCOP.Des.Record-class.html#__init__,Method Bio.SCOP.Dom.Record.__init__()=Bio.SCOP.Dom.Record-class.html#__init__,Method Bio.SCOP.Domain.__init__()=Bio.SCOP.Domain-class.html#__init__,Method Bio.SCOP.Hie.Record.__init__()=Bio.SCOP.Hie.Record-class.html#__init__,Method Bio.SCOP.Node.__init__()=Bio.SCOP.Node-class.html#__init__,Method Bio.SCOP.Raf.Res.__init__()=Bio.SCOP.Raf.Res-class.html#__init__,Method Bio.SCOP.Raf.SeqMap.__init__()=Bio.SCOP.Raf.SeqMap-class.html#__init__,Method Bio.SCOP.Raf.SeqMapIndex.__init__()=Bio.SCOP.Raf.SeqMapIndex-class.html#__init__,Method Bio.SCOP.Residues.Residues.__init__()=Bio.SCOP.Residues.Residues-class.html#__init__,Method Bio.SCOP.Scop.__init__()=Bio.SCOP.Scop-class.html#__init__,Method Bio.SVDSuperimposer.SVDSuperimposer.__init__()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabParser.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_text.BlastTextParser.__init__()=Bio.SearchIO.BlastIO.blast_text.BlastTextParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator.__init__()=Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslIndexer.__init__()=Bio.SearchIO.BlatIO.BlatPslIndexer-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslParser.__init__()=Bio.SearchIO.BlatIO.BlatPslParser-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslWriter.__init__()=Bio.SearchIO.BlatIO.BlatPslWriter-class.html#__init__,Method Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.__init__()=Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html#__init__,Method Bio.SearchIO.FastaIO.FastaM10Indexer.__init__()=Bio.SearchIO.FastaIO.FastaM10Indexer-class.html#__init__,Method Bio.SearchIO.FastaIO.FastaM10Parser.__init__()=Bio.SearchIO.FastaIO.FastaM10Parser-class.html#__init__,Method Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer.__init__()=Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser.__init__()=Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.__init__()=Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser.__init__()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.__init__()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser.__init__()=Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#__init__,Method Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser.__init__()=Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser-class.html#__init__,Method Bio.SearchIO._index.SearchIndexer.__init__()=Bio.SearchIO._index.SearchIndexer-class.html#__init__,Method Bio.SearchIO._model.hit.Hit.__init__()=Bio.SearchIO._model.hit.Hit-class.html#__init__,Method Bio.SearchIO._model.hsp.HSP.__init__()=Bio.SearchIO._model.hsp.HSP-class.html#__init__,Method Bio.SearchIO._model.hsp.HSPFragment.__init__()=Bio.SearchIO._model.hsp.HSPFragment-class.html#__init__,Method Bio.SearchIO._model.query.QueryResult.__init__()=Bio.SearchIO._model.query.QueryResult-class.html#__init__,Method Bio.Seq.MutableSeq.__init__()=Bio.Seq.MutableSeq-class.html#__init__,Method Bio.Seq.Seq.__init__()=Bio.Seq.Seq-class.html#__init__,Method Bio.Seq.UnknownSeq.__init__()=Bio.Seq.UnknownSeq-class.html#__init__,Method Bio.SeqFeature.CompoundLocation.__init__()=Bio.SeqFeature.CompoundLocation-class.html#__init__,Method Bio.SeqFeature.FeatureLocation.__init__()=Bio.SeqFeature.FeatureLocation-class.html#__init__,Method Bio.SeqFeature.PositionGap.__init__()=Bio.SeqFeature.PositionGap-class.html#__init__,Method Bio.SeqFeature.Reference.__init__()=Bio.SeqFeature.Reference-class.html#__init__,Method Bio.SeqFeature.SeqFeature.__init__()=Bio.SeqFeature.SeqFeature-class.html#__init__,Method Bio.SeqIO.FastaIO.FastaTwoLineWriter.__init__()=Bio.SeqIO.FastaIO.FastaTwoLineWriter-class.html#__init__,Method Bio.SeqIO.FastaIO.FastaWriter.__init__()=Bio.SeqIO.FastaIO.FastaWriter-class.html#__init__,Method Bio.SeqIO.Interfaces.SequenceIterator.__init__()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#__init__,Method Bio.SeqIO.Interfaces.SequenceWriter.__init__()=Bio.SeqIO.Interfaces.SequenceWriter-class.html#__init__,Method Bio.SeqIO.Interfaces.SequentialSequenceWriter.__init__()=Bio.SeqIO.Interfaces.SequentialSequenceWriter-class.html#__init__,Method Bio.SeqIO.PhdIO.PhdWriter.__init__()=Bio.SeqIO.PhdIO.PhdWriter-class.html#__init__,Method Bio.SeqIO.PirIO.PirWriter.__init__()=Bio.SeqIO.PirIO.PirWriter-class.html#__init__,Method Bio.SeqIO.QualityIO.QualPhredWriter.__init__()=Bio.SeqIO.QualityIO.QualPhredWriter-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.SeqXmlIterator.__init__()=Bio.SeqIO.SeqXmlIO.SeqXmlIterator-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.SeqXmlWriter.__init__()=Bio.SeqIO.SeqXmlIO.SeqXmlWriter-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.XMLRecordIterator.__init__()=Bio.SeqIO.SeqXmlIO.XMLRecordIterator-class.html#__init__,Method Bio.SeqIO.SffIO.SffWriter.__init__()=Bio.SeqIO.SffIO.SffWriter-class.html#__init__,Method Bio.SeqIO.SffIO._AddTellHandle.__init__()=Bio.SeqIO.SffIO._AddTellHandle-class.html#__init__,Method Bio.SeqIO.UniprotIO.Parser.__init__()=Bio.SeqIO.UniprotIO.Parser-class.html#__init__,Method Bio.SeqIO._index.IntelliGeneticsRandomAccess.__init__()=Bio.SeqIO._index.IntelliGeneticsRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SeqFileRandomAccess.__init__()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SequentialSeqFileRandomAccess.__init__()=Bio.SeqIO._index.SequentialSeqFileRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SffRandomAccess.__init__()=Bio.SeqIO._index.SffRandomAccess-class.html#__init__,Method Bio.SeqRecord.SeqRecord.__init__()=Bio.SeqRecord.SeqRecord-class.html#__init__,Method Bio.SeqRecord._RestrictedDict.__init__()=Bio.SeqRecord._RestrictedDict-class.html#__init__,Method Bio.SeqUtils.CodonUsage.CodonAdaptationIndex.__init__()=Bio.SeqUtils.CodonUsage.CodonAdaptationIndex-class.html#__init__,Method Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.__init__()=Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint-class.html#__init__,Method Bio.SeqUtils.ProtParam.ProteinAnalysis.__init__()=Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#__init__,Method Bio.Sequencing.Ace.ACEFileRecord.__init__()=Bio.Sequencing.Ace.ACEFileRecord-class.html#__init__,Method Bio.Sequencing.Ace.Contig.__init__()=Bio.Sequencing.Ace.Contig-class.html#__init__,Method Bio.Sequencing.Ace.Reads.__init__()=Bio.Sequencing.Ace.Reads-class.html#__init__,Method Bio.Sequencing.Ace.af.__init__()=Bio.Sequencing.Ace.af-class.html#__init__,Method Bio.Sequencing.Ace.bs.__init__()=Bio.Sequencing.Ace.bs-class.html#__init__,Method Bio.Sequencing.Ace.ct.__init__()=Bio.Sequencing.Ace.ct-class.html#__init__,Method Bio.Sequencing.Ace.ds.__init__()=Bio.Sequencing.Ace.ds-class.html#__init__,Method Bio.Sequencing.Ace.qa.__init__()=Bio.Sequencing.Ace.qa-class.html#__init__,Method Bio.Sequencing.Ace.rd.__init__()=Bio.Sequencing.Ace.rd-class.html#__init__,Method Bio.Sequencing.Ace.rt.__init__()=Bio.Sequencing.Ace.rt-class.html#__init__,Method Bio.Sequencing.Ace.wa.__init__()=Bio.Sequencing.Ace.wa-class.html#__init__,Method Bio.Sequencing.Ace.wr.__init__()=Bio.Sequencing.Ace.wr-class.html#__init__,Method Bio.Sequencing.Applications._Novoalign.NovoalignCommandline.__init__()=Bio.Sequencing.Applications._Novoalign.NovoalignCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaAlignCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaAlignCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaBwaswCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaBwaswCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaIndexCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaIndexCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaSampeCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaSampeCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaSamseCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaSamseCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsVersion0xSortCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsVersion0xSortCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsVersion1xSortCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsVersion1xSortCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline-class.html#__init__,Method Bio.Sequencing.Phd.Record.__init__()=Bio.Sequencing.Phd.Record-class.html#__init__,Method Bio.SubsMat.FreqTable.FreqTable.__init__()=Bio.SubsMat.FreqTable.FreqTable-class.html#__init__,Method Bio.SubsMat.SeqMat.__init__()=Bio.SubsMat.SeqMat-class.html#__init__,Method Bio.SwissProt.KeyWList.Record.__init__()=Bio.SwissProt.KeyWList.Record-class.html#__init__,Method Bio.SwissProt.Record.__init__()=Bio.SwissProt.Record-class.html#__init__,Method Bio.SwissProt.Reference.__init__()=Bio.SwissProt.Reference-class.html#__init__,Method Bio.UniGene.ProtsimLine.__init__()=Bio.UniGene.ProtsimLine-class.html#__init__,Method Bio.UniGene.Record.__init__()=Bio.UniGene.Record-class.html#__init__,Method Bio.UniGene.STSLine.__init__()=Bio.UniGene.STSLine-class.html#__init__,Method Bio.UniGene.SequenceLine.__init__()=Bio.UniGene.SequenceLine-class.html#__init__,Method Bio.Wise.dnal.Statistics.__init__()=Bio.Wise.dnal.Statistics-class.html#__init__,Method Bio.Wise.psw.AlignmentColumn.__init__()=Bio.Wise.psw.AlignmentColumn-class.html#__init__,Method Bio.Wise.psw.ColumnUnit.__init__()=Bio.Wise.psw.ColumnUnit-class.html#__init__,Method Bio.bgzf.BgzfReader.__init__()=Bio.bgzf.BgzfReader-class.html#__init__,Method Bio.bgzf.BgzfWriter.__init__()=Bio.bgzf.BgzfWriter-class.html#__init__,Method Bio.codonalign.codonalignment.CodonAlignment.__init__()=Bio.codonalign.codonalignment.CodonAlignment-class.html#__init__,Method Bio.codonalign.codonseq.CodonSeq.__init__()=Bio.codonalign.codonseq.CodonSeq-class.html#__init__,Method Bio.kNN.kNN.__init__()=Bio.kNN.kNN-class.html#__init__,Method Bio.motifs.Instances.__init__()=Bio.motifs.Instances-class.html#__init__,Method Bio.motifs.Motif.__init__()=Bio.motifs.Motif-class.html#__init__,Method Bio.motifs.alignace.Record.__init__()=Bio.motifs.alignace.Record-class.html#__init__,Method Bio.motifs.applications._xxmotif.XXmotifCommandline.__init__()=Bio.motifs.applications._xxmotif.XXmotifCommandline-class.html#__init__,Method Bio.motifs.jaspar.Motif.__init__()=Bio.motifs.jaspar.Motif-class.html#__init__,Method Bio.motifs.jaspar.Record.__init__()=Bio.motifs.jaspar.Record-class.html#__init__,Method Bio.motifs.jaspar.db.JASPAR5.__init__()=Bio.motifs.jaspar.db.JASPAR5-class.html#__init__,Method Bio.motifs.mast.Record.__init__()=Bio.motifs.mast.Record-class.html#__init__,Method Bio.motifs.matrix.GenericPositionMatrix.__init__()=Bio.motifs.matrix.GenericPositionMatrix-class.html#__init__,Method Bio.motifs.matrix.PositionWeightMatrix.__init__()=Bio.motifs.matrix.PositionWeightMatrix-class.html#__init__,Method Bio.motifs.meme.Instance.__init__()=Bio.motifs.meme.Instance-class.html#__init__,Method Bio.motifs.meme.Motif.__init__()=Bio.motifs.meme.Motif-class.html#__init__,Method Bio.motifs.meme.Record.__init__()=Bio.motifs.meme.Record-class.html#__init__,Method Bio.motifs.thresholds.ScoreDistribution.__init__()=Bio.motifs.thresholds.ScoreDistribution-class.html#__init__,Method Bio.motifs.transfac.Record.__init__()=Bio.motifs.transfac.Record-class.html#__init__,Method Bio.pairwise2.affine_penalty.__init__()=Bio.pairwise2.affine_penalty-class.html#__init__,Method Bio.pairwise2.dictionary_match.__init__()=Bio.pairwise2.dictionary_match-class.html#__init__,Method Bio.pairwise2.identity_match.__init__()=Bio.pairwise2.identity_match-class.html#__init__,Method Bio.phenotype.phen_micro.JsonWriter.__init__()=Bio.phenotype.phen_micro.JsonWriter-class.html#__init__,Method Bio.phenotype.phen_micro.PlateRecord.__init__()=Bio.phenotype.phen_micro.PlateRecord-class.html#__init__,Method Bio.phenotype.phen_micro.WellRecord.__init__()=Bio.phenotype.phen_micro.WellRecord-class.html#__init__,Method BioSQL.BioSeq.DBSeq.__init__()=BioSQL.BioSeq.DBSeq-class.html#__init__,Method BioSQL.BioSeq.DBSeqRecord.__init__()=BioSQL.BioSeq.DBSeqRecord-class.html#__init__,Method BioSQL.BioSeqDatabase.Adaptor.__init__()=BioSQL.BioSeqDatabase.Adaptor-class.html#__init__,Method BioSQL.BioSeqDatabase.BioSeqDatabase.__init__()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#__init__,Method BioSQL.BioSeqDatabase.DBServer.__init__()=BioSQL.BioSeqDatabase.DBServer-class.html#__init__,Method BioSQL.BioSeqDatabase._CursorWrapper.__init__()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#__init__,Method BioSQL.DBUtils.Generic_dbutils.__init__()=BioSQL.DBUtils.Generic_dbutils-class.html#__init__,Method BioSQL.Loader.DatabaseLoader.__init__()=BioSQL.Loader.DatabaseLoader-class.html#__init__,Method BioSQL.Loader.DatabaseRemover.__init__()=BioSQL.Loader.DatabaseRemover-class.html#__init__"><a title="Bio.Affy.CelFile.ParserError.__init__
Bio.Affy.CelFile.Record.__init__
Bio.Align.AlignInfo.PSSM.__init__
Bio.Align.AlignInfo.SummaryInfo.__init__
Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline.__init__
Bio.Align.Applications._Clustalw.ClustalwCommandline.__init__
Bio.Align.Applications._Dialign.DialignCommandline.__init__
Bio.Align.Applications._MSAProbs.MSAProbsCommandline.__init__
Bio.Align.Applications._Mafft.MafftCommandline.__init__
Bio.Align.Applications._Muscle.MuscleCommandline.__init__
Bio.Align.Applications._Prank.PrankCommandline.__init__
Bio.Align.Applications._Probcons.ProbconsCommandline.__init__
Bio.Align.Applications._TCoffee.TCoffeeCommandline.__init__
Bio.Align.MultipleSeqAlignment.__init__
Bio.Align.PairwiseAlignment.__init__
Bio.Align.PairwiseAlignments.__init__
Bio.AlignIO.Interfaces.AlignmentIterator.__init__
Bio.AlignIO.Interfaces.AlignmentWriter.__init__
Bio.AlignIO.Interfaces.SequentialAlignmentWriter.__init__
Bio.AlignIO.MafIO.MafIndex.__init__
Bio.AlignIO.MauveIO.MauveWriter.__init__
Bio.Alphabet.AlphabetEncoder.__init__
Bio.Alphabet.Gapped.__init__
Bio.Alphabet.HasStopCodon.__init__
Bio.Application.AbstractCommandline.__init__
Bio.Application.ApplicationError.__init__
Bio.Application._AbstractParameter.__init__
Bio.Application._Argument.__init__
Bio.Application._Option.__init__
Bio.Application._StaticArgument.__init__
Bio.Application._Switch.__init__
Bio.Blast.Applications.NcbiblastformatterCommandline.__init__
Bio.Blast.Applications.NcbiblastnCommandline.__init__
Bio.Blast.Applications.NcbiblastpCommandline.__init__
Bio.Blast.Applications.NcbiblastxCommandline.__init__
Bio.Blast.Applications.NcbideltablastCommandline.__init__
Bio.Blast.Applications.NcbipsiblastCommandline.__init__
Bio.Blast.Applications.NcbirpsblastCommandline.__init__
Bio.Blast.Applications.NcbirpstblastnCommandline.__init__
Bio.Blast.Applications.NcbitblastnCommandline.__init__
Bio.Blast.Applications.NcbitblastxCommandline.__init__
Bio.Blast.Applications._NcbibaseblastCommandline.__init__
Bio.Blast.Applications._Ncbiblast2SeqCommandline.__init__
Bio.Blast.Applications._NcbiblastCommandline.__init__
Bio.Blast.Applications._NcbiblastMain2SeqCommandline.__init__
Bio.Blast.NCBIStandalone.BlastErrorParser.__init__
Bio.Blast.NCBIStandalone.BlastParser.__init__
Bio.Blast.NCBIStandalone.Iterator.__init__
Bio.Blast.NCBIStandalone.PSIBlastParser.__init__
Bio.Blast.NCBIStandalone._BlastConsumer.__init__
Bio.Blast.NCBIStandalone._BlastErrorConsumer.__init__
Bio.Blast.NCBIStandalone._PSIBlastConsumer.__init__
Bio.Blast.NCBIXML.BlastParser.__init__
Bio.Blast.NCBIXML._XMLparser.__init__
Bio.Blast.ParseBlastTable.BlastTableEntry.__init__
Bio.Blast.ParseBlastTable.BlastTableReader.__init__
Bio.Blast.ParseBlastTable.BlastTableRec.__init__
Bio.Blast.Record.Alignment.__init__
Bio.Blast.Record.Blast.__init__
Bio.Blast.Record.DatabaseReport.__init__
Bio.Blast.Record.Description.__init__
Bio.Blast.Record.HSP.__init__
Bio.Blast.Record.Header.__init__
Bio.Blast.Record.MultipleAlignment.__init__
Bio.Blast.Record.PSIBlast.__init__
Bio.Blast.Record.Parameters.__init__
Bio.Blast.Record.Round.__init__
Bio.CAPS.CAPSMap.__init__
Bio.CAPS.DifferentialCutsite.__init__
Bio.Cluster.Record.__init__
Bio.Compass.Record.__init__
Bio.Crystal.Chain.__init__
Bio.Crystal.Crystal.__init__
Bio.Crystal.Hetero.__init__
Bio.Data.CodonTable.AmbiguousCodonTable.__init__
Bio.Data.CodonTable.AmbiguousForwardTable.__init__
Bio.Data.CodonTable.CodonTable.__init__
Bio.Data.CodonTable.NCBICodonTable.__init__
Bio.Emboss.Applications.DiffseqCommandline.__init__
Bio.Emboss.Applications.EInvertedCommandline.__init__
Bio.Emboss.Applications.ETandemCommandline.__init__
Bio.Emboss.Applications.Est2GenomeCommandline.__init__
Bio.Emboss.Applications.FConsenseCommandline.__init__
Bio.Emboss.Applications.FDNADistCommandline.__init__
Bio.Emboss.Applications.FDNAParsCommandline.__init__
Bio.Emboss.Applications.FNeighborCommandline.__init__
Bio.Emboss.Applications.FProtDistCommandline.__init__
Bio.Emboss.Applications.FProtParsCommandline.__init__
Bio.Emboss.Applications.FSeqBootCommandline.__init__
Bio.Emboss.Applications.FTreeDistCommandline.__init__
Bio.Emboss.Applications.FuzznucCommandline.__init__
Bio.Emboss.Applications.FuzzproCommandline.__init__
Bio.Emboss.Applications.IepCommandline.__init__
Bio.Emboss.Applications.NeedleCommandline.__init__
Bio.Emboss.Applications.NeedleallCommandline.__init__
Bio.Emboss.Applications.PalindromeCommandline.__init__
Bio.Emboss.Applications.Primer3Commandline.__init__
Bio.Emboss.Applications.PrimerSearchCommandline.__init__
Bio.Emboss.Applications.SeqmatchallCommandline.__init__
Bio.Emboss.Applications.SeqretCommandline.__init__
Bio.Emboss.Applications.StretcherCommandline.__init__
Bio.Emboss.Applications.TranalignCommandline.__init__
Bio.Emboss.Applications.WaterCommandline.__init__
Bio.Emboss.Applications._EmbossCommandLine.__init__
Bio.Emboss.Applications._EmbossMinimalCommandLine.__init__
Bio.Emboss.Primer3.Primers.__init__
Bio.Emboss.Primer3.Record.__init__
Bio.Emboss.PrimerSearch.Amplifier.__init__
Bio.Emboss.PrimerSearch.InputRecord.__init__
Bio.Emboss.PrimerSearch.OutputRecord.__init__
Bio.Entrez.Parser.Consumer.__init__
Bio.Entrez.Parser.CorruptedXMLError.__init__
Bio.Entrez.Parser.DataHandler.__init__
Bio.Entrez.Parser.DictionaryConsumer.__init__
Bio.Entrez.Parser.ErrorConsumer.__init__
Bio.Entrez.Parser.IntegerConsumer.__init__
Bio.Entrez.Parser.ListConsumer.__init__
Bio.Entrez.Parser.NotXMLError.__init__
Bio.Entrez.Parser.StringConsumer.__init__
Bio.Entrez.Parser.ValidationError.__init__
Bio.ExPASy.Enzyme.Record.__init__
Bio.ExPASy.Prodoc.Record.__init__
Bio.ExPASy.Prodoc.Reference.__init__
Bio.ExPASy.Prosite.Record.__init__
Bio.ExPASy.ScanProsite.ContentHandler.__init__
Bio.ExPASy.ScanProsite.Parser.__init__
Bio.ExPASy.ScanProsite.Record.__init__
Bio.ExPASy.cellosaurus.Record.__init__
Bio.FSSP.FSSPAlignDict.__init__
Bio.FSSP.FSSPAlignRec.__init__
Bio.FSSP.FSSPHeader.__init__
Bio.FSSP.FSSPSumRec.__init__
Bio.FSSP.FSSPTools.FSSPMultAlign.__init__
Bio.FSSP.PosAlign.__init__
Bio.FSSP.fssp_rec.fff_rec.__init__
Bio.File.UndoHandle.__init__
Bio.File._IndexedSeqFileDict.__init__
Bio.File._SQLiteManySeqFilesDict.__init__
Bio.GA.Crossover.General.SafeFitnessCrossover.__init__
Bio.GA.Crossover.GeneralPoint.GeneralPointCrossover.__init__
Bio.GA.Crossover.GeneralPoint.InterleaveCrossover.__init__
Bio.GA.Crossover.Point.SinglePointCrossover.__init__
Bio.GA.Crossover.TwoPoint.TwoPointCrossover.__init__
Bio.GA.Crossover.Uniform.UniformCrossover.__init__
Bio.GA.Evolver.GenerationEvolver.__init__
Bio.GA.Evolver.SteadyStateEvolver.__init__
Bio.GA.Mutation.General.SafeFitnessMutation.__init__
Bio.GA.Mutation.Simple.ConversionMutation.__init__
Bio.GA.Mutation.Simple.SinglePositionMutation.__init__
Bio.GA.Organism.Organism.__init__
Bio.GA.Repair.Stabilizing.AmbiguousRepair.__init__
Bio.GA.Selection.Abstract.AbstractSelection.__init__
Bio.GA.Selection.Diversity.DiversitySelection.__init__
Bio.GA.Selection.RouletteWheel.RouletteWheelSelection.__init__
Bio.GA.Selection.Tournament.TournamentSelection.__init__
Bio.GenBank.FeatureParser.__init__
Bio.GenBank.Iterator.__init__
Bio.GenBank.Record.Feature.__init__
Bio.GenBank.Record.Qualifier.__init__
Bio.GenBank.Record.Record.__init__
Bio.GenBank.Record.Reference.__init__
Bio.GenBank.RecordParser.__init__
Bio.GenBank.Scanner.InsdcScanner.__init__
Bio.GenBank._BaseGenBankConsumer.__init__
Bio.GenBank._FeatureConsumer.__init__
Bio.GenBank._RecordConsumer.__init__
Bio.GenBank.utils.FeatureValueCleaner.__init__
Bio.Geo.Record.Record.__init__
Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment.__init__
Bio.Graphics.BasicChromosome.Chromosome.__init__
Bio.Graphics.BasicChromosome.ChromosomeSegment.__init__
Bio.Graphics.BasicChromosome.Organism.__init__
Bio.Graphics.BasicChromosome.TelomereSegment.__init__
Bio.Graphics.BasicChromosome._ChromosomeComponent.__init__
Bio.Graphics.ColorSpiral.ColorSpiral.__init__
Bio.Graphics.Comparative.ComparativeScatterPlot.__init__
Bio.Graphics.DisplayRepresentation.ChromosomeCounts.__init__
Bio.Graphics.Distribution.BarChartDistribution.__init__
Bio.Graphics.Distribution.DistributionPage.__init__
Bio.Graphics.Distribution.LineDistribution.__init__
Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.__init__
Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.__init__
Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.__init__
Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.__init__
Bio.Graphics.GenomeDiagram._Diagram.Diagram.__init__
Bio.Graphics.GenomeDiagram._Feature.Feature.__init__
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__init__
Bio.Graphics.GenomeDiagram._Graph.GraphData.__init__
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__init__
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.__init__
Bio.Graphics.GenomeDiagram._Track.Track.__init__
Bio.Graphics.KGML_vis.KGMLCanvas.__init__
Bio.HMM.DynamicProgramming.AbstractDPAlgorithms.__init__
Bio.HMM.DynamicProgramming.LogDPAlgorithms.__init__
Bio.HMM.DynamicProgramming.ScaledDPAlgorithms.__init__
Bio.HMM.MarkovModel.HiddenMarkovModel.__init__
Bio.HMM.MarkovModel.MarkovModelBuilder.__init__
Bio.HMM.Trainer.AbstractTrainer.__init__
Bio.HMM.Trainer.BaumWelchTrainer.__init__
Bio.HMM.Trainer.KnownStateTrainer.__init__
Bio.HMM.Trainer.TrainingSequence.__init__
Bio.Index._InMemoryIndex.__init__
Bio.Index._ShelveIndex.__init__
Bio.KDTree.KDTree'.KDTree.__init__
Bio.KEGG.Compound.Record.__init__
Bio.KEGG.Enzyme.Record.__init__
Bio.KEGG.Gene.Record.__init__
Bio.KEGG.KGML.KGML_parser.KGMLParser.__init__
Bio.KEGG.KGML.KGML_pathway.Component.__init__
Bio.KEGG.KGML.KGML_pathway.Entry.__init__
Bio.KEGG.KGML.KGML_pathway.Graphics.__init__
Bio.KEGG.KGML.KGML_pathway.Pathway.__init__
Bio.KEGG.KGML.KGML_pathway.Reaction.__init__
Bio.KEGG.KGML.KGML_pathway.Relation.__init__
Bio.LogisticRegression.LogisticRegression.__init__
Bio.MarkovModel.MarkovModel.__init__
Bio.MaxEntropy.MaxEntropy.__init__
Bio.NMR.xpktools.Peaklist.__init__
Bio.NMR.xpktools.XpkEntry.__init__
Bio.NaiveBayes.NaiveBayes.__init__
Bio.NeuralNetwork.BackPropagation.Layer.AbstractLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.__init__
Bio.NeuralNetwork.BackPropagation.Network.BasicNetwork.__init__
Bio.NeuralNetwork.Gene.Motif.MotifCoder.__init__
Bio.NeuralNetwork.Gene.Motif.MotifFinder.__init__
Bio.NeuralNetwork.Gene.Pattern.PatternIO.__init__
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.__init__
Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness.__init__
Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder.__init__
Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness.__init__
Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator.__init__
Bio.NeuralNetwork.Gene.Schema.Schema.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaCoder.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaFactory.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaFinder.__init__
Bio.NeuralNetwork.Gene.Schema.SimpleFinisher.__init__
Bio.NeuralNetwork.Gene.Signature.SignatureCoder.__init__
Bio.NeuralNetwork.Gene.Signature.SignatureFinder.__init__
Bio.NeuralNetwork.StopTraining.ValidationIncreaseStop.__init__
Bio.NeuralNetwork.Training.ExampleManager.__init__
Bio.NeuralNetwork.Training.TrainingExample.__init__
Bio.Nexus.Nexus.Block.__init__
Bio.Nexus.Nexus.CharBuffer.__init__
Bio.Nexus.Nexus.Commandline.__init__
Bio.Nexus.Nexus.Nexus.__init__
Bio.Nexus.Nexus.StepMatrix.__init__
Bio.Nexus.Nodes.Chain.__init__
Bio.Nexus.Nodes.Node.__init__
Bio.Nexus.StandardData.StandardData.__init__
Bio.Nexus.Trees.NodeData.__init__
Bio.Nexus.Trees.Tree.__init__
Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap.__init__
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__init__
Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap.__init__
Bio.PDB.Atom.Atom.__init__
Bio.PDB.Atom.DisorderedAtom.__init__
Bio.PDB.Chain.Chain.__init__
Bio.PDB.DSSP'.DSSP.__init__
Bio.PDB.Dice.ChainSelector.__init__
Bio.PDB.Entity.DisorderedEntityWrapper.__init__
Bio.PDB.Entity.Entity.__init__
Bio.PDB.FragmentMapper'.Fragment.__init__
Bio.PDB.FragmentMapper'.FragmentMapper.__init__
Bio.PDB.HSExposure.ExposureCN.__init__
Bio.PDB.HSExposure.HSExposureCA.__init__
Bio.PDB.HSExposure.HSExposureCB.__init__
Bio.PDB.HSExposure._AbstractHSExposure.__init__
Bio.PDB.MMCIF2Dict.MMCIF2Dict.__init__
Bio.PDB.MMCIFParser'.FastMMCIFParser.__init__
Bio.PDB.MMCIFParser'.MMCIFParser.__init__
Bio.PDB.Model.Model.__init__
Bio.PDB.NACCESS.NACCESS.__init__
Bio.PDB.NACCESS.NACCESS_atomic.__init__
Bio.PDB.NeighborSearch'.NeighborSearch.__init__
Bio.PDB.PDBIO'.PDBIO.__init__
Bio.PDB.PDBList'.PDBList.__init__
Bio.PDB.PDBParser'.PDBParser.__init__
Bio.PDB.PSEA.PSEA.__init__
Bio.PDB.Polypeptide.CaPPBuilder.__init__
Bio.PDB.Polypeptide.PPBuilder.__init__
Bio.PDB.Polypeptide._PPBuilder.__init__
Bio.PDB.QCPSuperimposer.QCPSuperimposer.__init__
Bio.PDB.Residue.DisorderedResidue.__init__
Bio.PDB.Residue.Residue.__init__
Bio.PDB.ResidueDepth'.ResidueDepth.__init__
Bio.PDB.Structure.Structure.__init__
Bio.PDB.StructureAlignment'.StructureAlignment.__init__
Bio.PDB.StructureBuilder.StructureBuilder.__init__
Bio.PDB.Superimposer'.Superimposer.__init__
Bio.PDB.mmcifio.MMCIFIO.__init__
Bio.PDB.mmtf.DefaultParser.StructureDecoder.__init__
Bio.PDB.vectors.Vector.__init__
Bio.ParserSupport.TaggingConsumer.__init__
Bio.Pathway.Network.__init__
Bio.Pathway.Reaction.__init__
Bio.Pathway.Rep.Graph.Graph.__init__
Bio.Pathway.Rep.MultiGraph.MultiGraph.__init__
Bio.Pathway.System.__init__
Bio.Phylo.Applications._Fasttree.FastTreeCommandline.__init__
Bio.Phylo.Applications._Phyml.PhymlCommandline.__init__
Bio.Phylo.Applications._Raxml.RaxmlCommandline.__init__
Bio.Phylo.BaseTree.BranchColor.__init__
Bio.Phylo.BaseTree.Clade.__init__
Bio.Phylo.BaseTree.Tree.__init__
Bio.Phylo.CDAO.Clade.__init__
Bio.Phylo.CDAO.Tree.__init__
Bio.Phylo.CDAOIO.Parser.__init__
Bio.Phylo.CDAOIO.Writer.__init__
Bio.Phylo.NeXML.Clade.__init__
Bio.Phylo.NeXML.Tree.__init__
Bio.Phylo.NeXMLIO.Parser.__init__
Bio.Phylo.NeXMLIO.Writer.__init__
Bio.Phylo.Newick.Clade.__init__
Bio.Phylo.Newick.Tree.__init__
Bio.Phylo.NewickIO.Parser.__init__
Bio.Phylo.NewickIO.Writer.__init__
Bio.Phylo.PAML._paml.Paml.__init__
Bio.Phylo.PAML.baseml.Baseml.__init__
Bio.Phylo.PAML.codeml.Codeml.__init__
Bio.Phylo.PAML.yn00.Yn00.__init__
Bio.Phylo.PhyloXML.Accession.__init__
Bio.Phylo.PhyloXML.Annotation.__init__
Bio.Phylo.PhyloXML.BinaryCharacters.__init__
Bio.Phylo.PhyloXML.BranchColor.__init__
Bio.Phylo.PhyloXML.Clade.__init__
Bio.Phylo.PhyloXML.CladeRelation.__init__
Bio.Phylo.PhyloXML.Confidence.__init__
Bio.Phylo.PhyloXML.Date.__init__
Bio.Phylo.PhyloXML.Distribution.__init__
Bio.Phylo.PhyloXML.DomainArchitecture.__init__
Bio.Phylo.PhyloXML.Events.__init__
Bio.Phylo.PhyloXML.Id.__init__
Bio.Phylo.PhyloXML.MolSeq.__init__
Bio.Phylo.PhyloXML.Other.__init__
Bio.Phylo.PhyloXML.Phylogeny.__init__
Bio.Phylo.PhyloXML.Phyloxml.__init__
Bio.Phylo.PhyloXML.Point.__init__
Bio.Phylo.PhyloXML.Polygon.__init__
Bio.Phylo.PhyloXML.Property.__init__
Bio.Phylo.PhyloXML.ProteinDomain.__init__
Bio.Phylo.PhyloXML.Reference.__init__
Bio.Phylo.PhyloXML.Sequence.__init__
Bio.Phylo.PhyloXML.SequenceRelation.__init__
Bio.Phylo.PhyloXML.Taxonomy.__init__
Bio.Phylo.PhyloXML.Uri.__init__
Bio.Phylo.PhyloXMLIO.Parser.__init__
Bio.Phylo.PhyloXMLIO.Writer.__init__
Bio.Phylo.TreeConstruction.DistanceCalculator.__init__
Bio.Phylo.TreeConstruction.DistanceMatrix.__init__
Bio.Phylo.TreeConstruction.DistanceTreeConstructor.__init__
Bio.Phylo.TreeConstruction.NNITreeSearcher.__init__
Bio.Phylo.TreeConstruction.ParsimonyScorer.__init__
Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor.__init__
Bio.Phylo.TreeConstruction._Matrix.__init__
Bio.PopGen.GenePop.Controller.GenePopController.__init__
Bio.PopGen.GenePop.Controller._FileIterator.__init__
Bio.PopGen.GenePop.Controller._GenePopCommandline.__init__
Bio.PopGen.GenePop.EasyController.EasyController.__init__
Bio.PopGen.GenePop.FileParser.FileRecord.__init__
Bio.PopGen.GenePop.LargeFileParser.Record.__init__
Bio.PopGen.GenePop.Record.__init__
Bio.Restriction.PrintFormat.PrintFormat.__init__
Bio.Restriction.Restriction.Analysis.__init__
Bio.Restriction.Restriction.FormattedSeq.__init__
Bio.Restriction.Restriction.RestrictionBatch.__init__
Bio.Restriction.Restriction.RestrictionType.__init__
Bio.SCOP.Astral.__init__
Bio.SCOP.Cla.Index.__init__
Bio.SCOP.Cla.Record.__init__
Bio.SCOP.Des.Record.__init__
Bio.SCOP.Dom.Record.__init__
Bio.SCOP.Domain.__init__
Bio.SCOP.Hie.Record.__init__
Bio.SCOP.Node.__init__
Bio.SCOP.Raf.Res.__init__
Bio.SCOP.Raf.SeqMap.__init__
Bio.SCOP.Raf.SeqMapIndex.__init__
Bio.SCOP.Residues.Residues.__init__
Bio.SCOP.Scop.__init__
Bio.SVDSuperimposer.SVDSuperimposer.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabParser.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.__init__
Bio.SearchIO.BlastIO.blast_text.BlastTextParser.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.__init__
Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator.__init__
Bio.SearchIO.BlatIO.BlatPslIndexer.__init__
Bio.SearchIO.BlatIO.BlatPslParser.__init__
Bio.SearchIO.BlatIO.BlatPslWriter.__init__
Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.__init__
Bio.SearchIO.FastaIO.FastaM10Indexer.__init__
Bio.SearchIO.FastaIO.FastaM10Parser.__init__
Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer.__init__
Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser.__init__
Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.__init__
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser.__init__
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.__init__
Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser.__init__
Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser.__init__
Bio.SearchIO._index.SearchIndexer.__init__
Bio.SearchIO._model.hit.Hit.__init__
Bio.SearchIO._model.hsp.HSP.__init__
Bio.SearchIO._model.hsp.HSPFragment.__init__
Bio.SearchIO._model.query.QueryResult.__init__
Bio.Seq.MutableSeq.__init__
Bio.Seq.Seq.__init__
Bio.Seq.UnknownSeq.__init__
Bio.SeqFeature.CompoundLocation.__init__
Bio.SeqFeature.FeatureLocation.__init__
Bio.SeqFeature.PositionGap.__init__
Bio.SeqFeature.Reference.__init__
Bio.SeqFeature.SeqFeature.__init__
Bio.SeqIO.FastaIO.FastaTwoLineWriter.__init__
Bio.SeqIO.FastaIO.FastaWriter.__init__
Bio.SeqIO.Interfaces.SequenceIterator.__init__
Bio.SeqIO.Interfaces.SequenceWriter.__init__
Bio.SeqIO.Interfaces.SequentialSequenceWriter.__init__
Bio.SeqIO.PhdIO.PhdWriter.__init__
Bio.SeqIO.PirIO.PirWriter.__init__
Bio.SeqIO.QualityIO.QualPhredWriter.__init__
Bio.SeqIO.SeqXmlIO.SeqXmlIterator.__init__
Bio.SeqIO.SeqXmlIO.SeqXmlWriter.__init__
Bio.SeqIO.SeqXmlIO.XMLRecordIterator.__init__
Bio.SeqIO.SffIO.SffWriter.__init__
Bio.SeqIO.SffIO._AddTellHandle.__init__
Bio.SeqIO.UniprotIO.Parser.__init__
Bio.SeqIO._index.IntelliGeneticsRandomAccess.__init__
Bio.SeqIO._index.SeqFileRandomAccess.__init__
Bio.SeqIO._index.SequentialSeqFileRandomAccess.__init__
Bio.SeqIO._index.SffRandomAccess.__init__
Bio.SeqRecord.SeqRecord.__init__
Bio.SeqRecord._RestrictedDict.__init__
Bio.SeqUtils.CodonUsage.CodonAdaptationIndex.__init__
Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.__init__
Bio.SeqUtils.ProtParam.ProteinAnalysis.__init__
Bio.Sequencing.Ace.ACEFileRecord.__init__
Bio.Sequencing.Ace.Contig.__init__
Bio.Sequencing.Ace.Reads.__init__
Bio.Sequencing.Ace.af.__init__
Bio.Sequencing.Ace.bs.__init__
Bio.Sequencing.Ace.ct.__init__
Bio.Sequencing.Ace.ds.__init__
Bio.Sequencing.Ace.qa.__init__
Bio.Sequencing.Ace.rd.__init__
Bio.Sequencing.Ace.rt.__init__
Bio.Sequencing.Ace.wa.__init__
Bio.Sequencing.Ace.wr.__init__
Bio.Sequencing.Applications._Novoalign.NovoalignCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaAlignCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaBwaswCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaIndexCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaSampeCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaSamseCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsVersion0xSortCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsVersion1xSortCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline.__init__
Bio.Sequencing.Phd.Record.__init__
Bio.SubsMat.FreqTable.FreqTable.__init__
Bio.SubsMat.SeqMat.__init__
Bio.SwissProt.KeyWList.Record.__init__
Bio.SwissProt.Record.__init__
Bio.SwissProt.Reference.__init__
Bio.UniGene.ProtsimLine.__init__
Bio.UniGene.Record.__init__
Bio.UniGene.STSLine.__init__
Bio.UniGene.SequenceLine.__init__
Bio.Wise.dnal.Statistics.__init__
Bio.Wise.psw.AlignmentColumn.__init__
Bio.Wise.psw.ColumnUnit.__init__
Bio.bgzf.BgzfReader.__init__
Bio.bgzf.BgzfWriter.__init__
Bio.codonalign.codonalignment.CodonAlignment.__init__
Bio.codonalign.codonseq.CodonSeq.__init__
Bio.kNN.kNN.__init__
Bio.motifs.Instances.__init__
Bio.motifs.Motif.__init__
Bio.motifs.alignace.Record.__init__
Bio.motifs.applications._xxmotif.XXmotifCommandline.__init__
Bio.motifs.jaspar.Motif.__init__
Bio.motifs.jaspar.Record.__init__
Bio.motifs.jaspar.db.JASPAR5.__init__
Bio.motifs.mast.Record.__init__
Bio.motifs.matrix.GenericPositionMatrix.__init__
Bio.motifs.matrix.PositionWeightMatrix.__init__
Bio.motifs.meme.Instance.__init__
Bio.motifs.meme.Motif.__init__
Bio.motifs.meme.Record.__init__
Bio.motifs.thresholds.ScoreDistribution.__init__
Bio.motifs.transfac.Record.__init__
Bio.pairwise2.affine_penalty.__init__
Bio.pairwise2.dictionary_match.__init__
Bio.pairwise2.identity_match.__init__
Bio.phenotype.phen_micro.JsonWriter.__init__
Bio.phenotype.phen_micro.PlateRecord.__init__
Bio.phenotype.phen_micro.WellRecord.__init__
BioSQL.BioSeq.DBSeq.__init__
BioSQL.BioSeq.DBSeqRecord.__init__
BioSQL.BioSeqDatabase.Adaptor.__init__
BioSQL.BioSeqDatabase.BioSeqDatabase.__init__
BioSQL.BioSeqDatabase.DBServer.__init__
BioSQL.BioSeqDatabase._CursorWrapper.__init__
BioSQL.DBUtils.Generic_dbutils.__init__
BioSQL.Loader.DatabaseLoader.__init__
BioSQL.Loader.DatabaseRemover.__init__" class="py-name" href="#" onclick="return doclink('link-217', '__init__', 'link-217');">__init__</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-name">comments</tt><tt class="py-op">=</tt><tt class="py-name">comments</tt><tt class="py-op">,</tt> <tt class="py-name">fields</tt><tt class="py-op">=</tt><tt class="py-name">fields</tt><tt class="py-op">)</tt> </tt>
<a name="L526"></a><tt class="py-lineno">526</tt>  <tt class="py-line"> </tt>
<a name="L527"></a><tt class="py-lineno">527</tt>  <tt class="py-line">        <tt class="py-comment"># if the file doesn't have comments,</tt> </tt>
<a name="L528"></a><tt class="py-lineno">528</tt>  <tt class="py-line">        <tt class="py-comment"># get index of column used as the key (qseqid / qacc / qaccver)</tt> </tt>
<a name="L529"></a><tt class="py-lineno">529</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_kwargs</tt><tt class="py-op">[</tt><tt class="py-string">'comments'</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L530"></a><tt class="py-lineno">530</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-string">'qseqid'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">fields</tt><tt class="py-op">:</tt> </tt>
<a name="L531"></a><tt class="py-lineno">531</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_key_idx</tt> <tt class="py-op">=</tt> <tt class="py-name">fields</tt><tt class="py-op">.</tt><tt id="link-218" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-218', 'index', 'link-110');">index</a></tt><tt class="py-op">(</tt><tt class="py-string">'qseqid'</tt><tt class="py-op">)</tt> </tt>
<a name="L532"></a><tt class="py-lineno">532</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-string">'qacc'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">fields</tt><tt class="py-op">:</tt> </tt>
<a name="L533"></a><tt class="py-lineno">533</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_key_idx</tt> <tt class="py-op">=</tt> <tt class="py-name">fields</tt><tt class="py-op">.</tt><tt id="link-219" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-219', 'index', 'link-110');">index</a></tt><tt class="py-op">(</tt><tt class="py-string">'qacc'</tt><tt class="py-op">)</tt> </tt>
<a name="L534"></a><tt class="py-lineno">534</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-string">'qaccver'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">fields</tt><tt class="py-op">:</tt> </tt>
<a name="L535"></a><tt class="py-lineno">535</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_key_idx</tt> <tt class="py-op">=</tt> <tt class="py-name">fields</tt><tt class="py-op">.</tt><tt id="link-220" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-220', 'index', 'link-110');">index</a></tt><tt class="py-op">(</tt><tt class="py-string">'qaccver'</tt><tt class="py-op">)</tt> </tt>
<a name="L536"></a><tt class="py-lineno">536</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L537"></a><tt class="py-lineno">537</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Custom fields is missing an ID column. "</tt> </tt>
<a name="L538"></a><tt class="py-lineno">538</tt>  <tt class="py-line">                        <tt class="py-string">"One of these must be present: 'qseqid', 'qacc', or 'qaccver'."</tt><tt class="py-op">)</tt> </tt>
</div><a name="L539"></a><tt class="py-lineno">539</tt>  <tt class="py-line"> </tt>
<a name="BlastTabIndexer.__iter__"></a><div id="BlastTabIndexer.__iter__-def"><a name="L540"></a><tt class="py-lineno">540</tt> <a class="py-toggle" href="#" id="BlastTabIndexer.__iter__-toggle" onclick="return toggle('BlastTabIndexer.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabIndexer.__iter__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlastTabIndexer.__iter__-expanded"><a name="L541"></a><tt class="py-lineno">541</tt>  <tt class="py-line">        <tt class="py-docstring">"""Iterate over the file handle; yields key, start offset, and length."""</tt> </tt>
<a name="L542"></a><tt class="py-lineno">542</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_handle</tt> </tt>
<a name="L543"></a><tt class="py-lineno">543</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-221" class="py-name" targets="Method Bio.File.UndoHandle.seek()=Bio.File.UndoHandle-class.html#seek,Method Bio.SeqIO.SffIO._AddTellHandle.seek()=Bio.SeqIO.SffIO._AddTellHandle-class.html#seek,Method Bio.bgzf.BgzfReader.seek()=Bio.bgzf.BgzfReader-class.html#seek"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-221', 'seek', 'link-221');">seek</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L544"></a><tt class="py-lineno">544</tt>  <tt class="py-line"> </tt>
<a name="L545"></a><tt class="py-lineno">545</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_kwargs</tt><tt class="py-op">[</tt><tt class="py-string">'comments'</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L546"></a><tt class="py-lineno">546</tt>  <tt class="py-line">            <tt class="py-name">iterfunc</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-222" class="py-name" targets="Method Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer._qresult_index()=Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer-class.html#_qresult_index"><a title="Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer._qresult_index" class="py-name" href="#" onclick="return doclink('link-222', '_qresult_index', 'link-222');">_qresult_index</a></tt> </tt>
<a name="L547"></a><tt class="py-lineno">547</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L548"></a><tt class="py-lineno">548</tt>  <tt class="py-line">            <tt class="py-name">iterfunc</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-223" class="py-name" targets="Method Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer._qresult_index_commented()=Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer-class.html#_qresult_index_commented"><a title="Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer._qresult_index_commented" class="py-name" href="#" onclick="return doclink('link-223', '_qresult_index_commented', 'link-223');">_qresult_index_commented</a></tt> </tt>
<a name="L549"></a><tt class="py-lineno">549</tt>  <tt class="py-line"> </tt>
<a name="L550"></a><tt class="py-lineno">550</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">offset</tt><tt class="py-op">,</tt> <tt id="link-224" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.length()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#length"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-224', 'length', 'link-224');">length</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">iterfunc</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L551"></a><tt class="py-lineno">551</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt id="link-225" class="py-name"><a title="Bio._py3k._bytes_to_string" class="py-name" href="#" onclick="return doclink('link-225', '_bytes_to_string', 'link-3');">_bytes_to_string</a></tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">offset</tt><tt class="py-op">,</tt> <tt id="link-226" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-226', 'length', 'link-224');">length</a></tt> </tt>
</div><a name="L552"></a><tt class="py-lineno">552</tt>  <tt class="py-line"> </tt>
<a name="BlastTabIndexer._qresult_index_commented"></a><div id="BlastTabIndexer._qresult_index_commented-def"><a name="L553"></a><tt class="py-lineno">553</tt> <a class="py-toggle" href="#" id="BlastTabIndexer._qresult_index_commented-toggle" onclick="return toggle('BlastTabIndexer._qresult_index_commented');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer-class.html#_qresult_index_commented">_qresult_index_commented</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabIndexer._qresult_index_commented-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlastTabIndexer._qresult_index_commented-expanded"><a name="L554"></a><tt class="py-lineno">554</tt>  <tt class="py-line">        <tt class="py-docstring">"""Indexer for commented BLAST tabular files (PRIVATE)."""</tt> </tt>
<a name="L555"></a><tt class="py-lineno">555</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_handle</tt> </tt>
<a name="L556"></a><tt class="py-lineno">556</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-227" class="py-name"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-227', 'seek', 'link-221');">seek</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L557"></a><tt class="py-lineno">557</tt>  <tt class="py-line">        <tt class="py-name">start_offset</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L558"></a><tt class="py-lineno">558</tt>  <tt class="py-line">        <tt class="py-comment"># mark of a new query</tt> </tt>
<a name="L559"></a><tt class="py-lineno">559</tt>  <tt class="py-line">        <tt class="py-name">query_mark</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L560"></a><tt class="py-lineno">560</tt>  <tt class="py-line">        <tt class="py-comment"># mark of the query's ID</tt> </tt>
<a name="L561"></a><tt class="py-lineno">561</tt>  <tt class="py-line">        <tt class="py-name">qid_mark</tt> <tt class="py-op">=</tt> <tt id="link-228" class="py-name"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-228', '_as_bytes', 'link-2');">_as_bytes</a></tt><tt class="py-op">(</tt><tt class="py-string">'# Query: '</tt><tt class="py-op">)</tt> </tt>
<a name="L562"></a><tt class="py-lineno">562</tt>  <tt class="py-line">        <tt class="py-comment"># mark of the last line</tt> </tt>
<a name="L563"></a><tt class="py-lineno">563</tt>  <tt class="py-line">        <tt class="py-name">end_mark</tt> <tt class="py-op">=</tt> <tt id="link-229" class="py-name"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-229', '_as_bytes', 'link-2');">_as_bytes</a></tt><tt class="py-op">(</tt><tt class="py-string">'# BLAST processed'</tt><tt class="py-op">)</tt> </tt>
<a name="L564"></a><tt class="py-lineno">564</tt>  <tt class="py-line"> </tt>
<a name="L565"></a><tt class="py-lineno">565</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L566"></a><tt class="py-lineno">566</tt>  <tt class="py-line">            <tt class="py-name">end_offset</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-230" class="py-name" targets="Method Bio.File.UndoHandle.tell()=Bio.File.UndoHandle-class.html#tell,Method Bio.SeqIO.SffIO._AddTellHandle.tell()=Bio.SeqIO.SffIO._AddTellHandle-class.html#tell,Method Bio.bgzf.BgzfReader.tell()=Bio.bgzf.BgzfReader-class.html#tell,Method Bio.bgzf.BgzfWriter.tell()=Bio.bgzf.BgzfWriter-class.html#tell"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-230', 'tell', 'link-230');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L567"></a><tt class="py-lineno">567</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-231" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-231', 'readline', 'link-103');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L568"></a><tt class="py-lineno">568</tt>  <tt class="py-line"> </tt>
<a name="L569"></a><tt class="py-lineno">569</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">query_mark</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L570"></a><tt class="py-lineno">570</tt>  <tt class="py-line">                <tt class="py-name">query_mark</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> </tt>
<a name="L571"></a><tt class="py-lineno">571</tt>  <tt class="py-line">                <tt class="py-name">start_offset</tt> <tt class="py-op">=</tt> <tt class="py-name">end_offset</tt> </tt>
<a name="L572"></a><tt class="py-lineno">572</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-232" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-232', 'startswith', 'link-59');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-name">qid_mark</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L573"></a><tt class="py-lineno">573</tt>  <tt class="py-line">                <tt class="py-name">qresult_key</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">qid_mark</tt><tt class="py-op">)</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-233" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-233', 'split', 'link-21');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L574"></a><tt class="py-lineno">574</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt> <tt class="py-op">==</tt> <tt class="py-name">query_mark</tt> <tt class="py-keyword">or</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-234" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-234', 'startswith', 'link-59');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-name">end_mark</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L575"></a><tt class="py-lineno">575</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt class="py-name">qresult_key</tt><tt class="py-op">,</tt> <tt class="py-name">start_offset</tt><tt class="py-op">,</tt> <tt class="py-name">end_offset</tt> <tt class="py-op">-</tt> <tt class="py-name">start_offset</tt> </tt>
<a name="L576"></a><tt class="py-lineno">576</tt>  <tt class="py-line">                <tt class="py-name">start_offset</tt> <tt class="py-op">=</tt> <tt class="py-name">end_offset</tt> </tt>
<a name="L577"></a><tt class="py-lineno">577</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L578"></a><tt class="py-lineno">578</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
</div><a name="L579"></a><tt class="py-lineno">579</tt>  <tt class="py-line"> </tt>
<a name="BlastTabIndexer._qresult_index"></a><div id="BlastTabIndexer._qresult_index-def"><a name="L580"></a><tt class="py-lineno">580</tt> <a class="py-toggle" href="#" id="BlastTabIndexer._qresult_index-toggle" onclick="return toggle('BlastTabIndexer._qresult_index');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer-class.html#_qresult_index">_qresult_index</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabIndexer._qresult_index-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlastTabIndexer._qresult_index-expanded"><a name="L581"></a><tt class="py-lineno">581</tt>  <tt class="py-line">        <tt class="py-docstring">"""Indexer for noncommented BLAST tabular files (PRIVATE)."""</tt> </tt>
<a name="L582"></a><tt class="py-lineno">582</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_handle</tt> </tt>
<a name="L583"></a><tt class="py-lineno">583</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-235" class="py-name"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-235', 'seek', 'link-221');">seek</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L584"></a><tt class="py-lineno">584</tt>  <tt class="py-line">        <tt class="py-name">start_offset</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L585"></a><tt class="py-lineno">585</tt>  <tt class="py-line">        <tt class="py-name">qresult_key</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L586"></a><tt class="py-lineno">586</tt>  <tt class="py-line">        <tt class="py-name">key_idx</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_key_idx</tt> </tt>
<a name="L587"></a><tt class="py-lineno">587</tt>  <tt class="py-line">        <tt class="py-name">tab_char</tt> <tt class="py-op">=</tt> <tt id="link-236" class="py-name"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-236', '_as_bytes', 'link-2');">_as_bytes</a></tt><tt class="py-op">(</tt><tt class="py-string">'\t'</tt><tt class="py-op">)</tt> </tt>
<a name="L588"></a><tt class="py-lineno">588</tt>  <tt class="py-line"> </tt>
<a name="L589"></a><tt class="py-lineno">589</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L590"></a><tt class="py-lineno">590</tt>  <tt class="py-line">            <tt class="py-comment"># get end offset here since we only know a qresult ends after</tt> </tt>
<a name="L591"></a><tt class="py-lineno">591</tt>  <tt class="py-line">            <tt class="py-comment"># encountering the next one</tt> </tt>
<a name="L592"></a><tt class="py-lineno">592</tt>  <tt class="py-line">            <tt class="py-name">end_offset</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-237" class="py-name"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-237', 'tell', 'link-230');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L593"></a><tt class="py-lineno">593</tt>  <tt class="py-line">            <tt class="py-comment"># line = handle.readline()</tt> </tt>
<a name="L594"></a><tt class="py-lineno">594</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-238" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-238', 'readline', 'link-103');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L595"></a><tt class="py-lineno">595</tt>  <tt class="py-line"> </tt>
<a name="L596"></a><tt class="py-lineno">596</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">qresult_key</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L597"></a><tt class="py-lineno">597</tt>  <tt class="py-line">                <tt class="py-name">qresult_key</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-239" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-239', 'split', 'link-21');">split</a></tt><tt class="py-op">(</tt><tt class="py-name">tab_char</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-name">key_idx</tt><tt class="py-op">]</tt> </tt>
<a name="L598"></a><tt class="py-lineno">598</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L599"></a><tt class="py-lineno">599</tt>  <tt class="py-line">                <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L600"></a><tt class="py-lineno">600</tt>  <tt class="py-line">                    <tt class="py-name">curr_key</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-240" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-240', 'split', 'link-21');">split</a></tt><tt class="py-op">(</tt><tt class="py-name">tab_char</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-name">key_idx</tt><tt class="py-op">]</tt> </tt>
<a name="L601"></a><tt class="py-lineno">601</tt>  <tt class="py-line">                <tt class="py-keyword">except</tt> <tt class="py-name">IndexError</tt><tt class="py-op">:</tt> </tt>
<a name="L602"></a><tt class="py-lineno">602</tt>  <tt class="py-line">                    <tt class="py-name">curr_key</tt> <tt class="py-op">=</tt> <tt id="link-241" class="py-name"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-241', '_as_bytes', 'link-2');">_as_bytes</a></tt><tt class="py-op">(</tt><tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L603"></a><tt class="py-lineno">603</tt>  <tt class="py-line"> </tt>
<a name="L604"></a><tt class="py-lineno">604</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">curr_key</tt> <tt class="py-op">!=</tt> <tt class="py-name">qresult_key</tt><tt class="py-op">:</tt> </tt>
<a name="L605"></a><tt class="py-lineno">605</tt>  <tt class="py-line">                    <tt class="py-keyword">yield</tt> <tt class="py-name">qresult_key</tt><tt class="py-op">,</tt> <tt class="py-name">start_offset</tt><tt class="py-op">,</tt> <tt class="py-name">end_offset</tt> <tt class="py-op">-</tt> <tt class="py-name">start_offset</tt> </tt>
<a name="L606"></a><tt class="py-lineno">606</tt>  <tt class="py-line">                    <tt class="py-name">qresult_key</tt> <tt class="py-op">=</tt> <tt class="py-name">curr_key</tt> </tt>
<a name="L607"></a><tt class="py-lineno">607</tt>  <tt class="py-line">                    <tt class="py-name">start_offset</tt> <tt class="py-op">=</tt> <tt class="py-name">end_offset</tt> </tt>
<a name="L608"></a><tt class="py-lineno">608</tt>  <tt class="py-line"> </tt>
<a name="L609"></a><tt class="py-lineno">609</tt>  <tt class="py-line">            <tt class="py-comment"># break if we've reached EOF</tt> </tt>
<a name="L610"></a><tt class="py-lineno">610</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L611"></a><tt class="py-lineno">611</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
</div><a name="L612"></a><tt class="py-lineno">612</tt>  <tt class="py-line"> </tt>
<a name="BlastTabIndexer.get_raw"></a><div id="BlastTabIndexer.get_raw-def"><a name="L613"></a><tt class="py-lineno">613</tt> <a class="py-toggle" href="#" id="BlastTabIndexer.get_raw-toggle" onclick="return toggle('BlastTabIndexer.get_raw');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer-class.html#get_raw">get_raw</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">offset</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabIndexer.get_raw-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlastTabIndexer.get_raw-expanded"><a name="L614"></a><tt class="py-lineno">614</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the raw bytes string of a QueryResult object from the given offset."""</tt> </tt>
<a name="L615"></a><tt class="py-lineno">615</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_kwargs</tt><tt class="py-op">[</tt><tt class="py-string">'comments'</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L616"></a><tt class="py-lineno">616</tt>  <tt class="py-line">            <tt class="py-name">getfunc</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-242" class="py-name" targets="Method Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer._get_raw_qresult_commented()=Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer-class.html#_get_raw_qresult_commented"><a title="Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer._get_raw_qresult_commented" class="py-name" href="#" onclick="return doclink('link-242', '_get_raw_qresult_commented', 'link-242');">_get_raw_qresult_commented</a></tt> </tt>
<a name="L617"></a><tt class="py-lineno">617</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L618"></a><tt class="py-lineno">618</tt>  <tt class="py-line">            <tt class="py-name">getfunc</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-243" class="py-name" targets="Method Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer._get_raw_qresult()=Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer-class.html#_get_raw_qresult"><a title="Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer._get_raw_qresult" class="py-name" href="#" onclick="return doclink('link-243', '_get_raw_qresult', 'link-243');">_get_raw_qresult</a></tt> </tt>
<a name="L619"></a><tt class="py-lineno">619</tt>  <tt class="py-line"> </tt>
<a name="L620"></a><tt class="py-lineno">620</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">getfunc</tt><tt class="py-op">(</tt><tt class="py-name">offset</tt><tt class="py-op">)</tt> </tt>
</div><a name="L621"></a><tt class="py-lineno">621</tt>  <tt class="py-line"> </tt>
<a name="BlastTabIndexer._get_raw_qresult"></a><div id="BlastTabIndexer._get_raw_qresult-def"><a name="L622"></a><tt class="py-lineno">622</tt> <a class="py-toggle" href="#" id="BlastTabIndexer._get_raw_qresult-toggle" onclick="return toggle('BlastTabIndexer._get_raw_qresult');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer-class.html#_get_raw_qresult">_get_raw_qresult</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">offset</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabIndexer._get_raw_qresult-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlastTabIndexer._get_raw_qresult-expanded"><a name="L623"></a><tt class="py-lineno">623</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the raw bytes string of a single QueryResult from a noncommented file (PRIVATE)."""</tt> </tt>
<a name="L624"></a><tt class="py-lineno">624</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_handle</tt> </tt>
<a name="L625"></a><tt class="py-lineno">625</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-244" class="py-name"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-244', 'seek', 'link-221');">seek</a></tt><tt class="py-op">(</tt><tt class="py-name">offset</tt><tt class="py-op">)</tt> </tt>
<a name="L626"></a><tt class="py-lineno">626</tt>  <tt class="py-line">        <tt class="py-name">qresult_raw</tt> <tt class="py-op">=</tt> <tt id="link-245" class="py-name"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-245', '_as_bytes', 'link-2');">_as_bytes</a></tt><tt class="py-op">(</tt><tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L627"></a><tt class="py-lineno">627</tt>  <tt class="py-line">        <tt class="py-name">tab_char</tt> <tt class="py-op">=</tt> <tt id="link-246" class="py-name"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-246', '_as_bytes', 'link-2');">_as_bytes</a></tt><tt class="py-op">(</tt><tt class="py-string">'\t'</tt><tt class="py-op">)</tt> </tt>
<a name="L628"></a><tt class="py-lineno">628</tt>  <tt class="py-line">        <tt class="py-name">key_idx</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_key_idx</tt> </tt>
<a name="L629"></a><tt class="py-lineno">629</tt>  <tt class="py-line">        <tt class="py-name">qresult_key</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L630"></a><tt class="py-lineno">630</tt>  <tt class="py-line"> </tt>
<a name="L631"></a><tt class="py-lineno">631</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L632"></a><tt class="py-lineno">632</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-247" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-247', 'readline', 'link-103');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L633"></a><tt class="py-lineno">633</tt>  <tt class="py-line">            <tt class="py-comment"># get the key if the first line (qresult key)</tt> </tt>
<a name="L634"></a><tt class="py-lineno">634</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">qresult_key</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L635"></a><tt class="py-lineno">635</tt>  <tt class="py-line">                <tt class="py-name">qresult_key</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-248" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-248', 'split', 'link-21');">split</a></tt><tt class="py-op">(</tt><tt class="py-name">tab_char</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-name">key_idx</tt><tt class="py-op">]</tt> </tt>
<a name="L636"></a><tt class="py-lineno">636</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L637"></a><tt class="py-lineno">637</tt>  <tt class="py-line">                <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L638"></a><tt class="py-lineno">638</tt>  <tt class="py-line">                    <tt class="py-name">curr_key</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-249" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-249', 'split', 'link-21');">split</a></tt><tt class="py-op">(</tt><tt class="py-name">tab_char</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-name">key_idx</tt><tt class="py-op">]</tt> </tt>
<a name="L639"></a><tt class="py-lineno">639</tt>  <tt class="py-line">                <tt class="py-keyword">except</tt> <tt class="py-name">IndexError</tt><tt class="py-op">:</tt> </tt>
<a name="L640"></a><tt class="py-lineno">640</tt>  <tt class="py-line">                    <tt class="py-name">curr_key</tt> <tt class="py-op">=</tt> <tt id="link-250" class="py-name"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-250', '_as_bytes', 'link-2');">_as_bytes</a></tt><tt class="py-op">(</tt><tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L641"></a><tt class="py-lineno">641</tt>  <tt class="py-line">                <tt class="py-comment"># only break when qresult is finished (key is different)</tt> </tt>
<a name="L642"></a><tt class="py-lineno">642</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">curr_key</tt> <tt class="py-op">!=</tt> <tt class="py-name">qresult_key</tt><tt class="py-op">:</tt> </tt>
<a name="L643"></a><tt class="py-lineno">643</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L644"></a><tt class="py-lineno">644</tt>  <tt class="py-line">            <tt class="py-comment"># append to the raw string as long as qresult is the same</tt> </tt>
<a name="L645"></a><tt class="py-lineno">645</tt>  <tt class="py-line">            <tt class="py-name">qresult_raw</tt> <tt class="py-op">+=</tt> <tt class="py-name">line</tt> </tt>
<a name="L646"></a><tt class="py-lineno">646</tt>  <tt class="py-line"> </tt>
<a name="L647"></a><tt class="py-lineno">647</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">qresult_raw</tt> </tt>
</div><a name="L648"></a><tt class="py-lineno">648</tt>  <tt class="py-line"> </tt>
<a name="BlastTabIndexer._get_raw_qresult_commented"></a><div id="BlastTabIndexer._get_raw_qresult_commented-def"><a name="L649"></a><tt class="py-lineno">649</tt> <a class="py-toggle" href="#" id="BlastTabIndexer._get_raw_qresult_commented-toggle" onclick="return toggle('BlastTabIndexer._get_raw_qresult_commented');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer-class.html#_get_raw_qresult_commented">_get_raw_qresult_commented</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">offset</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabIndexer._get_raw_qresult_commented-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlastTabIndexer._get_raw_qresult_commented-expanded"><a name="L650"></a><tt class="py-lineno">650</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the bytes raw string of a single QueryResult from a commented file (PRIVATE)."""</tt> </tt>
<a name="L651"></a><tt class="py-lineno">651</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_handle</tt> </tt>
<a name="L652"></a><tt class="py-lineno">652</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-251" class="py-name"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-251', 'seek', 'link-221');">seek</a></tt><tt class="py-op">(</tt><tt class="py-name">offset</tt><tt class="py-op">)</tt> </tt>
<a name="L653"></a><tt class="py-lineno">653</tt>  <tt class="py-line">        <tt class="py-name">qresult_raw</tt> <tt class="py-op">=</tt> <tt id="link-252" class="py-name"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-252', '_as_bytes', 'link-2');">_as_bytes</a></tt><tt class="py-op">(</tt><tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L654"></a><tt class="py-lineno">654</tt>  <tt class="py-line">        <tt class="py-name">end_mark</tt> <tt class="py-op">=</tt> <tt id="link-253" class="py-name"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-253', '_as_bytes', 'link-2');">_as_bytes</a></tt><tt class="py-op">(</tt><tt class="py-string">'# BLAST processed'</tt><tt class="py-op">)</tt> </tt>
<a name="L655"></a><tt class="py-lineno">655</tt>  <tt class="py-line"> </tt>
<a name="L656"></a><tt class="py-lineno">656</tt>  <tt class="py-line">        <tt class="py-comment"># query mark is the line marking a new query</tt> </tt>
<a name="L657"></a><tt class="py-lineno">657</tt>  <tt class="py-line">        <tt class="py-comment"># something like '# TBLASTN 2.2.25+'</tt> </tt>
<a name="L658"></a><tt class="py-lineno">658</tt>  <tt class="py-line">        <tt class="py-name">query_mark</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L659"></a><tt class="py-lineno">659</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-254" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-254', 'readline', 'link-103');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L660"></a><tt class="py-lineno">660</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L661"></a><tt class="py-lineno">661</tt>  <tt class="py-line">            <tt class="py-comment"># since query_mark depends on the BLAST search, we need to obtain it</tt> </tt>
<a name="L662"></a><tt class="py-lineno">662</tt>  <tt class="py-line">            <tt class="py-comment"># first</tt> </tt>
<a name="L663"></a><tt class="py-lineno">663</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">query_mark</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L664"></a><tt class="py-lineno">664</tt>  <tt class="py-line">                <tt class="py-name">query_mark</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> </tt>
<a name="L665"></a><tt class="py-lineno">665</tt>  <tt class="py-line">            <tt class="py-comment"># break when we've reached the next qresult or the search ends</tt> </tt>
<a name="L666"></a><tt class="py-lineno">666</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt> <tt class="py-op">==</tt> <tt class="py-name">query_mark</tt> <tt class="py-keyword">or</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-255" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-255', 'startswith', 'link-59');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-name">end_mark</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L667"></a><tt class="py-lineno">667</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L668"></a><tt class="py-lineno">668</tt>  <tt class="py-line"> </tt>
<a name="L669"></a><tt class="py-lineno">669</tt>  <tt class="py-line">            <tt class="py-name">qresult_raw</tt> <tt class="py-op">+=</tt> <tt class="py-name">line</tt> </tt>
<a name="L670"></a><tt class="py-lineno">670</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-256" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-256', 'readline', 'link-103');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L671"></a><tt class="py-lineno">671</tt>  <tt class="py-line"> </tt>
<a name="L672"></a><tt class="py-lineno">672</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">qresult_raw</tt> </tt>
</div></div><a name="L673"></a><tt class="py-lineno">673</tt>  <tt class="py-line"> </tt>
<a name="L674"></a><tt class="py-lineno">674</tt>  <tt class="py-line"> </tt>
<a name="BlastTabWriter"></a><div id="BlastTabWriter-def"><a name="L675"></a><tt class="py-lineno">675</tt> <a class="py-toggle" href="#" id="BlastTabWriter-toggle" onclick="return toggle('BlastTabWriter');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter-class.html">BlastTabWriter</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabWriter-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="BlastTabWriter-expanded"><a name="L676"></a><tt class="py-lineno">676</tt>  <tt class="py-line">    <tt class="py-docstring">"""Writer for blast-tab output format."""</tt> </tt>
<a name="L677"></a><tt class="py-lineno">677</tt>  <tt class="py-line"> </tt>
<a name="BlastTabWriter.__init__"></a><div id="BlastTabWriter.__init__-def"><a name="L678"></a><tt class="py-lineno">678</tt> <a class="py-toggle" href="#" id="BlastTabWriter.__init__-toggle" onclick="return toggle('BlastTabWriter.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">comments</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> <tt class="py-param">fields</tt><tt class="py-op">=</tt><tt id="link-257" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._DEFAULT_FIELDS" class="py-name" href="#" onclick="return doclink('link-257', '_DEFAULT_FIELDS', 'link-44');">_DEFAULT_FIELDS</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabWriter.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlastTabWriter.__init__-expanded"><a name="L679"></a><tt class="py-lineno">679</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the class."""</tt> </tt>
<a name="L680"></a><tt class="py-lineno">680</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt> </tt>
<a name="L681"></a><tt class="py-lineno">681</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">has_comments</tt> <tt class="py-op">=</tt> <tt class="py-name">comments</tt> </tt>
<a name="L682"></a><tt class="py-lineno">682</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">fields</tt> <tt class="py-op">=</tt> <tt class="py-name">fields</tt> </tt>
</div><a name="L683"></a><tt class="py-lineno">683</tt>  <tt class="py-line"> </tt>
<a name="BlastTabWriter.write_file"></a><div id="BlastTabWriter.write_file-def"><a name="L684"></a><tt class="py-lineno">684</tt> <a class="py-toggle" href="#" id="BlastTabWriter.write_file-toggle" onclick="return toggle('BlastTabWriter.write_file');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter-class.html#write_file">write_file</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">qresults</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabWriter.write_file-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlastTabWriter.write_file-expanded"><a name="L685"></a><tt class="py-lineno">685</tt>  <tt class="py-line">        <tt class="py-docstring">"""Write to the handle, return how many QueryResult objects were written."""</tt> </tt>
<a name="L686"></a><tt class="py-lineno">686</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt> </tt>
<a name="L687"></a><tt class="py-lineno">687</tt>  <tt class="py-line">        <tt class="py-name">qresult_counter</tt><tt class="py-op">,</tt> <tt class="py-name">hit_counter</tt><tt class="py-op">,</tt> <tt class="py-name">hsp_counter</tt><tt class="py-op">,</tt> <tt class="py-name">frag_counter</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt> </tt>
<a name="L688"></a><tt class="py-lineno">688</tt>  <tt class="py-line"> </tt>
<a name="L689"></a><tt class="py-lineno">689</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">qresult</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qresults</tt><tt class="py-op">:</tt> </tt>
<a name="L690"></a><tt class="py-lineno">690</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">has_comments</tt><tt class="py-op">:</tt> </tt>
<a name="L691"></a><tt class="py-lineno">691</tt>  <tt class="py-line">                <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-258" class="py-name" targets="Function Bio.AlignIO.write()=Bio.AlignIO-module.html#write,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.write()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#write,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.write()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#write,Method Bio.Phylo.CDAOIO.Writer.write()=Bio.Phylo.CDAOIO.Writer-class.html#write,Function Bio.Phylo.CDAOIO.write()=Bio.Phylo.CDAOIO-module.html#write,Method Bio.Phylo.NeXMLIO.Writer.write()=Bio.Phylo.NeXMLIO.Writer-class.html#write,Function Bio.Phylo.NeXMLIO.write()=Bio.Phylo.NeXMLIO-module.html#write,Method Bio.Phylo.NewickIO.Writer.write()=Bio.Phylo.NewickIO.Writer-class.html#write,Function Bio.Phylo.NewickIO.write()=Bio.Phylo.NewickIO-module.html#write,Function Bio.Phylo.NexusIO.write()=Bio.Phylo.NexusIO-module.html#write,Method Bio.Phylo.PhyloXMLIO.Writer.write()=Bio.Phylo.PhyloXMLIO.Writer-class.html#write,Function Bio.Phylo.PhyloXMLIO.write()=Bio.Phylo.PhyloXMLIO-module.html#write,Function Bio.Phylo._io.write()=Bio.Phylo._io-module.html#write,Function Bio.SearchIO.write()=Bio.SearchIO-module.html#write,Function Bio.SeqIO.write()=Bio.SeqIO-module.html#write,Method Bio.bgzf.BgzfWriter.write()=Bio.bgzf.BgzfWriter-class.html#write,Function Bio.motifs.jaspar.write()=Bio.motifs.jaspar-module.html#write,Function Bio.motifs.transfac.write()=Bio.motifs.transfac-module.html#write,Function Bio.motifs.write()=Bio.motifs-module.html#write,Method Bio.phenotype.phen_micro.JsonWriter.write()=Bio.phenotype.phen_micro.JsonWriter-class.html#write,Function Bio.phenotype.write()=Bio.phenotype-module.html#write"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-258', 'write', 'link-258');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-259" class="py-name" targets="Method Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter._build_comments()=Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter-class.html#_build_comments"><a title="Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter._build_comments" class="py-name" href="#" onclick="return doclink('link-259', '_build_comments', 'link-259');">_build_comments</a></tt><tt class="py-op">(</tt><tt class="py-name">qresult</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L692"></a><tt class="py-lineno">692</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">qresult</tt><tt class="py-op">:</tt> </tt>
<a name="L693"></a><tt class="py-lineno">693</tt>  <tt class="py-line">                <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-260" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-260', 'write', 'link-258');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-261" class="py-name" targets="Method Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter._build_rows()=Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter-class.html#_build_rows"><a title="Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter._build_rows" class="py-name" href="#" onclick="return doclink('link-261', '_build_rows', 'link-261');">_build_rows</a></tt><tt class="py-op">(</tt><tt class="py-name">qresult</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L694"></a><tt class="py-lineno">694</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">has_comments</tt><tt class="py-op">:</tt> </tt>
<a name="L695"></a><tt class="py-lineno">695</tt>  <tt class="py-line">                    <tt class="py-name">qresult_counter</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L696"></a><tt class="py-lineno">696</tt>  <tt class="py-line">                <tt class="py-name">hit_counter</tt> <tt class="py-op">+=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">qresult</tt><tt class="py-op">)</tt> </tt>
<a name="L697"></a><tt class="py-lineno">697</tt>  <tt class="py-line">                <tt class="py-name">hsp_counter</tt> <tt class="py-op">+=</tt> <tt id="link-262" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.sum()=Bio.Nexus.Nexus.StepMatrix-class.html#sum,Method Bio.SubsMat.SeqMat.sum()=Bio.SubsMat.SeqMat-class.html#sum"><a title="Bio.Nexus.Nexus.StepMatrix.sum
Bio.SubsMat.SeqMat.sum" class="py-name" href="#" onclick="return doclink('link-262', 'sum', 'link-262');">sum</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-263" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-263', 'hit', 'link-135');">hit</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-264" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-264', 'hit', 'link-135');">hit</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">qresult</tt><tt class="py-op">)</tt> </tt>
<a name="L698"></a><tt class="py-lineno">698</tt>  <tt class="py-line">                <tt class="py-name">frag_counter</tt> <tt class="py-op">+=</tt> <tt id="link-265" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.sum
Bio.SubsMat.SeqMat.sum" class="py-name" href="#" onclick="return doclink('link-265', 'sum', 'link-262');">sum</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-266" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-266', 'hit', 'link-135');">hit</a></tt><tt class="py-op">.</tt><tt id="link-267" class="py-name" targets="Variable Bio.SearchIO._model.hit.Hit.fragments=Bio.SearchIO._model.hit.Hit-class.html#fragments,Variable Bio.SearchIO._model.hsp.HSP.fragments=Bio.SearchIO._model.hsp.HSP-class.html#fragments,Variable Bio.SearchIO._model.query.QueryResult.fragments=Bio.SearchIO._model.query.QueryResult-class.html#fragments"><a title="Bio.SearchIO._model.hit.Hit.fragments
Bio.SearchIO._model.hsp.HSP.fragments
Bio.SearchIO._model.query.QueryResult.fragments" class="py-name" href="#" onclick="return doclink('link-267', 'fragments', 'link-267');">fragments</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-268" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-268', 'hit', 'link-135');">hit</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">qresult</tt><tt class="py-op">)</tt> </tt>
<a name="L699"></a><tt class="py-lineno">699</tt>  <tt class="py-line">            <tt class="py-comment"># if it's commented and there are no hits in the qresult, we still</tt> </tt>
<a name="L700"></a><tt class="py-lineno">700</tt>  <tt class="py-line">            <tt class="py-comment"># increment the counter</tt> </tt>
<a name="L701"></a><tt class="py-lineno">701</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">has_comments</tt><tt class="py-op">:</tt> </tt>
<a name="L702"></a><tt class="py-lineno">702</tt>  <tt class="py-line">                <tt class="py-name">qresult_counter</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L703"></a><tt class="py-lineno">703</tt>  <tt class="py-line"> </tt>
<a name="L704"></a><tt class="py-lineno">704</tt>  <tt class="py-line">        <tt class="py-comment"># commented files have a line saying how many queries were processed</tt> </tt>
<a name="L705"></a><tt class="py-lineno">705</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">has_comments</tt><tt class="py-op">:</tt> </tt>
<a name="L706"></a><tt class="py-lineno">706</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-269" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-269', 'write', 'link-258');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">'# BLAST processed %i queries'</tt> <tt class="py-op">%</tt> <tt class="py-name">qresult_counter</tt><tt class="py-op">)</tt> </tt>
<a name="L707"></a><tt class="py-lineno">707</tt>  <tt class="py-line"> </tt>
<a name="L708"></a><tt class="py-lineno">708</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">qresult_counter</tt><tt class="py-op">,</tt> <tt class="py-name">hit_counter</tt><tt class="py-op">,</tt> <tt class="py-name">hsp_counter</tt><tt class="py-op">,</tt> <tt class="py-name">frag_counter</tt> </tt>
</div><a name="L709"></a><tt class="py-lineno">709</tt>  <tt class="py-line"> </tt>
<a name="BlastTabWriter._build_rows"></a><div id="BlastTabWriter._build_rows-def"><a name="L710"></a><tt class="py-lineno">710</tt> <a class="py-toggle" href="#" id="BlastTabWriter._build_rows-toggle" onclick="return toggle('BlastTabWriter._build_rows');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter-class.html#_build_rows">_build_rows</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">qresult</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabWriter._build_rows-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlastTabWriter._build_rows-expanded"><a name="L711"></a><tt class="py-lineno">711</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a string containing tabular rows of the QueryResult object (PRIVATE)."""</tt> </tt>
<a name="L712"></a><tt class="py-lineno">712</tt>  <tt class="py-line">        <tt class="py-name">coordinates</tt> <tt class="py-op">=</tt> <tt id="link-270" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-270', 'set', 'link-39');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-string">'qstart'</tt><tt class="py-op">,</tt> <tt class="py-string">'qend'</tt><tt class="py-op">,</tt> <tt class="py-string">'sstart'</tt><tt class="py-op">,</tt> <tt class="py-string">'send'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L713"></a><tt class="py-lineno">713</tt>  <tt class="py-line">        <tt class="py-name">qresult_lines</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L714"></a><tt class="py-lineno">714</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-271" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-271', 'hit', 'link-135');">hit</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">qresult</tt><tt class="py-op">:</tt> </tt>
<a name="L715"></a><tt class="py-lineno">715</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-272" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-272', 'hsp', 'link-51');">hsp</a></tt> <tt class="py-keyword">in</tt> <tt id="link-273" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-273', 'hit', 'link-135');">hit</a></tt><tt class="py-op">:</tt> </tt>
<a name="L716"></a><tt class="py-lineno">716</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L717"></a><tt class="py-lineno">717</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">field</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">fields</tt><tt class="py-op">:</tt> </tt>
<a name="L718"></a><tt class="py-lineno">718</tt>  <tt class="py-line">                    <tt class="py-comment"># get the column value ~ could either be an attribute</tt> </tt>
<a name="L719"></a><tt class="py-lineno">719</tt>  <tt class="py-line">                    <tt class="py-comment"># of qresult, hit, or hsp</tt> </tt>
<a name="L720"></a><tt class="py-lineno">720</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">field</tt> <tt class="py-keyword">in</tt> <tt id="link-274" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._COLUMN_QRESULT" class="py-name" href="#" onclick="return doclink('link-274', '_COLUMN_QRESULT', 'link-26');">_COLUMN_QRESULT</a></tt><tt class="py-op">:</tt> </tt>
<a name="L721"></a><tt class="py-lineno">721</tt>  <tt class="py-line">                        <tt id="link-275" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-275', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">qresult</tt><tt class="py-op">,</tt> <tt id="link-276" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._COLUMN_QRESULT" class="py-name" href="#" onclick="return doclink('link-276', '_COLUMN_QRESULT', 'link-26');">_COLUMN_QRESULT</a></tt><tt class="py-op">[</tt><tt class="py-name">field</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L722"></a><tt class="py-lineno">722</tt>  <tt class="py-line">                    <tt class="py-keyword">elif</tt> <tt class="py-name">field</tt> <tt class="py-keyword">in</tt> <tt id="link-277" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._COLUMN_HIT" class="py-name" href="#" onclick="return doclink('link-277', '_COLUMN_HIT', 'link-27');">_COLUMN_HIT</a></tt><tt class="py-op">:</tt> </tt>
<a name="L723"></a><tt class="py-lineno">723</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">field</tt> <tt class="py-op">==</tt> <tt class="py-string">'sallseqid'</tt><tt class="py-op">:</tt> </tt>
<a name="L724"></a><tt class="py-lineno">724</tt>  <tt class="py-line">                            <tt id="link-278" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-278', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt id="link-279" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-279', 'hit', 'link-135');">hit</a></tt><tt class="py-op">,</tt> <tt class="py-string">'id_all'</tt><tt class="py-op">)</tt> </tt>
<a name="L725"></a><tt class="py-lineno">725</tt>  <tt class="py-line">                        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L726"></a><tt class="py-lineno">726</tt>  <tt class="py-line">                            <tt id="link-280" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-280', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt id="link-281" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-281', 'hit', 'link-135');">hit</a></tt><tt class="py-op">,</tt> <tt id="link-282" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._COLUMN_HIT" class="py-name" href="#" onclick="return doclink('link-282', '_COLUMN_HIT', 'link-27');">_COLUMN_HIT</a></tt><tt class="py-op">[</tt><tt class="py-name">field</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L727"></a><tt class="py-lineno">727</tt>  <tt class="py-line">                    <tt class="py-comment"># special case, since 'frames' can be determined from</tt> </tt>
<a name="L728"></a><tt class="py-lineno">728</tt>  <tt class="py-line">                    <tt class="py-comment"># query frame and hit frame</tt> </tt>
<a name="L729"></a><tt class="py-lineno">729</tt>  <tt class="py-line">                    <tt class="py-keyword">elif</tt> <tt class="py-name">field</tt> <tt class="py-op">==</tt> <tt class="py-string">'frames'</tt><tt class="py-op">:</tt> </tt>
<a name="L730"></a><tt class="py-lineno">730</tt>  <tt class="py-line">                        <tt id="link-283" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-283', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'%i/%i'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-284" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-284', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-285" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.query_frame=Bio.SearchIO._model.hsp.HSP-class.html#query_frame,Variable Bio.SearchIO._model.hsp.HSPFragment.query_frame=Bio.SearchIO._model.hsp.HSPFragment-class.html#query_frame"><a title="Bio.SearchIO._model.hsp.HSP.query_frame
Bio.SearchIO._model.hsp.HSPFragment.query_frame" class="py-name" href="#" onclick="return doclink('link-285', 'query_frame', 'link-285');">query_frame</a></tt><tt class="py-op">,</tt> <tt id="link-286" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-286', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-287" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.hit_frame=Bio.SearchIO._model.hsp.HSP-class.html#hit_frame,Variable Bio.SearchIO._model.hsp.HSPFragment.hit_frame=Bio.SearchIO._model.hsp.HSPFragment-class.html#hit_frame"><a title="Bio.SearchIO._model.hsp.HSP.hit_frame
Bio.SearchIO._model.hsp.HSPFragment.hit_frame" class="py-name" href="#" onclick="return doclink('link-287', 'hit_frame', 'link-287');">hit_frame</a></tt><tt class="py-op">)</tt> </tt>
<a name="L731"></a><tt class="py-lineno">731</tt>  <tt class="py-line">                    <tt class="py-keyword">elif</tt> <tt class="py-name">field</tt> <tt class="py-keyword">in</tt> <tt id="link-288" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._COLUMN_HSP" class="py-name" href="#" onclick="return doclink('link-288', '_COLUMN_HSP', 'link-36');">_COLUMN_HSP</a></tt><tt class="py-op">:</tt> </tt>
<a name="L732"></a><tt class="py-lineno">732</tt>  <tt class="py-line">                        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L733"></a><tt class="py-lineno">733</tt>  <tt class="py-line">                            <tt id="link-289" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-289', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt id="link-290" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-290', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">,</tt> <tt id="link-291" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._COLUMN_HSP" class="py-name" href="#" onclick="return doclink('link-291', '_COLUMN_HSP', 'link-36');">_COLUMN_HSP</a></tt><tt class="py-op">[</tt><tt class="py-name">field</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L734"></a><tt class="py-lineno">734</tt>  <tt class="py-line">                        <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L735"></a><tt class="py-lineno">735</tt>  <tt class="py-line">                            <tt class="py-name">attr</tt> <tt class="py-op">=</tt> <tt id="link-292" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._COLUMN_HSP" class="py-name" href="#" onclick="return doclink('link-292', '_COLUMN_HSP', 'link-36');">_COLUMN_HSP</a></tt><tt class="py-op">[</tt><tt class="py-name">field</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L736"></a><tt class="py-lineno">736</tt>  <tt class="py-line">                            <tt id="link-293" class="py-name" targets="Function Bio.SearchIO.BlastIO.blast_tab._augment_blast_hsp()=Bio.SearchIO.BlastIO.blast_tab-module.html#_augment_blast_hsp"><a title="Bio.SearchIO.BlastIO.blast_tab._augment_blast_hsp" class="py-name" href="#" onclick="return doclink('link-293', '_augment_blast_hsp', 'link-293');">_augment_blast_hsp</a></tt><tt class="py-op">(</tt><tt id="link-294" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-294', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">,</tt> <tt class="py-name">attr</tt><tt class="py-op">)</tt> </tt>
<a name="L737"></a><tt class="py-lineno">737</tt>  <tt class="py-line">                            <tt id="link-295" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-295', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt id="link-296" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-296', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">,</tt> <tt class="py-name">attr</tt><tt class="py-op">)</tt> </tt>
<a name="L738"></a><tt class="py-lineno">738</tt>  <tt class="py-line">                    <tt class="py-keyword">elif</tt> <tt class="py-name">field</tt> <tt class="py-keyword">in</tt> <tt id="link-297" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._COLUMN_FRAG" class="py-name" href="#" onclick="return doclink('link-297', '_COLUMN_FRAG', 'link-37');">_COLUMN_FRAG</a></tt><tt class="py-op">:</tt> </tt>
<a name="L739"></a><tt class="py-lineno">739</tt>  <tt class="py-line">                        <tt id="link-298" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-298', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt id="link-299" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-299', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">,</tt> <tt id="link-300" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._COLUMN_FRAG" class="py-name" href="#" onclick="return doclink('link-300', '_COLUMN_FRAG', 'link-37');">_COLUMN_FRAG</a></tt><tt class="py-op">[</tt><tt class="py-name">field</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L740"></a><tt class="py-lineno">740</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L741"></a><tt class="py-lineno">741</tt>  <tt class="py-line">                        <tt class="py-keyword">assert</tt> <tt class="py-name">field</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt id="link-301" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._SUPPORTED_FIELDS" class="py-name" href="#" onclick="return doclink('link-301', '_SUPPORTED_FIELDS', 'link-38');">_SUPPORTED_FIELDS</a></tt> </tt>
<a name="L742"></a><tt class="py-lineno">742</tt>  <tt class="py-line">                        <tt class="py-keyword">continue</tt> </tt>
<a name="L743"></a><tt class="py-lineno">743</tt>  <tt class="py-line"> </tt>
<a name="L744"></a><tt class="py-lineno">744</tt>  <tt class="py-line">                    <tt class="py-comment"># adjust from and to according to strand, if from and to</tt> </tt>
<a name="L745"></a><tt class="py-lineno">745</tt>  <tt class="py-line">                    <tt class="py-comment"># is included in the output field</tt> </tt>
<a name="L746"></a><tt class="py-lineno">746</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">field</tt> <tt class="py-keyword">in</tt> <tt class="py-name">coordinates</tt><tt class="py-op">:</tt> </tt>
<a name="L747"></a><tt class="py-lineno">747</tt>  <tt class="py-line">                        <tt id="link-302" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-302', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-303" class="py-name" targets="Method Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter._adjust_coords()=Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter-class.html#_adjust_coords"><a title="Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter._adjust_coords" class="py-name" href="#" onclick="return doclink('link-303', '_adjust_coords', 'link-303');">_adjust_coords</a></tt><tt class="py-op">(</tt><tt class="py-name">field</tt><tt class="py-op">,</tt> <tt id="link-304" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-304', 'value', 'link-81');">value</a></tt><tt class="py-op">,</tt> <tt id="link-305" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-305', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">)</tt> </tt>
<a name="L748"></a><tt class="py-lineno">748</tt>  <tt class="py-line">                    <tt class="py-comment"># adjust output formatting</tt> </tt>
<a name="L749"></a><tt class="py-lineno">749</tt>  <tt class="py-line">                    <tt id="link-306" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-306', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-307" class="py-name" targets="Method Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter._adjust_output()=Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter-class.html#_adjust_output,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter._adjust_output()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter-class.html#_adjust_output"><a title="Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter._adjust_output
Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter._adjust_output" class="py-name" href="#" onclick="return doclink('link-307', '_adjust_output', 'link-307');">_adjust_output</a></tt><tt class="py-op">(</tt><tt class="py-name">field</tt><tt class="py-op">,</tt> <tt id="link-308" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-308', 'value', 'link-81');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L750"></a><tt class="py-lineno">750</tt>  <tt class="py-line"> </tt>
<a name="L751"></a><tt class="py-lineno">751</tt>  <tt class="py-line">                    <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-309" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-309', 'append', 'link-180');">append</a></tt><tt class="py-op">(</tt><tt id="link-310" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-310', 'value', 'link-81');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L752"></a><tt class="py-lineno">752</tt>  <tt class="py-line"> </tt>
<a name="L753"></a><tt class="py-lineno">753</tt>  <tt class="py-line">                <tt class="py-name">hsp_line</tt> <tt class="py-op">=</tt> <tt class="py-string">'\t'</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L754"></a><tt class="py-lineno">754</tt>  <tt class="py-line">                <tt class="py-name">qresult_lines</tt> <tt class="py-op">+=</tt> <tt class="py-name">hsp_line</tt> <tt class="py-op">+</tt> <tt class="py-string">'\n'</tt> </tt>
<a name="L755"></a><tt class="py-lineno">755</tt>  <tt class="py-line"> </tt>
<a name="L756"></a><tt class="py-lineno">756</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">qresult_lines</tt> </tt>
</div><a name="L757"></a><tt class="py-lineno">757</tt>  <tt class="py-line"> </tt>
<a name="BlastTabWriter._adjust_coords"></a><div id="BlastTabWriter._adjust_coords-def"><a name="L758"></a><tt class="py-lineno">758</tt> <a class="py-toggle" href="#" id="BlastTabWriter._adjust_coords-toggle" onclick="return toggle('BlastTabWriter._adjust_coords');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter-class.html#_adjust_coords">_adjust_coords</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">field</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">,</tt> <tt class="py-param">hsp</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabWriter._adjust_coords-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlastTabWriter._adjust_coords-expanded"><a name="L759"></a><tt class="py-lineno">759</tt>  <tt class="py-line">        <tt class="py-docstring">"""Adjust start and end coordinates according to strand (PRIVATE)."""</tt> </tt>
<a name="L760"></a><tt class="py-lineno">760</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">field</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'qstart'</tt><tt class="py-op">,</tt> <tt class="py-string">'qend'</tt><tt class="py-op">,</tt> <tt class="py-string">'sstart'</tt><tt class="py-op">,</tt> <tt class="py-string">'send'</tt><tt class="py-op">)</tt> </tt>
<a name="L761"></a><tt class="py-lineno">761</tt>  <tt class="py-line">        <tt class="py-comment"># determine sequence type to operate on based on field's first letter</tt> </tt>
<a name="L762"></a><tt class="py-lineno">762</tt>  <tt class="py-line">        <tt class="py-name">seq_type</tt> <tt class="py-op">=</tt> <tt class="py-string">'query'</tt> <tt class="py-keyword">if</tt> <tt class="py-name">field</tt><tt class="py-op">.</tt><tt id="link-311" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-311', 'startswith', 'link-59');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'q'</tt><tt class="py-op">)</tt> <tt class="py-keyword">else</tt> <tt class="py-string">'hit'</tt> </tt>
<a name="L763"></a><tt class="py-lineno">763</tt>  <tt class="py-line"> </tt>
<a name="L764"></a><tt class="py-lineno">764</tt>  <tt class="py-line">        <tt id="link-312" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-312', 'strand', 'link-171');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt id="link-313" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-313', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">,</tt> <tt class="py-string">'%s_strand'</tt> <tt class="py-op">%</tt> <tt class="py-name">seq_type</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
<a name="L765"></a><tt class="py-lineno">765</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-314" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-314', 'strand', 'link-171');">strand</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L766"></a><tt class="py-lineno">766</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Required attribute %r not found."</tt> <tt class="py-op">%</tt> </tt>
<a name="L767"></a><tt class="py-lineno">767</tt>  <tt class="py-line">                    <tt class="py-op">(</tt><tt class="py-string">'%s_strand'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">seq_type</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L768"></a><tt class="py-lineno">768</tt>  <tt class="py-line">        <tt class="py-comment"># switch start &lt;--&gt; end coordinates if strand is -1</tt> </tt>
<a name="L769"></a><tt class="py-lineno">769</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-315" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-315', 'strand', 'link-171');">strand</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L770"></a><tt class="py-lineno">770</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">field</tt><tt class="py-op">.</tt><tt id="link-316" class="py-name"><a title="Bio.Seq.Seq.endswith" class="py-name" href="#" onclick="return doclink('link-316', 'endswith', 'link-80');">endswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'start'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L771"></a><tt class="py-lineno">771</tt>  <tt class="py-line">                <tt id="link-317" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-317', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt id="link-318" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-318', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">,</tt> <tt class="py-string">'%s_end'</tt> <tt class="py-op">%</tt> <tt class="py-name">seq_type</tt><tt class="py-op">)</tt> </tt>
<a name="L772"></a><tt class="py-lineno">772</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">field</tt><tt class="py-op">.</tt><tt id="link-319" class="py-name"><a title="Bio.Seq.Seq.endswith" class="py-name" href="#" onclick="return doclink('link-319', 'endswith', 'link-80');">endswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'end'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L773"></a><tt class="py-lineno">773</tt>  <tt class="py-line">                <tt id="link-320" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-320', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt id="link-321" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-321', 'hsp', 'link-51');">hsp</a></tt><tt class="py-op">,</tt> <tt class="py-string">'%s_start'</tt> <tt class="py-op">%</tt> <tt class="py-name">seq_type</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt> </tt>
<a name="L774"></a><tt class="py-lineno">774</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">field</tt><tt class="py-op">.</tt><tt id="link-322" class="py-name"><a title="Bio.Seq.Seq.endswith" class="py-name" href="#" onclick="return doclink('link-322', 'endswith', 'link-80');">endswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'start'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L775"></a><tt class="py-lineno">775</tt>  <tt class="py-line">            <tt class="py-comment"># adjust start coordinate for positive strand</tt> </tt>
<a name="L776"></a><tt class="py-lineno">776</tt>  <tt class="py-line">            <tt id="link-323" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-323', 'value', 'link-81');">value</a></tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L777"></a><tt class="py-lineno">777</tt>  <tt class="py-line"> </tt>
<a name="L778"></a><tt class="py-lineno">778</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-324" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-324', 'value', 'link-81');">value</a></tt> </tt>
</div><a name="L779"></a><tt class="py-lineno">779</tt>  <tt class="py-line"> </tt>
<a name="BlastTabWriter._adjust_output"></a><div id="BlastTabWriter._adjust_output-def"><a name="L780"></a><tt class="py-lineno">780</tt> <a class="py-toggle" href="#" id="BlastTabWriter._adjust_output-toggle" onclick="return toggle('BlastTabWriter._adjust_output');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter-class.html#_adjust_output">_adjust_output</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">field</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabWriter._adjust_output-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlastTabWriter._adjust_output-expanded"><a name="L781"></a><tt class="py-lineno">781</tt>  <tt class="py-line">        <tt class="py-docstring">"""Adjust formatting of given field and value to mimic native tab output (PRIVATE)."""</tt> </tt>
<a name="L782"></a><tt class="py-lineno">782</tt>  <tt class="py-line">        <tt class="py-comment"># qseq and sseq are stored as SeqRecord, but here we only need the str</tt> </tt>
<a name="L783"></a><tt class="py-lineno">783</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">field</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'qseq'</tt><tt class="py-op">,</tt> <tt class="py-string">'sseq'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L784"></a><tt class="py-lineno">784</tt>  <tt class="py-line">            <tt id="link-325" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-325', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-326" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-326', 'value', 'link-81');">value</a></tt><tt class="py-op">.</tt><tt id="link-327" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-327', 'seq', 'link-50');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L785"></a><tt class="py-lineno">785</tt>  <tt class="py-line"> </tt>
<a name="L786"></a><tt class="py-lineno">786</tt>  <tt class="py-line">        <tt class="py-comment"># evalue formatting, adapted from BLAST+ source:</tt> </tt>
<a name="L787"></a><tt class="py-lineno">787</tt>  <tt class="py-line">        <tt class="py-comment"># src/objtools/align_format/align_format_util.cpp#L668</tt> </tt>
<a name="L788"></a><tt class="py-lineno">788</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">field</tt> <tt class="py-op">==</tt> <tt class="py-string">'evalue'</tt><tt class="py-op">:</tt> </tt>
<a name="L789"></a><tt class="py-lineno">789</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-328" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-328', 'value', 'link-81');">value</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-number">1.0e-180</tt><tt class="py-op">:</tt> </tt>
<a name="L790"></a><tt class="py-lineno">790</tt>  <tt class="py-line">                <tt id="link-329" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-329', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'0.0'</tt> </tt>
<a name="L791"></a><tt class="py-lineno">791</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-330" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-330', 'value', 'link-81');">value</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-number">1.0e-99</tt><tt class="py-op">:</tt> </tt>
<a name="L792"></a><tt class="py-lineno">792</tt>  <tt class="py-line">                <tt id="link-331" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-331', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'%2.0e'</tt> <tt class="py-op">%</tt> <tt id="link-332" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-332', 'value', 'link-81');">value</a></tt> </tt>
<a name="L793"></a><tt class="py-lineno">793</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-333" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-333', 'value', 'link-81');">value</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0.0009</tt><tt class="py-op">:</tt> </tt>
<a name="L794"></a><tt class="py-lineno">794</tt>  <tt class="py-line">                <tt id="link-334" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-334', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'%3.0e'</tt> <tt class="py-op">%</tt> <tt id="link-335" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-335', 'value', 'link-81');">value</a></tt> </tt>
<a name="L795"></a><tt class="py-lineno">795</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-336" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-336', 'value', 'link-81');">value</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0.1</tt><tt class="py-op">:</tt> </tt>
<a name="L796"></a><tt class="py-lineno">796</tt>  <tt class="py-line">                <tt id="link-337" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-337', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'%4.3f'</tt> <tt class="py-op">%</tt> <tt id="link-338" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-338', 'value', 'link-81');">value</a></tt> </tt>
<a name="L797"></a><tt class="py-lineno">797</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-339" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-339', 'value', 'link-81');">value</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-number">1.0</tt><tt class="py-op">:</tt> </tt>
<a name="L798"></a><tt class="py-lineno">798</tt>  <tt class="py-line">                <tt id="link-340" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-340', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'%3.2f'</tt> <tt class="py-op">%</tt> <tt id="link-341" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-341', 'value', 'link-81');">value</a></tt> </tt>
<a name="L799"></a><tt class="py-lineno">799</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-342" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-342', 'value', 'link-81');">value</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-number">10.0</tt><tt class="py-op">:</tt> </tt>
<a name="L800"></a><tt class="py-lineno">800</tt>  <tt class="py-line">                <tt id="link-343" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-343', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'%2.1f'</tt> <tt class="py-op">%</tt> <tt id="link-344" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-344', 'value', 'link-81');">value</a></tt> </tt>
<a name="L801"></a><tt class="py-lineno">801</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L802"></a><tt class="py-lineno">802</tt>  <tt class="py-line">                <tt id="link-345" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-345', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'%5.0f'</tt> <tt class="py-op">%</tt> <tt id="link-346" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-346', 'value', 'link-81');">value</a></tt> </tt>
<a name="L803"></a><tt class="py-lineno">803</tt>  <tt class="py-line"> </tt>
<a name="L804"></a><tt class="py-lineno">804</tt>  <tt class="py-line">        <tt class="py-comment"># pident and ppos formatting</tt> </tt>
<a name="L805"></a><tt class="py-lineno">805</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">field</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'pident'</tt><tt class="py-op">,</tt> <tt class="py-string">'ppos'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L806"></a><tt class="py-lineno">806</tt>  <tt class="py-line">            <tt id="link-347" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-347', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'%.2f'</tt> <tt class="py-op">%</tt> <tt id="link-348" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-348', 'value', 'link-81');">value</a></tt> </tt>
<a name="L807"></a><tt class="py-lineno">807</tt>  <tt class="py-line"> </tt>
<a name="L808"></a><tt class="py-lineno">808</tt>  <tt class="py-line">        <tt class="py-comment"># evalue formatting, adapted from BLAST+ source:</tt> </tt>
<a name="L809"></a><tt class="py-lineno">809</tt>  <tt class="py-line">        <tt class="py-comment"># src/objtools/align_format/align_format_util.cpp#L723</tt> </tt>
<a name="L810"></a><tt class="py-lineno">810</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">field</tt> <tt class="py-op">==</tt> <tt class="py-string">'bitscore'</tt><tt class="py-op">:</tt> </tt>
<a name="L811"></a><tt class="py-lineno">811</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-349" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-349', 'value', 'link-81');">value</a></tt> <tt class="py-op">&gt;</tt> <tt class="py-number">9999</tt><tt class="py-op">:</tt> </tt>
<a name="L812"></a><tt class="py-lineno">812</tt>  <tt class="py-line">                <tt id="link-350" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-350', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'%4.3e'</tt> <tt class="py-op">%</tt> <tt id="link-351" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-351', 'value', 'link-81');">value</a></tt> </tt>
<a name="L813"></a><tt class="py-lineno">813</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-352" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-352', 'value', 'link-81');">value</a></tt> <tt class="py-op">&gt;</tt> <tt class="py-number">99.9</tt><tt class="py-op">:</tt> </tt>
<a name="L814"></a><tt class="py-lineno">814</tt>  <tt class="py-line">                <tt id="link-353" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-353', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'%4.0d'</tt> <tt class="py-op">%</tt> <tt id="link-354" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-354', 'value', 'link-81');">value</a></tt> </tt>
<a name="L815"></a><tt class="py-lineno">815</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L816"></a><tt class="py-lineno">816</tt>  <tt class="py-line">                <tt id="link-355" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-355', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'%4.1f'</tt> <tt class="py-op">%</tt> <tt id="link-356" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-356', 'value', 'link-81');">value</a></tt> </tt>
<a name="L817"></a><tt class="py-lineno">817</tt>  <tt class="py-line"> </tt>
<a name="L818"></a><tt class="py-lineno">818</tt>  <tt class="py-line">        <tt class="py-comment"># coverages have no comma (using floats still ~ a more proper</tt> </tt>
<a name="L819"></a><tt class="py-lineno">819</tt>  <tt class="py-line">        <tt class="py-comment"># representation)</tt> </tt>
<a name="L820"></a><tt class="py-lineno">820</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">field</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'qcovhsp'</tt><tt class="py-op">,</tt> <tt class="py-string">'qcovs'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L821"></a><tt class="py-lineno">821</tt>  <tt class="py-line">            <tt id="link-357" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-357', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'%.0f'</tt> <tt class="py-op">%</tt> <tt id="link-358" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-358', 'value', 'link-81');">value</a></tt> </tt>
<a name="L822"></a><tt class="py-lineno">822</tt>  <tt class="py-line"> </tt>
<a name="L823"></a><tt class="py-lineno">823</tt>  <tt class="py-line">        <tt class="py-comment"># list into '&lt;&gt;'-delimited string</tt> </tt>
<a name="L824"></a><tt class="py-lineno">824</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">field</tt> <tt class="py-op">==</tt> <tt class="py-string">'salltitles'</tt><tt class="py-op">:</tt> </tt>
<a name="L825"></a><tt class="py-lineno">825</tt>  <tt class="py-line">            <tt id="link-359" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-359', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'&lt;&gt;'</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt id="link-360" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-360', 'value', 'link-81');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L826"></a><tt class="py-lineno">826</tt>  <tt class="py-line"> </tt>
<a name="L827"></a><tt class="py-lineno">827</tt>  <tt class="py-line">        <tt class="py-comment"># list into ';'-delimited string</tt> </tt>
<a name="L828"></a><tt class="py-lineno">828</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">field</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'sallseqid'</tt><tt class="py-op">,</tt> <tt class="py-string">'sallacc'</tt><tt class="py-op">,</tt> <tt class="py-string">'staxids'</tt><tt class="py-op">,</tt> <tt class="py-string">'sscinames'</tt><tt class="py-op">,</tt> </tt>
<a name="L829"></a><tt class="py-lineno">829</tt>  <tt class="py-line">                <tt class="py-string">'scomnames'</tt><tt class="py-op">,</tt> <tt class="py-string">'sblastnames'</tt><tt class="py-op">,</tt> <tt class="py-string">'sskingdoms'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L830"></a><tt class="py-lineno">830</tt>  <tt class="py-line">            <tt id="link-361" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-361', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-string">';'</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt id="link-362" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-362', 'value', 'link-81');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L831"></a><tt class="py-lineno">831</tt>  <tt class="py-line"> </tt>
<a name="L832"></a><tt class="py-lineno">832</tt>  <tt class="py-line">        <tt class="py-comment"># everything else</tt> </tt>
<a name="L833"></a><tt class="py-lineno">833</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L834"></a><tt class="py-lineno">834</tt>  <tt class="py-line">            <tt id="link-363" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-363', 'value', 'link-81');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-364" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-364', 'value', 'link-81');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L835"></a><tt class="py-lineno">835</tt>  <tt class="py-line"> </tt>
<a name="L836"></a><tt class="py-lineno">836</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-365" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-365', 'value', 'link-81');">value</a></tt> </tt>
</div><a name="L837"></a><tt class="py-lineno">837</tt>  <tt class="py-line"> </tt>
<a name="BlastTabWriter._build_comments"></a><div id="BlastTabWriter._build_comments-def"><a name="L838"></a><tt class="py-lineno">838</tt> <a class="py-toggle" href="#" id="BlastTabWriter._build_comments-toggle" onclick="return toggle('BlastTabWriter._build_comments');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter-class.html#_build_comments">_build_comments</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">qres</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastTabWriter._build_comments-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlastTabWriter._build_comments-expanded"><a name="L839"></a><tt class="py-lineno">839</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return QueryResult tabular comment as a string (PRIVATE)."""</tt> </tt>
<a name="L840"></a><tt class="py-lineno">840</tt>  <tt class="py-line">        <tt class="py-name">comments</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L841"></a><tt class="py-lineno">841</tt>  <tt class="py-line">        <tt class="py-comment"># inverse mapping of the long-short name map, required</tt> </tt>
<a name="L842"></a><tt class="py-lineno">842</tt>  <tt class="py-line">        <tt class="py-comment"># for writing comments</tt> </tt>
<a name="L843"></a><tt class="py-lineno">843</tt>  <tt class="py-line">        <tt class="py-name">inv_field_map</tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">v</tt><tt class="py-op">,</tt> <tt class="py-name">k</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt> <tt class="py-keyword">in</tt> <tt id="link-366" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_tab._LONG_SHORT_MAP" class="py-name" href="#" onclick="return doclink('link-366', '_LONG_SHORT_MAP', 'link-17');">_LONG_SHORT_MAP</a></tt><tt class="py-op">.</tt><tt id="link-367" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-367', 'items', 'link-120');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L844"></a><tt class="py-lineno">844</tt>  <tt class="py-line"> </tt>
<a name="L845"></a><tt class="py-lineno">845</tt>  <tt class="py-line">        <tt class="py-comment"># try to anticipate qress without version</tt> </tt>
<a name="L846"></a><tt class="py-lineno">846</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">qres</tt><tt class="py-op">,</tt> <tt class="py-string">'version'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L847"></a><tt class="py-lineno">847</tt>  <tt class="py-line">            <tt class="py-name">program_line</tt> <tt class="py-op">=</tt> <tt class="py-string">'# %s'</tt> <tt class="py-op">%</tt> <tt class="py-name">qres</tt><tt class="py-op">.</tt><tt class="py-name">program</tt><tt class="py-op">.</tt><tt id="link-368" class="py-name" targets="Method Bio.Seq.Seq.upper()=Bio.Seq.Seq-class.html#upper,Method Bio.Seq.UnknownSeq.upper()=Bio.Seq.UnknownSeq-class.html#upper,Method Bio.SeqRecord.SeqRecord.upper()=Bio.SeqRecord.SeqRecord-class.html#upper"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-368', 'upper', 'link-368');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L848"></a><tt class="py-lineno">848</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L849"></a><tt class="py-lineno">849</tt>  <tt class="py-line">            <tt class="py-name">program_line</tt> <tt class="py-op">=</tt> <tt class="py-string">'# %s %s'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">qres</tt><tt class="py-op">.</tt><tt class="py-name">program</tt><tt class="py-op">.</tt><tt id="link-369" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-369', 'upper', 'link-368');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">qres</tt><tt class="py-op">.</tt><tt id="link-370" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HeaderConsumer.version()=Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#version,Function Bio.Cluster._cluster.version()=Bio.Cluster._cluster-module.html#version,Method Bio.GenBank._FeatureConsumer.version()=Bio.GenBank._FeatureConsumer-class.html#version,Method Bio.GenBank._RecordConsumer.version()=Bio.GenBank._RecordConsumer-class.html#version,Variable Bio.motifs.jaspar.Motif.version=Bio.motifs.jaspar.Motif-class.html#version"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster._cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-370', 'version', 'link-370');">version</a></tt><tt class="py-op">)</tt> </tt>
<a name="L850"></a><tt class="py-lineno">850</tt>  <tt class="py-line">        <tt class="py-name">comments</tt><tt class="py-op">.</tt><tt id="link-371" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-371', 'append', 'link-180');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">program_line</tt><tt class="py-op">)</tt> </tt>
<a name="L851"></a><tt class="py-lineno">851</tt>  <tt class="py-line">        <tt class="py-comment"># description may or may not be None</tt> </tt>
<a name="L852"></a><tt class="py-lineno">852</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">qres</tt><tt class="py-op">.</tt><tt id="link-372" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DescriptionConsumer.description()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#description,Method Bio.Phylo.PhyloXMLIO.Writer.description()=Bio.Phylo.PhyloXMLIO.Writer-class.html#description,Variable Bio.SearchIO._model.hit.Hit.description=Bio.SearchIO._model.hit.Hit-class.html#description,Variable Bio.SearchIO._model.query.QueryResult.description=Bio.SearchIO._model.query.QueryResult-class.html#description"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-372', 'description', 'link-372');">description</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L853"></a><tt class="py-lineno">853</tt>  <tt class="py-line">            <tt class="py-name">comments</tt><tt class="py-op">.</tt><tt id="link-373" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-373', 'append', 'link-180');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">'# Query: %s'</tt> <tt class="py-op">%</tt> <tt class="py-name">qres</tt><tt class="py-op">.</tt><tt id="link-374" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Variable Bio.PDB.Entity.Entity.id=Bio.PDB.Entity.Entity-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-374', 'id', 'link-374');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L854"></a><tt class="py-lineno">854</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L855"></a><tt class="py-lineno">855</tt>  <tt class="py-line">            <tt class="py-name">comments</tt><tt class="py-op">.</tt><tt id="link-375" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-375', 'append', 'link-180');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">'# Query: %s %s'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">qres</tt><tt class="py-op">.</tt><tt id="link-376" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-376', 'id', 'link-374');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">qres</tt><tt class="py-op">.</tt><tt id="link-377" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-377', 'description', 'link-372');">description</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L856"></a><tt class="py-lineno">856</tt>  <tt class="py-line">        <tt class="py-comment"># try appending RID line, if present</tt> </tt>
<a name="L857"></a><tt class="py-lineno">857</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L858"></a><tt class="py-lineno">858</tt>  <tt class="py-line">            <tt class="py-name">comments</tt><tt class="py-op">.</tt><tt id="link-378" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-378', 'append', 'link-180');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">'# RID: %s'</tt> <tt class="py-op">%</tt> <tt class="py-name">qres</tt><tt class="py-op">.</tt><tt class="py-name">rid</tt><tt class="py-op">)</tt> </tt>
<a name="L859"></a><tt class="py-lineno">859</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L860"></a><tt class="py-lineno">860</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
<a name="L861"></a><tt class="py-lineno">861</tt>  <tt class="py-line">        <tt class="py-name">comments</tt><tt class="py-op">.</tt><tt id="link-379" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-379', 'append', 'link-180');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">'# Database: %s'</tt> <tt class="py-op">%</tt> <tt class="py-name">qres</tt><tt class="py-op">.</tt><tt class="py-name">target</tt><tt class="py-op">)</tt> </tt>
<a name="L862"></a><tt class="py-lineno">862</tt>  <tt class="py-line">        <tt class="py-comment"># qresults without hits don't show the Fields comment</tt> </tt>
<a name="L863"></a><tt class="py-lineno">863</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">qres</tt><tt class="py-op">:</tt> </tt>
<a name="L864"></a><tt class="py-lineno">864</tt>  <tt class="py-line">            <tt class="py-name">comments</tt><tt class="py-op">.</tt><tt id="link-380" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-380', 'append', 'link-180');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">'# Fields: %s'</tt> <tt class="py-op">%</tt> </tt>
<a name="L865"></a><tt class="py-lineno">865</tt>  <tt class="py-line">                            <tt class="py-string">', '</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">inv_field_map</tt><tt class="py-op">[</tt><tt class="py-name">field</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt class="py-name">field</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">fields</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L866"></a><tt class="py-lineno">866</tt>  <tt class="py-line">        <tt class="py-name">comments</tt><tt class="py-op">.</tt><tt id="link-381" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-381', 'append', 'link-180');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">'# %i hits found'</tt> <tt class="py-op">%</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">qres</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L867"></a><tt class="py-lineno">867</tt>  <tt class="py-line"> </tt>
<a name="L868"></a><tt class="py-lineno">868</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">'\n'</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">comments</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">'\n'</tt> </tt>
</div></div><a name="L869"></a><tt class="py-lineno">869</tt>  <tt class="py-line"> </tt>
<a name="L870"></a><tt class="py-lineno">870</tt>  <tt class="py-line"> </tt>
<a name="L871"></a><tt class="py-lineno">871</tt>  <tt class="py-line"><tt class="py-comment"># if not used as a module, run the doctest</tt> </tt>
<a name="L872"></a><tt class="py-lineno">872</tt>  <tt class="py-line"><tt class="py-keyword">if</tt> <tt class="py-name">__name__</tt> <tt class="py-op">==</tt> <tt class="py-string">"__main__"</tt><tt class="py-op">:</tt> </tt>
<a name="L873"></a><tt class="py-lineno">873</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-382" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-382', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-383" class="py-name" targets="Module Bio.Phylo._utils=Bio.Phylo._utils-module.html,Module Bio.SearchIO._utils=Bio.SearchIO._utils-module.html,Module Bio._utils=Bio._utils-module.html"><a title="Bio.Phylo._utils
Bio.SearchIO._utils
Bio._utils" class="py-name" href="#" onclick="return doclink('link-383', '_utils', 'link-383');">_utils</a></tt> <tt class="py-keyword">import</tt> <tt id="link-384" class="py-name" targets="Function Bio._utils.run_doctest()=Bio._utils-module.html#run_doctest"><a title="Bio._utils.run_doctest" class="py-name" href="#" onclick="return doclink('link-384', 'run_doctest', 'link-384');">run_doctest</a></tt> </tt>
<a name="L874"></a><tt class="py-lineno">874</tt>  <tt class="py-line">    <tt id="link-385" class="py-name"><a title="Bio._utils.run_doctest" class="py-name" href="#" onclick="return doclink('link-385', 'run_doctest', 'link-384');">run_doctest</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L875"></a><tt class="py-lineno">875</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Fri Jun 22 16:36:26 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
