<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.HMM.MarkovModel</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.HMM-module.html">Package&nbsp;HMM</a> ::
        Module&nbsp;MarkovModel
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.HMM.MarkovModel-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.HMM.MarkovModel-module.html">Module Bio.HMM.MarkovModel</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2001 Brad Chapman.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># This file is part of the Biopython distribution and governed by your</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># choice of the "Biopython License Agreement" or the "BSD 3-Clause License".</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-comment"># Please see the LICENSE file that should have been included as part of this</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-comment"># package.</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-docstring">"""Deal with representations of Markov Models."""</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-comment"># standard modules</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-0" class="py-name" targets="Method Bio.Align.substitution_matrices.Array.copy()=Bio.Align.substitution_matrices.Array-class.html#copy,Method Bio.Crystal.Crystal.copy()=Bio.Crystal.Crystal-class.html#copy,Method Bio.File._IndexedSeqFileDict.copy()=Bio.File._IndexedSeqFileDict-class.html#copy,Method Bio.PDB.Atom.Atom.copy()=Bio.PDB.Atom.Atom-class.html#copy,Method Bio.PDB.Entity.Entity.copy()=Bio.PDB.Entity.Entity-class.html#copy,Method Bio.PDB.vectors.Vector.copy()=Bio.PDB.vectors.Vector-class.html#copy"><a title="Bio.Align.substitution_matrices.Array.copy
Bio.Crystal.Crystal.copy
Bio.File._IndexedSeqFileDict.copy
Bio.PDB.Atom.Atom.copy
Bio.PDB.Entity.Entity.copy
Bio.PDB.vectors.Vector.copy" class="py-name" href="#" onclick="return doclink('link-0', 'copy', 'link-0');">copy</a></tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">math</tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">random</tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-comment"># TODO - Take advantage of defaultdict once Python 2.4 is dead?</tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-comment"># from collections import defaultdict</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-1" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-1', 'Bio', 'link-1');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-2" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-2', '_py3k', 'link-2');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt id="link-3" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-3', 'range', 'link-3');">range</a></tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-4" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-4', 'Bio', 'link-1');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-5" class="py-name" targets="Module Bio.Seq=Bio.Seq-module.html,Class Bio.Seq.Seq=Bio.Seq.Seq-class.html"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-5', 'Seq', 'link-5');">Seq</a></tt> <tt class="py-keyword">import</tt> <tt id="link-6" class="py-name" targets="Class Bio.Seq.MutableSeq=Bio.Seq.MutableSeq-class.html"><a title="Bio.Seq.MutableSeq" class="py-name" href="#" onclick="return doclink('link-6', 'MutableSeq', 'link-6');">MutableSeq</a></tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"> </tt>
<a name="_gen_random_array"></a><div id="_gen_random_array-def"><a name="L22"></a><tt class="py-lineno"> 22</tt> <a class="py-toggle" href="#" id="_gen_random_array-toggle" onclick="return toggle('_gen_random_array');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel-module.html#_gen_random_array">_gen_random_array</a><tt class="py-op">(</tt><tt class="py-param">n</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_gen_random_array-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_gen_random_array-expanded"><a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return an array of n random numbers summing to 1.0 (PRIVATE)."""</tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line">    <tt class="py-name">randArray</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">random</tt><tt class="py-op">.</tt><tt class="py-name">random</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-7" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-7', 'i', 'link-7');">i</a></tt> <tt class="py-keyword">in</tt> <tt id="link-8" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-8', 'range', 'link-3');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line">    <tt class="py-name">total</tt> <tt class="py-op">=</tt> <tt id="link-9" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.sum()=Bio.Nexus.Nexus.StepMatrix-class.html#sum,Method Bio.SubsMat.SeqMat.sum()=Bio.SubsMat.SeqMat-class.html#sum"><a title="Bio.Nexus.Nexus.StepMatrix.sum
Bio.SubsMat.SeqMat.sum" class="py-name" href="#" onclick="return doclink('link-9', 'sum', 'link-9');">sum</a></tt><tt class="py-op">(</tt><tt class="py-name">randArray</tt><tt class="py-op">)</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt id="link-10" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.x=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#x,Variable Bio.Statistics.lowess.x=Bio.Statistics.lowess-module.html#x"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-10', 'x', 'link-10');">x</a></tt> <tt class="py-op">/</tt> <tt class="py-name">total</tt> <tt class="py-keyword">for</tt> <tt id="link-11" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-11', 'x', 'link-10');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">randArray</tt><tt class="py-op">]</tt> </tt>
</div><a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"> </tt>
<a name="_calculate_emissions"></a><div id="_calculate_emissions-def"><a name="L29"></a><tt class="py-lineno"> 29</tt> <a class="py-toggle" href="#" id="_calculate_emissions-toggle" onclick="return toggle('_calculate_emissions');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel-module.html#_calculate_emissions">_calculate_emissions</a><tt class="py-op">(</tt><tt class="py-param">emission_probs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_calculate_emissions-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_calculate_emissions-expanded"><a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line">    <tt class="py-docstring">"""Calculate which symbols can be emitted in each state (PRIVATE)."""</tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line">    <tt class="py-comment"># loop over all of the state-symbol duples, mapping states to</tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line">    <tt class="py-comment"># lists of emitted symbols</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line">    <tt class="py-name">emissions</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">state</tt><tt class="py-op">,</tt> <tt id="link-12" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.symbol()=Bio.Phylo.PhyloXMLIO.Writer-class.html#symbol"><a title="Bio.Phylo.PhyloXMLIO.Writer.symbol" class="py-name" href="#" onclick="return doclink('link-12', 'symbol', 'link-12');">symbol</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">emission_probs</tt><tt class="py-op">:</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line">            <tt class="py-name">emissions</tt><tt class="py-op">[</tt><tt class="py-name">state</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-13" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-13', 'append', 'link-13');">append</a></tt><tt class="py-op">(</tt><tt id="link-14" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.symbol" class="py-name" href="#" onclick="return doclink('link-14', 'symbol', 'link-12');">symbol</a></tt><tt class="py-op">)</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line">            <tt class="py-name">emissions</tt><tt class="py-op">[</tt><tt class="py-name">state</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-15" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.symbol" class="py-name" href="#" onclick="return doclink('link-15', 'symbol', 'link-12');">symbol</a></tt><tt class="py-op">]</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">emissions</tt> </tt>
</div><a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"> </tt>
<a name="_calculate_from_transitions"></a><div id="_calculate_from_transitions-def"><a name="L43"></a><tt class="py-lineno"> 43</tt> <a class="py-toggle" href="#" id="_calculate_from_transitions-toggle" onclick="return toggle('_calculate_from_transitions');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel-module.html#_calculate_from_transitions">_calculate_from_transitions</a><tt class="py-op">(</tt><tt class="py-param">trans_probs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_calculate_from_transitions-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_calculate_from_transitions-expanded"><a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line">    <tt class="py-docstring">"""Calculate which 'from transitions' are allowed for each state (PRIVATE).</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line"><tt class="py-docstring">    This looks through all of the trans_probs, and uses this dictionary</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line"><tt class="py-docstring">    to determine allowed transitions. It converts this information into</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"><tt class="py-docstring">    a dictionary, whose keys are source states and whose values are</tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"><tt class="py-docstring">    lists of destination states reachable from the source state via a</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line"><tt class="py-docstring">    transition.</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line">    <tt class="py-name">transitions</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">from_state</tt><tt class="py-op">,</tt> <tt class="py-name">to_state</tt> <tt class="py-keyword">in</tt> <tt class="py-name">trans_probs</tt><tt class="py-op">:</tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line">            <tt class="py-name">transitions</tt><tt class="py-op">[</tt><tt class="py-name">from_state</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-16" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-16', 'append', 'link-13');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">to_state</tt><tt class="py-op">)</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line">            <tt class="py-name">transitions</tt><tt class="py-op">[</tt><tt class="py-name">from_state</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">to_state</tt><tt class="py-op">]</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line"> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">transitions</tt> </tt>
</div><a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line"> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line"> </tt>
<a name="_calculate_to_transitions"></a><div id="_calculate_to_transitions-def"><a name="L62"></a><tt class="py-lineno"> 62</tt> <a class="py-toggle" href="#" id="_calculate_to_transitions-toggle" onclick="return toggle('_calculate_to_transitions');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel-module.html#_calculate_to_transitions">_calculate_to_transitions</a><tt class="py-op">(</tt><tt class="py-param">trans_probs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_calculate_to_transitions-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_calculate_to_transitions-expanded"><a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line">    <tt class="py-docstring">"""Calculate which 'to transitions' are allowed for each state (PRIVATE).</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line"><tt class="py-docstring">    This looks through all of the trans_probs, and uses this dictionary</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line"><tt class="py-docstring">    to determine allowed transitions. It converts this information into</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line"><tt class="py-docstring">    a dictionary, whose keys are destination states and whose values are</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line"><tt class="py-docstring">    lists of source states from which the destination is reachable via a</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line"><tt class="py-docstring">    transition.</tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line">    <tt class="py-name">transitions</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">from_state</tt><tt class="py-op">,</tt> <tt class="py-name">to_state</tt> <tt class="py-keyword">in</tt> <tt class="py-name">trans_probs</tt><tt class="py-op">:</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">            <tt class="py-name">transitions</tt><tt class="py-op">[</tt><tt class="py-name">to_state</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-17" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-17', 'append', 'link-13');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">from_state</tt><tt class="py-op">)</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line">            <tt class="py-name">transitions</tt><tt class="py-op">[</tt><tt class="py-name">to_state</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">from_state</tt><tt class="py-op">]</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line"> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">transitions</tt> </tt>
</div><a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line"> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line"> </tt>
<a name="MarkovModelBuilder"></a><div id="MarkovModelBuilder-def"><a name="L81"></a><tt class="py-lineno"> 81</tt> <a class="py-toggle" href="#" id="MarkovModelBuilder-toggle" onclick="return toggle('MarkovModelBuilder');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.MarkovModelBuilder-class.html">MarkovModelBuilder</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MarkovModelBuilder-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="MarkovModelBuilder-expanded"><a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line">    <tt class="py-docstring">"""Interface to build up a Markov Model.</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line"><tt class="py-docstring">    This class is designed to try to separate the task of specifying the</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line"><tt class="py-docstring">    Markov Model from the actual model itself. This is in hopes of making</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line"><tt class="py-docstring">    the actual Markov Model classes smaller.</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line"><tt class="py-docstring">    So, this builder class should be used to create Markov models instead</tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line"><tt class="py-docstring">    of trying to initiate a Markov Model directly.</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line"> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line">    <tt class="py-comment"># the default pseudo counts to use</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">    <tt id="link-18" class="py-name" targets="Variable Bio.HMM.MarkovModel.MarkovModelBuilder.DEFAULT_PSEUDO=Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#DEFAULT_PSEUDO"><a title="Bio.HMM.MarkovModel.MarkovModelBuilder.DEFAULT_PSEUDO" class="py-name" href="#" onclick="return doclink('link-18', 'DEFAULT_PSEUDO', 'link-18');">DEFAULT_PSEUDO</a></tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line"> </tt>
<a name="MarkovModelBuilder.__init__"></a><div id="MarkovModelBuilder.__init__-def"><a name="L95"></a><tt class="py-lineno"> 95</tt> <a class="py-toggle" href="#" id="MarkovModelBuilder.__init__-toggle" onclick="return toggle('MarkovModelBuilder.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">state_alphabet</tt><tt class="py-op">,</tt> <tt class="py-param">emission_alphabet</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MarkovModelBuilder.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MarkovModelBuilder.__init__-expanded"><a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize a builder to create Markov Models.</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line"><tt class="py-docstring">         - state_alphabet -- An alphabet containing all of the letters that</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line"><tt class="py-docstring">           can appear in the states</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line"><tt class="py-docstring">         - emission_alphabet -- An alphabet containing all of the letters for</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line"><tt class="py-docstring">           states that can be emitted by the HMM.</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_state_alphabet</tt> <tt class="py-op">=</tt> <tt class="py-name">state_alphabet</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_emission_alphabet</tt> <tt class="py-op">=</tt> <tt class="py-name">emission_alphabet</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line"> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line">        <tt class="py-comment"># probabilities for the initial state, initialized by calling</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">        <tt class="py-comment"># set_initial_probabilities (required)</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">initial_prob</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line"> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">        <tt class="py-comment"># the probabilities for transitions and emissions</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">        <tt class="py-comment"># by default we have no transitions and all possible emissions</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_prob</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">emission_prob</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-19" class="py-name" targets="Method Bio.HMM.MarkovModel.MarkovModelBuilder._all_blank()=Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#_all_blank"><a title="Bio.HMM.MarkovModel.MarkovModelBuilder._all_blank" class="py-name" href="#" onclick="return doclink('link-19', '_all_blank', 'link-19');">_all_blank</a></tt><tt class="py-op">(</tt><tt class="py-name">state_alphabet</tt><tt class="py-op">,</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">                                             <tt class="py-name">emission_alphabet</tt><tt class="py-op">)</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line"> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line">        <tt class="py-comment"># the default pseudocounts for transition and emission counting</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_pseudo</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">emission_pseudo</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-20" class="py-name" targets="Method Bio.HMM.MarkovModel.MarkovModelBuilder._all_pseudo()=Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#_all_pseudo"><a title="Bio.HMM.MarkovModel.MarkovModelBuilder._all_pseudo" class="py-name" href="#" onclick="return doclink('link-20', '_all_pseudo', 'link-20');">_all_pseudo</a></tt><tt class="py-op">(</tt><tt class="py-name">state_alphabet</tt><tt class="py-op">,</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">                                                <tt class="py-name">emission_alphabet</tt><tt class="py-op">)</tt> </tt>
</div><a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line"> </tt>
<a name="MarkovModelBuilder._all_blank"></a><div id="MarkovModelBuilder._all_blank-def"><a name="L123"></a><tt class="py-lineno">123</tt> <a class="py-toggle" href="#" id="MarkovModelBuilder._all_blank-toggle" onclick="return toggle('MarkovModelBuilder._all_blank');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#_all_blank">_all_blank</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">first_alphabet</tt><tt class="py-op">,</tt> <tt class="py-param">second_alphabet</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MarkovModelBuilder._all_blank-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MarkovModelBuilder._all_blank-expanded"><a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a dictionary with all counts set to zero (PRIVATE).</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line"><tt class="py-docstring">        This uses the letters in the first and second alphabet to create</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line"><tt class="py-docstring">        a dictionary with keys of two tuples organized as</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line"><tt class="py-docstring">        (letter of first alphabet, letter of second alphabet). The values</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line"><tt class="py-docstring">        are all set to 0.</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line">        <tt class="py-name">all_blank</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">first_state</tt> <tt class="py-keyword">in</tt> <tt class="py-name">first_alphabet</tt><tt class="py-op">.</tt><tt id="link-21" class="py-name" targets="Variable Bio.Alphabet.Alphabet.letters=Bio.Alphabet.Alphabet-class.html#letters,Variable Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters=Bio.Alphabet.IUPAC.ExtendedIUPACDNA-class.html#letters,Variable Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters=Bio.Alphabet.IUPAC.ExtendedIUPACProtein-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters=Bio.Alphabet.IUPAC.IUPACAmbiguousDNA-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters=Bio.Alphabet.IUPAC.IUPACAmbiguousRNA-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACProtein.letters=Bio.Alphabet.IUPAC.IUPACProtein-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters=Bio.Alphabet.IUPAC.IUPACUnambiguousDNA-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters=Bio.Alphabet.IUPAC.IUPACUnambiguousRNA-class.html#letters,Variable Bio.Alphabet.Reduced.HPModel.letters=Bio.Alphabet.Reduced.HPModel-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy10.letters=Bio.Alphabet.Reduced.Murphy10-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy15.letters=Bio.Alphabet.Reduced.Murphy15-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy4.letters=Bio.Alphabet.Reduced.Murphy4-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy8.letters=Bio.Alphabet.Reduced.Murphy8-class.html#letters,Variable Bio.Alphabet.Reduced.PC5.letters=Bio.Alphabet.Reduced.PC5-class.html#letters,Variable Bio.Alphabet.SecondaryStructure.letters=Bio.Alphabet.SecondaryStructure-class.html#letters,Variable Bio.Alphabet.SingleLetterAlphabet.letters=Bio.Alphabet.SingleLetterAlphabet-class.html#letters,Variable Bio.Alphabet.ThreeLetterProtein.letters=Bio.Alphabet.ThreeLetterProtein-class.html#letters,Variable Bio.codonalign.codonalphabet.CodonAlphabet.letters=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#letters"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-21', 'letters', 'link-21');">letters</a></tt><tt class="py-op">:</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">second_state</tt> <tt class="py-keyword">in</tt> <tt class="py-name">second_alphabet</tt><tt class="py-op">.</tt><tt id="link-22" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-22', 'letters', 'link-21');">letters</a></tt><tt class="py-op">:</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line">                <tt class="py-name">all_blank</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">first_state</tt><tt class="py-op">,</tt> <tt class="py-name">second_state</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line"> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">all_blank</tt> </tt>
</div><a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line"> </tt>
<a name="MarkovModelBuilder._all_pseudo"></a><div id="MarkovModelBuilder._all_pseudo-def"><a name="L138"></a><tt class="py-lineno">138</tt> <a class="py-toggle" href="#" id="MarkovModelBuilder._all_pseudo-toggle" onclick="return toggle('MarkovModelBuilder._all_pseudo');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#_all_pseudo">_all_pseudo</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">first_alphabet</tt><tt class="py-op">,</tt> <tt class="py-param">second_alphabet</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MarkovModelBuilder._all_pseudo-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MarkovModelBuilder._all_pseudo-expanded"><a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a dictionary with all counts set to a default value (PRIVATE).</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line"><tt class="py-docstring">        This takes the letters in first alphabet and second alphabet and</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line"><tt class="py-docstring">        creates a dictionary with keys of two tuples organized as:</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line"><tt class="py-docstring">        (letter of first alphabet, letter of second alphabet). The values</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line"><tt class="py-docstring">        are all set to the value of the class attribute DEFAULT_PSEUDO.</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">        <tt class="py-name">all_counts</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">first_state</tt> <tt class="py-keyword">in</tt> <tt class="py-name">first_alphabet</tt><tt class="py-op">.</tt><tt id="link-23" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-23', 'letters', 'link-21');">letters</a></tt><tt class="py-op">:</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">second_state</tt> <tt class="py-keyword">in</tt> <tt class="py-name">second_alphabet</tt><tt class="py-op">.</tt><tt id="link-24" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-24', 'letters', 'link-21');">letters</a></tt><tt class="py-op">:</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line">                <tt class="py-name">all_counts</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">first_state</tt><tt class="py-op">,</tt> <tt class="py-name">second_state</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-25" class="py-name"><a title="Bio.HMM.MarkovModel.MarkovModelBuilder.DEFAULT_PSEUDO" class="py-name" href="#" onclick="return doclink('link-25', 'DEFAULT_PSEUDO', 'link-18');">DEFAULT_PSEUDO</a></tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line"> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">all_counts</tt> </tt>
</div><a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line"> </tt>
<a name="MarkovModelBuilder.get_markov_model"></a><div id="MarkovModelBuilder.get_markov_model-def"><a name="L153"></a><tt class="py-lineno">153</tt> <a class="py-toggle" href="#" id="MarkovModelBuilder.get_markov_model-toggle" onclick="return toggle('MarkovModelBuilder.get_markov_model');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#get_markov_model">get_markov_model</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MarkovModelBuilder.get_markov_model-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MarkovModelBuilder.get_markov_model-expanded"><a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the markov model corresponding with the current parameters.</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line"><tt class="py-docstring">        Each markov model returned by a call to this function is unique</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line"><tt class="py-docstring">        (ie. they don't influence each other).</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">        <tt class="py-comment"># user must set initial probabilities</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">initial_prob</tt><tt class="py-op">:</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">Exception</tt><tt class="py-op">(</tt><tt class="py-string">"set_initial_probabilities must be called to "</tt> <tt class="py-op">+</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">                            <tt class="py-string">"fully initialize the Markov model"</tt><tt class="py-op">)</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line"> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">        <tt class="py-name">initial_prob</tt> <tt class="py-op">=</tt> <tt id="link-26" class="py-name"><a title="Bio.Align.substitution_matrices.Array.copy
Bio.Crystal.Crystal.copy
Bio.File._IndexedSeqFileDict.copy
Bio.PDB.Atom.Atom.copy
Bio.PDB.Entity.Entity.copy
Bio.PDB.vectors.Vector.copy" class="py-name" href="#" onclick="return doclink('link-26', 'copy', 'link-0');">copy</a></tt><tt class="py-op">.</tt><tt class="py-name">deepcopy</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">initial_prob</tt><tt class="py-op">)</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">        <tt class="py-name">transition_prob</tt> <tt class="py-op">=</tt> <tt id="link-27" class="py-name"><a title="Bio.Align.substitution_matrices.Array.copy
Bio.Crystal.Crystal.copy
Bio.File._IndexedSeqFileDict.copy
Bio.PDB.Atom.Atom.copy
Bio.PDB.Entity.Entity.copy
Bio.PDB.vectors.Vector.copy" class="py-name" href="#" onclick="return doclink('link-27', 'copy', 'link-0');">copy</a></tt><tt class="py-op">.</tt><tt class="py-name">deepcopy</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_prob</tt><tt class="py-op">)</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">        <tt class="py-name">emission_prob</tt> <tt class="py-op">=</tt> <tt id="link-28" class="py-name"><a title="Bio.Align.substitution_matrices.Array.copy
Bio.Crystal.Crystal.copy
Bio.File._IndexedSeqFileDict.copy
Bio.PDB.Atom.Atom.copy
Bio.PDB.Entity.Entity.copy
Bio.PDB.vectors.Vector.copy" class="py-name" href="#" onclick="return doclink('link-28', 'copy', 'link-0');">copy</a></tt><tt class="py-op">.</tt><tt class="py-name">deepcopy</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">emission_prob</tt><tt class="py-op">)</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">        <tt class="py-name">transition_pseudo</tt> <tt class="py-op">=</tt> <tt id="link-29" class="py-name"><a title="Bio.Align.substitution_matrices.Array.copy
Bio.Crystal.Crystal.copy
Bio.File._IndexedSeqFileDict.copy
Bio.PDB.Atom.Atom.copy
Bio.PDB.Entity.Entity.copy
Bio.PDB.vectors.Vector.copy" class="py-name" href="#" onclick="return doclink('link-29', 'copy', 'link-0');">copy</a></tt><tt class="py-op">.</tt><tt class="py-name">deepcopy</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_pseudo</tt><tt class="py-op">)</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">        <tt class="py-name">emission_pseudo</tt> <tt class="py-op">=</tt> <tt id="link-30" class="py-name"><a title="Bio.Align.substitution_matrices.Array.copy
Bio.Crystal.Crystal.copy
Bio.File._IndexedSeqFileDict.copy
Bio.PDB.Atom.Atom.copy
Bio.PDB.Entity.Entity.copy
Bio.PDB.vectors.Vector.copy" class="py-name" href="#" onclick="return doclink('link-30', 'copy', 'link-0');">copy</a></tt><tt class="py-op">.</tt><tt class="py-name">deepcopy</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">emission_pseudo</tt><tt class="py-op">)</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line"> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-31" class="py-name" targets="Class Bio.HMM.MarkovModel.HiddenMarkovModel=Bio.HMM.MarkovModel.HiddenMarkovModel-class.html"><a title="Bio.HMM.MarkovModel.HiddenMarkovModel" class="py-name" href="#" onclick="return doclink('link-31', 'HiddenMarkovModel', 'link-31');">HiddenMarkovModel</a></tt><tt class="py-op">(</tt><tt class="py-name">initial_prob</tt><tt class="py-op">,</tt> <tt class="py-name">transition_prob</tt><tt class="py-op">,</tt> <tt class="py-name">emission_prob</tt><tt class="py-op">,</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line">                                 <tt class="py-name">transition_pseudo</tt><tt class="py-op">,</tt> <tt class="py-name">emission_pseudo</tt><tt class="py-op">)</tt> </tt>
</div><a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line"> </tt>
<a name="MarkovModelBuilder.set_initial_probabilities"></a><div id="MarkovModelBuilder.set_initial_probabilities-def"><a name="L173"></a><tt class="py-lineno">173</tt> <a class="py-toggle" href="#" id="MarkovModelBuilder.set_initial_probabilities-toggle" onclick="return toggle('MarkovModelBuilder.set_initial_probabilities');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#set_initial_probabilities">set_initial_probabilities</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">initial_prob</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MarkovModelBuilder.set_initial_probabilities-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MarkovModelBuilder.set_initial_probabilities-expanded"><a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line">        <tt class="py-docstring">"""Set initial state probabilities.</tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line"><tt class="py-docstring">        initial_prob is a dictionary mapping states to probabilities.</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line"><tt class="py-docstring">        Suppose, for example, that the state alphabet is ['A', 'B']. Call</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line"><tt class="py-docstring">        set_initial_prob({'A': 1}) to guarantee that the initial</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line"><tt class="py-docstring">        state will be 'A'. Call set_initial_prob({'A': 0.5, 'B': 0.5})</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line"><tt class="py-docstring">        to make each initial state equally probable.</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line"><tt class="py-docstring">        This method must now be called in order to use the Markov model</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line"><tt class="py-docstring">        because the calculation of initial probabilities has changed</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line"><tt class="py-docstring">        incompatibly; the previous calculation was incorrect.</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line"><tt class="py-docstring">        If initial probabilities are set for all states, then they should add up</tt> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line"><tt class="py-docstring">        to 1. Otherwise the sum should be &lt;= 1. The residual probability is</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line"><tt class="py-docstring">        divided up evenly between all the states for which the initial</tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line"><tt class="py-docstring">        probability has not been set. For example, calling</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line"><tt class="py-docstring">        set_initial_prob({}) results in P('A') = 0.5 and P('B') = 0.5,</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line"><tt class="py-docstring">        for the above example.</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">initial_prob</tt> <tt class="py-op">=</tt> <tt id="link-32" class="py-name"><a title="Bio.Align.substitution_matrices.Array.copy
Bio.Crystal.Crystal.copy
Bio.File._IndexedSeqFileDict.copy
Bio.PDB.Atom.Atom.copy
Bio.PDB.Entity.Entity.copy
Bio.PDB.vectors.Vector.copy" class="py-name" href="#" onclick="return doclink('link-32', 'copy', 'link-0');">copy</a></tt><tt class="py-op">.</tt><tt id="link-33" class="py-name"><a title="Bio.Align.substitution_matrices.Array.copy
Bio.Crystal.Crystal.copy
Bio.File._IndexedSeqFileDict.copy
Bio.PDB.Atom.Atom.copy
Bio.PDB.Entity.Entity.copy
Bio.PDB.vectors.Vector.copy" class="py-name" href="#" onclick="return doclink('link-33', 'copy', 'link-0');">copy</a></tt><tt class="py-op">(</tt><tt class="py-name">initial_prob</tt><tt class="py-op">)</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line"> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">        <tt class="py-comment"># ensure that all referenced states are valid</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">state</tt> <tt class="py-keyword">in</tt> <tt class="py-name">initial_prob</tt><tt class="py-op">:</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">state</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_state_alphabet</tt><tt class="py-op">.</tt><tt id="link-34" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-34', 'letters', 'link-21');">letters</a></tt><tt class="py-op">:</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"State %s was not found in the sequence "</tt> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line">                                 <tt class="py-string">"alphabet"</tt> <tt class="py-op">%</tt> <tt class="py-name">state</tt><tt class="py-op">)</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line"> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line">        <tt class="py-comment"># distribute the residual probability, if any</tt> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line">        <tt class="py-name">num_states_not_set</tt> <tt class="py-op">=</tt>\ </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line">            <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_state_alphabet</tt><tt class="py-op">.</tt><tt id="link-35" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-35', 'letters', 'link-21');">letters</a></tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">initial_prob</tt><tt class="py-op">)</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">num_states_not_set</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">Exception</tt><tt class="py-op">(</tt><tt class="py-string">"Initial probabilities can't exceed # of states"</tt><tt class="py-op">)</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line">        <tt class="py-name">prob_sum</tt> <tt class="py-op">=</tt> <tt id="link-36" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.sum
Bio.SubsMat.SeqMat.sum" class="py-name" href="#" onclick="return doclink('link-36', 'sum', 'link-9');">sum</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">initial_prob</tt><tt class="py-op">.</tt><tt id="link-37" class="py-name" targets="Method Bio.Align.substitution_matrices.Array.values()=Bio.Align.substitution_matrices.Array-class.html#values,Method Bio.Crystal.Crystal.values()=Bio.Crystal.Crystal-class.html#values,Method Bio.File._IndexedSeqFileDict.values()=Bio.File._IndexedSeqFileDict-class.html#values,Method Bio.Phylo.PhyloXML.Events.values()=Bio.Phylo.PhyloXML.Events-class.html#values,Method BioSQL.BioSeqDatabase.BioSeqDatabase.values()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#values,Method BioSQL.BioSeqDatabase.DBServer.values()=BioSQL.BioSeqDatabase.DBServer-class.html#values"><a title="Bio.Align.substitution_matrices.Array.values
Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-37', 'values', 'link-37');">values</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">prob_sum</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1.0</tt><tt class="py-op">:</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">Exception</tt><tt class="py-op">(</tt><tt class="py-string">"Total initial probability cannot exceed 1.0"</tt><tt class="py-op">)</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">num_states_not_set</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line">            <tt class="py-name">prob</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-number">1.0</tt> <tt class="py-op">-</tt> <tt class="py-name">prob_sum</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-name">num_states_not_set</tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">state</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_state_alphabet</tt><tt class="py-op">.</tt><tt id="link-38" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-38', 'letters', 'link-21');">letters</a></tt><tt class="py-op">:</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">state</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">initial_prob</tt><tt class="py-op">:</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">initial_prob</tt><tt class="py-op">[</tt><tt class="py-name">state</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">prob</tt> </tt>
</div><a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line"> </tt>
<a name="MarkovModelBuilder.set_equal_probabilities"></a><div id="MarkovModelBuilder.set_equal_probabilities-def"><a name="L215"></a><tt class="py-lineno">215</tt> <a class="py-toggle" href="#" id="MarkovModelBuilder.set_equal_probabilities-toggle" onclick="return toggle('MarkovModelBuilder.set_equal_probabilities');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#set_equal_probabilities">set_equal_probabilities</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MarkovModelBuilder.set_equal_probabilities-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MarkovModelBuilder.set_equal_probabilities-expanded"><a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line">        <tt class="py-docstring">"""Reset all probabilities to be an average value.</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line"><tt class="py-docstring">        Resets the values of all initial probabilities and all allowed</tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line"><tt class="py-docstring">        transitions and all allowed emissions to be equal to 1 divided by the</tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line"><tt class="py-docstring">        number of possible elements.</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line"><tt class="py-docstring">        This is useful if you just want to initialize a Markov Model to</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line"><tt class="py-docstring">        starting values (ie. if you have no prior notions of what the</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line"><tt class="py-docstring">        probabilities should be -- or if you are just feeling too lazy</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line"><tt class="py-docstring">        to calculate them :-).</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line"><tt class="py-docstring">        Warning 1 -- this will reset all currently set probabilities.</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line"><tt class="py-docstring">        Warning 2 -- This just sets all probabilities for transitions and</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line"><tt class="py-docstring">        emissions to total up to 1, so it doesn't ensure that the sum of</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line"><tt class="py-docstring">        each set of transitions adds up to 1.</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">        <tt class="py-comment"># set initial state probabilities</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line">        <tt class="py-name">new_initial_prob</tt> <tt class="py-op">=</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_prob</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">state</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_state_alphabet</tt><tt class="py-op">.</tt><tt id="link-39" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-39', 'letters', 'link-21');">letters</a></tt><tt class="py-op">:</tt> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">initial_prob</tt><tt class="py-op">[</tt><tt class="py-name">state</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">new_initial_prob</tt> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line"> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line">        <tt class="py-comment"># set the transitions</tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line">        <tt class="py-name">new_trans_prob</tt> <tt class="py-op">=</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_prob</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_prob</tt><tt class="py-op">:</tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_prob</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">new_trans_prob</tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line"> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line">        <tt class="py-comment"># set the emissions</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line">        <tt class="py-name">new_emission_prob</tt> <tt class="py-op">=</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">emission_prob</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">emission_prob</tt><tt class="py-op">:</tt> </tt>
<a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">emission_prob</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">new_emission_prob</tt> </tt>
</div><a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line"> </tt>
<a name="MarkovModelBuilder.set_random_initial_probabilities"></a><div id="MarkovModelBuilder.set_random_initial_probabilities-def"><a name="L248"></a><tt class="py-lineno">248</tt> <a class="py-toggle" href="#" id="MarkovModelBuilder.set_random_initial_probabilities-toggle" onclick="return toggle('MarkovModelBuilder.set_random_initial_probabilities');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#set_random_initial_probabilities">set_random_initial_probabilities</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MarkovModelBuilder.set_random_initial_probabilities-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MarkovModelBuilder.set_random_initial_probabilities-expanded"><a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line">        <tt class="py-docstring">"""Set all initial state probabilities to a randomly generated distribution.</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line"><tt class="py-docstring">        Returns the dictionary containing the initial probabilities.</tt> </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line">        <tt class="py-name">initial_freqs</tt> <tt class="py-op">=</tt> <tt id="link-40" class="py-name" targets="Function Bio.HMM.MarkovModel._gen_random_array()=Bio.HMM.MarkovModel-module.html#_gen_random_array"><a title="Bio.HMM.MarkovModel._gen_random_array" class="py-name" href="#" onclick="return doclink('link-40', '_gen_random_array', 'link-40');">_gen_random_array</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_state_alphabet</tt><tt class="py-op">.</tt><tt id="link-41" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-41', 'letters', 'link-21');">letters</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">state</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_state_alphabet</tt><tt class="py-op">.</tt><tt id="link-42" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-42', 'letters', 'link-21');">letters</a></tt><tt class="py-op">:</tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">initial_prob</tt><tt class="py-op">[</tt><tt class="py-name">state</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">initial_freqs</tt><tt class="py-op">.</tt><tt id="link-43" class="py-name" targets="Method Bio.File._IndexedSeqFileDict.pop()=Bio.File._IndexedSeqFileDict-class.html#pop,Method Bio.SearchIO._model.hit.Hit.pop()=Bio.SearchIO._model.hit.Hit-class.html#pop,Method Bio.SearchIO._model.query.QueryResult.pop()=Bio.SearchIO._model.query.QueryResult-class.html#pop,Method Bio.Seq.MutableSeq.pop()=Bio.Seq.MutableSeq-class.html#pop"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop" class="py-name" href="#" onclick="return doclink('link-43', 'pop', 'link-43');">pop</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line"> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">initial_prob</tt> </tt>
</div><a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line"> </tt>
<a name="MarkovModelBuilder.set_random_transition_probabilities"></a><div id="MarkovModelBuilder.set_random_transition_probabilities-def"><a name="L259"></a><tt class="py-lineno">259</tt> <a class="py-toggle" href="#" id="MarkovModelBuilder.set_random_transition_probabilities-toggle" onclick="return toggle('MarkovModelBuilder.set_random_transition_probabilities');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#set_random_transition_probabilities">set_random_transition_probabilities</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MarkovModelBuilder.set_random_transition_probabilities-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MarkovModelBuilder.set_random_transition_probabilities-expanded"><a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line">        <tt class="py-docstring">"""Set all allowed transition probabilities to a randomly generated distribution.</tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line"><tt class="py-docstring">        Returns the dictionary containing the transition probabilities.</tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_prob</tt><tt class="py-op">:</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">Exception</tt><tt class="py-op">(</tt><tt class="py-string">"No transitions have been allowed yet. "</tt> <tt class="py-op">+</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">                            <tt class="py-string">"Allow some or all transitions by calling "</tt> <tt class="py-op">+</tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">                            <tt class="py-string">"allow_transition or allow_all_transitions first."</tt><tt class="py-op">)</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line"> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">        <tt id="link-44" class="py-name" targets="Method Bio.HMM.MarkovModel.HiddenMarkovModel.transitions_from()=Bio.HMM.MarkovModel.HiddenMarkovModel-class.html#transitions_from"><a title="Bio.HMM.MarkovModel.HiddenMarkovModel.transitions_from" class="py-name" href="#" onclick="return doclink('link-44', 'transitions_from', 'link-44');">transitions_from</a></tt> <tt class="py-op">=</tt> <tt id="link-45" class="py-name" targets="Function Bio.HMM.MarkovModel._calculate_from_transitions()=Bio.HMM.MarkovModel-module.html#_calculate_from_transitions"><a title="Bio.HMM.MarkovModel._calculate_from_transitions" class="py-name" href="#" onclick="return doclink('link-45', '_calculate_from_transitions', 'link-45');">_calculate_from_transitions</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_prob</tt><tt class="py-op">)</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">from_state</tt> <tt class="py-keyword">in</tt> <tt id="link-46" class="py-name"><a title="Bio.HMM.MarkovModel.HiddenMarkovModel.transitions_from" class="py-name" href="#" onclick="return doclink('link-46', 'transitions_from', 'link-44');">transitions_from</a></tt><tt class="py-op">:</tt> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line">            <tt class="py-name">freqs</tt> <tt class="py-op">=</tt> <tt id="link-47" class="py-name"><a title="Bio.HMM.MarkovModel._gen_random_array" class="py-name" href="#" onclick="return doclink('link-47', '_gen_random_array', 'link-40');">_gen_random_array</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-48" class="py-name"><a title="Bio.HMM.MarkovModel.HiddenMarkovModel.transitions_from" class="py-name" href="#" onclick="return doclink('link-48', 'transitions_from', 'link-44');">transitions_from</a></tt><tt class="py-op">[</tt><tt class="py-name">from_state</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">to_state</tt> <tt class="py-keyword">in</tt> <tt id="link-49" class="py-name"><a title="Bio.HMM.MarkovModel.HiddenMarkovModel.transitions_from" class="py-name" href="#" onclick="return doclink('link-49', 'transitions_from', 'link-44');">transitions_from</a></tt><tt class="py-op">[</tt><tt class="py-name">from_state</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_prob</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">from_state</tt><tt class="py-op">,</tt> <tt class="py-name">to_state</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">freqs</tt><tt class="py-op">.</tt><tt id="link-50" class="py-name"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop" class="py-name" href="#" onclick="return doclink('link-50', 'pop', 'link-43');">pop</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line"> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_prob</tt> </tt>
</div><a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line"> </tt>
<a name="MarkovModelBuilder.set_random_emission_probabilities"></a><div id="MarkovModelBuilder.set_random_emission_probabilities-def"><a name="L277"></a><tt class="py-lineno">277</tt> <a class="py-toggle" href="#" id="MarkovModelBuilder.set_random_emission_probabilities-toggle" onclick="return toggle('MarkovModelBuilder.set_random_emission_probabilities');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#set_random_emission_probabilities">set_random_emission_probabilities</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MarkovModelBuilder.set_random_emission_probabilities-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MarkovModelBuilder.set_random_emission_probabilities-expanded"><a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line">        <tt class="py-docstring">"""Set all allowed emission probabilities to a randomly generated distribution.</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line"><tt class="py-docstring">        Returns the dictionary containing the emission probabilities.</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">emission_prob</tt><tt class="py-op">:</tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">Exception</tt><tt class="py-op">(</tt><tt class="py-string">"No emissions have been allowed yet. "</tt> <tt class="py-op">+</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">                            <tt class="py-string">"Allow some or all emissions."</tt><tt class="py-op">)</tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line"> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line">        <tt class="py-name">emissions</tt> <tt class="py-op">=</tt> <tt id="link-51" class="py-name" targets="Function Bio.HMM.MarkovModel._calculate_emissions()=Bio.HMM.MarkovModel-module.html#_calculate_emissions"><a title="Bio.HMM.MarkovModel._calculate_emissions" class="py-name" href="#" onclick="return doclink('link-51', '_calculate_emissions', 'link-51');">_calculate_emissions</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">emission_prob</tt><tt class="py-op">)</tt> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">state</tt> <tt class="py-keyword">in</tt> <tt class="py-name">emissions</tt><tt class="py-op">:</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">            <tt class="py-name">freqs</tt> <tt class="py-op">=</tt> <tt id="link-52" class="py-name"><a title="Bio.HMM.MarkovModel._gen_random_array" class="py-name" href="#" onclick="return doclink('link-52', '_gen_random_array', 'link-40');">_gen_random_array</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">emissions</tt><tt class="py-op">[</tt><tt class="py-name">state</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-53" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.symbol" class="py-name" href="#" onclick="return doclink('link-53', 'symbol', 'link-12');">symbol</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">emissions</tt><tt class="py-op">[</tt><tt class="py-name">state</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">emission_prob</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">state</tt><tt class="py-op">,</tt> <tt id="link-54" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.symbol" class="py-name" href="#" onclick="return doclink('link-54', 'symbol', 'link-12');">symbol</a></tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">freqs</tt><tt class="py-op">.</tt><tt id="link-55" class="py-name"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop" class="py-name" href="#" onclick="return doclink('link-55', 'pop', 'link-43');">pop</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line"> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">emission_prob</tt> </tt>
</div><a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line"> </tt>
<a name="MarkovModelBuilder.set_random_probabilities"></a><div id="MarkovModelBuilder.set_random_probabilities-def"><a name="L294"></a><tt class="py-lineno">294</tt> <a class="py-toggle" href="#" id="MarkovModelBuilder.set_random_probabilities-toggle" onclick="return toggle('MarkovModelBuilder.set_random_probabilities');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#set_random_probabilities">set_random_probabilities</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MarkovModelBuilder.set_random_probabilities-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MarkovModelBuilder.set_random_probabilities-expanded"><a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line">        <tt class="py-docstring">"""Set all probabilities to randomly generated numbers.</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line"><tt class="py-docstring">        Resets probabilities of all initial states, transitions, and</tt> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line"><tt class="py-docstring">        emissions to random values.</tt> </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-56" class="py-name" targets="Method Bio.HMM.MarkovModel.MarkovModelBuilder.set_random_initial_probabilities()=Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#set_random_initial_probabilities"><a title="Bio.HMM.MarkovModel.MarkovModelBuilder.set_random_initial_probabilities" class="py-name" href="#" onclick="return doclink('link-56', 'set_random_initial_probabilities', 'link-56');">set_random_initial_probabilities</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-57" class="py-name" targets="Method Bio.HMM.MarkovModel.MarkovModelBuilder.set_random_transition_probabilities()=Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#set_random_transition_probabilities"><a title="Bio.HMM.MarkovModel.MarkovModelBuilder.set_random_transition_probabilities" class="py-name" href="#" onclick="return doclink('link-57', 'set_random_transition_probabilities', 'link-57');">set_random_transition_probabilities</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-58" class="py-name" targets="Method Bio.HMM.MarkovModel.MarkovModelBuilder.set_random_emission_probabilities()=Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#set_random_emission_probabilities"><a title="Bio.HMM.MarkovModel.MarkovModelBuilder.set_random_emission_probabilities" class="py-name" href="#" onclick="return doclink('link-58', 'set_random_emission_probabilities', 'link-58');">set_random_emission_probabilities</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line"> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line">    <tt class="py-comment"># --- functions to deal with the transitions in the sequence</tt> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line"> </tt>
<a name="MarkovModelBuilder.allow_all_transitions"></a><div id="MarkovModelBuilder.allow_all_transitions-def"><a name="L306"></a><tt class="py-lineno">306</tt> <a class="py-toggle" href="#" id="MarkovModelBuilder.allow_all_transitions-toggle" onclick="return toggle('MarkovModelBuilder.allow_all_transitions');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#allow_all_transitions">allow_all_transitions</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MarkovModelBuilder.allow_all_transitions-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MarkovModelBuilder.allow_all_transitions-expanded"><a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create transitions between all states.</tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line"><tt class="py-docstring">        By default all transitions within the alphabet are disallowed;</tt> </tt>
<a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line"><tt class="py-docstring">        this is a convenience function to change this to allow all</tt> </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line"><tt class="py-docstring">        possible transitions.</tt> </tt>
<a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line">        <tt class="py-comment"># first get all probabilities and pseudo counts set</tt> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line">        <tt class="py-comment"># to the default values</tt> </tt>
<a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line">        <tt class="py-name">all_probs</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-59" class="py-name"><a title="Bio.HMM.MarkovModel.MarkovModelBuilder._all_blank" class="py-name" href="#" onclick="return doclink('link-59', '_all_blank', 'link-19');">_all_blank</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_state_alphabet</tt><tt class="py-op">,</tt> </tt>
<a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line">                                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_state_alphabet</tt><tt class="py-op">)</tt> </tt>
<a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line"> </tt>
<a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">        <tt class="py-name">all_pseudo</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-60" class="py-name"><a title="Bio.HMM.MarkovModel.MarkovModelBuilder._all_pseudo" class="py-name" href="#" onclick="return doclink('link-60', '_all_pseudo', 'link-20');">_all_pseudo</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_state_alphabet</tt><tt class="py-op">,</tt> </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line">                                      <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_state_alphabet</tt><tt class="py-op">)</tt> </tt>
<a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line"> </tt>
<a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line">        <tt class="py-comment"># now set any probabilities and pseudo counts that</tt> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">        <tt class="py-comment"># were previously set</tt> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">set_key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_prob</tt><tt class="py-op">:</tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line">            <tt class="py-name">all_probs</tt><tt class="py-op">[</tt><tt class="py-name">set_key</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_prob</tt><tt class="py-op">[</tt><tt class="py-name">set_key</tt><tt class="py-op">]</tt> </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line"> </tt>
<a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">set_key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_pseudo</tt><tt class="py-op">:</tt> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line">            <tt class="py-name">all_pseudo</tt><tt class="py-op">[</tt><tt class="py-name">set_key</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_pseudo</tt><tt class="py-op">[</tt><tt class="py-name">set_key</tt><tt class="py-op">]</tt> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line"> </tt>
<a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line">        <tt class="py-comment"># finally reinitialize the transition probs and pseudo counts</tt> </tt>
<a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_prob</tt> <tt class="py-op">=</tt> <tt class="py-name">all_probs</tt> </tt>
<a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_pseudo</tt> <tt class="py-op">=</tt> <tt class="py-name">all_pseudo</tt> </tt>
</div><a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line"> </tt>
<a name="MarkovModelBuilder.allow_transition"></a><div id="MarkovModelBuilder.allow_transition-def"><a name="L333"></a><tt class="py-lineno">333</tt> <a class="py-toggle" href="#" id="MarkovModelBuilder.allow_transition-toggle" onclick="return toggle('MarkovModelBuilder.allow_transition');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#allow_transition">allow_transition</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">from_state</tt><tt class="py-op">,</tt> <tt class="py-param">to_state</tt><tt class="py-op">,</tt> <tt class="py-param">probability</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line">                         <tt class="py-param">pseudocount</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MarkovModelBuilder.allow_transition-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MarkovModelBuilder.allow_transition-expanded"><a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line">        <tt class="py-docstring">"""Set a transition as being possible between the two states.</tt> </tt>
<a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line"><tt class="py-docstring">        probability and pseudocount are optional arguments</tt> </tt>
<a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line"><tt class="py-docstring">        specifying the probabilities and pseudo counts for the transition.</tt> </tt>
<a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line"><tt class="py-docstring">        If these are not supplied, then the values are set to the</tt> </tt>
<a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line"><tt class="py-docstring">        default values.</tt> </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line"><tt class="py-docstring">        Raises:</tt> </tt>
<a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line"><tt class="py-docstring">        KeyError -- if the two states already have an allowed transition.</tt> </tt>
<a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line">        <tt class="py-comment"># check the sanity of adding these states</tt> </tt>
<a name="L347"></a><tt class="py-lineno">347</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">state</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-name">from_state</tt><tt class="py-op">,</tt> <tt class="py-name">to_state</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">state</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_state_alphabet</tt><tt class="py-op">.</tt><tt id="link-61" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-61', 'letters', 'link-21');">letters</a></tt><tt class="py-op">:</tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"State %s was not found in the sequence "</tt> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line">                                 <tt class="py-string">"alphabet"</tt> <tt class="py-op">%</tt> <tt class="py-name">state</tt><tt class="py-op">)</tt> </tt>
<a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line"> </tt>
<a name="L352"></a><tt class="py-lineno">352</tt>  <tt class="py-line">        <tt class="py-comment"># ensure that the states are not already set</tt> </tt>
<a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-op">(</tt><tt class="py-name">from_state</tt><tt class="py-op">,</tt> <tt class="py-name">to_state</tt><tt class="py-op">)</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_prob</tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line">           <tt class="py-op">(</tt><tt class="py-name">from_state</tt><tt class="py-op">,</tt> <tt class="py-name">to_state</tt><tt class="py-op">)</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_pseudo</tt><tt class="py-op">:</tt> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">            <tt class="py-comment"># set the initial probability</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">probability</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line">                <tt class="py-name">probability</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L358"></a><tt class="py-lineno">358</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_prob</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">from_state</tt><tt class="py-op">,</tt> <tt class="py-name">to_state</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">probability</tt> </tt>
<a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line"> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line">            <tt class="py-comment"># set the initial pseudocounts</tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">pseudocount</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line">                <tt class="py-name">pseudocount</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-62" class="py-name"><a title="Bio.HMM.MarkovModel.MarkovModelBuilder.DEFAULT_PSEUDO" class="py-name" href="#" onclick="return doclink('link-62', 'DEFAULT_PSEUDO', 'link-18');">DEFAULT_PSEUDO</a></tt> </tt>
<a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_pseudo</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">from_state</tt><tt class="py-op">,</tt> <tt class="py-name">to_state</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">pseudocount</tt> </tt>
<a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">KeyError</tt><tt class="py-op">(</tt><tt class="py-string">"Transition from %s to %s is already allowed."</tt> </tt>
<a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line">                           <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">from_state</tt><tt class="py-op">,</tt> <tt class="py-name">to_state</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line"> </tt>
<a name="MarkovModelBuilder.destroy_transition"></a><div id="MarkovModelBuilder.destroy_transition-def"><a name="L368"></a><tt class="py-lineno">368</tt> <a class="py-toggle" href="#" id="MarkovModelBuilder.destroy_transition-toggle" onclick="return toggle('MarkovModelBuilder.destroy_transition');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#destroy_transition">destroy_transition</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">from_state</tt><tt class="py-op">,</tt> <tt class="py-param">to_state</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MarkovModelBuilder.destroy_transition-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MarkovModelBuilder.destroy_transition-expanded"><a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line">        <tt class="py-docstring">"""Restrict transitions between the two states.</tt> </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line"><tt class="py-docstring">        Raises:</tt> </tt>
<a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line"><tt class="py-docstring">        KeyError if the transition is not currently allowed.</tt> </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line">            <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_prob</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">from_state</tt><tt class="py-op">,</tt> <tt class="py-name">to_state</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line">            <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_pseudo</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">from_state</tt><tt class="py-op">,</tt> <tt class="py-name">to_state</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L378"></a><tt class="py-lineno">378</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L379"></a><tt class="py-lineno">379</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">KeyError</tt><tt class="py-op">(</tt><tt class="py-string">"Transition from %s to %s is already disallowed."</tt> </tt>
<a name="L380"></a><tt class="py-lineno">380</tt>  <tt class="py-line">                           <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">from_state</tt><tt class="py-op">,</tt> <tt class="py-name">to_state</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L381"></a><tt class="py-lineno">381</tt>  <tt class="py-line"> </tt>
<a name="MarkovModelBuilder.set_transition_score"></a><div id="MarkovModelBuilder.set_transition_score-def"><a name="L382"></a><tt class="py-lineno">382</tt> <a class="py-toggle" href="#" id="MarkovModelBuilder.set_transition_score-toggle" onclick="return toggle('MarkovModelBuilder.set_transition_score');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#set_transition_score">set_transition_score</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">from_state</tt><tt class="py-op">,</tt> <tt class="py-param">to_state</tt><tt class="py-op">,</tt> <tt class="py-param">probability</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MarkovModelBuilder.set_transition_score-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MarkovModelBuilder.set_transition_score-expanded"><a name="L383"></a><tt class="py-lineno">383</tt>  <tt class="py-line">        <tt class="py-docstring">"""Set the probability of a transition between two states.</tt> </tt>
<a name="L384"></a><tt class="py-lineno">384</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L385"></a><tt class="py-lineno">385</tt>  <tt class="py-line"><tt class="py-docstring">        Raises:</tt> </tt>
<a name="L386"></a><tt class="py-lineno">386</tt>  <tt class="py-line"><tt class="py-docstring">        KeyError if the transition is not allowed.</tt> </tt>
<a name="L387"></a><tt class="py-lineno">387</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L388"></a><tt class="py-lineno">388</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L389"></a><tt class="py-lineno">389</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-op">(</tt><tt class="py-name">from_state</tt><tt class="py-op">,</tt> <tt class="py-name">to_state</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_prob</tt><tt class="py-op">:</tt> </tt>
<a name="L390"></a><tt class="py-lineno">390</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_prob</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">from_state</tt><tt class="py-op">,</tt> <tt class="py-name">to_state</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">probability</tt> </tt>
<a name="L391"></a><tt class="py-lineno">391</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L392"></a><tt class="py-lineno">392</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">KeyError</tt><tt class="py-op">(</tt><tt class="py-string">"Transition from %s to %s is not allowed."</tt> </tt>
<a name="L393"></a><tt class="py-lineno">393</tt>  <tt class="py-line">                           <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">from_state</tt><tt class="py-op">,</tt> <tt class="py-name">to_state</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L394"></a><tt class="py-lineno">394</tt>  <tt class="py-line"> </tt>
<a name="MarkovModelBuilder.set_transition_pseudocount"></a><div id="MarkovModelBuilder.set_transition_pseudocount-def"><a name="L395"></a><tt class="py-lineno">395</tt> <a class="py-toggle" href="#" id="MarkovModelBuilder.set_transition_pseudocount-toggle" onclick="return toggle('MarkovModelBuilder.set_transition_pseudocount');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#set_transition_pseudocount">set_transition_pseudocount</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">from_state</tt><tt class="py-op">,</tt> <tt class="py-param">to_state</tt><tt class="py-op">,</tt> <tt class="py-param">count</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MarkovModelBuilder.set_transition_pseudocount-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MarkovModelBuilder.set_transition_pseudocount-expanded"><a name="L396"></a><tt class="py-lineno">396</tt>  <tt class="py-line">        <tt class="py-docstring">"""Set the default pseudocount for a transition.</tt> </tt>
<a name="L397"></a><tt class="py-lineno">397</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L398"></a><tt class="py-lineno">398</tt>  <tt class="py-line"><tt class="py-docstring">        To avoid computational problems, it is helpful to be able to</tt> </tt>
<a name="L399"></a><tt class="py-lineno">399</tt>  <tt class="py-line"><tt class="py-docstring">        set a 'default' pseudocount to start with for estimating</tt> </tt>
<a name="L400"></a><tt class="py-lineno">400</tt>  <tt class="py-line"><tt class="py-docstring">        transition and emission probabilities (see p62 in Durbin et al</tt> </tt>
<a name="L401"></a><tt class="py-lineno">401</tt>  <tt class="py-line"><tt class="py-docstring">        for more discussion on this. By default, all transitions have</tt> </tt>
<a name="L402"></a><tt class="py-lineno">402</tt>  <tt class="py-line"><tt class="py-docstring">        a pseudocount of 1.</tt> </tt>
<a name="L403"></a><tt class="py-lineno">403</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L404"></a><tt class="py-lineno">404</tt>  <tt class="py-line"><tt class="py-docstring">        Raises:</tt> </tt>
<a name="L405"></a><tt class="py-lineno">405</tt>  <tt class="py-line"><tt class="py-docstring">        KeyError if the transition is not allowed.</tt> </tt>
<a name="L406"></a><tt class="py-lineno">406</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L407"></a><tt class="py-lineno">407</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L408"></a><tt class="py-lineno">408</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-op">(</tt><tt class="py-name">from_state</tt><tt class="py-op">,</tt> <tt class="py-name">to_state</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_pseudo</tt><tt class="py-op">:</tt> </tt>
<a name="L409"></a><tt class="py-lineno">409</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_pseudo</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">from_state</tt><tt class="py-op">,</tt> <tt class="py-name">to_state</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-63" class="py-name" targets="Method Bio.Crystal.Chain.count()=Bio.Crystal.Chain-class.html#count,Method Bio.Seq.MutableSeq.count()=Bio.Seq.MutableSeq-class.html#count,Method Bio.Seq.Seq.count()=Bio.Seq.Seq-class.html#count,Method Bio.Seq.UnknownSeq.count()=Bio.Seq.UnknownSeq-class.html#count,Method Bio.motifs.Instances.count()=Bio.motifs.Instances-class.html#count"><a title="Bio.Crystal.Chain.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-63', 'count', 'link-63');">count</a></tt> </tt>
<a name="L410"></a><tt class="py-lineno">410</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L411"></a><tt class="py-lineno">411</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">KeyError</tt><tt class="py-op">(</tt><tt class="py-string">"Transition from %s to %s is not allowed."</tt> </tt>
<a name="L412"></a><tt class="py-lineno">412</tt>  <tt class="py-line">                           <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">from_state</tt><tt class="py-op">,</tt> <tt class="py-name">to_state</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L413"></a><tt class="py-lineno">413</tt>  <tt class="py-line"> </tt>
<a name="L414"></a><tt class="py-lineno">414</tt>  <tt class="py-line">    <tt class="py-comment"># --- functions to deal with emissions from the sequence</tt> </tt>
<a name="L415"></a><tt class="py-lineno">415</tt>  <tt class="py-line"> </tt>
<a name="MarkovModelBuilder.set_emission_score"></a><div id="MarkovModelBuilder.set_emission_score-def"><a name="L416"></a><tt class="py-lineno">416</tt> <a class="py-toggle" href="#" id="MarkovModelBuilder.set_emission_score-toggle" onclick="return toggle('MarkovModelBuilder.set_emission_score');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#set_emission_score">set_emission_score</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">seq_state</tt><tt class="py-op">,</tt> <tt class="py-param">emission_state</tt><tt class="py-op">,</tt> <tt class="py-param">probability</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MarkovModelBuilder.set_emission_score-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MarkovModelBuilder.set_emission_score-expanded"><a name="L417"></a><tt class="py-lineno">417</tt>  <tt class="py-line">        <tt class="py-docstring">"""Set the probability of a emission from a particular state.</tt> </tt>
<a name="L418"></a><tt class="py-lineno">418</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L419"></a><tt class="py-lineno">419</tt>  <tt class="py-line"><tt class="py-docstring">        Raises:</tt> </tt>
<a name="L420"></a><tt class="py-lineno">420</tt>  <tt class="py-line"><tt class="py-docstring">        KeyError if the emission from the given state is not allowed.</tt> </tt>
<a name="L421"></a><tt class="py-lineno">421</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L422"></a><tt class="py-lineno">422</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L423"></a><tt class="py-lineno">423</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-op">(</tt><tt class="py-name">seq_state</tt><tt class="py-op">,</tt> <tt class="py-name">emission_state</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">emission_prob</tt><tt class="py-op">:</tt> </tt>
<a name="L424"></a><tt class="py-lineno">424</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">emission_prob</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">seq_state</tt><tt class="py-op">,</tt> <tt class="py-name">emission_state</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">probability</tt> </tt>
<a name="L425"></a><tt class="py-lineno">425</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L426"></a><tt class="py-lineno">426</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">KeyError</tt><tt class="py-op">(</tt><tt class="py-string">"Emission of %s from %s is not allowed."</tt> </tt>
<a name="L427"></a><tt class="py-lineno">427</tt>  <tt class="py-line">                           <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">emission_state</tt><tt class="py-op">,</tt> <tt class="py-name">seq_state</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L428"></a><tt class="py-lineno">428</tt>  <tt class="py-line"> </tt>
<a name="MarkovModelBuilder.set_emission_pseudocount"></a><div id="MarkovModelBuilder.set_emission_pseudocount-def"><a name="L429"></a><tt class="py-lineno">429</tt> <a class="py-toggle" href="#" id="MarkovModelBuilder.set_emission_pseudocount-toggle" onclick="return toggle('MarkovModelBuilder.set_emission_pseudocount');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#set_emission_pseudocount">set_emission_pseudocount</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">seq_state</tt><tt class="py-op">,</tt> <tt class="py-param">emission_state</tt><tt class="py-op">,</tt> <tt class="py-param">count</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MarkovModelBuilder.set_emission_pseudocount-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MarkovModelBuilder.set_emission_pseudocount-expanded"><a name="L430"></a><tt class="py-lineno">430</tt>  <tt class="py-line">        <tt class="py-docstring">"""Set the default pseudocount for an emission.</tt> </tt>
<a name="L431"></a><tt class="py-lineno">431</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L432"></a><tt class="py-lineno">432</tt>  <tt class="py-line"><tt class="py-docstring">        To avoid computational problems, it is helpful to be able to</tt> </tt>
<a name="L433"></a><tt class="py-lineno">433</tt>  <tt class="py-line"><tt class="py-docstring">        set a 'default' pseudocount to start with for estimating</tt> </tt>
<a name="L434"></a><tt class="py-lineno">434</tt>  <tt class="py-line"><tt class="py-docstring">        transition and emission probabilities (see p62 in Durbin et al</tt> </tt>
<a name="L435"></a><tt class="py-lineno">435</tt>  <tt class="py-line"><tt class="py-docstring">        for more discussion on this. By default, all emissions have</tt> </tt>
<a name="L436"></a><tt class="py-lineno">436</tt>  <tt class="py-line"><tt class="py-docstring">        a pseudocount of 1.</tt> </tt>
<a name="L437"></a><tt class="py-lineno">437</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L438"></a><tt class="py-lineno">438</tt>  <tt class="py-line"><tt class="py-docstring">        Raises:</tt> </tt>
<a name="L439"></a><tt class="py-lineno">439</tt>  <tt class="py-line"><tt class="py-docstring">        KeyError if the emission from the given state is not allowed.</tt> </tt>
<a name="L440"></a><tt class="py-lineno">440</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L441"></a><tt class="py-lineno">441</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L442"></a><tt class="py-lineno">442</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-op">(</tt><tt class="py-name">seq_state</tt><tt class="py-op">,</tt> <tt class="py-name">emission_state</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">emission_pseudo</tt><tt class="py-op">:</tt> </tt>
<a name="L443"></a><tt class="py-lineno">443</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">emission_pseudo</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">seq_state</tt><tt class="py-op">,</tt> <tt class="py-name">emission_state</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-64" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-64', 'count', 'link-63');">count</a></tt> </tt>
<a name="L444"></a><tt class="py-lineno">444</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L445"></a><tt class="py-lineno">445</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">KeyError</tt><tt class="py-op">(</tt><tt class="py-string">"Emission of %s from %s is not allowed."</tt> </tt>
<a name="L446"></a><tt class="py-lineno">446</tt>  <tt class="py-line">                           <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">emission_state</tt><tt class="py-op">,</tt> <tt class="py-name">seq_state</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L447"></a><tt class="py-lineno">447</tt>  <tt class="py-line"> </tt>
<a name="L448"></a><tt class="py-lineno">448</tt>  <tt class="py-line"> </tt>
<a name="HiddenMarkovModel"></a><div id="HiddenMarkovModel-def"><a name="L449"></a><tt class="py-lineno">449</tt> <a class="py-toggle" href="#" id="HiddenMarkovModel-toggle" onclick="return toggle('HiddenMarkovModel');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.HiddenMarkovModel-class.html">HiddenMarkovModel</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="HiddenMarkovModel-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="HiddenMarkovModel-expanded"><a name="L450"></a><tt class="py-lineno">450</tt>  <tt class="py-line">    <tt class="py-docstring">"""Represent a hidden markov model that can be used for state estimation."""</tt> </tt>
<a name="L451"></a><tt class="py-lineno">451</tt>  <tt class="py-line"> </tt>
<a name="HiddenMarkovModel.__init__"></a><div id="HiddenMarkovModel.__init__-def"><a name="L452"></a><tt class="py-lineno">452</tt> <a class="py-toggle" href="#" id="HiddenMarkovModel.__init__-toggle" onclick="return toggle('HiddenMarkovModel.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.HiddenMarkovModel-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">initial_prob</tt><tt class="py-op">,</tt> <tt class="py-param">transition_prob</tt><tt class="py-op">,</tt> <tt class="py-param">emission_prob</tt><tt class="py-op">,</tt> </tt>
<a name="L453"></a><tt class="py-lineno">453</tt>  <tt class="py-line">                 <tt class="py-param">transition_pseudo</tt><tt class="py-op">,</tt> <tt class="py-param">emission_pseudo</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="HiddenMarkovModel.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="HiddenMarkovModel.__init__-expanded"><a name="L454"></a><tt class="py-lineno">454</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize a Markov Model.</tt> </tt>
<a name="L455"></a><tt class="py-lineno">455</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L456"></a><tt class="py-lineno">456</tt>  <tt class="py-line"><tt class="py-docstring">        Note: You should use the MarkovModelBuilder class instead of</tt> </tt>
<a name="L457"></a><tt class="py-lineno">457</tt>  <tt class="py-line"><tt class="py-docstring">        initiating this class directly.</tt> </tt>
<a name="L458"></a><tt class="py-lineno">458</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L459"></a><tt class="py-lineno">459</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L460"></a><tt class="py-lineno">460</tt>  <tt class="py-line"><tt class="py-docstring">         - initial_prob - A dictionary of initial probabilities for all states.</tt> </tt>
<a name="L461"></a><tt class="py-lineno">461</tt>  <tt class="py-line"><tt class="py-docstring">         - transition_prob -- A dictionary of transition probabilities for all</tt> </tt>
<a name="L462"></a><tt class="py-lineno">462</tt>  <tt class="py-line"><tt class="py-docstring">           possible transitions in the sequence.</tt> </tt>
<a name="L463"></a><tt class="py-lineno">463</tt>  <tt class="py-line"><tt class="py-docstring">         - emission_prob -- A dictionary of emission probabilities for all</tt> </tt>
<a name="L464"></a><tt class="py-lineno">464</tt>  <tt class="py-line"><tt class="py-docstring">           possible emissions from the sequence states.</tt> </tt>
<a name="L465"></a><tt class="py-lineno">465</tt>  <tt class="py-line"><tt class="py-docstring">         - transition_pseudo -- Pseudo-counts to be used for the transitions,</tt> </tt>
<a name="L466"></a><tt class="py-lineno">466</tt>  <tt class="py-line"><tt class="py-docstring">           when counting for purposes of estimating transition probabilities.</tt> </tt>
<a name="L467"></a><tt class="py-lineno">467</tt>  <tt class="py-line"><tt class="py-docstring">         - emission_pseudo -- Pseudo-counts to be used for the emissions,</tt> </tt>
<a name="L468"></a><tt class="py-lineno">468</tt>  <tt class="py-line"><tt class="py-docstring">           when counting for purposes of estimating emission probabilities.</tt> </tt>
<a name="L469"></a><tt class="py-lineno">469</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L470"></a><tt class="py-lineno">470</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L471"></a><tt class="py-lineno">471</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">initial_prob</tt> <tt class="py-op">=</tt> <tt class="py-name">initial_prob</tt> </tt>
<a name="L472"></a><tt class="py-lineno">472</tt>  <tt class="py-line"> </tt>
<a name="L473"></a><tt class="py-lineno">473</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_transition_pseudo</tt> <tt class="py-op">=</tt> <tt class="py-name">transition_pseudo</tt> </tt>
<a name="L474"></a><tt class="py-lineno">474</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_emission_pseudo</tt> <tt class="py-op">=</tt> <tt class="py-name">emission_pseudo</tt> </tt>
<a name="L475"></a><tt class="py-lineno">475</tt>  <tt class="py-line"> </tt>
<a name="L476"></a><tt class="py-lineno">476</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_prob</tt> <tt class="py-op">=</tt> <tt class="py-name">transition_prob</tt> </tt>
<a name="L477"></a><tt class="py-lineno">477</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">emission_prob</tt> <tt class="py-op">=</tt> <tt class="py-name">emission_prob</tt> </tt>
<a name="L478"></a><tt class="py-lineno">478</tt>  <tt class="py-line"> </tt>
<a name="L479"></a><tt class="py-lineno">479</tt>  <tt class="py-line">        <tt class="py-comment"># a dictionary of the possible transitions from each state</tt> </tt>
<a name="L480"></a><tt class="py-lineno">480</tt>  <tt class="py-line">        <tt class="py-comment"># each key is a source state, mapped to a list of the destination states</tt> </tt>
<a name="L481"></a><tt class="py-lineno">481</tt>  <tt class="py-line">        <tt class="py-comment"># that are reachable from the source state via a transition</tt> </tt>
<a name="L482"></a><tt class="py-lineno">482</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_transitions_from</tt> <tt class="py-op">=</tt> <tt id="link-65" class="py-name"><a title="Bio.HMM.MarkovModel._calculate_from_transitions" class="py-name" href="#" onclick="return doclink('link-65', '_calculate_from_transitions', 'link-45');">_calculate_from_transitions</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_prob</tt><tt class="py-op">)</tt> </tt>
<a name="L483"></a><tt class="py-lineno">483</tt>  <tt class="py-line"> </tt>
<a name="L484"></a><tt class="py-lineno">484</tt>  <tt class="py-line">        <tt class="py-comment"># a dictionary of the possible transitions to each state</tt> </tt>
<a name="L485"></a><tt class="py-lineno">485</tt>  <tt class="py-line">        <tt class="py-comment"># each key is a destination state, mapped to a list of source states</tt> </tt>
<a name="L486"></a><tt class="py-lineno">486</tt>  <tt class="py-line">        <tt class="py-comment"># from which the destination is reachable via a transition</tt> </tt>
<a name="L487"></a><tt class="py-lineno">487</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_transitions_to</tt> <tt class="py-op">=</tt> <tt id="link-66" class="py-name" targets="Function Bio.HMM.MarkovModel._calculate_to_transitions()=Bio.HMM.MarkovModel-module.html#_calculate_to_transitions"><a title="Bio.HMM.MarkovModel._calculate_to_transitions" class="py-name" href="#" onclick="return doclink('link-66', '_calculate_to_transitions', 'link-66');">_calculate_to_transitions</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_prob</tt><tt class="py-op">)</tt> </tt>
</div><a name="L488"></a><tt class="py-lineno">488</tt>  <tt class="py-line"> </tt>
<a name="HiddenMarkovModel.get_blank_transitions"></a><div id="HiddenMarkovModel.get_blank_transitions-def"><a name="L489"></a><tt class="py-lineno">489</tt> <a class="py-toggle" href="#" id="HiddenMarkovModel.get_blank_transitions-toggle" onclick="return toggle('HiddenMarkovModel.get_blank_transitions');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.HiddenMarkovModel-class.html#get_blank_transitions">get_blank_transitions</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="HiddenMarkovModel.get_blank_transitions-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="HiddenMarkovModel.get_blank_transitions-expanded"><a name="L490"></a><tt class="py-lineno">490</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get the default transitions for the model.</tt> </tt>
<a name="L491"></a><tt class="py-lineno">491</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L492"></a><tt class="py-lineno">492</tt>  <tt class="py-line"><tt class="py-docstring">        Returns a dictionary of all of the default transitions between any</tt> </tt>
<a name="L493"></a><tt class="py-lineno">493</tt>  <tt class="py-line"><tt class="py-docstring">        two letters in the sequence alphabet. The dictionary is structured</tt> </tt>
<a name="L494"></a><tt class="py-lineno">494</tt>  <tt class="py-line"><tt class="py-docstring">        with keys as (letter1, letter2) and values as the starting number</tt> </tt>
<a name="L495"></a><tt class="py-lineno">495</tt>  <tt class="py-line"><tt class="py-docstring">        of transitions.</tt> </tt>
<a name="L496"></a><tt class="py-lineno">496</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L497"></a><tt class="py-lineno">497</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_transition_pseudo</tt> </tt>
</div><a name="L498"></a><tt class="py-lineno">498</tt>  <tt class="py-line"> </tt>
<a name="HiddenMarkovModel.get_blank_emissions"></a><div id="HiddenMarkovModel.get_blank_emissions-def"><a name="L499"></a><tt class="py-lineno">499</tt> <a class="py-toggle" href="#" id="HiddenMarkovModel.get_blank_emissions-toggle" onclick="return toggle('HiddenMarkovModel.get_blank_emissions');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.HiddenMarkovModel-class.html#get_blank_emissions">get_blank_emissions</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="HiddenMarkovModel.get_blank_emissions-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="HiddenMarkovModel.get_blank_emissions-expanded"><a name="L500"></a><tt class="py-lineno">500</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get the starting default emmissions for each sequence.</tt> </tt>
<a name="L501"></a><tt class="py-lineno">501</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L502"></a><tt class="py-lineno">502</tt>  <tt class="py-line"><tt class="py-docstring">        This returns a dictionary of the default emmissions for each</tt> </tt>
<a name="L503"></a><tt class="py-lineno">503</tt>  <tt class="py-line"><tt class="py-docstring">        letter. The dictionary is structured with keys as</tt> </tt>
<a name="L504"></a><tt class="py-lineno">504</tt>  <tt class="py-line"><tt class="py-docstring">        (seq_letter, emmission_letter) and values as the starting number</tt> </tt>
<a name="L505"></a><tt class="py-lineno">505</tt>  <tt class="py-line"><tt class="py-docstring">        of emmissions.</tt> </tt>
<a name="L506"></a><tt class="py-lineno">506</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L507"></a><tt class="py-lineno">507</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_emission_pseudo</tt> </tt>
</div><a name="L508"></a><tt class="py-lineno">508</tt>  <tt class="py-line"> </tt>
<a name="HiddenMarkovModel.transitions_from"></a><div id="HiddenMarkovModel.transitions_from-def"><a name="L509"></a><tt class="py-lineno">509</tt> <a class="py-toggle" href="#" id="HiddenMarkovModel.transitions_from-toggle" onclick="return toggle('HiddenMarkovModel.transitions_from');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.HiddenMarkovModel-class.html#transitions_from">transitions_from</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">state_letter</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="HiddenMarkovModel.transitions_from-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="HiddenMarkovModel.transitions_from-expanded"><a name="L510"></a><tt class="py-lineno">510</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get all destination states which can transition from source state_letter.</tt> </tt>
<a name="L511"></a><tt class="py-lineno">511</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L512"></a><tt class="py-lineno">512</tt>  <tt class="py-line"><tt class="py-docstring">        This returns all letters which the given state_letter can transition</tt> </tt>
<a name="L513"></a><tt class="py-lineno">513</tt>  <tt class="py-line"><tt class="py-docstring">        to, i.e. all the destination states reachable from state_letter.</tt> </tt>
<a name="L514"></a><tt class="py-lineno">514</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L515"></a><tt class="py-lineno">515</tt>  <tt class="py-line"><tt class="py-docstring">        An empty list is returned if state_letter has no outgoing transitions.</tt> </tt>
<a name="L516"></a><tt class="py-lineno">516</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L517"></a><tt class="py-lineno">517</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">state_letter</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_transitions_from</tt><tt class="py-op">:</tt> </tt>
<a name="L518"></a><tt class="py-lineno">518</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_transitions_from</tt><tt class="py-op">[</tt><tt class="py-name">state_letter</tt><tt class="py-op">]</tt> </tt>
<a name="L519"></a><tt class="py-lineno">519</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L520"></a><tt class="py-lineno">520</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div><a name="L521"></a><tt class="py-lineno">521</tt>  <tt class="py-line"> </tt>
<a name="HiddenMarkovModel.transitions_to"></a><div id="HiddenMarkovModel.transitions_to-def"><a name="L522"></a><tt class="py-lineno">522</tt> <a class="py-toggle" href="#" id="HiddenMarkovModel.transitions_to-toggle" onclick="return toggle('HiddenMarkovModel.transitions_to');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.HiddenMarkovModel-class.html#transitions_to">transitions_to</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">state_letter</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="HiddenMarkovModel.transitions_to-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="HiddenMarkovModel.transitions_to-expanded"><a name="L523"></a><tt class="py-lineno">523</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get all source states which can transition to destination state_letter.</tt> </tt>
<a name="L524"></a><tt class="py-lineno">524</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L525"></a><tt class="py-lineno">525</tt>  <tt class="py-line"><tt class="py-docstring">        This returns all letters which the given state_letter is reachable</tt> </tt>
<a name="L526"></a><tt class="py-lineno">526</tt>  <tt class="py-line"><tt class="py-docstring">        from, i.e. all the source states which can reach state_later</tt> </tt>
<a name="L527"></a><tt class="py-lineno">527</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L528"></a><tt class="py-lineno">528</tt>  <tt class="py-line"><tt class="py-docstring">        An empty list is returned if state_letter is unreachable.</tt> </tt>
<a name="L529"></a><tt class="py-lineno">529</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L530"></a><tt class="py-lineno">530</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">state_letter</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_transitions_to</tt><tt class="py-op">:</tt> </tt>
<a name="L531"></a><tt class="py-lineno">531</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_transitions_to</tt><tt class="py-op">[</tt><tt class="py-name">state_letter</tt><tt class="py-op">]</tt> </tt>
<a name="L532"></a><tt class="py-lineno">532</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L533"></a><tt class="py-lineno">533</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div><a name="L534"></a><tt class="py-lineno">534</tt>  <tt class="py-line"> </tt>
<a name="HiddenMarkovModel.viterbi"></a><div id="HiddenMarkovModel.viterbi-def"><a name="L535"></a><tt class="py-lineno">535</tt> <a class="py-toggle" href="#" id="HiddenMarkovModel.viterbi-toggle" onclick="return toggle('HiddenMarkovModel.viterbi');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.HiddenMarkovModel-class.html#viterbi">viterbi</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sequence</tt><tt class="py-op">,</tt> <tt class="py-param">state_alphabet</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="HiddenMarkovModel.viterbi-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="HiddenMarkovModel.viterbi-expanded"><a name="L536"></a><tt class="py-lineno">536</tt>  <tt class="py-line">        <tt class="py-docstring">"""Calculate the most probable state path using the Viterbi algorithm.</tt> </tt>
<a name="L537"></a><tt class="py-lineno">537</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L538"></a><tt class="py-lineno">538</tt>  <tt class="py-line"><tt class="py-docstring">        This implements the Viterbi algorithm (see pgs 55-57 in Durbin et</tt> </tt>
<a name="L539"></a><tt class="py-lineno">539</tt>  <tt class="py-line"><tt class="py-docstring">        al for a full explanation -- this is where I took my implementation</tt> </tt>
<a name="L540"></a><tt class="py-lineno">540</tt>  <tt class="py-line"><tt class="py-docstring">        ideas from), to allow decoding of the state path, given a sequence</tt> </tt>
<a name="L541"></a><tt class="py-lineno">541</tt>  <tt class="py-line"><tt class="py-docstring">        of emissions.</tt> </tt>
<a name="L542"></a><tt class="py-lineno">542</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L543"></a><tt class="py-lineno">543</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L544"></a><tt class="py-lineno">544</tt>  <tt class="py-line"><tt class="py-docstring">         - sequence -- A Seq object with the emission sequence that we</tt> </tt>
<a name="L545"></a><tt class="py-lineno">545</tt>  <tt class="py-line"><tt class="py-docstring">           want to decode.</tt> </tt>
<a name="L546"></a><tt class="py-lineno">546</tt>  <tt class="py-line"><tt class="py-docstring">         - state_alphabet -- The alphabet of the possible state sequences</tt> </tt>
<a name="L547"></a><tt class="py-lineno">547</tt>  <tt class="py-line"><tt class="py-docstring">           that can be generated.</tt> </tt>
<a name="L548"></a><tt class="py-lineno">548</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L549"></a><tt class="py-lineno">549</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L550"></a><tt class="py-lineno">550</tt>  <tt class="py-line">        <tt class="py-comment"># calculate logarithms of the initial, transition, and emission probs</tt> </tt>
<a name="L551"></a><tt class="py-lineno">551</tt>  <tt class="py-line">        <tt class="py-name">log_initial</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-67" class="py-name" targets="Method Bio.HMM.MarkovModel.HiddenMarkovModel._log_transform()=Bio.HMM.MarkovModel.HiddenMarkovModel-class.html#_log_transform"><a title="Bio.HMM.MarkovModel.HiddenMarkovModel._log_transform" class="py-name" href="#" onclick="return doclink('link-67', '_log_transform', 'link-67');">_log_transform</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">initial_prob</tt><tt class="py-op">)</tt> </tt>
<a name="L552"></a><tt class="py-lineno">552</tt>  <tt class="py-line">        <tt class="py-name">log_trans</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-68" class="py-name"><a title="Bio.HMM.MarkovModel.HiddenMarkovModel._log_transform" class="py-name" href="#" onclick="return doclink('link-68', '_log_transform', 'link-67');">_log_transform</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">transition_prob</tt><tt class="py-op">)</tt> </tt>
<a name="L553"></a><tt class="py-lineno">553</tt>  <tt class="py-line">        <tt class="py-name">log_emission</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-69" class="py-name"><a title="Bio.HMM.MarkovModel.HiddenMarkovModel._log_transform" class="py-name" href="#" onclick="return doclink('link-69', '_log_transform', 'link-67');">_log_transform</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">emission_prob</tt><tt class="py-op">)</tt> </tt>
<a name="L554"></a><tt class="py-lineno">554</tt>  <tt class="py-line"> </tt>
<a name="L555"></a><tt class="py-lineno">555</tt>  <tt class="py-line">        <tt class="py-name">viterbi_probs</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L556"></a><tt class="py-lineno">556</tt>  <tt class="py-line">        <tt class="py-name">pred_state_seq</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L557"></a><tt class="py-lineno">557</tt>  <tt class="py-line">        <tt class="py-name">state_letters</tt> <tt class="py-op">=</tt> <tt class="py-name">state_alphabet</tt><tt class="py-op">.</tt><tt id="link-70" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-70', 'letters', 'link-21');">letters</a></tt> </tt>
<a name="L558"></a><tt class="py-lineno">558</tt>  <tt class="py-line"> </tt>
<a name="L559"></a><tt class="py-lineno">559</tt>  <tt class="py-line">        <tt class="py-comment"># --- recursion</tt> </tt>
<a name="L560"></a><tt class="py-lineno">560</tt>  <tt class="py-line">        <tt class="py-comment"># loop over the training squence (i = 1 .. L)</tt> </tt>
<a name="L561"></a><tt class="py-lineno">561</tt>  <tt class="py-line">        <tt class="py-comment"># NOTE: My index numbers are one less than what is given in Durbin</tt> </tt>
<a name="L562"></a><tt class="py-lineno">562</tt>  <tt class="py-line">        <tt class="py-comment"># et al, since we are indexing the sequence going from 0 to</tt> </tt>
<a name="L563"></a><tt class="py-lineno">563</tt>  <tt class="py-line">        <tt class="py-comment"># (Length - 1) not 1 to Length, like in Durbin et al.</tt> </tt>
<a name="L564"></a><tt class="py-lineno">564</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-71" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-71', 'i', 'link-7');">i</a></tt> <tt class="py-keyword">in</tt> <tt id="link-72" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-72', 'range', 'link-3');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-73" class="py-name" targets="Method Bio.FSSP.FSSPAlignDict.sequence()=Bio.FSSP.FSSPAlignDict-class.html#sequence,Method Bio.GenBank._FeatureConsumer.sequence()=Bio.GenBank._FeatureConsumer-class.html#sequence,Method Bio.GenBank._RecordConsumer.sequence()=Bio.GenBank._RecordConsumer-class.html#sequence,Method Bio.Phylo.PhyloXMLIO.Writer.sequence()=Bio.Phylo.PhyloXMLIO.Writer-class.html#sequence"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-73', 'sequence', 'link-73');">sequence</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L565"></a><tt class="py-lineno">565</tt>  <tt class="py-line">            <tt class="py-comment"># loop over all of the possible i-th states in the state path</tt> </tt>
<a name="L566"></a><tt class="py-lineno">566</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">cur_state</tt> <tt class="py-keyword">in</tt> <tt class="py-name">state_letters</tt><tt class="py-op">:</tt> </tt>
<a name="L567"></a><tt class="py-lineno">567</tt>  <tt class="py-line">                <tt class="py-comment"># e_{l}(x_{i})</tt> </tt>
<a name="L568"></a><tt class="py-lineno">568</tt>  <tt class="py-line">                <tt class="py-name">emission_part</tt> <tt class="py-op">=</tt> <tt class="py-name">log_emission</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">cur_state</tt><tt class="py-op">,</tt> <tt id="link-74" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-74', 'sequence', 'link-73');">sequence</a></tt><tt class="py-op">[</tt><tt id="link-75" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-75', 'i', 'link-7');">i</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L569"></a><tt class="py-lineno">569</tt>  <tt class="py-line"> </tt>
<a name="L570"></a><tt class="py-lineno">570</tt>  <tt class="py-line">                <tt class="py-name">max_prob</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L571"></a><tt class="py-lineno">571</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-76" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-76', 'i', 'link-7');">i</a></tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L572"></a><tt class="py-lineno">572</tt>  <tt class="py-line">                    <tt class="py-comment"># for the first state, use the initial probability rather</tt> </tt>
<a name="L573"></a><tt class="py-lineno">573</tt>  <tt class="py-line">                    <tt class="py-comment"># than looking back to previous states</tt> </tt>
<a name="L574"></a><tt class="py-lineno">574</tt>  <tt class="py-line">                    <tt class="py-name">max_prob</tt> <tt class="py-op">=</tt> <tt class="py-name">log_initial</tt><tt class="py-op">[</tt><tt class="py-name">cur_state</tt><tt class="py-op">]</tt> </tt>
<a name="L575"></a><tt class="py-lineno">575</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L576"></a><tt class="py-lineno">576</tt>  <tt class="py-line">                    <tt class="py-comment"># loop over all possible (i-1)-th previous states</tt> </tt>
<a name="L577"></a><tt class="py-lineno">577</tt>  <tt class="py-line">                    <tt class="py-name">possible_state_probs</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L578"></a><tt class="py-lineno">578</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-name">prev_state</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-77" class="py-name" targets="Method Bio.HMM.MarkovModel.HiddenMarkovModel.transitions_to()=Bio.HMM.MarkovModel.HiddenMarkovModel-class.html#transitions_to"><a title="Bio.HMM.MarkovModel.HiddenMarkovModel.transitions_to" class="py-name" href="#" onclick="return doclink('link-77', 'transitions_to', 'link-77');">transitions_to</a></tt><tt class="py-op">(</tt><tt class="py-name">cur_state</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L579"></a><tt class="py-lineno">579</tt>  <tt class="py-line">                        <tt class="py-comment"># a_{kl}</tt> </tt>
<a name="L580"></a><tt class="py-lineno">580</tt>  <tt class="py-line">                        <tt class="py-name">trans_part</tt> <tt class="py-op">=</tt> <tt class="py-name">log_trans</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">prev_state</tt><tt class="py-op">,</tt> <tt class="py-name">cur_state</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L581"></a><tt class="py-lineno">581</tt>  <tt class="py-line"> </tt>
<a name="L582"></a><tt class="py-lineno">582</tt>  <tt class="py-line">                        <tt class="py-comment"># v_{k}(i - 1)</tt> </tt>
<a name="L583"></a><tt class="py-lineno">583</tt>  <tt class="py-line">                        <tt class="py-name">viterbi_part</tt> <tt class="py-op">=</tt> <tt class="py-name">viterbi_probs</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">prev_state</tt><tt class="py-op">,</tt> <tt id="link-78" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-78', 'i', 'link-7');">i</a></tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L584"></a><tt class="py-lineno">584</tt>  <tt class="py-line">                        <tt class="py-name">cur_prob</tt> <tt class="py-op">=</tt> <tt class="py-name">viterbi_part</tt> <tt class="py-op">+</tt> <tt class="py-name">trans_part</tt> </tt>
<a name="L585"></a><tt class="py-lineno">585</tt>  <tt class="py-line"> </tt>
<a name="L586"></a><tt class="py-lineno">586</tt>  <tt class="py-line">                        <tt class="py-name">possible_state_probs</tt><tt class="py-op">[</tt><tt class="py-name">prev_state</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">cur_prob</tt> </tt>
<a name="L587"></a><tt class="py-lineno">587</tt>  <tt class="py-line"> </tt>
<a name="L588"></a><tt class="py-lineno">588</tt>  <tt class="py-line">                    <tt class="py-comment"># calculate the viterbi probability using the max</tt> </tt>
<a name="L589"></a><tt class="py-lineno">589</tt>  <tt class="py-line">                    <tt class="py-name">max_prob</tt> <tt class="py-op">=</tt> <tt id="link-79" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.max=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#max"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-79', 'max', 'link-79');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">possible_state_probs</tt><tt class="py-op">.</tt><tt id="link-80" class="py-name"><a title="Bio.Align.substitution_matrices.Array.values
Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-80', 'values', 'link-37');">values</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L590"></a><tt class="py-lineno">590</tt>  <tt class="py-line"> </tt>
<a name="L591"></a><tt class="py-lineno">591</tt>  <tt class="py-line">                <tt class="py-comment"># v_{k}(i)</tt> </tt>
<a name="L592"></a><tt class="py-lineno">592</tt>  <tt class="py-line">                <tt class="py-name">viterbi_probs</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">cur_state</tt><tt class="py-op">,</tt> <tt id="link-81" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-81', 'i', 'link-7');">i</a></tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">emission_part</tt> <tt class="py-op">+</tt> <tt class="py-name">max_prob</tt><tt class="py-op">)</tt> </tt>
<a name="L593"></a><tt class="py-lineno">593</tt>  <tt class="py-line"> </tt>
<a name="L594"></a><tt class="py-lineno">594</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-82" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-82', 'i', 'link-7');">i</a></tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L595"></a><tt class="py-lineno">595</tt>  <tt class="py-line">                    <tt class="py-comment"># get the most likely prev_state leading to cur_state</tt> </tt>
<a name="L596"></a><tt class="py-lineno">596</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-name">state</tt> <tt class="py-keyword">in</tt> <tt class="py-name">possible_state_probs</tt><tt class="py-op">:</tt> </tt>
<a name="L597"></a><tt class="py-lineno">597</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">possible_state_probs</tt><tt class="py-op">[</tt><tt class="py-name">state</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">max_prob</tt><tt class="py-op">:</tt> </tt>
<a name="L598"></a><tt class="py-lineno">598</tt>  <tt class="py-line">                            <tt class="py-name">pred_state_seq</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt id="link-83" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-83', 'i', 'link-7');">i</a></tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">cur_state</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">state</tt> </tt>
<a name="L599"></a><tt class="py-lineno">599</tt>  <tt class="py-line">                            <tt class="py-keyword">break</tt> </tt>
<a name="L600"></a><tt class="py-lineno">600</tt>  <tt class="py-line"> </tt>
<a name="L601"></a><tt class="py-lineno">601</tt>  <tt class="py-line">        <tt class="py-comment"># --- termination</tt> </tt>
<a name="L602"></a><tt class="py-lineno">602</tt>  <tt class="py-line">        <tt class="py-comment"># calculate the probability of the state path</tt> </tt>
<a name="L603"></a><tt class="py-lineno">603</tt>  <tt class="py-line">        <tt class="py-comment"># loop over all states</tt> </tt>
<a name="L604"></a><tt class="py-lineno">604</tt>  <tt class="py-line">        <tt class="py-name">all_probs</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L605"></a><tt class="py-lineno">605</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">state</tt> <tt class="py-keyword">in</tt> <tt class="py-name">state_letters</tt><tt class="py-op">:</tt> </tt>
<a name="L606"></a><tt class="py-lineno">606</tt>  <tt class="py-line">            <tt class="py-comment"># v_{k}(L)</tt> </tt>
<a name="L607"></a><tt class="py-lineno">607</tt>  <tt class="py-line">            <tt class="py-name">all_probs</tt><tt class="py-op">[</tt><tt class="py-name">state</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">viterbi_probs</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">state</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-84" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-84', 'sequence', 'link-73');">sequence</a></tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L608"></a><tt class="py-lineno">608</tt>  <tt class="py-line"> </tt>
<a name="L609"></a><tt class="py-lineno">609</tt>  <tt class="py-line">        <tt class="py-name">state_path_prob</tt> <tt class="py-op">=</tt> <tt id="link-85" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-85', 'max', 'link-79');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">all_probs</tt><tt class="py-op">.</tt><tt id="link-86" class="py-name"><a title="Bio.Align.substitution_matrices.Array.values
Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-86', 'values', 'link-37');">values</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L610"></a><tt class="py-lineno">610</tt>  <tt class="py-line"> </tt>
<a name="L611"></a><tt class="py-lineno">611</tt>  <tt class="py-line">        <tt class="py-comment"># find the last pointer we need to trace back from</tt> </tt>
<a name="L612"></a><tt class="py-lineno">612</tt>  <tt class="py-line">        <tt class="py-name">last_state</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L613"></a><tt class="py-lineno">613</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">state</tt> <tt class="py-keyword">in</tt> <tt class="py-name">all_probs</tt><tt class="py-op">:</tt> </tt>
<a name="L614"></a><tt class="py-lineno">614</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">all_probs</tt><tt class="py-op">[</tt><tt class="py-name">state</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">state_path_prob</tt><tt class="py-op">:</tt> </tt>
<a name="L615"></a><tt class="py-lineno">615</tt>  <tt class="py-line">                <tt class="py-name">last_state</tt> <tt class="py-op">=</tt> <tt class="py-name">state</tt> </tt>
<a name="L616"></a><tt class="py-lineno">616</tt>  <tt class="py-line"> </tt>
<a name="L617"></a><tt class="py-lineno">617</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">last_state</tt> <tt class="py-op">!=</tt> <tt class="py-string">""</tt><tt class="py-op">,</tt> <tt class="py-string">"Didn't find the last state to trace from!"</tt> </tt>
<a name="L618"></a><tt class="py-lineno">618</tt>  <tt class="py-line"> </tt>
<a name="L619"></a><tt class="py-lineno">619</tt>  <tt class="py-line">        <tt class="py-comment"># --- traceback</tt> </tt>
<a name="L620"></a><tt class="py-lineno">620</tt>  <tt class="py-line">        <tt class="py-name">traceback_seq</tt> <tt class="py-op">=</tt> <tt id="link-87" class="py-name"><a title="Bio.Seq.MutableSeq" class="py-name" href="#" onclick="return doclink('link-87', 'MutableSeq', 'link-6');">MutableSeq</a></tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">,</tt> <tt class="py-name">state_alphabet</tt><tt class="py-op">)</tt> </tt>
<a name="L621"></a><tt class="py-lineno">621</tt>  <tt class="py-line"> </tt>
<a name="L622"></a><tt class="py-lineno">622</tt>  <tt class="py-line">        <tt class="py-name">loop_seq</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-88" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-88', 'range', 'link-3');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-89" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-89', 'sequence', 'link-73');">sequence</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L623"></a><tt class="py-lineno">623</tt>  <tt class="py-line">        <tt class="py-name">loop_seq</tt><tt class="py-op">.</tt><tt id="link-90" class="py-name" targets="Method Bio.Pathway.Reaction.reverse()=Bio.Pathway.Reaction-class.html#reverse,Method Bio.Seq.MutableSeq.reverse()=Bio.Seq.MutableSeq-class.html#reverse"><a title="Bio.Pathway.Reaction.reverse
Bio.Seq.MutableSeq.reverse" class="py-name" href="#" onclick="return doclink('link-90', 'reverse', 'link-90');">reverse</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L624"></a><tt class="py-lineno">624</tt>  <tt class="py-line"> </tt>
<a name="L625"></a><tt class="py-lineno">625</tt>  <tt class="py-line">        <tt class="py-comment"># last_state is the last state in the most probable state sequence.</tt> </tt>
<a name="L626"></a><tt class="py-lineno">626</tt>  <tt class="py-line">        <tt class="py-comment"># Compute that sequence by walking backwards in time. From the i-th</tt> </tt>
<a name="L627"></a><tt class="py-lineno">627</tt>  <tt class="py-line">        <tt class="py-comment"># state in the sequence, find the (i-1)-th state as the most</tt> </tt>
<a name="L628"></a><tt class="py-lineno">628</tt>  <tt class="py-line">        <tt class="py-comment"># probable state preceding the i-th state.</tt> </tt>
<a name="L629"></a><tt class="py-lineno">629</tt>  <tt class="py-line">        <tt class="py-name">state</tt> <tt class="py-op">=</tt> <tt class="py-name">last_state</tt> </tt>
<a name="L630"></a><tt class="py-lineno">630</tt>  <tt class="py-line">        <tt class="py-name">traceback_seq</tt><tt class="py-op">.</tt><tt id="link-91" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-91', 'append', 'link-13');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">state</tt><tt class="py-op">)</tt> </tt>
<a name="L631"></a><tt class="py-lineno">631</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-92" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-92', 'i', 'link-7');">i</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">loop_seq</tt><tt class="py-op">:</tt> </tt>
<a name="L632"></a><tt class="py-lineno">632</tt>  <tt class="py-line">            <tt class="py-name">state</tt> <tt class="py-op">=</tt> <tt class="py-name">pred_state_seq</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt id="link-93" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-93', 'i', 'link-7');">i</a></tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">state</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L633"></a><tt class="py-lineno">633</tt>  <tt class="py-line">            <tt class="py-name">traceback_seq</tt><tt class="py-op">.</tt><tt id="link-94" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-94', 'append', 'link-13');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">state</tt><tt class="py-op">)</tt> </tt>
<a name="L634"></a><tt class="py-lineno">634</tt>  <tt class="py-line"> </tt>
<a name="L635"></a><tt class="py-lineno">635</tt>  <tt class="py-line">        <tt class="py-comment"># put the traceback sequence in the proper orientation</tt> </tt>
<a name="L636"></a><tt class="py-lineno">636</tt>  <tt class="py-line">        <tt class="py-name">traceback_seq</tt><tt class="py-op">.</tt><tt id="link-95" class="py-name"><a title="Bio.Pathway.Reaction.reverse
Bio.Seq.MutableSeq.reverse" class="py-name" href="#" onclick="return doclink('link-95', 'reverse', 'link-90');">reverse</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L637"></a><tt class="py-lineno">637</tt>  <tt class="py-line"> </tt>
<a name="L638"></a><tt class="py-lineno">638</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">traceback_seq</tt><tt class="py-op">.</tt><tt id="link-96" class="py-name" targets="Method Bio.Seq.MutableSeq.toseq()=Bio.Seq.MutableSeq-class.html#toseq,Method BioSQL.BioSeq.DBSeq.toseq()=BioSQL.BioSeq.DBSeq-class.html#toseq"><a title="Bio.Seq.MutableSeq.toseq
BioSQL.BioSeq.DBSeq.toseq" class="py-name" href="#" onclick="return doclink('link-96', 'toseq', 'link-96');">toseq</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">state_path_prob</tt> </tt>
</div><a name="L639"></a><tt class="py-lineno">639</tt>  <tt class="py-line"> </tt>
<a name="HiddenMarkovModel._log_transform"></a><div id="HiddenMarkovModel._log_transform-def"><a name="L640"></a><tt class="py-lineno">640</tt> <a class="py-toggle" href="#" id="HiddenMarkovModel._log_transform-toggle" onclick="return toggle('HiddenMarkovModel._log_transform');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.HMM.MarkovModel.HiddenMarkovModel-class.html#_log_transform">_log_transform</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">probability</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="HiddenMarkovModel._log_transform-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="HiddenMarkovModel._log_transform-expanded"><a name="L641"></a><tt class="py-lineno">641</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return log transform of the given probability dictionary (PRIVATE).</tt> </tt>
<a name="L642"></a><tt class="py-lineno">642</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L643"></a><tt class="py-lineno">643</tt>  <tt class="py-line"><tt class="py-docstring">        When calculating the Viterbi equation, add logs of probabilities rather</tt> </tt>
<a name="L644"></a><tt class="py-lineno">644</tt>  <tt class="py-line"><tt class="py-docstring">        than multiplying probabilities, to avoid underflow errors. This method</tt> </tt>
<a name="L645"></a><tt class="py-lineno">645</tt>  <tt class="py-line"><tt class="py-docstring">        returns a new dictionary with the same keys as the given dictionary</tt> </tt>
<a name="L646"></a><tt class="py-lineno">646</tt>  <tt class="py-line"><tt class="py-docstring">        and log-transformed values.</tt> </tt>
<a name="L647"></a><tt class="py-lineno">647</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L648"></a><tt class="py-lineno">648</tt>  <tt class="py-line">        <tt class="py-name">log_prob</tt> <tt class="py-op">=</tt> <tt id="link-97" class="py-name"><a title="Bio.Align.substitution_matrices.Array.copy
Bio.Crystal.Crystal.copy
Bio.File._IndexedSeqFileDict.copy
Bio.PDB.Atom.Atom.copy
Bio.PDB.Entity.Entity.copy
Bio.PDB.vectors.Vector.copy" class="py-name" href="#" onclick="return doclink('link-97', 'copy', 'link-0');">copy</a></tt><tt class="py-op">.</tt><tt id="link-98" class="py-name"><a title="Bio.Align.substitution_matrices.Array.copy
Bio.Crystal.Crystal.copy
Bio.File._IndexedSeqFileDict.copy
Bio.PDB.Atom.Atom.copy
Bio.PDB.Entity.Entity.copy
Bio.PDB.vectors.Vector.copy" class="py-name" href="#" onclick="return doclink('link-98', 'copy', 'link-0');">copy</a></tt><tt class="py-op">(</tt><tt class="py-name">probability</tt><tt class="py-op">)</tt> </tt>
<a name="L649"></a><tt class="py-lineno">649</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L650"></a><tt class="py-lineno">650</tt>  <tt class="py-line">            <tt class="py-name">neg_inf</tt> <tt class="py-op">=</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-string">"-inf"</tt><tt class="py-op">)</tt> </tt>
<a name="L651"></a><tt class="py-lineno">651</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L652"></a><tt class="py-lineno">652</tt>  <tt class="py-line">            <tt class="py-comment"># On Python 2.5 or older that was handled in C code,</tt> </tt>
<a name="L653"></a><tt class="py-lineno">653</tt>  <tt class="py-line">            <tt class="py-comment"># and failed on Windows XP 32bit</tt> </tt>
<a name="L654"></a><tt class="py-lineno">654</tt>  <tt class="py-line">            <tt class="py-name">neg_inf</tt> <tt class="py-op">=</tt> <tt class="py-op">-</tt> <tt class="py-number">1E400</tt> </tt>
<a name="L655"></a><tt class="py-lineno">655</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">log_prob</tt><tt class="py-op">:</tt> </tt>
<a name="L656"></a><tt class="py-lineno">656</tt>  <tt class="py-line">            <tt class="py-name">prob</tt> <tt class="py-op">=</tt> <tt class="py-name">log_prob</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> </tt>
<a name="L657"></a><tt class="py-lineno">657</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">prob</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L658"></a><tt class="py-lineno">658</tt>  <tt class="py-line">                <tt class="py-name">log_prob</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">math</tt><tt class="py-op">.</tt><tt class="py-name">log</tt><tt class="py-op">(</tt><tt class="py-name">log_prob</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L659"></a><tt class="py-lineno">659</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L660"></a><tt class="py-lineno">660</tt>  <tt class="py-line">                <tt class="py-name">log_prob</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">neg_inf</tt> </tt>
<a name="L661"></a><tt class="py-lineno">661</tt>  <tt class="py-line"> </tt>
<a name="L662"></a><tt class="py-lineno">662</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">log_prob</tt> </tt>
</div></div><a name="L663"></a><tt class="py-lineno">663</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Nov  6 13:33:55 2019
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
