<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.AlignIO.MafIO</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.AlignIO-module.html">Package&nbsp;AlignIO</a> ::
        Module&nbsp;MafIO
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.AlignIO.MafIO-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.AlignIO.MafIO-module.html">Module Bio.AlignIO.MafIO</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2011, 2012 by Andrew Sczesnak.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># Revisions Copyright 2011, 2017 by Peter Cock.  All rights reserved.</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># Revisions Copyright 2014, 2015 by Adam Novak.  All rights reserved.</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># Revisions Copyright 2015, 2017 by Blaise Li.  All rights reserved.</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-docstring">"""Bio.AlignIO support for the "maf" multiple alignment format.</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-docstring">The Multiple Alignment Format, described by UCSC, stores a series of</tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-docstring">multiple alignments in a single file. It is suitable for whole-genome</tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-docstring">to whole-genome alignments, metadata such as source chromosome, start</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-docstring">position, size, and strand can be stored.</tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-docstring">See http://genome.ucsc.edu/FAQ/FAQformat.html#format5</tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-docstring">You are expected to use this module via the Bio.AlignIO functions(or the</tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-docstring">Bio.SeqIO functions if you want to work directly with the gapped sequences).</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-docstring">Coordinates in the MAF format are defined in terms of zero-based start</tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-docstring">positions (like Python) and aligning region sizes.</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-docstring">A minimal aligned region of length one and starting at first position in the</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-docstring">source sequence would have ``start == 0`` and ``size == 1``.</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"><tt class="py-docstring">As we can see on this example, ``start + size`` will give one more than the</tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt class="py-docstring">zero-based end position. We can therefore manipulate ``start`` and</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"><tt class="py-docstring">``start + size`` as python list slice boundaries.</tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"><tt class="py-docstring">For an inclusive end coordinate, we need to use ``end = start + size - 1``.</tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"><tt class="py-docstring">A 1-column wide alignment would have ``start == end``.</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">os</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">itertools</tt> <tt class="py-keyword">import</tt> <tt class="py-name">islice</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"><tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt class="py-name">sqlite3</tt> <tt class="py-keyword">import</tt> <tt class="py-name">dbapi2</tt> <tt class="py-keyword">as</tt> <tt class="py-name">_sqlite</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"><tt class="py-keyword">except</tt> <tt class="py-name">ImportError</tt><tt class="py-op">:</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line">    <tt class="py-comment"># Not present on Jython, but should be included in Python 2.5</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line">    <tt class="py-comment"># or later (unless compiled from source without its dependencies)</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line">    <tt class="py-comment"># Still want to offer simple parsing/output</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line">    <tt class="py-name">_sqlite</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line"> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-1', 'Alphabet', 'link-1');">Alphabet</a></tt> <tt class="py-keyword">import</tt> <tt id="link-2" class="py-name" targets="Variable Bio.Alphabet.single_letter_alphabet=Bio.Alphabet-module.html#single_letter_alphabet"><a title="Bio.Alphabet.single_letter_alphabet" class="py-name" href="#" onclick="return doclink('link-2', 'single_letter_alphabet', 'link-2');">single_letter_alphabet</a></tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-3" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-3', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-4" class="py-name" targets="Module Bio.Seq=Bio.Seq-module.html,Class Bio.Seq.Seq=Bio.Seq.Seq-class.html"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-4', 'Seq', 'link-4');">Seq</a></tt> <tt class="py-keyword">import</tt> <tt id="link-5" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-5', 'Seq', 'link-4');">Seq</a></tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-6" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-6', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-7" class="py-name" targets="Module Bio.SeqRecord=Bio.SeqRecord-module.html,Class Bio.SeqRecord.SeqRecord=Bio.SeqRecord.SeqRecord-class.html"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-7', 'SeqRecord', 'link-7');">SeqRecord</a></tt> <tt class="py-keyword">import</tt> <tt id="link-8" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-8', 'SeqRecord', 'link-7');">SeqRecord</a></tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-9" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-9', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-10" class="py-name" targets="Package Bio.Align=Bio.Align-module.html"><a title="Bio.Align" class="py-name" href="#" onclick="return doclink('link-10', 'Align', 'link-10');">Align</a></tt> <tt class="py-keyword">import</tt> <tt id="link-11" class="py-name" targets="Class Bio.Align.MultipleSeqAlignment=Bio.Align.MultipleSeqAlignment-class.html"><a title="Bio.Align.MultipleSeqAlignment" class="py-name" href="#" onclick="return doclink('link-11', 'MultipleSeqAlignment', 'link-11');">MultipleSeqAlignment</a></tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt><tt id="link-12" class="py-name" targets="Module Bio.AlignIO.Interfaces=Bio.AlignIO.Interfaces-module.html,Module Bio.SeqIO.Interfaces=Bio.SeqIO.Interfaces-module.html"><a title="Bio.AlignIO.Interfaces
Bio.SeqIO.Interfaces" class="py-name" href="#" onclick="return doclink('link-12', 'Interfaces', 'link-12');">Interfaces</a></tt> <tt class="py-keyword">import</tt> <tt id="link-13" class="py-name" targets="Class Bio.AlignIO.Interfaces.SequentialAlignmentWriter=Bio.AlignIO.Interfaces.SequentialAlignmentWriter-class.html"><a title="Bio.AlignIO.Interfaces.SequentialAlignmentWriter" class="py-name" href="#" onclick="return doclink('link-13', 'SequentialAlignmentWriter', 'link-13');">SequentialAlignmentWriter</a></tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line"> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line"><tt id="link-14" class="py-name" targets="Variable Bio.AlignIO.MafIO.MAFINDEX_VERSION=Bio.AlignIO.MafIO-module.html#MAFINDEX_VERSION"><a title="Bio.AlignIO.MafIO.MAFINDEX_VERSION" class="py-name" href="#" onclick="return doclink('link-14', 'MAFINDEX_VERSION', 'link-14');">MAFINDEX_VERSION</a></tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="MafWriter"></a><div id="MafWriter-def"><a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line"> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line"> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt> <a class="py-toggle" href="#" id="MafWriter-toggle" onclick="return toggle('MafWriter');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafWriter-class.html">MafWriter</a><tt class="py-op">(</tt><tt class="py-base-class">SequentialAlignmentWriter</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafWriter-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="MafWriter-expanded"><a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line">    <tt class="py-docstring">"""Accepts a MultipleSeqAlignment object, writes a MAF file."""</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line"> </tt>
<a name="MafWriter.write_header"></a><div id="MafWriter.write_header-def"><a name="L57"></a><tt class="py-lineno"> 57</tt> <a class="py-toggle" href="#" id="MafWriter.write_header-toggle" onclick="return toggle('MafWriter.write_header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafWriter-class.html#write_header">write_header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafWriter.write_header-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafWriter.write_header-expanded"><a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line">        <tt class="py-docstring">"""Write the MAF header."""</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-15" class="py-name" targets="Function Bio.AlignIO.write()=Bio.AlignIO-module.html#write,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.write()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#write,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.write()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#write,Method Bio.Phylo.CDAOIO.Writer.write()=Bio.Phylo.CDAOIO.Writer-class.html#write,Function Bio.Phylo.CDAOIO.write()=Bio.Phylo.CDAOIO-module.html#write,Method Bio.Phylo.NeXMLIO.Writer.write()=Bio.Phylo.NeXMLIO.Writer-class.html#write,Function Bio.Phylo.NeXMLIO.write()=Bio.Phylo.NeXMLIO-module.html#write,Method Bio.Phylo.NewickIO.Writer.write()=Bio.Phylo.NewickIO.Writer-class.html#write,Function Bio.Phylo.NewickIO.write()=Bio.Phylo.NewickIO-module.html#write,Function Bio.Phylo.NexusIO.write()=Bio.Phylo.NexusIO-module.html#write,Method Bio.Phylo.PhyloXMLIO.Writer.write()=Bio.Phylo.PhyloXMLIO.Writer-class.html#write,Function Bio.Phylo.PhyloXMLIO.write()=Bio.Phylo.PhyloXMLIO-module.html#write,Function Bio.Phylo._io.write()=Bio.Phylo._io-module.html#write,Function Bio.SearchIO.write()=Bio.SearchIO-module.html#write,Function Bio.SeqIO.write()=Bio.SeqIO-module.html#write,Method Bio.bgzf.BgzfWriter.write()=Bio.bgzf.BgzfWriter-class.html#write,Function Bio.motifs.jaspar.write()=Bio.motifs.jaspar-module.html#write,Function Bio.motifs.transfac.write()=Bio.motifs.transfac-module.html#write,Function Bio.motifs.write()=Bio.motifs-module.html#write,Method Bio.phenotype.phen_micro.JsonWriter.write()=Bio.phenotype.phen_micro.JsonWriter-class.html#write,Function Bio.phenotype.write()=Bio.phenotype-module.html#write"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-15', 'write', 'link-15');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"##maf version=1 scoring=none\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-16" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-16', 'write', 'link-15');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"# generated by Biopython\n\n"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line"> </tt>
<a name="MafWriter._write_record"></a><div id="MafWriter._write_record-def"><a name="L62"></a><tt class="py-lineno"> 62</tt> <a class="py-toggle" href="#" id="MafWriter._write_record-toggle" onclick="return toggle('MafWriter._write_record');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafWriter-class.html#_write_record">_write_record</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafWriter._write_record-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafWriter._write_record-expanded"><a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line">        <tt class="py-docstring">"""Write a single SeqRecord object to an 's' line in a MAF block (PRIVATE)."""</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line">        <tt class="py-comment"># convert biopython-style 1/-1 strand to MAF-style +/- strand</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-17" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.annotations=BioSQL.BioSeq.DBSeqRecord-class.html#annotations"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-17', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-18" class="py-name" targets="Method Bio.Crystal.Crystal.get()=Bio.Crystal.Crystal-class.html#get,Method Bio.Data.CodonTable.AmbiguousForwardTable.get()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#get,Method Bio.File._IndexedSeqFileDict.get()=Bio.File._IndexedSeqFileDict-class.html#get,Method Bio.File._IndexedSeqFileProxy.get()=Bio.File._IndexedSeqFileProxy-class.html#get,Method Bio.File._SQLiteManySeqFilesDict.get()=Bio.File._SQLiteManySeqFilesDict-class.html#get,Method Bio.Restriction.Restriction.RestrictionBatch.get()=Bio.Restriction.Restriction.RestrictionBatch-class.html#get,Method Bio.SearchIO._index.SearchIndexer.get()=Bio.SearchIO._index.SearchIndexer-class.html#get,Method Bio.SeqIO._index.SeqFileRandomAccess.get()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#get,Method Bio.SeqIO._index.SffRandomAccess.get()=Bio.SeqIO._index.SffRandomAccess-class.html#get,Method Bio.SeqIO._index.SffTrimedRandomAccess.get()=Bio.SeqIO._index.SffTrimedRandomAccess-class.html#get,Method Bio.SeqIO._index.UniprotRandomAccess.get()=Bio.SeqIO._index.UniprotRandomAccess-class.html#get"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-18', 'get', 'link-18');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"strand"</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line">            <tt id="link-19" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.strand()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#strand,Variable Bio.SeqFeature.CompoundLocation.strand=Bio.SeqFeature.CompoundLocation-class.html#strand,Variable Bio.SeqFeature.FeatureLocation.strand=Bio.SeqFeature.FeatureLocation-class.html#strand,Variable Bio.SeqFeature.SeqFeature.strand=Bio.SeqFeature.SeqFeature-class.html#strand"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-19', 'strand', 'link-19');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"+"</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-20" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-20', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-21" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-21', 'get', 'link-18');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"strand"</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line">            <tt id="link-22" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-22', 'strand', 'link-19');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"-"</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line">            <tt class="py-comment"># TODO: issue warning?</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line">            <tt id="link-23" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-23', 'strand', 'link-19');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"+"</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line"> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line">        <tt class="py-name">fields</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">"s"</tt><tt class="py-op">,</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">                  <tt class="py-comment"># In the MAF file format, spaces are not allowed in the id</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line">                  <tt class="py-string">"%-40s"</tt> <tt class="py-op">%</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-24" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Variable Bio.PDB.Entity.Entity.id=Bio.PDB.Entity.Entity-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-24', 'id', 'link-24');">id</a></tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">" "</tt><tt class="py-op">,</tt> <tt class="py-string">"_"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line">                  <tt class="py-string">"%15s"</tt> <tt class="py-op">%</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-25" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-25', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-26" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-26', 'get', 'link-18');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"start"</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line">                  <tt class="py-string">"%5s"</tt> <tt class="py-op">%</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-27" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-27', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-28" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-28', 'get', 'link-18');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"size"</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-29" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.seq=Bio.SeqRecord.SeqRecord-class.html#seq,Variable BioSQL.BioSeq.DBSeqRecord.seq=BioSQL.BioSeq.DBSeqRecord-class.html#seq"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-29', 'seq', 'link-29');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"-"</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line">                  <tt id="link-30" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-30', 'strand', 'link-19');">strand</a></tt><tt class="py-op">,</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line">                  <tt class="py-string">"%15s"</tt> <tt class="py-op">%</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-31" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-31', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-32" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-32', 'get', 'link-18');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"srcSize"</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line">                  <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-33" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-33', 'seq', 'link-29');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-34" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-34', 'write', 'link-15');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"%s\n"</tt> <tt class="py-op">%</tt> <tt class="py-string">" "</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">fields</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line"> </tt>
<a name="MafWriter.write_alignment"></a><div id="MafWriter.write_alignment-def"><a name="L83"></a><tt class="py-lineno"> 83</tt> <a class="py-toggle" href="#" id="MafWriter.write_alignment-toggle" onclick="return toggle('MafWriter.write_alignment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafWriter-class.html#write_alignment">write_alignment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">alignment</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafWriter.write_alignment-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafWriter.write_alignment-expanded"><a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line">        <tt class="py-docstring">"""Write a complete alignment to a MAF block.</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line"><tt class="py-docstring">        Writes every SeqRecord in a MultipleSeqAlignment object to its own</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line"><tt class="py-docstring">        MAF block (beginning with an 'a' line, containing 's' lines).</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">alignment</tt><tt class="py-op">,</tt> <tt id="link-35" class="py-name"><a title="Bio.Align.MultipleSeqAlignment" class="py-name" href="#" onclick="return doclink('link-35', 'MultipleSeqAlignment', 'link-11');">MultipleSeqAlignment</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"Expected an alignment object"</tt><tt class="py-op">)</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line"> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-36" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.set()=Bio.Nexus.Nexus.StepMatrix-class.html#set,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.set()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#set,Method Bio.SVDSuperimposer.SVDSuperimposer.set()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#set"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-36', 'set', 'link-36');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-37" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.x=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#x,Variable Bio.Statistics.lowess.x=Bio.Statistics.lowess-module.html#x"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-37', 'x', 'link-37');">x</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-38" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-38', 'x', 'link-37');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">alignment</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Sequences must all be the same length"</tt><tt class="py-op">)</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line"> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line">        <tt class="py-comment"># We allow multiple sequences with the same IDs; for example, there may</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line">        <tt class="py-comment"># be a MAF aligning the + and - strands of the same sequence together.</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line"> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line">        <tt class="py-comment"># for now, use ._annotations private property, but restrict keys to those</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">        <tt class="py-comment"># specifically supported by the MAF format, according to spec</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line">            <tt class="py-name">anno</tt> <tt class="py-op">=</tt> <tt class="py-string">" "</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-string">"%s=%s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-39" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-39', 'x', 'link-37');">x</a></tt><tt class="py-op">,</tt> <tt id="link-40" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.y=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#y"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-40', 'y', 'link-40');">y</a></tt><tt class="py-op">)</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line">                             <tt class="py-keyword">for</tt> <tt id="link-41" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-41', 'x', 'link-37');">x</a></tt><tt class="py-op">,</tt> <tt id="link-42" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-42', 'y', 'link-40');">y</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">alignment</tt><tt class="py-op">.</tt><tt class="py-name">_annotations</tt><tt class="py-op">.</tt><tt id="link-43" class="py-name" targets="Method Bio.Crystal.Crystal.items()=Bio.Crystal.Crystal-class.html#items,Method Bio.File._IndexedSeqFileDict.items()=Bio.File._IndexedSeqFileDict-class.html#items,Method Bio.Phylo.PhyloXML.Events.items()=Bio.Phylo.PhyloXML.Events-class.html#items,Variable Bio.SearchIO._model.query.QueryResult.items=Bio.SearchIO._model.query.QueryResult-class.html#items,Method BioSQL.BioSeqDatabase.BioSeqDatabase.items()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#items,Method BioSQL.BioSeqDatabase.DBServer.items()=BioSQL.BioSeqDatabase.DBServer-class.html#items"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-43', 'items', 'link-43');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">                             <tt class="py-keyword">if</tt> <tt id="link-44" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-44', 'x', 'link-37');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">"score"</tt><tt class="py-op">,</tt> <tt class="py-string">"pass"</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">            <tt class="py-name">anno</tt> <tt class="py-op">=</tt> <tt class="py-string">"score=0.00"</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line"> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-45" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-45', 'write', 'link-15');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"a %s\n"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">anno</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line"> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">        <tt class="py-name">recs_out</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line"> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">alignment</tt><tt class="py-op">:</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-46" class="py-name" targets="Method Bio.AlignIO.MafIO.MafWriter._write_record()=Bio.AlignIO.MafIO.MafWriter-class.html#_write_record,Method Bio.AlignIO.MauveIO.MauveWriter._write_record()=Bio.AlignIO.MauveIO.MauveWriter-class.html#_write_record,Method Bio.AlignIO.StockholmIO.StockholmWriter._write_record()=Bio.AlignIO.StockholmIO.StockholmWriter-class.html#_write_record"><a title="Bio.AlignIO.MafIO.MafWriter._write_record
Bio.AlignIO.MauveIO.MauveWriter._write_record
Bio.AlignIO.StockholmIO.StockholmWriter._write_record" class="py-name" href="#" onclick="return doclink('link-46', '_write_record', 'link-46');">_write_record</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line"> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">            <tt class="py-name">recs_out</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line"> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-47" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-47', 'write', 'link-15');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line"> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">recs_out</tt> </tt>
</div></div><a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line"> </tt>
<a name="MafIterator"></a><div id="MafIterator-def"><a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line"> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line"><tt class="py-comment"># Invalid function name according to pylint, but kept for compatibility</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line"><tt class="py-comment"># with Bio* conventions.</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt> <a class="py-toggle" href="#" id="MafIterator-toggle" onclick="return toggle('MafIterator');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO-module.html#MafIterator">MafIterator</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">seq_count</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt id="link-48" class="py-name"><a title="Bio.Alphabet.single_letter_alphabet" class="py-name" href="#" onclick="return doclink('link-48', 'single_letter_alphabet', 'link-2');">single_letter_alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIterator-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="MafIterator-expanded"><a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">    <tt class="py-docstring">"""Iterate over a MAF file handle as MultipleSeqAlignment objects.</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line"><tt class="py-docstring">    Iterates over lines in a MAF file-like object (handle), yielding</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line"><tt class="py-docstring">    MultipleSeqAlignment objects. SeqRecord IDs generally correspond to</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line"><tt class="py-docstring">    species names.</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line">    <tt class="py-name">in_a_bundle</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line"> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line">    <tt id="link-49" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-49', 'annotations', 'link-17');">annotations</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line">    <tt class="py-name">records</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line"> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line">    <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">        <tt class="py-comment"># allows parsing of the last bundle without duplicating code</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-50" class="py-name" targets="Method Bio.AlignIO.Interfaces.AlignmentIterator.next()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#next,Method Bio.Blast.NCBIStandalone.Iterator.next()=Bio.Blast.NCBIStandalone.Iterator-class.html#next,Method Bio.Blast.ParseBlastTable.BlastTableReader.next()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#next,Method Bio.DocSQL.IterationCursor.next()=Bio.DocSQL.IterationCursor-class.html#next,Method Bio.File.UndoHandle.next()=Bio.File.UndoHandle-class.html#next,Method Bio.GenBank.Iterator.next()=Bio.GenBank.Iterator-class.html#next,Method Bio.Nexus.Nexus.CharBuffer.next()=Bio.Nexus.Nexus.CharBuffer-class.html#next,Method Bio.Nexus.StandardData.StandardData.next()=Bio.Nexus.StandardData.StandardData-class.html#next,Method Bio.PopGen.GenePop.Controller._FileIterator.next()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#next,Method Bio.SeqIO.Interfaces.SequenceIterator.next()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#next,Method Bio.bgzf.BgzfReader.next()=Bio.bgzf.BgzfReader-class.html#next"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.Nexus.StandardData.StandardData.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-50', 'next', 'link-50');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line"> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">in_a_bundle</tt><tt class="py-op">:</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-51" class="py-name" targets="Method Bio.Seq.Seq.startswith()=Bio.Seq.Seq-class.html#startswith"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-51', 'startswith', 'link-51');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"s"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line">                <tt class="py-comment"># add a SeqRecord to the bundle</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">                <tt class="py-name">line_split</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-52" class="py-name" targets="Method Bio.Seq.Seq.strip()=Bio.Seq.Seq-class.html#strip"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-52', 'strip', 'link-52');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-53" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-53', 'split', 'link-53');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line"> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">line_split</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-number">7</tt><tt class="py-op">:</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Error parsing alignment - 's' line must have 7 fields"</tt><tt class="py-op">)</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line"> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">                <tt class="py-comment"># convert MAF-style +/- strand to biopython-type 1/-1</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"+"</tt><tt class="py-op">:</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">                    <tt id="link-54" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-54', 'strand', 'link-19');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"-"</tt><tt class="py-op">:</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line">                    <tt id="link-55" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-55', 'strand', 'link-19');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">                    <tt class="py-comment"># TODO: issue warning, set to 0?</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">                    <tt id="link-56" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-56', 'strand', 'link-19');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line"> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">                <tt class="py-comment"># s (literal), src (ID), start, size, strand, srcSize, text (sequence)</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">                <tt class="py-name">anno</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">"start"</tt><tt class="py-op">:</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">                        <tt class="py-string">"size"</tt><tt class="py-op">:</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">                        <tt class="py-string">"strand"</tt><tt class="py-op">:</tt> <tt id="link-57" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-57', 'strand', 'link-19');">strand</a></tt><tt class="py-op">,</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">                        <tt class="py-string">"srcSize"</tt><tt class="py-op">:</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">5</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">}</tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line"> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">                <tt id="link-58" class="py-name" targets="Method Bio.FSSP.FSSPAlignDict.sequence()=Bio.FSSP.FSSPAlignDict-class.html#sequence,Method Bio.GenBank._FeatureConsumer.sequence()=Bio.GenBank._FeatureConsumer-class.html#sequence,Method Bio.GenBank._RecordConsumer.sequence()=Bio.GenBank._RecordConsumer-class.html#sequence,Method Bio.Phylo.PhyloXMLIO.Writer.sequence()=Bio.Phylo.PhyloXMLIO.Writer-class.html#sequence"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-58', 'sequence', 'link-58');">sequence</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">]</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line"> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">                <tt class="py-comment"># interpret a dot/period to mean the same as the first sequence</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-string">"."</tt> <tt class="py-keyword">in</tt> <tt id="link-59" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-59', 'sequence', 'link-58');">sequence</a></tt><tt class="py-op">:</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">records</tt><tt class="py-op">:</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line">                        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Found dot/period in first sequence of alignment"</tt><tt class="py-op">)</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line"> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line">                    <tt id="link-60" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.ref=Bio.SeqFeature.CompoundLocation-class.html#ref,Variable Bio.SeqFeature.SeqFeature.ref=Bio.SeqFeature.SeqFeature-class.html#ref"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-60', 'ref', 'link-60');">ref</a></tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">records</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-61" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-61', 'seq', 'link-29');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line">                    <tt class="py-name">new</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line"> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-op">(</tt><tt class="py-name">letter</tt><tt class="py-op">,</tt> <tt class="py-name">ref_letter</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt id="link-62" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-62', 'sequence', 'link-58');">sequence</a></tt><tt class="py-op">,</tt> <tt id="link-63" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-63', 'ref', 'link-60');">ref</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">                        <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-64" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-64', 'append', 'link-64');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">ref_letter</tt> <tt class="py-keyword">if</tt> <tt class="py-name">letter</tt> <tt class="py-op">==</tt> <tt class="py-string">"."</tt> <tt class="py-keyword">else</tt> <tt class="py-name">letter</tt><tt class="py-op">)</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line"> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">                    <tt id="link-65" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-65', 'sequence', 'link-58');">sequence</a></tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">new</tt><tt class="py-op">)</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line"> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">                <tt class="py-name">records</tt><tt class="py-op">.</tt><tt id="link-66" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-66', 'append', 'link-64');">append</a></tt><tt class="py-op">(</tt><tt id="link-67" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-67', 'SeqRecord', 'link-7');">SeqRecord</a></tt><tt class="py-op">(</tt><tt id="link-68" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-68', 'Seq', 'link-4');">Seq</a></tt><tt class="py-op">(</tt><tt id="link-69" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-69', 'sequence', 'link-58');">sequence</a></tt><tt class="py-op">,</tt> <tt id="link-70" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.alphabet=Bio.SearchIO._model.hsp.HSP-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSPFragment.alphabet=Bio.SearchIO._model.hsp.HSPFragment-class.html#alphabet"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-70', 'alphabet', 'link-70');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">                                         <tt id="link-71" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-71', 'id', 'link-24');">id</a></tt><tt class="py-op">=</tt><tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">                                         <tt id="link-72" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-72', 'name', 'link-72');">name</a></tt><tt class="py-op">=</tt><tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">                                         <tt id="link-73" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DescriptionConsumer.description()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#description,Method Bio.Phylo.PhyloXMLIO.Writer.description()=Bio.Phylo.PhyloXMLIO.Writer-class.html#description,Variable Bio.SearchIO._model.hit.Hit.description=Bio.SearchIO._model.hit.Hit-class.html#description,Variable Bio.SearchIO._model.query.QueryResult.description=Bio.SearchIO._model.query.QueryResult-class.html#description"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-73', 'description', 'link-73');">description</a></tt><tt class="py-op">=</tt><tt class="py-string">""</tt><tt class="py-op">,</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">                                         <tt id="link-74" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-74', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">=</tt><tt class="py-name">anno</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-75" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-75', 'startswith', 'link-51');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"i"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line">                <tt class="py-comment"># TODO: information about what is in the aligned species DNA before</tt> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">                <tt class="py-comment"># and after the immediately preceding "s" line</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">                <tt class="py-keyword">pass</tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-76" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-76', 'startswith', 'link-51');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"e"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">                <tt class="py-comment"># TODO: information about the size of the gap between the alignments</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">                <tt class="py-comment"># that span the current block</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">                <tt class="py-keyword">pass</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-77" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-77', 'startswith', 'link-51');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"q"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">                <tt class="py-comment"># TODO: quality of each aligned base for the species.</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">                <tt class="py-comment"># Need to find documentation on this, looks like ASCII 0-9 or gap?</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">                <tt class="py-comment"># Can then store in each SeqRecord's .letter_annotations dictionary,</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">                <tt class="py-comment"># perhaps as the raw string or turned into integers / None for gap?</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line">                <tt class="py-keyword">pass</tt> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-78" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-78', 'startswith', 'link-51');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"#"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line">                <tt class="py-comment"># ignore comments</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line">                <tt class="py-comment"># (not sure whether comments</tt> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line">                <tt class="py-comment"># are in the maf specification, though)</tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line">                <tt class="py-keyword">pass</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-79" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-79', 'strip', 'link-52');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line">                <tt class="py-comment"># end a bundle of records</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">seq_count</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line">                    <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">records</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">seq_count</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line"> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line">                <tt class="py-name">alignment</tt> <tt class="py-op">=</tt> <tt id="link-80" class="py-name"><a title="Bio.Align.MultipleSeqAlignment" class="py-name" href="#" onclick="return doclink('link-80', 'MultipleSeqAlignment', 'link-11');">MultipleSeqAlignment</a></tt><tt class="py-op">(</tt><tt class="py-name">records</tt><tt class="py-op">,</tt> <tt id="link-81" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-81', 'alphabet', 'link-70');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line">                <tt class="py-comment"># TODO - Introduce an annotated alignment class?</tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line">                <tt class="py-comment"># See also Bio/AlignIO/FastaIO.py for same requirement.</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line">                <tt class="py-comment"># For now, store the annotation a new private property:</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line">                <tt class="py-name">alignment</tt><tt class="py-op">.</tt><tt class="py-name">_annotations</tt> <tt class="py-op">=</tt> <tt id="link-82" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-82', 'annotations', 'link-17');">annotations</a></tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line"> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt class="py-name">alignment</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line"> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line">                <tt class="py-name">in_a_bundle</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line"> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line">                <tt id="link-83" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-83', 'annotations', 'link-17');">annotations</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">                <tt class="py-name">records</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Error parsing alignment - unexpected line:\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-84" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-84', 'startswith', 'link-51');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"a"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line">            <tt class="py-comment"># start a bundle of records</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line">            <tt class="py-name">in_a_bundle</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line">            <tt class="py-name">annot_strings</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-85" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-85', 'strip', 'link-52');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-86" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-86', 'split', 'link-53');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">annot_strings</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-87" class="py-name" targets="Method Bio.Crystal.Chain.count()=Bio.Crystal.Chain-class.html#count,Method Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count()=Bio.NeuralNetwork.Gene.Pattern.PatternRepository-class.html#count,Method Bio.Seq.MutableSeq.count()=Bio.Seq.MutableSeq-class.html#count,Method Bio.Seq.Seq.count()=Bio.Seq.Seq-class.html#count,Method Bio.Seq.UnknownSeq.count()=Bio.Seq.UnknownSeq-class.html#count,Method Bio.motifs.Instances.count()=Bio.motifs.Instances-class.html#count"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-87', 'count', 'link-87');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">"="</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Error parsing alignment - invalid key in 'a' line"</tt><tt class="py-op">)</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">            <tt id="link-88" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-88', 'annotations', 'link-17');">annotations</a></tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">a_string</tt><tt class="py-op">.</tt><tt id="link-89" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-89', 'split', 'link-53');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"="</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">a_string</tt> <tt class="py-keyword">in</tt> <tt class="py-name">annot_strings</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-90" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-90', 'startswith', 'link-51');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"#"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line">            <tt class="py-comment"># ignore comments</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line">            <tt class="py-keyword">break</tt> </tt>
</div><a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line"> </tt>
<a name="MafIndex"></a><div id="MafIndex-def"><a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line"> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt> <a class="py-toggle" href="#" id="MafIndex-toggle" onclick="return toggle('MafIndex');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafIndex-class.html">MafIndex</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIndex-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="MafIndex-expanded"><a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line">    <tt class="py-docstring">"""Index for a MAF file.</tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line"><tt class="py-docstring">    The index is a sqlite3 database that is built upon creation of the object</tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line"><tt class="py-docstring">    if necessary, and queried when methods *search* or *get_spliced* are</tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line"><tt class="py-docstring">    used.</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line"> </tt>
<a name="MafIndex.__init__"></a><div id="MafIndex.__init__-def"><a name="L245"></a><tt class="py-lineno">245</tt> <a class="py-toggle" href="#" id="MafIndex.__init__-toggle" onclick="return toggle('MafIndex.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafIndex-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sqlite_file</tt><tt class="py-op">,</tt> <tt class="py-param">maf_file</tt><tt class="py-op">,</tt> <tt class="py-param">target_seqname</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIndex.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafIndex.__init__-expanded"><a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line">        <tt class="py-docstring">"""Indexes or loads the index of a MAF file."""</tt> </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt> <tt class="py-op">=</tt> <tt class="py-name">target_seqname</tt> </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line">        <tt class="py-comment"># example: Tests/MAF/ucsc_mm9_chr10.mafindex</tt> </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_index_filename</tt> <tt class="py-op">=</tt> <tt class="py-name">sqlite_file</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line">        <tt class="py-comment"># example: /home/bli/src/biopython/Tests/MAF</tt> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_relative_path</tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">abspath</tt><tt class="py-op">(</tt><tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">dirname</tt><tt class="py-op">(</tt><tt class="py-name">sqlite_file</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line">        <tt class="py-comment"># example: Tests/MAF/ucsc_mm9_chr10.maf</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_file</tt> <tt class="py-op">=</tt> <tt class="py-name">maf_file</tt> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line"> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_fp</tt> <tt class="py-op">=</tt> <tt id="link-91" class="py-name" targets="Function Bio.bgzf.open()=Bio.bgzf-module.html#open"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-91', 'open', 'link-91');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_file</tt><tt class="py-op">,</tt> <tt class="py-string">"r"</tt><tt class="py-op">)</tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line"> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line">        <tt class="py-comment"># if sqlite_file exists, use the existing db, otherwise index the file</tt> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">isfile</tt><tt class="py-op">(</tt><tt class="py-name">sqlite_file</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt> <tt class="py-op">=</tt> <tt class="py-name">_sqlite</tt><tt class="py-op">.</tt><tt class="py-name">connect</tt><tt class="py-op">(</tt><tt class="py-name">sqlite_file</tt><tt class="py-op">)</tt> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_record_count</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-92" class="py-name" targets="Method Bio.AlignIO.MafIO.MafIndex.__check_existing_db()=Bio.AlignIO.MafIO.MafIndex-class.html#__check_existing_db"><a title="Bio.AlignIO.MafIO.MafIndex.__check_existing_db" class="py-name" href="#" onclick="return doclink('link-92', '__check_existing_db', 'link-92');">__check_existing_db</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt> <tt class="py-op">=</tt> <tt class="py-name">_sqlite</tt><tt class="py-op">.</tt><tt class="py-name">connect</tt><tt class="py-op">(</tt><tt class="py-name">sqlite_file</tt><tt class="py-op">)</tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_record_count</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-93" class="py-name" targets="Method Bio.AlignIO.MafIO.MafIndex.__make_new_index()=Bio.AlignIO.MafIO.MafIndex-class.html#__make_new_index"><a title="Bio.AlignIO.MafIO.MafIndex.__make_new_index" class="py-name" href="#" onclick="return doclink('link-93', '__make_new_index', 'link-93');">__make_new_index</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line"> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line">        <tt class="py-comment"># lastly, setup a MafIterator pointing at the open maf_file</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_mafiter</tt> <tt class="py-op">=</tt> <tt id="link-94" class="py-name" targets="Function Bio.AlignIO.MafIO.MafIterator()=Bio.AlignIO.MafIO-module.html#MafIterator"><a title="Bio.AlignIO.MafIO.MafIterator" class="py-name" href="#" onclick="return doclink('link-94', 'MafIterator', 'link-94');">MafIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_fp</tt><tt class="py-op">)</tt> </tt>
</div><a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line"> </tt>
<a name="MafIndex.__check_existing_db"></a><div id="MafIndex.__check_existing_db-def"><a name="L268"></a><tt class="py-lineno">268</tt> <a class="py-toggle" href="#" id="MafIndex.__check_existing_db-toggle" onclick="return toggle('MafIndex.__check_existing_db');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafIndex-class.html#__check_existing_db">__check_existing_db</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIndex.__check_existing_db-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafIndex.__check_existing_db-expanded"><a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">        <tt class="py-docstring">"""Perform basic sanity checks upon loading an existing index (PRIVATE)."""</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line">            <tt class="py-name">idx_version</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-95" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.execute()=BioSQL.BioSeqDatabase.Adaptor-class.html#execute,Method BioSQL.BioSeqDatabase._CursorWrapper.execute()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#execute,Method BioSQL.DBUtils.Generic_dbutils.execute()=BioSQL.DBUtils.Generic_dbutils-class.html#execute,Method BioSQL.DBUtils.Sqlite_dbutils.execute()=BioSQL.DBUtils.Sqlite_dbutils-class.html#execute"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-95', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line">                <tt class="py-string">"SELECT value FROM meta_data WHERE key = 'version'"</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-96" class="py-name" targets="Method BioSQL.BioSeqDatabase._CursorWrapper.fetchone()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#fetchone"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchone" class="py-name" href="#" onclick="return doclink('link-96', 'fetchone', 'link-96');">fetchone</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">idx_version</tt> <tt class="py-op">!=</tt> <tt id="link-97" class="py-name"><a title="Bio.AlignIO.MafIO.MAFINDEX_VERSION" class="py-name" href="#" onclick="return doclink('link-97', 'MAFINDEX_VERSION', 'link-14');">MAFINDEX_VERSION</a></tt><tt class="py-op">:</tt> </tt>
<a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Index version (%s) incompatible with "</tt> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line">                                 <tt class="py-string">"this version of MafIndex"</tt> <tt class="py-op">%</tt> <tt class="py-name">idx_version</tt><tt class="py-op">)</tt> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line"> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line">            <tt class="py-name">filename</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-98" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-98', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line">                <tt class="py-string">"SELECT value FROM meta_data WHERE key = 'filename'"</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-99" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchone" class="py-name" href="#" onclick="return doclink('link-99', 'fetchone', 'link-96');">fetchone</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line">            <tt class="py-comment"># Compute absolute path of the original maf file</tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">isabs</tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line">                <tt class="py-comment"># It was already stored as absolute</tt> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line">                <tt class="py-name">tmp_mafpath</tt> <tt class="py-op">=</tt> <tt class="py-name">filename</tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">                <tt class="py-comment"># It should otherwise have been stored as relative to the index</tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line">                <tt class="py-comment"># Would be stored with Unix / path separator, so convert</tt> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line">                <tt class="py-comment"># it to the local OS path separator here:</tt> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">                <tt class="py-name">tmp_mafpath</tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_relative_path</tt><tt class="py-op">,</tt> <tt class="py-name">filename</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"/"</tt><tt class="py-op">,</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">sep</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">tmp_mafpath</tt> <tt class="py-op">!=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">abspath</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_file</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">                <tt class="py-comment"># Original and given absolute paths differ.</tt> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Index uses a different file (%s != %s)"</tt> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line">                                 <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_file</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line"> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line">            <tt class="py-name">db_target</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-100" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-100', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line">                <tt class="py-string">"SELECT value FROM meta_data WHERE key = 'target_seqname'"</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-101" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchone" class="py-name" href="#" onclick="return doclink('link-101', 'fetchone', 'link-96');">fetchone</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">db_target</tt> <tt class="py-op">!=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">:</tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Provided database indexed for %s, expected %s"</tt> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line">                                 <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">db_target</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line"> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line">            <tt class="py-name">record_count</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-102" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-102', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line">                <tt class="py-string">"SELECT value FROM meta_data WHERE key = 'record_count'"</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-103" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchone" class="py-name" href="#" onclick="return doclink('link-103', 'fetchone', 'link-96');">fetchone</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">record_count</tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unfinished/partial database provided"</tt><tt class="py-op">)</tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line"> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line">            <tt class="py-name">records_found</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-104" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-104', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line">                <tt class="py-string">"SELECT COUNT(*) FROM offset_data"</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-105" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchone" class="py-name" href="#" onclick="return doclink('link-105', 'fetchone', 'link-96');">fetchone</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">records_found</tt> <tt class="py-op">!=</tt> <tt class="py-name">record_count</tt><tt class="py-op">:</tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Expected %s records, found %s.  Corrupt index?"</tt> </tt>
<a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line">                                 <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">record_count</tt><tt class="py-op">,</tt> <tt class="py-name">records_found</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line"> </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">records_found</tt> </tt>
<a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line"> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-op">(</tt><tt class="py-name">_sqlite</tt><tt class="py-op">.</tt><tt class="py-name">OperationalError</tt><tt class="py-op">,</tt> <tt class="py-name">_sqlite</tt><tt class="py-op">.</tt><tt class="py-name">DatabaseError</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">err</tt><tt class="py-op">:</tt> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Problem with SQLite database: %s"</tt> <tt class="py-op">%</tt> <tt class="py-name">err</tt><tt class="py-op">)</tt> </tt>
</div><a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line"> </tt>
<a name="MafIndex.__make_new_index"></a><div id="MafIndex.__make_new_index-def"><a name="L316"></a><tt class="py-lineno">316</tt> <a class="py-toggle" href="#" id="MafIndex.__make_new_index-toggle" onclick="return toggle('MafIndex.__make_new_index');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafIndex-class.html#__make_new_index">__make_new_index</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIndex.__make_new_index-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafIndex.__make_new_index-expanded"><a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line">        <tt class="py-docstring">"""Read MAF file and generate SQLite index (PRIVATE)."""</tt> </tt>
<a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">        <tt class="py-comment"># make the tables</tt> </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-106" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-106', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"CREATE TABLE meta_data (key TEXT, value TEXT);"</tt><tt class="py-op">)</tt> </tt>
<a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-107" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-107', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"INSERT INTO meta_data (key, value) VALUES ('version', 1);"</tt><tt class="py-op">)</tt> </tt>
<a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-108" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-108', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"INSERT INTO meta_data (key, value) VALUES ('record_count', -1);"</tt><tt class="py-op">)</tt> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-109" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-109', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"INSERT INTO meta_data (key, value) VALUES ('target_seqname', '%s');"</tt> <tt class="py-op">%</tt> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">                          <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line">        <tt class="py-comment"># Determine whether to store maf file as relative to the index or absolute</tt> </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line">        <tt class="py-comment"># See https://github.com/biopython/biopython/pull/381</tt> </tt>
<a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">isabs</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_file</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">isabs</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_index_filename</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line">            <tt class="py-comment"># Since the user gave both maf file and index as relative paths,</tt> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line">            <tt class="py-comment"># we will store the maf file relative to the index.</tt> </tt>
<a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line">            <tt class="py-comment"># Note for cross platform use (e.g. shared drive over SAMBA),</tt> </tt>
<a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line">            <tt class="py-comment"># convert any Windows slash into Unix style for rel paths.</tt> </tt>
<a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line">            <tt class="py-comment"># example: ucsc_mm9_chr10.maf</tt> </tt>
<a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line">            <tt class="py-name">mafpath</tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">relpath</tt><tt class="py-op">(</tt> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_file</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_relative_path</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">sep</tt><tt class="py-op">,</tt> <tt class="py-string">"/"</tt><tt class="py-op">)</tt> </tt>
<a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-op">(</tt><tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">dirname</tt><tt class="py-op">(</tt><tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">abspath</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_file</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> </tt>
<a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line">              <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">sep</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-110" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-110', 'startswith', 'link-51');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_relative_path</tt> <tt class="py-op">+</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">sep</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line">            <tt class="py-comment"># Since maf file is in same directory or sub directory,</tt> </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line">            <tt class="py-comment"># might as well make this into a relative path:</tt> </tt>
<a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line">            <tt class="py-name">mafpath</tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">relpath</tt><tt class="py-op">(</tt> </tt>
<a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_file</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_relative_path</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">sep</tt><tt class="py-op">,</tt> <tt class="py-string">"/"</tt><tt class="py-op">)</tt> </tt>
<a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line">            <tt class="py-comment"># Default to storing as an absolute path</tt> </tt>
<a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line">            <tt class="py-comment"># example: /home/bli/src/biopython/Tests/MAF/ucsc_mm9_chr10.maf</tt> </tt>
<a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line">            <tt class="py-name">mafpath</tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">abspath</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_file</tt><tt class="py-op">)</tt> </tt>
<a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-111" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-111', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"INSERT INTO meta_data (key, value) VALUES ('filename', '%s');"</tt> <tt class="py-op">%</tt> </tt>
<a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line">                          <tt class="py-op">(</tt><tt class="py-name">mafpath</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-112" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-112', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"CREATE TABLE offset_data (bin INTEGER, start INTEGER, end INTEGER, offset INTEGER);"</tt><tt class="py-op">)</tt> </tt>
<a name="L347"></a><tt class="py-lineno">347</tt>  <tt class="py-line"> </tt>
<a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line">        <tt class="py-name">insert_count</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line"> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line">        <tt class="py-comment"># iterate over the entire file and insert in batches</tt> </tt>
<a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line">        <tt class="py-name">mafindex_func</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-113" class="py-name" targets="Method Bio.AlignIO.MafIO.MafIndex.__maf_indexer()=Bio.AlignIO.MafIO.MafIndex-class.html#__maf_indexer"><a title="Bio.AlignIO.MafIO.MafIndex.__maf_indexer" class="py-name" href="#" onclick="return doclink('link-113', '__maf_indexer', 'link-113');">__maf_indexer</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L352"></a><tt class="py-lineno">352</tt>  <tt class="py-line"> </tt>
<a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line">            <tt class="py-name">batch</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">islice</tt><tt class="py-op">(</tt><tt class="py-name">mafindex_func</tt><tt class="py-op">,</tt> <tt class="py-number">100</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">batch</tt><tt class="py-op">:</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line"> </tt>
<a name="L358"></a><tt class="py-lineno">358</tt>  <tt class="py-line">            <tt class="py-comment"># batch is made from self.__maf_indexer(),</tt> </tt>
<a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-114" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.executemany()=BioSQL.BioSeqDatabase.Adaptor-class.html#executemany,Method BioSQL.BioSeqDatabase._CursorWrapper.executemany()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#executemany,Method BioSQL.DBUtils.Generic_dbutils.executemany()=BioSQL.DBUtils.Generic_dbutils-class.html#executemany,Method BioSQL.DBUtils.Sqlite_dbutils.executemany()=BioSQL.DBUtils.Sqlite_dbutils-class.html#executemany"><a title="BioSQL.BioSeqDatabase.Adaptor.executemany
BioSQL.BioSeqDatabase._CursorWrapper.executemany
BioSQL.DBUtils.Generic_dbutils.executemany
BioSQL.DBUtils.Sqlite_dbutils.executemany" class="py-name" href="#" onclick="return doclink('link-114', 'executemany', 'link-114');">executemany</a></tt><tt class="py-op">(</tt> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line">                <tt class="py-string">"INSERT INTO offset_data (bin, start, end, offset) VALUES (?,?,?,?);"</tt><tt class="py-op">,</tt> <tt class="py-name">batch</tt><tt class="py-op">)</tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-115" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.commit()=BioSQL.BioSeqDatabase.Adaptor-class.html#commit,Method BioSQL.BioSeqDatabase.DBServer.commit()=BioSQL.BioSeqDatabase.DBServer-class.html#commit"><a title="BioSQL.BioSeqDatabase.Adaptor.commit
BioSQL.BioSeqDatabase.DBServer.commit" class="py-name" href="#" onclick="return doclink('link-115', 'commit', 'link-115');">commit</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line">            <tt class="py-name">insert_count</tt> <tt class="py-op">+=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">batch</tt><tt class="py-op">)</tt> </tt>
<a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line"> </tt>
<a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line">        <tt class="py-comment"># then make indexes on the relevant fields</tt> </tt>
<a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-116" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-116', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"CREATE INDEX IF NOT EXISTS bin_index ON offset_data(bin);"</tt><tt class="py-op">)</tt> </tt>
<a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-117" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-117', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"CREATE INDEX IF NOT EXISTS start_index ON offset_data(start);"</tt><tt class="py-op">)</tt> </tt>
<a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-118" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-118', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"CREATE INDEX IF NOT EXISTS end_index ON offset_data(end);"</tt><tt class="py-op">)</tt> </tt>
<a name="L368"></a><tt class="py-lineno">368</tt>  <tt class="py-line"> </tt>
<a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-119" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-119', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line">            <tt class="py-string">"UPDATE meta_data SET value = '%s' WHERE key = 'record_count'"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">insert_count</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line"> </tt>
<a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-120" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.commit
BioSQL.BioSeqDatabase.DBServer.commit" class="py-name" href="#" onclick="return doclink('link-120', 'commit', 'link-115');">commit</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line"> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">insert_count</tt> </tt>
</div><a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line"> </tt>
<a name="MafIndex.__maf_indexer"></a><div id="MafIndex.__maf_indexer-def"><a name="L376"></a><tt class="py-lineno">376</tt> <a class="py-toggle" href="#" id="MafIndex.__maf_indexer-toggle" onclick="return toggle('MafIndex.__maf_indexer');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafIndex-class.html#__maf_indexer">__maf_indexer</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIndex.__maf_indexer-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafIndex.__maf_indexer-expanded"><a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return index information for each bundle (PRIVATE).</tt> </tt>
<a name="L378"></a><tt class="py-lineno">378</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L379"></a><tt class="py-lineno">379</tt>  <tt class="py-line"><tt class="py-docstring">        Yields index information for each bundle in the form of</tt> </tt>
<a name="L380"></a><tt class="py-lineno">380</tt>  <tt class="py-line"><tt class="py-docstring">        (bin, start, end, offset) tuples where start and end are</tt> </tt>
<a name="L381"></a><tt class="py-lineno">381</tt>  <tt class="py-line"><tt class="py-docstring">        0-based inclusive coordinates.</tt> </tt>
<a name="L382"></a><tt class="py-lineno">382</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L383"></a><tt class="py-lineno">383</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_fp</tt><tt class="py-op">.</tt><tt id="link-121" class="py-name" targets="Method Bio.File.UndoHandle.readline()=Bio.File.UndoHandle-class.html#readline,Method Bio.bgzf.BgzfReader.readline()=Bio.bgzf.BgzfReader-class.html#readline"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-121', 'readline', 'link-121');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L384"></a><tt class="py-lineno">384</tt>  <tt class="py-line"> </tt>
<a name="L385"></a><tt class="py-lineno">385</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L386"></a><tt class="py-lineno">386</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-122" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-122', 'startswith', 'link-51');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"a"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L387"></a><tt class="py-lineno">387</tt>  <tt class="py-line">                <tt class="py-comment"># note the offset</tt> </tt>
<a name="L388"></a><tt class="py-lineno">388</tt>  <tt class="py-line">                <tt class="py-name">offset</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_fp</tt><tt class="py-op">.</tt><tt id="link-123" class="py-name" targets="Method Bio.File.UndoHandle.tell()=Bio.File.UndoHandle-class.html#tell,Method Bio.SeqIO.SffIO._AddTellHandle.tell()=Bio.SeqIO.SffIO._AddTellHandle-class.html#tell,Method Bio.bgzf.BgzfReader.tell()=Bio.bgzf.BgzfReader-class.html#tell,Method Bio.bgzf.BgzfWriter.tell()=Bio.bgzf.BgzfWriter-class.html#tell"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-123', 'tell', 'link-123');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L389"></a><tt class="py-lineno">389</tt>  <tt class="py-line"> </tt>
<a name="L390"></a><tt class="py-lineno">390</tt>  <tt class="py-line">                <tt class="py-comment"># search the following lines for a match to target_seqname</tt> </tt>
<a name="L391"></a><tt class="py-lineno">391</tt>  <tt class="py-line">                <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L392"></a><tt class="py-lineno">392</tt>  <tt class="py-line">                    <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_fp</tt><tt class="py-op">.</tt><tt id="link-124" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-124', 'readline', 'link-121');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L393"></a><tt class="py-lineno">393</tt>  <tt class="py-line"> </tt>
<a name="L394"></a><tt class="py-lineno">394</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-125" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-125', 'strip', 'link-52');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-126" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-126', 'startswith', 'link-51');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"a"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L395"></a><tt class="py-lineno">395</tt>  <tt class="py-line">                        <tt class="py-comment"># Empty line or new alignment record</tt> </tt>
<a name="L396"></a><tt class="py-lineno">396</tt>  <tt class="py-line">                        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Target for indexing (%s) not found in this bundle"</tt> </tt>
<a name="L397"></a><tt class="py-lineno">397</tt>  <tt class="py-line">                                         <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L398"></a><tt class="py-lineno">398</tt>  <tt class="py-line">                    <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-127" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-127', 'startswith', 'link-51');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"s"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L399"></a><tt class="py-lineno">399</tt>  <tt class="py-line">                        <tt class="py-comment"># s (literal), src (ID), start, size, strand, srcSize, text (sequence)</tt> </tt>
<a name="L400"></a><tt class="py-lineno">400</tt>  <tt class="py-line">                        <tt class="py-name">line_split</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-128" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-128', 'strip', 'link-52');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-129" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-129', 'split', 'link-53');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L401"></a><tt class="py-lineno">401</tt>  <tt class="py-line"> </tt>
<a name="L402"></a><tt class="py-lineno">402</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">:</tt> </tt>
<a name="L403"></a><tt class="py-lineno">403</tt>  <tt class="py-line">                            <tt id="link-130" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.start=Bio.SeqFeature.CompoundLocation-class.html#start,Variable Bio.SeqFeature.FeatureLocation.start=Bio.SeqFeature.FeatureLocation-class.html#start"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-130', 'start', 'link-130');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L404"></a><tt class="py-lineno">404</tt>  <tt class="py-line">                            <tt id="link-131" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.end=Bio.SeqFeature.CompoundLocation-class.html#end,Variable Bio.SeqFeature.FeatureLocation.end=Bio.SeqFeature.FeatureLocation-class.html#end"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-131', 'end', 'link-131');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L405"></a><tt class="py-lineno">405</tt>  <tt class="py-line"> </tt>
<a name="L406"></a><tt class="py-lineno">406</tt>  <tt class="py-line">                            <tt class="py-keyword">if</tt> <tt id="link-132" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-132', 'end', 'link-131');">end</a></tt> <tt class="py-op">-</tt> <tt id="link-133" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-133', 'start', 'link-130');">start</a></tt> <tt class="py-op">!=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"-"</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L407"></a><tt class="py-lineno">407</tt>  <tt class="py-line">                                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid length for target coordinates (expected %s, found %s)"</tt> <tt class="py-op">%</tt> </tt>
<a name="L408"></a><tt class="py-lineno">408</tt>  <tt class="py-line">                                                 <tt class="py-op">(</tt><tt id="link-134" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-134', 'end', 'link-131');">end</a></tt> <tt class="py-op">-</tt> <tt id="link-135" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-135', 'start', 'link-130');">start</a></tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"-"</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L409"></a><tt class="py-lineno">409</tt>  <tt class="py-line"> </tt>
<a name="L410"></a><tt class="py-lineno">410</tt>  <tt class="py-line">                            <tt class="py-keyword">yield</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-136" class="py-name" targets="Static Method Bio.AlignIO.MafIO.MafIndex._ucscbin()=Bio.AlignIO.MafIO.MafIndex-class.html#_ucscbin"><a title="Bio.AlignIO.MafIO.MafIndex._ucscbin" class="py-name" href="#" onclick="return doclink('link-136', '_ucscbin', 'link-136');">_ucscbin</a></tt><tt class="py-op">(</tt><tt id="link-137" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-137', 'start', 'link-130');">start</a></tt><tt class="py-op">,</tt> <tt id="link-138" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-138', 'end', 'link-131');">end</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-139" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-139', 'start', 'link-130');">start</a></tt><tt class="py-op">,</tt> <tt id="link-140" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-140', 'end', 'link-131');">end</a></tt><tt class="py-op">,</tt> <tt class="py-name">offset</tt><tt class="py-op">)</tt> </tt>
<a name="L411"></a><tt class="py-lineno">411</tt>  <tt class="py-line"> </tt>
<a name="L412"></a><tt class="py-lineno">412</tt>  <tt class="py-line">                            <tt class="py-keyword">break</tt> </tt>
<a name="L413"></a><tt class="py-lineno">413</tt>  <tt class="py-line"> </tt>
<a name="L414"></a><tt class="py-lineno">414</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_fp</tt><tt class="py-op">.</tt><tt id="link-141" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-141', 'readline', 'link-121');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L415"></a><tt class="py-lineno">415</tt>  <tt class="py-line"> </tt>
<a name="L416"></a><tt class="py-lineno">416</tt>  <tt class="py-line">    <tt class="py-comment"># TODO: check coordinate correctness for the two bin-related static methods</tt> </tt>
<a name="L417"></a><tt class="py-lineno">417</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">staticmethod</tt> </tt>
<a name="MafIndex._region2bin"></a><div id="MafIndex._region2bin-def"><a name="L418"></a><tt class="py-lineno">418</tt> <a class="py-toggle" href="#" id="MafIndex._region2bin-toggle" onclick="return toggle('MafIndex._region2bin');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafIndex-class.html#_region2bin">_region2bin</a><tt class="py-op">(</tt><tt class="py-param">start</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIndex._region2bin-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafIndex._region2bin-expanded"><a name="L419"></a><tt class="py-lineno">419</tt>  <tt class="py-line">        <tt class="py-docstring">"""Find bins that a region may belong to (PRIVATE).</tt> </tt>
<a name="L420"></a><tt class="py-lineno">420</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L421"></a><tt class="py-lineno">421</tt>  <tt class="py-line"><tt class="py-docstring">        Converts a region to a list of bins that it may belong to, including largest</tt> </tt>
<a name="L422"></a><tt class="py-lineno">422</tt>  <tt class="py-line"><tt class="py-docstring">        and smallest bins.</tt> </tt>
<a name="L423"></a><tt class="py-lineno">423</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L424"></a><tt class="py-lineno">424</tt>  <tt class="py-line">        <tt class="py-name">bins</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L425"></a><tt class="py-lineno">425</tt>  <tt class="py-line"> </tt>
<a name="L426"></a><tt class="py-lineno">426</tt>  <tt class="py-line">        <tt class="py-name">bins</tt><tt class="py-op">.</tt><tt id="link-142" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.extend()=Bio.Align.MultipleSeqAlignment-class.html#extend,Method Bio.SCOP.Raf.SeqMap.extend()=Bio.SCOP.Raf.SeqMap-class.html#extend,Method Bio.Seq.MutableSeq.extend()=Bio.Seq.MutableSeq-class.html#extend"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-142', 'extend', 'link-142');">extend</a></tt><tt class="py-op">(</tt><tt id="link-143" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-143', 'range', 'link-143');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">1</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt id="link-144" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-144', 'start', 'link-130');">start</a></tt> <tt class="py-op">&gt;&gt;</tt> <tt class="py-number">26</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">2</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt class="py-op">(</tt><tt id="link-145" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-145', 'end', 'link-131');">end</a></tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;&gt;</tt> <tt class="py-number">26</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L427"></a><tt class="py-lineno">427</tt>  <tt class="py-line">        <tt class="py-name">bins</tt><tt class="py-op">.</tt><tt id="link-146" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-146', 'extend', 'link-142');">extend</a></tt><tt class="py-op">(</tt><tt id="link-147" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-147', 'range', 'link-143');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">9</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt id="link-148" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-148', 'start', 'link-130');">start</a></tt> <tt class="py-op">&gt;&gt;</tt> <tt class="py-number">23</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">10</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt class="py-op">(</tt><tt id="link-149" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-149', 'end', 'link-131');">end</a></tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;&gt;</tt> <tt class="py-number">23</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L428"></a><tt class="py-lineno">428</tt>  <tt class="py-line">        <tt class="py-name">bins</tt><tt class="py-op">.</tt><tt id="link-150" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-150', 'extend', 'link-142');">extend</a></tt><tt class="py-op">(</tt><tt id="link-151" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-151', 'range', 'link-143');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">73</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt id="link-152" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-152', 'start', 'link-130');">start</a></tt> <tt class="py-op">&gt;&gt;</tt> <tt class="py-number">20</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">74</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt class="py-op">(</tt><tt id="link-153" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-153', 'end', 'link-131');">end</a></tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;&gt;</tt> <tt class="py-number">20</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L429"></a><tt class="py-lineno">429</tt>  <tt class="py-line">        <tt class="py-name">bins</tt><tt class="py-op">.</tt><tt id="link-154" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-154', 'extend', 'link-142');">extend</a></tt><tt class="py-op">(</tt><tt id="link-155" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-155', 'range', 'link-143');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">585</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt id="link-156" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-156', 'start', 'link-130');">start</a></tt> <tt class="py-op">&gt;&gt;</tt> <tt class="py-number">17</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">586</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt class="py-op">(</tt><tt id="link-157" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-157', 'end', 'link-131');">end</a></tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;&gt;</tt> <tt class="py-number">17</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L430"></a><tt class="py-lineno">430</tt>  <tt class="py-line"> </tt>
<a name="L431"></a><tt class="py-lineno">431</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-158" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-158', 'set', 'link-36');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">bins</tt><tt class="py-op">)</tt> </tt>
</div><a name="L432"></a><tt class="py-lineno">432</tt>  <tt class="py-line"> </tt>
<a name="L433"></a><tt class="py-lineno">433</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">staticmethod</tt> </tt>
<a name="MafIndex._ucscbin"></a><div id="MafIndex._ucscbin-def"><a name="L434"></a><tt class="py-lineno">434</tt> <a class="py-toggle" href="#" id="MafIndex._ucscbin-toggle" onclick="return toggle('MafIndex._ucscbin');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafIndex-class.html#_ucscbin">_ucscbin</a><tt class="py-op">(</tt><tt class="py-param">start</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIndex._ucscbin-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafIndex._ucscbin-expanded"><a name="L435"></a><tt class="py-lineno">435</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the smallest bin a given region will fit into (PRIVATE).</tt> </tt>
<a name="L436"></a><tt class="py-lineno">436</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L437"></a><tt class="py-lineno">437</tt>  <tt class="py-line"><tt class="py-docstring">        Adapted from http://genomewiki.ucsc.edu/index.php/Bin_indexing_system</tt> </tt>
<a name="L438"></a><tt class="py-lineno">438</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L439"></a><tt class="py-lineno">439</tt>  <tt class="py-line">        <tt class="py-name">bin_offsets</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-number">512</tt> <tt class="py-op">+</tt> <tt class="py-number">64</tt> <tt class="py-op">+</tt> <tt class="py-number">8</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">64</tt> <tt class="py-op">+</tt> <tt class="py-number">8</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">8</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L440"></a><tt class="py-lineno">440</tt>  <tt class="py-line"> </tt>
<a name="L441"></a><tt class="py-lineno">441</tt>  <tt class="py-line">        <tt class="py-name">_bin_first_shift</tt> <tt class="py-op">=</tt> <tt class="py-number">17</tt> </tt>
<a name="L442"></a><tt class="py-lineno">442</tt>  <tt class="py-line">        <tt class="py-name">_bin_next_shift</tt> <tt class="py-op">=</tt> <tt class="py-number">3</tt> </tt>
<a name="L443"></a><tt class="py-lineno">443</tt>  <tt class="py-line"> </tt>
<a name="L444"></a><tt class="py-lineno">444</tt>  <tt class="py-line">        <tt class="py-name">start_bin</tt> <tt class="py-op">=</tt> <tt id="link-159" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-159', 'start', 'link-130');">start</a></tt> </tt>
<a name="L445"></a><tt class="py-lineno">445</tt>  <tt class="py-line">        <tt class="py-name">end_bin</tt> <tt class="py-op">=</tt> <tt id="link-160" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-160', 'end', 'link-131');">end</a></tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt> </tt>
<a name="L446"></a><tt class="py-lineno">446</tt>  <tt class="py-line"> </tt>
<a name="L447"></a><tt class="py-lineno">447</tt>  <tt class="py-line">        <tt class="py-name">start_bin</tt> <tt class="py-op">&gt;&gt;=</tt> <tt class="py-name">_bin_first_shift</tt> </tt>
<a name="L448"></a><tt class="py-lineno">448</tt>  <tt class="py-line">        <tt class="py-name">end_bin</tt> <tt class="py-op">&gt;&gt;=</tt> <tt class="py-name">_bin_first_shift</tt> </tt>
<a name="L449"></a><tt class="py-lineno">449</tt>  <tt class="py-line"> </tt>
<a name="L450"></a><tt class="py-lineno">450</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">bin_offset</tt> <tt class="py-keyword">in</tt> <tt class="py-name">bin_offsets</tt><tt class="py-op">:</tt> </tt>
<a name="L451"></a><tt class="py-lineno">451</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">start_bin</tt> <tt class="py-op">==</tt> <tt class="py-name">end_bin</tt><tt class="py-op">:</tt> </tt>
<a name="L452"></a><tt class="py-lineno">452</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">bin_offset</tt> <tt class="py-op">+</tt> <tt class="py-name">start_bin</tt> </tt>
<a name="L453"></a><tt class="py-lineno">453</tt>  <tt class="py-line">            <tt class="py-name">start_bin</tt> <tt class="py-op">&gt;&gt;=</tt> <tt class="py-name">_bin_next_shift</tt> </tt>
<a name="L454"></a><tt class="py-lineno">454</tt>  <tt class="py-line">            <tt class="py-name">end_bin</tt> <tt class="py-op">&gt;&gt;=</tt> <tt class="py-name">_bin_next_shift</tt> </tt>
<a name="L455"></a><tt class="py-lineno">455</tt>  <tt class="py-line"> </tt>
<a name="L456"></a><tt class="py-lineno">456</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-number">0</tt> </tt>
</div><a name="L457"></a><tt class="py-lineno">457</tt>  <tt class="py-line"> </tt>
<a name="MafIndex._get_record"></a><div id="MafIndex._get_record-def"><a name="L458"></a><tt class="py-lineno">458</tt> <a class="py-toggle" href="#" id="MafIndex._get_record-toggle" onclick="return toggle('MafIndex._get_record');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafIndex-class.html#_get_record">_get_record</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">offset</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIndex._get_record-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafIndex._get_record-expanded"><a name="L459"></a><tt class="py-lineno">459</tt>  <tt class="py-line">        <tt class="py-docstring">"""Retrieve a single MAF record located at the offset provided (PRIVATE)."""</tt> </tt>
<a name="L460"></a><tt class="py-lineno">460</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_fp</tt><tt class="py-op">.</tt><tt id="link-161" class="py-name" targets="Method Bio.File.UndoHandle.seek()=Bio.File.UndoHandle-class.html#seek,Method Bio.SeqIO.SffIO._AddTellHandle.seek()=Bio.SeqIO.SffIO._AddTellHandle-class.html#seek,Method Bio.bgzf.BgzfReader.seek()=Bio.bgzf.BgzfReader-class.html#seek"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-161', 'seek', 'link-161');">seek</a></tt><tt class="py-op">(</tt><tt class="py-name">offset</tt><tt class="py-op">)</tt> </tt>
<a name="L461"></a><tt class="py-lineno">461</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-162" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.Nexus.StandardData.StandardData.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-162', 'next', 'link-50');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_mafiter</tt><tt class="py-op">)</tt> </tt>
</div><a name="L462"></a><tt class="py-lineno">462</tt>  <tt class="py-line"> </tt>
<a name="MafIndex.search"></a><div id="MafIndex.search-def"><a name="L463"></a><tt class="py-lineno">463</tt> <a class="py-toggle" href="#" id="MafIndex.search-toggle" onclick="return toggle('MafIndex.search');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafIndex-class.html#search">search</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">starts</tt><tt class="py-op">,</tt> <tt class="py-param">ends</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIndex.search-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafIndex.search-expanded"><a name="L464"></a><tt class="py-lineno">464</tt>  <tt class="py-line">        <tt class="py-docstring">"""Search index database for MAF records overlapping ranges provided.</tt> </tt>
<a name="L465"></a><tt class="py-lineno">465</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L466"></a><tt class="py-lineno">466</tt>  <tt class="py-line"><tt class="py-docstring">        Returns *MultipleSeqAlignment* results in order by start, then end, then</tt> </tt>
<a name="L467"></a><tt class="py-lineno">467</tt>  <tt class="py-line"><tt class="py-docstring">        internal offset field.</tt> </tt>
<a name="L468"></a><tt class="py-lineno">468</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L469"></a><tt class="py-lineno">469</tt>  <tt class="py-line"><tt class="py-docstring">        *starts* should be a list of 0-based start coordinates of segments in the reference.</tt> </tt>
<a name="L470"></a><tt class="py-lineno">470</tt>  <tt class="py-line"><tt class="py-docstring">        *ends* should be the list of the corresponding segment ends</tt> </tt>
<a name="L471"></a><tt class="py-lineno">471</tt>  <tt class="py-line"><tt class="py-docstring">        (in the half-open UCSC convention:</tt> </tt>
<a name="L472"></a><tt class="py-lineno">472</tt>  <tt class="py-line"><tt class="py-docstring">        http://genome.ucsc.edu/blog/the-ucsc-genome-browser-coordinate-counting-systems/).</tt> </tt>
<a name="L473"></a><tt class="py-lineno">473</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L474"></a><tt class="py-lineno">474</tt>  <tt class="py-line">        <tt class="py-comment"># verify the provided exon coordinates</tt> </tt>
<a name="L475"></a><tt class="py-lineno">475</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">starts</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">ends</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L476"></a><tt class="py-lineno">476</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Every position in starts must have a match in ends"</tt><tt class="py-op">)</tt> </tt>
<a name="L477"></a><tt class="py-lineno">477</tt>  <tt class="py-line"> </tt>
<a name="L478"></a><tt class="py-lineno">478</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">exonstart</tt><tt class="py-op">,</tt> <tt class="py-name">exonend</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">starts</tt><tt class="py-op">,</tt> <tt class="py-name">ends</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L479"></a><tt class="py-lineno">479</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">exonstart</tt> <tt class="py-op">&gt;=</tt> <tt class="py-name">exonend</tt><tt class="py-op">:</tt> </tt>
<a name="L480"></a><tt class="py-lineno">480</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Exon coordinates invalid (%s &gt;= %s)"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">exonstart</tt><tt class="py-op">,</tt> <tt class="py-name">exonend</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L481"></a><tt class="py-lineno">481</tt>  <tt class="py-line">        <tt class="py-name">con</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt> </tt>
<a name="L482"></a><tt class="py-lineno">482</tt>  <tt class="py-line"> </tt>
<a name="L483"></a><tt class="py-lineno">483</tt>  <tt class="py-line">        <tt class="py-comment"># Keep track of what blocks have already been yielded</tt> </tt>
<a name="L484"></a><tt class="py-lineno">484</tt>  <tt class="py-line">        <tt class="py-comment"># in order to avoid duplicating them</tt> </tt>
<a name="L485"></a><tt class="py-lineno">485</tt>  <tt class="py-line">        <tt class="py-comment"># (see https://github.com/biopython/biopython/issues/1083)</tt> </tt>
<a name="L486"></a><tt class="py-lineno">486</tt>  <tt class="py-line">        <tt class="py-name">yielded_rec_coords</tt> <tt class="py-op">=</tt> <tt id="link-163" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-163', 'set', 'link-36');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L487"></a><tt class="py-lineno">487</tt>  <tt class="py-line">        <tt class="py-comment"># search for every exon</tt> </tt>
<a name="L488"></a><tt class="py-lineno">488</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">exonstart</tt><tt class="py-op">,</tt> <tt class="py-name">exonend</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">starts</tt><tt class="py-op">,</tt> <tt class="py-name">ends</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L489"></a><tt class="py-lineno">489</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L490"></a><tt class="py-lineno">490</tt>  <tt class="py-line">                <tt class="py-name">possible_bins</tt> <tt class="py-op">=</tt> <tt class="py-string">", "</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt id="link-164" class="py-name" targets="Method Bio.SearchIO._model.hit.Hit.map()=Bio.SearchIO._model.hit.Hit-class.html#map"><a title="Bio.SearchIO._model.hit.Hit.map" class="py-name" href="#" onclick="return doclink('link-164', 'map', 'link-164');">map</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-165" class="py-name" targets="Static Method Bio.AlignIO.MafIO.MafIndex._region2bin()=Bio.AlignIO.MafIO.MafIndex-class.html#_region2bin"><a title="Bio.AlignIO.MafIO.MafIndex._region2bin" class="py-name" href="#" onclick="return doclink('link-165', '_region2bin', 'link-165');">_region2bin</a></tt><tt class="py-op">(</tt><tt class="py-name">exonstart</tt><tt class="py-op">,</tt> <tt class="py-name">exonend</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L491"></a><tt class="py-lineno">491</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">TypeError</tt><tt class="py-op">:</tt> </tt>
<a name="L492"></a><tt class="py-lineno">492</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"Exon coordinates must be integers "</tt> </tt>
<a name="L493"></a><tt class="py-lineno">493</tt>  <tt class="py-line">                                <tt class="py-string">"(start=%d, end=%d)"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">exonstart</tt><tt class="py-op">,</tt> <tt class="py-name">exonend</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L494"></a><tt class="py-lineno">494</tt>  <tt class="py-line"> </tt>
<a name="L495"></a><tt class="py-lineno">495</tt>  <tt class="py-line">            <tt class="py-comment"># https://www.sqlite.org/lang_expr.html</tt> </tt>
<a name="L496"></a><tt class="py-lineno">496</tt>  <tt class="py-line">            <tt class="py-comment"># -----</tt> </tt>
<a name="L497"></a><tt class="py-lineno">497</tt>  <tt class="py-line">            <tt class="py-comment"># The BETWEEN operator</tt> </tt>
<a name="L498"></a><tt class="py-lineno">498</tt>  <tt class="py-line">            <tt class="py-comment">#</tt> </tt>
<a name="L499"></a><tt class="py-lineno">499</tt>  <tt class="py-line">            <tt class="py-comment"># The BETWEEN operator is logically equivalent to a pair of</tt> </tt>
<a name="L500"></a><tt class="py-lineno">500</tt>  <tt class="py-line">            <tt class="py-comment"># comparisons. "x BETWEEN y AND z" is equivalent to "x&gt;=y AND x&lt;=z"</tt> </tt>
<a name="L501"></a><tt class="py-lineno">501</tt>  <tt class="py-line">            <tt class="py-comment"># except that with BETWEEN, the x expression is only evaluated</tt> </tt>
<a name="L502"></a><tt class="py-lineno">502</tt>  <tt class="py-line">            <tt class="py-comment"># once. The precedence of the BETWEEN operator is the same as the</tt> </tt>
<a name="L503"></a><tt class="py-lineno">503</tt>  <tt class="py-line">            <tt class="py-comment"># precedence as operators == and != and LIKE and groups left to</tt> </tt>
<a name="L504"></a><tt class="py-lineno">504</tt>  <tt class="py-line">            <tt class="py-comment"># right.</tt> </tt>
<a name="L505"></a><tt class="py-lineno">505</tt>  <tt class="py-line">            <tt class="py-comment"># -----</tt> </tt>
<a name="L506"></a><tt class="py-lineno">506</tt>  <tt class="py-line"> </tt>
<a name="L507"></a><tt class="py-lineno">507</tt>  <tt class="py-line">            <tt class="py-name">result</tt> <tt class="py-op">=</tt> <tt class="py-name">con</tt><tt class="py-op">.</tt><tt id="link-166" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-166', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"SELECT DISTINCT start, end, offset FROM "</tt> </tt>
<a name="L508"></a><tt class="py-lineno">508</tt>  <tt class="py-line">                                 <tt class="py-string">"offset_data WHERE bin IN (%s) AND (end BETWEEN %s AND %s "</tt> </tt>
<a name="L509"></a><tt class="py-lineno">509</tt>  <tt class="py-line">                                 <tt class="py-string">"OR %s BETWEEN start AND end) ORDER BY start, end, "</tt> </tt>
<a name="L510"></a><tt class="py-lineno">510</tt>  <tt class="py-line">                                 <tt class="py-string">"offset ASC;"</tt> </tt>
<a name="L511"></a><tt class="py-lineno">511</tt>  <tt class="py-line">                                 <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">possible_bins</tt><tt class="py-op">,</tt> <tt class="py-name">exonstart</tt><tt class="py-op">,</tt> <tt class="py-name">exonend</tt><tt class="py-op">,</tt> <tt class="py-name">exonend</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L512"></a><tt class="py-lineno">512</tt>  <tt class="py-line"> </tt>
<a name="L513"></a><tt class="py-lineno">513</tt>  <tt class="py-line">            <tt class="py-name">rows</tt> <tt class="py-op">=</tt> <tt class="py-name">result</tt><tt class="py-op">.</tt><tt id="link-167" class="py-name" targets="Method BioSQL.BioSeqDatabase._CursorWrapper.fetchall()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#fetchall"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchall" class="py-name" href="#" onclick="return doclink('link-167', 'fetchall', 'link-167');">fetchall</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L514"></a><tt class="py-lineno">514</tt>  <tt class="py-line"> </tt>
<a name="L515"></a><tt class="py-lineno">515</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">rec_start</tt><tt class="py-op">,</tt> <tt class="py-name">rec_end</tt><tt class="py-op">,</tt> <tt class="py-name">offset</tt> <tt class="py-keyword">in</tt> <tt class="py-name">rows</tt><tt class="py-op">:</tt> </tt>
<a name="L516"></a><tt class="py-lineno">516</tt>  <tt class="py-line">                <tt class="py-comment"># Avoid yielding multiple time the same block</tt> </tt>
<a name="L517"></a><tt class="py-lineno">517</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-op">(</tt><tt class="py-name">rec_start</tt><tt class="py-op">,</tt> <tt class="py-name">rec_end</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">yielded_rec_coords</tt><tt class="py-op">:</tt> </tt>
<a name="L518"></a><tt class="py-lineno">518</tt>  <tt class="py-line">                    <tt class="py-keyword">continue</tt> </tt>
<a name="L519"></a><tt class="py-lineno">519</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L520"></a><tt class="py-lineno">520</tt>  <tt class="py-line">                    <tt class="py-name">yielded_rec_coords</tt><tt class="py-op">.</tt><tt id="link-168" class="py-name" targets="Method Bio.Graphics.BasicChromosome._ChromosomeComponent.add()=Bio.Graphics.BasicChromosome._ChromosomeComponent-class.html#add,Method Bio.Nexus.Nexus.StepMatrix.add()=Bio.Nexus.Nexus.StepMatrix-class.html#add,Method Bio.Nexus.Nodes.Chain.add()=Bio.Nexus.Nodes.Chain-class.html#add,Method Bio.PDB.Entity.Entity.add()=Bio.PDB.Entity.Entity-class.html#add,Method Bio.PDB.Residue.DisorderedResidue.add()=Bio.PDB.Residue.DisorderedResidue-class.html#add,Method Bio.PDB.Residue.Residue.add()=Bio.PDB.Residue.Residue-class.html#add,Method Bio.Restriction.Restriction.RestrictionBatch.add()=Bio.Restriction.Restriction.RestrictionBatch-class.html#add"><a title="Bio.Graphics.BasicChromosome._ChromosomeComponent.add
Bio.Nexus.Nexus.StepMatrix.add
Bio.Nexus.Nodes.Chain.add
Bio.PDB.Entity.Entity.add
Bio.PDB.Residue.DisorderedResidue.add
Bio.PDB.Residue.Residue.add
Bio.Restriction.Restriction.RestrictionBatch.add" class="py-name" href="#" onclick="return doclink('link-168', 'add', 'link-168');">add</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">rec_start</tt><tt class="py-op">,</tt> <tt class="py-name">rec_end</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L521"></a><tt class="py-lineno">521</tt>  <tt class="py-line">                <tt class="py-comment"># Iterate through hits, fetching alignments from the MAF file</tt> </tt>
<a name="L522"></a><tt class="py-lineno">522</tt>  <tt class="py-line">                <tt class="py-comment"># and checking to be sure we've retrieved the expected record.</tt> </tt>
<a name="L523"></a><tt class="py-lineno">523</tt>  <tt class="py-line"> </tt>
<a name="L524"></a><tt class="py-lineno">524</tt>  <tt class="py-line">                <tt class="py-name">fetched</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-169" class="py-name" targets="Method Bio.AlignIO.MafIO.MafIndex._get_record()=Bio.AlignIO.MafIO.MafIndex-class.html#_get_record"><a title="Bio.AlignIO.MafIO.MafIndex._get_record" class="py-name" href="#" onclick="return doclink('link-169', '_get_record', 'link-169');">_get_record</a></tt><tt class="py-op">(</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">offset</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L525"></a><tt class="py-lineno">525</tt>  <tt class="py-line"> </tt>
<a name="L526"></a><tt class="py-lineno">526</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">fetched</tt><tt class="py-op">:</tt> </tt>
<a name="L527"></a><tt class="py-lineno">527</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-170" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-170', 'id', 'link-24');">id</a></tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">:</tt> </tt>
<a name="L528"></a><tt class="py-lineno">528</tt>  <tt class="py-line">                        <tt class="py-comment"># start and size come from the maf lines</tt> </tt>
<a name="L529"></a><tt class="py-lineno">529</tt>  <tt class="py-line">                        <tt id="link-171" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-171', 'start', 'link-130');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-172" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-172', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"start"</tt><tt class="py-op">]</tt> </tt>
<a name="L530"></a><tt class="py-lineno">530</tt>  <tt class="py-line">                        <tt id="link-173" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-173', 'end', 'link-131');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-174" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-174', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"start"</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-175" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-175', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"size"</tt><tt class="py-op">]</tt> </tt>
<a name="L531"></a><tt class="py-lineno">531</tt>  <tt class="py-line"> </tt>
<a name="L532"></a><tt class="py-lineno">532</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-op">(</tt><tt id="link-176" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-176', 'start', 'link-130');">start</a></tt> <tt class="py-op">==</tt> <tt class="py-name">rec_start</tt> <tt class="py-keyword">and</tt> <tt id="link-177" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-177', 'end', 'link-131');">end</a></tt> <tt class="py-op">==</tt> <tt class="py-name">rec_end</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L533"></a><tt class="py-lineno">533</tt>  <tt class="py-line">                            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Expected %s-%s @ offset %s, found %s-%s"</tt> <tt class="py-op">%</tt> </tt>
<a name="L534"></a><tt class="py-lineno">534</tt>  <tt class="py-line">                                             <tt class="py-op">(</tt><tt class="py-name">rec_start</tt><tt class="py-op">,</tt> <tt class="py-name">rec_end</tt><tt class="py-op">,</tt> <tt class="py-name">offset</tt><tt class="py-op">,</tt> <tt id="link-178" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-178', 'start', 'link-130');">start</a></tt><tt class="py-op">,</tt> <tt id="link-179" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-179', 'end', 'link-131');">end</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L535"></a><tt class="py-lineno">535</tt>  <tt class="py-line"> </tt>
<a name="L536"></a><tt class="py-lineno">536</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt class="py-name">fetched</tt> </tt>
</div><a name="L537"></a><tt class="py-lineno">537</tt>  <tt class="py-line"> </tt>
<a name="MafIndex.get_spliced"></a><div id="MafIndex.get_spliced-def"><a name="L538"></a><tt class="py-lineno">538</tt> <a class="py-toggle" href="#" id="MafIndex.get_spliced-toggle" onclick="return toggle('MafIndex.get_spliced');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafIndex-class.html#get_spliced">get_spliced</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">starts</tt><tt class="py-op">,</tt> <tt class="py-param">ends</tt><tt class="py-op">,</tt> <tt class="py-param">strand</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIndex.get_spliced-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafIndex.get_spliced-expanded"><a name="L539"></a><tt class="py-lineno">539</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a multiple alignment of the exact sequence range provided.</tt> </tt>
<a name="L540"></a><tt class="py-lineno">540</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L541"></a><tt class="py-lineno">541</tt>  <tt class="py-line"><tt class="py-docstring">        Accepts two lists of start and end positions on target_seqname, representing</tt> </tt>
<a name="L542"></a><tt class="py-lineno">542</tt>  <tt class="py-line"><tt class="py-docstring">        exons to be spliced in silico.  Returns a *MultipleSeqAlignment* of the</tt> </tt>
<a name="L543"></a><tt class="py-lineno">543</tt>  <tt class="py-line"><tt class="py-docstring">        desired sequences spliced together.</tt> </tt>
<a name="L544"></a><tt class="py-lineno">544</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L545"></a><tt class="py-lineno">545</tt>  <tt class="py-line"><tt class="py-docstring">        *starts* should be a list of 0-based start coordinates of segments in the reference.</tt> </tt>
<a name="L546"></a><tt class="py-lineno">546</tt>  <tt class="py-line"><tt class="py-docstring">        *ends* should be the list of the corresponding segment ends</tt> </tt>
<a name="L547"></a><tt class="py-lineno">547</tt>  <tt class="py-line"><tt class="py-docstring">        (in the half-open UCSC convention:</tt> </tt>
<a name="L548"></a><tt class="py-lineno">548</tt>  <tt class="py-line"><tt class="py-docstring">        http://genome.ucsc.edu/blog/the-ucsc-genome-browser-coordinate-counting-systems/).</tt> </tt>
<a name="L549"></a><tt class="py-lineno">549</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L550"></a><tt class="py-lineno">550</tt>  <tt class="py-line"><tt class="py-docstring">        To ask for the alignment portion corresponding to the first 100</tt> </tt>
<a name="L551"></a><tt class="py-lineno">551</tt>  <tt class="py-line"><tt class="py-docstring">        nucleotides of the reference sequence, you would use</tt> </tt>
<a name="L552"></a><tt class="py-lineno">552</tt>  <tt class="py-line"><tt class="py-docstring">        ``search([0], [100])``</tt> </tt>
<a name="L553"></a><tt class="py-lineno">553</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L554"></a><tt class="py-lineno">554</tt>  <tt class="py-line">        <tt class="py-comment"># validate strand</tt> </tt>
<a name="L555"></a><tt class="py-lineno">555</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-180" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-180', 'strand', 'link-19');">strand</a></tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L556"></a><tt class="py-lineno">556</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Strand must be 1 or -1, got %s"</tt> <tt class="py-op">%</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-181" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-181', 'strand', 'link-19');">strand</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L557"></a><tt class="py-lineno">557</tt>  <tt class="py-line"> </tt>
<a name="L558"></a><tt class="py-lineno">558</tt>  <tt class="py-line">        <tt class="py-comment"># pull all alignments that span the desired intervals</tt> </tt>
<a name="L559"></a><tt class="py-lineno">559</tt>  <tt class="py-line">        <tt class="py-name">fetched</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">multiseq</tt> <tt class="py-keyword">for</tt> <tt class="py-name">multiseq</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-182" class="py-name" targets="Method Bio.AlignIO.MafIO.MafIndex.search()=Bio.AlignIO.MafIO.MafIndex-class.html#search,Method Bio.KDTree.KDTree'.KDTree.search()=Bio.KDTree.KDTree%27.KDTree-class.html#search,Method Bio.PDB.NeighborSearch'.NeighborSearch.search()=Bio.PDB.NeighborSearch%27.NeighborSearch-class.html#search,Method Bio.Phylo.TreeConstruction.NNITreeSearcher.search()=Bio.Phylo.TreeConstruction.NNITreeSearcher-class.html#search,Method Bio.Phylo.TreeConstruction.TreeSearcher.search()=Bio.Phylo.TreeConstruction.TreeSearcher-class.html#search,Class Method Bio.Restriction.Restriction.AbstractCut.search()=Bio.Restriction.Restriction.AbstractCut-class.html#search,Method Bio.Restriction.Restriction.RestrictionBatch.search()=Bio.Restriction.Restriction.RestrictionBatch-class.html#search,Function Bio.SCOP.search()=Bio.SCOP-module.html#search,Function Bio.TogoWS.search()=Bio.TogoWS-module.html#search,Method Bio.motifs.Instances.search()=Bio.motifs.Instances-class.html#search,Method Bio.motifs.matrix.PositionSpecificScoringMatrix.search()=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#search"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-182', 'search', 'link-182');">search</a></tt><tt class="py-op">(</tt><tt class="py-name">starts</tt><tt class="py-op">,</tt> <tt class="py-name">ends</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L560"></a><tt class="py-lineno">560</tt>  <tt class="py-line"> </tt>
<a name="L561"></a><tt class="py-lineno">561</tt>  <tt class="py-line">        <tt class="py-comment"># keep track of the expected letter count</tt> </tt>
<a name="L562"></a><tt class="py-lineno">562</tt>  <tt class="py-line">        <tt class="py-comment"># (sum of lengths of [start, end) segments,</tt> </tt>
<a name="L563"></a><tt class="py-lineno">563</tt>  <tt class="py-line">        <tt class="py-comment"># where [start, end) half-open)</tt> </tt>
<a name="L564"></a><tt class="py-lineno">564</tt>  <tt class="py-line">        <tt class="py-name">expected_letters</tt> <tt class="py-op">=</tt> <tt id="link-183" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.sum()=Bio.Nexus.Nexus.StepMatrix-class.html#sum,Method Bio.SubsMat.SeqMat.sum()=Bio.SubsMat.SeqMat-class.html#sum"><a title="Bio.Nexus.Nexus.StepMatrix.sum
Bio.SubsMat.SeqMat.sum" class="py-name" href="#" onclick="return doclink('link-183', 'sum', 'link-183');">sum</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt id="link-184" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-184', 'end', 'link-131');">end</a></tt> <tt class="py-op">-</tt> <tt id="link-185" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-185', 'start', 'link-130');">start</a></tt> <tt class="py-keyword">for</tt> <tt id="link-186" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-186', 'start', 'link-130');">start</a></tt><tt class="py-op">,</tt> <tt id="link-187" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-187', 'end', 'link-131');">end</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">starts</tt><tt class="py-op">,</tt> <tt class="py-name">ends</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L565"></a><tt class="py-lineno">565</tt>  <tt class="py-line"> </tt>
<a name="L566"></a><tt class="py-lineno">566</tt>  <tt class="py-line">        <tt class="py-comment"># if there's no alignment, return filler for the assembly of the length given</tt> </tt>
<a name="L567"></a><tt class="py-lineno">567</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">fetched</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L568"></a><tt class="py-lineno">568</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-188" class="py-name"><a title="Bio.Align.MultipleSeqAlignment" class="py-name" href="#" onclick="return doclink('link-188', 'MultipleSeqAlignment', 'link-11');">MultipleSeqAlignment</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt id="link-189" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-189', 'SeqRecord', 'link-7');">SeqRecord</a></tt><tt class="py-op">(</tt><tt id="link-190" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-190', 'Seq', 'link-4');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"N"</tt> <tt class="py-op">*</tt> <tt class="py-name">expected_letters</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L569"></a><tt class="py-lineno">569</tt>  <tt class="py-line">                                                   <tt id="link-191" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-191', 'id', 'link-24');">id</a></tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L570"></a><tt class="py-lineno">570</tt>  <tt class="py-line"> </tt>
<a name="L571"></a><tt class="py-lineno">571</tt>  <tt class="py-line">        <tt class="py-comment"># find the union of all IDs in these alignments</tt> </tt>
<a name="L572"></a><tt class="py-lineno">572</tt>  <tt class="py-line">        <tt class="py-name">all_seqnames</tt> <tt class="py-op">=</tt> <tt id="link-192" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-192', 'set', 'link-36');">set</a></tt><tt class="py-op">(</tt> </tt>
<a name="L573"></a><tt class="py-lineno">573</tt>  <tt class="py-line">            <tt class="py-op">[</tt><tt id="link-193" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-193', 'sequence', 'link-58');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-194" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-194', 'id', 'link-24');">id</a></tt> <tt class="py-keyword">for</tt> <tt class="py-name">multiseq</tt> <tt class="py-keyword">in</tt> <tt class="py-name">fetched</tt> <tt class="py-keyword">for</tt> <tt id="link-195" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-195', 'sequence', 'link-58');">sequence</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">multiseq</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L574"></a><tt class="py-lineno">574</tt>  <tt class="py-line"> </tt>
<a name="L575"></a><tt class="py-lineno">575</tt>  <tt class="py-line">        <tt class="py-comment"># split every record by base position</tt> </tt>
<a name="L576"></a><tt class="py-lineno">576</tt>  <tt class="py-line">        <tt class="py-comment"># key: sequence name</tt> </tt>
<a name="L577"></a><tt class="py-lineno">577</tt>  <tt class="py-line">        <tt class="py-comment"># value: dictionary</tt> </tt>
<a name="L578"></a><tt class="py-lineno">578</tt>  <tt class="py-line">        <tt class="py-comment">#        key: position in the reference sequence</tt> </tt>
<a name="L579"></a><tt class="py-lineno">579</tt>  <tt class="py-line">        <tt class="py-comment">#        value: letter(s) (including letters</tt> </tt>
<a name="L580"></a><tt class="py-lineno">580</tt>  <tt class="py-line">        <tt class="py-comment">#               aligned to the "-" preceding the letter</tt> </tt>
<a name="L581"></a><tt class="py-lineno">581</tt>  <tt class="py-line">        <tt class="py-comment">#               at the position in the reference, if any)</tt> </tt>
<a name="L582"></a><tt class="py-lineno">582</tt>  <tt class="py-line">        <tt class="py-name">split_by_position</tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">seq_name</tt><tt class="py-op">,</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">seq_name</tt> <tt class="py-keyword">in</tt> <tt class="py-name">all_seqnames</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L583"></a><tt class="py-lineno">583</tt>  <tt class="py-line"> </tt>
<a name="L584"></a><tt class="py-lineno">584</tt>  <tt class="py-line">        <tt class="py-comment"># keep track of what the total number of (unspliced) letters should be</tt> </tt>
<a name="L585"></a><tt class="py-lineno">585</tt>  <tt class="py-line">        <tt class="py-name">total_rec_length</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L586"></a><tt class="py-lineno">586</tt>  <tt class="py-line"> </tt>
<a name="L587"></a><tt class="py-lineno">587</tt>  <tt class="py-line">        <tt class="py-comment"># track first strand encountered on the target seqname</tt> </tt>
<a name="L588"></a><tt class="py-lineno">588</tt>  <tt class="py-line">        <tt class="py-name">ref_first_strand</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L589"></a><tt class="py-lineno">589</tt>  <tt class="py-line"> </tt>
<a name="L590"></a><tt class="py-lineno">590</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">multiseq</tt> <tt class="py-keyword">in</tt> <tt class="py-name">fetched</tt><tt class="py-op">:</tt> </tt>
<a name="L591"></a><tt class="py-lineno">591</tt>  <tt class="py-line">            <tt class="py-comment"># find the target_seqname in this MultipleSeqAlignment and use it to</tt> </tt>
<a name="L592"></a><tt class="py-lineno">592</tt>  <tt class="py-line">            <tt class="py-comment"># set the parameters for the rest of this iteration</tt> </tt>
<a name="L593"></a><tt class="py-lineno">593</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">seqrec</tt> <tt class="py-keyword">in</tt> <tt class="py-name">multiseq</tt><tt class="py-op">:</tt> </tt>
<a name="L594"></a><tt class="py-lineno">594</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-196" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-196', 'id', 'link-24');">id</a></tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">:</tt> </tt>
<a name="L595"></a><tt class="py-lineno">595</tt>  <tt class="py-line">                    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L596"></a><tt class="py-lineno">596</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">ref_first_strand</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L597"></a><tt class="py-lineno">597</tt>  <tt class="py-line">                            <tt class="py-name">ref_first_strand</tt> <tt class="py-op">=</tt> <tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-197" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-197', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"strand"</tt><tt class="py-op">]</tt> </tt>
<a name="L598"></a><tt class="py-lineno">598</tt>  <tt class="py-line"> </tt>
<a name="L599"></a><tt class="py-lineno">599</tt>  <tt class="py-line">                            <tt class="py-keyword">if</tt> <tt class="py-name">ref_first_strand</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L600"></a><tt class="py-lineno">600</tt>  <tt class="py-line">                                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Strand must be 1 or -1"</tt><tt class="py-op">)</tt> </tt>
<a name="L601"></a><tt class="py-lineno">601</tt>  <tt class="py-line">                        <tt class="py-keyword">elif</tt> <tt class="py-name">ref_first_strand</tt> <tt class="py-op">!=</tt> <tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-198" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-198', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"strand"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L602"></a><tt class="py-lineno">602</tt>  <tt class="py-line">                            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Encountered strand='%s' on target seqname, "</tt> </tt>
<a name="L603"></a><tt class="py-lineno">603</tt>  <tt class="py-line">                                             <tt class="py-string">"expected '%s'"</tt> <tt class="py-op">%</tt> </tt>
<a name="L604"></a><tt class="py-lineno">604</tt>  <tt class="py-line">                                             <tt class="py-op">(</tt><tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-199" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-199', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"strand"</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">ref_first_strand</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L605"></a><tt class="py-lineno">605</tt>  <tt class="py-line">                    <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L606"></a><tt class="py-lineno">606</tt>  <tt class="py-line">                        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"No strand information for target seqname (%s)"</tt> <tt class="py-op">%</tt> </tt>
<a name="L607"></a><tt class="py-lineno">607</tt>  <tt class="py-line">                                         <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">)</tt> </tt>
<a name="L608"></a><tt class="py-lineno">608</tt>  <tt class="py-line">                    <tt class="py-comment"># length including gaps (i.e. alignment length)</tt> </tt>
<a name="L609"></a><tt class="py-lineno">609</tt>  <tt class="py-line">                    <tt class="py-name">rec_length</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">seqrec</tt><tt class="py-op">)</tt> </tt>
<a name="L610"></a><tt class="py-lineno">610</tt>  <tt class="py-line">                    <tt class="py-name">rec_start</tt> <tt class="py-op">=</tt> <tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-200" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-200', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"start"</tt><tt class="py-op">]</tt> </tt>
<a name="L611"></a><tt class="py-lineno">611</tt>  <tt class="py-line">                    <tt class="py-name">rec_end</tt> <tt class="py-op">=</tt> <tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-201" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-201', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"start"</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-202" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-202', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"size"</tt><tt class="py-op">]</tt> </tt>
<a name="L612"></a><tt class="py-lineno">612</tt>  <tt class="py-line"> </tt>
<a name="L613"></a><tt class="py-lineno">613</tt>  <tt class="py-line">                    <tt class="py-name">total_rec_length</tt> <tt class="py-op">+=</tt> <tt class="py-name">rec_end</tt> <tt class="py-op">-</tt> <tt class="py-name">rec_start</tt> </tt>
<a name="L614"></a><tt class="py-lineno">614</tt>  <tt class="py-line"> </tt>
<a name="L615"></a><tt class="py-lineno">615</tt>  <tt class="py-line">                    <tt class="py-comment"># blank out these positions for every seqname</tt> </tt>
<a name="L616"></a><tt class="py-lineno">616</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-name">seqrec</tt> <tt class="py-keyword">in</tt> <tt class="py-name">multiseq</tt><tt class="py-op">:</tt> </tt>
<a name="L617"></a><tt class="py-lineno">617</tt>  <tt class="py-line">                        <tt class="py-keyword">for</tt> <tt class="py-name">pos</tt> <tt class="py-keyword">in</tt> <tt id="link-203" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-203', 'range', 'link-143');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">rec_start</tt><tt class="py-op">,</tt> <tt class="py-name">rec_end</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L618"></a><tt class="py-lineno">618</tt>  <tt class="py-line">                            <tt class="py-name">split_by_position</tt><tt class="py-op">[</tt><tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-204" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-204', 'id', 'link-24');">id</a></tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">pos</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L619"></a><tt class="py-lineno">619</tt>  <tt class="py-line"> </tt>
<a name="L620"></a><tt class="py-lineno">620</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L621"></a><tt class="py-lineno">621</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L622"></a><tt class="py-lineno">622</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Did not find %s in alignment bundle"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L623"></a><tt class="py-lineno">623</tt>  <tt class="py-line"> </tt>
<a name="L624"></a><tt class="py-lineno">624</tt>  <tt class="py-line">            <tt class="py-comment"># the true, chromosome/contig/etc position in the target seqname</tt> </tt>
<a name="L625"></a><tt class="py-lineno">625</tt>  <tt class="py-line">            <tt class="py-name">real_pos</tt> <tt class="py-op">=</tt> <tt class="py-name">rec_start</tt> </tt>
<a name="L626"></a><tt class="py-lineno">626</tt>  <tt class="py-line"> </tt>
<a name="L627"></a><tt class="py-lineno">627</tt>  <tt class="py-line">            <tt class="py-comment"># loop over the alignment to fill split_by_position</tt> </tt>
<a name="L628"></a><tt class="py-lineno">628</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">gapped_pos</tt> <tt class="py-keyword">in</tt> <tt id="link-205" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-205', 'range', 'link-143');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">rec_length</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L629"></a><tt class="py-lineno">629</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">seqrec</tt> <tt class="py-keyword">in</tt> <tt class="py-name">multiseq</tt><tt class="py-op">:</tt> </tt>
<a name="L630"></a><tt class="py-lineno">630</tt>  <tt class="py-line">                    <tt class="py-comment"># keep track of this position's value for the target seqname</tt> </tt>
<a name="L631"></a><tt class="py-lineno">631</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-206" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-206', 'id', 'link-24');">id</a></tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">:</tt> </tt>
<a name="L632"></a><tt class="py-lineno">632</tt>  <tt class="py-line">                        <tt class="py-name">track_val</tt> <tt class="py-op">=</tt> <tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-207" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-207', 'seq', 'link-29');">seq</a></tt><tt class="py-op">[</tt><tt class="py-name">gapped_pos</tt><tt class="py-op">]</tt> </tt>
<a name="L633"></a><tt class="py-lineno">633</tt>  <tt class="py-line"> </tt>
<a name="L634"></a><tt class="py-lineno">634</tt>  <tt class="py-line">                    <tt class="py-comment"># Here, a real_pos that corresponds to just after a series of "-"</tt> </tt>
<a name="L635"></a><tt class="py-lineno">635</tt>  <tt class="py-line">                    <tt class="py-comment"># in the reference will "accumulate" the letters found in other sequences</tt> </tt>
<a name="L636"></a><tt class="py-lineno">636</tt>  <tt class="py-line">                    <tt class="py-comment"># in front of the "-"s</tt> </tt>
<a name="L637"></a><tt class="py-lineno">637</tt>  <tt class="py-line">                    <tt class="py-name">split_by_position</tt><tt class="py-op">[</tt><tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-208" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-208', 'id', 'link-24');">id</a></tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">real_pos</tt><tt class="py-op">]</tt> <tt class="py-op">+=</tt> <tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-209" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-209', 'seq', 'link-29');">seq</a></tt><tt class="py-op">[</tt><tt class="py-name">gapped_pos</tt><tt class="py-op">]</tt> </tt>
<a name="L638"></a><tt class="py-lineno">638</tt>  <tt class="py-line"> </tt>
<a name="L639"></a><tt class="py-lineno">639</tt>  <tt class="py-line">                <tt class="py-comment"># increment the real_pos counter only when non-gaps are found in</tt> </tt>
<a name="L640"></a><tt class="py-lineno">640</tt>  <tt class="py-line">                <tt class="py-comment"># the target_seqname, and we haven't reached the end of the record</tt> </tt>
<a name="L641"></a><tt class="py-lineno">641</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">track_val</tt> <tt class="py-op">!=</tt> <tt class="py-string">"-"</tt> <tt class="py-keyword">and</tt> <tt class="py-name">real_pos</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">rec_end</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L642"></a><tt class="py-lineno">642</tt>  <tt class="py-line">                    <tt class="py-name">real_pos</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L643"></a><tt class="py-lineno">643</tt>  <tt class="py-line"> </tt>
<a name="L644"></a><tt class="py-lineno">644</tt>  <tt class="py-line">        <tt class="py-comment"># make sure the number of bp entries equals the sum of the record lengths</tt> </tt>
<a name="L645"></a><tt class="py-lineno">645</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">split_by_position</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">total_rec_length</tt><tt class="py-op">:</tt> </tt>
<a name="L646"></a><tt class="py-lineno">646</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Target seqname (%s) has %s records, expected %s"</tt> <tt class="py-op">%</tt> </tt>
<a name="L647"></a><tt class="py-lineno">647</tt>  <tt class="py-line">                             <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">,</tt> </tt>
<a name="L648"></a><tt class="py-lineno">648</tt>  <tt class="py-line">                              <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">split_by_position</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L649"></a><tt class="py-lineno">649</tt>  <tt class="py-line">                              <tt class="py-name">total_rec_length</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L650"></a><tt class="py-lineno">650</tt>  <tt class="py-line"> </tt>
<a name="L651"></a><tt class="py-lineno">651</tt>  <tt class="py-line">        <tt class="py-comment"># translates a position in the target_seqname sequence to its gapped length</tt> </tt>
<a name="L652"></a><tt class="py-lineno">652</tt>  <tt class="py-line">        <tt class="py-name">realpos_to_len</tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt id="link-210" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-210', 'x', 'link-37');">x</a></tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-211" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-211', 'y', 'link-40');">y</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L653"></a><tt class="py-lineno">653</tt>  <tt class="py-line">                               <tt class="py-keyword">for</tt> <tt id="link-212" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-212', 'x', 'link-37');">x</a></tt><tt class="py-op">,</tt> <tt id="link-213" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-213', 'y', 'link-40');">y</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">split_by_position</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-214" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-214', 'items', 'link-43');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L654"></a><tt class="py-lineno">654</tt>  <tt class="py-line">                               <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-215" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-215', 'y', 'link-40');">y</a></tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L655"></a><tt class="py-lineno">655</tt>  <tt class="py-line"> </tt>
<a name="L656"></a><tt class="py-lineno">656</tt>  <tt class="py-line">        <tt class="py-comment"># splice together the exons</tt> </tt>
<a name="L657"></a><tt class="py-lineno">657</tt>  <tt class="py-line">        <tt class="py-name">subseq</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L658"></a><tt class="py-lineno">658</tt>  <tt class="py-line"> </tt>
<a name="L659"></a><tt class="py-lineno">659</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">seqid</tt> <tt class="py-keyword">in</tt> <tt class="py-name">all_seqnames</tt><tt class="py-op">:</tt> </tt>
<a name="L660"></a><tt class="py-lineno">660</tt>  <tt class="py-line">            <tt class="py-name">seq_split</tt> <tt class="py-op">=</tt> <tt class="py-name">split_by_position</tt><tt class="py-op">[</tt><tt class="py-name">seqid</tt><tt class="py-op">]</tt> </tt>
<a name="L661"></a><tt class="py-lineno">661</tt>  <tt class="py-line">            <tt class="py-name">seq_splice</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L662"></a><tt class="py-lineno">662</tt>  <tt class="py-line"> </tt>
<a name="L663"></a><tt class="py-lineno">663</tt>  <tt class="py-line">            <tt class="py-name">filler_char</tt> <tt class="py-op">=</tt> <tt class="py-string">"N"</tt> <tt class="py-keyword">if</tt> <tt class="py-name">seqid</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt> <tt class="py-keyword">else</tt> <tt class="py-string">"-"</tt> </tt>
<a name="L664"></a><tt class="py-lineno">664</tt>  <tt class="py-line"> </tt>
<a name="L665"></a><tt class="py-lineno">665</tt>  <tt class="py-line">            <tt class="py-comment"># iterate from start to end, taking bases from split_by_position when</tt> </tt>
<a name="L666"></a><tt class="py-lineno">666</tt>  <tt class="py-line">            <tt class="py-comment"># they exist, using N or - for gaps when there is no alignment.</tt> </tt>
<a name="L667"></a><tt class="py-lineno">667</tt>  <tt class="py-line">            <tt id="link-216" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-216', 'append', 'link-64');">append</a></tt> <tt class="py-op">=</tt> <tt class="py-name">seq_splice</tt><tt class="py-op">.</tt><tt id="link-217" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-217', 'append', 'link-64');">append</a></tt> </tt>
<a name="L668"></a><tt class="py-lineno">668</tt>  <tt class="py-line"> </tt>
<a name="L669"></a><tt class="py-lineno">669</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">exonstart</tt><tt class="py-op">,</tt> <tt class="py-name">exonend</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">starts</tt><tt class="py-op">,</tt> <tt class="py-name">ends</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L670"></a><tt class="py-lineno">670</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">real_pos</tt> <tt class="py-keyword">in</tt> <tt id="link-218" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-218', 'range', 'link-143');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">exonstart</tt><tt class="py-op">,</tt> <tt class="py-name">exonend</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L671"></a><tt class="py-lineno">671</tt>  <tt class="py-line">                    <tt class="py-comment"># if this seqname has this position, add it</tt> </tt>
<a name="L672"></a><tt class="py-lineno">672</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">real_pos</tt> <tt class="py-keyword">in</tt> <tt class="py-name">seq_split</tt><tt class="py-op">:</tt> </tt>
<a name="L673"></a><tt class="py-lineno">673</tt>  <tt class="py-line">                        <tt id="link-219" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-219', 'append', 'link-64');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">seq_split</tt><tt class="py-op">[</tt><tt class="py-name">real_pos</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L674"></a><tt class="py-lineno">674</tt>  <tt class="py-line">                    <tt class="py-comment"># if not, but it's in the target_seqname, add length-matched filler</tt> </tt>
<a name="L675"></a><tt class="py-lineno">675</tt>  <tt class="py-line">                    <tt class="py-keyword">elif</tt> <tt class="py-name">real_pos</tt> <tt class="py-keyword">in</tt> <tt class="py-name">realpos_to_len</tt><tt class="py-op">:</tt> </tt>
<a name="L676"></a><tt class="py-lineno">676</tt>  <tt class="py-line">                        <tt id="link-220" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-220', 'append', 'link-64');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">filler_char</tt> <tt class="py-op">*</tt> <tt class="py-name">realpos_to_len</tt><tt class="py-op">[</tt><tt class="py-name">real_pos</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L677"></a><tt class="py-lineno">677</tt>  <tt class="py-line">                    <tt class="py-comment"># it's not in either, so add a single filler character</tt> </tt>
<a name="L678"></a><tt class="py-lineno">678</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L679"></a><tt class="py-lineno">679</tt>  <tt class="py-line">                        <tt id="link-221" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-221', 'append', 'link-64');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">filler_char</tt><tt class="py-op">)</tt> </tt>
<a name="L680"></a><tt class="py-lineno">680</tt>  <tt class="py-line"> </tt>
<a name="L681"></a><tt class="py-lineno">681</tt>  <tt class="py-line">            <tt class="py-name">subseq</tt><tt class="py-op">[</tt><tt class="py-name">seqid</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">seq_splice</tt><tt class="py-op">)</tt> </tt>
<a name="L682"></a><tt class="py-lineno">682</tt>  <tt class="py-line"> </tt>
<a name="L683"></a><tt class="py-lineno">683</tt>  <tt class="py-line">        <tt class="py-comment"># make sure we're returning the right number of letters</tt> </tt>
<a name="L684"></a><tt class="py-lineno">684</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">subseq</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"-"</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">expected_letters</tt><tt class="py-op">:</tt> </tt>
<a name="L685"></a><tt class="py-lineno">685</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Returning %s letters for target seqname (%s), expected %s"</tt> <tt class="py-op">%</tt> </tt>
<a name="L686"></a><tt class="py-lineno">686</tt>  <tt class="py-line">                             <tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">subseq</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"-"</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L687"></a><tt class="py-lineno">687</tt>  <tt class="py-line">                              <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">,</tt> <tt class="py-name">expected_letters</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L688"></a><tt class="py-lineno">688</tt>  <tt class="py-line"> </tt>
<a name="L689"></a><tt class="py-lineno">689</tt>  <tt class="py-line">        <tt class="py-comment"># check to make sure all sequences are the same length as the target seqname</tt> </tt>
<a name="L690"></a><tt class="py-lineno">690</tt>  <tt class="py-line">        <tt class="py-name">ref_subseq_len</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">subseq</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L691"></a><tt class="py-lineno">691</tt>  <tt class="py-line"> </tt>
<a name="L692"></a><tt class="py-lineno">692</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">seqid</tt><tt class="py-op">,</tt> <tt id="link-222" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-222', 'seq', 'link-29');">seq</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">subseq</tt><tt class="py-op">.</tt><tt id="link-223" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-223', 'items', 'link-43');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L693"></a><tt class="py-lineno">693</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-224" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-224', 'seq', 'link-29');">seq</a></tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">ref_subseq_len</tt><tt class="py-op">:</tt> </tt>
<a name="L694"></a><tt class="py-lineno">694</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Returning length %s for %s, expected %s"</tt> <tt class="py-op">%</tt> </tt>
<a name="L695"></a><tt class="py-lineno">695</tt>  <tt class="py-line">                                 <tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-225" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-225', 'seq', 'link-29');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">seqid</tt><tt class="py-op">,</tt> <tt class="py-name">ref_subseq_len</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L696"></a><tt class="py-lineno">696</tt>  <tt class="py-line"> </tt>
<a name="L697"></a><tt class="py-lineno">697</tt>  <tt class="py-line">        <tt class="py-comment"># finally, build a MultipleSeqAlignment object for our final sequences</tt> </tt>
<a name="L698"></a><tt class="py-lineno">698</tt>  <tt class="py-line">        <tt class="py-name">result_multiseq</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L699"></a><tt class="py-lineno">699</tt>  <tt class="py-line"> </tt>
<a name="L700"></a><tt class="py-lineno">700</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">seqid</tt><tt class="py-op">,</tt> <tt id="link-226" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-226', 'seq', 'link-29');">seq</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">subseq</tt><tt class="py-op">.</tt><tt id="link-227" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-227', 'items', 'link-43');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L701"></a><tt class="py-lineno">701</tt>  <tt class="py-line">            <tt id="link-228" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-228', 'seq', 'link-29');">seq</a></tt> <tt class="py-op">=</tt> <tt id="link-229" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-229', 'Seq', 'link-4');">Seq</a></tt><tt class="py-op">(</tt><tt id="link-230" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-230', 'seq', 'link-29');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L702"></a><tt class="py-lineno">702</tt>  <tt class="py-line"> </tt>
<a name="L703"></a><tt class="py-lineno">703</tt>  <tt class="py-line">            <tt id="link-231" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-231', 'seq', 'link-29');">seq</a></tt> <tt class="py-op">=</tt> <tt id="link-232" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-232', 'seq', 'link-29');">seq</a></tt> <tt class="py-keyword">if</tt> <tt id="link-233" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-233', 'strand', 'link-19');">strand</a></tt> <tt class="py-op">==</tt> <tt class="py-name">ref_first_strand</tt> <tt class="py-keyword">else</tt> <tt id="link-234" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-234', 'seq', 'link-29');">seq</a></tt><tt class="py-op">.</tt><tt id="link-235" class="py-name" targets="Method Bio.Seq.MutableSeq.reverse_complement()=Bio.Seq.MutableSeq-class.html#reverse_complement,Method Bio.Seq.Seq.reverse_complement()=Bio.Seq.Seq-class.html#reverse_complement,Method Bio.Seq.UnknownSeq.reverse_complement()=Bio.Seq.UnknownSeq-class.html#reverse_complement,Function Bio.Seq.reverse_complement()=Bio.Seq-module.html#reverse_complement,Method Bio.SeqRecord.SeqRecord.reverse_complement()=Bio.SeqRecord.SeqRecord-class.html#reverse_complement,Method Bio.motifs.Instances.reverse_complement()=Bio.motifs.Instances-class.html#reverse_complement,Method Bio.motifs.Motif.reverse_complement()=Bio.motifs.Motif-class.html#reverse_complement,Method Bio.motifs.matrix.GenericPositionMatrix.reverse_complement()=Bio.motifs.matrix.GenericPositionMatrix-class.html#reverse_complement"><a title="Bio.Seq.MutableSeq.reverse_complement
Bio.Seq.Seq.reverse_complement
Bio.Seq.UnknownSeq.reverse_complement
Bio.Seq.reverse_complement
Bio.SeqRecord.SeqRecord.reverse_complement
Bio.motifs.Instances.reverse_complement
Bio.motifs.Motif.reverse_complement
Bio.motifs.matrix.GenericPositionMatrix.reverse_complement" class="py-name" href="#" onclick="return doclink('link-235', 'reverse_complement', 'link-235');">reverse_complement</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L704"></a><tt class="py-lineno">704</tt>  <tt class="py-line"> </tt>
<a name="L705"></a><tt class="py-lineno">705</tt>  <tt class="py-line">            <tt class="py-name">result_multiseq</tt><tt class="py-op">.</tt><tt id="link-236" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-236', 'append', 'link-64');">append</a></tt><tt class="py-op">(</tt><tt id="link-237" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-237', 'SeqRecord', 'link-7');">SeqRecord</a></tt><tt class="py-op">(</tt><tt id="link-238" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-238', 'seq', 'link-29');">seq</a></tt><tt class="py-op">,</tt> </tt>
<a name="L706"></a><tt class="py-lineno">706</tt>  <tt class="py-line">                                             <tt id="link-239" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-239', 'id', 'link-24');">id</a></tt><tt class="py-op">=</tt><tt class="py-name">seqid</tt><tt class="py-op">,</tt> </tt>
<a name="L707"></a><tt class="py-lineno">707</tt>  <tt class="py-line">                                             <tt id="link-240" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-240', 'name', 'link-72');">name</a></tt><tt class="py-op">=</tt><tt class="py-name">seqid</tt><tt class="py-op">,</tt> </tt>
<a name="L708"></a><tt class="py-lineno">708</tt>  <tt class="py-line">                                             <tt id="link-241" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-241', 'description', 'link-73');">description</a></tt><tt class="py-op">=</tt><tt class="py-string">""</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L709"></a><tt class="py-lineno">709</tt>  <tt class="py-line"> </tt>
<a name="L710"></a><tt class="py-lineno">710</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-242" class="py-name"><a title="Bio.Align.MultipleSeqAlignment" class="py-name" href="#" onclick="return doclink('link-242', 'MultipleSeqAlignment', 'link-11');">MultipleSeqAlignment</a></tt><tt class="py-op">(</tt><tt class="py-name">result_multiseq</tt><tt class="py-op">)</tt> </tt>
</div><a name="L711"></a><tt class="py-lineno">711</tt>  <tt class="py-line"> </tt>
<a name="MafIndex.__repr__"></a><div id="MafIndex.__repr__-def"><a name="L712"></a><tt class="py-lineno">712</tt> <a class="py-toggle" href="#" id="MafIndex.__repr__-toggle" onclick="return toggle('MafIndex.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafIndex-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIndex.__repr__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafIndex.__repr__-expanded"><a name="L713"></a><tt class="py-lineno">713</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">"MafIO.MafIndex(%r, target_seqname=%r)"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_fp</tt><tt class="py-op">.</tt><tt id="link-243" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-243', 'name', 'link-72');">name</a></tt><tt class="py-op">,</tt> </tt>
<a name="L714"></a><tt class="py-lineno">714</tt>  <tt class="py-line">                                                          <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">)</tt> </tt>
</div><a name="L715"></a><tt class="py-lineno">715</tt>  <tt class="py-line"> </tt>
<a name="MafIndex.__len__"></a><div id="MafIndex.__len__-def"><a name="L716"></a><tt class="py-lineno">716</tt> <a class="py-toggle" href="#" id="MafIndex.__len__-toggle" onclick="return toggle('MafIndex.__len__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafIndex-class.html#__len__">__len__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIndex.__len__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafIndex.__len__-expanded"><a name="L717"></a><tt class="py-lineno">717</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the number of records in the index."""</tt> </tt>
<a name="L718"></a><tt class="py-lineno">718</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_record_count</tt> </tt>
</div></div><a name="L719"></a><tt class="py-lineno">719</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Apr  4 10:49:58 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
