<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.SeqIO._convert</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.SeqIO-module.html">Package&nbsp;SeqIO</a> ::
        Module&nbsp;_convert
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.SeqIO._convert-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.SeqIO._convert-module.html">Module Bio.SeqIO._convert</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2009 by Peter Cock.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-docstring">"""Optimised sequence conversion code (PRIVATE).</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-docstring">You are not expected to access this module, or any of its code, directly. This</tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-docstring">is all handled internally by the Bio.SeqIO.convert(...) function which is the</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-docstring">public interface for this.</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-docstring">The idea here is that while doing this will work::</tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-docstring">    from Bio import SeqIO</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-docstring">    records = SeqIO.parse(in_handle, in_format)</tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-docstring">    count = SeqIO.write(records, out_handle, out_format)</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-docstring">it is shorter to write::</tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-docstring">    from Bio import SeqIO</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-docstring">    count = SeqIO.convert(in_handle, in_format, out_handle, out_format)</tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-docstring">Also, the convert function can take a number of special case optimisations. This</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-docstring">means that using Bio.SeqIO.convert() may be faster, as well as more convenient.</tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-docstring">All these file format specific optimisations are handled by this (private) module.</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-1" class="py-name" targets="Package Bio.SeqIO=Bio.SeqIO-module.html"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-1', 'SeqIO', 'link-1');">SeqIO</a></tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt class="py-comment"># NOTE - Lots of lazy imports further on...</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"><tt class="py-name">__docformat__</tt> <tt class="py-op">=</tt> <tt class="py-string">"restructuredtext en"</tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"> </tt>
<a name="_genbank_convert_fasta"></a><div id="_genbank_convert_fasta-def"><a name="L33"></a><tt class="py-lineno"> 33</tt> <a class="py-toggle" href="#" id="_genbank_convert_fasta-toggle" onclick="return toggle('_genbank_convert_fasta');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO._convert-module.html#_genbank_convert_fasta">_genbank_convert_fasta</a><tt class="py-op">(</tt><tt class="py-param">in_handle</tt><tt class="py-op">,</tt> <tt class="py-param">out_handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_genbank_convert_fasta-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_genbank_convert_fasta-expanded"><a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line">    <tt class="py-docstring">"""Fast GenBank to FASTA (PRIVATE)."""</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line">    <tt class="py-comment"># We don't need to parse the features...</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-2" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-2', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-3" class="py-name" targets="Package Bio.GenBank=Bio.GenBank-module.html"><a title="Bio.GenBank" class="py-name" href="#" onclick="return doclink('link-3', 'GenBank', 'link-3');">GenBank</a></tt><tt class="py-op">.</tt><tt id="link-4" class="py-name" targets="Module Bio.GenBank.Scanner=Bio.GenBank.Scanner-module.html"><a title="Bio.GenBank.Scanner" class="py-name" href="#" onclick="return doclink('link-4', 'Scanner', 'link-4');">Scanner</a></tt> <tt class="py-keyword">import</tt> <tt id="link-5" class="py-name" targets="Class Bio.GenBank.Scanner.GenBankScanner=Bio.GenBank.Scanner.GenBankScanner-class.html"><a title="Bio.GenBank.Scanner.GenBankScanner" class="py-name" href="#" onclick="return doclink('link-5', 'GenBankScanner', 'link-5');">GenBankScanner</a></tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line">    <tt class="py-name">records</tt> <tt class="py-op">=</tt> <tt id="link-6" class="py-name"><a title="Bio.GenBank.Scanner.GenBankScanner" class="py-name" href="#" onclick="return doclink('link-6', 'GenBankScanner', 'link-5');">GenBankScanner</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-7" class="py-name" targets="Method Bio.GenBank.Scanner.InsdcScanner.parse_records()=Bio.GenBank.Scanner.InsdcScanner-class.html#parse_records"><a title="Bio.GenBank.Scanner.InsdcScanner.parse_records" class="py-name" href="#" onclick="return doclink('link-7', 'parse_records', 'link-7');">parse_records</a></tt><tt class="py-op">(</tt><tt class="py-name">in_handle</tt><tt class="py-op">,</tt> <tt class="py-name">do_features</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line">    <tt class="py-comment"># For FASTA output we can ignore the alphabet too</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-8" class="py-name"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-8', 'SeqIO', 'link-1');">SeqIO</a></tt><tt class="py-op">.</tt><tt id="link-9" class="py-name" targets="Function Bio.AlignIO.write()=Bio.AlignIO-module.html#write,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.write()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#write,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.write()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#write,Method Bio.Phylo.CDAOIO.Writer.write()=Bio.Phylo.CDAOIO.Writer-class.html#write,Function Bio.Phylo.CDAOIO.write()=Bio.Phylo.CDAOIO-module.html#write,Method Bio.Phylo.NeXMLIO.Writer.write()=Bio.Phylo.NeXMLIO.Writer-class.html#write,Function Bio.Phylo.NeXMLIO.write()=Bio.Phylo.NeXMLIO-module.html#write,Method Bio.Phylo.NewickIO.Writer.write()=Bio.Phylo.NewickIO.Writer-class.html#write,Function Bio.Phylo.NewickIO.write()=Bio.Phylo.NewickIO-module.html#write,Function Bio.Phylo.NexusIO.write()=Bio.Phylo.NexusIO-module.html#write,Method Bio.Phylo.PhyloXMLIO.Writer.write()=Bio.Phylo.PhyloXMLIO.Writer-class.html#write,Function Bio.Phylo.PhyloXMLIO.write()=Bio.Phylo.PhyloXMLIO-module.html#write,Function Bio.Phylo._io.write()=Bio.Phylo._io-module.html#write,Function Bio.SearchIO.write()=Bio.SearchIO-module.html#write,Function Bio.SeqIO.write()=Bio.SeqIO-module.html#write,Method Bio.bgzf.BgzfWriter.write()=Bio.bgzf.BgzfWriter-class.html#write,Function Bio.motifs.jaspar.write()=Bio.motifs.jaspar-module.html#write,Function Bio.motifs.transfac.write()=Bio.motifs.transfac-module.html#write,Function Bio.motifs.write()=Bio.motifs-module.html#write"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-9', 'write', 'link-9');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">records</tt><tt class="py-op">,</tt> <tt class="py-name">out_handle</tt><tt class="py-op">,</tt> <tt class="py-string">"fasta"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"> </tt>
<a name="_embl_convert_fasta"></a><div id="_embl_convert_fasta-def"><a name="L42"></a><tt class="py-lineno"> 42</tt> <a class="py-toggle" href="#" id="_embl_convert_fasta-toggle" onclick="return toggle('_embl_convert_fasta');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO._convert-module.html#_embl_convert_fasta">_embl_convert_fasta</a><tt class="py-op">(</tt><tt class="py-param">in_handle</tt><tt class="py-op">,</tt> <tt class="py-param">out_handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_embl_convert_fasta-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_embl_convert_fasta-expanded"><a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line">    <tt class="py-docstring">"""Fast EMBL to FASTA (PRIVATE)."""</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line">    <tt class="py-comment"># We don't need to parse the features...</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-10" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-10', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-11" class="py-name"><a title="Bio.GenBank" class="py-name" href="#" onclick="return doclink('link-11', 'GenBank', 'link-3');">GenBank</a></tt><tt class="py-op">.</tt><tt id="link-12" class="py-name"><a title="Bio.GenBank.Scanner" class="py-name" href="#" onclick="return doclink('link-12', 'Scanner', 'link-4');">Scanner</a></tt> <tt class="py-keyword">import</tt> <tt id="link-13" class="py-name" targets="Class Bio.GenBank.Scanner.EmblScanner=Bio.GenBank.Scanner.EmblScanner-class.html"><a title="Bio.GenBank.Scanner.EmblScanner" class="py-name" href="#" onclick="return doclink('link-13', 'EmblScanner', 'link-13');">EmblScanner</a></tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line">    <tt class="py-name">records</tt> <tt class="py-op">=</tt> <tt id="link-14" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner" class="py-name" href="#" onclick="return doclink('link-14', 'EmblScanner', 'link-13');">EmblScanner</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-15" class="py-name"><a title="Bio.GenBank.Scanner.InsdcScanner.parse_records" class="py-name" href="#" onclick="return doclink('link-15', 'parse_records', 'link-7');">parse_records</a></tt><tt class="py-op">(</tt><tt class="py-name">in_handle</tt><tt class="py-op">,</tt> <tt class="py-name">do_features</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line">    <tt class="py-comment"># For FASTA output we can ignore the alphabet too</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-16" class="py-name"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-16', 'SeqIO', 'link-1');">SeqIO</a></tt><tt class="py-op">.</tt><tt id="link-17" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-17', 'write', 'link-9');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">records</tt><tt class="py-op">,</tt> <tt class="py-name">out_handle</tt><tt class="py-op">,</tt> <tt class="py-string">"fasta"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line"> </tt>
<a name="_fastq_generic"></a><div id="_fastq_generic-def"><a name="L51"></a><tt class="py-lineno"> 51</tt> <a class="py-toggle" href="#" id="_fastq_generic-toggle" onclick="return toggle('_fastq_generic');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO._convert-module.html#_fastq_generic">_fastq_generic</a><tt class="py-op">(</tt><tt class="py-param">in_handle</tt><tt class="py-op">,</tt> <tt class="py-param">out_handle</tt><tt class="py-op">,</tt> <tt class="py-param">mapping</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_fastq_generic-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_fastq_generic-expanded"><a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line">    <tt class="py-docstring">"""FASTQ helper function where can't have data loss by truncation (PRIVATE)."""</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-18" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-18', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-19" class="py-name"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-19', 'SeqIO', 'link-1');">SeqIO</a></tt><tt class="py-op">.</tt><tt id="link-20" class="py-name" targets="Module Bio.SeqIO.QualityIO=Bio.SeqIO.QualityIO-module.html"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-20', 'QualityIO', 'link-20');">QualityIO</a></tt> <tt class="py-keyword">import</tt> <tt id="link-21" class="py-name" targets="Function Bio.SeqIO.QualityIO.FastqGeneralIterator()=Bio.SeqIO.QualityIO-module.html#FastqGeneralIterator"><a title="Bio.SeqIO.QualityIO.FastqGeneralIterator" class="py-name" href="#" onclick="return doclink('link-21', 'FastqGeneralIterator', 'link-21');">FastqGeneralIterator</a></tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line">    <tt class="py-comment"># For real speed, don't even make SeqRecord and Seq objects!</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line">    <tt id="link-22" class="py-name" targets="Method Bio.Crystal.Chain.count()=Bio.Crystal.Chain-class.html#count,Method Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count()=Bio.NeuralNetwork.Gene.Pattern.PatternRepository-class.html#count,Method Bio.Seq.MutableSeq.count()=Bio.Seq.MutableSeq-class.html#count,Method Bio.Seq.Seq.count()=Bio.Seq.Seq-class.html#count,Method Bio.Seq.UnknownSeq.count()=Bio.Seq.UnknownSeq-class.html#count,Method Bio.motifs.Instances.count()=Bio.motifs.Instances-class.html#count"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-22', 'count', 'link-22');">count</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line">    <tt class="py-name">null</tt> <tt class="py-op">=</tt> <tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt id="link-23" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.title()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#title,Method Bio.GenBank._FeatureConsumer.title()=Bio.GenBank._FeatureConsumer-class.html#title,Method Bio.GenBank._RecordConsumer.title()=Bio.GenBank._RecordConsumer-class.html#title"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-23', 'title', 'link-23');">title</a></tt><tt class="py-op">,</tt> <tt id="link-24" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.seq=Bio.SeqRecord.SeqRecord-class.html#seq,Variable BioSQL.BioSeq.DBSeqRecord.seq=BioSQL.BioSeq.DBSeqRecord-class.html#seq"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-24', 'seq', 'link-24');">seq</a></tt><tt class="py-op">,</tt> <tt class="py-name">old_qual</tt> <tt class="py-keyword">in</tt> <tt id="link-25" class="py-name"><a title="Bio.SeqIO.QualityIO.FastqGeneralIterator" class="py-name" href="#" onclick="return doclink('link-25', 'FastqGeneralIterator', 'link-21');">FastqGeneralIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">in_handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line">        <tt id="link-26" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-26', 'count', 'link-22');">count</a></tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line">        <tt class="py-comment"># map the qual...</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line">        <tt class="py-name">qual</tt> <tt class="py-op">=</tt> <tt class="py-name">old_qual</tt><tt class="py-op">.</tt><tt id="link-27" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.translate()=Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#translate,Method Bio.Seq.Seq.translate()=Bio.Seq.Seq-class.html#translate,Method Bio.Seq.UnknownSeq.translate()=Bio.Seq.UnknownSeq-class.html#translate,Function Bio.Seq.translate()=Bio.Seq-module.html#translate,Method Bio.codonalign.codonseq.CodonSeq.translate()=Bio.codonalign.codonseq.CodonSeq-class.html#translate"><a title="Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.translate
Bio.Seq.Seq.translate
Bio.Seq.UnknownSeq.translate
Bio.Seq.translate
Bio.codonalign.codonseq.CodonSeq.translate" class="py-name" href="#" onclick="return doclink('link-27', 'translate', 'link-27');">translate</a></tt><tt class="py-op">(</tt><tt class="py-name">mapping</tt><tt class="py-op">)</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">null</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qual</tt><tt class="py-op">:</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid character in quality string"</tt><tt class="py-op">)</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line">        <tt class="py-name">out_handle</tt><tt class="py-op">.</tt><tt id="link-28" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-28', 'write', 'link-9');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"@%s\n%s\n+\n%s\n"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-29" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-29', 'title', 'link-23');">title</a></tt><tt class="py-op">,</tt> <tt id="link-30" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-30', 'seq', 'link-24');">seq</a></tt><tt class="py-op">,</tt> <tt class="py-name">qual</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-31" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-31', 'count', 'link-22');">count</a></tt> </tt>
</div><a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line"> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line"> </tt>
<a name="_fastq_generic2"></a><div id="_fastq_generic2-def"><a name="L67"></a><tt class="py-lineno"> 67</tt> <a class="py-toggle" href="#" id="_fastq_generic2-toggle" onclick="return toggle('_fastq_generic2');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO._convert-module.html#_fastq_generic2">_fastq_generic2</a><tt class="py-op">(</tt><tt class="py-param">in_handle</tt><tt class="py-op">,</tt> <tt class="py-param">out_handle</tt><tt class="py-op">,</tt> <tt class="py-param">mapping</tt><tt class="py-op">,</tt> <tt class="py-param">truncate_char</tt><tt class="py-op">,</tt> <tt class="py-param">truncate_msg</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_fastq_generic2-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_fastq_generic2-expanded"><a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line">    <tt class="py-docstring">"""FASTQ helper function where there could be data loss by truncation (PRIVATE)."""</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-32" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-32', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-33" class="py-name"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-33', 'SeqIO', 'link-1');">SeqIO</a></tt><tt class="py-op">.</tt><tt id="link-34" class="py-name"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-34', 'QualityIO', 'link-20');">QualityIO</a></tt> <tt class="py-keyword">import</tt> <tt id="link-35" class="py-name"><a title="Bio.SeqIO.QualityIO.FastqGeneralIterator" class="py-name" href="#" onclick="return doclink('link-35', 'FastqGeneralIterator', 'link-21');">FastqGeneralIterator</a></tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line">    <tt class="py-comment"># For real speed, don't even make SeqRecord and Seq objects!</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line">    <tt id="link-36" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-36', 'count', 'link-22');">count</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line">    <tt class="py-name">null</tt> <tt class="py-op">=</tt> <tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt id="link-37" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-37', 'title', 'link-23');">title</a></tt><tt class="py-op">,</tt> <tt id="link-38" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-38', 'seq', 'link-24');">seq</a></tt><tt class="py-op">,</tt> <tt class="py-name">old_qual</tt> <tt class="py-keyword">in</tt> <tt id="link-39" class="py-name"><a title="Bio.SeqIO.QualityIO.FastqGeneralIterator" class="py-name" href="#" onclick="return doclink('link-39', 'FastqGeneralIterator', 'link-21');">FastqGeneralIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">in_handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">        <tt id="link-40" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-40', 'count', 'link-22');">count</a></tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line">        <tt class="py-comment"># map the qual...</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line">        <tt class="py-name">qual</tt> <tt class="py-op">=</tt> <tt class="py-name">old_qual</tt><tt class="py-op">.</tt><tt id="link-41" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.translate
Bio.Seq.Seq.translate
Bio.Seq.UnknownSeq.translate
Bio.Seq.translate
Bio.codonalign.codonseq.CodonSeq.translate" class="py-name" href="#" onclick="return doclink('link-41', 'translate', 'link-27');">translate</a></tt><tt class="py-op">(</tt><tt class="py-name">mapping</tt><tt class="py-op">)</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">null</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qual</tt><tt class="py-op">:</tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid character in quality string"</tt><tt class="py-op">)</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">truncate_char</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qual</tt><tt class="py-op">:</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line">            <tt class="py-name">qual</tt> <tt class="py-op">=</tt> <tt class="py-name">qual</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-name">truncate_char</tt><tt class="py-op">,</tt> <tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">126</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">            <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-name">truncate_msg</tt><tt class="py-op">)</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line">        <tt class="py-name">out_handle</tt><tt class="py-op">.</tt><tt id="link-42" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-42', 'write', 'link-9');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"@%s\n%s\n+\n%s\n"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-43" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-43', 'title', 'link-23');">title</a></tt><tt class="py-op">,</tt> <tt id="link-44" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-44', 'seq', 'link-24');">seq</a></tt><tt class="py-op">,</tt> <tt class="py-name">qual</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-45" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-45', 'count', 'link-22');">count</a></tt> </tt>
</div><a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line"> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line"> </tt>
<a name="_fastq_sanger_convert_fastq_sanger"></a><div id="_fastq_sanger_convert_fastq_sanger-def"><a name="L87"></a><tt class="py-lineno"> 87</tt> <a class="py-toggle" href="#" id="_fastq_sanger_convert_fastq_sanger-toggle" onclick="return toggle('_fastq_sanger_convert_fastq_sanger');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO._convert-module.html#_fastq_sanger_convert_fastq_sanger">_fastq_sanger_convert_fastq_sanger</a><tt class="py-op">(</tt><tt class="py-param">in_handle</tt><tt class="py-op">,</tt> <tt class="py-param">out_handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_fastq_sanger_convert_fastq_sanger-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_fastq_sanger_convert_fastq_sanger-expanded"><a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line">    <tt class="py-docstring">"""Fast Sanger FASTQ to Sanger FASTQ conversion (PRIVATE).</tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line"><tt class="py-docstring">    Useful for removing line wrapping and the redundant second identifier</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line"><tt class="py-docstring">    on the plus lines. Will check also check the quality string is valid.</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line"><tt class="py-docstring">    Avoids creating SeqRecord and Seq objects in order to speed up this</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line"><tt class="py-docstring">    conversion.</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line">    <tt class="py-comment"># Map unexpected chars to null</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line">    <tt class="py-name">mapping</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ascii</tt> <tt class="py-keyword">in</tt> <tt id="link-46" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-46', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">33</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line">                      <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-name">ascii</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ascii</tt> <tt class="py-keyword">in</tt> <tt id="link-47" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-47', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">33</tt><tt class="py-op">,</tt> <tt class="py-number">127</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">                      <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ascii</tt> <tt class="py-keyword">in</tt> <tt id="link-48" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-48', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">127</tt><tt class="py-op">,</tt> <tt class="py-number">256</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">mapping</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">256</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-49" class="py-name" targets="Function Bio.SeqIO._convert._fastq_generic()=Bio.SeqIO._convert-module.html#_fastq_generic"><a title="Bio.SeqIO._convert._fastq_generic" class="py-name" href="#" onclick="return doclink('link-49', '_fastq_generic', 'link-49');">_fastq_generic</a></tt><tt class="py-op">(</tt><tt class="py-name">in_handle</tt><tt class="py-op">,</tt> <tt class="py-name">out_handle</tt><tt class="py-op">,</tt> <tt class="py-name">mapping</tt><tt class="py-op">)</tt> </tt>
</div><a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line"> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line"> </tt>
<a name="_fastq_solexa_convert_fastq_solexa"></a><div id="_fastq_solexa_convert_fastq_solexa-def"><a name="L104"></a><tt class="py-lineno">104</tt> <a class="py-toggle" href="#" id="_fastq_solexa_convert_fastq_solexa-toggle" onclick="return toggle('_fastq_solexa_convert_fastq_solexa');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO._convert-module.html#_fastq_solexa_convert_fastq_solexa">_fastq_solexa_convert_fastq_solexa</a><tt class="py-op">(</tt><tt class="py-param">in_handle</tt><tt class="py-op">,</tt> <tt class="py-param">out_handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_fastq_solexa_convert_fastq_solexa-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_fastq_solexa_convert_fastq_solexa-expanded"><a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">    <tt class="py-docstring">"""Fast Solexa FASTQ to Solexa FASTQ conversion (PRIVATE).</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line"><tt class="py-docstring">    Useful for removing line wrapping and the redundant second identifier</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line"><tt class="py-docstring">    on the plus lines. Will check also check the quality string is valid.</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line"><tt class="py-docstring">    Avoids creating SeqRecord and Seq objects in order to speed up this</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line"><tt class="py-docstring">    conversion.</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">    <tt class="py-comment"># Map unexpected chars to null</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">    <tt class="py-name">mapping</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ascii</tt> <tt class="py-keyword">in</tt> <tt id="link-50" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-50', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">59</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">                      <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-name">ascii</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ascii</tt> <tt class="py-keyword">in</tt> <tt id="link-51" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-51', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">59</tt><tt class="py-op">,</tt> <tt class="py-number">127</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">                      <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ascii</tt> <tt class="py-keyword">in</tt> <tt id="link-52" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-52', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">127</tt><tt class="py-op">,</tt> <tt class="py-number">256</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">mapping</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">256</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-53" class="py-name"><a title="Bio.SeqIO._convert._fastq_generic" class="py-name" href="#" onclick="return doclink('link-53', '_fastq_generic', 'link-49');">_fastq_generic</a></tt><tt class="py-op">(</tt><tt class="py-name">in_handle</tt><tt class="py-op">,</tt> <tt class="py-name">out_handle</tt><tt class="py-op">,</tt> <tt class="py-name">mapping</tt><tt class="py-op">)</tt> </tt>
</div><a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line"> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line"> </tt>
<a name="_fastq_illumina_convert_fastq_illumina"></a><div id="_fastq_illumina_convert_fastq_illumina-def"><a name="L120"></a><tt class="py-lineno">120</tt> <a class="py-toggle" href="#" id="_fastq_illumina_convert_fastq_illumina-toggle" onclick="return toggle('_fastq_illumina_convert_fastq_illumina');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO._convert-module.html#_fastq_illumina_convert_fastq_illumina">_fastq_illumina_convert_fastq_illumina</a><tt class="py-op">(</tt><tt class="py-param">in_handle</tt><tt class="py-op">,</tt> <tt class="py-param">out_handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_fastq_illumina_convert_fastq_illumina-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_fastq_illumina_convert_fastq_illumina-expanded"><a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">    <tt class="py-docstring">"""Fast Illumina 1.3+ FASTQ to Illumina 1.3+ FASTQ conversion (PRIVATE).</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line"><tt class="py-docstring">    Useful for removing line wrapping and the redundant second identifier</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line"><tt class="py-docstring">    on the plus lines. Will check also check the quality string is valid.</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line"><tt class="py-docstring">    Avoids creating SeqRecord and Seq objects in order to speed up this</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line"><tt class="py-docstring">    conversion.</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line">    <tt class="py-comment"># Map unexpected chars to null</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line">    <tt class="py-name">mapping</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ascii</tt> <tt class="py-keyword">in</tt> <tt id="link-54" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-54', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">64</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line">                      <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-name">ascii</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ascii</tt> <tt class="py-keyword">in</tt> <tt id="link-55" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-55', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">64</tt><tt class="py-op">,</tt> <tt class="py-number">127</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line">                      <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ascii</tt> <tt class="py-keyword">in</tt> <tt id="link-56" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-56', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">127</tt><tt class="py-op">,</tt> <tt class="py-number">256</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">mapping</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">256</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-57" class="py-name"><a title="Bio.SeqIO._convert._fastq_generic" class="py-name" href="#" onclick="return doclink('link-57', '_fastq_generic', 'link-49');">_fastq_generic</a></tt><tt class="py-op">(</tt><tt class="py-name">in_handle</tt><tt class="py-op">,</tt> <tt class="py-name">out_handle</tt><tt class="py-op">,</tt> <tt class="py-name">mapping</tt><tt class="py-op">)</tt> </tt>
</div><a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line"> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line"> </tt>
<a name="_fastq_illumina_convert_fastq_sanger"></a><div id="_fastq_illumina_convert_fastq_sanger-def"><a name="L136"></a><tt class="py-lineno">136</tt> <a class="py-toggle" href="#" id="_fastq_illumina_convert_fastq_sanger-toggle" onclick="return toggle('_fastq_illumina_convert_fastq_sanger');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO._convert-module.html#_fastq_illumina_convert_fastq_sanger">_fastq_illumina_convert_fastq_sanger</a><tt class="py-op">(</tt><tt class="py-param">in_handle</tt><tt class="py-op">,</tt> <tt class="py-param">out_handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_fastq_illumina_convert_fastq_sanger-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_fastq_illumina_convert_fastq_sanger-expanded"><a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line">    <tt class="py-docstring">"""Fast Illumina 1.3+ FASTQ to Sanger FASTQ conversion (PRIVATE).</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line"><tt class="py-docstring">    Avoids creating SeqRecord and Seq objects in order to speed up this</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line"><tt class="py-docstring">    conversion.</tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line">    <tt class="py-comment"># Map unexpected chars to null</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line">    <tt class="py-name">mapping</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ascii</tt> <tt class="py-keyword">in</tt> <tt id="link-58" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-58', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">64</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line">                      <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">33</tt> <tt class="py-op">+</tt> <tt class="py-name">q</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">q</tt> <tt class="py-keyword">in</tt> <tt id="link-59" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-59', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">62</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">                      <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ascii</tt> <tt class="py-keyword">in</tt> <tt id="link-60" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-60', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">127</tt><tt class="py-op">,</tt> <tt class="py-number">256</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">mapping</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">256</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-61" class="py-name"><a title="Bio.SeqIO._convert._fastq_generic" class="py-name" href="#" onclick="return doclink('link-61', '_fastq_generic', 'link-49');">_fastq_generic</a></tt><tt class="py-op">(</tt><tt class="py-name">in_handle</tt><tt class="py-op">,</tt> <tt class="py-name">out_handle</tt><tt class="py-op">,</tt> <tt class="py-name">mapping</tt><tt class="py-op">)</tt> </tt>
</div><a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line"> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line"> </tt>
<a name="_fastq_sanger_convert_fastq_illumina"></a><div id="_fastq_sanger_convert_fastq_illumina-def"><a name="L150"></a><tt class="py-lineno">150</tt> <a class="py-toggle" href="#" id="_fastq_sanger_convert_fastq_illumina-toggle" onclick="return toggle('_fastq_sanger_convert_fastq_illumina');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO._convert-module.html#_fastq_sanger_convert_fastq_illumina">_fastq_sanger_convert_fastq_illumina</a><tt class="py-op">(</tt><tt class="py-param">in_handle</tt><tt class="py-op">,</tt> <tt class="py-param">out_handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_fastq_sanger_convert_fastq_illumina-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_fastq_sanger_convert_fastq_illumina-expanded"><a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">    <tt class="py-docstring">"""Fast Sanger FASTQ to Illumina 1.3+ FASTQ conversion (PRIVATE).</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line"><tt class="py-docstring">    Avoids creating SeqRecord and Seq objects in order to speed up this</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line"><tt class="py-docstring">    conversion. Will issue a warning if the scores had to be truncated at 62</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line"><tt class="py-docstring">    (maximum possible in the Illumina 1.3+ FASTQ format)</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">    <tt class="py-comment"># Map unexpected chars to null</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">    <tt class="py-name">trunc_char</tt> <tt class="py-op">=</tt> <tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">    <tt class="py-name">mapping</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ascii</tt> <tt class="py-keyword">in</tt> <tt id="link-62" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-62', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">33</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">                      <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">64</tt> <tt class="py-op">+</tt> <tt class="py-name">q</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">q</tt> <tt class="py-keyword">in</tt> <tt id="link-63" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-63', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">62</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">                      <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-name">trunc_char</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ascii</tt> <tt class="py-keyword">in</tt> <tt id="link-64" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-64', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">96</tt><tt class="py-op">,</tt> <tt class="py-number">127</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">                      <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ascii</tt> <tt class="py-keyword">in</tt> <tt id="link-65" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-65', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">127</tt><tt class="py-op">,</tt> <tt class="py-number">256</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">mapping</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">256</tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-66" class="py-name" targets="Function Bio.SeqIO._convert._fastq_generic2()=Bio.SeqIO._convert-module.html#_fastq_generic2"><a title="Bio.SeqIO._convert._fastq_generic2" class="py-name" href="#" onclick="return doclink('link-66', '_fastq_generic2', 'link-66');">_fastq_generic2</a></tt><tt class="py-op">(</tt><tt class="py-name">in_handle</tt><tt class="py-op">,</tt> <tt class="py-name">out_handle</tt><tt class="py-op">,</tt> <tt class="py-name">mapping</tt><tt class="py-op">,</tt> <tt class="py-name">trunc_char</tt><tt class="py-op">,</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">                           <tt class="py-string">"Data loss - max PHRED quality 62 in Illumina 1.3+ FASTQ"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line"> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line"> </tt>
<a name="_fastq_solexa_convert_fastq_sanger"></a><div id="_fastq_solexa_convert_fastq_sanger-def"><a name="L168"></a><tt class="py-lineno">168</tt> <a class="py-toggle" href="#" id="_fastq_solexa_convert_fastq_sanger-toggle" onclick="return toggle('_fastq_solexa_convert_fastq_sanger');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO._convert-module.html#_fastq_solexa_convert_fastq_sanger">_fastq_solexa_convert_fastq_sanger</a><tt class="py-op">(</tt><tt class="py-param">in_handle</tt><tt class="py-op">,</tt> <tt class="py-param">out_handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_fastq_solexa_convert_fastq_sanger-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_fastq_solexa_convert_fastq_sanger-expanded"><a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line">    <tt class="py-docstring">"""Fast Solexa FASTQ to Sanger FASTQ conversion (PRIVATE).</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line"><tt class="py-docstring">    Avoids creating SeqRecord and Seq objects in order to speed up this</tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line"><tt class="py-docstring">    conversion.</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line">    <tt class="py-comment"># Map unexpected chars to null</tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-67" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-67', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-68" class="py-name"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-68', 'SeqIO', 'link-1');">SeqIO</a></tt><tt class="py-op">.</tt><tt id="link-69" class="py-name"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-69', 'QualityIO', 'link-20');">QualityIO</a></tt> <tt class="py-keyword">import</tt> <tt id="link-70" class="py-name" targets="Function Bio.SeqIO.QualityIO.phred_quality_from_solexa()=Bio.SeqIO.QualityIO-module.html#phred_quality_from_solexa"><a title="Bio.SeqIO.QualityIO.phred_quality_from_solexa" class="py-name" href="#" onclick="return doclink('link-70', 'phred_quality_from_solexa', 'link-70');">phred_quality_from_solexa</a></tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">    <tt class="py-name">mapping</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ascii</tt> <tt class="py-keyword">in</tt> <tt id="link-71" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-71', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">59</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">                      <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">33</tt> <tt class="py-op">+</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-72" class="py-name" targets="Method Bio.Blast.NCBIStandalone._BlastConsumer.round()=Bio.Blast.NCBIStandalone._BlastConsumer-class.html#round,Method Bio.Blast.NCBIStandalone._DescriptionConsumer.round()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#round"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.round
Bio.Blast.NCBIStandalone._DescriptionConsumer.round" class="py-name" href="#" onclick="return doclink('link-72', 'round', 'link-72');">round</a></tt><tt class="py-op">(</tt><tt id="link-73" class="py-name"><a title="Bio.SeqIO.QualityIO.phred_quality_from_solexa" class="py-name" href="#" onclick="return doclink('link-73', 'phred_quality_from_solexa', 'link-70');">phred_quality_from_solexa</a></tt><tt class="py-op">(</tt><tt class="py-name">q</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">                         <tt class="py-keyword">for</tt> <tt class="py-name">q</tt> <tt class="py-keyword">in</tt> <tt id="link-74" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-74', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">5</tt><tt class="py-op">,</tt> <tt class="py-number">62</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">                      <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ascii</tt> <tt class="py-keyword">in</tt> <tt id="link-75" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-75', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">127</tt><tt class="py-op">,</tt> <tt class="py-number">256</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">mapping</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">256</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-76" class="py-name"><a title="Bio.SeqIO._convert._fastq_generic" class="py-name" href="#" onclick="return doclink('link-76', '_fastq_generic', 'link-49');">_fastq_generic</a></tt><tt class="py-op">(</tt><tt class="py-name">in_handle</tt><tt class="py-op">,</tt> <tt class="py-name">out_handle</tt><tt class="py-op">,</tt> <tt class="py-name">mapping</tt><tt class="py-op">)</tt> </tt>
</div><a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line"> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line"> </tt>
<a name="_fastq_sanger_convert_fastq_solexa"></a><div id="_fastq_sanger_convert_fastq_solexa-def"><a name="L184"></a><tt class="py-lineno">184</tt> <a class="py-toggle" href="#" id="_fastq_sanger_convert_fastq_solexa-toggle" onclick="return toggle('_fastq_sanger_convert_fastq_solexa');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO._convert-module.html#_fastq_sanger_convert_fastq_solexa">_fastq_sanger_convert_fastq_solexa</a><tt class="py-op">(</tt><tt class="py-param">in_handle</tt><tt class="py-op">,</tt> <tt class="py-param">out_handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_fastq_sanger_convert_fastq_solexa-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_fastq_sanger_convert_fastq_solexa-expanded"><a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">    <tt class="py-docstring">"""Fast Sanger FASTQ to Solexa FASTQ conversion (PRIVATE).</tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line"><tt class="py-docstring">    Avoids creating SeqRecord and Seq objects in order to speed up this</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line"><tt class="py-docstring">    conversion. Will issue a warning if the scores had to be truncated at 62</tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line"><tt class="py-docstring">    (maximum possible in the Solexa FASTQ format)</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">    <tt class="py-comment"># Map unexpected chars to null</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-77" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-77', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-78" class="py-name"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-78', 'SeqIO', 'link-1');">SeqIO</a></tt><tt class="py-op">.</tt><tt id="link-79" class="py-name"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-79', 'QualityIO', 'link-20');">QualityIO</a></tt> <tt class="py-keyword">import</tt> <tt id="link-80" class="py-name" targets="Function Bio.SeqIO.QualityIO.solexa_quality_from_phred()=Bio.SeqIO.QualityIO-module.html#solexa_quality_from_phred"><a title="Bio.SeqIO.QualityIO.solexa_quality_from_phred" class="py-name" href="#" onclick="return doclink('link-80', 'solexa_quality_from_phred', 'link-80');">solexa_quality_from_phred</a></tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">    <tt class="py-name">trunc_char</tt> <tt class="py-op">=</tt> <tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">    <tt class="py-name">mapping</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ascii</tt> <tt class="py-keyword">in</tt> <tt id="link-81" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-81', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">33</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">                      <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">64</tt> <tt class="py-op">+</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-82" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.round
Bio.Blast.NCBIStandalone._DescriptionConsumer.round" class="py-name" href="#" onclick="return doclink('link-82', 'round', 'link-72');">round</a></tt><tt class="py-op">(</tt><tt id="link-83" class="py-name"><a title="Bio.SeqIO.QualityIO.solexa_quality_from_phred" class="py-name" href="#" onclick="return doclink('link-83', 'solexa_quality_from_phred', 'link-80');">solexa_quality_from_phred</a></tt><tt class="py-op">(</tt><tt class="py-name">q</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">                         <tt class="py-keyword">for</tt> <tt class="py-name">q</tt> <tt class="py-keyword">in</tt> <tt id="link-84" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-84', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">62</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">                      <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-name">trunc_char</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ascii</tt> <tt class="py-keyword">in</tt> <tt id="link-85" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-85', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">96</tt><tt class="py-op">,</tt> <tt class="py-number">127</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line">                      <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ascii</tt> <tt class="py-keyword">in</tt> <tt id="link-86" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-86', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">127</tt><tt class="py-op">,</tt> <tt class="py-number">256</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">mapping</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">256</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-87" class="py-name"><a title="Bio.SeqIO._convert._fastq_generic2" class="py-name" href="#" onclick="return doclink('link-87', '_fastq_generic2', 'link-66');">_fastq_generic2</a></tt><tt class="py-op">(</tt><tt class="py-name">in_handle</tt><tt class="py-op">,</tt> <tt class="py-name">out_handle</tt><tt class="py-op">,</tt> <tt class="py-name">mapping</tt><tt class="py-op">,</tt> <tt class="py-name">trunc_char</tt><tt class="py-op">,</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line">                           <tt class="py-string">"Data loss - max Solexa quality 62 in Solexa FASTQ"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line"> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line"> </tt>
<a name="_fastq_solexa_convert_fastq_illumina"></a><div id="_fastq_solexa_convert_fastq_illumina-def"><a name="L204"></a><tt class="py-lineno">204</tt> <a class="py-toggle" href="#" id="_fastq_solexa_convert_fastq_illumina-toggle" onclick="return toggle('_fastq_solexa_convert_fastq_illumina');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO._convert-module.html#_fastq_solexa_convert_fastq_illumina">_fastq_solexa_convert_fastq_illumina</a><tt class="py-op">(</tt><tt class="py-param">in_handle</tt><tt class="py-op">,</tt> <tt class="py-param">out_handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_fastq_solexa_convert_fastq_illumina-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_fastq_solexa_convert_fastq_illumina-expanded"><a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line">    <tt class="py-docstring">"""Fast Solexa FASTQ to Illumina 1.3+ FASTQ conversion (PRIVATE).</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line"><tt class="py-docstring">    Avoids creating SeqRecord and Seq objects in order to speed up this</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line"><tt class="py-docstring">    conversion.</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line">    <tt class="py-comment"># Map unexpected chars to null</tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-88" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-88', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-89" class="py-name"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-89', 'SeqIO', 'link-1');">SeqIO</a></tt><tt class="py-op">.</tt><tt id="link-90" class="py-name"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-90', 'QualityIO', 'link-20');">QualityIO</a></tt> <tt class="py-keyword">import</tt> <tt id="link-91" class="py-name"><a title="Bio.SeqIO.QualityIO.phred_quality_from_solexa" class="py-name" href="#" onclick="return doclink('link-91', 'phred_quality_from_solexa', 'link-70');">phred_quality_from_solexa</a></tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line">    <tt class="py-name">mapping</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ascii</tt> <tt class="py-keyword">in</tt> <tt id="link-92" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-92', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">59</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line">                      <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">64</tt> <tt class="py-op">+</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-93" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.round
Bio.Blast.NCBIStandalone._DescriptionConsumer.round" class="py-name" href="#" onclick="return doclink('link-93', 'round', 'link-72');">round</a></tt><tt class="py-op">(</tt><tt id="link-94" class="py-name"><a title="Bio.SeqIO.QualityIO.phred_quality_from_solexa" class="py-name" href="#" onclick="return doclink('link-94', 'phred_quality_from_solexa', 'link-70');">phred_quality_from_solexa</a></tt><tt class="py-op">(</tt><tt class="py-name">q</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line">                         <tt class="py-keyword">for</tt> <tt class="py-name">q</tt> <tt class="py-keyword">in</tt> <tt id="link-95" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-95', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">5</tt><tt class="py-op">,</tt> <tt class="py-number">62</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">                      <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ascii</tt> <tt class="py-keyword">in</tt> <tt id="link-96" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-96', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">127</tt><tt class="py-op">,</tt> <tt class="py-number">256</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">mapping</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">256</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-97" class="py-name"><a title="Bio.SeqIO._convert._fastq_generic" class="py-name" href="#" onclick="return doclink('link-97', '_fastq_generic', 'link-49');">_fastq_generic</a></tt><tt class="py-op">(</tt><tt class="py-name">in_handle</tt><tt class="py-op">,</tt> <tt class="py-name">out_handle</tt><tt class="py-op">,</tt> <tt class="py-name">mapping</tt><tt class="py-op">)</tt> </tt>
</div><a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line"> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line"> </tt>
<a name="_fastq_illumina_convert_fastq_solexa"></a><div id="_fastq_illumina_convert_fastq_solexa-def"><a name="L220"></a><tt class="py-lineno">220</tt> <a class="py-toggle" href="#" id="_fastq_illumina_convert_fastq_solexa-toggle" onclick="return toggle('_fastq_illumina_convert_fastq_solexa');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO._convert-module.html#_fastq_illumina_convert_fastq_solexa">_fastq_illumina_convert_fastq_solexa</a><tt class="py-op">(</tt><tt class="py-param">in_handle</tt><tt class="py-op">,</tt> <tt class="py-param">out_handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_fastq_illumina_convert_fastq_solexa-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_fastq_illumina_convert_fastq_solexa-expanded"><a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line">    <tt class="py-docstring">"""Fast Illumina 1.3+ FASTQ to Solexa FASTQ conversion (PRIVATE).</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line"><tt class="py-docstring">    Avoids creating SeqRecord and Seq objects in order to speed up this</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line"><tt class="py-docstring">    conversion.</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line">    <tt class="py-comment"># Map unexpected chars to null</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-98" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-98', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-99" class="py-name"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-99', 'SeqIO', 'link-1');">SeqIO</a></tt><tt class="py-op">.</tt><tt id="link-100" class="py-name"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-100', 'QualityIO', 'link-20');">QualityIO</a></tt> <tt class="py-keyword">import</tt> <tt id="link-101" class="py-name"><a title="Bio.SeqIO.QualityIO.solexa_quality_from_phred" class="py-name" href="#" onclick="return doclink('link-101', 'solexa_quality_from_phred', 'link-80');">solexa_quality_from_phred</a></tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line">    <tt class="py-name">trunc_char</tt> <tt class="py-op">=</tt> <tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">    <tt class="py-name">mapping</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ascii</tt> <tt class="py-keyword">in</tt> <tt id="link-102" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-102', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">64</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line">                      <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">64</tt> <tt class="py-op">+</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-103" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.round
Bio.Blast.NCBIStandalone._DescriptionConsumer.round" class="py-name" href="#" onclick="return doclink('link-103', 'round', 'link-72');">round</a></tt><tt class="py-op">(</tt><tt id="link-104" class="py-name"><a title="Bio.SeqIO.QualityIO.solexa_quality_from_phred" class="py-name" href="#" onclick="return doclink('link-104', 'solexa_quality_from_phred', 'link-80');">solexa_quality_from_phred</a></tt><tt class="py-op">(</tt><tt class="py-name">q</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line">                         <tt class="py-keyword">for</tt> <tt class="py-name">q</tt> <tt class="py-keyword">in</tt> <tt id="link-105" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-105', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">62</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">                      <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ascii</tt> <tt class="py-keyword">in</tt> <tt id="link-106" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-106', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">127</tt><tt class="py-op">,</tt> <tt class="py-number">256</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">mapping</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">256</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-107" class="py-name"><a title="Bio.SeqIO._convert._fastq_generic" class="py-name" href="#" onclick="return doclink('link-107', '_fastq_generic', 'link-49');">_fastq_generic</a></tt><tt class="py-op">(</tt><tt class="py-name">in_handle</tt><tt class="py-op">,</tt> <tt class="py-name">out_handle</tt><tt class="py-op">,</tt> <tt class="py-name">mapping</tt><tt class="py-op">)</tt> </tt>
</div><a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line"> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line"> </tt>
<a name="_fastq_convert_fasta"></a><div id="_fastq_convert_fasta-def"><a name="L237"></a><tt class="py-lineno">237</tt> <a class="py-toggle" href="#" id="_fastq_convert_fasta-toggle" onclick="return toggle('_fastq_convert_fasta');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO._convert-module.html#_fastq_convert_fasta">_fastq_convert_fasta</a><tt class="py-op">(</tt><tt class="py-param">in_handle</tt><tt class="py-op">,</tt> <tt class="py-param">out_handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_fastq_convert_fasta-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_fastq_convert_fasta-expanded"><a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line">    <tt class="py-docstring">"""Fast FASTQ to FASTA conversion (PRIVATE).</tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line"><tt class="py-docstring">    Avoids dealing with the FASTQ quality encoding, and creating SeqRecord and</tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line"><tt class="py-docstring">    Seq objects in order to speed up this conversion.</tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line"><tt class="py-docstring">    NOTE - This does NOT check the characters used in the FASTQ quality string</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line"><tt class="py-docstring">    are valid!</tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-108" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-108', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-109" class="py-name"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-109', 'SeqIO', 'link-1');">SeqIO</a></tt><tt class="py-op">.</tt><tt id="link-110" class="py-name"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-110', 'QualityIO', 'link-20');">QualityIO</a></tt> <tt class="py-keyword">import</tt> <tt id="link-111" class="py-name"><a title="Bio.SeqIO.QualityIO.FastqGeneralIterator" class="py-name" href="#" onclick="return doclink('link-111', 'FastqGeneralIterator', 'link-21');">FastqGeneralIterator</a></tt> </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line">    <tt class="py-comment"># For real speed, don't even make SeqRecord and Seq objects!</tt> </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line">    <tt id="link-112" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-112', 'count', 'link-22');">count</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt id="link-113" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-113', 'title', 'link-23');">title</a></tt><tt class="py-op">,</tt> <tt id="link-114" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-114', 'seq', 'link-24');">seq</a></tt><tt class="py-op">,</tt> <tt class="py-name">qual</tt> <tt class="py-keyword">in</tt> <tt id="link-115" class="py-name"><a title="Bio.SeqIO.QualityIO.FastqGeneralIterator" class="py-name" href="#" onclick="return doclink('link-115', 'FastqGeneralIterator', 'link-21');">FastqGeneralIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">in_handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line">        <tt id="link-116" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-116', 'count', 'link-22');">count</a></tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line">        <tt class="py-name">out_handle</tt><tt class="py-op">.</tt><tt id="link-117" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-117', 'write', 'link-9');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"&gt;%s\n"</tt> <tt class="py-op">%</tt> <tt id="link-118" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-118', 'title', 'link-23');">title</a></tt><tt class="py-op">)</tt> </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line">        <tt class="py-comment"># Do line wrapping</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-119" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-119', 'i', 'link-119');">i</a></tt> <tt class="py-keyword">in</tt> <tt id="link-120" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-120', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-121" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-121', 'seq', 'link-24');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">60</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line">            <tt class="py-name">out_handle</tt><tt class="py-op">.</tt><tt id="link-122" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-122', 'write', 'link-9');">write</a></tt><tt class="py-op">(</tt><tt id="link-123" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-123', 'seq', 'link-24');">seq</a></tt><tt class="py-op">[</tt><tt id="link-124" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-124', 'i', 'link-119');">i</a></tt><tt class="py-op">:</tt><tt id="link-125" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-125', 'i', 'link-119');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-number">60</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-126" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-126', 'count', 'link-22');">count</a></tt> </tt>
</div><a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line"> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line"> </tt>
<a name="_fastq_convert_tab"></a><div id="_fastq_convert_tab-def"><a name="L258"></a><tt class="py-lineno">258</tt> <a class="py-toggle" href="#" id="_fastq_convert_tab-toggle" onclick="return toggle('_fastq_convert_tab');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO._convert-module.html#_fastq_convert_tab">_fastq_convert_tab</a><tt class="py-op">(</tt><tt class="py-param">in_handle</tt><tt class="py-op">,</tt> <tt class="py-param">out_handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_fastq_convert_tab-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_fastq_convert_tab-expanded"><a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line">    <tt class="py-docstring">"""Fast FASTQ to simple tabbed conversion (PRIVATE).</tt> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line"><tt class="py-docstring">    Avoids dealing with the FASTQ quality encoding, and creating SeqRecord and</tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line"><tt class="py-docstring">    Seq objects in order to speed up this conversion.</tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line"><tt class="py-docstring">    NOTE - This does NOT check the characters used in the FASTQ quality string</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line"><tt class="py-docstring">    are valid!</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-127" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-127', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-128" class="py-name"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-128', 'SeqIO', 'link-1');">SeqIO</a></tt><tt class="py-op">.</tt><tt id="link-129" class="py-name"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-129', 'QualityIO', 'link-20');">QualityIO</a></tt> <tt class="py-keyword">import</tt> <tt id="link-130" class="py-name"><a title="Bio.SeqIO.QualityIO.FastqGeneralIterator" class="py-name" href="#" onclick="return doclink('link-130', 'FastqGeneralIterator', 'link-21');">FastqGeneralIterator</a></tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line">    <tt class="py-comment"># For real speed, don't even make SeqRecord and Seq objects!</tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">    <tt id="link-131" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-131', 'count', 'link-22');">count</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt id="link-132" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-132', 'title', 'link-23');">title</a></tt><tt class="py-op">,</tt> <tt id="link-133" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-133', 'seq', 'link-24');">seq</a></tt><tt class="py-op">,</tt> <tt class="py-name">qual</tt> <tt class="py-keyword">in</tt> <tt id="link-134" class="py-name"><a title="Bio.SeqIO.QualityIO.FastqGeneralIterator" class="py-name" href="#" onclick="return doclink('link-134', 'FastqGeneralIterator', 'link-21');">FastqGeneralIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">in_handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line">        <tt id="link-135" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-135', 'count', 'link-22');">count</a></tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line">        <tt class="py-name">out_handle</tt><tt class="py-op">.</tt><tt id="link-136" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-136', 'write', 'link-9');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"%s\t%s\n"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-137" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-137', 'title', 'link-23');">title</a></tt><tt class="py-op">.</tt><tt id="link-138" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-138', 'split', 'link-138');">split</a></tt><tt class="py-op">(</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt id="link-139" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-139', 'seq', 'link-24');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-140" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-140', 'count', 'link-22');">count</a></tt> </tt>
</div><a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line"> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line"> </tt>
<a name="_fastq_convert_qual"></a><div id="_fastq_convert_qual-def"><a name="L276"></a><tt class="py-lineno">276</tt> <a class="py-toggle" href="#" id="_fastq_convert_qual-toggle" onclick="return toggle('_fastq_convert_qual');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO._convert-module.html#_fastq_convert_qual">_fastq_convert_qual</a><tt class="py-op">(</tt><tt class="py-param">in_handle</tt><tt class="py-op">,</tt> <tt class="py-param">out_handle</tt><tt class="py-op">,</tt> <tt class="py-param">mapping</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_fastq_convert_qual-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_fastq_convert_qual-expanded"><a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line">    <tt class="py-docstring">"""FASTQ helper function for QUAL output (PRIVATE).</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line"><tt class="py-docstring">    Mapping should be a dictionary mapping expected ASCII characters from the</tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line"><tt class="py-docstring">    FASTQ quality string to PHRED quality scores (as strings).</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-141" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-141', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-142" class="py-name"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-142', 'SeqIO', 'link-1');">SeqIO</a></tt><tt class="py-op">.</tt><tt id="link-143" class="py-name"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-143', 'QualityIO', 'link-20');">QualityIO</a></tt> <tt class="py-keyword">import</tt> <tt id="link-144" class="py-name"><a title="Bio.SeqIO.QualityIO.FastqGeneralIterator" class="py-name" href="#" onclick="return doclink('link-144', 'FastqGeneralIterator', 'link-21');">FastqGeneralIterator</a></tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line">    <tt class="py-comment"># For real speed, don't even make SeqRecord and Seq objects!</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">    <tt id="link-145" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-145', 'count', 'link-22');">count</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt id="link-146" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-146', 'title', 'link-23');">title</a></tt><tt class="py-op">,</tt> <tt id="link-147" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-147', 'seq', 'link-24');">seq</a></tt><tt class="py-op">,</tt> <tt class="py-name">qual</tt> <tt class="py-keyword">in</tt> <tt id="link-148" class="py-name"><a title="Bio.SeqIO.QualityIO.FastqGeneralIterator" class="py-name" href="#" onclick="return doclink('link-148', 'FastqGeneralIterator', 'link-21');">FastqGeneralIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">in_handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line">        <tt id="link-149" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-149', 'count', 'link-22');">count</a></tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">        <tt class="py-name">out_handle</tt><tt class="py-op">.</tt><tt id="link-150" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-150', 'write', 'link-9');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"&gt;%s\n"</tt> <tt class="py-op">%</tt> <tt id="link-151" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-151', 'title', 'link-23');">title</a></tt><tt class="py-op">)</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">        <tt class="py-comment"># map the qual... note even with Sanger encoding max 2 digits</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">            <tt class="py-name">qualities_strs</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">mapping</tt><tt class="py-op">[</tt><tt class="py-name">ascii</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ascii</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qual</tt><tt class="py-op">]</tt> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid character in quality string"</tt><tt class="py-op">)</tt> </tt>
<a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line">        <tt id="link-152" class="py-name" targets="Variable BioSQL.BioSeq.DBSeq.data=BioSQL.BioSeq.DBSeq-class.html#data"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-152', 'data', 'link-152');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-string">" "</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">qualities_strs</tt><tt class="py-op">)</tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-153" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-153', 'data', 'link-152');">data</a></tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">60</tt><tt class="py-op">:</tt> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line">            <tt class="py-comment"># Know quality scores are either 1 or 2 digits, so there</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line">            <tt class="py-comment"># must be a space in any three consecutive characters.</tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-154" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-154', 'data', 'link-152');">data</a></tt><tt class="py-op">[</tt><tt class="py-number">60</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">" "</tt><tt class="py-op">:</tt> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line">                <tt class="py-name">out_handle</tt><tt class="py-op">.</tt><tt id="link-155" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-155', 'write', 'link-9');">write</a></tt><tt class="py-op">(</tt><tt id="link-156" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-156', 'data', 'link-152');">data</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">60</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line">                <tt id="link-157" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-157', 'data', 'link-152');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-158" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-158', 'data', 'link-152');">data</a></tt><tt class="py-op">[</tt><tt class="py-number">61</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-159" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-159', 'data', 'link-152');">data</a></tt><tt class="py-op">[</tt><tt class="py-number">59</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">" "</tt><tt class="py-op">:</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line">                <tt class="py-name">out_handle</tt><tt class="py-op">.</tt><tt id="link-160" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-160', 'write', 'link-9');">write</a></tt><tt class="py-op">(</tt><tt id="link-161" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-161', 'data', 'link-152');">data</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">59</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">                <tt id="link-162" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-162', 'data', 'link-152');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-163" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-163', 'data', 'link-152');">data</a></tt><tt class="py-op">[</tt><tt class="py-number">60</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line">                <tt class="py-keyword">assert</tt> <tt id="link-164" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-164', 'data', 'link-152');">data</a></tt><tt class="py-op">[</tt><tt class="py-number">58</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">" "</tt><tt class="py-op">,</tt> <tt class="py-string">"Internal logic failure in wrapping"</tt> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line">                <tt class="py-name">out_handle</tt><tt class="py-op">.</tt><tt id="link-165" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-165', 'write', 'link-9');">write</a></tt><tt class="py-op">(</tt><tt id="link-166" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-166', 'data', 'link-152');">data</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">58</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line">                <tt id="link-167" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-167', 'data', 'link-152');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-168" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-168', 'data', 'link-152');">data</a></tt><tt class="py-op">[</tt><tt class="py-number">59</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">        <tt class="py-name">out_handle</tt><tt class="py-op">.</tt><tt id="link-169" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-169', 'write', 'link-9');">write</a></tt><tt class="py-op">(</tt><tt id="link-170" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-170', 'data', 'link-152');">data</a></tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-171" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-171', 'count', 'link-22');">count</a></tt> </tt>
</div><a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line"> </tt>
<a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line"> </tt>
<a name="_fastq_sanger_convert_qual"></a><div id="_fastq_sanger_convert_qual-def"><a name="L311"></a><tt class="py-lineno">311</tt> <a class="py-toggle" href="#" id="_fastq_sanger_convert_qual-toggle" onclick="return toggle('_fastq_sanger_convert_qual');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO._convert-module.html#_fastq_sanger_convert_qual">_fastq_sanger_convert_qual</a><tt class="py-op">(</tt><tt class="py-param">in_handle</tt><tt class="py-op">,</tt> <tt class="py-param">out_handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_fastq_sanger_convert_qual-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_fastq_sanger_convert_qual-expanded"><a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line">    <tt class="py-docstring">"""Fast Sanger FASTQ to QUAL conversion (PRIVATE)."""</tt> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line">    <tt class="py-name">mapping</tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-name">q</tt> <tt class="py-op">+</tt> <tt class="py-number">33</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">q</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">q</tt> <tt class="py-keyword">in</tt> <tt id="link-172" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-172', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">93</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-173" class="py-name" targets="Function Bio.SeqIO._convert._fastq_convert_qual()=Bio.SeqIO._convert-module.html#_fastq_convert_qual"><a title="Bio.SeqIO._convert._fastq_convert_qual" class="py-name" href="#" onclick="return doclink('link-173', '_fastq_convert_qual', 'link-173');">_fastq_convert_qual</a></tt><tt class="py-op">(</tt><tt class="py-name">in_handle</tt><tt class="py-op">,</tt> <tt class="py-name">out_handle</tt><tt class="py-op">,</tt> <tt class="py-name">mapping</tt><tt class="py-op">)</tt> </tt>
</div><a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line"> </tt>
<a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line"> </tt>
<a name="_fastq_solexa_convert_qual"></a><div id="_fastq_solexa_convert_qual-def"><a name="L317"></a><tt class="py-lineno">317</tt> <a class="py-toggle" href="#" id="_fastq_solexa_convert_qual-toggle" onclick="return toggle('_fastq_solexa_convert_qual');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO._convert-module.html#_fastq_solexa_convert_qual">_fastq_solexa_convert_qual</a><tt class="py-op">(</tt><tt class="py-param">in_handle</tt><tt class="py-op">,</tt> <tt class="py-param">out_handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_fastq_solexa_convert_qual-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_fastq_solexa_convert_qual-expanded"><a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">    <tt class="py-docstring">"""Fast Solexa FASTQ to QUAL conversion (PRIVATE)."""</tt> </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-174" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-174', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-175" class="py-name"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-175', 'SeqIO', 'link-1');">SeqIO</a></tt><tt class="py-op">.</tt><tt id="link-176" class="py-name"><a title="Bio.SeqIO.QualityIO" class="py-name" href="#" onclick="return doclink('link-176', 'QualityIO', 'link-20');">QualityIO</a></tt> <tt class="py-keyword">import</tt> <tt id="link-177" class="py-name"><a title="Bio.SeqIO.QualityIO.phred_quality_from_solexa" class="py-name" href="#" onclick="return doclink('link-177', 'phred_quality_from_solexa', 'link-70');">phred_quality_from_solexa</a></tt> </tt>
<a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line">    <tt class="py-name">mapping</tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-name">q</tt> <tt class="py-op">+</tt> <tt class="py-number">64</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-178" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.round
Bio.Blast.NCBIStandalone._DescriptionConsumer.round" class="py-name" href="#" onclick="return doclink('link-178', 'round', 'link-72');">round</a></tt><tt class="py-op">(</tt><tt id="link-179" class="py-name"><a title="Bio.SeqIO.QualityIO.phred_quality_from_solexa" class="py-name" href="#" onclick="return doclink('link-179', 'phred_quality_from_solexa', 'link-70');">phred_quality_from_solexa</a></tt><tt class="py-op">(</tt><tt class="py-name">q</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line">                   <tt class="py-keyword">for</tt> <tt class="py-name">q</tt> <tt class="py-keyword">in</tt> <tt id="link-180" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-180', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">5</tt><tt class="py-op">,</tt> <tt class="py-number">62</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-181" class="py-name"><a title="Bio.SeqIO._convert._fastq_convert_qual" class="py-name" href="#" onclick="return doclink('link-181', '_fastq_convert_qual', 'link-173');">_fastq_convert_qual</a></tt><tt class="py-op">(</tt><tt class="py-name">in_handle</tt><tt class="py-op">,</tt> <tt class="py-name">out_handle</tt><tt class="py-op">,</tt> <tt class="py-name">mapping</tt><tt class="py-op">)</tt> </tt>
</div><a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line"> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line"> </tt>
<a name="_fastq_illumina_convert_qual"></a><div id="_fastq_illumina_convert_qual-def"><a name="L325"></a><tt class="py-lineno">325</tt> <a class="py-toggle" href="#" id="_fastq_illumina_convert_qual-toggle" onclick="return toggle('_fastq_illumina_convert_qual');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO._convert-module.html#_fastq_illumina_convert_qual">_fastq_illumina_convert_qual</a><tt class="py-op">(</tt><tt class="py-param">in_handle</tt><tt class="py-op">,</tt> <tt class="py-param">out_handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_fastq_illumina_convert_qual-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_fastq_illumina_convert_qual-expanded"><a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">    <tt class="py-docstring">"""Fast Illumina 1.3+ FASTQ to QUAL conversion (PRIVATE)."""</tt> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line">    <tt class="py-name">mapping</tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-name">q</tt> <tt class="py-op">+</tt> <tt class="py-number">64</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">q</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">q</tt> <tt class="py-keyword">in</tt> <tt id="link-182" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-182', 'range', 'link-46');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">62</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-183" class="py-name"><a title="Bio.SeqIO._convert._fastq_convert_qual" class="py-name" href="#" onclick="return doclink('link-183', '_fastq_convert_qual', 'link-173');">_fastq_convert_qual</a></tt><tt class="py-op">(</tt><tt class="py-name">in_handle</tt><tt class="py-op">,</tt> <tt class="py-name">out_handle</tt><tt class="py-op">,</tt> <tt class="py-name">mapping</tt><tt class="py-op">)</tt> </tt>
</div><a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line"> </tt>
<a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line"> </tt>
<a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line"><tt class="py-comment"># TODO? - Handling aliases explicitly would let us shorten this list:</tt> </tt>
<a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line"><tt id="link-184" class="py-name" targets="Variable Bio.SeqIO._convert._converter=Bio.SeqIO._convert-module.html#_converter"><a title="Bio.SeqIO._convert._converter" class="py-name" href="#" onclick="return doclink('link-184', '_converter', 'link-184');">_converter</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"genbank"</tt><tt class="py-op">,</tt> <tt class="py-string">"fasta"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-185" class="py-name" targets="Function Bio.SeqIO._convert._genbank_convert_fasta()=Bio.SeqIO._convert-module.html#_genbank_convert_fasta"><a title="Bio.SeqIO._convert._genbank_convert_fasta" class="py-name" href="#" onclick="return doclink('link-185', '_genbank_convert_fasta', 'link-185');">_genbank_convert_fasta</a></tt><tt class="py-op">,</tt> </tt>
<a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"gb"</tt><tt class="py-op">,</tt> <tt class="py-string">"fasta"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-186" class="py-name"><a title="Bio.SeqIO._convert._genbank_convert_fasta" class="py-name" href="#" onclick="return doclink('link-186', '_genbank_convert_fasta', 'link-185');">_genbank_convert_fasta</a></tt><tt class="py-op">,</tt> </tt>
<a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"embl"</tt><tt class="py-op">,</tt> <tt class="py-string">"fasta"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-187" class="py-name" targets="Function Bio.SeqIO._convert._embl_convert_fasta()=Bio.SeqIO._convert-module.html#_embl_convert_fasta"><a title="Bio.SeqIO._convert._embl_convert_fasta" class="py-name" href="#" onclick="return doclink('link-187', '_embl_convert_fasta', 'link-187');">_embl_convert_fasta</a></tt><tt class="py-op">,</tt> </tt>
<a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq"</tt><tt class="py-op">,</tt> <tt class="py-string">"fasta"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-188" class="py-name" targets="Function Bio.SeqIO._convert._fastq_convert_fasta()=Bio.SeqIO._convert-module.html#_fastq_convert_fasta"><a title="Bio.SeqIO._convert._fastq_convert_fasta" class="py-name" href="#" onclick="return doclink('link-188', '_fastq_convert_fasta', 'link-188');">_fastq_convert_fasta</a></tt><tt class="py-op">,</tt> </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq-sanger"</tt><tt class="py-op">,</tt> <tt class="py-string">"fasta"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-189" class="py-name"><a title="Bio.SeqIO._convert._fastq_convert_fasta" class="py-name" href="#" onclick="return doclink('link-189', '_fastq_convert_fasta', 'link-188');">_fastq_convert_fasta</a></tt><tt class="py-op">,</tt> </tt>
<a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq-solexa"</tt><tt class="py-op">,</tt> <tt class="py-string">"fasta"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-190" class="py-name"><a title="Bio.SeqIO._convert._fastq_convert_fasta" class="py-name" href="#" onclick="return doclink('link-190', '_fastq_convert_fasta', 'link-188');">_fastq_convert_fasta</a></tt><tt class="py-op">,</tt> </tt>
<a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq-illumina"</tt><tt class="py-op">,</tt> <tt class="py-string">"fasta"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-191" class="py-name"><a title="Bio.SeqIO._convert._fastq_convert_fasta" class="py-name" href="#" onclick="return doclink('link-191', '_fastq_convert_fasta', 'link-188');">_fastq_convert_fasta</a></tt><tt class="py-op">,</tt> </tt>
<a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq"</tt><tt class="py-op">,</tt> <tt class="py-string">"tab"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-192" class="py-name" targets="Function Bio.SeqIO._convert._fastq_convert_tab()=Bio.SeqIO._convert-module.html#_fastq_convert_tab"><a title="Bio.SeqIO._convert._fastq_convert_tab" class="py-name" href="#" onclick="return doclink('link-192', '_fastq_convert_tab', 'link-192');">_fastq_convert_tab</a></tt><tt class="py-op">,</tt> </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq-sanger"</tt><tt class="py-op">,</tt> <tt class="py-string">"tab"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-193" class="py-name"><a title="Bio.SeqIO._convert._fastq_convert_tab" class="py-name" href="#" onclick="return doclink('link-193', '_fastq_convert_tab', 'link-192');">_fastq_convert_tab</a></tt><tt class="py-op">,</tt> </tt>
<a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq-solexa"</tt><tt class="py-op">,</tt> <tt class="py-string">"tab"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-194" class="py-name"><a title="Bio.SeqIO._convert._fastq_convert_tab" class="py-name" href="#" onclick="return doclink('link-194', '_fastq_convert_tab', 'link-192');">_fastq_convert_tab</a></tt><tt class="py-op">,</tt> </tt>
<a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq-illumina"</tt><tt class="py-op">,</tt> <tt class="py-string">"tab"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-195" class="py-name"><a title="Bio.SeqIO._convert._fastq_convert_tab" class="py-name" href="#" onclick="return doclink('link-195', '_fastq_convert_tab', 'link-192');">_fastq_convert_tab</a></tt><tt class="py-op">,</tt> </tt>
<a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq"</tt><tt class="py-op">,</tt> <tt class="py-string">"fastq"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-196" class="py-name" targets="Function Bio.SeqIO._convert._fastq_sanger_convert_fastq_sanger()=Bio.SeqIO._convert-module.html#_fastq_sanger_convert_fastq_sanger"><a title="Bio.SeqIO._convert._fastq_sanger_convert_fastq_sanger" class="py-name" href="#" onclick="return doclink('link-196', '_fastq_sanger_convert_fastq_sanger', 'link-196');">_fastq_sanger_convert_fastq_sanger</a></tt><tt class="py-op">,</tt> </tt>
<a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq-sanger"</tt><tt class="py-op">,</tt> <tt class="py-string">"fastq"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-197" class="py-name"><a title="Bio.SeqIO._convert._fastq_sanger_convert_fastq_sanger" class="py-name" href="#" onclick="return doclink('link-197', '_fastq_sanger_convert_fastq_sanger', 'link-196');">_fastq_sanger_convert_fastq_sanger</a></tt><tt class="py-op">,</tt> </tt>
<a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq-solexa"</tt><tt class="py-op">,</tt> <tt class="py-string">"fastq"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-198" class="py-name" targets="Function Bio.SeqIO._convert._fastq_solexa_convert_fastq_sanger()=Bio.SeqIO._convert-module.html#_fastq_solexa_convert_fastq_sanger"><a title="Bio.SeqIO._convert._fastq_solexa_convert_fastq_sanger" class="py-name" href="#" onclick="return doclink('link-198', '_fastq_solexa_convert_fastq_sanger', 'link-198');">_fastq_solexa_convert_fastq_sanger</a></tt><tt class="py-op">,</tt> </tt>
<a name="L347"></a><tt class="py-lineno">347</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq-illumina"</tt><tt class="py-op">,</tt> <tt class="py-string">"fastq"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-199" class="py-name" targets="Function Bio.SeqIO._convert._fastq_illumina_convert_fastq_sanger()=Bio.SeqIO._convert-module.html#_fastq_illumina_convert_fastq_sanger"><a title="Bio.SeqIO._convert._fastq_illumina_convert_fastq_sanger" class="py-name" href="#" onclick="return doclink('link-199', '_fastq_illumina_convert_fastq_sanger', 'link-199');">_fastq_illumina_convert_fastq_sanger</a></tt><tt class="py-op">,</tt> </tt>
<a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq"</tt><tt class="py-op">,</tt> <tt class="py-string">"fastq-sanger"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-200" class="py-name"><a title="Bio.SeqIO._convert._fastq_sanger_convert_fastq_sanger" class="py-name" href="#" onclick="return doclink('link-200', '_fastq_sanger_convert_fastq_sanger', 'link-196');">_fastq_sanger_convert_fastq_sanger</a></tt><tt class="py-op">,</tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq-sanger"</tt><tt class="py-op">,</tt> <tt class="py-string">"fastq-sanger"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-201" class="py-name"><a title="Bio.SeqIO._convert._fastq_sanger_convert_fastq_sanger" class="py-name" href="#" onclick="return doclink('link-201', '_fastq_sanger_convert_fastq_sanger', 'link-196');">_fastq_sanger_convert_fastq_sanger</a></tt><tt class="py-op">,</tt> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq-solexa"</tt><tt class="py-op">,</tt> <tt class="py-string">"fastq-sanger"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-202" class="py-name"><a title="Bio.SeqIO._convert._fastq_solexa_convert_fastq_sanger" class="py-name" href="#" onclick="return doclink('link-202', '_fastq_solexa_convert_fastq_sanger', 'link-198');">_fastq_solexa_convert_fastq_sanger</a></tt><tt class="py-op">,</tt> </tt>
<a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq-illumina"</tt><tt class="py-op">,</tt> <tt class="py-string">"fastq-sanger"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-203" class="py-name"><a title="Bio.SeqIO._convert._fastq_illumina_convert_fastq_sanger" class="py-name" href="#" onclick="return doclink('link-203', '_fastq_illumina_convert_fastq_sanger', 'link-199');">_fastq_illumina_convert_fastq_sanger</a></tt><tt class="py-op">,</tt> </tt>
<a name="L352"></a><tt class="py-lineno">352</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq"</tt><tt class="py-op">,</tt> <tt class="py-string">"fastq-solexa"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-204" class="py-name" targets="Function Bio.SeqIO._convert._fastq_sanger_convert_fastq_solexa()=Bio.SeqIO._convert-module.html#_fastq_sanger_convert_fastq_solexa"><a title="Bio.SeqIO._convert._fastq_sanger_convert_fastq_solexa" class="py-name" href="#" onclick="return doclink('link-204', '_fastq_sanger_convert_fastq_solexa', 'link-204');">_fastq_sanger_convert_fastq_solexa</a></tt><tt class="py-op">,</tt> </tt>
<a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq-sanger"</tt><tt class="py-op">,</tt> <tt class="py-string">"fastq-solexa"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-205" class="py-name"><a title="Bio.SeqIO._convert._fastq_sanger_convert_fastq_solexa" class="py-name" href="#" onclick="return doclink('link-205', '_fastq_sanger_convert_fastq_solexa', 'link-204');">_fastq_sanger_convert_fastq_solexa</a></tt><tt class="py-op">,</tt> </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq-solexa"</tt><tt class="py-op">,</tt> <tt class="py-string">"fastq-solexa"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-206" class="py-name" targets="Function Bio.SeqIO._convert._fastq_solexa_convert_fastq_solexa()=Bio.SeqIO._convert-module.html#_fastq_solexa_convert_fastq_solexa"><a title="Bio.SeqIO._convert._fastq_solexa_convert_fastq_solexa" class="py-name" href="#" onclick="return doclink('link-206', '_fastq_solexa_convert_fastq_solexa', 'link-206');">_fastq_solexa_convert_fastq_solexa</a></tt><tt class="py-op">,</tt> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq-illumina"</tt><tt class="py-op">,</tt> <tt class="py-string">"fastq-solexa"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-207" class="py-name" targets="Function Bio.SeqIO._convert._fastq_illumina_convert_fastq_solexa()=Bio.SeqIO._convert-module.html#_fastq_illumina_convert_fastq_solexa"><a title="Bio.SeqIO._convert._fastq_illumina_convert_fastq_solexa" class="py-name" href="#" onclick="return doclink('link-207', '_fastq_illumina_convert_fastq_solexa', 'link-207');">_fastq_illumina_convert_fastq_solexa</a></tt><tt class="py-op">,</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq"</tt><tt class="py-op">,</tt> <tt class="py-string">"fastq-illumina"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-208" class="py-name" targets="Function Bio.SeqIO._convert._fastq_sanger_convert_fastq_illumina()=Bio.SeqIO._convert-module.html#_fastq_sanger_convert_fastq_illumina"><a title="Bio.SeqIO._convert._fastq_sanger_convert_fastq_illumina" class="py-name" href="#" onclick="return doclink('link-208', '_fastq_sanger_convert_fastq_illumina', 'link-208');">_fastq_sanger_convert_fastq_illumina</a></tt><tt class="py-op">,</tt> </tt>
<a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq-sanger"</tt><tt class="py-op">,</tt> <tt class="py-string">"fastq-illumina"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-209" class="py-name"><a title="Bio.SeqIO._convert._fastq_sanger_convert_fastq_illumina" class="py-name" href="#" onclick="return doclink('link-209', '_fastq_sanger_convert_fastq_illumina', 'link-208');">_fastq_sanger_convert_fastq_illumina</a></tt><tt class="py-op">,</tt> </tt>
<a name="L358"></a><tt class="py-lineno">358</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq-solexa"</tt><tt class="py-op">,</tt> <tt class="py-string">"fastq-illumina"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-210" class="py-name" targets="Function Bio.SeqIO._convert._fastq_solexa_convert_fastq_illumina()=Bio.SeqIO._convert-module.html#_fastq_solexa_convert_fastq_illumina"><a title="Bio.SeqIO._convert._fastq_solexa_convert_fastq_illumina" class="py-name" href="#" onclick="return doclink('link-210', '_fastq_solexa_convert_fastq_illumina', 'link-210');">_fastq_solexa_convert_fastq_illumina</a></tt><tt class="py-op">,</tt> </tt>
<a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq-illumina"</tt><tt class="py-op">,</tt> <tt class="py-string">"fastq-illumina"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-211" class="py-name" targets="Function Bio.SeqIO._convert._fastq_illumina_convert_fastq_illumina()=Bio.SeqIO._convert-module.html#_fastq_illumina_convert_fastq_illumina"><a title="Bio.SeqIO._convert._fastq_illumina_convert_fastq_illumina" class="py-name" href="#" onclick="return doclink('link-211', '_fastq_illumina_convert_fastq_illumina', 'link-211');">_fastq_illumina_convert_fastq_illumina</a></tt><tt class="py-op">,</tt> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq"</tt><tt class="py-op">,</tt> <tt class="py-string">"qual"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-212" class="py-name" targets="Function Bio.SeqIO._convert._fastq_sanger_convert_qual()=Bio.SeqIO._convert-module.html#_fastq_sanger_convert_qual"><a title="Bio.SeqIO._convert._fastq_sanger_convert_qual" class="py-name" href="#" onclick="return doclink('link-212', '_fastq_sanger_convert_qual', 'link-212');">_fastq_sanger_convert_qual</a></tt><tt class="py-op">,</tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq-sanger"</tt><tt class="py-op">,</tt> <tt class="py-string">"qual"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-213" class="py-name"><a title="Bio.SeqIO._convert._fastq_sanger_convert_qual" class="py-name" href="#" onclick="return doclink('link-213', '_fastq_sanger_convert_qual', 'link-212');">_fastq_sanger_convert_qual</a></tt><tt class="py-op">,</tt> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq-solexa"</tt><tt class="py-op">,</tt> <tt class="py-string">"qual"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-214" class="py-name" targets="Function Bio.SeqIO._convert._fastq_solexa_convert_qual()=Bio.SeqIO._convert-module.html#_fastq_solexa_convert_qual"><a title="Bio.SeqIO._convert._fastq_solexa_convert_qual" class="py-name" href="#" onclick="return doclink('link-214', '_fastq_solexa_convert_qual', 'link-214');">_fastq_solexa_convert_qual</a></tt><tt class="py-op">,</tt> </tt>
<a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-string">"fastq-illumina"</tt><tt class="py-op">,</tt> <tt class="py-string">"qual"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt id="link-215" class="py-name" targets="Function Bio.SeqIO._convert._fastq_illumina_convert_qual()=Bio.SeqIO._convert-module.html#_fastq_illumina_convert_qual"><a title="Bio.SeqIO._convert._fastq_illumina_convert_qual" class="py-name" href="#" onclick="return doclink('link-215', '_fastq_illumina_convert_qual', 'link-215');">_fastq_illumina_convert_qual</a></tt><tt class="py-op">,</tt> </tt>
<a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line"><tt class="py-op">}</tt> </tt>
<a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line"> </tt>
<a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line"> </tt>
<a name="_handle_convert"></a><div id="_handle_convert-def"><a name="L367"></a><tt class="py-lineno">367</tt> <a class="py-toggle" href="#" id="_handle_convert-toggle" onclick="return toggle('_handle_convert');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO._convert-module.html#_handle_convert">_handle_convert</a><tt class="py-op">(</tt><tt class="py-param">in_handle</tt><tt class="py-op">,</tt> <tt class="py-param">in_format</tt><tt class="py-op">,</tt> <tt class="py-param">out_handle</tt><tt class="py-op">,</tt> <tt class="py-param">out_format</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_handle_convert-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_handle_convert-expanded"><a name="L368"></a><tt class="py-lineno">368</tt>  <tt class="py-line">    <tt class="py-docstring">"""SeqIO conversion function (PRIVATE)."""</tt> </tt>
<a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line">        <tt class="py-name">f</tt> <tt class="py-op">=</tt> <tt id="link-216" class="py-name"><a title="Bio.SeqIO._convert._converter" class="py-name" href="#" onclick="return doclink('link-216', '_converter', 'link-184');">_converter</a></tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">in_format</tt><tt class="py-op">,</tt> <tt class="py-name">out_format</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line">        <tt class="py-name">f</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">f</tt><tt class="py-op">:</tt> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">f</tt><tt class="py-op">(</tt><tt class="py-name">in_handle</tt><tt class="py-op">,</tt> <tt class="py-name">out_handle</tt><tt class="py-op">,</tt> <tt id="link-217" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.alphabet=Bio.SearchIO._model.hsp.HSP-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSPFragment.alphabet=Bio.SearchIO._model.hsp.HSPFragment-class.html#alphabet"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-217', 'alphabet', 'link-217');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line">        <tt class="py-name">records</tt> <tt class="py-op">=</tt> <tt id="link-218" class="py-name"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-218', 'SeqIO', 'link-1');">SeqIO</a></tt><tt class="py-op">.</tt><tt id="link-219" class="py-name" targets="Function Bio.AlignIO.parse()=Bio.AlignIO-module.html#parse,Method Bio.Blast.NCBIStandalone.BlastErrorParser.parse()=Bio.Blast.NCBIStandalone.BlastErrorParser-class.html#parse,Method Bio.Blast.NCBIStandalone.BlastParser.parse()=Bio.Blast.NCBIStandalone.BlastParser-class.html#parse,Method Bio.Blast.NCBIStandalone.PSIBlastParser.parse()=Bio.Blast.NCBIStandalone.PSIBlastParser-class.html#parse,Function Bio.Blast.NCBIXML.parse()=Bio.Blast.NCBIXML-module.html#parse,Function Bio.Compass.parse()=Bio.Compass-module.html#parse,Function Bio.Emboss.Primer3.parse()=Bio.Emboss.Primer3-module.html#parse,Method Bio.Entrez.Parser.DataHandler.parse()=Bio.Entrez.Parser.DataHandler-class.html#parse,Function Bio.Entrez.parse()=Bio.Entrez-module.html#parse,Function Bio.ExPASy.Enzyme.parse()=Bio.ExPASy.Enzyme-module.html#parse,Function Bio.ExPASy.Prodoc.parse()=Bio.ExPASy.Prodoc-module.html#parse,Function Bio.ExPASy.Prosite.parse()=Bio.ExPASy.Prosite-module.html#parse,Method Bio.GenBank.FeatureParser.parse()=Bio.GenBank.FeatureParser-class.html#parse,Method Bio.GenBank.RecordParser.parse()=Bio.GenBank.RecordParser-class.html#parse,Method Bio.GenBank.Scanner.InsdcScanner.parse()=Bio.GenBank.Scanner.InsdcScanner-class.html#parse,Function Bio.Geo.parse()=Bio.Geo-module.html#parse,Function Bio.KEGG.Compound.parse()=Bio.KEGG.Compound-module.html#parse,Function Bio.KEGG.Enzyme.parse()=Bio.KEGG.Enzyme-module.html#parse,Method Bio.KEGG.KGML.KGML_parser.KGMLParser.parse()=Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html#parse,Function Bio.KEGG.KGML.KGML_parser.parse()=Bio.KEGG.KGML.KGML_parser-module.html#parse,Function Bio.KEGG.Map.parse()=Bio.KEGG.Map-module.html#parse,Function Bio.Medline.parse()=Bio.Medline-module.html#parse,Function Bio.Motif.parse()=Bio.Motif-module.html#parse,Method Bio.ParserSupport.AbstractParser.parse()=Bio.ParserSupport.AbstractParser-class.html#parse,Method Bio.Phylo.CDAOIO.Parser.parse()=Bio.Phylo.CDAOIO.Parser-class.html#parse,Function Bio.Phylo.CDAOIO.parse()=Bio.Phylo.CDAOIO-module.html#parse,Method Bio.Phylo.NeXMLIO.Parser.parse()=Bio.Phylo.NeXMLIO.Parser-class.html#parse,Function Bio.Phylo.NeXMLIO.parse()=Bio.Phylo.NeXMLIO-module.html#parse,Method Bio.Phylo.NewickIO.Parser.parse()=Bio.Phylo.NewickIO.Parser-class.html#parse,Function Bio.Phylo.NewickIO.parse()=Bio.Phylo.NewickIO-module.html#parse,Function Bio.Phylo.NexusIO.parse()=Bio.Phylo.NexusIO-module.html#parse,Method Bio.Phylo.PhyloXMLIO.Parser.parse()=Bio.Phylo.PhyloXMLIO.Parser-class.html#parse,Function Bio.Phylo.PhyloXMLIO.parse()=Bio.Phylo.PhyloXMLIO-module.html#parse,Function Bio.Phylo._io.parse()=Bio.Phylo._io-module.html#parse,Function Bio.SCOP.Cla.parse()=Bio.SCOP.Cla-module.html#parse,Function Bio.SCOP.Des.parse()=Bio.SCOP.Des-module.html#parse,Function Bio.SCOP.Dom.parse()=Bio.SCOP.Dom-module.html#parse,Function Bio.SCOP.Hie.parse()=Bio.SCOP.Hie-module.html#parse,Function Bio.SCOP.Raf.parse()=Bio.SCOP.Raf-module.html#parse,Function Bio.SearchIO.parse()=Bio.SearchIO-module.html#parse,Method Bio.SeqIO.UniprotIO.Parser.parse()=Bio.SeqIO.UniprotIO.Parser-class.html#parse,Function Bio.SeqIO.parse()=Bio.SeqIO-module.html#parse,Function Bio.Sequencing.Ace.parse()=Bio.Sequencing.Ace-module.html#parse,Function Bio.Sequencing.Phd.parse()=Bio.Sequencing.Phd-module.html#parse,Function Bio.SwissProt.KeyWList.parse()=Bio.SwissProt.KeyWList-module.html#parse,Function Bio.SwissProt.parse()=Bio.SwissProt-module.html#parse,Function Bio.UniGene.parse()=Bio.UniGene-module.html#parse,Function Bio.Wise.psw.parse()=Bio.Wise.psw-module.html#parse,Function Bio.motifs.parse()=Bio.motifs-module.html#parse"><a title="Bio.AlignIO.parse
Bio.Blast.NCBIStandalone.BlastErrorParser.parse
Bio.Blast.NCBIStandalone.BlastParser.parse
Bio.Blast.NCBIStandalone.PSIBlastParser.parse
Bio.Blast.NCBIXML.parse
Bio.Compass.parse
Bio.Emboss.Primer3.parse
Bio.Entrez.Parser.DataHandler.parse
Bio.Entrez.parse
Bio.ExPASy.Enzyme.parse
Bio.ExPASy.Prodoc.parse
Bio.ExPASy.Prosite.parse
Bio.GenBank.FeatureParser.parse
Bio.GenBank.RecordParser.parse
Bio.GenBank.Scanner.InsdcScanner.parse
Bio.Geo.parse
Bio.KEGG.Compound.parse
Bio.KEGG.Enzyme.parse
Bio.KEGG.KGML.KGML_parser.KGMLParser.parse
Bio.KEGG.KGML.KGML_parser.parse
Bio.KEGG.Map.parse
Bio.Medline.parse
Bio.Motif.parse
Bio.ParserSupport.AbstractParser.parse
Bio.Phylo.CDAOIO.Parser.parse
Bio.Phylo.CDAOIO.parse
Bio.Phylo.NeXMLIO.Parser.parse
Bio.Phylo.NeXMLIO.parse
Bio.Phylo.NewickIO.Parser.parse
Bio.Phylo.NewickIO.parse
Bio.Phylo.NexusIO.parse
Bio.Phylo.PhyloXMLIO.Parser.parse
Bio.Phylo.PhyloXMLIO.parse
Bio.Phylo._io.parse
Bio.SCOP.Cla.parse
Bio.SCOP.Des.parse
Bio.SCOP.Dom.parse
Bio.SCOP.Hie.parse
Bio.SCOP.Raf.parse
Bio.SearchIO.parse
Bio.SeqIO.UniprotIO.Parser.parse
Bio.SeqIO.parse
Bio.Sequencing.Ace.parse
Bio.Sequencing.Phd.parse
Bio.SwissProt.KeyWList.parse
Bio.SwissProt.parse
Bio.UniGene.parse
Bio.Wise.psw.parse
Bio.motifs.parse" class="py-name" href="#" onclick="return doclink('link-219', 'parse', 'link-219');">parse</a></tt><tt class="py-op">(</tt><tt class="py-name">in_handle</tt><tt class="py-op">,</tt> <tt class="py-name">in_format</tt><tt class="py-op">,</tt> <tt id="link-220" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-220', 'alphabet', 'link-217');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-221" class="py-name"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-221', 'SeqIO', 'link-1');">SeqIO</a></tt><tt class="py-op">.</tt><tt id="link-222" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-222', 'write', 'link-9');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">records</tt><tt class="py-op">,</tt> <tt class="py-name">out_handle</tt><tt class="py-op">,</tt> <tt class="py-name">out_format</tt><tt class="py-op">)</tt> </tt>
</div><a name="L378"></a><tt class="py-lineno">378</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:04:50 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
