<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.Seq.Seq</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.Seq-module.html">Module&nbsp;Seq</a> ::
        Class&nbsp;Seq
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.Seq.Seq-class.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== CLASS DESCRIPTION ==================== -->
<h1 class="epydoc">Class Seq</h1><p class="nomargin-top"><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq">source&nbsp;code</a></span></p>
<pre class="base-tree">
object --+
         |
        <strong class="uidshort">Seq</strong>
</pre>

<dl><dt>Known Subclasses:</dt>
<dd>
      <ul class="subclass-list">
<li><a href="Bio.Seq.UnknownSeq-class.html">UnknownSeq</a></li><li>, <a href="Bio.codonalign.codonseq.CodonSeq-class.html">codonalign.codonseq.CodonSeq</a></li><li>, <a href="Bio.Motif.Parsers.MEME.MEMEInstance-class.html">Motif.Parsers.MEME.MEMEInstance</a></li><li>, <a href="Bio.motifs.meme.Instance-class.html">motifs.meme.Instance</a></li><li>, <a href="BioSQL.BioSeq.DBSeq-class.html">BioSQL.BioSeq.DBSeq</a></li>  </ul>
</dd></dl>

<hr />
<p>A read-only sequence object (essentially a string with an alphabet).</p>
<p>Like normal python strings, our basic sequence object is immutable.
This prevents you from doing my_seq[5] = &quot;A&quot; for example, but does allow
Seq objects to be used as dictionary keys.</p>
<p>The Seq object provides a number of string like methods (such as count,
find, split and strip), which are alphabet aware where appropriate.</p>
<p>In addition to the string like sequence, the Seq object has an alphabet
property. This is an instance of an Alphabet class from Bio.Alphabet,
for example generic DNA, or IUPAC DNA. This describes the type of molecule
(e.g. RNA, DNA, protein) and may also indicate the expected symbols
(letters).</p>
<p>The Seq object also provides some biological methods, such as complement,
reverse_complement, transcribe, back_transcribe and translate (which are
not applicable to sequences with a protein alphabet).</p>

<!-- ==================== INSTANCE METHODS ==================== -->
<a name="section-InstanceMethods"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Instance Methods</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-InstanceMethods"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#__init__" class="summary-sig-name">__init__</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">data</span>,
        <span class="summary-sig-arg">alphabet</span>=<span class="summary-sig-default">Alphabet()</span>)</span><br />
      Create a Seq object.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.__init__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#__repr__" class="summary-sig-name">__repr__</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Returns a (truncated) representation of the sequence for debugging.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.__repr__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#__str__" class="summary-sig-name">__str__</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Returns the full sequence as a python string, use str(my_seq).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.__str__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#__hash__" class="summary-sig-name">__hash__</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Hash for comparison.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.__hash__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#__eq__" class="summary-sig-name">__eq__</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">other</span>)</span><br />
      Compare the sequence to another sequence or a string (README).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.__eq__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__ne__"></a><span class="summary-sig-name">__ne__</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">other</span>)</span><br />
      Not equal, see __eq__ documentation.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.__ne__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__lt__"></a><span class="summary-sig-name">__lt__</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">other</span>)</span><br />
      Less than, see __eq__ documentation.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.__lt__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__le__"></a><span class="summary-sig-name">__le__</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">other</span>)</span><br />
      Less than or equal, see __eq__ documentation.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.__le__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__len__"></a><span class="summary-sig-name">__len__</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Returns the length of the sequence, use len(my_seq).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.__len__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__getitem__"></a><span class="summary-sig-name">__getitem__</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">index</span>)</span><br />
      Returns a subsequence of single letter, use my_seq[index].</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.__getitem__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#__add__" class="summary-sig-name">__add__</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">other</span>)</span><br />
      Add another sequence or string to this sequence.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.__add__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#__radd__" class="summary-sig-name">__radd__</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">other</span>)</span><br />
      Adding a sequence on the left.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.__radd__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#tostring" class="summary-sig-name">tostring</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Returns the full sequence as a python string (DEPRECATED).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.tostring">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#tomutable" class="summary-sig-name">tomutable</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Returns the full sequence as a MutableSeq object.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.tomutable">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#_get_seq_str_and_check_alphabet" class="summary-sig-name" onclick="show_private();">_get_seq_str_and_check_alphabet</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">other_sequence</span>)</span><br />
      string/Seq/MutableSeq to string, checking alphabet (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq._get_seq_str_and_check_alphabet">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#count" class="summary-sig-name">count</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">sub</span>,
        <span class="summary-sig-arg">start</span>=<span class="summary-sig-default">0</span>,
        <span class="summary-sig-arg">end</span>=<span class="summary-sig-default">9223372036854775807</span>)</span><br />
      Non-overlapping count method, like that of a python string.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.count">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#__contains__" class="summary-sig-name">__contains__</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">char</span>)</span><br />
      Implements the 'in' keyword, like a python string.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.__contains__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#find" class="summary-sig-name">find</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">sub</span>,
        <span class="summary-sig-arg">start</span>=<span class="summary-sig-default">0</span>,
        <span class="summary-sig-arg">end</span>=<span class="summary-sig-default">9223372036854775807</span>)</span><br />
      Find method, like that of a python string.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.find">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#rfind" class="summary-sig-name">rfind</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">sub</span>,
        <span class="summary-sig-arg">start</span>=<span class="summary-sig-default">0</span>,
        <span class="summary-sig-arg">end</span>=<span class="summary-sig-default">9223372036854775807</span>)</span><br />
      Find from right method, like that of a python string.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.rfind">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#startswith" class="summary-sig-name">startswith</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">prefix</span>,
        <span class="summary-sig-arg">start</span>=<span class="summary-sig-default">0</span>,
        <span class="summary-sig-arg">end</span>=<span class="summary-sig-default">9223372036854775807</span>)</span><br />
      Does the Seq start with the given prefix?  Returns True/False.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.startswith">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#endswith" class="summary-sig-name">endswith</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">suffix</span>,
        <span class="summary-sig-arg">start</span>=<span class="summary-sig-default">0</span>,
        <span class="summary-sig-arg">end</span>=<span class="summary-sig-default">9223372036854775807</span>)</span><br />
      Does the Seq end with the given suffix?  Returns True/False.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.endswith">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#split" class="summary-sig-name">split</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">sep</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">maxsplit</span>=<span class="summary-sig-default">-1</span>)</span><br />
      Split method, like that of a python string.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.split">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#rsplit" class="summary-sig-name">rsplit</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">sep</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">maxsplit</span>=<span class="summary-sig-default">-1</span>)</span><br />
      Right split method, like that of a python string.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.rsplit">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#strip" class="summary-sig-name">strip</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">chars</span>=<span class="summary-sig-default">None</span>)</span><br />
      Returns a new Seq object with leading and trailing ends stripped.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.strip">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#lstrip" class="summary-sig-name">lstrip</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">chars</span>=<span class="summary-sig-default">None</span>)</span><br />
      Returns a new Seq object with leading (left) end stripped.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.lstrip">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#rstrip" class="summary-sig-name">rstrip</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">chars</span>=<span class="summary-sig-default">None</span>)</span><br />
      Returns a new Seq object with trailing (right) end stripped.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.rstrip">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#upper" class="summary-sig-name">upper</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Returns an upper case copy of the sequence.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.upper">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#lower" class="summary-sig-name">lower</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Returns a lower case copy of the sequence.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.lower">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#complement" class="summary-sig-name">complement</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Returns the complement sequence.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.complement">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#reverse_complement" class="summary-sig-name">reverse_complement</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Returns the reverse complement sequence.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.reverse_complement">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#transcribe" class="summary-sig-name">transcribe</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Returns the RNA sequence from a DNA sequence.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.transcribe">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#back_transcribe" class="summary-sig-name">back_transcribe</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Returns the DNA sequence from an RNA sequence.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.back_transcribe">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#translate" class="summary-sig-name">translate</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">table</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">Standard</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">stop_symbol</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">*</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">to_stop</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">cds</span>=<span class="summary-sig-default">False</span>)</span><br />
      Turns a nucleotide sequence into a protein sequence.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.translate">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq.Seq-class.html#ungap" class="summary-sig-name">ungap</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">gap</span>=<span class="summary-sig-default">None</span>)</span><br />
      Return a copy of the sequence without the gap character(s).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.ungap">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__delattr__</code>,
      <code>__format__</code>,
      <code>__getattribute__</code>,
      <code>__new__</code>,
      <code>__reduce__</code>,
      <code>__reduce_ex__</code>,
      <code>__setattr__</code>,
      <code>__sizeof__</code>,
      <code>__subclasshook__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== PROPERTIES ==================== -->
<a name="section-Properties"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Properties</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Properties"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__class__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== METHOD DETAILS ==================== -->
<a name="section-MethodDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Method Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-MethodDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="__init__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__init__</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">data</span>,
        <span class="sig-arg">alphabet</span>=<span class="sig-default">Alphabet()</span>)</span>
    <br /><em class="fname">(Constructor)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.__init__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Create a Seq object.</p>
<dl class="rst-rst-docutils rst-docutils">
<dt>Arguments:</dt>
<dd><ul class="rst-rst-first rst-rst-last rst-rst-simple rst-first rst-last rst-simple">
<li><p class="rst-first rst-first">seq - Sequence, required (string)</p>
</li>
<li><p class="rst-first rst-first">alphabet - Optional argument, an Alphabet object from Bio.Alphabet</p>
</li>
</ul>
</dd>
</dl>
<p>You will typically use Bio.SeqIO to read in sequences from files as
SeqRecord objects, whose sequence will be exposed as a Seq object via
the seq property.</p>
<p>However, will often want to create your own Seq objects directly:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> IUPAC
<span class="py-prompt">&gt;&gt;&gt; </span>my_seq = Seq(<span class="py-string">&quot;MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF&quot;</span>,
<span class="py-more">... </span>             IUPAC.protein)
<span class="py-prompt">&gt;&gt;&gt; </span>my_seq
<span class="py-output">Seq('MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF', IUPACProtein())</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(my_seq)
<span class="py-output">MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>my_seq.alphabet
<span class="py-output">IUPACProtein()</span></pre>
  <dl class="fields">
    <dt>Overrides:
        object.__init__
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="__repr__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__repr__</span>(<span class="sig-arg">self</span>)</span>
    <br /><em class="fname">(Representation operator)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.__repr__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  Returns a (truncated) representation of the sequence for debugging.
  <dl class="fields">
    <dt>Overrides:
        object.__repr__
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="__str__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__str__</span>(<span class="sig-arg">self</span>)</span>
    <br /><em class="fname">(Informal representation operator)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.__str__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the full sequence as a python string, use str(my_seq).</p>
<p>Note that Biopython 1.44 and earlier would give a truncated
version of repr(my_seq) for str(my_seq).  If you are writing code
which need to be backwards compatible with old Biopython, you
should continue to use my_seq.tostring() rather than str(my_seq).</p>
  <dl class="fields">
    <dt>Overrides:
        object.__str__
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="__hash__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__hash__</span>(<span class="sig-arg">self</span>)</span>
    <br /><em class="fname">(Hashing function)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.__hash__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Hash for comparison.</p>
<p>See the __cmp__ documentation - this has changed from past
versions of Biopython!</p>
  <dl class="fields">
    <dt>Overrides:
        object.__hash__
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="__eq__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__eq__</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">other</span>)</span>
    <br /><em class="fname">(Equality operator)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.__eq__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Compare the sequence to another sequence or a string (README).</p>
<p>Historically comparing Seq objects has done Python object comparison.
After considerable discussion (keeping in mind constraints of the
Python language, hashes and dictionary support), Biopython now uses
simple string comparison (with a warning about the change).</p>
<p>Note that incompatible alphabets (e.g. DNA to RNA) will trigger a
warning.</p>
<p>During this transition period, please just do explicit comparisons:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> generic_dna
<span class="py-prompt">&gt;&gt;&gt; </span>seq1 = Seq(<span class="py-string">&quot;ACGT&quot;</span>)
<span class="py-prompt">&gt;&gt;&gt; </span>seq2 = Seq(<span class="py-string">&quot;ACGT&quot;</span>)
<span class="py-prompt">&gt;&gt;&gt; </span>id(seq1) == id(seq2)
<span class="py-output">False</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>str(seq1) == str(seq2)
<span class="py-output">True</span></pre>
<p>The new behaviour is to use string-like equality:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> generic_dna
<span class="py-prompt">&gt;&gt;&gt; </span>seq1 == seq2
<span class="py-output">True</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>seq1 == <span class="py-string">&quot;ACGT&quot;</span>
<span class="py-output">True</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>seq1 == Seq(<span class="py-string">&quot;ACGT&quot;</span>, generic_dna)
<span class="py-output">True</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="__add__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__add__</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">other</span>)</span>
    <br /><em class="fname">(Addition operator)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.__add__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Add another sequence or string to this sequence.</p>
<p>If adding a string to a Seq, the alphabet is preserved:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> generic_protein
<span class="py-prompt">&gt;&gt;&gt; </span>Seq(<span class="py-string">&quot;MELKI&quot;</span>, generic_protein) + <span class="py-string">&quot;LV&quot;</span>
<span class="py-output">Seq('MELKILV', ProteinAlphabet())</span></pre>
<p>When adding two Seq (like) objects, the alphabets are important.
Consider this example:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet.IUPAC <span class="py-keyword">import</span> unambiguous_dna, ambiguous_dna
<span class="py-prompt">&gt;&gt;&gt; </span>unamb_dna_seq = Seq(<span class="py-string">&quot;ACGT&quot;</span>, unambiguous_dna)
<span class="py-prompt">&gt;&gt;&gt; </span>ambig_dna_seq = Seq(<span class="py-string">&quot;ACRGT&quot;</span>, ambiguous_dna)
<span class="py-prompt">&gt;&gt;&gt; </span>unamb_dna_seq
<span class="py-output">Seq('ACGT', IUPACUnambiguousDNA())</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>ambig_dna_seq
<span class="py-output">Seq('ACRGT', IUPACAmbiguousDNA())</span></pre>
<p>If we add the ambiguous and unambiguous IUPAC DNA alphabets, we get
the more general ambiguous IUPAC DNA alphabet:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>unamb_dna_seq + ambig_dna_seq
<span class="py-output">Seq('ACGTACRGT', IUPACAmbiguousDNA())</span></pre>
<p>However, if the default generic alphabet is included, the result is
a generic alphabet:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>Seq(<span class="py-string">&quot;&quot;</span>) + ambig_dna_seq
<span class="py-output">Seq('ACRGT', Alphabet())</span></pre>
<p>You can't add RNA and DNA sequences:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> generic_dna, generic_rna
<span class="py-prompt">&gt;&gt;&gt; </span>Seq(<span class="py-string">&quot;ACGT&quot;</span>, generic_dna) + Seq(<span class="py-string">&quot;ACGU&quot;</span>, generic_rna)
<span class="py-except">Traceback (most recent call last):</span>
<span class="py-except">   ...</span>
<span class="py-except">TypeError: Incompatible alphabets DNAAlphabet() and RNAAlphabet()</span></pre>
<p>You can't add nucleotide and protein sequences:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> generic_dna, generic_protein
<span class="py-prompt">&gt;&gt;&gt; </span>Seq(<span class="py-string">&quot;ACGT&quot;</span>, generic_dna) + Seq(<span class="py-string">&quot;MELKI&quot;</span>, generic_protein)
<span class="py-except">Traceback (most recent call last):</span>
<span class="py-except">   ...</span>
<span class="py-except">TypeError: Incompatible alphabets DNAAlphabet() and ProteinAlphabet()</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="__radd__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__radd__</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">other</span>)</span>
    <br /><em class="fname">(Right-side addition operator)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.__radd__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Adding a sequence on the left.</p>
<p>If adding a string to a Seq, the alphabet is preserved:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> generic_protein
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-string">&quot;LV&quot;</span> + Seq(<span class="py-string">&quot;MELKI&quot;</span>, generic_protein)
<span class="py-output">Seq('LVMELKI', ProteinAlphabet())</span></pre>
<p>Adding two Seq (like) objects is handled via the __add__ method.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="tostring"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">tostring</span>(<span class="sig-arg">self</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.tostring">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the full sequence as a python string (DEPRECATED).</p>
<p>You are now encouraged to use str(my_seq) instead of
my_seq.tostring().</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="tomutable"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">tomutable</span>(<span class="sig-arg">self</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.tomutable">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the full sequence as a MutableSeq object.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> IUPAC
<span class="py-prompt">&gt;&gt;&gt; </span>my_seq = Seq(<span class="py-string">&quot;MKQHKAMIVALIVICITAVVAAL&quot;</span>,
<span class="py-more">... </span>             IUPAC.protein)
<span class="py-prompt">&gt;&gt;&gt; </span>my_seq
<span class="py-output">Seq('MKQHKAMIVALIVICITAVVAAL', IUPACProtein())</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>my_seq.tomutable()
<span class="py-output">MutableSeq('MKQHKAMIVALIVICITAVVAAL', IUPACProtein())</span></pre>
<p>Note that the alphabet is preserved.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_get_seq_str_and_check_alphabet"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_get_seq_str_and_check_alphabet</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">other_sequence</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq._get_seq_str_and_check_alphabet">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>string/Seq/MutableSeq to string, checking alphabet (PRIVATE).</p>
<p>For a string argument, returns the string.</p>
<p>For a Seq or MutableSeq, it checks the alphabet is compatible
(raising an exception if it isn't), and then returns a string.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="count"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">count</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">sub</span>,
        <span class="sig-arg">start</span>=<span class="sig-default">0</span>,
        <span class="sig-arg">end</span>=<span class="sig-default">9223372036854775807</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.count">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Non-overlapping count method, like that of a python string.</p>
<p>This behaves like the python string method of the same name,
which does a non-overlapping count!</p>
<p>Returns an integer, the number of occurrences of substring
argument sub in the (sub)sequence given by [start:end].
Optional arguments start and end are interpreted as in slice
notation.</p>
<dl class="rst-docutils">
<dt>Arguments:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>sub - a string or another Seq object to look for</li>
<li>start - optional integer, slice start</li>
<li>end - optional integer, slice end</li>
</ul>
</dd>
</dl>
<p>e.g.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span>my_seq = Seq(<span class="py-string">&quot;AAAATGA&quot;</span>)
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(my_seq.count(<span class="py-string">&quot;A&quot;</span>))
<span class="py-output">5</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(my_seq.count(<span class="py-string">&quot;ATG&quot;</span>))
<span class="py-output">1</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(my_seq.count(Seq(<span class="py-string">&quot;AT&quot;</span>)))
<span class="py-output">1</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(my_seq.count(<span class="py-string">&quot;AT&quot;</span>, 2, -1))
<span class="py-output">1</span></pre>
<p>HOWEVER, please note because python strings and Seq objects (and
MutableSeq objects) do a non-overlapping search, this may not give
the answer you expect:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-string">&quot;AAAA&quot;</span>.count(<span class="py-string">&quot;AA&quot;</span>)
<span class="py-output">2</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(Seq(<span class="py-string">&quot;AAAA&quot;</span>).count(<span class="py-string">&quot;AA&quot;</span>))
<span class="py-output">2</span></pre>
<p>An overlapping search would give the answer as three!</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="__contains__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__contains__</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">char</span>)</span>
    <br /><em class="fname">(In operator)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.__contains__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Implements the 'in' keyword, like a python string.</p>
<p>e.g.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> generic_dna, generic_rna, generic_protein
<span class="py-prompt">&gt;&gt;&gt; </span>my_dna = Seq(<span class="py-string">&quot;ATATGAAATTTGAAAA&quot;</span>, generic_dna)
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-string">&quot;AAA&quot;</span> <span class="py-keyword">in</span> my_dna
<span class="py-output">True</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>Seq(<span class="py-string">&quot;AAA&quot;</span>) <span class="py-keyword">in</span> my_dna
<span class="py-output">True</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>Seq(<span class="py-string">&quot;AAA&quot;</span>, generic_dna) <span class="py-keyword">in</span> my_dna
<span class="py-output">True</span></pre>
<p>Like other Seq methods, this will raise a type error if another Seq
(or Seq like) object with an incompatible alphabet is used:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>Seq(<span class="py-string">&quot;AAA&quot;</span>, generic_rna) <span class="py-keyword">in</span> my_dna
<span class="py-except">Traceback (most recent call last):</span>
<span class="py-except">   ...</span>
<span class="py-except">TypeError: Incompatible alphabets DNAAlphabet() and RNAAlphabet()</span>
<span class="py-except"></span><span class="py-prompt">&gt;&gt;&gt; </span>Seq(<span class="py-string">&quot;AAA&quot;</span>, generic_protein) <span class="py-keyword">in</span> my_dna
<span class="py-except">Traceback (most recent call last):</span>
<span class="py-except">   ...</span>
<span class="py-except">TypeError: Incompatible alphabets DNAAlphabet() and ProteinAlphabet()</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="find"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">find</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">sub</span>,
        <span class="sig-arg">start</span>=<span class="sig-default">0</span>,
        <span class="sig-arg">end</span>=<span class="sig-default">9223372036854775807</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.find">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Find method, like that of a python string.</p>
<p>This behaves like the python string method of the same name.</p>
<p>Returns an integer, the index of the first occurrence of substring
argument sub in the (sub)sequence given by [start:end].</p>
<dl class="rst-docutils">
<dt>Arguments:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>sub - a string or another Seq object to look for</li>
<li>start - optional integer, slice start</li>
<li>end - optional integer, slice end</li>
</ul>
</dd>
</dl>
<p>Returns -1 if the subsequence is NOT found.</p>
<p>e.g. Locating the first typical start codon, AUG, in an RNA sequence:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span>my_rna = Seq(<span class="py-string">&quot;GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG&quot;</span>)
<span class="py-prompt">&gt;&gt;&gt; </span>my_rna.find(<span class="py-string">&quot;AUG&quot;</span>)
<span class="py-output">3</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="rfind"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">rfind</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">sub</span>,
        <span class="sig-arg">start</span>=<span class="sig-default">0</span>,
        <span class="sig-arg">end</span>=<span class="sig-default">9223372036854775807</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.rfind">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Find from right method, like that of a python string.</p>
<p>This behaves like the python string method of the same name.</p>
<p>Returns an integer, the index of the last (right most) occurrence of
substring argument sub in the (sub)sequence given by [start:end].</p>
<dl class="rst-docutils">
<dt>Arguments:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>sub - a string or another Seq object to look for</li>
<li>start - optional integer, slice start</li>
<li>end - optional integer, slice end</li>
</ul>
</dd>
</dl>
<p>Returns -1 if the subsequence is NOT found.</p>
<p>e.g. Locating the last typical start codon, AUG, in an RNA sequence:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span>my_rna = Seq(<span class="py-string">&quot;GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG&quot;</span>)
<span class="py-prompt">&gt;&gt;&gt; </span>my_rna.rfind(<span class="py-string">&quot;AUG&quot;</span>)
<span class="py-output">15</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="startswith"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">startswith</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">prefix</span>,
        <span class="sig-arg">start</span>=<span class="sig-default">0</span>,
        <span class="sig-arg">end</span>=<span class="sig-default">9223372036854775807</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.startswith">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Does the Seq start with the given prefix?  Returns True/False.</p>
<p>This behaves like the python string method of the same name.</p>
<p>Return True if the sequence starts with the specified prefix
(a string or another Seq object), False otherwise.
With optional start, test sequence beginning at that position.
With optional end, stop comparing sequence at that position.
prefix can also be a tuple of strings to try.  e.g.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span>my_rna = Seq(<span class="py-string">&quot;GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG&quot;</span>)
<span class="py-prompt">&gt;&gt;&gt; </span>my_rna.startswith(<span class="py-string">&quot;GUC&quot;</span>)
<span class="py-output">True</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>my_rna.startswith(<span class="py-string">&quot;AUG&quot;</span>)
<span class="py-output">False</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>my_rna.startswith(<span class="py-string">&quot;AUG&quot;</span>, 3)
<span class="py-output">True</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>my_rna.startswith((<span class="py-string">&quot;UCC&quot;</span>, <span class="py-string">&quot;UCA&quot;</span>, <span class="py-string">&quot;UCG&quot;</span>), 1)
<span class="py-output">True</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="endswith"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">endswith</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">suffix</span>,
        <span class="sig-arg">start</span>=<span class="sig-default">0</span>,
        <span class="sig-arg">end</span>=<span class="sig-default">9223372036854775807</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.endswith">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Does the Seq end with the given suffix?  Returns True/False.</p>
<p>This behaves like the python string method of the same name.</p>
<p>Return True if the sequence ends with the specified suffix
(a string or another Seq object), False otherwise.
With optional start, test sequence beginning at that position.
With optional end, stop comparing sequence at that position.
suffix can also be a tuple of strings to try.  e.g.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span>my_rna = Seq(<span class="py-string">&quot;GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG&quot;</span>)
<span class="py-prompt">&gt;&gt;&gt; </span>my_rna.endswith(<span class="py-string">&quot;UUG&quot;</span>)
<span class="py-output">True</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>my_rna.endswith(<span class="py-string">&quot;AUG&quot;</span>)
<span class="py-output">False</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>my_rna.endswith(<span class="py-string">&quot;AUG&quot;</span>, 0, 18)
<span class="py-output">True</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>my_rna.endswith((<span class="py-string">&quot;UCC&quot;</span>, <span class="py-string">&quot;UCA&quot;</span>, <span class="py-string">&quot;UUG&quot;</span>))
<span class="py-output">True</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="split"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">split</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">sep</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">maxsplit</span>=<span class="sig-default">-1</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.split">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Split method, like that of a python string.</p>
<p>This behaves like the python string method of the same name.</p>
<p>Return a list of the 'words' in the string (as Seq objects),
using sep as the delimiter string.  If maxsplit is given, at
most maxsplit splits are done.  If maxsplit is omitted, all
splits are made.</p>
<p>Following the python string method, sep will by default be any
white space (tabs, spaces, newlines) but this is unlikely to
apply to biological sequences.</p>
<p>e.g.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span>my_rna = Seq(<span class="py-string">&quot;GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG&quot;</span>)
<span class="py-prompt">&gt;&gt;&gt; </span>my_aa = my_rna.translate()
<span class="py-prompt">&gt;&gt;&gt; </span>my_aa
<span class="py-output">Seq('VMAIVMGR*KGAR*L', HasStopCodon(ExtendedIUPACProtein(), '*'))</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>my_aa.split(<span class="py-string">&quot;*&quot;</span>)
<span class="py-output">[Seq('VMAIVMGR', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('KGAR', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('L', HasStopCodon(ExtendedIUPACProtein(), '*'))]</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>my_aa.split(<span class="py-string">&quot;*&quot;</span>, 1)
<span class="py-output">[Seq('VMAIVMGR', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('KGAR*L', HasStopCodon(ExtendedIUPACProtein(), '*'))]</span></pre>
<p>See also the rsplit method:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>my_aa.rsplit(<span class="py-string">&quot;*&quot;</span>, 1)
<span class="py-output">[Seq('VMAIVMGR*KGAR', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('L', HasStopCodon(ExtendedIUPACProtein(), '*'))]</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="rsplit"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">rsplit</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">sep</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">maxsplit</span>=<span class="sig-default">-1</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.rsplit">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Right split method, like that of a python string.</p>
<p>This behaves like the python string method of the same name.</p>
<p>Return a list of the 'words' in the string (as Seq objects),
using sep as the delimiter string.  If maxsplit is given, at
most maxsplit splits are done COUNTING FROM THE RIGHT.
If maxsplit is omitted, all splits are made.</p>
<p>Following the python string method, sep will by default be any
white space (tabs, spaces, newlines) but this is unlikely to
apply to biological sequences.</p>
<p>e.g. print(my_seq.rsplit(&quot;*&quot;,1))</p>
<p>See also the split method.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="strip"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">strip</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">chars</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.strip">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns a new Seq object with leading and trailing ends stripped.</p>
<p>This behaves like the python string method of the same name.</p>
<p>Optional argument chars defines which characters to remove.  If
omitted or None (default) then as for the python string method,
this defaults to removing any white space.</p>
<p>e.g. print(my_seq.strip(&quot;-&quot;))</p>
<p>See also the lstrip and rstrip methods.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="lstrip"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">lstrip</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">chars</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.lstrip">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns a new Seq object with leading (left) end stripped.</p>
<p>This behaves like the python string method of the same name.</p>
<p>Optional argument chars defines which characters to remove.  If
omitted or None (default) then as for the python string method,
this defaults to removing any white space.</p>
<p>e.g. print(my_seq.lstrip(&quot;-&quot;))</p>
<p>See also the strip and rstrip methods.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="rstrip"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">rstrip</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">chars</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.rstrip">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns a new Seq object with trailing (right) end stripped.</p>
<p>This behaves like the python string method of the same name.</p>
<p>Optional argument chars defines which characters to remove.  If
omitted or None (default) then as for the python string method,
this defaults to removing any white space.</p>
<p>e.g. Removing a nucleotide sequence's polyadenylation (poly-A tail):</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> IUPAC
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span>my_seq = Seq(<span class="py-string">&quot;CGGTACGCTTATGTCACGTAGAAAAAA&quot;</span>, IUPAC.unambiguous_dna)
<span class="py-prompt">&gt;&gt;&gt; </span>my_seq
<span class="py-output">Seq('CGGTACGCTTATGTCACGTAGAAAAAA', IUPACUnambiguousDNA())</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>my_seq.rstrip(<span class="py-string">&quot;A&quot;</span>)
<span class="py-output">Seq('CGGTACGCTTATGTCACGTAG', IUPACUnambiguousDNA())</span></pre>
<p>See also the strip and lstrip methods.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="upper"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">upper</span>(<span class="sig-arg">self</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.upper">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns an upper case copy of the sequence.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> HasStopCodon, generic_protein
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span>my_seq = Seq(<span class="py-string">&quot;VHLTPeeK*&quot;</span>, HasStopCodon(generic_protein))
<span class="py-prompt">&gt;&gt;&gt; </span>my_seq
<span class="py-output">Seq('VHLTPeeK*', HasStopCodon(ProteinAlphabet(), '*'))</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>my_seq.lower()
<span class="py-output">Seq('vhltpeek*', HasStopCodon(ProteinAlphabet(), '*'))</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>my_seq.upper()
<span class="py-output">Seq('VHLTPEEK*', HasStopCodon(ProteinAlphabet(), '*'))</span></pre>
<p>This will adjust the alphabet if required. See also the lower method.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="lower"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">lower</span>(<span class="sig-arg">self</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.lower">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns a lower case copy of the sequence.</p>
<p>This will adjust the alphabet if required. Note that the IUPAC alphabets
are upper case only, and thus a generic alphabet must be substituted.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> Gapped, generic_dna
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> IUPAC
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span>my_seq = Seq(<span class="py-string">&quot;CGGTACGCTTATGTCACGTAG*AAAAAA&quot;</span>, Gapped(IUPAC.unambiguous_dna, <span class="py-string">&quot;*&quot;</span>))
<span class="py-prompt">&gt;&gt;&gt; </span>my_seq
<span class="py-output">Seq('CGGTACGCTTATGTCACGTAG*AAAAAA', Gapped(IUPACUnambiguousDNA(), '*'))</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>my_seq.lower()
<span class="py-output">Seq('cggtacgcttatgtcacgtag*aaaaaa', Gapped(DNAAlphabet(), '*'))</span></pre>
<p>See also the upper method.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="complement"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">complement</span>(<span class="sig-arg">self</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.complement">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the complement sequence. New Seq object.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> IUPAC
<span class="py-prompt">&gt;&gt;&gt; </span>my_dna = Seq(<span class="py-string">&quot;CCCCCGATAG&quot;</span>, IUPAC.unambiguous_dna)
<span class="py-prompt">&gt;&gt;&gt; </span>my_dna
<span class="py-output">Seq('CCCCCGATAG', IUPACUnambiguousDNA())</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>my_dna.complement()
<span class="py-output">Seq('GGGGGCTATC', IUPACUnambiguousDNA())</span></pre>
<p>You can of course used mixed case sequences,</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> generic_dna
<span class="py-prompt">&gt;&gt;&gt; </span>my_dna = Seq(<span class="py-string">&quot;CCCCCgatA-GD&quot;</span>, generic_dna)
<span class="py-prompt">&gt;&gt;&gt; </span>my_dna
<span class="py-output">Seq('CCCCCgatA-GD', DNAAlphabet())</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>my_dna.complement()
<span class="py-output">Seq('GGGGGctaT-CH', DNAAlphabet())</span></pre>
<p>Note in the above example, ambiguous character D denotes
G, A or T so its complement is H (for C, T or A).</p>
<p>Trying to complement a protein sequence raises an exception.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>my_protein = Seq(<span class="py-string">&quot;MAIVMGR&quot;</span>, IUPAC.protein)
<span class="py-prompt">&gt;&gt;&gt; </span>my_protein.complement()
<span class="py-except">Traceback (most recent call last):</span>
<span class="py-except">   ...</span>
<span class="py-except">ValueError: Proteins do not have complements!</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="reverse_complement"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">reverse_complement</span>(<span class="sig-arg">self</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.reverse_complement">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the reverse complement sequence. New Seq object.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> IUPAC
<span class="py-prompt">&gt;&gt;&gt; </span>my_dna = Seq(<span class="py-string">&quot;CCCCCGATAGNR&quot;</span>, IUPAC.ambiguous_dna)
<span class="py-prompt">&gt;&gt;&gt; </span>my_dna
<span class="py-output">Seq('CCCCCGATAGNR', IUPACAmbiguousDNA())</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>my_dna.reverse_complement()
<span class="py-output">Seq('YNCTATCGGGGG', IUPACAmbiguousDNA())</span></pre>
<p>Note in the above example, since R = G or A, its complement
is Y (which denotes C or T).</p>
<p>You can of course used mixed case sequences,</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> generic_dna
<span class="py-prompt">&gt;&gt;&gt; </span>my_dna = Seq(<span class="py-string">&quot;CCCCCgatA-G&quot;</span>, generic_dna)
<span class="py-prompt">&gt;&gt;&gt; </span>my_dna
<span class="py-output">Seq('CCCCCgatA-G', DNAAlphabet())</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>my_dna.reverse_complement()
<span class="py-output">Seq('C-TatcGGGGG', DNAAlphabet())</span></pre>
<p>Trying to complement a protein sequence raises an exception:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>my_protein = Seq(<span class="py-string">&quot;MAIVMGR&quot;</span>, IUPAC.protein)
<span class="py-prompt">&gt;&gt;&gt; </span>my_protein.reverse_complement()
<span class="py-except">Traceback (most recent call last):</span>
<span class="py-except">   ...</span>
<span class="py-except">ValueError: Proteins do not have complements!</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="transcribe"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">transcribe</span>(<span class="sig-arg">self</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.transcribe">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the RNA sequence from a DNA sequence. New Seq object.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> IUPAC
<span class="py-prompt">&gt;&gt;&gt; </span>coding_dna = Seq(<span class="py-string">&quot;ATGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG&quot;</span>,
<span class="py-more">... </span>                 IUPAC.unambiguous_dna)
<span class="py-prompt">&gt;&gt;&gt; </span>coding_dna
<span class="py-output">Seq('ATGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG', IUPACUnambiguousDNA())</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>coding_dna.transcribe()
<span class="py-output">Seq('AUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAG', IUPACUnambiguousRNA())</span></pre>
<p>Trying to transcribe a protein or RNA sequence raises an exception:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>my_protein = Seq(<span class="py-string">&quot;MAIVMGR&quot;</span>, IUPAC.protein)
<span class="py-prompt">&gt;&gt;&gt; </span>my_protein.transcribe()
<span class="py-except">Traceback (most recent call last):</span>
<span class="py-except">   ...</span>
<span class="py-except">ValueError: Proteins cannot be transcribed!</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="back_transcribe"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">back_transcribe</span>(<span class="sig-arg">self</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.back_transcribe">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the DNA sequence from an RNA sequence. New Seq object.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> IUPAC
<span class="py-prompt">&gt;&gt;&gt; </span>messenger_rna = Seq(<span class="py-string">&quot;AUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAG&quot;</span>,
<span class="py-more">... </span>                    IUPAC.unambiguous_rna)
<span class="py-prompt">&gt;&gt;&gt; </span>messenger_rna
<span class="py-output">Seq('AUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAG', IUPACUnambiguousRNA())</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>messenger_rna.back_transcribe()
<span class="py-output">Seq('ATGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG', IUPACUnambiguousDNA())</span></pre>
<p>Trying to back-transcribe a protein or DNA sequence raises an
exception:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>my_protein = Seq(<span class="py-string">&quot;MAIVMGR&quot;</span>, IUPAC.protein)
<span class="py-prompt">&gt;&gt;&gt; </span>my_protein.back_transcribe()
<span class="py-except">Traceback (most recent call last):</span>
<span class="py-except">   ...</span>
<span class="py-except">ValueError: Proteins cannot be back transcribed!</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="translate"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">translate</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">table</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">Standard</code><code class="variable-quote">'</code></span>,
        <span class="sig-arg">stop_symbol</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">*</code><code class="variable-quote">'</code></span>,
        <span class="sig-arg">to_stop</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">cds</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.translate">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Turns a nucleotide sequence into a protein sequence. New Seq object.</p>
<p>This method will translate DNA or RNA sequences, and those with a
nucleotide or generic alphabet.  Trying to translate a protein
sequence raises an exception.</p>
<dl class="rst-docutils">
<dt>Arguments:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>table - Which codon table to use?  This can be either a name
(string), an NCBI identifier (integer), or a CodonTable
object (useful for non-standard genetic codes).  This
defaults to the &quot;Standard&quot; table.</li>
<li>stop_symbol - Single character string, what to use for terminators.
This defaults to the asterisk, &quot;*&quot;.</li>
<li>to_stop - Boolean, defaults to False meaning do a full translation
continuing on past any stop codons (translated as the
specified stop_symbol).  If True, translation is
terminated at the first in frame stop codon (and the
stop_symbol is not appended to the returned protein
sequence).</li>
<li>cds - Boolean, indicates this is a complete CDS.  If True,
this checks the sequence starts with a valid alternative start
codon (which will be translated as methionine, M), that the
sequence length is a multiple of three, and that there is a
single in frame stop codon at the end (this will be excluded
from the protein sequence, regardless of the to_stop option).
If these tests fail, an exception is raised.</li>
</ul>
</dd>
</dl>
<p>e.g. Using the standard table:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>coding_dna = Seq(<span class="py-string">&quot;GTGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG&quot;</span>)
<span class="py-prompt">&gt;&gt;&gt; </span>coding_dna.translate()
<span class="py-output">Seq('VAIVMGR*KGAR*', HasStopCodon(ExtendedIUPACProtein(), '*'))</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>coding_dna.translate(stop_symbol=<span class="py-string">&quot;@&quot;</span>)
<span class="py-output">Seq('VAIVMGR@KGAR@', HasStopCodon(ExtendedIUPACProtein(), '@'))</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>coding_dna.translate(to_stop=True)
<span class="py-output">Seq('VAIVMGR', ExtendedIUPACProtein())</span></pre>
<p>Now using NCBI table 2, where TGA is not a stop codon:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>coding_dna.translate(table=2)
<span class="py-output">Seq('VAIVMGRWKGAR*', HasStopCodon(ExtendedIUPACProtein(), '*'))</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>coding_dna.translate(table=2, to_stop=True)
<span class="py-output">Seq('VAIVMGRWKGAR', ExtendedIUPACProtein())</span></pre>
<p>In fact, GTG is an alternative start codon under NCBI table 2, meaning
this sequence could be a complete CDS:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>coding_dna.translate(table=2, cds=True)
<span class="py-output">Seq('MAIVMGRWKGAR', ExtendedIUPACProtein())</span></pre>
<p>It isn't a valid CDS under NCBI table 1, due to both the start codon and
also the in frame stop codons:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>coding_dna.translate(table=1, cds=True)
<span class="py-except">Traceback (most recent call last):</span>
<span class="py-except">    ...</span>
<span class="py-except">TranslationError: First codon 'GTG' is not a start codon</span></pre>
<p>If the sequence has no in-frame stop codon, then the to_stop argument
has no effect:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>coding_dna2 = Seq(<span class="py-string">&quot;TTGGCCATTGTAATGGGCCGC&quot;</span>)
<span class="py-prompt">&gt;&gt;&gt; </span>coding_dna2.translate()
<span class="py-output">Seq('LAIVMGR', ExtendedIUPACProtein())</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>coding_dna2.translate(to_stop=True)
<span class="py-output">Seq('LAIVMGR', ExtendedIUPACProtein())</span></pre>
<p>NOTE - Ambiguous codons like &quot;TAN&quot; or &quot;NNN&quot; could be an amino acid
or a stop codon.  These are translated as &quot;X&quot;.  Any invalid codon
(e.g. &quot;TA?&quot; or &quot;T-A&quot;) will throw a TranslationError.</p>
<p>NOTE - Does NOT support gapped sequences.</p>
<p>NOTE - This does NOT behave like the python string's translate
method.  For that use str(my_seq).translate(...) instead.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="ungap"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">ungap</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">gap</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#Seq.ungap">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Return a copy of the sequence without the gap character(s).</p>
<p>The gap character can be specified in two ways - either as an explicit
argument, or via the sequence's alphabet. For example:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> generic_dna
<span class="py-prompt">&gt;&gt;&gt; </span>my_dna = Seq(<span class="py-string">&quot;-ATA--TGAAAT-TTGAAAA&quot;</span>, generic_dna)
<span class="py-prompt">&gt;&gt;&gt; </span>my_dna
<span class="py-output">Seq('-ATA--TGAAAT-TTGAAAA', DNAAlphabet())</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>my_dna.ungap(<span class="py-string">&quot;-&quot;</span>)
<span class="py-output">Seq('ATATGAAATTTGAAAA', DNAAlphabet())</span></pre>
<p>If the gap character is not given as an argument, it will be taken from
the sequence's alphabet (if defined). Notice that the returned sequence's
alphabet is adjusted since it no longer requires a gapped alphabet:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> IUPAC, Gapped, HasStopCodon
<span class="py-prompt">&gt;&gt;&gt; </span>my_pro = Seq(<span class="py-string">&quot;MVVLE=AD*&quot;</span>, HasStopCodon(Gapped(IUPAC.protein, <span class="py-string">&quot;=&quot;</span>)))
<span class="py-prompt">&gt;&gt;&gt; </span>my_pro
<span class="py-output">Seq('MVVLE=AD*', HasStopCodon(Gapped(IUPACProtein(), '='), '*'))</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>my_pro.ungap()
<span class="py-output">Seq('MVVLEAD*', HasStopCodon(IUPACProtein(), '*'))</span></pre>
<p>Or, with a simpler gapped DNA example:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> IUPAC, Gapped
<span class="py-prompt">&gt;&gt;&gt; </span>my_seq = Seq(<span class="py-string">&quot;CGGGTAG=AAAAAA&quot;</span>, Gapped(IUPAC.unambiguous_dna, <span class="py-string">&quot;=&quot;</span>))
<span class="py-prompt">&gt;&gt;&gt; </span>my_seq
<span class="py-output">Seq('CGGGTAG=AAAAAA', Gapped(IUPACUnambiguousDNA(), '='))</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>my_seq.ungap()
<span class="py-output">Seq('CGGGTAGAAAAAA', IUPACUnambiguousDNA())</span></pre>
<p>As long as it is consistent with the alphabet, although it is redundant,
you can still supply the gap character as an argument to this method:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>my_seq
<span class="py-output">Seq('CGGGTAG=AAAAAA', Gapped(IUPACUnambiguousDNA(), '='))</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>my_seq.ungap(<span class="py-string">&quot;=&quot;</span>)
<span class="py-output">Seq('CGGGTAGAAAAAA', IUPACUnambiguousDNA())</span></pre>
<p>However, if the gap character given as the argument disagrees with that
declared in the alphabet, an exception is raised:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>my_seq
<span class="py-output">Seq('CGGGTAG=AAAAAA', Gapped(IUPACUnambiguousDNA(), '='))</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>my_seq.ungap(<span class="py-string">&quot;-&quot;</span>)
<span class="py-except">Traceback (most recent call last):</span>
<span class="py-except">   ...</span>
<span class="py-except">ValueError: Gap '-' does not match '=' from alphabet</span></pre>
<p>Finally, if a gap character is not supplied, and the alphabet does not
define one, an exception is raised:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> generic_dna
<span class="py-prompt">&gt;&gt;&gt; </span>my_dna = Seq(<span class="py-string">&quot;ATA--TGAAAT-TTGAAAA&quot;</span>, generic_dna)
<span class="py-prompt">&gt;&gt;&gt; </span>my_dna
<span class="py-output">Seq('ATA--TGAAAT-TTGAAAA', DNAAlphabet())</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>my_dna.ungap()
<span class="py-except">Traceback (most recent call last):</span>
<span class="py-except">   ...</span>
<span class="py-except">ValueError: Gap character not given and not defined in alphabet</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:04:18 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
