<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.codonalign</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        Package&nbsp;codonalign
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.codonalign-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.codonalign-module.html">Package Bio.codonalign</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2013 by Zheng Ruan (zruan1991@gmail.com).</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># All rights reserved.</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-docstring">"""Code for dealing with Codon Alignments."""</tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">__future__</tt> <tt class="py-keyword">import</tt> <tt class="py-name">print_function</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-1" class="py-name" targets="Class Bio.BiopythonWarning=Bio.BiopythonWarning-class.html"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-1', 'BiopythonWarning', 'link-1');">BiopythonWarning</a></tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-2" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-2', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-3" class="py-name" targets="Class Bio.BiopythonExperimentalWarning=Bio.BiopythonExperimentalWarning-class.html"><a title="Bio.BiopythonExperimentalWarning" class="py-name" href="#" onclick="return doclink('link-3', 'BiopythonExperimentalWarning', 'link-3');">BiopythonExperimentalWarning</a></tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-4" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-4', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-5" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-5', '_py3k', 'link-5');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">zip</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-6" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-6', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-7" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-7', 'Alphabet', 'link-7');">Alphabet</a></tt> <tt class="py-keyword">import</tt> <tt id="link-8" class="py-name" targets="Function Bio.Alphabet._get_base_alphabet()=Bio.Alphabet-module.html#_get_base_alphabet"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-8', '_get_base_alphabet', 'link-8');">_get_base_alphabet</a></tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-9" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-9', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-10" class="py-name" targets="Module Bio.SeqRecord=Bio.SeqRecord-module.html,Class Bio.SeqRecord.SeqRecord=Bio.SeqRecord.SeqRecord-class.html"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-10', 'SeqRecord', 'link-10');">SeqRecord</a></tt> <tt class="py-keyword">import</tt> <tt id="link-11" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-11', 'SeqRecord', 'link-10');">SeqRecord</a></tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-12" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-12', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-13" class="py-name" targets="Package Bio.codonalign=Bio.codonalign-module.html"><a title="Bio.codonalign" class="py-name" href="#" onclick="return doclink('link-13', 'codonalign', 'link-13');">codonalign</a></tt><tt class="py-op">.</tt><tt id="link-14" class="py-name" targets="Module Bio.codonalign.codonseq=Bio.codonalign.codonseq-module.html"><a title="Bio.codonalign.codonseq" class="py-name" href="#" onclick="return doclink('link-14', 'codonseq', 'link-14');">codonseq</a></tt> <tt class="py-keyword">import</tt> <tt id="link-15" class="py-name" targets="Class Bio.codonalign.codonseq.CodonSeq=Bio.codonalign.codonseq.CodonSeq-class.html"><a title="Bio.codonalign.codonseq.CodonSeq" class="py-name" href="#" onclick="return doclink('link-15', 'CodonSeq', 'link-15');">CodonSeq</a></tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-16" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-16', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-17" class="py-name"><a title="Bio.codonalign" class="py-name" href="#" onclick="return doclink('link-17', 'codonalign', 'link-13');">codonalign</a></tt><tt class="py-op">.</tt><tt id="link-18" class="py-name" targets="Module Bio.codonalign.codonalignment=Bio.codonalign.codonalignment-module.html"><a title="Bio.codonalign.codonalignment" class="py-name" href="#" onclick="return doclink('link-18', 'codonalignment', 'link-18');">codonalignment</a></tt> <tt class="py-keyword">import</tt> <tt id="link-19" class="py-name" targets="Class Bio.codonalign.codonalignment.CodonAlignment=Bio.codonalign.codonalignment.CodonAlignment-class.html"><a title="Bio.codonalign.codonalignment.CodonAlignment" class="py-name" href="#" onclick="return doclink('link-19', 'CodonAlignment', 'link-19');">CodonAlignment</a></tt><tt class="py-op">,</tt> <tt id="link-20" class="py-name" targets="Function Bio.codonalign.codonalignment.mktest()=Bio.codonalign.codonalignment-module.html#mktest"><a title="Bio.codonalign.codonalignment.mktest" class="py-name" href="#" onclick="return doclink('link-20', 'mktest', 'link-20');">mktest</a></tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-21" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-21', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-22" class="py-name"><a title="Bio.codonalign" class="py-name" href="#" onclick="return doclink('link-22', 'codonalign', 'link-13');">codonalign</a></tt><tt class="py-op">.</tt><tt id="link-23" class="py-name" targets="Module Bio.codonalign.codonalphabet=Bio.codonalign.codonalphabet-module.html"><a title="Bio.codonalign.codonalphabet" class="py-name" href="#" onclick="return doclink('link-23', 'codonalphabet', 'link-23');">codonalphabet</a></tt> <tt class="py-keyword">import</tt> <tt id="link-24" class="py-name" targets="Class Bio.codonalign.codonalphabet.CodonAlphabet=Bio.codonalign.codonalphabet.CodonAlphabet-class.html"><a title="Bio.codonalign.codonalphabet.CodonAlphabet" class="py-name" href="#" onclick="return doclink('link-24', 'CodonAlphabet', 'link-24');">CodonAlphabet</a></tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-25" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-25', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-26" class="py-name"><a title="Bio.codonalign" class="py-name" href="#" onclick="return doclink('link-26', 'codonalign', 'link-13');">codonalign</a></tt><tt class="py-op">.</tt><tt id="link-27" class="py-name"><a title="Bio.codonalign.codonalphabet" class="py-name" href="#" onclick="return doclink('link-27', 'codonalphabet', 'link-23');">codonalphabet</a></tt> <tt class="py-keyword">import</tt> <tt id="link-28" class="py-name" targets="Variable Bio.codonalign.codonalphabet.default_codon_table=Bio.codonalign.codonalphabet-module.html#default_codon_table"><a title="Bio.codonalign.codonalphabet.default_codon_table" class="py-name" href="#" onclick="return doclink('link-28', 'default_codon_table', 'link-28');">default_codon_table</a></tt><tt class="py-op">,</tt> <tt id="link-29" class="py-name" targets="Variable Bio.codonalign.codonalphabet.default_codon_alphabet=Bio.codonalign.codonalphabet-module.html#default_codon_alphabet"><a title="Bio.codonalign.codonalphabet.default_codon_alphabet" class="py-name" href="#" onclick="return doclink('link-29', 'default_codon_alphabet', 'link-29');">default_codon_alphabet</a></tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-30" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-30', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-31" class="py-name"><a title="Bio.codonalign" class="py-name" href="#" onclick="return doclink('link-31', 'codonalign', 'link-13');">codonalign</a></tt><tt class="py-op">.</tt><tt id="link-32" class="py-name"><a title="Bio.codonalign.codonalphabet" class="py-name" href="#" onclick="return doclink('link-32', 'codonalphabet', 'link-23');">codonalphabet</a></tt> <tt class="py-keyword">import</tt> <tt id="link-33" class="py-name" targets="Function Bio.codonalign.codonalphabet.get_codon_alphabet()=Bio.codonalign.codonalphabet-module.html#get_codon_alphabet"><a title="Bio.codonalign.codonalphabet.get_codon_alphabet" class="py-name" href="#" onclick="return doclink('link-33', 'get_codon_alphabet', 'link-33');">get_codon_alphabet</a></tt> <tt class="py-keyword">as</tt> <tt class="py-name">_get_codon_alphabet</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">'Bio.codonalign is an experimental module which may undergo '</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line">              <tt class="py-string">'significant changes prior to its future official release.'</tt><tt class="py-op">,</tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line">              <tt id="link-34" class="py-name"><a title="Bio.BiopythonExperimentalWarning" class="py-name" href="#" onclick="return doclink('link-34', 'BiopythonExperimentalWarning', 'link-3');">BiopythonExperimentalWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"> </tt>
<a name="build"></a><div id="build-def"><a name="L30"></a><tt class="py-lineno"> 30</tt> <a class="py-toggle" href="#" id="build-toggle" onclick="return toggle('build');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.codonalign-module.html#build">build</a><tt class="py-op">(</tt><tt class="py-param">pro_align</tt><tt class="py-op">,</tt> <tt class="py-param">nucl_seqs</tt><tt class="py-op">,</tt> <tt class="py-param">corr_dict</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">gap_char</tt><tt class="py-op">=</tt><tt class="py-string">'-'</tt><tt class="py-op">,</tt> <tt class="py-param">unknown</tt><tt class="py-op">=</tt><tt class="py-string">'X'</tt><tt class="py-op">,</tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line">          <tt class="py-param">codon_table</tt><tt class="py-op">=</tt><tt id="link-35" class="py-name"><a title="Bio.codonalign.codonalphabet.default_codon_table" class="py-name" href="#" onclick="return doclink('link-35', 'default_codon_table', 'link-28');">default_codon_table</a></tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line">          <tt class="py-param">complete_protein</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> <tt class="py-param">anchor_len</tt><tt class="py-op">=</tt><tt class="py-number">10</tt><tt class="py-op">,</tt> <tt class="py-param">max_score</tt><tt class="py-op">=</tt><tt class="py-number">10</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="build-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="build-expanded"><a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line">    <tt class="py-docstring">"""Build a codon alignment from protein alignment and corresponding nucleotides.</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-docstring">    Arguments:</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"><tt class="py-docstring">     - pro_align  - a protein MultipleSeqAlignment object</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"><tt class="py-docstring">     - nucl_seqs - an object returned by SeqIO.parse or SeqIO.index</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"><tt class="py-docstring">       or a collection of SeqRecord.</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"><tt class="py-docstring">     - alphabet   - alphabet for the returned codon alignment</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"><tt class="py-docstring">     - corr_dict  - a dict that maps protein id to nucleotide id</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"><tt class="py-docstring">     - complete_protein - whether the sequence begins with a start</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"><tt class="py-docstring">       codon</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line"><tt class="py-docstring">    Return a CodonAlignment object.</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line"><tt class="py-docstring">    The example below answers this Biostars question: https://www.biostars.org/p/89741/</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio.Alphabet import generic_dna, generic_protein</tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio.SeqRecord import SeqRecord</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio.Align import MultipleSeqAlignment</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio.codonalign import build</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; seq1 = SeqRecord(Seq('ATGTCTCGT', alphabet=generic_dna), id='pro1')</tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; seq2 = SeqRecord(Seq('ATGCGT', alphabet=generic_dna), id='pro2')</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; pro1 = SeqRecord(Seq('MSR', alphabet=generic_protein), id='pro1')</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; pro2 = SeqRecord(Seq('M-R', alphabet=generic_protein), id='pro2')</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; aln = MultipleSeqAlignment([pro1, pro2])</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; codon_aln = build(aln, [seq1, seq2])</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(codon_aln)</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line"><tt class="py-docstring">    CodonAlphabet(Standard) CodonAlignment with 2 rows and 9 columns (3 codons)</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line"><tt class="py-docstring">    ATGTCTCGT pro1</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line"><tt class="py-docstring">    ATG---CGT pro2</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line">    <tt class="py-comment"># TODO</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line">    <tt class="py-comment"># add an option to allow the user to specify the returned object?</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line"> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-36" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-36', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-37" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-37', 'Alphabet', 'link-7');">Alphabet</a></tt> <tt class="py-keyword">import</tt> <tt id="link-38" class="py-name" targets="Class Bio.Alphabet.ProteinAlphabet=Bio.Alphabet.ProteinAlphabet-class.html"><a title="Bio.Alphabet.ProteinAlphabet" class="py-name" href="#" onclick="return doclink('link-38', 'ProteinAlphabet', 'link-38');">ProteinAlphabet</a></tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-39" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-39', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-40" class="py-name" targets="Package Bio.Align=Bio.Align-module.html"><a title="Bio.Align" class="py-name" href="#" onclick="return doclink('link-40', 'Align', 'link-40');">Align</a></tt> <tt class="py-keyword">import</tt> <tt id="link-41" class="py-name" targets="Class Bio.Align.MultipleSeqAlignment=Bio.Align.MultipleSeqAlignment-class.html"><a title="Bio.Align.MultipleSeqAlignment" class="py-name" href="#" onclick="return doclink('link-41', 'MultipleSeqAlignment', 'link-41');">MultipleSeqAlignment</a></tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line"> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line">    <tt class="py-comment"># check the type of object of pro_align</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">pro_align</tt><tt class="py-op">,</tt> <tt id="link-42" class="py-name"><a title="Bio.Align.MultipleSeqAlignment" class="py-name" href="#" onclick="return doclink('link-42', 'MultipleSeqAlignment', 'link-41');">MultipleSeqAlignment</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"the first argument should be a MultipleSeqAlignment "</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">                        <tt class="py-string">"object"</tt><tt class="py-op">)</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line">    <tt class="py-comment"># check the alphabet of pro_align</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">pro</tt> <tt class="py-keyword">in</tt> <tt class="py-name">pro_align</tt><tt class="py-op">:</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-43" class="py-name"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-43', '_get_base_alphabet', 'link-8');">_get_base_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">pro</tt><tt class="py-op">.</tt><tt id="link-44" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.seq=Bio.SeqRecord.SeqRecord-class.html#seq,Variable BioSQL.BioSeq.DBSeqRecord.seq=BioSQL.BioSeq.DBSeqRecord-class.html#seq"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-44', 'seq', 'link-44');">seq</a></tt><tt class="py-op">.</tt><tt id="link-45" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.alphabet=Bio.SearchIO._model.hsp.HSP-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSPFragment.alphabet=Bio.SearchIO._model.hsp.HSPFragment-class.html#alphabet"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-45', 'alphabet', 'link-45');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-46" class="py-name"><a title="Bio.Alphabet.ProteinAlphabet" class="py-name" href="#" onclick="return doclink('link-46', 'ProteinAlphabet', 'link-38');">ProteinAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"Alphabet Error!\nThe input alignment should be "</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line">                            <tt class="py-string">"a *PROTEIN* alignemnt, found %r"</tt> <tt class="py-op">%</tt> <tt class="py-name">pro</tt><tt class="py-op">.</tt><tt id="link-47" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-47', 'seq', 'link-44');">seq</a></tt><tt class="py-op">.</tt><tt id="link-48" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-48', 'alphabet', 'link-45');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt id="link-49" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-49', 'alphabet', 'link-45');">alphabet</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">        <tt id="link-50" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-50', 'alphabet', 'link-45');">alphabet</a></tt> <tt class="py-op">=</tt> <tt class="py-name">_get_codon_alphabet</tt><tt class="py-op">(</tt><tt class="py-name">codon_table</tt><tt class="py-op">,</tt> <tt class="py-name">gap_char</tt><tt class="py-op">=</tt><tt class="py-name">gap_char</tt><tt class="py-op">)</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line">    <tt class="py-comment"># check whether the number of seqs in pro_align and nucl_seqs is</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line">    <tt class="py-comment"># the same</tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line">    <tt class="py-name">pro_num</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">pro_align</tt><tt class="py-op">)</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">corr_dict</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">nucl_seqs</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt> <tt class="py-op">==</tt> <tt class="py-string">"generator"</tt><tt class="py-op">:</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line">            <tt class="py-comment"># nucl_seqs will be a tuple if read by SeqIO.parse()</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line">            <tt class="py-name">nucl_seqs</tt> <tt class="py-op">=</tt> <tt class="py-name">tuple</tt><tt class="py-op">(</tt><tt class="py-name">nucl_seqs</tt><tt class="py-op">)</tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line">        <tt class="py-name">nucl_num</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">nucl_seqs</tt><tt class="py-op">)</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">pro_num</tt> <tt class="py-op">&gt;</tt> <tt class="py-name">nucl_num</tt><tt class="py-op">:</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Higher Number of SeqRecords in Protein Alignment "</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line">                             <tt class="py-string">"({0}) than the Number of Nucleotide SeqRecords "</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">                             <tt class="py-string">"({1}) are found!"</tt><tt class="py-op">.</tt><tt id="link-51" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.format()=Bio.Align.MultipleSeqAlignment-class.html#format,Method Bio.Phylo.BaseTree.Tree.format()=Bio.Phylo.BaseTree.Tree-class.html#format,Method Bio.Restriction.Restriction.RestrictionBatch.format()=Bio.Restriction.Restriction.RestrictionBatch-class.html#format,Method Bio.SeqRecord.SeqRecord.format()=Bio.SeqRecord.SeqRecord-class.html#format,Method Bio.motifs.Motif.format()=Bio.motifs.Motif-class.html#format"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-51', 'format', 'link-51');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">pro_num</tt><tt class="py-op">,</tt> <tt class="py-name">nucl_num</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line"> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line">        <tt class="py-comment"># Determine the protein sequences and nucl sequences</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line">        <tt class="py-comment"># correspondence. If nucl_seqs is a list, tuple or read by</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line">        <tt class="py-comment"># SeqIO.parse(), we assume the order of sequences in pro_align</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line">        <tt class="py-comment"># and nucl_seqs are the same. If nucl_seqs is a dict or read by</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">        <tt class="py-comment"># SeqIO.index(), we match seqs in pro_align and those in</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line">        <tt class="py-comment"># nucl_seq by their id.</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">nucl_seqs</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">"_IndexedSeqFileDict"</tt><tt class="py-op">,</tt> <tt class="py-string">"dict"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line">            <tt class="py-name">corr_method</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">nucl_seqs</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">"list"</tt><tt class="py-op">,</tt> <tt class="py-string">"tuple"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line">            <tt class="py-name">corr_method</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"Nucl Sequences Error, Unknown type to assign "</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line">                            <tt class="py-string">"correspondence method"</tt><tt class="py-op">)</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">corr_dict</tt><tt class="py-op">,</tt> <tt class="py-name">dict</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"corr_dict should be a dict that corresponds "</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">                            <tt class="py-string">"protein id to nucleotide id!"</tt><tt class="py-op">)</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">corr_dict</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;=</tt> <tt class="py-name">pro_num</tt><tt class="py-op">:</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">            <tt class="py-comment"># read by SeqIO.parse()</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">nucl_seqs</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt> <tt class="py-op">==</tt> <tt class="py-string">"generator"</tt><tt class="py-op">:</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">                <tt class="py-keyword">from</tt> <tt id="link-52" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-52', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-53" class="py-name" targets="Package Bio.SeqIO=Bio.SeqIO-module.html"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-53', 'SeqIO', 'link-53');">SeqIO</a></tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">                <tt class="py-name">nucl_seqs</tt> <tt class="py-op">=</tt> <tt id="link-54" class="py-name"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-54', 'SeqIO', 'link-53');">SeqIO</a></tt><tt class="py-op">.</tt><tt id="link-55" class="py-name" targets="Function Bio.SearchIO.to_dict()=Bio.SearchIO-module.html#to_dict,Function Bio.SeqIO.to_dict()=Bio.SeqIO-module.html#to_dict,Method Bio.motifs.jaspar.Record.to_dict()=Bio.motifs.jaspar.Record-class.html#to_dict"><a title="Bio.SearchIO.to_dict
Bio.SeqIO.to_dict
Bio.motifs.jaspar.Record.to_dict" class="py-name" href="#" onclick="return doclink('link-55', 'to_dict', 'link-55');">to_dict</a></tt><tt class="py-op">(</tt><tt class="py-name">nucl_seqs</tt><tt class="py-op">)</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">nucl_seqs</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">"list"</tt><tt class="py-op">,</tt> <tt class="py-string">"tuple"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line">                <tt class="py-name">nucl_seqs</tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt id="link-56" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-56', 'i', 'link-56');">i</a></tt><tt class="py-op">.</tt><tt id="link-57" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Variable Bio.PDB.Entity.Entity.id=Bio.PDB.Entity.Entity-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-57', 'id', 'link-57');">id</a></tt><tt class="py-op">,</tt> <tt id="link-58" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-58', 'i', 'link-56');">i</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-59" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-59', 'i', 'link-56');">i</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">nucl_seqs</tt><tt class="py-op">)</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line">                <tt class="py-comment"># nucl_seqs = {i.id: i for i in nucl_seqs}</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">nucl_seqs</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt> <tt class="py-keyword">in</tt> \ </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">                    <tt class="py-op">(</tt><tt class="py-string">"_IndexedSeqFileDict"</tt><tt class="py-op">,</tt> <tt class="py-string">"dict"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">                <tt class="py-keyword">pass</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"Nucl Sequences Error, Unknown type of "</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line">                                <tt class="py-string">"Nucleotide Records!"</tt><tt class="py-op">)</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line">            <tt class="py-name">corr_method</tt> <tt class="py-op">=</tt> <tt class="py-number">2</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">RuntimeError</tt><tt class="py-op">(</tt><tt class="py-string">"Number of items in corr_dict ({0}) is less "</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line">                               <tt class="py-string">"than number of protein records "</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line">                               <tt class="py-string">"({1})"</tt><tt class="py-op">.</tt><tt id="link-60" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-60', 'format', 'link-51');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">corr_dict</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">pro_num</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line"> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line">    <tt class="py-comment"># set up pro-nucl correspondence based on corr_method</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line">    <tt class="py-comment"># corr_method = 0, consecutive pairing</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">corr_method</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line">        <tt class="py-name">pro_nucl_pair</tt> <tt class="py-op">=</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">pro_align</tt><tt class="py-op">,</tt> <tt class="py-name">nucl_seqs</tt><tt class="py-op">)</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">    <tt class="py-comment"># corr_method = 1, keyword pairing</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">corr_method</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line">        <tt class="py-name">nucl_id</tt> <tt class="py-op">=</tt> <tt id="link-61" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.set()=Bio.Nexus.Nexus.StepMatrix-class.html#set,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.set()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#set,Method Bio.SVDSuperimposer.SVDSuperimposer.set()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#set"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-61', 'set', 'link-61');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">nucl_seqs</tt><tt class="py-op">.</tt><tt id="link-62" class="py-name" targets="Method Bio.Crystal.Crystal.keys()=Bio.Crystal.Crystal-class.html#keys,Variable Bio.Entrez.Parser.ListConsumer.keys=Bio.Entrez.Parser.ListConsumer-class.html#keys,Method Bio.File._IndexedSeqFileDict.keys()=Bio.File._IndexedSeqFileDict-class.html#keys,Method Bio.File._SQLiteManySeqFilesDict.keys()=Bio.File._SQLiteManySeqFilesDict-class.html#keys,Method Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys()=Bio.PDB.AbstractPropertyMap.AbstractPropertyMap-class.html#keys,Method Bio.Phylo.PhyloXML.Events.keys()=Bio.Phylo.PhyloXML.Events-class.html#keys,Method BioSQL.BioSeqDatabase.BioSeqDatabase.keys()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#keys,Method BioSQL.BioSeqDatabase.DBServer.keys()=BioSQL.BioSeqDatabase.DBServer-class.html#keys"><a title="Bio.Crystal.Crystal.keys
Bio.Entrez.Parser.ListConsumer.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-62', 'keys', 'link-62');">keys</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line">        <tt class="py-name">pro_id</tt> <tt class="py-op">=</tt> <tt id="link-63" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-63', 'set', 'link-61');">set</a></tt><tt class="py-op">(</tt><tt id="link-64" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-64', 'i', 'link-56');">i</a></tt><tt class="py-op">.</tt><tt id="link-65" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-65', 'id', 'link-57');">id</a></tt> <tt class="py-keyword">for</tt> <tt id="link-66" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-66', 'i', 'link-56');">i</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">pro_align</tt><tt class="py-op">)</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line">        <tt class="py-comment"># check if there is pro_id that does not have a nucleotide match</tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">pro_id</tt> <tt class="py-op">-</tt> <tt class="py-name">nucl_id</tt><tt class="py-op">:</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line">            <tt class="py-name">diff</tt> <tt class="py-op">=</tt> <tt class="py-name">pro_id</tt> <tt class="py-op">-</tt> <tt class="py-name">nucl_id</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Protein Record {0} cannot find a nucleotide "</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line">                             <tt class="py-string">"sequence match, please check the "</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">                             <tt class="py-string">"id"</tt><tt class="py-op">.</tt><tt id="link-67" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-67', 'format', 'link-51');">format</a></tt><tt class="py-op">(</tt><tt class="py-string">', '</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">diff</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line">            <tt class="py-name">pro_nucl_pair</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">pro_rec</tt> <tt class="py-keyword">in</tt> <tt class="py-name">pro_align</tt><tt class="py-op">:</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line">                <tt class="py-name">pro_nucl_pair</tt><tt class="py-op">.</tt><tt id="link-68" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-68', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">pro_rec</tt><tt class="py-op">,</tt> <tt class="py-name">nucl_seqs</tt><tt class="py-op">[</tt><tt class="py-name">pro_rec</tt><tt class="py-op">.</tt><tt id="link-69" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-69', 'id', 'link-57');">id</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">    <tt class="py-comment"># corr_method = 2, dict pairing</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">corr_method</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">        <tt class="py-name">pro_nucl_pair</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">pro_rec</tt> <tt class="py-keyword">in</tt> <tt class="py-name">pro_align</tt><tt class="py-op">:</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">                <tt class="py-name">nucl_id</tt> <tt class="py-op">=</tt> <tt class="py-name">corr_dict</tt><tt class="py-op">[</tt><tt class="py-name">pro_rec</tt><tt class="py-op">.</tt><tt id="link-70" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-70', 'id', 'link-57');">id</a></tt><tt class="py-op">]</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">                <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Protein record (%s) is not in corr_dict!"</tt> <tt class="py-op">%</tt> <tt class="py-name">pro_rec</tt><tt class="py-op">.</tt><tt id="link-71" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-71', 'id', 'link-57');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">                <tt class="py-name">exit</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">            <tt class="py-name">pro_nucl_pair</tt><tt class="py-op">.</tt><tt id="link-72" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-72', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">pro_rec</tt><tt class="py-op">,</tt> <tt class="py-name">nucl_seqs</tt><tt class="py-op">[</tt><tt class="py-name">nucl_id</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line"> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">    <tt class="py-name">codon_aln</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">    <tt class="py-name">shift</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">pair</tt> <tt class="py-keyword">in</tt> <tt class="py-name">pro_nucl_pair</tt><tt class="py-op">:</tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">        <tt class="py-comment"># Beware that the following span corresponds to an ungapped</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">        <tt class="py-comment"># nucleotide sequence.</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">        <tt class="py-name">corr_span</tt> <tt class="py-op">=</tt> <tt id="link-73" class="py-name" targets="Function Bio.codonalign._check_corr()=Bio.codonalign-module.html#_check_corr"><a title="Bio.codonalign._check_corr" class="py-name" href="#" onclick="return doclink('link-73', '_check_corr', 'link-73');">_check_corr</a></tt><tt class="py-op">(</tt><tt class="py-name">pair</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">pair</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">gap_char</tt><tt class="py-op">=</tt><tt class="py-name">gap_char</tt><tt class="py-op">,</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">                                <tt class="py-name">codon_table</tt><tt class="py-op">=</tt><tt class="py-name">codon_table</tt><tt class="py-op">,</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">                                <tt class="py-name">complete_protein</tt><tt class="py-op">=</tt><tt class="py-name">complete_protein</tt><tt class="py-op">,</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line">                                <tt class="py-name">anchor_len</tt><tt class="py-op">=</tt><tt class="py-name">anchor_len</tt><tt class="py-op">)</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">corr_span</tt><tt class="py-op">:</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Protein Record {0} and Nucleotide Record {1} do"</tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line">                             <tt class="py-string">" not match!"</tt><tt class="py-op">.</tt><tt id="link-74" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-74', 'format', 'link-51');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">pair</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-75" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-75', 'id', 'link-57');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">pair</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-76" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-76', 'id', 'link-57');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line">            <tt class="py-name">codon_rec</tt> <tt class="py-op">=</tt> <tt id="link-77" class="py-name" targets="Function Bio.codonalign._get_codon_rec()=Bio.codonalign-module.html#_get_codon_rec"><a title="Bio.codonalign._get_codon_rec" class="py-name" href="#" onclick="return doclink('link-77', '_get_codon_rec', 'link-77');">_get_codon_rec</a></tt><tt class="py-op">(</tt><tt class="py-name">pair</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">pair</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">corr_span</tt><tt class="py-op">,</tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line">                                       <tt id="link-78" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-78', 'alphabet', 'link-45');">alphabet</a></tt><tt class="py-op">=</tt><tt id="link-79" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-79', 'alphabet', 'link-45');">alphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">                                       <tt class="py-name">complete_protein</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">                                       <tt class="py-name">codon_table</tt><tt class="py-op">=</tt><tt class="py-name">codon_table</tt><tt class="py-op">,</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">                                       <tt class="py-name">max_score</tt><tt class="py-op">=</tt><tt class="py-name">max_score</tt><tt class="py-op">)</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">            <tt class="py-name">codon_aln</tt><tt class="py-op">.</tt><tt id="link-80" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-80', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">codon_rec</tt><tt class="py-op">)</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">corr_span</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">                <tt class="py-name">shift</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">shift</tt><tt class="py-op">:</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-81" class="py-name"><a title="Bio.codonalign.codonalignment.CodonAlignment" class="py-name" href="#" onclick="return doclink('link-81', 'CodonAlignment', 'link-19');">CodonAlignment</a></tt><tt class="py-op">(</tt><tt id="link-82" class="py-name" targets="Function Bio.codonalign._align_shift_recs()=Bio.codonalign-module.html#_align_shift_recs"><a title="Bio.codonalign._align_shift_recs" class="py-name" href="#" onclick="return doclink('link-82', '_align_shift_recs', 'link-82');">_align_shift_recs</a></tt><tt class="py-op">(</tt><tt class="py-name">codon_aln</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-83" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-83', 'alphabet', 'link-45');">alphabet</a></tt><tt class="py-op">=</tt><tt id="link-84" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-84', 'alphabet', 'link-45');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-85" class="py-name"><a title="Bio.codonalign.codonalignment.CodonAlignment" class="py-name" href="#" onclick="return doclink('link-85', 'CodonAlignment', 'link-19');">CodonAlignment</a></tt><tt class="py-op">(</tt><tt class="py-name">codon_aln</tt><tt class="py-op">,</tt> <tt id="link-86" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-86', 'alphabet', 'link-45');">alphabet</a></tt><tt class="py-op">=</tt><tt id="link-87" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-87', 'alphabet', 'link-45');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line"> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line"> </tt>
<a name="_codons2re"></a><div id="_codons2re-def"><a name="L188"></a><tt class="py-lineno">188</tt> <a class="py-toggle" href="#" id="_codons2re-toggle" onclick="return toggle('_codons2re');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.codonalign-module.html#_codons2re">_codons2re</a><tt class="py-op">(</tt><tt class="py-param">codons</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_codons2re-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_codons2re-expanded"><a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">    <tt class="py-docstring">"""Generate regular expression based on a given list of codons (PRIVATE)."""</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">    <tt class="py-name">reg</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt id="link-88" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-88', 'i', 'link-56');">i</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-op">*</tt><tt class="py-name">codons</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-89" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-89', 'set', 'link-61');">set</a></tt><tt class="py-op">(</tt><tt id="link-90" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-90', 'i', 'link-56');">i</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">            <tt class="py-name">reg</tt> <tt class="py-op">+=</tt> <tt class="py-string">''</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt id="link-91" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-91', 'set', 'link-61');">set</a></tt><tt class="py-op">(</tt><tt id="link-92" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-92', 'i', 'link-56');">i</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">            <tt class="py-name">reg</tt> <tt class="py-op">+=</tt> <tt class="py-string">'['</tt> <tt class="py-op">+</tt> <tt class="py-string">''</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt id="link-93" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-93', 'set', 'link-61');">set</a></tt><tt class="py-op">(</tt><tt id="link-94" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-94', 'i', 'link-56');">i</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">']'</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">reg</tt> </tt>
</div><a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line"> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line"> </tt>
<a name="_get_aa_regex"></a><div id="_get_aa_regex-def"><a name="L199"></a><tt class="py-lineno">199</tt> <a class="py-toggle" href="#" id="_get_aa_regex-toggle" onclick="return toggle('_get_aa_regex');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.codonalign-module.html#_get_aa_regex">_get_aa_regex</a><tt class="py-op">(</tt><tt class="py-param">codon_table</tt><tt class="py-op">,</tt> <tt class="py-param">stop</tt><tt class="py-op">=</tt><tt class="py-string">'*'</tt><tt class="py-op">,</tt> <tt class="py-param">unknown</tt><tt class="py-op">=</tt><tt class="py-string">'X'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_get_aa_regex-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_get_aa_regex-expanded"><a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line">    <tt class="py-docstring">"""Set up the regular expression of a given CodonTable (PRIVATE).</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio.Data.CodonTable import generic_by_id</tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; p = generic_by_id[1]</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; t = _get_aa_regex(p)</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(t['A'][0])</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line"><tt class="py-docstring">    G</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(t['A'][1])</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line"><tt class="py-docstring">    C</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(sorted(list(t['A'][2:])))</tt> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line"><tt class="py-docstring">    ['A', 'C', 'G', 'T', 'U', '[', ']']</tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(sorted(list(t['L'][:5])))</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line"><tt class="py-docstring">    ['C', 'T', 'U', '[', ']']</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(sorted(list(t['L'][5:9])))</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line"><tt class="py-docstring">    ['T', 'U', '[', ']']</tt> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(sorted(list(t['L'][9:])))</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line"><tt class="py-docstring">    ['A', 'C', 'G', 'T', 'U', '[', ']']</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-95" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-95', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-96" class="py-name" targets="Package Bio.Data=Bio.Data-module.html"><a title="Bio.Data" class="py-name" href="#" onclick="return doclink('link-96', 'Data', 'link-96');">Data</a></tt><tt class="py-op">.</tt><tt id="link-97" class="py-name" targets="Module Bio.Data.CodonTable=Bio.Data.CodonTable-module.html,Class Bio.Data.CodonTable.CodonTable=Bio.Data.CodonTable.CodonTable-class.html"><a title="Bio.Data.CodonTable
Bio.Data.CodonTable.CodonTable" class="py-name" href="#" onclick="return doclink('link-97', 'CodonTable', 'link-97');">CodonTable</a></tt> <tt class="py-keyword">import</tt> <tt id="link-98" class="py-name"><a title="Bio.Data.CodonTable
Bio.Data.CodonTable.CodonTable" class="py-name" href="#" onclick="return doclink('link-98', 'CodonTable', 'link-97');">CodonTable</a></tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">codon_table</tt><tt class="py-op">,</tt> <tt id="link-99" class="py-name"><a title="Bio.Data.CodonTable
Bio.Data.CodonTable.CodonTable" class="py-name" href="#" onclick="return doclink('link-99', 'CodonTable', 'link-97');">CodonTable</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"Input table is not a instance of "</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line">                        <tt class="py-string">"Bio.Data.CodonTable object"</tt><tt class="py-op">)</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line">    <tt class="py-name">aa2codon</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">codon</tt><tt class="py-op">,</tt> <tt class="py-name">aa</tt> <tt class="py-keyword">in</tt> <tt class="py-name">codon_table</tt><tt class="py-op">.</tt><tt id="link-100" class="py-name" targets="Variable Bio.Data.CodonTable.CodonTable.forward_table=Bio.Data.CodonTable.CodonTable-class.html#forward_table"><a title="Bio.Data.CodonTable.CodonTable.forward_table" class="py-name" href="#" onclick="return doclink('link-100', 'forward_table', 'link-100');">forward_table</a></tt><tt class="py-op">.</tt><tt id="link-101" class="py-name" targets="Method Bio.Crystal.Crystal.items()=Bio.Crystal.Crystal-class.html#items,Method Bio.File._IndexedSeqFileDict.items()=Bio.File._IndexedSeqFileDict-class.html#items,Method Bio.Phylo.PhyloXML.Events.items()=Bio.Phylo.PhyloXML.Events-class.html#items,Variable Bio.SearchIO._model.query.QueryResult.items=Bio.SearchIO._model.query.QueryResult-class.html#items,Method BioSQL.BioSeqDatabase.BioSeqDatabase.items()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#items,Method BioSQL.BioSeqDatabase.DBServer.items()=BioSQL.BioSeqDatabase.DBServer-class.html#items"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-101', 'items', 'link-101');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line">        <tt class="py-name">aa2codon</tt><tt class="py-op">.</tt><tt id="link-102" class="py-name" targets="Method Bio.Crystal.Crystal.setdefault()=Bio.Crystal.Crystal-class.html#setdefault"><a title="Bio.Crystal.Crystal.setdefault" class="py-name" href="#" onclick="return doclink('link-102', 'setdefault', 'link-102');">setdefault</a></tt><tt class="py-op">(</tt><tt class="py-name">aa</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-103" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-103', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">codon</tt><tt class="py-op">)</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">aa</tt><tt class="py-op">,</tt> <tt class="py-name">codons</tt> <tt class="py-keyword">in</tt> <tt class="py-name">aa2codon</tt><tt class="py-op">.</tt><tt id="link-104" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-104', 'items', 'link-101');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line">        <tt class="py-name">aa2codon</tt><tt class="py-op">[</tt><tt class="py-name">aa</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-105" class="py-name" targets="Function Bio.codonalign._codons2re()=Bio.codonalign-module.html#_codons2re"><a title="Bio.codonalign._codons2re" class="py-name" href="#" onclick="return doclink('link-105', '_codons2re', 'link-105');">_codons2re</a></tt><tt class="py-op">(</tt><tt class="py-name">codons</tt><tt class="py-op">)</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line">    <tt class="py-name">aa2codon</tt><tt class="py-op">[</tt><tt class="py-name">stop</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-106" class="py-name"><a title="Bio.codonalign._codons2re" class="py-name" href="#" onclick="return doclink('link-106', '_codons2re', 'link-105');">_codons2re</a></tt><tt class="py-op">(</tt><tt class="py-name">codon_table</tt><tt class="py-op">.</tt><tt id="link-107" class="py-name" targets="Variable Bio.Data.CodonTable.CodonTable.stop_codons=Bio.Data.CodonTable.CodonTable-class.html#stop_codons"><a title="Bio.Data.CodonTable.CodonTable.stop_codons" class="py-name" href="#" onclick="return doclink('link-107', 'stop_codons', 'link-107');">stop_codons</a></tt><tt class="py-op">)</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">    <tt class="py-name">aa2codon</tt><tt class="py-op">[</tt><tt class="py-name">unknown</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">'...'</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">aa2codon</tt> </tt>
</div><a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line"> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line"> </tt>
<a name="_check_corr"></a><div id="_check_corr-def"><a name="L233"></a><tt class="py-lineno">233</tt> <a class="py-toggle" href="#" id="_check_corr-toggle" onclick="return toggle('_check_corr');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.codonalign-module.html#_check_corr">_check_corr</a><tt class="py-op">(</tt><tt class="py-param">pro</tt><tt class="py-op">,</tt> <tt class="py-param">nucl</tt><tt class="py-op">,</tt> <tt class="py-param">gap_char</tt><tt class="py-op">=</tt><tt class="py-string">'-'</tt><tt class="py-op">,</tt> <tt class="py-param">codon_table</tt><tt class="py-op">=</tt><tt id="link-108" class="py-name"><a title="Bio.codonalign.codonalphabet.default_codon_table" class="py-name" href="#" onclick="return doclink('link-108', 'default_codon_table', 'link-28');">default_codon_table</a></tt><tt class="py-op">,</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line">                <tt class="py-param">complete_protein</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> <tt class="py-param">anchor_len</tt><tt class="py-op">=</tt><tt class="py-number">10</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_check_corr-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_check_corr-expanded"><a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line">    <tt class="py-docstring">"""Check if the nucleotide can be translated into the protein (PRIVATE).</tt> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line"><tt class="py-docstring">    Expects two SeqRecord objects.</tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line">    <tt class="py-keyword">import</tt> <tt class="py-name">re</tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-109" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-109', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-110" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-110', 'Alphabet', 'link-7');">Alphabet</a></tt> <tt class="py-keyword">import</tt> <tt id="link-111" class="py-name" targets="Class Bio.Alphabet.NucleotideAlphabet=Bio.Alphabet.NucleotideAlphabet-class.html"><a title="Bio.Alphabet.NucleotideAlphabet" class="py-name" href="#" onclick="return doclink('link-111', 'NucleotideAlphabet', 'link-111');">NucleotideAlphabet</a></tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line"> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">pro</tt><tt class="py-op">,</tt> <tt id="link-112" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-112', 'SeqRecord', 'link-10');">SeqRecord</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">nucl</tt><tt class="py-op">,</tt> <tt id="link-113" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-113', 'SeqRecord', 'link-10');">SeqRecord</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"_check_corr accepts two SeqRecord object. Please "</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line">                        <tt class="py-string">"check your input."</tt><tt class="py-op">)</tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line"> </tt>
<a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line">    <tt class="py-keyword">def</tt> <tt class="py-def-name">get_alpha</tt><tt class="py-op">(</tt><tt class="py-param">alpha</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">alpha</tt><tt class="py-op">,</tt> <tt class="py-string">'alphabet'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">get_alpha</tt><tt class="py-op">(</tt><tt class="py-name">alpha</tt><tt class="py-op">.</tt><tt id="link-114" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-114', 'alphabet', 'link-45');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">alpha</tt> </tt>
</div><a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line"> </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-115" class="py-name"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-115', '_get_base_alphabet', 'link-8');">_get_base_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">get_alpha</tt><tt class="py-op">(</tt><tt class="py-name">nucl</tt><tt class="py-op">.</tt><tt id="link-116" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-116', 'seq', 'link-44');">seq</a></tt><tt class="py-op">.</tt><tt id="link-117" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-117', 'alphabet', 'link-45');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line">                      <tt id="link-118" class="py-name"><a title="Bio.Alphabet.NucleotideAlphabet" class="py-name" href="#" onclick="return doclink('link-118', 'NucleotideAlphabet', 'link-111');">NucleotideAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"Alphabet for nucl should be an instance of "</tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line">                        <tt class="py-string">"NucleotideAlphabet, {0} "</tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line">                        <tt class="py-string">"detected"</tt><tt class="py-op">.</tt><tt id="link-119" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-119', 'format', 'link-51');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">nucl</tt><tt class="py-op">.</tt><tt id="link-120" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-120', 'seq', 'link-44');">seq</a></tt><tt class="py-op">.</tt><tt id="link-121" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-121', 'alphabet', 'link-45');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line"> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line">    <tt class="py-name">aa2re</tt> <tt class="py-op">=</tt> <tt id="link-122" class="py-name" targets="Function Bio.codonalign._get_aa_regex()=Bio.codonalign-module.html#_get_aa_regex"><a title="Bio.codonalign._get_aa_regex" class="py-name" href="#" onclick="return doclink('link-122', '_get_aa_regex', 'link-122');">_get_aa_regex</a></tt><tt class="py-op">(</tt><tt class="py-name">codon_table</tt><tt class="py-op">)</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line">    <tt class="py-name">pro_re</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">aa</tt> <tt class="py-keyword">in</tt> <tt class="py-name">pro</tt><tt class="py-op">.</tt><tt id="link-123" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-123', 'seq', 'link-44');">seq</a></tt><tt class="py-op">:</tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">aa</tt> <tt class="py-op">!=</tt> <tt class="py-name">gap_char</tt><tt class="py-op">:</tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line">            <tt class="py-name">pro_re</tt> <tt class="py-op">+=</tt> <tt class="py-name">aa2re</tt><tt class="py-op">[</tt><tt class="py-name">aa</tt><tt class="py-op">]</tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line"> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line">    <tt class="py-name">nucl_seq</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">nucl</tt><tt class="py-op">.</tt><tt id="link-124" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-124', 'seq', 'link-44');">seq</a></tt><tt class="py-op">.</tt><tt id="link-125" class="py-name" targets="Method Bio.Seq.Seq.upper()=Bio.Seq.Seq-class.html#upper,Method Bio.Seq.UnknownSeq.upper()=Bio.Seq.UnknownSeq-class.html#upper,Method Bio.SeqRecord.SeqRecord.upper()=Bio.SeqRecord.SeqRecord-class.html#upper"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-125', 'upper', 'link-125');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-126" class="py-name" targets="Method Bio.Seq.Seq.ungap()=Bio.Seq.Seq-class.html#ungap,Method Bio.Seq.UnknownSeq.ungap()=Bio.Seq.UnknownSeq-class.html#ungap,Method Bio.codonalign.codonseq.CodonSeq.ungap()=Bio.codonalign.codonseq.CodonSeq-class.html#ungap"><a title="Bio.Seq.Seq.ungap
Bio.Seq.UnknownSeq.ungap
Bio.codonalign.codonseq.CodonSeq.ungap" class="py-name" href="#" onclick="return doclink('link-126', 'ungap', 'link-126');">ungap</a></tt><tt class="py-op">(</tt><tt class="py-name">gap_char</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line">    <tt id="link-127" class="py-name" targets="Function Bio.triefind.match()=Bio.triefind-module.html#match"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-127', 'match', 'link-127');">match</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-128" class="py-name" targets="Method Bio.AlignIO.MafIO.MafIndex.search()=Bio.AlignIO.MafIO.MafIndex-class.html#search,Method Bio.KDTree.KDTree'.KDTree.search()=Bio.KDTree.KDTree%27.KDTree-class.html#search,Method Bio.PDB.NeighborSearch'.NeighborSearch.search()=Bio.PDB.NeighborSearch%27.NeighborSearch-class.html#search,Method Bio.Phylo.TreeConstruction.NNITreeSearcher.search()=Bio.Phylo.TreeConstruction.NNITreeSearcher-class.html#search,Method Bio.Phylo.TreeConstruction.TreeSearcher.search()=Bio.Phylo.TreeConstruction.TreeSearcher-class.html#search,Class Method Bio.Restriction.Restriction.AbstractCut.search()=Bio.Restriction.Restriction.AbstractCut-class.html#search,Method Bio.Restriction.Restriction.RestrictionBatch.search()=Bio.Restriction.Restriction.RestrictionBatch-class.html#search,Function Bio.SCOP.search()=Bio.SCOP-module.html#search,Function Bio.TogoWS.search()=Bio.TogoWS-module.html#search,Method Bio.motifs.Instances.search()=Bio.motifs.Instances-class.html#search,Method Bio.motifs.matrix.PositionSpecificScoringMatrix.search()=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#search"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-128', 'search', 'link-128');">search</a></tt><tt class="py-op">(</tt><tt class="py-name">pro_re</tt><tt class="py-op">,</tt> <tt class="py-name">nucl_seq</tt><tt class="py-op">)</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt id="link-129" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-129', 'match', 'link-127');">match</a></tt><tt class="py-op">:</tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">        <tt class="py-comment"># mode = 0, direct match</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt id="link-130" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-130', 'match', 'link-127');">match</a></tt><tt class="py-op">.</tt><tt class="py-name">span</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">        <tt class="py-comment"># Might caused by mismatches or frameshift, using anchors to</tt> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line">        <tt class="py-comment"># have a try</tt> </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line">        <tt class="py-comment"># anchor_len = 10 # adjust this value to test performance</tt> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line">        <tt class="py-name">pro_seq</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">pro</tt><tt class="py-op">.</tt><tt id="link-131" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-131', 'seq', 'link-44');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-name">gap_char</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt> </tt>
<a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line">        <tt class="py-name">anchors</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">pro_seq</tt><tt class="py-op">[</tt><tt id="link-132" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-132', 'i', 'link-56');">i</a></tt><tt class="py-op">:</tt><tt class="py-op">(</tt><tt id="link-133" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-133', 'i', 'link-56');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-name">anchor_len</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt id="link-134" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-134', 'i', 'link-56');">i</a></tt> <tt class="py-keyword">in</tt> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line">                   <tt id="link-135" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-135', 'range', 'link-135');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">pro_seq</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">anchor_len</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line">        <tt class="py-comment"># if the last anchor is less than the specified anchor</tt> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line">        <tt class="py-comment"># size, we combine the penultimate and the last anchor</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line">        <tt class="py-comment"># together as the last one.</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line">        <tt class="py-comment"># TODO: modify this to deal with short sequence with only</tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line">        <tt class="py-comment"># one anchor.</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">anchors</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">anchor_len</tt><tt class="py-op">:</tt> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line">            <tt class="py-name">anchors</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">anchors</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">anchors</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line"> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">        <tt class="py-name">pro_re</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line">        <tt class="py-name">anchor_distance</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line">        <tt class="py-name">anchor_pos</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-136" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-136', 'i', 'link-56');">i</a></tt><tt class="py-op">,</tt> <tt class="py-name">anchor</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt class="py-name">anchors</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">            <tt class="py-name">this_anchor_len</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">anchor</tt><tt class="py-op">)</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">            <tt class="py-name">qcodon</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">            <tt class="py-name">fncodon</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line">            <tt class="py-comment"># dirty code to deal with the last anchor</tt> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line">            <tt class="py-comment"># as the last anchor is combined in the steps</tt> </tt>
<a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line">            <tt class="py-comment"># above, we need to get the true last anchor to</tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line">            <tt class="py-comment"># pro_re</tt> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">this_anchor_len</tt> <tt class="py-op">==</tt> <tt class="py-name">anchor_len</tt><tt class="py-op">:</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">aa</tt> <tt class="py-keyword">in</tt> <tt class="py-name">anchor</tt><tt class="py-op">:</tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">complete_protein</tt> <tt class="py-keyword">and</tt> <tt id="link-137" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-137', 'i', 'link-56');">i</a></tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line">                        <tt class="py-name">qcodon</tt> <tt class="py-op">+=</tt> <tt id="link-138" class="py-name"><a title="Bio.codonalign._codons2re" class="py-name" href="#" onclick="return doclink('link-138', '_codons2re', 'link-105');">_codons2re</a></tt><tt class="py-op">(</tt><tt class="py-name">codon_table</tt><tt class="py-op">.</tt><tt id="link-139" class="py-name" targets="Variable Bio.Data.CodonTable.CodonTable.start_codons=Bio.Data.CodonTable.CodonTable-class.html#start_codons"><a title="Bio.Data.CodonTable.CodonTable.start_codons" class="py-name" href="#" onclick="return doclink('link-139', 'start_codons', 'link-139');">start_codons</a></tt><tt class="py-op">)</tt> </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line">                        <tt class="py-name">fncodon</tt> <tt class="py-op">+=</tt> <tt class="py-name">aa2re</tt><tt class="py-op">[</tt><tt class="py-string">'X'</tt><tt class="py-op">]</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line">                        <tt class="py-keyword">continue</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line">                    <tt class="py-name">qcodon</tt> <tt class="py-op">+=</tt> <tt class="py-name">aa2re</tt><tt class="py-op">[</tt><tt class="py-name">aa</tt><tt class="py-op">]</tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">                    <tt class="py-name">fncodon</tt> <tt class="py-op">+=</tt> <tt class="py-name">aa2re</tt><tt class="py-op">[</tt><tt class="py-string">'X'</tt><tt class="py-op">]</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line">                <tt id="link-140" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-140', 'match', 'link-127');">match</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-141" class="py-name"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-141', 'search', 'link-128');">search</a></tt><tt class="py-op">(</tt><tt class="py-name">qcodon</tt><tt class="py-op">,</tt> <tt class="py-name">nucl_seq</tt><tt class="py-op">)</tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">this_anchor_len</tt> <tt class="py-op">&gt;</tt> <tt class="py-name">anchor_len</tt><tt class="py-op">:</tt> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line">                <tt class="py-name">last_qcodon</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line">                <tt class="py-name">last_fcodon</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">j</tt> <tt class="py-keyword">in</tt> <tt id="link-142" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-142', 'range', 'link-135');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">anchor_len</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">anchor</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line">                    <tt class="py-name">last_qcodon</tt> <tt class="py-op">+=</tt> <tt class="py-name">aa2re</tt><tt class="py-op">[</tt><tt class="py-name">anchor</tt><tt class="py-op">[</tt><tt class="py-name">j</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line">                    <tt class="py-name">last_fcodon</tt> <tt class="py-op">+=</tt> <tt class="py-name">aa2re</tt><tt class="py-op">[</tt><tt class="py-string">'X'</tt><tt class="py-op">]</tt> </tt>
<a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line">                <tt id="link-143" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-143', 'match', 'link-127');">match</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-144" class="py-name"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-144', 'search', 'link-128');">search</a></tt><tt class="py-op">(</tt><tt class="py-name">last_qcodon</tt><tt class="py-op">,</tt> <tt class="py-name">nucl_seq</tt><tt class="py-op">)</tt> </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line">            <tt class="py-comment"># build full_pro_re from anchors</tt> </tt>
<a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-145" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-145', 'match', 'link-127');">match</a></tt><tt class="py-op">:</tt> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line">                <tt class="py-name">anchor_pos</tt><tt class="py-op">.</tt><tt id="link-146" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-146', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt id="link-147" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-147', 'match', 'link-127');">match</a></tt><tt class="py-op">.</tt><tt id="link-148" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.start=Bio.SeqFeature.CompoundLocation-class.html#start,Variable Bio.SeqFeature.FeatureLocation.start=Bio.SeqFeature.FeatureLocation-class.html#start"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-148', 'start', 'link-148');">start</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-149" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-149', 'match', 'link-127');">match</a></tt><tt class="py-op">.</tt><tt id="link-150" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.end=Bio.SeqFeature.CompoundLocation-class.html#end,Variable Bio.SeqFeature.FeatureLocation.end=Bio.SeqFeature.FeatureLocation-class.html#end"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-150', 'end', 'link-150');">end</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-151" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-151', 'i', 'link-56');">i</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">this_anchor_len</tt> <tt class="py-op">==</tt> <tt class="py-name">anchor_len</tt><tt class="py-op">:</tt> </tt>
<a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line">                    <tt class="py-name">pro_re</tt><tt class="py-op">.</tt><tt id="link-152" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-152', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">qcodon</tt><tt class="py-op">)</tt> </tt>
<a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line">                    <tt class="py-name">pro_re</tt><tt class="py-op">.</tt><tt id="link-153" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-153', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">last_qcodon</tt><tt class="py-op">)</tt> </tt>
<a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">this_anchor_len</tt> <tt class="py-op">==</tt> <tt class="py-name">anchor_len</tt><tt class="py-op">:</tt> </tt>
<a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line">                    <tt class="py-name">pro_re</tt><tt class="py-op">.</tt><tt id="link-154" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-154', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">fncodon</tt><tt class="py-op">)</tt> </tt>
<a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">                    <tt class="py-name">pro_re</tt><tt class="py-op">.</tt><tt id="link-155" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-155', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">last_fcodon</tt><tt class="py-op">)</tt> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">        <tt class="py-name">full_pro_re</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">pro_re</tt><tt class="py-op">)</tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line">        <tt id="link-156" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-156', 'match', 'link-127');">match</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-157" class="py-name"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-157', 'search', 'link-128');">search</a></tt><tt class="py-op">(</tt><tt class="py-name">full_pro_re</tt><tt class="py-op">,</tt> <tt class="py-name">nucl_seq</tt><tt class="py-op">)</tt> </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-158" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-158', 'match', 'link-127');">match</a></tt><tt class="py-op">:</tt> </tt>
<a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">            <tt class="py-comment"># mode = 1, mismatch</tt> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt id="link-159" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-159', 'match', 'link-127');">match</a></tt><tt class="py-op">.</tt><tt class="py-name">span</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line">            <tt class="py-comment"># check frames of anchors</tt> </tt>
<a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line">            <tt class="py-comment"># ten frameshift events are allowed in a sequence</tt> </tt>
<a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line">            <tt class="py-name">first_anchor</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line">            <tt class="py-name">shift_id_pos</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line">            <tt class="py-comment"># check the first anchor</tt> </tt>
<a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">first_anchor</tt> <tt class="py-keyword">and</tt> <tt class="py-name">anchor_pos</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line">                <tt class="py-name">shift_val_lst</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">2</tt><tt class="py-op">,</tt> <tt class="py-number">3</tt> <tt class="py-op">*</tt> <tt class="py-name">anchor_len</tt> <tt class="py-op">-</tt> <tt class="py-number">2</tt><tt class="py-op">,</tt> <tt class="py-number">3</tt> <tt class="py-op">*</tt> <tt class="py-name">anchor_len</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line">                <tt class="py-name">sh_anc</tt> <tt class="py-op">=</tt> <tt class="py-name">anchors</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">shift_val</tt> <tt class="py-keyword">in</tt> <tt class="py-name">shift_val_lst</tt><tt class="py-op">:</tt> </tt>
<a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">shift_val</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line">                        <tt class="py-name">qcodon</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line">                        <tt class="py-keyword">break</tt> </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">shift_val</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">2</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line">                        <tt class="py-name">sh_nuc_len</tt> <tt class="py-op">=</tt> <tt class="py-name">anchor_len</tt> <tt class="py-op">*</tt> <tt class="py-number">3</tt> <tt class="py-op">+</tt> <tt class="py-name">shift_val</tt> </tt>
<a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line">                    <tt class="py-keyword">elif</tt> <tt class="py-name">shift_val</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-number">3</tt> <tt class="py-op">*</tt> <tt class="py-name">anchor_len</tt> <tt class="py-op">-</tt> <tt class="py-number">2</tt><tt class="py-op">,</tt> <tt class="py-number">3</tt> <tt class="py-op">*</tt> <tt class="py-name">anchor_len</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line">                        <tt class="py-name">sh_nuc_len</tt> <tt class="py-op">=</tt> <tt class="py-name">anchor_len</tt> <tt class="py-op">*</tt> <tt class="py-number">3</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-number">3</tt> <tt class="py-op">*</tt> <tt class="py-name">anchor_len</tt> <tt class="py-op">-</tt> <tt class="py-name">shift_val</tt><tt class="py-op">)</tt> </tt>
<a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">anchor_pos</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">&gt;=</tt> <tt class="py-name">sh_nuc_len</tt><tt class="py-op">:</tt> </tt>
<a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line">                        <tt class="py-name">sh_nuc</tt> <tt class="py-op">=</tt> <tt class="py-name">nucl_seq</tt><tt class="py-op">[</tt><tt class="py-name">anchor_pos</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">-</tt> <tt class="py-name">sh_nuc_len</tt><tt class="py-op">:</tt><tt class="py-name">anchor_pos</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L347"></a><tt class="py-lineno">347</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line">                        <tt class="py-comment"># this is unlikely to produce the correct output</tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line">                        <tt class="py-name">sh_nuc</tt> <tt class="py-op">=</tt> <tt class="py-name">nucl_seq</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">anchor_pos</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line">                    <tt class="py-name">qcodon</tt><tt class="py-op">,</tt> <tt class="py-name">shift_id_pos</tt> <tt class="py-op">=</tt> <tt id="link-160" class="py-name" targets="Function Bio.codonalign._get_shift_anchor_re()=Bio.codonalign-module.html#_get_shift_anchor_re"><a title="Bio.codonalign._get_shift_anchor_re" class="py-name" href="#" onclick="return doclink('link-160', '_get_shift_anchor_re', 'link-160');">_get_shift_anchor_re</a></tt><tt class="py-op">(</tt><tt class="py-name">sh_anc</tt><tt class="py-op">,</tt> <tt class="py-name">sh_nuc</tt><tt class="py-op">,</tt> </tt>
<a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line">                                                                <tt class="py-name">shift_val</tt><tt class="py-op">,</tt> </tt>
<a name="L352"></a><tt class="py-lineno">352</tt>  <tt class="py-line">                                                                <tt class="py-name">aa2re</tt><tt class="py-op">,</tt> </tt>
<a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line">                                                                <tt class="py-name">anchor_len</tt><tt class="py-op">,</tt> </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line">                                                                <tt class="py-name">shift_id_pos</tt><tt class="py-op">)</tt> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">qcodon</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> <tt class="py-name">qcodon</tt> <tt class="py-op">!=</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line">                        <tt class="py-comment"># pro_re[0] should be '.'*anchor_len, therefore I</tt> </tt>
<a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line">                        <tt class="py-comment"># replace it.</tt> </tt>
<a name="L358"></a><tt class="py-lineno">358</tt>  <tt class="py-line">                        <tt class="py-name">pro_re</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">qcodon</tt> </tt>
<a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line">                        <tt class="py-keyword">break</tt> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">qcodon</tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line">                    <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"first frameshift detection failed for "</tt> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line">                                  <tt class="py-string">"{0}"</tt><tt class="py-op">.</tt><tt id="link-161" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-161', 'format', 'link-51');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">nucl</tt><tt class="py-op">.</tt><tt id="link-162" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-162', 'id', 'link-57');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-163" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-163', 'BiopythonWarning', 'link-1');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line">            <tt class="py-comment"># check anchors in the middle</tt> </tt>
<a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-164" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-164', 'i', 'link-56');">i</a></tt> <tt class="py-keyword">in</tt> <tt id="link-165" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-165', 'range', 'link-135');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">anchor_pos</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line">                <tt class="py-name">shift_val</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">anchor_pos</tt><tt class="py-op">[</tt><tt id="link-166" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-166', 'i', 'link-56');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">-</tt> <tt class="py-name">anchor_pos</tt><tt class="py-op">[</tt><tt id="link-167" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-167', 'i', 'link-56');">i</a></tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">%</tt> \ </tt>
<a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line">                            <tt class="py-op">(</tt><tt class="py-number">3</tt> <tt class="py-op">*</tt> <tt class="py-name">anchor_len</tt><tt class="py-op">)</tt> </tt>
<a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line">                <tt class="py-name">sh_anc</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">anchors</tt><tt class="py-op">[</tt><tt class="py-name">anchor_pos</tt><tt class="py-op">[</tt><tt id="link-168" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-168', 'i', 'link-56');">i</a></tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">:</tt><tt class="py-name">anchor_pos</tt><tt class="py-op">[</tt><tt id="link-169" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-169', 'i', 'link-56');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L368"></a><tt class="py-lineno">368</tt>  <tt class="py-line">                <tt class="py-name">sh_nuc</tt> <tt class="py-op">=</tt> <tt class="py-name">nucl_seq</tt><tt class="py-op">[</tt><tt class="py-name">anchor_pos</tt><tt class="py-op">[</tt><tt id="link-170" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-170', 'i', 'link-56');">i</a></tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">:</tt><tt class="py-name">anchor_pos</tt><tt class="py-op">[</tt><tt id="link-171" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-171', 'i', 'link-56');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line">                <tt class="py-name">qcodon</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">shift_val</tt> <tt class="py-op">!=</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line">                    <tt class="py-name">qcodon</tt><tt class="py-op">,</tt> <tt class="py-name">shift_id_pos</tt> <tt class="py-op">=</tt> <tt id="link-172" class="py-name"><a title="Bio.codonalign._get_shift_anchor_re" class="py-name" href="#" onclick="return doclink('link-172', '_get_shift_anchor_re', 'link-160');">_get_shift_anchor_re</a></tt><tt class="py-op">(</tt><tt class="py-name">sh_anc</tt><tt class="py-op">,</tt> <tt class="py-name">sh_nuc</tt><tt class="py-op">,</tt> </tt>
<a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line">                                                                <tt class="py-name">shift_val</tt><tt class="py-op">,</tt> </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line">                                                                <tt class="py-name">aa2re</tt><tt class="py-op">,</tt> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line">                                                                <tt class="py-name">anchor_len</tt><tt class="py-op">,</tt> </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line">                                                                <tt class="py-name">shift_id_pos</tt><tt class="py-op">)</tt> </tt>
<a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">qcodon</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> <tt class="py-name">qcodon</tt> <tt class="py-op">!=</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line">                    <tt class="py-name">pro_re</tt><tt class="py-op">[</tt><tt class="py-name">anchor_pos</tt><tt class="py-op">[</tt><tt id="link-173" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-173', 'i', 'link-56');">i</a></tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">:</tt><tt class="py-name">anchor_pos</tt><tt class="py-op">[</tt><tt id="link-174" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-174', 'i', 'link-56');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">qcodon</tt><tt class="py-op">]</tt> </tt>
<a name="L378"></a><tt class="py-lineno">378</tt>  <tt class="py-line">                    <tt class="py-name">qcodon</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L379"></a><tt class="py-lineno">379</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">qcodon</tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L380"></a><tt class="py-lineno">380</tt>  <tt class="py-line">                    <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"middle frameshift detection failed for "</tt> </tt>
<a name="L381"></a><tt class="py-lineno">381</tt>  <tt class="py-line">                                  <tt class="py-string">"{0}"</tt><tt class="py-op">.</tt><tt id="link-175" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-175', 'format', 'link-51');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">nucl</tt><tt class="py-op">.</tt><tt id="link-176" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-176', 'id', 'link-57');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-177" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-177', 'BiopythonWarning', 'link-1');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L382"></a><tt class="py-lineno">382</tt>  <tt class="py-line">            <tt class="py-comment"># check the last anchor</tt> </tt>
<a name="L383"></a><tt class="py-lineno">383</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">anchor_pos</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">anchors</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L384"></a><tt class="py-lineno">384</tt>  <tt class="py-line">                <tt class="py-name">sh_anc</tt> <tt class="py-op">=</tt> <tt class="py-name">anchors</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L385"></a><tt class="py-lineno">385</tt>  <tt class="py-line">                <tt class="py-name">this_anchor_len</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">sh_anc</tt><tt class="py-op">)</tt> </tt>
<a name="L386"></a><tt class="py-lineno">386</tt>  <tt class="py-line">                <tt class="py-name">shift_val_lst</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">2</tt><tt class="py-op">,</tt> <tt class="py-number">3</tt> <tt class="py-op">*</tt> <tt class="py-name">this_anchor_len</tt> <tt class="py-op">-</tt> <tt class="py-number">2</tt><tt class="py-op">,</tt> <tt class="py-number">3</tt> <tt class="py-op">*</tt> <tt class="py-name">this_anchor_len</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L387"></a><tt class="py-lineno">387</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">shift_val</tt> <tt class="py-keyword">in</tt> <tt class="py-name">shift_val_lst</tt><tt class="py-op">:</tt> </tt>
<a name="L388"></a><tt class="py-lineno">388</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">shift_val</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L389"></a><tt class="py-lineno">389</tt>  <tt class="py-line">                        <tt class="py-name">qcodon</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L390"></a><tt class="py-lineno">390</tt>  <tt class="py-line">                        <tt class="py-keyword">break</tt> </tt>
<a name="L391"></a><tt class="py-lineno">391</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">shift_val</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">2</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L392"></a><tt class="py-lineno">392</tt>  <tt class="py-line">                        <tt class="py-name">sh_nuc_len</tt> <tt class="py-op">=</tt> <tt class="py-name">this_anchor_len</tt> <tt class="py-op">*</tt> <tt class="py-number">3</tt> <tt class="py-op">+</tt> <tt class="py-name">shift_val</tt> </tt>
<a name="L393"></a><tt class="py-lineno">393</tt>  <tt class="py-line">                    <tt class="py-keyword">elif</tt> <tt class="py-name">shift_val</tt> <tt class="py-keyword">in</tt> \ </tt>
<a name="L394"></a><tt class="py-lineno">394</tt>  <tt class="py-line">                            <tt class="py-op">(</tt><tt class="py-number">3</tt> <tt class="py-op">*</tt> <tt class="py-name">this_anchor_len</tt> <tt class="py-op">-</tt> <tt class="py-number">2</tt><tt class="py-op">,</tt> <tt class="py-number">3</tt> <tt class="py-op">*</tt> <tt class="py-name">this_anchor_len</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L395"></a><tt class="py-lineno">395</tt>  <tt class="py-line">                        <tt class="py-name">sh_nuc_len</tt> <tt class="py-op">=</tt> <tt class="py-name">this_anchor_len</tt> <tt class="py-op">*</tt> <tt class="py-number">3</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-number">3</tt> <tt class="py-op">*</tt> <tt class="py-name">this_anchor_len</tt> <tt class="py-op">-</tt> <tt class="py-name">shift_val</tt><tt class="py-op">)</tt> </tt>
<a name="L396"></a><tt class="py-lineno">396</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">nucl_seq</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-name">anchor_pos</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">&gt;=</tt> <tt class="py-name">sh_nuc_len</tt><tt class="py-op">:</tt> </tt>
<a name="L397"></a><tt class="py-lineno">397</tt>  <tt class="py-line">                        <tt class="py-name">sh_nuc</tt> <tt class="py-op">=</tt> <tt class="py-name">nucl_seq</tt><tt class="py-op">[</tt><tt class="py-name">anchor_pos</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">:</tt><tt class="py-name">anchor_pos</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">sh_nuc_len</tt><tt class="py-op">]</tt> </tt>
<a name="L398"></a><tt class="py-lineno">398</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L399"></a><tt class="py-lineno">399</tt>  <tt class="py-line">                        <tt class="py-comment"># this is unlikely to produce the correct output</tt> </tt>
<a name="L400"></a><tt class="py-lineno">400</tt>  <tt class="py-line">                        <tt class="py-name">sh_nuc</tt> <tt class="py-op">=</tt> <tt class="py-name">nucl_seq</tt><tt class="py-op">[</tt><tt class="py-name">anchor_pos</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L401"></a><tt class="py-lineno">401</tt>  <tt class="py-line">                    <tt class="py-name">qcodon</tt><tt class="py-op">,</tt> <tt class="py-name">shift_id_pos</tt> <tt class="py-op">=</tt> <tt id="link-178" class="py-name"><a title="Bio.codonalign._get_shift_anchor_re" class="py-name" href="#" onclick="return doclink('link-178', '_get_shift_anchor_re', 'link-160');">_get_shift_anchor_re</a></tt><tt class="py-op">(</tt><tt class="py-name">sh_anc</tt><tt class="py-op">,</tt> <tt class="py-name">sh_nuc</tt><tt class="py-op">,</tt> </tt>
<a name="L402"></a><tt class="py-lineno">402</tt>  <tt class="py-line">                                                                <tt class="py-name">shift_val</tt><tt class="py-op">,</tt> </tt>
<a name="L403"></a><tt class="py-lineno">403</tt>  <tt class="py-line">                                                                <tt class="py-name">aa2re</tt><tt class="py-op">,</tt> </tt>
<a name="L404"></a><tt class="py-lineno">404</tt>  <tt class="py-line">                                                                <tt class="py-name">this_anchor_len</tt><tt class="py-op">,</tt> </tt>
<a name="L405"></a><tt class="py-lineno">405</tt>  <tt class="py-line">                                                                <tt class="py-name">shift_id_pos</tt><tt class="py-op">)</tt> </tt>
<a name="L406"></a><tt class="py-lineno">406</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">qcodon</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> <tt class="py-name">qcodon</tt> <tt class="py-op">!=</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L407"></a><tt class="py-lineno">407</tt>  <tt class="py-line">                        <tt class="py-name">pro_re</tt><tt class="py-op">.</tt><tt id="link-179" class="py-name" targets="Method Bio.File._IndexedSeqFileDict.pop()=Bio.File._IndexedSeqFileDict-class.html#pop,Method Bio.SearchIO._model.hit.Hit.pop()=Bio.SearchIO._model.hit.Hit-class.html#pop,Method Bio.SearchIO._model.query.QueryResult.pop()=Bio.SearchIO._model.query.QueryResult-class.html#pop,Method Bio.Seq.MutableSeq.pop()=Bio.Seq.MutableSeq-class.html#pop"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop" class="py-name" href="#" onclick="return doclink('link-179', 'pop', 'link-179');">pop</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L408"></a><tt class="py-lineno">408</tt>  <tt class="py-line">                        <tt class="py-name">pro_re</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">qcodon</tt> </tt>
<a name="L409"></a><tt class="py-lineno">409</tt>  <tt class="py-line">                        <tt class="py-keyword">break</tt> </tt>
<a name="L410"></a><tt class="py-lineno">410</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">qcodon</tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L411"></a><tt class="py-lineno">411</tt>  <tt class="py-line">                    <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"last frameshift detection failed for "</tt> </tt>
<a name="L412"></a><tt class="py-lineno">412</tt>  <tt class="py-line">                                  <tt class="py-string">"{0}"</tt><tt class="py-op">.</tt><tt id="link-180" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-180', 'format', 'link-51');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">nucl</tt><tt class="py-op">.</tt><tt id="link-181" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-181', 'id', 'link-57');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-182" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-182', 'BiopythonWarning', 'link-1');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L413"></a><tt class="py-lineno">413</tt>  <tt class="py-line">            <tt class="py-comment"># try global match</tt> </tt>
<a name="L414"></a><tt class="py-lineno">414</tt>  <tt class="py-line">            <tt class="py-name">full_pro_re</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">pro_re</tt><tt class="py-op">)</tt> </tt>
<a name="L415"></a><tt class="py-lineno">415</tt>  <tt class="py-line">            <tt id="link-183" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-183', 'match', 'link-127');">match</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-184" class="py-name"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-184', 'search', 'link-128');">search</a></tt><tt class="py-op">(</tt><tt class="py-name">full_pro_re</tt><tt class="py-op">,</tt> <tt class="py-name">nucl_seq</tt><tt class="py-op">)</tt> </tt>
<a name="L416"></a><tt class="py-lineno">416</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-185" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-185', 'match', 'link-127');">match</a></tt><tt class="py-op">:</tt> </tt>
<a name="L417"></a><tt class="py-lineno">417</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt id="link-186" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-186', 'match', 'link-127');">match</a></tt><tt class="py-op">.</tt><tt class="py-name">span</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">2</tt><tt class="py-op">,</tt> <tt id="link-187" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-187', 'match', 'link-127');">match</a></tt><tt class="py-op">)</tt> </tt>
<a name="L418"></a><tt class="py-lineno">418</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L419"></a><tt class="py-lineno">419</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">RuntimeError</tt><tt class="py-op">(</tt><tt class="py-string">"Protein SeqRecord ({0}) and Nucleotide "</tt> </tt>
<a name="L420"></a><tt class="py-lineno">420</tt>  <tt class="py-line">                                   <tt class="py-string">"SeqRecord ({1}) do not "</tt> </tt>
<a name="L421"></a><tt class="py-lineno">421</tt>  <tt class="py-line">                                   <tt class="py-string">"match!"</tt><tt class="py-op">.</tt><tt id="link-188" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-188', 'format', 'link-51');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">pro</tt><tt class="py-op">.</tt><tt id="link-189" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-189', 'id', 'link-57');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">nucl</tt><tt class="py-op">.</tt><tt id="link-190" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-190', 'id', 'link-57');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L422"></a><tt class="py-lineno">422</tt>  <tt class="py-line"> </tt>
<a name="L423"></a><tt class="py-lineno">423</tt>  <tt class="py-line"> </tt>
<a name="_get_shift_anchor_re"></a><div id="_get_shift_anchor_re-def"><a name="L424"></a><tt class="py-lineno">424</tt> <a class="py-toggle" href="#" id="_get_shift_anchor_re-toggle" onclick="return toggle('_get_shift_anchor_re');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.codonalign-module.html#_get_shift_anchor_re">_get_shift_anchor_re</a><tt class="py-op">(</tt><tt class="py-param">sh_anc</tt><tt class="py-op">,</tt> <tt class="py-param">sh_nuc</tt><tt class="py-op">,</tt> <tt class="py-param">shift_val</tt><tt class="py-op">,</tt> <tt class="py-param">aa2re</tt><tt class="py-op">,</tt> <tt class="py-param">anchor_len</tt><tt class="py-op">,</tt> </tt>
<a name="L425"></a><tt class="py-lineno">425</tt>  <tt class="py-line">                         <tt class="py-param">shift_id_pos</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_get_shift_anchor_re-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_get_shift_anchor_re-expanded"><a name="L426"></a><tt class="py-lineno">426</tt>  <tt class="py-line">    <tt class="py-docstring">"""Find a regular expression matching a potentially shifted anchor (PRIVATE).</tt> </tt>
<a name="L427"></a><tt class="py-lineno">427</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L428"></a><tt class="py-lineno">428</tt>  <tt class="py-line"><tt class="py-docstring">    Arguments:</tt> </tt>
<a name="L429"></a><tt class="py-lineno">429</tt>  <tt class="py-line"><tt class="py-docstring">     - sh_anc    - shifted anchor sequence</tt> </tt>
<a name="L430"></a><tt class="py-lineno">430</tt>  <tt class="py-line"><tt class="py-docstring">     - sh_nuc    - potentially corresponding nucleotide sequence</tt> </tt>
<a name="L431"></a><tt class="py-lineno">431</tt>  <tt class="py-line"><tt class="py-docstring">       of sh_anc</tt> </tt>
<a name="L432"></a><tt class="py-lineno">432</tt>  <tt class="py-line"><tt class="py-docstring">     - shift_val - 1 or 2 indicates forward frame shift, whereas</tt> </tt>
<a name="L433"></a><tt class="py-lineno">433</tt>  <tt class="py-line"><tt class="py-docstring">       3*anchor_len-1 or 3*anchor_len-2 indicates</tt> </tt>
<a name="L434"></a><tt class="py-lineno">434</tt>  <tt class="py-line"><tt class="py-docstring">       backward shift</tt> </tt>
<a name="L435"></a><tt class="py-lineno">435</tt>  <tt class="py-line"><tt class="py-docstring">     - aa2re     - aa to codon re dict</tt> </tt>
<a name="L436"></a><tt class="py-lineno">436</tt>  <tt class="py-line"><tt class="py-docstring">     - anchor_len - length of the anchor</tt> </tt>
<a name="L437"></a><tt class="py-lineno">437</tt>  <tt class="py-line"><tt class="py-docstring">     - shift_id_pos - specify current shift name we are at</tt> </tt>
<a name="L438"></a><tt class="py-lineno">438</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L439"></a><tt class="py-lineno">439</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L440"></a><tt class="py-lineno">440</tt>  <tt class="py-line">    <tt class="py-keyword">import</tt> <tt class="py-name">re</tt> </tt>
<a name="L441"></a><tt class="py-lineno">441</tt>  <tt class="py-line">    <tt class="py-name">shift_id</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt id="link-191" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-191', 'i', 'link-56');">i</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-192" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-192', 'i', 'link-56');">i</a></tt> <tt class="py-keyword">in</tt> <tt id="link-193" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-193', 'range', 'link-135');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">97</tt><tt class="py-op">,</tt> <tt class="py-number">107</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L442"></a><tt class="py-lineno">442</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-number">0</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">shift_val</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">3</tt> <tt class="py-op">*</tt> <tt class="py-name">anchor_len</tt> <tt class="py-op">-</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L443"></a><tt class="py-lineno">443</tt>  <tt class="py-line">        <tt class="py-comment"># if shift_val in (1, 2):</tt> </tt>
<a name="L444"></a><tt class="py-lineno">444</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">j</tt> <tt class="py-keyword">in</tt> <tt id="link-194" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-194', 'range', 'link-135');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">sh_anc</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L445"></a><tt class="py-lineno">445</tt>  <tt class="py-line">            <tt class="py-name">qcodon</tt> <tt class="py-op">=</tt> <tt class="py-string">"^"</tt> </tt>
<a name="L446"></a><tt class="py-lineno">446</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">aa</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt class="py-name">sh_anc</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L447"></a><tt class="py-lineno">447</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">k</tt> <tt class="py-op">==</tt> <tt class="py-name">j</tt><tt class="py-op">:</tt> </tt>
<a name="L448"></a><tt class="py-lineno">448</tt>  <tt class="py-line">                    <tt class="py-name">qcodon</tt> <tt class="py-op">+=</tt> <tt class="py-name">aa2re</tt><tt class="py-op">[</tt><tt class="py-name">aa</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">"(?P&lt;"</tt> <tt class="py-op">+</tt> <tt class="py-name">shift_id</tt><tt class="py-op">[</tt><tt class="py-name">shift_id_pos</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">"&gt;..*)"</tt> </tt>
<a name="L449"></a><tt class="py-lineno">449</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L450"></a><tt class="py-lineno">450</tt>  <tt class="py-line">                    <tt class="py-name">qcodon</tt> <tt class="py-op">+=</tt> <tt class="py-name">aa2re</tt><tt class="py-op">[</tt><tt class="py-name">aa</tt><tt class="py-op">]</tt> </tt>
<a name="L451"></a><tt class="py-lineno">451</tt>  <tt class="py-line">            <tt class="py-name">qcodon</tt> <tt class="py-op">+=</tt> <tt class="py-string">"$"</tt> </tt>
<a name="L452"></a><tt class="py-lineno">452</tt>  <tt class="py-line">            <tt id="link-195" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-195', 'match', 'link-127');">match</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-196" class="py-name"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-196', 'search', 'link-128');">search</a></tt><tt class="py-op">(</tt><tt class="py-name">qcodon</tt><tt class="py-op">,</tt> <tt class="py-name">sh_nuc</tt><tt class="py-op">)</tt> </tt>
<a name="L453"></a><tt class="py-lineno">453</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-197" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-197', 'match', 'link-127');">match</a></tt><tt class="py-op">:</tt> </tt>
<a name="L454"></a><tt class="py-lineno">454</tt>  <tt class="py-line">                <tt class="py-name">qcodon</tt> <tt class="py-op">=</tt> <tt class="py-name">qcodon</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'^'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'$'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L455"></a><tt class="py-lineno">455</tt>  <tt class="py-line">                <tt class="py-name">shift_id_pos</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L456"></a><tt class="py-lineno">456</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">qcodon</tt><tt class="py-op">,</tt> <tt class="py-name">shift_id_pos</tt> </tt>
<a name="L457"></a><tt class="py-lineno">457</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-198" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-198', 'match', 'link-127');">match</a></tt><tt class="py-op">:</tt> </tt>
<a name="L458"></a><tt class="py-lineno">458</tt>  <tt class="py-line">            <tt class="py-comment"># failed to find a match (frameshift)</tt> </tt>
<a name="L459"></a><tt class="py-lineno">459</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">shift_id_pos</tt> </tt>
<a name="L460"></a><tt class="py-lineno">460</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">shift_val</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-number">3</tt> <tt class="py-op">*</tt> <tt class="py-name">anchor_len</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">3</tt> <tt class="py-op">*</tt> <tt class="py-name">anchor_len</tt> <tt class="py-op">-</tt> <tt class="py-number">2</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L461"></a><tt class="py-lineno">461</tt>  <tt class="py-line">        <tt class="py-name">shift_val</tt> <tt class="py-op">=</tt> <tt class="py-number">3</tt> <tt class="py-op">*</tt> <tt class="py-name">anchor_len</tt> <tt class="py-op">-</tt> <tt class="py-name">shift_val</tt> </tt>
<a name="L462"></a><tt class="py-lineno">462</tt>  <tt class="py-line">        <tt class="py-comment"># obtain shifted anchor and corresponding nucl</tt> </tt>
<a name="L463"></a><tt class="py-lineno">463</tt>  <tt class="py-line">        <tt class="py-comment"># first check if the shifted pos is just at the end of the</tt> </tt>
<a name="L464"></a><tt class="py-lineno">464</tt>  <tt class="py-line">        <tt class="py-comment"># previous anchor.</tt> </tt>
<a name="L465"></a><tt class="py-lineno">465</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">j</tt> <tt class="py-keyword">in</tt> <tt id="link-199" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-199', 'range', 'link-135');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">sh_anc</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L466"></a><tt class="py-lineno">466</tt>  <tt class="py-line">            <tt class="py-name">qcodon</tt> <tt class="py-op">=</tt> <tt class="py-string">"^"</tt> </tt>
<a name="L467"></a><tt class="py-lineno">467</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">aa</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt class="py-name">sh_anc</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L468"></a><tt class="py-lineno">468</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">k</tt> <tt class="py-op">==</tt> <tt class="py-name">j</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L469"></a><tt class="py-lineno">469</tt>  <tt class="py-line">                    <tt class="py-comment"># will be considered in the next step</tt> </tt>
<a name="L470"></a><tt class="py-lineno">470</tt>  <tt class="py-line">                    <tt class="py-keyword">pass</tt> </tt>
<a name="L471"></a><tt class="py-lineno">471</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">k</tt> <tt class="py-op">==</tt> <tt class="py-name">j</tt><tt class="py-op">:</tt> </tt>
<a name="L472"></a><tt class="py-lineno">472</tt>  <tt class="py-line">                    <tt class="py-name">qcodon</tt> <tt class="py-op">+=</tt> <tt id="link-200" class="py-name" targets="Function Bio.codonalign._merge_aa2re()=Bio.codonalign-module.html#_merge_aa2re"><a title="Bio.codonalign._merge_aa2re" class="py-name" href="#" onclick="return doclink('link-200', '_merge_aa2re', 'link-200');">_merge_aa2re</a></tt><tt class="py-op">(</tt> </tt>
<a name="L473"></a><tt class="py-lineno">473</tt>  <tt class="py-line">                            <tt class="py-name">sh_anc</tt><tt class="py-op">[</tt><tt class="py-name">j</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">sh_anc</tt><tt class="py-op">[</tt><tt class="py-name">j</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">shift_val</tt><tt class="py-op">,</tt> <tt class="py-name">aa2re</tt><tt class="py-op">,</tt> </tt>
<a name="L474"></a><tt class="py-lineno">474</tt>  <tt class="py-line">                            <tt class="py-name">shift_id</tt><tt class="py-op">[</tt><tt class="py-name">shift_id_pos</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-201" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-201', 'upper', 'link-125');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L475"></a><tt class="py-lineno">475</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L476"></a><tt class="py-lineno">476</tt>  <tt class="py-line">                    <tt class="py-name">qcodon</tt> <tt class="py-op">+=</tt> <tt class="py-name">aa2re</tt><tt class="py-op">[</tt><tt class="py-name">aa</tt><tt class="py-op">]</tt> </tt>
<a name="L477"></a><tt class="py-lineno">477</tt>  <tt class="py-line">            <tt class="py-name">qcodon</tt> <tt class="py-op">+=</tt> <tt class="py-string">'$'</tt> </tt>
<a name="L478"></a><tt class="py-lineno">478</tt>  <tt class="py-line">            <tt id="link-202" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-202', 'match', 'link-127');">match</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-203" class="py-name"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-203', 'search', 'link-128');">search</a></tt><tt class="py-op">(</tt><tt class="py-name">qcodon</tt><tt class="py-op">,</tt> <tt class="py-name">sh_nuc</tt><tt class="py-op">)</tt> </tt>
<a name="L479"></a><tt class="py-lineno">479</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-204" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-204', 'match', 'link-127');">match</a></tt><tt class="py-op">:</tt> </tt>
<a name="L480"></a><tt class="py-lineno">480</tt>  <tt class="py-line">                <tt class="py-name">qcodon</tt> <tt class="py-op">=</tt> <tt class="py-name">qcodon</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'^'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'$'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L481"></a><tt class="py-lineno">481</tt>  <tt class="py-line">                <tt class="py-name">shift_id_pos</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L482"></a><tt class="py-lineno">482</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">qcodon</tt><tt class="py-op">,</tt> <tt class="py-name">shift_id_pos</tt> </tt>
<a name="L483"></a><tt class="py-lineno">483</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-205" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-205', 'match', 'link-127');">match</a></tt><tt class="py-op">:</tt> </tt>
<a name="L484"></a><tt class="py-lineno">484</tt>  <tt class="py-line">            <tt class="py-comment"># failed to find a match (frameshift)</tt> </tt>
<a name="L485"></a><tt class="py-lineno">485</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">shift_id_pos</tt> </tt>
</div><a name="L486"></a><tt class="py-lineno">486</tt>  <tt class="py-line"> </tt>
<a name="L487"></a><tt class="py-lineno">487</tt>  <tt class="py-line"> </tt>
<a name="_merge_aa2re"></a><div id="_merge_aa2re-def"><a name="L488"></a><tt class="py-lineno">488</tt> <a class="py-toggle" href="#" id="_merge_aa2re-toggle" onclick="return toggle('_merge_aa2re');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.codonalign-module.html#_merge_aa2re">_merge_aa2re</a><tt class="py-op">(</tt><tt class="py-param">aa1</tt><tt class="py-op">,</tt> <tt class="py-param">aa2</tt><tt class="py-op">,</tt> <tt class="py-param">shift_val</tt><tt class="py-op">,</tt> <tt class="py-param">aa2re</tt><tt class="py-op">,</tt> <tt class="py-param">reid</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_merge_aa2re-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_merge_aa2re-expanded"><a name="L489"></a><tt class="py-lineno">489</tt>  <tt class="py-line">    <tt class="py-docstring">"""Merge two amino acids based on detected frame shift value (PRIVATE)."""</tt> </tt>
<a name="L490"></a><tt class="py-lineno">490</tt>  <tt class="py-line">    <tt class="py-keyword">def</tt> <tt class="py-def-name">get_aa_from_codonre</tt><tt class="py-op">(</tt><tt class="py-param">re_aa</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L491"></a><tt class="py-lineno">491</tt>  <tt class="py-line">        <tt class="py-name">aas</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L492"></a><tt class="py-lineno">492</tt>  <tt class="py-line">        <tt class="py-name">m</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L493"></a><tt class="py-lineno">493</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-206" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-206', 'i', 'link-56');">i</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">re_aa</tt><tt class="py-op">:</tt> </tt>
<a name="L494"></a><tt class="py-lineno">494</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-207" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-207', 'i', 'link-56');">i</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'['</tt><tt class="py-op">:</tt> </tt>
<a name="L495"></a><tt class="py-lineno">495</tt>  <tt class="py-line">                <tt class="py-name">m</tt> <tt class="py-op">=</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt> </tt>
<a name="L496"></a><tt class="py-lineno">496</tt>  <tt class="py-line">                <tt class="py-name">aas</tt><tt class="py-op">.</tt><tt id="link-208" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-208', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L497"></a><tt class="py-lineno">497</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-209" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-209', 'i', 'link-56');">i</a></tt> <tt class="py-op">==</tt> <tt class="py-string">']'</tt><tt class="py-op">:</tt> </tt>
<a name="L498"></a><tt class="py-lineno">498</tt>  <tt class="py-line">                <tt class="py-name">m</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L499"></a><tt class="py-lineno">499</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L500"></a><tt class="py-lineno">500</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">m</tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L501"></a><tt class="py-lineno">501</tt>  <tt class="py-line">                <tt class="py-name">aas</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">aas</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt id="link-210" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-210', 'i', 'link-56');">i</a></tt> </tt>
<a name="L502"></a><tt class="py-lineno">502</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">m</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L503"></a><tt class="py-lineno">503</tt>  <tt class="py-line">                <tt class="py-name">aas</tt><tt class="py-op">.</tt><tt id="link-211" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-211', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt id="link-212" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-212', 'i', 'link-56');">i</a></tt><tt class="py-op">)</tt> </tt>
<a name="L504"></a><tt class="py-lineno">504</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">aas</tt> </tt>
</div><a name="L505"></a><tt class="py-lineno">505</tt>  <tt class="py-line">    <tt class="py-name">scodon</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-213" class="py-name" targets="Method Bio.SearchIO._model.hit.Hit.map()=Bio.SearchIO._model.hit.Hit-class.html#map"><a title="Bio.SearchIO._model.hit.Hit.map" class="py-name" href="#" onclick="return doclink('link-213', 'map', 'link-213');">map</a></tt><tt class="py-op">(</tt><tt class="py-name">get_aa_from_codonre</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">aa2re</tt><tt class="py-op">[</tt><tt id="link-214" class="py-name" targets="Variable Bio.PDB.Polypeptide.aa1=Bio.PDB.Polypeptide-module.html#aa1"><a title="Bio.PDB.Polypeptide.aa1" class="py-name" href="#" onclick="return doclink('link-214', 'aa1', 'link-214');">aa1</a></tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">aa2re</tt><tt class="py-op">[</tt><tt class="py-name">aa2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L506"></a><tt class="py-lineno">506</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">shift_val</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L507"></a><tt class="py-lineno">507</tt>  <tt class="py-line">        <tt class="py-name">intersect</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt id="link-215" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-215', 'set', 'link-61');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">scodon</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">&amp;</tt> <tt id="link-216" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-216', 'set', 'link-61');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">scodon</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L508"></a><tt class="py-lineno">508</tt>  <tt class="py-line">        <tt class="py-name">scodonre</tt> <tt class="py-op">=</tt> <tt class="py-string">'(?P&lt;'</tt> <tt class="py-op">+</tt> <tt class="py-name">reid</tt> <tt class="py-op">+</tt> <tt class="py-string">'&gt;'</tt> </tt>
<a name="L509"></a><tt class="py-lineno">509</tt>  <tt class="py-line">        <tt class="py-name">scodonre</tt> <tt class="py-op">+=</tt> <tt class="py-string">'['</tt> <tt class="py-op">+</tt> <tt class="py-name">scodon</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">']'</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L510"></a><tt class="py-lineno">510</tt>  <tt class="py-line">                    <tt class="py-string">'['</tt> <tt class="py-op">+</tt> <tt class="py-name">scodon</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">']'</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L511"></a><tt class="py-lineno">511</tt>  <tt class="py-line">                    <tt class="py-string">'['</tt> <tt class="py-op">+</tt> <tt class="py-name">intersect</tt> <tt class="py-op">+</tt> <tt class="py-string">']'</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L512"></a><tt class="py-lineno">512</tt>  <tt class="py-line">                    <tt class="py-string">'['</tt> <tt class="py-op">+</tt> <tt class="py-name">scodon</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">']'</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L513"></a><tt class="py-lineno">513</tt>  <tt class="py-line">                    <tt class="py-string">'['</tt> <tt class="py-op">+</tt> <tt class="py-name">scodon</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">']'</tt> </tt>
<a name="L514"></a><tt class="py-lineno">514</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">shift_val</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L515"></a><tt class="py-lineno">515</tt>  <tt class="py-line">        <tt class="py-name">intersect1</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt id="link-217" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-217', 'set', 'link-61');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">scodon</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">&amp;</tt> <tt id="link-218" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-218', 'set', 'link-61');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">scodon</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L516"></a><tt class="py-lineno">516</tt>  <tt class="py-line">        <tt class="py-name">intersect2</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt id="link-219" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-219', 'set', 'link-61');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">scodon</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">&amp;</tt> <tt id="link-220" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-220', 'set', 'link-61');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">scodon</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L517"></a><tt class="py-lineno">517</tt>  <tt class="py-line">        <tt class="py-name">scodonre</tt> <tt class="py-op">=</tt> <tt class="py-string">'(?P&lt;'</tt> <tt class="py-op">+</tt> <tt class="py-name">reid</tt> <tt class="py-op">+</tt> <tt class="py-string">'&gt;'</tt> </tt>
<a name="L518"></a><tt class="py-lineno">518</tt>  <tt class="py-line">        <tt class="py-name">scodonre</tt> <tt class="py-op">+=</tt> <tt class="py-string">'['</tt> <tt class="py-op">+</tt> <tt class="py-name">scodon</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">']'</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L519"></a><tt class="py-lineno">519</tt>  <tt class="py-line">                    <tt class="py-string">'['</tt> <tt class="py-op">+</tt> <tt class="py-name">intersect1</tt> <tt class="py-op">+</tt> <tt class="py-string">']'</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L520"></a><tt class="py-lineno">520</tt>  <tt class="py-line">                    <tt class="py-string">'['</tt> <tt class="py-op">+</tt> <tt class="py-name">intersect2</tt> <tt class="py-op">+</tt> <tt class="py-string">']'</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L521"></a><tt class="py-lineno">521</tt>  <tt class="py-line">                    <tt class="py-string">'['</tt> <tt class="py-op">+</tt> <tt class="py-name">scodon</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">']'</tt> </tt>
<a name="L522"></a><tt class="py-lineno">522</tt>  <tt class="py-line">    <tt class="py-name">scodonre</tt> <tt class="py-op">+=</tt> <tt class="py-string">')'</tt> </tt>
<a name="L523"></a><tt class="py-lineno">523</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">scodonre</tt> </tt>
</div><a name="L524"></a><tt class="py-lineno">524</tt>  <tt class="py-line"> </tt>
<a name="L525"></a><tt class="py-lineno">525</tt>  <tt class="py-line"> </tt>
<a name="_get_codon_rec"></a><div id="_get_codon_rec-def"><a name="L526"></a><tt class="py-lineno">526</tt> <a class="py-toggle" href="#" id="_get_codon_rec-toggle" onclick="return toggle('_get_codon_rec');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.codonalign-module.html#_get_codon_rec">_get_codon_rec</a><tt class="py-op">(</tt><tt class="py-param">pro</tt><tt class="py-op">,</tt> <tt class="py-param">nucl</tt><tt class="py-op">,</tt> <tt class="py-param">span_mode</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">,</tt> <tt class="py-param">gap_char</tt><tt class="py-op">=</tt><tt class="py-string">"-"</tt><tt class="py-op">,</tt> </tt>
<a name="L527"></a><tt class="py-lineno">527</tt>  <tt class="py-line">                   <tt class="py-param">codon_table</tt><tt class="py-op">=</tt><tt id="link-221" class="py-name"><a title="Bio.codonalign.codonalphabet.default_codon_table" class="py-name" href="#" onclick="return doclink('link-221', 'default_codon_table', 'link-28');">default_codon_table</a></tt><tt class="py-op">,</tt> <tt class="py-param">complete_protein</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> </tt>
<a name="L528"></a><tt class="py-lineno">528</tt>  <tt class="py-line">                   <tt class="py-param">max_score</tt><tt class="py-op">=</tt><tt class="py-number">10</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_get_codon_rec-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_get_codon_rec-expanded"><a name="L529"></a><tt class="py-lineno">529</tt>  <tt class="py-line">    <tt class="py-docstring">"""Generate codon alignment based on regular re match (PRIVATE).</tt> </tt>
<a name="L530"></a><tt class="py-lineno">530</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L531"></a><tt class="py-lineno">531</tt>  <tt class="py-line"><tt class="py-docstring">    span_mode is a tuple returned by _check_corr. The first element</tt> </tt>
<a name="L532"></a><tt class="py-lineno">532</tt>  <tt class="py-line"><tt class="py-docstring">    is the span of a re search, and the second element is the mode</tt> </tt>
<a name="L533"></a><tt class="py-lineno">533</tt>  <tt class="py-line"><tt class="py-docstring">    for the match.</tt> </tt>
<a name="L534"></a><tt class="py-lineno">534</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L535"></a><tt class="py-lineno">535</tt>  <tt class="py-line"><tt class="py-docstring">    mode</tt> </tt>
<a name="L536"></a><tt class="py-lineno">536</tt>  <tt class="py-line"><tt class="py-docstring">     - 0: direct match</tt> </tt>
<a name="L537"></a><tt class="py-lineno">537</tt>  <tt class="py-line"><tt class="py-docstring">     - 1: mismatch (no indels)</tt> </tt>
<a name="L538"></a><tt class="py-lineno">538</tt>  <tt class="py-line"><tt class="py-docstring">     - 2: frameshift</tt> </tt>
<a name="L539"></a><tt class="py-lineno">539</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L540"></a><tt class="py-lineno">540</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L541"></a><tt class="py-lineno">541</tt>  <tt class="py-line">    <tt class="py-keyword">import</tt> <tt class="py-name">re</tt> </tt>
<a name="L542"></a><tt class="py-lineno">542</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-222" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-222', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-223" class="py-name" targets="Module Bio.Seq=Bio.Seq-module.html,Class Bio.Seq.Seq=Bio.Seq.Seq-class.html"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-223', 'Seq', 'link-223');">Seq</a></tt> <tt class="py-keyword">import</tt> <tt id="link-224" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-224', 'Seq', 'link-223');">Seq</a></tt> </tt>
<a name="L543"></a><tt class="py-lineno">543</tt>  <tt class="py-line"> </tt>
<a name="L544"></a><tt class="py-lineno">544</tt>  <tt class="py-line">    <tt class="py-name">nucl_seq</tt> <tt class="py-op">=</tt> <tt class="py-name">nucl</tt><tt class="py-op">.</tt><tt id="link-225" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-225', 'seq', 'link-44');">seq</a></tt><tt class="py-op">.</tt><tt id="link-226" class="py-name"><a title="Bio.Seq.Seq.ungap
Bio.Seq.UnknownSeq.ungap
Bio.codonalign.codonseq.CodonSeq.ungap" class="py-name" href="#" onclick="return doclink('link-226', 'ungap', 'link-126');">ungap</a></tt><tt class="py-op">(</tt><tt class="py-name">gap_char</tt><tt class="py-op">)</tt> </tt>
<a name="L545"></a><tt class="py-lineno">545</tt>  <tt class="py-line">    <tt class="py-name">codon_seq</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L546"></a><tt class="py-lineno">546</tt>  <tt class="py-line">    <tt class="py-name">span</tt> <tt class="py-op">=</tt> <tt class="py-name">span_mode</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L547"></a><tt class="py-lineno">547</tt>  <tt class="py-line">    <tt class="py-name">mode</tt> <tt class="py-op">=</tt> <tt class="py-name">span_mode</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L548"></a><tt class="py-lineno">548</tt>  <tt class="py-line">    <tt class="py-name">aa2re</tt> <tt class="py-op">=</tt> <tt id="link-227" class="py-name"><a title="Bio.codonalign._get_aa_regex" class="py-name" href="#" onclick="return doclink('link-227', '_get_aa_regex', 'link-122');">_get_aa_regex</a></tt><tt class="py-op">(</tt><tt class="py-name">codon_table</tt><tt class="py-op">)</tt> </tt>
<a name="L549"></a><tt class="py-lineno">549</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">mode</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L550"></a><tt class="py-lineno">550</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">pro</tt><tt class="py-op">.</tt><tt id="link-228" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-228', 'seq', 'link-44');">seq</a></tt><tt class="py-op">.</tt><tt id="link-229" class="py-name"><a title="Bio.Seq.Seq.ungap
Bio.Seq.UnknownSeq.ungap
Bio.codonalign.codonseq.CodonSeq.ungap" class="py-name" href="#" onclick="return doclink('link-229', 'ungap', 'link-126');">ungap</a></tt><tt class="py-op">(</tt><tt class="py-name">gap_char</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-number">3</tt> <tt class="py-op">!=</tt> <tt class="py-op">(</tt><tt class="py-name">span</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">-</tt> <tt class="py-name">span</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L551"></a><tt class="py-lineno">551</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Protein Record {0} and Nucleotide Record {1} "</tt> </tt>
<a name="L552"></a><tt class="py-lineno">552</tt>  <tt class="py-line">                             <tt class="py-string">"do not match!"</tt><tt class="py-op">.</tt><tt id="link-230" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-230', 'format', 'link-51');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">pro</tt><tt class="py-op">.</tt><tt id="link-231" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-231', 'id', 'link-57');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">nucl</tt><tt class="py-op">.</tt><tt id="link-232" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-232', 'id', 'link-57');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L553"></a><tt class="py-lineno">553</tt>  <tt class="py-line">        <tt class="py-name">aa_num</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L554"></a><tt class="py-lineno">554</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">aa</tt> <tt class="py-keyword">in</tt> <tt class="py-name">pro</tt><tt class="py-op">.</tt><tt id="link-233" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-233', 'seq', 'link-44');">seq</a></tt><tt class="py-op">:</tt> </tt>
<a name="L555"></a><tt class="py-lineno">555</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">aa</tt> <tt class="py-op">==</tt> <tt class="py-string">"-"</tt><tt class="py-op">:</tt> </tt>
<a name="L556"></a><tt class="py-lineno">556</tt>  <tt class="py-line">                <tt class="py-name">codon_seq</tt> <tt class="py-op">+=</tt> <tt class="py-string">"---"</tt> </tt>
<a name="L557"></a><tt class="py-lineno">557</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">complete_protein</tt> <tt class="py-keyword">and</tt> <tt class="py-name">aa_num</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L558"></a><tt class="py-lineno">558</tt>  <tt class="py-line">                <tt class="py-name">this_codon</tt> <tt class="py-op">=</tt> <tt class="py-name">nucl_seq</tt><tt class="py-op">.</tt><tt class="py-name">_data</tt><tt class="py-op">[</tt><tt class="py-name">span</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">:</tt><tt class="py-name">span</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-number">3</tt><tt class="py-op">]</tt> </tt>
<a name="L559"></a><tt class="py-lineno">559</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-234" class="py-name"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-234', 'search', 'link-128');">search</a></tt><tt class="py-op">(</tt><tt id="link-235" class="py-name"><a title="Bio.codonalign._codons2re" class="py-name" href="#" onclick="return doclink('link-235', '_codons2re', 'link-105');">_codons2re</a></tt><tt class="py-op">[</tt><tt class="py-name">codon_table</tt><tt class="py-op">.</tt><tt id="link-236" class="py-name"><a title="Bio.Data.CodonTable.CodonTable.start_codons" class="py-name" href="#" onclick="return doclink('link-236', 'start_codons', 'link-139');">start_codons</a></tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L560"></a><tt class="py-lineno">560</tt>  <tt class="py-line">                                 <tt class="py-name">this_codon</tt><tt class="py-op">.</tt><tt id="link-237" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-237', 'upper', 'link-125');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L561"></a><tt class="py-lineno">561</tt>  <tt class="py-line">                    <tt class="py-name">max_score</tt> <tt class="py-op">-=</tt> <tt class="py-number">1</tt> </tt>
<a name="L562"></a><tt class="py-lineno">562</tt>  <tt class="py-line">                    <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"start codon of {0} ({1} {2}) does not "</tt> </tt>
<a name="L563"></a><tt class="py-lineno">563</tt>  <tt class="py-line">                                  <tt class="py-string">"correspond to {3} "</tt> </tt>
<a name="L564"></a><tt class="py-lineno">564</tt>  <tt class="py-line">                                  <tt class="py-string">"({4})"</tt><tt class="py-op">.</tt><tt id="link-238" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-238', 'format', 'link-51');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">pro</tt><tt class="py-op">.</tt><tt id="link-239" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-239', 'id', 'link-57');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">aa</tt><tt class="py-op">,</tt> <tt class="py-name">aa_num</tt><tt class="py-op">,</tt> </tt>
<a name="L565"></a><tt class="py-lineno">565</tt>  <tt class="py-line">                                                 <tt class="py-name">nucl</tt><tt class="py-op">.</tt><tt id="link-240" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-240', 'id', 'link-57');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">this_codon</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L566"></a><tt class="py-lineno">566</tt>  <tt class="py-line">                                  <tt id="link-241" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-241', 'BiopythonWarning', 'link-1');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L567"></a><tt class="py-lineno">567</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">max_score</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L568"></a><tt class="py-lineno">568</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">RuntimeError</tt><tt class="py-op">(</tt><tt class="py-string">"max_score reached for {0}! Please "</tt> </tt>
<a name="L569"></a><tt class="py-lineno">569</tt>  <tt class="py-line">                                       <tt class="py-string">"raise up the tolerance to get an "</tt> </tt>
<a name="L570"></a><tt class="py-lineno">570</tt>  <tt class="py-line">                                       <tt class="py-string">"alignment in anyway"</tt><tt class="py-op">.</tt><tt id="link-242" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-242', 'format', 'link-51');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">nucl</tt><tt class="py-op">.</tt><tt id="link-243" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-243', 'id', 'link-57');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L571"></a><tt class="py-lineno">571</tt>  <tt class="py-line">                <tt class="py-name">codon_seq</tt> <tt class="py-op">+=</tt> <tt class="py-name">this_codon</tt> </tt>
<a name="L572"></a><tt class="py-lineno">572</tt>  <tt class="py-line">                <tt class="py-name">aa_num</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L573"></a><tt class="py-lineno">573</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L574"></a><tt class="py-lineno">574</tt>  <tt class="py-line">                <tt class="py-name">this_codon</tt> <tt class="py-op">=</tt> <tt class="py-name">nucl_seq</tt><tt class="py-op">.</tt><tt class="py-name">_data</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">span</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-number">3</tt> <tt class="py-op">*</tt> <tt class="py-name">aa_num</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L575"></a><tt class="py-lineno">575</tt>  <tt class="py-line">                                            <tt class="py-op">(</tt><tt class="py-name">span</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-number">3</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">aa_num</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L576"></a><tt class="py-lineno">576</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-244" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-244', 'Seq', 'link-223');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">this_codon</tt><tt class="py-op">.</tt><tt id="link-245" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-245', 'upper', 'link-125');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-246" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.translate()=Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#translate,Method Bio.Seq.Seq.translate()=Bio.Seq.Seq-class.html#translate,Method Bio.Seq.UnknownSeq.translate()=Bio.Seq.UnknownSeq-class.html#translate,Function Bio.Seq.translate()=Bio.Seq-module.html#translate,Method Bio.SeqFeature.SeqFeature.translate()=Bio.SeqFeature.SeqFeature-class.html#translate,Method Bio.SeqRecord.SeqRecord.translate()=Bio.SeqRecord.SeqRecord-class.html#translate,Method Bio.codonalign.codonseq.CodonSeq.translate()=Bio.codonalign.codonseq.CodonSeq-class.html#translate"><a title="Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.translate
Bio.Seq.Seq.translate
Bio.Seq.UnknownSeq.translate
Bio.Seq.translate
Bio.SeqFeature.SeqFeature.translate
Bio.SeqRecord.SeqRecord.translate
Bio.codonalign.codonseq.CodonSeq.translate" class="py-name" href="#" onclick="return doclink('link-246', 'translate', 'link-246');">translate</a></tt><tt class="py-op">(</tt><tt class="py-name">table</tt><tt class="py-op">=</tt><tt class="py-name">codon_table</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">aa</tt><tt class="py-op">:</tt> </tt>
<a name="L577"></a><tt class="py-lineno">577</tt>  <tt class="py-line">                    <tt class="py-name">max_score</tt> <tt class="py-op">-=</tt> <tt class="py-number">1</tt> </tt>
<a name="L578"></a><tt class="py-lineno">578</tt>  <tt class="py-line">                    <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"%s(%s %d) does not correspond to %s(%s)"</tt> </tt>
<a name="L579"></a><tt class="py-lineno">579</tt>  <tt class="py-line">                                  <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">pro</tt><tt class="py-op">.</tt><tt id="link-247" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-247', 'id', 'link-57');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">aa</tt><tt class="py-op">,</tt> <tt class="py-name">aa_num</tt><tt class="py-op">,</tt> <tt class="py-name">nucl</tt><tt class="py-op">.</tt><tt id="link-248" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-248', 'id', 'link-57');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">this_codon</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L580"></a><tt class="py-lineno">580</tt>  <tt class="py-line">                                  <tt id="link-249" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-249', 'BiopythonWarning', 'link-1');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L581"></a><tt class="py-lineno">581</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">max_score</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L582"></a><tt class="py-lineno">582</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">RuntimeError</tt><tt class="py-op">(</tt><tt class="py-string">"max_score reached for {0}! Please "</tt> </tt>
<a name="L583"></a><tt class="py-lineno">583</tt>  <tt class="py-line">                                       <tt class="py-string">"raise up the tolerance to get an "</tt> </tt>
<a name="L584"></a><tt class="py-lineno">584</tt>  <tt class="py-line">                                       <tt class="py-string">"alignment in anyway"</tt><tt class="py-op">.</tt><tt id="link-250" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-250', 'format', 'link-51');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">nucl</tt><tt class="py-op">.</tt><tt id="link-251" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-251', 'id', 'link-57');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L585"></a><tt class="py-lineno">585</tt>  <tt class="py-line">                <tt class="py-name">codon_seq</tt> <tt class="py-op">+=</tt> <tt class="py-name">this_codon</tt> </tt>
<a name="L586"></a><tt class="py-lineno">586</tt>  <tt class="py-line">                <tt class="py-name">aa_num</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L587"></a><tt class="py-lineno">587</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-252" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-252', 'SeqRecord', 'link-10');">SeqRecord</a></tt><tt class="py-op">(</tt><tt id="link-253" class="py-name"><a title="Bio.codonalign.codonseq.CodonSeq" class="py-name" href="#" onclick="return doclink('link-253', 'CodonSeq', 'link-15');">CodonSeq</a></tt><tt class="py-op">(</tt><tt class="py-name">codon_seq</tt><tt class="py-op">,</tt> <tt id="link-254" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-254', 'alphabet', 'link-45');">alphabet</a></tt><tt class="py-op">=</tt><tt id="link-255" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-255', 'alphabet', 'link-45');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-256" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-256', 'id', 'link-57');">id</a></tt><tt class="py-op">=</tt><tt class="py-name">nucl</tt><tt class="py-op">.</tt><tt id="link-257" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-257', 'id', 'link-57');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L588"></a><tt class="py-lineno">588</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">mode</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L589"></a><tt class="py-lineno">589</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt class="py-name">collections</tt> <tt class="py-keyword">import</tt> <tt class="py-name">deque</tt> </tt>
<a name="L590"></a><tt class="py-lineno">590</tt>  <tt class="py-line">        <tt class="py-name">shift_pos</tt> <tt class="py-op">=</tt> <tt class="py-name">deque</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L591"></a><tt class="py-lineno">591</tt>  <tt class="py-line">        <tt class="py-name">shift_start</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L592"></a><tt class="py-lineno">592</tt>  <tt class="py-line">        <tt id="link-258" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-258', 'match', 'link-127');">match</a></tt> <tt class="py-op">=</tt> <tt class="py-name">span_mode</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> </tt>
<a name="L593"></a><tt class="py-lineno">593</tt>  <tt class="py-line">        <tt class="py-name">m_groupdict</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-259" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-259', 'match', 'link-127');">match</a></tt><tt class="py-op">.</tt><tt class="py-name">groupdict</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-260" class="py-name"><a title="Bio.Crystal.Crystal.keys
Bio.Entrez.Parser.ListConsumer.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-260', 'keys', 'link-62');">keys</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L594"></a><tt class="py-lineno">594</tt>  <tt class="py-line">        <tt class="py-comment"># backward frameshift</tt> </tt>
<a name="L595"></a><tt class="py-lineno">595</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-261" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-261', 'i', 'link-56');">i</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">m_groupdict</tt><tt class="py-op">:</tt> </tt>
<a name="L596"></a><tt class="py-lineno">596</tt>  <tt class="py-line">            <tt class="py-name">shift_pos</tt><tt class="py-op">.</tt><tt id="link-262" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-262', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt id="link-263" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-263', 'match', 'link-127');">match</a></tt><tt class="py-op">.</tt><tt class="py-name">span</tt><tt class="py-op">(</tt><tt id="link-264" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-264', 'i', 'link-56');">i</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L597"></a><tt class="py-lineno">597</tt>  <tt class="py-line">            <tt class="py-name">shift_start</tt><tt class="py-op">.</tt><tt id="link-265" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-265', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt id="link-266" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-266', 'match', 'link-127');">match</a></tt><tt class="py-op">.</tt><tt id="link-267" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-267', 'start', 'link-148');">start</a></tt><tt class="py-op">(</tt><tt id="link-268" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-268', 'i', 'link-56');">i</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L598"></a><tt class="py-lineno">598</tt>  <tt class="py-line">        <tt class="py-name">rf_table</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L599"></a><tt class="py-lineno">599</tt>  <tt class="py-line">        <tt id="link-269" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-269', 'i', 'link-56');">i</a></tt> <tt class="py-op">=</tt> <tt id="link-270" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-270', 'match', 'link-127');">match</a></tt><tt class="py-op">.</tt><tt id="link-271" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-271', 'start', 'link-148');">start</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L600"></a><tt class="py-lineno">600</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L601"></a><tt class="py-lineno">601</tt>  <tt class="py-line">            <tt class="py-name">rf_table</tt><tt class="py-op">.</tt><tt id="link-272" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-272', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt id="link-273" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-273', 'i', 'link-56');">i</a></tt><tt class="py-op">)</tt> </tt>
<a name="L602"></a><tt class="py-lineno">602</tt>  <tt class="py-line">            <tt id="link-274" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-274', 'i', 'link-56');">i</a></tt> <tt class="py-op">+=</tt> <tt class="py-number">3</tt> </tt>
<a name="L603"></a><tt class="py-lineno">603</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-275" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-275', 'i', 'link-56');">i</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">shift_start</tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L604"></a><tt class="py-lineno">604</tt>  <tt class="py-line">                    <tt class="py-name">m_groupdict</tt><tt class="py-op">[</tt><tt class="py-name">shift_start</tt><tt class="py-op">.</tt><tt id="link-276" class="py-name" targets="Method Bio.Crystal.Chain.index()=Bio.Crystal.Chain-class.html#index,Method Bio.SCOP.Raf.SeqMap.index()=Bio.SCOP.Raf.SeqMap-class.html#index,Method Bio.SearchIO._model.hit.Hit.index()=Bio.SearchIO._model.hit.Hit-class.html#index,Method Bio.SearchIO._model.query.QueryResult.index()=Bio.SearchIO._model.query.QueryResult-class.html#index,Function Bio.SearchIO.index()=Bio.SearchIO-module.html#index,Method Bio.Seq.MutableSeq.index()=Bio.Seq.MutableSeq-class.html#index,Function Bio.SeqIO.index()=Bio.SeqIO-module.html#index"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-276', 'index', 'link-276');">index</a></tt><tt class="py-op">(</tt><tt id="link-277" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-277', 'i', 'link-56');">i</a></tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">isupper</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L605"></a><tt class="py-lineno">605</tt>  <tt class="py-line">                <tt class="py-name">shift_index</tt> <tt class="py-op">=</tt> <tt class="py-name">shift_start</tt><tt class="py-op">.</tt><tt id="link-278" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-278', 'index', 'link-276');">index</a></tt><tt class="py-op">(</tt><tt id="link-279" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-279', 'i', 'link-56');">i</a></tt><tt class="py-op">)</tt> </tt>
<a name="L606"></a><tt class="py-lineno">606</tt>  <tt class="py-line">                <tt class="py-name">shift_val</tt> <tt class="py-op">=</tt> <tt class="py-number">6</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">shift_pos</tt><tt class="py-op">[</tt><tt class="py-name">shift_index</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">-</tt> </tt>
<a name="L607"></a><tt class="py-lineno">607</tt>  <tt class="py-line">                                 <tt class="py-name">shift_pos</tt><tt class="py-op">[</tt><tt class="py-name">shift_index</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L608"></a><tt class="py-lineno">608</tt>  <tt class="py-line">                <tt class="py-name">rf_table</tt><tt class="py-op">.</tt><tt id="link-280" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-280', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt id="link-281" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-281', 'i', 'link-56');">i</a></tt><tt class="py-op">)</tt> </tt>
<a name="L609"></a><tt class="py-lineno">609</tt>  <tt class="py-line">                <tt class="py-name">rf_table</tt><tt class="py-op">.</tt><tt id="link-282" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-282', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt id="link-283" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-283', 'i', 'link-56');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-number">3</tt> <tt class="py-op">-</tt> <tt class="py-name">shift_val</tt><tt class="py-op">)</tt> </tt>
<a name="L610"></a><tt class="py-lineno">610</tt>  <tt class="py-line">                <tt id="link-284" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-284', 'i', 'link-56');">i</a></tt> <tt class="py-op">=</tt> <tt class="py-name">shift_pos</tt><tt class="py-op">[</tt><tt class="py-name">shift_index</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L611"></a><tt class="py-lineno">611</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-285" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-285', 'i', 'link-56');">i</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">shift_start</tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L612"></a><tt class="py-lineno">612</tt>  <tt class="py-line">                    <tt class="py-name">m_groupdict</tt><tt class="py-op">[</tt><tt class="py-name">shift_start</tt><tt class="py-op">.</tt><tt id="link-286" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-286', 'index', 'link-276');">index</a></tt><tt class="py-op">(</tt><tt id="link-287" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-287', 'i', 'link-56');">i</a></tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">islower</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L613"></a><tt class="py-lineno">613</tt>  <tt class="py-line">                <tt id="link-288" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-288', 'i', 'link-56');">i</a></tt> <tt class="py-op">=</tt> <tt class="py-name">shift_pos</tt><tt class="py-op">[</tt><tt class="py-name">shift_start</tt><tt class="py-op">.</tt><tt id="link-289" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-289', 'index', 'link-276');">index</a></tt><tt class="py-op">(</tt><tt id="link-290" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-290', 'i', 'link-56');">i</a></tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L614"></a><tt class="py-lineno">614</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-291" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-291', 'i', 'link-56');">i</a></tt> <tt class="py-op">&gt;=</tt> <tt id="link-292" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-292', 'match', 'link-127');">match</a></tt><tt class="py-op">.</tt><tt id="link-293" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-293', 'end', 'link-150');">end</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L615"></a><tt class="py-lineno">615</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L616"></a><tt class="py-lineno">616</tt>  <tt class="py-line">        <tt class="py-name">aa_num</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L617"></a><tt class="py-lineno">617</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">aa</tt> <tt class="py-keyword">in</tt> <tt class="py-name">pro</tt><tt class="py-op">.</tt><tt id="link-294" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-294', 'seq', 'link-44');">seq</a></tt><tt class="py-op">:</tt> </tt>
<a name="L618"></a><tt class="py-lineno">618</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">aa</tt> <tt class="py-op">==</tt> <tt class="py-string">"-"</tt><tt class="py-op">:</tt> </tt>
<a name="L619"></a><tt class="py-lineno">619</tt>  <tt class="py-line">                <tt class="py-name">codon_seq</tt> <tt class="py-op">+=</tt> <tt class="py-string">"---"</tt> </tt>
<a name="L620"></a><tt class="py-lineno">620</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">complete_protein</tt> <tt class="py-keyword">and</tt> <tt class="py-name">aa_num</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L621"></a><tt class="py-lineno">621</tt>  <tt class="py-line">                <tt class="py-name">this_codon</tt> <tt class="py-op">=</tt> <tt class="py-name">nucl_seq</tt><tt class="py-op">.</tt><tt class="py-name">_data</tt><tt class="py-op">[</tt><tt class="py-name">rf_table</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">:</tt><tt class="py-name">rf_table</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-number">3</tt><tt class="py-op">]</tt> </tt>
<a name="L622"></a><tt class="py-lineno">622</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-295" class="py-name"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-295', 'search', 'link-128');">search</a></tt><tt class="py-op">(</tt><tt id="link-296" class="py-name"><a title="Bio.codonalign._codons2re" class="py-name" href="#" onclick="return doclink('link-296', '_codons2re', 'link-105');">_codons2re</a></tt><tt class="py-op">[</tt><tt class="py-name">codon_table</tt><tt class="py-op">.</tt><tt id="link-297" class="py-name"><a title="Bio.Data.CodonTable.CodonTable.start_codons" class="py-name" href="#" onclick="return doclink('link-297', 'start_codons', 'link-139');">start_codons</a></tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L623"></a><tt class="py-lineno">623</tt>  <tt class="py-line">                                 <tt class="py-name">this_codon</tt><tt class="py-op">.</tt><tt id="link-298" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-298', 'upper', 'link-125');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L624"></a><tt class="py-lineno">624</tt>  <tt class="py-line">                    <tt class="py-name">max_score</tt> <tt class="py-op">-=</tt> <tt class="py-number">1</tt> </tt>
<a name="L625"></a><tt class="py-lineno">625</tt>  <tt class="py-line">                    <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"start codon of {0}({1} {2}) does not "</tt> </tt>
<a name="L626"></a><tt class="py-lineno">626</tt>  <tt class="py-line">                                  <tt class="py-string">"correspond to {3}({4})"</tt><tt class="py-op">.</tt><tt id="link-299" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-299', 'format', 'link-51');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L627"></a><tt class="py-lineno">627</tt>  <tt class="py-line">                                      <tt class="py-name">pro</tt><tt class="py-op">.</tt><tt id="link-300" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-300', 'id', 'link-57');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">aa</tt><tt class="py-op">,</tt> <tt class="py-name">aa_num</tt><tt class="py-op">,</tt> <tt class="py-name">nucl</tt><tt class="py-op">.</tt><tt id="link-301" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-301', 'id', 'link-57');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">this_codon</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L628"></a><tt class="py-lineno">628</tt>  <tt class="py-line">                                  <tt id="link-302" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-302', 'BiopythonWarning', 'link-1');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L629"></a><tt class="py-lineno">629</tt>  <tt class="py-line">                    <tt class="py-name">codon_seq</tt> <tt class="py-op">+=</tt> <tt class="py-name">this_codon</tt> </tt>
<a name="L630"></a><tt class="py-lineno">630</tt>  <tt class="py-line">                    <tt class="py-name">aa_num</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L631"></a><tt class="py-lineno">631</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L632"></a><tt class="py-lineno">632</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">aa_num</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">pro</tt><tt class="py-op">.</tt><tt id="link-303" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-303', 'seq', 'link-44');">seq</a></tt><tt class="py-op">.</tt><tt id="link-304" class="py-name"><a title="Bio.Seq.Seq.ungap
Bio.Seq.UnknownSeq.ungap
Bio.codonalign.codonseq.CodonSeq.ungap" class="py-name" href="#" onclick="return doclink('link-304', 'ungap', 'link-126');">ungap</a></tt><tt class="py-op">(</tt><tt class="py-string">'-'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L633"></a><tt class="py-lineno">633</tt>  <tt class="py-line">                        <tt class="py-name">rf_table</tt><tt class="py-op">[</tt><tt class="py-name">aa_num</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">-</tt> <tt class="py-name">rf_table</tt><tt class="py-op">[</tt><tt class="py-name">aa_num</tt><tt class="py-op">]</tt> <tt class="py-op">-</tt> <tt class="py-number">3</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L634"></a><tt class="py-lineno">634</tt>  <tt class="py-line">                    <tt class="py-name">max_score</tt> <tt class="py-op">-=</tt> <tt class="py-number">1</tt> </tt>
<a name="L635"></a><tt class="py-lineno">635</tt>  <tt class="py-line">                    <tt id="link-305" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-305', 'start', 'link-148');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">rf_table</tt><tt class="py-op">[</tt><tt class="py-name">aa_num</tt><tt class="py-op">]</tt> </tt>
<a name="L636"></a><tt class="py-lineno">636</tt>  <tt class="py-line">                    <tt id="link-306" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-306', 'end', 'link-150');">end</a></tt> <tt class="py-op">=</tt> <tt id="link-307" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-307', 'start', 'link-148');">start</a></tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt class="py-number">3</tt> <tt class="py-op">-</tt> <tt class="py-name">shift_val</tt><tt class="py-op">)</tt> </tt>
<a name="L637"></a><tt class="py-lineno">637</tt>  <tt class="py-line">                    <tt class="py-name">ngap</tt> <tt class="py-op">=</tt> <tt class="py-name">shift_val</tt> </tt>
<a name="L638"></a><tt class="py-lineno">638</tt>  <tt class="py-line">                    <tt class="py-name">this_codon</tt> <tt class="py-op">=</tt> <tt class="py-name">nucl_seq</tt><tt class="py-op">.</tt><tt class="py-name">_data</tt><tt class="py-op">[</tt><tt id="link-308" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-308', 'start', 'link-148');">start</a></tt><tt class="py-op">:</tt><tt id="link-309" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-309', 'end', 'link-150');">end</a></tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">'-'</tt> <tt class="py-op">*</tt> <tt class="py-name">ngap</tt> </tt>
<a name="L639"></a><tt class="py-lineno">639</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">rf_table</tt><tt class="py-op">[</tt><tt class="py-name">aa_num</tt><tt class="py-op">]</tt> <tt class="py-op">-</tt> <tt class="py-name">rf_table</tt><tt class="py-op">[</tt><tt class="py-name">aa_num</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">-</tt> <tt class="py-number">3</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L640"></a><tt class="py-lineno">640</tt>  <tt class="py-line">                    <tt class="py-name">max_score</tt> <tt class="py-op">-=</tt> <tt class="py-number">1</tt> </tt>
<a name="L641"></a><tt class="py-lineno">641</tt>  <tt class="py-line">                    <tt id="link-310" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-310', 'start', 'link-148');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">rf_table</tt><tt class="py-op">[</tt><tt class="py-name">aa_num</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-number">3</tt> </tt>
<a name="L642"></a><tt class="py-lineno">642</tt>  <tt class="py-line">                    <tt id="link-311" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-311', 'end', 'link-150');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">rf_table</tt><tt class="py-op">[</tt><tt class="py-name">aa_num</tt><tt class="py-op">]</tt> </tt>
<a name="L643"></a><tt class="py-lineno">643</tt>  <tt class="py-line">                    <tt class="py-name">ngap</tt> <tt class="py-op">=</tt> <tt class="py-number">3</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">rf_table</tt><tt class="py-op">[</tt><tt class="py-name">aa_num</tt><tt class="py-op">]</tt> <tt class="py-op">-</tt> <tt class="py-name">rf_table</tt><tt class="py-op">[</tt><tt class="py-name">aa_num</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">-</tt> <tt class="py-number">3</tt><tt class="py-op">)</tt> </tt>
<a name="L644"></a><tt class="py-lineno">644</tt>  <tt class="py-line">                    <tt class="py-name">this_codon</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">nucl_seq</tt><tt class="py-op">.</tt><tt class="py-name">_data</tt><tt class="py-op">[</tt><tt id="link-312" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-312', 'start', 'link-148');">start</a></tt><tt class="py-op">:</tt><tt id="link-313" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-313', 'end', 'link-150');">end</a></tt><tt class="py-op">]</tt> </tt>
<a name="L645"></a><tt class="py-lineno">645</tt>  <tt class="py-line">                                  <tt class="py-op">+</tt> <tt class="py-string">'-'</tt> <tt class="py-op">*</tt> <tt class="py-name">ngap</tt> </tt>
<a name="L646"></a><tt class="py-lineno">646</tt>  <tt class="py-line">                                  <tt class="py-op">+</tt> <tt class="py-name">nucl_seq</tt><tt class="py-op">.</tt><tt class="py-name">_data</tt><tt class="py-op">[</tt><tt class="py-name">rf_table</tt><tt class="py-op">[</tt><tt class="py-name">aa_num</tt><tt class="py-op">]</tt><tt class="py-op">:</tt><tt class="py-name">rf_table</tt><tt class="py-op">[</tt><tt class="py-name">aa_num</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L647"></a><tt class="py-lineno">647</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L648"></a><tt class="py-lineno">648</tt>  <tt class="py-line">                    <tt id="link-314" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-314', 'start', 'link-148');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">rf_table</tt><tt class="py-op">[</tt><tt class="py-name">aa_num</tt><tt class="py-op">]</tt> </tt>
<a name="L649"></a><tt class="py-lineno">649</tt>  <tt class="py-line">                    <tt id="link-315" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-315', 'end', 'link-150');">end</a></tt> <tt class="py-op">=</tt> <tt id="link-316" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-316', 'start', 'link-148');">start</a></tt> <tt class="py-op">+</tt> <tt class="py-number">3</tt> </tt>
<a name="L650"></a><tt class="py-lineno">650</tt>  <tt class="py-line">                    <tt class="py-name">this_codon</tt> <tt class="py-op">=</tt> <tt class="py-name">nucl_seq</tt><tt class="py-op">.</tt><tt class="py-name">_data</tt><tt class="py-op">[</tt><tt id="link-317" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-317', 'start', 'link-148');">start</a></tt><tt class="py-op">:</tt><tt id="link-318" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-318', 'end', 'link-150');">end</a></tt><tt class="py-op">]</tt> </tt>
<a name="L651"></a><tt class="py-lineno">651</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-319" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-319', 'Seq', 'link-223');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">this_codon</tt><tt class="py-op">.</tt><tt id="link-320" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-320', 'upper', 'link-125');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-321" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.translate
Bio.Seq.Seq.translate
Bio.Seq.UnknownSeq.translate
Bio.Seq.translate
Bio.SeqFeature.SeqFeature.translate
Bio.SeqRecord.SeqRecord.translate
Bio.codonalign.codonseq.CodonSeq.translate" class="py-name" href="#" onclick="return doclink('link-321', 'translate', 'link-246');">translate</a></tt><tt class="py-op">(</tt><tt class="py-name">table</tt><tt class="py-op">=</tt><tt class="py-name">codon_table</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">aa</tt><tt class="py-op">:</tt> </tt>
<a name="L652"></a><tt class="py-lineno">652</tt>  <tt class="py-line">                        <tt class="py-name">max_score</tt> <tt class="py-op">-=</tt> <tt class="py-number">1</tt> </tt>
<a name="L653"></a><tt class="py-lineno">653</tt>  <tt class="py-line">                        <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Codon of {0}({1} {2}) does not "</tt> </tt>
<a name="L654"></a><tt class="py-lineno">654</tt>  <tt class="py-line">                                      <tt class="py-string">"correspond to {3}({4})"</tt><tt class="py-op">.</tt><tt id="link-322" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-322', 'format', 'link-51');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L655"></a><tt class="py-lineno">655</tt>  <tt class="py-line">                                          <tt class="py-name">pro</tt><tt class="py-op">.</tt><tt id="link-323" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-323', 'id', 'link-57');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">aa</tt><tt class="py-op">,</tt> <tt class="py-name">aa_num</tt><tt class="py-op">,</tt> <tt class="py-name">nucl</tt><tt class="py-op">.</tt><tt id="link-324" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-324', 'id', 'link-57');">id</a></tt><tt class="py-op">,</tt> </tt>
<a name="L656"></a><tt class="py-lineno">656</tt>  <tt class="py-line">                                          <tt class="py-name">this_codon</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L657"></a><tt class="py-lineno">657</tt>  <tt class="py-line">                                      <tt id="link-325" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-325', 'BiopythonWarning', 'link-1');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L658"></a><tt class="py-lineno">658</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">max_score</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L659"></a><tt class="py-lineno">659</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">RuntimeError</tt><tt class="py-op">(</tt><tt class="py-string">"max_score reached for {0}! Please "</tt> </tt>
<a name="L660"></a><tt class="py-lineno">660</tt>  <tt class="py-line">                                       <tt class="py-string">"raise up the tolerance to get an "</tt> </tt>
<a name="L661"></a><tt class="py-lineno">661</tt>  <tt class="py-line">                                       <tt class="py-string">"alignment in anyway"</tt><tt class="py-op">.</tt><tt id="link-326" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-326', 'format', 'link-51');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">nucl</tt><tt class="py-op">.</tt><tt id="link-327" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-327', 'id', 'link-57');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L662"></a><tt class="py-lineno">662</tt>  <tt class="py-line">                <tt class="py-name">codon_seq</tt> <tt class="py-op">+=</tt> <tt class="py-name">this_codon</tt> </tt>
<a name="L663"></a><tt class="py-lineno">663</tt>  <tt class="py-line">                <tt class="py-name">aa_num</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L664"></a><tt class="py-lineno">664</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-328" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-328', 'SeqRecord', 'link-10');">SeqRecord</a></tt><tt class="py-op">(</tt><tt id="link-329" class="py-name"><a title="Bio.codonalign.codonseq.CodonSeq" class="py-name" href="#" onclick="return doclink('link-329', 'CodonSeq', 'link-15');">CodonSeq</a></tt><tt class="py-op">(</tt><tt class="py-name">codon_seq</tt><tt class="py-op">,</tt> <tt id="link-330" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-330', 'alphabet', 'link-45');">alphabet</a></tt><tt class="py-op">=</tt><tt id="link-331" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-331', 'alphabet', 'link-45');">alphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L665"></a><tt class="py-lineno">665</tt>  <tt class="py-line">                         <tt class="py-name">rf_table</tt><tt class="py-op">=</tt><tt class="py-name">rf_table</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-332" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-332', 'id', 'link-57');">id</a></tt><tt class="py-op">=</tt><tt class="py-name">nucl</tt><tt class="py-op">.</tt><tt id="link-333" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-333', 'id', 'link-57');">id</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L666"></a><tt class="py-lineno">666</tt>  <tt class="py-line"> </tt>
<a name="L667"></a><tt class="py-lineno">667</tt>  <tt class="py-line"> </tt>
<a name="_align_shift_recs"></a><div id="_align_shift_recs-def"><a name="L668"></a><tt class="py-lineno">668</tt> <a class="py-toggle" href="#" id="_align_shift_recs-toggle" onclick="return toggle('_align_shift_recs');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.codonalign-module.html#_align_shift_recs">_align_shift_recs</a><tt class="py-op">(</tt><tt class="py-param">recs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_align_shift_recs-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_align_shift_recs-expanded"><a name="L669"></a><tt class="py-lineno">669</tt>  <tt class="py-line">    <tt class="py-docstring">"""Build alignment according to the frameshift detected by _check_corr (PRIVATE).</tt> </tt>
<a name="L670"></a><tt class="py-lineno">670</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L671"></a><tt class="py-lineno">671</tt>  <tt class="py-line"><tt class="py-docstring">    Argument:</tt> </tt>
<a name="L672"></a><tt class="py-lineno">672</tt>  <tt class="py-line"><tt class="py-docstring">     - recs - a list of SeqRecords containing a CodonSeq dictated</tt> </tt>
<a name="L673"></a><tt class="py-lineno">673</tt>  <tt class="py-line"><tt class="py-docstring">       by a rf_table (with frameshift in some of them).</tt> </tt>
<a name="L674"></a><tt class="py-lineno">674</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L675"></a><tt class="py-lineno">675</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L676"></a><tt class="py-lineno">676</tt>  <tt class="py-line">    <tt class="py-keyword">def</tt> <tt class="py-def-name">find_next_int</tt><tt class="py-op">(</tt><tt class="py-param">k</tt><tt class="py-op">,</tt> <tt class="py-param">lst</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L677"></a><tt class="py-lineno">677</tt>  <tt class="py-line">        <tt class="py-name">idx</tt> <tt class="py-op">=</tt> <tt class="py-name">lst</tt><tt class="py-op">.</tt><tt id="link-334" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-334', 'index', 'link-276');">index</a></tt><tt class="py-op">(</tt><tt class="py-name">k</tt><tt class="py-op">)</tt> </tt>
<a name="L678"></a><tt class="py-lineno">678</tt>  <tt class="py-line">        <tt class="py-name">p</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L679"></a><tt class="py-lineno">679</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L680"></a><tt class="py-lineno">680</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">lst</tt><tt class="py-op">[</tt><tt class="py-name">idx</tt> <tt class="py-op">+</tt> <tt class="py-name">p</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L681"></a><tt class="py-lineno">681</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">lst</tt><tt class="py-op">[</tt><tt class="py-name">idx</tt> <tt class="py-op">+</tt> <tt class="py-name">p</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">p</tt> </tt>
<a name="L682"></a><tt class="py-lineno">682</tt>  <tt class="py-line">            <tt class="py-name">p</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
</div><a name="L683"></a><tt class="py-lineno">683</tt>  <tt class="py-line">    <tt class="py-name">full_rf_table_lst</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">rec</tt><tt class="py-op">.</tt><tt id="link-335" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-335', 'seq', 'link-44');">seq</a></tt><tt class="py-op">.</tt><tt id="link-336" class="py-name" targets="Method Bio.codonalign.codonseq.CodonSeq.get_full_rf_table()=Bio.codonalign.codonseq.CodonSeq-class.html#get_full_rf_table"><a title="Bio.codonalign.codonseq.CodonSeq.get_full_rf_table" class="py-name" href="#" onclick="return doclink('link-336', 'get_full_rf_table', 'link-336');">get_full_rf_table</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">rec</tt> <tt class="py-keyword">in</tt> <tt class="py-name">recs</tt><tt class="py-op">]</tt> </tt>
<a name="L684"></a><tt class="py-lineno">684</tt>  <tt class="py-line">    <tt class="py-name">rf_num</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">*</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">recs</tt><tt class="py-op">)</tt> </tt>
<a name="L685"></a><tt class="py-lineno">685</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">rec</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt class="py-name">recs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L686"></a><tt class="py-lineno">686</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-337" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-337', 'i', 'link-56');">i</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">rec</tt><tt class="py-op">.</tt><tt id="link-338" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-338', 'seq', 'link-44');">seq</a></tt><tt class="py-op">.</tt><tt id="link-339" class="py-name"><a title="Bio.codonalign.codonseq.CodonSeq.get_full_rf_table" class="py-name" href="#" onclick="return doclink('link-339', 'get_full_rf_table', 'link-336');">get_full_rf_table</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L687"></a><tt class="py-lineno">687</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-340" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-340', 'i', 'link-56');">i</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L688"></a><tt class="py-lineno">688</tt>  <tt class="py-line">                <tt class="py-name">rf_num</tt><tt class="py-op">[</tt><tt class="py-name">k</tt><tt class="py-op">]</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L689"></a><tt class="py-lineno">689</tt>  <tt class="py-line">            <tt class="py-comment"># isinstance(i, float) should be True</tt> </tt>
<a name="L690"></a><tt class="py-lineno">690</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">rec</tt><tt class="py-op">.</tt><tt id="link-341" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-341', 'seq', 'link-44');">seq</a></tt><tt class="py-op">.</tt><tt class="py-name">_data</tt><tt class="py-op">[</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-342" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-342', 'i', 'link-56');">i</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-343" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-343', 'i', 'link-56');">i</a></tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-number">3</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"---"</tt><tt class="py-op">:</tt> </tt>
<a name="L691"></a><tt class="py-lineno">691</tt>  <tt class="py-line">                <tt class="py-name">rf_num</tt><tt class="py-op">[</tt><tt class="py-name">k</tt><tt class="py-op">]</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L692"></a><tt class="py-lineno">692</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-344" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-344', 'set', 'link-61');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">rf_num</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L693"></a><tt class="py-lineno">693</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">RuntimeError</tt><tt class="py-op">(</tt><tt class="py-string">"Number alignable codons unequal in given records"</tt><tt class="py-op">)</tt> </tt>
<a name="L694"></a><tt class="py-lineno">694</tt>  <tt class="py-line">    <tt id="link-345" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-345', 'i', 'link-56');">i</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L695"></a><tt class="py-lineno">695</tt>  <tt class="py-line">    <tt class="py-name">rec_num</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">recs</tt><tt class="py-op">)</tt> </tt>
<a name="L696"></a><tt class="py-lineno">696</tt>  <tt class="py-line">    <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L697"></a><tt class="py-lineno">697</tt>  <tt class="py-line">        <tt class="py-name">add_lst</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L698"></a><tt class="py-lineno">698</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L699"></a><tt class="py-lineno">699</tt>  <tt class="py-line">            <tt class="py-name">col_rf_lst</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">k</tt><tt class="py-op">[</tt><tt id="link-346" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-346', 'i', 'link-56');">i</a></tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt class="py-name">k</tt> <tt class="py-keyword">in</tt> <tt class="py-name">full_rf_table_lst</tt><tt class="py-op">]</tt> </tt>
<a name="L700"></a><tt class="py-lineno">700</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">IndexError</tt><tt class="py-op">:</tt> </tt>
<a name="L701"></a><tt class="py-lineno">701</tt>  <tt class="py-line">            <tt class="py-comment"># we probably reached the last codon</tt> </tt>
<a name="L702"></a><tt class="py-lineno">702</tt>  <tt class="py-line">            <tt class="py-keyword">break</tt> </tt>
<a name="L703"></a><tt class="py-lineno">703</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">j</tt><tt class="py-op">,</tt> <tt class="py-name">k</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt class="py-name">col_rf_lst</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L704"></a><tt class="py-lineno">704</tt>  <tt class="py-line">            <tt class="py-name">add_lst</tt><tt class="py-op">.</tt><tt id="link-347" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-347', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">j</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">k</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L705"></a><tt class="py-lineno">705</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">float</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L706"></a><tt class="py-lineno">706</tt>  <tt class="py-line">                    <tt class="py-name">recs</tt><tt class="py-op">[</tt><tt class="py-name">j</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-348" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-348', 'seq', 'link-44');">seq</a></tt><tt class="py-op">.</tt><tt class="py-name">_data</tt><tt class="py-op">[</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">k</tt><tt class="py-op">)</tt><tt class="py-op">:</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">k</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-number">3</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-string">"---"</tt><tt class="py-op">:</tt> </tt>
<a name="L707"></a><tt class="py-lineno">707</tt>  <tt class="py-line">                <tt class="py-name">m</tt><tt class="py-op">,</tt> <tt class="py-name">p</tt> <tt class="py-op">=</tt> <tt class="py-name">find_next_int</tt><tt class="py-op">(</tt><tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">full_rf_table_lst</tt><tt class="py-op">[</tt><tt class="py-name">j</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L708"></a><tt class="py-lineno">708</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-op">(</tt><tt class="py-name">m</tt> <tt class="py-op">-</tt> <tt class="py-name">k</tt><tt class="py-op">)</tt> <tt class="py-op">%</tt> <tt class="py-number">3</tt> <tt class="py-op">!=</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L709"></a><tt class="py-lineno">709</tt>  <tt class="py-line">                    <tt class="py-name">gap_num</tt> <tt class="py-op">=</tt> <tt class="py-number">3</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">m</tt> <tt class="py-op">-</tt> <tt class="py-name">k</tt><tt class="py-op">)</tt> <tt class="py-op">%</tt> <tt class="py-number">3</tt> </tt>
<a name="L710"></a><tt class="py-lineno">710</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L711"></a><tt class="py-lineno">711</tt>  <tt class="py-line">                    <tt class="py-name">gap_num</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L712"></a><tt class="py-lineno">712</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">gap_num</tt> <tt class="py-op">!=</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L713"></a><tt class="py-lineno">713</tt>  <tt class="py-line">                    <tt class="py-name">gaps</tt> <tt class="py-op">=</tt> <tt class="py-string">'-'</tt> <tt class="py-op">*</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">gap_num</tt><tt class="py-op">)</tt> </tt>
<a name="L714"></a><tt class="py-lineno">714</tt>  <tt class="py-line">                    <tt id="link-349" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-349', 'seq', 'link-44');">seq</a></tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">recs</tt><tt class="py-op">[</tt><tt class="py-name">j</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-350" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-350', 'seq', 'link-44');">seq</a></tt><tt class="py-op">.</tt><tt class="py-name">_data</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">k</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">gaps</tt> </tt>
<a name="L715"></a><tt class="py-lineno">715</tt>  <tt class="py-line">                           <tt class="py-op">+</tt> <tt class="py-name">recs</tt><tt class="py-op">[</tt><tt class="py-name">j</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-351" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-351', 'seq', 'link-44');">seq</a></tt><tt class="py-op">.</tt><tt class="py-name">_data</tt><tt class="py-op">[</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">k</tt><tt class="py-op">)</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L716"></a><tt class="py-lineno">716</tt>  <tt class="py-line">                    <tt class="py-name">full_rf_table</tt> <tt class="py-op">=</tt> <tt class="py-name">full_rf_table_lst</tt><tt class="py-op">[</tt><tt class="py-name">j</tt><tt class="py-op">]</tt> </tt>
<a name="L717"></a><tt class="py-lineno">717</tt>  <tt class="py-line">                    <tt class="py-name">bp</tt> <tt class="py-op">=</tt> <tt class="py-name">full_rf_table</tt><tt class="py-op">.</tt><tt id="link-352" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-352', 'index', 'link-276');">index</a></tt><tt class="py-op">(</tt><tt class="py-name">k</tt><tt class="py-op">)</tt> </tt>
<a name="L718"></a><tt class="py-lineno">718</tt>  <tt class="py-line">                    <tt class="py-name">full_rf_table</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">full_rf_table</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">bp</tt><tt class="py-op">]</tt> </tt>
<a name="L719"></a><tt class="py-lineno">719</tt>  <tt class="py-line">                                     <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-name">v</tt> <tt class="py-op">+</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">gap_num</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">v</tt> <tt class="py-keyword">in</tt> <tt class="py-name">full_rf_table</tt><tt class="py-op">[</tt><tt class="py-name">bp</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L720"></a><tt class="py-lineno">720</tt>  <tt class="py-line">                    <tt class="py-name">full_rf_table_lst</tt><tt class="py-op">[</tt><tt class="py-name">j</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">full_rf_table</tt> </tt>
<a name="L721"></a><tt class="py-lineno">721</tt>  <tt class="py-line">                    <tt class="py-name">recs</tt><tt class="py-op">[</tt><tt class="py-name">j</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-353" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-353', 'seq', 'link-44');">seq</a></tt> <tt class="py-op">=</tt> <tt id="link-354" class="py-name"><a title="Bio.codonalign.codonseq.CodonSeq" class="py-name" href="#" onclick="return doclink('link-354', 'CodonSeq', 'link-15');">CodonSeq</a></tt><tt class="py-op">(</tt><tt id="link-355" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-355', 'seq', 'link-44');">seq</a></tt><tt class="py-op">,</tt> </tt>
<a name="L722"></a><tt class="py-lineno">722</tt>  <tt class="py-line">                                           <tt class="py-name">rf_table</tt><tt class="py-op">=</tt><tt class="py-name">recs</tt><tt class="py-op">[</tt><tt class="py-name">j</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-356" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-356', 'seq', 'link-44');">seq</a></tt><tt class="py-op">.</tt><tt class="py-name">rf_table</tt><tt class="py-op">,</tt> </tt>
<a name="L723"></a><tt class="py-lineno">723</tt>  <tt class="py-line">                                           <tt id="link-357" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-357', 'alphabet', 'link-45');">alphabet</a></tt><tt class="py-op">=</tt><tt class="py-name">recs</tt><tt class="py-op">[</tt><tt class="py-name">j</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-358" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-358', 'seq', 'link-44');">seq</a></tt><tt class="py-op">.</tt><tt id="link-359" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-359', 'alphabet', 'link-45');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L724"></a><tt class="py-lineno">724</tt>  <tt class="py-line">                <tt class="py-name">add_lst</tt><tt class="py-op">.</tt><tt id="link-360" class="py-name"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop" class="py-name" href="#" onclick="return doclink('link-360', 'pop', 'link-179');">pop</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L725"></a><tt class="py-lineno">725</tt>  <tt class="py-line">                <tt class="py-name">gap_num</tt> <tt class="py-op">+=</tt> <tt class="py-name">m</tt> <tt class="py-op">-</tt> <tt class="py-name">k</tt> </tt>
<a name="L726"></a><tt class="py-lineno">726</tt>  <tt class="py-line">                <tt id="link-361" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-361', 'i', 'link-56');">i</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">p</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt> </tt>
<a name="L727"></a><tt class="py-lineno">727</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">add_lst</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">rec_num</tt><tt class="py-op">:</tt> </tt>
<a name="L728"></a><tt class="py-lineno">728</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">j</tt><tt class="py-op">,</tt> <tt class="py-name">k</tt> <tt class="py-keyword">in</tt> <tt class="py-name">add_lst</tt><tt class="py-op">:</tt> </tt>
<a name="L729"></a><tt class="py-lineno">729</tt>  <tt class="py-line">                <tt class="py-name">gaps</tt> <tt class="py-op">=</tt> <tt class="py-string">"-"</tt> <tt class="py-op">*</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">gap_num</tt><tt class="py-op">)</tt> </tt>
<a name="L730"></a><tt class="py-lineno">730</tt>  <tt class="py-line">                <tt id="link-362" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-362', 'seq', 'link-44');">seq</a></tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">recs</tt><tt class="py-op">[</tt><tt class="py-name">j</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-363" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-363', 'seq', 'link-44');">seq</a></tt><tt class="py-op">.</tt><tt class="py-name">_data</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">k</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">gaps</tt> </tt>
<a name="L731"></a><tt class="py-lineno">731</tt>  <tt class="py-line">                       <tt class="py-op">+</tt> <tt class="py-name">recs</tt><tt class="py-op">[</tt><tt class="py-name">j</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-364" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-364', 'seq', 'link-44');">seq</a></tt><tt class="py-op">.</tt><tt class="py-name">_data</tt><tt class="py-op">[</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">k</tt><tt class="py-op">)</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L732"></a><tt class="py-lineno">732</tt>  <tt class="py-line">                <tt class="py-name">full_rf_table</tt> <tt class="py-op">=</tt> <tt class="py-name">full_rf_table_lst</tt><tt class="py-op">[</tt><tt class="py-name">j</tt><tt class="py-op">]</tt> </tt>
<a name="L733"></a><tt class="py-lineno">733</tt>  <tt class="py-line">                <tt class="py-name">bp</tt> <tt class="py-op">=</tt> <tt class="py-name">full_rf_table</tt><tt class="py-op">.</tt><tt id="link-365" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-365', 'index', 'link-276');">index</a></tt><tt class="py-op">(</tt><tt class="py-name">k</tt><tt class="py-op">)</tt> </tt>
<a name="L734"></a><tt class="py-lineno">734</tt>  <tt class="py-line">                <tt class="py-name">inter_rf</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L735"></a><tt class="py-lineno">735</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">t</tt> <tt class="py-keyword">in</tt> <tt id="link-366" class="py-name" targets="Function Bio.FSSP.FSSPTools.filter()=Bio.FSSP.FSSPTools-module.html#filter,Method Bio.SearchIO._model.hit.Hit.filter()=Bio.SearchIO._model.hit.Hit-class.html#filter"><a title="Bio.FSSP.FSSPTools.filter
Bio.SearchIO._model.hit.Hit.filter" class="py-name" href="#" onclick="return doclink('link-366', 'filter', 'link-366');">filter</a></tt><tt class="py-op">(</tt><tt class="py-keyword">lambda</tt> <tt id="link-367" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.x=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#x,Variable Bio.Statistics.lowess.x=Bio.Statistics.lowess-module.html#x"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-367', 'x', 'link-367');">x</a></tt><tt class="py-op">:</tt> <tt id="link-368" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-368', 'x', 'link-367');">x</a></tt> <tt class="py-op">%</tt> <tt class="py-number">3</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt id="link-369" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-369', 'range', 'link-135');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">gaps</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L736"></a><tt class="py-lineno">736</tt>  <tt class="py-line">                    <tt class="py-name">inter_rf</tt><tt class="py-op">.</tt><tt id="link-370" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-370', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">k</tt> <tt class="py-op">+</tt> <tt class="py-name">t</tt> <tt class="py-op">+</tt> <tt class="py-number">3.0</tt><tt class="py-op">)</tt> </tt>
<a name="L737"></a><tt class="py-lineno">737</tt>  <tt class="py-line">                <tt class="py-name">full_rf_table</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">full_rf_table</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">bp</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">inter_rf</tt> </tt>
<a name="L738"></a><tt class="py-lineno">738</tt>  <tt class="py-line">                                 <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-name">v</tt> <tt class="py-op">+</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">gap_num</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">v</tt> <tt class="py-keyword">in</tt> <tt class="py-name">full_rf_table</tt><tt class="py-op">[</tt><tt class="py-name">bp</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L739"></a><tt class="py-lineno">739</tt>  <tt class="py-line">                <tt class="py-name">full_rf_table_lst</tt><tt class="py-op">[</tt><tt class="py-name">j</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">full_rf_table</tt> </tt>
<a name="L740"></a><tt class="py-lineno">740</tt>  <tt class="py-line">                <tt class="py-name">recs</tt><tt class="py-op">[</tt><tt class="py-name">j</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-371" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-371', 'seq', 'link-44');">seq</a></tt> <tt class="py-op">=</tt> <tt id="link-372" class="py-name"><a title="Bio.codonalign.codonseq.CodonSeq" class="py-name" href="#" onclick="return doclink('link-372', 'CodonSeq', 'link-15');">CodonSeq</a></tt><tt class="py-op">(</tt><tt id="link-373" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-373', 'seq', 'link-44');">seq</a></tt><tt class="py-op">,</tt> </tt>
<a name="L741"></a><tt class="py-lineno">741</tt>  <tt class="py-line">                                       <tt class="py-name">rf_table</tt><tt class="py-op">=</tt><tt class="py-name">recs</tt><tt class="py-op">[</tt><tt class="py-name">j</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-374" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-374', 'seq', 'link-44');">seq</a></tt><tt class="py-op">.</tt><tt class="py-name">rf_table</tt><tt class="py-op">,</tt> </tt>
<a name="L742"></a><tt class="py-lineno">742</tt>  <tt class="py-line">                                       <tt id="link-375" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-375', 'alphabet', 'link-45');">alphabet</a></tt><tt class="py-op">=</tt><tt class="py-name">recs</tt><tt class="py-op">[</tt><tt class="py-name">j</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-376" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-376', 'seq', 'link-44');">seq</a></tt><tt class="py-op">.</tt><tt id="link-377" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-377', 'alphabet', 'link-45');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L743"></a><tt class="py-lineno">743</tt>  <tt class="py-line">        <tt id="link-378" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-378', 'i', 'link-56');">i</a></tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L744"></a><tt class="py-lineno">744</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">recs</tt> </tt>
</div><a name="L745"></a><tt class="py-lineno">745</tt>  <tt class="py-line"> </tt>
<a name="L746"></a><tt class="py-lineno">746</tt>  <tt class="py-line"> </tt>
<a name="L747"></a><tt class="py-lineno">747</tt>  <tt class="py-line"><tt class="py-comment"># def toCodonAlignment(align, alphabet=default_codon_alphabet):</tt> </tt>
<a name="L748"></a><tt class="py-lineno">748</tt>  <tt class="py-line"><tt class="py-comment">#    """Function to convert a MultipleSeqAlignment to CodonAlignment.</tt> </tt>
<a name="L749"></a><tt class="py-lineno">749</tt>  <tt class="py-line"><tt class="py-comment">#    It is the user's responsibility to ensure all the requirement</tt> </tt>
<a name="L750"></a><tt class="py-lineno">750</tt>  <tt class="py-line"><tt class="py-comment">#    needed by CodonAlignment is met.</tt> </tt>
<a name="L751"></a><tt class="py-lineno">751</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L752"></a><tt class="py-lineno">752</tt>  <tt class="py-line"><tt class="py-comment">#    """</tt> </tt>
<a name="L753"></a><tt class="py-lineno">753</tt>  <tt class="py-line"><tt class="py-comment">#    rec = [SeqRecord(CodonSeq(str(i.seq), alphabet=alphabet), id=i.id) \</tt> </tt>
<a name="L754"></a><tt class="py-lineno">754</tt>  <tt class="py-line"><tt class="py-comment">#            for i in align._records]</tt> </tt>
<a name="L755"></a><tt class="py-lineno">755</tt>  <tt class="py-line"><tt class="py-comment">#    return CodonAlignment(rec, alphabet=align._alphabet)</tt> </tt>
<a name="L756"></a><tt class="py-lineno">756</tt>  <tt class="py-line"> </tt>
<a name="L757"></a><tt class="py-lineno">757</tt>  <tt class="py-line"> </tt>
<a name="L758"></a><tt class="py-lineno">758</tt>  <tt class="py-line"><tt class="py-keyword">if</tt> <tt class="py-name">__name__</tt> <tt class="py-op">==</tt> <tt class="py-string">"__main__"</tt><tt class="py-op">:</tt> </tt>
<a name="L759"></a><tt class="py-lineno">759</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-379" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-379', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-380" class="py-name" targets="Module Bio.Phylo._utils=Bio.Phylo._utils-module.html,Module Bio.SearchIO._utils=Bio.SearchIO._utils-module.html,Module Bio._utils=Bio._utils-module.html"><a title="Bio.Phylo._utils
Bio.SearchIO._utils
Bio._utils" class="py-name" href="#" onclick="return doclink('link-380', '_utils', 'link-380');">_utils</a></tt> <tt class="py-keyword">import</tt> <tt id="link-381" class="py-name" targets="Function Bio._utils.run_doctest()=Bio._utils-module.html#run_doctest"><a title="Bio._utils.run_doctest" class="py-name" href="#" onclick="return doclink('link-381', 'run_doctest', 'link-381');">run_doctest</a></tt> </tt>
<a name="L760"></a><tt class="py-lineno">760</tt>  <tt class="py-line">    <tt id="link-382" class="py-name"><a title="Bio._utils.run_doctest" class="py-name" href="#" onclick="return doclink('link-382', 'run_doctest', 'link-381');">run_doctest</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L761"></a><tt class="py-lineno">761</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Tue Dec 18 15:58:28 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
