<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.motifs.jaspar.db</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.motifs-module.html">Package&nbsp;motifs</a> ::
        <a href="Bio.motifs.jaspar-module.html">Package&nbsp;jaspar</a> ::
        Module&nbsp;db
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.motifs.jaspar.db-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.motifs.jaspar.db-module.html">Module Bio.motifs.jaspar.db</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2013 by David Arenillas and Anthony Mathelier. All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># license. Please see the LICENSE file that should have been included</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-docstring">"""Provides read access to a JASPAR5 formatted database.</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-docstring">This modules requires MySQLdb to be installed.</tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-docstring">Example, substitute the your database credentials as</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-docstring">appropriate:</tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio.motifs.jaspar.db import JASPAR5</tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt;</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; JASPAR_DB_HOST = "hostname.example.org"</tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; JASPAR_DB_NAME = "JASPAR_2013"</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; JASPAR_DB_USER = "guest"</tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; JASPAR_DB_PASS = "guest"</tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt;</tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; DFLT_COLLECTION = 'CORE'</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; jdb = JASPAR5(</tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-docstring">    ...     host=JASPAR_DB_HOST,</tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-docstring">    ...     name=JASPAR_DB_NAME,</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-docstring">    ...     user=JASPAR_DB_USER,</tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-docstring">    ...     password=JASPAR_DB_PASS</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-docstring">    ... )</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt;</tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt;</tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; ets1 = jdb.fetch_motif_by_id('MA0098')</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(ets1)</tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"><tt class="py-docstring">    TF name ETS1</tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"><tt class="py-docstring">    Matrix ID   MA0098.1</tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"><tt class="py-docstring">    Collection  CORE</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-docstring">    TF class    Winged Helix-Turn-Helix</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-docstring">    TF family   Ets</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-docstring">    Species 9606</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"><tt class="py-docstring">    Taxonomic group vertebrates</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"><tt class="py-docstring">    Accession   ['CAG47050']</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"><tt class="py-docstring">    Data type used  SELEX</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"><tt class="py-docstring">    Medline 1542566</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"><tt class="py-docstring">    PAZAR ID    TF0000070</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"><tt class="py-docstring">    Comments    -</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"><tt class="py-docstring">    Matrix:</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line"><tt class="py-docstring">            0      1      2      3      4      5</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line"><tt class="py-docstring">    A:   4.00  17.00   0.00   0.00   0.00   5.00</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line"><tt class="py-docstring">    C:  16.00   0.00   1.00  39.00  39.00   3.00</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line"><tt class="py-docstring">    G:   4.00   0.00   0.00   1.00   0.00  17.00</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line"><tt class="py-docstring">    T:  16.00  23.00  39.00   0.00   1.00  15.00</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt;</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; motifs = jdb.fetch_motifs(</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line"><tt class="py-docstring">    ...     collection = 'CORE',</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line"><tt class="py-docstring">    ...     tax_group = ['vertebrates', 'insects'],</tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line"><tt class="py-docstring">    ...     tf_class = 'Winged Helix-Turn-Helix',</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line"><tt class="py-docstring">    ...     tf_family = ['Forkhead', 'Ets'],</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line"><tt class="py-docstring">    ...     min_ic = 12</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"><tt class="py-docstring">    ... )</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt;</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; for motif in motifs:</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line"><tt class="py-docstring">    ...     pass # do something with the motif</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line"> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">__future__</tt> <tt class="py-keyword">import</tt> <tt class="py-name">print_function</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line"> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-1" class="py-name" targets="Class Bio.BiopythonWarning=Bio.BiopythonWarning-class.html"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-1', 'BiopythonWarning', 'link-1');">BiopythonWarning</a></tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-2" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-2', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-3" class="py-name" targets="Class Bio.MissingPythonDependencyError=Bio.MissingPythonDependencyError-class.html"><a title="Bio.MissingPythonDependencyError" class="py-name" href="#" onclick="return doclink('link-3', 'MissingPythonDependencyError', 'link-3');">MissingPythonDependencyError</a></tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line"> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line"><tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line">    <tt class="py-keyword">import</tt> <tt class="py-name">MySQLdb</tt> <tt class="py-keyword">as</tt> <tt class="py-name">mdb</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line"><tt class="py-keyword">except</tt><tt class="py-op">:</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line">    <tt class="py-keyword">raise</tt> <tt id="link-4" class="py-name"><a title="Bio.MissingPythonDependencyError" class="py-name" href="#" onclick="return doclink('link-4', 'MissingPythonDependencyError', 'link-3');">MissingPythonDependencyError</a></tt><tt class="py-op">(</tt><tt class="py-string">"Install MySQLdb if you want to use "</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">                                       <tt class="py-string">"Bio.motifs.jaspar.db"</tt><tt class="py-op">)</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line"> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-5" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-5', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-6" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-6', 'Alphabet', 'link-6');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-7" class="py-name" targets="Module Bio.Alphabet.IUPAC=Bio.Alphabet.IUPAC-module.html"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-7', 'IUPAC', 'link-7');">IUPAC</a></tt> <tt class="py-keyword">import</tt> <tt id="link-8" class="py-name" targets="Variable Bio.Alphabet.IUPAC.unambiguous_dna=Bio.Alphabet.IUPAC-module.html#unambiguous_dna"><a title="Bio.Alphabet.IUPAC.unambiguous_dna" class="py-name" href="#" onclick="return doclink('link-8', 'unambiguous_dna', 'link-8');">unambiguous_dna</a></tt> <tt class="py-keyword">as</tt> <tt id="link-9" class="py-name" targets="Variable Bio.Restriction.Restriction.AanI.dna=Bio.Restriction.Restriction.AanI-class.html#dna,Variable Bio.Restriction.Restriction.AarI.dna=Bio.Restriction.Restriction.AarI-class.html#dna,Variable Bio.Restriction.Restriction.AasI.dna=Bio.Restriction.Restriction.AasI-class.html#dna,Variable Bio.Restriction.Restriction.AatII.dna=Bio.Restriction.Restriction.AatII-class.html#dna,Variable Bio.Restriction.Restriction.AbaSI.dna=Bio.Restriction.Restriction.AbaSI-class.html#dna,Variable Bio.Restriction.Restriction.AbsI.dna=Bio.Restriction.Restriction.AbsI-class.html#dna,Variable Bio.Restriction.Restriction.Acc16I.dna=Bio.Restriction.Restriction.Acc16I-class.html#dna,Variable Bio.Restriction.Restriction.Acc36I.dna=Bio.Restriction.Restriction.Acc36I-class.html#dna,Variable Bio.Restriction.Restriction.Acc65I.dna=Bio.Restriction.Restriction.Acc65I-class.html#dna,Variable Bio.Restriction.Restriction.AccB1I.dna=Bio.Restriction.Restriction.AccB1I-class.html#dna,Variable Bio.Restriction.Restriction.AccB7I.dna=Bio.Restriction.Restriction.AccB7I-class.html#dna,Variable Bio.Restriction.Restriction.AccBSI.dna=Bio.Restriction.Restriction.AccBSI-class.html#dna,Variable Bio.Restriction.Restriction.AccI.dna=Bio.Restriction.Restriction.AccI-class.html#dna,Variable Bio.Restriction.Restriction.AccII.dna=Bio.Restriction.Restriction.AccII-class.html#dna,Variable Bio.Restriction.Restriction.AccIII.dna=Bio.Restriction.Restriction.AccIII-class.html#dna,Variable Bio.Restriction.Restriction.AceIII.dna=Bio.Restriction.Restriction.AceIII-class.html#dna,Variable Bio.Restriction.Restriction.AciI.dna=Bio.Restriction.Restriction.AciI-class.html#dna,Variable Bio.Restriction.Restriction.AclI.dna=Bio.Restriction.Restriction.AclI-class.html#dna,Variable Bio.Restriction.Restriction.AclWI.dna=Bio.Restriction.Restriction.AclWI-class.html#dna,Variable Bio.Restriction.Restriction.AcoI.dna=Bio.Restriction.Restriction.AcoI-class.html#dna,Variable Bio.Restriction.Restriction.AcsI.dna=Bio.Restriction.Restriction.AcsI-class.html#dna,Variable Bio.Restriction.Restriction.AcuI.dna=Bio.Restriction.Restriction.AcuI-class.html#dna,Variable Bio.Restriction.Restriction.AcvI.dna=Bio.Restriction.Restriction.AcvI-class.html#dna,Variable Bio.Restriction.Restriction.AcyI.dna=Bio.Restriction.Restriction.AcyI-class.html#dna,Variable Bio.Restriction.Restriction.AdeI.dna=Bio.Restriction.Restriction.AdeI-class.html#dna,Variable Bio.Restriction.Restriction.AfaI.dna=Bio.Restriction.Restriction.AfaI-class.html#dna,Variable Bio.Restriction.Restriction.AfeI.dna=Bio.Restriction.Restriction.AfeI-class.html#dna,Variable Bio.Restriction.Restriction.AfiI.dna=Bio.Restriction.Restriction.AfiI-class.html#dna,Variable Bio.Restriction.Restriction.AflII.dna=Bio.Restriction.Restriction.AflII-class.html#dna,Variable Bio.Restriction.Restriction.AflIII.dna=Bio.Restriction.Restriction.AflIII-class.html#dna,Variable Bio.Restriction.Restriction.AgeI.dna=Bio.Restriction.Restriction.AgeI-class.html#dna,Variable Bio.Restriction.Restriction.AgsI.dna=Bio.Restriction.Restriction.AgsI-class.html#dna,Variable Bio.Restriction.Restriction.AhaIII.dna=Bio.Restriction.Restriction.AhaIII-class.html#dna,Variable Bio.Restriction.Restriction.AhdI.dna=Bio.Restriction.Restriction.AhdI-class.html#dna,Variable Bio.Restriction.Restriction.AhlI.dna=Bio.Restriction.Restriction.AhlI-class.html#dna,Variable Bio.Restriction.Restriction.AjiI.dna=Bio.Restriction.Restriction.AjiI-class.html#dna,Variable Bio.Restriction.Restriction.AjnI.dna=Bio.Restriction.Restriction.AjnI-class.html#dna,Variable Bio.Restriction.Restriction.AjuI.dna=Bio.Restriction.Restriction.AjuI-class.html#dna,Variable Bio.Restriction.Restriction.AleI.dna=Bio.Restriction.Restriction.AleI-class.html#dna,Variable Bio.Restriction.Restriction.AlfI.dna=Bio.Restriction.Restriction.AlfI-class.html#dna,Variable Bio.Restriction.Restriction.AloI.dna=Bio.Restriction.Restriction.AloI-class.html#dna,Variable Bio.Restriction.Restriction.AluBI.dna=Bio.Restriction.Restriction.AluBI-class.html#dna,Variable Bio.Restriction.Restriction.AluI.dna=Bio.Restriction.Restriction.AluI-class.html#dna,Variable Bio.Restriction.Restriction.Alw21I.dna=Bio.Restriction.Restriction.Alw21I-class.html#dna,Variable Bio.Restriction.Restriction.Alw26I.dna=Bio.Restriction.Restriction.Alw26I-class.html#dna,Variable Bio.Restriction.Restriction.Alw44I.dna=Bio.Restriction.Restriction.Alw44I-class.html#dna,Variable Bio.Restriction.Restriction.AlwFI.dna=Bio.Restriction.Restriction.AlwFI-class.html#dna,Variable Bio.Restriction.Restriction.AlwI.dna=Bio.Restriction.Restriction.AlwI-class.html#dna,Variable Bio.Restriction.Restriction.AlwNI.dna=Bio.Restriction.Restriction.AlwNI-class.html#dna,Variable Bio.Restriction.Restriction.Ama87I.dna=Bio.Restriction.Restriction.Ama87I-class.html#dna,Variable Bio.Restriction.Restriction.Aor13HI.dna=Bio.Restriction.Restriction.Aor13HI-class.html#dna,Variable Bio.Restriction.Restriction.Aor51HI.dna=Bio.Restriction.Restriction.Aor51HI-class.html#dna,Variable Bio.Restriction.Restriction.AoxI.dna=Bio.Restriction.Restriction.AoxI-class.html#dna,Variable Bio.Restriction.Restriction.ApaBI.dna=Bio.Restriction.Restriction.ApaBI-class.html#dna,Variable Bio.Restriction.Restriction.ApaI.dna=Bio.Restriction.Restriction.ApaI-class.html#dna,Variable Bio.Restriction.Restriction.ApaLI.dna=Bio.Restriction.Restriction.ApaLI-class.html#dna,Variable Bio.Restriction.Restriction.ApeKI.dna=Bio.Restriction.Restriction.ApeKI-class.html#dna,Variable Bio.Restriction.Restriction.ApoI.dna=Bio.Restriction.Restriction.ApoI-class.html#dna,Variable Bio.Restriction.Restriction.ApyPI.dna=Bio.Restriction.Restriction.ApyPI-class.html#dna,Variable Bio.Restriction.Restriction.AquII.dna=Bio.Restriction.Restriction.AquII-class.html#dna,Variable Bio.Restriction.Restriction.AquIII.dna=Bio.Restriction.Restriction.AquIII-class.html#dna,Variable Bio.Restriction.Restriction.AquIV.dna=Bio.Restriction.Restriction.AquIV-class.html#dna,Variable Bio.Restriction.Restriction.ArsI.dna=Bio.Restriction.Restriction.ArsI-class.html#dna,Variable Bio.Restriction.Restriction.AscI.dna=Bio.Restriction.Restriction.AscI-class.html#dna,Variable Bio.Restriction.Restriction.AseI.dna=Bio.Restriction.Restriction.AseI-class.html#dna,Variable Bio.Restriction.Restriction.Asi256I.dna=Bio.Restriction.Restriction.Asi256I-class.html#dna,Variable Bio.Restriction.Restriction.AsiGI.dna=Bio.Restriction.Restriction.AsiGI-class.html#dna,Variable Bio.Restriction.Restriction.AsiSI.dna=Bio.Restriction.Restriction.AsiSI-class.html#dna,Variable Bio.Restriction.Restriction.Asp700I.dna=Bio.Restriction.Restriction.Asp700I-class.html#dna,Variable Bio.Restriction.Restriction.Asp718I.dna=Bio.Restriction.Restriction.Asp718I-class.html#dna,Variable Bio.Restriction.Restriction.AspA2I.dna=Bio.Restriction.Restriction.AspA2I-class.html#dna,Variable Bio.Restriction.Restriction.AspBHI.dna=Bio.Restriction.Restriction.AspBHI-class.html#dna,Variable Bio.Restriction.Restriction.AspLEI.dna=Bio.Restriction.Restriction.AspLEI-class.html#dna,Variable Bio.Restriction.Restriction.AspS9I.dna=Bio.Restriction.Restriction.AspS9I-class.html#dna,Variable Bio.Restriction.Restriction.AssI.dna=Bio.Restriction.Restriction.AssI-class.html#dna,Variable Bio.Restriction.Restriction.AsuC2I.dna=Bio.Restriction.Restriction.AsuC2I-class.html#dna,Variable Bio.Restriction.Restriction.AsuHPI.dna=Bio.Restriction.Restriction.AsuHPI-class.html#dna,Variable Bio.Restriction.Restriction.AsuI.dna=Bio.Restriction.Restriction.AsuI-class.html#dna,Variable Bio.Restriction.Restriction.AsuII.dna=Bio.Restriction.Restriction.AsuII-class.html#dna,Variable Bio.Restriction.Restriction.AsuNHI.dna=Bio.Restriction.Restriction.AsuNHI-class.html#dna,Variable Bio.Restriction.Restriction.AvaI.dna=Bio.Restriction.Restriction.AvaI-class.html#dna,Variable Bio.Restriction.Restriction.AvaII.dna=Bio.Restriction.Restriction.AvaII-class.html#dna,Variable Bio.Restriction.Restriction.AvaIII.dna=Bio.Restriction.Restriction.AvaIII-class.html#dna,Variable Bio.Restriction.Restriction.AvrII.dna=Bio.Restriction.Restriction.AvrII-class.html#dna,Variable Bio.Restriction.Restriction.AxyI.dna=Bio.Restriction.Restriction.AxyI-class.html#dna,Variable Bio.Restriction.Restriction.BaeGI.dna=Bio.Restriction.Restriction.BaeGI-class.html#dna,Variable Bio.Restriction.Restriction.BaeI.dna=Bio.Restriction.Restriction.BaeI-class.html#dna,Variable Bio.Restriction.Restriction.BalI.dna=Bio.Restriction.Restriction.BalI-class.html#dna,Variable Bio.Restriction.Restriction.BamHI.dna=Bio.Restriction.Restriction.BamHI-class.html#dna,Variable Bio.Restriction.Restriction.BanI.dna=Bio.Restriction.Restriction.BanI-class.html#dna,Variable Bio.Restriction.Restriction.BanII.dna=Bio.Restriction.Restriction.BanII-class.html#dna,Variable Bio.Restriction.Restriction.BarI.dna=Bio.Restriction.Restriction.BarI-class.html#dna,Variable Bio.Restriction.Restriction.BasI.dna=Bio.Restriction.Restriction.BasI-class.html#dna,Variable Bio.Restriction.Restriction.BauI.dna=Bio.Restriction.Restriction.BauI-class.html#dna,Variable Bio.Restriction.Restriction.Bbr7I.dna=Bio.Restriction.Restriction.Bbr7I-class.html#dna,Variable Bio.Restriction.Restriction.BbrPI.dna=Bio.Restriction.Restriction.BbrPI-class.html#dna,Variable Bio.Restriction.Restriction.BbsI.dna=Bio.Restriction.Restriction.BbsI-class.html#dna,Variable Bio.Restriction.Restriction.Bbv12I.dna=Bio.Restriction.Restriction.Bbv12I-class.html#dna,Variable Bio.Restriction.Restriction.BbvCI.dna=Bio.Restriction.Restriction.BbvCI-class.html#dna,Variable Bio.Restriction.Restriction.BbvI.dna=Bio.Restriction.Restriction.BbvI-class.html#dna,Variable Bio.Restriction.Restriction.BbvII.dna=Bio.Restriction.Restriction.BbvII-class.html#dna,Variable Bio.Restriction.Restriction.BccI.dna=Bio.Restriction.Restriction.BccI-class.html#dna,Variable Bio.Restriction.Restriction.Bce83I.dna=Bio.Restriction.Restriction.Bce83I-class.html#dna,Variable Bio.Restriction.Restriction.BceAI.dna=Bio.Restriction.Restriction.BceAI-class.html#dna,Variable Bio.Restriction.Restriction.BcefI.dna=Bio.Restriction.Restriction.BcefI-class.html#dna,Variable Bio.Restriction.Restriction.BcgI.dna=Bio.Restriction.Restriction.BcgI-class.html#dna,Variable Bio.Restriction.Restriction.BciT130I.dna=Bio.Restriction.Restriction.BciT130I-class.html#dna,Variable Bio.Restriction.Restriction.BciVI.dna=Bio.Restriction.Restriction.BciVI-class.html#dna,Variable Bio.Restriction.Restriction.BclI.dna=Bio.Restriction.Restriction.BclI-class.html#dna,Variable Bio.Restriction.Restriction.BcnI.dna=Bio.Restriction.Restriction.BcnI-class.html#dna,Variable Bio.Restriction.Restriction.BcoDI.dna=Bio.Restriction.Restriction.BcoDI-class.html#dna,Variable Bio.Restriction.Restriction.BcuI.dna=Bio.Restriction.Restriction.BcuI-class.html#dna,Variable Bio.Restriction.Restriction.BdaI.dna=Bio.Restriction.Restriction.BdaI-class.html#dna,Variable Bio.Restriction.Restriction.BetI.dna=Bio.Restriction.Restriction.BetI-class.html#dna,Variable Bio.Restriction.Restriction.BfaI.dna=Bio.Restriction.Restriction.BfaI-class.html#dna,Variable Bio.Restriction.Restriction.BfiI.dna=Bio.Restriction.Restriction.BfiI-class.html#dna,Variable Bio.Restriction.Restriction.BfmI.dna=Bio.Restriction.Restriction.BfmI-class.html#dna,Variable Bio.Restriction.Restriction.BfoI.dna=Bio.Restriction.Restriction.BfoI-class.html#dna,Variable Bio.Restriction.Restriction.BfrI.dna=Bio.Restriction.Restriction.BfrI-class.html#dna,Variable Bio.Restriction.Restriction.BfuAI.dna=Bio.Restriction.Restriction.BfuAI-class.html#dna,Variable Bio.Restriction.Restriction.BfuCI.dna=Bio.Restriction.Restriction.BfuCI-class.html#dna,Variable Bio.Restriction.Restriction.BfuI.dna=Bio.Restriction.Restriction.BfuI-class.html#dna,Variable Bio.Restriction.Restriction.BglI.dna=Bio.Restriction.Restriction.BglI-class.html#dna,Variable Bio.Restriction.Restriction.BglII.dna=Bio.Restriction.Restriction.BglII-class.html#dna,Variable Bio.Restriction.Restriction.BinI.dna=Bio.Restriction.Restriction.BinI-class.html#dna,Variable Bio.Restriction.Restriction.BisI.dna=Bio.Restriction.Restriction.BisI-class.html#dna,Variable Bio.Restriction.Restriction.BlnI.dna=Bio.Restriction.Restriction.BlnI-class.html#dna,Variable Bio.Restriction.Restriction.BlpI.dna=Bio.Restriction.Restriction.BlpI-class.html#dna,Variable Bio.Restriction.Restriction.BlsI.dna=Bio.Restriction.Restriction.BlsI-class.html#dna,Variable Bio.Restriction.Restriction.BmcAI.dna=Bio.Restriction.Restriction.BmcAI-class.html#dna,Variable Bio.Restriction.Restriction.Bme1390I.dna=Bio.Restriction.Restriction.Bme1390I-class.html#dna,Variable Bio.Restriction.Restriction.Bme18I.dna=Bio.Restriction.Restriction.Bme18I-class.html#dna,Variable Bio.Restriction.Restriction.BmeDI.dna=Bio.Restriction.Restriction.BmeDI-class.html#dna,Variable Bio.Restriction.Restriction.BmeRI.dna=Bio.Restriction.Restriction.BmeRI-class.html#dna,Variable Bio.Restriction.Restriction.BmeT110I.dna=Bio.Restriction.Restriction.BmeT110I-class.html#dna,Variable Bio.Restriction.Restriction.BmgBI.dna=Bio.Restriction.Restriction.BmgBI-class.html#dna,Variable Bio.Restriction.Restriction.BmgI.dna=Bio.Restriction.Restriction.BmgI-class.html#dna,Variable Bio.Restriction.Restriction.BmgT120I.dna=Bio.Restriction.Restriction.BmgT120I-class.html#dna,Variable Bio.Restriction.Restriction.BmiI.dna=Bio.Restriction.Restriction.BmiI-class.html#dna,Variable Bio.Restriction.Restriction.BmrFI.dna=Bio.Restriction.Restriction.BmrFI-class.html#dna,Variable Bio.Restriction.Restriction.BmrI.dna=Bio.Restriction.Restriction.BmrI-class.html#dna,Variable Bio.Restriction.Restriction.BmsI.dna=Bio.Restriction.Restriction.BmsI-class.html#dna,Variable Bio.Restriction.Restriction.BmtI.dna=Bio.Restriction.Restriction.BmtI-class.html#dna,Variable Bio.Restriction.Restriction.BmuI.dna=Bio.Restriction.Restriction.BmuI-class.html#dna,Variable Bio.Restriction.Restriction.BoxI.dna=Bio.Restriction.Restriction.BoxI-class.html#dna,Variable Bio.Restriction.Restriction.BpiI.dna=Bio.Restriction.Restriction.BpiI-class.html#dna,Variable Bio.Restriction.Restriction.BplI.dna=Bio.Restriction.Restriction.BplI-class.html#dna,Variable Bio.Restriction.Restriction.BpmI.dna=Bio.Restriction.Restriction.BpmI-class.html#dna,Variable Bio.Restriction.Restriction.Bpu10I.dna=Bio.Restriction.Restriction.Bpu10I-class.html#dna,Variable Bio.Restriction.Restriction.Bpu1102I.dna=Bio.Restriction.Restriction.Bpu1102I-class.html#dna,Variable Bio.Restriction.Restriction.Bpu14I.dna=Bio.Restriction.Restriction.Bpu14I-class.html#dna,Variable Bio.Restriction.Restriction.BpuEI.dna=Bio.Restriction.Restriction.BpuEI-class.html#dna,Variable Bio.Restriction.Restriction.BpuMI.dna=Bio.Restriction.Restriction.BpuMI-class.html#dna,Variable Bio.Restriction.Restriction.BpvUI.dna=Bio.Restriction.Restriction.BpvUI-class.html#dna,Variable Bio.Restriction.Restriction.Bsa29I.dna=Bio.Restriction.Restriction.Bsa29I-class.html#dna,Variable Bio.Restriction.Restriction.BsaAI.dna=Bio.Restriction.Restriction.BsaAI-class.html#dna,Variable Bio.Restriction.Restriction.BsaBI.dna=Bio.Restriction.Restriction.BsaBI-class.html#dna,Variable Bio.Restriction.Restriction.BsaHI.dna=Bio.Restriction.Restriction.BsaHI-class.html#dna,Variable Bio.Restriction.Restriction.BsaI.dna=Bio.Restriction.Restriction.BsaI-class.html#dna,Variable Bio.Restriction.Restriction.BsaJI.dna=Bio.Restriction.Restriction.BsaJI-class.html#dna,Variable Bio.Restriction.Restriction.BsaWI.dna=Bio.Restriction.Restriction.BsaWI-class.html#dna,Variable Bio.Restriction.Restriction.BsaXI.dna=Bio.Restriction.Restriction.BsaXI-class.html#dna,Variable Bio.Restriction.Restriction.BsbI.dna=Bio.Restriction.Restriction.BsbI-class.html#dna,Variable Bio.Restriction.Restriction.Bsc4I.dna=Bio.Restriction.Restriction.Bsc4I-class.html#dna,Variable Bio.Restriction.Restriction.BscAI.dna=Bio.Restriction.Restriction.BscAI-class.html#dna,Variable Bio.Restriction.Restriction.BscGI.dna=Bio.Restriction.Restriction.BscGI-class.html#dna,Variable Bio.Restriction.Restriction.Bse118I.dna=Bio.Restriction.Restriction.Bse118I-class.html#dna,Variable Bio.Restriction.Restriction.Bse1I.dna=Bio.Restriction.Restriction.Bse1I-class.html#dna,Variable Bio.Restriction.Restriction.Bse21I.dna=Bio.Restriction.Restriction.Bse21I-class.html#dna,Variable Bio.Restriction.Restriction.Bse3DI.dna=Bio.Restriction.Restriction.Bse3DI-class.html#dna,Variable Bio.Restriction.Restriction.Bse8I.dna=Bio.Restriction.Restriction.Bse8I-class.html#dna,Variable Bio.Restriction.Restriction.BseAI.dna=Bio.Restriction.Restriction.BseAI-class.html#dna,Variable Bio.Restriction.Restriction.BseBI.dna=Bio.Restriction.Restriction.BseBI-class.html#dna,Variable Bio.Restriction.Restriction.BseCI.dna=Bio.Restriction.Restriction.BseCI-class.html#dna,Variable Bio.Restriction.Restriction.BseDI.dna=Bio.Restriction.Restriction.BseDI-class.html#dna,Variable Bio.Restriction.Restriction.BseGI.dna=Bio.Restriction.Restriction.BseGI-class.html#dna,Variable Bio.Restriction.Restriction.BseJI.dna=Bio.Restriction.Restriction.BseJI-class.html#dna,Variable Bio.Restriction.Restriction.BseLI.dna=Bio.Restriction.Restriction.BseLI-class.html#dna,Variable Bio.Restriction.Restriction.BseMI.dna=Bio.Restriction.Restriction.BseMI-class.html#dna,Variable Bio.Restriction.Restriction.BseMII.dna=Bio.Restriction.Restriction.BseMII-class.html#dna,Variable Bio.Restriction.Restriction.BseNI.dna=Bio.Restriction.Restriction.BseNI-class.html#dna,Variable Bio.Restriction.Restriction.BsePI.dna=Bio.Restriction.Restriction.BsePI-class.html#dna,Variable Bio.Restriction.Restriction.BseRI.dna=Bio.Restriction.Restriction.BseRI-class.html#dna,Variable Bio.Restriction.Restriction.BseSI.dna=Bio.Restriction.Restriction.BseSI-class.html#dna,Variable Bio.Restriction.Restriction.BseX3I.dna=Bio.Restriction.Restriction.BseX3I-class.html#dna,Variable Bio.Restriction.Restriction.BseXI.dna=Bio.Restriction.Restriction.BseXI-class.html#dna,Variable Bio.Restriction.Restriction.BseYI.dna=Bio.Restriction.Restriction.BseYI-class.html#dna,Variable Bio.Restriction.Restriction.BsgI.dna=Bio.Restriction.Restriction.BsgI-class.html#dna,Variable Bio.Restriction.Restriction.Bsh1236I.dna=Bio.Restriction.Restriction.Bsh1236I-class.html#dna,Variable Bio.Restriction.Restriction.Bsh1285I.dna=Bio.Restriction.Restriction.Bsh1285I-class.html#dna,Variable Bio.Restriction.Restriction.BshFI.dna=Bio.Restriction.Restriction.BshFI-class.html#dna,Variable Bio.Restriction.Restriction.BshNI.dna=Bio.Restriction.Restriction.BshNI-class.html#dna,Variable Bio.Restriction.Restriction.BshTI.dna=Bio.Restriction.Restriction.BshTI-class.html#dna,Variable Bio.Restriction.Restriction.BshVI.dna=Bio.Restriction.Restriction.BshVI-class.html#dna,Variable Bio.Restriction.Restriction.BsiEI.dna=Bio.Restriction.Restriction.BsiEI-class.html#dna,Variable Bio.Restriction.Restriction.BsiHKAI.dna=Bio.Restriction.Restriction.BsiHKAI-class.html#dna,Variable Bio.Restriction.Restriction.BsiHKCI.dna=Bio.Restriction.Restriction.BsiHKCI-class.html#dna,Variable Bio.Restriction.Restriction.BsiI.dna=Bio.Restriction.Restriction.BsiI-class.html#dna,Variable Bio.Restriction.Restriction.BsiSI.dna=Bio.Restriction.Restriction.BsiSI-class.html#dna,Variable Bio.Restriction.Restriction.BsiWI.dna=Bio.Restriction.Restriction.BsiWI-class.html#dna,Variable Bio.Restriction.Restriction.BsiYI.dna=Bio.Restriction.Restriction.BsiYI-class.html#dna,Variable Bio.Restriction.Restriction.BslFI.dna=Bio.Restriction.Restriction.BslFI-class.html#dna,Variable Bio.Restriction.Restriction.BslI.dna=Bio.Restriction.Restriction.BslI-class.html#dna,Variable Bio.Restriction.Restriction.BsmAI.dna=Bio.Restriction.Restriction.BsmAI-class.html#dna,Variable Bio.Restriction.Restriction.BsmBI.dna=Bio.Restriction.Restriction.BsmBI-class.html#dna,Variable Bio.Restriction.Restriction.BsmFI.dna=Bio.Restriction.Restriction.BsmFI-class.html#dna,Variable Bio.Restriction.Restriction.BsmI.dna=Bio.Restriction.Restriction.BsmI-class.html#dna,Variable Bio.Restriction.Restriction.BsnI.dna=Bio.Restriction.Restriction.BsnI-class.html#dna,Variable Bio.Restriction.Restriction.Bso31I.dna=Bio.Restriction.Restriction.Bso31I-class.html#dna,Variable Bio.Restriction.Restriction.BsoBI.dna=Bio.Restriction.Restriction.BsoBI-class.html#dna,Variable Bio.Restriction.Restriction.Bsp119I.dna=Bio.Restriction.Restriction.Bsp119I-class.html#dna,Variable Bio.Restriction.Restriction.Bsp120I.dna=Bio.Restriction.Restriction.Bsp120I-class.html#dna,Variable Bio.Restriction.Restriction.Bsp1286I.dna=Bio.Restriction.Restriction.Bsp1286I-class.html#dna,Variable Bio.Restriction.Restriction.Bsp13I.dna=Bio.Restriction.Restriction.Bsp13I-class.html#dna,Variable Bio.Restriction.Restriction.Bsp1407I.dna=Bio.Restriction.Restriction.Bsp1407I-class.html#dna,Variable Bio.Restriction.Restriction.Bsp143I.dna=Bio.Restriction.Restriction.Bsp143I-class.html#dna,Variable Bio.Restriction.Restriction.Bsp1720I.dna=Bio.Restriction.Restriction.Bsp1720I-class.html#dna,Variable Bio.Restriction.Restriction.Bsp19I.dna=Bio.Restriction.Restriction.Bsp19I-class.html#dna,Variable Bio.Restriction.Restriction.Bsp24I.dna=Bio.Restriction.Restriction.Bsp24I-class.html#dna,Variable Bio.Restriction.Restriction.Bsp68I.dna=Bio.Restriction.Restriction.Bsp68I-class.html#dna,Variable Bio.Restriction.Restriction.BspACI.dna=Bio.Restriction.Restriction.BspACI-class.html#dna,Variable Bio.Restriction.Restriction.BspCNI.dna=Bio.Restriction.Restriction.BspCNI-class.html#dna,Variable Bio.Restriction.Restriction.BspD6I.dna=Bio.Restriction.Restriction.BspD6I-class.html#dna,Variable Bio.Restriction.Restriction.BspDI.dna=Bio.Restriction.Restriction.BspDI-class.html#dna,Variable Bio.Restriction.Restriction.BspEI.dna=Bio.Restriction.Restriction.BspEI-class.html#dna,Variable Bio.Restriction.Restriction.BspFNI.dna=Bio.Restriction.Restriction.BspFNI-class.html#dna,Variable Bio.Restriction.Restriction.BspGI.dna=Bio.Restriction.Restriction.BspGI-class.html#dna,Variable Bio.Restriction.Restriction.BspHI.dna=Bio.Restriction.Restriction.BspHI-class.html#dna,Variable Bio.Restriction.Restriction.BspLI.dna=Bio.Restriction.Restriction.BspLI-class.html#dna,Variable Bio.Restriction.Restriction.BspLU11I.dna=Bio.Restriction.Restriction.BspLU11I-class.html#dna,Variable Bio.Restriction.Restriction.BspMI.dna=Bio.Restriction.Restriction.BspMI-class.html#dna,Variable Bio.Restriction.Restriction.BspMII.dna=Bio.Restriction.Restriction.BspMII-class.html#dna,Variable Bio.Restriction.Restriction.BspNCI.dna=Bio.Restriction.Restriction.BspNCI-class.html#dna,Variable Bio.Restriction.Restriction.BspOI.dna=Bio.Restriction.Restriction.BspOI-class.html#dna,Variable Bio.Restriction.Restriction.BspPI.dna=Bio.Restriction.Restriction.BspPI-class.html#dna,Variable Bio.Restriction.Restriction.BspQI.dna=Bio.Restriction.Restriction.BspQI-class.html#dna,Variable Bio.Restriction.Restriction.BspT104I.dna=Bio.Restriction.Restriction.BspT104I-class.html#dna,Variable Bio.Restriction.Restriction.BspT107I.dna=Bio.Restriction.Restriction.BspT107I-class.html#dna,Variable Bio.Restriction.Restriction.BspTI.dna=Bio.Restriction.Restriction.BspTI-class.html#dna,Variable Bio.Restriction.Restriction.BsrBI.dna=Bio.Restriction.Restriction.BsrBI-class.html#dna,Variable Bio.Restriction.Restriction.BsrDI.dna=Bio.Restriction.Restriction.BsrDI-class.html#dna,Variable Bio.Restriction.Restriction.BsrFI.dna=Bio.Restriction.Restriction.BsrFI-class.html#dna,Variable Bio.Restriction.Restriction.BsrGI.dna=Bio.Restriction.Restriction.BsrGI-class.html#dna,Variable Bio.Restriction.Restriction.BsrI.dna=Bio.Restriction.Restriction.BsrI-class.html#dna,Variable Bio.Restriction.Restriction.BsrSI.dna=Bio.Restriction.Restriction.BsrSI-class.html#dna,Variable Bio.Restriction.Restriction.BssAI.dna=Bio.Restriction.Restriction.BssAI-class.html#dna,Variable Bio.Restriction.Restriction.BssECI.dna=Bio.Restriction.Restriction.BssECI-class.html#dna,Variable Bio.Restriction.Restriction.BssHII.dna=Bio.Restriction.Restriction.BssHII-class.html#dna,Variable Bio.Restriction.Restriction.BssKI.dna=Bio.Restriction.Restriction.BssKI-class.html#dna,Variable Bio.Restriction.Restriction.BssMI.dna=Bio.Restriction.Restriction.BssMI-class.html#dna,Variable Bio.Restriction.Restriction.BssNAI.dna=Bio.Restriction.Restriction.BssNAI-class.html#dna,Variable Bio.Restriction.Restriction.BssNI.dna=Bio.Restriction.Restriction.BssNI-class.html#dna,Variable Bio.Restriction.Restriction.BssSI.dna=Bio.Restriction.Restriction.BssSI-class.html#dna,Variable Bio.Restriction.Restriction.BssT1I.dna=Bio.Restriction.Restriction.BssT1I-class.html#dna,Variable Bio.Restriction.Restriction.Bst1107I.dna=Bio.Restriction.Restriction.Bst1107I-class.html#dna,Variable Bio.Restriction.Restriction.Bst2BI.dna=Bio.Restriction.Restriction.Bst2BI-class.html#dna,Variable Bio.Restriction.Restriction.Bst2UI.dna=Bio.Restriction.Restriction.Bst2UI-class.html#dna,Variable Bio.Restriction.Restriction.Bst4CI.dna=Bio.Restriction.Restriction.Bst4CI-class.html#dna,Variable Bio.Restriction.Restriction.Bst6I.dna=Bio.Restriction.Restriction.Bst6I-class.html#dna,Variable Bio.Restriction.Restriction.BstACI.dna=Bio.Restriction.Restriction.BstACI-class.html#dna,Variable Bio.Restriction.Restriction.BstAFI.dna=Bio.Restriction.Restriction.BstAFI-class.html#dna,Variable Bio.Restriction.Restriction.BstAPI.dna=Bio.Restriction.Restriction.BstAPI-class.html#dna,Variable Bio.Restriction.Restriction.BstAUI.dna=Bio.Restriction.Restriction.BstAUI-class.html#dna,Variable Bio.Restriction.Restriction.BstBAI.dna=Bio.Restriction.Restriction.BstBAI-class.html#dna,Variable Bio.Restriction.Restriction.BstBI.dna=Bio.Restriction.Restriction.BstBI-class.html#dna,Variable Bio.Restriction.Restriction.BstC8I.dna=Bio.Restriction.Restriction.BstC8I-class.html#dna,Variable Bio.Restriction.Restriction.BstDEI.dna=Bio.Restriction.Restriction.BstDEI-class.html#dna,Variable Bio.Restriction.Restriction.BstDSI.dna=Bio.Restriction.Restriction.BstDSI-class.html#dna,Variable Bio.Restriction.Restriction.BstEII.dna=Bio.Restriction.Restriction.BstEII-class.html#dna,Variable Bio.Restriction.Restriction.BstENI.dna=Bio.Restriction.Restriction.BstENI-class.html#dna,Variable Bio.Restriction.Restriction.BstF5I.dna=Bio.Restriction.Restriction.BstF5I-class.html#dna,Variable Bio.Restriction.Restriction.BstFNI.dna=Bio.Restriction.Restriction.BstFNI-class.html#dna,Variable Bio.Restriction.Restriction.BstH2I.dna=Bio.Restriction.Restriction.BstH2I-class.html#dna,Variable Bio.Restriction.Restriction.BstHHI.dna=Bio.Restriction.Restriction.BstHHI-class.html#dna,Variable Bio.Restriction.Restriction.BstKTI.dna=Bio.Restriction.Restriction.BstKTI-class.html#dna,Variable Bio.Restriction.Restriction.BstMAI.dna=Bio.Restriction.Restriction.BstMAI-class.html#dna,Variable Bio.Restriction.Restriction.BstMBI.dna=Bio.Restriction.Restriction.BstMBI-class.html#dna,Variable Bio.Restriction.Restriction.BstMCI.dna=Bio.Restriction.Restriction.BstMCI-class.html#dna,Variable Bio.Restriction.Restriction.BstMWI.dna=Bio.Restriction.Restriction.BstMWI-class.html#dna,Variable Bio.Restriction.Restriction.BstNI.dna=Bio.Restriction.Restriction.BstNI-class.html#dna,Variable Bio.Restriction.Restriction.BstNSI.dna=Bio.Restriction.Restriction.BstNSI-class.html#dna,Variable Bio.Restriction.Restriction.BstOI.dna=Bio.Restriction.Restriction.BstOI-class.html#dna,Variable Bio.Restriction.Restriction.BstPAI.dna=Bio.Restriction.Restriction.BstPAI-class.html#dna,Variable Bio.Restriction.Restriction.BstPI.dna=Bio.Restriction.Restriction.BstPI-class.html#dna,Variable Bio.Restriction.Restriction.BstSCI.dna=Bio.Restriction.Restriction.BstSCI-class.html#dna,Variable Bio.Restriction.Restriction.BstSFI.dna=Bio.Restriction.Restriction.BstSFI-class.html#dna,Variable Bio.Restriction.Restriction.BstSLI.dna=Bio.Restriction.Restriction.BstSLI-class.html#dna,Variable Bio.Restriction.Restriction.BstSNI.dna=Bio.Restriction.Restriction.BstSNI-class.html#dna,Variable Bio.Restriction.Restriction.BstUI.dna=Bio.Restriction.Restriction.BstUI-class.html#dna,Variable Bio.Restriction.Restriction.BstV1I.dna=Bio.Restriction.Restriction.BstV1I-class.html#dna,Variable Bio.Restriction.Restriction.BstV2I.dna=Bio.Restriction.Restriction.BstV2I-class.html#dna,Variable Bio.Restriction.Restriction.BstX2I.dna=Bio.Restriction.Restriction.BstX2I-class.html#dna,Variable Bio.Restriction.Restriction.BstXI.dna=Bio.Restriction.Restriction.BstXI-class.html#dna,Variable Bio.Restriction.Restriction.BstYI.dna=Bio.Restriction.Restriction.BstYI-class.html#dna,Variable Bio.Restriction.Restriction.BstZ17I.dna=Bio.Restriction.Restriction.BstZ17I-class.html#dna,Variable Bio.Restriction.Restriction.BstZI.dna=Bio.Restriction.Restriction.BstZI-class.html#dna,Variable Bio.Restriction.Restriction.Bsu15I.dna=Bio.Restriction.Restriction.Bsu15I-class.html#dna,Variable Bio.Restriction.Restriction.Bsu36I.dna=Bio.Restriction.Restriction.Bsu36I-class.html#dna,Variable Bio.Restriction.Restriction.BsuI.dna=Bio.Restriction.Restriction.BsuI-class.html#dna,Variable Bio.Restriction.Restriction.BsuRI.dna=Bio.Restriction.Restriction.BsuRI-class.html#dna,Variable Bio.Restriction.Restriction.BtgI.dna=Bio.Restriction.Restriction.BtgI-class.html#dna,Variable Bio.Restriction.Restriction.BtgZI.dna=Bio.Restriction.Restriction.BtgZI-class.html#dna,Variable Bio.Restriction.Restriction.BthCI.dna=Bio.Restriction.Restriction.BthCI-class.html#dna,Variable Bio.Restriction.Restriction.BtrI.dna=Bio.Restriction.Restriction.BtrI-class.html#dna,Variable Bio.Restriction.Restriction.BtsCI.dna=Bio.Restriction.Restriction.BtsCI-class.html#dna,Variable Bio.Restriction.Restriction.BtsI.dna=Bio.Restriction.Restriction.BtsI-class.html#dna,Variable Bio.Restriction.Restriction.BtsIMutI.dna=Bio.Restriction.Restriction.BtsIMutI-class.html#dna,Variable Bio.Restriction.Restriction.BtuMI.dna=Bio.Restriction.Restriction.BtuMI-class.html#dna,Variable Bio.Restriction.Restriction.BveI.dna=Bio.Restriction.Restriction.BveI-class.html#dna,Variable Bio.Restriction.Restriction.Cac8I.dna=Bio.Restriction.Restriction.Cac8I-class.html#dna,Variable Bio.Restriction.Restriction.CaiI.dna=Bio.Restriction.Restriction.CaiI-class.html#dna,Variable Bio.Restriction.Restriction.CauII.dna=Bio.Restriction.Restriction.CauII-class.html#dna,Variable Bio.Restriction.Restriction.CchII.dna=Bio.Restriction.Restriction.CchII-class.html#dna,Variable Bio.Restriction.Restriction.CchIII.dna=Bio.Restriction.Restriction.CchIII-class.html#dna,Variable Bio.Restriction.Restriction.CciI.dna=Bio.Restriction.Restriction.CciI-class.html#dna,Variable Bio.Restriction.Restriction.CciNI.dna=Bio.Restriction.Restriction.CciNI-class.html#dna,Variable Bio.Restriction.Restriction.Cdi630V.dna=Bio.Restriction.Restriction.Cdi630V-class.html#dna,Variable Bio.Restriction.Restriction.CdiI.dna=Bio.Restriction.Restriction.CdiI-class.html#dna,Variable Bio.Restriction.Restriction.CdpI.dna=Bio.Restriction.Restriction.CdpI-class.html#dna,Variable Bio.Restriction.Restriction.CfoI.dna=Bio.Restriction.Restriction.CfoI-class.html#dna,Variable Bio.Restriction.Restriction.Cfr10I.dna=Bio.Restriction.Restriction.Cfr10I-class.html#dna,Variable Bio.Restriction.Restriction.Cfr13I.dna=Bio.Restriction.Restriction.Cfr13I-class.html#dna,Variable Bio.Restriction.Restriction.Cfr42I.dna=Bio.Restriction.Restriction.Cfr42I-class.html#dna,Variable Bio.Restriction.Restriction.Cfr9I.dna=Bio.Restriction.Restriction.Cfr9I-class.html#dna,Variable Bio.Restriction.Restriction.CfrI.dna=Bio.Restriction.Restriction.CfrI-class.html#dna,Variable Bio.Restriction.Restriction.Cgl13032I.dna=Bio.Restriction.Restriction.Cgl13032I-class.html#dna,Variable Bio.Restriction.Restriction.Cgl13032II.dna=Bio.Restriction.Restriction.Cgl13032II-class.html#dna,Variable Bio.Restriction.Restriction.ChaI.dna=Bio.Restriction.Restriction.ChaI-class.html#dna,Variable Bio.Restriction.Restriction.CjeFIII.dna=Bio.Restriction.Restriction.CjeFIII-class.html#dna,Variable Bio.Restriction.Restriction.CjeFV.dna=Bio.Restriction.Restriction.CjeFV-class.html#dna,Variable Bio.Restriction.Restriction.CjeI.dna=Bio.Restriction.Restriction.CjeI-class.html#dna,Variable Bio.Restriction.Restriction.CjeNII.dna=Bio.Restriction.Restriction.CjeNII-class.html#dna,Variable Bio.Restriction.Restriction.CjeNIII.dna=Bio.Restriction.Restriction.CjeNIII-class.html#dna,Variable Bio.Restriction.Restriction.CjeP659IV.dna=Bio.Restriction.Restriction.CjeP659IV-class.html#dna,Variable Bio.Restriction.Restriction.CjePI.dna=Bio.Restriction.Restriction.CjePI-class.html#dna,Variable Bio.Restriction.Restriction.CjuI.dna=Bio.Restriction.Restriction.CjuI-class.html#dna,Variable Bio.Restriction.Restriction.CjuII.dna=Bio.Restriction.Restriction.CjuII-class.html#dna,Variable Bio.Restriction.Restriction.ClaI.dna=Bio.Restriction.Restriction.ClaI-class.html#dna,Variable Bio.Restriction.Restriction.CpoI.dna=Bio.Restriction.Restriction.CpoI-class.html#dna,Variable Bio.Restriction.Restriction.CseI.dna=Bio.Restriction.Restriction.CseI-class.html#dna,Variable Bio.Restriction.Restriction.CsiI.dna=Bio.Restriction.Restriction.CsiI-class.html#dna,Variable Bio.Restriction.Restriction.Csp6I.dna=Bio.Restriction.Restriction.Csp6I-class.html#dna,Variable Bio.Restriction.Restriction.CspAI.dna=Bio.Restriction.Restriction.CspAI-class.html#dna,Variable Bio.Restriction.Restriction.CspCI.dna=Bio.Restriction.Restriction.CspCI-class.html#dna,Variable Bio.Restriction.Restriction.CspI.dna=Bio.Restriction.Restriction.CspI-class.html#dna,Variable Bio.Restriction.Restriction.CstMI.dna=Bio.Restriction.Restriction.CstMI-class.html#dna,Variable Bio.Restriction.Restriction.CviAII.dna=Bio.Restriction.Restriction.CviAII-class.html#dna,Variable Bio.Restriction.Restriction.CviJI.dna=Bio.Restriction.Restriction.CviJI-class.html#dna,Variable Bio.Restriction.Restriction.CviKI_1.dna=Bio.Restriction.Restriction.CviKI_1-class.html#dna,Variable Bio.Restriction.Restriction.CviQI.dna=Bio.Restriction.Restriction.CviQI-class.html#dna,Variable Bio.Restriction.Restriction.CviRI.dna=Bio.Restriction.Restriction.CviRI-class.html#dna,Variable Bio.Restriction.Restriction.DdeI.dna=Bio.Restriction.Restriction.DdeI-class.html#dna,Variable Bio.Restriction.Restriction.DinI.dna=Bio.Restriction.Restriction.DinI-class.html#dna,Variable Bio.Restriction.Restriction.DpnI.dna=Bio.Restriction.Restriction.DpnI-class.html#dna,Variable Bio.Restriction.Restriction.DpnII.dna=Bio.Restriction.Restriction.DpnII-class.html#dna,Variable Bio.Restriction.Restriction.DraI.dna=Bio.Restriction.Restriction.DraI-class.html#dna,Variable Bio.Restriction.Restriction.DraII.dna=Bio.Restriction.Restriction.DraII-class.html#dna,Variable Bio.Restriction.Restriction.DraIII.dna=Bio.Restriction.Restriction.DraIII-class.html#dna,Variable Bio.Restriction.Restriction.DraRI.dna=Bio.Restriction.Restriction.DraRI-class.html#dna,Variable Bio.Restriction.Restriction.DrdI.dna=Bio.Restriction.Restriction.DrdI-class.html#dna,Variable Bio.Restriction.Restriction.DrdII.dna=Bio.Restriction.Restriction.DrdII-class.html#dna,Variable Bio.Restriction.Restriction.DriI.dna=Bio.Restriction.Restriction.DriI-class.html#dna,Variable Bio.Restriction.Restriction.DsaI.dna=Bio.Restriction.Restriction.DsaI-class.html#dna,Variable Bio.Restriction.Restriction.DseDI.dna=Bio.Restriction.Restriction.DseDI-class.html#dna,Variable Bio.Restriction.Restriction.EaeI.dna=Bio.Restriction.Restriction.EaeI-class.html#dna,Variable Bio.Restriction.Restriction.EagI.dna=Bio.Restriction.Restriction.EagI-class.html#dna,Variable Bio.Restriction.Restriction.Eam1104I.dna=Bio.Restriction.Restriction.Eam1104I-class.html#dna,Variable Bio.Restriction.Restriction.Eam1105I.dna=Bio.Restriction.Restriction.Eam1105I-class.html#dna,Variable Bio.Restriction.Restriction.EarI.dna=Bio.Restriction.Restriction.EarI-class.html#dna,Variable Bio.Restriction.Restriction.EciI.dna=Bio.Restriction.Restriction.EciI-class.html#dna,Variable Bio.Restriction.Restriction.Ecl136II.dna=Bio.Restriction.Restriction.Ecl136II-class.html#dna,Variable Bio.Restriction.Restriction.EclXI.dna=Bio.Restriction.Restriction.EclXI-class.html#dna,Variable Bio.Restriction.Restriction.Eco105I.dna=Bio.Restriction.Restriction.Eco105I-class.html#dna,Variable Bio.Restriction.Restriction.Eco130I.dna=Bio.Restriction.Restriction.Eco130I-class.html#dna,Variable Bio.Restriction.Restriction.Eco147I.dna=Bio.Restriction.Restriction.Eco147I-class.html#dna,Variable Bio.Restriction.Restriction.Eco24I.dna=Bio.Restriction.Restriction.Eco24I-class.html#dna,Variable Bio.Restriction.Restriction.Eco31I.dna=Bio.Restriction.Restriction.Eco31I-class.html#dna,Variable Bio.Restriction.Restriction.Eco32I.dna=Bio.Restriction.Restriction.Eco32I-class.html#dna,Variable Bio.Restriction.Restriction.Eco47I.dna=Bio.Restriction.Restriction.Eco47I-class.html#dna,Variable Bio.Restriction.Restriction.Eco47III.dna=Bio.Restriction.Restriction.Eco47III-class.html#dna,Variable Bio.Restriction.Restriction.Eco52I.dna=Bio.Restriction.Restriction.Eco52I-class.html#dna,Variable Bio.Restriction.Restriction.Eco53kI.dna=Bio.Restriction.Restriction.Eco53kI-class.html#dna,Variable Bio.Restriction.Restriction.Eco57I.dna=Bio.Restriction.Restriction.Eco57I-class.html#dna,Variable Bio.Restriction.Restriction.Eco57MI.dna=Bio.Restriction.Restriction.Eco57MI-class.html#dna,Variable Bio.Restriction.Restriction.Eco72I.dna=Bio.Restriction.Restriction.Eco72I-class.html#dna,Variable Bio.Restriction.Restriction.Eco81I.dna=Bio.Restriction.Restriction.Eco81I-class.html#dna,Variable Bio.Restriction.Restriction.Eco88I.dna=Bio.Restriction.Restriction.Eco88I-class.html#dna,Variable Bio.Restriction.Restriction.Eco91I.dna=Bio.Restriction.Restriction.Eco91I-class.html#dna,Variable Bio.Restriction.Restriction.EcoHI.dna=Bio.Restriction.Restriction.EcoHI-class.html#dna,Variable Bio.Restriction.Restriction.EcoICRI.dna=Bio.Restriction.Restriction.EcoICRI-class.html#dna,Variable Bio.Restriction.Restriction.EcoNI.dna=Bio.Restriction.Restriction.EcoNI-class.html#dna,Variable Bio.Restriction.Restriction.EcoO109I.dna=Bio.Restriction.Restriction.EcoO109I-class.html#dna,Variable Bio.Restriction.Restriction.EcoO65I.dna=Bio.Restriction.Restriction.EcoO65I-class.html#dna,Variable Bio.Restriction.Restriction.EcoRI.dna=Bio.Restriction.Restriction.EcoRI-class.html#dna,Variable Bio.Restriction.Restriction.EcoRII.dna=Bio.Restriction.Restriction.EcoRII-class.html#dna,Variable Bio.Restriction.Restriction.EcoRV.dna=Bio.Restriction.Restriction.EcoRV-class.html#dna,Variable Bio.Restriction.Restriction.EcoT14I.dna=Bio.Restriction.Restriction.EcoT14I-class.html#dna,Variable Bio.Restriction.Restriction.EcoT22I.dna=Bio.Restriction.Restriction.EcoT22I-class.html#dna,Variable Bio.Restriction.Restriction.EcoT38I.dna=Bio.Restriction.Restriction.EcoT38I-class.html#dna,Variable Bio.Restriction.Restriction.EgeI.dna=Bio.Restriction.Restriction.EgeI-class.html#dna,Variable Bio.Restriction.Restriction.EheI.dna=Bio.Restriction.Restriction.EheI-class.html#dna,Variable Bio.Restriction.Restriction.ErhI.dna=Bio.Restriction.Restriction.ErhI-class.html#dna,Variable Bio.Restriction.Restriction.EsaBC3I.dna=Bio.Restriction.Restriction.EsaBC3I-class.html#dna,Variable Bio.Restriction.Restriction.EsaSSI.dna=Bio.Restriction.Restriction.EsaSSI-class.html#dna,Variable Bio.Restriction.Restriction.Esp3I.dna=Bio.Restriction.Restriction.Esp3I-class.html#dna,Variable Bio.Restriction.Restriction.EspI.dna=Bio.Restriction.Restriction.EspI-class.html#dna,Variable Bio.Restriction.Restriction.FaeI.dna=Bio.Restriction.Restriction.FaeI-class.html#dna,Variable Bio.Restriction.Restriction.FaiI.dna=Bio.Restriction.Restriction.FaiI-class.html#dna,Variable Bio.Restriction.Restriction.FalI.dna=Bio.Restriction.Restriction.FalI-class.html#dna,Variable Bio.Restriction.Restriction.FaqI.dna=Bio.Restriction.Restriction.FaqI-class.html#dna,Variable Bio.Restriction.Restriction.FatI.dna=Bio.Restriction.Restriction.FatI-class.html#dna,Variable Bio.Restriction.Restriction.FauI.dna=Bio.Restriction.Restriction.FauI-class.html#dna,Variable Bio.Restriction.Restriction.FauNDI.dna=Bio.Restriction.Restriction.FauNDI-class.html#dna,Variable Bio.Restriction.Restriction.FbaI.dna=Bio.Restriction.Restriction.FbaI-class.html#dna,Variable Bio.Restriction.Restriction.FblI.dna=Bio.Restriction.Restriction.FblI-class.html#dna,Variable Bio.Restriction.Restriction.FinI.dna=Bio.Restriction.Restriction.FinI-class.html#dna,Variable Bio.Restriction.Restriction.FmuI.dna=Bio.Restriction.Restriction.FmuI-class.html#dna,Variable Bio.Restriction.Restriction.Fnu4HI.dna=Bio.Restriction.Restriction.Fnu4HI-class.html#dna,Variable Bio.Restriction.Restriction.FnuDII.dna=Bio.Restriction.Restriction.FnuDII-class.html#dna,Variable Bio.Restriction.Restriction.FokI.dna=Bio.Restriction.Restriction.FokI-class.html#dna,Variable Bio.Restriction.Restriction.FriOI.dna=Bio.Restriction.Restriction.FriOI-class.html#dna,Variable Bio.Restriction.Restriction.FseI.dna=Bio.Restriction.Restriction.FseI-class.html#dna,Variable Bio.Restriction.Restriction.Fsp4HI.dna=Bio.Restriction.Restriction.Fsp4HI-class.html#dna,Variable Bio.Restriction.Restriction.FspAI.dna=Bio.Restriction.Restriction.FspAI-class.html#dna,Variable Bio.Restriction.Restriction.FspBI.dna=Bio.Restriction.Restriction.FspBI-class.html#dna,Variable Bio.Restriction.Restriction.FspEI.dna=Bio.Restriction.Restriction.FspEI-class.html#dna,Variable Bio.Restriction.Restriction.FspI.dna=Bio.Restriction.Restriction.FspI-class.html#dna,Variable Bio.Restriction.Restriction.GauT27I.dna=Bio.Restriction.Restriction.GauT27I-class.html#dna,Variable Bio.Restriction.Restriction.GdiII.dna=Bio.Restriction.Restriction.GdiII-class.html#dna,Variable Bio.Restriction.Restriction.GlaI.dna=Bio.Restriction.Restriction.GlaI-class.html#dna,Variable Bio.Restriction.Restriction.GluI.dna=Bio.Restriction.Restriction.GluI-class.html#dna,Variable Bio.Restriction.Restriction.GsaI.dna=Bio.Restriction.Restriction.GsaI-class.html#dna,Variable Bio.Restriction.Restriction.GsuI.dna=Bio.Restriction.Restriction.GsuI-class.html#dna,Variable Bio.Restriction.Restriction.HaeI.dna=Bio.Restriction.Restriction.HaeI-class.html#dna,Variable Bio.Restriction.Restriction.HaeII.dna=Bio.Restriction.Restriction.HaeII-class.html#dna,Variable Bio.Restriction.Restriction.HaeIII.dna=Bio.Restriction.Restriction.HaeIII-class.html#dna,Variable Bio.Restriction.Restriction.HapII.dna=Bio.Restriction.Restriction.HapII-class.html#dna,Variable Bio.Restriction.Restriction.HauII.dna=Bio.Restriction.Restriction.HauII-class.html#dna,Variable Bio.Restriction.Restriction.HgaI.dna=Bio.Restriction.Restriction.HgaI-class.html#dna,Variable Bio.Restriction.Restriction.HgiAI.dna=Bio.Restriction.Restriction.HgiAI-class.html#dna,Variable Bio.Restriction.Restriction.HgiCI.dna=Bio.Restriction.Restriction.HgiCI-class.html#dna,Variable Bio.Restriction.Restriction.HgiEII.dna=Bio.Restriction.Restriction.HgiEII-class.html#dna,Variable Bio.Restriction.Restriction.HgiJII.dna=Bio.Restriction.Restriction.HgiJII-class.html#dna,Variable Bio.Restriction.Restriction.HhaI.dna=Bio.Restriction.Restriction.HhaI-class.html#dna,Variable Bio.Restriction.Restriction.Hin1I.dna=Bio.Restriction.Restriction.Hin1I-class.html#dna,Variable Bio.Restriction.Restriction.Hin1II.dna=Bio.Restriction.Restriction.Hin1II-class.html#dna,Variable Bio.Restriction.Restriction.Hin4I.dna=Bio.Restriction.Restriction.Hin4I-class.html#dna,Variable Bio.Restriction.Restriction.Hin4II.dna=Bio.Restriction.Restriction.Hin4II-class.html#dna,Variable Bio.Restriction.Restriction.Hin6I.dna=Bio.Restriction.Restriction.Hin6I-class.html#dna,Variable Bio.Restriction.Restriction.HinP1I.dna=Bio.Restriction.Restriction.HinP1I-class.html#dna,Variable Bio.Restriction.Restriction.HincII.dna=Bio.Restriction.Restriction.HincII-class.html#dna,Variable Bio.Restriction.Restriction.HindII.dna=Bio.Restriction.Restriction.HindII-class.html#dna,Variable Bio.Restriction.Restriction.HindIII.dna=Bio.Restriction.Restriction.HindIII-class.html#dna,Variable Bio.Restriction.Restriction.HinfI.dna=Bio.Restriction.Restriction.HinfI-class.html#dna,Variable Bio.Restriction.Restriction.HpaI.dna=Bio.Restriction.Restriction.HpaI-class.html#dna,Variable Bio.Restriction.Restriction.HpaII.dna=Bio.Restriction.Restriction.HpaII-class.html#dna,Variable Bio.Restriction.Restriction.HphI.dna=Bio.Restriction.Restriction.HphI-class.html#dna,Variable Bio.Restriction.Restriction.Hpy166II.dna=Bio.Restriction.Restriction.Hpy166II-class.html#dna,Variable Bio.Restriction.Restriction.Hpy178III.dna=Bio.Restriction.Restriction.Hpy178III-class.html#dna,Variable Bio.Restriction.Restriction.Hpy188I.dna=Bio.Restriction.Restriction.Hpy188I-class.html#dna,Variable Bio.Restriction.Restriction.Hpy188III.dna=Bio.Restriction.Restriction.Hpy188III-class.html#dna,Variable Bio.Restriction.Restriction.Hpy8I.dna=Bio.Restriction.Restriction.Hpy8I-class.html#dna,Variable Bio.Restriction.Restriction.Hpy99I.dna=Bio.Restriction.Restriction.Hpy99I-class.html#dna,Variable Bio.Restriction.Restriction.Hpy99XIII.dna=Bio.Restriction.Restriction.Hpy99XIII-class.html#dna,Variable Bio.Restriction.Restriction.Hpy99XIV.dna=Bio.Restriction.Restriction.Hpy99XIV-class.html#dna,Variable Bio.Restriction.Restriction.HpyAV.dna=Bio.Restriction.Restriction.HpyAV-class.html#dna,Variable Bio.Restriction.Restriction.HpyCH4III.dna=Bio.Restriction.Restriction.HpyCH4III-class.html#dna,Variable Bio.Restriction.Restriction.HpyCH4IV.dna=Bio.Restriction.Restriction.HpyCH4IV-class.html#dna,Variable Bio.Restriction.Restriction.HpyCH4V.dna=Bio.Restriction.Restriction.HpyCH4V-class.html#dna,Variable Bio.Restriction.Restriction.HpyF10VI.dna=Bio.Restriction.Restriction.HpyF10VI-class.html#dna,Variable Bio.Restriction.Restriction.HpyF3I.dna=Bio.Restriction.Restriction.HpyF3I-class.html#dna,Variable Bio.Restriction.Restriction.HpySE526I.dna=Bio.Restriction.Restriction.HpySE526I-class.html#dna,Variable Bio.Restriction.Restriction.Hsp92I.dna=Bio.Restriction.Restriction.Hsp92I-class.html#dna,Variable Bio.Restriction.Restriction.Hsp92II.dna=Bio.Restriction.Restriction.Hsp92II-class.html#dna,Variable Bio.Restriction.Restriction.HspAI.dna=Bio.Restriction.Restriction.HspAI-class.html#dna,Variable Bio.Restriction.Restriction.Jma19592I.dna=Bio.Restriction.Restriction.Jma19592I-class.html#dna,Variable Bio.Restriction.Restriction.KasI.dna=Bio.Restriction.Restriction.KasI-class.html#dna,Variable Bio.Restriction.Restriction.KflI.dna=Bio.Restriction.Restriction.KflI-class.html#dna,Variable Bio.Restriction.Restriction.Kpn2I.dna=Bio.Restriction.Restriction.Kpn2I-class.html#dna,Variable Bio.Restriction.Restriction.KpnI.dna=Bio.Restriction.Restriction.KpnI-class.html#dna,Variable Bio.Restriction.Restriction.KroI.dna=Bio.Restriction.Restriction.KroI-class.html#dna,Variable Bio.Restriction.Restriction.Ksp22I.dna=Bio.Restriction.Restriction.Ksp22I-class.html#dna,Variable Bio.Restriction.Restriction.Ksp632I.dna=Bio.Restriction.Restriction.Ksp632I-class.html#dna,Variable Bio.Restriction.Restriction.KspAI.dna=Bio.Restriction.Restriction.KspAI-class.html#dna,Variable Bio.Restriction.Restriction.KspI.dna=Bio.Restriction.Restriction.KspI-class.html#dna,Variable Bio.Restriction.Restriction.Kzo9I.dna=Bio.Restriction.Restriction.Kzo9I-class.html#dna,Variable Bio.Restriction.Restriction.LguI.dna=Bio.Restriction.Restriction.LguI-class.html#dna,Variable Bio.Restriction.Restriction.LpnI.dna=Bio.Restriction.Restriction.LpnI-class.html#dna,Variable Bio.Restriction.Restriction.LpnPI.dna=Bio.Restriction.Restriction.LpnPI-class.html#dna,Variable Bio.Restriction.Restriction.Lsp1109I.dna=Bio.Restriction.Restriction.Lsp1109I-class.html#dna,Variable Bio.Restriction.Restriction.LweI.dna=Bio.Restriction.Restriction.LweI-class.html#dna,Variable Bio.Restriction.Restriction.MabI.dna=Bio.Restriction.Restriction.MabI-class.html#dna,Variable Bio.Restriction.Restriction.MaeI.dna=Bio.Restriction.Restriction.MaeI-class.html#dna,Variable Bio.Restriction.Restriction.MaeII.dna=Bio.Restriction.Restriction.MaeII-class.html#dna,Variable Bio.Restriction.Restriction.MaeIII.dna=Bio.Restriction.Restriction.MaeIII-class.html#dna,Variable Bio.Restriction.Restriction.MalI.dna=Bio.Restriction.Restriction.MalI-class.html#dna,Variable Bio.Restriction.Restriction.MaqI.dna=Bio.Restriction.Restriction.MaqI-class.html#dna,Variable Bio.Restriction.Restriction.MauBI.dna=Bio.Restriction.Restriction.MauBI-class.html#dna,Variable Bio.Restriction.Restriction.MbiI.dna=Bio.Restriction.Restriction.MbiI-class.html#dna,Variable Bio.Restriction.Restriction.MboI.dna=Bio.Restriction.Restriction.MboI-class.html#dna,Variable Bio.Restriction.Restriction.MboII.dna=Bio.Restriction.Restriction.MboII-class.html#dna,Variable Bio.Restriction.Restriction.McaTI.dna=Bio.Restriction.Restriction.McaTI-class.html#dna,Variable Bio.Restriction.Restriction.McrI.dna=Bio.Restriction.Restriction.McrI-class.html#dna,Variable Bio.Restriction.Restriction.MfeI.dna=Bio.Restriction.Restriction.MfeI-class.html#dna,Variable Bio.Restriction.Restriction.MflI.dna=Bio.Restriction.Restriction.MflI-class.html#dna,Variable Bio.Restriction.Restriction.MhlI.dna=Bio.Restriction.Restriction.MhlI-class.html#dna,Variable Bio.Restriction.Restriction.MjaIV.dna=Bio.Restriction.Restriction.MjaIV-class.html#dna,Variable Bio.Restriction.Restriction.MkaDII.dna=Bio.Restriction.Restriction.MkaDII-class.html#dna,Variable Bio.Restriction.Restriction.MlsI.dna=Bio.Restriction.Restriction.MlsI-class.html#dna,Variable Bio.Restriction.Restriction.MluCI.dna=Bio.Restriction.Restriction.MluCI-class.html#dna,Variable Bio.Restriction.Restriction.MluI.dna=Bio.Restriction.Restriction.MluI-class.html#dna,Variable Bio.Restriction.Restriction.MluNI.dna=Bio.Restriction.Restriction.MluNI-class.html#dna,Variable Bio.Restriction.Restriction.Mly113I.dna=Bio.Restriction.Restriction.Mly113I-class.html#dna,Variable Bio.Restriction.Restriction.MlyI.dna=Bio.Restriction.Restriction.MlyI-class.html#dna,Variable Bio.Restriction.Restriction.MmeI.dna=Bio.Restriction.Restriction.MmeI-class.html#dna,Variable Bio.Restriction.Restriction.MnlI.dna=Bio.Restriction.Restriction.MnlI-class.html#dna,Variable Bio.Restriction.Restriction.Mph1103I.dna=Bio.Restriction.Restriction.Mph1103I-class.html#dna,Variable Bio.Restriction.Restriction.MreI.dna=Bio.Restriction.Restriction.MreI-class.html#dna,Variable Bio.Restriction.Restriction.MroI.dna=Bio.Restriction.Restriction.MroI-class.html#dna,Variable Bio.Restriction.Restriction.MroNI.dna=Bio.Restriction.Restriction.MroNI-class.html#dna,Variable Bio.Restriction.Restriction.MroXI.dna=Bio.Restriction.Restriction.MroXI-class.html#dna,Variable Bio.Restriction.Restriction.MscI.dna=Bio.Restriction.Restriction.MscI-class.html#dna,Variable Bio.Restriction.Restriction.MseI.dna=Bio.Restriction.Restriction.MseI-class.html#dna,Variable Bio.Restriction.Restriction.MslI.dna=Bio.Restriction.Restriction.MslI-class.html#dna,Variable Bio.Restriction.Restriction.Msp20I.dna=Bio.Restriction.Restriction.Msp20I-class.html#dna,Variable Bio.Restriction.Restriction.MspA1I.dna=Bio.Restriction.Restriction.MspA1I-class.html#dna,Variable Bio.Restriction.Restriction.MspCI.dna=Bio.Restriction.Restriction.MspCI-class.html#dna,Variable Bio.Restriction.Restriction.MspI.dna=Bio.Restriction.Restriction.MspI-class.html#dna,Variable Bio.Restriction.Restriction.MspJI.dna=Bio.Restriction.Restriction.MspJI-class.html#dna,Variable Bio.Restriction.Restriction.MspR9I.dna=Bio.Restriction.Restriction.MspR9I-class.html#dna,Variable Bio.Restriction.Restriction.MssI.dna=Bio.Restriction.Restriction.MssI-class.html#dna,Variable Bio.Restriction.Restriction.MstI.dna=Bio.Restriction.Restriction.MstI-class.html#dna,Variable Bio.Restriction.Restriction.MunI.dna=Bio.Restriction.Restriction.MunI-class.html#dna,Variable Bio.Restriction.Restriction.Mva1269I.dna=Bio.Restriction.Restriction.Mva1269I-class.html#dna,Variable Bio.Restriction.Restriction.MvaI.dna=Bio.Restriction.Restriction.MvaI-class.html#dna,Variable Bio.Restriction.Restriction.MvnI.dna=Bio.Restriction.Restriction.MvnI-class.html#dna,Variable Bio.Restriction.Restriction.MvrI.dna=Bio.Restriction.Restriction.MvrI-class.html#dna,Variable Bio.Restriction.Restriction.MwoI.dna=Bio.Restriction.Restriction.MwoI-class.html#dna,Variable Bio.Restriction.Restriction.NaeI.dna=Bio.Restriction.Restriction.NaeI-class.html#dna,Variable Bio.Restriction.Restriction.NarI.dna=Bio.Restriction.Restriction.NarI-class.html#dna,Variable Bio.Restriction.Restriction.NciI.dna=Bio.Restriction.Restriction.NciI-class.html#dna,Variable Bio.Restriction.Restriction.NcoI.dna=Bio.Restriction.Restriction.NcoI-class.html#dna,Variable Bio.Restriction.Restriction.NdeI.dna=Bio.Restriction.Restriction.NdeI-class.html#dna,Variable Bio.Restriction.Restriction.NdeII.dna=Bio.Restriction.Restriction.NdeII-class.html#dna,Variable Bio.Restriction.Restriction.NgoAVIII.dna=Bio.Restriction.Restriction.NgoAVIII-class.html#dna,Variable Bio.Restriction.Restriction.NgoMIV.dna=Bio.Restriction.Restriction.NgoMIV-class.html#dna,Variable Bio.Restriction.Restriction.NhaXI.dna=Bio.Restriction.Restriction.NhaXI-class.html#dna,Variable Bio.Restriction.Restriction.NheI.dna=Bio.Restriction.Restriction.NheI-class.html#dna,Variable Bio.Restriction.Restriction.NlaCI.dna=Bio.Restriction.Restriction.NlaCI-class.html#dna,Variable Bio.Restriction.Restriction.NlaIII.dna=Bio.Restriction.Restriction.NlaIII-class.html#dna,Variable Bio.Restriction.Restriction.NlaIV.dna=Bio.Restriction.Restriction.NlaIV-class.html#dna,Variable Bio.Restriction.Restriction.Nli3877I.dna=Bio.Restriction.Restriction.Nli3877I-class.html#dna,Variable Bio.Restriction.Restriction.NmeAIII.dna=Bio.Restriction.Restriction.NmeAIII-class.html#dna,Variable Bio.Restriction.Restriction.NmeDI.dna=Bio.Restriction.Restriction.NmeDI-class.html#dna,Variable Bio.Restriction.Restriction.NmuCI.dna=Bio.Restriction.Restriction.NmuCI-class.html#dna,Variable Bio.Restriction.Restriction.NotI.dna=Bio.Restriction.Restriction.NotI-class.html#dna,Variable Bio.Restriction.Restriction.NruI.dna=Bio.Restriction.Restriction.NruI-class.html#dna,Variable Bio.Restriction.Restriction.NsbI.dna=Bio.Restriction.Restriction.NsbI-class.html#dna,Variable Bio.Restriction.Restriction.NsiI.dna=Bio.Restriction.Restriction.NsiI-class.html#dna,Variable Bio.Restriction.Restriction.NspBII.dna=Bio.Restriction.Restriction.NspBII-class.html#dna,Variable Bio.Restriction.Restriction.NspI.dna=Bio.Restriction.Restriction.NspI-class.html#dna,Variable Bio.Restriction.Restriction.NspV.dna=Bio.Restriction.Restriction.NspV-class.html#dna,Variable Bio.Restriction.Restriction.OliI.dna=Bio.Restriction.Restriction.OliI-class.html#dna,Variable Bio.Restriction.Restriction.PabI.dna=Bio.Restriction.Restriction.PabI-class.html#dna,Variable Bio.Restriction.Restriction.PacI.dna=Bio.Restriction.Restriction.PacI-class.html#dna,Variable Bio.Restriction.Restriction.PaeI.dna=Bio.Restriction.Restriction.PaeI-class.html#dna,Variable Bio.Restriction.Restriction.PaeR7I.dna=Bio.Restriction.Restriction.PaeR7I-class.html#dna,Variable Bio.Restriction.Restriction.PagI.dna=Bio.Restriction.Restriction.PagI-class.html#dna,Variable Bio.Restriction.Restriction.PalAI.dna=Bio.Restriction.Restriction.PalAI-class.html#dna,Variable Bio.Restriction.Restriction.PasI.dna=Bio.Restriction.Restriction.PasI-class.html#dna,Variable Bio.Restriction.Restriction.PauI.dna=Bio.Restriction.Restriction.PauI-class.html#dna,Variable Bio.Restriction.Restriction.PceI.dna=Bio.Restriction.Restriction.PceI-class.html#dna,Variable Bio.Restriction.Restriction.PciI.dna=Bio.Restriction.Restriction.PciI-class.html#dna,Variable Bio.Restriction.Restriction.PciSI.dna=Bio.Restriction.Restriction.PciSI-class.html#dna,Variable Bio.Restriction.Restriction.PcsI.dna=Bio.Restriction.Restriction.PcsI-class.html#dna,Variable Bio.Restriction.Restriction.PctI.dna=Bio.Restriction.Restriction.PctI-class.html#dna,Variable Bio.Restriction.Restriction.PdiI.dna=Bio.Restriction.Restriction.PdiI-class.html#dna,Variable Bio.Restriction.Restriction.PdmI.dna=Bio.Restriction.Restriction.PdmI-class.html#dna,Variable Bio.Restriction.Restriction.PenI.dna=Bio.Restriction.Restriction.PenI-class.html#dna,Variable Bio.Restriction.Restriction.PfeI.dna=Bio.Restriction.Restriction.PfeI-class.html#dna,Variable Bio.Restriction.Restriction.Pfl1108I.dna=Bio.Restriction.Restriction.Pfl1108I-class.html#dna,Variable Bio.Restriction.Restriction.Pfl23II.dna=Bio.Restriction.Restriction.Pfl23II-class.html#dna,Variable Bio.Restriction.Restriction.PflFI.dna=Bio.Restriction.Restriction.PflFI-class.html#dna,Variable Bio.Restriction.Restriction.PflMI.dna=Bio.Restriction.Restriction.PflMI-class.html#dna,Variable Bio.Restriction.Restriction.PfoI.dna=Bio.Restriction.Restriction.PfoI-class.html#dna,Variable Bio.Restriction.Restriction.PinAI.dna=Bio.Restriction.Restriction.PinAI-class.html#dna,Variable Bio.Restriction.Restriction.PlaDI.dna=Bio.Restriction.Restriction.PlaDI-class.html#dna,Variable Bio.Restriction.Restriction.Ple19I.dna=Bio.Restriction.Restriction.Ple19I-class.html#dna,Variable Bio.Restriction.Restriction.PleI.dna=Bio.Restriction.Restriction.PleI-class.html#dna,Variable Bio.Restriction.Restriction.PluTI.dna=Bio.Restriction.Restriction.PluTI-class.html#dna,Variable Bio.Restriction.Restriction.PmaCI.dna=Bio.Restriction.Restriction.PmaCI-class.html#dna,Variable Bio.Restriction.Restriction.PmeI.dna=Bio.Restriction.Restriction.PmeI-class.html#dna,Variable Bio.Restriction.Restriction.PmlI.dna=Bio.Restriction.Restriction.PmlI-class.html#dna,Variable Bio.Restriction.Restriction.PpiI.dna=Bio.Restriction.Restriction.PpiI-class.html#dna,Variable Bio.Restriction.Restriction.PpsI.dna=Bio.Restriction.Restriction.PpsI-class.html#dna,Variable Bio.Restriction.Restriction.Ppu10I.dna=Bio.Restriction.Restriction.Ppu10I-class.html#dna,Variable Bio.Restriction.Restriction.Ppu21I.dna=Bio.Restriction.Restriction.Ppu21I-class.html#dna,Variable Bio.Restriction.Restriction.PpuMI.dna=Bio.Restriction.Restriction.PpuMI-class.html#dna,Variable Bio.Restriction.Restriction.PscI.dna=Bio.Restriction.Restriction.PscI-class.html#dna,Variable Bio.Restriction.Restriction.PshAI.dna=Bio.Restriction.Restriction.PshAI-class.html#dna,Variable Bio.Restriction.Restriction.PshBI.dna=Bio.Restriction.Restriction.PshBI-class.html#dna,Variable Bio.Restriction.Restriction.PsiI.dna=Bio.Restriction.Restriction.PsiI-class.html#dna,Variable Bio.Restriction.Restriction.Psp03I.dna=Bio.Restriction.Restriction.Psp03I-class.html#dna,Variable Bio.Restriction.Restriction.Psp124BI.dna=Bio.Restriction.Restriction.Psp124BI-class.html#dna,Variable Bio.Restriction.Restriction.Psp1406I.dna=Bio.Restriction.Restriction.Psp1406I-class.html#dna,Variable Bio.Restriction.Restriction.Psp5II.dna=Bio.Restriction.Restriction.Psp5II-class.html#dna,Variable Bio.Restriction.Restriction.Psp6I.dna=Bio.Restriction.Restriction.Psp6I-class.html#dna,Variable Bio.Restriction.Restriction.PspCI.dna=Bio.Restriction.Restriction.PspCI-class.html#dna,Variable Bio.Restriction.Restriction.PspEI.dna=Bio.Restriction.Restriction.PspEI-class.html#dna,Variable Bio.Restriction.Restriction.PspGI.dna=Bio.Restriction.Restriction.PspGI-class.html#dna,Variable Bio.Restriction.Restriction.PspLI.dna=Bio.Restriction.Restriction.PspLI-class.html#dna,Variable Bio.Restriction.Restriction.PspN4I.dna=Bio.Restriction.Restriction.PspN4I-class.html#dna,Variable Bio.Restriction.Restriction.PspOMI.dna=Bio.Restriction.Restriction.PspOMI-class.html#dna,Variable Bio.Restriction.Restriction.PspOMII.dna=Bio.Restriction.Restriction.PspOMII-class.html#dna,Variable Bio.Restriction.Restriction.PspPI.dna=Bio.Restriction.Restriction.PspPI-class.html#dna,Variable Bio.Restriction.Restriction.PspPPI.dna=Bio.Restriction.Restriction.PspPPI-class.html#dna,Variable Bio.Restriction.Restriction.PspPRI.dna=Bio.Restriction.Restriction.PspPRI-class.html#dna,Variable Bio.Restriction.Restriction.PspXI.dna=Bio.Restriction.Restriction.PspXI-class.html#dna,Variable Bio.Restriction.Restriction.PsrI.dna=Bio.Restriction.Restriction.PsrI-class.html#dna,Variable Bio.Restriction.Restriction.PssI.dna=Bio.Restriction.Restriction.PssI-class.html#dna,Variable Bio.Restriction.Restriction.PstI.dna=Bio.Restriction.Restriction.PstI-class.html#dna,Variable Bio.Restriction.Restriction.PstNI.dna=Bio.Restriction.Restriction.PstNI-class.html#dna,Variable Bio.Restriction.Restriction.PsuI.dna=Bio.Restriction.Restriction.PsuI-class.html#dna,Variable Bio.Restriction.Restriction.PsyI.dna=Bio.Restriction.Restriction.PsyI-class.html#dna,Variable Bio.Restriction.Restriction.PteI.dna=Bio.Restriction.Restriction.PteI-class.html#dna,Variable Bio.Restriction.Restriction.PvuI.dna=Bio.Restriction.Restriction.PvuI-class.html#dna,Variable Bio.Restriction.Restriction.PvuII.dna=Bio.Restriction.Restriction.PvuII-class.html#dna,Variable Bio.Restriction.Restriction.R2_BceSIV.dna=Bio.Restriction.Restriction.R2_BceSIV-class.html#dna,Variable Bio.Restriction.Restriction.RceI.dna=Bio.Restriction.Restriction.RceI-class.html#dna,Variable Bio.Restriction.Restriction.RdeGBI.dna=Bio.Restriction.Restriction.RdeGBI-class.html#dna,Variable Bio.Restriction.Restriction.RdeGBII.dna=Bio.Restriction.Restriction.RdeGBII-class.html#dna,Variable Bio.Restriction.Restriction.RdeGBIII.dna=Bio.Restriction.Restriction.RdeGBIII-class.html#dna,Variable Bio.Restriction.Restriction.RflFIII.dna=Bio.Restriction.Restriction.RflFIII-class.html#dna,Variable Bio.Restriction.Restriction.RgaI.dna=Bio.Restriction.Restriction.RgaI-class.html#dna,Variable Bio.Restriction.Restriction.RigI.dna=Bio.Restriction.Restriction.RigI-class.html#dna,Variable Bio.Restriction.Restriction.RlaI.dna=Bio.Restriction.Restriction.RlaI-class.html#dna,Variable Bio.Restriction.Restriction.RleAI.dna=Bio.Restriction.Restriction.RleAI-class.html#dna,Variable Bio.Restriction.Restriction.RpaB5I.dna=Bio.Restriction.Restriction.RpaB5I-class.html#dna,Variable Bio.Restriction.Restriction.RpaBI.dna=Bio.Restriction.Restriction.RpaBI-class.html#dna,Variable Bio.Restriction.Restriction.RpaI.dna=Bio.Restriction.Restriction.RpaI-class.html#dna,Variable Bio.Restriction.Restriction.RpaTI.dna=Bio.Restriction.Restriction.RpaTI-class.html#dna,Variable Bio.Restriction.Restriction.RruI.dna=Bio.Restriction.Restriction.RruI-class.html#dna,Variable Bio.Restriction.Restriction.RsaI.dna=Bio.Restriction.Restriction.RsaI-class.html#dna,Variable Bio.Restriction.Restriction.RsaNI.dna=Bio.Restriction.Restriction.RsaNI-class.html#dna,Variable Bio.Restriction.Restriction.RseI.dna=Bio.Restriction.Restriction.RseI-class.html#dna,Variable Bio.Restriction.Restriction.Rsr2I.dna=Bio.Restriction.Restriction.Rsr2I-class.html#dna,Variable Bio.Restriction.Restriction.RsrII.dna=Bio.Restriction.Restriction.RsrII-class.html#dna,Variable Bio.Restriction.Restriction.SacI.dna=Bio.Restriction.Restriction.SacI-class.html#dna,Variable Bio.Restriction.Restriction.SacII.dna=Bio.Restriction.Restriction.SacII-class.html#dna,Variable Bio.Restriction.Restriction.SalI.dna=Bio.Restriction.Restriction.SalI-class.html#dna,Variable Bio.Restriction.Restriction.SanDI.dna=Bio.Restriction.Restriction.SanDI-class.html#dna,Variable Bio.Restriction.Restriction.SapI.dna=Bio.Restriction.Restriction.SapI-class.html#dna,Variable Bio.Restriction.Restriction.SaqAI.dna=Bio.Restriction.Restriction.SaqAI-class.html#dna,Variable Bio.Restriction.Restriction.SatI.dna=Bio.Restriction.Restriction.SatI-class.html#dna,Variable Bio.Restriction.Restriction.Sau3AI.dna=Bio.Restriction.Restriction.Sau3AI-class.html#dna,Variable Bio.Restriction.Restriction.Sau96I.dna=Bio.Restriction.Restriction.Sau96I-class.html#dna,Variable Bio.Restriction.Restriction.SauI.dna=Bio.Restriction.Restriction.SauI-class.html#dna,Variable Bio.Restriction.Restriction.SbfI.dna=Bio.Restriction.Restriction.SbfI-class.html#dna,Variable Bio.Restriction.Restriction.ScaI.dna=Bio.Restriction.Restriction.ScaI-class.html#dna,Variable Bio.Restriction.Restriction.SchI.dna=Bio.Restriction.Restriction.SchI-class.html#dna,Variable Bio.Restriction.Restriction.SciI.dna=Bio.Restriction.Restriction.SciI-class.html#dna,Variable Bio.Restriction.Restriction.ScrFI.dna=Bio.Restriction.Restriction.ScrFI-class.html#dna,Variable Bio.Restriction.Restriction.SdaI.dna=Bio.Restriction.Restriction.SdaI-class.html#dna,Variable Bio.Restriction.Restriction.SdeAI.dna=Bio.Restriction.Restriction.SdeAI-class.html#dna,Variable Bio.Restriction.Restriction.SdeOSI.dna=Bio.Restriction.Restriction.SdeOSI-class.html#dna,Variable Bio.Restriction.Restriction.SduI.dna=Bio.Restriction.Restriction.SduI-class.html#dna,Variable Bio.Restriction.Restriction.SecI.dna=Bio.Restriction.Restriction.SecI-class.html#dna,Variable Bio.Restriction.Restriction.SelI.dna=Bio.Restriction.Restriction.SelI-class.html#dna,Variable Bio.Restriction.Restriction.SetI.dna=Bio.Restriction.Restriction.SetI-class.html#dna,Variable Bio.Restriction.Restriction.SexAI.dna=Bio.Restriction.Restriction.SexAI-class.html#dna,Variable Bio.Restriction.Restriction.SfaAI.dna=Bio.Restriction.Restriction.SfaAI-class.html#dna,Variable Bio.Restriction.Restriction.SfaNI.dna=Bio.Restriction.Restriction.SfaNI-class.html#dna,Variable Bio.Restriction.Restriction.SfcI.dna=Bio.Restriction.Restriction.SfcI-class.html#dna,Variable Bio.Restriction.Restriction.SfeI.dna=Bio.Restriction.Restriction.SfeI-class.html#dna,Variable Bio.Restriction.Restriction.SfiI.dna=Bio.Restriction.Restriction.SfiI-class.html#dna,Variable Bio.Restriction.Restriction.SfoI.dna=Bio.Restriction.Restriction.SfoI-class.html#dna,Variable Bio.Restriction.Restriction.Sfr274I.dna=Bio.Restriction.Restriction.Sfr274I-class.html#dna,Variable Bio.Restriction.Restriction.Sfr303I.dna=Bio.Restriction.Restriction.Sfr303I-class.html#dna,Variable Bio.Restriction.Restriction.SfuI.dna=Bio.Restriction.Restriction.SfuI-class.html#dna,Variable Bio.Restriction.Restriction.SgeI.dna=Bio.Restriction.Restriction.SgeI-class.html#dna,Variable Bio.Restriction.Restriction.SgfI.dna=Bio.Restriction.Restriction.SgfI-class.html#dna,Variable Bio.Restriction.Restriction.SgrAI.dna=Bio.Restriction.Restriction.SgrAI-class.html#dna,Variable Bio.Restriction.Restriction.SgrBI.dna=Bio.Restriction.Restriction.SgrBI-class.html#dna,Variable Bio.Restriction.Restriction.SgrDI.dna=Bio.Restriction.Restriction.SgrDI-class.html#dna,Variable Bio.Restriction.Restriction.SgrTI.dna=Bio.Restriction.Restriction.SgrTI-class.html#dna,Variable Bio.Restriction.Restriction.SgsI.dna=Bio.Restriction.Restriction.SgsI-class.html#dna,Variable Bio.Restriction.Restriction.SimI.dna=Bio.Restriction.Restriction.SimI-class.html#dna,Variable Bio.Restriction.Restriction.SlaI.dna=Bio.Restriction.Restriction.SlaI-class.html#dna,Variable Bio.Restriction.Restriction.SmaI.dna=Bio.Restriction.Restriction.SmaI-class.html#dna,Variable Bio.Restriction.Restriction.SmiI.dna=Bio.Restriction.Restriction.SmiI-class.html#dna,Variable Bio.Restriction.Restriction.SmiMI.dna=Bio.Restriction.Restriction.SmiMI-class.html#dna,Variable Bio.Restriction.Restriction.SmlI.dna=Bio.Restriction.Restriction.SmlI-class.html#dna,Variable Bio.Restriction.Restriction.SmoI.dna=Bio.Restriction.Restriction.SmoI-class.html#dna,Variable Bio.Restriction.Restriction.SnaBI.dna=Bio.Restriction.Restriction.SnaBI-class.html#dna,Variable Bio.Restriction.Restriction.SnaI.dna=Bio.Restriction.Restriction.SnaI-class.html#dna,Variable Bio.Restriction.Restriction.Sno506I.dna=Bio.Restriction.Restriction.Sno506I-class.html#dna,Variable Bio.Restriction.Restriction.SpeI.dna=Bio.Restriction.Restriction.SpeI-class.html#dna,Variable Bio.Restriction.Restriction.SphI.dna=Bio.Restriction.Restriction.SphI-class.html#dna,Variable Bio.Restriction.Restriction.SplI.dna=Bio.Restriction.Restriction.SplI-class.html#dna,Variable Bio.Restriction.Restriction.SpoDI.dna=Bio.Restriction.Restriction.SpoDI-class.html#dna,Variable Bio.Restriction.Restriction.SrfI.dna=Bio.Restriction.Restriction.SrfI-class.html#dna,Variable Bio.Restriction.Restriction.Sse232I.dna=Bio.Restriction.Restriction.Sse232I-class.html#dna,Variable Bio.Restriction.Restriction.Sse8387I.dna=Bio.Restriction.Restriction.Sse8387I-class.html#dna,Variable Bio.Restriction.Restriction.Sse8647I.dna=Bio.Restriction.Restriction.Sse8647I-class.html#dna,Variable Bio.Restriction.Restriction.Sse9I.dna=Bio.Restriction.Restriction.Sse9I-class.html#dna,Variable Bio.Restriction.Restriction.SseBI.dna=Bio.Restriction.Restriction.SseBI-class.html#dna,Variable Bio.Restriction.Restriction.SsiI.dna=Bio.Restriction.Restriction.SsiI-class.html#dna,Variable Bio.Restriction.Restriction.SspD5I.dna=Bio.Restriction.Restriction.SspD5I-class.html#dna,Variable Bio.Restriction.Restriction.SspDI.dna=Bio.Restriction.Restriction.SspDI-class.html#dna,Variable Bio.Restriction.Restriction.SspI.dna=Bio.Restriction.Restriction.SspI-class.html#dna,Variable Bio.Restriction.Restriction.SstE37I.dna=Bio.Restriction.Restriction.SstE37I-class.html#dna,Variable Bio.Restriction.Restriction.SstI.dna=Bio.Restriction.Restriction.SstI-class.html#dna,Variable Bio.Restriction.Restriction.Sth132I.dna=Bio.Restriction.Restriction.Sth132I-class.html#dna,Variable Bio.Restriction.Restriction.Sth302II.dna=Bio.Restriction.Restriction.Sth302II-class.html#dna,Variable Bio.Restriction.Restriction.StrI.dna=Bio.Restriction.Restriction.StrI-class.html#dna,Variable Bio.Restriction.Restriction.StsI.dna=Bio.Restriction.Restriction.StsI-class.html#dna,Variable Bio.Restriction.Restriction.StuI.dna=Bio.Restriction.Restriction.StuI-class.html#dna,Variable Bio.Restriction.Restriction.StyD4I.dna=Bio.Restriction.Restriction.StyD4I-class.html#dna,Variable Bio.Restriction.Restriction.StyI.dna=Bio.Restriction.Restriction.StyI-class.html#dna,Variable Bio.Restriction.Restriction.SwaI.dna=Bio.Restriction.Restriction.SwaI-class.html#dna,Variable Bio.Restriction.Restriction.TaaI.dna=Bio.Restriction.Restriction.TaaI-class.html#dna,Variable Bio.Restriction.Restriction.TaiI.dna=Bio.Restriction.Restriction.TaiI-class.html#dna,Variable Bio.Restriction.Restriction.TaqI.dna=Bio.Restriction.Restriction.TaqI-class.html#dna,Variable Bio.Restriction.Restriction.TaqII.dna=Bio.Restriction.Restriction.TaqII-class.html#dna,Variable Bio.Restriction.Restriction.TasI.dna=Bio.Restriction.Restriction.TasI-class.html#dna,Variable Bio.Restriction.Restriction.TatI.dna=Bio.Restriction.Restriction.TatI-class.html#dna,Variable Bio.Restriction.Restriction.TauI.dna=Bio.Restriction.Restriction.TauI-class.html#dna,Variable Bio.Restriction.Restriction.TfiI.dna=Bio.Restriction.Restriction.TfiI-class.html#dna,Variable Bio.Restriction.Restriction.Tru1I.dna=Bio.Restriction.Restriction.Tru1I-class.html#dna,Variable Bio.Restriction.Restriction.Tru9I.dna=Bio.Restriction.Restriction.Tru9I-class.html#dna,Variable Bio.Restriction.Restriction.TscAI.dna=Bio.Restriction.Restriction.TscAI-class.html#dna,Variable Bio.Restriction.Restriction.TseFI.dna=Bio.Restriction.Restriction.TseFI-class.html#dna,Variable Bio.Restriction.Restriction.TseI.dna=Bio.Restriction.Restriction.TseI-class.html#dna,Variable Bio.Restriction.Restriction.TsoI.dna=Bio.Restriction.Restriction.TsoI-class.html#dna,Variable Bio.Restriction.Restriction.Tsp45I.dna=Bio.Restriction.Restriction.Tsp45I-class.html#dna,Variable Bio.Restriction.Restriction.Tsp4CI.dna=Bio.Restriction.Restriction.Tsp4CI-class.html#dna,Variable Bio.Restriction.Restriction.TspDTI.dna=Bio.Restriction.Restriction.TspDTI-class.html#dna,Variable Bio.Restriction.Restriction.TspEI.dna=Bio.Restriction.Restriction.TspEI-class.html#dna,Variable Bio.Restriction.Restriction.TspGWI.dna=Bio.Restriction.Restriction.TspGWI-class.html#dna,Variable Bio.Restriction.Restriction.TspMI.dna=Bio.Restriction.Restriction.TspMI-class.html#dna,Variable Bio.Restriction.Restriction.TspRI.dna=Bio.Restriction.Restriction.TspRI-class.html#dna,Variable Bio.Restriction.Restriction.TssI.dna=Bio.Restriction.Restriction.TssI-class.html#dna,Variable Bio.Restriction.Restriction.TstI.dna=Bio.Restriction.Restriction.TstI-class.html#dna,Variable Bio.Restriction.Restriction.TsuI.dna=Bio.Restriction.Restriction.TsuI-class.html#dna,Variable Bio.Restriction.Restriction.Tth111I.dna=Bio.Restriction.Restriction.Tth111I-class.html#dna,Variable Bio.Restriction.Restriction.Tth111II.dna=Bio.Restriction.Restriction.Tth111II-class.html#dna,Variable Bio.Restriction.Restriction.UbaF11I.dna=Bio.Restriction.Restriction.UbaF11I-class.html#dna,Variable Bio.Restriction.Restriction.UbaF12I.dna=Bio.Restriction.Restriction.UbaF12I-class.html#dna,Variable Bio.Restriction.Restriction.UbaF13I.dna=Bio.Restriction.Restriction.UbaF13I-class.html#dna,Variable Bio.Restriction.Restriction.UbaF14I.dna=Bio.Restriction.Restriction.UbaF14I-class.html#dna,Variable Bio.Restriction.Restriction.UbaF9I.dna=Bio.Restriction.Restriction.UbaF9I-class.html#dna,Variable Bio.Restriction.Restriction.UbaPI.dna=Bio.Restriction.Restriction.UbaPI-class.html#dna,Variable Bio.Restriction.Restriction.UcoMSI.dna=Bio.Restriction.Restriction.UcoMSI-class.html#dna,Variable Bio.Restriction.Restriction.UnbI.dna=Bio.Restriction.Restriction.UnbI-class.html#dna,Variable Bio.Restriction.Restriction.Van91I.dna=Bio.Restriction.Restriction.Van91I-class.html#dna,Variable Bio.Restriction.Restriction.Vha464I.dna=Bio.Restriction.Restriction.Vha464I-class.html#dna,Variable Bio.Restriction.Restriction.VneI.dna=Bio.Restriction.Restriction.VneI-class.html#dna,Variable Bio.Restriction.Restriction.VpaK11AI.dna=Bio.Restriction.Restriction.VpaK11AI-class.html#dna,Variable Bio.Restriction.Restriction.VpaK11BI.dna=Bio.Restriction.Restriction.VpaK11BI-class.html#dna,Variable Bio.Restriction.Restriction.VspI.dna=Bio.Restriction.Restriction.VspI-class.html#dna,Variable Bio.Restriction.Restriction.WviI.dna=Bio.Restriction.Restriction.WviI-class.html#dna,Variable Bio.Restriction.Restriction.XagI.dna=Bio.Restriction.Restriction.XagI-class.html#dna,Variable Bio.Restriction.Restriction.XapI.dna=Bio.Restriction.Restriction.XapI-class.html#dna,Variable Bio.Restriction.Restriction.XbaI.dna=Bio.Restriction.Restriction.XbaI-class.html#dna,Variable Bio.Restriction.Restriction.XceI.dna=Bio.Restriction.Restriction.XceI-class.html#dna,Variable Bio.Restriction.Restriction.XcmI.dna=Bio.Restriction.Restriction.XcmI-class.html#dna,Variable Bio.Restriction.Restriction.XhoI.dna=Bio.Restriction.Restriction.XhoI-class.html#dna,Variable Bio.Restriction.Restriction.XhoII.dna=Bio.Restriction.Restriction.XhoII-class.html#dna,Variable Bio.Restriction.Restriction.XmaI.dna=Bio.Restriction.Restriction.XmaI-class.html#dna,Variable Bio.Restriction.Restriction.XmaIII.dna=Bio.Restriction.Restriction.XmaIII-class.html#dna,Variable Bio.Restriction.Restriction.XmaJI.dna=Bio.Restriction.Restriction.XmaJI-class.html#dna,Variable Bio.Restriction.Restriction.XmiI.dna=Bio.Restriction.Restriction.XmiI-class.html#dna,Variable Bio.Restriction.Restriction.XmnI.dna=Bio.Restriction.Restriction.XmnI-class.html#dna,Variable Bio.Restriction.Restriction.XspI.dna=Bio.Restriction.Restriction.XspI-class.html#dna,Variable Bio.Restriction.Restriction.YkrI.dna=Bio.Restriction.Restriction.YkrI-class.html#dna,Variable Bio.Restriction.Restriction.ZraI.dna=Bio.Restriction.Restriction.ZraI-class.html#dna,Variable Bio.Restriction.Restriction.ZrmI.dna=Bio.Restriction.Restriction.ZrmI-class.html#dna,Variable Bio.Restriction.Restriction.Zsp2I.dna=Bio.Restriction.Restriction.Zsp2I-class.html#dna"><a title="Bio.Restriction.Restriction.AanI.dna
Bio.Restriction.Restriction.AarI.dna
Bio.Restriction.Restriction.AasI.dna
Bio.Restriction.Restriction.AatII.dna
Bio.Restriction.Restriction.AbaSI.dna
Bio.Restriction.Restriction.AbsI.dna
Bio.Restriction.Restriction.Acc16I.dna
Bio.Restriction.Restriction.Acc36I.dna
Bio.Restriction.Restriction.Acc65I.dna
Bio.Restriction.Restriction.AccB1I.dna
Bio.Restriction.Restriction.AccB7I.dna
Bio.Restriction.Restriction.AccBSI.dna
Bio.Restriction.Restriction.AccI.dna
Bio.Restriction.Restriction.AccII.dna
Bio.Restriction.Restriction.AccIII.dna
Bio.Restriction.Restriction.AceIII.dna
Bio.Restriction.Restriction.AciI.dna
Bio.Restriction.Restriction.AclI.dna
Bio.Restriction.Restriction.AclWI.dna
Bio.Restriction.Restriction.AcoI.dna
Bio.Restriction.Restriction.AcsI.dna
Bio.Restriction.Restriction.AcuI.dna
Bio.Restriction.Restriction.AcvI.dna
Bio.Restriction.Restriction.AcyI.dna
Bio.Restriction.Restriction.AdeI.dna
Bio.Restriction.Restriction.AfaI.dna
Bio.Restriction.Restriction.AfeI.dna
Bio.Restriction.Restriction.AfiI.dna
Bio.Restriction.Restriction.AflII.dna
Bio.Restriction.Restriction.AflIII.dna
Bio.Restriction.Restriction.AgeI.dna
Bio.Restriction.Restriction.AgsI.dna
Bio.Restriction.Restriction.AhaIII.dna
Bio.Restriction.Restriction.AhdI.dna
Bio.Restriction.Restriction.AhlI.dna
Bio.Restriction.Restriction.AjiI.dna
Bio.Restriction.Restriction.AjnI.dna
Bio.Restriction.Restriction.AjuI.dna
Bio.Restriction.Restriction.AleI.dna
Bio.Restriction.Restriction.AlfI.dna
Bio.Restriction.Restriction.AloI.dna
Bio.Restriction.Restriction.AluBI.dna
Bio.Restriction.Restriction.AluI.dna
Bio.Restriction.Restriction.Alw21I.dna
Bio.Restriction.Restriction.Alw26I.dna
Bio.Restriction.Restriction.Alw44I.dna
Bio.Restriction.Restriction.AlwFI.dna
Bio.Restriction.Restriction.AlwI.dna
Bio.Restriction.Restriction.AlwNI.dna
Bio.Restriction.Restriction.Ama87I.dna
Bio.Restriction.Restriction.Aor13HI.dna
Bio.Restriction.Restriction.Aor51HI.dna
Bio.Restriction.Restriction.AoxI.dna
Bio.Restriction.Restriction.ApaBI.dna
Bio.Restriction.Restriction.ApaI.dna
Bio.Restriction.Restriction.ApaLI.dna
Bio.Restriction.Restriction.ApeKI.dna
Bio.Restriction.Restriction.ApoI.dna
Bio.Restriction.Restriction.ApyPI.dna
Bio.Restriction.Restriction.AquII.dna
Bio.Restriction.Restriction.AquIII.dna
Bio.Restriction.Restriction.AquIV.dna
Bio.Restriction.Restriction.ArsI.dna
Bio.Restriction.Restriction.AscI.dna
Bio.Restriction.Restriction.AseI.dna
Bio.Restriction.Restriction.Asi256I.dna
Bio.Restriction.Restriction.AsiGI.dna
Bio.Restriction.Restriction.AsiSI.dna
Bio.Restriction.Restriction.Asp700I.dna
Bio.Restriction.Restriction.Asp718I.dna
Bio.Restriction.Restriction.AspA2I.dna
Bio.Restriction.Restriction.AspBHI.dna
Bio.Restriction.Restriction.AspLEI.dna
Bio.Restriction.Restriction.AspS9I.dna
Bio.Restriction.Restriction.AssI.dna
Bio.Restriction.Restriction.AsuC2I.dna
Bio.Restriction.Restriction.AsuHPI.dna
Bio.Restriction.Restriction.AsuI.dna
Bio.Restriction.Restriction.AsuII.dna
Bio.Restriction.Restriction.AsuNHI.dna
Bio.Restriction.Restriction.AvaI.dna
Bio.Restriction.Restriction.AvaII.dna
Bio.Restriction.Restriction.AvaIII.dna
Bio.Restriction.Restriction.AvrII.dna
Bio.Restriction.Restriction.AxyI.dna
Bio.Restriction.Restriction.BaeGI.dna
Bio.Restriction.Restriction.BaeI.dna
Bio.Restriction.Restriction.BalI.dna
Bio.Restriction.Restriction.BamHI.dna
Bio.Restriction.Restriction.BanI.dna
Bio.Restriction.Restriction.BanII.dna
Bio.Restriction.Restriction.BarI.dna
Bio.Restriction.Restriction.BasI.dna
Bio.Restriction.Restriction.BauI.dna
Bio.Restriction.Restriction.Bbr7I.dna
Bio.Restriction.Restriction.BbrPI.dna
Bio.Restriction.Restriction.BbsI.dna
Bio.Restriction.Restriction.Bbv12I.dna
Bio.Restriction.Restriction.BbvCI.dna
Bio.Restriction.Restriction.BbvI.dna
Bio.Restriction.Restriction.BbvII.dna
Bio.Restriction.Restriction.BccI.dna
Bio.Restriction.Restriction.Bce83I.dna
Bio.Restriction.Restriction.BceAI.dna
Bio.Restriction.Restriction.BcefI.dna
Bio.Restriction.Restriction.BcgI.dna
Bio.Restriction.Restriction.BciT130I.dna
Bio.Restriction.Restriction.BciVI.dna
Bio.Restriction.Restriction.BclI.dna
Bio.Restriction.Restriction.BcnI.dna
Bio.Restriction.Restriction.BcoDI.dna
Bio.Restriction.Restriction.BcuI.dna
Bio.Restriction.Restriction.BdaI.dna
Bio.Restriction.Restriction.BetI.dna
Bio.Restriction.Restriction.BfaI.dna
Bio.Restriction.Restriction.BfiI.dna
Bio.Restriction.Restriction.BfmI.dna
Bio.Restriction.Restriction.BfoI.dna
Bio.Restriction.Restriction.BfrI.dna
Bio.Restriction.Restriction.BfuAI.dna
Bio.Restriction.Restriction.BfuCI.dna
Bio.Restriction.Restriction.BfuI.dna
Bio.Restriction.Restriction.BglI.dna
Bio.Restriction.Restriction.BglII.dna
Bio.Restriction.Restriction.BinI.dna
Bio.Restriction.Restriction.BisI.dna
Bio.Restriction.Restriction.BlnI.dna
Bio.Restriction.Restriction.BlpI.dna
Bio.Restriction.Restriction.BlsI.dna
Bio.Restriction.Restriction.BmcAI.dna
Bio.Restriction.Restriction.Bme1390I.dna
Bio.Restriction.Restriction.Bme18I.dna
Bio.Restriction.Restriction.BmeDI.dna
Bio.Restriction.Restriction.BmeRI.dna
Bio.Restriction.Restriction.BmeT110I.dna
Bio.Restriction.Restriction.BmgBI.dna
Bio.Restriction.Restriction.BmgI.dna
Bio.Restriction.Restriction.BmgT120I.dna
Bio.Restriction.Restriction.BmiI.dna
Bio.Restriction.Restriction.BmrFI.dna
Bio.Restriction.Restriction.BmrI.dna
Bio.Restriction.Restriction.BmsI.dna
Bio.Restriction.Restriction.BmtI.dna
Bio.Restriction.Restriction.BmuI.dna
Bio.Restriction.Restriction.BoxI.dna
Bio.Restriction.Restriction.BpiI.dna
Bio.Restriction.Restriction.BplI.dna
Bio.Restriction.Restriction.BpmI.dna
Bio.Restriction.Restriction.Bpu10I.dna
Bio.Restriction.Restriction.Bpu1102I.dna
Bio.Restriction.Restriction.Bpu14I.dna
Bio.Restriction.Restriction.BpuEI.dna
Bio.Restriction.Restriction.BpuMI.dna
Bio.Restriction.Restriction.BpvUI.dna
Bio.Restriction.Restriction.Bsa29I.dna
Bio.Restriction.Restriction.BsaAI.dna
Bio.Restriction.Restriction.BsaBI.dna
Bio.Restriction.Restriction.BsaHI.dna
Bio.Restriction.Restriction.BsaI.dna
Bio.Restriction.Restriction.BsaJI.dna
Bio.Restriction.Restriction.BsaWI.dna
Bio.Restriction.Restriction.BsaXI.dna
Bio.Restriction.Restriction.BsbI.dna
Bio.Restriction.Restriction.Bsc4I.dna
Bio.Restriction.Restriction.BscAI.dna
Bio.Restriction.Restriction.BscGI.dna
Bio.Restriction.Restriction.Bse118I.dna
Bio.Restriction.Restriction.Bse1I.dna
Bio.Restriction.Restriction.Bse21I.dna
Bio.Restriction.Restriction.Bse3DI.dna
Bio.Restriction.Restriction.Bse8I.dna
Bio.Restriction.Restriction.BseAI.dna
Bio.Restriction.Restriction.BseBI.dna
Bio.Restriction.Restriction.BseCI.dna
Bio.Restriction.Restriction.BseDI.dna
Bio.Restriction.Restriction.BseGI.dna
Bio.Restriction.Restriction.BseJI.dna
Bio.Restriction.Restriction.BseLI.dna
Bio.Restriction.Restriction.BseMI.dna
Bio.Restriction.Restriction.BseMII.dna
Bio.Restriction.Restriction.BseNI.dna
Bio.Restriction.Restriction.BsePI.dna
Bio.Restriction.Restriction.BseRI.dna
Bio.Restriction.Restriction.BseSI.dna
Bio.Restriction.Restriction.BseX3I.dna
Bio.Restriction.Restriction.BseXI.dna
Bio.Restriction.Restriction.BseYI.dna
Bio.Restriction.Restriction.BsgI.dna
Bio.Restriction.Restriction.Bsh1236I.dna
Bio.Restriction.Restriction.Bsh1285I.dna
Bio.Restriction.Restriction.BshFI.dna
Bio.Restriction.Restriction.BshNI.dna
Bio.Restriction.Restriction.BshTI.dna
Bio.Restriction.Restriction.BshVI.dna
Bio.Restriction.Restriction.BsiEI.dna
Bio.Restriction.Restriction.BsiHKAI.dna
Bio.Restriction.Restriction.BsiHKCI.dna
Bio.Restriction.Restriction.BsiI.dna
Bio.Restriction.Restriction.BsiSI.dna
Bio.Restriction.Restriction.BsiWI.dna
Bio.Restriction.Restriction.BsiYI.dna
Bio.Restriction.Restriction.BslFI.dna
Bio.Restriction.Restriction.BslI.dna
Bio.Restriction.Restriction.BsmAI.dna
Bio.Restriction.Restriction.BsmBI.dna
Bio.Restriction.Restriction.BsmFI.dna
Bio.Restriction.Restriction.BsmI.dna
Bio.Restriction.Restriction.BsnI.dna
Bio.Restriction.Restriction.Bso31I.dna
Bio.Restriction.Restriction.BsoBI.dna
Bio.Restriction.Restriction.Bsp119I.dna
Bio.Restriction.Restriction.Bsp120I.dna
Bio.Restriction.Restriction.Bsp1286I.dna
Bio.Restriction.Restriction.Bsp13I.dna
Bio.Restriction.Restriction.Bsp1407I.dna
Bio.Restriction.Restriction.Bsp143I.dna
Bio.Restriction.Restriction.Bsp1720I.dna
Bio.Restriction.Restriction.Bsp19I.dna
Bio.Restriction.Restriction.Bsp24I.dna
Bio.Restriction.Restriction.Bsp68I.dna
Bio.Restriction.Restriction.BspACI.dna
Bio.Restriction.Restriction.BspCNI.dna
Bio.Restriction.Restriction.BspD6I.dna
Bio.Restriction.Restriction.BspDI.dna
Bio.Restriction.Restriction.BspEI.dna
Bio.Restriction.Restriction.BspFNI.dna
Bio.Restriction.Restriction.BspGI.dna
Bio.Restriction.Restriction.BspHI.dna
Bio.Restriction.Restriction.BspLI.dna
Bio.Restriction.Restriction.BspLU11I.dna
Bio.Restriction.Restriction.BspMI.dna
Bio.Restriction.Restriction.BspMII.dna
Bio.Restriction.Restriction.BspNCI.dna
Bio.Restriction.Restriction.BspOI.dna
Bio.Restriction.Restriction.BspPI.dna
Bio.Restriction.Restriction.BspQI.dna
Bio.Restriction.Restriction.BspT104I.dna
Bio.Restriction.Restriction.BspT107I.dna
Bio.Restriction.Restriction.BspTI.dna
Bio.Restriction.Restriction.BsrBI.dna
Bio.Restriction.Restriction.BsrDI.dna
Bio.Restriction.Restriction.BsrFI.dna
Bio.Restriction.Restriction.BsrGI.dna
Bio.Restriction.Restriction.BsrI.dna
Bio.Restriction.Restriction.BsrSI.dna
Bio.Restriction.Restriction.BssAI.dna
Bio.Restriction.Restriction.BssECI.dna
Bio.Restriction.Restriction.BssHII.dna
Bio.Restriction.Restriction.BssKI.dna
Bio.Restriction.Restriction.BssMI.dna
Bio.Restriction.Restriction.BssNAI.dna
Bio.Restriction.Restriction.BssNI.dna
Bio.Restriction.Restriction.BssSI.dna
Bio.Restriction.Restriction.BssT1I.dna
Bio.Restriction.Restriction.Bst1107I.dna
Bio.Restriction.Restriction.Bst2BI.dna
Bio.Restriction.Restriction.Bst2UI.dna
Bio.Restriction.Restriction.Bst4CI.dna
Bio.Restriction.Restriction.Bst6I.dna
Bio.Restriction.Restriction.BstACI.dna
Bio.Restriction.Restriction.BstAFI.dna
Bio.Restriction.Restriction.BstAPI.dna
Bio.Restriction.Restriction.BstAUI.dna
Bio.Restriction.Restriction.BstBAI.dna
Bio.Restriction.Restriction.BstBI.dna
Bio.Restriction.Restriction.BstC8I.dna
Bio.Restriction.Restriction.BstDEI.dna
Bio.Restriction.Restriction.BstDSI.dna
Bio.Restriction.Restriction.BstEII.dna
Bio.Restriction.Restriction.BstENI.dna
Bio.Restriction.Restriction.BstF5I.dna
Bio.Restriction.Restriction.BstFNI.dna
Bio.Restriction.Restriction.BstH2I.dna
Bio.Restriction.Restriction.BstHHI.dna
Bio.Restriction.Restriction.BstKTI.dna
Bio.Restriction.Restriction.BstMAI.dna
Bio.Restriction.Restriction.BstMBI.dna
Bio.Restriction.Restriction.BstMCI.dna
Bio.Restriction.Restriction.BstMWI.dna
Bio.Restriction.Restriction.BstNI.dna
Bio.Restriction.Restriction.BstNSI.dna
Bio.Restriction.Restriction.BstOI.dna
Bio.Restriction.Restriction.BstPAI.dna
Bio.Restriction.Restriction.BstPI.dna
Bio.Restriction.Restriction.BstSCI.dna
Bio.Restriction.Restriction.BstSFI.dna
Bio.Restriction.Restriction.BstSLI.dna
Bio.Restriction.Restriction.BstSNI.dna
Bio.Restriction.Restriction.BstUI.dna
Bio.Restriction.Restriction.BstV1I.dna
Bio.Restriction.Restriction.BstV2I.dna
Bio.Restriction.Restriction.BstX2I.dna
Bio.Restriction.Restriction.BstXI.dna
Bio.Restriction.Restriction.BstYI.dna
Bio.Restriction.Restriction.BstZ17I.dna
Bio.Restriction.Restriction.BstZI.dna
Bio.Restriction.Restriction.Bsu15I.dna
Bio.Restriction.Restriction.Bsu36I.dna
Bio.Restriction.Restriction.BsuI.dna
Bio.Restriction.Restriction.BsuRI.dna
Bio.Restriction.Restriction.BtgI.dna
Bio.Restriction.Restriction.BtgZI.dna
Bio.Restriction.Restriction.BthCI.dna
Bio.Restriction.Restriction.BtrI.dna
Bio.Restriction.Restriction.BtsCI.dna
Bio.Restriction.Restriction.BtsI.dna
Bio.Restriction.Restriction.BtsIMutI.dna
Bio.Restriction.Restriction.BtuMI.dna
Bio.Restriction.Restriction.BveI.dna
Bio.Restriction.Restriction.Cac8I.dna
Bio.Restriction.Restriction.CaiI.dna
Bio.Restriction.Restriction.CauII.dna
Bio.Restriction.Restriction.CchII.dna
Bio.Restriction.Restriction.CchIII.dna
Bio.Restriction.Restriction.CciI.dna
Bio.Restriction.Restriction.CciNI.dna
Bio.Restriction.Restriction.Cdi630V.dna
Bio.Restriction.Restriction.CdiI.dna
Bio.Restriction.Restriction.CdpI.dna
Bio.Restriction.Restriction.CfoI.dna
Bio.Restriction.Restriction.Cfr10I.dna
Bio.Restriction.Restriction.Cfr13I.dna
Bio.Restriction.Restriction.Cfr42I.dna
Bio.Restriction.Restriction.Cfr9I.dna
Bio.Restriction.Restriction.CfrI.dna
Bio.Restriction.Restriction.Cgl13032I.dna
Bio.Restriction.Restriction.Cgl13032II.dna
Bio.Restriction.Restriction.ChaI.dna
Bio.Restriction.Restriction.CjeFIII.dna
Bio.Restriction.Restriction.CjeFV.dna
Bio.Restriction.Restriction.CjeI.dna
Bio.Restriction.Restriction.CjeNII.dna
Bio.Restriction.Restriction.CjeNIII.dna
Bio.Restriction.Restriction.CjeP659IV.dna
Bio.Restriction.Restriction.CjePI.dna
Bio.Restriction.Restriction.CjuI.dna
Bio.Restriction.Restriction.CjuII.dna
Bio.Restriction.Restriction.ClaI.dna
Bio.Restriction.Restriction.CpoI.dna
Bio.Restriction.Restriction.CseI.dna
Bio.Restriction.Restriction.CsiI.dna
Bio.Restriction.Restriction.Csp6I.dna
Bio.Restriction.Restriction.CspAI.dna
Bio.Restriction.Restriction.CspCI.dna
Bio.Restriction.Restriction.CspI.dna
Bio.Restriction.Restriction.CstMI.dna
Bio.Restriction.Restriction.CviAII.dna
Bio.Restriction.Restriction.CviJI.dna
Bio.Restriction.Restriction.CviKI_1.dna
Bio.Restriction.Restriction.CviQI.dna
Bio.Restriction.Restriction.CviRI.dna
Bio.Restriction.Restriction.DdeI.dna
Bio.Restriction.Restriction.DinI.dna
Bio.Restriction.Restriction.DpnI.dna
Bio.Restriction.Restriction.DpnII.dna
Bio.Restriction.Restriction.DraI.dna
Bio.Restriction.Restriction.DraII.dna
Bio.Restriction.Restriction.DraIII.dna
Bio.Restriction.Restriction.DraRI.dna
Bio.Restriction.Restriction.DrdI.dna
Bio.Restriction.Restriction.DrdII.dna
Bio.Restriction.Restriction.DriI.dna
Bio.Restriction.Restriction.DsaI.dna
Bio.Restriction.Restriction.DseDI.dna
Bio.Restriction.Restriction.EaeI.dna
Bio.Restriction.Restriction.EagI.dna
Bio.Restriction.Restriction.Eam1104I.dna
Bio.Restriction.Restriction.Eam1105I.dna
Bio.Restriction.Restriction.EarI.dna
Bio.Restriction.Restriction.EciI.dna
Bio.Restriction.Restriction.Ecl136II.dna
Bio.Restriction.Restriction.EclXI.dna
Bio.Restriction.Restriction.Eco105I.dna
Bio.Restriction.Restriction.Eco130I.dna
Bio.Restriction.Restriction.Eco147I.dna
Bio.Restriction.Restriction.Eco24I.dna
Bio.Restriction.Restriction.Eco31I.dna
Bio.Restriction.Restriction.Eco32I.dna
Bio.Restriction.Restriction.Eco47I.dna
Bio.Restriction.Restriction.Eco47III.dna
Bio.Restriction.Restriction.Eco52I.dna
Bio.Restriction.Restriction.Eco53kI.dna
Bio.Restriction.Restriction.Eco57I.dna
Bio.Restriction.Restriction.Eco57MI.dna
Bio.Restriction.Restriction.Eco72I.dna
Bio.Restriction.Restriction.Eco81I.dna
Bio.Restriction.Restriction.Eco88I.dna
Bio.Restriction.Restriction.Eco91I.dna
Bio.Restriction.Restriction.EcoHI.dna
Bio.Restriction.Restriction.EcoICRI.dna
Bio.Restriction.Restriction.EcoNI.dna
Bio.Restriction.Restriction.EcoO109I.dna
Bio.Restriction.Restriction.EcoO65I.dna
Bio.Restriction.Restriction.EcoRI.dna
Bio.Restriction.Restriction.EcoRII.dna
Bio.Restriction.Restriction.EcoRV.dna
Bio.Restriction.Restriction.EcoT14I.dna
Bio.Restriction.Restriction.EcoT22I.dna
Bio.Restriction.Restriction.EcoT38I.dna
Bio.Restriction.Restriction.EgeI.dna
Bio.Restriction.Restriction.EheI.dna
Bio.Restriction.Restriction.ErhI.dna
Bio.Restriction.Restriction.EsaBC3I.dna
Bio.Restriction.Restriction.EsaSSI.dna
Bio.Restriction.Restriction.Esp3I.dna
Bio.Restriction.Restriction.EspI.dna
Bio.Restriction.Restriction.FaeI.dna
Bio.Restriction.Restriction.FaiI.dna
Bio.Restriction.Restriction.FalI.dna
Bio.Restriction.Restriction.FaqI.dna
Bio.Restriction.Restriction.FatI.dna
Bio.Restriction.Restriction.FauI.dna
Bio.Restriction.Restriction.FauNDI.dna
Bio.Restriction.Restriction.FbaI.dna
Bio.Restriction.Restriction.FblI.dna
Bio.Restriction.Restriction.FinI.dna
Bio.Restriction.Restriction.FmuI.dna
Bio.Restriction.Restriction.Fnu4HI.dna
Bio.Restriction.Restriction.FnuDII.dna
Bio.Restriction.Restriction.FokI.dna
Bio.Restriction.Restriction.FriOI.dna
Bio.Restriction.Restriction.FseI.dna
Bio.Restriction.Restriction.Fsp4HI.dna
Bio.Restriction.Restriction.FspAI.dna
Bio.Restriction.Restriction.FspBI.dna
Bio.Restriction.Restriction.FspEI.dna
Bio.Restriction.Restriction.FspI.dna
Bio.Restriction.Restriction.GauT27I.dna
Bio.Restriction.Restriction.GdiII.dna
Bio.Restriction.Restriction.GlaI.dna
Bio.Restriction.Restriction.GluI.dna
Bio.Restriction.Restriction.GsaI.dna
Bio.Restriction.Restriction.GsuI.dna
Bio.Restriction.Restriction.HaeI.dna
Bio.Restriction.Restriction.HaeII.dna
Bio.Restriction.Restriction.HaeIII.dna
Bio.Restriction.Restriction.HapII.dna
Bio.Restriction.Restriction.HauII.dna
Bio.Restriction.Restriction.HgaI.dna
Bio.Restriction.Restriction.HgiAI.dna
Bio.Restriction.Restriction.HgiCI.dna
Bio.Restriction.Restriction.HgiEII.dna
Bio.Restriction.Restriction.HgiJII.dna
Bio.Restriction.Restriction.HhaI.dna
Bio.Restriction.Restriction.Hin1I.dna
Bio.Restriction.Restriction.Hin1II.dna
Bio.Restriction.Restriction.Hin4I.dna
Bio.Restriction.Restriction.Hin4II.dna
Bio.Restriction.Restriction.Hin6I.dna
Bio.Restriction.Restriction.HinP1I.dna
Bio.Restriction.Restriction.HincII.dna
Bio.Restriction.Restriction.HindII.dna
Bio.Restriction.Restriction.HindIII.dna
Bio.Restriction.Restriction.HinfI.dna
Bio.Restriction.Restriction.HpaI.dna
Bio.Restriction.Restriction.HpaII.dna
Bio.Restriction.Restriction.HphI.dna
Bio.Restriction.Restriction.Hpy166II.dna
Bio.Restriction.Restriction.Hpy178III.dna
Bio.Restriction.Restriction.Hpy188I.dna
Bio.Restriction.Restriction.Hpy188III.dna
Bio.Restriction.Restriction.Hpy8I.dna
Bio.Restriction.Restriction.Hpy99I.dna
Bio.Restriction.Restriction.Hpy99XIII.dna
Bio.Restriction.Restriction.Hpy99XIV.dna
Bio.Restriction.Restriction.HpyAV.dna
Bio.Restriction.Restriction.HpyCH4III.dna
Bio.Restriction.Restriction.HpyCH4IV.dna
Bio.Restriction.Restriction.HpyCH4V.dna
Bio.Restriction.Restriction.HpyF10VI.dna
Bio.Restriction.Restriction.HpyF3I.dna
Bio.Restriction.Restriction.HpySE526I.dna
Bio.Restriction.Restriction.Hsp92I.dna
Bio.Restriction.Restriction.Hsp92II.dna
Bio.Restriction.Restriction.HspAI.dna
Bio.Restriction.Restriction.Jma19592I.dna
Bio.Restriction.Restriction.KasI.dna
Bio.Restriction.Restriction.KflI.dna
Bio.Restriction.Restriction.Kpn2I.dna
Bio.Restriction.Restriction.KpnI.dna
Bio.Restriction.Restriction.KroI.dna
Bio.Restriction.Restriction.Ksp22I.dna
Bio.Restriction.Restriction.Ksp632I.dna
Bio.Restriction.Restriction.KspAI.dna
Bio.Restriction.Restriction.KspI.dna
Bio.Restriction.Restriction.Kzo9I.dna
Bio.Restriction.Restriction.LguI.dna
Bio.Restriction.Restriction.LpnI.dna
Bio.Restriction.Restriction.LpnPI.dna
Bio.Restriction.Restriction.Lsp1109I.dna
Bio.Restriction.Restriction.LweI.dna
Bio.Restriction.Restriction.MabI.dna
Bio.Restriction.Restriction.MaeI.dna
Bio.Restriction.Restriction.MaeII.dna
Bio.Restriction.Restriction.MaeIII.dna
Bio.Restriction.Restriction.MalI.dna
Bio.Restriction.Restriction.MaqI.dna
Bio.Restriction.Restriction.MauBI.dna
Bio.Restriction.Restriction.MbiI.dna
Bio.Restriction.Restriction.MboI.dna
Bio.Restriction.Restriction.MboII.dna
Bio.Restriction.Restriction.McaTI.dna
Bio.Restriction.Restriction.McrI.dna
Bio.Restriction.Restriction.MfeI.dna
Bio.Restriction.Restriction.MflI.dna
Bio.Restriction.Restriction.MhlI.dna
Bio.Restriction.Restriction.MjaIV.dna
Bio.Restriction.Restriction.MkaDII.dna
Bio.Restriction.Restriction.MlsI.dna
Bio.Restriction.Restriction.MluCI.dna
Bio.Restriction.Restriction.MluI.dna
Bio.Restriction.Restriction.MluNI.dna
Bio.Restriction.Restriction.Mly113I.dna
Bio.Restriction.Restriction.MlyI.dna
Bio.Restriction.Restriction.MmeI.dna
Bio.Restriction.Restriction.MnlI.dna
Bio.Restriction.Restriction.Mph1103I.dna
Bio.Restriction.Restriction.MreI.dna
Bio.Restriction.Restriction.MroI.dna
Bio.Restriction.Restriction.MroNI.dna
Bio.Restriction.Restriction.MroXI.dna
Bio.Restriction.Restriction.MscI.dna
Bio.Restriction.Restriction.MseI.dna
Bio.Restriction.Restriction.MslI.dna
Bio.Restriction.Restriction.Msp20I.dna
Bio.Restriction.Restriction.MspA1I.dna
Bio.Restriction.Restriction.MspCI.dna
Bio.Restriction.Restriction.MspI.dna
Bio.Restriction.Restriction.MspJI.dna
Bio.Restriction.Restriction.MspR9I.dna
Bio.Restriction.Restriction.MssI.dna
Bio.Restriction.Restriction.MstI.dna
Bio.Restriction.Restriction.MunI.dna
Bio.Restriction.Restriction.Mva1269I.dna
Bio.Restriction.Restriction.MvaI.dna
Bio.Restriction.Restriction.MvnI.dna
Bio.Restriction.Restriction.MvrI.dna
Bio.Restriction.Restriction.MwoI.dna
Bio.Restriction.Restriction.NaeI.dna
Bio.Restriction.Restriction.NarI.dna
Bio.Restriction.Restriction.NciI.dna
Bio.Restriction.Restriction.NcoI.dna
Bio.Restriction.Restriction.NdeI.dna
Bio.Restriction.Restriction.NdeII.dna
Bio.Restriction.Restriction.NgoAVIII.dna
Bio.Restriction.Restriction.NgoMIV.dna
Bio.Restriction.Restriction.NhaXI.dna
Bio.Restriction.Restriction.NheI.dna
Bio.Restriction.Restriction.NlaCI.dna
Bio.Restriction.Restriction.NlaIII.dna
Bio.Restriction.Restriction.NlaIV.dna
Bio.Restriction.Restriction.Nli3877I.dna
Bio.Restriction.Restriction.NmeAIII.dna
Bio.Restriction.Restriction.NmeDI.dna
Bio.Restriction.Restriction.NmuCI.dna
Bio.Restriction.Restriction.NotI.dna
Bio.Restriction.Restriction.NruI.dna
Bio.Restriction.Restriction.NsbI.dna
Bio.Restriction.Restriction.NsiI.dna
Bio.Restriction.Restriction.NspBII.dna
Bio.Restriction.Restriction.NspI.dna
Bio.Restriction.Restriction.NspV.dna
Bio.Restriction.Restriction.OliI.dna
Bio.Restriction.Restriction.PabI.dna
Bio.Restriction.Restriction.PacI.dna
Bio.Restriction.Restriction.PaeI.dna
Bio.Restriction.Restriction.PaeR7I.dna
Bio.Restriction.Restriction.PagI.dna
Bio.Restriction.Restriction.PalAI.dna
Bio.Restriction.Restriction.PasI.dna
Bio.Restriction.Restriction.PauI.dna
Bio.Restriction.Restriction.PceI.dna
Bio.Restriction.Restriction.PciI.dna
Bio.Restriction.Restriction.PciSI.dna
Bio.Restriction.Restriction.PcsI.dna
Bio.Restriction.Restriction.PctI.dna
Bio.Restriction.Restriction.PdiI.dna
Bio.Restriction.Restriction.PdmI.dna
Bio.Restriction.Restriction.PenI.dna
Bio.Restriction.Restriction.PfeI.dna
Bio.Restriction.Restriction.Pfl1108I.dna
Bio.Restriction.Restriction.Pfl23II.dna
Bio.Restriction.Restriction.PflFI.dna
Bio.Restriction.Restriction.PflMI.dna
Bio.Restriction.Restriction.PfoI.dna
Bio.Restriction.Restriction.PinAI.dna
Bio.Restriction.Restriction.PlaDI.dna
Bio.Restriction.Restriction.Ple19I.dna
Bio.Restriction.Restriction.PleI.dna
Bio.Restriction.Restriction.PluTI.dna
Bio.Restriction.Restriction.PmaCI.dna
Bio.Restriction.Restriction.PmeI.dna
Bio.Restriction.Restriction.PmlI.dna
Bio.Restriction.Restriction.PpiI.dna
Bio.Restriction.Restriction.PpsI.dna
Bio.Restriction.Restriction.Ppu10I.dna
Bio.Restriction.Restriction.Ppu21I.dna
Bio.Restriction.Restriction.PpuMI.dna
Bio.Restriction.Restriction.PscI.dna
Bio.Restriction.Restriction.PshAI.dna
Bio.Restriction.Restriction.PshBI.dna
Bio.Restriction.Restriction.PsiI.dna
Bio.Restriction.Restriction.Psp03I.dna
Bio.Restriction.Restriction.Psp124BI.dna
Bio.Restriction.Restriction.Psp1406I.dna
Bio.Restriction.Restriction.Psp5II.dna
Bio.Restriction.Restriction.Psp6I.dna
Bio.Restriction.Restriction.PspCI.dna
Bio.Restriction.Restriction.PspEI.dna
Bio.Restriction.Restriction.PspGI.dna
Bio.Restriction.Restriction.PspLI.dna
Bio.Restriction.Restriction.PspN4I.dna
Bio.Restriction.Restriction.PspOMI.dna
Bio.Restriction.Restriction.PspOMII.dna
Bio.Restriction.Restriction.PspPI.dna
Bio.Restriction.Restriction.PspPPI.dna
Bio.Restriction.Restriction.PspPRI.dna
Bio.Restriction.Restriction.PspXI.dna
Bio.Restriction.Restriction.PsrI.dna
Bio.Restriction.Restriction.PssI.dna
Bio.Restriction.Restriction.PstI.dna
Bio.Restriction.Restriction.PstNI.dna
Bio.Restriction.Restriction.PsuI.dna
Bio.Restriction.Restriction.PsyI.dna
Bio.Restriction.Restriction.PteI.dna
Bio.Restriction.Restriction.PvuI.dna
Bio.Restriction.Restriction.PvuII.dna
Bio.Restriction.Restriction.R2_BceSIV.dna
Bio.Restriction.Restriction.RceI.dna
Bio.Restriction.Restriction.RdeGBI.dna
Bio.Restriction.Restriction.RdeGBII.dna
Bio.Restriction.Restriction.RdeGBIII.dna
Bio.Restriction.Restriction.RflFIII.dna
Bio.Restriction.Restriction.RgaI.dna
Bio.Restriction.Restriction.RigI.dna
Bio.Restriction.Restriction.RlaI.dna
Bio.Restriction.Restriction.RleAI.dna
Bio.Restriction.Restriction.RpaB5I.dna
Bio.Restriction.Restriction.RpaBI.dna
Bio.Restriction.Restriction.RpaI.dna
Bio.Restriction.Restriction.RpaTI.dna
Bio.Restriction.Restriction.RruI.dna
Bio.Restriction.Restriction.RsaI.dna
Bio.Restriction.Restriction.RsaNI.dna
Bio.Restriction.Restriction.RseI.dna
Bio.Restriction.Restriction.Rsr2I.dna
Bio.Restriction.Restriction.RsrII.dna
Bio.Restriction.Restriction.SacI.dna
Bio.Restriction.Restriction.SacII.dna
Bio.Restriction.Restriction.SalI.dna
Bio.Restriction.Restriction.SanDI.dna
Bio.Restriction.Restriction.SapI.dna
Bio.Restriction.Restriction.SaqAI.dna
Bio.Restriction.Restriction.SatI.dna
Bio.Restriction.Restriction.Sau3AI.dna
Bio.Restriction.Restriction.Sau96I.dna
Bio.Restriction.Restriction.SauI.dna
Bio.Restriction.Restriction.SbfI.dna
Bio.Restriction.Restriction.ScaI.dna
Bio.Restriction.Restriction.SchI.dna
Bio.Restriction.Restriction.SciI.dna
Bio.Restriction.Restriction.ScrFI.dna
Bio.Restriction.Restriction.SdaI.dna
Bio.Restriction.Restriction.SdeAI.dna
Bio.Restriction.Restriction.SdeOSI.dna
Bio.Restriction.Restriction.SduI.dna
Bio.Restriction.Restriction.SecI.dna
Bio.Restriction.Restriction.SelI.dna
Bio.Restriction.Restriction.SetI.dna
Bio.Restriction.Restriction.SexAI.dna
Bio.Restriction.Restriction.SfaAI.dna
Bio.Restriction.Restriction.SfaNI.dna
Bio.Restriction.Restriction.SfcI.dna
Bio.Restriction.Restriction.SfeI.dna
Bio.Restriction.Restriction.SfiI.dna
Bio.Restriction.Restriction.SfoI.dna
Bio.Restriction.Restriction.Sfr274I.dna
Bio.Restriction.Restriction.Sfr303I.dna
Bio.Restriction.Restriction.SfuI.dna
Bio.Restriction.Restriction.SgeI.dna
Bio.Restriction.Restriction.SgfI.dna
Bio.Restriction.Restriction.SgrAI.dna
Bio.Restriction.Restriction.SgrBI.dna
Bio.Restriction.Restriction.SgrDI.dna
Bio.Restriction.Restriction.SgrTI.dna
Bio.Restriction.Restriction.SgsI.dna
Bio.Restriction.Restriction.SimI.dna
Bio.Restriction.Restriction.SlaI.dna
Bio.Restriction.Restriction.SmaI.dna
Bio.Restriction.Restriction.SmiI.dna
Bio.Restriction.Restriction.SmiMI.dna
Bio.Restriction.Restriction.SmlI.dna
Bio.Restriction.Restriction.SmoI.dna
Bio.Restriction.Restriction.SnaBI.dna
Bio.Restriction.Restriction.SnaI.dna
Bio.Restriction.Restriction.Sno506I.dna
Bio.Restriction.Restriction.SpeI.dna
Bio.Restriction.Restriction.SphI.dna
Bio.Restriction.Restriction.SplI.dna
Bio.Restriction.Restriction.SpoDI.dna
Bio.Restriction.Restriction.SrfI.dna
Bio.Restriction.Restriction.Sse232I.dna
Bio.Restriction.Restriction.Sse8387I.dna
Bio.Restriction.Restriction.Sse8647I.dna
Bio.Restriction.Restriction.Sse9I.dna
Bio.Restriction.Restriction.SseBI.dna
Bio.Restriction.Restriction.SsiI.dna
Bio.Restriction.Restriction.SspD5I.dna
Bio.Restriction.Restriction.SspDI.dna
Bio.Restriction.Restriction.SspI.dna
Bio.Restriction.Restriction.SstE37I.dna
Bio.Restriction.Restriction.SstI.dna
Bio.Restriction.Restriction.Sth132I.dna
Bio.Restriction.Restriction.Sth302II.dna
Bio.Restriction.Restriction.StrI.dna
Bio.Restriction.Restriction.StsI.dna
Bio.Restriction.Restriction.StuI.dna
Bio.Restriction.Restriction.StyD4I.dna
Bio.Restriction.Restriction.StyI.dna
Bio.Restriction.Restriction.SwaI.dna
Bio.Restriction.Restriction.TaaI.dna
Bio.Restriction.Restriction.TaiI.dna
Bio.Restriction.Restriction.TaqI.dna
Bio.Restriction.Restriction.TaqII.dna
Bio.Restriction.Restriction.TasI.dna
Bio.Restriction.Restriction.TatI.dna
Bio.Restriction.Restriction.TauI.dna
Bio.Restriction.Restriction.TfiI.dna
Bio.Restriction.Restriction.Tru1I.dna
Bio.Restriction.Restriction.Tru9I.dna
Bio.Restriction.Restriction.TscAI.dna
Bio.Restriction.Restriction.TseFI.dna
Bio.Restriction.Restriction.TseI.dna
Bio.Restriction.Restriction.TsoI.dna
Bio.Restriction.Restriction.Tsp45I.dna
Bio.Restriction.Restriction.Tsp4CI.dna
Bio.Restriction.Restriction.TspDTI.dna
Bio.Restriction.Restriction.TspEI.dna
Bio.Restriction.Restriction.TspGWI.dna
Bio.Restriction.Restriction.TspMI.dna
Bio.Restriction.Restriction.TspRI.dna
Bio.Restriction.Restriction.TssI.dna
Bio.Restriction.Restriction.TstI.dna
Bio.Restriction.Restriction.TsuI.dna
Bio.Restriction.Restriction.Tth111I.dna
Bio.Restriction.Restriction.Tth111II.dna
Bio.Restriction.Restriction.UbaF11I.dna
Bio.Restriction.Restriction.UbaF12I.dna
Bio.Restriction.Restriction.UbaF13I.dna
Bio.Restriction.Restriction.UbaF14I.dna
Bio.Restriction.Restriction.UbaF9I.dna
Bio.Restriction.Restriction.UbaPI.dna
Bio.Restriction.Restriction.UcoMSI.dna
Bio.Restriction.Restriction.UnbI.dna
Bio.Restriction.Restriction.Van91I.dna
Bio.Restriction.Restriction.Vha464I.dna
Bio.Restriction.Restriction.VneI.dna
Bio.Restriction.Restriction.VpaK11AI.dna
Bio.Restriction.Restriction.VpaK11BI.dna
Bio.Restriction.Restriction.VspI.dna
Bio.Restriction.Restriction.WviI.dna
Bio.Restriction.Restriction.XagI.dna
Bio.Restriction.Restriction.XapI.dna
Bio.Restriction.Restriction.XbaI.dna
Bio.Restriction.Restriction.XceI.dna
Bio.Restriction.Restriction.XcmI.dna
Bio.Restriction.Restriction.XhoI.dna
Bio.Restriction.Restriction.XhoII.dna
Bio.Restriction.Restriction.XmaI.dna
Bio.Restriction.Restriction.XmaIII.dna
Bio.Restriction.Restriction.XmaJI.dna
Bio.Restriction.Restriction.XmiI.dna
Bio.Restriction.Restriction.XmnI.dna
Bio.Restriction.Restriction.XspI.dna
Bio.Restriction.Restriction.YkrI.dna
Bio.Restriction.Restriction.ZraI.dna
Bio.Restriction.Restriction.ZrmI.dna
Bio.Restriction.Restriction.Zsp2I.dna" class="py-name" href="#" onclick="return doclink('link-9', 'dna', 'link-9');">dna</a></tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-10" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-10', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-11" class="py-name" targets="Package Bio.motifs=Bio.motifs-module.html"><a title="Bio.motifs" class="py-name" href="#" onclick="return doclink('link-11', 'motifs', 'link-11');">motifs</a></tt> <tt class="py-keyword">import</tt> <tt id="link-12" class="py-name" targets="Package Bio.motifs.jaspar=Bio.motifs.jaspar-module.html"><a title="Bio.motifs.jaspar" class="py-name" href="#" onclick="return doclink('link-12', 'jaspar', 'link-12');">jaspar</a></tt><tt class="py-op">,</tt> <tt id="link-13" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.matrix()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#matrix,Module Bio.motifs.matrix=Bio.motifs.matrix-module.html"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.matrix
Bio.motifs.matrix" class="py-name" href="#" onclick="return doclink('link-13', 'matrix', 'link-13');">matrix</a></tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line"> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line"><tt class="py-name">__docformat__</tt> <tt class="py-op">=</tt> <tt class="py-string">"restructuredtext en"</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line"> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line"><tt id="link-14" class="py-name" targets="Variable Bio.motifs.jaspar.db.JASPAR_DFLT_COLLECTION=Bio.motifs.jaspar.db-module.html#JASPAR_DFLT_COLLECTION"><a title="Bio.motifs.jaspar.db.JASPAR_DFLT_COLLECTION" class="py-name" href="#" onclick="return doclink('link-14', 'JASPAR_DFLT_COLLECTION', 'link-14');">JASPAR_DFLT_COLLECTION</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'CORE'</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line"> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line"> </tt>
<a name="JASPAR5"></a><div id="JASPAR5-def"><a name="L84"></a><tt class="py-lineno"> 84</tt> <a class="py-toggle" href="#" id="JASPAR5-toggle" onclick="return toggle('JASPAR5');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.motifs.jaspar.db.JASPAR5-class.html">JASPAR5</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="JASPAR5-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="JASPAR5-expanded"><a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line"><tt class="py-docstring">    Class representing a JASPAR5 DB. The methods within are loosely based</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line"><tt class="py-docstring">    on the perl TFBS::DB::JASPAR5 module.</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line"><tt class="py-docstring">    Note: We will only implement reading of JASPAR motifs from the DB.</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line"><tt class="py-docstring">    Unlike the perl module, we will not attempt to implement any methods to</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line"><tt class="py-docstring">    store JASPAR motifs or create a new DB at this time.</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line"> </tt>
<a name="JASPAR5.__init__"></a><div id="JASPAR5.__init__-def"><a name="L95"></a><tt class="py-lineno"> 95</tt> <a class="py-toggle" href="#" id="JASPAR5.__init__-toggle" onclick="return toggle('JASPAR5.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.jaspar.db.JASPAR5-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">host</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">user</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">password</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="JASPAR5.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="JASPAR5.__init__-expanded"><a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line"><tt class="py-docstring">        Construct a JASPAR5 instance and connect to specified DB</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line"><tt class="py-docstring">        host - host name of the the JASPAR DB server</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line"><tt class="py-docstring">        name - name of the JASPAR database</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line"><tt class="py-docstring">        user - user name to connect to the JASPAR DB</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line"><tt class="py-docstring">        password - JASPAR DB password</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line"> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-15" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-15', 'name', 'link-15');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-16" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-16', 'name', 'link-15');">name</a></tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">host</tt> <tt class="py-op">=</tt> <tt class="py-name">host</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">user</tt> <tt class="py-op">=</tt> <tt class="py-name">user</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">password</tt> <tt class="py-op">=</tt> <tt class="py-name">password</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line"> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbh</tt> <tt class="py-op">=</tt> <tt class="py-name">mdb</tt><tt class="py-op">.</tt><tt class="py-name">connect</tt><tt class="py-op">(</tt><tt class="py-name">host</tt><tt class="py-op">,</tt> <tt class="py-name">user</tt><tt class="py-op">,</tt> <tt class="py-name">password</tt><tt class="py-op">,</tt> <tt id="link-17" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-17', 'name', 'link-15');">name</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line"> </tt>
<a name="JASPAR5.__str__"></a><div id="JASPAR5.__str__-def"><a name="L114"></a><tt class="py-lineno">114</tt> <a class="py-toggle" href="#" id="JASPAR5.__str__-toggle" onclick="return toggle('JASPAR5.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.jaspar.db.JASPAR5-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="JASPAR5.__str__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="JASPAR5.__str__-expanded"><a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line"><tt class="py-docstring">        Return a string represention of the JASPAR5 DB connection.</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line"> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">        <tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt class="py-string">"%s\@%s:%s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">user</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">host</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-18" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-18', 'name', 'link-15');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line"> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">text</tt> </tt>
</div><a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line"> </tt>
<a name="JASPAR5.fetch_motif_by_id"></a><div id="JASPAR5.fetch_motif_by_id-def"><a name="L124"></a><tt class="py-lineno">124</tt> <a class="py-toggle" href="#" id="JASPAR5.fetch_motif_by_id-toggle" onclick="return toggle('JASPAR5.fetch_motif_by_id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.jaspar.db.JASPAR5-class.html#fetch_motif_by_id">fetch_motif_by_id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="JASPAR5.fetch_motif_by_id-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="JASPAR5.fetch_motif_by_id-expanded"><a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line"><tt class="py-docstring">        Fetch a single JASPAR motif from the DB by it's JASPAR matrix ID</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line"><tt class="py-docstring">        (e.g. 'MA0001.1').</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line"><tt class="py-docstring">            - id - JASPAR matrix ID. This may be a fully specified ID including</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line"><tt class="py-docstring">              the version number (e.g. MA0049.2) or just the base ID (e.g.</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line"><tt class="py-docstring">              MA0049). If only a base ID is provided, the latest version is</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line"><tt class="py-docstring">              returned.</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line"><tt class="py-docstring">        Returns:</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line"><tt class="py-docstring">            - A Bio.motifs.jaspar.Motif object</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line"><tt class="py-docstring">        **NOTE:** The perl TFBS module allows you to specify the type of matrix</tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line"><tt class="py-docstring">        to return (PFM, PWM, ICM) but matrices are always stored in JASPAR as</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line"><tt class="py-docstring">        PFMs so this does not really belong here. Once a PFM is fetched the</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line"><tt class="py-docstring">        pwm() and pssm() methods can be called to return the normalized and</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line"><tt class="py-docstring">        log-odds matrices.</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line"> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">        <tt class="py-comment"># separate stable ID and version number</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line">        <tt class="py-op">(</tt><tt id="link-19" class="py-name" targets="Variable Bio.motifs.jaspar.Motif.base_id=Bio.motifs.jaspar.Motif-class.html#base_id"><a title="Bio.motifs.jaspar.Motif.base_id" class="py-name" href="#" onclick="return doclink('link-19', 'base_id', 'link-19');">base_id</a></tt><tt class="py-op">,</tt> <tt id="link-20" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HeaderConsumer.version()=Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#version,Function Bio.Cluster.cluster.version()=Bio.Cluster.cluster-module.html#version,Method Bio.GenBank._FeatureConsumer.version()=Bio.GenBank._FeatureConsumer-class.html#version,Method Bio.GenBank._RecordConsumer.version()=Bio.GenBank._RecordConsumer-class.html#version,Variable Bio.motifs.jaspar.Motif.version=Bio.motifs.jaspar.Motif-class.html#version"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-20', 'version', 'link-20');">version</a></tt><tt class="py-op">)</tt> <tt class="py-op">=</tt> <tt id="link-21" class="py-name"><a title="Bio.motifs.jaspar" class="py-name" href="#" onclick="return doclink('link-21', 'jaspar', 'link-12');">jaspar</a></tt><tt class="py-op">.</tt><tt id="link-22" class="py-name" targets="Function Bio.motifs.jaspar.split_jaspar_id()=Bio.motifs.jaspar-module.html#split_jaspar_id"><a title="Bio.motifs.jaspar.split_jaspar_id" class="py-name" href="#" onclick="return doclink('link-22', 'split_jaspar_id', 'link-22');">split_jaspar_id</a></tt><tt class="py-op">(</tt><tt id="link-23" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-23', 'id', 'link-23');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-24" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-24', 'version', 'link-20');">version</a></tt><tt class="py-op">:</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">            <tt class="py-comment"># if ID contains no version portion, fetch the latest version</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">            <tt id="link-25" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-25', 'version', 'link-20');">version</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-26" class="py-name" targets="Method Bio.motifs.jaspar.db.JASPAR5._fetch_latest_version()=Bio.motifs.jaspar.db.JASPAR5-class.html#_fetch_latest_version"><a title="Bio.motifs.jaspar.db.JASPAR5._fetch_latest_version" class="py-name" href="#" onclick="return doclink('link-26', '_fetch_latest_version', 'link-26');">_fetch_latest_version</a></tt><tt class="py-op">(</tt><tt id="link-27" class="py-name"><a title="Bio.motifs.jaspar.Motif.base_id" class="py-name" href="#" onclick="return doclink('link-27', 'base_id', 'link-19');">base_id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line"> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line">        <tt class="py-comment"># fetch internal JASPAR matrix ID - also a check for validity</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">        <tt class="py-name">int_id</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-28" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-28', 'version', 'link-20');">version</a></tt><tt class="py-op">:</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">            <tt class="py-name">int_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-29" class="py-name" targets="Method Bio.motifs.jaspar.db.JASPAR5._fetch_internal_id()=Bio.motifs.jaspar.db.JASPAR5-class.html#_fetch_internal_id"><a title="Bio.motifs.jaspar.db.JASPAR5._fetch_internal_id" class="py-name" href="#" onclick="return doclink('link-29', '_fetch_internal_id', 'link-29');">_fetch_internal_id</a></tt><tt class="py-op">(</tt><tt id="link-30" class="py-name"><a title="Bio.motifs.jaspar.Motif.base_id" class="py-name" href="#" onclick="return doclink('link-30', 'base_id', 'link-19');">base_id</a></tt><tt class="py-op">,</tt> <tt id="link-31" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-31', 'version', 'link-20');">version</a></tt><tt class="py-op">)</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line"> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">        <tt class="py-comment"># fetch JASPAR motif using internal ID</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">        <tt class="py-name">motif</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">int_id</tt><tt class="py-op">:</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">            <tt class="py-name">motif</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-32" class="py-name" targets="Method Bio.motifs.jaspar.db.JASPAR5._fetch_motif_by_internal_id()=Bio.motifs.jaspar.db.JASPAR5-class.html#_fetch_motif_by_internal_id"><a title="Bio.motifs.jaspar.db.JASPAR5._fetch_motif_by_internal_id" class="py-name" href="#" onclick="return doclink('link-32', '_fetch_motif_by_internal_id', 'link-32');">_fetch_motif_by_internal_id</a></tt><tt class="py-op">(</tt><tt class="py-name">int_id</tt><tt class="py-op">)</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line"> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">motif</tt> </tt>
</div><a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line"> </tt>
<a name="JASPAR5.fetch_motifs_by_name"></a><div id="JASPAR5.fetch_motifs_by_name-def"><a name="L166"></a><tt class="py-lineno">166</tt> <a class="py-toggle" href="#" id="JASPAR5.fetch_motifs_by_name-toggle" onclick="return toggle('JASPAR5.fetch_motifs_by_name');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.jaspar.db.JASPAR5-class.html#fetch_motifs_by_name">fetch_motifs_by_name</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="JASPAR5.fetch_motifs_by_name-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="JASPAR5.fetch_motifs_by_name-expanded"><a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line"><tt class="py-docstring">        Fetch a list of JASPAR motifs from a JASPAR DB by the given TF name(s).</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line"><tt class="py-docstring">        name - a single name or list of names</tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line"><tt class="py-docstring">        Returns:</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line"><tt class="py-docstring">        A list of Bio.motifs.Motif.japar objects</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line"><tt class="py-docstring">        Notes:</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line"><tt class="py-docstring">        Names are not guaranteed to be unique. There may be more than one</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line"><tt class="py-docstring">        motif with the same name. Therefore even if name specifies a single</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line"><tt class="py-docstring">        name, a list of motifs is returned. This just calls</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line"><tt class="py-docstring">        self.fetch_motifs(collection = None, tf_name = name).</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line"><tt class="py-docstring">        This behaviour is different from the TFBS perl module's</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line"><tt class="py-docstring">        get_Matrix_by_name() method which always returns a single matrix,</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line"><tt class="py-docstring">        issuing a warning message and returning the first matrix retrieved</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line"><tt class="py-docstring">        in the case where multiple matrices have the same name.</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line"> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-33" class="py-name" targets="Method Bio.motifs.jaspar.db.JASPAR5.fetch_motifs()=Bio.motifs.jaspar.db.JASPAR5-class.html#fetch_motifs"><a title="Bio.motifs.jaspar.db.JASPAR5.fetch_motifs" class="py-name" href="#" onclick="return doclink('link-33', 'fetch_motifs', 'link-33');">fetch_motifs</a></tt><tt class="py-op">(</tt><tt class="py-name">collection</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-name">tf_name</tt><tt class="py-op">=</tt><tt id="link-34" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-34', 'name', 'link-15');">name</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line"> </tt>
<a name="JASPAR5.fetch_motifs"></a><div id="JASPAR5.fetch_motifs-def"><a name="L190"></a><tt class="py-lineno">190</tt> <a class="py-toggle" href="#" id="JASPAR5.fetch_motifs-toggle" onclick="return toggle('JASPAR5.fetch_motifs');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.jaspar.db.JASPAR5-class.html#fetch_motifs">fetch_motifs</a><tt class="py-op">(</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">        <tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">collection</tt><tt class="py-op">=</tt><tt id="link-35" class="py-name"><a title="Bio.motifs.jaspar.db.JASPAR_DFLT_COLLECTION" class="py-name" href="#" onclick="return doclink('link-35', 'JASPAR_DFLT_COLLECTION', 'link-14');">JASPAR_DFLT_COLLECTION</a></tt><tt class="py-op">,</tt> <tt class="py-param">tf_name</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">tf_class</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">        <tt class="py-param">tf_family</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">matrix_id</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">tax_group</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">species</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">        <tt class="py-param">pazar_id</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">data_type</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">medline</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">min_ic</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-param">min_length</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">        <tt class="py-param">min_sites</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-param">all</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> <tt class="py-param">all_versions</tt><tt class="py-op">=</tt><tt class="py-name">False</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">    <tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="JASPAR5.fetch_motifs-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="JASPAR5.fetch_motifs-expanded"><a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line"><tt class="py-docstring">        Fetch a jaspar.Record (list) of motifs based on the provided selection</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line"><tt class="py-docstring">        criteria.</tt> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments::</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line"><tt class="py-docstring">            Except where obvious, all selection criteria arguments may be</tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line"><tt class="py-docstring">            specified as a single value or a list of values. Motifs must</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line"><tt class="py-docstring">            meet ALL the specified selection criteria to be returned with</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line"><tt class="py-docstring">            the precedent exceptions noted below.</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line"><tt class="py-docstring">            all         - Takes precedent of all other selection criteria.</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line"><tt class="py-docstring">                          Every motif is returned. If 'all_versions' is also</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line"><tt class="py-docstring">                          specified, all versions of every motif are returned,</tt> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line"><tt class="py-docstring">                          otherwise just the latest version of every motif is</tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line"><tt class="py-docstring">                          returned.</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line"><tt class="py-docstring">            matrix_id   - Takes precedence over all other selection criteria</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line"><tt class="py-docstring">                          except 'all'.  Only motifs with the given JASPAR</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line"><tt class="py-docstring">                          matrix ID(s) are returned. A matrix ID may be</tt> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line"><tt class="py-docstring">                          specified as just a base ID or full JASPAR IDs</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line"><tt class="py-docstring">                          including version number. If only a base ID is</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line"><tt class="py-docstring">                          provided for specific motif(s), then just the latest</tt> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line"><tt class="py-docstring">                          version of those motif(s) are returned unless</tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line"><tt class="py-docstring">                          'all_versions' is also specified.</tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line"><tt class="py-docstring">            collection  - Only motifs from the specified JASPAR collection(s)</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line"><tt class="py-docstring">                          are returned. NOTE - if not specified, the collection</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line"><tt class="py-docstring">                          defaults to CORE for all other selection criteria</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line"><tt class="py-docstring">                          except 'all' and 'matrix_id'. To apply the other</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line"><tt class="py-docstring">                          selection criteria across all JASPAR collections,</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line"><tt class="py-docstring">                          explicitly set collection=None.</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line"><tt class="py-docstring">            tf_name     - Only motifs with the given name(s) are returned.</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line"><tt class="py-docstring">            tf_class    - Only motifs of the given TF class(es) are returned.</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line"><tt class="py-docstring">            tf_family   - Only motifs from the given TF families are returned.</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line"><tt class="py-docstring">            tax_group   - Only motifs belonging to the given taxonomic</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line"><tt class="py-docstring">                          supergroups are returned (e.g. 'vertebrates',</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line"><tt class="py-docstring">                          'insects', 'nematodes' etc.)</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line"><tt class="py-docstring">            species     - Only motifs derived from the given species are</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line"><tt class="py-docstring">                          returned.  Species are specified as taxonomy IDs.</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line"><tt class="py-docstring">            data_type   - Only motifs generated with the given data type (e.g.</tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line"><tt class="py-docstring">                          ('ChIP-seq', 'PBM', 'SELEX' etc.) are returned.</tt> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line"><tt class="py-docstring">                          NOTE - must match exactly as stored in the database.</tt> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line"><tt class="py-docstring">            pazar_id    - Only motifs with the given PAZAR TF ID are returned.</tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line"><tt class="py-docstring">            medline     - Only motifs with the given medline (PubmMed IDs) are </tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line"><tt class="py-docstring">                          returned.</tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line"><tt class="py-docstring">            min_ic      - Only motifs whose profile matrices have at least this</tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line"><tt class="py-docstring">                          information content (specificty) are returned.</tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line"><tt class="py-docstring">            min_length  - Only motifs whose profiles are of at least this</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line"><tt class="py-docstring">                          length are returned.</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line"><tt class="py-docstring">            min_sites   - Only motifs compiled from at least these many binding</tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line"><tt class="py-docstring">                          sites are returned.</tt> </tt>
<a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line"><tt class="py-docstring">            all_versions- Unless specified, just the latest version of motifs</tt> </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line"><tt class="py-docstring">                          determined by the other selection criteria are</tt> </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line"><tt class="py-docstring">                          returned. Otherwise all versions of the selected</tt> </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line"><tt class="py-docstring">                          motifs are returned.</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line"><tt class="py-docstring">        Returns:</tt> </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line"><tt class="py-docstring">            - A Bio.motifs.jaspar.Record (list) of motifs.</tt> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line"> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line">        <tt class="py-comment"># Fetch the internal IDs of the motifs using the criteria provided</tt> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line">        <tt class="py-name">int_ids</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-36" class="py-name" targets="Method Bio.motifs.jaspar.db.JASPAR5._fetch_internal_id_list()=Bio.motifs.jaspar.db.JASPAR5-class.html#_fetch_internal_id_list"><a title="Bio.motifs.jaspar.db.JASPAR5._fetch_internal_id_list" class="py-name" href="#" onclick="return doclink('link-36', '_fetch_internal_id_list', 'link-36');">_fetch_internal_id_list</a></tt><tt class="py-op">(</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line">            <tt class="py-name">collection</tt><tt class="py-op">=</tt><tt class="py-name">collection</tt><tt class="py-op">,</tt> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line">            <tt class="py-name">tf_name</tt><tt class="py-op">=</tt><tt class="py-name">tf_name</tt><tt class="py-op">,</tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line">            <tt class="py-name">tf_class</tt><tt class="py-op">=</tt><tt class="py-name">tf_class</tt><tt class="py-op">,</tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line">            <tt class="py-name">tf_family</tt><tt class="py-op">=</tt><tt class="py-name">tf_family</tt><tt class="py-op">,</tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line">            <tt class="py-name">matrix_id</tt><tt class="py-op">=</tt><tt class="py-name">matrix_id</tt><tt class="py-op">,</tt> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line">            <tt class="py-name">tax_group</tt><tt class="py-op">=</tt><tt class="py-name">tax_group</tt><tt class="py-op">,</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line">            <tt id="link-37" class="py-name" targets="Method Bio.Pathway.Network.species()=Bio.Pathway.Network-class.html#species,Method Bio.Pathway.Reaction.species()=Bio.Pathway.Reaction-class.html#species,Method Bio.Pathway.System.species()=Bio.Pathway.System-class.html#species"><a title="Bio.Pathway.Network.species
Bio.Pathway.Reaction.species
Bio.Pathway.System.species" class="py-name" href="#" onclick="return doclink('link-37', 'species', 'link-37');">species</a></tt><tt class="py-op">=</tt><tt id="link-38" class="py-name"><a title="Bio.Pathway.Network.species
Bio.Pathway.Reaction.species
Bio.Pathway.System.species" class="py-name" href="#" onclick="return doclink('link-38', 'species', 'link-37');">species</a></tt><tt class="py-op">,</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">            <tt class="py-name">pazar_id</tt><tt class="py-op">=</tt><tt class="py-name">pazar_id</tt><tt class="py-op">,</tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">            <tt class="py-name">data_type</tt><tt class="py-op">=</tt><tt class="py-name">data_type</tt><tt class="py-op">,</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line">            <tt class="py-name">medline</tt><tt class="py-op">=</tt><tt class="py-name">medline</tt><tt class="py-op">,</tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">            <tt class="py-name">all</tt><tt class="py-op">=</tt><tt class="py-name">all</tt><tt class="py-op">,</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">            <tt class="py-name">all_versions</tt><tt class="py-op">=</tt><tt class="py-name">all_versions</tt> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line">        <tt class="py-op">)</tt> </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line"> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line">        <tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt id="link-39" class="py-name"><a title="Bio.motifs.jaspar" class="py-name" href="#" onclick="return doclink('link-39', 'jaspar', 'link-12');">jaspar</a></tt><tt class="py-op">.</tt><tt id="link-40" class="py-name" targets="Class Bio.Affy.CelFile.Record=Bio.Affy.CelFile.Record-class.html,Module Bio.Blast.Record=Bio.Blast.Record-module.html,Class Bio.Cluster.Record=Bio.Cluster.Record-class.html,Class Bio.Compass.Record=Bio.Compass.Record-class.html,Class Bio.Emboss.Primer3.Record=Bio.Emboss.Primer3.Record-class.html,Class Bio.ExPASy.Enzyme.Record=Bio.ExPASy.Enzyme.Record-class.html,Class Bio.ExPASy.Prodoc.Record=Bio.ExPASy.Prodoc.Record-class.html,Class Bio.ExPASy.Prosite.Record=Bio.ExPASy.Prosite.Record-class.html,Class Bio.ExPASy.ScanProsite.Record=Bio.ExPASy.ScanProsite.Record-class.html,Module Bio.GenBank.Record=Bio.GenBank.Record-module.html,Class Bio.GenBank.Record.Record=Bio.GenBank.Record.Record-class.html,Module Bio.Geo.Record=Bio.Geo.Record-module.html,Class Bio.Geo.Record.Record=Bio.Geo.Record.Record-class.html,Class Bio.KEGG.Compound.Record=Bio.KEGG.Compound.Record-class.html,Class Bio.KEGG.Enzyme.Record=Bio.KEGG.Enzyme.Record-class.html,Class Bio.Medline.Record=Bio.Medline.Record-class.html,Class Bio.Motif.Parsers.AlignAce.Record=Bio.Motif.Parsers.AlignAce.Record-class.html,Class Bio.Motif.Parsers.MAST.Record=Bio.Motif.Parsers.MAST.Record-class.html,Class Bio.PopGen.FDist.Record=Bio.PopGen.FDist.Record-class.html,Class Bio.PopGen.GenePop.LargeFileParser.Record=Bio.PopGen.GenePop.LargeFileParser.Record-class.html,Class Bio.PopGen.GenePop.Record=Bio.PopGen.GenePop.Record-class.html,Class Bio.SCOP.Cla.Record=Bio.SCOP.Cla.Record-class.html,Class Bio.SCOP.Des.Record=Bio.SCOP.Des.Record-class.html,Class Bio.SCOP.Dom.Record=Bio.SCOP.Dom.Record-class.html,Class Bio.SCOP.Hie.Record=Bio.SCOP.Hie.Record-class.html,Class Bio.Sequencing.Phd.Record=Bio.Sequencing.Phd.Record-class.html,Class Bio.SwissProt.KeyWList.Record=Bio.SwissProt.KeyWList.Record-class.html,Class Bio.SwissProt.Record=Bio.SwissProt.Record-class.html,Class Bio.UniGene.Record=Bio.UniGene.Record-class.html,Class Bio.motifs.alignace.Record=Bio.motifs.alignace.Record-class.html,Class Bio.motifs.jaspar.Record=Bio.motifs.jaspar.Record-class.html,Class Bio.motifs.mast.Record=Bio.motifs.mast.Record-class.html,Class Bio.motifs.meme.Record=Bio.motifs.meme.Record-class.html,Class Bio.motifs.transfac.Record=Bio.motifs.transfac.Record-class.html"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-40', 'Record', 'link-40');">Record</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line"> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line">        <tt class="py-string">"""</tt> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line"><tt class="py-string">        Now further filter motifs returned above based on any specified</tt> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line"><tt class="py-string">        matrix specific criteria.</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line"><tt class="py-string">        """</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">int_id</tt> <tt class="py-keyword">in</tt> <tt class="py-name">int_ids</tt><tt class="py-op">:</tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line">            <tt class="py-name">motif</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-41" class="py-name"><a title="Bio.motifs.jaspar.db.JASPAR5._fetch_motif_by_internal_id" class="py-name" href="#" onclick="return doclink('link-41', '_fetch_motif_by_internal_id', 'link-32');">_fetch_motif_by_internal_id</a></tt><tt class="py-op">(</tt><tt class="py-name">int_id</tt><tt class="py-op">)</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line"> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line">            <tt class="py-comment"># Filter motifs to those with matrix IC greater than min_ic</tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">min_ic</tt><tt class="py-op">:</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt id="link-42" class="py-name" targets="Variable Bio.motifs.Motif.pssm=Bio.motifs.Motif-class.html#pssm"><a title="Bio.motifs.Motif.pssm" class="py-name" href="#" onclick="return doclink('link-42', 'pssm', 'link-42');">pssm</a></tt><tt class="py-op">.</tt><tt id="link-43" class="py-name" targets="Function Bio.Cluster.cluster.mean()=Bio.Cluster.cluster-module.html#mean,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.mean()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#mean,Method Bio.motifs.matrix.PositionSpecificScoringMatrix.mean()=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#mean"><a title="Bio.Cluster.cluster.mean
Bio.Graphics.GenomeDiagram._Graph.GraphData.mean
Bio.motifs.matrix.PositionSpecificScoringMatrix.mean" class="py-name" href="#" onclick="return doclink('link-43', 'mean', 'link-43');">mean</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">min_ic</tt><tt class="py-op">:</tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line">                    <tt class="py-keyword">continue</tt> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line"> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">            <tt class="py-comment"># Filter motifs to those with minimum length of min_length</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">min_length</tt><tt class="py-op">:</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt id="link-44" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.length()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#length"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-44', 'length', 'link-44');">length</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-name">min_length</tt><tt class="py-op">:</tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">                    <tt class="py-keyword">continue</tt> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line"> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line">            <tt class="py-comment"># XXX We could also supply a max_length filter.</tt> </tt>
<a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line"> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line">            <tt class="py-string">"""</tt> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line"><tt class="py-string">            Filter motifs to those composed of at least this many sites.</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line"><tt class="py-string">            The perl TFBS module assumes column sums may be different but</tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line"><tt class="py-string">            this should be strictly enforced here we will ignore this and</tt> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line"><tt class="py-string">            just use the first column sum.</tt> </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line"><tt class="py-string">            """</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">min_sites</tt><tt class="py-op">:</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line">                <tt class="py-name">num_sites</tt> <tt class="py-op">=</tt> <tt id="link-45" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.sum()=Bio.Nexus.Nexus.StepMatrix-class.html#sum,Method Bio.SubsMat.SeqMat.sum()=Bio.SubsMat.SeqMat-class.html#sum"><a title="Bio.Nexus.Nexus.StepMatrix.sum
Bio.SubsMat.SeqMat.sum" class="py-name" href="#" onclick="return doclink('link-45', 'sum', 'link-45');">sum</a></tt><tt class="py-op">(</tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">                    <tt class="py-op">[</tt><tt class="py-name">motif</tt><tt class="py-op">.</tt><tt class="py-name">counts</tt><tt class="py-op">[</tt><tt class="py-name">nt</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt class="py-name">nt</tt> <tt class="py-keyword">in</tt> <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt id="link-46" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.alphabet=Bio.SearchIO._model.hsp.HSP-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSPFragment.alphabet=Bio.SearchIO._model.hsp.HSPFragment-class.html#alphabet"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-46', 'alphabet', 'link-46');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-47" class="py-name" targets="Variable Bio.Alphabet.Alphabet.letters=Bio.Alphabet.Alphabet-class.html#letters,Variable Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters=Bio.Alphabet.IUPAC.ExtendedIUPACDNA-class.html#letters,Variable Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters=Bio.Alphabet.IUPAC.ExtendedIUPACProtein-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters=Bio.Alphabet.IUPAC.IUPACAmbiguousDNA-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters=Bio.Alphabet.IUPAC.IUPACAmbiguousRNA-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACProtein.letters=Bio.Alphabet.IUPAC.IUPACProtein-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters=Bio.Alphabet.IUPAC.IUPACUnambiguousDNA-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters=Bio.Alphabet.IUPAC.IUPACUnambiguousRNA-class.html#letters,Variable Bio.Alphabet.Reduced.HPModel.letters=Bio.Alphabet.Reduced.HPModel-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy10.letters=Bio.Alphabet.Reduced.Murphy10-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy15.letters=Bio.Alphabet.Reduced.Murphy15-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy4.letters=Bio.Alphabet.Reduced.Murphy4-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy8.letters=Bio.Alphabet.Reduced.Murphy8-class.html#letters,Variable Bio.Alphabet.Reduced.PC5.letters=Bio.Alphabet.Reduced.PC5-class.html#letters,Variable Bio.Alphabet.SecondaryStructure.letters=Bio.Alphabet.SecondaryStructure-class.html#letters,Variable Bio.Alphabet.SingleLetterAlphabet.letters=Bio.Alphabet.SingleLetterAlphabet-class.html#letters,Variable Bio.Alphabet.ThreeLetterProtein.letters=Bio.Alphabet.ThreeLetterProtein-class.html#letters,Variable Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.letters=Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet-class.html#letters,Variable Bio.codonalign.codonalphabet.CodonAlphabet.letters=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#letters"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-47', 'letters', 'link-47');">letters</a></tt><tt class="py-op">]</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line">                <tt class="py-op">)</tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">num_sites</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">min_sites</tt><tt class="py-op">:</tt> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line">                    <tt class="py-keyword">continue</tt> </tt>
<a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line"> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-48" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-48', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">motif</tt><tt class="py-op">)</tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line"> </tt>
<a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">record</tt> </tt>
</div><a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line"> </tt>
<a name="JASPAR5._fetch_latest_version"></a><div id="JASPAR5._fetch_latest_version-def"><a name="L311"></a><tt class="py-lineno">311</tt> <a class="py-toggle" href="#" id="JASPAR5._fetch_latest_version-toggle" onclick="return toggle('JASPAR5._fetch_latest_version');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.jaspar.db.JASPAR5-class.html#_fetch_latest_version">_fetch_latest_version</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">base_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="JASPAR5._fetch_latest_version-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="JASPAR5._fetch_latest_version-expanded"><a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line"><tt class="py-docstring">        Get the latest version number for the given base_id,</tt> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line"> </tt>
<a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line">        <tt class="py-name">cur</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbh</tt><tt class="py-op">.</tt><tt id="link-49" class="py-name" targets="Method Bio.DocSQL.Query.cursor()=Bio.DocSQL.Query-class.html#cursor,Method Bio.DocSQL.QuerySingle.cursor()=Bio.DocSQL.QuerySingle-class.html#cursor"><a title="Bio.DocSQL.Query.cursor
Bio.DocSQL.QuerySingle.cursor" class="py-name" href="#" onclick="return doclink('link-49', 'cursor', 'link-49');">cursor</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">        <tt class="py-name">cur</tt><tt class="py-op">.</tt><tt id="link-50" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.execute()=BioSQL.BioSeqDatabase.Adaptor-class.html#execute,Method BioSQL.BioSeqDatabase._CursorWrapper.execute()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#execute,Method BioSQL.DBUtils.Generic_dbutils.execute()=BioSQL.DBUtils.Generic_dbutils-class.html#execute,Method BioSQL.DBUtils.Sqlite_dbutils.execute()=BioSQL.DBUtils.Sqlite_dbutils-class.html#execute"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-50', 'execute', 'link-50');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"""select VERSION from MATRIX where BASE_id = %s</tt> </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line"><tt class="py-string">                       order by VERSION desc limit 1"""</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt id="link-51" class="py-name"><a title="Bio.motifs.jaspar.Motif.base_id" class="py-name" href="#" onclick="return doclink('link-51', 'base_id', 'link-19');">base_id</a></tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line"> </tt>
<a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line">        <tt class="py-name">row</tt> <tt class="py-op">=</tt> <tt class="py-name">cur</tt><tt class="py-op">.</tt><tt id="link-52" class="py-name" targets="Method BioSQL.BioSeqDatabase._CursorWrapper.fetchone()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#fetchone"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchone" class="py-name" href="#" onclick="return doclink('link-52', 'fetchone', 'link-52');">fetchone</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line"> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">        <tt class="py-name">latest</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">row</tt><tt class="py-op">:</tt> </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line">            <tt class="py-name">latest</tt> <tt class="py-op">=</tt> <tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Failed to fetch latest version number for JASPAR motif with base ID '{0}'. No JASPAR motif with this base ID appears to exist in the database."</tt><tt class="py-op">.</tt><tt id="link-53" class="py-name" targets="Method Bio.Align.Generic.Alignment.format()=Bio.Align.Generic.Alignment-class.html#format,Method Bio.Motif._Motif.Motif.format()=Bio.Motif._Motif.Motif-class.html#format,Method Bio.Phylo.BaseTree.Tree.format()=Bio.Phylo.BaseTree.Tree-class.html#format,Method Bio.Restriction.Restriction.RestrictionBatch.format()=Bio.Restriction.Restriction.RestrictionBatch-class.html#format,Method Bio.SeqRecord.SeqRecord.format()=Bio.SeqRecord.SeqRecord-class.html#format,Method Bio.motifs.Motif.format()=Bio.motifs.Motif-class.html#format"><a title="Bio.Align.Generic.Alignment.format
Bio.Motif._Motif.Motif.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-53', 'format', 'link-53');">format</a></tt><tt class="py-op">(</tt><tt id="link-54" class="py-name"><a title="Bio.motifs.jaspar.Motif.base_id" class="py-name" href="#" onclick="return doclink('link-54', 'base_id', 'link-19');">base_id</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-55" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-55', 'BiopythonWarning', 'link-1');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line"> </tt>
<a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">latest</tt> </tt>
</div><a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line"> </tt>
<a name="JASPAR5._fetch_internal_id"></a><div id="JASPAR5._fetch_internal_id-def"><a name="L331"></a><tt class="py-lineno">331</tt> <a class="py-toggle" href="#" id="JASPAR5._fetch_internal_id-toggle" onclick="return toggle('JASPAR5._fetch_internal_id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.jaspar.db.JASPAR5-class.html#_fetch_internal_id">_fetch_internal_id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">base_id</tt><tt class="py-op">,</tt> <tt class="py-param">version</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="JASPAR5._fetch_internal_id-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="JASPAR5._fetch_internal_id-expanded"><a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line"><tt class="py-docstring">        Fetch the internal id for a base id + version. Also checks if this</tt> </tt>
<a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line"><tt class="py-docstring">        combo exists or not</tt> </tt>
<a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line"> </tt>
<a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line">        <tt class="py-name">cur</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbh</tt><tt class="py-op">.</tt><tt id="link-56" class="py-name"><a title="Bio.DocSQL.Query.cursor
Bio.DocSQL.QuerySingle.cursor" class="py-name" href="#" onclick="return doclink('link-56', 'cursor', 'link-49');">cursor</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line">        <tt class="py-name">cur</tt><tt class="py-op">.</tt><tt id="link-57" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-57', 'execute', 'link-50');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"""select id from MATRIX where BASE_id = %s</tt> </tt>
<a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line"><tt class="py-string">                       and VERSION = %s"""</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt id="link-58" class="py-name"><a title="Bio.motifs.jaspar.Motif.base_id" class="py-name" href="#" onclick="return doclink('link-58', 'base_id', 'link-19');">base_id</a></tt><tt class="py-op">,</tt> <tt id="link-59" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-59', 'version', 'link-20');">version</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line"> </tt>
<a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line">        <tt class="py-name">row</tt> <tt class="py-op">=</tt> <tt class="py-name">cur</tt><tt class="py-op">.</tt><tt id="link-60" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchone" class="py-name" href="#" onclick="return doclink('link-60', 'fetchone', 'link-52');">fetchone</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line"> </tt>
<a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line">        <tt class="py-name">int_id</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">row</tt><tt class="py-op">:</tt> </tt>
<a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line">            <tt class="py-name">int_id</tt> <tt class="py-op">=</tt> <tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L347"></a><tt class="py-lineno">347</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Failed to fetch internal database ID for JASPAR motif with matrix ID '{0}.{1}'. No JASPAR motif with this matrix ID appears to exist."</tt><tt class="py-op">.</tt><tt id="link-61" class="py-name"><a title="Bio.Align.Generic.Alignment.format
Bio.Motif._Motif.Motif.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-61', 'format', 'link-53');">format</a></tt><tt class="py-op">(</tt><tt id="link-62" class="py-name"><a title="Bio.motifs.jaspar.Motif.base_id" class="py-name" href="#" onclick="return doclink('link-62', 'base_id', 'link-19');">base_id</a></tt><tt class="py-op">,</tt> <tt id="link-63" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-63', 'version', 'link-20');">version</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-64" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-64', 'BiopythonWarning', 'link-1');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line"> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">int_id</tt> </tt>
</div><a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line"> </tt>
<a name="JASPAR5._fetch_motif_by_internal_id"></a><div id="JASPAR5._fetch_motif_by_internal_id-def"><a name="L352"></a><tt class="py-lineno">352</tt> <a class="py-toggle" href="#" id="JASPAR5._fetch_motif_by_internal_id-toggle" onclick="return toggle('JASPAR5._fetch_motif_by_internal_id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.jaspar.db.JASPAR5-class.html#_fetch_motif_by_internal_id">_fetch_motif_by_internal_id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">int_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="JASPAR5._fetch_motif_by_internal_id-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="JASPAR5._fetch_motif_by_internal_id-expanded"><a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line">        <tt class="py-comment"># fetch basic motif information</tt> </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line">        <tt class="py-name">cur</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbh</tt><tt class="py-op">.</tt><tt id="link-65" class="py-name"><a title="Bio.DocSQL.Query.cursor
Bio.DocSQL.QuerySingle.cursor" class="py-name" href="#" onclick="return doclink('link-65', 'cursor', 'link-49');">cursor</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">        <tt class="py-name">cur</tt><tt class="py-op">.</tt><tt id="link-66" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-66', 'execute', 'link-50');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"""select BASE_ID, VERSION, COLLECTION, NAME from MATRIX</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line"><tt class="py-string">                       where id = %s"""</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">int_id</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line"> </tt>
<a name="L358"></a><tt class="py-lineno">358</tt>  <tt class="py-line">        <tt class="py-name">row</tt> <tt class="py-op">=</tt> <tt class="py-name">cur</tt><tt class="py-op">.</tt><tt id="link-67" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchone" class="py-name" href="#" onclick="return doclink('link-67', 'fetchone', 'link-52');">fetchone</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line"> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line">        <tt class="py-comment"># This should never happen as it is an internal method. If it does</tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line">        <tt class="py-comment"># we should probably raise an exception</tt> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">row</tt><tt class="py-op">:</tt> </tt>
<a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Could not fetch JASPAR motif with internal ID = {0}"</tt><tt class="py-op">.</tt><tt id="link-68" class="py-name"><a title="Bio.Align.Generic.Alignment.format
Bio.Motif._Motif.Motif.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-68', 'format', 'link-53');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">int_id</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-69" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-69', 'BiopythonWarning', 'link-1');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
<a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line"> </tt>
<a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line">        <tt id="link-70" class="py-name"><a title="Bio.motifs.jaspar.Motif.base_id" class="py-name" href="#" onclick="return doclink('link-70', 'base_id', 'link-19');">base_id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line">        <tt id="link-71" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-71', 'version', 'link-20');">version</a></tt> <tt class="py-op">=</tt> <tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L368"></a><tt class="py-lineno">368</tt>  <tt class="py-line">        <tt class="py-name">collection</tt> <tt class="py-op">=</tt> <tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> </tt>
<a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line">        <tt id="link-72" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-72', 'name', 'link-15');">name</a></tt> <tt class="py-op">=</tt> <tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt> </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line"> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line">        <tt class="py-name">matrix_id</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt id="link-73" class="py-name"><a title="Bio.motifs.jaspar.Motif.base_id" class="py-name" href="#" onclick="return doclink('link-73', 'base_id', 'link-19');">base_id</a></tt><tt class="py-op">,</tt> <tt class="py-string">'.'</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-74" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-74', 'version', 'link-20');">version</a></tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line"> </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line">        <tt class="py-comment"># fetch the counts matrix</tt> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line">        <tt class="py-name">counts</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-75" class="py-name" targets="Method Bio.motifs.jaspar.db.JASPAR5._fetch_counts_matrix()=Bio.motifs.jaspar.db.JASPAR5-class.html#_fetch_counts_matrix"><a title="Bio.motifs.jaspar.db.JASPAR5._fetch_counts_matrix" class="py-name" href="#" onclick="return doclink('link-75', '_fetch_counts_matrix', 'link-75');">_fetch_counts_matrix</a></tt><tt class="py-op">(</tt><tt class="py-name">int_id</tt><tt class="py-op">)</tt> </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line"> </tt>
<a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line">        <tt class="py-comment"># Create new JASPAR motif</tt> </tt>
<a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line">        <tt class="py-name">motif</tt> <tt class="py-op">=</tt> <tt id="link-76" class="py-name"><a title="Bio.motifs.jaspar" class="py-name" href="#" onclick="return doclink('link-76', 'jaspar', 'link-12');">jaspar</a></tt><tt class="py-op">.</tt><tt id="link-77" class="py-name" targets="Package Bio.Motif=Bio.Motif-module.html,Class Bio.Motif._Motif.Motif=Bio.Motif._Motif.Motif-class.html,Module Bio.NeuralNetwork.Gene.Motif=Bio.NeuralNetwork.Gene.Motif-module.html,Class Bio.motifs.Motif=Bio.motifs.Motif-class.html,Class Bio.motifs.jaspar.Motif=Bio.motifs.jaspar.Motif-class.html,Class Bio.motifs.meme.Motif=Bio.motifs.meme.Motif-class.html,Class Bio.motifs.transfac.Motif=Bio.motifs.transfac.Motif-class.html"><a title="Bio.Motif
Bio.Motif._Motif.Motif
Bio.NeuralNetwork.Gene.Motif
Bio.motifs.Motif
Bio.motifs.jaspar.Motif
Bio.motifs.meme.Motif
Bio.motifs.transfac.Motif" class="py-name" href="#" onclick="return doclink('link-77', 'Motif', 'link-77');">Motif</a></tt><tt class="py-op">(</tt> </tt>
<a name="L378"></a><tt class="py-lineno">378</tt>  <tt class="py-line">            <tt class="py-name">matrix_id</tt><tt class="py-op">,</tt> <tt id="link-78" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-78', 'name', 'link-15');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">collection</tt><tt class="py-op">=</tt><tt class="py-name">collection</tt><tt class="py-op">,</tt> <tt class="py-name">counts</tt><tt class="py-op">=</tt><tt class="py-name">counts</tt> </tt>
<a name="L379"></a><tt class="py-lineno">379</tt>  <tt class="py-line">        <tt class="py-op">)</tt> </tt>
<a name="L380"></a><tt class="py-lineno">380</tt>  <tt class="py-line"> </tt>
<a name="L381"></a><tt class="py-lineno">381</tt>  <tt class="py-line">        <tt class="py-comment"># fetch species</tt> </tt>
<a name="L382"></a><tt class="py-lineno">382</tt>  <tt class="py-line">        <tt class="py-name">cur</tt><tt class="py-op">.</tt><tt id="link-79" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-79', 'execute', 'link-50');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"""select TAX_ID from MATRIX_SPECIES</tt> </tt>
<a name="L383"></a><tt class="py-lineno">383</tt>  <tt class="py-line"><tt class="py-string">                       where id = %s"""</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">int_id</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L384"></a><tt class="py-lineno">384</tt>  <tt class="py-line">        <tt class="py-name">tax_ids</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L385"></a><tt class="py-lineno">385</tt>  <tt class="py-line">        <tt class="py-name">rows</tt> <tt class="py-op">=</tt> <tt class="py-name">cur</tt><tt class="py-op">.</tt><tt id="link-80" class="py-name" targets="Method BioSQL.BioSeqDatabase._CursorWrapper.fetchall()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#fetchall"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchall" class="py-name" href="#" onclick="return doclink('link-80', 'fetchall', 'link-80');">fetchall</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L386"></a><tt class="py-lineno">386</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">row</tt> <tt class="py-keyword">in</tt> <tt class="py-name">rows</tt><tt class="py-op">:</tt> </tt>
<a name="L387"></a><tt class="py-lineno">387</tt>  <tt class="py-line">            <tt class="py-name">tax_ids</tt><tt class="py-op">.</tt><tt id="link-81" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-81', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L388"></a><tt class="py-lineno">388</tt>  <tt class="py-line"> </tt>
<a name="L389"></a><tt class="py-lineno">389</tt>  <tt class="py-line">        <tt class="py-comment"># Many JASPAR motifs (especially those not in the CORE collection)</tt> </tt>
<a name="L390"></a><tt class="py-lineno">390</tt>  <tt class="py-line">        <tt class="py-comment"># do not have taxonomy IDs. So this warning would get annoying.</tt> </tt>
<a name="L391"></a><tt class="py-lineno">391</tt>  <tt class="py-line">        <tt class="py-comment">#if not tax_ids:</tt> </tt>
<a name="L392"></a><tt class="py-lineno">392</tt>  <tt class="py-line">        <tt class="py-comment">#    warnings.warn("Could not fetch any taxonomy IDs for JASPAR motif {0}".format(motif.matrix_id), BiopythonWarning)</tt> </tt>
<a name="L393"></a><tt class="py-lineno">393</tt>  <tt class="py-line"> </tt>
<a name="L394"></a><tt class="py-lineno">394</tt>  <tt class="py-line">        <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt id="link-82" class="py-name"><a title="Bio.Pathway.Network.species
Bio.Pathway.Reaction.species
Bio.Pathway.System.species" class="py-name" href="#" onclick="return doclink('link-82', 'species', 'link-37');">species</a></tt> <tt class="py-op">=</tt> <tt class="py-name">tax_ids</tt> </tt>
<a name="L395"></a><tt class="py-lineno">395</tt>  <tt class="py-line"> </tt>
<a name="L396"></a><tt class="py-lineno">396</tt>  <tt class="py-line">        <tt class="py-comment"># fetch protein accession numbers</tt> </tt>
<a name="L397"></a><tt class="py-lineno">397</tt>  <tt class="py-line">        <tt class="py-name">cur</tt><tt class="py-op">.</tt><tt id="link-83" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-83', 'execute', 'link-50');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"select ACC FROM MATRIX_PROTEIN where id = %s"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">int_id</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L398"></a><tt class="py-lineno">398</tt>  <tt class="py-line">        <tt class="py-name">accs</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L399"></a><tt class="py-lineno">399</tt>  <tt class="py-line">        <tt class="py-name">rows</tt> <tt class="py-op">=</tt> <tt class="py-name">cur</tt><tt class="py-op">.</tt><tt id="link-84" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchall" class="py-name" href="#" onclick="return doclink('link-84', 'fetchall', 'link-80');">fetchall</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L400"></a><tt class="py-lineno">400</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">row</tt> <tt class="py-keyword">in</tt> <tt class="py-name">rows</tt><tt class="py-op">:</tt> </tt>
<a name="L401"></a><tt class="py-lineno">401</tt>  <tt class="py-line">            <tt class="py-name">accs</tt><tt class="py-op">.</tt><tt id="link-85" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-85', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L402"></a><tt class="py-lineno">402</tt>  <tt class="py-line"> </tt>
<a name="L403"></a><tt class="py-lineno">403</tt>  <tt class="py-line">        <tt class="py-comment"># Similarly as for taxonomy IDs, it would get annoying to print</tt> </tt>
<a name="L404"></a><tt class="py-lineno">404</tt>  <tt class="py-line">        <tt class="py-comment"># warnings for JASPAR motifs which do not have accession numbers.</tt> </tt>
<a name="L405"></a><tt class="py-lineno">405</tt>  <tt class="py-line"> </tt>
<a name="L406"></a><tt class="py-lineno">406</tt>  <tt class="py-line">        <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt id="link-86" class="py-name" targets="Variable Bio.FSSP.fssp_rec.align.acc=Bio.FSSP.fssp_rec.align-class.html#acc"><a title="Bio.FSSP.fssp_rec.align.acc" class="py-name" href="#" onclick="return doclink('link-86', 'acc', 'link-86');">acc</a></tt> <tt class="py-op">=</tt> <tt class="py-name">accs</tt> </tt>
<a name="L407"></a><tt class="py-lineno">407</tt>  <tt class="py-line"> </tt>
<a name="L408"></a><tt class="py-lineno">408</tt>  <tt class="py-line">        <tt class="py-comment"># fetch remaining annotation as tags from the ANNOTATION table</tt> </tt>
<a name="L409"></a><tt class="py-lineno">409</tt>  <tt class="py-line">        <tt class="py-name">cur</tt><tt class="py-op">.</tt><tt id="link-87" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-87', 'execute', 'link-50');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"""select TAG, VAL from MATRIX_ANNOTATION</tt> </tt>
<a name="L410"></a><tt class="py-lineno">410</tt>  <tt class="py-line"><tt class="py-string">                       where id = %s"""</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">int_id</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L411"></a><tt class="py-lineno">411</tt>  <tt class="py-line">        <tt class="py-name">rows</tt> <tt class="py-op">=</tt> <tt class="py-name">cur</tt><tt class="py-op">.</tt><tt id="link-88" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchall" class="py-name" href="#" onclick="return doclink('link-88', 'fetchall', 'link-80');">fetchall</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L412"></a><tt class="py-lineno">412</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">row</tt> <tt class="py-keyword">in</tt> <tt class="py-name">rows</tt><tt class="py-op">:</tt> </tt>
<a name="L413"></a><tt class="py-lineno">413</tt>  <tt class="py-line">            <tt class="py-name">attr</tt> <tt class="py-op">=</tt> <tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L414"></a><tt class="py-lineno">414</tt>  <tt class="py-line">            <tt class="py-name">val</tt> <tt class="py-op">=</tt> <tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L415"></a><tt class="py-lineno">415</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">attr</tt> <tt class="py-op">==</tt> <tt class="py-string">'class'</tt><tt class="py-op">:</tt> </tt>
<a name="L416"></a><tt class="py-lineno">416</tt>  <tt class="py-line">                <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt class="py-name">tf_class</tt> <tt class="py-op">=</tt> <tt class="py-name">val</tt> </tt>
<a name="L417"></a><tt class="py-lineno">417</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">attr</tt> <tt class="py-op">==</tt> <tt class="py-string">'family'</tt><tt class="py-op">:</tt> </tt>
<a name="L418"></a><tt class="py-lineno">418</tt>  <tt class="py-line">                <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt class="py-name">tf_family</tt> <tt class="py-op">=</tt> <tt class="py-name">val</tt> </tt>
<a name="L419"></a><tt class="py-lineno">419</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">attr</tt> <tt class="py-op">==</tt> <tt class="py-string">'tax_group'</tt><tt class="py-op">:</tt> </tt>
<a name="L420"></a><tt class="py-lineno">420</tt>  <tt class="py-line">                <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt class="py-name">tax_group</tt> <tt class="py-op">=</tt> <tt class="py-name">val</tt> </tt>
<a name="L421"></a><tt class="py-lineno">421</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">attr</tt> <tt class="py-op">==</tt> <tt class="py-string">'type'</tt><tt class="py-op">:</tt> </tt>
<a name="L422"></a><tt class="py-lineno">422</tt>  <tt class="py-line">                <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt class="py-name">data_type</tt> <tt class="py-op">=</tt> <tt class="py-name">val</tt> </tt>
<a name="L423"></a><tt class="py-lineno">423</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">attr</tt> <tt class="py-op">==</tt> <tt class="py-string">'pazar_tf_id'</tt><tt class="py-op">:</tt> </tt>
<a name="L424"></a><tt class="py-lineno">424</tt>  <tt class="py-line">                <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt class="py-name">pazar_id</tt> <tt class="py-op">=</tt> <tt class="py-name">val</tt> </tt>
<a name="L425"></a><tt class="py-lineno">425</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">attr</tt> <tt class="py-op">==</tt> <tt class="py-string">'medline'</tt><tt class="py-op">:</tt> </tt>
<a name="L426"></a><tt class="py-lineno">426</tt>  <tt class="py-line">                <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt class="py-name">medline</tt> <tt class="py-op">=</tt> <tt class="py-name">val</tt> </tt>
<a name="L427"></a><tt class="py-lineno">427</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">attr</tt> <tt class="py-op">==</tt> <tt class="py-string">'comment'</tt><tt class="py-op">:</tt> </tt>
<a name="L428"></a><tt class="py-lineno">428</tt>  <tt class="py-line">                <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt id="link-89" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.comment()=Bio.GenBank._FeatureConsumer-class.html#comment,Method Bio.GenBank._RecordConsumer.comment()=Bio.GenBank._RecordConsumer-class.html#comment"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-89', 'comment', 'link-89');">comment</a></tt> <tt class="py-op">=</tt> <tt class="py-name">val</tt> </tt>
<a name="L429"></a><tt class="py-lineno">429</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L430"></a><tt class="py-lineno">430</tt>  <tt class="py-line">                <tt class="py-string">"""</tt> </tt>
<a name="L431"></a><tt class="py-lineno">431</tt>  <tt class="py-line"><tt class="py-string">                TODO If we were to implement additional abitrary tags</tt> </tt>
<a name="L432"></a><tt class="py-lineno">432</tt>  <tt class="py-line"><tt class="py-string">                motif.tag(attr, val)</tt> </tt>
<a name="L433"></a><tt class="py-lineno">433</tt>  <tt class="py-line"><tt class="py-string">                """</tt> </tt>
<a name="L434"></a><tt class="py-lineno">434</tt>  <tt class="py-line">                <tt class="py-keyword">pass</tt> </tt>
<a name="L435"></a><tt class="py-lineno">435</tt>  <tt class="py-line"> </tt>
<a name="L436"></a><tt class="py-lineno">436</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">motif</tt> </tt>
</div><a name="L437"></a><tt class="py-lineno">437</tt>  <tt class="py-line"> </tt>
<a name="JASPAR5._fetch_counts_matrix"></a><div id="JASPAR5._fetch_counts_matrix-def"><a name="L438"></a><tt class="py-lineno">438</tt> <a class="py-toggle" href="#" id="JASPAR5._fetch_counts_matrix-toggle" onclick="return toggle('JASPAR5._fetch_counts_matrix');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.jaspar.db.JASPAR5-class.html#_fetch_counts_matrix">_fetch_counts_matrix</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">int_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="JASPAR5._fetch_counts_matrix-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="JASPAR5._fetch_counts_matrix-expanded"><a name="L439"></a><tt class="py-lineno">439</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L440"></a><tt class="py-lineno">440</tt>  <tt class="py-line"><tt class="py-docstring">        Fetch the counts matrix from the JASPAR DB by the internal ID</tt> </tt>
<a name="L441"></a><tt class="py-lineno">441</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L442"></a><tt class="py-lineno">442</tt>  <tt class="py-line"><tt class="py-docstring">        Returns a Bio.motifs.matrix.GenericPositionMatrix</tt> </tt>
<a name="L443"></a><tt class="py-lineno">443</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L444"></a><tt class="py-lineno">444</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L445"></a><tt class="py-lineno">445</tt>  <tt class="py-line">        <tt class="py-name">counts</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L446"></a><tt class="py-lineno">446</tt>  <tt class="py-line">        <tt class="py-name">cur</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbh</tt><tt class="py-op">.</tt><tt id="link-90" class="py-name"><a title="Bio.DocSQL.Query.cursor
Bio.DocSQL.QuerySingle.cursor" class="py-name" href="#" onclick="return doclink('link-90', 'cursor', 'link-49');">cursor</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L447"></a><tt class="py-lineno">447</tt>  <tt class="py-line"> </tt>
<a name="L448"></a><tt class="py-lineno">448</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">base</tt> <tt class="py-keyword">in</tt> <tt id="link-91" class="py-name"><a title="Bio.Restriction.Restriction.AanI.dna
Bio.Restriction.Restriction.AarI.dna
Bio.Restriction.Restriction.AasI.dna
Bio.Restriction.Restriction.AatII.dna
Bio.Restriction.Restriction.AbaSI.dna
Bio.Restriction.Restriction.AbsI.dna
Bio.Restriction.Restriction.Acc16I.dna
Bio.Restriction.Restriction.Acc36I.dna
Bio.Restriction.Restriction.Acc65I.dna
Bio.Restriction.Restriction.AccB1I.dna
Bio.Restriction.Restriction.AccB7I.dna
Bio.Restriction.Restriction.AccBSI.dna
Bio.Restriction.Restriction.AccI.dna
Bio.Restriction.Restriction.AccII.dna
Bio.Restriction.Restriction.AccIII.dna
Bio.Restriction.Restriction.AceIII.dna
Bio.Restriction.Restriction.AciI.dna
Bio.Restriction.Restriction.AclI.dna
Bio.Restriction.Restriction.AclWI.dna
Bio.Restriction.Restriction.AcoI.dna
Bio.Restriction.Restriction.AcsI.dna
Bio.Restriction.Restriction.AcuI.dna
Bio.Restriction.Restriction.AcvI.dna
Bio.Restriction.Restriction.AcyI.dna
Bio.Restriction.Restriction.AdeI.dna
Bio.Restriction.Restriction.AfaI.dna
Bio.Restriction.Restriction.AfeI.dna
Bio.Restriction.Restriction.AfiI.dna
Bio.Restriction.Restriction.AflII.dna
Bio.Restriction.Restriction.AflIII.dna
Bio.Restriction.Restriction.AgeI.dna
Bio.Restriction.Restriction.AgsI.dna
Bio.Restriction.Restriction.AhaIII.dna
Bio.Restriction.Restriction.AhdI.dna
Bio.Restriction.Restriction.AhlI.dna
Bio.Restriction.Restriction.AjiI.dna
Bio.Restriction.Restriction.AjnI.dna
Bio.Restriction.Restriction.AjuI.dna
Bio.Restriction.Restriction.AleI.dna
Bio.Restriction.Restriction.AlfI.dna
Bio.Restriction.Restriction.AloI.dna
Bio.Restriction.Restriction.AluBI.dna
Bio.Restriction.Restriction.AluI.dna
Bio.Restriction.Restriction.Alw21I.dna
Bio.Restriction.Restriction.Alw26I.dna
Bio.Restriction.Restriction.Alw44I.dna
Bio.Restriction.Restriction.AlwFI.dna
Bio.Restriction.Restriction.AlwI.dna
Bio.Restriction.Restriction.AlwNI.dna
Bio.Restriction.Restriction.Ama87I.dna
Bio.Restriction.Restriction.Aor13HI.dna
Bio.Restriction.Restriction.Aor51HI.dna
Bio.Restriction.Restriction.AoxI.dna
Bio.Restriction.Restriction.ApaBI.dna
Bio.Restriction.Restriction.ApaI.dna
Bio.Restriction.Restriction.ApaLI.dna
Bio.Restriction.Restriction.ApeKI.dna
Bio.Restriction.Restriction.ApoI.dna
Bio.Restriction.Restriction.ApyPI.dna
Bio.Restriction.Restriction.AquII.dna
Bio.Restriction.Restriction.AquIII.dna
Bio.Restriction.Restriction.AquIV.dna
Bio.Restriction.Restriction.ArsI.dna
Bio.Restriction.Restriction.AscI.dna
Bio.Restriction.Restriction.AseI.dna
Bio.Restriction.Restriction.Asi256I.dna
Bio.Restriction.Restriction.AsiGI.dna
Bio.Restriction.Restriction.AsiSI.dna
Bio.Restriction.Restriction.Asp700I.dna
Bio.Restriction.Restriction.Asp718I.dna
Bio.Restriction.Restriction.AspA2I.dna
Bio.Restriction.Restriction.AspBHI.dna
Bio.Restriction.Restriction.AspLEI.dna
Bio.Restriction.Restriction.AspS9I.dna
Bio.Restriction.Restriction.AssI.dna
Bio.Restriction.Restriction.AsuC2I.dna
Bio.Restriction.Restriction.AsuHPI.dna
Bio.Restriction.Restriction.AsuI.dna
Bio.Restriction.Restriction.AsuII.dna
Bio.Restriction.Restriction.AsuNHI.dna
Bio.Restriction.Restriction.AvaI.dna
Bio.Restriction.Restriction.AvaII.dna
Bio.Restriction.Restriction.AvaIII.dna
Bio.Restriction.Restriction.AvrII.dna
Bio.Restriction.Restriction.AxyI.dna
Bio.Restriction.Restriction.BaeGI.dna
Bio.Restriction.Restriction.BaeI.dna
Bio.Restriction.Restriction.BalI.dna
Bio.Restriction.Restriction.BamHI.dna
Bio.Restriction.Restriction.BanI.dna
Bio.Restriction.Restriction.BanII.dna
Bio.Restriction.Restriction.BarI.dna
Bio.Restriction.Restriction.BasI.dna
Bio.Restriction.Restriction.BauI.dna
Bio.Restriction.Restriction.Bbr7I.dna
Bio.Restriction.Restriction.BbrPI.dna
Bio.Restriction.Restriction.BbsI.dna
Bio.Restriction.Restriction.Bbv12I.dna
Bio.Restriction.Restriction.BbvCI.dna
Bio.Restriction.Restriction.BbvI.dna
Bio.Restriction.Restriction.BbvII.dna
Bio.Restriction.Restriction.BccI.dna
Bio.Restriction.Restriction.Bce83I.dna
Bio.Restriction.Restriction.BceAI.dna
Bio.Restriction.Restriction.BcefI.dna
Bio.Restriction.Restriction.BcgI.dna
Bio.Restriction.Restriction.BciT130I.dna
Bio.Restriction.Restriction.BciVI.dna
Bio.Restriction.Restriction.BclI.dna
Bio.Restriction.Restriction.BcnI.dna
Bio.Restriction.Restriction.BcoDI.dna
Bio.Restriction.Restriction.BcuI.dna
Bio.Restriction.Restriction.BdaI.dna
Bio.Restriction.Restriction.BetI.dna
Bio.Restriction.Restriction.BfaI.dna
Bio.Restriction.Restriction.BfiI.dna
Bio.Restriction.Restriction.BfmI.dna
Bio.Restriction.Restriction.BfoI.dna
Bio.Restriction.Restriction.BfrI.dna
Bio.Restriction.Restriction.BfuAI.dna
Bio.Restriction.Restriction.BfuCI.dna
Bio.Restriction.Restriction.BfuI.dna
Bio.Restriction.Restriction.BglI.dna
Bio.Restriction.Restriction.BglII.dna
Bio.Restriction.Restriction.BinI.dna
Bio.Restriction.Restriction.BisI.dna
Bio.Restriction.Restriction.BlnI.dna
Bio.Restriction.Restriction.BlpI.dna
Bio.Restriction.Restriction.BlsI.dna
Bio.Restriction.Restriction.BmcAI.dna
Bio.Restriction.Restriction.Bme1390I.dna
Bio.Restriction.Restriction.Bme18I.dna
Bio.Restriction.Restriction.BmeDI.dna
Bio.Restriction.Restriction.BmeRI.dna
Bio.Restriction.Restriction.BmeT110I.dna
Bio.Restriction.Restriction.BmgBI.dna
Bio.Restriction.Restriction.BmgI.dna
Bio.Restriction.Restriction.BmgT120I.dna
Bio.Restriction.Restriction.BmiI.dna
Bio.Restriction.Restriction.BmrFI.dna
Bio.Restriction.Restriction.BmrI.dna
Bio.Restriction.Restriction.BmsI.dna
Bio.Restriction.Restriction.BmtI.dna
Bio.Restriction.Restriction.BmuI.dna
Bio.Restriction.Restriction.BoxI.dna
Bio.Restriction.Restriction.BpiI.dna
Bio.Restriction.Restriction.BplI.dna
Bio.Restriction.Restriction.BpmI.dna
Bio.Restriction.Restriction.Bpu10I.dna
Bio.Restriction.Restriction.Bpu1102I.dna
Bio.Restriction.Restriction.Bpu14I.dna
Bio.Restriction.Restriction.BpuEI.dna
Bio.Restriction.Restriction.BpuMI.dna
Bio.Restriction.Restriction.BpvUI.dna
Bio.Restriction.Restriction.Bsa29I.dna
Bio.Restriction.Restriction.BsaAI.dna
Bio.Restriction.Restriction.BsaBI.dna
Bio.Restriction.Restriction.BsaHI.dna
Bio.Restriction.Restriction.BsaI.dna
Bio.Restriction.Restriction.BsaJI.dna
Bio.Restriction.Restriction.BsaWI.dna
Bio.Restriction.Restriction.BsaXI.dna
Bio.Restriction.Restriction.BsbI.dna
Bio.Restriction.Restriction.Bsc4I.dna
Bio.Restriction.Restriction.BscAI.dna
Bio.Restriction.Restriction.BscGI.dna
Bio.Restriction.Restriction.Bse118I.dna
Bio.Restriction.Restriction.Bse1I.dna
Bio.Restriction.Restriction.Bse21I.dna
Bio.Restriction.Restriction.Bse3DI.dna
Bio.Restriction.Restriction.Bse8I.dna
Bio.Restriction.Restriction.BseAI.dna
Bio.Restriction.Restriction.BseBI.dna
Bio.Restriction.Restriction.BseCI.dna
Bio.Restriction.Restriction.BseDI.dna
Bio.Restriction.Restriction.BseGI.dna
Bio.Restriction.Restriction.BseJI.dna
Bio.Restriction.Restriction.BseLI.dna
Bio.Restriction.Restriction.BseMI.dna
Bio.Restriction.Restriction.BseMII.dna
Bio.Restriction.Restriction.BseNI.dna
Bio.Restriction.Restriction.BsePI.dna
Bio.Restriction.Restriction.BseRI.dna
Bio.Restriction.Restriction.BseSI.dna
Bio.Restriction.Restriction.BseX3I.dna
Bio.Restriction.Restriction.BseXI.dna
Bio.Restriction.Restriction.BseYI.dna
Bio.Restriction.Restriction.BsgI.dna
Bio.Restriction.Restriction.Bsh1236I.dna
Bio.Restriction.Restriction.Bsh1285I.dna
Bio.Restriction.Restriction.BshFI.dna
Bio.Restriction.Restriction.BshNI.dna
Bio.Restriction.Restriction.BshTI.dna
Bio.Restriction.Restriction.BshVI.dna
Bio.Restriction.Restriction.BsiEI.dna
Bio.Restriction.Restriction.BsiHKAI.dna
Bio.Restriction.Restriction.BsiHKCI.dna
Bio.Restriction.Restriction.BsiI.dna
Bio.Restriction.Restriction.BsiSI.dna
Bio.Restriction.Restriction.BsiWI.dna
Bio.Restriction.Restriction.BsiYI.dna
Bio.Restriction.Restriction.BslFI.dna
Bio.Restriction.Restriction.BslI.dna
Bio.Restriction.Restriction.BsmAI.dna
Bio.Restriction.Restriction.BsmBI.dna
Bio.Restriction.Restriction.BsmFI.dna
Bio.Restriction.Restriction.BsmI.dna
Bio.Restriction.Restriction.BsnI.dna
Bio.Restriction.Restriction.Bso31I.dna
Bio.Restriction.Restriction.BsoBI.dna
Bio.Restriction.Restriction.Bsp119I.dna
Bio.Restriction.Restriction.Bsp120I.dna
Bio.Restriction.Restriction.Bsp1286I.dna
Bio.Restriction.Restriction.Bsp13I.dna
Bio.Restriction.Restriction.Bsp1407I.dna
Bio.Restriction.Restriction.Bsp143I.dna
Bio.Restriction.Restriction.Bsp1720I.dna
Bio.Restriction.Restriction.Bsp19I.dna
Bio.Restriction.Restriction.Bsp24I.dna
Bio.Restriction.Restriction.Bsp68I.dna
Bio.Restriction.Restriction.BspACI.dna
Bio.Restriction.Restriction.BspCNI.dna
Bio.Restriction.Restriction.BspD6I.dna
Bio.Restriction.Restriction.BspDI.dna
Bio.Restriction.Restriction.BspEI.dna
Bio.Restriction.Restriction.BspFNI.dna
Bio.Restriction.Restriction.BspGI.dna
Bio.Restriction.Restriction.BspHI.dna
Bio.Restriction.Restriction.BspLI.dna
Bio.Restriction.Restriction.BspLU11I.dna
Bio.Restriction.Restriction.BspMI.dna
Bio.Restriction.Restriction.BspMII.dna
Bio.Restriction.Restriction.BspNCI.dna
Bio.Restriction.Restriction.BspOI.dna
Bio.Restriction.Restriction.BspPI.dna
Bio.Restriction.Restriction.BspQI.dna
Bio.Restriction.Restriction.BspT104I.dna
Bio.Restriction.Restriction.BspT107I.dna
Bio.Restriction.Restriction.BspTI.dna
Bio.Restriction.Restriction.BsrBI.dna
Bio.Restriction.Restriction.BsrDI.dna
Bio.Restriction.Restriction.BsrFI.dna
Bio.Restriction.Restriction.BsrGI.dna
Bio.Restriction.Restriction.BsrI.dna
Bio.Restriction.Restriction.BsrSI.dna
Bio.Restriction.Restriction.BssAI.dna
Bio.Restriction.Restriction.BssECI.dna
Bio.Restriction.Restriction.BssHII.dna
Bio.Restriction.Restriction.BssKI.dna
Bio.Restriction.Restriction.BssMI.dna
Bio.Restriction.Restriction.BssNAI.dna
Bio.Restriction.Restriction.BssNI.dna
Bio.Restriction.Restriction.BssSI.dna
Bio.Restriction.Restriction.BssT1I.dna
Bio.Restriction.Restriction.Bst1107I.dna
Bio.Restriction.Restriction.Bst2BI.dna
Bio.Restriction.Restriction.Bst2UI.dna
Bio.Restriction.Restriction.Bst4CI.dna
Bio.Restriction.Restriction.Bst6I.dna
Bio.Restriction.Restriction.BstACI.dna
Bio.Restriction.Restriction.BstAFI.dna
Bio.Restriction.Restriction.BstAPI.dna
Bio.Restriction.Restriction.BstAUI.dna
Bio.Restriction.Restriction.BstBAI.dna
Bio.Restriction.Restriction.BstBI.dna
Bio.Restriction.Restriction.BstC8I.dna
Bio.Restriction.Restriction.BstDEI.dna
Bio.Restriction.Restriction.BstDSI.dna
Bio.Restriction.Restriction.BstEII.dna
Bio.Restriction.Restriction.BstENI.dna
Bio.Restriction.Restriction.BstF5I.dna
Bio.Restriction.Restriction.BstFNI.dna
Bio.Restriction.Restriction.BstH2I.dna
Bio.Restriction.Restriction.BstHHI.dna
Bio.Restriction.Restriction.BstKTI.dna
Bio.Restriction.Restriction.BstMAI.dna
Bio.Restriction.Restriction.BstMBI.dna
Bio.Restriction.Restriction.BstMCI.dna
Bio.Restriction.Restriction.BstMWI.dna
Bio.Restriction.Restriction.BstNI.dna
Bio.Restriction.Restriction.BstNSI.dna
Bio.Restriction.Restriction.BstOI.dna
Bio.Restriction.Restriction.BstPAI.dna
Bio.Restriction.Restriction.BstPI.dna
Bio.Restriction.Restriction.BstSCI.dna
Bio.Restriction.Restriction.BstSFI.dna
Bio.Restriction.Restriction.BstSLI.dna
Bio.Restriction.Restriction.BstSNI.dna
Bio.Restriction.Restriction.BstUI.dna
Bio.Restriction.Restriction.BstV1I.dna
Bio.Restriction.Restriction.BstV2I.dna
Bio.Restriction.Restriction.BstX2I.dna
Bio.Restriction.Restriction.BstXI.dna
Bio.Restriction.Restriction.BstYI.dna
Bio.Restriction.Restriction.BstZ17I.dna
Bio.Restriction.Restriction.BstZI.dna
Bio.Restriction.Restriction.Bsu15I.dna
Bio.Restriction.Restriction.Bsu36I.dna
Bio.Restriction.Restriction.BsuI.dna
Bio.Restriction.Restriction.BsuRI.dna
Bio.Restriction.Restriction.BtgI.dna
Bio.Restriction.Restriction.BtgZI.dna
Bio.Restriction.Restriction.BthCI.dna
Bio.Restriction.Restriction.BtrI.dna
Bio.Restriction.Restriction.BtsCI.dna
Bio.Restriction.Restriction.BtsI.dna
Bio.Restriction.Restriction.BtsIMutI.dna
Bio.Restriction.Restriction.BtuMI.dna
Bio.Restriction.Restriction.BveI.dna
Bio.Restriction.Restriction.Cac8I.dna
Bio.Restriction.Restriction.CaiI.dna
Bio.Restriction.Restriction.CauII.dna
Bio.Restriction.Restriction.CchII.dna
Bio.Restriction.Restriction.CchIII.dna
Bio.Restriction.Restriction.CciI.dna
Bio.Restriction.Restriction.CciNI.dna
Bio.Restriction.Restriction.Cdi630V.dna
Bio.Restriction.Restriction.CdiI.dna
Bio.Restriction.Restriction.CdpI.dna
Bio.Restriction.Restriction.CfoI.dna
Bio.Restriction.Restriction.Cfr10I.dna
Bio.Restriction.Restriction.Cfr13I.dna
Bio.Restriction.Restriction.Cfr42I.dna
Bio.Restriction.Restriction.Cfr9I.dna
Bio.Restriction.Restriction.CfrI.dna
Bio.Restriction.Restriction.Cgl13032I.dna
Bio.Restriction.Restriction.Cgl13032II.dna
Bio.Restriction.Restriction.ChaI.dna
Bio.Restriction.Restriction.CjeFIII.dna
Bio.Restriction.Restriction.CjeFV.dna
Bio.Restriction.Restriction.CjeI.dna
Bio.Restriction.Restriction.CjeNII.dna
Bio.Restriction.Restriction.CjeNIII.dna
Bio.Restriction.Restriction.CjeP659IV.dna
Bio.Restriction.Restriction.CjePI.dna
Bio.Restriction.Restriction.CjuI.dna
Bio.Restriction.Restriction.CjuII.dna
Bio.Restriction.Restriction.ClaI.dna
Bio.Restriction.Restriction.CpoI.dna
Bio.Restriction.Restriction.CseI.dna
Bio.Restriction.Restriction.CsiI.dna
Bio.Restriction.Restriction.Csp6I.dna
Bio.Restriction.Restriction.CspAI.dna
Bio.Restriction.Restriction.CspCI.dna
Bio.Restriction.Restriction.CspI.dna
Bio.Restriction.Restriction.CstMI.dna
Bio.Restriction.Restriction.CviAII.dna
Bio.Restriction.Restriction.CviJI.dna
Bio.Restriction.Restriction.CviKI_1.dna
Bio.Restriction.Restriction.CviQI.dna
Bio.Restriction.Restriction.CviRI.dna
Bio.Restriction.Restriction.DdeI.dna
Bio.Restriction.Restriction.DinI.dna
Bio.Restriction.Restriction.DpnI.dna
Bio.Restriction.Restriction.DpnII.dna
Bio.Restriction.Restriction.DraI.dna
Bio.Restriction.Restriction.DraII.dna
Bio.Restriction.Restriction.DraIII.dna
Bio.Restriction.Restriction.DraRI.dna
Bio.Restriction.Restriction.DrdI.dna
Bio.Restriction.Restriction.DrdII.dna
Bio.Restriction.Restriction.DriI.dna
Bio.Restriction.Restriction.DsaI.dna
Bio.Restriction.Restriction.DseDI.dna
Bio.Restriction.Restriction.EaeI.dna
Bio.Restriction.Restriction.EagI.dna
Bio.Restriction.Restriction.Eam1104I.dna
Bio.Restriction.Restriction.Eam1105I.dna
Bio.Restriction.Restriction.EarI.dna
Bio.Restriction.Restriction.EciI.dna
Bio.Restriction.Restriction.Ecl136II.dna
Bio.Restriction.Restriction.EclXI.dna
Bio.Restriction.Restriction.Eco105I.dna
Bio.Restriction.Restriction.Eco130I.dna
Bio.Restriction.Restriction.Eco147I.dna
Bio.Restriction.Restriction.Eco24I.dna
Bio.Restriction.Restriction.Eco31I.dna
Bio.Restriction.Restriction.Eco32I.dna
Bio.Restriction.Restriction.Eco47I.dna
Bio.Restriction.Restriction.Eco47III.dna
Bio.Restriction.Restriction.Eco52I.dna
Bio.Restriction.Restriction.Eco53kI.dna
Bio.Restriction.Restriction.Eco57I.dna
Bio.Restriction.Restriction.Eco57MI.dna
Bio.Restriction.Restriction.Eco72I.dna
Bio.Restriction.Restriction.Eco81I.dna
Bio.Restriction.Restriction.Eco88I.dna
Bio.Restriction.Restriction.Eco91I.dna
Bio.Restriction.Restriction.EcoHI.dna
Bio.Restriction.Restriction.EcoICRI.dna
Bio.Restriction.Restriction.EcoNI.dna
Bio.Restriction.Restriction.EcoO109I.dna
Bio.Restriction.Restriction.EcoO65I.dna
Bio.Restriction.Restriction.EcoRI.dna
Bio.Restriction.Restriction.EcoRII.dna
Bio.Restriction.Restriction.EcoRV.dna
Bio.Restriction.Restriction.EcoT14I.dna
Bio.Restriction.Restriction.EcoT22I.dna
Bio.Restriction.Restriction.EcoT38I.dna
Bio.Restriction.Restriction.EgeI.dna
Bio.Restriction.Restriction.EheI.dna
Bio.Restriction.Restriction.ErhI.dna
Bio.Restriction.Restriction.EsaBC3I.dna
Bio.Restriction.Restriction.EsaSSI.dna
Bio.Restriction.Restriction.Esp3I.dna
Bio.Restriction.Restriction.EspI.dna
Bio.Restriction.Restriction.FaeI.dna
Bio.Restriction.Restriction.FaiI.dna
Bio.Restriction.Restriction.FalI.dna
Bio.Restriction.Restriction.FaqI.dna
Bio.Restriction.Restriction.FatI.dna
Bio.Restriction.Restriction.FauI.dna
Bio.Restriction.Restriction.FauNDI.dna
Bio.Restriction.Restriction.FbaI.dna
Bio.Restriction.Restriction.FblI.dna
Bio.Restriction.Restriction.FinI.dna
Bio.Restriction.Restriction.FmuI.dna
Bio.Restriction.Restriction.Fnu4HI.dna
Bio.Restriction.Restriction.FnuDII.dna
Bio.Restriction.Restriction.FokI.dna
Bio.Restriction.Restriction.FriOI.dna
Bio.Restriction.Restriction.FseI.dna
Bio.Restriction.Restriction.Fsp4HI.dna
Bio.Restriction.Restriction.FspAI.dna
Bio.Restriction.Restriction.FspBI.dna
Bio.Restriction.Restriction.FspEI.dna
Bio.Restriction.Restriction.FspI.dna
Bio.Restriction.Restriction.GauT27I.dna
Bio.Restriction.Restriction.GdiII.dna
Bio.Restriction.Restriction.GlaI.dna
Bio.Restriction.Restriction.GluI.dna
Bio.Restriction.Restriction.GsaI.dna
Bio.Restriction.Restriction.GsuI.dna
Bio.Restriction.Restriction.HaeI.dna
Bio.Restriction.Restriction.HaeII.dna
Bio.Restriction.Restriction.HaeIII.dna
Bio.Restriction.Restriction.HapII.dna
Bio.Restriction.Restriction.HauII.dna
Bio.Restriction.Restriction.HgaI.dna
Bio.Restriction.Restriction.HgiAI.dna
Bio.Restriction.Restriction.HgiCI.dna
Bio.Restriction.Restriction.HgiEII.dna
Bio.Restriction.Restriction.HgiJII.dna
Bio.Restriction.Restriction.HhaI.dna
Bio.Restriction.Restriction.Hin1I.dna
Bio.Restriction.Restriction.Hin1II.dna
Bio.Restriction.Restriction.Hin4I.dna
Bio.Restriction.Restriction.Hin4II.dna
Bio.Restriction.Restriction.Hin6I.dna
Bio.Restriction.Restriction.HinP1I.dna
Bio.Restriction.Restriction.HincII.dna
Bio.Restriction.Restriction.HindII.dna
Bio.Restriction.Restriction.HindIII.dna
Bio.Restriction.Restriction.HinfI.dna
Bio.Restriction.Restriction.HpaI.dna
Bio.Restriction.Restriction.HpaII.dna
Bio.Restriction.Restriction.HphI.dna
Bio.Restriction.Restriction.Hpy166II.dna
Bio.Restriction.Restriction.Hpy178III.dna
Bio.Restriction.Restriction.Hpy188I.dna
Bio.Restriction.Restriction.Hpy188III.dna
Bio.Restriction.Restriction.Hpy8I.dna
Bio.Restriction.Restriction.Hpy99I.dna
Bio.Restriction.Restriction.Hpy99XIII.dna
Bio.Restriction.Restriction.Hpy99XIV.dna
Bio.Restriction.Restriction.HpyAV.dna
Bio.Restriction.Restriction.HpyCH4III.dna
Bio.Restriction.Restriction.HpyCH4IV.dna
Bio.Restriction.Restriction.HpyCH4V.dna
Bio.Restriction.Restriction.HpyF10VI.dna
Bio.Restriction.Restriction.HpyF3I.dna
Bio.Restriction.Restriction.HpySE526I.dna
Bio.Restriction.Restriction.Hsp92I.dna
Bio.Restriction.Restriction.Hsp92II.dna
Bio.Restriction.Restriction.HspAI.dna
Bio.Restriction.Restriction.Jma19592I.dna
Bio.Restriction.Restriction.KasI.dna
Bio.Restriction.Restriction.KflI.dna
Bio.Restriction.Restriction.Kpn2I.dna
Bio.Restriction.Restriction.KpnI.dna
Bio.Restriction.Restriction.KroI.dna
Bio.Restriction.Restriction.Ksp22I.dna
Bio.Restriction.Restriction.Ksp632I.dna
Bio.Restriction.Restriction.KspAI.dna
Bio.Restriction.Restriction.KspI.dna
Bio.Restriction.Restriction.Kzo9I.dna
Bio.Restriction.Restriction.LguI.dna
Bio.Restriction.Restriction.LpnI.dna
Bio.Restriction.Restriction.LpnPI.dna
Bio.Restriction.Restriction.Lsp1109I.dna
Bio.Restriction.Restriction.LweI.dna
Bio.Restriction.Restriction.MabI.dna
Bio.Restriction.Restriction.MaeI.dna
Bio.Restriction.Restriction.MaeII.dna
Bio.Restriction.Restriction.MaeIII.dna
Bio.Restriction.Restriction.MalI.dna
Bio.Restriction.Restriction.MaqI.dna
Bio.Restriction.Restriction.MauBI.dna
Bio.Restriction.Restriction.MbiI.dna
Bio.Restriction.Restriction.MboI.dna
Bio.Restriction.Restriction.MboII.dna
Bio.Restriction.Restriction.McaTI.dna
Bio.Restriction.Restriction.McrI.dna
Bio.Restriction.Restriction.MfeI.dna
Bio.Restriction.Restriction.MflI.dna
Bio.Restriction.Restriction.MhlI.dna
Bio.Restriction.Restriction.MjaIV.dna
Bio.Restriction.Restriction.MkaDII.dna
Bio.Restriction.Restriction.MlsI.dna
Bio.Restriction.Restriction.MluCI.dna
Bio.Restriction.Restriction.MluI.dna
Bio.Restriction.Restriction.MluNI.dna
Bio.Restriction.Restriction.Mly113I.dna
Bio.Restriction.Restriction.MlyI.dna
Bio.Restriction.Restriction.MmeI.dna
Bio.Restriction.Restriction.MnlI.dna
Bio.Restriction.Restriction.Mph1103I.dna
Bio.Restriction.Restriction.MreI.dna
Bio.Restriction.Restriction.MroI.dna
Bio.Restriction.Restriction.MroNI.dna
Bio.Restriction.Restriction.MroXI.dna
Bio.Restriction.Restriction.MscI.dna
Bio.Restriction.Restriction.MseI.dna
Bio.Restriction.Restriction.MslI.dna
Bio.Restriction.Restriction.Msp20I.dna
Bio.Restriction.Restriction.MspA1I.dna
Bio.Restriction.Restriction.MspCI.dna
Bio.Restriction.Restriction.MspI.dna
Bio.Restriction.Restriction.MspJI.dna
Bio.Restriction.Restriction.MspR9I.dna
Bio.Restriction.Restriction.MssI.dna
Bio.Restriction.Restriction.MstI.dna
Bio.Restriction.Restriction.MunI.dna
Bio.Restriction.Restriction.Mva1269I.dna
Bio.Restriction.Restriction.MvaI.dna
Bio.Restriction.Restriction.MvnI.dna
Bio.Restriction.Restriction.MvrI.dna
Bio.Restriction.Restriction.MwoI.dna
Bio.Restriction.Restriction.NaeI.dna
Bio.Restriction.Restriction.NarI.dna
Bio.Restriction.Restriction.NciI.dna
Bio.Restriction.Restriction.NcoI.dna
Bio.Restriction.Restriction.NdeI.dna
Bio.Restriction.Restriction.NdeII.dna
Bio.Restriction.Restriction.NgoAVIII.dna
Bio.Restriction.Restriction.NgoMIV.dna
Bio.Restriction.Restriction.NhaXI.dna
Bio.Restriction.Restriction.NheI.dna
Bio.Restriction.Restriction.NlaCI.dna
Bio.Restriction.Restriction.NlaIII.dna
Bio.Restriction.Restriction.NlaIV.dna
Bio.Restriction.Restriction.Nli3877I.dna
Bio.Restriction.Restriction.NmeAIII.dna
Bio.Restriction.Restriction.NmeDI.dna
Bio.Restriction.Restriction.NmuCI.dna
Bio.Restriction.Restriction.NotI.dna
Bio.Restriction.Restriction.NruI.dna
Bio.Restriction.Restriction.NsbI.dna
Bio.Restriction.Restriction.NsiI.dna
Bio.Restriction.Restriction.NspBII.dna
Bio.Restriction.Restriction.NspI.dna
Bio.Restriction.Restriction.NspV.dna
Bio.Restriction.Restriction.OliI.dna
Bio.Restriction.Restriction.PabI.dna
Bio.Restriction.Restriction.PacI.dna
Bio.Restriction.Restriction.PaeI.dna
Bio.Restriction.Restriction.PaeR7I.dna
Bio.Restriction.Restriction.PagI.dna
Bio.Restriction.Restriction.PalAI.dna
Bio.Restriction.Restriction.PasI.dna
Bio.Restriction.Restriction.PauI.dna
Bio.Restriction.Restriction.PceI.dna
Bio.Restriction.Restriction.PciI.dna
Bio.Restriction.Restriction.PciSI.dna
Bio.Restriction.Restriction.PcsI.dna
Bio.Restriction.Restriction.PctI.dna
Bio.Restriction.Restriction.PdiI.dna
Bio.Restriction.Restriction.PdmI.dna
Bio.Restriction.Restriction.PenI.dna
Bio.Restriction.Restriction.PfeI.dna
Bio.Restriction.Restriction.Pfl1108I.dna
Bio.Restriction.Restriction.Pfl23II.dna
Bio.Restriction.Restriction.PflFI.dna
Bio.Restriction.Restriction.PflMI.dna
Bio.Restriction.Restriction.PfoI.dna
Bio.Restriction.Restriction.PinAI.dna
Bio.Restriction.Restriction.PlaDI.dna
Bio.Restriction.Restriction.Ple19I.dna
Bio.Restriction.Restriction.PleI.dna
Bio.Restriction.Restriction.PluTI.dna
Bio.Restriction.Restriction.PmaCI.dna
Bio.Restriction.Restriction.PmeI.dna
Bio.Restriction.Restriction.PmlI.dna
Bio.Restriction.Restriction.PpiI.dna
Bio.Restriction.Restriction.PpsI.dna
Bio.Restriction.Restriction.Ppu10I.dna
Bio.Restriction.Restriction.Ppu21I.dna
Bio.Restriction.Restriction.PpuMI.dna
Bio.Restriction.Restriction.PscI.dna
Bio.Restriction.Restriction.PshAI.dna
Bio.Restriction.Restriction.PshBI.dna
Bio.Restriction.Restriction.PsiI.dna
Bio.Restriction.Restriction.Psp03I.dna
Bio.Restriction.Restriction.Psp124BI.dna
Bio.Restriction.Restriction.Psp1406I.dna
Bio.Restriction.Restriction.Psp5II.dna
Bio.Restriction.Restriction.Psp6I.dna
Bio.Restriction.Restriction.PspCI.dna
Bio.Restriction.Restriction.PspEI.dna
Bio.Restriction.Restriction.PspGI.dna
Bio.Restriction.Restriction.PspLI.dna
Bio.Restriction.Restriction.PspN4I.dna
Bio.Restriction.Restriction.PspOMI.dna
Bio.Restriction.Restriction.PspOMII.dna
Bio.Restriction.Restriction.PspPI.dna
Bio.Restriction.Restriction.PspPPI.dna
Bio.Restriction.Restriction.PspPRI.dna
Bio.Restriction.Restriction.PspXI.dna
Bio.Restriction.Restriction.PsrI.dna
Bio.Restriction.Restriction.PssI.dna
Bio.Restriction.Restriction.PstI.dna
Bio.Restriction.Restriction.PstNI.dna
Bio.Restriction.Restriction.PsuI.dna
Bio.Restriction.Restriction.PsyI.dna
Bio.Restriction.Restriction.PteI.dna
Bio.Restriction.Restriction.PvuI.dna
Bio.Restriction.Restriction.PvuII.dna
Bio.Restriction.Restriction.R2_BceSIV.dna
Bio.Restriction.Restriction.RceI.dna
Bio.Restriction.Restriction.RdeGBI.dna
Bio.Restriction.Restriction.RdeGBII.dna
Bio.Restriction.Restriction.RdeGBIII.dna
Bio.Restriction.Restriction.RflFIII.dna
Bio.Restriction.Restriction.RgaI.dna
Bio.Restriction.Restriction.RigI.dna
Bio.Restriction.Restriction.RlaI.dna
Bio.Restriction.Restriction.RleAI.dna
Bio.Restriction.Restriction.RpaB5I.dna
Bio.Restriction.Restriction.RpaBI.dna
Bio.Restriction.Restriction.RpaI.dna
Bio.Restriction.Restriction.RpaTI.dna
Bio.Restriction.Restriction.RruI.dna
Bio.Restriction.Restriction.RsaI.dna
Bio.Restriction.Restriction.RsaNI.dna
Bio.Restriction.Restriction.RseI.dna
Bio.Restriction.Restriction.Rsr2I.dna
Bio.Restriction.Restriction.RsrII.dna
Bio.Restriction.Restriction.SacI.dna
Bio.Restriction.Restriction.SacII.dna
Bio.Restriction.Restriction.SalI.dna
Bio.Restriction.Restriction.SanDI.dna
Bio.Restriction.Restriction.SapI.dna
Bio.Restriction.Restriction.SaqAI.dna
Bio.Restriction.Restriction.SatI.dna
Bio.Restriction.Restriction.Sau3AI.dna
Bio.Restriction.Restriction.Sau96I.dna
Bio.Restriction.Restriction.SauI.dna
Bio.Restriction.Restriction.SbfI.dna
Bio.Restriction.Restriction.ScaI.dna
Bio.Restriction.Restriction.SchI.dna
Bio.Restriction.Restriction.SciI.dna
Bio.Restriction.Restriction.ScrFI.dna
Bio.Restriction.Restriction.SdaI.dna
Bio.Restriction.Restriction.SdeAI.dna
Bio.Restriction.Restriction.SdeOSI.dna
Bio.Restriction.Restriction.SduI.dna
Bio.Restriction.Restriction.SecI.dna
Bio.Restriction.Restriction.SelI.dna
Bio.Restriction.Restriction.SetI.dna
Bio.Restriction.Restriction.SexAI.dna
Bio.Restriction.Restriction.SfaAI.dna
Bio.Restriction.Restriction.SfaNI.dna
Bio.Restriction.Restriction.SfcI.dna
Bio.Restriction.Restriction.SfeI.dna
Bio.Restriction.Restriction.SfiI.dna
Bio.Restriction.Restriction.SfoI.dna
Bio.Restriction.Restriction.Sfr274I.dna
Bio.Restriction.Restriction.Sfr303I.dna
Bio.Restriction.Restriction.SfuI.dna
Bio.Restriction.Restriction.SgeI.dna
Bio.Restriction.Restriction.SgfI.dna
Bio.Restriction.Restriction.SgrAI.dna
Bio.Restriction.Restriction.SgrBI.dna
Bio.Restriction.Restriction.SgrDI.dna
Bio.Restriction.Restriction.SgrTI.dna
Bio.Restriction.Restriction.SgsI.dna
Bio.Restriction.Restriction.SimI.dna
Bio.Restriction.Restriction.SlaI.dna
Bio.Restriction.Restriction.SmaI.dna
Bio.Restriction.Restriction.SmiI.dna
Bio.Restriction.Restriction.SmiMI.dna
Bio.Restriction.Restriction.SmlI.dna
Bio.Restriction.Restriction.SmoI.dna
Bio.Restriction.Restriction.SnaBI.dna
Bio.Restriction.Restriction.SnaI.dna
Bio.Restriction.Restriction.Sno506I.dna
Bio.Restriction.Restriction.SpeI.dna
Bio.Restriction.Restriction.SphI.dna
Bio.Restriction.Restriction.SplI.dna
Bio.Restriction.Restriction.SpoDI.dna
Bio.Restriction.Restriction.SrfI.dna
Bio.Restriction.Restriction.Sse232I.dna
Bio.Restriction.Restriction.Sse8387I.dna
Bio.Restriction.Restriction.Sse8647I.dna
Bio.Restriction.Restriction.Sse9I.dna
Bio.Restriction.Restriction.SseBI.dna
Bio.Restriction.Restriction.SsiI.dna
Bio.Restriction.Restriction.SspD5I.dna
Bio.Restriction.Restriction.SspDI.dna
Bio.Restriction.Restriction.SspI.dna
Bio.Restriction.Restriction.SstE37I.dna
Bio.Restriction.Restriction.SstI.dna
Bio.Restriction.Restriction.Sth132I.dna
Bio.Restriction.Restriction.Sth302II.dna
Bio.Restriction.Restriction.StrI.dna
Bio.Restriction.Restriction.StsI.dna
Bio.Restriction.Restriction.StuI.dna
Bio.Restriction.Restriction.StyD4I.dna
Bio.Restriction.Restriction.StyI.dna
Bio.Restriction.Restriction.SwaI.dna
Bio.Restriction.Restriction.TaaI.dna
Bio.Restriction.Restriction.TaiI.dna
Bio.Restriction.Restriction.TaqI.dna
Bio.Restriction.Restriction.TaqII.dna
Bio.Restriction.Restriction.TasI.dna
Bio.Restriction.Restriction.TatI.dna
Bio.Restriction.Restriction.TauI.dna
Bio.Restriction.Restriction.TfiI.dna
Bio.Restriction.Restriction.Tru1I.dna
Bio.Restriction.Restriction.Tru9I.dna
Bio.Restriction.Restriction.TscAI.dna
Bio.Restriction.Restriction.TseFI.dna
Bio.Restriction.Restriction.TseI.dna
Bio.Restriction.Restriction.TsoI.dna
Bio.Restriction.Restriction.Tsp45I.dna
Bio.Restriction.Restriction.Tsp4CI.dna
Bio.Restriction.Restriction.TspDTI.dna
Bio.Restriction.Restriction.TspEI.dna
Bio.Restriction.Restriction.TspGWI.dna
Bio.Restriction.Restriction.TspMI.dna
Bio.Restriction.Restriction.TspRI.dna
Bio.Restriction.Restriction.TssI.dna
Bio.Restriction.Restriction.TstI.dna
Bio.Restriction.Restriction.TsuI.dna
Bio.Restriction.Restriction.Tth111I.dna
Bio.Restriction.Restriction.Tth111II.dna
Bio.Restriction.Restriction.UbaF11I.dna
Bio.Restriction.Restriction.UbaF12I.dna
Bio.Restriction.Restriction.UbaF13I.dna
Bio.Restriction.Restriction.UbaF14I.dna
Bio.Restriction.Restriction.UbaF9I.dna
Bio.Restriction.Restriction.UbaPI.dna
Bio.Restriction.Restriction.UcoMSI.dna
Bio.Restriction.Restriction.UnbI.dna
Bio.Restriction.Restriction.Van91I.dna
Bio.Restriction.Restriction.Vha464I.dna
Bio.Restriction.Restriction.VneI.dna
Bio.Restriction.Restriction.VpaK11AI.dna
Bio.Restriction.Restriction.VpaK11BI.dna
Bio.Restriction.Restriction.VspI.dna
Bio.Restriction.Restriction.WviI.dna
Bio.Restriction.Restriction.XagI.dna
Bio.Restriction.Restriction.XapI.dna
Bio.Restriction.Restriction.XbaI.dna
Bio.Restriction.Restriction.XceI.dna
Bio.Restriction.Restriction.XcmI.dna
Bio.Restriction.Restriction.XhoI.dna
Bio.Restriction.Restriction.XhoII.dna
Bio.Restriction.Restriction.XmaI.dna
Bio.Restriction.Restriction.XmaIII.dna
Bio.Restriction.Restriction.XmaJI.dna
Bio.Restriction.Restriction.XmiI.dna
Bio.Restriction.Restriction.XmnI.dna
Bio.Restriction.Restriction.XspI.dna
Bio.Restriction.Restriction.YkrI.dna
Bio.Restriction.Restriction.ZraI.dna
Bio.Restriction.Restriction.ZrmI.dna
Bio.Restriction.Restriction.Zsp2I.dna" class="py-name" href="#" onclick="return doclink('link-91', 'dna', 'link-9');">dna</a></tt><tt class="py-op">.</tt><tt id="link-92" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-92', 'letters', 'link-47');">letters</a></tt><tt class="py-op">:</tt> </tt>
<a name="L449"></a><tt class="py-lineno">449</tt>  <tt class="py-line">            <tt class="py-name">base_counts</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L450"></a><tt class="py-lineno">450</tt>  <tt class="py-line"> </tt>
<a name="L451"></a><tt class="py-lineno">451</tt>  <tt class="py-line">            <tt class="py-name">cur</tt><tt class="py-op">.</tt><tt id="link-93" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-93', 'execute', 'link-50');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"""select val from MATRIX_DATA where ID = %s</tt> </tt>
<a name="L452"></a><tt class="py-lineno">452</tt>  <tt class="py-line"><tt class="py-string">                           and row = %s order by col"""</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">int_id</tt><tt class="py-op">,</tt> <tt class="py-name">base</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L453"></a><tt class="py-lineno">453</tt>  <tt class="py-line"> </tt>
<a name="L454"></a><tt class="py-lineno">454</tt>  <tt class="py-line">            <tt class="py-name">rows</tt> <tt class="py-op">=</tt> <tt class="py-name">cur</tt><tt class="py-op">.</tt><tt id="link-94" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchall" class="py-name" href="#" onclick="return doclink('link-94', 'fetchall', 'link-80');">fetchall</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L455"></a><tt class="py-lineno">455</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">row</tt> <tt class="py-keyword">in</tt> <tt class="py-name">rows</tt><tt class="py-op">:</tt> </tt>
<a name="L456"></a><tt class="py-lineno">456</tt>  <tt class="py-line">                <tt class="py-name">base_counts</tt><tt class="py-op">.</tt><tt id="link-95" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-95', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L457"></a><tt class="py-lineno">457</tt>  <tt class="py-line"> </tt>
<a name="L458"></a><tt class="py-lineno">458</tt>  <tt class="py-line">            <tt class="py-name">counts</tt><tt class="py-op">[</tt><tt class="py-name">base</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">float</tt><tt class="py-op">(</tt><tt id="link-96" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.x=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#x"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-96', 'x', 'link-96');">x</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-97" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-97', 'x', 'link-96');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">base_counts</tt><tt class="py-op">]</tt> </tt>
<a name="L459"></a><tt class="py-lineno">459</tt>  <tt class="py-line"> </tt>
<a name="L460"></a><tt class="py-lineno">460</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-98" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.matrix
Bio.motifs.matrix" class="py-name" href="#" onclick="return doclink('link-98', 'matrix', 'link-13');">matrix</a></tt><tt class="py-op">.</tt><tt id="link-99" class="py-name" targets="Class Bio.motifs.matrix.GenericPositionMatrix=Bio.motifs.matrix.GenericPositionMatrix-class.html"><a title="Bio.motifs.matrix.GenericPositionMatrix" class="py-name" href="#" onclick="return doclink('link-99', 'GenericPositionMatrix', 'link-99');">GenericPositionMatrix</a></tt><tt class="py-op">(</tt><tt id="link-100" class="py-name"><a title="Bio.Restriction.Restriction.AanI.dna
Bio.Restriction.Restriction.AarI.dna
Bio.Restriction.Restriction.AasI.dna
Bio.Restriction.Restriction.AatII.dna
Bio.Restriction.Restriction.AbaSI.dna
Bio.Restriction.Restriction.AbsI.dna
Bio.Restriction.Restriction.Acc16I.dna
Bio.Restriction.Restriction.Acc36I.dna
Bio.Restriction.Restriction.Acc65I.dna
Bio.Restriction.Restriction.AccB1I.dna
Bio.Restriction.Restriction.AccB7I.dna
Bio.Restriction.Restriction.AccBSI.dna
Bio.Restriction.Restriction.AccI.dna
Bio.Restriction.Restriction.AccII.dna
Bio.Restriction.Restriction.AccIII.dna
Bio.Restriction.Restriction.AceIII.dna
Bio.Restriction.Restriction.AciI.dna
Bio.Restriction.Restriction.AclI.dna
Bio.Restriction.Restriction.AclWI.dna
Bio.Restriction.Restriction.AcoI.dna
Bio.Restriction.Restriction.AcsI.dna
Bio.Restriction.Restriction.AcuI.dna
Bio.Restriction.Restriction.AcvI.dna
Bio.Restriction.Restriction.AcyI.dna
Bio.Restriction.Restriction.AdeI.dna
Bio.Restriction.Restriction.AfaI.dna
Bio.Restriction.Restriction.AfeI.dna
Bio.Restriction.Restriction.AfiI.dna
Bio.Restriction.Restriction.AflII.dna
Bio.Restriction.Restriction.AflIII.dna
Bio.Restriction.Restriction.AgeI.dna
Bio.Restriction.Restriction.AgsI.dna
Bio.Restriction.Restriction.AhaIII.dna
Bio.Restriction.Restriction.AhdI.dna
Bio.Restriction.Restriction.AhlI.dna
Bio.Restriction.Restriction.AjiI.dna
Bio.Restriction.Restriction.AjnI.dna
Bio.Restriction.Restriction.AjuI.dna
Bio.Restriction.Restriction.AleI.dna
Bio.Restriction.Restriction.AlfI.dna
Bio.Restriction.Restriction.AloI.dna
Bio.Restriction.Restriction.AluBI.dna
Bio.Restriction.Restriction.AluI.dna
Bio.Restriction.Restriction.Alw21I.dna
Bio.Restriction.Restriction.Alw26I.dna
Bio.Restriction.Restriction.Alw44I.dna
Bio.Restriction.Restriction.AlwFI.dna
Bio.Restriction.Restriction.AlwI.dna
Bio.Restriction.Restriction.AlwNI.dna
Bio.Restriction.Restriction.Ama87I.dna
Bio.Restriction.Restriction.Aor13HI.dna
Bio.Restriction.Restriction.Aor51HI.dna
Bio.Restriction.Restriction.AoxI.dna
Bio.Restriction.Restriction.ApaBI.dna
Bio.Restriction.Restriction.ApaI.dna
Bio.Restriction.Restriction.ApaLI.dna
Bio.Restriction.Restriction.ApeKI.dna
Bio.Restriction.Restriction.ApoI.dna
Bio.Restriction.Restriction.ApyPI.dna
Bio.Restriction.Restriction.AquII.dna
Bio.Restriction.Restriction.AquIII.dna
Bio.Restriction.Restriction.AquIV.dna
Bio.Restriction.Restriction.ArsI.dna
Bio.Restriction.Restriction.AscI.dna
Bio.Restriction.Restriction.AseI.dna
Bio.Restriction.Restriction.Asi256I.dna
Bio.Restriction.Restriction.AsiGI.dna
Bio.Restriction.Restriction.AsiSI.dna
Bio.Restriction.Restriction.Asp700I.dna
Bio.Restriction.Restriction.Asp718I.dna
Bio.Restriction.Restriction.AspA2I.dna
Bio.Restriction.Restriction.AspBHI.dna
Bio.Restriction.Restriction.AspLEI.dna
Bio.Restriction.Restriction.AspS9I.dna
Bio.Restriction.Restriction.AssI.dna
Bio.Restriction.Restriction.AsuC2I.dna
Bio.Restriction.Restriction.AsuHPI.dna
Bio.Restriction.Restriction.AsuI.dna
Bio.Restriction.Restriction.AsuII.dna
Bio.Restriction.Restriction.AsuNHI.dna
Bio.Restriction.Restriction.AvaI.dna
Bio.Restriction.Restriction.AvaII.dna
Bio.Restriction.Restriction.AvaIII.dna
Bio.Restriction.Restriction.AvrII.dna
Bio.Restriction.Restriction.AxyI.dna
Bio.Restriction.Restriction.BaeGI.dna
Bio.Restriction.Restriction.BaeI.dna
Bio.Restriction.Restriction.BalI.dna
Bio.Restriction.Restriction.BamHI.dna
Bio.Restriction.Restriction.BanI.dna
Bio.Restriction.Restriction.BanII.dna
Bio.Restriction.Restriction.BarI.dna
Bio.Restriction.Restriction.BasI.dna
Bio.Restriction.Restriction.BauI.dna
Bio.Restriction.Restriction.Bbr7I.dna
Bio.Restriction.Restriction.BbrPI.dna
Bio.Restriction.Restriction.BbsI.dna
Bio.Restriction.Restriction.Bbv12I.dna
Bio.Restriction.Restriction.BbvCI.dna
Bio.Restriction.Restriction.BbvI.dna
Bio.Restriction.Restriction.BbvII.dna
Bio.Restriction.Restriction.BccI.dna
Bio.Restriction.Restriction.Bce83I.dna
Bio.Restriction.Restriction.BceAI.dna
Bio.Restriction.Restriction.BcefI.dna
Bio.Restriction.Restriction.BcgI.dna
Bio.Restriction.Restriction.BciT130I.dna
Bio.Restriction.Restriction.BciVI.dna
Bio.Restriction.Restriction.BclI.dna
Bio.Restriction.Restriction.BcnI.dna
Bio.Restriction.Restriction.BcoDI.dna
Bio.Restriction.Restriction.BcuI.dna
Bio.Restriction.Restriction.BdaI.dna
Bio.Restriction.Restriction.BetI.dna
Bio.Restriction.Restriction.BfaI.dna
Bio.Restriction.Restriction.BfiI.dna
Bio.Restriction.Restriction.BfmI.dna
Bio.Restriction.Restriction.BfoI.dna
Bio.Restriction.Restriction.BfrI.dna
Bio.Restriction.Restriction.BfuAI.dna
Bio.Restriction.Restriction.BfuCI.dna
Bio.Restriction.Restriction.BfuI.dna
Bio.Restriction.Restriction.BglI.dna
Bio.Restriction.Restriction.BglII.dna
Bio.Restriction.Restriction.BinI.dna
Bio.Restriction.Restriction.BisI.dna
Bio.Restriction.Restriction.BlnI.dna
Bio.Restriction.Restriction.BlpI.dna
Bio.Restriction.Restriction.BlsI.dna
Bio.Restriction.Restriction.BmcAI.dna
Bio.Restriction.Restriction.Bme1390I.dna
Bio.Restriction.Restriction.Bme18I.dna
Bio.Restriction.Restriction.BmeDI.dna
Bio.Restriction.Restriction.BmeRI.dna
Bio.Restriction.Restriction.BmeT110I.dna
Bio.Restriction.Restriction.BmgBI.dna
Bio.Restriction.Restriction.BmgI.dna
Bio.Restriction.Restriction.BmgT120I.dna
Bio.Restriction.Restriction.BmiI.dna
Bio.Restriction.Restriction.BmrFI.dna
Bio.Restriction.Restriction.BmrI.dna
Bio.Restriction.Restriction.BmsI.dna
Bio.Restriction.Restriction.BmtI.dna
Bio.Restriction.Restriction.BmuI.dna
Bio.Restriction.Restriction.BoxI.dna
Bio.Restriction.Restriction.BpiI.dna
Bio.Restriction.Restriction.BplI.dna
Bio.Restriction.Restriction.BpmI.dna
Bio.Restriction.Restriction.Bpu10I.dna
Bio.Restriction.Restriction.Bpu1102I.dna
Bio.Restriction.Restriction.Bpu14I.dna
Bio.Restriction.Restriction.BpuEI.dna
Bio.Restriction.Restriction.BpuMI.dna
Bio.Restriction.Restriction.BpvUI.dna
Bio.Restriction.Restriction.Bsa29I.dna
Bio.Restriction.Restriction.BsaAI.dna
Bio.Restriction.Restriction.BsaBI.dna
Bio.Restriction.Restriction.BsaHI.dna
Bio.Restriction.Restriction.BsaI.dna
Bio.Restriction.Restriction.BsaJI.dna
Bio.Restriction.Restriction.BsaWI.dna
Bio.Restriction.Restriction.BsaXI.dna
Bio.Restriction.Restriction.BsbI.dna
Bio.Restriction.Restriction.Bsc4I.dna
Bio.Restriction.Restriction.BscAI.dna
Bio.Restriction.Restriction.BscGI.dna
Bio.Restriction.Restriction.Bse118I.dna
Bio.Restriction.Restriction.Bse1I.dna
Bio.Restriction.Restriction.Bse21I.dna
Bio.Restriction.Restriction.Bse3DI.dna
Bio.Restriction.Restriction.Bse8I.dna
Bio.Restriction.Restriction.BseAI.dna
Bio.Restriction.Restriction.BseBI.dna
Bio.Restriction.Restriction.BseCI.dna
Bio.Restriction.Restriction.BseDI.dna
Bio.Restriction.Restriction.BseGI.dna
Bio.Restriction.Restriction.BseJI.dna
Bio.Restriction.Restriction.BseLI.dna
Bio.Restriction.Restriction.BseMI.dna
Bio.Restriction.Restriction.BseMII.dna
Bio.Restriction.Restriction.BseNI.dna
Bio.Restriction.Restriction.BsePI.dna
Bio.Restriction.Restriction.BseRI.dna
Bio.Restriction.Restriction.BseSI.dna
Bio.Restriction.Restriction.BseX3I.dna
Bio.Restriction.Restriction.BseXI.dna
Bio.Restriction.Restriction.BseYI.dna
Bio.Restriction.Restriction.BsgI.dna
Bio.Restriction.Restriction.Bsh1236I.dna
Bio.Restriction.Restriction.Bsh1285I.dna
Bio.Restriction.Restriction.BshFI.dna
Bio.Restriction.Restriction.BshNI.dna
Bio.Restriction.Restriction.BshTI.dna
Bio.Restriction.Restriction.BshVI.dna
Bio.Restriction.Restriction.BsiEI.dna
Bio.Restriction.Restriction.BsiHKAI.dna
Bio.Restriction.Restriction.BsiHKCI.dna
Bio.Restriction.Restriction.BsiI.dna
Bio.Restriction.Restriction.BsiSI.dna
Bio.Restriction.Restriction.BsiWI.dna
Bio.Restriction.Restriction.BsiYI.dna
Bio.Restriction.Restriction.BslFI.dna
Bio.Restriction.Restriction.BslI.dna
Bio.Restriction.Restriction.BsmAI.dna
Bio.Restriction.Restriction.BsmBI.dna
Bio.Restriction.Restriction.BsmFI.dna
Bio.Restriction.Restriction.BsmI.dna
Bio.Restriction.Restriction.BsnI.dna
Bio.Restriction.Restriction.Bso31I.dna
Bio.Restriction.Restriction.BsoBI.dna
Bio.Restriction.Restriction.Bsp119I.dna
Bio.Restriction.Restriction.Bsp120I.dna
Bio.Restriction.Restriction.Bsp1286I.dna
Bio.Restriction.Restriction.Bsp13I.dna
Bio.Restriction.Restriction.Bsp1407I.dna
Bio.Restriction.Restriction.Bsp143I.dna
Bio.Restriction.Restriction.Bsp1720I.dna
Bio.Restriction.Restriction.Bsp19I.dna
Bio.Restriction.Restriction.Bsp24I.dna
Bio.Restriction.Restriction.Bsp68I.dna
Bio.Restriction.Restriction.BspACI.dna
Bio.Restriction.Restriction.BspCNI.dna
Bio.Restriction.Restriction.BspD6I.dna
Bio.Restriction.Restriction.BspDI.dna
Bio.Restriction.Restriction.BspEI.dna
Bio.Restriction.Restriction.BspFNI.dna
Bio.Restriction.Restriction.BspGI.dna
Bio.Restriction.Restriction.BspHI.dna
Bio.Restriction.Restriction.BspLI.dna
Bio.Restriction.Restriction.BspLU11I.dna
Bio.Restriction.Restriction.BspMI.dna
Bio.Restriction.Restriction.BspMII.dna
Bio.Restriction.Restriction.BspNCI.dna
Bio.Restriction.Restriction.BspOI.dna
Bio.Restriction.Restriction.BspPI.dna
Bio.Restriction.Restriction.BspQI.dna
Bio.Restriction.Restriction.BspT104I.dna
Bio.Restriction.Restriction.BspT107I.dna
Bio.Restriction.Restriction.BspTI.dna
Bio.Restriction.Restriction.BsrBI.dna
Bio.Restriction.Restriction.BsrDI.dna
Bio.Restriction.Restriction.BsrFI.dna
Bio.Restriction.Restriction.BsrGI.dna
Bio.Restriction.Restriction.BsrI.dna
Bio.Restriction.Restriction.BsrSI.dna
Bio.Restriction.Restriction.BssAI.dna
Bio.Restriction.Restriction.BssECI.dna
Bio.Restriction.Restriction.BssHII.dna
Bio.Restriction.Restriction.BssKI.dna
Bio.Restriction.Restriction.BssMI.dna
Bio.Restriction.Restriction.BssNAI.dna
Bio.Restriction.Restriction.BssNI.dna
Bio.Restriction.Restriction.BssSI.dna
Bio.Restriction.Restriction.BssT1I.dna
Bio.Restriction.Restriction.Bst1107I.dna
Bio.Restriction.Restriction.Bst2BI.dna
Bio.Restriction.Restriction.Bst2UI.dna
Bio.Restriction.Restriction.Bst4CI.dna
Bio.Restriction.Restriction.Bst6I.dna
Bio.Restriction.Restriction.BstACI.dna
Bio.Restriction.Restriction.BstAFI.dna
Bio.Restriction.Restriction.BstAPI.dna
Bio.Restriction.Restriction.BstAUI.dna
Bio.Restriction.Restriction.BstBAI.dna
Bio.Restriction.Restriction.BstBI.dna
Bio.Restriction.Restriction.BstC8I.dna
Bio.Restriction.Restriction.BstDEI.dna
Bio.Restriction.Restriction.BstDSI.dna
Bio.Restriction.Restriction.BstEII.dna
Bio.Restriction.Restriction.BstENI.dna
Bio.Restriction.Restriction.BstF5I.dna
Bio.Restriction.Restriction.BstFNI.dna
Bio.Restriction.Restriction.BstH2I.dna
Bio.Restriction.Restriction.BstHHI.dna
Bio.Restriction.Restriction.BstKTI.dna
Bio.Restriction.Restriction.BstMAI.dna
Bio.Restriction.Restriction.BstMBI.dna
Bio.Restriction.Restriction.BstMCI.dna
Bio.Restriction.Restriction.BstMWI.dna
Bio.Restriction.Restriction.BstNI.dna
Bio.Restriction.Restriction.BstNSI.dna
Bio.Restriction.Restriction.BstOI.dna
Bio.Restriction.Restriction.BstPAI.dna
Bio.Restriction.Restriction.BstPI.dna
Bio.Restriction.Restriction.BstSCI.dna
Bio.Restriction.Restriction.BstSFI.dna
Bio.Restriction.Restriction.BstSLI.dna
Bio.Restriction.Restriction.BstSNI.dna
Bio.Restriction.Restriction.BstUI.dna
Bio.Restriction.Restriction.BstV1I.dna
Bio.Restriction.Restriction.BstV2I.dna
Bio.Restriction.Restriction.BstX2I.dna
Bio.Restriction.Restriction.BstXI.dna
Bio.Restriction.Restriction.BstYI.dna
Bio.Restriction.Restriction.BstZ17I.dna
Bio.Restriction.Restriction.BstZI.dna
Bio.Restriction.Restriction.Bsu15I.dna
Bio.Restriction.Restriction.Bsu36I.dna
Bio.Restriction.Restriction.BsuI.dna
Bio.Restriction.Restriction.BsuRI.dna
Bio.Restriction.Restriction.BtgI.dna
Bio.Restriction.Restriction.BtgZI.dna
Bio.Restriction.Restriction.BthCI.dna
Bio.Restriction.Restriction.BtrI.dna
Bio.Restriction.Restriction.BtsCI.dna
Bio.Restriction.Restriction.BtsI.dna
Bio.Restriction.Restriction.BtsIMutI.dna
Bio.Restriction.Restriction.BtuMI.dna
Bio.Restriction.Restriction.BveI.dna
Bio.Restriction.Restriction.Cac8I.dna
Bio.Restriction.Restriction.CaiI.dna
Bio.Restriction.Restriction.CauII.dna
Bio.Restriction.Restriction.CchII.dna
Bio.Restriction.Restriction.CchIII.dna
Bio.Restriction.Restriction.CciI.dna
Bio.Restriction.Restriction.CciNI.dna
Bio.Restriction.Restriction.Cdi630V.dna
Bio.Restriction.Restriction.CdiI.dna
Bio.Restriction.Restriction.CdpI.dna
Bio.Restriction.Restriction.CfoI.dna
Bio.Restriction.Restriction.Cfr10I.dna
Bio.Restriction.Restriction.Cfr13I.dna
Bio.Restriction.Restriction.Cfr42I.dna
Bio.Restriction.Restriction.Cfr9I.dna
Bio.Restriction.Restriction.CfrI.dna
Bio.Restriction.Restriction.Cgl13032I.dna
Bio.Restriction.Restriction.Cgl13032II.dna
Bio.Restriction.Restriction.ChaI.dna
Bio.Restriction.Restriction.CjeFIII.dna
Bio.Restriction.Restriction.CjeFV.dna
Bio.Restriction.Restriction.CjeI.dna
Bio.Restriction.Restriction.CjeNII.dna
Bio.Restriction.Restriction.CjeNIII.dna
Bio.Restriction.Restriction.CjeP659IV.dna
Bio.Restriction.Restriction.CjePI.dna
Bio.Restriction.Restriction.CjuI.dna
Bio.Restriction.Restriction.CjuII.dna
Bio.Restriction.Restriction.ClaI.dna
Bio.Restriction.Restriction.CpoI.dna
Bio.Restriction.Restriction.CseI.dna
Bio.Restriction.Restriction.CsiI.dna
Bio.Restriction.Restriction.Csp6I.dna
Bio.Restriction.Restriction.CspAI.dna
Bio.Restriction.Restriction.CspCI.dna
Bio.Restriction.Restriction.CspI.dna
Bio.Restriction.Restriction.CstMI.dna
Bio.Restriction.Restriction.CviAII.dna
Bio.Restriction.Restriction.CviJI.dna
Bio.Restriction.Restriction.CviKI_1.dna
Bio.Restriction.Restriction.CviQI.dna
Bio.Restriction.Restriction.CviRI.dna
Bio.Restriction.Restriction.DdeI.dna
Bio.Restriction.Restriction.DinI.dna
Bio.Restriction.Restriction.DpnI.dna
Bio.Restriction.Restriction.DpnII.dna
Bio.Restriction.Restriction.DraI.dna
Bio.Restriction.Restriction.DraII.dna
Bio.Restriction.Restriction.DraIII.dna
Bio.Restriction.Restriction.DraRI.dna
Bio.Restriction.Restriction.DrdI.dna
Bio.Restriction.Restriction.DrdII.dna
Bio.Restriction.Restriction.DriI.dna
Bio.Restriction.Restriction.DsaI.dna
Bio.Restriction.Restriction.DseDI.dna
Bio.Restriction.Restriction.EaeI.dna
Bio.Restriction.Restriction.EagI.dna
Bio.Restriction.Restriction.Eam1104I.dna
Bio.Restriction.Restriction.Eam1105I.dna
Bio.Restriction.Restriction.EarI.dna
Bio.Restriction.Restriction.EciI.dna
Bio.Restriction.Restriction.Ecl136II.dna
Bio.Restriction.Restriction.EclXI.dna
Bio.Restriction.Restriction.Eco105I.dna
Bio.Restriction.Restriction.Eco130I.dna
Bio.Restriction.Restriction.Eco147I.dna
Bio.Restriction.Restriction.Eco24I.dna
Bio.Restriction.Restriction.Eco31I.dna
Bio.Restriction.Restriction.Eco32I.dna
Bio.Restriction.Restriction.Eco47I.dna
Bio.Restriction.Restriction.Eco47III.dna
Bio.Restriction.Restriction.Eco52I.dna
Bio.Restriction.Restriction.Eco53kI.dna
Bio.Restriction.Restriction.Eco57I.dna
Bio.Restriction.Restriction.Eco57MI.dna
Bio.Restriction.Restriction.Eco72I.dna
Bio.Restriction.Restriction.Eco81I.dna
Bio.Restriction.Restriction.Eco88I.dna
Bio.Restriction.Restriction.Eco91I.dna
Bio.Restriction.Restriction.EcoHI.dna
Bio.Restriction.Restriction.EcoICRI.dna
Bio.Restriction.Restriction.EcoNI.dna
Bio.Restriction.Restriction.EcoO109I.dna
Bio.Restriction.Restriction.EcoO65I.dna
Bio.Restriction.Restriction.EcoRI.dna
Bio.Restriction.Restriction.EcoRII.dna
Bio.Restriction.Restriction.EcoRV.dna
Bio.Restriction.Restriction.EcoT14I.dna
Bio.Restriction.Restriction.EcoT22I.dna
Bio.Restriction.Restriction.EcoT38I.dna
Bio.Restriction.Restriction.EgeI.dna
Bio.Restriction.Restriction.EheI.dna
Bio.Restriction.Restriction.ErhI.dna
Bio.Restriction.Restriction.EsaBC3I.dna
Bio.Restriction.Restriction.EsaSSI.dna
Bio.Restriction.Restriction.Esp3I.dna
Bio.Restriction.Restriction.EspI.dna
Bio.Restriction.Restriction.FaeI.dna
Bio.Restriction.Restriction.FaiI.dna
Bio.Restriction.Restriction.FalI.dna
Bio.Restriction.Restriction.FaqI.dna
Bio.Restriction.Restriction.FatI.dna
Bio.Restriction.Restriction.FauI.dna
Bio.Restriction.Restriction.FauNDI.dna
Bio.Restriction.Restriction.FbaI.dna
Bio.Restriction.Restriction.FblI.dna
Bio.Restriction.Restriction.FinI.dna
Bio.Restriction.Restriction.FmuI.dna
Bio.Restriction.Restriction.Fnu4HI.dna
Bio.Restriction.Restriction.FnuDII.dna
Bio.Restriction.Restriction.FokI.dna
Bio.Restriction.Restriction.FriOI.dna
Bio.Restriction.Restriction.FseI.dna
Bio.Restriction.Restriction.Fsp4HI.dna
Bio.Restriction.Restriction.FspAI.dna
Bio.Restriction.Restriction.FspBI.dna
Bio.Restriction.Restriction.FspEI.dna
Bio.Restriction.Restriction.FspI.dna
Bio.Restriction.Restriction.GauT27I.dna
Bio.Restriction.Restriction.GdiII.dna
Bio.Restriction.Restriction.GlaI.dna
Bio.Restriction.Restriction.GluI.dna
Bio.Restriction.Restriction.GsaI.dna
Bio.Restriction.Restriction.GsuI.dna
Bio.Restriction.Restriction.HaeI.dna
Bio.Restriction.Restriction.HaeII.dna
Bio.Restriction.Restriction.HaeIII.dna
Bio.Restriction.Restriction.HapII.dna
Bio.Restriction.Restriction.HauII.dna
Bio.Restriction.Restriction.HgaI.dna
Bio.Restriction.Restriction.HgiAI.dna
Bio.Restriction.Restriction.HgiCI.dna
Bio.Restriction.Restriction.HgiEII.dna
Bio.Restriction.Restriction.HgiJII.dna
Bio.Restriction.Restriction.HhaI.dna
Bio.Restriction.Restriction.Hin1I.dna
Bio.Restriction.Restriction.Hin1II.dna
Bio.Restriction.Restriction.Hin4I.dna
Bio.Restriction.Restriction.Hin4II.dna
Bio.Restriction.Restriction.Hin6I.dna
Bio.Restriction.Restriction.HinP1I.dna
Bio.Restriction.Restriction.HincII.dna
Bio.Restriction.Restriction.HindII.dna
Bio.Restriction.Restriction.HindIII.dna
Bio.Restriction.Restriction.HinfI.dna
Bio.Restriction.Restriction.HpaI.dna
Bio.Restriction.Restriction.HpaII.dna
Bio.Restriction.Restriction.HphI.dna
Bio.Restriction.Restriction.Hpy166II.dna
Bio.Restriction.Restriction.Hpy178III.dna
Bio.Restriction.Restriction.Hpy188I.dna
Bio.Restriction.Restriction.Hpy188III.dna
Bio.Restriction.Restriction.Hpy8I.dna
Bio.Restriction.Restriction.Hpy99I.dna
Bio.Restriction.Restriction.Hpy99XIII.dna
Bio.Restriction.Restriction.Hpy99XIV.dna
Bio.Restriction.Restriction.HpyAV.dna
Bio.Restriction.Restriction.HpyCH4III.dna
Bio.Restriction.Restriction.HpyCH4IV.dna
Bio.Restriction.Restriction.HpyCH4V.dna
Bio.Restriction.Restriction.HpyF10VI.dna
Bio.Restriction.Restriction.HpyF3I.dna
Bio.Restriction.Restriction.HpySE526I.dna
Bio.Restriction.Restriction.Hsp92I.dna
Bio.Restriction.Restriction.Hsp92II.dna
Bio.Restriction.Restriction.HspAI.dna
Bio.Restriction.Restriction.Jma19592I.dna
Bio.Restriction.Restriction.KasI.dna
Bio.Restriction.Restriction.KflI.dna
Bio.Restriction.Restriction.Kpn2I.dna
Bio.Restriction.Restriction.KpnI.dna
Bio.Restriction.Restriction.KroI.dna
Bio.Restriction.Restriction.Ksp22I.dna
Bio.Restriction.Restriction.Ksp632I.dna
Bio.Restriction.Restriction.KspAI.dna
Bio.Restriction.Restriction.KspI.dna
Bio.Restriction.Restriction.Kzo9I.dna
Bio.Restriction.Restriction.LguI.dna
Bio.Restriction.Restriction.LpnI.dna
Bio.Restriction.Restriction.LpnPI.dna
Bio.Restriction.Restriction.Lsp1109I.dna
Bio.Restriction.Restriction.LweI.dna
Bio.Restriction.Restriction.MabI.dna
Bio.Restriction.Restriction.MaeI.dna
Bio.Restriction.Restriction.MaeII.dna
Bio.Restriction.Restriction.MaeIII.dna
Bio.Restriction.Restriction.MalI.dna
Bio.Restriction.Restriction.MaqI.dna
Bio.Restriction.Restriction.MauBI.dna
Bio.Restriction.Restriction.MbiI.dna
Bio.Restriction.Restriction.MboI.dna
Bio.Restriction.Restriction.MboII.dna
Bio.Restriction.Restriction.McaTI.dna
Bio.Restriction.Restriction.McrI.dna
Bio.Restriction.Restriction.MfeI.dna
Bio.Restriction.Restriction.MflI.dna
Bio.Restriction.Restriction.MhlI.dna
Bio.Restriction.Restriction.MjaIV.dna
Bio.Restriction.Restriction.MkaDII.dna
Bio.Restriction.Restriction.MlsI.dna
Bio.Restriction.Restriction.MluCI.dna
Bio.Restriction.Restriction.MluI.dna
Bio.Restriction.Restriction.MluNI.dna
Bio.Restriction.Restriction.Mly113I.dna
Bio.Restriction.Restriction.MlyI.dna
Bio.Restriction.Restriction.MmeI.dna
Bio.Restriction.Restriction.MnlI.dna
Bio.Restriction.Restriction.Mph1103I.dna
Bio.Restriction.Restriction.MreI.dna
Bio.Restriction.Restriction.MroI.dna
Bio.Restriction.Restriction.MroNI.dna
Bio.Restriction.Restriction.MroXI.dna
Bio.Restriction.Restriction.MscI.dna
Bio.Restriction.Restriction.MseI.dna
Bio.Restriction.Restriction.MslI.dna
Bio.Restriction.Restriction.Msp20I.dna
Bio.Restriction.Restriction.MspA1I.dna
Bio.Restriction.Restriction.MspCI.dna
Bio.Restriction.Restriction.MspI.dna
Bio.Restriction.Restriction.MspJI.dna
Bio.Restriction.Restriction.MspR9I.dna
Bio.Restriction.Restriction.MssI.dna
Bio.Restriction.Restriction.MstI.dna
Bio.Restriction.Restriction.MunI.dna
Bio.Restriction.Restriction.Mva1269I.dna
Bio.Restriction.Restriction.MvaI.dna
Bio.Restriction.Restriction.MvnI.dna
Bio.Restriction.Restriction.MvrI.dna
Bio.Restriction.Restriction.MwoI.dna
Bio.Restriction.Restriction.NaeI.dna
Bio.Restriction.Restriction.NarI.dna
Bio.Restriction.Restriction.NciI.dna
Bio.Restriction.Restriction.NcoI.dna
Bio.Restriction.Restriction.NdeI.dna
Bio.Restriction.Restriction.NdeII.dna
Bio.Restriction.Restriction.NgoAVIII.dna
Bio.Restriction.Restriction.NgoMIV.dna
Bio.Restriction.Restriction.NhaXI.dna
Bio.Restriction.Restriction.NheI.dna
Bio.Restriction.Restriction.NlaCI.dna
Bio.Restriction.Restriction.NlaIII.dna
Bio.Restriction.Restriction.NlaIV.dna
Bio.Restriction.Restriction.Nli3877I.dna
Bio.Restriction.Restriction.NmeAIII.dna
Bio.Restriction.Restriction.NmeDI.dna
Bio.Restriction.Restriction.NmuCI.dna
Bio.Restriction.Restriction.NotI.dna
Bio.Restriction.Restriction.NruI.dna
Bio.Restriction.Restriction.NsbI.dna
Bio.Restriction.Restriction.NsiI.dna
Bio.Restriction.Restriction.NspBII.dna
Bio.Restriction.Restriction.NspI.dna
Bio.Restriction.Restriction.NspV.dna
Bio.Restriction.Restriction.OliI.dna
Bio.Restriction.Restriction.PabI.dna
Bio.Restriction.Restriction.PacI.dna
Bio.Restriction.Restriction.PaeI.dna
Bio.Restriction.Restriction.PaeR7I.dna
Bio.Restriction.Restriction.PagI.dna
Bio.Restriction.Restriction.PalAI.dna
Bio.Restriction.Restriction.PasI.dna
Bio.Restriction.Restriction.PauI.dna
Bio.Restriction.Restriction.PceI.dna
Bio.Restriction.Restriction.PciI.dna
Bio.Restriction.Restriction.PciSI.dna
Bio.Restriction.Restriction.PcsI.dna
Bio.Restriction.Restriction.PctI.dna
Bio.Restriction.Restriction.PdiI.dna
Bio.Restriction.Restriction.PdmI.dna
Bio.Restriction.Restriction.PenI.dna
Bio.Restriction.Restriction.PfeI.dna
Bio.Restriction.Restriction.Pfl1108I.dna
Bio.Restriction.Restriction.Pfl23II.dna
Bio.Restriction.Restriction.PflFI.dna
Bio.Restriction.Restriction.PflMI.dna
Bio.Restriction.Restriction.PfoI.dna
Bio.Restriction.Restriction.PinAI.dna
Bio.Restriction.Restriction.PlaDI.dna
Bio.Restriction.Restriction.Ple19I.dna
Bio.Restriction.Restriction.PleI.dna
Bio.Restriction.Restriction.PluTI.dna
Bio.Restriction.Restriction.PmaCI.dna
Bio.Restriction.Restriction.PmeI.dna
Bio.Restriction.Restriction.PmlI.dna
Bio.Restriction.Restriction.PpiI.dna
Bio.Restriction.Restriction.PpsI.dna
Bio.Restriction.Restriction.Ppu10I.dna
Bio.Restriction.Restriction.Ppu21I.dna
Bio.Restriction.Restriction.PpuMI.dna
Bio.Restriction.Restriction.PscI.dna
Bio.Restriction.Restriction.PshAI.dna
Bio.Restriction.Restriction.PshBI.dna
Bio.Restriction.Restriction.PsiI.dna
Bio.Restriction.Restriction.Psp03I.dna
Bio.Restriction.Restriction.Psp124BI.dna
Bio.Restriction.Restriction.Psp1406I.dna
Bio.Restriction.Restriction.Psp5II.dna
Bio.Restriction.Restriction.Psp6I.dna
Bio.Restriction.Restriction.PspCI.dna
Bio.Restriction.Restriction.PspEI.dna
Bio.Restriction.Restriction.PspGI.dna
Bio.Restriction.Restriction.PspLI.dna
Bio.Restriction.Restriction.PspN4I.dna
Bio.Restriction.Restriction.PspOMI.dna
Bio.Restriction.Restriction.PspOMII.dna
Bio.Restriction.Restriction.PspPI.dna
Bio.Restriction.Restriction.PspPPI.dna
Bio.Restriction.Restriction.PspPRI.dna
Bio.Restriction.Restriction.PspXI.dna
Bio.Restriction.Restriction.PsrI.dna
Bio.Restriction.Restriction.PssI.dna
Bio.Restriction.Restriction.PstI.dna
Bio.Restriction.Restriction.PstNI.dna
Bio.Restriction.Restriction.PsuI.dna
Bio.Restriction.Restriction.PsyI.dna
Bio.Restriction.Restriction.PteI.dna
Bio.Restriction.Restriction.PvuI.dna
Bio.Restriction.Restriction.PvuII.dna
Bio.Restriction.Restriction.R2_BceSIV.dna
Bio.Restriction.Restriction.RceI.dna
Bio.Restriction.Restriction.RdeGBI.dna
Bio.Restriction.Restriction.RdeGBII.dna
Bio.Restriction.Restriction.RdeGBIII.dna
Bio.Restriction.Restriction.RflFIII.dna
Bio.Restriction.Restriction.RgaI.dna
Bio.Restriction.Restriction.RigI.dna
Bio.Restriction.Restriction.RlaI.dna
Bio.Restriction.Restriction.RleAI.dna
Bio.Restriction.Restriction.RpaB5I.dna
Bio.Restriction.Restriction.RpaBI.dna
Bio.Restriction.Restriction.RpaI.dna
Bio.Restriction.Restriction.RpaTI.dna
Bio.Restriction.Restriction.RruI.dna
Bio.Restriction.Restriction.RsaI.dna
Bio.Restriction.Restriction.RsaNI.dna
Bio.Restriction.Restriction.RseI.dna
Bio.Restriction.Restriction.Rsr2I.dna
Bio.Restriction.Restriction.RsrII.dna
Bio.Restriction.Restriction.SacI.dna
Bio.Restriction.Restriction.SacII.dna
Bio.Restriction.Restriction.SalI.dna
Bio.Restriction.Restriction.SanDI.dna
Bio.Restriction.Restriction.SapI.dna
Bio.Restriction.Restriction.SaqAI.dna
Bio.Restriction.Restriction.SatI.dna
Bio.Restriction.Restriction.Sau3AI.dna
Bio.Restriction.Restriction.Sau96I.dna
Bio.Restriction.Restriction.SauI.dna
Bio.Restriction.Restriction.SbfI.dna
Bio.Restriction.Restriction.ScaI.dna
Bio.Restriction.Restriction.SchI.dna
Bio.Restriction.Restriction.SciI.dna
Bio.Restriction.Restriction.ScrFI.dna
Bio.Restriction.Restriction.SdaI.dna
Bio.Restriction.Restriction.SdeAI.dna
Bio.Restriction.Restriction.SdeOSI.dna
Bio.Restriction.Restriction.SduI.dna
Bio.Restriction.Restriction.SecI.dna
Bio.Restriction.Restriction.SelI.dna
Bio.Restriction.Restriction.SetI.dna
Bio.Restriction.Restriction.SexAI.dna
Bio.Restriction.Restriction.SfaAI.dna
Bio.Restriction.Restriction.SfaNI.dna
Bio.Restriction.Restriction.SfcI.dna
Bio.Restriction.Restriction.SfeI.dna
Bio.Restriction.Restriction.SfiI.dna
Bio.Restriction.Restriction.SfoI.dna
Bio.Restriction.Restriction.Sfr274I.dna
Bio.Restriction.Restriction.Sfr303I.dna
Bio.Restriction.Restriction.SfuI.dna
Bio.Restriction.Restriction.SgeI.dna
Bio.Restriction.Restriction.SgfI.dna
Bio.Restriction.Restriction.SgrAI.dna
Bio.Restriction.Restriction.SgrBI.dna
Bio.Restriction.Restriction.SgrDI.dna
Bio.Restriction.Restriction.SgrTI.dna
Bio.Restriction.Restriction.SgsI.dna
Bio.Restriction.Restriction.SimI.dna
Bio.Restriction.Restriction.SlaI.dna
Bio.Restriction.Restriction.SmaI.dna
Bio.Restriction.Restriction.SmiI.dna
Bio.Restriction.Restriction.SmiMI.dna
Bio.Restriction.Restriction.SmlI.dna
Bio.Restriction.Restriction.SmoI.dna
Bio.Restriction.Restriction.SnaBI.dna
Bio.Restriction.Restriction.SnaI.dna
Bio.Restriction.Restriction.Sno506I.dna
Bio.Restriction.Restriction.SpeI.dna
Bio.Restriction.Restriction.SphI.dna
Bio.Restriction.Restriction.SplI.dna
Bio.Restriction.Restriction.SpoDI.dna
Bio.Restriction.Restriction.SrfI.dna
Bio.Restriction.Restriction.Sse232I.dna
Bio.Restriction.Restriction.Sse8387I.dna
Bio.Restriction.Restriction.Sse8647I.dna
Bio.Restriction.Restriction.Sse9I.dna
Bio.Restriction.Restriction.SseBI.dna
Bio.Restriction.Restriction.SsiI.dna
Bio.Restriction.Restriction.SspD5I.dna
Bio.Restriction.Restriction.SspDI.dna
Bio.Restriction.Restriction.SspI.dna
Bio.Restriction.Restriction.SstE37I.dna
Bio.Restriction.Restriction.SstI.dna
Bio.Restriction.Restriction.Sth132I.dna
Bio.Restriction.Restriction.Sth302II.dna
Bio.Restriction.Restriction.StrI.dna
Bio.Restriction.Restriction.StsI.dna
Bio.Restriction.Restriction.StuI.dna
Bio.Restriction.Restriction.StyD4I.dna
Bio.Restriction.Restriction.StyI.dna
Bio.Restriction.Restriction.SwaI.dna
Bio.Restriction.Restriction.TaaI.dna
Bio.Restriction.Restriction.TaiI.dna
Bio.Restriction.Restriction.TaqI.dna
Bio.Restriction.Restriction.TaqII.dna
Bio.Restriction.Restriction.TasI.dna
Bio.Restriction.Restriction.TatI.dna
Bio.Restriction.Restriction.TauI.dna
Bio.Restriction.Restriction.TfiI.dna
Bio.Restriction.Restriction.Tru1I.dna
Bio.Restriction.Restriction.Tru9I.dna
Bio.Restriction.Restriction.TscAI.dna
Bio.Restriction.Restriction.TseFI.dna
Bio.Restriction.Restriction.TseI.dna
Bio.Restriction.Restriction.TsoI.dna
Bio.Restriction.Restriction.Tsp45I.dna
Bio.Restriction.Restriction.Tsp4CI.dna
Bio.Restriction.Restriction.TspDTI.dna
Bio.Restriction.Restriction.TspEI.dna
Bio.Restriction.Restriction.TspGWI.dna
Bio.Restriction.Restriction.TspMI.dna
Bio.Restriction.Restriction.TspRI.dna
Bio.Restriction.Restriction.TssI.dna
Bio.Restriction.Restriction.TstI.dna
Bio.Restriction.Restriction.TsuI.dna
Bio.Restriction.Restriction.Tth111I.dna
Bio.Restriction.Restriction.Tth111II.dna
Bio.Restriction.Restriction.UbaF11I.dna
Bio.Restriction.Restriction.UbaF12I.dna
Bio.Restriction.Restriction.UbaF13I.dna
Bio.Restriction.Restriction.UbaF14I.dna
Bio.Restriction.Restriction.UbaF9I.dna
Bio.Restriction.Restriction.UbaPI.dna
Bio.Restriction.Restriction.UcoMSI.dna
Bio.Restriction.Restriction.UnbI.dna
Bio.Restriction.Restriction.Van91I.dna
Bio.Restriction.Restriction.Vha464I.dna
Bio.Restriction.Restriction.VneI.dna
Bio.Restriction.Restriction.VpaK11AI.dna
Bio.Restriction.Restriction.VpaK11BI.dna
Bio.Restriction.Restriction.VspI.dna
Bio.Restriction.Restriction.WviI.dna
Bio.Restriction.Restriction.XagI.dna
Bio.Restriction.Restriction.XapI.dna
Bio.Restriction.Restriction.XbaI.dna
Bio.Restriction.Restriction.XceI.dna
Bio.Restriction.Restriction.XcmI.dna
Bio.Restriction.Restriction.XhoI.dna
Bio.Restriction.Restriction.XhoII.dna
Bio.Restriction.Restriction.XmaI.dna
Bio.Restriction.Restriction.XmaIII.dna
Bio.Restriction.Restriction.XmaJI.dna
Bio.Restriction.Restriction.XmiI.dna
Bio.Restriction.Restriction.XmnI.dna
Bio.Restriction.Restriction.XspI.dna
Bio.Restriction.Restriction.YkrI.dna
Bio.Restriction.Restriction.ZraI.dna
Bio.Restriction.Restriction.ZrmI.dna
Bio.Restriction.Restriction.Zsp2I.dna" class="py-name" href="#" onclick="return doclink('link-100', 'dna', 'link-9');">dna</a></tt><tt class="py-op">,</tt> <tt class="py-name">counts</tt><tt class="py-op">)</tt> </tt>
</div><a name="L461"></a><tt class="py-lineno">461</tt>  <tt class="py-line"> </tt>
<a name="JASPAR5._fetch_internal_id_list"></a><div id="JASPAR5._fetch_internal_id_list-def"><a name="L462"></a><tt class="py-lineno">462</tt> <a class="py-toggle" href="#" id="JASPAR5._fetch_internal_id_list-toggle" onclick="return toggle('JASPAR5._fetch_internal_id_list');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.jaspar.db.JASPAR5-class.html#_fetch_internal_id_list">_fetch_internal_id_list</a><tt class="py-op">(</tt> </tt>
<a name="L463"></a><tt class="py-lineno">463</tt>  <tt class="py-line">        <tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">collection</tt><tt class="py-op">=</tt><tt id="link-101" class="py-name"><a title="Bio.motifs.jaspar.db.JASPAR_DFLT_COLLECTION" class="py-name" href="#" onclick="return doclink('link-101', 'JASPAR_DFLT_COLLECTION', 'link-14');">JASPAR_DFLT_COLLECTION</a></tt><tt class="py-op">,</tt> <tt class="py-param">tf_name</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">tf_class</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L464"></a><tt class="py-lineno">464</tt>  <tt class="py-line">        <tt class="py-param">tf_family</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">matrix_id</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">tax_group</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">species</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L465"></a><tt class="py-lineno">465</tt>  <tt class="py-line">        <tt class="py-param">pazar_id</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">data_type</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">medline</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">all</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> </tt>
<a name="L466"></a><tt class="py-lineno">466</tt>  <tt class="py-line">        <tt class="py-param">all_versions</tt><tt class="py-op">=</tt><tt class="py-name">False</tt> </tt>
<a name="L467"></a><tt class="py-lineno">467</tt>  <tt class="py-line">    <tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="JASPAR5._fetch_internal_id_list-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="JASPAR5._fetch_internal_id_list-expanded"><a name="L468"></a><tt class="py-lineno">468</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L469"></a><tt class="py-lineno">469</tt>  <tt class="py-line"><tt class="py-docstring">        Fetch a list of internal JASPAR motif IDs based on various passed</tt> </tt>
<a name="L470"></a><tt class="py-lineno">470</tt>  <tt class="py-line"><tt class="py-docstring">        parameters which may then be used to fetch the rest of the motif data.</tt> </tt>
<a name="L471"></a><tt class="py-lineno">471</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L472"></a><tt class="py-lineno">472</tt>  <tt class="py-line"><tt class="py-docstring">        Caller:</tt> </tt>
<a name="L473"></a><tt class="py-lineno">473</tt>  <tt class="py-line"><tt class="py-docstring">            fetch_motifs()</tt> </tt>
<a name="L474"></a><tt class="py-lineno">474</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L475"></a><tt class="py-lineno">475</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L476"></a><tt class="py-lineno">476</tt>  <tt class="py-line"><tt class="py-docstring">            See arguments sections of fetch_motifs()</tt> </tt>
<a name="L477"></a><tt class="py-lineno">477</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L478"></a><tt class="py-lineno">478</tt>  <tt class="py-line"><tt class="py-docstring">        Returns:</tt> </tt>
<a name="L479"></a><tt class="py-lineno">479</tt>  <tt class="py-line"><tt class="py-docstring">            A list of internal JASPAR motif IDs which match the given</tt> </tt>
<a name="L480"></a><tt class="py-lineno">480</tt>  <tt class="py-line"><tt class="py-docstring">            selection criteria arguments.</tt> </tt>
<a name="L481"></a><tt class="py-lineno">481</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L482"></a><tt class="py-lineno">482</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L483"></a><tt class="py-lineno">483</tt>  <tt class="py-line"><tt class="py-docstring">        Build an SQL query based on the selection arguments provided.</tt> </tt>
<a name="L484"></a><tt class="py-lineno">484</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L485"></a><tt class="py-lineno">485</tt>  <tt class="py-line"><tt class="py-docstring">        1: First add table joins and sub-clauses for criteria corresponding to</tt> </tt>
<a name="L486"></a><tt class="py-lineno">486</tt>  <tt class="py-line"><tt class="py-docstring">           named fields from the MATRIX and MATRIX_SPECIES tables such as</tt> </tt>
<a name="L487"></a><tt class="py-lineno">487</tt>  <tt class="py-line"><tt class="py-docstring">           collection, matrix ID, name, species etc.</tt> </tt>
<a name="L488"></a><tt class="py-lineno">488</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L489"></a><tt class="py-lineno">489</tt>  <tt class="py-line"><tt class="py-docstring">        2: Then add joins/sub-clauses for tag/value parameters from the</tt> </tt>
<a name="L490"></a><tt class="py-lineno">490</tt>  <tt class="py-line"><tt class="py-docstring">           MATRIX_ANNOTATION table.</tt> </tt>
<a name="L491"></a><tt class="py-lineno">491</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L492"></a><tt class="py-lineno">492</tt>  <tt class="py-line"><tt class="py-docstring">        For the surviving matrices, the responsibility to do matrix-based</tt> </tt>
<a name="L493"></a><tt class="py-lineno">493</tt>  <tt class="py-line"><tt class="py-docstring">        feature filtering such as ic, number of sites etc, fall on the</tt> </tt>
<a name="L494"></a><tt class="py-lineno">494</tt>  <tt class="py-line"><tt class="py-docstring">        calling fetch_motifs() method.</tt> </tt>
<a name="L495"></a><tt class="py-lineno">495</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L496"></a><tt class="py-lineno">496</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L497"></a><tt class="py-lineno">497</tt>  <tt class="py-line"> </tt>
<a name="L498"></a><tt class="py-lineno">498</tt>  <tt class="py-line">        <tt class="py-name">int_ids</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L499"></a><tt class="py-lineno">499</tt>  <tt class="py-line"> </tt>
<a name="L500"></a><tt class="py-lineno">500</tt>  <tt class="py-line">        <tt class="py-name">cur</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbh</tt><tt class="py-op">.</tt><tt id="link-102" class="py-name"><a title="Bio.DocSQL.Query.cursor
Bio.DocSQL.QuerySingle.cursor" class="py-name" href="#" onclick="return doclink('link-102', 'cursor', 'link-49');">cursor</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L501"></a><tt class="py-lineno">501</tt>  <tt class="py-line"> </tt>
<a name="L502"></a><tt class="py-lineno">502</tt>  <tt class="py-line">        <tt class="py-string">"""</tt> </tt>
<a name="L503"></a><tt class="py-lineno">503</tt>  <tt class="py-line"><tt class="py-string">        Special case 1: fetch ALL motifs. Highest priority.</tt> </tt>
<a name="L504"></a><tt class="py-lineno">504</tt>  <tt class="py-line"><tt class="py-string">        Ignore all other selection arguments.</tt> </tt>
<a name="L505"></a><tt class="py-lineno">505</tt>  <tt class="py-line"><tt class="py-string">        """</tt> </tt>
<a name="L506"></a><tt class="py-lineno">506</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">all</tt><tt class="py-op">:</tt> </tt>
<a name="L507"></a><tt class="py-lineno">507</tt>  <tt class="py-line">            <tt class="py-name">cur</tt><tt class="py-op">.</tt><tt id="link-103" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-103', 'execute', 'link-50');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"select ID from MATRIX"</tt><tt class="py-op">)</tt> </tt>
<a name="L508"></a><tt class="py-lineno">508</tt>  <tt class="py-line">            <tt class="py-name">rows</tt> <tt class="py-op">=</tt> <tt class="py-name">cur</tt><tt class="py-op">.</tt><tt id="link-104" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchall" class="py-name" href="#" onclick="return doclink('link-104', 'fetchall', 'link-80');">fetchall</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L509"></a><tt class="py-lineno">509</tt>  <tt class="py-line"> </tt>
<a name="L510"></a><tt class="py-lineno">510</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">row</tt> <tt class="py-keyword">in</tt> <tt class="py-name">rows</tt><tt class="py-op">:</tt> </tt>
<a name="L511"></a><tt class="py-lineno">511</tt>  <tt class="py-line">                <tt class="py-name">int_ids</tt><tt class="py-op">.</tt><tt id="link-105" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-105', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L512"></a><tt class="py-lineno">512</tt>  <tt class="py-line"> </tt>
<a name="L513"></a><tt class="py-lineno">513</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">int_ids</tt> </tt>
<a name="L514"></a><tt class="py-lineno">514</tt>  <tt class="py-line"> </tt>
<a name="L515"></a><tt class="py-lineno">515</tt>  <tt class="py-line">        <tt class="py-string">"""</tt> </tt>
<a name="L516"></a><tt class="py-lineno">516</tt>  <tt class="py-line"><tt class="py-string">        Special case 2: fetch specific motifs by their JASPAR IDs. This</tt> </tt>
<a name="L517"></a><tt class="py-lineno">517</tt>  <tt class="py-line"><tt class="py-string">        has higher priority than any other except the above 'all' case.</tt> </tt>
<a name="L518"></a><tt class="py-lineno">518</tt>  <tt class="py-line"><tt class="py-string">        Ignore all other selection arguments.</tt> </tt>
<a name="L519"></a><tt class="py-lineno">519</tt>  <tt class="py-line"><tt class="py-string">        """</tt> </tt>
<a name="L520"></a><tt class="py-lineno">520</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">matrix_id</tt><tt class="py-op">:</tt> </tt>
<a name="L521"></a><tt class="py-lineno">521</tt>  <tt class="py-line">            <tt class="py-string">"""</tt> </tt>
<a name="L522"></a><tt class="py-lineno">522</tt>  <tt class="py-line"><tt class="py-string">            These might be either stable IDs or stable_ID.version.</tt> </tt>
<a name="L523"></a><tt class="py-lineno">523</tt>  <tt class="py-line"><tt class="py-string">            If just stable ID and if all_versions == 1, return all versions,</tt> </tt>
<a name="L524"></a><tt class="py-lineno">524</tt>  <tt class="py-line"><tt class="py-string">            otherwise just the latest</tt> </tt>
<a name="L525"></a><tt class="py-lineno">525</tt>  <tt class="py-line"><tt class="py-string">            """</tt> </tt>
<a name="L526"></a><tt class="py-lineno">526</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">all_versions</tt><tt class="py-op">:</tt> </tt>
<a name="L527"></a><tt class="py-lineno">527</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt id="link-106" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-106', 'id', 'link-23');">id</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">matrix_id</tt><tt class="py-op">:</tt> </tt>
<a name="L528"></a><tt class="py-lineno">528</tt>  <tt class="py-line">                    <tt class="py-comment"># ignore vesion here, this is a stupidity filter</tt> </tt>
<a name="L529"></a><tt class="py-lineno">529</tt>  <tt class="py-line">                    <tt class="py-op">(</tt><tt id="link-107" class="py-name"><a title="Bio.motifs.jaspar.Motif.base_id" class="py-name" href="#" onclick="return doclink('link-107', 'base_id', 'link-19');">base_id</a></tt><tt class="py-op">,</tt> <tt id="link-108" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-108', 'version', 'link-20');">version</a></tt><tt class="py-op">)</tt> <tt class="py-op">=</tt> <tt id="link-109" class="py-name"><a title="Bio.motifs.jaspar" class="py-name" href="#" onclick="return doclink('link-109', 'jaspar', 'link-12');">jaspar</a></tt><tt class="py-op">.</tt><tt id="link-110" class="py-name"><a title="Bio.motifs.jaspar.split_jaspar_id" class="py-name" href="#" onclick="return doclink('link-110', 'split_jaspar_id', 'link-22');">split_jaspar_id</a></tt><tt class="py-op">(</tt><tt id="link-111" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-111', 'id', 'link-23');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L530"></a><tt class="py-lineno">530</tt>  <tt class="py-line">                    <tt class="py-name">cur</tt><tt class="py-op">.</tt><tt id="link-112" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-112', 'execute', 'link-50');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L531"></a><tt class="py-lineno">531</tt>  <tt class="py-line">                        <tt class="py-string">"select ID from MATRIX where BASE_ID = %s"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt id="link-113" class="py-name"><a title="Bio.motifs.jaspar.Motif.base_id" class="py-name" href="#" onclick="return doclink('link-113', 'base_id', 'link-19');">base_id</a></tt><tt class="py-op">,</tt><tt class="py-op">)</tt> </tt>
<a name="L532"></a><tt class="py-lineno">532</tt>  <tt class="py-line">                    <tt class="py-op">)</tt> </tt>
<a name="L533"></a><tt class="py-lineno">533</tt>  <tt class="py-line"> </tt>
<a name="L534"></a><tt class="py-lineno">534</tt>  <tt class="py-line">                    <tt class="py-name">rows</tt> <tt class="py-op">=</tt> <tt class="py-name">cur</tt><tt class="py-op">.</tt><tt id="link-114" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchall" class="py-name" href="#" onclick="return doclink('link-114', 'fetchall', 'link-80');">fetchall</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L535"></a><tt class="py-lineno">535</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-name">row</tt> <tt class="py-keyword">in</tt> <tt class="py-name">rows</tt><tt class="py-op">:</tt> </tt>
<a name="L536"></a><tt class="py-lineno">536</tt>  <tt class="py-line">                        <tt class="py-name">int_ids</tt><tt class="py-op">.</tt><tt id="link-115" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-115', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L537"></a><tt class="py-lineno">537</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L538"></a><tt class="py-lineno">538</tt>  <tt class="py-line">                <tt class="py-comment"># only the lastest version, or the requested version</tt> </tt>
<a name="L539"></a><tt class="py-lineno">539</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt id="link-116" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-116', 'id', 'link-23');">id</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">matrix_id</tt><tt class="py-op">:</tt> </tt>
<a name="L540"></a><tt class="py-lineno">540</tt>  <tt class="py-line">                    <tt class="py-op">(</tt><tt id="link-117" class="py-name"><a title="Bio.motifs.jaspar.Motif.base_id" class="py-name" href="#" onclick="return doclink('link-117', 'base_id', 'link-19');">base_id</a></tt><tt class="py-op">,</tt> <tt id="link-118" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-118', 'version', 'link-20');">version</a></tt><tt class="py-op">)</tt> <tt class="py-op">=</tt> <tt id="link-119" class="py-name"><a title="Bio.motifs.jaspar" class="py-name" href="#" onclick="return doclink('link-119', 'jaspar', 'link-12');">jaspar</a></tt><tt class="py-op">.</tt><tt id="link-120" class="py-name"><a title="Bio.motifs.jaspar.split_jaspar_id" class="py-name" href="#" onclick="return doclink('link-120', 'split_jaspar_id', 'link-22');">split_jaspar_id</a></tt><tt class="py-op">(</tt><tt id="link-121" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-121', 'id', 'link-23');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L541"></a><tt class="py-lineno">541</tt>  <tt class="py-line"> </tt>
<a name="L542"></a><tt class="py-lineno">542</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-122" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-122', 'version', 'link-20');">version</a></tt><tt class="py-op">:</tt> </tt>
<a name="L543"></a><tt class="py-lineno">543</tt>  <tt class="py-line">                        <tt id="link-123" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-123', 'version', 'link-20');">version</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-124" class="py-name"><a title="Bio.motifs.jaspar.db.JASPAR5._fetch_latest_version" class="py-name" href="#" onclick="return doclink('link-124', '_fetch_latest_version', 'link-26');">_fetch_latest_version</a></tt><tt class="py-op">(</tt><tt id="link-125" class="py-name"><a title="Bio.motifs.jaspar.Motif.base_id" class="py-name" href="#" onclick="return doclink('link-125', 'base_id', 'link-19');">base_id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L544"></a><tt class="py-lineno">544</tt>  <tt class="py-line"> </tt>
<a name="L545"></a><tt class="py-lineno">545</tt>  <tt class="py-line">                    <tt class="py-name">int_id</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L546"></a><tt class="py-lineno">546</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt id="link-126" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-126', 'version', 'link-20');">version</a></tt><tt class="py-op">:</tt> </tt>
<a name="L547"></a><tt class="py-lineno">547</tt>  <tt class="py-line">                        <tt class="py-name">int_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-127" class="py-name"><a title="Bio.motifs.jaspar.db.JASPAR5._fetch_internal_id" class="py-name" href="#" onclick="return doclink('link-127', '_fetch_internal_id', 'link-29');">_fetch_internal_id</a></tt><tt class="py-op">(</tt><tt id="link-128" class="py-name"><a title="Bio.motifs.jaspar.Motif.base_id" class="py-name" href="#" onclick="return doclink('link-128', 'base_id', 'link-19');">base_id</a></tt><tt class="py-op">,</tt> <tt id="link-129" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-129', 'version', 'link-20');">version</a></tt><tt class="py-op">)</tt> </tt>
<a name="L548"></a><tt class="py-lineno">548</tt>  <tt class="py-line"> </tt>
<a name="L549"></a><tt class="py-lineno">549</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">int_id</tt><tt class="py-op">:</tt> </tt>
<a name="L550"></a><tt class="py-lineno">550</tt>  <tt class="py-line">                        <tt class="py-name">int_ids</tt><tt class="py-op">.</tt><tt id="link-130" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-130', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">int_id</tt><tt class="py-op">)</tt> </tt>
<a name="L551"></a><tt class="py-lineno">551</tt>  <tt class="py-line"> </tt>
<a name="L552"></a><tt class="py-lineno">552</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">int_ids</tt> </tt>
<a name="L553"></a><tt class="py-lineno">553</tt>  <tt class="py-line"> </tt>
<a name="L554"></a><tt class="py-lineno">554</tt>  <tt class="py-line">        <tt class="py-name">tables</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">"MATRIX m"</tt><tt class="py-op">]</tt> </tt>
<a name="L555"></a><tt class="py-lineno">555</tt>  <tt class="py-line">        <tt class="py-name">where_clauses</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L556"></a><tt class="py-lineno">556</tt>  <tt class="py-line"> </tt>
<a name="L557"></a><tt class="py-lineno">557</tt>  <tt class="py-line">        <tt class="py-comment"># Select by MATRIX.COLLECTION</tt> </tt>
<a name="L558"></a><tt class="py-lineno">558</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">collection</tt><tt class="py-op">:</tt> </tt>
<a name="L559"></a><tt class="py-lineno">559</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">collection</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L560"></a><tt class="py-lineno">560</tt>  <tt class="py-line">                <tt class="py-comment"># Multiple collections passed in as a list</tt> </tt>
<a name="L561"></a><tt class="py-lineno">561</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">"m.COLLECTION in ('"</tt> </tt>
<a name="L562"></a><tt class="py-lineno">562</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">"','"</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">collection</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L563"></a><tt class="py-lineno">563</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">"')"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L564"></a><tt class="py-lineno">564</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L565"></a><tt class="py-lineno">565</tt>  <tt class="py-line">                <tt class="py-comment"># A single collection - typical usage</tt> </tt>
<a name="L566"></a><tt class="py-lineno">566</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">"m.COLLECTION = '%s'"</tt> <tt class="py-op">%</tt> <tt class="py-name">collection</tt> </tt>
<a name="L567"></a><tt class="py-lineno">567</tt>  <tt class="py-line"> </tt>
<a name="L568"></a><tt class="py-lineno">568</tt>  <tt class="py-line">            <tt class="py-name">where_clauses</tt><tt class="py-op">.</tt><tt id="link-131" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-131', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">clause</tt><tt class="py-op">)</tt> </tt>
<a name="L569"></a><tt class="py-lineno">569</tt>  <tt class="py-line"> </tt>
<a name="L570"></a><tt class="py-lineno">570</tt>  <tt class="py-line">        <tt class="py-comment"># Select by MATRIX.NAME</tt> </tt>
<a name="L571"></a><tt class="py-lineno">571</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">tf_name</tt><tt class="py-op">:</tt> </tt>
<a name="L572"></a><tt class="py-lineno">572</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">tf_name</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L573"></a><tt class="py-lineno">573</tt>  <tt class="py-line">                <tt class="py-comment"># Multiple names passed in as a list</tt> </tt>
<a name="L574"></a><tt class="py-lineno">574</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">"m.NAME in ('"</tt> </tt>
<a name="L575"></a><tt class="py-lineno">575</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">"','"</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">tf_name</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L576"></a><tt class="py-lineno">576</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">"')"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L577"></a><tt class="py-lineno">577</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L578"></a><tt class="py-lineno">578</tt>  <tt class="py-line">                <tt class="py-comment"># A single name</tt> </tt>
<a name="L579"></a><tt class="py-lineno">579</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">"m.NAME = '%s'"</tt> <tt class="py-op">%</tt> <tt class="py-name">tf_name</tt> </tt>
<a name="L580"></a><tt class="py-lineno">580</tt>  <tt class="py-line"> </tt>
<a name="L581"></a><tt class="py-lineno">581</tt>  <tt class="py-line">            <tt class="py-name">where_clauses</tt><tt class="py-op">.</tt><tt id="link-132" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-132', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">clause</tt><tt class="py-op">)</tt> </tt>
<a name="L582"></a><tt class="py-lineno">582</tt>  <tt class="py-line"> </tt>
<a name="L583"></a><tt class="py-lineno">583</tt>  <tt class="py-line">        <tt class="py-comment"># Select by MATRIX_SPECIES.TAX_ID</tt> </tt>
<a name="L584"></a><tt class="py-lineno">584</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-133" class="py-name"><a title="Bio.Pathway.Network.species
Bio.Pathway.Reaction.species
Bio.Pathway.System.species" class="py-name" href="#" onclick="return doclink('link-133', 'species', 'link-37');">species</a></tt><tt class="py-op">:</tt> </tt>
<a name="L585"></a><tt class="py-lineno">585</tt>  <tt class="py-line">            <tt class="py-name">tables</tt><tt class="py-op">.</tt><tt id="link-134" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-134', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"MATRIX_SPECIES ms"</tt><tt class="py-op">)</tt> </tt>
<a name="L586"></a><tt class="py-lineno">586</tt>  <tt class="py-line">            <tt class="py-name">where_clauses</tt><tt class="py-op">.</tt><tt id="link-135" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-135', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"m.ID = ms.ID"</tt><tt class="py-op">)</tt> </tt>
<a name="L587"></a><tt class="py-lineno">587</tt>  <tt class="py-line"> </tt>
<a name="L588"></a><tt class="py-lineno">588</tt>  <tt class="py-line">            <tt class="py-string">"""</tt> </tt>
<a name="L589"></a><tt class="py-lineno">589</tt>  <tt class="py-line"><tt class="py-string">            NOTE: species are numeric taxonomy IDs but stored as varchars</tt> </tt>
<a name="L590"></a><tt class="py-lineno">590</tt>  <tt class="py-line"><tt class="py-string">            in the DB.</tt> </tt>
<a name="L591"></a><tt class="py-lineno">591</tt>  <tt class="py-line"><tt class="py-string">            """</tt> </tt>
<a name="L592"></a><tt class="py-lineno">592</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-136" class="py-name"><a title="Bio.Pathway.Network.species
Bio.Pathway.Reaction.species
Bio.Pathway.System.species" class="py-name" href="#" onclick="return doclink('link-136', 'species', 'link-37');">species</a></tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L593"></a><tt class="py-lineno">593</tt>  <tt class="py-line">                <tt class="py-comment"># Multiple tax IDs passed in as a list</tt> </tt>
<a name="L594"></a><tt class="py-lineno">594</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">"ms.TAX_ID in ('"</tt> </tt>
<a name="L595"></a><tt class="py-lineno">595</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">"','"</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">s</tt> <tt class="py-keyword">in</tt> <tt id="link-137" class="py-name"><a title="Bio.Pathway.Network.species
Bio.Pathway.Reaction.species
Bio.Pathway.System.species" class="py-name" href="#" onclick="return doclink('link-137', 'species', 'link-37');">species</a></tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L596"></a><tt class="py-lineno">596</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">"')"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L597"></a><tt class="py-lineno">597</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L598"></a><tt class="py-lineno">598</tt>  <tt class="py-line">                <tt class="py-comment"># A single tax ID</tt> </tt>
<a name="L599"></a><tt class="py-lineno">599</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">"ms.TAX_ID = '%s'"</tt> <tt class="py-op">%</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-138" class="py-name"><a title="Bio.Pathway.Network.species
Bio.Pathway.Reaction.species
Bio.Pathway.System.species" class="py-name" href="#" onclick="return doclink('link-138', 'species', 'link-37');">species</a></tt><tt class="py-op">)</tt> </tt>
<a name="L600"></a><tt class="py-lineno">600</tt>  <tt class="py-line"> </tt>
<a name="L601"></a><tt class="py-lineno">601</tt>  <tt class="py-line">            <tt class="py-name">where_clauses</tt><tt class="py-op">.</tt><tt id="link-139" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-139', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">clause</tt><tt class="py-op">)</tt> </tt>
<a name="L602"></a><tt class="py-lineno">602</tt>  <tt class="py-line"> </tt>
<a name="L603"></a><tt class="py-lineno">603</tt>  <tt class="py-line">        <tt class="py-string">"""</tt> </tt>
<a name="L604"></a><tt class="py-lineno">604</tt>  <tt class="py-line"><tt class="py-string">        Tag based selection from MATRIX_ANNOTATION</tt> </tt>
<a name="L605"></a><tt class="py-lineno">605</tt>  <tt class="py-line"><tt class="py-string">        Differs from perl TFBS module in that the matrix class explicitly</tt> </tt>
<a name="L606"></a><tt class="py-lineno">606</tt>  <tt class="py-line"><tt class="py-string">        has a tag attribute corresponding to the tags in the database. This</tt> </tt>
<a name="L607"></a><tt class="py-lineno">607</tt>  <tt class="py-line"><tt class="py-string">        provides tremendous flexibility in adding new tags to the DB and</tt> </tt>
<a name="L608"></a><tt class="py-lineno">608</tt>  <tt class="py-line"><tt class="py-string">        being able to select based on those tags with out adding new code.</tt> </tt>
<a name="L609"></a><tt class="py-lineno">609</tt>  <tt class="py-line"><tt class="py-string">        In the JASPAR Motif class we have elected to use specific attributes</tt> </tt>
<a name="L610"></a><tt class="py-lineno">610</tt>  <tt class="py-line"><tt class="py-string">        for the most commonly used tags and here correspondingly only allow</tt> </tt>
<a name="L611"></a><tt class="py-lineno">611</tt>  <tt class="py-line"><tt class="py-string">        selection on these attributes.</tt> </tt>
<a name="L612"></a><tt class="py-lineno">612</tt>  <tt class="py-line"><tt class="py-string"></tt> </tt>
<a name="L613"></a><tt class="py-lineno">613</tt>  <tt class="py-line"><tt class="py-string">        The attributes corresponding to the tags for which selection is</tt> </tt>
<a name="L614"></a><tt class="py-lineno">614</tt>  <tt class="py-line"><tt class="py-string">        provided are:</tt> </tt>
<a name="L615"></a><tt class="py-lineno">615</tt>  <tt class="py-line"><tt class="py-string"></tt> </tt>
<a name="L616"></a><tt class="py-lineno">616</tt>  <tt class="py-line"><tt class="py-string">           Attribute   Tag</tt> </tt>
<a name="L617"></a><tt class="py-lineno">617</tt>  <tt class="py-line"><tt class="py-string">           tf_class    class</tt> </tt>
<a name="L618"></a><tt class="py-lineno">618</tt>  <tt class="py-line"><tt class="py-string">           tf_family   family</tt> </tt>
<a name="L619"></a><tt class="py-lineno">619</tt>  <tt class="py-line"><tt class="py-string">           pazar_id    pazar_tf_id</tt> </tt>
<a name="L620"></a><tt class="py-lineno">620</tt>  <tt class="py-line"><tt class="py-string">           medline     medline</tt> </tt>
<a name="L621"></a><tt class="py-lineno">621</tt>  <tt class="py-line"><tt class="py-string">           data_type   type</tt> </tt>
<a name="L622"></a><tt class="py-lineno">622</tt>  <tt class="py-line"><tt class="py-string">           tax_group   tax_group</tt> </tt>
<a name="L623"></a><tt class="py-lineno">623</tt>  <tt class="py-line"><tt class="py-string">        """</tt> </tt>
<a name="L624"></a><tt class="py-lineno">624</tt>  <tt class="py-line"> </tt>
<a name="L625"></a><tt class="py-lineno">625</tt>  <tt class="py-line">        <tt class="py-comment"># Select by TF class(es) (MATRIX_ANNOTATION.TAG="class")</tt> </tt>
<a name="L626"></a><tt class="py-lineno">626</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">tf_class</tt><tt class="py-op">:</tt> </tt>
<a name="L627"></a><tt class="py-lineno">627</tt>  <tt class="py-line">            <tt class="py-name">tables</tt><tt class="py-op">.</tt><tt id="link-140" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-140', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"MATRIX_ANNOTATION ma1"</tt><tt class="py-op">)</tt> </tt>
<a name="L628"></a><tt class="py-lineno">628</tt>  <tt class="py-line">            <tt class="py-name">where_clauses</tt><tt class="py-op">.</tt><tt id="link-141" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-141', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"m.ID = ma1.ID"</tt><tt class="py-op">)</tt> </tt>
<a name="L629"></a><tt class="py-lineno">629</tt>  <tt class="py-line"> </tt>
<a name="L630"></a><tt class="py-lineno">630</tt>  <tt class="py-line">            <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">"ma1.TAG = 'class'"</tt> </tt>
<a name="L631"></a><tt class="py-lineno">631</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">tf_class</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L632"></a><tt class="py-lineno">632</tt>  <tt class="py-line">                <tt class="py-comment"># A list of TF classes</tt> </tt>
<a name="L633"></a><tt class="py-lineno">633</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">" and ma1.VAL in ('"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L634"></a><tt class="py-lineno">634</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">"','"</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">tf_class</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L635"></a><tt class="py-lineno">635</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">"')"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L636"></a><tt class="py-lineno">636</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L637"></a><tt class="py-lineno">637</tt>  <tt class="py-line">                <tt class="py-comment"># A single TF class</tt> </tt>
<a name="L638"></a><tt class="py-lineno">638</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">" and ma1.VAL = '%s' "</tt> <tt class="py-op">%</tt> <tt class="py-name">tf_class</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L639"></a><tt class="py-lineno">639</tt>  <tt class="py-line"> </tt>
<a name="L640"></a><tt class="py-lineno">640</tt>  <tt class="py-line">            <tt class="py-name">where_clauses</tt><tt class="py-op">.</tt><tt id="link-142" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-142', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">clause</tt><tt class="py-op">)</tt> </tt>
<a name="L641"></a><tt class="py-lineno">641</tt>  <tt class="py-line"> </tt>
<a name="L642"></a><tt class="py-lineno">642</tt>  <tt class="py-line">        <tt class="py-comment"># Select by TF families (MATRIX_ANNOTATION.TAG="family")</tt> </tt>
<a name="L643"></a><tt class="py-lineno">643</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">tf_family</tt><tt class="py-op">:</tt> </tt>
<a name="L644"></a><tt class="py-lineno">644</tt>  <tt class="py-line">            <tt class="py-name">tables</tt><tt class="py-op">.</tt><tt id="link-143" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-143', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"MATRIX_ANNOTATION ma2"</tt><tt class="py-op">)</tt> </tt>
<a name="L645"></a><tt class="py-lineno">645</tt>  <tt class="py-line">            <tt class="py-name">where_clauses</tt><tt class="py-op">.</tt><tt id="link-144" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-144', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"m.ID = ma2.ID"</tt><tt class="py-op">)</tt> </tt>
<a name="L646"></a><tt class="py-lineno">646</tt>  <tt class="py-line"> </tt>
<a name="L647"></a><tt class="py-lineno">647</tt>  <tt class="py-line">            <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">"ma2.TAG = 'family'"</tt> </tt>
<a name="L648"></a><tt class="py-lineno">648</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">tf_family</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L649"></a><tt class="py-lineno">649</tt>  <tt class="py-line">                <tt class="py-comment"># A list of TF families</tt> </tt>
<a name="L650"></a><tt class="py-lineno">650</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">" and ma2.VAL in ('"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L651"></a><tt class="py-lineno">651</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">"','"</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">tf_family</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L652"></a><tt class="py-lineno">652</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">"')"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L653"></a><tt class="py-lineno">653</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L654"></a><tt class="py-lineno">654</tt>  <tt class="py-line">                <tt class="py-comment"># A single TF family</tt> </tt>
<a name="L655"></a><tt class="py-lineno">655</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">" and ma2.VAL = '%s' "</tt> <tt class="py-op">%</tt> <tt class="py-name">tf_family</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L656"></a><tt class="py-lineno">656</tt>  <tt class="py-line"> </tt>
<a name="L657"></a><tt class="py-lineno">657</tt>  <tt class="py-line">            <tt class="py-name">where_clauses</tt><tt class="py-op">.</tt><tt id="link-145" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-145', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">clause</tt><tt class="py-op">)</tt> </tt>
<a name="L658"></a><tt class="py-lineno">658</tt>  <tt class="py-line"> </tt>
<a name="L659"></a><tt class="py-lineno">659</tt>  <tt class="py-line">        <tt class="py-comment"># Select by PAZAR TF ID(s) (MATRIX_ANNOTATION.TAG="pazar_tf_id")</tt> </tt>
<a name="L660"></a><tt class="py-lineno">660</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">pazar_id</tt><tt class="py-op">:</tt> </tt>
<a name="L661"></a><tt class="py-lineno">661</tt>  <tt class="py-line">            <tt class="py-name">tables</tt><tt class="py-op">.</tt><tt id="link-146" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-146', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"MATRIX_ANNOTATION ma3"</tt><tt class="py-op">)</tt> </tt>
<a name="L662"></a><tt class="py-lineno">662</tt>  <tt class="py-line">            <tt class="py-name">where_clauses</tt><tt class="py-op">.</tt><tt id="link-147" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-147', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"m.ID = ma3.ID"</tt><tt class="py-op">)</tt> </tt>
<a name="L663"></a><tt class="py-lineno">663</tt>  <tt class="py-line"> </tt>
<a name="L664"></a><tt class="py-lineno">664</tt>  <tt class="py-line">            <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">"ma3.TAG = 'pazar_tf_id'"</tt> </tt>
<a name="L665"></a><tt class="py-lineno">665</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">pazar_id</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L666"></a><tt class="py-lineno">666</tt>  <tt class="py-line">                <tt class="py-comment"># A list of PAZAR IDs</tt> </tt>
<a name="L667"></a><tt class="py-lineno">667</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">" and ma3.VAL in ('"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L668"></a><tt class="py-lineno">668</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">"','"</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">pazar_id</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L669"></a><tt class="py-lineno">669</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">"')"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L670"></a><tt class="py-lineno">670</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L671"></a><tt class="py-lineno">671</tt>  <tt class="py-line">                <tt class="py-comment"># A single PAZAR ID</tt> </tt>
<a name="L672"></a><tt class="py-lineno">672</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-string">" and ma3.VAL = '%s' "</tt> <tt class="py-op">%</tt> <tt class="py-name">pazar_id</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L673"></a><tt class="py-lineno">673</tt>  <tt class="py-line"> </tt>
<a name="L674"></a><tt class="py-lineno">674</tt>  <tt class="py-line">            <tt class="py-name">where_clauses</tt><tt class="py-op">.</tt><tt id="link-148" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-148', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">clause</tt><tt class="py-op">)</tt> </tt>
<a name="L675"></a><tt class="py-lineno">675</tt>  <tt class="py-line"> </tt>
<a name="L676"></a><tt class="py-lineno">676</tt>  <tt class="py-line">        <tt class="py-comment"># Select by PubMed ID(s) (MATRIX_ANNOTATION.TAG="medline")</tt> </tt>
<a name="L677"></a><tt class="py-lineno">677</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">medline</tt><tt class="py-op">:</tt> </tt>
<a name="L678"></a><tt class="py-lineno">678</tt>  <tt class="py-line">            <tt class="py-name">tables</tt><tt class="py-op">.</tt><tt id="link-149" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-149', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"MATRIX_ANNOTATION ma4"</tt><tt class="py-op">)</tt> </tt>
<a name="L679"></a><tt class="py-lineno">679</tt>  <tt class="py-line">            <tt class="py-name">where_clauses</tt><tt class="py-op">.</tt><tt id="link-150" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-150', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"m.ID = ma4.ID"</tt><tt class="py-op">)</tt> </tt>
<a name="L680"></a><tt class="py-lineno">680</tt>  <tt class="py-line"> </tt>
<a name="L681"></a><tt class="py-lineno">681</tt>  <tt class="py-line">            <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">"ma4.TAG = 'medline'"</tt> </tt>
<a name="L682"></a><tt class="py-lineno">682</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">medline</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L683"></a><tt class="py-lineno">683</tt>  <tt class="py-line">                <tt class="py-comment"># A list of PubMed IDs</tt> </tt>
<a name="L684"></a><tt class="py-lineno">684</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">" and ma4.VAL in ('"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L685"></a><tt class="py-lineno">685</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">"','"</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">medline</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L686"></a><tt class="py-lineno">686</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">"')"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L687"></a><tt class="py-lineno">687</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L688"></a><tt class="py-lineno">688</tt>  <tt class="py-line">                <tt class="py-comment"># A single PubMed ID</tt> </tt>
<a name="L689"></a><tt class="py-lineno">689</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-string">" and ma4.VAL = '%s' "</tt> <tt class="py-op">%</tt> <tt class="py-name">medline</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L690"></a><tt class="py-lineno">690</tt>  <tt class="py-line"> </tt>
<a name="L691"></a><tt class="py-lineno">691</tt>  <tt class="py-line">            <tt class="py-name">where_clauses</tt><tt class="py-op">.</tt><tt id="link-151" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-151', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">clause</tt><tt class="py-op">)</tt> </tt>
<a name="L692"></a><tt class="py-lineno">692</tt>  <tt class="py-line"> </tt>
<a name="L693"></a><tt class="py-lineno">693</tt>  <tt class="py-line">        <tt class="py-comment"># Select by data type(s) used to compile the matrix</tt> </tt>
<a name="L694"></a><tt class="py-lineno">694</tt>  <tt class="py-line">        <tt class="py-comment"># (MATRIX_ANNOTATION.TAG="type")</tt> </tt>
<a name="L695"></a><tt class="py-lineno">695</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">data_type</tt><tt class="py-op">:</tt> </tt>
<a name="L696"></a><tt class="py-lineno">696</tt>  <tt class="py-line">            <tt class="py-name">tables</tt><tt class="py-op">.</tt><tt id="link-152" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-152', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"MATRIX_ANNOTATION ma5"</tt><tt class="py-op">)</tt> </tt>
<a name="L697"></a><tt class="py-lineno">697</tt>  <tt class="py-line">            <tt class="py-name">where_clauses</tt><tt class="py-op">.</tt><tt id="link-153" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-153', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"m.ID = ma5.ID"</tt><tt class="py-op">)</tt> </tt>
<a name="L698"></a><tt class="py-lineno">698</tt>  <tt class="py-line"> </tt>
<a name="L699"></a><tt class="py-lineno">699</tt>  <tt class="py-line">            <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">"ma5.TAG = 'type'"</tt> </tt>
<a name="L700"></a><tt class="py-lineno">700</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">data_type</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L701"></a><tt class="py-lineno">701</tt>  <tt class="py-line">                <tt class="py-comment"># A list of data types</tt> </tt>
<a name="L702"></a><tt class="py-lineno">702</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">" and ma5.VAL in ('"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L703"></a><tt class="py-lineno">703</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">"','"</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">data_type</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L704"></a><tt class="py-lineno">704</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">"')"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L705"></a><tt class="py-lineno">705</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L706"></a><tt class="py-lineno">706</tt>  <tt class="py-line">                <tt class="py-comment"># A single data type</tt> </tt>
<a name="L707"></a><tt class="py-lineno">707</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-string">" and ma5.VAL = '%s' "</tt> <tt class="py-op">%</tt> <tt class="py-name">data_type</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L708"></a><tt class="py-lineno">708</tt>  <tt class="py-line"> </tt>
<a name="L709"></a><tt class="py-lineno">709</tt>  <tt class="py-line">            <tt class="py-name">where_clauses</tt><tt class="py-op">.</tt><tt id="link-154" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-154', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">clause</tt><tt class="py-op">)</tt> </tt>
<a name="L710"></a><tt class="py-lineno">710</tt>  <tt class="py-line"> </tt>
<a name="L711"></a><tt class="py-lineno">711</tt>  <tt class="py-line">        <tt class="py-comment"># Select by taxonomic supergroup(s) (MATRIX_ANNOTATION.TAG="tax_group")</tt> </tt>
<a name="L712"></a><tt class="py-lineno">712</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">tax_group</tt><tt class="py-op">:</tt> </tt>
<a name="L713"></a><tt class="py-lineno">713</tt>  <tt class="py-line">            <tt class="py-name">tables</tt><tt class="py-op">.</tt><tt id="link-155" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-155', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"MATRIX_ANNOTATION ma6"</tt><tt class="py-op">)</tt> </tt>
<a name="L714"></a><tt class="py-lineno">714</tt>  <tt class="py-line">            <tt class="py-name">where_clauses</tt><tt class="py-op">.</tt><tt id="link-156" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-156', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"m.ID = ma6.ID"</tt><tt class="py-op">)</tt> </tt>
<a name="L715"></a><tt class="py-lineno">715</tt>  <tt class="py-line"> </tt>
<a name="L716"></a><tt class="py-lineno">716</tt>  <tt class="py-line">            <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">"ma6.TAG = 'tax_group'"</tt> </tt>
<a name="L717"></a><tt class="py-lineno">717</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">tax_group</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L718"></a><tt class="py-lineno">718</tt>  <tt class="py-line">                <tt class="py-comment"># A list of tax IDs</tt> </tt>
<a name="L719"></a><tt class="py-lineno">719</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">" and ma6.VAL in ('"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L720"></a><tt class="py-lineno">720</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">"','"</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">tax_group</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L721"></a><tt class="py-lineno">721</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">"')"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L722"></a><tt class="py-lineno">722</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L723"></a><tt class="py-lineno">723</tt>  <tt class="py-line">                <tt class="py-comment"># A single tax ID</tt> </tt>
<a name="L724"></a><tt class="py-lineno">724</tt>  <tt class="py-line">                <tt class="py-name">clause</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">clause</tt><tt class="py-op">,</tt> <tt class="py-string">" and ma6.VAL = '%s' "</tt> <tt class="py-op">%</tt> <tt class="py-name">tax_group</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L725"></a><tt class="py-lineno">725</tt>  <tt class="py-line"> </tt>
<a name="L726"></a><tt class="py-lineno">726</tt>  <tt class="py-line">            <tt class="py-name">where_clauses</tt><tt class="py-op">.</tt><tt id="link-157" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-157', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">clause</tt><tt class="py-op">)</tt> </tt>
<a name="L727"></a><tt class="py-lineno">727</tt>  <tt class="py-line"> </tt>
<a name="L728"></a><tt class="py-lineno">728</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-string">"select distinct(m.ID) from "</tt><tt class="py-op">,</tt> <tt class="py-string">", "</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">tables</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L729"></a><tt class="py-lineno">729</tt>  <tt class="py-line"> </tt>
<a name="L730"></a><tt class="py-lineno">730</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">where_clauses</tt><tt class="py-op">:</tt> </tt>
<a name="L731"></a><tt class="py-lineno">731</tt>  <tt class="py-line">            <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-string">" where "</tt><tt class="py-op">,</tt> <tt class="py-string">" and "</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">where_clauses</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L732"></a><tt class="py-lineno">732</tt>  <tt class="py-line"> </tt>
<a name="L733"></a><tt class="py-lineno">733</tt>  <tt class="py-line">        <tt class="py-comment"># print "sql = %s" % sql</tt> </tt>
<a name="L734"></a><tt class="py-lineno">734</tt>  <tt class="py-line"> </tt>
<a name="L735"></a><tt class="py-lineno">735</tt>  <tt class="py-line">        <tt class="py-name">cur</tt><tt class="py-op">.</tt><tt id="link-158" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-158', 'execute', 'link-50');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">)</tt> </tt>
<a name="L736"></a><tt class="py-lineno">736</tt>  <tt class="py-line">        <tt class="py-name">rows</tt> <tt class="py-op">=</tt> <tt class="py-name">cur</tt><tt class="py-op">.</tt><tt id="link-159" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchall" class="py-name" href="#" onclick="return doclink('link-159', 'fetchall', 'link-80');">fetchall</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L737"></a><tt class="py-lineno">737</tt>  <tt class="py-line"> </tt>
<a name="L738"></a><tt class="py-lineno">738</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">row</tt> <tt class="py-keyword">in</tt> <tt class="py-name">rows</tt><tt class="py-op">:</tt> </tt>
<a name="L739"></a><tt class="py-lineno">739</tt>  <tt class="py-line">            <tt id="link-160" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-160', 'id', 'link-23');">id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L740"></a><tt class="py-lineno">740</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">all_versions</tt><tt class="py-op">:</tt> </tt>
<a name="L741"></a><tt class="py-lineno">741</tt>  <tt class="py-line">                <tt class="py-name">int_ids</tt><tt class="py-op">.</tt><tt id="link-161" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-161', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt id="link-162" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-162', 'id', 'link-23');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L742"></a><tt class="py-lineno">742</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L743"></a><tt class="py-lineno">743</tt>  <tt class="py-line">                <tt class="py-comment"># is the latest version?</tt> </tt>
<a name="L744"></a><tt class="py-lineno">744</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-163" class="py-name" targets="Method Bio.motifs.jaspar.db.JASPAR5._is_latest_version()=Bio.motifs.jaspar.db.JASPAR5-class.html#_is_latest_version"><a title="Bio.motifs.jaspar.db.JASPAR5._is_latest_version" class="py-name" href="#" onclick="return doclink('link-163', '_is_latest_version', 'link-163');">_is_latest_version</a></tt><tt class="py-op">(</tt><tt id="link-164" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-164', 'id', 'link-23');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L745"></a><tt class="py-lineno">745</tt>  <tt class="py-line">                    <tt class="py-name">int_ids</tt><tt class="py-op">.</tt><tt id="link-165" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-165', 'append', 'link-48');">append</a></tt><tt class="py-op">(</tt><tt id="link-166" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-166', 'id', 'link-23');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L746"></a><tt class="py-lineno">746</tt>  <tt class="py-line"> </tt>
<a name="L747"></a><tt class="py-lineno">747</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">int_ids</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L748"></a><tt class="py-lineno">748</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Zero motifs returned with current select critera"</tt><tt class="py-op">,</tt> <tt id="link-167" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-167', 'BiopythonWarning', 'link-1');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L749"></a><tt class="py-lineno">749</tt>  <tt class="py-line"> </tt>
<a name="L750"></a><tt class="py-lineno">750</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">int_ids</tt> </tt>
</div><a name="L751"></a><tt class="py-lineno">751</tt>  <tt class="py-line"> </tt>
<a name="JASPAR5._is_latest_version"></a><div id="JASPAR5._is_latest_version-def"><a name="L752"></a><tt class="py-lineno">752</tt> <a class="py-toggle" href="#" id="JASPAR5._is_latest_version-toggle" onclick="return toggle('JASPAR5._is_latest_version');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.jaspar.db.JASPAR5-class.html#_is_latest_version">_is_latest_version</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">int_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="JASPAR5._is_latest_version-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="JASPAR5._is_latest_version-expanded"><a name="L753"></a><tt class="py-lineno">753</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L754"></a><tt class="py-lineno">754</tt>  <tt class="py-line"><tt class="py-docstring">        Does this internal ID represent the latest version of the JASPAR</tt> </tt>
<a name="L755"></a><tt class="py-lineno">755</tt>  <tt class="py-line"><tt class="py-docstring">        matrix (collapse on base ids)</tt> </tt>
<a name="L756"></a><tt class="py-lineno">756</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L757"></a><tt class="py-lineno">757</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L758"></a><tt class="py-lineno">758</tt>  <tt class="py-line">        <tt class="py-name">cur</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbh</tt><tt class="py-op">.</tt><tt id="link-168" class="py-name"><a title="Bio.DocSQL.Query.cursor
Bio.DocSQL.QuerySingle.cursor" class="py-name" href="#" onclick="return doclink('link-168', 'cursor', 'link-49');">cursor</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L759"></a><tt class="py-lineno">759</tt>  <tt class="py-line"> </tt>
<a name="L760"></a><tt class="py-lineno">760</tt>  <tt class="py-line">        <tt class="py-name">cur</tt><tt class="py-op">.</tt><tt id="link-169" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-169', 'execute', 'link-50');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L761"></a><tt class="py-lineno">761</tt>  <tt class="py-line">            <tt class="py-string">"""select count(*) from MATRIX</tt> </tt>
<a name="L762"></a><tt class="py-lineno">762</tt>  <tt class="py-line"><tt class="py-string">               where BASE_ID = (select BASE_ID from MATRIX where ID = %s)</tt> </tt>
<a name="L763"></a><tt class="py-lineno">763</tt>  <tt class="py-line"><tt class="py-string">               and VERSION &gt; (select VERSION from MATRIX where ID = %s)"""</tt><tt class="py-op">,</tt> </tt>
<a name="L764"></a><tt class="py-lineno">764</tt>  <tt class="py-line">               <tt class="py-op">(</tt><tt class="py-name">int_id</tt><tt class="py-op">,</tt> <tt class="py-name">int_id</tt><tt class="py-op">)</tt> </tt>
<a name="L765"></a><tt class="py-lineno">765</tt>  <tt class="py-line">        <tt class="py-op">)</tt> </tt>
<a name="L766"></a><tt class="py-lineno">766</tt>  <tt class="py-line"> </tt>
<a name="L767"></a><tt class="py-lineno">767</tt>  <tt class="py-line">        <tt class="py-name">row</tt> <tt class="py-op">=</tt> <tt class="py-name">cur</tt><tt class="py-op">.</tt><tt id="link-170" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchone" class="py-name" href="#" onclick="return doclink('link-170', 'fetchone', 'link-52');">fetchone</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L768"></a><tt class="py-lineno">768</tt>  <tt class="py-line"> </tt>
<a name="L769"></a><tt class="py-lineno">769</tt>  <tt class="py-line">        <tt id="link-171" class="py-name" targets="Method Bio.Crystal.Chain.count()=Bio.Crystal.Chain-class.html#count,Method Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count()=Bio.NeuralNetwork.Gene.Pattern.PatternRepository-class.html#count,Method Bio.Seq.MutableSeq.count()=Bio.Seq.MutableSeq-class.html#count,Method Bio.Seq.Seq.count()=Bio.Seq.Seq-class.html#count,Method Bio.Seq.UnknownSeq.count()=Bio.Seq.UnknownSeq-class.html#count,Method Bio.motifs.Instances.count()=Bio.motifs.Instances-class.html#count"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-171', 'count', 'link-171');">count</a></tt> <tt class="py-op">=</tt> <tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L770"></a><tt class="py-lineno">770</tt>  <tt class="py-line"> </tt>
<a name="L771"></a><tt class="py-lineno">771</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-172" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-172', 'count', 'link-171');">count</a></tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L772"></a><tt class="py-lineno">772</tt>  <tt class="py-line">            <tt class="py-comment"># no matrices with higher version ID and same base id</tt> </tt>
<a name="L773"></a><tt class="py-lineno">773</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">True</tt> </tt>
<a name="L774"></a><tt class="py-lineno">774</tt>  <tt class="py-line"> </tt>
<a name="L775"></a><tt class="py-lineno">775</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
</div></div><a name="L776"></a><tt class="py-lineno">776</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:04:41 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
