<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.Align.AlignInfo</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.Align-module.html">Package&nbsp;Align</a> ::
        Module&nbsp;AlignInfo
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.Align.AlignInfo-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.Align.AlignInfo-module.html">Module Bio.Align.AlignInfo</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-docstring">"""Extract information from alignment objects.</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-docstring">In order to try and avoid huge alignment objects with tons of functions,</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-docstring">functions which return summary type information about alignments should</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-docstring">be put into classes in this module.</tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">__future__</tt> <tt class="py-keyword">import</tt> <tt class="py-name">print_function</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">math</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">sys</tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-1" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-1', 'Alphabet', 'link-1');">Alphabet</a></tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-2" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-2', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-3" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-3', 'Alphabet', 'link-1');">Alphabet</a></tt> <tt class="py-keyword">import</tt> <tt id="link-4" class="py-name" targets="Module Bio.Alphabet.IUPAC=Bio.Alphabet.IUPAC-module.html"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-4', 'IUPAC', 'link-4');">IUPAC</a></tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-5" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-5', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-6" class="py-name" targets="Module Bio.Seq=Bio.Seq-module.html,Class Bio.Seq.Seq=Bio.Seq.Seq-class.html"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-6', 'Seq', 'link-6');">Seq</a></tt> <tt class="py-keyword">import</tt> <tt id="link-7" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-7', 'Seq', 'link-6');">Seq</a></tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-8" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-8', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-9" class="py-name" targets="Package Bio.SubsMat=Bio.SubsMat-module.html"><a title="Bio.SubsMat" class="py-name" href="#" onclick="return doclink('link-9', 'SubsMat', 'link-9');">SubsMat</a></tt> <tt class="py-keyword">import</tt> <tt id="link-10" class="py-name" targets="Module Bio.SubsMat.FreqTable=Bio.SubsMat.FreqTable-module.html,Class Bio.SubsMat.FreqTable.FreqTable=Bio.SubsMat.FreqTable.FreqTable-class.html"><a title="Bio.SubsMat.FreqTable
Bio.SubsMat.FreqTable.FreqTable" class="py-name" href="#" onclick="return doclink('link-10', 'FreqTable', 'link-10');">FreqTable</a></tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-name">__docformat__</tt> <tt class="py-op">=</tt> <tt class="py-string">"restructuredtext en"</tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-comment"># Expected random distributions for 20-letter protein, and</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-comment"># for 4-letter nucleotide alphabets</tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"><tt id="link-11" class="py-name" targets="Variable Bio.Align.AlignInfo.Protein20Random=Bio.Align.AlignInfo-module.html#Protein20Random"><a title="Bio.Align.AlignInfo.Protein20Random" class="py-name" href="#" onclick="return doclink('link-11', 'Protein20Random', 'link-11');">Protein20Random</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0.05</tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt id="link-12" class="py-name" targets="Variable Bio.Align.AlignInfo.Nucleotide4Random=Bio.Align.AlignInfo-module.html#Nucleotide4Random"><a title="Bio.Align.AlignInfo.Nucleotide4Random" class="py-name" href="#" onclick="return doclink('link-12', 'Nucleotide4Random', 'link-12');">Nucleotide4Random</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0.25</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"> </tt>
<a name="SummaryInfo"></a><div id="SummaryInfo-def"><a name="L31"></a><tt class="py-lineno"> 31</tt> <a class="py-toggle" href="#" id="SummaryInfo-toggle" onclick="return toggle('SummaryInfo');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Align.AlignInfo.SummaryInfo-class.html">SummaryInfo</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SummaryInfo-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="SummaryInfo-expanded"><a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line">    <tt class="py-docstring">"""Calculate summary info about the alignment.</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-docstring">    This class should be used to caclculate information summarizing the</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-docstring">    results of an alignment. This may either be straight consensus info</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"><tt class="py-docstring">    or more complicated things.</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="SummaryInfo.__init__"></a><div id="SummaryInfo.__init__-def"><a name="L38"></a><tt class="py-lineno"> 38</tt> <a class="py-toggle" href="#" id="SummaryInfo.__init__-toggle" onclick="return toggle('SummaryInfo.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Align.AlignInfo.SummaryInfo-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">alignment</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SummaryInfo.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SummaryInfo.__init__-expanded"><a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize with the alignment to calculate information on.</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"><tt class="py-docstring">        ic_vector attribute. A dictionary. Keys: column numbers. Values:</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt> <tt class="py-op">=</tt> <tt class="py-name">alignment</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ic_vector</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
</div><a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line"> </tt>
<a name="SummaryInfo.dumb_consensus"></a><div id="SummaryInfo.dumb_consensus-def"><a name="L46"></a><tt class="py-lineno"> 46</tt> <a class="py-toggle" href="#" id="SummaryInfo.dumb_consensus-toggle" onclick="return toggle('SummaryInfo.dumb_consensus');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Align.AlignInfo.SummaryInfo-class.html#dumb_consensus">dumb_consensus</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">threshold</tt><tt class="py-op">=</tt><tt class="py-number">.7</tt><tt class="py-op">,</tt> <tt class="py-param">ambiguous</tt><tt class="py-op">=</tt><tt class="py-string">"X"</tt><tt class="py-op">,</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line">                       <tt class="py-param">consensus_alpha</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">require_multiple</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SummaryInfo.dumb_consensus-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SummaryInfo.dumb_consensus-expanded"><a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line">        <tt class="py-docstring">"""Output a fast consensus sequence of the alignment.</tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line"><tt class="py-docstring">        This doesn't do anything fancy at all. It will just go through the</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line"><tt class="py-docstring">        sequence residue by residue and count up the number of each type</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line"><tt class="py-docstring">        of residue (ie. A or G or T or C for DNA) in all sequences in the</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line"><tt class="py-docstring">        alignment. If the percentage of the most common residue type is</tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line"><tt class="py-docstring">        greater then the passed threshold, then we will add that residue type,</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line"><tt class="py-docstring">        otherwise an ambiguous character will be added.</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"><tt class="py-docstring">        This could be made a lot fancier (ie. to take a substitution matrix</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line"><tt class="py-docstring">        into account), but it just meant for a quick and dirty consensus.</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line"><tt class="py-docstring">            - threshold - The threshold value that is required to add a particular</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line"><tt class="py-docstring">              atom.</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line"><tt class="py-docstring">            - ambiguous - The ambiguous character to be added when the threshold is</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line"><tt class="py-docstring">              not reached.</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line"><tt class="py-docstring">            - consensus_alpha - The alphabet to return for the consensus sequence.</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line"><tt class="py-docstring">              If this is None, then we will try to guess the alphabet.</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line"><tt class="py-docstring">            - require_multiple - If set as 1, this will require that more than</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line"><tt class="py-docstring">              1 sequence be part of an alignment to put it in the consensus (ie.</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line"><tt class="py-docstring">              not just 1 sequence and gaps).</tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line">        <tt class="py-comment"># Iddo Friedberg, 1-JUL-2004: changed ambiguous default to "X"</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line">        <tt id="link-13" class="py-name" targets="Method Bio.Motif._Motif.Motif.consensus()=Bio.Motif._Motif.Motif-class.html#consensus,Function Bio.Nexus.Trees.consensus()=Bio.Nexus.Trees-module.html#consensus,Variable Bio.motifs.Motif.consensus=Bio.motifs.Motif-class.html#consensus,Variable Bio.motifs.matrix.GenericPositionMatrix.consensus=Bio.motifs.matrix.GenericPositionMatrix-class.html#consensus"><a title="Bio.Motif._Motif.Motif.consensus
Bio.Nexus.Trees.consensus
Bio.motifs.Motif.consensus
Bio.motifs.matrix.GenericPositionMatrix.consensus" class="py-name" href="#" onclick="return doclink('link-13', 'consensus', 'link-13');">consensus</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line"> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">        <tt class="py-comment"># find the length of the consensus we are creating</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line">        <tt class="py-name">con_len</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">.</tt><tt id="link-14" class="py-name" targets="Method Bio.Align.Generic.Alignment.get_alignment_length()=Bio.Align.Generic.Alignment-class.html#get_alignment_length"><a title="Bio.Align.Generic.Alignment.get_alignment_length" class="py-name" href="#" onclick="return doclink('link-14', 'get_alignment_length', 'link-14');">get_alignment_length</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line"> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line">        <tt class="py-comment"># go through each seq item</tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">n</tt> <tt class="py-keyword">in</tt> <tt id="link-15" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-15', 'range', 'link-15');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">con_len</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line">            <tt class="py-comment"># keep track of the counts of the different atoms we get</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line">            <tt class="py-name">atom_dict</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">            <tt class="py-name">num_atoms</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line"> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">:</tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line">                <tt class="py-comment"># make sure we haven't run past the end of any sequences</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line">                <tt class="py-comment"># if they are of different lengths</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">n</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-16" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.seq=Bio.SeqRecord.SeqRecord-class.html#seq,Variable BioSQL.BioSeq.DBSeqRecord.seq=BioSQL.BioSeq.DBSeqRecord-class.html#seq"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-16', 'seq', 'link-16');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-17" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-17', 'seq', 'link-16');">seq</a></tt><tt class="py-op">[</tt><tt class="py-name">n</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-string">'-'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-18" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-18', 'seq', 'link-16');">seq</a></tt><tt class="py-op">[</tt><tt class="py-name">n</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-string">'.'</tt><tt class="py-op">:</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-19" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-19', 'seq', 'link-16');">seq</a></tt><tt class="py-op">[</tt><tt class="py-name">n</tt><tt class="py-op">]</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">atom_dict</tt><tt class="py-op">:</tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line">                            <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-20" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-20', 'seq', 'link-16');">seq</a></tt><tt class="py-op">[</tt><tt class="py-name">n</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line">                        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">                            <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-21" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-21', 'seq', 'link-16');">seq</a></tt><tt class="py-op">[</tt><tt class="py-name">n</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line"> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">                        <tt class="py-name">num_atoms</tt> <tt class="py-op">=</tt> <tt class="py-name">num_atoms</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line"> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line">            <tt class="py-name">max_atoms</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line">            <tt class="py-name">max_size</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line"> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">atom</tt> <tt class="py-keyword">in</tt> <tt class="py-name">atom_dict</tt><tt class="py-op">:</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-name">atom</tt><tt class="py-op">]</tt> <tt class="py-op">&gt;</tt> <tt class="py-name">max_size</tt><tt class="py-op">:</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line">                    <tt class="py-name">max_atoms</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">atom</tt><tt class="py-op">]</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line">                    <tt class="py-name">max_size</tt> <tt class="py-op">=</tt> <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-name">atom</tt><tt class="py-op">]</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-name">atom</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">max_size</tt><tt class="py-op">:</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">                    <tt class="py-name">max_atoms</tt><tt class="py-op">.</tt><tt id="link-22" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-22', 'append', 'link-22');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">atom</tt><tt class="py-op">)</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line"> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">require_multiple</tt> <tt class="py-keyword">and</tt> <tt class="py-name">num_atoms</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">                <tt id="link-23" class="py-name"><a title="Bio.Motif._Motif.Motif.consensus
Bio.Nexus.Trees.consensus
Bio.motifs.Motif.consensus
Bio.motifs.matrix.GenericPositionMatrix.consensus" class="py-name" href="#" onclick="return doclink('link-23', 'consensus', 'link-13');">consensus</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">ambiguous</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">max_atoms</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">max_size</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">num_atoms</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line">                                            <tt class="py-op">&gt;=</tt> <tt id="link-24" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.threshold()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#threshold"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.threshold" class="py-name" href="#" onclick="return doclink('link-24', 'threshold', 'link-24');">threshold</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">                <tt id="link-25" class="py-name"><a title="Bio.Motif._Motif.Motif.consensus
Bio.Nexus.Trees.consensus
Bio.motifs.Motif.consensus
Bio.motifs.matrix.GenericPositionMatrix.consensus" class="py-name" href="#" onclick="return doclink('link-25', 'consensus', 'link-13');">consensus</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">max_atoms</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">                <tt id="link-26" class="py-name"><a title="Bio.Motif._Motif.Motif.consensus
Bio.Nexus.Trees.consensus
Bio.motifs.Motif.consensus
Bio.motifs.matrix.GenericPositionMatrix.consensus" class="py-name" href="#" onclick="return doclink('link-26', 'consensus', 'link-13');">consensus</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">ambiguous</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line"> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">        <tt class="py-comment"># we need to guess a consensus alphabet if one isn't specified</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">consensus_alpha</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">            <tt class="py-name">consensus_alpha</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-27" class="py-name" targets="Method Bio.Align.AlignInfo.SummaryInfo._guess_consensus_alphabet()=Bio.Align.AlignInfo.SummaryInfo-class.html#_guess_consensus_alphabet"><a title="Bio.Align.AlignInfo.SummaryInfo._guess_consensus_alphabet" class="py-name" href="#" onclick="return doclink('link-27', '_guess_consensus_alphabet', 'link-27');">_guess_consensus_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">ambiguous</tt><tt class="py-op">)</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line"> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-28" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-28', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt id="link-29" class="py-name"><a title="Bio.Motif._Motif.Motif.consensus
Bio.Nexus.Trees.consensus
Bio.motifs.Motif.consensus
Bio.motifs.matrix.GenericPositionMatrix.consensus" class="py-name" href="#" onclick="return doclink('link-29', 'consensus', 'link-13');">consensus</a></tt><tt class="py-op">,</tt> <tt class="py-name">consensus_alpha</tt><tt class="py-op">)</tt> </tt>
</div><a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line"> </tt>
<a name="SummaryInfo.gap_consensus"></a><div id="SummaryInfo.gap_consensus-def"><a name="L119"></a><tt class="py-lineno">119</tt> <a class="py-toggle" href="#" id="SummaryInfo.gap_consensus-toggle" onclick="return toggle('SummaryInfo.gap_consensus');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Align.AlignInfo.SummaryInfo-class.html#gap_consensus">gap_consensus</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">threshold</tt><tt class="py-op">=</tt><tt class="py-number">.7</tt><tt class="py-op">,</tt> <tt class="py-param">ambiguous</tt><tt class="py-op">=</tt><tt class="py-string">"X"</tt><tt class="py-op">,</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">                       <tt class="py-param">consensus_alpha</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">require_multiple</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SummaryInfo.gap_consensus-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SummaryInfo.gap_consensus-expanded"><a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">        <tt class="py-docstring">"""Same as dumb_consensus(), but allows gap on the output.</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line"><tt class="py-docstring">        Things to do:</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line"><tt class="py-docstring">            - Let the user define that with only one gap, the result</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line"><tt class="py-docstring">              character in consensus is gap.</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line"><tt class="py-docstring">            - Let the user select gap character, now</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line"><tt class="py-docstring">              it takes the same as input.</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line">        <tt class="py-comment"># Iddo Friedberg, 1-JUL-2004: changed ambiguous default to "X"</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line">        <tt id="link-30" class="py-name"><a title="Bio.Motif._Motif.Motif.consensus
Bio.Nexus.Trees.consensus
Bio.motifs.Motif.consensus
Bio.motifs.matrix.GenericPositionMatrix.consensus" class="py-name" href="#" onclick="return doclink('link-30', 'consensus', 'link-13');">consensus</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line"> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line">        <tt class="py-comment"># find the length of the consensus we are creating</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line">        <tt class="py-name">con_len</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">.</tt><tt id="link-31" class="py-name"><a title="Bio.Align.Generic.Alignment.get_alignment_length" class="py-name" href="#" onclick="return doclink('link-31', 'get_alignment_length', 'link-14');">get_alignment_length</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line"> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line">        <tt class="py-comment"># go through each seq item</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">n</tt> <tt class="py-keyword">in</tt> <tt id="link-32" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-32', 'range', 'link-15');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">con_len</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line">            <tt class="py-comment"># keep track of the counts of the different atoms we get</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line">            <tt class="py-name">atom_dict</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line">            <tt class="py-name">num_atoms</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line"> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">:</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line">                <tt class="py-comment"># make sure we haven't run past the end of any sequences</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line">                <tt class="py-comment"># if they are of different lengths</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">n</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-33" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-33', 'seq', 'link-16');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-34" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-34', 'seq', 'link-16');">seq</a></tt><tt class="py-op">[</tt><tt class="py-name">n</tt><tt class="py-op">]</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">atom_dict</tt><tt class="py-op">:</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">                        <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-35" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-35', 'seq', 'link-16');">seq</a></tt><tt class="py-op">[</tt><tt class="py-name">n</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">                        <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-36" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-36', 'seq', 'link-16');">seq</a></tt><tt class="py-op">[</tt><tt class="py-name">n</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line"> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">                    <tt class="py-name">num_atoms</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line"> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">            <tt class="py-name">max_atoms</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">            <tt class="py-name">max_size</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line"> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">atom</tt> <tt class="py-keyword">in</tt> <tt class="py-name">atom_dict</tt><tt class="py-op">:</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-name">atom</tt><tt class="py-op">]</tt> <tt class="py-op">&gt;</tt> <tt class="py-name">max_size</tt><tt class="py-op">:</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">                    <tt class="py-name">max_atoms</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">atom</tt><tt class="py-op">]</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">                    <tt class="py-name">max_size</tt> <tt class="py-op">=</tt> <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-name">atom</tt><tt class="py-op">]</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-name">atom</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">max_size</tt><tt class="py-op">:</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">                    <tt class="py-name">max_atoms</tt><tt class="py-op">.</tt><tt id="link-37" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-37', 'append', 'link-22');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">atom</tt><tt class="py-op">)</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line"> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">require_multiple</tt> <tt class="py-keyword">and</tt> <tt class="py-name">num_atoms</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">                <tt id="link-38" class="py-name"><a title="Bio.Motif._Motif.Motif.consensus
Bio.Nexus.Trees.consensus
Bio.motifs.Motif.consensus
Bio.motifs.matrix.GenericPositionMatrix.consensus" class="py-name" href="#" onclick="return doclink('link-38', 'consensus', 'link-13');">consensus</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">ambiguous</tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">max_atoms</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">max_size</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">num_atoms</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">                                            <tt class="py-op">&gt;=</tt> <tt id="link-39" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.threshold" class="py-name" href="#" onclick="return doclink('link-39', 'threshold', 'link-24');">threshold</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">                <tt id="link-40" class="py-name"><a title="Bio.Motif._Motif.Motif.consensus
Bio.Nexus.Trees.consensus
Bio.motifs.Motif.consensus
Bio.motifs.matrix.GenericPositionMatrix.consensus" class="py-name" href="#" onclick="return doclink('link-40', 'consensus', 'link-13');">consensus</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">max_atoms</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">                <tt id="link-41" class="py-name"><a title="Bio.Motif._Motif.Motif.consensus
Bio.Nexus.Trees.consensus
Bio.motifs.Motif.consensus
Bio.motifs.matrix.GenericPositionMatrix.consensus" class="py-name" href="#" onclick="return doclink('link-41', 'consensus', 'link-13');">consensus</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">ambiguous</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line"> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line">        <tt class="py-comment"># we need to guess a consensus alphabet if one isn't specified</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">consensus_alpha</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line">            <tt class="py-comment"># TODO - Should we make this into a Gapped alphabet?</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line">            <tt class="py-name">consensus_alpha</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-42" class="py-name"><a title="Bio.Align.AlignInfo.SummaryInfo._guess_consensus_alphabet" class="py-name" href="#" onclick="return doclink('link-42', '_guess_consensus_alphabet', 'link-27');">_guess_consensus_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">ambiguous</tt><tt class="py-op">)</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line"> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-43" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-43', 'Seq', 'link-6');">Seq</a></tt><tt class="py-op">(</tt><tt id="link-44" class="py-name"><a title="Bio.Motif._Motif.Motif.consensus
Bio.Nexus.Trees.consensus
Bio.motifs.Motif.consensus
Bio.motifs.matrix.GenericPositionMatrix.consensus" class="py-name" href="#" onclick="return doclink('link-44', 'consensus', 'link-13');">consensus</a></tt><tt class="py-op">,</tt> <tt class="py-name">consensus_alpha</tt><tt class="py-op">)</tt> </tt>
</div><a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line"> </tt>
<a name="SummaryInfo._guess_consensus_alphabet"></a><div id="SummaryInfo._guess_consensus_alphabet-def"><a name="L177"></a><tt class="py-lineno">177</tt> <a class="py-toggle" href="#" id="SummaryInfo._guess_consensus_alphabet-toggle" onclick="return toggle('SummaryInfo._guess_consensus_alphabet');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Align.AlignInfo.SummaryInfo-class.html#_guess_consensus_alphabet">_guess_consensus_alphabet</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">ambiguous</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SummaryInfo._guess_consensus_alphabet-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SummaryInfo._guess_consensus_alphabet-expanded"><a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">        <tt class="py-docstring">"""Pick an (ungapped) alphabet for an alignment consesus sequence.</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line"><tt class="py-docstring">        This just looks at the sequences we have, checks their type, and</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line"><tt class="py-docstring">        returns as appropriate type which seems to make sense with the</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line"><tt class="py-docstring">        sequences we've got.</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">        <tt class="py-comment"># Start with the (un-gapped version of) the alignment alphabet</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">        <tt id="link-45" class="py-name" targets="Variable Bio.Graphics.ColorSpiral.ColorSpiral.a=Bio.Graphics.ColorSpiral.ColorSpiral-class.html#a"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-45', 'a', 'link-45');">a</a></tt> <tt class="py-op">=</tt> <tt id="link-46" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-46', 'Alphabet', 'link-1');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-47" class="py-name" targets="Function Bio.Alphabet._get_base_alphabet()=Bio.Alphabet-module.html#_get_base_alphabet"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-47', '_get_base_alphabet', 'link-47');">_get_base_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">.</tt><tt class="py-name">_alphabet</tt><tt class="py-op">)</tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line"> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">        <tt class="py-comment"># Now check its compatible with all the rest of the sequences</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">:</tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">            <tt class="py-comment"># Get the (un-gapped version of) the sequence's alphabet</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">            <tt id="link-48" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.alt()=Bio.Phylo.PhyloXMLIO.Writer-class.html#alt"><a title="Bio.Phylo.PhyloXMLIO.Writer.alt" class="py-name" href="#" onclick="return doclink('link-48', 'alt', 'link-48');">alt</a></tt> <tt class="py-op">=</tt> <tt id="link-49" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-49', 'Alphabet', 'link-1');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-50" class="py-name"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-50', '_get_base_alphabet', 'link-47');">_get_base_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-51" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-51', 'seq', 'link-16');">seq</a></tt><tt class="py-op">.</tt><tt id="link-52" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.alphabet=Bio.SearchIO._model.hsp.HSP-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSPFragment.alphabet=Bio.SearchIO._model.hsp.HSPFragment-class.html#alphabet"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-52', 'alphabet', 'link-52');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-53" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.alt" class="py-name" href="#" onclick="return doclink('link-53', 'alt', 'link-48');">alt</a></tt><tt class="py-op">,</tt> <tt id="link-54" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-54', 'a', 'link-45');">a</a></tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Alignment contains a sequence with \</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line"><tt class="py-string">                                an incompatible alphabet."</tt><tt class="py-op">)</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line"> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">        <tt class="py-comment"># Check the ambiguous character we are going to use in the consensus</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">        <tt class="py-comment"># is in the alphabet's list of valid letters (if defined).</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-55" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-55', 'a', 'link-45');">a</a></tt><tt class="py-op">,</tt> <tt class="py-string">"letters"</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt id="link-56" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-56', 'a', 'link-45');">a</a></tt><tt class="py-op">.</tt><tt id="link-57" class="py-name" targets="Variable Bio.Alphabet.Alphabet.letters=Bio.Alphabet.Alphabet-class.html#letters,Variable Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters=Bio.Alphabet.IUPAC.ExtendedIUPACDNA-class.html#letters,Variable Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters=Bio.Alphabet.IUPAC.ExtendedIUPACProtein-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters=Bio.Alphabet.IUPAC.IUPACAmbiguousDNA-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters=Bio.Alphabet.IUPAC.IUPACAmbiguousRNA-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACProtein.letters=Bio.Alphabet.IUPAC.IUPACProtein-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters=Bio.Alphabet.IUPAC.IUPACUnambiguousDNA-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters=Bio.Alphabet.IUPAC.IUPACUnambiguousRNA-class.html#letters,Variable Bio.Alphabet.Reduced.HPModel.letters=Bio.Alphabet.Reduced.HPModel-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy10.letters=Bio.Alphabet.Reduced.Murphy10-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy15.letters=Bio.Alphabet.Reduced.Murphy15-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy4.letters=Bio.Alphabet.Reduced.Murphy4-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy8.letters=Bio.Alphabet.Reduced.Murphy8-class.html#letters,Variable Bio.Alphabet.Reduced.PC5.letters=Bio.Alphabet.Reduced.PC5-class.html#letters,Variable Bio.Alphabet.SecondaryStructure.letters=Bio.Alphabet.SecondaryStructure-class.html#letters,Variable Bio.Alphabet.SingleLetterAlphabet.letters=Bio.Alphabet.SingleLetterAlphabet-class.html#letters,Variable Bio.Alphabet.ThreeLetterProtein.letters=Bio.Alphabet.ThreeLetterProtein-class.html#letters,Variable Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.letters=Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet-class.html#letters,Variable Bio.codonalign.codonalphabet.CodonAlphabet.letters=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#letters"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-57', 'letters', 'link-57');">letters</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> \ </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line">        <tt class="py-keyword">and</tt> <tt class="py-name">ambiguous</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt id="link-58" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-58', 'a', 'link-45');">a</a></tt><tt class="py-op">.</tt><tt id="link-59" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-59', 'letters', 'link-57');">letters</a></tt><tt class="py-op">:</tt> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line">            <tt class="py-comment"># We'll need to pick a more generic alphabet...</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-60" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-60', 'a', 'link-45');">a</a></tt><tt class="py-op">,</tt> <tt id="link-61" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-61', 'IUPAC', 'link-4');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-62" class="py-name" targets="Class Bio.Alphabet.IUPAC.IUPACUnambiguousDNA=Bio.Alphabet.IUPAC.IUPACUnambiguousDNA-class.html"><a title="Bio.Alphabet.IUPAC.IUPACUnambiguousDNA" class="py-name" href="#" onclick="return doclink('link-62', 'IUPACUnambiguousDNA', 'link-62');">IUPACUnambiguousDNA</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">ambiguous</tt> <tt class="py-keyword">in</tt> <tt id="link-63" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-63', 'IUPAC', 'link-4');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-64" class="py-name"><a title="Bio.Alphabet.IUPAC.IUPACUnambiguousDNA" class="py-name" href="#" onclick="return doclink('link-64', 'IUPACUnambiguousDNA', 'link-62');">IUPACUnambiguousDNA</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-65" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-65', 'letters', 'link-57');">letters</a></tt><tt class="py-op">:</tt> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line">                    <tt id="link-66" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-66', 'a', 'link-45');">a</a></tt> <tt class="py-op">=</tt> <tt id="link-67" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-67', 'IUPAC', 'link-4');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-68" class="py-name"><a title="Bio.Alphabet.IUPAC.IUPACUnambiguousDNA" class="py-name" href="#" onclick="return doclink('link-68', 'IUPACUnambiguousDNA', 'link-62');">IUPACUnambiguousDNA</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line">                    <tt id="link-69" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-69', 'a', 'link-45');">a</a></tt> <tt class="py-op">=</tt> <tt id="link-70" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-70', 'Alphabet', 'link-1');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-71" class="py-name" targets="Variable Bio.Alphabet.generic_dna=Bio.Alphabet-module.html#generic_dna"><a title="Bio.Alphabet.generic_dna" class="py-name" href="#" onclick="return doclink('link-71', 'generic_dna', 'link-71');">generic_dna</a></tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-72" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-72', 'a', 'link-45');">a</a></tt><tt class="py-op">,</tt> <tt id="link-73" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-73', 'IUPAC', 'link-4');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-74" class="py-name" targets="Class Bio.Alphabet.IUPAC.IUPACUnambiguousRNA=Bio.Alphabet.IUPAC.IUPACUnambiguousRNA-class.html"><a title="Bio.Alphabet.IUPAC.IUPACUnambiguousRNA" class="py-name" href="#" onclick="return doclink('link-74', 'IUPACUnambiguousRNA', 'link-74');">IUPACUnambiguousRNA</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">ambiguous</tt> <tt class="py-keyword">in</tt> <tt id="link-75" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-75', 'IUPAC', 'link-4');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-76" class="py-name"><a title="Bio.Alphabet.IUPAC.IUPACUnambiguousRNA" class="py-name" href="#" onclick="return doclink('link-76', 'IUPACUnambiguousRNA', 'link-74');">IUPACUnambiguousRNA</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-77" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-77', 'letters', 'link-57');">letters</a></tt><tt class="py-op">:</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line">                    <tt id="link-78" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-78', 'a', 'link-45');">a</a></tt> <tt class="py-op">=</tt> <tt id="link-79" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-79', 'IUPAC', 'link-4');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-80" class="py-name"><a title="Bio.Alphabet.IUPAC.IUPACUnambiguousRNA" class="py-name" href="#" onclick="return doclink('link-80', 'IUPACUnambiguousRNA', 'link-74');">IUPACUnambiguousRNA</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line">                    <tt id="link-81" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-81', 'a', 'link-45');">a</a></tt> <tt class="py-op">=</tt> <tt id="link-82" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-82', 'Alphabet', 'link-1');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-83" class="py-name" targets="Variable Bio.Alphabet.generic_rna=Bio.Alphabet-module.html#generic_rna"><a title="Bio.Alphabet.generic_rna" class="py-name" href="#" onclick="return doclink('link-83', 'generic_rna', 'link-83');">generic_rna</a></tt> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-84" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-84', 'a', 'link-45');">a</a></tt><tt class="py-op">,</tt> <tt id="link-85" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-85', 'IUPAC', 'link-4');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-86" class="py-name" targets="Class Bio.Alphabet.IUPAC.IUPACProtein=Bio.Alphabet.IUPAC.IUPACProtein-class.html"><a title="Bio.Alphabet.IUPAC.IUPACProtein" class="py-name" href="#" onclick="return doclink('link-86', 'IUPACProtein', 'link-86');">IUPACProtein</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">ambiguous</tt> <tt class="py-keyword">in</tt> <tt id="link-87" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-87', 'IUPAC', 'link-4');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-88" class="py-name" targets="Class Bio.Alphabet.IUPAC.ExtendedIUPACProtein=Bio.Alphabet.IUPAC.ExtendedIUPACProtein-class.html"><a title="Bio.Alphabet.IUPAC.ExtendedIUPACProtein" class="py-name" href="#" onclick="return doclink('link-88', 'ExtendedIUPACProtein', 'link-88');">ExtendedIUPACProtein</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-89" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-89', 'letters', 'link-57');">letters</a></tt><tt class="py-op">:</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line">                    <tt id="link-90" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-90', 'a', 'link-45');">a</a></tt> <tt class="py-op">=</tt> <tt id="link-91" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-91', 'IUPAC', 'link-4');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-92" class="py-name"><a title="Bio.Alphabet.IUPAC.ExtendedIUPACProtein" class="py-name" href="#" onclick="return doclink('link-92', 'ExtendedIUPACProtein', 'link-88');">ExtendedIUPACProtein</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line">                    <tt id="link-93" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-93', 'a', 'link-45');">a</a></tt> <tt class="py-op">=</tt> <tt id="link-94" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-94', 'Alphabet', 'link-1');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-95" class="py-name" targets="Variable Bio.Alphabet.generic_protein=Bio.Alphabet-module.html#generic_protein"><a title="Bio.Alphabet.generic_protein" class="py-name" href="#" onclick="return doclink('link-95', 'generic_protein', 'link-95');">generic_protein</a></tt> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line">                <tt id="link-96" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-96', 'a', 'link-45');">a</a></tt> <tt class="py-op">=</tt> <tt id="link-97" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-97', 'Alphabet', 'link-1');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-98" class="py-name" targets="Variable Bio.Alphabet.single_letter_alphabet=Bio.Alphabet-module.html#single_letter_alphabet"><a title="Bio.Alphabet.single_letter_alphabet" class="py-name" href="#" onclick="return doclink('link-98', 'single_letter_alphabet', 'link-98');">single_letter_alphabet</a></tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-99" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-99', 'a', 'link-45');">a</a></tt> </tt>
</div><a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line"> </tt>
<a name="SummaryInfo.replacement_dictionary"></a><div id="SummaryInfo.replacement_dictionary-def"><a name="L219"></a><tt class="py-lineno">219</tt> <a class="py-toggle" href="#" id="SummaryInfo.replacement_dictionary-toggle" onclick="return toggle('SummaryInfo.replacement_dictionary');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Align.AlignInfo.SummaryInfo-class.html#replacement_dictionary">replacement_dictionary</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">skip_chars</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SummaryInfo.replacement_dictionary-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SummaryInfo.replacement_dictionary-expanded"><a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">        <tt class="py-docstring">"""Generate a replacement dictionary to plug into a substitution matrix</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line"><tt class="py-docstring">        This should look at an alignment, and be able to generate the number</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line"><tt class="py-docstring">        of substitutions of different residues for each other in the</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line"><tt class="py-docstring">        aligned object.</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line"><tt class="py-docstring">        Will then return a dictionary with this information::</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line"><tt class="py-docstring">            {('A', 'C') : 10, ('C', 'A') : 12, ('G', 'C') : 15 ....}</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line"><tt class="py-docstring">        This also treats weighted sequences. The following example shows how</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line"><tt class="py-docstring">        we calculate the replacement dictionary. Given the following</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line"><tt class="py-docstring">        multiple sequence alignment::</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line"><tt class="py-docstring">            GTATC  0.5</tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line"><tt class="py-docstring">            AT--C  0.8</tt> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line"><tt class="py-docstring">            CTGTC  1.0</tt> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line"><tt class="py-docstring">        For the first column we have::</tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line"><tt class="py-docstring">            ('A', 'G') : 0.5 * 0.8 = 0.4</tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line"><tt class="py-docstring">            ('C', 'G') : 0.5 * 1.0 = 0.5</tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line"><tt class="py-docstring">            ('A', 'C') : 0.8 * 1.0 = 0.8</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line"><tt class="py-docstring">        We then continue this for all of the columns in the alignment, summing</tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line"><tt class="py-docstring">        the information for each substitution in each column, until we end</tt> </tt>
<a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line"><tt class="py-docstring">        up with the replacement dictionary.</tt> </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line"><tt class="py-docstring">         - skip_chars - A list of characters to skip when creating the dictionary.</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line"><tt class="py-docstring">           This defaults to an empty list.</tt> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line"><tt class="py-docstring">        For instance, you might have Xs (screened stuff) or Ns, and not want</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line"><tt class="py-docstring">        to include the ambiguity characters in the dictionary.</tt> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line">        <tt class="py-comment"># get a starting dictionary based on the alphabet of the alignment</tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line">        <tt class="py-name">rep_dict</tt><tt class="py-op">,</tt> <tt class="py-name">skip_items</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-100" class="py-name" targets="Method Bio.Align.AlignInfo.SummaryInfo._get_base_replacements()=Bio.Align.AlignInfo.SummaryInfo-class.html#_get_base_replacements"><a title="Bio.Align.AlignInfo.SummaryInfo._get_base_replacements" class="py-name" href="#" onclick="return doclink('link-100', '_get_base_replacements', 'link-100');">_get_base_replacements</a></tt><tt class="py-op">(</tt><tt class="py-name">skip_chars</tt><tt class="py-op">)</tt> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line"> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line">        <tt class="py-comment"># iterate through each record</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">rec_num1</tt> <tt class="py-keyword">in</tt> <tt id="link-101" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-101', 'range', 'link-15');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line">            <tt class="py-comment"># iterate through each record from one beyond the current record</tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line">            <tt class="py-comment"># to the end of the list of records</tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">rec_num2</tt> <tt class="py-keyword">in</tt> <tt id="link-102" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-102', 'range', 'link-15');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">rec_num1</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line">                <tt class="py-comment"># for each pair of records, compare the sequences and add</tt> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line">                <tt class="py-comment"># the pertinent info to the dictionary</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line">                <tt class="py-name">rep_dict</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-103" class="py-name" targets="Method Bio.Align.AlignInfo.SummaryInfo._pair_replacement()=Bio.Align.AlignInfo.SummaryInfo-class.html#_pair_replacement"><a title="Bio.Align.AlignInfo.SummaryInfo._pair_replacement" class="py-name" href="#" onclick="return doclink('link-103', '_pair_replacement', 'link-103');">_pair_replacement</a></tt><tt class="py-op">(</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">[</tt><tt class="py-name">rec_num1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-104" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-104', 'seq', 'link-16');">seq</a></tt><tt class="py-op">,</tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">[</tt><tt class="py-name">rec_num2</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-105" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-105', 'seq', 'link-16');">seq</a></tt><tt class="py-op">,</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">[</tt><tt class="py-name">rec_num1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-106" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.annotations=BioSQL.BioSeq.DBSeqRecord-class.html#annotations"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-106', 'annotations', 'link-106');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-107" class="py-name" targets="Method Bio.Crystal.Crystal.get()=Bio.Crystal.Crystal-class.html#get,Method Bio.Data.CodonTable.AmbiguousForwardTable.get()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#get,Method Bio.File._IndexedSeqFileDict.get()=Bio.File._IndexedSeqFileDict-class.html#get,Method Bio.File._IndexedSeqFileProxy.get()=Bio.File._IndexedSeqFileProxy-class.html#get,Method Bio.File._SQLiteManySeqFilesDict.get()=Bio.File._SQLiteManySeqFilesDict-class.html#get,Method Bio.Restriction.Restriction.RestrictionBatch.get()=Bio.Restriction.Restriction.RestrictionBatch-class.html#get,Method Bio.SearchIO._index.SearchIndexer.get()=Bio.SearchIO._index.SearchIndexer-class.html#get,Method Bio.SeqIO._index.SeqFileRandomAccess.get()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#get,Method Bio.SeqIO._index.SffRandomAccess.get()=Bio.SeqIO._index.SffRandomAccess-class.html#get,Method Bio.SeqIO._index.SffTrimedRandomAccess.get()=Bio.SeqIO._index.SffTrimedRandomAccess-class.html#get,Method Bio.SeqIO._index.UniprotRandomAccess.get()=Bio.SeqIO._index.UniprotRandomAccess-class.html#get"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-107', 'get', 'link-107');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'weight'</tt><tt class="py-op">,</tt> <tt class="py-number">1.0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">[</tt><tt class="py-name">rec_num2</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-108" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-108', 'annotations', 'link-106');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-109" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-109', 'get', 'link-107');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'weight'</tt><tt class="py-op">,</tt> <tt class="py-number">1.0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">                    <tt class="py-name">rep_dict</tt><tt class="py-op">,</tt> <tt class="py-name">skip_items</tt><tt class="py-op">)</tt> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line"> </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">rep_dict</tt> </tt>
</div><a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line"> </tt>
<a name="SummaryInfo._pair_replacement"></a><div id="SummaryInfo._pair_replacement-def"><a name="L274"></a><tt class="py-lineno">274</tt> <a class="py-toggle" href="#" id="SummaryInfo._pair_replacement-toggle" onclick="return toggle('SummaryInfo._pair_replacement');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Align.AlignInfo.SummaryInfo-class.html#_pair_replacement">_pair_replacement</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">seq1</tt><tt class="py-op">,</tt> <tt class="py-param">seq2</tt><tt class="py-op">,</tt> <tt class="py-param">weight1</tt><tt class="py-op">,</tt> <tt class="py-param">weight2</tt><tt class="py-op">,</tt> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line">                          <tt class="py-param">start_dict</tt><tt class="py-op">,</tt> <tt class="py-param">ignore_chars</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SummaryInfo._pair_replacement-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SummaryInfo._pair_replacement-expanded"><a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line">        <tt class="py-docstring">"""Compare two sequences and generate info on the replacements seen.</tt> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line"><tt class="py-docstring">            - seq1, seq2 - The two sequences to compare.</tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line"><tt class="py-docstring">            - weight1, weight2 - The relative weights of seq1 and seq2.</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line"><tt class="py-docstring">            - start_dict - The dictionary containing the starting replacement</tt> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line"><tt class="py-docstring">              info that we will modify.</tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line"><tt class="py-docstring">            - ignore_chars - A list of characters to ignore when calculating</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line"><tt class="py-docstring">              replacements (ie. '-').</tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line"><tt class="py-docstring">        Returns:</tt> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line"><tt class="py-docstring">            - A replacment dictionary which is modified from initial_dict with</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line"><tt class="py-docstring">              the information from the sequence comparison.</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">        <tt class="py-comment"># loop through each residue in the sequences</tt> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">residue_num</tt> <tt class="py-keyword">in</tt> <tt id="link-110" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-110', 'range', 'link-15');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-111" class="py-name" targets="Function Bio.SeqUtils.seq1()=Bio.SeqUtils-module.html#seq1"><a title="Bio.SeqUtils.seq1" class="py-name" href="#" onclick="return doclink('link-111', 'seq1', 'link-111');">seq1</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line">            <tt class="py-name">residue1</tt> <tt class="py-op">=</tt> <tt id="link-112" class="py-name"><a title="Bio.SeqUtils.seq1" class="py-name" href="#" onclick="return doclink('link-112', 'seq1', 'link-111');">seq1</a></tt><tt class="py-op">[</tt><tt class="py-name">residue_num</tt><tt class="py-op">]</tt> </tt>
<a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line">                <tt class="py-name">residue2</tt> <tt class="py-op">=</tt> <tt class="py-name">seq2</tt><tt class="py-op">[</tt><tt class="py-name">residue_num</tt><tt class="py-op">]</tt> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line">            <tt class="py-comment"># if seq2 is shorter, then we just stop looking at replacements</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line">            <tt class="py-comment"># and return the information</tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">IndexError</tt><tt class="py-op">:</tt> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">start_dict</tt> </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line"> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line">            <tt class="py-comment"># if the two residues are characters we want to count</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-op">(</tt><tt class="py-name">residue1</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">ignore_chars</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-op">(</tt><tt class="py-name">residue2</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">ignore_chars</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">                <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line">                    <tt class="py-comment"># add info about the replacement to the dictionary,</tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line">                    <tt class="py-comment"># modified by the sequence weights</tt> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line">                    <tt class="py-name">start_dict</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">residue1</tt><tt class="py-op">,</tt> <tt class="py-name">residue2</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-op">+=</tt> <tt class="py-name">weight1</tt> <tt class="py-op">*</tt> <tt class="py-name">weight2</tt> </tt>
<a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line"> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">                <tt class="py-comment"># if we get a key error, then we've got a problem with alphabets</tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line">                <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Residues %s, %s not found in alphabet %s"</tt> </tt>
<a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line">                                     <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">residue1</tt><tt class="py-op">,</tt> <tt class="py-name">residue2</tt><tt class="py-op">,</tt> </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line">                                        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">.</tt><tt class="py-name">_alphabet</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line"> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">start_dict</tt> </tt>
</div><a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line"> </tt>
<a name="SummaryInfo._get_all_letters"></a><div id="SummaryInfo._get_all_letters-def"><a name="L315"></a><tt class="py-lineno">315</tt> <a class="py-toggle" href="#" id="SummaryInfo._get_all_letters-toggle" onclick="return toggle('SummaryInfo._get_all_letters');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Align.AlignInfo.SummaryInfo-class.html#_get_all_letters">_get_all_letters</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SummaryInfo._get_all_letters-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SummaryInfo._get_all_letters-expanded"><a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line">        <tt class="py-docstring">"""Returns a string containing the expected letters in the alignment."""</tt> </tt>
<a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line">        <tt class="py-name">all_letters</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">.</tt><tt class="py-name">_alphabet</tt><tt class="py-op">.</tt><tt id="link-113" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-113', 'letters', 'link-57');">letters</a></tt> </tt>
<a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">all_letters</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt> \ </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line">        <tt class="py-keyword">or</tt> <tt class="py-op">(</tt><tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">.</tt><tt class="py-name">_alphabet</tt><tt class="py-op">,</tt> <tt id="link-114" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-114', 'Alphabet', 'link-1');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-115" class="py-name" targets="Class Bio.Alphabet.Gapped=Bio.Alphabet.Gapped-class.html"><a title="Bio.Alphabet.Gapped" class="py-name" href="#" onclick="return doclink('link-115', 'Gapped', 'link-115');">Gapped</a></tt><tt class="py-op">)</tt> </tt>
<a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line">        <tt class="py-keyword">and</tt> <tt class="py-name">all_letters</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">.</tt><tt class="py-name">_alphabet</tt><tt class="py-op">.</tt><tt class="py-name">gap_char</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line">            <tt class="py-comment"># We are dealing with a generic alphabet class where the</tt> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">            <tt class="py-comment"># letters are not defined!  We must build a list of the</tt> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">            <tt class="py-comment"># letters used...</tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line">            <tt class="py-name">set_letters</tt> <tt class="py-op">=</tt> <tt id="link-116" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.set()=Bio.Nexus.Nexus.StepMatrix-class.html#set,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.set()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#set,Method Bio.SVDSuperimposer.SVDSuperimposer.set()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#set"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-116', 'set', 'link-116');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">:</tt> </tt>
<a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">                <tt class="py-comment"># Note the built in set does not have a union_update</tt> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line">                <tt class="py-comment"># which was provided by the sets module's Set</tt> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line">                <tt class="py-name">set_letters</tt> <tt class="py-op">=</tt> <tt class="py-name">set_letters</tt><tt class="py-op">.</tt><tt class="py-name">union</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-117" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-117', 'seq', 'link-16');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line">            <tt class="py-name">list_letters</tt> <tt class="py-op">=</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt class="py-name">set_letters</tt><tt class="py-op">)</tt> </tt>
<a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line">            <tt class="py-name">all_letters</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">list_letters</tt><tt class="py-op">)</tt> </tt>
<a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">all_letters</tt> </tt>
</div><a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line"> </tt>
<a name="SummaryInfo._get_base_replacements"></a><div id="SummaryInfo._get_base_replacements-def"><a name="L333"></a><tt class="py-lineno">333</tt> <a class="py-toggle" href="#" id="SummaryInfo._get_base_replacements-toggle" onclick="return toggle('SummaryInfo._get_base_replacements');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Align.AlignInfo.SummaryInfo-class.html#_get_base_replacements">_get_base_replacements</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">skip_items</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SummaryInfo._get_base_replacements-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SummaryInfo._get_base_replacements-expanded"><a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get a zeroed dictionary of all possible letter combinations.</tt> </tt>
<a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line"><tt class="py-docstring">        This looks at the type of alphabet and gets the letters for it.</tt> </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line"><tt class="py-docstring">        It then creates a dictionary with all possible combinations of these</tt> </tt>
<a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line"><tt class="py-docstring">        letters as keys (ie. ('A', 'G')) and sets the values as zero.</tt> </tt>
<a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line"><tt class="py-docstring">        Returns:</tt> </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line"><tt class="py-docstring">            - The base dictionary created</tt> </tt>
<a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line"><tt class="py-docstring">            - A list of alphabet items to skip when filling the dictionary.</tt> </tt>
<a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line"><tt class="py-docstring">              (Right now the only thing I can imagine in this list is gap</tt> </tt>
<a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line"><tt class="py-docstring">              characters, but maybe X's or something else might be useful later.</tt> </tt>
<a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line"><tt class="py-docstring">              This will also include any characters that are specified to be</tt> </tt>
<a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line"><tt class="py-docstring">              skipped.) Defaults to an empty list.</tt> </tt>
<a name="L347"></a><tt class="py-lineno">347</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">skip_items</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line">            <tt class="py-name">skip_items</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line">        <tt class="py-name">base_dictionary</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line">        <tt class="py-name">all_letters</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-118" class="py-name" targets="Method Bio.Align.AlignInfo.SummaryInfo._get_all_letters()=Bio.Align.AlignInfo.SummaryInfo-class.html#_get_all_letters"><a title="Bio.Align.AlignInfo.SummaryInfo._get_all_letters" class="py-name" href="#" onclick="return doclink('link-118', '_get_all_letters', 'link-118');">_get_all_letters</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L352"></a><tt class="py-lineno">352</tt>  <tt class="py-line"> </tt>
<a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line">        <tt class="py-comment"># if we have a gapped alphabet we need to find the gap character</tt> </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line">        <tt class="py-comment"># and drop it out</tt> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">.</tt><tt class="py-name">_alphabet</tt><tt class="py-op">,</tt> <tt id="link-119" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-119', 'Alphabet', 'link-1');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-120" class="py-name"><a title="Bio.Alphabet.Gapped" class="py-name" href="#" onclick="return doclink('link-120', 'Gapped', 'link-115');">Gapped</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line">            <tt class="py-name">skip_items</tt><tt class="py-op">.</tt><tt id="link-121" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-121', 'append', 'link-22');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">.</tt><tt class="py-name">_alphabet</tt><tt class="py-op">.</tt><tt class="py-name">gap_char</tt><tt class="py-op">)</tt> </tt>
<a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line">            <tt class="py-name">all_letters</tt> <tt class="py-op">=</tt> <tt class="py-name">all_letters</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">.</tt><tt class="py-name">_alphabet</tt><tt class="py-op">.</tt><tt class="py-name">gap_char</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L358"></a><tt class="py-lineno">358</tt>  <tt class="py-line"> </tt>
<a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line">        <tt class="py-comment"># now create the dictionary</tt> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">first_letter</tt> <tt class="py-keyword">in</tt> <tt class="py-name">all_letters</tt><tt class="py-op">:</tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">second_letter</tt> <tt class="py-keyword">in</tt> <tt class="py-name">all_letters</tt><tt class="py-op">:</tt> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">first_letter</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">skip_items</tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line">                   <tt class="py-name">second_letter</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">skip_items</tt><tt class="py-op">:</tt> </tt>
<a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line">                    <tt class="py-name">base_dictionary</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">first_letter</tt><tt class="py-op">,</tt> <tt class="py-name">second_letter</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line"> </tt>
<a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">base_dictionary</tt><tt class="py-op">,</tt> <tt class="py-name">skip_items</tt> </tt>
</div><a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line"> </tt>
<a name="SummaryInfo.pos_specific_score_matrix"></a><div id="SummaryInfo.pos_specific_score_matrix-def"><a name="L368"></a><tt class="py-lineno">368</tt> <a class="py-toggle" href="#" id="SummaryInfo.pos_specific_score_matrix-toggle" onclick="return toggle('SummaryInfo.pos_specific_score_matrix');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Align.AlignInfo.SummaryInfo-class.html#pos_specific_score_matrix">pos_specific_score_matrix</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">axis_seq</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line">                                  <tt class="py-param">chars_to_ignore</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SummaryInfo.pos_specific_score_matrix-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SummaryInfo.pos_specific_score_matrix-expanded"><a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a position specific score matrix object for the alignment.</tt> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line"><tt class="py-docstring">        This creates a position specific score matrix (pssm) which is an</tt> </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line"><tt class="py-docstring">        alternative method to look at a consensus sequence.</tt> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line"><tt class="py-docstring">            - chars_to_ignore - A list of all characters not to include in</tt> </tt>
<a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line"><tt class="py-docstring">              the pssm.  If the alignment alphabet declares a gap character,</tt> </tt>
<a name="L378"></a><tt class="py-lineno">378</tt>  <tt class="py-line"><tt class="py-docstring">              then it will be excluded automatically.</tt> </tt>
<a name="L379"></a><tt class="py-lineno">379</tt>  <tt class="py-line"><tt class="py-docstring">            - axis_seq - An optional argument specifying the sequence to</tt> </tt>
<a name="L380"></a><tt class="py-lineno">380</tt>  <tt class="py-line"><tt class="py-docstring">              put on the axis of the PSSM. This should be a Seq object. If nothing</tt> </tt>
<a name="L381"></a><tt class="py-lineno">381</tt>  <tt class="py-line"><tt class="py-docstring">              is specified, the consensus sequence, calculated with default</tt> </tt>
<a name="L382"></a><tt class="py-lineno">382</tt>  <tt class="py-line"><tt class="py-docstring">              parameters, will be used.</tt> </tt>
<a name="L383"></a><tt class="py-lineno">383</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L384"></a><tt class="py-lineno">384</tt>  <tt class="py-line"><tt class="py-docstring">        Returns:</tt> </tt>
<a name="L385"></a><tt class="py-lineno">385</tt>  <tt class="py-line"><tt class="py-docstring">            - A PSSM (position specific score matrix) object.</tt> </tt>
<a name="L386"></a><tt class="py-lineno">386</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L387"></a><tt class="py-lineno">387</tt>  <tt class="py-line">        <tt class="py-comment"># determine all of the letters we have to deal with</tt> </tt>
<a name="L388"></a><tt class="py-lineno">388</tt>  <tt class="py-line">        <tt class="py-name">all_letters</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-122" class="py-name"><a title="Bio.Align.AlignInfo.SummaryInfo._get_all_letters" class="py-name" href="#" onclick="return doclink('link-122', '_get_all_letters', 'link-118');">_get_all_letters</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L389"></a><tt class="py-lineno">389</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">all_letters</tt> </tt>
<a name="L390"></a><tt class="py-lineno">390</tt>  <tt class="py-line"> </tt>
<a name="L391"></a><tt class="py-lineno">391</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">chars_to_ignore</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L392"></a><tt class="py-lineno">392</tt>  <tt class="py-line">            <tt class="py-name">chars_to_ignore</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L393"></a><tt class="py-lineno">393</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">chars_to_ignore</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L394"></a><tt class="py-lineno">394</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"chars_to_ignore should be a list."</tt><tt class="py-op">)</tt> </tt>
<a name="L395"></a><tt class="py-lineno">395</tt>  <tt class="py-line"> </tt>
<a name="L396"></a><tt class="py-lineno">396</tt>  <tt class="py-line">        <tt class="py-comment"># if we have a gap char, add it to stuff to ignore</tt> </tt>
<a name="L397"></a><tt class="py-lineno">397</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">.</tt><tt class="py-name">_alphabet</tt><tt class="py-op">,</tt> <tt id="link-123" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-123', 'Alphabet', 'link-1');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-124" class="py-name"><a title="Bio.Alphabet.Gapped" class="py-name" href="#" onclick="return doclink('link-124', 'Gapped', 'link-115');">Gapped</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L398"></a><tt class="py-lineno">398</tt>  <tt class="py-line">            <tt class="py-name">chars_to_ignore</tt><tt class="py-op">.</tt><tt id="link-125" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-125', 'append', 'link-22');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">.</tt><tt class="py-name">_alphabet</tt><tt class="py-op">.</tt><tt class="py-name">gap_char</tt><tt class="py-op">)</tt> </tt>
<a name="L399"></a><tt class="py-lineno">399</tt>  <tt class="py-line"> </tt>
<a name="L400"></a><tt class="py-lineno">400</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">char</tt> <tt class="py-keyword">in</tt> <tt class="py-name">chars_to_ignore</tt><tt class="py-op">:</tt> </tt>
<a name="L401"></a><tt class="py-lineno">401</tt>  <tt class="py-line">            <tt class="py-name">all_letters</tt> <tt class="py-op">=</tt> <tt class="py-name">all_letters</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-name">char</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L402"></a><tt class="py-lineno">402</tt>  <tt class="py-line"> </tt>
<a name="L403"></a><tt class="py-lineno">403</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">axis_seq</tt><tt class="py-op">:</tt> </tt>
<a name="L404"></a><tt class="py-lineno">404</tt>  <tt class="py-line">            <tt class="py-name">left_seq</tt> <tt class="py-op">=</tt> <tt class="py-name">axis_seq</tt> </tt>
<a name="L405"></a><tt class="py-lineno">405</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">axis_seq</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">.</tt><tt id="link-126" class="py-name"><a title="Bio.Align.Generic.Alignment.get_alignment_length" class="py-name" href="#" onclick="return doclink('link-126', 'get_alignment_length', 'link-14');">get_alignment_length</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L406"></a><tt class="py-lineno">406</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L407"></a><tt class="py-lineno">407</tt>  <tt class="py-line">            <tt class="py-name">left_seq</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-127" class="py-name" targets="Method Bio.Align.AlignInfo.SummaryInfo.dumb_consensus()=Bio.Align.AlignInfo.SummaryInfo-class.html#dumb_consensus"><a title="Bio.Align.AlignInfo.SummaryInfo.dumb_consensus" class="py-name" href="#" onclick="return doclink('link-127', 'dumb_consensus', 'link-127');">dumb_consensus</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L408"></a><tt class="py-lineno">408</tt>  <tt class="py-line"> </tt>
<a name="L409"></a><tt class="py-lineno">409</tt>  <tt class="py-line">        <tt class="py-name">pssm_info</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L410"></a><tt class="py-lineno">410</tt>  <tt class="py-line">        <tt class="py-comment"># now start looping through all of the sequences and getting info</tt> </tt>
<a name="L411"></a><tt class="py-lineno">411</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">residue_num</tt> <tt class="py-keyword">in</tt> <tt id="link-128" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-128', 'range', 'link-15');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">left_seq</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L412"></a><tt class="py-lineno">412</tt>  <tt class="py-line">            <tt class="py-name">score_dict</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-129" class="py-name" targets="Method Bio.Align.AlignInfo.SummaryInfo._get_base_letters()=Bio.Align.AlignInfo.SummaryInfo-class.html#_get_base_letters"><a title="Bio.Align.AlignInfo.SummaryInfo._get_base_letters" class="py-name" href="#" onclick="return doclink('link-129', '_get_base_letters', 'link-129');">_get_base_letters</a></tt><tt class="py-op">(</tt><tt class="py-name">all_letters</tt><tt class="py-op">)</tt> </tt>
<a name="L413"></a><tt class="py-lineno">413</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">:</tt> </tt>
<a name="L414"></a><tt class="py-lineno">414</tt>  <tt class="py-line">                <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L415"></a><tt class="py-lineno">415</tt>  <tt class="py-line">                    <tt class="py-name">this_residue</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-130" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-130', 'seq', 'link-16');">seq</a></tt><tt class="py-op">[</tt><tt class="py-name">residue_num</tt><tt class="py-op">]</tt> </tt>
<a name="L416"></a><tt class="py-lineno">416</tt>  <tt class="py-line">                <tt class="py-comment"># if we hit an index error we've run out of sequence and</tt> </tt>
<a name="L417"></a><tt class="py-lineno">417</tt>  <tt class="py-line">                <tt class="py-comment"># should not add new residues</tt> </tt>
<a name="L418"></a><tt class="py-lineno">418</tt>  <tt class="py-line">                <tt class="py-keyword">except</tt> <tt class="py-name">IndexError</tt><tt class="py-op">:</tt> </tt>
<a name="L419"></a><tt class="py-lineno">419</tt>  <tt class="py-line">                    <tt class="py-name">this_residue</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L420"></a><tt class="py-lineno">420</tt>  <tt class="py-line"> </tt>
<a name="L421"></a><tt class="py-lineno">421</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">this_residue</tt> <tt class="py-keyword">and</tt> <tt class="py-name">this_residue</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">chars_to_ignore</tt><tt class="py-op">:</tt> </tt>
<a name="L422"></a><tt class="py-lineno">422</tt>  <tt class="py-line">                    <tt class="py-name">weight</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-131" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-131', 'annotations', 'link-106');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-132" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-132', 'get', 'link-107');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'weight'</tt><tt class="py-op">,</tt> <tt class="py-number">1.0</tt><tt class="py-op">)</tt> </tt>
<a name="L423"></a><tt class="py-lineno">423</tt>  <tt class="py-line">                    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L424"></a><tt class="py-lineno">424</tt>  <tt class="py-line">                        <tt class="py-name">score_dict</tt><tt class="py-op">[</tt><tt class="py-name">this_residue</tt><tt class="py-op">]</tt> <tt class="py-op">+=</tt> <tt class="py-name">weight</tt> </tt>
<a name="L425"></a><tt class="py-lineno">425</tt>  <tt class="py-line">                    <tt class="py-comment"># if we get a KeyError then we have an alphabet problem</tt> </tt>
<a name="L426"></a><tt class="py-lineno">426</tt>  <tt class="py-line">                    <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L427"></a><tt class="py-lineno">427</tt>  <tt class="py-line">                        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Residue %s not found in alphabet %s"</tt> </tt>
<a name="L428"></a><tt class="py-lineno">428</tt>  <tt class="py-line">                                     <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">this_residue</tt><tt class="py-op">,</tt> </tt>
<a name="L429"></a><tt class="py-lineno">429</tt>  <tt class="py-line">                                        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">.</tt><tt class="py-name">_alphabet</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L430"></a><tt class="py-lineno">430</tt>  <tt class="py-line"> </tt>
<a name="L431"></a><tt class="py-lineno">431</tt>  <tt class="py-line">            <tt class="py-name">pssm_info</tt><tt class="py-op">.</tt><tt id="link-133" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-133', 'append', 'link-22');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">left_seq</tt><tt class="py-op">[</tt><tt class="py-name">residue_num</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L432"></a><tt class="py-lineno">432</tt>  <tt class="py-line">                              <tt class="py-name">score_dict</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L433"></a><tt class="py-lineno">433</tt>  <tt class="py-line"> </tt>
<a name="L434"></a><tt class="py-lineno">434</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-134" class="py-name" targets="Class Bio.Align.AlignInfo.PSSM=Bio.Align.AlignInfo.PSSM-class.html"><a title="Bio.Align.AlignInfo.PSSM" class="py-name" href="#" onclick="return doclink('link-134', 'PSSM', 'link-134');">PSSM</a></tt><tt class="py-op">(</tt><tt class="py-name">pssm_info</tt><tt class="py-op">)</tt> </tt>
</div><a name="L435"></a><tt class="py-lineno">435</tt>  <tt class="py-line"> </tt>
<a name="SummaryInfo._get_base_letters"></a><div id="SummaryInfo._get_base_letters-def"><a name="L436"></a><tt class="py-lineno">436</tt> <a class="py-toggle" href="#" id="SummaryInfo._get_base_letters-toggle" onclick="return toggle('SummaryInfo._get_base_letters');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Align.AlignInfo.SummaryInfo-class.html#_get_base_letters">_get_base_letters</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">letters</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SummaryInfo._get_base_letters-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SummaryInfo._get_base_letters-expanded"><a name="L437"></a><tt class="py-lineno">437</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a zeroed dictionary with all of the specified letters.</tt> </tt>
<a name="L438"></a><tt class="py-lineno">438</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L439"></a><tt class="py-lineno">439</tt>  <tt class="py-line">        <tt class="py-name">base_info</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L440"></a><tt class="py-lineno">440</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">letter</tt> <tt class="py-keyword">in</tt> <tt id="link-135" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-135', 'letters', 'link-57');">letters</a></tt><tt class="py-op">:</tt> </tt>
<a name="L441"></a><tt class="py-lineno">441</tt>  <tt class="py-line">            <tt class="py-name">base_info</tt><tt class="py-op">[</tt><tt class="py-name">letter</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L442"></a><tt class="py-lineno">442</tt>  <tt class="py-line"> </tt>
<a name="L443"></a><tt class="py-lineno">443</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">base_info</tt> </tt>
</div><a name="L444"></a><tt class="py-lineno">444</tt>  <tt class="py-line"> </tt>
<a name="SummaryInfo.information_content"></a><div id="SummaryInfo.information_content-def"><a name="L445"></a><tt class="py-lineno">445</tt> <a class="py-toggle" href="#" id="SummaryInfo.information_content-toggle" onclick="return toggle('SummaryInfo.information_content');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Align.AlignInfo.SummaryInfo-class.html#information_content">information_content</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> </tt>
<a name="L446"></a><tt class="py-lineno">446</tt>  <tt class="py-line">                            <tt class="py-param">end</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L447"></a><tt class="py-lineno">447</tt>  <tt class="py-line">                            <tt class="py-param">e_freq_table</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">log_base</tt><tt class="py-op">=</tt><tt class="py-number">2</tt><tt class="py-op">,</tt> </tt>
<a name="L448"></a><tt class="py-lineno">448</tt>  <tt class="py-line">                            <tt class="py-param">chars_to_ignore</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SummaryInfo.information_content-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SummaryInfo.information_content-expanded"><a name="L449"></a><tt class="py-lineno">449</tt>  <tt class="py-line">        <tt class="py-docstring">"""Calculate the information content for each residue along an alignment.</tt> </tt>
<a name="L450"></a><tt class="py-lineno">450</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L451"></a><tt class="py-lineno">451</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L452"></a><tt class="py-lineno">452</tt>  <tt class="py-line"><tt class="py-docstring">            - start, end - The starting an ending points to calculate the</tt> </tt>
<a name="L453"></a><tt class="py-lineno">453</tt>  <tt class="py-line"><tt class="py-docstring">              information content. These points should be relative to the first</tt> </tt>
<a name="L454"></a><tt class="py-lineno">454</tt>  <tt class="py-line"><tt class="py-docstring">              sequence in the alignment, starting at zero (ie. even if the 'real'</tt> </tt>
<a name="L455"></a><tt class="py-lineno">455</tt>  <tt class="py-line"><tt class="py-docstring">              first position in the seq is 203 in the initial sequence, for</tt> </tt>
<a name="L456"></a><tt class="py-lineno">456</tt>  <tt class="py-line"><tt class="py-docstring">              the info content, we need to use zero). This defaults to the entire</tt> </tt>
<a name="L457"></a><tt class="py-lineno">457</tt>  <tt class="py-line"><tt class="py-docstring">              length of the first sequence.</tt> </tt>
<a name="L458"></a><tt class="py-lineno">458</tt>  <tt class="py-line"><tt class="py-docstring">            - e_freq_table - A FreqTable object specifying the expected frequencies</tt> </tt>
<a name="L459"></a><tt class="py-lineno">459</tt>  <tt class="py-line"><tt class="py-docstring">              for each letter in the alphabet we are using (e.g. {'G' : 0.4,</tt> </tt>
<a name="L460"></a><tt class="py-lineno">460</tt>  <tt class="py-line"><tt class="py-docstring">              'C' : 0.4, 'T' : 0.1, 'A' : 0.1}). Gap characters should not be</tt> </tt>
<a name="L461"></a><tt class="py-lineno">461</tt>  <tt class="py-line"><tt class="py-docstring">              included, since these should not have expected frequencies.</tt> </tt>
<a name="L462"></a><tt class="py-lineno">462</tt>  <tt class="py-line"><tt class="py-docstring">            - log_base - The base of the logathrim to use in calculating the</tt> </tt>
<a name="L463"></a><tt class="py-lineno">463</tt>  <tt class="py-line"><tt class="py-docstring">              information content. This defaults to 2 so the info is in bits.</tt> </tt>
<a name="L464"></a><tt class="py-lineno">464</tt>  <tt class="py-line"><tt class="py-docstring">            - chars_to_ignore - A listing of characters which should be ignored</tt> </tt>
<a name="L465"></a><tt class="py-lineno">465</tt>  <tt class="py-line"><tt class="py-docstring">              in calculating the info content. Defaults to none.</tt> </tt>
<a name="L466"></a><tt class="py-lineno">466</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L467"></a><tt class="py-lineno">467</tt>  <tt class="py-line"><tt class="py-docstring">        Returns:</tt> </tt>
<a name="L468"></a><tt class="py-lineno">468</tt>  <tt class="py-line"><tt class="py-docstring">            - A number representing the info content for the specified region.</tt> </tt>
<a name="L469"></a><tt class="py-lineno">469</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L470"></a><tt class="py-lineno">470</tt>  <tt class="py-line"><tt class="py-docstring">        Please see the Biopython manual for more information on how information</tt> </tt>
<a name="L471"></a><tt class="py-lineno">471</tt>  <tt class="py-line"><tt class="py-docstring">        content is calculated.</tt> </tt>
<a name="L472"></a><tt class="py-lineno">472</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L473"></a><tt class="py-lineno">473</tt>  <tt class="py-line">        <tt class="py-comment"># if no end was specified, then we default to the end of the sequence</tt> </tt>
<a name="L474"></a><tt class="py-lineno">474</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-136" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.end=Bio.SeqFeature.CompoundLocation-class.html#end,Variable Bio.SeqFeature.FeatureLocation.end=Bio.SeqFeature.FeatureLocation-class.html#end"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-136', 'end', 'link-136');">end</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L475"></a><tt class="py-lineno">475</tt>  <tt class="py-line">            <tt id="link-137" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-137', 'end', 'link-136');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-138" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-138', 'seq', 'link-16');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L476"></a><tt class="py-lineno">476</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">chars_to_ignore</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L477"></a><tt class="py-lineno">477</tt>  <tt class="py-line">            <tt class="py-name">chars_to_ignore</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L478"></a><tt class="py-lineno">478</tt>  <tt class="py-line"> </tt>
<a name="L479"></a><tt class="py-lineno">479</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-139" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.start=Bio.SeqFeature.CompoundLocation-class.html#start,Variable Bio.SeqFeature.FeatureLocation.start=Bio.SeqFeature.FeatureLocation-class.html#start"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-139', 'start', 'link-139');">start</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt> <tt class="py-keyword">or</tt> <tt id="link-140" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-140', 'end', 'link-136');">end</a></tt> <tt class="py-op">&gt;</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-141" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-141', 'seq', 'link-16');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L480"></a><tt class="py-lineno">480</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Start (%s) and end (%s) are not in the \</tt> </tt>
<a name="L481"></a><tt class="py-lineno">481</tt>  <tt class="py-line"><tt class="py-string">                    range %s to %s"</tt> </tt>
<a name="L482"></a><tt class="py-lineno">482</tt>  <tt class="py-line">                    <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-142" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-142', 'start', 'link-139');">start</a></tt><tt class="py-op">,</tt> <tt id="link-143" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-143', 'end', 'link-136');">end</a></tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-144" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-144', 'seq', 'link-16');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L483"></a><tt class="py-lineno">483</tt>  <tt class="py-line">        <tt class="py-comment"># determine random expected frequencies, if necessary</tt> </tt>
<a name="L484"></a><tt class="py-lineno">484</tt>  <tt class="py-line">        <tt class="py-name">random_expected</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L485"></a><tt class="py-lineno">485</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">e_freq_table</tt><tt class="py-op">:</tt> </tt>
<a name="L486"></a><tt class="py-lineno">486</tt>  <tt class="py-line">            <tt class="py-comment"># TODO - What about ambiguous alphabets?</tt> </tt>
<a name="L487"></a><tt class="py-lineno">487</tt>  <tt class="py-line">            <tt class="py-name">base_alpha</tt> <tt class="py-op">=</tt> <tt id="link-145" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-145', 'Alphabet', 'link-1');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-146" class="py-name"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-146', '_get_base_alphabet', 'link-47');">_get_base_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">.</tt><tt class="py-name">_alphabet</tt><tt class="py-op">)</tt> </tt>
<a name="L488"></a><tt class="py-lineno">488</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">base_alpha</tt><tt class="py-op">,</tt> <tt id="link-147" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-147', 'Alphabet', 'link-1');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-148" class="py-name" targets="Class Bio.Alphabet.ProteinAlphabet=Bio.Alphabet.ProteinAlphabet-class.html"><a title="Bio.Alphabet.ProteinAlphabet" class="py-name" href="#" onclick="return doclink('link-148', 'ProteinAlphabet', 'link-148');">ProteinAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L489"></a><tt class="py-lineno">489</tt>  <tt class="py-line">                <tt class="py-name">random_expected</tt> <tt class="py-op">=</tt> <tt id="link-149" class="py-name"><a title="Bio.Align.AlignInfo.Protein20Random" class="py-name" href="#" onclick="return doclink('link-149', 'Protein20Random', 'link-11');">Protein20Random</a></tt> </tt>
<a name="L490"></a><tt class="py-lineno">490</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">base_alpha</tt><tt class="py-op">,</tt> <tt id="link-150" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-150', 'Alphabet', 'link-1');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-151" class="py-name" targets="Class Bio.Alphabet.NucleotideAlphabet=Bio.Alphabet.NucleotideAlphabet-class.html"><a title="Bio.Alphabet.NucleotideAlphabet" class="py-name" href="#" onclick="return doclink('link-151', 'NucleotideAlphabet', 'link-151');">NucleotideAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L491"></a><tt class="py-lineno">491</tt>  <tt class="py-line">                <tt class="py-name">random_expected</tt> <tt class="py-op">=</tt> <tt id="link-152" class="py-name"><a title="Bio.Align.AlignInfo.Nucleotide4Random" class="py-name" href="#" onclick="return doclink('link-152', 'Nucleotide4Random', 'link-12');">Nucleotide4Random</a></tt> </tt>
<a name="L492"></a><tt class="py-lineno">492</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L493"></a><tt class="py-lineno">493</tt>  <tt class="py-line">                <tt class="py-name">errstr</tt> <tt class="py-op">=</tt> <tt class="py-string">"Error in alphabet: not Nucleotide or Protein, "</tt> </tt>
<a name="L494"></a><tt class="py-lineno">494</tt>  <tt class="py-line">                <tt class="py-name">errstr</tt> <tt class="py-op">+=</tt> <tt class="py-string">"supply expected frequencies"</tt> </tt>
<a name="L495"></a><tt class="py-lineno">495</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-name">errstr</tt><tt class="py-op">)</tt> </tt>
<a name="L496"></a><tt class="py-lineno">496</tt>  <tt class="py-line">            <tt class="py-keyword">del</tt> <tt class="py-name">base_alpha</tt> </tt>
<a name="L497"></a><tt class="py-lineno">497</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">e_freq_table</tt><tt class="py-op">,</tt> <tt id="link-153" class="py-name"><a title="Bio.SubsMat.FreqTable
Bio.SubsMat.FreqTable.FreqTable" class="py-name" href="#" onclick="return doclink('link-153', 'FreqTable', 'link-10');">FreqTable</a></tt><tt class="py-op">.</tt><tt id="link-154" class="py-name"><a title="Bio.SubsMat.FreqTable
Bio.SubsMat.FreqTable.FreqTable" class="py-name" href="#" onclick="return doclink('link-154', 'FreqTable', 'link-10');">FreqTable</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L498"></a><tt class="py-lineno">498</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"e_freq_table should be a FreqTable object"</tt><tt class="py-op">)</tt> </tt>
<a name="L499"></a><tt class="py-lineno">499</tt>  <tt class="py-line"> </tt>
<a name="L500"></a><tt class="py-lineno">500</tt>  <tt class="py-line">        <tt class="py-comment"># determine all of the letters we have to deal with</tt> </tt>
<a name="L501"></a><tt class="py-lineno">501</tt>  <tt class="py-line">        <tt class="py-name">all_letters</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-155" class="py-name"><a title="Bio.Align.AlignInfo.SummaryInfo._get_all_letters" class="py-name" href="#" onclick="return doclink('link-155', '_get_all_letters', 'link-118');">_get_all_letters</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L502"></a><tt class="py-lineno">502</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">char</tt> <tt class="py-keyword">in</tt> <tt class="py-name">chars_to_ignore</tt><tt class="py-op">:</tt> </tt>
<a name="L503"></a><tt class="py-lineno">503</tt>  <tt class="py-line">            <tt class="py-name">all_letters</tt> <tt class="py-op">=</tt> <tt class="py-name">all_letters</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-name">char</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L504"></a><tt class="py-lineno">504</tt>  <tt class="py-line"> </tt>
<a name="L505"></a><tt class="py-lineno">505</tt>  <tt class="py-line">        <tt class="py-name">info_content</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L506"></a><tt class="py-lineno">506</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">residue_num</tt> <tt class="py-keyword">in</tt> <tt id="link-156" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-156', 'range', 'link-15');">range</a></tt><tt class="py-op">(</tt><tt id="link-157" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-157', 'start', 'link-139');">start</a></tt><tt class="py-op">,</tt> <tt id="link-158" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-158', 'end', 'link-136');">end</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L507"></a><tt class="py-lineno">507</tt>  <tt class="py-line">            <tt class="py-name">freq_dict</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-159" class="py-name" targets="Method Bio.Align.AlignInfo.SummaryInfo._get_letter_freqs()=Bio.Align.AlignInfo.SummaryInfo-class.html#_get_letter_freqs"><a title="Bio.Align.AlignInfo.SummaryInfo._get_letter_freqs" class="py-name" href="#" onclick="return doclink('link-159', '_get_letter_freqs', 'link-159');">_get_letter_freqs</a></tt><tt class="py-op">(</tt><tt class="py-name">residue_num</tt><tt class="py-op">,</tt> </tt>
<a name="L508"></a><tt class="py-lineno">508</tt>  <tt class="py-line">                                               <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">,</tt> </tt>
<a name="L509"></a><tt class="py-lineno">509</tt>  <tt class="py-line">                                               <tt class="py-name">all_letters</tt><tt class="py-op">,</tt> <tt class="py-name">chars_to_ignore</tt><tt class="py-op">)</tt> </tt>
<a name="L510"></a><tt class="py-lineno">510</tt>  <tt class="py-line">            <tt class="py-comment"># print freq_dict,</tt> </tt>
<a name="L511"></a><tt class="py-lineno">511</tt>  <tt class="py-line">            <tt class="py-name">column_score</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-160" class="py-name" targets="Method Bio.Align.AlignInfo.SummaryInfo._get_column_info_content()=Bio.Align.AlignInfo.SummaryInfo-class.html#_get_column_info_content"><a title="Bio.Align.AlignInfo.SummaryInfo._get_column_info_content" class="py-name" href="#" onclick="return doclink('link-160', '_get_column_info_content', 'link-160');">_get_column_info_content</a></tt><tt class="py-op">(</tt><tt class="py-name">freq_dict</tt><tt class="py-op">,</tt> </tt>
<a name="L512"></a><tt class="py-lineno">512</tt>  <tt class="py-line">                                                         <tt class="py-name">e_freq_table</tt><tt class="py-op">,</tt> </tt>
<a name="L513"></a><tt class="py-lineno">513</tt>  <tt class="py-line">                                                         <tt class="py-name">log_base</tt><tt class="py-op">,</tt> </tt>
<a name="L514"></a><tt class="py-lineno">514</tt>  <tt class="py-line">                                                         <tt class="py-name">random_expected</tt><tt class="py-op">)</tt> </tt>
<a name="L515"></a><tt class="py-lineno">515</tt>  <tt class="py-line"> </tt>
<a name="L516"></a><tt class="py-lineno">516</tt>  <tt class="py-line">            <tt class="py-name">info_content</tt><tt class="py-op">[</tt><tt class="py-name">residue_num</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">column_score</tt> </tt>
<a name="L517"></a><tt class="py-lineno">517</tt>  <tt class="py-line">        <tt class="py-comment"># sum up the score</tt> </tt>
<a name="L518"></a><tt class="py-lineno">518</tt>  <tt class="py-line">        <tt class="py-name">total_info</tt> <tt class="py-op">=</tt> <tt id="link-161" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.sum()=Bio.Nexus.Nexus.StepMatrix-class.html#sum,Method Bio.SubsMat.SeqMat.sum()=Bio.SubsMat.SeqMat-class.html#sum"><a title="Bio.Nexus.Nexus.StepMatrix.sum
Bio.SubsMat.SeqMat.sum" class="py-name" href="#" onclick="return doclink('link-161', 'sum', 'link-161');">sum</a></tt><tt class="py-op">(</tt><tt class="py-name">info_content</tt><tt class="py-op">.</tt><tt id="link-162" class="py-name" targets="Method Bio.Crystal.Crystal.values()=Bio.Crystal.Crystal-class.html#values,Method Bio.File._IndexedSeqFileDict.values()=Bio.File._IndexedSeqFileDict-class.html#values,Method Bio.Phylo.PhyloXML.Events.values()=Bio.Phylo.PhyloXML.Events-class.html#values,Method Bio._py3k._ordereddict.OrderedDict.values()=Bio._py3k._ordereddict.OrderedDict-class.html#values,Method BioSQL.BioSeqDatabase.BioSeqDatabase.values()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#values,Method BioSQL.BioSeqDatabase.DBServer.values()=BioSQL.BioSeqDatabase.DBServer-class.html#values"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
Bio._py3k._ordereddict.OrderedDict.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-162', 'values', 'link-162');">values</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L519"></a><tt class="py-lineno">519</tt>  <tt class="py-line">        <tt class="py-comment"># fill in the ic_vector member: holds IC for each column</tt> </tt>
<a name="L520"></a><tt class="py-lineno">520</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-163" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-163', 'i', 'link-163');">i</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">info_content</tt><tt class="py-op">:</tt> </tt>
<a name="L521"></a><tt class="py-lineno">521</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ic_vector</tt><tt class="py-op">[</tt><tt id="link-164" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-164', 'i', 'link-163');">i</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">info_content</tt><tt class="py-op">[</tt><tt id="link-165" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-165', 'i', 'link-163');">i</a></tt><tt class="py-op">]</tt> </tt>
<a name="L522"></a><tt class="py-lineno">522</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">total_info</tt> </tt>
</div><a name="L523"></a><tt class="py-lineno">523</tt>  <tt class="py-line"> </tt>
<a name="SummaryInfo._get_letter_freqs"></a><div id="SummaryInfo._get_letter_freqs-def"><a name="L524"></a><tt class="py-lineno">524</tt> <a class="py-toggle" href="#" id="SummaryInfo._get_letter_freqs-toggle" onclick="return toggle('SummaryInfo._get_letter_freqs');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Align.AlignInfo.SummaryInfo-class.html#_get_letter_freqs">_get_letter_freqs</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">residue_num</tt><tt class="py-op">,</tt> <tt class="py-param">all_records</tt><tt class="py-op">,</tt> <tt class="py-param">letters</tt><tt class="py-op">,</tt> <tt class="py-param">to_ignore</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SummaryInfo._get_letter_freqs-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SummaryInfo._get_letter_freqs-expanded"><a name="L525"></a><tt class="py-lineno">525</tt>  <tt class="py-line">        <tt class="py-docstring">"""Determine the frequency of specific letters in the alignment.</tt> </tt>
<a name="L526"></a><tt class="py-lineno">526</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L527"></a><tt class="py-lineno">527</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L528"></a><tt class="py-lineno">528</tt>  <tt class="py-line"><tt class="py-docstring">            - residue_num - The number of the column we are getting frequencies</tt> </tt>
<a name="L529"></a><tt class="py-lineno">529</tt>  <tt class="py-line"><tt class="py-docstring">              from.</tt> </tt>
<a name="L530"></a><tt class="py-lineno">530</tt>  <tt class="py-line"><tt class="py-docstring">            - all_records - All of the SeqRecords in the alignment.</tt> </tt>
<a name="L531"></a><tt class="py-lineno">531</tt>  <tt class="py-line"><tt class="py-docstring">            - letters - The letters we are interested in getting the frequency</tt> </tt>
<a name="L532"></a><tt class="py-lineno">532</tt>  <tt class="py-line"><tt class="py-docstring">              for.</tt> </tt>
<a name="L533"></a><tt class="py-lineno">533</tt>  <tt class="py-line"><tt class="py-docstring">            - to_ignore - Letters we are specifically supposed to ignore.</tt> </tt>
<a name="L534"></a><tt class="py-lineno">534</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L535"></a><tt class="py-lineno">535</tt>  <tt class="py-line"><tt class="py-docstring">        This will calculate the frequencies of each of the specified letters</tt> </tt>
<a name="L536"></a><tt class="py-lineno">536</tt>  <tt class="py-line"><tt class="py-docstring">        in the alignment at the given frequency, and return this as a</tt> </tt>
<a name="L537"></a><tt class="py-lineno">537</tt>  <tt class="py-line"><tt class="py-docstring">        dictionary where the keys are the letters and the values are the</tt> </tt>
<a name="L538"></a><tt class="py-lineno">538</tt>  <tt class="py-line"><tt class="py-docstring">        frequencies.</tt> </tt>
<a name="L539"></a><tt class="py-lineno">539</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L540"></a><tt class="py-lineno">540</tt>  <tt class="py-line">        <tt class="py-name">freq_info</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-166" class="py-name"><a title="Bio.Align.AlignInfo.SummaryInfo._get_base_letters" class="py-name" href="#" onclick="return doclink('link-166', '_get_base_letters', 'link-129');">_get_base_letters</a></tt><tt class="py-op">(</tt><tt id="link-167" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-167', 'letters', 'link-57');">letters</a></tt><tt class="py-op">)</tt> </tt>
<a name="L541"></a><tt class="py-lineno">541</tt>  <tt class="py-line"> </tt>
<a name="L542"></a><tt class="py-lineno">542</tt>  <tt class="py-line">        <tt class="py-name">total_count</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L543"></a><tt class="py-lineno">543</tt>  <tt class="py-line">        <tt class="py-comment"># collect the count info into the dictionary for all the records</tt> </tt>
<a name="L544"></a><tt class="py-lineno">544</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">all_records</tt><tt class="py-op">:</tt> </tt>
<a name="L545"></a><tt class="py-lineno">545</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L546"></a><tt class="py-lineno">546</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-168" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-168', 'seq', 'link-16');">seq</a></tt><tt class="py-op">[</tt><tt class="py-name">residue_num</tt><tt class="py-op">]</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">to_ignore</tt><tt class="py-op">:</tt> </tt>
<a name="L547"></a><tt class="py-lineno">547</tt>  <tt class="py-line">                    <tt class="py-name">weight</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-169" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-169', 'annotations', 'link-106');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-170" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-170', 'get', 'link-107');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'weight'</tt><tt class="py-op">,</tt> <tt class="py-number">1.0</tt><tt class="py-op">)</tt> </tt>
<a name="L548"></a><tt class="py-lineno">548</tt>  <tt class="py-line">                    <tt class="py-name">freq_info</tt><tt class="py-op">[</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-171" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-171', 'seq', 'link-16');">seq</a></tt><tt class="py-op">[</tt><tt class="py-name">residue_num</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> <tt class="py-op">+=</tt> <tt class="py-name">weight</tt> </tt>
<a name="L549"></a><tt class="py-lineno">549</tt>  <tt class="py-line">                    <tt class="py-name">total_count</tt> <tt class="py-op">+=</tt> <tt class="py-name">weight</tt> </tt>
<a name="L550"></a><tt class="py-lineno">550</tt>  <tt class="py-line">            <tt class="py-comment"># getting a key error means we've got a problem with the alphabet</tt> </tt>
<a name="L551"></a><tt class="py-lineno">551</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L552"></a><tt class="py-lineno">552</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Residue %s not found in alphabet %s"</tt> </tt>
<a name="L553"></a><tt class="py-lineno">553</tt>  <tt class="py-line">                                 <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-172" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-172', 'seq', 'link-16');">seq</a></tt><tt class="py-op">[</tt><tt class="py-name">residue_num</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L554"></a><tt class="py-lineno">554</tt>  <tt class="py-line">                                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">.</tt><tt class="py-name">_alphabet</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L555"></a><tt class="py-lineno">555</tt>  <tt class="py-line"> </tt>
<a name="L556"></a><tt class="py-lineno">556</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">total_count</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L557"></a><tt class="py-lineno">557</tt>  <tt class="py-line">            <tt class="py-comment"># This column must be entirely ignored characters</tt> </tt>
<a name="L558"></a><tt class="py-lineno">558</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">letter</tt> <tt class="py-keyword">in</tt> <tt class="py-name">freq_info</tt><tt class="py-op">:</tt> </tt>
<a name="L559"></a><tt class="py-lineno">559</tt>  <tt class="py-line">                <tt class="py-keyword">assert</tt> <tt class="py-name">freq_info</tt><tt class="py-op">[</tt><tt class="py-name">letter</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt> </tt>
<a name="L560"></a><tt class="py-lineno">560</tt>  <tt class="py-line">                <tt class="py-comment"># TODO - Map this to NA or NaN?</tt> </tt>
<a name="L561"></a><tt class="py-lineno">561</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L562"></a><tt class="py-lineno">562</tt>  <tt class="py-line">            <tt class="py-comment"># now convert the counts into frequencies</tt> </tt>
<a name="L563"></a><tt class="py-lineno">563</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">letter</tt> <tt class="py-keyword">in</tt> <tt class="py-name">freq_info</tt><tt class="py-op">:</tt> </tt>
<a name="L564"></a><tt class="py-lineno">564</tt>  <tt class="py-line">                <tt class="py-name">freq_info</tt><tt class="py-op">[</tt><tt class="py-name">letter</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">freq_info</tt><tt class="py-op">[</tt><tt class="py-name">letter</tt><tt class="py-op">]</tt> <tt class="py-op">/</tt> <tt class="py-name">total_count</tt> </tt>
<a name="L565"></a><tt class="py-lineno">565</tt>  <tt class="py-line"> </tt>
<a name="L566"></a><tt class="py-lineno">566</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">freq_info</tt> </tt>
</div><a name="L567"></a><tt class="py-lineno">567</tt>  <tt class="py-line"> </tt>
<a name="SummaryInfo._get_column_info_content"></a><div id="SummaryInfo._get_column_info_content-def"><a name="L568"></a><tt class="py-lineno">568</tt> <a class="py-toggle" href="#" id="SummaryInfo._get_column_info_content-toggle" onclick="return toggle('SummaryInfo._get_column_info_content');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Align.AlignInfo.SummaryInfo-class.html#_get_column_info_content">_get_column_info_content</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">obs_freq</tt><tt class="py-op">,</tt> <tt class="py-param">e_freq_table</tt><tt class="py-op">,</tt> <tt class="py-param">log_base</tt><tt class="py-op">,</tt> </tt>
<a name="L569"></a><tt class="py-lineno">569</tt>  <tt class="py-line">                                 <tt class="py-param">random_expected</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SummaryInfo._get_column_info_content-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SummaryInfo._get_column_info_content-expanded"><a name="L570"></a><tt class="py-lineno">570</tt>  <tt class="py-line">        <tt class="py-docstring">"""Calculate the information content for a column.</tt> </tt>
<a name="L571"></a><tt class="py-lineno">571</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L572"></a><tt class="py-lineno">572</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L573"></a><tt class="py-lineno">573</tt>  <tt class="py-line"><tt class="py-docstring">            - obs_freq - The frequencies observed for each letter in the column.</tt> </tt>
<a name="L574"></a><tt class="py-lineno">574</tt>  <tt class="py-line"><tt class="py-docstring">            - e_freq_table - An optional argument specifying the expected</tt> </tt>
<a name="L575"></a><tt class="py-lineno">575</tt>  <tt class="py-line"><tt class="py-docstring">              frequencies for each letter. This is a SubsMat.FreqTable instance.</tt> </tt>
<a name="L576"></a><tt class="py-lineno">576</tt>  <tt class="py-line"><tt class="py-docstring">            - log_base - The base of the logathrim to use in calculating the</tt> </tt>
<a name="L577"></a><tt class="py-lineno">577</tt>  <tt class="py-line"><tt class="py-docstring">              info content.</tt> </tt>
<a name="L578"></a><tt class="py-lineno">578</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L579"></a><tt class="py-lineno">579</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L580"></a><tt class="py-lineno">580</tt>  <tt class="py-line">            <tt class="py-name">gap_char</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">.</tt><tt class="py-name">_alphabet</tt><tt class="py-op">.</tt><tt class="py-name">gap_char</tt> </tt>
<a name="L581"></a><tt class="py-lineno">581</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L582"></a><tt class="py-lineno">582</tt>  <tt class="py-line">            <tt class="py-comment"># The alphabet doesn't declare a gap - there could be none</tt> </tt>
<a name="L583"></a><tt class="py-lineno">583</tt>  <tt class="py-line">            <tt class="py-comment"># in the sequence... or just a vague alphabet.</tt> </tt>
<a name="L584"></a><tt class="py-lineno">584</tt>  <tt class="py-line">            <tt class="py-name">gap_char</tt> <tt class="py-op">=</tt> <tt class="py-string">"-"</tt>  <tt class="py-comment"># Safe?</tt> </tt>
<a name="L585"></a><tt class="py-lineno">585</tt>  <tt class="py-line"> </tt>
<a name="L586"></a><tt class="py-lineno">586</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">e_freq_table</tt><tt class="py-op">:</tt> </tt>
<a name="L587"></a><tt class="py-lineno">587</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">e_freq_table</tt><tt class="py-op">,</tt> <tt id="link-173" class="py-name"><a title="Bio.SubsMat.FreqTable
Bio.SubsMat.FreqTable.FreqTable" class="py-name" href="#" onclick="return doclink('link-173', 'FreqTable', 'link-10');">FreqTable</a></tt><tt class="py-op">.</tt><tt id="link-174" class="py-name"><a title="Bio.SubsMat.FreqTable
Bio.SubsMat.FreqTable.FreqTable" class="py-name" href="#" onclick="return doclink('link-174', 'FreqTable', 'link-10');">FreqTable</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L588"></a><tt class="py-lineno">588</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"e_freq_table should be a FreqTable object"</tt><tt class="py-op">)</tt> </tt>
<a name="L589"></a><tt class="py-lineno">589</tt>  <tt class="py-line">            <tt class="py-comment"># check the expected freq information to make sure it is good</tt> </tt>
<a name="L590"></a><tt class="py-lineno">590</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">obs_freq</tt><tt class="py-op">:</tt> </tt>
<a name="L591"></a><tt class="py-lineno">591</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-op">(</tt><tt class="py-name">key</tt> <tt class="py-op">!=</tt> <tt class="py-name">gap_char</tt> <tt class="py-keyword">and</tt> <tt class="py-name">key</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">e_freq_table</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L592"></a><tt class="py-lineno">592</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Expected frequency letters %s "</tt> </tt>
<a name="L593"></a><tt class="py-lineno">593</tt>  <tt class="py-line">                                     <tt class="py-string">"do not match observed %s"</tt> </tt>
<a name="L594"></a><tt class="py-lineno">594</tt>  <tt class="py-line">                                     <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">e_freq_table</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L595"></a><tt class="py-lineno">595</tt>  <tt class="py-line">                                        <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">obs_freq</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-op">[</tt><tt class="py-name">gap_char</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L596"></a><tt class="py-lineno">596</tt>  <tt class="py-line"> </tt>
<a name="L597"></a><tt class="py-lineno">597</tt>  <tt class="py-line">        <tt class="py-name">total_info</tt> <tt class="py-op">=</tt> <tt class="py-number">0.0</tt> </tt>
<a name="L598"></a><tt class="py-lineno">598</tt>  <tt class="py-line"> </tt>
<a name="L599"></a><tt class="py-lineno">599</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">letter</tt> <tt class="py-keyword">in</tt> <tt class="py-name">obs_freq</tt><tt class="py-op">:</tt> </tt>
<a name="L600"></a><tt class="py-lineno">600</tt>  <tt class="py-line">            <tt class="py-name">inner_log</tt> <tt class="py-op">=</tt> <tt class="py-number">0.0</tt> </tt>
<a name="L601"></a><tt class="py-lineno">601</tt>  <tt class="py-line">            <tt class="py-comment"># if we have expected frequencies, modify the log value by them</tt> </tt>
<a name="L602"></a><tt class="py-lineno">602</tt>  <tt class="py-line">            <tt class="py-comment"># gap characters do not have expected frequencies, so they</tt> </tt>
<a name="L603"></a><tt class="py-lineno">603</tt>  <tt class="py-line">            <tt class="py-comment"># should just be the observed frequency.</tt> </tt>
<a name="L604"></a><tt class="py-lineno">604</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">letter</tt> <tt class="py-op">!=</tt> <tt class="py-name">gap_char</tt><tt class="py-op">:</tt> </tt>
<a name="L605"></a><tt class="py-lineno">605</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">e_freq_table</tt><tt class="py-op">:</tt> </tt>
<a name="L606"></a><tt class="py-lineno">606</tt>  <tt class="py-line">                    <tt class="py-name">inner_log</tt> <tt class="py-op">=</tt> <tt class="py-name">obs_freq</tt><tt class="py-op">[</tt><tt class="py-name">letter</tt><tt class="py-op">]</tt> <tt class="py-op">/</tt> <tt class="py-name">e_freq_table</tt><tt class="py-op">[</tt><tt class="py-name">letter</tt><tt class="py-op">]</tt> </tt>
<a name="L607"></a><tt class="py-lineno">607</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L608"></a><tt class="py-lineno">608</tt>  <tt class="py-line">                    <tt class="py-name">inner_log</tt> <tt class="py-op">=</tt> <tt class="py-name">obs_freq</tt><tt class="py-op">[</tt><tt class="py-name">letter</tt><tt class="py-op">]</tt> <tt class="py-op">/</tt> <tt class="py-name">random_expected</tt> </tt>
<a name="L609"></a><tt class="py-lineno">609</tt>  <tt class="py-line">            <tt class="py-comment"># if the observed frequency is zero, we don't add any info to the</tt> </tt>
<a name="L610"></a><tt class="py-lineno">610</tt>  <tt class="py-line">            <tt class="py-comment"># total information content</tt> </tt>
<a name="L611"></a><tt class="py-lineno">611</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">inner_log</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L612"></a><tt class="py-lineno">612</tt>  <tt class="py-line">                <tt class="py-name">letter_info</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">obs_freq</tt><tt class="py-op">[</tt><tt class="py-name">letter</tt><tt class="py-op">]</tt> <tt class="py-op">*</tt> </tt>
<a name="L613"></a><tt class="py-lineno">613</tt>  <tt class="py-line">                               <tt class="py-name">math</tt><tt class="py-op">.</tt><tt class="py-name">log</tt><tt class="py-op">(</tt><tt class="py-name">inner_log</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-name">math</tt><tt class="py-op">.</tt><tt class="py-name">log</tt><tt class="py-op">(</tt><tt class="py-name">log_base</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L614"></a><tt class="py-lineno">614</tt>  <tt class="py-line">                <tt class="py-name">total_info</tt> <tt class="py-op">+=</tt> <tt class="py-name">letter_info</tt> </tt>
<a name="L615"></a><tt class="py-lineno">615</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">total_info</tt> </tt>
</div><a name="L616"></a><tt class="py-lineno">616</tt>  <tt class="py-line"> </tt>
<a name="SummaryInfo.get_column"></a><div id="SummaryInfo.get_column-def"><a name="L617"></a><tt class="py-lineno">617</tt> <a class="py-toggle" href="#" id="SummaryInfo.get_column-toggle" onclick="return toggle('SummaryInfo.get_column');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Align.AlignInfo.SummaryInfo-class.html#get_column">get_column</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">col</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SummaryInfo.get_column-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SummaryInfo.get_column-expanded"><a name="L618"></a><tt class="py-lineno">618</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Deprecate this and implement slicing?</tt> </tt>
<a name="L619"></a><tt class="py-lineno">619</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">,</tt> <tt class="py-name">col</tt><tt class="py-op">]</tt> </tt>
</div></div><a name="L620"></a><tt class="py-lineno">620</tt>  <tt class="py-line"> </tt>
<a name="L621"></a><tt class="py-lineno">621</tt>  <tt class="py-line"> </tt>
<a name="PSSM"></a><div id="PSSM-def"><a name="L622"></a><tt class="py-lineno">622</tt> <a class="py-toggle" href="#" id="PSSM-toggle" onclick="return toggle('PSSM');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Align.AlignInfo.PSSM-class.html">PSSM</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PSSM-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="PSSM-expanded"><a name="L623"></a><tt class="py-lineno">623</tt>  <tt class="py-line">    <tt class="py-docstring">"""Represent a position specific score matrix.</tt> </tt>
<a name="L624"></a><tt class="py-lineno">624</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L625"></a><tt class="py-lineno">625</tt>  <tt class="py-line"><tt class="py-docstring">    This class is meant to make it easy to access the info within a PSSM</tt> </tt>
<a name="L626"></a><tt class="py-lineno">626</tt>  <tt class="py-line"><tt class="py-docstring">    and also make it easy to print out the information in a nice table.</tt> </tt>
<a name="L627"></a><tt class="py-lineno">627</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L628"></a><tt class="py-lineno">628</tt>  <tt class="py-line"><tt class="py-docstring">    Let's say you had an alignment like this::</tt> </tt>
<a name="L629"></a><tt class="py-lineno">629</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L630"></a><tt class="py-lineno">630</tt>  <tt class="py-line"><tt class="py-docstring">        GTATC</tt> </tt>
<a name="L631"></a><tt class="py-lineno">631</tt>  <tt class="py-line"><tt class="py-docstring">        AT--C</tt> </tt>
<a name="L632"></a><tt class="py-lineno">632</tt>  <tt class="py-line"><tt class="py-docstring">        CTGTC</tt> </tt>
<a name="L633"></a><tt class="py-lineno">633</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L634"></a><tt class="py-lineno">634</tt>  <tt class="py-line"><tt class="py-docstring">    The position specific score matrix (when printed) looks like::</tt> </tt>
<a name="L635"></a><tt class="py-lineno">635</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L636"></a><tt class="py-lineno">636</tt>  <tt class="py-line"><tt class="py-docstring">          G A T C</tt> </tt>
<a name="L637"></a><tt class="py-lineno">637</tt>  <tt class="py-line"><tt class="py-docstring">        G 1 1 0 1</tt> </tt>
<a name="L638"></a><tt class="py-lineno">638</tt>  <tt class="py-line"><tt class="py-docstring">        T 0 0 3 0</tt> </tt>
<a name="L639"></a><tt class="py-lineno">639</tt>  <tt class="py-line"><tt class="py-docstring">        A 1 1 0 0</tt> </tt>
<a name="L640"></a><tt class="py-lineno">640</tt>  <tt class="py-line"><tt class="py-docstring">        T 0 0 2 0</tt> </tt>
<a name="L641"></a><tt class="py-lineno">641</tt>  <tt class="py-line"><tt class="py-docstring">        C 0 0 0 3</tt> </tt>
<a name="L642"></a><tt class="py-lineno">642</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L643"></a><tt class="py-lineno">643</tt>  <tt class="py-line"><tt class="py-docstring">    You can access a single element of the PSSM using the following::</tt> </tt>
<a name="L644"></a><tt class="py-lineno">644</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L645"></a><tt class="py-lineno">645</tt>  <tt class="py-line"><tt class="py-docstring">        your_pssm[sequence_number][residue_count_name]</tt> </tt>
<a name="L646"></a><tt class="py-lineno">646</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L647"></a><tt class="py-lineno">647</tt>  <tt class="py-line"><tt class="py-docstring">    For instance, to get the 'T' residue for the second element in the</tt> </tt>
<a name="L648"></a><tt class="py-lineno">648</tt>  <tt class="py-line"><tt class="py-docstring">    above alignment you would need to do:</tt> </tt>
<a name="L649"></a><tt class="py-lineno">649</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L650"></a><tt class="py-lineno">650</tt>  <tt class="py-line"><tt class="py-docstring">    your_pssm[1]['T']</tt> </tt>
<a name="L651"></a><tt class="py-lineno">651</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="PSSM.__init__"></a><div id="PSSM.__init__-def"><a name="L652"></a><tt class="py-lineno">652</tt> <a class="py-toggle" href="#" id="PSSM.__init__-toggle" onclick="return toggle('PSSM.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Align.AlignInfo.PSSM-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">pssm</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PSSM.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="PSSM.__init__-expanded"><a name="L653"></a><tt class="py-lineno">653</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize with pssm data to represent.</tt> </tt>
<a name="L654"></a><tt class="py-lineno">654</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L655"></a><tt class="py-lineno">655</tt>  <tt class="py-line"><tt class="py-docstring">        The pssm passed should be a list with the following structure:</tt> </tt>
<a name="L656"></a><tt class="py-lineno">656</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L657"></a><tt class="py-lineno">657</tt>  <tt class="py-line"><tt class="py-docstring">        list[0] - The letter of the residue being represented (for instance,</tt> </tt>
<a name="L658"></a><tt class="py-lineno">658</tt>  <tt class="py-line"><tt class="py-docstring">        from the example above, the first few list[0]s would be GTAT...</tt> </tt>
<a name="L659"></a><tt class="py-lineno">659</tt>  <tt class="py-line"><tt class="py-docstring">        list[1] - A dictionary with the letter substitutions and counts.</tt> </tt>
<a name="L660"></a><tt class="py-lineno">660</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L661"></a><tt class="py-lineno">661</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-175" class="py-name" targets="Variable Bio.motifs.Motif.pssm=Bio.motifs.Motif-class.html#pssm"><a title="Bio.motifs.Motif.pssm" class="py-name" href="#" onclick="return doclink('link-175', 'pssm', 'link-175');">pssm</a></tt> <tt class="py-op">=</tt> <tt id="link-176" class="py-name"><a title="Bio.motifs.Motif.pssm" class="py-name" href="#" onclick="return doclink('link-176', 'pssm', 'link-175');">pssm</a></tt> </tt>
</div><a name="L662"></a><tt class="py-lineno">662</tt>  <tt class="py-line"> </tt>
<a name="PSSM.__getitem__"></a><div id="PSSM.__getitem__-def"><a name="L663"></a><tt class="py-lineno">663</tt> <a class="py-toggle" href="#" id="PSSM.__getitem__-toggle" onclick="return toggle('PSSM.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Align.AlignInfo.PSSM-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">pos</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PSSM.__getitem__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="PSSM.__getitem__-expanded"><a name="L664"></a><tt class="py-lineno">664</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-177" class="py-name"><a title="Bio.motifs.Motif.pssm" class="py-name" href="#" onclick="return doclink('link-177', 'pssm', 'link-175');">pssm</a></tt><tt class="py-op">[</tt><tt class="py-name">pos</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
</div><a name="L665"></a><tt class="py-lineno">665</tt>  <tt class="py-line"> </tt>
<a name="PSSM.__str__"></a><div id="PSSM.__str__-def"><a name="L666"></a><tt class="py-lineno">666</tt> <a class="py-toggle" href="#" id="PSSM.__str__-toggle" onclick="return toggle('PSSM.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Align.AlignInfo.PSSM-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PSSM.__str__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="PSSM.__str__-expanded"><a name="L667"></a><tt class="py-lineno">667</tt>  <tt class="py-line">        <tt class="py-name">out</tt> <tt class="py-op">=</tt> <tt class="py-string">" "</tt> </tt>
<a name="L668"></a><tt class="py-lineno">668</tt>  <tt class="py-line">        <tt class="py-name">all_residues</tt> <tt class="py-op">=</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-178" class="py-name"><a title="Bio.motifs.Motif.pssm" class="py-name" href="#" onclick="return doclink('link-178', 'pssm', 'link-175');">pssm</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L669"></a><tt class="py-lineno">669</tt>  <tt class="py-line"> </tt>
<a name="L670"></a><tt class="py-lineno">670</tt>  <tt class="py-line">        <tt class="py-comment"># first print out the top header</tt> </tt>
<a name="L671"></a><tt class="py-lineno">671</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">res</tt> <tt class="py-keyword">in</tt> <tt class="py-name">all_residues</tt><tt class="py-op">:</tt> </tt>
<a name="L672"></a><tt class="py-lineno">672</tt>  <tt class="py-line">            <tt class="py-name">out</tt> <tt class="py-op">+=</tt> <tt class="py-string">"   %s"</tt> <tt class="py-op">%</tt> <tt class="py-name">res</tt> </tt>
<a name="L673"></a><tt class="py-lineno">673</tt>  <tt class="py-line">        <tt class="py-name">out</tt> <tt class="py-op">+=</tt> <tt class="py-string">"\n"</tt> </tt>
<a name="L674"></a><tt class="py-lineno">674</tt>  <tt class="py-line"> </tt>
<a name="L675"></a><tt class="py-lineno">675</tt>  <tt class="py-line">        <tt class="py-comment"># for each item, write out the substitutions</tt> </tt>
<a name="L676"></a><tt class="py-lineno">676</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">item</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-179" class="py-name"><a title="Bio.motifs.Motif.pssm" class="py-name" href="#" onclick="return doclink('link-179', 'pssm', 'link-175');">pssm</a></tt><tt class="py-op">:</tt> </tt>
<a name="L677"></a><tt class="py-lineno">677</tt>  <tt class="py-line">            <tt class="py-name">out</tt> <tt class="py-op">+=</tt> <tt class="py-string">"%s "</tt> <tt class="py-op">%</tt> <tt class="py-name">item</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L678"></a><tt class="py-lineno">678</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">res</tt> <tt class="py-keyword">in</tt> <tt class="py-name">all_residues</tt><tt class="py-op">:</tt> </tt>
<a name="L679"></a><tt class="py-lineno">679</tt>  <tt class="py-line">                <tt class="py-name">out</tt> <tt class="py-op">+=</tt> <tt class="py-string">" %.1f"</tt> <tt class="py-op">%</tt> <tt class="py-name">item</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">res</tt><tt class="py-op">]</tt> </tt>
<a name="L680"></a><tt class="py-lineno">680</tt>  <tt class="py-line"> </tt>
<a name="L681"></a><tt class="py-lineno">681</tt>  <tt class="py-line">            <tt class="py-name">out</tt> <tt class="py-op">+=</tt> <tt class="py-string">"\n"</tt> </tt>
<a name="L682"></a><tt class="py-lineno">682</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">out</tt> </tt>
</div><a name="L683"></a><tt class="py-lineno">683</tt>  <tt class="py-line"> </tt>
<a name="PSSM.get_residue"></a><div id="PSSM.get_residue-def"><a name="L684"></a><tt class="py-lineno">684</tt> <a class="py-toggle" href="#" id="PSSM.get_residue-toggle" onclick="return toggle('PSSM.get_residue');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Align.AlignInfo.PSSM-class.html#get_residue">get_residue</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">pos</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PSSM.get_residue-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="PSSM.get_residue-expanded"><a name="L685"></a><tt class="py-lineno">685</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the residue letter at the specified position.</tt> </tt>
<a name="L686"></a><tt class="py-lineno">686</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L687"></a><tt class="py-lineno">687</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-180" class="py-name"><a title="Bio.motifs.Motif.pssm" class="py-name" href="#" onclick="return doclink('link-180', 'pssm', 'link-175');">pssm</a></tt><tt class="py-op">[</tt><tt class="py-name">pos</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
</div></div><a name="L688"></a><tt class="py-lineno">688</tt>  <tt class="py-line"> </tt>
<a name="L689"></a><tt class="py-lineno">689</tt>  <tt class="py-line"> </tt>
<a name="print_info_content"></a><div id="print_info_content-def"><a name="L690"></a><tt class="py-lineno">690</tt> <a class="py-toggle" href="#" id="print_info_content-toggle" onclick="return toggle('print_info_content');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Align.AlignInfo-module.html#print_info_content">print_info_content</a><tt class="py-op">(</tt><tt class="py-param">summary_info</tt><tt class="py-op">,</tt> <tt class="py-param">fout</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">rep_record</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="print_info_content-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="print_info_content-expanded"><a name="L691"></a><tt class="py-lineno">691</tt>  <tt class="py-line">    <tt class="py-docstring">""" Three column output: position, aa in representative sequence,</tt> </tt>
<a name="L692"></a><tt class="py-lineno">692</tt>  <tt class="py-line"><tt class="py-docstring">        ic_vector value"""</tt> </tt>
<a name="L693"></a><tt class="py-lineno">693</tt>  <tt class="py-line">    <tt class="py-name">fout</tt> <tt class="py-op">=</tt> <tt class="py-name">fout</tt> <tt class="py-keyword">or</tt> <tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">stdout</tt> </tt>
<a name="L694"></a><tt class="py-lineno">694</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">summary_info</tt><tt class="py-op">.</tt><tt class="py-name">ic_vector</tt><tt class="py-op">:</tt> </tt>
<a name="L695"></a><tt class="py-lineno">695</tt>  <tt class="py-line">        <tt class="py-name">summary_info</tt><tt class="py-op">.</tt><tt id="link-181" class="py-name" targets="Method Bio.Align.AlignInfo.SummaryInfo.information_content()=Bio.Align.AlignInfo.SummaryInfo-class.html#information_content"><a title="Bio.Align.AlignInfo.SummaryInfo.information_content" class="py-name" href="#" onclick="return doclink('link-181', 'information_content', 'link-181');">information_content</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L696"></a><tt class="py-lineno">696</tt>  <tt class="py-line">    <tt class="py-name">rep_sequence</tt> <tt class="py-op">=</tt> <tt class="py-name">summary_info</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">[</tt><tt class="py-name">rep_record</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-182" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-182', 'seq', 'link-16');">seq</a></tt> </tt>
<a name="L697"></a><tt class="py-lineno">697</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">pos</tt> <tt class="py-keyword">in</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt class="py-name">summary_info</tt><tt class="py-op">.</tt><tt class="py-name">ic_vector</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L698"></a><tt class="py-lineno">698</tt>  <tt class="py-line">        <tt class="py-name">fout</tt><tt class="py-op">.</tt><tt id="link-183" class="py-name" targets="Function Bio.AlignIO.write()=Bio.AlignIO-module.html#write,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.write()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#write,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.write()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#write,Method Bio.Phylo.CDAOIO.Writer.write()=Bio.Phylo.CDAOIO.Writer-class.html#write,Function Bio.Phylo.CDAOIO.write()=Bio.Phylo.CDAOIO-module.html#write,Method Bio.Phylo.NeXMLIO.Writer.write()=Bio.Phylo.NeXMLIO.Writer-class.html#write,Function Bio.Phylo.NeXMLIO.write()=Bio.Phylo.NeXMLIO-module.html#write,Method Bio.Phylo.NewickIO.Writer.write()=Bio.Phylo.NewickIO.Writer-class.html#write,Function Bio.Phylo.NewickIO.write()=Bio.Phylo.NewickIO-module.html#write,Function Bio.Phylo.NexusIO.write()=Bio.Phylo.NexusIO-module.html#write,Method Bio.Phylo.PhyloXMLIO.Writer.write()=Bio.Phylo.PhyloXMLIO.Writer-class.html#write,Function Bio.Phylo.PhyloXMLIO.write()=Bio.Phylo.PhyloXMLIO-module.html#write,Function Bio.Phylo._io.write()=Bio.Phylo._io-module.html#write,Function Bio.SearchIO.write()=Bio.SearchIO-module.html#write,Function Bio.SeqIO.write()=Bio.SeqIO-module.html#write,Method Bio.bgzf.BgzfWriter.write()=Bio.bgzf.BgzfWriter-class.html#write,Function Bio.motifs.jaspar.write()=Bio.motifs.jaspar-module.html#write,Function Bio.motifs.transfac.write()=Bio.motifs.transfac-module.html#write,Function Bio.motifs.write()=Bio.motifs-module.html#write"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-183', 'write', 'link-183');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"%d %s %.3f\n"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">pos</tt><tt class="py-op">,</tt> <tt class="py-name">rep_sequence</tt><tt class="py-op">[</tt><tt class="py-name">pos</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L699"></a><tt class="py-lineno">699</tt>  <tt class="py-line">                   <tt class="py-name">summary_info</tt><tt class="py-op">.</tt><tt class="py-name">ic_vector</tt><tt class="py-op">[</tt><tt class="py-name">pos</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L700"></a><tt class="py-lineno">700</tt>  <tt class="py-line"> </tt>
<a name="L701"></a><tt class="py-lineno">701</tt>  <tt class="py-line"><tt class="py-keyword">if</tt> <tt class="py-name">__name__</tt> <tt class="py-op">==</tt> <tt class="py-string">"__main__"</tt><tt class="py-op">:</tt> </tt>
<a name="L702"></a><tt class="py-lineno">702</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Quick test"</tt><tt class="py-op">)</tt> </tt>
<a name="L703"></a><tt class="py-lineno">703</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-184" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-184', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-185" class="py-name" targets="Package Bio.AlignIO=Bio.AlignIO-module.html"><a title="Bio.AlignIO" class="py-name" href="#" onclick="return doclink('link-185', 'AlignIO', 'link-185');">AlignIO</a></tt> </tt>
<a name="L704"></a><tt class="py-lineno">704</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-186" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-186', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-187" class="py-name" targets="Package Bio.Align=Bio.Align-module.html"><a title="Bio.Align" class="py-name" href="#" onclick="return doclink('link-187', 'Align', 'link-187');">Align</a></tt><tt class="py-op">.</tt><tt id="link-188" class="py-name" targets="Module Bio.Align.Generic=Bio.Align.Generic-module.html"><a title="Bio.Align.Generic" class="py-name" href="#" onclick="return doclink('link-188', 'Generic', 'link-188');">Generic</a></tt> <tt class="py-keyword">import</tt> <tt id="link-189" class="py-name" targets="Class Bio.Align.Generic.Alignment=Bio.Align.Generic.Alignment-class.html,Class Bio.Blast.Record.Alignment=Bio.Blast.Record.Alignment-class.html,Class Bio.Wise.psw.Alignment=Bio.Wise.psw.Alignment-class.html"><a title="Bio.Align.Generic.Alignment
Bio.Blast.Record.Alignment
Bio.Wise.psw.Alignment" class="py-name" href="#" onclick="return doclink('link-189', 'Alignment', 'link-189');">Alignment</a></tt> </tt>
<a name="L705"></a><tt class="py-lineno">705</tt>  <tt class="py-line"> </tt>
<a name="L706"></a><tt class="py-lineno">706</tt>  <tt class="py-line">    <tt class="py-name">filename</tt> <tt class="py-op">=</tt> <tt class="py-string">"../../Tests/GFF/multi.fna"</tt> </tt>
<a name="L707"></a><tt class="py-lineno">707</tt>  <tt class="py-line">    <tt id="link-190" class="py-name" targets="Method Bio.Align.Generic.Alignment.format()=Bio.Align.Generic.Alignment-class.html#format,Method Bio.Motif._Motif.Motif.format()=Bio.Motif._Motif.Motif-class.html#format,Method Bio.Phylo.BaseTree.Tree.format()=Bio.Phylo.BaseTree.Tree-class.html#format,Method Bio.Restriction.Restriction.RestrictionBatch.format()=Bio.Restriction.Restriction.RestrictionBatch-class.html#format,Method Bio.SeqRecord.SeqRecord.format()=Bio.SeqRecord.SeqRecord-class.html#format,Method Bio.motifs.Motif.format()=Bio.motifs.Motif-class.html#format"><a title="Bio.Align.Generic.Alignment.format
Bio.Motif._Motif.Motif.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-190', 'format', 'link-190');">format</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"fasta"</tt> </tt>
<a name="L708"></a><tt class="py-lineno">708</tt>  <tt class="py-line">    <tt class="py-name">expected</tt> <tt class="py-op">=</tt> <tt id="link-191" class="py-name"><a title="Bio.SubsMat.FreqTable
Bio.SubsMat.FreqTable.FreqTable" class="py-name" href="#" onclick="return doclink('link-191', 'FreqTable', 'link-10');">FreqTable</a></tt><tt class="py-op">.</tt><tt id="link-192" class="py-name"><a title="Bio.SubsMat.FreqTable
Bio.SubsMat.FreqTable.FreqTable" class="py-name" href="#" onclick="return doclink('link-192', 'FreqTable', 'link-10');">FreqTable</a></tt><tt class="py-op">(</tt><tt class="py-op">{</tt><tt class="py-string">"A"</tt><tt class="py-op">:</tt> <tt class="py-number">0.25</tt><tt class="py-op">,</tt> <tt class="py-string">"G"</tt><tt class="py-op">:</tt> <tt class="py-number">0.25</tt><tt class="py-op">,</tt> <tt class="py-string">"T"</tt><tt class="py-op">:</tt> <tt class="py-number">0.25</tt><tt class="py-op">,</tt> <tt class="py-string">"C"</tt><tt class="py-op">:</tt> <tt class="py-number">0.25</tt><tt class="py-op">}</tt><tt class="py-op">,</tt> </tt>
<a name="L709"></a><tt class="py-lineno">709</tt>  <tt class="py-line">                                   <tt id="link-193" class="py-name"><a title="Bio.SubsMat.FreqTable
Bio.SubsMat.FreqTable.FreqTable" class="py-name" href="#" onclick="return doclink('link-193', 'FreqTable', 'link-10');">FreqTable</a></tt><tt class="py-op">.</tt><tt id="link-194" class="py-name" targets="Variable Bio.SubsMat.FreqTable.FREQ=Bio.SubsMat.FreqTable-module.html#FREQ"><a title="Bio.SubsMat.FreqTable.FREQ" class="py-name" href="#" onclick="return doclink('link-194', 'FREQ', 'link-194');">FREQ</a></tt><tt class="py-op">,</tt> </tt>
<a name="L710"></a><tt class="py-lineno">710</tt>  <tt class="py-line">                                   <tt id="link-195" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-195', 'IUPAC', 'link-4');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-196" class="py-name" targets="Variable Bio.Alphabet.IUPAC.unambiguous_dna=Bio.Alphabet.IUPAC-module.html#unambiguous_dna"><a title="Bio.Alphabet.IUPAC.unambiguous_dna" class="py-name" href="#" onclick="return doclink('link-196', 'unambiguous_dna', 'link-196');">unambiguous_dna</a></tt><tt class="py-op">)</tt> </tt>
<a name="L711"></a><tt class="py-lineno">711</tt>  <tt class="py-line"> </tt>
<a name="L712"></a><tt class="py-lineno">712</tt>  <tt class="py-line">    <tt class="py-name">alignment</tt> <tt class="py-op">=</tt> <tt id="link-197" class="py-name"><a title="Bio.AlignIO" class="py-name" href="#" onclick="return doclink('link-197', 'AlignIO', 'link-185');">AlignIO</a></tt><tt class="py-op">.</tt><tt id="link-198" class="py-name" targets="Function Bio.Affy.CelFile.read()=Bio.Affy.CelFile-module.html#read,Function Bio.AlignIO.read()=Bio.AlignIO-module.html#read,Function Bio.Blast.NCBIXML.read()=Bio.Blast.NCBIXML-module.html#read,Function Bio.Cluster.read()=Bio.Cluster-module.html#read,Function Bio.Compass.read()=Bio.Compass-module.html#read,Function Bio.Emboss.Primer3.read()=Bio.Emboss.Primer3-module.html#read,Function Bio.Emboss.PrimerSearch.read()=Bio.Emboss.PrimerSearch-module.html#read,Method Bio.Entrez.Parser.DataHandler.read()=Bio.Entrez.Parser.DataHandler-class.html#read,Function Bio.Entrez.read()=Bio.Entrez-module.html#read,Function Bio.ExPASy.Enzyme.read()=Bio.ExPASy.Enzyme-module.html#read,Function Bio.ExPASy.Prodoc.read()=Bio.ExPASy.Prodoc-module.html#read,Function Bio.ExPASy.Prosite.read()=Bio.ExPASy.Prosite-module.html#read,Function Bio.ExPASy.ScanProsite.read()=Bio.ExPASy.ScanProsite-module.html#read,Method Bio.File.UndoHandle.read()=Bio.File.UndoHandle-class.html#read,Function Bio.KEGG.KGML.KGML_parser.read()=Bio.KEGG.KGML.KGML_parser-module.html#read,Function Bio.Medline.read()=Bio.Medline-module.html#read,Function Bio.Motif.Parsers.AlignAce.read()=Bio.Motif.Parsers.AlignAce-module.html#read,Function Bio.Motif.Parsers.MAST.read()=Bio.Motif.Parsers.MAST-module.html#read,Function Bio.Motif.Parsers.MEME.read()=Bio.Motif.Parsers.MEME-module.html#read,Function Bio.Motif.read()=Bio.Motif-module.html#read,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.read()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#read,Method Bio.Nexus.Nexus.Nexus.read()=Bio.Nexus.Nexus.Nexus-class.html#read,Function Bio.Phylo.PAML.baseml.read()=Bio.Phylo.PAML.baseml-module.html#read,Function Bio.Phylo.PAML.codeml.read()=Bio.Phylo.PAML.codeml-module.html#read,Function Bio.Phylo.PAML.yn00.read()=Bio.Phylo.PAML.yn00-module.html#read,Method Bio.Phylo.PhyloXMLIO.Parser.read()=Bio.Phylo.PhyloXMLIO.Parser-class.html#read,Function Bio.Phylo.PhyloXMLIO.read()=Bio.Phylo.PhyloXMLIO-module.html#read,Function Bio.Phylo._io.read()=Bio.Phylo._io-module.html#read,Function Bio.PopGen.GenePop.FileParser.read()=Bio.PopGen.GenePop.FileParser-module.html#read,Function Bio.PopGen.GenePop.LargeFileParser.read()=Bio.PopGen.GenePop.LargeFileParser-module.html#read,Function Bio.PopGen.GenePop.read()=Bio.PopGen.GenePop-module.html#read,Function Bio.SearchIO.read()=Bio.SearchIO-module.html#read,Method Bio.SeqIO.SffIO._AddTellHandle.read()=Bio.SeqIO.SffIO._AddTellHandle-class.html#read,Function Bio.SeqIO.read()=Bio.SeqIO-module.html#read,Function Bio.Sequencing.Ace.read()=Bio.Sequencing.Ace-module.html#read,Function Bio.Sequencing.Phd.read()=Bio.Sequencing.Phd-module.html#read,Function Bio.SwissProt.read()=Bio.SwissProt-module.html#read,Function Bio.UniGene.read()=Bio.UniGene-module.html#read,Method Bio.bgzf.BgzfReader.read()=Bio.bgzf.BgzfReader-class.html#read,Function Bio.motifs.alignace.read()=Bio.motifs.alignace-module.html#read,Function Bio.motifs.jaspar.read()=Bio.motifs.jaspar-module.html#read,Function Bio.motifs.mast.read()=Bio.motifs.mast-module.html#read,Function Bio.motifs.meme.read()=Bio.motifs.meme-module.html#read,Function Bio.motifs.read()=Bio.motifs-module.html#read,Function Bio.motifs.transfac.read()=Bio.motifs.transfac-module.html#read"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-198', 'read', 'link-198');">read</a></tt><tt class="py-op">(</tt><tt id="link-199" class="py-name" targets="Function Bio.bgzf.open()=Bio.bgzf-module.html#open"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-199', 'open', 'link-199');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-200" class="py-name"><a title="Bio.Align.Generic.Alignment.format
Bio.Motif._Motif.Motif.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-200', 'format', 'link-190');">format</a></tt><tt class="py-op">)</tt> </tt>
<a name="L713"></a><tt class="py-lineno">713</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">alignment</tt><tt class="py-op">:</tt> </tt>
<a name="L714"></a><tt class="py-lineno">714</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-201" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-201', 'seq', 'link-16');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L715"></a><tt class="py-lineno">715</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"="</tt> <tt class="py-op">*</tt> <tt class="py-name">alignment</tt><tt class="py-op">.</tt><tt id="link-202" class="py-name"><a title="Bio.Align.Generic.Alignment.get_alignment_length" class="py-name" href="#" onclick="return doclink('link-202', 'get_alignment_length', 'link-14');">get_alignment_length</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L716"></a><tt class="py-lineno">716</tt>  <tt class="py-line"> </tt>
<a name="L717"></a><tt class="py-lineno">717</tt>  <tt class="py-line">    <tt class="py-name">summary</tt> <tt class="py-op">=</tt> <tt id="link-203" class="py-name" targets="Class Bio.Align.AlignInfo.SummaryInfo=Bio.Align.AlignInfo.SummaryInfo-class.html"><a title="Bio.Align.AlignInfo.SummaryInfo" class="py-name" href="#" onclick="return doclink('link-203', 'SummaryInfo', 'link-203');">SummaryInfo</a></tt><tt class="py-op">(</tt><tt class="py-name">alignment</tt><tt class="py-op">)</tt> </tt>
<a name="L718"></a><tt class="py-lineno">718</tt>  <tt class="py-line">    <tt id="link-204" class="py-name"><a title="Bio.Motif._Motif.Motif.consensus
Bio.Nexus.Trees.consensus
Bio.motifs.Motif.consensus
Bio.motifs.matrix.GenericPositionMatrix.consensus" class="py-name" href="#" onclick="return doclink('link-204', 'consensus', 'link-13');">consensus</a></tt> <tt class="py-op">=</tt> <tt class="py-name">summary</tt><tt class="py-op">.</tt><tt id="link-205" class="py-name"><a title="Bio.Align.AlignInfo.SummaryInfo.dumb_consensus" class="py-name" href="#" onclick="return doclink('link-205', 'dumb_consensus', 'link-127');">dumb_consensus</a></tt><tt class="py-op">(</tt><tt class="py-name">ambiguous</tt><tt class="py-op">=</tt><tt class="py-string">"N"</tt><tt class="py-op">)</tt> </tt>
<a name="L719"></a><tt class="py-lineno">719</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt id="link-206" class="py-name"><a title="Bio.Motif._Motif.Motif.consensus
Bio.Nexus.Trees.consensus
Bio.motifs.Motif.consensus
Bio.motifs.matrix.GenericPositionMatrix.consensus" class="py-name" href="#" onclick="return doclink('link-206', 'consensus', 'link-13');">consensus</a></tt><tt class="py-op">)</tt> </tt>
<a name="L720"></a><tt class="py-lineno">720</tt>  <tt class="py-line">    <tt id="link-207" class="py-name"><a title="Bio.Motif._Motif.Motif.consensus
Bio.Nexus.Trees.consensus
Bio.motifs.Motif.consensus
Bio.motifs.matrix.GenericPositionMatrix.consensus" class="py-name" href="#" onclick="return doclink('link-207', 'consensus', 'link-13');">consensus</a></tt> <tt class="py-op">=</tt> <tt class="py-name">summary</tt><tt class="py-op">.</tt><tt id="link-208" class="py-name" targets="Method Bio.Align.AlignInfo.SummaryInfo.gap_consensus()=Bio.Align.AlignInfo.SummaryInfo-class.html#gap_consensus"><a title="Bio.Align.AlignInfo.SummaryInfo.gap_consensus" class="py-name" href="#" onclick="return doclink('link-208', 'gap_consensus', 'link-208');">gap_consensus</a></tt><tt class="py-op">(</tt><tt class="py-name">ambiguous</tt><tt class="py-op">=</tt><tt class="py-string">"N"</tt><tt class="py-op">)</tt> </tt>
<a name="L721"></a><tt class="py-lineno">721</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt id="link-209" class="py-name"><a title="Bio.Motif._Motif.Motif.consensus
Bio.Nexus.Trees.consensus
Bio.motifs.Motif.consensus
Bio.motifs.matrix.GenericPositionMatrix.consensus" class="py-name" href="#" onclick="return doclink('link-209', 'consensus', 'link-13');">consensus</a></tt><tt class="py-op">)</tt> </tt>
<a name="L722"></a><tt class="py-lineno">722</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">)</tt> </tt>
<a name="L723"></a><tt class="py-lineno">723</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">summary</tt><tt class="py-op">.</tt><tt id="link-210" class="py-name" targets="Method Bio.Align.AlignInfo.SummaryInfo.pos_specific_score_matrix()=Bio.Align.AlignInfo.SummaryInfo-class.html#pos_specific_score_matrix"><a title="Bio.Align.AlignInfo.SummaryInfo.pos_specific_score_matrix" class="py-name" href="#" onclick="return doclink('link-210', 'pos_specific_score_matrix', 'link-210');">pos_specific_score_matrix</a></tt><tt class="py-op">(</tt><tt class="py-name">chars_to_ignore</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-string">'-'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L724"></a><tt class="py-lineno">724</tt>  <tt class="py-line">                                            <tt class="py-name">axis_seq</tt><tt class="py-op">=</tt><tt id="link-211" class="py-name"><a title="Bio.Motif._Motif.Motif.consensus
Bio.Nexus.Trees.consensus
Bio.motifs.Motif.consensus
Bio.motifs.matrix.GenericPositionMatrix.consensus" class="py-name" href="#" onclick="return doclink('link-211', 'consensus', 'link-13');">consensus</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L725"></a><tt class="py-lineno">725</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">)</tt> </tt>
<a name="L726"></a><tt class="py-lineno">726</tt>  <tt class="py-line">    <tt class="py-comment"># Have a generic alphabet, without a declared gap char, so must tell</tt> </tt>
<a name="L727"></a><tt class="py-lineno">727</tt>  <tt class="py-line">    <tt class="py-comment"># provide the frequencies and chars to ignore explicitly.</tt> </tt>
<a name="L728"></a><tt class="py-lineno">728</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">summary</tt><tt class="py-op">.</tt><tt id="link-212" class="py-name"><a title="Bio.Align.AlignInfo.SummaryInfo.information_content" class="py-name" href="#" onclick="return doclink('link-212', 'information_content', 'link-181');">information_content</a></tt><tt class="py-op">(</tt><tt class="py-name">e_freq_table</tt><tt class="py-op">=</tt><tt class="py-name">expected</tt><tt class="py-op">,</tt> </tt>
<a name="L729"></a><tt class="py-lineno">729</tt>  <tt class="py-line">                                      <tt class="py-name">chars_to_ignore</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-string">'-'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L730"></a><tt class="py-lineno">730</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">)</tt> </tt>
<a name="L731"></a><tt class="py-lineno">731</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Trying a protein sequence with gaps and stops"</tt><tt class="py-op">)</tt> </tt>
<a name="L732"></a><tt class="py-lineno">732</tt>  <tt class="py-line"> </tt>
<a name="L733"></a><tt class="py-lineno">733</tt>  <tt class="py-line">    <tt class="py-name">alpha</tt> <tt class="py-op">=</tt> <tt id="link-213" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-213', 'Alphabet', 'link-1');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-214" class="py-name" targets="Class Bio.Alphabet.HasStopCodon=Bio.Alphabet.HasStopCodon-class.html"><a title="Bio.Alphabet.HasStopCodon" class="py-name" href="#" onclick="return doclink('link-214', 'HasStopCodon', 'link-214');">HasStopCodon</a></tt><tt class="py-op">(</tt><tt id="link-215" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-215', 'Alphabet', 'link-1');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-216" class="py-name"><a title="Bio.Alphabet.Gapped" class="py-name" href="#" onclick="return doclink('link-216', 'Gapped', 'link-115');">Gapped</a></tt><tt class="py-op">(</tt><tt id="link-217" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-217', 'Alphabet', 'link-1');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-218" class="py-name"><a title="Bio.Alphabet.generic_protein" class="py-name" href="#" onclick="return doclink('link-218', 'generic_protein', 'link-95');">generic_protein</a></tt><tt class="py-op">,</tt> <tt class="py-string">"-"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"*"</tt><tt class="py-op">)</tt> </tt>
<a name="L734"></a><tt class="py-lineno">734</tt>  <tt class="py-line">    <tt id="link-219" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-219', 'a', 'link-45');">a</a></tt> <tt class="py-op">=</tt> <tt id="link-220" class="py-name"><a title="Bio.Align.Generic.Alignment
Bio.Blast.Record.Alignment
Bio.Wise.psw.Alignment" class="py-name" href="#" onclick="return doclink('link-220', 'Alignment', 'link-189');">Alignment</a></tt><tt class="py-op">(</tt><tt class="py-name">alpha</tt><tt class="py-op">)</tt> </tt>
<a name="L735"></a><tt class="py-lineno">735</tt>  <tt class="py-line">    <tt id="link-221" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-221', 'a', 'link-45');">a</a></tt><tt class="py-op">.</tt><tt id="link-222" class="py-name" targets="Method Bio.Align.Generic.Alignment.add_sequence()=Bio.Align.Generic.Alignment-class.html#add_sequence,Method Bio.Align.MultipleSeqAlignment.add_sequence()=Bio.Align.MultipleSeqAlignment-class.html#add_sequence,Method Bio.Nexus.Nexus.Nexus.add_sequence()=Bio.Nexus.Nexus.Nexus-class.html#add_sequence"><a title="Bio.Align.Generic.Alignment.add_sequence
Bio.Align.MultipleSeqAlignment.add_sequence
Bio.Nexus.Nexus.Nexus.add_sequence" class="py-name" href="#" onclick="return doclink('link-222', 'add_sequence', 'link-222');">add_sequence</a></tt><tt class="py-op">(</tt><tt class="py-string">"ID001"</tt><tt class="py-op">,</tt> <tt class="py-string">"MHQAIFIYQIGYP*LKSGYIQSIRSPEYDNW-"</tt><tt class="py-op">)</tt> </tt>
<a name="L736"></a><tt class="py-lineno">736</tt>  <tt class="py-line">    <tt id="link-223" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-223', 'a', 'link-45');">a</a></tt><tt class="py-op">.</tt><tt id="link-224" class="py-name"><a title="Bio.Align.Generic.Alignment.add_sequence
Bio.Align.MultipleSeqAlignment.add_sequence
Bio.Nexus.Nexus.Nexus.add_sequence" class="py-name" href="#" onclick="return doclink('link-224', 'add_sequence', 'link-222');">add_sequence</a></tt><tt class="py-op">(</tt><tt class="py-string">"ID002"</tt><tt class="py-op">,</tt> <tt class="py-string">"MH--IFIYQIGYAYLKSGYIQSIRSPEY-NW*"</tt><tt class="py-op">)</tt> </tt>
<a name="L737"></a><tt class="py-lineno">737</tt>  <tt class="py-line">    <tt id="link-225" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-225', 'a', 'link-45');">a</a></tt><tt class="py-op">.</tt><tt id="link-226" class="py-name"><a title="Bio.Align.Generic.Alignment.add_sequence
Bio.Align.MultipleSeqAlignment.add_sequence
Bio.Nexus.Nexus.Nexus.add_sequence" class="py-name" href="#" onclick="return doclink('link-226', 'add_sequence', 'link-222');">add_sequence</a></tt><tt class="py-op">(</tt><tt class="py-string">"ID003"</tt><tt class="py-op">,</tt> <tt class="py-string">"MHQAIFIYQIGYPYLKSGYIQSIRSPEYDNW*"</tt><tt class="py-op">)</tt> </tt>
<a name="L738"></a><tt class="py-lineno">738</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt id="link-227" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-227', 'a', 'link-45');">a</a></tt><tt class="py-op">)</tt> </tt>
<a name="L739"></a><tt class="py-lineno">739</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"="</tt> <tt class="py-op">*</tt> <tt id="link-228" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-228', 'a', 'link-45');">a</a></tt><tt class="py-op">.</tt><tt id="link-229" class="py-name"><a title="Bio.Align.Generic.Alignment.get_alignment_length" class="py-name" href="#" onclick="return doclink('link-229', 'get_alignment_length', 'link-14');">get_alignment_length</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L740"></a><tt class="py-lineno">740</tt>  <tt class="py-line"> </tt>
<a name="L741"></a><tt class="py-lineno">741</tt>  <tt class="py-line">    <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt id="link-230" class="py-name"><a title="Bio.Align.AlignInfo.SummaryInfo" class="py-name" href="#" onclick="return doclink('link-230', 'SummaryInfo', 'link-203');">SummaryInfo</a></tt><tt class="py-op">(</tt><tt id="link-231" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-231', 'a', 'link-45');">a</a></tt><tt class="py-op">)</tt> </tt>
<a name="L742"></a><tt class="py-lineno">742</tt>  <tt class="py-line">    <tt class="py-name">c</tt> <tt class="py-op">=</tt> <tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-232" class="py-name"><a title="Bio.Align.AlignInfo.SummaryInfo.dumb_consensus" class="py-name" href="#" onclick="return doclink('link-232', 'dumb_consensus', 'link-127');">dumb_consensus</a></tt><tt class="py-op">(</tt><tt class="py-name">ambiguous</tt><tt class="py-op">=</tt><tt class="py-string">"X"</tt><tt class="py-op">)</tt> </tt>
<a name="L743"></a><tt class="py-lineno">743</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">)</tt> </tt>
<a name="L744"></a><tt class="py-lineno">744</tt>  <tt class="py-line">    <tt class="py-name">c</tt> <tt class="py-op">=</tt> <tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-233" class="py-name"><a title="Bio.Align.AlignInfo.SummaryInfo.gap_consensus" class="py-name" href="#" onclick="return doclink('link-233', 'gap_consensus', 'link-208');">gap_consensus</a></tt><tt class="py-op">(</tt><tt class="py-name">ambiguous</tt><tt class="py-op">=</tt><tt class="py-string">"X"</tt><tt class="py-op">)</tt> </tt>
<a name="L745"></a><tt class="py-lineno">745</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">)</tt> </tt>
<a name="L746"></a><tt class="py-lineno">746</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">)</tt> </tt>
<a name="L747"></a><tt class="py-lineno">747</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-234" class="py-name"><a title="Bio.Align.AlignInfo.SummaryInfo.pos_specific_score_matrix" class="py-name" href="#" onclick="return doclink('link-234', 'pos_specific_score_matrix', 'link-210');">pos_specific_score_matrix</a></tt><tt class="py-op">(</tt><tt class="py-name">chars_to_ignore</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-string">'-'</tt><tt class="py-op">,</tt> <tt class="py-string">'*'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">axis_seq</tt><tt class="py-op">=</tt><tt class="py-name">c</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L748"></a><tt class="py-lineno">748</tt>  <tt class="py-line"> </tt>
<a name="L749"></a><tt class="py-lineno">749</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-235" class="py-name"><a title="Bio.Align.AlignInfo.SummaryInfo.information_content" class="py-name" href="#" onclick="return doclink('link-235', 'information_content', 'link-181');">information_content</a></tt><tt class="py-op">(</tt><tt class="py-name">chars_to_ignore</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-string">'-'</tt><tt class="py-op">,</tt> <tt class="py-string">'*'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L750"></a><tt class="py-lineno">750</tt>  <tt class="py-line"> </tt>
<a name="L751"></a><tt class="py-lineno">751</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Done"</tt><tt class="py-op">)</tt> </tt>
<a name="L752"></a><tt class="py-lineno">752</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:05:00 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
