<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.SearchIO.ExonerateIO.exonerate_text</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.SearchIO-module.html">Package&nbsp;SearchIO</a> ::
        <a href="Bio.SearchIO.ExonerateIO-module.html">Package&nbsp;ExonerateIO</a> ::
        Module&nbsp;exonerate_text
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.SearchIO.ExonerateIO.exonerate_text-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.SearchIO.ExonerateIO.exonerate_text-module.html">Module Bio.SearchIO.ExonerateIO.exonerate_text</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2012 by Wibowo Arindrarto.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-docstring">"""Bio.SearchIO parser for Exonerate plain text output format."""</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">re</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">itertools</tt> <tt class="py-keyword">import</tt> <tt class="py-name">chain</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-1', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt id="link-2" class="py-name" targets="Function Bio._py3k._bytes_to_string()=Bio._py3k-module.html#_bytes_to_string"><a title="Bio._py3k._bytes_to_string" class="py-name" href="#" onclick="return doclink('link-2', '_bytes_to_string', 'link-2');">_bytes_to_string</a></tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-3" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-3', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-4" class="py-name"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-4', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">zip</tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-5" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-5', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-6" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-6', 'Alphabet', 'link-6');">Alphabet</a></tt> <tt class="py-keyword">import</tt> <tt id="link-7" class="py-name" targets="Variable Bio.Alphabet.generic_protein=Bio.Alphabet-module.html#generic_protein"><a title="Bio.Alphabet.generic_protein" class="py-name" href="#" onclick="return doclink('link-7', 'generic_protein', 'link-7');">generic_protein</a></tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt><tt id="link-8" class="py-name" targets="Module Bio.SearchIO.ExonerateIO._base=Bio.SearchIO.ExonerateIO._base-module.html,Module Bio.SearchIO.HmmerIO._base=Bio.SearchIO.HmmerIO._base-module.html,Module Bio.SearchIO._model._base=Bio.SearchIO._model._base-module.html"><a title="Bio.SearchIO.ExonerateIO._base
Bio.SearchIO.HmmerIO._base
Bio.SearchIO._model._base" class="py-name" href="#" onclick="return doclink('link-8', '_base', 'link-8');">_base</a></tt> <tt class="py-keyword">import</tt> <tt id="link-9" class="py-name" targets="Class Bio.SearchIO.ExonerateIO._base._BaseExonerateParser=Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateParser" class="py-name" href="#" onclick="return doclink('link-9', '_BaseExonerateParser', 'link-9');">_BaseExonerateParser</a></tt><tt class="py-op">,</tt> <tt id="link-10" class="py-name" targets="Class Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer=Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer-class.html"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer" class="py-name" href="#" onclick="return doclink('link-10', '_BaseExonerateIndexer', 'link-10');">_BaseExonerateIndexer</a></tt><tt class="py-op">,</tt> <tt id="link-11" class="py-name" targets="Variable Bio.SearchIO.ExonerateIO._base._STRAND_MAP=Bio.SearchIO.ExonerateIO._base-module.html#_STRAND_MAP"><a title="Bio.SearchIO.ExonerateIO._base._STRAND_MAP" class="py-name" href="#" onclick="return doclink('link-11', '_STRAND_MAP', 'link-11');">_STRAND_MAP</a></tt><tt class="py-op">,</tt> \ </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line">        <tt id="link-12" class="py-name" targets="Function Bio.SearchIO.ExonerateIO._base._parse_hit_or_query_line()=Bio.SearchIO.ExonerateIO._base-module.html#_parse_hit_or_query_line"><a title="Bio.SearchIO.ExonerateIO._base._parse_hit_or_query_line" class="py-name" href="#" onclick="return doclink('link-12', '_parse_hit_or_query_line', 'link-12');">_parse_hit_or_query_line</a></tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt><tt id="link-13" class="py-name" targets="Module Bio.SearchIO.ExonerateIO.exonerate_vulgar=Bio.SearchIO.ExonerateIO.exonerate_vulgar-module.html"><a title="Bio.SearchIO.ExonerateIO.exonerate_vulgar" class="py-name" href="#" onclick="return doclink('link-13', 'exonerate_vulgar', 'link-13');">exonerate_vulgar</a></tt> <tt class="py-keyword">import</tt> <tt id="link-14" class="py-name" targets="Variable Bio.SearchIO.ExonerateIO.exonerate_vulgar._RE_VULGAR=Bio.SearchIO.ExonerateIO.exonerate_vulgar-module.html#_RE_VULGAR"><a title="Bio.SearchIO.ExonerateIO.exonerate_vulgar._RE_VULGAR" class="py-name" href="#" onclick="return doclink('link-14', '_RE_VULGAR', 'link-14');">_RE_VULGAR</a></tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-name">__all__</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-string">'ExonerateTextParser'</tt><tt class="py-op">,</tt> <tt class="py-string">'ExonerateTextIndexer'</tt><tt class="py-op">)</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-comment"># for capturing sequences in alignment blocks</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-comment"># e.g. ' 529 : ATCCCTTATCTCTTTATCTTGTA :    472'</tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"><tt id="link-15" class="py-name" targets="Variable Bio.SearchIO.ExonerateIO.exonerate_text._RE_ALN_ROW=Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_RE_ALN_ROW"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._RE_ALN_ROW" class="py-name" href="#" onclick="return doclink('link-15', '_RE_ALN_ROW', 'link-15');">_RE_ALN_ROW</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'\s*\d+\s+: (.*) :\s+\d+'</tt><tt class="py-op">)</tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt class="py-comment"># for splitting the line based on intron annotations</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"><tt class="py-comment"># e.g. '  &gt;&gt;&gt;&gt; Target Intron 1 &gt;&gt;&gt;&gt;  ' or 'gt.........................ag'</tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"><tt id="link-16" class="py-name" targets="Variable Bio.SearchIO.ExonerateIO.exonerate_text._RE_EXON=Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_RE_EXON"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._RE_EXON" class="py-name" href="#" onclick="return doclink('link-16', '_RE_EXON', 'link-16');">_RE_EXON</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'[atgc ]{2,}?(?:(?:[&lt;&gt;]+ \w+ Intron \d+ [&lt;&gt;]+)|(?:\.+))[atgc ]{2,}?'</tt><tt class="py-op">)</tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"><tt class="py-comment"># captures the intron length</tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"><tt class="py-comment"># from e.g. '61 bp // 154295 bp' (joint intron lengths) or '177446 bp'</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt id="link-17" class="py-name" targets="Variable Bio.SearchIO.ExonerateIO.exonerate_text._RE_EXON_LEN=Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_RE_EXON_LEN"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._RE_EXON_LEN" class="py-name" href="#" onclick="return doclink('link-17', '_RE_EXON_LEN', 'link-17');">_RE_EXON_LEN</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'(?:(\d+) bp // (\d+) bp)|(?:(\d+) bp)'</tt><tt class="py-op">)</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-comment"># for splitting lines in the NER model</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt id="link-18" class="py-name" targets="Variable Bio.SearchIO.ExonerateIO.exonerate_text._RE_NER=Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_RE_NER"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._RE_NER" class="py-name" href="#" onclick="return doclink('link-18', '_RE_NER', 'link-18');">_RE_NER</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'--&lt;\s+\d+\s+&gt;--'</tt><tt class="py-op">)</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"><tt class="py-comment"># for capturing NER gap lengths</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"><tt id="link-19" class="py-name" targets="Variable Bio.SearchIO.ExonerateIO.exonerate_text._RE_NER_LEN=Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_RE_NER_LEN"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._RE_NER_LEN" class="py-name" href="#" onclick="return doclink('link-19', '_RE_NER_LEN', 'link-19');">_RE_NER_LEN</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'--&lt;\s+(\d+)\s+&gt;--'</tt><tt class="py-op">)</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"><tt class="py-comment"># regexes for capturing the letters inside curly braces</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"><tt class="py-comment"># no. of letters is either 1 or 2, since they are split codons</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"><tt id="link-20" class="py-name" targets="Variable Bio.SearchIO.ExonerateIO.exonerate_text._RE_SCODON_START=Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_RE_SCODON_START"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._RE_SCODON_START" class="py-name" href="#" onclick="return doclink('link-20', '_RE_SCODON_START', 'link-20');">_RE_SCODON_START</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'\{(\w{1,2})\}$'</tt><tt class="py-op">)</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"><tt id="link-21" class="py-name" targets="Variable Bio.SearchIO.ExonerateIO.exonerate_text._RE_SCODON_END=Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_RE_SCODON_END"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._RE_SCODON_END" class="py-name" href="#" onclick="return doclink('link-21', '_RE_SCODON_END', 'link-21');">_RE_SCODON_END</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'^\{(\w{1,2})\}'</tt><tt class="py-op">)</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line"> </tt>
<a name="_flip_codons"></a><div id="_flip_codons-def"><a name="L44"></a><tt class="py-lineno"> 44</tt> <a class="py-toggle" href="#" id="_flip_codons-toggle" onclick="return toggle('_flip_codons');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_flip_codons">_flip_codons</a><tt class="py-op">(</tt><tt class="py-param">codon_seq</tt><tt class="py-op">,</tt> <tt class="py-param">target_seq</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_flip_codons-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_flip_codons-expanded"><a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line">    <tt class="py-docstring">"""Flips the codon characters from one seq to another (PRIVATE)."""</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line">    <tt id="link-22" class="py-name" targets="Variable Bio.Graphics.ColorSpiral.ColorSpiral.a=Bio.Graphics.ColorSpiral.ColorSpiral-class.html#a"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-22', 'a', 'link-22');">a</a></tt><tt class="py-op">,</tt> <tt id="link-23" class="py-name" targets="Variable Bio.Graphics.ColorSpiral.ColorSpiral.b=Bio.Graphics.ColorSpiral.ColorSpiral-class.html#b"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-23', 'b', 'link-23');">b</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">char1</tt><tt class="py-op">,</tt> <tt class="py-name">char2</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">codon_seq</tt><tt class="py-op">,</tt> <tt class="py-name">target_seq</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line">        <tt class="py-comment"># no need to do anything if the codon seq line has nothing</tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">char1</tt> <tt class="py-op">==</tt> <tt class="py-string">' '</tt><tt class="py-op">:</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line">            <tt id="link-24" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-24', 'a', 'link-22');">a</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">char1</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line">            <tt id="link-25" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-25', 'b', 'link-23');">b</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">char2</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line">            <tt id="link-26" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-26', 'a', 'link-22');">a</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">char2</tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line">            <tt id="link-27" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-27', 'b', 'link-23');">b</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">char1</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line"> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-28" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-28', 'a', 'link-22');">a</a></tt><tt class="py-op">,</tt> <tt id="link-29" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-29', 'b', 'link-23');">b</a></tt> </tt>
</div><a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line"> </tt>
<a name="_get_block_coords"></a><div id="_get_block_coords-def"><a name="L59"></a><tt class="py-lineno"> 59</tt> <a class="py-toggle" href="#" id="_get_block_coords-toggle" onclick="return toggle('_get_block_coords');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_get_block_coords">_get_block_coords</a><tt class="py-op">(</tt><tt class="py-param">parsed_seq</tt><tt class="py-op">,</tt> <tt class="py-param">row_dict</tt><tt class="py-op">,</tt> <tt class="py-param">has_ner</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_get_block_coords-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_get_block_coords-expanded"><a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return a list of start, end coordinates for each given block in the sequence (PRIVATE)."""</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line">    <tt id="link-30" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.start=Bio.SeqFeature.CompoundLocation-class.html#start,Variable Bio.SeqFeature.FeatureLocation.start=Bio.SeqFeature.FeatureLocation-class.html#start"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-30', 'start', 'link-30');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line">    <tt id="link-31" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.coords=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#coords"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.coords" class="py-name" href="#" onclick="return doclink('link-31', 'coords', 'link-31');">coords</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">has_ner</tt><tt class="py-op">:</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line">        <tt class="py-name">splitter</tt> <tt class="py-op">=</tt> <tt id="link-32" class="py-name"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._RE_EXON" class="py-name" href="#" onclick="return doclink('link-32', '_RE_EXON', 'link-16');">_RE_EXON</a></tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line">        <tt class="py-name">splitter</tt> <tt class="py-op">=</tt> <tt id="link-33" class="py-name"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._RE_NER" class="py-name" href="#" onclick="return doclink('link-33', '_RE_NER', 'link-18');">_RE_NER</a></tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line"> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line">    <tt class="py-comment"># use the query line for reference</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line">    <tt id="link-34" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.seq=Bio.SeqRecord.SeqRecord-class.html#seq,Variable BioSQL.BioSeq.DBSeqRecord.seq=BioSQL.BioSeq.DBSeqRecord-class.html#seq"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-34', 'seq', 'link-34');">seq</a></tt> <tt class="py-op">=</tt> <tt class="py-name">parsed_seq</tt><tt class="py-op">[</tt><tt class="py-name">row_dict</tt><tt class="py-op">[</tt><tt class="py-string">'query'</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line"> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">block</tt> <tt class="py-keyword">in</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-35" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-35', 'split', 'link-35');">split</a></tt><tt class="py-op">(</tt><tt class="py-name">splitter</tt><tt class="py-op">,</tt> <tt id="link-36" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-36', 'seq', 'link-34');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line">        <tt id="link-37" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-37', 'start', 'link-30');">start</a></tt> <tt class="py-op">+=</tt> <tt id="link-38" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-38', 'seq', 'link-34');">seq</a></tt><tt class="py-op">[</tt><tt id="link-39" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-39', 'start', 'link-30');">start</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-40" class="py-name" targets="Method Bio.NeuralNetwork.Gene.Motif.MotifFinder.find()=Bio.NeuralNetwork.Gene.Motif.MotifFinder-class.html#find,Method Bio.NeuralNetwork.Gene.Schema.SchemaFinder.find()=Bio.NeuralNetwork.Gene.Schema.SchemaFinder-class.html#find,Method Bio.NeuralNetwork.Gene.Signature.SignatureFinder.find()=Bio.NeuralNetwork.Gene.Signature.SignatureFinder-class.html#find,Method Bio.Seq.Seq.find()=Bio.Seq.Seq-class.html#find,Function Bio.triefind.find()=Bio.triefind-module.html#find"><a title="Bio.NeuralNetwork.Gene.Motif.MotifFinder.find
Bio.NeuralNetwork.Gene.Schema.SchemaFinder.find
Bio.NeuralNetwork.Gene.Signature.SignatureFinder.find
Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-40', 'find', 'link-40');">find</a></tt><tt class="py-op">(</tt><tt class="py-name">block</tt><tt class="py-op">)</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line">        <tt id="link-41" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.end=Bio.SeqFeature.CompoundLocation-class.html#end,Variable Bio.SeqFeature.FeatureLocation.end=Bio.SeqFeature.FeatureLocation-class.html#end"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-41', 'end', 'link-41');">end</a></tt> <tt class="py-op">=</tt> <tt id="link-42" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-42', 'start', 'link-30');">start</a></tt> <tt class="py-op">+</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">block</tt><tt class="py-op">)</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">        <tt id="link-43" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.coords" class="py-name" href="#" onclick="return doclink('link-43', 'coords', 'link-31');">coords</a></tt><tt class="py-op">.</tt><tt id="link-44" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-44', 'append', 'link-44');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt id="link-45" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-45', 'start', 'link-30');">start</a></tt><tt class="py-op">,</tt> <tt id="link-46" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-46', 'end', 'link-41');">end</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line"> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-47" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.coords" class="py-name" href="#" onclick="return doclink('link-47', 'coords', 'link-31');">coords</a></tt> </tt>
</div><a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line"> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line"> </tt>
<a name="_get_inter_coords"></a><div id="_get_inter_coords-def"><a name="L79"></a><tt class="py-lineno"> 79</tt> <a class="py-toggle" href="#" id="_get_inter_coords-toggle" onclick="return toggle('_get_inter_coords');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_get_inter_coords">_get_inter_coords</a><tt class="py-op">(</tt><tt class="py-param">coords</tt><tt class="py-op">,</tt> <tt class="py-param">strand</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_get_inter_coords-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_get_inter_coords-expanded"><a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return list of pairs covering intervening ranges (PRIVATE).</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line"><tt class="py-docstring">    From the given pairs of coordinates, returns a list of pairs</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line"><tt class="py-docstring">    covering the intervening ranges.</tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line">    <tt class="py-comment"># adapted from Python's itertools guide</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line">    <tt class="py-comment"># if strand is -1, adjust coords to the ends and starts are chained</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt id="link-48" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.strand()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#strand,Variable Bio.SeqFeature.CompoundLocation.strand=Bio.SeqFeature.CompoundLocation-class.html#strand,Variable Bio.SeqFeature.FeatureLocation.strand=Bio.SeqFeature.FeatureLocation-class.html#strand,Variable Bio.SeqFeature.SeqFeature.strand=Bio.SeqFeature.SeqFeature-class.html#strand"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-48', 'strand', 'link-48');">strand</a></tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line">        <tt class="py-name">sorted_coords</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">(</tt><tt id="link-49" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.max=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#max"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-49', 'max', 'link-49');">max</a></tt><tt class="py-op">(</tt><tt id="link-50" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-50', 'a', 'link-22');">a</a></tt><tt class="py-op">,</tt> <tt id="link-51" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-51', 'b', 'link-23');">b</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-52" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.min=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#min"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-52', 'min', 'link-52');">min</a></tt><tt class="py-op">(</tt><tt id="link-53" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-53', 'a', 'link-22');">a</a></tt><tt class="py-op">,</tt> <tt id="link-54" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-54', 'b', 'link-23');">b</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-55" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-55', 'a', 'link-22');">a</a></tt><tt class="py-op">,</tt> <tt id="link-56" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-56', 'b', 'link-23');">b</a></tt> <tt class="py-keyword">in</tt> <tt id="link-57" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.coords" class="py-name" href="#" onclick="return doclink('link-57', 'coords', 'link-31');">coords</a></tt><tt class="py-op">]</tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line">        <tt class="py-name">inter_coords</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">chain</tt><tt class="py-op">(</tt><tt class="py-op">*</tt><tt class="py-name">sorted_coords</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">inter_coords</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">inter_coords</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line">        <tt class="py-name">inter_coords</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">chain</tt><tt class="py-op">(</tt><tt class="py-op">*</tt><tt id="link-58" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.coords" class="py-name" href="#" onclick="return doclink('link-58', 'coords', 'link-31');">coords</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">inter_coords</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">inter_coords</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line"> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line"> </tt>
<a name="_stitch_rows"></a><div id="_stitch_rows-def"><a name="L96"></a><tt class="py-lineno"> 96</tt> <a class="py-toggle" href="#" id="_stitch_rows-toggle" onclick="return toggle('_stitch_rows');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_stitch_rows">_stitch_rows</a><tt class="py-op">(</tt><tt class="py-param">raw_rows</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_stitch_rows-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_stitch_rows-expanded"><a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line">    <tt class="py-docstring">"""Stitches together the parsed alignment rows and returns them in a list (PRIVATE)."""</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line">    <tt class="py-comment"># deal with possible codon surprise!</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">    <tt class="py-comment"># (i.e. alignments with codons using cdna2genome model)</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line">    <tt class="py-comment"># by creating additional rows to contain the codons</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line">        <tt class="py-name">max_len</tt> <tt class="py-op">=</tt> <tt id="link-59" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-59', 'max', 'link-49');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-60" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.x=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#x,Variable Bio.Statistics.lowess.x=Bio.Statistics.lowess-module.html#x"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-60', 'x', 'link-60');">x</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-61" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-61', 'x', 'link-60');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">raw_rows</tt><tt class="py-op">)</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">row</tt> <tt class="py-keyword">in</tt> <tt class="py-name">raw_rows</tt><tt class="py-op">:</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">max_len</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">AssertionError</tt><tt class="py-op">:</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">idx</tt><tt class="py-op">,</tt> <tt class="py-name">row</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt class="py-name">raw_rows</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">max_len</tt><tt class="py-op">:</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line">                <tt class="py-comment"># codons must be present in the query and hit (so +2)</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">                <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-number">2</tt> <tt class="py-op">==</tt> <tt class="py-name">max_len</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">                <tt class="py-comment"># add additional empty lines to contain codons</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">                <tt class="py-name">raw_rows</tt><tt class="py-op">[</tt><tt class="py-name">idx</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">' '</tt> <tt class="py-op">*</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">row</tt> <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-string">' '</tt> <tt class="py-op">*</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line"> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">    <tt class="py-name">cmbn_rows</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">idx</tt><tt class="py-op">,</tt> <tt class="py-name">row</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt class="py-name">raw_rows</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">        <tt class="py-name">cmbn_row</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">aln_row</tt><tt class="py-op">[</tt><tt class="py-name">idx</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt class="py-name">aln_row</tt> <tt class="py-keyword">in</tt> <tt class="py-name">raw_rows</tt><tt class="py-op">)</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">        <tt class="py-name">cmbn_rows</tt><tt class="py-op">.</tt><tt id="link-62" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-62', 'append', 'link-44');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">cmbn_row</tt><tt class="py-op">)</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line"> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line">    <tt class="py-comment"># the real aligned sequence is always the 'outer' one, so we want</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line">    <tt class="py-comment"># to flip them with their 'inner' pairs</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">cmbn_rows</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">5</tt><tt class="py-op">:</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">        <tt class="py-comment"># flip query sequence</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">        <tt class="py-name">cmbn_rows</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">cmbn_rows</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-63" class="py-name" targets="Function Bio.SearchIO.ExonerateIO.exonerate_text._flip_codons()=Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_flip_codons"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._flip_codons" class="py-name" href="#" onclick="return doclink('link-63', '_flip_codons', 'link-63');">_flip_codons</a></tt><tt class="py-op">(</tt><tt class="py-name">cmbn_rows</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">cmbn_rows</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">        <tt class="py-comment"># flip hit sequence</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">        <tt class="py-name">cmbn_rows</tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">cmbn_rows</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-64" class="py-name"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._flip_codons" class="py-name" href="#" onclick="return doclink('link-64', '_flip_codons', 'link-63');">_flip_codons</a></tt><tt class="py-op">(</tt><tt class="py-name">cmbn_rows</tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">cmbn_rows</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line"> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">cmbn_rows</tt> </tt>
</div><a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line"> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line"> </tt>
<a name="_get_row_dict"></a><div id="_get_row_dict-def"><a name="L129"></a><tt class="py-lineno">129</tt> <a class="py-toggle" href="#" id="_get_row_dict-toggle" onclick="return toggle('_get_row_dict');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_get_row_dict">_get_row_dict</a><tt class="py-op">(</tt><tt class="py-param">row_len</tt><tt class="py-op">,</tt> <tt class="py-param">model</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_get_row_dict-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_get_row_dict-expanded"><a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return a dictionary of row indices for parsing alignment blocks (PRIVATE)."""</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line">    <tt class="py-name">idx</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line">    <tt class="py-comment"># 3 lines, usually in dna vs dna models</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">row_len</tt> <tt class="py-op">==</tt> <tt class="py-number">3</tt><tt class="py-op">:</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line">        <tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'query'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line">        <tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'midline'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">        <tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'hit'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">2</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line">        <tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'qannot'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'hannot'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line">    <tt class="py-comment"># 4 lines, in protein vs dna models or dna vs protein models</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line">    <tt class="py-comment"># TODO: currently we check this from the model string; is there</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line">    <tt class="py-comment"># a better way to do it?</tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">row_len</tt> <tt class="py-op">==</tt> <tt class="py-number">4</tt><tt class="py-op">:</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">'protein2'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">model</tt><tt class="py-op">:</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line">            <tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'query'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line">            <tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'midline'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">            <tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'hit'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">2</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">            <tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'hannot'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">3</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line">            <tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'qannot'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-string">'2protein'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">model</tt><tt class="py-op">:</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line">            <tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'query'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">            <tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'midline'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">2</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">            <tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'hit'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">3</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">            <tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'hannot'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">            <tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'qannot'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected model: "</tt> <tt class="py-op">+</tt> <tt class="py-name">model</tt><tt class="py-op">)</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">    <tt class="py-comment"># 5 lines, translated dna vs translated dna</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">row_len</tt> <tt class="py-op">==</tt> <tt class="py-number">5</tt><tt class="py-op">:</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">        <tt class="py-comment"># set sequence indexes</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">        <tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'qannot'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">        <tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'query'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">        <tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'midline'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">2</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">        <tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'hit'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">3</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">        <tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'hannot'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">4</tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected row count in alignment block: "</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">                <tt class="py-string">"%i"</tt> <tt class="py-op">%</tt> <tt class="py-name">row_len</tt><tt class="py-op">)</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">idx</tt> </tt>
</div><a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line"> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line"> </tt>
<a name="_get_blocks"></a><div id="_get_blocks-def"><a name="L170"></a><tt class="py-lineno">170</tt> <a class="py-toggle" href="#" id="_get_blocks-toggle" onclick="return toggle('_get_blocks');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_get_blocks">_get_blocks</a><tt class="py-op">(</tt><tt class="py-param">rows</tt><tt class="py-op">,</tt> <tt class="py-param">coords</tt><tt class="py-op">,</tt> <tt class="py-param">idx</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_get_blocks-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_get_blocks-expanded"><a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return a list of dictionaries of sequences split by the coordinates (PRIVATE)."""</tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">idx_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'query'</tt><tt class="py-op">,</tt> <tt class="py-string">'hit'</tt><tt class="py-op">,</tt> <tt class="py-string">'midline'</tt><tt class="py-op">,</tt> <tt class="py-string">'qannot'</tt><tt class="py-op">,</tt> <tt class="py-string">'hannot'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">idx_name</tt> <tt class="py-keyword">in</tt> <tt class="py-name">idx</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line">    <tt class="py-name">blocks</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt id="link-65" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-65', 'start', 'link-30');">start</a></tt><tt class="py-op">,</tt> <tt id="link-66" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-66', 'end', 'link-41');">end</a></tt> <tt class="py-keyword">in</tt> <tt id="link-67" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.coords" class="py-name" href="#" onclick="return doclink('link-67', 'coords', 'link-31');">coords</a></tt><tt class="py-op">:</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">        <tt class="py-name">block</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">        <tt class="py-comment"># get seqs according to index</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">        <tt class="py-name">block</tt><tt class="py-op">[</tt><tt class="py-string">'query'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">rows</tt><tt class="py-op">[</tt><tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'query'</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt id="link-68" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-68', 'start', 'link-30');">start</a></tt><tt class="py-op">:</tt><tt id="link-69" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-69', 'end', 'link-41');">end</a></tt><tt class="py-op">]</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">        <tt class="py-name">block</tt><tt class="py-op">[</tt><tt class="py-string">'hit'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">rows</tt><tt class="py-op">[</tt><tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'hit'</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt id="link-70" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-70', 'start', 'link-30');">start</a></tt><tt class="py-op">:</tt><tt id="link-71" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-71', 'end', 'link-41');">end</a></tt><tt class="py-op">]</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">        <tt class="py-name">block</tt><tt class="py-op">[</tt><tt class="py-string">'similarity'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">rows</tt><tt class="py-op">[</tt><tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'midline'</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt id="link-72" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-72', 'start', 'link-30');">start</a></tt><tt class="py-op">:</tt><tt id="link-73" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-73', 'end', 'link-41');">end</a></tt><tt class="py-op">]</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'qannot'</tt><tt class="py-op">]</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">            <tt class="py-name">block</tt><tt class="py-op">[</tt><tt class="py-string">'query_annotation'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">rows</tt><tt class="py-op">[</tt><tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'qannot'</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt id="link-74" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-74', 'start', 'link-30');">start</a></tt><tt class="py-op">:</tt><tt id="link-75" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-75', 'end', 'link-41');">end</a></tt><tt class="py-op">]</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'hannot'</tt><tt class="py-op">]</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">            <tt class="py-name">block</tt><tt class="py-op">[</tt><tt class="py-string">'hit_annotation'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">rows</tt><tt class="py-op">[</tt><tt class="py-name">idx</tt><tt class="py-op">[</tt><tt class="py-string">'hannot'</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt id="link-76" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-76', 'start', 'link-30');">start</a></tt><tt class="py-op">:</tt><tt id="link-77" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-77', 'end', 'link-41');">end</a></tt><tt class="py-op">]</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">        <tt class="py-name">blocks</tt><tt class="py-op">.</tt><tt id="link-78" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-78', 'append', 'link-44');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">block</tt><tt class="py-op">)</tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line"> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">blocks</tt> </tt>
</div><a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line"> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line"> </tt>
<a name="_get_scodon_moves"></a><div id="_get_scodon_moves-def"><a name="L190"></a><tt class="py-lineno">190</tt> <a class="py-toggle" href="#" id="_get_scodon_moves-toggle" onclick="return toggle('_get_scodon_moves');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_get_scodon_moves">_get_scodon_moves</a><tt class="py-op">(</tt><tt class="py-param">tmp_seq_blocks</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_get_scodon_moves-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_get_scodon_moves-expanded"><a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">    <tt class="py-docstring">"""Get a dictionary of split codon locations relative to each fragment end (PRIVATE)."""</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">    <tt class="py-name">scodon_moves</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'query'</tt><tt class="py-op">:</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-string">'hit'</tt><tt class="py-op">:</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">}</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">seq_type</tt> <tt class="py-keyword">in</tt> <tt class="py-name">scodon_moves</tt><tt class="py-op">:</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">        <tt class="py-name">scoords</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">block</tt> <tt class="py-keyword">in</tt> <tt class="py-name">tmp_seq_blocks</tt><tt class="py-op">:</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">            <tt class="py-comment"># check both ends of the sequence for residues in curly braces</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">            <tt class="py-name">m_start</tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-79" class="py-name" targets="Method Bio.AlignIO.MafIO.MafIndex.search()=Bio.AlignIO.MafIO.MafIndex-class.html#search,Method Bio.KDTree.KDTree'.KDTree.search()=Bio.KDTree.KDTree%27.KDTree-class.html#search,Method Bio.PDB.NeighborSearch'.NeighborSearch.search()=Bio.PDB.NeighborSearch%27.NeighborSearch-class.html#search,Method Bio.Phylo.TreeConstruction.NNITreeSearcher.search()=Bio.Phylo.TreeConstruction.NNITreeSearcher-class.html#search,Method Bio.Phylo.TreeConstruction.TreeSearcher.search()=Bio.Phylo.TreeConstruction.TreeSearcher-class.html#search,Class Method Bio.Restriction.Restriction.AbstractCut.search()=Bio.Restriction.Restriction.AbstractCut-class.html#search,Method Bio.Restriction.Restriction.RestrictionBatch.search()=Bio.Restriction.Restriction.RestrictionBatch-class.html#search,Function Bio.SCOP.search()=Bio.SCOP-module.html#search,Function Bio.TogoWS.search()=Bio.TogoWS-module.html#search,Method Bio.motifs.Instances.search()=Bio.motifs.Instances-class.html#search,Method Bio.motifs.matrix.PositionSpecificScoringMatrix.search()=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#search"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-79', 'search', 'link-79');">search</a></tt><tt class="py-op">(</tt><tt id="link-80" class="py-name"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._RE_SCODON_START" class="py-name" href="#" onclick="return doclink('link-80', '_RE_SCODON_START', 'link-20');">_RE_SCODON_START</a></tt><tt class="py-op">,</tt> <tt class="py-name">block</tt><tt class="py-op">[</tt><tt class="py-name">seq_type</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line">            <tt class="py-name">m_end</tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-81" class="py-name"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-81', 'search', 'link-79');">search</a></tt><tt class="py-op">(</tt><tt id="link-82" class="py-name"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._RE_SCODON_END" class="py-name" href="#" onclick="return doclink('link-82', '_RE_SCODON_END', 'link-21');">_RE_SCODON_END</a></tt><tt class="py-op">,</tt> <tt class="py-name">block</tt><tt class="py-op">[</tt><tt class="py-name">seq_type</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">m_start</tt><tt class="py-op">:</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line">                <tt class="py-name">m_start</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">m_start</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line">                <tt class="py-name">scoords</tt><tt class="py-op">.</tt><tt id="link-83" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-83', 'append', 'link-44');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">m_start</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line">                <tt class="py-name">scoords</tt><tt class="py-op">.</tt><tt id="link-84" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-84', 'append', 'link-44');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">m_end</tt><tt class="py-op">:</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line">                <tt class="py-name">m_end</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">m_end</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line">                <tt class="py-name">scoords</tt><tt class="py-op">.</tt><tt id="link-85" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-85', 'append', 'link-44');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">m_end</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line">                <tt class="py-name">scoords</tt><tt class="py-op">.</tt><tt id="link-86" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-86', 'append', 'link-44');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line">        <tt class="py-name">scodon_moves</tt><tt class="py-op">[</tt><tt class="py-name">seq_type</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">scoords</tt> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line"> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">scodon_moves</tt> </tt>
</div><a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line"> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line"> </tt>
<a name="_clean_blocks"></a><div id="_clean_blocks-def"><a name="L214"></a><tt class="py-lineno">214</tt> <a class="py-toggle" href="#" id="_clean_blocks-toggle" onclick="return toggle('_clean_blocks');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_clean_blocks">_clean_blocks</a><tt class="py-op">(</tt><tt class="py-param">tmp_seq_blocks</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_clean_blocks-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_clean_blocks-expanded"><a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">    <tt class="py-docstring">"""Remove curly braces (split codon markers) from the given sequences (PRIVATE)."""</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line">    <tt class="py-name">seq_blocks</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">seq_block</tt> <tt class="py-keyword">in</tt> <tt class="py-name">tmp_seq_blocks</tt><tt class="py-op">:</tt> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">line_name</tt> <tt class="py-keyword">in</tt> <tt class="py-name">seq_block</tt><tt class="py-op">:</tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line">            <tt class="py-name">seq_block</tt><tt class="py-op">[</tt><tt class="py-name">line_name</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> \ </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">                    <tt class="py-name">seq_block</tt><tt class="py-op">[</tt><tt class="py-name">line_name</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'{'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'}'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line">        <tt class="py-name">seq_blocks</tt><tt class="py-op">.</tt><tt id="link-87" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-87', 'append', 'link-44');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">seq_block</tt><tt class="py-op">)</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line"> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">seq_blocks</tt> </tt>
</div><a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line"> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line"> </tt>
<a name="_comp_intron_lens"></a><div id="_comp_intron_lens-def"><a name="L226"></a><tt class="py-lineno">226</tt> <a class="py-toggle" href="#" id="_comp_intron_lens-toggle" onclick="return toggle('_comp_intron_lens');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_comp_intron_lens">_comp_intron_lens</a><tt class="py-op">(</tt><tt class="py-param">seq_type</tt><tt class="py-op">,</tt> <tt class="py-param">inter_blocks</tt><tt class="py-op">,</tt> <tt class="py-param">raw_inter_lens</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_comp_intron_lens-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_comp_intron_lens-expanded"><a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return the length of introns between fragments (PRIVATE)."""</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line">    <tt class="py-comment"># set opposite type, for setting introns</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">    <tt class="py-name">opp_type</tt> <tt class="py-op">=</tt> <tt class="py-string">'hit'</tt> <tt class="py-keyword">if</tt> <tt class="py-name">seq_type</tt> <tt class="py-op">==</tt> <tt class="py-string">'query'</tt> <tt class="py-keyword">else</tt> <tt class="py-string">'query'</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line">    <tt class="py-comment"># list of flags to denote if an intron follows a block</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line">    <tt class="py-comment"># it reads e.g. this line:</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">    <tt class="py-comment"># "ATGTT{TT}  &gt;&gt;&gt;&gt; Target Intron 1 &gt;&gt;&gt;&gt;  {G}TGTGTGTACATT"</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">    <tt class="py-comment"># and sets the opposing sequence type's intron (since this</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line">    <tt class="py-comment"># line is present on the opposite sequence type line)</tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line">    <tt class="py-name">has_intron_after</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">'Intron'</tt> <tt class="py-keyword">in</tt> <tt id="link-88" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-88', 'x', 'link-60');">x</a></tt><tt class="py-op">[</tt><tt class="py-name">seq_type</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt id="link-89" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-89', 'x', 'link-60');">x</a></tt> <tt class="py-keyword">in</tt> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line">            <tt class="py-name">inter_blocks</tt><tt class="py-op">]</tt> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">has_intron_after</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">raw_inter_lens</tt><tt class="py-op">)</tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line">    <tt class="py-comment"># create list containing coord adjustments incorporating</tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line">    <tt class="py-comment"># intron lengths</tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line">    <tt class="py-name">inter_lens</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">flag</tt><tt class="py-op">,</tt> <tt class="py-name">parsed_len</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">has_intron_after</tt><tt class="py-op">,</tt> <tt class="py-name">raw_inter_lens</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">flag</tt><tt class="py-op">:</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line">            <tt class="py-comment"># joint introns</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">all</tt><tt class="py-op">(</tt><tt class="py-name">parsed_len</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line">                <tt class="py-comment"># intron len is [0] if opp_type is query, otherwise it's [1]</tt> </tt>
<a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line">                <tt class="py-name">intron_len</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">parsed_len</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt class="py-name">opp_type</tt> <tt class="py-op">==</tt> <tt class="py-string">'query'</tt> \ </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line">                        <tt class="py-keyword">else</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">parsed_len</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line">            <tt class="py-comment"># single hit/query introns</tt> </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">parsed_len</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line">                <tt class="py-name">intron_len</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">parsed_len</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected intron parsing "</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line">                        <tt class="py-string">"result: %r"</tt> <tt class="py-op">%</tt> <tt class="py-name">parsed_len</tt><tt class="py-op">)</tt> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line">            <tt class="py-name">intron_len</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line"> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line">        <tt class="py-name">inter_lens</tt><tt class="py-op">.</tt><tt id="link-90" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-90', 'append', 'link-44');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">intron_len</tt><tt class="py-op">)</tt> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line"> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">inter_lens</tt> </tt>
</div><a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line"> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line"> </tt>
<a name="_comp_coords"></a><div id="_comp_coords-def"><a name="L262"></a><tt class="py-lineno">262</tt> <a class="py-toggle" href="#" id="_comp_coords-toggle" onclick="return toggle('_comp_coords');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_comp_coords">_comp_coords</a><tt class="py-op">(</tt><tt class="py-param">hsp</tt><tt class="py-op">,</tt> <tt class="py-param">seq_type</tt><tt class="py-op">,</tt> <tt class="py-param">inter_lens</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_comp_coords-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_comp_coords-expanded"><a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line">    <tt class="py-docstring">"""Fill the block coordinates of the given hsp dictionary (PRIVATE)."""</tt> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">seq_type</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'hit'</tt><tt class="py-op">,</tt> <tt class="py-string">'query'</tt><tt class="py-op">)</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line">    <tt class="py-comment"># manually fill the first coord</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">    <tt class="py-name">seq_step</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> <tt class="py-keyword">if</tt> <tt id="link-91" class="py-name" targets="Module Bio.SearchIO._model.hsp=Bio.SearchIO._model.hsp-module.html"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-91', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'%s_strand'</tt> <tt class="py-op">%</tt> <tt class="py-name">seq_type</tt><tt class="py-op">]</tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">0</tt> <tt class="py-keyword">else</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">    <tt class="py-name">fstart</tt> <tt class="py-op">=</tt> <tt id="link-92" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-92', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'%s_start'</tt> <tt class="py-op">%</tt> <tt class="py-name">seq_type</tt><tt class="py-op">]</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line">    <tt class="py-comment"># fend is fstart + number of residues in the sequence, minus gaps</tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">    <tt class="py-name">fend</tt> <tt class="py-op">=</tt> <tt class="py-name">fstart</tt> <tt class="py-op">+</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">            <tt id="link-93" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-93', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-name">seq_type</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'-'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'&gt;'</tt><tt class="py-op">,</tt> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line">            <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'&lt;'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-name">seq_step</tt> </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line">    <tt id="link-94" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.coords" class="py-name" href="#" onclick="return doclink('link-94', 'coords', 'link-31');">coords</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">fstart</tt><tt class="py-op">,</tt> <tt class="py-name">fend</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line">    <tt class="py-comment"># and start from the second block, after the first inter seq</tt> </tt>
<a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">idx</tt><tt class="py-op">,</tt> <tt class="py-name">block</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt id="link-95" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-95', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-name">seq_type</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line">        <tt class="py-name">bstart</tt> <tt class="py-op">=</tt> <tt id="link-96" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.coords" class="py-name" href="#" onclick="return doclink('link-96', 'coords', 'link-31');">coords</a></tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">inter_lens</tt><tt class="py-op">[</tt><tt class="py-name">idx</tt><tt class="py-op">]</tt> <tt class="py-op">*</tt> <tt class="py-name">seq_step</tt> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line">        <tt class="py-name">bend</tt> <tt class="py-op">=</tt> <tt class="py-name">bstart</tt> <tt class="py-op">+</tt> <tt class="py-name">seq_step</tt> <tt class="py-op">*</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">block</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'-'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line">        <tt id="link-97" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.coords" class="py-name" href="#" onclick="return doclink('link-97', 'coords', 'link-31');">coords</a></tt><tt class="py-op">.</tt><tt id="link-98" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-98', 'append', 'link-44');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">bstart</tt><tt class="py-op">,</tt> <tt class="py-name">bend</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line"> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line">    <tt class="py-comment"># adjust the coords so the smallest is [0], if strand is -1</tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line">    <tt class="py-comment"># couldn't do this in the previous steps since we need the initial</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line">    <tt class="py-comment"># block ordering</tt> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">seq_step</tt> <tt class="py-op">!=</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">idx</tt><tt class="py-op">,</tt> <tt class="py-name">coord</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt id="link-99" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.coords" class="py-name" href="#" onclick="return doclink('link-99', 'coords', 'link-31');">coords</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">            <tt id="link-100" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.coords" class="py-name" href="#" onclick="return doclink('link-100', 'coords', 'link-31');">coords</a></tt><tt class="py-op">[</tt><tt class="py-name">idx</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-101" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.coords" class="py-name" href="#" onclick="return doclink('link-101', 'coords', 'link-31');">coords</a></tt><tt class="py-op">[</tt><tt class="py-name">idx</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt id="link-102" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.coords" class="py-name" href="#" onclick="return doclink('link-102', 'coords', 'link-31');">coords</a></tt><tt class="py-op">[</tt><tt class="py-name">idx</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line"> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-103" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.coords" class="py-name" href="#" onclick="return doclink('link-103', 'coords', 'link-31');">coords</a></tt> </tt>
</div><a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line"> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line"> </tt>
<a name="_comp_split_codons"></a><div id="_comp_split_codons-def"><a name="L289"></a><tt class="py-lineno">289</tt> <a class="py-toggle" href="#" id="_comp_split_codons-toggle" onclick="return toggle('_comp_split_codons');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_comp_split_codons">_comp_split_codons</a><tt class="py-op">(</tt><tt class="py-param">hsp</tt><tt class="py-op">,</tt> <tt class="py-param">seq_type</tt><tt class="py-op">,</tt> <tt class="py-param">scodon_moves</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_comp_split_codons-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_comp_split_codons-expanded"><a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">    <tt class="py-docstring">"""Compute positions of split codons, store in given HSP dictionary (PRIVATE)."""</tt> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line">    <tt class="py-name">scodons</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">idx</tt> <tt class="py-keyword">in</tt> <tt id="link-104" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-104', 'range', 'link-104');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">scodon_moves</tt><tt class="py-op">[</tt><tt class="py-name">seq_type</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line">        <tt class="py-name">pair</tt> <tt class="py-op">=</tt> <tt class="py-name">scodon_moves</tt><tt class="py-op">[</tt><tt class="py-name">seq_type</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">idx</tt><tt class="py-op">]</tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">any</tt><tt class="py-op">(</tt><tt class="py-name">pair</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line">            <tt class="py-keyword">continue</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt class="py-name">all</tt><tt class="py-op">(</tt><tt class="py-name">pair</tt><tt class="py-op">)</tt> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line">        <tt id="link-105" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-105', 'a', 'link-22');">a</a></tt><tt class="py-op">,</tt> <tt id="link-106" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-106', 'b', 'link-23');">b</a></tt> <tt class="py-op">=</tt> <tt class="py-name">pair</tt> </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line">        <tt class="py-name">anchor_pair</tt> <tt class="py-op">=</tt> <tt id="link-107" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-107', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'%s_ranges'</tt> <tt class="py-op">%</tt> <tt class="py-name">seq_type</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">idx</tt> <tt class="py-op">//</tt> <tt class="py-number">2</tt><tt class="py-op">]</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line">        <tt id="link-108" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-108', 'strand', 'link-48');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> <tt class="py-keyword">if</tt> <tt id="link-109" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-109', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'%s_strand'</tt> <tt class="py-op">%</tt> <tt class="py-name">seq_type</tt><tt class="py-op">]</tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">0</tt> <tt class="py-keyword">else</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line"> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-110" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-110', 'a', 'link-22');">a</a></tt><tt class="py-op">:</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line">            <tt class="py-name">func</tt> <tt class="py-op">=</tt> <tt id="link-111" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-111', 'max', 'link-49');">max</a></tt> <tt class="py-keyword">if</tt> <tt id="link-112" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-112', 'strand', 'link-48');">strand</a></tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt> <tt class="py-keyword">else</tt> <tt id="link-113" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-113', 'min', 'link-52');">min</a></tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line">            <tt class="py-name">anchor</tt> <tt class="py-op">=</tt> <tt class="py-name">func</tt><tt class="py-op">(</tt><tt class="py-name">anchor_pair</tt><tt class="py-op">)</tt> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line">            <tt class="py-name">start_c</tt><tt class="py-op">,</tt> <tt class="py-name">end_c</tt> <tt class="py-op">=</tt> <tt class="py-name">anchor</tt> <tt class="py-op">+</tt> <tt id="link-114" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-114', 'a', 'link-22');">a</a></tt> <tt class="py-op">*</tt> <tt id="link-115" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-115', 'strand', 'link-48');">strand</a></tt> <tt class="py-op">*</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">anchor</tt> </tt>
<a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-116" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-116', 'b', 'link-23');">b</a></tt><tt class="py-op">:</tt> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">            <tt class="py-name">func</tt> <tt class="py-op">=</tt> <tt id="link-117" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-117', 'min', 'link-52');">min</a></tt> <tt class="py-keyword">if</tt> <tt id="link-118" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-118', 'strand', 'link-48');">strand</a></tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt> <tt class="py-keyword">else</tt> <tt id="link-119" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-119', 'max', 'link-49');">max</a></tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line">            <tt class="py-name">anchor</tt> <tt class="py-op">=</tt> <tt class="py-name">func</tt><tt class="py-op">(</tt><tt class="py-name">anchor_pair</tt><tt class="py-op">)</tt> </tt>
<a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line">            <tt class="py-name">start_c</tt><tt class="py-op">,</tt> <tt class="py-name">end_c</tt> <tt class="py-op">=</tt> <tt class="py-name">anchor</tt> <tt class="py-op">+</tt> <tt id="link-120" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-120', 'b', 'link-23');">b</a></tt> <tt class="py-op">*</tt> <tt id="link-121" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-121', 'strand', 'link-48');">strand</a></tt><tt class="py-op">,</tt> <tt class="py-name">anchor</tt> </tt>
<a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line">        <tt class="py-name">scodons</tt><tt class="py-op">.</tt><tt id="link-122" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-122', 'append', 'link-44');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt id="link-123" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-123', 'min', 'link-52');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">start_c</tt><tt class="py-op">,</tt> <tt class="py-name">end_c</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-124" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-124', 'max', 'link-49');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">start_c</tt><tt class="py-op">,</tt> <tt class="py-name">end_c</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line"> </tt>
<a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">scodons</tt> </tt>
</div><a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line"> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line"> </tt>
<a name="ExonerateTextParser"></a><div id="ExonerateTextParser-def"><a name="L315"></a><tt class="py-lineno">315</tt> <a class="py-toggle" href="#" id="ExonerateTextParser-toggle" onclick="return toggle('ExonerateTextParser');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextParser-class.html">ExonerateTextParser</a><tt class="py-op">(</tt><tt class="py-base-class">_BaseExonerateParser</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ExonerateTextParser-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="ExonerateTextParser-expanded"><a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line">    <tt class="py-docstring">"""Parser for Exonerate plain text output."""</tt> </tt>
<a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line"> </tt>
<a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">    <tt id="link-125" class="py-name" targets="Variable Bio.SearchIO.ExonerateIO._base._BaseExonerateParser._ALN_MARK=Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html#_ALN_MARK,Variable Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarParser._ALN_MARK=Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarParser-class.html#_ALN_MARK,Variable Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextParser._ALN_MARK=Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextParser-class.html#_ALN_MARK,Variable Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarParser._ALN_MARK=Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarParser-class.html#_ALN_MARK"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateParser._ALN_MARK
Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarParser._ALN_MARK
Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextParser._ALN_MARK
Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarParser._ALN_MARK" class="py-name" href="#" onclick="return doclink('link-125', '_ALN_MARK', 'link-125');">_ALN_MARK</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'C4 Alignment:'</tt> </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line"> </tt>
<a name="ExonerateTextParser.parse_alignment_block"></a><div id="ExonerateTextParser.parse_alignment_block-def"><a name="L320"></a><tt class="py-lineno">320</tt> <a class="py-toggle" href="#" id="ExonerateTextParser.parse_alignment_block-toggle" onclick="return toggle('ExonerateTextParser.parse_alignment_block');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextParser-class.html#parse_alignment_block">parse_alignment_block</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">header</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ExonerateTextParser.parse_alignment_block-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ExonerateTextParser.parse_alignment_block-expanded"><a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line">        <tt class="py-name">qresult</tt> <tt class="py-op">=</tt> <tt id="link-126" class="py-name" targets="Variable Bio.Wise.dnal.Statistics.header=Bio.Wise.dnal.Statistics-class.html#header"><a title="Bio.Wise.dnal.Statistics.header" class="py-name" href="#" onclick="return doclink('link-126', 'header', 'link-126');">header</a></tt><tt class="py-op">[</tt><tt class="py-string">'qresult'</tt><tt class="py-op">]</tt> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">        <tt id="link-127" class="py-name" targets="Module Bio.SearchIO._model.hit=Bio.SearchIO._model.hit-module.html,Variable Bio.SearchIO._model.hsp.HSP.hit=Bio.SearchIO._model.hsp.HSP-class.html#hit,Variable Bio.SearchIO._model.hsp.HSPFragment.hit=Bio.SearchIO._model.hsp.HSPFragment-class.html#hit"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-127', 'hit', 'link-127');">hit</a></tt> <tt class="py-op">=</tt> <tt id="link-128" class="py-name"><a title="Bio.Wise.dnal.Statistics.header" class="py-name" href="#" onclick="return doclink('link-128', 'header', 'link-126');">header</a></tt><tt class="py-op">[</tt><tt class="py-string">'hit'</tt><tt class="py-op">]</tt> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">        <tt id="link-129" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-129', 'hsp', 'link-91');">hsp</a></tt> <tt class="py-op">=</tt> <tt id="link-130" class="py-name"><a title="Bio.Wise.dnal.Statistics.header" class="py-name" href="#" onclick="return doclink('link-130', 'header', 'link-126');">header</a></tt><tt class="py-op">[</tt><tt class="py-string">'hsp'</tt><tt class="py-op">]</tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line">        <tt class="py-comment"># check for values that must have been set by previous methods</tt> </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">val_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'query_start'</tt><tt class="py-op">,</tt> <tt class="py-string">'query_end'</tt><tt class="py-op">,</tt> <tt class="py-string">'hit_start'</tt><tt class="py-op">,</tt> <tt class="py-string">'hit_end'</tt><tt class="py-op">,</tt> </tt>
<a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">                <tt class="py-string">'query_strand'</tt><tt class="py-op">,</tt> <tt class="py-string">'hit_strand'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">val_name</tt> <tt class="py-keyword">in</tt> <tt id="link-131" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-131', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">,</tt> <tt id="link-132" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-132', 'hsp', 'link-91');">hsp</a></tt> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line"> </tt>
<a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line">        <tt class="py-comment"># get the alignment rows</tt> </tt>
<a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line">        <tt class="py-comment"># and stitch them so we have the full sequences in single strings</tt> </tt>
<a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line">        <tt class="py-name">raw_aln_blocks</tt><tt class="py-op">,</tt> <tt class="py-name">vulgar_comp</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-133" class="py-name" targets="Method Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextParser._read_alignment()=Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextParser-class.html#_read_alignment"><a title="Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextParser._read_alignment" class="py-name" href="#" onclick="return doclink('link-133', '_read_alignment', 'link-133');">_read_alignment</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line">        <tt class="py-comment"># cmbn_rows still has split codon markers (curly braces)</tt> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line">        <tt class="py-name">cmbn_rows</tt> <tt class="py-op">=</tt> <tt id="link-134" class="py-name" targets="Function Bio.SearchIO.ExonerateIO.exonerate_text._stitch_rows()=Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_stitch_rows"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._stitch_rows" class="py-name" href="#" onclick="return doclink('link-134', '_stitch_rows', 'link-134');">_stitch_rows</a></tt><tt class="py-op">(</tt><tt class="py-name">raw_aln_blocks</tt><tt class="py-op">)</tt> </tt>
<a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line">        <tt class="py-name">row_dict</tt> <tt class="py-op">=</tt> <tt id="link-135" class="py-name" targets="Function Bio.SearchIO.ExonerateIO.exonerate_text._get_row_dict()=Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_get_row_dict"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._get_row_dict" class="py-name" href="#" onclick="return doclink('link-135', '_get_row_dict', 'link-135');">_get_row_dict</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">cmbn_rows</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">qresult</tt><tt class="py-op">[</tt><tt class="py-string">'model'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line">        <tt class="py-comment"># get the sequence blocks</tt> </tt>
<a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line">        <tt class="py-name">has_ner</tt> <tt class="py-op">=</tt> <tt class="py-string">'NER'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qresult</tt><tt class="py-op">[</tt><tt class="py-string">'model'</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-136" class="py-name" targets="Method Bio.Seq.Seq.upper()=Bio.Seq.Seq-class.html#upper,Method Bio.Seq.UnknownSeq.upper()=Bio.Seq.UnknownSeq-class.html#upper,Method Bio.SeqRecord.SeqRecord.upper()=Bio.SeqRecord.SeqRecord-class.html#upper"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-136', 'upper', 'link-136');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line">        <tt class="py-name">seq_coords</tt> <tt class="py-op">=</tt> <tt id="link-137" class="py-name" targets="Function Bio.SearchIO.ExonerateIO.exonerate_text._get_block_coords()=Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_get_block_coords"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._get_block_coords" class="py-name" href="#" onclick="return doclink('link-137', '_get_block_coords', 'link-137');">_get_block_coords</a></tt><tt class="py-op">(</tt><tt class="py-name">cmbn_rows</tt><tt class="py-op">,</tt> <tt class="py-name">row_dict</tt><tt class="py-op">,</tt> <tt class="py-name">has_ner</tt><tt class="py-op">)</tt> </tt>
<a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line">        <tt class="py-name">tmp_seq_blocks</tt> <tt class="py-op">=</tt> <tt id="link-138" class="py-name" targets="Function Bio.SearchIO.ExonerateIO.exonerate_text._get_blocks()=Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_get_blocks"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._get_blocks" class="py-name" href="#" onclick="return doclink('link-138', '_get_blocks', 'link-138');">_get_blocks</a></tt><tt class="py-op">(</tt><tt class="py-name">cmbn_rows</tt><tt class="py-op">,</tt> <tt class="py-name">seq_coords</tt><tt class="py-op">,</tt> <tt class="py-name">row_dict</tt><tt class="py-op">)</tt> </tt>
<a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line">        <tt class="py-comment"># get split codon temp coords for later use</tt> </tt>
<a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line">        <tt class="py-comment"># this result in pairs of base movement for both ends of each row</tt> </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line">        <tt class="py-name">scodon_moves</tt> <tt class="py-op">=</tt> <tt id="link-139" class="py-name" targets="Function Bio.SearchIO.ExonerateIO.exonerate_text._get_scodon_moves()=Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_get_scodon_moves"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._get_scodon_moves" class="py-name" href="#" onclick="return doclink('link-139', '_get_scodon_moves', 'link-139');">_get_scodon_moves</a></tt><tt class="py-op">(</tt><tt class="py-name">tmp_seq_blocks</tt><tt class="py-op">)</tt> </tt>
<a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line">        <tt class="py-comment"># remove the split codon markers</tt> </tt>
<a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line">        <tt class="py-name">seq_blocks</tt> <tt class="py-op">=</tt> <tt id="link-140" class="py-name" targets="Function Bio.SearchIO.ExonerateIO.exonerate_text._clean_blocks()=Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_clean_blocks"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._clean_blocks" class="py-name" href="#" onclick="return doclink('link-140', '_clean_blocks', 'link-140');">_clean_blocks</a></tt><tt class="py-op">(</tt><tt class="py-name">tmp_seq_blocks</tt><tt class="py-op">)</tt> </tt>
<a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line"> </tt>
<a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line">        <tt class="py-comment"># adjust strands</tt> </tt>
<a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line">        <tt id="link-141" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-141', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'query_strand'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-142" class="py-name"><a title="Bio.SearchIO.ExonerateIO._base._STRAND_MAP" class="py-name" href="#" onclick="return doclink('link-142', '_STRAND_MAP', 'link-11');">_STRAND_MAP</a></tt><tt class="py-op">[</tt><tt id="link-143" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-143', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'query_strand'</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L347"></a><tt class="py-lineno">347</tt>  <tt class="py-line">        <tt id="link-144" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-144', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'hit_strand'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-145" class="py-name"><a title="Bio.SearchIO.ExonerateIO._base._STRAND_MAP" class="py-name" href="#" onclick="return doclink('link-145', '_STRAND_MAP', 'link-11');">_STRAND_MAP</a></tt><tt class="py-op">[</tt><tt id="link-146" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-146', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'hit_strand'</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line">        <tt class="py-comment"># cast coords into ints</tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line">        <tt id="link-147" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-147', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'query_start'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-148" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-148', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'query_start'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line">        <tt id="link-149" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-149', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'query_end'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-150" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-150', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'query_end'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line">        <tt id="link-151" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-151', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'hit_start'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-152" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-152', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'hit_start'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L352"></a><tt class="py-lineno">352</tt>  <tt class="py-line">        <tt id="link-153" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-153', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'hit_end'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-154" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-154', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'hit_end'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line">        <tt class="py-comment"># cast score into ints</tt> </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line">        <tt id="link-155" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-155', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'score'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-156" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-156', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'score'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">        <tt class="py-comment"># set sequences</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line">        <tt id="link-157" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-157', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'query'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-158" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-158', 'x', 'link-60');">x</a></tt><tt class="py-op">[</tt><tt class="py-string">'query'</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt id="link-159" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-159', 'x', 'link-60');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">seq_blocks</tt><tt class="py-op">]</tt> </tt>
<a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line">        <tt id="link-160" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-160', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'hit'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-161" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-161', 'x', 'link-60');">x</a></tt><tt class="py-op">[</tt><tt class="py-string">'hit'</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt id="link-162" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-162', 'x', 'link-60');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">seq_blocks</tt><tt class="py-op">]</tt> </tt>
<a name="L358"></a><tt class="py-lineno">358</tt>  <tt class="py-line">        <tt id="link-163" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-163', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'aln_annotation'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line">        <tt class="py-comment"># set the alphabet</tt> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line">        <tt class="py-comment"># currently only limited to models with protein queries</tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">'protein2'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qresult</tt><tt class="py-op">[</tt><tt class="py-string">'model'</tt><tt class="py-op">]</tt> <tt class="py-keyword">or</tt> <tt class="py-string">'coding2'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qresult</tt><tt class="py-op">[</tt><tt class="py-string">'model'</tt><tt class="py-op">]</tt> \ </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line">                <tt class="py-keyword">or</tt> <tt class="py-string">'2protein'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qresult</tt><tt class="py-op">[</tt><tt class="py-string">'model'</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line">            <tt id="link-164" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-164', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'alphabet'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-165" class="py-name"><a title="Bio.Alphabet.generic_protein" class="py-name" href="#" onclick="return doclink('link-165', 'generic_protein', 'link-7');">generic_protein</a></tt> </tt>
<a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line">        <tt class="py-comment"># get the annotations if they exist</tt> </tt>
<a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">annot_type</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'similarity'</tt><tt class="py-op">,</tt> <tt class="py-string">'query_annotation'</tt><tt class="py-op">,</tt> <tt class="py-string">'hit_annotation'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line">                <tt id="link-166" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-166', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'aln_annotation'</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">annot_type</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> \ </tt>
<a name="L368"></a><tt class="py-lineno">368</tt>  <tt class="py-line">                        <tt class="py-op">[</tt><tt id="link-167" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-167', 'x', 'link-60');">x</a></tt><tt class="py-op">[</tt><tt class="py-name">annot_type</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt id="link-168" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-168', 'x', 'link-60');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">seq_blocks</tt><tt class="py-op">]</tt> </tt>
<a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line">                <tt class="py-keyword">pass</tt> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line"> </tt>
<a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line">        <tt class="py-comment"># use vulgar coordinates if vulgar line is present and return</tt> </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line">        <tt class="py-comment"># if vulgar_comp is not None:</tt> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line">        <tt class="py-comment">#    hsp = parse_vulgar_comp(hsp, vulgar_comp)</tt> </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line"> </tt>
<a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line">        <tt class="py-comment">#    return {'qresult': qresult, 'hit': hit, 'hsp': hsp}</tt> </tt>
<a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line"> </tt>
<a name="L378"></a><tt class="py-lineno">378</tt>  <tt class="py-line">        <tt class="py-comment"># otherwise we need to get the coordinates from the alignment</tt> </tt>
<a name="L379"></a><tt class="py-lineno">379</tt>  <tt class="py-line">        <tt class="py-comment"># get the intervening blocks first, so we can use them</tt> </tt>
<a name="L380"></a><tt class="py-lineno">380</tt>  <tt class="py-line">        <tt class="py-comment"># to adjust the coordinates</tt> </tt>
<a name="L381"></a><tt class="py-lineno">381</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">has_ner</tt><tt class="py-op">:</tt> </tt>
<a name="L382"></a><tt class="py-lineno">382</tt>  <tt class="py-line">            <tt class="py-comment"># get intervening coordinates and blocks, only if model is not ner</tt> </tt>
<a name="L383"></a><tt class="py-lineno">383</tt>  <tt class="py-line">            <tt class="py-comment"># ner models have a much more simple coordinate calculation</tt> </tt>
<a name="L384"></a><tt class="py-lineno">384</tt>  <tt class="py-line">            <tt class="py-name">inter_coords</tt> <tt class="py-op">=</tt> <tt id="link-169" class="py-name" targets="Function Bio.SearchIO.ExonerateIO.exonerate_text._get_inter_coords()=Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_get_inter_coords"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._get_inter_coords" class="py-name" href="#" onclick="return doclink('link-169', '_get_inter_coords', 'link-169');">_get_inter_coords</a></tt><tt class="py-op">(</tt><tt class="py-name">seq_coords</tt><tt class="py-op">)</tt> </tt>
<a name="L385"></a><tt class="py-lineno">385</tt>  <tt class="py-line">            <tt class="py-name">inter_blocks</tt> <tt class="py-op">=</tt> <tt id="link-170" class="py-name"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._get_blocks" class="py-name" href="#" onclick="return doclink('link-170', '_get_blocks', 'link-138');">_get_blocks</a></tt><tt class="py-op">(</tt><tt class="py-name">cmbn_rows</tt><tt class="py-op">,</tt> <tt class="py-name">inter_coords</tt><tt class="py-op">,</tt> <tt class="py-name">row_dict</tt><tt class="py-op">)</tt> </tt>
<a name="L386"></a><tt class="py-lineno">386</tt>  <tt class="py-line">            <tt class="py-comment"># returns a three-component tuple of intron lengths</tt> </tt>
<a name="L387"></a><tt class="py-lineno">387</tt>  <tt class="py-line">            <tt class="py-comment"># first two component filled == intron in hit and query</tt> </tt>
<a name="L388"></a><tt class="py-lineno">388</tt>  <tt class="py-line">            <tt class="py-comment"># last component filled == intron in hit or query</tt> </tt>
<a name="L389"></a><tt class="py-lineno">389</tt>  <tt class="py-line">            <tt class="py-name">raw_inter_lens</tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">findall</tt><tt class="py-op">(</tt><tt id="link-171" class="py-name"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._RE_EXON_LEN" class="py-name" href="#" onclick="return doclink('link-171', '_RE_EXON_LEN', 'link-17');">_RE_EXON_LEN</a></tt><tt class="py-op">,</tt> </tt>
<a name="L390"></a><tt class="py-lineno">390</tt>  <tt class="py-line">                    <tt class="py-name">cmbn_rows</tt><tt class="py-op">[</tt><tt class="py-name">row_dict</tt><tt class="py-op">[</tt><tt class="py-string">'midline'</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L391"></a><tt class="py-lineno">391</tt>  <tt class="py-line"> </tt>
<a name="L392"></a><tt class="py-lineno">392</tt>  <tt class="py-line">        <tt class="py-comment"># compute start and end coords for each block</tt> </tt>
<a name="L393"></a><tt class="py-lineno">393</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">seq_type</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'query'</tt><tt class="py-op">,</tt> <tt class="py-string">'hit'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L394"></a><tt class="py-lineno">394</tt>  <tt class="py-line"> </tt>
<a name="L395"></a><tt class="py-lineno">395</tt>  <tt class="py-line">            <tt class="py-comment"># ner blocks and intron blocks require different adjustments</tt> </tt>
<a name="L396"></a><tt class="py-lineno">396</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">has_ner</tt><tt class="py-op">:</tt> </tt>
<a name="L397"></a><tt class="py-lineno">397</tt>  <tt class="py-line">                <tt class="py-name">opp_type</tt> <tt class="py-op">=</tt> <tt class="py-string">'hit'</tt> <tt class="py-keyword">if</tt> <tt class="py-name">seq_type</tt> <tt class="py-op">==</tt> <tt class="py-string">'query'</tt> <tt class="py-keyword">else</tt> <tt class="py-string">'query'</tt> </tt>
<a name="L398"></a><tt class="py-lineno">398</tt>  <tt class="py-line">                <tt class="py-name">inter_lens</tt> <tt class="py-op">=</tt> <tt id="link-172" class="py-name" targets="Function Bio.SearchIO.ExonerateIO.exonerate_text._comp_intron_lens()=Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_comp_intron_lens"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._comp_intron_lens" class="py-name" href="#" onclick="return doclink('link-172', '_comp_intron_lens', 'link-172');">_comp_intron_lens</a></tt><tt class="py-op">(</tt><tt class="py-name">seq_type</tt><tt class="py-op">,</tt> <tt class="py-name">inter_blocks</tt><tt class="py-op">,</tt> </tt>
<a name="L399"></a><tt class="py-lineno">399</tt>  <tt class="py-line">                        <tt class="py-name">raw_inter_lens</tt><tt class="py-op">)</tt> </tt>
<a name="L400"></a><tt class="py-lineno">400</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L401"></a><tt class="py-lineno">401</tt>  <tt class="py-line">                <tt class="py-comment"># for NER blocks, the length of the inter-fragment gaps is</tt> </tt>
<a name="L402"></a><tt class="py-lineno">402</tt>  <tt class="py-line">                <tt class="py-comment"># written on the same strand, so opp_type is seq_type</tt> </tt>
<a name="L403"></a><tt class="py-lineno">403</tt>  <tt class="py-line">                <tt class="py-name">opp_type</tt> <tt class="py-op">=</tt> <tt class="py-name">seq_type</tt> </tt>
<a name="L404"></a><tt class="py-lineno">404</tt>  <tt class="py-line">                <tt class="py-name">inter_lens</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-173" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-173', 'x', 'link-60');">x</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-174" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-174', 'x', 'link-60');">x</a></tt> <tt class="py-keyword">in</tt> </tt>
<a name="L405"></a><tt class="py-lineno">405</tt>  <tt class="py-line">                        <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">findall</tt><tt class="py-op">(</tt><tt id="link-175" class="py-name"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._RE_NER_LEN" class="py-name" href="#" onclick="return doclink('link-175', '_RE_NER_LEN', 'link-19');">_RE_NER_LEN</a></tt><tt class="py-op">,</tt> <tt class="py-name">cmbn_rows</tt><tt class="py-op">[</tt><tt class="py-name">row_dict</tt><tt class="py-op">[</tt><tt class="py-name">seq_type</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L406"></a><tt class="py-lineno">406</tt>  <tt class="py-line"> </tt>
<a name="L407"></a><tt class="py-lineno">407</tt>  <tt class="py-line">            <tt class="py-comment"># check that inter_lens's length is len opp_type block - 1</tt> </tt>
<a name="L408"></a><tt class="py-lineno">408</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">inter_lens</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-176" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-176', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-name">opp_type</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> \ </tt>
<a name="L409"></a><tt class="py-lineno">409</tt>  <tt class="py-line">                    <tt class="py-string">"%r vs %r"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">inter_lens</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-177" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-177', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-name">opp_type</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L410"></a><tt class="py-lineno">410</tt>  <tt class="py-line">            <tt class="py-comment"># fill the hsp query and hit coordinates</tt> </tt>
<a name="L411"></a><tt class="py-lineno">411</tt>  <tt class="py-line">            <tt id="link-178" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-178', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'%s_ranges'</tt> <tt class="py-op">%</tt> <tt class="py-name">opp_type</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-179" class="py-name" targets="Function Bio.SearchIO.ExonerateIO.exonerate_text._comp_coords()=Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_comp_coords"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._comp_coords" class="py-name" href="#" onclick="return doclink('link-179', '_comp_coords', 'link-179');">_comp_coords</a></tt><tt class="py-op">(</tt><tt id="link-180" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-180', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">,</tt> <tt class="py-name">opp_type</tt><tt class="py-op">,</tt> <tt class="py-name">inter_lens</tt><tt class="py-op">)</tt> </tt>
<a name="L412"></a><tt class="py-lineno">412</tt>  <tt class="py-line">            <tt class="py-comment"># and fill the split codon coordinates, if model != ner</tt> </tt>
<a name="L413"></a><tt class="py-lineno">413</tt>  <tt class="py-line">            <tt class="py-comment"># can't do this in the if-else clause above since we need to</tt> </tt>
<a name="L414"></a><tt class="py-lineno">414</tt>  <tt class="py-line">            <tt class="py-comment"># compute the ranges first</tt> </tt>
<a name="L415"></a><tt class="py-lineno">415</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">has_ner</tt><tt class="py-op">:</tt> </tt>
<a name="L416"></a><tt class="py-lineno">416</tt>  <tt class="py-line">                <tt id="link-181" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-181', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'%s_split_codons'</tt> <tt class="py-op">%</tt> <tt class="py-name">opp_type</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> \ </tt>
<a name="L417"></a><tt class="py-lineno">417</tt>  <tt class="py-line">                        <tt id="link-182" class="py-name" targets="Function Bio.SearchIO.ExonerateIO.exonerate_text._comp_split_codons()=Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_comp_split_codons"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._comp_split_codons" class="py-name" href="#" onclick="return doclink('link-182', '_comp_split_codons', 'link-182');">_comp_split_codons</a></tt><tt class="py-op">(</tt><tt id="link-183" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-183', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">,</tt> <tt class="py-name">opp_type</tt><tt class="py-op">,</tt> <tt class="py-name">scodon_moves</tt><tt class="py-op">)</tt> </tt>
<a name="L418"></a><tt class="py-lineno">418</tt>  <tt class="py-line"> </tt>
<a name="L419"></a><tt class="py-lineno">419</tt>  <tt class="py-line">        <tt class="py-comment"># now that we've finished parsing coords, we can set the hit and start</tt> </tt>
<a name="L420"></a><tt class="py-lineno">420</tt>  <tt class="py-line">        <tt class="py-comment"># coord according to Biopython's convention (start &lt;= end)</tt> </tt>
<a name="L421"></a><tt class="py-lineno">421</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">seq_type</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'query'</tt><tt class="py-op">,</tt> <tt class="py-string">'hit'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L422"></a><tt class="py-lineno">422</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-184" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-184', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'%s_strand'</tt> <tt class="py-op">%</tt> <tt class="py-name">seq_type</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L423"></a><tt class="py-lineno">423</tt>  <tt class="py-line">                <tt class="py-name">n_start</tt> <tt class="py-op">=</tt> <tt class="py-string">'%s_start'</tt> <tt class="py-op">%</tt> <tt class="py-name">seq_type</tt> </tt>
<a name="L424"></a><tt class="py-lineno">424</tt>  <tt class="py-line">                <tt class="py-name">n_end</tt> <tt class="py-op">=</tt> <tt class="py-string">'%s_end'</tt> <tt class="py-op">%</tt> <tt class="py-name">seq_type</tt> </tt>
<a name="L425"></a><tt class="py-lineno">425</tt>  <tt class="py-line">                <tt id="link-185" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-185', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-name">n_start</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt id="link-186" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-186', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-name">n_end</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-187" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-187', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-name">n_end</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt id="link-188" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-188', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-name">n_start</tt><tt class="py-op">]</tt> </tt>
<a name="L426"></a><tt class="py-lineno">426</tt>  <tt class="py-line"> </tt>
<a name="L427"></a><tt class="py-lineno">427</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">{</tt><tt class="py-string">'qresult'</tt><tt class="py-op">:</tt> <tt class="py-name">qresult</tt><tt class="py-op">,</tt> <tt class="py-string">'hit'</tt><tt class="py-op">:</tt> <tt id="link-189" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-189', 'hit', 'link-127');">hit</a></tt><tt class="py-op">,</tt> <tt class="py-string">'hsp'</tt><tt class="py-op">:</tt> <tt id="link-190" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-190', 'hsp', 'link-91');">hsp</a></tt><tt class="py-op">}</tt> </tt>
</div><a name="L428"></a><tt class="py-lineno">428</tt>  <tt class="py-line"> </tt>
<a name="ExonerateTextParser._read_alignment"></a><div id="ExonerateTextParser._read_alignment-def"><a name="L429"></a><tt class="py-lineno">429</tt> <a class="py-toggle" href="#" id="ExonerateTextParser._read_alignment-toggle" onclick="return toggle('ExonerateTextParser._read_alignment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextParser-class.html#_read_alignment">_read_alignment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ExonerateTextParser._read_alignment-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ExonerateTextParser._read_alignment-expanded"><a name="L430"></a><tt class="py-lineno">430</tt>  <tt class="py-line">        <tt class="py-docstring">"""Read the raw alignment block strings, returns them in a list (PRIVATE)."""</tt> </tt>
<a name="L431"></a><tt class="py-lineno">431</tt>  <tt class="py-line">        <tt class="py-name">raw_aln_blocks</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L432"></a><tt class="py-lineno">432</tt>  <tt class="py-line">        <tt class="py-comment"># flag to check whether we're in an aligment row</tt> </tt>
<a name="L433"></a><tt class="py-lineno">433</tt>  <tt class="py-line">        <tt class="py-name">in_aln_row</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L434"></a><tt class="py-lineno">434</tt>  <tt class="py-line">        <tt class="py-comment"># flag for vulgar line, if present, we can parse coordinates from it</tt> </tt>
<a name="L435"></a><tt class="py-lineno">435</tt>  <tt class="py-line">        <tt class="py-name">vulgar_comp</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L436"></a><tt class="py-lineno">436</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L437"></a><tt class="py-lineno">437</tt>  <tt class="py-line"> </tt>
<a name="L438"></a><tt class="py-lineno">438</tt>  <tt class="py-line">            <tt id="link-191" class="py-name" targets="Function Bio.triefind.match()=Bio.triefind-module.html#match"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-191', 'match', 'link-191');">match</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-192" class="py-name"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-192', 'search', 'link-79');">search</a></tt><tt class="py-op">(</tt><tt id="link-193" class="py-name"><a title="Bio.SearchIO.ExonerateIO.exonerate_text._RE_ALN_ROW" class="py-name" href="#" onclick="return doclink('link-193', '_RE_ALN_ROW', 'link-15');">_RE_ALN_ROW</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-194" class="py-name" targets="Method Bio.Seq.Seq.strip()=Bio.Seq.Seq-class.html#strip"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-194', 'strip', 'link-194');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L439"></a><tt class="py-lineno">439</tt>  <tt class="py-line">            <tt class="py-comment"># if we have a match, set flags and values</tt> </tt>
<a name="L440"></a><tt class="py-lineno">440</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-195" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-195', 'match', 'link-191');">match</a></tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt class="py-name">in_aln_row</tt><tt class="py-op">:</tt> </tt>
<a name="L441"></a><tt class="py-lineno">441</tt>  <tt class="py-line">                <tt class="py-name">start_idx</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-196" class="py-name" targets="Method Bio.Crystal.Chain.index()=Bio.Crystal.Chain-class.html#index,Method Bio.SCOP.Raf.SeqMap.index()=Bio.SCOP.Raf.SeqMap-class.html#index,Method Bio.SearchIO._model.hit.Hit.index()=Bio.SearchIO._model.hit.Hit-class.html#index,Method Bio.SearchIO._model.query.QueryResult.index()=Bio.SearchIO._model.query.QueryResult-class.html#index,Function Bio.SearchIO.index()=Bio.SearchIO-module.html#index,Method Bio.Seq.MutableSeq.index()=Bio.Seq.MutableSeq-class.html#index,Function Bio.SeqIO.index()=Bio.SeqIO-module.html#index"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-196', 'index', 'link-196');">index</a></tt><tt class="py-op">(</tt><tt id="link-197" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-197', 'match', 'link-191');">match</a></tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L442"></a><tt class="py-lineno">442</tt>  <tt class="py-line">                <tt class="py-name">row_len</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-198" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-198', 'match', 'link-191');">match</a></tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L443"></a><tt class="py-lineno">443</tt>  <tt class="py-line">                <tt class="py-name">in_aln_row</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L444"></a><tt class="py-lineno">444</tt>  <tt class="py-line">                <tt class="py-name">raw_aln_block</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L445"></a><tt class="py-lineno">445</tt>  <tt class="py-line">            <tt class="py-comment"># if we're in an alignment row, grab the sequence</tt> </tt>
<a name="L446"></a><tt class="py-lineno">446</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">in_aln_row</tt><tt class="py-op">:</tt> </tt>
<a name="L447"></a><tt class="py-lineno">447</tt>  <tt class="py-line">                <tt class="py-name">raw_aln_block</tt><tt class="py-op">.</tt><tt id="link-199" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-199', 'append', 'link-44');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">start_idx</tt><tt class="py-op">:</tt><tt class="py-name">start_idx</tt> <tt class="py-op">+</tt> <tt class="py-name">row_len</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L448"></a><tt class="py-lineno">448</tt>  <tt class="py-line">            <tt class="py-comment"># reset flags and values if the line matches, we're in an alignment</tt> </tt>
<a name="L449"></a><tt class="py-lineno">449</tt>  <tt class="py-line">            <tt class="py-comment"># row, and there are more than 1 line in rows</tt> </tt>
<a name="L450"></a><tt class="py-lineno">450</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-200" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-200', 'match', 'link-191');">match</a></tt> <tt class="py-keyword">and</tt> <tt class="py-name">in_aln_row</tt> <tt class="py-keyword">and</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">raw_aln_block</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L451"></a><tt class="py-lineno">451</tt>  <tt class="py-line">                <tt class="py-name">raw_aln_blocks</tt><tt class="py-op">.</tt><tt id="link-201" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-201', 'append', 'link-44');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">raw_aln_block</tt><tt class="py-op">)</tt> </tt>
<a name="L452"></a><tt class="py-lineno">452</tt>  <tt class="py-line">                <tt class="py-name">start_idx</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L453"></a><tt class="py-lineno">453</tt>  <tt class="py-line">                <tt class="py-name">row_len</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L454"></a><tt class="py-lineno">454</tt>  <tt class="py-line">                <tt class="py-name">in_aln_row</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L455"></a><tt class="py-lineno">455</tt>  <tt class="py-line"> </tt>
<a name="L456"></a><tt class="py-lineno">456</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-202" class="py-name" targets="Method Bio.File.UndoHandle.readline()=Bio.File.UndoHandle-class.html#readline,Method Bio.bgzf.BgzfReader.readline()=Bio.bgzf.BgzfReader-class.html#readline"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-202', 'readline', 'link-202');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L457"></a><tt class="py-lineno">457</tt>  <tt class="py-line">            <tt class="py-comment"># try to parse vulgar line if present</tt> </tt>
<a name="L458"></a><tt class="py-lineno">458</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-203" class="py-name" targets="Method Bio.Seq.Seq.startswith()=Bio.Seq.Seq-class.html#startswith"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-203', 'startswith', 'link-203');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'vulgar'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L459"></a><tt class="py-lineno">459</tt>  <tt class="py-line">                <tt class="py-name">vulgar</tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-204" class="py-name"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-204', 'search', 'link-79');">search</a></tt><tt class="py-op">(</tt><tt id="link-205" class="py-name"><a title="Bio.SearchIO.ExonerateIO.exonerate_vulgar._RE_VULGAR" class="py-name" href="#" onclick="return doclink('link-205', '_RE_VULGAR', 'link-14');">_RE_VULGAR</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L460"></a><tt class="py-lineno">460</tt>  <tt class="py-line">                <tt class="py-name">vulgar_comp</tt> <tt class="py-op">=</tt> <tt class="py-name">vulgar</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">10</tt><tt class="py-op">)</tt> </tt>
<a name="L461"></a><tt class="py-lineno">461</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-keyword">or</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-206" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-206', 'startswith', 'link-203');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-207" class="py-name"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateParser._ALN_MARK
Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarParser._ALN_MARK
Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextParser._ALN_MARK
Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarParser._ALN_MARK" class="py-name" href="#" onclick="return doclink('link-207', '_ALN_MARK', 'link-125');">_ALN_MARK</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L462"></a><tt class="py-lineno">462</tt>  <tt class="py-line">                <tt class="py-comment"># HACK: this is so that the parse_qresult method does not</tt> </tt>
<a name="L463"></a><tt class="py-lineno">463</tt>  <tt class="py-line">                <tt class="py-comment"># yield the objects before appending the last HSP. We are doing</tt> </tt>
<a name="L464"></a><tt class="py-lineno">464</tt>  <tt class="py-line">                <tt class="py-comment"># this to keep the parser compatible with outputs without</tt> </tt>
<a name="L465"></a><tt class="py-lineno">465</tt>  <tt class="py-line">                <tt class="py-comment"># human-readable alignment outputs. This also relies on the</tt> </tt>
<a name="L466"></a><tt class="py-lineno">466</tt>  <tt class="py-line">                <tt class="py-comment"># fact that repeated readline() always returns '' on EOF.</tt> </tt>
<a name="L467"></a><tt class="py-lineno">467</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L468"></a><tt class="py-lineno">468</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-string">'mock'</tt> </tt>
<a name="L469"></a><tt class="py-lineno">469</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L470"></a><tt class="py-lineno">470</tt>  <tt class="py-line"> </tt>
<a name="L471"></a><tt class="py-lineno">471</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">raw_aln_blocks</tt><tt class="py-op">,</tt> <tt class="py-name">vulgar_comp</tt> </tt>
</div></div><a name="L472"></a><tt class="py-lineno">472</tt>  <tt class="py-line"> </tt>
<a name="L473"></a><tt class="py-lineno">473</tt>  <tt class="py-line"> </tt>
<a name="ExonerateTextIndexer"></a><div id="ExonerateTextIndexer-def"><a name="L474"></a><tt class="py-lineno">474</tt> <a class="py-toggle" href="#" id="ExonerateTextIndexer-toggle" onclick="return toggle('ExonerateTextIndexer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextIndexer-class.html">ExonerateTextIndexer</a><tt class="py-op">(</tt><tt class="py-base-class">_BaseExonerateIndexer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ExonerateTextIndexer-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="ExonerateTextIndexer-expanded"><a name="L475"></a><tt class="py-lineno">475</tt>  <tt class="py-line">    <tt class="py-docstring">"""Indexer class for Exonerate plain text."""</tt> </tt>
<a name="L476"></a><tt class="py-lineno">476</tt>  <tt class="py-line"> </tt>
<a name="L477"></a><tt class="py-lineno">477</tt>  <tt class="py-line">    <tt id="link-208" class="py-name" targets="Variable Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer._parser=Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer-class.html#_parser"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer._parser" class="py-name" href="#" onclick="return doclink('link-208', '_parser', 'link-208');">_parser</a></tt> <tt class="py-op">=</tt> <tt id="link-209" class="py-name" targets="Class Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextParser=Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextParser-class.html"><a title="Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextParser" class="py-name" href="#" onclick="return doclink('link-209', 'ExonerateTextParser', 'link-209');">ExonerateTextParser</a></tt> </tt>
<a name="L478"></a><tt class="py-lineno">478</tt>  <tt class="py-line">    <tt id="link-210" class="py-name" targets="Variable Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer._query_mark=Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer-class.html#_query_mark,Variable Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarIndexer._query_mark=Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarIndexer-class.html#_query_mark,Variable Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextIndexer._query_mark=Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextIndexer-class.html#_query_mark,Variable Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarIndexer._query_mark=Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarIndexer-class.html#_query_mark"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer._query_mark
Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarIndexer._query_mark
Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextIndexer._query_mark
Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarIndexer._query_mark" class="py-name" href="#" onclick="return doclink('link-210', '_query_mark', 'link-210');">_query_mark</a></tt> <tt class="py-op">=</tt> <tt class="py-string">b"C4 Alignment"</tt> </tt>
<a name="L479"></a><tt class="py-lineno">479</tt>  <tt class="py-line"> </tt>
<a name="ExonerateTextIndexer.get_qresult_id"></a><div id="ExonerateTextIndexer.get_qresult_id-def"><a name="L480"></a><tt class="py-lineno">480</tt> <a class="py-toggle" href="#" id="ExonerateTextIndexer.get_qresult_id-toggle" onclick="return toggle('ExonerateTextIndexer.get_qresult_id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextIndexer-class.html#get_qresult_id">get_qresult_id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">pos</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ExonerateTextIndexer.get_qresult_id-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ExonerateTextIndexer.get_qresult_id-expanded"><a name="L481"></a><tt class="py-lineno">481</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the query ID from the nearest "Query:" line."""</tt> </tt>
<a name="L482"></a><tt class="py-lineno">482</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_handle</tt> </tt>
<a name="L483"></a><tt class="py-lineno">483</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-211" class="py-name" targets="Method Bio.File.UndoHandle.seek()=Bio.File.UndoHandle-class.html#seek,Method Bio.SeqIO.SffIO._AddTellHandle.seek()=Bio.SeqIO.SffIO._AddTellHandle-class.html#seek,Method Bio.bgzf.BgzfReader.seek()=Bio.bgzf.BgzfReader-class.html#seek"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-211', 'seek', 'link-211');">seek</a></tt><tt class="py-op">(</tt><tt class="py-name">pos</tt><tt class="py-op">)</tt> </tt>
<a name="L484"></a><tt class="py-lineno">484</tt>  <tt class="py-line">        <tt class="py-name">sentinel</tt> <tt class="py-op">=</tt> <tt class="py-string">b"Query:"</tt> </tt>
<a name="L485"></a><tt class="py-lineno">485</tt>  <tt class="py-line"> </tt>
<a name="L486"></a><tt class="py-lineno">486</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L487"></a><tt class="py-lineno">487</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-212" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-212', 'readline', 'link-202');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-213" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-213', 'strip', 'link-194');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L488"></a><tt class="py-lineno">488</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-214" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-214', 'startswith', 'link-203');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-name">sentinel</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L489"></a><tt class="py-lineno">489</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L490"></a><tt class="py-lineno">490</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L491"></a><tt class="py-lineno">491</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">StopIteration</tt> </tt>
<a name="L492"></a><tt class="py-lineno">492</tt>  <tt class="py-line">        <tt class="py-name">qid</tt><tt class="py-op">,</tt> <tt id="link-215" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.desc()=Bio.Phylo.PhyloXMLIO.Writer-class.html#desc"><a title="Bio.Phylo.PhyloXMLIO.Writer.desc" class="py-name" href="#" onclick="return doclink('link-215', 'desc', 'link-215');">desc</a></tt> <tt class="py-op">=</tt> <tt id="link-216" class="py-name"><a title="Bio.SearchIO.ExonerateIO._base._parse_hit_or_query_line" class="py-name" href="#" onclick="return doclink('link-216', '_parse_hit_or_query_line', 'link-12');">_parse_hit_or_query_line</a></tt><tt class="py-op">(</tt><tt id="link-217" class="py-name"><a title="Bio._py3k._bytes_to_string" class="py-name" href="#" onclick="return doclink('link-217', '_bytes_to_string', 'link-2');">_bytes_to_string</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L493"></a><tt class="py-lineno">493</tt>  <tt class="py-line"> </tt>
<a name="L494"></a><tt class="py-lineno">494</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">qid</tt> </tt>
</div><a name="L495"></a><tt class="py-lineno">495</tt>  <tt class="py-line"> </tt>
<a name="ExonerateTextIndexer.get_raw"></a><div id="ExonerateTextIndexer.get_raw-def"><a name="L496"></a><tt class="py-lineno">496</tt> <a class="py-toggle" href="#" id="ExonerateTextIndexer.get_raw-toggle" onclick="return toggle('ExonerateTextIndexer.get_raw');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextIndexer-class.html#get_raw">get_raw</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">offset</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ExonerateTextIndexer.get_raw-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ExonerateTextIndexer.get_raw-expanded"><a name="L497"></a><tt class="py-lineno">497</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the raw string of a QueryResult object from the given offset."""</tt> </tt>
<a name="L498"></a><tt class="py-lineno">498</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_handle</tt> </tt>
<a name="L499"></a><tt class="py-lineno">499</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-218" class="py-name"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-218', 'seek', 'link-211');">seek</a></tt><tt class="py-op">(</tt><tt class="py-name">offset</tt><tt class="py-op">)</tt> </tt>
<a name="L500"></a><tt class="py-lineno">500</tt>  <tt class="py-line">        <tt class="py-name">qresult_key</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L501"></a><tt class="py-lineno">501</tt>  <tt class="py-line">        <tt class="py-name">qresult_raw</tt> <tt class="py-op">=</tt> <tt class="py-string">b""</tt> </tt>
<a name="L502"></a><tt class="py-lineno">502</tt>  <tt class="py-line"> </tt>
<a name="L503"></a><tt class="py-lineno">503</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L504"></a><tt class="py-lineno">504</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-219" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-219', 'readline', 'link-202');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L505"></a><tt class="py-lineno">505</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L506"></a><tt class="py-lineno">506</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L507"></a><tt class="py-lineno">507</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-220" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-220', 'startswith', 'link-203');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-221" class="py-name"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer._query_mark
Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarIndexer._query_mark
Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextIndexer._query_mark
Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarIndexer._query_mark" class="py-name" href="#" onclick="return doclink('link-221', '_query_mark', 'link-210');">_query_mark</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L508"></a><tt class="py-lineno">508</tt>  <tt class="py-line">                <tt class="py-name">cur_pos</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-222" class="py-name" targets="Method Bio.File.UndoHandle.tell()=Bio.File.UndoHandle-class.html#tell,Method Bio.SeqIO.SffIO._AddTellHandle.tell()=Bio.SeqIO.SffIO._AddTellHandle-class.html#tell,Method Bio.bgzf.BgzfReader.tell()=Bio.bgzf.BgzfReader-class.html#tell,Method Bio.bgzf.BgzfWriter.tell()=Bio.bgzf.BgzfWriter-class.html#tell"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-222', 'tell', 'link-222');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L509"></a><tt class="py-lineno">509</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">qresult_key</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L510"></a><tt class="py-lineno">510</tt>  <tt class="py-line">                    <tt class="py-name">qresult_key</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-223" class="py-name" targets="Method Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer.get_qresult_id()=Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer-class.html#get_qresult_id,Method Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarIndexer.get_qresult_id()=Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarIndexer-class.html#get_qresult_id,Method Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextIndexer.get_qresult_id()=Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextIndexer-class.html#get_qresult_id,Method Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarIndexer.get_qresult_id()=Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarIndexer-class.html#get_qresult_id"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer.get_qresult_id
Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarIndexer.get_qresult_id
Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextIndexer.get_qresult_id
Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarIndexer.get_qresult_id" class="py-name" href="#" onclick="return doclink('link-223', 'get_qresult_id', 'link-223');">get_qresult_id</a></tt><tt class="py-op">(</tt><tt class="py-name">cur_pos</tt><tt class="py-op">)</tt> </tt>
<a name="L511"></a><tt class="py-lineno">511</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L512"></a><tt class="py-lineno">512</tt>  <tt class="py-line">                    <tt class="py-name">curr_key</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-224" class="py-name"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer.get_qresult_id
Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarIndexer.get_qresult_id
Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextIndexer.get_qresult_id
Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarIndexer.get_qresult_id" class="py-name" href="#" onclick="return doclink('link-224', 'get_qresult_id', 'link-223');">get_qresult_id</a></tt><tt class="py-op">(</tt><tt class="py-name">cur_pos</tt><tt class="py-op">)</tt> </tt>
<a name="L513"></a><tt class="py-lineno">513</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">curr_key</tt> <tt class="py-op">!=</tt> <tt class="py-name">qresult_key</tt><tt class="py-op">:</tt> </tt>
<a name="L514"></a><tt class="py-lineno">514</tt>  <tt class="py-line">                        <tt class="py-keyword">break</tt> </tt>
<a name="L515"></a><tt class="py-lineno">515</tt>  <tt class="py-line">                <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-225" class="py-name"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-225', 'seek', 'link-211');">seek</a></tt><tt class="py-op">(</tt><tt class="py-name">cur_pos</tt><tt class="py-op">)</tt> </tt>
<a name="L516"></a><tt class="py-lineno">516</tt>  <tt class="py-line">            <tt class="py-name">qresult_raw</tt> <tt class="py-op">+=</tt> <tt class="py-name">line</tt> </tt>
<a name="L517"></a><tt class="py-lineno">517</tt>  <tt class="py-line"> </tt>
<a name="L518"></a><tt class="py-lineno">518</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">qresult_raw</tt> </tt>
</div></div><a name="L519"></a><tt class="py-lineno">519</tt>  <tt class="py-line"> </tt>
<a name="L520"></a><tt class="py-lineno">520</tt>  <tt class="py-line"> </tt>
<a name="L521"></a><tt class="py-lineno">521</tt>  <tt class="py-line"><tt class="py-comment"># if not used as a module, run the doctest</tt> </tt>
<a name="L522"></a><tt class="py-lineno">522</tt>  <tt class="py-line"><tt class="py-keyword">if</tt> <tt class="py-name">__name__</tt> <tt class="py-op">==</tt> <tt class="py-string">"__main__"</tt><tt class="py-op">:</tt> </tt>
<a name="L523"></a><tt class="py-lineno">523</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-226" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-226', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-227" class="py-name" targets="Module Bio.Phylo._utils=Bio.Phylo._utils-module.html,Module Bio.SearchIO._utils=Bio.SearchIO._utils-module.html,Module Bio._utils=Bio._utils-module.html"><a title="Bio.Phylo._utils
Bio.SearchIO._utils
Bio._utils" class="py-name" href="#" onclick="return doclink('link-227', '_utils', 'link-227');">_utils</a></tt> <tt class="py-keyword">import</tt> <tt id="link-228" class="py-name" targets="Function Bio._utils.run_doctest()=Bio._utils-module.html#run_doctest"><a title="Bio._utils.run_doctest" class="py-name" href="#" onclick="return doclink('link-228', 'run_doctest', 'link-228');">run_doctest</a></tt> </tt>
<a name="L524"></a><tt class="py-lineno">524</tt>  <tt class="py-line">    <tt id="link-229" class="py-name"><a title="Bio._utils.run_doctest" class="py-name" href="#" onclick="return doclink('link-229', 'run_doctest', 'link-228');">run_doctest</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L525"></a><tt class="py-lineno">525</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Fri Jun 22 16:36:14 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
