<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.motifs.pfm</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.motifs-module.html">Package&nbsp;motifs</a> ::
        Module&nbsp;pfm
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.motifs.pfm-module.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== MODULE DESCRIPTION ==================== -->
<h1 class="epydoc">Module pfm</h1><p class="nomargin-top"><span class="codelink"><a href="Bio.motifs.pfm-pysrc.html">source&nbsp;code</a></span></p>
Parse various position frequency matrix format files.

<!-- ==================== CLASSES ==================== -->
<a name="section-Classes"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Classes</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Classes"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.motifs.pfm.Record-class.html" class="summary-name">Record</a><br />
      Class to store the information in a position frequency matrix table.
    </td>
  </tr>
</table>
<!-- ==================== FUNCTIONS ==================== -->
<a name="section-Functions"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Functions</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Functions"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.motifs.pfm-module.html#read" class="summary-sig-name">read</a>(<span class="summary-sig-arg">handle</span>,
        <span class="summary-sig-arg">pfm_format</span>)</span><br />
      Read motif(s) from a file in various position frequency matrix formats.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.motifs.pfm-pysrc.html#read">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.motifs.pfm-module.html#_read_pfm_four_columns" class="summary-sig-name" onclick="show_private();">_read_pfm_four_columns</a>(<span class="summary-sig-arg">handle</span>)</span><br />
      Read motifs in Cluster Buster position frequency matrix format from a file handle.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.motifs.pfm-pysrc.html#_read_pfm_four_columns">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.motifs.pfm-module.html#_read_pfm_four_rows" class="summary-sig-name" onclick="show_private();">_read_pfm_four_rows</a>(<span class="summary-sig-arg">handle</span>)</span><br />
      Read motifs in position frequency matrix format from a file handle.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.motifs.pfm-pysrc.html#_read_pfm_four_rows">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="write"></a><span class="summary-sig-name">write</span>(<span class="summary-sig-arg">motifs</span>)</span><br />
      Return the representation of motifs in Cluster Buster position frequency matrix format.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.motifs.pfm-pysrc.html#write">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
</table>
<!-- ==================== VARIABLES ==================== -->
<a name="section-Variables"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Variables</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Variables"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="__package__"></a><span class="summary-name">__package__</span> = <code title="'Bio.motifs'"><code class="variable-quote">'</code><code class="variable-string">Bio.motifs</code><code class="variable-quote">'</code></code>
    </td>
  </tr>
</table>
<!-- ==================== FUNCTION DETAILS ==================== -->
<a name="section-FunctionDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Function Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-FunctionDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="read"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">read</span>(<span class="sig-arg">handle</span>,
        <span class="sig-arg">pfm_format</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.motifs.pfm-pysrc.html#read">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Read motif(s) from a file in various position frequency matrix formats.</p>
<p>Return the record of PFM(s).
Call the appropriate routine based on the format passed.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_read_pfm_four_columns"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_read_pfm_four_columns</span>(<span class="sig-arg">handle</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.motifs.pfm-pysrc.html#_read_pfm_four_columns">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Read motifs in Cluster Buster position frequency matrix format from a file handle.</p>
<p>Cluster Buster motif format: <a class="rst-reference external" href="http://zlab.bu.edu/cluster-buster/help/cis-format.html" target="_top">http://zlab.bu.edu/cluster-buster/help/cis-format.html</a></p>
<p>#cisbp
Pos A   C   G   T
1   0.00961538461538462 0.00961538461538462 0.00961538461538462 0.971153846153846
2   0.00961538461538462 0.00961538461538462 0.00961538461538462 0.971153846153846
3   0.971153846153846   0.00961538461538462 0.00961538461538462 0.00961538461538462
4   0.00961538461538462 0.00961538461538462 0.00961538461538462 0.971153846153846
5   0.00961538461538462 0.971153846153846   0.00961538461538462 0.00961538461538462
6   0.971153846153846   0.00961538461538462 0.00961538461538462 0.00961538461538462
7   0.00961538461538462 0.971153846153846   0.00961538461538462 0.00961538461538462
8   0.00961538461538462 0.00961538461538462 0.00961538461538462 0.971153846153846</p>
<p>#c2h2 zfs
Gene    ENSG00000197372
Pos A   C   G   T
1   0.341303    0.132427    0.117054    0.409215
2   0.283785    0.077066    0.364552    0.274597
3   0.491055    0.078208    0.310520    0.120217
4   0.492621    0.076117    0.131007    0.300256
5   0.250645    0.361464    0.176504    0.211387
6   0.276694    0.498070    0.197793    0.027444
7   0.056317    0.014631    0.926202    0.002850
8   0.004470    0.007769    0.983797    0.003964
9   0.936213    0.058787    0.002387    0.002613
10  0.004352    0.004030    0.002418    0.989200
11  0.013277    0.008165    0.001991    0.976567
12  0.968132    0.002263    0.002868    0.026737
13  0.397623    0.052017    0.350783    0.199577
14  0.000000    0.000000    1.000000    0.000000
15  1.000000    0.000000    0.000000    0.000000
16  0.000000    0.000000    1.000000    0.000000
17  0.000000    0.000000    1.000000    0.000000
18  1.000000    0.000000    0.000000    0.000000
19  0.000000    1.000000    0.000000    0.000000
20  1.000000    0.000000    0.000000    0.000000</p>
<p>#c2h2 zfs
Gene    FBgn0000210
Motif   M1734_0.90
Pos A   C   G   T
1   0.25    0.0833333   0.0833333   0.583333
2   0.75    0.166667    0.0833333   0
3   0.833333    0   0   0.166667
4   1   0   0   0
5   0   0.833333    0.0833333   0.0833333
6   0.333333    0   0   0.666667
7   0.833333    0   0   0.166667
8   0.5 0   0.333333    0.166667
9   0.5 0.0833333   0.166667    0.25
10  0.333333    0.25    0.166667    0.25
11  0.166667    0.25    0.416667    0.166667</p>
<p># flyfactorsurvey (cluster buster)
&gt;AbdA_Cell_FBgn0000014
1   3   0   14
0   0   0   18
16  0   0   2
18  0   0   0
1   0   0   17
0   0   6   12
15  1   2   0</p>
<p># homer
&gt;ATGACTCATC AP-1(bZIP)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer    6.049537    -1.782996e+03   0   9805.3,5781.0,3085.1,2715.0,0.00e+00
0.419   0.275   0.277   0.028
0.001   0.001   0.001   0.997
0.010   0.002   0.965   0.023
0.984   0.003   0.001   0.012
0.062   0.579   0.305   0.054
0.026   0.001   0.001   0.972
0.043   0.943   0.001   0.012
0.980   0.005   0.001   0.014
0.050   0.172   0.307   0.471
0.149   0.444   0.211   0.195</p>
<p># hocomoco
&gt; AHR_si
40.51343240527031  18.259112547756697  56.41253757072521  38.77363485291994
10.877470982533044  11.870876719950774  34.66312982331297  96.54723985087516
21.7165707818416  43.883079837598544  20.706746561638717  67.6523201955933
2.5465132509466635  1.3171620263517245  145.8637051322628  4.231336967110781
0.0  150.35847450464382  1.4927836298652875  2.1074592421627525
3.441039751299748  0.7902972158110341  149.37613720253387  0.3512432070271259
0.0  3.441039751299748  0.7024864140542533  149.81519121131782
0.0  0.0  153.95871737667187  0.0
43.07922333291745  66.87558226865211  16.159862546986584  27.844049228115868</p>
<p># neph
UW.Motif.0001   atgactca
0.772949    0.089579    0.098612    0.038860
0.026652    0.004653    0.025056    0.943639
0.017663    0.023344    0.918728    0.040264
0.919596    0.025414    0.029759    0.025231
0.060312    0.772259    0.104968    0.062462
0.037406    0.020643    0.006667    0.935284
0.047316    0.899024    0.026928    0.026732
0.948639    0.019497    0.005737    0.026128</p>
<p># tiffin
T   A   G   C
30  0   28  40
0   0   0   99
0   55  14  29
0   99  0   0
20  78  0   0
0   52  7   39
19  46  11  22
0   60  38  0
0   33  0   66
73  0   25  0
99  0   0   0</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_read_pfm_four_rows"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_read_pfm_four_rows</span>(<span class="sig-arg">handle</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.motifs.pfm-pysrc.html#_read_pfm_four_rows">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Read motifs in position frequency matrix format from a file handle.</p>
<p>Cluster Buster motif format: <a class="rst-reference external" href="http://zlab.bu.edu/cluster-buster/help/cis-format.html" target="_top">http://zlab.bu.edu/cluster-buster/help/cis-format.html</a></p>
<p>#hdpi
A   0   5   6   5   1   0
C   1   1   0   0   0   4
G   5   0   0   0   3   0
T   0   0   0   1   2   2</p>
<p># yetfasco
A   0.5 0.0 0.0 0.25    0.25    0.25    0.25    0.25    0.25    0.25    0.25    0.25    0.5 0.0 0.0833333334583333
T   0.0 0.0 0.0 0.25    0.25    0.25    0.25    0.25    0.25    0.25    0.25    0.25    0.0 0.0 0.0833333334583333
G   0.0 1.0 0.0 0.25    0.25    0.25    0.25    0.25    0.25    0.25    0.25    0.25    0.0 1.0 0.249999999875
C   0.5 0.0 1.0 0.25    0.25    0.25    0.25    0.25    0.25    0.25    0.25    0.25    0.5 0.0 0.583333333208333</p>
<p>#flyfactorsurvey ZFP finger
A |     92    106    231    135      0      1    780     28      0    700    739     94     60    127    130
C |    138     82    129     81    774      1      3      1      0      6     17     49    193    122    148
G |    270    398     54    164      7    659      1    750    755     65      1     41    202    234    205
T |    290    204    375    411      9    127      6     11     36     20     31    605    335    307    308</p>
<p># scertf pcm
A | 9 1 1 97 1 94
T | 80 1 97 1 1 2
C | 9 97 1 1 1 2
G | 2 1 1 1 97 2</p>
<p># scertf pfm
A | 0.090 0.010 0.010 0.970 0.010 0.940
C | 0.090 0.970 0.010 0.010 0.010 0.020
G | 0.020 0.010 0.010 0.010 0.970 0.020
T | 0.800 0.010 0.970 0.010 0.010 0.020</p>
<p>#idmmpmm
&gt; abd-A
0.218451749734889 0.0230646871686108 0.656680805938494 0.898197242841994 0.040694591728526 0.132953340402969 0.74907211028632 0.628313891834571
0.0896076352067868 0.317338282078473 0.321580063626723 0.0461293743372216 0.0502386002120891 0.040694591728526 0.0284994697773065 0.0339342523860021
0.455991516436904 0.0691940615058324 0.0108695652173913 0.0217391304347826 0.0284994697773065 0.0284994697773065 0.016304347826087 0.160127253446448
0.235949098621421 0.590402969247084 0.0108695652173913 0.0339342523860021 0.880567338282079 0.797852598091198 0.206124072110286 0.17762460233298</p>
<dl class="rst-docutils">
<dt># JASPAR</dt>
<dd>&gt;MA0001.1 AGL3
A  [ 0  3 79 40 66 48 65 11 65  0 ]
C  [94 75  4  3  1  2  5  2  3  3 ]
G  [ 1  0  3  4  1  0  5  3 28 88 ]
T  [ 2 19 11 50 29 47 22 81  1  6 ]</dd>
</dl>
<p>or:</p>
<pre class="rst-literal-block">
&gt;MA0001.1 AGL3
0  3 79 40 66 48 65 11 65  0
94 75  4  3  1  2  5  2  3  3
1  0  3  4  1  0  5  3 28 88
2 19 11 50 29 47 22 81  1  6
</pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Nov  6 13:32:24 2019
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
