<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.AlignIO.MafIO</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.AlignIO-module.html">Package&nbsp;AlignIO</a> ::
        Module&nbsp;MafIO
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.AlignIO.MafIO-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.AlignIO.MafIO-module.html">Module Bio.AlignIO.MafIO</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2011, 2012 by Andrew Sczesnak.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># Revisions Copyright 2011, 2017 by Peter Cock.  All rights reserved.</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># Revisions Copyright 2014, 2015 by Adam Novak.  All rights reserved.</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># Revisions Copyright 2015, 2017 by Blaise Li.  All rights reserved.</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-comment"># This file is part of the Biopython distribution and governed by your</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-comment"># choice of the "Biopython License Agreement" or the "BSD 3-Clause License".</tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-comment"># Please see the LICENSE file that should have been included as part of this</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-comment"># package.</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-docstring">"""Bio.AlignIO support for the "maf" multiple alignment format.</tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-docstring">The Multiple Alignment Format, described by UCSC, stores a series of</tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-docstring">multiple alignments in a single file. It is suitable for whole-genome</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-docstring">to whole-genome alignments, metadata such as source chromosome, start</tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-docstring">position, size, and strand can be stored.</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-docstring">See http://genome.ucsc.edu/FAQ/FAQformat.html#format5</tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-docstring">You are expected to use this module via the Bio.AlignIO functions(or the</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-docstring">Bio.SeqIO functions if you want to work directly with the gapped sequences).</tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-docstring">Coordinates in the MAF format are defined in terms of zero-based start</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-docstring">positions (like Python) and aligning region sizes.</tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-docstring">A minimal aligned region of length one and starting at first position in the</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-docstring">source sequence would have ``start == 0`` and ``size == 1``.</tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt class="py-docstring">As we can see on this example, ``start + size`` will give one more than the</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"><tt class="py-docstring">zero-based end position. We can therefore manipulate ``start`` and</tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"><tt class="py-docstring">``start + size`` as python list slice boundaries.</tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"><tt class="py-docstring">For an inclusive end coordinate, we need to use ``end = start + size - 1``.</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-docstring">A 1-column wide alignment would have ``start == end``.</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">os</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">itertools</tt> <tt class="py-keyword">import</tt> <tt class="py-name">islice</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"><tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt class="py-name">sqlite3</tt> <tt class="py-keyword">import</tt> <tt class="py-name">dbapi2</tt> <tt class="py-keyword">as</tt> <tt class="py-name">_sqlite</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"><tt class="py-keyword">except</tt> <tt class="py-name">ImportError</tt><tt class="py-op">:</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line">    <tt class="py-comment"># Not present on Jython, but should be included in Python 2.5</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line">    <tt class="py-comment"># or later (unless compiled from source without its dependencies)</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line">    <tt class="py-comment"># Still want to offer simple parsing/output</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line">    <tt class="py-name">_sqlite</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line"> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-1', 'Alphabet', 'link-1');">Alphabet</a></tt> <tt class="py-keyword">import</tt> <tt id="link-2" class="py-name" targets="Variable Bio.Alphabet.single_letter_alphabet=Bio.Alphabet-module.html#single_letter_alphabet"><a title="Bio.Alphabet.single_letter_alphabet" class="py-name" href="#" onclick="return doclink('link-2', 'single_letter_alphabet', 'link-2');">single_letter_alphabet</a></tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-3" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-3', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-4" class="py-name" targets="Module Bio.Seq=Bio.Seq-module.html,Class Bio.Seq.Seq=Bio.Seq.Seq-class.html"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-4', 'Seq', 'link-4');">Seq</a></tt> <tt class="py-keyword">import</tt> <tt id="link-5" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-5', 'Seq', 'link-4');">Seq</a></tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-6" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-6', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-7" class="py-name" targets="Module Bio.SeqRecord=Bio.SeqRecord-module.html,Class Bio.SeqRecord.SeqRecord=Bio.SeqRecord.SeqRecord-class.html"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-7', 'SeqRecord', 'link-7');">SeqRecord</a></tt> <tt class="py-keyword">import</tt> <tt id="link-8" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-8', 'SeqRecord', 'link-7');">SeqRecord</a></tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-9" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-9', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-10" class="py-name" targets="Package Bio.Align=Bio.Align-module.html"><a title="Bio.Align" class="py-name" href="#" onclick="return doclink('link-10', 'Align', 'link-10');">Align</a></tt> <tt class="py-keyword">import</tt> <tt id="link-11" class="py-name" targets="Class Bio.Align.MultipleSeqAlignment=Bio.Align.MultipleSeqAlignment-class.html"><a title="Bio.Align.MultipleSeqAlignment" class="py-name" href="#" onclick="return doclink('link-11', 'MultipleSeqAlignment', 'link-11');">MultipleSeqAlignment</a></tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt><tt id="link-12" class="py-name" targets="Module Bio.AlignIO.Interfaces=Bio.AlignIO.Interfaces-module.html,Module Bio.SeqIO.Interfaces=Bio.SeqIO.Interfaces-module.html"><a title="Bio.AlignIO.Interfaces
Bio.SeqIO.Interfaces" class="py-name" href="#" onclick="return doclink('link-12', 'Interfaces', 'link-12');">Interfaces</a></tt> <tt class="py-keyword">import</tt> <tt id="link-13" class="py-name" targets="Class Bio.AlignIO.Interfaces.SequentialAlignmentWriter=Bio.AlignIO.Interfaces.SequentialAlignmentWriter-class.html"><a title="Bio.AlignIO.Interfaces.SequentialAlignmentWriter" class="py-name" href="#" onclick="return doclink('link-13', 'SequentialAlignmentWriter', 'link-13');">SequentialAlignmentWriter</a></tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line"> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line"><tt id="link-14" class="py-name" targets="Variable Bio.AlignIO.MafIO.MAFINDEX_VERSION=Bio.AlignIO.MafIO-module.html#MAFINDEX_VERSION"><a title="Bio.AlignIO.MafIO.MAFINDEX_VERSION" class="py-name" href="#" onclick="return doclink('link-14', 'MAFINDEX_VERSION', 'link-14');">MAFINDEX_VERSION</a></tt> <tt class="py-op">=</tt> <tt class="py-number">2</tt> </tt>
<a name="MafWriter"></a><div id="MafWriter-def"><a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line"> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line"> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt> <a class="py-toggle" href="#" id="MafWriter-toggle" onclick="return toggle('MafWriter');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafWriter-class.html">MafWriter</a><tt class="py-op">(</tt><tt class="py-base-class">SequentialAlignmentWriter</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafWriter-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="MafWriter-expanded"><a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line">    <tt class="py-docstring">"""Accepts a MultipleSeqAlignment object, writes a MAF file."""</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"> </tt>
<a name="MafWriter.write_header"></a><div id="MafWriter.write_header-def"><a name="L58"></a><tt class="py-lineno"> 58</tt> <a class="py-toggle" href="#" id="MafWriter.write_header-toggle" onclick="return toggle('MafWriter.write_header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafWriter-class.html#write_header">write_header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafWriter.write_header-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafWriter.write_header-expanded"><a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line">        <tt class="py-docstring">"""Write the MAF header."""</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-15" class="py-name" targets="Function Bio.AlignIO.write()=Bio.AlignIO-module.html#write,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.write()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#write,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.write()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#write,Method Bio.Phylo.CDAOIO.Writer.write()=Bio.Phylo.CDAOIO.Writer-class.html#write,Function Bio.Phylo.CDAOIO.write()=Bio.Phylo.CDAOIO-module.html#write,Method Bio.Phylo.NeXMLIO.Writer.write()=Bio.Phylo.NeXMLIO.Writer-class.html#write,Function Bio.Phylo.NeXMLIO.write()=Bio.Phylo.NeXMLIO-module.html#write,Method Bio.Phylo.NewickIO.Writer.write()=Bio.Phylo.NewickIO.Writer-class.html#write,Function Bio.Phylo.NewickIO.write()=Bio.Phylo.NewickIO-module.html#write,Function Bio.Phylo.NexusIO.write()=Bio.Phylo.NexusIO-module.html#write,Method Bio.Phylo.PhyloXMLIO.Writer.write()=Bio.Phylo.PhyloXMLIO.Writer-class.html#write,Function Bio.Phylo.PhyloXMLIO.write()=Bio.Phylo.PhyloXMLIO-module.html#write,Function Bio.Phylo._io.write()=Bio.Phylo._io-module.html#write,Function Bio.SearchIO.write()=Bio.SearchIO-module.html#write,Function Bio.SeqIO.write()=Bio.SeqIO-module.html#write,Method Bio.bgzf.BgzfWriter.write()=Bio.bgzf.BgzfWriter-class.html#write,Function Bio.motifs.jaspar.write()=Bio.motifs.jaspar-module.html#write,Function Bio.motifs.transfac.write()=Bio.motifs.transfac-module.html#write,Function Bio.motifs.write()=Bio.motifs-module.html#write,Method Bio.phenotype.phen_micro.JsonWriter.write()=Bio.phenotype.phen_micro.JsonWriter-class.html#write,Function Bio.phenotype.write()=Bio.phenotype-module.html#write"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-15', 'write', 'link-15');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"##maf version=1 scoring=none\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-16" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-16', 'write', 'link-15');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"# generated by Biopython\n\n"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line"> </tt>
<a name="MafWriter._write_record"></a><div id="MafWriter._write_record-def"><a name="L63"></a><tt class="py-lineno"> 63</tt> <a class="py-toggle" href="#" id="MafWriter._write_record-toggle" onclick="return toggle('MafWriter._write_record');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafWriter-class.html#_write_record">_write_record</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafWriter._write_record-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafWriter._write_record-expanded"><a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line">        <tt class="py-docstring">"""Write a single SeqRecord object to an 's' line in a MAF block (PRIVATE)."""</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line">        <tt class="py-comment"># convert biopython-style 1/-1 strand to MAF-style +/- strand</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-17" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.annotations=BioSQL.BioSeq.DBSeqRecord-class.html#annotations"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-17', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-18" class="py-name" targets="Method Bio.Crystal.Crystal.get()=Bio.Crystal.Crystal-class.html#get,Method Bio.Data.CodonTable.AmbiguousForwardTable.get()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#get,Method Bio.File._IndexedSeqFileDict.get()=Bio.File._IndexedSeqFileDict-class.html#get,Method Bio.File._IndexedSeqFileProxy.get()=Bio.File._IndexedSeqFileProxy-class.html#get,Method Bio.File._SQLiteManySeqFilesDict.get()=Bio.File._SQLiteManySeqFilesDict-class.html#get,Method Bio.Restriction.Restriction.RestrictionBatch.get()=Bio.Restriction.Restriction.RestrictionBatch-class.html#get,Method Bio.SearchIO._index.SearchIndexer.get()=Bio.SearchIO._index.SearchIndexer-class.html#get,Method Bio.SeqIO._index.SeqFileRandomAccess.get()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#get,Method Bio.SeqIO._index.SffRandomAccess.get()=Bio.SeqIO._index.SffRandomAccess-class.html#get,Method Bio.SeqIO._index.SffTrimedRandomAccess.get()=Bio.SeqIO._index.SffTrimedRandomAccess-class.html#get,Method Bio.SeqIO._index.UniprotRandomAccess.get()=Bio.SeqIO._index.UniprotRandomAccess-class.html#get"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-18', 'get', 'link-18');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"strand"</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line">            <tt id="link-19" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.strand()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#strand,Variable Bio.SeqFeature.CompoundLocation.strand=Bio.SeqFeature.CompoundLocation-class.html#strand,Variable Bio.SeqFeature.FeatureLocation.strand=Bio.SeqFeature.FeatureLocation-class.html#strand,Variable Bio.SeqFeature.SeqFeature.strand=Bio.SeqFeature.SeqFeature-class.html#strand"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-19', 'strand', 'link-19');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"+"</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-20" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-20', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-21" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-21', 'get', 'link-18');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"strand"</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line">            <tt id="link-22" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-22', 'strand', 'link-19');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"-"</tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line">            <tt class="py-comment"># TODO: issue warning?</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line">            <tt id="link-23" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-23', 'strand', 'link-19');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"+"</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line"> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">        <tt class="py-name">fields</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">"s"</tt><tt class="py-op">,</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line">                  <tt class="py-comment"># In the MAF file format, spaces are not allowed in the id</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line">                  <tt class="py-string">"%-40s"</tt> <tt class="py-op">%</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-24" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Variable Bio.PDB.Entity.Entity.id=Bio.PDB.Entity.Entity-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-24', 'id', 'link-24');">id</a></tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">" "</tt><tt class="py-op">,</tt> <tt class="py-string">"_"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line">                  <tt class="py-string">"%15s"</tt> <tt class="py-op">%</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-25" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-25', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-26" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-26', 'get', 'link-18');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"start"</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line">                  <tt class="py-string">"%5s"</tt> <tt class="py-op">%</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-27" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-27', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-28" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-28', 'get', 'link-18');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"size"</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-29" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.seq=Bio.SeqRecord.SeqRecord-class.html#seq,Variable BioSQL.BioSeq.DBSeqRecord.seq=BioSQL.BioSeq.DBSeqRecord-class.html#seq"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-29', 'seq', 'link-29');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"-"</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line">                  <tt id="link-30" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-30', 'strand', 'link-19');">strand</a></tt><tt class="py-op">,</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line">                  <tt class="py-string">"%15s"</tt> <tt class="py-op">%</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-31" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-31', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-32" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-32', 'get', 'link-18');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"srcSize"</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">                  <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-33" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-33', 'seq', 'link-29');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-34" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-34', 'write', 'link-15');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"%s\n"</tt> <tt class="py-op">%</tt> <tt class="py-string">" "</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">fields</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line"> </tt>
<a name="MafWriter.write_alignment"></a><div id="MafWriter.write_alignment-def"><a name="L84"></a><tt class="py-lineno"> 84</tt> <a class="py-toggle" href="#" id="MafWriter.write_alignment-toggle" onclick="return toggle('MafWriter.write_alignment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafWriter-class.html#write_alignment">write_alignment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">alignment</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafWriter.write_alignment-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafWriter.write_alignment-expanded"><a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line">        <tt class="py-docstring">"""Write a complete alignment to a MAF block.</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line"><tt class="py-docstring">        Writes every SeqRecord in a MultipleSeqAlignment object to its own</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line"><tt class="py-docstring">        MAF block (beginning with an 'a' line, containing 's' lines).</tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">alignment</tt><tt class="py-op">,</tt> <tt id="link-35" class="py-name"><a title="Bio.Align.MultipleSeqAlignment" class="py-name" href="#" onclick="return doclink('link-35', 'MultipleSeqAlignment', 'link-11');">MultipleSeqAlignment</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"Expected an alignment object"</tt><tt class="py-op">)</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line"> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-36" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.set()=Bio.Nexus.Nexus.StepMatrix-class.html#set,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.set()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#set,Method Bio.SVDSuperimposer.SVDSuperimposer.set()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#set"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-36', 'set', 'link-36');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-37" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.x=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#x,Variable Bio.Statistics.lowess.x=Bio.Statistics.lowess-module.html#x"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-37', 'x', 'link-37');">x</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-38" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-38', 'x', 'link-37');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">alignment</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Sequences must all be the same length"</tt><tt class="py-op">)</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line"> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line">        <tt class="py-comment"># We allow multiple sequences with the same IDs; for example, there may</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line">        <tt class="py-comment"># be a MAF aligning the + and - strands of the same sequence together.</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line"> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">        <tt class="py-comment"># for now, use ._annotations private property, but restrict keys to those</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line">        <tt class="py-comment"># specifically supported by the MAF format, according to spec</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line">            <tt class="py-name">anno</tt> <tt class="py-op">=</tt> <tt class="py-string">" "</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-string">"%s=%s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-39" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-39', 'x', 'link-37');">x</a></tt><tt class="py-op">,</tt> <tt id="link-40" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.y=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#y"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-40', 'y', 'link-40');">y</a></tt><tt class="py-op">)</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">                             <tt class="py-keyword">for</tt> <tt id="link-41" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-41', 'x', 'link-37');">x</a></tt><tt class="py-op">,</tt> <tt id="link-42" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-42', 'y', 'link-40');">y</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">alignment</tt><tt class="py-op">.</tt><tt class="py-name">_annotations</tt><tt class="py-op">.</tt><tt id="link-43" class="py-name" targets="Method Bio.Crystal.Crystal.items()=Bio.Crystal.Crystal-class.html#items,Method Bio.File._IndexedSeqFileDict.items()=Bio.File._IndexedSeqFileDict-class.html#items,Method Bio.Phylo.PhyloXML.Events.items()=Bio.Phylo.PhyloXML.Events-class.html#items,Variable Bio.SearchIO._model.query.QueryResult.items=Bio.SearchIO._model.query.QueryResult-class.html#items,Method BioSQL.BioSeqDatabase.BioSeqDatabase.items()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#items,Method BioSQL.BioSeqDatabase.DBServer.items()=BioSQL.BioSeqDatabase.DBServer-class.html#items"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-43', 'items', 'link-43');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line">                             <tt class="py-keyword">if</tt> <tt id="link-44" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-44', 'x', 'link-37');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">"score"</tt><tt class="py-op">,</tt> <tt class="py-string">"pass"</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">            <tt class="py-name">anno</tt> <tt class="py-op">=</tt> <tt class="py-string">"score=0.00"</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line"> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-45" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-45', 'write', 'link-15');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"a %s\n"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">anno</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line"> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">        <tt class="py-name">recs_out</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line"> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">alignment</tt><tt class="py-op">:</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-46" class="py-name" targets="Method Bio.AlignIO.MafIO.MafWriter._write_record()=Bio.AlignIO.MafIO.MafWriter-class.html#_write_record,Method Bio.AlignIO.MauveIO.MauveWriter._write_record()=Bio.AlignIO.MauveIO.MauveWriter-class.html#_write_record,Method Bio.AlignIO.StockholmIO.StockholmWriter._write_record()=Bio.AlignIO.StockholmIO.StockholmWriter-class.html#_write_record"><a title="Bio.AlignIO.MafIO.MafWriter._write_record
Bio.AlignIO.MauveIO.MauveWriter._write_record
Bio.AlignIO.StockholmIO.StockholmWriter._write_record" class="py-name" href="#" onclick="return doclink('link-46', '_write_record', 'link-46');">_write_record</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line"> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">            <tt class="py-name">recs_out</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line"> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-47" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-47', 'write', 'link-15');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line"> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">recs_out</tt> </tt>
</div></div><a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line"> </tt>
<a name="MafIterator"></a><div id="MafIterator-def"><a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line"> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line"><tt class="py-comment"># Invalid function name according to pylint, but kept for compatibility</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line"><tt class="py-comment"># with Bio* conventions.</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt> <a class="py-toggle" href="#" id="MafIterator-toggle" onclick="return toggle('MafIterator');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO-module.html#MafIterator">MafIterator</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">seq_count</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt id="link-48" class="py-name"><a title="Bio.Alphabet.single_letter_alphabet" class="py-name" href="#" onclick="return doclink('link-48', 'single_letter_alphabet', 'link-2');">single_letter_alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIterator-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="MafIterator-expanded"><a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line">    <tt class="py-docstring">"""Iterate over a MAF file handle as MultipleSeqAlignment objects.</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line"><tt class="py-docstring">    Iterates over lines in a MAF file-like object (handle), yielding</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line"><tt class="py-docstring">    MultipleSeqAlignment objects. SeqRecord IDs generally correspond to</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line"><tt class="py-docstring">    species names.</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line">    <tt class="py-name">in_a_bundle</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line"> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line">    <tt id="link-49" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-49', 'annotations', 'link-17');">annotations</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line">    <tt class="py-name">records</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line"> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">    <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line">        <tt class="py-comment"># allows parsing of the last bundle without duplicating code</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-50" class="py-name" targets="Method Bio.AlignIO.Interfaces.AlignmentIterator.next()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#next,Method Bio.Blast.NCBIStandalone.Iterator.next()=Bio.Blast.NCBIStandalone.Iterator-class.html#next,Method Bio.Blast.ParseBlastTable.BlastTableReader.next()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#next,Method Bio.File.UndoHandle.next()=Bio.File.UndoHandle-class.html#next,Method Bio.GenBank.Iterator.next()=Bio.GenBank.Iterator-class.html#next,Method Bio.Nexus.Nexus.CharBuffer.next()=Bio.Nexus.Nexus.CharBuffer-class.html#next,Method Bio.Nexus.StandardData.StandardData.next()=Bio.Nexus.StandardData.StandardData-class.html#next,Method Bio.PopGen.GenePop.Controller._FileIterator.next()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#next,Method Bio.SeqIO.Interfaces.SequenceIterator.next()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#next,Method Bio.bgzf.BgzfReader.next()=Bio.bgzf.BgzfReader-class.html#next"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.Nexus.StandardData.StandardData.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-50', 'next', 'link-50');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line"> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">in_a_bundle</tt><tt class="py-op">:</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-51" class="py-name" targets="Method Bio.Seq.Seq.startswith()=Bio.Seq.Seq-class.html#startswith"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-51', 'startswith', 'link-51');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"s"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">                <tt class="py-comment"># add a SeqRecord to the bundle</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">                <tt class="py-name">line_split</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-52" class="py-name" targets="Method Bio.Seq.Seq.strip()=Bio.Seq.Seq-class.html#strip"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-52', 'strip', 'link-52');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-53" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-53', 'split', 'link-53');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line"> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">line_split</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-number">7</tt><tt class="py-op">:</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Error parsing alignment - 's' line must have 7 fields"</tt><tt class="py-op">)</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line"> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">                <tt class="py-comment"># convert MAF-style +/- strand to biopython-type 1/-1</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"+"</tt><tt class="py-op">:</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">                    <tt id="link-54" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-54', 'strand', 'link-19');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"-"</tt><tt class="py-op">:</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">                    <tt id="link-55" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-55', 'strand', 'link-19');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">                    <tt class="py-comment"># TODO: issue warning, set to 0?</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">                    <tt id="link-56" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-56', 'strand', 'link-19');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line"> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">                <tt class="py-comment"># s (literal), src (ID), start, size, strand, srcSize, text (sequence)</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">                <tt class="py-name">anno</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">"start"</tt><tt class="py-op">:</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">                        <tt class="py-string">"size"</tt><tt class="py-op">:</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">                        <tt class="py-string">"strand"</tt><tt class="py-op">:</tt> <tt id="link-57" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-57', 'strand', 'link-19');">strand</a></tt><tt class="py-op">,</tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">                        <tt class="py-string">"srcSize"</tt><tt class="py-op">:</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">5</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">}</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line"> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">                <tt id="link-58" class="py-name" targets="Method Bio.FSSP.FSSPAlignDict.sequence()=Bio.FSSP.FSSPAlignDict-class.html#sequence,Method Bio.GenBank._FeatureConsumer.sequence()=Bio.GenBank._FeatureConsumer-class.html#sequence,Method Bio.GenBank._RecordConsumer.sequence()=Bio.GenBank._RecordConsumer-class.html#sequence,Method Bio.Phylo.PhyloXMLIO.Writer.sequence()=Bio.Phylo.PhyloXMLIO.Writer-class.html#sequence"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-58', 'sequence', 'link-58');">sequence</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">]</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line"> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">                <tt class="py-comment"># interpret a dot/period to mean the same as the first sequence</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-string">"."</tt> <tt class="py-keyword">in</tt> <tt id="link-59" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-59', 'sequence', 'link-58');">sequence</a></tt><tt class="py-op">:</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">records</tt><tt class="py-op">:</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line">                        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Found dot/period in first sequence of alignment"</tt><tt class="py-op">)</tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line"> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line">                    <tt id="link-60" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.ref=Bio.SeqFeature.CompoundLocation-class.html#ref,Variable Bio.SeqFeature.SeqFeature.ref=Bio.SeqFeature.SeqFeature-class.html#ref"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-60', 'ref', 'link-60');">ref</a></tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">records</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-61" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-61', 'seq', 'link-29');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line">                    <tt class="py-name">new</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line"> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-op">(</tt><tt class="py-name">letter</tt><tt class="py-op">,</tt> <tt class="py-name">ref_letter</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt id="link-62" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-62', 'sequence', 'link-58');">sequence</a></tt><tt class="py-op">,</tt> <tt id="link-63" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-63', 'ref', 'link-60');">ref</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">                        <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-64" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-64', 'append', 'link-64');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">ref_letter</tt> <tt class="py-keyword">if</tt> <tt class="py-name">letter</tt> <tt class="py-op">==</tt> <tt class="py-string">"."</tt> <tt class="py-keyword">else</tt> <tt class="py-name">letter</tt><tt class="py-op">)</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line"> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">                    <tt id="link-65" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-65', 'sequence', 'link-58');">sequence</a></tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">new</tt><tt class="py-op">)</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line"> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">                <tt class="py-name">records</tt><tt class="py-op">.</tt><tt id="link-66" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-66', 'append', 'link-64');">append</a></tt><tt class="py-op">(</tt><tt id="link-67" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-67', 'SeqRecord', 'link-7');">SeqRecord</a></tt><tt class="py-op">(</tt><tt id="link-68" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-68', 'Seq', 'link-4');">Seq</a></tt><tt class="py-op">(</tt><tt id="link-69" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-69', 'sequence', 'link-58');">sequence</a></tt><tt class="py-op">,</tt> <tt id="link-70" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.alphabet=Bio.SearchIO._model.hsp.HSP-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSPFragment.alphabet=Bio.SearchIO._model.hsp.HSPFragment-class.html#alphabet"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-70', 'alphabet', 'link-70');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">                                         <tt id="link-71" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-71', 'id', 'link-24');">id</a></tt><tt class="py-op">=</tt><tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">                                         <tt id="link-72" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-72', 'name', 'link-72');">name</a></tt><tt class="py-op">=</tt><tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">                                         <tt id="link-73" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DescriptionConsumer.description()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#description,Method Bio.Phylo.PhyloXMLIO.Writer.description()=Bio.Phylo.PhyloXMLIO.Writer-class.html#description,Variable Bio.SearchIO._model.hit.Hit.description=Bio.SearchIO._model.hit.Hit-class.html#description,Variable Bio.SearchIO._model.query.QueryResult.description=Bio.SearchIO._model.query.QueryResult-class.html#description"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-73', 'description', 'link-73');">description</a></tt><tt class="py-op">=</tt><tt class="py-string">""</tt><tt class="py-op">,</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">                                         <tt id="link-74" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-74', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">=</tt><tt class="py-name">anno</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-75" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-75', 'startswith', 'link-51');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"i"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">                <tt class="py-comment"># TODO: information about what is in the aligned species DNA before</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">                <tt class="py-comment"># and after the immediately preceding "s" line</tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">                <tt class="py-keyword">pass</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-76" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-76', 'startswith', 'link-51');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"e"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">                <tt class="py-comment"># TODO: information about the size of the gap between the alignments</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">                <tt class="py-comment"># that span the current block</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">                <tt class="py-keyword">pass</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-77" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-77', 'startswith', 'link-51');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"q"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">                <tt class="py-comment"># TODO: quality of each aligned base for the species.</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">                <tt class="py-comment"># Need to find documentation on this, looks like ASCII 0-9 or gap?</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">                <tt class="py-comment"># Can then store in each SeqRecord's .letter_annotations dictionary,</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line">                <tt class="py-comment"># perhaps as the raw string or turned into integers / None for gap?</tt> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line">                <tt class="py-keyword">pass</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-78" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-78', 'startswith', 'link-51');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"#"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line">                <tt class="py-comment"># ignore comments</tt> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line">                <tt class="py-comment"># (not sure whether comments</tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line">                <tt class="py-comment"># are in the maf specification, though)</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line">                <tt class="py-keyword">pass</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-79" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-79', 'strip', 'link-52');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line">                <tt class="py-comment"># end a bundle of records</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">seq_count</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line">                    <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">records</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">seq_count</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line"> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line">                <tt class="py-name">alignment</tt> <tt class="py-op">=</tt> <tt id="link-80" class="py-name"><a title="Bio.Align.MultipleSeqAlignment" class="py-name" href="#" onclick="return doclink('link-80', 'MultipleSeqAlignment', 'link-11');">MultipleSeqAlignment</a></tt><tt class="py-op">(</tt><tt class="py-name">records</tt><tt class="py-op">,</tt> <tt id="link-81" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-81', 'alphabet', 'link-70');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line">                <tt class="py-comment"># TODO - Introduce an annotated alignment class?</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line">                <tt class="py-comment"># See also Bio/AlignIO/FastaIO.py for same requirement.</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line">                <tt class="py-comment"># For now, store the annotation a new private property:</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line">                <tt class="py-name">alignment</tt><tt class="py-op">.</tt><tt class="py-name">_annotations</tt> <tt class="py-op">=</tt> <tt id="link-82" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-82', 'annotations', 'link-17');">annotations</a></tt> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line"> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt class="py-name">alignment</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line"> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line">                <tt class="py-name">in_a_bundle</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line"> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">                <tt id="link-83" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-83', 'annotations', 'link-17');">annotations</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line">                <tt class="py-name">records</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Error parsing alignment - unexpected line:\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-84" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-84', 'startswith', 'link-51');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"a"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line">            <tt class="py-comment"># start a bundle of records</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line">            <tt class="py-name">in_a_bundle</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line">            <tt class="py-name">annot_strings</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-85" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-85', 'strip', 'link-52');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-86" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-86', 'split', 'link-53');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">annot_strings</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-87" class="py-name" targets="Method Bio.Crystal.Chain.count()=Bio.Crystal.Chain-class.html#count,Method Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count()=Bio.NeuralNetwork.Gene.Pattern.PatternRepository-class.html#count,Method Bio.Seq.MutableSeq.count()=Bio.Seq.MutableSeq-class.html#count,Method Bio.Seq.Seq.count()=Bio.Seq.Seq-class.html#count,Method Bio.Seq.UnknownSeq.count()=Bio.Seq.UnknownSeq-class.html#count,Method Bio.motifs.Instances.count()=Bio.motifs.Instances-class.html#count"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-87', 'count', 'link-87');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">"="</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Error parsing alignment - invalid key in 'a' line"</tt><tt class="py-op">)</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line">            <tt id="link-88" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-88', 'annotations', 'link-17');">annotations</a></tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">a_string</tt><tt class="py-op">.</tt><tt id="link-89" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-89', 'split', 'link-53');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"="</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">a_string</tt> <tt class="py-keyword">in</tt> <tt class="py-name">annot_strings</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-90" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-90', 'startswith', 'link-51');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"#"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">            <tt class="py-comment"># ignore comments</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line">            <tt class="py-keyword">break</tt> </tt>
</div><a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line"> </tt>
<a name="MafIndex"></a><div id="MafIndex-def"><a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line"> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt> <a class="py-toggle" href="#" id="MafIndex-toggle" onclick="return toggle('MafIndex');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafIndex-class.html">MafIndex</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIndex-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="MafIndex-expanded"><a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line">    <tt class="py-docstring">"""Index for a MAF file.</tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line"><tt class="py-docstring">    The index is a sqlite3 database that is built upon creation of the object</tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line"><tt class="py-docstring">    if necessary, and queried when methods *search* or *get_spliced* are</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line"><tt class="py-docstring">    used.</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line"> </tt>
<a name="MafIndex.__init__"></a><div id="MafIndex.__init__-def"><a name="L246"></a><tt class="py-lineno">246</tt> <a class="py-toggle" href="#" id="MafIndex.__init__-toggle" onclick="return toggle('MafIndex.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafIndex-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sqlite_file</tt><tt class="py-op">,</tt> <tt class="py-param">maf_file</tt><tt class="py-op">,</tt> <tt class="py-param">target_seqname</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIndex.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafIndex.__init__-expanded"><a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line">        <tt class="py-docstring">"""Indexes or loads the index of a MAF file."""</tt> </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt> <tt class="py-op">=</tt> <tt class="py-name">target_seqname</tt> </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line">        <tt class="py-comment"># example: Tests/MAF/ucsc_mm9_chr10.mafindex</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_index_filename</tt> <tt class="py-op">=</tt> <tt class="py-name">sqlite_file</tt> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line">        <tt class="py-comment"># example: /home/bli/src/biopython/Tests/MAF</tt> </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_relative_path</tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">abspath</tt><tt class="py-op">(</tt><tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">dirname</tt><tt class="py-op">(</tt><tt class="py-name">sqlite_file</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line">        <tt class="py-comment"># example: Tests/MAF/ucsc_mm9_chr10.maf</tt> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_file</tt> <tt class="py-op">=</tt> <tt class="py-name">maf_file</tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line"> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_fp</tt> <tt class="py-op">=</tt> <tt id="link-91" class="py-name" targets="Function Bio.bgzf.open()=Bio.bgzf-module.html#open"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-91', 'open', 'link-91');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_file</tt><tt class="py-op">,</tt> <tt class="py-string">"r"</tt><tt class="py-op">)</tt> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line"> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line">        <tt class="py-comment"># if sqlite_file exists, use the existing db, otherwise index the file</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">isfile</tt><tt class="py-op">(</tt><tt class="py-name">sqlite_file</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt> <tt class="py-op">=</tt> <tt class="py-name">_sqlite</tt><tt class="py-op">.</tt><tt class="py-name">connect</tt><tt class="py-op">(</tt><tt class="py-name">sqlite_file</tt><tt class="py-op">)</tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_record_count</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-92" class="py-name" targets="Method Bio.AlignIO.MafIO.MafIndex.__check_existing_db()=Bio.AlignIO.MafIO.MafIndex-class.html#__check_existing_db"><a title="Bio.AlignIO.MafIO.MafIndex.__check_existing_db" class="py-name" href="#" onclick="return doclink('link-92', '__check_existing_db', 'link-92');">__check_existing_db</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt> <tt class="py-op">=</tt> <tt class="py-name">_sqlite</tt><tt class="py-op">.</tt><tt class="py-name">connect</tt><tt class="py-op">(</tt><tt class="py-name">sqlite_file</tt><tt class="py-op">)</tt> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_record_count</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-93" class="py-name" targets="Method Bio.AlignIO.MafIO.MafIndex.__make_new_index()=Bio.AlignIO.MafIO.MafIndex-class.html#__make_new_index"><a title="Bio.AlignIO.MafIO.MafIndex.__make_new_index" class="py-name" href="#" onclick="return doclink('link-93', '__make_new_index', 'link-93');">__make_new_index</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line"> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">        <tt class="py-comment"># lastly, setup a MafIterator pointing at the open maf_file</tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_mafiter</tt> <tt class="py-op">=</tt> <tt id="link-94" class="py-name" targets="Function Bio.AlignIO.MafIO.MafIterator()=Bio.AlignIO.MafIO-module.html#MafIterator"><a title="Bio.AlignIO.MafIO.MafIterator" class="py-name" href="#" onclick="return doclink('link-94', 'MafIterator', 'link-94');">MafIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_fp</tt><tt class="py-op">)</tt> </tt>
</div><a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line"> </tt>
<a name="MafIndex.__check_existing_db"></a><div id="MafIndex.__check_existing_db-def"><a name="L269"></a><tt class="py-lineno">269</tt> <a class="py-toggle" href="#" id="MafIndex.__check_existing_db-toggle" onclick="return toggle('MafIndex.__check_existing_db');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafIndex-class.html#__check_existing_db">__check_existing_db</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIndex.__check_existing_db-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafIndex.__check_existing_db-expanded"><a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">        <tt class="py-docstring">"""Perform basic sanity checks upon loading an existing index (PRIVATE)."""</tt> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line">            <tt class="py-name">idx_version</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-95" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.execute()=BioSQL.BioSeqDatabase.Adaptor-class.html#execute,Method BioSQL.BioSeqDatabase._CursorWrapper.execute()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#execute,Method BioSQL.DBUtils.Generic_dbutils.execute()=BioSQL.DBUtils.Generic_dbutils-class.html#execute,Method BioSQL.DBUtils.Sqlite_dbutils.execute()=BioSQL.DBUtils.Sqlite_dbutils-class.html#execute"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-95', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line">                <tt class="py-string">"SELECT value FROM meta_data WHERE key = 'version'"</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-96" class="py-name" targets="Method BioSQL.BioSeqDatabase._CursorWrapper.fetchone()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#fetchone"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchone" class="py-name" href="#" onclick="return doclink('link-96', 'fetchone', 'link-96');">fetchone</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">idx_version</tt> <tt class="py-op">!=</tt> <tt id="link-97" class="py-name"><a title="Bio.AlignIO.MafIO.MAFINDEX_VERSION" class="py-name" href="#" onclick="return doclink('link-97', 'MAFINDEX_VERSION', 'link-14');">MAFINDEX_VERSION</a></tt><tt class="py-op">:</tt> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line">                <tt class="py-name">msg</tt> <tt class="py-op">=</tt> <tt class="py-string">"\n"</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line">                    <tt class="py-string">"Index version (%s) incompatible with this version "</tt> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line">                    <tt class="py-string">"of MafIndex"</tt> <tt class="py-op">%</tt> <tt class="py-name">idx_version</tt><tt class="py-op">,</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line">                    <tt class="py-string">"You might erase the existing index %s "</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line">                    <tt class="py-string">"for it to be rebuilt."</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_index_filename</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-name">msg</tt><tt class="py-op">)</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line"> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line">            <tt class="py-name">filename</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-98" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-98', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line">                <tt class="py-string">"SELECT value FROM meta_data WHERE key = 'filename'"</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-99" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchone" class="py-name" href="#" onclick="return doclink('link-99', 'fetchone', 'link-96');">fetchone</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">            <tt class="py-comment"># Compute absolute path of the original maf file</tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">isabs</tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line">                <tt class="py-comment"># It was already stored as absolute</tt> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">                <tt class="py-name">tmp_mafpath</tt> <tt class="py-op">=</tt> <tt class="py-name">filename</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">                <tt class="py-comment"># It should otherwise have been stored as relative to the index</tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">                <tt class="py-comment"># Would be stored with Unix / path separator, so convert</tt> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line">                <tt class="py-comment"># it to the local OS path separator here:</tt> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line">                <tt class="py-name">tmp_mafpath</tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt> </tt>
<a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_relative_path</tt><tt class="py-op">,</tt> <tt class="py-name">filename</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"/"</tt><tt class="py-op">,</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">sep</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">tmp_mafpath</tt> <tt class="py-op">!=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">abspath</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_file</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line">                <tt class="py-comment"># Original and given absolute paths differ.</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Index uses a different file (%s != %s)"</tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line">                                 <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_file</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line"> </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line">            <tt class="py-name">db_target</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-100" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-100', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line">                <tt class="py-string">"SELECT value FROM meta_data WHERE key = 'target_seqname'"</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-101" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchone" class="py-name" href="#" onclick="return doclink('link-101', 'fetchone', 'link-96');">fetchone</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">db_target</tt> <tt class="py-op">!=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">:</tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Provided database indexed for %s, expected %s"</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line">                                 <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">db_target</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line"> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line">            <tt class="py-name">record_count</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-102" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-102', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line">                <tt class="py-string">"SELECT value FROM meta_data WHERE key = 'record_count'"</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-103" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchone" class="py-name" href="#" onclick="return doclink('link-103', 'fetchone', 'link-96');">fetchone</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">record_count</tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unfinished/partial database provided"</tt><tt class="py-op">)</tt> </tt>
<a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line"> </tt>
<a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line">            <tt class="py-name">records_found</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-104" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-104', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line">                <tt class="py-string">"SELECT COUNT(*) FROM offset_data"</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-105" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchone" class="py-name" href="#" onclick="return doclink('link-105', 'fetchone', 'link-96');">fetchone</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">records_found</tt> <tt class="py-op">!=</tt> <tt class="py-name">record_count</tt><tt class="py-op">:</tt> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Expected %s records, found %s.  Corrupt index?"</tt> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line">                                 <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">record_count</tt><tt class="py-op">,</tt> <tt class="py-name">records_found</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line"> </tt>
<a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">records_found</tt> </tt>
<a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line"> </tt>
<a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-op">(</tt><tt class="py-name">_sqlite</tt><tt class="py-op">.</tt><tt class="py-name">OperationalError</tt><tt class="py-op">,</tt> <tt class="py-name">_sqlite</tt><tt class="py-op">.</tt><tt class="py-name">DatabaseError</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">err</tt><tt class="py-op">:</tt> </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Problem with SQLite database: %s"</tt> <tt class="py-op">%</tt> <tt class="py-name">err</tt><tt class="py-op">)</tt> </tt>
</div><a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line"> </tt>
<a name="MafIndex.__make_new_index"></a><div id="MafIndex.__make_new_index-def"><a name="L321"></a><tt class="py-lineno">321</tt> <a class="py-toggle" href="#" id="MafIndex.__make_new_index-toggle" onclick="return toggle('MafIndex.__make_new_index');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafIndex-class.html#__make_new_index">__make_new_index</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIndex.__make_new_index-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafIndex.__make_new_index-expanded"><a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">        <tt class="py-docstring">"""Read MAF file and generate SQLite index (PRIVATE)."""</tt> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">        <tt class="py-comment"># make the tables</tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-106" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-106', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"CREATE TABLE meta_data (key TEXT, value TEXT);"</tt><tt class="py-op">)</tt> </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-107" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-107', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"INSERT INTO meta_data (key, value) VALUES ('version', %s);"</tt> <tt class="py-op">%</tt> <tt id="link-108" class="py-name"><a title="Bio.AlignIO.MafIO.MAFINDEX_VERSION" class="py-name" href="#" onclick="return doclink('link-108', 'MAFINDEX_VERSION', 'link-14');">MAFINDEX_VERSION</a></tt><tt class="py-op">)</tt> </tt>
<a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-109" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-109', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"INSERT INTO meta_data (key, value) VALUES ('record_count', -1);"</tt><tt class="py-op">)</tt> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-110" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-110', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"INSERT INTO meta_data (key, value) VALUES ('target_seqname', '%s');"</tt> <tt class="py-op">%</tt> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line">                          <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line">        <tt class="py-comment"># Determine whether to store maf file as relative to the index or absolute</tt> </tt>
<a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line">        <tt class="py-comment"># See https://github.com/biopython/biopython/pull/381</tt> </tt>
<a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">isabs</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_file</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">isabs</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_index_filename</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line">            <tt class="py-comment"># Since the user gave both maf file and index as relative paths,</tt> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line">            <tt class="py-comment"># we will store the maf file relative to the index.</tt> </tt>
<a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line">            <tt class="py-comment"># Note for cross platform use (e.g. shared drive over SAMBA),</tt> </tt>
<a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line">            <tt class="py-comment"># convert any Windows slash into Unix style for rel paths.</tt> </tt>
<a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line">            <tt class="py-comment"># example: ucsc_mm9_chr10.maf</tt> </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line">            <tt class="py-name">mafpath</tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">relpath</tt><tt class="py-op">(</tt> </tt>
<a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_file</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_relative_path</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">sep</tt><tt class="py-op">,</tt> <tt class="py-string">"/"</tt><tt class="py-op">)</tt> </tt>
<a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-op">(</tt><tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">dirname</tt><tt class="py-op">(</tt><tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">abspath</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_file</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> </tt>
<a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line">              <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">sep</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-111" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-111', 'startswith', 'link-51');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_relative_path</tt> <tt class="py-op">+</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">sep</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line">            <tt class="py-comment"># Since maf file is in same directory or sub directory,</tt> </tt>
<a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line">            <tt class="py-comment"># might as well make this into a relative path:</tt> </tt>
<a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line">            <tt class="py-name">mafpath</tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">relpath</tt><tt class="py-op">(</tt> </tt>
<a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_file</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_relative_path</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">sep</tt><tt class="py-op">,</tt> <tt class="py-string">"/"</tt><tt class="py-op">)</tt> </tt>
<a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line">            <tt class="py-comment"># Default to storing as an absolute path</tt> </tt>
<a name="L347"></a><tt class="py-lineno">347</tt>  <tt class="py-line">            <tt class="py-comment"># example: /home/bli/src/biopython/Tests/MAF/ucsc_mm9_chr10.maf</tt> </tt>
<a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line">            <tt class="py-name">mafpath</tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">abspath</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_file</tt><tt class="py-op">)</tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-112" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-112', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"INSERT INTO meta_data (key, value) VALUES ('filename', '%s');"</tt> <tt class="py-op">%</tt> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line">                          <tt class="py-op">(</tt><tt class="py-name">mafpath</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-113" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-113', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"CREATE TABLE offset_data (bin INTEGER, start INTEGER, end INTEGER, offset INTEGER);"</tt><tt class="py-op">)</tt> </tt>
<a name="L352"></a><tt class="py-lineno">352</tt>  <tt class="py-line"> </tt>
<a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line">        <tt class="py-name">insert_count</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line"> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">        <tt class="py-comment"># iterate over the entire file and insert in batches</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line">        <tt class="py-name">mafindex_func</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-114" class="py-name" targets="Method Bio.AlignIO.MafIO.MafIndex.__maf_indexer()=Bio.AlignIO.MafIO.MafIndex-class.html#__maf_indexer"><a title="Bio.AlignIO.MafIO.MafIndex.__maf_indexer" class="py-name" href="#" onclick="return doclink('link-114', '__maf_indexer', 'link-114');">__maf_indexer</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line"> </tt>
<a name="L358"></a><tt class="py-lineno">358</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line">            <tt class="py-name">batch</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">islice</tt><tt class="py-op">(</tt><tt class="py-name">mafindex_func</tt><tt class="py-op">,</tt> <tt class="py-number">100</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">batch</tt><tt class="py-op">:</tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line"> </tt>
<a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line">            <tt class="py-comment"># batch is made from self.__maf_indexer(),</tt> </tt>
<a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line">            <tt class="py-comment"># which yields zero-based "inclusive" start and end coordinates</tt> </tt>
<a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-115" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.executemany()=BioSQL.BioSeqDatabase.Adaptor-class.html#executemany,Method BioSQL.BioSeqDatabase._CursorWrapper.executemany()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#executemany,Method BioSQL.DBUtils.Generic_dbutils.executemany()=BioSQL.DBUtils.Generic_dbutils-class.html#executemany,Method BioSQL.DBUtils.Sqlite_dbutils.executemany()=BioSQL.DBUtils.Sqlite_dbutils-class.html#executemany"><a title="BioSQL.BioSeqDatabase.Adaptor.executemany
BioSQL.BioSeqDatabase._CursorWrapper.executemany
BioSQL.DBUtils.Generic_dbutils.executemany
BioSQL.DBUtils.Sqlite_dbutils.executemany" class="py-name" href="#" onclick="return doclink('link-115', 'executemany', 'link-115');">executemany</a></tt><tt class="py-op">(</tt> </tt>
<a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line">                <tt class="py-string">"INSERT INTO offset_data (bin, start, end, offset) VALUES (?,?,?,?);"</tt><tt class="py-op">,</tt> <tt class="py-name">batch</tt><tt class="py-op">)</tt> </tt>
<a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-116" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.commit()=BioSQL.BioSeqDatabase.Adaptor-class.html#commit,Method BioSQL.BioSeqDatabase.DBServer.commit()=BioSQL.BioSeqDatabase.DBServer-class.html#commit"><a title="BioSQL.BioSeqDatabase.Adaptor.commit
BioSQL.BioSeqDatabase.DBServer.commit" class="py-name" href="#" onclick="return doclink('link-116', 'commit', 'link-116');">commit</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L368"></a><tt class="py-lineno">368</tt>  <tt class="py-line">            <tt class="py-name">insert_count</tt> <tt class="py-op">+=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">batch</tt><tt class="py-op">)</tt> </tt>
<a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line"> </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line">        <tt class="py-comment"># then make indexes on the relevant fields</tt> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-117" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-117', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"CREATE INDEX IF NOT EXISTS bin_index ON offset_data(bin);"</tt><tt class="py-op">)</tt> </tt>
<a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-118" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-118', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"CREATE INDEX IF NOT EXISTS start_index ON offset_data(start);"</tt><tt class="py-op">)</tt> </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-119" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-119', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">"CREATE INDEX IF NOT EXISTS end_index ON offset_data(end);"</tt><tt class="py-op">)</tt> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line"> </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-120" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-120', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line">            <tt class="py-string">"UPDATE meta_data SET value = '%s' WHERE key = 'record_count'"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">insert_count</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line"> </tt>
<a name="L378"></a><tt class="py-lineno">378</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt><tt class="py-op">.</tt><tt id="link-121" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.commit
BioSQL.BioSeqDatabase.DBServer.commit" class="py-name" href="#" onclick="return doclink('link-121', 'commit', 'link-116');">commit</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L379"></a><tt class="py-lineno">379</tt>  <tt class="py-line"> </tt>
<a name="L380"></a><tt class="py-lineno">380</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">insert_count</tt> </tt>
</div><a name="L381"></a><tt class="py-lineno">381</tt>  <tt class="py-line"> </tt>
<a name="MafIndex.__maf_indexer"></a><div id="MafIndex.__maf_indexer-def"><a name="L382"></a><tt class="py-lineno">382</tt> <a class="py-toggle" href="#" id="MafIndex.__maf_indexer-toggle" onclick="return toggle('MafIndex.__maf_indexer');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafIndex-class.html#__maf_indexer">__maf_indexer</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIndex.__maf_indexer-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafIndex.__maf_indexer-expanded"><a name="L383"></a><tt class="py-lineno">383</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return index information for each bundle (PRIVATE).</tt> </tt>
<a name="L384"></a><tt class="py-lineno">384</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L385"></a><tt class="py-lineno">385</tt>  <tt class="py-line"><tt class="py-docstring">        Yields index information for each bundle in the form of</tt> </tt>
<a name="L386"></a><tt class="py-lineno">386</tt>  <tt class="py-line"><tt class="py-docstring">        (bin, start, end, offset) tuples where start and end are</tt> </tt>
<a name="L387"></a><tt class="py-lineno">387</tt>  <tt class="py-line"><tt class="py-docstring">        0-based inclusive coordinates.</tt> </tt>
<a name="L388"></a><tt class="py-lineno">388</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L389"></a><tt class="py-lineno">389</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_fp</tt><tt class="py-op">.</tt><tt id="link-122" class="py-name" targets="Method Bio.File.UndoHandle.readline()=Bio.File.UndoHandle-class.html#readline,Method Bio.bgzf.BgzfReader.readline()=Bio.bgzf.BgzfReader-class.html#readline"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-122', 'readline', 'link-122');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L390"></a><tt class="py-lineno">390</tt>  <tt class="py-line"> </tt>
<a name="L391"></a><tt class="py-lineno">391</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L392"></a><tt class="py-lineno">392</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-123" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-123', 'startswith', 'link-51');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"a"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L393"></a><tt class="py-lineno">393</tt>  <tt class="py-line">                <tt class="py-comment"># note the offset</tt> </tt>
<a name="L394"></a><tt class="py-lineno">394</tt>  <tt class="py-line">                <tt class="py-name">offset</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_fp</tt><tt class="py-op">.</tt><tt id="link-124" class="py-name" targets="Method Bio.File.UndoHandle.tell()=Bio.File.UndoHandle-class.html#tell,Method Bio.SeqIO.SffIO._AddTellHandle.tell()=Bio.SeqIO.SffIO._AddTellHandle-class.html#tell,Method Bio.bgzf.BgzfReader.tell()=Bio.bgzf.BgzfReader-class.html#tell,Method Bio.bgzf.BgzfWriter.tell()=Bio.bgzf.BgzfWriter-class.html#tell"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-124', 'tell', 'link-124');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L395"></a><tt class="py-lineno">395</tt>  <tt class="py-line"> </tt>
<a name="L396"></a><tt class="py-lineno">396</tt>  <tt class="py-line">                <tt class="py-comment"># search the following lines for a match to target_seqname</tt> </tt>
<a name="L397"></a><tt class="py-lineno">397</tt>  <tt class="py-line">                <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L398"></a><tt class="py-lineno">398</tt>  <tt class="py-line">                    <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_fp</tt><tt class="py-op">.</tt><tt id="link-125" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-125', 'readline', 'link-122');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L399"></a><tt class="py-lineno">399</tt>  <tt class="py-line"> </tt>
<a name="L400"></a><tt class="py-lineno">400</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-126" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-126', 'strip', 'link-52');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-127" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-127', 'startswith', 'link-51');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"a"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L401"></a><tt class="py-lineno">401</tt>  <tt class="py-line">                        <tt class="py-comment"># Empty line or new alignment record</tt> </tt>
<a name="L402"></a><tt class="py-lineno">402</tt>  <tt class="py-line">                        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Target for indexing (%s) not found in this bundle"</tt> </tt>
<a name="L403"></a><tt class="py-lineno">403</tt>  <tt class="py-line">                                         <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L404"></a><tt class="py-lineno">404</tt>  <tt class="py-line">                    <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-128" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-128', 'startswith', 'link-51');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"s"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L405"></a><tt class="py-lineno">405</tt>  <tt class="py-line">                        <tt class="py-comment"># s (literal), src (ID), start, size, strand, srcSize, text (sequence)</tt> </tt>
<a name="L406"></a><tt class="py-lineno">406</tt>  <tt class="py-line">                        <tt class="py-name">line_split</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-129" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-129', 'strip', 'link-52');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-130" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-130', 'split', 'link-53');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L407"></a><tt class="py-lineno">407</tt>  <tt class="py-line"> </tt>
<a name="L408"></a><tt class="py-lineno">408</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">:</tt> </tt>
<a name="L409"></a><tt class="py-lineno">409</tt>  <tt class="py-line">                            <tt id="link-131" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.start=Bio.SeqFeature.CompoundLocation-class.html#start,Variable Bio.SeqFeature.FeatureLocation.start=Bio.SeqFeature.FeatureLocation-class.html#start"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-131', 'start', 'link-131');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L410"></a><tt class="py-lineno">410</tt>  <tt class="py-line">                            <tt id="link-132" class="py-name" targets="Variable Bio.Alphabet.Alphabet.size=Bio.Alphabet.Alphabet-class.html#size,Variable Bio.Alphabet.Reduced.HPModel.size=Bio.Alphabet.Reduced.HPModel-class.html#size,Variable Bio.Alphabet.Reduced.Murphy10.size=Bio.Alphabet.Reduced.Murphy10-class.html#size,Variable Bio.Alphabet.Reduced.Murphy15.size=Bio.Alphabet.Reduced.Murphy15-class.html#size,Variable Bio.Alphabet.Reduced.Murphy4.size=Bio.Alphabet.Reduced.Murphy4-class.html#size,Variable Bio.Alphabet.Reduced.Murphy8.size=Bio.Alphabet.Reduced.Murphy8-class.html#size,Variable Bio.Alphabet.Reduced.PC5.size=Bio.Alphabet.Reduced.PC5-class.html#size,Variable Bio.Alphabet.SingleLetterAlphabet.size=Bio.Alphabet.SingleLetterAlphabet-class.html#size,Variable Bio.Alphabet.ThreeLetterProtein.size=Bio.Alphabet.ThreeLetterProtein-class.html#size,Method Bio.GenBank._FeatureConsumer.size()=Bio.GenBank._FeatureConsumer-class.html#size,Method Bio.GenBank._RecordConsumer.size()=Bio.GenBank._RecordConsumer-class.html#size,Variable Bio.Restriction.Restriction.AanI.size=Bio.Restriction.Restriction.AanI-class.html#size,Variable Bio.Restriction.Restriction.AarI.size=Bio.Restriction.Restriction.AarI-class.html#size,Variable Bio.Restriction.Restriction.AasI.size=Bio.Restriction.Restriction.AasI-class.html#size,Variable Bio.Restriction.Restriction.AatII.size=Bio.Restriction.Restriction.AatII-class.html#size,Variable Bio.Restriction.Restriction.Aba6411II.size=Bio.Restriction.Restriction.Aba6411II-class.html#size,Variable Bio.Restriction.Restriction.AbaCIII.size=Bio.Restriction.Restriction.AbaCIII-class.html#size,Variable Bio.Restriction.Restriction.AbaSI.size=Bio.Restriction.Restriction.AbaSI-class.html#size,Variable Bio.Restriction.Restriction.AbaUMB2I.size=Bio.Restriction.Restriction.AbaUMB2I-class.html#size,Variable Bio.Restriction.Restriction.AbsI.size=Bio.Restriction.Restriction.AbsI-class.html#size,Variable Bio.Restriction.Restriction.Acc16I.size=Bio.Restriction.Restriction.Acc16I-class.html#size,Variable Bio.Restriction.Restriction.Acc36I.size=Bio.Restriction.Restriction.Acc36I-class.html#size,Variable Bio.Restriction.Restriction.Acc65I.size=Bio.Restriction.Restriction.Acc65I-class.html#size,Variable Bio.Restriction.Restriction.Acc65V.size=Bio.Restriction.Restriction.Acc65V-class.html#size,Variable Bio.Restriction.Restriction.AccB1I.size=Bio.Restriction.Restriction.AccB1I-class.html#size,Variable Bio.Restriction.Restriction.AccB7I.size=Bio.Restriction.Restriction.AccB7I-class.html#size,Variable Bio.Restriction.Restriction.AccBSI.size=Bio.Restriction.Restriction.AccBSI-class.html#size,Variable Bio.Restriction.Restriction.AccI.size=Bio.Restriction.Restriction.AccI-class.html#size,Variable Bio.Restriction.Restriction.AccII.size=Bio.Restriction.Restriction.AccII-class.html#size,Variable Bio.Restriction.Restriction.AccIII.size=Bio.Restriction.Restriction.AccIII-class.html#size,Variable Bio.Restriction.Restriction.AceIII.size=Bio.Restriction.Restriction.AceIII-class.html#size,Variable Bio.Restriction.Restriction.AchA6III.size=Bio.Restriction.Restriction.AchA6III-class.html#size,Variable Bio.Restriction.Restriction.AciI.size=Bio.Restriction.Restriction.AciI-class.html#size,Variable Bio.Restriction.Restriction.AclI.size=Bio.Restriction.Restriction.AclI-class.html#size,Variable Bio.Restriction.Restriction.AclWI.size=Bio.Restriction.Restriction.AclWI-class.html#size,Variable Bio.Restriction.Restriction.Aco12261II.size=Bio.Restriction.Restriction.Aco12261II-class.html#size,Variable Bio.Restriction.Restriction.AcoI.size=Bio.Restriction.Restriction.AcoI-class.html#size,Variable Bio.Restriction.Restriction.AcsI.size=Bio.Restriction.Restriction.AcsI-class.html#size,Variable Bio.Restriction.Restriction.AcuI.size=Bio.Restriction.Restriction.AcuI-class.html#size,Variable Bio.Restriction.Restriction.AcvI.size=Bio.Restriction.Restriction.AcvI-class.html#size,Variable Bio.Restriction.Restriction.AcyI.size=Bio.Restriction.Restriction.AcyI-class.html#size,Variable Bio.Restriction.Restriction.AdeI.size=Bio.Restriction.Restriction.AdeI-class.html#size,Variable Bio.Restriction.Restriction.Adh6U21I.size=Bio.Restriction.Restriction.Adh6U21I-class.html#size,Variable Bio.Restriction.Restriction.AfaI.size=Bio.Restriction.Restriction.AfaI-class.html#size,Variable Bio.Restriction.Restriction.AfeI.size=Bio.Restriction.Restriction.AfeI-class.html#size,Variable Bio.Restriction.Restriction.AfiI.size=Bio.Restriction.Restriction.AfiI-class.html#size,Variable Bio.Restriction.Restriction.AflII.size=Bio.Restriction.Restriction.AflII-class.html#size,Variable Bio.Restriction.Restriction.AflIII.size=Bio.Restriction.Restriction.AflIII-class.html#size,Variable Bio.Restriction.Restriction.AgeI.size=Bio.Restriction.Restriction.AgeI-class.html#size,Variable Bio.Restriction.Restriction.AgsI.size=Bio.Restriction.Restriction.AgsI-class.html#size,Variable Bio.Restriction.Restriction.AhaIII.size=Bio.Restriction.Restriction.AhaIII-class.html#size,Variable Bio.Restriction.Restriction.AhdI.size=Bio.Restriction.Restriction.AhdI-class.html#size,Variable Bio.Restriction.Restriction.AhlI.size=Bio.Restriction.Restriction.AhlI-class.html#size,Variable Bio.Restriction.Restriction.AhyRBAHI.size=Bio.Restriction.Restriction.AhyRBAHI-class.html#size,Variable Bio.Restriction.Restriction.AhyYL17I.size=Bio.Restriction.Restriction.AhyYL17I-class.html#size,Variable Bio.Restriction.Restriction.AjiI.size=Bio.Restriction.Restriction.AjiI-class.html#size,Variable Bio.Restriction.Restriction.AjnI.size=Bio.Restriction.Restriction.AjnI-class.html#size,Variable Bio.Restriction.Restriction.AjuI.size=Bio.Restriction.Restriction.AjuI-class.html#size,Variable Bio.Restriction.Restriction.AleI.size=Bio.Restriction.Restriction.AleI-class.html#size,Variable Bio.Restriction.Restriction.AlfI.size=Bio.Restriction.Restriction.AlfI-class.html#size,Variable Bio.Restriction.Restriction.AloI.size=Bio.Restriction.Restriction.AloI-class.html#size,Variable Bio.Restriction.Restriction.AluBI.size=Bio.Restriction.Restriction.AluBI-class.html#size,Variable Bio.Restriction.Restriction.AluI.size=Bio.Restriction.Restriction.AluI-class.html#size,Variable Bio.Restriction.Restriction.Alw21I.size=Bio.Restriction.Restriction.Alw21I-class.html#size,Variable Bio.Restriction.Restriction.Alw26I.size=Bio.Restriction.Restriction.Alw26I-class.html#size,Variable Bio.Restriction.Restriction.Alw44I.size=Bio.Restriction.Restriction.Alw44I-class.html#size,Variable Bio.Restriction.Restriction.AlwFI.size=Bio.Restriction.Restriction.AlwFI-class.html#size,Variable Bio.Restriction.Restriction.AlwI.size=Bio.Restriction.Restriction.AlwI-class.html#size,Variable Bio.Restriction.Restriction.AlwNI.size=Bio.Restriction.Restriction.AlwNI-class.html#size,Variable Bio.Restriction.Restriction.Ama87I.size=Bio.Restriction.Restriction.Ama87I-class.html#size,Variable Bio.Restriction.Restriction.AmaCSI.size=Bio.Restriction.Restriction.AmaCSI-class.html#size,Variable Bio.Restriction.Restriction.Aor13HI.size=Bio.Restriction.Restriction.Aor13HI-class.html#size,Variable Bio.Restriction.Restriction.Aor51HI.size=Bio.Restriction.Restriction.Aor51HI-class.html#size,Variable Bio.Restriction.Restriction.AoxI.size=Bio.Restriction.Restriction.AoxI-class.html#size,Variable Bio.Restriction.Restriction.ApaBI.size=Bio.Restriction.Restriction.ApaBI-class.html#size,Variable Bio.Restriction.Restriction.ApaI.size=Bio.Restriction.Restriction.ApaI-class.html#size,Variable Bio.Restriction.Restriction.ApaLI.size=Bio.Restriction.Restriction.ApaLI-class.html#size,Variable Bio.Restriction.Restriction.ApeKI.size=Bio.Restriction.Restriction.ApeKI-class.html#size,Variable Bio.Restriction.Restriction.ApoI.size=Bio.Restriction.Restriction.ApoI-class.html#size,Variable Bio.Restriction.Restriction.ApyPI.size=Bio.Restriction.Restriction.ApyPI-class.html#size,Variable Bio.Restriction.Restriction.AquII.size=Bio.Restriction.Restriction.AquII-class.html#size,Variable Bio.Restriction.Restriction.AquIII.size=Bio.Restriction.Restriction.AquIII-class.html#size,Variable Bio.Restriction.Restriction.AquIV.size=Bio.Restriction.Restriction.AquIV-class.html#size,Variable Bio.Restriction.Restriction.ArsI.size=Bio.Restriction.Restriction.ArsI-class.html#size,Variable Bio.Restriction.Restriction.AscI.size=Bio.Restriction.Restriction.AscI-class.html#size,Variable Bio.Restriction.Restriction.AseI.size=Bio.Restriction.Restriction.AseI-class.html#size,Variable Bio.Restriction.Restriction.Asi256I.size=Bio.Restriction.Restriction.Asi256I-class.html#size,Variable Bio.Restriction.Restriction.AsiGI.size=Bio.Restriction.Restriction.AsiGI-class.html#size,Variable Bio.Restriction.Restriction.AsiSI.size=Bio.Restriction.Restriction.AsiSI-class.html#size,Variable Bio.Restriction.Restriction.Asp114II.size=Bio.Restriction.Restriction.Asp114II-class.html#size,Variable Bio.Restriction.Restriction.Asp700I.size=Bio.Restriction.Restriction.Asp700I-class.html#size,Variable Bio.Restriction.Restriction.Asp718I.size=Bio.Restriction.Restriction.Asp718I-class.html#size,Variable Bio.Restriction.Restriction.AspA2I.size=Bio.Restriction.Restriction.AspA2I-class.html#size,Variable Bio.Restriction.Restriction.AspBHI.size=Bio.Restriction.Restriction.AspBHI-class.html#size,Variable Bio.Restriction.Restriction.AspDUT2V.size=Bio.Restriction.Restriction.AspDUT2V-class.html#size,Variable Bio.Restriction.Restriction.AspJHL3II.size=Bio.Restriction.Restriction.AspJHL3II-class.html#size,Variable Bio.Restriction.Restriction.AspLEI.size=Bio.Restriction.Restriction.AspLEI-class.html#size,Variable Bio.Restriction.Restriction.AspS9I.size=Bio.Restriction.Restriction.AspS9I-class.html#size,Variable Bio.Restriction.Restriction.Asu14238IV.size=Bio.Restriction.Restriction.Asu14238IV-class.html#size,Variable Bio.Restriction.Restriction.AsuC2I.size=Bio.Restriction.Restriction.AsuC2I-class.html#size,Variable Bio.Restriction.Restriction.AsuHPI.size=Bio.Restriction.Restriction.AsuHPI-class.html#size,Variable Bio.Restriction.Restriction.AsuI.size=Bio.Restriction.Restriction.AsuI-class.html#size,Variable Bio.Restriction.Restriction.AsuII.size=Bio.Restriction.Restriction.AsuII-class.html#size,Variable Bio.Restriction.Restriction.AsuNHI.size=Bio.Restriction.Restriction.AsuNHI-class.html#size,Variable Bio.Restriction.Restriction.AteTI.size=Bio.Restriction.Restriction.AteTI-class.html#size,Variable Bio.Restriction.Restriction.AvaI.size=Bio.Restriction.Restriction.AvaI-class.html#size,Variable Bio.Restriction.Restriction.AvaII.size=Bio.Restriction.Restriction.AvaII-class.html#size,Variable Bio.Restriction.Restriction.AvaIII.size=Bio.Restriction.Restriction.AvaIII-class.html#size,Variable Bio.Restriction.Restriction.AvrII.size=Bio.Restriction.Restriction.AvrII-class.html#size,Variable Bio.Restriction.Restriction.Awo1030IV.size=Bio.Restriction.Restriction.Awo1030IV-class.html#size,Variable Bio.Restriction.Restriction.AxyI.size=Bio.Restriction.Restriction.AxyI-class.html#size,Variable Bio.Restriction.Restriction.BaeGI.size=Bio.Restriction.Restriction.BaeGI-class.html#size,Variable Bio.Restriction.Restriction.BaeI.size=Bio.Restriction.Restriction.BaeI-class.html#size,Variable Bio.Restriction.Restriction.Bag18758I.size=Bio.Restriction.Restriction.Bag18758I-class.html#size,Variable Bio.Restriction.Restriction.BalI.size=Bio.Restriction.Restriction.BalI-class.html#size,Variable Bio.Restriction.Restriction.BamHI.size=Bio.Restriction.Restriction.BamHI-class.html#size,Variable Bio.Restriction.Restriction.BanI.size=Bio.Restriction.Restriction.BanI-class.html#size,Variable Bio.Restriction.Restriction.BanII.size=Bio.Restriction.Restriction.BanII-class.html#size,Variable Bio.Restriction.Restriction.BanLI.size=Bio.Restriction.Restriction.BanLI-class.html#size,Variable Bio.Restriction.Restriction.BarI.size=Bio.Restriction.Restriction.BarI-class.html#size,Variable Bio.Restriction.Restriction.BauI.size=Bio.Restriction.Restriction.BauI-class.html#size,Variable Bio.Restriction.Restriction.BbrPI.size=Bio.Restriction.Restriction.BbrPI-class.html#size,Variable Bio.Restriction.Restriction.BbsI.size=Bio.Restriction.Restriction.BbsI-class.html#size,Variable Bio.Restriction.Restriction.Bbv12I.size=Bio.Restriction.Restriction.Bbv12I-class.html#size,Variable Bio.Restriction.Restriction.BbvCI.size=Bio.Restriction.Restriction.BbvCI-class.html#size,Variable Bio.Restriction.Restriction.BbvI.size=Bio.Restriction.Restriction.BbvI-class.html#size,Variable Bio.Restriction.Restriction.BbvII.size=Bio.Restriction.Restriction.BbvII-class.html#size,Variable Bio.Restriction.Restriction.BccI.size=Bio.Restriction.Restriction.BccI-class.html#size,Variable Bio.Restriction.Restriction.Bce3081I.size=Bio.Restriction.Restriction.Bce3081I-class.html#size,Variable Bio.Restriction.Restriction.Bce83I.size=Bio.Restriction.Restriction.Bce83I-class.html#size,Variable Bio.Restriction.Restriction.BceAI.size=Bio.Restriction.Restriction.BceAI-class.html#size,Variable Bio.Restriction.Restriction.BceSIV.size=Bio.Restriction.Restriction.BceSIV-class.html#size,Variable Bio.Restriction.Restriction.BcefI.size=Bio.Restriction.Restriction.BcefI-class.html#size,Variable Bio.Restriction.Restriction.BcgI.size=Bio.Restriction.Restriction.BcgI-class.html#size,Variable Bio.Restriction.Restriction.BciT130I.size=Bio.Restriction.Restriction.BciT130I-class.html#size,Variable Bio.Restriction.Restriction.BciVI.size=Bio.Restriction.Restriction.BciVI-class.html#size,Variable Bio.Restriction.Restriction.BclI.size=Bio.Restriction.Restriction.BclI-class.html#size,Variable Bio.Restriction.Restriction.BcnI.size=Bio.Restriction.Restriction.BcnI-class.html#size,Variable Bio.Restriction.Restriction.BcoDI.size=Bio.Restriction.Restriction.BcoDI-class.html#size,Variable Bio.Restriction.Restriction.BcuI.size=Bio.Restriction.Restriction.BcuI-class.html#size,Variable Bio.Restriction.Restriction.BdaI.size=Bio.Restriction.Restriction.BdaI-class.html#size,Variable Bio.Restriction.Restriction.BetI.size=Bio.Restriction.Restriction.BetI-class.html#size,Variable Bio.Restriction.Restriction.BfaI.size=Bio.Restriction.Restriction.BfaI-class.html#size,Variable Bio.Restriction.Restriction.BfaSII.size=Bio.Restriction.Restriction.BfaSII-class.html#size,Variable Bio.Restriction.Restriction.BfiI.size=Bio.Restriction.Restriction.BfiI-class.html#size,Variable Bio.Restriction.Restriction.BfmI.size=Bio.Restriction.Restriction.BfmI-class.html#size,Variable Bio.Restriction.Restriction.BfoI.size=Bio.Restriction.Restriction.BfoI-class.html#size,Variable Bio.Restriction.Restriction.BfrI.size=Bio.Restriction.Restriction.BfrI-class.html#size,Variable Bio.Restriction.Restriction.BfuAI.size=Bio.Restriction.Restriction.BfuAI-class.html#size,Variable Bio.Restriction.Restriction.BfuI.size=Bio.Restriction.Restriction.BfuI-class.html#size,Variable Bio.Restriction.Restriction.BglI.size=Bio.Restriction.Restriction.BglI-class.html#size,Variable Bio.Restriction.Restriction.BglII.size=Bio.Restriction.Restriction.BglII-class.html#size,Variable Bio.Restriction.Restriction.BinI.size=Bio.Restriction.Restriction.BinI-class.html#size,Variable Bio.Restriction.Restriction.BisI.size=Bio.Restriction.Restriction.BisI-class.html#size,Variable Bio.Restriction.Restriction.Ble402II.size=Bio.Restriction.Restriction.Ble402II-class.html#size,Variable Bio.Restriction.Restriction.BlnI.size=Bio.Restriction.Restriction.BlnI-class.html#size,Variable Bio.Restriction.Restriction.BloAII.size=Bio.Restriction.Restriction.BloAII-class.html#size,Variable Bio.Restriction.Restriction.BlpI.size=Bio.Restriction.Restriction.BlpI-class.html#size,Variable Bio.Restriction.Restriction.BlsI.size=Bio.Restriction.Restriction.BlsI-class.html#size,Variable Bio.Restriction.Restriction.BmcAI.size=Bio.Restriction.Restriction.BmcAI-class.html#size,Variable Bio.Restriction.Restriction.Bme1390I.size=Bio.Restriction.Restriction.Bme1390I-class.html#size,Variable Bio.Restriction.Restriction.Bme18I.size=Bio.Restriction.Restriction.Bme18I-class.html#size,Variable Bio.Restriction.Restriction.BmeDI.size=Bio.Restriction.Restriction.BmeDI-class.html#size,Variable Bio.Restriction.Restriction.BmeRI.size=Bio.Restriction.Restriction.BmeRI-class.html#size,Variable Bio.Restriction.Restriction.BmeT110I.size=Bio.Restriction.Restriction.BmeT110I-class.html#size,Variable Bio.Restriction.Restriction.BmgBI.size=Bio.Restriction.Restriction.BmgBI-class.html#size,Variable Bio.Restriction.Restriction.BmgI.size=Bio.Restriction.Restriction.BmgI-class.html#size,Variable Bio.Restriction.Restriction.BmgT120I.size=Bio.Restriction.Restriction.BmgT120I-class.html#size,Variable Bio.Restriction.Restriction.BmiI.size=Bio.Restriction.Restriction.BmiI-class.html#size,Variable Bio.Restriction.Restriction.BmrFI.size=Bio.Restriction.Restriction.BmrFI-class.html#size,Variable Bio.Restriction.Restriction.BmrI.size=Bio.Restriction.Restriction.BmrI-class.html#size,Variable Bio.Restriction.Restriction.BmsI.size=Bio.Restriction.Restriction.BmsI-class.html#size,Variable Bio.Restriction.Restriction.BmtI.size=Bio.Restriction.Restriction.BmtI-class.html#size,Variable Bio.Restriction.Restriction.BmuI.size=Bio.Restriction.Restriction.BmuI-class.html#size,Variable Bio.Restriction.Restriction.BoxI.size=Bio.Restriction.Restriction.BoxI-class.html#size,Variable Bio.Restriction.Restriction.BpiI.size=Bio.Restriction.Restriction.BpiI-class.html#size,Variable Bio.Restriction.Restriction.BplI.size=Bio.Restriction.Restriction.BplI-class.html#size,Variable Bio.Restriction.Restriction.BpmI.size=Bio.Restriction.Restriction.BpmI-class.html#size,Variable Bio.Restriction.Restriction.Bpu10I.size=Bio.Restriction.Restriction.Bpu10I-class.html#size,Variable Bio.Restriction.Restriction.Bpu1102I.size=Bio.Restriction.Restriction.Bpu1102I-class.html#size,Variable Bio.Restriction.Restriction.Bpu14I.size=Bio.Restriction.Restriction.Bpu14I-class.html#size,Variable Bio.Restriction.Restriction.BpuEI.size=Bio.Restriction.Restriction.BpuEI-class.html#size,Variable Bio.Restriction.Restriction.BpuMI.size=Bio.Restriction.Restriction.BpuMI-class.html#size,Variable Bio.Restriction.Restriction.Bsa29I.size=Bio.Restriction.Restriction.Bsa29I-class.html#size,Variable Bio.Restriction.Restriction.BsaAI.size=Bio.Restriction.Restriction.BsaAI-class.html#size,Variable Bio.Restriction.Restriction.BsaBI.size=Bio.Restriction.Restriction.BsaBI-class.html#size,Variable Bio.Restriction.Restriction.BsaHI.size=Bio.Restriction.Restriction.BsaHI-class.html#size,Variable Bio.Restriction.Restriction.BsaI.size=Bio.Restriction.Restriction.BsaI-class.html#size,Variable Bio.Restriction.Restriction.BsaJI.size=Bio.Restriction.Restriction.BsaJI-class.html#size,Variable Bio.Restriction.Restriction.BsaWI.size=Bio.Restriction.Restriction.BsaWI-class.html#size,Variable Bio.Restriction.Restriction.BsaXI.size=Bio.Restriction.Restriction.BsaXI-class.html#size,Variable Bio.Restriction.Restriction.BsbI.size=Bio.Restriction.Restriction.BsbI-class.html#size,Variable Bio.Restriction.Restriction.Bsc4I.size=Bio.Restriction.Restriction.Bsc4I-class.html#size,Variable Bio.Restriction.Restriction.BscAI.size=Bio.Restriction.Restriction.BscAI-class.html#size,Variable Bio.Restriction.Restriction.BscGI.size=Bio.Restriction.Restriction.BscGI-class.html#size,Variable Bio.Restriction.Restriction.Bse118I.size=Bio.Restriction.Restriction.Bse118I-class.html#size,Variable Bio.Restriction.Restriction.Bse1I.size=Bio.Restriction.Restriction.Bse1I-class.html#size,Variable Bio.Restriction.Restriction.Bse21I.size=Bio.Restriction.Restriction.Bse21I-class.html#size,Variable Bio.Restriction.Restriction.Bse3DI.size=Bio.Restriction.Restriction.Bse3DI-class.html#size,Variable Bio.Restriction.Restriction.Bse8I.size=Bio.Restriction.Restriction.Bse8I-class.html#size,Variable Bio.Restriction.Restriction.BseAI.size=Bio.Restriction.Restriction.BseAI-class.html#size,Variable Bio.Restriction.Restriction.BseBI.size=Bio.Restriction.Restriction.BseBI-class.html#size,Variable Bio.Restriction.Restriction.BseCI.size=Bio.Restriction.Restriction.BseCI-class.html#size,Variable Bio.Restriction.Restriction.BseDI.size=Bio.Restriction.Restriction.BseDI-class.html#size,Variable Bio.Restriction.Restriction.BseGI.size=Bio.Restriction.Restriction.BseGI-class.html#size,Variable Bio.Restriction.Restriction.BseJI.size=Bio.Restriction.Restriction.BseJI-class.html#size,Variable Bio.Restriction.Restriction.BseLI.size=Bio.Restriction.Restriction.BseLI-class.html#size,Variable Bio.Restriction.Restriction.BseMI.size=Bio.Restriction.Restriction.BseMI-class.html#size,Variable Bio.Restriction.Restriction.BseMII.size=Bio.Restriction.Restriction.BseMII-class.html#size,Variable Bio.Restriction.Restriction.BseNI.size=Bio.Restriction.Restriction.BseNI-class.html#size,Variable Bio.Restriction.Restriction.BsePI.size=Bio.Restriction.Restriction.BsePI-class.html#size,Variable Bio.Restriction.Restriction.BseRI.size=Bio.Restriction.Restriction.BseRI-class.html#size,Variable Bio.Restriction.Restriction.BseSI.size=Bio.Restriction.Restriction.BseSI-class.html#size,Variable Bio.Restriction.Restriction.BseX3I.size=Bio.Restriction.Restriction.BseX3I-class.html#size,Variable Bio.Restriction.Restriction.BseXI.size=Bio.Restriction.Restriction.BseXI-class.html#size,Variable Bio.Restriction.Restriction.BseYI.size=Bio.Restriction.Restriction.BseYI-class.html#size,Variable Bio.Restriction.Restriction.BsgI.size=Bio.Restriction.Restriction.BsgI-class.html#size,Variable Bio.Restriction.Restriction.Bsh1236I.size=Bio.Restriction.Restriction.Bsh1236I-class.html#size,Variable Bio.Restriction.Restriction.Bsh1285I.size=Bio.Restriction.Restriction.Bsh1285I-class.html#size,Variable Bio.Restriction.Restriction.BshFI.size=Bio.Restriction.Restriction.BshFI-class.html#size,Variable Bio.Restriction.Restriction.BshNI.size=Bio.Restriction.Restriction.BshNI-class.html#size,Variable Bio.Restriction.Restriction.BshTI.size=Bio.Restriction.Restriction.BshTI-class.html#size,Variable Bio.Restriction.Restriction.BshVI.size=Bio.Restriction.Restriction.BshVI-class.html#size,Variable Bio.Restriction.Restriction.BsiEI.size=Bio.Restriction.Restriction.BsiEI-class.html#size,Variable Bio.Restriction.Restriction.BsiHKAI.size=Bio.Restriction.Restriction.BsiHKAI-class.html#size,Variable Bio.Restriction.Restriction.BsiHKCI.size=Bio.Restriction.Restriction.BsiHKCI-class.html#size,Variable Bio.Restriction.Restriction.BsiI.size=Bio.Restriction.Restriction.BsiI-class.html#size,Variable Bio.Restriction.Restriction.BsiSI.size=Bio.Restriction.Restriction.BsiSI-class.html#size,Variable Bio.Restriction.Restriction.BsiWI.size=Bio.Restriction.Restriction.BsiWI-class.html#size,Variable Bio.Restriction.Restriction.BsiYI.size=Bio.Restriction.Restriction.BsiYI-class.html#size,Variable Bio.Restriction.Restriction.BslFI.size=Bio.Restriction.Restriction.BslFI-class.html#size,Variable Bio.Restriction.Restriction.BslI.size=Bio.Restriction.Restriction.BslI-class.html#size,Variable Bio.Restriction.Restriction.BsmAI.size=Bio.Restriction.Restriction.BsmAI-class.html#size,Variable Bio.Restriction.Restriction.BsmBI.size=Bio.Restriction.Restriction.BsmBI-class.html#size,Variable Bio.Restriction.Restriction.BsmFI.size=Bio.Restriction.Restriction.BsmFI-class.html#size,Variable Bio.Restriction.Restriction.BsmI.size=Bio.Restriction.Restriction.BsmI-class.html#size,Variable Bio.Restriction.Restriction.BsnI.size=Bio.Restriction.Restriction.BsnI-class.html#size,Variable Bio.Restriction.Restriction.Bso31I.size=Bio.Restriction.Restriction.Bso31I-class.html#size,Variable Bio.Restriction.Restriction.BsoBI.size=Bio.Restriction.Restriction.BsoBI-class.html#size,Variable Bio.Restriction.Restriction.Bsp119I.size=Bio.Restriction.Restriction.Bsp119I-class.html#size,Variable Bio.Restriction.Restriction.Bsp120I.size=Bio.Restriction.Restriction.Bsp120I-class.html#size,Variable Bio.Restriction.Restriction.Bsp1286I.size=Bio.Restriction.Restriction.Bsp1286I-class.html#size,Variable Bio.Restriction.Restriction.Bsp13I.size=Bio.Restriction.Restriction.Bsp13I-class.html#size,Variable Bio.Restriction.Restriction.Bsp1407I.size=Bio.Restriction.Restriction.Bsp1407I-class.html#size,Variable Bio.Restriction.Restriction.Bsp143I.size=Bio.Restriction.Restriction.Bsp143I-class.html#size,Variable Bio.Restriction.Restriction.Bsp1720I.size=Bio.Restriction.Restriction.Bsp1720I-class.html#size,Variable Bio.Restriction.Restriction.Bsp19I.size=Bio.Restriction.Restriction.Bsp19I-class.html#size,Variable Bio.Restriction.Restriction.Bsp24I.size=Bio.Restriction.Restriction.Bsp24I-class.html#size,Variable Bio.Restriction.Restriction.Bsp3004IV.size=Bio.Restriction.Restriction.Bsp3004IV-class.html#size,Variable Bio.Restriction.Restriction.Bsp460III.size=Bio.Restriction.Restriction.Bsp460III-class.html#size,Variable Bio.Restriction.Restriction.Bsp68I.size=Bio.Restriction.Restriction.Bsp68I-class.html#size,Variable Bio.Restriction.Restriction.BspACI.size=Bio.Restriction.Restriction.BspACI-class.html#size,Variable Bio.Restriction.Restriction.BspANI.size=Bio.Restriction.Restriction.BspANI-class.html#size,Variable Bio.Restriction.Restriction.BspCNI.size=Bio.Restriction.Restriction.BspCNI-class.html#size,Variable Bio.Restriction.Restriction.BspD6I.size=Bio.Restriction.Restriction.BspD6I-class.html#size,Variable Bio.Restriction.Restriction.BspDI.size=Bio.Restriction.Restriction.BspDI-class.html#size,Variable Bio.Restriction.Restriction.BspEI.size=Bio.Restriction.Restriction.BspEI-class.html#size,Variable Bio.Restriction.Restriction.BspFNI.size=Bio.Restriction.Restriction.BspFNI-class.html#size,Variable Bio.Restriction.Restriction.BspGI.size=Bio.Restriction.Restriction.BspGI-class.html#size,Variable Bio.Restriction.Restriction.BspHI.size=Bio.Restriction.Restriction.BspHI-class.html#size,Variable Bio.Restriction.Restriction.BspLI.size=Bio.Restriction.Restriction.BspLI-class.html#size,Variable Bio.Restriction.Restriction.BspLU11I.size=Bio.Restriction.Restriction.BspLU11I-class.html#size,Variable Bio.Restriction.Restriction.BspMAI.size=Bio.Restriction.Restriction.BspMAI-class.html#size,Variable Bio.Restriction.Restriction.BspMI.size=Bio.Restriction.Restriction.BspMI-class.html#size,Variable Bio.Restriction.Restriction.BspMII.size=Bio.Restriction.Restriction.BspMII-class.html#size,Variable Bio.Restriction.Restriction.BspNCI.size=Bio.Restriction.Restriction.BspNCI-class.html#size,Variable Bio.Restriction.Restriction.BspOI.size=Bio.Restriction.Restriction.BspOI-class.html#size,Variable Bio.Restriction.Restriction.BspPI.size=Bio.Restriction.Restriction.BspPI-class.html#size,Variable Bio.Restriction.Restriction.BspQI.size=Bio.Restriction.Restriction.BspQI-class.html#size,Variable Bio.Restriction.Restriction.BspT104I.size=Bio.Restriction.Restriction.BspT104I-class.html#size,Variable Bio.Restriction.Restriction.BspT107I.size=Bio.Restriction.Restriction.BspT107I-class.html#size,Variable Bio.Restriction.Restriction.BspTI.size=Bio.Restriction.Restriction.BspTI-class.html#size,Variable Bio.Restriction.Restriction.BspTNI.size=Bio.Restriction.Restriction.BspTNI-class.html#size,Variable Bio.Restriction.Restriction.BsrBI.size=Bio.Restriction.Restriction.BsrBI-class.html#size,Variable Bio.Restriction.Restriction.BsrDI.size=Bio.Restriction.Restriction.BsrDI-class.html#size,Variable Bio.Restriction.Restriction.BsrFI.size=Bio.Restriction.Restriction.BsrFI-class.html#size,Variable Bio.Restriction.Restriction.BsrGI.size=Bio.Restriction.Restriction.BsrGI-class.html#size,Variable Bio.Restriction.Restriction.BsrI.size=Bio.Restriction.Restriction.BsrI-class.html#size,Variable Bio.Restriction.Restriction.BssAI.size=Bio.Restriction.Restriction.BssAI-class.html#size,Variable Bio.Restriction.Restriction.BssECI.size=Bio.Restriction.Restriction.BssECI-class.html#size,Variable Bio.Restriction.Restriction.BssHII.size=Bio.Restriction.Restriction.BssHII-class.html#size,Variable Bio.Restriction.Restriction.BssMI.size=Bio.Restriction.Restriction.BssMI-class.html#size,Variable Bio.Restriction.Restriction.BssNAI.size=Bio.Restriction.Restriction.BssNAI-class.html#size,Variable Bio.Restriction.Restriction.BssNI.size=Bio.Restriction.Restriction.BssNI-class.html#size,Variable Bio.Restriction.Restriction.BssSI.size=Bio.Restriction.Restriction.BssSI-class.html#size,Variable Bio.Restriction.Restriction.BssT1I.size=Bio.Restriction.Restriction.BssT1I-class.html#size,Variable Bio.Restriction.Restriction.Bst1107I.size=Bio.Restriction.Restriction.Bst1107I-class.html#size,Variable Bio.Restriction.Restriction.Bst2BI.size=Bio.Restriction.Restriction.Bst2BI-class.html#size,Variable Bio.Restriction.Restriction.Bst2UI.size=Bio.Restriction.Restriction.Bst2UI-class.html#size,Variable Bio.Restriction.Restriction.Bst4CI.size=Bio.Restriction.Restriction.Bst4CI-class.html#size,Variable Bio.Restriction.Restriction.Bst6I.size=Bio.Restriction.Restriction.Bst6I-class.html#size,Variable Bio.Restriction.Restriction.BstACI.size=Bio.Restriction.Restriction.BstACI-class.html#size,Variable Bio.Restriction.Restriction.BstAFI.size=Bio.Restriction.Restriction.BstAFI-class.html#size,Variable Bio.Restriction.Restriction.BstAPI.size=Bio.Restriction.Restriction.BstAPI-class.html#size,Variable Bio.Restriction.Restriction.BstAUI.size=Bio.Restriction.Restriction.BstAUI-class.html#size,Variable Bio.Restriction.Restriction.BstBAI.size=Bio.Restriction.Restriction.BstBAI-class.html#size,Variable Bio.Restriction.Restriction.BstBI.size=Bio.Restriction.Restriction.BstBI-class.html#size,Variable Bio.Restriction.Restriction.BstC8I.size=Bio.Restriction.Restriction.BstC8I-class.html#size,Variable Bio.Restriction.Restriction.BstDEI.size=Bio.Restriction.Restriction.BstDEI-class.html#size,Variable Bio.Restriction.Restriction.BstDSI.size=Bio.Restriction.Restriction.BstDSI-class.html#size,Variable Bio.Restriction.Restriction.BstEII.size=Bio.Restriction.Restriction.BstEII-class.html#size,Variable Bio.Restriction.Restriction.BstENI.size=Bio.Restriction.Restriction.BstENI-class.html#size,Variable Bio.Restriction.Restriction.BstF5I.size=Bio.Restriction.Restriction.BstF5I-class.html#size,Variable Bio.Restriction.Restriction.BstFNI.size=Bio.Restriction.Restriction.BstFNI-class.html#size,Variable Bio.Restriction.Restriction.BstH2I.size=Bio.Restriction.Restriction.BstH2I-class.html#size,Variable Bio.Restriction.Restriction.BstHHI.size=Bio.Restriction.Restriction.BstHHI-class.html#size,Variable Bio.Restriction.Restriction.BstKTI.size=Bio.Restriction.Restriction.BstKTI-class.html#size,Variable Bio.Restriction.Restriction.BstMAI.size=Bio.Restriction.Restriction.BstMAI-class.html#size,Variable Bio.Restriction.Restriction.BstMBI.size=Bio.Restriction.Restriction.BstMBI-class.html#size,Variable Bio.Restriction.Restriction.BstMCI.size=Bio.Restriction.Restriction.BstMCI-class.html#size,Variable Bio.Restriction.Restriction.BstMWI.size=Bio.Restriction.Restriction.BstMWI-class.html#size,Variable Bio.Restriction.Restriction.BstNI.size=Bio.Restriction.Restriction.BstNI-class.html#size,Variable Bio.Restriction.Restriction.BstNSI.size=Bio.Restriction.Restriction.BstNSI-class.html#size,Variable Bio.Restriction.Restriction.BstPAI.size=Bio.Restriction.Restriction.BstPAI-class.html#size,Variable Bio.Restriction.Restriction.BstPI.size=Bio.Restriction.Restriction.BstPI-class.html#size,Variable Bio.Restriction.Restriction.BstSCI.size=Bio.Restriction.Restriction.BstSCI-class.html#size,Variable Bio.Restriction.Restriction.BstSFI.size=Bio.Restriction.Restriction.BstSFI-class.html#size,Variable Bio.Restriction.Restriction.BstSLI.size=Bio.Restriction.Restriction.BstSLI-class.html#size,Variable Bio.Restriction.Restriction.BstSNI.size=Bio.Restriction.Restriction.BstSNI-class.html#size,Variable Bio.Restriction.Restriction.BstUI.size=Bio.Restriction.Restriction.BstUI-class.html#size,Variable Bio.Restriction.Restriction.BstV1I.size=Bio.Restriction.Restriction.BstV1I-class.html#size,Variable Bio.Restriction.Restriction.BstV2I.size=Bio.Restriction.Restriction.BstV2I-class.html#size,Variable Bio.Restriction.Restriction.BstX2I.size=Bio.Restriction.Restriction.BstX2I-class.html#size,Variable Bio.Restriction.Restriction.BstXI.size=Bio.Restriction.Restriction.BstXI-class.html#size,Variable Bio.Restriction.Restriction.BstYI.size=Bio.Restriction.Restriction.BstYI-class.html#size,Variable Bio.Restriction.Restriction.BstZ17I.size=Bio.Restriction.Restriction.BstZ17I-class.html#size,Variable Bio.Restriction.Restriction.BstZI.size=Bio.Restriction.Restriction.BstZI-class.html#size,Variable Bio.Restriction.Restriction.Bsu15I.size=Bio.Restriction.Restriction.Bsu15I-class.html#size,Variable Bio.Restriction.Restriction.Bsu36I.size=Bio.Restriction.Restriction.Bsu36I-class.html#size,Variable Bio.Restriction.Restriction.BsuI.size=Bio.Restriction.Restriction.BsuI-class.html#size,Variable Bio.Restriction.Restriction.BsuRI.size=Bio.Restriction.Restriction.BsuRI-class.html#size,Variable Bio.Restriction.Restriction.BsuTUI.size=Bio.Restriction.Restriction.BsuTUI-class.html#size,Variable Bio.Restriction.Restriction.BtgI.size=Bio.Restriction.Restriction.BtgI-class.html#size,Variable Bio.Restriction.Restriction.BtgZI.size=Bio.Restriction.Restriction.BtgZI-class.html#size,Variable Bio.Restriction.Restriction.BthCI.size=Bio.Restriction.Restriction.BthCI-class.html#size,Variable Bio.Restriction.Restriction.BtrI.size=Bio.Restriction.Restriction.BtrI-class.html#size,Variable Bio.Restriction.Restriction.BtsCI.size=Bio.Restriction.Restriction.BtsCI-class.html#size,Variable Bio.Restriction.Restriction.BtsI.size=Bio.Restriction.Restriction.BtsI-class.html#size,Variable Bio.Restriction.Restriction.BtsIMutI.size=Bio.Restriction.Restriction.BtsIMutI-class.html#size,Variable Bio.Restriction.Restriction.BtuMI.size=Bio.Restriction.Restriction.BtuMI-class.html#size,Variable Bio.Restriction.Restriction.BveI.size=Bio.Restriction.Restriction.BveI-class.html#size,Variable Bio.Restriction.Restriction.Cac8I.size=Bio.Restriction.Restriction.Cac8I-class.html#size,Variable Bio.Restriction.Restriction.CaiI.size=Bio.Restriction.Restriction.CaiI-class.html#size,Variable Bio.Restriction.Restriction.Cal14237I.size=Bio.Restriction.Restriction.Cal14237I-class.html#size,Variable Bio.Restriction.Restriction.CalB3II.size=Bio.Restriction.Restriction.CalB3II-class.html#size,Variable Bio.Restriction.Restriction.Cau10061II.size=Bio.Restriction.Restriction.Cau10061II-class.html#size,Variable Bio.Restriction.Restriction.CauII.size=Bio.Restriction.Restriction.CauII-class.html#size,Variable Bio.Restriction.Restriction.Cba13II.size=Bio.Restriction.Restriction.Cba13II-class.html#size,Variable Bio.Restriction.Restriction.Cba16038I.size=Bio.Restriction.Restriction.Cba16038I-class.html#size,Variable Bio.Restriction.Restriction.Cbo67071IV.size=Bio.Restriction.Restriction.Cbo67071IV-class.html#size,Variable Bio.Restriction.Restriction.Cch467III.size=Bio.Restriction.Restriction.Cch467III-class.html#size,Variable Bio.Restriction.Restriction.CchII.size=Bio.Restriction.Restriction.CchII-class.html#size,Variable Bio.Restriction.Restriction.CchIII.size=Bio.Restriction.Restriction.CchIII-class.html#size,Variable Bio.Restriction.Restriction.CciI.size=Bio.Restriction.Restriction.CciI-class.html#size,Variable Bio.Restriction.Restriction.CciNI.size=Bio.Restriction.Restriction.CciNI-class.html#size,Variable Bio.Restriction.Restriction.CcrNAIII.size=Bio.Restriction.Restriction.CcrNAIII-class.html#size,Variable Bio.Restriction.Restriction.Cdi11397I.size=Bio.Restriction.Restriction.Cdi11397I-class.html#size,Variable Bio.Restriction.Restriction.Cdi81III.size=Bio.Restriction.Restriction.Cdi81III-class.html#size,Variable Bio.Restriction.Restriction.CdiI.size=Bio.Restriction.Restriction.CdiI-class.html#size,Variable Bio.Restriction.Restriction.CdpI.size=Bio.Restriction.Restriction.CdpI-class.html#size,Variable Bio.Restriction.Restriction.Cdu23823II.size=Bio.Restriction.Restriction.Cdu23823II-class.html#size,Variable Bio.Restriction.Restriction.CfoI.size=Bio.Restriction.Restriction.CfoI-class.html#size,Variable Bio.Restriction.Restriction.Cfr10I.size=Bio.Restriction.Restriction.Cfr10I-class.html#size,Variable Bio.Restriction.Restriction.Cfr13I.size=Bio.Restriction.Restriction.Cfr13I-class.html#size,Variable Bio.Restriction.Restriction.Cfr42I.size=Bio.Restriction.Restriction.Cfr42I-class.html#size,Variable Bio.Restriction.Restriction.Cfr9I.size=Bio.Restriction.Restriction.Cfr9I-class.html#size,Variable Bio.Restriction.Restriction.CfrI.size=Bio.Restriction.Restriction.CfrI-class.html#size,Variable Bio.Restriction.Restriction.Cfupf3II.size=Bio.Restriction.Restriction.Cfupf3II-class.html#size,Variable Bio.Restriction.Restriction.Cgl13032I.size=Bio.Restriction.Restriction.Cgl13032I-class.html#size,Variable Bio.Restriction.Restriction.Cgl13032II.size=Bio.Restriction.Restriction.Cgl13032II-class.html#size,Variable Bio.Restriction.Restriction.ChaI.size=Bio.Restriction.Restriction.ChaI-class.html#size,Variable Bio.Restriction.Restriction.CjeFIII.size=Bio.Restriction.Restriction.CjeFIII-class.html#size,Variable Bio.Restriction.Restriction.CjeFV.size=Bio.Restriction.Restriction.CjeFV-class.html#size,Variable Bio.Restriction.Restriction.CjeI.size=Bio.Restriction.Restriction.CjeI-class.html#size,Variable Bio.Restriction.Restriction.CjeNII.size=Bio.Restriction.Restriction.CjeNII-class.html#size,Variable Bio.Restriction.Restriction.CjeNIII.size=Bio.Restriction.Restriction.CjeNIII-class.html#size,Variable Bio.Restriction.Restriction.CjeNV.size=Bio.Restriction.Restriction.CjeNV-class.html#size,Variable Bio.Restriction.Restriction.CjeP659IV.size=Bio.Restriction.Restriction.CjeP659IV-class.html#size,Variable Bio.Restriction.Restriction.CjePI.size=Bio.Restriction.Restriction.CjePI-class.html#size,Variable Bio.Restriction.Restriction.CjuI.size=Bio.Restriction.Restriction.CjuI-class.html#size,Variable Bio.Restriction.Restriction.CjuII.size=Bio.Restriction.Restriction.CjuII-class.html#size,Variable Bio.Restriction.Restriction.Cla11845III.size=Bio.Restriction.Restriction.Cla11845III-class.html#size,Variable Bio.Restriction.Restriction.ClaI.size=Bio.Restriction.Restriction.ClaI-class.html#size,Variable Bio.Restriction.Restriction.Cly7489II.size=Bio.Restriction.Restriction.Cly7489II-class.html#size,Variable Bio.Restriction.Restriction.Cma23826I.size=Bio.Restriction.Restriction.Cma23826I-class.html#size,Variable Bio.Restriction.Restriction.CpoI.size=Bio.Restriction.Restriction.CpoI-class.html#size,Variable Bio.Restriction.Restriction.CseI.size=Bio.Restriction.Restriction.CseI-class.html#size,Variable Bio.Restriction.Restriction.CsiI.size=Bio.Restriction.Restriction.CsiI-class.html#size,Variable Bio.Restriction.Restriction.Csp2014I.size=Bio.Restriction.Restriction.Csp2014I-class.html#size,Variable Bio.Restriction.Restriction.Csp6I.size=Bio.Restriction.Restriction.Csp6I-class.html#size,Variable Bio.Restriction.Restriction.CspAI.size=Bio.Restriction.Restriction.CspAI-class.html#size,Variable Bio.Restriction.Restriction.CspCI.size=Bio.Restriction.Restriction.CspCI-class.html#size,Variable Bio.Restriction.Restriction.CspI.size=Bio.Restriction.Restriction.CspI-class.html#size,Variable Bio.Restriction.Restriction.CstMI.size=Bio.Restriction.Restriction.CstMI-class.html#size,Variable Bio.Restriction.Restriction.CviAII.size=Bio.Restriction.Restriction.CviAII-class.html#size,Variable Bio.Restriction.Restriction.CviJI.size=Bio.Restriction.Restriction.CviJI-class.html#size,Variable Bio.Restriction.Restriction.CviKI_1.size=Bio.Restriction.Restriction.CviKI_1-class.html#size,Variable Bio.Restriction.Restriction.CviQI.size=Bio.Restriction.Restriction.CviQI-class.html#size,Variable Bio.Restriction.Restriction.CviRI.size=Bio.Restriction.Restriction.CviRI-class.html#size,Variable Bio.Restriction.Restriction.Dde51507I.size=Bio.Restriction.Restriction.Dde51507I-class.html#size,Variable Bio.Restriction.Restriction.DdeI.size=Bio.Restriction.Restriction.DdeI-class.html#size,Variable Bio.Restriction.Restriction.DinI.size=Bio.Restriction.Restriction.DinI-class.html#size,Variable Bio.Restriction.Restriction.DpnI.size=Bio.Restriction.Restriction.DpnI-class.html#size,Variable Bio.Restriction.Restriction.DpnII.size=Bio.Restriction.Restriction.DpnII-class.html#size,Variable Bio.Restriction.Restriction.DraI.size=Bio.Restriction.Restriction.DraI-class.html#size,Variable Bio.Restriction.Restriction.DraII.size=Bio.Restriction.Restriction.DraII-class.html#size,Variable Bio.Restriction.Restriction.DraIII.size=Bio.Restriction.Restriction.DraIII-class.html#size,Variable Bio.Restriction.Restriction.DraRI.size=Bio.Restriction.Restriction.DraRI-class.html#size,Variable Bio.Restriction.Restriction.DrdI.size=Bio.Restriction.Restriction.DrdI-class.html#size,Variable Bio.Restriction.Restriction.DrdII.size=Bio.Restriction.Restriction.DrdII-class.html#size,Variable Bio.Restriction.Restriction.DriI.size=Bio.Restriction.Restriction.DriI-class.html#size,Variable Bio.Restriction.Restriction.DsaI.size=Bio.Restriction.Restriction.DsaI-class.html#size,Variable Bio.Restriction.Restriction.DseDI.size=Bio.Restriction.Restriction.DseDI-class.html#size,Variable Bio.Restriction.Restriction.EaeI.size=Bio.Restriction.Restriction.EaeI-class.html#size,Variable Bio.Restriction.Restriction.EagI.size=Bio.Restriction.Restriction.EagI-class.html#size,Variable Bio.Restriction.Restriction.Eam1104I.size=Bio.Restriction.Restriction.Eam1104I-class.html#size,Variable Bio.Restriction.Restriction.Eam1105I.size=Bio.Restriction.Restriction.Eam1105I-class.html#size,Variable Bio.Restriction.Restriction.EarI.size=Bio.Restriction.Restriction.EarI-class.html#size,Variable Bio.Restriction.Restriction.EciI.size=Bio.Restriction.Restriction.EciI-class.html#size,Variable Bio.Restriction.Restriction.Ecl136II.size=Bio.Restriction.Restriction.Ecl136II-class.html#size,Variable Bio.Restriction.Restriction.Ecl35734I.size=Bio.Restriction.Restriction.Ecl35734I-class.html#size,Variable Bio.Restriction.Restriction.EclXI.size=Bio.Restriction.Restriction.EclXI-class.html#size,Variable Bio.Restriction.Restriction.Eco105I.size=Bio.Restriction.Restriction.Eco105I-class.html#size,Variable Bio.Restriction.Restriction.Eco130I.size=Bio.Restriction.Restriction.Eco130I-class.html#size,Variable Bio.Restriction.Restriction.Eco147I.size=Bio.Restriction.Restriction.Eco147I-class.html#size,Variable Bio.Restriction.Restriction.Eco24I.size=Bio.Restriction.Restriction.Eco24I-class.html#size,Variable Bio.Restriction.Restriction.Eco31I.size=Bio.Restriction.Restriction.Eco31I-class.html#size,Variable Bio.Restriction.Restriction.Eco32I.size=Bio.Restriction.Restriction.Eco32I-class.html#size,Variable Bio.Restriction.Restriction.Eco4465II.size=Bio.Restriction.Restriction.Eco4465II-class.html#size,Variable Bio.Restriction.Restriction.Eco47I.size=Bio.Restriction.Restriction.Eco47I-class.html#size,Variable Bio.Restriction.Restriction.Eco47III.size=Bio.Restriction.Restriction.Eco47III-class.html#size,Variable Bio.Restriction.Restriction.Eco52I.size=Bio.Restriction.Restriction.Eco52I-class.html#size,Variable Bio.Restriction.Restriction.Eco53kI.size=Bio.Restriction.Restriction.Eco53kI-class.html#size,Variable Bio.Restriction.Restriction.Eco57I.size=Bio.Restriction.Restriction.Eco57I-class.html#size,Variable Bio.Restriction.Restriction.Eco57MI.size=Bio.Restriction.Restriction.Eco57MI-class.html#size,Variable Bio.Restriction.Restriction.Eco72I.size=Bio.Restriction.Restriction.Eco72I-class.html#size,Variable Bio.Restriction.Restriction.Eco81I.size=Bio.Restriction.Restriction.Eco81I-class.html#size,Variable Bio.Restriction.Restriction.Eco88I.size=Bio.Restriction.Restriction.Eco88I-class.html#size,Variable Bio.Restriction.Restriction.Eco91I.size=Bio.Restriction.Restriction.Eco91I-class.html#size,Variable Bio.Restriction.Restriction.EcoBLMcrX.size=Bio.Restriction.Restriction.EcoBLMcrX-class.html#size,Variable Bio.Restriction.Restriction.EcoE1140I.size=Bio.Restriction.Restriction.EcoE1140I-class.html#size,Variable Bio.Restriction.Restriction.EcoHI.size=Bio.Restriction.Restriction.EcoHI-class.html#size,Variable Bio.Restriction.Restriction.EcoICRI.size=Bio.Restriction.Restriction.EcoICRI-class.html#size,Variable Bio.Restriction.Restriction.EcoMVII.size=Bio.Restriction.Restriction.EcoMVII-class.html#size,Variable Bio.Restriction.Restriction.EcoNI.size=Bio.Restriction.Restriction.EcoNI-class.html#size,Variable Bio.Restriction.Restriction.EcoO109I.size=Bio.Restriction.Restriction.EcoO109I-class.html#size,Variable Bio.Restriction.Restriction.EcoO65I.size=Bio.Restriction.Restriction.EcoO65I-class.html#size,Variable Bio.Restriction.Restriction.EcoRI.size=Bio.Restriction.Restriction.EcoRI-class.html#size,Variable Bio.Restriction.Restriction.EcoRII.size=Bio.Restriction.Restriction.EcoRII-class.html#size,Variable Bio.Restriction.Restriction.EcoRV.size=Bio.Restriction.Restriction.EcoRV-class.html#size,Variable Bio.Restriction.Restriction.EcoT14I.size=Bio.Restriction.Restriction.EcoT14I-class.html#size,Variable Bio.Restriction.Restriction.EcoT22I.size=Bio.Restriction.Restriction.EcoT22I-class.html#size,Variable Bio.Restriction.Restriction.EcoT38I.size=Bio.Restriction.Restriction.EcoT38I-class.html#size,Variable Bio.Restriction.Restriction.EgeI.size=Bio.Restriction.Restriction.EgeI-class.html#size,Variable Bio.Restriction.Restriction.EheI.size=Bio.Restriction.Restriction.EheI-class.html#size,Variable Bio.Restriction.Restriction.Eli8509II.size=Bio.Restriction.Restriction.Eli8509II-class.html#size,Variable Bio.Restriction.Restriction.ErhI.size=Bio.Restriction.Restriction.ErhI-class.html#size,Variable Bio.Restriction.Restriction.EsaBC3I.size=Bio.Restriction.Restriction.EsaBC3I-class.html#size,Variable Bio.Restriction.Restriction.EsaSSI.size=Bio.Restriction.Restriction.EsaSSI-class.html#size,Variable Bio.Restriction.Restriction.Esp3007I.size=Bio.Restriction.Restriction.Esp3007I-class.html#size,Variable Bio.Restriction.Restriction.Esp3I.size=Bio.Restriction.Restriction.Esp3I-class.html#size,Variable Bio.Restriction.Restriction.EspI.size=Bio.Restriction.Restriction.EspI-class.html#size,Variable Bio.Restriction.Restriction.Exi27195I.size=Bio.Restriction.Restriction.Exi27195I-class.html#size,Variable Bio.Restriction.Restriction.FaeI.size=Bio.Restriction.Restriction.FaeI-class.html#size,Variable Bio.Restriction.Restriction.FaiI.size=Bio.Restriction.Restriction.FaiI-class.html#size,Variable Bio.Restriction.Restriction.FalI.size=Bio.Restriction.Restriction.FalI-class.html#size,Variable Bio.Restriction.Restriction.FaqI.size=Bio.Restriction.Restriction.FaqI-class.html#size,Variable Bio.Restriction.Restriction.FatI.size=Bio.Restriction.Restriction.FatI-class.html#size,Variable Bio.Restriction.Restriction.FauI.size=Bio.Restriction.Restriction.FauI-class.html#size,Variable Bio.Restriction.Restriction.FauNDI.size=Bio.Restriction.Restriction.FauNDI-class.html#size,Variable Bio.Restriction.Restriction.FbaI.size=Bio.Restriction.Restriction.FbaI-class.html#size,Variable Bio.Restriction.Restriction.FblI.size=Bio.Restriction.Restriction.FblI-class.html#size,Variable Bio.Restriction.Restriction.FinI.size=Bio.Restriction.Restriction.FinI-class.html#size,Variable Bio.Restriction.Restriction.FmuI.size=Bio.Restriction.Restriction.FmuI-class.html#size,Variable Bio.Restriction.Restriction.Fnu4HI.size=Bio.Restriction.Restriction.Fnu4HI-class.html#size,Variable Bio.Restriction.Restriction.FnuDII.size=Bio.Restriction.Restriction.FnuDII-class.html#size,Variable Bio.Restriction.Restriction.FokI.size=Bio.Restriction.Restriction.FokI-class.html#size,Variable Bio.Restriction.Restriction.FriOI.size=Bio.Restriction.Restriction.FriOI-class.html#size,Variable Bio.Restriction.Restriction.FseI.size=Bio.Restriction.Restriction.FseI-class.html#size,Variable Bio.Restriction.Restriction.Fsp4HI.size=Bio.Restriction.Restriction.Fsp4HI-class.html#size,Variable Bio.Restriction.Restriction.FspAI.size=Bio.Restriction.Restriction.FspAI-class.html#size,Variable Bio.Restriction.Restriction.FspBI.size=Bio.Restriction.Restriction.FspBI-class.html#size,Variable Bio.Restriction.Restriction.FspEI.size=Bio.Restriction.Restriction.FspEI-class.html#size,Variable Bio.Restriction.Restriction.FspI.size=Bio.Restriction.Restriction.FspI-class.html#size,Variable Bio.Restriction.Restriction.FtnUV.size=Bio.Restriction.Restriction.FtnUV-class.html#size,Variable Bio.Restriction.Restriction.GauT27I.size=Bio.Restriction.Restriction.GauT27I-class.html#size,Variable Bio.Restriction.Restriction.Gba708II.size=Bio.Restriction.Restriction.Gba708II-class.html#size,Variable Bio.Restriction.Restriction.GdiII.size=Bio.Restriction.Restriction.GdiII-class.html#size,Variable Bio.Restriction.Restriction.GlaI.size=Bio.Restriction.Restriction.GlaI-class.html#size,Variable Bio.Restriction.Restriction.GluI.size=Bio.Restriction.Restriction.GluI-class.html#size,Variable Bio.Restriction.Restriction.GsaI.size=Bio.Restriction.Restriction.GsaI-class.html#size,Variable Bio.Restriction.Restriction.GsuI.size=Bio.Restriction.Restriction.GsuI-class.html#size,Variable Bio.Restriction.Restriction.HaeI.size=Bio.Restriction.Restriction.HaeI-class.html#size,Variable Bio.Restriction.Restriction.HaeII.size=Bio.Restriction.Restriction.HaeII-class.html#size,Variable Bio.Restriction.Restriction.HaeIII.size=Bio.Restriction.Restriction.HaeIII-class.html#size,Variable Bio.Restriction.Restriction.HapII.size=Bio.Restriction.Restriction.HapII-class.html#size,Variable Bio.Restriction.Restriction.HauII.size=Bio.Restriction.Restriction.HauII-class.html#size,Variable Bio.Restriction.Restriction.HbaII.size=Bio.Restriction.Restriction.HbaII-class.html#size,Variable Bio.Restriction.Restriction.HgaI.size=Bio.Restriction.Restriction.HgaI-class.html#size,Variable Bio.Restriction.Restriction.HgiAI.size=Bio.Restriction.Restriction.HgiAI-class.html#size,Variable Bio.Restriction.Restriction.HgiCI.size=Bio.Restriction.Restriction.HgiCI-class.html#size,Variable Bio.Restriction.Restriction.HgiEII.size=Bio.Restriction.Restriction.HgiEII-class.html#size,Variable Bio.Restriction.Restriction.HgiJII.size=Bio.Restriction.Restriction.HgiJII-class.html#size,Variable Bio.Restriction.Restriction.HhaI.size=Bio.Restriction.Restriction.HhaI-class.html#size,Variable Bio.Restriction.Restriction.Hin1I.size=Bio.Restriction.Restriction.Hin1I-class.html#size,Variable Bio.Restriction.Restriction.Hin1II.size=Bio.Restriction.Restriction.Hin1II-class.html#size,Variable Bio.Restriction.Restriction.Hin4I.size=Bio.Restriction.Restriction.Hin4I-class.html#size,Variable Bio.Restriction.Restriction.Hin4II.size=Bio.Restriction.Restriction.Hin4II-class.html#size,Variable Bio.Restriction.Restriction.Hin6I.size=Bio.Restriction.Restriction.Hin6I-class.html#size,Variable Bio.Restriction.Restriction.HinP1I.size=Bio.Restriction.Restriction.HinP1I-class.html#size,Variable Bio.Restriction.Restriction.HincII.size=Bio.Restriction.Restriction.HincII-class.html#size,Variable Bio.Restriction.Restriction.HindII.size=Bio.Restriction.Restriction.HindII-class.html#size,Variable Bio.Restriction.Restriction.HindIII.size=Bio.Restriction.Restriction.HindIII-class.html#size,Variable Bio.Restriction.Restriction.HinfI.size=Bio.Restriction.Restriction.HinfI-class.html#size,Variable Bio.Restriction.Restriction.HpaI.size=Bio.Restriction.Restriction.HpaI-class.html#size,Variable Bio.Restriction.Restriction.HpaII.size=Bio.Restriction.Restriction.HpaII-class.html#size,Variable Bio.Restriction.Restriction.HphI.size=Bio.Restriction.Restriction.HphI-class.html#size,Variable Bio.Restriction.Restriction.Hpy166II.size=Bio.Restriction.Restriction.Hpy166II-class.html#size,Variable Bio.Restriction.Restriction.Hpy178III.size=Bio.Restriction.Restriction.Hpy178III-class.html#size,Variable Bio.Restriction.Restriction.Hpy188I.size=Bio.Restriction.Restriction.Hpy188I-class.html#size,Variable Bio.Restriction.Restriction.Hpy188III.size=Bio.Restriction.Restriction.Hpy188III-class.html#size,Variable Bio.Restriction.Restriction.Hpy8I.size=Bio.Restriction.Restriction.Hpy8I-class.html#size,Variable Bio.Restriction.Restriction.Hpy99I.size=Bio.Restriction.Restriction.Hpy99I-class.html#size,Variable Bio.Restriction.Restriction.Hpy99XIII.size=Bio.Restriction.Restriction.Hpy99XIII-class.html#size,Variable Bio.Restriction.Restriction.Hpy99XIV.size=Bio.Restriction.Restriction.Hpy99XIV-class.html#size,Variable Bio.Restriction.Restriction.Hpy99XIV_mut1.size=Bio.Restriction.Restriction.Hpy99XIV_mut1-class.html#size,Variable Bio.Restriction.Restriction.Hpy99XXII.size=Bio.Restriction.Restriction.Hpy99XXII-class.html#size,Variable Bio.Restriction.Restriction.HpyAV.size=Bio.Restriction.Restriction.HpyAV-class.html#size,Variable Bio.Restriction.Restriction.HpyAXIV.size=Bio.Restriction.Restriction.HpyAXIV-class.html#size,Variable Bio.Restriction.Restriction.HpyAXVI_mut1.size=Bio.Restriction.Restriction.HpyAXVI_mut1-class.html#size,Variable Bio.Restriction.Restriction.HpyAXVI_mut2.size=Bio.Restriction.Restriction.HpyAXVI_mut2-class.html#size,Variable Bio.Restriction.Restriction.HpyCH4III.size=Bio.Restriction.Restriction.HpyCH4III-class.html#size,Variable Bio.Restriction.Restriction.HpyCH4IV.size=Bio.Restriction.Restriction.HpyCH4IV-class.html#size,Variable Bio.Restriction.Restriction.HpyCH4V.size=Bio.Restriction.Restriction.HpyCH4V-class.html#size,Variable Bio.Restriction.Restriction.HpyF10VI.size=Bio.Restriction.Restriction.HpyF10VI-class.html#size,Variable Bio.Restriction.Restriction.HpyF3I.size=Bio.Restriction.Restriction.HpyF3I-class.html#size,Variable Bio.Restriction.Restriction.HpySE526I.size=Bio.Restriction.Restriction.HpySE526I-class.html#size,Variable Bio.Restriction.Restriction.HpyUM032XIII.size=Bio.Restriction.Restriction.HpyUM032XIII-class.html#size,Variable Bio.Restriction.Restriction.HpyUM032XIII_mut1.size=Bio.Restriction.Restriction.HpyUM032XIII_mut1-class.html#size,Variable Bio.Restriction.Restriction.HpyUM032XIV.size=Bio.Restriction.Restriction.HpyUM032XIV-class.html#size,Variable Bio.Restriction.Restriction.HpyUM037X.size=Bio.Restriction.Restriction.HpyUM037X-class.html#size,Variable Bio.Restriction.Restriction.Hsp92I.size=Bio.Restriction.Restriction.Hsp92I-class.html#size,Variable Bio.Restriction.Restriction.Hsp92II.size=Bio.Restriction.Restriction.Hsp92II-class.html#size,Variable Bio.Restriction.Restriction.HspAI.size=Bio.Restriction.Restriction.HspAI-class.html#size,Variable Bio.Restriction.Restriction.Jma19592I.size=Bio.Restriction.Restriction.Jma19592I-class.html#size,Variable Bio.Restriction.Restriction.Jma19592II.size=Bio.Restriction.Restriction.Jma19592II-class.html#size,Variable Bio.Restriction.Restriction.Jsp2502II.size=Bio.Restriction.Restriction.Jsp2502II-class.html#size,Variable Bio.Restriction.Restriction.KasI.size=Bio.Restriction.Restriction.KasI-class.html#size,Variable Bio.Restriction.Restriction.KflI.size=Bio.Restriction.Restriction.KflI-class.html#size,Variable Bio.Restriction.Restriction.Kor51II.size=Bio.Restriction.Restriction.Kor51II-class.html#size,Variable Bio.Restriction.Restriction.Kpn2I.size=Bio.Restriction.Restriction.Kpn2I-class.html#size,Variable Bio.Restriction.Restriction.KpnI.size=Bio.Restriction.Restriction.KpnI-class.html#size,Variable Bio.Restriction.Restriction.KpnNIH30III.size=Bio.Restriction.Restriction.KpnNIH30III-class.html#size,Variable Bio.Restriction.Restriction.KroI.size=Bio.Restriction.Restriction.KroI-class.html#size,Variable Bio.Restriction.Restriction.Ksp22I.size=Bio.Restriction.Restriction.Ksp22I-class.html#size,Variable Bio.Restriction.Restriction.Ksp632I.size=Bio.Restriction.Restriction.Ksp632I-class.html#size,Variable Bio.Restriction.Restriction.KspAI.size=Bio.Restriction.Restriction.KspAI-class.html#size,Variable Bio.Restriction.Restriction.KspI.size=Bio.Restriction.Restriction.KspI-class.html#size,Variable Bio.Restriction.Restriction.Kzo9I.size=Bio.Restriction.Restriction.Kzo9I-class.html#size,Variable Bio.Restriction.Restriction.Lba2029III.size=Bio.Restriction.Restriction.Lba2029III-class.html#size,Variable Bio.Restriction.Restriction.LguI.size=Bio.Restriction.Restriction.LguI-class.html#size,Variable Bio.Restriction.Restriction.LmnI.size=Bio.Restriction.Restriction.LmnI-class.html#size,Variable Bio.Restriction.Restriction.Lmo370I.size=Bio.Restriction.Restriction.Lmo370I-class.html#size,Variable Bio.Restriction.Restriction.Lmo911II.size=Bio.Restriction.Restriction.Lmo911II-class.html#size,Variable Bio.Restriction.Restriction.LpnI.size=Bio.Restriction.Restriction.LpnI-class.html#size,Variable Bio.Restriction.Restriction.LpnPI.size=Bio.Restriction.Restriction.LpnPI-class.html#size,Variable Bio.Restriction.Restriction.Lsp1109I.size=Bio.Restriction.Restriction.Lsp1109I-class.html#size,Variable Bio.Restriction.Restriction.Lsp48III.size=Bio.Restriction.Restriction.Lsp48III-class.html#size,Variable Bio.Restriction.Restriction.Lsp6406VI.size=Bio.Restriction.Restriction.Lsp6406VI-class.html#size,Variable Bio.Restriction.Restriction.LweI.size=Bio.Restriction.Restriction.LweI-class.html#size,Variable Bio.Restriction.Restriction.MabI.size=Bio.Restriction.Restriction.MabI-class.html#size,Variable Bio.Restriction.Restriction.MaeI.size=Bio.Restriction.Restriction.MaeI-class.html#size,Variable Bio.Restriction.Restriction.MaeII.size=Bio.Restriction.Restriction.MaeII-class.html#size,Variable Bio.Restriction.Restriction.MaeIII.size=Bio.Restriction.Restriction.MaeIII-class.html#size,Variable Bio.Restriction.Restriction.MalI.size=Bio.Restriction.Restriction.MalI-class.html#size,Variable Bio.Restriction.Restriction.MaqI.size=Bio.Restriction.Restriction.MaqI-class.html#size,Variable Bio.Restriction.Restriction.MauBI.size=Bio.Restriction.Restriction.MauBI-class.html#size,Variable Bio.Restriction.Restriction.Mba11I.size=Bio.Restriction.Restriction.Mba11I-class.html#size,Variable Bio.Restriction.Restriction.MbiI.size=Bio.Restriction.Restriction.MbiI-class.html#size,Variable Bio.Restriction.Restriction.MboI.size=Bio.Restriction.Restriction.MboI-class.html#size,Variable Bio.Restriction.Restriction.MboII.size=Bio.Restriction.Restriction.MboII-class.html#size,Variable Bio.Restriction.Restriction.McaTI.size=Bio.Restriction.Restriction.McaTI-class.html#size,Variable Bio.Restriction.Restriction.Mcr10I.size=Bio.Restriction.Restriction.Mcr10I-class.html#size,Variable Bio.Restriction.Restriction.McrI.size=Bio.Restriction.Restriction.McrI-class.html#size,Variable Bio.Restriction.Restriction.MfeI.size=Bio.Restriction.Restriction.MfeI-class.html#size,Variable Bio.Restriction.Restriction.MflI.size=Bio.Restriction.Restriction.MflI-class.html#size,Variable Bio.Restriction.Restriction.MhlI.size=Bio.Restriction.Restriction.MhlI-class.html#size,Variable Bio.Restriction.Restriction.MjaIV.size=Bio.Restriction.Restriction.MjaIV-class.html#size,Variable Bio.Restriction.Restriction.MkaDII.size=Bio.Restriction.Restriction.MkaDII-class.html#size,Variable Bio.Restriction.Restriction.MlsI.size=Bio.Restriction.Restriction.MlsI-class.html#size,Variable Bio.Restriction.Restriction.MluCI.size=Bio.Restriction.Restriction.MluCI-class.html#size,Variable Bio.Restriction.Restriction.MluI.size=Bio.Restriction.Restriction.MluI-class.html#size,Variable Bio.Restriction.Restriction.MluNI.size=Bio.Restriction.Restriction.MluNI-class.html#size,Variable Bio.Restriction.Restriction.Mly113I.size=Bio.Restriction.Restriction.Mly113I-class.html#size,Variable Bio.Restriction.Restriction.MlyI.size=Bio.Restriction.Restriction.MlyI-class.html#size,Variable Bio.Restriction.Restriction.MmeI.size=Bio.Restriction.Restriction.MmeI-class.html#size,Variable Bio.Restriction.Restriction.MnlI.size=Bio.Restriction.Restriction.MnlI-class.html#size,Variable Bio.Restriction.Restriction.Mox20I.size=Bio.Restriction.Restriction.Mox20I-class.html#size,Variable Bio.Restriction.Restriction.Mph1103I.size=Bio.Restriction.Restriction.Mph1103I-class.html#size,Variable Bio.Restriction.Restriction.MreI.size=Bio.Restriction.Restriction.MreI-class.html#size,Variable Bio.Restriction.Restriction.MroI.size=Bio.Restriction.Restriction.MroI-class.html#size,Variable Bio.Restriction.Restriction.MroNI.size=Bio.Restriction.Restriction.MroNI-class.html#size,Variable Bio.Restriction.Restriction.MroXI.size=Bio.Restriction.Restriction.MroXI-class.html#size,Variable Bio.Restriction.Restriction.MscI.size=Bio.Restriction.Restriction.MscI-class.html#size,Variable Bio.Restriction.Restriction.MseI.size=Bio.Restriction.Restriction.MseI-class.html#size,Variable Bio.Restriction.Restriction.MslI.size=Bio.Restriction.Restriction.MslI-class.html#size,Variable Bio.Restriction.Restriction.Msp20I.size=Bio.Restriction.Restriction.Msp20I-class.html#size,Variable Bio.Restriction.Restriction.MspA1I.size=Bio.Restriction.Restriction.MspA1I-class.html#size,Variable Bio.Restriction.Restriction.MspCI.size=Bio.Restriction.Restriction.MspCI-class.html#size,Variable Bio.Restriction.Restriction.MspI.size=Bio.Restriction.Restriction.MspI-class.html#size,Variable Bio.Restriction.Restriction.MspI7II.size=Bio.Restriction.Restriction.MspI7II-class.html#size,Variable Bio.Restriction.Restriction.MspJI.size=Bio.Restriction.Restriction.MspJI-class.html#size,Variable Bio.Restriction.Restriction.MspR9I.size=Bio.Restriction.Restriction.MspR9I-class.html#size,Variable Bio.Restriction.Restriction.MssI.size=Bio.Restriction.Restriction.MssI-class.html#size,Variable Bio.Restriction.Restriction.MstI.size=Bio.Restriction.Restriction.MstI-class.html#size,Variable Bio.Restriction.Restriction.MteI.size=Bio.Restriction.Restriction.MteI-class.html#size,Variable Bio.Restriction.Restriction.MtuHN878II.size=Bio.Restriction.Restriction.MtuHN878II-class.html#size,Variable Bio.Restriction.Restriction.MunI.size=Bio.Restriction.Restriction.MunI-class.html#size,Variable Bio.Restriction.Restriction.Mva1269I.size=Bio.Restriction.Restriction.Mva1269I-class.html#size,Variable Bio.Restriction.Restriction.MvaI.size=Bio.Restriction.Restriction.MvaI-class.html#size,Variable Bio.Restriction.Restriction.MvnI.size=Bio.Restriction.Restriction.MvnI-class.html#size,Variable Bio.Restriction.Restriction.MwoI.size=Bio.Restriction.Restriction.MwoI-class.html#size,Variable Bio.Restriction.Restriction.NaeI.size=Bio.Restriction.Restriction.NaeI-class.html#size,Variable Bio.Restriction.Restriction.Nal45188II.size=Bio.Restriction.Restriction.Nal45188II-class.html#size,Variable Bio.Restriction.Restriction.NarI.size=Bio.Restriction.Restriction.NarI-class.html#size,Variable Bio.Restriction.Restriction.Nbr128II.size=Bio.Restriction.Restriction.Nbr128II-class.html#size,Variable Bio.Restriction.Restriction.NciI.size=Bio.Restriction.Restriction.NciI-class.html#size,Variable Bio.Restriction.Restriction.NcoI.size=Bio.Restriction.Restriction.NcoI-class.html#size,Variable Bio.Restriction.Restriction.NdeI.size=Bio.Restriction.Restriction.NdeI-class.html#size,Variable Bio.Restriction.Restriction.NdeII.size=Bio.Restriction.Restriction.NdeII-class.html#size,Variable Bio.Restriction.Restriction.NgoAVII.size=Bio.Restriction.Restriction.NgoAVII-class.html#size,Variable Bio.Restriction.Restriction.NgoAVIII.size=Bio.Restriction.Restriction.NgoAVIII-class.html#size,Variable Bio.Restriction.Restriction.NgoMIV.size=Bio.Restriction.Restriction.NgoMIV-class.html#size,Variable Bio.Restriction.Restriction.NhaXI.size=Bio.Restriction.Restriction.NhaXI-class.html#size,Variable Bio.Restriction.Restriction.NheI.size=Bio.Restriction.Restriction.NheI-class.html#size,Variable Bio.Restriction.Restriction.NhoI.size=Bio.Restriction.Restriction.NhoI-class.html#size,Variable Bio.Restriction.Restriction.NlaCI.size=Bio.Restriction.Restriction.NlaCI-class.html#size,Variable Bio.Restriction.Restriction.NlaIII.size=Bio.Restriction.Restriction.NlaIII-class.html#size,Variable Bio.Restriction.Restriction.NlaIV.size=Bio.Restriction.Restriction.NlaIV-class.html#size,Variable Bio.Restriction.Restriction.Nli3877I.size=Bio.Restriction.Restriction.Nli3877I-class.html#size,Variable Bio.Restriction.Restriction.NmeAIII.size=Bio.Restriction.Restriction.NmeAIII-class.html#size,Variable Bio.Restriction.Restriction.NmeDI.size=Bio.Restriction.Restriction.NmeDI-class.html#size,Variable Bio.Restriction.Restriction.NmuCI.size=Bio.Restriction.Restriction.NmuCI-class.html#size,Variable Bio.Restriction.Restriction.NotI.size=Bio.Restriction.Restriction.NotI-class.html#size,Variable Bio.Restriction.Restriction.NpeUS61II.size=Bio.Restriction.Restriction.NpeUS61II-class.html#size,Variable Bio.Restriction.Restriction.NruI.size=Bio.Restriction.Restriction.NruI-class.html#size,Variable Bio.Restriction.Restriction.NsbI.size=Bio.Restriction.Restriction.NsbI-class.html#size,Variable Bio.Restriction.Restriction.NsiI.size=Bio.Restriction.Restriction.NsiI-class.html#size,Variable Bio.Restriction.Restriction.NspBII.size=Bio.Restriction.Restriction.NspBII-class.html#size,Variable Bio.Restriction.Restriction.NspI.size=Bio.Restriction.Restriction.NspI-class.html#size,Variable Bio.Restriction.Restriction.NspV.size=Bio.Restriction.Restriction.NspV-class.html#size,Variable Bio.Restriction.Restriction.OliI.size=Bio.Restriction.Restriction.OliI-class.html#size,Variable Bio.Restriction.Restriction.OspHL35III.size=Bio.Restriction.Restriction.OspHL35III-class.html#size,Variable Bio.Restriction.Restriction.PabI.size=Bio.Restriction.Restriction.PabI-class.html#size,Variable Bio.Restriction.Restriction.Pac19842II.size=Bio.Restriction.Restriction.Pac19842II-class.html#size,Variable Bio.Restriction.Restriction.PacI.size=Bio.Restriction.Restriction.PacI-class.html#size,Variable Bio.Restriction.Restriction.PacIII.size=Bio.Restriction.Restriction.PacIII-class.html#size,Variable Bio.Restriction.Restriction.PaeI.size=Bio.Restriction.Restriction.PaeI-class.html#size,Variable Bio.Restriction.Restriction.PaeR7I.size=Bio.Restriction.Restriction.PaeR7I-class.html#size,Variable Bio.Restriction.Restriction.PagI.size=Bio.Restriction.Restriction.PagI-class.html#size,Variable Bio.Restriction.Restriction.Pal408I.size=Bio.Restriction.Restriction.Pal408I-class.html#size,Variable Bio.Restriction.Restriction.PalAI.size=Bio.Restriction.Restriction.PalAI-class.html#size,Variable Bio.Restriction.Restriction.PasI.size=Bio.Restriction.Restriction.PasI-class.html#size,Variable Bio.Restriction.Restriction.PauI.size=Bio.Restriction.Restriction.PauI-class.html#size,Variable Bio.Restriction.Restriction.PcaII.size=Bio.Restriction.Restriction.PcaII-class.html#size,Variable Bio.Restriction.Restriction.PceI.size=Bio.Restriction.Restriction.PceI-class.html#size,Variable Bio.Restriction.Restriction.PciI.size=Bio.Restriction.Restriction.PciI-class.html#size,Variable Bio.Restriction.Restriction.PciSI.size=Bio.Restriction.Restriction.PciSI-class.html#size,Variable Bio.Restriction.Restriction.PcsI.size=Bio.Restriction.Restriction.PcsI-class.html#size,Variable Bio.Restriction.Restriction.PctI.size=Bio.Restriction.Restriction.PctI-class.html#size,Variable Bio.Restriction.Restriction.Pdi8503III.size=Bio.Restriction.Restriction.Pdi8503III-class.html#size,Variable Bio.Restriction.Restriction.PdiI.size=Bio.Restriction.Restriction.PdiI-class.html#size,Variable Bio.Restriction.Restriction.PdmI.size=Bio.Restriction.Restriction.PdmI-class.html#size,Variable Bio.Restriction.Restriction.Pdu1735I.size=Bio.Restriction.Restriction.Pdu1735I-class.html#size,Variable Bio.Restriction.Restriction.PenI.size=Bio.Restriction.Restriction.PenI-class.html#size,Variable Bio.Restriction.Restriction.PfeI.size=Bio.Restriction.Restriction.PfeI-class.html#size,Variable Bio.Restriction.Restriction.Pfl1108I.size=Bio.Restriction.Restriction.Pfl1108I-class.html#size,Variable Bio.Restriction.Restriction.Pfl23II.size=Bio.Restriction.Restriction.Pfl23II-class.html#size,Variable Bio.Restriction.Restriction.Pfl8569I.size=Bio.Restriction.Restriction.Pfl8569I-class.html#size,Variable Bio.Restriction.Restriction.PflFI.size=Bio.Restriction.Restriction.PflFI-class.html#size,Variable Bio.Restriction.Restriction.PflMI.size=Bio.Restriction.Restriction.PflMI-class.html#size,Variable Bio.Restriction.Restriction.PfoI.size=Bio.Restriction.Restriction.PfoI-class.html#size,Variable Bio.Restriction.Restriction.PfrJS12IV.size=Bio.Restriction.Restriction.PfrJS12IV-class.html#size,Variable Bio.Restriction.Restriction.PfrJS12V.size=Bio.Restriction.Restriction.PfrJS12V-class.html#size,Variable Bio.Restriction.Restriction.PfrJS15III.size=Bio.Restriction.Restriction.PfrJS15III-class.html#size,Variable Bio.Restriction.Restriction.Pin17FIII.size=Bio.Restriction.Restriction.Pin17FIII-class.html#size,Variable Bio.Restriction.Restriction.PinAI.size=Bio.Restriction.Restriction.PinAI-class.html#size,Variable Bio.Restriction.Restriction.PkrI.size=Bio.Restriction.Restriction.PkrI-class.html#size,Variable Bio.Restriction.Restriction.PlaDI.size=Bio.Restriction.Restriction.PlaDI-class.html#size,Variable Bio.Restriction.Restriction.Ple19I.size=Bio.Restriction.Restriction.Ple19I-class.html#size,Variable Bio.Restriction.Restriction.PleI.size=Bio.Restriction.Restriction.PleI-class.html#size,Variable Bio.Restriction.Restriction.PliMI.size=Bio.Restriction.Restriction.PliMI-class.html#size,Variable Bio.Restriction.Restriction.PluTI.size=Bio.Restriction.Restriction.PluTI-class.html#size,Variable Bio.Restriction.Restriction.PmaCI.size=Bio.Restriction.Restriction.PmaCI-class.html#size,Variable Bio.Restriction.Restriction.PmeI.size=Bio.Restriction.Restriction.PmeI-class.html#size,Variable Bio.Restriction.Restriction.PmlI.size=Bio.Restriction.Restriction.PmlI-class.html#size,Variable Bio.Restriction.Restriction.PpiI.size=Bio.Restriction.Restriction.PpiI-class.html#size,Variable Bio.Restriction.Restriction.PpsI.size=Bio.Restriction.Restriction.PpsI-class.html#size,Variable Bio.Restriction.Restriction.Ppu10I.size=Bio.Restriction.Restriction.Ppu10I-class.html#size,Variable Bio.Restriction.Restriction.Ppu21I.size=Bio.Restriction.Restriction.Ppu21I-class.html#size,Variable Bio.Restriction.Restriction.PpuMI.size=Bio.Restriction.Restriction.PpuMI-class.html#size,Variable Bio.Restriction.Restriction.PscI.size=Bio.Restriction.Restriction.PscI-class.html#size,Variable Bio.Restriction.Restriction.Pse18267I.size=Bio.Restriction.Restriction.Pse18267I-class.html#size,Variable Bio.Restriction.Restriction.PshAI.size=Bio.Restriction.Restriction.PshAI-class.html#size,Variable Bio.Restriction.Restriction.PshBI.size=Bio.Restriction.Restriction.PshBI-class.html#size,Variable Bio.Restriction.Restriction.PsiI.size=Bio.Restriction.Restriction.PsiI-class.html#size,Variable Bio.Restriction.Restriction.Psp0357II.size=Bio.Restriction.Restriction.Psp0357II-class.html#size,Variable Bio.Restriction.Restriction.Psp03I.size=Bio.Restriction.Restriction.Psp03I-class.html#size,Variable Bio.Restriction.Restriction.Psp124BI.size=Bio.Restriction.Restriction.Psp124BI-class.html#size,Variable Bio.Restriction.Restriction.Psp1406I.size=Bio.Restriction.Restriction.Psp1406I-class.html#size,Variable Bio.Restriction.Restriction.Psp5II.size=Bio.Restriction.Restriction.Psp5II-class.html#size,Variable Bio.Restriction.Restriction.Psp6I.size=Bio.Restriction.Restriction.Psp6I-class.html#size,Variable Bio.Restriction.Restriction.PspCI.size=Bio.Restriction.Restriction.PspCI-class.html#size,Variable Bio.Restriction.Restriction.PspEI.size=Bio.Restriction.Restriction.PspEI-class.html#size,Variable Bio.Restriction.Restriction.PspFI.size=Bio.Restriction.Restriction.PspFI-class.html#size,Variable Bio.Restriction.Restriction.PspGI.size=Bio.Restriction.Restriction.PspGI-class.html#size,Variable Bio.Restriction.Restriction.PspLI.size=Bio.Restriction.Restriction.PspLI-class.html#size,Variable Bio.Restriction.Restriction.PspN4I.size=Bio.Restriction.Restriction.PspN4I-class.html#size,Variable Bio.Restriction.Restriction.PspOMI.size=Bio.Restriction.Restriction.PspOMI-class.html#size,Variable Bio.Restriction.Restriction.PspOMII.size=Bio.Restriction.Restriction.PspOMII-class.html#size,Variable Bio.Restriction.Restriction.PspPI.size=Bio.Restriction.Restriction.PspPI-class.html#size,Variable Bio.Restriction.Restriction.PspPPI.size=Bio.Restriction.Restriction.PspPPI-class.html#size,Variable Bio.Restriction.Restriction.PspPRI.size=Bio.Restriction.Restriction.PspPRI-class.html#size,Variable Bio.Restriction.Restriction.PspXI.size=Bio.Restriction.Restriction.PspXI-class.html#size,Variable Bio.Restriction.Restriction.PsrI.size=Bio.Restriction.Restriction.PsrI-class.html#size,Variable Bio.Restriction.Restriction.PssI.size=Bio.Restriction.Restriction.PssI-class.html#size,Variable Bio.Restriction.Restriction.Pst14472I.size=Bio.Restriction.Restriction.Pst14472I-class.html#size,Variable Bio.Restriction.Restriction.Pst273I.size=Bio.Restriction.Restriction.Pst273I-class.html#size,Variable Bio.Restriction.Restriction.PstI.size=Bio.Restriction.Restriction.PstI-class.html#size,Variable Bio.Restriction.Restriction.PstNI.size=Bio.Restriction.Restriction.PstNI-class.html#size,Variable Bio.Restriction.Restriction.PsuGI.size=Bio.Restriction.Restriction.PsuGI-class.html#size,Variable Bio.Restriction.Restriction.PsuI.size=Bio.Restriction.Restriction.PsuI-class.html#size,Variable Bio.Restriction.Restriction.PsyI.size=Bio.Restriction.Restriction.PsyI-class.html#size,Variable Bio.Restriction.Restriction.PteI.size=Bio.Restriction.Restriction.PteI-class.html#size,Variable Bio.Restriction.Restriction.PvuI.size=Bio.Restriction.Restriction.PvuI-class.html#size,Variable Bio.Restriction.Restriction.PvuII.size=Bio.Restriction.Restriction.PvuII-class.html#size,Variable Bio.Restriction.Restriction.Rba2021I.size=Bio.Restriction.Restriction.Rba2021I-class.html#size,Variable Bio.Restriction.Restriction.RceI.size=Bio.Restriction.Restriction.RceI-class.html#size,Variable Bio.Restriction.Restriction.RdeGBI.size=Bio.Restriction.Restriction.RdeGBI-class.html#size,Variable Bio.Restriction.Restriction.RdeGBII.size=Bio.Restriction.Restriction.RdeGBII-class.html#size,Variable Bio.Restriction.Restriction.RdeGBIII.size=Bio.Restriction.Restriction.RdeGBIII-class.html#size,Variable Bio.Restriction.Restriction.RflFIII.size=Bio.Restriction.Restriction.RflFIII-class.html#size,Variable Bio.Restriction.Restriction.RgaI.size=Bio.Restriction.Restriction.RgaI-class.html#size,Variable Bio.Restriction.Restriction.RigI.size=Bio.Restriction.Restriction.RigI-class.html#size,Variable Bio.Restriction.Restriction.RlaI.size=Bio.Restriction.Restriction.RlaI-class.html#size,Variable Bio.Restriction.Restriction.RlaII.size=Bio.Restriction.Restriction.RlaII-class.html#size,Variable Bio.Restriction.Restriction.RleAI.size=Bio.Restriction.Restriction.RleAI-class.html#size,Variable Bio.Restriction.Restriction.RpaB5I.size=Bio.Restriction.Restriction.RpaB5I-class.html#size,Variable Bio.Restriction.Restriction.RpaBI.size=Bio.Restriction.Restriction.RpaBI-class.html#size,Variable Bio.Restriction.Restriction.RpaI.size=Bio.Restriction.Restriction.RpaI-class.html#size,Variable Bio.Restriction.Restriction.RpaTI.size=Bio.Restriction.Restriction.RpaTI-class.html#size,Variable Bio.Restriction.Restriction.RruI.size=Bio.Restriction.Restriction.RruI-class.html#size,Variable Bio.Restriction.Restriction.RsaI.size=Bio.Restriction.Restriction.RsaI-class.html#size,Variable Bio.Restriction.Restriction.RsaNI.size=Bio.Restriction.Restriction.RsaNI-class.html#size,Variable Bio.Restriction.Restriction.RseI.size=Bio.Restriction.Restriction.RseI-class.html#size,Variable Bio.Restriction.Restriction.Rsp008IV.size=Bio.Restriction.Restriction.Rsp008IV-class.html#size,Variable Bio.Restriction.Restriction.Rsp008V.size=Bio.Restriction.Restriction.Rsp008V-class.html#size,Variable Bio.Restriction.Restriction.RspPBTS2III.size=Bio.Restriction.Restriction.RspPBTS2III-class.html#size,Variable Bio.Restriction.Restriction.Rsr2I.size=Bio.Restriction.Restriction.Rsr2I-class.html#size,Variable Bio.Restriction.Restriction.RsrII.size=Bio.Restriction.Restriction.RsrII-class.html#size,Variable Bio.Restriction.Restriction.SacI.size=Bio.Restriction.Restriction.SacI-class.html#size,Variable Bio.Restriction.Restriction.SacII.size=Bio.Restriction.Restriction.SacII-class.html#size,Variable Bio.Restriction.Restriction.Saf8902III.size=Bio.Restriction.Restriction.Saf8902III-class.html#size,Variable Bio.Restriction.Restriction.SalI.size=Bio.Restriction.Restriction.SalI-class.html#size,Variable Bio.Restriction.Restriction.SanDI.size=Bio.Restriction.Restriction.SanDI-class.html#size,Variable Bio.Restriction.Restriction.SapI.size=Bio.Restriction.Restriction.SapI-class.html#size,Variable Bio.Restriction.Restriction.SaqAI.size=Bio.Restriction.Restriction.SaqAI-class.html#size,Variable Bio.Restriction.Restriction.SatI.size=Bio.Restriction.Restriction.SatI-class.html#size,Variable Bio.Restriction.Restriction.Sau3AI.size=Bio.Restriction.Restriction.Sau3AI-class.html#size,Variable Bio.Restriction.Restriction.Sau96I.size=Bio.Restriction.Restriction.Sau96I-class.html#size,Variable Bio.Restriction.Restriction.SauI.size=Bio.Restriction.Restriction.SauI-class.html#size,Variable Bio.Restriction.Restriction.SbfI.size=Bio.Restriction.Restriction.SbfI-class.html#size,Variable Bio.Restriction.Restriction.Sbo46I.size=Bio.Restriction.Restriction.Sbo46I-class.html#size,Variable Bio.Restriction.Restriction.ScaI.size=Bio.Restriction.Restriction.ScaI-class.html#size,Variable Bio.Restriction.Restriction.SchI.size=Bio.Restriction.Restriction.SchI-class.html#size,Variable Bio.Restriction.Restriction.SciI.size=Bio.Restriction.Restriction.SciI-class.html#size,Variable Bio.Restriction.Restriction.ScoDS2II.size=Bio.Restriction.Restriction.ScoDS2II-class.html#size,Variable Bio.Restriction.Restriction.ScrFI.size=Bio.Restriction.Restriction.ScrFI-class.html#size,Variable Bio.Restriction.Restriction.SdaI.size=Bio.Restriction.Restriction.SdaI-class.html#size,Variable Bio.Restriction.Restriction.SdeAI.size=Bio.Restriction.Restriction.SdeAI-class.html#size,Variable Bio.Restriction.Restriction.SdeOSI.size=Bio.Restriction.Restriction.SdeOSI-class.html#size,Variable Bio.Restriction.Restriction.SduI.size=Bio.Restriction.Restriction.SduI-class.html#size,Variable Bio.Restriction.Restriction.SecI.size=Bio.Restriction.Restriction.SecI-class.html#size,Variable Bio.Restriction.Restriction.SelI.size=Bio.Restriction.Restriction.SelI-class.html#size,Variable Bio.Restriction.Restriction.Sen13311III.size=Bio.Restriction.Restriction.Sen13311III-class.html#size,Variable Bio.Restriction.Restriction.Sen13III.size=Bio.Restriction.Restriction.Sen13III-class.html#size,Variable Bio.Restriction.Restriction.Sen1736II.size=Bio.Restriction.Restriction.Sen1736II-class.html#size,Variable Bio.Restriction.Restriction.Sen1878IV.size=Bio.Restriction.Restriction.Sen1878IV-class.html#size,Variable Bio.Restriction.Restriction.Sen1880IV.size=Bio.Restriction.Restriction.Sen1880IV-class.html#size,Variable Bio.Restriction.Restriction.Sen1896IV.size=Bio.Restriction.Restriction.Sen1896IV-class.html#size,Variable Bio.Restriction.Restriction.Sen1898IV.size=Bio.Restriction.Restriction.Sen1898IV-class.html#size,Variable Bio.Restriction.Restriction.Sen1906IV.size=Bio.Restriction.Restriction.Sen1906IV-class.html#size,Variable Bio.Restriction.Restriction.Sen1908IV.size=Bio.Restriction.Restriction.Sen1908IV-class.html#size,Variable Bio.Restriction.Restriction.Sen1910IV.size=Bio.Restriction.Restriction.Sen1910IV-class.html#size,Variable Bio.Restriction.Restriction.SenC1808III.size=Bio.Restriction.Restriction.SenC1808III-class.html#size,Variable Bio.Restriction.Restriction.SenC1810IV.size=Bio.Restriction.Restriction.SenC1810IV-class.html#size,Variable Bio.Restriction.Restriction.SenSARA26III.size=Bio.Restriction.Restriction.SenSARA26III-class.html#size,Variable Bio.Restriction.Restriction.SenTFIV.size=Bio.Restriction.Restriction.SenTFIV-class.html#size,Variable Bio.Restriction.Restriction.SetI.size=Bio.Restriction.Restriction.SetI-class.html#size,Variable Bio.Restriction.Restriction.SexAI.size=Bio.Restriction.Restriction.SexAI-class.html#size,Variable Bio.Restriction.Restriction.SfaAI.size=Bio.Restriction.Restriction.SfaAI-class.html#size,Variable Bio.Restriction.Restriction.SfaNI.size=Bio.Restriction.Restriction.SfaNI-class.html#size,Variable Bio.Restriction.Restriction.SfcI.size=Bio.Restriction.Restriction.SfcI-class.html#size,Variable Bio.Restriction.Restriction.SfeI.size=Bio.Restriction.Restriction.SfeI-class.html#size,Variable Bio.Restriction.Restriction.SfiI.size=Bio.Restriction.Restriction.SfiI-class.html#size,Variable Bio.Restriction.Restriction.SfoI.size=Bio.Restriction.Restriction.SfoI-class.html#size,Variable Bio.Restriction.Restriction.Sfr274I.size=Bio.Restriction.Restriction.Sfr274I-class.html#size,Variable Bio.Restriction.Restriction.Sfr303I.size=Bio.Restriction.Restriction.Sfr303I-class.html#size,Variable Bio.Restriction.Restriction.SfuI.size=Bio.Restriction.Restriction.SfuI-class.html#size,Variable Bio.Restriction.Restriction.SgeI.size=Bio.Restriction.Restriction.SgeI-class.html#size,Variable Bio.Restriction.Restriction.SgfI.size=Bio.Restriction.Restriction.SgfI-class.html#size,Variable Bio.Restriction.Restriction.SgrAI.size=Bio.Restriction.Restriction.SgrAI-class.html#size,Variable Bio.Restriction.Restriction.SgrBI.size=Bio.Restriction.Restriction.SgrBI-class.html#size,Variable Bio.Restriction.Restriction.SgrDI.size=Bio.Restriction.Restriction.SgrDI-class.html#size,Variable Bio.Restriction.Restriction.SgrTI.size=Bio.Restriction.Restriction.SgrTI-class.html#size,Variable Bio.Restriction.Restriction.SgsI.size=Bio.Restriction.Restriction.SgsI-class.html#size,Variable Bio.Restriction.Restriction.SimI.size=Bio.Restriction.Restriction.SimI-class.html#size,Variable Bio.Restriction.Restriction.SinI.size=Bio.Restriction.Restriction.SinI-class.html#size,Variable Bio.Restriction.Restriction.SlaI.size=Bio.Restriction.Restriction.SlaI-class.html#size,Variable Bio.Restriction.Restriction.SmaI.size=Bio.Restriction.Restriction.SmaI-class.html#size,Variable Bio.Restriction.Restriction.SmiI.size=Bio.Restriction.Restriction.SmiI-class.html#size,Variable Bio.Restriction.Restriction.SmiMI.size=Bio.Restriction.Restriction.SmiMI-class.html#size,Variable Bio.Restriction.Restriction.SmlI.size=Bio.Restriction.Restriction.SmlI-class.html#size,Variable Bio.Restriction.Restriction.SmoI.size=Bio.Restriction.Restriction.SmoI-class.html#size,Variable Bio.Restriction.Restriction.SnaBI.size=Bio.Restriction.Restriction.SnaBI-class.html#size,Variable Bio.Restriction.Restriction.SnaI.size=Bio.Restriction.Restriction.SnaI-class.html#size,Variable Bio.Restriction.Restriction.Sno506I.size=Bio.Restriction.Restriction.Sno506I-class.html#size,Variable Bio.Restriction.Restriction.SpeI.size=Bio.Restriction.Restriction.SpeI-class.html#size,Variable Bio.Restriction.Restriction.SphI.size=Bio.Restriction.Restriction.SphI-class.html#size,Variable Bio.Restriction.Restriction.SplI.size=Bio.Restriction.Restriction.SplI-class.html#size,Variable Bio.Restriction.Restriction.SpnRII.size=Bio.Restriction.Restriction.SpnRII-class.html#size,Variable Bio.Restriction.Restriction.SpoDI.size=Bio.Restriction.Restriction.SpoDI-class.html#size,Variable Bio.Restriction.Restriction.SrfI.size=Bio.Restriction.Restriction.SrfI-class.html#size,Variable Bio.Restriction.Restriction.Sse232I.size=Bio.Restriction.Restriction.Sse232I-class.html#size,Variable Bio.Restriction.Restriction.Sse8387I.size=Bio.Restriction.Restriction.Sse8387I-class.html#size,Variable Bio.Restriction.Restriction.Sse8647I.size=Bio.Restriction.Restriction.Sse8647I-class.html#size,Variable Bio.Restriction.Restriction.Sse9I.size=Bio.Restriction.Restriction.Sse9I-class.html#size,Variable Bio.Restriction.Restriction.SseBI.size=Bio.Restriction.Restriction.SseBI-class.html#size,Variable Bio.Restriction.Restriction.SsiI.size=Bio.Restriction.Restriction.SsiI-class.html#size,Variable Bio.Restriction.Restriction.Ssp6803IV.size=Bio.Restriction.Restriction.Ssp6803IV-class.html#size,Variable Bio.Restriction.Restriction.Ssp714II.size=Bio.Restriction.Restriction.Ssp714II-class.html#size,Variable Bio.Restriction.Restriction.SspD5I.size=Bio.Restriction.Restriction.SspD5I-class.html#size,Variable Bio.Restriction.Restriction.SspDI.size=Bio.Restriction.Restriction.SspDI-class.html#size,Variable Bio.Restriction.Restriction.SspI.size=Bio.Restriction.Restriction.SspI-class.html#size,Variable Bio.Restriction.Restriction.SspMI.size=Bio.Restriction.Restriction.SspMI-class.html#size,Variable Bio.Restriction.Restriction.SstE37I.size=Bio.Restriction.Restriction.SstE37I-class.html#size,Variable Bio.Restriction.Restriction.SstI.size=Bio.Restriction.Restriction.SstI-class.html#size,Variable Bio.Restriction.Restriction.Sth132I.size=Bio.Restriction.Restriction.Sth132I-class.html#size,Variable Bio.Restriction.Restriction.Sth20745III.size=Bio.Restriction.Restriction.Sth20745III-class.html#size,Variable Bio.Restriction.Restriction.Sth302II.size=Bio.Restriction.Restriction.Sth302II-class.html#size,Variable Bio.Restriction.Restriction.StsI.size=Bio.Restriction.Restriction.StsI-class.html#size,Variable Bio.Restriction.Restriction.StuI.size=Bio.Restriction.Restriction.StuI-class.html#size,Variable Bio.Restriction.Restriction.Sty13348III.size=Bio.Restriction.Restriction.Sty13348III-class.html#size,Variable Bio.Restriction.Restriction.StyD4I.size=Bio.Restriction.Restriction.StyD4I-class.html#size,Variable Bio.Restriction.Restriction.StyI.size=Bio.Restriction.Restriction.StyI-class.html#size,Variable Bio.Restriction.Restriction.SwaI.size=Bio.Restriction.Restriction.SwaI-class.html#size,Variable Bio.Restriction.Restriction.TaaI.size=Bio.Restriction.Restriction.TaaI-class.html#size,Variable Bio.Restriction.Restriction.TaiI.size=Bio.Restriction.Restriction.TaiI-class.html#size,Variable Bio.Restriction.Restriction.TaqI.size=Bio.Restriction.Restriction.TaqI-class.html#size,Variable Bio.Restriction.Restriction.TaqII.size=Bio.Restriction.Restriction.TaqII-class.html#size,Variable Bio.Restriction.Restriction.TaqIII.size=Bio.Restriction.Restriction.TaqIII-class.html#size,Variable Bio.Restriction.Restriction.TasI.size=Bio.Restriction.Restriction.TasI-class.html#size,Variable Bio.Restriction.Restriction.TatI.size=Bio.Restriction.Restriction.TatI-class.html#size,Variable Bio.Restriction.Restriction.TauI.size=Bio.Restriction.Restriction.TauI-class.html#size,Variable Bio.Restriction.Restriction.TfiI.size=Bio.Restriction.Restriction.TfiI-class.html#size,Variable Bio.Restriction.Restriction.Tru1I.size=Bio.Restriction.Restriction.Tru1I-class.html#size,Variable Bio.Restriction.Restriction.Tru9I.size=Bio.Restriction.Restriction.Tru9I-class.html#size,Variable Bio.Restriction.Restriction.TscAI.size=Bio.Restriction.Restriction.TscAI-class.html#size,Variable Bio.Restriction.Restriction.TseFI.size=Bio.Restriction.Restriction.TseFI-class.html#size,Variable Bio.Restriction.Restriction.TseI.size=Bio.Restriction.Restriction.TseI-class.html#size,Variable Bio.Restriction.Restriction.TsoI.size=Bio.Restriction.Restriction.TsoI-class.html#size,Variable Bio.Restriction.Restriction.Tsp45I.size=Bio.Restriction.Restriction.Tsp45I-class.html#size,Variable Bio.Restriction.Restriction.Tsp4CI.size=Bio.Restriction.Restriction.Tsp4CI-class.html#size,Variable Bio.Restriction.Restriction.TspARh3I.size=Bio.Restriction.Restriction.TspARh3I-class.html#size,Variable Bio.Restriction.Restriction.TspDTI.size=Bio.Restriction.Restriction.TspDTI-class.html#size,Variable Bio.Restriction.Restriction.TspEI.size=Bio.Restriction.Restriction.TspEI-class.html#size,Variable Bio.Restriction.Restriction.TspGWI.size=Bio.Restriction.Restriction.TspGWI-class.html#size,Variable Bio.Restriction.Restriction.TspMI.size=Bio.Restriction.Restriction.TspMI-class.html#size,Variable Bio.Restriction.Restriction.TspRI.size=Bio.Restriction.Restriction.TspRI-class.html#size,Variable Bio.Restriction.Restriction.TssI.size=Bio.Restriction.Restriction.TssI-class.html#size,Variable Bio.Restriction.Restriction.TstI.size=Bio.Restriction.Restriction.TstI-class.html#size,Variable Bio.Restriction.Restriction.TsuI.size=Bio.Restriction.Restriction.TsuI-class.html#size,Variable Bio.Restriction.Restriction.Tth111I.size=Bio.Restriction.Restriction.Tth111I-class.html#size,Variable Bio.Restriction.Restriction.Tth111II.size=Bio.Restriction.Restriction.Tth111II-class.html#size,Variable Bio.Restriction.Restriction.UbaF11I.size=Bio.Restriction.Restriction.UbaF11I-class.html#size,Variable Bio.Restriction.Restriction.UbaF12I.size=Bio.Restriction.Restriction.UbaF12I-class.html#size,Variable Bio.Restriction.Restriction.UbaF13I.size=Bio.Restriction.Restriction.UbaF13I-class.html#size,Variable Bio.Restriction.Restriction.UbaF14I.size=Bio.Restriction.Restriction.UbaF14I-class.html#size,Variable Bio.Restriction.Restriction.UbaF9I.size=Bio.Restriction.Restriction.UbaF9I-class.html#size,Variable Bio.Restriction.Restriction.UbaPI.size=Bio.Restriction.Restriction.UbaPI-class.html#size,Variable Bio.Restriction.Restriction.UcoMSI.size=Bio.Restriction.Restriction.UcoMSI-class.html#size,Variable Bio.Restriction.Restriction.UnbI.size=Bio.Restriction.Restriction.UnbI-class.html#size,Variable Bio.Restriction.Restriction.Van91I.size=Bio.Restriction.Restriction.Van91I-class.html#size,Variable Bio.Restriction.Restriction.Vha464I.size=Bio.Restriction.Restriction.Vha464I-class.html#size,Variable Bio.Restriction.Restriction.VneI.size=Bio.Restriction.Restriction.VneI-class.html#size,Variable Bio.Restriction.Restriction.VpaK11AI.size=Bio.Restriction.Restriction.VpaK11AI-class.html#size,Variable Bio.Restriction.Restriction.VpaK11BI.size=Bio.Restriction.Restriction.VpaK11BI-class.html#size,Variable Bio.Restriction.Restriction.VspI.size=Bio.Restriction.Restriction.VspI-class.html#size,Variable Bio.Restriction.Restriction.Vtu19109I.size=Bio.Restriction.Restriction.Vtu19109I-class.html#size,Variable Bio.Restriction.Restriction.WviI.size=Bio.Restriction.Restriction.WviI-class.html#size,Variable Bio.Restriction.Restriction.XagI.size=Bio.Restriction.Restriction.XagI-class.html#size,Variable Bio.Restriction.Restriction.XapI.size=Bio.Restriction.Restriction.XapI-class.html#size,Variable Bio.Restriction.Restriction.XbaI.size=Bio.Restriction.Restriction.XbaI-class.html#size,Variable Bio.Restriction.Restriction.Xca85IV.size=Bio.Restriction.Restriction.Xca85IV-class.html#size,Variable Bio.Restriction.Restriction.XceI.size=Bio.Restriction.Restriction.XceI-class.html#size,Variable Bio.Restriction.Restriction.XcmI.size=Bio.Restriction.Restriction.XcmI-class.html#size,Variable Bio.Restriction.Restriction.XhoI.size=Bio.Restriction.Restriction.XhoI-class.html#size,Variable Bio.Restriction.Restriction.XhoII.size=Bio.Restriction.Restriction.XhoII-class.html#size,Variable Bio.Restriction.Restriction.XmaI.size=Bio.Restriction.Restriction.XmaI-class.html#size,Variable Bio.Restriction.Restriction.XmaIII.size=Bio.Restriction.Restriction.XmaIII-class.html#size,Variable Bio.Restriction.Restriction.XmaJI.size=Bio.Restriction.Restriction.XmaJI-class.html#size,Variable Bio.Restriction.Restriction.XmiI.size=Bio.Restriction.Restriction.XmiI-class.html#size,Variable Bio.Restriction.Restriction.XmnI.size=Bio.Restriction.Restriction.XmnI-class.html#size,Variable Bio.Restriction.Restriction.XspI.size=Bio.Restriction.Restriction.XspI-class.html#size,Variable Bio.Restriction.Restriction.YkrI.size=Bio.Restriction.Restriction.YkrI-class.html#size,Variable Bio.Restriction.Restriction.Yps3606I.size=Bio.Restriction.Restriction.Yps3606I-class.html#size,Variable Bio.Restriction.Restriction.ZraI.size=Bio.Restriction.Restriction.ZraI-class.html#size,Variable Bio.Restriction.Restriction.ZrmI.size=Bio.Restriction.Restriction.ZrmI-class.html#size,Variable Bio.Restriction.Restriction.Zsp2I.size=Bio.Restriction.Restriction.Zsp2I-class.html#size,Variable Bio.codonalign.codonalphabet.CodonAlphabet.size=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#size"><a title="Bio.Alphabet.Alphabet.size
Bio.Alphabet.Reduced.HPModel.size
Bio.Alphabet.Reduced.Murphy10.size
Bio.Alphabet.Reduced.Murphy15.size
Bio.Alphabet.Reduced.Murphy4.size
Bio.Alphabet.Reduced.Murphy8.size
Bio.Alphabet.Reduced.PC5.size
Bio.Alphabet.SingleLetterAlphabet.size
Bio.Alphabet.ThreeLetterProtein.size
Bio.GenBank._FeatureConsumer.size
Bio.GenBank._RecordConsumer.size
Bio.Restriction.Restriction.AanI.size
Bio.Restriction.Restriction.AarI.size
Bio.Restriction.Restriction.AasI.size
Bio.Restriction.Restriction.AatII.size
Bio.Restriction.Restriction.Aba6411II.size
Bio.Restriction.Restriction.AbaCIII.size
Bio.Restriction.Restriction.AbaSI.size
Bio.Restriction.Restriction.AbaUMB2I.size
Bio.Restriction.Restriction.AbsI.size
Bio.Restriction.Restriction.Acc16I.size
Bio.Restriction.Restriction.Acc36I.size
Bio.Restriction.Restriction.Acc65I.size
Bio.Restriction.Restriction.Acc65V.size
Bio.Restriction.Restriction.AccB1I.size
Bio.Restriction.Restriction.AccB7I.size
Bio.Restriction.Restriction.AccBSI.size
Bio.Restriction.Restriction.AccI.size
Bio.Restriction.Restriction.AccII.size
Bio.Restriction.Restriction.AccIII.size
Bio.Restriction.Restriction.AceIII.size
Bio.Restriction.Restriction.AchA6III.size
Bio.Restriction.Restriction.AciI.size
Bio.Restriction.Restriction.AclI.size
Bio.Restriction.Restriction.AclWI.size
Bio.Restriction.Restriction.Aco12261II.size
Bio.Restriction.Restriction.AcoI.size
Bio.Restriction.Restriction.AcsI.size
Bio.Restriction.Restriction.AcuI.size
Bio.Restriction.Restriction.AcvI.size
Bio.Restriction.Restriction.AcyI.size
Bio.Restriction.Restriction.AdeI.size
Bio.Restriction.Restriction.Adh6U21I.size
Bio.Restriction.Restriction.AfaI.size
Bio.Restriction.Restriction.AfeI.size
Bio.Restriction.Restriction.AfiI.size
Bio.Restriction.Restriction.AflII.size
Bio.Restriction.Restriction.AflIII.size
Bio.Restriction.Restriction.AgeI.size
Bio.Restriction.Restriction.AgsI.size
Bio.Restriction.Restriction.AhaIII.size
Bio.Restriction.Restriction.AhdI.size
Bio.Restriction.Restriction.AhlI.size
Bio.Restriction.Restriction.AhyRBAHI.size
Bio.Restriction.Restriction.AhyYL17I.size
Bio.Restriction.Restriction.AjiI.size
Bio.Restriction.Restriction.AjnI.size
Bio.Restriction.Restriction.AjuI.size
Bio.Restriction.Restriction.AleI.size
Bio.Restriction.Restriction.AlfI.size
Bio.Restriction.Restriction.AloI.size
Bio.Restriction.Restriction.AluBI.size
Bio.Restriction.Restriction.AluI.size
Bio.Restriction.Restriction.Alw21I.size
Bio.Restriction.Restriction.Alw26I.size
Bio.Restriction.Restriction.Alw44I.size
Bio.Restriction.Restriction.AlwFI.size
Bio.Restriction.Restriction.AlwI.size
Bio.Restriction.Restriction.AlwNI.size
Bio.Restriction.Restriction.Ama87I.size
Bio.Restriction.Restriction.AmaCSI.size
Bio.Restriction.Restriction.Aor13HI.size
Bio.Restriction.Restriction.Aor51HI.size
Bio.Restriction.Restriction.AoxI.size
Bio.Restriction.Restriction.ApaBI.size
Bio.Restriction.Restriction.ApaI.size
Bio.Restriction.Restriction.ApaLI.size
Bio.Restriction.Restriction.ApeKI.size
Bio.Restriction.Restriction.ApoI.size
Bio.Restriction.Restriction.ApyPI.size
Bio.Restriction.Restriction.AquII.size
Bio.Restriction.Restriction.AquIII.size
Bio.Restriction.Restriction.AquIV.size
Bio.Restriction.Restriction.ArsI.size
Bio.Restriction.Restriction.AscI.size
Bio.Restriction.Restriction.AseI.size
Bio.Restriction.Restriction.Asi256I.size
Bio.Restriction.Restriction.AsiGI.size
Bio.Restriction.Restriction.AsiSI.size
Bio.Restriction.Restriction.Asp114II.size
Bio.Restriction.Restriction.Asp700I.size
Bio.Restriction.Restriction.Asp718I.size
Bio.Restriction.Restriction.AspA2I.size
Bio.Restriction.Restriction.AspBHI.size
Bio.Restriction.Restriction.AspDUT2V.size
Bio.Restriction.Restriction.AspJHL3II.size
Bio.Restriction.Restriction.AspLEI.size
Bio.Restriction.Restriction.AspS9I.size
Bio.Restriction.Restriction.Asu14238IV.size
Bio.Restriction.Restriction.AsuC2I.size
Bio.Restriction.Restriction.AsuHPI.size
Bio.Restriction.Restriction.AsuI.size
Bio.Restriction.Restriction.AsuII.size
Bio.Restriction.Restriction.AsuNHI.size
Bio.Restriction.Restriction.AteTI.size
Bio.Restriction.Restriction.AvaI.size
Bio.Restriction.Restriction.AvaII.size
Bio.Restriction.Restriction.AvaIII.size
Bio.Restriction.Restriction.AvrII.size
Bio.Restriction.Restriction.Awo1030IV.size
Bio.Restriction.Restriction.AxyI.size
Bio.Restriction.Restriction.BaeGI.size
Bio.Restriction.Restriction.BaeI.size
Bio.Restriction.Restriction.Bag18758I.size
Bio.Restriction.Restriction.BalI.size
Bio.Restriction.Restriction.BamHI.size
Bio.Restriction.Restriction.BanI.size
Bio.Restriction.Restriction.BanII.size
Bio.Restriction.Restriction.BanLI.size
Bio.Restriction.Restriction.BarI.size
Bio.Restriction.Restriction.BauI.size
Bio.Restriction.Restriction.BbrPI.size
Bio.Restriction.Restriction.BbsI.size
Bio.Restriction.Restriction.Bbv12I.size
Bio.Restriction.Restriction.BbvCI.size
Bio.Restriction.Restriction.BbvI.size
Bio.Restriction.Restriction.BbvII.size
Bio.Restriction.Restriction.BccI.size
Bio.Restriction.Restriction.Bce3081I.size
Bio.Restriction.Restriction.Bce83I.size
Bio.Restriction.Restriction.BceAI.size
Bio.Restriction.Restriction.BceSIV.size
Bio.Restriction.Restriction.BcefI.size
Bio.Restriction.Restriction.BcgI.size
Bio.Restriction.Restriction.BciT130I.size
Bio.Restriction.Restriction.BciVI.size
Bio.Restriction.Restriction.BclI.size
Bio.Restriction.Restriction.BcnI.size
Bio.Restriction.Restriction.BcoDI.size
Bio.Restriction.Restriction.BcuI.size
Bio.Restriction.Restriction.BdaI.size
Bio.Restriction.Restriction.BetI.size
Bio.Restriction.Restriction.BfaI.size
Bio.Restriction.Restriction.BfaSII.size
Bio.Restriction.Restriction.BfiI.size
Bio.Restriction.Restriction.BfmI.size
Bio.Restriction.Restriction.BfoI.size
Bio.Restriction.Restriction.BfrI.size
Bio.Restriction.Restriction.BfuAI.size
Bio.Restriction.Restriction.BfuI.size
Bio.Restriction.Restriction.BglI.size
Bio.Restriction.Restriction.BglII.size
Bio.Restriction.Restriction.BinI.size
Bio.Restriction.Restriction.BisI.size
Bio.Restriction.Restriction.Ble402II.size
Bio.Restriction.Restriction.BlnI.size
Bio.Restriction.Restriction.BloAII.size
Bio.Restriction.Restriction.BlpI.size
Bio.Restriction.Restriction.BlsI.size
Bio.Restriction.Restriction.BmcAI.size
Bio.Restriction.Restriction.Bme1390I.size
Bio.Restriction.Restriction.Bme18I.size
Bio.Restriction.Restriction.BmeDI.size
Bio.Restriction.Restriction.BmeRI.size
Bio.Restriction.Restriction.BmeT110I.size
Bio.Restriction.Restriction.BmgBI.size
Bio.Restriction.Restriction.BmgI.size
Bio.Restriction.Restriction.BmgT120I.size
Bio.Restriction.Restriction.BmiI.size
Bio.Restriction.Restriction.BmrFI.size
Bio.Restriction.Restriction.BmrI.size
Bio.Restriction.Restriction.BmsI.size
Bio.Restriction.Restriction.BmtI.size
Bio.Restriction.Restriction.BmuI.size
Bio.Restriction.Restriction.BoxI.size
Bio.Restriction.Restriction.BpiI.size
Bio.Restriction.Restriction.BplI.size
Bio.Restriction.Restriction.BpmI.size
Bio.Restriction.Restriction.Bpu10I.size
Bio.Restriction.Restriction.Bpu1102I.size
Bio.Restriction.Restriction.Bpu14I.size
Bio.Restriction.Restriction.BpuEI.size
Bio.Restriction.Restriction.BpuMI.size
Bio.Restriction.Restriction.Bsa29I.size
Bio.Restriction.Restriction.BsaAI.size
Bio.Restriction.Restriction.BsaBI.size
Bio.Restriction.Restriction.BsaHI.size
Bio.Restriction.Restriction.BsaI.size
Bio.Restriction.Restriction.BsaJI.size
Bio.Restriction.Restriction.BsaWI.size
Bio.Restriction.Restriction.BsaXI.size
Bio.Restriction.Restriction.BsbI.size
Bio.Restriction.Restriction.Bsc4I.size
Bio.Restriction.Restriction.BscAI.size
Bio.Restriction.Restriction.BscGI.size
Bio.Restriction.Restriction.Bse118I.size
Bio.Restriction.Restriction.Bse1I.size
Bio.Restriction.Restriction.Bse21I.size
Bio.Restriction.Restriction.Bse3DI.size
Bio.Restriction.Restriction.Bse8I.size
Bio.Restriction.Restriction.BseAI.size
Bio.Restriction.Restriction.BseBI.size
Bio.Restriction.Restriction.BseCI.size
Bio.Restriction.Restriction.BseDI.size
Bio.Restriction.Restriction.BseGI.size
Bio.Restriction.Restriction.BseJI.size
Bio.Restriction.Restriction.BseLI.size
Bio.Restriction.Restriction.BseMI.size
Bio.Restriction.Restriction.BseMII.size
Bio.Restriction.Restriction.BseNI.size
Bio.Restriction.Restriction.BsePI.size
Bio.Restriction.Restriction.BseRI.size
Bio.Restriction.Restriction.BseSI.size
Bio.Restriction.Restriction.BseX3I.size
Bio.Restriction.Restriction.BseXI.size
Bio.Restriction.Restriction.BseYI.size
Bio.Restriction.Restriction.BsgI.size
Bio.Restriction.Restriction.Bsh1236I.size
Bio.Restriction.Restriction.Bsh1285I.size
Bio.Restriction.Restriction.BshFI.size
Bio.Restriction.Restriction.BshNI.size
Bio.Restriction.Restriction.BshTI.size
Bio.Restriction.Restriction.BshVI.size
Bio.Restriction.Restriction.BsiEI.size
Bio.Restriction.Restriction.BsiHKAI.size
Bio.Restriction.Restriction.BsiHKCI.size
Bio.Restriction.Restriction.BsiI.size
Bio.Restriction.Restriction.BsiSI.size
Bio.Restriction.Restriction.BsiWI.size
Bio.Restriction.Restriction.BsiYI.size
Bio.Restriction.Restriction.BslFI.size
Bio.Restriction.Restriction.BslI.size
Bio.Restriction.Restriction.BsmAI.size
Bio.Restriction.Restriction.BsmBI.size
Bio.Restriction.Restriction.BsmFI.size
Bio.Restriction.Restriction.BsmI.size
Bio.Restriction.Restriction.BsnI.size
Bio.Restriction.Restriction.Bso31I.size
Bio.Restriction.Restriction.BsoBI.size
Bio.Restriction.Restriction.Bsp119I.size
Bio.Restriction.Restriction.Bsp120I.size
Bio.Restriction.Restriction.Bsp1286I.size
Bio.Restriction.Restriction.Bsp13I.size
Bio.Restriction.Restriction.Bsp1407I.size
Bio.Restriction.Restriction.Bsp143I.size
Bio.Restriction.Restriction.Bsp1720I.size
Bio.Restriction.Restriction.Bsp19I.size
Bio.Restriction.Restriction.Bsp24I.size
Bio.Restriction.Restriction.Bsp3004IV.size
Bio.Restriction.Restriction.Bsp460III.size
Bio.Restriction.Restriction.Bsp68I.size
Bio.Restriction.Restriction.BspACI.size
Bio.Restriction.Restriction.BspANI.size
Bio.Restriction.Restriction.BspCNI.size
Bio.Restriction.Restriction.BspD6I.size
Bio.Restriction.Restriction.BspDI.size
Bio.Restriction.Restriction.BspEI.size
Bio.Restriction.Restriction.BspFNI.size
Bio.Restriction.Restriction.BspGI.size
Bio.Restriction.Restriction.BspHI.size
Bio.Restriction.Restriction.BspLI.size
Bio.Restriction.Restriction.BspLU11I.size
Bio.Restriction.Restriction.BspMAI.size
Bio.Restriction.Restriction.BspMI.size
Bio.Restriction.Restriction.BspMII.size
Bio.Restriction.Restriction.BspNCI.size
Bio.Restriction.Restriction.BspOI.size
Bio.Restriction.Restriction.BspPI.size
Bio.Restriction.Restriction.BspQI.size
Bio.Restriction.Restriction.BspT104I.size
Bio.Restriction.Restriction.BspT107I.size
Bio.Restriction.Restriction.BspTI.size
Bio.Restriction.Restriction.BspTNI.size
Bio.Restriction.Restriction.BsrBI.size
Bio.Restriction.Restriction.BsrDI.size
Bio.Restriction.Restriction.BsrFI.size
Bio.Restriction.Restriction.BsrGI.size
Bio.Restriction.Restriction.BsrI.size
Bio.Restriction.Restriction.BssAI.size
Bio.Restriction.Restriction.BssECI.size
Bio.Restriction.Restriction.BssHII.size
Bio.Restriction.Restriction.BssMI.size
Bio.Restriction.Restriction.BssNAI.size
Bio.Restriction.Restriction.BssNI.size
Bio.Restriction.Restriction.BssSI.size
Bio.Restriction.Restriction.BssT1I.size
Bio.Restriction.Restriction.Bst1107I.size
Bio.Restriction.Restriction.Bst2BI.size
Bio.Restriction.Restriction.Bst2UI.size
Bio.Restriction.Restriction.Bst4CI.size
Bio.Restriction.Restriction.Bst6I.size
Bio.Restriction.Restriction.BstACI.size
Bio.Restriction.Restriction.BstAFI.size
Bio.Restriction.Restriction.BstAPI.size
Bio.Restriction.Restriction.BstAUI.size
Bio.Restriction.Restriction.BstBAI.size
Bio.Restriction.Restriction.BstBI.size
Bio.Restriction.Restriction.BstC8I.size
Bio.Restriction.Restriction.BstDEI.size
Bio.Restriction.Restriction.BstDSI.size
Bio.Restriction.Restriction.BstEII.size
Bio.Restriction.Restriction.BstENI.size
Bio.Restriction.Restriction.BstF5I.size
Bio.Restriction.Restriction.BstFNI.size
Bio.Restriction.Restriction.BstH2I.size
Bio.Restriction.Restriction.BstHHI.size
Bio.Restriction.Restriction.BstKTI.size
Bio.Restriction.Restriction.BstMAI.size
Bio.Restriction.Restriction.BstMBI.size
Bio.Restriction.Restriction.BstMCI.size
Bio.Restriction.Restriction.BstMWI.size
Bio.Restriction.Restriction.BstNI.size
Bio.Restriction.Restriction.BstNSI.size
Bio.Restriction.Restriction.BstPAI.size
Bio.Restriction.Restriction.BstPI.size
Bio.Restriction.Restriction.BstSCI.size
Bio.Restriction.Restriction.BstSFI.size
Bio.Restriction.Restriction.BstSLI.size
Bio.Restriction.Restriction.BstSNI.size
Bio.Restriction.Restriction.BstUI.size
Bio.Restriction.Restriction.BstV1I.size
Bio.Restriction.Restriction.BstV2I.size
Bio.Restriction.Restriction.BstX2I.size
Bio.Restriction.Restriction.BstXI.size
Bio.Restriction.Restriction.BstYI.size
Bio.Restriction.Restriction.BstZ17I.size
Bio.Restriction.Restriction.BstZI.size
Bio.Restriction.Restriction.Bsu15I.size
Bio.Restriction.Restriction.Bsu36I.size
Bio.Restriction.Restriction.BsuI.size
Bio.Restriction.Restriction.BsuRI.size
Bio.Restriction.Restriction.BsuTUI.size
Bio.Restriction.Restriction.BtgI.size
Bio.Restriction.Restriction.BtgZI.size
Bio.Restriction.Restriction.BthCI.size
Bio.Restriction.Restriction.BtrI.size
Bio.Restriction.Restriction.BtsCI.size
Bio.Restriction.Restriction.BtsI.size
Bio.Restriction.Restriction.BtsIMutI.size
Bio.Restriction.Restriction.BtuMI.size
Bio.Restriction.Restriction.BveI.size
Bio.Restriction.Restriction.Cac8I.size
Bio.Restriction.Restriction.CaiI.size
Bio.Restriction.Restriction.Cal14237I.size
Bio.Restriction.Restriction.CalB3II.size
Bio.Restriction.Restriction.Cau10061II.size
Bio.Restriction.Restriction.CauII.size
Bio.Restriction.Restriction.Cba13II.size
Bio.Restriction.Restriction.Cba16038I.size
Bio.Restriction.Restriction.Cbo67071IV.size
Bio.Restriction.Restriction.Cch467III.size
Bio.Restriction.Restriction.CchII.size
Bio.Restriction.Restriction.CchIII.size
Bio.Restriction.Restriction.CciI.size
Bio.Restriction.Restriction.CciNI.size
Bio.Restriction.Restriction.CcrNAIII.size
Bio.Restriction.Restriction.Cdi11397I.size
Bio.Restriction.Restriction.Cdi81III.size
Bio.Restriction.Restriction.CdiI.size
Bio.Restriction.Restriction.CdpI.size
Bio.Restriction.Restriction.Cdu23823II.size
Bio.Restriction.Restriction.CfoI.size
Bio.Restriction.Restriction.Cfr10I.size
Bio.Restriction.Restriction.Cfr13I.size
Bio.Restriction.Restriction.Cfr42I.size
Bio.Restriction.Restriction.Cfr9I.size
Bio.Restriction.Restriction.CfrI.size
Bio.Restriction.Restriction.Cfupf3II.size
Bio.Restriction.Restriction.Cgl13032I.size
Bio.Restriction.Restriction.Cgl13032II.size
Bio.Restriction.Restriction.ChaI.size
Bio.Restriction.Restriction.CjeFIII.size
Bio.Restriction.Restriction.CjeFV.size
Bio.Restriction.Restriction.CjeI.size
Bio.Restriction.Restriction.CjeNII.size
Bio.Restriction.Restriction.CjeNIII.size
Bio.Restriction.Restriction.CjeNV.size
Bio.Restriction.Restriction.CjeP659IV.size
Bio.Restriction.Restriction.CjePI.size
Bio.Restriction.Restriction.CjuI.size
Bio.Restriction.Restriction.CjuII.size
Bio.Restriction.Restriction.Cla11845III.size
Bio.Restriction.Restriction.ClaI.size
Bio.Restriction.Restriction.Cly7489II.size
Bio.Restriction.Restriction.Cma23826I.size
Bio.Restriction.Restriction.CpoI.size
Bio.Restriction.Restriction.CseI.size
Bio.Restriction.Restriction.CsiI.size
Bio.Restriction.Restriction.Csp2014I.size
Bio.Restriction.Restriction.Csp6I.size
Bio.Restriction.Restriction.CspAI.size
Bio.Restriction.Restriction.CspCI.size
Bio.Restriction.Restriction.CspI.size
Bio.Restriction.Restriction.CstMI.size
Bio.Restriction.Restriction.CviAII.size
Bio.Restriction.Restriction.CviJI.size
Bio.Restriction.Restriction.CviKI_1.size
Bio.Restriction.Restriction.CviQI.size
Bio.Restriction.Restriction.CviRI.size
Bio.Restriction.Restriction.Dde51507I.size
Bio.Restriction.Restriction.DdeI.size
Bio.Restriction.Restriction.DinI.size
Bio.Restriction.Restriction.DpnI.size
Bio.Restriction.Restriction.DpnII.size
Bio.Restriction.Restriction.DraI.size
Bio.Restriction.Restriction.DraII.size
Bio.Restriction.Restriction.DraIII.size
Bio.Restriction.Restriction.DraRI.size
Bio.Restriction.Restriction.DrdI.size
Bio.Restriction.Restriction.DrdII.size
Bio.Restriction.Restriction.DriI.size
Bio.Restriction.Restriction.DsaI.size
Bio.Restriction.Restriction.DseDI.size
Bio.Restriction.Restriction.EaeI.size
Bio.Restriction.Restriction.EagI.size
Bio.Restriction.Restriction.Eam1104I.size
Bio.Restriction.Restriction.Eam1105I.size
Bio.Restriction.Restriction.EarI.size
Bio.Restriction.Restriction.EciI.size
Bio.Restriction.Restriction.Ecl136II.size
Bio.Restriction.Restriction.Ecl35734I.size
Bio.Restriction.Restriction.EclXI.size
Bio.Restriction.Restriction.Eco105I.size
Bio.Restriction.Restriction.Eco130I.size
Bio.Restriction.Restriction.Eco147I.size
Bio.Restriction.Restriction.Eco24I.size
Bio.Restriction.Restriction.Eco31I.size
Bio.Restriction.Restriction.Eco32I.size
Bio.Restriction.Restriction.Eco4465II.size
Bio.Restriction.Restriction.Eco47I.size
Bio.Restriction.Restriction.Eco47III.size
Bio.Restriction.Restriction.Eco52I.size
Bio.Restriction.Restriction.Eco53kI.size
Bio.Restriction.Restriction.Eco57I.size
Bio.Restriction.Restriction.Eco57MI.size
Bio.Restriction.Restriction.Eco72I.size
Bio.Restriction.Restriction.Eco81I.size
Bio.Restriction.Restriction.Eco88I.size
Bio.Restriction.Restriction.Eco91I.size
Bio.Restriction.Restriction.EcoBLMcrX.size
Bio.Restriction.Restriction.EcoE1140I.size
Bio.Restriction.Restriction.EcoHI.size
Bio.Restriction.Restriction.EcoICRI.size
Bio.Restriction.Restriction.EcoMVII.size
Bio.Restriction.Restriction.EcoNI.size
Bio.Restriction.Restriction.EcoO109I.size
Bio.Restriction.Restriction.EcoO65I.size
Bio.Restriction.Restriction.EcoRI.size
Bio.Restriction.Restriction.EcoRII.size
Bio.Restriction.Restriction.EcoRV.size
Bio.Restriction.Restriction.EcoT14I.size
Bio.Restriction.Restriction.EcoT22I.size
Bio.Restriction.Restriction.EcoT38I.size
Bio.Restriction.Restriction.EgeI.size
Bio.Restriction.Restriction.EheI.size
Bio.Restriction.Restriction.Eli8509II.size
Bio.Restriction.Restriction.ErhI.size
Bio.Restriction.Restriction.EsaBC3I.size
Bio.Restriction.Restriction.EsaSSI.size
Bio.Restriction.Restriction.Esp3007I.size
Bio.Restriction.Restriction.Esp3I.size
Bio.Restriction.Restriction.EspI.size
Bio.Restriction.Restriction.Exi27195I.size
Bio.Restriction.Restriction.FaeI.size
Bio.Restriction.Restriction.FaiI.size
Bio.Restriction.Restriction.FalI.size
Bio.Restriction.Restriction.FaqI.size
Bio.Restriction.Restriction.FatI.size
Bio.Restriction.Restriction.FauI.size
Bio.Restriction.Restriction.FauNDI.size
Bio.Restriction.Restriction.FbaI.size
Bio.Restriction.Restriction.FblI.size
Bio.Restriction.Restriction.FinI.size
Bio.Restriction.Restriction.FmuI.size
Bio.Restriction.Restriction.Fnu4HI.size
Bio.Restriction.Restriction.FnuDII.size
Bio.Restriction.Restriction.FokI.size
Bio.Restriction.Restriction.FriOI.size
Bio.Restriction.Restriction.FseI.size
Bio.Restriction.Restriction.Fsp4HI.size
Bio.Restriction.Restriction.FspAI.size
Bio.Restriction.Restriction.FspBI.size
Bio.Restriction.Restriction.FspEI.size
Bio.Restriction.Restriction.FspI.size
Bio.Restriction.Restriction.FtnUV.size
Bio.Restriction.Restriction.GauT27I.size
Bio.Restriction.Restriction.Gba708II.size
Bio.Restriction.Restriction.GdiII.size
Bio.Restriction.Restriction.GlaI.size
Bio.Restriction.Restriction.GluI.size
Bio.Restriction.Restriction.GsaI.size
Bio.Restriction.Restriction.GsuI.size
Bio.Restriction.Restriction.HaeI.size
Bio.Restriction.Restriction.HaeII.size
Bio.Restriction.Restriction.HaeIII.size
Bio.Restriction.Restriction.HapII.size
Bio.Restriction.Restriction.HauII.size
Bio.Restriction.Restriction.HbaII.size
Bio.Restriction.Restriction.HgaI.size
Bio.Restriction.Restriction.HgiAI.size
Bio.Restriction.Restriction.HgiCI.size
Bio.Restriction.Restriction.HgiEII.size
Bio.Restriction.Restriction.HgiJII.size
Bio.Restriction.Restriction.HhaI.size
Bio.Restriction.Restriction.Hin1I.size
Bio.Restriction.Restriction.Hin1II.size
Bio.Restriction.Restriction.Hin4I.size
Bio.Restriction.Restriction.Hin4II.size
Bio.Restriction.Restriction.Hin6I.size
Bio.Restriction.Restriction.HinP1I.size
Bio.Restriction.Restriction.HincII.size
Bio.Restriction.Restriction.HindII.size
Bio.Restriction.Restriction.HindIII.size
Bio.Restriction.Restriction.HinfI.size
Bio.Restriction.Restriction.HpaI.size
Bio.Restriction.Restriction.HpaII.size
Bio.Restriction.Restriction.HphI.size
Bio.Restriction.Restriction.Hpy166II.size
Bio.Restriction.Restriction.Hpy178III.size
Bio.Restriction.Restriction.Hpy188I.size
Bio.Restriction.Restriction.Hpy188III.size
Bio.Restriction.Restriction.Hpy8I.size
Bio.Restriction.Restriction.Hpy99I.size
Bio.Restriction.Restriction.Hpy99XIII.size
Bio.Restriction.Restriction.Hpy99XIV.size
Bio.Restriction.Restriction.Hpy99XIV_mut1.size
Bio.Restriction.Restriction.Hpy99XXII.size
Bio.Restriction.Restriction.HpyAV.size
Bio.Restriction.Restriction.HpyAXIV.size
Bio.Restriction.Restriction.HpyAXVI_mut1.size
Bio.Restriction.Restriction.HpyAXVI_mut2.size
Bio.Restriction.Restriction.HpyCH4III.size
Bio.Restriction.Restriction.HpyCH4IV.size
Bio.Restriction.Restriction.HpyCH4V.size
Bio.Restriction.Restriction.HpyF10VI.size
Bio.Restriction.Restriction.HpyF3I.size
Bio.Restriction.Restriction.HpySE526I.size
Bio.Restriction.Restriction.HpyUM032XIII.size
Bio.Restriction.Restriction.HpyUM032XIII_mut1.size
Bio.Restriction.Restriction.HpyUM032XIV.size
Bio.Restriction.Restriction.HpyUM037X.size
Bio.Restriction.Restriction.Hsp92I.size
Bio.Restriction.Restriction.Hsp92II.size
Bio.Restriction.Restriction.HspAI.size
Bio.Restriction.Restriction.Jma19592I.size
Bio.Restriction.Restriction.Jma19592II.size
Bio.Restriction.Restriction.Jsp2502II.size
Bio.Restriction.Restriction.KasI.size
Bio.Restriction.Restriction.KflI.size
Bio.Restriction.Restriction.Kor51II.size
Bio.Restriction.Restriction.Kpn2I.size
Bio.Restriction.Restriction.KpnI.size
Bio.Restriction.Restriction.KpnNIH30III.size
Bio.Restriction.Restriction.KroI.size
Bio.Restriction.Restriction.Ksp22I.size
Bio.Restriction.Restriction.Ksp632I.size
Bio.Restriction.Restriction.KspAI.size
Bio.Restriction.Restriction.KspI.size
Bio.Restriction.Restriction.Kzo9I.size
Bio.Restriction.Restriction.Lba2029III.size
Bio.Restriction.Restriction.LguI.size
Bio.Restriction.Restriction.LmnI.size
Bio.Restriction.Restriction.Lmo370I.size
Bio.Restriction.Restriction.Lmo911II.size
Bio.Restriction.Restriction.LpnI.size
Bio.Restriction.Restriction.LpnPI.size
Bio.Restriction.Restriction.Lsp1109I.size
Bio.Restriction.Restriction.Lsp48III.size
Bio.Restriction.Restriction.Lsp6406VI.size
Bio.Restriction.Restriction.LweI.size
Bio.Restriction.Restriction.MabI.size
Bio.Restriction.Restriction.MaeI.size
Bio.Restriction.Restriction.MaeII.size
Bio.Restriction.Restriction.MaeIII.size
Bio.Restriction.Restriction.MalI.size
Bio.Restriction.Restriction.MaqI.size
Bio.Restriction.Restriction.MauBI.size
Bio.Restriction.Restriction.Mba11I.size
Bio.Restriction.Restriction.MbiI.size
Bio.Restriction.Restriction.MboI.size
Bio.Restriction.Restriction.MboII.size
Bio.Restriction.Restriction.McaTI.size
Bio.Restriction.Restriction.Mcr10I.size
Bio.Restriction.Restriction.McrI.size
Bio.Restriction.Restriction.MfeI.size
Bio.Restriction.Restriction.MflI.size
Bio.Restriction.Restriction.MhlI.size
Bio.Restriction.Restriction.MjaIV.size
Bio.Restriction.Restriction.MkaDII.size
Bio.Restriction.Restriction.MlsI.size
Bio.Restriction.Restriction.MluCI.size
Bio.Restriction.Restriction.MluI.size
Bio.Restriction.Restriction.MluNI.size
Bio.Restriction.Restriction.Mly113I.size
Bio.Restriction.Restriction.MlyI.size
Bio.Restriction.Restriction.MmeI.size
Bio.Restriction.Restriction.MnlI.size
Bio.Restriction.Restriction.Mox20I.size
Bio.Restriction.Restriction.Mph1103I.size
Bio.Restriction.Restriction.MreI.size
Bio.Restriction.Restriction.MroI.size
Bio.Restriction.Restriction.MroNI.size
Bio.Restriction.Restriction.MroXI.size
Bio.Restriction.Restriction.MscI.size
Bio.Restriction.Restriction.MseI.size
Bio.Restriction.Restriction.MslI.size
Bio.Restriction.Restriction.Msp20I.size
Bio.Restriction.Restriction.MspA1I.size
Bio.Restriction.Restriction.MspCI.size
Bio.Restriction.Restriction.MspI.size
Bio.Restriction.Restriction.MspI7II.size
Bio.Restriction.Restriction.MspJI.size
Bio.Restriction.Restriction.MspR9I.size
Bio.Restriction.Restriction.MssI.size
Bio.Restriction.Restriction.MstI.size
Bio.Restriction.Restriction.MteI.size
Bio.Restriction.Restriction.MtuHN878II.size
Bio.Restriction.Restriction.MunI.size
Bio.Restriction.Restriction.Mva1269I.size
Bio.Restriction.Restriction.MvaI.size
Bio.Restriction.Restriction.MvnI.size
Bio.Restriction.Restriction.MwoI.size
Bio.Restriction.Restriction.NaeI.size
Bio.Restriction.Restriction.Nal45188II.size
Bio.Restriction.Restriction.NarI.size
Bio.Restriction.Restriction.Nbr128II.size
Bio.Restriction.Restriction.NciI.size
Bio.Restriction.Restriction.NcoI.size
Bio.Restriction.Restriction.NdeI.size
Bio.Restriction.Restriction.NdeII.size
Bio.Restriction.Restriction.NgoAVII.size
Bio.Restriction.Restriction.NgoAVIII.size
Bio.Restriction.Restriction.NgoMIV.size
Bio.Restriction.Restriction.NhaXI.size
Bio.Restriction.Restriction.NheI.size
Bio.Restriction.Restriction.NhoI.size
Bio.Restriction.Restriction.NlaCI.size
Bio.Restriction.Restriction.NlaIII.size
Bio.Restriction.Restriction.NlaIV.size
Bio.Restriction.Restriction.Nli3877I.size
Bio.Restriction.Restriction.NmeAIII.size
Bio.Restriction.Restriction.NmeDI.size
Bio.Restriction.Restriction.NmuCI.size
Bio.Restriction.Restriction.NotI.size
Bio.Restriction.Restriction.NpeUS61II.size
Bio.Restriction.Restriction.NruI.size
Bio.Restriction.Restriction.NsbI.size
Bio.Restriction.Restriction.NsiI.size
Bio.Restriction.Restriction.NspBII.size
Bio.Restriction.Restriction.NspI.size
Bio.Restriction.Restriction.NspV.size
Bio.Restriction.Restriction.OliI.size
Bio.Restriction.Restriction.OspHL35III.size
Bio.Restriction.Restriction.PabI.size
Bio.Restriction.Restriction.Pac19842II.size
Bio.Restriction.Restriction.PacI.size
Bio.Restriction.Restriction.PacIII.size
Bio.Restriction.Restriction.PaeI.size
Bio.Restriction.Restriction.PaeR7I.size
Bio.Restriction.Restriction.PagI.size
Bio.Restriction.Restriction.Pal408I.size
Bio.Restriction.Restriction.PalAI.size
Bio.Restriction.Restriction.PasI.size
Bio.Restriction.Restriction.PauI.size
Bio.Restriction.Restriction.PcaII.size
Bio.Restriction.Restriction.PceI.size
Bio.Restriction.Restriction.PciI.size
Bio.Restriction.Restriction.PciSI.size
Bio.Restriction.Restriction.PcsI.size
Bio.Restriction.Restriction.PctI.size
Bio.Restriction.Restriction.Pdi8503III.size
Bio.Restriction.Restriction.PdiI.size
Bio.Restriction.Restriction.PdmI.size
Bio.Restriction.Restriction.Pdu1735I.size
Bio.Restriction.Restriction.PenI.size
Bio.Restriction.Restriction.PfeI.size
Bio.Restriction.Restriction.Pfl1108I.size
Bio.Restriction.Restriction.Pfl23II.size
Bio.Restriction.Restriction.Pfl8569I.size
Bio.Restriction.Restriction.PflFI.size
Bio.Restriction.Restriction.PflMI.size
Bio.Restriction.Restriction.PfoI.size
Bio.Restriction.Restriction.PfrJS12IV.size
Bio.Restriction.Restriction.PfrJS12V.size
Bio.Restriction.Restriction.PfrJS15III.size
Bio.Restriction.Restriction.Pin17FIII.size
Bio.Restriction.Restriction.PinAI.size
Bio.Restriction.Restriction.PkrI.size
Bio.Restriction.Restriction.PlaDI.size
Bio.Restriction.Restriction.Ple19I.size
Bio.Restriction.Restriction.PleI.size
Bio.Restriction.Restriction.PliMI.size
Bio.Restriction.Restriction.PluTI.size
Bio.Restriction.Restriction.PmaCI.size
Bio.Restriction.Restriction.PmeI.size
Bio.Restriction.Restriction.PmlI.size
Bio.Restriction.Restriction.PpiI.size
Bio.Restriction.Restriction.PpsI.size
Bio.Restriction.Restriction.Ppu10I.size
Bio.Restriction.Restriction.Ppu21I.size
Bio.Restriction.Restriction.PpuMI.size
Bio.Restriction.Restriction.PscI.size
Bio.Restriction.Restriction.Pse18267I.size
Bio.Restriction.Restriction.PshAI.size
Bio.Restriction.Restriction.PshBI.size
Bio.Restriction.Restriction.PsiI.size
Bio.Restriction.Restriction.Psp0357II.size
Bio.Restriction.Restriction.Psp03I.size
Bio.Restriction.Restriction.Psp124BI.size
Bio.Restriction.Restriction.Psp1406I.size
Bio.Restriction.Restriction.Psp5II.size
Bio.Restriction.Restriction.Psp6I.size
Bio.Restriction.Restriction.PspCI.size
Bio.Restriction.Restriction.PspEI.size
Bio.Restriction.Restriction.PspFI.size
Bio.Restriction.Restriction.PspGI.size
Bio.Restriction.Restriction.PspLI.size
Bio.Restriction.Restriction.PspN4I.size
Bio.Restriction.Restriction.PspOMI.size
Bio.Restriction.Restriction.PspOMII.size
Bio.Restriction.Restriction.PspPI.size
Bio.Restriction.Restriction.PspPPI.size
Bio.Restriction.Restriction.PspPRI.size
Bio.Restriction.Restriction.PspXI.size
Bio.Restriction.Restriction.PsrI.size
Bio.Restriction.Restriction.PssI.size
Bio.Restriction.Restriction.Pst14472I.size
Bio.Restriction.Restriction.Pst273I.size
Bio.Restriction.Restriction.PstI.size
Bio.Restriction.Restriction.PstNI.size
Bio.Restriction.Restriction.PsuGI.size
Bio.Restriction.Restriction.PsuI.size
Bio.Restriction.Restriction.PsyI.size
Bio.Restriction.Restriction.PteI.size
Bio.Restriction.Restriction.PvuI.size
Bio.Restriction.Restriction.PvuII.size
Bio.Restriction.Restriction.Rba2021I.size
Bio.Restriction.Restriction.RceI.size
Bio.Restriction.Restriction.RdeGBI.size
Bio.Restriction.Restriction.RdeGBII.size
Bio.Restriction.Restriction.RdeGBIII.size
Bio.Restriction.Restriction.RflFIII.size
Bio.Restriction.Restriction.RgaI.size
Bio.Restriction.Restriction.RigI.size
Bio.Restriction.Restriction.RlaI.size
Bio.Restriction.Restriction.RlaII.size
Bio.Restriction.Restriction.RleAI.size
Bio.Restriction.Restriction.RpaB5I.size
Bio.Restriction.Restriction.RpaBI.size
Bio.Restriction.Restriction.RpaI.size
Bio.Restriction.Restriction.RpaTI.size
Bio.Restriction.Restriction.RruI.size
Bio.Restriction.Restriction.RsaI.size
Bio.Restriction.Restriction.RsaNI.size
Bio.Restriction.Restriction.RseI.size
Bio.Restriction.Restriction.Rsp008IV.size
Bio.Restriction.Restriction.Rsp008V.size
Bio.Restriction.Restriction.RspPBTS2III.size
Bio.Restriction.Restriction.Rsr2I.size
Bio.Restriction.Restriction.RsrII.size
Bio.Restriction.Restriction.SacI.size
Bio.Restriction.Restriction.SacII.size
Bio.Restriction.Restriction.Saf8902III.size
Bio.Restriction.Restriction.SalI.size
Bio.Restriction.Restriction.SanDI.size
Bio.Restriction.Restriction.SapI.size
Bio.Restriction.Restriction.SaqAI.size
Bio.Restriction.Restriction.SatI.size
Bio.Restriction.Restriction.Sau3AI.size
Bio.Restriction.Restriction.Sau96I.size
Bio.Restriction.Restriction.SauI.size
Bio.Restriction.Restriction.SbfI.size
Bio.Restriction.Restriction.Sbo46I.size
Bio.Restriction.Restriction.ScaI.size
Bio.Restriction.Restriction.SchI.size
Bio.Restriction.Restriction.SciI.size
Bio.Restriction.Restriction.ScoDS2II.size
Bio.Restriction.Restriction.ScrFI.size
Bio.Restriction.Restriction.SdaI.size
Bio.Restriction.Restriction.SdeAI.size
Bio.Restriction.Restriction.SdeOSI.size
Bio.Restriction.Restriction.SduI.size
Bio.Restriction.Restriction.SecI.size
Bio.Restriction.Restriction.SelI.size
Bio.Restriction.Restriction.Sen13311III.size
Bio.Restriction.Restriction.Sen13III.size
Bio.Restriction.Restriction.Sen1736II.size
Bio.Restriction.Restriction.Sen1878IV.size
Bio.Restriction.Restriction.Sen1880IV.size
Bio.Restriction.Restriction.Sen1896IV.size
Bio.Restriction.Restriction.Sen1898IV.size
Bio.Restriction.Restriction.Sen1906IV.size
Bio.Restriction.Restriction.Sen1908IV.size
Bio.Restriction.Restriction.Sen1910IV.size
Bio.Restriction.Restriction.SenC1808III.size
Bio.Restriction.Restriction.SenC1810IV.size
Bio.Restriction.Restriction.SenSARA26III.size
Bio.Restriction.Restriction.SenTFIV.size
Bio.Restriction.Restriction.SetI.size
Bio.Restriction.Restriction.SexAI.size
Bio.Restriction.Restriction.SfaAI.size
Bio.Restriction.Restriction.SfaNI.size
Bio.Restriction.Restriction.SfcI.size
Bio.Restriction.Restriction.SfeI.size
Bio.Restriction.Restriction.SfiI.size
Bio.Restriction.Restriction.SfoI.size
Bio.Restriction.Restriction.Sfr274I.size
Bio.Restriction.Restriction.Sfr303I.size
Bio.Restriction.Restriction.SfuI.size
Bio.Restriction.Restriction.SgeI.size
Bio.Restriction.Restriction.SgfI.size
Bio.Restriction.Restriction.SgrAI.size
Bio.Restriction.Restriction.SgrBI.size
Bio.Restriction.Restriction.SgrDI.size
Bio.Restriction.Restriction.SgrTI.size
Bio.Restriction.Restriction.SgsI.size
Bio.Restriction.Restriction.SimI.size
Bio.Restriction.Restriction.SinI.size
Bio.Restriction.Restriction.SlaI.size
Bio.Restriction.Restriction.SmaI.size
Bio.Restriction.Restriction.SmiI.size
Bio.Restriction.Restriction.SmiMI.size
Bio.Restriction.Restriction.SmlI.size
Bio.Restriction.Restriction.SmoI.size
Bio.Restriction.Restriction.SnaBI.size
Bio.Restriction.Restriction.SnaI.size
Bio.Restriction.Restriction.Sno506I.size
Bio.Restriction.Restriction.SpeI.size
Bio.Restriction.Restriction.SphI.size
Bio.Restriction.Restriction.SplI.size
Bio.Restriction.Restriction.SpnRII.size
Bio.Restriction.Restriction.SpoDI.size
Bio.Restriction.Restriction.SrfI.size
Bio.Restriction.Restriction.Sse232I.size
Bio.Restriction.Restriction.Sse8387I.size
Bio.Restriction.Restriction.Sse8647I.size
Bio.Restriction.Restriction.Sse9I.size
Bio.Restriction.Restriction.SseBI.size
Bio.Restriction.Restriction.SsiI.size
Bio.Restriction.Restriction.Ssp6803IV.size
Bio.Restriction.Restriction.Ssp714II.size
Bio.Restriction.Restriction.SspD5I.size
Bio.Restriction.Restriction.SspDI.size
Bio.Restriction.Restriction.SspI.size
Bio.Restriction.Restriction.SspMI.size
Bio.Restriction.Restriction.SstE37I.size
Bio.Restriction.Restriction.SstI.size
Bio.Restriction.Restriction.Sth132I.size
Bio.Restriction.Restriction.Sth20745III.size
Bio.Restriction.Restriction.Sth302II.size
Bio.Restriction.Restriction.StsI.size
Bio.Restriction.Restriction.StuI.size
Bio.Restriction.Restriction.Sty13348III.size
Bio.Restriction.Restriction.StyD4I.size
Bio.Restriction.Restriction.StyI.size
Bio.Restriction.Restriction.SwaI.size
Bio.Restriction.Restriction.TaaI.size
Bio.Restriction.Restriction.TaiI.size
Bio.Restriction.Restriction.TaqI.size
Bio.Restriction.Restriction.TaqII.size
Bio.Restriction.Restriction.TaqIII.size
Bio.Restriction.Restriction.TasI.size
Bio.Restriction.Restriction.TatI.size
Bio.Restriction.Restriction.TauI.size
Bio.Restriction.Restriction.TfiI.size
Bio.Restriction.Restriction.Tru1I.size
Bio.Restriction.Restriction.Tru9I.size
Bio.Restriction.Restriction.TscAI.size
Bio.Restriction.Restriction.TseFI.size
Bio.Restriction.Restriction.TseI.size
Bio.Restriction.Restriction.TsoI.size
Bio.Restriction.Restriction.Tsp45I.size
Bio.Restriction.Restriction.Tsp4CI.size
Bio.Restriction.Restriction.TspARh3I.size
Bio.Restriction.Restriction.TspDTI.size
Bio.Restriction.Restriction.TspEI.size
Bio.Restriction.Restriction.TspGWI.size
Bio.Restriction.Restriction.TspMI.size
Bio.Restriction.Restriction.TspRI.size
Bio.Restriction.Restriction.TssI.size
Bio.Restriction.Restriction.TstI.size
Bio.Restriction.Restriction.TsuI.size
Bio.Restriction.Restriction.Tth111I.size
Bio.Restriction.Restriction.Tth111II.size
Bio.Restriction.Restriction.UbaF11I.size
Bio.Restriction.Restriction.UbaF12I.size
Bio.Restriction.Restriction.UbaF13I.size
Bio.Restriction.Restriction.UbaF14I.size
Bio.Restriction.Restriction.UbaF9I.size
Bio.Restriction.Restriction.UbaPI.size
Bio.Restriction.Restriction.UcoMSI.size
Bio.Restriction.Restriction.UnbI.size
Bio.Restriction.Restriction.Van91I.size
Bio.Restriction.Restriction.Vha464I.size
Bio.Restriction.Restriction.VneI.size
Bio.Restriction.Restriction.VpaK11AI.size
Bio.Restriction.Restriction.VpaK11BI.size
Bio.Restriction.Restriction.VspI.size
Bio.Restriction.Restriction.Vtu19109I.size
Bio.Restriction.Restriction.WviI.size
Bio.Restriction.Restriction.XagI.size
Bio.Restriction.Restriction.XapI.size
Bio.Restriction.Restriction.XbaI.size
Bio.Restriction.Restriction.Xca85IV.size
Bio.Restriction.Restriction.XceI.size
Bio.Restriction.Restriction.XcmI.size
Bio.Restriction.Restriction.XhoI.size
Bio.Restriction.Restriction.XhoII.size
Bio.Restriction.Restriction.XmaI.size
Bio.Restriction.Restriction.XmaIII.size
Bio.Restriction.Restriction.XmaJI.size
Bio.Restriction.Restriction.XmiI.size
Bio.Restriction.Restriction.XmnI.size
Bio.Restriction.Restriction.XspI.size
Bio.Restriction.Restriction.YkrI.size
Bio.Restriction.Restriction.Yps3606I.size
Bio.Restriction.Restriction.ZraI.size
Bio.Restriction.Restriction.ZrmI.size
Bio.Restriction.Restriction.Zsp2I.size
Bio.codonalign.codonalphabet.CodonAlphabet.size" class="py-name" href="#" onclick="return doclink('link-132', 'size', 'link-132');">size</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L411"></a><tt class="py-lineno">411</tt>  <tt class="py-line">                            <tt class="py-keyword">if</tt> <tt id="link-133" class="py-name"><a title="Bio.Alphabet.Alphabet.size
Bio.Alphabet.Reduced.HPModel.size
Bio.Alphabet.Reduced.Murphy10.size
Bio.Alphabet.Reduced.Murphy15.size
Bio.Alphabet.Reduced.Murphy4.size
Bio.Alphabet.Reduced.Murphy8.size
Bio.Alphabet.Reduced.PC5.size
Bio.Alphabet.SingleLetterAlphabet.size
Bio.Alphabet.ThreeLetterProtein.size
Bio.GenBank._FeatureConsumer.size
Bio.GenBank._RecordConsumer.size
Bio.Restriction.Restriction.AanI.size
Bio.Restriction.Restriction.AarI.size
Bio.Restriction.Restriction.AasI.size
Bio.Restriction.Restriction.AatII.size
Bio.Restriction.Restriction.Aba6411II.size
Bio.Restriction.Restriction.AbaCIII.size
Bio.Restriction.Restriction.AbaSI.size
Bio.Restriction.Restriction.AbaUMB2I.size
Bio.Restriction.Restriction.AbsI.size
Bio.Restriction.Restriction.Acc16I.size
Bio.Restriction.Restriction.Acc36I.size
Bio.Restriction.Restriction.Acc65I.size
Bio.Restriction.Restriction.Acc65V.size
Bio.Restriction.Restriction.AccB1I.size
Bio.Restriction.Restriction.AccB7I.size
Bio.Restriction.Restriction.AccBSI.size
Bio.Restriction.Restriction.AccI.size
Bio.Restriction.Restriction.AccII.size
Bio.Restriction.Restriction.AccIII.size
Bio.Restriction.Restriction.AceIII.size
Bio.Restriction.Restriction.AchA6III.size
Bio.Restriction.Restriction.AciI.size
Bio.Restriction.Restriction.AclI.size
Bio.Restriction.Restriction.AclWI.size
Bio.Restriction.Restriction.Aco12261II.size
Bio.Restriction.Restriction.AcoI.size
Bio.Restriction.Restriction.AcsI.size
Bio.Restriction.Restriction.AcuI.size
Bio.Restriction.Restriction.AcvI.size
Bio.Restriction.Restriction.AcyI.size
Bio.Restriction.Restriction.AdeI.size
Bio.Restriction.Restriction.Adh6U21I.size
Bio.Restriction.Restriction.AfaI.size
Bio.Restriction.Restriction.AfeI.size
Bio.Restriction.Restriction.AfiI.size
Bio.Restriction.Restriction.AflII.size
Bio.Restriction.Restriction.AflIII.size
Bio.Restriction.Restriction.AgeI.size
Bio.Restriction.Restriction.AgsI.size
Bio.Restriction.Restriction.AhaIII.size
Bio.Restriction.Restriction.AhdI.size
Bio.Restriction.Restriction.AhlI.size
Bio.Restriction.Restriction.AhyRBAHI.size
Bio.Restriction.Restriction.AhyYL17I.size
Bio.Restriction.Restriction.AjiI.size
Bio.Restriction.Restriction.AjnI.size
Bio.Restriction.Restriction.AjuI.size
Bio.Restriction.Restriction.AleI.size
Bio.Restriction.Restriction.AlfI.size
Bio.Restriction.Restriction.AloI.size
Bio.Restriction.Restriction.AluBI.size
Bio.Restriction.Restriction.AluI.size
Bio.Restriction.Restriction.Alw21I.size
Bio.Restriction.Restriction.Alw26I.size
Bio.Restriction.Restriction.Alw44I.size
Bio.Restriction.Restriction.AlwFI.size
Bio.Restriction.Restriction.AlwI.size
Bio.Restriction.Restriction.AlwNI.size
Bio.Restriction.Restriction.Ama87I.size
Bio.Restriction.Restriction.AmaCSI.size
Bio.Restriction.Restriction.Aor13HI.size
Bio.Restriction.Restriction.Aor51HI.size
Bio.Restriction.Restriction.AoxI.size
Bio.Restriction.Restriction.ApaBI.size
Bio.Restriction.Restriction.ApaI.size
Bio.Restriction.Restriction.ApaLI.size
Bio.Restriction.Restriction.ApeKI.size
Bio.Restriction.Restriction.ApoI.size
Bio.Restriction.Restriction.ApyPI.size
Bio.Restriction.Restriction.AquII.size
Bio.Restriction.Restriction.AquIII.size
Bio.Restriction.Restriction.AquIV.size
Bio.Restriction.Restriction.ArsI.size
Bio.Restriction.Restriction.AscI.size
Bio.Restriction.Restriction.AseI.size
Bio.Restriction.Restriction.Asi256I.size
Bio.Restriction.Restriction.AsiGI.size
Bio.Restriction.Restriction.AsiSI.size
Bio.Restriction.Restriction.Asp114II.size
Bio.Restriction.Restriction.Asp700I.size
Bio.Restriction.Restriction.Asp718I.size
Bio.Restriction.Restriction.AspA2I.size
Bio.Restriction.Restriction.AspBHI.size
Bio.Restriction.Restriction.AspDUT2V.size
Bio.Restriction.Restriction.AspJHL3II.size
Bio.Restriction.Restriction.AspLEI.size
Bio.Restriction.Restriction.AspS9I.size
Bio.Restriction.Restriction.Asu14238IV.size
Bio.Restriction.Restriction.AsuC2I.size
Bio.Restriction.Restriction.AsuHPI.size
Bio.Restriction.Restriction.AsuI.size
Bio.Restriction.Restriction.AsuII.size
Bio.Restriction.Restriction.AsuNHI.size
Bio.Restriction.Restriction.AteTI.size
Bio.Restriction.Restriction.AvaI.size
Bio.Restriction.Restriction.AvaII.size
Bio.Restriction.Restriction.AvaIII.size
Bio.Restriction.Restriction.AvrII.size
Bio.Restriction.Restriction.Awo1030IV.size
Bio.Restriction.Restriction.AxyI.size
Bio.Restriction.Restriction.BaeGI.size
Bio.Restriction.Restriction.BaeI.size
Bio.Restriction.Restriction.Bag18758I.size
Bio.Restriction.Restriction.BalI.size
Bio.Restriction.Restriction.BamHI.size
Bio.Restriction.Restriction.BanI.size
Bio.Restriction.Restriction.BanII.size
Bio.Restriction.Restriction.BanLI.size
Bio.Restriction.Restriction.BarI.size
Bio.Restriction.Restriction.BauI.size
Bio.Restriction.Restriction.BbrPI.size
Bio.Restriction.Restriction.BbsI.size
Bio.Restriction.Restriction.Bbv12I.size
Bio.Restriction.Restriction.BbvCI.size
Bio.Restriction.Restriction.BbvI.size
Bio.Restriction.Restriction.BbvII.size
Bio.Restriction.Restriction.BccI.size
Bio.Restriction.Restriction.Bce3081I.size
Bio.Restriction.Restriction.Bce83I.size
Bio.Restriction.Restriction.BceAI.size
Bio.Restriction.Restriction.BceSIV.size
Bio.Restriction.Restriction.BcefI.size
Bio.Restriction.Restriction.BcgI.size
Bio.Restriction.Restriction.BciT130I.size
Bio.Restriction.Restriction.BciVI.size
Bio.Restriction.Restriction.BclI.size
Bio.Restriction.Restriction.BcnI.size
Bio.Restriction.Restriction.BcoDI.size
Bio.Restriction.Restriction.BcuI.size
Bio.Restriction.Restriction.BdaI.size
Bio.Restriction.Restriction.BetI.size
Bio.Restriction.Restriction.BfaI.size
Bio.Restriction.Restriction.BfaSII.size
Bio.Restriction.Restriction.BfiI.size
Bio.Restriction.Restriction.BfmI.size
Bio.Restriction.Restriction.BfoI.size
Bio.Restriction.Restriction.BfrI.size
Bio.Restriction.Restriction.BfuAI.size
Bio.Restriction.Restriction.BfuI.size
Bio.Restriction.Restriction.BglI.size
Bio.Restriction.Restriction.BglII.size
Bio.Restriction.Restriction.BinI.size
Bio.Restriction.Restriction.BisI.size
Bio.Restriction.Restriction.Ble402II.size
Bio.Restriction.Restriction.BlnI.size
Bio.Restriction.Restriction.BloAII.size
Bio.Restriction.Restriction.BlpI.size
Bio.Restriction.Restriction.BlsI.size
Bio.Restriction.Restriction.BmcAI.size
Bio.Restriction.Restriction.Bme1390I.size
Bio.Restriction.Restriction.Bme18I.size
Bio.Restriction.Restriction.BmeDI.size
Bio.Restriction.Restriction.BmeRI.size
Bio.Restriction.Restriction.BmeT110I.size
Bio.Restriction.Restriction.BmgBI.size
Bio.Restriction.Restriction.BmgI.size
Bio.Restriction.Restriction.BmgT120I.size
Bio.Restriction.Restriction.BmiI.size
Bio.Restriction.Restriction.BmrFI.size
Bio.Restriction.Restriction.BmrI.size
Bio.Restriction.Restriction.BmsI.size
Bio.Restriction.Restriction.BmtI.size
Bio.Restriction.Restriction.BmuI.size
Bio.Restriction.Restriction.BoxI.size
Bio.Restriction.Restriction.BpiI.size
Bio.Restriction.Restriction.BplI.size
Bio.Restriction.Restriction.BpmI.size
Bio.Restriction.Restriction.Bpu10I.size
Bio.Restriction.Restriction.Bpu1102I.size
Bio.Restriction.Restriction.Bpu14I.size
Bio.Restriction.Restriction.BpuEI.size
Bio.Restriction.Restriction.BpuMI.size
Bio.Restriction.Restriction.Bsa29I.size
Bio.Restriction.Restriction.BsaAI.size
Bio.Restriction.Restriction.BsaBI.size
Bio.Restriction.Restriction.BsaHI.size
Bio.Restriction.Restriction.BsaI.size
Bio.Restriction.Restriction.BsaJI.size
Bio.Restriction.Restriction.BsaWI.size
Bio.Restriction.Restriction.BsaXI.size
Bio.Restriction.Restriction.BsbI.size
Bio.Restriction.Restriction.Bsc4I.size
Bio.Restriction.Restriction.BscAI.size
Bio.Restriction.Restriction.BscGI.size
Bio.Restriction.Restriction.Bse118I.size
Bio.Restriction.Restriction.Bse1I.size
Bio.Restriction.Restriction.Bse21I.size
Bio.Restriction.Restriction.Bse3DI.size
Bio.Restriction.Restriction.Bse8I.size
Bio.Restriction.Restriction.BseAI.size
Bio.Restriction.Restriction.BseBI.size
Bio.Restriction.Restriction.BseCI.size
Bio.Restriction.Restriction.BseDI.size
Bio.Restriction.Restriction.BseGI.size
Bio.Restriction.Restriction.BseJI.size
Bio.Restriction.Restriction.BseLI.size
Bio.Restriction.Restriction.BseMI.size
Bio.Restriction.Restriction.BseMII.size
Bio.Restriction.Restriction.BseNI.size
Bio.Restriction.Restriction.BsePI.size
Bio.Restriction.Restriction.BseRI.size
Bio.Restriction.Restriction.BseSI.size
Bio.Restriction.Restriction.BseX3I.size
Bio.Restriction.Restriction.BseXI.size
Bio.Restriction.Restriction.BseYI.size
Bio.Restriction.Restriction.BsgI.size
Bio.Restriction.Restriction.Bsh1236I.size
Bio.Restriction.Restriction.Bsh1285I.size
Bio.Restriction.Restriction.BshFI.size
Bio.Restriction.Restriction.BshNI.size
Bio.Restriction.Restriction.BshTI.size
Bio.Restriction.Restriction.BshVI.size
Bio.Restriction.Restriction.BsiEI.size
Bio.Restriction.Restriction.BsiHKAI.size
Bio.Restriction.Restriction.BsiHKCI.size
Bio.Restriction.Restriction.BsiI.size
Bio.Restriction.Restriction.BsiSI.size
Bio.Restriction.Restriction.BsiWI.size
Bio.Restriction.Restriction.BsiYI.size
Bio.Restriction.Restriction.BslFI.size
Bio.Restriction.Restriction.BslI.size
Bio.Restriction.Restriction.BsmAI.size
Bio.Restriction.Restriction.BsmBI.size
Bio.Restriction.Restriction.BsmFI.size
Bio.Restriction.Restriction.BsmI.size
Bio.Restriction.Restriction.BsnI.size
Bio.Restriction.Restriction.Bso31I.size
Bio.Restriction.Restriction.BsoBI.size
Bio.Restriction.Restriction.Bsp119I.size
Bio.Restriction.Restriction.Bsp120I.size
Bio.Restriction.Restriction.Bsp1286I.size
Bio.Restriction.Restriction.Bsp13I.size
Bio.Restriction.Restriction.Bsp1407I.size
Bio.Restriction.Restriction.Bsp143I.size
Bio.Restriction.Restriction.Bsp1720I.size
Bio.Restriction.Restriction.Bsp19I.size
Bio.Restriction.Restriction.Bsp24I.size
Bio.Restriction.Restriction.Bsp3004IV.size
Bio.Restriction.Restriction.Bsp460III.size
Bio.Restriction.Restriction.Bsp68I.size
Bio.Restriction.Restriction.BspACI.size
Bio.Restriction.Restriction.BspANI.size
Bio.Restriction.Restriction.BspCNI.size
Bio.Restriction.Restriction.BspD6I.size
Bio.Restriction.Restriction.BspDI.size
Bio.Restriction.Restriction.BspEI.size
Bio.Restriction.Restriction.BspFNI.size
Bio.Restriction.Restriction.BspGI.size
Bio.Restriction.Restriction.BspHI.size
Bio.Restriction.Restriction.BspLI.size
Bio.Restriction.Restriction.BspLU11I.size
Bio.Restriction.Restriction.BspMAI.size
Bio.Restriction.Restriction.BspMI.size
Bio.Restriction.Restriction.BspMII.size
Bio.Restriction.Restriction.BspNCI.size
Bio.Restriction.Restriction.BspOI.size
Bio.Restriction.Restriction.BspPI.size
Bio.Restriction.Restriction.BspQI.size
Bio.Restriction.Restriction.BspT104I.size
Bio.Restriction.Restriction.BspT107I.size
Bio.Restriction.Restriction.BspTI.size
Bio.Restriction.Restriction.BspTNI.size
Bio.Restriction.Restriction.BsrBI.size
Bio.Restriction.Restriction.BsrDI.size
Bio.Restriction.Restriction.BsrFI.size
Bio.Restriction.Restriction.BsrGI.size
Bio.Restriction.Restriction.BsrI.size
Bio.Restriction.Restriction.BssAI.size
Bio.Restriction.Restriction.BssECI.size
Bio.Restriction.Restriction.BssHII.size
Bio.Restriction.Restriction.BssMI.size
Bio.Restriction.Restriction.BssNAI.size
Bio.Restriction.Restriction.BssNI.size
Bio.Restriction.Restriction.BssSI.size
Bio.Restriction.Restriction.BssT1I.size
Bio.Restriction.Restriction.Bst1107I.size
Bio.Restriction.Restriction.Bst2BI.size
Bio.Restriction.Restriction.Bst2UI.size
Bio.Restriction.Restriction.Bst4CI.size
Bio.Restriction.Restriction.Bst6I.size
Bio.Restriction.Restriction.BstACI.size
Bio.Restriction.Restriction.BstAFI.size
Bio.Restriction.Restriction.BstAPI.size
Bio.Restriction.Restriction.BstAUI.size
Bio.Restriction.Restriction.BstBAI.size
Bio.Restriction.Restriction.BstBI.size
Bio.Restriction.Restriction.BstC8I.size
Bio.Restriction.Restriction.BstDEI.size
Bio.Restriction.Restriction.BstDSI.size
Bio.Restriction.Restriction.BstEII.size
Bio.Restriction.Restriction.BstENI.size
Bio.Restriction.Restriction.BstF5I.size
Bio.Restriction.Restriction.BstFNI.size
Bio.Restriction.Restriction.BstH2I.size
Bio.Restriction.Restriction.BstHHI.size
Bio.Restriction.Restriction.BstKTI.size
Bio.Restriction.Restriction.BstMAI.size
Bio.Restriction.Restriction.BstMBI.size
Bio.Restriction.Restriction.BstMCI.size
Bio.Restriction.Restriction.BstMWI.size
Bio.Restriction.Restriction.BstNI.size
Bio.Restriction.Restriction.BstNSI.size
Bio.Restriction.Restriction.BstPAI.size
Bio.Restriction.Restriction.BstPI.size
Bio.Restriction.Restriction.BstSCI.size
Bio.Restriction.Restriction.BstSFI.size
Bio.Restriction.Restriction.BstSLI.size
Bio.Restriction.Restriction.BstSNI.size
Bio.Restriction.Restriction.BstUI.size
Bio.Restriction.Restriction.BstV1I.size
Bio.Restriction.Restriction.BstV2I.size
Bio.Restriction.Restriction.BstX2I.size
Bio.Restriction.Restriction.BstXI.size
Bio.Restriction.Restriction.BstYI.size
Bio.Restriction.Restriction.BstZ17I.size
Bio.Restriction.Restriction.BstZI.size
Bio.Restriction.Restriction.Bsu15I.size
Bio.Restriction.Restriction.Bsu36I.size
Bio.Restriction.Restriction.BsuI.size
Bio.Restriction.Restriction.BsuRI.size
Bio.Restriction.Restriction.BsuTUI.size
Bio.Restriction.Restriction.BtgI.size
Bio.Restriction.Restriction.BtgZI.size
Bio.Restriction.Restriction.BthCI.size
Bio.Restriction.Restriction.BtrI.size
Bio.Restriction.Restriction.BtsCI.size
Bio.Restriction.Restriction.BtsI.size
Bio.Restriction.Restriction.BtsIMutI.size
Bio.Restriction.Restriction.BtuMI.size
Bio.Restriction.Restriction.BveI.size
Bio.Restriction.Restriction.Cac8I.size
Bio.Restriction.Restriction.CaiI.size
Bio.Restriction.Restriction.Cal14237I.size
Bio.Restriction.Restriction.CalB3II.size
Bio.Restriction.Restriction.Cau10061II.size
Bio.Restriction.Restriction.CauII.size
Bio.Restriction.Restriction.Cba13II.size
Bio.Restriction.Restriction.Cba16038I.size
Bio.Restriction.Restriction.Cbo67071IV.size
Bio.Restriction.Restriction.Cch467III.size
Bio.Restriction.Restriction.CchII.size
Bio.Restriction.Restriction.CchIII.size
Bio.Restriction.Restriction.CciI.size
Bio.Restriction.Restriction.CciNI.size
Bio.Restriction.Restriction.CcrNAIII.size
Bio.Restriction.Restriction.Cdi11397I.size
Bio.Restriction.Restriction.Cdi81III.size
Bio.Restriction.Restriction.CdiI.size
Bio.Restriction.Restriction.CdpI.size
Bio.Restriction.Restriction.Cdu23823II.size
Bio.Restriction.Restriction.CfoI.size
Bio.Restriction.Restriction.Cfr10I.size
Bio.Restriction.Restriction.Cfr13I.size
Bio.Restriction.Restriction.Cfr42I.size
Bio.Restriction.Restriction.Cfr9I.size
Bio.Restriction.Restriction.CfrI.size
Bio.Restriction.Restriction.Cfupf3II.size
Bio.Restriction.Restriction.Cgl13032I.size
Bio.Restriction.Restriction.Cgl13032II.size
Bio.Restriction.Restriction.ChaI.size
Bio.Restriction.Restriction.CjeFIII.size
Bio.Restriction.Restriction.CjeFV.size
Bio.Restriction.Restriction.CjeI.size
Bio.Restriction.Restriction.CjeNII.size
Bio.Restriction.Restriction.CjeNIII.size
Bio.Restriction.Restriction.CjeNV.size
Bio.Restriction.Restriction.CjeP659IV.size
Bio.Restriction.Restriction.CjePI.size
Bio.Restriction.Restriction.CjuI.size
Bio.Restriction.Restriction.CjuII.size
Bio.Restriction.Restriction.Cla11845III.size
Bio.Restriction.Restriction.ClaI.size
Bio.Restriction.Restriction.Cly7489II.size
Bio.Restriction.Restriction.Cma23826I.size
Bio.Restriction.Restriction.CpoI.size
Bio.Restriction.Restriction.CseI.size
Bio.Restriction.Restriction.CsiI.size
Bio.Restriction.Restriction.Csp2014I.size
Bio.Restriction.Restriction.Csp6I.size
Bio.Restriction.Restriction.CspAI.size
Bio.Restriction.Restriction.CspCI.size
Bio.Restriction.Restriction.CspI.size
Bio.Restriction.Restriction.CstMI.size
Bio.Restriction.Restriction.CviAII.size
Bio.Restriction.Restriction.CviJI.size
Bio.Restriction.Restriction.CviKI_1.size
Bio.Restriction.Restriction.CviQI.size
Bio.Restriction.Restriction.CviRI.size
Bio.Restriction.Restriction.Dde51507I.size
Bio.Restriction.Restriction.DdeI.size
Bio.Restriction.Restriction.DinI.size
Bio.Restriction.Restriction.DpnI.size
Bio.Restriction.Restriction.DpnII.size
Bio.Restriction.Restriction.DraI.size
Bio.Restriction.Restriction.DraII.size
Bio.Restriction.Restriction.DraIII.size
Bio.Restriction.Restriction.DraRI.size
Bio.Restriction.Restriction.DrdI.size
Bio.Restriction.Restriction.DrdII.size
Bio.Restriction.Restriction.DriI.size
Bio.Restriction.Restriction.DsaI.size
Bio.Restriction.Restriction.DseDI.size
Bio.Restriction.Restriction.EaeI.size
Bio.Restriction.Restriction.EagI.size
Bio.Restriction.Restriction.Eam1104I.size
Bio.Restriction.Restriction.Eam1105I.size
Bio.Restriction.Restriction.EarI.size
Bio.Restriction.Restriction.EciI.size
Bio.Restriction.Restriction.Ecl136II.size
Bio.Restriction.Restriction.Ecl35734I.size
Bio.Restriction.Restriction.EclXI.size
Bio.Restriction.Restriction.Eco105I.size
Bio.Restriction.Restriction.Eco130I.size
Bio.Restriction.Restriction.Eco147I.size
Bio.Restriction.Restriction.Eco24I.size
Bio.Restriction.Restriction.Eco31I.size
Bio.Restriction.Restriction.Eco32I.size
Bio.Restriction.Restriction.Eco4465II.size
Bio.Restriction.Restriction.Eco47I.size
Bio.Restriction.Restriction.Eco47III.size
Bio.Restriction.Restriction.Eco52I.size
Bio.Restriction.Restriction.Eco53kI.size
Bio.Restriction.Restriction.Eco57I.size
Bio.Restriction.Restriction.Eco57MI.size
Bio.Restriction.Restriction.Eco72I.size
Bio.Restriction.Restriction.Eco81I.size
Bio.Restriction.Restriction.Eco88I.size
Bio.Restriction.Restriction.Eco91I.size
Bio.Restriction.Restriction.EcoBLMcrX.size
Bio.Restriction.Restriction.EcoE1140I.size
Bio.Restriction.Restriction.EcoHI.size
Bio.Restriction.Restriction.EcoICRI.size
Bio.Restriction.Restriction.EcoMVII.size
Bio.Restriction.Restriction.EcoNI.size
Bio.Restriction.Restriction.EcoO109I.size
Bio.Restriction.Restriction.EcoO65I.size
Bio.Restriction.Restriction.EcoRI.size
Bio.Restriction.Restriction.EcoRII.size
Bio.Restriction.Restriction.EcoRV.size
Bio.Restriction.Restriction.EcoT14I.size
Bio.Restriction.Restriction.EcoT22I.size
Bio.Restriction.Restriction.EcoT38I.size
Bio.Restriction.Restriction.EgeI.size
Bio.Restriction.Restriction.EheI.size
Bio.Restriction.Restriction.Eli8509II.size
Bio.Restriction.Restriction.ErhI.size
Bio.Restriction.Restriction.EsaBC3I.size
Bio.Restriction.Restriction.EsaSSI.size
Bio.Restriction.Restriction.Esp3007I.size
Bio.Restriction.Restriction.Esp3I.size
Bio.Restriction.Restriction.EspI.size
Bio.Restriction.Restriction.Exi27195I.size
Bio.Restriction.Restriction.FaeI.size
Bio.Restriction.Restriction.FaiI.size
Bio.Restriction.Restriction.FalI.size
Bio.Restriction.Restriction.FaqI.size
Bio.Restriction.Restriction.FatI.size
Bio.Restriction.Restriction.FauI.size
Bio.Restriction.Restriction.FauNDI.size
Bio.Restriction.Restriction.FbaI.size
Bio.Restriction.Restriction.FblI.size
Bio.Restriction.Restriction.FinI.size
Bio.Restriction.Restriction.FmuI.size
Bio.Restriction.Restriction.Fnu4HI.size
Bio.Restriction.Restriction.FnuDII.size
Bio.Restriction.Restriction.FokI.size
Bio.Restriction.Restriction.FriOI.size
Bio.Restriction.Restriction.FseI.size
Bio.Restriction.Restriction.Fsp4HI.size
Bio.Restriction.Restriction.FspAI.size
Bio.Restriction.Restriction.FspBI.size
Bio.Restriction.Restriction.FspEI.size
Bio.Restriction.Restriction.FspI.size
Bio.Restriction.Restriction.FtnUV.size
Bio.Restriction.Restriction.GauT27I.size
Bio.Restriction.Restriction.Gba708II.size
Bio.Restriction.Restriction.GdiII.size
Bio.Restriction.Restriction.GlaI.size
Bio.Restriction.Restriction.GluI.size
Bio.Restriction.Restriction.GsaI.size
Bio.Restriction.Restriction.GsuI.size
Bio.Restriction.Restriction.HaeI.size
Bio.Restriction.Restriction.HaeII.size
Bio.Restriction.Restriction.HaeIII.size
Bio.Restriction.Restriction.HapII.size
Bio.Restriction.Restriction.HauII.size
Bio.Restriction.Restriction.HbaII.size
Bio.Restriction.Restriction.HgaI.size
Bio.Restriction.Restriction.HgiAI.size
Bio.Restriction.Restriction.HgiCI.size
Bio.Restriction.Restriction.HgiEII.size
Bio.Restriction.Restriction.HgiJII.size
Bio.Restriction.Restriction.HhaI.size
Bio.Restriction.Restriction.Hin1I.size
Bio.Restriction.Restriction.Hin1II.size
Bio.Restriction.Restriction.Hin4I.size
Bio.Restriction.Restriction.Hin4II.size
Bio.Restriction.Restriction.Hin6I.size
Bio.Restriction.Restriction.HinP1I.size
Bio.Restriction.Restriction.HincII.size
Bio.Restriction.Restriction.HindII.size
Bio.Restriction.Restriction.HindIII.size
Bio.Restriction.Restriction.HinfI.size
Bio.Restriction.Restriction.HpaI.size
Bio.Restriction.Restriction.HpaII.size
Bio.Restriction.Restriction.HphI.size
Bio.Restriction.Restriction.Hpy166II.size
Bio.Restriction.Restriction.Hpy178III.size
Bio.Restriction.Restriction.Hpy188I.size
Bio.Restriction.Restriction.Hpy188III.size
Bio.Restriction.Restriction.Hpy8I.size
Bio.Restriction.Restriction.Hpy99I.size
Bio.Restriction.Restriction.Hpy99XIII.size
Bio.Restriction.Restriction.Hpy99XIV.size
Bio.Restriction.Restriction.Hpy99XIV_mut1.size
Bio.Restriction.Restriction.Hpy99XXII.size
Bio.Restriction.Restriction.HpyAV.size
Bio.Restriction.Restriction.HpyAXIV.size
Bio.Restriction.Restriction.HpyAXVI_mut1.size
Bio.Restriction.Restriction.HpyAXVI_mut2.size
Bio.Restriction.Restriction.HpyCH4III.size
Bio.Restriction.Restriction.HpyCH4IV.size
Bio.Restriction.Restriction.HpyCH4V.size
Bio.Restriction.Restriction.HpyF10VI.size
Bio.Restriction.Restriction.HpyF3I.size
Bio.Restriction.Restriction.HpySE526I.size
Bio.Restriction.Restriction.HpyUM032XIII.size
Bio.Restriction.Restriction.HpyUM032XIII_mut1.size
Bio.Restriction.Restriction.HpyUM032XIV.size
Bio.Restriction.Restriction.HpyUM037X.size
Bio.Restriction.Restriction.Hsp92I.size
Bio.Restriction.Restriction.Hsp92II.size
Bio.Restriction.Restriction.HspAI.size
Bio.Restriction.Restriction.Jma19592I.size
Bio.Restriction.Restriction.Jma19592II.size
Bio.Restriction.Restriction.Jsp2502II.size
Bio.Restriction.Restriction.KasI.size
Bio.Restriction.Restriction.KflI.size
Bio.Restriction.Restriction.Kor51II.size
Bio.Restriction.Restriction.Kpn2I.size
Bio.Restriction.Restriction.KpnI.size
Bio.Restriction.Restriction.KpnNIH30III.size
Bio.Restriction.Restriction.KroI.size
Bio.Restriction.Restriction.Ksp22I.size
Bio.Restriction.Restriction.Ksp632I.size
Bio.Restriction.Restriction.KspAI.size
Bio.Restriction.Restriction.KspI.size
Bio.Restriction.Restriction.Kzo9I.size
Bio.Restriction.Restriction.Lba2029III.size
Bio.Restriction.Restriction.LguI.size
Bio.Restriction.Restriction.LmnI.size
Bio.Restriction.Restriction.Lmo370I.size
Bio.Restriction.Restriction.Lmo911II.size
Bio.Restriction.Restriction.LpnI.size
Bio.Restriction.Restriction.LpnPI.size
Bio.Restriction.Restriction.Lsp1109I.size
Bio.Restriction.Restriction.Lsp48III.size
Bio.Restriction.Restriction.Lsp6406VI.size
Bio.Restriction.Restriction.LweI.size
Bio.Restriction.Restriction.MabI.size
Bio.Restriction.Restriction.MaeI.size
Bio.Restriction.Restriction.MaeII.size
Bio.Restriction.Restriction.MaeIII.size
Bio.Restriction.Restriction.MalI.size
Bio.Restriction.Restriction.MaqI.size
Bio.Restriction.Restriction.MauBI.size
Bio.Restriction.Restriction.Mba11I.size
Bio.Restriction.Restriction.MbiI.size
Bio.Restriction.Restriction.MboI.size
Bio.Restriction.Restriction.MboII.size
Bio.Restriction.Restriction.McaTI.size
Bio.Restriction.Restriction.Mcr10I.size
Bio.Restriction.Restriction.McrI.size
Bio.Restriction.Restriction.MfeI.size
Bio.Restriction.Restriction.MflI.size
Bio.Restriction.Restriction.MhlI.size
Bio.Restriction.Restriction.MjaIV.size
Bio.Restriction.Restriction.MkaDII.size
Bio.Restriction.Restriction.MlsI.size
Bio.Restriction.Restriction.MluCI.size
Bio.Restriction.Restriction.MluI.size
Bio.Restriction.Restriction.MluNI.size
Bio.Restriction.Restriction.Mly113I.size
Bio.Restriction.Restriction.MlyI.size
Bio.Restriction.Restriction.MmeI.size
Bio.Restriction.Restriction.MnlI.size
Bio.Restriction.Restriction.Mox20I.size
Bio.Restriction.Restriction.Mph1103I.size
Bio.Restriction.Restriction.MreI.size
Bio.Restriction.Restriction.MroI.size
Bio.Restriction.Restriction.MroNI.size
Bio.Restriction.Restriction.MroXI.size
Bio.Restriction.Restriction.MscI.size
Bio.Restriction.Restriction.MseI.size
Bio.Restriction.Restriction.MslI.size
Bio.Restriction.Restriction.Msp20I.size
Bio.Restriction.Restriction.MspA1I.size
Bio.Restriction.Restriction.MspCI.size
Bio.Restriction.Restriction.MspI.size
Bio.Restriction.Restriction.MspI7II.size
Bio.Restriction.Restriction.MspJI.size
Bio.Restriction.Restriction.MspR9I.size
Bio.Restriction.Restriction.MssI.size
Bio.Restriction.Restriction.MstI.size
Bio.Restriction.Restriction.MteI.size
Bio.Restriction.Restriction.MtuHN878II.size
Bio.Restriction.Restriction.MunI.size
Bio.Restriction.Restriction.Mva1269I.size
Bio.Restriction.Restriction.MvaI.size
Bio.Restriction.Restriction.MvnI.size
Bio.Restriction.Restriction.MwoI.size
Bio.Restriction.Restriction.NaeI.size
Bio.Restriction.Restriction.Nal45188II.size
Bio.Restriction.Restriction.NarI.size
Bio.Restriction.Restriction.Nbr128II.size
Bio.Restriction.Restriction.NciI.size
Bio.Restriction.Restriction.NcoI.size
Bio.Restriction.Restriction.NdeI.size
Bio.Restriction.Restriction.NdeII.size
Bio.Restriction.Restriction.NgoAVII.size
Bio.Restriction.Restriction.NgoAVIII.size
Bio.Restriction.Restriction.NgoMIV.size
Bio.Restriction.Restriction.NhaXI.size
Bio.Restriction.Restriction.NheI.size
Bio.Restriction.Restriction.NhoI.size
Bio.Restriction.Restriction.NlaCI.size
Bio.Restriction.Restriction.NlaIII.size
Bio.Restriction.Restriction.NlaIV.size
Bio.Restriction.Restriction.Nli3877I.size
Bio.Restriction.Restriction.NmeAIII.size
Bio.Restriction.Restriction.NmeDI.size
Bio.Restriction.Restriction.NmuCI.size
Bio.Restriction.Restriction.NotI.size
Bio.Restriction.Restriction.NpeUS61II.size
Bio.Restriction.Restriction.NruI.size
Bio.Restriction.Restriction.NsbI.size
Bio.Restriction.Restriction.NsiI.size
Bio.Restriction.Restriction.NspBII.size
Bio.Restriction.Restriction.NspI.size
Bio.Restriction.Restriction.NspV.size
Bio.Restriction.Restriction.OliI.size
Bio.Restriction.Restriction.OspHL35III.size
Bio.Restriction.Restriction.PabI.size
Bio.Restriction.Restriction.Pac19842II.size
Bio.Restriction.Restriction.PacI.size
Bio.Restriction.Restriction.PacIII.size
Bio.Restriction.Restriction.PaeI.size
Bio.Restriction.Restriction.PaeR7I.size
Bio.Restriction.Restriction.PagI.size
Bio.Restriction.Restriction.Pal408I.size
Bio.Restriction.Restriction.PalAI.size
Bio.Restriction.Restriction.PasI.size
Bio.Restriction.Restriction.PauI.size
Bio.Restriction.Restriction.PcaII.size
Bio.Restriction.Restriction.PceI.size
Bio.Restriction.Restriction.PciI.size
Bio.Restriction.Restriction.PciSI.size
Bio.Restriction.Restriction.PcsI.size
Bio.Restriction.Restriction.PctI.size
Bio.Restriction.Restriction.Pdi8503III.size
Bio.Restriction.Restriction.PdiI.size
Bio.Restriction.Restriction.PdmI.size
Bio.Restriction.Restriction.Pdu1735I.size
Bio.Restriction.Restriction.PenI.size
Bio.Restriction.Restriction.PfeI.size
Bio.Restriction.Restriction.Pfl1108I.size
Bio.Restriction.Restriction.Pfl23II.size
Bio.Restriction.Restriction.Pfl8569I.size
Bio.Restriction.Restriction.PflFI.size
Bio.Restriction.Restriction.PflMI.size
Bio.Restriction.Restriction.PfoI.size
Bio.Restriction.Restriction.PfrJS12IV.size
Bio.Restriction.Restriction.PfrJS12V.size
Bio.Restriction.Restriction.PfrJS15III.size
Bio.Restriction.Restriction.Pin17FIII.size
Bio.Restriction.Restriction.PinAI.size
Bio.Restriction.Restriction.PkrI.size
Bio.Restriction.Restriction.PlaDI.size
Bio.Restriction.Restriction.Ple19I.size
Bio.Restriction.Restriction.PleI.size
Bio.Restriction.Restriction.PliMI.size
Bio.Restriction.Restriction.PluTI.size
Bio.Restriction.Restriction.PmaCI.size
Bio.Restriction.Restriction.PmeI.size
Bio.Restriction.Restriction.PmlI.size
Bio.Restriction.Restriction.PpiI.size
Bio.Restriction.Restriction.PpsI.size
Bio.Restriction.Restriction.Ppu10I.size
Bio.Restriction.Restriction.Ppu21I.size
Bio.Restriction.Restriction.PpuMI.size
Bio.Restriction.Restriction.PscI.size
Bio.Restriction.Restriction.Pse18267I.size
Bio.Restriction.Restriction.PshAI.size
Bio.Restriction.Restriction.PshBI.size
Bio.Restriction.Restriction.PsiI.size
Bio.Restriction.Restriction.Psp0357II.size
Bio.Restriction.Restriction.Psp03I.size
Bio.Restriction.Restriction.Psp124BI.size
Bio.Restriction.Restriction.Psp1406I.size
Bio.Restriction.Restriction.Psp5II.size
Bio.Restriction.Restriction.Psp6I.size
Bio.Restriction.Restriction.PspCI.size
Bio.Restriction.Restriction.PspEI.size
Bio.Restriction.Restriction.PspFI.size
Bio.Restriction.Restriction.PspGI.size
Bio.Restriction.Restriction.PspLI.size
Bio.Restriction.Restriction.PspN4I.size
Bio.Restriction.Restriction.PspOMI.size
Bio.Restriction.Restriction.PspOMII.size
Bio.Restriction.Restriction.PspPI.size
Bio.Restriction.Restriction.PspPPI.size
Bio.Restriction.Restriction.PspPRI.size
Bio.Restriction.Restriction.PspXI.size
Bio.Restriction.Restriction.PsrI.size
Bio.Restriction.Restriction.PssI.size
Bio.Restriction.Restriction.Pst14472I.size
Bio.Restriction.Restriction.Pst273I.size
Bio.Restriction.Restriction.PstI.size
Bio.Restriction.Restriction.PstNI.size
Bio.Restriction.Restriction.PsuGI.size
Bio.Restriction.Restriction.PsuI.size
Bio.Restriction.Restriction.PsyI.size
Bio.Restriction.Restriction.PteI.size
Bio.Restriction.Restriction.PvuI.size
Bio.Restriction.Restriction.PvuII.size
Bio.Restriction.Restriction.Rba2021I.size
Bio.Restriction.Restriction.RceI.size
Bio.Restriction.Restriction.RdeGBI.size
Bio.Restriction.Restriction.RdeGBII.size
Bio.Restriction.Restriction.RdeGBIII.size
Bio.Restriction.Restriction.RflFIII.size
Bio.Restriction.Restriction.RgaI.size
Bio.Restriction.Restriction.RigI.size
Bio.Restriction.Restriction.RlaI.size
Bio.Restriction.Restriction.RlaII.size
Bio.Restriction.Restriction.RleAI.size
Bio.Restriction.Restriction.RpaB5I.size
Bio.Restriction.Restriction.RpaBI.size
Bio.Restriction.Restriction.RpaI.size
Bio.Restriction.Restriction.RpaTI.size
Bio.Restriction.Restriction.RruI.size
Bio.Restriction.Restriction.RsaI.size
Bio.Restriction.Restriction.RsaNI.size
Bio.Restriction.Restriction.RseI.size
Bio.Restriction.Restriction.Rsp008IV.size
Bio.Restriction.Restriction.Rsp008V.size
Bio.Restriction.Restriction.RspPBTS2III.size
Bio.Restriction.Restriction.Rsr2I.size
Bio.Restriction.Restriction.RsrII.size
Bio.Restriction.Restriction.SacI.size
Bio.Restriction.Restriction.SacII.size
Bio.Restriction.Restriction.Saf8902III.size
Bio.Restriction.Restriction.SalI.size
Bio.Restriction.Restriction.SanDI.size
Bio.Restriction.Restriction.SapI.size
Bio.Restriction.Restriction.SaqAI.size
Bio.Restriction.Restriction.SatI.size
Bio.Restriction.Restriction.Sau3AI.size
Bio.Restriction.Restriction.Sau96I.size
Bio.Restriction.Restriction.SauI.size
Bio.Restriction.Restriction.SbfI.size
Bio.Restriction.Restriction.Sbo46I.size
Bio.Restriction.Restriction.ScaI.size
Bio.Restriction.Restriction.SchI.size
Bio.Restriction.Restriction.SciI.size
Bio.Restriction.Restriction.ScoDS2II.size
Bio.Restriction.Restriction.ScrFI.size
Bio.Restriction.Restriction.SdaI.size
Bio.Restriction.Restriction.SdeAI.size
Bio.Restriction.Restriction.SdeOSI.size
Bio.Restriction.Restriction.SduI.size
Bio.Restriction.Restriction.SecI.size
Bio.Restriction.Restriction.SelI.size
Bio.Restriction.Restriction.Sen13311III.size
Bio.Restriction.Restriction.Sen13III.size
Bio.Restriction.Restriction.Sen1736II.size
Bio.Restriction.Restriction.Sen1878IV.size
Bio.Restriction.Restriction.Sen1880IV.size
Bio.Restriction.Restriction.Sen1896IV.size
Bio.Restriction.Restriction.Sen1898IV.size
Bio.Restriction.Restriction.Sen1906IV.size
Bio.Restriction.Restriction.Sen1908IV.size
Bio.Restriction.Restriction.Sen1910IV.size
Bio.Restriction.Restriction.SenC1808III.size
Bio.Restriction.Restriction.SenC1810IV.size
Bio.Restriction.Restriction.SenSARA26III.size
Bio.Restriction.Restriction.SenTFIV.size
Bio.Restriction.Restriction.SetI.size
Bio.Restriction.Restriction.SexAI.size
Bio.Restriction.Restriction.SfaAI.size
Bio.Restriction.Restriction.SfaNI.size
Bio.Restriction.Restriction.SfcI.size
Bio.Restriction.Restriction.SfeI.size
Bio.Restriction.Restriction.SfiI.size
Bio.Restriction.Restriction.SfoI.size
Bio.Restriction.Restriction.Sfr274I.size
Bio.Restriction.Restriction.Sfr303I.size
Bio.Restriction.Restriction.SfuI.size
Bio.Restriction.Restriction.SgeI.size
Bio.Restriction.Restriction.SgfI.size
Bio.Restriction.Restriction.SgrAI.size
Bio.Restriction.Restriction.SgrBI.size
Bio.Restriction.Restriction.SgrDI.size
Bio.Restriction.Restriction.SgrTI.size
Bio.Restriction.Restriction.SgsI.size
Bio.Restriction.Restriction.SimI.size
Bio.Restriction.Restriction.SinI.size
Bio.Restriction.Restriction.SlaI.size
Bio.Restriction.Restriction.SmaI.size
Bio.Restriction.Restriction.SmiI.size
Bio.Restriction.Restriction.SmiMI.size
Bio.Restriction.Restriction.SmlI.size
Bio.Restriction.Restriction.SmoI.size
Bio.Restriction.Restriction.SnaBI.size
Bio.Restriction.Restriction.SnaI.size
Bio.Restriction.Restriction.Sno506I.size
Bio.Restriction.Restriction.SpeI.size
Bio.Restriction.Restriction.SphI.size
Bio.Restriction.Restriction.SplI.size
Bio.Restriction.Restriction.SpnRII.size
Bio.Restriction.Restriction.SpoDI.size
Bio.Restriction.Restriction.SrfI.size
Bio.Restriction.Restriction.Sse232I.size
Bio.Restriction.Restriction.Sse8387I.size
Bio.Restriction.Restriction.Sse8647I.size
Bio.Restriction.Restriction.Sse9I.size
Bio.Restriction.Restriction.SseBI.size
Bio.Restriction.Restriction.SsiI.size
Bio.Restriction.Restriction.Ssp6803IV.size
Bio.Restriction.Restriction.Ssp714II.size
Bio.Restriction.Restriction.SspD5I.size
Bio.Restriction.Restriction.SspDI.size
Bio.Restriction.Restriction.SspI.size
Bio.Restriction.Restriction.SspMI.size
Bio.Restriction.Restriction.SstE37I.size
Bio.Restriction.Restriction.SstI.size
Bio.Restriction.Restriction.Sth132I.size
Bio.Restriction.Restriction.Sth20745III.size
Bio.Restriction.Restriction.Sth302II.size
Bio.Restriction.Restriction.StsI.size
Bio.Restriction.Restriction.StuI.size
Bio.Restriction.Restriction.Sty13348III.size
Bio.Restriction.Restriction.StyD4I.size
Bio.Restriction.Restriction.StyI.size
Bio.Restriction.Restriction.SwaI.size
Bio.Restriction.Restriction.TaaI.size
Bio.Restriction.Restriction.TaiI.size
Bio.Restriction.Restriction.TaqI.size
Bio.Restriction.Restriction.TaqII.size
Bio.Restriction.Restriction.TaqIII.size
Bio.Restriction.Restriction.TasI.size
Bio.Restriction.Restriction.TatI.size
Bio.Restriction.Restriction.TauI.size
Bio.Restriction.Restriction.TfiI.size
Bio.Restriction.Restriction.Tru1I.size
Bio.Restriction.Restriction.Tru9I.size
Bio.Restriction.Restriction.TscAI.size
Bio.Restriction.Restriction.TseFI.size
Bio.Restriction.Restriction.TseI.size
Bio.Restriction.Restriction.TsoI.size
Bio.Restriction.Restriction.Tsp45I.size
Bio.Restriction.Restriction.Tsp4CI.size
Bio.Restriction.Restriction.TspARh3I.size
Bio.Restriction.Restriction.TspDTI.size
Bio.Restriction.Restriction.TspEI.size
Bio.Restriction.Restriction.TspGWI.size
Bio.Restriction.Restriction.TspMI.size
Bio.Restriction.Restriction.TspRI.size
Bio.Restriction.Restriction.TssI.size
Bio.Restriction.Restriction.TstI.size
Bio.Restriction.Restriction.TsuI.size
Bio.Restriction.Restriction.Tth111I.size
Bio.Restriction.Restriction.Tth111II.size
Bio.Restriction.Restriction.UbaF11I.size
Bio.Restriction.Restriction.UbaF12I.size
Bio.Restriction.Restriction.UbaF13I.size
Bio.Restriction.Restriction.UbaF14I.size
Bio.Restriction.Restriction.UbaF9I.size
Bio.Restriction.Restriction.UbaPI.size
Bio.Restriction.Restriction.UcoMSI.size
Bio.Restriction.Restriction.UnbI.size
Bio.Restriction.Restriction.Van91I.size
Bio.Restriction.Restriction.Vha464I.size
Bio.Restriction.Restriction.VneI.size
Bio.Restriction.Restriction.VpaK11AI.size
Bio.Restriction.Restriction.VpaK11BI.size
Bio.Restriction.Restriction.VspI.size
Bio.Restriction.Restriction.Vtu19109I.size
Bio.Restriction.Restriction.WviI.size
Bio.Restriction.Restriction.XagI.size
Bio.Restriction.Restriction.XapI.size
Bio.Restriction.Restriction.XbaI.size
Bio.Restriction.Restriction.Xca85IV.size
Bio.Restriction.Restriction.XceI.size
Bio.Restriction.Restriction.XcmI.size
Bio.Restriction.Restriction.XhoI.size
Bio.Restriction.Restriction.XhoII.size
Bio.Restriction.Restriction.XmaI.size
Bio.Restriction.Restriction.XmaIII.size
Bio.Restriction.Restriction.XmaJI.size
Bio.Restriction.Restriction.XmiI.size
Bio.Restriction.Restriction.XmnI.size
Bio.Restriction.Restriction.XspI.size
Bio.Restriction.Restriction.YkrI.size
Bio.Restriction.Restriction.Yps3606I.size
Bio.Restriction.Restriction.ZraI.size
Bio.Restriction.Restriction.ZrmI.size
Bio.Restriction.Restriction.Zsp2I.size
Bio.codonalign.codonalphabet.CodonAlphabet.size" class="py-name" href="#" onclick="return doclink('link-133', 'size', 'link-132');">size</a></tt> <tt class="py-op">!=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"-"</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L412"></a><tt class="py-lineno">412</tt>  <tt class="py-line">                                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt> </tt>
<a name="L413"></a><tt class="py-lineno">413</tt>  <tt class="py-line">                                    <tt class="py-string">"Invalid length for target coordinates "</tt> </tt>
<a name="L414"></a><tt class="py-lineno">414</tt>  <tt class="py-line">                                    <tt class="py-string">"(expected %s, found %s)"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt> </tt>
<a name="L415"></a><tt class="py-lineno">415</tt>  <tt class="py-line">                                        <tt id="link-134" class="py-name"><a title="Bio.Alphabet.Alphabet.size
Bio.Alphabet.Reduced.HPModel.size
Bio.Alphabet.Reduced.Murphy10.size
Bio.Alphabet.Reduced.Murphy15.size
Bio.Alphabet.Reduced.Murphy4.size
Bio.Alphabet.Reduced.Murphy8.size
Bio.Alphabet.Reduced.PC5.size
Bio.Alphabet.SingleLetterAlphabet.size
Bio.Alphabet.ThreeLetterProtein.size
Bio.GenBank._FeatureConsumer.size
Bio.GenBank._RecordConsumer.size
Bio.Restriction.Restriction.AanI.size
Bio.Restriction.Restriction.AarI.size
Bio.Restriction.Restriction.AasI.size
Bio.Restriction.Restriction.AatII.size
Bio.Restriction.Restriction.Aba6411II.size
Bio.Restriction.Restriction.AbaCIII.size
Bio.Restriction.Restriction.AbaSI.size
Bio.Restriction.Restriction.AbaUMB2I.size
Bio.Restriction.Restriction.AbsI.size
Bio.Restriction.Restriction.Acc16I.size
Bio.Restriction.Restriction.Acc36I.size
Bio.Restriction.Restriction.Acc65I.size
Bio.Restriction.Restriction.Acc65V.size
Bio.Restriction.Restriction.AccB1I.size
Bio.Restriction.Restriction.AccB7I.size
Bio.Restriction.Restriction.AccBSI.size
Bio.Restriction.Restriction.AccI.size
Bio.Restriction.Restriction.AccII.size
Bio.Restriction.Restriction.AccIII.size
Bio.Restriction.Restriction.AceIII.size
Bio.Restriction.Restriction.AchA6III.size
Bio.Restriction.Restriction.AciI.size
Bio.Restriction.Restriction.AclI.size
Bio.Restriction.Restriction.AclWI.size
Bio.Restriction.Restriction.Aco12261II.size
Bio.Restriction.Restriction.AcoI.size
Bio.Restriction.Restriction.AcsI.size
Bio.Restriction.Restriction.AcuI.size
Bio.Restriction.Restriction.AcvI.size
Bio.Restriction.Restriction.AcyI.size
Bio.Restriction.Restriction.AdeI.size
Bio.Restriction.Restriction.Adh6U21I.size
Bio.Restriction.Restriction.AfaI.size
Bio.Restriction.Restriction.AfeI.size
Bio.Restriction.Restriction.AfiI.size
Bio.Restriction.Restriction.AflII.size
Bio.Restriction.Restriction.AflIII.size
Bio.Restriction.Restriction.AgeI.size
Bio.Restriction.Restriction.AgsI.size
Bio.Restriction.Restriction.AhaIII.size
Bio.Restriction.Restriction.AhdI.size
Bio.Restriction.Restriction.AhlI.size
Bio.Restriction.Restriction.AhyRBAHI.size
Bio.Restriction.Restriction.AhyYL17I.size
Bio.Restriction.Restriction.AjiI.size
Bio.Restriction.Restriction.AjnI.size
Bio.Restriction.Restriction.AjuI.size
Bio.Restriction.Restriction.AleI.size
Bio.Restriction.Restriction.AlfI.size
Bio.Restriction.Restriction.AloI.size
Bio.Restriction.Restriction.AluBI.size
Bio.Restriction.Restriction.AluI.size
Bio.Restriction.Restriction.Alw21I.size
Bio.Restriction.Restriction.Alw26I.size
Bio.Restriction.Restriction.Alw44I.size
Bio.Restriction.Restriction.AlwFI.size
Bio.Restriction.Restriction.AlwI.size
Bio.Restriction.Restriction.AlwNI.size
Bio.Restriction.Restriction.Ama87I.size
Bio.Restriction.Restriction.AmaCSI.size
Bio.Restriction.Restriction.Aor13HI.size
Bio.Restriction.Restriction.Aor51HI.size
Bio.Restriction.Restriction.AoxI.size
Bio.Restriction.Restriction.ApaBI.size
Bio.Restriction.Restriction.ApaI.size
Bio.Restriction.Restriction.ApaLI.size
Bio.Restriction.Restriction.ApeKI.size
Bio.Restriction.Restriction.ApoI.size
Bio.Restriction.Restriction.ApyPI.size
Bio.Restriction.Restriction.AquII.size
Bio.Restriction.Restriction.AquIII.size
Bio.Restriction.Restriction.AquIV.size
Bio.Restriction.Restriction.ArsI.size
Bio.Restriction.Restriction.AscI.size
Bio.Restriction.Restriction.AseI.size
Bio.Restriction.Restriction.Asi256I.size
Bio.Restriction.Restriction.AsiGI.size
Bio.Restriction.Restriction.AsiSI.size
Bio.Restriction.Restriction.Asp114II.size
Bio.Restriction.Restriction.Asp700I.size
Bio.Restriction.Restriction.Asp718I.size
Bio.Restriction.Restriction.AspA2I.size
Bio.Restriction.Restriction.AspBHI.size
Bio.Restriction.Restriction.AspDUT2V.size
Bio.Restriction.Restriction.AspJHL3II.size
Bio.Restriction.Restriction.AspLEI.size
Bio.Restriction.Restriction.AspS9I.size
Bio.Restriction.Restriction.Asu14238IV.size
Bio.Restriction.Restriction.AsuC2I.size
Bio.Restriction.Restriction.AsuHPI.size
Bio.Restriction.Restriction.AsuI.size
Bio.Restriction.Restriction.AsuII.size
Bio.Restriction.Restriction.AsuNHI.size
Bio.Restriction.Restriction.AteTI.size
Bio.Restriction.Restriction.AvaI.size
Bio.Restriction.Restriction.AvaII.size
Bio.Restriction.Restriction.AvaIII.size
Bio.Restriction.Restriction.AvrII.size
Bio.Restriction.Restriction.Awo1030IV.size
Bio.Restriction.Restriction.AxyI.size
Bio.Restriction.Restriction.BaeGI.size
Bio.Restriction.Restriction.BaeI.size
Bio.Restriction.Restriction.Bag18758I.size
Bio.Restriction.Restriction.BalI.size
Bio.Restriction.Restriction.BamHI.size
Bio.Restriction.Restriction.BanI.size
Bio.Restriction.Restriction.BanII.size
Bio.Restriction.Restriction.BanLI.size
Bio.Restriction.Restriction.BarI.size
Bio.Restriction.Restriction.BauI.size
Bio.Restriction.Restriction.BbrPI.size
Bio.Restriction.Restriction.BbsI.size
Bio.Restriction.Restriction.Bbv12I.size
Bio.Restriction.Restriction.BbvCI.size
Bio.Restriction.Restriction.BbvI.size
Bio.Restriction.Restriction.BbvII.size
Bio.Restriction.Restriction.BccI.size
Bio.Restriction.Restriction.Bce3081I.size
Bio.Restriction.Restriction.Bce83I.size
Bio.Restriction.Restriction.BceAI.size
Bio.Restriction.Restriction.BceSIV.size
Bio.Restriction.Restriction.BcefI.size
Bio.Restriction.Restriction.BcgI.size
Bio.Restriction.Restriction.BciT130I.size
Bio.Restriction.Restriction.BciVI.size
Bio.Restriction.Restriction.BclI.size
Bio.Restriction.Restriction.BcnI.size
Bio.Restriction.Restriction.BcoDI.size
Bio.Restriction.Restriction.BcuI.size
Bio.Restriction.Restriction.BdaI.size
Bio.Restriction.Restriction.BetI.size
Bio.Restriction.Restriction.BfaI.size
Bio.Restriction.Restriction.BfaSII.size
Bio.Restriction.Restriction.BfiI.size
Bio.Restriction.Restriction.BfmI.size
Bio.Restriction.Restriction.BfoI.size
Bio.Restriction.Restriction.BfrI.size
Bio.Restriction.Restriction.BfuAI.size
Bio.Restriction.Restriction.BfuI.size
Bio.Restriction.Restriction.BglI.size
Bio.Restriction.Restriction.BglII.size
Bio.Restriction.Restriction.BinI.size
Bio.Restriction.Restriction.BisI.size
Bio.Restriction.Restriction.Ble402II.size
Bio.Restriction.Restriction.BlnI.size
Bio.Restriction.Restriction.BloAII.size
Bio.Restriction.Restriction.BlpI.size
Bio.Restriction.Restriction.BlsI.size
Bio.Restriction.Restriction.BmcAI.size
Bio.Restriction.Restriction.Bme1390I.size
Bio.Restriction.Restriction.Bme18I.size
Bio.Restriction.Restriction.BmeDI.size
Bio.Restriction.Restriction.BmeRI.size
Bio.Restriction.Restriction.BmeT110I.size
Bio.Restriction.Restriction.BmgBI.size
Bio.Restriction.Restriction.BmgI.size
Bio.Restriction.Restriction.BmgT120I.size
Bio.Restriction.Restriction.BmiI.size
Bio.Restriction.Restriction.BmrFI.size
Bio.Restriction.Restriction.BmrI.size
Bio.Restriction.Restriction.BmsI.size
Bio.Restriction.Restriction.BmtI.size
Bio.Restriction.Restriction.BmuI.size
Bio.Restriction.Restriction.BoxI.size
Bio.Restriction.Restriction.BpiI.size
Bio.Restriction.Restriction.BplI.size
Bio.Restriction.Restriction.BpmI.size
Bio.Restriction.Restriction.Bpu10I.size
Bio.Restriction.Restriction.Bpu1102I.size
Bio.Restriction.Restriction.Bpu14I.size
Bio.Restriction.Restriction.BpuEI.size
Bio.Restriction.Restriction.BpuMI.size
Bio.Restriction.Restriction.Bsa29I.size
Bio.Restriction.Restriction.BsaAI.size
Bio.Restriction.Restriction.BsaBI.size
Bio.Restriction.Restriction.BsaHI.size
Bio.Restriction.Restriction.BsaI.size
Bio.Restriction.Restriction.BsaJI.size
Bio.Restriction.Restriction.BsaWI.size
Bio.Restriction.Restriction.BsaXI.size
Bio.Restriction.Restriction.BsbI.size
Bio.Restriction.Restriction.Bsc4I.size
Bio.Restriction.Restriction.BscAI.size
Bio.Restriction.Restriction.BscGI.size
Bio.Restriction.Restriction.Bse118I.size
Bio.Restriction.Restriction.Bse1I.size
Bio.Restriction.Restriction.Bse21I.size
Bio.Restriction.Restriction.Bse3DI.size
Bio.Restriction.Restriction.Bse8I.size
Bio.Restriction.Restriction.BseAI.size
Bio.Restriction.Restriction.BseBI.size
Bio.Restriction.Restriction.BseCI.size
Bio.Restriction.Restriction.BseDI.size
Bio.Restriction.Restriction.BseGI.size
Bio.Restriction.Restriction.BseJI.size
Bio.Restriction.Restriction.BseLI.size
Bio.Restriction.Restriction.BseMI.size
Bio.Restriction.Restriction.BseMII.size
Bio.Restriction.Restriction.BseNI.size
Bio.Restriction.Restriction.BsePI.size
Bio.Restriction.Restriction.BseRI.size
Bio.Restriction.Restriction.BseSI.size
Bio.Restriction.Restriction.BseX3I.size
Bio.Restriction.Restriction.BseXI.size
Bio.Restriction.Restriction.BseYI.size
Bio.Restriction.Restriction.BsgI.size
Bio.Restriction.Restriction.Bsh1236I.size
Bio.Restriction.Restriction.Bsh1285I.size
Bio.Restriction.Restriction.BshFI.size
Bio.Restriction.Restriction.BshNI.size
Bio.Restriction.Restriction.BshTI.size
Bio.Restriction.Restriction.BshVI.size
Bio.Restriction.Restriction.BsiEI.size
Bio.Restriction.Restriction.BsiHKAI.size
Bio.Restriction.Restriction.BsiHKCI.size
Bio.Restriction.Restriction.BsiI.size
Bio.Restriction.Restriction.BsiSI.size
Bio.Restriction.Restriction.BsiWI.size
Bio.Restriction.Restriction.BsiYI.size
Bio.Restriction.Restriction.BslFI.size
Bio.Restriction.Restriction.BslI.size
Bio.Restriction.Restriction.BsmAI.size
Bio.Restriction.Restriction.BsmBI.size
Bio.Restriction.Restriction.BsmFI.size
Bio.Restriction.Restriction.BsmI.size
Bio.Restriction.Restriction.BsnI.size
Bio.Restriction.Restriction.Bso31I.size
Bio.Restriction.Restriction.BsoBI.size
Bio.Restriction.Restriction.Bsp119I.size
Bio.Restriction.Restriction.Bsp120I.size
Bio.Restriction.Restriction.Bsp1286I.size
Bio.Restriction.Restriction.Bsp13I.size
Bio.Restriction.Restriction.Bsp1407I.size
Bio.Restriction.Restriction.Bsp143I.size
Bio.Restriction.Restriction.Bsp1720I.size
Bio.Restriction.Restriction.Bsp19I.size
Bio.Restriction.Restriction.Bsp24I.size
Bio.Restriction.Restriction.Bsp3004IV.size
Bio.Restriction.Restriction.Bsp460III.size
Bio.Restriction.Restriction.Bsp68I.size
Bio.Restriction.Restriction.BspACI.size
Bio.Restriction.Restriction.BspANI.size
Bio.Restriction.Restriction.BspCNI.size
Bio.Restriction.Restriction.BspD6I.size
Bio.Restriction.Restriction.BspDI.size
Bio.Restriction.Restriction.BspEI.size
Bio.Restriction.Restriction.BspFNI.size
Bio.Restriction.Restriction.BspGI.size
Bio.Restriction.Restriction.BspHI.size
Bio.Restriction.Restriction.BspLI.size
Bio.Restriction.Restriction.BspLU11I.size
Bio.Restriction.Restriction.BspMAI.size
Bio.Restriction.Restriction.BspMI.size
Bio.Restriction.Restriction.BspMII.size
Bio.Restriction.Restriction.BspNCI.size
Bio.Restriction.Restriction.BspOI.size
Bio.Restriction.Restriction.BspPI.size
Bio.Restriction.Restriction.BspQI.size
Bio.Restriction.Restriction.BspT104I.size
Bio.Restriction.Restriction.BspT107I.size
Bio.Restriction.Restriction.BspTI.size
Bio.Restriction.Restriction.BspTNI.size
Bio.Restriction.Restriction.BsrBI.size
Bio.Restriction.Restriction.BsrDI.size
Bio.Restriction.Restriction.BsrFI.size
Bio.Restriction.Restriction.BsrGI.size
Bio.Restriction.Restriction.BsrI.size
Bio.Restriction.Restriction.BssAI.size
Bio.Restriction.Restriction.BssECI.size
Bio.Restriction.Restriction.BssHII.size
Bio.Restriction.Restriction.BssMI.size
Bio.Restriction.Restriction.BssNAI.size
Bio.Restriction.Restriction.BssNI.size
Bio.Restriction.Restriction.BssSI.size
Bio.Restriction.Restriction.BssT1I.size
Bio.Restriction.Restriction.Bst1107I.size
Bio.Restriction.Restriction.Bst2BI.size
Bio.Restriction.Restriction.Bst2UI.size
Bio.Restriction.Restriction.Bst4CI.size
Bio.Restriction.Restriction.Bst6I.size
Bio.Restriction.Restriction.BstACI.size
Bio.Restriction.Restriction.BstAFI.size
Bio.Restriction.Restriction.BstAPI.size
Bio.Restriction.Restriction.BstAUI.size
Bio.Restriction.Restriction.BstBAI.size
Bio.Restriction.Restriction.BstBI.size
Bio.Restriction.Restriction.BstC8I.size
Bio.Restriction.Restriction.BstDEI.size
Bio.Restriction.Restriction.BstDSI.size
Bio.Restriction.Restriction.BstEII.size
Bio.Restriction.Restriction.BstENI.size
Bio.Restriction.Restriction.BstF5I.size
Bio.Restriction.Restriction.BstFNI.size
Bio.Restriction.Restriction.BstH2I.size
Bio.Restriction.Restriction.BstHHI.size
Bio.Restriction.Restriction.BstKTI.size
Bio.Restriction.Restriction.BstMAI.size
Bio.Restriction.Restriction.BstMBI.size
Bio.Restriction.Restriction.BstMCI.size
Bio.Restriction.Restriction.BstMWI.size
Bio.Restriction.Restriction.BstNI.size
Bio.Restriction.Restriction.BstNSI.size
Bio.Restriction.Restriction.BstPAI.size
Bio.Restriction.Restriction.BstPI.size
Bio.Restriction.Restriction.BstSCI.size
Bio.Restriction.Restriction.BstSFI.size
Bio.Restriction.Restriction.BstSLI.size
Bio.Restriction.Restriction.BstSNI.size
Bio.Restriction.Restriction.BstUI.size
Bio.Restriction.Restriction.BstV1I.size
Bio.Restriction.Restriction.BstV2I.size
Bio.Restriction.Restriction.BstX2I.size
Bio.Restriction.Restriction.BstXI.size
Bio.Restriction.Restriction.BstYI.size
Bio.Restriction.Restriction.BstZ17I.size
Bio.Restriction.Restriction.BstZI.size
Bio.Restriction.Restriction.Bsu15I.size
Bio.Restriction.Restriction.Bsu36I.size
Bio.Restriction.Restriction.BsuI.size
Bio.Restriction.Restriction.BsuRI.size
Bio.Restriction.Restriction.BsuTUI.size
Bio.Restriction.Restriction.BtgI.size
Bio.Restriction.Restriction.BtgZI.size
Bio.Restriction.Restriction.BthCI.size
Bio.Restriction.Restriction.BtrI.size
Bio.Restriction.Restriction.BtsCI.size
Bio.Restriction.Restriction.BtsI.size
Bio.Restriction.Restriction.BtsIMutI.size
Bio.Restriction.Restriction.BtuMI.size
Bio.Restriction.Restriction.BveI.size
Bio.Restriction.Restriction.Cac8I.size
Bio.Restriction.Restriction.CaiI.size
Bio.Restriction.Restriction.Cal14237I.size
Bio.Restriction.Restriction.CalB3II.size
Bio.Restriction.Restriction.Cau10061II.size
Bio.Restriction.Restriction.CauII.size
Bio.Restriction.Restriction.Cba13II.size
Bio.Restriction.Restriction.Cba16038I.size
Bio.Restriction.Restriction.Cbo67071IV.size
Bio.Restriction.Restriction.Cch467III.size
Bio.Restriction.Restriction.CchII.size
Bio.Restriction.Restriction.CchIII.size
Bio.Restriction.Restriction.CciI.size
Bio.Restriction.Restriction.CciNI.size
Bio.Restriction.Restriction.CcrNAIII.size
Bio.Restriction.Restriction.Cdi11397I.size
Bio.Restriction.Restriction.Cdi81III.size
Bio.Restriction.Restriction.CdiI.size
Bio.Restriction.Restriction.CdpI.size
Bio.Restriction.Restriction.Cdu23823II.size
Bio.Restriction.Restriction.CfoI.size
Bio.Restriction.Restriction.Cfr10I.size
Bio.Restriction.Restriction.Cfr13I.size
Bio.Restriction.Restriction.Cfr42I.size
Bio.Restriction.Restriction.Cfr9I.size
Bio.Restriction.Restriction.CfrI.size
Bio.Restriction.Restriction.Cfupf3II.size
Bio.Restriction.Restriction.Cgl13032I.size
Bio.Restriction.Restriction.Cgl13032II.size
Bio.Restriction.Restriction.ChaI.size
Bio.Restriction.Restriction.CjeFIII.size
Bio.Restriction.Restriction.CjeFV.size
Bio.Restriction.Restriction.CjeI.size
Bio.Restriction.Restriction.CjeNII.size
Bio.Restriction.Restriction.CjeNIII.size
Bio.Restriction.Restriction.CjeNV.size
Bio.Restriction.Restriction.CjeP659IV.size
Bio.Restriction.Restriction.CjePI.size
Bio.Restriction.Restriction.CjuI.size
Bio.Restriction.Restriction.CjuII.size
Bio.Restriction.Restriction.Cla11845III.size
Bio.Restriction.Restriction.ClaI.size
Bio.Restriction.Restriction.Cly7489II.size
Bio.Restriction.Restriction.Cma23826I.size
Bio.Restriction.Restriction.CpoI.size
Bio.Restriction.Restriction.CseI.size
Bio.Restriction.Restriction.CsiI.size
Bio.Restriction.Restriction.Csp2014I.size
Bio.Restriction.Restriction.Csp6I.size
Bio.Restriction.Restriction.CspAI.size
Bio.Restriction.Restriction.CspCI.size
Bio.Restriction.Restriction.CspI.size
Bio.Restriction.Restriction.CstMI.size
Bio.Restriction.Restriction.CviAII.size
Bio.Restriction.Restriction.CviJI.size
Bio.Restriction.Restriction.CviKI_1.size
Bio.Restriction.Restriction.CviQI.size
Bio.Restriction.Restriction.CviRI.size
Bio.Restriction.Restriction.Dde51507I.size
Bio.Restriction.Restriction.DdeI.size
Bio.Restriction.Restriction.DinI.size
Bio.Restriction.Restriction.DpnI.size
Bio.Restriction.Restriction.DpnII.size
Bio.Restriction.Restriction.DraI.size
Bio.Restriction.Restriction.DraII.size
Bio.Restriction.Restriction.DraIII.size
Bio.Restriction.Restriction.DraRI.size
Bio.Restriction.Restriction.DrdI.size
Bio.Restriction.Restriction.DrdII.size
Bio.Restriction.Restriction.DriI.size
Bio.Restriction.Restriction.DsaI.size
Bio.Restriction.Restriction.DseDI.size
Bio.Restriction.Restriction.EaeI.size
Bio.Restriction.Restriction.EagI.size
Bio.Restriction.Restriction.Eam1104I.size
Bio.Restriction.Restriction.Eam1105I.size
Bio.Restriction.Restriction.EarI.size
Bio.Restriction.Restriction.EciI.size
Bio.Restriction.Restriction.Ecl136II.size
Bio.Restriction.Restriction.Ecl35734I.size
Bio.Restriction.Restriction.EclXI.size
Bio.Restriction.Restriction.Eco105I.size
Bio.Restriction.Restriction.Eco130I.size
Bio.Restriction.Restriction.Eco147I.size
Bio.Restriction.Restriction.Eco24I.size
Bio.Restriction.Restriction.Eco31I.size
Bio.Restriction.Restriction.Eco32I.size
Bio.Restriction.Restriction.Eco4465II.size
Bio.Restriction.Restriction.Eco47I.size
Bio.Restriction.Restriction.Eco47III.size
Bio.Restriction.Restriction.Eco52I.size
Bio.Restriction.Restriction.Eco53kI.size
Bio.Restriction.Restriction.Eco57I.size
Bio.Restriction.Restriction.Eco57MI.size
Bio.Restriction.Restriction.Eco72I.size
Bio.Restriction.Restriction.Eco81I.size
Bio.Restriction.Restriction.Eco88I.size
Bio.Restriction.Restriction.Eco91I.size
Bio.Restriction.Restriction.EcoBLMcrX.size
Bio.Restriction.Restriction.EcoE1140I.size
Bio.Restriction.Restriction.EcoHI.size
Bio.Restriction.Restriction.EcoICRI.size
Bio.Restriction.Restriction.EcoMVII.size
Bio.Restriction.Restriction.EcoNI.size
Bio.Restriction.Restriction.EcoO109I.size
Bio.Restriction.Restriction.EcoO65I.size
Bio.Restriction.Restriction.EcoRI.size
Bio.Restriction.Restriction.EcoRII.size
Bio.Restriction.Restriction.EcoRV.size
Bio.Restriction.Restriction.EcoT14I.size
Bio.Restriction.Restriction.EcoT22I.size
Bio.Restriction.Restriction.EcoT38I.size
Bio.Restriction.Restriction.EgeI.size
Bio.Restriction.Restriction.EheI.size
Bio.Restriction.Restriction.Eli8509II.size
Bio.Restriction.Restriction.ErhI.size
Bio.Restriction.Restriction.EsaBC3I.size
Bio.Restriction.Restriction.EsaSSI.size
Bio.Restriction.Restriction.Esp3007I.size
Bio.Restriction.Restriction.Esp3I.size
Bio.Restriction.Restriction.EspI.size
Bio.Restriction.Restriction.Exi27195I.size
Bio.Restriction.Restriction.FaeI.size
Bio.Restriction.Restriction.FaiI.size
Bio.Restriction.Restriction.FalI.size
Bio.Restriction.Restriction.FaqI.size
Bio.Restriction.Restriction.FatI.size
Bio.Restriction.Restriction.FauI.size
Bio.Restriction.Restriction.FauNDI.size
Bio.Restriction.Restriction.FbaI.size
Bio.Restriction.Restriction.FblI.size
Bio.Restriction.Restriction.FinI.size
Bio.Restriction.Restriction.FmuI.size
Bio.Restriction.Restriction.Fnu4HI.size
Bio.Restriction.Restriction.FnuDII.size
Bio.Restriction.Restriction.FokI.size
Bio.Restriction.Restriction.FriOI.size
Bio.Restriction.Restriction.FseI.size
Bio.Restriction.Restriction.Fsp4HI.size
Bio.Restriction.Restriction.FspAI.size
Bio.Restriction.Restriction.FspBI.size
Bio.Restriction.Restriction.FspEI.size
Bio.Restriction.Restriction.FspI.size
Bio.Restriction.Restriction.FtnUV.size
Bio.Restriction.Restriction.GauT27I.size
Bio.Restriction.Restriction.Gba708II.size
Bio.Restriction.Restriction.GdiII.size
Bio.Restriction.Restriction.GlaI.size
Bio.Restriction.Restriction.GluI.size
Bio.Restriction.Restriction.GsaI.size
Bio.Restriction.Restriction.GsuI.size
Bio.Restriction.Restriction.HaeI.size
Bio.Restriction.Restriction.HaeII.size
Bio.Restriction.Restriction.HaeIII.size
Bio.Restriction.Restriction.HapII.size
Bio.Restriction.Restriction.HauII.size
Bio.Restriction.Restriction.HbaII.size
Bio.Restriction.Restriction.HgaI.size
Bio.Restriction.Restriction.HgiAI.size
Bio.Restriction.Restriction.HgiCI.size
Bio.Restriction.Restriction.HgiEII.size
Bio.Restriction.Restriction.HgiJII.size
Bio.Restriction.Restriction.HhaI.size
Bio.Restriction.Restriction.Hin1I.size
Bio.Restriction.Restriction.Hin1II.size
Bio.Restriction.Restriction.Hin4I.size
Bio.Restriction.Restriction.Hin4II.size
Bio.Restriction.Restriction.Hin6I.size
Bio.Restriction.Restriction.HinP1I.size
Bio.Restriction.Restriction.HincII.size
Bio.Restriction.Restriction.HindII.size
Bio.Restriction.Restriction.HindIII.size
Bio.Restriction.Restriction.HinfI.size
Bio.Restriction.Restriction.HpaI.size
Bio.Restriction.Restriction.HpaII.size
Bio.Restriction.Restriction.HphI.size
Bio.Restriction.Restriction.Hpy166II.size
Bio.Restriction.Restriction.Hpy178III.size
Bio.Restriction.Restriction.Hpy188I.size
Bio.Restriction.Restriction.Hpy188III.size
Bio.Restriction.Restriction.Hpy8I.size
Bio.Restriction.Restriction.Hpy99I.size
Bio.Restriction.Restriction.Hpy99XIII.size
Bio.Restriction.Restriction.Hpy99XIV.size
Bio.Restriction.Restriction.Hpy99XIV_mut1.size
Bio.Restriction.Restriction.Hpy99XXII.size
Bio.Restriction.Restriction.HpyAV.size
Bio.Restriction.Restriction.HpyAXIV.size
Bio.Restriction.Restriction.HpyAXVI_mut1.size
Bio.Restriction.Restriction.HpyAXVI_mut2.size
Bio.Restriction.Restriction.HpyCH4III.size
Bio.Restriction.Restriction.HpyCH4IV.size
Bio.Restriction.Restriction.HpyCH4V.size
Bio.Restriction.Restriction.HpyF10VI.size
Bio.Restriction.Restriction.HpyF3I.size
Bio.Restriction.Restriction.HpySE526I.size
Bio.Restriction.Restriction.HpyUM032XIII.size
Bio.Restriction.Restriction.HpyUM032XIII_mut1.size
Bio.Restriction.Restriction.HpyUM032XIV.size
Bio.Restriction.Restriction.HpyUM037X.size
Bio.Restriction.Restriction.Hsp92I.size
Bio.Restriction.Restriction.Hsp92II.size
Bio.Restriction.Restriction.HspAI.size
Bio.Restriction.Restriction.Jma19592I.size
Bio.Restriction.Restriction.Jma19592II.size
Bio.Restriction.Restriction.Jsp2502II.size
Bio.Restriction.Restriction.KasI.size
Bio.Restriction.Restriction.KflI.size
Bio.Restriction.Restriction.Kor51II.size
Bio.Restriction.Restriction.Kpn2I.size
Bio.Restriction.Restriction.KpnI.size
Bio.Restriction.Restriction.KpnNIH30III.size
Bio.Restriction.Restriction.KroI.size
Bio.Restriction.Restriction.Ksp22I.size
Bio.Restriction.Restriction.Ksp632I.size
Bio.Restriction.Restriction.KspAI.size
Bio.Restriction.Restriction.KspI.size
Bio.Restriction.Restriction.Kzo9I.size
Bio.Restriction.Restriction.Lba2029III.size
Bio.Restriction.Restriction.LguI.size
Bio.Restriction.Restriction.LmnI.size
Bio.Restriction.Restriction.Lmo370I.size
Bio.Restriction.Restriction.Lmo911II.size
Bio.Restriction.Restriction.LpnI.size
Bio.Restriction.Restriction.LpnPI.size
Bio.Restriction.Restriction.Lsp1109I.size
Bio.Restriction.Restriction.Lsp48III.size
Bio.Restriction.Restriction.Lsp6406VI.size
Bio.Restriction.Restriction.LweI.size
Bio.Restriction.Restriction.MabI.size
Bio.Restriction.Restriction.MaeI.size
Bio.Restriction.Restriction.MaeII.size
Bio.Restriction.Restriction.MaeIII.size
Bio.Restriction.Restriction.MalI.size
Bio.Restriction.Restriction.MaqI.size
Bio.Restriction.Restriction.MauBI.size
Bio.Restriction.Restriction.Mba11I.size
Bio.Restriction.Restriction.MbiI.size
Bio.Restriction.Restriction.MboI.size
Bio.Restriction.Restriction.MboII.size
Bio.Restriction.Restriction.McaTI.size
Bio.Restriction.Restriction.Mcr10I.size
Bio.Restriction.Restriction.McrI.size
Bio.Restriction.Restriction.MfeI.size
Bio.Restriction.Restriction.MflI.size
Bio.Restriction.Restriction.MhlI.size
Bio.Restriction.Restriction.MjaIV.size
Bio.Restriction.Restriction.MkaDII.size
Bio.Restriction.Restriction.MlsI.size
Bio.Restriction.Restriction.MluCI.size
Bio.Restriction.Restriction.MluI.size
Bio.Restriction.Restriction.MluNI.size
Bio.Restriction.Restriction.Mly113I.size
Bio.Restriction.Restriction.MlyI.size
Bio.Restriction.Restriction.MmeI.size
Bio.Restriction.Restriction.MnlI.size
Bio.Restriction.Restriction.Mox20I.size
Bio.Restriction.Restriction.Mph1103I.size
Bio.Restriction.Restriction.MreI.size
Bio.Restriction.Restriction.MroI.size
Bio.Restriction.Restriction.MroNI.size
Bio.Restriction.Restriction.MroXI.size
Bio.Restriction.Restriction.MscI.size
Bio.Restriction.Restriction.MseI.size
Bio.Restriction.Restriction.MslI.size
Bio.Restriction.Restriction.Msp20I.size
Bio.Restriction.Restriction.MspA1I.size
Bio.Restriction.Restriction.MspCI.size
Bio.Restriction.Restriction.MspI.size
Bio.Restriction.Restriction.MspI7II.size
Bio.Restriction.Restriction.MspJI.size
Bio.Restriction.Restriction.MspR9I.size
Bio.Restriction.Restriction.MssI.size
Bio.Restriction.Restriction.MstI.size
Bio.Restriction.Restriction.MteI.size
Bio.Restriction.Restriction.MtuHN878II.size
Bio.Restriction.Restriction.MunI.size
Bio.Restriction.Restriction.Mva1269I.size
Bio.Restriction.Restriction.MvaI.size
Bio.Restriction.Restriction.MvnI.size
Bio.Restriction.Restriction.MwoI.size
Bio.Restriction.Restriction.NaeI.size
Bio.Restriction.Restriction.Nal45188II.size
Bio.Restriction.Restriction.NarI.size
Bio.Restriction.Restriction.Nbr128II.size
Bio.Restriction.Restriction.NciI.size
Bio.Restriction.Restriction.NcoI.size
Bio.Restriction.Restriction.NdeI.size
Bio.Restriction.Restriction.NdeII.size
Bio.Restriction.Restriction.NgoAVII.size
Bio.Restriction.Restriction.NgoAVIII.size
Bio.Restriction.Restriction.NgoMIV.size
Bio.Restriction.Restriction.NhaXI.size
Bio.Restriction.Restriction.NheI.size
Bio.Restriction.Restriction.NhoI.size
Bio.Restriction.Restriction.NlaCI.size
Bio.Restriction.Restriction.NlaIII.size
Bio.Restriction.Restriction.NlaIV.size
Bio.Restriction.Restriction.Nli3877I.size
Bio.Restriction.Restriction.NmeAIII.size
Bio.Restriction.Restriction.NmeDI.size
Bio.Restriction.Restriction.NmuCI.size
Bio.Restriction.Restriction.NotI.size
Bio.Restriction.Restriction.NpeUS61II.size
Bio.Restriction.Restriction.NruI.size
Bio.Restriction.Restriction.NsbI.size
Bio.Restriction.Restriction.NsiI.size
Bio.Restriction.Restriction.NspBII.size
Bio.Restriction.Restriction.NspI.size
Bio.Restriction.Restriction.NspV.size
Bio.Restriction.Restriction.OliI.size
Bio.Restriction.Restriction.OspHL35III.size
Bio.Restriction.Restriction.PabI.size
Bio.Restriction.Restriction.Pac19842II.size
Bio.Restriction.Restriction.PacI.size
Bio.Restriction.Restriction.PacIII.size
Bio.Restriction.Restriction.PaeI.size
Bio.Restriction.Restriction.PaeR7I.size
Bio.Restriction.Restriction.PagI.size
Bio.Restriction.Restriction.Pal408I.size
Bio.Restriction.Restriction.PalAI.size
Bio.Restriction.Restriction.PasI.size
Bio.Restriction.Restriction.PauI.size
Bio.Restriction.Restriction.PcaII.size
Bio.Restriction.Restriction.PceI.size
Bio.Restriction.Restriction.PciI.size
Bio.Restriction.Restriction.PciSI.size
Bio.Restriction.Restriction.PcsI.size
Bio.Restriction.Restriction.PctI.size
Bio.Restriction.Restriction.Pdi8503III.size
Bio.Restriction.Restriction.PdiI.size
Bio.Restriction.Restriction.PdmI.size
Bio.Restriction.Restriction.Pdu1735I.size
Bio.Restriction.Restriction.PenI.size
Bio.Restriction.Restriction.PfeI.size
Bio.Restriction.Restriction.Pfl1108I.size
Bio.Restriction.Restriction.Pfl23II.size
Bio.Restriction.Restriction.Pfl8569I.size
Bio.Restriction.Restriction.PflFI.size
Bio.Restriction.Restriction.PflMI.size
Bio.Restriction.Restriction.PfoI.size
Bio.Restriction.Restriction.PfrJS12IV.size
Bio.Restriction.Restriction.PfrJS12V.size
Bio.Restriction.Restriction.PfrJS15III.size
Bio.Restriction.Restriction.Pin17FIII.size
Bio.Restriction.Restriction.PinAI.size
Bio.Restriction.Restriction.PkrI.size
Bio.Restriction.Restriction.PlaDI.size
Bio.Restriction.Restriction.Ple19I.size
Bio.Restriction.Restriction.PleI.size
Bio.Restriction.Restriction.PliMI.size
Bio.Restriction.Restriction.PluTI.size
Bio.Restriction.Restriction.PmaCI.size
Bio.Restriction.Restriction.PmeI.size
Bio.Restriction.Restriction.PmlI.size
Bio.Restriction.Restriction.PpiI.size
Bio.Restriction.Restriction.PpsI.size
Bio.Restriction.Restriction.Ppu10I.size
Bio.Restriction.Restriction.Ppu21I.size
Bio.Restriction.Restriction.PpuMI.size
Bio.Restriction.Restriction.PscI.size
Bio.Restriction.Restriction.Pse18267I.size
Bio.Restriction.Restriction.PshAI.size
Bio.Restriction.Restriction.PshBI.size
Bio.Restriction.Restriction.PsiI.size
Bio.Restriction.Restriction.Psp0357II.size
Bio.Restriction.Restriction.Psp03I.size
Bio.Restriction.Restriction.Psp124BI.size
Bio.Restriction.Restriction.Psp1406I.size
Bio.Restriction.Restriction.Psp5II.size
Bio.Restriction.Restriction.Psp6I.size
Bio.Restriction.Restriction.PspCI.size
Bio.Restriction.Restriction.PspEI.size
Bio.Restriction.Restriction.PspFI.size
Bio.Restriction.Restriction.PspGI.size
Bio.Restriction.Restriction.PspLI.size
Bio.Restriction.Restriction.PspN4I.size
Bio.Restriction.Restriction.PspOMI.size
Bio.Restriction.Restriction.PspOMII.size
Bio.Restriction.Restriction.PspPI.size
Bio.Restriction.Restriction.PspPPI.size
Bio.Restriction.Restriction.PspPRI.size
Bio.Restriction.Restriction.PspXI.size
Bio.Restriction.Restriction.PsrI.size
Bio.Restriction.Restriction.PssI.size
Bio.Restriction.Restriction.Pst14472I.size
Bio.Restriction.Restriction.Pst273I.size
Bio.Restriction.Restriction.PstI.size
Bio.Restriction.Restriction.PstNI.size
Bio.Restriction.Restriction.PsuGI.size
Bio.Restriction.Restriction.PsuI.size
Bio.Restriction.Restriction.PsyI.size
Bio.Restriction.Restriction.PteI.size
Bio.Restriction.Restriction.PvuI.size
Bio.Restriction.Restriction.PvuII.size
Bio.Restriction.Restriction.Rba2021I.size
Bio.Restriction.Restriction.RceI.size
Bio.Restriction.Restriction.RdeGBI.size
Bio.Restriction.Restriction.RdeGBII.size
Bio.Restriction.Restriction.RdeGBIII.size
Bio.Restriction.Restriction.RflFIII.size
Bio.Restriction.Restriction.RgaI.size
Bio.Restriction.Restriction.RigI.size
Bio.Restriction.Restriction.RlaI.size
Bio.Restriction.Restriction.RlaII.size
Bio.Restriction.Restriction.RleAI.size
Bio.Restriction.Restriction.RpaB5I.size
Bio.Restriction.Restriction.RpaBI.size
Bio.Restriction.Restriction.RpaI.size
Bio.Restriction.Restriction.RpaTI.size
Bio.Restriction.Restriction.RruI.size
Bio.Restriction.Restriction.RsaI.size
Bio.Restriction.Restriction.RsaNI.size
Bio.Restriction.Restriction.RseI.size
Bio.Restriction.Restriction.Rsp008IV.size
Bio.Restriction.Restriction.Rsp008V.size
Bio.Restriction.Restriction.RspPBTS2III.size
Bio.Restriction.Restriction.Rsr2I.size
Bio.Restriction.Restriction.RsrII.size
Bio.Restriction.Restriction.SacI.size
Bio.Restriction.Restriction.SacII.size
Bio.Restriction.Restriction.Saf8902III.size
Bio.Restriction.Restriction.SalI.size
Bio.Restriction.Restriction.SanDI.size
Bio.Restriction.Restriction.SapI.size
Bio.Restriction.Restriction.SaqAI.size
Bio.Restriction.Restriction.SatI.size
Bio.Restriction.Restriction.Sau3AI.size
Bio.Restriction.Restriction.Sau96I.size
Bio.Restriction.Restriction.SauI.size
Bio.Restriction.Restriction.SbfI.size
Bio.Restriction.Restriction.Sbo46I.size
Bio.Restriction.Restriction.ScaI.size
Bio.Restriction.Restriction.SchI.size
Bio.Restriction.Restriction.SciI.size
Bio.Restriction.Restriction.ScoDS2II.size
Bio.Restriction.Restriction.ScrFI.size
Bio.Restriction.Restriction.SdaI.size
Bio.Restriction.Restriction.SdeAI.size
Bio.Restriction.Restriction.SdeOSI.size
Bio.Restriction.Restriction.SduI.size
Bio.Restriction.Restriction.SecI.size
Bio.Restriction.Restriction.SelI.size
Bio.Restriction.Restriction.Sen13311III.size
Bio.Restriction.Restriction.Sen13III.size
Bio.Restriction.Restriction.Sen1736II.size
Bio.Restriction.Restriction.Sen1878IV.size
Bio.Restriction.Restriction.Sen1880IV.size
Bio.Restriction.Restriction.Sen1896IV.size
Bio.Restriction.Restriction.Sen1898IV.size
Bio.Restriction.Restriction.Sen1906IV.size
Bio.Restriction.Restriction.Sen1908IV.size
Bio.Restriction.Restriction.Sen1910IV.size
Bio.Restriction.Restriction.SenC1808III.size
Bio.Restriction.Restriction.SenC1810IV.size
Bio.Restriction.Restriction.SenSARA26III.size
Bio.Restriction.Restriction.SenTFIV.size
Bio.Restriction.Restriction.SetI.size
Bio.Restriction.Restriction.SexAI.size
Bio.Restriction.Restriction.SfaAI.size
Bio.Restriction.Restriction.SfaNI.size
Bio.Restriction.Restriction.SfcI.size
Bio.Restriction.Restriction.SfeI.size
Bio.Restriction.Restriction.SfiI.size
Bio.Restriction.Restriction.SfoI.size
Bio.Restriction.Restriction.Sfr274I.size
Bio.Restriction.Restriction.Sfr303I.size
Bio.Restriction.Restriction.SfuI.size
Bio.Restriction.Restriction.SgeI.size
Bio.Restriction.Restriction.SgfI.size
Bio.Restriction.Restriction.SgrAI.size
Bio.Restriction.Restriction.SgrBI.size
Bio.Restriction.Restriction.SgrDI.size
Bio.Restriction.Restriction.SgrTI.size
Bio.Restriction.Restriction.SgsI.size
Bio.Restriction.Restriction.SimI.size
Bio.Restriction.Restriction.SinI.size
Bio.Restriction.Restriction.SlaI.size
Bio.Restriction.Restriction.SmaI.size
Bio.Restriction.Restriction.SmiI.size
Bio.Restriction.Restriction.SmiMI.size
Bio.Restriction.Restriction.SmlI.size
Bio.Restriction.Restriction.SmoI.size
Bio.Restriction.Restriction.SnaBI.size
Bio.Restriction.Restriction.SnaI.size
Bio.Restriction.Restriction.Sno506I.size
Bio.Restriction.Restriction.SpeI.size
Bio.Restriction.Restriction.SphI.size
Bio.Restriction.Restriction.SplI.size
Bio.Restriction.Restriction.SpnRII.size
Bio.Restriction.Restriction.SpoDI.size
Bio.Restriction.Restriction.SrfI.size
Bio.Restriction.Restriction.Sse232I.size
Bio.Restriction.Restriction.Sse8387I.size
Bio.Restriction.Restriction.Sse8647I.size
Bio.Restriction.Restriction.Sse9I.size
Bio.Restriction.Restriction.SseBI.size
Bio.Restriction.Restriction.SsiI.size
Bio.Restriction.Restriction.Ssp6803IV.size
Bio.Restriction.Restriction.Ssp714II.size
Bio.Restriction.Restriction.SspD5I.size
Bio.Restriction.Restriction.SspDI.size
Bio.Restriction.Restriction.SspI.size
Bio.Restriction.Restriction.SspMI.size
Bio.Restriction.Restriction.SstE37I.size
Bio.Restriction.Restriction.SstI.size
Bio.Restriction.Restriction.Sth132I.size
Bio.Restriction.Restriction.Sth20745III.size
Bio.Restriction.Restriction.Sth302II.size
Bio.Restriction.Restriction.StsI.size
Bio.Restriction.Restriction.StuI.size
Bio.Restriction.Restriction.Sty13348III.size
Bio.Restriction.Restriction.StyD4I.size
Bio.Restriction.Restriction.StyI.size
Bio.Restriction.Restriction.SwaI.size
Bio.Restriction.Restriction.TaaI.size
Bio.Restriction.Restriction.TaiI.size
Bio.Restriction.Restriction.TaqI.size
Bio.Restriction.Restriction.TaqII.size
Bio.Restriction.Restriction.TaqIII.size
Bio.Restriction.Restriction.TasI.size
Bio.Restriction.Restriction.TatI.size
Bio.Restriction.Restriction.TauI.size
Bio.Restriction.Restriction.TfiI.size
Bio.Restriction.Restriction.Tru1I.size
Bio.Restriction.Restriction.Tru9I.size
Bio.Restriction.Restriction.TscAI.size
Bio.Restriction.Restriction.TseFI.size
Bio.Restriction.Restriction.TseI.size
Bio.Restriction.Restriction.TsoI.size
Bio.Restriction.Restriction.Tsp45I.size
Bio.Restriction.Restriction.Tsp4CI.size
Bio.Restriction.Restriction.TspARh3I.size
Bio.Restriction.Restriction.TspDTI.size
Bio.Restriction.Restriction.TspEI.size
Bio.Restriction.Restriction.TspGWI.size
Bio.Restriction.Restriction.TspMI.size
Bio.Restriction.Restriction.TspRI.size
Bio.Restriction.Restriction.TssI.size
Bio.Restriction.Restriction.TstI.size
Bio.Restriction.Restriction.TsuI.size
Bio.Restriction.Restriction.Tth111I.size
Bio.Restriction.Restriction.Tth111II.size
Bio.Restriction.Restriction.UbaF11I.size
Bio.Restriction.Restriction.UbaF12I.size
Bio.Restriction.Restriction.UbaF13I.size
Bio.Restriction.Restriction.UbaF14I.size
Bio.Restriction.Restriction.UbaF9I.size
Bio.Restriction.Restriction.UbaPI.size
Bio.Restriction.Restriction.UcoMSI.size
Bio.Restriction.Restriction.UnbI.size
Bio.Restriction.Restriction.Van91I.size
Bio.Restriction.Restriction.Vha464I.size
Bio.Restriction.Restriction.VneI.size
Bio.Restriction.Restriction.VpaK11AI.size
Bio.Restriction.Restriction.VpaK11BI.size
Bio.Restriction.Restriction.VspI.size
Bio.Restriction.Restriction.Vtu19109I.size
Bio.Restriction.Restriction.WviI.size
Bio.Restriction.Restriction.XagI.size
Bio.Restriction.Restriction.XapI.size
Bio.Restriction.Restriction.XbaI.size
Bio.Restriction.Restriction.Xca85IV.size
Bio.Restriction.Restriction.XceI.size
Bio.Restriction.Restriction.XcmI.size
Bio.Restriction.Restriction.XhoI.size
Bio.Restriction.Restriction.XhoII.size
Bio.Restriction.Restriction.XmaI.size
Bio.Restriction.Restriction.XmaIII.size
Bio.Restriction.Restriction.XmaJI.size
Bio.Restriction.Restriction.XmiI.size
Bio.Restriction.Restriction.XmnI.size
Bio.Restriction.Restriction.XspI.size
Bio.Restriction.Restriction.YkrI.size
Bio.Restriction.Restriction.Yps3606I.size
Bio.Restriction.Restriction.ZraI.size
Bio.Restriction.Restriction.ZrmI.size
Bio.Restriction.Restriction.Zsp2I.size
Bio.codonalign.codonalphabet.CodonAlphabet.size" class="py-name" href="#" onclick="return doclink('link-134', 'size', 'link-132');">size</a></tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">line_split</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"-"</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L416"></a><tt class="py-lineno">416</tt>  <tt class="py-line"> </tt>
<a name="L417"></a><tt class="py-lineno">417</tt>  <tt class="py-line">                            <tt class="py-comment"># "inclusive" end position is start + length - 1</tt> </tt>
<a name="L418"></a><tt class="py-lineno">418</tt>  <tt class="py-line">                            <tt id="link-135" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.end=Bio.SeqFeature.CompoundLocation-class.html#end,Variable Bio.SeqFeature.FeatureLocation.end=Bio.SeqFeature.FeatureLocation-class.html#end"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-135', 'end', 'link-135');">end</a></tt> <tt class="py-op">=</tt> <tt id="link-136" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-136', 'start', 'link-131');">start</a></tt> <tt class="py-op">+</tt> <tt id="link-137" class="py-name"><a title="Bio.Alphabet.Alphabet.size
Bio.Alphabet.Reduced.HPModel.size
Bio.Alphabet.Reduced.Murphy10.size
Bio.Alphabet.Reduced.Murphy15.size
Bio.Alphabet.Reduced.Murphy4.size
Bio.Alphabet.Reduced.Murphy8.size
Bio.Alphabet.Reduced.PC5.size
Bio.Alphabet.SingleLetterAlphabet.size
Bio.Alphabet.ThreeLetterProtein.size
Bio.GenBank._FeatureConsumer.size
Bio.GenBank._RecordConsumer.size
Bio.Restriction.Restriction.AanI.size
Bio.Restriction.Restriction.AarI.size
Bio.Restriction.Restriction.AasI.size
Bio.Restriction.Restriction.AatII.size
Bio.Restriction.Restriction.Aba6411II.size
Bio.Restriction.Restriction.AbaCIII.size
Bio.Restriction.Restriction.AbaSI.size
Bio.Restriction.Restriction.AbaUMB2I.size
Bio.Restriction.Restriction.AbsI.size
Bio.Restriction.Restriction.Acc16I.size
Bio.Restriction.Restriction.Acc36I.size
Bio.Restriction.Restriction.Acc65I.size
Bio.Restriction.Restriction.Acc65V.size
Bio.Restriction.Restriction.AccB1I.size
Bio.Restriction.Restriction.AccB7I.size
Bio.Restriction.Restriction.AccBSI.size
Bio.Restriction.Restriction.AccI.size
Bio.Restriction.Restriction.AccII.size
Bio.Restriction.Restriction.AccIII.size
Bio.Restriction.Restriction.AceIII.size
Bio.Restriction.Restriction.AchA6III.size
Bio.Restriction.Restriction.AciI.size
Bio.Restriction.Restriction.AclI.size
Bio.Restriction.Restriction.AclWI.size
Bio.Restriction.Restriction.Aco12261II.size
Bio.Restriction.Restriction.AcoI.size
Bio.Restriction.Restriction.AcsI.size
Bio.Restriction.Restriction.AcuI.size
Bio.Restriction.Restriction.AcvI.size
Bio.Restriction.Restriction.AcyI.size
Bio.Restriction.Restriction.AdeI.size
Bio.Restriction.Restriction.Adh6U21I.size
Bio.Restriction.Restriction.AfaI.size
Bio.Restriction.Restriction.AfeI.size
Bio.Restriction.Restriction.AfiI.size
Bio.Restriction.Restriction.AflII.size
Bio.Restriction.Restriction.AflIII.size
Bio.Restriction.Restriction.AgeI.size
Bio.Restriction.Restriction.AgsI.size
Bio.Restriction.Restriction.AhaIII.size
Bio.Restriction.Restriction.AhdI.size
Bio.Restriction.Restriction.AhlI.size
Bio.Restriction.Restriction.AhyRBAHI.size
Bio.Restriction.Restriction.AhyYL17I.size
Bio.Restriction.Restriction.AjiI.size
Bio.Restriction.Restriction.AjnI.size
Bio.Restriction.Restriction.AjuI.size
Bio.Restriction.Restriction.AleI.size
Bio.Restriction.Restriction.AlfI.size
Bio.Restriction.Restriction.AloI.size
Bio.Restriction.Restriction.AluBI.size
Bio.Restriction.Restriction.AluI.size
Bio.Restriction.Restriction.Alw21I.size
Bio.Restriction.Restriction.Alw26I.size
Bio.Restriction.Restriction.Alw44I.size
Bio.Restriction.Restriction.AlwFI.size
Bio.Restriction.Restriction.AlwI.size
Bio.Restriction.Restriction.AlwNI.size
Bio.Restriction.Restriction.Ama87I.size
Bio.Restriction.Restriction.AmaCSI.size
Bio.Restriction.Restriction.Aor13HI.size
Bio.Restriction.Restriction.Aor51HI.size
Bio.Restriction.Restriction.AoxI.size
Bio.Restriction.Restriction.ApaBI.size
Bio.Restriction.Restriction.ApaI.size
Bio.Restriction.Restriction.ApaLI.size
Bio.Restriction.Restriction.ApeKI.size
Bio.Restriction.Restriction.ApoI.size
Bio.Restriction.Restriction.ApyPI.size
Bio.Restriction.Restriction.AquII.size
Bio.Restriction.Restriction.AquIII.size
Bio.Restriction.Restriction.AquIV.size
Bio.Restriction.Restriction.ArsI.size
Bio.Restriction.Restriction.AscI.size
Bio.Restriction.Restriction.AseI.size
Bio.Restriction.Restriction.Asi256I.size
Bio.Restriction.Restriction.AsiGI.size
Bio.Restriction.Restriction.AsiSI.size
Bio.Restriction.Restriction.Asp114II.size
Bio.Restriction.Restriction.Asp700I.size
Bio.Restriction.Restriction.Asp718I.size
Bio.Restriction.Restriction.AspA2I.size
Bio.Restriction.Restriction.AspBHI.size
Bio.Restriction.Restriction.AspDUT2V.size
Bio.Restriction.Restriction.AspJHL3II.size
Bio.Restriction.Restriction.AspLEI.size
Bio.Restriction.Restriction.AspS9I.size
Bio.Restriction.Restriction.Asu14238IV.size
Bio.Restriction.Restriction.AsuC2I.size
Bio.Restriction.Restriction.AsuHPI.size
Bio.Restriction.Restriction.AsuI.size
Bio.Restriction.Restriction.AsuII.size
Bio.Restriction.Restriction.AsuNHI.size
Bio.Restriction.Restriction.AteTI.size
Bio.Restriction.Restriction.AvaI.size
Bio.Restriction.Restriction.AvaII.size
Bio.Restriction.Restriction.AvaIII.size
Bio.Restriction.Restriction.AvrII.size
Bio.Restriction.Restriction.Awo1030IV.size
Bio.Restriction.Restriction.AxyI.size
Bio.Restriction.Restriction.BaeGI.size
Bio.Restriction.Restriction.BaeI.size
Bio.Restriction.Restriction.Bag18758I.size
Bio.Restriction.Restriction.BalI.size
Bio.Restriction.Restriction.BamHI.size
Bio.Restriction.Restriction.BanI.size
Bio.Restriction.Restriction.BanII.size
Bio.Restriction.Restriction.BanLI.size
Bio.Restriction.Restriction.BarI.size
Bio.Restriction.Restriction.BauI.size
Bio.Restriction.Restriction.BbrPI.size
Bio.Restriction.Restriction.BbsI.size
Bio.Restriction.Restriction.Bbv12I.size
Bio.Restriction.Restriction.BbvCI.size
Bio.Restriction.Restriction.BbvI.size
Bio.Restriction.Restriction.BbvII.size
Bio.Restriction.Restriction.BccI.size
Bio.Restriction.Restriction.Bce3081I.size
Bio.Restriction.Restriction.Bce83I.size
Bio.Restriction.Restriction.BceAI.size
Bio.Restriction.Restriction.BceSIV.size
Bio.Restriction.Restriction.BcefI.size
Bio.Restriction.Restriction.BcgI.size
Bio.Restriction.Restriction.BciT130I.size
Bio.Restriction.Restriction.BciVI.size
Bio.Restriction.Restriction.BclI.size
Bio.Restriction.Restriction.BcnI.size
Bio.Restriction.Restriction.BcoDI.size
Bio.Restriction.Restriction.BcuI.size
Bio.Restriction.Restriction.BdaI.size
Bio.Restriction.Restriction.BetI.size
Bio.Restriction.Restriction.BfaI.size
Bio.Restriction.Restriction.BfaSII.size
Bio.Restriction.Restriction.BfiI.size
Bio.Restriction.Restriction.BfmI.size
Bio.Restriction.Restriction.BfoI.size
Bio.Restriction.Restriction.BfrI.size
Bio.Restriction.Restriction.BfuAI.size
Bio.Restriction.Restriction.BfuI.size
Bio.Restriction.Restriction.BglI.size
Bio.Restriction.Restriction.BglII.size
Bio.Restriction.Restriction.BinI.size
Bio.Restriction.Restriction.BisI.size
Bio.Restriction.Restriction.Ble402II.size
Bio.Restriction.Restriction.BlnI.size
Bio.Restriction.Restriction.BloAII.size
Bio.Restriction.Restriction.BlpI.size
Bio.Restriction.Restriction.BlsI.size
Bio.Restriction.Restriction.BmcAI.size
Bio.Restriction.Restriction.Bme1390I.size
Bio.Restriction.Restriction.Bme18I.size
Bio.Restriction.Restriction.BmeDI.size
Bio.Restriction.Restriction.BmeRI.size
Bio.Restriction.Restriction.BmeT110I.size
Bio.Restriction.Restriction.BmgBI.size
Bio.Restriction.Restriction.BmgI.size
Bio.Restriction.Restriction.BmgT120I.size
Bio.Restriction.Restriction.BmiI.size
Bio.Restriction.Restriction.BmrFI.size
Bio.Restriction.Restriction.BmrI.size
Bio.Restriction.Restriction.BmsI.size
Bio.Restriction.Restriction.BmtI.size
Bio.Restriction.Restriction.BmuI.size
Bio.Restriction.Restriction.BoxI.size
Bio.Restriction.Restriction.BpiI.size
Bio.Restriction.Restriction.BplI.size
Bio.Restriction.Restriction.BpmI.size
Bio.Restriction.Restriction.Bpu10I.size
Bio.Restriction.Restriction.Bpu1102I.size
Bio.Restriction.Restriction.Bpu14I.size
Bio.Restriction.Restriction.BpuEI.size
Bio.Restriction.Restriction.BpuMI.size
Bio.Restriction.Restriction.Bsa29I.size
Bio.Restriction.Restriction.BsaAI.size
Bio.Restriction.Restriction.BsaBI.size
Bio.Restriction.Restriction.BsaHI.size
Bio.Restriction.Restriction.BsaI.size
Bio.Restriction.Restriction.BsaJI.size
Bio.Restriction.Restriction.BsaWI.size
Bio.Restriction.Restriction.BsaXI.size
Bio.Restriction.Restriction.BsbI.size
Bio.Restriction.Restriction.Bsc4I.size
Bio.Restriction.Restriction.BscAI.size
Bio.Restriction.Restriction.BscGI.size
Bio.Restriction.Restriction.Bse118I.size
Bio.Restriction.Restriction.Bse1I.size
Bio.Restriction.Restriction.Bse21I.size
Bio.Restriction.Restriction.Bse3DI.size
Bio.Restriction.Restriction.Bse8I.size
Bio.Restriction.Restriction.BseAI.size
Bio.Restriction.Restriction.BseBI.size
Bio.Restriction.Restriction.BseCI.size
Bio.Restriction.Restriction.BseDI.size
Bio.Restriction.Restriction.BseGI.size
Bio.Restriction.Restriction.BseJI.size
Bio.Restriction.Restriction.BseLI.size
Bio.Restriction.Restriction.BseMI.size
Bio.Restriction.Restriction.BseMII.size
Bio.Restriction.Restriction.BseNI.size
Bio.Restriction.Restriction.BsePI.size
Bio.Restriction.Restriction.BseRI.size
Bio.Restriction.Restriction.BseSI.size
Bio.Restriction.Restriction.BseX3I.size
Bio.Restriction.Restriction.BseXI.size
Bio.Restriction.Restriction.BseYI.size
Bio.Restriction.Restriction.BsgI.size
Bio.Restriction.Restriction.Bsh1236I.size
Bio.Restriction.Restriction.Bsh1285I.size
Bio.Restriction.Restriction.BshFI.size
Bio.Restriction.Restriction.BshNI.size
Bio.Restriction.Restriction.BshTI.size
Bio.Restriction.Restriction.BshVI.size
Bio.Restriction.Restriction.BsiEI.size
Bio.Restriction.Restriction.BsiHKAI.size
Bio.Restriction.Restriction.BsiHKCI.size
Bio.Restriction.Restriction.BsiI.size
Bio.Restriction.Restriction.BsiSI.size
Bio.Restriction.Restriction.BsiWI.size
Bio.Restriction.Restriction.BsiYI.size
Bio.Restriction.Restriction.BslFI.size
Bio.Restriction.Restriction.BslI.size
Bio.Restriction.Restriction.BsmAI.size
Bio.Restriction.Restriction.BsmBI.size
Bio.Restriction.Restriction.BsmFI.size
Bio.Restriction.Restriction.BsmI.size
Bio.Restriction.Restriction.BsnI.size
Bio.Restriction.Restriction.Bso31I.size
Bio.Restriction.Restriction.BsoBI.size
Bio.Restriction.Restriction.Bsp119I.size
Bio.Restriction.Restriction.Bsp120I.size
Bio.Restriction.Restriction.Bsp1286I.size
Bio.Restriction.Restriction.Bsp13I.size
Bio.Restriction.Restriction.Bsp1407I.size
Bio.Restriction.Restriction.Bsp143I.size
Bio.Restriction.Restriction.Bsp1720I.size
Bio.Restriction.Restriction.Bsp19I.size
Bio.Restriction.Restriction.Bsp24I.size
Bio.Restriction.Restriction.Bsp3004IV.size
Bio.Restriction.Restriction.Bsp460III.size
Bio.Restriction.Restriction.Bsp68I.size
Bio.Restriction.Restriction.BspACI.size
Bio.Restriction.Restriction.BspANI.size
Bio.Restriction.Restriction.BspCNI.size
Bio.Restriction.Restriction.BspD6I.size
Bio.Restriction.Restriction.BspDI.size
Bio.Restriction.Restriction.BspEI.size
Bio.Restriction.Restriction.BspFNI.size
Bio.Restriction.Restriction.BspGI.size
Bio.Restriction.Restriction.BspHI.size
Bio.Restriction.Restriction.BspLI.size
Bio.Restriction.Restriction.BspLU11I.size
Bio.Restriction.Restriction.BspMAI.size
Bio.Restriction.Restriction.BspMI.size
Bio.Restriction.Restriction.BspMII.size
Bio.Restriction.Restriction.BspNCI.size
Bio.Restriction.Restriction.BspOI.size
Bio.Restriction.Restriction.BspPI.size
Bio.Restriction.Restriction.BspQI.size
Bio.Restriction.Restriction.BspT104I.size
Bio.Restriction.Restriction.BspT107I.size
Bio.Restriction.Restriction.BspTI.size
Bio.Restriction.Restriction.BspTNI.size
Bio.Restriction.Restriction.BsrBI.size
Bio.Restriction.Restriction.BsrDI.size
Bio.Restriction.Restriction.BsrFI.size
Bio.Restriction.Restriction.BsrGI.size
Bio.Restriction.Restriction.BsrI.size
Bio.Restriction.Restriction.BssAI.size
Bio.Restriction.Restriction.BssECI.size
Bio.Restriction.Restriction.BssHII.size
Bio.Restriction.Restriction.BssMI.size
Bio.Restriction.Restriction.BssNAI.size
Bio.Restriction.Restriction.BssNI.size
Bio.Restriction.Restriction.BssSI.size
Bio.Restriction.Restriction.BssT1I.size
Bio.Restriction.Restriction.Bst1107I.size
Bio.Restriction.Restriction.Bst2BI.size
Bio.Restriction.Restriction.Bst2UI.size
Bio.Restriction.Restriction.Bst4CI.size
Bio.Restriction.Restriction.Bst6I.size
Bio.Restriction.Restriction.BstACI.size
Bio.Restriction.Restriction.BstAFI.size
Bio.Restriction.Restriction.BstAPI.size
Bio.Restriction.Restriction.BstAUI.size
Bio.Restriction.Restriction.BstBAI.size
Bio.Restriction.Restriction.BstBI.size
Bio.Restriction.Restriction.BstC8I.size
Bio.Restriction.Restriction.BstDEI.size
Bio.Restriction.Restriction.BstDSI.size
Bio.Restriction.Restriction.BstEII.size
Bio.Restriction.Restriction.BstENI.size
Bio.Restriction.Restriction.BstF5I.size
Bio.Restriction.Restriction.BstFNI.size
Bio.Restriction.Restriction.BstH2I.size
Bio.Restriction.Restriction.BstHHI.size
Bio.Restriction.Restriction.BstKTI.size
Bio.Restriction.Restriction.BstMAI.size
Bio.Restriction.Restriction.BstMBI.size
Bio.Restriction.Restriction.BstMCI.size
Bio.Restriction.Restriction.BstMWI.size
Bio.Restriction.Restriction.BstNI.size
Bio.Restriction.Restriction.BstNSI.size
Bio.Restriction.Restriction.BstPAI.size
Bio.Restriction.Restriction.BstPI.size
Bio.Restriction.Restriction.BstSCI.size
Bio.Restriction.Restriction.BstSFI.size
Bio.Restriction.Restriction.BstSLI.size
Bio.Restriction.Restriction.BstSNI.size
Bio.Restriction.Restriction.BstUI.size
Bio.Restriction.Restriction.BstV1I.size
Bio.Restriction.Restriction.BstV2I.size
Bio.Restriction.Restriction.BstX2I.size
Bio.Restriction.Restriction.BstXI.size
Bio.Restriction.Restriction.BstYI.size
Bio.Restriction.Restriction.BstZ17I.size
Bio.Restriction.Restriction.BstZI.size
Bio.Restriction.Restriction.Bsu15I.size
Bio.Restriction.Restriction.Bsu36I.size
Bio.Restriction.Restriction.BsuI.size
Bio.Restriction.Restriction.BsuRI.size
Bio.Restriction.Restriction.BsuTUI.size
Bio.Restriction.Restriction.BtgI.size
Bio.Restriction.Restriction.BtgZI.size
Bio.Restriction.Restriction.BthCI.size
Bio.Restriction.Restriction.BtrI.size
Bio.Restriction.Restriction.BtsCI.size
Bio.Restriction.Restriction.BtsI.size
Bio.Restriction.Restriction.BtsIMutI.size
Bio.Restriction.Restriction.BtuMI.size
Bio.Restriction.Restriction.BveI.size
Bio.Restriction.Restriction.Cac8I.size
Bio.Restriction.Restriction.CaiI.size
Bio.Restriction.Restriction.Cal14237I.size
Bio.Restriction.Restriction.CalB3II.size
Bio.Restriction.Restriction.Cau10061II.size
Bio.Restriction.Restriction.CauII.size
Bio.Restriction.Restriction.Cba13II.size
Bio.Restriction.Restriction.Cba16038I.size
Bio.Restriction.Restriction.Cbo67071IV.size
Bio.Restriction.Restriction.Cch467III.size
Bio.Restriction.Restriction.CchII.size
Bio.Restriction.Restriction.CchIII.size
Bio.Restriction.Restriction.CciI.size
Bio.Restriction.Restriction.CciNI.size
Bio.Restriction.Restriction.CcrNAIII.size
Bio.Restriction.Restriction.Cdi11397I.size
Bio.Restriction.Restriction.Cdi81III.size
Bio.Restriction.Restriction.CdiI.size
Bio.Restriction.Restriction.CdpI.size
Bio.Restriction.Restriction.Cdu23823II.size
Bio.Restriction.Restriction.CfoI.size
Bio.Restriction.Restriction.Cfr10I.size
Bio.Restriction.Restriction.Cfr13I.size
Bio.Restriction.Restriction.Cfr42I.size
Bio.Restriction.Restriction.Cfr9I.size
Bio.Restriction.Restriction.CfrI.size
Bio.Restriction.Restriction.Cfupf3II.size
Bio.Restriction.Restriction.Cgl13032I.size
Bio.Restriction.Restriction.Cgl13032II.size
Bio.Restriction.Restriction.ChaI.size
Bio.Restriction.Restriction.CjeFIII.size
Bio.Restriction.Restriction.CjeFV.size
Bio.Restriction.Restriction.CjeI.size
Bio.Restriction.Restriction.CjeNII.size
Bio.Restriction.Restriction.CjeNIII.size
Bio.Restriction.Restriction.CjeNV.size
Bio.Restriction.Restriction.CjeP659IV.size
Bio.Restriction.Restriction.CjePI.size
Bio.Restriction.Restriction.CjuI.size
Bio.Restriction.Restriction.CjuII.size
Bio.Restriction.Restriction.Cla11845III.size
Bio.Restriction.Restriction.ClaI.size
Bio.Restriction.Restriction.Cly7489II.size
Bio.Restriction.Restriction.Cma23826I.size
Bio.Restriction.Restriction.CpoI.size
Bio.Restriction.Restriction.CseI.size
Bio.Restriction.Restriction.CsiI.size
Bio.Restriction.Restriction.Csp2014I.size
Bio.Restriction.Restriction.Csp6I.size
Bio.Restriction.Restriction.CspAI.size
Bio.Restriction.Restriction.CspCI.size
Bio.Restriction.Restriction.CspI.size
Bio.Restriction.Restriction.CstMI.size
Bio.Restriction.Restriction.CviAII.size
Bio.Restriction.Restriction.CviJI.size
Bio.Restriction.Restriction.CviKI_1.size
Bio.Restriction.Restriction.CviQI.size
Bio.Restriction.Restriction.CviRI.size
Bio.Restriction.Restriction.Dde51507I.size
Bio.Restriction.Restriction.DdeI.size
Bio.Restriction.Restriction.DinI.size
Bio.Restriction.Restriction.DpnI.size
Bio.Restriction.Restriction.DpnII.size
Bio.Restriction.Restriction.DraI.size
Bio.Restriction.Restriction.DraII.size
Bio.Restriction.Restriction.DraIII.size
Bio.Restriction.Restriction.DraRI.size
Bio.Restriction.Restriction.DrdI.size
Bio.Restriction.Restriction.DrdII.size
Bio.Restriction.Restriction.DriI.size
Bio.Restriction.Restriction.DsaI.size
Bio.Restriction.Restriction.DseDI.size
Bio.Restriction.Restriction.EaeI.size
Bio.Restriction.Restriction.EagI.size
Bio.Restriction.Restriction.Eam1104I.size
Bio.Restriction.Restriction.Eam1105I.size
Bio.Restriction.Restriction.EarI.size
Bio.Restriction.Restriction.EciI.size
Bio.Restriction.Restriction.Ecl136II.size
Bio.Restriction.Restriction.Ecl35734I.size
Bio.Restriction.Restriction.EclXI.size
Bio.Restriction.Restriction.Eco105I.size
Bio.Restriction.Restriction.Eco130I.size
Bio.Restriction.Restriction.Eco147I.size
Bio.Restriction.Restriction.Eco24I.size
Bio.Restriction.Restriction.Eco31I.size
Bio.Restriction.Restriction.Eco32I.size
Bio.Restriction.Restriction.Eco4465II.size
Bio.Restriction.Restriction.Eco47I.size
Bio.Restriction.Restriction.Eco47III.size
Bio.Restriction.Restriction.Eco52I.size
Bio.Restriction.Restriction.Eco53kI.size
Bio.Restriction.Restriction.Eco57I.size
Bio.Restriction.Restriction.Eco57MI.size
Bio.Restriction.Restriction.Eco72I.size
Bio.Restriction.Restriction.Eco81I.size
Bio.Restriction.Restriction.Eco88I.size
Bio.Restriction.Restriction.Eco91I.size
Bio.Restriction.Restriction.EcoBLMcrX.size
Bio.Restriction.Restriction.EcoE1140I.size
Bio.Restriction.Restriction.EcoHI.size
Bio.Restriction.Restriction.EcoICRI.size
Bio.Restriction.Restriction.EcoMVII.size
Bio.Restriction.Restriction.EcoNI.size
Bio.Restriction.Restriction.EcoO109I.size
Bio.Restriction.Restriction.EcoO65I.size
Bio.Restriction.Restriction.EcoRI.size
Bio.Restriction.Restriction.EcoRII.size
Bio.Restriction.Restriction.EcoRV.size
Bio.Restriction.Restriction.EcoT14I.size
Bio.Restriction.Restriction.EcoT22I.size
Bio.Restriction.Restriction.EcoT38I.size
Bio.Restriction.Restriction.EgeI.size
Bio.Restriction.Restriction.EheI.size
Bio.Restriction.Restriction.Eli8509II.size
Bio.Restriction.Restriction.ErhI.size
Bio.Restriction.Restriction.EsaBC3I.size
Bio.Restriction.Restriction.EsaSSI.size
Bio.Restriction.Restriction.Esp3007I.size
Bio.Restriction.Restriction.Esp3I.size
Bio.Restriction.Restriction.EspI.size
Bio.Restriction.Restriction.Exi27195I.size
Bio.Restriction.Restriction.FaeI.size
Bio.Restriction.Restriction.FaiI.size
Bio.Restriction.Restriction.FalI.size
Bio.Restriction.Restriction.FaqI.size
Bio.Restriction.Restriction.FatI.size
Bio.Restriction.Restriction.FauI.size
Bio.Restriction.Restriction.FauNDI.size
Bio.Restriction.Restriction.FbaI.size
Bio.Restriction.Restriction.FblI.size
Bio.Restriction.Restriction.FinI.size
Bio.Restriction.Restriction.FmuI.size
Bio.Restriction.Restriction.Fnu4HI.size
Bio.Restriction.Restriction.FnuDII.size
Bio.Restriction.Restriction.FokI.size
Bio.Restriction.Restriction.FriOI.size
Bio.Restriction.Restriction.FseI.size
Bio.Restriction.Restriction.Fsp4HI.size
Bio.Restriction.Restriction.FspAI.size
Bio.Restriction.Restriction.FspBI.size
Bio.Restriction.Restriction.FspEI.size
Bio.Restriction.Restriction.FspI.size
Bio.Restriction.Restriction.FtnUV.size
Bio.Restriction.Restriction.GauT27I.size
Bio.Restriction.Restriction.Gba708II.size
Bio.Restriction.Restriction.GdiII.size
Bio.Restriction.Restriction.GlaI.size
Bio.Restriction.Restriction.GluI.size
Bio.Restriction.Restriction.GsaI.size
Bio.Restriction.Restriction.GsuI.size
Bio.Restriction.Restriction.HaeI.size
Bio.Restriction.Restriction.HaeII.size
Bio.Restriction.Restriction.HaeIII.size
Bio.Restriction.Restriction.HapII.size
Bio.Restriction.Restriction.HauII.size
Bio.Restriction.Restriction.HbaII.size
Bio.Restriction.Restriction.HgaI.size
Bio.Restriction.Restriction.HgiAI.size
Bio.Restriction.Restriction.HgiCI.size
Bio.Restriction.Restriction.HgiEII.size
Bio.Restriction.Restriction.HgiJII.size
Bio.Restriction.Restriction.HhaI.size
Bio.Restriction.Restriction.Hin1I.size
Bio.Restriction.Restriction.Hin1II.size
Bio.Restriction.Restriction.Hin4I.size
Bio.Restriction.Restriction.Hin4II.size
Bio.Restriction.Restriction.Hin6I.size
Bio.Restriction.Restriction.HinP1I.size
Bio.Restriction.Restriction.HincII.size
Bio.Restriction.Restriction.HindII.size
Bio.Restriction.Restriction.HindIII.size
Bio.Restriction.Restriction.HinfI.size
Bio.Restriction.Restriction.HpaI.size
Bio.Restriction.Restriction.HpaII.size
Bio.Restriction.Restriction.HphI.size
Bio.Restriction.Restriction.Hpy166II.size
Bio.Restriction.Restriction.Hpy178III.size
Bio.Restriction.Restriction.Hpy188I.size
Bio.Restriction.Restriction.Hpy188III.size
Bio.Restriction.Restriction.Hpy8I.size
Bio.Restriction.Restriction.Hpy99I.size
Bio.Restriction.Restriction.Hpy99XIII.size
Bio.Restriction.Restriction.Hpy99XIV.size
Bio.Restriction.Restriction.Hpy99XIV_mut1.size
Bio.Restriction.Restriction.Hpy99XXII.size
Bio.Restriction.Restriction.HpyAV.size
Bio.Restriction.Restriction.HpyAXIV.size
Bio.Restriction.Restriction.HpyAXVI_mut1.size
Bio.Restriction.Restriction.HpyAXVI_mut2.size
Bio.Restriction.Restriction.HpyCH4III.size
Bio.Restriction.Restriction.HpyCH4IV.size
Bio.Restriction.Restriction.HpyCH4V.size
Bio.Restriction.Restriction.HpyF10VI.size
Bio.Restriction.Restriction.HpyF3I.size
Bio.Restriction.Restriction.HpySE526I.size
Bio.Restriction.Restriction.HpyUM032XIII.size
Bio.Restriction.Restriction.HpyUM032XIII_mut1.size
Bio.Restriction.Restriction.HpyUM032XIV.size
Bio.Restriction.Restriction.HpyUM037X.size
Bio.Restriction.Restriction.Hsp92I.size
Bio.Restriction.Restriction.Hsp92II.size
Bio.Restriction.Restriction.HspAI.size
Bio.Restriction.Restriction.Jma19592I.size
Bio.Restriction.Restriction.Jma19592II.size
Bio.Restriction.Restriction.Jsp2502II.size
Bio.Restriction.Restriction.KasI.size
Bio.Restriction.Restriction.KflI.size
Bio.Restriction.Restriction.Kor51II.size
Bio.Restriction.Restriction.Kpn2I.size
Bio.Restriction.Restriction.KpnI.size
Bio.Restriction.Restriction.KpnNIH30III.size
Bio.Restriction.Restriction.KroI.size
Bio.Restriction.Restriction.Ksp22I.size
Bio.Restriction.Restriction.Ksp632I.size
Bio.Restriction.Restriction.KspAI.size
Bio.Restriction.Restriction.KspI.size
Bio.Restriction.Restriction.Kzo9I.size
Bio.Restriction.Restriction.Lba2029III.size
Bio.Restriction.Restriction.LguI.size
Bio.Restriction.Restriction.LmnI.size
Bio.Restriction.Restriction.Lmo370I.size
Bio.Restriction.Restriction.Lmo911II.size
Bio.Restriction.Restriction.LpnI.size
Bio.Restriction.Restriction.LpnPI.size
Bio.Restriction.Restriction.Lsp1109I.size
Bio.Restriction.Restriction.Lsp48III.size
Bio.Restriction.Restriction.Lsp6406VI.size
Bio.Restriction.Restriction.LweI.size
Bio.Restriction.Restriction.MabI.size
Bio.Restriction.Restriction.MaeI.size
Bio.Restriction.Restriction.MaeII.size
Bio.Restriction.Restriction.MaeIII.size
Bio.Restriction.Restriction.MalI.size
Bio.Restriction.Restriction.MaqI.size
Bio.Restriction.Restriction.MauBI.size
Bio.Restriction.Restriction.Mba11I.size
Bio.Restriction.Restriction.MbiI.size
Bio.Restriction.Restriction.MboI.size
Bio.Restriction.Restriction.MboII.size
Bio.Restriction.Restriction.McaTI.size
Bio.Restriction.Restriction.Mcr10I.size
Bio.Restriction.Restriction.McrI.size
Bio.Restriction.Restriction.MfeI.size
Bio.Restriction.Restriction.MflI.size
Bio.Restriction.Restriction.MhlI.size
Bio.Restriction.Restriction.MjaIV.size
Bio.Restriction.Restriction.MkaDII.size
Bio.Restriction.Restriction.MlsI.size
Bio.Restriction.Restriction.MluCI.size
Bio.Restriction.Restriction.MluI.size
Bio.Restriction.Restriction.MluNI.size
Bio.Restriction.Restriction.Mly113I.size
Bio.Restriction.Restriction.MlyI.size
Bio.Restriction.Restriction.MmeI.size
Bio.Restriction.Restriction.MnlI.size
Bio.Restriction.Restriction.Mox20I.size
Bio.Restriction.Restriction.Mph1103I.size
Bio.Restriction.Restriction.MreI.size
Bio.Restriction.Restriction.MroI.size
Bio.Restriction.Restriction.MroNI.size
Bio.Restriction.Restriction.MroXI.size
Bio.Restriction.Restriction.MscI.size
Bio.Restriction.Restriction.MseI.size
Bio.Restriction.Restriction.MslI.size
Bio.Restriction.Restriction.Msp20I.size
Bio.Restriction.Restriction.MspA1I.size
Bio.Restriction.Restriction.MspCI.size
Bio.Restriction.Restriction.MspI.size
Bio.Restriction.Restriction.MspI7II.size
Bio.Restriction.Restriction.MspJI.size
Bio.Restriction.Restriction.MspR9I.size
Bio.Restriction.Restriction.MssI.size
Bio.Restriction.Restriction.MstI.size
Bio.Restriction.Restriction.MteI.size
Bio.Restriction.Restriction.MtuHN878II.size
Bio.Restriction.Restriction.MunI.size
Bio.Restriction.Restriction.Mva1269I.size
Bio.Restriction.Restriction.MvaI.size
Bio.Restriction.Restriction.MvnI.size
Bio.Restriction.Restriction.MwoI.size
Bio.Restriction.Restriction.NaeI.size
Bio.Restriction.Restriction.Nal45188II.size
Bio.Restriction.Restriction.NarI.size
Bio.Restriction.Restriction.Nbr128II.size
Bio.Restriction.Restriction.NciI.size
Bio.Restriction.Restriction.NcoI.size
Bio.Restriction.Restriction.NdeI.size
Bio.Restriction.Restriction.NdeII.size
Bio.Restriction.Restriction.NgoAVII.size
Bio.Restriction.Restriction.NgoAVIII.size
Bio.Restriction.Restriction.NgoMIV.size
Bio.Restriction.Restriction.NhaXI.size
Bio.Restriction.Restriction.NheI.size
Bio.Restriction.Restriction.NhoI.size
Bio.Restriction.Restriction.NlaCI.size
Bio.Restriction.Restriction.NlaIII.size
Bio.Restriction.Restriction.NlaIV.size
Bio.Restriction.Restriction.Nli3877I.size
Bio.Restriction.Restriction.NmeAIII.size
Bio.Restriction.Restriction.NmeDI.size
Bio.Restriction.Restriction.NmuCI.size
Bio.Restriction.Restriction.NotI.size
Bio.Restriction.Restriction.NpeUS61II.size
Bio.Restriction.Restriction.NruI.size
Bio.Restriction.Restriction.NsbI.size
Bio.Restriction.Restriction.NsiI.size
Bio.Restriction.Restriction.NspBII.size
Bio.Restriction.Restriction.NspI.size
Bio.Restriction.Restriction.NspV.size
Bio.Restriction.Restriction.OliI.size
Bio.Restriction.Restriction.OspHL35III.size
Bio.Restriction.Restriction.PabI.size
Bio.Restriction.Restriction.Pac19842II.size
Bio.Restriction.Restriction.PacI.size
Bio.Restriction.Restriction.PacIII.size
Bio.Restriction.Restriction.PaeI.size
Bio.Restriction.Restriction.PaeR7I.size
Bio.Restriction.Restriction.PagI.size
Bio.Restriction.Restriction.Pal408I.size
Bio.Restriction.Restriction.PalAI.size
Bio.Restriction.Restriction.PasI.size
Bio.Restriction.Restriction.PauI.size
Bio.Restriction.Restriction.PcaII.size
Bio.Restriction.Restriction.PceI.size
Bio.Restriction.Restriction.PciI.size
Bio.Restriction.Restriction.PciSI.size
Bio.Restriction.Restriction.PcsI.size
Bio.Restriction.Restriction.PctI.size
Bio.Restriction.Restriction.Pdi8503III.size
Bio.Restriction.Restriction.PdiI.size
Bio.Restriction.Restriction.PdmI.size
Bio.Restriction.Restriction.Pdu1735I.size
Bio.Restriction.Restriction.PenI.size
Bio.Restriction.Restriction.PfeI.size
Bio.Restriction.Restriction.Pfl1108I.size
Bio.Restriction.Restriction.Pfl23II.size
Bio.Restriction.Restriction.Pfl8569I.size
Bio.Restriction.Restriction.PflFI.size
Bio.Restriction.Restriction.PflMI.size
Bio.Restriction.Restriction.PfoI.size
Bio.Restriction.Restriction.PfrJS12IV.size
Bio.Restriction.Restriction.PfrJS12V.size
Bio.Restriction.Restriction.PfrJS15III.size
Bio.Restriction.Restriction.Pin17FIII.size
Bio.Restriction.Restriction.PinAI.size
Bio.Restriction.Restriction.PkrI.size
Bio.Restriction.Restriction.PlaDI.size
Bio.Restriction.Restriction.Ple19I.size
Bio.Restriction.Restriction.PleI.size
Bio.Restriction.Restriction.PliMI.size
Bio.Restriction.Restriction.PluTI.size
Bio.Restriction.Restriction.PmaCI.size
Bio.Restriction.Restriction.PmeI.size
Bio.Restriction.Restriction.PmlI.size
Bio.Restriction.Restriction.PpiI.size
Bio.Restriction.Restriction.PpsI.size
Bio.Restriction.Restriction.Ppu10I.size
Bio.Restriction.Restriction.Ppu21I.size
Bio.Restriction.Restriction.PpuMI.size
Bio.Restriction.Restriction.PscI.size
Bio.Restriction.Restriction.Pse18267I.size
Bio.Restriction.Restriction.PshAI.size
Bio.Restriction.Restriction.PshBI.size
Bio.Restriction.Restriction.PsiI.size
Bio.Restriction.Restriction.Psp0357II.size
Bio.Restriction.Restriction.Psp03I.size
Bio.Restriction.Restriction.Psp124BI.size
Bio.Restriction.Restriction.Psp1406I.size
Bio.Restriction.Restriction.Psp5II.size
Bio.Restriction.Restriction.Psp6I.size
Bio.Restriction.Restriction.PspCI.size
Bio.Restriction.Restriction.PspEI.size
Bio.Restriction.Restriction.PspFI.size
Bio.Restriction.Restriction.PspGI.size
Bio.Restriction.Restriction.PspLI.size
Bio.Restriction.Restriction.PspN4I.size
Bio.Restriction.Restriction.PspOMI.size
Bio.Restriction.Restriction.PspOMII.size
Bio.Restriction.Restriction.PspPI.size
Bio.Restriction.Restriction.PspPPI.size
Bio.Restriction.Restriction.PspPRI.size
Bio.Restriction.Restriction.PspXI.size
Bio.Restriction.Restriction.PsrI.size
Bio.Restriction.Restriction.PssI.size
Bio.Restriction.Restriction.Pst14472I.size
Bio.Restriction.Restriction.Pst273I.size
Bio.Restriction.Restriction.PstI.size
Bio.Restriction.Restriction.PstNI.size
Bio.Restriction.Restriction.PsuGI.size
Bio.Restriction.Restriction.PsuI.size
Bio.Restriction.Restriction.PsyI.size
Bio.Restriction.Restriction.PteI.size
Bio.Restriction.Restriction.PvuI.size
Bio.Restriction.Restriction.PvuII.size
Bio.Restriction.Restriction.Rba2021I.size
Bio.Restriction.Restriction.RceI.size
Bio.Restriction.Restriction.RdeGBI.size
Bio.Restriction.Restriction.RdeGBII.size
Bio.Restriction.Restriction.RdeGBIII.size
Bio.Restriction.Restriction.RflFIII.size
Bio.Restriction.Restriction.RgaI.size
Bio.Restriction.Restriction.RigI.size
Bio.Restriction.Restriction.RlaI.size
Bio.Restriction.Restriction.RlaII.size
Bio.Restriction.Restriction.RleAI.size
Bio.Restriction.Restriction.RpaB5I.size
Bio.Restriction.Restriction.RpaBI.size
Bio.Restriction.Restriction.RpaI.size
Bio.Restriction.Restriction.RpaTI.size
Bio.Restriction.Restriction.RruI.size
Bio.Restriction.Restriction.RsaI.size
Bio.Restriction.Restriction.RsaNI.size
Bio.Restriction.Restriction.RseI.size
Bio.Restriction.Restriction.Rsp008IV.size
Bio.Restriction.Restriction.Rsp008V.size
Bio.Restriction.Restriction.RspPBTS2III.size
Bio.Restriction.Restriction.Rsr2I.size
Bio.Restriction.Restriction.RsrII.size
Bio.Restriction.Restriction.SacI.size
Bio.Restriction.Restriction.SacII.size
Bio.Restriction.Restriction.Saf8902III.size
Bio.Restriction.Restriction.SalI.size
Bio.Restriction.Restriction.SanDI.size
Bio.Restriction.Restriction.SapI.size
Bio.Restriction.Restriction.SaqAI.size
Bio.Restriction.Restriction.SatI.size
Bio.Restriction.Restriction.Sau3AI.size
Bio.Restriction.Restriction.Sau96I.size
Bio.Restriction.Restriction.SauI.size
Bio.Restriction.Restriction.SbfI.size
Bio.Restriction.Restriction.Sbo46I.size
Bio.Restriction.Restriction.ScaI.size
Bio.Restriction.Restriction.SchI.size
Bio.Restriction.Restriction.SciI.size
Bio.Restriction.Restriction.ScoDS2II.size
Bio.Restriction.Restriction.ScrFI.size
Bio.Restriction.Restriction.SdaI.size
Bio.Restriction.Restriction.SdeAI.size
Bio.Restriction.Restriction.SdeOSI.size
Bio.Restriction.Restriction.SduI.size
Bio.Restriction.Restriction.SecI.size
Bio.Restriction.Restriction.SelI.size
Bio.Restriction.Restriction.Sen13311III.size
Bio.Restriction.Restriction.Sen13III.size
Bio.Restriction.Restriction.Sen1736II.size
Bio.Restriction.Restriction.Sen1878IV.size
Bio.Restriction.Restriction.Sen1880IV.size
Bio.Restriction.Restriction.Sen1896IV.size
Bio.Restriction.Restriction.Sen1898IV.size
Bio.Restriction.Restriction.Sen1906IV.size
Bio.Restriction.Restriction.Sen1908IV.size
Bio.Restriction.Restriction.Sen1910IV.size
Bio.Restriction.Restriction.SenC1808III.size
Bio.Restriction.Restriction.SenC1810IV.size
Bio.Restriction.Restriction.SenSARA26III.size
Bio.Restriction.Restriction.SenTFIV.size
Bio.Restriction.Restriction.SetI.size
Bio.Restriction.Restriction.SexAI.size
Bio.Restriction.Restriction.SfaAI.size
Bio.Restriction.Restriction.SfaNI.size
Bio.Restriction.Restriction.SfcI.size
Bio.Restriction.Restriction.SfeI.size
Bio.Restriction.Restriction.SfiI.size
Bio.Restriction.Restriction.SfoI.size
Bio.Restriction.Restriction.Sfr274I.size
Bio.Restriction.Restriction.Sfr303I.size
Bio.Restriction.Restriction.SfuI.size
Bio.Restriction.Restriction.SgeI.size
Bio.Restriction.Restriction.SgfI.size
Bio.Restriction.Restriction.SgrAI.size
Bio.Restriction.Restriction.SgrBI.size
Bio.Restriction.Restriction.SgrDI.size
Bio.Restriction.Restriction.SgrTI.size
Bio.Restriction.Restriction.SgsI.size
Bio.Restriction.Restriction.SimI.size
Bio.Restriction.Restriction.SinI.size
Bio.Restriction.Restriction.SlaI.size
Bio.Restriction.Restriction.SmaI.size
Bio.Restriction.Restriction.SmiI.size
Bio.Restriction.Restriction.SmiMI.size
Bio.Restriction.Restriction.SmlI.size
Bio.Restriction.Restriction.SmoI.size
Bio.Restriction.Restriction.SnaBI.size
Bio.Restriction.Restriction.SnaI.size
Bio.Restriction.Restriction.Sno506I.size
Bio.Restriction.Restriction.SpeI.size
Bio.Restriction.Restriction.SphI.size
Bio.Restriction.Restriction.SplI.size
Bio.Restriction.Restriction.SpnRII.size
Bio.Restriction.Restriction.SpoDI.size
Bio.Restriction.Restriction.SrfI.size
Bio.Restriction.Restriction.Sse232I.size
Bio.Restriction.Restriction.Sse8387I.size
Bio.Restriction.Restriction.Sse8647I.size
Bio.Restriction.Restriction.Sse9I.size
Bio.Restriction.Restriction.SseBI.size
Bio.Restriction.Restriction.SsiI.size
Bio.Restriction.Restriction.Ssp6803IV.size
Bio.Restriction.Restriction.Ssp714II.size
Bio.Restriction.Restriction.SspD5I.size
Bio.Restriction.Restriction.SspDI.size
Bio.Restriction.Restriction.SspI.size
Bio.Restriction.Restriction.SspMI.size
Bio.Restriction.Restriction.SstE37I.size
Bio.Restriction.Restriction.SstI.size
Bio.Restriction.Restriction.Sth132I.size
Bio.Restriction.Restriction.Sth20745III.size
Bio.Restriction.Restriction.Sth302II.size
Bio.Restriction.Restriction.StsI.size
Bio.Restriction.Restriction.StuI.size
Bio.Restriction.Restriction.Sty13348III.size
Bio.Restriction.Restriction.StyD4I.size
Bio.Restriction.Restriction.StyI.size
Bio.Restriction.Restriction.SwaI.size
Bio.Restriction.Restriction.TaaI.size
Bio.Restriction.Restriction.TaiI.size
Bio.Restriction.Restriction.TaqI.size
Bio.Restriction.Restriction.TaqII.size
Bio.Restriction.Restriction.TaqIII.size
Bio.Restriction.Restriction.TasI.size
Bio.Restriction.Restriction.TatI.size
Bio.Restriction.Restriction.TauI.size
Bio.Restriction.Restriction.TfiI.size
Bio.Restriction.Restriction.Tru1I.size
Bio.Restriction.Restriction.Tru9I.size
Bio.Restriction.Restriction.TscAI.size
Bio.Restriction.Restriction.TseFI.size
Bio.Restriction.Restriction.TseI.size
Bio.Restriction.Restriction.TsoI.size
Bio.Restriction.Restriction.Tsp45I.size
Bio.Restriction.Restriction.Tsp4CI.size
Bio.Restriction.Restriction.TspARh3I.size
Bio.Restriction.Restriction.TspDTI.size
Bio.Restriction.Restriction.TspEI.size
Bio.Restriction.Restriction.TspGWI.size
Bio.Restriction.Restriction.TspMI.size
Bio.Restriction.Restriction.TspRI.size
Bio.Restriction.Restriction.TssI.size
Bio.Restriction.Restriction.TstI.size
Bio.Restriction.Restriction.TsuI.size
Bio.Restriction.Restriction.Tth111I.size
Bio.Restriction.Restriction.Tth111II.size
Bio.Restriction.Restriction.UbaF11I.size
Bio.Restriction.Restriction.UbaF12I.size
Bio.Restriction.Restriction.UbaF13I.size
Bio.Restriction.Restriction.UbaF14I.size
Bio.Restriction.Restriction.UbaF9I.size
Bio.Restriction.Restriction.UbaPI.size
Bio.Restriction.Restriction.UcoMSI.size
Bio.Restriction.Restriction.UnbI.size
Bio.Restriction.Restriction.Van91I.size
Bio.Restriction.Restriction.Vha464I.size
Bio.Restriction.Restriction.VneI.size
Bio.Restriction.Restriction.VpaK11AI.size
Bio.Restriction.Restriction.VpaK11BI.size
Bio.Restriction.Restriction.VspI.size
Bio.Restriction.Restriction.Vtu19109I.size
Bio.Restriction.Restriction.WviI.size
Bio.Restriction.Restriction.XagI.size
Bio.Restriction.Restriction.XapI.size
Bio.Restriction.Restriction.XbaI.size
Bio.Restriction.Restriction.Xca85IV.size
Bio.Restriction.Restriction.XceI.size
Bio.Restriction.Restriction.XcmI.size
Bio.Restriction.Restriction.XhoI.size
Bio.Restriction.Restriction.XhoII.size
Bio.Restriction.Restriction.XmaI.size
Bio.Restriction.Restriction.XmaIII.size
Bio.Restriction.Restriction.XmaJI.size
Bio.Restriction.Restriction.XmiI.size
Bio.Restriction.Restriction.XmnI.size
Bio.Restriction.Restriction.XspI.size
Bio.Restriction.Restriction.YkrI.size
Bio.Restriction.Restriction.Yps3606I.size
Bio.Restriction.Restriction.ZraI.size
Bio.Restriction.Restriction.ZrmI.size
Bio.Restriction.Restriction.Zsp2I.size
Bio.codonalign.codonalphabet.CodonAlphabet.size" class="py-name" href="#" onclick="return doclink('link-137', 'size', 'link-132');">size</a></tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt> </tt>
<a name="L419"></a><tt class="py-lineno">419</tt>  <tt class="py-line"> </tt>
<a name="L420"></a><tt class="py-lineno">420</tt>  <tt class="py-line">                            <tt class="py-comment"># _ucscbin takes end-exclusive coordinates</tt> </tt>
<a name="L421"></a><tt class="py-lineno">421</tt>  <tt class="py-line">                            <tt class="py-keyword">yield</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-138" class="py-name" targets="Static Method Bio.AlignIO.MafIO.MafIndex._ucscbin()=Bio.AlignIO.MafIO.MafIndex-class.html#_ucscbin"><a title="Bio.AlignIO.MafIO.MafIndex._ucscbin" class="py-name" href="#" onclick="return doclink('link-138', '_ucscbin', 'link-138');">_ucscbin</a></tt><tt class="py-op">(</tt><tt id="link-139" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-139', 'start', 'link-131');">start</a></tt><tt class="py-op">,</tt> <tt id="link-140" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-140', 'end', 'link-135');">end</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-141" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-141', 'start', 'link-131');">start</a></tt><tt class="py-op">,</tt> <tt id="link-142" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-142', 'end', 'link-135');">end</a></tt><tt class="py-op">,</tt> <tt class="py-name">offset</tt><tt class="py-op">)</tt> </tt>
<a name="L422"></a><tt class="py-lineno">422</tt>  <tt class="py-line"> </tt>
<a name="L423"></a><tt class="py-lineno">423</tt>  <tt class="py-line">                            <tt class="py-keyword">break</tt> </tt>
<a name="L424"></a><tt class="py-lineno">424</tt>  <tt class="py-line"> </tt>
<a name="L425"></a><tt class="py-lineno">425</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_fp</tt><tt class="py-op">.</tt><tt id="link-143" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-143', 'readline', 'link-122');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L426"></a><tt class="py-lineno">426</tt>  <tt class="py-line"> </tt>
<a name="L427"></a><tt class="py-lineno">427</tt>  <tt class="py-line">    <tt class="py-comment"># TODO: check coordinate correctness for the two bin-related static methods</tt> </tt>
<a name="L428"></a><tt class="py-lineno">428</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">staticmethod</tt> </tt>
<a name="MafIndex._region2bin"></a><div id="MafIndex._region2bin-def"><a name="L429"></a><tt class="py-lineno">429</tt> <a class="py-toggle" href="#" id="MafIndex._region2bin-toggle" onclick="return toggle('MafIndex._region2bin');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafIndex-class.html#_region2bin">_region2bin</a><tt class="py-op">(</tt><tt class="py-param">start</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIndex._region2bin-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafIndex._region2bin-expanded"><a name="L430"></a><tt class="py-lineno">430</tt>  <tt class="py-line">        <tt class="py-docstring">"""Find bins that a region may belong to (PRIVATE).</tt> </tt>
<a name="L431"></a><tt class="py-lineno">431</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L432"></a><tt class="py-lineno">432</tt>  <tt class="py-line"><tt class="py-docstring">        Converts a region to a list of bins that it may belong to, including largest</tt> </tt>
<a name="L433"></a><tt class="py-lineno">433</tt>  <tt class="py-line"><tt class="py-docstring">        and smallest bins.</tt> </tt>
<a name="L434"></a><tt class="py-lineno">434</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L435"></a><tt class="py-lineno">435</tt>  <tt class="py-line">        <tt class="py-name">bins</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L436"></a><tt class="py-lineno">436</tt>  <tt class="py-line"> </tt>
<a name="L437"></a><tt class="py-lineno">437</tt>  <tt class="py-line">        <tt class="py-name">bins</tt><tt class="py-op">.</tt><tt id="link-144" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.extend()=Bio.Align.MultipleSeqAlignment-class.html#extend,Method Bio.SCOP.Raf.SeqMap.extend()=Bio.SCOP.Raf.SeqMap-class.html#extend,Method Bio.Seq.MutableSeq.extend()=Bio.Seq.MutableSeq-class.html#extend"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-144', 'extend', 'link-144');">extend</a></tt><tt class="py-op">(</tt><tt id="link-145" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-145', 'range', 'link-145');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">1</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt id="link-146" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-146', 'start', 'link-131');">start</a></tt> <tt class="py-op">&gt;&gt;</tt> <tt class="py-number">26</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">2</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt class="py-op">(</tt><tt id="link-147" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-147', 'end', 'link-135');">end</a></tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;&gt;</tt> <tt class="py-number">26</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L438"></a><tt class="py-lineno">438</tt>  <tt class="py-line">        <tt class="py-name">bins</tt><tt class="py-op">.</tt><tt id="link-148" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-148', 'extend', 'link-144');">extend</a></tt><tt class="py-op">(</tt><tt id="link-149" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-149', 'range', 'link-145');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">9</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt id="link-150" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-150', 'start', 'link-131');">start</a></tt> <tt class="py-op">&gt;&gt;</tt> <tt class="py-number">23</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">10</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt class="py-op">(</tt><tt id="link-151" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-151', 'end', 'link-135');">end</a></tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;&gt;</tt> <tt class="py-number">23</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L439"></a><tt class="py-lineno">439</tt>  <tt class="py-line">        <tt class="py-name">bins</tt><tt class="py-op">.</tt><tt id="link-152" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-152', 'extend', 'link-144');">extend</a></tt><tt class="py-op">(</tt><tt id="link-153" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-153', 'range', 'link-145');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">73</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt id="link-154" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-154', 'start', 'link-131');">start</a></tt> <tt class="py-op">&gt;&gt;</tt> <tt class="py-number">20</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">74</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt class="py-op">(</tt><tt id="link-155" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-155', 'end', 'link-135');">end</a></tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;&gt;</tt> <tt class="py-number">20</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L440"></a><tt class="py-lineno">440</tt>  <tt class="py-line">        <tt class="py-name">bins</tt><tt class="py-op">.</tt><tt id="link-156" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-156', 'extend', 'link-144');">extend</a></tt><tt class="py-op">(</tt><tt id="link-157" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-157', 'range', 'link-145');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">585</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt id="link-158" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-158', 'start', 'link-131');">start</a></tt> <tt class="py-op">&gt;&gt;</tt> <tt class="py-number">17</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">586</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt class="py-op">(</tt><tt id="link-159" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-159', 'end', 'link-135');">end</a></tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;&gt;</tt> <tt class="py-number">17</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L441"></a><tt class="py-lineno">441</tt>  <tt class="py-line"> </tt>
<a name="L442"></a><tt class="py-lineno">442</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-160" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-160', 'set', 'link-36');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">bins</tt><tt class="py-op">)</tt> </tt>
</div><a name="L443"></a><tt class="py-lineno">443</tt>  <tt class="py-line"> </tt>
<a name="L444"></a><tt class="py-lineno">444</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">staticmethod</tt> </tt>
<a name="MafIndex._ucscbin"></a><div id="MafIndex._ucscbin-def"><a name="L445"></a><tt class="py-lineno">445</tt> <a class="py-toggle" href="#" id="MafIndex._ucscbin-toggle" onclick="return toggle('MafIndex._ucscbin');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafIndex-class.html#_ucscbin">_ucscbin</a><tt class="py-op">(</tt><tt class="py-param">start</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIndex._ucscbin-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafIndex._ucscbin-expanded"><a name="L446"></a><tt class="py-lineno">446</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the smallest bin a given region will fit into (PRIVATE).</tt> </tt>
<a name="L447"></a><tt class="py-lineno">447</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L448"></a><tt class="py-lineno">448</tt>  <tt class="py-line"><tt class="py-docstring">        Adapted from http://genomewiki.ucsc.edu/index.php/Bin_indexing_system</tt> </tt>
<a name="L449"></a><tt class="py-lineno">449</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L450"></a><tt class="py-lineno">450</tt>  <tt class="py-line">        <tt class="py-name">bin_offsets</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-number">512</tt> <tt class="py-op">+</tt> <tt class="py-number">64</tt> <tt class="py-op">+</tt> <tt class="py-number">8</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">64</tt> <tt class="py-op">+</tt> <tt class="py-number">8</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">8</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L451"></a><tt class="py-lineno">451</tt>  <tt class="py-line"> </tt>
<a name="L452"></a><tt class="py-lineno">452</tt>  <tt class="py-line">        <tt class="py-name">_bin_first_shift</tt> <tt class="py-op">=</tt> <tt class="py-number">17</tt> </tt>
<a name="L453"></a><tt class="py-lineno">453</tt>  <tt class="py-line">        <tt class="py-name">_bin_next_shift</tt> <tt class="py-op">=</tt> <tt class="py-number">3</tt> </tt>
<a name="L454"></a><tt class="py-lineno">454</tt>  <tt class="py-line"> </tt>
<a name="L455"></a><tt class="py-lineno">455</tt>  <tt class="py-line">        <tt class="py-name">start_bin</tt> <tt class="py-op">=</tt> <tt id="link-161" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-161', 'start', 'link-131');">start</a></tt> </tt>
<a name="L456"></a><tt class="py-lineno">456</tt>  <tt class="py-line">        <tt class="py-name">end_bin</tt> <tt class="py-op">=</tt> <tt id="link-162" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-162', 'end', 'link-135');">end</a></tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt> </tt>
<a name="L457"></a><tt class="py-lineno">457</tt>  <tt class="py-line"> </tt>
<a name="L458"></a><tt class="py-lineno">458</tt>  <tt class="py-line">        <tt class="py-name">start_bin</tt> <tt class="py-op">&gt;&gt;=</tt> <tt class="py-name">_bin_first_shift</tt> </tt>
<a name="L459"></a><tt class="py-lineno">459</tt>  <tt class="py-line">        <tt class="py-name">end_bin</tt> <tt class="py-op">&gt;&gt;=</tt> <tt class="py-name">_bin_first_shift</tt> </tt>
<a name="L460"></a><tt class="py-lineno">460</tt>  <tt class="py-line"> </tt>
<a name="L461"></a><tt class="py-lineno">461</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">bin_offset</tt> <tt class="py-keyword">in</tt> <tt class="py-name">bin_offsets</tt><tt class="py-op">:</tt> </tt>
<a name="L462"></a><tt class="py-lineno">462</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">start_bin</tt> <tt class="py-op">==</tt> <tt class="py-name">end_bin</tt><tt class="py-op">:</tt> </tt>
<a name="L463"></a><tt class="py-lineno">463</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">bin_offset</tt> <tt class="py-op">+</tt> <tt class="py-name">start_bin</tt> </tt>
<a name="L464"></a><tt class="py-lineno">464</tt>  <tt class="py-line">            <tt class="py-name">start_bin</tt> <tt class="py-op">&gt;&gt;=</tt> <tt class="py-name">_bin_next_shift</tt> </tt>
<a name="L465"></a><tt class="py-lineno">465</tt>  <tt class="py-line">            <tt class="py-name">end_bin</tt> <tt class="py-op">&gt;&gt;=</tt> <tt class="py-name">_bin_next_shift</tt> </tt>
<a name="L466"></a><tt class="py-lineno">466</tt>  <tt class="py-line"> </tt>
<a name="L467"></a><tt class="py-lineno">467</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-number">0</tt> </tt>
</div><a name="L468"></a><tt class="py-lineno">468</tt>  <tt class="py-line"> </tt>
<a name="MafIndex._get_record"></a><div id="MafIndex._get_record-def"><a name="L469"></a><tt class="py-lineno">469</tt> <a class="py-toggle" href="#" id="MafIndex._get_record-toggle" onclick="return toggle('MafIndex._get_record');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafIndex-class.html#_get_record">_get_record</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">offset</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIndex._get_record-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafIndex._get_record-expanded"><a name="L470"></a><tt class="py-lineno">470</tt>  <tt class="py-line">        <tt class="py-docstring">"""Retrieve a single MAF record located at the offset provided (PRIVATE)."""</tt> </tt>
<a name="L471"></a><tt class="py-lineno">471</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_fp</tt><tt class="py-op">.</tt><tt id="link-163" class="py-name" targets="Method Bio.File.UndoHandle.seek()=Bio.File.UndoHandle-class.html#seek,Method Bio.SeqIO.SffIO._AddTellHandle.seek()=Bio.SeqIO.SffIO._AddTellHandle-class.html#seek,Method Bio.bgzf.BgzfReader.seek()=Bio.bgzf.BgzfReader-class.html#seek"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-163', 'seek', 'link-163');">seek</a></tt><tt class="py-op">(</tt><tt class="py-name">offset</tt><tt class="py-op">)</tt> </tt>
<a name="L472"></a><tt class="py-lineno">472</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-164" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.Nexus.StandardData.StandardData.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-164', 'next', 'link-50');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_mafiter</tt><tt class="py-op">)</tt> </tt>
</div><a name="L473"></a><tt class="py-lineno">473</tt>  <tt class="py-line"> </tt>
<a name="MafIndex.search"></a><div id="MafIndex.search-def"><a name="L474"></a><tt class="py-lineno">474</tt> <a class="py-toggle" href="#" id="MafIndex.search-toggle" onclick="return toggle('MafIndex.search');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafIndex-class.html#search">search</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">starts</tt><tt class="py-op">,</tt> <tt class="py-param">ends</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIndex.search-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafIndex.search-expanded"><a name="L475"></a><tt class="py-lineno">475</tt>  <tt class="py-line">        <tt class="py-docstring">"""Search index database for MAF records overlapping ranges provided.</tt> </tt>
<a name="L476"></a><tt class="py-lineno">476</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L477"></a><tt class="py-lineno">477</tt>  <tt class="py-line"><tt class="py-docstring">        Returns *MultipleSeqAlignment* results in order by start, then end, then</tt> </tt>
<a name="L478"></a><tt class="py-lineno">478</tt>  <tt class="py-line"><tt class="py-docstring">        internal offset field.</tt> </tt>
<a name="L479"></a><tt class="py-lineno">479</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L480"></a><tt class="py-lineno">480</tt>  <tt class="py-line"><tt class="py-docstring">        *starts* should be a list of 0-based start coordinates of segments in the reference.</tt> </tt>
<a name="L481"></a><tt class="py-lineno">481</tt>  <tt class="py-line"><tt class="py-docstring">        *ends* should be the list of the corresponding segment ends</tt> </tt>
<a name="L482"></a><tt class="py-lineno">482</tt>  <tt class="py-line"><tt class="py-docstring">        (in the half-open UCSC convention:</tt> </tt>
<a name="L483"></a><tt class="py-lineno">483</tt>  <tt class="py-line"><tt class="py-docstring">        http://genome.ucsc.edu/blog/the-ucsc-genome-browser-coordinate-counting-systems/).</tt> </tt>
<a name="L484"></a><tt class="py-lineno">484</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L485"></a><tt class="py-lineno">485</tt>  <tt class="py-line">        <tt class="py-comment"># verify the provided exon coordinates</tt> </tt>
<a name="L486"></a><tt class="py-lineno">486</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">starts</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">ends</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L487"></a><tt class="py-lineno">487</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Every position in starts must have a match in ends"</tt><tt class="py-op">)</tt> </tt>
<a name="L488"></a><tt class="py-lineno">488</tt>  <tt class="py-line"> </tt>
<a name="L489"></a><tt class="py-lineno">489</tt>  <tt class="py-line">        <tt class="py-comment"># Could it be safer to sort the (exonstart, exonend) pairs?</tt> </tt>
<a name="L490"></a><tt class="py-lineno">490</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">exonstart</tt><tt class="py-op">,</tt> <tt class="py-name">exonend</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">starts</tt><tt class="py-op">,</tt> <tt class="py-name">ends</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L491"></a><tt class="py-lineno">491</tt>  <tt class="py-line">            <tt class="py-name">exonlen</tt> <tt class="py-op">=</tt> <tt class="py-name">exonend</tt> <tt class="py-op">-</tt> <tt class="py-name">exonstart</tt> </tt>
<a name="L492"></a><tt class="py-lineno">492</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">exonlen</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L493"></a><tt class="py-lineno">493</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Exon coordinates (%d, %d) invalid: exon length (%d) &lt; 1"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt> </tt>
<a name="L494"></a><tt class="py-lineno">494</tt>  <tt class="py-line">                    <tt class="py-name">exonstart</tt><tt class="py-op">,</tt> <tt class="py-name">exonend</tt><tt class="py-op">,</tt> <tt class="py-name">exonlen</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L495"></a><tt class="py-lineno">495</tt>  <tt class="py-line">        <tt class="py-name">con</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_con</tt> </tt>
<a name="L496"></a><tt class="py-lineno">496</tt>  <tt class="py-line"> </tt>
<a name="L497"></a><tt class="py-lineno">497</tt>  <tt class="py-line">        <tt class="py-comment"># Keep track of what blocks have already been yielded</tt> </tt>
<a name="L498"></a><tt class="py-lineno">498</tt>  <tt class="py-line">        <tt class="py-comment"># in order to avoid duplicating them</tt> </tt>
<a name="L499"></a><tt class="py-lineno">499</tt>  <tt class="py-line">        <tt class="py-comment"># (see https://github.com/biopython/biopython/issues/1083)</tt> </tt>
<a name="L500"></a><tt class="py-lineno">500</tt>  <tt class="py-line">        <tt class="py-name">yielded_rec_coords</tt> <tt class="py-op">=</tt> <tt id="link-165" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-165', 'set', 'link-36');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L501"></a><tt class="py-lineno">501</tt>  <tt class="py-line">        <tt class="py-comment"># search for every exon</tt> </tt>
<a name="L502"></a><tt class="py-lineno">502</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">exonstart</tt><tt class="py-op">,</tt> <tt class="py-name">exonend</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">starts</tt><tt class="py-op">,</tt> <tt class="py-name">ends</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L503"></a><tt class="py-lineno">503</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L504"></a><tt class="py-lineno">504</tt>  <tt class="py-line">                <tt class="py-name">possible_bins</tt> <tt class="py-op">=</tt> <tt class="py-string">", "</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt id="link-166" class="py-name" targets="Method Bio.SearchIO._model.hit.Hit.map()=Bio.SearchIO._model.hit.Hit-class.html#map"><a title="Bio.SearchIO._model.hit.Hit.map" class="py-name" href="#" onclick="return doclink('link-166', 'map', 'link-166');">map</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-167" class="py-name" targets="Static Method Bio.AlignIO.MafIO.MafIndex._region2bin()=Bio.AlignIO.MafIO.MafIndex-class.html#_region2bin"><a title="Bio.AlignIO.MafIO.MafIndex._region2bin" class="py-name" href="#" onclick="return doclink('link-167', '_region2bin', 'link-167');">_region2bin</a></tt><tt class="py-op">(</tt><tt class="py-name">exonstart</tt><tt class="py-op">,</tt> <tt class="py-name">exonend</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L505"></a><tt class="py-lineno">505</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">TypeError</tt><tt class="py-op">:</tt> </tt>
<a name="L506"></a><tt class="py-lineno">506</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"Exon coordinates must be integers "</tt> </tt>
<a name="L507"></a><tt class="py-lineno">507</tt>  <tt class="py-line">                                <tt class="py-string">"(start=%d, end=%d)"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">exonstart</tt><tt class="py-op">,</tt> <tt class="py-name">exonend</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L508"></a><tt class="py-lineno">508</tt>  <tt class="py-line"> </tt>
<a name="L509"></a><tt class="py-lineno">509</tt>  <tt class="py-line">            <tt class="py-comment"># https://www.sqlite.org/lang_expr.html</tt> </tt>
<a name="L510"></a><tt class="py-lineno">510</tt>  <tt class="py-line">            <tt class="py-comment"># -----</tt> </tt>
<a name="L511"></a><tt class="py-lineno">511</tt>  <tt class="py-line">            <tt class="py-comment"># The BETWEEN operator</tt> </tt>
<a name="L512"></a><tt class="py-lineno">512</tt>  <tt class="py-line">            <tt class="py-comment">#</tt> </tt>
<a name="L513"></a><tt class="py-lineno">513</tt>  <tt class="py-line">            <tt class="py-comment"># The BETWEEN operator is logically equivalent to a pair of</tt> </tt>
<a name="L514"></a><tt class="py-lineno">514</tt>  <tt class="py-line">            <tt class="py-comment"># comparisons. "x BETWEEN y AND z" is equivalent to "x&gt;=y AND x&lt;=z"</tt> </tt>
<a name="L515"></a><tt class="py-lineno">515</tt>  <tt class="py-line">            <tt class="py-comment"># except that with BETWEEN, the x expression is only evaluated</tt> </tt>
<a name="L516"></a><tt class="py-lineno">516</tt>  <tt class="py-line">            <tt class="py-comment"># once. The precedence of the BETWEEN operator is the same as the</tt> </tt>
<a name="L517"></a><tt class="py-lineno">517</tt>  <tt class="py-line">            <tt class="py-comment"># precedence as operators == and != and LIKE and groups left to</tt> </tt>
<a name="L518"></a><tt class="py-lineno">518</tt>  <tt class="py-line">            <tt class="py-comment"># right.</tt> </tt>
<a name="L519"></a><tt class="py-lineno">519</tt>  <tt class="py-line">            <tt class="py-comment"># -----</tt> </tt>
<a name="L520"></a><tt class="py-lineno">520</tt>  <tt class="py-line"> </tt>
<a name="L521"></a><tt class="py-lineno">521</tt>  <tt class="py-line">            <tt class="py-comment"># We are testing overlap between the query segment and records in</tt> </tt>
<a name="L522"></a><tt class="py-lineno">522</tt>  <tt class="py-line">            <tt class="py-comment"># the index, using non-strict coordinates comparisons.</tt> </tt>
<a name="L523"></a><tt class="py-lineno">523</tt>  <tt class="py-line">            <tt class="py-comment"># The query segment end must be passed as end-inclusive</tt> </tt>
<a name="L524"></a><tt class="py-lineno">524</tt>  <tt class="py-line">            <tt class="py-comment"># The index should also have been build with end-inclusive</tt> </tt>
<a name="L525"></a><tt class="py-lineno">525</tt>  <tt class="py-line">            <tt class="py-comment"># end coordinates.</tt> </tt>
<a name="L526"></a><tt class="py-lineno">526</tt>  <tt class="py-line">            <tt class="py-comment"># See https://github.com/biopython/biopython/pull/1086#issuecomment-285069073</tt> </tt>
<a name="L527"></a><tt class="py-lineno">527</tt>  <tt class="py-line"> </tt>
<a name="L528"></a><tt class="py-lineno">528</tt>  <tt class="py-line">            <tt class="py-name">result</tt> <tt class="py-op">=</tt> <tt class="py-name">con</tt><tt class="py-op">.</tt><tt id="link-168" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-168', 'execute', 'link-95');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L529"></a><tt class="py-lineno">529</tt>  <tt class="py-line">                <tt class="py-string">"SELECT DISTINCT start, end, offset FROM offset_data "</tt> </tt>
<a name="L530"></a><tt class="py-lineno">530</tt>  <tt class="py-line">                <tt class="py-string">"WHERE bin IN (%s) "</tt> </tt>
<a name="L531"></a><tt class="py-lineno">531</tt>  <tt class="py-line">                <tt class="py-string">"AND (end BETWEEN %s AND %s OR %s BETWEEN start AND end) "</tt> </tt>
<a name="L532"></a><tt class="py-lineno">532</tt>  <tt class="py-line">                <tt class="py-string">"ORDER BY start, end, offset ASC;"</tt> <tt class="py-op">%</tt> </tt>
<a name="L533"></a><tt class="py-lineno">533</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-name">possible_bins</tt><tt class="py-op">,</tt> <tt class="py-name">exonstart</tt><tt class="py-op">,</tt> <tt class="py-name">exonend</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">exonend</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L534"></a><tt class="py-lineno">534</tt>  <tt class="py-line"> </tt>
<a name="L535"></a><tt class="py-lineno">535</tt>  <tt class="py-line">            <tt class="py-name">rows</tt> <tt class="py-op">=</tt> <tt class="py-name">result</tt><tt class="py-op">.</tt><tt id="link-169" class="py-name" targets="Method BioSQL.BioSeqDatabase._CursorWrapper.fetchall()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#fetchall"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchall" class="py-name" href="#" onclick="return doclink('link-169', 'fetchall', 'link-169');">fetchall</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L536"></a><tt class="py-lineno">536</tt>  <tt class="py-line"> </tt>
<a name="L537"></a><tt class="py-lineno">537</tt>  <tt class="py-line">            <tt class="py-comment"># rows come from the sqlite index,</tt> </tt>
<a name="L538"></a><tt class="py-lineno">538</tt>  <tt class="py-line">            <tt class="py-comment"># which should have been written using __make_new_index,</tt> </tt>
<a name="L539"></a><tt class="py-lineno">539</tt>  <tt class="py-line">            <tt class="py-comment"># so rec_start and rec_end should be zero-based "inclusive" coordinates</tt> </tt>
<a name="L540"></a><tt class="py-lineno">540</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">rec_start</tt><tt class="py-op">,</tt> <tt class="py-name">rec_end</tt><tt class="py-op">,</tt> <tt class="py-name">offset</tt> <tt class="py-keyword">in</tt> <tt class="py-name">rows</tt><tt class="py-op">:</tt> </tt>
<a name="L541"></a><tt class="py-lineno">541</tt>  <tt class="py-line">                <tt class="py-comment"># Avoid yielding multiple time the same block</tt> </tt>
<a name="L542"></a><tt class="py-lineno">542</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-op">(</tt><tt class="py-name">rec_start</tt><tt class="py-op">,</tt> <tt class="py-name">rec_end</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">yielded_rec_coords</tt><tt class="py-op">:</tt> </tt>
<a name="L543"></a><tt class="py-lineno">543</tt>  <tt class="py-line">                    <tt class="py-keyword">continue</tt> </tt>
<a name="L544"></a><tt class="py-lineno">544</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L545"></a><tt class="py-lineno">545</tt>  <tt class="py-line">                    <tt class="py-name">yielded_rec_coords</tt><tt class="py-op">.</tt><tt id="link-170" class="py-name" targets="Method Bio.Graphics.BasicChromosome._ChromosomeComponent.add()=Bio.Graphics.BasicChromosome._ChromosomeComponent-class.html#add,Method Bio.Nexus.Nexus.StepMatrix.add()=Bio.Nexus.Nexus.StepMatrix-class.html#add,Method Bio.Nexus.Nodes.Chain.add()=Bio.Nexus.Nodes.Chain-class.html#add,Method Bio.PDB.Entity.Entity.add()=Bio.PDB.Entity.Entity-class.html#add,Method Bio.PDB.Residue.DisorderedResidue.add()=Bio.PDB.Residue.DisorderedResidue-class.html#add,Method Bio.PDB.Residue.Residue.add()=Bio.PDB.Residue.Residue-class.html#add,Method Bio.Restriction.Restriction.RestrictionBatch.add()=Bio.Restriction.Restriction.RestrictionBatch-class.html#add"><a title="Bio.Graphics.BasicChromosome._ChromosomeComponent.add
Bio.Nexus.Nexus.StepMatrix.add
Bio.Nexus.Nodes.Chain.add
Bio.PDB.Entity.Entity.add
Bio.PDB.Residue.DisorderedResidue.add
Bio.PDB.Residue.Residue.add
Bio.Restriction.Restriction.RestrictionBatch.add" class="py-name" href="#" onclick="return doclink('link-170', 'add', 'link-170');">add</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">rec_start</tt><tt class="py-op">,</tt> <tt class="py-name">rec_end</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L546"></a><tt class="py-lineno">546</tt>  <tt class="py-line">                <tt class="py-comment"># Iterate through hits, fetching alignments from the MAF file</tt> </tt>
<a name="L547"></a><tt class="py-lineno">547</tt>  <tt class="py-line">                <tt class="py-comment"># and checking to be sure we've retrieved the expected record.</tt> </tt>
<a name="L548"></a><tt class="py-lineno">548</tt>  <tt class="py-line"> </tt>
<a name="L549"></a><tt class="py-lineno">549</tt>  <tt class="py-line">                <tt class="py-name">fetched</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-171" class="py-name" targets="Method Bio.AlignIO.MafIO.MafIndex._get_record()=Bio.AlignIO.MafIO.MafIndex-class.html#_get_record"><a title="Bio.AlignIO.MafIO.MafIndex._get_record" class="py-name" href="#" onclick="return doclink('link-171', '_get_record', 'link-171');">_get_record</a></tt><tt class="py-op">(</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">offset</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L550"></a><tt class="py-lineno">550</tt>  <tt class="py-line"> </tt>
<a name="L551"></a><tt class="py-lineno">551</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">fetched</tt><tt class="py-op">:</tt> </tt>
<a name="L552"></a><tt class="py-lineno">552</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-172" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-172', 'id', 'link-24');">id</a></tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">:</tt> </tt>
<a name="L553"></a><tt class="py-lineno">553</tt>  <tt class="py-line">                        <tt class="py-comment"># start and size come from the maf lines</tt> </tt>
<a name="L554"></a><tt class="py-lineno">554</tt>  <tt class="py-line">                        <tt id="link-173" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-173', 'start', 'link-131');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-174" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-174', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"start"</tt><tt class="py-op">]</tt> </tt>
<a name="L555"></a><tt class="py-lineno">555</tt>  <tt class="py-line">                        <tt class="py-comment"># "inclusive" end is start + length - 1</tt> </tt>
<a name="L556"></a><tt class="py-lineno">556</tt>  <tt class="py-line">                        <tt id="link-175" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-175', 'end', 'link-135');">end</a></tt> <tt class="py-op">=</tt> <tt id="link-176" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-176', 'start', 'link-131');">start</a></tt> <tt class="py-op">+</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-177" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-177', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"size"</tt><tt class="py-op">]</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt> </tt>
<a name="L557"></a><tt class="py-lineno">557</tt>  <tt class="py-line"> </tt>
<a name="L558"></a><tt class="py-lineno">558</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-op">(</tt><tt id="link-178" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-178', 'start', 'link-131');">start</a></tt> <tt class="py-op">==</tt> <tt class="py-name">rec_start</tt> <tt class="py-keyword">and</tt> <tt id="link-179" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-179', 'end', 'link-135');">end</a></tt> <tt class="py-op">==</tt> <tt class="py-name">rec_end</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L559"></a><tt class="py-lineno">559</tt>  <tt class="py-line">                            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Expected %s-%s @ offset %s, found %s-%s"</tt> <tt class="py-op">%</tt> </tt>
<a name="L560"></a><tt class="py-lineno">560</tt>  <tt class="py-line">                                             <tt class="py-op">(</tt><tt class="py-name">rec_start</tt><tt class="py-op">,</tt> <tt class="py-name">rec_end</tt><tt class="py-op">,</tt> <tt class="py-name">offset</tt><tt class="py-op">,</tt> <tt id="link-180" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-180', 'start', 'link-131');">start</a></tt><tt class="py-op">,</tt> <tt id="link-181" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-181', 'end', 'link-135');">end</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L561"></a><tt class="py-lineno">561</tt>  <tt class="py-line"> </tt>
<a name="L562"></a><tt class="py-lineno">562</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt class="py-name">fetched</tt> </tt>
</div><a name="L563"></a><tt class="py-lineno">563</tt>  <tt class="py-line"> </tt>
<a name="MafIndex.get_spliced"></a><div id="MafIndex.get_spliced-def"><a name="L564"></a><tt class="py-lineno">564</tt> <a class="py-toggle" href="#" id="MafIndex.get_spliced-toggle" onclick="return toggle('MafIndex.get_spliced');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafIndex-class.html#get_spliced">get_spliced</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">starts</tt><tt class="py-op">,</tt> <tt class="py-param">ends</tt><tt class="py-op">,</tt> <tt class="py-param">strand</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIndex.get_spliced-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafIndex.get_spliced-expanded"><a name="L565"></a><tt class="py-lineno">565</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a multiple alignment of the exact sequence range provided.</tt> </tt>
<a name="L566"></a><tt class="py-lineno">566</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L567"></a><tt class="py-lineno">567</tt>  <tt class="py-line"><tt class="py-docstring">        Accepts two lists of start and end positions on target_seqname, representing</tt> </tt>
<a name="L568"></a><tt class="py-lineno">568</tt>  <tt class="py-line"><tt class="py-docstring">        exons to be spliced in silico.  Returns a *MultipleSeqAlignment* of the</tt> </tt>
<a name="L569"></a><tt class="py-lineno">569</tt>  <tt class="py-line"><tt class="py-docstring">        desired sequences spliced together.</tt> </tt>
<a name="L570"></a><tt class="py-lineno">570</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L571"></a><tt class="py-lineno">571</tt>  <tt class="py-line"><tt class="py-docstring">        *starts* should be a list of 0-based start coordinates of segments in the reference.</tt> </tt>
<a name="L572"></a><tt class="py-lineno">572</tt>  <tt class="py-line"><tt class="py-docstring">        *ends* should be the list of the corresponding segment ends</tt> </tt>
<a name="L573"></a><tt class="py-lineno">573</tt>  <tt class="py-line"><tt class="py-docstring">        (in the half-open UCSC convention:</tt> </tt>
<a name="L574"></a><tt class="py-lineno">574</tt>  <tt class="py-line"><tt class="py-docstring">        http://genome.ucsc.edu/blog/the-ucsc-genome-browser-coordinate-counting-systems/).</tt> </tt>
<a name="L575"></a><tt class="py-lineno">575</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L576"></a><tt class="py-lineno">576</tt>  <tt class="py-line"><tt class="py-docstring">        To ask for the alignment portion corresponding to the first 100</tt> </tt>
<a name="L577"></a><tt class="py-lineno">577</tt>  <tt class="py-line"><tt class="py-docstring">        nucleotides of the reference sequence, you would use</tt> </tt>
<a name="L578"></a><tt class="py-lineno">578</tt>  <tt class="py-line"><tt class="py-docstring">        ``search([0], [100])``</tt> </tt>
<a name="L579"></a><tt class="py-lineno">579</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L580"></a><tt class="py-lineno">580</tt>  <tt class="py-line">        <tt class="py-comment"># validate strand</tt> </tt>
<a name="L581"></a><tt class="py-lineno">581</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-182" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-182', 'strand', 'link-19');">strand</a></tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L582"></a><tt class="py-lineno">582</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Strand must be 1 or -1, got %s"</tt> <tt class="py-op">%</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-183" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-183', 'strand', 'link-19');">strand</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L583"></a><tt class="py-lineno">583</tt>  <tt class="py-line"> </tt>
<a name="L584"></a><tt class="py-lineno">584</tt>  <tt class="py-line">        <tt class="py-comment"># pull all alignments that span the desired intervals</tt> </tt>
<a name="L585"></a><tt class="py-lineno">585</tt>  <tt class="py-line">        <tt class="py-name">fetched</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">multiseq</tt> <tt class="py-keyword">for</tt> <tt class="py-name">multiseq</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-184" class="py-name" targets="Method Bio.AlignIO.MafIO.MafIndex.search()=Bio.AlignIO.MafIO.MafIndex-class.html#search,Method Bio.KDTree.KDTree'.KDTree.search()=Bio.KDTree.KDTree%27.KDTree-class.html#search,Method Bio.PDB.NeighborSearch'.NeighborSearch.search()=Bio.PDB.NeighborSearch%27.NeighborSearch-class.html#search,Method Bio.Phylo.TreeConstruction.NNITreeSearcher.search()=Bio.Phylo.TreeConstruction.NNITreeSearcher-class.html#search,Method Bio.Phylo.TreeConstruction.TreeSearcher.search()=Bio.Phylo.TreeConstruction.TreeSearcher-class.html#search,Class Method Bio.Restriction.Restriction.AbstractCut.search()=Bio.Restriction.Restriction.AbstractCut-class.html#search,Method Bio.Restriction.Restriction.RestrictionBatch.search()=Bio.Restriction.Restriction.RestrictionBatch-class.html#search,Function Bio.SCOP.search()=Bio.SCOP-module.html#search,Function Bio.TogoWS.search()=Bio.TogoWS-module.html#search,Method Bio.motifs.Instances.search()=Bio.motifs.Instances-class.html#search,Method Bio.motifs.matrix.PositionSpecificScoringMatrix.search()=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#search"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-184', 'search', 'link-184');">search</a></tt><tt class="py-op">(</tt><tt class="py-name">starts</tt><tt class="py-op">,</tt> <tt class="py-name">ends</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L586"></a><tt class="py-lineno">586</tt>  <tt class="py-line"> </tt>
<a name="L587"></a><tt class="py-lineno">587</tt>  <tt class="py-line">        <tt class="py-comment"># keep track of the expected letter count</tt> </tt>
<a name="L588"></a><tt class="py-lineno">588</tt>  <tt class="py-line">        <tt class="py-comment"># (sum of lengths of [start, end) segments,</tt> </tt>
<a name="L589"></a><tt class="py-lineno">589</tt>  <tt class="py-line">        <tt class="py-comment"># where [start, end) half-open)</tt> </tt>
<a name="L590"></a><tt class="py-lineno">590</tt>  <tt class="py-line">        <tt class="py-name">expected_letters</tt> <tt class="py-op">=</tt> <tt id="link-185" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.sum()=Bio.Nexus.Nexus.StepMatrix-class.html#sum,Method Bio.SubsMat.SeqMat.sum()=Bio.SubsMat.SeqMat-class.html#sum"><a title="Bio.Nexus.Nexus.StepMatrix.sum
Bio.SubsMat.SeqMat.sum" class="py-name" href="#" onclick="return doclink('link-185', 'sum', 'link-185');">sum</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt id="link-186" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-186', 'end', 'link-135');">end</a></tt> <tt class="py-op">-</tt> <tt id="link-187" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-187', 'start', 'link-131');">start</a></tt> <tt class="py-keyword">for</tt> <tt id="link-188" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-188', 'start', 'link-131');">start</a></tt><tt class="py-op">,</tt> <tt id="link-189" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-189', 'end', 'link-135');">end</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">starts</tt><tt class="py-op">,</tt> <tt class="py-name">ends</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L591"></a><tt class="py-lineno">591</tt>  <tt class="py-line"> </tt>
<a name="L592"></a><tt class="py-lineno">592</tt>  <tt class="py-line">        <tt class="py-comment"># if there's no alignment, return filler for the assembly of the length given</tt> </tt>
<a name="L593"></a><tt class="py-lineno">593</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">fetched</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L594"></a><tt class="py-lineno">594</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-190" class="py-name"><a title="Bio.Align.MultipleSeqAlignment" class="py-name" href="#" onclick="return doclink('link-190', 'MultipleSeqAlignment', 'link-11');">MultipleSeqAlignment</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt id="link-191" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-191', 'SeqRecord', 'link-7');">SeqRecord</a></tt><tt class="py-op">(</tt><tt id="link-192" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-192', 'Seq', 'link-4');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">"N"</tt> <tt class="py-op">*</tt> <tt class="py-name">expected_letters</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L595"></a><tt class="py-lineno">595</tt>  <tt class="py-line">                                                   <tt id="link-193" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-193', 'id', 'link-24');">id</a></tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L596"></a><tt class="py-lineno">596</tt>  <tt class="py-line"> </tt>
<a name="L597"></a><tt class="py-lineno">597</tt>  <tt class="py-line">        <tt class="py-comment"># find the union of all IDs in these alignments</tt> </tt>
<a name="L598"></a><tt class="py-lineno">598</tt>  <tt class="py-line">        <tt class="py-name">all_seqnames</tt> <tt class="py-op">=</tt> <tt id="link-194" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-194', 'set', 'link-36');">set</a></tt><tt class="py-op">(</tt> </tt>
<a name="L599"></a><tt class="py-lineno">599</tt>  <tt class="py-line">            <tt class="py-op">[</tt><tt id="link-195" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-195', 'sequence', 'link-58');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-196" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-196', 'id', 'link-24');">id</a></tt> <tt class="py-keyword">for</tt> <tt class="py-name">multiseq</tt> <tt class="py-keyword">in</tt> <tt class="py-name">fetched</tt> <tt class="py-keyword">for</tt> <tt id="link-197" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-197', 'sequence', 'link-58');">sequence</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">multiseq</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L600"></a><tt class="py-lineno">600</tt>  <tt class="py-line"> </tt>
<a name="L601"></a><tt class="py-lineno">601</tt>  <tt class="py-line">        <tt class="py-comment"># split every record by base position</tt> </tt>
<a name="L602"></a><tt class="py-lineno">602</tt>  <tt class="py-line">        <tt class="py-comment"># key: sequence name</tt> </tt>
<a name="L603"></a><tt class="py-lineno">603</tt>  <tt class="py-line">        <tt class="py-comment"># value: dictionary</tt> </tt>
<a name="L604"></a><tt class="py-lineno">604</tt>  <tt class="py-line">        <tt class="py-comment">#        key: position in the reference sequence</tt> </tt>
<a name="L605"></a><tt class="py-lineno">605</tt>  <tt class="py-line">        <tt class="py-comment">#        value: letter(s) (including letters</tt> </tt>
<a name="L606"></a><tt class="py-lineno">606</tt>  <tt class="py-line">        <tt class="py-comment">#               aligned to the "-" preceding the letter</tt> </tt>
<a name="L607"></a><tt class="py-lineno">607</tt>  <tt class="py-line">        <tt class="py-comment">#               at the position in the reference, if any)</tt> </tt>
<a name="L608"></a><tt class="py-lineno">608</tt>  <tt class="py-line">        <tt class="py-name">split_by_position</tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">seq_name</tt><tt class="py-op">,</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">seq_name</tt> <tt class="py-keyword">in</tt> <tt class="py-name">all_seqnames</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L609"></a><tt class="py-lineno">609</tt>  <tt class="py-line"> </tt>
<a name="L610"></a><tt class="py-lineno">610</tt>  <tt class="py-line">        <tt class="py-comment"># keep track of what the total number of (unspliced) letters should be</tt> </tt>
<a name="L611"></a><tt class="py-lineno">611</tt>  <tt class="py-line">        <tt class="py-name">total_rec_length</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L612"></a><tt class="py-lineno">612</tt>  <tt class="py-line"> </tt>
<a name="L613"></a><tt class="py-lineno">613</tt>  <tt class="py-line">        <tt class="py-comment"># track first strand encountered on the target seqname</tt> </tt>
<a name="L614"></a><tt class="py-lineno">614</tt>  <tt class="py-line">        <tt class="py-name">ref_first_strand</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L615"></a><tt class="py-lineno">615</tt>  <tt class="py-line"> </tt>
<a name="L616"></a><tt class="py-lineno">616</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">multiseq</tt> <tt class="py-keyword">in</tt> <tt class="py-name">fetched</tt><tt class="py-op">:</tt> </tt>
<a name="L617"></a><tt class="py-lineno">617</tt>  <tt class="py-line">            <tt class="py-comment"># find the target_seqname in this MultipleSeqAlignment and use it to</tt> </tt>
<a name="L618"></a><tt class="py-lineno">618</tt>  <tt class="py-line">            <tt class="py-comment"># set the parameters for the rest of this iteration</tt> </tt>
<a name="L619"></a><tt class="py-lineno">619</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">seqrec</tt> <tt class="py-keyword">in</tt> <tt class="py-name">multiseq</tt><tt class="py-op">:</tt> </tt>
<a name="L620"></a><tt class="py-lineno">620</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-198" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-198', 'id', 'link-24');">id</a></tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">:</tt> </tt>
<a name="L621"></a><tt class="py-lineno">621</tt>  <tt class="py-line">                    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L622"></a><tt class="py-lineno">622</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">ref_first_strand</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L623"></a><tt class="py-lineno">623</tt>  <tt class="py-line">                            <tt class="py-name">ref_first_strand</tt> <tt class="py-op">=</tt> <tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-199" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-199', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"strand"</tt><tt class="py-op">]</tt> </tt>
<a name="L624"></a><tt class="py-lineno">624</tt>  <tt class="py-line"> </tt>
<a name="L625"></a><tt class="py-lineno">625</tt>  <tt class="py-line">                            <tt class="py-keyword">if</tt> <tt class="py-name">ref_first_strand</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L626"></a><tt class="py-lineno">626</tt>  <tt class="py-line">                                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Strand must be 1 or -1"</tt><tt class="py-op">)</tt> </tt>
<a name="L627"></a><tt class="py-lineno">627</tt>  <tt class="py-line">                        <tt class="py-keyword">elif</tt> <tt class="py-name">ref_first_strand</tt> <tt class="py-op">!=</tt> <tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-200" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-200', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"strand"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L628"></a><tt class="py-lineno">628</tt>  <tt class="py-line">                            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Encountered strand='%s' on target seqname, "</tt> </tt>
<a name="L629"></a><tt class="py-lineno">629</tt>  <tt class="py-line">                                             <tt class="py-string">"expected '%s'"</tt> <tt class="py-op">%</tt> </tt>
<a name="L630"></a><tt class="py-lineno">630</tt>  <tt class="py-line">                                             <tt class="py-op">(</tt><tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-201" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-201', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"strand"</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">ref_first_strand</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L631"></a><tt class="py-lineno">631</tt>  <tt class="py-line">                    <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L632"></a><tt class="py-lineno">632</tt>  <tt class="py-line">                        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"No strand information for target seqname (%s)"</tt> <tt class="py-op">%</tt> </tt>
<a name="L633"></a><tt class="py-lineno">633</tt>  <tt class="py-line">                                         <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">)</tt> </tt>
<a name="L634"></a><tt class="py-lineno">634</tt>  <tt class="py-line">                    <tt class="py-comment"># length including gaps (i.e. alignment length)</tt> </tt>
<a name="L635"></a><tt class="py-lineno">635</tt>  <tt class="py-line">                    <tt class="py-name">rec_length</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">seqrec</tt><tt class="py-op">)</tt> </tt>
<a name="L636"></a><tt class="py-lineno">636</tt>  <tt class="py-line">                    <tt class="py-name">rec_start</tt> <tt class="py-op">=</tt> <tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-202" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-202', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"start"</tt><tt class="py-op">]</tt> </tt>
<a name="L637"></a><tt class="py-lineno">637</tt>  <tt class="py-line">                    <tt class="py-name">ungapped_length</tt> <tt class="py-op">=</tt> <tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-203" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-203', 'annotations', 'link-17');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"size"</tt><tt class="py-op">]</tt> </tt>
<a name="L638"></a><tt class="py-lineno">638</tt>  <tt class="py-line">                    <tt class="py-comment"># inclusive end in zero-based coordinates of the reference</tt> </tt>
<a name="L639"></a><tt class="py-lineno">639</tt>  <tt class="py-line">                    <tt class="py-name">rec_end</tt> <tt class="py-op">=</tt> <tt class="py-name">rec_start</tt> <tt class="py-op">+</tt> <tt class="py-name">ungapped_length</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt> </tt>
<a name="L640"></a><tt class="py-lineno">640</tt>  <tt class="py-line">                    <tt class="py-comment"># This is length in terms of actual letters in the reference</tt> </tt>
<a name="L641"></a><tt class="py-lineno">641</tt>  <tt class="py-line">                    <tt class="py-name">total_rec_length</tt> <tt class="py-op">+=</tt> <tt class="py-name">ungapped_length</tt> </tt>
<a name="L642"></a><tt class="py-lineno">642</tt>  <tt class="py-line"> </tt>
<a name="L643"></a><tt class="py-lineno">643</tt>  <tt class="py-line">                    <tt class="py-comment"># blank out these positions for every seqname</tt> </tt>
<a name="L644"></a><tt class="py-lineno">644</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-name">seqrec</tt> <tt class="py-keyword">in</tt> <tt class="py-name">multiseq</tt><tt class="py-op">:</tt> </tt>
<a name="L645"></a><tt class="py-lineno">645</tt>  <tt class="py-line">                        <tt class="py-keyword">for</tt> <tt class="py-name">pos</tt> <tt class="py-keyword">in</tt> <tt id="link-204" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-204', 'range', 'link-145');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">rec_start</tt><tt class="py-op">,</tt> <tt class="py-name">rec_end</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L646"></a><tt class="py-lineno">646</tt>  <tt class="py-line">                            <tt class="py-name">split_by_position</tt><tt class="py-op">[</tt><tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-205" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-205', 'id', 'link-24');">id</a></tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">pos</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L647"></a><tt class="py-lineno">647</tt>  <tt class="py-line"> </tt>
<a name="L648"></a><tt class="py-lineno">648</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L649"></a><tt class="py-lineno">649</tt>  <tt class="py-line">            <tt class="py-comment"># http://psung.blogspot.fr/2007/12/for-else-in-python.html</tt> </tt>
<a name="L650"></a><tt class="py-lineno">650</tt>  <tt class="py-line">            <tt class="py-comment"># https://docs.python.org/2/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops</tt> </tt>
<a name="L651"></a><tt class="py-lineno">651</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L652"></a><tt class="py-lineno">652</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Did not find %s in alignment bundle"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L653"></a><tt class="py-lineno">653</tt>  <tt class="py-line"> </tt>
<a name="L654"></a><tt class="py-lineno">654</tt>  <tt class="py-line">            <tt class="py-comment"># the true, chromosome/contig/etc position in the target seqname</tt> </tt>
<a name="L655"></a><tt class="py-lineno">655</tt>  <tt class="py-line">            <tt class="py-name">real_pos</tt> <tt class="py-op">=</tt> <tt class="py-name">rec_start</tt> </tt>
<a name="L656"></a><tt class="py-lineno">656</tt>  <tt class="py-line"> </tt>
<a name="L657"></a><tt class="py-lineno">657</tt>  <tt class="py-line">            <tt class="py-comment"># loop over the alignment to fill split_by_position</tt> </tt>
<a name="L658"></a><tt class="py-lineno">658</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">gapped_pos</tt> <tt class="py-keyword">in</tt> <tt id="link-206" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-206', 'range', 'link-145');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">rec_length</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L659"></a><tt class="py-lineno">659</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">seqrec</tt> <tt class="py-keyword">in</tt> <tt class="py-name">multiseq</tt><tt class="py-op">:</tt> </tt>
<a name="L660"></a><tt class="py-lineno">660</tt>  <tt class="py-line">                    <tt class="py-comment"># keep track of this position's value for the target seqname</tt> </tt>
<a name="L661"></a><tt class="py-lineno">661</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-207" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-207', 'id', 'link-24');">id</a></tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">:</tt> </tt>
<a name="L662"></a><tt class="py-lineno">662</tt>  <tt class="py-line">                        <tt class="py-name">track_val</tt> <tt class="py-op">=</tt> <tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-208" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-208', 'seq', 'link-29');">seq</a></tt><tt class="py-op">[</tt><tt class="py-name">gapped_pos</tt><tt class="py-op">]</tt> </tt>
<a name="L663"></a><tt class="py-lineno">663</tt>  <tt class="py-line"> </tt>
<a name="L664"></a><tt class="py-lineno">664</tt>  <tt class="py-line">                    <tt class="py-comment"># Here, a real_pos that corresponds to just after a series of "-"</tt> </tt>
<a name="L665"></a><tt class="py-lineno">665</tt>  <tt class="py-line">                    <tt class="py-comment"># in the reference will "accumulate" the letters found in other sequences</tt> </tt>
<a name="L666"></a><tt class="py-lineno">666</tt>  <tt class="py-line">                    <tt class="py-comment"># in front of the "-"s</tt> </tt>
<a name="L667"></a><tt class="py-lineno">667</tt>  <tt class="py-line">                    <tt class="py-name">split_by_position</tt><tt class="py-op">[</tt><tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-209" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-209', 'id', 'link-24');">id</a></tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">real_pos</tt><tt class="py-op">]</tt> <tt class="py-op">+=</tt> <tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-210" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-210', 'seq', 'link-29');">seq</a></tt><tt class="py-op">[</tt><tt class="py-name">gapped_pos</tt><tt class="py-op">]</tt> </tt>
<a name="L668"></a><tt class="py-lineno">668</tt>  <tt class="py-line"> </tt>
<a name="L669"></a><tt class="py-lineno">669</tt>  <tt class="py-line">                <tt class="py-comment"># increment the real_pos counter only when non-gaps are found in</tt> </tt>
<a name="L670"></a><tt class="py-lineno">670</tt>  <tt class="py-line">                <tt class="py-comment"># the target_seqname, and we haven't reached the end of the record</tt> </tt>
<a name="L671"></a><tt class="py-lineno">671</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">track_val</tt> <tt class="py-op">!=</tt> <tt class="py-string">"-"</tt> <tt class="py-keyword">and</tt> <tt class="py-name">real_pos</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">rec_end</tt><tt class="py-op">:</tt> </tt>
<a name="L672"></a><tt class="py-lineno">672</tt>  <tt class="py-line">                    <tt class="py-name">real_pos</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L673"></a><tt class="py-lineno">673</tt>  <tt class="py-line"> </tt>
<a name="L674"></a><tt class="py-lineno">674</tt>  <tt class="py-line">        <tt class="py-comment"># make sure the number of bp entries equals the sum of the record lengths</tt> </tt>
<a name="L675"></a><tt class="py-lineno">675</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">split_by_position</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">total_rec_length</tt><tt class="py-op">:</tt> </tt>
<a name="L676"></a><tt class="py-lineno">676</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Target seqname (%s) has %s records, expected %s"</tt> <tt class="py-op">%</tt> </tt>
<a name="L677"></a><tt class="py-lineno">677</tt>  <tt class="py-line">                             <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">,</tt> </tt>
<a name="L678"></a><tt class="py-lineno">678</tt>  <tt class="py-line">                              <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">split_by_position</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L679"></a><tt class="py-lineno">679</tt>  <tt class="py-line">                              <tt class="py-name">total_rec_length</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L680"></a><tt class="py-lineno">680</tt>  <tt class="py-line"> </tt>
<a name="L681"></a><tt class="py-lineno">681</tt>  <tt class="py-line">        <tt class="py-comment"># translates a position in the target_seqname sequence to its gapped length</tt> </tt>
<a name="L682"></a><tt class="py-lineno">682</tt>  <tt class="py-line">        <tt class="py-name">realpos_to_len</tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">pos</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">gapped_fragment</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L683"></a><tt class="py-lineno">683</tt>  <tt class="py-line">                               <tt class="py-keyword">for</tt> <tt class="py-name">pos</tt><tt class="py-op">,</tt> <tt class="py-name">gapped_fragment</tt> <tt class="py-keyword">in</tt> <tt class="py-name">split_by_position</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-211" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-211', 'items', 'link-43');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L684"></a><tt class="py-lineno">684</tt>  <tt class="py-line">                               <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">gapped_fragment</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L685"></a><tt class="py-lineno">685</tt>  <tt class="py-line"> </tt>
<a name="L686"></a><tt class="py-lineno">686</tt>  <tt class="py-line">        <tt class="py-comment"># splice together the exons</tt> </tt>
<a name="L687"></a><tt class="py-lineno">687</tt>  <tt class="py-line">        <tt class="py-name">subseq</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L688"></a><tt class="py-lineno">688</tt>  <tt class="py-line"> </tt>
<a name="L689"></a><tt class="py-lineno">689</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">seqid</tt> <tt class="py-keyword">in</tt> <tt class="py-name">all_seqnames</tt><tt class="py-op">:</tt> </tt>
<a name="L690"></a><tt class="py-lineno">690</tt>  <tt class="py-line">            <tt class="py-name">seq_split</tt> <tt class="py-op">=</tt> <tt class="py-name">split_by_position</tt><tt class="py-op">[</tt><tt class="py-name">seqid</tt><tt class="py-op">]</tt> </tt>
<a name="L691"></a><tt class="py-lineno">691</tt>  <tt class="py-line">            <tt class="py-name">seq_splice</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L692"></a><tt class="py-lineno">692</tt>  <tt class="py-line"> </tt>
<a name="L693"></a><tt class="py-lineno">693</tt>  <tt class="py-line">            <tt class="py-name">filler_char</tt> <tt class="py-op">=</tt> <tt class="py-string">"N"</tt> <tt class="py-keyword">if</tt> <tt class="py-name">seqid</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt> <tt class="py-keyword">else</tt> <tt class="py-string">"-"</tt> </tt>
<a name="L694"></a><tt class="py-lineno">694</tt>  <tt class="py-line"> </tt>
<a name="L695"></a><tt class="py-lineno">695</tt>  <tt class="py-line">            <tt class="py-comment"># iterate from start to end, taking bases from split_by_position when</tt> </tt>
<a name="L696"></a><tt class="py-lineno">696</tt>  <tt class="py-line">            <tt class="py-comment"># they exist, using N or - for gaps when there is no alignment.</tt> </tt>
<a name="L697"></a><tt class="py-lineno">697</tt>  <tt class="py-line">            <tt id="link-212" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-212', 'append', 'link-64');">append</a></tt> <tt class="py-op">=</tt> <tt class="py-name">seq_splice</tt><tt class="py-op">.</tt><tt id="link-213" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-213', 'append', 'link-64');">append</a></tt> </tt>
<a name="L698"></a><tt class="py-lineno">698</tt>  <tt class="py-line"> </tt>
<a name="L699"></a><tt class="py-lineno">699</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">exonstart</tt><tt class="py-op">,</tt> <tt class="py-name">exonend</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">starts</tt><tt class="py-op">,</tt> <tt class="py-name">ends</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L700"></a><tt class="py-lineno">700</tt>  <tt class="py-line">                <tt class="py-comment"># exonend is exclusive</tt> </tt>
<a name="L701"></a><tt class="py-lineno">701</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">real_pos</tt> <tt class="py-keyword">in</tt> <tt id="link-214" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-214', 'range', 'link-145');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">exonstart</tt><tt class="py-op">,</tt> <tt class="py-name">exonend</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L702"></a><tt class="py-lineno">702</tt>  <tt class="py-line">                    <tt class="py-comment"># if this seqname has this position, add it</tt> </tt>
<a name="L703"></a><tt class="py-lineno">703</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">real_pos</tt> <tt class="py-keyword">in</tt> <tt class="py-name">seq_split</tt><tt class="py-op">:</tt> </tt>
<a name="L704"></a><tt class="py-lineno">704</tt>  <tt class="py-line">                        <tt id="link-215" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-215', 'append', 'link-64');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">seq_split</tt><tt class="py-op">[</tt><tt class="py-name">real_pos</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L705"></a><tt class="py-lineno">705</tt>  <tt class="py-line">                    <tt class="py-comment"># if not, but it's in the target_seqname, add length-matched filler</tt> </tt>
<a name="L706"></a><tt class="py-lineno">706</tt>  <tt class="py-line">                    <tt class="py-keyword">elif</tt> <tt class="py-name">real_pos</tt> <tt class="py-keyword">in</tt> <tt class="py-name">realpos_to_len</tt><tt class="py-op">:</tt> </tt>
<a name="L707"></a><tt class="py-lineno">707</tt>  <tt class="py-line">                        <tt id="link-216" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-216', 'append', 'link-64');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">filler_char</tt> <tt class="py-op">*</tt> <tt class="py-name">realpos_to_len</tt><tt class="py-op">[</tt><tt class="py-name">real_pos</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L708"></a><tt class="py-lineno">708</tt>  <tt class="py-line">                    <tt class="py-comment"># it's not in either, so add a single filler character</tt> </tt>
<a name="L709"></a><tt class="py-lineno">709</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L710"></a><tt class="py-lineno">710</tt>  <tt class="py-line">                        <tt id="link-217" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-217', 'append', 'link-64');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">filler_char</tt><tt class="py-op">)</tt> </tt>
<a name="L711"></a><tt class="py-lineno">711</tt>  <tt class="py-line"> </tt>
<a name="L712"></a><tt class="py-lineno">712</tt>  <tt class="py-line">            <tt class="py-name">subseq</tt><tt class="py-op">[</tt><tt class="py-name">seqid</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">seq_splice</tt><tt class="py-op">)</tt> </tt>
<a name="L713"></a><tt class="py-lineno">713</tt>  <tt class="py-line"> </tt>
<a name="L714"></a><tt class="py-lineno">714</tt>  <tt class="py-line">        <tt class="py-comment"># make sure we're returning the right number of letters</tt> </tt>
<a name="L715"></a><tt class="py-lineno">715</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">subseq</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"-"</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">expected_letters</tt><tt class="py-op">:</tt> </tt>
<a name="L716"></a><tt class="py-lineno">716</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Returning %s letters for target seqname (%s), expected %s"</tt> <tt class="py-op">%</tt> </tt>
<a name="L717"></a><tt class="py-lineno">717</tt>  <tt class="py-line">                             <tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">subseq</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"-"</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L718"></a><tt class="py-lineno">718</tt>  <tt class="py-line">                              <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">,</tt> <tt class="py-name">expected_letters</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L719"></a><tt class="py-lineno">719</tt>  <tt class="py-line"> </tt>
<a name="L720"></a><tt class="py-lineno">720</tt>  <tt class="py-line">        <tt class="py-comment"># check to make sure all sequences are the same length as the target seqname</tt> </tt>
<a name="L721"></a><tt class="py-lineno">721</tt>  <tt class="py-line">        <tt class="py-name">ref_subseq_len</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">subseq</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L722"></a><tt class="py-lineno">722</tt>  <tt class="py-line"> </tt>
<a name="L723"></a><tt class="py-lineno">723</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">seqid</tt><tt class="py-op">,</tt> <tt id="link-218" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-218', 'seq', 'link-29');">seq</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">subseq</tt><tt class="py-op">.</tt><tt id="link-219" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-219', 'items', 'link-43');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L724"></a><tt class="py-lineno">724</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-220" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-220', 'seq', 'link-29');">seq</a></tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">ref_subseq_len</tt><tt class="py-op">:</tt> </tt>
<a name="L725"></a><tt class="py-lineno">725</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Returning length %s for %s, expected %s"</tt> <tt class="py-op">%</tt> </tt>
<a name="L726"></a><tt class="py-lineno">726</tt>  <tt class="py-line">                                 <tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-221" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-221', 'seq', 'link-29');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">seqid</tt><tt class="py-op">,</tt> <tt class="py-name">ref_subseq_len</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L727"></a><tt class="py-lineno">727</tt>  <tt class="py-line"> </tt>
<a name="L728"></a><tt class="py-lineno">728</tt>  <tt class="py-line">        <tt class="py-comment"># finally, build a MultipleSeqAlignment object for our final sequences</tt> </tt>
<a name="L729"></a><tt class="py-lineno">729</tt>  <tt class="py-line">        <tt class="py-name">result_multiseq</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L730"></a><tt class="py-lineno">730</tt>  <tt class="py-line"> </tt>
<a name="L731"></a><tt class="py-lineno">731</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">seqid</tt><tt class="py-op">,</tt> <tt id="link-222" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-222', 'seq', 'link-29');">seq</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">subseq</tt><tt class="py-op">.</tt><tt id="link-223" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-223', 'items', 'link-43');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L732"></a><tt class="py-lineno">732</tt>  <tt class="py-line">            <tt id="link-224" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-224', 'seq', 'link-29');">seq</a></tt> <tt class="py-op">=</tt> <tt id="link-225" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-225', 'Seq', 'link-4');">Seq</a></tt><tt class="py-op">(</tt><tt id="link-226" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-226', 'seq', 'link-29');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L733"></a><tt class="py-lineno">733</tt>  <tt class="py-line"> </tt>
<a name="L734"></a><tt class="py-lineno">734</tt>  <tt class="py-line">            <tt id="link-227" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-227', 'seq', 'link-29');">seq</a></tt> <tt class="py-op">=</tt> <tt id="link-228" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-228', 'seq', 'link-29');">seq</a></tt> <tt class="py-keyword">if</tt> <tt id="link-229" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-229', 'strand', 'link-19');">strand</a></tt> <tt class="py-op">==</tt> <tt class="py-name">ref_first_strand</tt> <tt class="py-keyword">else</tt> <tt id="link-230" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-230', 'seq', 'link-29');">seq</a></tt><tt class="py-op">.</tt><tt id="link-231" class="py-name" targets="Method Bio.Seq.MutableSeq.reverse_complement()=Bio.Seq.MutableSeq-class.html#reverse_complement,Method Bio.Seq.Seq.reverse_complement()=Bio.Seq.Seq-class.html#reverse_complement,Method Bio.Seq.UnknownSeq.reverse_complement()=Bio.Seq.UnknownSeq-class.html#reverse_complement,Function Bio.Seq.reverse_complement()=Bio.Seq-module.html#reverse_complement,Method Bio.SeqRecord.SeqRecord.reverse_complement()=Bio.SeqRecord.SeqRecord-class.html#reverse_complement,Method Bio.motifs.Instances.reverse_complement()=Bio.motifs.Instances-class.html#reverse_complement,Method Bio.motifs.Motif.reverse_complement()=Bio.motifs.Motif-class.html#reverse_complement,Method Bio.motifs.matrix.GenericPositionMatrix.reverse_complement()=Bio.motifs.matrix.GenericPositionMatrix-class.html#reverse_complement"><a title="Bio.Seq.MutableSeq.reverse_complement
Bio.Seq.Seq.reverse_complement
Bio.Seq.UnknownSeq.reverse_complement
Bio.Seq.reverse_complement
Bio.SeqRecord.SeqRecord.reverse_complement
Bio.motifs.Instances.reverse_complement
Bio.motifs.Motif.reverse_complement
Bio.motifs.matrix.GenericPositionMatrix.reverse_complement" class="py-name" href="#" onclick="return doclink('link-231', 'reverse_complement', 'link-231');">reverse_complement</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L735"></a><tt class="py-lineno">735</tt>  <tt class="py-line"> </tt>
<a name="L736"></a><tt class="py-lineno">736</tt>  <tt class="py-line">            <tt class="py-name">result_multiseq</tt><tt class="py-op">.</tt><tt id="link-232" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-232', 'append', 'link-64');">append</a></tt><tt class="py-op">(</tt><tt id="link-233" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-233', 'SeqRecord', 'link-7');">SeqRecord</a></tt><tt class="py-op">(</tt><tt id="link-234" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-234', 'seq', 'link-29');">seq</a></tt><tt class="py-op">,</tt> </tt>
<a name="L737"></a><tt class="py-lineno">737</tt>  <tt class="py-line">                                             <tt id="link-235" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-235', 'id', 'link-24');">id</a></tt><tt class="py-op">=</tt><tt class="py-name">seqid</tt><tt class="py-op">,</tt> </tt>
<a name="L738"></a><tt class="py-lineno">738</tt>  <tt class="py-line">                                             <tt id="link-236" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-236', 'name', 'link-72');">name</a></tt><tt class="py-op">=</tt><tt class="py-name">seqid</tt><tt class="py-op">,</tt> </tt>
<a name="L739"></a><tt class="py-lineno">739</tt>  <tt class="py-line">                                             <tt id="link-237" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-237', 'description', 'link-73');">description</a></tt><tt class="py-op">=</tt><tt class="py-string">""</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L740"></a><tt class="py-lineno">740</tt>  <tt class="py-line"> </tt>
<a name="L741"></a><tt class="py-lineno">741</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-238" class="py-name"><a title="Bio.Align.MultipleSeqAlignment" class="py-name" href="#" onclick="return doclink('link-238', 'MultipleSeqAlignment', 'link-11');">MultipleSeqAlignment</a></tt><tt class="py-op">(</tt><tt class="py-name">result_multiseq</tt><tt class="py-op">)</tt> </tt>
</div><a name="L742"></a><tt class="py-lineno">742</tt>  <tt class="py-line"> </tt>
<a name="MafIndex.__repr__"></a><div id="MafIndex.__repr__-def"><a name="L743"></a><tt class="py-lineno">743</tt> <a class="py-toggle" href="#" id="MafIndex.__repr__-toggle" onclick="return toggle('MafIndex.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafIndex-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIndex.__repr__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafIndex.__repr__-expanded"><a name="L744"></a><tt class="py-lineno">744</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a string representation of the index."""</tt> </tt>
<a name="L745"></a><tt class="py-lineno">745</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">"MafIO.MafIndex(%r, target_seqname=%r)"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_maf_fp</tt><tt class="py-op">.</tt><tt id="link-239" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-239', 'name', 'link-72');">name</a></tt><tt class="py-op">,</tt> </tt>
<a name="L746"></a><tt class="py-lineno">746</tt>  <tt class="py-line">                                                          <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_target_seqname</tt><tt class="py-op">)</tt> </tt>
</div><a name="L747"></a><tt class="py-lineno">747</tt>  <tt class="py-line"> </tt>
<a name="MafIndex.__len__"></a><div id="MafIndex.__len__-def"><a name="L748"></a><tt class="py-lineno">748</tt> <a class="py-toggle" href="#" id="MafIndex.__len__-toggle" onclick="return toggle('MafIndex.__len__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.MafIO.MafIndex-class.html#__len__">__len__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MafIndex.__len__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MafIndex.__len__-expanded"><a name="L749"></a><tt class="py-lineno">749</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the number of records in the index."""</tt> </tt>
<a name="L750"></a><tt class="py-lineno">750</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_record_count</tt> </tt>
</div></div><a name="L751"></a><tt class="py-lineno">751</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Fri Jun 22 16:36:10 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
