<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.SeqIO.SffIO</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.SeqIO-module.html">Package&nbsp;SeqIO</a> ::
        Module&nbsp;SffIO
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.SeqIO.SffIO-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.SeqIO.SffIO-module.html">Module Bio.SeqIO.SffIO</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">   1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2009-2010 by Peter Cock.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">   2</tt>  <tt class="py-line"><tt class="py-comment"># Based on code contributed and copyright 2009 by Jose Blanca (COMAV-UPV).</tt> </tt>
<a name="L3"></a><tt class="py-lineno">   3</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L4"></a><tt class="py-lineno">   4</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L5"></a><tt class="py-lineno">   5</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L6"></a><tt class="py-lineno">   6</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L7"></a><tt class="py-lineno">   7</tt>  <tt class="py-line"><tt class="py-docstring">"""Bio.SeqIO support for the binary Standard Flowgram Format (SFF) file format.</tt> </tt>
<a name="L8"></a><tt class="py-lineno">   8</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L9"></a><tt class="py-lineno">   9</tt>  <tt class="py-line"><tt class="py-docstring">SFF was designed by 454 Life Sciences (Roche), the Whitehead Institute for</tt> </tt>
<a name="L10"></a><tt class="py-lineno">  10</tt>  <tt class="py-line"><tt class="py-docstring">Biomedical Research and the Wellcome Trust Sanger Institute. SFF was also used</tt> </tt>
<a name="L11"></a><tt class="py-lineno">  11</tt>  <tt class="py-line"><tt class="py-docstring">as the native output format from early versions of Ion Torrent's PGM platform</tt> </tt>
<a name="L12"></a><tt class="py-lineno">  12</tt>  <tt class="py-line"><tt class="py-docstring">as well. You are expected to use this module via the Bio.SeqIO functions under</tt> </tt>
<a name="L13"></a><tt class="py-lineno">  13</tt>  <tt class="py-line"><tt class="py-docstring">the format name "sff" (or "sff-trim" as described below).</tt> </tt>
<a name="L14"></a><tt class="py-lineno">  14</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L15"></a><tt class="py-lineno">  15</tt>  <tt class="py-line"><tt class="py-docstring">For example, to iterate over the records in an SFF file,</tt> </tt>
<a name="L16"></a><tt class="py-lineno">  16</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L17"></a><tt class="py-lineno">  17</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L18"></a><tt class="py-lineno">  18</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; for record in SeqIO.parse("Roche/E3MFGYR02_random_10_reads.sff", "sff"):</tt> </tt>
<a name="L19"></a><tt class="py-lineno">  19</tt>  <tt class="py-line"><tt class="py-docstring">    ...     print("%s %i %s..." % (record.id, len(record), record.seq[:20]))</tt> </tt>
<a name="L20"></a><tt class="py-lineno">  20</tt>  <tt class="py-line"><tt class="py-docstring">    ...</tt> </tt>
<a name="L21"></a><tt class="py-lineno">  21</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02JWQ7T 265 tcagGGTCTACATGTTGGTT...</tt> </tt>
<a name="L22"></a><tt class="py-lineno">  22</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02JA6IL 271 tcagTTTTTTTTGGAAAGGA...</tt> </tt>
<a name="L23"></a><tt class="py-lineno">  23</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02JHD4H 310 tcagAAAGACAAGTGGTATC...</tt> </tt>
<a name="L24"></a><tt class="py-lineno">  24</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02GFKUC 299 tcagCGGCCGGGCCTCTCAT...</tt> </tt>
<a name="L25"></a><tt class="py-lineno">  25</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02FTGED 281 tcagTGGTAATGGGGGGAAA...</tt> </tt>
<a name="L26"></a><tt class="py-lineno">  26</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02FR9G7 261 tcagCTCCGTAAGAAGGTGC...</tt> </tt>
<a name="L27"></a><tt class="py-lineno">  27</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02GAZMS 278 tcagAAAGAAGTAAGGTAAA...</tt> </tt>
<a name="L28"></a><tt class="py-lineno">  28</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02HHZ8O 221 tcagACTTTCTTCTTTACCG...</tt> </tt>
<a name="L29"></a><tt class="py-lineno">  29</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02GPGB1 269 tcagAAGCAGTGGTATCAAC...</tt> </tt>
<a name="L30"></a><tt class="py-lineno">  30</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02F7Z7G 219 tcagAATCATCCACTTTTTA...</tt> </tt>
<a name="L31"></a><tt class="py-lineno">  31</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L32"></a><tt class="py-lineno">  32</tt>  <tt class="py-line"><tt class="py-docstring">Each SeqRecord object will contain all the annotation from the SFF file,</tt> </tt>
<a name="L33"></a><tt class="py-lineno">  33</tt>  <tt class="py-line"><tt class="py-docstring">including the PHRED quality scores.</tt> </tt>
<a name="L34"></a><tt class="py-lineno">  34</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L35"></a><tt class="py-lineno">  35</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%s %i" % (record.id, len(record)))</tt> </tt>
<a name="L36"></a><tt class="py-lineno">  36</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02F7Z7G 219</tt> </tt>
<a name="L37"></a><tt class="py-lineno">  37</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%s..." % record.seq[:10])</tt> </tt>
<a name="L38"></a><tt class="py-lineno">  38</tt>  <tt class="py-line"><tt class="py-docstring">    tcagAATCAT...</tt> </tt>
<a name="L39"></a><tt class="py-lineno">  39</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%r..." % (record.letter_annotations["phred_quality"][:10]))</tt> </tt>
<a name="L40"></a><tt class="py-lineno">  40</tt>  <tt class="py-line"><tt class="py-docstring">    [22, 21, 23, 28, 26, 15, 12, 21, 28, 21]...</tt> </tt>
<a name="L41"></a><tt class="py-lineno">  41</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L42"></a><tt class="py-lineno">  42</tt>  <tt class="py-line"><tt class="py-docstring">Notice that the sequence is given in mixed case, the central upper case region</tt> </tt>
<a name="L43"></a><tt class="py-lineno">  43</tt>  <tt class="py-line"><tt class="py-docstring">corresponds to the trimmed sequence. This matches the output of the Roche</tt> </tt>
<a name="L44"></a><tt class="py-lineno">  44</tt>  <tt class="py-line"><tt class="py-docstring">tools (and the 3rd party tool sff_extract) for SFF to FASTA.</tt> </tt>
<a name="L45"></a><tt class="py-lineno">  45</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L46"></a><tt class="py-lineno">  46</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(record.annotations["clip_qual_left"])</tt> </tt>
<a name="L47"></a><tt class="py-lineno">  47</tt>  <tt class="py-line"><tt class="py-docstring">    4</tt> </tt>
<a name="L48"></a><tt class="py-lineno">  48</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(record.annotations["clip_qual_right"])</tt> </tt>
<a name="L49"></a><tt class="py-lineno">  49</tt>  <tt class="py-line"><tt class="py-docstring">    134</tt> </tt>
<a name="L50"></a><tt class="py-lineno">  50</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(record.seq[:4])</tt> </tt>
<a name="L51"></a><tt class="py-lineno">  51</tt>  <tt class="py-line"><tt class="py-docstring">    tcag</tt> </tt>
<a name="L52"></a><tt class="py-lineno">  52</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%s...%s" % (record.seq[4:20], record.seq[120:134]))</tt> </tt>
<a name="L53"></a><tt class="py-lineno">  53</tt>  <tt class="py-line"><tt class="py-docstring">    AATCATCCACTTTTTA...CAAAACACAAACAG</tt> </tt>
<a name="L54"></a><tt class="py-lineno">  54</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(record.seq[134:])</tt> </tt>
<a name="L55"></a><tt class="py-lineno">  55</tt>  <tt class="py-line"><tt class="py-docstring">    atcttatcaacaaaactcaaagttcctaactgagacacgcaacaggggataagacaaggcacacaggggataggnnnnnnnnnnn</tt> </tt>
<a name="L56"></a><tt class="py-lineno">  56</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L57"></a><tt class="py-lineno">  57</tt>  <tt class="py-line"><tt class="py-docstring">The annotations dictionary also contains any adapter clip positions</tt> </tt>
<a name="L58"></a><tt class="py-lineno">  58</tt>  <tt class="py-line"><tt class="py-docstring">(usually zero), and information about the flows. e.g.</tt> </tt>
<a name="L59"></a><tt class="py-lineno">  59</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L60"></a><tt class="py-lineno">  60</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; len(record.annotations)</tt> </tt>
<a name="L61"></a><tt class="py-lineno">  61</tt>  <tt class="py-line"><tt class="py-docstring">    11</tt> </tt>
<a name="L62"></a><tt class="py-lineno">  62</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(record.annotations["flow_key"])</tt> </tt>
<a name="L63"></a><tt class="py-lineno">  63</tt>  <tt class="py-line"><tt class="py-docstring">    TCAG</tt> </tt>
<a name="L64"></a><tt class="py-lineno">  64</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(record.annotations["flow_values"][:10])</tt> </tt>
<a name="L65"></a><tt class="py-lineno">  65</tt>  <tt class="py-line"><tt class="py-docstring">    (83, 1, 128, 7, 4, 84, 6, 106, 3, 172)</tt> </tt>
<a name="L66"></a><tt class="py-lineno">  66</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(len(record.annotations["flow_values"]))</tt> </tt>
<a name="L67"></a><tt class="py-lineno">  67</tt>  <tt class="py-line"><tt class="py-docstring">    400</tt> </tt>
<a name="L68"></a><tt class="py-lineno">  68</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(record.annotations["flow_index"][:10])</tt> </tt>
<a name="L69"></a><tt class="py-lineno">  69</tt>  <tt class="py-line"><tt class="py-docstring">    (1, 2, 3, 2, 2, 0, 3, 2, 3, 3)</tt> </tt>
<a name="L70"></a><tt class="py-lineno">  70</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(len(record.annotations["flow_index"]))</tt> </tt>
<a name="L71"></a><tt class="py-lineno">  71</tt>  <tt class="py-line"><tt class="py-docstring">    219</tt> </tt>
<a name="L72"></a><tt class="py-lineno">  72</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L73"></a><tt class="py-lineno">  73</tt>  <tt class="py-line"><tt class="py-docstring">Note that to convert from a raw reading in flow_values to the corresponding</tt> </tt>
<a name="L74"></a><tt class="py-lineno">  74</tt>  <tt class="py-line"><tt class="py-docstring">homopolymer stretch estimate, the value should be rounded to the nearest 100:</tt> </tt>
<a name="L75"></a><tt class="py-lineno">  75</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L76"></a><tt class="py-lineno">  76</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%r..." % [int(round(value, -2)) // 100</tt> </tt>
<a name="L77"></a><tt class="py-lineno">  77</tt>  <tt class="py-line"><tt class="py-docstring">    ...                  for value in record.annotations["flow_values"][:10]])</tt> </tt>
<a name="L78"></a><tt class="py-lineno">  78</tt>  <tt class="py-line"><tt class="py-docstring">    ...</tt> </tt>
<a name="L79"></a><tt class="py-lineno">  79</tt>  <tt class="py-line"><tt class="py-docstring">    [1, 0, 1, 0, 0, 1, 0, 1, 0, 2]...</tt> </tt>
<a name="L80"></a><tt class="py-lineno">  80</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L81"></a><tt class="py-lineno">  81</tt>  <tt class="py-line"><tt class="py-docstring">If a read name is exactly 14 alphanumeric characters, the annotations</tt> </tt>
<a name="L82"></a><tt class="py-lineno">  82</tt>  <tt class="py-line"><tt class="py-docstring">dictionary will also contain meta-data about the read extracted by</tt> </tt>
<a name="L83"></a><tt class="py-lineno">  83</tt>  <tt class="py-line"><tt class="py-docstring">interpretting the name as a 454 Sequencing System "Universal" Accession</tt> </tt>
<a name="L84"></a><tt class="py-lineno">  84</tt>  <tt class="py-line"><tt class="py-docstring">Number. Note that if a read name happens to be exactly 14 alphanumeric</tt> </tt>
<a name="L85"></a><tt class="py-lineno">  85</tt>  <tt class="py-line"><tt class="py-docstring">characters but was not generated automatically, these annotation records</tt> </tt>
<a name="L86"></a><tt class="py-lineno">  86</tt>  <tt class="py-line"><tt class="py-docstring">will contain nonsense information.</tt> </tt>
<a name="L87"></a><tt class="py-lineno">  87</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L88"></a><tt class="py-lineno">  88</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(record.annotations["region"])</tt> </tt>
<a name="L89"></a><tt class="py-lineno">  89</tt>  <tt class="py-line"><tt class="py-docstring">    2</tt> </tt>
<a name="L90"></a><tt class="py-lineno">  90</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(record.annotations["time"])</tt> </tt>
<a name="L91"></a><tt class="py-lineno">  91</tt>  <tt class="py-line"><tt class="py-docstring">    [2008, 1, 9, 16, 16, 0]</tt> </tt>
<a name="L92"></a><tt class="py-lineno">  92</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(record.annotations["coords"])</tt> </tt>
<a name="L93"></a><tt class="py-lineno">  93</tt>  <tt class="py-line"><tt class="py-docstring">    (2434, 1658)</tt> </tt>
<a name="L94"></a><tt class="py-lineno">  94</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L95"></a><tt class="py-lineno">  95</tt>  <tt class="py-line"><tt class="py-docstring">As a convenience method, you can read the file with SeqIO format name "sff-trim"</tt> </tt>
<a name="L96"></a><tt class="py-lineno">  96</tt>  <tt class="py-line"><tt class="py-docstring">instead of "sff" to get just the trimmed sequences (without any annotation</tt> </tt>
<a name="L97"></a><tt class="py-lineno">  97</tt>  <tt class="py-line"><tt class="py-docstring">except for the PHRED quality scores and anything encoded in the read names):</tt> </tt>
<a name="L98"></a><tt class="py-lineno">  98</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L99"></a><tt class="py-lineno">  99</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L100"></a><tt class="py-lineno"> 100</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; for record in SeqIO.parse("Roche/E3MFGYR02_random_10_reads.sff", "sff-trim"):</tt> </tt>
<a name="L101"></a><tt class="py-lineno"> 101</tt>  <tt class="py-line"><tt class="py-docstring">    ...     print("%s %i %s..." % (record.id, len(record), record.seq[:20]))</tt> </tt>
<a name="L102"></a><tt class="py-lineno"> 102</tt>  <tt class="py-line"><tt class="py-docstring">    ...</tt> </tt>
<a name="L103"></a><tt class="py-lineno"> 103</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02JWQ7T 260 GGTCTACATGTTGGTTAACC...</tt> </tt>
<a name="L104"></a><tt class="py-lineno"> 104</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02JA6IL 265 TTTTTTTTGGAAAGGAAAAC...</tt> </tt>
<a name="L105"></a><tt class="py-lineno"> 105</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02JHD4H 292 AAAGACAAGTGGTATCAACG...</tt> </tt>
<a name="L106"></a><tt class="py-lineno"> 106</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02GFKUC 295 CGGCCGGGCCTCTCATCGGT...</tt> </tt>
<a name="L107"></a><tt class="py-lineno"> 107</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02FTGED 277 TGGTAATGGGGGGAAATTTA...</tt> </tt>
<a name="L108"></a><tt class="py-lineno"> 108</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02FR9G7 256 CTCCGTAAGAAGGTGCTGCC...</tt> </tt>
<a name="L109"></a><tt class="py-lineno"> 109</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02GAZMS 271 AAAGAAGTAAGGTAAATAAC...</tt> </tt>
<a name="L110"></a><tt class="py-lineno"> 110</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02HHZ8O 150 ACTTTCTTCTTTACCGTAAC...</tt> </tt>
<a name="L111"></a><tt class="py-lineno"> 111</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02GPGB1 221 AAGCAGTGGTATCAACGCAG...</tt> </tt>
<a name="L112"></a><tt class="py-lineno"> 112</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02F7Z7G 130 AATCATCCACTTTTTAACGT...</tt> </tt>
<a name="L113"></a><tt class="py-lineno"> 113</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L114"></a><tt class="py-lineno"> 114</tt>  <tt class="py-line"><tt class="py-docstring">Looking at the final record in more detail, note how this differs to the</tt> </tt>
<a name="L115"></a><tt class="py-lineno"> 115</tt>  <tt class="py-line"><tt class="py-docstring">example above:</tt> </tt>
<a name="L116"></a><tt class="py-lineno"> 116</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L117"></a><tt class="py-lineno"> 117</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%s %i" % (record.id, len(record)))</tt> </tt>
<a name="L118"></a><tt class="py-lineno"> 118</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02F7Z7G 130</tt> </tt>
<a name="L119"></a><tt class="py-lineno"> 119</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%s..." % record.seq[:10])</tt> </tt>
<a name="L120"></a><tt class="py-lineno"> 120</tt>  <tt class="py-line"><tt class="py-docstring">    AATCATCCAC...</tt> </tt>
<a name="L121"></a><tt class="py-lineno"> 121</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%r..." % record.letter_annotations["phred_quality"][:10])</tt> </tt>
<a name="L122"></a><tt class="py-lineno"> 122</tt>  <tt class="py-line"><tt class="py-docstring">    [26, 15, 12, 21, 28, 21, 36, 28, 27, 27]...</tt> </tt>
<a name="L123"></a><tt class="py-lineno"> 123</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; len(record.annotations)</tt> </tt>
<a name="L124"></a><tt class="py-lineno"> 124</tt>  <tt class="py-line"><tt class="py-docstring">    3</tt> </tt>
<a name="L125"></a><tt class="py-lineno"> 125</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(record.annotations["region"])</tt> </tt>
<a name="L126"></a><tt class="py-lineno"> 126</tt>  <tt class="py-line"><tt class="py-docstring">    2</tt> </tt>
<a name="L127"></a><tt class="py-lineno"> 127</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(record.annotations["coords"])</tt> </tt>
<a name="L128"></a><tt class="py-lineno"> 128</tt>  <tt class="py-line"><tt class="py-docstring">    (2434, 1658)</tt> </tt>
<a name="L129"></a><tt class="py-lineno"> 129</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(record.annotations["time"])</tt> </tt>
<a name="L130"></a><tt class="py-lineno"> 130</tt>  <tt class="py-line"><tt class="py-docstring">    [2008, 1, 9, 16, 16, 0]</tt> </tt>
<a name="L131"></a><tt class="py-lineno"> 131</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L132"></a><tt class="py-lineno"> 132</tt>  <tt class="py-line"><tt class="py-docstring">You might use the Bio.SeqIO.convert() function to convert the (trimmed) SFF</tt> </tt>
<a name="L133"></a><tt class="py-lineno"> 133</tt>  <tt class="py-line"><tt class="py-docstring">reads into a FASTQ file (or a FASTA file and a QUAL file), e.g.</tt> </tt>
<a name="L134"></a><tt class="py-lineno"> 134</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L135"></a><tt class="py-lineno"> 135</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L136"></a><tt class="py-lineno"> 136</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; try:</tt> </tt>
<a name="L137"></a><tt class="py-lineno"> 137</tt>  <tt class="py-line"><tt class="py-docstring">    ...     from StringIO import StringIO # Python 2</tt> </tt>
<a name="L138"></a><tt class="py-lineno"> 138</tt>  <tt class="py-line"><tt class="py-docstring">    ... except ImportError:</tt> </tt>
<a name="L139"></a><tt class="py-lineno"> 139</tt>  <tt class="py-line"><tt class="py-docstring">    ...     from io import StringIO # Python 3</tt> </tt>
<a name="L140"></a><tt class="py-lineno"> 140</tt>  <tt class="py-line"><tt class="py-docstring">    ...</tt> </tt>
<a name="L141"></a><tt class="py-lineno"> 141</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; out_handle = StringIO()</tt> </tt>
<a name="L142"></a><tt class="py-lineno"> 142</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; count = SeqIO.convert("Roche/E3MFGYR02_random_10_reads.sff", "sff",</tt> </tt>
<a name="L143"></a><tt class="py-lineno"> 143</tt>  <tt class="py-line"><tt class="py-docstring">    ...                       out_handle, "fastq")</tt> </tt>
<a name="L144"></a><tt class="py-lineno"> 144</tt>  <tt class="py-line"><tt class="py-docstring">    ...</tt> </tt>
<a name="L145"></a><tt class="py-lineno"> 145</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("Converted %i records" % count)</tt> </tt>
<a name="L146"></a><tt class="py-lineno"> 146</tt>  <tt class="py-line"><tt class="py-docstring">    Converted 10 records</tt> </tt>
<a name="L147"></a><tt class="py-lineno"> 147</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L148"></a><tt class="py-lineno"> 148</tt>  <tt class="py-line"><tt class="py-docstring">The output FASTQ file would start like this:</tt> </tt>
<a name="L149"></a><tt class="py-lineno"> 149</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L150"></a><tt class="py-lineno"> 150</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%s..." % out_handle.getvalue()[:50])</tt> </tt>
<a name="L151"></a><tt class="py-lineno"> 151</tt>  <tt class="py-line"><tt class="py-docstring">    @E3MFGYR02JWQ7T</tt> </tt>
<a name="L152"></a><tt class="py-lineno"> 152</tt>  <tt class="py-line"><tt class="py-docstring">    tcagGGTCTACATGTTGGTTAACCCGTACTGATT...</tt> </tt>
<a name="L153"></a><tt class="py-lineno"> 153</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L154"></a><tt class="py-lineno"> 154</tt>  <tt class="py-line"><tt class="py-docstring">Bio.SeqIO.index() provides memory efficient random access to the reads in an</tt> </tt>
<a name="L155"></a><tt class="py-lineno"> 155</tt>  <tt class="py-line"><tt class="py-docstring">SFF file by name. SFF files can include an index within the file, which can</tt> </tt>
<a name="L156"></a><tt class="py-lineno"> 156</tt>  <tt class="py-line"><tt class="py-docstring">be read in making this very fast. If the index is missing (or in a format not</tt> </tt>
<a name="L157"></a><tt class="py-lineno"> 157</tt>  <tt class="py-line"><tt class="py-docstring">yet supported in Biopython) the file is indexed by scanning all the reads -</tt> </tt>
<a name="L158"></a><tt class="py-lineno"> 158</tt>  <tt class="py-line"><tt class="py-docstring">which is a little slower. For example,</tt> </tt>
<a name="L159"></a><tt class="py-lineno"> 159</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L160"></a><tt class="py-lineno"> 160</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L161"></a><tt class="py-lineno"> 161</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; reads = SeqIO.index("Roche/E3MFGYR02_random_10_reads.sff", "sff")</tt> </tt>
<a name="L162"></a><tt class="py-lineno"> 162</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; record = reads["E3MFGYR02JHD4H"]</tt> </tt>
<a name="L163"></a><tt class="py-lineno"> 163</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%s %i %s..." % (record.id, len(record), record.seq[:20]))</tt> </tt>
<a name="L164"></a><tt class="py-lineno"> 164</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02JHD4H 310 tcagAAAGACAAGTGGTATC...</tt> </tt>
<a name="L165"></a><tt class="py-lineno"> 165</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; reads.close()</tt> </tt>
<a name="L166"></a><tt class="py-lineno"> 166</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L167"></a><tt class="py-lineno"> 167</tt>  <tt class="py-line"><tt class="py-docstring">Or, using the trimmed reads:</tt> </tt>
<a name="L168"></a><tt class="py-lineno"> 168</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L169"></a><tt class="py-lineno"> 169</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L170"></a><tt class="py-lineno"> 170</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; reads = SeqIO.index("Roche/E3MFGYR02_random_10_reads.sff", "sff-trim")</tt> </tt>
<a name="L171"></a><tt class="py-lineno"> 171</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; record = reads["E3MFGYR02JHD4H"]</tt> </tt>
<a name="L172"></a><tt class="py-lineno"> 172</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%s %i %s..." % (record.id, len(record), record.seq[:20]))</tt> </tt>
<a name="L173"></a><tt class="py-lineno"> 173</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02JHD4H 292 AAAGACAAGTGGTATCAACG...</tt> </tt>
<a name="L174"></a><tt class="py-lineno"> 174</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; reads.close()</tt> </tt>
<a name="L175"></a><tt class="py-lineno"> 175</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L176"></a><tt class="py-lineno"> 176</tt>  <tt class="py-line"><tt class="py-docstring">You can also use the Bio.SeqIO.write() function with the "sff" format. Note</tt> </tt>
<a name="L177"></a><tt class="py-lineno"> 177</tt>  <tt class="py-line"><tt class="py-docstring">that this requires all the flow information etc, and thus is probably only</tt> </tt>
<a name="L178"></a><tt class="py-lineno"> 178</tt>  <tt class="py-line"><tt class="py-docstring">useful for SeqRecord objects originally from reading another SFF file (and</tt> </tt>
<a name="L179"></a><tt class="py-lineno"> 179</tt>  <tt class="py-line"><tt class="py-docstring">not the trimmed SeqRecord objects from parsing an SFF file as "sff-trim").</tt> </tt>
<a name="L180"></a><tt class="py-lineno"> 180</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L181"></a><tt class="py-lineno"> 181</tt>  <tt class="py-line"><tt class="py-docstring">As an example, let's pretend this example SFF file represents some DNA which</tt> </tt>
<a name="L182"></a><tt class="py-lineno"> 182</tt>  <tt class="py-line"><tt class="py-docstring">was pre-amplified with a PCR primers AAAGANNNNN. The following script would</tt> </tt>
<a name="L183"></a><tt class="py-lineno"> 183</tt>  <tt class="py-line"><tt class="py-docstring">produce a sub-file containing all those reads whose post-quality clipping</tt> </tt>
<a name="L184"></a><tt class="py-lineno"> 184</tt>  <tt class="py-line"><tt class="py-docstring">region (i.e. the sequence after trimming) starts with AAAGA exactly (the non-</tt> </tt>
<a name="L185"></a><tt class="py-lineno"> 185</tt>  <tt class="py-line"><tt class="py-docstring">degenerate bit of this pretend primer):</tt> </tt>
<a name="L186"></a><tt class="py-lineno"> 186</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L187"></a><tt class="py-lineno"> 187</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L188"></a><tt class="py-lineno"> 188</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; records = (record for record in</tt> </tt>
<a name="L189"></a><tt class="py-lineno"> 189</tt>  <tt class="py-line"><tt class="py-docstring">    ...            SeqIO.parse("Roche/E3MFGYR02_random_10_reads.sff", "sff")</tt> </tt>
<a name="L190"></a><tt class="py-lineno"> 190</tt>  <tt class="py-line"><tt class="py-docstring">    ...            if record.seq[record.annotations["clip_qual_left"]:].startswith("AAAGA"))</tt> </tt>
<a name="L191"></a><tt class="py-lineno"> 191</tt>  <tt class="py-line"><tt class="py-docstring">    ...</tt> </tt>
<a name="L192"></a><tt class="py-lineno"> 192</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; count = SeqIO.write(records, "temp_filtered.sff", "sff")</tt> </tt>
<a name="L193"></a><tt class="py-lineno"> 193</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("Selected %i records" % count)</tt> </tt>
<a name="L194"></a><tt class="py-lineno"> 194</tt>  <tt class="py-line"><tt class="py-docstring">    Selected 2 records</tt> </tt>
<a name="L195"></a><tt class="py-lineno"> 195</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L196"></a><tt class="py-lineno"> 196</tt>  <tt class="py-line"><tt class="py-docstring">Of course, for an assembly you would probably want to remove these primers.</tt> </tt>
<a name="L197"></a><tt class="py-lineno"> 197</tt>  <tt class="py-line"><tt class="py-docstring">If you want FASTA or FASTQ output, you could just slice the SeqRecord. However,</tt> </tt>
<a name="L198"></a><tt class="py-lineno"> 198</tt>  <tt class="py-line"><tt class="py-docstring">if you want SFF output we have to preserve all the flow information - the trick</tt> </tt>
<a name="L199"></a><tt class="py-lineno"> 199</tt>  <tt class="py-line"><tt class="py-docstring">is just to adjust the left clip position!</tt> </tt>
<a name="L200"></a><tt class="py-lineno"> 200</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L201"></a><tt class="py-lineno"> 201</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L202"></a><tt class="py-lineno"> 202</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; def filter_and_trim(records, primer):</tt> </tt>
<a name="L203"></a><tt class="py-lineno"> 203</tt>  <tt class="py-line"><tt class="py-docstring">    ...     for record in records:</tt> </tt>
<a name="L204"></a><tt class="py-lineno"> 204</tt>  <tt class="py-line"><tt class="py-docstring">    ...         if record.seq[record.annotations["clip_qual_left"]:].startswith(primer):</tt> </tt>
<a name="L205"></a><tt class="py-lineno"> 205</tt>  <tt class="py-line"><tt class="py-docstring">    ...             record.annotations["clip_qual_left"] += len(primer)</tt> </tt>
<a name="L206"></a><tt class="py-lineno"> 206</tt>  <tt class="py-line"><tt class="py-docstring">    ...             yield record</tt> </tt>
<a name="L207"></a><tt class="py-lineno"> 207</tt>  <tt class="py-line"><tt class="py-docstring">    ...</tt> </tt>
<a name="L208"></a><tt class="py-lineno"> 208</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; records = SeqIO.parse("Roche/E3MFGYR02_random_10_reads.sff", "sff")</tt> </tt>
<a name="L209"></a><tt class="py-lineno"> 209</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; count = SeqIO.write(filter_and_trim(records, "AAAGA"),</tt> </tt>
<a name="L210"></a><tt class="py-lineno"> 210</tt>  <tt class="py-line"><tt class="py-docstring">    ...                     "temp_filtered.sff", "sff")</tt> </tt>
<a name="L211"></a><tt class="py-lineno"> 211</tt>  <tt class="py-line"><tt class="py-docstring">    ...</tt> </tt>
<a name="L212"></a><tt class="py-lineno"> 212</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("Selected %i records" % count)</tt> </tt>
<a name="L213"></a><tt class="py-lineno"> 213</tt>  <tt class="py-line"><tt class="py-docstring">    Selected 2 records</tt> </tt>
<a name="L214"></a><tt class="py-lineno"> 214</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L215"></a><tt class="py-lineno"> 215</tt>  <tt class="py-line"><tt class="py-docstring">We can check the results, note the lower case clipped region now includes the "AAAGA"</tt> </tt>
<a name="L216"></a><tt class="py-lineno"> 216</tt>  <tt class="py-line"><tt class="py-docstring">sequence:</tt> </tt>
<a name="L217"></a><tt class="py-lineno"> 217</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L218"></a><tt class="py-lineno"> 218</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; for record in SeqIO.parse("temp_filtered.sff", "sff"):</tt> </tt>
<a name="L219"></a><tt class="py-lineno"> 219</tt>  <tt class="py-line"><tt class="py-docstring">    ...     print("%s %i %s..." % (record.id, len(record), record.seq[:20]))</tt> </tt>
<a name="L220"></a><tt class="py-lineno"> 220</tt>  <tt class="py-line"><tt class="py-docstring">    ...</tt> </tt>
<a name="L221"></a><tt class="py-lineno"> 221</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02JHD4H 310 tcagaaagaCAAGTGGTATC...</tt> </tt>
<a name="L222"></a><tt class="py-lineno"> 222</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02GAZMS 278 tcagaaagaAGTAAGGTAAA...</tt> </tt>
<a name="L223"></a><tt class="py-lineno"> 223</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; for record in SeqIO.parse("temp_filtered.sff", "sff-trim"):</tt> </tt>
<a name="L224"></a><tt class="py-lineno"> 224</tt>  <tt class="py-line"><tt class="py-docstring">    ...     print("%s %i %s..." % (record.id, len(record), record.seq[:20]))</tt> </tt>
<a name="L225"></a><tt class="py-lineno"> 225</tt>  <tt class="py-line"><tt class="py-docstring">    ...</tt> </tt>
<a name="L226"></a><tt class="py-lineno"> 226</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02JHD4H 287 CAAGTGGTATCAACGCAGAG...</tt> </tt>
<a name="L227"></a><tt class="py-lineno"> 227</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02GAZMS 266 AGTAAGGTAAATAACAAACG...</tt> </tt>
<a name="L228"></a><tt class="py-lineno"> 228</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; import os</tt> </tt>
<a name="L229"></a><tt class="py-lineno"> 229</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; os.remove("temp_filtered.sff")</tt> </tt>
<a name="L230"></a><tt class="py-lineno"> 230</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L231"></a><tt class="py-lineno"> 231</tt>  <tt class="py-line"><tt class="py-docstring">For a description of the file format, please see the Roche manuals and:</tt> </tt>
<a name="L232"></a><tt class="py-lineno"> 232</tt>  <tt class="py-line"><tt class="py-docstring">http://www.ncbi.nlm.nih.gov/Traces/trace.cgi?cmd=show&amp;f=formats&amp;m=doc&amp;s=formats</tt> </tt>
<a name="L233"></a><tt class="py-lineno"> 233</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L234"></a><tt class="py-lineno"> 234</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L235"></a><tt class="py-lineno"> 235</tt>  <tt class="py-line"> </tt>
<a name="L236"></a><tt class="py-lineno"> 236</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">__future__</tt> <tt class="py-keyword">import</tt> <tt class="py-name">print_function</tt> </tt>
<a name="L237"></a><tt class="py-lineno"> 237</tt>  <tt class="py-line"> </tt>
<a name="L238"></a><tt class="py-lineno"> 238</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio.SeqIO=Bio.SeqIO-module.html"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-1', 'SeqIO', 'link-1');">SeqIO</a></tt><tt class="py-op">.</tt><tt id="link-2" class="py-name" targets="Module Bio.AlignIO.Interfaces=Bio.AlignIO.Interfaces-module.html,Module Bio.SeqIO.Interfaces=Bio.SeqIO.Interfaces-module.html"><a title="Bio.AlignIO.Interfaces
Bio.SeqIO.Interfaces" class="py-name" href="#" onclick="return doclink('link-2', 'Interfaces', 'link-2');">Interfaces</a></tt> <tt class="py-keyword">import</tt> <tt id="link-3" class="py-name" targets="Class Bio.SeqIO.Interfaces.SequenceWriter=Bio.SeqIO.Interfaces.SequenceWriter-class.html"><a title="Bio.SeqIO.Interfaces.SequenceWriter" class="py-name" href="#" onclick="return doclink('link-3', 'SequenceWriter', 'link-3');">SequenceWriter</a></tt> </tt>
<a name="L239"></a><tt class="py-lineno"> 239</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-4" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-4', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-5" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-5', 'Alphabet', 'link-5');">Alphabet</a></tt> </tt>
<a name="L240"></a><tt class="py-lineno"> 240</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-6" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-6', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-7" class="py-name" targets="Module Bio.Seq=Bio.Seq-module.html,Class Bio.Seq.Seq=Bio.Seq.Seq-class.html"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-7', 'Seq', 'link-7');">Seq</a></tt> <tt class="py-keyword">import</tt> <tt id="link-8" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-8', 'Seq', 'link-7');">Seq</a></tt> </tt>
<a name="L241"></a><tt class="py-lineno"> 241</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-9" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-9', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-10" class="py-name" targets="Module Bio.SeqRecord=Bio.SeqRecord-module.html,Class Bio.SeqRecord.SeqRecord=Bio.SeqRecord.SeqRecord-class.html"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-10', 'SeqRecord', 'link-10');">SeqRecord</a></tt> <tt class="py-keyword">import</tt> <tt id="link-11" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-11', 'SeqRecord', 'link-10');">SeqRecord</a></tt> </tt>
<a name="L242"></a><tt class="py-lineno"> 242</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">struct</tt> </tt>
<a name="L243"></a><tt class="py-lineno"> 243</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">sys</tt> </tt>
<a name="L244"></a><tt class="py-lineno"> 244</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">re</tt> </tt>
<a name="L245"></a><tt class="py-lineno"> 245</tt>  <tt class="py-line"> </tt>
<a name="L246"></a><tt class="py-lineno"> 246</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-12" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-12', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-13" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-13', '_py3k', 'link-13');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt id="link-14" class="py-name" targets="Function Bio._py3k._bytes_to_string()=Bio._py3k-module.html#_bytes_to_string"><a title="Bio._py3k._bytes_to_string" class="py-name" href="#" onclick="return doclink('link-14', '_bytes_to_string', 'link-14');">_bytes_to_string</a></tt><tt class="py-op">,</tt> <tt id="link-15" class="py-name" targets="Function Bio._py3k._as_bytes()=Bio._py3k-module.html#_as_bytes"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-15', '_as_bytes', 'link-15');">_as_bytes</a></tt> </tt>
<a name="L247"></a><tt class="py-lineno"> 247</tt>  <tt class="py-line"><tt id="link-16" class="py-name" targets="Variable Bio.SeqIO.SffIO._null=Bio.SeqIO.SffIO-module.html#_null"><a title="Bio.SeqIO.SffIO._null" class="py-name" href="#" onclick="return doclink('link-16', '_null', 'link-16');">_null</a></tt> <tt class="py-op">=</tt> <tt id="link-17" class="py-name" targets="Variable Bio.Graphics.ColorSpiral.ColorSpiral.b=Bio.Graphics.ColorSpiral.ColorSpiral-class.html#b"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-17', 'b', 'link-17');">b</a></tt><tt class="py-string">"\0"</tt> </tt>
<a name="L248"></a><tt class="py-lineno"> 248</tt>  <tt class="py-line"><tt id="link-18" class="py-name" targets="Variable Bio.SeqIO.SffIO._sff=Bio.SeqIO.SffIO-module.html#_sff"><a title="Bio.SeqIO.SffIO._sff" class="py-name" href="#" onclick="return doclink('link-18', '_sff', 'link-18');">_sff</a></tt> <tt class="py-op">=</tt> <tt id="link-19" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-19', 'b', 'link-17');">b</a></tt><tt class="py-string">".sff"</tt> </tt>
<a name="L249"></a><tt class="py-lineno"> 249</tt>  <tt class="py-line"><tt id="link-20" class="py-name" targets="Variable Bio.SeqIO.SffIO._hsh=Bio.SeqIO.SffIO-module.html#_hsh"><a title="Bio.SeqIO.SffIO._hsh" class="py-name" href="#" onclick="return doclink('link-20', '_hsh', 'link-20');">_hsh</a></tt> <tt class="py-op">=</tt> <tt id="link-21" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-21', 'b', 'link-17');">b</a></tt><tt class="py-string">".hsh"</tt> </tt>
<a name="L250"></a><tt class="py-lineno"> 250</tt>  <tt class="py-line"><tt id="link-22" class="py-name" targets="Variable Bio.SeqIO.SffIO._srt=Bio.SeqIO.SffIO-module.html#_srt"><a title="Bio.SeqIO.SffIO._srt" class="py-name" href="#" onclick="return doclink('link-22', '_srt', 'link-22');">_srt</a></tt> <tt class="py-op">=</tt> <tt id="link-23" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-23', 'b', 'link-17');">b</a></tt><tt class="py-string">".srt"</tt> </tt>
<a name="L251"></a><tt class="py-lineno"> 251</tt>  <tt class="py-line"><tt id="link-24" class="py-name" targets="Variable Bio.SeqIO.SffIO._mft=Bio.SeqIO.SffIO-module.html#_mft"><a title="Bio.SeqIO.SffIO._mft" class="py-name" href="#" onclick="return doclink('link-24', '_mft', 'link-24');">_mft</a></tt> <tt class="py-op">=</tt> <tt id="link-25" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-25', 'b', 'link-17');">b</a></tt><tt class="py-string">".mft"</tt> </tt>
<a name="L252"></a><tt class="py-lineno"> 252</tt>  <tt class="py-line"><tt id="link-26" class="py-name" targets="Variable Bio.SeqIO.SffIO._flag=Bio.SeqIO.SffIO-module.html#_flag"><a title="Bio.SeqIO.SffIO._flag" class="py-name" href="#" onclick="return doclink('link-26', '_flag', 'link-26');">_flag</a></tt> <tt class="py-op">=</tt> <tt id="link-27" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-27', 'b', 'link-17');">b</a></tt><tt class="py-string">"\xff"</tt> </tt>
<a name="L253"></a><tt class="py-lineno"> 253</tt>  <tt class="py-line"> </tt>
<a name="L254"></a><tt class="py-lineno"> 254</tt>  <tt class="py-line"><tt class="py-name">__docformat__</tt> <tt class="py-op">=</tt> <tt class="py-string">"restructuredtext en"</tt> </tt>
<a name="L255"></a><tt class="py-lineno"> 255</tt>  <tt class="py-line"> </tt>
<a name="L256"></a><tt class="py-lineno"> 256</tt>  <tt class="py-line"> </tt>
<a name="_check_mode"></a><div id="_check_mode-def"><a name="L257"></a><tt class="py-lineno"> 257</tt> <a class="py-toggle" href="#" id="_check_mode-toggle" onclick="return toggle('_check_mode');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO-module.html#_check_mode">_check_mode</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_check_mode-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_check_mode-expanded"><a name="L258"></a><tt class="py-lineno"> 258</tt>  <tt class="py-line">    <tt class="py-docstring">"""Ensure handle not opened in text mode.</tt> </tt>
<a name="L259"></a><tt class="py-lineno"> 259</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L260"></a><tt class="py-lineno"> 260</tt>  <tt class="py-line"><tt class="py-docstring">    Ensures mode is not set for Universal new line</tt> </tt>
<a name="L261"></a><tt class="py-lineno"> 261</tt>  <tt class="py-line"><tt class="py-docstring">    and ensures mode is binary for Windows</tt> </tt>
<a name="L262"></a><tt class="py-lineno"> 262</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L263"></a><tt class="py-lineno"> 263</tt>  <tt class="py-line">    <tt class="py-comment"># TODO - Does this need to be stricter under Python 3?</tt> </tt>
<a name="L264"></a><tt class="py-lineno"> 264</tt>  <tt class="py-line">    <tt class="py-name">mode</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L265"></a><tt class="py-lineno"> 265</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-string">"mode"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L266"></a><tt class="py-lineno"> 266</tt>  <tt class="py-line">        <tt class="py-name">mode</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt class="py-name">mode</tt> </tt>
<a name="L267"></a><tt class="py-lineno"> 267</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">mode</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L268"></a><tt class="py-lineno"> 268</tt>  <tt class="py-line">            <tt class="py-comment"># gzip.open(...) does this, fine</tt> </tt>
<a name="L269"></a><tt class="py-lineno"> 269</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L270"></a><tt class="py-lineno"> 270</tt>  <tt class="py-line">        <tt class="py-name">mode</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">mode</tt><tt class="py-op">)</tt> </tt>
<a name="L271"></a><tt class="py-lineno"> 271</tt>  <tt class="py-line"> </tt>
<a name="L272"></a><tt class="py-lineno"> 272</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">mode</tt> <tt class="py-keyword">and</tt> <tt class="py-string">"U"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">mode</tt><tt class="py-op">.</tt><tt id="link-28" class="py-name" targets="Method Bio.Seq.Seq.upper()=Bio.Seq.Seq-class.html#upper,Method Bio.Seq.UnknownSeq.upper()=Bio.Seq.UnknownSeq-class.html#upper,Method Bio.SeqRecord.SeqRecord.upper()=Bio.SeqRecord.SeqRecord-class.html#upper"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-28', 'upper', 'link-28');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L273"></a><tt class="py-lineno"> 273</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"SFF files must NOT be opened in universal new "</tt> </tt>
<a name="L274"></a><tt class="py-lineno"> 274</tt>  <tt class="py-line">                         <tt class="py-string">"lines mode. Binary mode is recommended (although "</tt> </tt>
<a name="L275"></a><tt class="py-lineno"> 275</tt>  <tt class="py-line">                         <tt class="py-string">"on Unix the default mode is also fine)."</tt><tt class="py-op">)</tt> </tt>
<a name="L276"></a><tt class="py-lineno"> 276</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">mode</tt> <tt class="py-keyword">and</tt> <tt class="py-string">"B"</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">mode</tt><tt class="py-op">.</tt><tt id="link-29" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-29', 'upper', 'link-28');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> \ </tt>
<a name="L277"></a><tt class="py-lineno"> 277</tt>  <tt class="py-line">            <tt class="py-keyword">and</tt> <tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">platform</tt> <tt class="py-op">==</tt> <tt class="py-string">"win32"</tt><tt class="py-op">:</tt> </tt>
<a name="L278"></a><tt class="py-lineno"> 278</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"SFF files must be opened in binary mode on Windows"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L279"></a><tt class="py-lineno"> 279</tt>  <tt class="py-line"> </tt>
<a name="L280"></a><tt class="py-lineno"> 280</tt>  <tt class="py-line"> </tt>
<a name="_sff_file_header"></a><div id="_sff_file_header-def"><a name="L281"></a><tt class="py-lineno"> 281</tt> <a class="py-toggle" href="#" id="_sff_file_header-toggle" onclick="return toggle('_sff_file_header');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO-module.html#_sff_file_header">_sff_file_header</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_sff_file_header-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_sff_file_header-expanded"><a name="L282"></a><tt class="py-lineno"> 282</tt>  <tt class="py-line">    <tt class="py-docstring">"""Read in an SFF file header (PRIVATE).</tt> </tt>
<a name="L283"></a><tt class="py-lineno"> 283</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L284"></a><tt class="py-lineno"> 284</tt>  <tt class="py-line"><tt class="py-docstring">    Assumes the handle is at the start of the file, will read forwards</tt> </tt>
<a name="L285"></a><tt class="py-lineno"> 285</tt>  <tt class="py-line"><tt class="py-docstring">    though the header and leave the handle pointing at the first record.</tt> </tt>
<a name="L286"></a><tt class="py-lineno"> 286</tt>  <tt class="py-line"><tt class="py-docstring">    Returns a tuple of values from the header (header_length, index_offset,</tt> </tt>
<a name="L287"></a><tt class="py-lineno"> 287</tt>  <tt class="py-line"><tt class="py-docstring">    index_length, number_of_reads, flows_per_read, flow_chars, key_sequence)</tt> </tt>
<a name="L288"></a><tt class="py-lineno"> 288</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L289"></a><tt class="py-lineno"> 289</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; with open("Roche/greek.sff", "rb") as handle:</tt> </tt>
<a name="L290"></a><tt class="py-lineno"> 290</tt>  <tt class="py-line"><tt class="py-docstring">    ...     values = _sff_file_header(handle)</tt> </tt>
<a name="L291"></a><tt class="py-lineno"> 291</tt>  <tt class="py-line"><tt class="py-docstring">    ...</tt> </tt>
<a name="L292"></a><tt class="py-lineno"> 292</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(values[0])</tt> </tt>
<a name="L293"></a><tt class="py-lineno"> 293</tt>  <tt class="py-line"><tt class="py-docstring">    840</tt> </tt>
<a name="L294"></a><tt class="py-lineno"> 294</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(values[1])</tt> </tt>
<a name="L295"></a><tt class="py-lineno"> 295</tt>  <tt class="py-line"><tt class="py-docstring">    65040</tt> </tt>
<a name="L296"></a><tt class="py-lineno"> 296</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(values[2])</tt> </tt>
<a name="L297"></a><tt class="py-lineno"> 297</tt>  <tt class="py-line"><tt class="py-docstring">    256</tt> </tt>
<a name="L298"></a><tt class="py-lineno"> 298</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(values[3])</tt> </tt>
<a name="L299"></a><tt class="py-lineno"> 299</tt>  <tt class="py-line"><tt class="py-docstring">    24</tt> </tt>
<a name="L300"></a><tt class="py-lineno"> 300</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(values[4])</tt> </tt>
<a name="L301"></a><tt class="py-lineno"> 301</tt>  <tt class="py-line"><tt class="py-docstring">    800</tt> </tt>
<a name="L302"></a><tt class="py-lineno"> 302</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; values[-1]</tt> </tt>
<a name="L303"></a><tt class="py-lineno"> 303</tt>  <tt class="py-line"><tt class="py-docstring">    'TCAG'</tt> </tt>
<a name="L304"></a><tt class="py-lineno"> 304</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L305"></a><tt class="py-lineno"> 305</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L306"></a><tt class="py-lineno"> 306</tt>  <tt class="py-line">    <tt id="link-30" class="py-name" targets="Function Bio.SeqIO.SffIO._check_mode()=Bio.SeqIO.SffIO-module.html#_check_mode"><a title="Bio.SeqIO.SffIO._check_mode" class="py-name" href="#" onclick="return doclink('link-30', '_check_mode', 'link-30');">_check_mode</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L307"></a><tt class="py-lineno"> 307</tt>  <tt class="py-line">    <tt class="py-comment"># file header (part one)</tt> </tt>
<a name="L308"></a><tt class="py-lineno"> 308</tt>  <tt class="py-line">    <tt class="py-comment"># use big endiean encdoing   &gt;</tt> </tt>
<a name="L309"></a><tt class="py-lineno"> 309</tt>  <tt class="py-line">    <tt class="py-comment"># magic_number               I</tt> </tt>
<a name="L310"></a><tt class="py-lineno"> 310</tt>  <tt class="py-line">    <tt class="py-comment"># version                    4B</tt> </tt>
<a name="L311"></a><tt class="py-lineno"> 311</tt>  <tt class="py-line">    <tt class="py-comment"># index_offset               Q</tt> </tt>
<a name="L312"></a><tt class="py-lineno"> 312</tt>  <tt class="py-line">    <tt class="py-comment"># index_length               I</tt> </tt>
<a name="L313"></a><tt class="py-lineno"> 313</tt>  <tt class="py-line">    <tt class="py-comment"># number_of_reads            I</tt> </tt>
<a name="L314"></a><tt class="py-lineno"> 314</tt>  <tt class="py-line">    <tt class="py-comment"># header_length              H</tt> </tt>
<a name="L315"></a><tt class="py-lineno"> 315</tt>  <tt class="py-line">    <tt class="py-comment"># key_length                 H</tt> </tt>
<a name="L316"></a><tt class="py-lineno"> 316</tt>  <tt class="py-line">    <tt class="py-comment"># number_of_flows_per_read   H</tt> </tt>
<a name="L317"></a><tt class="py-lineno"> 317</tt>  <tt class="py-line">    <tt class="py-comment"># flowgram_format_code       B</tt> </tt>
<a name="L318"></a><tt class="py-lineno"> 318</tt>  <tt class="py-line">    <tt class="py-comment"># [rest of file header depends on the number of flows and how many keys]</tt> </tt>
<a name="L319"></a><tt class="py-lineno"> 319</tt>  <tt class="py-line">    <tt class="py-name">fmt</tt> <tt class="py-op">=</tt> <tt class="py-string">'&gt;4s4BQIIHHHB'</tt> </tt>
<a name="L320"></a><tt class="py-lineno"> 320</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-number">31</tt> <tt class="py-op">==</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-name">fmt</tt><tt class="py-op">)</tt> </tt>
<a name="L321"></a><tt class="py-lineno"> 321</tt>  <tt class="py-line">    <tt id="link-31" class="py-name" targets="Variable BioSQL.BioSeq.DBSeq.data=BioSQL.BioSeq.DBSeq-class.html#data"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-31', 'data', 'link-31');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-32" class="py-name" targets="Function Bio.Affy.CelFile.read()=Bio.Affy.CelFile-module.html#read,Function Bio.AlignIO.read()=Bio.AlignIO-module.html#read,Function Bio.Blast.NCBIXML.read()=Bio.Blast.NCBIXML-module.html#read,Function Bio.Cluster.read()=Bio.Cluster-module.html#read,Function Bio.Compass.read()=Bio.Compass-module.html#read,Function Bio.Emboss.Primer3.read()=Bio.Emboss.Primer3-module.html#read,Function Bio.Emboss.PrimerSearch.read()=Bio.Emboss.PrimerSearch-module.html#read,Method Bio.Entrez.Parser.DataHandler.read()=Bio.Entrez.Parser.DataHandler-class.html#read,Function Bio.Entrez.read()=Bio.Entrez-module.html#read,Function Bio.ExPASy.Enzyme.read()=Bio.ExPASy.Enzyme-module.html#read,Function Bio.ExPASy.Prodoc.read()=Bio.ExPASy.Prodoc-module.html#read,Function Bio.ExPASy.Prosite.read()=Bio.ExPASy.Prosite-module.html#read,Function Bio.ExPASy.ScanProsite.read()=Bio.ExPASy.ScanProsite-module.html#read,Method Bio.File.UndoHandle.read()=Bio.File.UndoHandle-class.html#read,Function Bio.KEGG.KGML.KGML_parser.read()=Bio.KEGG.KGML.KGML_parser-module.html#read,Function Bio.Medline.read()=Bio.Medline-module.html#read,Function Bio.Motif.Parsers.AlignAce.read()=Bio.Motif.Parsers.AlignAce-module.html#read,Function Bio.Motif.Parsers.MAST.read()=Bio.Motif.Parsers.MAST-module.html#read,Function Bio.Motif.Parsers.MEME.read()=Bio.Motif.Parsers.MEME-module.html#read,Function Bio.Motif.read()=Bio.Motif-module.html#read,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.read()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#read,Method Bio.Nexus.Nexus.Nexus.read()=Bio.Nexus.Nexus.Nexus-class.html#read,Function Bio.Phylo.PAML.baseml.read()=Bio.Phylo.PAML.baseml-module.html#read,Function Bio.Phylo.PAML.codeml.read()=Bio.Phylo.PAML.codeml-module.html#read,Function Bio.Phylo.PAML.yn00.read()=Bio.Phylo.PAML.yn00-module.html#read,Method Bio.Phylo.PhyloXMLIO.Parser.read()=Bio.Phylo.PhyloXMLIO.Parser-class.html#read,Function Bio.Phylo.PhyloXMLIO.read()=Bio.Phylo.PhyloXMLIO-module.html#read,Function Bio.Phylo._io.read()=Bio.Phylo._io-module.html#read,Function Bio.PopGen.GenePop.FileParser.read()=Bio.PopGen.GenePop.FileParser-module.html#read,Function Bio.PopGen.GenePop.LargeFileParser.read()=Bio.PopGen.GenePop.LargeFileParser-module.html#read,Function Bio.PopGen.GenePop.read()=Bio.PopGen.GenePop-module.html#read,Function Bio.SearchIO.read()=Bio.SearchIO-module.html#read,Method Bio.SeqIO.SffIO._AddTellHandle.read()=Bio.SeqIO.SffIO._AddTellHandle-class.html#read,Function Bio.SeqIO.read()=Bio.SeqIO-module.html#read,Function Bio.Sequencing.Ace.read()=Bio.Sequencing.Ace-module.html#read,Function Bio.Sequencing.Phd.read()=Bio.Sequencing.Phd-module.html#read,Function Bio.SwissProt.read()=Bio.SwissProt-module.html#read,Function Bio.UniGene.read()=Bio.UniGene-module.html#read,Method Bio.bgzf.BgzfReader.read()=Bio.bgzf.BgzfReader-class.html#read,Function Bio.motifs.alignace.read()=Bio.motifs.alignace-module.html#read,Function Bio.motifs.jaspar.read()=Bio.motifs.jaspar-module.html#read,Function Bio.motifs.mast.read()=Bio.motifs.mast-module.html#read,Function Bio.motifs.meme.read()=Bio.motifs.meme-module.html#read,Function Bio.motifs.read()=Bio.motifs-module.html#read,Function Bio.motifs.transfac.read()=Bio.motifs.transfac-module.html#read"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-32', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-number">31</tt><tt class="py-op">)</tt> </tt>
<a name="L322"></a><tt class="py-lineno"> 322</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-33" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-33', 'data', 'link-31');">data</a></tt><tt class="py-op">:</tt> </tt>
<a name="L323"></a><tt class="py-lineno"> 323</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Empty file."</tt><tt class="py-op">)</tt> </tt>
<a name="L324"></a><tt class="py-lineno"> 324</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-34" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-34', 'data', 'link-31');">data</a></tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">13</tt><tt class="py-op">:</tt> </tt>
<a name="L325"></a><tt class="py-lineno"> 325</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"File too small to hold a valid SFF header."</tt><tt class="py-op">)</tt> </tt>
<a name="L326"></a><tt class="py-lineno"> 326</tt>  <tt class="py-line">    <tt class="py-name">magic_number</tt><tt class="py-op">,</tt> <tt class="py-name">ver0</tt><tt class="py-op">,</tt> <tt class="py-name">ver1</tt><tt class="py-op">,</tt> <tt class="py-name">ver2</tt><tt class="py-op">,</tt> <tt class="py-name">ver3</tt><tt class="py-op">,</tt> <tt class="py-name">index_offset</tt><tt class="py-op">,</tt> <tt class="py-name">index_length</tt><tt class="py-op">,</tt> \ </tt>
<a name="L327"></a><tt class="py-lineno"> 327</tt>  <tt class="py-line">        <tt class="py-name">number_of_reads</tt><tt class="py-op">,</tt> <tt class="py-name">header_length</tt><tt class="py-op">,</tt> <tt class="py-name">key_length</tt><tt class="py-op">,</tt> <tt class="py-name">number_of_flows_per_read</tt><tt class="py-op">,</tt> \ </tt>
<a name="L328"></a><tt class="py-lineno"> 328</tt>  <tt class="py-line">        <tt class="py-name">flowgram_format</tt> <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">unpack</tt><tt class="py-op">(</tt><tt class="py-name">fmt</tt><tt class="py-op">,</tt> <tt id="link-35" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-35', 'data', 'link-31');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L329"></a><tt class="py-lineno"> 329</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">magic_number</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt id="link-36" class="py-name"><a title="Bio.SeqIO.SffIO._hsh" class="py-name" href="#" onclick="return doclink('link-36', '_hsh', 'link-20');">_hsh</a></tt><tt class="py-op">,</tt> <tt id="link-37" class="py-name"><a title="Bio.SeqIO.SffIO._srt" class="py-name" href="#" onclick="return doclink('link-37', '_srt', 'link-22');">_srt</a></tt><tt class="py-op">,</tt> <tt id="link-38" class="py-name"><a title="Bio.SeqIO.SffIO._mft" class="py-name" href="#" onclick="return doclink('link-38', '_mft', 'link-24');">_mft</a></tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L330"></a><tt class="py-lineno"> 330</tt>  <tt class="py-line">        <tt class="py-comment"># Probably user error, calling Bio.SeqIO.parse() twice!</tt> </tt>
<a name="L331"></a><tt class="py-lineno"> 331</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Handle seems to be at SFF index block, not start"</tt><tt class="py-op">)</tt> </tt>
<a name="L332"></a><tt class="py-lineno"> 332</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">magic_number</tt> <tt class="py-op">!=</tt> <tt id="link-39" class="py-name"><a title="Bio.SeqIO.SffIO._sff" class="py-name" href="#" onclick="return doclink('link-39', '_sff', 'link-18');">_sff</a></tt><tt class="py-op">:</tt>  <tt class="py-comment"># 779314790</tt> </tt>
<a name="L333"></a><tt class="py-lineno"> 333</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"SFF file did not start '.sff', but %s"</tt> </tt>
<a name="L334"></a><tt class="py-lineno"> 334</tt>  <tt class="py-line">                         <tt class="py-op">%</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">magic_number</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L335"></a><tt class="py-lineno"> 335</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-op">(</tt><tt class="py-name">ver0</tt><tt class="py-op">,</tt> <tt class="py-name">ver1</tt><tt class="py-op">,</tt> <tt class="py-name">ver2</tt><tt class="py-op">,</tt> <tt class="py-name">ver3</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L336"></a><tt class="py-lineno"> 336</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unsupported SFF version in header, %i.%i.%i.%i"</tt> </tt>
<a name="L337"></a><tt class="py-lineno"> 337</tt>  <tt class="py-line">                         <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">ver0</tt><tt class="py-op">,</tt> <tt class="py-name">ver1</tt><tt class="py-op">,</tt> <tt class="py-name">ver2</tt><tt class="py-op">,</tt> <tt class="py-name">ver3</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L338"></a><tt class="py-lineno"> 338</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">flowgram_format</tt> <tt class="py-op">!=</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L339"></a><tt class="py-lineno"> 339</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Flowgram format code %i not supported"</tt> </tt>
<a name="L340"></a><tt class="py-lineno"> 340</tt>  <tt class="py-line">                         <tt class="py-op">%</tt> <tt class="py-name">flowgram_format</tt><tt class="py-op">)</tt> </tt>
<a name="L341"></a><tt class="py-lineno"> 341</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-op">(</tt><tt class="py-name">index_offset</tt> <tt class="py-op">!=</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt> <tt class="py-op">^</tt> <tt class="py-op">(</tt><tt class="py-name">index_length</tt> <tt class="py-op">!=</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L342"></a><tt class="py-lineno"> 342</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Index offset %i but index length %i"</tt> </tt>
<a name="L343"></a><tt class="py-lineno"> 343</tt>  <tt class="py-line">                         <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">index_offset</tt><tt class="py-op">,</tt> <tt class="py-name">index_length</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L344"></a><tt class="py-lineno"> 344</tt>  <tt class="py-line">    <tt class="py-name">flow_chars</tt> <tt class="py-op">=</tt> <tt id="link-40" class="py-name"><a title="Bio._py3k._bytes_to_string" class="py-name" href="#" onclick="return doclink('link-40', '_bytes_to_string', 'link-14');">_bytes_to_string</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-41" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-41', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">number_of_flows_per_read</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L345"></a><tt class="py-lineno"> 345</tt>  <tt class="py-line">    <tt class="py-name">key_sequence</tt> <tt class="py-op">=</tt> <tt id="link-42" class="py-name"><a title="Bio._py3k._bytes_to_string" class="py-name" href="#" onclick="return doclink('link-42', '_bytes_to_string', 'link-14');">_bytes_to_string</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-43" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-43', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">key_length</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L346"></a><tt class="py-lineno"> 346</tt>  <tt class="py-line">    <tt class="py-comment"># According to the spec, the header_length field should be the total number</tt> </tt>
<a name="L347"></a><tt class="py-lineno"> 347</tt>  <tt class="py-line">    <tt class="py-comment"># of bytes required by this set of header fields, and should be equal to</tt> </tt>
<a name="L348"></a><tt class="py-lineno"> 348</tt>  <tt class="py-line">    <tt class="py-comment"># "31 + number_of_flows_per_read + key_length" rounded up to the next value</tt> </tt>
<a name="L349"></a><tt class="py-lineno"> 349</tt>  <tt class="py-line">    <tt class="py-comment"># divisible by 8.</tt> </tt>
<a name="L350"></a><tt class="py-lineno"> 350</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">header_length</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt> </tt>
<a name="L351"></a><tt class="py-lineno"> 351</tt>  <tt class="py-line">    <tt class="py-name">padding</tt> <tt class="py-op">=</tt> <tt class="py-name">header_length</tt> <tt class="py-op">-</tt> <tt class="py-name">number_of_flows_per_read</tt> <tt class="py-op">-</tt> <tt class="py-name">key_length</tt> <tt class="py-op">-</tt> <tt class="py-number">31</tt> </tt>
<a name="L352"></a><tt class="py-lineno"> 352</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-number">0</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">padding</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">8</tt><tt class="py-op">,</tt> <tt class="py-name">padding</tt> </tt>
<a name="L353"></a><tt class="py-lineno"> 353</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-44" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-44', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">padding</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-45" class="py-name" targets="Method Bio.Crystal.Chain.count()=Bio.Crystal.Chain-class.html#count,Method Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count()=Bio.NeuralNetwork.Gene.Pattern.PatternRepository-class.html#count,Method Bio.Seq.MutableSeq.count()=Bio.Seq.MutableSeq-class.html#count,Method Bio.Seq.Seq.count()=Bio.Seq.Seq-class.html#count,Method Bio.Seq.UnknownSeq.count()=Bio.Seq.UnknownSeq-class.html#count,Method Bio.motifs.Instances.count()=Bio.motifs.Instances-class.html#count"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-45', 'count', 'link-45');">count</a></tt><tt class="py-op">(</tt><tt id="link-46" class="py-name"><a title="Bio.SeqIO.SffIO._null" class="py-name" href="#" onclick="return doclink('link-46', '_null', 'link-16');">_null</a></tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">padding</tt><tt class="py-op">:</tt> </tt>
<a name="L354"></a><tt class="py-lineno"> 354</tt>  <tt class="py-line">        <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L355"></a><tt class="py-lineno"> 355</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt id="link-47" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-47', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-48" class="py-name" targets="Class Bio.BiopythonParserWarning=Bio.BiopythonParserWarning-class.html"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-48', 'BiopythonParserWarning', 'link-48');">BiopythonParserWarning</a></tt> </tt>
<a name="L356"></a><tt class="py-lineno"> 356</tt>  <tt class="py-line">        <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Your SFF file is invalid, post header %i byte "</tt> </tt>
<a name="L357"></a><tt class="py-lineno"> 357</tt>  <tt class="py-line">                      <tt class="py-string">"null padding region contained data."</tt> <tt class="py-op">%</tt> <tt class="py-name">padding</tt><tt class="py-op">,</tt> </tt>
<a name="L358"></a><tt class="py-lineno"> 358</tt>  <tt class="py-line">                      <tt id="link-49" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-49', 'BiopythonParserWarning', 'link-48');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L359"></a><tt class="py-lineno"> 359</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">header_length</tt><tt class="py-op">,</tt> <tt class="py-name">index_offset</tt><tt class="py-op">,</tt> <tt class="py-name">index_length</tt><tt class="py-op">,</tt> \ </tt>
<a name="L360"></a><tt class="py-lineno"> 360</tt>  <tt class="py-line">        <tt class="py-name">number_of_reads</tt><tt class="py-op">,</tt> <tt class="py-name">number_of_flows_per_read</tt><tt class="py-op">,</tt> \ </tt>
<a name="L361"></a><tt class="py-lineno"> 361</tt>  <tt class="py-line">        <tt class="py-name">flow_chars</tt><tt class="py-op">,</tt> <tt class="py-name">key_sequence</tt> </tt>
</div><a name="L362"></a><tt class="py-lineno"> 362</tt>  <tt class="py-line"> </tt>
<a name="L363"></a><tt class="py-lineno"> 363</tt>  <tt class="py-line"> </tt>
<a name="_sff_do_slow_index"></a><div id="_sff_do_slow_index-def"><a name="L364"></a><tt class="py-lineno"> 364</tt> <a class="py-toggle" href="#" id="_sff_do_slow_index-toggle" onclick="return toggle('_sff_do_slow_index');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO-module.html#_sff_do_slow_index">_sff_do_slow_index</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_sff_do_slow_index-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_sff_do_slow_index-expanded"><a name="L365"></a><tt class="py-lineno"> 365</tt>  <tt class="py-line">    <tt class="py-docstring">"""Generates an index by scanning though all the reads in an SFF file (PRIVATE).</tt> </tt>
<a name="L366"></a><tt class="py-lineno"> 366</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L367"></a><tt class="py-lineno"> 367</tt>  <tt class="py-line"><tt class="py-docstring">    This is a slow but generic approach if we can't parse the provided index</tt> </tt>
<a name="L368"></a><tt class="py-lineno"> 368</tt>  <tt class="py-line"><tt class="py-docstring">    (if present).</tt> </tt>
<a name="L369"></a><tt class="py-lineno"> 369</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L370"></a><tt class="py-lineno"> 370</tt>  <tt class="py-line"><tt class="py-docstring">    Will use the handle seek/tell functions.</tt> </tt>
<a name="L371"></a><tt class="py-lineno"> 371</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L372"></a><tt class="py-lineno"> 372</tt>  <tt class="py-line">    <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-50" class="py-name" targets="Method Bio.File.UndoHandle.seek()=Bio.File.UndoHandle-class.html#seek,Method Bio.SeqIO.SffIO._AddTellHandle.seek()=Bio.SeqIO.SffIO._AddTellHandle-class.html#seek,Method Bio.bgzf.BgzfReader.seek()=Bio.bgzf.BgzfReader-class.html#seek"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-50', 'seek', 'link-50');">seek</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L373"></a><tt class="py-lineno"> 373</tt>  <tt class="py-line">    <tt class="py-name">header_length</tt><tt class="py-op">,</tt> <tt class="py-name">index_offset</tt><tt class="py-op">,</tt> <tt class="py-name">index_length</tt><tt class="py-op">,</tt> <tt class="py-name">number_of_reads</tt><tt class="py-op">,</tt> \ </tt>
<a name="L374"></a><tt class="py-lineno"> 374</tt>  <tt class="py-line">        <tt class="py-name">number_of_flows_per_read</tt><tt class="py-op">,</tt> <tt class="py-name">flow_chars</tt><tt class="py-op">,</tt> <tt class="py-name">key_sequence</tt> \ </tt>
<a name="L375"></a><tt class="py-lineno"> 375</tt>  <tt class="py-line">        <tt class="py-op">=</tt> <tt id="link-51" class="py-name" targets="Function Bio.SeqIO.SffIO._sff_file_header()=Bio.SeqIO.SffIO-module.html#_sff_file_header"><a title="Bio.SeqIO.SffIO._sff_file_header" class="py-name" href="#" onclick="return doclink('link-51', '_sff_file_header', 'link-51');">_sff_file_header</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L376"></a><tt class="py-lineno"> 376</tt>  <tt class="py-line">    <tt class="py-comment"># Now on to the reads...</tt> </tt>
<a name="L377"></a><tt class="py-lineno"> 377</tt>  <tt class="py-line">    <tt class="py-name">read_header_fmt</tt> <tt class="py-op">=</tt> <tt class="py-string">'&gt;2HI4H'</tt> </tt>
<a name="L378"></a><tt class="py-lineno"> 378</tt>  <tt class="py-line">    <tt class="py-name">read_header_size</tt> <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-name">read_header_fmt</tt><tt class="py-op">)</tt> </tt>
<a name="L379"></a><tt class="py-lineno"> 379</tt>  <tt class="py-line">    <tt class="py-comment"># NOTE - assuming flowgram_format==1, which means struct type H</tt> </tt>
<a name="L380"></a><tt class="py-lineno"> 380</tt>  <tt class="py-line">    <tt class="py-name">read_flow_fmt</tt> <tt class="py-op">=</tt> <tt class="py-string">"&gt;%iH"</tt> <tt class="py-op">%</tt> <tt class="py-name">number_of_flows_per_read</tt> </tt>
<a name="L381"></a><tt class="py-lineno"> 381</tt>  <tt class="py-line">    <tt class="py-name">read_flow_size</tt> <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-name">read_flow_fmt</tt><tt class="py-op">)</tt> </tt>
<a name="L382"></a><tt class="py-lineno"> 382</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-number">1</tt> <tt class="py-op">==</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-string">"&gt;B"</tt><tt class="py-op">)</tt> </tt>
<a name="L383"></a><tt class="py-lineno"> 383</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-number">1</tt> <tt class="py-op">==</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-string">"&gt;s"</tt><tt class="py-op">)</tt> </tt>
<a name="L384"></a><tt class="py-lineno"> 384</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-number">1</tt> <tt class="py-op">==</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-string">"&gt;c"</tt><tt class="py-op">)</tt> </tt>
<a name="L385"></a><tt class="py-lineno"> 385</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">read_header_size</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt>  <tt class="py-comment"># Important for padding calc later!</tt> </tt>
<a name="L386"></a><tt class="py-lineno"> 386</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt id="link-52" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-52', 'read', 'link-32');">read</a></tt> <tt class="py-keyword">in</tt> <tt id="link-53" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-53', 'range', 'link-53');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">number_of_reads</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L387"></a><tt class="py-lineno"> 387</tt>  <tt class="py-line">        <tt class="py-name">record_offset</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-54" class="py-name" targets="Method Bio.File.UndoHandle.tell()=Bio.File.UndoHandle-class.html#tell,Method Bio.SeqIO.SffIO._AddTellHandle.tell()=Bio.SeqIO.SffIO._AddTellHandle-class.html#tell,Method Bio.bgzf.BgzfReader.tell()=Bio.bgzf.BgzfReader-class.html#tell,Method Bio.bgzf.BgzfWriter.tell()=Bio.bgzf.BgzfWriter-class.html#tell"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-54', 'tell', 'link-54');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L388"></a><tt class="py-lineno"> 388</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">record_offset</tt> <tt class="py-op">==</tt> <tt class="py-name">index_offset</tt><tt class="py-op">:</tt> </tt>
<a name="L389"></a><tt class="py-lineno"> 389</tt>  <tt class="py-line">            <tt class="py-comment"># Found index block within reads, ignore it:</tt> </tt>
<a name="L390"></a><tt class="py-lineno"> 390</tt>  <tt class="py-line">            <tt class="py-name">offset</tt> <tt class="py-op">=</tt> <tt class="py-name">index_offset</tt> <tt class="py-op">+</tt> <tt class="py-name">index_length</tt> </tt>
<a name="L391"></a><tt class="py-lineno"> 391</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">offset</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt><tt class="py-op">:</tt> </tt>
<a name="L392"></a><tt class="py-lineno"> 392</tt>  <tt class="py-line">                <tt class="py-name">offset</tt> <tt class="py-op">+=</tt> <tt class="py-number">8</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">offset</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt><tt class="py-op">)</tt> </tt>
<a name="L393"></a><tt class="py-lineno"> 393</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">offset</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt> </tt>
<a name="L394"></a><tt class="py-lineno"> 394</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-55" class="py-name"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-55', 'seek', 'link-50');">seek</a></tt><tt class="py-op">(</tt><tt class="py-name">offset</tt><tt class="py-op">)</tt> </tt>
<a name="L395"></a><tt class="py-lineno"> 395</tt>  <tt class="py-line">            <tt class="py-name">record_offset</tt> <tt class="py-op">=</tt> <tt class="py-name">offset</tt> </tt>
<a name="L396"></a><tt class="py-lineno"> 396</tt>  <tt class="py-line">        <tt class="py-comment"># assert record_offset%8 == 0 # Worth checking, but slow</tt> </tt>
<a name="L397"></a><tt class="py-lineno"> 397</tt>  <tt class="py-line">        <tt class="py-comment"># First the fixed header</tt> </tt>
<a name="L398"></a><tt class="py-lineno"> 398</tt>  <tt class="py-line">        <tt id="link-56" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-56', 'data', 'link-31');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-57" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-57', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">read_header_size</tt><tt class="py-op">)</tt> </tt>
<a name="L399"></a><tt class="py-lineno"> 399</tt>  <tt class="py-line">        <tt class="py-name">read_header_length</tt><tt class="py-op">,</tt> <tt class="py-name">name_length</tt><tt class="py-op">,</tt> <tt class="py-name">seq_len</tt><tt class="py-op">,</tt> <tt class="py-name">clip_qual_left</tt><tt class="py-op">,</tt> \ </tt>
<a name="L400"></a><tt class="py-lineno"> 400</tt>  <tt class="py-line">            <tt class="py-name">clip_qual_right</tt><tt class="py-op">,</tt> <tt class="py-name">clip_adapter_left</tt><tt class="py-op">,</tt> <tt class="py-name">clip_adapter_right</tt> \ </tt>
<a name="L401"></a><tt class="py-lineno"> 401</tt>  <tt class="py-line">            <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">unpack</tt><tt class="py-op">(</tt><tt class="py-name">read_header_fmt</tt><tt class="py-op">,</tt> <tt id="link-58" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-58', 'data', 'link-31');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L402"></a><tt class="py-lineno"> 402</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">read_header_length</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">10</tt> <tt class="py-keyword">or</tt> <tt class="py-name">read_header_length</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt> <tt class="py-op">!=</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L403"></a><tt class="py-lineno"> 403</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Malformed read header, says length is %i:\n%s"</tt> </tt>
<a name="L404"></a><tt class="py-lineno"> 404</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">read_header_length</tt><tt class="py-op">,</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt id="link-59" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-59', 'data', 'link-31');">data</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L405"></a><tt class="py-lineno"> 405</tt>  <tt class="py-line">        <tt class="py-comment"># now the name and any padding (remainder of header)</tt> </tt>
<a name="L406"></a><tt class="py-lineno"> 406</tt>  <tt class="py-line">        <tt id="link-60" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-60', 'name', 'link-60');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-61" class="py-name"><a title="Bio._py3k._bytes_to_string" class="py-name" href="#" onclick="return doclink('link-61', '_bytes_to_string', 'link-14');">_bytes_to_string</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-62" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-62', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">name_length</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L407"></a><tt class="py-lineno"> 407</tt>  <tt class="py-line">        <tt class="py-name">padding</tt> <tt class="py-op">=</tt> <tt class="py-name">read_header_length</tt> <tt class="py-op">-</tt> <tt class="py-name">read_header_size</tt> <tt class="py-op">-</tt> <tt class="py-name">name_length</tt> </tt>
<a name="L408"></a><tt class="py-lineno"> 408</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-63" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-63', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">padding</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-64" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-64', 'count', 'link-45');">count</a></tt><tt class="py-op">(</tt><tt id="link-65" class="py-name"><a title="Bio.SeqIO.SffIO._null" class="py-name" href="#" onclick="return doclink('link-65', '_null', 'link-16');">_null</a></tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">padding</tt><tt class="py-op">:</tt> </tt>
<a name="L409"></a><tt class="py-lineno"> 409</tt>  <tt class="py-line">            <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L410"></a><tt class="py-lineno"> 410</tt>  <tt class="py-line">            <tt class="py-keyword">from</tt> <tt id="link-66" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-66', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-67" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-67', 'BiopythonParserWarning', 'link-48');">BiopythonParserWarning</a></tt> </tt>
<a name="L411"></a><tt class="py-lineno"> 411</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Your SFF file is invalid, post name %i byte "</tt> </tt>
<a name="L412"></a><tt class="py-lineno"> 412</tt>  <tt class="py-line">                          <tt class="py-string">"padding region contained data"</tt> <tt class="py-op">%</tt> <tt class="py-name">padding</tt><tt class="py-op">,</tt> </tt>
<a name="L413"></a><tt class="py-lineno"> 413</tt>  <tt class="py-line">                          <tt id="link-68" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-68', 'BiopythonParserWarning', 'link-48');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L414"></a><tt class="py-lineno"> 414</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">record_offset</tt> <tt class="py-op">+</tt> <tt class="py-name">read_header_length</tt> <tt class="py-op">==</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-69" class="py-name"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-69', 'tell', 'link-54');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L415"></a><tt class="py-lineno"> 415</tt>  <tt class="py-line">        <tt class="py-comment"># now the flowgram values, flowgram index, bases and qualities</tt> </tt>
<a name="L416"></a><tt class="py-lineno"> 416</tt>  <tt class="py-line">        <tt id="link-70" class="py-name" targets="Variable Bio.Alphabet.Alphabet.size=Bio.Alphabet.Alphabet-class.html#size,Variable Bio.Alphabet.Reduced.HPModel.size=Bio.Alphabet.Reduced.HPModel-class.html#size,Variable Bio.Alphabet.Reduced.Murphy10.size=Bio.Alphabet.Reduced.Murphy10-class.html#size,Variable Bio.Alphabet.Reduced.Murphy15.size=Bio.Alphabet.Reduced.Murphy15-class.html#size,Variable Bio.Alphabet.Reduced.Murphy4.size=Bio.Alphabet.Reduced.Murphy4-class.html#size,Variable Bio.Alphabet.Reduced.Murphy8.size=Bio.Alphabet.Reduced.Murphy8-class.html#size,Variable Bio.Alphabet.Reduced.PC5.size=Bio.Alphabet.Reduced.PC5-class.html#size,Variable Bio.Alphabet.SingleLetterAlphabet.size=Bio.Alphabet.SingleLetterAlphabet-class.html#size,Variable Bio.Alphabet.ThreeLetterProtein.size=Bio.Alphabet.ThreeLetterProtein-class.html#size,Method Bio.GenBank._FeatureConsumer.size()=Bio.GenBank._FeatureConsumer-class.html#size,Method Bio.GenBank._RecordConsumer.size()=Bio.GenBank._RecordConsumer-class.html#size,Variable Bio.Restriction.Restriction.AanI.size=Bio.Restriction.Restriction.AanI-class.html#size,Variable Bio.Restriction.Restriction.AarI.size=Bio.Restriction.Restriction.AarI-class.html#size,Variable Bio.Restriction.Restriction.AasI.size=Bio.Restriction.Restriction.AasI-class.html#size,Variable Bio.Restriction.Restriction.AatII.size=Bio.Restriction.Restriction.AatII-class.html#size,Variable Bio.Restriction.Restriction.AbaSI.size=Bio.Restriction.Restriction.AbaSI-class.html#size,Variable Bio.Restriction.Restriction.AbsI.size=Bio.Restriction.Restriction.AbsI-class.html#size,Variable Bio.Restriction.Restriction.Acc16I.size=Bio.Restriction.Restriction.Acc16I-class.html#size,Variable Bio.Restriction.Restriction.Acc36I.size=Bio.Restriction.Restriction.Acc36I-class.html#size,Variable Bio.Restriction.Restriction.Acc65I.size=Bio.Restriction.Restriction.Acc65I-class.html#size,Variable Bio.Restriction.Restriction.AccB1I.size=Bio.Restriction.Restriction.AccB1I-class.html#size,Variable Bio.Restriction.Restriction.AccB7I.size=Bio.Restriction.Restriction.AccB7I-class.html#size,Variable Bio.Restriction.Restriction.AccBSI.size=Bio.Restriction.Restriction.AccBSI-class.html#size,Variable Bio.Restriction.Restriction.AccI.size=Bio.Restriction.Restriction.AccI-class.html#size,Variable Bio.Restriction.Restriction.AccII.size=Bio.Restriction.Restriction.AccII-class.html#size,Variable Bio.Restriction.Restriction.AccIII.size=Bio.Restriction.Restriction.AccIII-class.html#size,Variable Bio.Restriction.Restriction.AceIII.size=Bio.Restriction.Restriction.AceIII-class.html#size,Variable Bio.Restriction.Restriction.AciI.size=Bio.Restriction.Restriction.AciI-class.html#size,Variable Bio.Restriction.Restriction.AclI.size=Bio.Restriction.Restriction.AclI-class.html#size,Variable Bio.Restriction.Restriction.AclWI.size=Bio.Restriction.Restriction.AclWI-class.html#size,Variable Bio.Restriction.Restriction.AcoI.size=Bio.Restriction.Restriction.AcoI-class.html#size,Variable Bio.Restriction.Restriction.AcsI.size=Bio.Restriction.Restriction.AcsI-class.html#size,Variable Bio.Restriction.Restriction.AcuI.size=Bio.Restriction.Restriction.AcuI-class.html#size,Variable Bio.Restriction.Restriction.AcvI.size=Bio.Restriction.Restriction.AcvI-class.html#size,Variable Bio.Restriction.Restriction.AcyI.size=Bio.Restriction.Restriction.AcyI-class.html#size,Variable Bio.Restriction.Restriction.AdeI.size=Bio.Restriction.Restriction.AdeI-class.html#size,Variable Bio.Restriction.Restriction.AfaI.size=Bio.Restriction.Restriction.AfaI-class.html#size,Variable Bio.Restriction.Restriction.AfeI.size=Bio.Restriction.Restriction.AfeI-class.html#size,Variable Bio.Restriction.Restriction.AfiI.size=Bio.Restriction.Restriction.AfiI-class.html#size,Variable Bio.Restriction.Restriction.AflII.size=Bio.Restriction.Restriction.AflII-class.html#size,Variable Bio.Restriction.Restriction.AflIII.size=Bio.Restriction.Restriction.AflIII-class.html#size,Variable Bio.Restriction.Restriction.AgeI.size=Bio.Restriction.Restriction.AgeI-class.html#size,Variable Bio.Restriction.Restriction.AgsI.size=Bio.Restriction.Restriction.AgsI-class.html#size,Variable Bio.Restriction.Restriction.AhaIII.size=Bio.Restriction.Restriction.AhaIII-class.html#size,Variable Bio.Restriction.Restriction.AhdI.size=Bio.Restriction.Restriction.AhdI-class.html#size,Variable Bio.Restriction.Restriction.AhlI.size=Bio.Restriction.Restriction.AhlI-class.html#size,Variable Bio.Restriction.Restriction.AjiI.size=Bio.Restriction.Restriction.AjiI-class.html#size,Variable Bio.Restriction.Restriction.AjnI.size=Bio.Restriction.Restriction.AjnI-class.html#size,Variable Bio.Restriction.Restriction.AjuI.size=Bio.Restriction.Restriction.AjuI-class.html#size,Variable Bio.Restriction.Restriction.AleI.size=Bio.Restriction.Restriction.AleI-class.html#size,Variable Bio.Restriction.Restriction.AlfI.size=Bio.Restriction.Restriction.AlfI-class.html#size,Variable Bio.Restriction.Restriction.AloI.size=Bio.Restriction.Restriction.AloI-class.html#size,Variable Bio.Restriction.Restriction.AluBI.size=Bio.Restriction.Restriction.AluBI-class.html#size,Variable Bio.Restriction.Restriction.AluI.size=Bio.Restriction.Restriction.AluI-class.html#size,Variable Bio.Restriction.Restriction.Alw21I.size=Bio.Restriction.Restriction.Alw21I-class.html#size,Variable Bio.Restriction.Restriction.Alw26I.size=Bio.Restriction.Restriction.Alw26I-class.html#size,Variable Bio.Restriction.Restriction.Alw44I.size=Bio.Restriction.Restriction.Alw44I-class.html#size,Variable Bio.Restriction.Restriction.AlwFI.size=Bio.Restriction.Restriction.AlwFI-class.html#size,Variable Bio.Restriction.Restriction.AlwI.size=Bio.Restriction.Restriction.AlwI-class.html#size,Variable Bio.Restriction.Restriction.AlwNI.size=Bio.Restriction.Restriction.AlwNI-class.html#size,Variable Bio.Restriction.Restriction.Ama87I.size=Bio.Restriction.Restriction.Ama87I-class.html#size,Variable Bio.Restriction.Restriction.Aor13HI.size=Bio.Restriction.Restriction.Aor13HI-class.html#size,Variable Bio.Restriction.Restriction.Aor51HI.size=Bio.Restriction.Restriction.Aor51HI-class.html#size,Variable Bio.Restriction.Restriction.AoxI.size=Bio.Restriction.Restriction.AoxI-class.html#size,Variable Bio.Restriction.Restriction.ApaBI.size=Bio.Restriction.Restriction.ApaBI-class.html#size,Variable Bio.Restriction.Restriction.ApaI.size=Bio.Restriction.Restriction.ApaI-class.html#size,Variable Bio.Restriction.Restriction.ApaLI.size=Bio.Restriction.Restriction.ApaLI-class.html#size,Variable Bio.Restriction.Restriction.ApeKI.size=Bio.Restriction.Restriction.ApeKI-class.html#size,Variable Bio.Restriction.Restriction.ApoI.size=Bio.Restriction.Restriction.ApoI-class.html#size,Variable Bio.Restriction.Restriction.ApyPI.size=Bio.Restriction.Restriction.ApyPI-class.html#size,Variable Bio.Restriction.Restriction.AquII.size=Bio.Restriction.Restriction.AquII-class.html#size,Variable Bio.Restriction.Restriction.AquIII.size=Bio.Restriction.Restriction.AquIII-class.html#size,Variable Bio.Restriction.Restriction.AquIV.size=Bio.Restriction.Restriction.AquIV-class.html#size,Variable Bio.Restriction.Restriction.ArsI.size=Bio.Restriction.Restriction.ArsI-class.html#size,Variable Bio.Restriction.Restriction.AscI.size=Bio.Restriction.Restriction.AscI-class.html#size,Variable Bio.Restriction.Restriction.AseI.size=Bio.Restriction.Restriction.AseI-class.html#size,Variable Bio.Restriction.Restriction.Asi256I.size=Bio.Restriction.Restriction.Asi256I-class.html#size,Variable Bio.Restriction.Restriction.AsiGI.size=Bio.Restriction.Restriction.AsiGI-class.html#size,Variable Bio.Restriction.Restriction.AsiSI.size=Bio.Restriction.Restriction.AsiSI-class.html#size,Variable Bio.Restriction.Restriction.Asp700I.size=Bio.Restriction.Restriction.Asp700I-class.html#size,Variable Bio.Restriction.Restriction.Asp718I.size=Bio.Restriction.Restriction.Asp718I-class.html#size,Variable Bio.Restriction.Restriction.AspA2I.size=Bio.Restriction.Restriction.AspA2I-class.html#size,Variable Bio.Restriction.Restriction.AspBHI.size=Bio.Restriction.Restriction.AspBHI-class.html#size,Variable Bio.Restriction.Restriction.AspLEI.size=Bio.Restriction.Restriction.AspLEI-class.html#size,Variable Bio.Restriction.Restriction.AspS9I.size=Bio.Restriction.Restriction.AspS9I-class.html#size,Variable Bio.Restriction.Restriction.AssI.size=Bio.Restriction.Restriction.AssI-class.html#size,Variable Bio.Restriction.Restriction.AsuC2I.size=Bio.Restriction.Restriction.AsuC2I-class.html#size,Variable Bio.Restriction.Restriction.AsuHPI.size=Bio.Restriction.Restriction.AsuHPI-class.html#size,Variable Bio.Restriction.Restriction.AsuI.size=Bio.Restriction.Restriction.AsuI-class.html#size,Variable Bio.Restriction.Restriction.AsuII.size=Bio.Restriction.Restriction.AsuII-class.html#size,Variable Bio.Restriction.Restriction.AsuNHI.size=Bio.Restriction.Restriction.AsuNHI-class.html#size,Variable Bio.Restriction.Restriction.AvaI.size=Bio.Restriction.Restriction.AvaI-class.html#size,Variable Bio.Restriction.Restriction.AvaII.size=Bio.Restriction.Restriction.AvaII-class.html#size,Variable Bio.Restriction.Restriction.AvaIII.size=Bio.Restriction.Restriction.AvaIII-class.html#size,Variable Bio.Restriction.Restriction.AvrII.size=Bio.Restriction.Restriction.AvrII-class.html#size,Variable Bio.Restriction.Restriction.AxyI.size=Bio.Restriction.Restriction.AxyI-class.html#size,Variable Bio.Restriction.Restriction.BaeGI.size=Bio.Restriction.Restriction.BaeGI-class.html#size,Variable Bio.Restriction.Restriction.BaeI.size=Bio.Restriction.Restriction.BaeI-class.html#size,Variable Bio.Restriction.Restriction.BalI.size=Bio.Restriction.Restriction.BalI-class.html#size,Variable Bio.Restriction.Restriction.BamHI.size=Bio.Restriction.Restriction.BamHI-class.html#size,Variable Bio.Restriction.Restriction.BanI.size=Bio.Restriction.Restriction.BanI-class.html#size,Variable Bio.Restriction.Restriction.BanII.size=Bio.Restriction.Restriction.BanII-class.html#size,Variable Bio.Restriction.Restriction.BarI.size=Bio.Restriction.Restriction.BarI-class.html#size,Variable Bio.Restriction.Restriction.BasI.size=Bio.Restriction.Restriction.BasI-class.html#size,Variable Bio.Restriction.Restriction.BauI.size=Bio.Restriction.Restriction.BauI-class.html#size,Variable Bio.Restriction.Restriction.Bbr7I.size=Bio.Restriction.Restriction.Bbr7I-class.html#size,Variable Bio.Restriction.Restriction.BbrPI.size=Bio.Restriction.Restriction.BbrPI-class.html#size,Variable Bio.Restriction.Restriction.BbsI.size=Bio.Restriction.Restriction.BbsI-class.html#size,Variable Bio.Restriction.Restriction.Bbv12I.size=Bio.Restriction.Restriction.Bbv12I-class.html#size,Variable Bio.Restriction.Restriction.BbvCI.size=Bio.Restriction.Restriction.BbvCI-class.html#size,Variable Bio.Restriction.Restriction.BbvI.size=Bio.Restriction.Restriction.BbvI-class.html#size,Variable Bio.Restriction.Restriction.BbvII.size=Bio.Restriction.Restriction.BbvII-class.html#size,Variable Bio.Restriction.Restriction.BccI.size=Bio.Restriction.Restriction.BccI-class.html#size,Variable Bio.Restriction.Restriction.Bce83I.size=Bio.Restriction.Restriction.Bce83I-class.html#size,Variable Bio.Restriction.Restriction.BceAI.size=Bio.Restriction.Restriction.BceAI-class.html#size,Variable Bio.Restriction.Restriction.BcefI.size=Bio.Restriction.Restriction.BcefI-class.html#size,Variable Bio.Restriction.Restriction.BcgI.size=Bio.Restriction.Restriction.BcgI-class.html#size,Variable Bio.Restriction.Restriction.BciT130I.size=Bio.Restriction.Restriction.BciT130I-class.html#size,Variable Bio.Restriction.Restriction.BciVI.size=Bio.Restriction.Restriction.BciVI-class.html#size,Variable Bio.Restriction.Restriction.BclI.size=Bio.Restriction.Restriction.BclI-class.html#size,Variable Bio.Restriction.Restriction.BcnI.size=Bio.Restriction.Restriction.BcnI-class.html#size,Variable Bio.Restriction.Restriction.BcoDI.size=Bio.Restriction.Restriction.BcoDI-class.html#size,Variable Bio.Restriction.Restriction.BcuI.size=Bio.Restriction.Restriction.BcuI-class.html#size,Variable Bio.Restriction.Restriction.BdaI.size=Bio.Restriction.Restriction.BdaI-class.html#size,Variable Bio.Restriction.Restriction.BetI.size=Bio.Restriction.Restriction.BetI-class.html#size,Variable Bio.Restriction.Restriction.BfaI.size=Bio.Restriction.Restriction.BfaI-class.html#size,Variable Bio.Restriction.Restriction.BfiI.size=Bio.Restriction.Restriction.BfiI-class.html#size,Variable Bio.Restriction.Restriction.BfmI.size=Bio.Restriction.Restriction.BfmI-class.html#size,Variable Bio.Restriction.Restriction.BfoI.size=Bio.Restriction.Restriction.BfoI-class.html#size,Variable Bio.Restriction.Restriction.BfrI.size=Bio.Restriction.Restriction.BfrI-class.html#size,Variable Bio.Restriction.Restriction.BfuAI.size=Bio.Restriction.Restriction.BfuAI-class.html#size,Variable Bio.Restriction.Restriction.BfuCI.size=Bio.Restriction.Restriction.BfuCI-class.html#size,Variable Bio.Restriction.Restriction.BfuI.size=Bio.Restriction.Restriction.BfuI-class.html#size,Variable Bio.Restriction.Restriction.BglI.size=Bio.Restriction.Restriction.BglI-class.html#size,Variable Bio.Restriction.Restriction.BglII.size=Bio.Restriction.Restriction.BglII-class.html#size,Variable Bio.Restriction.Restriction.BinI.size=Bio.Restriction.Restriction.BinI-class.html#size,Variable Bio.Restriction.Restriction.BisI.size=Bio.Restriction.Restriction.BisI-class.html#size,Variable Bio.Restriction.Restriction.BlnI.size=Bio.Restriction.Restriction.BlnI-class.html#size,Variable Bio.Restriction.Restriction.BlpI.size=Bio.Restriction.Restriction.BlpI-class.html#size,Variable Bio.Restriction.Restriction.BlsI.size=Bio.Restriction.Restriction.BlsI-class.html#size,Variable Bio.Restriction.Restriction.BmcAI.size=Bio.Restriction.Restriction.BmcAI-class.html#size,Variable Bio.Restriction.Restriction.Bme1390I.size=Bio.Restriction.Restriction.Bme1390I-class.html#size,Variable Bio.Restriction.Restriction.Bme18I.size=Bio.Restriction.Restriction.Bme18I-class.html#size,Variable Bio.Restriction.Restriction.BmeDI.size=Bio.Restriction.Restriction.BmeDI-class.html#size,Variable Bio.Restriction.Restriction.BmeRI.size=Bio.Restriction.Restriction.BmeRI-class.html#size,Variable Bio.Restriction.Restriction.BmeT110I.size=Bio.Restriction.Restriction.BmeT110I-class.html#size,Variable Bio.Restriction.Restriction.BmgBI.size=Bio.Restriction.Restriction.BmgBI-class.html#size,Variable Bio.Restriction.Restriction.BmgI.size=Bio.Restriction.Restriction.BmgI-class.html#size,Variable Bio.Restriction.Restriction.BmgT120I.size=Bio.Restriction.Restriction.BmgT120I-class.html#size,Variable Bio.Restriction.Restriction.BmiI.size=Bio.Restriction.Restriction.BmiI-class.html#size,Variable Bio.Restriction.Restriction.BmrFI.size=Bio.Restriction.Restriction.BmrFI-class.html#size,Variable Bio.Restriction.Restriction.BmrI.size=Bio.Restriction.Restriction.BmrI-class.html#size,Variable Bio.Restriction.Restriction.BmsI.size=Bio.Restriction.Restriction.BmsI-class.html#size,Variable Bio.Restriction.Restriction.BmtI.size=Bio.Restriction.Restriction.BmtI-class.html#size,Variable Bio.Restriction.Restriction.BmuI.size=Bio.Restriction.Restriction.BmuI-class.html#size,Variable Bio.Restriction.Restriction.BoxI.size=Bio.Restriction.Restriction.BoxI-class.html#size,Variable Bio.Restriction.Restriction.BpiI.size=Bio.Restriction.Restriction.BpiI-class.html#size,Variable Bio.Restriction.Restriction.BplI.size=Bio.Restriction.Restriction.BplI-class.html#size,Variable Bio.Restriction.Restriction.BpmI.size=Bio.Restriction.Restriction.BpmI-class.html#size,Variable Bio.Restriction.Restriction.Bpu10I.size=Bio.Restriction.Restriction.Bpu10I-class.html#size,Variable Bio.Restriction.Restriction.Bpu1102I.size=Bio.Restriction.Restriction.Bpu1102I-class.html#size,Variable Bio.Restriction.Restriction.Bpu14I.size=Bio.Restriction.Restriction.Bpu14I-class.html#size,Variable Bio.Restriction.Restriction.BpuEI.size=Bio.Restriction.Restriction.BpuEI-class.html#size,Variable Bio.Restriction.Restriction.BpuMI.size=Bio.Restriction.Restriction.BpuMI-class.html#size,Variable Bio.Restriction.Restriction.BpvUI.size=Bio.Restriction.Restriction.BpvUI-class.html#size,Variable Bio.Restriction.Restriction.Bsa29I.size=Bio.Restriction.Restriction.Bsa29I-class.html#size,Variable Bio.Restriction.Restriction.BsaAI.size=Bio.Restriction.Restriction.BsaAI-class.html#size,Variable Bio.Restriction.Restriction.BsaBI.size=Bio.Restriction.Restriction.BsaBI-class.html#size,Variable Bio.Restriction.Restriction.BsaHI.size=Bio.Restriction.Restriction.BsaHI-class.html#size,Variable Bio.Restriction.Restriction.BsaI.size=Bio.Restriction.Restriction.BsaI-class.html#size,Variable Bio.Restriction.Restriction.BsaJI.size=Bio.Restriction.Restriction.BsaJI-class.html#size,Variable Bio.Restriction.Restriction.BsaWI.size=Bio.Restriction.Restriction.BsaWI-class.html#size,Variable Bio.Restriction.Restriction.BsaXI.size=Bio.Restriction.Restriction.BsaXI-class.html#size,Variable Bio.Restriction.Restriction.BsbI.size=Bio.Restriction.Restriction.BsbI-class.html#size,Variable Bio.Restriction.Restriction.Bsc4I.size=Bio.Restriction.Restriction.Bsc4I-class.html#size,Variable Bio.Restriction.Restriction.BscAI.size=Bio.Restriction.Restriction.BscAI-class.html#size,Variable Bio.Restriction.Restriction.BscGI.size=Bio.Restriction.Restriction.BscGI-class.html#size,Variable Bio.Restriction.Restriction.Bse118I.size=Bio.Restriction.Restriction.Bse118I-class.html#size,Variable Bio.Restriction.Restriction.Bse1I.size=Bio.Restriction.Restriction.Bse1I-class.html#size,Variable Bio.Restriction.Restriction.Bse21I.size=Bio.Restriction.Restriction.Bse21I-class.html#size,Variable Bio.Restriction.Restriction.Bse3DI.size=Bio.Restriction.Restriction.Bse3DI-class.html#size,Variable Bio.Restriction.Restriction.Bse8I.size=Bio.Restriction.Restriction.Bse8I-class.html#size,Variable Bio.Restriction.Restriction.BseAI.size=Bio.Restriction.Restriction.BseAI-class.html#size,Variable Bio.Restriction.Restriction.BseBI.size=Bio.Restriction.Restriction.BseBI-class.html#size,Variable Bio.Restriction.Restriction.BseCI.size=Bio.Restriction.Restriction.BseCI-class.html#size,Variable Bio.Restriction.Restriction.BseDI.size=Bio.Restriction.Restriction.BseDI-class.html#size,Variable Bio.Restriction.Restriction.BseGI.size=Bio.Restriction.Restriction.BseGI-class.html#size,Variable Bio.Restriction.Restriction.BseJI.size=Bio.Restriction.Restriction.BseJI-class.html#size,Variable Bio.Restriction.Restriction.BseLI.size=Bio.Restriction.Restriction.BseLI-class.html#size,Variable Bio.Restriction.Restriction.BseMI.size=Bio.Restriction.Restriction.BseMI-class.html#size,Variable Bio.Restriction.Restriction.BseMII.size=Bio.Restriction.Restriction.BseMII-class.html#size,Variable Bio.Restriction.Restriction.BseNI.size=Bio.Restriction.Restriction.BseNI-class.html#size,Variable Bio.Restriction.Restriction.BsePI.size=Bio.Restriction.Restriction.BsePI-class.html#size,Variable Bio.Restriction.Restriction.BseRI.size=Bio.Restriction.Restriction.BseRI-class.html#size,Variable Bio.Restriction.Restriction.BseSI.size=Bio.Restriction.Restriction.BseSI-class.html#size,Variable Bio.Restriction.Restriction.BseX3I.size=Bio.Restriction.Restriction.BseX3I-class.html#size,Variable Bio.Restriction.Restriction.BseXI.size=Bio.Restriction.Restriction.BseXI-class.html#size,Variable Bio.Restriction.Restriction.BseYI.size=Bio.Restriction.Restriction.BseYI-class.html#size,Variable Bio.Restriction.Restriction.BsgI.size=Bio.Restriction.Restriction.BsgI-class.html#size,Variable Bio.Restriction.Restriction.Bsh1236I.size=Bio.Restriction.Restriction.Bsh1236I-class.html#size,Variable Bio.Restriction.Restriction.Bsh1285I.size=Bio.Restriction.Restriction.Bsh1285I-class.html#size,Variable Bio.Restriction.Restriction.BshFI.size=Bio.Restriction.Restriction.BshFI-class.html#size,Variable Bio.Restriction.Restriction.BshNI.size=Bio.Restriction.Restriction.BshNI-class.html#size,Variable Bio.Restriction.Restriction.BshTI.size=Bio.Restriction.Restriction.BshTI-class.html#size,Variable Bio.Restriction.Restriction.BshVI.size=Bio.Restriction.Restriction.BshVI-class.html#size,Variable Bio.Restriction.Restriction.BsiEI.size=Bio.Restriction.Restriction.BsiEI-class.html#size,Variable Bio.Restriction.Restriction.BsiHKAI.size=Bio.Restriction.Restriction.BsiHKAI-class.html#size,Variable Bio.Restriction.Restriction.BsiHKCI.size=Bio.Restriction.Restriction.BsiHKCI-class.html#size,Variable Bio.Restriction.Restriction.BsiI.size=Bio.Restriction.Restriction.BsiI-class.html#size,Variable Bio.Restriction.Restriction.BsiSI.size=Bio.Restriction.Restriction.BsiSI-class.html#size,Variable Bio.Restriction.Restriction.BsiWI.size=Bio.Restriction.Restriction.BsiWI-class.html#size,Variable Bio.Restriction.Restriction.BsiYI.size=Bio.Restriction.Restriction.BsiYI-class.html#size,Variable Bio.Restriction.Restriction.BslFI.size=Bio.Restriction.Restriction.BslFI-class.html#size,Variable Bio.Restriction.Restriction.BslI.size=Bio.Restriction.Restriction.BslI-class.html#size,Variable Bio.Restriction.Restriction.BsmAI.size=Bio.Restriction.Restriction.BsmAI-class.html#size,Variable Bio.Restriction.Restriction.BsmBI.size=Bio.Restriction.Restriction.BsmBI-class.html#size,Variable Bio.Restriction.Restriction.BsmFI.size=Bio.Restriction.Restriction.BsmFI-class.html#size,Variable Bio.Restriction.Restriction.BsmI.size=Bio.Restriction.Restriction.BsmI-class.html#size,Variable Bio.Restriction.Restriction.BsnI.size=Bio.Restriction.Restriction.BsnI-class.html#size,Variable Bio.Restriction.Restriction.Bso31I.size=Bio.Restriction.Restriction.Bso31I-class.html#size,Variable Bio.Restriction.Restriction.BsoBI.size=Bio.Restriction.Restriction.BsoBI-class.html#size,Variable Bio.Restriction.Restriction.Bsp119I.size=Bio.Restriction.Restriction.Bsp119I-class.html#size,Variable Bio.Restriction.Restriction.Bsp120I.size=Bio.Restriction.Restriction.Bsp120I-class.html#size,Variable Bio.Restriction.Restriction.Bsp1286I.size=Bio.Restriction.Restriction.Bsp1286I-class.html#size,Variable Bio.Restriction.Restriction.Bsp13I.size=Bio.Restriction.Restriction.Bsp13I-class.html#size,Variable Bio.Restriction.Restriction.Bsp1407I.size=Bio.Restriction.Restriction.Bsp1407I-class.html#size,Variable Bio.Restriction.Restriction.Bsp143I.size=Bio.Restriction.Restriction.Bsp143I-class.html#size,Variable Bio.Restriction.Restriction.Bsp1720I.size=Bio.Restriction.Restriction.Bsp1720I-class.html#size,Variable Bio.Restriction.Restriction.Bsp19I.size=Bio.Restriction.Restriction.Bsp19I-class.html#size,Variable Bio.Restriction.Restriction.Bsp24I.size=Bio.Restriction.Restriction.Bsp24I-class.html#size,Variable Bio.Restriction.Restriction.Bsp68I.size=Bio.Restriction.Restriction.Bsp68I-class.html#size,Variable Bio.Restriction.Restriction.BspACI.size=Bio.Restriction.Restriction.BspACI-class.html#size,Variable Bio.Restriction.Restriction.BspCNI.size=Bio.Restriction.Restriction.BspCNI-class.html#size,Variable Bio.Restriction.Restriction.BspD6I.size=Bio.Restriction.Restriction.BspD6I-class.html#size,Variable Bio.Restriction.Restriction.BspDI.size=Bio.Restriction.Restriction.BspDI-class.html#size,Variable Bio.Restriction.Restriction.BspEI.size=Bio.Restriction.Restriction.BspEI-class.html#size,Variable Bio.Restriction.Restriction.BspFNI.size=Bio.Restriction.Restriction.BspFNI-class.html#size,Variable Bio.Restriction.Restriction.BspGI.size=Bio.Restriction.Restriction.BspGI-class.html#size,Variable Bio.Restriction.Restriction.BspHI.size=Bio.Restriction.Restriction.BspHI-class.html#size,Variable Bio.Restriction.Restriction.BspLI.size=Bio.Restriction.Restriction.BspLI-class.html#size,Variable Bio.Restriction.Restriction.BspLU11I.size=Bio.Restriction.Restriction.BspLU11I-class.html#size,Variable Bio.Restriction.Restriction.BspMI.size=Bio.Restriction.Restriction.BspMI-class.html#size,Variable Bio.Restriction.Restriction.BspMII.size=Bio.Restriction.Restriction.BspMII-class.html#size,Variable Bio.Restriction.Restriction.BspNCI.size=Bio.Restriction.Restriction.BspNCI-class.html#size,Variable Bio.Restriction.Restriction.BspOI.size=Bio.Restriction.Restriction.BspOI-class.html#size,Variable Bio.Restriction.Restriction.BspPI.size=Bio.Restriction.Restriction.BspPI-class.html#size,Variable Bio.Restriction.Restriction.BspQI.size=Bio.Restriction.Restriction.BspQI-class.html#size,Variable Bio.Restriction.Restriction.BspT104I.size=Bio.Restriction.Restriction.BspT104I-class.html#size,Variable Bio.Restriction.Restriction.BspT107I.size=Bio.Restriction.Restriction.BspT107I-class.html#size,Variable Bio.Restriction.Restriction.BspTI.size=Bio.Restriction.Restriction.BspTI-class.html#size,Variable Bio.Restriction.Restriction.BsrBI.size=Bio.Restriction.Restriction.BsrBI-class.html#size,Variable Bio.Restriction.Restriction.BsrDI.size=Bio.Restriction.Restriction.BsrDI-class.html#size,Variable Bio.Restriction.Restriction.BsrFI.size=Bio.Restriction.Restriction.BsrFI-class.html#size,Variable Bio.Restriction.Restriction.BsrGI.size=Bio.Restriction.Restriction.BsrGI-class.html#size,Variable Bio.Restriction.Restriction.BsrI.size=Bio.Restriction.Restriction.BsrI-class.html#size,Variable Bio.Restriction.Restriction.BsrSI.size=Bio.Restriction.Restriction.BsrSI-class.html#size,Variable Bio.Restriction.Restriction.BssAI.size=Bio.Restriction.Restriction.BssAI-class.html#size,Variable Bio.Restriction.Restriction.BssECI.size=Bio.Restriction.Restriction.BssECI-class.html#size,Variable Bio.Restriction.Restriction.BssHII.size=Bio.Restriction.Restriction.BssHII-class.html#size,Variable Bio.Restriction.Restriction.BssKI.size=Bio.Restriction.Restriction.BssKI-class.html#size,Variable Bio.Restriction.Restriction.BssMI.size=Bio.Restriction.Restriction.BssMI-class.html#size,Variable Bio.Restriction.Restriction.BssNAI.size=Bio.Restriction.Restriction.BssNAI-class.html#size,Variable Bio.Restriction.Restriction.BssNI.size=Bio.Restriction.Restriction.BssNI-class.html#size,Variable Bio.Restriction.Restriction.BssSI.size=Bio.Restriction.Restriction.BssSI-class.html#size,Variable Bio.Restriction.Restriction.BssT1I.size=Bio.Restriction.Restriction.BssT1I-class.html#size,Variable Bio.Restriction.Restriction.Bst1107I.size=Bio.Restriction.Restriction.Bst1107I-class.html#size,Variable Bio.Restriction.Restriction.Bst2BI.size=Bio.Restriction.Restriction.Bst2BI-class.html#size,Variable Bio.Restriction.Restriction.Bst2UI.size=Bio.Restriction.Restriction.Bst2UI-class.html#size,Variable Bio.Restriction.Restriction.Bst4CI.size=Bio.Restriction.Restriction.Bst4CI-class.html#size,Variable Bio.Restriction.Restriction.Bst6I.size=Bio.Restriction.Restriction.Bst6I-class.html#size,Variable Bio.Restriction.Restriction.BstACI.size=Bio.Restriction.Restriction.BstACI-class.html#size,Variable Bio.Restriction.Restriction.BstAFI.size=Bio.Restriction.Restriction.BstAFI-class.html#size,Variable Bio.Restriction.Restriction.BstAPI.size=Bio.Restriction.Restriction.BstAPI-class.html#size,Variable Bio.Restriction.Restriction.BstAUI.size=Bio.Restriction.Restriction.BstAUI-class.html#size,Variable Bio.Restriction.Restriction.BstBAI.size=Bio.Restriction.Restriction.BstBAI-class.html#size,Variable Bio.Restriction.Restriction.BstBI.size=Bio.Restriction.Restriction.BstBI-class.html#size,Variable Bio.Restriction.Restriction.BstC8I.size=Bio.Restriction.Restriction.BstC8I-class.html#size,Variable Bio.Restriction.Restriction.BstDEI.size=Bio.Restriction.Restriction.BstDEI-class.html#size,Variable Bio.Restriction.Restriction.BstDSI.size=Bio.Restriction.Restriction.BstDSI-class.html#size,Variable Bio.Restriction.Restriction.BstEII.size=Bio.Restriction.Restriction.BstEII-class.html#size,Variable Bio.Restriction.Restriction.BstENI.size=Bio.Restriction.Restriction.BstENI-class.html#size,Variable Bio.Restriction.Restriction.BstF5I.size=Bio.Restriction.Restriction.BstF5I-class.html#size,Variable Bio.Restriction.Restriction.BstFNI.size=Bio.Restriction.Restriction.BstFNI-class.html#size,Variable Bio.Restriction.Restriction.BstH2I.size=Bio.Restriction.Restriction.BstH2I-class.html#size,Variable Bio.Restriction.Restriction.BstHHI.size=Bio.Restriction.Restriction.BstHHI-class.html#size,Variable Bio.Restriction.Restriction.BstKTI.size=Bio.Restriction.Restriction.BstKTI-class.html#size,Variable Bio.Restriction.Restriction.BstMAI.size=Bio.Restriction.Restriction.BstMAI-class.html#size,Variable Bio.Restriction.Restriction.BstMBI.size=Bio.Restriction.Restriction.BstMBI-class.html#size,Variable Bio.Restriction.Restriction.BstMCI.size=Bio.Restriction.Restriction.BstMCI-class.html#size,Variable Bio.Restriction.Restriction.BstMWI.size=Bio.Restriction.Restriction.BstMWI-class.html#size,Variable Bio.Restriction.Restriction.BstNI.size=Bio.Restriction.Restriction.BstNI-class.html#size,Variable Bio.Restriction.Restriction.BstNSI.size=Bio.Restriction.Restriction.BstNSI-class.html#size,Variable Bio.Restriction.Restriction.BstOI.size=Bio.Restriction.Restriction.BstOI-class.html#size,Variable Bio.Restriction.Restriction.BstPAI.size=Bio.Restriction.Restriction.BstPAI-class.html#size,Variable Bio.Restriction.Restriction.BstPI.size=Bio.Restriction.Restriction.BstPI-class.html#size,Variable Bio.Restriction.Restriction.BstSCI.size=Bio.Restriction.Restriction.BstSCI-class.html#size,Variable Bio.Restriction.Restriction.BstSFI.size=Bio.Restriction.Restriction.BstSFI-class.html#size,Variable Bio.Restriction.Restriction.BstSLI.size=Bio.Restriction.Restriction.BstSLI-class.html#size,Variable Bio.Restriction.Restriction.BstSNI.size=Bio.Restriction.Restriction.BstSNI-class.html#size,Variable Bio.Restriction.Restriction.BstUI.size=Bio.Restriction.Restriction.BstUI-class.html#size,Variable Bio.Restriction.Restriction.BstV1I.size=Bio.Restriction.Restriction.BstV1I-class.html#size,Variable Bio.Restriction.Restriction.BstV2I.size=Bio.Restriction.Restriction.BstV2I-class.html#size,Variable Bio.Restriction.Restriction.BstX2I.size=Bio.Restriction.Restriction.BstX2I-class.html#size,Variable Bio.Restriction.Restriction.BstXI.size=Bio.Restriction.Restriction.BstXI-class.html#size,Variable Bio.Restriction.Restriction.BstYI.size=Bio.Restriction.Restriction.BstYI-class.html#size,Variable Bio.Restriction.Restriction.BstZ17I.size=Bio.Restriction.Restriction.BstZ17I-class.html#size,Variable Bio.Restriction.Restriction.BstZI.size=Bio.Restriction.Restriction.BstZI-class.html#size,Variable Bio.Restriction.Restriction.Bsu15I.size=Bio.Restriction.Restriction.Bsu15I-class.html#size,Variable Bio.Restriction.Restriction.Bsu36I.size=Bio.Restriction.Restriction.Bsu36I-class.html#size,Variable Bio.Restriction.Restriction.BsuI.size=Bio.Restriction.Restriction.BsuI-class.html#size,Variable Bio.Restriction.Restriction.BsuRI.size=Bio.Restriction.Restriction.BsuRI-class.html#size,Variable Bio.Restriction.Restriction.BtgI.size=Bio.Restriction.Restriction.BtgI-class.html#size,Variable Bio.Restriction.Restriction.BtgZI.size=Bio.Restriction.Restriction.BtgZI-class.html#size,Variable Bio.Restriction.Restriction.BthCI.size=Bio.Restriction.Restriction.BthCI-class.html#size,Variable Bio.Restriction.Restriction.BtrI.size=Bio.Restriction.Restriction.BtrI-class.html#size,Variable Bio.Restriction.Restriction.BtsCI.size=Bio.Restriction.Restriction.BtsCI-class.html#size,Variable Bio.Restriction.Restriction.BtsI.size=Bio.Restriction.Restriction.BtsI-class.html#size,Variable Bio.Restriction.Restriction.BtsIMutI.size=Bio.Restriction.Restriction.BtsIMutI-class.html#size,Variable Bio.Restriction.Restriction.BtuMI.size=Bio.Restriction.Restriction.BtuMI-class.html#size,Variable Bio.Restriction.Restriction.BveI.size=Bio.Restriction.Restriction.BveI-class.html#size,Variable Bio.Restriction.Restriction.Cac8I.size=Bio.Restriction.Restriction.Cac8I-class.html#size,Variable Bio.Restriction.Restriction.CaiI.size=Bio.Restriction.Restriction.CaiI-class.html#size,Variable Bio.Restriction.Restriction.CauII.size=Bio.Restriction.Restriction.CauII-class.html#size,Variable Bio.Restriction.Restriction.CchII.size=Bio.Restriction.Restriction.CchII-class.html#size,Variable Bio.Restriction.Restriction.CchIII.size=Bio.Restriction.Restriction.CchIII-class.html#size,Variable Bio.Restriction.Restriction.CciI.size=Bio.Restriction.Restriction.CciI-class.html#size,Variable Bio.Restriction.Restriction.CciNI.size=Bio.Restriction.Restriction.CciNI-class.html#size,Variable Bio.Restriction.Restriction.Cdi630V.size=Bio.Restriction.Restriction.Cdi630V-class.html#size,Variable Bio.Restriction.Restriction.CdiI.size=Bio.Restriction.Restriction.CdiI-class.html#size,Variable Bio.Restriction.Restriction.CdpI.size=Bio.Restriction.Restriction.CdpI-class.html#size,Variable Bio.Restriction.Restriction.CfoI.size=Bio.Restriction.Restriction.CfoI-class.html#size,Variable Bio.Restriction.Restriction.Cfr10I.size=Bio.Restriction.Restriction.Cfr10I-class.html#size,Variable Bio.Restriction.Restriction.Cfr13I.size=Bio.Restriction.Restriction.Cfr13I-class.html#size,Variable Bio.Restriction.Restriction.Cfr42I.size=Bio.Restriction.Restriction.Cfr42I-class.html#size,Variable Bio.Restriction.Restriction.Cfr9I.size=Bio.Restriction.Restriction.Cfr9I-class.html#size,Variable Bio.Restriction.Restriction.CfrI.size=Bio.Restriction.Restriction.CfrI-class.html#size,Variable Bio.Restriction.Restriction.Cgl13032I.size=Bio.Restriction.Restriction.Cgl13032I-class.html#size,Variable Bio.Restriction.Restriction.Cgl13032II.size=Bio.Restriction.Restriction.Cgl13032II-class.html#size,Variable Bio.Restriction.Restriction.ChaI.size=Bio.Restriction.Restriction.ChaI-class.html#size,Variable Bio.Restriction.Restriction.CjeFIII.size=Bio.Restriction.Restriction.CjeFIII-class.html#size,Variable Bio.Restriction.Restriction.CjeFV.size=Bio.Restriction.Restriction.CjeFV-class.html#size,Variable Bio.Restriction.Restriction.CjeI.size=Bio.Restriction.Restriction.CjeI-class.html#size,Variable Bio.Restriction.Restriction.CjeNII.size=Bio.Restriction.Restriction.CjeNII-class.html#size,Variable Bio.Restriction.Restriction.CjeNIII.size=Bio.Restriction.Restriction.CjeNIII-class.html#size,Variable Bio.Restriction.Restriction.CjeP659IV.size=Bio.Restriction.Restriction.CjeP659IV-class.html#size,Variable Bio.Restriction.Restriction.CjePI.size=Bio.Restriction.Restriction.CjePI-class.html#size,Variable Bio.Restriction.Restriction.CjuI.size=Bio.Restriction.Restriction.CjuI-class.html#size,Variable Bio.Restriction.Restriction.CjuII.size=Bio.Restriction.Restriction.CjuII-class.html#size,Variable Bio.Restriction.Restriction.ClaI.size=Bio.Restriction.Restriction.ClaI-class.html#size,Variable Bio.Restriction.Restriction.CpoI.size=Bio.Restriction.Restriction.CpoI-class.html#size,Variable Bio.Restriction.Restriction.CseI.size=Bio.Restriction.Restriction.CseI-class.html#size,Variable Bio.Restriction.Restriction.CsiI.size=Bio.Restriction.Restriction.CsiI-class.html#size,Variable Bio.Restriction.Restriction.Csp6I.size=Bio.Restriction.Restriction.Csp6I-class.html#size,Variable Bio.Restriction.Restriction.CspAI.size=Bio.Restriction.Restriction.CspAI-class.html#size,Variable Bio.Restriction.Restriction.CspCI.size=Bio.Restriction.Restriction.CspCI-class.html#size,Variable Bio.Restriction.Restriction.CspI.size=Bio.Restriction.Restriction.CspI-class.html#size,Variable Bio.Restriction.Restriction.CstMI.size=Bio.Restriction.Restriction.CstMI-class.html#size,Variable Bio.Restriction.Restriction.CviAII.size=Bio.Restriction.Restriction.CviAII-class.html#size,Variable Bio.Restriction.Restriction.CviJI.size=Bio.Restriction.Restriction.CviJI-class.html#size,Variable Bio.Restriction.Restriction.CviKI_1.size=Bio.Restriction.Restriction.CviKI_1-class.html#size,Variable Bio.Restriction.Restriction.CviQI.size=Bio.Restriction.Restriction.CviQI-class.html#size,Variable Bio.Restriction.Restriction.CviRI.size=Bio.Restriction.Restriction.CviRI-class.html#size,Variable Bio.Restriction.Restriction.DdeI.size=Bio.Restriction.Restriction.DdeI-class.html#size,Variable Bio.Restriction.Restriction.DinI.size=Bio.Restriction.Restriction.DinI-class.html#size,Variable Bio.Restriction.Restriction.DpnI.size=Bio.Restriction.Restriction.DpnI-class.html#size,Variable Bio.Restriction.Restriction.DpnII.size=Bio.Restriction.Restriction.DpnII-class.html#size,Variable Bio.Restriction.Restriction.DraI.size=Bio.Restriction.Restriction.DraI-class.html#size,Variable Bio.Restriction.Restriction.DraII.size=Bio.Restriction.Restriction.DraII-class.html#size,Variable Bio.Restriction.Restriction.DraIII.size=Bio.Restriction.Restriction.DraIII-class.html#size,Variable Bio.Restriction.Restriction.DraRI.size=Bio.Restriction.Restriction.DraRI-class.html#size,Variable Bio.Restriction.Restriction.DrdI.size=Bio.Restriction.Restriction.DrdI-class.html#size,Variable Bio.Restriction.Restriction.DrdII.size=Bio.Restriction.Restriction.DrdII-class.html#size,Variable Bio.Restriction.Restriction.DriI.size=Bio.Restriction.Restriction.DriI-class.html#size,Variable Bio.Restriction.Restriction.DsaI.size=Bio.Restriction.Restriction.DsaI-class.html#size,Variable Bio.Restriction.Restriction.DseDI.size=Bio.Restriction.Restriction.DseDI-class.html#size,Variable Bio.Restriction.Restriction.EaeI.size=Bio.Restriction.Restriction.EaeI-class.html#size,Variable Bio.Restriction.Restriction.EagI.size=Bio.Restriction.Restriction.EagI-class.html#size,Variable Bio.Restriction.Restriction.Eam1104I.size=Bio.Restriction.Restriction.Eam1104I-class.html#size,Variable Bio.Restriction.Restriction.Eam1105I.size=Bio.Restriction.Restriction.Eam1105I-class.html#size,Variable Bio.Restriction.Restriction.EarI.size=Bio.Restriction.Restriction.EarI-class.html#size,Variable Bio.Restriction.Restriction.EciI.size=Bio.Restriction.Restriction.EciI-class.html#size,Variable Bio.Restriction.Restriction.Ecl136II.size=Bio.Restriction.Restriction.Ecl136II-class.html#size,Variable Bio.Restriction.Restriction.EclXI.size=Bio.Restriction.Restriction.EclXI-class.html#size,Variable Bio.Restriction.Restriction.Eco105I.size=Bio.Restriction.Restriction.Eco105I-class.html#size,Variable Bio.Restriction.Restriction.Eco130I.size=Bio.Restriction.Restriction.Eco130I-class.html#size,Variable Bio.Restriction.Restriction.Eco147I.size=Bio.Restriction.Restriction.Eco147I-class.html#size,Variable Bio.Restriction.Restriction.Eco24I.size=Bio.Restriction.Restriction.Eco24I-class.html#size,Variable Bio.Restriction.Restriction.Eco31I.size=Bio.Restriction.Restriction.Eco31I-class.html#size,Variable Bio.Restriction.Restriction.Eco32I.size=Bio.Restriction.Restriction.Eco32I-class.html#size,Variable Bio.Restriction.Restriction.Eco47I.size=Bio.Restriction.Restriction.Eco47I-class.html#size,Variable Bio.Restriction.Restriction.Eco47III.size=Bio.Restriction.Restriction.Eco47III-class.html#size,Variable Bio.Restriction.Restriction.Eco52I.size=Bio.Restriction.Restriction.Eco52I-class.html#size,Variable Bio.Restriction.Restriction.Eco53kI.size=Bio.Restriction.Restriction.Eco53kI-class.html#size,Variable Bio.Restriction.Restriction.Eco57I.size=Bio.Restriction.Restriction.Eco57I-class.html#size,Variable Bio.Restriction.Restriction.Eco57MI.size=Bio.Restriction.Restriction.Eco57MI-class.html#size,Variable Bio.Restriction.Restriction.Eco72I.size=Bio.Restriction.Restriction.Eco72I-class.html#size,Variable Bio.Restriction.Restriction.Eco81I.size=Bio.Restriction.Restriction.Eco81I-class.html#size,Variable Bio.Restriction.Restriction.Eco88I.size=Bio.Restriction.Restriction.Eco88I-class.html#size,Variable Bio.Restriction.Restriction.Eco91I.size=Bio.Restriction.Restriction.Eco91I-class.html#size,Variable Bio.Restriction.Restriction.EcoHI.size=Bio.Restriction.Restriction.EcoHI-class.html#size,Variable Bio.Restriction.Restriction.EcoICRI.size=Bio.Restriction.Restriction.EcoICRI-class.html#size,Variable Bio.Restriction.Restriction.EcoNI.size=Bio.Restriction.Restriction.EcoNI-class.html#size,Variable Bio.Restriction.Restriction.EcoO109I.size=Bio.Restriction.Restriction.EcoO109I-class.html#size,Variable Bio.Restriction.Restriction.EcoO65I.size=Bio.Restriction.Restriction.EcoO65I-class.html#size,Variable Bio.Restriction.Restriction.EcoRI.size=Bio.Restriction.Restriction.EcoRI-class.html#size,Variable Bio.Restriction.Restriction.EcoRII.size=Bio.Restriction.Restriction.EcoRII-class.html#size,Variable Bio.Restriction.Restriction.EcoRV.size=Bio.Restriction.Restriction.EcoRV-class.html#size,Variable Bio.Restriction.Restriction.EcoT14I.size=Bio.Restriction.Restriction.EcoT14I-class.html#size,Variable Bio.Restriction.Restriction.EcoT22I.size=Bio.Restriction.Restriction.EcoT22I-class.html#size,Variable Bio.Restriction.Restriction.EcoT38I.size=Bio.Restriction.Restriction.EcoT38I-class.html#size,Variable Bio.Restriction.Restriction.EgeI.size=Bio.Restriction.Restriction.EgeI-class.html#size,Variable Bio.Restriction.Restriction.EheI.size=Bio.Restriction.Restriction.EheI-class.html#size,Variable Bio.Restriction.Restriction.ErhI.size=Bio.Restriction.Restriction.ErhI-class.html#size,Variable Bio.Restriction.Restriction.EsaBC3I.size=Bio.Restriction.Restriction.EsaBC3I-class.html#size,Variable Bio.Restriction.Restriction.EsaSSI.size=Bio.Restriction.Restriction.EsaSSI-class.html#size,Variable Bio.Restriction.Restriction.Esp3I.size=Bio.Restriction.Restriction.Esp3I-class.html#size,Variable Bio.Restriction.Restriction.EspI.size=Bio.Restriction.Restriction.EspI-class.html#size,Variable Bio.Restriction.Restriction.FaeI.size=Bio.Restriction.Restriction.FaeI-class.html#size,Variable Bio.Restriction.Restriction.FaiI.size=Bio.Restriction.Restriction.FaiI-class.html#size,Variable Bio.Restriction.Restriction.FalI.size=Bio.Restriction.Restriction.FalI-class.html#size,Variable Bio.Restriction.Restriction.FaqI.size=Bio.Restriction.Restriction.FaqI-class.html#size,Variable Bio.Restriction.Restriction.FatI.size=Bio.Restriction.Restriction.FatI-class.html#size,Variable Bio.Restriction.Restriction.FauI.size=Bio.Restriction.Restriction.FauI-class.html#size,Variable Bio.Restriction.Restriction.FauNDI.size=Bio.Restriction.Restriction.FauNDI-class.html#size,Variable Bio.Restriction.Restriction.FbaI.size=Bio.Restriction.Restriction.FbaI-class.html#size,Variable Bio.Restriction.Restriction.FblI.size=Bio.Restriction.Restriction.FblI-class.html#size,Variable Bio.Restriction.Restriction.FinI.size=Bio.Restriction.Restriction.FinI-class.html#size,Variable Bio.Restriction.Restriction.FmuI.size=Bio.Restriction.Restriction.FmuI-class.html#size,Variable Bio.Restriction.Restriction.Fnu4HI.size=Bio.Restriction.Restriction.Fnu4HI-class.html#size,Variable Bio.Restriction.Restriction.FnuDII.size=Bio.Restriction.Restriction.FnuDII-class.html#size,Variable Bio.Restriction.Restriction.FokI.size=Bio.Restriction.Restriction.FokI-class.html#size,Variable Bio.Restriction.Restriction.FriOI.size=Bio.Restriction.Restriction.FriOI-class.html#size,Variable Bio.Restriction.Restriction.FseI.size=Bio.Restriction.Restriction.FseI-class.html#size,Variable Bio.Restriction.Restriction.Fsp4HI.size=Bio.Restriction.Restriction.Fsp4HI-class.html#size,Variable Bio.Restriction.Restriction.FspAI.size=Bio.Restriction.Restriction.FspAI-class.html#size,Variable Bio.Restriction.Restriction.FspBI.size=Bio.Restriction.Restriction.FspBI-class.html#size,Variable Bio.Restriction.Restriction.FspEI.size=Bio.Restriction.Restriction.FspEI-class.html#size,Variable Bio.Restriction.Restriction.FspI.size=Bio.Restriction.Restriction.FspI-class.html#size,Variable Bio.Restriction.Restriction.GauT27I.size=Bio.Restriction.Restriction.GauT27I-class.html#size,Variable Bio.Restriction.Restriction.GdiII.size=Bio.Restriction.Restriction.GdiII-class.html#size,Variable Bio.Restriction.Restriction.GlaI.size=Bio.Restriction.Restriction.GlaI-class.html#size,Variable Bio.Restriction.Restriction.GluI.size=Bio.Restriction.Restriction.GluI-class.html#size,Variable Bio.Restriction.Restriction.GsaI.size=Bio.Restriction.Restriction.GsaI-class.html#size,Variable Bio.Restriction.Restriction.GsuI.size=Bio.Restriction.Restriction.GsuI-class.html#size,Variable Bio.Restriction.Restriction.HaeI.size=Bio.Restriction.Restriction.HaeI-class.html#size,Variable Bio.Restriction.Restriction.HaeII.size=Bio.Restriction.Restriction.HaeII-class.html#size,Variable Bio.Restriction.Restriction.HaeIII.size=Bio.Restriction.Restriction.HaeIII-class.html#size,Variable Bio.Restriction.Restriction.HapII.size=Bio.Restriction.Restriction.HapII-class.html#size,Variable Bio.Restriction.Restriction.HauII.size=Bio.Restriction.Restriction.HauII-class.html#size,Variable Bio.Restriction.Restriction.HgaI.size=Bio.Restriction.Restriction.HgaI-class.html#size,Variable Bio.Restriction.Restriction.HgiAI.size=Bio.Restriction.Restriction.HgiAI-class.html#size,Variable Bio.Restriction.Restriction.HgiCI.size=Bio.Restriction.Restriction.HgiCI-class.html#size,Variable Bio.Restriction.Restriction.HgiEII.size=Bio.Restriction.Restriction.HgiEII-class.html#size,Variable Bio.Restriction.Restriction.HgiJII.size=Bio.Restriction.Restriction.HgiJII-class.html#size,Variable Bio.Restriction.Restriction.HhaI.size=Bio.Restriction.Restriction.HhaI-class.html#size,Variable Bio.Restriction.Restriction.Hin1I.size=Bio.Restriction.Restriction.Hin1I-class.html#size,Variable Bio.Restriction.Restriction.Hin1II.size=Bio.Restriction.Restriction.Hin1II-class.html#size,Variable Bio.Restriction.Restriction.Hin4I.size=Bio.Restriction.Restriction.Hin4I-class.html#size,Variable Bio.Restriction.Restriction.Hin4II.size=Bio.Restriction.Restriction.Hin4II-class.html#size,Variable Bio.Restriction.Restriction.Hin6I.size=Bio.Restriction.Restriction.Hin6I-class.html#size,Variable Bio.Restriction.Restriction.HinP1I.size=Bio.Restriction.Restriction.HinP1I-class.html#size,Variable Bio.Restriction.Restriction.HincII.size=Bio.Restriction.Restriction.HincII-class.html#size,Variable Bio.Restriction.Restriction.HindII.size=Bio.Restriction.Restriction.HindII-class.html#size,Variable Bio.Restriction.Restriction.HindIII.size=Bio.Restriction.Restriction.HindIII-class.html#size,Variable Bio.Restriction.Restriction.HinfI.size=Bio.Restriction.Restriction.HinfI-class.html#size,Variable Bio.Restriction.Restriction.HpaI.size=Bio.Restriction.Restriction.HpaI-class.html#size,Variable Bio.Restriction.Restriction.HpaII.size=Bio.Restriction.Restriction.HpaII-class.html#size,Variable Bio.Restriction.Restriction.HphI.size=Bio.Restriction.Restriction.HphI-class.html#size,Variable Bio.Restriction.Restriction.Hpy166II.size=Bio.Restriction.Restriction.Hpy166II-class.html#size,Variable Bio.Restriction.Restriction.Hpy178III.size=Bio.Restriction.Restriction.Hpy178III-class.html#size,Variable Bio.Restriction.Restriction.Hpy188I.size=Bio.Restriction.Restriction.Hpy188I-class.html#size,Variable Bio.Restriction.Restriction.Hpy188III.size=Bio.Restriction.Restriction.Hpy188III-class.html#size,Variable Bio.Restriction.Restriction.Hpy8I.size=Bio.Restriction.Restriction.Hpy8I-class.html#size,Variable Bio.Restriction.Restriction.Hpy99I.size=Bio.Restriction.Restriction.Hpy99I-class.html#size,Variable Bio.Restriction.Restriction.Hpy99XIII.size=Bio.Restriction.Restriction.Hpy99XIII-class.html#size,Variable Bio.Restriction.Restriction.Hpy99XIV.size=Bio.Restriction.Restriction.Hpy99XIV-class.html#size,Variable Bio.Restriction.Restriction.HpyAV.size=Bio.Restriction.Restriction.HpyAV-class.html#size,Variable Bio.Restriction.Restriction.HpyCH4III.size=Bio.Restriction.Restriction.HpyCH4III-class.html#size,Variable Bio.Restriction.Restriction.HpyCH4IV.size=Bio.Restriction.Restriction.HpyCH4IV-class.html#size,Variable Bio.Restriction.Restriction.HpyCH4V.size=Bio.Restriction.Restriction.HpyCH4V-class.html#size,Variable Bio.Restriction.Restriction.HpyF10VI.size=Bio.Restriction.Restriction.HpyF10VI-class.html#size,Variable Bio.Restriction.Restriction.HpyF3I.size=Bio.Restriction.Restriction.HpyF3I-class.html#size,Variable Bio.Restriction.Restriction.HpySE526I.size=Bio.Restriction.Restriction.HpySE526I-class.html#size,Variable Bio.Restriction.Restriction.Hsp92I.size=Bio.Restriction.Restriction.Hsp92I-class.html#size,Variable Bio.Restriction.Restriction.Hsp92II.size=Bio.Restriction.Restriction.Hsp92II-class.html#size,Variable Bio.Restriction.Restriction.HspAI.size=Bio.Restriction.Restriction.HspAI-class.html#size,Variable Bio.Restriction.Restriction.Jma19592I.size=Bio.Restriction.Restriction.Jma19592I-class.html#size,Variable Bio.Restriction.Restriction.KasI.size=Bio.Restriction.Restriction.KasI-class.html#size,Variable Bio.Restriction.Restriction.KflI.size=Bio.Restriction.Restriction.KflI-class.html#size,Variable Bio.Restriction.Restriction.Kpn2I.size=Bio.Restriction.Restriction.Kpn2I-class.html#size,Variable Bio.Restriction.Restriction.KpnI.size=Bio.Restriction.Restriction.KpnI-class.html#size,Variable Bio.Restriction.Restriction.KroI.size=Bio.Restriction.Restriction.KroI-class.html#size,Variable Bio.Restriction.Restriction.Ksp22I.size=Bio.Restriction.Restriction.Ksp22I-class.html#size,Variable Bio.Restriction.Restriction.Ksp632I.size=Bio.Restriction.Restriction.Ksp632I-class.html#size,Variable Bio.Restriction.Restriction.KspAI.size=Bio.Restriction.Restriction.KspAI-class.html#size,Variable Bio.Restriction.Restriction.KspI.size=Bio.Restriction.Restriction.KspI-class.html#size,Variable Bio.Restriction.Restriction.Kzo9I.size=Bio.Restriction.Restriction.Kzo9I-class.html#size,Variable Bio.Restriction.Restriction.LguI.size=Bio.Restriction.Restriction.LguI-class.html#size,Variable Bio.Restriction.Restriction.LpnI.size=Bio.Restriction.Restriction.LpnI-class.html#size,Variable Bio.Restriction.Restriction.LpnPI.size=Bio.Restriction.Restriction.LpnPI-class.html#size,Variable Bio.Restriction.Restriction.Lsp1109I.size=Bio.Restriction.Restriction.Lsp1109I-class.html#size,Variable Bio.Restriction.Restriction.LweI.size=Bio.Restriction.Restriction.LweI-class.html#size,Variable Bio.Restriction.Restriction.MabI.size=Bio.Restriction.Restriction.MabI-class.html#size,Variable Bio.Restriction.Restriction.MaeI.size=Bio.Restriction.Restriction.MaeI-class.html#size,Variable Bio.Restriction.Restriction.MaeII.size=Bio.Restriction.Restriction.MaeII-class.html#size,Variable Bio.Restriction.Restriction.MaeIII.size=Bio.Restriction.Restriction.MaeIII-class.html#size,Variable Bio.Restriction.Restriction.MalI.size=Bio.Restriction.Restriction.MalI-class.html#size,Variable Bio.Restriction.Restriction.MaqI.size=Bio.Restriction.Restriction.MaqI-class.html#size,Variable Bio.Restriction.Restriction.MauBI.size=Bio.Restriction.Restriction.MauBI-class.html#size,Variable Bio.Restriction.Restriction.MbiI.size=Bio.Restriction.Restriction.MbiI-class.html#size,Variable Bio.Restriction.Restriction.MboI.size=Bio.Restriction.Restriction.MboI-class.html#size,Variable Bio.Restriction.Restriction.MboII.size=Bio.Restriction.Restriction.MboII-class.html#size,Variable Bio.Restriction.Restriction.McaTI.size=Bio.Restriction.Restriction.McaTI-class.html#size,Variable Bio.Restriction.Restriction.McrI.size=Bio.Restriction.Restriction.McrI-class.html#size,Variable Bio.Restriction.Restriction.MfeI.size=Bio.Restriction.Restriction.MfeI-class.html#size,Variable Bio.Restriction.Restriction.MflI.size=Bio.Restriction.Restriction.MflI-class.html#size,Variable Bio.Restriction.Restriction.MhlI.size=Bio.Restriction.Restriction.MhlI-class.html#size,Variable Bio.Restriction.Restriction.MjaIV.size=Bio.Restriction.Restriction.MjaIV-class.html#size,Variable Bio.Restriction.Restriction.MkaDII.size=Bio.Restriction.Restriction.MkaDII-class.html#size,Variable Bio.Restriction.Restriction.MlsI.size=Bio.Restriction.Restriction.MlsI-class.html#size,Variable Bio.Restriction.Restriction.MluCI.size=Bio.Restriction.Restriction.MluCI-class.html#size,Variable Bio.Restriction.Restriction.MluI.size=Bio.Restriction.Restriction.MluI-class.html#size,Variable Bio.Restriction.Restriction.MluNI.size=Bio.Restriction.Restriction.MluNI-class.html#size,Variable Bio.Restriction.Restriction.Mly113I.size=Bio.Restriction.Restriction.Mly113I-class.html#size,Variable Bio.Restriction.Restriction.MlyI.size=Bio.Restriction.Restriction.MlyI-class.html#size,Variable Bio.Restriction.Restriction.MmeI.size=Bio.Restriction.Restriction.MmeI-class.html#size,Variable Bio.Restriction.Restriction.MnlI.size=Bio.Restriction.Restriction.MnlI-class.html#size,Variable Bio.Restriction.Restriction.Mph1103I.size=Bio.Restriction.Restriction.Mph1103I-class.html#size,Variable Bio.Restriction.Restriction.MreI.size=Bio.Restriction.Restriction.MreI-class.html#size,Variable Bio.Restriction.Restriction.MroI.size=Bio.Restriction.Restriction.MroI-class.html#size,Variable Bio.Restriction.Restriction.MroNI.size=Bio.Restriction.Restriction.MroNI-class.html#size,Variable Bio.Restriction.Restriction.MroXI.size=Bio.Restriction.Restriction.MroXI-class.html#size,Variable Bio.Restriction.Restriction.MscI.size=Bio.Restriction.Restriction.MscI-class.html#size,Variable Bio.Restriction.Restriction.MseI.size=Bio.Restriction.Restriction.MseI-class.html#size,Variable Bio.Restriction.Restriction.MslI.size=Bio.Restriction.Restriction.MslI-class.html#size,Variable Bio.Restriction.Restriction.Msp20I.size=Bio.Restriction.Restriction.Msp20I-class.html#size,Variable Bio.Restriction.Restriction.MspA1I.size=Bio.Restriction.Restriction.MspA1I-class.html#size,Variable Bio.Restriction.Restriction.MspCI.size=Bio.Restriction.Restriction.MspCI-class.html#size,Variable Bio.Restriction.Restriction.MspI.size=Bio.Restriction.Restriction.MspI-class.html#size,Variable Bio.Restriction.Restriction.MspJI.size=Bio.Restriction.Restriction.MspJI-class.html#size,Variable Bio.Restriction.Restriction.MspR9I.size=Bio.Restriction.Restriction.MspR9I-class.html#size,Variable Bio.Restriction.Restriction.MssI.size=Bio.Restriction.Restriction.MssI-class.html#size,Variable Bio.Restriction.Restriction.MstI.size=Bio.Restriction.Restriction.MstI-class.html#size,Variable Bio.Restriction.Restriction.MunI.size=Bio.Restriction.Restriction.MunI-class.html#size,Variable Bio.Restriction.Restriction.Mva1269I.size=Bio.Restriction.Restriction.Mva1269I-class.html#size,Variable Bio.Restriction.Restriction.MvaI.size=Bio.Restriction.Restriction.MvaI-class.html#size,Variable Bio.Restriction.Restriction.MvnI.size=Bio.Restriction.Restriction.MvnI-class.html#size,Variable Bio.Restriction.Restriction.MvrI.size=Bio.Restriction.Restriction.MvrI-class.html#size,Variable Bio.Restriction.Restriction.MwoI.size=Bio.Restriction.Restriction.MwoI-class.html#size,Variable Bio.Restriction.Restriction.NaeI.size=Bio.Restriction.Restriction.NaeI-class.html#size,Variable Bio.Restriction.Restriction.NarI.size=Bio.Restriction.Restriction.NarI-class.html#size,Variable Bio.Restriction.Restriction.NciI.size=Bio.Restriction.Restriction.NciI-class.html#size,Variable Bio.Restriction.Restriction.NcoI.size=Bio.Restriction.Restriction.NcoI-class.html#size,Variable Bio.Restriction.Restriction.NdeI.size=Bio.Restriction.Restriction.NdeI-class.html#size,Variable Bio.Restriction.Restriction.NdeII.size=Bio.Restriction.Restriction.NdeII-class.html#size,Variable Bio.Restriction.Restriction.NgoAVIII.size=Bio.Restriction.Restriction.NgoAVIII-class.html#size,Variable Bio.Restriction.Restriction.NgoMIV.size=Bio.Restriction.Restriction.NgoMIV-class.html#size,Variable Bio.Restriction.Restriction.NhaXI.size=Bio.Restriction.Restriction.NhaXI-class.html#size,Variable Bio.Restriction.Restriction.NheI.size=Bio.Restriction.Restriction.NheI-class.html#size,Variable Bio.Restriction.Restriction.NlaCI.size=Bio.Restriction.Restriction.NlaCI-class.html#size,Variable Bio.Restriction.Restriction.NlaIII.size=Bio.Restriction.Restriction.NlaIII-class.html#size,Variable Bio.Restriction.Restriction.NlaIV.size=Bio.Restriction.Restriction.NlaIV-class.html#size,Variable Bio.Restriction.Restriction.Nli3877I.size=Bio.Restriction.Restriction.Nli3877I-class.html#size,Variable Bio.Restriction.Restriction.NmeAIII.size=Bio.Restriction.Restriction.NmeAIII-class.html#size,Variable Bio.Restriction.Restriction.NmeDI.size=Bio.Restriction.Restriction.NmeDI-class.html#size,Variable Bio.Restriction.Restriction.NmuCI.size=Bio.Restriction.Restriction.NmuCI-class.html#size,Variable Bio.Restriction.Restriction.NotI.size=Bio.Restriction.Restriction.NotI-class.html#size,Variable Bio.Restriction.Restriction.NruI.size=Bio.Restriction.Restriction.NruI-class.html#size,Variable Bio.Restriction.Restriction.NsbI.size=Bio.Restriction.Restriction.NsbI-class.html#size,Variable Bio.Restriction.Restriction.NsiI.size=Bio.Restriction.Restriction.NsiI-class.html#size,Variable Bio.Restriction.Restriction.NspBII.size=Bio.Restriction.Restriction.NspBII-class.html#size,Variable Bio.Restriction.Restriction.NspI.size=Bio.Restriction.Restriction.NspI-class.html#size,Variable Bio.Restriction.Restriction.NspV.size=Bio.Restriction.Restriction.NspV-class.html#size,Variable Bio.Restriction.Restriction.OliI.size=Bio.Restriction.Restriction.OliI-class.html#size,Variable Bio.Restriction.Restriction.PabI.size=Bio.Restriction.Restriction.PabI-class.html#size,Variable Bio.Restriction.Restriction.PacI.size=Bio.Restriction.Restriction.PacI-class.html#size,Variable Bio.Restriction.Restriction.PaeI.size=Bio.Restriction.Restriction.PaeI-class.html#size,Variable Bio.Restriction.Restriction.PaeR7I.size=Bio.Restriction.Restriction.PaeR7I-class.html#size,Variable Bio.Restriction.Restriction.PagI.size=Bio.Restriction.Restriction.PagI-class.html#size,Variable Bio.Restriction.Restriction.PalAI.size=Bio.Restriction.Restriction.PalAI-class.html#size,Variable Bio.Restriction.Restriction.PasI.size=Bio.Restriction.Restriction.PasI-class.html#size,Variable Bio.Restriction.Restriction.PauI.size=Bio.Restriction.Restriction.PauI-class.html#size,Variable Bio.Restriction.Restriction.PceI.size=Bio.Restriction.Restriction.PceI-class.html#size,Variable Bio.Restriction.Restriction.PciI.size=Bio.Restriction.Restriction.PciI-class.html#size,Variable Bio.Restriction.Restriction.PciSI.size=Bio.Restriction.Restriction.PciSI-class.html#size,Variable Bio.Restriction.Restriction.PcsI.size=Bio.Restriction.Restriction.PcsI-class.html#size,Variable Bio.Restriction.Restriction.PctI.size=Bio.Restriction.Restriction.PctI-class.html#size,Variable Bio.Restriction.Restriction.PdiI.size=Bio.Restriction.Restriction.PdiI-class.html#size,Variable Bio.Restriction.Restriction.PdmI.size=Bio.Restriction.Restriction.PdmI-class.html#size,Variable Bio.Restriction.Restriction.PenI.size=Bio.Restriction.Restriction.PenI-class.html#size,Variable Bio.Restriction.Restriction.PfeI.size=Bio.Restriction.Restriction.PfeI-class.html#size,Variable Bio.Restriction.Restriction.Pfl1108I.size=Bio.Restriction.Restriction.Pfl1108I-class.html#size,Variable Bio.Restriction.Restriction.Pfl23II.size=Bio.Restriction.Restriction.Pfl23II-class.html#size,Variable Bio.Restriction.Restriction.PflFI.size=Bio.Restriction.Restriction.PflFI-class.html#size,Variable Bio.Restriction.Restriction.PflMI.size=Bio.Restriction.Restriction.PflMI-class.html#size,Variable Bio.Restriction.Restriction.PfoI.size=Bio.Restriction.Restriction.PfoI-class.html#size,Variable Bio.Restriction.Restriction.PinAI.size=Bio.Restriction.Restriction.PinAI-class.html#size,Variable Bio.Restriction.Restriction.PlaDI.size=Bio.Restriction.Restriction.PlaDI-class.html#size,Variable Bio.Restriction.Restriction.Ple19I.size=Bio.Restriction.Restriction.Ple19I-class.html#size,Variable Bio.Restriction.Restriction.PleI.size=Bio.Restriction.Restriction.PleI-class.html#size,Variable Bio.Restriction.Restriction.PluTI.size=Bio.Restriction.Restriction.PluTI-class.html#size,Variable Bio.Restriction.Restriction.PmaCI.size=Bio.Restriction.Restriction.PmaCI-class.html#size,Variable Bio.Restriction.Restriction.PmeI.size=Bio.Restriction.Restriction.PmeI-class.html#size,Variable Bio.Restriction.Restriction.PmlI.size=Bio.Restriction.Restriction.PmlI-class.html#size,Variable Bio.Restriction.Restriction.PpiI.size=Bio.Restriction.Restriction.PpiI-class.html#size,Variable Bio.Restriction.Restriction.PpsI.size=Bio.Restriction.Restriction.PpsI-class.html#size,Variable Bio.Restriction.Restriction.Ppu10I.size=Bio.Restriction.Restriction.Ppu10I-class.html#size,Variable Bio.Restriction.Restriction.Ppu21I.size=Bio.Restriction.Restriction.Ppu21I-class.html#size,Variable Bio.Restriction.Restriction.PpuMI.size=Bio.Restriction.Restriction.PpuMI-class.html#size,Variable Bio.Restriction.Restriction.PscI.size=Bio.Restriction.Restriction.PscI-class.html#size,Variable Bio.Restriction.Restriction.PshAI.size=Bio.Restriction.Restriction.PshAI-class.html#size,Variable Bio.Restriction.Restriction.PshBI.size=Bio.Restriction.Restriction.PshBI-class.html#size,Variable Bio.Restriction.Restriction.PsiI.size=Bio.Restriction.Restriction.PsiI-class.html#size,Variable Bio.Restriction.Restriction.Psp03I.size=Bio.Restriction.Restriction.Psp03I-class.html#size,Variable Bio.Restriction.Restriction.Psp124BI.size=Bio.Restriction.Restriction.Psp124BI-class.html#size,Variable Bio.Restriction.Restriction.Psp1406I.size=Bio.Restriction.Restriction.Psp1406I-class.html#size,Variable Bio.Restriction.Restriction.Psp5II.size=Bio.Restriction.Restriction.Psp5II-class.html#size,Variable Bio.Restriction.Restriction.Psp6I.size=Bio.Restriction.Restriction.Psp6I-class.html#size,Variable Bio.Restriction.Restriction.PspCI.size=Bio.Restriction.Restriction.PspCI-class.html#size,Variable Bio.Restriction.Restriction.PspEI.size=Bio.Restriction.Restriction.PspEI-class.html#size,Variable Bio.Restriction.Restriction.PspGI.size=Bio.Restriction.Restriction.PspGI-class.html#size,Variable Bio.Restriction.Restriction.PspLI.size=Bio.Restriction.Restriction.PspLI-class.html#size,Variable Bio.Restriction.Restriction.PspN4I.size=Bio.Restriction.Restriction.PspN4I-class.html#size,Variable Bio.Restriction.Restriction.PspOMI.size=Bio.Restriction.Restriction.PspOMI-class.html#size,Variable Bio.Restriction.Restriction.PspOMII.size=Bio.Restriction.Restriction.PspOMII-class.html#size,Variable Bio.Restriction.Restriction.PspPI.size=Bio.Restriction.Restriction.PspPI-class.html#size,Variable Bio.Restriction.Restriction.PspPPI.size=Bio.Restriction.Restriction.PspPPI-class.html#size,Variable Bio.Restriction.Restriction.PspPRI.size=Bio.Restriction.Restriction.PspPRI-class.html#size,Variable Bio.Restriction.Restriction.PspXI.size=Bio.Restriction.Restriction.PspXI-class.html#size,Variable Bio.Restriction.Restriction.PsrI.size=Bio.Restriction.Restriction.PsrI-class.html#size,Variable Bio.Restriction.Restriction.PssI.size=Bio.Restriction.Restriction.PssI-class.html#size,Variable Bio.Restriction.Restriction.PstI.size=Bio.Restriction.Restriction.PstI-class.html#size,Variable Bio.Restriction.Restriction.PstNI.size=Bio.Restriction.Restriction.PstNI-class.html#size,Variable Bio.Restriction.Restriction.PsuI.size=Bio.Restriction.Restriction.PsuI-class.html#size,Variable Bio.Restriction.Restriction.PsyI.size=Bio.Restriction.Restriction.PsyI-class.html#size,Variable Bio.Restriction.Restriction.PteI.size=Bio.Restriction.Restriction.PteI-class.html#size,Variable Bio.Restriction.Restriction.PvuI.size=Bio.Restriction.Restriction.PvuI-class.html#size,Variable Bio.Restriction.Restriction.PvuII.size=Bio.Restriction.Restriction.PvuII-class.html#size,Variable Bio.Restriction.Restriction.R2_BceSIV.size=Bio.Restriction.Restriction.R2_BceSIV-class.html#size,Variable Bio.Restriction.Restriction.RceI.size=Bio.Restriction.Restriction.RceI-class.html#size,Variable Bio.Restriction.Restriction.RdeGBI.size=Bio.Restriction.Restriction.RdeGBI-class.html#size,Variable Bio.Restriction.Restriction.RdeGBII.size=Bio.Restriction.Restriction.RdeGBII-class.html#size,Variable Bio.Restriction.Restriction.RdeGBIII.size=Bio.Restriction.Restriction.RdeGBIII-class.html#size,Variable Bio.Restriction.Restriction.RflFIII.size=Bio.Restriction.Restriction.RflFIII-class.html#size,Variable Bio.Restriction.Restriction.RgaI.size=Bio.Restriction.Restriction.RgaI-class.html#size,Variable Bio.Restriction.Restriction.RigI.size=Bio.Restriction.Restriction.RigI-class.html#size,Variable Bio.Restriction.Restriction.RlaI.size=Bio.Restriction.Restriction.RlaI-class.html#size,Variable Bio.Restriction.Restriction.RleAI.size=Bio.Restriction.Restriction.RleAI-class.html#size,Variable Bio.Restriction.Restriction.RpaB5I.size=Bio.Restriction.Restriction.RpaB5I-class.html#size,Variable Bio.Restriction.Restriction.RpaBI.size=Bio.Restriction.Restriction.RpaBI-class.html#size,Variable Bio.Restriction.Restriction.RpaI.size=Bio.Restriction.Restriction.RpaI-class.html#size,Variable Bio.Restriction.Restriction.RpaTI.size=Bio.Restriction.Restriction.RpaTI-class.html#size,Variable Bio.Restriction.Restriction.RruI.size=Bio.Restriction.Restriction.RruI-class.html#size,Variable Bio.Restriction.Restriction.RsaI.size=Bio.Restriction.Restriction.RsaI-class.html#size,Variable Bio.Restriction.Restriction.RsaNI.size=Bio.Restriction.Restriction.RsaNI-class.html#size,Variable Bio.Restriction.Restriction.RseI.size=Bio.Restriction.Restriction.RseI-class.html#size,Variable Bio.Restriction.Restriction.Rsr2I.size=Bio.Restriction.Restriction.Rsr2I-class.html#size,Variable Bio.Restriction.Restriction.RsrII.size=Bio.Restriction.Restriction.RsrII-class.html#size,Variable Bio.Restriction.Restriction.SacI.size=Bio.Restriction.Restriction.SacI-class.html#size,Variable Bio.Restriction.Restriction.SacII.size=Bio.Restriction.Restriction.SacII-class.html#size,Variable Bio.Restriction.Restriction.SalI.size=Bio.Restriction.Restriction.SalI-class.html#size,Variable Bio.Restriction.Restriction.SanDI.size=Bio.Restriction.Restriction.SanDI-class.html#size,Variable Bio.Restriction.Restriction.SapI.size=Bio.Restriction.Restriction.SapI-class.html#size,Variable Bio.Restriction.Restriction.SaqAI.size=Bio.Restriction.Restriction.SaqAI-class.html#size,Variable Bio.Restriction.Restriction.SatI.size=Bio.Restriction.Restriction.SatI-class.html#size,Variable Bio.Restriction.Restriction.Sau3AI.size=Bio.Restriction.Restriction.Sau3AI-class.html#size,Variable Bio.Restriction.Restriction.Sau96I.size=Bio.Restriction.Restriction.Sau96I-class.html#size,Variable Bio.Restriction.Restriction.SauI.size=Bio.Restriction.Restriction.SauI-class.html#size,Variable Bio.Restriction.Restriction.SbfI.size=Bio.Restriction.Restriction.SbfI-class.html#size,Variable Bio.Restriction.Restriction.ScaI.size=Bio.Restriction.Restriction.ScaI-class.html#size,Variable Bio.Restriction.Restriction.SchI.size=Bio.Restriction.Restriction.SchI-class.html#size,Variable Bio.Restriction.Restriction.SciI.size=Bio.Restriction.Restriction.SciI-class.html#size,Variable Bio.Restriction.Restriction.ScrFI.size=Bio.Restriction.Restriction.ScrFI-class.html#size,Variable Bio.Restriction.Restriction.SdaI.size=Bio.Restriction.Restriction.SdaI-class.html#size,Variable Bio.Restriction.Restriction.SdeAI.size=Bio.Restriction.Restriction.SdeAI-class.html#size,Variable Bio.Restriction.Restriction.SdeOSI.size=Bio.Restriction.Restriction.SdeOSI-class.html#size,Variable Bio.Restriction.Restriction.SduI.size=Bio.Restriction.Restriction.SduI-class.html#size,Variable Bio.Restriction.Restriction.SecI.size=Bio.Restriction.Restriction.SecI-class.html#size,Variable Bio.Restriction.Restriction.SelI.size=Bio.Restriction.Restriction.SelI-class.html#size,Variable Bio.Restriction.Restriction.SetI.size=Bio.Restriction.Restriction.SetI-class.html#size,Variable Bio.Restriction.Restriction.SexAI.size=Bio.Restriction.Restriction.SexAI-class.html#size,Variable Bio.Restriction.Restriction.SfaAI.size=Bio.Restriction.Restriction.SfaAI-class.html#size,Variable Bio.Restriction.Restriction.SfaNI.size=Bio.Restriction.Restriction.SfaNI-class.html#size,Variable Bio.Restriction.Restriction.SfcI.size=Bio.Restriction.Restriction.SfcI-class.html#size,Variable Bio.Restriction.Restriction.SfeI.size=Bio.Restriction.Restriction.SfeI-class.html#size,Variable Bio.Restriction.Restriction.SfiI.size=Bio.Restriction.Restriction.SfiI-class.html#size,Variable Bio.Restriction.Restriction.SfoI.size=Bio.Restriction.Restriction.SfoI-class.html#size,Variable Bio.Restriction.Restriction.Sfr274I.size=Bio.Restriction.Restriction.Sfr274I-class.html#size,Variable Bio.Restriction.Restriction.Sfr303I.size=Bio.Restriction.Restriction.Sfr303I-class.html#size,Variable Bio.Restriction.Restriction.SfuI.size=Bio.Restriction.Restriction.SfuI-class.html#size,Variable Bio.Restriction.Restriction.SgeI.size=Bio.Restriction.Restriction.SgeI-class.html#size,Variable Bio.Restriction.Restriction.SgfI.size=Bio.Restriction.Restriction.SgfI-class.html#size,Variable Bio.Restriction.Restriction.SgrAI.size=Bio.Restriction.Restriction.SgrAI-class.html#size,Variable Bio.Restriction.Restriction.SgrBI.size=Bio.Restriction.Restriction.SgrBI-class.html#size,Variable Bio.Restriction.Restriction.SgrDI.size=Bio.Restriction.Restriction.SgrDI-class.html#size,Variable Bio.Restriction.Restriction.SgrTI.size=Bio.Restriction.Restriction.SgrTI-class.html#size,Variable Bio.Restriction.Restriction.SgsI.size=Bio.Restriction.Restriction.SgsI-class.html#size,Variable Bio.Restriction.Restriction.SimI.size=Bio.Restriction.Restriction.SimI-class.html#size,Variable Bio.Restriction.Restriction.SlaI.size=Bio.Restriction.Restriction.SlaI-class.html#size,Variable Bio.Restriction.Restriction.SmaI.size=Bio.Restriction.Restriction.SmaI-class.html#size,Variable Bio.Restriction.Restriction.SmiI.size=Bio.Restriction.Restriction.SmiI-class.html#size,Variable Bio.Restriction.Restriction.SmiMI.size=Bio.Restriction.Restriction.SmiMI-class.html#size,Variable Bio.Restriction.Restriction.SmlI.size=Bio.Restriction.Restriction.SmlI-class.html#size,Variable Bio.Restriction.Restriction.SmoI.size=Bio.Restriction.Restriction.SmoI-class.html#size,Variable Bio.Restriction.Restriction.SnaBI.size=Bio.Restriction.Restriction.SnaBI-class.html#size,Variable Bio.Restriction.Restriction.SnaI.size=Bio.Restriction.Restriction.SnaI-class.html#size,Variable Bio.Restriction.Restriction.Sno506I.size=Bio.Restriction.Restriction.Sno506I-class.html#size,Variable Bio.Restriction.Restriction.SpeI.size=Bio.Restriction.Restriction.SpeI-class.html#size,Variable Bio.Restriction.Restriction.SphI.size=Bio.Restriction.Restriction.SphI-class.html#size,Variable Bio.Restriction.Restriction.SplI.size=Bio.Restriction.Restriction.SplI-class.html#size,Variable Bio.Restriction.Restriction.SpoDI.size=Bio.Restriction.Restriction.SpoDI-class.html#size,Variable Bio.Restriction.Restriction.SrfI.size=Bio.Restriction.Restriction.SrfI-class.html#size,Variable Bio.Restriction.Restriction.Sse232I.size=Bio.Restriction.Restriction.Sse232I-class.html#size,Variable Bio.Restriction.Restriction.Sse8387I.size=Bio.Restriction.Restriction.Sse8387I-class.html#size,Variable Bio.Restriction.Restriction.Sse8647I.size=Bio.Restriction.Restriction.Sse8647I-class.html#size,Variable Bio.Restriction.Restriction.Sse9I.size=Bio.Restriction.Restriction.Sse9I-class.html#size,Variable Bio.Restriction.Restriction.SseBI.size=Bio.Restriction.Restriction.SseBI-class.html#size,Variable Bio.Restriction.Restriction.SsiI.size=Bio.Restriction.Restriction.SsiI-class.html#size,Variable Bio.Restriction.Restriction.SspD5I.size=Bio.Restriction.Restriction.SspD5I-class.html#size,Variable Bio.Restriction.Restriction.SspDI.size=Bio.Restriction.Restriction.SspDI-class.html#size,Variable Bio.Restriction.Restriction.SspI.size=Bio.Restriction.Restriction.SspI-class.html#size,Variable Bio.Restriction.Restriction.SstE37I.size=Bio.Restriction.Restriction.SstE37I-class.html#size,Variable Bio.Restriction.Restriction.SstI.size=Bio.Restriction.Restriction.SstI-class.html#size,Variable Bio.Restriction.Restriction.Sth132I.size=Bio.Restriction.Restriction.Sth132I-class.html#size,Variable Bio.Restriction.Restriction.Sth302II.size=Bio.Restriction.Restriction.Sth302II-class.html#size,Variable Bio.Restriction.Restriction.StrI.size=Bio.Restriction.Restriction.StrI-class.html#size,Variable Bio.Restriction.Restriction.StsI.size=Bio.Restriction.Restriction.StsI-class.html#size,Variable Bio.Restriction.Restriction.StuI.size=Bio.Restriction.Restriction.StuI-class.html#size,Variable Bio.Restriction.Restriction.StyD4I.size=Bio.Restriction.Restriction.StyD4I-class.html#size,Variable Bio.Restriction.Restriction.StyI.size=Bio.Restriction.Restriction.StyI-class.html#size,Variable Bio.Restriction.Restriction.SwaI.size=Bio.Restriction.Restriction.SwaI-class.html#size,Variable Bio.Restriction.Restriction.TaaI.size=Bio.Restriction.Restriction.TaaI-class.html#size,Variable Bio.Restriction.Restriction.TaiI.size=Bio.Restriction.Restriction.TaiI-class.html#size,Variable Bio.Restriction.Restriction.TaqI.size=Bio.Restriction.Restriction.TaqI-class.html#size,Variable Bio.Restriction.Restriction.TaqII.size=Bio.Restriction.Restriction.TaqII-class.html#size,Variable Bio.Restriction.Restriction.TasI.size=Bio.Restriction.Restriction.TasI-class.html#size,Variable Bio.Restriction.Restriction.TatI.size=Bio.Restriction.Restriction.TatI-class.html#size,Variable Bio.Restriction.Restriction.TauI.size=Bio.Restriction.Restriction.TauI-class.html#size,Variable Bio.Restriction.Restriction.TfiI.size=Bio.Restriction.Restriction.TfiI-class.html#size,Variable Bio.Restriction.Restriction.Tru1I.size=Bio.Restriction.Restriction.Tru1I-class.html#size,Variable Bio.Restriction.Restriction.Tru9I.size=Bio.Restriction.Restriction.Tru9I-class.html#size,Variable Bio.Restriction.Restriction.TscAI.size=Bio.Restriction.Restriction.TscAI-class.html#size,Variable Bio.Restriction.Restriction.TseFI.size=Bio.Restriction.Restriction.TseFI-class.html#size,Variable Bio.Restriction.Restriction.TseI.size=Bio.Restriction.Restriction.TseI-class.html#size,Variable Bio.Restriction.Restriction.TsoI.size=Bio.Restriction.Restriction.TsoI-class.html#size,Variable Bio.Restriction.Restriction.Tsp45I.size=Bio.Restriction.Restriction.Tsp45I-class.html#size,Variable Bio.Restriction.Restriction.Tsp4CI.size=Bio.Restriction.Restriction.Tsp4CI-class.html#size,Variable Bio.Restriction.Restriction.TspDTI.size=Bio.Restriction.Restriction.TspDTI-class.html#size,Variable Bio.Restriction.Restriction.TspEI.size=Bio.Restriction.Restriction.TspEI-class.html#size,Variable Bio.Restriction.Restriction.TspGWI.size=Bio.Restriction.Restriction.TspGWI-class.html#size,Variable Bio.Restriction.Restriction.TspMI.size=Bio.Restriction.Restriction.TspMI-class.html#size,Variable Bio.Restriction.Restriction.TspRI.size=Bio.Restriction.Restriction.TspRI-class.html#size,Variable Bio.Restriction.Restriction.TssI.size=Bio.Restriction.Restriction.TssI-class.html#size,Variable Bio.Restriction.Restriction.TstI.size=Bio.Restriction.Restriction.TstI-class.html#size,Variable Bio.Restriction.Restriction.TsuI.size=Bio.Restriction.Restriction.TsuI-class.html#size,Variable Bio.Restriction.Restriction.Tth111I.size=Bio.Restriction.Restriction.Tth111I-class.html#size,Variable Bio.Restriction.Restriction.Tth111II.size=Bio.Restriction.Restriction.Tth111II-class.html#size,Variable Bio.Restriction.Restriction.UbaF11I.size=Bio.Restriction.Restriction.UbaF11I-class.html#size,Variable Bio.Restriction.Restriction.UbaF12I.size=Bio.Restriction.Restriction.UbaF12I-class.html#size,Variable Bio.Restriction.Restriction.UbaF13I.size=Bio.Restriction.Restriction.UbaF13I-class.html#size,Variable Bio.Restriction.Restriction.UbaF14I.size=Bio.Restriction.Restriction.UbaF14I-class.html#size,Variable Bio.Restriction.Restriction.UbaF9I.size=Bio.Restriction.Restriction.UbaF9I-class.html#size,Variable Bio.Restriction.Restriction.UbaPI.size=Bio.Restriction.Restriction.UbaPI-class.html#size,Variable Bio.Restriction.Restriction.UcoMSI.size=Bio.Restriction.Restriction.UcoMSI-class.html#size,Variable Bio.Restriction.Restriction.UnbI.size=Bio.Restriction.Restriction.UnbI-class.html#size,Variable Bio.Restriction.Restriction.Van91I.size=Bio.Restriction.Restriction.Van91I-class.html#size,Variable Bio.Restriction.Restriction.Vha464I.size=Bio.Restriction.Restriction.Vha464I-class.html#size,Variable Bio.Restriction.Restriction.VneI.size=Bio.Restriction.Restriction.VneI-class.html#size,Variable Bio.Restriction.Restriction.VpaK11AI.size=Bio.Restriction.Restriction.VpaK11AI-class.html#size,Variable Bio.Restriction.Restriction.VpaK11BI.size=Bio.Restriction.Restriction.VpaK11BI-class.html#size,Variable Bio.Restriction.Restriction.VspI.size=Bio.Restriction.Restriction.VspI-class.html#size,Variable Bio.Restriction.Restriction.WviI.size=Bio.Restriction.Restriction.WviI-class.html#size,Variable Bio.Restriction.Restriction.XagI.size=Bio.Restriction.Restriction.XagI-class.html#size,Variable Bio.Restriction.Restriction.XapI.size=Bio.Restriction.Restriction.XapI-class.html#size,Variable Bio.Restriction.Restriction.XbaI.size=Bio.Restriction.Restriction.XbaI-class.html#size,Variable Bio.Restriction.Restriction.XceI.size=Bio.Restriction.Restriction.XceI-class.html#size,Variable Bio.Restriction.Restriction.XcmI.size=Bio.Restriction.Restriction.XcmI-class.html#size,Variable Bio.Restriction.Restriction.XhoI.size=Bio.Restriction.Restriction.XhoI-class.html#size,Variable Bio.Restriction.Restriction.XhoII.size=Bio.Restriction.Restriction.XhoII-class.html#size,Variable Bio.Restriction.Restriction.XmaI.size=Bio.Restriction.Restriction.XmaI-class.html#size,Variable Bio.Restriction.Restriction.XmaIII.size=Bio.Restriction.Restriction.XmaIII-class.html#size,Variable Bio.Restriction.Restriction.XmaJI.size=Bio.Restriction.Restriction.XmaJI-class.html#size,Variable Bio.Restriction.Restriction.XmiI.size=Bio.Restriction.Restriction.XmiI-class.html#size,Variable Bio.Restriction.Restriction.XmnI.size=Bio.Restriction.Restriction.XmnI-class.html#size,Variable Bio.Restriction.Restriction.XspI.size=Bio.Restriction.Restriction.XspI-class.html#size,Variable Bio.Restriction.Restriction.YkrI.size=Bio.Restriction.Restriction.YkrI-class.html#size,Variable Bio.Restriction.Restriction.ZraI.size=Bio.Restriction.Restriction.ZraI-class.html#size,Variable Bio.Restriction.Restriction.ZrmI.size=Bio.Restriction.Restriction.ZrmI-class.html#size,Variable Bio.Restriction.Restriction.Zsp2I.size=Bio.Restriction.Restriction.Zsp2I-class.html#size,Variable Bio.codonalign.codonalphabet.CodonAlphabet.size=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#size"><a title="Bio.Alphabet.Alphabet.size
Bio.Alphabet.Reduced.HPModel.size
Bio.Alphabet.Reduced.Murphy10.size
Bio.Alphabet.Reduced.Murphy15.size
Bio.Alphabet.Reduced.Murphy4.size
Bio.Alphabet.Reduced.Murphy8.size
Bio.Alphabet.Reduced.PC5.size
Bio.Alphabet.SingleLetterAlphabet.size
Bio.Alphabet.ThreeLetterProtein.size
Bio.GenBank._FeatureConsumer.size
Bio.GenBank._RecordConsumer.size
Bio.Restriction.Restriction.AanI.size
Bio.Restriction.Restriction.AarI.size
Bio.Restriction.Restriction.AasI.size
Bio.Restriction.Restriction.AatII.size
Bio.Restriction.Restriction.AbaSI.size
Bio.Restriction.Restriction.AbsI.size
Bio.Restriction.Restriction.Acc16I.size
Bio.Restriction.Restriction.Acc36I.size
Bio.Restriction.Restriction.Acc65I.size
Bio.Restriction.Restriction.AccB1I.size
Bio.Restriction.Restriction.AccB7I.size
Bio.Restriction.Restriction.AccBSI.size
Bio.Restriction.Restriction.AccI.size
Bio.Restriction.Restriction.AccII.size
Bio.Restriction.Restriction.AccIII.size
Bio.Restriction.Restriction.AceIII.size
Bio.Restriction.Restriction.AciI.size
Bio.Restriction.Restriction.AclI.size
Bio.Restriction.Restriction.AclWI.size
Bio.Restriction.Restriction.AcoI.size
Bio.Restriction.Restriction.AcsI.size
Bio.Restriction.Restriction.AcuI.size
Bio.Restriction.Restriction.AcvI.size
Bio.Restriction.Restriction.AcyI.size
Bio.Restriction.Restriction.AdeI.size
Bio.Restriction.Restriction.AfaI.size
Bio.Restriction.Restriction.AfeI.size
Bio.Restriction.Restriction.AfiI.size
Bio.Restriction.Restriction.AflII.size
Bio.Restriction.Restriction.AflIII.size
Bio.Restriction.Restriction.AgeI.size
Bio.Restriction.Restriction.AgsI.size
Bio.Restriction.Restriction.AhaIII.size
Bio.Restriction.Restriction.AhdI.size
Bio.Restriction.Restriction.AhlI.size
Bio.Restriction.Restriction.AjiI.size
Bio.Restriction.Restriction.AjnI.size
Bio.Restriction.Restriction.AjuI.size
Bio.Restriction.Restriction.AleI.size
Bio.Restriction.Restriction.AlfI.size
Bio.Restriction.Restriction.AloI.size
Bio.Restriction.Restriction.AluBI.size
Bio.Restriction.Restriction.AluI.size
Bio.Restriction.Restriction.Alw21I.size
Bio.Restriction.Restriction.Alw26I.size
Bio.Restriction.Restriction.Alw44I.size
Bio.Restriction.Restriction.AlwFI.size
Bio.Restriction.Restriction.AlwI.size
Bio.Restriction.Restriction.AlwNI.size
Bio.Restriction.Restriction.Ama87I.size
Bio.Restriction.Restriction.Aor13HI.size
Bio.Restriction.Restriction.Aor51HI.size
Bio.Restriction.Restriction.AoxI.size
Bio.Restriction.Restriction.ApaBI.size
Bio.Restriction.Restriction.ApaI.size
Bio.Restriction.Restriction.ApaLI.size
Bio.Restriction.Restriction.ApeKI.size
Bio.Restriction.Restriction.ApoI.size
Bio.Restriction.Restriction.ApyPI.size
Bio.Restriction.Restriction.AquII.size
Bio.Restriction.Restriction.AquIII.size
Bio.Restriction.Restriction.AquIV.size
Bio.Restriction.Restriction.ArsI.size
Bio.Restriction.Restriction.AscI.size
Bio.Restriction.Restriction.AseI.size
Bio.Restriction.Restriction.Asi256I.size
Bio.Restriction.Restriction.AsiGI.size
Bio.Restriction.Restriction.AsiSI.size
Bio.Restriction.Restriction.Asp700I.size
Bio.Restriction.Restriction.Asp718I.size
Bio.Restriction.Restriction.AspA2I.size
Bio.Restriction.Restriction.AspBHI.size
Bio.Restriction.Restriction.AspLEI.size
Bio.Restriction.Restriction.AspS9I.size
Bio.Restriction.Restriction.AssI.size
Bio.Restriction.Restriction.AsuC2I.size
Bio.Restriction.Restriction.AsuHPI.size
Bio.Restriction.Restriction.AsuI.size
Bio.Restriction.Restriction.AsuII.size
Bio.Restriction.Restriction.AsuNHI.size
Bio.Restriction.Restriction.AvaI.size
Bio.Restriction.Restriction.AvaII.size
Bio.Restriction.Restriction.AvaIII.size
Bio.Restriction.Restriction.AvrII.size
Bio.Restriction.Restriction.AxyI.size
Bio.Restriction.Restriction.BaeGI.size
Bio.Restriction.Restriction.BaeI.size
Bio.Restriction.Restriction.BalI.size
Bio.Restriction.Restriction.BamHI.size
Bio.Restriction.Restriction.BanI.size
Bio.Restriction.Restriction.BanII.size
Bio.Restriction.Restriction.BarI.size
Bio.Restriction.Restriction.BasI.size
Bio.Restriction.Restriction.BauI.size
Bio.Restriction.Restriction.Bbr7I.size
Bio.Restriction.Restriction.BbrPI.size
Bio.Restriction.Restriction.BbsI.size
Bio.Restriction.Restriction.Bbv12I.size
Bio.Restriction.Restriction.BbvCI.size
Bio.Restriction.Restriction.BbvI.size
Bio.Restriction.Restriction.BbvII.size
Bio.Restriction.Restriction.BccI.size
Bio.Restriction.Restriction.Bce83I.size
Bio.Restriction.Restriction.BceAI.size
Bio.Restriction.Restriction.BcefI.size
Bio.Restriction.Restriction.BcgI.size
Bio.Restriction.Restriction.BciT130I.size
Bio.Restriction.Restriction.BciVI.size
Bio.Restriction.Restriction.BclI.size
Bio.Restriction.Restriction.BcnI.size
Bio.Restriction.Restriction.BcoDI.size
Bio.Restriction.Restriction.BcuI.size
Bio.Restriction.Restriction.BdaI.size
Bio.Restriction.Restriction.BetI.size
Bio.Restriction.Restriction.BfaI.size
Bio.Restriction.Restriction.BfiI.size
Bio.Restriction.Restriction.BfmI.size
Bio.Restriction.Restriction.BfoI.size
Bio.Restriction.Restriction.BfrI.size
Bio.Restriction.Restriction.BfuAI.size
Bio.Restriction.Restriction.BfuCI.size
Bio.Restriction.Restriction.BfuI.size
Bio.Restriction.Restriction.BglI.size
Bio.Restriction.Restriction.BglII.size
Bio.Restriction.Restriction.BinI.size
Bio.Restriction.Restriction.BisI.size
Bio.Restriction.Restriction.BlnI.size
Bio.Restriction.Restriction.BlpI.size
Bio.Restriction.Restriction.BlsI.size
Bio.Restriction.Restriction.BmcAI.size
Bio.Restriction.Restriction.Bme1390I.size
Bio.Restriction.Restriction.Bme18I.size
Bio.Restriction.Restriction.BmeDI.size
Bio.Restriction.Restriction.BmeRI.size
Bio.Restriction.Restriction.BmeT110I.size
Bio.Restriction.Restriction.BmgBI.size
Bio.Restriction.Restriction.BmgI.size
Bio.Restriction.Restriction.BmgT120I.size
Bio.Restriction.Restriction.BmiI.size
Bio.Restriction.Restriction.BmrFI.size
Bio.Restriction.Restriction.BmrI.size
Bio.Restriction.Restriction.BmsI.size
Bio.Restriction.Restriction.BmtI.size
Bio.Restriction.Restriction.BmuI.size
Bio.Restriction.Restriction.BoxI.size
Bio.Restriction.Restriction.BpiI.size
Bio.Restriction.Restriction.BplI.size
Bio.Restriction.Restriction.BpmI.size
Bio.Restriction.Restriction.Bpu10I.size
Bio.Restriction.Restriction.Bpu1102I.size
Bio.Restriction.Restriction.Bpu14I.size
Bio.Restriction.Restriction.BpuEI.size
Bio.Restriction.Restriction.BpuMI.size
Bio.Restriction.Restriction.BpvUI.size
Bio.Restriction.Restriction.Bsa29I.size
Bio.Restriction.Restriction.BsaAI.size
Bio.Restriction.Restriction.BsaBI.size
Bio.Restriction.Restriction.BsaHI.size
Bio.Restriction.Restriction.BsaI.size
Bio.Restriction.Restriction.BsaJI.size
Bio.Restriction.Restriction.BsaWI.size
Bio.Restriction.Restriction.BsaXI.size
Bio.Restriction.Restriction.BsbI.size
Bio.Restriction.Restriction.Bsc4I.size
Bio.Restriction.Restriction.BscAI.size
Bio.Restriction.Restriction.BscGI.size
Bio.Restriction.Restriction.Bse118I.size
Bio.Restriction.Restriction.Bse1I.size
Bio.Restriction.Restriction.Bse21I.size
Bio.Restriction.Restriction.Bse3DI.size
Bio.Restriction.Restriction.Bse8I.size
Bio.Restriction.Restriction.BseAI.size
Bio.Restriction.Restriction.BseBI.size
Bio.Restriction.Restriction.BseCI.size
Bio.Restriction.Restriction.BseDI.size
Bio.Restriction.Restriction.BseGI.size
Bio.Restriction.Restriction.BseJI.size
Bio.Restriction.Restriction.BseLI.size
Bio.Restriction.Restriction.BseMI.size
Bio.Restriction.Restriction.BseMII.size
Bio.Restriction.Restriction.BseNI.size
Bio.Restriction.Restriction.BsePI.size
Bio.Restriction.Restriction.BseRI.size
Bio.Restriction.Restriction.BseSI.size
Bio.Restriction.Restriction.BseX3I.size
Bio.Restriction.Restriction.BseXI.size
Bio.Restriction.Restriction.BseYI.size
Bio.Restriction.Restriction.BsgI.size
Bio.Restriction.Restriction.Bsh1236I.size
Bio.Restriction.Restriction.Bsh1285I.size
Bio.Restriction.Restriction.BshFI.size
Bio.Restriction.Restriction.BshNI.size
Bio.Restriction.Restriction.BshTI.size
Bio.Restriction.Restriction.BshVI.size
Bio.Restriction.Restriction.BsiEI.size
Bio.Restriction.Restriction.BsiHKAI.size
Bio.Restriction.Restriction.BsiHKCI.size
Bio.Restriction.Restriction.BsiI.size
Bio.Restriction.Restriction.BsiSI.size
Bio.Restriction.Restriction.BsiWI.size
Bio.Restriction.Restriction.BsiYI.size
Bio.Restriction.Restriction.BslFI.size
Bio.Restriction.Restriction.BslI.size
Bio.Restriction.Restriction.BsmAI.size
Bio.Restriction.Restriction.BsmBI.size
Bio.Restriction.Restriction.BsmFI.size
Bio.Restriction.Restriction.BsmI.size
Bio.Restriction.Restriction.BsnI.size
Bio.Restriction.Restriction.Bso31I.size
Bio.Restriction.Restriction.BsoBI.size
Bio.Restriction.Restriction.Bsp119I.size
Bio.Restriction.Restriction.Bsp120I.size
Bio.Restriction.Restriction.Bsp1286I.size
Bio.Restriction.Restriction.Bsp13I.size
Bio.Restriction.Restriction.Bsp1407I.size
Bio.Restriction.Restriction.Bsp143I.size
Bio.Restriction.Restriction.Bsp1720I.size
Bio.Restriction.Restriction.Bsp19I.size
Bio.Restriction.Restriction.Bsp24I.size
Bio.Restriction.Restriction.Bsp68I.size
Bio.Restriction.Restriction.BspACI.size
Bio.Restriction.Restriction.BspCNI.size
Bio.Restriction.Restriction.BspD6I.size
Bio.Restriction.Restriction.BspDI.size
Bio.Restriction.Restriction.BspEI.size
Bio.Restriction.Restriction.BspFNI.size
Bio.Restriction.Restriction.BspGI.size
Bio.Restriction.Restriction.BspHI.size
Bio.Restriction.Restriction.BspLI.size
Bio.Restriction.Restriction.BspLU11I.size
Bio.Restriction.Restriction.BspMI.size
Bio.Restriction.Restriction.BspMII.size
Bio.Restriction.Restriction.BspNCI.size
Bio.Restriction.Restriction.BspOI.size
Bio.Restriction.Restriction.BspPI.size
Bio.Restriction.Restriction.BspQI.size
Bio.Restriction.Restriction.BspT104I.size
Bio.Restriction.Restriction.BspT107I.size
Bio.Restriction.Restriction.BspTI.size
Bio.Restriction.Restriction.BsrBI.size
Bio.Restriction.Restriction.BsrDI.size
Bio.Restriction.Restriction.BsrFI.size
Bio.Restriction.Restriction.BsrGI.size
Bio.Restriction.Restriction.BsrI.size
Bio.Restriction.Restriction.BsrSI.size
Bio.Restriction.Restriction.BssAI.size
Bio.Restriction.Restriction.BssECI.size
Bio.Restriction.Restriction.BssHII.size
Bio.Restriction.Restriction.BssKI.size
Bio.Restriction.Restriction.BssMI.size
Bio.Restriction.Restriction.BssNAI.size
Bio.Restriction.Restriction.BssNI.size
Bio.Restriction.Restriction.BssSI.size
Bio.Restriction.Restriction.BssT1I.size
Bio.Restriction.Restriction.Bst1107I.size
Bio.Restriction.Restriction.Bst2BI.size
Bio.Restriction.Restriction.Bst2UI.size
Bio.Restriction.Restriction.Bst4CI.size
Bio.Restriction.Restriction.Bst6I.size
Bio.Restriction.Restriction.BstACI.size
Bio.Restriction.Restriction.BstAFI.size
Bio.Restriction.Restriction.BstAPI.size
Bio.Restriction.Restriction.BstAUI.size
Bio.Restriction.Restriction.BstBAI.size
Bio.Restriction.Restriction.BstBI.size
Bio.Restriction.Restriction.BstC8I.size
Bio.Restriction.Restriction.BstDEI.size
Bio.Restriction.Restriction.BstDSI.size
Bio.Restriction.Restriction.BstEII.size
Bio.Restriction.Restriction.BstENI.size
Bio.Restriction.Restriction.BstF5I.size
Bio.Restriction.Restriction.BstFNI.size
Bio.Restriction.Restriction.BstH2I.size
Bio.Restriction.Restriction.BstHHI.size
Bio.Restriction.Restriction.BstKTI.size
Bio.Restriction.Restriction.BstMAI.size
Bio.Restriction.Restriction.BstMBI.size
Bio.Restriction.Restriction.BstMCI.size
Bio.Restriction.Restriction.BstMWI.size
Bio.Restriction.Restriction.BstNI.size
Bio.Restriction.Restriction.BstNSI.size
Bio.Restriction.Restriction.BstOI.size
Bio.Restriction.Restriction.BstPAI.size
Bio.Restriction.Restriction.BstPI.size
Bio.Restriction.Restriction.BstSCI.size
Bio.Restriction.Restriction.BstSFI.size
Bio.Restriction.Restriction.BstSLI.size
Bio.Restriction.Restriction.BstSNI.size
Bio.Restriction.Restriction.BstUI.size
Bio.Restriction.Restriction.BstV1I.size
Bio.Restriction.Restriction.BstV2I.size
Bio.Restriction.Restriction.BstX2I.size
Bio.Restriction.Restriction.BstXI.size
Bio.Restriction.Restriction.BstYI.size
Bio.Restriction.Restriction.BstZ17I.size
Bio.Restriction.Restriction.BstZI.size
Bio.Restriction.Restriction.Bsu15I.size
Bio.Restriction.Restriction.Bsu36I.size
Bio.Restriction.Restriction.BsuI.size
Bio.Restriction.Restriction.BsuRI.size
Bio.Restriction.Restriction.BtgI.size
Bio.Restriction.Restriction.BtgZI.size
Bio.Restriction.Restriction.BthCI.size
Bio.Restriction.Restriction.BtrI.size
Bio.Restriction.Restriction.BtsCI.size
Bio.Restriction.Restriction.BtsI.size
Bio.Restriction.Restriction.BtsIMutI.size
Bio.Restriction.Restriction.BtuMI.size
Bio.Restriction.Restriction.BveI.size
Bio.Restriction.Restriction.Cac8I.size
Bio.Restriction.Restriction.CaiI.size
Bio.Restriction.Restriction.CauII.size
Bio.Restriction.Restriction.CchII.size
Bio.Restriction.Restriction.CchIII.size
Bio.Restriction.Restriction.CciI.size
Bio.Restriction.Restriction.CciNI.size
Bio.Restriction.Restriction.Cdi630V.size
Bio.Restriction.Restriction.CdiI.size
Bio.Restriction.Restriction.CdpI.size
Bio.Restriction.Restriction.CfoI.size
Bio.Restriction.Restriction.Cfr10I.size
Bio.Restriction.Restriction.Cfr13I.size
Bio.Restriction.Restriction.Cfr42I.size
Bio.Restriction.Restriction.Cfr9I.size
Bio.Restriction.Restriction.CfrI.size
Bio.Restriction.Restriction.Cgl13032I.size
Bio.Restriction.Restriction.Cgl13032II.size
Bio.Restriction.Restriction.ChaI.size
Bio.Restriction.Restriction.CjeFIII.size
Bio.Restriction.Restriction.CjeFV.size
Bio.Restriction.Restriction.CjeI.size
Bio.Restriction.Restriction.CjeNII.size
Bio.Restriction.Restriction.CjeNIII.size
Bio.Restriction.Restriction.CjeP659IV.size
Bio.Restriction.Restriction.CjePI.size
Bio.Restriction.Restriction.CjuI.size
Bio.Restriction.Restriction.CjuII.size
Bio.Restriction.Restriction.ClaI.size
Bio.Restriction.Restriction.CpoI.size
Bio.Restriction.Restriction.CseI.size
Bio.Restriction.Restriction.CsiI.size
Bio.Restriction.Restriction.Csp6I.size
Bio.Restriction.Restriction.CspAI.size
Bio.Restriction.Restriction.CspCI.size
Bio.Restriction.Restriction.CspI.size
Bio.Restriction.Restriction.CstMI.size
Bio.Restriction.Restriction.CviAII.size
Bio.Restriction.Restriction.CviJI.size
Bio.Restriction.Restriction.CviKI_1.size
Bio.Restriction.Restriction.CviQI.size
Bio.Restriction.Restriction.CviRI.size
Bio.Restriction.Restriction.DdeI.size
Bio.Restriction.Restriction.DinI.size
Bio.Restriction.Restriction.DpnI.size
Bio.Restriction.Restriction.DpnII.size
Bio.Restriction.Restriction.DraI.size
Bio.Restriction.Restriction.DraII.size
Bio.Restriction.Restriction.DraIII.size
Bio.Restriction.Restriction.DraRI.size
Bio.Restriction.Restriction.DrdI.size
Bio.Restriction.Restriction.DrdII.size
Bio.Restriction.Restriction.DriI.size
Bio.Restriction.Restriction.DsaI.size
Bio.Restriction.Restriction.DseDI.size
Bio.Restriction.Restriction.EaeI.size
Bio.Restriction.Restriction.EagI.size
Bio.Restriction.Restriction.Eam1104I.size
Bio.Restriction.Restriction.Eam1105I.size
Bio.Restriction.Restriction.EarI.size
Bio.Restriction.Restriction.EciI.size
Bio.Restriction.Restriction.Ecl136II.size
Bio.Restriction.Restriction.EclXI.size
Bio.Restriction.Restriction.Eco105I.size
Bio.Restriction.Restriction.Eco130I.size
Bio.Restriction.Restriction.Eco147I.size
Bio.Restriction.Restriction.Eco24I.size
Bio.Restriction.Restriction.Eco31I.size
Bio.Restriction.Restriction.Eco32I.size
Bio.Restriction.Restriction.Eco47I.size
Bio.Restriction.Restriction.Eco47III.size
Bio.Restriction.Restriction.Eco52I.size
Bio.Restriction.Restriction.Eco53kI.size
Bio.Restriction.Restriction.Eco57I.size
Bio.Restriction.Restriction.Eco57MI.size
Bio.Restriction.Restriction.Eco72I.size
Bio.Restriction.Restriction.Eco81I.size
Bio.Restriction.Restriction.Eco88I.size
Bio.Restriction.Restriction.Eco91I.size
Bio.Restriction.Restriction.EcoHI.size
Bio.Restriction.Restriction.EcoICRI.size
Bio.Restriction.Restriction.EcoNI.size
Bio.Restriction.Restriction.EcoO109I.size
Bio.Restriction.Restriction.EcoO65I.size
Bio.Restriction.Restriction.EcoRI.size
Bio.Restriction.Restriction.EcoRII.size
Bio.Restriction.Restriction.EcoRV.size
Bio.Restriction.Restriction.EcoT14I.size
Bio.Restriction.Restriction.EcoT22I.size
Bio.Restriction.Restriction.EcoT38I.size
Bio.Restriction.Restriction.EgeI.size
Bio.Restriction.Restriction.EheI.size
Bio.Restriction.Restriction.ErhI.size
Bio.Restriction.Restriction.EsaBC3I.size
Bio.Restriction.Restriction.EsaSSI.size
Bio.Restriction.Restriction.Esp3I.size
Bio.Restriction.Restriction.EspI.size
Bio.Restriction.Restriction.FaeI.size
Bio.Restriction.Restriction.FaiI.size
Bio.Restriction.Restriction.FalI.size
Bio.Restriction.Restriction.FaqI.size
Bio.Restriction.Restriction.FatI.size
Bio.Restriction.Restriction.FauI.size
Bio.Restriction.Restriction.FauNDI.size
Bio.Restriction.Restriction.FbaI.size
Bio.Restriction.Restriction.FblI.size
Bio.Restriction.Restriction.FinI.size
Bio.Restriction.Restriction.FmuI.size
Bio.Restriction.Restriction.Fnu4HI.size
Bio.Restriction.Restriction.FnuDII.size
Bio.Restriction.Restriction.FokI.size
Bio.Restriction.Restriction.FriOI.size
Bio.Restriction.Restriction.FseI.size
Bio.Restriction.Restriction.Fsp4HI.size
Bio.Restriction.Restriction.FspAI.size
Bio.Restriction.Restriction.FspBI.size
Bio.Restriction.Restriction.FspEI.size
Bio.Restriction.Restriction.FspI.size
Bio.Restriction.Restriction.GauT27I.size
Bio.Restriction.Restriction.GdiII.size
Bio.Restriction.Restriction.GlaI.size
Bio.Restriction.Restriction.GluI.size
Bio.Restriction.Restriction.GsaI.size
Bio.Restriction.Restriction.GsuI.size
Bio.Restriction.Restriction.HaeI.size
Bio.Restriction.Restriction.HaeII.size
Bio.Restriction.Restriction.HaeIII.size
Bio.Restriction.Restriction.HapII.size
Bio.Restriction.Restriction.HauII.size
Bio.Restriction.Restriction.HgaI.size
Bio.Restriction.Restriction.HgiAI.size
Bio.Restriction.Restriction.HgiCI.size
Bio.Restriction.Restriction.HgiEII.size
Bio.Restriction.Restriction.HgiJII.size
Bio.Restriction.Restriction.HhaI.size
Bio.Restriction.Restriction.Hin1I.size
Bio.Restriction.Restriction.Hin1II.size
Bio.Restriction.Restriction.Hin4I.size
Bio.Restriction.Restriction.Hin4II.size
Bio.Restriction.Restriction.Hin6I.size
Bio.Restriction.Restriction.HinP1I.size
Bio.Restriction.Restriction.HincII.size
Bio.Restriction.Restriction.HindII.size
Bio.Restriction.Restriction.HindIII.size
Bio.Restriction.Restriction.HinfI.size
Bio.Restriction.Restriction.HpaI.size
Bio.Restriction.Restriction.HpaII.size
Bio.Restriction.Restriction.HphI.size
Bio.Restriction.Restriction.Hpy166II.size
Bio.Restriction.Restriction.Hpy178III.size
Bio.Restriction.Restriction.Hpy188I.size
Bio.Restriction.Restriction.Hpy188III.size
Bio.Restriction.Restriction.Hpy8I.size
Bio.Restriction.Restriction.Hpy99I.size
Bio.Restriction.Restriction.Hpy99XIII.size
Bio.Restriction.Restriction.Hpy99XIV.size
Bio.Restriction.Restriction.HpyAV.size
Bio.Restriction.Restriction.HpyCH4III.size
Bio.Restriction.Restriction.HpyCH4IV.size
Bio.Restriction.Restriction.HpyCH4V.size
Bio.Restriction.Restriction.HpyF10VI.size
Bio.Restriction.Restriction.HpyF3I.size
Bio.Restriction.Restriction.HpySE526I.size
Bio.Restriction.Restriction.Hsp92I.size
Bio.Restriction.Restriction.Hsp92II.size
Bio.Restriction.Restriction.HspAI.size
Bio.Restriction.Restriction.Jma19592I.size
Bio.Restriction.Restriction.KasI.size
Bio.Restriction.Restriction.KflI.size
Bio.Restriction.Restriction.Kpn2I.size
Bio.Restriction.Restriction.KpnI.size
Bio.Restriction.Restriction.KroI.size
Bio.Restriction.Restriction.Ksp22I.size
Bio.Restriction.Restriction.Ksp632I.size
Bio.Restriction.Restriction.KspAI.size
Bio.Restriction.Restriction.KspI.size
Bio.Restriction.Restriction.Kzo9I.size
Bio.Restriction.Restriction.LguI.size
Bio.Restriction.Restriction.LpnI.size
Bio.Restriction.Restriction.LpnPI.size
Bio.Restriction.Restriction.Lsp1109I.size
Bio.Restriction.Restriction.LweI.size
Bio.Restriction.Restriction.MabI.size
Bio.Restriction.Restriction.MaeI.size
Bio.Restriction.Restriction.MaeII.size
Bio.Restriction.Restriction.MaeIII.size
Bio.Restriction.Restriction.MalI.size
Bio.Restriction.Restriction.MaqI.size
Bio.Restriction.Restriction.MauBI.size
Bio.Restriction.Restriction.MbiI.size
Bio.Restriction.Restriction.MboI.size
Bio.Restriction.Restriction.MboII.size
Bio.Restriction.Restriction.McaTI.size
Bio.Restriction.Restriction.McrI.size
Bio.Restriction.Restriction.MfeI.size
Bio.Restriction.Restriction.MflI.size
Bio.Restriction.Restriction.MhlI.size
Bio.Restriction.Restriction.MjaIV.size
Bio.Restriction.Restriction.MkaDII.size
Bio.Restriction.Restriction.MlsI.size
Bio.Restriction.Restriction.MluCI.size
Bio.Restriction.Restriction.MluI.size
Bio.Restriction.Restriction.MluNI.size
Bio.Restriction.Restriction.Mly113I.size
Bio.Restriction.Restriction.MlyI.size
Bio.Restriction.Restriction.MmeI.size
Bio.Restriction.Restriction.MnlI.size
Bio.Restriction.Restriction.Mph1103I.size
Bio.Restriction.Restriction.MreI.size
Bio.Restriction.Restriction.MroI.size
Bio.Restriction.Restriction.MroNI.size
Bio.Restriction.Restriction.MroXI.size
Bio.Restriction.Restriction.MscI.size
Bio.Restriction.Restriction.MseI.size
Bio.Restriction.Restriction.MslI.size
Bio.Restriction.Restriction.Msp20I.size
Bio.Restriction.Restriction.MspA1I.size
Bio.Restriction.Restriction.MspCI.size
Bio.Restriction.Restriction.MspI.size
Bio.Restriction.Restriction.MspJI.size
Bio.Restriction.Restriction.MspR9I.size
Bio.Restriction.Restriction.MssI.size
Bio.Restriction.Restriction.MstI.size
Bio.Restriction.Restriction.MunI.size
Bio.Restriction.Restriction.Mva1269I.size
Bio.Restriction.Restriction.MvaI.size
Bio.Restriction.Restriction.MvnI.size
Bio.Restriction.Restriction.MvrI.size
Bio.Restriction.Restriction.MwoI.size
Bio.Restriction.Restriction.NaeI.size
Bio.Restriction.Restriction.NarI.size
Bio.Restriction.Restriction.NciI.size
Bio.Restriction.Restriction.NcoI.size
Bio.Restriction.Restriction.NdeI.size
Bio.Restriction.Restriction.NdeII.size
Bio.Restriction.Restriction.NgoAVIII.size
Bio.Restriction.Restriction.NgoMIV.size
Bio.Restriction.Restriction.NhaXI.size
Bio.Restriction.Restriction.NheI.size
Bio.Restriction.Restriction.NlaCI.size
Bio.Restriction.Restriction.NlaIII.size
Bio.Restriction.Restriction.NlaIV.size
Bio.Restriction.Restriction.Nli3877I.size
Bio.Restriction.Restriction.NmeAIII.size
Bio.Restriction.Restriction.NmeDI.size
Bio.Restriction.Restriction.NmuCI.size
Bio.Restriction.Restriction.NotI.size
Bio.Restriction.Restriction.NruI.size
Bio.Restriction.Restriction.NsbI.size
Bio.Restriction.Restriction.NsiI.size
Bio.Restriction.Restriction.NspBII.size
Bio.Restriction.Restriction.NspI.size
Bio.Restriction.Restriction.NspV.size
Bio.Restriction.Restriction.OliI.size
Bio.Restriction.Restriction.PabI.size
Bio.Restriction.Restriction.PacI.size
Bio.Restriction.Restriction.PaeI.size
Bio.Restriction.Restriction.PaeR7I.size
Bio.Restriction.Restriction.PagI.size
Bio.Restriction.Restriction.PalAI.size
Bio.Restriction.Restriction.PasI.size
Bio.Restriction.Restriction.PauI.size
Bio.Restriction.Restriction.PceI.size
Bio.Restriction.Restriction.PciI.size
Bio.Restriction.Restriction.PciSI.size
Bio.Restriction.Restriction.PcsI.size
Bio.Restriction.Restriction.PctI.size
Bio.Restriction.Restriction.PdiI.size
Bio.Restriction.Restriction.PdmI.size
Bio.Restriction.Restriction.PenI.size
Bio.Restriction.Restriction.PfeI.size
Bio.Restriction.Restriction.Pfl1108I.size
Bio.Restriction.Restriction.Pfl23II.size
Bio.Restriction.Restriction.PflFI.size
Bio.Restriction.Restriction.PflMI.size
Bio.Restriction.Restriction.PfoI.size
Bio.Restriction.Restriction.PinAI.size
Bio.Restriction.Restriction.PlaDI.size
Bio.Restriction.Restriction.Ple19I.size
Bio.Restriction.Restriction.PleI.size
Bio.Restriction.Restriction.PluTI.size
Bio.Restriction.Restriction.PmaCI.size
Bio.Restriction.Restriction.PmeI.size
Bio.Restriction.Restriction.PmlI.size
Bio.Restriction.Restriction.PpiI.size
Bio.Restriction.Restriction.PpsI.size
Bio.Restriction.Restriction.Ppu10I.size
Bio.Restriction.Restriction.Ppu21I.size
Bio.Restriction.Restriction.PpuMI.size
Bio.Restriction.Restriction.PscI.size
Bio.Restriction.Restriction.PshAI.size
Bio.Restriction.Restriction.PshBI.size
Bio.Restriction.Restriction.PsiI.size
Bio.Restriction.Restriction.Psp03I.size
Bio.Restriction.Restriction.Psp124BI.size
Bio.Restriction.Restriction.Psp1406I.size
Bio.Restriction.Restriction.Psp5II.size
Bio.Restriction.Restriction.Psp6I.size
Bio.Restriction.Restriction.PspCI.size
Bio.Restriction.Restriction.PspEI.size
Bio.Restriction.Restriction.PspGI.size
Bio.Restriction.Restriction.PspLI.size
Bio.Restriction.Restriction.PspN4I.size
Bio.Restriction.Restriction.PspOMI.size
Bio.Restriction.Restriction.PspOMII.size
Bio.Restriction.Restriction.PspPI.size
Bio.Restriction.Restriction.PspPPI.size
Bio.Restriction.Restriction.PspPRI.size
Bio.Restriction.Restriction.PspXI.size
Bio.Restriction.Restriction.PsrI.size
Bio.Restriction.Restriction.PssI.size
Bio.Restriction.Restriction.PstI.size
Bio.Restriction.Restriction.PstNI.size
Bio.Restriction.Restriction.PsuI.size
Bio.Restriction.Restriction.PsyI.size
Bio.Restriction.Restriction.PteI.size
Bio.Restriction.Restriction.PvuI.size
Bio.Restriction.Restriction.PvuII.size
Bio.Restriction.Restriction.R2_BceSIV.size
Bio.Restriction.Restriction.RceI.size
Bio.Restriction.Restriction.RdeGBI.size
Bio.Restriction.Restriction.RdeGBII.size
Bio.Restriction.Restriction.RdeGBIII.size
Bio.Restriction.Restriction.RflFIII.size
Bio.Restriction.Restriction.RgaI.size
Bio.Restriction.Restriction.RigI.size
Bio.Restriction.Restriction.RlaI.size
Bio.Restriction.Restriction.RleAI.size
Bio.Restriction.Restriction.RpaB5I.size
Bio.Restriction.Restriction.RpaBI.size
Bio.Restriction.Restriction.RpaI.size
Bio.Restriction.Restriction.RpaTI.size
Bio.Restriction.Restriction.RruI.size
Bio.Restriction.Restriction.RsaI.size
Bio.Restriction.Restriction.RsaNI.size
Bio.Restriction.Restriction.RseI.size
Bio.Restriction.Restriction.Rsr2I.size
Bio.Restriction.Restriction.RsrII.size
Bio.Restriction.Restriction.SacI.size
Bio.Restriction.Restriction.SacII.size
Bio.Restriction.Restriction.SalI.size
Bio.Restriction.Restriction.SanDI.size
Bio.Restriction.Restriction.SapI.size
Bio.Restriction.Restriction.SaqAI.size
Bio.Restriction.Restriction.SatI.size
Bio.Restriction.Restriction.Sau3AI.size
Bio.Restriction.Restriction.Sau96I.size
Bio.Restriction.Restriction.SauI.size
Bio.Restriction.Restriction.SbfI.size
Bio.Restriction.Restriction.ScaI.size
Bio.Restriction.Restriction.SchI.size
Bio.Restriction.Restriction.SciI.size
Bio.Restriction.Restriction.ScrFI.size
Bio.Restriction.Restriction.SdaI.size
Bio.Restriction.Restriction.SdeAI.size
Bio.Restriction.Restriction.SdeOSI.size
Bio.Restriction.Restriction.SduI.size
Bio.Restriction.Restriction.SecI.size
Bio.Restriction.Restriction.SelI.size
Bio.Restriction.Restriction.SetI.size
Bio.Restriction.Restriction.SexAI.size
Bio.Restriction.Restriction.SfaAI.size
Bio.Restriction.Restriction.SfaNI.size
Bio.Restriction.Restriction.SfcI.size
Bio.Restriction.Restriction.SfeI.size
Bio.Restriction.Restriction.SfiI.size
Bio.Restriction.Restriction.SfoI.size
Bio.Restriction.Restriction.Sfr274I.size
Bio.Restriction.Restriction.Sfr303I.size
Bio.Restriction.Restriction.SfuI.size
Bio.Restriction.Restriction.SgeI.size
Bio.Restriction.Restriction.SgfI.size
Bio.Restriction.Restriction.SgrAI.size
Bio.Restriction.Restriction.SgrBI.size
Bio.Restriction.Restriction.SgrDI.size
Bio.Restriction.Restriction.SgrTI.size
Bio.Restriction.Restriction.SgsI.size
Bio.Restriction.Restriction.SimI.size
Bio.Restriction.Restriction.SlaI.size
Bio.Restriction.Restriction.SmaI.size
Bio.Restriction.Restriction.SmiI.size
Bio.Restriction.Restriction.SmiMI.size
Bio.Restriction.Restriction.SmlI.size
Bio.Restriction.Restriction.SmoI.size
Bio.Restriction.Restriction.SnaBI.size
Bio.Restriction.Restriction.SnaI.size
Bio.Restriction.Restriction.Sno506I.size
Bio.Restriction.Restriction.SpeI.size
Bio.Restriction.Restriction.SphI.size
Bio.Restriction.Restriction.SplI.size
Bio.Restriction.Restriction.SpoDI.size
Bio.Restriction.Restriction.SrfI.size
Bio.Restriction.Restriction.Sse232I.size
Bio.Restriction.Restriction.Sse8387I.size
Bio.Restriction.Restriction.Sse8647I.size
Bio.Restriction.Restriction.Sse9I.size
Bio.Restriction.Restriction.SseBI.size
Bio.Restriction.Restriction.SsiI.size
Bio.Restriction.Restriction.SspD5I.size
Bio.Restriction.Restriction.SspDI.size
Bio.Restriction.Restriction.SspI.size
Bio.Restriction.Restriction.SstE37I.size
Bio.Restriction.Restriction.SstI.size
Bio.Restriction.Restriction.Sth132I.size
Bio.Restriction.Restriction.Sth302II.size
Bio.Restriction.Restriction.StrI.size
Bio.Restriction.Restriction.StsI.size
Bio.Restriction.Restriction.StuI.size
Bio.Restriction.Restriction.StyD4I.size
Bio.Restriction.Restriction.StyI.size
Bio.Restriction.Restriction.SwaI.size
Bio.Restriction.Restriction.TaaI.size
Bio.Restriction.Restriction.TaiI.size
Bio.Restriction.Restriction.TaqI.size
Bio.Restriction.Restriction.TaqII.size
Bio.Restriction.Restriction.TasI.size
Bio.Restriction.Restriction.TatI.size
Bio.Restriction.Restriction.TauI.size
Bio.Restriction.Restriction.TfiI.size
Bio.Restriction.Restriction.Tru1I.size
Bio.Restriction.Restriction.Tru9I.size
Bio.Restriction.Restriction.TscAI.size
Bio.Restriction.Restriction.TseFI.size
Bio.Restriction.Restriction.TseI.size
Bio.Restriction.Restriction.TsoI.size
Bio.Restriction.Restriction.Tsp45I.size
Bio.Restriction.Restriction.Tsp4CI.size
Bio.Restriction.Restriction.TspDTI.size
Bio.Restriction.Restriction.TspEI.size
Bio.Restriction.Restriction.TspGWI.size
Bio.Restriction.Restriction.TspMI.size
Bio.Restriction.Restriction.TspRI.size
Bio.Restriction.Restriction.TssI.size
Bio.Restriction.Restriction.TstI.size
Bio.Restriction.Restriction.TsuI.size
Bio.Restriction.Restriction.Tth111I.size
Bio.Restriction.Restriction.Tth111II.size
Bio.Restriction.Restriction.UbaF11I.size
Bio.Restriction.Restriction.UbaF12I.size
Bio.Restriction.Restriction.UbaF13I.size
Bio.Restriction.Restriction.UbaF14I.size
Bio.Restriction.Restriction.UbaF9I.size
Bio.Restriction.Restriction.UbaPI.size
Bio.Restriction.Restriction.UcoMSI.size
Bio.Restriction.Restriction.UnbI.size
Bio.Restriction.Restriction.Van91I.size
Bio.Restriction.Restriction.Vha464I.size
Bio.Restriction.Restriction.VneI.size
Bio.Restriction.Restriction.VpaK11AI.size
Bio.Restriction.Restriction.VpaK11BI.size
Bio.Restriction.Restriction.VspI.size
Bio.Restriction.Restriction.WviI.size
Bio.Restriction.Restriction.XagI.size
Bio.Restriction.Restriction.XapI.size
Bio.Restriction.Restriction.XbaI.size
Bio.Restriction.Restriction.XceI.size
Bio.Restriction.Restriction.XcmI.size
Bio.Restriction.Restriction.XhoI.size
Bio.Restriction.Restriction.XhoII.size
Bio.Restriction.Restriction.XmaI.size
Bio.Restriction.Restriction.XmaIII.size
Bio.Restriction.Restriction.XmaJI.size
Bio.Restriction.Restriction.XmiI.size
Bio.Restriction.Restriction.XmnI.size
Bio.Restriction.Restriction.XspI.size
Bio.Restriction.Restriction.YkrI.size
Bio.Restriction.Restriction.ZraI.size
Bio.Restriction.Restriction.ZrmI.size
Bio.Restriction.Restriction.Zsp2I.size
Bio.codonalign.codonalphabet.CodonAlphabet.size" class="py-name" href="#" onclick="return doclink('link-70', 'size', 'link-70');">size</a></tt> <tt class="py-op">=</tt> <tt class="py-name">read_flow_size</tt> <tt class="py-op">+</tt> <tt class="py-number">3</tt> <tt class="py-op">*</tt> <tt class="py-name">seq_len</tt> </tt>
<a name="L417"></a><tt class="py-lineno"> 417</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-71" class="py-name"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-71', 'seek', 'link-50');">seek</a></tt><tt class="py-op">(</tt><tt id="link-72" class="py-name"><a title="Bio.Alphabet.Alphabet.size
Bio.Alphabet.Reduced.HPModel.size
Bio.Alphabet.Reduced.Murphy10.size
Bio.Alphabet.Reduced.Murphy15.size
Bio.Alphabet.Reduced.Murphy4.size
Bio.Alphabet.Reduced.Murphy8.size
Bio.Alphabet.Reduced.PC5.size
Bio.Alphabet.SingleLetterAlphabet.size
Bio.Alphabet.ThreeLetterProtein.size
Bio.GenBank._FeatureConsumer.size
Bio.GenBank._RecordConsumer.size
Bio.Restriction.Restriction.AanI.size
Bio.Restriction.Restriction.AarI.size
Bio.Restriction.Restriction.AasI.size
Bio.Restriction.Restriction.AatII.size
Bio.Restriction.Restriction.AbaSI.size
Bio.Restriction.Restriction.AbsI.size
Bio.Restriction.Restriction.Acc16I.size
Bio.Restriction.Restriction.Acc36I.size
Bio.Restriction.Restriction.Acc65I.size
Bio.Restriction.Restriction.AccB1I.size
Bio.Restriction.Restriction.AccB7I.size
Bio.Restriction.Restriction.AccBSI.size
Bio.Restriction.Restriction.AccI.size
Bio.Restriction.Restriction.AccII.size
Bio.Restriction.Restriction.AccIII.size
Bio.Restriction.Restriction.AceIII.size
Bio.Restriction.Restriction.AciI.size
Bio.Restriction.Restriction.AclI.size
Bio.Restriction.Restriction.AclWI.size
Bio.Restriction.Restriction.AcoI.size
Bio.Restriction.Restriction.AcsI.size
Bio.Restriction.Restriction.AcuI.size
Bio.Restriction.Restriction.AcvI.size
Bio.Restriction.Restriction.AcyI.size
Bio.Restriction.Restriction.AdeI.size
Bio.Restriction.Restriction.AfaI.size
Bio.Restriction.Restriction.AfeI.size
Bio.Restriction.Restriction.AfiI.size
Bio.Restriction.Restriction.AflII.size
Bio.Restriction.Restriction.AflIII.size
Bio.Restriction.Restriction.AgeI.size
Bio.Restriction.Restriction.AgsI.size
Bio.Restriction.Restriction.AhaIII.size
Bio.Restriction.Restriction.AhdI.size
Bio.Restriction.Restriction.AhlI.size
Bio.Restriction.Restriction.AjiI.size
Bio.Restriction.Restriction.AjnI.size
Bio.Restriction.Restriction.AjuI.size
Bio.Restriction.Restriction.AleI.size
Bio.Restriction.Restriction.AlfI.size
Bio.Restriction.Restriction.AloI.size
Bio.Restriction.Restriction.AluBI.size
Bio.Restriction.Restriction.AluI.size
Bio.Restriction.Restriction.Alw21I.size
Bio.Restriction.Restriction.Alw26I.size
Bio.Restriction.Restriction.Alw44I.size
Bio.Restriction.Restriction.AlwFI.size
Bio.Restriction.Restriction.AlwI.size
Bio.Restriction.Restriction.AlwNI.size
Bio.Restriction.Restriction.Ama87I.size
Bio.Restriction.Restriction.Aor13HI.size
Bio.Restriction.Restriction.Aor51HI.size
Bio.Restriction.Restriction.AoxI.size
Bio.Restriction.Restriction.ApaBI.size
Bio.Restriction.Restriction.ApaI.size
Bio.Restriction.Restriction.ApaLI.size
Bio.Restriction.Restriction.ApeKI.size
Bio.Restriction.Restriction.ApoI.size
Bio.Restriction.Restriction.ApyPI.size
Bio.Restriction.Restriction.AquII.size
Bio.Restriction.Restriction.AquIII.size
Bio.Restriction.Restriction.AquIV.size
Bio.Restriction.Restriction.ArsI.size
Bio.Restriction.Restriction.AscI.size
Bio.Restriction.Restriction.AseI.size
Bio.Restriction.Restriction.Asi256I.size
Bio.Restriction.Restriction.AsiGI.size
Bio.Restriction.Restriction.AsiSI.size
Bio.Restriction.Restriction.Asp700I.size
Bio.Restriction.Restriction.Asp718I.size
Bio.Restriction.Restriction.AspA2I.size
Bio.Restriction.Restriction.AspBHI.size
Bio.Restriction.Restriction.AspLEI.size
Bio.Restriction.Restriction.AspS9I.size
Bio.Restriction.Restriction.AssI.size
Bio.Restriction.Restriction.AsuC2I.size
Bio.Restriction.Restriction.AsuHPI.size
Bio.Restriction.Restriction.AsuI.size
Bio.Restriction.Restriction.AsuII.size
Bio.Restriction.Restriction.AsuNHI.size
Bio.Restriction.Restriction.AvaI.size
Bio.Restriction.Restriction.AvaII.size
Bio.Restriction.Restriction.AvaIII.size
Bio.Restriction.Restriction.AvrII.size
Bio.Restriction.Restriction.AxyI.size
Bio.Restriction.Restriction.BaeGI.size
Bio.Restriction.Restriction.BaeI.size
Bio.Restriction.Restriction.BalI.size
Bio.Restriction.Restriction.BamHI.size
Bio.Restriction.Restriction.BanI.size
Bio.Restriction.Restriction.BanII.size
Bio.Restriction.Restriction.BarI.size
Bio.Restriction.Restriction.BasI.size
Bio.Restriction.Restriction.BauI.size
Bio.Restriction.Restriction.Bbr7I.size
Bio.Restriction.Restriction.BbrPI.size
Bio.Restriction.Restriction.BbsI.size
Bio.Restriction.Restriction.Bbv12I.size
Bio.Restriction.Restriction.BbvCI.size
Bio.Restriction.Restriction.BbvI.size
Bio.Restriction.Restriction.BbvII.size
Bio.Restriction.Restriction.BccI.size
Bio.Restriction.Restriction.Bce83I.size
Bio.Restriction.Restriction.BceAI.size
Bio.Restriction.Restriction.BcefI.size
Bio.Restriction.Restriction.BcgI.size
Bio.Restriction.Restriction.BciT130I.size
Bio.Restriction.Restriction.BciVI.size
Bio.Restriction.Restriction.BclI.size
Bio.Restriction.Restriction.BcnI.size
Bio.Restriction.Restriction.BcoDI.size
Bio.Restriction.Restriction.BcuI.size
Bio.Restriction.Restriction.BdaI.size
Bio.Restriction.Restriction.BetI.size
Bio.Restriction.Restriction.BfaI.size
Bio.Restriction.Restriction.BfiI.size
Bio.Restriction.Restriction.BfmI.size
Bio.Restriction.Restriction.BfoI.size
Bio.Restriction.Restriction.BfrI.size
Bio.Restriction.Restriction.BfuAI.size
Bio.Restriction.Restriction.BfuCI.size
Bio.Restriction.Restriction.BfuI.size
Bio.Restriction.Restriction.BglI.size
Bio.Restriction.Restriction.BglII.size
Bio.Restriction.Restriction.BinI.size
Bio.Restriction.Restriction.BisI.size
Bio.Restriction.Restriction.BlnI.size
Bio.Restriction.Restriction.BlpI.size
Bio.Restriction.Restriction.BlsI.size
Bio.Restriction.Restriction.BmcAI.size
Bio.Restriction.Restriction.Bme1390I.size
Bio.Restriction.Restriction.Bme18I.size
Bio.Restriction.Restriction.BmeDI.size
Bio.Restriction.Restriction.BmeRI.size
Bio.Restriction.Restriction.BmeT110I.size
Bio.Restriction.Restriction.BmgBI.size
Bio.Restriction.Restriction.BmgI.size
Bio.Restriction.Restriction.BmgT120I.size
Bio.Restriction.Restriction.BmiI.size
Bio.Restriction.Restriction.BmrFI.size
Bio.Restriction.Restriction.BmrI.size
Bio.Restriction.Restriction.BmsI.size
Bio.Restriction.Restriction.BmtI.size
Bio.Restriction.Restriction.BmuI.size
Bio.Restriction.Restriction.BoxI.size
Bio.Restriction.Restriction.BpiI.size
Bio.Restriction.Restriction.BplI.size
Bio.Restriction.Restriction.BpmI.size
Bio.Restriction.Restriction.Bpu10I.size
Bio.Restriction.Restriction.Bpu1102I.size
Bio.Restriction.Restriction.Bpu14I.size
Bio.Restriction.Restriction.BpuEI.size
Bio.Restriction.Restriction.BpuMI.size
Bio.Restriction.Restriction.BpvUI.size
Bio.Restriction.Restriction.Bsa29I.size
Bio.Restriction.Restriction.BsaAI.size
Bio.Restriction.Restriction.BsaBI.size
Bio.Restriction.Restriction.BsaHI.size
Bio.Restriction.Restriction.BsaI.size
Bio.Restriction.Restriction.BsaJI.size
Bio.Restriction.Restriction.BsaWI.size
Bio.Restriction.Restriction.BsaXI.size
Bio.Restriction.Restriction.BsbI.size
Bio.Restriction.Restriction.Bsc4I.size
Bio.Restriction.Restriction.BscAI.size
Bio.Restriction.Restriction.BscGI.size
Bio.Restriction.Restriction.Bse118I.size
Bio.Restriction.Restriction.Bse1I.size
Bio.Restriction.Restriction.Bse21I.size
Bio.Restriction.Restriction.Bse3DI.size
Bio.Restriction.Restriction.Bse8I.size
Bio.Restriction.Restriction.BseAI.size
Bio.Restriction.Restriction.BseBI.size
Bio.Restriction.Restriction.BseCI.size
Bio.Restriction.Restriction.BseDI.size
Bio.Restriction.Restriction.BseGI.size
Bio.Restriction.Restriction.BseJI.size
Bio.Restriction.Restriction.BseLI.size
Bio.Restriction.Restriction.BseMI.size
Bio.Restriction.Restriction.BseMII.size
Bio.Restriction.Restriction.BseNI.size
Bio.Restriction.Restriction.BsePI.size
Bio.Restriction.Restriction.BseRI.size
Bio.Restriction.Restriction.BseSI.size
Bio.Restriction.Restriction.BseX3I.size
Bio.Restriction.Restriction.BseXI.size
Bio.Restriction.Restriction.BseYI.size
Bio.Restriction.Restriction.BsgI.size
Bio.Restriction.Restriction.Bsh1236I.size
Bio.Restriction.Restriction.Bsh1285I.size
Bio.Restriction.Restriction.BshFI.size
Bio.Restriction.Restriction.BshNI.size
Bio.Restriction.Restriction.BshTI.size
Bio.Restriction.Restriction.BshVI.size
Bio.Restriction.Restriction.BsiEI.size
Bio.Restriction.Restriction.BsiHKAI.size
Bio.Restriction.Restriction.BsiHKCI.size
Bio.Restriction.Restriction.BsiI.size
Bio.Restriction.Restriction.BsiSI.size
Bio.Restriction.Restriction.BsiWI.size
Bio.Restriction.Restriction.BsiYI.size
Bio.Restriction.Restriction.BslFI.size
Bio.Restriction.Restriction.BslI.size
Bio.Restriction.Restriction.BsmAI.size
Bio.Restriction.Restriction.BsmBI.size
Bio.Restriction.Restriction.BsmFI.size
Bio.Restriction.Restriction.BsmI.size
Bio.Restriction.Restriction.BsnI.size
Bio.Restriction.Restriction.Bso31I.size
Bio.Restriction.Restriction.BsoBI.size
Bio.Restriction.Restriction.Bsp119I.size
Bio.Restriction.Restriction.Bsp120I.size
Bio.Restriction.Restriction.Bsp1286I.size
Bio.Restriction.Restriction.Bsp13I.size
Bio.Restriction.Restriction.Bsp1407I.size
Bio.Restriction.Restriction.Bsp143I.size
Bio.Restriction.Restriction.Bsp1720I.size
Bio.Restriction.Restriction.Bsp19I.size
Bio.Restriction.Restriction.Bsp24I.size
Bio.Restriction.Restriction.Bsp68I.size
Bio.Restriction.Restriction.BspACI.size
Bio.Restriction.Restriction.BspCNI.size
Bio.Restriction.Restriction.BspD6I.size
Bio.Restriction.Restriction.BspDI.size
Bio.Restriction.Restriction.BspEI.size
Bio.Restriction.Restriction.BspFNI.size
Bio.Restriction.Restriction.BspGI.size
Bio.Restriction.Restriction.BspHI.size
Bio.Restriction.Restriction.BspLI.size
Bio.Restriction.Restriction.BspLU11I.size
Bio.Restriction.Restriction.BspMI.size
Bio.Restriction.Restriction.BspMII.size
Bio.Restriction.Restriction.BspNCI.size
Bio.Restriction.Restriction.BspOI.size
Bio.Restriction.Restriction.BspPI.size
Bio.Restriction.Restriction.BspQI.size
Bio.Restriction.Restriction.BspT104I.size
Bio.Restriction.Restriction.BspT107I.size
Bio.Restriction.Restriction.BspTI.size
Bio.Restriction.Restriction.BsrBI.size
Bio.Restriction.Restriction.BsrDI.size
Bio.Restriction.Restriction.BsrFI.size
Bio.Restriction.Restriction.BsrGI.size
Bio.Restriction.Restriction.BsrI.size
Bio.Restriction.Restriction.BsrSI.size
Bio.Restriction.Restriction.BssAI.size
Bio.Restriction.Restriction.BssECI.size
Bio.Restriction.Restriction.BssHII.size
Bio.Restriction.Restriction.BssKI.size
Bio.Restriction.Restriction.BssMI.size
Bio.Restriction.Restriction.BssNAI.size
Bio.Restriction.Restriction.BssNI.size
Bio.Restriction.Restriction.BssSI.size
Bio.Restriction.Restriction.BssT1I.size
Bio.Restriction.Restriction.Bst1107I.size
Bio.Restriction.Restriction.Bst2BI.size
Bio.Restriction.Restriction.Bst2UI.size
Bio.Restriction.Restriction.Bst4CI.size
Bio.Restriction.Restriction.Bst6I.size
Bio.Restriction.Restriction.BstACI.size
Bio.Restriction.Restriction.BstAFI.size
Bio.Restriction.Restriction.BstAPI.size
Bio.Restriction.Restriction.BstAUI.size
Bio.Restriction.Restriction.BstBAI.size
Bio.Restriction.Restriction.BstBI.size
Bio.Restriction.Restriction.BstC8I.size
Bio.Restriction.Restriction.BstDEI.size
Bio.Restriction.Restriction.BstDSI.size
Bio.Restriction.Restriction.BstEII.size
Bio.Restriction.Restriction.BstENI.size
Bio.Restriction.Restriction.BstF5I.size
Bio.Restriction.Restriction.BstFNI.size
Bio.Restriction.Restriction.BstH2I.size
Bio.Restriction.Restriction.BstHHI.size
Bio.Restriction.Restriction.BstKTI.size
Bio.Restriction.Restriction.BstMAI.size
Bio.Restriction.Restriction.BstMBI.size
Bio.Restriction.Restriction.BstMCI.size
Bio.Restriction.Restriction.BstMWI.size
Bio.Restriction.Restriction.BstNI.size
Bio.Restriction.Restriction.BstNSI.size
Bio.Restriction.Restriction.BstOI.size
Bio.Restriction.Restriction.BstPAI.size
Bio.Restriction.Restriction.BstPI.size
Bio.Restriction.Restriction.BstSCI.size
Bio.Restriction.Restriction.BstSFI.size
Bio.Restriction.Restriction.BstSLI.size
Bio.Restriction.Restriction.BstSNI.size
Bio.Restriction.Restriction.BstUI.size
Bio.Restriction.Restriction.BstV1I.size
Bio.Restriction.Restriction.BstV2I.size
Bio.Restriction.Restriction.BstX2I.size
Bio.Restriction.Restriction.BstXI.size
Bio.Restriction.Restriction.BstYI.size
Bio.Restriction.Restriction.BstZ17I.size
Bio.Restriction.Restriction.BstZI.size
Bio.Restriction.Restriction.Bsu15I.size
Bio.Restriction.Restriction.Bsu36I.size
Bio.Restriction.Restriction.BsuI.size
Bio.Restriction.Restriction.BsuRI.size
Bio.Restriction.Restriction.BtgI.size
Bio.Restriction.Restriction.BtgZI.size
Bio.Restriction.Restriction.BthCI.size
Bio.Restriction.Restriction.BtrI.size
Bio.Restriction.Restriction.BtsCI.size
Bio.Restriction.Restriction.BtsI.size
Bio.Restriction.Restriction.BtsIMutI.size
Bio.Restriction.Restriction.BtuMI.size
Bio.Restriction.Restriction.BveI.size
Bio.Restriction.Restriction.Cac8I.size
Bio.Restriction.Restriction.CaiI.size
Bio.Restriction.Restriction.CauII.size
Bio.Restriction.Restriction.CchII.size
Bio.Restriction.Restriction.CchIII.size
Bio.Restriction.Restriction.CciI.size
Bio.Restriction.Restriction.CciNI.size
Bio.Restriction.Restriction.Cdi630V.size
Bio.Restriction.Restriction.CdiI.size
Bio.Restriction.Restriction.CdpI.size
Bio.Restriction.Restriction.CfoI.size
Bio.Restriction.Restriction.Cfr10I.size
Bio.Restriction.Restriction.Cfr13I.size
Bio.Restriction.Restriction.Cfr42I.size
Bio.Restriction.Restriction.Cfr9I.size
Bio.Restriction.Restriction.CfrI.size
Bio.Restriction.Restriction.Cgl13032I.size
Bio.Restriction.Restriction.Cgl13032II.size
Bio.Restriction.Restriction.ChaI.size
Bio.Restriction.Restriction.CjeFIII.size
Bio.Restriction.Restriction.CjeFV.size
Bio.Restriction.Restriction.CjeI.size
Bio.Restriction.Restriction.CjeNII.size
Bio.Restriction.Restriction.CjeNIII.size
Bio.Restriction.Restriction.CjeP659IV.size
Bio.Restriction.Restriction.CjePI.size
Bio.Restriction.Restriction.CjuI.size
Bio.Restriction.Restriction.CjuII.size
Bio.Restriction.Restriction.ClaI.size
Bio.Restriction.Restriction.CpoI.size
Bio.Restriction.Restriction.CseI.size
Bio.Restriction.Restriction.CsiI.size
Bio.Restriction.Restriction.Csp6I.size
Bio.Restriction.Restriction.CspAI.size
Bio.Restriction.Restriction.CspCI.size
Bio.Restriction.Restriction.CspI.size
Bio.Restriction.Restriction.CstMI.size
Bio.Restriction.Restriction.CviAII.size
Bio.Restriction.Restriction.CviJI.size
Bio.Restriction.Restriction.CviKI_1.size
Bio.Restriction.Restriction.CviQI.size
Bio.Restriction.Restriction.CviRI.size
Bio.Restriction.Restriction.DdeI.size
Bio.Restriction.Restriction.DinI.size
Bio.Restriction.Restriction.DpnI.size
Bio.Restriction.Restriction.DpnII.size
Bio.Restriction.Restriction.DraI.size
Bio.Restriction.Restriction.DraII.size
Bio.Restriction.Restriction.DraIII.size
Bio.Restriction.Restriction.DraRI.size
Bio.Restriction.Restriction.DrdI.size
Bio.Restriction.Restriction.DrdII.size
Bio.Restriction.Restriction.DriI.size
Bio.Restriction.Restriction.DsaI.size
Bio.Restriction.Restriction.DseDI.size
Bio.Restriction.Restriction.EaeI.size
Bio.Restriction.Restriction.EagI.size
Bio.Restriction.Restriction.Eam1104I.size
Bio.Restriction.Restriction.Eam1105I.size
Bio.Restriction.Restriction.EarI.size
Bio.Restriction.Restriction.EciI.size
Bio.Restriction.Restriction.Ecl136II.size
Bio.Restriction.Restriction.EclXI.size
Bio.Restriction.Restriction.Eco105I.size
Bio.Restriction.Restriction.Eco130I.size
Bio.Restriction.Restriction.Eco147I.size
Bio.Restriction.Restriction.Eco24I.size
Bio.Restriction.Restriction.Eco31I.size
Bio.Restriction.Restriction.Eco32I.size
Bio.Restriction.Restriction.Eco47I.size
Bio.Restriction.Restriction.Eco47III.size
Bio.Restriction.Restriction.Eco52I.size
Bio.Restriction.Restriction.Eco53kI.size
Bio.Restriction.Restriction.Eco57I.size
Bio.Restriction.Restriction.Eco57MI.size
Bio.Restriction.Restriction.Eco72I.size
Bio.Restriction.Restriction.Eco81I.size
Bio.Restriction.Restriction.Eco88I.size
Bio.Restriction.Restriction.Eco91I.size
Bio.Restriction.Restriction.EcoHI.size
Bio.Restriction.Restriction.EcoICRI.size
Bio.Restriction.Restriction.EcoNI.size
Bio.Restriction.Restriction.EcoO109I.size
Bio.Restriction.Restriction.EcoO65I.size
Bio.Restriction.Restriction.EcoRI.size
Bio.Restriction.Restriction.EcoRII.size
Bio.Restriction.Restriction.EcoRV.size
Bio.Restriction.Restriction.EcoT14I.size
Bio.Restriction.Restriction.EcoT22I.size
Bio.Restriction.Restriction.EcoT38I.size
Bio.Restriction.Restriction.EgeI.size
Bio.Restriction.Restriction.EheI.size
Bio.Restriction.Restriction.ErhI.size
Bio.Restriction.Restriction.EsaBC3I.size
Bio.Restriction.Restriction.EsaSSI.size
Bio.Restriction.Restriction.Esp3I.size
Bio.Restriction.Restriction.EspI.size
Bio.Restriction.Restriction.FaeI.size
Bio.Restriction.Restriction.FaiI.size
Bio.Restriction.Restriction.FalI.size
Bio.Restriction.Restriction.FaqI.size
Bio.Restriction.Restriction.FatI.size
Bio.Restriction.Restriction.FauI.size
Bio.Restriction.Restriction.FauNDI.size
Bio.Restriction.Restriction.FbaI.size
Bio.Restriction.Restriction.FblI.size
Bio.Restriction.Restriction.FinI.size
Bio.Restriction.Restriction.FmuI.size
Bio.Restriction.Restriction.Fnu4HI.size
Bio.Restriction.Restriction.FnuDII.size
Bio.Restriction.Restriction.FokI.size
Bio.Restriction.Restriction.FriOI.size
Bio.Restriction.Restriction.FseI.size
Bio.Restriction.Restriction.Fsp4HI.size
Bio.Restriction.Restriction.FspAI.size
Bio.Restriction.Restriction.FspBI.size
Bio.Restriction.Restriction.FspEI.size
Bio.Restriction.Restriction.FspI.size
Bio.Restriction.Restriction.GauT27I.size
Bio.Restriction.Restriction.GdiII.size
Bio.Restriction.Restriction.GlaI.size
Bio.Restriction.Restriction.GluI.size
Bio.Restriction.Restriction.GsaI.size
Bio.Restriction.Restriction.GsuI.size
Bio.Restriction.Restriction.HaeI.size
Bio.Restriction.Restriction.HaeII.size
Bio.Restriction.Restriction.HaeIII.size
Bio.Restriction.Restriction.HapII.size
Bio.Restriction.Restriction.HauII.size
Bio.Restriction.Restriction.HgaI.size
Bio.Restriction.Restriction.HgiAI.size
Bio.Restriction.Restriction.HgiCI.size
Bio.Restriction.Restriction.HgiEII.size
Bio.Restriction.Restriction.HgiJII.size
Bio.Restriction.Restriction.HhaI.size
Bio.Restriction.Restriction.Hin1I.size
Bio.Restriction.Restriction.Hin1II.size
Bio.Restriction.Restriction.Hin4I.size
Bio.Restriction.Restriction.Hin4II.size
Bio.Restriction.Restriction.Hin6I.size
Bio.Restriction.Restriction.HinP1I.size
Bio.Restriction.Restriction.HincII.size
Bio.Restriction.Restriction.HindII.size
Bio.Restriction.Restriction.HindIII.size
Bio.Restriction.Restriction.HinfI.size
Bio.Restriction.Restriction.HpaI.size
Bio.Restriction.Restriction.HpaII.size
Bio.Restriction.Restriction.HphI.size
Bio.Restriction.Restriction.Hpy166II.size
Bio.Restriction.Restriction.Hpy178III.size
Bio.Restriction.Restriction.Hpy188I.size
Bio.Restriction.Restriction.Hpy188III.size
Bio.Restriction.Restriction.Hpy8I.size
Bio.Restriction.Restriction.Hpy99I.size
Bio.Restriction.Restriction.Hpy99XIII.size
Bio.Restriction.Restriction.Hpy99XIV.size
Bio.Restriction.Restriction.HpyAV.size
Bio.Restriction.Restriction.HpyCH4III.size
Bio.Restriction.Restriction.HpyCH4IV.size
Bio.Restriction.Restriction.HpyCH4V.size
Bio.Restriction.Restriction.HpyF10VI.size
Bio.Restriction.Restriction.HpyF3I.size
Bio.Restriction.Restriction.HpySE526I.size
Bio.Restriction.Restriction.Hsp92I.size
Bio.Restriction.Restriction.Hsp92II.size
Bio.Restriction.Restriction.HspAI.size
Bio.Restriction.Restriction.Jma19592I.size
Bio.Restriction.Restriction.KasI.size
Bio.Restriction.Restriction.KflI.size
Bio.Restriction.Restriction.Kpn2I.size
Bio.Restriction.Restriction.KpnI.size
Bio.Restriction.Restriction.KroI.size
Bio.Restriction.Restriction.Ksp22I.size
Bio.Restriction.Restriction.Ksp632I.size
Bio.Restriction.Restriction.KspAI.size
Bio.Restriction.Restriction.KspI.size
Bio.Restriction.Restriction.Kzo9I.size
Bio.Restriction.Restriction.LguI.size
Bio.Restriction.Restriction.LpnI.size
Bio.Restriction.Restriction.LpnPI.size
Bio.Restriction.Restriction.Lsp1109I.size
Bio.Restriction.Restriction.LweI.size
Bio.Restriction.Restriction.MabI.size
Bio.Restriction.Restriction.MaeI.size
Bio.Restriction.Restriction.MaeII.size
Bio.Restriction.Restriction.MaeIII.size
Bio.Restriction.Restriction.MalI.size
Bio.Restriction.Restriction.MaqI.size
Bio.Restriction.Restriction.MauBI.size
Bio.Restriction.Restriction.MbiI.size
Bio.Restriction.Restriction.MboI.size
Bio.Restriction.Restriction.MboII.size
Bio.Restriction.Restriction.McaTI.size
Bio.Restriction.Restriction.McrI.size
Bio.Restriction.Restriction.MfeI.size
Bio.Restriction.Restriction.MflI.size
Bio.Restriction.Restriction.MhlI.size
Bio.Restriction.Restriction.MjaIV.size
Bio.Restriction.Restriction.MkaDII.size
Bio.Restriction.Restriction.MlsI.size
Bio.Restriction.Restriction.MluCI.size
Bio.Restriction.Restriction.MluI.size
Bio.Restriction.Restriction.MluNI.size
Bio.Restriction.Restriction.Mly113I.size
Bio.Restriction.Restriction.MlyI.size
Bio.Restriction.Restriction.MmeI.size
Bio.Restriction.Restriction.MnlI.size
Bio.Restriction.Restriction.Mph1103I.size
Bio.Restriction.Restriction.MreI.size
Bio.Restriction.Restriction.MroI.size
Bio.Restriction.Restriction.MroNI.size
Bio.Restriction.Restriction.MroXI.size
Bio.Restriction.Restriction.MscI.size
Bio.Restriction.Restriction.MseI.size
Bio.Restriction.Restriction.MslI.size
Bio.Restriction.Restriction.Msp20I.size
Bio.Restriction.Restriction.MspA1I.size
Bio.Restriction.Restriction.MspCI.size
Bio.Restriction.Restriction.MspI.size
Bio.Restriction.Restriction.MspJI.size
Bio.Restriction.Restriction.MspR9I.size
Bio.Restriction.Restriction.MssI.size
Bio.Restriction.Restriction.MstI.size
Bio.Restriction.Restriction.MunI.size
Bio.Restriction.Restriction.Mva1269I.size
Bio.Restriction.Restriction.MvaI.size
Bio.Restriction.Restriction.MvnI.size
Bio.Restriction.Restriction.MvrI.size
Bio.Restriction.Restriction.MwoI.size
Bio.Restriction.Restriction.NaeI.size
Bio.Restriction.Restriction.NarI.size
Bio.Restriction.Restriction.NciI.size
Bio.Restriction.Restriction.NcoI.size
Bio.Restriction.Restriction.NdeI.size
Bio.Restriction.Restriction.NdeII.size
Bio.Restriction.Restriction.NgoAVIII.size
Bio.Restriction.Restriction.NgoMIV.size
Bio.Restriction.Restriction.NhaXI.size
Bio.Restriction.Restriction.NheI.size
Bio.Restriction.Restriction.NlaCI.size
Bio.Restriction.Restriction.NlaIII.size
Bio.Restriction.Restriction.NlaIV.size
Bio.Restriction.Restriction.Nli3877I.size
Bio.Restriction.Restriction.NmeAIII.size
Bio.Restriction.Restriction.NmeDI.size
Bio.Restriction.Restriction.NmuCI.size
Bio.Restriction.Restriction.NotI.size
Bio.Restriction.Restriction.NruI.size
Bio.Restriction.Restriction.NsbI.size
Bio.Restriction.Restriction.NsiI.size
Bio.Restriction.Restriction.NspBII.size
Bio.Restriction.Restriction.NspI.size
Bio.Restriction.Restriction.NspV.size
Bio.Restriction.Restriction.OliI.size
Bio.Restriction.Restriction.PabI.size
Bio.Restriction.Restriction.PacI.size
Bio.Restriction.Restriction.PaeI.size
Bio.Restriction.Restriction.PaeR7I.size
Bio.Restriction.Restriction.PagI.size
Bio.Restriction.Restriction.PalAI.size
Bio.Restriction.Restriction.PasI.size
Bio.Restriction.Restriction.PauI.size
Bio.Restriction.Restriction.PceI.size
Bio.Restriction.Restriction.PciI.size
Bio.Restriction.Restriction.PciSI.size
Bio.Restriction.Restriction.PcsI.size
Bio.Restriction.Restriction.PctI.size
Bio.Restriction.Restriction.PdiI.size
Bio.Restriction.Restriction.PdmI.size
Bio.Restriction.Restriction.PenI.size
Bio.Restriction.Restriction.PfeI.size
Bio.Restriction.Restriction.Pfl1108I.size
Bio.Restriction.Restriction.Pfl23II.size
Bio.Restriction.Restriction.PflFI.size
Bio.Restriction.Restriction.PflMI.size
Bio.Restriction.Restriction.PfoI.size
Bio.Restriction.Restriction.PinAI.size
Bio.Restriction.Restriction.PlaDI.size
Bio.Restriction.Restriction.Ple19I.size
Bio.Restriction.Restriction.PleI.size
Bio.Restriction.Restriction.PluTI.size
Bio.Restriction.Restriction.PmaCI.size
Bio.Restriction.Restriction.PmeI.size
Bio.Restriction.Restriction.PmlI.size
Bio.Restriction.Restriction.PpiI.size
Bio.Restriction.Restriction.PpsI.size
Bio.Restriction.Restriction.Ppu10I.size
Bio.Restriction.Restriction.Ppu21I.size
Bio.Restriction.Restriction.PpuMI.size
Bio.Restriction.Restriction.PscI.size
Bio.Restriction.Restriction.PshAI.size
Bio.Restriction.Restriction.PshBI.size
Bio.Restriction.Restriction.PsiI.size
Bio.Restriction.Restriction.Psp03I.size
Bio.Restriction.Restriction.Psp124BI.size
Bio.Restriction.Restriction.Psp1406I.size
Bio.Restriction.Restriction.Psp5II.size
Bio.Restriction.Restriction.Psp6I.size
Bio.Restriction.Restriction.PspCI.size
Bio.Restriction.Restriction.PspEI.size
Bio.Restriction.Restriction.PspGI.size
Bio.Restriction.Restriction.PspLI.size
Bio.Restriction.Restriction.PspN4I.size
Bio.Restriction.Restriction.PspOMI.size
Bio.Restriction.Restriction.PspOMII.size
Bio.Restriction.Restriction.PspPI.size
Bio.Restriction.Restriction.PspPPI.size
Bio.Restriction.Restriction.PspPRI.size
Bio.Restriction.Restriction.PspXI.size
Bio.Restriction.Restriction.PsrI.size
Bio.Restriction.Restriction.PssI.size
Bio.Restriction.Restriction.PstI.size
Bio.Restriction.Restriction.PstNI.size
Bio.Restriction.Restriction.PsuI.size
Bio.Restriction.Restriction.PsyI.size
Bio.Restriction.Restriction.PteI.size
Bio.Restriction.Restriction.PvuI.size
Bio.Restriction.Restriction.PvuII.size
Bio.Restriction.Restriction.R2_BceSIV.size
Bio.Restriction.Restriction.RceI.size
Bio.Restriction.Restriction.RdeGBI.size
Bio.Restriction.Restriction.RdeGBII.size
Bio.Restriction.Restriction.RdeGBIII.size
Bio.Restriction.Restriction.RflFIII.size
Bio.Restriction.Restriction.RgaI.size
Bio.Restriction.Restriction.RigI.size
Bio.Restriction.Restriction.RlaI.size
Bio.Restriction.Restriction.RleAI.size
Bio.Restriction.Restriction.RpaB5I.size
Bio.Restriction.Restriction.RpaBI.size
Bio.Restriction.Restriction.RpaI.size
Bio.Restriction.Restriction.RpaTI.size
Bio.Restriction.Restriction.RruI.size
Bio.Restriction.Restriction.RsaI.size
Bio.Restriction.Restriction.RsaNI.size
Bio.Restriction.Restriction.RseI.size
Bio.Restriction.Restriction.Rsr2I.size
Bio.Restriction.Restriction.RsrII.size
Bio.Restriction.Restriction.SacI.size
Bio.Restriction.Restriction.SacII.size
Bio.Restriction.Restriction.SalI.size
Bio.Restriction.Restriction.SanDI.size
Bio.Restriction.Restriction.SapI.size
Bio.Restriction.Restriction.SaqAI.size
Bio.Restriction.Restriction.SatI.size
Bio.Restriction.Restriction.Sau3AI.size
Bio.Restriction.Restriction.Sau96I.size
Bio.Restriction.Restriction.SauI.size
Bio.Restriction.Restriction.SbfI.size
Bio.Restriction.Restriction.ScaI.size
Bio.Restriction.Restriction.SchI.size
Bio.Restriction.Restriction.SciI.size
Bio.Restriction.Restriction.ScrFI.size
Bio.Restriction.Restriction.SdaI.size
Bio.Restriction.Restriction.SdeAI.size
Bio.Restriction.Restriction.SdeOSI.size
Bio.Restriction.Restriction.SduI.size
Bio.Restriction.Restriction.SecI.size
Bio.Restriction.Restriction.SelI.size
Bio.Restriction.Restriction.SetI.size
Bio.Restriction.Restriction.SexAI.size
Bio.Restriction.Restriction.SfaAI.size
Bio.Restriction.Restriction.SfaNI.size
Bio.Restriction.Restriction.SfcI.size
Bio.Restriction.Restriction.SfeI.size
Bio.Restriction.Restriction.SfiI.size
Bio.Restriction.Restriction.SfoI.size
Bio.Restriction.Restriction.Sfr274I.size
Bio.Restriction.Restriction.Sfr303I.size
Bio.Restriction.Restriction.SfuI.size
Bio.Restriction.Restriction.SgeI.size
Bio.Restriction.Restriction.SgfI.size
Bio.Restriction.Restriction.SgrAI.size
Bio.Restriction.Restriction.SgrBI.size
Bio.Restriction.Restriction.SgrDI.size
Bio.Restriction.Restriction.SgrTI.size
Bio.Restriction.Restriction.SgsI.size
Bio.Restriction.Restriction.SimI.size
Bio.Restriction.Restriction.SlaI.size
Bio.Restriction.Restriction.SmaI.size
Bio.Restriction.Restriction.SmiI.size
Bio.Restriction.Restriction.SmiMI.size
Bio.Restriction.Restriction.SmlI.size
Bio.Restriction.Restriction.SmoI.size
Bio.Restriction.Restriction.SnaBI.size
Bio.Restriction.Restriction.SnaI.size
Bio.Restriction.Restriction.Sno506I.size
Bio.Restriction.Restriction.SpeI.size
Bio.Restriction.Restriction.SphI.size
Bio.Restriction.Restriction.SplI.size
Bio.Restriction.Restriction.SpoDI.size
Bio.Restriction.Restriction.SrfI.size
Bio.Restriction.Restriction.Sse232I.size
Bio.Restriction.Restriction.Sse8387I.size
Bio.Restriction.Restriction.Sse8647I.size
Bio.Restriction.Restriction.Sse9I.size
Bio.Restriction.Restriction.SseBI.size
Bio.Restriction.Restriction.SsiI.size
Bio.Restriction.Restriction.SspD5I.size
Bio.Restriction.Restriction.SspDI.size
Bio.Restriction.Restriction.SspI.size
Bio.Restriction.Restriction.SstE37I.size
Bio.Restriction.Restriction.SstI.size
Bio.Restriction.Restriction.Sth132I.size
Bio.Restriction.Restriction.Sth302II.size
Bio.Restriction.Restriction.StrI.size
Bio.Restriction.Restriction.StsI.size
Bio.Restriction.Restriction.StuI.size
Bio.Restriction.Restriction.StyD4I.size
Bio.Restriction.Restriction.StyI.size
Bio.Restriction.Restriction.SwaI.size
Bio.Restriction.Restriction.TaaI.size
Bio.Restriction.Restriction.TaiI.size
Bio.Restriction.Restriction.TaqI.size
Bio.Restriction.Restriction.TaqII.size
Bio.Restriction.Restriction.TasI.size
Bio.Restriction.Restriction.TatI.size
Bio.Restriction.Restriction.TauI.size
Bio.Restriction.Restriction.TfiI.size
Bio.Restriction.Restriction.Tru1I.size
Bio.Restriction.Restriction.Tru9I.size
Bio.Restriction.Restriction.TscAI.size
Bio.Restriction.Restriction.TseFI.size
Bio.Restriction.Restriction.TseI.size
Bio.Restriction.Restriction.TsoI.size
Bio.Restriction.Restriction.Tsp45I.size
Bio.Restriction.Restriction.Tsp4CI.size
Bio.Restriction.Restriction.TspDTI.size
Bio.Restriction.Restriction.TspEI.size
Bio.Restriction.Restriction.TspGWI.size
Bio.Restriction.Restriction.TspMI.size
Bio.Restriction.Restriction.TspRI.size
Bio.Restriction.Restriction.TssI.size
Bio.Restriction.Restriction.TstI.size
Bio.Restriction.Restriction.TsuI.size
Bio.Restriction.Restriction.Tth111I.size
Bio.Restriction.Restriction.Tth111II.size
Bio.Restriction.Restriction.UbaF11I.size
Bio.Restriction.Restriction.UbaF12I.size
Bio.Restriction.Restriction.UbaF13I.size
Bio.Restriction.Restriction.UbaF14I.size
Bio.Restriction.Restriction.UbaF9I.size
Bio.Restriction.Restriction.UbaPI.size
Bio.Restriction.Restriction.UcoMSI.size
Bio.Restriction.Restriction.UnbI.size
Bio.Restriction.Restriction.Van91I.size
Bio.Restriction.Restriction.Vha464I.size
Bio.Restriction.Restriction.VneI.size
Bio.Restriction.Restriction.VpaK11AI.size
Bio.Restriction.Restriction.VpaK11BI.size
Bio.Restriction.Restriction.VspI.size
Bio.Restriction.Restriction.WviI.size
Bio.Restriction.Restriction.XagI.size
Bio.Restriction.Restriction.XapI.size
Bio.Restriction.Restriction.XbaI.size
Bio.Restriction.Restriction.XceI.size
Bio.Restriction.Restriction.XcmI.size
Bio.Restriction.Restriction.XhoI.size
Bio.Restriction.Restriction.XhoII.size
Bio.Restriction.Restriction.XmaI.size
Bio.Restriction.Restriction.XmaIII.size
Bio.Restriction.Restriction.XmaJI.size
Bio.Restriction.Restriction.XmiI.size
Bio.Restriction.Restriction.XmnI.size
Bio.Restriction.Restriction.XspI.size
Bio.Restriction.Restriction.YkrI.size
Bio.Restriction.Restriction.ZraI.size
Bio.Restriction.Restriction.ZrmI.size
Bio.Restriction.Restriction.Zsp2I.size
Bio.codonalign.codonalphabet.CodonAlphabet.size" class="py-name" href="#" onclick="return doclink('link-72', 'size', 'link-70');">size</a></tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L418"></a><tt class="py-lineno"> 418</tt>  <tt class="py-line">        <tt class="py-comment"># now any padding...</tt> </tt>
<a name="L419"></a><tt class="py-lineno"> 419</tt>  <tt class="py-line">        <tt class="py-name">padding</tt> <tt class="py-op">=</tt> <tt id="link-73" class="py-name"><a title="Bio.Alphabet.Alphabet.size
Bio.Alphabet.Reduced.HPModel.size
Bio.Alphabet.Reduced.Murphy10.size
Bio.Alphabet.Reduced.Murphy15.size
Bio.Alphabet.Reduced.Murphy4.size
Bio.Alphabet.Reduced.Murphy8.size
Bio.Alphabet.Reduced.PC5.size
Bio.Alphabet.SingleLetterAlphabet.size
Bio.Alphabet.ThreeLetterProtein.size
Bio.GenBank._FeatureConsumer.size
Bio.GenBank._RecordConsumer.size
Bio.Restriction.Restriction.AanI.size
Bio.Restriction.Restriction.AarI.size
Bio.Restriction.Restriction.AasI.size
Bio.Restriction.Restriction.AatII.size
Bio.Restriction.Restriction.AbaSI.size
Bio.Restriction.Restriction.AbsI.size
Bio.Restriction.Restriction.Acc16I.size
Bio.Restriction.Restriction.Acc36I.size
Bio.Restriction.Restriction.Acc65I.size
Bio.Restriction.Restriction.AccB1I.size
Bio.Restriction.Restriction.AccB7I.size
Bio.Restriction.Restriction.AccBSI.size
Bio.Restriction.Restriction.AccI.size
Bio.Restriction.Restriction.AccII.size
Bio.Restriction.Restriction.AccIII.size
Bio.Restriction.Restriction.AceIII.size
Bio.Restriction.Restriction.AciI.size
Bio.Restriction.Restriction.AclI.size
Bio.Restriction.Restriction.AclWI.size
Bio.Restriction.Restriction.AcoI.size
Bio.Restriction.Restriction.AcsI.size
Bio.Restriction.Restriction.AcuI.size
Bio.Restriction.Restriction.AcvI.size
Bio.Restriction.Restriction.AcyI.size
Bio.Restriction.Restriction.AdeI.size
Bio.Restriction.Restriction.AfaI.size
Bio.Restriction.Restriction.AfeI.size
Bio.Restriction.Restriction.AfiI.size
Bio.Restriction.Restriction.AflII.size
Bio.Restriction.Restriction.AflIII.size
Bio.Restriction.Restriction.AgeI.size
Bio.Restriction.Restriction.AgsI.size
Bio.Restriction.Restriction.AhaIII.size
Bio.Restriction.Restriction.AhdI.size
Bio.Restriction.Restriction.AhlI.size
Bio.Restriction.Restriction.AjiI.size
Bio.Restriction.Restriction.AjnI.size
Bio.Restriction.Restriction.AjuI.size
Bio.Restriction.Restriction.AleI.size
Bio.Restriction.Restriction.AlfI.size
Bio.Restriction.Restriction.AloI.size
Bio.Restriction.Restriction.AluBI.size
Bio.Restriction.Restriction.AluI.size
Bio.Restriction.Restriction.Alw21I.size
Bio.Restriction.Restriction.Alw26I.size
Bio.Restriction.Restriction.Alw44I.size
Bio.Restriction.Restriction.AlwFI.size
Bio.Restriction.Restriction.AlwI.size
Bio.Restriction.Restriction.AlwNI.size
Bio.Restriction.Restriction.Ama87I.size
Bio.Restriction.Restriction.Aor13HI.size
Bio.Restriction.Restriction.Aor51HI.size
Bio.Restriction.Restriction.AoxI.size
Bio.Restriction.Restriction.ApaBI.size
Bio.Restriction.Restriction.ApaI.size
Bio.Restriction.Restriction.ApaLI.size
Bio.Restriction.Restriction.ApeKI.size
Bio.Restriction.Restriction.ApoI.size
Bio.Restriction.Restriction.ApyPI.size
Bio.Restriction.Restriction.AquII.size
Bio.Restriction.Restriction.AquIII.size
Bio.Restriction.Restriction.AquIV.size
Bio.Restriction.Restriction.ArsI.size
Bio.Restriction.Restriction.AscI.size
Bio.Restriction.Restriction.AseI.size
Bio.Restriction.Restriction.Asi256I.size
Bio.Restriction.Restriction.AsiGI.size
Bio.Restriction.Restriction.AsiSI.size
Bio.Restriction.Restriction.Asp700I.size
Bio.Restriction.Restriction.Asp718I.size
Bio.Restriction.Restriction.AspA2I.size
Bio.Restriction.Restriction.AspBHI.size
Bio.Restriction.Restriction.AspLEI.size
Bio.Restriction.Restriction.AspS9I.size
Bio.Restriction.Restriction.AssI.size
Bio.Restriction.Restriction.AsuC2I.size
Bio.Restriction.Restriction.AsuHPI.size
Bio.Restriction.Restriction.AsuI.size
Bio.Restriction.Restriction.AsuII.size
Bio.Restriction.Restriction.AsuNHI.size
Bio.Restriction.Restriction.AvaI.size
Bio.Restriction.Restriction.AvaII.size
Bio.Restriction.Restriction.AvaIII.size
Bio.Restriction.Restriction.AvrII.size
Bio.Restriction.Restriction.AxyI.size
Bio.Restriction.Restriction.BaeGI.size
Bio.Restriction.Restriction.BaeI.size
Bio.Restriction.Restriction.BalI.size
Bio.Restriction.Restriction.BamHI.size
Bio.Restriction.Restriction.BanI.size
Bio.Restriction.Restriction.BanII.size
Bio.Restriction.Restriction.BarI.size
Bio.Restriction.Restriction.BasI.size
Bio.Restriction.Restriction.BauI.size
Bio.Restriction.Restriction.Bbr7I.size
Bio.Restriction.Restriction.BbrPI.size
Bio.Restriction.Restriction.BbsI.size
Bio.Restriction.Restriction.Bbv12I.size
Bio.Restriction.Restriction.BbvCI.size
Bio.Restriction.Restriction.BbvI.size
Bio.Restriction.Restriction.BbvII.size
Bio.Restriction.Restriction.BccI.size
Bio.Restriction.Restriction.Bce83I.size
Bio.Restriction.Restriction.BceAI.size
Bio.Restriction.Restriction.BcefI.size
Bio.Restriction.Restriction.BcgI.size
Bio.Restriction.Restriction.BciT130I.size
Bio.Restriction.Restriction.BciVI.size
Bio.Restriction.Restriction.BclI.size
Bio.Restriction.Restriction.BcnI.size
Bio.Restriction.Restriction.BcoDI.size
Bio.Restriction.Restriction.BcuI.size
Bio.Restriction.Restriction.BdaI.size
Bio.Restriction.Restriction.BetI.size
Bio.Restriction.Restriction.BfaI.size
Bio.Restriction.Restriction.BfiI.size
Bio.Restriction.Restriction.BfmI.size
Bio.Restriction.Restriction.BfoI.size
Bio.Restriction.Restriction.BfrI.size
Bio.Restriction.Restriction.BfuAI.size
Bio.Restriction.Restriction.BfuCI.size
Bio.Restriction.Restriction.BfuI.size
Bio.Restriction.Restriction.BglI.size
Bio.Restriction.Restriction.BglII.size
Bio.Restriction.Restriction.BinI.size
Bio.Restriction.Restriction.BisI.size
Bio.Restriction.Restriction.BlnI.size
Bio.Restriction.Restriction.BlpI.size
Bio.Restriction.Restriction.BlsI.size
Bio.Restriction.Restriction.BmcAI.size
Bio.Restriction.Restriction.Bme1390I.size
Bio.Restriction.Restriction.Bme18I.size
Bio.Restriction.Restriction.BmeDI.size
Bio.Restriction.Restriction.BmeRI.size
Bio.Restriction.Restriction.BmeT110I.size
Bio.Restriction.Restriction.BmgBI.size
Bio.Restriction.Restriction.BmgI.size
Bio.Restriction.Restriction.BmgT120I.size
Bio.Restriction.Restriction.BmiI.size
Bio.Restriction.Restriction.BmrFI.size
Bio.Restriction.Restriction.BmrI.size
Bio.Restriction.Restriction.BmsI.size
Bio.Restriction.Restriction.BmtI.size
Bio.Restriction.Restriction.BmuI.size
Bio.Restriction.Restriction.BoxI.size
Bio.Restriction.Restriction.BpiI.size
Bio.Restriction.Restriction.BplI.size
Bio.Restriction.Restriction.BpmI.size
Bio.Restriction.Restriction.Bpu10I.size
Bio.Restriction.Restriction.Bpu1102I.size
Bio.Restriction.Restriction.Bpu14I.size
Bio.Restriction.Restriction.BpuEI.size
Bio.Restriction.Restriction.BpuMI.size
Bio.Restriction.Restriction.BpvUI.size
Bio.Restriction.Restriction.Bsa29I.size
Bio.Restriction.Restriction.BsaAI.size
Bio.Restriction.Restriction.BsaBI.size
Bio.Restriction.Restriction.BsaHI.size
Bio.Restriction.Restriction.BsaI.size
Bio.Restriction.Restriction.BsaJI.size
Bio.Restriction.Restriction.BsaWI.size
Bio.Restriction.Restriction.BsaXI.size
Bio.Restriction.Restriction.BsbI.size
Bio.Restriction.Restriction.Bsc4I.size
Bio.Restriction.Restriction.BscAI.size
Bio.Restriction.Restriction.BscGI.size
Bio.Restriction.Restriction.Bse118I.size
Bio.Restriction.Restriction.Bse1I.size
Bio.Restriction.Restriction.Bse21I.size
Bio.Restriction.Restriction.Bse3DI.size
Bio.Restriction.Restriction.Bse8I.size
Bio.Restriction.Restriction.BseAI.size
Bio.Restriction.Restriction.BseBI.size
Bio.Restriction.Restriction.BseCI.size
Bio.Restriction.Restriction.BseDI.size
Bio.Restriction.Restriction.BseGI.size
Bio.Restriction.Restriction.BseJI.size
Bio.Restriction.Restriction.BseLI.size
Bio.Restriction.Restriction.BseMI.size
Bio.Restriction.Restriction.BseMII.size
Bio.Restriction.Restriction.BseNI.size
Bio.Restriction.Restriction.BsePI.size
Bio.Restriction.Restriction.BseRI.size
Bio.Restriction.Restriction.BseSI.size
Bio.Restriction.Restriction.BseX3I.size
Bio.Restriction.Restriction.BseXI.size
Bio.Restriction.Restriction.BseYI.size
Bio.Restriction.Restriction.BsgI.size
Bio.Restriction.Restriction.Bsh1236I.size
Bio.Restriction.Restriction.Bsh1285I.size
Bio.Restriction.Restriction.BshFI.size
Bio.Restriction.Restriction.BshNI.size
Bio.Restriction.Restriction.BshTI.size
Bio.Restriction.Restriction.BshVI.size
Bio.Restriction.Restriction.BsiEI.size
Bio.Restriction.Restriction.BsiHKAI.size
Bio.Restriction.Restriction.BsiHKCI.size
Bio.Restriction.Restriction.BsiI.size
Bio.Restriction.Restriction.BsiSI.size
Bio.Restriction.Restriction.BsiWI.size
Bio.Restriction.Restriction.BsiYI.size
Bio.Restriction.Restriction.BslFI.size
Bio.Restriction.Restriction.BslI.size
Bio.Restriction.Restriction.BsmAI.size
Bio.Restriction.Restriction.BsmBI.size
Bio.Restriction.Restriction.BsmFI.size
Bio.Restriction.Restriction.BsmI.size
Bio.Restriction.Restriction.BsnI.size
Bio.Restriction.Restriction.Bso31I.size
Bio.Restriction.Restriction.BsoBI.size
Bio.Restriction.Restriction.Bsp119I.size
Bio.Restriction.Restriction.Bsp120I.size
Bio.Restriction.Restriction.Bsp1286I.size
Bio.Restriction.Restriction.Bsp13I.size
Bio.Restriction.Restriction.Bsp1407I.size
Bio.Restriction.Restriction.Bsp143I.size
Bio.Restriction.Restriction.Bsp1720I.size
Bio.Restriction.Restriction.Bsp19I.size
Bio.Restriction.Restriction.Bsp24I.size
Bio.Restriction.Restriction.Bsp68I.size
Bio.Restriction.Restriction.BspACI.size
Bio.Restriction.Restriction.BspCNI.size
Bio.Restriction.Restriction.BspD6I.size
Bio.Restriction.Restriction.BspDI.size
Bio.Restriction.Restriction.BspEI.size
Bio.Restriction.Restriction.BspFNI.size
Bio.Restriction.Restriction.BspGI.size
Bio.Restriction.Restriction.BspHI.size
Bio.Restriction.Restriction.BspLI.size
Bio.Restriction.Restriction.BspLU11I.size
Bio.Restriction.Restriction.BspMI.size
Bio.Restriction.Restriction.BspMII.size
Bio.Restriction.Restriction.BspNCI.size
Bio.Restriction.Restriction.BspOI.size
Bio.Restriction.Restriction.BspPI.size
Bio.Restriction.Restriction.BspQI.size
Bio.Restriction.Restriction.BspT104I.size
Bio.Restriction.Restriction.BspT107I.size
Bio.Restriction.Restriction.BspTI.size
Bio.Restriction.Restriction.BsrBI.size
Bio.Restriction.Restriction.BsrDI.size
Bio.Restriction.Restriction.BsrFI.size
Bio.Restriction.Restriction.BsrGI.size
Bio.Restriction.Restriction.BsrI.size
Bio.Restriction.Restriction.BsrSI.size
Bio.Restriction.Restriction.BssAI.size
Bio.Restriction.Restriction.BssECI.size
Bio.Restriction.Restriction.BssHII.size
Bio.Restriction.Restriction.BssKI.size
Bio.Restriction.Restriction.BssMI.size
Bio.Restriction.Restriction.BssNAI.size
Bio.Restriction.Restriction.BssNI.size
Bio.Restriction.Restriction.BssSI.size
Bio.Restriction.Restriction.BssT1I.size
Bio.Restriction.Restriction.Bst1107I.size
Bio.Restriction.Restriction.Bst2BI.size
Bio.Restriction.Restriction.Bst2UI.size
Bio.Restriction.Restriction.Bst4CI.size
Bio.Restriction.Restriction.Bst6I.size
Bio.Restriction.Restriction.BstACI.size
Bio.Restriction.Restriction.BstAFI.size
Bio.Restriction.Restriction.BstAPI.size
Bio.Restriction.Restriction.BstAUI.size
Bio.Restriction.Restriction.BstBAI.size
Bio.Restriction.Restriction.BstBI.size
Bio.Restriction.Restriction.BstC8I.size
Bio.Restriction.Restriction.BstDEI.size
Bio.Restriction.Restriction.BstDSI.size
Bio.Restriction.Restriction.BstEII.size
Bio.Restriction.Restriction.BstENI.size
Bio.Restriction.Restriction.BstF5I.size
Bio.Restriction.Restriction.BstFNI.size
Bio.Restriction.Restriction.BstH2I.size
Bio.Restriction.Restriction.BstHHI.size
Bio.Restriction.Restriction.BstKTI.size
Bio.Restriction.Restriction.BstMAI.size
Bio.Restriction.Restriction.BstMBI.size
Bio.Restriction.Restriction.BstMCI.size
Bio.Restriction.Restriction.BstMWI.size
Bio.Restriction.Restriction.BstNI.size
Bio.Restriction.Restriction.BstNSI.size
Bio.Restriction.Restriction.BstOI.size
Bio.Restriction.Restriction.BstPAI.size
Bio.Restriction.Restriction.BstPI.size
Bio.Restriction.Restriction.BstSCI.size
Bio.Restriction.Restriction.BstSFI.size
Bio.Restriction.Restriction.BstSLI.size
Bio.Restriction.Restriction.BstSNI.size
Bio.Restriction.Restriction.BstUI.size
Bio.Restriction.Restriction.BstV1I.size
Bio.Restriction.Restriction.BstV2I.size
Bio.Restriction.Restriction.BstX2I.size
Bio.Restriction.Restriction.BstXI.size
Bio.Restriction.Restriction.BstYI.size
Bio.Restriction.Restriction.BstZ17I.size
Bio.Restriction.Restriction.BstZI.size
Bio.Restriction.Restriction.Bsu15I.size
Bio.Restriction.Restriction.Bsu36I.size
Bio.Restriction.Restriction.BsuI.size
Bio.Restriction.Restriction.BsuRI.size
Bio.Restriction.Restriction.BtgI.size
Bio.Restriction.Restriction.BtgZI.size
Bio.Restriction.Restriction.BthCI.size
Bio.Restriction.Restriction.BtrI.size
Bio.Restriction.Restriction.BtsCI.size
Bio.Restriction.Restriction.BtsI.size
Bio.Restriction.Restriction.BtsIMutI.size
Bio.Restriction.Restriction.BtuMI.size
Bio.Restriction.Restriction.BveI.size
Bio.Restriction.Restriction.Cac8I.size
Bio.Restriction.Restriction.CaiI.size
Bio.Restriction.Restriction.CauII.size
Bio.Restriction.Restriction.CchII.size
Bio.Restriction.Restriction.CchIII.size
Bio.Restriction.Restriction.CciI.size
Bio.Restriction.Restriction.CciNI.size
Bio.Restriction.Restriction.Cdi630V.size
Bio.Restriction.Restriction.CdiI.size
Bio.Restriction.Restriction.CdpI.size
Bio.Restriction.Restriction.CfoI.size
Bio.Restriction.Restriction.Cfr10I.size
Bio.Restriction.Restriction.Cfr13I.size
Bio.Restriction.Restriction.Cfr42I.size
Bio.Restriction.Restriction.Cfr9I.size
Bio.Restriction.Restriction.CfrI.size
Bio.Restriction.Restriction.Cgl13032I.size
Bio.Restriction.Restriction.Cgl13032II.size
Bio.Restriction.Restriction.ChaI.size
Bio.Restriction.Restriction.CjeFIII.size
Bio.Restriction.Restriction.CjeFV.size
Bio.Restriction.Restriction.CjeI.size
Bio.Restriction.Restriction.CjeNII.size
Bio.Restriction.Restriction.CjeNIII.size
Bio.Restriction.Restriction.CjeP659IV.size
Bio.Restriction.Restriction.CjePI.size
Bio.Restriction.Restriction.CjuI.size
Bio.Restriction.Restriction.CjuII.size
Bio.Restriction.Restriction.ClaI.size
Bio.Restriction.Restriction.CpoI.size
Bio.Restriction.Restriction.CseI.size
Bio.Restriction.Restriction.CsiI.size
Bio.Restriction.Restriction.Csp6I.size
Bio.Restriction.Restriction.CspAI.size
Bio.Restriction.Restriction.CspCI.size
Bio.Restriction.Restriction.CspI.size
Bio.Restriction.Restriction.CstMI.size
Bio.Restriction.Restriction.CviAII.size
Bio.Restriction.Restriction.CviJI.size
Bio.Restriction.Restriction.CviKI_1.size
Bio.Restriction.Restriction.CviQI.size
Bio.Restriction.Restriction.CviRI.size
Bio.Restriction.Restriction.DdeI.size
Bio.Restriction.Restriction.DinI.size
Bio.Restriction.Restriction.DpnI.size
Bio.Restriction.Restriction.DpnII.size
Bio.Restriction.Restriction.DraI.size
Bio.Restriction.Restriction.DraII.size
Bio.Restriction.Restriction.DraIII.size
Bio.Restriction.Restriction.DraRI.size
Bio.Restriction.Restriction.DrdI.size
Bio.Restriction.Restriction.DrdII.size
Bio.Restriction.Restriction.DriI.size
Bio.Restriction.Restriction.DsaI.size
Bio.Restriction.Restriction.DseDI.size
Bio.Restriction.Restriction.EaeI.size
Bio.Restriction.Restriction.EagI.size
Bio.Restriction.Restriction.Eam1104I.size
Bio.Restriction.Restriction.Eam1105I.size
Bio.Restriction.Restriction.EarI.size
Bio.Restriction.Restriction.EciI.size
Bio.Restriction.Restriction.Ecl136II.size
Bio.Restriction.Restriction.EclXI.size
Bio.Restriction.Restriction.Eco105I.size
Bio.Restriction.Restriction.Eco130I.size
Bio.Restriction.Restriction.Eco147I.size
Bio.Restriction.Restriction.Eco24I.size
Bio.Restriction.Restriction.Eco31I.size
Bio.Restriction.Restriction.Eco32I.size
Bio.Restriction.Restriction.Eco47I.size
Bio.Restriction.Restriction.Eco47III.size
Bio.Restriction.Restriction.Eco52I.size
Bio.Restriction.Restriction.Eco53kI.size
Bio.Restriction.Restriction.Eco57I.size
Bio.Restriction.Restriction.Eco57MI.size
Bio.Restriction.Restriction.Eco72I.size
Bio.Restriction.Restriction.Eco81I.size
Bio.Restriction.Restriction.Eco88I.size
Bio.Restriction.Restriction.Eco91I.size
Bio.Restriction.Restriction.EcoHI.size
Bio.Restriction.Restriction.EcoICRI.size
Bio.Restriction.Restriction.EcoNI.size
Bio.Restriction.Restriction.EcoO109I.size
Bio.Restriction.Restriction.EcoO65I.size
Bio.Restriction.Restriction.EcoRI.size
Bio.Restriction.Restriction.EcoRII.size
Bio.Restriction.Restriction.EcoRV.size
Bio.Restriction.Restriction.EcoT14I.size
Bio.Restriction.Restriction.EcoT22I.size
Bio.Restriction.Restriction.EcoT38I.size
Bio.Restriction.Restriction.EgeI.size
Bio.Restriction.Restriction.EheI.size
Bio.Restriction.Restriction.ErhI.size
Bio.Restriction.Restriction.EsaBC3I.size
Bio.Restriction.Restriction.EsaSSI.size
Bio.Restriction.Restriction.Esp3I.size
Bio.Restriction.Restriction.EspI.size
Bio.Restriction.Restriction.FaeI.size
Bio.Restriction.Restriction.FaiI.size
Bio.Restriction.Restriction.FalI.size
Bio.Restriction.Restriction.FaqI.size
Bio.Restriction.Restriction.FatI.size
Bio.Restriction.Restriction.FauI.size
Bio.Restriction.Restriction.FauNDI.size
Bio.Restriction.Restriction.FbaI.size
Bio.Restriction.Restriction.FblI.size
Bio.Restriction.Restriction.FinI.size
Bio.Restriction.Restriction.FmuI.size
Bio.Restriction.Restriction.Fnu4HI.size
Bio.Restriction.Restriction.FnuDII.size
Bio.Restriction.Restriction.FokI.size
Bio.Restriction.Restriction.FriOI.size
Bio.Restriction.Restriction.FseI.size
Bio.Restriction.Restriction.Fsp4HI.size
Bio.Restriction.Restriction.FspAI.size
Bio.Restriction.Restriction.FspBI.size
Bio.Restriction.Restriction.FspEI.size
Bio.Restriction.Restriction.FspI.size
Bio.Restriction.Restriction.GauT27I.size
Bio.Restriction.Restriction.GdiII.size
Bio.Restriction.Restriction.GlaI.size
Bio.Restriction.Restriction.GluI.size
Bio.Restriction.Restriction.GsaI.size
Bio.Restriction.Restriction.GsuI.size
Bio.Restriction.Restriction.HaeI.size
Bio.Restriction.Restriction.HaeII.size
Bio.Restriction.Restriction.HaeIII.size
Bio.Restriction.Restriction.HapII.size
Bio.Restriction.Restriction.HauII.size
Bio.Restriction.Restriction.HgaI.size
Bio.Restriction.Restriction.HgiAI.size
Bio.Restriction.Restriction.HgiCI.size
Bio.Restriction.Restriction.HgiEII.size
Bio.Restriction.Restriction.HgiJII.size
Bio.Restriction.Restriction.HhaI.size
Bio.Restriction.Restriction.Hin1I.size
Bio.Restriction.Restriction.Hin1II.size
Bio.Restriction.Restriction.Hin4I.size
Bio.Restriction.Restriction.Hin4II.size
Bio.Restriction.Restriction.Hin6I.size
Bio.Restriction.Restriction.HinP1I.size
Bio.Restriction.Restriction.HincII.size
Bio.Restriction.Restriction.HindII.size
Bio.Restriction.Restriction.HindIII.size
Bio.Restriction.Restriction.HinfI.size
Bio.Restriction.Restriction.HpaI.size
Bio.Restriction.Restriction.HpaII.size
Bio.Restriction.Restriction.HphI.size
Bio.Restriction.Restriction.Hpy166II.size
Bio.Restriction.Restriction.Hpy178III.size
Bio.Restriction.Restriction.Hpy188I.size
Bio.Restriction.Restriction.Hpy188III.size
Bio.Restriction.Restriction.Hpy8I.size
Bio.Restriction.Restriction.Hpy99I.size
Bio.Restriction.Restriction.Hpy99XIII.size
Bio.Restriction.Restriction.Hpy99XIV.size
Bio.Restriction.Restriction.HpyAV.size
Bio.Restriction.Restriction.HpyCH4III.size
Bio.Restriction.Restriction.HpyCH4IV.size
Bio.Restriction.Restriction.HpyCH4V.size
Bio.Restriction.Restriction.HpyF10VI.size
Bio.Restriction.Restriction.HpyF3I.size
Bio.Restriction.Restriction.HpySE526I.size
Bio.Restriction.Restriction.Hsp92I.size
Bio.Restriction.Restriction.Hsp92II.size
Bio.Restriction.Restriction.HspAI.size
Bio.Restriction.Restriction.Jma19592I.size
Bio.Restriction.Restriction.KasI.size
Bio.Restriction.Restriction.KflI.size
Bio.Restriction.Restriction.Kpn2I.size
Bio.Restriction.Restriction.KpnI.size
Bio.Restriction.Restriction.KroI.size
Bio.Restriction.Restriction.Ksp22I.size
Bio.Restriction.Restriction.Ksp632I.size
Bio.Restriction.Restriction.KspAI.size
Bio.Restriction.Restriction.KspI.size
Bio.Restriction.Restriction.Kzo9I.size
Bio.Restriction.Restriction.LguI.size
Bio.Restriction.Restriction.LpnI.size
Bio.Restriction.Restriction.LpnPI.size
Bio.Restriction.Restriction.Lsp1109I.size
Bio.Restriction.Restriction.LweI.size
Bio.Restriction.Restriction.MabI.size
Bio.Restriction.Restriction.MaeI.size
Bio.Restriction.Restriction.MaeII.size
Bio.Restriction.Restriction.MaeIII.size
Bio.Restriction.Restriction.MalI.size
Bio.Restriction.Restriction.MaqI.size
Bio.Restriction.Restriction.MauBI.size
Bio.Restriction.Restriction.MbiI.size
Bio.Restriction.Restriction.MboI.size
Bio.Restriction.Restriction.MboII.size
Bio.Restriction.Restriction.McaTI.size
Bio.Restriction.Restriction.McrI.size
Bio.Restriction.Restriction.MfeI.size
Bio.Restriction.Restriction.MflI.size
Bio.Restriction.Restriction.MhlI.size
Bio.Restriction.Restriction.MjaIV.size
Bio.Restriction.Restriction.MkaDII.size
Bio.Restriction.Restriction.MlsI.size
Bio.Restriction.Restriction.MluCI.size
Bio.Restriction.Restriction.MluI.size
Bio.Restriction.Restriction.MluNI.size
Bio.Restriction.Restriction.Mly113I.size
Bio.Restriction.Restriction.MlyI.size
Bio.Restriction.Restriction.MmeI.size
Bio.Restriction.Restriction.MnlI.size
Bio.Restriction.Restriction.Mph1103I.size
Bio.Restriction.Restriction.MreI.size
Bio.Restriction.Restriction.MroI.size
Bio.Restriction.Restriction.MroNI.size
Bio.Restriction.Restriction.MroXI.size
Bio.Restriction.Restriction.MscI.size
Bio.Restriction.Restriction.MseI.size
Bio.Restriction.Restriction.MslI.size
Bio.Restriction.Restriction.Msp20I.size
Bio.Restriction.Restriction.MspA1I.size
Bio.Restriction.Restriction.MspCI.size
Bio.Restriction.Restriction.MspI.size
Bio.Restriction.Restriction.MspJI.size
Bio.Restriction.Restriction.MspR9I.size
Bio.Restriction.Restriction.MssI.size
Bio.Restriction.Restriction.MstI.size
Bio.Restriction.Restriction.MunI.size
Bio.Restriction.Restriction.Mva1269I.size
Bio.Restriction.Restriction.MvaI.size
Bio.Restriction.Restriction.MvnI.size
Bio.Restriction.Restriction.MvrI.size
Bio.Restriction.Restriction.MwoI.size
Bio.Restriction.Restriction.NaeI.size
Bio.Restriction.Restriction.NarI.size
Bio.Restriction.Restriction.NciI.size
Bio.Restriction.Restriction.NcoI.size
Bio.Restriction.Restriction.NdeI.size
Bio.Restriction.Restriction.NdeII.size
Bio.Restriction.Restriction.NgoAVIII.size
Bio.Restriction.Restriction.NgoMIV.size
Bio.Restriction.Restriction.NhaXI.size
Bio.Restriction.Restriction.NheI.size
Bio.Restriction.Restriction.NlaCI.size
Bio.Restriction.Restriction.NlaIII.size
Bio.Restriction.Restriction.NlaIV.size
Bio.Restriction.Restriction.Nli3877I.size
Bio.Restriction.Restriction.NmeAIII.size
Bio.Restriction.Restriction.NmeDI.size
Bio.Restriction.Restriction.NmuCI.size
Bio.Restriction.Restriction.NotI.size
Bio.Restriction.Restriction.NruI.size
Bio.Restriction.Restriction.NsbI.size
Bio.Restriction.Restriction.NsiI.size
Bio.Restriction.Restriction.NspBII.size
Bio.Restriction.Restriction.NspI.size
Bio.Restriction.Restriction.NspV.size
Bio.Restriction.Restriction.OliI.size
Bio.Restriction.Restriction.PabI.size
Bio.Restriction.Restriction.PacI.size
Bio.Restriction.Restriction.PaeI.size
Bio.Restriction.Restriction.PaeR7I.size
Bio.Restriction.Restriction.PagI.size
Bio.Restriction.Restriction.PalAI.size
Bio.Restriction.Restriction.PasI.size
Bio.Restriction.Restriction.PauI.size
Bio.Restriction.Restriction.PceI.size
Bio.Restriction.Restriction.PciI.size
Bio.Restriction.Restriction.PciSI.size
Bio.Restriction.Restriction.PcsI.size
Bio.Restriction.Restriction.PctI.size
Bio.Restriction.Restriction.PdiI.size
Bio.Restriction.Restriction.PdmI.size
Bio.Restriction.Restriction.PenI.size
Bio.Restriction.Restriction.PfeI.size
Bio.Restriction.Restriction.Pfl1108I.size
Bio.Restriction.Restriction.Pfl23II.size
Bio.Restriction.Restriction.PflFI.size
Bio.Restriction.Restriction.PflMI.size
Bio.Restriction.Restriction.PfoI.size
Bio.Restriction.Restriction.PinAI.size
Bio.Restriction.Restriction.PlaDI.size
Bio.Restriction.Restriction.Ple19I.size
Bio.Restriction.Restriction.PleI.size
Bio.Restriction.Restriction.PluTI.size
Bio.Restriction.Restriction.PmaCI.size
Bio.Restriction.Restriction.PmeI.size
Bio.Restriction.Restriction.PmlI.size
Bio.Restriction.Restriction.PpiI.size
Bio.Restriction.Restriction.PpsI.size
Bio.Restriction.Restriction.Ppu10I.size
Bio.Restriction.Restriction.Ppu21I.size
Bio.Restriction.Restriction.PpuMI.size
Bio.Restriction.Restriction.PscI.size
Bio.Restriction.Restriction.PshAI.size
Bio.Restriction.Restriction.PshBI.size
Bio.Restriction.Restriction.PsiI.size
Bio.Restriction.Restriction.Psp03I.size
Bio.Restriction.Restriction.Psp124BI.size
Bio.Restriction.Restriction.Psp1406I.size
Bio.Restriction.Restriction.Psp5II.size
Bio.Restriction.Restriction.Psp6I.size
Bio.Restriction.Restriction.PspCI.size
Bio.Restriction.Restriction.PspEI.size
Bio.Restriction.Restriction.PspGI.size
Bio.Restriction.Restriction.PspLI.size
Bio.Restriction.Restriction.PspN4I.size
Bio.Restriction.Restriction.PspOMI.size
Bio.Restriction.Restriction.PspOMII.size
Bio.Restriction.Restriction.PspPI.size
Bio.Restriction.Restriction.PspPPI.size
Bio.Restriction.Restriction.PspPRI.size
Bio.Restriction.Restriction.PspXI.size
Bio.Restriction.Restriction.PsrI.size
Bio.Restriction.Restriction.PssI.size
Bio.Restriction.Restriction.PstI.size
Bio.Restriction.Restriction.PstNI.size
Bio.Restriction.Restriction.PsuI.size
Bio.Restriction.Restriction.PsyI.size
Bio.Restriction.Restriction.PteI.size
Bio.Restriction.Restriction.PvuI.size
Bio.Restriction.Restriction.PvuII.size
Bio.Restriction.Restriction.R2_BceSIV.size
Bio.Restriction.Restriction.RceI.size
Bio.Restriction.Restriction.RdeGBI.size
Bio.Restriction.Restriction.RdeGBII.size
Bio.Restriction.Restriction.RdeGBIII.size
Bio.Restriction.Restriction.RflFIII.size
Bio.Restriction.Restriction.RgaI.size
Bio.Restriction.Restriction.RigI.size
Bio.Restriction.Restriction.RlaI.size
Bio.Restriction.Restriction.RleAI.size
Bio.Restriction.Restriction.RpaB5I.size
Bio.Restriction.Restriction.RpaBI.size
Bio.Restriction.Restriction.RpaI.size
Bio.Restriction.Restriction.RpaTI.size
Bio.Restriction.Restriction.RruI.size
Bio.Restriction.Restriction.RsaI.size
Bio.Restriction.Restriction.RsaNI.size
Bio.Restriction.Restriction.RseI.size
Bio.Restriction.Restriction.Rsr2I.size
Bio.Restriction.Restriction.RsrII.size
Bio.Restriction.Restriction.SacI.size
Bio.Restriction.Restriction.SacII.size
Bio.Restriction.Restriction.SalI.size
Bio.Restriction.Restriction.SanDI.size
Bio.Restriction.Restriction.SapI.size
Bio.Restriction.Restriction.SaqAI.size
Bio.Restriction.Restriction.SatI.size
Bio.Restriction.Restriction.Sau3AI.size
Bio.Restriction.Restriction.Sau96I.size
Bio.Restriction.Restriction.SauI.size
Bio.Restriction.Restriction.SbfI.size
Bio.Restriction.Restriction.ScaI.size
Bio.Restriction.Restriction.SchI.size
Bio.Restriction.Restriction.SciI.size
Bio.Restriction.Restriction.ScrFI.size
Bio.Restriction.Restriction.SdaI.size
Bio.Restriction.Restriction.SdeAI.size
Bio.Restriction.Restriction.SdeOSI.size
Bio.Restriction.Restriction.SduI.size
Bio.Restriction.Restriction.SecI.size
Bio.Restriction.Restriction.SelI.size
Bio.Restriction.Restriction.SetI.size
Bio.Restriction.Restriction.SexAI.size
Bio.Restriction.Restriction.SfaAI.size
Bio.Restriction.Restriction.SfaNI.size
Bio.Restriction.Restriction.SfcI.size
Bio.Restriction.Restriction.SfeI.size
Bio.Restriction.Restriction.SfiI.size
Bio.Restriction.Restriction.SfoI.size
Bio.Restriction.Restriction.Sfr274I.size
Bio.Restriction.Restriction.Sfr303I.size
Bio.Restriction.Restriction.SfuI.size
Bio.Restriction.Restriction.SgeI.size
Bio.Restriction.Restriction.SgfI.size
Bio.Restriction.Restriction.SgrAI.size
Bio.Restriction.Restriction.SgrBI.size
Bio.Restriction.Restriction.SgrDI.size
Bio.Restriction.Restriction.SgrTI.size
Bio.Restriction.Restriction.SgsI.size
Bio.Restriction.Restriction.SimI.size
Bio.Restriction.Restriction.SlaI.size
Bio.Restriction.Restriction.SmaI.size
Bio.Restriction.Restriction.SmiI.size
Bio.Restriction.Restriction.SmiMI.size
Bio.Restriction.Restriction.SmlI.size
Bio.Restriction.Restriction.SmoI.size
Bio.Restriction.Restriction.SnaBI.size
Bio.Restriction.Restriction.SnaI.size
Bio.Restriction.Restriction.Sno506I.size
Bio.Restriction.Restriction.SpeI.size
Bio.Restriction.Restriction.SphI.size
Bio.Restriction.Restriction.SplI.size
Bio.Restriction.Restriction.SpoDI.size
Bio.Restriction.Restriction.SrfI.size
Bio.Restriction.Restriction.Sse232I.size
Bio.Restriction.Restriction.Sse8387I.size
Bio.Restriction.Restriction.Sse8647I.size
Bio.Restriction.Restriction.Sse9I.size
Bio.Restriction.Restriction.SseBI.size
Bio.Restriction.Restriction.SsiI.size
Bio.Restriction.Restriction.SspD5I.size
Bio.Restriction.Restriction.SspDI.size
Bio.Restriction.Restriction.SspI.size
Bio.Restriction.Restriction.SstE37I.size
Bio.Restriction.Restriction.SstI.size
Bio.Restriction.Restriction.Sth132I.size
Bio.Restriction.Restriction.Sth302II.size
Bio.Restriction.Restriction.StrI.size
Bio.Restriction.Restriction.StsI.size
Bio.Restriction.Restriction.StuI.size
Bio.Restriction.Restriction.StyD4I.size
Bio.Restriction.Restriction.StyI.size
Bio.Restriction.Restriction.SwaI.size
Bio.Restriction.Restriction.TaaI.size
Bio.Restriction.Restriction.TaiI.size
Bio.Restriction.Restriction.TaqI.size
Bio.Restriction.Restriction.TaqII.size
Bio.Restriction.Restriction.TasI.size
Bio.Restriction.Restriction.TatI.size
Bio.Restriction.Restriction.TauI.size
Bio.Restriction.Restriction.TfiI.size
Bio.Restriction.Restriction.Tru1I.size
Bio.Restriction.Restriction.Tru9I.size
Bio.Restriction.Restriction.TscAI.size
Bio.Restriction.Restriction.TseFI.size
Bio.Restriction.Restriction.TseI.size
Bio.Restriction.Restriction.TsoI.size
Bio.Restriction.Restriction.Tsp45I.size
Bio.Restriction.Restriction.Tsp4CI.size
Bio.Restriction.Restriction.TspDTI.size
Bio.Restriction.Restriction.TspEI.size
Bio.Restriction.Restriction.TspGWI.size
Bio.Restriction.Restriction.TspMI.size
Bio.Restriction.Restriction.TspRI.size
Bio.Restriction.Restriction.TssI.size
Bio.Restriction.Restriction.TstI.size
Bio.Restriction.Restriction.TsuI.size
Bio.Restriction.Restriction.Tth111I.size
Bio.Restriction.Restriction.Tth111II.size
Bio.Restriction.Restriction.UbaF11I.size
Bio.Restriction.Restriction.UbaF12I.size
Bio.Restriction.Restriction.UbaF13I.size
Bio.Restriction.Restriction.UbaF14I.size
Bio.Restriction.Restriction.UbaF9I.size
Bio.Restriction.Restriction.UbaPI.size
Bio.Restriction.Restriction.UcoMSI.size
Bio.Restriction.Restriction.UnbI.size
Bio.Restriction.Restriction.Van91I.size
Bio.Restriction.Restriction.Vha464I.size
Bio.Restriction.Restriction.VneI.size
Bio.Restriction.Restriction.VpaK11AI.size
Bio.Restriction.Restriction.VpaK11BI.size
Bio.Restriction.Restriction.VspI.size
Bio.Restriction.Restriction.WviI.size
Bio.Restriction.Restriction.XagI.size
Bio.Restriction.Restriction.XapI.size
Bio.Restriction.Restriction.XbaI.size
Bio.Restriction.Restriction.XceI.size
Bio.Restriction.Restriction.XcmI.size
Bio.Restriction.Restriction.XhoI.size
Bio.Restriction.Restriction.XhoII.size
Bio.Restriction.Restriction.XmaI.size
Bio.Restriction.Restriction.XmaIII.size
Bio.Restriction.Restriction.XmaJI.size
Bio.Restriction.Restriction.XmiI.size
Bio.Restriction.Restriction.XmnI.size
Bio.Restriction.Restriction.XspI.size
Bio.Restriction.Restriction.YkrI.size
Bio.Restriction.Restriction.ZraI.size
Bio.Restriction.Restriction.ZrmI.size
Bio.Restriction.Restriction.Zsp2I.size
Bio.codonalign.codonalphabet.CodonAlphabet.size" class="py-name" href="#" onclick="return doclink('link-73', 'size', 'link-70');">size</a></tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt> </tt>
<a name="L420"></a><tt class="py-lineno"> 420</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">padding</tt><tt class="py-op">:</tt> </tt>
<a name="L421"></a><tt class="py-lineno"> 421</tt>  <tt class="py-line">            <tt class="py-name">padding</tt> <tt class="py-op">=</tt> <tt class="py-number">8</tt> <tt class="py-op">-</tt> <tt class="py-name">padding</tt> </tt>
<a name="L422"></a><tt class="py-lineno"> 422</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-74" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-74', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">padding</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-75" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-75', 'count', 'link-45');">count</a></tt><tt class="py-op">(</tt><tt id="link-76" class="py-name"><a title="Bio.SeqIO.SffIO._null" class="py-name" href="#" onclick="return doclink('link-76', '_null', 'link-16');">_null</a></tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">padding</tt><tt class="py-op">:</tt> </tt>
<a name="L423"></a><tt class="py-lineno"> 423</tt>  <tt class="py-line">                <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L424"></a><tt class="py-lineno"> 424</tt>  <tt class="py-line">                <tt class="py-keyword">from</tt> <tt id="link-77" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-77', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-78" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-78', 'BiopythonParserWarning', 'link-48');">BiopythonParserWarning</a></tt> </tt>
<a name="L425"></a><tt class="py-lineno"> 425</tt>  <tt class="py-line">                <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Your SFF file is invalid, post quality %i "</tt> </tt>
<a name="L426"></a><tt class="py-lineno"> 426</tt>  <tt class="py-line">                              <tt class="py-string">"byte padding region contained data"</tt> <tt class="py-op">%</tt> <tt class="py-name">padding</tt><tt class="py-op">,</tt> </tt>
<a name="L427"></a><tt class="py-lineno"> 427</tt>  <tt class="py-line">                              <tt id="link-79" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-79', 'BiopythonParserWarning', 'link-48');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L428"></a><tt class="py-lineno"> 428</tt>  <tt class="py-line">        <tt class="py-comment"># print("%s %s %i" % (read, name, record_offset))</tt> </tt>
<a name="L429"></a><tt class="py-lineno"> 429</tt>  <tt class="py-line">        <tt class="py-keyword">yield</tt> <tt id="link-80" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-80', 'name', 'link-60');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">record_offset</tt> </tt>
<a name="L430"></a><tt class="py-lineno"> 430</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-81" class="py-name"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-81', 'tell', 'link-54');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt> <tt class="py-op">!=</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L431"></a><tt class="py-lineno"> 431</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt> </tt>
<a name="L432"></a><tt class="py-lineno"> 432</tt>  <tt class="py-line">            <tt class="py-string">"After scanning reads, did not end on a multiple of 8"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L433"></a><tt class="py-lineno"> 433</tt>  <tt class="py-line"> </tt>
<a name="L434"></a><tt class="py-lineno"> 434</tt>  <tt class="py-line"> </tt>
<a name="_sff_find_roche_index"></a><div id="_sff_find_roche_index-def"><a name="L435"></a><tt class="py-lineno"> 435</tt> <a class="py-toggle" href="#" id="_sff_find_roche_index-toggle" onclick="return toggle('_sff_find_roche_index');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO-module.html#_sff_find_roche_index">_sff_find_roche_index</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_sff_find_roche_index-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_sff_find_roche_index-expanded"><a name="L436"></a><tt class="py-lineno"> 436</tt>  <tt class="py-line">    <tt class="py-docstring">"""Locate any existing Roche style XML meta data and read index (PRIVATE).</tt> </tt>
<a name="L437"></a><tt class="py-lineno"> 437</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L438"></a><tt class="py-lineno"> 438</tt>  <tt class="py-line"><tt class="py-docstring">    Makes a number of hard coded assumptions based on reverse engineered SFF</tt> </tt>
<a name="L439"></a><tt class="py-lineno"> 439</tt>  <tt class="py-line"><tt class="py-docstring">    files from Roche 454 machines.</tt> </tt>
<a name="L440"></a><tt class="py-lineno"> 440</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L441"></a><tt class="py-lineno"> 441</tt>  <tt class="py-line"><tt class="py-docstring">    Returns a tuple of read count, SFF "index" offset and size, XML offset</tt> </tt>
<a name="L442"></a><tt class="py-lineno"> 442</tt>  <tt class="py-line"><tt class="py-docstring">    and size, and the actual read index offset and size.</tt> </tt>
<a name="L443"></a><tt class="py-lineno"> 443</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L444"></a><tt class="py-lineno"> 444</tt>  <tt class="py-line"><tt class="py-docstring">    Raises a ValueError for unsupported or non-Roche index blocks.</tt> </tt>
<a name="L445"></a><tt class="py-lineno"> 445</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L446"></a><tt class="py-lineno"> 446</tt>  <tt class="py-line">    <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-82" class="py-name"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-82', 'seek', 'link-50');">seek</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L447"></a><tt class="py-lineno"> 447</tt>  <tt class="py-line">    <tt class="py-name">header_length</tt><tt class="py-op">,</tt> <tt class="py-name">index_offset</tt><tt class="py-op">,</tt> <tt class="py-name">index_length</tt><tt class="py-op">,</tt> <tt class="py-name">number_of_reads</tt><tt class="py-op">,</tt> \ </tt>
<a name="L448"></a><tt class="py-lineno"> 448</tt>  <tt class="py-line">        <tt class="py-name">number_of_flows_per_read</tt><tt class="py-op">,</tt> <tt class="py-name">flow_chars</tt><tt class="py-op">,</tt> <tt class="py-name">key_sequence</tt> \ </tt>
<a name="L449"></a><tt class="py-lineno"> 449</tt>  <tt class="py-line">        <tt class="py-op">=</tt> <tt id="link-83" class="py-name"><a title="Bio.SeqIO.SffIO._sff_file_header" class="py-name" href="#" onclick="return doclink('link-83', '_sff_file_header', 'link-51');">_sff_file_header</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L450"></a><tt class="py-lineno"> 450</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-84" class="py-name"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-84', 'tell', 'link-54');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">header_length</tt> </tt>
<a name="L451"></a><tt class="py-lineno"> 451</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">index_offset</tt> <tt class="py-keyword">or</tt> <tt class="py-keyword">not</tt> <tt class="py-name">index_offset</tt><tt class="py-op">:</tt> </tt>
<a name="L452"></a><tt class="py-lineno"> 452</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"No index present in this SFF file"</tt><tt class="py-op">)</tt> </tt>
<a name="L453"></a><tt class="py-lineno"> 453</tt>  <tt class="py-line">    <tt class="py-comment"># Now jump to the header...</tt> </tt>
<a name="L454"></a><tt class="py-lineno"> 454</tt>  <tt class="py-line">    <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-85" class="py-name"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-85', 'seek', 'link-50');">seek</a></tt><tt class="py-op">(</tt><tt class="py-name">index_offset</tt><tt class="py-op">)</tt> </tt>
<a name="L455"></a><tt class="py-lineno"> 455</tt>  <tt class="py-line">    <tt class="py-name">fmt</tt> <tt class="py-op">=</tt> <tt class="py-string">"&gt;4s4B"</tt> </tt>
<a name="L456"></a><tt class="py-lineno"> 456</tt>  <tt class="py-line">    <tt class="py-name">fmt_size</tt> <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-name">fmt</tt><tt class="py-op">)</tt> </tt>
<a name="L457"></a><tt class="py-lineno"> 457</tt>  <tt class="py-line">    <tt id="link-86" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-86', 'data', 'link-31');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-87" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-87', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">fmt_size</tt><tt class="py-op">)</tt> </tt>
<a name="L458"></a><tt class="py-lineno"> 458</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-88" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-88', 'data', 'link-31');">data</a></tt><tt class="py-op">:</tt> </tt>
<a name="L459"></a><tt class="py-lineno"> 459</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Premature end of file? Expected index of size %i at offest %i, found nothing"</tt> </tt>
<a name="L460"></a><tt class="py-lineno"> 460</tt>  <tt class="py-line">                         <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">index_length</tt><tt class="py-op">,</tt> <tt class="py-name">index_offset</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L461"></a><tt class="py-lineno"> 461</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-89" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-89', 'data', 'link-31');">data</a></tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">fmt_size</tt><tt class="py-op">:</tt> </tt>
<a name="L462"></a><tt class="py-lineno"> 462</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Premature end of file? Expected index of size %i at offest %i, found %s"</tt> </tt>
<a name="L463"></a><tt class="py-lineno"> 463</tt>  <tt class="py-line">                         <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">index_length</tt><tt class="py-op">,</tt> <tt class="py-name">index_offset</tt><tt class="py-op">,</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt id="link-90" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-90', 'data', 'link-31');">data</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L464"></a><tt class="py-lineno"> 464</tt>  <tt class="py-line">    <tt class="py-name">magic_number</tt><tt class="py-op">,</tt> <tt class="py-name">ver0</tt><tt class="py-op">,</tt> <tt class="py-name">ver1</tt><tt class="py-op">,</tt> <tt class="py-name">ver2</tt><tt class="py-op">,</tt> <tt class="py-name">ver3</tt> <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">unpack</tt><tt class="py-op">(</tt><tt class="py-name">fmt</tt><tt class="py-op">,</tt> <tt id="link-91" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-91', 'data', 'link-31');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L465"></a><tt class="py-lineno"> 465</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">magic_number</tt> <tt class="py-op">==</tt> <tt id="link-92" class="py-name"><a title="Bio.SeqIO.SffIO._mft" class="py-name" href="#" onclick="return doclink('link-92', '_mft', 'link-24');">_mft</a></tt><tt class="py-op">:</tt>  <tt class="py-comment"># 778921588</tt> </tt>
<a name="L466"></a><tt class="py-lineno"> 466</tt>  <tt class="py-line">        <tt class="py-comment"># Roche 454 manifest index</tt> </tt>
<a name="L467"></a><tt class="py-lineno"> 467</tt>  <tt class="py-line">        <tt class="py-comment"># This is typical from raw Roche 454 SFF files (2009), and includes</tt> </tt>
<a name="L468"></a><tt class="py-lineno"> 468</tt>  <tt class="py-line">        <tt class="py-comment"># both an XML manifest and the sorted index.</tt> </tt>
<a name="L469"></a><tt class="py-lineno"> 469</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-op">(</tt><tt class="py-name">ver0</tt><tt class="py-op">,</tt> <tt class="py-name">ver1</tt><tt class="py-op">,</tt> <tt class="py-name">ver2</tt><tt class="py-op">,</tt> <tt class="py-name">ver3</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-op">(</tt><tt class="py-number">49</tt><tt class="py-op">,</tt> <tt class="py-number">46</tt><tt class="py-op">,</tt> <tt class="py-number">48</tt><tt class="py-op">,</tt> <tt class="py-number">48</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L470"></a><tt class="py-lineno"> 470</tt>  <tt class="py-line">            <tt class="py-comment"># This is "1.00" as a string</tt> </tt>
<a name="L471"></a><tt class="py-lineno"> 471</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unsupported version in .mft index header, %i.%i.%i.%i"</tt> </tt>
<a name="L472"></a><tt class="py-lineno"> 472</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">ver0</tt><tt class="py-op">,</tt> <tt class="py-name">ver1</tt><tt class="py-op">,</tt> <tt class="py-name">ver2</tt><tt class="py-op">,</tt> <tt class="py-name">ver3</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L473"></a><tt class="py-lineno"> 473</tt>  <tt class="py-line">        <tt class="py-name">fmt2</tt> <tt class="py-op">=</tt> <tt class="py-string">"&gt;LL"</tt> </tt>
<a name="L474"></a><tt class="py-lineno"> 474</tt>  <tt class="py-line">        <tt class="py-name">fmt2_size</tt> <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-name">fmt2</tt><tt class="py-op">)</tt> </tt>
<a name="L475"></a><tt class="py-lineno"> 475</tt>  <tt class="py-line">        <tt class="py-name">xml_size</tt><tt class="py-op">,</tt> <tt class="py-name">data_size</tt> <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">unpack</tt><tt class="py-op">(</tt><tt class="py-name">fmt2</tt><tt class="py-op">,</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-93" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-93', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">fmt2_size</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L476"></a><tt class="py-lineno"> 476</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">index_length</tt> <tt class="py-op">!=</tt> <tt class="py-name">fmt_size</tt> <tt class="py-op">+</tt> <tt class="py-name">fmt2_size</tt> <tt class="py-op">+</tt> <tt class="py-name">xml_size</tt> <tt class="py-op">+</tt> <tt class="py-name">data_size</tt><tt class="py-op">:</tt> </tt>
<a name="L477"></a><tt class="py-lineno"> 477</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Problem understanding .mft index header, %i != %i + %i + %i + %i"</tt> </tt>
<a name="L478"></a><tt class="py-lineno"> 478</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">index_length</tt><tt class="py-op">,</tt> <tt class="py-name">fmt_size</tt><tt class="py-op">,</tt> <tt class="py-name">fmt2_size</tt><tt class="py-op">,</tt> <tt class="py-name">xml_size</tt><tt class="py-op">,</tt> <tt class="py-name">data_size</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L479"></a><tt class="py-lineno"> 479</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">number_of_reads</tt><tt class="py-op">,</tt> <tt class="py-name">header_length</tt><tt class="py-op">,</tt> \ </tt>
<a name="L480"></a><tt class="py-lineno"> 480</tt>  <tt class="py-line">            <tt class="py-name">index_offset</tt><tt class="py-op">,</tt> <tt class="py-name">index_length</tt><tt class="py-op">,</tt> \ </tt>
<a name="L481"></a><tt class="py-lineno"> 481</tt>  <tt class="py-line">            <tt class="py-name">index_offset</tt> <tt class="py-op">+</tt> <tt class="py-name">fmt_size</tt> <tt class="py-op">+</tt> <tt class="py-name">fmt2_size</tt><tt class="py-op">,</tt> <tt class="py-name">xml_size</tt><tt class="py-op">,</tt> \ </tt>
<a name="L482"></a><tt class="py-lineno"> 482</tt>  <tt class="py-line">            <tt class="py-name">index_offset</tt> <tt class="py-op">+</tt> <tt class="py-name">fmt_size</tt> <tt class="py-op">+</tt> <tt class="py-name">fmt2_size</tt> <tt class="py-op">+</tt> <tt class="py-name">xml_size</tt><tt class="py-op">,</tt> <tt class="py-name">data_size</tt> </tt>
<a name="L483"></a><tt class="py-lineno"> 483</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">magic_number</tt> <tt class="py-op">==</tt> <tt id="link-94" class="py-name"><a title="Bio.SeqIO.SffIO._srt" class="py-name" href="#" onclick="return doclink('link-94', '_srt', 'link-22');">_srt</a></tt><tt class="py-op">:</tt>  <tt class="py-comment"># 779317876</tt> </tt>
<a name="L484"></a><tt class="py-lineno"> 484</tt>  <tt class="py-line">        <tt class="py-comment"># Roche 454 sorted index</tt> </tt>
<a name="L485"></a><tt class="py-lineno"> 485</tt>  <tt class="py-line">        <tt class="py-comment"># I've had this from Roche tool sfffile when the read identifiers</tt> </tt>
<a name="L486"></a><tt class="py-lineno"> 486</tt>  <tt class="py-line">        <tt class="py-comment"># had nonstandard lengths and there was no XML manifest.</tt> </tt>
<a name="L487"></a><tt class="py-lineno"> 487</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-op">(</tt><tt class="py-name">ver0</tt><tt class="py-op">,</tt> <tt class="py-name">ver1</tt><tt class="py-op">,</tt> <tt class="py-name">ver2</tt><tt class="py-op">,</tt> <tt class="py-name">ver3</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-op">(</tt><tt class="py-number">49</tt><tt class="py-op">,</tt> <tt class="py-number">46</tt><tt class="py-op">,</tt> <tt class="py-number">48</tt><tt class="py-op">,</tt> <tt class="py-number">48</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L488"></a><tt class="py-lineno"> 488</tt>  <tt class="py-line">            <tt class="py-comment"># This is "1.00" as a string</tt> </tt>
<a name="L489"></a><tt class="py-lineno"> 489</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unsupported version in .srt index header, %i.%i.%i.%i"</tt> </tt>
<a name="L490"></a><tt class="py-lineno"> 490</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">ver0</tt><tt class="py-op">,</tt> <tt class="py-name">ver1</tt><tt class="py-op">,</tt> <tt class="py-name">ver2</tt><tt class="py-op">,</tt> <tt class="py-name">ver3</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L491"></a><tt class="py-lineno"> 491</tt>  <tt class="py-line">        <tt id="link-95" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-95', 'data', 'link-31');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-96" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-96', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-number">4</tt><tt class="py-op">)</tt> </tt>
<a name="L492"></a><tt class="py-lineno"> 492</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-97" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-97', 'data', 'link-31');">data</a></tt> <tt class="py-op">!=</tt> <tt id="link-98" class="py-name"><a title="Bio.SeqIO.SffIO._null" class="py-name" href="#" onclick="return doclink('link-98', '_null', 'link-16');">_null</a></tt> <tt class="py-op">*</tt> <tt class="py-number">4</tt><tt class="py-op">:</tt> </tt>
<a name="L493"></a><tt class="py-lineno"> 493</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt> </tt>
<a name="L494"></a><tt class="py-lineno"> 494</tt>  <tt class="py-line">                <tt class="py-string">"Did not find expected null four bytes in .srt index"</tt><tt class="py-op">)</tt> </tt>
<a name="L495"></a><tt class="py-lineno"> 495</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">number_of_reads</tt><tt class="py-op">,</tt> <tt class="py-name">header_length</tt><tt class="py-op">,</tt> \ </tt>
<a name="L496"></a><tt class="py-lineno"> 496</tt>  <tt class="py-line">            <tt class="py-name">index_offset</tt><tt class="py-op">,</tt> <tt class="py-name">index_length</tt><tt class="py-op">,</tt> \ </tt>
<a name="L497"></a><tt class="py-lineno"> 497</tt>  <tt class="py-line">            <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> \ </tt>
<a name="L498"></a><tt class="py-lineno"> 498</tt>  <tt class="py-line">            <tt class="py-name">index_offset</tt> <tt class="py-op">+</tt> <tt class="py-name">fmt_size</tt> <tt class="py-op">+</tt> <tt class="py-number">4</tt><tt class="py-op">,</tt> <tt class="py-name">index_length</tt> <tt class="py-op">-</tt> <tt class="py-name">fmt_size</tt> <tt class="py-op">-</tt> <tt class="py-number">4</tt> </tt>
<a name="L499"></a><tt class="py-lineno"> 499</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">magic_number</tt> <tt class="py-op">==</tt> <tt id="link-99" class="py-name"><a title="Bio.SeqIO.SffIO._hsh" class="py-name" href="#" onclick="return doclink('link-99', '_hsh', 'link-20');">_hsh</a></tt><tt class="py-op">:</tt> </tt>
<a name="L500"></a><tt class="py-lineno"> 500</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Hash table style indexes (.hsh) in SFF files are "</tt> </tt>
<a name="L501"></a><tt class="py-lineno"> 501</tt>  <tt class="py-line">                         <tt class="py-string">"not (yet) supported"</tt><tt class="py-op">)</tt> </tt>
<a name="L502"></a><tt class="py-lineno"> 502</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L503"></a><tt class="py-lineno"> 503</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unknown magic number %s in SFF index header:\n%s"</tt> </tt>
<a name="L504"></a><tt class="py-lineno"> 504</tt>  <tt class="py-line">                         <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">magic_number</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt id="link-100" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-100', 'data', 'link-31');">data</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L505"></a><tt class="py-lineno"> 505</tt>  <tt class="py-line"> </tt>
<a name="L506"></a><tt class="py-lineno"> 506</tt>  <tt class="py-line"> </tt>
<a name="ReadRocheXmlManifest"></a><div id="ReadRocheXmlManifest-def"><a name="L507"></a><tt class="py-lineno"> 507</tt> <a class="py-toggle" href="#" id="ReadRocheXmlManifest-toggle" onclick="return toggle('ReadRocheXmlManifest');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO-module.html#ReadRocheXmlManifest">ReadRocheXmlManifest</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ReadRocheXmlManifest-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="ReadRocheXmlManifest-expanded"><a name="L508"></a><tt class="py-lineno"> 508</tt>  <tt class="py-line">    <tt class="py-docstring">"""Reads any Roche style XML manifest data in the SFF "index".</tt> </tt>
<a name="L509"></a><tt class="py-lineno"> 509</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L510"></a><tt class="py-lineno"> 510</tt>  <tt class="py-line"><tt class="py-docstring">    The SFF file format allows for multiple different index blocks, and Roche</tt> </tt>
<a name="L511"></a><tt class="py-lineno"> 511</tt>  <tt class="py-line"><tt class="py-docstring">    took advantage of this to define their own index block which also embeds</tt> </tt>
<a name="L512"></a><tt class="py-lineno"> 512</tt>  <tt class="py-line"><tt class="py-docstring">    an XML manifest string. This is not a publically documented extension to</tt> </tt>
<a name="L513"></a><tt class="py-lineno"> 513</tt>  <tt class="py-line"><tt class="py-docstring">    the SFF file format, this was reverse engineered.</tt> </tt>
<a name="L514"></a><tt class="py-lineno"> 514</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L515"></a><tt class="py-lineno"> 515</tt>  <tt class="py-line"><tt class="py-docstring">    The handle should be to an SFF file opened in binary mode. This function</tt> </tt>
<a name="L516"></a><tt class="py-lineno"> 516</tt>  <tt class="py-line"><tt class="py-docstring">    will use the handle seek/tell functions and leave the handle in an</tt> </tt>
<a name="L517"></a><tt class="py-lineno"> 517</tt>  <tt class="py-line"><tt class="py-docstring">    arbitrary location.</tt> </tt>
<a name="L518"></a><tt class="py-lineno"> 518</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L519"></a><tt class="py-lineno"> 519</tt>  <tt class="py-line"><tt class="py-docstring">    Any XML manifest found is returned as a Python string, which you can then</tt> </tt>
<a name="L520"></a><tt class="py-lineno"> 520</tt>  <tt class="py-line"><tt class="py-docstring">    parse as appropriate, or reuse when writing out SFF files with the</tt> </tt>
<a name="L521"></a><tt class="py-lineno"> 521</tt>  <tt class="py-line"><tt class="py-docstring">    SffWriter class.</tt> </tt>
<a name="L522"></a><tt class="py-lineno"> 522</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L523"></a><tt class="py-lineno"> 523</tt>  <tt class="py-line"><tt class="py-docstring">    Returns a string, or raises a ValueError if an Roche manifest could not be</tt> </tt>
<a name="L524"></a><tt class="py-lineno"> 524</tt>  <tt class="py-line"><tt class="py-docstring">    found.</tt> </tt>
<a name="L525"></a><tt class="py-lineno"> 525</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L526"></a><tt class="py-lineno"> 526</tt>  <tt class="py-line">    <tt class="py-name">number_of_reads</tt><tt class="py-op">,</tt> <tt class="py-name">header_length</tt><tt class="py-op">,</tt> <tt class="py-name">index_offset</tt><tt class="py-op">,</tt> <tt class="py-name">index_length</tt><tt class="py-op">,</tt> <tt class="py-name">xml_offset</tt><tt class="py-op">,</tt> \ </tt>
<a name="L527"></a><tt class="py-lineno"> 527</tt>  <tt class="py-line">        <tt class="py-name">xml_size</tt><tt class="py-op">,</tt> <tt class="py-name">read_index_offset</tt><tt class="py-op">,</tt> <tt class="py-name">read_index_size</tt> <tt class="py-op">=</tt> <tt id="link-101" class="py-name" targets="Function Bio.SeqIO.SffIO._sff_find_roche_index()=Bio.SeqIO.SffIO-module.html#_sff_find_roche_index"><a title="Bio.SeqIO.SffIO._sff_find_roche_index" class="py-name" href="#" onclick="return doclink('link-101', '_sff_find_roche_index', 'link-101');">_sff_find_roche_index</a></tt><tt class="py-op">(</tt> </tt>
<a name="L528"></a><tt class="py-lineno"> 528</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L529"></a><tt class="py-lineno"> 529</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">xml_offset</tt> <tt class="py-keyword">or</tt> <tt class="py-keyword">not</tt> <tt class="py-name">xml_size</tt><tt class="py-op">:</tt> </tt>
<a name="L530"></a><tt class="py-lineno"> 530</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"No XML manifest found"</tt><tt class="py-op">)</tt> </tt>
<a name="L531"></a><tt class="py-lineno"> 531</tt>  <tt class="py-line">    <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-102" class="py-name"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-102', 'seek', 'link-50');">seek</a></tt><tt class="py-op">(</tt><tt class="py-name">xml_offset</tt><tt class="py-op">)</tt> </tt>
<a name="L532"></a><tt class="py-lineno"> 532</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-103" class="py-name"><a title="Bio._py3k._bytes_to_string" class="py-name" href="#" onclick="return doclink('link-103', '_bytes_to_string', 'link-14');">_bytes_to_string</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-104" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-104', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">xml_size</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L533"></a><tt class="py-lineno"> 533</tt>  <tt class="py-line"> </tt>
<a name="L534"></a><tt class="py-lineno"> 534</tt>  <tt class="py-line"> </tt>
<a name="L535"></a><tt class="py-lineno"> 535</tt>  <tt class="py-line"><tt class="py-comment"># This is a generator function!</tt> </tt>
<a name="_sff_read_roche_index"></a><div id="_sff_read_roche_index-def"><a name="L536"></a><tt class="py-lineno"> 536</tt> <a class="py-toggle" href="#" id="_sff_read_roche_index-toggle" onclick="return toggle('_sff_read_roche_index');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO-module.html#_sff_read_roche_index">_sff_read_roche_index</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_sff_read_roche_index-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_sff_read_roche_index-expanded"><a name="L537"></a><tt class="py-lineno"> 537</tt>  <tt class="py-line">    <tt class="py-docstring">"""Reads any existing Roche style read index provided in the SFF file (PRIVATE).</tt> </tt>
<a name="L538"></a><tt class="py-lineno"> 538</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L539"></a><tt class="py-lineno"> 539</tt>  <tt class="py-line"><tt class="py-docstring">    Will use the handle seek/tell functions.</tt> </tt>
<a name="L540"></a><tt class="py-lineno"> 540</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L541"></a><tt class="py-lineno"> 541</tt>  <tt class="py-line"><tt class="py-docstring">    This works on ".srt1.00" and ".mft1.00" style Roche SFF index blocks.</tt> </tt>
<a name="L542"></a><tt class="py-lineno"> 542</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L543"></a><tt class="py-lineno"> 543</tt>  <tt class="py-line"><tt class="py-docstring">    Roche SFF indices use base 255 not 256, meaning we see bytes in range the</tt> </tt>
<a name="L544"></a><tt class="py-lineno"> 544</tt>  <tt class="py-line"><tt class="py-docstring">    range 0 to 254 only. This appears to be so that byte 0xFF (character 255)</tt> </tt>
<a name="L545"></a><tt class="py-lineno"> 545</tt>  <tt class="py-line"><tt class="py-docstring">    can be used as a marker character to separate entries (required if the</tt> </tt>
<a name="L546"></a><tt class="py-lineno"> 546</tt>  <tt class="py-line"><tt class="py-docstring">    read name lengths vary).</tt> </tt>
<a name="L547"></a><tt class="py-lineno"> 547</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L548"></a><tt class="py-lineno"> 548</tt>  <tt class="py-line"><tt class="py-docstring">    Note that since only four bytes are used for the read offset, this is</tt> </tt>
<a name="L549"></a><tt class="py-lineno"> 549</tt>  <tt class="py-line"><tt class="py-docstring">    limited to 255^4 bytes (nearly 4GB). If you try to use the Roche sfffile</tt> </tt>
<a name="L550"></a><tt class="py-lineno"> 550</tt>  <tt class="py-line"><tt class="py-docstring">    tool to combine SFF files beyound this limit, they issue a warning and</tt> </tt>
<a name="L551"></a><tt class="py-lineno"> 551</tt>  <tt class="py-line"><tt class="py-docstring">    omit the index (and manifest).</tt> </tt>
<a name="L552"></a><tt class="py-lineno"> 552</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L553"></a><tt class="py-lineno"> 553</tt>  <tt class="py-line">    <tt class="py-name">number_of_reads</tt><tt class="py-op">,</tt> <tt class="py-name">header_length</tt><tt class="py-op">,</tt> <tt class="py-name">index_offset</tt><tt class="py-op">,</tt> <tt class="py-name">index_length</tt><tt class="py-op">,</tt> <tt class="py-name">xml_offset</tt><tt class="py-op">,</tt> \ </tt>
<a name="L554"></a><tt class="py-lineno"> 554</tt>  <tt class="py-line">        <tt class="py-name">xml_size</tt><tt class="py-op">,</tt> <tt class="py-name">read_index_offset</tt><tt class="py-op">,</tt> <tt class="py-name">read_index_size</tt> <tt class="py-op">=</tt> <tt id="link-105" class="py-name"><a title="Bio.SeqIO.SffIO._sff_find_roche_index" class="py-name" href="#" onclick="return doclink('link-105', '_sff_find_roche_index', 'link-101');">_sff_find_roche_index</a></tt><tt class="py-op">(</tt> </tt>
<a name="L555"></a><tt class="py-lineno"> 555</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L556"></a><tt class="py-lineno"> 556</tt>  <tt class="py-line">    <tt class="py-comment"># Now parse the read index...</tt> </tt>
<a name="L557"></a><tt class="py-lineno"> 557</tt>  <tt class="py-line">    <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-106" class="py-name"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-106', 'seek', 'link-50');">seek</a></tt><tt class="py-op">(</tt><tt class="py-name">read_index_offset</tt><tt class="py-op">)</tt> </tt>
<a name="L558"></a><tt class="py-lineno"> 558</tt>  <tt class="py-line">    <tt class="py-name">fmt</tt> <tt class="py-op">=</tt> <tt class="py-string">"&gt;5B"</tt> </tt>
<a name="L559"></a><tt class="py-lineno"> 559</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt id="link-107" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-107', 'read', 'link-32');">read</a></tt> <tt class="py-keyword">in</tt> <tt id="link-108" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-108', 'range', 'link-53');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">number_of_reads</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L560"></a><tt class="py-lineno"> 560</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Be more aware of when the index should end?</tt> </tt>
<a name="L561"></a><tt class="py-lineno"> 561</tt>  <tt class="py-line">        <tt id="link-109" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-109', 'data', 'link-31');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-110" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-110', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-number">6</tt><tt class="py-op">)</tt> </tt>
<a name="L562"></a><tt class="py-lineno"> 562</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L563"></a><tt class="py-lineno"> 563</tt>  <tt class="py-line">            <tt class="py-name">more</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-111" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-111', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L564"></a><tt class="py-lineno"> 564</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">more</tt><tt class="py-op">:</tt> </tt>
<a name="L565"></a><tt class="py-lineno"> 565</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Premature end of file!"</tt><tt class="py-op">)</tt> </tt>
<a name="L566"></a><tt class="py-lineno"> 566</tt>  <tt class="py-line">            <tt id="link-112" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-112', 'data', 'link-31');">data</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">more</tt> </tt>
<a name="L567"></a><tt class="py-lineno"> 567</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">more</tt> <tt class="py-op">==</tt> <tt id="link-113" class="py-name"><a title="Bio.SeqIO.SffIO._flag" class="py-name" href="#" onclick="return doclink('link-113', '_flag', 'link-26');">_flag</a></tt><tt class="py-op">:</tt> </tt>
<a name="L568"></a><tt class="py-lineno"> 568</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L569"></a><tt class="py-lineno"> 569</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt id="link-114" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-114', 'data', 'link-31');">data</a></tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt id="link-115" class="py-name"><a title="Bio.SeqIO.SffIO._flag" class="py-name" href="#" onclick="return doclink('link-115', '_flag', 'link-26');">_flag</a></tt><tt class="py-op">,</tt> <tt id="link-116" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-116', 'data', 'link-31');">data</a></tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L570"></a><tt class="py-lineno"> 570</tt>  <tt class="py-line">        <tt id="link-117" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-117', 'name', 'link-60');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-118" class="py-name"><a title="Bio._py3k._bytes_to_string" class="py-name" href="#" onclick="return doclink('link-118', '_bytes_to_string', 'link-14');">_bytes_to_string</a></tt><tt class="py-op">(</tt><tt id="link-119" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-119', 'data', 'link-31');">data</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">6</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L571"></a><tt class="py-lineno"> 571</tt>  <tt class="py-line">        <tt class="py-name">off4</tt><tt class="py-op">,</tt> <tt class="py-name">off3</tt><tt class="py-op">,</tt> <tt class="py-name">off2</tt><tt class="py-op">,</tt> <tt class="py-name">off1</tt><tt class="py-op">,</tt> <tt class="py-name">off0</tt> <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">unpack</tt><tt class="py-op">(</tt><tt class="py-name">fmt</tt><tt class="py-op">,</tt> <tt id="link-120" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-120', 'data', 'link-31');">data</a></tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">6</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L572"></a><tt class="py-lineno"> 572</tt>  <tt class="py-line">        <tt class="py-name">offset</tt> <tt class="py-op">=</tt> <tt class="py-name">off0</tt> <tt class="py-op">+</tt> <tt class="py-number">255</tt> <tt class="py-op">*</tt> <tt class="py-name">off1</tt> <tt class="py-op">+</tt> <tt class="py-number">65025</tt> <tt class="py-op">*</tt> <tt class="py-name">off2</tt> <tt class="py-op">+</tt> <tt class="py-number">16581375</tt> <tt class="py-op">*</tt> <tt class="py-name">off3</tt> </tt>
<a name="L573"></a><tt class="py-lineno"> 573</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">off4</tt><tt class="py-op">:</tt> </tt>
<a name="L574"></a><tt class="py-lineno"> 574</tt>  <tt class="py-line">            <tt class="py-comment"># Could in theory be used as a fifth piece of offset information,</tt> </tt>
<a name="L575"></a><tt class="py-lineno"> 575</tt>  <tt class="py-line">            <tt class="py-comment"># i.e. offset =+ 4228250625L*off4, but testing the Roche tools this</tt> </tt>
<a name="L576"></a><tt class="py-lineno"> 576</tt>  <tt class="py-line">            <tt class="py-comment"># is not the case. They simple don't support such large indexes.</tt> </tt>
<a name="L577"></a><tt class="py-lineno"> 577</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Expected a null terminator to the read name."</tt><tt class="py-op">)</tt> </tt>
<a name="L578"></a><tt class="py-lineno"> 578</tt>  <tt class="py-line">        <tt class="py-keyword">yield</tt> <tt id="link-121" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-121', 'name', 'link-60');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">offset</tt> </tt>
<a name="L579"></a><tt class="py-lineno"> 579</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-122" class="py-name"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-122', 'tell', 'link-54');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">read_index_offset</tt> <tt class="py-op">+</tt> <tt class="py-name">read_index_size</tt><tt class="py-op">:</tt> </tt>
<a name="L580"></a><tt class="py-lineno"> 580</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Problem with index length? %i vs %i"</tt> </tt>
<a name="L581"></a><tt class="py-lineno"> 581</tt>  <tt class="py-line">                         <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-123" class="py-name"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-123', 'tell', 'link-54');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">read_index_offset</tt> <tt class="py-op">+</tt> <tt class="py-name">read_index_size</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L582"></a><tt class="py-lineno"> 582</tt>  <tt class="py-line"> </tt>
<a name="L583"></a><tt class="py-lineno"> 583</tt>  <tt class="py-line"><tt id="link-124" class="py-name" targets="Variable Bio.SeqIO.SffIO._valid_UAN_read_name=Bio.SeqIO.SffIO-module.html#_valid_UAN_read_name"><a title="Bio.SeqIO.SffIO._valid_UAN_read_name" class="py-name" href="#" onclick="return doclink('link-124', '_valid_UAN_read_name', 'link-124');">_valid_UAN_read_name</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'^[a-zA-Z0-9]{14}$'</tt><tt class="py-op">)</tt> </tt>
<a name="L584"></a><tt class="py-lineno"> 584</tt>  <tt class="py-line"> </tt>
<a name="L585"></a><tt class="py-lineno"> 585</tt>  <tt class="py-line"> </tt>
<a name="_sff_read_seq_record"></a><div id="_sff_read_seq_record-def"><a name="L586"></a><tt class="py-lineno"> 586</tt> <a class="py-toggle" href="#" id="_sff_read_seq_record-toggle" onclick="return toggle('_sff_read_seq_record');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO-module.html#_sff_read_seq_record">_sff_read_seq_record</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">number_of_flows_per_read</tt><tt class="py-op">,</tt> <tt class="py-param">flow_chars</tt><tt class="py-op">,</tt> </tt>
<a name="L587"></a><tt class="py-lineno"> 587</tt>  <tt class="py-line">                         <tt class="py-param">key_sequence</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">,</tt> <tt class="py-param">trim</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_sff_read_seq_record-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_sff_read_seq_record-expanded"><a name="L588"></a><tt class="py-lineno"> 588</tt>  <tt class="py-line">    <tt class="py-docstring">"""Parse the next read in the file, return data as a SeqRecord (PRIVATE)."""</tt> </tt>
<a name="L589"></a><tt class="py-lineno"> 589</tt>  <tt class="py-line">    <tt class="py-comment"># Now on to the reads...</tt> </tt>
<a name="L590"></a><tt class="py-lineno"> 590</tt>  <tt class="py-line">    <tt class="py-comment"># the read header format (fixed part):</tt> </tt>
<a name="L591"></a><tt class="py-lineno"> 591</tt>  <tt class="py-line">    <tt class="py-comment"># read_header_length     H</tt> </tt>
<a name="L592"></a><tt class="py-lineno"> 592</tt>  <tt class="py-line">    <tt class="py-comment"># name_length            H</tt> </tt>
<a name="L593"></a><tt class="py-lineno"> 593</tt>  <tt class="py-line">    <tt class="py-comment"># seq_len                I</tt> </tt>
<a name="L594"></a><tt class="py-lineno"> 594</tt>  <tt class="py-line">    <tt class="py-comment"># clip_qual_left         H</tt> </tt>
<a name="L595"></a><tt class="py-lineno"> 595</tt>  <tt class="py-line">    <tt class="py-comment"># clip_qual_right        H</tt> </tt>
<a name="L596"></a><tt class="py-lineno"> 596</tt>  <tt class="py-line">    <tt class="py-comment"># clip_adapter_left      H</tt> </tt>
<a name="L597"></a><tt class="py-lineno"> 597</tt>  <tt class="py-line">    <tt class="py-comment"># clip_adapter_right     H</tt> </tt>
<a name="L598"></a><tt class="py-lineno"> 598</tt>  <tt class="py-line">    <tt class="py-comment"># [rest of read header depends on the name length etc]</tt> </tt>
<a name="L599"></a><tt class="py-lineno"> 599</tt>  <tt class="py-line">    <tt class="py-name">read_header_fmt</tt> <tt class="py-op">=</tt> <tt class="py-string">'&gt;2HI4H'</tt> </tt>
<a name="L600"></a><tt class="py-lineno"> 600</tt>  <tt class="py-line">    <tt class="py-name">read_header_size</tt> <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-name">read_header_fmt</tt><tt class="py-op">)</tt> </tt>
<a name="L601"></a><tt class="py-lineno"> 601</tt>  <tt class="py-line">    <tt class="py-name">read_flow_fmt</tt> <tt class="py-op">=</tt> <tt class="py-string">"&gt;%iH"</tt> <tt class="py-op">%</tt> <tt class="py-name">number_of_flows_per_read</tt> </tt>
<a name="L602"></a><tt class="py-lineno"> 602</tt>  <tt class="py-line">    <tt class="py-name">read_flow_size</tt> <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-name">read_flow_fmt</tt><tt class="py-op">)</tt> </tt>
<a name="L603"></a><tt class="py-lineno"> 603</tt>  <tt class="py-line"> </tt>
<a name="L604"></a><tt class="py-lineno"> 604</tt>  <tt class="py-line">    <tt class="py-name">read_header_length</tt><tt class="py-op">,</tt> <tt class="py-name">name_length</tt><tt class="py-op">,</tt> <tt class="py-name">seq_len</tt><tt class="py-op">,</tt> <tt class="py-name">clip_qual_left</tt><tt class="py-op">,</tt> \ </tt>
<a name="L605"></a><tt class="py-lineno"> 605</tt>  <tt class="py-line">        <tt class="py-name">clip_qual_right</tt><tt class="py-op">,</tt> <tt class="py-name">clip_adapter_left</tt><tt class="py-op">,</tt> <tt class="py-name">clip_adapter_right</tt> \ </tt>
<a name="L606"></a><tt class="py-lineno"> 606</tt>  <tt class="py-line">        <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">unpack</tt><tt class="py-op">(</tt><tt class="py-name">read_header_fmt</tt><tt class="py-op">,</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-125" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-125', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">read_header_size</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L607"></a><tt class="py-lineno"> 607</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">clip_qual_left</tt><tt class="py-op">:</tt> </tt>
<a name="L608"></a><tt class="py-lineno"> 608</tt>  <tt class="py-line">        <tt class="py-name">clip_qual_left</tt> <tt class="py-op">-=</tt> <tt class="py-number">1</tt>  <tt class="py-comment"># python counting</tt> </tt>
<a name="L609"></a><tt class="py-lineno"> 609</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">clip_adapter_left</tt><tt class="py-op">:</tt> </tt>
<a name="L610"></a><tt class="py-lineno"> 610</tt>  <tt class="py-line">        <tt class="py-name">clip_adapter_left</tt> <tt class="py-op">-=</tt> <tt class="py-number">1</tt>  <tt class="py-comment"># python counting</tt> </tt>
<a name="L611"></a><tt class="py-lineno"> 611</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">read_header_length</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">10</tt> <tt class="py-keyword">or</tt> <tt class="py-name">read_header_length</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt> <tt class="py-op">!=</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L612"></a><tt class="py-lineno"> 612</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Malformed read header, says length is %i"</tt> </tt>
<a name="L613"></a><tt class="py-lineno"> 613</tt>  <tt class="py-line">                         <tt class="py-op">%</tt> <tt class="py-name">read_header_length</tt><tt class="py-op">)</tt> </tt>
<a name="L614"></a><tt class="py-lineno"> 614</tt>  <tt class="py-line">    <tt class="py-comment"># now the name and any padding (remainder of header)</tt> </tt>
<a name="L615"></a><tt class="py-lineno"> 615</tt>  <tt class="py-line">    <tt id="link-126" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-126', 'name', 'link-60');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-127" class="py-name"><a title="Bio._py3k._bytes_to_string" class="py-name" href="#" onclick="return doclink('link-127', '_bytes_to_string', 'link-14');">_bytes_to_string</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-128" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-128', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">name_length</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L616"></a><tt class="py-lineno"> 616</tt>  <tt class="py-line">    <tt class="py-name">padding</tt> <tt class="py-op">=</tt> <tt class="py-name">read_header_length</tt> <tt class="py-op">-</tt> <tt class="py-name">read_header_size</tt> <tt class="py-op">-</tt> <tt class="py-name">name_length</tt> </tt>
<a name="L617"></a><tt class="py-lineno"> 617</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-129" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-129', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">padding</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-130" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-130', 'count', 'link-45');">count</a></tt><tt class="py-op">(</tt><tt id="link-131" class="py-name"><a title="Bio.SeqIO.SffIO._null" class="py-name" href="#" onclick="return doclink('link-131', '_null', 'link-16');">_null</a></tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">padding</tt><tt class="py-op">:</tt> </tt>
<a name="L618"></a><tt class="py-lineno"> 618</tt>  <tt class="py-line">        <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L619"></a><tt class="py-lineno"> 619</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt id="link-132" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-132', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-133" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-133', 'BiopythonParserWarning', 'link-48');">BiopythonParserWarning</a></tt> </tt>
<a name="L620"></a><tt class="py-lineno"> 620</tt>  <tt class="py-line">        <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Your SFF file is invalid, post name %i "</tt> </tt>
<a name="L621"></a><tt class="py-lineno"> 621</tt>  <tt class="py-line">                      <tt class="py-string">"byte padding region contained data"</tt> <tt class="py-op">%</tt> <tt class="py-name">padding</tt><tt class="py-op">,</tt> </tt>
<a name="L622"></a><tt class="py-lineno"> 622</tt>  <tt class="py-line">                      <tt id="link-134" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-134', 'BiopythonParserWarning', 'link-48');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L623"></a><tt class="py-lineno"> 623</tt>  <tt class="py-line">    <tt class="py-comment"># now the flowgram values, flowgram index, bases and qualities</tt> </tt>
<a name="L624"></a><tt class="py-lineno"> 624</tt>  <tt class="py-line">    <tt class="py-comment"># NOTE - assuming flowgram_format==1, which means struct type H</tt> </tt>
<a name="L625"></a><tt class="py-lineno"> 625</tt>  <tt class="py-line">    <tt class="py-name">flow_values</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-135" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-135', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">read_flow_size</tt><tt class="py-op">)</tt>  <tt class="py-comment"># unpack later if needed</tt> </tt>
<a name="L626"></a><tt class="py-lineno"> 626</tt>  <tt class="py-line">    <tt class="py-name">temp_fmt</tt> <tt class="py-op">=</tt> <tt class="py-string">"&gt;%iB"</tt> <tt class="py-op">%</tt> <tt class="py-name">seq_len</tt>  <tt class="py-comment"># used for flow index and quals</tt> </tt>
<a name="L627"></a><tt class="py-lineno"> 627</tt>  <tt class="py-line">    <tt class="py-name">flow_index</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-136" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-136', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">seq_len</tt><tt class="py-op">)</tt>  <tt class="py-comment"># unpack later if needed</tt> </tt>
<a name="L628"></a><tt class="py-lineno"> 628</tt>  <tt class="py-line">    <tt id="link-137" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.seq=Bio.SeqRecord.SeqRecord-class.html#seq,Variable BioSQL.BioSeq.DBSeqRecord.seq=BioSQL.BioSeq.DBSeqRecord-class.html#seq"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-137', 'seq', 'link-137');">seq</a></tt> <tt class="py-op">=</tt> <tt id="link-138" class="py-name"><a title="Bio._py3k._bytes_to_string" class="py-name" href="#" onclick="return doclink('link-138', '_bytes_to_string', 'link-14');">_bytes_to_string</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-139" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-139', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">seq_len</tt><tt class="py-op">)</tt><tt class="py-op">)</tt>  <tt class="py-comment"># TODO - Use bytes in Seq?</tt> </tt>
<a name="L629"></a><tt class="py-lineno"> 629</tt>  <tt class="py-line">    <tt class="py-name">quals</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">unpack</tt><tt class="py-op">(</tt><tt class="py-name">temp_fmt</tt><tt class="py-op">,</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-140" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-140', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">seq_len</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L630"></a><tt class="py-lineno"> 630</tt>  <tt class="py-line">    <tt class="py-comment"># now any padding...</tt> </tt>
<a name="L631"></a><tt class="py-lineno"> 631</tt>  <tt class="py-line">    <tt class="py-name">padding</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">read_flow_size</tt> <tt class="py-op">+</tt> <tt class="py-name">seq_len</tt> <tt class="py-op">*</tt> <tt class="py-number">3</tt><tt class="py-op">)</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt> </tt>
<a name="L632"></a><tt class="py-lineno"> 632</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">padding</tt><tt class="py-op">:</tt> </tt>
<a name="L633"></a><tt class="py-lineno"> 633</tt>  <tt class="py-line">        <tt class="py-name">padding</tt> <tt class="py-op">=</tt> <tt class="py-number">8</tt> <tt class="py-op">-</tt> <tt class="py-name">padding</tt> </tt>
<a name="L634"></a><tt class="py-lineno"> 634</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-141" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-141', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">padding</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-142" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-142', 'count', 'link-45');">count</a></tt><tt class="py-op">(</tt><tt id="link-143" class="py-name"><a title="Bio.SeqIO.SffIO._null" class="py-name" href="#" onclick="return doclink('link-143', '_null', 'link-16');">_null</a></tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">padding</tt><tt class="py-op">:</tt> </tt>
<a name="L635"></a><tt class="py-lineno"> 635</tt>  <tt class="py-line">            <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L636"></a><tt class="py-lineno"> 636</tt>  <tt class="py-line">            <tt class="py-keyword">from</tt> <tt id="link-144" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-144', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-145" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-145', 'BiopythonParserWarning', 'link-48');">BiopythonParserWarning</a></tt> </tt>
<a name="L637"></a><tt class="py-lineno"> 637</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Your SFF file is invalid, post quality %i "</tt> </tt>
<a name="L638"></a><tt class="py-lineno"> 638</tt>  <tt class="py-line">                          <tt class="py-string">"byte padding region contained data"</tt> <tt class="py-op">%</tt> <tt class="py-name">padding</tt><tt class="py-op">,</tt> </tt>
<a name="L639"></a><tt class="py-lineno"> 639</tt>  <tt class="py-line">                          <tt id="link-146" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-146', 'BiopythonParserWarning', 'link-48');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L640"></a><tt class="py-lineno"> 640</tt>  <tt class="py-line">    <tt class="py-comment"># Follow Roche and apply most aggressive of qual and adapter clipping.</tt> </tt>
<a name="L641"></a><tt class="py-lineno"> 641</tt>  <tt class="py-line">    <tt class="py-comment"># Note Roche seems to ignore adapter clip fields when writing SFF,</tt> </tt>
<a name="L642"></a><tt class="py-lineno"> 642</tt>  <tt class="py-line">    <tt class="py-comment"># and uses just the quality clipping values for any clipping.</tt> </tt>
<a name="L643"></a><tt class="py-lineno"> 643</tt>  <tt class="py-line">    <tt class="py-name">clip_left</tt> <tt class="py-op">=</tt> <tt id="link-147" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.max=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#max"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-147', 'max', 'link-147');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">clip_qual_left</tt><tt class="py-op">,</tt> <tt class="py-name">clip_adapter_left</tt><tt class="py-op">)</tt> </tt>
<a name="L644"></a><tt class="py-lineno"> 644</tt>  <tt class="py-line">    <tt class="py-comment"># Right clipping of zero means no clipping</tt> </tt>
<a name="L645"></a><tt class="py-lineno"> 645</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">clip_qual_right</tt><tt class="py-op">:</tt> </tt>
<a name="L646"></a><tt class="py-lineno"> 646</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">clip_adapter_right</tt><tt class="py-op">:</tt> </tt>
<a name="L647"></a><tt class="py-lineno"> 647</tt>  <tt class="py-line">            <tt class="py-name">clip_right</tt> <tt class="py-op">=</tt> <tt id="link-148" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.min=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#min"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-148', 'min', 'link-148');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">clip_qual_right</tt><tt class="py-op">,</tt> <tt class="py-name">clip_adapter_right</tt><tt class="py-op">)</tt> </tt>
<a name="L648"></a><tt class="py-lineno"> 648</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L649"></a><tt class="py-lineno"> 649</tt>  <tt class="py-line">            <tt class="py-comment"># Typical case with Roche SFF files</tt> </tt>
<a name="L650"></a><tt class="py-lineno"> 650</tt>  <tt class="py-line">            <tt class="py-name">clip_right</tt> <tt class="py-op">=</tt> <tt class="py-name">clip_qual_right</tt> </tt>
<a name="L651"></a><tt class="py-lineno"> 651</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">clip_adapter_right</tt><tt class="py-op">:</tt> </tt>
<a name="L652"></a><tt class="py-lineno"> 652</tt>  <tt class="py-line">        <tt class="py-name">clip_right</tt> <tt class="py-op">=</tt> <tt class="py-name">clip_adapter_right</tt> </tt>
<a name="L653"></a><tt class="py-lineno"> 653</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L654"></a><tt class="py-lineno"> 654</tt>  <tt class="py-line">        <tt class="py-name">clip_right</tt> <tt class="py-op">=</tt> <tt class="py-name">seq_len</tt> </tt>
<a name="L655"></a><tt class="py-lineno"> 655</tt>  <tt class="py-line">    <tt class="py-comment"># Now build a SeqRecord</tt> </tt>
<a name="L656"></a><tt class="py-lineno"> 656</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">trim</tt><tt class="py-op">:</tt> </tt>
<a name="L657"></a><tt class="py-lineno"> 657</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">clip_left</tt> <tt class="py-op">&gt;=</tt> <tt class="py-name">clip_right</tt><tt class="py-op">:</tt> </tt>
<a name="L658"></a><tt class="py-lineno"> 658</tt>  <tt class="py-line">            <tt class="py-comment"># Raise an error?</tt> </tt>
<a name="L659"></a><tt class="py-lineno"> 659</tt>  <tt class="py-line">            <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L660"></a><tt class="py-lineno"> 660</tt>  <tt class="py-line">            <tt class="py-keyword">from</tt> <tt id="link-149" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-149', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-150" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-150', 'BiopythonParserWarning', 'link-48');">BiopythonParserWarning</a></tt> </tt>
<a name="L661"></a><tt class="py-lineno"> 661</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Overlapping clip values in SFF record, trimmed to nothing"</tt><tt class="py-op">,</tt> </tt>
<a name="L662"></a><tt class="py-lineno"> 662</tt>  <tt class="py-line">                          <tt id="link-151" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-151', 'BiopythonParserWarning', 'link-48');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L663"></a><tt class="py-lineno"> 663</tt>  <tt class="py-line">            <tt id="link-152" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-152', 'seq', 'link-137');">seq</a></tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L664"></a><tt class="py-lineno"> 664</tt>  <tt class="py-line">            <tt class="py-name">quals</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L665"></a><tt class="py-lineno"> 665</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L666"></a><tt class="py-lineno"> 666</tt>  <tt class="py-line">            <tt id="link-153" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-153', 'seq', 'link-137');">seq</a></tt> <tt class="py-op">=</tt> <tt id="link-154" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-154', 'seq', 'link-137');">seq</a></tt><tt class="py-op">[</tt><tt class="py-name">clip_left</tt><tt class="py-op">:</tt><tt class="py-name">clip_right</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-155" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-155', 'upper', 'link-28');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L667"></a><tt class="py-lineno"> 667</tt>  <tt class="py-line">            <tt class="py-name">quals</tt> <tt class="py-op">=</tt> <tt class="py-name">quals</tt><tt class="py-op">[</tt><tt class="py-name">clip_left</tt><tt class="py-op">:</tt><tt class="py-name">clip_right</tt><tt class="py-op">]</tt> </tt>
<a name="L668"></a><tt class="py-lineno"> 668</tt>  <tt class="py-line">        <tt class="py-comment"># Don't record the clipping values, flow etc, they make no sense now:</tt> </tt>
<a name="L669"></a><tt class="py-lineno"> 669</tt>  <tt class="py-line">        <tt id="link-156" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.annotations=BioSQL.BioSeq.DBSeqRecord-class.html#annotations"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-156', 'annotations', 'link-156');">annotations</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L670"></a><tt class="py-lineno"> 670</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L671"></a><tt class="py-lineno"> 671</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">clip_left</tt> <tt class="py-op">&gt;=</tt> <tt class="py-name">clip_right</tt><tt class="py-op">:</tt> </tt>
<a name="L672"></a><tt class="py-lineno"> 672</tt>  <tt class="py-line">            <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L673"></a><tt class="py-lineno"> 673</tt>  <tt class="py-line">            <tt class="py-keyword">from</tt> <tt id="link-157" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-157', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-158" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-158', 'BiopythonParserWarning', 'link-48');">BiopythonParserWarning</a></tt> </tt>
<a name="L674"></a><tt class="py-lineno"> 674</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Overlapping clip values in SFF record"</tt><tt class="py-op">,</tt> <tt id="link-159" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-159', 'BiopythonParserWarning', 'link-48');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L675"></a><tt class="py-lineno"> 675</tt>  <tt class="py-line">            <tt id="link-160" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-160', 'seq', 'link-137');">seq</a></tt> <tt class="py-op">=</tt> <tt id="link-161" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-161', 'seq', 'link-137');">seq</a></tt><tt class="py-op">.</tt><tt id="link-162" class="py-name" targets="Method Bio.Seq.Seq.lower()=Bio.Seq.Seq-class.html#lower,Method Bio.Seq.UnknownSeq.lower()=Bio.Seq.UnknownSeq-class.html#lower,Method Bio.SeqRecord.SeqRecord.lower()=Bio.SeqRecord.SeqRecord-class.html#lower"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-162', 'lower', 'link-162');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L676"></a><tt class="py-lineno"> 676</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L677"></a><tt class="py-lineno"> 677</tt>  <tt class="py-line">            <tt class="py-comment"># This use of mixed case mimics the Roche SFF tool's FASTA output</tt> </tt>
<a name="L678"></a><tt class="py-lineno"> 678</tt>  <tt class="py-line">            <tt id="link-163" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-163', 'seq', 'link-137');">seq</a></tt> <tt class="py-op">=</tt> <tt id="link-164" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-164', 'seq', 'link-137');">seq</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">clip_left</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-165" class="py-name"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-165', 'lower', 'link-162');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L679"></a><tt class="py-lineno"> 679</tt>  <tt class="py-line">                <tt id="link-166" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-166', 'seq', 'link-137');">seq</a></tt><tt class="py-op">[</tt><tt class="py-name">clip_left</tt><tt class="py-op">:</tt><tt class="py-name">clip_right</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-167" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-167', 'upper', 'link-28');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L680"></a><tt class="py-lineno"> 680</tt>  <tt class="py-line">                <tt id="link-168" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-168', 'seq', 'link-137');">seq</a></tt><tt class="py-op">[</tt><tt class="py-name">clip_right</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-169" class="py-name"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-169', 'lower', 'link-162');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L681"></a><tt class="py-lineno"> 681</tt>  <tt class="py-line">        <tt id="link-170" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-170', 'annotations', 'link-156');">annotations</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">"flow_values"</tt><tt class="py-op">:</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">unpack</tt><tt class="py-op">(</tt><tt class="py-name">read_flow_fmt</tt><tt class="py-op">,</tt> <tt class="py-name">flow_values</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L682"></a><tt class="py-lineno"> 682</tt>  <tt class="py-line">                       <tt class="py-string">"flow_index"</tt><tt class="py-op">:</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">unpack</tt><tt class="py-op">(</tt><tt class="py-name">temp_fmt</tt><tt class="py-op">,</tt> <tt class="py-name">flow_index</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L683"></a><tt class="py-lineno"> 683</tt>  <tt class="py-line">                       <tt class="py-string">"flow_chars"</tt><tt class="py-op">:</tt> <tt class="py-name">flow_chars</tt><tt class="py-op">,</tt> </tt>
<a name="L684"></a><tt class="py-lineno"> 684</tt>  <tt class="py-line">                       <tt class="py-string">"flow_key"</tt><tt class="py-op">:</tt> <tt class="py-name">key_sequence</tt><tt class="py-op">,</tt> </tt>
<a name="L685"></a><tt class="py-lineno"> 685</tt>  <tt class="py-line">                       <tt class="py-string">"clip_qual_left"</tt><tt class="py-op">:</tt> <tt class="py-name">clip_qual_left</tt><tt class="py-op">,</tt> </tt>
<a name="L686"></a><tt class="py-lineno"> 686</tt>  <tt class="py-line">                       <tt class="py-string">"clip_qual_right"</tt><tt class="py-op">:</tt> <tt class="py-name">clip_qual_right</tt><tt class="py-op">,</tt> </tt>
<a name="L687"></a><tt class="py-lineno"> 687</tt>  <tt class="py-line">                       <tt class="py-string">"clip_adapter_left"</tt><tt class="py-op">:</tt> <tt class="py-name">clip_adapter_left</tt><tt class="py-op">,</tt> </tt>
<a name="L688"></a><tt class="py-lineno"> 688</tt>  <tt class="py-line">                       <tt class="py-string">"clip_adapter_right"</tt><tt class="py-op">:</tt> <tt class="py-name">clip_adapter_right</tt><tt class="py-op">}</tt> </tt>
<a name="L689"></a><tt class="py-lineno"> 689</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-171" class="py-name" targets="Function Bio.triefind.match()=Bio.triefind-module.html#match"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-171', 'match', 'link-171');">match</a></tt><tt class="py-op">(</tt><tt id="link-172" class="py-name"><a title="Bio.SeqIO.SffIO._valid_UAN_read_name" class="py-name" href="#" onclick="return doclink('link-172', '_valid_UAN_read_name', 'link-124');">_valid_UAN_read_name</a></tt><tt class="py-op">,</tt> <tt id="link-173" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-173', 'name', 'link-60');">name</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L690"></a><tt class="py-lineno"> 690</tt>  <tt class="py-line">        <tt id="link-174" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-174', 'annotations', 'link-156');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"time"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-175" class="py-name" targets="Function Bio.SeqIO.SffIO._get_read_time()=Bio.SeqIO.SffIO-module.html#_get_read_time"><a title="Bio.SeqIO.SffIO._get_read_time" class="py-name" href="#" onclick="return doclink('link-175', '_get_read_time', 'link-175');">_get_read_time</a></tt><tt class="py-op">(</tt><tt id="link-176" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-176', 'name', 'link-60');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L691"></a><tt class="py-lineno"> 691</tt>  <tt class="py-line">        <tt id="link-177" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-177', 'annotations', 'link-156');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"region"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-178" class="py-name" targets="Function Bio.SeqIO.SffIO._get_read_region()=Bio.SeqIO.SffIO-module.html#_get_read_region"><a title="Bio.SeqIO.SffIO._get_read_region" class="py-name" href="#" onclick="return doclink('link-178', '_get_read_region', 'link-178');">_get_read_region</a></tt><tt class="py-op">(</tt><tt id="link-179" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-179', 'name', 'link-60');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L692"></a><tt class="py-lineno"> 692</tt>  <tt class="py-line">        <tt id="link-180" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-180', 'annotations', 'link-156');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"coords"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-181" class="py-name" targets="Function Bio.SeqIO.SffIO._get_read_xy()=Bio.SeqIO.SffIO-module.html#_get_read_xy"><a title="Bio.SeqIO.SffIO._get_read_xy" class="py-name" href="#" onclick="return doclink('link-181', '_get_read_xy', 'link-181');">_get_read_xy</a></tt><tt class="py-op">(</tt><tt id="link-182" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-182', 'name', 'link-60');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L693"></a><tt class="py-lineno"> 693</tt>  <tt class="py-line">    <tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt id="link-183" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-183', 'SeqRecord', 'link-10');">SeqRecord</a></tt><tt class="py-op">(</tt><tt id="link-184" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-184', 'Seq', 'link-7');">Seq</a></tt><tt class="py-op">(</tt><tt id="link-185" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-185', 'seq', 'link-137');">seq</a></tt><tt class="py-op">,</tt> <tt id="link-186" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.alphabet=Bio.SearchIO._model.hsp.HSP-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSPFragment.alphabet=Bio.SearchIO._model.hsp.HSPFragment-class.html#alphabet"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-186', 'alphabet', 'link-186');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L694"></a><tt class="py-lineno"> 694</tt>  <tt class="py-line">                       <tt id="link-187" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-187', 'id', 'link-187');">id</a></tt><tt class="py-op">=</tt><tt id="link-188" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-188', 'name', 'link-60');">name</a></tt><tt class="py-op">,</tt> </tt>
<a name="L695"></a><tt class="py-lineno"> 695</tt>  <tt class="py-line">                       <tt id="link-189" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-189', 'name', 'link-60');">name</a></tt><tt class="py-op">=</tt><tt id="link-190" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-190', 'name', 'link-60');">name</a></tt><tt class="py-op">,</tt> </tt>
<a name="L696"></a><tt class="py-lineno"> 696</tt>  <tt class="py-line">                       <tt id="link-191" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DescriptionConsumer.description()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#description,Method Bio.Phylo.PhyloXMLIO.Writer.description()=Bio.Phylo.PhyloXMLIO.Writer-class.html#description,Variable Bio.SearchIO._model.hit.Hit.description=Bio.SearchIO._model.hit.Hit-class.html#description,Variable Bio.SearchIO._model.query.QueryResult.description=Bio.SearchIO._model.query.QueryResult-class.html#description"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-191', 'description', 'link-191');">description</a></tt><tt class="py-op">=</tt><tt class="py-string">""</tt><tt class="py-op">,</tt> </tt>
<a name="L697"></a><tt class="py-lineno"> 697</tt>  <tt class="py-line">                       <tt id="link-192" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-192', 'annotations', 'link-156');">annotations</a></tt><tt class="py-op">=</tt><tt id="link-193" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-193', 'annotations', 'link-156');">annotations</a></tt><tt class="py-op">)</tt> </tt>
<a name="L698"></a><tt class="py-lineno"> 698</tt>  <tt class="py-line">    <tt class="py-comment"># Dirty trick to speed up this line:</tt> </tt>
<a name="L699"></a><tt class="py-lineno"> 699</tt>  <tt class="py-line">    <tt class="py-comment"># record.letter_annotations["phred_quality"] = quals</tt> </tt>
<a name="L700"></a><tt class="py-lineno"> 700</tt>  <tt class="py-line">    <tt class="py-name">dict</tt><tt class="py-op">.</tt><tt id="link-194" class="py-name" targets="Method Bio.Crystal.Chain.__setitem__()=Bio.Crystal.Chain-class.html#__setitem__,Method Bio.Crystal.Crystal.__setitem__()=Bio.Crystal.Crystal-class.html#__setitem__,Method Bio.Entrez.Parser.StructureElement.__setitem__()=Bio.Entrez.Parser.StructureElement-class.html#__setitem__,Method Bio.File._IndexedSeqFileDict.__setitem__()=Bio.File._IndexedSeqFileDict-class.html#__setitem__,Method Bio.Index._InMemoryIndex.__setitem__()=Bio.Index._InMemoryIndex-class.html#__setitem__,Method Bio.PDB.Entity.DisorderedEntityWrapper.__setitem__()=Bio.PDB.Entity.DisorderedEntityWrapper-class.html#__setitem__,Method Bio.PDB.Vector'.Vector.__setitem__()=Bio.PDB.Vector%27.Vector-class.html#__setitem__,Method Bio.Phylo.PhyloXML.Events.__setitem__()=Bio.Phylo.PhyloXML.Events-class.html#__setitem__,Method Bio.Phylo.TreeConstruction._DistanceMatrix.__setitem__()=Bio.Phylo.TreeConstruction._DistanceMatrix-class.html#__setitem__,Method Bio.Phylo.TreeConstruction._Matrix.__setitem__()=Bio.Phylo.TreeConstruction._Matrix-class.html#__setitem__,Method Bio.SearchIO._model.hit.Hit.__setitem__()=Bio.SearchIO._model.hit.Hit-class.html#__setitem__,Method Bio.SearchIO._model.hsp.HSP.__setitem__()=Bio.SearchIO._model.hsp.HSP-class.html#__setitem__,Method Bio.SearchIO._model.query.QueryResult.__setitem__()=Bio.SearchIO._model.query.QueryResult-class.html#__setitem__,Method Bio.Seq.MutableSeq.__setitem__()=Bio.Seq.MutableSeq-class.html#__setitem__,Method Bio.SeqRecord._RestrictedDict.__setitem__()=Bio.SeqRecord._RestrictedDict-class.html#__setitem__,Method Bio._py3k._ordereddict.OrderedDict.__setitem__()=Bio._py3k._ordereddict.OrderedDict-class.html#__setitem__"><a title="Bio.Crystal.Chain.__setitem__
Bio.Crystal.Crystal.__setitem__
Bio.Entrez.Parser.StructureElement.__setitem__
Bio.File._IndexedSeqFileDict.__setitem__
Bio.Index._InMemoryIndex.__setitem__
Bio.PDB.Entity.DisorderedEntityWrapper.__setitem__
Bio.PDB.Vector'.Vector.__setitem__
Bio.Phylo.PhyloXML.Events.__setitem__
Bio.Phylo.TreeConstruction._DistanceMatrix.__setitem__
Bio.Phylo.TreeConstruction._Matrix.__setitem__
Bio.SearchIO._model.hit.Hit.__setitem__
Bio.SearchIO._model.hsp.HSP.__setitem__
Bio.SearchIO._model.query.QueryResult.__setitem__
Bio.Seq.MutableSeq.__setitem__
Bio.SeqRecord._RestrictedDict.__setitem__
Bio._py3k._ordereddict.OrderedDict.__setitem__" class="py-name" href="#" onclick="return doclink('link-194', '__setitem__', 'link-194');">__setitem__</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">_per_letter_annotations</tt><tt class="py-op">,</tt> </tt>
<a name="L701"></a><tt class="py-lineno"> 701</tt>  <tt class="py-line">                     <tt class="py-string">"phred_quality"</tt><tt class="py-op">,</tt> <tt class="py-name">quals</tt><tt class="py-op">)</tt> </tt>
<a name="L702"></a><tt class="py-lineno"> 702</tt>  <tt class="py-line">    <tt class="py-comment"># Return the record and then continue...</tt> </tt>
<a name="L703"></a><tt class="py-lineno"> 703</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">record</tt> </tt>
</div><a name="L704"></a><tt class="py-lineno"> 704</tt>  <tt class="py-line"> </tt>
<a name="L705"></a><tt class="py-lineno"> 705</tt>  <tt class="py-line"><tt id="link-195" class="py-name" targets="Variable Bio.SeqIO.SffIO._powers_of_36=Bio.SeqIO.SffIO-module.html#_powers_of_36"><a title="Bio.SeqIO.SffIO._powers_of_36" class="py-name" href="#" onclick="return doclink('link-195', '_powers_of_36', 'link-195');">_powers_of_36</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-number">36</tt> <tt class="py-op">**</tt> <tt id="link-196" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-196', 'i', 'link-196');">i</a></tt> <tt class="py-keyword">for</tt> <tt id="link-197" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-197', 'i', 'link-196');">i</a></tt> <tt class="py-keyword">in</tt> <tt id="link-198" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-198', 'range', 'link-53');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">6</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L706"></a><tt class="py-lineno"> 706</tt>  <tt class="py-line"> </tt>
<a name="L707"></a><tt class="py-lineno"> 707</tt>  <tt class="py-line"> </tt>
<a name="_string_as_base_36"></a><div id="_string_as_base_36-def"><a name="L708"></a><tt class="py-lineno"> 708</tt> <a class="py-toggle" href="#" id="_string_as_base_36-toggle" onclick="return toggle('_string_as_base_36');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO-module.html#_string_as_base_36">_string_as_base_36</a><tt class="py-op">(</tt><tt class="py-param">string</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_string_as_base_36-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_string_as_base_36-expanded"><a name="L709"></a><tt class="py-lineno"> 709</tt>  <tt class="py-line">    <tt class="py-docstring">"""Interpret a string as a base-36 number as per 454 manual."""</tt> </tt>
<a name="L710"></a><tt class="py-lineno"> 710</tt>  <tt class="py-line">    <tt class="py-name">total</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L711"></a><tt class="py-lineno"> 711</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">c</tt><tt class="py-op">,</tt> <tt class="py-name">power</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">string</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt id="link-199" class="py-name"><a title="Bio.SeqIO.SffIO._powers_of_36" class="py-name" href="#" onclick="return doclink('link-199', '_powers_of_36', 'link-195');">_powers_of_36</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L712"></a><tt class="py-lineno"> 712</tt>  <tt class="py-line">        <tt class="py-comment"># For reference: ord('0') = 48, ord('9') = 57</tt> </tt>
<a name="L713"></a><tt class="py-lineno"> 713</tt>  <tt class="py-line">        <tt class="py-comment"># For reference: ord('A') = 65, ord('Z') = 90</tt> </tt>
<a name="L714"></a><tt class="py-lineno"> 714</tt>  <tt class="py-line">        <tt class="py-comment"># For reference: ord('a') = 97, ord('z') = 122</tt> </tt>
<a name="L715"></a><tt class="py-lineno"> 715</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-number">48</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">ord</tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;=</tt> <tt class="py-number">57</tt><tt class="py-op">:</tt> </tt>
<a name="L716"></a><tt class="py-lineno"> 716</tt>  <tt class="py-line">            <tt class="py-name">val</tt> <tt class="py-op">=</tt> <tt class="py-name">ord</tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-number">22</tt>  <tt class="py-comment"># equivalent to: - ord('0') + 26</tt> </tt>
<a name="L717"></a><tt class="py-lineno"> 717</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-number">65</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">ord</tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;=</tt> <tt class="py-number">90</tt><tt class="py-op">:</tt> </tt>
<a name="L718"></a><tt class="py-lineno"> 718</tt>  <tt class="py-line">            <tt class="py-name">val</tt> <tt class="py-op">=</tt> <tt class="py-name">ord</tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-number">65</tt> </tt>
<a name="L719"></a><tt class="py-lineno"> 719</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-number">97</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">ord</tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;=</tt> <tt class="py-number">122</tt><tt class="py-op">:</tt> </tt>
<a name="L720"></a><tt class="py-lineno"> 720</tt>  <tt class="py-line">            <tt class="py-name">val</tt> <tt class="py-op">=</tt> <tt class="py-name">ord</tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-number">97</tt> </tt>
<a name="L721"></a><tt class="py-lineno"> 721</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L722"></a><tt class="py-lineno"> 722</tt>  <tt class="py-line">            <tt class="py-comment"># Invalid character</tt> </tt>
<a name="L723"></a><tt class="py-lineno"> 723</tt>  <tt class="py-line">            <tt class="py-name">val</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L724"></a><tt class="py-lineno"> 724</tt>  <tt class="py-line">        <tt class="py-name">total</tt> <tt class="py-op">+=</tt> <tt class="py-name">val</tt> <tt class="py-op">*</tt> <tt class="py-name">power</tt> </tt>
<a name="L725"></a><tt class="py-lineno"> 725</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">total</tt> </tt>
</div><a name="L726"></a><tt class="py-lineno"> 726</tt>  <tt class="py-line"> </tt>
<a name="L727"></a><tt class="py-lineno"> 727</tt>  <tt class="py-line"> </tt>
<a name="_get_read_xy"></a><div id="_get_read_xy-def"><a name="L728"></a><tt class="py-lineno"> 728</tt> <a class="py-toggle" href="#" id="_get_read_xy-toggle" onclick="return toggle('_get_read_xy');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO-module.html#_get_read_xy">_get_read_xy</a><tt class="py-op">(</tt><tt class="py-param">read_name</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_get_read_xy-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_get_read_xy-expanded"><a name="L729"></a><tt class="py-lineno"> 729</tt>  <tt class="py-line">    <tt class="py-docstring">"""Extract coordinates from last 5 characters of read name."""</tt> </tt>
<a name="L730"></a><tt class="py-lineno"> 730</tt>  <tt class="py-line">    <tt id="link-200" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Pathway.number=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#number"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.number" class="py-name" href="#" onclick="return doclink('link-200', 'number', 'link-200');">number</a></tt> <tt class="py-op">=</tt> <tt id="link-201" class="py-name" targets="Function Bio.SeqIO.SffIO._string_as_base_36()=Bio.SeqIO.SffIO-module.html#_string_as_base_36"><a title="Bio.SeqIO.SffIO._string_as_base_36" class="py-name" href="#" onclick="return doclink('link-201', '_string_as_base_36', 'link-201');">_string_as_base_36</a></tt><tt class="py-op">(</tt><tt class="py-name">read_name</tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L731"></a><tt class="py-lineno"> 731</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">divmod</tt><tt class="py-op">(</tt><tt id="link-202" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.number" class="py-name" href="#" onclick="return doclink('link-202', 'number', 'link-200');">number</a></tt><tt class="py-op">,</tt> <tt class="py-number">4096</tt><tt class="py-op">)</tt> </tt>
</div><a name="L732"></a><tt class="py-lineno"> 732</tt>  <tt class="py-line"> </tt>
<a name="L733"></a><tt class="py-lineno"> 733</tt>  <tt class="py-line"><tt id="link-203" class="py-name" targets="Variable Bio.SeqIO.SffIO._time_denominators=Bio.SeqIO.SffIO-module.html#_time_denominators"><a title="Bio.SeqIO.SffIO._time_denominators" class="py-name" href="#" onclick="return doclink('link-203', '_time_denominators', 'link-203');">_time_denominators</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-number">13</tt> <tt class="py-op">*</tt> <tt class="py-number">32</tt> <tt class="py-op">*</tt> <tt class="py-number">24</tt> <tt class="py-op">*</tt> <tt class="py-number">60</tt> <tt class="py-op">*</tt> <tt class="py-number">60</tt><tt class="py-op">,</tt> </tt>
<a name="L734"></a><tt class="py-lineno"> 734</tt>  <tt class="py-line">                      <tt class="py-number">32</tt> <tt class="py-op">*</tt> <tt class="py-number">24</tt> <tt class="py-op">*</tt> <tt class="py-number">60</tt> <tt class="py-op">*</tt> <tt class="py-number">60</tt><tt class="py-op">,</tt> </tt>
<a name="L735"></a><tt class="py-lineno"> 735</tt>  <tt class="py-line">                      <tt class="py-number">24</tt> <tt class="py-op">*</tt> <tt class="py-number">60</tt> <tt class="py-op">*</tt> <tt class="py-number">60</tt><tt class="py-op">,</tt> </tt>
<a name="L736"></a><tt class="py-lineno"> 736</tt>  <tt class="py-line">                      <tt class="py-number">60</tt> <tt class="py-op">*</tt> <tt class="py-number">60</tt><tt class="py-op">,</tt> </tt>
<a name="L737"></a><tt class="py-lineno"> 737</tt>  <tt class="py-line">                      <tt class="py-number">60</tt><tt class="py-op">]</tt> </tt>
<a name="L738"></a><tt class="py-lineno"> 738</tt>  <tt class="py-line"> </tt>
<a name="L739"></a><tt class="py-lineno"> 739</tt>  <tt class="py-line"> </tt>
<a name="_get_read_time"></a><div id="_get_read_time-def"><a name="L740"></a><tt class="py-lineno"> 740</tt> <a class="py-toggle" href="#" id="_get_read_time-toggle" onclick="return toggle('_get_read_time');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO-module.html#_get_read_time">_get_read_time</a><tt class="py-op">(</tt><tt class="py-param">read_name</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_get_read_time-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_get_read_time-expanded"><a name="L741"></a><tt class="py-lineno"> 741</tt>  <tt class="py-line">    <tt class="py-docstring">"""Extract time from first 6 characters of read name."""</tt> </tt>
<a name="L742"></a><tt class="py-lineno"> 742</tt>  <tt class="py-line">    <tt class="py-name">time_list</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L743"></a><tt class="py-lineno"> 743</tt>  <tt class="py-line">    <tt class="py-name">remainder</tt> <tt class="py-op">=</tt> <tt id="link-204" class="py-name"><a title="Bio.SeqIO.SffIO._string_as_base_36" class="py-name" href="#" onclick="return doclink('link-204', '_string_as_base_36', 'link-201');">_string_as_base_36</a></tt><tt class="py-op">(</tt><tt class="py-name">read_name</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">6</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L744"></a><tt class="py-lineno"> 744</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">denominator</tt> <tt class="py-keyword">in</tt> <tt id="link-205" class="py-name"><a title="Bio.SeqIO.SffIO._time_denominators" class="py-name" href="#" onclick="return doclink('link-205', '_time_denominators', 'link-203');">_time_denominators</a></tt><tt class="py-op">:</tt> </tt>
<a name="L745"></a><tt class="py-lineno"> 745</tt>  <tt class="py-line">        <tt class="py-name">this_term</tt><tt class="py-op">,</tt> <tt class="py-name">remainder</tt> <tt class="py-op">=</tt> <tt class="py-name">divmod</tt><tt class="py-op">(</tt><tt class="py-name">remainder</tt><tt class="py-op">,</tt> <tt class="py-name">denominator</tt><tt class="py-op">)</tt> </tt>
<a name="L746"></a><tt class="py-lineno"> 746</tt>  <tt class="py-line">        <tt class="py-name">time_list</tt><tt class="py-op">.</tt><tt id="link-206" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-206', 'append', 'link-206');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">this_term</tt><tt class="py-op">)</tt> </tt>
<a name="L747"></a><tt class="py-lineno"> 747</tt>  <tt class="py-line">    <tt class="py-name">time_list</tt><tt class="py-op">.</tt><tt id="link-207" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-207', 'append', 'link-206');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">remainder</tt><tt class="py-op">)</tt> </tt>
<a name="L748"></a><tt class="py-lineno"> 748</tt>  <tt class="py-line">    <tt class="py-name">time_list</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">+=</tt> <tt class="py-number">2000</tt> </tt>
<a name="L749"></a><tt class="py-lineno"> 749</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">time_list</tt> </tt>
</div><a name="L750"></a><tt class="py-lineno"> 750</tt>  <tt class="py-line"> </tt>
<a name="L751"></a><tt class="py-lineno"> 751</tt>  <tt class="py-line"> </tt>
<a name="_get_read_region"></a><div id="_get_read_region-def"><a name="L752"></a><tt class="py-lineno"> 752</tt> <a class="py-toggle" href="#" id="_get_read_region-toggle" onclick="return toggle('_get_read_region');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO-module.html#_get_read_region">_get_read_region</a><tt class="py-op">(</tt><tt class="py-param">read_name</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_get_read_region-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_get_read_region-expanded"><a name="L753"></a><tt class="py-lineno"> 753</tt>  <tt class="py-line">    <tt class="py-docstring">"""Extract region from read name."""</tt> </tt>
<a name="L754"></a><tt class="py-lineno"> 754</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">read_name</tt><tt class="py-op">[</tt><tt class="py-number">8</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L755"></a><tt class="py-lineno"> 755</tt>  <tt class="py-line"> </tt>
<a name="L756"></a><tt class="py-lineno"> 756</tt>  <tt class="py-line"> </tt>
<a name="_sff_read_raw_record"></a><div id="_sff_read_raw_record-def"><a name="L757"></a><tt class="py-lineno"> 757</tt> <a class="py-toggle" href="#" id="_sff_read_raw_record-toggle" onclick="return toggle('_sff_read_raw_record');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO-module.html#_sff_read_raw_record">_sff_read_raw_record</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">number_of_flows_per_read</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_sff_read_raw_record-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_sff_read_raw_record-expanded"><a name="L758"></a><tt class="py-lineno"> 758</tt>  <tt class="py-line">    <tt class="py-docstring">"""Extract the next read in the file as a raw (bytes) string (PRIVATE)."""</tt> </tt>
<a name="L759"></a><tt class="py-lineno"> 759</tt>  <tt class="py-line">    <tt class="py-name">read_header_fmt</tt> <tt class="py-op">=</tt> <tt class="py-string">'&gt;2HI'</tt> </tt>
<a name="L760"></a><tt class="py-lineno"> 760</tt>  <tt class="py-line">    <tt class="py-name">read_header_size</tt> <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-name">read_header_fmt</tt><tt class="py-op">)</tt> </tt>
<a name="L761"></a><tt class="py-lineno"> 761</tt>  <tt class="py-line">    <tt class="py-name">read_flow_fmt</tt> <tt class="py-op">=</tt> <tt class="py-string">"&gt;%iH"</tt> <tt class="py-op">%</tt> <tt class="py-name">number_of_flows_per_read</tt> </tt>
<a name="L762"></a><tt class="py-lineno"> 762</tt>  <tt class="py-line">    <tt class="py-name">read_flow_size</tt> <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-name">read_flow_fmt</tt><tt class="py-op">)</tt> </tt>
<a name="L763"></a><tt class="py-lineno"> 763</tt>  <tt class="py-line"> </tt>
<a name="L764"></a><tt class="py-lineno"> 764</tt>  <tt class="py-line">    <tt class="py-name">raw</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-208" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-208', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">read_header_size</tt><tt class="py-op">)</tt> </tt>
<a name="L765"></a><tt class="py-lineno"> 765</tt>  <tt class="py-line">    <tt class="py-name">read_header_length</tt><tt class="py-op">,</tt> <tt class="py-name">name_length</tt><tt class="py-op">,</tt> <tt class="py-name">seq_len</tt> \ </tt>
<a name="L766"></a><tt class="py-lineno"> 766</tt>  <tt class="py-line">        <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">unpack</tt><tt class="py-op">(</tt><tt class="py-name">read_header_fmt</tt><tt class="py-op">,</tt> <tt class="py-name">raw</tt><tt class="py-op">)</tt> </tt>
<a name="L767"></a><tt class="py-lineno"> 767</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">read_header_length</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">10</tt> <tt class="py-keyword">or</tt> <tt class="py-name">read_header_length</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt> <tt class="py-op">!=</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L768"></a><tt class="py-lineno"> 768</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Malformed read header, says length is %i"</tt> </tt>
<a name="L769"></a><tt class="py-lineno"> 769</tt>  <tt class="py-line">                         <tt class="py-op">%</tt> <tt class="py-name">read_header_length</tt><tt class="py-op">)</tt> </tt>
<a name="L770"></a><tt class="py-lineno"> 770</tt>  <tt class="py-line">    <tt class="py-comment"># now the four clip values (4H = 8 bytes), and read name</tt> </tt>
<a name="L771"></a><tt class="py-lineno"> 771</tt>  <tt class="py-line">    <tt class="py-name">raw</tt> <tt class="py-op">+=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-209" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-209', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-number">8</tt> <tt class="py-op">+</tt> <tt class="py-name">name_length</tt><tt class="py-op">)</tt> </tt>
<a name="L772"></a><tt class="py-lineno"> 772</tt>  <tt class="py-line">    <tt class="py-comment"># and any padding (remainder of header)</tt> </tt>
<a name="L773"></a><tt class="py-lineno"> 773</tt>  <tt class="py-line">    <tt class="py-name">padding</tt> <tt class="py-op">=</tt> <tt class="py-name">read_header_length</tt> <tt class="py-op">-</tt> <tt class="py-name">read_header_size</tt> <tt class="py-op">-</tt> <tt class="py-number">8</tt> <tt class="py-op">-</tt> <tt class="py-name">name_length</tt> </tt>
<a name="L774"></a><tt class="py-lineno"> 774</tt>  <tt class="py-line">    <tt class="py-name">pad</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-210" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-210', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">padding</tt><tt class="py-op">)</tt> </tt>
<a name="L775"></a><tt class="py-lineno"> 775</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">pad</tt><tt class="py-op">.</tt><tt id="link-211" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-211', 'count', 'link-45');">count</a></tt><tt class="py-op">(</tt><tt id="link-212" class="py-name"><a title="Bio.SeqIO.SffIO._null" class="py-name" href="#" onclick="return doclink('link-212', '_null', 'link-16');">_null</a></tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">padding</tt><tt class="py-op">:</tt> </tt>
<a name="L776"></a><tt class="py-lineno"> 776</tt>  <tt class="py-line">        <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L777"></a><tt class="py-lineno"> 777</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt id="link-213" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-213', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-214" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-214', 'BiopythonParserWarning', 'link-48');">BiopythonParserWarning</a></tt> </tt>
<a name="L778"></a><tt class="py-lineno"> 778</tt>  <tt class="py-line">        <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Your SFF file is invalid, post name %i "</tt> </tt>
<a name="L779"></a><tt class="py-lineno"> 779</tt>  <tt class="py-line">                      <tt class="py-string">"byte padding region contained data"</tt> <tt class="py-op">%</tt> <tt class="py-name">padding</tt><tt class="py-op">,</tt> </tt>
<a name="L780"></a><tt class="py-lineno"> 780</tt>  <tt class="py-line">                      <tt id="link-215" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-215', 'BiopythonParserWarning', 'link-48');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L781"></a><tt class="py-lineno"> 781</tt>  <tt class="py-line">    <tt class="py-name">raw</tt> <tt class="py-op">+=</tt> <tt class="py-name">pad</tt> </tt>
<a name="L782"></a><tt class="py-lineno"> 782</tt>  <tt class="py-line">    <tt class="py-comment"># now the flowgram values, flowgram index, bases and qualities</tt> </tt>
<a name="L783"></a><tt class="py-lineno"> 783</tt>  <tt class="py-line">    <tt class="py-name">raw</tt> <tt class="py-op">+=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-216" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-216', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">read_flow_size</tt> <tt class="py-op">+</tt> <tt class="py-name">seq_len</tt> <tt class="py-op">*</tt> <tt class="py-number">3</tt><tt class="py-op">)</tt> </tt>
<a name="L784"></a><tt class="py-lineno"> 784</tt>  <tt class="py-line">    <tt class="py-name">padding</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">read_flow_size</tt> <tt class="py-op">+</tt> <tt class="py-name">seq_len</tt> <tt class="py-op">*</tt> <tt class="py-number">3</tt><tt class="py-op">)</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt> </tt>
<a name="L785"></a><tt class="py-lineno"> 785</tt>  <tt class="py-line">    <tt class="py-comment"># now any padding...</tt> </tt>
<a name="L786"></a><tt class="py-lineno"> 786</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">padding</tt><tt class="py-op">:</tt> </tt>
<a name="L787"></a><tt class="py-lineno"> 787</tt>  <tt class="py-line">        <tt class="py-name">padding</tt> <tt class="py-op">=</tt> <tt class="py-number">8</tt> <tt class="py-op">-</tt> <tt class="py-name">padding</tt> </tt>
<a name="L788"></a><tt class="py-lineno"> 788</tt>  <tt class="py-line">        <tt class="py-name">pad</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-217" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-217', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">padding</tt><tt class="py-op">)</tt> </tt>
<a name="L789"></a><tt class="py-lineno"> 789</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">pad</tt><tt class="py-op">.</tt><tt id="link-218" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-218', 'count', 'link-45');">count</a></tt><tt class="py-op">(</tt><tt id="link-219" class="py-name"><a title="Bio.SeqIO.SffIO._null" class="py-name" href="#" onclick="return doclink('link-219', '_null', 'link-16');">_null</a></tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">padding</tt><tt class="py-op">:</tt> </tt>
<a name="L790"></a><tt class="py-lineno"> 790</tt>  <tt class="py-line">            <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L791"></a><tt class="py-lineno"> 791</tt>  <tt class="py-line">            <tt class="py-keyword">from</tt> <tt id="link-220" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-220', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-221" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-221', 'BiopythonParserWarning', 'link-48');">BiopythonParserWarning</a></tt> </tt>
<a name="L792"></a><tt class="py-lineno"> 792</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Your SFF file is invalid, post quality %i "</tt> </tt>
<a name="L793"></a><tt class="py-lineno"> 793</tt>  <tt class="py-line">                          <tt class="py-string">"byte padding region contained data"</tt> <tt class="py-op">%</tt> <tt class="py-name">padding</tt><tt class="py-op">,</tt> </tt>
<a name="L794"></a><tt class="py-lineno"> 794</tt>  <tt class="py-line">                          <tt id="link-222" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-222', 'BiopythonParserWarning', 'link-48');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L795"></a><tt class="py-lineno"> 795</tt>  <tt class="py-line">        <tt class="py-name">raw</tt> <tt class="py-op">+=</tt> <tt class="py-name">pad</tt> </tt>
<a name="L796"></a><tt class="py-lineno"> 796</tt>  <tt class="py-line">    <tt class="py-comment"># Return the raw bytes</tt> </tt>
<a name="L797"></a><tt class="py-lineno"> 797</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">raw</tt> </tt>
</div><a name="L798"></a><tt class="py-lineno"> 798</tt>  <tt class="py-line"> </tt>
<a name="L799"></a><tt class="py-lineno"> 799</tt>  <tt class="py-line"> </tt>
<a name="_AddTellHandle"></a><div id="_AddTellHandle-def"><a name="L800"></a><tt class="py-lineno"> 800</tt> <a class="py-toggle" href="#" id="_AddTellHandle-toggle" onclick="return toggle('_AddTellHandle');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO._AddTellHandle-class.html">_AddTellHandle</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_AddTellHandle-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_AddTellHandle-expanded"><a name="L801"></a><tt class="py-lineno"> 801</tt>  <tt class="py-line">    <tt class="py-docstring">"""Wrapper for handles which do not support the tell method (PRIVATE).</tt> </tt>
<a name="L802"></a><tt class="py-lineno"> 802</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L803"></a><tt class="py-lineno"> 803</tt>  <tt class="py-line"><tt class="py-docstring">    Intended for use with things like network handles where tell (and reverse</tt> </tt>
<a name="L804"></a><tt class="py-lineno"> 804</tt>  <tt class="py-line"><tt class="py-docstring">    seek) are not supported. The SFF file needs to track the current offset in</tt> </tt>
<a name="L805"></a><tt class="py-lineno"> 805</tt>  <tt class="py-line"><tt class="py-docstring">    order to deal with the index block.</tt> </tt>
<a name="L806"></a><tt class="py-lineno"> 806</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="_AddTellHandle.__init__"></a><div id="_AddTellHandle.__init__-def"><a name="L807"></a><tt class="py-lineno"> 807</tt> <a class="py-toggle" href="#" id="_AddTellHandle.__init__-toggle" onclick="return toggle('_AddTellHandle.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO._AddTellHandle-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_AddTellHandle.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_AddTellHandle.__init__-expanded"><a name="L808"></a><tt class="py-lineno"> 808</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_handle</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt> </tt>
<a name="L809"></a><tt class="py-lineno"> 809</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_offset</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
</div><a name="L810"></a><tt class="py-lineno"> 810</tt>  <tt class="py-line"> </tt>
<a name="_AddTellHandle.read"></a><div id="_AddTellHandle.read-def"><a name="L811"></a><tt class="py-lineno"> 811</tt> <a class="py-toggle" href="#" id="_AddTellHandle.read-toggle" onclick="return toggle('_AddTellHandle.read');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO._AddTellHandle-class.html#read">read</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">length</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_AddTellHandle.read-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_AddTellHandle.read-expanded"><a name="L812"></a><tt class="py-lineno"> 812</tt>  <tt class="py-line">        <tt id="link-223" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-223', 'data', 'link-31');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_handle</tt><tt class="py-op">.</tt><tt id="link-224" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-224', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt id="link-225" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.length()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#length"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-225', 'length', 'link-225');">length</a></tt><tt class="py-op">)</tt> </tt>
<a name="L813"></a><tt class="py-lineno"> 813</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_offset</tt> <tt class="py-op">+=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-226" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-226', 'data', 'link-31');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L814"></a><tt class="py-lineno"> 814</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-227" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-227', 'data', 'link-31');">data</a></tt> </tt>
</div><a name="L815"></a><tt class="py-lineno"> 815</tt>  <tt class="py-line"> </tt>
<a name="_AddTellHandle.tell"></a><div id="_AddTellHandle.tell-def"><a name="L816"></a><tt class="py-lineno"> 816</tt> <a class="py-toggle" href="#" id="_AddTellHandle.tell-toggle" onclick="return toggle('_AddTellHandle.tell');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO._AddTellHandle-class.html#tell">tell</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_AddTellHandle.tell-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_AddTellHandle.tell-expanded"><a name="L817"></a><tt class="py-lineno"> 817</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_offset</tt> </tt>
</div><a name="L818"></a><tt class="py-lineno"> 818</tt>  <tt class="py-line"> </tt>
<a name="_AddTellHandle.seek"></a><div id="_AddTellHandle.seek-def"><a name="L819"></a><tt class="py-lineno"> 819</tt> <a class="py-toggle" href="#" id="_AddTellHandle.seek-toggle" onclick="return toggle('_AddTellHandle.seek');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO._AddTellHandle-class.html#seek">seek</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">offset</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_AddTellHandle.seek-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_AddTellHandle.seek-expanded"><a name="L820"></a><tt class="py-lineno"> 820</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">offset</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_offset</tt><tt class="py-op">:</tt> </tt>
<a name="L821"></a><tt class="py-lineno"> 821</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">RuntimeError</tt><tt class="py-op">(</tt><tt class="py-string">"Can't seek backwards"</tt><tt class="py-op">)</tt> </tt>
<a name="L822"></a><tt class="py-lineno"> 822</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_handle</tt><tt class="py-op">.</tt><tt id="link-228" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-228', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">offset</tt> <tt class="py-op">-</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_offset</tt><tt class="py-op">)</tt> </tt>
</div><a name="L823"></a><tt class="py-lineno"> 823</tt>  <tt class="py-line"> </tt>
<a name="_AddTellHandle.close"></a><div id="_AddTellHandle.close-def"><a name="L824"></a><tt class="py-lineno"> 824</tt> <a class="py-toggle" href="#" id="_AddTellHandle.close-toggle" onclick="return toggle('_AddTellHandle.close');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO._AddTellHandle-class.html#close">close</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_AddTellHandle.close-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_AddTellHandle.close-expanded"><a name="L825"></a><tt class="py-lineno"> 825</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_handle</tt><tt class="py-op">.</tt><tt id="link-229" class="py-name" targets="Method Bio.File._IndexedSeqFileDict.close()=Bio.File._IndexedSeqFileDict-class.html#close,Method Bio.File._SQLiteManySeqFilesDict.close()=Bio.File._SQLiteManySeqFilesDict-class.html#close,Method Bio.SeqIO.SffIO._AddTellHandle.close()=Bio.SeqIO.SffIO._AddTellHandle-class.html#close,Method Bio.bgzf.BgzfReader.close()=Bio.bgzf.BgzfReader-class.html#close,Method Bio.bgzf.BgzfWriter.close()=Bio.bgzf.BgzfWriter-class.html#close,Method BioSQL.BioSeqDatabase.Adaptor.close()=BioSQL.BioSeqDatabase.Adaptor-class.html#close,Method BioSQL.BioSeqDatabase.DBServer.close()=BioSQL.BioSeqDatabase.DBServer-class.html#close"><a title="Bio.File._IndexedSeqFileDict.close
Bio.File._SQLiteManySeqFilesDict.close
Bio.SeqIO.SffIO._AddTellHandle.close
Bio.bgzf.BgzfReader.close
Bio.bgzf.BgzfWriter.close
BioSQL.BioSeqDatabase.Adaptor.close
BioSQL.BioSeqDatabase.DBServer.close" class="py-name" href="#" onclick="return doclink('link-229', 'close', 'link-229');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L826"></a><tt class="py-lineno"> 826</tt>  <tt class="py-line"> </tt>
<a name="L827"></a><tt class="py-lineno"> 827</tt>  <tt class="py-line"> </tt>
<a name="L828"></a><tt class="py-lineno"> 828</tt>  <tt class="py-line"><tt class="py-comment"># This is a generator function!</tt> </tt>
<a name="SffIterator"></a><div id="SffIterator-def"><a name="L829"></a><tt class="py-lineno"> 829</tt> <a class="py-toggle" href="#" id="SffIterator-toggle" onclick="return toggle('SffIterator');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO-module.html#SffIterator">SffIterator</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt id="link-230" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-230', 'Alphabet', 'link-5');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-231" class="py-name" targets="Variable Bio.Alphabet.generic_dna=Bio.Alphabet-module.html#generic_dna"><a title="Bio.Alphabet.generic_dna" class="py-name" href="#" onclick="return doclink('link-231', 'generic_dna', 'link-231');">generic_dna</a></tt><tt class="py-op">,</tt> <tt class="py-param">trim</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SffIterator-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="SffIterator-expanded"><a name="L830"></a><tt class="py-lineno"> 830</tt>  <tt class="py-line">    <tt class="py-docstring">"""Iterate over Standard Flowgram Format (SFF) reads (as SeqRecord objects).</tt> </tt>
<a name="L831"></a><tt class="py-lineno"> 831</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L832"></a><tt class="py-lineno"> 832</tt>  <tt class="py-line"><tt class="py-docstring">        - handle - input file, an SFF file, e.g. from Roche 454 sequencing.</tt> </tt>
<a name="L833"></a><tt class="py-lineno"> 833</tt>  <tt class="py-line"><tt class="py-docstring">          This must NOT be opened in universal read lines mode!</tt> </tt>
<a name="L834"></a><tt class="py-lineno"> 834</tt>  <tt class="py-line"><tt class="py-docstring">        - alphabet - optional alphabet, defaults to generic DNA.</tt> </tt>
<a name="L835"></a><tt class="py-lineno"> 835</tt>  <tt class="py-line"><tt class="py-docstring">        - trim - should the sequences be trimmed?</tt> </tt>
<a name="L836"></a><tt class="py-lineno"> 836</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L837"></a><tt class="py-lineno"> 837</tt>  <tt class="py-line"><tt class="py-docstring">    The resulting SeqRecord objects should match those from a paired FASTA</tt> </tt>
<a name="L838"></a><tt class="py-lineno"> 838</tt>  <tt class="py-line"><tt class="py-docstring">    and QUAL file converted from the SFF file using the Roche 454 tool</tt> </tt>
<a name="L839"></a><tt class="py-lineno"> 839</tt>  <tt class="py-line"><tt class="py-docstring">    ssfinfo. i.e. The sequence will be mixed case, with the trim regions</tt> </tt>
<a name="L840"></a><tt class="py-lineno"> 840</tt>  <tt class="py-line"><tt class="py-docstring">    shown in lower case.</tt> </tt>
<a name="L841"></a><tt class="py-lineno"> 841</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L842"></a><tt class="py-lineno"> 842</tt>  <tt class="py-line"><tt class="py-docstring">    This function is used internally via the Bio.SeqIO functions:</tt> </tt>
<a name="L843"></a><tt class="py-lineno"> 843</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L844"></a><tt class="py-lineno"> 844</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L845"></a><tt class="py-lineno"> 845</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; for record in SeqIO.parse("Roche/E3MFGYR02_random_10_reads.sff", "sff"):</tt> </tt>
<a name="L846"></a><tt class="py-lineno"> 846</tt>  <tt class="py-line"><tt class="py-docstring">    ...     print("%s %i" % (record.id, len(record)))</tt> </tt>
<a name="L847"></a><tt class="py-lineno"> 847</tt>  <tt class="py-line"><tt class="py-docstring">    ...</tt> </tt>
<a name="L848"></a><tt class="py-lineno"> 848</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02JWQ7T 265</tt> </tt>
<a name="L849"></a><tt class="py-lineno"> 849</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02JA6IL 271</tt> </tt>
<a name="L850"></a><tt class="py-lineno"> 850</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02JHD4H 310</tt> </tt>
<a name="L851"></a><tt class="py-lineno"> 851</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02GFKUC 299</tt> </tt>
<a name="L852"></a><tt class="py-lineno"> 852</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02FTGED 281</tt> </tt>
<a name="L853"></a><tt class="py-lineno"> 853</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02FR9G7 261</tt> </tt>
<a name="L854"></a><tt class="py-lineno"> 854</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02GAZMS 278</tt> </tt>
<a name="L855"></a><tt class="py-lineno"> 855</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02HHZ8O 221</tt> </tt>
<a name="L856"></a><tt class="py-lineno"> 856</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02GPGB1 269</tt> </tt>
<a name="L857"></a><tt class="py-lineno"> 857</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02F7Z7G 219</tt> </tt>
<a name="L858"></a><tt class="py-lineno"> 858</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L859"></a><tt class="py-lineno"> 859</tt>  <tt class="py-line"><tt class="py-docstring">    You can also call it directly:</tt> </tt>
<a name="L860"></a><tt class="py-lineno"> 860</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L861"></a><tt class="py-lineno"> 861</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; with open("Roche/E3MFGYR02_random_10_reads.sff", "rb") as handle:</tt> </tt>
<a name="L862"></a><tt class="py-lineno"> 862</tt>  <tt class="py-line"><tt class="py-docstring">    ...     for record in SffIterator(handle):</tt> </tt>
<a name="L863"></a><tt class="py-lineno"> 863</tt>  <tt class="py-line"><tt class="py-docstring">    ...         print("%s %i" % (record.id, len(record)))</tt> </tt>
<a name="L864"></a><tt class="py-lineno"> 864</tt>  <tt class="py-line"><tt class="py-docstring">    ...</tt> </tt>
<a name="L865"></a><tt class="py-lineno"> 865</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02JWQ7T 265</tt> </tt>
<a name="L866"></a><tt class="py-lineno"> 866</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02JA6IL 271</tt> </tt>
<a name="L867"></a><tt class="py-lineno"> 867</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02JHD4H 310</tt> </tt>
<a name="L868"></a><tt class="py-lineno"> 868</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02GFKUC 299</tt> </tt>
<a name="L869"></a><tt class="py-lineno"> 869</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02FTGED 281</tt> </tt>
<a name="L870"></a><tt class="py-lineno"> 870</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02FR9G7 261</tt> </tt>
<a name="L871"></a><tt class="py-lineno"> 871</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02GAZMS 278</tt> </tt>
<a name="L872"></a><tt class="py-lineno"> 872</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02HHZ8O 221</tt> </tt>
<a name="L873"></a><tt class="py-lineno"> 873</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02GPGB1 269</tt> </tt>
<a name="L874"></a><tt class="py-lineno"> 874</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02F7Z7G 219</tt> </tt>
<a name="L875"></a><tt class="py-lineno"> 875</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L876"></a><tt class="py-lineno"> 876</tt>  <tt class="py-line"><tt class="py-docstring">    Or, with the trim option:</tt> </tt>
<a name="L877"></a><tt class="py-lineno"> 877</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L878"></a><tt class="py-lineno"> 878</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; with open("Roche/E3MFGYR02_random_10_reads.sff", "rb") as handle:</tt> </tt>
<a name="L879"></a><tt class="py-lineno"> 879</tt>  <tt class="py-line"><tt class="py-docstring">    ...     for record in SffIterator(handle, trim=True):</tt> </tt>
<a name="L880"></a><tt class="py-lineno"> 880</tt>  <tt class="py-line"><tt class="py-docstring">    ...         print("%s %i" % (record.id, len(record)))</tt> </tt>
<a name="L881"></a><tt class="py-lineno"> 881</tt>  <tt class="py-line"><tt class="py-docstring">    ...</tt> </tt>
<a name="L882"></a><tt class="py-lineno"> 882</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02JWQ7T 260</tt> </tt>
<a name="L883"></a><tt class="py-lineno"> 883</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02JA6IL 265</tt> </tt>
<a name="L884"></a><tt class="py-lineno"> 884</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02JHD4H 292</tt> </tt>
<a name="L885"></a><tt class="py-lineno"> 885</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02GFKUC 295</tt> </tt>
<a name="L886"></a><tt class="py-lineno"> 886</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02FTGED 277</tt> </tt>
<a name="L887"></a><tt class="py-lineno"> 887</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02FR9G7 256</tt> </tt>
<a name="L888"></a><tt class="py-lineno"> 888</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02GAZMS 271</tt> </tt>
<a name="L889"></a><tt class="py-lineno"> 889</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02HHZ8O 150</tt> </tt>
<a name="L890"></a><tt class="py-lineno"> 890</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02GPGB1 221</tt> </tt>
<a name="L891"></a><tt class="py-lineno"> 891</tt>  <tt class="py-line"><tt class="py-docstring">    E3MFGYR02F7Z7G 130</tt> </tt>
<a name="L892"></a><tt class="py-lineno"> 892</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L893"></a><tt class="py-lineno"> 893</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L894"></a><tt class="py-lineno"> 894</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-232" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-232', 'Alphabet', 'link-5');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-233" class="py-name" targets="Function Bio.Alphabet._get_base_alphabet()=Bio.Alphabet-module.html#_get_base_alphabet"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-233', '_get_base_alphabet', 'link-233');">_get_base_alphabet</a></tt><tt class="py-op">(</tt><tt id="link-234" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-234', 'alphabet', 'link-186');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L895"></a><tt class="py-lineno"> 895</tt>  <tt class="py-line">                  <tt id="link-235" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-235', 'Alphabet', 'link-5');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-236" class="py-name" targets="Class Bio.Alphabet.ProteinAlphabet=Bio.Alphabet.ProteinAlphabet-class.html"><a title="Bio.Alphabet.ProteinAlphabet" class="py-name" href="#" onclick="return doclink('link-236', 'ProteinAlphabet', 'link-236');">ProteinAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L896"></a><tt class="py-lineno"> 896</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid alphabet, SFF files do not hold proteins."</tt><tt class="py-op">)</tt> </tt>
<a name="L897"></a><tt class="py-lineno"> 897</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-237" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-237', 'Alphabet', 'link-5');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-238" class="py-name"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-238', '_get_base_alphabet', 'link-233');">_get_base_alphabet</a></tt><tt class="py-op">(</tt><tt id="link-239" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-239', 'alphabet', 'link-186');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L898"></a><tt class="py-lineno"> 898</tt>  <tt class="py-line">                  <tt id="link-240" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-240', 'Alphabet', 'link-5');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-241" class="py-name" targets="Class Bio.Alphabet.RNAAlphabet=Bio.Alphabet.RNAAlphabet-class.html"><a title="Bio.Alphabet.RNAAlphabet" class="py-name" href="#" onclick="return doclink('link-241', 'RNAAlphabet', 'link-241');">RNAAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L899"></a><tt class="py-lineno"> 899</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid alphabet, SFF files do not hold RNA."</tt><tt class="py-op">)</tt> </tt>
<a name="L900"></a><tt class="py-lineno"> 900</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L901"></a><tt class="py-lineno"> 901</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-number">0</tt> <tt class="py-op">==</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-242" class="py-name"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-242', 'tell', 'link-54');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"Not at start of file, offset %i"</tt> <tt class="py-op">%</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-243" class="py-name"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-243', 'tell', 'link-54');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L902"></a><tt class="py-lineno"> 902</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L903"></a><tt class="py-lineno"> 903</tt>  <tt class="py-line">        <tt class="py-comment"># Probably a network handle or something like that</tt> </tt>
<a name="L904"></a><tt class="py-lineno"> 904</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt id="link-244" class="py-name" targets="Class Bio.SeqIO.SffIO._AddTellHandle=Bio.SeqIO.SffIO._AddTellHandle-class.html"><a title="Bio.SeqIO.SffIO._AddTellHandle" class="py-name" href="#" onclick="return doclink('link-244', '_AddTellHandle', 'link-244');">_AddTellHandle</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L905"></a><tt class="py-lineno"> 905</tt>  <tt class="py-line">    <tt class="py-name">header_length</tt><tt class="py-op">,</tt> <tt class="py-name">index_offset</tt><tt class="py-op">,</tt> <tt class="py-name">index_length</tt><tt class="py-op">,</tt> <tt class="py-name">number_of_reads</tt><tt class="py-op">,</tt> \ </tt>
<a name="L906"></a><tt class="py-lineno"> 906</tt>  <tt class="py-line">        <tt class="py-name">number_of_flows_per_read</tt><tt class="py-op">,</tt> <tt class="py-name">flow_chars</tt><tt class="py-op">,</tt> <tt class="py-name">key_sequence</tt> \ </tt>
<a name="L907"></a><tt class="py-lineno"> 907</tt>  <tt class="py-line">        <tt class="py-op">=</tt> <tt id="link-245" class="py-name"><a title="Bio.SeqIO.SffIO._sff_file_header" class="py-name" href="#" onclick="return doclink('link-245', '_sff_file_header', 'link-51');">_sff_file_header</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L908"></a><tt class="py-lineno"> 908</tt>  <tt class="py-line">    <tt class="py-comment"># Now on to the reads...</tt> </tt>
<a name="L909"></a><tt class="py-lineno"> 909</tt>  <tt class="py-line">    <tt class="py-comment"># the read header format (fixed part):</tt> </tt>
<a name="L910"></a><tt class="py-lineno"> 910</tt>  <tt class="py-line">    <tt class="py-comment"># read_header_length     H</tt> </tt>
<a name="L911"></a><tt class="py-lineno"> 911</tt>  <tt class="py-line">    <tt class="py-comment"># name_length            H</tt> </tt>
<a name="L912"></a><tt class="py-lineno"> 912</tt>  <tt class="py-line">    <tt class="py-comment"># seq_len                I</tt> </tt>
<a name="L913"></a><tt class="py-lineno"> 913</tt>  <tt class="py-line">    <tt class="py-comment"># clip_qual_left         H</tt> </tt>
<a name="L914"></a><tt class="py-lineno"> 914</tt>  <tt class="py-line">    <tt class="py-comment"># clip_qual_right        H</tt> </tt>
<a name="L915"></a><tt class="py-lineno"> 915</tt>  <tt class="py-line">    <tt class="py-comment"># clip_adapter_left      H</tt> </tt>
<a name="L916"></a><tt class="py-lineno"> 916</tt>  <tt class="py-line">    <tt class="py-comment"># clip_adapter_right     H</tt> </tt>
<a name="L917"></a><tt class="py-lineno"> 917</tt>  <tt class="py-line">    <tt class="py-comment"># [rest of read header depends on the name length etc]</tt> </tt>
<a name="L918"></a><tt class="py-lineno"> 918</tt>  <tt class="py-line">    <tt class="py-name">read_header_fmt</tt> <tt class="py-op">=</tt> <tt class="py-string">'&gt;2HI4H'</tt> </tt>
<a name="L919"></a><tt class="py-lineno"> 919</tt>  <tt class="py-line">    <tt class="py-name">read_header_size</tt> <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-name">read_header_fmt</tt><tt class="py-op">)</tt> </tt>
<a name="L920"></a><tt class="py-lineno"> 920</tt>  <tt class="py-line">    <tt class="py-name">read_flow_fmt</tt> <tt class="py-op">=</tt> <tt class="py-string">"&gt;%iH"</tt> <tt class="py-op">%</tt> <tt class="py-name">number_of_flows_per_read</tt> </tt>
<a name="L921"></a><tt class="py-lineno"> 921</tt>  <tt class="py-line">    <tt class="py-name">read_flow_size</tt> <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-name">read_flow_fmt</tt><tt class="py-op">)</tt> </tt>
<a name="L922"></a><tt class="py-lineno"> 922</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-number">1</tt> <tt class="py-op">==</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-string">"&gt;B"</tt><tt class="py-op">)</tt> </tt>
<a name="L923"></a><tt class="py-lineno"> 923</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-number">1</tt> <tt class="py-op">==</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-string">"&gt;s"</tt><tt class="py-op">)</tt> </tt>
<a name="L924"></a><tt class="py-lineno"> 924</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-number">1</tt> <tt class="py-op">==</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-string">"&gt;c"</tt><tt class="py-op">)</tt> </tt>
<a name="L925"></a><tt class="py-lineno"> 925</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">read_header_size</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt>  <tt class="py-comment"># Important for padding calc later!</tt> </tt>
<a name="L926"></a><tt class="py-lineno"> 926</tt>  <tt class="py-line">    <tt class="py-comment"># The spec allows for the index block to be before or even in the middle</tt> </tt>
<a name="L927"></a><tt class="py-lineno"> 927</tt>  <tt class="py-line">    <tt class="py-comment"># of the reads. We can check that if we keep track of our position</tt> </tt>
<a name="L928"></a><tt class="py-lineno"> 928</tt>  <tt class="py-line">    <tt class="py-comment"># in the file...</tt> </tt>
<a name="L929"></a><tt class="py-lineno"> 929</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt id="link-246" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-246', 'read', 'link-32');">read</a></tt> <tt class="py-keyword">in</tt> <tt id="link-247" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-247', 'range', 'link-53');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">number_of_reads</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L930"></a><tt class="py-lineno"> 930</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">index_offset</tt> <tt class="py-keyword">and</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-248" class="py-name"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-248', 'tell', 'link-54');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">index_offset</tt><tt class="py-op">:</tt> </tt>
<a name="L931"></a><tt class="py-lineno"> 931</tt>  <tt class="py-line">            <tt class="py-name">offset</tt> <tt class="py-op">=</tt> <tt class="py-name">index_offset</tt> <tt class="py-op">+</tt> <tt class="py-name">index_length</tt> </tt>
<a name="L932"></a><tt class="py-lineno"> 932</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">offset</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt><tt class="py-op">:</tt> </tt>
<a name="L933"></a><tt class="py-lineno"> 933</tt>  <tt class="py-line">                <tt class="py-name">offset</tt> <tt class="py-op">+=</tt> <tt class="py-number">8</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">offset</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt><tt class="py-op">)</tt> </tt>
<a name="L934"></a><tt class="py-lineno"> 934</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">offset</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt> </tt>
<a name="L935"></a><tt class="py-lineno"> 935</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-249" class="py-name"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-249', 'seek', 'link-50');">seek</a></tt><tt class="py-op">(</tt><tt class="py-name">offset</tt><tt class="py-op">)</tt> </tt>
<a name="L936"></a><tt class="py-lineno"> 936</tt>  <tt class="py-line">            <tt class="py-comment"># Now that we've done this, we don't need to do it again. Clear</tt> </tt>
<a name="L937"></a><tt class="py-lineno"> 937</tt>  <tt class="py-line">            <tt class="py-comment"># the index_offset so we can skip extra handle.tell() calls:</tt> </tt>
<a name="L938"></a><tt class="py-lineno"> 938</tt>  <tt class="py-line">            <tt class="py-name">index_offset</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L939"></a><tt class="py-lineno"> 939</tt>  <tt class="py-line">        <tt class="py-keyword">yield</tt> <tt id="link-250" class="py-name" targets="Function Bio.SeqIO.SffIO._sff_read_seq_record()=Bio.SeqIO.SffIO-module.html#_sff_read_seq_record"><a title="Bio.SeqIO.SffIO._sff_read_seq_record" class="py-name" href="#" onclick="return doclink('link-250', '_sff_read_seq_record', 'link-250');">_sff_read_seq_record</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> </tt>
<a name="L940"></a><tt class="py-lineno"> 940</tt>  <tt class="py-line">                                   <tt class="py-name">number_of_flows_per_read</tt><tt class="py-op">,</tt> </tt>
<a name="L941"></a><tt class="py-lineno"> 941</tt>  <tt class="py-line">                                   <tt class="py-name">flow_chars</tt><tt class="py-op">,</tt> </tt>
<a name="L942"></a><tt class="py-lineno"> 942</tt>  <tt class="py-line">                                   <tt class="py-name">key_sequence</tt><tt class="py-op">,</tt> </tt>
<a name="L943"></a><tt class="py-lineno"> 943</tt>  <tt class="py-line">                                   <tt id="link-251" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-251', 'alphabet', 'link-186');">alphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L944"></a><tt class="py-lineno"> 944</tt>  <tt class="py-line">                                   <tt class="py-name">trim</tt><tt class="py-op">)</tt> </tt>
<a name="L945"></a><tt class="py-lineno"> 945</tt>  <tt class="py-line">    <tt id="link-252" class="py-name" targets="Function Bio.SeqIO.SffIO._check_eof()=Bio.SeqIO.SffIO-module.html#_check_eof"><a title="Bio.SeqIO.SffIO._check_eof" class="py-name" href="#" onclick="return doclink('link-252', '_check_eof', 'link-252');">_check_eof</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-name">index_offset</tt><tt class="py-op">,</tt> <tt class="py-name">index_length</tt><tt class="py-op">)</tt> </tt>
</div><a name="L946"></a><tt class="py-lineno"> 946</tt>  <tt class="py-line"> </tt>
<a name="L947"></a><tt class="py-lineno"> 947</tt>  <tt class="py-line"> </tt>
<a name="_check_eof"></a><div id="_check_eof-def"><a name="L948"></a><tt class="py-lineno"> 948</tt> <a class="py-toggle" href="#" id="_check_eof-toggle" onclick="return toggle('_check_eof');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO-module.html#_check_eof">_check_eof</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">index_offset</tt><tt class="py-op">,</tt> <tt class="py-param">index_length</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_check_eof-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_check_eof-expanded"><a name="L949"></a><tt class="py-lineno"> 949</tt>  <tt class="py-line">    <tt class="py-docstring">"""Check final padding is OK (8 byte alignment) and file ends (PRIVATE).</tt> </tt>
<a name="L950"></a><tt class="py-lineno"> 950</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L951"></a><tt class="py-lineno"> 951</tt>  <tt class="py-line"><tt class="py-docstring">    Will attempt to spot apparent SFF file concatenation and give an error.</tt> </tt>
<a name="L952"></a><tt class="py-lineno"> 952</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L953"></a><tt class="py-lineno"> 953</tt>  <tt class="py-line"><tt class="py-docstring">    Will not attempt to seek, only moves the handle forward.</tt> </tt>
<a name="L954"></a><tt class="py-lineno"> 954</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L955"></a><tt class="py-lineno"> 955</tt>  <tt class="py-line">    <tt class="py-name">offset</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-253" class="py-name"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-253', 'tell', 'link-54');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L956"></a><tt class="py-lineno"> 956</tt>  <tt class="py-line">    <tt class="py-name">extra</tt> <tt class="py-op">=</tt> <tt id="link-254" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-254', 'b', 'link-17');">b</a></tt><tt class="py-string">""</tt> </tt>
<a name="L957"></a><tt class="py-lineno"> 957</tt>  <tt class="py-line">    <tt class="py-name">padding</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L958"></a><tt class="py-lineno"> 958</tt>  <tt class="py-line"> </tt>
<a name="L959"></a><tt class="py-lineno"> 959</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">index_offset</tt> <tt class="py-keyword">and</tt> <tt class="py-name">offset</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">index_offset</tt><tt class="py-op">:</tt> </tt>
<a name="L960"></a><tt class="py-lineno"> 960</tt>  <tt class="py-line">        <tt class="py-comment"># Index block then end of file...</tt> </tt>
<a name="L961"></a><tt class="py-lineno"> 961</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">offset</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">index_offset</tt><tt class="py-op">:</tt> </tt>
<a name="L962"></a><tt class="py-lineno"> 962</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Gap of %i bytes after final record end %i, "</tt> </tt>
<a name="L963"></a><tt class="py-lineno"> 963</tt>  <tt class="py-line">                             <tt class="py-string">"before %i where index starts?"</tt> </tt>
<a name="L964"></a><tt class="py-lineno"> 964</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">index_offset</tt> <tt class="py-op">-</tt> <tt class="py-name">offset</tt><tt class="py-op">,</tt> <tt class="py-name">offset</tt><tt class="py-op">,</tt> <tt class="py-name">index_offset</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L965"></a><tt class="py-lineno"> 965</tt>  <tt class="py-line">        <tt class="py-comment"># Doing read to jump the index rather than a seek</tt> </tt>
<a name="L966"></a><tt class="py-lineno"> 966</tt>  <tt class="py-line">        <tt class="py-comment"># in case this is a network handle or similar</tt> </tt>
<a name="L967"></a><tt class="py-lineno"> 967</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-255" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-255', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">index_offset</tt> <tt class="py-op">+</tt> <tt class="py-name">index_length</tt> <tt class="py-op">-</tt> <tt class="py-name">offset</tt><tt class="py-op">)</tt> </tt>
<a name="L968"></a><tt class="py-lineno"> 968</tt>  <tt class="py-line">        <tt class="py-name">offset</tt> <tt class="py-op">=</tt> <tt class="py-name">index_offset</tt> <tt class="py-op">+</tt> <tt class="py-name">index_length</tt> </tt>
<a name="L969"></a><tt class="py-lineno"> 969</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">offset</tt> <tt class="py-op">==</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-256" class="py-name"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-256', 'tell', 'link-54');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> \ </tt>
<a name="L970"></a><tt class="py-lineno"> 970</tt>  <tt class="py-line">            <tt class="py-string">"Wanted %i, got %i, index is %i to %i"</tt> \ </tt>
<a name="L971"></a><tt class="py-lineno"> 971</tt>  <tt class="py-line">            <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">offset</tt><tt class="py-op">,</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-257" class="py-name"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-257', 'tell', 'link-54');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">index_offset</tt><tt class="py-op">,</tt> <tt class="py-name">index_offset</tt> <tt class="py-op">+</tt> <tt class="py-name">index_length</tt><tt class="py-op">)</tt> </tt>
<a name="L972"></a><tt class="py-lineno"> 972</tt>  <tt class="py-line"> </tt>
<a name="L973"></a><tt class="py-lineno"> 973</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">offset</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt><tt class="py-op">:</tt> </tt>
<a name="L974"></a><tt class="py-lineno"> 974</tt>  <tt class="py-line">        <tt class="py-name">padding</tt> <tt class="py-op">=</tt> <tt class="py-number">8</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">offset</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt><tt class="py-op">)</tt> </tt>
<a name="L975"></a><tt class="py-lineno"> 975</tt>  <tt class="py-line">        <tt class="py-name">extra</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-258" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-258', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">padding</tt><tt class="py-op">)</tt> </tt>
<a name="L976"></a><tt class="py-lineno"> 976</tt>  <tt class="py-line"> </tt>
<a name="L977"></a><tt class="py-lineno"> 977</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">padding</tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">4</tt> <tt class="py-keyword">and</tt> <tt class="py-name">extra</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">4</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt id="link-259" class="py-name"><a title="Bio.SeqIO.SffIO._sff" class="py-name" href="#" onclick="return doclink('link-259', '_sff', 'link-18');">_sff</a></tt><tt class="py-op">:</tt> </tt>
<a name="L978"></a><tt class="py-lineno"> 978</tt>  <tt class="py-line">        <tt class="py-comment"># Seen this in one user supplied file, should have been</tt> </tt>
<a name="L979"></a><tt class="py-lineno"> 979</tt>  <tt class="py-line">        <tt class="py-comment"># four bytes of null padding but was actually .sff and</tt> </tt>
<a name="L980"></a><tt class="py-lineno"> 980</tt>  <tt class="py-line">        <tt class="py-comment"># the start of a new concatenated SFF file!</tt> </tt>
<a name="L981"></a><tt class="py-lineno"> 981</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Your SFF file is invalid, post index %i byte "</tt> </tt>
<a name="L982"></a><tt class="py-lineno"> 982</tt>  <tt class="py-line">                         <tt class="py-string">"null padding region ended '.sff' which could "</tt> </tt>
<a name="L983"></a><tt class="py-lineno"> 983</tt>  <tt class="py-line">                         <tt class="py-string">"be the start of a concatenated SFF file? "</tt> </tt>
<a name="L984"></a><tt class="py-lineno"> 984</tt>  <tt class="py-line">                         <tt class="py-string">"See offset %i"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">padding</tt><tt class="py-op">,</tt> <tt class="py-name">offset</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L985"></a><tt class="py-lineno"> 985</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">padding</tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt class="py-name">extra</tt><tt class="py-op">:</tt> </tt>
<a name="L986"></a><tt class="py-lineno"> 986</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Is this error harmless enough to just ignore?</tt> </tt>
<a name="L987"></a><tt class="py-lineno"> 987</tt>  <tt class="py-line">        <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L988"></a><tt class="py-lineno"> 988</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt id="link-260" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-260', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-261" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-261', 'BiopythonParserWarning', 'link-48');">BiopythonParserWarning</a></tt> </tt>
<a name="L989"></a><tt class="py-lineno"> 989</tt>  <tt class="py-line">        <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Your SFF file is technically invalid as it is missing "</tt> </tt>
<a name="L990"></a><tt class="py-lineno"> 990</tt>  <tt class="py-line">                      <tt class="py-string">"a terminal %i byte null padding region."</tt> <tt class="py-op">%</tt> <tt class="py-name">padding</tt><tt class="py-op">,</tt> </tt>
<a name="L991"></a><tt class="py-lineno"> 991</tt>  <tt class="py-line">                      <tt id="link-262" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-262', 'BiopythonParserWarning', 'link-48');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L992"></a><tt class="py-lineno"> 992</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> </tt>
<a name="L993"></a><tt class="py-lineno"> 993</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">extra</tt><tt class="py-op">.</tt><tt id="link-263" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-263', 'count', 'link-45');">count</a></tt><tt class="py-op">(</tt><tt id="link-264" class="py-name"><a title="Bio.SeqIO.SffIO._null" class="py-name" href="#" onclick="return doclink('link-264', '_null', 'link-16');">_null</a></tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">padding</tt><tt class="py-op">:</tt> </tt>
<a name="L994"></a><tt class="py-lineno"> 994</tt>  <tt class="py-line">        <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L995"></a><tt class="py-lineno"> 995</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt id="link-265" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-265', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-266" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-266', 'BiopythonParserWarning', 'link-48');">BiopythonParserWarning</a></tt> </tt>
<a name="L996"></a><tt class="py-lineno"> 996</tt>  <tt class="py-line">        <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Your SFF file is invalid, post index %i byte "</tt> </tt>
<a name="L997"></a><tt class="py-lineno"> 997</tt>  <tt class="py-line">                      <tt class="py-string">"null padding region contained data: %r"</tt> </tt>
<a name="L998"></a><tt class="py-lineno"> 998</tt>  <tt class="py-line">                      <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">padding</tt><tt class="py-op">,</tt> <tt class="py-name">extra</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-267" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-267', 'BiopythonParserWarning', 'link-48');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L999"></a><tt class="py-lineno"> 999</tt>  <tt class="py-line"> </tt>
<a name="L1000"></a><tt class="py-lineno">1000</tt>  <tt class="py-line">    <tt class="py-name">offset</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-268" class="py-name"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-268', 'tell', 'link-54');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1001"></a><tt class="py-lineno">1001</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">offset</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> \ </tt>
<a name="L1002"></a><tt class="py-lineno">1002</tt>  <tt class="py-line">        <tt class="py-string">"Wanted offset %i %% 8 = %i to be zero"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">offset</tt><tt class="py-op">,</tt> <tt class="py-name">offset</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt><tt class="py-op">)</tt> </tt>
<a name="L1003"></a><tt class="py-lineno">1003</tt>  <tt class="py-line">    <tt class="py-comment"># Should now be at the end of the file...</tt> </tt>
<a name="L1004"></a><tt class="py-lineno">1004</tt>  <tt class="py-line">    <tt class="py-name">extra</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-269" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-269', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-number">4</tt><tt class="py-op">)</tt> </tt>
<a name="L1005"></a><tt class="py-lineno">1005</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">extra</tt> <tt class="py-op">==</tt> <tt id="link-270" class="py-name"><a title="Bio.SeqIO.SffIO._sff" class="py-name" href="#" onclick="return doclink('link-270', '_sff', 'link-18');">_sff</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1006"></a><tt class="py-lineno">1006</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Additional data at end of SFF file, "</tt> </tt>
<a name="L1007"></a><tt class="py-lineno">1007</tt>  <tt class="py-line">                         <tt class="py-string">"perhaps multiple SFF files concatenated? "</tt> </tt>
<a name="L1008"></a><tt class="py-lineno">1008</tt>  <tt class="py-line">                         <tt class="py-string">"See offset %i"</tt> <tt class="py-op">%</tt> <tt class="py-name">offset</tt><tt class="py-op">)</tt> </tt>
<a name="L1009"></a><tt class="py-lineno">1009</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">extra</tt><tt class="py-op">:</tt> </tt>
<a name="L1010"></a><tt class="py-lineno">1010</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Additional data at end of SFF file, "</tt> </tt>
<a name="L1011"></a><tt class="py-lineno">1011</tt>  <tt class="py-line">                         <tt class="py-string">"see offset %i"</tt> <tt class="py-op">%</tt> <tt class="py-name">offset</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1012"></a><tt class="py-lineno">1012</tt>  <tt class="py-line"> </tt>
<a name="L1013"></a><tt class="py-lineno">1013</tt>  <tt class="py-line"> </tt>
<a name="L1014"></a><tt class="py-lineno">1014</tt>  <tt class="py-line"><tt class="py-comment"># This is a generator function!</tt> </tt>
<a name="_SffTrimIterator"></a><div id="_SffTrimIterator-def"><a name="L1015"></a><tt class="py-lineno">1015</tt> <a class="py-toggle" href="#" id="_SffTrimIterator-toggle" onclick="return toggle('_SffTrimIterator');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO-module.html#_SffTrimIterator">_SffTrimIterator</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt id="link-271" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-271', 'Alphabet', 'link-5');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-272" class="py-name"><a title="Bio.Alphabet.generic_dna" class="py-name" href="#" onclick="return doclink('link-272', 'generic_dna', 'link-231');">generic_dna</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_SffTrimIterator-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_SffTrimIterator-expanded"><a name="L1016"></a><tt class="py-lineno">1016</tt>  <tt class="py-line">    <tt class="py-docstring">"""Iterate over SFF reads (as SeqRecord objects) with trimming (PRIVATE)."""</tt> </tt>
<a name="L1017"></a><tt class="py-lineno">1017</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-273" class="py-name" targets="Function Bio.SeqIO.SffIO.SffIterator()=Bio.SeqIO.SffIO-module.html#SffIterator"><a title="Bio.SeqIO.SffIO.SffIterator" class="py-name" href="#" onclick="return doclink('link-273', 'SffIterator', 'link-273');">SffIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt id="link-274" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-274', 'alphabet', 'link-186');">alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-name">trim</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1018"></a><tt class="py-lineno">1018</tt>  <tt class="py-line"> </tt>
<a name="L1019"></a><tt class="py-lineno">1019</tt>  <tt class="py-line"> </tt>
<a name="SffWriter"></a><div id="SffWriter-def"><a name="L1020"></a><tt class="py-lineno">1020</tt> <a class="py-toggle" href="#" id="SffWriter-toggle" onclick="return toggle('SffWriter');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO.SffWriter-class.html">SffWriter</a><tt class="py-op">(</tt><tt class="py-base-class">SequenceWriter</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SffWriter-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="SffWriter-expanded"><a name="L1021"></a><tt class="py-lineno">1021</tt>  <tt class="py-line">    <tt class="py-docstring">"""SFF file writer."""</tt> </tt>
<a name="L1022"></a><tt class="py-lineno">1022</tt>  <tt class="py-line"> </tt>
<a name="SffWriter.__init__"></a><div id="SffWriter.__init__-def"><a name="L1023"></a><tt class="py-lineno">1023</tt> <a class="py-toggle" href="#" id="SffWriter.__init__-toggle" onclick="return toggle('SffWriter.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO.SffWriter-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">index</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">,</tt> <tt class="py-param">xml</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SffWriter.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SffWriter.__init__-expanded"><a name="L1024"></a><tt class="py-lineno">1024</tt>  <tt class="py-line">        <tt class="py-docstring">"""Creates the writer object.</tt> </tt>
<a name="L1025"></a><tt class="py-lineno">1025</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1026"></a><tt class="py-lineno">1026</tt>  <tt class="py-line"><tt class="py-docstring">            - handle - Output handle, ideally in binary write mode.</tt> </tt>
<a name="L1027"></a><tt class="py-lineno">1027</tt>  <tt class="py-line"><tt class="py-docstring">            - index - Boolean argument, should we try and write an index?</tt> </tt>
<a name="L1028"></a><tt class="py-lineno">1028</tt>  <tt class="py-line"><tt class="py-docstring">            - xml - Optional string argument, xml manifest to be recorded in the index</tt> </tt>
<a name="L1029"></a><tt class="py-lineno">1029</tt>  <tt class="py-line"><tt class="py-docstring">              block (see function ReadRocheXmlManifest for reading this data).</tt> </tt>
<a name="L1030"></a><tt class="py-lineno">1030</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1031"></a><tt class="py-lineno">1031</tt>  <tt class="py-line">        <tt id="link-275" class="py-name"><a title="Bio.SeqIO.SffIO._check_mode" class="py-name" href="#" onclick="return doclink('link-275', '_check_mode', 'link-30');">_check_mode</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L1032"></a><tt class="py-lineno">1032</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt> </tt>
<a name="L1033"></a><tt class="py-lineno">1033</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_xml</tt> <tt class="py-op">=</tt> <tt class="py-name">xml</tt> </tt>
<a name="L1034"></a><tt class="py-lineno">1034</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-276" class="py-name" targets="Method Bio.Crystal.Chain.index()=Bio.Crystal.Chain-class.html#index,Method Bio.SCOP.Raf.SeqMap.index()=Bio.SCOP.Raf.SeqMap-class.html#index,Method Bio.SearchIO._model.hit.Hit.index()=Bio.SearchIO._model.hit.Hit-class.html#index,Method Bio.SearchIO._model.query.QueryResult.index()=Bio.SearchIO._model.query.QueryResult-class.html#index,Function Bio.SearchIO.index()=Bio.SearchIO-module.html#index,Method Bio.Seq.MutableSeq.index()=Bio.Seq.MutableSeq-class.html#index,Function Bio.SeqIO.index()=Bio.SeqIO-module.html#index"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-276', 'index', 'link-276');">index</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1035"></a><tt class="py-lineno">1035</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-277" class="py-name" targets="Module Bio.SearchIO._index=Bio.SearchIO._index-module.html,Module Bio.SeqIO._index=Bio.SeqIO._index-module.html"><a title="Bio.SearchIO._index
Bio.SeqIO._index" class="py-name" href="#" onclick="return doclink('link-277', '_index', 'link-277');">_index</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1036"></a><tt class="py-lineno">1036</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1037"></a><tt class="py-lineno">1037</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-278" class="py-name"><a title="Bio.SearchIO._index
Bio.SeqIO._index" class="py-name" href="#" onclick="return doclink('link-278', '_index', 'link-277');">_index</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L1038"></a><tt class="py-lineno">1038</tt>  <tt class="py-line"> </tt>
<a name="SffWriter.write_file"></a><div id="SffWriter.write_file-def"><a name="L1039"></a><tt class="py-lineno">1039</tt> <a class="py-toggle" href="#" id="SffWriter.write_file-toggle" onclick="return toggle('SffWriter.write_file');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO.SffWriter-class.html#write_file">write_file</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">records</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SffWriter.write_file-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SffWriter.write_file-expanded"><a name="L1040"></a><tt class="py-lineno">1040</tt>  <tt class="py-line">        <tt class="py-docstring">"""Use this to write an entire file containing the given records."""</tt> </tt>
<a name="L1041"></a><tt class="py-lineno">1041</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1042"></a><tt class="py-lineno">1042</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_number_of_reads</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">records</tt><tt class="py-op">)</tt> </tt>
<a name="L1043"></a><tt class="py-lineno">1043</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">TypeError</tt><tt class="py-op">:</tt> </tt>
<a name="L1044"></a><tt class="py-lineno">1044</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_number_of_reads</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt>  <tt class="py-comment"># dummy value</tt> </tt>
<a name="L1045"></a><tt class="py-lineno">1045</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-string">"seek"</tt><tt class="py-op">)</tt> \ </tt>
<a name="L1046"></a><tt class="py-lineno">1046</tt>  <tt class="py-line">                    <tt class="py-keyword">or</tt> <tt class="py-keyword">not</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-string">"tell"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1047"></a><tt class="py-lineno">1047</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"A handle with a seek/tell methods is "</tt> </tt>
<a name="L1048"></a><tt class="py-lineno">1048</tt>  <tt class="py-line">                                 <tt class="py-string">"required in order to record the total "</tt> </tt>
<a name="L1049"></a><tt class="py-lineno">1049</tt>  <tt class="py-line">                                 <tt class="py-string">"record count in the file header (once it "</tt> </tt>
<a name="L1050"></a><tt class="py-lineno">1050</tt>  <tt class="py-line">                                 <tt class="py-string">"is known at the end)."</tt><tt class="py-op">)</tt> </tt>
<a name="L1051"></a><tt class="py-lineno">1051</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-279" class="py-name"><a title="Bio.SearchIO._index
Bio.SeqIO._index" class="py-name" href="#" onclick="return doclink('link-279', '_index', 'link-277');">_index</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L1052"></a><tt class="py-lineno">1052</tt>  <tt class="py-line">                <tt class="py-keyword">not</tt> <tt class="py-op">(</tt><tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-string">"seek"</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-string">"tell"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1053"></a><tt class="py-lineno">1053</tt>  <tt class="py-line">            <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L1054"></a><tt class="py-lineno">1054</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"A handle with a seek/tell methods is required in "</tt> </tt>
<a name="L1055"></a><tt class="py-lineno">1055</tt>  <tt class="py-line">                          <tt class="py-string">"order to record an SFF index."</tt><tt class="py-op">)</tt> </tt>
<a name="L1056"></a><tt class="py-lineno">1056</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-280" class="py-name"><a title="Bio.SearchIO._index
Bio.SeqIO._index" class="py-name" href="#" onclick="return doclink('link-280', '_index', 'link-277');">_index</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L1057"></a><tt class="py-lineno">1057</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_index_start</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L1058"></a><tt class="py-lineno">1058</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_index_length</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L1059"></a><tt class="py-lineno">1059</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">records</tt><tt class="py-op">,</tt> <tt class="py-string">"next"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1060"></a><tt class="py-lineno">1060</tt>  <tt class="py-line">            <tt class="py-name">records</tt> <tt class="py-op">=</tt> <tt class="py-name">iter</tt><tt class="py-op">(</tt><tt class="py-name">records</tt><tt class="py-op">)</tt> </tt>
<a name="L1061"></a><tt class="py-lineno">1061</tt>  <tt class="py-line">        <tt class="py-comment"># Get the first record in order to find the flow information</tt> </tt>
<a name="L1062"></a><tt class="py-lineno">1062</tt>  <tt class="py-line">        <tt class="py-comment"># we will need for the header.</tt> </tt>
<a name="L1063"></a><tt class="py-lineno">1063</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1064"></a><tt class="py-lineno">1064</tt>  <tt class="py-line">            <tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt id="link-281" class="py-name" targets="Method Bio.AlignIO.Interfaces.AlignmentIterator.next()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#next,Method Bio.Blast.NCBIStandalone.Iterator.next()=Bio.Blast.NCBIStandalone.Iterator-class.html#next,Method Bio.Blast.ParseBlastTable.BlastTableReader.next()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#next,Method Bio.DocSQL.IterationCursor.next()=Bio.DocSQL.IterationCursor-class.html#next,Method Bio.File.UndoHandle.next()=Bio.File.UndoHandle-class.html#next,Method Bio.GenBank.Iterator.next()=Bio.GenBank.Iterator-class.html#next,Method Bio.Nexus.Nexus.CharBuffer.next()=Bio.Nexus.Nexus.CharBuffer-class.html#next,Method Bio.PopGen.GenePop.Controller._FileIterator.next()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#next,Method Bio.SeqIO.Interfaces.SequenceIterator.next()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#next,Method Bio.bgzf.BgzfReader.next()=Bio.bgzf.BgzfReader-class.html#next"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-281', 'next', 'link-281');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">records</tt><tt class="py-op">)</tt> </tt>
<a name="L1065"></a><tt class="py-lineno">1065</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L1066"></a><tt class="py-lineno">1066</tt>  <tt class="py-line">            <tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L1067"></a><tt class="py-lineno">1067</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">record</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1068"></a><tt class="py-lineno">1068</tt>  <tt class="py-line">            <tt class="py-comment"># No records -&gt; empty SFF file (or an error)?</tt> </tt>
<a name="L1069"></a><tt class="py-lineno">1069</tt>  <tt class="py-line">            <tt class="py-comment"># We can't write a header without the flow information.</tt> </tt>
<a name="L1070"></a><tt class="py-lineno">1070</tt>  <tt class="py-line">            <tt class="py-comment"># return 0</tt> </tt>
<a name="L1071"></a><tt class="py-lineno">1071</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Must have at least one sequence"</tt><tt class="py-op">)</tt> </tt>
<a name="L1072"></a><tt class="py-lineno">1072</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1073"></a><tt class="py-lineno">1073</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_key_sequence</tt> <tt class="py-op">=</tt> <tt id="link-282" class="py-name"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-282', '_as_bytes', 'link-15');">_as_bytes</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-283" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-283', 'annotations', 'link-156');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"flow_key"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1074"></a><tt class="py-lineno">1074</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_flow_chars</tt> <tt class="py-op">=</tt> <tt id="link-284" class="py-name"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-284', '_as_bytes', 'link-15');">_as_bytes</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-285" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-285', 'annotations', 'link-156');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"flow_chars"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1075"></a><tt class="py-lineno">1075</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_number_of_flows_per_read</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_flow_chars</tt><tt class="py-op">)</tt> </tt>
<a name="L1076"></a><tt class="py-lineno">1076</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L1077"></a><tt class="py-lineno">1077</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Missing SFF flow information"</tt><tt class="py-op">)</tt> </tt>
<a name="L1078"></a><tt class="py-lineno">1078</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-286" class="py-name" targets="Method Bio.AlignIO.EmbossIO.EmbossWriter.write_header()=Bio.AlignIO.EmbossIO.EmbossWriter-class.html#write_header,Method Bio.AlignIO.Interfaces.SequentialAlignmentWriter.write_header()=Bio.AlignIO.Interfaces.SequentialAlignmentWriter-class.html#write_header,Method Bio.NMR.xpktools.Peaklist.write_header()=Bio.NMR.xpktools.Peaklist-class.html#write_header,Method Bio.SeqIO.Interfaces.SequentialSequenceWriter.write_header()=Bio.SeqIO.Interfaces.SequentialSequenceWriter-class.html#write_header,Method Bio.SeqIO.SeqXmlIO.SeqXmlWriter.write_header()=Bio.SeqIO.SeqXmlIO.SeqXmlWriter-class.html#write_header,Method Bio.SeqIO.SffIO.SffWriter.write_header()=Bio.SeqIO.SffIO.SffWriter-class.html#write_header"><a title="Bio.AlignIO.EmbossIO.EmbossWriter.write_header
Bio.AlignIO.Interfaces.SequentialAlignmentWriter.write_header
Bio.NMR.xpktools.Peaklist.write_header
Bio.SeqIO.Interfaces.SequentialSequenceWriter.write_header
Bio.SeqIO.SeqXmlIO.SeqXmlWriter.write_header
Bio.SeqIO.SffIO.SffWriter.write_header" class="py-name" href="#" onclick="return doclink('link-286', 'write_header', 'link-286');">write_header</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1079"></a><tt class="py-lineno">1079</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-287" class="py-name" targets="Method Bio.SeqIO.FastaIO.FastaWriter.write_record()=Bio.SeqIO.FastaIO.FastaWriter-class.html#write_record,Method Bio.SeqIO.InsdcIO.EmblWriter.write_record()=Bio.SeqIO.InsdcIO.EmblWriter-class.html#write_record,Method Bio.SeqIO.InsdcIO.GenBankWriter.write_record()=Bio.SeqIO.InsdcIO.GenBankWriter-class.html#write_record,Method Bio.SeqIO.Interfaces.SequentialSequenceWriter.write_record()=Bio.SeqIO.Interfaces.SequentialSequenceWriter-class.html#write_record,Method Bio.SeqIO.PhdIO.PhdWriter.write_record()=Bio.SeqIO.PhdIO.PhdWriter-class.html#write_record,Method Bio.SeqIO.QualityIO.FastqIlluminaWriter.write_record()=Bio.SeqIO.QualityIO.FastqIlluminaWriter-class.html#write_record,Method Bio.SeqIO.QualityIO.FastqPhredWriter.write_record()=Bio.SeqIO.QualityIO.FastqPhredWriter-class.html#write_record,Method Bio.SeqIO.QualityIO.FastqSolexaWriter.write_record()=Bio.SeqIO.QualityIO.FastqSolexaWriter-class.html#write_record,Method Bio.SeqIO.QualityIO.QualPhredWriter.write_record()=Bio.SeqIO.QualityIO.QualPhredWriter-class.html#write_record,Method Bio.SeqIO.SeqXmlIO.SeqXmlWriter.write_record()=Bio.SeqIO.SeqXmlIO.SeqXmlWriter-class.html#write_record,Method Bio.SeqIO.SffIO.SffWriter.write_record()=Bio.SeqIO.SffIO.SffWriter-class.html#write_record,Method Bio.SeqIO.TabIO.TabWriter.write_record()=Bio.SeqIO.TabIO.TabWriter-class.html#write_record"><a title="Bio.SeqIO.FastaIO.FastaWriter.write_record
Bio.SeqIO.InsdcIO.EmblWriter.write_record
Bio.SeqIO.InsdcIO.GenBankWriter.write_record
Bio.SeqIO.Interfaces.SequentialSequenceWriter.write_record
Bio.SeqIO.PhdIO.PhdWriter.write_record
Bio.SeqIO.QualityIO.FastqIlluminaWriter.write_record
Bio.SeqIO.QualityIO.FastqPhredWriter.write_record
Bio.SeqIO.QualityIO.FastqSolexaWriter.write_record
Bio.SeqIO.QualityIO.QualPhredWriter.write_record
Bio.SeqIO.SeqXmlIO.SeqXmlWriter.write_record
Bio.SeqIO.SffIO.SffWriter.write_record
Bio.SeqIO.TabIO.TabWriter.write_record" class="py-name" href="#" onclick="return doclink('link-287', 'write_record', 'link-287');">write_record</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L1080"></a><tt class="py-lineno">1080</tt>  <tt class="py-line">        <tt id="link-288" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-288', 'count', 'link-45');">count</a></tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L1081"></a><tt class="py-lineno">1081</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">records</tt><tt class="py-op">:</tt> </tt>
<a name="L1082"></a><tt class="py-lineno">1082</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-289" class="py-name"><a title="Bio.SeqIO.FastaIO.FastaWriter.write_record
Bio.SeqIO.InsdcIO.EmblWriter.write_record
Bio.SeqIO.InsdcIO.GenBankWriter.write_record
Bio.SeqIO.Interfaces.SequentialSequenceWriter.write_record
Bio.SeqIO.PhdIO.PhdWriter.write_record
Bio.SeqIO.QualityIO.FastqIlluminaWriter.write_record
Bio.SeqIO.QualityIO.FastqPhredWriter.write_record
Bio.SeqIO.QualityIO.FastqSolexaWriter.write_record
Bio.SeqIO.QualityIO.QualPhredWriter.write_record
Bio.SeqIO.SeqXmlIO.SeqXmlWriter.write_record
Bio.SeqIO.SffIO.SffWriter.write_record
Bio.SeqIO.TabIO.TabWriter.write_record" class="py-name" href="#" onclick="return doclink('link-289', 'write_record', 'link-287');">write_record</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L1083"></a><tt class="py-lineno">1083</tt>  <tt class="py-line">            <tt id="link-290" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-290', 'count', 'link-45');">count</a></tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L1084"></a><tt class="py-lineno">1084</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_number_of_reads</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1085"></a><tt class="py-lineno">1085</tt>  <tt class="py-line">            <tt class="py-comment"># Must go back and record the record count...</tt> </tt>
<a name="L1086"></a><tt class="py-lineno">1086</tt>  <tt class="py-line">            <tt class="py-name">offset</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-291" class="py-name"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-291', 'tell', 'link-54');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1087"></a><tt class="py-lineno">1087</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-292" class="py-name"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-292', 'seek', 'link-50');">seek</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L1088"></a><tt class="py-lineno">1088</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_number_of_reads</tt> <tt class="py-op">=</tt> <tt id="link-293" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-293', 'count', 'link-45');">count</a></tt> </tt>
<a name="L1089"></a><tt class="py-lineno">1089</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-294" class="py-name"><a title="Bio.AlignIO.EmbossIO.EmbossWriter.write_header
Bio.AlignIO.Interfaces.SequentialAlignmentWriter.write_header
Bio.NMR.xpktools.Peaklist.write_header
Bio.SeqIO.Interfaces.SequentialSequenceWriter.write_header
Bio.SeqIO.SeqXmlIO.SeqXmlWriter.write_header
Bio.SeqIO.SffIO.SffWriter.write_header" class="py-name" href="#" onclick="return doclink('link-294', 'write_header', 'link-286');">write_header</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1090"></a><tt class="py-lineno">1090</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-295" class="py-name"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-295', 'seek', 'link-50');">seek</a></tt><tt class="py-op">(</tt><tt class="py-name">offset</tt><tt class="py-op">)</tt>  <tt class="py-comment"># not essential?</tt> </tt>
<a name="L1091"></a><tt class="py-lineno">1091</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1092"></a><tt class="py-lineno">1092</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt id="link-296" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-296', 'count', 'link-45');">count</a></tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_number_of_reads</tt> </tt>
<a name="L1093"></a><tt class="py-lineno">1093</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-297" class="py-name"><a title="Bio.SearchIO._index
Bio.SeqIO._index" class="py-name" href="#" onclick="return doclink('link-297', '_index', 'link-277');">_index</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1094"></a><tt class="py-lineno">1094</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-298" class="py-name" targets="Method Bio.SeqIO.SffIO.SffWriter._write_index()=Bio.SeqIO.SffIO.SffWriter-class.html#_write_index"><a title="Bio.SeqIO.SffIO.SffWriter._write_index" class="py-name" href="#" onclick="return doclink('link-298', '_write_index', 'link-298');">_write_index</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1095"></a><tt class="py-lineno">1095</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-299" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-299', 'count', 'link-45');">count</a></tt> </tt>
</div><a name="L1096"></a><tt class="py-lineno">1096</tt>  <tt class="py-line"> </tt>
<a name="SffWriter._write_index"></a><div id="SffWriter._write_index-def"><a name="L1097"></a><tt class="py-lineno">1097</tt> <a class="py-toggle" href="#" id="SffWriter._write_index-toggle" onclick="return toggle('SffWriter._write_index');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO.SffWriter-class.html#_write_index">_write_index</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SffWriter._write_index-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SffWriter._write_index-expanded"><a name="L1098"></a><tt class="py-lineno">1098</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-300" class="py-name"><a title="Bio.SearchIO._index
Bio.SeqIO._index" class="py-name" href="#" onclick="return doclink('link-300', '_index', 'link-277');">_index</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_number_of_reads</tt> </tt>
<a name="L1099"></a><tt class="py-lineno">1099</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt> </tt>
<a name="L1100"></a><tt class="py-lineno">1100</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-301" class="py-name"><a title="Bio.SearchIO._index
Bio.SeqIO._index" class="py-name" href="#" onclick="return doclink('link-301', '_index', 'link-277');">_index</a></tt><tt class="py-op">.</tt><tt id="link-302" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.sort()=Bio.Align.MultipleSeqAlignment-class.html#sort,Method Bio.PDB.Residue.DisorderedResidue.sort()=Bio.PDB.Residue.DisorderedResidue-class.html#sort,Method Bio.PDB.Residue.Residue.sort()=Bio.PDB.Residue.Residue-class.html#sort,Method Bio.SearchIO._model.hit.Hit.sort()=Bio.SearchIO._model.hit.Hit-class.html#sort,Method Bio.SearchIO._model.query.QueryResult.sort()=Bio.SearchIO._model.query.QueryResult-class.html#sort,Method Bio.Sequencing.Ace.ACEFileRecord.sort()=Bio.Sequencing.Ace.ACEFileRecord-class.html#sort"><a title="Bio.Align.MultipleSeqAlignment.sort
Bio.PDB.Residue.DisorderedResidue.sort
Bio.PDB.Residue.Residue.sort
Bio.SearchIO._model.hit.Hit.sort
Bio.SearchIO._model.query.QueryResult.sort
Bio.Sequencing.Ace.ACEFileRecord.sort" class="py-name" href="#" onclick="return doclink('link-302', 'sort', 'link-302');">sort</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1101"></a><tt class="py-lineno">1101</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_index_start</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-303" class="py-name"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-303', 'tell', 'link-54');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt>  <tt class="py-comment"># need for header</tt> </tt>
<a name="L1102"></a><tt class="py-lineno">1102</tt>  <tt class="py-line">        <tt class="py-comment"># XML...</tt> </tt>
<a name="L1103"></a><tt class="py-lineno">1103</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_xml</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1104"></a><tt class="py-lineno">1104</tt>  <tt class="py-line">            <tt class="py-name">xml</tt> <tt class="py-op">=</tt> <tt id="link-304" class="py-name"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-304', '_as_bytes', 'link-15');">_as_bytes</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_xml</tt><tt class="py-op">)</tt> </tt>
<a name="L1105"></a><tt class="py-lineno">1105</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1106"></a><tt class="py-lineno">1106</tt>  <tt class="py-line">            <tt class="py-keyword">from</tt> <tt id="link-305" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-305', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">__version__</tt> </tt>
<a name="L1107"></a><tt class="py-lineno">1107</tt>  <tt class="py-line">            <tt class="py-name">xml</tt> <tt class="py-op">=</tt> <tt class="py-string">"&lt;!-- This file was output with Biopython %s --&gt;\n"</tt> <tt class="py-op">%</tt> <tt class="py-name">__version__</tt> </tt>
<a name="L1108"></a><tt class="py-lineno">1108</tt>  <tt class="py-line">            <tt class="py-name">xml</tt> <tt class="py-op">+=</tt> <tt class="py-string">"&lt;!-- This XML and index block attempts to mimic Roche SFF files --&gt;\n"</tt> </tt>
<a name="L1109"></a><tt class="py-lineno">1109</tt>  <tt class="py-line">            <tt class="py-name">xml</tt> <tt class="py-op">+=</tt> <tt class="py-string">"&lt;!-- This file may be a combination of multiple SFF files etc --&gt;\n"</tt> </tt>
<a name="L1110"></a><tt class="py-lineno">1110</tt>  <tt class="py-line">            <tt class="py-name">xml</tt> <tt class="py-op">=</tt> <tt id="link-306" class="py-name"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-306', '_as_bytes', 'link-15');">_as_bytes</a></tt><tt class="py-op">(</tt><tt class="py-name">xml</tt><tt class="py-op">)</tt> </tt>
<a name="L1111"></a><tt class="py-lineno">1111</tt>  <tt class="py-line">        <tt class="py-name">xml_len</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">xml</tt><tt class="py-op">)</tt> </tt>
<a name="L1112"></a><tt class="py-lineno">1112</tt>  <tt class="py-line">        <tt class="py-comment"># Write to the file...</tt> </tt>
<a name="L1113"></a><tt class="py-lineno">1113</tt>  <tt class="py-line">        <tt class="py-name">fmt</tt> <tt class="py-op">=</tt> <tt class="py-string">"&gt;I4BLL"</tt> </tt>
<a name="L1114"></a><tt class="py-lineno">1114</tt>  <tt class="py-line">        <tt class="py-name">fmt_size</tt> <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-name">fmt</tt><tt class="py-op">)</tt> </tt>
<a name="L1115"></a><tt class="py-lineno">1115</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-307" class="py-name" targets="Function Bio.AlignIO.write()=Bio.AlignIO-module.html#write,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.write()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#write,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.write()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#write,Method Bio.Phylo.CDAOIO.Writer.write()=Bio.Phylo.CDAOIO.Writer-class.html#write,Function Bio.Phylo.CDAOIO.write()=Bio.Phylo.CDAOIO-module.html#write,Method Bio.Phylo.NeXMLIO.Writer.write()=Bio.Phylo.NeXMLIO.Writer-class.html#write,Function Bio.Phylo.NeXMLIO.write()=Bio.Phylo.NeXMLIO-module.html#write,Method Bio.Phylo.NewickIO.Writer.write()=Bio.Phylo.NewickIO.Writer-class.html#write,Function Bio.Phylo.NewickIO.write()=Bio.Phylo.NewickIO-module.html#write,Function Bio.Phylo.NexusIO.write()=Bio.Phylo.NexusIO-module.html#write,Method Bio.Phylo.PhyloXMLIO.Writer.write()=Bio.Phylo.PhyloXMLIO.Writer-class.html#write,Function Bio.Phylo.PhyloXMLIO.write()=Bio.Phylo.PhyloXMLIO-module.html#write,Function Bio.Phylo._io.write()=Bio.Phylo._io-module.html#write,Function Bio.SearchIO.write()=Bio.SearchIO-module.html#write,Function Bio.SeqIO.write()=Bio.SeqIO-module.html#write,Method Bio.bgzf.BgzfWriter.write()=Bio.bgzf.BgzfWriter-class.html#write,Function Bio.motifs.jaspar.write()=Bio.motifs.jaspar-module.html#write,Function Bio.motifs.transfac.write()=Bio.motifs.transfac-module.html#write,Function Bio.motifs.write()=Bio.motifs-module.html#write"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-307', 'write', 'link-307');">write</a></tt><tt class="py-op">(</tt><tt id="link-308" class="py-name"><a title="Bio.SeqIO.SffIO._null" class="py-name" href="#" onclick="return doclink('link-308', '_null', 'link-16');">_null</a></tt> <tt class="py-op">*</tt> <tt class="py-name">fmt_size</tt> <tt class="py-op">+</tt> <tt class="py-name">xml</tt><tt class="py-op">)</tt>  <tt class="py-comment"># fill this later</tt> </tt>
<a name="L1116"></a><tt class="py-lineno">1116</tt>  <tt class="py-line">        <tt class="py-name">fmt2</tt> <tt class="py-op">=</tt> <tt class="py-string">"&gt;6B"</tt> </tt>
<a name="L1117"></a><tt class="py-lineno">1117</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-number">6</tt> <tt class="py-op">==</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-name">fmt2</tt><tt class="py-op">)</tt> </tt>
<a name="L1118"></a><tt class="py-lineno">1118</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-309" class="py-name"><a title="Bio.SearchIO._index
Bio.SeqIO._index" class="py-name" href="#" onclick="return doclink('link-309', '_index', 'link-277');">_index</a></tt><tt class="py-op">.</tt><tt id="link-310" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.sort
Bio.PDB.Residue.DisorderedResidue.sort
Bio.PDB.Residue.Residue.sort
Bio.SearchIO._model.hit.Hit.sort
Bio.SearchIO._model.query.QueryResult.sort
Bio.Sequencing.Ace.ACEFileRecord.sort" class="py-name" href="#" onclick="return doclink('link-310', 'sort', 'link-302');">sort</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1119"></a><tt class="py-lineno">1119</tt>  <tt class="py-line">        <tt class="py-name">index_len</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt>  <tt class="py-comment"># don't know yet!</tt> </tt>
<a name="L1120"></a><tt class="py-lineno">1120</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-311" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-311', 'name', 'link-60');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">offset</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-312" class="py-name"><a title="Bio.SearchIO._index
Bio.SeqIO._index" class="py-name" href="#" onclick="return doclink('link-312', '_index', 'link-277');">_index</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1121"></a><tt class="py-lineno">1121</tt>  <tt class="py-line">            <tt class="py-comment"># Roche files record the offsets using base 255 not 256.</tt> </tt>
<a name="L1122"></a><tt class="py-lineno">1122</tt>  <tt class="py-line">            <tt class="py-comment"># See comments for parsing the index block. There may be a faster</tt> </tt>
<a name="L1123"></a><tt class="py-lineno">1123</tt>  <tt class="py-line">            <tt class="py-comment"># way to code this, but we can't easily use shifts due to odd base</tt> </tt>
<a name="L1124"></a><tt class="py-lineno">1124</tt>  <tt class="py-line">            <tt class="py-name">off3</tt> <tt class="py-op">=</tt> <tt class="py-name">offset</tt> </tt>
<a name="L1125"></a><tt class="py-lineno">1125</tt>  <tt class="py-line">            <tt class="py-name">off0</tt> <tt class="py-op">=</tt> <tt class="py-name">off3</tt> <tt class="py-op">%</tt> <tt class="py-number">255</tt> </tt>
<a name="L1126"></a><tt class="py-lineno">1126</tt>  <tt class="py-line">            <tt class="py-name">off3</tt> <tt class="py-op">-=</tt> <tt class="py-name">off0</tt> </tt>
<a name="L1127"></a><tt class="py-lineno">1127</tt>  <tt class="py-line">            <tt class="py-name">off1</tt> <tt class="py-op">=</tt> <tt class="py-name">off3</tt> <tt class="py-op">%</tt> <tt class="py-number">65025</tt> </tt>
<a name="L1128"></a><tt class="py-lineno">1128</tt>  <tt class="py-line">            <tt class="py-name">off3</tt> <tt class="py-op">-=</tt> <tt class="py-name">off1</tt> </tt>
<a name="L1129"></a><tt class="py-lineno">1129</tt>  <tt class="py-line">            <tt class="py-name">off2</tt> <tt class="py-op">=</tt> <tt class="py-name">off3</tt> <tt class="py-op">%</tt> <tt class="py-number">16581375</tt> </tt>
<a name="L1130"></a><tt class="py-lineno">1130</tt>  <tt class="py-line">            <tt class="py-name">off3</tt> <tt class="py-op">-=</tt> <tt class="py-name">off2</tt> </tt>
<a name="L1131"></a><tt class="py-lineno">1131</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">offset</tt> <tt class="py-op">==</tt> <tt class="py-name">off0</tt> <tt class="py-op">+</tt> <tt class="py-name">off1</tt> <tt class="py-op">+</tt> <tt class="py-name">off2</tt> <tt class="py-op">+</tt> <tt class="py-name">off3</tt><tt class="py-op">,</tt> \ </tt>
<a name="L1132"></a><tt class="py-lineno">1132</tt>  <tt class="py-line">                <tt class="py-string">"%i -&gt; %i %i %i %i"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">offset</tt><tt class="py-op">,</tt> <tt class="py-name">off0</tt><tt class="py-op">,</tt> <tt class="py-name">off1</tt><tt class="py-op">,</tt> <tt class="py-name">off2</tt><tt class="py-op">,</tt> <tt class="py-name">off3</tt><tt class="py-op">)</tt> </tt>
<a name="L1133"></a><tt class="py-lineno">1133</tt>  <tt class="py-line">            <tt class="py-name">off3</tt><tt class="py-op">,</tt> <tt class="py-name">off2</tt><tt class="py-op">,</tt> <tt class="py-name">off1</tt><tt class="py-op">,</tt> <tt class="py-name">off0</tt> <tt class="py-op">=</tt> <tt class="py-name">off3</tt> <tt class="py-op">//</tt> <tt class="py-number">16581375</tt><tt class="py-op">,</tt> <tt class="py-name">off2</tt> <tt class="py-op">//</tt> <tt class="py-number">65025</tt><tt class="py-op">,</tt> \ </tt>
<a name="L1134"></a><tt class="py-lineno">1134</tt>  <tt class="py-line">                <tt class="py-name">off1</tt> <tt class="py-op">//</tt> <tt class="py-number">255</tt><tt class="py-op">,</tt> <tt class="py-name">off0</tt> </tt>
<a name="L1135"></a><tt class="py-lineno">1135</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">off0</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">255</tt> <tt class="py-keyword">and</tt> <tt class="py-name">off1</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">255</tt> <tt class="py-keyword">and</tt> <tt class="py-name">off2</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">255</tt> <tt class="py-keyword">and</tt> <tt class="py-name">off3</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">255</tt><tt class="py-op">,</tt> \ </tt>
<a name="L1136"></a><tt class="py-lineno">1136</tt>  <tt class="py-line">                <tt class="py-string">"%i -&gt; %i %i %i %i"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">offset</tt><tt class="py-op">,</tt> <tt class="py-name">off0</tt><tt class="py-op">,</tt> <tt class="py-name">off1</tt><tt class="py-op">,</tt> <tt class="py-name">off2</tt><tt class="py-op">,</tt> <tt class="py-name">off3</tt><tt class="py-op">)</tt> </tt>
<a name="L1137"></a><tt class="py-lineno">1137</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-313" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-313', 'write', 'link-307');">write</a></tt><tt class="py-op">(</tt><tt id="link-314" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-314', 'name', 'link-60');">name</a></tt> <tt class="py-op">+</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">pack</tt><tt class="py-op">(</tt><tt class="py-name">fmt2</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> </tt>
<a name="L1138"></a><tt class="py-lineno">1138</tt>  <tt class="py-line">                                            <tt class="py-name">off3</tt><tt class="py-op">,</tt> <tt class="py-name">off2</tt><tt class="py-op">,</tt> <tt class="py-name">off1</tt><tt class="py-op">,</tt> <tt class="py-name">off0</tt><tt class="py-op">,</tt> <tt class="py-number">255</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1139"></a><tt class="py-lineno">1139</tt>  <tt class="py-line">            <tt class="py-name">index_len</tt> <tt class="py-op">+=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-315" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-315', 'name', 'link-60');">name</a></tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-number">6</tt> </tt>
<a name="L1140"></a><tt class="py-lineno">1140</tt>  <tt class="py-line">        <tt class="py-comment"># Note any padding in not included:</tt> </tt>
<a name="L1141"></a><tt class="py-lineno">1141</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_index_length</tt> <tt class="py-op">=</tt> <tt class="py-name">fmt_size</tt> <tt class="py-op">+</tt> <tt class="py-name">xml_len</tt> <tt class="py-op">+</tt> <tt class="py-name">index_len</tt>  <tt class="py-comment"># need for header</tt> </tt>
<a name="L1142"></a><tt class="py-lineno">1142</tt>  <tt class="py-line">        <tt class="py-comment"># Pad out to an 8 byte boundary (although I have noticed some</tt> </tt>
<a name="L1143"></a><tt class="py-lineno">1143</tt>  <tt class="py-line">        <tt class="py-comment"># real Roche SFF files neglect to do this depsite their manual</tt> </tt>
<a name="L1144"></a><tt class="py-lineno">1144</tt>  <tt class="py-line">        <tt class="py-comment"># suggesting this padding should be there):</tt> </tt>
<a name="L1145"></a><tt class="py-lineno">1145</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_index_length</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt><tt class="py-op">:</tt> </tt>
<a name="L1146"></a><tt class="py-lineno">1146</tt>  <tt class="py-line">            <tt class="py-name">padding</tt> <tt class="py-op">=</tt> <tt class="py-number">8</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_index_length</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt><tt class="py-op">)</tt> </tt>
<a name="L1147"></a><tt class="py-lineno">1147</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-316" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-316', 'write', 'link-307');">write</a></tt><tt class="py-op">(</tt><tt id="link-317" class="py-name"><a title="Bio.SeqIO.SffIO._null" class="py-name" href="#" onclick="return doclink('link-317', '_null', 'link-16');">_null</a></tt> <tt class="py-op">*</tt> <tt class="py-name">padding</tt><tt class="py-op">)</tt> </tt>
<a name="L1148"></a><tt class="py-lineno">1148</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1149"></a><tt class="py-lineno">1149</tt>  <tt class="py-line">            <tt class="py-name">padding</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L1150"></a><tt class="py-lineno">1150</tt>  <tt class="py-line">        <tt class="py-name">offset</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-318" class="py-name"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-318', 'tell', 'link-54');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1151"></a><tt class="py-lineno">1151</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">offset</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_index_start</tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_index_length</tt> <tt class="py-op">+</tt> <tt class="py-name">padding</tt><tt class="py-op">,</tt> \ </tt>
<a name="L1152"></a><tt class="py-lineno">1152</tt>  <tt class="py-line">            <tt class="py-string">"%i vs %i + %i + %i"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">offset</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_index_start</tt><tt class="py-op">,</tt> </tt>
<a name="L1153"></a><tt class="py-lineno">1153</tt>  <tt class="py-line">                                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_index_length</tt><tt class="py-op">,</tt> <tt class="py-name">padding</tt><tt class="py-op">)</tt> </tt>
<a name="L1154"></a><tt class="py-lineno">1154</tt>  <tt class="py-line">        <tt class="py-comment"># Must now go back and update the index header with index size...</tt> </tt>
<a name="L1155"></a><tt class="py-lineno">1155</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-319" class="py-name"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-319', 'seek', 'link-50');">seek</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_index_start</tt><tt class="py-op">)</tt> </tt>
<a name="L1156"></a><tt class="py-lineno">1156</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-320" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-320', 'write', 'link-307');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">pack</tt><tt class="py-op">(</tt><tt class="py-name">fmt</tt><tt class="py-op">,</tt> <tt class="py-number">778921588</tt><tt class="py-op">,</tt>  <tt class="py-comment"># magic number</tt> </tt>
<a name="L1157"></a><tt class="py-lineno">1157</tt>  <tt class="py-line">                                 <tt class="py-number">49</tt><tt class="py-op">,</tt> <tt class="py-number">46</tt><tt class="py-op">,</tt> <tt class="py-number">48</tt><tt class="py-op">,</tt> <tt class="py-number">48</tt><tt class="py-op">,</tt>  <tt class="py-comment"># Roche index version, "1.00"</tt> </tt>
<a name="L1158"></a><tt class="py-lineno">1158</tt>  <tt class="py-line">                                 <tt class="py-name">xml_len</tt><tt class="py-op">,</tt> <tt class="py-name">index_len</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-name">xml</tt><tt class="py-op">)</tt> </tt>
<a name="L1159"></a><tt class="py-lineno">1159</tt>  <tt class="py-line">        <tt class="py-comment"># Must now go back and update the header...</tt> </tt>
<a name="L1160"></a><tt class="py-lineno">1160</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-321" class="py-name"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-321', 'seek', 'link-50');">seek</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L1161"></a><tt class="py-lineno">1161</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-322" class="py-name"><a title="Bio.AlignIO.EmbossIO.EmbossWriter.write_header
Bio.AlignIO.Interfaces.SequentialAlignmentWriter.write_header
Bio.NMR.xpktools.Peaklist.write_header
Bio.SeqIO.Interfaces.SequentialSequenceWriter.write_header
Bio.SeqIO.SeqXmlIO.SeqXmlWriter.write_header
Bio.SeqIO.SffIO.SffWriter.write_header" class="py-name" href="#" onclick="return doclink('link-322', 'write_header', 'link-286');">write_header</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1162"></a><tt class="py-lineno">1162</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-323" class="py-name"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-323', 'seek', 'link-50');">seek</a></tt><tt class="py-op">(</tt><tt class="py-name">offset</tt><tt class="py-op">)</tt>  <tt class="py-comment"># not essential?</tt> </tt>
</div><a name="L1163"></a><tt class="py-lineno">1163</tt>  <tt class="py-line"> </tt>
<a name="SffWriter.write_header"></a><div id="SffWriter.write_header-def"><a name="L1164"></a><tt class="py-lineno">1164</tt> <a class="py-toggle" href="#" id="SffWriter.write_header-toggle" onclick="return toggle('SffWriter.write_header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO.SffWriter-class.html#write_header">write_header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SffWriter.write_header-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SffWriter.write_header-expanded"><a name="L1165"></a><tt class="py-lineno">1165</tt>  <tt class="py-line">        <tt class="py-comment"># Do header...</tt> </tt>
<a name="L1166"></a><tt class="py-lineno">1166</tt>  <tt class="py-line">        <tt class="py-name">key_length</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_key_sequence</tt><tt class="py-op">)</tt> </tt>
<a name="L1167"></a><tt class="py-lineno">1167</tt>  <tt class="py-line">        <tt class="py-comment"># file header (part one)</tt> </tt>
<a name="L1168"></a><tt class="py-lineno">1168</tt>  <tt class="py-line">        <tt class="py-comment"># use big endiean encdoing   &gt;</tt> </tt>
<a name="L1169"></a><tt class="py-lineno">1169</tt>  <tt class="py-line">        <tt class="py-comment"># magic_number               I</tt> </tt>
<a name="L1170"></a><tt class="py-lineno">1170</tt>  <tt class="py-line">        <tt class="py-comment"># version                    4B</tt> </tt>
<a name="L1171"></a><tt class="py-lineno">1171</tt>  <tt class="py-line">        <tt class="py-comment"># index_offset               Q</tt> </tt>
<a name="L1172"></a><tt class="py-lineno">1172</tt>  <tt class="py-line">        <tt class="py-comment"># index_length               I</tt> </tt>
<a name="L1173"></a><tt class="py-lineno">1173</tt>  <tt class="py-line">        <tt class="py-comment"># number_of_reads            I</tt> </tt>
<a name="L1174"></a><tt class="py-lineno">1174</tt>  <tt class="py-line">        <tt class="py-comment"># header_length              H</tt> </tt>
<a name="L1175"></a><tt class="py-lineno">1175</tt>  <tt class="py-line">        <tt class="py-comment"># key_length                 H</tt> </tt>
<a name="L1176"></a><tt class="py-lineno">1176</tt>  <tt class="py-line">        <tt class="py-comment"># number_of_flows_per_read   H</tt> </tt>
<a name="L1177"></a><tt class="py-lineno">1177</tt>  <tt class="py-line">        <tt class="py-comment"># flowgram_format_code       B</tt> </tt>
<a name="L1178"></a><tt class="py-lineno">1178</tt>  <tt class="py-line">        <tt class="py-comment"># [rest of file header depends on the number of flows and how many keys]</tt> </tt>
<a name="L1179"></a><tt class="py-lineno">1179</tt>  <tt class="py-line">        <tt class="py-name">fmt</tt> <tt class="py-op">=</tt> <tt class="py-string">'&gt;I4BQIIHHHB%is%is'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt> </tt>
<a name="L1180"></a><tt class="py-lineno">1180</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_number_of_flows_per_read</tt><tt class="py-op">,</tt> <tt class="py-name">key_length</tt><tt class="py-op">)</tt> </tt>
<a name="L1181"></a><tt class="py-lineno">1181</tt>  <tt class="py-line">        <tt class="py-comment"># According to the spec, the header_length field should be the total</tt> </tt>
<a name="L1182"></a><tt class="py-lineno">1182</tt>  <tt class="py-line">        <tt class="py-comment"># number of bytes required by this set of header fields, and should be</tt> </tt>
<a name="L1183"></a><tt class="py-lineno">1183</tt>  <tt class="py-line">        <tt class="py-comment"># equal to "31 + number_of_flows_per_read + key_length" rounded up to</tt> </tt>
<a name="L1184"></a><tt class="py-lineno">1184</tt>  <tt class="py-line">        <tt class="py-comment"># the next value divisible by 8.</tt> </tt>
<a name="L1185"></a><tt class="py-lineno">1185</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-name">fmt</tt><tt class="py-op">)</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1186"></a><tt class="py-lineno">1186</tt>  <tt class="py-line">            <tt class="py-name">padding</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L1187"></a><tt class="py-lineno">1187</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1188"></a><tt class="py-lineno">1188</tt>  <tt class="py-line">            <tt class="py-name">padding</tt> <tt class="py-op">=</tt> <tt class="py-number">8</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-name">fmt</tt><tt class="py-op">)</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt><tt class="py-op">)</tt> </tt>
<a name="L1189"></a><tt class="py-lineno">1189</tt>  <tt class="py-line">        <tt class="py-name">header_length</tt> <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-name">fmt</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-name">padding</tt> </tt>
<a name="L1190"></a><tt class="py-lineno">1190</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">header_length</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt> </tt>
<a name="L1191"></a><tt class="py-lineno">1191</tt>  <tt class="py-line">        <tt id="link-324" class="py-name" targets="Variable Bio.Wise.dnal.Statistics.header=Bio.Wise.dnal.Statistics-class.html#header"><a title="Bio.Wise.dnal.Statistics.header" class="py-name" href="#" onclick="return doclink('link-324', 'header', 'link-324');">header</a></tt> <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">pack</tt><tt class="py-op">(</tt><tt class="py-name">fmt</tt><tt class="py-op">,</tt> <tt class="py-number">779314790</tt><tt class="py-op">,</tt>  <tt class="py-comment"># magic number 0x2E736666</tt> </tt>
<a name="L1192"></a><tt class="py-lineno">1192</tt>  <tt class="py-line">                             <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt>  <tt class="py-comment"># version</tt> </tt>
<a name="L1193"></a><tt class="py-lineno">1193</tt>  <tt class="py-line">                             <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_index_start</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_index_length</tt><tt class="py-op">,</tt> </tt>
<a name="L1194"></a><tt class="py-lineno">1194</tt>  <tt class="py-line">                             <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_number_of_reads</tt><tt class="py-op">,</tt> </tt>
<a name="L1195"></a><tt class="py-lineno">1195</tt>  <tt class="py-line">                             <tt class="py-name">header_length</tt><tt class="py-op">,</tt> <tt class="py-name">key_length</tt><tt class="py-op">,</tt> </tt>
<a name="L1196"></a><tt class="py-lineno">1196</tt>  <tt class="py-line">                             <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_number_of_flows_per_read</tt><tt class="py-op">,</tt> </tt>
<a name="L1197"></a><tt class="py-lineno">1197</tt>  <tt class="py-line">                             <tt class="py-number">1</tt><tt class="py-op">,</tt>  <tt class="py-comment"># the only flowgram format code we support</tt> </tt>
<a name="L1198"></a><tt class="py-lineno">1198</tt>  <tt class="py-line">                             <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_flow_chars</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_key_sequence</tt><tt class="py-op">)</tt> </tt>
<a name="L1199"></a><tt class="py-lineno">1199</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-325" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-325', 'write', 'link-307');">write</a></tt><tt class="py-op">(</tt><tt id="link-326" class="py-name"><a title="Bio.Wise.dnal.Statistics.header" class="py-name" href="#" onclick="return doclink('link-326', 'header', 'link-324');">header</a></tt> <tt class="py-op">+</tt> <tt id="link-327" class="py-name"><a title="Bio.SeqIO.SffIO._null" class="py-name" href="#" onclick="return doclink('link-327', '_null', 'link-16');">_null</a></tt> <tt class="py-op">*</tt> <tt class="py-name">padding</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1200"></a><tt class="py-lineno">1200</tt>  <tt class="py-line"> </tt>
<a name="SffWriter.write_record"></a><div id="SffWriter.write_record-def"><a name="L1201"></a><tt class="py-lineno">1201</tt> <a class="py-toggle" href="#" id="SffWriter.write_record-toggle" onclick="return toggle('SffWriter.write_record');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SffIO.SffWriter-class.html#write_record">write_record</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SffWriter.write_record-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SffWriter.write_record-expanded"><a name="L1202"></a><tt class="py-lineno">1202</tt>  <tt class="py-line">        <tt class="py-docstring">"""Write a single additional record to the output file.</tt> </tt>
<a name="L1203"></a><tt class="py-lineno">1203</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1204"></a><tt class="py-lineno">1204</tt>  <tt class="py-line"><tt class="py-docstring">        This assumes the header has been done.</tt> </tt>
<a name="L1205"></a><tt class="py-lineno">1205</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1206"></a><tt class="py-lineno">1206</tt>  <tt class="py-line">        <tt class="py-comment"># Basics</tt> </tt>
<a name="L1207"></a><tt class="py-lineno">1207</tt>  <tt class="py-line">        <tt id="link-328" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-328', 'name', 'link-60');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-329" class="py-name"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-329', '_as_bytes', 'link-15');">_as_bytes</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-330" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-330', 'id', 'link-187');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1208"></a><tt class="py-lineno">1208</tt>  <tt class="py-line">        <tt class="py-name">name_len</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-331" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-331', 'name', 'link-60');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1209"></a><tt class="py-lineno">1209</tt>  <tt class="py-line">        <tt id="link-332" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-332', 'seq', 'link-137');">seq</a></tt> <tt class="py-op">=</tt> <tt id="link-333" class="py-name"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-333', '_as_bytes', 'link-15');">_as_bytes</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-334" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-334', 'seq', 'link-137');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-335" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-335', 'upper', 'link-28');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1210"></a><tt class="py-lineno">1210</tt>  <tt class="py-line">        <tt class="py-name">seq_len</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-336" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-336', 'seq', 'link-137');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1211"></a><tt class="py-lineno">1211</tt>  <tt class="py-line">        <tt class="py-comment"># Qualities</tt> </tt>
<a name="L1212"></a><tt class="py-lineno">1212</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1213"></a><tt class="py-lineno">1213</tt>  <tt class="py-line">            <tt class="py-name">quals</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-337" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.letter_annotations=Bio.SeqRecord.SeqRecord-class.html#letter_annotations"><a title="Bio.SeqRecord.SeqRecord.letter_annotations" class="py-name" href="#" onclick="return doclink('link-337', 'letter_annotations', 'link-337');">letter_annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"phred_quality"</tt><tt class="py-op">]</tt> </tt>
<a name="L1214"></a><tt class="py-lineno">1214</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L1215"></a><tt class="py-lineno">1215</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Missing PHRED qualities information for %s"</tt> <tt class="py-op">%</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-338" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-338', 'id', 'link-187');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1216"></a><tt class="py-lineno">1216</tt>  <tt class="py-line">        <tt class="py-comment"># Flow</tt> </tt>
<a name="L1217"></a><tt class="py-lineno">1217</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1218"></a><tt class="py-lineno">1218</tt>  <tt class="py-line">            <tt class="py-name">flow_values</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-339" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-339', 'annotations', 'link-156');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"flow_values"</tt><tt class="py-op">]</tt> </tt>
<a name="L1219"></a><tt class="py-lineno">1219</tt>  <tt class="py-line">            <tt class="py-name">flow_index</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-340" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-340', 'annotations', 'link-156');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"flow_index"</tt><tt class="py-op">]</tt> </tt>
<a name="L1220"></a><tt class="py-lineno">1220</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_key_sequence</tt> <tt class="py-op">!=</tt> <tt id="link-341" class="py-name"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-341', '_as_bytes', 'link-15');">_as_bytes</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-342" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-342', 'annotations', 'link-156');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"flow_key"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> \ </tt>
<a name="L1221"></a><tt class="py-lineno">1221</tt>  <tt class="py-line">                    <tt class="py-keyword">or</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_flow_chars</tt> <tt class="py-op">!=</tt> <tt id="link-343" class="py-name"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-343', '_as_bytes', 'link-15');">_as_bytes</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-344" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-344', 'annotations', 'link-156');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"flow_chars"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1222"></a><tt class="py-lineno">1222</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Records have inconsistent SFF flow data"</tt><tt class="py-op">)</tt> </tt>
<a name="L1223"></a><tt class="py-lineno">1223</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L1224"></a><tt class="py-lineno">1224</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Missing SFF flow information for %s"</tt> <tt class="py-op">%</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-345" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-345', 'id', 'link-187');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1225"></a><tt class="py-lineno">1225</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L1226"></a><tt class="py-lineno">1226</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Header not written yet?"</tt><tt class="py-op">)</tt> </tt>
<a name="L1227"></a><tt class="py-lineno">1227</tt>  <tt class="py-line">        <tt class="py-comment"># Clipping</tt> </tt>
<a name="L1228"></a><tt class="py-lineno">1228</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1229"></a><tt class="py-lineno">1229</tt>  <tt class="py-line">            <tt class="py-name">clip_qual_left</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-346" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-346', 'annotations', 'link-156');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"clip_qual_left"</tt><tt class="py-op">]</tt> </tt>
<a name="L1230"></a><tt class="py-lineno">1230</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">clip_qual_left</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1231"></a><tt class="py-lineno">1231</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Negative SFF clip_qual_left value for %s"</tt> <tt class="py-op">%</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-347" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-347', 'id', 'link-187');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1232"></a><tt class="py-lineno">1232</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">clip_qual_left</tt><tt class="py-op">:</tt> </tt>
<a name="L1233"></a><tt class="py-lineno">1233</tt>  <tt class="py-line">                <tt class="py-name">clip_qual_left</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L1234"></a><tt class="py-lineno">1234</tt>  <tt class="py-line">            <tt class="py-name">clip_qual_right</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-348" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-348', 'annotations', 'link-156');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"clip_qual_right"</tt><tt class="py-op">]</tt> </tt>
<a name="L1235"></a><tt class="py-lineno">1235</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">clip_qual_right</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1236"></a><tt class="py-lineno">1236</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Negative SFF clip_qual_right value for %s"</tt> <tt class="py-op">%</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-349" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-349', 'id', 'link-187');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1237"></a><tt class="py-lineno">1237</tt>  <tt class="py-line">            <tt class="py-name">clip_adapter_left</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-350" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-350', 'annotations', 'link-156');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"clip_adapter_left"</tt><tt class="py-op">]</tt> </tt>
<a name="L1238"></a><tt class="py-lineno">1238</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">clip_adapter_left</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1239"></a><tt class="py-lineno">1239</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Negative SFF clip_adapter_left value for %s"</tt> <tt class="py-op">%</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-351" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-351', 'id', 'link-187');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1240"></a><tt class="py-lineno">1240</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">clip_adapter_left</tt><tt class="py-op">:</tt> </tt>
<a name="L1241"></a><tt class="py-lineno">1241</tt>  <tt class="py-line">                <tt class="py-name">clip_adapter_left</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L1242"></a><tt class="py-lineno">1242</tt>  <tt class="py-line">            <tt class="py-name">clip_adapter_right</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-352" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-352', 'annotations', 'link-156');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"clip_adapter_right"</tt><tt class="py-op">]</tt> </tt>
<a name="L1243"></a><tt class="py-lineno">1243</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">clip_adapter_right</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1244"></a><tt class="py-lineno">1244</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Negative SFF clip_adapter_right value for %s"</tt> <tt class="py-op">%</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-353" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-353', 'id', 'link-187');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1245"></a><tt class="py-lineno">1245</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L1246"></a><tt class="py-lineno">1246</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Missing SFF clipping information for %s"</tt> <tt class="py-op">%</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-354" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-354', 'id', 'link-187');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1247"></a><tt class="py-lineno">1247</tt>  <tt class="py-line"> </tt>
<a name="L1248"></a><tt class="py-lineno">1248</tt>  <tt class="py-line">        <tt class="py-comment"># Capture information for index</tt> </tt>
<a name="L1249"></a><tt class="py-lineno">1249</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-355" class="py-name"><a title="Bio.SearchIO._index
Bio.SeqIO._index" class="py-name" href="#" onclick="return doclink('link-355', '_index', 'link-277');">_index</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1250"></a><tt class="py-lineno">1250</tt>  <tt class="py-line">            <tt class="py-name">offset</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-356" class="py-name"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-356', 'tell', 'link-54');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1251"></a><tt class="py-lineno">1251</tt>  <tt class="py-line">            <tt class="py-comment"># Check the position of the final record (before sort by name)</tt> </tt>
<a name="L1252"></a><tt class="py-lineno">1252</tt>  <tt class="py-line">            <tt class="py-comment"># Using a four-digit base 255 number, so the upper bound is</tt> </tt>
<a name="L1253"></a><tt class="py-lineno">1253</tt>  <tt class="py-line">            <tt class="py-comment"># 254*(1)+254*(255)+254*(255**2)+254*(255**3) = 4228250624</tt> </tt>
<a name="L1254"></a><tt class="py-lineno">1254</tt>  <tt class="py-line">            <tt class="py-comment"># or equivalently it overflows at 255**4 = 4228250625</tt> </tt>
<a name="L1255"></a><tt class="py-lineno">1255</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">offset</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">4228250624</tt><tt class="py-op">:</tt> </tt>
<a name="L1256"></a><tt class="py-lineno">1256</tt>  <tt class="py-line">                <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L1257"></a><tt class="py-lineno">1257</tt>  <tt class="py-line">                <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Read %s has file offset %i, which is too large "</tt> </tt>
<a name="L1258"></a><tt class="py-lineno">1258</tt>  <tt class="py-line">                              <tt class="py-string">"to store in the Roche SFF index structure. No "</tt> </tt>
<a name="L1259"></a><tt class="py-lineno">1259</tt>  <tt class="py-line">                              <tt class="py-string">"index block will be recorded."</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-357" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-357', 'name', 'link-60');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">offset</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1260"></a><tt class="py-lineno">1260</tt>  <tt class="py-line">                <tt class="py-comment"># No point recoring the offsets now</tt> </tt>
<a name="L1261"></a><tt class="py-lineno">1261</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-358" class="py-name"><a title="Bio.SearchIO._index
Bio.SeqIO._index" class="py-name" href="#" onclick="return doclink('link-358', '_index', 'link-277');">_index</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L1262"></a><tt class="py-lineno">1262</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1263"></a><tt class="py-lineno">1263</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-359" class="py-name"><a title="Bio.SearchIO._index
Bio.SeqIO._index" class="py-name" href="#" onclick="return doclink('link-359', '_index', 'link-277');">_index</a></tt><tt class="py-op">.</tt><tt id="link-360" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-360', 'append', 'link-206');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt id="link-361" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-361', 'name', 'link-60');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-362" class="py-name"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-362', 'tell', 'link-54');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1264"></a><tt class="py-lineno">1264</tt>  <tt class="py-line"> </tt>
<a name="L1265"></a><tt class="py-lineno">1265</tt>  <tt class="py-line">        <tt class="py-comment"># the read header format (fixed part):</tt> </tt>
<a name="L1266"></a><tt class="py-lineno">1266</tt>  <tt class="py-line">        <tt class="py-comment"># read_header_length     H</tt> </tt>
<a name="L1267"></a><tt class="py-lineno">1267</tt>  <tt class="py-line">        <tt class="py-comment"># name_length            H</tt> </tt>
<a name="L1268"></a><tt class="py-lineno">1268</tt>  <tt class="py-line">        <tt class="py-comment"># seq_len                I</tt> </tt>
<a name="L1269"></a><tt class="py-lineno">1269</tt>  <tt class="py-line">        <tt class="py-comment"># clip_qual_left         H</tt> </tt>
<a name="L1270"></a><tt class="py-lineno">1270</tt>  <tt class="py-line">        <tt class="py-comment"># clip_qual_right        H</tt> </tt>
<a name="L1271"></a><tt class="py-lineno">1271</tt>  <tt class="py-line">        <tt class="py-comment"># clip_adapter_left      H</tt> </tt>
<a name="L1272"></a><tt class="py-lineno">1272</tt>  <tt class="py-line">        <tt class="py-comment"># clip_adapter_right     H</tt> </tt>
<a name="L1273"></a><tt class="py-lineno">1273</tt>  <tt class="py-line">        <tt class="py-comment"># [rest of read header depends on the name length etc]</tt> </tt>
<a name="L1274"></a><tt class="py-lineno">1274</tt>  <tt class="py-line">        <tt class="py-comment"># name</tt> </tt>
<a name="L1275"></a><tt class="py-lineno">1275</tt>  <tt class="py-line">        <tt class="py-comment"># flow values</tt> </tt>
<a name="L1276"></a><tt class="py-lineno">1276</tt>  <tt class="py-line">        <tt class="py-comment"># flow index</tt> </tt>
<a name="L1277"></a><tt class="py-lineno">1277</tt>  <tt class="py-line">        <tt class="py-comment"># sequence</tt> </tt>
<a name="L1278"></a><tt class="py-lineno">1278</tt>  <tt class="py-line">        <tt class="py-comment"># padding</tt> </tt>
<a name="L1279"></a><tt class="py-lineno">1279</tt>  <tt class="py-line">        <tt class="py-name">read_header_fmt</tt> <tt class="py-op">=</tt> <tt class="py-string">'&gt;2HI4H%is'</tt> <tt class="py-op">%</tt> <tt class="py-name">name_len</tt> </tt>
<a name="L1280"></a><tt class="py-lineno">1280</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-name">read_header_fmt</tt><tt class="py-op">)</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1281"></a><tt class="py-lineno">1281</tt>  <tt class="py-line">            <tt class="py-name">padding</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L1282"></a><tt class="py-lineno">1282</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1283"></a><tt class="py-lineno">1283</tt>  <tt class="py-line">            <tt class="py-name">padding</tt> <tt class="py-op">=</tt> <tt class="py-number">8</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-name">read_header_fmt</tt><tt class="py-op">)</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt><tt class="py-op">)</tt> </tt>
<a name="L1284"></a><tt class="py-lineno">1284</tt>  <tt class="py-line">        <tt class="py-name">read_header_length</tt> <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-name">read_header_fmt</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-name">padding</tt> </tt>
<a name="L1285"></a><tt class="py-lineno">1285</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">read_header_length</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt> </tt>
<a name="L1286"></a><tt class="py-lineno">1286</tt>  <tt class="py-line">        <tt id="link-363" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-363', 'data', 'link-31');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">pack</tt><tt class="py-op">(</tt><tt class="py-name">read_header_fmt</tt><tt class="py-op">,</tt> </tt>
<a name="L1287"></a><tt class="py-lineno">1287</tt>  <tt class="py-line">                           <tt class="py-name">read_header_length</tt><tt class="py-op">,</tt> </tt>
<a name="L1288"></a><tt class="py-lineno">1288</tt>  <tt class="py-line">                           <tt class="py-name">name_len</tt><tt class="py-op">,</tt> <tt class="py-name">seq_len</tt><tt class="py-op">,</tt> </tt>
<a name="L1289"></a><tt class="py-lineno">1289</tt>  <tt class="py-line">                           <tt class="py-name">clip_qual_left</tt><tt class="py-op">,</tt> <tt class="py-name">clip_qual_right</tt><tt class="py-op">,</tt> </tt>
<a name="L1290"></a><tt class="py-lineno">1290</tt>  <tt class="py-line">                           <tt class="py-name">clip_adapter_left</tt><tt class="py-op">,</tt> <tt class="py-name">clip_adapter_right</tt><tt class="py-op">,</tt> </tt>
<a name="L1291"></a><tt class="py-lineno">1291</tt>  <tt class="py-line">                           <tt id="link-364" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-364', 'name', 'link-60');">name</a></tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt id="link-365" class="py-name"><a title="Bio.SeqIO.SffIO._null" class="py-name" href="#" onclick="return doclink('link-365', '_null', 'link-16');">_null</a></tt> <tt class="py-op">*</tt> <tt class="py-name">padding</tt> </tt>
<a name="L1292"></a><tt class="py-lineno">1292</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-366" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-366', 'data', 'link-31');">data</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">read_header_length</tt> </tt>
<a name="L1293"></a><tt class="py-lineno">1293</tt>  <tt class="py-line">        <tt class="py-comment"># now the flowgram values, flowgram index, bases and qualities</tt> </tt>
<a name="L1294"></a><tt class="py-lineno">1294</tt>  <tt class="py-line">        <tt class="py-comment"># NOTE - assuming flowgram_format==1, which means struct type H</tt> </tt>
<a name="L1295"></a><tt class="py-lineno">1295</tt>  <tt class="py-line">        <tt class="py-name">read_flow_fmt</tt> <tt class="py-op">=</tt> <tt class="py-string">"&gt;%iH"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_number_of_flows_per_read</tt> </tt>
<a name="L1296"></a><tt class="py-lineno">1296</tt>  <tt class="py-line">        <tt class="py-name">read_flow_size</tt> <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-name">read_flow_fmt</tt><tt class="py-op">)</tt> </tt>
<a name="L1297"></a><tt class="py-lineno">1297</tt>  <tt class="py-line">        <tt class="py-name">temp_fmt</tt> <tt class="py-op">=</tt> <tt class="py-string">"&gt;%iB"</tt> <tt class="py-op">%</tt> <tt class="py-name">seq_len</tt>  <tt class="py-comment"># used for flow index and quals</tt> </tt>
<a name="L1298"></a><tt class="py-lineno">1298</tt>  <tt class="py-line">        <tt id="link-367" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-367', 'data', 'link-31');">data</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">pack</tt><tt class="py-op">(</tt><tt class="py-name">read_flow_fmt</tt><tt class="py-op">,</tt> <tt class="py-op">*</tt><tt class="py-name">flow_values</tt><tt class="py-op">)</tt> \ </tt>
<a name="L1299"></a><tt class="py-lineno">1299</tt>  <tt class="py-line">            <tt class="py-op">+</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">pack</tt><tt class="py-op">(</tt><tt class="py-name">temp_fmt</tt><tt class="py-op">,</tt> <tt class="py-op">*</tt><tt class="py-name">flow_index</tt><tt class="py-op">)</tt> \ </tt>
<a name="L1300"></a><tt class="py-lineno">1300</tt>  <tt class="py-line">            <tt class="py-op">+</tt> <tt id="link-368" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-368', 'seq', 'link-137');">seq</a></tt> \ </tt>
<a name="L1301"></a><tt class="py-lineno">1301</tt>  <tt class="py-line">            <tt class="py-op">+</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">pack</tt><tt class="py-op">(</tt><tt class="py-name">temp_fmt</tt><tt class="py-op">,</tt> <tt class="py-op">*</tt><tt class="py-name">quals</tt><tt class="py-op">)</tt> </tt>
<a name="L1302"></a><tt class="py-lineno">1302</tt>  <tt class="py-line">        <tt class="py-comment"># now any final padding...</tt> </tt>
<a name="L1303"></a><tt class="py-lineno">1303</tt>  <tt class="py-line">        <tt class="py-name">padding</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">read_flow_size</tt> <tt class="py-op">+</tt> <tt class="py-name">seq_len</tt> <tt class="py-op">*</tt> <tt class="py-number">3</tt><tt class="py-op">)</tt> <tt class="py-op">%</tt> <tt class="py-number">8</tt> </tt>
<a name="L1304"></a><tt class="py-lineno">1304</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">padding</tt><tt class="py-op">:</tt> </tt>
<a name="L1305"></a><tt class="py-lineno">1305</tt>  <tt class="py-line">            <tt class="py-name">padding</tt> <tt class="py-op">=</tt> <tt class="py-number">8</tt> <tt class="py-op">-</tt> <tt class="py-name">padding</tt> </tt>
<a name="L1306"></a><tt class="py-lineno">1306</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-369" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-369', 'write', 'link-307');">write</a></tt><tt class="py-op">(</tt><tt id="link-370" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-370', 'data', 'link-31');">data</a></tt> <tt class="py-op">+</tt> <tt id="link-371" class="py-name"><a title="Bio.SeqIO.SffIO._null" class="py-name" href="#" onclick="return doclink('link-371', '_null', 'link-16');">_null</a></tt> <tt class="py-op">*</tt> <tt class="py-name">padding</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1307"></a><tt class="py-lineno">1307</tt>  <tt class="py-line"> </tt>
<a name="L1308"></a><tt class="py-lineno">1308</tt>  <tt class="py-line"> </tt>
<a name="L1309"></a><tt class="py-lineno">1309</tt>  <tt class="py-line"><tt class="py-keyword">if</tt> <tt class="py-name">__name__</tt> <tt class="py-op">==</tt> <tt class="py-string">"__main__"</tt><tt class="py-op">:</tt> </tt>
<a name="L1310"></a><tt class="py-lineno">1310</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Running quick self test"</tt><tt class="py-op">)</tt> </tt>
<a name="L1311"></a><tt class="py-lineno">1311</tt>  <tt class="py-line">    <tt class="py-name">filename</tt> <tt class="py-op">=</tt> <tt class="py-string">"../../Tests/Roche/E3MFGYR02_random_10_reads.sff"</tt> </tt>
<a name="L1312"></a><tt class="py-lineno">1312</tt>  <tt class="py-line">    <tt class="py-keyword">with</tt> <tt id="link-372" class="py-name" targets="Function Bio.bgzf.open()=Bio.bgzf-module.html#open"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-372', 'open', 'link-372');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-string">"rb"</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1313"></a><tt class="py-lineno">1313</tt>  <tt class="py-line">        <tt class="py-name">metadata</tt> <tt class="py-op">=</tt> <tt id="link-373" class="py-name" targets="Function Bio.SeqIO.SffIO.ReadRocheXmlManifest()=Bio.SeqIO.SffIO-module.html#ReadRocheXmlManifest"><a title="Bio.SeqIO.SffIO.ReadRocheXmlManifest" class="py-name" href="#" onclick="return doclink('link-373', 'ReadRocheXmlManifest', 'link-373');">ReadRocheXmlManifest</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L1314"></a><tt class="py-lineno">1314</tt>  <tt class="py-line">    <tt class="py-keyword">with</tt> <tt id="link-374" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-374', 'open', 'link-372');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-string">"rb"</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1315"></a><tt class="py-lineno">1315</tt>  <tt class="py-line">        <tt class="py-name">index1</tt> <tt class="py-op">=</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt id="link-375" class="py-name" targets="Function Bio.SeqIO.SffIO._sff_read_roche_index()=Bio.SeqIO.SffIO-module.html#_sff_read_roche_index"><a title="Bio.SeqIO.SffIO._sff_read_roche_index" class="py-name" href="#" onclick="return doclink('link-375', '_sff_read_roche_index', 'link-375');">_sff_read_roche_index</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1316"></a><tt class="py-lineno">1316</tt>  <tt class="py-line">    <tt class="py-keyword">with</tt> <tt id="link-376" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-376', 'open', 'link-372');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-string">"rb"</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1317"></a><tt class="py-lineno">1317</tt>  <tt class="py-line">        <tt class="py-name">index2</tt> <tt class="py-op">=</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt id="link-377" class="py-name" targets="Function Bio.SeqIO.SffIO._sff_do_slow_index()=Bio.SeqIO.SffIO-module.html#_sff_do_slow_index"><a title="Bio.SeqIO.SffIO._sff_do_slow_index" class="py-name" href="#" onclick="return doclink('link-377', '_sff_do_slow_index', 'link-377');">_sff_do_slow_index</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1318"></a><tt class="py-lineno">1318</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">index1</tt> <tt class="py-op">==</tt> <tt class="py-name">index2</tt> </tt>
<a name="L1319"></a><tt class="py-lineno">1319</tt>  <tt class="py-line">    <tt class="py-keyword">with</tt> <tt id="link-378" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-378', 'open', 'link-372');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-string">"rb"</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1320"></a><tt class="py-lineno">1320</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">index1</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-379" class="py-name"><a title="Bio.SeqIO.SffIO.SffIterator" class="py-name" href="#" onclick="return doclink('link-379', 'SffIterator', 'link-273');">SffIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1321"></a><tt class="py-lineno">1321</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-380" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-380', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-381" class="py-name"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-381', '_py3k', 'link-13');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">StringIO</tt> </tt>
<a name="L1322"></a><tt class="py-lineno">1322</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt class="py-name">io</tt> <tt class="py-keyword">import</tt> <tt class="py-name">BytesIO</tt> </tt>
<a name="L1323"></a><tt class="py-lineno">1323</tt>  <tt class="py-line">    <tt class="py-keyword">with</tt> <tt id="link-382" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-382', 'open', 'link-372');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-string">"rb"</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1324"></a><tt class="py-lineno">1324</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">index1</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-383" class="py-name"><a title="Bio.SeqIO.SffIO.SffIterator" class="py-name" href="#" onclick="return doclink('link-383', 'SffIterator', 'link-273');">SffIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">BytesIO</tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-384" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-384', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1325"></a><tt class="py-lineno">1325</tt>  <tt class="py-line"> </tt>
<a name="L1326"></a><tt class="py-lineno">1326</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">platform</tt> <tt class="py-op">!=</tt> <tt class="py-string">"win32"</tt> <tt class="py-keyword">and</tt> <tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">version_info</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">3</tt><tt class="py-op">:</tt> </tt>
<a name="L1327"></a><tt class="py-lineno">1327</tt>  <tt class="py-line">        <tt class="py-comment"># Can be lazy and treat as binary...</tt> </tt>
<a name="L1328"></a><tt class="py-lineno">1328</tt>  <tt class="py-line">        <tt class="py-keyword">with</tt> <tt id="link-385" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-385', 'open', 'link-372');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-string">"r"</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1329"></a><tt class="py-lineno">1329</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">index1</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-386" class="py-name"><a title="Bio.SeqIO.SffIO.SffIterator" class="py-name" href="#" onclick="return doclink('link-386', 'SffIterator', 'link-273');">SffIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1330"></a><tt class="py-lineno">1330</tt>  <tt class="py-line">        <tt class="py-keyword">with</tt> <tt id="link-387" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-387', 'open', 'link-372');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1331"></a><tt class="py-lineno">1331</tt>  <tt class="py-line">            <tt class="py-name">index2</tt> <tt class="py-op">=</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt id="link-388" class="py-name"><a title="Bio.SeqIO.SffIO._sff_read_roche_index" class="py-name" href="#" onclick="return doclink('link-388', '_sff_read_roche_index', 'link-375');">_sff_read_roche_index</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1332"></a><tt class="py-lineno">1332</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">index1</tt> <tt class="py-op">==</tt> <tt class="py-name">index2</tt> </tt>
<a name="L1333"></a><tt class="py-lineno">1333</tt>  <tt class="py-line">        <tt class="py-keyword">with</tt> <tt id="link-389" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-389', 'open', 'link-372');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-string">"r"</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1334"></a><tt class="py-lineno">1334</tt>  <tt class="py-line">            <tt class="py-name">index2</tt> <tt class="py-op">=</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt id="link-390" class="py-name"><a title="Bio.SeqIO.SffIO._sff_do_slow_index" class="py-name" href="#" onclick="return doclink('link-390', '_sff_do_slow_index', 'link-377');">_sff_do_slow_index</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1335"></a><tt class="py-lineno">1335</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">index1</tt> <tt class="py-op">==</tt> <tt class="py-name">index2</tt> </tt>
<a name="L1336"></a><tt class="py-lineno">1336</tt>  <tt class="py-line">        <tt class="py-keyword">with</tt> <tt id="link-391" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-391', 'open', 'link-372');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-string">"r"</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1337"></a><tt class="py-lineno">1337</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">index1</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-392" class="py-name"><a title="Bio.SeqIO.SffIO.SffIterator" class="py-name" href="#" onclick="return doclink('link-392', 'SffIterator', 'link-273');">SffIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1338"></a><tt class="py-lineno">1338</tt>  <tt class="py-line">        <tt class="py-keyword">with</tt> <tt id="link-393" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-393', 'open', 'link-372');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-string">"r"</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1339"></a><tt class="py-lineno">1339</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">index1</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-394" class="py-name"><a title="Bio.SeqIO.SffIO.SffIterator" class="py-name" href="#" onclick="return doclink('link-394', 'SffIterator', 'link-273');">SffIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">BytesIO</tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-395" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-395', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1340"></a><tt class="py-lineno">1340</tt>  <tt class="py-line"> </tt>
<a name="L1341"></a><tt class="py-lineno">1341</tt>  <tt class="py-line">    <tt class="py-keyword">with</tt> <tt id="link-396" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-396', 'open', 'link-372');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-string">"rb"</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1342"></a><tt class="py-lineno">1342</tt>  <tt class="py-line">        <tt class="py-name">sff</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-397" class="py-name"><a title="Bio.SeqIO.SffIO.SffIterator" class="py-name" href="#" onclick="return doclink('link-397', 'SffIterator', 'link-273');">SffIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1343"></a><tt class="py-lineno">1343</tt>  <tt class="py-line"> </tt>
<a name="L1344"></a><tt class="py-lineno">1344</tt>  <tt class="py-line">    <tt class="py-keyword">with</tt> <tt id="link-398" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-398', 'open', 'link-372');">open</a></tt><tt class="py-op">(</tt><tt class="py-string">"../../Tests/Roche/E3MFGYR02_alt_index_at_end.sff"</tt><tt class="py-op">,</tt> <tt class="py-string">"rb"</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1345"></a><tt class="py-lineno">1345</tt>  <tt class="py-line">        <tt class="py-name">sff2</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-399" class="py-name"><a title="Bio.SeqIO.SffIO.SffIterator" class="py-name" href="#" onclick="return doclink('link-399', 'SffIterator', 'link-273');">SffIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1346"></a><tt class="py-lineno">1346</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">sff</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">sff2</tt><tt class="py-op">)</tt> </tt>
<a name="L1347"></a><tt class="py-lineno">1347</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">old</tt><tt class="py-op">,</tt> <tt class="py-name">new</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">sff</tt><tt class="py-op">,</tt> <tt class="py-name">sff2</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1348"></a><tt class="py-lineno">1348</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-400" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-400', 'id', 'link-187');">id</a></tt> <tt class="py-op">==</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-401" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-401', 'id', 'link-187');">id</a></tt> </tt>
<a name="L1349"></a><tt class="py-lineno">1349</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-402" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-402', 'seq', 'link-137');">seq</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-403" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-403', 'seq', 'link-137');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1350"></a><tt class="py-lineno">1350</tt>  <tt class="py-line"> </tt>
<a name="L1351"></a><tt class="py-lineno">1351</tt>  <tt class="py-line">    <tt class="py-keyword">with</tt> <tt id="link-404" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-404', 'open', 'link-372');">open</a></tt><tt class="py-op">(</tt><tt class="py-string">"../../Tests/Roche/E3MFGYR02_alt_index_at_start.sff"</tt><tt class="py-op">,</tt> <tt class="py-string">"rb"</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1352"></a><tt class="py-lineno">1352</tt>  <tt class="py-line">        <tt class="py-name">sff2</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-405" class="py-name"><a title="Bio.SeqIO.SffIO.SffIterator" class="py-name" href="#" onclick="return doclink('link-405', 'SffIterator', 'link-273');">SffIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1353"></a><tt class="py-lineno">1353</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">sff</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">sff2</tt><tt class="py-op">)</tt> </tt>
<a name="L1354"></a><tt class="py-lineno">1354</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">old</tt><tt class="py-op">,</tt> <tt class="py-name">new</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">sff</tt><tt class="py-op">,</tt> <tt class="py-name">sff2</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1355"></a><tt class="py-lineno">1355</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-406" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-406', 'id', 'link-187');">id</a></tt> <tt class="py-op">==</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-407" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-407', 'id', 'link-187');">id</a></tt> </tt>
<a name="L1356"></a><tt class="py-lineno">1356</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-408" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-408', 'seq', 'link-137');">seq</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-409" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-409', 'seq', 'link-137');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1357"></a><tt class="py-lineno">1357</tt>  <tt class="py-line"> </tt>
<a name="L1358"></a><tt class="py-lineno">1358</tt>  <tt class="py-line">    <tt class="py-keyword">with</tt> <tt id="link-410" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-410', 'open', 'link-372');">open</a></tt><tt class="py-op">(</tt><tt class="py-string">"../../Tests/Roche/E3MFGYR02_alt_index_in_middle.sff"</tt><tt class="py-op">,</tt> <tt class="py-string">"rb"</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1359"></a><tt class="py-lineno">1359</tt>  <tt class="py-line">        <tt class="py-name">sff2</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-411" class="py-name"><a title="Bio.SeqIO.SffIO.SffIterator" class="py-name" href="#" onclick="return doclink('link-411', 'SffIterator', 'link-273');">SffIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1360"></a><tt class="py-lineno">1360</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">sff</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">sff2</tt><tt class="py-op">)</tt> </tt>
<a name="L1361"></a><tt class="py-lineno">1361</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">old</tt><tt class="py-op">,</tt> <tt class="py-name">new</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">sff</tt><tt class="py-op">,</tt> <tt class="py-name">sff2</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1362"></a><tt class="py-lineno">1362</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-412" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-412', 'id', 'link-187');">id</a></tt> <tt class="py-op">==</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-413" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-413', 'id', 'link-187');">id</a></tt> </tt>
<a name="L1363"></a><tt class="py-lineno">1363</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-414" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-414', 'seq', 'link-137');">seq</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-415" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-415', 'seq', 'link-137');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1364"></a><tt class="py-lineno">1364</tt>  <tt class="py-line"> </tt>
<a name="L1365"></a><tt class="py-lineno">1365</tt>  <tt class="py-line">    <tt class="py-keyword">with</tt> <tt id="link-416" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-416', 'open', 'link-372');">open</a></tt><tt class="py-op">(</tt><tt class="py-string">"../../Tests/Roche/E3MFGYR02_index_at_start.sff"</tt><tt class="py-op">,</tt> <tt class="py-string">"rb"</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1366"></a><tt class="py-lineno">1366</tt>  <tt class="py-line">        <tt class="py-name">sff2</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-417" class="py-name"><a title="Bio.SeqIO.SffIO.SffIterator" class="py-name" href="#" onclick="return doclink('link-417', 'SffIterator', 'link-273');">SffIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1367"></a><tt class="py-lineno">1367</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">sff</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">sff2</tt><tt class="py-op">)</tt> </tt>
<a name="L1368"></a><tt class="py-lineno">1368</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">old</tt><tt class="py-op">,</tt> <tt class="py-name">new</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">sff</tt><tt class="py-op">,</tt> <tt class="py-name">sff2</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1369"></a><tt class="py-lineno">1369</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-418" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-418', 'id', 'link-187');">id</a></tt> <tt class="py-op">==</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-419" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-419', 'id', 'link-187');">id</a></tt> </tt>
<a name="L1370"></a><tt class="py-lineno">1370</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-420" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-420', 'seq', 'link-137');">seq</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-421" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-421', 'seq', 'link-137');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1371"></a><tt class="py-lineno">1371</tt>  <tt class="py-line"> </tt>
<a name="L1372"></a><tt class="py-lineno">1372</tt>  <tt class="py-line">    <tt class="py-keyword">with</tt> <tt id="link-422" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-422', 'open', 'link-372');">open</a></tt><tt class="py-op">(</tt><tt class="py-string">"../../Tests/Roche/E3MFGYR02_index_in_middle.sff"</tt><tt class="py-op">,</tt> <tt class="py-string">"rb"</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1373"></a><tt class="py-lineno">1373</tt>  <tt class="py-line">        <tt class="py-name">sff2</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-423" class="py-name"><a title="Bio.SeqIO.SffIO.SffIterator" class="py-name" href="#" onclick="return doclink('link-423', 'SffIterator', 'link-273');">SffIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1374"></a><tt class="py-lineno">1374</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">sff</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">sff2</tt><tt class="py-op">)</tt> </tt>
<a name="L1375"></a><tt class="py-lineno">1375</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">old</tt><tt class="py-op">,</tt> <tt class="py-name">new</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">sff</tt><tt class="py-op">,</tt> <tt class="py-name">sff2</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1376"></a><tt class="py-lineno">1376</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-424" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-424', 'id', 'link-187');">id</a></tt> <tt class="py-op">==</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-425" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-425', 'id', 'link-187');">id</a></tt> </tt>
<a name="L1377"></a><tt class="py-lineno">1377</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-426" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-426', 'seq', 'link-137');">seq</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-427" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-427', 'seq', 'link-137');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1378"></a><tt class="py-lineno">1378</tt>  <tt class="py-line"> </tt>
<a name="L1379"></a><tt class="py-lineno">1379</tt>  <tt class="py-line">    <tt class="py-keyword">with</tt> <tt id="link-428" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-428', 'open', 'link-372');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-string">"rb"</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1380"></a><tt class="py-lineno">1380</tt>  <tt class="py-line">        <tt class="py-name">sff_trim</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-429" class="py-name"><a title="Bio.SeqIO.SffIO.SffIterator" class="py-name" href="#" onclick="return doclink('link-429', 'SffIterator', 'link-273');">SffIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-name">trim</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1381"></a><tt class="py-lineno">1381</tt>  <tt class="py-line"> </tt>
<a name="L1382"></a><tt class="py-lineno">1382</tt>  <tt class="py-line">    <tt class="py-keyword">with</tt> <tt id="link-430" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-430', 'open', 'link-372');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-string">"rb"</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1383"></a><tt class="py-lineno">1383</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt id="link-431" class="py-name"><a title="Bio.SeqIO.SffIO.ReadRocheXmlManifest" class="py-name" href="#" onclick="return doclink('link-431', 'ReadRocheXmlManifest', 'link-373');">ReadRocheXmlManifest</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1384"></a><tt class="py-lineno">1384</tt>  <tt class="py-line"> </tt>
<a name="L1385"></a><tt class="py-lineno">1385</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-432" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-432', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-433" class="py-name"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-433', 'SeqIO', 'link-1');">SeqIO</a></tt> </tt>
<a name="L1386"></a><tt class="py-lineno">1386</tt>  <tt class="py-line">    <tt class="py-name">filename</tt> <tt class="py-op">=</tt> <tt class="py-string">"../../Tests/Roche/E3MFGYR02_random_10_reads_no_trim.fasta"</tt> </tt>
<a name="L1387"></a><tt class="py-lineno">1387</tt>  <tt class="py-line">    <tt class="py-name">fasta_no_trim</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-434" class="py-name"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-434', 'SeqIO', 'link-1');">SeqIO</a></tt><tt class="py-op">.</tt><tt id="link-435" class="py-name" targets="Function Bio.AlignIO.parse()=Bio.AlignIO-module.html#parse,Method Bio.Blast.NCBIStandalone.BlastErrorParser.parse()=Bio.Blast.NCBIStandalone.BlastErrorParser-class.html#parse,Method Bio.Blast.NCBIStandalone.BlastParser.parse()=Bio.Blast.NCBIStandalone.BlastParser-class.html#parse,Method Bio.Blast.NCBIStandalone.PSIBlastParser.parse()=Bio.Blast.NCBIStandalone.PSIBlastParser-class.html#parse,Function Bio.Blast.NCBIXML.parse()=Bio.Blast.NCBIXML-module.html#parse,Function Bio.Compass.parse()=Bio.Compass-module.html#parse,Function Bio.Emboss.Primer3.parse()=Bio.Emboss.Primer3-module.html#parse,Method Bio.Entrez.Parser.DataHandler.parse()=Bio.Entrez.Parser.DataHandler-class.html#parse,Function Bio.Entrez.parse()=Bio.Entrez-module.html#parse,Function Bio.ExPASy.Enzyme.parse()=Bio.ExPASy.Enzyme-module.html#parse,Function Bio.ExPASy.Prodoc.parse()=Bio.ExPASy.Prodoc-module.html#parse,Function Bio.ExPASy.Prosite.parse()=Bio.ExPASy.Prosite-module.html#parse,Method Bio.GenBank.FeatureParser.parse()=Bio.GenBank.FeatureParser-class.html#parse,Method Bio.GenBank.RecordParser.parse()=Bio.GenBank.RecordParser-class.html#parse,Method Bio.GenBank.Scanner.InsdcScanner.parse()=Bio.GenBank.Scanner.InsdcScanner-class.html#parse,Function Bio.Geo.parse()=Bio.Geo-module.html#parse,Function Bio.KEGG.Compound.parse()=Bio.KEGG.Compound-module.html#parse,Function Bio.KEGG.Enzyme.parse()=Bio.KEGG.Enzyme-module.html#parse,Method Bio.KEGG.KGML.KGML_parser.KGMLParser.parse()=Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html#parse,Function Bio.KEGG.KGML.KGML_parser.parse()=Bio.KEGG.KGML.KGML_parser-module.html#parse,Function Bio.KEGG.Map.parse()=Bio.KEGG.Map-module.html#parse,Function Bio.Medline.parse()=Bio.Medline-module.html#parse,Function Bio.Motif.parse()=Bio.Motif-module.html#parse,Method Bio.ParserSupport.AbstractParser.parse()=Bio.ParserSupport.AbstractParser-class.html#parse,Method Bio.Phylo.CDAOIO.Parser.parse()=Bio.Phylo.CDAOIO.Parser-class.html#parse,Function Bio.Phylo.CDAOIO.parse()=Bio.Phylo.CDAOIO-module.html#parse,Method Bio.Phylo.NeXMLIO.Parser.parse()=Bio.Phylo.NeXMLIO.Parser-class.html#parse,Function Bio.Phylo.NeXMLIO.parse()=Bio.Phylo.NeXMLIO-module.html#parse,Method Bio.Phylo.NewickIO.Parser.parse()=Bio.Phylo.NewickIO.Parser-class.html#parse,Function Bio.Phylo.NewickIO.parse()=Bio.Phylo.NewickIO-module.html#parse,Function Bio.Phylo.NexusIO.parse()=Bio.Phylo.NexusIO-module.html#parse,Method Bio.Phylo.PhyloXMLIO.Parser.parse()=Bio.Phylo.PhyloXMLIO.Parser-class.html#parse,Function Bio.Phylo.PhyloXMLIO.parse()=Bio.Phylo.PhyloXMLIO-module.html#parse,Function Bio.Phylo._io.parse()=Bio.Phylo._io-module.html#parse,Function Bio.SCOP.Cla.parse()=Bio.SCOP.Cla-module.html#parse,Function Bio.SCOP.Des.parse()=Bio.SCOP.Des-module.html#parse,Function Bio.SCOP.Dom.parse()=Bio.SCOP.Dom-module.html#parse,Function Bio.SCOP.Hie.parse()=Bio.SCOP.Hie-module.html#parse,Function Bio.SCOP.Raf.parse()=Bio.SCOP.Raf-module.html#parse,Function Bio.SearchIO.parse()=Bio.SearchIO-module.html#parse,Method Bio.SeqIO.UniprotIO.Parser.parse()=Bio.SeqIO.UniprotIO.Parser-class.html#parse,Function Bio.SeqIO.parse()=Bio.SeqIO-module.html#parse,Function Bio.Sequencing.Ace.parse()=Bio.Sequencing.Ace-module.html#parse,Function Bio.Sequencing.Phd.parse()=Bio.Sequencing.Phd-module.html#parse,Function Bio.SwissProt.KeyWList.parse()=Bio.SwissProt.KeyWList-module.html#parse,Function Bio.SwissProt.parse()=Bio.SwissProt-module.html#parse,Function Bio.UniGene.parse()=Bio.UniGene-module.html#parse,Function Bio.Wise.psw.parse()=Bio.Wise.psw-module.html#parse,Function Bio.motifs.parse()=Bio.motifs-module.html#parse"><a title="Bio.AlignIO.parse
Bio.Blast.NCBIStandalone.BlastErrorParser.parse
Bio.Blast.NCBIStandalone.BlastParser.parse
Bio.Blast.NCBIStandalone.PSIBlastParser.parse
Bio.Blast.NCBIXML.parse
Bio.Compass.parse
Bio.Emboss.Primer3.parse
Bio.Entrez.Parser.DataHandler.parse
Bio.Entrez.parse
Bio.ExPASy.Enzyme.parse
Bio.ExPASy.Prodoc.parse
Bio.ExPASy.Prosite.parse
Bio.GenBank.FeatureParser.parse
Bio.GenBank.RecordParser.parse
Bio.GenBank.Scanner.InsdcScanner.parse
Bio.Geo.parse
Bio.KEGG.Compound.parse
Bio.KEGG.Enzyme.parse
Bio.KEGG.KGML.KGML_parser.KGMLParser.parse
Bio.KEGG.KGML.KGML_parser.parse
Bio.KEGG.Map.parse
Bio.Medline.parse
Bio.Motif.parse
Bio.ParserSupport.AbstractParser.parse
Bio.Phylo.CDAOIO.Parser.parse
Bio.Phylo.CDAOIO.parse
Bio.Phylo.NeXMLIO.Parser.parse
Bio.Phylo.NeXMLIO.parse
Bio.Phylo.NewickIO.Parser.parse
Bio.Phylo.NewickIO.parse
Bio.Phylo.NexusIO.parse
Bio.Phylo.PhyloXMLIO.Parser.parse
Bio.Phylo.PhyloXMLIO.parse
Bio.Phylo._io.parse
Bio.SCOP.Cla.parse
Bio.SCOP.Des.parse
Bio.SCOP.Dom.parse
Bio.SCOP.Hie.parse
Bio.SCOP.Raf.parse
Bio.SearchIO.parse
Bio.SeqIO.UniprotIO.Parser.parse
Bio.SeqIO.parse
Bio.Sequencing.Ace.parse
Bio.Sequencing.Phd.parse
Bio.SwissProt.KeyWList.parse
Bio.SwissProt.parse
Bio.UniGene.parse
Bio.Wise.psw.parse
Bio.motifs.parse" class="py-name" href="#" onclick="return doclink('link-435', 'parse', 'link-435');">parse</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-string">"fasta"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1388"></a><tt class="py-lineno">1388</tt>  <tt class="py-line">    <tt class="py-name">filename</tt> <tt class="py-op">=</tt> <tt class="py-string">"../../Tests/Roche/E3MFGYR02_random_10_reads_no_trim.qual"</tt> </tt>
<a name="L1389"></a><tt class="py-lineno">1389</tt>  <tt class="py-line">    <tt class="py-name">qual_no_trim</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-436" class="py-name"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-436', 'SeqIO', 'link-1');">SeqIO</a></tt><tt class="py-op">.</tt><tt id="link-437" class="py-name"><a title="Bio.AlignIO.parse
Bio.Blast.NCBIStandalone.BlastErrorParser.parse
Bio.Blast.NCBIStandalone.BlastParser.parse
Bio.Blast.NCBIStandalone.PSIBlastParser.parse
Bio.Blast.NCBIXML.parse
Bio.Compass.parse
Bio.Emboss.Primer3.parse
Bio.Entrez.Parser.DataHandler.parse
Bio.Entrez.parse
Bio.ExPASy.Enzyme.parse
Bio.ExPASy.Prodoc.parse
Bio.ExPASy.Prosite.parse
Bio.GenBank.FeatureParser.parse
Bio.GenBank.RecordParser.parse
Bio.GenBank.Scanner.InsdcScanner.parse
Bio.Geo.parse
Bio.KEGG.Compound.parse
Bio.KEGG.Enzyme.parse
Bio.KEGG.KGML.KGML_parser.KGMLParser.parse
Bio.KEGG.KGML.KGML_parser.parse
Bio.KEGG.Map.parse
Bio.Medline.parse
Bio.Motif.parse
Bio.ParserSupport.AbstractParser.parse
Bio.Phylo.CDAOIO.Parser.parse
Bio.Phylo.CDAOIO.parse
Bio.Phylo.NeXMLIO.Parser.parse
Bio.Phylo.NeXMLIO.parse
Bio.Phylo.NewickIO.Parser.parse
Bio.Phylo.NewickIO.parse
Bio.Phylo.NexusIO.parse
Bio.Phylo.PhyloXMLIO.Parser.parse
Bio.Phylo.PhyloXMLIO.parse
Bio.Phylo._io.parse
Bio.SCOP.Cla.parse
Bio.SCOP.Des.parse
Bio.SCOP.Dom.parse
Bio.SCOP.Hie.parse
Bio.SCOP.Raf.parse
Bio.SearchIO.parse
Bio.SeqIO.UniprotIO.Parser.parse
Bio.SeqIO.parse
Bio.Sequencing.Ace.parse
Bio.Sequencing.Phd.parse
Bio.SwissProt.KeyWList.parse
Bio.SwissProt.parse
Bio.UniGene.parse
Bio.Wise.psw.parse
Bio.motifs.parse" class="py-name" href="#" onclick="return doclink('link-437', 'parse', 'link-435');">parse</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-string">"qual"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1390"></a><tt class="py-lineno">1390</tt>  <tt class="py-line"> </tt>
<a name="L1391"></a><tt class="py-lineno">1391</tt>  <tt class="py-line">    <tt class="py-name">filename</tt> <tt class="py-op">=</tt> <tt class="py-string">"../../Tests/Roche/E3MFGYR02_random_10_reads.fasta"</tt> </tt>
<a name="L1392"></a><tt class="py-lineno">1392</tt>  <tt class="py-line">    <tt class="py-name">fasta_trim</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-438" class="py-name"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-438', 'SeqIO', 'link-1');">SeqIO</a></tt><tt class="py-op">.</tt><tt id="link-439" class="py-name"><a title="Bio.AlignIO.parse
Bio.Blast.NCBIStandalone.BlastErrorParser.parse
Bio.Blast.NCBIStandalone.BlastParser.parse
Bio.Blast.NCBIStandalone.PSIBlastParser.parse
Bio.Blast.NCBIXML.parse
Bio.Compass.parse
Bio.Emboss.Primer3.parse
Bio.Entrez.Parser.DataHandler.parse
Bio.Entrez.parse
Bio.ExPASy.Enzyme.parse
Bio.ExPASy.Prodoc.parse
Bio.ExPASy.Prosite.parse
Bio.GenBank.FeatureParser.parse
Bio.GenBank.RecordParser.parse
Bio.GenBank.Scanner.InsdcScanner.parse
Bio.Geo.parse
Bio.KEGG.Compound.parse
Bio.KEGG.Enzyme.parse
Bio.KEGG.KGML.KGML_parser.KGMLParser.parse
Bio.KEGG.KGML.KGML_parser.parse
Bio.KEGG.Map.parse
Bio.Medline.parse
Bio.Motif.parse
Bio.ParserSupport.AbstractParser.parse
Bio.Phylo.CDAOIO.Parser.parse
Bio.Phylo.CDAOIO.parse
Bio.Phylo.NeXMLIO.Parser.parse
Bio.Phylo.NeXMLIO.parse
Bio.Phylo.NewickIO.Parser.parse
Bio.Phylo.NewickIO.parse
Bio.Phylo.NexusIO.parse
Bio.Phylo.PhyloXMLIO.Parser.parse
Bio.Phylo.PhyloXMLIO.parse
Bio.Phylo._io.parse
Bio.SCOP.Cla.parse
Bio.SCOP.Des.parse
Bio.SCOP.Dom.parse
Bio.SCOP.Hie.parse
Bio.SCOP.Raf.parse
Bio.SearchIO.parse
Bio.SeqIO.UniprotIO.Parser.parse
Bio.SeqIO.parse
Bio.Sequencing.Ace.parse
Bio.Sequencing.Phd.parse
Bio.SwissProt.KeyWList.parse
Bio.SwissProt.parse
Bio.UniGene.parse
Bio.Wise.psw.parse
Bio.motifs.parse" class="py-name" href="#" onclick="return doclink('link-439', 'parse', 'link-435');">parse</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-string">"fasta"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1393"></a><tt class="py-lineno">1393</tt>  <tt class="py-line">    <tt class="py-name">filename</tt> <tt class="py-op">=</tt> <tt class="py-string">"../../Tests/Roche/E3MFGYR02_random_10_reads.qual"</tt> </tt>
<a name="L1394"></a><tt class="py-lineno">1394</tt>  <tt class="py-line">    <tt class="py-name">qual_trim</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-440" class="py-name"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-440', 'SeqIO', 'link-1');">SeqIO</a></tt><tt class="py-op">.</tt><tt id="link-441" class="py-name"><a title="Bio.AlignIO.parse
Bio.Blast.NCBIStandalone.BlastErrorParser.parse
Bio.Blast.NCBIStandalone.BlastParser.parse
Bio.Blast.NCBIStandalone.PSIBlastParser.parse
Bio.Blast.NCBIXML.parse
Bio.Compass.parse
Bio.Emboss.Primer3.parse
Bio.Entrez.Parser.DataHandler.parse
Bio.Entrez.parse
Bio.ExPASy.Enzyme.parse
Bio.ExPASy.Prodoc.parse
Bio.ExPASy.Prosite.parse
Bio.GenBank.FeatureParser.parse
Bio.GenBank.RecordParser.parse
Bio.GenBank.Scanner.InsdcScanner.parse
Bio.Geo.parse
Bio.KEGG.Compound.parse
Bio.KEGG.Enzyme.parse
Bio.KEGG.KGML.KGML_parser.KGMLParser.parse
Bio.KEGG.KGML.KGML_parser.parse
Bio.KEGG.Map.parse
Bio.Medline.parse
Bio.Motif.parse
Bio.ParserSupport.AbstractParser.parse
Bio.Phylo.CDAOIO.Parser.parse
Bio.Phylo.CDAOIO.parse
Bio.Phylo.NeXMLIO.Parser.parse
Bio.Phylo.NeXMLIO.parse
Bio.Phylo.NewickIO.Parser.parse
Bio.Phylo.NewickIO.parse
Bio.Phylo.NexusIO.parse
Bio.Phylo.PhyloXMLIO.Parser.parse
Bio.Phylo.PhyloXMLIO.parse
Bio.Phylo._io.parse
Bio.SCOP.Cla.parse
Bio.SCOP.Des.parse
Bio.SCOP.Dom.parse
Bio.SCOP.Hie.parse
Bio.SCOP.Raf.parse
Bio.SearchIO.parse
Bio.SeqIO.UniprotIO.Parser.parse
Bio.SeqIO.parse
Bio.Sequencing.Ace.parse
Bio.Sequencing.Phd.parse
Bio.SwissProt.KeyWList.parse
Bio.SwissProt.parse
Bio.UniGene.parse
Bio.Wise.psw.parse
Bio.motifs.parse" class="py-name" href="#" onclick="return doclink('link-441', 'parse', 'link-435');">parse</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-string">"qual"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1395"></a><tt class="py-lineno">1395</tt>  <tt class="py-line"> </tt>
<a name="L1396"></a><tt class="py-lineno">1396</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">s</tt><tt class="py-op">,</tt> <tt class="py-name">sT</tt><tt class="py-op">,</tt> <tt class="py-name">f</tt><tt class="py-op">,</tt> <tt class="py-name">q</tt><tt class="py-op">,</tt> <tt class="py-name">fT</tt><tt class="py-op">,</tt> <tt class="py-name">qT</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">sff</tt><tt class="py-op">,</tt> <tt class="py-name">sff_trim</tt><tt class="py-op">,</tt> <tt class="py-name">fasta_no_trim</tt><tt class="py-op">,</tt> </tt>
<a name="L1397"></a><tt class="py-lineno">1397</tt>  <tt class="py-line">                                   <tt class="py-name">qual_no_trim</tt><tt class="py-op">,</tt> <tt class="py-name">fasta_trim</tt><tt class="py-op">,</tt> <tt class="py-name">qual_trim</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1398"></a><tt class="py-lineno">1398</tt>  <tt class="py-line">        <tt class="py-comment"># print("")</tt> </tt>
<a name="L1399"></a><tt class="py-lineno">1399</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-442" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-442', 'id', 'link-187');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1400"></a><tt class="py-lineno">1400</tt>  <tt class="py-line">        <tt class="py-comment"># print(s.seq)</tt> </tt>
<a name="L1401"></a><tt class="py-lineno">1401</tt>  <tt class="py-line">        <tt class="py-comment"># print(s.letter_annotations["phred_quality"])</tt> </tt>
<a name="L1402"></a><tt class="py-lineno">1402</tt>  <tt class="py-line"> </tt>
<a name="L1403"></a><tt class="py-lineno">1403</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-443" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-443', 'id', 'link-187');">id</a></tt> <tt class="py-op">==</tt> <tt class="py-name">f</tt><tt class="py-op">.</tt><tt id="link-444" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-444', 'id', 'link-187');">id</a></tt> <tt class="py-op">==</tt> <tt class="py-name">q</tt><tt class="py-op">.</tt><tt id="link-445" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-445', 'id', 'link-187');">id</a></tt> </tt>
<a name="L1404"></a><tt class="py-lineno">1404</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-446" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-446', 'seq', 'link-137');">seq</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">f</tt><tt class="py-op">.</tt><tt id="link-447" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-447', 'seq', 'link-137');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1405"></a><tt class="py-lineno">1405</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-448" class="py-name"><a title="Bio.SeqRecord.SeqRecord.letter_annotations" class="py-name" href="#" onclick="return doclink('link-448', 'letter_annotations', 'link-337');">letter_annotations</a></tt><tt class="py-op">[</tt> </tt>
<a name="L1406"></a><tt class="py-lineno">1406</tt>  <tt class="py-line">            <tt class="py-string">"phred_quality"</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">q</tt><tt class="py-op">.</tt><tt id="link-449" class="py-name"><a title="Bio.SeqRecord.SeqRecord.letter_annotations" class="py-name" href="#" onclick="return doclink('link-449', 'letter_annotations', 'link-337');">letter_annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"phred_quality"</tt><tt class="py-op">]</tt> </tt>
<a name="L1407"></a><tt class="py-lineno">1407</tt>  <tt class="py-line"> </tt>
<a name="L1408"></a><tt class="py-lineno">1408</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-450" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-450', 'id', 'link-187');">id</a></tt> <tt class="py-op">==</tt> <tt class="py-name">sT</tt><tt class="py-op">.</tt><tt id="link-451" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-451', 'id', 'link-187');">id</a></tt> <tt class="py-op">==</tt> <tt class="py-name">fT</tt><tt class="py-op">.</tt><tt id="link-452" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-452', 'id', 'link-187');">id</a></tt> <tt class="py-op">==</tt> <tt class="py-name">qT</tt><tt class="py-op">.</tt><tt id="link-453" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-453', 'id', 'link-187');">id</a></tt> </tt>
<a name="L1409"></a><tt class="py-lineno">1409</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">sT</tt><tt class="py-op">.</tt><tt id="link-454" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-454', 'seq', 'link-137');">seq</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">fT</tt><tt class="py-op">.</tt><tt id="link-455" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-455', 'seq', 'link-137');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1410"></a><tt class="py-lineno">1410</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">sT</tt><tt class="py-op">.</tt><tt id="link-456" class="py-name"><a title="Bio.SeqRecord.SeqRecord.letter_annotations" class="py-name" href="#" onclick="return doclink('link-456', 'letter_annotations', 'link-337');">letter_annotations</a></tt><tt class="py-op">[</tt> </tt>
<a name="L1411"></a><tt class="py-lineno">1411</tt>  <tt class="py-line">            <tt class="py-string">"phred_quality"</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">qT</tt><tt class="py-op">.</tt><tt id="link-457" class="py-name"><a title="Bio.SeqRecord.SeqRecord.letter_annotations" class="py-name" href="#" onclick="return doclink('link-457', 'letter_annotations', 'link-337');">letter_annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"phred_quality"</tt><tt class="py-op">]</tt> </tt>
<a name="L1412"></a><tt class="py-lineno">1412</tt>  <tt class="py-line"> </tt>
<a name="L1413"></a><tt class="py-lineno">1413</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Writing with a list of SeqRecords..."</tt><tt class="py-op">)</tt> </tt>
<a name="L1414"></a><tt class="py-lineno">1414</tt>  <tt class="py-line">    <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">BytesIO</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1415"></a><tt class="py-lineno">1415</tt>  <tt class="py-line">    <tt class="py-name">w</tt> <tt class="py-op">=</tt> <tt id="link-458" class="py-name" targets="Class Bio.SeqIO.SffIO.SffWriter=Bio.SeqIO.SffIO.SffWriter-class.html"><a title="Bio.SeqIO.SffIO.SffWriter" class="py-name" href="#" onclick="return doclink('link-458', 'SffWriter', 'link-458');">SffWriter</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-name">xml</tt><tt class="py-op">=</tt><tt class="py-name">metadata</tt><tt class="py-op">)</tt> </tt>
<a name="L1416"></a><tt class="py-lineno">1416</tt>  <tt class="py-line">    <tt class="py-name">w</tt><tt class="py-op">.</tt><tt id="link-459" class="py-name" targets="Method Bio.AlignIO.Interfaces.AlignmentWriter.write_file()=Bio.AlignIO.Interfaces.AlignmentWriter-class.html#write_file,Method Bio.AlignIO.Interfaces.SequentialAlignmentWriter.write_file()=Bio.AlignIO.Interfaces.SequentialAlignmentWriter-class.html#write_file,Method Bio.AlignIO.NexusIO.NexusWriter.write_file()=Bio.AlignIO.NexusIO.NexusWriter-class.html#write_file,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.write_file()=Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter-class.html#write_file,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.write_file()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter-class.html#write_file,Method Bio.SearchIO.BlatIO.BlatPslWriter.write_file()=Bio.SearchIO.BlatIO.BlatPslWriter-class.html#write_file,Method Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.write_file()=Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter-class.html#write_file,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.write_file()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter-class.html#write_file,Method Bio.SeqIO.Interfaces.SequenceWriter.write_file()=Bio.SeqIO.Interfaces.SequenceWriter-class.html#write_file,Method Bio.SeqIO.Interfaces.SequentialSequenceWriter.write_file()=Bio.SeqIO.Interfaces.SequentialSequenceWriter-class.html#write_file,Method Bio.SeqIO.SffIO.SffWriter.write_file()=Bio.SeqIO.SffIO.SffWriter-class.html#write_file"><a title="Bio.AlignIO.Interfaces.AlignmentWriter.write_file
Bio.AlignIO.Interfaces.SequentialAlignmentWriter.write_file
Bio.AlignIO.NexusIO.NexusWriter.write_file
Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.write_file
Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.write_file
Bio.SearchIO.BlatIO.BlatPslWriter.write_file
Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.write_file
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.write_file
Bio.SeqIO.Interfaces.SequenceWriter.write_file
Bio.SeqIO.Interfaces.SequentialSequenceWriter.write_file
Bio.SeqIO.SffIO.SffWriter.write_file" class="py-name" href="#" onclick="return doclink('link-459', 'write_file', 'link-459');">write_file</a></tt><tt class="py-op">(</tt><tt class="py-name">sff</tt><tt class="py-op">)</tt>  <tt class="py-comment"># list</tt> </tt>
<a name="L1417"></a><tt class="py-lineno">1417</tt>  <tt class="py-line">    <tt id="link-460" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-460', 'data', 'link-31');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt class="py-name">getvalue</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1418"></a><tt class="py-lineno">1418</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"And again with an iterator..."</tt><tt class="py-op">)</tt> </tt>
<a name="L1419"></a><tt class="py-lineno">1419</tt>  <tt class="py-line">    <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">BytesIO</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1420"></a><tt class="py-lineno">1420</tt>  <tt class="py-line">    <tt class="py-name">w</tt> <tt class="py-op">=</tt> <tt id="link-461" class="py-name"><a title="Bio.SeqIO.SffIO.SffWriter" class="py-name" href="#" onclick="return doclink('link-461', 'SffWriter', 'link-458');">SffWriter</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-name">xml</tt><tt class="py-op">=</tt><tt class="py-name">metadata</tt><tt class="py-op">)</tt> </tt>
<a name="L1421"></a><tt class="py-lineno">1421</tt>  <tt class="py-line">    <tt class="py-name">w</tt><tt class="py-op">.</tt><tt id="link-462" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentWriter.write_file
Bio.AlignIO.Interfaces.SequentialAlignmentWriter.write_file
Bio.AlignIO.NexusIO.NexusWriter.write_file
Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.write_file
Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.write_file
Bio.SearchIO.BlatIO.BlatPslWriter.write_file
Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.write_file
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.write_file
Bio.SeqIO.Interfaces.SequenceWriter.write_file
Bio.SeqIO.Interfaces.SequentialSequenceWriter.write_file
Bio.SeqIO.SffIO.SffWriter.write_file" class="py-name" href="#" onclick="return doclink('link-462', 'write_file', 'link-459');">write_file</a></tt><tt class="py-op">(</tt><tt class="py-name">iter</tt><tt class="py-op">(</tt><tt class="py-name">sff</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1422"></a><tt class="py-lineno">1422</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt id="link-463" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-463', 'data', 'link-31');">data</a></tt> <tt class="py-op">==</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt class="py-name">getvalue</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1423"></a><tt class="py-lineno">1423</tt>  <tt class="py-line">    <tt class="py-comment"># Check 100% identical to the original:</tt> </tt>
<a name="L1424"></a><tt class="py-lineno">1424</tt>  <tt class="py-line">    <tt class="py-name">filename</tt> <tt class="py-op">=</tt> <tt class="py-string">"../../Tests/Roche/E3MFGYR02_random_10_reads.sff"</tt> </tt>
<a name="L1425"></a><tt class="py-lineno">1425</tt>  <tt class="py-line">    <tt class="py-keyword">with</tt> <tt id="link-464" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-464', 'open', 'link-372');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-string">"rb"</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1426"></a><tt class="py-lineno">1426</tt>  <tt class="py-line">        <tt class="py-name">original</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-465" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-465', 'read', 'link-32');">read</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1427"></a><tt class="py-lineno">1427</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-466" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-466', 'data', 'link-31');">data</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">original</tt><tt class="py-op">)</tt> </tt>
<a name="L1428"></a><tt class="py-lineno">1428</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt id="link-467" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-467', 'data', 'link-31');">data</a></tt> <tt class="py-op">==</tt> <tt class="py-name">original</tt> </tt>
<a name="L1429"></a><tt class="py-lineno">1429</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt id="link-468" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-468', 'data', 'link-31');">data</a></tt> </tt>
<a name="L1430"></a><tt class="py-lineno">1430</tt>  <tt class="py-line"> </tt>
<a name="L1431"></a><tt class="py-lineno">1431</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"-"</tt> <tt class="py-op">*</tt> <tt class="py-number">50</tt><tt class="py-op">)</tt> </tt>
<a name="L1432"></a><tt class="py-lineno">1432</tt>  <tt class="py-line">    <tt class="py-name">filename</tt> <tt class="py-op">=</tt> <tt class="py-string">"../../Tests/Roche/greek.sff"</tt> </tt>
<a name="L1433"></a><tt class="py-lineno">1433</tt>  <tt class="py-line">    <tt class="py-keyword">with</tt> <tt id="link-469" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-469', 'open', 'link-372');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-string">"rb"</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1434"></a><tt class="py-lineno">1434</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt id="link-470" class="py-name"><a title="Bio.SeqIO.SffIO.SffIterator" class="py-name" href="#" onclick="return doclink('link-470', 'SffIterator', 'link-273');">SffIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1435"></a><tt class="py-lineno">1435</tt>  <tt class="py-line">            <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-471" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-471', 'id', 'link-187');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1436"></a><tt class="py-lineno">1436</tt>  <tt class="py-line">    <tt class="py-keyword">with</tt> <tt id="link-472" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-472', 'open', 'link-372');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-string">"rb"</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1437"></a><tt class="py-lineno">1437</tt>  <tt class="py-line">        <tt class="py-name">index1</tt> <tt class="py-op">=</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt id="link-473" class="py-name"><a title="Bio.SeqIO.SffIO._sff_read_roche_index" class="py-name" href="#" onclick="return doclink('link-473', '_sff_read_roche_index', 'link-375');">_sff_read_roche_index</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1438"></a><tt class="py-lineno">1438</tt>  <tt class="py-line">    <tt class="py-keyword">with</tt> <tt id="link-474" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-474', 'open', 'link-372');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-string">"rb"</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1439"></a><tt class="py-lineno">1439</tt>  <tt class="py-line">        <tt class="py-name">index2</tt> <tt class="py-op">=</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt id="link-475" class="py-name"><a title="Bio.SeqIO.SffIO._sff_do_slow_index" class="py-name" href="#" onclick="return doclink('link-475', '_sff_do_slow_index', 'link-377');">_sff_do_slow_index</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1440"></a><tt class="py-lineno">1440</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">index1</tt> <tt class="py-op">==</tt> <tt class="py-name">index2</tt> </tt>
<a name="L1441"></a><tt class="py-lineno">1441</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1442"></a><tt class="py-lineno">1442</tt>  <tt class="py-line">        <tt class="py-keyword">with</tt> <tt id="link-476" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-476', 'open', 'link-372');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-string">"rb"</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1443"></a><tt class="py-lineno">1443</tt>  <tt class="py-line">            <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt id="link-477" class="py-name"><a title="Bio.SeqIO.SffIO.ReadRocheXmlManifest" class="py-name" href="#" onclick="return doclink('link-477', 'ReadRocheXmlManifest', 'link-373');">ReadRocheXmlManifest</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1444"></a><tt class="py-lineno">1444</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">False</tt><tt class="py-op">,</tt> <tt class="py-string">"Should fail!"</tt> </tt>
<a name="L1445"></a><tt class="py-lineno">1445</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L1446"></a><tt class="py-lineno">1446</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
<a name="L1447"></a><tt class="py-lineno">1447</tt>  <tt class="py-line"> </tt>
<a name="L1448"></a><tt class="py-lineno">1448</tt>  <tt class="py-line">    <tt class="py-keyword">with</tt> <tt id="link-478" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-478', 'open', 'link-372');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-string">"rb"</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1449"></a><tt class="py-lineno">1449</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt id="link-479" class="py-name"><a title="Bio.SeqIO.SffIO.SffIterator" class="py-name" href="#" onclick="return doclink('link-479', 'SffIterator', 'link-273');">SffIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1450"></a><tt class="py-lineno">1450</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
<a name="L1451"></a><tt class="py-lineno">1451</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1452"></a><tt class="py-lineno">1452</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">record</tt> <tt class="py-keyword">in</tt> <tt id="link-480" class="py-name"><a title="Bio.SeqIO.SffIO.SffIterator" class="py-name" href="#" onclick="return doclink('link-480', 'SffIterator', 'link-273');">SffIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1453"></a><tt class="py-lineno">1453</tt>  <tt class="py-line">                <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-481" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-481', 'id', 'link-187');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1454"></a><tt class="py-lineno">1454</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">False</tt><tt class="py-op">,</tt> <tt class="py-string">"Should have failed"</tt> </tt>
<a name="L1455"></a><tt class="py-lineno">1455</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt> <tt class="py-keyword">as</tt> <tt class="py-name">err</tt><tt class="py-op">:</tt> </tt>
<a name="L1456"></a><tt class="py-lineno">1456</tt>  <tt class="py-line">            <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Checking what happens on re-reading a handle:"</tt><tt class="py-op">)</tt> </tt>
<a name="L1457"></a><tt class="py-lineno">1457</tt>  <tt class="py-line">            <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">err</tt><tt class="py-op">)</tt> </tt>
<a name="L1458"></a><tt class="py-lineno">1458</tt>  <tt class="py-line"> </tt>
<a name="L1459"></a><tt class="py-lineno">1459</tt>  <tt class="py-line">    <tt class="py-string">"""</tt> </tt>
<a name="L1460"></a><tt class="py-lineno">1460</tt>  <tt class="py-line"><tt class="py-string">    # Ugly code to make test files...</tt> </tt>
<a name="L1461"></a><tt class="py-lineno">1461</tt>  <tt class="py-line"><tt class="py-string">    index = ".diy1.00This is a fake index block (DIY = Do It Yourself), which is allowed under the SFF standard.\0"</tt> </tt>
<a name="L1462"></a><tt class="py-lineno">1462</tt>  <tt class="py-line"><tt class="py-string">    padding = len(index)%8</tt> </tt>
<a name="L1463"></a><tt class="py-lineno">1463</tt>  <tt class="py-line"><tt class="py-string">    if padding:</tt> </tt>
<a name="L1464"></a><tt class="py-lineno">1464</tt>  <tt class="py-line"><tt class="py-string">        padding = 8 - padding</tt> </tt>
<a name="L1465"></a><tt class="py-lineno">1465</tt>  <tt class="py-line"><tt class="py-string">    index += chr(0)*padding</tt> </tt>
<a name="L1466"></a><tt class="py-lineno">1466</tt>  <tt class="py-line"><tt class="py-string">    assert len(index)%8 == 0</tt> </tt>
<a name="L1467"></a><tt class="py-lineno">1467</tt>  <tt class="py-line"><tt class="py-string"></tt> </tt>
<a name="L1468"></a><tt class="py-lineno">1468</tt>  <tt class="py-line"><tt class="py-string">    # Ugly bit of code to make a fake index at start</tt> </tt>
<a name="L1469"></a><tt class="py-lineno">1469</tt>  <tt class="py-line"><tt class="py-string">    records = list(SffIterator(</tt> </tt>
<a name="L1470"></a><tt class="py-lineno">1470</tt>  <tt class="py-line"><tt class="py-string">        open("../../Tests/Roche/E3MFGYR02_random_10_reads.sff", "rb")))</tt> </tt>
<a name="L1471"></a><tt class="py-lineno">1471</tt>  <tt class="py-line"><tt class="py-string">    out_handle = open(</tt> </tt>
<a name="L1472"></a><tt class="py-lineno">1472</tt>  <tt class="py-line"><tt class="py-string">        "../../Tests/Roche/E3MFGYR02_alt_index_at_start.sff", "w")</tt> </tt>
<a name="L1473"></a><tt class="py-lineno">1473</tt>  <tt class="py-line"><tt class="py-string">    index = ".diy1.00This is a fake index block (DIY = Do It Yourself), which is allowed under the SFF standard.\0"</tt> </tt>
<a name="L1474"></a><tt class="py-lineno">1474</tt>  <tt class="py-line"><tt class="py-string">    padding = len(index)%8</tt> </tt>
<a name="L1475"></a><tt class="py-lineno">1475</tt>  <tt class="py-line"><tt class="py-string">    if padding:</tt> </tt>
<a name="L1476"></a><tt class="py-lineno">1476</tt>  <tt class="py-line"><tt class="py-string">        padding = 8 - padding</tt> </tt>
<a name="L1477"></a><tt class="py-lineno">1477</tt>  <tt class="py-line"><tt class="py-string">    index += chr(0)*padding</tt> </tt>
<a name="L1478"></a><tt class="py-lineno">1478</tt>  <tt class="py-line"><tt class="py-string">    w = SffWriter(out_handle, index=False, xml=None)</tt> </tt>
<a name="L1479"></a><tt class="py-lineno">1479</tt>  <tt class="py-line"><tt class="py-string">    # Fake the header...</tt> </tt>
<a name="L1480"></a><tt class="py-lineno">1480</tt>  <tt class="py-line"><tt class="py-string">    w._number_of_reads = len(records)</tt> </tt>
<a name="L1481"></a><tt class="py-lineno">1481</tt>  <tt class="py-line"><tt class="py-string">    w._index_start = 0</tt> </tt>
<a name="L1482"></a><tt class="py-lineno">1482</tt>  <tt class="py-line"><tt class="py-string">    w._index_length = 0</tt> </tt>
<a name="L1483"></a><tt class="py-lineno">1483</tt>  <tt class="py-line"><tt class="py-string">    w._key_sequence = records[0].annotations["flow_key"]</tt> </tt>
<a name="L1484"></a><tt class="py-lineno">1484</tt>  <tt class="py-line"><tt class="py-string">    w._flow_chars = records[0].annotations["flow_chars"]</tt> </tt>
<a name="L1485"></a><tt class="py-lineno">1485</tt>  <tt class="py-line"><tt class="py-string">    w._number_of_flows_per_read = len(w._flow_chars)</tt> </tt>
<a name="L1486"></a><tt class="py-lineno">1486</tt>  <tt class="py-line"><tt class="py-string">    w.write_header()</tt> </tt>
<a name="L1487"></a><tt class="py-lineno">1487</tt>  <tt class="py-line"><tt class="py-string">    w._index_start = out_handle.tell()</tt> </tt>
<a name="L1488"></a><tt class="py-lineno">1488</tt>  <tt class="py-line"><tt class="py-string">    w._index_length = len(index)</tt> </tt>
<a name="L1489"></a><tt class="py-lineno">1489</tt>  <tt class="py-line"><tt class="py-string">    out_handle.seek(0)</tt> </tt>
<a name="L1490"></a><tt class="py-lineno">1490</tt>  <tt class="py-line"><tt class="py-string">    w.write_header() # this time with index info</tt> </tt>
<a name="L1491"></a><tt class="py-lineno">1491</tt>  <tt class="py-line"><tt class="py-string">    w.handle.write(index)</tt> </tt>
<a name="L1492"></a><tt class="py-lineno">1492</tt>  <tt class="py-line"><tt class="py-string">    for record in records:</tt> </tt>
<a name="L1493"></a><tt class="py-lineno">1493</tt>  <tt class="py-line"><tt class="py-string">        w.write_record(record)</tt> </tt>
<a name="L1494"></a><tt class="py-lineno">1494</tt>  <tt class="py-line"><tt class="py-string">    out_handle.close()</tt> </tt>
<a name="L1495"></a><tt class="py-lineno">1495</tt>  <tt class="py-line"><tt class="py-string">    records2 = list(SffIterator(</tt> </tt>
<a name="L1496"></a><tt class="py-lineno">1496</tt>  <tt class="py-line"><tt class="py-string">        open("../../Tests/Roche/E3MFGYR02_alt_index_at_start.sff", "rb")))</tt> </tt>
<a name="L1497"></a><tt class="py-lineno">1497</tt>  <tt class="py-line"><tt class="py-string">    for old, new in zip(records, records2):</tt> </tt>
<a name="L1498"></a><tt class="py-lineno">1498</tt>  <tt class="py-line"><tt class="py-string">        assert str(old.seq)==str(new.seq)</tt> </tt>
<a name="L1499"></a><tt class="py-lineno">1499</tt>  <tt class="py-line"><tt class="py-string">    i = list(_sff_do_slow_index(</tt> </tt>
<a name="L1500"></a><tt class="py-lineno">1500</tt>  <tt class="py-line"><tt class="py-string">        open("../../Tests/Roche/E3MFGYR02_alt_index_at_start.sff", "rb")))</tt> </tt>
<a name="L1501"></a><tt class="py-lineno">1501</tt>  <tt class="py-line"><tt class="py-string"></tt> </tt>
<a name="L1502"></a><tt class="py-lineno">1502</tt>  <tt class="py-line"><tt class="py-string">    # Ugly bit of code to make a fake index in middle</tt> </tt>
<a name="L1503"></a><tt class="py-lineno">1503</tt>  <tt class="py-line"><tt class="py-string">    records = list(SffIterator(</tt> </tt>
<a name="L1504"></a><tt class="py-lineno">1504</tt>  <tt class="py-line"><tt class="py-string">        open("../../Tests/Roche/E3MFGYR02_random_10_reads.sff", "rb")))</tt> </tt>
<a name="L1505"></a><tt class="py-lineno">1505</tt>  <tt class="py-line"><tt class="py-string">    out_handle = open(</tt> </tt>
<a name="L1506"></a><tt class="py-lineno">1506</tt>  <tt class="py-line"><tt class="py-string">        "../../Tests/Roche/E3MFGYR02_alt_index_in_middle.sff", "w")</tt> </tt>
<a name="L1507"></a><tt class="py-lineno">1507</tt>  <tt class="py-line"><tt class="py-string">    index = ".diy1.00This is a fake index block (DIY = Do It Yourself), which is allowed under the SFF standard.\0"</tt> </tt>
<a name="L1508"></a><tt class="py-lineno">1508</tt>  <tt class="py-line"><tt class="py-string">    padding = len(index)%8</tt> </tt>
<a name="L1509"></a><tt class="py-lineno">1509</tt>  <tt class="py-line"><tt class="py-string">    if padding:</tt> </tt>
<a name="L1510"></a><tt class="py-lineno">1510</tt>  <tt class="py-line"><tt class="py-string">        padding = 8 - padding</tt> </tt>
<a name="L1511"></a><tt class="py-lineno">1511</tt>  <tt class="py-line"><tt class="py-string">    index += chr(0)*padding</tt> </tt>
<a name="L1512"></a><tt class="py-lineno">1512</tt>  <tt class="py-line"><tt class="py-string">    w = SffWriter(out_handle, index=False, xml=None)</tt> </tt>
<a name="L1513"></a><tt class="py-lineno">1513</tt>  <tt class="py-line"><tt class="py-string">    # Fake the header...</tt> </tt>
<a name="L1514"></a><tt class="py-lineno">1514</tt>  <tt class="py-line"><tt class="py-string">    w._number_of_reads = len(records)</tt> </tt>
<a name="L1515"></a><tt class="py-lineno">1515</tt>  <tt class="py-line"><tt class="py-string">    w._index_start = 0</tt> </tt>
<a name="L1516"></a><tt class="py-lineno">1516</tt>  <tt class="py-line"><tt class="py-string">    w._index_length = 0</tt> </tt>
<a name="L1517"></a><tt class="py-lineno">1517</tt>  <tt class="py-line"><tt class="py-string">    w._key_sequence = records[0].annotations["flow_key"]</tt> </tt>
<a name="L1518"></a><tt class="py-lineno">1518</tt>  <tt class="py-line"><tt class="py-string">    w._flow_chars = records[0].annotations["flow_chars"]</tt> </tt>
<a name="L1519"></a><tt class="py-lineno">1519</tt>  <tt class="py-line"><tt class="py-string">    w._number_of_flows_per_read = len(w._flow_chars)</tt> </tt>
<a name="L1520"></a><tt class="py-lineno">1520</tt>  <tt class="py-line"><tt class="py-string">    w.write_header()</tt> </tt>
<a name="L1521"></a><tt class="py-lineno">1521</tt>  <tt class="py-line"><tt class="py-string">    for record in records[:5]:</tt> </tt>
<a name="L1522"></a><tt class="py-lineno">1522</tt>  <tt class="py-line"><tt class="py-string">        w.write_record(record)</tt> </tt>
<a name="L1523"></a><tt class="py-lineno">1523</tt>  <tt class="py-line"><tt class="py-string">    w._index_start = out_handle.tell()</tt> </tt>
<a name="L1524"></a><tt class="py-lineno">1524</tt>  <tt class="py-line"><tt class="py-string">    w._index_length = len(index)</tt> </tt>
<a name="L1525"></a><tt class="py-lineno">1525</tt>  <tt class="py-line"><tt class="py-string">    w.handle.write(index)</tt> </tt>
<a name="L1526"></a><tt class="py-lineno">1526</tt>  <tt class="py-line"><tt class="py-string">    for record in records[5:]:</tt> </tt>
<a name="L1527"></a><tt class="py-lineno">1527</tt>  <tt class="py-line"><tt class="py-string">        w.write_record(record)</tt> </tt>
<a name="L1528"></a><tt class="py-lineno">1528</tt>  <tt class="py-line"><tt class="py-string">    out_handle.seek(0)</tt> </tt>
<a name="L1529"></a><tt class="py-lineno">1529</tt>  <tt class="py-line"><tt class="py-string">    w.write_header() # this time with index info</tt> </tt>
<a name="L1530"></a><tt class="py-lineno">1530</tt>  <tt class="py-line"><tt class="py-string">    out_handle.close()</tt> </tt>
<a name="L1531"></a><tt class="py-lineno">1531</tt>  <tt class="py-line"><tt class="py-string">    records2 = list(SffIterator(</tt> </tt>
<a name="L1532"></a><tt class="py-lineno">1532</tt>  <tt class="py-line"><tt class="py-string">        open("../../Tests/Roche/E3MFGYR02_alt_index_in_middle.sff", "rb")))</tt> </tt>
<a name="L1533"></a><tt class="py-lineno">1533</tt>  <tt class="py-line"><tt class="py-string">    for old, new in zip(records, records2):</tt> </tt>
<a name="L1534"></a><tt class="py-lineno">1534</tt>  <tt class="py-line"><tt class="py-string">        assert str(old.seq)==str(new.seq)</tt> </tt>
<a name="L1535"></a><tt class="py-lineno">1535</tt>  <tt class="py-line"><tt class="py-string">    j = list(_sff_do_slow_index(</tt> </tt>
<a name="L1536"></a><tt class="py-lineno">1536</tt>  <tt class="py-line"><tt class="py-string">        open("../../Tests/Roche/E3MFGYR02_alt_index_in_middle.sff", "rb")))</tt> </tt>
<a name="L1537"></a><tt class="py-lineno">1537</tt>  <tt class="py-line"><tt class="py-string"></tt> </tt>
<a name="L1538"></a><tt class="py-lineno">1538</tt>  <tt class="py-line"><tt class="py-string">    # Ugly bit of code to make a fake index at end</tt> </tt>
<a name="L1539"></a><tt class="py-lineno">1539</tt>  <tt class="py-line"><tt class="py-string">    records = list(SffIterator(</tt> </tt>
<a name="L1540"></a><tt class="py-lineno">1540</tt>  <tt class="py-line"><tt class="py-string">        open("../../Tests/Roche/E3MFGYR02_random_10_reads.sff", "rb")))</tt> </tt>
<a name="L1541"></a><tt class="py-lineno">1541</tt>  <tt class="py-line"><tt class="py-string">    with open("../../Tests/Roche/E3MFGYR02_alt_index_at_end.sff", "w") as out_handle:</tt> </tt>
<a name="L1542"></a><tt class="py-lineno">1542</tt>  <tt class="py-line"><tt class="py-string">        w = SffWriter(out_handle, index=False, xml=None)</tt> </tt>
<a name="L1543"></a><tt class="py-lineno">1543</tt>  <tt class="py-line"><tt class="py-string">        # Fake the header...</tt> </tt>
<a name="L1544"></a><tt class="py-lineno">1544</tt>  <tt class="py-line"><tt class="py-string">        w._number_of_reads = len(records)</tt> </tt>
<a name="L1545"></a><tt class="py-lineno">1545</tt>  <tt class="py-line"><tt class="py-string">        w._index_start = 0</tt> </tt>
<a name="L1546"></a><tt class="py-lineno">1546</tt>  <tt class="py-line"><tt class="py-string">        w._index_length = 0</tt> </tt>
<a name="L1547"></a><tt class="py-lineno">1547</tt>  <tt class="py-line"><tt class="py-string">        w._key_sequence = records[0].annotations["flow_key"]</tt> </tt>
<a name="L1548"></a><tt class="py-lineno">1548</tt>  <tt class="py-line"><tt class="py-string">        w._flow_chars = records[0].annotations["flow_chars"]</tt> </tt>
<a name="L1549"></a><tt class="py-lineno">1549</tt>  <tt class="py-line"><tt class="py-string">        w._number_of_flows_per_read = len(w._flow_chars)</tt> </tt>
<a name="L1550"></a><tt class="py-lineno">1550</tt>  <tt class="py-line"><tt class="py-string">        w.write_header()</tt> </tt>
<a name="L1551"></a><tt class="py-lineno">1551</tt>  <tt class="py-line"><tt class="py-string">        for record in records:</tt> </tt>
<a name="L1552"></a><tt class="py-lineno">1552</tt>  <tt class="py-line"><tt class="py-string">            w.write_record(record)</tt> </tt>
<a name="L1553"></a><tt class="py-lineno">1553</tt>  <tt class="py-line"><tt class="py-string">        w._index_start = out_handle.tell()</tt> </tt>
<a name="L1554"></a><tt class="py-lineno">1554</tt>  <tt class="py-line"><tt class="py-string">        w._index_length = len(index)</tt> </tt>
<a name="L1555"></a><tt class="py-lineno">1555</tt>  <tt class="py-line"><tt class="py-string">        out_handle.write(index)</tt> </tt>
<a name="L1556"></a><tt class="py-lineno">1556</tt>  <tt class="py-line"><tt class="py-string">        out_handle.seek(0)</tt> </tt>
<a name="L1557"></a><tt class="py-lineno">1557</tt>  <tt class="py-line"><tt class="py-string">        w.write_header() # this time with index info</tt> </tt>
<a name="L1558"></a><tt class="py-lineno">1558</tt>  <tt class="py-line"><tt class="py-string">    records2 = list(SffIterator(</tt> </tt>
<a name="L1559"></a><tt class="py-lineno">1559</tt>  <tt class="py-line"><tt class="py-string">        open("../../Tests/Roche/E3MFGYR02_alt_index_at_end.sff", "rb")))</tt> </tt>
<a name="L1560"></a><tt class="py-lineno">1560</tt>  <tt class="py-line"><tt class="py-string">    for old, new in zip(records, records2):</tt> </tt>
<a name="L1561"></a><tt class="py-lineno">1561</tt>  <tt class="py-line"><tt class="py-string">        assert str(old.seq)==str(new.seq)</tt> </tt>
<a name="L1562"></a><tt class="py-lineno">1562</tt>  <tt class="py-line"><tt class="py-string">    try:</tt> </tt>
<a name="L1563"></a><tt class="py-lineno">1563</tt>  <tt class="py-line"><tt class="py-string">        print(ReadRocheXmlManifest(</tt> </tt>
<a name="L1564"></a><tt class="py-lineno">1564</tt>  <tt class="py-line"><tt class="py-string">            open("../../Tests/Roche/E3MFGYR02_alt_index_at_end.sff", "rb")))</tt> </tt>
<a name="L1565"></a><tt class="py-lineno">1565</tt>  <tt class="py-line"><tt class="py-string">        assert False, "Should fail!"</tt> </tt>
<a name="L1566"></a><tt class="py-lineno">1566</tt>  <tt class="py-line"><tt class="py-string">    except ValueError:</tt> </tt>
<a name="L1567"></a><tt class="py-lineno">1567</tt>  <tt class="py-line"><tt class="py-string">        pass</tt> </tt>
<a name="L1568"></a><tt class="py-lineno">1568</tt>  <tt class="py-line"><tt class="py-string">    k = list(_sff_do_slow_index(</tt> </tt>
<a name="L1569"></a><tt class="py-lineno">1569</tt>  <tt class="py-line"><tt class="py-string">        open("../../Tests/Roche/E3MFGYR02_alt_index_at_end.sff", "rb")))</tt> </tt>
<a name="L1570"></a><tt class="py-lineno">1570</tt>  <tt class="py-line"><tt class="py-string">    """</tt> </tt>
<a name="L1571"></a><tt class="py-lineno">1571</tt>  <tt class="py-line"> </tt>
<a name="L1572"></a><tt class="py-lineno">1572</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Done"</tt><tt class="py-op">)</tt> </tt>
<a name="L1573"></a><tt class="py-lineno">1573</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:04:53 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
