<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.AlignIO.Interfaces</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.AlignIO-module.html">Package&nbsp;AlignIO</a> ::
        Module&nbsp;Interfaces
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.AlignIO.Interfaces-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.AlignIO.Interfaces-module.html">Module Bio.AlignIO.Interfaces</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2008-2013 by Peter Cock.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-docstring">"""AlignIO support module (not for general use).</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-docstring">Unless you are writing a new parser or writer for Bio.AlignIO, you should not</tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-docstring">use this module.  It provides base classes to try and simplify things.</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">__future__</tt> <tt class="py-keyword">import</tt> <tt class="py-name">print_function</tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">sys</tt>  <tt class="py-comment"># for checking if Python 2</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-1', 'Alphabet', 'link-1');">Alphabet</a></tt> <tt class="py-keyword">import</tt> <tt id="link-2" class="py-name" targets="Variable Bio.Alphabet.single_letter_alphabet=Bio.Alphabet-module.html#single_letter_alphabet"><a title="Bio.Alphabet.single_letter_alphabet" class="py-name" href="#" onclick="return doclink('link-2', 'single_letter_alphabet', 'link-2');">single_letter_alphabet</a></tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-name">__docformat__</tt> <tt class="py-op">=</tt> <tt class="py-string">"restructuredtext en"</tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"> </tt>
<a name="AlignmentIterator"></a><div id="AlignmentIterator-def"><a name="L20"></a><tt class="py-lineno"> 20</tt> <a class="py-toggle" href="#" id="AlignmentIterator-toggle" onclick="return toggle('AlignmentIterator');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.AlignIO.Interfaces.AlignmentIterator-class.html">AlignmentIterator</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignmentIterator-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="AlignmentIterator-expanded"><a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line">    <tt class="py-docstring">"""Base class for building MultipleSeqAlignment iterators.</tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-docstring">    You should write a next() method to return Aligment</tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-docstring">    objects.  You may wish to redefine the __init__</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-docstring">    method as well.</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line">    <tt class="py-comment"># TODO - Should the default be Gapped(single_letter_alphabet) instead?</tt> </tt>
<a name="AlignmentIterator.__init__"></a><div id="AlignmentIterator.__init__-def"><a name="L28"></a><tt class="py-lineno"> 28</tt> <a class="py-toggle" href="#" id="AlignmentIterator.__init__-toggle" onclick="return toggle('AlignmentIterator.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.Interfaces.AlignmentIterator-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">seq_count</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line">                 <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt id="link-3" class="py-name"><a title="Bio.Alphabet.single_letter_alphabet" class="py-name" href="#" onclick="return doclink('link-3', 'single_letter_alphabet', 'link-2');">single_letter_alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignmentIterator.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="AlignmentIterator.__init__-expanded"><a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create an AlignmentIterator object.</tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"><tt class="py-docstring">            - handle   - input file</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-docstring">            - count    - optional, expected number of records per alignment</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-docstring">              Recommend for fasta file format.</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-docstring">            - alphabet - optional, e.g. Bio.Alphabet.generic_protein</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"><tt class="py-docstring">        Note when subclassing:</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"><tt class="py-docstring">            - there should be a single non-optional argument, the handle,</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"><tt class="py-docstring">              and optional count and alphabet IN THAT ORDER.</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"><tt class="py-docstring">            - you do not have to require an alphabet (?).</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"><tt class="py-docstring">            - you can add additional optional arguments.</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">records_per_alignment</tt> <tt class="py-op">=</tt> <tt class="py-name">seq_count</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-4" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.alphabet=Bio.SearchIO._model.hsp.HSP-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSPFragment.alphabet=Bio.SearchIO._model.hsp.HSPFragment-class.html#alphabet"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-4', 'alphabet', 'link-4');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-5" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-5', 'alphabet', 'link-4');">alphabet</a></tt> </tt>
</div><a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line">        <tt class="py-comment">#####################################################</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line">        <tt class="py-comment"># You may want to subclass this, for example        #</tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line">        <tt class="py-comment"># to read through the file to find the first record,#</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line">        <tt class="py-comment"># or if additional arguments are required.          #</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line">        <tt class="py-comment">#####################################################</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line"> </tt>
<a name="AlignmentIterator.__next__"></a><div id="AlignmentIterator.__next__-def"><a name="L53"></a><tt class="py-lineno"> 53</tt> <a class="py-toggle" href="#" id="AlignmentIterator.__next__-toggle" onclick="return toggle('AlignmentIterator.__next__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.Interfaces.AlignmentIterator-class.html#__next__">__next__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignmentIterator.__next__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="AlignmentIterator.__next__-expanded"><a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the next alignment in the file.</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line"><tt class="py-docstring">        This method should be replaced by any derived class to do something</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"><tt class="py-docstring">        useful."""</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">NotImplementedError</tt><tt class="py-op">(</tt><tt class="py-string">"This object should be subclassed"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line">        <tt class="py-comment">#####################################################</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line">        <tt class="py-comment"># You SHOULD subclass this, to split the file up    #</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line">        <tt class="py-comment"># into your individual alignments and convert these #</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line">        <tt class="py-comment"># into MultipleSeqAlignment objects.                #</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line">        <tt class="py-comment">#####################################################</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line"> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">version_info</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">3</tt><tt class="py-op">:</tt> </tt>
<a name="AlignmentIterator.next"></a><div id="AlignmentIterator.next-def"><a name="L66"></a><tt class="py-lineno"> 66</tt> <a class="py-toggle" href="#" id="AlignmentIterator.next-toggle" onclick="return toggle('AlignmentIterator.next');">-</a><tt class="py-line">        <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.Interfaces.AlignmentIterator-class.html#next">next</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignmentIterator.next-collapsed" style="display:none;" pad="+++" indent="++++++++++++++++"></div><div id="AlignmentIterator.next-expanded"><a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line">            <tt class="py-docstring">"""Python 2 style alias for Python 3 style __next__ method."""</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-6" class="py-name" targets="Method Bio.AlignIO.ClustalIO.ClustalIterator.__next__()=Bio.AlignIO.ClustalIO.ClustalIterator-class.html#__next__,Method Bio.AlignIO.EmbossIO.EmbossIterator.__next__()=Bio.AlignIO.EmbossIO.EmbossIterator-class.html#__next__,Method Bio.AlignIO.Interfaces.AlignmentIterator.__next__()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#__next__,Method Bio.AlignIO.PhylipIO.PhylipIterator.__next__()=Bio.AlignIO.PhylipIO.PhylipIterator-class.html#__next__,Method Bio.AlignIO.PhylipIO.SequentialPhylipIterator.__next__()=Bio.AlignIO.PhylipIO.SequentialPhylipIterator-class.html#__next__,Method Bio.AlignIO.StockholmIO.StockholmIterator.__next__()=Bio.AlignIO.StockholmIO.StockholmIterator-class.html#__next__,Method Bio.Blast.NCBIStandalone.Iterator.__next__()=Bio.Blast.NCBIStandalone.Iterator-class.html#__next__,Method Bio.Blast.ParseBlastTable.BlastTableReader.__next__()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#__next__,Method Bio.DocSQL.IterationCursor.__next__()=Bio.DocSQL.IterationCursor-class.html#__next__,Method Bio.File.UndoHandle.__next__()=Bio.File.UndoHandle-class.html#__next__,Method Bio.GenBank.Iterator.__next__()=Bio.GenBank.Iterator-class.html#__next__,Method Bio.Nexus.Nexus.CharBuffer.__next__()=Bio.Nexus.Nexus.CharBuffer-class.html#__next__,Method Bio.PopGen.GenePop.Controller._FileIterator.__next__()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#__next__,Method Bio.SeqIO.Interfaces.SequenceIterator.__next__()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#__next__,Method Bio.bgzf.BgzfReader.__next__()=Bio.bgzf.BgzfReader-class.html#__next__"><a title="Bio.AlignIO.ClustalIO.ClustalIterator.__next__
Bio.AlignIO.EmbossIO.EmbossIterator.__next__
Bio.AlignIO.Interfaces.AlignmentIterator.__next__
Bio.AlignIO.PhylipIO.PhylipIterator.__next__
Bio.AlignIO.PhylipIO.SequentialPhylipIterator.__next__
Bio.AlignIO.StockholmIO.StockholmIterator.__next__
Bio.Blast.NCBIStandalone.Iterator.__next__
Bio.Blast.ParseBlastTable.BlastTableReader.__next__
Bio.DocSQL.IterationCursor.__next__
Bio.File.UndoHandle.__next__
Bio.GenBank.Iterator.__next__
Bio.Nexus.Nexus.CharBuffer.__next__
Bio.PopGen.GenePop.Controller._FileIterator.__next__
Bio.SeqIO.Interfaces.SequenceIterator.__next__
Bio.bgzf.BgzfReader.__next__" class="py-name" href="#" onclick="return doclink('link-6', '__next__', 'link-6');">__next__</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line"> </tt>
<a name="AlignmentIterator.__iter__"></a><div id="AlignmentIterator.__iter__-def"><a name="L70"></a><tt class="py-lineno"> 70</tt> <a class="py-toggle" href="#" id="AlignmentIterator.__iter__-toggle" onclick="return toggle('AlignmentIterator.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.Interfaces.AlignmentIterator-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignmentIterator.__iter__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="AlignmentIterator.__iter__-expanded"><a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line">        <tt class="py-docstring">"""Iterate over the entries as MultipleSeqAlignment objects.</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line"><tt class="py-docstring">        Example usage for (concatenated) PHYLIP files::</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line"><tt class="py-docstring">            with open("many.phy","r") as myFile:</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line"><tt class="py-docstring">                for alignment in PhylipIterator(myFile):</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line"><tt class="py-docstring">                    print "New alignment:"</tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line"><tt class="py-docstring">                    for record in alignment:</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line"><tt class="py-docstring">                        print record.id</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line"><tt class="py-docstring">                        print record.seq</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">iter</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-7" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator.__next__
Bio.AlignIO.EmbossIO.EmbossIterator.__next__
Bio.AlignIO.Interfaces.AlignmentIterator.__next__
Bio.AlignIO.PhylipIO.PhylipIterator.__next__
Bio.AlignIO.PhylipIO.SequentialPhylipIterator.__next__
Bio.AlignIO.StockholmIO.StockholmIterator.__next__
Bio.Blast.NCBIStandalone.Iterator.__next__
Bio.Blast.ParseBlastTable.BlastTableReader.__next__
Bio.DocSQL.IterationCursor.__next__
Bio.File.UndoHandle.__next__
Bio.GenBank.Iterator.__next__
Bio.Nexus.Nexus.CharBuffer.__next__
Bio.PopGen.GenePop.Controller._FileIterator.__next__
Bio.SeqIO.Interfaces.SequenceIterator.__next__
Bio.bgzf.BgzfReader.__next__" class="py-name" href="#" onclick="return doclink('link-7', '__next__', 'link-6');">__next__</a></tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line"> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line"> </tt>
<a name="AlignmentWriter"></a><div id="AlignmentWriter-def"><a name="L85"></a><tt class="py-lineno"> 85</tt> <a class="py-toggle" href="#" id="AlignmentWriter-toggle" onclick="return toggle('AlignmentWriter');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.AlignIO.Interfaces.AlignmentWriter-class.html">AlignmentWriter</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignmentWriter-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="AlignmentWriter-expanded"><a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line">    <tt class="py-docstring">"""Base class for building MultipleSeqAlignment writers.</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line"><tt class="py-docstring">    You should write a write_alignment() method.</tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line"><tt class="py-docstring">    You may wish to redefine the __init__ method as well"""</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line"> </tt>
<a name="AlignmentWriter.__init__"></a><div id="AlignmentWriter.__init__-def"><a name="L91"></a><tt class="py-lineno"> 91</tt> <a class="py-toggle" href="#" id="AlignmentWriter.__init__-toggle" onclick="return toggle('AlignmentWriter.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.Interfaces.AlignmentWriter-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignmentWriter.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="AlignmentWriter.__init__-expanded"><a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt> </tt>
</div><a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line"> </tt>
<a name="AlignmentWriter.write_file"></a><div id="AlignmentWriter.write_file-def"><a name="L94"></a><tt class="py-lineno"> 94</tt> <a class="py-toggle" href="#" id="AlignmentWriter.write_file-toggle" onclick="return toggle('AlignmentWriter.write_file');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.Interfaces.AlignmentWriter-class.html#write_file">write_file</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">alignments</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignmentWriter.write_file-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="AlignmentWriter.write_file-expanded"><a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line">        <tt class="py-docstring">"""Use this to write an entire file containing the given alignments.</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line"><tt class="py-docstring">        alignments - A list or iterator returning MultipleSeqAlignment objects</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line"><tt class="py-docstring">        In general, this method can only be called once per file.</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line"><tt class="py-docstring">        This method should be replaced by any derived class to do something</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line"><tt class="py-docstring">        useful.  It should return the number of alignments"""</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">NotImplementedError</tt><tt class="py-op">(</tt><tt class="py-string">"This object should be subclassed"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line">        <tt class="py-comment">#####################################################</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">        <tt class="py-comment"># You SHOULD subclass this, to write the alignment  #</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">        <tt class="py-comment"># objecta to the file handle                        #</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line">        <tt class="py-comment">#####################################################</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line"> </tt>
<a name="AlignmentWriter.clean"></a><div id="AlignmentWriter.clean-def"><a name="L109"></a><tt class="py-lineno">109</tt> <a class="py-toggle" href="#" id="AlignmentWriter.clean-toggle" onclick="return toggle('AlignmentWriter.clean');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.Interfaces.AlignmentWriter-class.html#clean">clean</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignmentWriter.clean-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="AlignmentWriter.clean-expanded"><a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">        <tt class="py-docstring">"""Use this to avoid getting newlines in the output."""</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">text</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"\n"</tt><tt class="py-op">,</tt> <tt class="py-string">" "</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"\r"</tt><tt class="py-op">,</tt> <tt class="py-string">" "</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"  "</tt><tt class="py-op">,</tt> <tt class="py-string">" "</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line"> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line"> </tt>
<a name="SequentialAlignmentWriter"></a><div id="SequentialAlignmentWriter-def"><a name="L114"></a><tt class="py-lineno">114</tt> <a class="py-toggle" href="#" id="SequentialAlignmentWriter-toggle" onclick="return toggle('SequentialAlignmentWriter');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.AlignIO.Interfaces.SequentialAlignmentWriter-class.html">SequentialAlignmentWriter</a><tt class="py-op">(</tt><tt class="py-base-class">AlignmentWriter</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequentialAlignmentWriter-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="SequentialAlignmentWriter-expanded"><a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">    <tt class="py-docstring">"""Base class for building MultipleSeqAlignment writers.</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line"><tt class="py-docstring">    This assumes each alignment can be simply appended to the file.</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line"><tt class="py-docstring">    You should write a write_alignment() method.</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line"><tt class="py-docstring">    You may wish to redefine the __init__ method as well"""</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line"> </tt>
<a name="SequentialAlignmentWriter.__init__"></a><div id="SequentialAlignmentWriter.__init__-def"><a name="L121"></a><tt class="py-lineno">121</tt> <a class="py-toggle" href="#" id="SequentialAlignmentWriter.__init__-toggle" onclick="return toggle('SequentialAlignmentWriter.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.Interfaces.SequentialAlignmentWriter-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequentialAlignmentWriter.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SequentialAlignmentWriter.__init__-expanded"><a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt> </tt>
</div><a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line"> </tt>
<a name="SequentialAlignmentWriter.write_file"></a><div id="SequentialAlignmentWriter.write_file-def"><a name="L124"></a><tt class="py-lineno">124</tt> <a class="py-toggle" href="#" id="SequentialAlignmentWriter.write_file-toggle" onclick="return toggle('SequentialAlignmentWriter.write_file');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.Interfaces.SequentialAlignmentWriter-class.html#write_file">write_file</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">alignments</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequentialAlignmentWriter.write_file-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SequentialAlignmentWriter.write_file-expanded"><a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line">        <tt class="py-docstring">"""Use this to write an entire file containing the given alignments.</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line"><tt class="py-docstring">        alignments - A list or iterator returning MultipleSeqAlignment objects</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line"><tt class="py-docstring">        In general, this method can only be called once per file."""</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-8" class="py-name" targets="Method Bio.AlignIO.EmbossIO.EmbossWriter.write_header()=Bio.AlignIO.EmbossIO.EmbossWriter-class.html#write_header,Method Bio.AlignIO.Interfaces.SequentialAlignmentWriter.write_header()=Bio.AlignIO.Interfaces.SequentialAlignmentWriter-class.html#write_header,Method Bio.NMR.xpktools.Peaklist.write_header()=Bio.NMR.xpktools.Peaklist-class.html#write_header,Method Bio.SeqIO.Interfaces.SequentialSequenceWriter.write_header()=Bio.SeqIO.Interfaces.SequentialSequenceWriter-class.html#write_header,Method Bio.SeqIO.SeqXmlIO.SeqXmlWriter.write_header()=Bio.SeqIO.SeqXmlIO.SeqXmlWriter-class.html#write_header,Method Bio.SeqIO.SffIO.SffWriter.write_header()=Bio.SeqIO.SffIO.SffWriter-class.html#write_header"><a title="Bio.AlignIO.EmbossIO.EmbossWriter.write_header
Bio.AlignIO.Interfaces.SequentialAlignmentWriter.write_header
Bio.NMR.xpktools.Peaklist.write_header
Bio.SeqIO.Interfaces.SequentialSequenceWriter.write_header
Bio.SeqIO.SeqXmlIO.SeqXmlWriter.write_header
Bio.SeqIO.SffIO.SffWriter.write_header" class="py-name" href="#" onclick="return doclink('link-8', 'write_header', 'link-8');">write_header</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line">        <tt id="link-9" class="py-name" targets="Method Bio.Crystal.Chain.count()=Bio.Crystal.Chain-class.html#count,Method Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count()=Bio.NeuralNetwork.Gene.Pattern.PatternRepository-class.html#count,Method Bio.Seq.MutableSeq.count()=Bio.Seq.MutableSeq-class.html#count,Method Bio.Seq.Seq.count()=Bio.Seq.Seq-class.html#count,Method Bio.Seq.UnknownSeq.count()=Bio.Seq.UnknownSeq-class.html#count,Method Bio.motifs.Instances.count()=Bio.motifs.Instances-class.html#count"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-9', 'count', 'link-9');">count</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">alignment</tt> <tt class="py-keyword">in</tt> <tt class="py-name">alignments</tt><tt class="py-op">:</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-10" class="py-name" targets="Method Bio.AlignIO.ClustalIO.ClustalWriter.write_alignment()=Bio.AlignIO.ClustalIO.ClustalWriter-class.html#write_alignment,Method Bio.AlignIO.EmbossIO.EmbossWriter.write_alignment()=Bio.AlignIO.EmbossIO.EmbossWriter-class.html#write_alignment,Method Bio.AlignIO.Interfaces.SequentialAlignmentWriter.write_alignment()=Bio.AlignIO.Interfaces.SequentialAlignmentWriter-class.html#write_alignment,Method Bio.AlignIO.NexusIO.NexusWriter.write_alignment()=Bio.AlignIO.NexusIO.NexusWriter-class.html#write_alignment,Method Bio.AlignIO.PhylipIO.PhylipWriter.write_alignment()=Bio.AlignIO.PhylipIO.PhylipWriter-class.html#write_alignment,Method Bio.AlignIO.PhylipIO.RelaxedPhylipWriter.write_alignment()=Bio.AlignIO.PhylipIO.RelaxedPhylipWriter-class.html#write_alignment,Method Bio.AlignIO.PhylipIO.SequentialPhylipWriter.write_alignment()=Bio.AlignIO.PhylipIO.SequentialPhylipWriter-class.html#write_alignment,Method Bio.AlignIO.StockholmIO.StockholmWriter.write_alignment()=Bio.AlignIO.StockholmIO.StockholmWriter-class.html#write_alignment"><a title="Bio.AlignIO.ClustalIO.ClustalWriter.write_alignment
Bio.AlignIO.EmbossIO.EmbossWriter.write_alignment
Bio.AlignIO.Interfaces.SequentialAlignmentWriter.write_alignment
Bio.AlignIO.NexusIO.NexusWriter.write_alignment
Bio.AlignIO.PhylipIO.PhylipWriter.write_alignment
Bio.AlignIO.PhylipIO.RelaxedPhylipWriter.write_alignment
Bio.AlignIO.PhylipIO.SequentialPhylipWriter.write_alignment
Bio.AlignIO.StockholmIO.StockholmWriter.write_alignment" class="py-name" href="#" onclick="return doclink('link-10', 'write_alignment', 'link-10');">write_alignment</a></tt><tt class="py-op">(</tt><tt class="py-name">alignment</tt><tt class="py-op">)</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line">            <tt id="link-11" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-11', 'count', 'link-9');">count</a></tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-12" class="py-name" targets="Method Bio.AlignIO.EmbossIO.EmbossWriter.write_footer()=Bio.AlignIO.EmbossIO.EmbossWriter-class.html#write_footer,Method Bio.AlignIO.Interfaces.SequentialAlignmentWriter.write_footer()=Bio.AlignIO.Interfaces.SequentialAlignmentWriter-class.html#write_footer,Method Bio.SeqIO.Interfaces.SequentialSequenceWriter.write_footer()=Bio.SeqIO.Interfaces.SequentialSequenceWriter-class.html#write_footer,Method Bio.SeqIO.SeqXmlIO.SeqXmlWriter.write_footer()=Bio.SeqIO.SeqXmlIO.SeqXmlWriter-class.html#write_footer"><a title="Bio.AlignIO.EmbossIO.EmbossWriter.write_footer
Bio.AlignIO.Interfaces.SequentialAlignmentWriter.write_footer
Bio.SeqIO.Interfaces.SequentialSequenceWriter.write_footer
Bio.SeqIO.SeqXmlIO.SeqXmlWriter.write_footer" class="py-name" href="#" onclick="return doclink('link-12', 'write_footer', 'link-12');">write_footer</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-13" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-13', 'count', 'link-9');">count</a></tt> </tt>
</div><a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line"> </tt>
<a name="SequentialAlignmentWriter.write_header"></a><div id="SequentialAlignmentWriter.write_header-def"><a name="L138"></a><tt class="py-lineno">138</tt> <a class="py-toggle" href="#" id="SequentialAlignmentWriter.write_header-toggle" onclick="return toggle('SequentialAlignmentWriter.write_header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.Interfaces.SequentialAlignmentWriter-class.html#write_header">write_header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequentialAlignmentWriter.write_header-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SequentialAlignmentWriter.write_header-expanded"><a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line">        <tt class="py-docstring">"""Use this to write any header.</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line"><tt class="py-docstring">        This method should be replaced by any derived class to do something</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line"><tt class="py-docstring">        useful."""</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line"> </tt>
<a name="SequentialAlignmentWriter.write_footer"></a><div id="SequentialAlignmentWriter.write_footer-def"><a name="L145"></a><tt class="py-lineno">145</tt> <a class="py-toggle" href="#" id="SequentialAlignmentWriter.write_footer-toggle" onclick="return toggle('SequentialAlignmentWriter.write_footer');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.Interfaces.SequentialAlignmentWriter-class.html#write_footer">write_footer</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequentialAlignmentWriter.write_footer-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SequentialAlignmentWriter.write_footer-expanded"><a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">        <tt class="py-docstring">"""Use this to write any footer.</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line"><tt class="py-docstring">        This method should be replaced by any derived class to do something</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line"><tt class="py-docstring">        useful."""</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line"> </tt>
<a name="SequentialAlignmentWriter.write_alignment"></a><div id="SequentialAlignmentWriter.write_alignment-def"><a name="L152"></a><tt class="py-lineno">152</tt> <a class="py-toggle" href="#" id="SequentialAlignmentWriter.write_alignment-toggle" onclick="return toggle('SequentialAlignmentWriter.write_alignment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.AlignIO.Interfaces.SequentialAlignmentWriter-class.html#write_alignment">write_alignment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">alignment</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequentialAlignmentWriter.write_alignment-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SequentialAlignmentWriter.write_alignment-expanded"><a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">        <tt class="py-docstring">"""Use this to write a single alignment.</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line"><tt class="py-docstring">        This method should be replaced by any derived class to do something</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line"><tt class="py-docstring">        useful."""</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">NotImplementedError</tt><tt class="py-op">(</tt><tt class="py-string">"This object should be subclassed"</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">        <tt class="py-comment">#####################################################</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">        <tt class="py-comment"># You SHOULD subclass this, to write the alignment  #</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">        <tt class="py-comment"># objecta to the file handle                        #</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">        <tt class="py-comment">#####################################################</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:05:06 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
