<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.motifs.mast</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.motifs-module.html">Package&nbsp;motifs</a> ::
        Module&nbsp;mast
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.motifs.mast-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.motifs.mast-module.html">Module Bio.motifs.mast</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2008 by Bartek Wilczynski.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># Adapted from Bio.MEME.Parser by Jason A. Hackney.  All rights reserved.</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-docstring">"""Module for the support of Motif Alignment and Search Tool (MAST)."""</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">__future__</tt> <tt class="py-keyword">import</tt> <tt class="py-name">print_function</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">xml</tt><tt class="py-op">.</tt><tt class="py-name">etree</tt><tt class="py-op">.</tt><tt class="py-name">ElementTree</tt> <tt class="py-keyword">as</tt> <tt class="py-name">ET</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio.motifs=Bio.motifs-module.html"><a title="Bio.motifs" class="py-name" href="#" onclick="return doclink('link-1', 'motifs', 'link-1');">motifs</a></tt> <tt class="py-keyword">import</tt> <tt id="link-2" class="py-name" targets="Module Bio.motifs.meme=Bio.motifs.meme-module.html"><a title="Bio.motifs.meme" class="py-name" href="#" onclick="return doclink('link-2', 'meme', 'link-2');">meme</a></tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"> </tt>
<a name="Record"></a><div id="Record-def"><a name="L14"></a><tt class="py-lineno"> 14</tt> <a class="py-toggle" href="#" id="Record-toggle" onclick="return toggle('Record');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.motifs.mast.Record-class.html">Record</a><tt class="py-op">(</tt><tt class="py-base-class">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Record-expanded"><a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line">    <tt class="py-docstring">"""The class for holding the results from a MAST run.</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-docstring">    A mast.Record holds data about matches between motifs and sequences.</tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-docstring">    The motifs held by the Record are objects of the class meme.Motif.</tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-docstring">    The mast.Record class inherits from list, so you can access individual</tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-docstring">    motifs in the record by their index. Alternatively, you can find a motif</tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-docstring">    by its name:</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import motifs</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; with open("motifs/mast.crp0.de.oops.txt.xml") as f:</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-docstring">    ...     record = motifs.parse(f, 'MAST')</tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; motif = record[0]</tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(motif.name)</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"><tt class="py-docstring">    1</tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; motif = record['1']</tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(motif.name)</tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"><tt class="py-docstring">    1</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"> </tt>
<a name="Record.__init__"></a><div id="Record.__init__-def"><a name="L35"></a><tt class="py-lineno"> 35</tt> <a class="py-toggle" href="#" id="Record.__init__-toggle" onclick="return toggle('Record.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.mast.Record-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record.__init__-expanded"><a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the class."""</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">sequences</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-3" class="py-name" targets="Function Bio.Cluster._cluster.version()=Bio.Cluster._cluster-module.html#version,Method Bio.GenBank._FeatureConsumer.version()=Bio.GenBank._FeatureConsumer-class.html#version,Method Bio.GenBank._RecordConsumer.version()=Bio.GenBank._RecordConsumer-class.html#version,Method Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.version()=Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer-class.html#version,Variable Bio.motifs.jaspar.Motif.version=Bio.motifs.jaspar.Motif-class.html#version"><a title="Bio.Cluster._cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-3', 'version', 'link-3');">version</a></tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-4" class="py-name" targets="Method Bio.SearchIO._legacy.NCBIStandalone._DatabaseReportConsumer.database()=Bio.SearchIO._legacy.NCBIStandalone._DatabaseReportConsumer-class.html#database"><a title="Bio.SearchIO._legacy.NCBIStandalone._DatabaseReportConsumer.database" class="py-name" href="#" onclick="return doclink('link-4', 'database', 'link-4');">database</a></tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">diagrams</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-5" class="py-name" targets="Variable Bio.Align.substitution_matrices.Array.alphabet=Bio.Align.substitution_matrices.Array-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSP.alphabet=Bio.SearchIO._model.hsp.HSP-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSPFragment.alphabet=Bio.SearchIO._model.hsp.HSPFragment-class.html#alphabet"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-5', 'alphabet', 'link-5');">alphabet</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">strand_handling</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
</div><a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line"> </tt>
<a name="Record.__getitem__"></a><div id="Record.__getitem__-def"><a name="L44"></a><tt class="py-lineno"> 44</tt> <a class="py-toggle" href="#" id="Record.__getitem__-toggle" onclick="return toggle('Record.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.mast.Record-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">key</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record.__getitem__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record.__getitem__-expanded"><a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the motif of index key."""</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">motif</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">:</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt id="link-6" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-6', 'name', 'link-6');">name</a></tt> <tt class="py-op">==</tt> <tt class="py-name">key</tt><tt class="py-op">:</tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line">                    <tt class="py-keyword">return</tt> <tt class="py-name">motif</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">list</tt><tt class="py-op">.</tt><tt id="link-7" class="py-name" targets="Method Bio.Align.AlignInfo.PSSM.__getitem__()=Bio.Align.AlignInfo.PSSM-class.html#__getitem__,Method Bio.Align.MultipleSeqAlignment.__getitem__()=Bio.Align.MultipleSeqAlignment-class.html#__getitem__,Method Bio.Align.PairwiseAlignments.__getitem__()=Bio.Align.PairwiseAlignments-class.html#__getitem__,Method Bio.Align.substitution_matrices.Array.__getitem__()=Bio.Align.substitution_matrices.Array-class.html#__getitem__,Method Bio.Crystal.Chain.__getitem__()=Bio.Crystal.Chain-class.html#__getitem__,Method Bio.Crystal.Crystal.__getitem__()=Bio.Crystal.Crystal-class.html#__getitem__,Method Bio.Data.CodonTable.AmbiguousForwardTable.__getitem__()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#__getitem__,Method Bio.FSSP.fssp_rec.fff_rec.__getitem__()=Bio.FSSP.fssp_rec.fff_rec-class.html#__getitem__,Method Bio.File._IndexedSeqFileDict.__getitem__()=Bio.File._IndexedSeqFileDict-class.html#__getitem__,Method Bio.File._SQLiteManySeqFilesDict.__getitem__()=Bio.File._SQLiteManySeqFilesDict-class.html#__getitem__,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.__getitem__()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#__getitem__,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__getitem__()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#__getitem__,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.__getitem__()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#__getitem__,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__getitem__()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#__getitem__,Method Bio.Graphics.GenomeDiagram._Track.Track.__getitem__()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#__getitem__,Method Bio.Nexus.StandardData.StandardData.__getitem__()=Bio.Nexus.StandardData.StandardData-class.html#__getitem__,Method Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__getitem__()=Bio.PDB.AbstractPropertyMap.AbstractPropertyMap-class.html#__getitem__,Method Bio.PDB.Chain.Chain.__getitem__()=Bio.PDB.Chain.Chain-class.html#__getitem__,Method Bio.PDB.Entity.DisorderedEntityWrapper.__getitem__()=Bio.PDB.Entity.DisorderedEntityWrapper-class.html#__getitem__,Method Bio.PDB.Entity.Entity.__getitem__()=Bio.PDB.Entity.Entity-class.html#__getitem__,Method Bio.PDB.FragmentMapper'.FragmentMapper.__getitem__()=Bio.PDB.FragmentMapper%27.FragmentMapper-class.html#__getitem__,Method Bio.PDB.vectors.Vector.__getitem__()=Bio.PDB.vectors.Vector-class.html#__getitem__,Method Bio.Phylo.BaseTree.Clade.__getitem__()=Bio.Phylo.BaseTree.Clade-class.html#__getitem__,Method Bio.Phylo.PhyloXML.Events.__getitem__()=Bio.Phylo.PhyloXML.Events-class.html#__getitem__,Method Bio.Phylo.PhyloXML.Phyloxml.__getitem__()=Bio.Phylo.PhyloXML.Phyloxml-class.html#__getitem__,Method Bio.Phylo.TreeConstruction._Matrix.__getitem__()=Bio.Phylo.TreeConstruction._Matrix-class.html#__getitem__,Method Bio.Restriction.Restriction.FormattedSeq.__getitem__()=Bio.Restriction.Restriction.FormattedSeq-class.html#__getitem__,Method Bio.SCOP.Cla.Index.__getitem__()=Bio.SCOP.Cla.Index-class.html#__getitem__,Method Bio.SCOP.Raf.SeqMap.__getitem__()=Bio.SCOP.Raf.SeqMap-class.html#__getitem__,Method Bio.SCOP.Raf.SeqMapIndex.__getitem__()=Bio.SCOP.Raf.SeqMapIndex-class.html#__getitem__,Method Bio.SearchIO._model.hit.Hit.__getitem__()=Bio.SearchIO._model.hit.Hit-class.html#__getitem__,Method Bio.SearchIO._model.hsp.HSP.__getitem__()=Bio.SearchIO._model.hsp.HSP-class.html#__getitem__,Method Bio.SearchIO._model.hsp.HSPFragment.__getitem__()=Bio.SearchIO._model.hsp.HSPFragment-class.html#__getitem__,Method Bio.SearchIO._model.query.QueryResult.__getitem__()=Bio.SearchIO._model.query.QueryResult-class.html#__getitem__,Method Bio.Seq.MutableSeq.__getitem__()=Bio.Seq.MutableSeq-class.html#__getitem__,Method Bio.Seq.Seq.__getitem__()=Bio.Seq.Seq-class.html#__getitem__,Method Bio.Seq.UnknownSeq.__getitem__()=Bio.Seq.UnknownSeq-class.html#__getitem__,Method Bio.SeqRecord.SeqRecord.__getitem__()=Bio.SeqRecord.SeqRecord-class.html#__getitem__,Method Bio.codonalign.codonalignment.CodonAlignment.__getitem__()=Bio.codonalign.codonalignment.CodonAlignment-class.html#__getitem__,Method Bio.codonalign.codonseq.CodonSeq.__getitem__()=Bio.codonalign.codonseq.CodonSeq-class.html#__getitem__,Method Bio.motifs.mast.Record.__getitem__()=Bio.motifs.mast.Record-class.html#__getitem__,Method Bio.motifs.matrix.GenericPositionMatrix.__getitem__()=Bio.motifs.matrix.GenericPositionMatrix-class.html#__getitem__,Method Bio.motifs.meme.Record.__getitem__()=Bio.motifs.meme.Record-class.html#__getitem__,Method Bio.motifs.minimal.Record.__getitem__()=Bio.motifs.minimal.Record-class.html#__getitem__,Method Bio.phenotype.phen_micro.PlateRecord.__getitem__()=Bio.phenotype.phen_micro.PlateRecord-class.html#__getitem__,Method Bio.phenotype.phen_micro.WellRecord.__getitem__()=Bio.phenotype.phen_micro.WellRecord-class.html#__getitem__,Method BioSQL.BioSeq.DBSeq.__getitem__()=BioSQL.BioSeq.DBSeq-class.html#__getitem__,Method BioSQL.BioSeqDatabase.BioSeqDatabase.__getitem__()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#__getitem__,Method BioSQL.BioSeqDatabase.DBServer.__getitem__()=BioSQL.BioSeqDatabase.DBServer-class.html#__getitem__"><a title="Bio.Align.AlignInfo.PSSM.__getitem__
Bio.Align.MultipleSeqAlignment.__getitem__
Bio.Align.PairwiseAlignments.__getitem__
Bio.Align.substitution_matrices.Array.__getitem__
Bio.Crystal.Chain.__getitem__
Bio.Crystal.Crystal.__getitem__
Bio.Data.CodonTable.AmbiguousForwardTable.__getitem__
Bio.FSSP.fssp_rec.fff_rec.__getitem__
Bio.File._IndexedSeqFileDict.__getitem__
Bio.File._SQLiteManySeqFilesDict.__getitem__
Bio.Graphics.GenomeDiagram._Diagram.Diagram.__getitem__
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__getitem__
Bio.Graphics.GenomeDiagram._Graph.GraphData.__getitem__
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__getitem__
Bio.Graphics.GenomeDiagram._Track.Track.__getitem__
Bio.Nexus.StandardData.StandardData.__getitem__
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__getitem__
Bio.PDB.Chain.Chain.__getitem__
Bio.PDB.Entity.DisorderedEntityWrapper.__getitem__
Bio.PDB.Entity.Entity.__getitem__
Bio.PDB.FragmentMapper'.FragmentMapper.__getitem__
Bio.PDB.vectors.Vector.__getitem__
Bio.Phylo.BaseTree.Clade.__getitem__
Bio.Phylo.PhyloXML.Events.__getitem__
Bio.Phylo.PhyloXML.Phyloxml.__getitem__
Bio.Phylo.TreeConstruction._Matrix.__getitem__
Bio.Restriction.Restriction.FormattedSeq.__getitem__
Bio.SCOP.Cla.Index.__getitem__
Bio.SCOP.Raf.SeqMap.__getitem__
Bio.SCOP.Raf.SeqMapIndex.__getitem__
Bio.SearchIO._model.hit.Hit.__getitem__
Bio.SearchIO._model.hsp.HSP.__getitem__
Bio.SearchIO._model.hsp.HSPFragment.__getitem__
Bio.SearchIO._model.query.QueryResult.__getitem__
Bio.Seq.MutableSeq.__getitem__
Bio.Seq.Seq.__getitem__
Bio.Seq.UnknownSeq.__getitem__
Bio.SeqRecord.SeqRecord.__getitem__
Bio.codonalign.codonalignment.CodonAlignment.__getitem__
Bio.codonalign.codonseq.CodonSeq.__getitem__
Bio.motifs.mast.Record.__getitem__
Bio.motifs.matrix.GenericPositionMatrix.__getitem__
Bio.motifs.meme.Record.__getitem__
Bio.motifs.minimal.Record.__getitem__
Bio.phenotype.phen_micro.PlateRecord.__getitem__
Bio.phenotype.phen_micro.WellRecord.__getitem__
BioSQL.BioSeq.DBSeq.__getitem__
BioSQL.BioSeqDatabase.BioSeqDatabase.__getitem__
BioSQL.BioSeqDatabase.DBServer.__getitem__" class="py-name" href="#" onclick="return doclink('link-7', '__getitem__', 'link-7');">__getitem__</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-name">key</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line"> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line"> </tt>
<a name="read"></a><div id="read-def"><a name="L54"></a><tt class="py-lineno"> 54</tt> <a class="py-toggle" href="#" id="read-toggle" onclick="return toggle('read');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.mast-module.html#read">read</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="read-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="read-expanded"><a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line">    <tt class="py-docstring">"""Parse a MAST XML format handle as a Record object."""</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line">    <tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt id="link-8" class="py-name" targets="Class Bio.Affy.CelFile.Record=Bio.Affy.CelFile.Record-class.html,Module Bio.Blast.Record=Bio.Blast.Record-module.html,Class Bio.Cluster.Record=Bio.Cluster.Record-class.html,Class Bio.Compass.Record=Bio.Compass.Record-class.html,Class Bio.Emboss.Primer3.Record=Bio.Emboss.Primer3.Record-class.html,Class Bio.ExPASy.Enzyme.Record=Bio.ExPASy.Enzyme.Record-class.html,Class Bio.ExPASy.Prodoc.Record=Bio.ExPASy.Prodoc.Record-class.html,Class Bio.ExPASy.Prosite.Record=Bio.ExPASy.Prosite.Record-class.html,Class Bio.ExPASy.ScanProsite.Record=Bio.ExPASy.ScanProsite.Record-class.html,Class Bio.ExPASy.cellosaurus.Record=Bio.ExPASy.cellosaurus.Record-class.html,Module Bio.GenBank.Record=Bio.GenBank.Record-module.html,Class Bio.GenBank.Record.Record=Bio.GenBank.Record.Record-class.html,Module Bio.Geo.Record=Bio.Geo.Record-module.html,Class Bio.Geo.Record.Record=Bio.Geo.Record.Record-class.html,Class Bio.KEGG.Compound.Record=Bio.KEGG.Compound.Record-class.html,Class Bio.KEGG.Enzyme.Record=Bio.KEGG.Enzyme.Record-class.html,Class Bio.KEGG.Gene.Record=Bio.KEGG.Gene.Record-class.html,Class Bio.Medline.Record=Bio.Medline.Record-class.html,Class Bio.PopGen.GenePop.LargeFileParser.Record=Bio.PopGen.GenePop.LargeFileParser.Record-class.html,Class Bio.PopGen.GenePop.Record=Bio.PopGen.GenePop.Record-class.html,Class Bio.SCOP.Cla.Record=Bio.SCOP.Cla.Record-class.html,Class Bio.SCOP.Des.Record=Bio.SCOP.Des.Record-class.html,Class Bio.SCOP.Dom.Record=Bio.SCOP.Dom.Record-class.html,Class Bio.SCOP.Hie.Record=Bio.SCOP.Hie.Record-class.html,Class Bio.Sequencing.Phd.Record=Bio.Sequencing.Phd.Record-class.html,Class Bio.SwissProt.KeyWList.Record=Bio.SwissProt.KeyWList.Record-class.html,Class Bio.SwissProt.Record=Bio.SwissProt.Record-class.html,Class Bio.UniGene.Record=Bio.UniGene.Record-class.html,Class Bio.motifs.alignace.Record=Bio.motifs.alignace.Record-class.html,Class Bio.motifs.clusterbuster.Record=Bio.motifs.clusterbuster.Record-class.html,Class Bio.motifs.jaspar.Record=Bio.motifs.jaspar.Record-class.html,Class Bio.motifs.mast.Record=Bio.motifs.mast.Record-class.html,Class Bio.motifs.meme.Record=Bio.motifs.meme.Record-class.html,Class Bio.motifs.minimal.Record=Bio.motifs.minimal.Record-class.html,Class Bio.motifs.pfm.Record=Bio.motifs.pfm.Record-class.html,Class Bio.motifs.transfac.Record=Bio.motifs.transfac.Record-class.html,Class Bio.motifs.xms.Record=Bio.motifs.xms.Record-class.html"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.ExPASy.cellosaurus.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.KEGG.Gene.Record
Bio.Medline.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.clusterbuster.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.minimal.Record
Bio.motifs.pfm.Record
Bio.motifs.transfac.Record
Bio.motifs.xms.Record" class="py-name" href="#" onclick="return doclink('link-8', 'Record', 'link-8');">Record</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line">        <tt class="py-name">xml_tree</tt> <tt class="py-op">=</tt> <tt class="py-name">ET</tt><tt class="py-op">.</tt><tt id="link-9" class="py-name" targets="Function Bio.AlignIO.parse()=Bio.AlignIO-module.html#parse,Function Bio.Blast.NCBIXML.parse()=Bio.Blast.NCBIXML-module.html#parse,Function Bio.Compass.parse()=Bio.Compass-module.html#parse,Function Bio.Emboss.Primer3.parse()=Bio.Emboss.Primer3-module.html#parse,Method Bio.Entrez.Parser.DataHandler.parse()=Bio.Entrez.Parser.DataHandler-class.html#parse,Function Bio.Entrez.parse()=Bio.Entrez-module.html#parse,Function Bio.ExPASy.Enzyme.parse()=Bio.ExPASy.Enzyme-module.html#parse,Function Bio.ExPASy.Prodoc.parse()=Bio.ExPASy.Prodoc-module.html#parse,Function Bio.ExPASy.Prosite.parse()=Bio.ExPASy.Prosite-module.html#parse,Function Bio.ExPASy.cellosaurus.parse()=Bio.ExPASy.cellosaurus-module.html#parse,Method Bio.GenBank.FeatureParser.parse()=Bio.GenBank.FeatureParser-class.html#parse,Method Bio.GenBank.RecordParser.parse()=Bio.GenBank.RecordParser-class.html#parse,Method Bio.GenBank.Scanner.InsdcScanner.parse()=Bio.GenBank.Scanner.InsdcScanner-class.html#parse,Function Bio.Geo.parse()=Bio.Geo-module.html#parse,Function Bio.KEGG.Compound.parse()=Bio.KEGG.Compound-module.html#parse,Function Bio.KEGG.Enzyme.parse()=Bio.KEGG.Enzyme-module.html#parse,Function Bio.KEGG.Gene.parse()=Bio.KEGG.Gene-module.html#parse,Method Bio.KEGG.KGML.KGML_parser.KGMLParser.parse()=Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html#parse,Function Bio.KEGG.KGML.KGML_parser.parse()=Bio.KEGG.KGML.KGML_parser-module.html#parse,Function Bio.KEGG.Map.parse()=Bio.KEGG.Map-module.html#parse,Function Bio.Medline.parse()=Bio.Medline-module.html#parse,Method Bio.Phylo.CDAOIO.Parser.parse()=Bio.Phylo.CDAOIO.Parser-class.html#parse,Function Bio.Phylo.CDAOIO.parse()=Bio.Phylo.CDAOIO-module.html#parse,Method Bio.Phylo.NeXMLIO.Parser.parse()=Bio.Phylo.NeXMLIO.Parser-class.html#parse,Function Bio.Phylo.NeXMLIO.parse()=Bio.Phylo.NeXMLIO-module.html#parse,Method Bio.Phylo.NewickIO.Parser.parse()=Bio.Phylo.NewickIO.Parser-class.html#parse,Function Bio.Phylo.NewickIO.parse()=Bio.Phylo.NewickIO-module.html#parse,Function Bio.Phylo.NexusIO.parse()=Bio.Phylo.NexusIO-module.html#parse,Method Bio.Phylo.PhyloXMLIO.Parser.parse()=Bio.Phylo.PhyloXMLIO.Parser-class.html#parse,Function Bio.Phylo.PhyloXMLIO.parse()=Bio.Phylo.PhyloXMLIO-module.html#parse,Function Bio.Phylo._io.parse()=Bio.Phylo._io-module.html#parse,Function Bio.SCOP.Cla.parse()=Bio.SCOP.Cla-module.html#parse,Function Bio.SCOP.Des.parse()=Bio.SCOP.Des-module.html#parse,Function Bio.SCOP.Dom.parse()=Bio.SCOP.Dom-module.html#parse,Function Bio.SCOP.Hie.parse()=Bio.SCOP.Hie-module.html#parse,Function Bio.SCOP.Raf.parse()=Bio.SCOP.Raf-module.html#parse,Method Bio.SearchIO._legacy.NCBIStandalone.BlastErrorParser.parse()=Bio.SearchIO._legacy.NCBIStandalone.BlastErrorParser-class.html#parse,Method Bio.SearchIO._legacy.NCBIStandalone.BlastParser.parse()=Bio.SearchIO._legacy.NCBIStandalone.BlastParser-class.html#parse,Method Bio.SearchIO._legacy.NCBIStandalone.PSIBlastParser.parse()=Bio.SearchIO._legacy.NCBIStandalone.PSIBlastParser-class.html#parse,Method Bio.SearchIO._legacy.ParserSupport.AbstractParser.parse()=Bio.SearchIO._legacy.ParserSupport.AbstractParser-class.html#parse,Function Bio.SearchIO.parse()=Bio.SearchIO-module.html#parse,Method Bio.SeqIO.UniprotIO.Parser.parse()=Bio.SeqIO.UniprotIO.Parser-class.html#parse,Function Bio.SeqIO.parse()=Bio.SeqIO-module.html#parse,Function Bio.Sequencing.Ace.parse()=Bio.Sequencing.Ace-module.html#parse,Function Bio.Sequencing.Phd.parse()=Bio.Sequencing.Phd-module.html#parse,Function Bio.SwissProt.KeyWList.parse()=Bio.SwissProt.KeyWList-module.html#parse,Function Bio.SwissProt.parse()=Bio.SwissProt-module.html#parse,Function Bio.UniGene.parse()=Bio.UniGene-module.html#parse,Function Bio.Wise.psw.parse()=Bio.Wise.psw-module.html#parse,Function Bio.motifs.parse()=Bio.motifs-module.html#parse,Function Bio.phenotype.parse()=Bio.phenotype-module.html#parse"><a title="Bio.AlignIO.parse
Bio.Blast.NCBIXML.parse
Bio.Compass.parse
Bio.Emboss.Primer3.parse
Bio.Entrez.Parser.DataHandler.parse
Bio.Entrez.parse
Bio.ExPASy.Enzyme.parse
Bio.ExPASy.Prodoc.parse
Bio.ExPASy.Prosite.parse
Bio.ExPASy.cellosaurus.parse
Bio.GenBank.FeatureParser.parse
Bio.GenBank.RecordParser.parse
Bio.GenBank.Scanner.InsdcScanner.parse
Bio.Geo.parse
Bio.KEGG.Compound.parse
Bio.KEGG.Enzyme.parse
Bio.KEGG.Gene.parse
Bio.KEGG.KGML.KGML_parser.KGMLParser.parse
Bio.KEGG.KGML.KGML_parser.parse
Bio.KEGG.Map.parse
Bio.Medline.parse
Bio.Phylo.CDAOIO.Parser.parse
Bio.Phylo.CDAOIO.parse
Bio.Phylo.NeXMLIO.Parser.parse
Bio.Phylo.NeXMLIO.parse
Bio.Phylo.NewickIO.Parser.parse
Bio.Phylo.NewickIO.parse
Bio.Phylo.NexusIO.parse
Bio.Phylo.PhyloXMLIO.Parser.parse
Bio.Phylo.PhyloXMLIO.parse
Bio.Phylo._io.parse
Bio.SCOP.Cla.parse
Bio.SCOP.Des.parse
Bio.SCOP.Dom.parse
Bio.SCOP.Hie.parse
Bio.SCOP.Raf.parse
Bio.SearchIO._legacy.NCBIStandalone.BlastErrorParser.parse
Bio.SearchIO._legacy.NCBIStandalone.BlastParser.parse
Bio.SearchIO._legacy.NCBIStandalone.PSIBlastParser.parse
Bio.SearchIO._legacy.ParserSupport.AbstractParser.parse
Bio.SearchIO.parse
Bio.SeqIO.UniprotIO.Parser.parse
Bio.SeqIO.parse
Bio.Sequencing.Ace.parse
Bio.Sequencing.Phd.parse
Bio.SwissProt.KeyWList.parse
Bio.SwissProt.parse
Bio.UniGene.parse
Bio.Wise.psw.parse
Bio.motifs.parse
Bio.phenotype.parse" class="py-name" href="#" onclick="return doclink('link-9', 'parse', 'link-9');">parse</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">ET</tt><tt class="py-op">.</tt><tt class="py-name">ParseError</tt><tt class="py-op">:</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Improper MAST XML input file. XML root tag should start with &lt;mast version= ..."</tt><tt class="py-op">)</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line">    <tt id="link-10" class="py-name" targets="Function Bio.motifs.mast.__read_metadata()=Bio.motifs.mast-module.html#__read_metadata"><a title="Bio.motifs.mast.__read_metadata" class="py-name" href="#" onclick="return doclink('link-10', '__read_metadata', 'link-10');">__read_metadata</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">,</tt> <tt class="py-name">xml_tree</tt><tt class="py-op">)</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line">    <tt id="link-11" class="py-name" targets="Function Bio.motifs.mast.__read_sequences()=Bio.motifs.mast-module.html#__read_sequences,Function Bio.motifs.meme.__read_sequences()=Bio.motifs.meme-module.html#__read_sequences"><a title="Bio.motifs.mast.__read_sequences
Bio.motifs.meme.__read_sequences" class="py-name" href="#" onclick="return doclink('link-11', '__read_sequences', 'link-11');">__read_sequences</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">,</tt> <tt class="py-name">xml_tree</tt><tt class="py-op">)</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">record</tt> </tt>
</div><a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line"> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line"> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line"><tt class="py-comment"># Everything below is private</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line"> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line"> </tt>
<a name="__read_metadata"></a><div id="__read_metadata-def"><a name="L69"></a><tt class="py-lineno"> 69</tt> <a class="py-toggle" href="#" id="__read_metadata-toggle" onclick="return toggle('__read_metadata');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.mast-module.html#__read_metadata">__read_metadata</a><tt class="py-op">(</tt><tt class="py-param">record</tt><tt class="py-op">,</tt> <tt class="py-param">xml_tree</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="__read_metadata-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="__read_metadata-expanded"><a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line">    <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-12" class="py-name"><a title="Bio.Cluster._cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-12', 'version', 'link-3');">version</a></tt> <tt class="py-op">=</tt> <tt class="py-name">xml_tree</tt><tt class="py-op">.</tt><tt class="py-name">getroot</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-13" class="py-name" targets="Method Bio.Align.substitution_matrices.Array.get()=Bio.Align.substitution_matrices.Array-class.html#get,Method Bio.Crystal.Crystal.get()=Bio.Crystal.Crystal-class.html#get,Method Bio.Data.CodonTable.AmbiguousForwardTable.get()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#get,Method Bio.File._IndexedSeqFileDict.get()=Bio.File._IndexedSeqFileDict-class.html#get,Method Bio.File._IndexedSeqFileProxy.get()=Bio.File._IndexedSeqFileProxy-class.html#get,Method Bio.File._SQLiteManySeqFilesDict.get()=Bio.File._SQLiteManySeqFilesDict-class.html#get,Method Bio.Restriction.Restriction.RestrictionBatch.get()=Bio.Restriction.Restriction.RestrictionBatch-class.html#get,Method Bio.SearchIO._index.SearchIndexer.get()=Bio.SearchIO._index.SearchIndexer-class.html#get,Method Bio.SeqIO._index.SeqFileRandomAccess.get()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#get,Method Bio.SeqIO._index.SffRandomAccess.get()=Bio.SeqIO._index.SffRandomAccess-class.html#get,Method Bio.SeqIO._index.SffTrimedRandomAccess.get()=Bio.SeqIO._index.SffTrimedRandomAccess-class.html#get,Method Bio.SeqIO._index.UniprotRandomAccess.get()=Bio.SeqIO._index.UniprotRandomAccess-class.html#get"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-13', 'get', 'link-13');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"version"</tt><tt class="py-op">)</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line">    <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-14" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._DatabaseReportConsumer.database" class="py-name" href="#" onclick="return doclink('link-14', 'database', 'link-4');">database</a></tt> <tt class="py-op">=</tt> <tt class="py-name">xml_tree</tt><tt class="py-op">.</tt><tt id="link-15" class="py-name" targets="Method Bio.Seq.Seq.find()=Bio.Seq.Seq-class.html#find,Function Bio.triefind.find()=Bio.triefind-module.html#find"><a title="Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-15', 'find', 'link-15');">find</a></tt><tt class="py-op">(</tt><tt class="py-string">"sequence_dbs"</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-16" class="py-name"><a title="Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-16', 'find', 'link-15');">find</a></tt><tt class="py-op">(</tt><tt class="py-string">"sequence_db"</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-17" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-17', 'get', 'link-13');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"source"</tt><tt class="py-op">)</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line">    <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-18" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-18', 'alphabet', 'link-5');">alphabet</a></tt> <tt class="py-op">=</tt> <tt class="py-name">xml_tree</tt><tt class="py-op">.</tt><tt id="link-19" class="py-name"><a title="Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-19', 'find', 'link-15');">find</a></tt><tt class="py-op">(</tt><tt class="py-string">"alphabet"</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-20" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-20', 'get', 'link-13');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"name"</tt><tt class="py-op">)</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line">    <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">strand_handling</tt> <tt class="py-op">=</tt> <tt class="py-name">xml_tree</tt><tt class="py-op">.</tt><tt id="link-21" class="py-name"><a title="Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-21', 'find', 'link-15');">find</a></tt><tt class="py-op">(</tt><tt class="py-string">"settings"</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-22" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-22', 'get', 'link-13');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"strand_handling"</tt><tt class="py-op">)</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">    <tt class="py-comment"># TODO - read other metadata</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt id="link-23" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-23', 'i', 'link-23');">i</a></tt><tt class="py-op">,</tt> <tt class="py-name">motif_tree</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt class="py-name">xml_tree</tt><tt class="py-op">.</tt><tt id="link-24" class="py-name"><a title="Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-24', 'find', 'link-15');">find</a></tt><tt class="py-op">(</tt><tt class="py-string">"motifs"</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">findall</tt><tt class="py-op">(</tt><tt class="py-string">"motif"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line">        <tt class="py-name">motif</tt> <tt class="py-op">=</tt> <tt id="link-25" class="py-name"><a title="Bio.motifs.meme" class="py-name" href="#" onclick="return doclink('link-25', 'meme', 'link-2');">meme</a></tt><tt class="py-op">.</tt><tt id="link-26" class="py-name" targets="Class Bio.motifs.Motif=Bio.motifs.Motif-class.html,Class Bio.motifs.jaspar.Motif=Bio.motifs.jaspar.Motif-class.html,Class Bio.motifs.meme.Motif=Bio.motifs.meme.Motif-class.html,Class Bio.motifs.transfac.Motif=Bio.motifs.transfac.Motif-class.html"><a title="Bio.motifs.Motif
Bio.motifs.jaspar.Motif
Bio.motifs.meme.Motif
Bio.motifs.transfac.Motif" class="py-name" href="#" onclick="return doclink('link-26', 'Motif', 'link-26');">Motif</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-27" class="py-name"><a title="Bio.Align.substitution_matrices.Array.alphabet
Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-27', 'alphabet', 'link-5');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - motif.name not in XML - always index?</tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line">        <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt id="link-28" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-28', 'name', 'link-6');">name</a></tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-29" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-29', 'i', 'link-23');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line">        <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt id="link-30" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Variable Bio.PDB.Entity.Entity.id=Bio.PDB.Entity.Entity-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-30', 'id', 'link-30');">id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">motif_tree</tt><tt class="py-op">.</tt><tt id="link-31" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-31', 'get', 'link-13');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"id"</tt><tt class="py-op">)</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line">        <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt class="py-name">alt_id</tt> <tt class="py-op">=</tt> <tt class="py-name">motif_tree</tt><tt class="py-op">.</tt><tt id="link-32" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-32', 'get', 'link-13');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"alt"</tt><tt class="py-op">)</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">        <tt class="py-name">motif</tt><tt class="py-op">.</tt><tt id="link-33" class="py-name" targets="Method Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length()=Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer-class.html#length"><a title="Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-33', 'length', 'link-33');">length</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">motif_tree</tt><tt class="py-op">.</tt><tt id="link-34" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-34', 'get', 'link-13');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"length"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - add nsites, evalue</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line">        <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-35" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-35', 'append', 'link-35');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">motif</tt><tt class="py-op">)</tt> </tt>
</div><a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line"> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line"> </tt>
<a name="__read_sequences"></a><div id="__read_sequences-def"><a name="L86"></a><tt class="py-lineno"> 86</tt> <a class="py-toggle" href="#" id="__read_sequences-toggle" onclick="return toggle('__read_sequences');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.mast-module.html#__read_sequences">__read_sequences</a><tt class="py-op">(</tt><tt class="py-param">record</tt><tt class="py-op">,</tt> <tt class="py-param">xml_tree</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="__read_sequences-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="__read_sequences-expanded"><a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line">    <tt class="py-docstring">"""Read sequences from XML ElementTree object."""</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">sequence_tree</tt> <tt class="py-keyword">in</tt> <tt class="py-name">xml_tree</tt><tt class="py-op">.</tt><tt id="link-36" class="py-name"><a title="Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-36', 'find', 'link-15');">find</a></tt><tt class="py-op">(</tt><tt class="py-string">"sequences"</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">findall</tt><tt class="py-op">(</tt><tt class="py-string">"sequence"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line">        <tt class="py-name">sequence_name</tt> <tt class="py-op">=</tt> <tt class="py-name">sequence_tree</tt><tt class="py-op">.</tt><tt id="link-37" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-37', 'get', 'link-13');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"name"</tt><tt class="py-op">)</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line">        <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">sequences</tt><tt class="py-op">.</tt><tt id="link-38" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-38', 'append', 'link-35');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">sequence_name</tt><tt class="py-op">)</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">        <tt class="py-name">diagram_str</tt> <tt class="py-op">=</tt> <tt id="link-39" class="py-name" targets="Function Bio.motifs.mast.__make_diagram()=Bio.motifs.mast-module.html#__make_diagram"><a title="Bio.motifs.mast.__make_diagram" class="py-name" href="#" onclick="return doclink('link-39', '__make_diagram', 'link-39');">__make_diagram</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">,</tt> <tt class="py-name">sequence_tree</tt><tt class="py-op">)</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line">        <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">diagrams</tt><tt class="py-op">[</tt><tt class="py-name">sequence_name</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">diagram_str</tt> </tt>
</div><a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - add description, evalue, length, combined_pvalue</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line"> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line"> </tt>
<a name="__make_diagram"></a><div id="__make_diagram-def"><a name="L96"></a><tt class="py-lineno"> 96</tt> <a class="py-toggle" href="#" id="__make_diagram-toggle" onclick="return toggle('__make_diagram');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.motifs.mast-module.html#__make_diagram">__make_diagram</a><tt class="py-op">(</tt><tt class="py-param">record</tt><tt class="py-op">,</tt> <tt class="py-param">sequence_tree</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="__make_diagram-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="__make_diagram-expanded"><a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line">    <tt class="py-docstring">"""Make diagram string found in text file based on motif hit info."""</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line">    <tt class="py-name">sequence_length</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">sequence_tree</tt><tt class="py-op">.</tt><tt id="link-40" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-40', 'get', 'link-13');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"length"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">    <tt class="py-name">hit_eles</tt><tt class="py-op">,</tt> <tt class="py-name">hit_motifs</tt><tt class="py-op">,</tt> <tt class="py-name">gaps</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">seg_tree</tt> <tt class="py-keyword">in</tt> <tt class="py-name">sequence_tree</tt><tt class="py-op">.</tt><tt class="py-name">findall</tt><tt class="py-op">(</tt><tt class="py-string">"seg"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">hit_ele</tt> <tt class="py-keyword">in</tt> <tt class="py-name">seg_tree</tt><tt class="py-op">.</tt><tt class="py-name">findall</tt><tt class="py-op">(</tt><tt class="py-string">"hit"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line">            <tt class="py-name">hit_pos</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">hit_ele</tt><tt class="py-op">.</tt><tt id="link-41" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-41', 'get', 'link-13');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"pos"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">hit_eles</tt><tt class="py-op">:</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line">                <tt class="py-name">gap</tt> <tt class="py-op">=</tt> <tt class="py-name">hit_pos</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">                <tt class="py-name">gap</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">hit_pos</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line">                       <tt class="py-op">-</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">hit_eles</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-42" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-42', 'get', 'link-13');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"pos"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line">                       <tt class="py-op">-</tt> <tt class="py-name">hit_motifs</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-43" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-43', 'length', 'link-33');">length</a></tt><tt class="py-op">)</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">            <tt class="py-name">gaps</tt><tt class="py-op">.</tt><tt id="link-44" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-44', 'append', 'link-35');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">gap</tt><tt class="py-op">)</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">            <tt class="py-name">hit_motifs</tt><tt class="py-op">.</tt><tt id="link-45" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-45', 'append', 'link-35');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">[</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">hit_ele</tt><tt class="py-op">.</tt><tt id="link-46" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-46', 'get', 'link-13');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"idx"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">            <tt class="py-name">hit_eles</tt><tt class="py-op">.</tt><tt id="link-47" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-47', 'append', 'link-35');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">hit_ele</tt><tt class="py-op">)</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">hit_eles</tt><tt class="py-op">:</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">sequence_length</tt><tt class="py-op">)</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">strand_handling</tt> <tt class="py-op">==</tt> <tt class="py-string">"combine"</tt><tt class="py-op">:</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">        <tt class="py-name">motif_strs</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">"[{}{}]"</tt><tt class="py-op">.</tt><tt id="link-48" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.format()=Bio.Align.MultipleSeqAlignment-class.html#format,Method Bio.Align.PairwiseAlignment.format()=Bio.Align.PairwiseAlignment-class.html#format,Method Bio.Phylo.BaseTree.Tree.format()=Bio.Phylo.BaseTree.Tree-class.html#format,Method Bio.Restriction.Restriction.RestrictionBatch.format()=Bio.Restriction.Restriction.RestrictionBatch-class.html#format,Method Bio.SeqRecord.SeqRecord.format()=Bio.SeqRecord.SeqRecord-class.html#format,Method Bio.SubsMat.SeqMat.format()=Bio.SubsMat.SeqMat-class.html#format,Method Bio.motifs.Motif.format()=Bio.motifs.Motif-class.html#format"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-48', 'format', 'link-48');">format</a></tt><tt class="py-op">(</tt><tt class="py-string">"-"</tt> <tt class="py-keyword">if</tt> <tt class="py-name">hit_ele</tt><tt class="py-op">.</tt><tt id="link-49" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-49', 'get', 'link-13');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"rc"</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">"y"</tt> <tt class="py-keyword">else</tt> <tt class="py-string">"+"</tt><tt class="py-op">,</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">                                      <tt class="py-name">hit_motif</tt><tt class="py-op">.</tt><tt id="link-50" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-50', 'name', 'link-6');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">                      <tt class="py-keyword">for</tt> <tt class="py-name">hit_ele</tt><tt class="py-op">,</tt> <tt class="py-name">hit_motif</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">hit_eles</tt><tt class="py-op">,</tt> <tt class="py-name">hit_motifs</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">strand_handling</tt> <tt class="py-op">==</tt> <tt class="py-string">"unstranded"</tt><tt class="py-op">:</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line">        <tt class="py-name">motif_strs</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">"[{}]"</tt><tt class="py-op">.</tt><tt id="link-51" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-51', 'format', 'link-48');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">hit_motif</tt><tt class="py-op">.</tt><tt id="link-52" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-52', 'name', 'link-6');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">                      <tt class="py-keyword">for</tt> <tt class="py-name">hit_ele</tt><tt class="py-op">,</tt> <tt class="py-name">hit_motif</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">hit_eles</tt><tt class="py-op">,</tt> <tt class="py-name">hit_motifs</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - more strand_handling possibilities?</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">Exception</tt><tt class="py-op">(</tt><tt class="py-string">"Strand handling option {} not parsable"</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">                        <tt class="py-op">.</tt><tt id="link-53" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-53', 'format', 'link-48');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">strand_handling</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line">    <tt class="py-name">tail_length</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">sequence_length</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line">                   <tt class="py-op">-</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">hit_eles</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-54" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-54', 'get', 'link-13');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"pos"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line">                   <tt class="py-op">-</tt> <tt class="py-name">hit_motifs</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-55" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-55', 'length', 'link-33');">length</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line">    <tt class="py-name">motifs_with_gaps</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line">                         <tt class="py-keyword">for</tt> <tt class="py-name">pair</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">gaps</tt><tt class="py-op">,</tt> <tt class="py-name">motif_strs</tt><tt class="py-op">)</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line">                         <tt class="py-keyword">for</tt> <tt class="py-name">s</tt> <tt class="py-keyword">in</tt> <tt class="py-name">pair</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line">                        <tt class="py-op">[</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">tail_length</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line">    <tt class="py-comment"># remove 0-length gaps</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line">    <tt class="py-name">motifs_with_gaps</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">s</tt> <tt class="py-keyword">for</tt> <tt class="py-name">s</tt> <tt class="py-keyword">in</tt> <tt class="py-name">motifs_with_gaps</tt> <tt class="py-keyword">if</tt> <tt class="py-name">s</tt> <tt class="py-op">!=</tt> <tt class="py-string">"0"</tt><tt class="py-op">]</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-string">"-"</tt><tt class="py-op">.</tt><tt id="link-56" class="py-name" targets="Method Bio.Seq.MutableSeq.join()=Bio.Seq.MutableSeq-class.html#join,Method Bio.Seq.Seq.join()=Bio.Seq.Seq-class.html#join,Method Bio.Seq.UnknownSeq.join()=Bio.Seq.UnknownSeq-class.html#join"><a title="Bio.Seq.MutableSeq.join
Bio.Seq.Seq.join
Bio.Seq.UnknownSeq.join" class="py-name" href="#" onclick="return doclink('link-56', 'join', 'link-56');">join</a></tt><tt class="py-op">(</tt><tt class="py-name">motifs_with_gaps</tt><tt class="py-op">)</tt> </tt>
</div><a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Nov  6 13:33:57 2019
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
