<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.NeuralNetwork.Gene.Schema</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.NeuralNetwork-module.html">Package&nbsp;NeuralNetwork</a> ::
        <a href="Bio.NeuralNetwork.Gene-module.html">Package&nbsp;Gene</a> ::
        Module&nbsp;Schema
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.NeuralNetwork.Gene.Schema-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.NeuralNetwork.Gene.Schema-module.html">Module Bio.NeuralNetwork.Gene.Schema</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-docstring">"""Deal with Motifs or Signatures allowing ambiguity in the sequences.</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-docstring">This class contains Schema which deal with Motifs and Signatures at</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-docstring">a higher level, by introducing \`don't care\` (ambiguity) symbols into</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-docstring">the sequences. For instance, you could combine the following Motifs:</tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-docstring">'GATC', 'GATG', 'GATG', 'GATT'</tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-docstring">as all falling under a schema like 'GAT*', where the star indicates a</tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-docstring">character can be anything. This helps us condense a whole ton of</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-docstring">motifs or signatures.</tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-comment"># standard modules</tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">__future__</tt> <tt class="py-keyword">import</tt> <tt class="py-name">print_function</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">random</tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">re</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-1', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt id="link-2" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-2', 'range', 'link-2');">range</a></tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-3" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-3', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-4" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-4', 'Alphabet', 'link-4');">Alphabet</a></tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-5" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-5', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-6" class="py-name" targets="Module Bio.Seq=Bio.Seq-module.html,Class Bio.Seq.Seq=Bio.Seq.Seq-class.html"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-6', 'Seq', 'link-6');">Seq</a></tt> <tt class="py-keyword">import</tt> <tt id="link-7" class="py-name" targets="Class Bio.Seq.MutableSeq=Bio.Seq.MutableSeq-class.html"><a title="Bio.Seq.MutableSeq" class="py-name" href="#" onclick="return doclink('link-7', 'MutableSeq', 'link-7');">MutableSeq</a></tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"><tt class="py-comment"># neural network libraries</tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt><tt id="link-8" class="py-name" targets="Module Bio.NeuralNetwork.Gene.Pattern=Bio.NeuralNetwork.Gene.Pattern-module.html"><a title="Bio.NeuralNetwork.Gene.Pattern" class="py-name" href="#" onclick="return doclink('link-8', 'Pattern', 'link-8');">Pattern</a></tt> <tt class="py-keyword">import</tt> <tt id="link-9" class="py-name" targets="Class Bio.NeuralNetwork.Gene.Pattern.PatternRepository=Bio.NeuralNetwork.Gene.Pattern.PatternRepository-class.html"><a title="Bio.NeuralNetwork.Gene.Pattern.PatternRepository" class="py-name" href="#" onclick="return doclink('link-9', 'PatternRepository', 'link-9');">PatternRepository</a></tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"><tt class="py-comment"># genetic algorithm libraries</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-10" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-10', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-11" class="py-name" targets="Package Bio.GA=Bio.GA-module.html"><a title="Bio.GA" class="py-name" href="#" onclick="return doclink('link-11', 'GA', 'link-11');">GA</a></tt> <tt class="py-keyword">import</tt> <tt id="link-12" class="py-name" targets="Module Bio.GA.Organism=Bio.GA.Organism-module.html,Class Bio.GA.Organism.Organism=Bio.GA.Organism.Organism-class.html,Class Bio.Graphics.BasicChromosome.Organism=Bio.Graphics.BasicChromosome.Organism-class.html"><a title="Bio.GA.Organism
Bio.GA.Organism.Organism
Bio.Graphics.BasicChromosome.Organism" class="py-name" href="#" onclick="return doclink('link-12', 'Organism', 'link-12');">Organism</a></tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-13" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-13', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-14" class="py-name"><a title="Bio.GA" class="py-name" href="#" onclick="return doclink('link-14', 'GA', 'link-11');">GA</a></tt><tt class="py-op">.</tt><tt id="link-15" class="py-name" targets="Module Bio.GA.Evolver=Bio.GA.Evolver-module.html"><a title="Bio.GA.Evolver" class="py-name" href="#" onclick="return doclink('link-15', 'Evolver', 'link-15');">Evolver</a></tt> <tt class="py-keyword">import</tt> <tt id="link-16" class="py-name" targets="Class Bio.GA.Evolver.GenerationEvolver=Bio.GA.Evolver.GenerationEvolver-class.html"><a title="Bio.GA.Evolver.GenerationEvolver" class="py-name" href="#" onclick="return doclink('link-16', 'GenerationEvolver', 'link-16');">GenerationEvolver</a></tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-17" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-17', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-18" class="py-name"><a title="Bio.GA" class="py-name" href="#" onclick="return doclink('link-18', 'GA', 'link-11');">GA</a></tt><tt class="py-op">.</tt><tt id="link-19" class="py-name" targets="Package Bio.GA.Mutation=Bio.GA.Mutation-module.html"><a title="Bio.GA.Mutation" class="py-name" href="#" onclick="return doclink('link-19', 'Mutation', 'link-19');">Mutation</a></tt><tt class="py-op">.</tt><tt id="link-20" class="py-name" targets="Module Bio.GA.Mutation.Simple=Bio.GA.Mutation.Simple-module.html"><a title="Bio.GA.Mutation.Simple" class="py-name" href="#" onclick="return doclink('link-20', 'Simple', 'link-20');">Simple</a></tt> <tt class="py-keyword">import</tt> <tt id="link-21" class="py-name" targets="Class Bio.GA.Mutation.Simple.SinglePositionMutation=Bio.GA.Mutation.Simple.SinglePositionMutation-class.html"><a title="Bio.GA.Mutation.Simple.SinglePositionMutation" class="py-name" href="#" onclick="return doclink('link-21', 'SinglePositionMutation', 'link-21');">SinglePositionMutation</a></tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-22" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-22', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-23" class="py-name"><a title="Bio.GA" class="py-name" href="#" onclick="return doclink('link-23', 'GA', 'link-11');">GA</a></tt><tt class="py-op">.</tt><tt id="link-24" class="py-name" targets="Package Bio.GA.Crossover=Bio.GA.Crossover-module.html"><a title="Bio.GA.Crossover" class="py-name" href="#" onclick="return doclink('link-24', 'Crossover', 'link-24');">Crossover</a></tt><tt class="py-op">.</tt><tt id="link-25" class="py-name" targets="Module Bio.GA.Crossover.Point=Bio.GA.Crossover.Point-module.html,Class Bio.Phylo.PhyloXML.Point=Bio.Phylo.PhyloXML.Point-class.html"><a title="Bio.GA.Crossover.Point
Bio.Phylo.PhyloXML.Point" class="py-name" href="#" onclick="return doclink('link-25', 'Point', 'link-25');">Point</a></tt> <tt class="py-keyword">import</tt> <tt id="link-26" class="py-name" targets="Class Bio.GA.Crossover.Point.SinglePointCrossover=Bio.GA.Crossover.Point.SinglePointCrossover-class.html"><a title="Bio.GA.Crossover.Point.SinglePointCrossover" class="py-name" href="#" onclick="return doclink('link-26', 'SinglePointCrossover', 'link-26');">SinglePointCrossover</a></tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-27" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-27', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-28" class="py-name"><a title="Bio.GA" class="py-name" href="#" onclick="return doclink('link-28', 'GA', 'link-11');">GA</a></tt><tt class="py-op">.</tt><tt id="link-29" class="py-name" targets="Package Bio.GA.Repair=Bio.GA.Repair-module.html"><a title="Bio.GA.Repair" class="py-name" href="#" onclick="return doclink('link-29', 'Repair', 'link-29');">Repair</a></tt><tt class="py-op">.</tt><tt id="link-30" class="py-name" targets="Module Bio.GA.Repair.Stabilizing=Bio.GA.Repair.Stabilizing-module.html"><a title="Bio.GA.Repair.Stabilizing" class="py-name" href="#" onclick="return doclink('link-30', 'Stabilizing', 'link-30');">Stabilizing</a></tt> <tt class="py-keyword">import</tt> <tt id="link-31" class="py-name" targets="Class Bio.GA.Repair.Stabilizing.AmbiguousRepair=Bio.GA.Repair.Stabilizing.AmbiguousRepair-class.html"><a title="Bio.GA.Repair.Stabilizing.AmbiguousRepair" class="py-name" href="#" onclick="return doclink('link-31', 'AmbiguousRepair', 'link-31');">AmbiguousRepair</a></tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-32" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-32', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-33" class="py-name"><a title="Bio.GA" class="py-name" href="#" onclick="return doclink('link-33', 'GA', 'link-11');">GA</a></tt><tt class="py-op">.</tt><tt id="link-34" class="py-name" targets="Package Bio.GA.Selection=Bio.GA.Selection-module.html,Module Bio.PDB.Selection=Bio.PDB.Selection-module.html"><a title="Bio.GA.Selection
Bio.PDB.Selection" class="py-name" href="#" onclick="return doclink('link-34', 'Selection', 'link-34');">Selection</a></tt><tt class="py-op">.</tt><tt id="link-35" class="py-name" targets="Module Bio.GA.Selection.Tournament=Bio.GA.Selection.Tournament-module.html"><a title="Bio.GA.Selection.Tournament" class="py-name" href="#" onclick="return doclink('link-35', 'Tournament', 'link-35');">Tournament</a></tt> <tt class="py-keyword">import</tt> <tt id="link-36" class="py-name" targets="Class Bio.GA.Selection.Tournament.TournamentSelection=Bio.GA.Selection.Tournament.TournamentSelection-class.html"><a title="Bio.GA.Selection.Tournament.TournamentSelection" class="py-name" href="#" onclick="return doclink('link-36', 'TournamentSelection', 'link-36');">TournamentSelection</a></tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-37" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-37', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-38" class="py-name"><a title="Bio.GA" class="py-name" href="#" onclick="return doclink('link-38', 'GA', 'link-11');">GA</a></tt><tt class="py-op">.</tt><tt id="link-39" class="py-name"><a title="Bio.GA.Selection
Bio.PDB.Selection" class="py-name" href="#" onclick="return doclink('link-39', 'Selection', 'link-34');">Selection</a></tt><tt class="py-op">.</tt><tt id="link-40" class="py-name" targets="Module Bio.GA.Selection.Diversity=Bio.GA.Selection.Diversity-module.html"><a title="Bio.GA.Selection.Diversity" class="py-name" href="#" onclick="return doclink('link-40', 'Diversity', 'link-40');">Diversity</a></tt> <tt class="py-keyword">import</tt> <tt id="link-41" class="py-name" targets="Class Bio.GA.Selection.Diversity.DiversitySelection=Bio.GA.Selection.Diversity.DiversitySelection-class.html"><a title="Bio.GA.Selection.Diversity.DiversitySelection" class="py-name" href="#" onclick="return doclink('link-41', 'DiversitySelection', 'link-41');">DiversitySelection</a></tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"><tt class="py-name">__docformat__</tt> <tt class="py-op">=</tt> <tt class="py-string">"restructuredtext en"</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line"> </tt>
<a name="Schema"></a><div id="Schema-def"><a name="L44"></a><tt class="py-lineno"> 44</tt> <a class="py-toggle" href="#" id="Schema-toggle" onclick="return toggle('Schema');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.Schema-class.html">Schema</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Schema-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Schema-expanded"><a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line">    <tt class="py-docstring">"""Deal with motifs that have ambiguity characters in it.</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line"><tt class="py-docstring">    This motif class allows specific ambiguity characters and tries to</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"><tt class="py-docstring">    speed up finding motifs using regular expressions.</tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line"><tt class="py-docstring">    This is likely to be a replacement for the Schema representation,</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line"><tt class="py-docstring">    since it allows multiple ambiguity characters to be used.</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="Schema.__init__"></a><div id="Schema.__init__-def"><a name="L53"></a><tt class="py-lineno"> 53</tt> <a class="py-toggle" href="#" id="Schema.__init__-toggle" onclick="return toggle('Schema.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.Schema-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">ambiguity_info</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Schema.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Schema.__init__-expanded"><a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize with ambiguity information.</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line"><tt class="py-docstring">        o ambiguity_info - A dictionary which maps letters in the motifs to</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line"><tt class="py-docstring">        the ambiguous characters which they might represent. For example,</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line"><tt class="py-docstring">        {'R' : 'AG'} specifies that Rs in the motif can match an A or a G.</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line"><tt class="py-docstring">        All letters in the motif must be represented in the ambiguity_info</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line"><tt class="py-docstring">        dictionary.</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_ambiguity_info</tt> <tt class="py-op">=</tt> <tt class="py-name">ambiguity_info</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line"> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line">        <tt class="py-comment"># a cache of all encoded motifs</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_motif_cache</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
</div><a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line"> </tt>
<a name="Schema.encode_motif"></a><div id="Schema.encode_motif-def"><a name="L69"></a><tt class="py-lineno"> 69</tt> <a class="py-toggle" href="#" id="Schema.encode_motif-toggle" onclick="return toggle('Schema.encode_motif');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.Schema-class.html#encode_motif">encode_motif</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">motif</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Schema.encode_motif-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Schema.encode_motif-expanded"><a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line">        <tt class="py-docstring">"""Encode the passed motif as a regular expression pattern object.</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line"><tt class="py-docstring">        o motif - The motif we want to encode. This should be a string.</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line"><tt class="py-docstring">        Returns:</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line"><tt class="py-docstring">        A compiled regular expression pattern object that can be used</tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line"><tt class="py-docstring">        for searching strings.</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line">        <tt class="py-name">regexp_string</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line"> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">motif_letter</tt> <tt class="py-keyword">in</tt> <tt class="py-name">motif</tt><tt class="py-op">:</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line">                <tt class="py-name">letter_matches</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_ambiguity_info</tt><tt class="py-op">[</tt><tt class="py-name">motif_letter</tt><tt class="py-op">]</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">KeyError</tt><tt class="py-op">(</tt><tt class="py-string">"No match information for letter %s"</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line">                               <tt class="py-op">%</tt> <tt class="py-name">motif_letter</tt><tt class="py-op">)</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line"> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">letter_matches</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line">                <tt class="py-name">regexp_match</tt> <tt class="py-op">=</tt> <tt class="py-string">"["</tt> <tt class="py-op">+</tt> <tt class="py-name">letter_matches</tt> <tt class="py-op">+</tt> <tt class="py-string">"]"</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">letter_matches</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line">                <tt class="py-name">regexp_match</tt> <tt class="py-op">=</tt> <tt class="py-name">letter_matches</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected match information %s"</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line">                                 <tt class="py-op">%</tt> <tt class="py-name">letter_matches</tt><tt class="py-op">)</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line"> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line">            <tt class="py-name">regexp_string</tt> <tt class="py-op">+=</tt> <tt class="py-name">regexp_match</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line"> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-name">regexp_string</tt><tt class="py-op">)</tt> </tt>
</div><a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line"> </tt>
<a name="Schema.find_ambiguous"></a><div id="Schema.find_ambiguous-def"><a name="L101"></a><tt class="py-lineno">101</tt> <a class="py-toggle" href="#" id="Schema.find_ambiguous-toggle" onclick="return toggle('Schema.find_ambiguous');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.Schema-class.html#find_ambiguous">find_ambiguous</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">motif</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Schema.find_ambiguous-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Schema.find_ambiguous-expanded"><a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the location of ambiguous items in the motif.</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line"><tt class="py-docstring">        This just checks through the motif and compares each letter</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line"><tt class="py-docstring">        against the ambiguity information. If a letter stands for multiple</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line"><tt class="py-docstring">        items, it is ambiguous.</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line">        <tt class="py-name">ambig_positions</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">motif_letter_pos</tt> <tt class="py-keyword">in</tt> <tt id="link-42" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-42', 'range', 'link-2');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">motif</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">            <tt class="py-name">motif_letter</tt> <tt class="py-op">=</tt> <tt class="py-name">motif</tt><tt class="py-op">[</tt><tt class="py-name">motif_letter_pos</tt><tt class="py-op">]</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">                <tt class="py-name">letter_matches</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_ambiguity_info</tt><tt class="py-op">[</tt><tt class="py-name">motif_letter</tt><tt class="py-op">]</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">KeyError</tt><tt class="py-op">(</tt><tt class="py-string">"No match information for letter %s"</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">                               <tt class="py-op">%</tt> <tt class="py-name">motif_letter</tt><tt class="py-op">)</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line"> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">letter_matches</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line">                <tt class="py-name">ambig_positions</tt><tt class="py-op">.</tt><tt id="link-43" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-43', 'append', 'link-43');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">motif_letter_pos</tt><tt class="py-op">)</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line"> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">ambig_positions</tt> </tt>
</div><a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line"> </tt>
<a name="Schema.num_ambiguous"></a><div id="Schema.num_ambiguous-def"><a name="L122"></a><tt class="py-lineno">122</tt> <a class="py-toggle" href="#" id="Schema.num_ambiguous-toggle" onclick="return toggle('Schema.num_ambiguous');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.Schema-class.html#num_ambiguous">num_ambiguous</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">motif</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Schema.num_ambiguous-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Schema.num_ambiguous-expanded"><a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the number of ambiguous letters in a given motif.</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line">        <tt class="py-name">ambig_positions</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-44" class="py-name" targets="Method Bio.NeuralNetwork.Gene.Schema.Schema.find_ambiguous()=Bio.NeuralNetwork.Gene.Schema.Schema-class.html#find_ambiguous"><a title="Bio.NeuralNetwork.Gene.Schema.Schema.find_ambiguous" class="py-name" href="#" onclick="return doclink('link-44', 'find_ambiguous', 'link-44');">find_ambiguous</a></tt><tt class="py-op">(</tt><tt class="py-name">motif</tt><tt class="py-op">)</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">ambig_positions</tt><tt class="py-op">)</tt> </tt>
</div><a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line"> </tt>
<a name="Schema.find_matches"></a><div id="Schema.find_matches-def"><a name="L128"></a><tt class="py-lineno">128</tt> <a class="py-toggle" href="#" id="Schema.find_matches-toggle" onclick="return toggle('Schema.find_matches');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.Schema-class.html#find_matches">find_matches</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">motif</tt><tt class="py-op">,</tt> <tt class="py-param">query</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Schema.find_matches-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Schema.find_matches-expanded"><a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return all non-overlapping motif matches in the query string.</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line"><tt class="py-docstring">        This utilizes the regular expression findall function, and will</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line"><tt class="py-docstring">        return a list of all non-overlapping occurances in query that</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line"><tt class="py-docstring">        match the ambiguous motif.</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">            <tt class="py-name">motif_pattern</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_motif_cache</tt><tt class="py-op">[</tt><tt class="py-name">motif</tt><tt class="py-op">]</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line">            <tt class="py-name">motif_pattern</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-45" class="py-name" targets="Method Bio.NeuralNetwork.Gene.Schema.Schema.encode_motif()=Bio.NeuralNetwork.Gene.Schema.Schema-class.html#encode_motif"><a title="Bio.NeuralNetwork.Gene.Schema.Schema.encode_motif" class="py-name" href="#" onclick="return doclink('link-45', 'encode_motif', 'link-45');">encode_motif</a></tt><tt class="py-op">(</tt><tt class="py-name">motif</tt><tt class="py-op">)</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_motif_cache</tt><tt class="py-op">[</tt><tt class="py-name">motif</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">motif_pattern</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line"> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">motif_pattern</tt><tt class="py-op">.</tt><tt class="py-name">findall</tt><tt class="py-op">(</tt><tt id="link-46" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.query()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#query,Variable Bio.SearchIO._model.hsp.HSP.query=Bio.SearchIO._model.hsp.HSP-class.html#query,Variable Bio.SearchIO._model.hsp.HSPFragment.query=Bio.SearchIO._model.hsp.HSPFragment-class.html#query,Module Bio.SearchIO._model.query=Bio.SearchIO._model.query-module.html"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-46', 'query', 'link-46');">query</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line"> </tt>
<a name="Schema.num_matches"></a><div id="Schema.num_matches-def"><a name="L143"></a><tt class="py-lineno">143</tt> <a class="py-toggle" href="#" id="Schema.num_matches-toggle" onclick="return toggle('Schema.num_matches');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.Schema-class.html#num_matches">num_matches</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">motif</tt><tt class="py-op">,</tt> <tt class="py-param">query</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Schema.num_matches-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Schema.num_matches-expanded"><a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line">        <tt class="py-docstring">"""Find the number of non-overlapping times motif occurs in query.</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">        <tt class="py-name">all_matches</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-47" class="py-name" targets="Method Bio.NeuralNetwork.Gene.Schema.Schema.find_matches()=Bio.NeuralNetwork.Gene.Schema.Schema-class.html#find_matches"><a title="Bio.NeuralNetwork.Gene.Schema.Schema.find_matches" class="py-name" href="#" onclick="return doclink('link-47', 'find_matches', 'link-47');">find_matches</a></tt><tt class="py-op">(</tt><tt class="py-name">motif</tt><tt class="py-op">,</tt> <tt id="link-48" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-48', 'query', 'link-46');">query</a></tt><tt class="py-op">)</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">all_matches</tt><tt class="py-op">)</tt> </tt>
</div><a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line"> </tt>
<a name="Schema.all_unambiguous"></a><div id="Schema.all_unambiguous-def"><a name="L149"></a><tt class="py-lineno">149</tt> <a class="py-toggle" href="#" id="Schema.all_unambiguous-toggle" onclick="return toggle('Schema.all_unambiguous');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.Schema-class.html#all_unambiguous">all_unambiguous</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Schema.all_unambiguous-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Schema.all_unambiguous-expanded"><a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a listing of all unambiguous letters allowed in motifs.</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">        <tt class="py-name">all_letters</tt> <tt class="py-op">=</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_ambiguity_info</tt><tt class="py-op">)</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">        <tt class="py-name">unambig_letters</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line"> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">letter</tt> <tt class="py-keyword">in</tt> <tt class="py-name">all_letters</tt><tt class="py-op">:</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">            <tt class="py-name">possible_matches</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_ambiguity_info</tt><tt class="py-op">[</tt><tt class="py-name">letter</tt><tt class="py-op">]</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">possible_matches</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">                <tt class="py-name">unambig_letters</tt><tt class="py-op">.</tt><tt id="link-49" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-49', 'append', 'link-43');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">letter</tt><tt class="py-op">)</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line"> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">unambig_letters</tt> </tt>
</div></div><a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line"> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line"><tt class="py-comment"># --- helper classes and functions for the default SchemaFinder</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line"> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line"><tt class="py-comment"># -- Alphabets</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line"> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line"> </tt>
<a name="SchemaDNAAlphabet"></a><div id="SchemaDNAAlphabet-def"><a name="L167"></a><tt class="py-lineno">167</tt> <a class="py-toggle" href="#" id="SchemaDNAAlphabet-toggle" onclick="return toggle('SchemaDNAAlphabet');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet-class.html">SchemaDNAAlphabet</a><tt class="py-op">(</tt><tt class="py-base-class">Alphabet</tt><tt class="py-op">.</tt><tt class="py-base-class">Alphabet</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SchemaDNAAlphabet-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="SchemaDNAAlphabet-expanded"><a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">    <tt class="py-docstring">"""Alphabet of a simple Schema for DNA sequences.</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line"><tt class="py-docstring">    This defines a simple alphabet for DNA sequences that has a single</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line"><tt class="py-docstring">    character which can match any other character.</tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line"><tt class="py-docstring">    o G,A,T,C - The standard unambiguous DNA alphabet.</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line"><tt class="py-docstring">    o * - Any letter</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">    <tt id="link-50" class="py-name" targets="Variable Bio.Alphabet.Alphabet.letters=Bio.Alphabet.Alphabet-class.html#letters,Variable Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters=Bio.Alphabet.IUPAC.ExtendedIUPACDNA-class.html#letters,Variable Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters=Bio.Alphabet.IUPAC.ExtendedIUPACProtein-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters=Bio.Alphabet.IUPAC.IUPACAmbiguousDNA-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters=Bio.Alphabet.IUPAC.IUPACAmbiguousRNA-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACProtein.letters=Bio.Alphabet.IUPAC.IUPACProtein-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters=Bio.Alphabet.IUPAC.IUPACUnambiguousDNA-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters=Bio.Alphabet.IUPAC.IUPACUnambiguousRNA-class.html#letters,Variable Bio.Alphabet.Reduced.HPModel.letters=Bio.Alphabet.Reduced.HPModel-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy10.letters=Bio.Alphabet.Reduced.Murphy10-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy15.letters=Bio.Alphabet.Reduced.Murphy15-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy4.letters=Bio.Alphabet.Reduced.Murphy4-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy8.letters=Bio.Alphabet.Reduced.Murphy8-class.html#letters,Variable Bio.Alphabet.Reduced.PC5.letters=Bio.Alphabet.Reduced.PC5-class.html#letters,Variable Bio.Alphabet.SecondaryStructure.letters=Bio.Alphabet.SecondaryStructure-class.html#letters,Variable Bio.Alphabet.SingleLetterAlphabet.letters=Bio.Alphabet.SingleLetterAlphabet-class.html#letters,Variable Bio.Alphabet.ThreeLetterProtein.letters=Bio.Alphabet.ThreeLetterProtein-class.html#letters,Variable Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.letters=Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet-class.html#letters,Variable Bio.codonalign.codonalphabet.CodonAlphabet.letters=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#letters"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-50', 'letters', 'link-50');">letters</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">"G"</tt><tt class="py-op">,</tt> <tt class="py-string">"A"</tt><tt class="py-op">,</tt> <tt class="py-string">"T"</tt><tt class="py-op">,</tt> <tt class="py-string">"C"</tt><tt class="py-op">,</tt> <tt class="py-string">"*"</tt><tt class="py-op">]</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line"> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">    <tt id="link-51" class="py-name" targets="Variable Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.alphabet_matches=Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet-class.html#alphabet_matches"><a title="Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.alphabet_matches" class="py-name" href="#" onclick="return doclink('link-51', 'alphabet_matches', 'link-51');">alphabet_matches</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">"G"</tt><tt class="py-op">:</tt> <tt class="py-string">"G"</tt><tt class="py-op">,</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">                        <tt class="py-string">"A"</tt><tt class="py-op">:</tt> <tt class="py-string">"A"</tt><tt class="py-op">,</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">                        <tt class="py-string">"T"</tt><tt class="py-op">:</tt> <tt class="py-string">"T"</tt><tt class="py-op">,</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">                        <tt class="py-string">"C"</tt><tt class="py-op">:</tt> <tt class="py-string">"C"</tt><tt class="py-op">,</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">                        <tt class="py-string">"*"</tt><tt class="py-op">:</tt> <tt class="py-string">"GATC"</tt><tt class="py-op">}</tt> </tt>
</div><a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line"> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line"><tt class="py-comment"># -- GA schema finder</tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line"> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line"> </tt>
<a name="GeneticAlgorithmFinder"></a><div id="GeneticAlgorithmFinder-def"><a name="L188"></a><tt class="py-lineno">188</tt> <a class="py-toggle" href="#" id="GeneticAlgorithmFinder-toggle" onclick="return toggle('GeneticAlgorithmFinder');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder-class.html">GeneticAlgorithmFinder</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GeneticAlgorithmFinder-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="GeneticAlgorithmFinder-expanded"><a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">    <tt class="py-docstring">"""Find schemas using a genetic algorithm approach.</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line"><tt class="py-docstring">    This approach to finding schema uses Genetic Algorithms to evolve</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line"><tt class="py-docstring">    a set of schema and find the best schema for a specific set of</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line"><tt class="py-docstring">    records.</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line"><tt class="py-docstring">    The 'default' finder searches for ambiguous DNA elements. This</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line"><tt class="py-docstring">    can be overridden easily by creating a GeneticAlgorithmFinder</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line"><tt class="py-docstring">    with a different alphabet.</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="GeneticAlgorithmFinder.__init__"></a><div id="GeneticAlgorithmFinder.__init__-def"><a name="L199"></a><tt class="py-lineno">199</tt> <a class="py-toggle" href="#" id="GeneticAlgorithmFinder.__init__-toggle" onclick="return toggle('GeneticAlgorithmFinder.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt id="link-52" class="py-name" targets="Class Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet=Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet-class.html"><a title="Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet" class="py-name" href="#" onclick="return doclink('link-52', 'SchemaDNAAlphabet', 'link-52');">SchemaDNAAlphabet</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GeneticAlgorithmFinder.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="GeneticAlgorithmFinder.__init__-expanded"><a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize a finder to get schemas using Genetic Algorithms.</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line"><tt class="py-docstring">        o alphabet -- The alphabet which specifies the contents of the</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line"><tt class="py-docstring">        schemas we'll be generating. This alphabet must contain the</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line"><tt class="py-docstring">        attribute 'alphabet_matches', which is a dictionary specifying</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line"><tt class="py-docstring">        the potential ambiguities of each letter in the alphabet. These</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line"><tt class="py-docstring">        ambiguities will be used in building up the schema.</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-53" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.alphabet=Bio.SearchIO._model.hsp.HSP-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSPFragment.alphabet=Bio.SearchIO._model.hsp.HSPFragment-class.html#alphabet"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-53', 'alphabet', 'link-53');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-54" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-54', 'alphabet', 'link-53');">alphabet</a></tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line"> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">initial_population</tt> <tt class="py-op">=</tt> <tt class="py-number">500</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">min_generations</tt> <tt class="py-op">=</tt> <tt class="py-number">10</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line"> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-55" class="py-name" targets="Method Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder._set_up_genetic_algorithm()=Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder-class.html#_set_up_genetic_algorithm"><a title="Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder._set_up_genetic_algorithm" class="py-name" href="#" onclick="return doclink('link-55', '_set_up_genetic_algorithm', 'link-55');">_set_up_genetic_algorithm</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line"> </tt>
<a name="GeneticAlgorithmFinder._set_up_genetic_algorithm"></a><div id="GeneticAlgorithmFinder._set_up_genetic_algorithm-def"><a name="L217"></a><tt class="py-lineno">217</tt> <a class="py-toggle" href="#" id="GeneticAlgorithmFinder._set_up_genetic_algorithm-toggle" onclick="return toggle('GeneticAlgorithmFinder._set_up_genetic_algorithm');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder-class.html#_set_up_genetic_algorithm">_set_up_genetic_algorithm</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GeneticAlgorithmFinder._set_up_genetic_algorithm-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="GeneticAlgorithmFinder._set_up_genetic_algorithm-expanded"><a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line">        <tt class="py-docstring">"""Overrideable function to set up the genetic algorithm parameters.</tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line"><tt class="py-docstring">        This functions sole job is to set up the different genetic</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line"><tt class="py-docstring">        algorithm functionality. Since this can be quite complicated, this</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line"><tt class="py-docstring">        allows cusotmizablity of all of the parameters. If you want to</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line"><tt class="py-docstring">        customize specially, you can inherit from this class and override</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line"><tt class="py-docstring">        this function.</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">motif_generator</tt> <tt class="py-op">=</tt> <tt id="link-56" class="py-name" targets="Class Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator=Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator-class.html"><a title="Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator" class="py-name" href="#" onclick="return doclink('link-56', 'RandomMotifGenerator', 'link-56');">RandomMotifGenerator</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-57" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-57', 'alphabet', 'link-53');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line"> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">mutator</tt> <tt class="py-op">=</tt> <tt id="link-58" class="py-name"><a title="Bio.GA.Mutation.Simple.SinglePositionMutation" class="py-name" href="#" onclick="return doclink('link-58', 'SinglePositionMutation', 'link-21');">SinglePositionMutation</a></tt><tt class="py-op">(</tt><tt class="py-name">mutation_rate</tt><tt class="py-op">=</tt><tt class="py-number">0.1</tt><tt class="py-op">)</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">crossover</tt> <tt class="py-op">=</tt> <tt id="link-59" class="py-name"><a title="Bio.GA.Crossover.Point.SinglePointCrossover" class="py-name" href="#" onclick="return doclink('link-59', 'SinglePointCrossover', 'link-26');">SinglePointCrossover</a></tt><tt class="py-op">(</tt><tt class="py-name">crossover_prob</tt><tt class="py-op">=</tt><tt class="py-number">0.25</tt><tt class="py-op">)</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-60" class="py-name" targets="Method Bio.GA.Repair.Stabilizing.AmbiguousRepair.repair()=Bio.GA.Repair.Stabilizing.AmbiguousRepair-class.html#repair"><a title="Bio.GA.Repair.Stabilizing.AmbiguousRepair.repair" class="py-name" href="#" onclick="return doclink('link-60', 'repair', 'link-60');">repair</a></tt> <tt class="py-op">=</tt> <tt id="link-61" class="py-name"><a title="Bio.GA.Repair.Stabilizing.AmbiguousRepair" class="py-name" href="#" onclick="return doclink('link-61', 'AmbiguousRepair', 'link-31');">AmbiguousRepair</a></tt><tt class="py-op">(</tt><tt id="link-62" class="py-name" targets="Module Bio.NeuralNetwork.Gene.Schema=Bio.NeuralNetwork.Gene.Schema-module.html,Class Bio.NeuralNetwork.Gene.Schema.Schema=Bio.NeuralNetwork.Gene.Schema.Schema-class.html"><a title="Bio.NeuralNetwork.Gene.Schema
Bio.NeuralNetwork.Gene.Schema.Schema" class="py-name" href="#" onclick="return doclink('link-62', 'Schema', 'link-62');">Schema</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-63" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-63', 'alphabet', 'link-53');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-64" class="py-name"><a title="Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.alphabet_matches" class="py-name" href="#" onclick="return doclink('link-64', 'alphabet_matches', 'link-51');">alphabet_matches</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line">                                      <tt class="py-number">4</tt><tt class="py-op">)</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">base_selector</tt> <tt class="py-op">=</tt> <tt id="link-65" class="py-name"><a title="Bio.GA.Selection.Tournament.TournamentSelection" class="py-name" href="#" onclick="return doclink('link-65', 'TournamentSelection', 'link-36');">TournamentSelection</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">mutator</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">crossover</tt><tt class="py-op">,</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">                                                 <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-66" class="py-name"><a title="Bio.GA.Repair.Stabilizing.AmbiguousRepair.repair" class="py-name" href="#" onclick="return doclink('link-66', 'repair', 'link-60');">repair</a></tt><tt class="py-op">,</tt> <tt class="py-number">2</tt><tt class="py-op">)</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">selector</tt> <tt class="py-op">=</tt> <tt id="link-67" class="py-name"><a title="Bio.GA.Selection.Diversity.DiversitySelection" class="py-name" href="#" onclick="return doclink('link-67', 'DiversitySelection', 'link-41');">DiversitySelection</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">base_selector</tt><tt class="py-op">,</tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line">                                           <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">motif_generator</tt><tt class="py-op">.</tt><tt id="link-68" class="py-name" targets="Method Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator.random_motif()=Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator-class.html#random_motif"><a title="Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator.random_motif" class="py-name" href="#" onclick="return doclink('link-68', 'random_motif', 'link-68');">random_motif</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line"> </tt>
<a name="GeneticAlgorithmFinder.find_schemas"></a><div id="GeneticAlgorithmFinder.find_schemas-def"><a name="L237"></a><tt class="py-lineno">237</tt> <a class="py-toggle" href="#" id="GeneticAlgorithmFinder.find_schemas-toggle" onclick="return toggle('GeneticAlgorithmFinder.find_schemas');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder-class.html#find_schemas">find_schemas</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">fitness</tt><tt class="py-op">,</tt> <tt class="py-param">num_schemas</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GeneticAlgorithmFinder.find_schemas-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="GeneticAlgorithmFinder.find_schemas-expanded"><a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line">        <tt class="py-docstring">"""Find the given number of unique schemas using a genetic algorithm</tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line"><tt class="py-docstring">        o fitness - A callable object (ie. function) which will evaluate</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line"><tt class="py-docstring">        the fitness of a motif.</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line"><tt class="py-docstring">        o num_schemas - The number of unique schemas with good fitness</tt> </tt>
<a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line"><tt class="py-docstring">        that we want to generate.</tt> </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line">        <tt class="py-name">start_population</tt> <tt class="py-op">=</tt> \ </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line">           <tt id="link-69" class="py-name"><a title="Bio.GA.Organism
Bio.GA.Organism.Organism
Bio.Graphics.BasicChromosome.Organism" class="py-name" href="#" onclick="return doclink('link-69', 'Organism', 'link-12');">Organism</a></tt><tt class="py-op">.</tt><tt id="link-70" class="py-name" targets="Function Bio.GA.Organism.function_population()=Bio.GA.Organism-module.html#function_population"><a title="Bio.GA.Organism.function_population" class="py-name" href="#" onclick="return doclink('link-70', 'function_population', 'link-70');">function_population</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">motif_generator</tt><tt class="py-op">.</tt><tt id="link-71" class="py-name"><a title="Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator.random_motif" class="py-name" href="#" onclick="return doclink('link-71', 'random_motif', 'link-68');">random_motif</a></tt><tt class="py-op">,</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line">                                        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">initial_population</tt><tt class="py-op">,</tt> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line">                                        <tt class="py-name">fitness</tt><tt class="py-op">)</tt> </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line">        <tt class="py-name">finisher</tt> <tt class="py-op">=</tt> <tt id="link-72" class="py-name" targets="Class Bio.NeuralNetwork.Gene.Schema.SimpleFinisher=Bio.NeuralNetwork.Gene.Schema.SimpleFinisher-class.html"><a title="Bio.NeuralNetwork.Gene.Schema.SimpleFinisher" class="py-name" href="#" onclick="return doclink('link-72', 'SimpleFinisher', 'link-72');">SimpleFinisher</a></tt><tt class="py-op">(</tt><tt class="py-name">num_schemas</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">min_generations</tt><tt class="py-op">)</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line"> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line">        <tt class="py-comment"># set up the evolver and do the evolution</tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line">        <tt class="py-name">evolver</tt> <tt class="py-op">=</tt> <tt id="link-73" class="py-name"><a title="Bio.GA.Evolver.GenerationEvolver" class="py-name" href="#" onclick="return doclink('link-73', 'GenerationEvolver', 'link-16');">GenerationEvolver</a></tt><tt class="py-op">(</tt><tt class="py-name">start_population</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">selector</tt><tt class="py-op">)</tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line">        <tt class="py-name">evolved_pop</tt> <tt class="py-op">=</tt> <tt class="py-name">evolver</tt><tt class="py-op">.</tt><tt id="link-74" class="py-name" targets="Method Bio.GA.Evolver.GenerationEvolver.evolve()=Bio.GA.Evolver.GenerationEvolver-class.html#evolve"><a title="Bio.GA.Evolver.GenerationEvolver.evolve" class="py-name" href="#" onclick="return doclink('link-74', 'evolve', 'link-74');">evolve</a></tt><tt class="py-op">(</tt><tt class="py-name">finisher</tt><tt class="py-op">.</tt><tt id="link-75" class="py-name" targets="Method Bio.NeuralNetwork.Gene.Schema.SimpleFinisher.is_finished()=Bio.NeuralNetwork.Gene.Schema.SimpleFinisher-class.html#is_finished"><a title="Bio.NeuralNetwork.Gene.Schema.SimpleFinisher.is_finished" class="py-name" href="#" onclick="return doclink('link-75', 'is_finished', 'link-75');">is_finished</a></tt><tt class="py-op">)</tt> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line"> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line">        <tt class="py-comment"># convert the evolved population into a PatternRepository</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line">        <tt class="py-name">schema_info</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">org</tt> <tt class="py-keyword">in</tt> <tt class="py-name">evolved_pop</tt><tt class="py-op">:</tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line">            <tt class="py-comment"># convert the Genome from a MutableSeq to a Seq so that</tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line">            <tt class="py-comment"># the schemas are just strings (and not array("c")s)</tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line">            <tt class="py-name">seq_genome</tt> <tt class="py-op">=</tt> <tt class="py-name">org</tt><tt class="py-op">.</tt><tt class="py-name">genome</tt><tt class="py-op">.</tt><tt id="link-76" class="py-name" targets="Method Bio.Seq.MutableSeq.toseq()=Bio.Seq.MutableSeq-class.html#toseq,Method BioSQL.BioSeq.DBSeq.toseq()=BioSQL.BioSeq.DBSeq-class.html#toseq"><a title="Bio.Seq.MutableSeq.toseq
BioSQL.BioSeq.DBSeq.toseq" class="py-name" href="#" onclick="return doclink('link-76', 'toseq', 'link-76');">toseq</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line">            <tt class="py-name">schema_info</tt><tt class="py-op">[</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">seq_genome</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">org</tt><tt class="py-op">.</tt><tt class="py-name">fitness</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line"> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-77" class="py-name"><a title="Bio.NeuralNetwork.Gene.Pattern.PatternRepository" class="py-name" href="#" onclick="return doclink('link-77', 'PatternRepository', 'link-9');">PatternRepository</a></tt><tt class="py-op">(</tt><tt class="py-name">schema_info</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line"> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line"><tt class="py-comment"># -- fitness classes</tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line"> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line"> </tt>
<a name="DifferentialSchemaFitness"></a><div id="DifferentialSchemaFitness-def"><a name="L271"></a><tt class="py-lineno">271</tt> <a class="py-toggle" href="#" id="DifferentialSchemaFitness-toggle" onclick="return toggle('DifferentialSchemaFitness');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness-class.html">DifferentialSchemaFitness</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DifferentialSchemaFitness-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="DifferentialSchemaFitness-expanded"><a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line">    <tt class="py-docstring">"""Calculate fitness for schemas that differentiate between sequences.</tt> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="DifferentialSchemaFitness.__init__"></a><div id="DifferentialSchemaFitness.__init__-def"><a name="L274"></a><tt class="py-lineno">274</tt> <a class="py-toggle" href="#" id="DifferentialSchemaFitness.__init__-toggle" onclick="return toggle('DifferentialSchemaFitness.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">positive_seqs</tt><tt class="py-op">,</tt> <tt class="py-param">negative_seqs</tt><tt class="py-op">,</tt> <tt class="py-param">schema_evaluator</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DifferentialSchemaFitness.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DifferentialSchemaFitness.__init__-expanded"><a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize with different sequences to evaluate</tt> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line"><tt class="py-docstring">        o positive_seq - A list of SeqRecord objects which are the 'positive'</tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line"><tt class="py-docstring">        sequences -- the ones we want to select for.</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line"><tt class="py-docstring">        o negative_seq - A list of SeqRecord objects which are the 'negative'</tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line"><tt class="py-docstring">        sequences that we want to avoid selecting.</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line"><tt class="py-docstring">        o schema_evaluator - An Schema class which can be used to</tt> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line"><tt class="py-docstring">        evaluate find motif matches in sequences.</tt> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_pos_seqs</tt> <tt class="py-op">=</tt> <tt class="py-name">positive_seqs</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_neg_seqs</tt> <tt class="py-op">=</tt> <tt class="py-name">negative_seqs</tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_schema_eval</tt> <tt class="py-op">=</tt> <tt class="py-name">schema_evaluator</tt> </tt>
</div><a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line"> </tt>
<a name="DifferentialSchemaFitness.calculate_fitness"></a><div id="DifferentialSchemaFitness.calculate_fitness-def"><a name="L292"></a><tt class="py-lineno">292</tt> <a class="py-toggle" href="#" id="DifferentialSchemaFitness.calculate_fitness-toggle" onclick="return toggle('DifferentialSchemaFitness.calculate_fitness');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness-class.html#calculate_fitness">calculate_fitness</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">genome</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DifferentialSchemaFitness.calculate_fitness-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DifferentialSchemaFitness.calculate_fitness-expanded"><a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line">        <tt class="py-docstring">"""Calculate the fitness for a given schema.</tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line"><tt class="py-docstring">        Fitness is specified by the number of occurances of the schema in</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line"><tt class="py-docstring">        the positive sequences minus the number of occurances in the</tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line"><tt class="py-docstring">        negative examples.</tt> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line"><tt class="py-docstring">        This fitness is then modified by multiplying by the length of the</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line"><tt class="py-docstring">        schema and then dividing by the number of ambiguous characters in</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line"><tt class="py-docstring">        the schema. This helps select for schema which are longer and have</tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line"><tt class="py-docstring">        less redundancy.</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line">        <tt class="py-comment"># convert the genome into a string</tt> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line">        <tt class="py-name">seq_motif</tt> <tt class="py-op">=</tt> <tt class="py-name">genome</tt><tt class="py-op">.</tt><tt id="link-78" class="py-name"><a title="Bio.Seq.MutableSeq.toseq
BioSQL.BioSeq.DBSeq.toseq" class="py-name" href="#" onclick="return doclink('link-78', 'toseq', 'link-76');">toseq</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line">        <tt class="py-name">motif</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">seq_motif</tt><tt class="py-op">)</tt> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line"> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line">        <tt class="py-comment"># get the counts in the positive examples</tt> </tt>
<a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line">        <tt class="py-name">num_pos</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">seq_record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_pos_seqs</tt><tt class="py-op">:</tt> </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line">            <tt class="py-name">cur_counts</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_schema_eval</tt><tt class="py-op">.</tt><tt id="link-79" class="py-name" targets="Method Bio.NeuralNetwork.Gene.Schema.Schema.num_matches()=Bio.NeuralNetwork.Gene.Schema.Schema-class.html#num_matches"><a title="Bio.NeuralNetwork.Gene.Schema.Schema.num_matches" class="py-name" href="#" onclick="return doclink('link-79', 'num_matches', 'link-79');">num_matches</a></tt><tt class="py-op">(</tt><tt class="py-name">motif</tt><tt class="py-op">,</tt> </tt>
<a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line">                                                      <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">seq_record</tt><tt class="py-op">.</tt><tt id="link-80" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.seq=Bio.SeqRecord.SeqRecord-class.html#seq,Variable BioSQL.BioSeq.DBSeqRecord.seq=BioSQL.BioSeq.DBSeqRecord-class.html#seq"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-80', 'seq', 'link-80');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line">            <tt class="py-name">num_pos</tt> <tt class="py-op">+=</tt> <tt class="py-name">cur_counts</tt> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line"> </tt>
<a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line">        <tt class="py-comment"># get the counts in the negative examples</tt> </tt>
<a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line">        <tt class="py-name">num_neg</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">seq_record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_neg_seqs</tt><tt class="py-op">:</tt> </tt>
<a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">            <tt class="py-name">cur_counts</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_schema_eval</tt><tt class="py-op">.</tt><tt id="link-81" class="py-name"><a title="Bio.NeuralNetwork.Gene.Schema.Schema.num_matches" class="py-name" href="#" onclick="return doclink('link-81', 'num_matches', 'link-79');">num_matches</a></tt><tt class="py-op">(</tt><tt class="py-name">motif</tt><tt class="py-op">,</tt> </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line">                                                      <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">seq_record</tt><tt class="py-op">.</tt><tt id="link-82" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-82', 'seq', 'link-80');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line"> </tt>
<a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line">            <tt class="py-name">num_neg</tt> <tt class="py-op">+=</tt> <tt class="py-name">cur_counts</tt> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line"> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">        <tt id="link-83" class="py-name" targets="Method Bio.NeuralNetwork.Gene.Schema.Schema.num_ambiguous()=Bio.NeuralNetwork.Gene.Schema.Schema-class.html#num_ambiguous"><a title="Bio.NeuralNetwork.Gene.Schema.Schema.num_ambiguous" class="py-name" href="#" onclick="return doclink('link-83', 'num_ambiguous', 'link-83');">num_ambiguous</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_schema_eval</tt><tt class="py-op">.</tt><tt id="link-84" class="py-name"><a title="Bio.NeuralNetwork.Gene.Schema.Schema.num_ambiguous" class="py-name" href="#" onclick="return doclink('link-84', 'num_ambiguous', 'link-83');">num_ambiguous</a></tt><tt class="py-op">(</tt><tt class="py-name">motif</tt><tt class="py-op">)</tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line">        <tt class="py-comment"># weight the ambiguous stuff more highly</tt> </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line">        <tt id="link-85" class="py-name"><a title="Bio.NeuralNetwork.Gene.Schema.Schema.num_ambiguous" class="py-name" href="#" onclick="return doclink('link-85', 'num_ambiguous', 'link-83');">num_ambiguous</a></tt> <tt class="py-op">=</tt> <tt class="py-name">pow</tt><tt class="py-op">(</tt><tt class="py-number">2.0</tt><tt class="py-op">,</tt> <tt id="link-86" class="py-name"><a title="Bio.NeuralNetwork.Gene.Schema.Schema.num_ambiguous" class="py-name" href="#" onclick="return doclink('link-86', 'num_ambiguous', 'link-83');">num_ambiguous</a></tt><tt class="py-op">)</tt> </tt>
<a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">        <tt class="py-comment"># increment num ambiguous to prevent division by zero errors.</tt> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line">        <tt id="link-87" class="py-name"><a title="Bio.NeuralNetwork.Gene.Schema.Schema.num_ambiguous" class="py-name" href="#" onclick="return doclink('link-87', 'num_ambiguous', 'link-83');">num_ambiguous</a></tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line"> </tt>
<a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line">        <tt class="py-name">motif_size</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">motif</tt><tt class="py-op">)</tt> </tt>
<a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line">        <tt class="py-name">motif_size</tt> <tt class="py-op">=</tt> <tt class="py-name">motif_size</tt> <tt class="py-op">*</tt> <tt class="py-number">4.0</tt> </tt>
<a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line"> </tt>
<a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line">        <tt class="py-name">discerning_power</tt> <tt class="py-op">=</tt> <tt class="py-name">num_pos</tt> <tt class="py-op">-</tt> <tt class="py-name">num_neg</tt> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line"> </tt>
<a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line">        <tt class="py-name">diff</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">discerning_power</tt> <tt class="py-op">*</tt> <tt class="py-name">motif_size</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt id="link-88" class="py-name"><a title="Bio.NeuralNetwork.Gene.Schema.Schema.num_ambiguous" class="py-name" href="#" onclick="return doclink('link-88', 'num_ambiguous', 'link-83');">num_ambiguous</a></tt><tt class="py-op">)</tt> </tt>
<a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">diff</tt> </tt>
</div></div><a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line"> </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line"> </tt>
<a name="MostCountSchemaFitness"></a><div id="MostCountSchemaFitness-def"><a name="L338"></a><tt class="py-lineno">338</tt> <a class="py-toggle" href="#" id="MostCountSchemaFitness-toggle" onclick="return toggle('MostCountSchemaFitness');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness-class.html">MostCountSchemaFitness</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MostCountSchemaFitness-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="MostCountSchemaFitness-expanded"><a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line">    <tt class="py-docstring">"""Calculate a fitness giving weight to schemas that match many times.</tt> </tt>
<a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line"><tt class="py-docstring">    This fitness function tries to maximize schemas which are found many</tt> </tt>
<a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line"><tt class="py-docstring">    times in a group of sequences.</tt> </tt>
<a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="MostCountSchemaFitness.__init__"></a><div id="MostCountSchemaFitness.__init__-def"><a name="L344"></a><tt class="py-lineno">344</tt> <a class="py-toggle" href="#" id="MostCountSchemaFitness.__init__-toggle" onclick="return toggle('MostCountSchemaFitness.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">seq_records</tt><tt class="py-op">,</tt> <tt class="py-param">schema_evaluator</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MostCountSchemaFitness.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MostCountSchemaFitness.__init__-expanded"><a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize with sequences to evaluate.</tt> </tt>
<a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L347"></a><tt class="py-lineno">347</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line"><tt class="py-docstring">        o seq_records -- A set of SeqRecord objects which we use to</tt> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line"><tt class="py-docstring">        calculate the fitness.</tt> </tt>
<a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L352"></a><tt class="py-lineno">352</tt>  <tt class="py-line"><tt class="py-docstring">        o schema_evaluator - An Schema class which can be used to</tt> </tt>
<a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line"><tt class="py-docstring">        evaluate find motif matches in sequences.</tt> </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_records</tt> <tt class="py-op">=</tt> <tt class="py-name">seq_records</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_evaluator</tt> <tt class="py-op">=</tt> <tt class="py-name">schema_evaluator</tt> </tt>
</div><a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line"> </tt>
<a name="MostCountSchemaFitness.calculate_fitness"></a><div id="MostCountSchemaFitness.calculate_fitness-def"><a name="L358"></a><tt class="py-lineno">358</tt> <a class="py-toggle" href="#" id="MostCountSchemaFitness.calculate_fitness-toggle" onclick="return toggle('MostCountSchemaFitness.calculate_fitness');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness-class.html#calculate_fitness">calculate_fitness</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">genome</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MostCountSchemaFitness.calculate_fitness-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MostCountSchemaFitness.calculate_fitness-expanded"><a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line">        <tt class="py-docstring">"""Calculate the fitness of a genome based on schema matches.</tt> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line"><tt class="py-docstring">        This bases the fitness of a genome completely on the number of times</tt> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line"><tt class="py-docstring">        it matches in the set of seq_records. Matching more times gives a</tt> </tt>
<a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line"><tt class="py-docstring">        better fitness</tt> </tt>
<a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line">        <tt class="py-comment"># convert the genome into a string</tt> </tt>
<a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line">        <tt class="py-name">seq_motif</tt> <tt class="py-op">=</tt> <tt class="py-name">genome</tt><tt class="py-op">.</tt><tt id="link-89" class="py-name"><a title="Bio.Seq.MutableSeq.toseq
BioSQL.BioSeq.DBSeq.toseq" class="py-name" href="#" onclick="return doclink('link-89', 'toseq', 'link-76');">toseq</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line">        <tt class="py-name">motif</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">seq_motif</tt><tt class="py-op">)</tt> </tt>
<a name="L368"></a><tt class="py-lineno">368</tt>  <tt class="py-line"> </tt>
<a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line">        <tt class="py-comment"># find the number of times the genome matches</tt> </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line">        <tt class="py-name">num_times</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">seq_record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_records</tt><tt class="py-op">:</tt> </tt>
<a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line">            <tt class="py-name">cur_counts</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_evaluator</tt><tt class="py-op">.</tt><tt id="link-90" class="py-name"><a title="Bio.NeuralNetwork.Gene.Schema.Schema.num_matches" class="py-name" href="#" onclick="return doclink('link-90', 'num_matches', 'link-79');">num_matches</a></tt><tt class="py-op">(</tt><tt class="py-name">motif</tt><tt class="py-op">,</tt> </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line">                                                     <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">seq_record</tt><tt class="py-op">.</tt><tt id="link-91" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-91', 'seq', 'link-80');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line">            <tt class="py-name">num_times</tt> <tt class="py-op">+=</tt> <tt class="py-name">cur_counts</tt> </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line"> </tt>
<a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">num_times</tt> </tt>
</div></div><a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line"> </tt>
<a name="L378"></a><tt class="py-lineno">378</tt>  <tt class="py-line"> </tt>
<a name="L379"></a><tt class="py-lineno">379</tt>  <tt class="py-line"><tt class="py-comment"># -- Helper classes</tt> </tt>
<a name="RandomMotifGenerator"></a><div id="RandomMotifGenerator-def"><a name="L380"></a><tt class="py-lineno">380</tt> <a class="py-toggle" href="#" id="RandomMotifGenerator-toggle" onclick="return toggle('RandomMotifGenerator');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator-class.html">RandomMotifGenerator</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RandomMotifGenerator-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="RandomMotifGenerator-expanded"><a name="L381"></a><tt class="py-lineno">381</tt>  <tt class="py-line">    <tt class="py-docstring">"""Generate a random motif within given parameters.</tt> </tt>
<a name="L382"></a><tt class="py-lineno">382</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="RandomMotifGenerator.__init__"></a><div id="RandomMotifGenerator.__init__-def"><a name="L383"></a><tt class="py-lineno">383</tt> <a class="py-toggle" href="#" id="RandomMotifGenerator.__init__-toggle" onclick="return toggle('RandomMotifGenerator.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">,</tt> <tt class="py-param">min_size</tt><tt class="py-op">=</tt><tt class="py-number">12</tt><tt class="py-op">,</tt> <tt class="py-param">max_size</tt><tt class="py-op">=</tt><tt class="py-number">17</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RandomMotifGenerator.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="RandomMotifGenerator.__init__-expanded"><a name="L384"></a><tt class="py-lineno">384</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize with the motif parameters.</tt> </tt>
<a name="L385"></a><tt class="py-lineno">385</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L386"></a><tt class="py-lineno">386</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L387"></a><tt class="py-lineno">387</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L388"></a><tt class="py-lineno">388</tt>  <tt class="py-line"><tt class="py-docstring">        o alphabet - An alphabet specifying what letters can be inserted in</tt> </tt>
<a name="L389"></a><tt class="py-lineno">389</tt>  <tt class="py-line"><tt class="py-docstring">        a motif.</tt> </tt>
<a name="L390"></a><tt class="py-lineno">390</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L391"></a><tt class="py-lineno">391</tt>  <tt class="py-line"><tt class="py-docstring">        o min_size, max_size - Specify the range of sizes for motifs.</tt> </tt>
<a name="L392"></a><tt class="py-lineno">392</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L393"></a><tt class="py-lineno">393</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_alphabet</tt> <tt class="py-op">=</tt> <tt id="link-92" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-92', 'alphabet', 'link-53');">alphabet</a></tt> </tt>
<a name="L394"></a><tt class="py-lineno">394</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_min_size</tt> <tt class="py-op">=</tt> <tt class="py-name">min_size</tt> </tt>
<a name="L395"></a><tt class="py-lineno">395</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_max_size</tt> <tt class="py-op">=</tt> <tt class="py-name">max_size</tt> </tt>
</div><a name="L396"></a><tt class="py-lineno">396</tt>  <tt class="py-line"> </tt>
<a name="RandomMotifGenerator.random_motif"></a><div id="RandomMotifGenerator.random_motif-def"><a name="L397"></a><tt class="py-lineno">397</tt> <a class="py-toggle" href="#" id="RandomMotifGenerator.random_motif-toggle" onclick="return toggle('RandomMotifGenerator.random_motif');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator-class.html#random_motif">random_motif</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RandomMotifGenerator.random_motif-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="RandomMotifGenerator.random_motif-expanded"><a name="L398"></a><tt class="py-lineno">398</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a random motif within the given parameters.</tt> </tt>
<a name="L399"></a><tt class="py-lineno">399</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L400"></a><tt class="py-lineno">400</tt>  <tt class="py-line"><tt class="py-docstring">        This returns a single motif string with letters from the given</tt> </tt>
<a name="L401"></a><tt class="py-lineno">401</tt>  <tt class="py-line"><tt class="py-docstring">        alphabet. The size of the motif will be randomly chosen between</tt> </tt>
<a name="L402"></a><tt class="py-lineno">402</tt>  <tt class="py-line"><tt class="py-docstring">        max_size and min_size.</tt> </tt>
<a name="L403"></a><tt class="py-lineno">403</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L404"></a><tt class="py-lineno">404</tt>  <tt class="py-line">        <tt class="py-name">motif_size</tt> <tt class="py-op">=</tt> <tt class="py-name">random</tt><tt class="py-op">.</tt><tt class="py-name">randrange</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_min_size</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_max_size</tt><tt class="py-op">)</tt> </tt>
<a name="L405"></a><tt class="py-lineno">405</tt>  <tt class="py-line"> </tt>
<a name="L406"></a><tt class="py-lineno">406</tt>  <tt class="py-line">        <tt class="py-name">motif</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L407"></a><tt class="py-lineno">407</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">letter_num</tt> <tt class="py-keyword">in</tt> <tt id="link-93" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-93', 'range', 'link-2');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">motif_size</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L408"></a><tt class="py-lineno">408</tt>  <tt class="py-line">            <tt class="py-name">cur_letter</tt> <tt class="py-op">=</tt> <tt class="py-name">random</tt><tt class="py-op">.</tt><tt class="py-name">choice</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_alphabet</tt><tt class="py-op">.</tt><tt id="link-94" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-94', 'letters', 'link-50');">letters</a></tt><tt class="py-op">)</tt> </tt>
<a name="L409"></a><tt class="py-lineno">409</tt>  <tt class="py-line">            <tt class="py-name">motif</tt> <tt class="py-op">+=</tt> <tt class="py-name">cur_letter</tt> </tt>
<a name="L410"></a><tt class="py-lineno">410</tt>  <tt class="py-line"> </tt>
<a name="L411"></a><tt class="py-lineno">411</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-95" class="py-name"><a title="Bio.Seq.MutableSeq" class="py-name" href="#" onclick="return doclink('link-95', 'MutableSeq', 'link-7');">MutableSeq</a></tt><tt class="py-op">(</tt><tt class="py-name">motif</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_alphabet</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L412"></a><tt class="py-lineno">412</tt>  <tt class="py-line"> </tt>
<a name="L413"></a><tt class="py-lineno">413</tt>  <tt class="py-line"> </tt>
<a name="SimpleFinisher"></a><div id="SimpleFinisher-def"><a name="L414"></a><tt class="py-lineno">414</tt> <a class="py-toggle" href="#" id="SimpleFinisher-toggle" onclick="return toggle('SimpleFinisher');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.SimpleFinisher-class.html">SimpleFinisher</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SimpleFinisher-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="SimpleFinisher-expanded"><a name="L415"></a><tt class="py-lineno">415</tt>  <tt class="py-line">    <tt class="py-docstring">"""Determine when we are done evolving motifs.</tt> </tt>
<a name="L416"></a><tt class="py-lineno">416</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L417"></a><tt class="py-lineno">417</tt>  <tt class="py-line"><tt class="py-docstring">    This takes the very simple approach of halting evolution when the</tt> </tt>
<a name="L418"></a><tt class="py-lineno">418</tt>  <tt class="py-line"><tt class="py-docstring">    GA has proceeded for a specified number of generations and has</tt> </tt>
<a name="L419"></a><tt class="py-lineno">419</tt>  <tt class="py-line"><tt class="py-docstring">    a given number of unique schema with positive fitness.</tt> </tt>
<a name="L420"></a><tt class="py-lineno">420</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="SimpleFinisher.__init__"></a><div id="SimpleFinisher.__init__-def"><a name="L421"></a><tt class="py-lineno">421</tt> <a class="py-toggle" href="#" id="SimpleFinisher.__init__-toggle" onclick="return toggle('SimpleFinisher.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.SimpleFinisher-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">num_schemas</tt><tt class="py-op">,</tt> <tt class="py-param">min_generations</tt><tt class="py-op">=</tt><tt class="py-number">100</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SimpleFinisher.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SimpleFinisher.__init__-expanded"><a name="L422"></a><tt class="py-lineno">422</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the finisher with its parameters.</tt> </tt>
<a name="L423"></a><tt class="py-lineno">423</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L424"></a><tt class="py-lineno">424</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L425"></a><tt class="py-lineno">425</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L426"></a><tt class="py-lineno">426</tt>  <tt class="py-line"><tt class="py-docstring">        o num_schemas -- the number of useful (positive fitness) schemas</tt> </tt>
<a name="L427"></a><tt class="py-lineno">427</tt>  <tt class="py-line"><tt class="py-docstring">        we want to generation</tt> </tt>
<a name="L428"></a><tt class="py-lineno">428</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L429"></a><tt class="py-lineno">429</tt>  <tt class="py-line"><tt class="py-docstring">        o min_generations -- The minimum number of generations to allow</tt> </tt>
<a name="L430"></a><tt class="py-lineno">430</tt>  <tt class="py-line"><tt class="py-docstring">        the GA to proceed.</tt> </tt>
<a name="L431"></a><tt class="py-lineno">431</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L432"></a><tt class="py-lineno">432</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">num_generations</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L433"></a><tt class="py-lineno">433</tt>  <tt class="py-line"> </tt>
<a name="L434"></a><tt class="py-lineno">434</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">num_schemas</tt> <tt class="py-op">=</tt> <tt class="py-name">num_schemas</tt> </tt>
<a name="L435"></a><tt class="py-lineno">435</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">min_generations</tt> <tt class="py-op">=</tt> <tt class="py-name">min_generations</tt> </tt>
</div><a name="L436"></a><tt class="py-lineno">436</tt>  <tt class="py-line"> </tt>
<a name="SimpleFinisher.is_finished"></a><div id="SimpleFinisher.is_finished-def"><a name="L437"></a><tt class="py-lineno">437</tt> <a class="py-toggle" href="#" id="SimpleFinisher.is_finished-toggle" onclick="return toggle('SimpleFinisher.is_finished');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.SimpleFinisher-class.html#is_finished">is_finished</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">organisms</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SimpleFinisher.is_finished-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SimpleFinisher.is_finished-expanded"><a name="L438"></a><tt class="py-lineno">438</tt>  <tt class="py-line">        <tt class="py-docstring">"""Determine when we can stop evolving the population.</tt> </tt>
<a name="L439"></a><tt class="py-lineno">439</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L440"></a><tt class="py-lineno">440</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">num_generations</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L441"></a><tt class="py-lineno">441</tt>  <tt class="py-line">        <tt class="py-comment"># print "generation %s" % self.num_generations</tt> </tt>
<a name="L442"></a><tt class="py-lineno">442</tt>  <tt class="py-line"> </tt>
<a name="L443"></a><tt class="py-lineno">443</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">num_generations</tt> <tt class="py-op">&gt;=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">min_generations</tt><tt class="py-op">:</tt> </tt>
<a name="L444"></a><tt class="py-lineno">444</tt>  <tt class="py-line">            <tt class="py-name">all_seqs</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L445"></a><tt class="py-lineno">445</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">org</tt> <tt class="py-keyword">in</tt> <tt class="py-name">organisms</tt><tt class="py-op">:</tt> </tt>
<a name="L446"></a><tt class="py-lineno">446</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">org</tt><tt class="py-op">.</tt><tt class="py-name">fitness</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L447"></a><tt class="py-lineno">447</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">org</tt><tt class="py-op">.</tt><tt class="py-name">genome</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">all_seqs</tt><tt class="py-op">:</tt> </tt>
<a name="L448"></a><tt class="py-lineno">448</tt>  <tt class="py-line">                        <tt class="py-name">all_seqs</tt><tt class="py-op">.</tt><tt id="link-96" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-96', 'append', 'link-43');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">org</tt><tt class="py-op">.</tt><tt class="py-name">genome</tt><tt class="py-op">)</tt> </tt>
<a name="L449"></a><tt class="py-lineno">449</tt>  <tt class="py-line"> </tt>
<a name="L450"></a><tt class="py-lineno">450</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">all_seqs</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">num_schemas</tt><tt class="py-op">:</tt> </tt>
<a name="L451"></a><tt class="py-lineno">451</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-number">1</tt> </tt>
<a name="L452"></a><tt class="py-lineno">452</tt>  <tt class="py-line"> </tt>
<a name="L453"></a><tt class="py-lineno">453</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-number">0</tt> </tt>
</div></div><a name="L454"></a><tt class="py-lineno">454</tt>  <tt class="py-line"><tt class="py-comment"># ---</tt> </tt>
<a name="L455"></a><tt class="py-lineno">455</tt>  <tt class="py-line"> </tt>
<a name="L456"></a><tt class="py-lineno">456</tt>  <tt class="py-line"> </tt>
<a name="SchemaFinder"></a><div id="SchemaFinder-def"><a name="L457"></a><tt class="py-lineno">457</tt> <a class="py-toggle" href="#" id="SchemaFinder-toggle" onclick="return toggle('SchemaFinder');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.SchemaFinder-class.html">SchemaFinder</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SchemaFinder-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="SchemaFinder-expanded"><a name="L458"></a><tt class="py-lineno">458</tt>  <tt class="py-line">    <tt class="py-docstring">"""Find schema in a set of sequences using a genetic algorithm approach.</tt> </tt>
<a name="L459"></a><tt class="py-lineno">459</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L460"></a><tt class="py-lineno">460</tt>  <tt class="py-line"><tt class="py-docstring">    Finding good schemas is very difficult because it takes forever to</tt> </tt>
<a name="L461"></a><tt class="py-lineno">461</tt>  <tt class="py-line"><tt class="py-docstring">    enumerate all of the potential schemas. This finder using a genetic</tt> </tt>
<a name="L462"></a><tt class="py-lineno">462</tt>  <tt class="py-line"><tt class="py-docstring">    algorithm approach to evolve good schema which match many times in</tt> </tt>
<a name="L463"></a><tt class="py-lineno">463</tt>  <tt class="py-line"><tt class="py-docstring">    a set of sequences.</tt> </tt>
<a name="L464"></a><tt class="py-lineno">464</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L465"></a><tt class="py-lineno">465</tt>  <tt class="py-line"><tt class="py-docstring">    The default implementation of the finder is ready to find schemas</tt> </tt>
<a name="L466"></a><tt class="py-lineno">466</tt>  <tt class="py-line"><tt class="py-docstring">    in a set of DNA sequences, but the finder can be customized to deal</tt> </tt>
<a name="L467"></a><tt class="py-lineno">467</tt>  <tt class="py-line"><tt class="py-docstring">    with any type of data.</tt> </tt>
<a name="L468"></a><tt class="py-lineno">468</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="SchemaFinder.__init__"></a><div id="SchemaFinder.__init__-def"><a name="L469"></a><tt class="py-lineno">469</tt> <a class="py-toggle" href="#" id="SchemaFinder.__init__-toggle" onclick="return toggle('SchemaFinder.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.SchemaFinder-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">num_schemas</tt><tt class="py-op">=</tt><tt class="py-number">100</tt><tt class="py-op">,</tt> </tt>
<a name="L470"></a><tt class="py-lineno">470</tt>  <tt class="py-line">                 <tt class="py-param">schema_finder</tt><tt class="py-op">=</tt><tt id="link-97" class="py-name" targets="Class Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder=Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder-class.html"><a title="Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder" class="py-name" href="#" onclick="return doclink('link-97', 'GeneticAlgorithmFinder', 'link-97');">GeneticAlgorithmFinder</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SchemaFinder.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SchemaFinder.__init__-expanded"><a name="L471"></a><tt class="py-lineno">471</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">num_schemas</tt> <tt class="py-op">=</tt> <tt class="py-name">num_schemas</tt> </tt>
<a name="L472"></a><tt class="py-lineno">472</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_finder</tt> <tt class="py-op">=</tt> <tt class="py-name">schema_finder</tt> </tt>
<a name="L473"></a><tt class="py-lineno">473</tt>  <tt class="py-line"> </tt>
<a name="L474"></a><tt class="py-lineno">474</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">evaluator</tt> <tt class="py-op">=</tt> <tt id="link-98" class="py-name"><a title="Bio.NeuralNetwork.Gene.Schema
Bio.NeuralNetwork.Gene.Schema.Schema" class="py-name" href="#" onclick="return doclink('link-98', 'Schema', 'link-62');">Schema</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_finder</tt><tt class="py-op">.</tt><tt id="link-99" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-99', 'alphabet', 'link-53');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-100" class="py-name"><a title="Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.alphabet_matches" class="py-name" href="#" onclick="return doclink('link-100', 'alphabet_matches', 'link-51');">alphabet_matches</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L475"></a><tt class="py-lineno">475</tt>  <tt class="py-line"> </tt>
<a name="SchemaFinder.find"></a><div id="SchemaFinder.find-def"><a name="L476"></a><tt class="py-lineno">476</tt> <a class="py-toggle" href="#" id="SchemaFinder.find-toggle" onclick="return toggle('SchemaFinder.find');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.SchemaFinder-class.html#find">find</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">seq_records</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SchemaFinder.find-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SchemaFinder.find-expanded"><a name="L477"></a><tt class="py-lineno">477</tt>  <tt class="py-line">        <tt class="py-docstring">"""Find well-represented schemas in the given set of SeqRecords.</tt> </tt>
<a name="L478"></a><tt class="py-lineno">478</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L479"></a><tt class="py-lineno">479</tt>  <tt class="py-line">        <tt class="py-name">fitness_evaluator</tt> <tt class="py-op">=</tt> <tt id="link-101" class="py-name" targets="Class Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness=Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness-class.html"><a title="Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness" class="py-name" href="#" onclick="return doclink('link-101', 'MostCountSchemaFitness', 'link-101');">MostCountSchemaFitness</a></tt><tt class="py-op">(</tt><tt class="py-name">seq_records</tt><tt class="py-op">,</tt> </tt>
<a name="L480"></a><tt class="py-lineno">480</tt>  <tt class="py-line">                                                   <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">evaluator</tt><tt class="py-op">)</tt> </tt>
<a name="L481"></a><tt class="py-lineno">481</tt>  <tt class="py-line"> </tt>
<a name="L482"></a><tt class="py-lineno">482</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_finder</tt><tt class="py-op">.</tt><tt id="link-102" class="py-name" targets="Method Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder.find_schemas()=Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder-class.html#find_schemas"><a title="Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder.find_schemas" class="py-name" href="#" onclick="return doclink('link-102', 'find_schemas', 'link-102');">find_schemas</a></tt><tt class="py-op">(</tt><tt class="py-name">fitness_evaluator</tt><tt class="py-op">.</tt><tt id="link-103" class="py-name" targets="Method Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness.calculate_fitness()=Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness-class.html#calculate_fitness,Method Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness.calculate_fitness()=Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness-class.html#calculate_fitness"><a title="Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness.calculate_fitness
Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness.calculate_fitness" class="py-name" href="#" onclick="return doclink('link-103', 'calculate_fitness', 'link-103');">calculate_fitness</a></tt><tt class="py-op">,</tt> </tt>
<a name="L483"></a><tt class="py-lineno">483</tt>  <tt class="py-line">                                         <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">num_schemas</tt><tt class="py-op">)</tt> </tt>
</div><a name="L484"></a><tt class="py-lineno">484</tt>  <tt class="py-line"> </tt>
<a name="SchemaFinder.find_differences"></a><div id="SchemaFinder.find_differences-def"><a name="L485"></a><tt class="py-lineno">485</tt> <a class="py-toggle" href="#" id="SchemaFinder.find_differences-toggle" onclick="return toggle('SchemaFinder.find_differences');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.SchemaFinder-class.html#find_differences">find_differences</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">first_records</tt><tt class="py-op">,</tt> <tt class="py-param">second_records</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SchemaFinder.find_differences-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SchemaFinder.find_differences-expanded"><a name="L486"></a><tt class="py-lineno">486</tt>  <tt class="py-line">        <tt class="py-docstring">"""Find schemas which differentiate between the two sets of SeqRecords.</tt> </tt>
<a name="L487"></a><tt class="py-lineno">487</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L488"></a><tt class="py-lineno">488</tt>  <tt class="py-line">        <tt class="py-name">fitness_evaluator</tt> <tt class="py-op">=</tt> <tt id="link-104" class="py-name" targets="Class Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness=Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness-class.html"><a title="Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness" class="py-name" href="#" onclick="return doclink('link-104', 'DifferentialSchemaFitness', 'link-104');">DifferentialSchemaFitness</a></tt><tt class="py-op">(</tt><tt class="py-name">first_records</tt><tt class="py-op">,</tt> </tt>
<a name="L489"></a><tt class="py-lineno">489</tt>  <tt class="py-line">                                                      <tt class="py-name">second_records</tt><tt class="py-op">,</tt> </tt>
<a name="L490"></a><tt class="py-lineno">490</tt>  <tt class="py-line">                                                      <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">evaluator</tt><tt class="py-op">)</tt> </tt>
<a name="L491"></a><tt class="py-lineno">491</tt>  <tt class="py-line"> </tt>
<a name="L492"></a><tt class="py-lineno">492</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_finder</tt><tt class="py-op">.</tt><tt id="link-105" class="py-name"><a title="Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder.find_schemas" class="py-name" href="#" onclick="return doclink('link-105', 'find_schemas', 'link-102');">find_schemas</a></tt><tt class="py-op">(</tt><tt class="py-name">fitness_evaluator</tt><tt class="py-op">.</tt><tt id="link-106" class="py-name"><a title="Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness.calculate_fitness
Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness.calculate_fitness" class="py-name" href="#" onclick="return doclink('link-106', 'calculate_fitness', 'link-103');">calculate_fitness</a></tt><tt class="py-op">,</tt> </tt>
<a name="L493"></a><tt class="py-lineno">493</tt>  <tt class="py-line">                                         <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">num_schemas</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L494"></a><tt class="py-lineno">494</tt>  <tt class="py-line"> </tt>
<a name="L495"></a><tt class="py-lineno">495</tt>  <tt class="py-line"> </tt>
<a name="SchemaCoder"></a><div id="SchemaCoder-def"><a name="L496"></a><tt class="py-lineno">496</tt> <a class="py-toggle" href="#" id="SchemaCoder-toggle" onclick="return toggle('SchemaCoder');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.SchemaCoder-class.html">SchemaCoder</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SchemaCoder-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="SchemaCoder-expanded"><a name="L497"></a><tt class="py-lineno">497</tt>  <tt class="py-line">    <tt class="py-docstring">"""Convert a sequence into a representation of ambiguous motifs (schemas).</tt> </tt>
<a name="L498"></a><tt class="py-lineno">498</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L499"></a><tt class="py-lineno">499</tt>  <tt class="py-line"><tt class="py-docstring">    This takes a sequence, and returns the number of times specified</tt> </tt>
<a name="L500"></a><tt class="py-lineno">500</tt>  <tt class="py-line"><tt class="py-docstring">    motifs are found in the sequence. This lets you represent a sequence</tt> </tt>
<a name="L501"></a><tt class="py-lineno">501</tt>  <tt class="py-line"><tt class="py-docstring">    as just a count of (possibly ambiguous) motifs.</tt> </tt>
<a name="L502"></a><tt class="py-lineno">502</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="SchemaCoder.__init__"></a><div id="SchemaCoder.__init__-def"><a name="L503"></a><tt class="py-lineno">503</tt> <a class="py-toggle" href="#" id="SchemaCoder.__init__-toggle" onclick="return toggle('SchemaCoder.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.SchemaCoder-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">schemas</tt><tt class="py-op">,</tt> <tt class="py-param">ambiguous_converter</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SchemaCoder.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SchemaCoder.__init__-expanded"><a name="L504"></a><tt class="py-lineno">504</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the coder to convert sequences</tt> </tt>
<a name="L505"></a><tt class="py-lineno">505</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L506"></a><tt class="py-lineno">506</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L507"></a><tt class="py-lineno">507</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L508"></a><tt class="py-lineno">508</tt>  <tt class="py-line"><tt class="py-docstring">        o schema - A list of all of the schemas we want to search for</tt> </tt>
<a name="L509"></a><tt class="py-lineno">509</tt>  <tt class="py-line"><tt class="py-docstring">        in input sequences.</tt> </tt>
<a name="L510"></a><tt class="py-lineno">510</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L511"></a><tt class="py-lineno">511</tt>  <tt class="py-line"><tt class="py-docstring">        o ambiguous_converter - An Schema class which can be</tt> </tt>
<a name="L512"></a><tt class="py-lineno">512</tt>  <tt class="py-line"><tt class="py-docstring">        used to convert motifs into regular expressions for searching.</tt> </tt>
<a name="L513"></a><tt class="py-lineno">513</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L514"></a><tt class="py-lineno">514</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_schemas</tt> <tt class="py-op">=</tt> <tt class="py-name">schemas</tt> </tt>
<a name="L515"></a><tt class="py-lineno">515</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-107" class="py-name" targets="Variable Bio.SeqIO._convert._converter=Bio.SeqIO._convert-module.html#_converter"><a title="Bio.SeqIO._convert._converter" class="py-name" href="#" onclick="return doclink('link-107', '_converter', 'link-107');">_converter</a></tt> <tt class="py-op">=</tt> <tt class="py-name">ambiguous_converter</tt> </tt>
</div><a name="L516"></a><tt class="py-lineno">516</tt>  <tt class="py-line"> </tt>
<a name="SchemaCoder.representation"></a><div id="SchemaCoder.representation-def"><a name="L517"></a><tt class="py-lineno">517</tt> <a class="py-toggle" href="#" id="SchemaCoder.representation-toggle" onclick="return toggle('SchemaCoder.representation');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.SchemaCoder-class.html#representation">representation</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sequence</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SchemaCoder.representation-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SchemaCoder.representation-expanded"><a name="L518"></a><tt class="py-lineno">518</tt>  <tt class="py-line">        <tt class="py-docstring">"""Represent the given input sequence as a bunch of motif counts.</tt> </tt>
<a name="L519"></a><tt class="py-lineno">519</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L520"></a><tt class="py-lineno">520</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L521"></a><tt class="py-lineno">521</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L522"></a><tt class="py-lineno">522</tt>  <tt class="py-line"><tt class="py-docstring">        o sequence - A Bio.Seq object we are going to represent as schemas.</tt> </tt>
<a name="L523"></a><tt class="py-lineno">523</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L524"></a><tt class="py-lineno">524</tt>  <tt class="py-line"><tt class="py-docstring">        This takes the sequence, searches for the motifs within it, and then</tt> </tt>
<a name="L525"></a><tt class="py-lineno">525</tt>  <tt class="py-line"><tt class="py-docstring">        returns counts specifying the relative number of times each motifs</tt> </tt>
<a name="L526"></a><tt class="py-lineno">526</tt>  <tt class="py-line"><tt class="py-docstring">        was found. The frequencies are in the order the original motifs were</tt> </tt>
<a name="L527"></a><tt class="py-lineno">527</tt>  <tt class="py-line"><tt class="py-docstring">        passed into the initializer.</tt> </tt>
<a name="L528"></a><tt class="py-lineno">528</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L529"></a><tt class="py-lineno">529</tt>  <tt class="py-line">        <tt class="py-name">schema_counts</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L530"></a><tt class="py-lineno">530</tt>  <tt class="py-line"> </tt>
<a name="L531"></a><tt class="py-lineno">531</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">schema</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_schemas</tt><tt class="py-op">:</tt> </tt>
<a name="L532"></a><tt class="py-lineno">532</tt>  <tt class="py-line">            <tt class="py-name">num_counts</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-108" class="py-name"><a title="Bio.SeqIO._convert._converter" class="py-name" href="#" onclick="return doclink('link-108', '_converter', 'link-107');">_converter</a></tt><tt class="py-op">.</tt><tt id="link-109" class="py-name"><a title="Bio.NeuralNetwork.Gene.Schema.Schema.num_matches" class="py-name" href="#" onclick="return doclink('link-109', 'num_matches', 'link-79');">num_matches</a></tt><tt class="py-op">(</tt><tt class="py-name">schema</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-110" class="py-name" targets="Method Bio.FSSP.FSSPAlignDict.sequence()=Bio.FSSP.FSSPAlignDict-class.html#sequence,Method Bio.GenBank._FeatureConsumer.sequence()=Bio.GenBank._FeatureConsumer-class.html#sequence,Method Bio.GenBank._RecordConsumer.sequence()=Bio.GenBank._RecordConsumer-class.html#sequence,Method Bio.Phylo.PhyloXMLIO.Writer.sequence()=Bio.Phylo.PhyloXMLIO.Writer-class.html#sequence"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-110', 'sequence', 'link-110');">sequence</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L533"></a><tt class="py-lineno">533</tt>  <tt class="py-line">            <tt class="py-name">schema_counts</tt><tt class="py-op">.</tt><tt id="link-111" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-111', 'append', 'link-43');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">num_counts</tt><tt class="py-op">)</tt> </tt>
<a name="L534"></a><tt class="py-lineno">534</tt>  <tt class="py-line"> </tt>
<a name="L535"></a><tt class="py-lineno">535</tt>  <tt class="py-line">        <tt class="py-comment"># normalize the counts to go between zero and one</tt> </tt>
<a name="L536"></a><tt class="py-lineno">536</tt>  <tt class="py-line">        <tt class="py-name">min_count</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L537"></a><tt class="py-lineno">537</tt>  <tt class="py-line">        <tt class="py-name">max_count</tt> <tt class="py-op">=</tt> <tt id="link-112" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.max=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#max"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-112', 'max', 'link-112');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">schema_counts</tt><tt class="py-op">)</tt> </tt>
<a name="L538"></a><tt class="py-lineno">538</tt>  <tt class="py-line"> </tt>
<a name="L539"></a><tt class="py-lineno">539</tt>  <tt class="py-line">        <tt class="py-comment"># only normalize if we've actually found something, otherwise</tt> </tt>
<a name="L540"></a><tt class="py-lineno">540</tt>  <tt class="py-line">        <tt class="py-comment"># we'll just return 0 for everything</tt> </tt>
<a name="L541"></a><tt class="py-lineno">541</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">max_count</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L542"></a><tt class="py-lineno">542</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">count_num</tt> <tt class="py-keyword">in</tt> <tt id="link-113" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-113', 'range', 'link-2');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">schema_counts</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L543"></a><tt class="py-lineno">543</tt>  <tt class="py-line">                <tt class="py-name">schema_counts</tt><tt class="py-op">[</tt><tt class="py-name">count_num</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">schema_counts</tt><tt class="py-op">[</tt><tt class="py-name">count_num</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> </tt>
<a name="L544"></a><tt class="py-lineno">544</tt>  <tt class="py-line">                                           <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">min_count</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">max_count</tt><tt class="py-op">)</tt> </tt>
<a name="L545"></a><tt class="py-lineno">545</tt>  <tt class="py-line"> </tt>
<a name="L546"></a><tt class="py-lineno">546</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">schema_counts</tt> </tt>
</div></div><a name="L547"></a><tt class="py-lineno">547</tt>  <tt class="py-line"> </tt>
<a name="L548"></a><tt class="py-lineno">548</tt>  <tt class="py-line"> </tt>
<a name="matches_schema"></a><div id="matches_schema-def"><a name="L549"></a><tt class="py-lineno">549</tt> <a class="py-toggle" href="#" id="matches_schema-toggle" onclick="return toggle('matches_schema');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema-module.html#matches_schema">matches_schema</a><tt class="py-op">(</tt><tt class="py-param">pattern</tt><tt class="py-op">,</tt> <tt class="py-param">schema</tt><tt class="py-op">,</tt> <tt class="py-param">ambiguity_character</tt><tt class="py-op">=</tt><tt class="py-string">'*'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="matches_schema-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="matches_schema-expanded"><a name="L550"></a><tt class="py-lineno">550</tt>  <tt class="py-line">    <tt class="py-docstring">"""Determine whether or not the given pattern matches the schema.</tt> </tt>
<a name="L551"></a><tt class="py-lineno">551</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L552"></a><tt class="py-lineno">552</tt>  <tt class="py-line"><tt class="py-docstring">    Arguments:</tt> </tt>
<a name="L553"></a><tt class="py-lineno">553</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L554"></a><tt class="py-lineno">554</tt>  <tt class="py-line"><tt class="py-docstring">    o pattern - A string representing the pattern we want to check for</tt> </tt>
<a name="L555"></a><tt class="py-lineno">555</tt>  <tt class="py-line"><tt class="py-docstring">    matching. This pattern can contain ambiguity characters (which are</tt> </tt>
<a name="L556"></a><tt class="py-lineno">556</tt>  <tt class="py-line"><tt class="py-docstring">    assumed to be the same as those in the schema).</tt> </tt>
<a name="L557"></a><tt class="py-lineno">557</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L558"></a><tt class="py-lineno">558</tt>  <tt class="py-line"><tt class="py-docstring">    o schema - A string schema with ambiguity characters.</tt> </tt>
<a name="L559"></a><tt class="py-lineno">559</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L560"></a><tt class="py-lineno">560</tt>  <tt class="py-line"><tt class="py-docstring">    o ambiguity_character - The character used for ambiguity in the schema.</tt> </tt>
<a name="L561"></a><tt class="py-lineno">561</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L562"></a><tt class="py-lineno">562</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">pattern</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">schema</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L563"></a><tt class="py-lineno">563</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-number">0</tt> </tt>
<a name="L564"></a><tt class="py-lineno">564</tt>  <tt class="py-line"> </tt>
<a name="L565"></a><tt class="py-lineno">565</tt>  <tt class="py-line">    <tt class="py-comment"># check each position, and return a non match if the schema and pattern</tt> </tt>
<a name="L566"></a><tt class="py-lineno">566</tt>  <tt class="py-line">    <tt class="py-comment"># are non ambiguous and don't match</tt> </tt>
<a name="L567"></a><tt class="py-lineno">567</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">pos</tt> <tt class="py-keyword">in</tt> <tt id="link-114" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-114', 'range', 'link-2');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">pattern</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L568"></a><tt class="py-lineno">568</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">schema</tt><tt class="py-op">[</tt><tt class="py-name">pos</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-name">ambiguity_character</tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L569"></a><tt class="py-lineno">569</tt>  <tt class="py-line">           <tt class="py-name">pattern</tt><tt class="py-op">[</tt><tt class="py-name">pos</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-name">ambiguity_character</tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L570"></a><tt class="py-lineno">570</tt>  <tt class="py-line">           <tt class="py-name">pattern</tt><tt class="py-op">[</tt><tt class="py-name">pos</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-name">schema</tt><tt class="py-op">[</tt><tt class="py-name">pos</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L571"></a><tt class="py-lineno">571</tt>  <tt class="py-line"> </tt>
<a name="L572"></a><tt class="py-lineno">572</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-number">0</tt> </tt>
<a name="L573"></a><tt class="py-lineno">573</tt>  <tt class="py-line"> </tt>
<a name="L574"></a><tt class="py-lineno">574</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-number">1</tt> </tt>
</div><a name="L575"></a><tt class="py-lineno">575</tt>  <tt class="py-line"> </tt>
<a name="L576"></a><tt class="py-lineno">576</tt>  <tt class="py-line"> </tt>
<a name="SchemaFactory"></a><div id="SchemaFactory-def"><a name="L577"></a><tt class="py-lineno">577</tt> <a class="py-toggle" href="#" id="SchemaFactory-toggle" onclick="return toggle('SchemaFactory');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.SchemaFactory-class.html">SchemaFactory</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SchemaFactory-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="SchemaFactory-expanded"><a name="L578"></a><tt class="py-lineno">578</tt>  <tt class="py-line">    <tt class="py-docstring">"""Generate Schema from inputs of Motifs or Signatures.</tt> </tt>
<a name="L579"></a><tt class="py-lineno">579</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="SchemaFactory.__init__"></a><div id="SchemaFactory.__init__-def"><a name="L580"></a><tt class="py-lineno">580</tt> <a class="py-toggle" href="#" id="SchemaFactory.__init__-toggle" onclick="return toggle('SchemaFactory.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.SchemaFactory-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">ambiguity_symbol</tt><tt class="py-op">=</tt><tt class="py-string">'*'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SchemaFactory.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SchemaFactory.__init__-expanded"><a name="L581"></a><tt class="py-lineno">581</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the SchemaFactory</tt> </tt>
<a name="L582"></a><tt class="py-lineno">582</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L583"></a><tt class="py-lineno">583</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L584"></a><tt class="py-lineno">584</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L585"></a><tt class="py-lineno">585</tt>  <tt class="py-line"><tt class="py-docstring">        o ambiguity_symbol -- The symbol to use when specifying that</tt> </tt>
<a name="L586"></a><tt class="py-lineno">586</tt>  <tt class="py-line"><tt class="py-docstring">        a position is arbitrary.</tt> </tt>
<a name="L587"></a><tt class="py-lineno">587</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L588"></a><tt class="py-lineno">588</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_ambiguity_symbol</tt> <tt class="py-op">=</tt> <tt class="py-name">ambiguity_symbol</tt> </tt>
</div><a name="L589"></a><tt class="py-lineno">589</tt>  <tt class="py-line"> </tt>
<a name="SchemaFactory.from_motifs"></a><div id="SchemaFactory.from_motifs-def"><a name="L590"></a><tt class="py-lineno">590</tt> <a class="py-toggle" href="#" id="SchemaFactory.from_motifs-toggle" onclick="return toggle('SchemaFactory.from_motifs');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.SchemaFactory-class.html#from_motifs">from_motifs</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">motif_repository</tt><tt class="py-op">,</tt> <tt class="py-param">motif_percent</tt><tt class="py-op">,</tt> <tt class="py-param">num_ambiguous</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SchemaFactory.from_motifs-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SchemaFactory.from_motifs-expanded"><a name="L591"></a><tt class="py-lineno">591</tt>  <tt class="py-line">        <tt class="py-docstring">"""Generate schema from a list of motifs.</tt> </tt>
<a name="L592"></a><tt class="py-lineno">592</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L593"></a><tt class="py-lineno">593</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L594"></a><tt class="py-lineno">594</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L595"></a><tt class="py-lineno">595</tt>  <tt class="py-line"><tt class="py-docstring">        o motif_repository - A MotifRepository class that has all of the</tt> </tt>
<a name="L596"></a><tt class="py-lineno">596</tt>  <tt class="py-line"><tt class="py-docstring">        motifs we want to convert to Schema.</tt> </tt>
<a name="L597"></a><tt class="py-lineno">597</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L598"></a><tt class="py-lineno">598</tt>  <tt class="py-line"><tt class="py-docstring">        o motif_percent - The percentage of motifs in the motif bank which</tt> </tt>
<a name="L599"></a><tt class="py-lineno">599</tt>  <tt class="py-line"><tt class="py-docstring">        should be matches. We'll try to create schema that match this</tt> </tt>
<a name="L600"></a><tt class="py-lineno">600</tt>  <tt class="py-line"><tt class="py-docstring">        percentage of motifs.</tt> </tt>
<a name="L601"></a><tt class="py-lineno">601</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L602"></a><tt class="py-lineno">602</tt>  <tt class="py-line"><tt class="py-docstring">        o num_ambiguous - The number of ambiguous characters to include</tt> </tt>
<a name="L603"></a><tt class="py-lineno">603</tt>  <tt class="py-line"><tt class="py-docstring">        in each schema. The positions of these ambiguous characters will</tt> </tt>
<a name="L604"></a><tt class="py-lineno">604</tt>  <tt class="py-line"><tt class="py-docstring">        be randomly selected.</tt> </tt>
<a name="L605"></a><tt class="py-lineno">605</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L606"></a><tt class="py-lineno">606</tt>  <tt class="py-line">        <tt class="py-comment"># get all of the motifs we can deal with</tt> </tt>
<a name="L607"></a><tt class="py-lineno">607</tt>  <tt class="py-line">        <tt class="py-name">all_motifs</tt> <tt class="py-op">=</tt> <tt class="py-name">motif_repository</tt><tt class="py-op">.</tt><tt id="link-115" class="py-name" targets="Method Bio.NeuralNetwork.Gene.Pattern.PatternRepository.get_top_percentage()=Bio.NeuralNetwork.Gene.Pattern.PatternRepository-class.html#get_top_percentage"><a title="Bio.NeuralNetwork.Gene.Pattern.PatternRepository.get_top_percentage" class="py-name" href="#" onclick="return doclink('link-115', 'get_top_percentage', 'link-115');">get_top_percentage</a></tt><tt class="py-op">(</tt><tt class="py-name">motif_percent</tt><tt class="py-op">)</tt> </tt>
<a name="L608"></a><tt class="py-lineno">608</tt>  <tt class="py-line"> </tt>
<a name="L609"></a><tt class="py-lineno">609</tt>  <tt class="py-line">        <tt class="py-comment"># start building up schemas</tt> </tt>
<a name="L610"></a><tt class="py-lineno">610</tt>  <tt class="py-line">        <tt class="py-name">schema_info</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L611"></a><tt class="py-lineno">611</tt>  <tt class="py-line">        <tt class="py-comment"># continue until we've built schema matching the desired percentage</tt> </tt>
<a name="L612"></a><tt class="py-lineno">612</tt>  <tt class="py-line">        <tt class="py-comment"># of motifs</tt> </tt>
<a name="L613"></a><tt class="py-lineno">613</tt>  <tt class="py-line">        <tt class="py-name">total_count</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-116" class="py-name" targets="Method Bio.NeuralNetwork.Gene.Schema.SchemaFactory._get_num_motifs()=Bio.NeuralNetwork.Gene.Schema.SchemaFactory-class.html#_get_num_motifs"><a title="Bio.NeuralNetwork.Gene.Schema.SchemaFactory._get_num_motifs" class="py-name" href="#" onclick="return doclink('link-116', '_get_num_motifs', 'link-116');">_get_num_motifs</a></tt><tt class="py-op">(</tt><tt class="py-name">motif_repository</tt><tt class="py-op">,</tt> <tt class="py-name">all_motifs</tt><tt class="py-op">)</tt> </tt>
<a name="L614"></a><tt class="py-lineno">614</tt>  <tt class="py-line">        <tt class="py-name">matched_count</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L615"></a><tt class="py-lineno">615</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">total_count</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-string">"Expected to have motifs to match"</tt> </tt>
<a name="L616"></a><tt class="py-lineno">616</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-op">(</tt><tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">matched_count</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">total_count</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">motif_percent</tt><tt class="py-op">:</tt> </tt>
<a name="L617"></a><tt class="py-lineno">617</tt>  <tt class="py-line">            <tt class="py-name">new_schema</tt><tt class="py-op">,</tt> <tt class="py-name">matching_motifs</tt> <tt class="py-op">=</tt> \ </tt>
<a name="L618"></a><tt class="py-lineno">618</tt>  <tt class="py-line">                        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-117" class="py-name" targets="Method Bio.NeuralNetwork.Gene.Schema.SchemaFactory._get_unique_schema()=Bio.NeuralNetwork.Gene.Schema.SchemaFactory-class.html#_get_unique_schema"><a title="Bio.NeuralNetwork.Gene.Schema.SchemaFactory._get_unique_schema" class="py-name" href="#" onclick="return doclink('link-117', '_get_unique_schema', 'link-117');">_get_unique_schema</a></tt><tt class="py-op">(</tt><tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">schema_info</tt><tt class="py-op">.</tt><tt id="link-118" class="py-name" targets="Method Bio.Crystal.Crystal.keys()=Bio.Crystal.Crystal-class.html#keys,Method Bio.File._IndexedSeqFileDict.keys()=Bio.File._IndexedSeqFileDict-class.html#keys,Method Bio.File._SQLiteManySeqFilesDict.keys()=Bio.File._SQLiteManySeqFilesDict-class.html#keys,Method Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys()=Bio.PDB.AbstractPropertyMap.AbstractPropertyMap-class.html#keys,Method Bio.Phylo.PhyloXML.Events.keys()=Bio.Phylo.PhyloXML.Events-class.html#keys,Method Bio._py3k._ordereddict.OrderedDict.keys()=Bio._py3k._ordereddict.OrderedDict-class.html#keys,Method BioSQL.BioSeqDatabase.BioSeqDatabase.keys()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#keys,Method BioSQL.BioSeqDatabase.DBServer.keys()=BioSQL.BioSeqDatabase.DBServer-class.html#keys"><a title="Bio.Crystal.Crystal.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
Bio._py3k._ordereddict.OrderedDict.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-118', 'keys', 'link-118');">keys</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L619"></a><tt class="py-lineno">619</tt>  <tt class="py-line">                                                <tt class="py-name">all_motifs</tt><tt class="py-op">,</tt> <tt id="link-119" class="py-name"><a title="Bio.NeuralNetwork.Gene.Schema.Schema.num_ambiguous" class="py-name" href="#" onclick="return doclink('link-119', 'num_ambiguous', 'link-83');">num_ambiguous</a></tt><tt class="py-op">)</tt> </tt>
<a name="L620"></a><tt class="py-lineno">620</tt>  <tt class="py-line"> </tt>
<a name="L621"></a><tt class="py-lineno">621</tt>  <tt class="py-line">            <tt class="py-comment"># get the number of counts for the new schema and clean up</tt> </tt>
<a name="L622"></a><tt class="py-lineno">622</tt>  <tt class="py-line">            <tt class="py-comment"># the motif list</tt> </tt>
<a name="L623"></a><tt class="py-lineno">623</tt>  <tt class="py-line">            <tt class="py-name">schema_counts</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L624"></a><tt class="py-lineno">624</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">motif</tt> <tt class="py-keyword">in</tt> <tt class="py-name">matching_motifs</tt><tt class="py-op">:</tt> </tt>
<a name="L625"></a><tt class="py-lineno">625</tt>  <tt class="py-line">                <tt class="py-comment"># get the counts for the motif</tt> </tt>
<a name="L626"></a><tt class="py-lineno">626</tt>  <tt class="py-line">                <tt class="py-name">schema_counts</tt> <tt class="py-op">+=</tt> <tt class="py-name">motif_repository</tt><tt class="py-op">.</tt><tt id="link-120" class="py-name" targets="Method Bio.Crystal.Chain.count()=Bio.Crystal.Chain-class.html#count,Method Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count()=Bio.NeuralNetwork.Gene.Pattern.PatternRepository-class.html#count,Method Bio.Seq.MutableSeq.count()=Bio.Seq.MutableSeq-class.html#count,Method Bio.Seq.Seq.count()=Bio.Seq.Seq-class.html#count,Method Bio.Seq.UnknownSeq.count()=Bio.Seq.UnknownSeq-class.html#count,Method Bio.motifs.Instances.count()=Bio.motifs.Instances-class.html#count"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-120', 'count', 'link-120');">count</a></tt><tt class="py-op">(</tt><tt class="py-name">motif</tt><tt class="py-op">)</tt> </tt>
<a name="L627"></a><tt class="py-lineno">627</tt>  <tt class="py-line"> </tt>
<a name="L628"></a><tt class="py-lineno">628</tt>  <tt class="py-line">                <tt class="py-comment"># remove the motif from the motif list since it is already</tt> </tt>
<a name="L629"></a><tt class="py-lineno">629</tt>  <tt class="py-line">                <tt class="py-comment"># represented by this schema</tt> </tt>
<a name="L630"></a><tt class="py-lineno">630</tt>  <tt class="py-line">                <tt class="py-name">all_motifs</tt><tt class="py-op">.</tt><tt id="link-121" class="py-name" targets="Method Bio.Crystal.Chain.remove()=Bio.Crystal.Chain-class.html#remove,Method Bio.Graphics.BasicChromosome._ChromosomeComponent.remove()=Bio.Graphics.BasicChromosome._ChromosomeComponent-class.html#remove,Method Bio.Restriction.Restriction.RestrictionBatch.remove()=Bio.Restriction.Restriction.RestrictionBatch-class.html#remove,Method Bio.Seq.MutableSeq.remove()=Bio.Seq.MutableSeq-class.html#remove,Method BioSQL.Loader.DatabaseRemover.remove()=BioSQL.Loader.DatabaseRemover-class.html#remove"><a title="Bio.Crystal.Chain.remove
Bio.Graphics.BasicChromosome._ChromosomeComponent.remove
Bio.Restriction.Restriction.RestrictionBatch.remove
Bio.Seq.MutableSeq.remove
BioSQL.Loader.DatabaseRemover.remove" class="py-name" href="#" onclick="return doclink('link-121', 'remove', 'link-121');">remove</a></tt><tt class="py-op">(</tt><tt class="py-name">motif</tt><tt class="py-op">)</tt> </tt>
<a name="L631"></a><tt class="py-lineno">631</tt>  <tt class="py-line"> </tt>
<a name="L632"></a><tt class="py-lineno">632</tt>  <tt class="py-line">            <tt class="py-comment"># all the schema info</tt> </tt>
<a name="L633"></a><tt class="py-lineno">633</tt>  <tt class="py-line">            <tt class="py-name">schema_info</tt><tt class="py-op">[</tt><tt class="py-name">new_schema</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">schema_counts</tt> </tt>
<a name="L634"></a><tt class="py-lineno">634</tt>  <tt class="py-line"> </tt>
<a name="L635"></a><tt class="py-lineno">635</tt>  <tt class="py-line">            <tt class="py-name">matched_count</tt> <tt class="py-op">+=</tt> <tt class="py-name">schema_counts</tt> </tt>
<a name="L636"></a><tt class="py-lineno">636</tt>  <tt class="py-line"> </tt>
<a name="L637"></a><tt class="py-lineno">637</tt>  <tt class="py-line">            <tt class="py-comment"># print "percentage:", float(matched_count) / float(total_count)</tt> </tt>
<a name="L638"></a><tt class="py-lineno">638</tt>  <tt class="py-line"> </tt>
<a name="L639"></a><tt class="py-lineno">639</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-122" class="py-name"><a title="Bio.NeuralNetwork.Gene.Pattern.PatternRepository" class="py-name" href="#" onclick="return doclink('link-122', 'PatternRepository', 'link-9');">PatternRepository</a></tt><tt class="py-op">(</tt><tt class="py-name">schema_info</tt><tt class="py-op">)</tt> </tt>
</div><a name="L640"></a><tt class="py-lineno">640</tt>  <tt class="py-line"> </tt>
<a name="SchemaFactory._get_num_motifs"></a><div id="SchemaFactory._get_num_motifs-def"><a name="L641"></a><tt class="py-lineno">641</tt> <a class="py-toggle" href="#" id="SchemaFactory._get_num_motifs-toggle" onclick="return toggle('SchemaFactory._get_num_motifs');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.SchemaFactory-class.html#_get_num_motifs">_get_num_motifs</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">repository</tt><tt class="py-op">,</tt> <tt class="py-param">motif_list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SchemaFactory._get_num_motifs-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SchemaFactory._get_num_motifs-expanded"><a name="L642"></a><tt class="py-lineno">642</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the number of motif counts for the list of motifs.</tt> </tt>
<a name="L643"></a><tt class="py-lineno">643</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L644"></a><tt class="py-lineno">644</tt>  <tt class="py-line">        <tt class="py-name">motif_count</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L645"></a><tt class="py-lineno">645</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">motif</tt> <tt class="py-keyword">in</tt> <tt class="py-name">motif_list</tt><tt class="py-op">:</tt> </tt>
<a name="L646"></a><tt class="py-lineno">646</tt>  <tt class="py-line">            <tt class="py-name">motif_count</tt> <tt class="py-op">+=</tt> <tt class="py-name">repository</tt><tt class="py-op">.</tt><tt id="link-123" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-123', 'count', 'link-120');">count</a></tt><tt class="py-op">(</tt><tt class="py-name">motif</tt><tt class="py-op">)</tt> </tt>
<a name="L647"></a><tt class="py-lineno">647</tt>  <tt class="py-line"> </tt>
<a name="L648"></a><tt class="py-lineno">648</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">motif_count</tt> </tt>
</div><a name="L649"></a><tt class="py-lineno">649</tt>  <tt class="py-line"> </tt>
<a name="SchemaFactory._get_unique_schema"></a><div id="SchemaFactory._get_unique_schema-def"><a name="L650"></a><tt class="py-lineno">650</tt> <a class="py-toggle" href="#" id="SchemaFactory._get_unique_schema-toggle" onclick="return toggle('SchemaFactory._get_unique_schema');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.SchemaFactory-class.html#_get_unique_schema">_get_unique_schema</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">cur_schemas</tt><tt class="py-op">,</tt> <tt class="py-param">motif_list</tt><tt class="py-op">,</tt> <tt class="py-param">num_ambiguous</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SchemaFactory._get_unique_schema-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SchemaFactory._get_unique_schema-expanded"><a name="L651"></a><tt class="py-lineno">651</tt>  <tt class="py-line">        <tt class="py-docstring">"""Retrieve a unique schema from a motif.</tt> </tt>
<a name="L652"></a><tt class="py-lineno">652</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L653"></a><tt class="py-lineno">653</tt>  <tt class="py-line"><tt class="py-docstring">        We don't want to end up with schema that match the same thing,</tt> </tt>
<a name="L654"></a><tt class="py-lineno">654</tt>  <tt class="py-line"><tt class="py-docstring">        since this could lead to ambiguous results, and be messy. This</tt> </tt>
<a name="L655"></a><tt class="py-lineno">655</tt>  <tt class="py-line"><tt class="py-docstring">        tries to create schema, and checks that they do not match any</tt> </tt>
<a name="L656"></a><tt class="py-lineno">656</tt>  <tt class="py-line"><tt class="py-docstring">        currently existing schema.</tt> </tt>
<a name="L657"></a><tt class="py-lineno">657</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L658"></a><tt class="py-lineno">658</tt>  <tt class="py-line">        <tt class="py-comment"># create a schema starting with a random motif</tt> </tt>
<a name="L659"></a><tt class="py-lineno">659</tt>  <tt class="py-line">        <tt class="py-comment"># we'll keep doing this until we get a completely new schema that</tt> </tt>
<a name="L660"></a><tt class="py-lineno">660</tt>  <tt class="py-line">        <tt class="py-comment"># doesn't match any old schema</tt> </tt>
<a name="L661"></a><tt class="py-lineno">661</tt>  <tt class="py-line">        <tt class="py-name">num_tries</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L662"></a><tt class="py-lineno">662</tt>  <tt class="py-line"> </tt>
<a name="L663"></a><tt class="py-lineno">663</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L664"></a><tt class="py-lineno">664</tt>  <tt class="py-line">            <tt class="py-comment"># pick a motif to work from and make a schema from it</tt> </tt>
<a name="L665"></a><tt class="py-lineno">665</tt>  <tt class="py-line">            <tt class="py-name">cur_motif</tt> <tt class="py-op">=</tt> <tt class="py-name">random</tt><tt class="py-op">.</tt><tt class="py-name">choice</tt><tt class="py-op">(</tt><tt class="py-name">motif_list</tt><tt class="py-op">)</tt> </tt>
<a name="L666"></a><tt class="py-lineno">666</tt>  <tt class="py-line"> </tt>
<a name="L667"></a><tt class="py-lineno">667</tt>  <tt class="py-line">            <tt class="py-name">num_tries</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L668"></a><tt class="py-lineno">668</tt>  <tt class="py-line"> </tt>
<a name="L669"></a><tt class="py-lineno">669</tt>  <tt class="py-line">            <tt class="py-name">new_schema</tt><tt class="py-op">,</tt> <tt class="py-name">matching_motifs</tt> <tt class="py-op">=</tt> \ </tt>
<a name="L670"></a><tt class="py-lineno">670</tt>  <tt class="py-line">                        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-124" class="py-name" targets="Method Bio.NeuralNetwork.Gene.Schema.SchemaFactory._schema_from_motif()=Bio.NeuralNetwork.Gene.Schema.SchemaFactory-class.html#_schema_from_motif"><a title="Bio.NeuralNetwork.Gene.Schema.SchemaFactory._schema_from_motif" class="py-name" href="#" onclick="return doclink('link-124', '_schema_from_motif', 'link-124');">_schema_from_motif</a></tt><tt class="py-op">(</tt><tt class="py-name">cur_motif</tt><tt class="py-op">,</tt> <tt class="py-name">motif_list</tt><tt class="py-op">,</tt> </tt>
<a name="L671"></a><tt class="py-lineno">671</tt>  <tt class="py-line">                                                <tt id="link-125" class="py-name"><a title="Bio.NeuralNetwork.Gene.Schema.Schema.num_ambiguous" class="py-name" href="#" onclick="return doclink('link-125', 'num_ambiguous', 'link-83');">num_ambiguous</a></tt><tt class="py-op">)</tt> </tt>
<a name="L672"></a><tt class="py-lineno">672</tt>  <tt class="py-line"> </tt>
<a name="L673"></a><tt class="py-lineno">673</tt>  <tt class="py-line">            <tt class="py-name">has_match</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L674"></a><tt class="py-lineno">674</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">old_schema</tt> <tt class="py-keyword">in</tt> <tt class="py-name">cur_schemas</tt><tt class="py-op">:</tt> </tt>
<a name="L675"></a><tt class="py-lineno">675</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-126" class="py-name" targets="Function Bio.NeuralNetwork.Gene.Schema.matches_schema()=Bio.NeuralNetwork.Gene.Schema-module.html#matches_schema"><a title="Bio.NeuralNetwork.Gene.Schema.matches_schema" class="py-name" href="#" onclick="return doclink('link-126', 'matches_schema', 'link-126');">matches_schema</a></tt><tt class="py-op">(</tt><tt class="py-name">new_schema</tt><tt class="py-op">,</tt> <tt class="py-name">old_schema</tt><tt class="py-op">,</tt> </tt>
<a name="L676"></a><tt class="py-lineno">676</tt>  <tt class="py-line">                                  <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_ambiguity_symbol</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L677"></a><tt class="py-lineno">677</tt>  <tt class="py-line">                    <tt class="py-name">has_match</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L678"></a><tt class="py-lineno">678</tt>  <tt class="py-line"> </tt>
<a name="L679"></a><tt class="py-lineno">679</tt>  <tt class="py-line">            <tt class="py-comment"># if the schema doesn't match any other schema we've got</tt> </tt>
<a name="L680"></a><tt class="py-lineno">680</tt>  <tt class="py-line">            <tt class="py-comment"># a good one</tt> </tt>
<a name="L681"></a><tt class="py-lineno">681</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt><tt class="py-op">(</tt><tt class="py-name">has_match</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L682"></a><tt class="py-lineno">682</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L683"></a><tt class="py-lineno">683</tt>  <tt class="py-line"> </tt>
<a name="L684"></a><tt class="py-lineno">684</tt>  <tt class="py-line">            <tt class="py-comment"># check for big loops in which we can't find a new schema</tt> </tt>
<a name="L685"></a><tt class="py-lineno">685</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">num_tries</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">150</tt><tt class="py-op">,</tt> \ </tt>
<a name="L686"></a><tt class="py-lineno">686</tt>  <tt class="py-line">                   <tt class="py-string">"Could not generate schema in %s tries from %s with %s"</tt> \ </tt>
<a name="L687"></a><tt class="py-lineno">687</tt>  <tt class="py-line">                   <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">num_tries</tt><tt class="py-op">,</tt> <tt class="py-name">motif_list</tt><tt class="py-op">,</tt> <tt class="py-name">cur_schemas</tt><tt class="py-op">)</tt> </tt>
<a name="L688"></a><tt class="py-lineno">688</tt>  <tt class="py-line"> </tt>
<a name="L689"></a><tt class="py-lineno">689</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">new_schema</tt><tt class="py-op">,</tt> <tt class="py-name">matching_motifs</tt> </tt>
</div><a name="L690"></a><tt class="py-lineno">690</tt>  <tt class="py-line"> </tt>
<a name="SchemaFactory._schema_from_motif"></a><div id="SchemaFactory._schema_from_motif-def"><a name="L691"></a><tt class="py-lineno">691</tt> <a class="py-toggle" href="#" id="SchemaFactory._schema_from_motif-toggle" onclick="return toggle('SchemaFactory._schema_from_motif');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.SchemaFactory-class.html#_schema_from_motif">_schema_from_motif</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">motif</tt><tt class="py-op">,</tt> <tt class="py-param">motif_list</tt><tt class="py-op">,</tt> <tt class="py-param">num_ambiguous</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SchemaFactory._schema_from_motif-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SchemaFactory._schema_from_motif-expanded"><a name="L692"></a><tt class="py-lineno">692</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a schema from a given starting motif.</tt> </tt>
<a name="L693"></a><tt class="py-lineno">693</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L694"></a><tt class="py-lineno">694</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L695"></a><tt class="py-lineno">695</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L696"></a><tt class="py-lineno">696</tt>  <tt class="py-line"><tt class="py-docstring">        o motif - A motif with the pattern we will start from.</tt> </tt>
<a name="L697"></a><tt class="py-lineno">697</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L698"></a><tt class="py-lineno">698</tt>  <tt class="py-line"><tt class="py-docstring">        o motif_list - The total motifs we have.to match to.</tt> </tt>
<a name="L699"></a><tt class="py-lineno">699</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L700"></a><tt class="py-lineno">700</tt>  <tt class="py-line"><tt class="py-docstring">        o num_ambiguous - The number of ambiguous characters that should</tt> </tt>
<a name="L701"></a><tt class="py-lineno">701</tt>  <tt class="py-line"><tt class="py-docstring">        be present in the schema.</tt> </tt>
<a name="L702"></a><tt class="py-lineno">702</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L703"></a><tt class="py-lineno">703</tt>  <tt class="py-line"><tt class="py-docstring">        Returns:</tt> </tt>
<a name="L704"></a><tt class="py-lineno">704</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L705"></a><tt class="py-lineno">705</tt>  <tt class="py-line"><tt class="py-docstring">        o A string representing the newly generated schema.</tt> </tt>
<a name="L706"></a><tt class="py-lineno">706</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L707"></a><tt class="py-lineno">707</tt>  <tt class="py-line"><tt class="py-docstring">        o A list of all of the motifs in motif_list that match the schema.</tt> </tt>
<a name="L708"></a><tt class="py-lineno">708</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L709"></a><tt class="py-lineno">709</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">motif</tt> <tt class="py-keyword">in</tt> <tt class="py-name">motif_list</tt><tt class="py-op">,</tt> \ </tt>
<a name="L710"></a><tt class="py-lineno">710</tt>  <tt class="py-line">               <tt class="py-string">"Expected starting motif present in remaining motifs."</tt> </tt>
<a name="L711"></a><tt class="py-lineno">711</tt>  <tt class="py-line"> </tt>
<a name="L712"></a><tt class="py-lineno">712</tt>  <tt class="py-line">        <tt class="py-comment"># convert random positions in the motif to ambiguous characters</tt> </tt>
<a name="L713"></a><tt class="py-lineno">713</tt>  <tt class="py-line">        <tt class="py-comment"># convert the motif into a list of characters so we can manipulate it</tt> </tt>
<a name="L714"></a><tt class="py-lineno">714</tt>  <tt class="py-line">        <tt class="py-name">new_schema_list</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">motif</tt><tt class="py-op">)</tt> </tt>
<a name="L715"></a><tt class="py-lineno">715</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">add_ambiguous</tt> <tt class="py-keyword">in</tt> <tt id="link-127" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-127', 'range', 'link-2');">range</a></tt><tt class="py-op">(</tt><tt id="link-128" class="py-name"><a title="Bio.NeuralNetwork.Gene.Schema.Schema.num_ambiguous" class="py-name" href="#" onclick="return doclink('link-128', 'num_ambiguous', 'link-83');">num_ambiguous</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L716"></a><tt class="py-lineno">716</tt>  <tt class="py-line">            <tt class="py-comment"># add an ambiguous position in a new place in the motif</tt> </tt>
<a name="L717"></a><tt class="py-lineno">717</tt>  <tt class="py-line">            <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L718"></a><tt class="py-lineno">718</tt>  <tt class="py-line">                <tt class="py-name">ambig_pos</tt> <tt class="py-op">=</tt> <tt class="py-name">random</tt><tt class="py-op">.</tt><tt class="py-name">choice</tt><tt class="py-op">(</tt><tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-129" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-129', 'range', 'link-2');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">new_schema_list</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L719"></a><tt class="py-lineno">719</tt>  <tt class="py-line"> </tt>
<a name="L720"></a><tt class="py-lineno">720</tt>  <tt class="py-line">                <tt class="py-comment"># only add a position if it isn't already ambiguous</tt> </tt>
<a name="L721"></a><tt class="py-lineno">721</tt>  <tt class="py-line">                <tt class="py-comment"># otherwise, we'll try again</tt> </tt>
<a name="L722"></a><tt class="py-lineno">722</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">new_schema_list</tt><tt class="py-op">[</tt><tt class="py-name">ambig_pos</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_ambiguity_symbol</tt><tt class="py-op">:</tt> </tt>
<a name="L723"></a><tt class="py-lineno">723</tt>  <tt class="py-line">                    <tt class="py-name">new_schema_list</tt><tt class="py-op">[</tt><tt class="py-name">ambig_pos</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_ambiguity_symbol</tt> </tt>
<a name="L724"></a><tt class="py-lineno">724</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L725"></a><tt class="py-lineno">725</tt>  <tt class="py-line"> </tt>
<a name="L726"></a><tt class="py-lineno">726</tt>  <tt class="py-line">        <tt class="py-comment"># convert the schema back to a string</tt> </tt>
<a name="L727"></a><tt class="py-lineno">727</tt>  <tt class="py-line">        <tt class="py-name">new_schema</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">new_schema_list</tt><tt class="py-op">)</tt> </tt>
<a name="L728"></a><tt class="py-lineno">728</tt>  <tt class="py-line"> </tt>
<a name="L729"></a><tt class="py-lineno">729</tt>  <tt class="py-line">        <tt class="py-comment"># get the motifs that the schema matches</tt> </tt>
<a name="L730"></a><tt class="py-lineno">730</tt>  <tt class="py-line">        <tt class="py-name">matched_motifs</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L731"></a><tt class="py-lineno">731</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">motif</tt> <tt class="py-keyword">in</tt> <tt class="py-name">motif_list</tt><tt class="py-op">:</tt> </tt>
<a name="L732"></a><tt class="py-lineno">732</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-130" class="py-name"><a title="Bio.NeuralNetwork.Gene.Schema.matches_schema" class="py-name" href="#" onclick="return doclink('link-130', 'matches_schema', 'link-126');">matches_schema</a></tt><tt class="py-op">(</tt><tt class="py-name">motif</tt><tt class="py-op">,</tt> <tt class="py-name">new_schema</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_ambiguity_symbol</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L733"></a><tt class="py-lineno">733</tt>  <tt class="py-line">                <tt class="py-name">matched_motifs</tt><tt class="py-op">.</tt><tt id="link-131" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-131', 'append', 'link-43');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">motif</tt><tt class="py-op">)</tt> </tt>
<a name="L734"></a><tt class="py-lineno">734</tt>  <tt class="py-line"> </tt>
<a name="L735"></a><tt class="py-lineno">735</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">new_schema</tt><tt class="py-op">,</tt> <tt class="py-name">matched_motifs</tt> </tt>
</div><a name="L736"></a><tt class="py-lineno">736</tt>  <tt class="py-line"> </tt>
<a name="SchemaFactory.from_signatures"></a><div id="SchemaFactory.from_signatures-def"><a name="L737"></a><tt class="py-lineno">737</tt> <a class="py-toggle" href="#" id="SchemaFactory.from_signatures-toggle" onclick="return toggle('SchemaFactory.from_signatures');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.NeuralNetwork.Gene.Schema.SchemaFactory-class.html#from_signatures">from_signatures</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">signature_repository</tt><tt class="py-op">,</tt> <tt class="py-param">num_ambiguous</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SchemaFactory.from_signatures-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SchemaFactory.from_signatures-expanded"><a name="L738"></a><tt class="py-lineno">738</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">NotImplementedError</tt><tt class="py-op">(</tt><tt class="py-string">"Still need to code this."</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L739"></a><tt class="py-lineno">739</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:04:50 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
