<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.GenBank.Record.Record</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.GenBank-module.html">Package&nbsp;GenBank</a> ::
        <a href="Bio.GenBank.Record-module.html">Module&nbsp;Record</a> ::
        Class&nbsp;Record
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.GenBank.Record.Record-class.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== CLASS DESCRIPTION ==================== -->
<h1 class="epydoc">Class Record</h1><p class="nomargin-top"><span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record">source&nbsp;code</a></span></p>
<pre class="base-tree">
object --+
         |
        <strong class="uidshort">Record</strong>
</pre>

<hr />
<p>Hold GenBank information in a format similar to the original record.</p>
<p>The Record class is meant to make data easy to get to when you are
just interested in looking at GenBank data.</p>
<dl class="rst-docutils">
<dt>Attributes:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>locus - The name specified after the LOCUS keyword in the GenBank
record. This may be the accession number, or a clone id or something else.</li>
<li>size - The size of the record.</li>
<li>residue_type - The type of residues making up the sequence in this
record. Normally something like RNA, DNA or PROTEIN, but may be as
esoteric as 'ss-RNA circular'.</li>
<li>data_file_division - The division this record is stored under in
GenBank (ie. PLN -&gt; plants; PRI -&gt; humans, primates; BCT -&gt; bacteria...)</li>
<li>date - The date of submission of the record, in a form like '28-JUL-1998'</li>
<li>accession - list of all accession numbers for the sequence.</li>
<li>nid - Nucleotide identifier number.</li>
<li>pid - Proteint identifier number</li>
<li>version - The accession number + version (ie. AB01234.2)</li>
<li>db_source - Information about the database the record came from</li>
<li>gi - The NCBI gi identifier for the record.</li>
<li>keywords - A list of keywords related to the record.</li>
<li>segment - If the record is one of a series, this is info about which
segment this record is (something like '1 of 6').</li>
<li>source - The source of material where the sequence came from.</li>
<li>organism - The genus and species of the organism (ie. 'Homo sapiens')</li>
<li>taxonomy - A listing of the taxonomic classification of the organism,
starting general and getting more specific.</li>
<li>references - A list of Reference objects.</li>
<li>comment - Text with any kind of comment about the record.</li>
<li>features - A listing of Features making up the feature table.</li>
<li>base_counts - A string with the counts of bases for the sequence.</li>
<li>origin - A string specifying info about the origin of the sequence.</li>
<li>sequence - A string with the sequence itself.</li>
<li>contig - A string of location information for a CONTIG in a RefSeq file</li>
<li>project - The genome sequencing project numbers
(will be replaced by the dblink cross-references in 2009).</li>
<li>dblinks - The genome sequencing project number(s) and other links.
(will replace the project information in 2009).</li>
</ul>
</dd>
</dl>

<!-- ==================== INSTANCE METHODS ==================== -->
<a name="section-InstanceMethods"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Instance Methods</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-InstanceMethods"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.GenBank.Record.Record-class.html#__init__" class="summary-sig-name">__init__</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Initialize.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record.__init__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.GenBank.Record.Record-class.html#__str__" class="summary-sig-name">__str__</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Provide a GenBank formatted output option for a Record.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record.__str__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_locus_line"></a><span class="summary-sig-name">_locus_line</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Provide the output string for the LOCUS line (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record._locus_line">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_definition_line"></a><span class="summary-sig-name">_definition_line</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Provide output for the DEFINITION line (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record._definition_line">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_accession_line"></a><span class="summary-sig-name">_accession_line</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Output for the ACCESSION line (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record._accession_line">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_version_line"></a><span class="summary-sig-name">_version_line</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Output for the VERSION line (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record._version_line">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_project_line"></a><span class="summary-sig-name">_project_line</span>(<span class="summary-sig-arg">self</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record._project_line">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_dblink_line"></a><span class="summary-sig-name">_dblink_line</span>(<span class="summary-sig-arg">self</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record._dblink_line">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_nid_line"></a><span class="summary-sig-name">_nid_line</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Output for the NID line. Use of NID is obsolete in GenBank files (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record._nid_line">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_pid_line"></a><span class="summary-sig-name">_pid_line</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Output for PID line. Presumedly, PID usage is also obsolete (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record._pid_line">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_keywords_line"></a><span class="summary-sig-name">_keywords_line</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Output for the KEYWORDS line (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record._keywords_line">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_db_source_line"></a><span class="summary-sig-name">_db_source_line</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Output for DBSOURCE line (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record._db_source_line">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_segment_line"></a><span class="summary-sig-name">_segment_line</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Output for the SEGMENT line (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record._segment_line">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_source_line"></a><span class="summary-sig-name">_source_line</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Output for SOURCE line on where the sample came from (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record._source_line">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_organism_line"></a><span class="summary-sig-name">_organism_line</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Output for ORGANISM line with taxonomy info (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record._organism_line">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_comment_line"></a><span class="summary-sig-name">_comment_line</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Output for the COMMENT lines (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record._comment_line">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_features_line"></a><span class="summary-sig-name">_features_line</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Output for the FEATURES line (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record._features_line">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_base_count_line"></a><span class="summary-sig-name">_base_count_line</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Output for the BASE COUNT line with base information (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record._base_count_line">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_origin_line"></a><span class="summary-sig-name">_origin_line</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Output for the ORIGIN line (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record._origin_line">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_sequence_line"></a><span class="summary-sig-name">_sequence_line</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Output for all of the sequence (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record._sequence_line">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_wgs_line"></a><span class="summary-sig-name">_wgs_line</span>(<span class="summary-sig-arg">self</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record._wgs_line">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_wgs_scafld_line"></a><span class="summary-sig-name">_wgs_scafld_line</span>(<span class="summary-sig-arg">self</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record._wgs_scafld_line">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_contig_line"></a><span class="summary-sig-name">_contig_line</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Output for CONTIG location information from RefSeq (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record._contig_line">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__delattr__</code>,
      <code>__format__</code>,
      <code>__getattribute__</code>,
      <code>__hash__</code>,
      <code>__new__</code>,
      <code>__reduce__</code>,
      <code>__reduce_ex__</code>,
      <code>__repr__</code>,
      <code>__setattr__</code>,
      <code>__sizeof__</code>,
      <code>__subclasshook__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== CLASS VARIABLES ==================== -->
<a name="section-ClassVariables"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Class Variables</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-ClassVariables"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="GB_LINE_LENGTH"></a><span class="summary-name">GB_LINE_LENGTH</span> = <code title="79">79</code>
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="GB_BASE_INDENT"></a><span class="summary-name">GB_BASE_INDENT</span> = <code title="12">12</code>
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="GB_FEATURE_INDENT"></a><span class="summary-name">GB_FEATURE_INDENT</span> = <code title="21">21</code>
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="GB_INTERNAL_INDENT"></a><span class="summary-name">GB_INTERNAL_INDENT</span> = <code title="2">2</code>
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="GB_OTHER_INTERNAL_INDENT"></a><span class="summary-name">GB_OTHER_INTERNAL_INDENT</span> = <code title="3">3</code><br />
      hash(x)
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="GB_FEATURE_INTERNAL_INDENT"></a><span class="summary-name">GB_FEATURE_INTERNAL_INDENT</span> = <code title="5">5</code>
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="GB_SEQUENCE_INDENT"></a><span class="summary-name">GB_SEQUENCE_INDENT</span> = <code title="9">9</code>
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="BASE_FORMAT"></a><span class="summary-name">BASE_FORMAT</span> = <code title="'%-12s'"><code class="variable-quote">'</code><code class="variable-string">%-12s</code><code class="variable-quote">'</code></code>
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="INTERNAL_FORMAT"></a><span class="summary-name">INTERNAL_FORMAT</span> = <code title="'  %-10s'"><code class="variable-quote">'</code><code class="variable-string">  %-10s</code><code class="variable-quote">'</code></code>
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="OTHER_INTERNAL_FORMAT"></a><span class="summary-name">OTHER_INTERNAL_FORMAT</span> = <code title="'   %-9s'"><code class="variable-quote">'</code><code class="variable-string">   %-9s</code><code class="variable-quote">'</code></code>
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="BASE_FEATURE_FORMAT"></a><span class="summary-name">BASE_FEATURE_FORMAT</span> = <code title="'%-21s'"><code class="variable-quote">'</code><code class="variable-string">%-21s</code><code class="variable-quote">'</code></code>
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="INTERNAL_FEATURE_FORMAT"></a><span class="summary-name">INTERNAL_FEATURE_FORMAT</span> = <code title="'     %-16s'"><code class="variable-quote">'</code><code class="variable-string">     %-16s</code><code class="variable-quote">'</code></code>
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="SEQUENCE_FORMAT"></a><span class="summary-name">SEQUENCE_FORMAT</span> = <code title="'%9s'"><code class="variable-quote">'</code><code class="variable-string">%9s</code><code class="variable-quote">'</code></code>
    </td>
  </tr>
</table>
<!-- ==================== PROPERTIES ==================== -->
<a name="section-Properties"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Properties</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Properties"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__class__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== METHOD DETAILS ==================== -->
<a name="section-MethodDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Method Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-MethodDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="__init__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__init__</span>(<span class="sig-arg">self</span>)</span>
    <br /><em class="fname">(Constructor)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record.__init__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  Initialize.
  <dl class="fields">
    <dt>Overrides:
        object.__init__
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="__str__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__str__</span>(<span class="sig-arg">self</span>)</span>
    <br /><em class="fname">(Informal representation operator)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.GenBank.Record-pysrc.html#Record.__str__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Provide a GenBank formatted output option for a Record.</p>
<p>The objective of this is to provide an easy way to read in a GenBank
record, modify it somehow, and then output it in 'GenBank format.'
We are striving to make this work so that a parsed Record that is
output using this function will look exactly like the original
record.</p>
<p>Much of the output is based on format description info at:</p>
<p><a class="rst-reference external" href="ftp://ncbi.nlm.nih.gov/genbank/gbrel.txt" target="_top">ftp://ncbi.nlm.nih.gov/genbank/gbrel.txt</a></p>
  <dl class="fields">
    <dt>Overrides:
        object.__str__
    </dt>
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Tue Dec 18 15:56:28 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
