<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.Entrez.Parser</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.Entrez-module.html">Package&nbsp;Entrez</a> ::
        Module&nbsp;Parser
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.Entrez.Parser-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.Entrez.Parser-module.html">Module Bio.Entrez.Parser</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2008-2014 by Michiel de Hoon.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># Revisions copyright 2008-2015 by Peter Cock. All rights reserved.</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-docstring">"""Parser for XML results returned by NCBI's Entrez Utilities.</tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-docstring">This parser is used by the read() function in Bio.Entrez, and is not</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-docstring">intended be used directly.</tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-docstring">The question is how to represent an XML file as Python objects. Some</tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-docstring">XML files returned by NCBI look like lists, others look like dictionaries,</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-docstring">and others look like a mix of lists and dictionaries.</tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-docstring">My approach is to classify each possible element in the XML as a plain</tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-docstring">string, an integer, a list, a dictionary, or a structure. The latter is a</tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-docstring">dictionary where the same key can occur multiple times; in Python, it is</tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-docstring">represented as a dictionary where that key occurs once, pointing to a list</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-docstring">of values found in the XML file.</tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-docstring">The parser then goes through the XML and creates the appropriate Python</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-docstring">object for each element. The different levels encountered in the XML are</tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-docstring">preserved on the Python side. So a subelement of a subelement of an element</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-docstring">is a value in a dictionary that is stored in a list which is a value in</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-docstring">some other dictionary (or a value in a list which itself belongs to a list</tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"><tt class="py-docstring">which is a value in a dictionary, and so on). Attributes encountered in</tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt class="py-docstring">the XML are stored as a dictionary in a member .attributes of each element,</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"><tt class="py-docstring">and the tag name is saved in a member .tag.</tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"><tt class="py-docstring">To decide which kind of Python object corresponds to each element in the</tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"><tt class="py-docstring">XML, the parser analyzes the DTD referred at the top of (almost) every</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-docstring">XML file returned by the Entrez Utilities. This is preferred over a hand-</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-docstring">written solution, since the number of DTDs is rather large and their</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-docstring">contents may change over time. About half the code in this parser deals</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"><tt class="py-docstring">with parsing the DTD, and the other half with the XML itself.</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">sys</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">re</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">os</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">collections</tt> <tt class="py-keyword">import</tt> <tt class="py-name">Counter</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">xml</tt><tt class="py-op">.</tt><tt class="py-name">parsers</tt> <tt class="py-keyword">import</tt> <tt class="py-name">expat</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">io</tt> <tt class="py-keyword">import</tt> <tt class="py-name">BytesIO</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">xml</tt><tt class="py-op">.</tt><tt class="py-name">etree</tt><tt class="py-op">.</tt><tt class="py-name">ElementTree</tt> <tt class="py-keyword">as</tt> <tt class="py-name">ET</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">xml</tt><tt class="py-op">.</tt><tt class="py-name">sax</tt><tt class="py-op">.</tt><tt class="py-name">saxutils</tt> <tt class="py-keyword">import</tt> <tt class="py-name">escape</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line"> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"><tt class="py-comment"># Importing these functions with leading underscore as not intended for reuse</tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-1', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">urlopen</tt> <tt class="py-keyword">as</tt> <tt class="py-name">_urlopen</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-2" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-2', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-3" class="py-name"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-3', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">urlparse</tt> <tt class="py-keyword">as</tt> <tt class="py-name">_urlparse</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-4" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-4', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-5" class="py-name"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-5', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">unicode</tt> </tt>
<a name="IntegerElement"></a><div id="IntegerElement-def"><a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line"> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line"> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line"><tt class="py-comment"># The following four classes are used to add a member .attributes to integers,</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line"><tt class="py-comment"># strings, lists, and dictionaries, respectively.</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line"> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt> <a class="py-toggle" href="#" id="IntegerElement-toggle" onclick="return toggle('IntegerElement');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Entrez.Parser.IntegerElement-class.html">IntegerElement</a><tt class="py-op">(</tt><tt class="py-base-class">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="IntegerElement-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="IntegerElement-expanded"><a name="IntegerElement.__repr__"></a><div id="IntegerElement.__repr__-def"><a name="L59"></a><tt class="py-lineno"> 59</tt> <a class="py-toggle" href="#" id="IntegerElement.__repr__-toggle" onclick="return toggle('IntegerElement.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.IntegerElement-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="IntegerElement.__repr__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="IntegerElement.__repr__-expanded"><a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line">        <tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">.</tt><tt id="link-6" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.__repr__()=Bio.Align.MultipleSeqAlignment-class.html#__repr__,Method Bio.AlignIO.MafIO.MafIndex.__repr__()=Bio.AlignIO.MafIO.MafIndex-class.html#__repr__,Method Bio.Alphabet.Alphabet.__repr__()=Bio.Alphabet.Alphabet-class.html#__repr__,Method Bio.Alphabet.AlphabetEncoder.__repr__()=Bio.Alphabet.AlphabetEncoder-class.html#__repr__,Method Bio.Application.AbstractCommandline.__repr__()=Bio.Application.AbstractCommandline-class.html#__repr__,Method Bio.Application.ApplicationError.__repr__()=Bio.Application.ApplicationError-class.html#__repr__,Method Bio.Crystal.Crystal.__repr__()=Bio.Crystal.Crystal-class.html#__repr__,Method Bio.Crystal.Hetero.__repr__()=Bio.Crystal.Hetero-class.html#__repr__,Method Bio.Entrez.Parser.DictionaryElement.__repr__()=Bio.Entrez.Parser.DictionaryElement-class.html#__repr__,Method Bio.Entrez.Parser.IntegerElement.__repr__()=Bio.Entrez.Parser.IntegerElement-class.html#__repr__,Method Bio.Entrez.Parser.ListElement.__repr__()=Bio.Entrez.Parser.ListElement-class.html#__repr__,Method Bio.Entrez.Parser.StringElement.__repr__()=Bio.Entrez.Parser.StringElement-class.html#__repr__,Method Bio.Entrez.Parser.UnicodeElement.__repr__()=Bio.Entrez.Parser.UnicodeElement-class.html#__repr__,Method Bio.ExPASy.Enzyme.Record.__repr__()=Bio.ExPASy.Enzyme.Record-class.html#__repr__,Method Bio.ExPASy.cellosaurus.Record.__repr__()=Bio.ExPASy.cellosaurus.Record-class.html#__repr__,Method Bio.FSSP.FSSPSumRec.__repr__()=Bio.FSSP.FSSPSumRec-class.html#__repr__,Method Bio.FSSP.PosAlign.__repr__()=Bio.FSSP.PosAlign-class.html#__repr__,Method Bio.FSSP.fssp_rec.fff_rec.__repr__()=Bio.FSSP.fssp_rec.fff_rec-class.html#__repr__,Method Bio.File._IndexedSeqFileDict.__repr__()=Bio.File._IndexedSeqFileDict-class.html#__repr__,Method Bio.File._SQLiteManySeqFilesDict.__repr__()=Bio.File._SQLiteManySeqFilesDict-class.html#__repr__,Method Bio.GenBank.Record.Feature.__repr__()=Bio.GenBank.Record.Feature-class.html#__repr__,Method Bio.GenBank.Record.Qualifier.__repr__()=Bio.GenBank.Record.Qualifier-class.html#__repr__,Method Bio.PDB.Atom.Atom.__repr__()=Bio.PDB.Atom.Atom-class.html#__repr__,Method Bio.PDB.Atom.DisorderedAtom.__repr__()=Bio.PDB.Atom.DisorderedAtom-class.html#__repr__,Method Bio.PDB.Chain.Chain.__repr__()=Bio.PDB.Chain.Chain-class.html#__repr__,Method Bio.PDB.FragmentMapper'.Fragment.__repr__()=Bio.PDB.FragmentMapper%27.Fragment-class.html#__repr__,Method Bio.PDB.Model.Model.__repr__()=Bio.PDB.Model.Model-class.html#__repr__,Method Bio.PDB.PDBIO'.Select.__repr__()=Bio.PDB.PDBIO%27.Select-class.html#__repr__,Method Bio.PDB.Polypeptide.Polypeptide.__repr__()=Bio.PDB.Polypeptide.Polypeptide-class.html#__repr__,Method Bio.PDB.Residue.DisorderedResidue.__repr__()=Bio.PDB.Residue.DisorderedResidue-class.html#__repr__,Method Bio.PDB.Residue.Residue.__repr__()=Bio.PDB.Residue.Residue-class.html#__repr__,Method Bio.PDB.Structure.Structure.__repr__()=Bio.PDB.Structure.Structure-class.html#__repr__,Method Bio.PDB.vectors.Vector.__repr__()=Bio.PDB.vectors.Vector-class.html#__repr__,Method Bio.Pathway.Interaction.__repr__()=Bio.Pathway.Interaction-class.html#__repr__,Method Bio.Pathway.Network.__repr__()=Bio.Pathway.Network-class.html#__repr__,Method Bio.Pathway.Reaction.__repr__()=Bio.Pathway.Reaction-class.html#__repr__,Method Bio.Pathway.Rep.Graph.Graph.__repr__()=Bio.Pathway.Rep.Graph.Graph-class.html#__repr__,Method Bio.Pathway.Rep.MultiGraph.MultiGraph.__repr__()=Bio.Pathway.Rep.MultiGraph.MultiGraph-class.html#__repr__,Method Bio.Pathway.System.__repr__()=Bio.Pathway.System-class.html#__repr__,Method Bio.Phylo.BaseTree.BranchColor.__repr__()=Bio.Phylo.BaseTree.BranchColor-class.html#__repr__,Method Bio.Phylo.BaseTree.TreeElement.__repr__()=Bio.Phylo.BaseTree.TreeElement-class.html#__repr__,Method Bio.Phylo.Consensus._BitString.__repr__()=Bio.Phylo.Consensus._BitString-class.html#__repr__,Method Bio.Phylo.TreeConstruction._Matrix.__repr__()=Bio.Phylo.TreeConstruction._Matrix-class.html#__repr__,Method Bio.Restriction.Restriction.Analysis.__repr__()=Bio.Restriction.Restriction.Analysis-class.html#__repr__,Method Bio.Restriction.Restriction.FormattedSeq.__repr__()=Bio.Restriction.Restriction.FormattedSeq-class.html#__repr__,Method Bio.Restriction.Restriction.RestrictionBatch.__repr__()=Bio.Restriction.Restriction.RestrictionBatch-class.html#__repr__,Method Bio.Restriction.Restriction.RestrictionType.__repr__()=Bio.Restriction.Restriction.RestrictionType-class.html#__repr__,Method Bio.SearchIO._model.hit.Hit.__repr__()=Bio.SearchIO._model.hit.Hit-class.html#__repr__,Method Bio.SearchIO._model.hsp.HSP.__repr__()=Bio.SearchIO._model.hsp.HSP-class.html#__repr__,Method Bio.SearchIO._model.hsp.HSPFragment.__repr__()=Bio.SearchIO._model.hsp.HSPFragment-class.html#__repr__,Method Bio.SearchIO._model.query.QueryResult.__repr__()=Bio.SearchIO._model.query.QueryResult-class.html#__repr__,Method Bio.Seq.MutableSeq.__repr__()=Bio.Seq.MutableSeq-class.html#__repr__,Method Bio.Seq.Seq.__repr__()=Bio.Seq.Seq-class.html#__repr__,Method Bio.Seq.UnknownSeq.__repr__()=Bio.Seq.UnknownSeq-class.html#__repr__,Method Bio.SeqFeature.AbstractPosition.__repr__()=Bio.SeqFeature.AbstractPosition-class.html#__repr__,Method Bio.SeqFeature.AfterPosition.__repr__()=Bio.SeqFeature.AfterPosition-class.html#__repr__,Method Bio.SeqFeature.BeforePosition.__repr__()=Bio.SeqFeature.BeforePosition-class.html#__repr__,Method Bio.SeqFeature.BetweenPosition.__repr__()=Bio.SeqFeature.BetweenPosition-class.html#__repr__,Method Bio.SeqFeature.CompoundLocation.__repr__()=Bio.SeqFeature.CompoundLocation-class.html#__repr__,Method Bio.SeqFeature.ExactPosition.__repr__()=Bio.SeqFeature.ExactPosition-class.html#__repr__,Method Bio.SeqFeature.FeatureLocation.__repr__()=Bio.SeqFeature.FeatureLocation-class.html#__repr__,Method Bio.SeqFeature.OneOfPosition.__repr__()=Bio.SeqFeature.OneOfPosition-class.html#__repr__,Method Bio.SeqFeature.PositionGap.__repr__()=Bio.SeqFeature.PositionGap-class.html#__repr__,Method Bio.SeqFeature.Reference.__repr__()=Bio.SeqFeature.Reference-class.html#__repr__,Method Bio.SeqFeature.SeqFeature.__repr__()=Bio.SeqFeature.SeqFeature-class.html#__repr__,Method Bio.SeqFeature.UnknownPosition.__repr__()=Bio.SeqFeature.UnknownPosition-class.html#__repr__,Method Bio.SeqFeature.WithinPosition.__repr__()=Bio.SeqFeature.WithinPosition-class.html#__repr__,Method Bio.SeqRecord.SeqRecord.__repr__()=Bio.SeqRecord.SeqRecord-class.html#__repr__,Method Bio.UniGene.ProtsimLine.__repr__()=Bio.UniGene.ProtsimLine-class.html#__repr__,Method Bio.UniGene.Record.__repr__()=Bio.UniGene.Record-class.html#__repr__,Method Bio.UniGene.STSLine.__repr__()=Bio.UniGene.STSLine-class.html#__repr__,Method Bio.UniGene.SequenceLine.__repr__()=Bio.UniGene.SequenceLine-class.html#__repr__,Method Bio.Wise.psw.AlignmentColumn.__repr__()=Bio.Wise.psw.AlignmentColumn-class.html#__repr__,Method Bio.codonalign.codonalphabet.CodonAlphabet.__repr__()=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#__repr__,Method Bio.phenotype.phen_micro.PlateRecord.__repr__()=Bio.phenotype.phen_micro.PlateRecord-class.html#__repr__,Method Bio.phenotype.phen_micro.WellRecord.__repr__()=Bio.phenotype.phen_micro.WellRecord-class.html#__repr__,Method BioSQL.BioSeqDatabase.BioSeqDatabase.__repr__()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#__repr__,Method BioSQL.BioSeqDatabase.DBServer.__repr__()=BioSQL.BioSeqDatabase.DBServer-class.html#__repr__"><a title="Bio.Align.MultipleSeqAlignment.__repr__
Bio.AlignIO.MafIO.MafIndex.__repr__
Bio.Alphabet.Alphabet.__repr__
Bio.Alphabet.AlphabetEncoder.__repr__
Bio.Application.AbstractCommandline.__repr__
Bio.Application.ApplicationError.__repr__
Bio.Crystal.Crystal.__repr__
Bio.Crystal.Hetero.__repr__
Bio.Entrez.Parser.DictionaryElement.__repr__
Bio.Entrez.Parser.IntegerElement.__repr__
Bio.Entrez.Parser.ListElement.__repr__
Bio.Entrez.Parser.StringElement.__repr__
Bio.Entrez.Parser.UnicodeElement.__repr__
Bio.ExPASy.Enzyme.Record.__repr__
Bio.ExPASy.cellosaurus.Record.__repr__
Bio.FSSP.FSSPSumRec.__repr__
Bio.FSSP.PosAlign.__repr__
Bio.FSSP.fssp_rec.fff_rec.__repr__
Bio.File._IndexedSeqFileDict.__repr__
Bio.File._SQLiteManySeqFilesDict.__repr__
Bio.GenBank.Record.Feature.__repr__
Bio.GenBank.Record.Qualifier.__repr__
Bio.PDB.Atom.Atom.__repr__
Bio.PDB.Atom.DisorderedAtom.__repr__
Bio.PDB.Chain.Chain.__repr__
Bio.PDB.FragmentMapper'.Fragment.__repr__
Bio.PDB.Model.Model.__repr__
Bio.PDB.PDBIO'.Select.__repr__
Bio.PDB.Polypeptide.Polypeptide.__repr__
Bio.PDB.Residue.DisorderedResidue.__repr__
Bio.PDB.Residue.Residue.__repr__
Bio.PDB.Structure.Structure.__repr__
Bio.PDB.vectors.Vector.__repr__
Bio.Pathway.Interaction.__repr__
Bio.Pathway.Network.__repr__
Bio.Pathway.Reaction.__repr__
Bio.Pathway.Rep.Graph.Graph.__repr__
Bio.Pathway.Rep.MultiGraph.MultiGraph.__repr__
Bio.Pathway.System.__repr__
Bio.Phylo.BaseTree.BranchColor.__repr__
Bio.Phylo.BaseTree.TreeElement.__repr__
Bio.Phylo.Consensus._BitString.__repr__
Bio.Phylo.TreeConstruction._Matrix.__repr__
Bio.Restriction.Restriction.Analysis.__repr__
Bio.Restriction.Restriction.FormattedSeq.__repr__
Bio.Restriction.Restriction.RestrictionBatch.__repr__
Bio.Restriction.Restriction.RestrictionType.__repr__
Bio.SearchIO._model.hit.Hit.__repr__
Bio.SearchIO._model.hsp.HSP.__repr__
Bio.SearchIO._model.hsp.HSPFragment.__repr__
Bio.SearchIO._model.query.QueryResult.__repr__
Bio.Seq.MutableSeq.__repr__
Bio.Seq.Seq.__repr__
Bio.Seq.UnknownSeq.__repr__
Bio.SeqFeature.AbstractPosition.__repr__
Bio.SeqFeature.AfterPosition.__repr__
Bio.SeqFeature.BeforePosition.__repr__
Bio.SeqFeature.BetweenPosition.__repr__
Bio.SeqFeature.CompoundLocation.__repr__
Bio.SeqFeature.ExactPosition.__repr__
Bio.SeqFeature.FeatureLocation.__repr__
Bio.SeqFeature.OneOfPosition.__repr__
Bio.SeqFeature.PositionGap.__repr__
Bio.SeqFeature.Reference.__repr__
Bio.SeqFeature.SeqFeature.__repr__
Bio.SeqFeature.UnknownPosition.__repr__
Bio.SeqFeature.WithinPosition.__repr__
Bio.SeqRecord.SeqRecord.__repr__
Bio.UniGene.ProtsimLine.__repr__
Bio.UniGene.Record.__repr__
Bio.UniGene.STSLine.__repr__
Bio.UniGene.SequenceLine.__repr__
Bio.Wise.psw.AlignmentColumn.__repr__
Bio.codonalign.codonalphabet.CodonAlphabet.__repr__
Bio.phenotype.phen_micro.PlateRecord.__repr__
Bio.phenotype.phen_micro.WellRecord.__repr__
BioSQL.BioSeqDatabase.BioSeqDatabase.__repr__
BioSQL.BioSeqDatabase.DBServer.__repr__" class="py-name" href="#" onclick="return doclink('link-6', '__repr__', 'link-6');">__repr__</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line">            <tt class="py-name">attributes</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">attributes</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">text</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">"IntegerElement(%s, attributes=%s)"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">,</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">attributes</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line"> </tt>
<a name="StringElement"></a><div id="StringElement-def"><a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line"> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt> <a class="py-toggle" href="#" id="StringElement-toggle" onclick="return toggle('StringElement');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Entrez.Parser.StringElement-class.html">StringElement</a><tt class="py-op">(</tt><tt class="py-base-class">str</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StringElement-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="StringElement-expanded"><a name="StringElement.__repr__"></a><div id="StringElement.__repr__-def"><a name="L69"></a><tt class="py-lineno"> 69</tt> <a class="py-toggle" href="#" id="StringElement.__repr__-toggle" onclick="return toggle('StringElement.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.StringElement-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StringElement.__repr__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="StringElement.__repr__-expanded"><a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line">        <tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">.</tt><tt id="link-7" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.__repr__
Bio.AlignIO.MafIO.MafIndex.__repr__
Bio.Alphabet.Alphabet.__repr__
Bio.Alphabet.AlphabetEncoder.__repr__
Bio.Application.AbstractCommandline.__repr__
Bio.Application.ApplicationError.__repr__
Bio.Crystal.Crystal.__repr__
Bio.Crystal.Hetero.__repr__
Bio.Entrez.Parser.DictionaryElement.__repr__
Bio.Entrez.Parser.IntegerElement.__repr__
Bio.Entrez.Parser.ListElement.__repr__
Bio.Entrez.Parser.StringElement.__repr__
Bio.Entrez.Parser.UnicodeElement.__repr__
Bio.ExPASy.Enzyme.Record.__repr__
Bio.ExPASy.cellosaurus.Record.__repr__
Bio.FSSP.FSSPSumRec.__repr__
Bio.FSSP.PosAlign.__repr__
Bio.FSSP.fssp_rec.fff_rec.__repr__
Bio.File._IndexedSeqFileDict.__repr__
Bio.File._SQLiteManySeqFilesDict.__repr__
Bio.GenBank.Record.Feature.__repr__
Bio.GenBank.Record.Qualifier.__repr__
Bio.PDB.Atom.Atom.__repr__
Bio.PDB.Atom.DisorderedAtom.__repr__
Bio.PDB.Chain.Chain.__repr__
Bio.PDB.FragmentMapper'.Fragment.__repr__
Bio.PDB.Model.Model.__repr__
Bio.PDB.PDBIO'.Select.__repr__
Bio.PDB.Polypeptide.Polypeptide.__repr__
Bio.PDB.Residue.DisorderedResidue.__repr__
Bio.PDB.Residue.Residue.__repr__
Bio.PDB.Structure.Structure.__repr__
Bio.PDB.vectors.Vector.__repr__
Bio.Pathway.Interaction.__repr__
Bio.Pathway.Network.__repr__
Bio.Pathway.Reaction.__repr__
Bio.Pathway.Rep.Graph.Graph.__repr__
Bio.Pathway.Rep.MultiGraph.MultiGraph.__repr__
Bio.Pathway.System.__repr__
Bio.Phylo.BaseTree.BranchColor.__repr__
Bio.Phylo.BaseTree.TreeElement.__repr__
Bio.Phylo.Consensus._BitString.__repr__
Bio.Phylo.TreeConstruction._Matrix.__repr__
Bio.Restriction.Restriction.Analysis.__repr__
Bio.Restriction.Restriction.FormattedSeq.__repr__
Bio.Restriction.Restriction.RestrictionBatch.__repr__
Bio.Restriction.Restriction.RestrictionType.__repr__
Bio.SearchIO._model.hit.Hit.__repr__
Bio.SearchIO._model.hsp.HSP.__repr__
Bio.SearchIO._model.hsp.HSPFragment.__repr__
Bio.SearchIO._model.query.QueryResult.__repr__
Bio.Seq.MutableSeq.__repr__
Bio.Seq.Seq.__repr__
Bio.Seq.UnknownSeq.__repr__
Bio.SeqFeature.AbstractPosition.__repr__
Bio.SeqFeature.AfterPosition.__repr__
Bio.SeqFeature.BeforePosition.__repr__
Bio.SeqFeature.BetweenPosition.__repr__
Bio.SeqFeature.CompoundLocation.__repr__
Bio.SeqFeature.ExactPosition.__repr__
Bio.SeqFeature.FeatureLocation.__repr__
Bio.SeqFeature.OneOfPosition.__repr__
Bio.SeqFeature.PositionGap.__repr__
Bio.SeqFeature.Reference.__repr__
Bio.SeqFeature.SeqFeature.__repr__
Bio.SeqFeature.UnknownPosition.__repr__
Bio.SeqFeature.WithinPosition.__repr__
Bio.SeqRecord.SeqRecord.__repr__
Bio.UniGene.ProtsimLine.__repr__
Bio.UniGene.Record.__repr__
Bio.UniGene.STSLine.__repr__
Bio.UniGene.SequenceLine.__repr__
Bio.Wise.psw.AlignmentColumn.__repr__
Bio.codonalign.codonalphabet.CodonAlphabet.__repr__
Bio.phenotype.phen_micro.PlateRecord.__repr__
Bio.phenotype.phen_micro.WellRecord.__repr__
BioSQL.BioSeqDatabase.BioSeqDatabase.__repr__
BioSQL.BioSeqDatabase.DBServer.__repr__" class="py-name" href="#" onclick="return doclink('link-7', '__repr__', 'link-6');">__repr__</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line">            <tt class="py-name">attributes</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">attributes</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">text</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">"StringElement(%s, attributes=%s)"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">,</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">attributes</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line"> </tt>
<a name="UnicodeElement"></a><div id="UnicodeElement-def"><a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line"> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt> <a class="py-toggle" href="#" id="UnicodeElement-toggle" onclick="return toggle('UnicodeElement');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Entrez.Parser.UnicodeElement-class.html">UnicodeElement</a><tt class="py-op">(</tt><tt class="py-base-class">unicode</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnicodeElement-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="UnicodeElement-expanded"><a name="UnicodeElement.__repr__"></a><div id="UnicodeElement.__repr__-def"><a name="L79"></a><tt class="py-lineno"> 79</tt> <a class="py-toggle" href="#" id="UnicodeElement.__repr__-toggle" onclick="return toggle('UnicodeElement.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.UnicodeElement-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnicodeElement.__repr__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="UnicodeElement.__repr__-expanded"><a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line">        <tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt class="py-name">unicode</tt><tt class="py-op">.</tt><tt id="link-8" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.__repr__
Bio.AlignIO.MafIO.MafIndex.__repr__
Bio.Alphabet.Alphabet.__repr__
Bio.Alphabet.AlphabetEncoder.__repr__
Bio.Application.AbstractCommandline.__repr__
Bio.Application.ApplicationError.__repr__
Bio.Crystal.Crystal.__repr__
Bio.Crystal.Hetero.__repr__
Bio.Entrez.Parser.DictionaryElement.__repr__
Bio.Entrez.Parser.IntegerElement.__repr__
Bio.Entrez.Parser.ListElement.__repr__
Bio.Entrez.Parser.StringElement.__repr__
Bio.Entrez.Parser.UnicodeElement.__repr__
Bio.ExPASy.Enzyme.Record.__repr__
Bio.ExPASy.cellosaurus.Record.__repr__
Bio.FSSP.FSSPSumRec.__repr__
Bio.FSSP.PosAlign.__repr__
Bio.FSSP.fssp_rec.fff_rec.__repr__
Bio.File._IndexedSeqFileDict.__repr__
Bio.File._SQLiteManySeqFilesDict.__repr__
Bio.GenBank.Record.Feature.__repr__
Bio.GenBank.Record.Qualifier.__repr__
Bio.PDB.Atom.Atom.__repr__
Bio.PDB.Atom.DisorderedAtom.__repr__
Bio.PDB.Chain.Chain.__repr__
Bio.PDB.FragmentMapper'.Fragment.__repr__
Bio.PDB.Model.Model.__repr__
Bio.PDB.PDBIO'.Select.__repr__
Bio.PDB.Polypeptide.Polypeptide.__repr__
Bio.PDB.Residue.DisorderedResidue.__repr__
Bio.PDB.Residue.Residue.__repr__
Bio.PDB.Structure.Structure.__repr__
Bio.PDB.vectors.Vector.__repr__
Bio.Pathway.Interaction.__repr__
Bio.Pathway.Network.__repr__
Bio.Pathway.Reaction.__repr__
Bio.Pathway.Rep.Graph.Graph.__repr__
Bio.Pathway.Rep.MultiGraph.MultiGraph.__repr__
Bio.Pathway.System.__repr__
Bio.Phylo.BaseTree.BranchColor.__repr__
Bio.Phylo.BaseTree.TreeElement.__repr__
Bio.Phylo.Consensus._BitString.__repr__
Bio.Phylo.TreeConstruction._Matrix.__repr__
Bio.Restriction.Restriction.Analysis.__repr__
Bio.Restriction.Restriction.FormattedSeq.__repr__
Bio.Restriction.Restriction.RestrictionBatch.__repr__
Bio.Restriction.Restriction.RestrictionType.__repr__
Bio.SearchIO._model.hit.Hit.__repr__
Bio.SearchIO._model.hsp.HSP.__repr__
Bio.SearchIO._model.hsp.HSPFragment.__repr__
Bio.SearchIO._model.query.QueryResult.__repr__
Bio.Seq.MutableSeq.__repr__
Bio.Seq.Seq.__repr__
Bio.Seq.UnknownSeq.__repr__
Bio.SeqFeature.AbstractPosition.__repr__
Bio.SeqFeature.AfterPosition.__repr__
Bio.SeqFeature.BeforePosition.__repr__
Bio.SeqFeature.BetweenPosition.__repr__
Bio.SeqFeature.CompoundLocation.__repr__
Bio.SeqFeature.ExactPosition.__repr__
Bio.SeqFeature.FeatureLocation.__repr__
Bio.SeqFeature.OneOfPosition.__repr__
Bio.SeqFeature.PositionGap.__repr__
Bio.SeqFeature.Reference.__repr__
Bio.SeqFeature.SeqFeature.__repr__
Bio.SeqFeature.UnknownPosition.__repr__
Bio.SeqFeature.WithinPosition.__repr__
Bio.SeqRecord.SeqRecord.__repr__
Bio.UniGene.ProtsimLine.__repr__
Bio.UniGene.Record.__repr__
Bio.UniGene.STSLine.__repr__
Bio.UniGene.SequenceLine.__repr__
Bio.Wise.psw.AlignmentColumn.__repr__
Bio.codonalign.codonalphabet.CodonAlphabet.__repr__
Bio.phenotype.phen_micro.PlateRecord.__repr__
Bio.phenotype.phen_micro.WellRecord.__repr__
BioSQL.BioSeqDatabase.BioSeqDatabase.__repr__
BioSQL.BioSeqDatabase.DBServer.__repr__" class="py-name" href="#" onclick="return doclink('link-8', '__repr__', 'link-6');">__repr__</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line">            <tt class="py-name">attributes</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">attributes</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">text</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">"UnicodeElement(%s, attributes=%s)"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">,</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">attributes</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line"> </tt>
<a name="ListElement"></a><div id="ListElement-def"><a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line"> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt> <a class="py-toggle" href="#" id="ListElement-toggle" onclick="return toggle('ListElement');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Entrez.Parser.ListElement-class.html">ListElement</a><tt class="py-op">(</tt><tt class="py-base-class">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ListElement-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="ListElement-expanded"><a name="ListElement.__repr__"></a><div id="ListElement.__repr__-def"><a name="L89"></a><tt class="py-lineno"> 89</tt> <a class="py-toggle" href="#" id="ListElement.__repr__-toggle" onclick="return toggle('ListElement.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.ListElement-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ListElement.__repr__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ListElement.__repr__-expanded"><a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line">        <tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">.</tt><tt id="link-9" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.__repr__
Bio.AlignIO.MafIO.MafIndex.__repr__
Bio.Alphabet.Alphabet.__repr__
Bio.Alphabet.AlphabetEncoder.__repr__
Bio.Application.AbstractCommandline.__repr__
Bio.Application.ApplicationError.__repr__
Bio.Crystal.Crystal.__repr__
Bio.Crystal.Hetero.__repr__
Bio.Entrez.Parser.DictionaryElement.__repr__
Bio.Entrez.Parser.IntegerElement.__repr__
Bio.Entrez.Parser.ListElement.__repr__
Bio.Entrez.Parser.StringElement.__repr__
Bio.Entrez.Parser.UnicodeElement.__repr__
Bio.ExPASy.Enzyme.Record.__repr__
Bio.ExPASy.cellosaurus.Record.__repr__
Bio.FSSP.FSSPSumRec.__repr__
Bio.FSSP.PosAlign.__repr__
Bio.FSSP.fssp_rec.fff_rec.__repr__
Bio.File._IndexedSeqFileDict.__repr__
Bio.File._SQLiteManySeqFilesDict.__repr__
Bio.GenBank.Record.Feature.__repr__
Bio.GenBank.Record.Qualifier.__repr__
Bio.PDB.Atom.Atom.__repr__
Bio.PDB.Atom.DisorderedAtom.__repr__
Bio.PDB.Chain.Chain.__repr__
Bio.PDB.FragmentMapper'.Fragment.__repr__
Bio.PDB.Model.Model.__repr__
Bio.PDB.PDBIO'.Select.__repr__
Bio.PDB.Polypeptide.Polypeptide.__repr__
Bio.PDB.Residue.DisorderedResidue.__repr__
Bio.PDB.Residue.Residue.__repr__
Bio.PDB.Structure.Structure.__repr__
Bio.PDB.vectors.Vector.__repr__
Bio.Pathway.Interaction.__repr__
Bio.Pathway.Network.__repr__
Bio.Pathway.Reaction.__repr__
Bio.Pathway.Rep.Graph.Graph.__repr__
Bio.Pathway.Rep.MultiGraph.MultiGraph.__repr__
Bio.Pathway.System.__repr__
Bio.Phylo.BaseTree.BranchColor.__repr__
Bio.Phylo.BaseTree.TreeElement.__repr__
Bio.Phylo.Consensus._BitString.__repr__
Bio.Phylo.TreeConstruction._Matrix.__repr__
Bio.Restriction.Restriction.Analysis.__repr__
Bio.Restriction.Restriction.FormattedSeq.__repr__
Bio.Restriction.Restriction.RestrictionBatch.__repr__
Bio.Restriction.Restriction.RestrictionType.__repr__
Bio.SearchIO._model.hit.Hit.__repr__
Bio.SearchIO._model.hsp.HSP.__repr__
Bio.SearchIO._model.hsp.HSPFragment.__repr__
Bio.SearchIO._model.query.QueryResult.__repr__
Bio.Seq.MutableSeq.__repr__
Bio.Seq.Seq.__repr__
Bio.Seq.UnknownSeq.__repr__
Bio.SeqFeature.AbstractPosition.__repr__
Bio.SeqFeature.AfterPosition.__repr__
Bio.SeqFeature.BeforePosition.__repr__
Bio.SeqFeature.BetweenPosition.__repr__
Bio.SeqFeature.CompoundLocation.__repr__
Bio.SeqFeature.ExactPosition.__repr__
Bio.SeqFeature.FeatureLocation.__repr__
Bio.SeqFeature.OneOfPosition.__repr__
Bio.SeqFeature.PositionGap.__repr__
Bio.SeqFeature.Reference.__repr__
Bio.SeqFeature.SeqFeature.__repr__
Bio.SeqFeature.UnknownPosition.__repr__
Bio.SeqFeature.WithinPosition.__repr__
Bio.SeqRecord.SeqRecord.__repr__
Bio.UniGene.ProtsimLine.__repr__
Bio.UniGene.Record.__repr__
Bio.UniGene.STSLine.__repr__
Bio.UniGene.SequenceLine.__repr__
Bio.Wise.psw.AlignmentColumn.__repr__
Bio.codonalign.codonalphabet.CodonAlphabet.__repr__
Bio.phenotype.phen_micro.PlateRecord.__repr__
Bio.phenotype.phen_micro.WellRecord.__repr__
BioSQL.BioSeqDatabase.BioSeqDatabase.__repr__
BioSQL.BioSeqDatabase.DBServer.__repr__" class="py-name" href="#" onclick="return doclink('link-9', '__repr__', 'link-6');">__repr__</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line">            <tt class="py-name">attributes</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">attributes</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">text</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">"ListElement(%s, attributes=%s)"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">,</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">attributes</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line"> </tt>
<a name="DictionaryElement"></a><div id="DictionaryElement-def"><a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line"> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt> <a class="py-toggle" href="#" id="DictionaryElement-toggle" onclick="return toggle('DictionaryElement');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DictionaryElement-class.html">DictionaryElement</a><tt class="py-op">(</tt><tt class="py-base-class">dict</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DictionaryElement-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="DictionaryElement-expanded"><a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line"> </tt>
<a name="DictionaryElement.__repr__"></a><div id="DictionaryElement.__repr__-def"><a name="L100"></a><tt class="py-lineno">100</tt> <a class="py-toggle" href="#" id="DictionaryElement.__repr__-toggle" onclick="return toggle('DictionaryElement.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DictionaryElement-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DictionaryElement.__repr__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DictionaryElement.__repr__-expanded"><a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line">        <tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">.</tt><tt id="link-10" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.__repr__
Bio.AlignIO.MafIO.MafIndex.__repr__
Bio.Alphabet.Alphabet.__repr__
Bio.Alphabet.AlphabetEncoder.__repr__
Bio.Application.AbstractCommandline.__repr__
Bio.Application.ApplicationError.__repr__
Bio.Crystal.Crystal.__repr__
Bio.Crystal.Hetero.__repr__
Bio.Entrez.Parser.DictionaryElement.__repr__
Bio.Entrez.Parser.IntegerElement.__repr__
Bio.Entrez.Parser.ListElement.__repr__
Bio.Entrez.Parser.StringElement.__repr__
Bio.Entrez.Parser.UnicodeElement.__repr__
Bio.ExPASy.Enzyme.Record.__repr__
Bio.ExPASy.cellosaurus.Record.__repr__
Bio.FSSP.FSSPSumRec.__repr__
Bio.FSSP.PosAlign.__repr__
Bio.FSSP.fssp_rec.fff_rec.__repr__
Bio.File._IndexedSeqFileDict.__repr__
Bio.File._SQLiteManySeqFilesDict.__repr__
Bio.GenBank.Record.Feature.__repr__
Bio.GenBank.Record.Qualifier.__repr__
Bio.PDB.Atom.Atom.__repr__
Bio.PDB.Atom.DisorderedAtom.__repr__
Bio.PDB.Chain.Chain.__repr__
Bio.PDB.FragmentMapper'.Fragment.__repr__
Bio.PDB.Model.Model.__repr__
Bio.PDB.PDBIO'.Select.__repr__
Bio.PDB.Polypeptide.Polypeptide.__repr__
Bio.PDB.Residue.DisorderedResidue.__repr__
Bio.PDB.Residue.Residue.__repr__
Bio.PDB.Structure.Structure.__repr__
Bio.PDB.vectors.Vector.__repr__
Bio.Pathway.Interaction.__repr__
Bio.Pathway.Network.__repr__
Bio.Pathway.Reaction.__repr__
Bio.Pathway.Rep.Graph.Graph.__repr__
Bio.Pathway.Rep.MultiGraph.MultiGraph.__repr__
Bio.Pathway.System.__repr__
Bio.Phylo.BaseTree.BranchColor.__repr__
Bio.Phylo.BaseTree.TreeElement.__repr__
Bio.Phylo.Consensus._BitString.__repr__
Bio.Phylo.TreeConstruction._Matrix.__repr__
Bio.Restriction.Restriction.Analysis.__repr__
Bio.Restriction.Restriction.FormattedSeq.__repr__
Bio.Restriction.Restriction.RestrictionBatch.__repr__
Bio.Restriction.Restriction.RestrictionType.__repr__
Bio.SearchIO._model.hit.Hit.__repr__
Bio.SearchIO._model.hsp.HSP.__repr__
Bio.SearchIO._model.hsp.HSPFragment.__repr__
Bio.SearchIO._model.query.QueryResult.__repr__
Bio.Seq.MutableSeq.__repr__
Bio.Seq.Seq.__repr__
Bio.Seq.UnknownSeq.__repr__
Bio.SeqFeature.AbstractPosition.__repr__
Bio.SeqFeature.AfterPosition.__repr__
Bio.SeqFeature.BeforePosition.__repr__
Bio.SeqFeature.BetweenPosition.__repr__
Bio.SeqFeature.CompoundLocation.__repr__
Bio.SeqFeature.ExactPosition.__repr__
Bio.SeqFeature.FeatureLocation.__repr__
Bio.SeqFeature.OneOfPosition.__repr__
Bio.SeqFeature.PositionGap.__repr__
Bio.SeqFeature.Reference.__repr__
Bio.SeqFeature.SeqFeature.__repr__
Bio.SeqFeature.UnknownPosition.__repr__
Bio.SeqFeature.WithinPosition.__repr__
Bio.SeqRecord.SeqRecord.__repr__
Bio.UniGene.ProtsimLine.__repr__
Bio.UniGene.Record.__repr__
Bio.UniGene.STSLine.__repr__
Bio.UniGene.SequenceLine.__repr__
Bio.Wise.psw.AlignmentColumn.__repr__
Bio.codonalign.codonalphabet.CodonAlphabet.__repr__
Bio.phenotype.phen_micro.PlateRecord.__repr__
Bio.phenotype.phen_micro.WellRecord.__repr__
BioSQL.BioSeqDatabase.BioSeqDatabase.__repr__
BioSQL.BioSeqDatabase.DBServer.__repr__" class="py-name" href="#" onclick="return doclink('link-10', '__repr__', 'link-6');">__repr__</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">            <tt class="py-name">attributes</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">attributes</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">text</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">"DictElement(%s, attributes=%s)"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">,</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">attributes</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line"> </tt>
<a name="NotXMLError"></a><div id="NotXMLError-def"><a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line"> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt> <a class="py-toggle" href="#" id="NotXMLError-toggle" onclick="return toggle('NotXMLError');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Entrez.Parser.NotXMLError-class.html">NotXMLError</a><tt class="py-op">(</tt><tt class="py-base-class">ValueError</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="NotXMLError-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="NotXMLError-expanded"><a name="NotXMLError.__init__"></a><div id="NotXMLError.__init__-def"><a name="L110"></a><tt class="py-lineno">110</tt> <a class="py-toggle" href="#" id="NotXMLError.__init__-toggle" onclick="return toggle('NotXMLError.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.NotXMLError-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">message</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="NotXMLError.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="NotXMLError.__init__-expanded"><a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the class."""</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">msg</tt> <tt class="py-op">=</tt> <tt class="py-name">message</tt> </tt>
</div><a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line"> </tt>
<a name="NotXMLError.__str__"></a><div id="NotXMLError.__str__-def"><a name="L114"></a><tt class="py-lineno">114</tt> <a class="py-toggle" href="#" id="NotXMLError.__str__-toggle" onclick="return toggle('NotXMLError.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.NotXMLError-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="NotXMLError.__str__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="NotXMLError.__str__-expanded"><a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">"Failed to parse the XML data (%s). Please make sure that the input data are in XML format."</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">msg</tt> </tt>
</div></div><a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line"> </tt>
<a name="CorruptedXMLError"></a><div id="CorruptedXMLError-def"><a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line"> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt> <a class="py-toggle" href="#" id="CorruptedXMLError-toggle" onclick="return toggle('CorruptedXMLError');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Entrez.Parser.CorruptedXMLError-class.html">CorruptedXMLError</a><tt class="py-op">(</tt><tt class="py-base-class">ValueError</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CorruptedXMLError-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="CorruptedXMLError-expanded"><a name="CorruptedXMLError.__init__"></a><div id="CorruptedXMLError.__init__-def"><a name="L119"></a><tt class="py-lineno">119</tt> <a class="py-toggle" href="#" id="CorruptedXMLError.__init__-toggle" onclick="return toggle('CorruptedXMLError.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.CorruptedXMLError-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">message</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CorruptedXMLError.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="CorruptedXMLError.__init__-expanded"><a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the class."""</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">msg</tt> <tt class="py-op">=</tt> <tt class="py-name">message</tt> </tt>
</div><a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line"> </tt>
<a name="CorruptedXMLError.__str__"></a><div id="CorruptedXMLError.__str__-def"><a name="L123"></a><tt class="py-lineno">123</tt> <a class="py-toggle" href="#" id="CorruptedXMLError.__str__-toggle" onclick="return toggle('CorruptedXMLError.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.CorruptedXMLError-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CorruptedXMLError.__str__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="CorruptedXMLError.__str__-expanded"><a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">"Failed to parse the XML data (%s). Please make sure that the input data are not corrupted."</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">msg</tt> </tt>
</div></div><a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line"> </tt>
<a name="ValidationError"></a><div id="ValidationError-def"><a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line"> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt> <a class="py-toggle" href="#" id="ValidationError-toggle" onclick="return toggle('ValidationError');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Entrez.Parser.ValidationError-class.html">ValidationError</a><tt class="py-op">(</tt><tt class="py-base-class">ValueError</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ValidationError-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="ValidationError-expanded"><a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line">    <tt class="py-docstring">"""XML tag found which was not defined in the DTD.</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line"><tt class="py-docstring">    Validating parsers raise this error if the parser finds a tag in the XML</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line"><tt class="py-docstring">    that is not defined in the DTD. Non-validating parsers do not raise this</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line"><tt class="py-docstring">    error. The Bio.Entrez.read and Bio.Entrez.parse functions use validating</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line"><tt class="py-docstring">    parsers by default (see those functions for more information).</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line"> </tt>
<a name="ValidationError.__init__"></a><div id="ValidationError.__init__-def"><a name="L136"></a><tt class="py-lineno">136</tt> <a class="py-toggle" href="#" id="ValidationError.__init__-toggle" onclick="return toggle('ValidationError.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.ValidationError-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ValidationError.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ValidationError.__init__-expanded"><a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the class."""</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-11" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-11', 'name', 'link-11');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-12" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-12', 'name', 'link-11');">name</a></tt> </tt>
</div><a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line"> </tt>
<a name="ValidationError.__str__"></a><div id="ValidationError.__str__-def"><a name="L140"></a><tt class="py-lineno">140</tt> <a class="py-toggle" href="#" id="ValidationError.__str__-toggle" onclick="return toggle('ValidationError.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.ValidationError-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ValidationError.__str__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ValidationError.__str__-expanded"><a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-string">"Failed to find tag '%s' in the DTD. To skip all tags that "</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line">                <tt class="py-string">"are not represented in the DTD, please call Bio.Entrez.read "</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line">                <tt class="py-string">"or Bio.Entrez.parse with validate=False."</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-13" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-13', 'name', 'link-11');">name</a></tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line"> </tt>
<a name="Consumer"></a><div id="Consumer-def"><a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line"> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt> <a class="py-toggle" href="#" id="Consumer-toggle" onclick="return toggle('Consumer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Entrez.Parser.Consumer-class.html">Consumer</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Consumer-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Consumer-expanded"><a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line"> </tt>
<a name="Consumer.__init__"></a><div id="Consumer.__init__-def"><a name="L148"></a><tt class="py-lineno">148</tt> <a class="py-toggle" href="#" id="Consumer.__init__-toggle" onclick="return toggle('Consumer.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.Consumer-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">,</tt> <tt class="py-param">attrs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Consumer.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Consumer.__init__-expanded"><a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a do-nothing Consumer object."""</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> </tt>
</div><a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line"> </tt>
<a name="Consumer.startElementHandler"></a><div id="Consumer.startElementHandler-def"><a name="L152"></a><tt class="py-lineno">152</tt> <a class="py-toggle" href="#" id="Consumer.startElementHandler-toggle" onclick="return toggle('Consumer.startElementHandler');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.Consumer-class.html#startElementHandler">startElementHandler</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">,</tt> <tt class="py-param">attrs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Consumer.startElementHandler-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Consumer.startElementHandler-expanded"><a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
</div><a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line"> </tt>
<a name="Consumer.endElementHandler"></a><div id="Consumer.endElementHandler-def"><a name="L155"></a><tt class="py-lineno">155</tt> <a class="py-toggle" href="#" id="Consumer.endElementHandler-toggle" onclick="return toggle('Consumer.endElementHandler');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.Consumer-class.html#endElementHandler">endElementHandler</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Consumer.endElementHandler-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Consumer.endElementHandler-expanded"><a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
</div><a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line"> </tt>
<a name="Consumer.consume"></a><div id="Consumer.consume-def"><a name="L158"></a><tt class="py-lineno">158</tt> <a class="py-toggle" href="#" id="Consumer.consume-toggle" onclick="return toggle('Consumer.consume');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.Consumer-class.html#consume">consume</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Consumer.consume-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Consumer.consume-expanded"><a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> </tt>
</div><a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line"> </tt>
<a name="Consumer.store"></a><div id="Consumer.store-def"><a name="L161"></a><tt class="py-lineno">161</tt> <a class="py-toggle" href="#" id="Consumer.store-toggle" onclick="return toggle('Consumer.store');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.Consumer-class.html#store">store</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">key</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Consumer.store-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Consumer.store-expanded"><a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> </tt>
</div><a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line"> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Consumer.value"></a><div id="Consumer.value-def"><a name="L165"></a><tt class="py-lineno">165</tt> <a class="py-toggle" href="#" id="Consumer.value-toggle" onclick="return toggle('Consumer.value');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.Consumer-class.html#value">value</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Consumer.value-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Consumer.value-expanded"><a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> </tt>
</div></div><a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line"> </tt>
<a name="ErrorConsumer"></a><div id="ErrorConsumer-def"><a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line"> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt> <a class="py-toggle" href="#" id="ErrorConsumer-toggle" onclick="return toggle('ErrorConsumer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Entrez.Parser.ErrorConsumer-class.html">ErrorConsumer</a><tt class="py-op">(</tt><tt class="py-base-class">Consumer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ErrorConsumer-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="ErrorConsumer-expanded"><a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line"> </tt>
<a name="ErrorConsumer.__init__"></a><div id="ErrorConsumer.__init__-def"><a name="L171"></a><tt class="py-lineno">171</tt> <a class="py-toggle" href="#" id="ErrorConsumer.__init__-toggle" onclick="return toggle('ErrorConsumer.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.ErrorConsumer-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">,</tt> <tt class="py-param">attrs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ErrorConsumer.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ErrorConsumer.__init__-expanded"><a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a Consumer for ERROR messages in the XML data."""</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-14" class="py-name" targets="Variable BioSQL.BioSeq.DBSeq.data=BioSQL.BioSeq.DBSeq-class.html#data"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-14', 'data', 'link-14');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div><a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line"> </tt>
<a name="ErrorConsumer.consume"></a><div id="ErrorConsumer.consume-def"><a name="L175"></a><tt class="py-lineno">175</tt> <a class="py-toggle" href="#" id="ErrorConsumer.consume-toggle" onclick="return toggle('ErrorConsumer.consume');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.ErrorConsumer-class.html#consume">consume</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ErrorConsumer.consume-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ErrorConsumer.consume-expanded"><a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-15" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-15', 'data', 'link-14');">data</a></tt><tt class="py-op">.</tt><tt id="link-16" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-16', 'append', 'link-16');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">)</tt> </tt>
</div><a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line"> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="ErrorConsumer.value"></a><div id="ErrorConsumer.value-def"><a name="L179"></a><tt class="py-lineno">179</tt> <a class="py-toggle" href="#" id="ErrorConsumer.value-toggle" onclick="return toggle('ErrorConsumer.value');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.ErrorConsumer-class.html#value">value</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ErrorConsumer.value-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ErrorConsumer.value-expanded"><a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">        <tt id="link-17" class="py-name" targets="Variable Bio.Entrez.Parser.Consumer.value=Bio.Entrez.Parser.Consumer-class.html#value,Variable Bio.Entrez.Parser.DictionaryConsumer.value=Bio.Entrez.Parser.DictionaryConsumer-class.html#value,Variable Bio.Entrez.Parser.ErrorConsumer.value=Bio.Entrez.Parser.ErrorConsumer-class.html#value,Variable Bio.Entrez.Parser.IntegerConsumer.value=Bio.Entrez.Parser.IntegerConsumer-class.html#value,Variable Bio.Entrez.Parser.ListConsumer.value=Bio.Entrez.Parser.ListConsumer-class.html#value,Variable Bio.Entrez.Parser.StringConsumer.value=Bio.Entrez.Parser.StringConsumer-class.html#value,Method Bio.Phylo.PhyloXMLIO.Writer.value()=Bio.Phylo.PhyloXMLIO.Writer-class.html#value"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-17', 'value', 'link-17');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-18" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-18', 'data', 'link-14');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-19" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-19', 'value', 'link-17');">value</a></tt> <tt class="py-op">==</tt> <tt class="py-string">""</tt><tt class="py-op">:</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">RuntimeError</tt><tt class="py-op">(</tt><tt id="link-20" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-20', 'value', 'link-17');">value</a></tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line"> </tt>
<a name="StringConsumer"></a><div id="StringConsumer-def"><a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line"> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt> <a class="py-toggle" href="#" id="StringConsumer-toggle" onclick="return toggle('StringConsumer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Entrez.Parser.StringConsumer-class.html">StringConsumer</a><tt class="py-op">(</tt><tt class="py-base-class">Consumer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StringConsumer-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="StringConsumer-expanded"><a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line"> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">    <tt id="link-21" class="py-name" targets="Variable Bio.Entrez.Parser.StringConsumer.consumable=Bio.Entrez.Parser.StringConsumer-class.html#consumable"><a title="Bio.Entrez.Parser.StringConsumer.consumable" class="py-name" href="#" onclick="return doclink('link-21', 'consumable', 'link-21');">consumable</a></tt> <tt class="py-op">=</tt> <tt id="link-22" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.set()=Bio.Nexus.Nexus.StepMatrix-class.html#set,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.set()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#set,Method Bio.SVDSuperimposer.SVDSuperimposer.set()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#set"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-22', 'set', 'link-22');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line"> </tt>
<a name="StringConsumer.__init__"></a><div id="StringConsumer.__init__-def"><a name="L191"></a><tt class="py-lineno">191</tt> <a class="py-toggle" href="#" id="StringConsumer.__init__-toggle" onclick="return toggle('StringConsumer.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.StringConsumer-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">,</tt> <tt class="py-param">attrs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StringConsumer.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="StringConsumer.__init__-expanded"><a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a Consumer for plain text elements in the XML data."""</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-23" class="py-name" targets="Variable Bio.SeqIO.AbiIO.tag=Bio.SeqIO.AbiIO-module.html#tag"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-23', 'tag', 'link-23');">tag</a></tt> <tt class="py-op">=</tt> <tt id="link-24" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-24', 'name', 'link-11');">name</a></tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">attributes</tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-name">attrs</tt><tt class="py-op">)</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-25" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-25', 'data', 'link-14');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div><a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line"> </tt>
<a name="StringConsumer.startElementHandler"></a><div id="StringConsumer.startElementHandler-def"><a name="L197"></a><tt class="py-lineno">197</tt> <a class="py-toggle" href="#" id="StringConsumer.startElementHandler-toggle" onclick="return toggle('StringConsumer.startElementHandler');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.StringConsumer-class.html#startElementHandler">startElementHandler</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">,</tt> <tt class="py-param">attrs</tt><tt class="py-op">,</tt> <tt class="py-param">prefix</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StringConsumer.startElementHandler-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="StringConsumer.startElementHandler-expanded"><a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-26" class="py-name" targets="Variable Bio.Phylo.NeXMLIO.prefix=Bio.Phylo.NeXMLIO-module.html#prefix,Variable Bio.Phylo.PhyloXMLIO.prefix=Bio.Phylo.PhyloXMLIO-module.html#prefix"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-26', 'prefix', 'link-26');">prefix</a></tt><tt class="py-op">:</tt> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line">            <tt class="py-name">key</tt> <tt class="py-op">=</tt> <tt class="py-string">"%s:%s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-27" class="py-name"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-27', 'prefix', 'link-26');">prefix</a></tt><tt class="py-op">,</tt> <tt id="link-28" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-28', 'name', 'link-11');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line">            <tt class="py-name">key</tt> <tt class="py-op">=</tt> <tt id="link-29" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-29', 'name', 'link-11');">name</a></tt> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-30" class="py-name"><a title="Bio.Entrez.Parser.StringConsumer.consumable" class="py-name" href="#" onclick="return doclink('link-30', 'consumable', 'link-21');">consumable</a></tt><tt class="py-op">:</tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line">            <tt id="link-31" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-31', 'tag', 'link-23');">tag</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"&lt;%s"</tt> <tt class="py-op">%</tt> <tt id="link-32" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-32', 'name', 'link-11');">name</a></tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt id="link-33" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-33', 'value', 'link-17');">value</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">attrs</tt><tt class="py-op">.</tt><tt id="link-34" class="py-name" targets="Method Bio.Crystal.Crystal.items()=Bio.Crystal.Crystal-class.html#items,Method Bio.File._IndexedSeqFileDict.items()=Bio.File._IndexedSeqFileDict-class.html#items,Method Bio.Phylo.PhyloXML.Events.items()=Bio.Phylo.PhyloXML.Events-class.html#items,Variable Bio.SearchIO._model.query.QueryResult.items=Bio.SearchIO._model.query.QueryResult-class.html#items,Method BioSQL.BioSeqDatabase.BioSeqDatabase.items()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#items,Method BioSQL.BioSeqDatabase.DBServer.items()=BioSQL.BioSeqDatabase.DBServer-class.html#items"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-34', 'items', 'link-34');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line">                <tt id="link-35" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-35', 'tag', 'link-23');">tag</a></tt> <tt class="py-op">+=</tt> <tt class="py-string">' %s="%s"'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt id="link-36" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-36', 'value', 'link-17');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line">            <tt id="link-37" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-37', 'tag', 'link-23');">tag</a></tt> <tt class="py-op">+=</tt> <tt class="py-string">"&gt;"</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-38" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-38', 'data', 'link-14');">data</a></tt><tt class="py-op">.</tt><tt id="link-39" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-39', 'append', 'link-16');">append</a></tt><tt class="py-op">(</tt><tt id="link-40" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-40', 'tag', 'link-23');">tag</a></tt><tt class="py-op">)</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">True</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
</div><a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line"> </tt>
<a name="StringConsumer.endElementHandler"></a><div id="StringConsumer.endElementHandler-def"><a name="L211"></a><tt class="py-lineno">211</tt> <a class="py-toggle" href="#" id="StringConsumer.endElementHandler-toggle" onclick="return toggle('StringConsumer.endElementHandler');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.StringConsumer-class.html#endElementHandler">endElementHandler</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">,</tt> <tt class="py-param">prefix</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StringConsumer.endElementHandler-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="StringConsumer.endElementHandler-expanded"><a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-41" class="py-name"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-41', 'prefix', 'link-26');">prefix</a></tt><tt class="py-op">:</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line">            <tt class="py-name">key</tt> <tt class="py-op">=</tt> <tt class="py-string">"%s:%s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-42" class="py-name"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-42', 'prefix', 'link-26');">prefix</a></tt><tt class="py-op">,</tt> <tt id="link-43" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-43', 'name', 'link-11');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">            <tt class="py-name">key</tt> <tt class="py-op">=</tt> <tt id="link-44" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-44', 'name', 'link-11');">name</a></tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-45" class="py-name"><a title="Bio.Entrez.Parser.StringConsumer.consumable" class="py-name" href="#" onclick="return doclink('link-45', 'consumable', 'link-21');">consumable</a></tt><tt class="py-op">:</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line">            <tt id="link-46" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-46', 'tag', 'link-23');">tag</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"&lt;/%s&gt;"</tt> <tt class="py-op">%</tt> <tt id="link-47" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-47', 'name', 'link-11');">name</a></tt> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-48" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-48', 'data', 'link-14');">data</a></tt><tt class="py-op">.</tt><tt id="link-49" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-49', 'append', 'link-16');">append</a></tt><tt class="py-op">(</tt><tt id="link-50" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-50', 'tag', 'link-23');">tag</a></tt><tt class="py-op">)</tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">True</tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
</div><a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line"> </tt>
<a name="StringConsumer.consume"></a><div id="StringConsumer.consume-def"><a name="L222"></a><tt class="py-lineno">222</tt> <a class="py-toggle" href="#" id="StringConsumer.consume-toggle" onclick="return toggle('StringConsumer.consume');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.StringConsumer-class.html#consume">consume</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StringConsumer.consume-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="StringConsumer.consume-expanded"><a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-51" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-51', 'data', 'link-14');">data</a></tt><tt class="py-op">.</tt><tt id="link-52" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-52', 'append', 'link-16');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">)</tt> </tt>
</div><a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line"> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="StringConsumer.value"></a><div id="StringConsumer.value-def"><a name="L226"></a><tt class="py-lineno">226</tt> <a class="py-toggle" href="#" id="StringConsumer.value-toggle" onclick="return toggle('StringConsumer.value');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.StringConsumer-class.html#value">value</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StringConsumer.value-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="StringConsumer.value-expanded"><a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line">        <tt id="link-53" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-53', 'value', 'link-17');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-54" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-54', 'data', 'link-14');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line">        <tt class="py-comment"># Convert Unicode strings to plain strings if possible</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line">            <tt id="link-55" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-55', 'value', 'link-17');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-56" class="py-name" targets="Class Bio.Entrez.Parser.StringElement=Bio.Entrez.Parser.StringElement-class.html"><a title="Bio.Entrez.Parser.StringElement" class="py-name" href="#" onclick="return doclink('link-56', 'StringElement', 'link-56');">StringElement</a></tt><tt class="py-op">(</tt><tt id="link-57" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-57', 'value', 'link-17');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">UnicodeEncodeError</tt><tt class="py-op">:</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">            <tt id="link-58" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-58', 'value', 'link-17');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-59" class="py-name" targets="Class Bio.Entrez.Parser.UnicodeElement=Bio.Entrez.Parser.UnicodeElement-class.html"><a title="Bio.Entrez.Parser.UnicodeElement" class="py-name" href="#" onclick="return doclink('link-59', 'UnicodeElement', 'link-59');">UnicodeElement</a></tt><tt class="py-op">(</tt><tt id="link-60" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-60', 'value', 'link-17');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">        <tt id="link-61" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-61', 'value', 'link-17');">value</a></tt><tt class="py-op">.</tt><tt id="link-62" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-62', 'tag', 'link-23');">tag</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-63" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-63', 'tag', 'link-23');">tag</a></tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">attributes</tt><tt class="py-op">:</tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line">            <tt id="link-64" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-64', 'value', 'link-17');">value</a></tt><tt class="py-op">.</tt><tt class="py-name">attributes</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">attributes</tt> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-65" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-65', 'value', 'link-17');">value</a></tt> </tt>
</div></div><a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line"> </tt>
<a name="IntegerConsumer"></a><div id="IntegerConsumer-def"><a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line"> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt> <a class="py-toggle" href="#" id="IntegerConsumer-toggle" onclick="return toggle('IntegerConsumer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Entrez.Parser.IntegerConsumer-class.html">IntegerConsumer</a><tt class="py-op">(</tt><tt class="py-base-class">Consumer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="IntegerConsumer-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="IntegerConsumer-expanded"><a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line"> </tt>
<a name="IntegerConsumer.__init__"></a><div id="IntegerConsumer.__init__-def"><a name="L241"></a><tt class="py-lineno">241</tt> <a class="py-toggle" href="#" id="IntegerConsumer.__init__-toggle" onclick="return toggle('IntegerConsumer.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.IntegerConsumer-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">,</tt> <tt class="py-param">attrs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="IntegerConsumer.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="IntegerConsumer.__init__-expanded"><a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a Consumer for integer elements in the XML data."""</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-66" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-66', 'tag', 'link-23');">tag</a></tt> <tt class="py-op">=</tt> <tt id="link-67" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-67', 'name', 'link-11');">name</a></tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">attributes</tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-name">attrs</tt><tt class="py-op">)</tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-68" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-68', 'data', 'link-14');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div><a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line"> </tt>
<a name="IntegerConsumer.consume"></a><div id="IntegerConsumer.consume-def"><a name="L247"></a><tt class="py-lineno">247</tt> <a class="py-toggle" href="#" id="IntegerConsumer.consume-toggle" onclick="return toggle('IntegerConsumer.consume');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.IntegerConsumer-class.html#consume">consume</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="IntegerConsumer.consume-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="IntegerConsumer.consume-expanded"><a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-69" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-69', 'data', 'link-14');">data</a></tt><tt class="py-op">.</tt><tt id="link-70" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-70', 'append', 'link-16');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">)</tt> </tt>
</div><a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line"> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="IntegerConsumer.value"></a><div id="IntegerConsumer.value-def"><a name="L251"></a><tt class="py-lineno">251</tt> <a class="py-toggle" href="#" id="IntegerConsumer.value-toggle" onclick="return toggle('IntegerConsumer.value');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.IntegerConsumer-class.html#value">value</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="IntegerConsumer.value-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="IntegerConsumer.value-expanded"><a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line">        <tt id="link-71" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-71', 'value', 'link-17');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-72" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-72', 'data', 'link-14');">data</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line">        <tt id="link-73" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-73', 'value', 'link-17');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-74" class="py-name" targets="Class Bio.Entrez.Parser.IntegerElement=Bio.Entrez.Parser.IntegerElement-class.html"><a title="Bio.Entrez.Parser.IntegerElement" class="py-name" href="#" onclick="return doclink('link-74', 'IntegerElement', 'link-74');">IntegerElement</a></tt><tt class="py-op">(</tt><tt id="link-75" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-75', 'value', 'link-17');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line">        <tt id="link-76" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-76', 'value', 'link-17');">value</a></tt><tt class="py-op">.</tt><tt id="link-77" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-77', 'tag', 'link-23');">tag</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-78" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-78', 'tag', 'link-23');">tag</a></tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line">        <tt id="link-79" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-79', 'value', 'link-17');">value</a></tt><tt class="py-op">.</tt><tt class="py-name">attributes</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">attributes</tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-80" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-80', 'value', 'link-17');">value</a></tt> </tt>
</div></div><a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line"> </tt>
<a name="ListConsumer"></a><div id="ListConsumer-def"><a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line"> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt> <a class="py-toggle" href="#" id="ListConsumer-toggle" onclick="return toggle('ListConsumer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Entrez.Parser.ListConsumer-class.html">ListConsumer</a><tt class="py-op">(</tt><tt class="py-base-class">Consumer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ListConsumer-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="ListConsumer-expanded"><a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line"> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line">    <tt id="link-81" class="py-name" targets="Method Bio.Crystal.Crystal.keys()=Bio.Crystal.Crystal-class.html#keys,Variable Bio.Entrez.Parser.ListConsumer.keys=Bio.Entrez.Parser.ListConsumer-class.html#keys,Method Bio.File._IndexedSeqFileDict.keys()=Bio.File._IndexedSeqFileDict-class.html#keys,Method Bio.File._SQLiteManySeqFilesDict.keys()=Bio.File._SQLiteManySeqFilesDict-class.html#keys,Method Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys()=Bio.PDB.AbstractPropertyMap.AbstractPropertyMap-class.html#keys,Method Bio.Phylo.PhyloXML.Events.keys()=Bio.Phylo.PhyloXML.Events-class.html#keys,Method BioSQL.BioSeqDatabase.BioSeqDatabase.keys()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#keys,Method BioSQL.BioSeqDatabase.DBServer.keys()=BioSQL.BioSeqDatabase.DBServer-class.html#keys"><a title="Bio.Crystal.Crystal.keys
Bio.Entrez.Parser.ListConsumer.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-81', 'keys', 'link-81');">keys</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line"> </tt>
<a name="ListConsumer.__init__"></a><div id="ListConsumer.__init__-def"><a name="L263"></a><tt class="py-lineno">263</tt> <a class="py-toggle" href="#" id="ListConsumer.__init__-toggle" onclick="return toggle('ListConsumer.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.ListConsumer-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">,</tt> <tt class="py-param">attrs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ListConsumer.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ListConsumer.__init__-expanded"><a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a Consumer for list elements in the XML data."""</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line">        <tt id="link-82" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-82', 'data', 'link-14');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-83" class="py-name" targets="Class Bio.Entrez.Parser.ListElement=Bio.Entrez.Parser.ListElement-class.html"><a title="Bio.Entrez.Parser.ListElement" class="py-name" href="#" onclick="return doclink('link-83', 'ListElement', 'link-83');">ListElement</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">        <tt id="link-84" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-84', 'data', 'link-14');">data</a></tt><tt class="py-op">.</tt><tt id="link-85" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-85', 'tag', 'link-23');">tag</a></tt> <tt class="py-op">=</tt> <tt id="link-86" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-86', 'name', 'link-11');">name</a></tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">attrs</tt><tt class="py-op">:</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line">            <tt id="link-87" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-87', 'data', 'link-14');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">attributes</tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-name">attrs</tt><tt class="py-op">)</tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-88" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-88', 'data', 'link-14');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-89" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-89', 'data', 'link-14');">data</a></tt> </tt>
</div><a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line"> </tt>
<a name="ListConsumer.store"></a><div id="ListConsumer.store-def"><a name="L271"></a><tt class="py-lineno">271</tt> <a class="py-toggle" href="#" id="ListConsumer.store-toggle" onclick="return toggle('ListConsumer.store');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.ListConsumer-class.html#store">store</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">key</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ListConsumer.store-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ListConsumer.store-expanded"><a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-90" class="py-name"><a title="Bio.Crystal.Crystal.keys
Bio.Entrez.Parser.ListConsumer.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-90', 'keys', 'link-81');">keys</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> <tt class="py-name">key</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-91" class="py-name"><a title="Bio.Crystal.Crystal.keys
Bio.Entrez.Parser.ListConsumer.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-91', 'keys', 'link-81');">keys</a></tt><tt class="py-op">:</tt> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected item '%s' in list"</tt> <tt class="py-op">%</tt> <tt class="py-name">key</tt><tt class="py-op">)</tt> </tt>
<a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-92" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-92', 'data', 'link-14');">data</a></tt><tt class="py-op">.</tt><tt id="link-93" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-93', 'append', 'link-16');">append</a></tt><tt class="py-op">(</tt><tt id="link-94" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-94', 'value', 'link-17');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line"> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="ListConsumer.value"></a><div id="ListConsumer.value-def"><a name="L277"></a><tt class="py-lineno">277</tt> <a class="py-toggle" href="#" id="ListConsumer.value-toggle" onclick="return toggle('ListConsumer.value');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.ListConsumer-class.html#value">value</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ListConsumer.value-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ListConsumer.value-expanded"><a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-95" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-95', 'data', 'link-14');">data</a></tt> </tt>
</div></div><a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line"> </tt>
<a name="DictionaryConsumer"></a><div id="DictionaryConsumer-def"><a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line"> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt> <a class="py-toggle" href="#" id="DictionaryConsumer-toggle" onclick="return toggle('DictionaryConsumer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DictionaryConsumer-class.html">DictionaryConsumer</a><tt class="py-op">(</tt><tt class="py-base-class">Consumer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DictionaryConsumer-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="DictionaryConsumer-expanded"><a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line"> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line">    <tt id="link-96" class="py-name" targets="Variable Bio.Entrez.Parser.DictionaryConsumer.multiple=Bio.Entrez.Parser.DictionaryConsumer-class.html#multiple"><a title="Bio.Entrez.Parser.DictionaryConsumer.multiple" class="py-name" href="#" onclick="return doclink('link-96', 'multiple', 'link-96');">multiple</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line"> </tt>
<a name="DictionaryConsumer.__init__"></a><div id="DictionaryConsumer.__init__-def"><a name="L285"></a><tt class="py-lineno">285</tt> <a class="py-toggle" href="#" id="DictionaryConsumer.__init__-toggle" onclick="return toggle('DictionaryConsumer.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DictionaryConsumer-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">,</tt> <tt class="py-param">attrs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DictionaryConsumer.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DictionaryConsumer.__init__-expanded"><a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a Consumer for dictionary elements in the XML data."""</tt> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">        <tt id="link-97" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-97', 'data', 'link-14');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-98" class="py-name" targets="Class Bio.Entrez.Parser.DictionaryElement=Bio.Entrez.Parser.DictionaryElement-class.html"><a title="Bio.Entrez.Parser.DictionaryElement" class="py-name" href="#" onclick="return doclink('link-98', 'DictionaryElement', 'link-98');">DictionaryElement</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">        <tt id="link-99" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-99', 'data', 'link-14');">data</a></tt><tt class="py-op">.</tt><tt id="link-100" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-100', 'tag', 'link-23');">tag</a></tt> <tt class="py-op">=</tt> <tt id="link-101" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-101', 'name', 'link-11');">name</a></tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">        <tt id="link-102" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-102', 'data', 'link-14');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">attributes</tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-name">attrs</tt><tt class="py-op">)</tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-103" class="py-name"><a title="Bio.Entrez.Parser.DictionaryConsumer.multiple" class="py-name" href="#" onclick="return doclink('link-103', 'multiple', 'link-96');">multiple</a></tt><tt class="py-op">:</tt> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line">            <tt id="link-104" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-104', 'data', 'link-14');">data</a></tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-105" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-105', 'data', 'link-14');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-106" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-106', 'data', 'link-14');">data</a></tt> </tt>
</div><a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line"> </tt>
<a name="DictionaryConsumer.store"></a><div id="DictionaryConsumer.store-def"><a name="L294"></a><tt class="py-lineno">294</tt> <a class="py-toggle" href="#" id="DictionaryConsumer.store-toggle" onclick="return toggle('DictionaryConsumer.store');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DictionaryConsumer-class.html#store">store</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">key</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DictionaryConsumer.store-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DictionaryConsumer.store-expanded"><a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-107" class="py-name"><a title="Bio.Entrez.Parser.DictionaryConsumer.multiple" class="py-name" href="#" onclick="return doclink('link-107', 'multiple', 'link-96');">multiple</a></tt><tt class="py-op">:</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-108" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-108', 'data', 'link-14');">data</a></tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-109" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-109', 'append', 'link-16');">append</a></tt><tt class="py-op">(</tt><tt id="link-110" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-110', 'value', 'link-17');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-111" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-111', 'data', 'link-14');">data</a></tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-112" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-112', 'value', 'link-17');">value</a></tt> </tt>
</div><a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line"> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="DictionaryConsumer.value"></a><div id="DictionaryConsumer.value-def"><a name="L301"></a><tt class="py-lineno">301</tt> <a class="py-toggle" href="#" id="DictionaryConsumer.value-toggle" onclick="return toggle('DictionaryConsumer.value');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DictionaryConsumer-class.html#value">value</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DictionaryConsumer.value-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DictionaryConsumer.value-expanded"><a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-113" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-113', 'data', 'link-14');">data</a></tt> </tt>
</div></div><a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line"> </tt>
<a name="select_item_consumer"></a><div id="select_item_consumer-def"><a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line"> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt> <a class="py-toggle" href="#" id="select_item_consumer-toggle" onclick="return toggle('select_item_consumer');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser-module.html#select_item_consumer">select_item_consumer</a><tt class="py-op">(</tt><tt class="py-param">name</tt><tt class="py-op">,</tt> <tt class="py-param">attrs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="select_item_consumer-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="select_item_consumer-expanded"><a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt id="link-114" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-114', 'name', 'link-11');">name</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'Item'</tt> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">    <tt id="link-115" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-115', 'name', 'link-11');">name</a></tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">attrs</tt><tt class="py-op">[</tt><tt class="py-string">"Name"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>  <tt class="py-comment"># convert from Unicode</tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line">    <tt class="py-keyword">del</tt> <tt class="py-name">attrs</tt><tt class="py-op">[</tt><tt class="py-string">"Name"</tt><tt class="py-op">]</tt> </tt>
<a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line">    <tt class="py-name">itemtype</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">attrs</tt><tt class="py-op">[</tt><tt class="py-string">"Type"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>  <tt class="py-comment"># convert from Unicode</tt> </tt>
<a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line">    <tt class="py-keyword">del</tt> <tt class="py-name">attrs</tt><tt class="py-op">[</tt><tt class="py-string">"Type"</tt><tt class="py-op">]</tt> </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">itemtype</tt> <tt class="py-op">==</tt> <tt class="py-string">"Structure"</tt><tt class="py-op">:</tt> </tt>
<a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line">        <tt class="py-name">cls</tt> <tt class="py-op">=</tt> <tt id="link-116" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.type()=Bio.Phylo.PhyloXMLIO.Writer-class.html#type"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-116', 'type', 'link-116');">type</a></tt><tt class="py-op">(</tt><tt id="link-117" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-117', 'name', 'link-11');">name</a></tt><tt class="py-op">,</tt> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line">                   <tt class="py-op">(</tt><tt id="link-118" class="py-name" targets="Class Bio.Entrez.Parser.DictionaryConsumer=Bio.Entrez.Parser.DictionaryConsumer-class.html"><a title="Bio.Entrez.Parser.DictionaryConsumer" class="py-name" href="#" onclick="return doclink('link-118', 'DictionaryConsumer', 'link-118');">DictionaryConsumer</a></tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line">                   <tt class="py-op">{</tt><tt class="py-string">"multiple"</tt><tt class="py-op">:</tt> <tt id="link-119" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-119', 'set', 'link-22');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt> <tt class="py-op">=</tt> <tt class="py-name">cls</tt><tt class="py-op">(</tt><tt id="link-120" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-120', 'name', 'link-11');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">attrs</tt><tt class="py-op">)</tt> </tt>
<a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt id="link-121" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-121', 'name', 'link-11');">name</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">"ArticleIds"</tt><tt class="py-op">,</tt> <tt class="py-string">"History"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line">        <tt class="py-name">cls</tt> <tt class="py-op">=</tt> <tt id="link-122" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-122', 'type', 'link-116');">type</a></tt><tt class="py-op">(</tt><tt id="link-123" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-123', 'name', 'link-11');">name</a></tt><tt class="py-op">,</tt> </tt>
<a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">                   <tt class="py-op">(</tt><tt id="link-124" class="py-name"><a title="Bio.Entrez.Parser.DictionaryConsumer" class="py-name" href="#" onclick="return doclink('link-124', 'DictionaryConsumer', 'link-118');">DictionaryConsumer</a></tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line">                   <tt class="py-op">{</tt><tt class="py-string">"multiple"</tt><tt class="py-op">:</tt> <tt id="link-125" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-125', 'set', 'link-22');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-string">"pubmed"</tt><tt class="py-op">,</tt> <tt class="py-string">"medline"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt> <tt class="py-op">=</tt> <tt class="py-name">cls</tt><tt class="py-op">(</tt><tt id="link-126" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-126', 'name', 'link-11');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">attrs</tt><tt class="py-op">)</tt> </tt>
<a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">itemtype</tt> <tt class="py-op">==</tt> <tt class="py-string">"List"</tt><tt class="py-op">:</tt> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">        <tt class="py-comment"># Keys are unknown in this case</tt> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt> <tt class="py-op">=</tt> <tt id="link-127" class="py-name" targets="Class Bio.Entrez.Parser.ListConsumer=Bio.Entrez.Parser.ListConsumer-class.html"><a title="Bio.Entrez.Parser.ListConsumer" class="py-name" href="#" onclick="return doclink('link-127', 'ListConsumer', 'link-127');">ListConsumer</a></tt><tt class="py-op">(</tt><tt id="link-128" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-128', 'name', 'link-11');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">attrs</tt><tt class="py-op">)</tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">itemtype</tt> <tt class="py-op">==</tt> <tt class="py-string">"Integer"</tt><tt class="py-op">:</tt> </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt> <tt class="py-op">=</tt> <tt id="link-129" class="py-name" targets="Class Bio.Entrez.Parser.IntegerConsumer=Bio.Entrez.Parser.IntegerConsumer-class.html"><a title="Bio.Entrez.Parser.IntegerConsumer" class="py-name" href="#" onclick="return doclink('link-129', 'IntegerConsumer', 'link-129');">IntegerConsumer</a></tt><tt class="py-op">(</tt><tt id="link-130" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-130', 'name', 'link-11');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">attrs</tt><tt class="py-op">)</tt> </tt>
<a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">itemtype</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">"String"</tt><tt class="py-op">,</tt> <tt class="py-string">"Unknown"</tt><tt class="py-op">,</tt> <tt class="py-string">"Date"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt> <tt class="py-op">=</tt> <tt id="link-131" class="py-name" targets="Class Bio.Entrez.Parser.StringConsumer=Bio.Entrez.Parser.StringConsumer-class.html"><a title="Bio.Entrez.Parser.StringConsumer" class="py-name" href="#" onclick="return doclink('link-131', 'StringConsumer', 'link-131');">StringConsumer</a></tt><tt class="py-op">(</tt><tt id="link-132" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-132', 'name', 'link-11');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">attrs</tt><tt class="py-op">)</tt> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unknown item type %s"</tt> <tt class="py-op">%</tt> <tt id="link-133" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-133', 'name', 'link-11');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">consumer</tt> </tt>
</div><a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line"> </tt>
<a name="DataHandler"></a><div id="DataHandler-def"><a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line"> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt> <a class="py-toggle" href="#" id="DataHandler-toggle" onclick="return toggle('DataHandler');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DataHandler-class.html">DataHandler</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DataHandler-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="DataHandler-expanded"><a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line"> </tt>
<a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-134" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-134', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-135" class="py-name" targets="Package Bio.Entrez=Bio.Entrez-module.html"><a title="Bio.Entrez" class="py-name" href="#" onclick="return doclink('link-135', 'Entrez', 'link-135');">Entrez</a></tt> </tt>
<a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line">    <tt id="link-136" class="py-name" targets="Variable Bio.Entrez.Parser.DataHandler.global_dtd_dir=Bio.Entrez.Parser.DataHandler-class.html#global_dtd_dir"><a title="Bio.Entrez.Parser.DataHandler.global_dtd_dir" class="py-name" href="#" onclick="return doclink('link-136', 'global_dtd_dir', 'link-136');">global_dtd_dir</a></tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-137" class="py-name"><a title="Bio.Entrez" class="py-name" href="#" onclick="return doclink('link-137', 'Entrez', 'link-135');">Entrez</a></tt><tt class="py-op">.</tt><tt class="py-name">__path__</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"DTDs"</tt><tt class="py-op">)</tt> </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line">    <tt id="link-138" class="py-name" targets="Variable Bio.Entrez.Parser.DataHandler.global_xsd_dir=Bio.Entrez.Parser.DataHandler-class.html#global_xsd_dir"><a title="Bio.Entrez.Parser.DataHandler.global_xsd_dir" class="py-name" href="#" onclick="return doclink('link-138', 'global_xsd_dir', 'link-138');">global_xsd_dir</a></tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-139" class="py-name"><a title="Bio.Entrez" class="py-name" href="#" onclick="return doclink('link-139', 'Entrez', 'link-135');">Entrez</a></tt><tt class="py-op">.</tt><tt class="py-name">__path__</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"XSDs"</tt><tt class="py-op">)</tt> </tt>
<a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line">    <tt id="link-140" class="py-name" targets="Variable Bio.Entrez.Parser.DataHandler.local_dtd_dir=Bio.Entrez.Parser.DataHandler-class.html#local_dtd_dir"><a title="Bio.Entrez.Parser.DataHandler.local_dtd_dir" class="py-name" href="#" onclick="return doclink('link-140', 'local_dtd_dir', 'link-140');">local_dtd_dir</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line">    <tt id="link-141" class="py-name" targets="Variable Bio.Entrez.Parser.DataHandler.local_xsd_dir=Bio.Entrez.Parser.DataHandler-class.html#local_xsd_dir"><a title="Bio.Entrez.Parser.DataHandler.local_xsd_dir" class="py-name" href="#" onclick="return doclink('link-141', 'local_xsd_dir', 'link-141');">local_xsd_dir</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line"> </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line">    <tt class="py-keyword">del</tt> <tt id="link-142" class="py-name"><a title="Bio.Entrez" class="py-name" href="#" onclick="return doclink('link-142', 'Entrez', 'link-135');">Entrez</a></tt> </tt>
<a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line"> </tt>
<a name="DataHandler.__init__"></a><div id="DataHandler.__init__-def"><a name="L343"></a><tt class="py-lineno">343</tt> <a class="py-toggle" href="#" id="DataHandler.__init__-toggle" onclick="return toggle('DataHandler.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DataHandler-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">validate</tt><tt class="py-op">,</tt> <tt class="py-param">escape</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DataHandler.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DataHandler.__init__-expanded"><a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a DataHandler object."""</tt> </tt>
<a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dtd_urls</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">classes</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L347"></a><tt class="py-lineno">347</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">consumer</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">validating</tt> <tt class="py-op">=</tt> <tt id="link-143" class="py-name" targets="Method Bio.Crystal.Chain.validate()=Bio.Crystal.Chain-class.html#validate"><a title="Bio.Crystal.Chain.validate" class="py-name" href="#" onclick="return doclink('link-143', 'validate', 'link-143');">validate</a></tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">escaping</tt> <tt class="py-op">=</tt> <tt class="py-name">escape</tt> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">parser</tt> <tt class="py-op">=</tt> <tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">ParserCreate</tt><tt class="py-op">(</tt><tt class="py-name">namespace_separator</tt><tt class="py-op">=</tt><tt class="py-string">" "</tt><tt class="py-op">)</tt> </tt>
<a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">parser</tt><tt class="py-op">.</tt><tt class="py-name">SetParamEntityParsing</tt><tt class="py-op">(</tt><tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">XML_PARAM_ENTITY_PARSING_ALWAYS</tt><tt class="py-op">)</tt> </tt>
<a name="L352"></a><tt class="py-lineno">352</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">parser</tt><tt class="py-op">.</tt><tt class="py-name">XmlDeclHandler</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-144" class="py-name" targets="Method Bio.Entrez.Parser.DataHandler.xmlDeclHandler()=Bio.Entrez.Parser.DataHandler-class.html#xmlDeclHandler"><a title="Bio.Entrez.Parser.DataHandler.xmlDeclHandler" class="py-name" href="#" onclick="return doclink('link-144', 'xmlDeclHandler', 'link-144');">xmlDeclHandler</a></tt> </tt>
<a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">schema_namespace</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">namespace_level</tt> <tt class="py-op">=</tt> <tt class="py-name">Counter</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">namespace_prefix</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_directory</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line"> </tt>
<a name="DataHandler.read"></a><div id="DataHandler.read-def"><a name="L358"></a><tt class="py-lineno">358</tt> <a class="py-toggle" href="#" id="DataHandler.read-toggle" onclick="return toggle('DataHandler.read');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DataHandler-class.html#read">read</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DataHandler.read-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DataHandler.read-expanded"><a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line">        <tt class="py-docstring">"""Set up the parser and let it parse the XML results."""</tt> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line">        <tt class="py-comment"># HACK: remove Bio._py3k handle conversion, since the Entrez XML parser</tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line">        <tt class="py-comment"># expects binary data</tt> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt> <tt class="py-op">==</tt> <tt class="py-string">'EvilHandleHack'</tt><tt class="py-op">:</tt> </tt>
<a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line">            <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt class="py-name">_handle</tt> </tt>
<a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt> <tt class="py-op">==</tt> <tt class="py-string">'TextIOWrapper'</tt><tt class="py-op">:</tt> </tt>
<a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line">            <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt class="py-name">buffer</tt> </tt>
<a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-string">"closed"</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt class="py-name">closed</tt><tt class="py-op">:</tt> </tt>
<a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line">            <tt class="py-comment"># Should avoid a possible Segmentation Fault, see:</tt> </tt>
<a name="L368"></a><tt class="py-lineno">368</tt>  <tt class="py-line">            <tt class="py-comment"># http://bugs.python.org/issue4877</tt> </tt>
<a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">IOError</tt><tt class="py-op">(</tt><tt class="py-string">"Can't parse a closed handle"</tt><tt class="py-op">)</tt> </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">version_info</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">3</tt><tt class="py-op">:</tt> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line">            <tt class="py-comment"># Another nasty hack to cope with a unicode StringIO handle</tt> </tt>
<a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line">            <tt class="py-comment"># since the Entrez XML parser expects binary data (bytes)</tt> </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line">            <tt class="py-keyword">from</tt> <tt class="py-name">io</tt> <tt class="py-keyword">import</tt> <tt class="py-name">StringIO</tt> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-name">StringIO</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line">                <tt class="py-keyword">from</tt> <tt id="link-145" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-145', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-146" class="py-name"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-146', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt id="link-147" class="py-name" targets="Function Bio._py3k._as_bytes()=Bio._py3k-module.html#_as_bytes"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-147', '_as_bytes', 'link-147');">_as_bytes</a></tt> </tt>
<a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line">                <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">BytesIO</tt><tt class="py-op">(</tt><tt id="link-148" class="py-name"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-148', '_as_bytes', 'link-147');">_as_bytes</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-149" class="py-name" targets="Function Bio.Affy.CelFile.read()=Bio.Affy.CelFile-module.html#read,Function Bio.AlignIO.read()=Bio.AlignIO-module.html#read,Function Bio.Blast.NCBIXML.read()=Bio.Blast.NCBIXML-module.html#read,Function Bio.Cluster.read()=Bio.Cluster-module.html#read,Function Bio.Compass.read()=Bio.Compass-module.html#read,Function Bio.Emboss.Primer3.read()=Bio.Emboss.Primer3-module.html#read,Function Bio.Emboss.PrimerSearch.read()=Bio.Emboss.PrimerSearch-module.html#read,Method Bio.Entrez.Parser.DataHandler.read()=Bio.Entrez.Parser.DataHandler-class.html#read,Function Bio.Entrez.read()=Bio.Entrez-module.html#read,Function Bio.ExPASy.Enzyme.read()=Bio.ExPASy.Enzyme-module.html#read,Function Bio.ExPASy.Prodoc.read()=Bio.ExPASy.Prodoc-module.html#read,Function Bio.ExPASy.Prosite.read()=Bio.ExPASy.Prosite-module.html#read,Function Bio.ExPASy.ScanProsite.read()=Bio.ExPASy.ScanProsite-module.html#read,Function Bio.ExPASy.cellosaurus.read()=Bio.ExPASy.cellosaurus-module.html#read,Method Bio.File.UndoHandle.read()=Bio.File.UndoHandle-class.html#read,Function Bio.KEGG.Enzyme.read()=Bio.KEGG.Enzyme-module.html#read,Function Bio.KEGG.KGML.KGML_parser.read()=Bio.KEGG.KGML.KGML_parser-module.html#read,Function Bio.Medline.read()=Bio.Medline-module.html#read,Method Bio.Nexus.Nexus.Nexus.read()=Bio.Nexus.Nexus.Nexus-class.html#read,Function Bio.Phylo.PAML.baseml.read()=Bio.Phylo.PAML.baseml-module.html#read,Function Bio.Phylo.PAML.codeml.read()=Bio.Phylo.PAML.codeml-module.html#read,Function Bio.Phylo.PAML.yn00.read()=Bio.Phylo.PAML.yn00-module.html#read,Method Bio.Phylo.PhyloXMLIO.Parser.read()=Bio.Phylo.PhyloXMLIO.Parser-class.html#read,Function Bio.Phylo.PhyloXMLIO.read()=Bio.Phylo.PhyloXMLIO-module.html#read,Function Bio.Phylo._io.read()=Bio.Phylo._io-module.html#read,Function Bio.PopGen.GenePop.FileParser.read()=Bio.PopGen.GenePop.FileParser-module.html#read,Function Bio.PopGen.GenePop.LargeFileParser.read()=Bio.PopGen.GenePop.LargeFileParser-module.html#read,Function Bio.PopGen.GenePop.read()=Bio.PopGen.GenePop-module.html#read,Function Bio.SearchIO.read()=Bio.SearchIO-module.html#read,Method Bio.SeqIO.SffIO._AddTellHandle.read()=Bio.SeqIO.SffIO._AddTellHandle-class.html#read,Function Bio.SeqIO.read()=Bio.SeqIO-module.html#read,Function Bio.Sequencing.Ace.read()=Bio.Sequencing.Ace-module.html#read,Function Bio.Sequencing.Phd.read()=Bio.Sequencing.Phd-module.html#read,Function Bio.SwissProt.read()=Bio.SwissProt-module.html#read,Function Bio.UniGene.read()=Bio.UniGene-module.html#read,Method Bio.bgzf.BgzfReader.read()=Bio.bgzf.BgzfReader-class.html#read,Function Bio.motifs.alignace.read()=Bio.motifs.alignace-module.html#read,Function Bio.motifs.jaspar.read()=Bio.motifs.jaspar-module.html#read,Function Bio.motifs.mast.read()=Bio.motifs.mast-module.html#read,Function Bio.motifs.meme.read()=Bio.motifs.meme-module.html#read,Function Bio.motifs.minimal.read()=Bio.motifs.minimal-module.html#read,Function Bio.motifs.read()=Bio.motifs-module.html#read,Function Bio.motifs.transfac.read()=Bio.motifs.transfac-module.html#read,Function Bio.phenotype.read()=Bio.phenotype-module.html#read"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.ExPASy.cellosaurus.read
Bio.File.UndoHandle.read
Bio.KEGG.Enzyme.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.minimal.read
Bio.motifs.read
Bio.motifs.transfac.read
Bio.phenotype.read" class="py-name" href="#" onclick="return doclink('link-149', 'read', 'link-149');">read</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L378"></a><tt class="py-lineno">378</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">parser</tt><tt class="py-op">.</tt><tt class="py-name">ParseFile</tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L379"></a><tt class="py-lineno">379</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">ExpatError</tt> <tt class="py-keyword">as</tt> <tt class="py-name">e</tt><tt class="py-op">:</tt> </tt>
<a name="L380"></a><tt class="py-lineno">380</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">parser</tt><tt class="py-op">.</tt><tt class="py-name">StartElementHandler</tt><tt class="py-op">:</tt> </tt>
<a name="L381"></a><tt class="py-lineno">381</tt>  <tt class="py-line">                <tt class="py-comment"># We saw the initial &lt;!xml declaration, so we can be sure that</tt> </tt>
<a name="L382"></a><tt class="py-lineno">382</tt>  <tt class="py-line">                <tt class="py-comment"># we are parsing XML data. Most likely, the XML file is</tt> </tt>
<a name="L383"></a><tt class="py-lineno">383</tt>  <tt class="py-line">                <tt class="py-comment"># corrupted.</tt> </tt>
<a name="L384"></a><tt class="py-lineno">384</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt id="link-150" class="py-name" targets="Class Bio.Entrez.Parser.CorruptedXMLError=Bio.Entrez.Parser.CorruptedXMLError-class.html"><a title="Bio.Entrez.Parser.CorruptedXMLError" class="py-name" href="#" onclick="return doclink('link-150', 'CorruptedXMLError', 'link-150');">CorruptedXMLError</a></tt><tt class="py-op">(</tt><tt class="py-name">e</tt><tt class="py-op">)</tt> </tt>
<a name="L385"></a><tt class="py-lineno">385</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L386"></a><tt class="py-lineno">386</tt>  <tt class="py-line">                <tt class="py-comment"># We have not seen the initial &lt;!xml declaration, so probably</tt> </tt>
<a name="L387"></a><tt class="py-lineno">387</tt>  <tt class="py-line">                <tt class="py-comment"># the input data is not in XML format.</tt> </tt>
<a name="L388"></a><tt class="py-lineno">388</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt id="link-151" class="py-name" targets="Class Bio.Entrez.Parser.NotXMLError=Bio.Entrez.Parser.NotXMLError-class.html"><a title="Bio.Entrez.Parser.NotXMLError" class="py-name" href="#" onclick="return doclink('link-151', 'NotXMLError', 'link-151');">NotXMLError</a></tt><tt class="py-op">(</tt><tt class="py-name">e</tt><tt class="py-op">)</tt> </tt>
<a name="L389"></a><tt class="py-lineno">389</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L390"></a><tt class="py-lineno">390</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">record</tt> </tt>
<a name="L391"></a><tt class="py-lineno">391</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L392"></a><tt class="py-lineno">392</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">parser</tt><tt class="py-op">.</tt><tt class="py-name">StartElementHandler</tt><tt class="py-op">:</tt> </tt>
<a name="L393"></a><tt class="py-lineno">393</tt>  <tt class="py-line">                <tt class="py-comment"># We saw the initial &lt;!xml declaration, and expat didn't notice</tt> </tt>
<a name="L394"></a><tt class="py-lineno">394</tt>  <tt class="py-line">                <tt class="py-comment"># any errors, so self.record should be defined. If not, this is</tt> </tt>
<a name="L395"></a><tt class="py-lineno">395</tt>  <tt class="py-line">                <tt class="py-comment"># a bug.</tt> </tt>
<a name="L396"></a><tt class="py-lineno">396</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">RuntimeError</tt><tt class="py-op">(</tt><tt class="py-string">"Failed to parse the XML file correctly, possibly due to a bug in Bio.Entrez. Please contact the Biopython developers via the mailing list or GitHub for assistance."</tt><tt class="py-op">)</tt> </tt>
<a name="L397"></a><tt class="py-lineno">397</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L398"></a><tt class="py-lineno">398</tt>  <tt class="py-line">                <tt class="py-comment"># We did not see the initial &lt;!xml declaration, so probably</tt> </tt>
<a name="L399"></a><tt class="py-lineno">399</tt>  <tt class="py-line">                <tt class="py-comment"># the input data is not in XML format.</tt> </tt>
<a name="L400"></a><tt class="py-lineno">400</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt id="link-152" class="py-name"><a title="Bio.Entrez.Parser.NotXMLError" class="py-name" href="#" onclick="return doclink('link-152', 'NotXMLError', 'link-151');">NotXMLError</a></tt><tt class="py-op">(</tt><tt class="py-string">"XML declaration not found"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L401"></a><tt class="py-lineno">401</tt>  <tt class="py-line"> </tt>
<a name="DataHandler.parse"></a><div id="DataHandler.parse-def"><a name="L402"></a><tt class="py-lineno">402</tt> <a class="py-toggle" href="#" id="DataHandler.parse-toggle" onclick="return toggle('DataHandler.parse');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DataHandler-class.html#parse">parse</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DataHandler.parse-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DataHandler.parse-expanded"><a name="L403"></a><tt class="py-lineno">403</tt>  <tt class="py-line">        <tt class="py-name">BLOCK</tt> <tt class="py-op">=</tt> <tt class="py-number">1024</tt> </tt>
<a name="L404"></a><tt class="py-lineno">404</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L405"></a><tt class="py-lineno">405</tt>  <tt class="py-line">            <tt class="py-comment"># Read in another block of the file...</tt> </tt>
<a name="L406"></a><tt class="py-lineno">406</tt>  <tt class="py-line">            <tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-153" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.ExPASy.cellosaurus.read
Bio.File.UndoHandle.read
Bio.KEGG.Enzyme.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.minimal.read
Bio.motifs.read
Bio.motifs.transfac.read
Bio.phenotype.read" class="py-name" href="#" onclick="return doclink('link-153', 'read', 'link-149');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">BLOCK</tt><tt class="py-op">)</tt> </tt>
<a name="L407"></a><tt class="py-lineno">407</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L408"></a><tt class="py-lineno">408</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">parser</tt><tt class="py-op">.</tt><tt class="py-name">Parse</tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">,</tt> <tt class="py-name">False</tt><tt class="py-op">)</tt> </tt>
<a name="L409"></a><tt class="py-lineno">409</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">ExpatError</tt> <tt class="py-keyword">as</tt> <tt class="py-name">e</tt><tt class="py-op">:</tt> </tt>
<a name="L410"></a><tt class="py-lineno">410</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">parser</tt><tt class="py-op">.</tt><tt class="py-name">StartElementHandler</tt><tt class="py-op">:</tt> </tt>
<a name="L411"></a><tt class="py-lineno">411</tt>  <tt class="py-line">                    <tt class="py-comment"># We saw the initial &lt;!xml declaration, so we can be sure</tt> </tt>
<a name="L412"></a><tt class="py-lineno">412</tt>  <tt class="py-line">                    <tt class="py-comment"># that we are parsing XML data. Most likely, the XML file</tt> </tt>
<a name="L413"></a><tt class="py-lineno">413</tt>  <tt class="py-line">                    <tt class="py-comment"># is corrupted.</tt> </tt>
<a name="L414"></a><tt class="py-lineno">414</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt id="link-154" class="py-name"><a title="Bio.Entrez.Parser.CorruptedXMLError" class="py-name" href="#" onclick="return doclink('link-154', 'CorruptedXMLError', 'link-150');">CorruptedXMLError</a></tt><tt class="py-op">(</tt><tt class="py-name">e</tt><tt class="py-op">)</tt> </tt>
<a name="L415"></a><tt class="py-lineno">415</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L416"></a><tt class="py-lineno">416</tt>  <tt class="py-line">                    <tt class="py-comment"># We have not seen the initial &lt;!xml declaration, so</tt> </tt>
<a name="L417"></a><tt class="py-lineno">417</tt>  <tt class="py-line">                    <tt class="py-comment"># probably the input data is not in XML format.</tt> </tt>
<a name="L418"></a><tt class="py-lineno">418</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt id="link-155" class="py-name"><a title="Bio.Entrez.Parser.NotXMLError" class="py-name" href="#" onclick="return doclink('link-155', 'NotXMLError', 'link-151');">NotXMLError</a></tt><tt class="py-op">(</tt><tt class="py-name">e</tt><tt class="py-op">)</tt> </tt>
<a name="L419"></a><tt class="py-lineno">419</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L420"></a><tt class="py-lineno">420</tt>  <tt class="py-line">                <tt class="py-name">records</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">record</tt> </tt>
<a name="L421"></a><tt class="py-lineno">421</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L422"></a><tt class="py-lineno">422</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">parser</tt><tt class="py-op">.</tt><tt class="py-name">StartElementHandler</tt><tt class="py-op">:</tt> </tt>
<a name="L423"></a><tt class="py-lineno">423</tt>  <tt class="py-line">                    <tt class="py-comment"># We saw the initial &lt;!xml declaration, and expat</tt> </tt>
<a name="L424"></a><tt class="py-lineno">424</tt>  <tt class="py-line">                    <tt class="py-comment"># didn't notice any errors, so self.record should be</tt> </tt>
<a name="L425"></a><tt class="py-lineno">425</tt>  <tt class="py-line">                    <tt class="py-comment"># defined. If not, this is a bug.</tt> </tt>
<a name="L426"></a><tt class="py-lineno">426</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">RuntimeError</tt><tt class="py-op">(</tt><tt class="py-string">"Failed to parse the XML file correctly, possibly due to a bug in Bio.Entrez. Please contact the Biopython developers via the mailing list or GitHub for assistance."</tt><tt class="py-op">)</tt> </tt>
<a name="L427"></a><tt class="py-lineno">427</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L428"></a><tt class="py-lineno">428</tt>  <tt class="py-line">                    <tt class="py-comment"># We did not see the initial &lt;!xml declaration, so</tt> </tt>
<a name="L429"></a><tt class="py-lineno">429</tt>  <tt class="py-line">                    <tt class="py-comment"># probably the input data is not in XML format.</tt> </tt>
<a name="L430"></a><tt class="py-lineno">430</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt id="link-156" class="py-name"><a title="Bio.Entrez.Parser.NotXMLError" class="py-name" href="#" onclick="return doclink('link-156', 'NotXMLError', 'link-151');">NotXMLError</a></tt><tt class="py-op">(</tt><tt class="py-string">"XML declaration not found"</tt><tt class="py-op">)</tt> </tt>
<a name="L431"></a><tt class="py-lineno">431</tt>  <tt class="py-line"> </tt>
<a name="L432"></a><tt class="py-lineno">432</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">records</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L433"></a><tt class="py-lineno">433</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"The XML file does not represent a list. Please use Entrez.read instead of Entrez.parse"</tt><tt class="py-op">)</tt> </tt>
<a name="L434"></a><tt class="py-lineno">434</tt>  <tt class="py-line"> </tt>
<a name="L435"></a><tt class="py-lineno">435</tt>  <tt class="py-line">            <tt class="py-keyword">while</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">records</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt>  <tt class="py-comment"># Then the top record is finished</tt> </tt>
<a name="L436"></a><tt class="py-lineno">436</tt>  <tt class="py-line">                <tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt class="py-name">records</tt><tt class="py-op">.</tt><tt id="link-157" class="py-name" targets="Method Bio.File._IndexedSeqFileDict.pop()=Bio.File._IndexedSeqFileDict-class.html#pop,Method Bio.SearchIO._model.hit.Hit.pop()=Bio.SearchIO._model.hit.Hit-class.html#pop,Method Bio.SearchIO._model.query.QueryResult.pop()=Bio.SearchIO._model.query.QueryResult-class.html#pop,Method Bio.Seq.MutableSeq.pop()=Bio.Seq.MutableSeq-class.html#pop"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop" class="py-name" href="#" onclick="return doclink('link-157', 'pop', 'link-157');">pop</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L437"></a><tt class="py-lineno">437</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt class="py-name">record</tt> </tt>
<a name="L438"></a><tt class="py-lineno">438</tt>  <tt class="py-line"> </tt>
<a name="L439"></a><tt class="py-lineno">439</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">text</tt><tt class="py-op">:</tt> </tt>
<a name="L440"></a><tt class="py-lineno">440</tt>  <tt class="py-line">                <tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">stdout</tt><tt class="py-op">.</tt><tt id="link-158" class="py-name" targets="Method Bio.bgzf.BgzfWriter.flush()=Bio.bgzf.BgzfWriter-class.html#flush"><a title="Bio.bgzf.BgzfWriter.flush" class="py-name" href="#" onclick="return doclink('link-158', 'flush', 'link-158');">flush</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L441"></a><tt class="py-lineno">441</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">parser</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L442"></a><tt class="py-lineno">442</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">consumer</tt><tt class="py-op">:</tt> </tt>
<a name="L443"></a><tt class="py-lineno">443</tt>  <tt class="py-line">                    <tt class="py-comment"># We have reached the end of the XML file</tt> </tt>
<a name="L444"></a><tt class="py-lineno">444</tt>  <tt class="py-line">                    <tt class="py-comment"># No more XML data, but there is still some unfinished</tt> </tt>
<a name="L445"></a><tt class="py-lineno">445</tt>  <tt class="py-line">                    <tt class="py-comment"># business</tt> </tt>
<a name="L446"></a><tt class="py-lineno">446</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt id="link-159" class="py-name"><a title="Bio.Entrez.Parser.CorruptedXMLError" class="py-name" href="#" onclick="return doclink('link-159', 'CorruptedXMLError', 'link-150');">CorruptedXMLError</a></tt><tt class="py-op">(</tt><tt class="py-string">"Premature end of XML stream"</tt><tt class="py-op">)</tt> </tt>
<a name="L447"></a><tt class="py-lineno">447</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> </tt>
</div><a name="L448"></a><tt class="py-lineno">448</tt>  <tt class="py-line"> </tt>
<a name="DataHandler.xmlDeclHandler"></a><div id="DataHandler.xmlDeclHandler-def"><a name="L449"></a><tt class="py-lineno">449</tt> <a class="py-toggle" href="#" id="DataHandler.xmlDeclHandler-toggle" onclick="return toggle('DataHandler.xmlDeclHandler');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DataHandler-class.html#xmlDeclHandler">xmlDeclHandler</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">version</tt><tt class="py-op">,</tt> <tt class="py-param">encoding</tt><tt class="py-op">,</tt> <tt class="py-param">standalone</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DataHandler.xmlDeclHandler-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DataHandler.xmlDeclHandler-expanded"><a name="L450"></a><tt class="py-lineno">450</tt>  <tt class="py-line">        <tt class="py-comment"># XML declaration found; set the handlers</tt> </tt>
<a name="L451"></a><tt class="py-lineno">451</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">parser</tt><tt class="py-op">.</tt><tt class="py-name">StartElementHandler</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-160" class="py-name" targets="Method Bio.Entrez.Parser.Consumer.startElementHandler()=Bio.Entrez.Parser.Consumer-class.html#startElementHandler,Method Bio.Entrez.Parser.DataHandler.startElementHandler()=Bio.Entrez.Parser.DataHandler-class.html#startElementHandler,Method Bio.Entrez.Parser.StringConsumer.startElementHandler()=Bio.Entrez.Parser.StringConsumer-class.html#startElementHandler"><a title="Bio.Entrez.Parser.Consumer.startElementHandler
Bio.Entrez.Parser.DataHandler.startElementHandler
Bio.Entrez.Parser.StringConsumer.startElementHandler" class="py-name" href="#" onclick="return doclink('link-160', 'startElementHandler', 'link-160');">startElementHandler</a></tt> </tt>
<a name="L452"></a><tt class="py-lineno">452</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">parser</tt><tt class="py-op">.</tt><tt class="py-name">EndElementHandler</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-161" class="py-name" targets="Method Bio.Entrez.Parser.Consumer.endElementHandler()=Bio.Entrez.Parser.Consumer-class.html#endElementHandler,Method Bio.Entrez.Parser.DataHandler.endElementHandler()=Bio.Entrez.Parser.DataHandler-class.html#endElementHandler,Method Bio.Entrez.Parser.StringConsumer.endElementHandler()=Bio.Entrez.Parser.StringConsumer-class.html#endElementHandler"><a title="Bio.Entrez.Parser.Consumer.endElementHandler
Bio.Entrez.Parser.DataHandler.endElementHandler
Bio.Entrez.Parser.StringConsumer.endElementHandler" class="py-name" href="#" onclick="return doclink('link-161', 'endElementHandler', 'link-161');">endElementHandler</a></tt> </tt>
<a name="L453"></a><tt class="py-lineno">453</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">escaping</tt><tt class="py-op">:</tt> </tt>
<a name="L454"></a><tt class="py-lineno">454</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">parser</tt><tt class="py-op">.</tt><tt class="py-name">CharacterDataHandler</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-162" class="py-name" targets="Method Bio.Entrez.Parser.DataHandler.characterDataHandlerEscape()=Bio.Entrez.Parser.DataHandler-class.html#characterDataHandlerEscape"><a title="Bio.Entrez.Parser.DataHandler.characterDataHandlerEscape" class="py-name" href="#" onclick="return doclink('link-162', 'characterDataHandlerEscape', 'link-162');">characterDataHandlerEscape</a></tt> </tt>
<a name="L455"></a><tt class="py-lineno">455</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L456"></a><tt class="py-lineno">456</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">parser</tt><tt class="py-op">.</tt><tt class="py-name">CharacterDataHandler</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-163" class="py-name" targets="Method Bio.Entrez.Parser.DataHandler.characterDataHandlerRaw()=Bio.Entrez.Parser.DataHandler-class.html#characterDataHandlerRaw"><a title="Bio.Entrez.Parser.DataHandler.characterDataHandlerRaw" class="py-name" href="#" onclick="return doclink('link-163', 'characterDataHandlerRaw', 'link-163');">characterDataHandlerRaw</a></tt> </tt>
<a name="L457"></a><tt class="py-lineno">457</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">parser</tt><tt class="py-op">.</tt><tt class="py-name">ExternalEntityRefHandler</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-164" class="py-name" targets="Method Bio.Entrez.Parser.DataHandler.externalEntityRefHandler()=Bio.Entrez.Parser.DataHandler-class.html#externalEntityRefHandler"><a title="Bio.Entrez.Parser.DataHandler.externalEntityRefHandler" class="py-name" href="#" onclick="return doclink('link-164', 'externalEntityRefHandler', 'link-164');">externalEntityRefHandler</a></tt> </tt>
<a name="L458"></a><tt class="py-lineno">458</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">parser</tt><tt class="py-op">.</tt><tt class="py-name">StartNamespaceDeclHandler</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-165" class="py-name" targets="Method Bio.Entrez.Parser.DataHandler.startNamespaceDeclHandler()=Bio.Entrez.Parser.DataHandler-class.html#startNamespaceDeclHandler"><a title="Bio.Entrez.Parser.DataHandler.startNamespaceDeclHandler" class="py-name" href="#" onclick="return doclink('link-165', 'startNamespaceDeclHandler', 'link-165');">startNamespaceDeclHandler</a></tt> </tt>
<a name="L459"></a><tt class="py-lineno">459</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">parser</tt><tt class="py-op">.</tt><tt class="py-name">EndNamespaceDeclHandler</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-166" class="py-name" targets="Method Bio.Entrez.Parser.DataHandler.endNamespaceDeclHandler()=Bio.Entrez.Parser.DataHandler-class.html#endNamespaceDeclHandler"><a title="Bio.Entrez.Parser.DataHandler.endNamespaceDeclHandler" class="py-name" href="#" onclick="return doclink('link-166', 'endNamespaceDeclHandler', 'link-166');">endNamespaceDeclHandler</a></tt> </tt>
</div><a name="L460"></a><tt class="py-lineno">460</tt>  <tt class="py-line"> </tt>
<a name="DataHandler.startNamespaceDeclHandler"></a><div id="DataHandler.startNamespaceDeclHandler-def"><a name="L461"></a><tt class="py-lineno">461</tt> <a class="py-toggle" href="#" id="DataHandler.startNamespaceDeclHandler-toggle" onclick="return toggle('DataHandler.startNamespaceDeclHandler');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DataHandler-class.html#startNamespaceDeclHandler">startNamespaceDeclHandler</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">prefix</tt><tt class="py-op">,</tt> <tt class="py-param">uri</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DataHandler.startNamespaceDeclHandler-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DataHandler.startNamespaceDeclHandler-expanded"><a name="L462"></a><tt class="py-lineno">462</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-167" class="py-name"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-167', 'prefix', 'link-26');">prefix</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'xsi'</tt><tt class="py-op">:</tt> </tt>
<a name="L463"></a><tt class="py-lineno">463</tt>  <tt class="py-line">            <tt class="py-comment"># This is an xml schema</tt> </tt>
<a name="L464"></a><tt class="py-lineno">464</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">schema_namespace</tt> <tt class="py-op">=</tt> <tt id="link-168" class="py-name" targets="Variable Bio.Phylo.NeXMLIO.uri=Bio.Phylo.NeXMLIO-module.html#uri,Method Bio.Phylo.PhyloXMLIO.Parser.uri()=Bio.Phylo.PhyloXMLIO.Parser-class.html#uri,Method Bio.Phylo.PhyloXMLIO.Writer.uri()=Bio.Phylo.PhyloXMLIO.Writer-class.html#uri,Variable Bio.Phylo.PhyloXMLIO.uri=Bio.Phylo.PhyloXMLIO-module.html#uri"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-168', 'uri', 'link-168');">uri</a></tt> </tt>
<a name="L465"></a><tt class="py-lineno">465</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">parser</tt><tt class="py-op">.</tt><tt class="py-name">StartElementHandler</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-169" class="py-name" targets="Method Bio.Entrez.Parser.DataHandler.schemaHandler()=Bio.Entrez.Parser.DataHandler-class.html#schemaHandler"><a title="Bio.Entrez.Parser.DataHandler.schemaHandler" class="py-name" href="#" onclick="return doclink('link-169', 'schemaHandler', 'link-169');">schemaHandler</a></tt> </tt>
<a name="L466"></a><tt class="py-lineno">466</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L467"></a><tt class="py-lineno">467</tt>  <tt class="py-line">            <tt class="py-comment"># Note that the DTD for MathML specifies a default attribute</tt> </tt>
<a name="L468"></a><tt class="py-lineno">468</tt>  <tt class="py-line">            <tt class="py-comment"># that declares the namespace for each MathML element. This means</tt> </tt>
<a name="L469"></a><tt class="py-lineno">469</tt>  <tt class="py-line">            <tt class="py-comment"># that MathML element in the XML has an invisible MathML namespace</tt> </tt>
<a name="L470"></a><tt class="py-lineno">470</tt>  <tt class="py-line">            <tt class="py-comment"># declaration that triggers a call to startNamespaceDeclHandler</tt> </tt>
<a name="L471"></a><tt class="py-lineno">471</tt>  <tt class="py-line">            <tt class="py-comment"># and endNamespaceDeclHandler. Therefore we need to count how often</tt> </tt>
<a name="L472"></a><tt class="py-lineno">472</tt>  <tt class="py-line">            <tt class="py-comment"># startNamespaceDeclHandler and endNamespaceDeclHandler were called</tt> </tt>
<a name="L473"></a><tt class="py-lineno">473</tt>  <tt class="py-line">            <tt class="py-comment"># to find out their first and last invocation for each namespace.</tt> </tt>
<a name="L474"></a><tt class="py-lineno">474</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">namespace_level</tt><tt class="py-op">[</tt><tt id="link-170" class="py-name"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-170', 'prefix', 'link-26');">prefix</a></tt><tt class="py-op">]</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L475"></a><tt class="py-lineno">475</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">namespace_prefix</tt><tt class="py-op">[</tt><tt id="link-171" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-171', 'uri', 'link-168');">uri</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-172" class="py-name"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-172', 'prefix', 'link-26');">prefix</a></tt> </tt>
</div><a name="L476"></a><tt class="py-lineno">476</tt>  <tt class="py-line"> </tt>
<a name="DataHandler.endNamespaceDeclHandler"></a><div id="DataHandler.endNamespaceDeclHandler-def"><a name="L477"></a><tt class="py-lineno">477</tt> <a class="py-toggle" href="#" id="DataHandler.endNamespaceDeclHandler-toggle" onclick="return toggle('DataHandler.endNamespaceDeclHandler');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DataHandler-class.html#endNamespaceDeclHandler">endNamespaceDeclHandler</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">prefix</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DataHandler.endNamespaceDeclHandler-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DataHandler.endNamespaceDeclHandler-expanded"><a name="L478"></a><tt class="py-lineno">478</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-173" class="py-name"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-173', 'prefix', 'link-26');">prefix</a></tt> <tt class="py-op">!=</tt> <tt class="py-string">'xsi'</tt><tt class="py-op">:</tt> </tt>
<a name="L479"></a><tt class="py-lineno">479</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">namespace_level</tt><tt class="py-op">[</tt><tt id="link-174" class="py-name"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-174', 'prefix', 'link-26');">prefix</a></tt><tt class="py-op">]</tt> <tt class="py-op">-=</tt> <tt class="py-number">1</tt> </tt>
<a name="L480"></a><tt class="py-lineno">480</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">namespace_level</tt><tt class="py-op">[</tt><tt id="link-175" class="py-name"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-175', 'prefix', 'link-26');">prefix</a></tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L481"></a><tt class="py-lineno">481</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt id="link-176" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-176', 'value', 'link-17');">value</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">namespace_prefix</tt><tt class="py-op">.</tt><tt id="link-177" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-177', 'items', 'link-34');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L482"></a><tt class="py-lineno">482</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt id="link-178" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-178', 'value', 'link-17');">value</a></tt> <tt class="py-op">==</tt> <tt id="link-179" class="py-name"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-179', 'prefix', 'link-26');">prefix</a></tt><tt class="py-op">:</tt> </tt>
<a name="L483"></a><tt class="py-lineno">483</tt>  <tt class="py-line">                        <tt class="py-keyword">break</tt> </tt>
<a name="L484"></a><tt class="py-lineno">484</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L485"></a><tt class="py-lineno">485</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">RuntimeError</tt><tt class="py-op">(</tt><tt class="py-string">"Failed to find namespace prefix"</tt><tt class="py-op">)</tt> </tt>
<a name="L486"></a><tt class="py-lineno">486</tt>  <tt class="py-line">                <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">namespace_prefix</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> </tt>
</div><a name="L487"></a><tt class="py-lineno">487</tt>  <tt class="py-line"> </tt>
<a name="DataHandler.schemaHandler"></a><div id="DataHandler.schemaHandler-def"><a name="L488"></a><tt class="py-lineno">488</tt> <a class="py-toggle" href="#" id="DataHandler.schemaHandler-toggle" onclick="return toggle('DataHandler.schemaHandler');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DataHandler-class.html#schemaHandler">schemaHandler</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">,</tt> <tt class="py-param">attrs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DataHandler.schemaHandler-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DataHandler.schemaHandler-expanded"><a name="L489"></a><tt class="py-lineno">489</tt>  <tt class="py-line">        <tt class="py-comment"># process the XML schema before processing the element</tt> </tt>
<a name="L490"></a><tt class="py-lineno">490</tt>  <tt class="py-line">        <tt class="py-name">key</tt> <tt class="py-op">=</tt> <tt class="py-string">"%s noNamespaceSchemaLocation"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">schema_namespace</tt> </tt>
<a name="L491"></a><tt class="py-lineno">491</tt>  <tt class="py-line">        <tt class="py-name">schema</tt> <tt class="py-op">=</tt> <tt class="py-name">attrs</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> </tt>
<a name="L492"></a><tt class="py-lineno">492</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-180" class="py-name" targets="Method Bio.Entrez.Parser.DataHandler.open_xsd_file()=Bio.Entrez.Parser.DataHandler-class.html#open_xsd_file"><a title="Bio.Entrez.Parser.DataHandler.open_xsd_file" class="py-name" href="#" onclick="return doclink('link-180', 'open_xsd_file', 'link-180');">open_xsd_file</a></tt><tt class="py-op">(</tt><tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">basename</tt><tt class="py-op">(</tt><tt class="py-name">schema</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L493"></a><tt class="py-lineno">493</tt>  <tt class="py-line">        <tt class="py-comment"># if there is no local xsd file grab the url and parse the file</tt> </tt>
<a name="L494"></a><tt class="py-lineno">494</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L495"></a><tt class="py-lineno">495</tt>  <tt class="py-line">            <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">_urlopen</tt><tt class="py-op">(</tt><tt class="py-name">schema</tt><tt class="py-op">)</tt> </tt>
<a name="L496"></a><tt class="py-lineno">496</tt>  <tt class="py-line">            <tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-181" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.ExPASy.cellosaurus.read
Bio.File.UndoHandle.read
Bio.KEGG.Enzyme.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.minimal.read
Bio.motifs.read
Bio.motifs.transfac.read
Bio.phenotype.read" class="py-name" href="#" onclick="return doclink('link-181', 'read', 'link-149');">read</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L497"></a><tt class="py-lineno">497</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-182" class="py-name" targets="Method Bio.Entrez.Parser.DataHandler.save_xsd_file()=Bio.Entrez.Parser.DataHandler-class.html#save_xsd_file"><a title="Bio.Entrez.Parser.DataHandler.save_xsd_file" class="py-name" href="#" onclick="return doclink('link-182', 'save_xsd_file', 'link-182');">save_xsd_file</a></tt><tt class="py-op">(</tt><tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">basename</tt><tt class="py-op">(</tt><tt class="py-name">schema</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
<a name="L498"></a><tt class="py-lineno">498</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-183" class="py-name" targets="Method Bio.File._IndexedSeqFileDict.close()=Bio.File._IndexedSeqFileDict-class.html#close,Method Bio.File._SQLiteManySeqFilesDict.close()=Bio.File._SQLiteManySeqFilesDict-class.html#close,Method Bio.SeqIO.SffIO._AddTellHandle.close()=Bio.SeqIO.SffIO._AddTellHandle-class.html#close,Method Bio.bgzf.BgzfReader.close()=Bio.bgzf.BgzfReader-class.html#close,Method Bio.bgzf.BgzfWriter.close()=Bio.bgzf.BgzfWriter-class.html#close,Method BioSQL.BioSeqDatabase.Adaptor.close()=BioSQL.BioSeqDatabase.Adaptor-class.html#close,Method BioSQL.BioSeqDatabase.DBServer.close()=BioSQL.BioSeqDatabase.DBServer-class.html#close"><a title="Bio.File._IndexedSeqFileDict.close
Bio.File._SQLiteManySeqFilesDict.close
Bio.SeqIO.SffIO._AddTellHandle.close
Bio.bgzf.BgzfReader.close
Bio.bgzf.BgzfWriter.close
BioSQL.BioSeqDatabase.Adaptor.close
BioSQL.BioSeqDatabase.DBServer.close" class="py-name" href="#" onclick="return doclink('link-183', 'close', 'link-183');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L499"></a><tt class="py-lineno">499</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-184" class="py-name" targets="Method Bio.Entrez.Parser.DataHandler.parse_xsd()=Bio.Entrez.Parser.DataHandler-class.html#parse_xsd"><a title="Bio.Entrez.Parser.DataHandler.parse_xsd" class="py-name" href="#" onclick="return doclink('link-184', 'parse_xsd', 'link-184');">parse_xsd</a></tt><tt class="py-op">(</tt><tt class="py-name">ET</tt><tt class="py-op">.</tt><tt class="py-name">fromstring</tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L500"></a><tt class="py-lineno">500</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L501"></a><tt class="py-lineno">501</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-185" class="py-name"><a title="Bio.Entrez.Parser.DataHandler.parse_xsd" class="py-name" href="#" onclick="return doclink('link-185', 'parse_xsd', 'link-184');">parse_xsd</a></tt><tt class="py-op">(</tt><tt class="py-name">ET</tt><tt class="py-op">.</tt><tt class="py-name">fromstring</tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-186" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.ExPASy.cellosaurus.read
Bio.File.UndoHandle.read
Bio.KEGG.Enzyme.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.minimal.read
Bio.motifs.read
Bio.motifs.transfac.read
Bio.phenotype.read" class="py-name" href="#" onclick="return doclink('link-186', 'read', 'link-149');">read</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L502"></a><tt class="py-lineno">502</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-187" class="py-name"><a title="Bio.File._IndexedSeqFileDict.close
Bio.File._SQLiteManySeqFilesDict.close
Bio.SeqIO.SffIO._AddTellHandle.close
Bio.bgzf.BgzfReader.close
Bio.bgzf.BgzfWriter.close
BioSQL.BioSeqDatabase.Adaptor.close
BioSQL.BioSeqDatabase.DBServer.close" class="py-name" href="#" onclick="return doclink('link-187', 'close', 'link-183');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L503"></a><tt class="py-lineno">503</tt>  <tt class="py-line">        <tt class="py-comment"># continue handling the element</tt> </tt>
<a name="L504"></a><tt class="py-lineno">504</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-188" class="py-name"><a title="Bio.Entrez.Parser.Consumer.startElementHandler
Bio.Entrez.Parser.DataHandler.startElementHandler
Bio.Entrez.Parser.StringConsumer.startElementHandler" class="py-name" href="#" onclick="return doclink('link-188', 'startElementHandler', 'link-160');">startElementHandler</a></tt><tt class="py-op">(</tt><tt id="link-189" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-189', 'name', 'link-11');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">attrs</tt><tt class="py-op">)</tt> </tt>
<a name="L505"></a><tt class="py-lineno">505</tt>  <tt class="py-line">        <tt class="py-comment"># reset the element handler</tt> </tt>
<a name="L506"></a><tt class="py-lineno">506</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">parser</tt><tt class="py-op">.</tt><tt class="py-name">StartElementHandler</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-190" class="py-name"><a title="Bio.Entrez.Parser.Consumer.startElementHandler
Bio.Entrez.Parser.DataHandler.startElementHandler
Bio.Entrez.Parser.StringConsumer.startElementHandler" class="py-name" href="#" onclick="return doclink('link-190', 'startElementHandler', 'link-160');">startElementHandler</a></tt> </tt>
</div><a name="L507"></a><tt class="py-lineno">507</tt>  <tt class="py-line"> </tt>
<a name="DataHandler.startElementHandler"></a><div id="DataHandler.startElementHandler-def"><a name="L508"></a><tt class="py-lineno">508</tt> <a class="py-toggle" href="#" id="DataHandler.startElementHandler-toggle" onclick="return toggle('DataHandler.startElementHandler');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DataHandler-class.html#startElementHandler">startElementHandler</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">,</tt> <tt class="py-param">attrs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DataHandler.startElementHandler-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DataHandler.startElementHandler-expanded"><a name="L509"></a><tt class="py-lineno">509</tt>  <tt class="py-line">        <tt class="py-comment"># check if the name is in a namespace</tt> </tt>
<a name="L510"></a><tt class="py-lineno">510</tt>  <tt class="py-line">        <tt id="link-191" class="py-name"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-191', 'prefix', 'link-26');">prefix</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L511"></a><tt class="py-lineno">511</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">namespace_prefix</tt><tt class="py-op">:</tt> </tt>
<a name="L512"></a><tt class="py-lineno">512</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L513"></a><tt class="py-lineno">513</tt>  <tt class="py-line">                <tt id="link-192" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-192', 'uri', 'link-168');">uri</a></tt><tt class="py-op">,</tt> <tt id="link-193" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-193', 'name', 'link-11');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-194" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-194', 'name', 'link-11');">name</a></tt><tt class="py-op">.</tt><tt id="link-195" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-195', 'split', 'link-195');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L514"></a><tt class="py-lineno">514</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L515"></a><tt class="py-lineno">515</tt>  <tt class="py-line">                <tt class="py-keyword">pass</tt> </tt>
<a name="L516"></a><tt class="py-lineno">516</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L517"></a><tt class="py-lineno">517</tt>  <tt class="py-line">                <tt id="link-196" class="py-name"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-196', 'prefix', 'link-26');">prefix</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">namespace_prefix</tt><tt class="py-op">[</tt><tt id="link-197" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-197', 'uri', 'link-168');">uri</a></tt><tt class="py-op">]</tt> </tt>
<a name="L518"></a><tt class="py-lineno">518</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">namespace_level</tt><tt class="py-op">[</tt><tt id="link-198" class="py-name"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-198', 'prefix', 'link-26');">prefix</a></tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L519"></a><tt class="py-lineno">519</tt>  <tt class="py-line">                    <tt class="py-name">attrs</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'xmlns'</tt><tt class="py-op">:</tt> <tt id="link-199" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-199', 'uri', 'link-168');">uri</a></tt><tt class="py-op">}</tt> </tt>
<a name="L520"></a><tt class="py-lineno">520</tt>  <tt class="py-line">        <tt class="py-comment"># First, check if the current consumer can use the tag</tt> </tt>
<a name="L521"></a><tt class="py-lineno">521</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">consumer</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L522"></a><tt class="py-lineno">522</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-200" class="py-name"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-200', 'prefix', 'link-26');">prefix</a></tt><tt class="py-op">:</tt> </tt>
<a name="L523"></a><tt class="py-lineno">523</tt>  <tt class="py-line">                <tt class="py-name">consumed</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-201" class="py-name"><a title="Bio.Entrez.Parser.Consumer.startElementHandler
Bio.Entrez.Parser.DataHandler.startElementHandler
Bio.Entrez.Parser.StringConsumer.startElementHandler" class="py-name" href="#" onclick="return doclink('link-201', 'startElementHandler', 'link-160');">startElementHandler</a></tt><tt class="py-op">(</tt><tt id="link-202" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-202', 'name', 'link-11');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">attrs</tt><tt class="py-op">,</tt> <tt id="link-203" class="py-name"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-203', 'prefix', 'link-26');">prefix</a></tt><tt class="py-op">)</tt> </tt>
<a name="L524"></a><tt class="py-lineno">524</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L525"></a><tt class="py-lineno">525</tt>  <tt class="py-line">                <tt class="py-name">consumed</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-204" class="py-name"><a title="Bio.Entrez.Parser.Consumer.startElementHandler
Bio.Entrez.Parser.DataHandler.startElementHandler
Bio.Entrez.Parser.StringConsumer.startElementHandler" class="py-name" href="#" onclick="return doclink('link-204', 'startElementHandler', 'link-160');">startElementHandler</a></tt><tt class="py-op">(</tt><tt id="link-205" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-205', 'name', 'link-11');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">attrs</tt><tt class="py-op">)</tt> </tt>
<a name="L526"></a><tt class="py-lineno">526</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">consumed</tt><tt class="py-op">:</tt> </tt>
<a name="L527"></a><tt class="py-lineno">527</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> </tt>
<a name="L528"></a><tt class="py-lineno">528</tt>  <tt class="py-line">        <tt class="py-name">cls</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">classes</tt><tt class="py-op">.</tt><tt id="link-206" class="py-name" targets="Method Bio.Crystal.Crystal.get()=Bio.Crystal.Crystal-class.html#get,Method Bio.Data.CodonTable.AmbiguousForwardTable.get()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#get,Method Bio.File._IndexedSeqFileDict.get()=Bio.File._IndexedSeqFileDict-class.html#get,Method Bio.File._IndexedSeqFileProxy.get()=Bio.File._IndexedSeqFileProxy-class.html#get,Method Bio.File._SQLiteManySeqFilesDict.get()=Bio.File._SQLiteManySeqFilesDict-class.html#get,Method Bio.Restriction.Restriction.RestrictionBatch.get()=Bio.Restriction.Restriction.RestrictionBatch-class.html#get,Method Bio.SearchIO._index.SearchIndexer.get()=Bio.SearchIO._index.SearchIndexer-class.html#get,Method Bio.SeqIO._index.SeqFileRandomAccess.get()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#get,Method Bio.SeqIO._index.SffRandomAccess.get()=Bio.SeqIO._index.SffRandomAccess-class.html#get,Method Bio.SeqIO._index.SffTrimedRandomAccess.get()=Bio.SeqIO._index.SffTrimedRandomAccess-class.html#get,Method Bio.SeqIO._index.UniprotRandomAccess.get()=Bio.SeqIO._index.UniprotRandomAccess-class.html#get"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-206', 'get', 'link-206');">get</a></tt><tt class="py-op">(</tt><tt id="link-207" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-207', 'name', 'link-11');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L529"></a><tt class="py-lineno">529</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">cls</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L530"></a><tt class="py-lineno">530</tt>  <tt class="py-line">            <tt class="py-comment"># Element not found in DTD</tt> </tt>
<a name="L531"></a><tt class="py-lineno">531</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">validating</tt><tt class="py-op">:</tt> </tt>
<a name="L532"></a><tt class="py-lineno">532</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt id="link-208" class="py-name" targets="Class Bio.Entrez.Parser.ValidationError=Bio.Entrez.Parser.ValidationError-class.html"><a title="Bio.Entrez.Parser.ValidationError" class="py-name" href="#" onclick="return doclink('link-208', 'ValidationError', 'link-208');">ValidationError</a></tt><tt class="py-op">(</tt><tt id="link-209" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-209', 'name', 'link-11');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L533"></a><tt class="py-lineno">533</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L534"></a><tt class="py-lineno">534</tt>  <tt class="py-line">                <tt class="py-comment"># this will not be stored in the record</tt> </tt>
<a name="L535"></a><tt class="py-lineno">535</tt>  <tt class="py-line">                <tt class="py-name">consumer</tt> <tt class="py-op">=</tt> <tt id="link-210" class="py-name" targets="Class Bio.Entrez.Parser.Consumer=Bio.Entrez.Parser.Consumer-class.html"><a title="Bio.Entrez.Parser.Consumer" class="py-name" href="#" onclick="return doclink('link-210', 'Consumer', 'link-210');">Consumer</a></tt><tt class="py-op">(</tt><tt id="link-211" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-211', 'name', 'link-11');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">attrs</tt><tt class="py-op">)</tt> </tt>
<a name="L536"></a><tt class="py-lineno">536</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L537"></a><tt class="py-lineno">537</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt> <tt class="py-op">=</tt> <tt class="py-name">cls</tt><tt class="py-op">(</tt><tt id="link-212" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-212', 'name', 'link-11');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">attrs</tt><tt class="py-op">)</tt> </tt>
<a name="L538"></a><tt class="py-lineno">538</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt class="py-name">parent</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">consumer</tt> </tt>
<a name="L539"></a><tt class="py-lineno">539</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">consumer</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L540"></a><tt class="py-lineno">540</tt>  <tt class="py-line">            <tt class="py-comment"># This is relevant only for Entrez.parse, not for Entrez.read.</tt> </tt>
<a name="L541"></a><tt class="py-lineno">541</tt>  <tt class="py-line">            <tt class="py-comment"># If self.consumer is None, then this is the first start tag we</tt> </tt>
<a name="L542"></a><tt class="py-lineno">542</tt>  <tt class="py-line">            <tt class="py-comment"># encounter, and it should refer to a list. Store this list in</tt> </tt>
<a name="L543"></a><tt class="py-lineno">543</tt>  <tt class="py-line">            <tt class="py-comment"># the record attribute, so that Entrez.parse can iterate over it.</tt> </tt>
<a name="L544"></a><tt class="py-lineno">544</tt>  <tt class="py-line">            <tt class="py-comment"># The record attribute will be set again at the last end tag;</tt> </tt>
<a name="L545"></a><tt class="py-lineno">545</tt>  <tt class="py-line">            <tt class="py-comment"># However, it doesn't hurt to set it twice.</tt> </tt>
<a name="L546"></a><tt class="py-lineno">546</tt>  <tt class="py-line">            <tt id="link-213" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-213', 'value', 'link-17');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-214" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-214', 'value', 'link-17');">value</a></tt> </tt>
<a name="L547"></a><tt class="py-lineno">547</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-215" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-215', 'value', 'link-17');">value</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L548"></a><tt class="py-lineno">548</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt id="link-216" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-216', 'value', 'link-17');">value</a></tt> </tt>
<a name="L549"></a><tt class="py-lineno">549</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">consumer</tt> <tt class="py-op">=</tt> <tt class="py-name">consumer</tt> </tt>
</div><a name="L550"></a><tt class="py-lineno">550</tt>  <tt class="py-line"> </tt>
<a name="DataHandler.endElementHandler"></a><div id="DataHandler.endElementHandler-def"><a name="L551"></a><tt class="py-lineno">551</tt> <a class="py-toggle" href="#" id="DataHandler.endElementHandler-toggle" onclick="return toggle('DataHandler.endElementHandler');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DataHandler-class.html#endElementHandler">endElementHandler</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DataHandler.endElementHandler-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DataHandler.endElementHandler-expanded"><a name="L552"></a><tt class="py-lineno">552</tt>  <tt class="py-line">        <tt id="link-217" class="py-name"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-217', 'prefix', 'link-26');">prefix</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L553"></a><tt class="py-lineno">553</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">namespace_prefix</tt><tt class="py-op">:</tt> </tt>
<a name="L554"></a><tt class="py-lineno">554</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L555"></a><tt class="py-lineno">555</tt>  <tt class="py-line">                <tt id="link-218" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-218', 'uri', 'link-168');">uri</a></tt><tt class="py-op">,</tt> <tt id="link-219" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-219', 'name', 'link-11');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-220" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-220', 'name', 'link-11');">name</a></tt><tt class="py-op">.</tt><tt id="link-221" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-221', 'split', 'link-195');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L556"></a><tt class="py-lineno">556</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L557"></a><tt class="py-lineno">557</tt>  <tt class="py-line">                <tt class="py-keyword">pass</tt> </tt>
<a name="L558"></a><tt class="py-lineno">558</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L559"></a><tt class="py-lineno">559</tt>  <tt class="py-line">                <tt id="link-222" class="py-name"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-222', 'prefix', 'link-26');">prefix</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">namespace_prefix</tt><tt class="py-op">[</tt><tt id="link-223" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-223', 'uri', 'link-168');">uri</a></tt><tt class="py-op">]</tt> </tt>
<a name="L560"></a><tt class="py-lineno">560</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">consumer</tt> </tt>
<a name="L561"></a><tt class="py-lineno">561</tt>  <tt class="py-line">        <tt class="py-comment"># First, check if the current consumer can use the tag</tt> </tt>
<a name="L562"></a><tt class="py-lineno">562</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">consumer</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L563"></a><tt class="py-lineno">563</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-224" class="py-name"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-224', 'prefix', 'link-26');">prefix</a></tt><tt class="py-op">:</tt> </tt>
<a name="L564"></a><tt class="py-lineno">564</tt>  <tt class="py-line">                <tt class="py-name">consumed</tt> <tt class="py-op">=</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-225" class="py-name"><a title="Bio.Entrez.Parser.Consumer.endElementHandler
Bio.Entrez.Parser.DataHandler.endElementHandler
Bio.Entrez.Parser.StringConsumer.endElementHandler" class="py-name" href="#" onclick="return doclink('link-225', 'endElementHandler', 'link-161');">endElementHandler</a></tt><tt class="py-op">(</tt><tt id="link-226" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-226', 'name', 'link-11');">name</a></tt><tt class="py-op">,</tt> <tt id="link-227" class="py-name"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-227', 'prefix', 'link-26');">prefix</a></tt><tt class="py-op">)</tt> </tt>
<a name="L565"></a><tt class="py-lineno">565</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L566"></a><tt class="py-lineno">566</tt>  <tt class="py-line">                <tt class="py-name">consumed</tt> <tt class="py-op">=</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-228" class="py-name"><a title="Bio.Entrez.Parser.Consumer.endElementHandler
Bio.Entrez.Parser.DataHandler.endElementHandler
Bio.Entrez.Parser.StringConsumer.endElementHandler" class="py-name" href="#" onclick="return doclink('link-228', 'endElementHandler', 'link-161');">endElementHandler</a></tt><tt class="py-op">(</tt><tt id="link-229" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-229', 'name', 'link-11');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L567"></a><tt class="py-lineno">567</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">consumed</tt><tt class="py-op">:</tt> </tt>
<a name="L568"></a><tt class="py-lineno">568</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> </tt>
<a name="L569"></a><tt class="py-lineno">569</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">consumer</tt> <tt class="py-op">=</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt class="py-name">parent</tt> </tt>
<a name="L570"></a><tt class="py-lineno">570</tt>  <tt class="py-line">        <tt id="link-230" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-230', 'value', 'link-17');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-231" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-231', 'value', 'link-17');">value</a></tt> </tt>
<a name="L571"></a><tt class="py-lineno">571</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">consumer</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L572"></a><tt class="py-lineno">572</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt id="link-232" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-232', 'value', 'link-17');">value</a></tt> </tt>
<a name="L573"></a><tt class="py-lineno">573</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-233" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-233', 'value', 'link-17');">value</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L574"></a><tt class="py-lineno">574</tt>  <tt class="py-line">            <tt id="link-234" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-234', 'name', 'link-11');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-235" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-235', 'value', 'link-17');">value</a></tt><tt class="py-op">.</tt><tt id="link-236" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-236', 'tag', 'link-23');">tag</a></tt> </tt>
<a name="L575"></a><tt class="py-lineno">575</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-237" class="py-name" targets="Method Bio.Entrez.Parser.Consumer.store()=Bio.Entrez.Parser.Consumer-class.html#store,Method Bio.Entrez.Parser.DictionaryConsumer.store()=Bio.Entrez.Parser.DictionaryConsumer-class.html#store,Method Bio.Entrez.Parser.ListConsumer.store()=Bio.Entrez.Parser.ListConsumer-class.html#store"><a title="Bio.Entrez.Parser.Consumer.store
Bio.Entrez.Parser.DictionaryConsumer.store
Bio.Entrez.Parser.ListConsumer.store" class="py-name" href="#" onclick="return doclink('link-237', 'store', 'link-237');">store</a></tt><tt class="py-op">(</tt><tt id="link-238" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-238', 'name', 'link-11');">name</a></tt><tt class="py-op">,</tt> <tt id="link-239" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-239', 'value', 'link-17');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L576"></a><tt class="py-lineno">576</tt>  <tt class="py-line"> </tt>
<a name="DataHandler.characterDataHandlerRaw"></a><div id="DataHandler.characterDataHandlerRaw-def"><a name="L577"></a><tt class="py-lineno">577</tt> <a class="py-toggle" href="#" id="DataHandler.characterDataHandlerRaw-toggle" onclick="return toggle('DataHandler.characterDataHandlerRaw');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DataHandler-class.html#characterDataHandlerRaw">characterDataHandlerRaw</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DataHandler.characterDataHandlerRaw-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DataHandler.characterDataHandlerRaw-expanded"><a name="L578"></a><tt class="py-lineno">578</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-240" class="py-name" targets="Method Bio.Entrez.Parser.Consumer.consume()=Bio.Entrez.Parser.Consumer-class.html#consume,Method Bio.Entrez.Parser.ErrorConsumer.consume()=Bio.Entrez.Parser.ErrorConsumer-class.html#consume,Method Bio.Entrez.Parser.IntegerConsumer.consume()=Bio.Entrez.Parser.IntegerConsumer-class.html#consume,Method Bio.Entrez.Parser.StringConsumer.consume()=Bio.Entrez.Parser.StringConsumer-class.html#consume"><a title="Bio.Entrez.Parser.Consumer.consume
Bio.Entrez.Parser.ErrorConsumer.consume
Bio.Entrez.Parser.IntegerConsumer.consume
Bio.Entrez.Parser.StringConsumer.consume" class="py-name" href="#" onclick="return doclink('link-240', 'consume', 'link-240');">consume</a></tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">)</tt> </tt>
</div><a name="L579"></a><tt class="py-lineno">579</tt>  <tt class="py-line"> </tt>
<a name="DataHandler.characterDataHandlerEscape"></a><div id="DataHandler.characterDataHandlerEscape-def"><a name="L580"></a><tt class="py-lineno">580</tt> <a class="py-toggle" href="#" id="DataHandler.characterDataHandlerEscape-toggle" onclick="return toggle('DataHandler.characterDataHandlerEscape');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DataHandler-class.html#characterDataHandlerEscape">characterDataHandlerEscape</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DataHandler.characterDataHandlerEscape-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DataHandler.characterDataHandlerEscape-expanded"><a name="L581"></a><tt class="py-lineno">581</tt>  <tt class="py-line">        <tt class="py-name">content</tt> <tt class="py-op">=</tt> <tt class="py-name">escape</tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">)</tt> </tt>
<a name="L582"></a><tt class="py-lineno">582</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-241" class="py-name"><a title="Bio.Entrez.Parser.Consumer.consume
Bio.Entrez.Parser.ErrorConsumer.consume
Bio.Entrez.Parser.IntegerConsumer.consume
Bio.Entrez.Parser.StringConsumer.consume" class="py-name" href="#" onclick="return doclink('link-241', 'consume', 'link-240');">consume</a></tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">)</tt> </tt>
</div><a name="L583"></a><tt class="py-lineno">583</tt>  <tt class="py-line"> </tt>
<a name="DataHandler.parse_xsd"></a><div id="DataHandler.parse_xsd-def"><a name="L584"></a><tt class="py-lineno">584</tt> <a class="py-toggle" href="#" id="DataHandler.parse_xsd-toggle" onclick="return toggle('DataHandler.parse_xsd');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DataHandler-class.html#parse_xsd">parse_xsd</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">root</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DataHandler.parse_xsd-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DataHandler.parse_xsd-expanded"><a name="L585"></a><tt class="py-lineno">585</tt>  <tt class="py-line">        <tt id="link-242" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-242', 'name', 'link-11');">name</a></tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L586"></a><tt class="py-lineno">586</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">child</tt> <tt class="py-keyword">in</tt> <tt id="link-243" class="py-name" targets="Variable Bio.Phylo.BaseTree.Clade.root=Bio.Phylo.BaseTree.Clade-class.html#root,Variable Bio.Phylo._cdao_owl.root=Bio.Phylo._cdao_owl-module.html#root"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-243', 'root', 'link-243');">root</a></tt><tt class="py-op">:</tt> </tt>
<a name="L587"></a><tt class="py-lineno">587</tt>  <tt class="py-line">            <tt class="py-name">is_dictionary</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L588"></a><tt class="py-lineno">588</tt>  <tt class="py-line">            <tt id="link-244" class="py-name"><a title="Bio.Entrez.Parser.DictionaryConsumer.multiple" class="py-name" href="#" onclick="return doclink('link-244', 'multiple', 'link-96');">multiple</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L589"></a><tt class="py-lineno">589</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-245" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.element=Bio.KEGG.KGML.KGML_pathway.Component-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Entry.element=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Graphics.element=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.element=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.element=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Relation.element=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#element,Variable Bio.Phylo._cdao_owl.element=Bio.Phylo._cdao_owl-module.html#element"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-245', 'element', 'link-245');">element</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">child</tt><tt class="py-op">.</tt><tt class="py-name">getiterator</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L590"></a><tt class="py-lineno">590</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-string">"element"</tt> <tt class="py-keyword">in</tt> <tt id="link-246" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-246', 'element', 'link-245');">element</a></tt><tt class="py-op">.</tt><tt id="link-247" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-247', 'tag', 'link-23');">tag</a></tt><tt class="py-op">:</tt> </tt>
<a name="L591"></a><tt class="py-lineno">591</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-string">"name"</tt> <tt class="py-keyword">in</tt> <tt id="link-248" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-248', 'element', 'link-245');">element</a></tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">:</tt> </tt>
<a name="L592"></a><tt class="py-lineno">592</tt>  <tt class="py-line">                        <tt id="link-249" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-249', 'name', 'link-11');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-250" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-250', 'element', 'link-245');">element</a></tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">[</tt><tt class="py-string">'name'</tt><tt class="py-op">]</tt> </tt>
<a name="L593"></a><tt class="py-lineno">593</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-string">"attribute"</tt> <tt class="py-keyword">in</tt> <tt id="link-251" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-251', 'element', 'link-245');">element</a></tt><tt class="py-op">.</tt><tt id="link-252" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-252', 'tag', 'link-23');">tag</a></tt><tt class="py-op">:</tt> </tt>
<a name="L594"></a><tt class="py-lineno">594</tt>  <tt class="py-line">                    <tt class="py-name">is_dictionary</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L595"></a><tt class="py-lineno">595</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-string">"sequence"</tt> <tt class="py-keyword">in</tt> <tt id="link-253" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-253', 'element', 'link-245');">element</a></tt><tt class="py-op">.</tt><tt id="link-254" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-254', 'tag', 'link-23');">tag</a></tt><tt class="py-op">:</tt> </tt>
<a name="L596"></a><tt class="py-lineno">596</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-name">grandchild</tt> <tt class="py-keyword">in</tt> <tt id="link-255" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-255', 'element', 'link-245');">element</a></tt><tt class="py-op">:</tt> </tt>
<a name="L597"></a><tt class="py-lineno">597</tt>  <tt class="py-line">                        <tt class="py-name">key</tt> <tt class="py-op">=</tt> <tt class="py-name">grandchild</tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">[</tt><tt class="py-string">'ref'</tt><tt class="py-op">]</tt> </tt>
<a name="L598"></a><tt class="py-lineno">598</tt>  <tt class="py-line">                        <tt id="link-256" class="py-name"><a title="Bio.Entrez.Parser.DictionaryConsumer.multiple" class="py-name" href="#" onclick="return doclink('link-256', 'multiple', 'link-96');">multiple</a></tt><tt class="py-op">.</tt><tt id="link-257" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-257', 'append', 'link-16');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">)</tt> </tt>
<a name="L599"></a><tt class="py-lineno">599</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">is_dictionary</tt><tt class="py-op">:</tt> </tt>
<a name="L600"></a><tt class="py-lineno">600</tt>  <tt class="py-line">                <tt class="py-name">bases</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt id="link-258" class="py-name"><a title="Bio.Entrez.Parser.DictionaryConsumer" class="py-name" href="#" onclick="return doclink('link-258', 'DictionaryConsumer', 'link-118');">DictionaryConsumer</a></tt><tt class="py-op">,</tt><tt class="py-op">)</tt> </tt>
<a name="L601"></a><tt class="py-lineno">601</tt>  <tt class="py-line">                <tt id="link-259" class="py-name"><a title="Bio.Entrez.Parser.DictionaryConsumer.multiple" class="py-name" href="#" onclick="return doclink('link-259', 'multiple', 'link-96');">multiple</a></tt> <tt class="py-op">=</tt> <tt id="link-260" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-260', 'set', 'link-22');">set</a></tt><tt class="py-op">(</tt><tt id="link-261" class="py-name"><a title="Bio.Entrez.Parser.DictionaryConsumer.multiple" class="py-name" href="#" onclick="return doclink('link-261', 'multiple', 'link-96');">multiple</a></tt><tt class="py-op">)</tt> </tt>
<a name="L602"></a><tt class="py-lineno">602</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">classes</tt><tt class="py-op">[</tt><tt id="link-262" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-262', 'name', 'link-11');">name</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-263" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-263', 'type', 'link-116');">type</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-264" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-264', 'name', 'link-11');">name</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L603"></a><tt class="py-lineno">603</tt>  <tt class="py-line">                                          <tt class="py-name">bases</tt><tt class="py-op">,</tt> </tt>
<a name="L604"></a><tt class="py-lineno">604</tt>  <tt class="py-line">                                          <tt class="py-op">{</tt><tt class="py-string">"multiple"</tt><tt class="py-op">:</tt> <tt id="link-265" class="py-name"><a title="Bio.Entrez.Parser.DictionaryConsumer.multiple" class="py-name" href="#" onclick="return doclink('link-265', 'multiple', 'link-96');">multiple</a></tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L605"></a><tt class="py-lineno">605</tt>  <tt class="py-line">                <tt class="py-name">is_dictionary</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L606"></a><tt class="py-lineno">606</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L607"></a><tt class="py-lineno">607</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">classes</tt><tt class="py-op">[</tt><tt id="link-266" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-266', 'name', 'link-11');">name</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-267" class="py-name"><a title="Bio.Entrez.Parser.ListConsumer" class="py-name" href="#" onclick="return doclink('link-267', 'ListConsumer', 'link-127');">ListConsumer</a></tt> </tt>
</div><a name="L608"></a><tt class="py-lineno">608</tt>  <tt class="py-line"> </tt>
<a name="DataHandler.elementDecl"></a><div id="DataHandler.elementDecl-def"><a name="L609"></a><tt class="py-lineno">609</tt> <a class="py-toggle" href="#" id="DataHandler.elementDecl-toggle" onclick="return toggle('DataHandler.elementDecl');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DataHandler-class.html#elementDecl">elementDecl</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">,</tt> <tt class="py-param">model</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DataHandler.elementDecl-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DataHandler.elementDecl-expanded"><a name="L610"></a><tt class="py-lineno">610</tt>  <tt class="py-line">        <tt class="py-docstring">"""Call a call-back function for each element declaration in a DTD.</tt> </tt>
<a name="L611"></a><tt class="py-lineno">611</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L612"></a><tt class="py-lineno">612</tt>  <tt class="py-line"><tt class="py-docstring">        This is used for each element declaration in a DTD like::</tt> </tt>
<a name="L613"></a><tt class="py-lineno">613</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L614"></a><tt class="py-lineno">614</tt>  <tt class="py-line"><tt class="py-docstring">            &lt;!ELEMENT       name          (...)&gt;</tt> </tt>
<a name="L615"></a><tt class="py-lineno">615</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L616"></a><tt class="py-lineno">616</tt>  <tt class="py-line"><tt class="py-docstring">        The purpose of this function is to determine whether this element</tt> </tt>
<a name="L617"></a><tt class="py-lineno">617</tt>  <tt class="py-line"><tt class="py-docstring">        should be regarded as a string, integer, list, dictionary, structure,</tt> </tt>
<a name="L618"></a><tt class="py-lineno">618</tt>  <tt class="py-line"><tt class="py-docstring">        or error.</tt> </tt>
<a name="L619"></a><tt class="py-lineno">619</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L620"></a><tt class="py-lineno">620</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-268" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-268', 'name', 'link-11');">name</a></tt><tt class="py-op">.</tt><tt id="link-269" class="py-name" targets="Method Bio.Seq.Seq.upper()=Bio.Seq.Seq-class.html#upper,Method Bio.Seq.UnknownSeq.upper()=Bio.Seq.UnknownSeq-class.html#upper,Method Bio.SeqRecord.SeqRecord.upper()=Bio.SeqRecord.SeqRecord-class.html#upper"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-269', 'upper', 'link-269');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">"ERROR"</tt><tt class="py-op">:</tt> </tt>
<a name="L621"></a><tt class="py-lineno">621</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">classes</tt><tt class="py-op">[</tt><tt id="link-270" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-270', 'name', 'link-11');">name</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-271" class="py-name" targets="Class Bio.Entrez.Parser.ErrorConsumer=Bio.Entrez.Parser.ErrorConsumer-class.html"><a title="Bio.Entrez.Parser.ErrorConsumer" class="py-name" href="#" onclick="return doclink('link-271', 'ErrorConsumer', 'link-271');">ErrorConsumer</a></tt> </tt>
<a name="L622"></a><tt class="py-lineno">622</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L623"></a><tt class="py-lineno">623</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-272" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-272', 'name', 'link-11');">name</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'Item'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">model</tt> <tt class="py-op">==</tt> <tt class="py-op">(</tt><tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">model</tt><tt class="py-op">.</tt><tt class="py-name">XML_CTYPE_MIXED</tt><tt class="py-op">,</tt> </tt>
<a name="L624"></a><tt class="py-lineno">624</tt>  <tt class="py-line">                                        <tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">model</tt><tt class="py-op">.</tt><tt class="py-name">XML_CQUANT_REP</tt><tt class="py-op">,</tt> </tt>
<a name="L625"></a><tt class="py-lineno">625</tt>  <tt class="py-line">                                        <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">model</tt><tt class="py-op">.</tt><tt class="py-name">XML_CTYPE_NAME</tt><tt class="py-op">,</tt> </tt>
<a name="L626"></a><tt class="py-lineno">626</tt>  <tt class="py-line">                                                <tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">model</tt><tt class="py-op">.</tt><tt class="py-name">XML_CQUANT_NONE</tt><tt class="py-op">,</tt> </tt>
<a name="L627"></a><tt class="py-lineno">627</tt>  <tt class="py-line">                                                <tt class="py-string">'Item'</tt><tt class="py-op">,</tt> </tt>
<a name="L628"></a><tt class="py-lineno">628</tt>  <tt class="py-line">                                                <tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L629"></a><tt class="py-lineno">629</tt>  <tt class="py-line">                                                <tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L630"></a><tt class="py-lineno">630</tt>  <tt class="py-line">                                               <tt class="py-op">)</tt> </tt>
<a name="L631"></a><tt class="py-lineno">631</tt>  <tt class="py-line">                                        <tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L632"></a><tt class="py-lineno">632</tt>  <tt class="py-line">            <tt class="py-comment"># Special case. As far as I can tell, this only occurs in the</tt> </tt>
<a name="L633"></a><tt class="py-lineno">633</tt>  <tt class="py-line">            <tt class="py-comment"># eSummary DTD.</tt> </tt>
<a name="L634"></a><tt class="py-lineno">634</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">classes</tt><tt class="py-op">[</tt><tt id="link-273" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-273', 'name', 'link-11');">name</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-274" class="py-name" targets="Function Bio.Entrez.Parser.select_item_consumer()=Bio.Entrez.Parser-module.html#select_item_consumer"><a title="Bio.Entrez.Parser.select_item_consumer" class="py-name" href="#" onclick="return doclink('link-274', 'select_item_consumer', 'link-274');">select_item_consumer</a></tt> </tt>
<a name="L635"></a><tt class="py-lineno">635</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L636"></a><tt class="py-lineno">636</tt>  <tt class="py-line">        <tt class="py-comment"># First, remove ignorable parentheses around declarations</tt> </tt>
<a name="L637"></a><tt class="py-lineno">637</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-op">(</tt><tt class="py-name">model</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">model</tt><tt class="py-op">.</tt><tt class="py-name">XML_CTYPE_SEQ</tt><tt class="py-op">,</tt> </tt>
<a name="L638"></a><tt class="py-lineno">638</tt>  <tt class="py-line">                            <tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">model</tt><tt class="py-op">.</tt><tt class="py-name">XML_CTYPE_CHOICE</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> </tt>
<a name="L639"></a><tt class="py-lineno">639</tt>  <tt class="py-line">               <tt class="py-name">model</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">model</tt><tt class="py-op">.</tt><tt class="py-name">XML_CQUANT_NONE</tt><tt class="py-op">,</tt> </tt>
<a name="L640"></a><tt class="py-lineno">640</tt>  <tt class="py-line">                            <tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">model</tt><tt class="py-op">.</tt><tt class="py-name">XML_CQUANT_OPT</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> </tt>
<a name="L641"></a><tt class="py-lineno">641</tt>  <tt class="py-line">               <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">model</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L642"></a><tt class="py-lineno">642</tt>  <tt class="py-line">            <tt class="py-name">model</tt> <tt class="py-op">=</tt> <tt class="py-name">model</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L643"></a><tt class="py-lineno">643</tt>  <tt class="py-line">        <tt class="py-comment"># PCDATA declarations correspond to strings</tt> </tt>
<a name="L644"></a><tt class="py-lineno">644</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">model</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">model</tt><tt class="py-op">.</tt><tt class="py-name">XML_CTYPE_MIXED</tt><tt class="py-op">,</tt> </tt>
<a name="L645"></a><tt class="py-lineno">645</tt>  <tt class="py-line">                        <tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">model</tt><tt class="py-op">.</tt><tt class="py-name">XML_CTYPE_EMPTY</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L646"></a><tt class="py-lineno">646</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">model</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">model</tt><tt class="py-op">.</tt><tt class="py-name">XML_CQUANT_REP</tt><tt class="py-op">:</tt> </tt>
<a name="L647"></a><tt class="py-lineno">647</tt>  <tt class="py-line">                <tt class="py-name">tags</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L648"></a><tt class="py-lineno">648</tt>  <tt class="py-line">                <tt id="link-275" class="py-name" targets="Method Bio.Pathway.Rep.Graph.Graph.children()=Bio.Pathway.Rep.Graph.Graph-class.html#children,Method Bio.Pathway.Rep.MultiGraph.MultiGraph.children()=Bio.Pathway.Rep.MultiGraph.MultiGraph-class.html#children"><a title="Bio.Pathway.Rep.Graph.Graph.children
Bio.Pathway.Rep.MultiGraph.MultiGraph.children" class="py-name" href="#" onclick="return doclink('link-275', 'children', 'link-275');">children</a></tt> <tt class="py-op">=</tt> <tt class="py-name">model</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt> </tt>
<a name="L649"></a><tt class="py-lineno">649</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">child</tt> <tt class="py-keyword">in</tt> <tt id="link-276" class="py-name"><a title="Bio.Pathway.Rep.Graph.Graph.children
Bio.Pathway.Rep.MultiGraph.MultiGraph.children" class="py-name" href="#" onclick="return doclink('link-276', 'children', 'link-275');">children</a></tt><tt class="py-op">:</tt> </tt>
<a name="L650"></a><tt class="py-lineno">650</tt>  <tt class="py-line">                    <tt id="link-277" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-277', 'tag', 'link-23');">tag</a></tt> <tt class="py-op">=</tt> <tt class="py-name">child</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> </tt>
<a name="L651"></a><tt class="py-lineno">651</tt>  <tt class="py-line">                    <tt class="py-name">tags</tt><tt class="py-op">.</tt><tt id="link-278" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-278', 'append', 'link-16');">append</a></tt><tt class="py-op">(</tt><tt id="link-279" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-279', 'tag', 'link-23');">tag</a></tt><tt class="py-op">)</tt> </tt>
<a name="L652"></a><tt class="py-lineno">652</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt id="link-280" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-280', 'tag', 'link-23');">tag</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">classes</tt><tt class="py-op">:</tt> </tt>
<a name="L653"></a><tt class="py-lineno">653</tt>  <tt class="py-line">                        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L654"></a><tt class="py-lineno">654</tt>  <tt class="py-line">                            <tt id="link-281" class="py-name"><a title="Bio.Crystal.Crystal.keys
Bio.Entrez.Parser.ListConsumer.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-281', 'keys', 'link-81');">keys</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">classes</tt><tt class="py-op">[</tt><tt id="link-282" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-282', 'tag', 'link-23');">tag</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-283" class="py-name"><a title="Bio.Crystal.Crystal.keys
Bio.Entrez.Parser.ListConsumer.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-283', 'keys', 'link-81');">keys</a></tt> </tt>
<a name="L655"></a><tt class="py-lineno">655</tt>  <tt class="py-line">                        <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L656"></a><tt class="py-lineno">656</tt>  <tt class="py-line">                            <tt class="py-keyword">continue</tt> </tt>
<a name="L657"></a><tt class="py-lineno">657</tt>  <tt class="py-line">                        <tt class="py-name">tags</tt><tt class="py-op">.</tt><tt id="link-284" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.extend()=Bio.Align.MultipleSeqAlignment-class.html#extend,Method Bio.SCOP.Raf.SeqMap.extend()=Bio.SCOP.Raf.SeqMap-class.html#extend,Method Bio.Seq.MutableSeq.extend()=Bio.Seq.MutableSeq-class.html#extend"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-284', 'extend', 'link-284');">extend</a></tt><tt class="py-op">(</tt><tt id="link-285" class="py-name"><a title="Bio.Crystal.Crystal.keys
Bio.Entrez.Parser.ListConsumer.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-285', 'keys', 'link-81');">keys</a></tt><tt class="py-op">)</tt> </tt>
<a name="L658"></a><tt class="py-lineno">658</tt>  <tt class="py-line">                <tt class="py-name">bases</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt id="link-286" class="py-name"><a title="Bio.Entrez.Parser.StringConsumer" class="py-name" href="#" onclick="return doclink('link-286', 'StringConsumer', 'link-131');">StringConsumer</a></tt><tt class="py-op">,</tt> <tt class="py-op">)</tt> </tt>
<a name="L659"></a><tt class="py-lineno">659</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">classes</tt><tt class="py-op">[</tt><tt id="link-287" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-287', 'name', 'link-11');">name</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-288" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-288', 'type', 'link-116');">type</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-289" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-289', 'name', 'link-11');">name</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">bases</tt><tt class="py-op">,</tt> <tt class="py-op">{</tt><tt class="py-string">'consumable'</tt><tt class="py-op">:</tt> <tt class="py-name">tags</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L660"></a><tt class="py-lineno">660</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L661"></a><tt class="py-lineno">661</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">classes</tt><tt class="py-op">[</tt><tt id="link-290" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-290', 'name', 'link-11');">name</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-291" class="py-name"><a title="Bio.Entrez.Parser.StringConsumer" class="py-name" href="#" onclick="return doclink('link-291', 'StringConsumer', 'link-131');">StringConsumer</a></tt> </tt>
<a name="L662"></a><tt class="py-lineno">662</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L663"></a><tt class="py-lineno">663</tt>  <tt class="py-line">        <tt class="py-comment"># List-type elements</tt> </tt>
<a name="L664"></a><tt class="py-lineno">664</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-op">(</tt><tt class="py-name">model</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">model</tt><tt class="py-op">.</tt><tt class="py-name">XML_CTYPE_CHOICE</tt><tt class="py-op">,</tt> </tt>
<a name="L665"></a><tt class="py-lineno">665</tt>  <tt class="py-line">                         <tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">model</tt><tt class="py-op">.</tt><tt class="py-name">XML_CTYPE_SEQ</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> </tt>
<a name="L666"></a><tt class="py-lineno">666</tt>  <tt class="py-line">            <tt class="py-name">model</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">model</tt><tt class="py-op">.</tt><tt class="py-name">XML_CQUANT_PLUS</tt><tt class="py-op">,</tt> </tt>
<a name="L667"></a><tt class="py-lineno">667</tt>  <tt class="py-line">                         <tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">model</tt><tt class="py-op">.</tt><tt class="py-name">XML_CQUANT_REP</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L668"></a><tt class="py-lineno">668</tt>  <tt class="py-line">            <tt id="link-292" class="py-name"><a title="Bio.Pathway.Rep.Graph.Graph.children
Bio.Pathway.Rep.MultiGraph.MultiGraph.children" class="py-name" href="#" onclick="return doclink('link-292', 'children', 'link-275');">children</a></tt> <tt class="py-op">=</tt> <tt class="py-name">model</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt> </tt>
<a name="L669"></a><tt class="py-lineno">669</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">model</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">model</tt><tt class="py-op">.</tt><tt class="py-name">XML_CTYPE_SEQ</tt><tt class="py-op">:</tt> </tt>
<a name="L670"></a><tt class="py-lineno">670</tt>  <tt class="py-line">                <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-293" class="py-name"><a title="Bio.Pathway.Rep.Graph.Graph.children
Bio.Pathway.Rep.MultiGraph.MultiGraph.children" class="py-name" href="#" onclick="return doclink('link-293', 'children', 'link-275');">children</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt> </tt>
<a name="L671"></a><tt class="py-lineno">671</tt>  <tt class="py-line">            <tt id="link-294" class="py-name"><a title="Bio.Crystal.Crystal.keys
Bio.Entrez.Parser.ListConsumer.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-294', 'keys', 'link-81');">keys</a></tt> <tt class="py-op">=</tt> <tt id="link-295" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-295', 'set', 'link-22');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">child</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt class="py-name">child</tt> <tt class="py-keyword">in</tt> <tt id="link-296" class="py-name"><a title="Bio.Pathway.Rep.Graph.Graph.children
Bio.Pathway.Rep.MultiGraph.MultiGraph.children" class="py-name" href="#" onclick="return doclink('link-296', 'children', 'link-275');">children</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L672"></a><tt class="py-lineno">672</tt>  <tt class="py-line">            <tt class="py-name">bases</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt id="link-297" class="py-name"><a title="Bio.Entrez.Parser.ListConsumer" class="py-name" href="#" onclick="return doclink('link-297', 'ListConsumer', 'link-127');">ListConsumer</a></tt><tt class="py-op">,</tt><tt class="py-op">)</tt> </tt>
<a name="L673"></a><tt class="py-lineno">673</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">classes</tt><tt class="py-op">[</tt><tt id="link-298" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-298', 'name', 'link-11');">name</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-299" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-299', 'type', 'link-116');">type</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-300" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-300', 'name', 'link-11');">name</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">bases</tt><tt class="py-op">,</tt> <tt class="py-op">{</tt><tt class="py-string">'keys'</tt><tt class="py-op">:</tt> <tt id="link-301" class="py-name"><a title="Bio.Crystal.Crystal.keys
Bio.Entrez.Parser.ListConsumer.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-301', 'keys', 'link-81');">keys</a></tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L674"></a><tt class="py-lineno">674</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L675"></a><tt class="py-lineno">675</tt>  <tt class="py-line">        <tt class="py-comment"># This is the tricky case. Check which keys can occur multiple</tt> </tt>
<a name="L676"></a><tt class="py-lineno">676</tt>  <tt class="py-line">        <tt class="py-comment"># times. If only one key is possible, and it can occur multiple</tt> </tt>
<a name="L677"></a><tt class="py-lineno">677</tt>  <tt class="py-line">        <tt class="py-comment"># times, then this is a list. If more than one key is possible,</tt> </tt>
<a name="L678"></a><tt class="py-lineno">678</tt>  <tt class="py-line">        <tt class="py-comment"># but none of them can occur multiple times, then this is a</tt> </tt>
<a name="L679"></a><tt class="py-lineno">679</tt>  <tt class="py-line">        <tt class="py-comment"># dictionary. Otherwise, this is a structure.</tt> </tt>
<a name="L680"></a><tt class="py-lineno">680</tt>  <tt class="py-line">        <tt class="py-comment"># In 'single' and 'multiple', we keep track which keys can occur</tt> </tt>
<a name="L681"></a><tt class="py-lineno">681</tt>  <tt class="py-line">        <tt class="py-comment"># only once, and which can occur multiple times.</tt> </tt>
<a name="L682"></a><tt class="py-lineno">682</tt>  <tt class="py-line">        <tt class="py-name">single</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L683"></a><tt class="py-lineno">683</tt>  <tt class="py-line">        <tt id="link-302" class="py-name"><a title="Bio.Entrez.Parser.DictionaryConsumer.multiple" class="py-name" href="#" onclick="return doclink('link-302', 'multiple', 'link-96');">multiple</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L684"></a><tt class="py-lineno">684</tt>  <tt class="py-line">        <tt class="py-comment"># The 'count' function is called recursively to make sure all the</tt> </tt>
<a name="L685"></a><tt class="py-lineno">685</tt>  <tt class="py-line">        <tt class="py-comment"># children in this model are counted. Error keys are ignored;</tt> </tt>
<a name="L686"></a><tt class="py-lineno">686</tt>  <tt class="py-line">        <tt class="py-comment"># they raise an exception in Python.</tt> </tt>
<a name="L687"></a><tt class="py-lineno">687</tt>  <tt class="py-line"> </tt>
<a name="L688"></a><tt class="py-lineno">688</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">count</tt><tt class="py-op">(</tt><tt class="py-param">model</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L689"></a><tt class="py-lineno">689</tt>  <tt class="py-line">            <tt class="py-name">quantifier</tt><tt class="py-op">,</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt id="link-303" class="py-name"><a title="Bio.Pathway.Rep.Graph.Graph.children
Bio.Pathway.Rep.MultiGraph.MultiGraph.children" class="py-name" href="#" onclick="return doclink('link-303', 'children', 'link-275');">children</a></tt> <tt class="py-op">=</tt> <tt class="py-name">model</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L690"></a><tt class="py-lineno">690</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L691"></a><tt class="py-lineno">691</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">quantifier</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">model</tt><tt class="py-op">.</tt><tt class="py-name">XML_CQUANT_PLUS</tt><tt class="py-op">,</tt> </tt>
<a name="L692"></a><tt class="py-lineno">692</tt>  <tt class="py-line">                                  <tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">model</tt><tt class="py-op">.</tt><tt class="py-name">XML_CQUANT_REP</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L693"></a><tt class="py-lineno">693</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-name">child</tt> <tt class="py-keyword">in</tt> <tt id="link-304" class="py-name"><a title="Bio.Pathway.Rep.Graph.Graph.children
Bio.Pathway.Rep.MultiGraph.MultiGraph.children" class="py-name" href="#" onclick="return doclink('link-304', 'children', 'link-275');">children</a></tt><tt class="py-op">:</tt> </tt>
<a name="L694"></a><tt class="py-lineno">694</tt>  <tt class="py-line">                        <tt id="link-305" class="py-name"><a title="Bio.Entrez.Parser.DictionaryConsumer.multiple" class="py-name" href="#" onclick="return doclink('link-305', 'multiple', 'link-96');">multiple</a></tt><tt class="py-op">.</tt><tt id="link-306" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-306', 'append', 'link-16');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">child</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L695"></a><tt class="py-lineno">695</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L696"></a><tt class="py-lineno">696</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-name">child</tt> <tt class="py-keyword">in</tt> <tt id="link-307" class="py-name"><a title="Bio.Pathway.Rep.Graph.Graph.children
Bio.Pathway.Rep.MultiGraph.MultiGraph.children" class="py-name" href="#" onclick="return doclink('link-307', 'children', 'link-275');">children</a></tt><tt class="py-op">:</tt> </tt>
<a name="L697"></a><tt class="py-lineno">697</tt>  <tt class="py-line">                        <tt id="link-308" class="py-name" targets="Method Bio.Crystal.Chain.count()=Bio.Crystal.Chain-class.html#count,Method Bio.Seq.MutableSeq.count()=Bio.Seq.MutableSeq-class.html#count,Method Bio.Seq.Seq.count()=Bio.Seq.Seq-class.html#count,Method Bio.Seq.UnknownSeq.count()=Bio.Seq.UnknownSeq-class.html#count,Method Bio.motifs.Instances.count()=Bio.motifs.Instances-class.html#count"><a title="Bio.Crystal.Chain.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-308', 'count', 'link-308');">count</a></tt><tt class="py-op">(</tt><tt class="py-name">child</tt><tt class="py-op">)</tt> </tt>
<a name="L698"></a><tt class="py-lineno">698</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">key</tt><tt class="py-op">.</tt><tt id="link-309" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-309', 'upper', 'link-269');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-string">"ERROR"</tt><tt class="py-op">:</tt> </tt>
<a name="L699"></a><tt class="py-lineno">699</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">quantifier</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">model</tt><tt class="py-op">.</tt><tt class="py-name">XML_CQUANT_NONE</tt><tt class="py-op">,</tt> </tt>
<a name="L700"></a><tt class="py-lineno">700</tt>  <tt class="py-line">                                  <tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">model</tt><tt class="py-op">.</tt><tt class="py-name">XML_CQUANT_OPT</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L701"></a><tt class="py-lineno">701</tt>  <tt class="py-line">                    <tt class="py-name">single</tt><tt class="py-op">.</tt><tt id="link-310" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-310', 'append', 'link-16');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">)</tt> </tt>
<a name="L702"></a><tt class="py-lineno">702</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">quantifier</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">model</tt><tt class="py-op">.</tt><tt class="py-name">XML_CQUANT_PLUS</tt><tt class="py-op">,</tt> </tt>
<a name="L703"></a><tt class="py-lineno">703</tt>  <tt class="py-line">                                    <tt class="py-name">expat</tt><tt class="py-op">.</tt><tt class="py-name">model</tt><tt class="py-op">.</tt><tt class="py-name">XML_CQUANT_REP</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L704"></a><tt class="py-lineno">704</tt>  <tt class="py-line">                    <tt id="link-311" class="py-name"><a title="Bio.Entrez.Parser.DictionaryConsumer.multiple" class="py-name" href="#" onclick="return doclink('link-311', 'multiple', 'link-96');">multiple</a></tt><tt class="py-op">.</tt><tt id="link-312" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-312', 'append', 'link-16');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">)</tt> </tt>
</div><a name="L705"></a><tt class="py-lineno">705</tt>  <tt class="py-line">        <tt id="link-313" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-313', 'count', 'link-308');">count</a></tt><tt class="py-op">(</tt><tt class="py-name">model</tt><tt class="py-op">)</tt> </tt>
<a name="L706"></a><tt class="py-lineno">706</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">single</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt> <tt class="py-keyword">and</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-314" class="py-name"><a title="Bio.Entrez.Parser.DictionaryConsumer.multiple" class="py-name" href="#" onclick="return doclink('link-314', 'multiple', 'link-96');">multiple</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L707"></a><tt class="py-lineno">707</tt>  <tt class="py-line">            <tt id="link-315" class="py-name"><a title="Bio.Crystal.Crystal.keys
Bio.Entrez.Parser.ListConsumer.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-315', 'keys', 'link-81');">keys</a></tt> <tt class="py-op">=</tt> <tt id="link-316" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-316', 'set', 'link-22');">set</a></tt><tt class="py-op">(</tt><tt id="link-317" class="py-name"><a title="Bio.Entrez.Parser.DictionaryConsumer.multiple" class="py-name" href="#" onclick="return doclink('link-317', 'multiple', 'link-96');">multiple</a></tt><tt class="py-op">)</tt> </tt>
<a name="L708"></a><tt class="py-lineno">708</tt>  <tt class="py-line">            <tt class="py-name">bases</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt id="link-318" class="py-name"><a title="Bio.Entrez.Parser.ListConsumer" class="py-name" href="#" onclick="return doclink('link-318', 'ListConsumer', 'link-127');">ListConsumer</a></tt><tt class="py-op">,</tt> <tt class="py-op">)</tt> </tt>
<a name="L709"></a><tt class="py-lineno">709</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">classes</tt><tt class="py-op">[</tt><tt id="link-319" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-319', 'name', 'link-11');">name</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-320" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-320', 'type', 'link-116');">type</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-321" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-321', 'name', 'link-11');">name</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">bases</tt><tt class="py-op">,</tt> <tt class="py-op">{</tt><tt class="py-string">'keys'</tt><tt class="py-op">:</tt> <tt id="link-322" class="py-name"><a title="Bio.Crystal.Crystal.keys
Bio.Entrez.Parser.ListConsumer.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-322', 'keys', 'link-81');">keys</a></tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L710"></a><tt class="py-lineno">710</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L711"></a><tt class="py-lineno">711</tt>  <tt class="py-line">            <tt id="link-323" class="py-name"><a title="Bio.Entrez.Parser.DictionaryConsumer.multiple" class="py-name" href="#" onclick="return doclink('link-323', 'multiple', 'link-96');">multiple</a></tt> <tt class="py-op">=</tt> <tt id="link-324" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-324', 'set', 'link-22');">set</a></tt><tt class="py-op">(</tt><tt id="link-325" class="py-name"><a title="Bio.Entrez.Parser.DictionaryConsumer.multiple" class="py-name" href="#" onclick="return doclink('link-325', 'multiple', 'link-96');">multiple</a></tt><tt class="py-op">)</tt> </tt>
<a name="L712"></a><tt class="py-lineno">712</tt>  <tt class="py-line">            <tt class="py-name">bases</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt id="link-326" class="py-name"><a title="Bio.Entrez.Parser.DictionaryConsumer" class="py-name" href="#" onclick="return doclink('link-326', 'DictionaryConsumer', 'link-118');">DictionaryConsumer</a></tt><tt class="py-op">,</tt><tt class="py-op">)</tt> </tt>
<a name="L713"></a><tt class="py-lineno">713</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">classes</tt><tt class="py-op">[</tt><tt id="link-327" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-327', 'name', 'link-11');">name</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-328" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-328', 'type', 'link-116');">type</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-329" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-329', 'name', 'link-11');">name</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L714"></a><tt class="py-lineno">714</tt>  <tt class="py-line">                                      <tt class="py-name">bases</tt><tt class="py-op">,</tt> </tt>
<a name="L715"></a><tt class="py-lineno">715</tt>  <tt class="py-line">                                      <tt class="py-op">{</tt><tt class="py-string">"multiple"</tt><tt class="py-op">:</tt> <tt id="link-330" class="py-name"><a title="Bio.Entrez.Parser.DictionaryConsumer.multiple" class="py-name" href="#" onclick="return doclink('link-330', 'multiple', 'link-96');">multiple</a></tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
</div><a name="L716"></a><tt class="py-lineno">716</tt>  <tt class="py-line"> </tt>
<a name="DataHandler.open_dtd_file"></a><div id="DataHandler.open_dtd_file-def"><a name="L717"></a><tt class="py-lineno">717</tt> <a class="py-toggle" href="#" id="DataHandler.open_dtd_file-toggle" onclick="return toggle('DataHandler.open_dtd_file');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DataHandler-class.html#open_dtd_file">open_dtd_file</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">filename</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DataHandler.open_dtd_file-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DataHandler.open_dtd_file-expanded"><a name="L718"></a><tt class="py-lineno">718</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-331" class="py-name" targets="Method Bio.Entrez.Parser.DataHandler._initialize_directory()=Bio.Entrez.Parser.DataHandler-class.html#_initialize_directory"><a title="Bio.Entrez.Parser.DataHandler._initialize_directory" class="py-name" href="#" onclick="return doclink('link-331', '_initialize_directory', 'link-331');">_initialize_directory</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L719"></a><tt class="py-lineno">719</tt>  <tt class="py-line">        <tt class="py-name">path</tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-332" class="py-name"><a title="Bio.Entrez.Parser.DataHandler.local_dtd_dir" class="py-name" href="#" onclick="return doclink('link-332', 'local_dtd_dir', 'link-140');">local_dtd_dir</a></tt><tt class="py-op">,</tt> <tt class="py-name">filename</tt><tt class="py-op">)</tt> </tt>
<a name="L720"></a><tt class="py-lineno">720</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L721"></a><tt class="py-lineno">721</tt>  <tt class="py-line">            <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt id="link-333" class="py-name" targets="Function Bio.bgzf.open()=Bio.bgzf-module.html#open"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-333', 'open', 'link-333');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">path</tt><tt class="py-op">,</tt> <tt class="py-string">"rb"</tt><tt class="py-op">)</tt> </tt>
<a name="L722"></a><tt class="py-lineno">722</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">IOError</tt><tt class="py-op">:</tt> </tt>
<a name="L723"></a><tt class="py-lineno">723</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
<a name="L724"></a><tt class="py-lineno">724</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L725"></a><tt class="py-lineno">725</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">handle</tt> </tt>
<a name="L726"></a><tt class="py-lineno">726</tt>  <tt class="py-line">        <tt class="py-name">path</tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-334" class="py-name"><a title="Bio.Entrez.Parser.DataHandler.global_dtd_dir" class="py-name" href="#" onclick="return doclink('link-334', 'global_dtd_dir', 'link-136');">global_dtd_dir</a></tt><tt class="py-op">,</tt> <tt class="py-name">filename</tt><tt class="py-op">)</tt> </tt>
<a name="L727"></a><tt class="py-lineno">727</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L728"></a><tt class="py-lineno">728</tt>  <tt class="py-line">            <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt id="link-335" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-335', 'open', 'link-333');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">path</tt><tt class="py-op">,</tt> <tt class="py-string">"rb"</tt><tt class="py-op">)</tt> </tt>
<a name="L729"></a><tt class="py-lineno">729</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">IOError</tt><tt class="py-op">:</tt> </tt>
<a name="L730"></a><tt class="py-lineno">730</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
<a name="L731"></a><tt class="py-lineno">731</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L732"></a><tt class="py-lineno">732</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">handle</tt> </tt>
<a name="L733"></a><tt class="py-lineno">733</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L734"></a><tt class="py-lineno">734</tt>  <tt class="py-line"> </tt>
<a name="DataHandler.open_xsd_file"></a><div id="DataHandler.open_xsd_file-def"><a name="L735"></a><tt class="py-lineno">735</tt> <a class="py-toggle" href="#" id="DataHandler.open_xsd_file-toggle" onclick="return toggle('DataHandler.open_xsd_file');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DataHandler-class.html#open_xsd_file">open_xsd_file</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">filename</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DataHandler.open_xsd_file-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DataHandler.open_xsd_file-expanded"><a name="L736"></a><tt class="py-lineno">736</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-336" class="py-name"><a title="Bio.Entrez.Parser.DataHandler._initialize_directory" class="py-name" href="#" onclick="return doclink('link-336', '_initialize_directory', 'link-331');">_initialize_directory</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L737"></a><tt class="py-lineno">737</tt>  <tt class="py-line">        <tt class="py-name">path</tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-337" class="py-name"><a title="Bio.Entrez.Parser.DataHandler.local_xsd_dir" class="py-name" href="#" onclick="return doclink('link-337', 'local_xsd_dir', 'link-141');">local_xsd_dir</a></tt><tt class="py-op">,</tt> <tt class="py-name">filename</tt><tt class="py-op">)</tt> </tt>
<a name="L738"></a><tt class="py-lineno">738</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L739"></a><tt class="py-lineno">739</tt>  <tt class="py-line">            <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt id="link-338" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-338', 'open', 'link-333');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">path</tt><tt class="py-op">,</tt> <tt class="py-string">"rb"</tt><tt class="py-op">)</tt> </tt>
<a name="L740"></a><tt class="py-lineno">740</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">IOError</tt><tt class="py-op">:</tt> </tt>
<a name="L741"></a><tt class="py-lineno">741</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
<a name="L742"></a><tt class="py-lineno">742</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L743"></a><tt class="py-lineno">743</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">handle</tt> </tt>
<a name="L744"></a><tt class="py-lineno">744</tt>  <tt class="py-line">        <tt class="py-name">path</tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-339" class="py-name"><a title="Bio.Entrez.Parser.DataHandler.global_xsd_dir" class="py-name" href="#" onclick="return doclink('link-339', 'global_xsd_dir', 'link-138');">global_xsd_dir</a></tt><tt class="py-op">,</tt> <tt class="py-name">filename</tt><tt class="py-op">)</tt> </tt>
<a name="L745"></a><tt class="py-lineno">745</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L746"></a><tt class="py-lineno">746</tt>  <tt class="py-line">            <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt id="link-340" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-340', 'open', 'link-333');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">path</tt><tt class="py-op">,</tt> <tt class="py-string">"rb"</tt><tt class="py-op">)</tt> </tt>
<a name="L747"></a><tt class="py-lineno">747</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">IOError</tt><tt class="py-op">:</tt> </tt>
<a name="L748"></a><tt class="py-lineno">748</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
<a name="L749"></a><tt class="py-lineno">749</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L750"></a><tt class="py-lineno">750</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">handle</tt> </tt>
<a name="L751"></a><tt class="py-lineno">751</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L752"></a><tt class="py-lineno">752</tt>  <tt class="py-line"> </tt>
<a name="DataHandler.save_dtd_file"></a><div id="DataHandler.save_dtd_file-def"><a name="L753"></a><tt class="py-lineno">753</tt> <a class="py-toggle" href="#" id="DataHandler.save_dtd_file-toggle" onclick="return toggle('DataHandler.save_dtd_file');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DataHandler-class.html#save_dtd_file">save_dtd_file</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">filename</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DataHandler.save_dtd_file-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DataHandler.save_dtd_file-expanded"><a name="L754"></a><tt class="py-lineno">754</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-341" class="py-name"><a title="Bio.Entrez.Parser.DataHandler._initialize_directory" class="py-name" href="#" onclick="return doclink('link-341', '_initialize_directory', 'link-331');">_initialize_directory</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L755"></a><tt class="py-lineno">755</tt>  <tt class="py-line">        <tt class="py-name">path</tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-342" class="py-name"><a title="Bio.Entrez.Parser.DataHandler.local_dtd_dir" class="py-name" href="#" onclick="return doclink('link-342', 'local_dtd_dir', 'link-140');">local_dtd_dir</a></tt><tt class="py-op">,</tt> <tt class="py-name">filename</tt><tt class="py-op">)</tt> </tt>
<a name="L756"></a><tt class="py-lineno">756</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L757"></a><tt class="py-lineno">757</tt>  <tt class="py-line">            <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt id="link-343" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-343', 'open', 'link-333');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">path</tt><tt class="py-op">,</tt> <tt class="py-string">"wb"</tt><tt class="py-op">)</tt> </tt>
<a name="L758"></a><tt class="py-lineno">758</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">IOError</tt><tt class="py-op">:</tt> </tt>
<a name="L759"></a><tt class="py-lineno">759</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Failed to save %s at %s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-name">path</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L760"></a><tt class="py-lineno">760</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L761"></a><tt class="py-lineno">761</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-344" class="py-name" targets="Function Bio.AlignIO.write()=Bio.AlignIO-module.html#write,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.write()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#write,Method Bio.Phylo.CDAOIO.Writer.write()=Bio.Phylo.CDAOIO.Writer-class.html#write,Function Bio.Phylo.CDAOIO.write()=Bio.Phylo.CDAOIO-module.html#write,Method Bio.Phylo.NeXMLIO.Writer.write()=Bio.Phylo.NeXMLIO.Writer-class.html#write,Function Bio.Phylo.NeXMLIO.write()=Bio.Phylo.NeXMLIO-module.html#write,Method Bio.Phylo.NewickIO.Writer.write()=Bio.Phylo.NewickIO.Writer-class.html#write,Function Bio.Phylo.NewickIO.write()=Bio.Phylo.NewickIO-module.html#write,Function Bio.Phylo.NexusIO.write()=Bio.Phylo.NexusIO-module.html#write,Method Bio.Phylo.PhyloXMLIO.Writer.write()=Bio.Phylo.PhyloXMLIO.Writer-class.html#write,Function Bio.Phylo.PhyloXMLIO.write()=Bio.Phylo.PhyloXMLIO-module.html#write,Function Bio.Phylo._io.write()=Bio.Phylo._io-module.html#write,Function Bio.SearchIO.write()=Bio.SearchIO-module.html#write,Function Bio.SeqIO.write()=Bio.SeqIO-module.html#write,Method Bio.bgzf.BgzfWriter.write()=Bio.bgzf.BgzfWriter-class.html#write,Function Bio.motifs.jaspar.write()=Bio.motifs.jaspar-module.html#write,Function Bio.motifs.transfac.write()=Bio.motifs.transfac-module.html#write,Function Bio.motifs.write()=Bio.motifs-module.html#write,Method Bio.phenotype.phen_micro.JsonWriter.write()=Bio.phenotype.phen_micro.JsonWriter-class.html#write,Function Bio.phenotype.write()=Bio.phenotype-module.html#write"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-344', 'write', 'link-344');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
<a name="L762"></a><tt class="py-lineno">762</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-345" class="py-name"><a title="Bio.File._IndexedSeqFileDict.close
Bio.File._SQLiteManySeqFilesDict.close
Bio.SeqIO.SffIO._AddTellHandle.close
Bio.bgzf.BgzfReader.close
Bio.bgzf.BgzfWriter.close
BioSQL.BioSeqDatabase.Adaptor.close
BioSQL.BioSeqDatabase.DBServer.close" class="py-name" href="#" onclick="return doclink('link-345', 'close', 'link-183');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L763"></a><tt class="py-lineno">763</tt>  <tt class="py-line"> </tt>
<a name="DataHandler.save_xsd_file"></a><div id="DataHandler.save_xsd_file-def"><a name="L764"></a><tt class="py-lineno">764</tt> <a class="py-toggle" href="#" id="DataHandler.save_xsd_file-toggle" onclick="return toggle('DataHandler.save_xsd_file');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DataHandler-class.html#save_xsd_file">save_xsd_file</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">filename</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DataHandler.save_xsd_file-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DataHandler.save_xsd_file-expanded"><a name="L765"></a><tt class="py-lineno">765</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-346" class="py-name"><a title="Bio.Entrez.Parser.DataHandler._initialize_directory" class="py-name" href="#" onclick="return doclink('link-346', '_initialize_directory', 'link-331');">_initialize_directory</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L766"></a><tt class="py-lineno">766</tt>  <tt class="py-line">        <tt class="py-name">path</tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-347" class="py-name"><a title="Bio.Entrez.Parser.DataHandler.local_xsd_dir" class="py-name" href="#" onclick="return doclink('link-347', 'local_xsd_dir', 'link-141');">local_xsd_dir</a></tt><tt class="py-op">,</tt> <tt class="py-name">filename</tt><tt class="py-op">)</tt> </tt>
<a name="L767"></a><tt class="py-lineno">767</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L768"></a><tt class="py-lineno">768</tt>  <tt class="py-line">            <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt id="link-348" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-348', 'open', 'link-333');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">path</tt><tt class="py-op">,</tt> <tt class="py-string">"wb"</tt><tt class="py-op">)</tt> </tt>
<a name="L769"></a><tt class="py-lineno">769</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">IOError</tt><tt class="py-op">:</tt> </tt>
<a name="L770"></a><tt class="py-lineno">770</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Failed to save %s at %s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-name">path</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L771"></a><tt class="py-lineno">771</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L772"></a><tt class="py-lineno">772</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-349" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-349', 'write', 'link-344');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
<a name="L773"></a><tt class="py-lineno">773</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-350" class="py-name"><a title="Bio.File._IndexedSeqFileDict.close
Bio.File._SQLiteManySeqFilesDict.close
Bio.SeqIO.SffIO._AddTellHandle.close
Bio.bgzf.BgzfReader.close
Bio.bgzf.BgzfWriter.close
BioSQL.BioSeqDatabase.Adaptor.close
BioSQL.BioSeqDatabase.DBServer.close" class="py-name" href="#" onclick="return doclink('link-350', 'close', 'link-183');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L774"></a><tt class="py-lineno">774</tt>  <tt class="py-line"> </tt>
<a name="DataHandler.externalEntityRefHandler"></a><div id="DataHandler.externalEntityRefHandler-def"><a name="L775"></a><tt class="py-lineno">775</tt> <a class="py-toggle" href="#" id="DataHandler.externalEntityRefHandler-toggle" onclick="return toggle('DataHandler.externalEntityRefHandler');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DataHandler-class.html#externalEntityRefHandler">externalEntityRefHandler</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">context</tt><tt class="py-op">,</tt> <tt class="py-param">base</tt><tt class="py-op">,</tt> <tt class="py-param">systemId</tt><tt class="py-op">,</tt> <tt class="py-param">publicId</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DataHandler.externalEntityRefHandler-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DataHandler.externalEntityRefHandler-expanded"><a name="L776"></a><tt class="py-lineno">776</tt>  <tt class="py-line">        <tt class="py-docstring">"""Handle external entiry reference in order to cache DTD locally.</tt> </tt>
<a name="L777"></a><tt class="py-lineno">777</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L778"></a><tt class="py-lineno">778</tt>  <tt class="py-line"><tt class="py-docstring">        The purpose of this function is to load the DTD locally, instead</tt> </tt>
<a name="L779"></a><tt class="py-lineno">779</tt>  <tt class="py-line"><tt class="py-docstring">        of downloading it from the URL specified in the XML. Using the local</tt> </tt>
<a name="L780"></a><tt class="py-lineno">780</tt>  <tt class="py-line"><tt class="py-docstring">        DTD results in much faster parsing. If the DTD is not found locally,</tt> </tt>
<a name="L781"></a><tt class="py-lineno">781</tt>  <tt class="py-line"><tt class="py-docstring">        we try to download it. If new DTDs become available from NCBI,</tt> </tt>
<a name="L782"></a><tt class="py-lineno">782</tt>  <tt class="py-line"><tt class="py-docstring">        putting them in Bio/Entrez/DTDs will allow the parser to see them.</tt> </tt>
<a name="L783"></a><tt class="py-lineno">783</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L784"></a><tt class="py-lineno">784</tt>  <tt class="py-line">        <tt class="py-name">urlinfo</tt> <tt class="py-op">=</tt> <tt class="py-name">_urlparse</tt><tt class="py-op">(</tt><tt class="py-name">systemId</tt><tt class="py-op">)</tt> </tt>
<a name="L785"></a><tt class="py-lineno">785</tt>  <tt class="py-line">        <tt class="py-comment"># Following attribute requires Python 2.5+</tt> </tt>
<a name="L786"></a><tt class="py-lineno">786</tt>  <tt class="py-line">        <tt class="py-comment"># if urlinfo.scheme=='http':</tt> </tt>
<a name="L787"></a><tt class="py-lineno">787</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">urlinfo</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">'http'</tt><tt class="py-op">,</tt> <tt class="py-string">'https'</tt><tt class="py-op">,</tt> <tt class="py-string">'ftp'</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L788"></a><tt class="py-lineno">788</tt>  <tt class="py-line">            <tt class="py-comment"># Then this is an absolute path to the DTD.</tt> </tt>
<a name="L789"></a><tt class="py-lineno">789</tt>  <tt class="py-line">            <tt class="py-name">url</tt> <tt class="py-op">=</tt> <tt class="py-name">systemId</tt> </tt>
<a name="L790"></a><tt class="py-lineno">790</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">urlinfo</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">''</tt><tt class="py-op">:</tt> </tt>
<a name="L791"></a><tt class="py-lineno">791</tt>  <tt class="py-line">            <tt class="py-comment"># Then this is a relative path to the DTD.</tt> </tt>
<a name="L792"></a><tt class="py-lineno">792</tt>  <tt class="py-line">            <tt class="py-comment"># Look at the parent URL to find the full path.</tt> </tt>
<a name="L793"></a><tt class="py-lineno">793</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L794"></a><tt class="py-lineno">794</tt>  <tt class="py-line">                <tt id="link-351" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.source()=Bio.GenBank._FeatureConsumer-class.html#source,Method Bio.GenBank._RecordConsumer.source()=Bio.GenBank._RecordConsumer-class.html#source,Method Bio.Pathway.Network.source()=Bio.Pathway.Network-class.html#source"><a title="Bio.GenBank._FeatureConsumer.source
Bio.GenBank._RecordConsumer.source
Bio.Pathway.Network.source" class="py-name" href="#" onclick="return doclink('link-351', 'source', 'link-351');">source</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dtd_urls</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L795"></a><tt class="py-lineno">795</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">IndexError</tt><tt class="py-op">:</tt> </tt>
<a name="L796"></a><tt class="py-lineno">796</tt>  <tt class="py-line">                <tt class="py-comment"># Assume the default URL for DTDs if the top parent</tt> </tt>
<a name="L797"></a><tt class="py-lineno">797</tt>  <tt class="py-line">                <tt class="py-comment"># does not contain an absolute path</tt> </tt>
<a name="L798"></a><tt class="py-lineno">798</tt>  <tt class="py-line">                <tt id="link-352" class="py-name"><a title="Bio.GenBank._FeatureConsumer.source
Bio.GenBank._RecordConsumer.source
Bio.Pathway.Network.source" class="py-name" href="#" onclick="return doclink('link-352', 'source', 'link-351');">source</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"http://www.ncbi.nlm.nih.gov/dtd/"</tt> </tt>
<a name="L799"></a><tt class="py-lineno">799</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L800"></a><tt class="py-lineno">800</tt>  <tt class="py-line">                <tt id="link-353" class="py-name"><a title="Bio.GenBank._FeatureConsumer.source
Bio.GenBank._RecordConsumer.source
Bio.Pathway.Network.source" class="py-name" href="#" onclick="return doclink('link-353', 'source', 'link-351');">source</a></tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">dirname</tt><tt class="py-op">(</tt><tt id="link-354" class="py-name"><a title="Bio.GenBank._FeatureConsumer.source
Bio.GenBank._RecordConsumer.source
Bio.Pathway.Network.source" class="py-name" href="#" onclick="return doclink('link-354', 'source', 'link-351');">source</a></tt><tt class="py-op">)</tt> </tt>
<a name="L801"></a><tt class="py-lineno">801</tt>  <tt class="py-line">            <tt class="py-comment"># urls always have a forward slash, don't use os.path.join</tt> </tt>
<a name="L802"></a><tt class="py-lineno">802</tt>  <tt class="py-line">            <tt class="py-name">url</tt> <tt class="py-op">=</tt> <tt id="link-355" class="py-name"><a title="Bio.GenBank._FeatureConsumer.source
Bio.GenBank._RecordConsumer.source
Bio.Pathway.Network.source" class="py-name" href="#" onclick="return doclink('link-355', 'source', 'link-351');">source</a></tt><tt class="py-op">.</tt><tt id="link-356" class="py-name" targets="Method Bio.Seq.Seq.rstrip()=Bio.Seq.Seq-class.html#rstrip"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-356', 'rstrip', 'link-356');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-string">"/"</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">"/"</tt> <tt class="py-op">+</tt> <tt class="py-name">systemId</tt> </tt>
<a name="L803"></a><tt class="py-lineno">803</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L804"></a><tt class="py-lineno">804</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected URL scheme %r"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">urlinfo</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L805"></a><tt class="py-lineno">805</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dtd_urls</tt><tt class="py-op">.</tt><tt id="link-357" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-357', 'append', 'link-16');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">url</tt><tt class="py-op">)</tt> </tt>
<a name="L806"></a><tt class="py-lineno">806</tt>  <tt class="py-line">        <tt class="py-comment"># First, try to load the local version of the DTD file</tt> </tt>
<a name="L807"></a><tt class="py-lineno">807</tt>  <tt class="py-line">        <tt id="link-358" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.location()=Bio.GenBank._FeatureConsumer-class.html#location,Method Bio.GenBank._RecordConsumer.location()=Bio.GenBank._RecordConsumer-class.html#location,Method Bio.Phylo.PhyloXMLIO.Writer.location()=Bio.Phylo.PhyloXMLIO.Writer-class.html#location"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-358', 'location', 'link-358');">location</a></tt><tt class="py-op">,</tt> <tt class="py-name">filename</tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt id="link-359" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-359', 'split', 'link-195');">split</a></tt><tt class="py-op">(</tt><tt class="py-name">systemId</tt><tt class="py-op">)</tt> </tt>
<a name="L808"></a><tt class="py-lineno">808</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-360" class="py-name" targets="Method Bio.Entrez.Parser.DataHandler.open_dtd_file()=Bio.Entrez.Parser.DataHandler-class.html#open_dtd_file"><a title="Bio.Entrez.Parser.DataHandler.open_dtd_file" class="py-name" href="#" onclick="return doclink('link-360', 'open_dtd_file', 'link-360');">open_dtd_file</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">)</tt> </tt>
<a name="L809"></a><tt class="py-lineno">809</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L810"></a><tt class="py-lineno">810</tt>  <tt class="py-line">            <tt class="py-comment"># DTD is not available as a local file. Try accessing it through</tt> </tt>
<a name="L811"></a><tt class="py-lineno">811</tt>  <tt class="py-line">            <tt class="py-comment"># the internet instead.</tt> </tt>
<a name="L812"></a><tt class="py-lineno">812</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L813"></a><tt class="py-lineno">813</tt>  <tt class="py-line">                <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">_urlopen</tt><tt class="py-op">(</tt><tt class="py-name">url</tt><tt class="py-op">)</tt> </tt>
<a name="L814"></a><tt class="py-lineno">814</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">IOError</tt><tt class="py-op">:</tt> </tt>
<a name="L815"></a><tt class="py-lineno">815</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">RuntimeError</tt><tt class="py-op">(</tt><tt class="py-string">"Failed to access %s at %s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-name">url</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L816"></a><tt class="py-lineno">816</tt>  <tt class="py-line">            <tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-361" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.ExPASy.cellosaurus.read
Bio.File.UndoHandle.read
Bio.KEGG.Enzyme.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.minimal.read
Bio.motifs.read
Bio.motifs.transfac.read
Bio.phenotype.read" class="py-name" href="#" onclick="return doclink('link-361', 'read', 'link-149');">read</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L817"></a><tt class="py-lineno">817</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-362" class="py-name"><a title="Bio.File._IndexedSeqFileDict.close
Bio.File._SQLiteManySeqFilesDict.close
Bio.SeqIO.SffIO._AddTellHandle.close
Bio.bgzf.BgzfReader.close
Bio.bgzf.BgzfWriter.close
BioSQL.BioSeqDatabase.Adaptor.close
BioSQL.BioSeqDatabase.DBServer.close" class="py-name" href="#" onclick="return doclink('link-362', 'close', 'link-183');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L818"></a><tt class="py-lineno">818</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-363" class="py-name" targets="Method Bio.Entrez.Parser.DataHandler.save_dtd_file()=Bio.Entrez.Parser.DataHandler-class.html#save_dtd_file"><a title="Bio.Entrez.Parser.DataHandler.save_dtd_file" class="py-name" href="#" onclick="return doclink('link-363', 'save_dtd_file', 'link-363');">save_dtd_file</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
<a name="L819"></a><tt class="py-lineno">819</tt>  <tt class="py-line">            <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">BytesIO</tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
<a name="L820"></a><tt class="py-lineno">820</tt>  <tt class="py-line"> </tt>
<a name="L821"></a><tt class="py-lineno">821</tt>  <tt class="py-line">        <tt class="py-name">parser</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">parser</tt><tt class="py-op">.</tt><tt class="py-name">ExternalEntityParserCreate</tt><tt class="py-op">(</tt><tt class="py-name">context</tt><tt class="py-op">)</tt> </tt>
<a name="L822"></a><tt class="py-lineno">822</tt>  <tt class="py-line">        <tt class="py-name">parser</tt><tt class="py-op">.</tt><tt class="py-name">ElementDeclHandler</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-364" class="py-name" targets="Method Bio.Entrez.Parser.DataHandler.elementDecl()=Bio.Entrez.Parser.DataHandler-class.html#elementDecl"><a title="Bio.Entrez.Parser.DataHandler.elementDecl" class="py-name" href="#" onclick="return doclink('link-364', 'elementDecl', 'link-364');">elementDecl</a></tt> </tt>
<a name="L823"></a><tt class="py-lineno">823</tt>  <tt class="py-line">        <tt class="py-name">parser</tt><tt class="py-op">.</tt><tt class="py-name">ParseFile</tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L824"></a><tt class="py-lineno">824</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-365" class="py-name"><a title="Bio.File._IndexedSeqFileDict.close
Bio.File._SQLiteManySeqFilesDict.close
Bio.SeqIO.SffIO._AddTellHandle.close
Bio.bgzf.BgzfReader.close
Bio.bgzf.BgzfWriter.close
BioSQL.BioSeqDatabase.Adaptor.close
BioSQL.BioSeqDatabase.DBServer.close" class="py-name" href="#" onclick="return doclink('link-365', 'close', 'link-183');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L825"></a><tt class="py-lineno">825</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dtd_urls</tt><tt class="py-op">.</tt><tt id="link-366" class="py-name"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop" class="py-name" href="#" onclick="return doclink('link-366', 'pop', 'link-157');">pop</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L826"></a><tt class="py-lineno">826</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-number">1</tt> </tt>
</div><a name="L827"></a><tt class="py-lineno">827</tt>  <tt class="py-line"> </tt>
<a name="DataHandler._initialize_directory"></a><div id="DataHandler._initialize_directory-def"><a name="L828"></a><tt class="py-lineno">828</tt> <a class="py-toggle" href="#" id="DataHandler._initialize_directory-toggle" onclick="return toggle('DataHandler._initialize_directory');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DataHandler-class.html#_initialize_directory">_initialize_directory</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DataHandler._initialize_directory-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DataHandler._initialize_directory-expanded"><a name="L829"></a><tt class="py-lineno">829</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the local DTD/XSD directories (PRIVATE).</tt> </tt>
<a name="L830"></a><tt class="py-lineno">830</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L831"></a><tt class="py-lineno">831</tt>  <tt class="py-line"><tt class="py-docstring">        Added to allow for custom directory (cache) locations,</tt> </tt>
<a name="L832"></a><tt class="py-lineno">832</tt>  <tt class="py-line"><tt class="py-docstring">        for example when code is deployed on AWS Lambda.</tt> </tt>
<a name="L833"></a><tt class="py-lineno">833</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L834"></a><tt class="py-lineno">834</tt>  <tt class="py-line">        <tt class="py-comment"># If user hasn't set a custom cache location, initialize it.</tt> </tt>
<a name="L835"></a><tt class="py-lineno">835</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-367" class="py-name" targets="Variable Bio.Entrez.Parser.DataHandler.directory=Bio.Entrez.Parser.DataHandler-class.html#directory"><a title="Bio.Entrez.Parser.DataHandler.directory" class="py-name" href="#" onclick="return doclink('link-367', 'directory', 'link-367');">directory</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L836"></a><tt class="py-lineno">836</tt>  <tt class="py-line">            <tt class="py-keyword">import</tt> <tt class="py-name">platform</tt> </tt>
<a name="L837"></a><tt class="py-lineno">837</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">platform</tt><tt class="py-op">.</tt><tt class="py-name">system</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">'Windows'</tt><tt class="py-op">:</tt> </tt>
<a name="L838"></a><tt class="py-lineno">838</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-368" class="py-name"><a title="Bio.Entrez.Parser.DataHandler.directory" class="py-name" href="#" onclick="return doclink('link-368', 'directory', 'link-367');">directory</a></tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">getenv</tt><tt class="py-op">(</tt><tt class="py-string">"APPDATA"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"biopython"</tt><tt class="py-op">)</tt> </tt>
<a name="L839"></a><tt class="py-lineno">839</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt>  <tt class="py-comment"># Unix/Linux/Mac</tt> </tt>
<a name="L840"></a><tt class="py-lineno">840</tt>  <tt class="py-line">                <tt class="py-name">home</tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">expanduser</tt><tt class="py-op">(</tt><tt class="py-string">'~'</tt><tt class="py-op">)</tt> </tt>
<a name="L841"></a><tt class="py-lineno">841</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-369" class="py-name"><a title="Bio.Entrez.Parser.DataHandler.directory" class="py-name" href="#" onclick="return doclink('link-369', 'directory', 'link-367');">directory</a></tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">home</tt><tt class="py-op">,</tt> <tt class="py-string">'.config'</tt><tt class="py-op">,</tt> <tt class="py-string">'biopython'</tt><tt class="py-op">)</tt> </tt>
<a name="L842"></a><tt class="py-lineno">842</tt>  <tt class="py-line">                <tt class="py-keyword">del</tt> <tt class="py-name">home</tt> </tt>
<a name="L843"></a><tt class="py-lineno">843</tt>  <tt class="py-line">            <tt class="py-keyword">del</tt> <tt class="py-name">platform</tt> </tt>
<a name="L844"></a><tt class="py-lineno">844</tt>  <tt class="py-line">        <tt class="py-comment"># Create DTD local directory</tt> </tt>
<a name="L845"></a><tt class="py-lineno">845</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-370" class="py-name"><a title="Bio.Entrez.Parser.DataHandler.local_dtd_dir" class="py-name" href="#" onclick="return doclink('link-370', 'local_dtd_dir', 'link-140');">local_dtd_dir</a></tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-371" class="py-name"><a title="Bio.Entrez.Parser.DataHandler.directory" class="py-name" href="#" onclick="return doclink('link-371', 'directory', 'link-367');">directory</a></tt><tt class="py-op">,</tt> <tt class="py-string">'Bio'</tt><tt class="py-op">,</tt> <tt class="py-string">'Entrez'</tt><tt class="py-op">,</tt> <tt class="py-string">'DTDs'</tt><tt class="py-op">)</tt> </tt>
<a name="L846"></a><tt class="py-lineno">846</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L847"></a><tt class="py-lineno">847</tt>  <tt class="py-line">            <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">makedirs</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-372" class="py-name"><a title="Bio.Entrez.Parser.DataHandler.local_dtd_dir" class="py-name" href="#" onclick="return doclink('link-372', 'local_dtd_dir', 'link-140');">local_dtd_dir</a></tt><tt class="py-op">)</tt>  <tt class="py-comment"># use exist_ok=True on Python &gt;= 3.2</tt> </tt>
<a name="L848"></a><tt class="py-lineno">848</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">OSError</tt> <tt class="py-keyword">as</tt> <tt class="py-name">exception</tt><tt class="py-op">:</tt> </tt>
<a name="L849"></a><tt class="py-lineno">849</tt>  <tt class="py-line">            <tt class="py-comment"># Check if local_dtd_dir already exists, and that it is a directory.</tt> </tt>
<a name="L850"></a><tt class="py-lineno">850</tt>  <tt class="py-line">            <tt class="py-comment"># Trying os.makedirs first and then checking for os.path.isdir avoids</tt> </tt>
<a name="L851"></a><tt class="py-lineno">851</tt>  <tt class="py-line">            <tt class="py-comment"># a race condition.</tt> </tt>
<a name="L852"></a><tt class="py-lineno">852</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">isdir</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-373" class="py-name"><a title="Bio.Entrez.Parser.DataHandler.local_dtd_dir" class="py-name" href="#" onclick="return doclink('link-373', 'local_dtd_dir', 'link-140');">local_dtd_dir</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L853"></a><tt class="py-lineno">853</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">exception</tt> </tt>
<a name="L854"></a><tt class="py-lineno">854</tt>  <tt class="py-line">        <tt class="py-comment"># Create XSD local directory</tt> </tt>
<a name="L855"></a><tt class="py-lineno">855</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-374" class="py-name"><a title="Bio.Entrez.Parser.DataHandler.local_xsd_dir" class="py-name" href="#" onclick="return doclink('link-374', 'local_xsd_dir', 'link-141');">local_xsd_dir</a></tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-375" class="py-name"><a title="Bio.Entrez.Parser.DataHandler.directory" class="py-name" href="#" onclick="return doclink('link-375', 'directory', 'link-367');">directory</a></tt><tt class="py-op">,</tt> <tt class="py-string">'Bio'</tt><tt class="py-op">,</tt> <tt class="py-string">'Entrez'</tt><tt class="py-op">,</tt> <tt class="py-string">'XSDs'</tt><tt class="py-op">)</tt> </tt>
<a name="L856"></a><tt class="py-lineno">856</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L857"></a><tt class="py-lineno">857</tt>  <tt class="py-line">            <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">makedirs</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-376" class="py-name"><a title="Bio.Entrez.Parser.DataHandler.local_xsd_dir" class="py-name" href="#" onclick="return doclink('link-376', 'local_xsd_dir', 'link-141');">local_xsd_dir</a></tt><tt class="py-op">)</tt>  <tt class="py-comment"># use exist_ok=True on Python &gt;= 3.2</tt> </tt>
<a name="L858"></a><tt class="py-lineno">858</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">OSError</tt> <tt class="py-keyword">as</tt> <tt class="py-name">exception</tt><tt class="py-op">:</tt> </tt>
<a name="L859"></a><tt class="py-lineno">859</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">isdir</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-377" class="py-name"><a title="Bio.Entrez.Parser.DataHandler.local_xsd_dir" class="py-name" href="#" onclick="return doclink('link-377', 'local_xsd_dir', 'link-141');">local_xsd_dir</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L860"></a><tt class="py-lineno">860</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">exception</tt> </tt>
</div><a name="L861"></a><tt class="py-lineno">861</tt>  <tt class="py-line"> </tt>
<a name="L862"></a><tt class="py-lineno">862</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="DataHandler.directory"></a><div id="DataHandler.directory-def"><a name="L863"></a><tt class="py-lineno">863</tt> <a class="py-toggle" href="#" id="DataHandler.directory-toggle" onclick="return toggle('DataHandler.directory');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DataHandler-class.html#directory">directory</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DataHandler.directory-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DataHandler.directory-expanded"><a name="L864"></a><tt class="py-lineno">864</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_directory</tt> </tt>
</div><a name="L865"></a><tt class="py-lineno">865</tt>  <tt class="py-line"> </tt>
<a name="L866"></a><tt class="py-lineno">866</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">directory</tt><tt class="py-op">.</tt><tt class="py-name">setter</tt> </tt>
<a name="DataHandler.directory"></a><div id="DataHandler.directory-def"><a name="L867"></a><tt class="py-lineno">867</tt> <a class="py-toggle" href="#" id="DataHandler.directory-toggle" onclick="return toggle('DataHandler.directory');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Entrez.Parser.DataHandler-class.html#directory">directory</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">directory</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DataHandler.directory-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DataHandler.directory-expanded"><a name="L868"></a><tt class="py-lineno">868</tt>  <tt class="py-line">        <tt class="py-docstring">"""Allow user to set a custom directory, also triggering subdirectory initialization."""</tt> </tt>
<a name="L869"></a><tt class="py-lineno">869</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_directory</tt> <tt class="py-op">=</tt> <tt id="link-378" class="py-name"><a title="Bio.Entrez.Parser.DataHandler.directory" class="py-name" href="#" onclick="return doclink('link-378', 'directory', 'link-367');">directory</a></tt> </tt>
<a name="L870"></a><tt class="py-lineno">870</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-379" class="py-name"><a title="Bio.Entrez.Parser.DataHandler._initialize_directory" class="py-name" href="#" onclick="return doclink('link-379', '_initialize_directory', 'link-331');">_initialize_directory</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L871"></a><tt class="py-lineno">871</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Tue Dec 18 15:58:13 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
