<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.Cluster</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        Package&nbsp;Cluster
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.Cluster-module.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== PACKAGE DESCRIPTION ==================== -->
<h1 class="epydoc">Package Cluster</h1><p class="nomargin-top"><span class="codelink"><a href="Bio.Cluster-pysrc.html">source&nbsp;code</a></span></p>
<p>Cluster Analysis.</p>
<p>The Bio.Cluster provides commonly used clustering algorithms and was
designed with the application to gene expression data in mind. However,
this module can also be used for cluster analysis of other types of data.</p>
<p>Bio.Cluster and the underlying C Clustering Library is described in
M. de Hoon et al. (2004) <a class="rst-reference external" href="https://doi.org/10.1093/bioinformatics/bth078" target="_top">https://doi.org/10.1093/bioinformatics/bth078</a></p>

<hr />
<div class="fields">      <p><strong>Version:</strong>
        1.55
      </p>
</div><!-- ==================== SUBMODULES ==================== -->
<a name="section-Submodules"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Submodules</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Submodules"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
  <tr><td class="summary">
  <ul class="nomargin">
    <li class="private"> <strong class="uidlink"><a href="Bio.Cluster._cluster-module.html" onclick="show_private();">Bio.Cluster._cluster</a></strong>: <em class="summary">C Clustering Library</em>    </li>
  </ul></td></tr>
</table>

<br />
<!-- ==================== CLASSES ==================== -->
<a name="section-Classes"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Classes</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Classes"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.Cluster.Node-class.html" class="summary-name">Node</a><br />
      A Node object describes a single node in a hierarchical clustering tree.
The integer attributes 'left' and 'right' represent the two members that
make up this node; the floating point attribute 'distance' contains the
distance between the two members of this node.
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.Cluster.Tree-class.html" class="summary-name">Tree</a>
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.Cluster.Record-class.html" class="summary-name">Record</a><br />
      Store gene expression data.
    </td>
  </tr>
</table>
<!-- ==================== FUNCTIONS ==================== -->
<a name="section-Functions"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Functions</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Functions"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Cluster-module.html#kcluster" class="summary-sig-name">kcluster</a>(<span class="summary-sig-arg">data</span>,
        <span class="summary-sig-arg">nclusters</span>=<span class="summary-sig-default">2</span>,
        <span class="summary-sig-arg">mask</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">weight</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">transpose</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">npass</span>=<span class="summary-sig-default">1</span>,
        <span class="summary-sig-arg">method</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">a</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">dist</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">e</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">initialid</span>=<span class="summary-sig-default">None</span>)</span><br />
      Perform k-means clustering.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Cluster-pysrc.html#kcluster">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Cluster-module.html#kmedoids" class="summary-sig-name">kmedoids</a>(<span class="summary-sig-arg">distance</span>,
        <span class="summary-sig-arg">nclusters</span>=<span class="summary-sig-default">2</span>,
        <span class="summary-sig-arg">npass</span>=<span class="summary-sig-default">1</span>,
        <span class="summary-sig-arg">initialid</span>=<span class="summary-sig-default">None</span>)</span><br />
      Perform k-medoids clustering.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Cluster-pysrc.html#kmedoids">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Cluster-module.html#treecluster" class="summary-sig-name">treecluster</a>(<span class="summary-sig-arg">data</span>,
        <span class="summary-sig-arg">mask</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">weight</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">transpose</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">method</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">m</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">dist</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">e</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">distancematrix</span>=<span class="summary-sig-default">None</span>)</span><br />
      Perform hierarchical clustering, and return a Tree object.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Cluster-pysrc.html#treecluster">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Cluster-module.html#somcluster" class="summary-sig-name">somcluster</a>(<span class="summary-sig-arg">data</span>,
        <span class="summary-sig-arg">mask</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">weight</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">transpose</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">nxgrid</span>=<span class="summary-sig-default">2</span>,
        <span class="summary-sig-arg">nygrid</span>=<span class="summary-sig-default">1</span>,
        <span class="summary-sig-arg">inittau</span>=<span class="summary-sig-default">0.02</span>,
        <span class="summary-sig-arg">niter</span>=<span class="summary-sig-default">1</span>,
        <span class="summary-sig-arg">dist</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">e</code><code class="variable-quote">'</code></span>)</span><br />
      Calculate a Self-Organizing Map.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Cluster-pysrc.html#somcluster">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Cluster-module.html#clusterdistance" class="summary-sig-name">clusterdistance</a>(<span class="summary-sig-arg">data</span>,
        <span class="summary-sig-arg">mask</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">weight</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">index1</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">index2</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">method</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">a</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">dist</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">e</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">transpose</span>=<span class="summary-sig-default">False</span>)</span><br />
      Calculate and return the distance between two clusters.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Cluster-pysrc.html#clusterdistance">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Cluster-module.html#clustercentroids" class="summary-sig-name">clustercentroids</a>(<span class="summary-sig-arg">data</span>,
        <span class="summary-sig-arg">mask</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">clusterid</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">method</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">a</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">transpose</span>=<span class="summary-sig-default">False</span>)</span><br />
      Calculate and return the centroid of each cluster.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Cluster-pysrc.html#clustercentroids">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Cluster-module.html#distancematrix" class="summary-sig-name">distancematrix</a>(<span class="summary-sig-arg">data</span>,
        <span class="summary-sig-arg">mask</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">weight</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">transpose</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">dist</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">e</code><code class="variable-quote">'</code></span>)</span><br />
      Calculate and return a distance matrix from the data.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Cluster-pysrc.html#distancematrix">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Cluster-module.html#pca" class="summary-sig-name">pca</a>(<span class="summary-sig-arg">data</span>)</span><br />
      Perform principal component analysis.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Cluster-pysrc.html#pca">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Cluster-module.html#read" class="summary-sig-name">read</a>(<span class="summary-sig-arg">handle</span>)</span><br />
      Read gene expression data from the file handle and return a Record.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Cluster-pysrc.html#read">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__check_data"></a><span class="summary-sig-name">__check_data</span>(<span class="summary-sig-arg">data</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Cluster-pysrc.html#__check_data">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__check_mask"></a><span class="summary-sig-name">__check_mask</span>(<span class="summary-sig-arg">mask</span>,
        <span class="summary-sig-arg">shape</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Cluster-pysrc.html#__check_mask">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__check_weight"></a><span class="summary-sig-name">__check_weight</span>(<span class="summary-sig-arg">weight</span>,
        <span class="summary-sig-arg">ndata</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Cluster-pysrc.html#__check_weight">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__check_initialid"></a><span class="summary-sig-name">__check_initialid</span>(<span class="summary-sig-arg">initialid</span>,
        <span class="summary-sig-arg">npass</span>,
        <span class="summary-sig-arg">nitems</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Cluster-pysrc.html#__check_initialid">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__check_index"></a><span class="summary-sig-name">__check_index</span>(<span class="summary-sig-arg">index</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Cluster-pysrc.html#__check_index">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__check_distancematrix"></a><span class="summary-sig-name">__check_distancematrix</span>(<span class="summary-sig-arg">distancematrix</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Cluster-pysrc.html#__check_distancematrix">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
</table>
<!-- ==================== VARIABLES ==================== -->
<a name="section-Variables"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Variables</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Variables"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="__package__"></a><span class="summary-name">__package__</span> = <code title="'Bio.Cluster'"><code class="variable-quote">'</code><code class="variable-string">Bio.Cluster</code><code class="variable-quote">'</code></code>
    </td>
  </tr>
</table>
<!-- ==================== FUNCTION DETAILS ==================== -->
<a name="section-FunctionDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Function Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-FunctionDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="kcluster"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">kcluster</span>(<span class="sig-arg">data</span>,
        <span class="sig-arg">nclusters</span>=<span class="sig-default">2</span>,
        <span class="sig-arg">mask</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">weight</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">transpose</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">npass</span>=<span class="sig-default">1</span>,
        <span class="sig-arg">method</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">a</code><code class="variable-quote">'</code></span>,
        <span class="sig-arg">dist</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">e</code><code class="variable-quote">'</code></span>,
        <span class="sig-arg">initialid</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Cluster-pysrc.html#kcluster">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Perform k-means clustering.</p>
<p>This function performs k-means clustering on the values in data, and
returns the cluster assignments, the within-cluster sum of distances
of the optimal k-means clustering solution, and the number of times
the optimal solution was found.</p>
<dl class="rst-docutils">
<dt>Keyword arguments:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>data: nrows x ncolumns array containing the data values.</li>
<li>nclusters: number of clusters (the 'k' in k-means).</li>
<li>mask: nrows x ncolumns array of integers, showing which data
are missing. If mask[i,j]==0, then data[i,j] is missing.</li>
<li>weight: the weights to be used when calculating distances</li>
<li>transpose:
- if False: rows are clustered;
- if True: columns are clustered.</li>
<li>npass: number of times the k-means clustering algorithm is
performed, each time with a different (random) initial
condition.</li>
<li>method: specifies how the center of a cluster is found:
- method == 'a': arithmetic mean;
- method == 'm': median.</li>
<li>dist: specifies the distance function to be used:
- dist == 'e': Euclidean distance;
- dist == 'b': City Block distance;
- dist == 'c': Pearson correlation;
- dist == 'a': absolute value of the correlation;
- dist == 'u': uncentered correlation;
- dist == 'x': absolute uncentered correlation;
- dist == 's': Spearman's rank correlation;
- dist == 'k': Kendall's tau.</li>
<li>initialid: the initial clustering from which the algorithm
should start.
If initialid is None, the routine carries out npass
repetitions of the EM algorithm, each time starting from a
different random initial clustering. If initialid is given,
the routine carries out the EM algorithm only once, starting
from the given initial clustering and without randomizing the
order in which items are assigned to clusters (i.e., using
the same order as in the data matrix). In that case, the
k-means algorithm is fully deterministic.</li>
</ul>
</dd>
<dt>Return values:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>clusterid: array containing the number of the cluster to which each
item was assigned in the best k-means clustering solution that was
found in the npass runs;</li>
<li>error: the within-cluster sum of distances for the returned k-means
clustering solution;</li>
<li>nfound: the number of times this solution was found.</li>
</ul>
</dd>
</dl>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="kmedoids"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">kmedoids</span>(<span class="sig-arg">distance</span>,
        <span class="sig-arg">nclusters</span>=<span class="sig-default">2</span>,
        <span class="sig-arg">npass</span>=<span class="sig-default">1</span>,
        <span class="sig-arg">initialid</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Cluster-pysrc.html#kmedoids">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Perform k-medoids clustering.</p>
<p>This function performs k-medoids clustering, and returns the cluster
assignments, the within-cluster sum of distances of the optimal
k-medoids clustering solution, and the number of times the optimal
solution was found.</p>
<dl class="rst-docutils">
<dt>Keyword arguments:</dt>
<dd><ul class="rst-first rst-last">
<li><p class="rst-first">distance: The distance matrix between the items. There are three
ways in which you can pass a distance matrix:
1. a 2D Numerical Python array (in which only the left-lower
part of the array will be accessed);
2. a 1D Numerical Python array containing the distances
consecutively;
3. a list of rows containing the lower-triangular part of
the distance matrix.</p>
<p>Examples are:</p>
<blockquote>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-comment"># option 1.:</span>
<span class="py-prompt">&gt;&gt;&gt; </span>distance = array([[0.0, 1.1, 2.3],
<span class="py-more">... </span>                  [1.1, 0.0, 4.5],
<span class="py-more">... </span>                  [2.3, 4.5, 0.0]])
<span class="py-more">...</span>
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-comment"># option 2.:</span>
<span class="py-prompt">&gt;&gt;&gt; </span>distance = array([1.1, 2.3, 4.5])
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-comment"># option 3.:</span>
<span class="py-prompt">&gt;&gt;&gt; </span>distance = [array([]),
<span class="py-more">... </span>            array([1.1]),
<span class="py-more">... </span>            array([2.3, 4.5])]
<span class="py-more">...</span>
<span class="py-prompt">&gt;&gt;&gt;</span></pre>
</blockquote>
<p>These three correspond to the same distance matrix.</p>
</li>
<li><p class="rst-first">nclusters: number of clusters (the 'k' in k-medoids)</p>
</li>
<li><p class="rst-first">npass: the number of times the k-medoids clustering algorithm
is performed, each time with a different (random) initial
condition.</p>
</li>
<li><p class="rst-first">initialid: the initial clustering from which the algorithm should start.
If initialid is not given, the routine carries out npass
repetitions of the EM algorithm, each time starting from a
different random initial clustering. If initialid is given,
the routine carries out the EM algorithm only once, starting
from the initial clustering specified by initialid and
without randomizing the order in which items are assigned to
clusters (i.e., using the same order as in the data matrix).
In that case, the k-medoids algorithm is fully deterministic.</p>
</li>
</ul>
</dd>
<dt>Return values:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>clusterid: array containing the number of the cluster to which each
item was assigned in the best k-means clustering solution that was
found in the npass runs;</li>
<li>error: the within-cluster sum of distances for the returned k-means
clustering solution;</li>
<li>nfound: the number of times this solution was found.</li>
</ul>
</dd>
</dl>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="treecluster"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">treecluster</span>(<span class="sig-arg">data</span>,
        <span class="sig-arg">mask</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">weight</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">transpose</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">method</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">m</code><code class="variable-quote">'</code></span>,
        <span class="sig-arg">dist</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">e</code><code class="variable-quote">'</code></span>,
        <span class="sig-arg">distancematrix</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Cluster-pysrc.html#treecluster">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Perform hierarchical clustering, and return a Tree object.</p>
<p>This function implements the pairwise single, complete, centroid, and
average linkage hierarchical clustering methods.</p>
<dl class="rst-docutils">
<dt>Keyword arguments:</dt>
<dd><ul class="rst-first rst-last">
<li><p class="rst-first">data: nrows x ncolumns array containing the data values.</p>
</li>
<li><p class="rst-first">mask: nrows x ncolumns array of integers, showing which data are
missing. If mask[i][j]==0, then data[i][j] is missing.</p>
</li>
<li><p class="rst-first">weight: the weights to be used when calculating distances.</p>
</li>
<li><p class="rst-first">transpose:
- if False, rows are clustered;
- if True, columns are clustered.</p>
</li>
<li><p class="rst-first">dist: specifies the distance function to be used:
- dist == 'e': Euclidean distance
- dist == 'b': City Block distance
- dist == 'c': Pearson correlation
- dist == 'a': absolute value of the correlation
- dist == 'u': uncentered correlation
- dist == 'x': absolute uncentered correlation
- dist == 's': Spearman's rank correlation
- dist == 'k': Kendall's tau</p>
</li>
<li><p class="rst-first">method: specifies which linkage method is used:
- method == 's': Single pairwise linkage
- method == 'm': Complete (maximum) pairwise linkage (default)
- method == 'c': Centroid linkage
- method == 'a': Average pairwise linkage</p>
</li>
<li><p class="rst-first">distancematrix:  The distance matrix between the items. There are
three ways in which you can pass a distance matrix:
1. a 2D Numerical Python array (in which only the left-lower
part of the array will be accessed);
2. a 1D Numerical Python array containing the distances
consecutively;
3. a list of rows containing the lower-triangular part of
the distance matrix.</p>
<p>Examples are:</p>
<blockquote>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-comment"># option 1.:</span>
<span class="py-prompt">&gt;&gt;&gt; </span>distance = array([[0.0, 1.1, 2.3],
<span class="py-more">... </span>                  [1.1, 0.0, 4.5],
<span class="py-more">... </span>                  [2.3, 4.5, 0.0]])
<span class="py-more">...</span>
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-comment"># option 2.:</span>
<span class="py-prompt">&gt;&gt;&gt; </span>distance = array([1.1, 2.3, 4.5])
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-comment"># option 3.:</span>
<span class="py-prompt">&gt;&gt;&gt; </span>distance = [array([]),
<span class="py-more">... </span>            array([1.1]),
<span class="py-more">... </span>            array([2.3, 4.5])]
<span class="py-more">...</span>
<span class="py-prompt">&gt;&gt;&gt;</span></pre>
</blockquote>
<p>These three correspond to the same distance matrix.</p>
<p>PLEASE NOTE:
As the treecluster routine may shuffle the values in the
distance matrix as part of the clustering algorithm, be sure
to save this array in a different variable before calling
treecluster if you need it later.</p>
</li>
</ul>
</dd>
</dl>
<p>Either data or distancematrix should be None. If distancematrix is None,
the hierarchical clustering solution is calculated from the values stored
in the argument data. If data is None, the hierarchical clustering solution
is instead calculated from the distance matrix. Pairwise centroid-linkage
clustering can be performed only from the data values and not from the
distance matrix. Pairwise single-, maximum-, and average-linkage clustering
can be calculated from the data values or from the distance matrix.</p>
<p>Return value:
treecluster returns a Tree object describing the hierarchical clustering
result. See the description of the Tree class for more information.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="somcluster"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">somcluster</span>(<span class="sig-arg">data</span>,
        <span class="sig-arg">mask</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">weight</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">transpose</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">nxgrid</span>=<span class="sig-default">2</span>,
        <span class="sig-arg">nygrid</span>=<span class="sig-default">1</span>,
        <span class="sig-arg">inittau</span>=<span class="sig-default">0.02</span>,
        <span class="sig-arg">niter</span>=<span class="sig-default">1</span>,
        <span class="sig-arg">dist</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">e</code><code class="variable-quote">'</code></span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Cluster-pysrc.html#somcluster">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculate a Self-Organizing Map.</p>
<p>This function implements a Self-Organizing Map on a rectangular grid.</p>
<dl class="rst-docutils">
<dt>Keyword arguments:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>data: nrows x ncolumns array containing the data values;</li>
<li>mask: nrows x ncolumns array of integers, showing which data are
missing. If mask[i][j]==0, then data[i][j] is missing.</li>
<li>weight: the weights to be used when calculating distances</li>
<li>transpose:
- if False: rows are clustered;
- if True: columns are clustered.</li>
<li>nxgrid: the horizontal dimension of the rectangular SOM map</li>
<li>nygrid: the vertical dimension of the rectangular SOM map</li>
<li>inittau: the initial value of tau (the neighborbood function)</li>
<li>niter: the number of iterations</li>
<li>dist: specifies the distance function to be used:
- dist == 'e': Euclidean distance
- dist == 'b': City Block distance
- dist == 'c': Pearson correlation
- dist == 'a': absolute value of the correlation
- dist == 'u': uncentered correlation
- dist == 'x': absolute uncentered correlation
- dist == 's': Spearman's rank correlation
- dist == 'k': Kendall's tau</li>
</ul>
</dd>
</dl>
<p>Return values:</p>
<blockquote>
<ul class="rst-simple">
<li>clusterid: array with two columns, with the number of rows equal to
the items that are being clustered. Each row in the array contains
the x and y coordinates of the cell in the rectangular SOM grid to
which the item was assigned.</li>
<li>celldata:  an array with dimensions [nxgrid, nygrid, number of columns]
if rows are being clustered, or [nxgrid, nygrid, number of rows) if
columns are being clustered.
Each element [ix, iy] of this array is a 1D vector containing the
data values for the centroid of the cluster in the SOM grid cell
with coordinates [ix, iy].</li>
</ul>
</blockquote>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="clusterdistance"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">clusterdistance</span>(<span class="sig-arg">data</span>,
        <span class="sig-arg">mask</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">weight</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">index1</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">index2</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">method</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">a</code><code class="variable-quote">'</code></span>,
        <span class="sig-arg">dist</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">e</code><code class="variable-quote">'</code></span>,
        <span class="sig-arg">transpose</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Cluster-pysrc.html#clusterdistance">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculate and return the distance between two clusters.</p>
<dl class="rst-docutils">
<dt>Keyword arguments:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>data: nrows x ncolumns array containing the data values.</li>
<li>mask: nrows x ncolumns array of integers, showing which data are
missing. If mask[i, j]==0, then data[i, j] is missing.</li>
<li>weight: the weights to be used when calculating distances</li>
<li>index1: 1D array identifying which items belong to the
first cluster. If the cluster contains only one item, then
index1 can also be written as a single integer.</li>
<li>index2: 1D array identifying which items belong to the
second cluster. If the cluster contains only one item, then
index2 can also be written as a single integer.</li>
<li>dist: specifies the distance function to be used:
- dist == 'e': Euclidean distance
- dist == 'b': City Block distance
- dist == 'c': Pearson correlation
- dist == 'a': absolute value of the correlation
- dist == 'u': uncentered correlation
- dist == 'x': absolute uncentered correlation
- dist == 's': Spearman's rank correlation
- dist == 'k': Kendall's tau</li>
<li>method: specifies how the distance between two clusters is defined:
- method == 'a': the distance between the arithmetic means
of the two clusters
- method == 'm': the distance between the medians of the two clusters
- method == 's': the smallest pairwise distance between members
of the two clusters
- method == 'x': the largest pairwise distance between members
of the two clusters
- method == 'v': average of the pairwise distances between members
of the two clusters</li>
<li>transpose:
- if False: clusters of rows are considered;
- if True: clusters of columns are considered.</li>
</ul>
</dd>
</dl>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="clustercentroids"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">clustercentroids</span>(<span class="sig-arg">data</span>,
        <span class="sig-arg">mask</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">clusterid</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">method</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">a</code><code class="variable-quote">'</code></span>,
        <span class="sig-arg">transpose</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Cluster-pysrc.html#clustercentroids">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculate and return the centroid of each cluster.</p>
<p>The clustercentroids routine calculates the cluster centroids, given to
which cluster each item belongs. The centroid is defined as either
the mean or the median over all items for each dimension.</p>
<dl class="rst-docutils">
<dt>Keyword arguments:</dt>
<dd><ul class="rst-first rst-last">
<li><p class="rst-first">data: nrows x ncolumns array containing the data values.</p>
</li>
<li><p class="rst-first">mask: nrows x ncolumns array of integers, showing which data are
missing. If mask[i, j]==0, then data[i, j] is missing.</p>
</li>
<li><p class="rst-first">clusterid: array containing the cluster number for each item.
The cluster number should be non-negative.</p>
</li>
<li><p class="rst-first">method: specifies whether the centroid is calculated from the
arithmetic mean (method == 'a', default) or the median (method == 'm')
over each dimension.</p>
</li>
<li><dl class="rst-first rst-docutils">
<dt>transpose: if False, each row contains the data for one item;</dt>
<dd><p class="rst-first rst-last">if True, each column contains the data for one item.</p>
</dd>
</dl>
</li>
</ul>
</dd>
<dt>Return values:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>cdata: 2D array containing the cluster centroids.
If transpose is False, then the dimensions of cdata are
nclusters x ncolumns.
If transpose is True, then the dimensions of cdata are
nrows x nclusters.</li>
<li>cmask: 2D array of integers describing which items in cdata,
if any, are missing.</li>
</ul>
</dd>
</dl>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="distancematrix"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">distancematrix</span>(<span class="sig-arg">data</span>,
        <span class="sig-arg">mask</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">weight</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">transpose</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">dist</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">e</code><code class="variable-quote">'</code></span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Cluster-pysrc.html#distancematrix">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculate and return a distance matrix from the data.</p>
<p>This function returns the distance matrix calculated from the data.</p>
<dl class="rst-docutils">
<dt>Keyword arguments:</dt>
<dd><ul class="rst-first rst-last">
<li><p class="rst-first">data: nrows x ncolumns array containing the data values.</p>
</li>
<li><p class="rst-first">mask: nrows x ncolumns array of integers, showing which data are
missing. If mask[i, j]==0, then data[i, j] is missing.</p>
</li>
<li><p class="rst-first">weight: the weights to be used when calculating distances.</p>
</li>
<li><dl class="rst-first rst-docutils">
<dt>transpose: if False: the distances between rows are calculated;</dt>
<dd><p class="rst-first rst-last">if True:  the distances between columns are calculated.</p>
</dd>
</dl>
</li>
<li><p class="rst-first">dist: specifies the distance function to be used:
- dist == 'e': Euclidean distance
- dist == 'b': City Block distance
- dist == 'c': Pearson correlation
- dist == 'a': absolute value of the correlation
- dist == 'u': uncentered correlation
- dist == 'x': absolute uncentered correlation
- dist == 's': Spearman's rank correlation
- dist == 'k': Kendall's tau</p>
</li>
</ul>
</dd>
</dl>
<p>Return value:
The distance matrix is returned as a list of 1D arrays containing the
distance matrix calculated from the data. The number of columns in eac
row is equal to the row number. Hence, the first row has zero length.
An example of the return value is:</p>
<blockquote>
<dl class="rst-docutils">
<dt>matrix = [[],</dt>
<dd>array([1.]),
array([7., 3.]),
array([4., 2., 6.])]</dd>
</dl>
</blockquote>
<p>This corresponds to the distance matrix:</p>
<blockquote>
[0., 1., 7., 4.]
[1., 0., 3., 2.]
[7., 3., 0., 6.]
[4., 2., 6., 0.]</blockquote>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="pca"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">pca</span>(<span class="sig-arg">data</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Cluster-pysrc.html#pca">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Perform principal component analysis.</p>
<dl class="rst-docutils">
<dt>Keyword arguments:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>data: nrows x ncolumns array containing the data values.</li>
</ul>
</dd>
</dl>
<p>Return value:
This function returns an array containing the mean of each column, the
principal components as an nmin x ncolumns array, as well as the
coordinates (an nrows x nmin array) of the data along the principal
components, and the associated eigenvalues. The principal components, the
coordinates, and the eigenvalues are sorted by the magnitude of the
eigenvalue, with the largest eigenvalues appearing first. Here, nmin is
the smaller of nrows and ncolumns.
Adding the column means to the dot product of the coordinates and the
principal components,</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>columnmean + dot(coordinates, pc)</pre>
<p>recreates the data matrix.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="read"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">read</span>(<span class="sig-arg">handle</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Cluster-pysrc.html#read">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Read gene expression data from the file handle and return a Record.</p>
<p>The file should be in the file format defined for Michael Eisen's
Cluster/TreeView program.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Fri Jun 22 16:34:19 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
