<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.Phylo.PhyloXMLIO</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.Phylo-module.html">Package&nbsp;Phylo</a> ::
        Module&nbsp;PhyloXMLIO
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.Phylo.PhyloXMLIO-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.Phylo.PhyloXMLIO-module.html">Module Bio.Phylo.PhyloXMLIO</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright (C) 2009 by Eric Talevich (eric.talevich@gmail.com)</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># license. Please see the LICENSE file that should have been included</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-docstring">"""PhyloXML reader/parser, writer, and associated functions.</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-docstring">Instantiates tree elements from a parsed PhyloXML file, and constructs an XML</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-docstring">file from a ``Bio.Phylo.PhyloXML`` object.</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-docstring">About capitalization:</tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-docstring"> - phyloXML means the file format specification</tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-docstring"> - PhyloXML means the Biopython module ``Bio.Phylo.PhyloXML`` and its classes</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-docstring"> - Phyloxml means the top-level class used by ``PhyloXMLIO.read`` (but not</tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-docstring">   ``Bio.Phylo.read``!), containing a list of Phylogenies (objects derived from</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-docstring">   ``BaseTree.Tree``)</tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">sys</tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-1', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">basestring</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-2" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-2', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-3" class="py-name"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-3', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">unicode</tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-4" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-4', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-5" class="py-name" targets="Package Bio.Phylo=Bio.Phylo-module.html"><a title="Bio.Phylo" class="py-name" href="#" onclick="return doclink('link-5', 'Phylo', 'link-5');">Phylo</a></tt> <tt class="py-keyword">import</tt> <tt id="link-6" class="py-name" targets="Module Bio.Phylo.PhyloXML=Bio.Phylo.PhyloXML-module.html"><a title="Bio.Phylo.PhyloXML" class="py-name" href="#" onclick="return doclink('link-6', 'PhyloXML', 'link-6');">PhyloXML</a></tt> <tt class="py-keyword">as</tt> <tt class="py-name">PX</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"><tt class="py-comment"># For speed try to use cElementTree rather than ElementTree</tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-op">(</tt><tt class="py-number">3</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">version_info</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> <tt class="py-op">&lt;=</tt> <tt class="py-op">(</tt><tt class="py-number">3</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line">        <tt class="py-comment"># Workaround for bug in python 3.0 and 3.1,</tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line">        <tt class="py-comment"># see http://bugs.python.org/issue9257</tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt class="py-name">xml</tt><tt class="py-op">.</tt><tt class="py-name">etree</tt> <tt class="py-keyword">import</tt> <tt class="py-name">ElementTree</tt> <tt class="py-keyword">as</tt> <tt class="py-name">ElementTree</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt class="py-name">xml</tt><tt class="py-op">.</tt><tt class="py-name">etree</tt> <tt class="py-keyword">import</tt> <tt class="py-name">cElementTree</tt> <tt class="py-keyword">as</tt> <tt class="py-name">ElementTree</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-keyword">except</tt> <tt class="py-name">ImportError</tt><tt class="py-op">:</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt class="py-name">xml</tt><tt class="py-op">.</tt><tt class="py-name">etree</tt> <tt class="py-keyword">import</tt> <tt class="py-name">ElementTree</tt> <tt class="py-keyword">as</tt> <tt class="py-name">ElementTree</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"><tt class="py-comment"># Recognize the phyloXML namespace when parsing</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"><tt class="py-comment"># See http://effbot.org/zone/element-namespaces.htm</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"><tt id="link-7" class="py-name" targets="Variable Bio.Phylo.NeXMLIO.NAMESPACES=Bio.Phylo.NeXMLIO-module.html#NAMESPACES,Variable Bio.Phylo.PhyloXMLIO.NAMESPACES=Bio.Phylo.PhyloXMLIO-module.html#NAMESPACES"><a title="Bio.Phylo.NeXMLIO.NAMESPACES
Bio.Phylo.PhyloXMLIO.NAMESPACES" class="py-name" href="#" onclick="return doclink('link-7', 'NAMESPACES', 'link-7');">NAMESPACES</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line">    <tt class="py-string">"phy"</tt><tt class="py-op">:</tt> <tt class="py-string">"http://www.phyloxml.org"</tt><tt class="py-op">,</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line"><tt class="py-op">}</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line"> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line"><tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line">    <tt class="py-name">register_namespace</tt> <tt class="py-op">=</tt> <tt class="py-name">ElementTree</tt><tt class="py-op">.</tt><tt class="py-name">register_namespace</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line"><tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">ElementTree</tt><tt class="py-op">,</tt> <tt class="py-string">"_namespace_map"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line">        <tt class="py-comment"># cElementTree needs the pure-Python xml.etree.ElementTree</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt class="py-name">xml</tt><tt class="py-op">.</tt><tt class="py-name">etree</tt> <tt class="py-keyword">import</tt> <tt class="py-name">ElementTree</tt> <tt class="py-keyword">as</tt> <tt class="py-name">ET_py</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line">        <tt class="py-name">ElementTree</tt><tt class="py-op">.</tt><tt class="py-name">_namespace_map</tt> <tt class="py-op">=</tt> <tt class="py-name">ET_py</tt><tt class="py-op">.</tt><tt class="py-name">_namespace_map</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line"> </tt>
<a name="register_namespace"></a><div id="register_namespace-def"><a name="L53"></a><tt class="py-lineno"> 53</tt> <a class="py-toggle" href="#" id="register_namespace-toggle" onclick="return toggle('register_namespace');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO-module.html#register_namespace">register_namespace</a><tt class="py-op">(</tt><tt class="py-param">prefix</tt><tt class="py-op">,</tt> <tt class="py-param">uri</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="register_namespace-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="register_namespace-expanded"><a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line">        <tt class="py-docstring">"""Set the namespace for ElementTree."""</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line">        <tt class="py-name">ElementTree</tt><tt class="py-op">.</tt><tt class="py-name">_namespace_map</tt><tt class="py-op">[</tt><tt id="link-8" class="py-name" targets="Variable Bio.Phylo.NeXMLIO.uri=Bio.Phylo.NeXMLIO-module.html#uri,Method Bio.Phylo.PhyloXMLIO.Parser.uri()=Bio.Phylo.PhyloXMLIO.Parser-class.html#uri,Method Bio.Phylo.PhyloXMLIO.Writer.uri()=Bio.Phylo.PhyloXMLIO.Writer-class.html#uri,Variable Bio.Phylo.PhyloXMLIO.uri=Bio.Phylo.PhyloXMLIO-module.html#uri"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-8', 'uri', 'link-8');">uri</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-9" class="py-name" targets="Variable Bio.Phylo.NeXMLIO.prefix=Bio.Phylo.NeXMLIO-module.html#prefix,Variable Bio.Phylo.PhyloXMLIO.prefix=Bio.Phylo.PhyloXMLIO-module.html#prefix"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-9', 'prefix', 'link-9');">prefix</a></tt> </tt>
</div><a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line"> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"><tt class="py-keyword">for</tt> <tt id="link-10" class="py-name"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-10', 'prefix', 'link-9');">prefix</a></tt><tt class="py-op">,</tt> <tt id="link-11" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-11', 'uri', 'link-8');">uri</a></tt> <tt class="py-keyword">in</tt> <tt id="link-12" class="py-name"><a title="Bio.Phylo.NeXMLIO.NAMESPACES
Bio.Phylo.PhyloXMLIO.NAMESPACES" class="py-name" href="#" onclick="return doclink('link-12', 'NAMESPACES', 'link-7');">NAMESPACES</a></tt><tt class="py-op">.</tt><tt id="link-13" class="py-name" targets="Method Bio.Align.substitution_matrices.Array.items()=Bio.Align.substitution_matrices.Array-class.html#items,Method Bio.Crystal.Crystal.items()=Bio.Crystal.Crystal-class.html#items,Method Bio.File._IndexedSeqFileDict.items()=Bio.File._IndexedSeqFileDict-class.html#items,Method Bio.Phylo.PhyloXML.Events.items()=Bio.Phylo.PhyloXML.Events-class.html#items,Variable Bio.SearchIO._model.query.QueryResult.items=Bio.SearchIO._model.query.QueryResult-class.html#items,Method BioSQL.BioSeqDatabase.BioSeqDatabase.items()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#items,Method BioSQL.BioSeqDatabase.DBServer.items()=BioSQL.BioSeqDatabase.DBServer-class.html#items"><a title="Bio.Align.substitution_matrices.Array.items
Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-13', 'items', 'link-13');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line">    <tt class="py-name">register_namespace</tt><tt class="py-op">(</tt><tt id="link-14" class="py-name"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-14', 'prefix', 'link-9');">prefix</a></tt><tt class="py-op">,</tt> <tt id="link-15" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-15', 'uri', 'link-8');">uri</a></tt><tt class="py-op">)</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line"> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line"><tt class="py-comment"># Tell ElementTree how to write to text handles</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line"><tt id="link-16" class="py-name" targets="Variable Bio.Phylo.PhyloXMLIO.DEFAULT_ENCODING=Bio.Phylo.PhyloXMLIO-module.html#DEFAULT_ENCODING"><a title="Bio.Phylo.PhyloXMLIO.DEFAULT_ENCODING" class="py-name" href="#" onclick="return doclink('link-16', 'DEFAULT_ENCODING', 'link-16');">DEFAULT_ENCODING</a></tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-string">"unicode"</tt> <tt class="py-keyword">if</tt> <tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">version_info</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">3</tt> <tt class="py-keyword">else</tt> <tt class="py-string">"utf-8"</tt><tt class="py-op">)</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line"> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line"> </tt>
<a name="PhyloXMLError"></a><div id="PhyloXMLError-def"><a name="L64"></a><tt class="py-lineno"> 64</tt> <a class="py-toggle" href="#" id="PhyloXMLError-toggle" onclick="return toggle('PhyloXMLError');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.PhyloXMLError-class.html">PhyloXMLError</a><tt class="py-op">(</tt><tt class="py-base-class">Exception</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PhyloXMLError-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="PhyloXMLError-expanded"><a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line">    <tt class="py-docstring">"""Exception raised when PhyloXML object construction cannot continue.</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line"><tt class="py-docstring">    XML syntax errors will be found and raised by the underlying ElementTree</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line"><tt class="py-docstring">    module; this exception is for valid XML that breaks the phyloXML</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line"><tt class="py-docstring">    specification.</tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line"> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line">    <tt class="py-keyword">pass</tt> </tt>
</div><a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line"> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line"> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line"><tt class="py-comment"># ---------------------------------------------------------</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line"><tt class="py-comment"># Public API</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line"> </tt>
<a name="read"></a><div id="read-def"><a name="L78"></a><tt class="py-lineno"> 78</tt> <a class="py-toggle" href="#" id="read-toggle" onclick="return toggle('read');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO-module.html#read">read</a><tt class="py-op">(</tt><tt class="py-param">file</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="read-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="read-expanded"><a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line">    <tt class="py-docstring">"""Parse a phyloXML file or stream and build a tree of Biopython objects.</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line"><tt class="py-docstring">    The children of the root node are phylogenies and possibly other arbitrary</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line"><tt class="py-docstring">    (non-phyloXML) objects.</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line"><tt class="py-docstring">    :returns: a single ``Bio.Phylo.PhyloXML.Phyloxml`` object.</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-17" class="py-name" targets="Module Bio.Entrez.Parser=Bio.Entrez.Parser-module.html,Class Bio.ExPASy.ScanProsite.Parser=Bio.ExPASy.ScanProsite.Parser-class.html,Class Bio.Phylo.CDAOIO.Parser=Bio.Phylo.CDAOIO.Parser-class.html,Class Bio.Phylo.NeXMLIO.Parser=Bio.Phylo.NeXMLIO.Parser-class.html,Class Bio.Phylo.NewickIO.Parser=Bio.Phylo.NewickIO.Parser-class.html,Class Bio.Phylo.PhyloXMLIO.Parser=Bio.Phylo.PhyloXMLIO.Parser-class.html,Class Bio.SeqIO.UniprotIO.Parser=Bio.SeqIO.UniprotIO.Parser-class.html"><a title="Bio.Entrez.Parser
Bio.ExPASy.ScanProsite.Parser
Bio.Phylo.CDAOIO.Parser
Bio.Phylo.NeXMLIO.Parser
Bio.Phylo.NewickIO.Parser
Bio.Phylo.PhyloXMLIO.Parser
Bio.SeqIO.UniprotIO.Parser" class="py-name" href="#" onclick="return doclink('link-17', 'Parser', 'link-17');">Parser</a></tt><tt class="py-op">(</tt><tt class="py-name">file</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-18" class="py-name" targets="Function Bio.Affy.CelFile.read()=Bio.Affy.CelFile-module.html#read,Function Bio.Align.substitution_matrices.read()=Bio.Align.substitution_matrices-module.html#read,Function Bio.AlignIO.read()=Bio.AlignIO-module.html#read,Function Bio.Blast.NCBIXML.read()=Bio.Blast.NCBIXML-module.html#read,Function Bio.Cluster.read()=Bio.Cluster-module.html#read,Function Bio.Compass.read()=Bio.Compass-module.html#read,Function Bio.Emboss.Primer3.read()=Bio.Emboss.Primer3-module.html#read,Function Bio.Emboss.PrimerSearch.read()=Bio.Emboss.PrimerSearch-module.html#read,Method Bio.Entrez.Parser.DataHandler.read()=Bio.Entrez.Parser.DataHandler-class.html#read,Function Bio.Entrez.read()=Bio.Entrez-module.html#read,Function Bio.ExPASy.Enzyme.read()=Bio.ExPASy.Enzyme-module.html#read,Function Bio.ExPASy.Prodoc.read()=Bio.ExPASy.Prodoc-module.html#read,Function Bio.ExPASy.Prosite.read()=Bio.ExPASy.Prosite-module.html#read,Function Bio.ExPASy.ScanProsite.read()=Bio.ExPASy.ScanProsite-module.html#read,Function Bio.ExPASy.cellosaurus.read()=Bio.ExPASy.cellosaurus-module.html#read,Method Bio.File.UndoHandle.read()=Bio.File.UndoHandle-class.html#read,Function Bio.KEGG.Enzyme.read()=Bio.KEGG.Enzyme-module.html#read,Function Bio.KEGG.KGML.KGML_parser.read()=Bio.KEGG.KGML.KGML_parser-module.html#read,Function Bio.Medline.read()=Bio.Medline-module.html#read,Method Bio.Nexus.Nexus.Nexus.read()=Bio.Nexus.Nexus.Nexus-class.html#read,Function Bio.Phylo.PAML.baseml.read()=Bio.Phylo.PAML.baseml-module.html#read,Function Bio.Phylo.PAML.codeml.read()=Bio.Phylo.PAML.codeml-module.html#read,Function Bio.Phylo.PAML.yn00.read()=Bio.Phylo.PAML.yn00-module.html#read,Method Bio.Phylo.PhyloXMLIO.Parser.read()=Bio.Phylo.PhyloXMLIO.Parser-class.html#read,Function Bio.Phylo.PhyloXMLIO.read()=Bio.Phylo.PhyloXMLIO-module.html#read,Function Bio.Phylo._io.read()=Bio.Phylo._io-module.html#read,Function Bio.PopGen.GenePop.FileParser.read()=Bio.PopGen.GenePop.FileParser-module.html#read,Function Bio.PopGen.GenePop.LargeFileParser.read()=Bio.PopGen.GenePop.LargeFileParser-module.html#read,Function Bio.PopGen.GenePop.read()=Bio.PopGen.GenePop-module.html#read,Function Bio.SearchIO.read()=Bio.SearchIO-module.html#read,Method Bio.SeqIO.SffIO._AddTellHandle.read()=Bio.SeqIO.SffIO._AddTellHandle-class.html#read,Function Bio.SeqIO.read()=Bio.SeqIO-module.html#read,Function Bio.Sequencing.Ace.read()=Bio.Sequencing.Ace-module.html#read,Function Bio.Sequencing.Phd.read()=Bio.Sequencing.Phd-module.html#read,Function Bio.SwissProt.read()=Bio.SwissProt-module.html#read,Function Bio.UniGene.read()=Bio.UniGene-module.html#read,Method Bio.bgzf.BgzfReader.read()=Bio.bgzf.BgzfReader-class.html#read,Function Bio.motifs.alignace.read()=Bio.motifs.alignace-module.html#read,Function Bio.motifs.clusterbuster.read()=Bio.motifs.clusterbuster-module.html#read,Function Bio.motifs.jaspar.read()=Bio.motifs.jaspar-module.html#read,Function Bio.motifs.mast.read()=Bio.motifs.mast-module.html#read,Function Bio.motifs.meme.read()=Bio.motifs.meme-module.html#read,Function Bio.motifs.minimal.read()=Bio.motifs.minimal-module.html#read,Function Bio.motifs.pfm.read()=Bio.motifs.pfm-module.html#read,Function Bio.motifs.read()=Bio.motifs-module.html#read,Function Bio.motifs.transfac.read()=Bio.motifs.transfac-module.html#read,Function Bio.motifs.xms.read()=Bio.motifs.xms-module.html#read,Function Bio.phenotype.read()=Bio.phenotype-module.html#read"><a title="Bio.Affy.CelFile.read
Bio.Align.substitution_matrices.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.ExPASy.cellosaurus.read
Bio.File.UndoHandle.read
Bio.KEGG.Enzyme.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.clusterbuster.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.minimal.read
Bio.motifs.pfm.read
Bio.motifs.read
Bio.motifs.transfac.read
Bio.motifs.xms.read
Bio.phenotype.read" class="py-name" href="#" onclick="return doclink('link-18', 'read', 'link-18');">read</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line"> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line"> </tt>
<a name="parse"></a><div id="parse-def"><a name="L90"></a><tt class="py-lineno"> 90</tt> <a class="py-toggle" href="#" id="parse-toggle" onclick="return toggle('parse');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO-module.html#parse">parse</a><tt class="py-op">(</tt><tt class="py-param">file</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="parse-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="parse-expanded"><a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">    <tt class="py-docstring">"""Iterate over the phylogenetic trees in a phyloXML file.</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line"><tt class="py-docstring">    This ignores any additional data stored at the top level, but may be more</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line"><tt class="py-docstring">    memory-efficient than the ``read`` function.</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line"><tt class="py-docstring">    :returns: a generator of ``Bio.Phylo.PhyloXML.Phylogeny`` objects.</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-19" class="py-name"><a title="Bio.Entrez.Parser
Bio.ExPASy.ScanProsite.Parser
Bio.Phylo.CDAOIO.Parser
Bio.Phylo.NeXMLIO.Parser
Bio.Phylo.NewickIO.Parser
Bio.Phylo.PhyloXMLIO.Parser
Bio.SeqIO.UniprotIO.Parser" class="py-name" href="#" onclick="return doclink('link-19', 'Parser', 'link-17');">Parser</a></tt><tt class="py-op">(</tt><tt class="py-name">file</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-20" class="py-name" targets="Function Bio.AlignIO.parse()=Bio.AlignIO-module.html#parse,Function Bio.Blast.NCBIXML.parse()=Bio.Blast.NCBIXML-module.html#parse,Function Bio.Compass.parse()=Bio.Compass-module.html#parse,Function Bio.Emboss.Primer3.parse()=Bio.Emboss.Primer3-module.html#parse,Method Bio.Entrez.Parser.DataHandler.parse()=Bio.Entrez.Parser.DataHandler-class.html#parse,Function Bio.Entrez.parse()=Bio.Entrez-module.html#parse,Function Bio.ExPASy.Enzyme.parse()=Bio.ExPASy.Enzyme-module.html#parse,Function Bio.ExPASy.Prodoc.parse()=Bio.ExPASy.Prodoc-module.html#parse,Function Bio.ExPASy.Prosite.parse()=Bio.ExPASy.Prosite-module.html#parse,Function Bio.ExPASy.cellosaurus.parse()=Bio.ExPASy.cellosaurus-module.html#parse,Method Bio.GenBank.FeatureParser.parse()=Bio.GenBank.FeatureParser-class.html#parse,Method Bio.GenBank.RecordParser.parse()=Bio.GenBank.RecordParser-class.html#parse,Method Bio.GenBank.Scanner.InsdcScanner.parse()=Bio.GenBank.Scanner.InsdcScanner-class.html#parse,Function Bio.Geo.parse()=Bio.Geo-module.html#parse,Function Bio.KEGG.Compound.parse()=Bio.KEGG.Compound-module.html#parse,Function Bio.KEGG.Enzyme.parse()=Bio.KEGG.Enzyme-module.html#parse,Function Bio.KEGG.Gene.parse()=Bio.KEGG.Gene-module.html#parse,Method Bio.KEGG.KGML.KGML_parser.KGMLParser.parse()=Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html#parse,Function Bio.KEGG.KGML.KGML_parser.parse()=Bio.KEGG.KGML.KGML_parser-module.html#parse,Function Bio.KEGG.Map.parse()=Bio.KEGG.Map-module.html#parse,Function Bio.Medline.parse()=Bio.Medline-module.html#parse,Method Bio.Phylo.CDAOIO.Parser.parse()=Bio.Phylo.CDAOIO.Parser-class.html#parse,Function Bio.Phylo.CDAOIO.parse()=Bio.Phylo.CDAOIO-module.html#parse,Method Bio.Phylo.NeXMLIO.Parser.parse()=Bio.Phylo.NeXMLIO.Parser-class.html#parse,Function Bio.Phylo.NeXMLIO.parse()=Bio.Phylo.NeXMLIO-module.html#parse,Method Bio.Phylo.NewickIO.Parser.parse()=Bio.Phylo.NewickIO.Parser-class.html#parse,Function Bio.Phylo.NewickIO.parse()=Bio.Phylo.NewickIO-module.html#parse,Function Bio.Phylo.NexusIO.parse()=Bio.Phylo.NexusIO-module.html#parse,Method Bio.Phylo.PhyloXMLIO.Parser.parse()=Bio.Phylo.PhyloXMLIO.Parser-class.html#parse,Function Bio.Phylo.PhyloXMLIO.parse()=Bio.Phylo.PhyloXMLIO-module.html#parse,Function Bio.Phylo._io.parse()=Bio.Phylo._io-module.html#parse,Function Bio.SCOP.Cla.parse()=Bio.SCOP.Cla-module.html#parse,Function Bio.SCOP.Des.parse()=Bio.SCOP.Des-module.html#parse,Function Bio.SCOP.Dom.parse()=Bio.SCOP.Dom-module.html#parse,Function Bio.SCOP.Hie.parse()=Bio.SCOP.Hie-module.html#parse,Function Bio.SCOP.Raf.parse()=Bio.SCOP.Raf-module.html#parse,Method Bio.SearchIO._legacy.NCBIStandalone.BlastErrorParser.parse()=Bio.SearchIO._legacy.NCBIStandalone.BlastErrorParser-class.html#parse,Method Bio.SearchIO._legacy.NCBIStandalone.BlastParser.parse()=Bio.SearchIO._legacy.NCBIStandalone.BlastParser-class.html#parse,Method Bio.SearchIO._legacy.NCBIStandalone.PSIBlastParser.parse()=Bio.SearchIO._legacy.NCBIStandalone.PSIBlastParser-class.html#parse,Method Bio.SearchIO._legacy.ParserSupport.AbstractParser.parse()=Bio.SearchIO._legacy.ParserSupport.AbstractParser-class.html#parse,Function Bio.SearchIO.parse()=Bio.SearchIO-module.html#parse,Method Bio.SeqIO.UniprotIO.Parser.parse()=Bio.SeqIO.UniprotIO.Parser-class.html#parse,Function Bio.SeqIO.parse()=Bio.SeqIO-module.html#parse,Function Bio.Sequencing.Ace.parse()=Bio.Sequencing.Ace-module.html#parse,Function Bio.Sequencing.Phd.parse()=Bio.Sequencing.Phd-module.html#parse,Function Bio.SwissProt.KeyWList.parse()=Bio.SwissProt.KeyWList-module.html#parse,Function Bio.SwissProt.parse()=Bio.SwissProt-module.html#parse,Function Bio.UniGene.parse()=Bio.UniGene-module.html#parse,Function Bio.Wise.psw.parse()=Bio.Wise.psw-module.html#parse,Function Bio.motifs.parse()=Bio.motifs-module.html#parse,Function Bio.phenotype.parse()=Bio.phenotype-module.html#parse"><a title="Bio.AlignIO.parse
Bio.Blast.NCBIXML.parse
Bio.Compass.parse
Bio.Emboss.Primer3.parse
Bio.Entrez.Parser.DataHandler.parse
Bio.Entrez.parse
Bio.ExPASy.Enzyme.parse
Bio.ExPASy.Prodoc.parse
Bio.ExPASy.Prosite.parse
Bio.ExPASy.cellosaurus.parse
Bio.GenBank.FeatureParser.parse
Bio.GenBank.RecordParser.parse
Bio.GenBank.Scanner.InsdcScanner.parse
Bio.Geo.parse
Bio.KEGG.Compound.parse
Bio.KEGG.Enzyme.parse
Bio.KEGG.Gene.parse
Bio.KEGG.KGML.KGML_parser.KGMLParser.parse
Bio.KEGG.KGML.KGML_parser.parse
Bio.KEGG.Map.parse
Bio.Medline.parse
Bio.Phylo.CDAOIO.Parser.parse
Bio.Phylo.CDAOIO.parse
Bio.Phylo.NeXMLIO.Parser.parse
Bio.Phylo.NeXMLIO.parse
Bio.Phylo.NewickIO.Parser.parse
Bio.Phylo.NewickIO.parse
Bio.Phylo.NexusIO.parse
Bio.Phylo.PhyloXMLIO.Parser.parse
Bio.Phylo.PhyloXMLIO.parse
Bio.Phylo._io.parse
Bio.SCOP.Cla.parse
Bio.SCOP.Des.parse
Bio.SCOP.Dom.parse
Bio.SCOP.Hie.parse
Bio.SCOP.Raf.parse
Bio.SearchIO._legacy.NCBIStandalone.BlastErrorParser.parse
Bio.SearchIO._legacy.NCBIStandalone.BlastParser.parse
Bio.SearchIO._legacy.NCBIStandalone.PSIBlastParser.parse
Bio.SearchIO._legacy.ParserSupport.AbstractParser.parse
Bio.SearchIO.parse
Bio.SeqIO.UniprotIO.Parser.parse
Bio.SeqIO.parse
Bio.Sequencing.Ace.parse
Bio.Sequencing.Phd.parse
Bio.SwissProt.KeyWList.parse
Bio.SwissProt.parse
Bio.UniGene.parse
Bio.Wise.psw.parse
Bio.motifs.parse
Bio.phenotype.parse" class="py-name" href="#" onclick="return doclink('link-20', 'parse', 'link-20');">parse</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line"> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line"> </tt>
<a name="write"></a><div id="write-def"><a name="L102"></a><tt class="py-lineno">102</tt> <a class="py-toggle" href="#" id="write-toggle" onclick="return toggle('write');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO-module.html#write">write</a><tt class="py-op">(</tt><tt class="py-param">obj</tt><tt class="py-op">,</tt> <tt class="py-param">file</tt><tt class="py-op">,</tt> <tt class="py-param">encoding</tt><tt class="py-op">=</tt><tt id="link-21" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.DEFAULT_ENCODING" class="py-name" href="#" onclick="return doclink('link-21', 'DEFAULT_ENCODING', 'link-16');">DEFAULT_ENCODING</a></tt><tt class="py-op">,</tt> <tt class="py-param">indent</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="write-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="write-expanded"><a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">    <tt class="py-docstring">"""Write a phyloXML file.</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line"><tt class="py-docstring">    :Parameters:</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line"><tt class="py-docstring">        obj</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line"><tt class="py-docstring">            an instance of ``Phyloxml``, ``Phylogeny`` or ``BaseTree.Tree``,</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line"><tt class="py-docstring">            or an iterable of either of the latter two. The object will be</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line"><tt class="py-docstring">            converted to a Phyloxml object before serialization.</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line"><tt class="py-docstring">        file</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line"><tt class="py-docstring">            either an open handle or a file name.</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">    <tt class="py-keyword">def</tt> <tt class="py-def-name">fix_single</tt><tt class="py-op">(</tt><tt class="py-param">tree</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">tree</tt><tt class="py-op">,</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-22" class="py-name" targets="Class Bio.Phylo.PhyloXML.Phylogeny=Bio.Phylo.PhyloXML.Phylogeny-class.html"><a title="Bio.Phylo.PhyloXML.Phylogeny" class="py-name" href="#" onclick="return doclink('link-22', 'Phylogeny', 'link-22');">Phylogeny</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">tree</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">tree</tt><tt class="py-op">,</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-23" class="py-name" targets="Class Bio.Phylo.BaseTree.Clade=Bio.Phylo.BaseTree.Clade-class.html,Class Bio.Phylo.CDAO.Clade=Bio.Phylo.CDAO.Clade-class.html,Class Bio.Phylo.NeXML.Clade=Bio.Phylo.NeXML.Clade-class.html,Class Bio.Phylo.Newick.Clade=Bio.Phylo.Newick.Clade-class.html,Class Bio.Phylo.PhyloXML.Clade=Bio.Phylo.PhyloXML.Clade-class.html"><a title="Bio.Phylo.BaseTree.Clade
Bio.Phylo.CDAO.Clade
Bio.Phylo.NeXML.Clade
Bio.Phylo.Newick.Clade
Bio.Phylo.PhyloXML.Clade" class="py-name" href="#" onclick="return doclink('link-23', 'Clade', 'link-23');">Clade</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">tree</tt><tt class="py-op">.</tt><tt id="link-24" class="py-name" targets="Method Bio.Phylo.PhyloXML.Clade.to_phylogeny()=Bio.Phylo.PhyloXML.Clade-class.html#to_phylogeny"><a title="Bio.Phylo.PhyloXML.Clade.to_phylogeny" class="py-name" href="#" onclick="return doclink('link-24', 'to_phylogeny', 'link-24');">to_phylogeny</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">tree</tt><tt class="py-op">,</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-25" class="py-name" targets="Module Bio.Phylo.BaseTree=Bio.Phylo.BaseTree-module.html"><a title="Bio.Phylo.BaseTree" class="py-name" href="#" onclick="return doclink('link-25', 'BaseTree', 'link-25');">BaseTree</a></tt><tt class="py-op">.</tt><tt id="link-26" class="py-name" targets="Class Bio.Cluster.Tree=Bio.Cluster.Tree-class.html,Class Bio.Nexus.Trees.Tree=Bio.Nexus.Trees.Tree-class.html,Class Bio.Phylo.BaseTree.Tree=Bio.Phylo.BaseTree.Tree-class.html,Class Bio.Phylo.CDAO.Tree=Bio.Phylo.CDAO.Tree-class.html,Class Bio.Phylo.NeXML.Tree=Bio.Phylo.NeXML.Tree-class.html,Class Bio.Phylo.Newick.Tree=Bio.Phylo.Newick.Tree-class.html"><a title="Bio.Cluster.Tree
Bio.Nexus.Trees.Tree
Bio.Phylo.BaseTree.Tree
Bio.Phylo.CDAO.Tree
Bio.Phylo.NeXML.Tree
Bio.Phylo.Newick.Tree" class="py-name" href="#" onclick="return doclink('link-26', 'Tree', 'link-26');">Tree</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-27" class="py-name"><a title="Bio.Phylo.PhyloXML.Phylogeny" class="py-name" href="#" onclick="return doclink('link-27', 'Phylogeny', 'link-22');">Phylogeny</a></tt><tt class="py-op">.</tt><tt id="link-28" class="py-name" targets="Class Method Bio.Phylo.PhyloXML.Phylogeny.from_tree()=Bio.Phylo.PhyloXML.Phylogeny-class.html#from_tree"><a title="Bio.Phylo.PhyloXML.Phylogeny.from_tree" class="py-name" href="#" onclick="return doclink('link-28', 'from_tree', 'link-28');">from_tree</a></tt><tt class="py-op">(</tt><tt class="py-name">tree</tt><tt class="py-op">)</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">tree</tt><tt class="py-op">,</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-29" class="py-name"><a title="Bio.Phylo.BaseTree" class="py-name" href="#" onclick="return doclink('link-29', 'BaseTree', 'link-25');">BaseTree</a></tt><tt class="py-op">.</tt><tt id="link-30" class="py-name"><a title="Bio.Phylo.BaseTree.Clade
Bio.Phylo.CDAO.Clade
Bio.Phylo.NeXML.Clade
Bio.Phylo.Newick.Clade
Bio.Phylo.PhyloXML.Clade" class="py-name" href="#" onclick="return doclink('link-30', 'Clade', 'link-23');">Clade</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-31" class="py-name"><a title="Bio.Phylo.PhyloXML.Phylogeny" class="py-name" href="#" onclick="return doclink('link-31', 'Phylogeny', 'link-22');">Phylogeny</a></tt><tt class="py-op">.</tt><tt id="link-32" class="py-name"><a title="Bio.Phylo.PhyloXML.Phylogeny.from_tree" class="py-name" href="#" onclick="return doclink('link-32', 'from_tree', 'link-28');">from_tree</a></tt><tt class="py-op">(</tt><tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-33" class="py-name"><a title="Bio.Phylo.BaseTree" class="py-name" href="#" onclick="return doclink('link-33', 'BaseTree', 'link-25');">BaseTree</a></tt><tt class="py-op">.</tt><tt id="link-34" class="py-name"><a title="Bio.Cluster.Tree
Bio.Nexus.Trees.Tree
Bio.Phylo.BaseTree.Tree
Bio.Phylo.CDAO.Tree
Bio.Phylo.NeXML.Tree
Bio.Phylo.Newick.Tree" class="py-name" href="#" onclick="return doclink('link-34', 'Tree', 'link-26');">Tree</a></tt><tt class="py-op">(</tt><tt id="link-35" class="py-name" targets="Variable Bio.Phylo.BaseTree.Clade.root=Bio.Phylo.BaseTree.Clade-class.html#root,Variable Bio.Phylo._cdao_owl.root=Bio.Phylo._cdao_owl-module.html#root"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-35', 'root', 'link-35');">root</a></tt><tt class="py-op">=</tt><tt class="py-name">tree</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"iterable must contain Tree or Clade types"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line"> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">,</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-36" class="py-name" targets="Class Bio.Phylo.PhyloXML.Phyloxml=Bio.Phylo.PhyloXML.Phyloxml-class.html"><a title="Bio.Phylo.PhyloXML.Phyloxml" class="py-name" href="#" onclick="return doclink('link-36', 'Phyloxml', 'link-36');">Phyloxml</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-37" class="py-name"><a title="Bio.Phylo.BaseTree" class="py-name" href="#" onclick="return doclink('link-37', 'BaseTree', 'link-25');">BaseTree</a></tt><tt class="py-op">.</tt><tt id="link-38" class="py-name"><a title="Bio.Cluster.Tree
Bio.Nexus.Trees.Tree
Bio.Phylo.BaseTree.Tree
Bio.Phylo.CDAO.Tree
Bio.Phylo.NeXML.Tree
Bio.Phylo.Newick.Tree" class="py-name" href="#" onclick="return doclink('link-38', 'Tree', 'link-26');">Tree</a></tt><tt class="py-op">,</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-39" class="py-name"><a title="Bio.Phylo.BaseTree" class="py-name" href="#" onclick="return doclink('link-39', 'BaseTree', 'link-25');">BaseTree</a></tt><tt class="py-op">.</tt><tt id="link-40" class="py-name"><a title="Bio.Phylo.BaseTree.Clade
Bio.Phylo.CDAO.Clade
Bio.Phylo.NeXML.Clade
Bio.Phylo.Newick.Clade
Bio.Phylo.PhyloXML.Clade" class="py-name" href="#" onclick="return doclink('link-40', 'Clade', 'link-23');">Clade</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line">        <tt class="py-name">obj</tt> <tt class="py-op">=</tt> <tt class="py-name">fix_single</tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">to_phyloxml</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">,</tt> <tt class="py-string">"__iter__"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line">        <tt class="py-name">obj</tt> <tt class="py-op">=</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-41" class="py-name"><a title="Bio.Phylo.PhyloXML.Phyloxml" class="py-name" href="#" onclick="return doclink('link-41', 'Phyloxml', 'link-36');">Phyloxml</a></tt><tt class="py-op">(</tt><tt class="py-op">{</tt><tt class="py-op">}</tt><tt class="py-op">,</tt> <tt class="py-name">phylogenies</tt><tt class="py-op">=</tt><tt class="py-op">(</tt><tt class="py-name">fix_single</tt><tt class="py-op">(</tt><tt class="py-name">t</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">t</tt> <tt class="py-keyword">in</tt> <tt class="py-name">obj</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"First argument must be a Phyloxml, Phylogeny, "</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line">                         <tt class="py-string">"Tree, or iterable of Trees or Phylogenies."</tt><tt class="py-op">)</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-42" class="py-name" targets="Class Bio.Phylo.CDAOIO.Writer=Bio.Phylo.CDAOIO.Writer-class.html,Class Bio.Phylo.NeXMLIO.Writer=Bio.Phylo.NeXMLIO.Writer-class.html,Class Bio.Phylo.NewickIO.Writer=Bio.Phylo.NewickIO.Writer-class.html,Class Bio.Phylo.PhyloXMLIO.Writer=Bio.Phylo.PhyloXMLIO.Writer-class.html"><a title="Bio.Phylo.CDAOIO.Writer
Bio.Phylo.NeXMLIO.Writer
Bio.Phylo.NewickIO.Writer
Bio.Phylo.PhyloXMLIO.Writer" class="py-name" href="#" onclick="return doclink('link-42', 'Writer', 'link-42');">Writer</a></tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-43" class="py-name" targets="Function Bio.AlignIO.write()=Bio.AlignIO-module.html#write,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.write()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#write,Method Bio.Phylo.CDAOIO.Writer.write()=Bio.Phylo.CDAOIO.Writer-class.html#write,Function Bio.Phylo.CDAOIO.write()=Bio.Phylo.CDAOIO-module.html#write,Method Bio.Phylo.NeXMLIO.Writer.write()=Bio.Phylo.NeXMLIO.Writer-class.html#write,Function Bio.Phylo.NeXMLIO.write()=Bio.Phylo.NeXMLIO-module.html#write,Method Bio.Phylo.NewickIO.Writer.write()=Bio.Phylo.NewickIO.Writer-class.html#write,Function Bio.Phylo.NewickIO.write()=Bio.Phylo.NewickIO-module.html#write,Function Bio.Phylo.NexusIO.write()=Bio.Phylo.NexusIO-module.html#write,Method Bio.Phylo.PhyloXMLIO.Writer.write()=Bio.Phylo.PhyloXMLIO.Writer-class.html#write,Function Bio.Phylo.PhyloXMLIO.write()=Bio.Phylo.PhyloXMLIO-module.html#write,Function Bio.Phylo._io.write()=Bio.Phylo._io-module.html#write,Function Bio.SearchIO.write()=Bio.SearchIO-module.html#write,Function Bio.SeqIO.write()=Bio.SeqIO-module.html#write,Method Bio.bgzf.BgzfWriter.write()=Bio.bgzf.BgzfWriter-class.html#write,Function Bio.motifs.clusterbuster.write()=Bio.motifs.clusterbuster-module.html#write,Function Bio.motifs.jaspar.write()=Bio.motifs.jaspar-module.html#write,Function Bio.motifs.pfm.write()=Bio.motifs.pfm-module.html#write,Function Bio.motifs.transfac.write()=Bio.motifs.transfac-module.html#write,Function Bio.motifs.write()=Bio.motifs-module.html#write,Method Bio.phenotype.phen_micro.JsonWriter.write()=Bio.phenotype.phen_micro.JsonWriter-class.html#write,Function Bio.phenotype.write()=Bio.phenotype-module.html#write"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.clusterbuster.write
Bio.motifs.jaspar.write
Bio.motifs.pfm.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-43', 'write', 'link-43');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">file</tt><tt class="py-op">,</tt> <tt class="py-name">encoding</tt><tt class="py-op">=</tt><tt class="py-name">encoding</tt><tt class="py-op">,</tt> <tt class="py-name">indent</tt><tt class="py-op">=</tt><tt class="py-name">indent</tt><tt class="py-op">)</tt> </tt>
</div><a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line"> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line"> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line"><tt class="py-comment"># ---------------------------------------------------------</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line"><tt class="py-comment"># Functions I wish ElementTree had</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line"> </tt>
<a name="_local"></a><div id="_local-def"><a name="L141"></a><tt class="py-lineno">141</tt> <a class="py-toggle" href="#" id="_local-toggle" onclick="return toggle('_local');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO-module.html#_local">_local</a><tt class="py-op">(</tt><tt class="py-param">tag</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_local-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_local-expanded"><a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line">    <tt class="py-docstring">"""Extract the local tag from a namespaced tag name (PRIVATE)."""</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt id="link-44" class="py-name" targets="Variable Bio.SeqIO.AbiIO.tag=Bio.SeqIO.AbiIO-module.html#tag"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-44', 'tag', 'link-44');">tag</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"{"</tt><tt class="py-op">:</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-45" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-45', 'tag', 'link-44');">tag</a></tt><tt class="py-op">[</tt><tt id="link-46" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-46', 'tag', 'link-44');">tag</a></tt><tt class="py-op">.</tt><tt id="link-47" class="py-name" targets="Method Bio.Crystal.Chain.index()=Bio.Crystal.Chain-class.html#index,Method Bio.SCOP.Raf.SeqMap.index()=Bio.SCOP.Raf.SeqMap-class.html#index,Method Bio.SearchIO._model.hit.Hit.index()=Bio.SearchIO._model.hit.Hit-class.html#index,Method Bio.SearchIO._model.query.QueryResult.index()=Bio.SearchIO._model.query.QueryResult-class.html#index,Function Bio.SearchIO.index()=Bio.SearchIO-module.html#index,Method Bio.Seq.MutableSeq.index()=Bio.Seq.MutableSeq-class.html#index,Method Bio.Seq.Seq.index()=Bio.Seq.Seq-class.html#index,Function Bio.SeqIO.index()=Bio.SeqIO-module.html#index"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.Seq.Seq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-47', 'index', 'link-47');">index</a></tt><tt class="py-op">(</tt><tt class="py-string">"}"</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-48" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-48', 'tag', 'link-44');">tag</a></tt> </tt>
</div><a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line"> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line"> </tt>
<a name="_split_namespace"></a><div id="_split_namespace-def"><a name="L148"></a><tt class="py-lineno">148</tt> <a class="py-toggle" href="#" id="_split_namespace-toggle" onclick="return toggle('_split_namespace');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO-module.html#_split_namespace">_split_namespace</a><tt class="py-op">(</tt><tt class="py-param">tag</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_split_namespace-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_split_namespace-expanded"><a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line">    <tt class="py-docstring">"""Split a tag into namespace and local tag strings (PRIVATE)."""</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-49" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-49', 'tag', 'link-44');">tag</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-50" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-50', 'split', 'link-50');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"}"</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">,</tt> <tt id="link-51" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-51', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line"> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line"> </tt>
<a name="_ns"></a><div id="_ns-def"><a name="L156"></a><tt class="py-lineno">156</tt> <a class="py-toggle" href="#" id="_ns-toggle" onclick="return toggle('_ns');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO-module.html#_ns">_ns</a><tt class="py-op">(</tt><tt class="py-param">tag</tt><tt class="py-op">,</tt> <tt class="py-param">namespace</tt><tt class="py-op">=</tt><tt id="link-52" class="py-name"><a title="Bio.Phylo.NeXMLIO.NAMESPACES
Bio.Phylo.PhyloXMLIO.NAMESPACES" class="py-name" href="#" onclick="return doclink('link-52', 'NAMESPACES', 'link-7');">NAMESPACES</a></tt><tt class="py-op">[</tt><tt class="py-string">"phy"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ns-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_ns-expanded"><a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">    <tt class="py-docstring">"""Format an XML tag with the given namespace (PRIVATE)."""</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-string">"{%s}%s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">namespace</tt><tt class="py-op">,</tt> <tt id="link-53" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-53', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line"> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line"> </tt>
<a name="_get_child_as"></a><div id="_get_child_as-def"><a name="L161"></a><tt class="py-lineno">161</tt> <a class="py-toggle" href="#" id="_get_child_as-toggle" onclick="return toggle('_get_child_as');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO-module.html#_get_child_as">_get_child_as</a><tt class="py-op">(</tt><tt class="py-param">parent</tt><tt class="py-op">,</tt> <tt class="py-param">tag</tt><tt class="py-op">,</tt> <tt class="py-param">construct</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_get_child_as-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_get_child_as-expanded"><a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">    <tt class="py-docstring">"""Find a child node by tag, and pass it through a constructor (PRIVATE).</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line"><tt class="py-docstring">    Returns None if no matching child is found.</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">    <tt class="py-name">child</tt> <tt class="py-op">=</tt> <tt class="py-name">parent</tt><tt class="py-op">.</tt><tt id="link-54" class="py-name" targets="Method Bio.Seq.Seq.find()=Bio.Seq.Seq-class.html#find,Function Bio.triefind.find()=Bio.triefind-module.html#find"><a title="Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-54', 'find', 'link-54');">find</a></tt><tt class="py-op">(</tt><tt id="link-55" class="py-name" targets="Function Bio.Phylo.PhyloXMLIO._ns()=Bio.Phylo.PhyloXMLIO-module.html#_ns"><a title="Bio.Phylo.PhyloXMLIO._ns" class="py-name" href="#" onclick="return doclink('link-55', '_ns', 'link-55');">_ns</a></tt><tt class="py-op">(</tt><tt id="link-56" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-56', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">child</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">construct</tt><tt class="py-op">(</tt><tt class="py-name">child</tt><tt class="py-op">)</tt> </tt>
</div><a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line"> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line"> </tt>
<a name="_get_child_text"></a><div id="_get_child_text-def"><a name="L171"></a><tt class="py-lineno">171</tt> <a class="py-toggle" href="#" id="_get_child_text-toggle" onclick="return toggle('_get_child_text');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO-module.html#_get_child_text">_get_child_text</a><tt class="py-op">(</tt><tt class="py-param">parent</tt><tt class="py-op">,</tt> <tt class="py-param">tag</tt><tt class="py-op">,</tt> <tt class="py-param">construct</tt><tt class="py-op">=</tt><tt class="py-name">unicode</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_get_child_text-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_get_child_text-expanded"><a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line">    <tt class="py-docstring">"""Find a child node by tag; pass its text through a constructor (PRIVATE).</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line"><tt class="py-docstring">    Returns None if no matching child is found.</tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">    <tt class="py-name">child</tt> <tt class="py-op">=</tt> <tt class="py-name">parent</tt><tt class="py-op">.</tt><tt id="link-57" class="py-name"><a title="Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-57', 'find', 'link-54');">find</a></tt><tt class="py-op">(</tt><tt id="link-58" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._ns" class="py-name" href="#" onclick="return doclink('link-58', '_ns', 'link-55');">_ns</a></tt><tt class="py-op">(</tt><tt id="link-59" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-59', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">child</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> <tt class="py-name">child</tt><tt class="py-op">.</tt><tt class="py-name">text</tt><tt class="py-op">:</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">construct</tt><tt class="py-op">(</tt><tt class="py-name">child</tt><tt class="py-op">.</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
</div><a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line"> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line"> </tt>
<a name="_get_children_as"></a><div id="_get_children_as-def"><a name="L181"></a><tt class="py-lineno">181</tt> <a class="py-toggle" href="#" id="_get_children_as-toggle" onclick="return toggle('_get_children_as');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO-module.html#_get_children_as">_get_children_as</a><tt class="py-op">(</tt><tt class="py-param">parent</tt><tt class="py-op">,</tt> <tt class="py-param">tag</tt><tt class="py-op">,</tt> <tt class="py-param">construct</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_get_children_as-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_get_children_as-expanded"><a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">    <tt class="py-docstring">"""Find child nodes by tag; pass each through a constructor (PRIVATE).</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line"><tt class="py-docstring">    Returns an empty list if no matching child is found.</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">construct</tt><tt class="py-op">(</tt><tt class="py-name">child</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">child</tt> <tt class="py-keyword">in</tt> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">            <tt class="py-name">parent</tt><tt class="py-op">.</tt><tt class="py-name">findall</tt><tt class="py-op">(</tt><tt id="link-60" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._ns" class="py-name" href="#" onclick="return doclink('link-60', '_ns', 'link-55');">_ns</a></tt><tt class="py-op">(</tt><tt id="link-61" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-61', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
</div><a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line"> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line"> </tt>
<a name="_get_children_text"></a><div id="_get_children_text-def"><a name="L190"></a><tt class="py-lineno">190</tt> <a class="py-toggle" href="#" id="_get_children_text-toggle" onclick="return toggle('_get_children_text');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO-module.html#_get_children_text">_get_children_text</a><tt class="py-op">(</tt><tt class="py-param">parent</tt><tt class="py-op">,</tt> <tt class="py-param">tag</tt><tt class="py-op">,</tt> <tt class="py-param">construct</tt><tt class="py-op">=</tt><tt class="py-name">unicode</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_get_children_text-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_get_children_text-expanded"><a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">    <tt class="py-docstring">"""Find child nodes by tag; pass each node's text through a constructor (PRIVATE).</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line"><tt class="py-docstring">    Returns an empty list if no matching child is found.</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">construct</tt><tt class="py-op">(</tt><tt class="py-name">child</tt><tt class="py-op">.</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">child</tt> <tt class="py-keyword">in</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">            <tt class="py-name">parent</tt><tt class="py-op">.</tt><tt class="py-name">findall</tt><tt class="py-op">(</tt><tt id="link-62" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._ns" class="py-name" href="#" onclick="return doclink('link-62', '_ns', 'link-55');">_ns</a></tt><tt class="py-op">(</tt><tt id="link-63" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-63', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">child</tt><tt class="py-op">.</tt><tt class="py-name">text</tt><tt class="py-op">]</tt> </tt>
</div><a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line"> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line"> </tt>
<a name="_indent"></a><div id="_indent-def"><a name="L200"></a><tt class="py-lineno">200</tt> <a class="py-toggle" href="#" id="_indent-toggle" onclick="return toggle('_indent');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO-module.html#_indent">_indent</a><tt class="py-op">(</tt><tt class="py-param">elem</tt><tt class="py-op">,</tt> <tt class="py-param">level</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_indent-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_indent-expanded"><a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line">    <tt class="py-docstring">"""Add line breaks and indentation to ElementTree in-place (PRIVATE).</tt> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line"><tt class="py-docstring">    Sources:</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line"><tt class="py-docstring">     - http://effbot.org/zone/element-lib.htm#prettyprint</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line"><tt class="py-docstring">     - http://infix.se/2007/02/06/gentlemen-indent-your-xml</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line">    <tt id="link-64" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-64', 'i', 'link-64');">i</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"\n"</tt> <tt class="py-op">+</tt> <tt class="py-name">level</tt> <tt class="py-op">*</tt> <tt class="py-string">"  "</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt> <tt class="py-keyword">or</tt> <tt class="py-keyword">not</tt> <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt><tt class="py-op">.</tt><tt id="link-65" class="py-name" targets="Method Bio.Seq.Seq.strip()=Bio.Seq.Seq-class.html#strip"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-65', 'strip', 'link-65');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line">            <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt id="link-66" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-66', 'i', 'link-64');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-string">"  "</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">e</tt> <tt class="py-keyword">in</tt> <tt class="py-name">elem</tt><tt class="py-op">:</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line">            <tt id="link-67" class="py-name" targets="Function Bio.Phylo.PhyloXMLIO._indent()=Bio.Phylo.PhyloXMLIO-module.html#_indent"><a title="Bio.Phylo.PhyloXMLIO._indent" class="py-name" href="#" onclick="return doclink('link-67', '_indent', 'link-67');">_indent</a></tt><tt class="py-op">(</tt><tt class="py-name">e</tt><tt class="py-op">,</tt> <tt class="py-name">level</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">e</tt><tt class="py-op">.</tt><tt class="py-name">tail</tt> <tt class="py-keyword">or</tt> <tt class="py-keyword">not</tt> <tt class="py-name">e</tt><tt class="py-op">.</tt><tt class="py-name">tail</tt><tt class="py-op">.</tt><tt id="link-68" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-68', 'strip', 'link-65');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">                <tt class="py-name">e</tt><tt class="py-op">.</tt><tt class="py-name">tail</tt> <tt class="py-op">=</tt> <tt id="link-69" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-69', 'i', 'link-64');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-string">"  "</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">e</tt><tt class="py-op">.</tt><tt class="py-name">tail</tt> <tt class="py-keyword">or</tt> <tt class="py-keyword">not</tt> <tt class="py-name">e</tt><tt class="py-op">.</tt><tt class="py-name">tail</tt><tt class="py-op">.</tt><tt id="link-70" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-70', 'strip', 'link-65');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line">            <tt class="py-name">e</tt><tt class="py-op">.</tt><tt class="py-name">tail</tt> <tt class="py-op">=</tt> <tt id="link-71" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-71', 'i', 'link-64');">i</a></tt> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">level</tt> <tt class="py-keyword">and</tt> <tt class="py-op">(</tt><tt class="py-keyword">not</tt> <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">tail</tt> <tt class="py-keyword">or</tt> <tt class="py-keyword">not</tt> <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">tail</tt><tt class="py-op">.</tt><tt id="link-72" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-72', 'strip', 'link-65');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">            <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">tail</tt> <tt class="py-op">=</tt> <tt id="link-73" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-73', 'i', 'link-64');">i</a></tt> </tt>
</div><a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line"> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line"><tt class="py-comment"># ---------------------------------------------------------</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line"><tt class="py-comment"># INPUT</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line"><tt class="py-comment"># ---------------------------------------------------------</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line"> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line"> </tt>
<a name="_str2bool"></a><div id="_str2bool-def"><a name="L227"></a><tt class="py-lineno">227</tt> <a class="py-toggle" href="#" id="_str2bool-toggle" onclick="return toggle('_str2bool');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO-module.html#_str2bool">_str2bool</a><tt class="py-op">(</tt><tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_str2bool-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_str2bool-expanded"><a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line">    <tt class="py-docstring">"""Convert string to boolean (PRIVATE)."""</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">text</tt> <tt class="py-op">==</tt> <tt class="py-string">"true"</tt> <tt class="py-keyword">or</tt> <tt class="py-name">text</tt> <tt class="py-op">==</tt> <tt class="py-string">"1"</tt><tt class="py-op">:</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">True</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">text</tt> <tt class="py-op">==</tt> <tt class="py-string">"false"</tt> <tt class="py-keyword">or</tt> <tt class="py-name">text</tt> <tt class="py-op">==</tt> <tt class="py-string">"0"</tt><tt class="py-op">:</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">    <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"String could not be converted to boolean: "</tt> <tt class="py-op">+</tt> <tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
</div><a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line"> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line"> </tt>
<a name="_dict_str2bool"></a><div id="_dict_str2bool-def"><a name="L236"></a><tt class="py-lineno">236</tt> <a class="py-toggle" href="#" id="_dict_str2bool-toggle" onclick="return toggle('_dict_str2bool');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO-module.html#_dict_str2bool">_dict_str2bool</a><tt class="py-op">(</tt><tt class="py-param">dct</tt><tt class="py-op">,</tt> <tt class="py-param">keys</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_dict_str2bool-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_dict_str2bool-expanded"><a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return a new dictionary where string values are replaced with booleans (PRIVATE)."""</tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line">    <tt class="py-name">out</tt> <tt class="py-op">=</tt> <tt class="py-name">dct</tt><tt class="py-op">.</tt><tt id="link-74" class="py-name" targets="Method Bio.Align.substitution_matrices.Array.copy()=Bio.Align.substitution_matrices.Array-class.html#copy,Method Bio.Crystal.Crystal.copy()=Bio.Crystal.Crystal-class.html#copy,Method Bio.File._IndexedSeqFileDict.copy()=Bio.File._IndexedSeqFileDict-class.html#copy,Method Bio.PDB.Atom.Atom.copy()=Bio.PDB.Atom.Atom-class.html#copy,Method Bio.PDB.Entity.Entity.copy()=Bio.PDB.Entity.Entity-class.html#copy,Method Bio.PDB.vectors.Vector.copy()=Bio.PDB.vectors.Vector-class.html#copy"><a title="Bio.Align.substitution_matrices.Array.copy
Bio.Crystal.Crystal.copy
Bio.File._IndexedSeqFileDict.copy
Bio.PDB.Atom.Atom.copy
Bio.PDB.Entity.Entity.copy
Bio.PDB.vectors.Vector.copy" class="py-name" href="#" onclick="return doclink('link-74', 'copy', 'link-74');">copy</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt id="link-75" class="py-name" targets="Method Bio.Align.substitution_matrices.Array.keys()=Bio.Align.substitution_matrices.Array-class.html#keys,Method Bio.Crystal.Crystal.keys()=Bio.Crystal.Crystal-class.html#keys,Method Bio.File._IndexedSeqFileDict.keys()=Bio.File._IndexedSeqFileDict-class.html#keys,Method Bio.File._SQLiteManySeqFilesDict.keys()=Bio.File._SQLiteManySeqFilesDict-class.html#keys,Method Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys()=Bio.PDB.AbstractPropertyMap.AbstractPropertyMap-class.html#keys,Method Bio.Phylo.PhyloXML.Events.keys()=Bio.Phylo.PhyloXML.Events-class.html#keys,Method BioSQL.BioSeqDatabase.BioSeqDatabase.keys()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#keys,Method BioSQL.BioSeqDatabase.DBServer.keys()=BioSQL.BioSeqDatabase.DBServer-class.html#keys"><a title="Bio.Align.substitution_matrices.Array.keys
Bio.Crystal.Crystal.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-75', 'keys', 'link-75');">keys</a></tt><tt class="py-op">:</tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">out</tt><tt class="py-op">:</tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line">            <tt class="py-name">out</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-76" class="py-name" targets="Function Bio.Phylo.PhyloXMLIO._str2bool()=Bio.Phylo.PhyloXMLIO-module.html#_str2bool"><a title="Bio.Phylo.PhyloXMLIO._str2bool" class="py-name" href="#" onclick="return doclink('link-76', '_str2bool', 'link-76');">_str2bool</a></tt><tt class="py-op">(</tt><tt class="py-name">out</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">out</tt> </tt>
</div><a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line"> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line"> </tt>
<a name="_int"></a><div id="_int-def"><a name="L245"></a><tt class="py-lineno">245</tt> <a class="py-toggle" href="#" id="_int-toggle" onclick="return toggle('_int');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO-module.html#_int">_int</a><tt class="py-op">(</tt><tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_int-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_int-expanded"><a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return text as an integer (PRIVATE)."""</tt> </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">text</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">Exception</tt><tt class="py-op">:</tt> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line"> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line"> </tt>
<a name="_float"></a><div id="_float-def"><a name="L254"></a><tt class="py-lineno">254</tt> <a class="py-toggle" href="#" id="_float-toggle" onclick="return toggle('_float');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO-module.html#_float">_float</a><tt class="py-op">(</tt><tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_float-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_float-expanded"><a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return text as a float (PRIVATE)."""</tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">text</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">Exception</tt><tt class="py-op">:</tt> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line"> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line"> </tt>
<a name="_collapse_wspace"></a><div id="_collapse_wspace-def"><a name="L263"></a><tt class="py-lineno">263</tt> <a class="py-toggle" href="#" id="_collapse_wspace-toggle" onclick="return toggle('_collapse_wspace');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO-module.html#_collapse_wspace">_collapse_wspace</a><tt class="py-op">(</tt><tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_collapse_wspace-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_collapse_wspace-expanded"><a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line">    <tt class="py-docstring">"""Replace all spans of whitespace with a single space character (PRIVATE).</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line"><tt class="py-docstring">    Also remove leading and trailing whitespace. See "Collapse Whitespace</tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line"><tt class="py-docstring">    Policy" in the phyloXML spec glossary:</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line"><tt class="py-docstring">    http://phyloxml.org/documentation/version_100/phyloxml.xsd.html#Glossary</tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">text</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">" "</tt><tt class="py-op">.</tt><tt id="link-77" class="py-name" targets="Method Bio.Seq.MutableSeq.join()=Bio.Seq.MutableSeq-class.html#join,Method Bio.Seq.Seq.join()=Bio.Seq.Seq-class.html#join,Method Bio.Seq.UnknownSeq.join()=Bio.Seq.UnknownSeq-class.html#join"><a title="Bio.Seq.MutableSeq.join
Bio.Seq.Seq.join
Bio.Seq.UnknownSeq.join" class="py-name" href="#" onclick="return doclink('link-77', 'join', 'link-77');">join</a></tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">.</tt><tt id="link-78" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-78', 'split', 'link-50');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line"> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line"> </tt>
<a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line"><tt class="py-comment"># NB: Not currently used</tt> </tt>
<a name="_replace_wspace"></a><div id="_replace_wspace-def"><a name="L275"></a><tt class="py-lineno">275</tt> <a class="py-toggle" href="#" id="_replace_wspace-toggle" onclick="return toggle('_replace_wspace');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO-module.html#_replace_wspace">_replace_wspace</a><tt class="py-op">(</tt><tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_replace_wspace-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_replace_wspace-expanded"><a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line">    <tt class="py-docstring">"""Replace tab, LF and CR characters with spaces, but don't collapse (PRIVATE).</tt> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line"><tt class="py-docstring">    See "Replace Whitespace Policy" in the phyloXML spec glossary:</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line"><tt class="py-docstring">    http://phyloxml.org/documentation/version_100/phyloxml.xsd.html#Glossary</tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">char</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">"\t"</tt><tt class="py-op">,</tt> <tt class="py-string">"\n"</tt><tt class="py-op">,</tt> <tt class="py-string">"\r"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">char</tt> <tt class="py-keyword">in</tt> <tt class="py-name">text</tt><tt class="py-op">:</tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line">            <tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt class="py-name">text</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-name">char</tt><tt class="py-op">,</tt> <tt class="py-string">" "</tt><tt class="py-op">)</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">text</tt> </tt>
</div><a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line"> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line"> </tt>
<a name="Parser"></a><div id="Parser-def"><a name="L287"></a><tt class="py-lineno">287</tt> <a class="py-toggle" href="#" id="Parser-toggle" onclick="return toggle('Parser');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html">Parser</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Parser-expanded"><a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">    <tt class="py-docstring">"""Methods for parsing all phyloXML nodes from an XML stream.</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line"><tt class="py-docstring">    To minimize memory use, the tree of ElementTree parsing events is cleared</tt> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line"><tt class="py-docstring">    after completing each phylogeny, clade, and top-level 'other' element.</tt> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line"><tt class="py-docstring">    Elements below the clade level are kept in memory until parsing of the</tt> </tt>
<a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line"><tt class="py-docstring">    current clade is finished -- this shouldn't be a problem because clade is</tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line"><tt class="py-docstring">    the only recursive element, and non-clade nodes below this level are of</tt> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line"><tt class="py-docstring">    bounded size.</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line"> </tt>
<a name="Parser.__init__"></a><div id="Parser.__init__-def"><a name="L298"></a><tt class="py-lineno">298</tt> <a class="py-toggle" href="#" id="Parser.__init__-toggle" onclick="return toggle('Parser.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">file</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.__init__-expanded"><a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the class."""</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line">        <tt class="py-comment"># Get an iterable context for XML parsing events</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line">        <tt class="py-name">context</tt> <tt class="py-op">=</tt> <tt class="py-name">iter</tt><tt class="py-op">(</tt><tt class="py-name">ElementTree</tt><tt class="py-op">.</tt><tt class="py-name">iterparse</tt><tt class="py-op">(</tt><tt class="py-name">file</tt><tt class="py-op">,</tt> <tt id="link-79" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.events()=Bio.Phylo.PhyloXMLIO.Parser-class.html#events,Method Bio.Phylo.PhyloXMLIO.Writer.events()=Bio.Phylo.PhyloXMLIO.Writer-class.html#events"><a title="Bio.Phylo.PhyloXMLIO.Parser.events
Bio.Phylo.PhyloXMLIO.Writer.events" class="py-name" href="#" onclick="return doclink('link-79', 'events', 'link-79');">events</a></tt><tt class="py-op">=</tt><tt class="py-op">(</tt><tt class="py-string">"start"</tt><tt class="py-op">,</tt> <tt class="py-string">"end"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">        <tt class="py-name">event</tt><tt class="py-op">,</tt> <tt id="link-80" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-80', 'root', 'link-35');">root</a></tt> <tt class="py-op">=</tt> <tt id="link-81" class="py-name" targets="Method Bio.AlignIO.Interfaces.AlignmentIterator.next()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#next,Method Bio.Blast.ParseBlastTable.BlastTableReader.next()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#next,Method Bio.File.UndoHandle.next()=Bio.File.UndoHandle-class.html#next,Method Bio.GenBank.Iterator.next()=Bio.GenBank.Iterator-class.html#next,Method Bio.Nexus.Nexus.CharBuffer.next()=Bio.Nexus.Nexus.CharBuffer-class.html#next,Method Bio.Nexus.StandardData.StandardData.next()=Bio.Nexus.StandardData.StandardData-class.html#next,Method Bio.PopGen.GenePop.Controller._FileIterator.next()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#next,Method Bio.SearchIO._legacy.NCBIStandalone.Iterator.next()=Bio.SearchIO._legacy.NCBIStandalone.Iterator-class.html#next,Method Bio.SeqIO.Interfaces.SequenceIterator.next()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#next,Method Bio.SeqIO.SnapGeneIO._PacketIterator.next()=Bio.SeqIO.SnapGeneIO._PacketIterator-class.html#next,Method Bio.bgzf.BgzfReader.next()=Bio.bgzf.BgzfReader-class.html#next"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.Nexus.StandardData.StandardData.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SearchIO._legacy.NCBIStandalone.Iterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.SeqIO.SnapGeneIO._PacketIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-81', 'next', 'link-81');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">context</tt><tt class="py-op">)</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-82" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-82', 'root', 'link-35');">root</a></tt> <tt class="py-op">=</tt> <tt id="link-83" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-83', 'root', 'link-35');">root</a></tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">context</tt> <tt class="py-op">=</tt> <tt class="py-name">context</tt> </tt>
</div><a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line"> </tt>
<a name="Parser.read"></a><div id="Parser.read-def"><a name="L306"></a><tt class="py-lineno">306</tt> <a class="py-toggle" href="#" id="Parser.read-toggle" onclick="return toggle('Parser.read');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#read">read</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.read-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.read-expanded"><a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">        <tt class="py-docstring">"""Parse the phyloXML file and create a single Phyloxml object."""</tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line">        <tt id="link-84" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.phyloxml()=Bio.Phylo.PhyloXMLIO.Writer-class.html#phyloxml"><a title="Bio.Phylo.PhyloXMLIO.Writer.phyloxml" class="py-name" href="#" onclick="return doclink('link-84', 'phyloxml', 'link-84');">phyloxml</a></tt> <tt class="py-op">=</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-85" class="py-name"><a title="Bio.Phylo.PhyloXML.Phyloxml" class="py-name" href="#" onclick="return doclink('link-85', 'Phyloxml', 'link-36');">Phyloxml</a></tt><tt class="py-op">(</tt><tt class="py-op">{</tt><tt id="link-86" class="py-name" targets="Function Bio.Phylo.PhyloXMLIO._local()=Bio.Phylo.PhyloXMLIO-module.html#_local"><a title="Bio.Phylo.PhyloXMLIO._local" class="py-name" href="#" onclick="return doclink('link-86', '_local', 'link-86');">_local</a></tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt class="py-name">val</tt> </tt>
<a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line">                                <tt class="py-keyword">for</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">val</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-87" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-87', 'root', 'link-35');">root</a></tt><tt class="py-op">.</tt><tt id="link-88" class="py-name"><a title="Bio.Align.substitution_matrices.Array.items
Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-88', 'items', 'link-13');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line">        <tt class="py-name">other_depth</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">event</tt><tt class="py-op">,</tt> <tt class="py-name">elem</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">context</tt><tt class="py-op">:</tt> </tt>
<a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line">            <tt class="py-name">namespace</tt><tt class="py-op">,</tt> <tt class="py-name">localtag</tt> <tt class="py-op">=</tt> <tt id="link-89" class="py-name" targets="Function Bio.Phylo.PhyloXMLIO._split_namespace()=Bio.Phylo.PhyloXMLIO-module.html#_split_namespace"><a title="Bio.Phylo.PhyloXMLIO._split_namespace" class="py-name" href="#" onclick="return doclink('link-89', '_split_namespace', 'link-89');">_split_namespace</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-90" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-90', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">event</tt> <tt class="py-op">==</tt> <tt class="py-string">"start"</tt><tt class="py-op">:</tt> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">namespace</tt> <tt class="py-op">!=</tt> <tt id="link-91" class="py-name"><a title="Bio.Phylo.NeXMLIO.NAMESPACES
Bio.Phylo.PhyloXMLIO.NAMESPACES" class="py-name" href="#" onclick="return doclink('link-91', 'NAMESPACES', 'link-7');">NAMESPACES</a></tt><tt class="py-op">[</tt><tt class="py-string">"phy"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line">                    <tt class="py-name">other_depth</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line">                    <tt class="py-keyword">continue</tt> </tt>
<a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">localtag</tt> <tt class="py-op">==</tt> <tt class="py-string">"phylogeny"</tt><tt class="py-op">:</tt> </tt>
<a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">                    <tt id="link-92" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.phylogeny()=Bio.Phylo.PhyloXMLIO.Writer-class.html#phylogeny"><a title="Bio.Phylo.PhyloXMLIO.Writer.phylogeny" class="py-name" href="#" onclick="return doclink('link-92', 'phylogeny', 'link-92');">phylogeny</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-93" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser._parse_phylogeny()=Bio.Phylo.PhyloXMLIO.Parser-class.html#_parse_phylogeny"><a title="Bio.Phylo.PhyloXMLIO.Parser._parse_phylogeny" class="py-name" href="#" onclick="return doclink('link-93', '_parse_phylogeny', 'link-93');">_parse_phylogeny</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">)</tt> </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line">                    <tt id="link-94" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.phyloxml" class="py-name" href="#" onclick="return doclink('link-94', 'phyloxml', 'link-84');">phyloxml</a></tt><tt class="py-op">.</tt><tt class="py-name">phylogenies</tt><tt class="py-op">.</tt><tt id="link-95" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-95', 'append', 'link-95');">append</a></tt><tt class="py-op">(</tt><tt id="link-96" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.phylogeny" class="py-name" href="#" onclick="return doclink('link-96', 'phylogeny', 'link-92');">phylogeny</a></tt><tt class="py-op">)</tt> </tt>
<a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">event</tt> <tt class="py-op">==</tt> <tt class="py-string">"end"</tt> <tt class="py-keyword">and</tt> <tt class="py-name">namespace</tt> <tt class="py-op">!=</tt> <tt id="link-97" class="py-name"><a title="Bio.Phylo.NeXMLIO.NAMESPACES
Bio.Phylo.PhyloXMLIO.NAMESPACES" class="py-name" href="#" onclick="return doclink('link-97', 'NAMESPACES', 'link-7');">NAMESPACES</a></tt><tt class="py-op">[</tt><tt class="py-string">"phy"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line">                <tt class="py-comment"># Deal with items not specified by phyloXML</tt> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">                <tt class="py-name">other_depth</tt> <tt class="py-op">-=</tt> <tt class="py-number">1</tt> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">other_depth</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line">                    <tt class="py-comment"># We're directly under the root node -- evaluate</tt> </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line">                    <tt class="py-name">otr</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-98" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.other()=Bio.Phylo.PhyloXMLIO.Parser-class.html#other,Method Bio.Phylo.PhyloXMLIO.Writer.other()=Bio.Phylo.PhyloXMLIO.Writer-class.html#other"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-98', 'other', 'link-98');">other</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-name">namespace</tt><tt class="py-op">,</tt> <tt class="py-name">localtag</tt><tt class="py-op">)</tt> </tt>
<a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">                    <tt id="link-99" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.phyloxml" class="py-name" href="#" onclick="return doclink('link-99', 'phyloxml', 'link-84');">phyloxml</a></tt><tt class="py-op">.</tt><tt id="link-100" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-100', 'other', 'link-98');">other</a></tt><tt class="py-op">.</tt><tt id="link-101" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-101', 'append', 'link-95');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">otr</tt><tt class="py-op">)</tt> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-102" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-102', 'root', 'link-35');">root</a></tt><tt class="py-op">.</tt><tt id="link-103" class="py-name" targets="Method Bio.Crystal.Crystal.clear()=Bio.Crystal.Crystal-class.html#clear,Method Bio.File._IndexedSeqFileDict.clear()=Bio.File._IndexedSeqFileDict-class.html#clear,Method Bio.Index._InMemoryIndex.clear()=Bio.Index._InMemoryIndex-class.html#clear"><a title="Bio.Crystal.Crystal.clear
Bio.File._IndexedSeqFileDict.clear
Bio.Index._InMemoryIndex.clear" class="py-name" href="#" onclick="return doclink('link-103', 'clear', 'link-103');">clear</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-104" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.phyloxml" class="py-name" href="#" onclick="return doclink('link-104', 'phyloxml', 'link-84');">phyloxml</a></tt> </tt>
</div><a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line"> </tt>
<a name="Parser.parse"></a><div id="Parser.parse-def"><a name="L330"></a><tt class="py-lineno">330</tt> <a class="py-toggle" href="#" id="Parser.parse-toggle" onclick="return toggle('Parser.parse');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#parse">parse</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.parse-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.parse-expanded"><a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line">        <tt class="py-docstring">"""Parse the phyloXML file incrementally and return each phylogeny."""</tt> </tt>
<a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line">        <tt class="py-name">phytag</tt> <tt class="py-op">=</tt> <tt id="link-105" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._ns" class="py-name" href="#" onclick="return doclink('link-105', '_ns', 'link-55');">_ns</a></tt><tt class="py-op">(</tt><tt class="py-string">"phylogeny"</tt><tt class="py-op">)</tt> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">event</tt><tt class="py-op">,</tt> <tt class="py-name">elem</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">context</tt><tt class="py-op">:</tt> </tt>
<a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">event</tt> <tt class="py-op">==</tt> <tt class="py-string">"start"</tt> <tt class="py-keyword">and</tt> <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-106" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-106', 'tag', 'link-44');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-name">phytag</tt><tt class="py-op">:</tt> </tt>
<a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-107" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser._parse_phylogeny" class="py-name" href="#" onclick="return doclink('link-107', '_parse_phylogeny', 'link-93');">_parse_phylogeny</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">)</tt> </tt>
</div><a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line"> </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line">    <tt class="py-comment"># Special parsing cases -- incremental, using self.context</tt> </tt>
<a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line"> </tt>
<a name="Parser._parse_phylogeny"></a><div id="Parser._parse_phylogeny-def"><a name="L339"></a><tt class="py-lineno">339</tt> <a class="py-toggle" href="#" id="Parser._parse_phylogeny-toggle" onclick="return toggle('Parser._parse_phylogeny');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#_parse_phylogeny">_parse_phylogeny</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">parent</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser._parse_phylogeny-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser._parse_phylogeny-expanded"><a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line">        <tt class="py-docstring">"""Parse a single phylogeny within the phyloXML tree (PRIVATE).</tt> </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line"><tt class="py-docstring">        Recursively builds a phylogenetic tree with help from parse_clade, then</tt> </tt>
<a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line"><tt class="py-docstring">        clears the XML event history for the phylogeny element and returns</tt> </tt>
<a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line"><tt class="py-docstring">        control to the top-level parsing function.</tt> </tt>
<a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line">        <tt id="link-108" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.phylogeny" class="py-name" href="#" onclick="return doclink('link-108', 'phylogeny', 'link-92');">phylogeny</a></tt> <tt class="py-op">=</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-109" class="py-name"><a title="Bio.Phylo.PhyloXML.Phylogeny" class="py-name" href="#" onclick="return doclink('link-109', 'Phylogeny', 'link-22');">Phylogeny</a></tt><tt class="py-op">(</tt><tt class="py-op">**</tt><tt id="link-110" class="py-name" targets="Function Bio.Phylo.PhyloXMLIO._dict_str2bool()=Bio.Phylo.PhyloXMLIO-module.html#_dict_str2bool"><a title="Bio.Phylo.PhyloXMLIO._dict_str2bool" class="py-name" href="#" onclick="return doclink('link-110', '_dict_str2bool', 'link-110');">_dict_str2bool</a></tt><tt class="py-op">(</tt><tt class="py-name">parent</tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">,</tt> </tt>
<a name="L347"></a><tt class="py-lineno">347</tt>  <tt class="py-line">                                                  <tt class="py-op">[</tt><tt class="py-string">"rooted"</tt><tt class="py-op">,</tt> <tt class="py-string">"rerootable"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line">        <tt class="py-name">list_types</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line">            <tt class="py-comment"># XML tag, plural attribute</tt> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line">            <tt class="py-string">"confidence"</tt><tt class="py-op">:</tt> <tt class="py-string">"confidences"</tt><tt class="py-op">,</tt> </tt>
<a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line">            <tt class="py-string">"property"</tt><tt class="py-op">:</tt> <tt class="py-string">"properties"</tt><tt class="py-op">,</tt> </tt>
<a name="L352"></a><tt class="py-lineno">352</tt>  <tt class="py-line">            <tt class="py-string">"clade_relation"</tt><tt class="py-op">:</tt> <tt class="py-string">"clade_relations"</tt><tt class="py-op">,</tt> </tt>
<a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line">            <tt class="py-string">"sequence_relation"</tt><tt class="py-op">:</tt> <tt class="py-string">"sequence_relations"</tt><tt class="py-op">,</tt> </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line">        <tt class="py-op">}</tt> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">event</tt><tt class="py-op">,</tt> <tt class="py-name">elem</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">context</tt><tt class="py-op">:</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line">            <tt class="py-name">namespace</tt><tt class="py-op">,</tt> <tt id="link-111" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-111', 'tag', 'link-44');">tag</a></tt> <tt class="py-op">=</tt> <tt id="link-112" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._split_namespace" class="py-name" href="#" onclick="return doclink('link-112', '_split_namespace', 'link-89');">_split_namespace</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-113" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-113', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt> </tt>
<a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">event</tt> <tt class="py-op">==</tt> <tt class="py-string">"start"</tt> <tt class="py-keyword">and</tt> <tt id="link-114" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-114', 'tag', 'link-44');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"clade"</tt><tt class="py-op">:</tt> </tt>
<a name="L358"></a><tt class="py-lineno">358</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-115" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.phylogeny" class="py-name" href="#" onclick="return doclink('link-115', 'phylogeny', 'link-92');">phylogeny</a></tt><tt class="py-op">.</tt><tt id="link-116" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-116', 'root', 'link-35');">root</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Phylogeny object should only have "</tt> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line">                                     <tt class="py-string">"1 clade"</tt><tt class="py-op">)</tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line">                <tt id="link-117" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.phylogeny" class="py-name" href="#" onclick="return doclink('link-117', 'phylogeny', 'link-92');">phylogeny</a></tt><tt class="py-op">.</tt><tt id="link-118" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-118', 'root', 'link-35');">root</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-119" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser._parse_clade()=Bio.Phylo.PhyloXMLIO.Parser-class.html#_parse_clade"><a title="Bio.Phylo.PhyloXMLIO.Parser._parse_clade" class="py-name" href="#" onclick="return doclink('link-119', '_parse_clade', 'link-119');">_parse_clade</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">)</tt> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">event</tt> <tt class="py-op">==</tt> <tt class="py-string">"end"</tt><tt class="py-op">:</tt> </tt>
<a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-120" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-120', 'tag', 'link-44');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"phylogeny"</tt><tt class="py-op">:</tt> </tt>
<a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line">                    <tt class="py-name">parent</tt><tt class="py-op">.</tt><tt id="link-121" class="py-name"><a title="Bio.Crystal.Crystal.clear
Bio.File._IndexedSeqFileDict.clear
Bio.Index._InMemoryIndex.clear" class="py-name" href="#" onclick="return doclink('link-121', 'clear', 'link-103');">clear</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line">                <tt class="py-comment"># Handle the other non-recursive children</tt> </tt>
<a name="L368"></a><tt class="py-lineno">368</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-122" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-122', 'tag', 'link-44');">tag</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">list_types</tt><tt class="py-op">:</tt> </tt>
<a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line">                    <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt id="link-123" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.phylogeny" class="py-name" href="#" onclick="return doclink('link-123', 'phylogeny', 'link-92');">phylogeny</a></tt><tt class="py-op">,</tt> <tt class="py-name">list_types</tt><tt class="py-op">[</tt><tt id="link-124" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-124', 'tag', 'link-44');">tag</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-125" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-125', 'append', 'link-95');">append</a></tt><tt class="py-op">(</tt> </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line">                        <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt id="link-126" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-126', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line">                <tt class="py-comment"># Complex types</tt> </tt>
<a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt id="link-127" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-127', 'tag', 'link-44');">tag</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">"date"</tt><tt class="py-op">,</tt> <tt class="py-string">"id"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line">                    <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt id="link-128" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.phylogeny" class="py-name" href="#" onclick="return doclink('link-128', 'phylogeny', 'link-92');">phylogeny</a></tt><tt class="py-op">,</tt> <tt id="link-129" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-129', 'tag', 'link-44');">tag</a></tt><tt class="py-op">,</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt id="link-130" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-130', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line">                <tt class="py-comment"># Simple types</tt> </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt id="link-131" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-131', 'tag', 'link-44');">tag</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">"name"</tt><tt class="py-op">,</tt> <tt class="py-string">"description"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line">                    <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt id="link-132" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.phylogeny" class="py-name" href="#" onclick="return doclink('link-132', 'phylogeny', 'link-92');">phylogeny</a></tt><tt class="py-op">,</tt> <tt id="link-133" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-133', 'tag', 'link-44');">tag</a></tt><tt class="py-op">,</tt> <tt id="link-134" class="py-name" targets="Function Bio.Phylo.PhyloXMLIO._collapse_wspace()=Bio.Phylo.PhyloXMLIO-module.html#_collapse_wspace"><a title="Bio.Phylo.PhyloXMLIO._collapse_wspace" class="py-name" href="#" onclick="return doclink('link-134', '_collapse_wspace', 'link-134');">_collapse_wspace</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line">                <tt class="py-comment"># Unknown tags</tt> </tt>
<a name="L378"></a><tt class="py-lineno">378</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">namespace</tt> <tt class="py-op">!=</tt> <tt id="link-135" class="py-name"><a title="Bio.Phylo.NeXMLIO.NAMESPACES
Bio.Phylo.PhyloXMLIO.NAMESPACES" class="py-name" href="#" onclick="return doclink('link-135', 'NAMESPACES', 'link-7');">NAMESPACES</a></tt><tt class="py-op">[</tt><tt class="py-string">"phy"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L379"></a><tt class="py-lineno">379</tt>  <tt class="py-line">                    <tt id="link-136" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.phylogeny" class="py-name" href="#" onclick="return doclink('link-136', 'phylogeny', 'link-92');">phylogeny</a></tt><tt class="py-op">.</tt><tt id="link-137" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-137', 'other', 'link-98');">other</a></tt><tt class="py-op">.</tt><tt id="link-138" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-138', 'append', 'link-95');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-139" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-139', 'other', 'link-98');">other</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-name">namespace</tt><tt class="py-op">,</tt> <tt id="link-140" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-140', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L380"></a><tt class="py-lineno">380</tt>  <tt class="py-line">                    <tt class="py-name">parent</tt><tt class="py-op">.</tt><tt id="link-141" class="py-name"><a title="Bio.Crystal.Crystal.clear
Bio.File._IndexedSeqFileDict.clear
Bio.Index._InMemoryIndex.clear" class="py-name" href="#" onclick="return doclink('link-141', 'clear', 'link-103');">clear</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L381"></a><tt class="py-lineno">381</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L382"></a><tt class="py-lineno">382</tt>  <tt class="py-line">                    <tt class="py-comment"># NB: This shouldn't happen in valid files</tt> </tt>
<a name="L383"></a><tt class="py-lineno">383</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt id="link-142" class="py-name" targets="Class Bio.Phylo.PhyloXMLIO.PhyloXMLError=Bio.Phylo.PhyloXMLIO.PhyloXMLError-class.html"><a title="Bio.Phylo.PhyloXMLIO.PhyloXMLError" class="py-name" href="#" onclick="return doclink('link-142', 'PhyloXMLError', 'link-142');">PhyloXMLError</a></tt><tt class="py-op">(</tt><tt class="py-string">"Misidentified tag: "</tt> <tt class="py-op">+</tt> <tt id="link-143" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-143', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt> </tt>
<a name="L384"></a><tt class="py-lineno">384</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-144" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.phylogeny" class="py-name" href="#" onclick="return doclink('link-144', 'phylogeny', 'link-92');">phylogeny</a></tt> </tt>
</div><a name="L385"></a><tt class="py-lineno">385</tt>  <tt class="py-line"> </tt>
<a name="L386"></a><tt class="py-lineno">386</tt>  <tt class="py-line">    <tt id="link-145" class="py-name" targets="Variable Bio.Phylo.PhyloXMLIO.Parser._clade_complex_types=Bio.Phylo.PhyloXMLIO.Parser-class.html#_clade_complex_types"><a title="Bio.Phylo.PhyloXMLIO.Parser._clade_complex_types" class="py-name" href="#" onclick="return doclink('link-145', '_clade_complex_types', 'link-145');">_clade_complex_types</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">"color"</tt><tt class="py-op">,</tt> <tt class="py-string">"events"</tt><tt class="py-op">,</tt> <tt class="py-string">"binary_characters"</tt><tt class="py-op">,</tt> <tt class="py-string">"date"</tt><tt class="py-op">]</tt> </tt>
<a name="L387"></a><tt class="py-lineno">387</tt>  <tt class="py-line">    <tt id="link-146" class="py-name" targets="Variable Bio.Phylo.PhyloXMLIO.Parser._clade_list_types=Bio.Phylo.PhyloXMLIO.Parser-class.html#_clade_list_types"><a title="Bio.Phylo.PhyloXMLIO.Parser._clade_list_types" class="py-name" href="#" onclick="return doclink('link-146', '_clade_list_types', 'link-146');">_clade_list_types</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L388"></a><tt class="py-lineno">388</tt>  <tt class="py-line">        <tt class="py-string">"confidence"</tt><tt class="py-op">:</tt> <tt class="py-string">"confidences"</tt><tt class="py-op">,</tt> </tt>
<a name="L389"></a><tt class="py-lineno">389</tt>  <tt class="py-line">        <tt class="py-string">"distribution"</tt><tt class="py-op">:</tt> <tt class="py-string">"distributions"</tt><tt class="py-op">,</tt> </tt>
<a name="L390"></a><tt class="py-lineno">390</tt>  <tt class="py-line">        <tt class="py-string">"reference"</tt><tt class="py-op">:</tt> <tt class="py-string">"references"</tt><tt class="py-op">,</tt> </tt>
<a name="L391"></a><tt class="py-lineno">391</tt>  <tt class="py-line">        <tt class="py-string">"property"</tt><tt class="py-op">:</tt> <tt class="py-string">"properties"</tt><tt class="py-op">,</tt> </tt>
<a name="L392"></a><tt class="py-lineno">392</tt>  <tt class="py-line">    <tt class="py-op">}</tt> </tt>
<a name="L393"></a><tt class="py-lineno">393</tt>  <tt class="py-line">    <tt id="link-147" class="py-name" targets="Variable Bio.Phylo.PhyloXMLIO.Parser._clade_tracked_tags=Bio.Phylo.PhyloXMLIO.Parser-class.html#_clade_tracked_tags"><a title="Bio.Phylo.PhyloXMLIO.Parser._clade_tracked_tags" class="py-name" href="#" onclick="return doclink('link-147', '_clade_tracked_tags', 'link-147');">_clade_tracked_tags</a></tt> <tt class="py-op">=</tt> <tt id="link-148" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.set()=Bio.Nexus.Nexus.StepMatrix-class.html#set,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.set()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#set,Method Bio.SVDSuperimposer.SVDSuperimposer.set()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#set"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-148', 'set', 'link-148');">set</a></tt><tt class="py-op">(</tt><tt id="link-149" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser._clade_complex_types" class="py-name" href="#" onclick="return doclink('link-149', '_clade_complex_types', 'link-145');">_clade_complex_types</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">union</tt><tt class="py-op">(</tt><tt id="link-150" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser._clade_list_types" class="py-name" href="#" onclick="return doclink('link-150', '_clade_list_types', 'link-146');">_clade_list_types</a></tt><tt class="py-op">.</tt><tt id="link-151" class="py-name"><a title="Bio.Align.substitution_matrices.Array.keys
Bio.Crystal.Crystal.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-151', 'keys', 'link-75');">keys</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">union</tt><tt class="py-op">(</tt> </tt>
<a name="L394"></a><tt class="py-lineno">394</tt>  <tt class="py-line">        <tt class="py-op">[</tt><tt class="py-string">"branch_length"</tt><tt class="py-op">,</tt> <tt class="py-string">"name"</tt><tt class="py-op">,</tt> <tt class="py-string">"node_id"</tt><tt class="py-op">,</tt> <tt class="py-string">"width"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L395"></a><tt class="py-lineno">395</tt>  <tt class="py-line"> </tt>
<a name="Parser._parse_clade"></a><div id="Parser._parse_clade-def"><a name="L396"></a><tt class="py-lineno">396</tt> <a class="py-toggle" href="#" id="Parser._parse_clade-toggle" onclick="return toggle('Parser._parse_clade');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#_parse_clade">_parse_clade</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">parent</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser._parse_clade-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser._parse_clade-expanded"><a name="L397"></a><tt class="py-lineno">397</tt>  <tt class="py-line">        <tt class="py-docstring">"""Parse a Clade node and its children, recursively (PRIVATE)."""</tt> </tt>
<a name="L398"></a><tt class="py-lineno">398</tt>  <tt class="py-line">        <tt id="link-152" class="py-name" targets="Variable Bio.Phylo.BaseTree.Tree.clade=Bio.Phylo.BaseTree.Tree-class.html#clade,Method Bio.Phylo.PhyloXMLIO.Writer.clade()=Bio.Phylo.PhyloXMLIO.Writer-class.html#clade"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-152', 'clade', 'link-152');">clade</a></tt> <tt class="py-op">=</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-153" class="py-name"><a title="Bio.Phylo.BaseTree.Clade
Bio.Phylo.CDAO.Clade
Bio.Phylo.NeXML.Clade
Bio.Phylo.Newick.Clade
Bio.Phylo.PhyloXML.Clade" class="py-name" href="#" onclick="return doclink('link-153', 'Clade', 'link-23');">Clade</a></tt><tt class="py-op">(</tt><tt class="py-op">**</tt><tt class="py-name">parent</tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">)</tt> </tt>
<a name="L399"></a><tt class="py-lineno">399</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-154" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-154', 'clade', 'link-152');">clade</a></tt><tt class="py-op">.</tt><tt id="link-155" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.branch_length()=Bio.Phylo.PhyloXMLIO.Writer-class.html#branch_length"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-155', 'branch_length', 'link-155');">branch_length</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L400"></a><tt class="py-lineno">400</tt>  <tt class="py-line">            <tt id="link-156" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-156', 'clade', 'link-152');">clade</a></tt><tt class="py-op">.</tt><tt id="link-157" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-157', 'branch_length', 'link-155');">branch_length</a></tt> <tt class="py-op">=</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt id="link-158" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-158', 'clade', 'link-152');">clade</a></tt><tt class="py-op">.</tt><tt id="link-159" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-159', 'branch_length', 'link-155');">branch_length</a></tt><tt class="py-op">)</tt> </tt>
<a name="L401"></a><tt class="py-lineno">401</tt>  <tt class="py-line">        <tt class="py-comment"># NB: Only evaluate nodes at the current level</tt> </tt>
<a name="L402"></a><tt class="py-lineno">402</tt>  <tt class="py-line">        <tt class="py-name">tag_stack</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L403"></a><tt class="py-lineno">403</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">event</tt><tt class="py-op">,</tt> <tt class="py-name">elem</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">context</tt><tt class="py-op">:</tt> </tt>
<a name="L404"></a><tt class="py-lineno">404</tt>  <tt class="py-line">            <tt class="py-name">namespace</tt><tt class="py-op">,</tt> <tt id="link-160" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-160', 'tag', 'link-44');">tag</a></tt> <tt class="py-op">=</tt> <tt id="link-161" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._split_namespace" class="py-name" href="#" onclick="return doclink('link-161', '_split_namespace', 'link-89');">_split_namespace</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-162" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-162', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt> </tt>
<a name="L405"></a><tt class="py-lineno">405</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">event</tt> <tt class="py-op">==</tt> <tt class="py-string">"start"</tt><tt class="py-op">:</tt> </tt>
<a name="L406"></a><tt class="py-lineno">406</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-163" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-163', 'tag', 'link-44');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"clade"</tt><tt class="py-op">:</tt> </tt>
<a name="L407"></a><tt class="py-lineno">407</tt>  <tt class="py-line">                    <tt id="link-164" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-164', 'clade', 'link-152');">clade</a></tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">.</tt><tt id="link-165" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-165', 'append', 'link-95');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-166" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser._parse_clade" class="py-name" href="#" onclick="return doclink('link-166', '_parse_clade', 'link-119');">_parse_clade</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L408"></a><tt class="py-lineno">408</tt>  <tt class="py-line">                    <tt class="py-keyword">continue</tt> </tt>
<a name="L409"></a><tt class="py-lineno">409</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-167" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-167', 'tag', 'link-44');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"taxonomy"</tt><tt class="py-op">:</tt> </tt>
<a name="L410"></a><tt class="py-lineno">410</tt>  <tt class="py-line">                    <tt id="link-168" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-168', 'clade', 'link-152');">clade</a></tt><tt class="py-op">.</tt><tt class="py-name">taxonomies</tt><tt class="py-op">.</tt><tt id="link-169" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-169', 'append', 'link-95');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-170" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser._parse_taxonomy()=Bio.Phylo.PhyloXMLIO.Parser-class.html#_parse_taxonomy"><a title="Bio.Phylo.PhyloXMLIO.Parser._parse_taxonomy" class="py-name" href="#" onclick="return doclink('link-170', '_parse_taxonomy', 'link-170');">_parse_taxonomy</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L411"></a><tt class="py-lineno">411</tt>  <tt class="py-line">                    <tt class="py-keyword">continue</tt> </tt>
<a name="L412"></a><tt class="py-lineno">412</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-171" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-171', 'tag', 'link-44');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"sequence"</tt><tt class="py-op">:</tt> </tt>
<a name="L413"></a><tt class="py-lineno">413</tt>  <tt class="py-line">                    <tt id="link-172" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-172', 'clade', 'link-152');">clade</a></tt><tt class="py-op">.</tt><tt class="py-name">sequences</tt><tt class="py-op">.</tt><tt id="link-173" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-173', 'append', 'link-95');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-174" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser._parse_sequence()=Bio.Phylo.PhyloXMLIO.Parser-class.html#_parse_sequence"><a title="Bio.Phylo.PhyloXMLIO.Parser._parse_sequence" class="py-name" href="#" onclick="return doclink('link-174', '_parse_sequence', 'link-174');">_parse_sequence</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L414"></a><tt class="py-lineno">414</tt>  <tt class="py-line">                    <tt class="py-keyword">continue</tt> </tt>
<a name="L415"></a><tt class="py-lineno">415</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-175" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-175', 'tag', 'link-44');">tag</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-176" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser._clade_tracked_tags" class="py-name" href="#" onclick="return doclink('link-176', '_clade_tracked_tags', 'link-147');">_clade_tracked_tags</a></tt><tt class="py-op">:</tt> </tt>
<a name="L416"></a><tt class="py-lineno">416</tt>  <tt class="py-line">                    <tt class="py-name">tag_stack</tt><tt class="py-op">.</tt><tt id="link-177" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-177', 'append', 'link-95');">append</a></tt><tt class="py-op">(</tt><tt id="link-178" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-178', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt> </tt>
<a name="L417"></a><tt class="py-lineno">417</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">event</tt> <tt class="py-op">==</tt> <tt class="py-string">"end"</tt><tt class="py-op">:</tt> </tt>
<a name="L418"></a><tt class="py-lineno">418</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-179" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-179', 'tag', 'link-44');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"clade"</tt><tt class="py-op">:</tt> </tt>
<a name="L419"></a><tt class="py-lineno">419</tt>  <tt class="py-line">                    <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-180" class="py-name"><a title="Bio.Crystal.Crystal.clear
Bio.File._IndexedSeqFileDict.clear
Bio.Index._InMemoryIndex.clear" class="py-name" href="#" onclick="return doclink('link-180', 'clear', 'link-103');">clear</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L420"></a><tt class="py-lineno">420</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L421"></a><tt class="py-lineno">421</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-181" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-181', 'tag', 'link-44');">tag</a></tt> <tt class="py-op">!=</tt> <tt class="py-name">tag_stack</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L422"></a><tt class="py-lineno">422</tt>  <tt class="py-line">                    <tt class="py-keyword">continue</tt> </tt>
<a name="L423"></a><tt class="py-lineno">423</tt>  <tt class="py-line">                <tt class="py-name">tag_stack</tt><tt class="py-op">.</tt><tt id="link-182" class="py-name" targets="Method Bio.File._IndexedSeqFileDict.pop()=Bio.File._IndexedSeqFileDict-class.html#pop,Method Bio.SearchIO._model.hit.Hit.pop()=Bio.SearchIO._model.hit.Hit-class.html#pop,Method Bio.SearchIO._model.query.QueryResult.pop()=Bio.SearchIO._model.query.QueryResult-class.html#pop,Method Bio.Seq.MutableSeq.pop()=Bio.Seq.MutableSeq-class.html#pop"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop" class="py-name" href="#" onclick="return doclink('link-182', 'pop', 'link-182');">pop</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L424"></a><tt class="py-lineno">424</tt>  <tt class="py-line">                <tt class="py-comment"># Handle the other non-recursive children</tt> </tt>
<a name="L425"></a><tt class="py-lineno">425</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-183" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-183', 'tag', 'link-44');">tag</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-184" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser._clade_list_types" class="py-name" href="#" onclick="return doclink('link-184', '_clade_list_types', 'link-146');">_clade_list_types</a></tt><tt class="py-op">:</tt> </tt>
<a name="L426"></a><tt class="py-lineno">426</tt>  <tt class="py-line">                    <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt id="link-185" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-185', 'clade', 'link-152');">clade</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-186" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser._clade_list_types" class="py-name" href="#" onclick="return doclink('link-186', '_clade_list_types', 'link-146');">_clade_list_types</a></tt><tt class="py-op">[</tt><tt id="link-187" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-187', 'tag', 'link-44');">tag</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-188" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-188', 'append', 'link-95');">append</a></tt><tt class="py-op">(</tt> </tt>
<a name="L427"></a><tt class="py-lineno">427</tt>  <tt class="py-line">                        <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt id="link-189" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-189', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L428"></a><tt class="py-lineno">428</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt id="link-190" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-190', 'tag', 'link-44');">tag</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-191" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser._clade_complex_types" class="py-name" href="#" onclick="return doclink('link-191', '_clade_complex_types', 'link-145');">_clade_complex_types</a></tt><tt class="py-op">:</tt> </tt>
<a name="L429"></a><tt class="py-lineno">429</tt>  <tt class="py-line">                    <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt id="link-192" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-192', 'clade', 'link-152');">clade</a></tt><tt class="py-op">,</tt> <tt id="link-193" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-193', 'tag', 'link-44');">tag</a></tt><tt class="py-op">,</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt id="link-194" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-194', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L430"></a><tt class="py-lineno">430</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt id="link-195" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-195', 'tag', 'link-44');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"branch_length"</tt><tt class="py-op">:</tt> </tt>
<a name="L431"></a><tt class="py-lineno">431</tt>  <tt class="py-line">                    <tt class="py-comment"># NB: possible collision with the attribute</tt> </tt>
<a name="L432"></a><tt class="py-lineno">432</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt id="link-196" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-196', 'clade', 'link-152');">clade</a></tt><tt class="py-op">.</tt><tt id="link-197" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-197', 'branch_length', 'link-155');">branch_length</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L433"></a><tt class="py-lineno">433</tt>  <tt class="py-line">                        <tt class="py-keyword">raise</tt> <tt id="link-198" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.PhyloXMLError" class="py-name" href="#" onclick="return doclink('link-198', 'PhyloXMLError', 'link-142');">PhyloXMLError</a></tt><tt class="py-op">(</tt> </tt>
<a name="L434"></a><tt class="py-lineno">434</tt>  <tt class="py-line">                            <tt class="py-string">"Attribute branch_length was already set "</tt> </tt>
<a name="L435"></a><tt class="py-lineno">435</tt>  <tt class="py-line">                            <tt class="py-string">"for this Clade."</tt><tt class="py-op">)</tt> </tt>
<a name="L436"></a><tt class="py-lineno">436</tt>  <tt class="py-line">                    <tt id="link-199" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-199', 'clade', 'link-152');">clade</a></tt><tt class="py-op">.</tt><tt id="link-200" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-200', 'branch_length', 'link-155');">branch_length</a></tt> <tt class="py-op">=</tt> <tt id="link-201" class="py-name" targets="Function Bio.Phylo.PhyloXMLIO._float()=Bio.Phylo.PhyloXMLIO-module.html#_float"><a title="Bio.Phylo.PhyloXMLIO._float" class="py-name" href="#" onclick="return doclink('link-201', '_float', 'link-201');">_float</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
<a name="L437"></a><tt class="py-lineno">437</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt id="link-202" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-202', 'tag', 'link-44');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"width"</tt><tt class="py-op">:</tt> </tt>
<a name="L438"></a><tt class="py-lineno">438</tt>  <tt class="py-line">                    <tt id="link-203" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-203', 'clade', 'link-152');">clade</a></tt><tt class="py-op">.</tt><tt id="link-204" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.width=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#width,Method Bio.Phylo.PhyloXMLIO.Writer.width()=Bio.Phylo.PhyloXMLIO.Writer-class.html#width"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.width
Bio.Phylo.PhyloXMLIO.Writer.width" class="py-name" href="#" onclick="return doclink('link-204', 'width', 'link-204');">width</a></tt> <tt class="py-op">=</tt> <tt id="link-205" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._float" class="py-name" href="#" onclick="return doclink('link-205', '_float', 'link-201');">_float</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
<a name="L439"></a><tt class="py-lineno">439</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt id="link-206" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-206', 'tag', 'link-44');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"name"</tt><tt class="py-op">:</tt> </tt>
<a name="L440"></a><tt class="py-lineno">440</tt>  <tt class="py-line">                    <tt id="link-207" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-207', 'clade', 'link-152');">clade</a></tt><tt class="py-op">.</tt><tt id="link-208" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-208', 'name', 'link-208');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-209" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._collapse_wspace" class="py-name" href="#" onclick="return doclink('link-209', '_collapse_wspace', 'link-134');">_collapse_wspace</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
<a name="L441"></a><tt class="py-lineno">441</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt id="link-210" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-210', 'tag', 'link-44');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"node_id"</tt><tt class="py-op">:</tt> </tt>
<a name="L442"></a><tt class="py-lineno">442</tt>  <tt class="py-line">                    <tt id="link-211" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-211', 'clade', 'link-152');">clade</a></tt><tt class="py-op">.</tt><tt id="link-212" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.node_id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#node_id"><a title="Bio.Phylo.PhyloXMLIO.Writer.node_id" class="py-name" href="#" onclick="return doclink('link-212', 'node_id', 'link-212');">node_id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-213" class="py-name" targets="Class Bio.Phylo.PhyloXML.Id=Bio.Phylo.PhyloXML.Id-class.html"><a title="Bio.Phylo.PhyloXML.Id" class="py-name" href="#" onclick="return doclink('link-213', 'Id', 'link-213');">Id</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt><tt class="py-op">.</tt><tt id="link-214" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-214', 'strip', 'link-65');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L443"></a><tt class="py-lineno">443</tt>  <tt class="py-line">                                          <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">.</tt><tt id="link-215" class="py-name" targets="Method Bio.Align.substitution_matrices.Array.get()=Bio.Align.substitution_matrices.Array-class.html#get,Method Bio.Crystal.Crystal.get()=Bio.Crystal.Crystal-class.html#get,Method Bio.Data.CodonTable.AmbiguousForwardTable.get()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#get,Method Bio.File._IndexedSeqFileDict.get()=Bio.File._IndexedSeqFileDict-class.html#get,Method Bio.File._IndexedSeqFileProxy.get()=Bio.File._IndexedSeqFileProxy-class.html#get,Method Bio.File._SQLiteManySeqFilesDict.get()=Bio.File._SQLiteManySeqFilesDict-class.html#get,Method Bio.Restriction.Restriction.RestrictionBatch.get()=Bio.Restriction.Restriction.RestrictionBatch-class.html#get,Method Bio.SearchIO._index.SearchIndexer.get()=Bio.SearchIO._index.SearchIndexer-class.html#get,Method Bio.SeqIO._index.SeqFileRandomAccess.get()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#get,Method Bio.SeqIO._index.SffRandomAccess.get()=Bio.SeqIO._index.SffRandomAccess-class.html#get,Method Bio.SeqIO._index.SffTrimedRandomAccess.get()=Bio.SeqIO._index.SffTrimedRandomAccess-class.html#get,Method Bio.SeqIO._index.UniprotRandomAccess.get()=Bio.SeqIO._index.UniprotRandomAccess-class.html#get"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-215', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"provider"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L444"></a><tt class="py-lineno">444</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">namespace</tt> <tt class="py-op">!=</tt> <tt id="link-216" class="py-name"><a title="Bio.Phylo.NeXMLIO.NAMESPACES
Bio.Phylo.PhyloXMLIO.NAMESPACES" class="py-name" href="#" onclick="return doclink('link-216', 'NAMESPACES', 'link-7');">NAMESPACES</a></tt><tt class="py-op">[</tt><tt class="py-string">"phy"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L445"></a><tt class="py-lineno">445</tt>  <tt class="py-line">                    <tt id="link-217" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-217', 'clade', 'link-152');">clade</a></tt><tt class="py-op">.</tt><tt id="link-218" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-218', 'other', 'link-98');">other</a></tt><tt class="py-op">.</tt><tt id="link-219" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-219', 'append', 'link-95');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-220" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-220', 'other', 'link-98');">other</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-name">namespace</tt><tt class="py-op">,</tt> <tt id="link-221" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-221', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L446"></a><tt class="py-lineno">446</tt>  <tt class="py-line">                    <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-222" class="py-name"><a title="Bio.Crystal.Crystal.clear
Bio.File._IndexedSeqFileDict.clear
Bio.Index._InMemoryIndex.clear" class="py-name" href="#" onclick="return doclink('link-222', 'clear', 'link-103');">clear</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L447"></a><tt class="py-lineno">447</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L448"></a><tt class="py-lineno">448</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt id="link-223" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.PhyloXMLError" class="py-name" href="#" onclick="return doclink('link-223', 'PhyloXMLError', 'link-142');">PhyloXMLError</a></tt><tt class="py-op">(</tt><tt class="py-string">"Misidentified tag: "</tt> <tt class="py-op">+</tt> <tt id="link-224" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-224', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt> </tt>
<a name="L449"></a><tt class="py-lineno">449</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-225" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-225', 'clade', 'link-152');">clade</a></tt> </tt>
</div><a name="L450"></a><tt class="py-lineno">450</tt>  <tt class="py-line"> </tt>
<a name="Parser._parse_sequence"></a><div id="Parser._parse_sequence-def"><a name="L451"></a><tt class="py-lineno">451</tt> <a class="py-toggle" href="#" id="Parser._parse_sequence-toggle" onclick="return toggle('Parser._parse_sequence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#_parse_sequence">_parse_sequence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">parent</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser._parse_sequence-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser._parse_sequence-expanded"><a name="L452"></a><tt class="py-lineno">452</tt>  <tt class="py-line">        <tt class="py-docstring">"""Parse a molecular sequence (PRIVATE)."""</tt> </tt>
<a name="L453"></a><tt class="py-lineno">453</tt>  <tt class="py-line">        <tt id="link-226" class="py-name" targets="Method Bio.FSSP.FSSPAlignDict.sequence()=Bio.FSSP.FSSPAlignDict-class.html#sequence,Method Bio.GenBank._FeatureConsumer.sequence()=Bio.GenBank._FeatureConsumer-class.html#sequence,Method Bio.GenBank._RecordConsumer.sequence()=Bio.GenBank._RecordConsumer-class.html#sequence,Method Bio.Phylo.PhyloXMLIO.Writer.sequence()=Bio.Phylo.PhyloXMLIO.Writer-class.html#sequence"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-226', 'sequence', 'link-226');">sequence</a></tt> <tt class="py-op">=</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-227" class="py-name" targets="Class Bio.Phylo.PhyloXML.Sequence=Bio.Phylo.PhyloXML.Sequence-class.html"><a title="Bio.Phylo.PhyloXML.Sequence" class="py-name" href="#" onclick="return doclink('link-227', 'Sequence', 'link-227');">Sequence</a></tt><tt class="py-op">(</tt><tt class="py-op">**</tt><tt class="py-name">parent</tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">)</tt> </tt>
<a name="L454"></a><tt class="py-lineno">454</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">event</tt><tt class="py-op">,</tt> <tt class="py-name">elem</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">context</tt><tt class="py-op">:</tt> </tt>
<a name="L455"></a><tt class="py-lineno">455</tt>  <tt class="py-line">            <tt class="py-name">namespace</tt><tt class="py-op">,</tt> <tt id="link-228" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-228', 'tag', 'link-44');">tag</a></tt> <tt class="py-op">=</tt> <tt id="link-229" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._split_namespace" class="py-name" href="#" onclick="return doclink('link-229', '_split_namespace', 'link-89');">_split_namespace</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-230" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-230', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt> </tt>
<a name="L456"></a><tt class="py-lineno">456</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">event</tt> <tt class="py-op">==</tt> <tt class="py-string">"end"</tt><tt class="py-op">:</tt> </tt>
<a name="L457"></a><tt class="py-lineno">457</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-231" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-231', 'tag', 'link-44');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"sequence"</tt><tt class="py-op">:</tt> </tt>
<a name="L458"></a><tt class="py-lineno">458</tt>  <tt class="py-line">                    <tt class="py-name">parent</tt><tt class="py-op">.</tt><tt id="link-232" class="py-name"><a title="Bio.Crystal.Crystal.clear
Bio.File._IndexedSeqFileDict.clear
Bio.Index._InMemoryIndex.clear" class="py-name" href="#" onclick="return doclink('link-232', 'clear', 'link-103');">clear</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L459"></a><tt class="py-lineno">459</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L460"></a><tt class="py-lineno">460</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-233" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-233', 'tag', 'link-44');">tag</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">"accession"</tt><tt class="py-op">,</tt> <tt class="py-string">"mol_seq"</tt><tt class="py-op">,</tt> <tt class="py-string">"uri"</tt><tt class="py-op">,</tt> </tt>
<a name="L461"></a><tt class="py-lineno">461</tt>  <tt class="py-line">                           <tt class="py-string">"domain_architecture"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L462"></a><tt class="py-lineno">462</tt>  <tt class="py-line">                    <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt id="link-234" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-234', 'sequence', 'link-226');">sequence</a></tt><tt class="py-op">,</tt> <tt id="link-235" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-235', 'tag', 'link-44');">tag</a></tt><tt class="py-op">,</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt id="link-236" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-236', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L463"></a><tt class="py-lineno">463</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt id="link-237" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-237', 'tag', 'link-44');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"annotation"</tt><tt class="py-op">:</tt> </tt>
<a name="L464"></a><tt class="py-lineno">464</tt>  <tt class="py-line">                    <tt id="link-238" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-238', 'sequence', 'link-226');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-239" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.annotations=BioSQL.BioSeq.DBSeqRecord-class.html#annotations"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-239', 'annotations', 'link-239');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-240" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-240', 'append', 'link-95');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-241" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.annotation()=Bio.Phylo.PhyloXMLIO.Parser-class.html#annotation,Method Bio.Phylo.PhyloXMLIO.Writer.annotation()=Bio.Phylo.PhyloXMLIO.Writer-class.html#annotation"><a title="Bio.Phylo.PhyloXMLIO.Parser.annotation
Bio.Phylo.PhyloXMLIO.Writer.annotation" class="py-name" href="#" onclick="return doclink('link-241', 'annotation', 'link-241');">annotation</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L465"></a><tt class="py-lineno">465</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt id="link-242" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-242', 'tag', 'link-44');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"name"</tt><tt class="py-op">:</tt> </tt>
<a name="L466"></a><tt class="py-lineno">466</tt>  <tt class="py-line">                    <tt id="link-243" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-243', 'sequence', 'link-226');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-244" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-244', 'name', 'link-208');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-245" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._collapse_wspace" class="py-name" href="#" onclick="return doclink('link-245', '_collapse_wspace', 'link-134');">_collapse_wspace</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
<a name="L467"></a><tt class="py-lineno">467</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt id="link-246" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-246', 'tag', 'link-44');">tag</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">"symbol"</tt><tt class="py-op">,</tt> <tt class="py-string">"location"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L468"></a><tt class="py-lineno">468</tt>  <tt class="py-line">                    <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt id="link-247" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-247', 'sequence', 'link-226');">sequence</a></tt><tt class="py-op">,</tt> <tt id="link-248" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-248', 'tag', 'link-44');">tag</a></tt><tt class="py-op">,</tt> <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
<a name="L469"></a><tt class="py-lineno">469</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">namespace</tt> <tt class="py-op">!=</tt> <tt id="link-249" class="py-name"><a title="Bio.Phylo.NeXMLIO.NAMESPACES
Bio.Phylo.PhyloXMLIO.NAMESPACES" class="py-name" href="#" onclick="return doclink('link-249', 'NAMESPACES', 'link-7');">NAMESPACES</a></tt><tt class="py-op">[</tt><tt class="py-string">"phy"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L470"></a><tt class="py-lineno">470</tt>  <tt class="py-line">                    <tt id="link-250" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-250', 'sequence', 'link-226');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-251" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-251', 'other', 'link-98');">other</a></tt><tt class="py-op">.</tt><tt id="link-252" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-252', 'append', 'link-95');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-253" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-253', 'other', 'link-98');">other</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-name">namespace</tt><tt class="py-op">,</tt> <tt id="link-254" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-254', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L471"></a><tt class="py-lineno">471</tt>  <tt class="py-line">                    <tt class="py-name">parent</tt><tt class="py-op">.</tt><tt id="link-255" class="py-name"><a title="Bio.Crystal.Crystal.clear
Bio.File._IndexedSeqFileDict.clear
Bio.Index._InMemoryIndex.clear" class="py-name" href="#" onclick="return doclink('link-255', 'clear', 'link-103');">clear</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L472"></a><tt class="py-lineno">472</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-256" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-256', 'sequence', 'link-226');">sequence</a></tt> </tt>
</div><a name="L473"></a><tt class="py-lineno">473</tt>  <tt class="py-line"> </tt>
<a name="Parser._parse_taxonomy"></a><div id="Parser._parse_taxonomy-def"><a name="L474"></a><tt class="py-lineno">474</tt> <a class="py-toggle" href="#" id="Parser._parse_taxonomy-toggle" onclick="return toggle('Parser._parse_taxonomy');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#_parse_taxonomy">_parse_taxonomy</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">parent</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser._parse_taxonomy-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser._parse_taxonomy-expanded"><a name="L475"></a><tt class="py-lineno">475</tt>  <tt class="py-line">        <tt class="py-docstring">"""Parse taxonomic information for a clade (PRIVATE)."""</tt> </tt>
<a name="L476"></a><tt class="py-lineno">476</tt>  <tt class="py-line">        <tt id="link-257" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.taxonomy()=Bio.GenBank._FeatureConsumer-class.html#taxonomy,Method Bio.GenBank._RecordConsumer.taxonomy()=Bio.GenBank._RecordConsumer-class.html#taxonomy,Variable Bio.Phylo.PhyloXML.Clade.taxonomy=Bio.Phylo.PhyloXML.Clade-class.html#taxonomy,Method Bio.Phylo.PhyloXMLIO.Writer.taxonomy()=Bio.Phylo.PhyloXMLIO.Writer-class.html#taxonomy"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-257', 'taxonomy', 'link-257');">taxonomy</a></tt> <tt class="py-op">=</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-258" class="py-name" targets="Class Bio.Phylo.PhyloXML.Taxonomy=Bio.Phylo.PhyloXML.Taxonomy-class.html"><a title="Bio.Phylo.PhyloXML.Taxonomy" class="py-name" href="#" onclick="return doclink('link-258', 'Taxonomy', 'link-258');">Taxonomy</a></tt><tt class="py-op">(</tt><tt class="py-op">**</tt><tt class="py-name">parent</tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">)</tt> </tt>
<a name="L477"></a><tt class="py-lineno">477</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">event</tt><tt class="py-op">,</tt> <tt class="py-name">elem</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">context</tt><tt class="py-op">:</tt> </tt>
<a name="L478"></a><tt class="py-lineno">478</tt>  <tt class="py-line">            <tt class="py-name">namespace</tt><tt class="py-op">,</tt> <tt id="link-259" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-259', 'tag', 'link-44');">tag</a></tt> <tt class="py-op">=</tt> <tt id="link-260" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._split_namespace" class="py-name" href="#" onclick="return doclink('link-260', '_split_namespace', 'link-89');">_split_namespace</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-261" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-261', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt> </tt>
<a name="L479"></a><tt class="py-lineno">479</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">event</tt> <tt class="py-op">==</tt> <tt class="py-string">"end"</tt><tt class="py-op">:</tt> </tt>
<a name="L480"></a><tt class="py-lineno">480</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-262" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-262', 'tag', 'link-44');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"taxonomy"</tt><tt class="py-op">:</tt> </tt>
<a name="L481"></a><tt class="py-lineno">481</tt>  <tt class="py-line">                    <tt class="py-name">parent</tt><tt class="py-op">.</tt><tt id="link-263" class="py-name"><a title="Bio.Crystal.Crystal.clear
Bio.File._IndexedSeqFileDict.clear
Bio.Index._InMemoryIndex.clear" class="py-name" href="#" onclick="return doclink('link-263', 'clear', 'link-103');">clear</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L482"></a><tt class="py-lineno">482</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L483"></a><tt class="py-lineno">483</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-264" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-264', 'tag', 'link-44');">tag</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">"id"</tt><tt class="py-op">,</tt> <tt class="py-string">"uri"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L484"></a><tt class="py-lineno">484</tt>  <tt class="py-line">                    <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt id="link-265" class="py-name"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-265', 'taxonomy', 'link-257');">taxonomy</a></tt><tt class="py-op">,</tt> <tt id="link-266" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-266', 'tag', 'link-44');">tag</a></tt><tt class="py-op">,</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt id="link-267" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-267', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L485"></a><tt class="py-lineno">485</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt id="link-268" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-268', 'tag', 'link-44');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"common_name"</tt><tt class="py-op">:</tt> </tt>
<a name="L486"></a><tt class="py-lineno">486</tt>  <tt class="py-line">                    <tt id="link-269" class="py-name"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-269', 'taxonomy', 'link-257');">taxonomy</a></tt><tt class="py-op">.</tt><tt class="py-name">common_names</tt><tt class="py-op">.</tt><tt id="link-270" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-270', 'append', 'link-95');">append</a></tt><tt class="py-op">(</tt><tt id="link-271" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._collapse_wspace" class="py-name" href="#" onclick="return doclink('link-271', '_collapse_wspace', 'link-134');">_collapse_wspace</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L487"></a><tt class="py-lineno">487</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt id="link-272" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-272', 'tag', 'link-44');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"synonym"</tt><tt class="py-op">:</tt> </tt>
<a name="L488"></a><tt class="py-lineno">488</tt>  <tt class="py-line">                    <tt id="link-273" class="py-name"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-273', 'taxonomy', 'link-257');">taxonomy</a></tt><tt class="py-op">.</tt><tt class="py-name">synonyms</tt><tt class="py-op">.</tt><tt id="link-274" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-274', 'append', 'link-95');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
<a name="L489"></a><tt class="py-lineno">489</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt id="link-275" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-275', 'tag', 'link-44');">tag</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">"code"</tt><tt class="py-op">,</tt> <tt class="py-string">"scientific_name"</tt><tt class="py-op">,</tt> <tt class="py-string">"authority"</tt><tt class="py-op">,</tt> <tt class="py-string">"rank"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L490"></a><tt class="py-lineno">490</tt>  <tt class="py-line">                    <tt class="py-comment"># ENH: check_str on rank</tt> </tt>
<a name="L491"></a><tt class="py-lineno">491</tt>  <tt class="py-line">                    <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt id="link-276" class="py-name"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-276', 'taxonomy', 'link-257');">taxonomy</a></tt><tt class="py-op">,</tt> <tt id="link-277" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-277', 'tag', 'link-44');">tag</a></tt><tt class="py-op">,</tt> <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
<a name="L492"></a><tt class="py-lineno">492</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">namespace</tt> <tt class="py-op">!=</tt> <tt id="link-278" class="py-name"><a title="Bio.Phylo.NeXMLIO.NAMESPACES
Bio.Phylo.PhyloXMLIO.NAMESPACES" class="py-name" href="#" onclick="return doclink('link-278', 'NAMESPACES', 'link-7');">NAMESPACES</a></tt><tt class="py-op">[</tt><tt class="py-string">"phy"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L493"></a><tt class="py-lineno">493</tt>  <tt class="py-line">                    <tt id="link-279" class="py-name"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-279', 'taxonomy', 'link-257');">taxonomy</a></tt><tt class="py-op">.</tt><tt id="link-280" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-280', 'other', 'link-98');">other</a></tt><tt class="py-op">.</tt><tt id="link-281" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-281', 'append', 'link-95');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-282" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-282', 'other', 'link-98');">other</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-name">namespace</tt><tt class="py-op">,</tt> <tt id="link-283" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-283', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L494"></a><tt class="py-lineno">494</tt>  <tt class="py-line">                    <tt class="py-name">parent</tt><tt class="py-op">.</tt><tt id="link-284" class="py-name"><a title="Bio.Crystal.Crystal.clear
Bio.File._IndexedSeqFileDict.clear
Bio.Index._InMemoryIndex.clear" class="py-name" href="#" onclick="return doclink('link-284', 'clear', 'link-103');">clear</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L495"></a><tt class="py-lineno">495</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-285" class="py-name"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-285', 'taxonomy', 'link-257');">taxonomy</a></tt> </tt>
</div><a name="L496"></a><tt class="py-lineno">496</tt>  <tt class="py-line"> </tt>
<a name="Parser.other"></a><div id="Parser.other-def"><a name="L497"></a><tt class="py-lineno">497</tt> <a class="py-toggle" href="#" id="Parser.other-toggle" onclick="return toggle('Parser.other');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#other">other</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">elem</tt><tt class="py-op">,</tt> <tt class="py-param">namespace</tt><tt class="py-op">,</tt> <tt class="py-param">localtag</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.other-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.other-expanded"><a name="L498"></a><tt class="py-lineno">498</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create an Other object, a non-phyloXML element."""</tt> </tt>
<a name="L499"></a><tt class="py-lineno">499</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-286" class="py-name" targets="Class Bio.Phylo.PhyloXML.Other=Bio.Phylo.PhyloXML.Other-class.html"><a title="Bio.Phylo.PhyloXML.Other" class="py-name" href="#" onclick="return doclink('link-286', 'Other', 'link-286');">Other</a></tt><tt class="py-op">(</tt><tt class="py-name">localtag</tt><tt class="py-op">,</tt> <tt class="py-name">namespace</tt><tt class="py-op">,</tt> <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">,</tt> </tt>
<a name="L500"></a><tt class="py-lineno">500</tt>  <tt class="py-line">                        <tt id="link-287" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.value()=Bio.Phylo.PhyloXMLIO.Writer-class.html#value"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-287', 'value', 'link-287');">value</a></tt><tt class="py-op">=</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt> <tt class="py-keyword">and</tt> <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt><tt class="py-op">.</tt><tt id="link-288" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-288', 'strip', 'link-65');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L501"></a><tt class="py-lineno">501</tt>  <tt class="py-line">                        <tt id="link-289" class="py-name" targets="Method Bio.Pathway.Rep.Graph.Graph.children()=Bio.Pathway.Rep.Graph.Graph-class.html#children,Method Bio.Pathway.Rep.MultiGraph.MultiGraph.children()=Bio.Pathway.Rep.MultiGraph.MultiGraph-class.html#children"><a title="Bio.Pathway.Rep.Graph.Graph.children
Bio.Pathway.Rep.MultiGraph.MultiGraph.children" class="py-name" href="#" onclick="return doclink('link-289', 'children', 'link-289');">children</a></tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-290" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-290', 'other', 'link-98');">other</a></tt><tt class="py-op">(</tt><tt class="py-name">child</tt><tt class="py-op">,</tt> <tt class="py-op">*</tt><tt id="link-291" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._split_namespace" class="py-name" href="#" onclick="return doclink('link-291', '_split_namespace', 'link-89');">_split_namespace</a></tt><tt class="py-op">(</tt><tt class="py-name">child</tt><tt class="py-op">.</tt><tt id="link-292" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-292', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L502"></a><tt class="py-lineno">502</tt>  <tt class="py-line">                                  <tt class="py-keyword">for</tt> <tt class="py-name">child</tt> <tt class="py-keyword">in</tt> <tt class="py-name">elem</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L503"></a><tt class="py-lineno">503</tt>  <tt class="py-line"> </tt>
<a name="L504"></a><tt class="py-lineno">504</tt>  <tt class="py-line">    <tt class="py-comment"># Complex types</tt> </tt>
<a name="L505"></a><tt class="py-lineno">505</tt>  <tt class="py-line"> </tt>
<a name="Parser.accession"></a><div id="Parser.accession-def"><a name="L506"></a><tt class="py-lineno">506</tt> <a class="py-toggle" href="#" id="Parser.accession-toggle" onclick="return toggle('Parser.accession');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#accession">accession</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.accession-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.accession-expanded"><a name="L507"></a><tt class="py-lineno">507</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create accession object."""</tt> </tt>
<a name="L508"></a><tt class="py-lineno">508</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-293" class="py-name" targets="Class Bio.Phylo.PhyloXML.Accession=Bio.Phylo.PhyloXML.Accession-class.html"><a title="Bio.Phylo.PhyloXML.Accession" class="py-name" href="#" onclick="return doclink('link-293', 'Accession', 'link-293');">Accession</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt><tt class="py-op">.</tt><tt id="link-294" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-294', 'strip', 'link-65');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-295" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-295', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"source"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L509"></a><tt class="py-lineno">509</tt>  <tt class="py-line"> </tt>
<a name="Parser.annotation"></a><div id="Parser.annotation-def"><a name="L510"></a><tt class="py-lineno">510</tt> <a class="py-toggle" href="#" id="Parser.annotation-toggle" onclick="return toggle('Parser.annotation');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#annotation">annotation</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.annotation-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.annotation-expanded"><a name="L511"></a><tt class="py-lineno">511</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create annotation object."""</tt> </tt>
<a name="L512"></a><tt class="py-lineno">512</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-296" class="py-name" targets="Class Bio.Phylo.PhyloXML.Annotation=Bio.Phylo.PhyloXML.Annotation-class.html"><a title="Bio.Phylo.PhyloXML.Annotation" class="py-name" href="#" onclick="return doclink('link-296', 'Annotation', 'link-296');">Annotation</a></tt><tt class="py-op">(</tt> </tt>
<a name="L513"></a><tt class="py-lineno">513</tt>  <tt class="py-line">            <tt id="link-297" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.desc()=Bio.Phylo.PhyloXMLIO.Writer-class.html#desc"><a title="Bio.Phylo.PhyloXMLIO.Writer.desc" class="py-name" href="#" onclick="return doclink('link-297', 'desc', 'link-297');">desc</a></tt><tt class="py-op">=</tt><tt id="link-298" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._collapse_wspace" class="py-name" href="#" onclick="return doclink('link-298', '_collapse_wspace', 'link-134');">_collapse_wspace</a></tt><tt class="py-op">(</tt><tt id="link-299" class="py-name" targets="Function Bio.Phylo.PhyloXMLIO._get_child_text()=Bio.Phylo.PhyloXMLIO-module.html#_get_child_text"><a title="Bio.Phylo.PhyloXMLIO._get_child_text" class="py-name" href="#" onclick="return doclink('link-299', '_get_child_text', 'link-299');">_get_child_text</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"desc"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L514"></a><tt class="py-lineno">514</tt>  <tt class="py-line">            <tt id="link-300" class="py-name" targets="Variable Bio.Phylo.PhyloXML.Clade.confidence=Bio.Phylo.PhyloXML.Clade-class.html#confidence,Variable Bio.Phylo.PhyloXML.Phylogeny.confidence=Bio.Phylo.PhyloXML.Phylogeny-class.html#confidence,Method Bio.Phylo.PhyloXMLIO.Parser.confidence()=Bio.Phylo.PhyloXMLIO.Parser-class.html#confidence,Method Bio.Phylo.PhyloXMLIO.Writer.confidence()=Bio.Phylo.PhyloXMLIO.Writer-class.html#confidence"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-300', 'confidence', 'link-300');">confidence</a></tt><tt class="py-op">=</tt><tt id="link-301" class="py-name" targets="Function Bio.Phylo.PhyloXMLIO._get_child_as()=Bio.Phylo.PhyloXMLIO-module.html#_get_child_as"><a title="Bio.Phylo.PhyloXMLIO._get_child_as" class="py-name" href="#" onclick="return doclink('link-301', '_get_child_as', 'link-301');">_get_child_as</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"confidence"</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-302" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-302', 'confidence', 'link-300');">confidence</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L515"></a><tt class="py-lineno">515</tt>  <tt class="py-line">            <tt class="py-name">properties</tt><tt class="py-op">=</tt><tt id="link-303" class="py-name" targets="Function Bio.Phylo.PhyloXMLIO._get_children_as()=Bio.Phylo.PhyloXMLIO-module.html#_get_children_as"><a title="Bio.Phylo.PhyloXMLIO._get_children_as" class="py-name" href="#" onclick="return doclink('link-303', '_get_children_as', 'link-303');">_get_children_as</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"property"</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-304" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.property()=Bio.Phylo.PhyloXMLIO.Parser-class.html#property,Method Bio.Phylo.PhyloXMLIO.Writer.property()=Bio.Phylo.PhyloXMLIO.Writer-class.html#property"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-304', 'property', 'link-304');">property</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L516"></a><tt class="py-lineno">516</tt>  <tt class="py-line">            <tt id="link-305" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-305', 'uri', 'link-8');">uri</a></tt><tt class="py-op">=</tt><tt id="link-306" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_child_as" class="py-name" href="#" onclick="return doclink('link-306', '_get_child_as', 'link-301');">_get_child_as</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"uri"</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-307" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-307', 'uri', 'link-8');">uri</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L517"></a><tt class="py-lineno">517</tt>  <tt class="py-line">            <tt class="py-op">**</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">)</tt> </tt>
</div><a name="L518"></a><tt class="py-lineno">518</tt>  <tt class="py-line"> </tt>
<a name="Parser.binary_characters"></a><div id="Parser.binary_characters-def"><a name="L519"></a><tt class="py-lineno">519</tt> <a class="py-toggle" href="#" id="Parser.binary_characters-toggle" onclick="return toggle('Parser.binary_characters');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#binary_characters">binary_characters</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.binary_characters-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.binary_characters-expanded"><a name="L520"></a><tt class="py-lineno">520</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create binary characters object."""</tt> </tt>
<a name="L521"></a><tt class="py-lineno">521</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">bc_getter</tt><tt class="py-op">(</tt><tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L522"></a><tt class="py-lineno">522</tt>  <tt class="py-line">            <tt class="py-docstring">"""Get binary characters from subnodes."""</tt> </tt>
<a name="L523"></a><tt class="py-lineno">523</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-308" class="py-name" targets="Function Bio.Phylo.PhyloXMLIO._get_children_text()=Bio.Phylo.PhyloXMLIO-module.html#_get_children_text"><a title="Bio.Phylo.PhyloXMLIO._get_children_text" class="py-name" href="#" onclick="return doclink('link-308', '_get_children_text', 'link-308');">_get_children_text</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"bc"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L524"></a><tt class="py-lineno">524</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-309" class="py-name" targets="Class Bio.Phylo.PhyloXML.BinaryCharacters=Bio.Phylo.PhyloXML.BinaryCharacters-class.html"><a title="Bio.Phylo.PhyloXML.BinaryCharacters" class="py-name" href="#" onclick="return doclink('link-309', 'BinaryCharacters', 'link-309');">BinaryCharacters</a></tt><tt class="py-op">(</tt> </tt>
<a name="L525"></a><tt class="py-lineno">525</tt>  <tt class="py-line">            <tt id="link-310" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.type()=Bio.Phylo.PhyloXMLIO.Writer-class.html#type"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-310', 'type', 'link-310');">type</a></tt><tt class="py-op">=</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-311" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-311', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"type"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L526"></a><tt class="py-lineno">526</tt>  <tt class="py-line">            <tt class="py-name">gained_count</tt><tt class="py-op">=</tt><tt id="link-312" class="py-name" targets="Function Bio.Phylo.PhyloXMLIO._int()=Bio.Phylo.PhyloXMLIO-module.html#_int"><a title="Bio.Phylo.PhyloXMLIO._int" class="py-name" href="#" onclick="return doclink('link-312', '_int', 'link-312');">_int</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-313" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-313', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"gained_count"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L527"></a><tt class="py-lineno">527</tt>  <tt class="py-line">            <tt class="py-name">lost_count</tt><tt class="py-op">=</tt><tt id="link-314" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._int" class="py-name" href="#" onclick="return doclink('link-314', '_int', 'link-312');">_int</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-315" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-315', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"lost_count"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L528"></a><tt class="py-lineno">528</tt>  <tt class="py-line">            <tt class="py-name">present_count</tt><tt class="py-op">=</tt><tt id="link-316" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._int" class="py-name" href="#" onclick="return doclink('link-316', '_int', 'link-312');">_int</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-317" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-317', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"present_count"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L529"></a><tt class="py-lineno">529</tt>  <tt class="py-line">            <tt class="py-name">absent_count</tt><tt class="py-op">=</tt><tt id="link-318" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._int" class="py-name" href="#" onclick="return doclink('link-318', '_int', 'link-312');">_int</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-319" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-319', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"absent_count"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L530"></a><tt class="py-lineno">530</tt>  <tt class="py-line">            <tt class="py-comment"># Flatten BinaryCharacterList sub-nodes into lists of strings</tt> </tt>
<a name="L531"></a><tt class="py-lineno">531</tt>  <tt class="py-line">            <tt class="py-name">gained</tt><tt class="py-op">=</tt><tt id="link-320" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_child_as" class="py-name" href="#" onclick="return doclink('link-320', '_get_child_as', 'link-301');">_get_child_as</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"gained"</tt><tt class="py-op">,</tt> <tt class="py-name">bc_getter</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L532"></a><tt class="py-lineno">532</tt>  <tt class="py-line">            <tt class="py-name">lost</tt><tt class="py-op">=</tt><tt id="link-321" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_child_as" class="py-name" href="#" onclick="return doclink('link-321', '_get_child_as', 'link-301');">_get_child_as</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"lost"</tt><tt class="py-op">,</tt> <tt class="py-name">bc_getter</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L533"></a><tt class="py-lineno">533</tt>  <tt class="py-line">            <tt class="py-name">present</tt><tt class="py-op">=</tt><tt id="link-322" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_child_as" class="py-name" href="#" onclick="return doclink('link-322', '_get_child_as', 'link-301');">_get_child_as</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"present"</tt><tt class="py-op">,</tt> <tt class="py-name">bc_getter</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L534"></a><tt class="py-lineno">534</tt>  <tt class="py-line">            <tt class="py-name">absent</tt><tt class="py-op">=</tt><tt id="link-323" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_child_as" class="py-name" href="#" onclick="return doclink('link-323', '_get_child_as', 'link-301');">_get_child_as</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"absent"</tt><tt class="py-op">,</tt> <tt class="py-name">bc_getter</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L535"></a><tt class="py-lineno">535</tt>  <tt class="py-line"> </tt>
<a name="Parser.clade_relation"></a><div id="Parser.clade_relation-def"><a name="L536"></a><tt class="py-lineno">536</tt> <a class="py-toggle" href="#" id="Parser.clade_relation-toggle" onclick="return toggle('Parser.clade_relation');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#clade_relation">clade_relation</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.clade_relation-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.clade_relation-expanded"><a name="L537"></a><tt class="py-lineno">537</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create clade relationship object."""</tt> </tt>
<a name="L538"></a><tt class="py-lineno">538</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-324" class="py-name" targets="Class Bio.Phylo.PhyloXML.CladeRelation=Bio.Phylo.PhyloXML.CladeRelation-class.html"><a title="Bio.Phylo.PhyloXML.CladeRelation" class="py-name" href="#" onclick="return doclink('link-324', 'CladeRelation', 'link-324');">CladeRelation</a></tt><tt class="py-op">(</tt> </tt>
<a name="L539"></a><tt class="py-lineno">539</tt>  <tt class="py-line">            <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-325" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-325', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"type"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-326" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-326', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"id_ref_0"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-327" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-327', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"id_ref_1"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L540"></a><tt class="py-lineno">540</tt>  <tt class="py-line">            <tt id="link-328" class="py-name" targets="Method Bio.Nexus.Trees.Tree.distance()=Bio.Nexus.Trees.Tree-class.html#distance,Method Bio.Phylo.BaseTree.TreeMixin.distance()=Bio.Phylo.BaseTree.TreeMixin-class.html#distance"><a title="Bio.Nexus.Trees.Tree.distance
Bio.Phylo.BaseTree.TreeMixin.distance" class="py-name" href="#" onclick="return doclink('link-328', 'distance', 'link-328');">distance</a></tt><tt class="py-op">=</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-329" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-329', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"distance"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L541"></a><tt class="py-lineno">541</tt>  <tt class="py-line">            <tt id="link-330" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-330', 'confidence', 'link-300');">confidence</a></tt><tt class="py-op">=</tt><tt id="link-331" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_child_as" class="py-name" href="#" onclick="return doclink('link-331', '_get_child_as', 'link-301');">_get_child_as</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"confidence"</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-332" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-332', 'confidence', 'link-300');">confidence</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L542"></a><tt class="py-lineno">542</tt>  <tt class="py-line"> </tt>
<a name="Parser.color"></a><div id="Parser.color-def"><a name="L543"></a><tt class="py-lineno">543</tt> <a class="py-toggle" href="#" id="Parser.color-toggle" onclick="return toggle('Parser.color');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#color">color</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.color-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.color-expanded"><a name="L544"></a><tt class="py-lineno">544</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create branch color object."""</tt> </tt>
<a name="L545"></a><tt class="py-lineno">545</tt>  <tt class="py-line">        <tt id="link-333" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.red()=Bio.Phylo.PhyloXMLIO.Writer-class.html#red"><a title="Bio.Phylo.PhyloXMLIO.Writer.red" class="py-name" href="#" onclick="return doclink('link-333', 'red', 'link-333');">red</a></tt><tt class="py-op">,</tt> <tt id="link-334" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.green()=Bio.Phylo.PhyloXMLIO.Writer-class.html#green"><a title="Bio.Phylo.PhyloXMLIO.Writer.green" class="py-name" href="#" onclick="return doclink('link-334', 'green', 'link-334');">green</a></tt><tt class="py-op">,</tt> <tt id="link-335" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.blue()=Bio.Phylo.PhyloXMLIO.Writer-class.html#blue"><a title="Bio.Phylo.PhyloXMLIO.Writer.blue" class="py-name" href="#" onclick="return doclink('link-335', 'blue', 'link-335');">blue</a></tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt id="link-336" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_child_text" class="py-name" href="#" onclick="return doclink('link-336', '_get_child_text', 'link-299');">_get_child_text</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt id="link-337" class="py-name" targets="Variable Bio.Phylo.BaseTree.Clade.color=Bio.Phylo.BaseTree.Clade-class.html#color,Method Bio.Phylo.PhyloXMLIO.Parser.color()=Bio.Phylo.PhyloXMLIO.Parser-class.html#color,Method Bio.Phylo.PhyloXMLIO.Writer.color()=Bio.Phylo.PhyloXMLIO.Writer-class.html#color"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-337', 'color', 'link-337');">color</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-338" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-338', 'color', 'link-337');">color</a></tt> <tt class="py-keyword">in</tt> </tt>
<a name="L546"></a><tt class="py-lineno">546</tt>  <tt class="py-line">                            <tt class="py-op">(</tt><tt class="py-string">"red"</tt><tt class="py-op">,</tt> <tt class="py-string">"green"</tt><tt class="py-op">,</tt> <tt class="py-string">"blue"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L547"></a><tt class="py-lineno">547</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-339" class="py-name" targets="Class Bio.Phylo.BaseTree.BranchColor=Bio.Phylo.BaseTree.BranchColor-class.html,Class Bio.Phylo.PhyloXML.BranchColor=Bio.Phylo.PhyloXML.BranchColor-class.html"><a title="Bio.Phylo.BaseTree.BranchColor
Bio.Phylo.PhyloXML.BranchColor" class="py-name" href="#" onclick="return doclink('link-339', 'BranchColor', 'link-339');">BranchColor</a></tt><tt class="py-op">(</tt><tt id="link-340" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.red" class="py-name" href="#" onclick="return doclink('link-340', 'red', 'link-333');">red</a></tt><tt class="py-op">,</tt> <tt id="link-341" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.green" class="py-name" href="#" onclick="return doclink('link-341', 'green', 'link-334');">green</a></tt><tt class="py-op">,</tt> <tt id="link-342" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.blue" class="py-name" href="#" onclick="return doclink('link-342', 'blue', 'link-335');">blue</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L548"></a><tt class="py-lineno">548</tt>  <tt class="py-line"> </tt>
<a name="Parser.confidence"></a><div id="Parser.confidence-def"><a name="L549"></a><tt class="py-lineno">549</tt> <a class="py-toggle" href="#" id="Parser.confidence-toggle" onclick="return toggle('Parser.confidence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#confidence">confidence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.confidence-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.confidence-expanded"><a name="L550"></a><tt class="py-lineno">550</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create confidence object."""</tt> </tt>
<a name="L551"></a><tt class="py-lineno">551</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-343" class="py-name" targets="Class Bio.Phylo.PhyloXML.Confidence=Bio.Phylo.PhyloXML.Confidence-class.html"><a title="Bio.Phylo.PhyloXML.Confidence" class="py-name" href="#" onclick="return doclink('link-343', 'Confidence', 'link-343');">Confidence</a></tt><tt class="py-op">(</tt> </tt>
<a name="L552"></a><tt class="py-lineno">552</tt>  <tt class="py-line">            <tt id="link-344" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._float" class="py-name" href="#" onclick="return doclink('link-344', '_float', 'link-201');">_float</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L553"></a><tt class="py-lineno">553</tt>  <tt class="py-line">            <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-345" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-345', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"type"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L554"></a><tt class="py-lineno">554</tt>  <tt class="py-line"> </tt>
<a name="Parser.date"></a><div id="Parser.date-def"><a name="L555"></a><tt class="py-lineno">555</tt> <a class="py-toggle" href="#" id="Parser.date-toggle" onclick="return toggle('Parser.date');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#date">date</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.date-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.date-expanded"><a name="L556"></a><tt class="py-lineno">556</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create date object."""</tt> </tt>
<a name="L557"></a><tt class="py-lineno">557</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-346" class="py-name" targets="Class Bio.Phylo.PhyloXML.Date=Bio.Phylo.PhyloXML.Date-class.html"><a title="Bio.Phylo.PhyloXML.Date" class="py-name" href="#" onclick="return doclink('link-346', 'Date', 'link-346');">Date</a></tt><tt class="py-op">(</tt> </tt>
<a name="L558"></a><tt class="py-lineno">558</tt>  <tt class="py-line">            <tt class="py-name">unit</tt><tt class="py-op">=</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-347" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-347', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"unit"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L559"></a><tt class="py-lineno">559</tt>  <tt class="py-line">            <tt id="link-348" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.desc" class="py-name" href="#" onclick="return doclink('link-348', 'desc', 'link-297');">desc</a></tt><tt class="py-op">=</tt><tt id="link-349" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._collapse_wspace" class="py-name" href="#" onclick="return doclink('link-349', '_collapse_wspace', 'link-134');">_collapse_wspace</a></tt><tt class="py-op">(</tt><tt id="link-350" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_child_text" class="py-name" href="#" onclick="return doclink('link-350', '_get_child_text', 'link-299');">_get_child_text</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"desc"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L560"></a><tt class="py-lineno">560</tt>  <tt class="py-line">            <tt id="link-351" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-351', 'value', 'link-287');">value</a></tt><tt class="py-op">=</tt><tt id="link-352" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_child_text" class="py-name" href="#" onclick="return doclink('link-352', '_get_child_text', 'link-299');">_get_child_text</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"value"</tt><tt class="py-op">,</tt> <tt class="py-name">float</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L561"></a><tt class="py-lineno">561</tt>  <tt class="py-line">            <tt id="link-353" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.minimum()=Bio.Phylo.PhyloXMLIO.Writer-class.html#minimum"><a title="Bio.Phylo.PhyloXMLIO.Writer.minimum" class="py-name" href="#" onclick="return doclink('link-353', 'minimum', 'link-353');">minimum</a></tt><tt class="py-op">=</tt><tt id="link-354" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_child_text" class="py-name" href="#" onclick="return doclink('link-354', '_get_child_text', 'link-299');">_get_child_text</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"minimum"</tt><tt class="py-op">,</tt> <tt class="py-name">float</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L562"></a><tt class="py-lineno">562</tt>  <tt class="py-line">            <tt id="link-355" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.maximum()=Bio.Phylo.PhyloXMLIO.Writer-class.html#maximum"><a title="Bio.Phylo.PhyloXMLIO.Writer.maximum" class="py-name" href="#" onclick="return doclink('link-355', 'maximum', 'link-355');">maximum</a></tt><tt class="py-op">=</tt><tt id="link-356" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_child_text" class="py-name" href="#" onclick="return doclink('link-356', '_get_child_text', 'link-299');">_get_child_text</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"maximum"</tt><tt class="py-op">,</tt> <tt class="py-name">float</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L563"></a><tt class="py-lineno">563</tt>  <tt class="py-line">        <tt class="py-op">)</tt> </tt>
</div><a name="L564"></a><tt class="py-lineno">564</tt>  <tt class="py-line"> </tt>
<a name="Parser.distribution"></a><div id="Parser.distribution-def"><a name="L565"></a><tt class="py-lineno">565</tt> <a class="py-toggle" href="#" id="Parser.distribution-toggle" onclick="return toggle('Parser.distribution');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#distribution">distribution</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.distribution-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.distribution-expanded"><a name="L566"></a><tt class="py-lineno">566</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create geographic distribution object."""</tt> </tt>
<a name="L567"></a><tt class="py-lineno">567</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-357" class="py-name" targets="Module Bio.Graphics.Distribution=Bio.Graphics.Distribution-module.html,Class Bio.Phylo.PhyloXML.Distribution=Bio.Phylo.PhyloXML.Distribution-class.html"><a title="Bio.Graphics.Distribution
Bio.Phylo.PhyloXML.Distribution" class="py-name" href="#" onclick="return doclink('link-357', 'Distribution', 'link-357');">Distribution</a></tt><tt class="py-op">(</tt> </tt>
<a name="L568"></a><tt class="py-lineno">568</tt>  <tt class="py-line">            <tt id="link-358" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.desc" class="py-name" href="#" onclick="return doclink('link-358', 'desc', 'link-297');">desc</a></tt><tt class="py-op">=</tt><tt id="link-359" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._collapse_wspace" class="py-name" href="#" onclick="return doclink('link-359', '_collapse_wspace', 'link-134');">_collapse_wspace</a></tt><tt class="py-op">(</tt><tt id="link-360" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_child_text" class="py-name" href="#" onclick="return doclink('link-360', '_get_child_text', 'link-299');">_get_child_text</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"desc"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L569"></a><tt class="py-lineno">569</tt>  <tt class="py-line">            <tt class="py-name">points</tt><tt class="py-op">=</tt><tt id="link-361" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_children_as" class="py-name" href="#" onclick="return doclink('link-361', '_get_children_as', 'link-303');">_get_children_as</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"point"</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-362" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.point()=Bio.Phylo.PhyloXMLIO.Parser-class.html#point,Method Bio.Phylo.PhyloXMLIO.Writer.point()=Bio.Phylo.PhyloXMLIO.Writer-class.html#point"><a title="Bio.Phylo.PhyloXMLIO.Parser.point
Bio.Phylo.PhyloXMLIO.Writer.point" class="py-name" href="#" onclick="return doclink('link-362', 'point', 'link-362');">point</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L570"></a><tt class="py-lineno">570</tt>  <tt class="py-line">            <tt class="py-name">polygons</tt><tt class="py-op">=</tt><tt id="link-363" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_children_as" class="py-name" href="#" onclick="return doclink('link-363', '_get_children_as', 'link-303');">_get_children_as</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"polygon"</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-364" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.polygon()=Bio.Phylo.PhyloXMLIO.Parser-class.html#polygon,Method Bio.Phylo.PhyloXMLIO.Writer.polygon()=Bio.Phylo.PhyloXMLIO.Writer-class.html#polygon"><a title="Bio.Phylo.PhyloXMLIO.Parser.polygon
Bio.Phylo.PhyloXMLIO.Writer.polygon" class="py-name" href="#" onclick="return doclink('link-364', 'polygon', 'link-364');">polygon</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L571"></a><tt class="py-lineno">571</tt>  <tt class="py-line"> </tt>
<a name="Parser.domain"></a><div id="Parser.domain-def"><a name="L572"></a><tt class="py-lineno">572</tt> <a class="py-toggle" href="#" id="Parser.domain-toggle" onclick="return toggle('Parser.domain');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#domain">domain</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.domain-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.domain-expanded"><a name="L573"></a><tt class="py-lineno">573</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create protein domain object."""</tt> </tt>
<a name="L574"></a><tt class="py-lineno">574</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-365" class="py-name" targets="Class Bio.Phylo.PhyloXML.ProteinDomain=Bio.Phylo.PhyloXML.ProteinDomain-class.html"><a title="Bio.Phylo.PhyloXML.ProteinDomain" class="py-name" href="#" onclick="return doclink('link-365', 'ProteinDomain', 'link-365');">ProteinDomain</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt><tt class="py-op">.</tt><tt id="link-366" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-366', 'strip', 'link-65');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L575"></a><tt class="py-lineno">575</tt>  <tt class="py-line">                                <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-367" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-367', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"from"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> </tt>
<a name="L576"></a><tt class="py-lineno">576</tt>  <tt class="py-line">                                <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-368" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-368', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"to"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L577"></a><tt class="py-lineno">577</tt>  <tt class="py-line">                                <tt id="link-369" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-369', 'confidence', 'link-300');">confidence</a></tt><tt class="py-op">=</tt><tt id="link-370" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._float" class="py-name" href="#" onclick="return doclink('link-370', '_float', 'link-201');">_float</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-371" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-371', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"confidence"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L578"></a><tt class="py-lineno">578</tt>  <tt class="py-line">                                <tt id="link-372" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Variable Bio.PDB.Entity.Entity.id=Bio.PDB.Entity.Entity-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-372', 'id', 'link-372');">id</a></tt><tt class="py-op">=</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-373" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-373', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"id"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L579"></a><tt class="py-lineno">579</tt>  <tt class="py-line"> </tt>
<a name="Parser.domain_architecture"></a><div id="Parser.domain_architecture-def"><a name="L580"></a><tt class="py-lineno">580</tt> <a class="py-toggle" href="#" id="Parser.domain_architecture-toggle" onclick="return toggle('Parser.domain_architecture');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#domain_architecture">domain_architecture</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.domain_architecture-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.domain_architecture-expanded"><a name="L581"></a><tt class="py-lineno">581</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create domain architecture object."""</tt> </tt>
<a name="L582"></a><tt class="py-lineno">582</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-374" class="py-name" targets="Class Bio.Phylo.PhyloXML.DomainArchitecture=Bio.Phylo.PhyloXML.DomainArchitecture-class.html"><a title="Bio.Phylo.PhyloXML.DomainArchitecture" class="py-name" href="#" onclick="return doclink('link-374', 'DomainArchitecture', 'link-374');">DomainArchitecture</a></tt><tt class="py-op">(</tt> </tt>
<a name="L583"></a><tt class="py-lineno">583</tt>  <tt class="py-line">            <tt id="link-375" class="py-name" targets="Method Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length()=Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer-class.html#length"><a title="Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-375', 'length', 'link-375');">length</a></tt><tt class="py-op">=</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-376" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-376', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"length"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L584"></a><tt class="py-lineno">584</tt>  <tt class="py-line">            <tt class="py-name">domains</tt><tt class="py-op">=</tt><tt id="link-377" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_children_as" class="py-name" href="#" onclick="return doclink('link-377', '_get_children_as', 'link-303');">_get_children_as</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"domain"</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-378" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.domain()=Bio.Phylo.PhyloXMLIO.Parser-class.html#domain,Method Bio.Phylo.PhyloXMLIO.Writer.domain()=Bio.Phylo.PhyloXMLIO.Writer-class.html#domain"><a title="Bio.Phylo.PhyloXMLIO.Parser.domain
Bio.Phylo.PhyloXMLIO.Writer.domain" class="py-name" href="#" onclick="return doclink('link-378', 'domain', 'link-378');">domain</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L585"></a><tt class="py-lineno">585</tt>  <tt class="py-line"> </tt>
<a name="Parser.events"></a><div id="Parser.events-def"><a name="L586"></a><tt class="py-lineno">586</tt> <a class="py-toggle" href="#" id="Parser.events-toggle" onclick="return toggle('Parser.events');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#events">events</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.events-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.events-expanded"><a name="L587"></a><tt class="py-lineno">587</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create events object."""</tt> </tt>
<a name="L588"></a><tt class="py-lineno">588</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-379" class="py-name" targets="Class Bio.Phylo.PhyloXML.Events=Bio.Phylo.PhyloXML.Events-class.html"><a title="Bio.Phylo.PhyloXML.Events" class="py-name" href="#" onclick="return doclink('link-379', 'Events', 'link-379');">Events</a></tt><tt class="py-op">(</tt> </tt>
<a name="L589"></a><tt class="py-lineno">589</tt>  <tt class="py-line">            <tt id="link-380" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-380', 'type', 'link-310');">type</a></tt><tt class="py-op">=</tt><tt id="link-381" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_child_text" class="py-name" href="#" onclick="return doclink('link-381', '_get_child_text', 'link-299');">_get_child_text</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"type"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L590"></a><tt class="py-lineno">590</tt>  <tt class="py-line">            <tt id="link-382" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.duplications()=Bio.Phylo.PhyloXMLIO.Writer-class.html#duplications"><a title="Bio.Phylo.PhyloXMLIO.Writer.duplications" class="py-name" href="#" onclick="return doclink('link-382', 'duplications', 'link-382');">duplications</a></tt><tt class="py-op">=</tt><tt id="link-383" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_child_text" class="py-name" href="#" onclick="return doclink('link-383', '_get_child_text', 'link-299');">_get_child_text</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"duplications"</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L591"></a><tt class="py-lineno">591</tt>  <tt class="py-line">            <tt id="link-384" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.speciations()=Bio.Phylo.PhyloXMLIO.Writer-class.html#speciations"><a title="Bio.Phylo.PhyloXMLIO.Writer.speciations" class="py-name" href="#" onclick="return doclink('link-384', 'speciations', 'link-384');">speciations</a></tt><tt class="py-op">=</tt><tt id="link-385" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_child_text" class="py-name" href="#" onclick="return doclink('link-385', '_get_child_text', 'link-299');">_get_child_text</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"speciations"</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L592"></a><tt class="py-lineno">592</tt>  <tt class="py-line">            <tt id="link-386" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.losses()=Bio.Phylo.PhyloXMLIO.Writer-class.html#losses"><a title="Bio.Phylo.PhyloXMLIO.Writer.losses" class="py-name" href="#" onclick="return doclink('link-386', 'losses', 'link-386');">losses</a></tt><tt class="py-op">=</tt><tt id="link-387" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_child_text" class="py-name" href="#" onclick="return doclink('link-387', '_get_child_text', 'link-299');">_get_child_text</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"losses"</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L593"></a><tt class="py-lineno">593</tt>  <tt class="py-line">            <tt id="link-388" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-388', 'confidence', 'link-300');">confidence</a></tt><tt class="py-op">=</tt><tt id="link-389" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_child_as" class="py-name" href="#" onclick="return doclink('link-389', '_get_child_as', 'link-301');">_get_child_as</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"confidence"</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-390" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-390', 'confidence', 'link-300');">confidence</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L594"></a><tt class="py-lineno">594</tt>  <tt class="py-line"> </tt>
<a name="Parser.id"></a><div id="Parser.id-def"><a name="L595"></a><tt class="py-lineno">595</tt> <a class="py-toggle" href="#" id="Parser.id-toggle" onclick="return toggle('Parser.id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#id">id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.id-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.id-expanded"><a name="L596"></a><tt class="py-lineno">596</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create identifier object."""</tt> </tt>
<a name="L597"></a><tt class="py-lineno">597</tt>  <tt class="py-line">        <tt class="py-name">provider</tt> <tt class="py-op">=</tt> <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-391" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-391', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"provider"</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-392" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-392', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"type"</tt><tt class="py-op">)</tt> </tt>
<a name="L598"></a><tt class="py-lineno">598</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-393" class="py-name"><a title="Bio.Phylo.PhyloXML.Id" class="py-name" href="#" onclick="return doclink('link-393', 'Id', 'link-213');">Id</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt><tt class="py-op">.</tt><tt id="link-394" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-394', 'strip', 'link-65');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">provider</tt><tt class="py-op">)</tt> </tt>
</div><a name="L599"></a><tt class="py-lineno">599</tt>  <tt class="py-line"> </tt>
<a name="Parser.mol_seq"></a><div id="Parser.mol_seq-def"><a name="L600"></a><tt class="py-lineno">600</tt> <a class="py-toggle" href="#" id="Parser.mol_seq-toggle" onclick="return toggle('Parser.mol_seq');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#mol_seq">mol_seq</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.mol_seq-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.mol_seq-expanded"><a name="L601"></a><tt class="py-lineno">601</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create molecular sequence object."""</tt> </tt>
<a name="L602"></a><tt class="py-lineno">602</tt>  <tt class="py-line">        <tt class="py-name">is_aligned</tt> <tt class="py-op">=</tt> <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-395" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-395', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"is_aligned"</tt><tt class="py-op">)</tt> </tt>
<a name="L603"></a><tt class="py-lineno">603</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">is_aligned</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L604"></a><tt class="py-lineno">604</tt>  <tt class="py-line">            <tt class="py-name">is_aligned</tt> <tt class="py-op">=</tt> <tt id="link-396" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._str2bool" class="py-name" href="#" onclick="return doclink('link-396', '_str2bool', 'link-76');">_str2bool</a></tt><tt class="py-op">(</tt><tt class="py-name">is_aligned</tt><tt class="py-op">)</tt> </tt>
<a name="L605"></a><tt class="py-lineno">605</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-397" class="py-name" targets="Class Bio.Phylo.PhyloXML.MolSeq=Bio.Phylo.PhyloXML.MolSeq-class.html"><a title="Bio.Phylo.PhyloXML.MolSeq" class="py-name" href="#" onclick="return doclink('link-397', 'MolSeq', 'link-397');">MolSeq</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt><tt class="py-op">.</tt><tt id="link-398" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-398', 'strip', 'link-65');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">is_aligned</tt><tt class="py-op">=</tt><tt class="py-name">is_aligned</tt><tt class="py-op">)</tt> </tt>
</div><a name="L606"></a><tt class="py-lineno">606</tt>  <tt class="py-line"> </tt>
<a name="Parser.point"></a><div id="Parser.point-def"><a name="L607"></a><tt class="py-lineno">607</tt> <a class="py-toggle" href="#" id="Parser.point-toggle" onclick="return toggle('Parser.point');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#point">point</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.point-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.point-expanded"><a name="L608"></a><tt class="py-lineno">608</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create point object, coordinates of a point."""</tt> </tt>
<a name="L609"></a><tt class="py-lineno">609</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-399" class="py-name" targets="Class Bio.Phylo.PhyloXML.Point=Bio.Phylo.PhyloXML.Point-class.html"><a title="Bio.Phylo.PhyloXML.Point" class="py-name" href="#" onclick="return doclink('link-399', 'Point', 'link-399');">Point</a></tt><tt class="py-op">(</tt> </tt>
<a name="L610"></a><tt class="py-lineno">610</tt>  <tt class="py-line">            <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-400" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-400', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"geodetic_datum"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L611"></a><tt class="py-lineno">611</tt>  <tt class="py-line">            <tt id="link-401" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_child_text" class="py-name" href="#" onclick="return doclink('link-401', '_get_child_text', 'link-299');">_get_child_text</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"lat"</tt><tt class="py-op">,</tt> <tt class="py-name">float</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L612"></a><tt class="py-lineno">612</tt>  <tt class="py-line">            <tt id="link-402" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_child_text" class="py-name" href="#" onclick="return doclink('link-402', '_get_child_text', 'link-299');">_get_child_text</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"long"</tt><tt class="py-op">,</tt> <tt class="py-name">float</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L613"></a><tt class="py-lineno">613</tt>  <tt class="py-line">            <tt id="link-403" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.alt()=Bio.Phylo.PhyloXMLIO.Writer-class.html#alt"><a title="Bio.Phylo.PhyloXMLIO.Writer.alt" class="py-name" href="#" onclick="return doclink('link-403', 'alt', 'link-403');">alt</a></tt><tt class="py-op">=</tt><tt id="link-404" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_child_text" class="py-name" href="#" onclick="return doclink('link-404', '_get_child_text', 'link-299');">_get_child_text</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"alt"</tt><tt class="py-op">,</tt> <tt class="py-name">float</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L614"></a><tt class="py-lineno">614</tt>  <tt class="py-line">            <tt class="py-name">alt_unit</tt><tt class="py-op">=</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-405" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-405', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"alt_unit"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L615"></a><tt class="py-lineno">615</tt>  <tt class="py-line"> </tt>
<a name="Parser.polygon"></a><div id="Parser.polygon-def"><a name="L616"></a><tt class="py-lineno">616</tt> <a class="py-toggle" href="#" id="Parser.polygon-toggle" onclick="return toggle('Parser.polygon');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#polygon">polygon</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.polygon-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.polygon-expanded"><a name="L617"></a><tt class="py-lineno">617</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create polygon object, list of points."""</tt> </tt>
<a name="L618"></a><tt class="py-lineno">618</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-406" class="py-name" targets="Class Bio.Phylo.PhyloXML.Polygon=Bio.Phylo.PhyloXML.Polygon-class.html"><a title="Bio.Phylo.PhyloXML.Polygon" class="py-name" href="#" onclick="return doclink('link-406', 'Polygon', 'link-406');">Polygon</a></tt><tt class="py-op">(</tt> </tt>
<a name="L619"></a><tt class="py-lineno">619</tt>  <tt class="py-line">            <tt class="py-name">points</tt><tt class="py-op">=</tt><tt id="link-407" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_children_as" class="py-name" href="#" onclick="return doclink('link-407', '_get_children_as', 'link-303');">_get_children_as</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"point"</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-408" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.point
Bio.Phylo.PhyloXMLIO.Writer.point" class="py-name" href="#" onclick="return doclink('link-408', 'point', 'link-362');">point</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L620"></a><tt class="py-lineno">620</tt>  <tt class="py-line"> </tt>
<a name="Parser.property"></a><div id="Parser.property-def"><a name="L621"></a><tt class="py-lineno">621</tt> <a class="py-toggle" href="#" id="Parser.property-toggle" onclick="return toggle('Parser.property');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#property">property</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.property-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.property-expanded"><a name="L622"></a><tt class="py-lineno">622</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create properties from external resources."""</tt> </tt>
<a name="L623"></a><tt class="py-lineno">623</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-409" class="py-name" targets="Class Bio.Phylo.PhyloXML.Property=Bio.Phylo.PhyloXML.Property-class.html"><a title="Bio.Phylo.PhyloXML.Property" class="py-name" href="#" onclick="return doclink('link-409', 'Property', 'link-409');">Property</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt><tt class="py-op">.</tt><tt id="link-410" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-410', 'strip', 'link-65');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L624"></a><tt class="py-lineno">624</tt>  <tt class="py-line">                           <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-411" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-411', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"ref"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L625"></a><tt class="py-lineno">625</tt>  <tt class="py-line">                           <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-412" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-412', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"applies_to"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L626"></a><tt class="py-lineno">626</tt>  <tt class="py-line">                           <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-413" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-413', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"datatype"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L627"></a><tt class="py-lineno">627</tt>  <tt class="py-line">                           <tt class="py-name">unit</tt><tt class="py-op">=</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-414" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-414', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"unit"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L628"></a><tt class="py-lineno">628</tt>  <tt class="py-line">                           <tt class="py-name">id_ref</tt><tt class="py-op">=</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-415" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-415', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"id_ref"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L629"></a><tt class="py-lineno">629</tt>  <tt class="py-line"> </tt>
<a name="Parser.reference"></a><div id="Parser.reference-def"><a name="L630"></a><tt class="py-lineno">630</tt> <a class="py-toggle" href="#" id="Parser.reference-toggle" onclick="return toggle('Parser.reference');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#reference">reference</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.reference-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.reference-expanded"><a name="L631"></a><tt class="py-lineno">631</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create literature reference object."""</tt> </tt>
<a name="L632"></a><tt class="py-lineno">632</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-416" class="py-name" targets="Class Bio.ExPASy.Prodoc.Reference=Bio.ExPASy.Prodoc.Reference-class.html,Class Bio.GenBank.Record.Reference=Bio.GenBank.Record.Reference-class.html,Class Bio.Phylo.PhyloXML.Reference=Bio.Phylo.PhyloXML.Reference-class.html,Class Bio.SeqFeature.Reference=Bio.SeqFeature.Reference-class.html,Class Bio.SwissProt.Reference=Bio.SwissProt.Reference-class.html"><a title="Bio.ExPASy.Prodoc.Reference
Bio.GenBank.Record.Reference
Bio.Phylo.PhyloXML.Reference
Bio.SeqFeature.Reference
Bio.SwissProt.Reference" class="py-name" href="#" onclick="return doclink('link-416', 'Reference', 'link-416');">Reference</a></tt><tt class="py-op">(</tt> </tt>
<a name="L633"></a><tt class="py-lineno">633</tt>  <tt class="py-line">            <tt class="py-name">doi</tt><tt class="py-op">=</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-417" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-417', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"doi"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L634"></a><tt class="py-lineno">634</tt>  <tt class="py-line">            <tt id="link-418" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.desc" class="py-name" href="#" onclick="return doclink('link-418', 'desc', 'link-297');">desc</a></tt><tt class="py-op">=</tt><tt id="link-419" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_child_text" class="py-name" href="#" onclick="return doclink('link-419', '_get_child_text', 'link-299');">_get_child_text</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"desc"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L635"></a><tt class="py-lineno">635</tt>  <tt class="py-line"> </tt>
<a name="Parser.sequence_relation"></a><div id="Parser.sequence_relation-def"><a name="L636"></a><tt class="py-lineno">636</tt> <a class="py-toggle" href="#" id="Parser.sequence_relation-toggle" onclick="return toggle('Parser.sequence_relation');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#sequence_relation">sequence_relation</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.sequence_relation-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.sequence_relation-expanded"><a name="L637"></a><tt class="py-lineno">637</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create sequence relationship object, relationship between two sequences."""</tt> </tt>
<a name="L638"></a><tt class="py-lineno">638</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-420" class="py-name" targets="Class Bio.Phylo.PhyloXML.SequenceRelation=Bio.Phylo.PhyloXML.SequenceRelation-class.html"><a title="Bio.Phylo.PhyloXML.SequenceRelation" class="py-name" href="#" onclick="return doclink('link-420', 'SequenceRelation', 'link-420');">SequenceRelation</a></tt><tt class="py-op">(</tt> </tt>
<a name="L639"></a><tt class="py-lineno">639</tt>  <tt class="py-line">            <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-421" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-421', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"type"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-422" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-422', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"id_ref_0"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-423" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-423', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"id_ref_1"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L640"></a><tt class="py-lineno">640</tt>  <tt class="py-line">            <tt id="link-424" class="py-name"><a title="Bio.Nexus.Trees.Tree.distance
Bio.Phylo.BaseTree.TreeMixin.distance" class="py-name" href="#" onclick="return doclink('link-424', 'distance', 'link-328');">distance</a></tt><tt class="py-op">=</tt><tt id="link-425" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._float" class="py-name" href="#" onclick="return doclink('link-425', '_float', 'link-201');">_float</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-426" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-426', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"distance"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L641"></a><tt class="py-lineno">641</tt>  <tt class="py-line">            <tt id="link-427" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-427', 'confidence', 'link-300');">confidence</a></tt><tt class="py-op">=</tt><tt id="link-428" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._get_child_as" class="py-name" href="#" onclick="return doclink('link-428', '_get_child_as', 'link-301');">_get_child_as</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-string">"confidence"</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-429" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-429', 'confidence', 'link-300');">confidence</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L642"></a><tt class="py-lineno">642</tt>  <tt class="py-line"> </tt>
<a name="Parser.uri"></a><div id="Parser.uri-def"><a name="L643"></a><tt class="py-lineno">643</tt> <a class="py-toggle" href="#" id="Parser.uri-toggle" onclick="return toggle('Parser.uri');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Parser-class.html#uri">uri</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.uri-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.uri-expanded"><a name="L644"></a><tt class="py-lineno">644</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create uri object, expected to be a url."""</tt> </tt>
<a name="L645"></a><tt class="py-lineno">645</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-430" class="py-name" targets="Class Bio.Phylo.PhyloXML.Uri=Bio.Phylo.PhyloXML.Uri-class.html"><a title="Bio.Phylo.PhyloXML.Uri" class="py-name" href="#" onclick="return doclink('link-430', 'Uri', 'link-430');">Uri</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt><tt class="py-op">.</tt><tt id="link-431" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-431', 'strip', 'link-65');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L646"></a><tt class="py-lineno">646</tt>  <tt class="py-line">                      <tt id="link-432" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.desc" class="py-name" href="#" onclick="return doclink('link-432', 'desc', 'link-297');">desc</a></tt><tt class="py-op">=</tt><tt id="link-433" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._collapse_wspace" class="py-name" href="#" onclick="return doclink('link-433', '_collapse_wspace', 'link-134');">_collapse_wspace</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-434" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-434', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"desc"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L647"></a><tt class="py-lineno">647</tt>  <tt class="py-line">                      <tt id="link-435" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-435', 'type', 'link-310');">type</a></tt><tt class="py-op">=</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-436" class="py-name"><a title="Bio.Align.substitution_matrices.Array.get
Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-436', 'get', 'link-215');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"type"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L648"></a><tt class="py-lineno">648</tt>  <tt class="py-line"> </tt>
<a name="L649"></a><tt class="py-lineno">649</tt>  <tt class="py-line"> </tt>
<a name="L650"></a><tt class="py-lineno">650</tt>  <tt class="py-line"><tt class="py-comment"># ---------------------------------------------------------</tt> </tt>
<a name="L651"></a><tt class="py-lineno">651</tt>  <tt class="py-line"><tt class="py-comment"># OUTPUT</tt> </tt>
<a name="L652"></a><tt class="py-lineno">652</tt>  <tt class="py-line"><tt class="py-comment"># ---------------------------------------------------------</tt> </tt>
<a name="L653"></a><tt class="py-lineno">653</tt>  <tt class="py-line"> </tt>
<a name="_serialize"></a><div id="_serialize-def"><a name="L654"></a><tt class="py-lineno">654</tt> <a class="py-toggle" href="#" id="_serialize-toggle" onclick="return toggle('_serialize');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO-module.html#_serialize">_serialize</a><tt class="py-op">(</tt><tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_serialize-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_serialize-expanded"><a name="L655"></a><tt class="py-lineno">655</tt>  <tt class="py-line">    <tt class="py-docstring">"""Convert a Python primitive to a phyloXML-compatible Unicode string (PRIVATE)."""</tt> </tt>
<a name="L656"></a><tt class="py-lineno">656</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-437" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-437', 'value', 'link-287');">value</a></tt><tt class="py-op">,</tt> <tt class="py-name">float</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L657"></a><tt class="py-lineno">657</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">unicode</tt><tt class="py-op">(</tt><tt id="link-438" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-438', 'value', 'link-287');">value</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-439" class="py-name" targets="Method Bio.Seq.Seq.upper()=Bio.Seq.Seq-class.html#upper,Method Bio.Seq.UnknownSeq.upper()=Bio.Seq.UnknownSeq-class.html#upper,Method Bio.SeqRecord.SeqRecord.upper()=Bio.SeqRecord.SeqRecord-class.html#upper"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-439', 'upper', 'link-439');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L658"></a><tt class="py-lineno">658</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-440" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-440', 'value', 'link-287');">value</a></tt><tt class="py-op">,</tt> <tt class="py-name">bool</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L659"></a><tt class="py-lineno">659</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">unicode</tt><tt class="py-op">(</tt><tt id="link-441" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-441', 'value', 'link-287');">value</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-442" class="py-name" targets="Method Bio.Seq.Seq.lower()=Bio.Seq.Seq-class.html#lower,Method Bio.Seq.UnknownSeq.lower()=Bio.Seq.UnknownSeq-class.html#lower,Method Bio.SeqRecord.SeqRecord.lower()=Bio.SeqRecord.SeqRecord-class.html#lower"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-442', 'lower', 'link-442');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L660"></a><tt class="py-lineno">660</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">unicode</tt><tt class="py-op">(</tt><tt id="link-443" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-443', 'value', 'link-287');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L661"></a><tt class="py-lineno">661</tt>  <tt class="py-line"> </tt>
<a name="L662"></a><tt class="py-lineno">662</tt>  <tt class="py-line"> </tt>
<a name="_clean_attrib"></a><div id="_clean_attrib-def"><a name="L663"></a><tt class="py-lineno">663</tt> <a class="py-toggle" href="#" id="_clean_attrib-toggle" onclick="return toggle('_clean_attrib');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO-module.html#_clean_attrib">_clean_attrib</a><tt class="py-op">(</tt><tt class="py-param">obj</tt><tt class="py-op">,</tt> <tt class="py-param">attrs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_clean_attrib-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_clean_attrib-expanded"><a name="L664"></a><tt class="py-lineno">664</tt>  <tt class="py-line">    <tt class="py-docstring">"""Create a dictionary from an object's specified, non-None attributes (PRIVATE)."""</tt> </tt>
<a name="L665"></a><tt class="py-lineno">665</tt>  <tt class="py-line">    <tt class="py-name">out</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L666"></a><tt class="py-lineno">666</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">attrs</tt><tt class="py-op">:</tt> </tt>
<a name="L667"></a><tt class="py-lineno">667</tt>  <tt class="py-line">        <tt class="py-name">val</tt> <tt class="py-op">=</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">,</tt> <tt class="py-name">key</tt><tt class="py-op">)</tt> </tt>
<a name="L668"></a><tt class="py-lineno">668</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">val</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L669"></a><tt class="py-lineno">669</tt>  <tt class="py-line">            <tt class="py-name">out</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-444" class="py-name" targets="Function Bio.Phylo.PhyloXMLIO._serialize()=Bio.Phylo.PhyloXMLIO-module.html#_serialize"><a title="Bio.Phylo.PhyloXMLIO._serialize" class="py-name" href="#" onclick="return doclink('link-444', '_serialize', 'link-444');">_serialize</a></tt><tt class="py-op">(</tt><tt class="py-name">val</tt><tt class="py-op">)</tt> </tt>
<a name="L670"></a><tt class="py-lineno">670</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">out</tt> </tt>
</div><a name="L671"></a><tt class="py-lineno">671</tt>  <tt class="py-line"> </tt>
<a name="L672"></a><tt class="py-lineno">672</tt>  <tt class="py-line"> </tt>
<a name="_handle_complex"></a><div id="_handle_complex-def"><a name="L673"></a><tt class="py-lineno">673</tt> <a class="py-toggle" href="#" id="_handle_complex-toggle" onclick="return toggle('_handle_complex');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO-module.html#_handle_complex">_handle_complex</a><tt class="py-op">(</tt><tt class="py-param">tag</tt><tt class="py-op">,</tt> <tt class="py-param">attribs</tt><tt class="py-op">,</tt> <tt class="py-param">subnodes</tt><tt class="py-op">,</tt> <tt class="py-param">has_text</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_handle_complex-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_handle_complex-expanded"><a name="L674"></a><tt class="py-lineno">674</tt>  <tt class="py-line">    <tt class="py-docstring">"""Handle to serialize nodes with subnodes (PRIVATE)."""</tt> </tt>
<a name="L675"></a><tt class="py-lineno">675</tt>  <tt class="py-line">    <tt class="py-keyword">def</tt> <tt class="py-def-name">wrapped</tt><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">obj</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L676"></a><tt class="py-lineno">676</tt>  <tt class="py-line">        <tt class="py-docstring">"""Wrap nodes and subnodes as elements."""</tt> </tt>
<a name="L677"></a><tt class="py-lineno">677</tt>  <tt class="py-line">        <tt class="py-name">elem</tt> <tt class="py-op">=</tt> <tt class="py-name">ElementTree</tt><tt class="py-op">.</tt><tt class="py-name">Element</tt><tt class="py-op">(</tt><tt id="link-445" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-445', 'tag', 'link-44');">tag</a></tt><tt class="py-op">,</tt> <tt id="link-446" class="py-name" targets="Function Bio.Phylo.PhyloXMLIO._clean_attrib()=Bio.Phylo.PhyloXMLIO-module.html#_clean_attrib"><a title="Bio.Phylo.PhyloXMLIO._clean_attrib" class="py-name" href="#" onclick="return doclink('link-446', '_clean_attrib', 'link-446');">_clean_attrib</a></tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">,</tt> <tt class="py-name">attribs</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L678"></a><tt class="py-lineno">678</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">subn</tt> <tt class="py-keyword">in</tt> <tt class="py-name">subnodes</tt><tt class="py-op">:</tt> </tt>
<a name="L679"></a><tt class="py-lineno">679</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">subn</tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L680"></a><tt class="py-lineno">680</tt>  <tt class="py-line">                <tt class="py-comment"># singular object: method and attribute names are the same</tt> </tt>
<a name="L681"></a><tt class="py-lineno">681</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">,</tt> <tt class="py-name">subn</tt><tt class="py-op">)</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L682"></a><tt class="py-lineno">682</tt>  <tt class="py-line">                    <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-447" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-447', 'append', 'link-95');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-name">subn</tt><tt class="py-op">)</tt><tt class="py-op">(</tt><tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">,</tt> <tt class="py-name">subn</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L683"></a><tt class="py-lineno">683</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L684"></a><tt class="py-lineno">684</tt>  <tt class="py-line">                <tt class="py-comment"># list: singular method, pluralized attribute name</tt> </tt>
<a name="L685"></a><tt class="py-lineno">685</tt>  <tt class="py-line">                <tt class="py-name">method</tt><tt class="py-op">,</tt> <tt class="py-name">plural</tt> <tt class="py-op">=</tt> <tt class="py-name">subn</tt> </tt>
<a name="L686"></a><tt class="py-lineno">686</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">item</tt> <tt class="py-keyword">in</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">,</tt> <tt class="py-name">plural</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L687"></a><tt class="py-lineno">687</tt>  <tt class="py-line">                    <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-448" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-448', 'append', 'link-95');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-name">method</tt><tt class="py-op">)</tt><tt class="py-op">(</tt><tt class="py-name">item</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L688"></a><tt class="py-lineno">688</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">has_text</tt><tt class="py-op">:</tt> </tt>
<a name="L689"></a><tt class="py-lineno">689</tt>  <tt class="py-line">            <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt id="link-449" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._serialize" class="py-name" href="#" onclick="return doclink('link-449', '_serialize', 'link-444');">_serialize</a></tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">.</tt><tt id="link-450" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-450', 'value', 'link-287');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L690"></a><tt class="py-lineno">690</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">elem</tt> </tt>
</div><a name="L691"></a><tt class="py-lineno">691</tt>  <tt class="py-line">    <tt class="py-name">wrapped</tt><tt class="py-op">.</tt><tt id="link-451" class="py-name" targets="Variable Bio.Cluster.Node.__doc__=Bio.Cluster.Node-class.html#__doc__"><a title="Bio.Cluster.Node.__doc__" class="py-name" href="#" onclick="return doclink('link-451', '__doc__', 'link-451');">__doc__</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"Serialize a %s and its subnodes, in order."</tt> <tt class="py-op">%</tt> <tt id="link-452" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-452', 'tag', 'link-44');">tag</a></tt> </tt>
<a name="L692"></a><tt class="py-lineno">692</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">wrapped</tt> </tt>
</div><a name="L693"></a><tt class="py-lineno">693</tt>  <tt class="py-line"> </tt>
<a name="L694"></a><tt class="py-lineno">694</tt>  <tt class="py-line"> </tt>
<a name="_handle_simple"></a><div id="_handle_simple-def"><a name="L695"></a><tt class="py-lineno">695</tt> <a class="py-toggle" href="#" id="_handle_simple-toggle" onclick="return toggle('_handle_simple');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO-module.html#_handle_simple">_handle_simple</a><tt class="py-op">(</tt><tt class="py-param">tag</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_handle_simple-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_handle_simple-expanded"><a name="L696"></a><tt class="py-lineno">696</tt>  <tt class="py-line">    <tt class="py-docstring">"""Handle to serialize simple nodes (PRIVATE)."""</tt> </tt>
<a name="L697"></a><tt class="py-lineno">697</tt>  <tt class="py-line">    <tt class="py-keyword">def</tt> <tt class="py-def-name">wrapped</tt><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">obj</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L698"></a><tt class="py-lineno">698</tt>  <tt class="py-line">        <tt class="py-docstring">"""Wrap node as element."""</tt> </tt>
<a name="L699"></a><tt class="py-lineno">699</tt>  <tt class="py-line">        <tt class="py-name">elem</tt> <tt class="py-op">=</tt> <tt class="py-name">ElementTree</tt><tt class="py-op">.</tt><tt class="py-name">Element</tt><tt class="py-op">(</tt><tt id="link-453" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-453', 'tag', 'link-44');">tag</a></tt><tt class="py-op">)</tt> </tt>
<a name="L700"></a><tt class="py-lineno">700</tt>  <tt class="py-line">        <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt id="link-454" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._serialize" class="py-name" href="#" onclick="return doclink('link-454', '_serialize', 'link-444');">_serialize</a></tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">)</tt> </tt>
<a name="L701"></a><tt class="py-lineno">701</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">elem</tt> </tt>
</div><a name="L702"></a><tt class="py-lineno">702</tt>  <tt class="py-line">    <tt class="py-name">wrapped</tt><tt class="py-op">.</tt><tt id="link-455" class="py-name"><a title="Bio.Cluster.Node.__doc__" class="py-name" href="#" onclick="return doclink('link-455', '__doc__', 'link-451');">__doc__</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"Serialize a simple %s node."</tt> <tt class="py-op">%</tt> <tt id="link-456" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-456', 'tag', 'link-44');">tag</a></tt> </tt>
<a name="L703"></a><tt class="py-lineno">703</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">wrapped</tt> </tt>
</div><a name="L704"></a><tt class="py-lineno">704</tt>  <tt class="py-line"> </tt>
<a name="L705"></a><tt class="py-lineno">705</tt>  <tt class="py-line"> </tt>
<a name="Writer"></a><div id="Writer-def"><a name="L706"></a><tt class="py-lineno">706</tt> <a class="py-toggle" href="#" id="Writer-toggle" onclick="return toggle('Writer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Writer-class.html">Writer</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Writer-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Writer-expanded"><a name="L707"></a><tt class="py-lineno">707</tt>  <tt class="py-line">    <tt class="py-docstring">"""Methods for serializing a PhyloXML object to XML."""</tt> </tt>
<a name="L708"></a><tt class="py-lineno">708</tt>  <tt class="py-line"> </tt>
<a name="Writer.__init__"></a><div id="Writer.__init__-def"><a name="L709"></a><tt class="py-lineno">709</tt> <a class="py-toggle" href="#" id="Writer.__init__-toggle" onclick="return toggle('Writer.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Writer-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">phyloxml</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Writer.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Writer.__init__-expanded"><a name="L710"></a><tt class="py-lineno">710</tt>  <tt class="py-line">        <tt class="py-docstring">"""Build an ElementTree from a PhyloXML object."""</tt> </tt>
<a name="L711"></a><tt class="py-lineno">711</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-457" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.phyloxml" class="py-name" href="#" onclick="return doclink('link-457', 'phyloxml', 'link-84');">phyloxml</a></tt><tt class="py-op">,</tt> <tt class="py-name">PX</tt><tt class="py-op">.</tt><tt id="link-458" class="py-name"><a title="Bio.Phylo.PhyloXML.Phyloxml" class="py-name" href="#" onclick="return doclink('link-458', 'Phyloxml', 'link-36');">Phyloxml</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"Not a Phyloxml object"</tt> </tt>
<a name="L712"></a><tt class="py-lineno">712</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-459" class="py-name" targets="Method Bio.Nexus.Nexus.Nexus._tree()=Bio.Nexus.Nexus.Nexus-class.html#_tree"><a title="Bio.Nexus.Nexus.Nexus._tree" class="py-name" href="#" onclick="return doclink('link-459', '_tree', 'link-459');">_tree</a></tt> <tt class="py-op">=</tt> <tt class="py-name">ElementTree</tt><tt class="py-op">.</tt><tt class="py-name">ElementTree</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-460" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.phyloxml" class="py-name" href="#" onclick="return doclink('link-460', 'phyloxml', 'link-84');">phyloxml</a></tt><tt class="py-op">(</tt><tt id="link-461" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.phyloxml" class="py-name" href="#" onclick="return doclink('link-461', 'phyloxml', 'link-84');">phyloxml</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L713"></a><tt class="py-lineno">713</tt>  <tt class="py-line"> </tt>
<a name="Writer.write"></a><div id="Writer.write-def"><a name="L714"></a><tt class="py-lineno">714</tt> <a class="py-toggle" href="#" id="Writer.write-toggle" onclick="return toggle('Writer.write');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Writer-class.html#write">write</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">file</tt><tt class="py-op">,</tt> <tt class="py-param">encoding</tt><tt class="py-op">=</tt><tt id="link-462" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.DEFAULT_ENCODING" class="py-name" href="#" onclick="return doclink('link-462', 'DEFAULT_ENCODING', 'link-16');">DEFAULT_ENCODING</a></tt><tt class="py-op">,</tt> <tt class="py-param">indent</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Writer.write-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Writer.write-expanded"><a name="L715"></a><tt class="py-lineno">715</tt>  <tt class="py-line">        <tt class="py-docstring">"""Write PhyloXML to a file."""</tt> </tt>
<a name="L716"></a><tt class="py-lineno">716</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">indent</tt><tt class="py-op">:</tt> </tt>
<a name="L717"></a><tt class="py-lineno">717</tt>  <tt class="py-line">            <tt id="link-463" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._indent" class="py-name" href="#" onclick="return doclink('link-463', '_indent', 'link-67');">_indent</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-464" class="py-name"><a title="Bio.Nexus.Nexus.Nexus._tree" class="py-name" href="#" onclick="return doclink('link-464', '_tree', 'link-459');">_tree</a></tt><tt class="py-op">.</tt><tt class="py-name">getroot</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L718"></a><tt class="py-lineno">718</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-465" class="py-name"><a title="Bio.Nexus.Nexus.Nexus._tree" class="py-name" href="#" onclick="return doclink('link-465', '_tree', 'link-459');">_tree</a></tt><tt class="py-op">.</tt><tt id="link-466" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.clusterbuster.write
Bio.motifs.jaspar.write
Bio.motifs.pfm.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-466', 'write', 'link-43');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">file</tt><tt class="py-op">,</tt> <tt class="py-name">encoding</tt><tt class="py-op">)</tt> </tt>
<a name="L719"></a><tt class="py-lineno">719</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-467" class="py-name"><a title="Bio.Nexus.Nexus.Nexus._tree" class="py-name" href="#" onclick="return doclink('link-467', '_tree', 'link-459');">_tree</a></tt><tt class="py-op">.</tt><tt class="py-name">getroot</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L720"></a><tt class="py-lineno">720</tt>  <tt class="py-line"> </tt>
<a name="L721"></a><tt class="py-lineno">721</tt>  <tt class="py-line">    <tt class="py-comment"># Convert classes to ETree elements</tt> </tt>
<a name="L722"></a><tt class="py-lineno">722</tt>  <tt class="py-line"> </tt>
<a name="Writer.phyloxml"></a><div id="Writer.phyloxml-def"><a name="L723"></a><tt class="py-lineno">723</tt> <a class="py-toggle" href="#" id="Writer.phyloxml-toggle" onclick="return toggle('Writer.phyloxml');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Writer-class.html#phyloxml">phyloxml</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">obj</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Writer.phyloxml-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Writer.phyloxml-expanded"><a name="L724"></a><tt class="py-lineno">724</tt>  <tt class="py-line">        <tt class="py-docstring">"""Convert phyloxml to Etree element."""</tt> </tt>
<a name="L725"></a><tt class="py-lineno">725</tt>  <tt class="py-line">        <tt class="py-name">elem</tt> <tt class="py-op">=</tt> <tt class="py-name">ElementTree</tt><tt class="py-op">.</tt><tt class="py-name">Element</tt><tt class="py-op">(</tt><tt class="py-string">"phyloxml"</tt><tt class="py-op">,</tt> <tt class="py-name">obj</tt><tt class="py-op">.</tt><tt class="py-name">attributes</tt><tt class="py-op">)</tt>  <tt class="py-comment"># Namespaces</tt> </tt>
<a name="L726"></a><tt class="py-lineno">726</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">tree</tt> <tt class="py-keyword">in</tt> <tt class="py-name">obj</tt><tt class="py-op">.</tt><tt class="py-name">phylogenies</tt><tt class="py-op">:</tt> </tt>
<a name="L727"></a><tt class="py-lineno">727</tt>  <tt class="py-line">            <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-468" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-468', 'append', 'link-95');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-469" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.phylogeny" class="py-name" href="#" onclick="return doclink('link-469', 'phylogeny', 'link-92');">phylogeny</a></tt><tt class="py-op">(</tt><tt class="py-name">tree</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L728"></a><tt class="py-lineno">728</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">otr</tt> <tt class="py-keyword">in</tt> <tt class="py-name">obj</tt><tt class="py-op">.</tt><tt id="link-470" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-470', 'other', 'link-98');">other</a></tt><tt class="py-op">:</tt> </tt>
<a name="L729"></a><tt class="py-lineno">729</tt>  <tt class="py-line">            <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-471" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-471', 'append', 'link-95');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-472" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-472', 'other', 'link-98');">other</a></tt><tt class="py-op">(</tt><tt class="py-name">otr</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L730"></a><tt class="py-lineno">730</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">elem</tt> </tt>
</div><a name="L731"></a><tt class="py-lineno">731</tt>  <tt class="py-line"> </tt>
<a name="Writer.other"></a><div id="Writer.other-def"><a name="L732"></a><tt class="py-lineno">732</tt> <a class="py-toggle" href="#" id="Writer.other-toggle" onclick="return toggle('Writer.other');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Writer-class.html#other">other</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">obj</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Writer.other-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Writer.other-expanded"><a name="L733"></a><tt class="py-lineno">733</tt>  <tt class="py-line">        <tt class="py-docstring">"""Convert other to Etree element."""</tt> </tt>
<a name="L734"></a><tt class="py-lineno">734</tt>  <tt class="py-line">        <tt class="py-name">elem</tt> <tt class="py-op">=</tt> <tt class="py-name">ElementTree</tt><tt class="py-op">.</tt><tt class="py-name">Element</tt><tt class="py-op">(</tt><tt id="link-473" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._ns" class="py-name" href="#" onclick="return doclink('link-473', '_ns', 'link-55');">_ns</a></tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">.</tt><tt id="link-474" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-474', 'tag', 'link-44');">tag</a></tt><tt class="py-op">,</tt> <tt class="py-name">obj</tt><tt class="py-op">.</tt><tt class="py-name">namespace</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">obj</tt><tt class="py-op">.</tt><tt class="py-name">attributes</tt><tt class="py-op">)</tt> </tt>
<a name="L735"></a><tt class="py-lineno">735</tt>  <tt class="py-line">        <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt class="py-name">obj</tt><tt class="py-op">.</tt><tt id="link-475" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-475', 'value', 'link-287');">value</a></tt> </tt>
<a name="L736"></a><tt class="py-lineno">736</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">child</tt> <tt class="py-keyword">in</tt> <tt class="py-name">obj</tt><tt class="py-op">.</tt><tt id="link-476" class="py-name"><a title="Bio.Pathway.Rep.Graph.Graph.children
Bio.Pathway.Rep.MultiGraph.MultiGraph.children" class="py-name" href="#" onclick="return doclink('link-476', 'children', 'link-289');">children</a></tt><tt class="py-op">:</tt> </tt>
<a name="L737"></a><tt class="py-lineno">737</tt>  <tt class="py-line">            <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-477" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-477', 'append', 'link-95');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-478" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-478', 'other', 'link-98');">other</a></tt><tt class="py-op">(</tt><tt class="py-name">child</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L738"></a><tt class="py-lineno">738</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">elem</tt> </tt>
</div><a name="L739"></a><tt class="py-lineno">739</tt>  <tt class="py-line"> </tt>
<a name="L740"></a><tt class="py-lineno">740</tt>  <tt class="py-line">    <tt id="link-479" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.phylogeny" class="py-name" href="#" onclick="return doclink('link-479', 'phylogeny', 'link-92');">phylogeny</a></tt> <tt class="py-op">=</tt> <tt id="link-480" class="py-name" targets="Function Bio.Phylo.PhyloXMLIO._handle_complex()=Bio.Phylo.PhyloXMLIO-module.html#_handle_complex"><a title="Bio.Phylo.PhyloXMLIO._handle_complex" class="py-name" href="#" onclick="return doclink('link-480', '_handle_complex', 'link-480');">_handle_complex</a></tt><tt class="py-op">(</tt><tt class="py-string">"phylogeny"</tt><tt class="py-op">,</tt> </tt>
<a name="L741"></a><tt class="py-lineno">741</tt>  <tt class="py-line">                                <tt class="py-op">(</tt><tt class="py-string">"rooted"</tt><tt class="py-op">,</tt> <tt class="py-string">"rerootable"</tt><tt class="py-op">,</tt> </tt>
<a name="L742"></a><tt class="py-lineno">742</tt>  <tt class="py-line">                                 <tt class="py-string">"branch_length_unit"</tt><tt class="py-op">,</tt> <tt class="py-string">"type"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L743"></a><tt class="py-lineno">743</tt>  <tt class="py-line">                                <tt class="py-op">(</tt><tt class="py-string">"name"</tt><tt class="py-op">,</tt> </tt>
<a name="L744"></a><tt class="py-lineno">744</tt>  <tt class="py-line">                                 <tt class="py-string">"id"</tt><tt class="py-op">,</tt> </tt>
<a name="L745"></a><tt class="py-lineno">745</tt>  <tt class="py-line">                                 <tt class="py-string">"description"</tt><tt class="py-op">,</tt> </tt>
<a name="L746"></a><tt class="py-lineno">746</tt>  <tt class="py-line">                                 <tt class="py-string">"date"</tt><tt class="py-op">,</tt> </tt>
<a name="L747"></a><tt class="py-lineno">747</tt>  <tt class="py-line">                                 <tt class="py-op">(</tt><tt class="py-string">"confidence"</tt><tt class="py-op">,</tt> <tt class="py-string">"confidences"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L748"></a><tt class="py-lineno">748</tt>  <tt class="py-line">                                 <tt class="py-string">"clade"</tt><tt class="py-op">,</tt> </tt>
<a name="L749"></a><tt class="py-lineno">749</tt>  <tt class="py-line">                                 <tt class="py-op">(</tt><tt class="py-string">"clade_relation"</tt><tt class="py-op">,</tt> <tt class="py-string">"clade_relations"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L750"></a><tt class="py-lineno">750</tt>  <tt class="py-line">                                 <tt class="py-op">(</tt><tt class="py-string">"sequence_relation"</tt><tt class="py-op">,</tt> </tt>
<a name="L751"></a><tt class="py-lineno">751</tt>  <tt class="py-line">                                  <tt class="py-string">"sequence_relations"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L752"></a><tt class="py-lineno">752</tt>  <tt class="py-line">                                 <tt class="py-op">(</tt><tt class="py-string">"property"</tt><tt class="py-op">,</tt> <tt class="py-string">"properties"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L753"></a><tt class="py-lineno">753</tt>  <tt class="py-line">                                 <tt class="py-op">(</tt><tt class="py-string">"other"</tt><tt class="py-op">,</tt> <tt class="py-string">"other"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L754"></a><tt class="py-lineno">754</tt>  <tt class="py-line">                                 <tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L755"></a><tt class="py-lineno">755</tt>  <tt class="py-line"> </tt>
<a name="L756"></a><tt class="py-lineno">756</tt>  <tt class="py-line">    <tt id="link-481" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-481', 'clade', 'link-152');">clade</a></tt> <tt class="py-op">=</tt> <tt id="link-482" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_complex" class="py-name" href="#" onclick="return doclink('link-482', '_handle_complex', 'link-480');">_handle_complex</a></tt><tt class="py-op">(</tt><tt class="py-string">"clade"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-string">"id_source"</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L757"></a><tt class="py-lineno">757</tt>  <tt class="py-line">                            <tt class="py-op">(</tt><tt class="py-string">"name"</tt><tt class="py-op">,</tt> </tt>
<a name="L758"></a><tt class="py-lineno">758</tt>  <tt class="py-line">                             <tt class="py-string">"branch_length"</tt><tt class="py-op">,</tt> </tt>
<a name="L759"></a><tt class="py-lineno">759</tt>  <tt class="py-line">                             <tt class="py-op">(</tt><tt class="py-string">"confidence"</tt><tt class="py-op">,</tt> <tt class="py-string">"confidences"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L760"></a><tt class="py-lineno">760</tt>  <tt class="py-line">                             <tt class="py-string">"width"</tt><tt class="py-op">,</tt> </tt>
<a name="L761"></a><tt class="py-lineno">761</tt>  <tt class="py-line">                             <tt class="py-string">"color"</tt><tt class="py-op">,</tt> </tt>
<a name="L762"></a><tt class="py-lineno">762</tt>  <tt class="py-line">                             <tt class="py-string">"node_id"</tt><tt class="py-op">,</tt> </tt>
<a name="L763"></a><tt class="py-lineno">763</tt>  <tt class="py-line">                             <tt class="py-op">(</tt><tt class="py-string">"taxonomy"</tt><tt class="py-op">,</tt> <tt class="py-string">"taxonomies"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L764"></a><tt class="py-lineno">764</tt>  <tt class="py-line">                             <tt class="py-op">(</tt><tt class="py-string">"sequence"</tt><tt class="py-op">,</tt> <tt class="py-string">"sequences"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L765"></a><tt class="py-lineno">765</tt>  <tt class="py-line">                             <tt class="py-string">"events"</tt><tt class="py-op">,</tt> </tt>
<a name="L766"></a><tt class="py-lineno">766</tt>  <tt class="py-line">                             <tt class="py-string">"binary_characters"</tt><tt class="py-op">,</tt> </tt>
<a name="L767"></a><tt class="py-lineno">767</tt>  <tt class="py-line">                             <tt class="py-op">(</tt><tt class="py-string">"distribution"</tt><tt class="py-op">,</tt> <tt class="py-string">"distributions"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L768"></a><tt class="py-lineno">768</tt>  <tt class="py-line">                             <tt class="py-string">"date"</tt><tt class="py-op">,</tt> </tt>
<a name="L769"></a><tt class="py-lineno">769</tt>  <tt class="py-line">                             <tt class="py-op">(</tt><tt class="py-string">"reference"</tt><tt class="py-op">,</tt> <tt class="py-string">"references"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L770"></a><tt class="py-lineno">770</tt>  <tt class="py-line">                             <tt class="py-op">(</tt><tt class="py-string">"property"</tt><tt class="py-op">,</tt> <tt class="py-string">"properties"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L771"></a><tt class="py-lineno">771</tt>  <tt class="py-line">                             <tt class="py-op">(</tt><tt class="py-string">"clade"</tt><tt class="py-op">,</tt> <tt class="py-string">"clades"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L772"></a><tt class="py-lineno">772</tt>  <tt class="py-line">                             <tt class="py-op">(</tt><tt class="py-string">"other"</tt><tt class="py-op">,</tt> <tt class="py-string">"other"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L773"></a><tt class="py-lineno">773</tt>  <tt class="py-line">                             <tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L774"></a><tt class="py-lineno">774</tt>  <tt class="py-line"> </tt>
<a name="L775"></a><tt class="py-lineno">775</tt>  <tt class="py-line">    <tt id="link-483" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.accession()=Bio.GenBank._FeatureConsumer-class.html#accession,Method Bio.GenBank._RecordConsumer.accession()=Bio.GenBank._RecordConsumer-class.html#accession,Method Bio.Phylo.PhyloXMLIO.Parser.accession()=Bio.Phylo.PhyloXMLIO.Parser-class.html#accession,Method Bio.Phylo.PhyloXMLIO.Writer.accession()=Bio.Phylo.PhyloXMLIO.Writer-class.html#accession"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-483', 'accession', 'link-483');">accession</a></tt> <tt class="py-op">=</tt> <tt id="link-484" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_complex" class="py-name" href="#" onclick="return doclink('link-484', '_handle_complex', 'link-480');">_handle_complex</a></tt><tt class="py-op">(</tt><tt class="py-string">"accession"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-string">"source"</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L776"></a><tt class="py-lineno">776</tt>  <tt class="py-line">                                <tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">has_text</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L777"></a><tt class="py-lineno">777</tt>  <tt class="py-line"> </tt>
<a name="L778"></a><tt class="py-lineno">778</tt>  <tt class="py-line">    <tt id="link-485" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.annotation
Bio.Phylo.PhyloXMLIO.Writer.annotation" class="py-name" href="#" onclick="return doclink('link-485', 'annotation', 'link-241');">annotation</a></tt> <tt class="py-op">=</tt> <tt id="link-486" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_complex" class="py-name" href="#" onclick="return doclink('link-486', '_handle_complex', 'link-480');">_handle_complex</a></tt><tt class="py-op">(</tt><tt class="py-string">"annotation"</tt><tt class="py-op">,</tt> </tt>
<a name="L779"></a><tt class="py-lineno">779</tt>  <tt class="py-line">                                 <tt class="py-op">(</tt><tt class="py-string">"ref"</tt><tt class="py-op">,</tt> <tt class="py-string">"source"</tt><tt class="py-op">,</tt> <tt class="py-string">"evidence"</tt><tt class="py-op">,</tt> <tt class="py-string">"type"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L780"></a><tt class="py-lineno">780</tt>  <tt class="py-line">                                 <tt class="py-op">(</tt><tt class="py-string">"desc"</tt><tt class="py-op">,</tt> </tt>
<a name="L781"></a><tt class="py-lineno">781</tt>  <tt class="py-line">                                  <tt class="py-string">"confidence"</tt><tt class="py-op">,</tt> </tt>
<a name="L782"></a><tt class="py-lineno">782</tt>  <tt class="py-line">                                  <tt class="py-op">(</tt><tt class="py-string">"property"</tt><tt class="py-op">,</tt> <tt class="py-string">"properties"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L783"></a><tt class="py-lineno">783</tt>  <tt class="py-line">                                  <tt class="py-string">"uri"</tt><tt class="py-op">,</tt> </tt>
<a name="L784"></a><tt class="py-lineno">784</tt>  <tt class="py-line">                                  <tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L785"></a><tt class="py-lineno">785</tt>  <tt class="py-line"> </tt>
<a name="Writer.binary_characters"></a><div id="Writer.binary_characters-def"><a name="L786"></a><tt class="py-lineno">786</tt> <a class="py-toggle" href="#" id="Writer.binary_characters-toggle" onclick="return toggle('Writer.binary_characters');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Writer-class.html#binary_characters">binary_characters</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">obj</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Writer.binary_characters-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Writer.binary_characters-expanded"><a name="L787"></a><tt class="py-lineno">787</tt>  <tt class="py-line">        <tt class="py-docstring">"""Serialize a binary_characters node and its subnodes."""</tt> </tt>
<a name="L788"></a><tt class="py-lineno">788</tt>  <tt class="py-line">        <tt class="py-name">elem</tt> <tt class="py-op">=</tt> <tt class="py-name">ElementTree</tt><tt class="py-op">.</tt><tt class="py-name">Element</tt><tt class="py-op">(</tt><tt class="py-string">"binary_characters"</tt><tt class="py-op">,</tt> </tt>
<a name="L789"></a><tt class="py-lineno">789</tt>  <tt class="py-line">                                   <tt id="link-487" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._clean_attrib" class="py-name" href="#" onclick="return doclink('link-487', '_clean_attrib', 'link-446');">_clean_attrib</a></tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">,</tt> </tt>
<a name="L790"></a><tt class="py-lineno">790</tt>  <tt class="py-line">                                                 <tt class="py-op">(</tt><tt class="py-string">"type"</tt><tt class="py-op">,</tt> <tt class="py-string">"gained_count"</tt><tt class="py-op">,</tt> <tt class="py-string">"lost_count"</tt><tt class="py-op">,</tt> </tt>
<a name="L791"></a><tt class="py-lineno">791</tt>  <tt class="py-line">                                                  <tt class="py-string">"present_count"</tt><tt class="py-op">,</tt> <tt class="py-string">"absent_count"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L792"></a><tt class="py-lineno">792</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">subn</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">"gained"</tt><tt class="py-op">,</tt> <tt class="py-string">"lost"</tt><tt class="py-op">,</tt> <tt class="py-string">"present"</tt><tt class="py-op">,</tt> <tt class="py-string">"absent"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L793"></a><tt class="py-lineno">793</tt>  <tt class="py-line">            <tt class="py-name">subelem</tt> <tt class="py-op">=</tt> <tt class="py-name">ElementTree</tt><tt class="py-op">.</tt><tt class="py-name">Element</tt><tt class="py-op">(</tt><tt class="py-name">subn</tt><tt class="py-op">)</tt> </tt>
<a name="L794"></a><tt class="py-lineno">794</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">token</tt> <tt class="py-keyword">in</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">,</tt> <tt class="py-name">subn</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L795"></a><tt class="py-lineno">795</tt>  <tt class="py-line">                <tt class="py-name">subelem</tt><tt class="py-op">.</tt><tt id="link-488" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-488', 'append', 'link-95');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-489" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.bc()=Bio.Phylo.PhyloXMLIO.Writer-class.html#bc"><a title="Bio.Phylo.PhyloXMLIO.Writer.bc" class="py-name" href="#" onclick="return doclink('link-489', 'bc', 'link-489');">bc</a></tt><tt class="py-op">(</tt><tt class="py-name">token</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L796"></a><tt class="py-lineno">796</tt>  <tt class="py-line">            <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-490" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-490', 'append', 'link-95');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">subelem</tt><tt class="py-op">)</tt> </tt>
<a name="L797"></a><tt class="py-lineno">797</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">elem</tt> </tt>
</div><a name="L798"></a><tt class="py-lineno">798</tt>  <tt class="py-line"> </tt>
<a name="L799"></a><tt class="py-lineno">799</tt>  <tt class="py-line">    <tt id="link-491" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.clade_relation()=Bio.Phylo.PhyloXMLIO.Parser-class.html#clade_relation,Method Bio.Phylo.PhyloXMLIO.Writer.clade_relation()=Bio.Phylo.PhyloXMLIO.Writer-class.html#clade_relation"><a title="Bio.Phylo.PhyloXMLIO.Parser.clade_relation
Bio.Phylo.PhyloXMLIO.Writer.clade_relation" class="py-name" href="#" onclick="return doclink('link-491', 'clade_relation', 'link-491');">clade_relation</a></tt> <tt class="py-op">=</tt> <tt id="link-492" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_complex" class="py-name" href="#" onclick="return doclink('link-492', '_handle_complex', 'link-480');">_handle_complex</a></tt><tt class="py-op">(</tt><tt class="py-string">"clade_relation"</tt><tt class="py-op">,</tt> </tt>
<a name="L800"></a><tt class="py-lineno">800</tt>  <tt class="py-line">                                     <tt class="py-op">(</tt><tt class="py-string">"id_ref_0"</tt><tt class="py-op">,</tt> <tt class="py-string">"id_ref_1"</tt><tt class="py-op">,</tt> </tt>
<a name="L801"></a><tt class="py-lineno">801</tt>  <tt class="py-line">                                      <tt class="py-string">"distance"</tt><tt class="py-op">,</tt> <tt class="py-string">"type"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L802"></a><tt class="py-lineno">802</tt>  <tt class="py-line">                                     <tt class="py-op">(</tt><tt class="py-string">"confidence"</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L803"></a><tt class="py-lineno">803</tt>  <tt class="py-line"> </tt>
<a name="L804"></a><tt class="py-lineno">804</tt>  <tt class="py-line">    <tt id="link-493" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-493', 'color', 'link-337');">color</a></tt> <tt class="py-op">=</tt> <tt id="link-494" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_complex" class="py-name" href="#" onclick="return doclink('link-494', '_handle_complex', 'link-480');">_handle_complex</a></tt><tt class="py-op">(</tt><tt class="py-string">"color"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-string">"red"</tt><tt class="py-op">,</tt> <tt class="py-string">"green"</tt><tt class="py-op">,</tt> <tt class="py-string">"blue"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L805"></a><tt class="py-lineno">805</tt>  <tt class="py-line"> </tt>
<a name="L806"></a><tt class="py-lineno">806</tt>  <tt class="py-line">    <tt id="link-495" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-495', 'confidence', 'link-300');">confidence</a></tt> <tt class="py-op">=</tt> <tt id="link-496" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_complex" class="py-name" href="#" onclick="return doclink('link-496', '_handle_complex', 'link-480');">_handle_complex</a></tt><tt class="py-op">(</tt><tt class="py-string">"confidence"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-string">"type"</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L807"></a><tt class="py-lineno">807</tt>  <tt class="py-line">                                 <tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">has_text</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L808"></a><tt class="py-lineno">808</tt>  <tt class="py-line"> </tt>
<a name="L809"></a><tt class="py-lineno">809</tt>  <tt class="py-line">    <tt id="link-497" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.date()=Bio.GenBank._FeatureConsumer-class.html#date,Method Bio.GenBank._RecordConsumer.date()=Bio.GenBank._RecordConsumer-class.html#date,Method Bio.Phylo.PhyloXMLIO.Parser.date()=Bio.Phylo.PhyloXMLIO.Parser-class.html#date,Method Bio.Phylo.PhyloXMLIO.Writer.date()=Bio.Phylo.PhyloXMLIO.Writer-class.html#date"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-497', 'date', 'link-497');">date</a></tt> <tt class="py-op">=</tt> <tt id="link-498" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_complex" class="py-name" href="#" onclick="return doclink('link-498', '_handle_complex', 'link-480');">_handle_complex</a></tt><tt class="py-op">(</tt><tt class="py-string">"date"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-string">"unit"</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L810"></a><tt class="py-lineno">810</tt>  <tt class="py-line">                           <tt class="py-op">(</tt><tt class="py-string">"desc"</tt><tt class="py-op">,</tt> <tt class="py-string">"value"</tt><tt class="py-op">,</tt> <tt class="py-string">"minimum"</tt><tt class="py-op">,</tt> <tt class="py-string">"maximum"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L811"></a><tt class="py-lineno">811</tt>  <tt class="py-line"> </tt>
<a name="L812"></a><tt class="py-lineno">812</tt>  <tt class="py-line">    <tt id="link-499" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.distribution()=Bio.Phylo.PhyloXMLIO.Parser-class.html#distribution,Method Bio.Phylo.PhyloXMLIO.Writer.distribution()=Bio.Phylo.PhyloXMLIO.Writer-class.html#distribution,Method Bio.motifs.matrix.PositionSpecificScoringMatrix.distribution()=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#distribution"><a title="Bio.Phylo.PhyloXMLIO.Parser.distribution
Bio.Phylo.PhyloXMLIO.Writer.distribution
Bio.motifs.matrix.PositionSpecificScoringMatrix.distribution" class="py-name" href="#" onclick="return doclink('link-499', 'distribution', 'link-499');">distribution</a></tt> <tt class="py-op">=</tt> <tt id="link-500" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_complex" class="py-name" href="#" onclick="return doclink('link-500', '_handle_complex', 'link-480');">_handle_complex</a></tt><tt class="py-op">(</tt><tt class="py-string">"distribution"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L813"></a><tt class="py-lineno">813</tt>  <tt class="py-line">                                   <tt class="py-op">(</tt><tt class="py-string">"desc"</tt><tt class="py-op">,</tt> </tt>
<a name="L814"></a><tt class="py-lineno">814</tt>  <tt class="py-line">                                    <tt class="py-op">(</tt><tt class="py-string">"point"</tt><tt class="py-op">,</tt> <tt class="py-string">"points"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L815"></a><tt class="py-lineno">815</tt>  <tt class="py-line">                                    <tt class="py-op">(</tt><tt class="py-string">"polygon"</tt><tt class="py-op">,</tt> <tt class="py-string">"polygons"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L816"></a><tt class="py-lineno">816</tt>  <tt class="py-line">                                    <tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L817"></a><tt class="py-lineno">817</tt>  <tt class="py-line"> </tt>
<a name="Writer.domain"></a><div id="Writer.domain-def"><a name="L818"></a><tt class="py-lineno">818</tt> <a class="py-toggle" href="#" id="Writer.domain-toggle" onclick="return toggle('Writer.domain');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXMLIO.Writer-class.html#domain">domain</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">obj</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Writer.domain-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Writer.domain-expanded"><a name="L819"></a><tt class="py-lineno">819</tt>  <tt class="py-line">        <tt class="py-docstring">"""Serialize a domain node."""</tt> </tt>
<a name="L820"></a><tt class="py-lineno">820</tt>  <tt class="py-line">        <tt class="py-name">elem</tt> <tt class="py-op">=</tt> <tt class="py-name">ElementTree</tt><tt class="py-op">.</tt><tt class="py-name">Element</tt><tt class="py-op">(</tt><tt class="py-string">"domain"</tt><tt class="py-op">,</tt> </tt>
<a name="L821"></a><tt class="py-lineno">821</tt>  <tt class="py-line">                                   <tt class="py-op">{</tt><tt class="py-string">"from"</tt><tt class="py-op">:</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">.</tt><tt id="link-501" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.start=Bio.SeqFeature.CompoundLocation-class.html#start,Variable Bio.SeqFeature.FeatureLocation.start=Bio.SeqFeature.FeatureLocation-class.html#start"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-501', 'start', 'link-501');">start</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"to"</tt><tt class="py-op">:</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">.</tt><tt id="link-502" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.end=Bio.SeqFeature.CompoundLocation-class.html#end,Variable Bio.SeqFeature.FeatureLocation.end=Bio.SeqFeature.FeatureLocation-class.html#end"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-502', 'end', 'link-502');">end</a></tt><tt class="py-op">)</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L822"></a><tt class="py-lineno">822</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">obj</tt><tt class="py-op">.</tt><tt id="link-503" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-503', 'confidence', 'link-300');">confidence</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L823"></a><tt class="py-lineno">823</tt>  <tt class="py-line">            <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-504" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-504', 'set', 'link-148');">set</a></tt><tt class="py-op">(</tt><tt class="py-string">"confidence"</tt><tt class="py-op">,</tt> <tt id="link-505" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._serialize" class="py-name" href="#" onclick="return doclink('link-505', '_serialize', 'link-444');">_serialize</a></tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">.</tt><tt id="link-506" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-506', 'confidence', 'link-300');">confidence</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L824"></a><tt class="py-lineno">824</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">obj</tt><tt class="py-op">.</tt><tt id="link-507" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-507', 'id', 'link-372');">id</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L825"></a><tt class="py-lineno">825</tt>  <tt class="py-line">            <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-508" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-508', 'set', 'link-148');">set</a></tt><tt class="py-op">(</tt><tt class="py-string">"id"</tt><tt class="py-op">,</tt> <tt class="py-name">obj</tt><tt class="py-op">.</tt><tt id="link-509" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-509', 'id', 'link-372');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L826"></a><tt class="py-lineno">826</tt>  <tt class="py-line">        <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt id="link-510" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._serialize" class="py-name" href="#" onclick="return doclink('link-510', '_serialize', 'link-444');">_serialize</a></tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">.</tt><tt id="link-511" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-511', 'value', 'link-287');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L827"></a><tt class="py-lineno">827</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">elem</tt> </tt>
</div><a name="L828"></a><tt class="py-lineno">828</tt>  <tt class="py-line"> </tt>
<a name="L829"></a><tt class="py-lineno">829</tt>  <tt class="py-line">    <tt id="link-512" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.domain_architecture()=Bio.Phylo.PhyloXMLIO.Parser-class.html#domain_architecture,Method Bio.Phylo.PhyloXMLIO.Writer.domain_architecture()=Bio.Phylo.PhyloXMLIO.Writer-class.html#domain_architecture"><a title="Bio.Phylo.PhyloXMLIO.Parser.domain_architecture
Bio.Phylo.PhyloXMLIO.Writer.domain_architecture" class="py-name" href="#" onclick="return doclink('link-512', 'domain_architecture', 'link-512');">domain_architecture</a></tt> <tt class="py-op">=</tt> <tt id="link-513" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_complex" class="py-name" href="#" onclick="return doclink('link-513', '_handle_complex', 'link-480');">_handle_complex</a></tt><tt class="py-op">(</tt><tt class="py-string">"domain_architecture"</tt><tt class="py-op">,</tt> </tt>
<a name="L830"></a><tt class="py-lineno">830</tt>  <tt class="py-line">                                          <tt class="py-op">(</tt><tt class="py-string">"length"</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L831"></a><tt class="py-lineno">831</tt>  <tt class="py-line">                                          <tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">"domain"</tt><tt class="py-op">,</tt> <tt class="py-string">"domains"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L832"></a><tt class="py-lineno">832</tt>  <tt class="py-line"> </tt>
<a name="L833"></a><tt class="py-lineno">833</tt>  <tt class="py-line">    <tt id="link-514" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.events
Bio.Phylo.PhyloXMLIO.Writer.events" class="py-name" href="#" onclick="return doclink('link-514', 'events', 'link-79');">events</a></tt> <tt class="py-op">=</tt> <tt id="link-515" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_complex" class="py-name" href="#" onclick="return doclink('link-515', '_handle_complex', 'link-480');">_handle_complex</a></tt><tt class="py-op">(</tt><tt class="py-string">"events"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L834"></a><tt class="py-lineno">834</tt>  <tt class="py-line">                             <tt class="py-op">(</tt><tt class="py-string">"type"</tt><tt class="py-op">,</tt> </tt>
<a name="L835"></a><tt class="py-lineno">835</tt>  <tt class="py-line">                              <tt class="py-string">"duplications"</tt><tt class="py-op">,</tt> </tt>
<a name="L836"></a><tt class="py-lineno">836</tt>  <tt class="py-line">                              <tt class="py-string">"speciations"</tt><tt class="py-op">,</tt> </tt>
<a name="L837"></a><tt class="py-lineno">837</tt>  <tt class="py-line">                              <tt class="py-string">"losses"</tt><tt class="py-op">,</tt> </tt>
<a name="L838"></a><tt class="py-lineno">838</tt>  <tt class="py-line">                              <tt class="py-string">"confidence"</tt><tt class="py-op">,</tt> </tt>
<a name="L839"></a><tt class="py-lineno">839</tt>  <tt class="py-line">                              <tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L840"></a><tt class="py-lineno">840</tt>  <tt class="py-line"> </tt>
<a name="L841"></a><tt class="py-lineno">841</tt>  <tt class="py-line">    <tt id="link-516" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-516', 'id', 'link-372');">id</a></tt> <tt class="py-op">=</tt> <tt id="link-517" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_complex" class="py-name" href="#" onclick="return doclink('link-517', '_handle_complex', 'link-480');">_handle_complex</a></tt><tt class="py-op">(</tt><tt class="py-string">"id"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-string">"provider"</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">has_text</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L842"></a><tt class="py-lineno">842</tt>  <tt class="py-line"> </tt>
<a name="L843"></a><tt class="py-lineno">843</tt>  <tt class="py-line">    <tt id="link-518" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.mol_seq()=Bio.Phylo.PhyloXMLIO.Parser-class.html#mol_seq,Method Bio.Phylo.PhyloXMLIO.Writer.mol_seq()=Bio.Phylo.PhyloXMLIO.Writer-class.html#mol_seq"><a title="Bio.Phylo.PhyloXMLIO.Parser.mol_seq
Bio.Phylo.PhyloXMLIO.Writer.mol_seq" class="py-name" href="#" onclick="return doclink('link-518', 'mol_seq', 'link-518');">mol_seq</a></tt> <tt class="py-op">=</tt> <tt id="link-519" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_complex" class="py-name" href="#" onclick="return doclink('link-519', '_handle_complex', 'link-480');">_handle_complex</a></tt><tt class="py-op">(</tt><tt class="py-string">"mol_seq"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-string">"is_aligned"</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L844"></a><tt class="py-lineno">844</tt>  <tt class="py-line">                              <tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">has_text</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L845"></a><tt class="py-lineno">845</tt>  <tt class="py-line"> </tt>
<a name="L846"></a><tt class="py-lineno">846</tt>  <tt class="py-line">    <tt id="link-520" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.node_id" class="py-name" href="#" onclick="return doclink('link-520', 'node_id', 'link-212');">node_id</a></tt> <tt class="py-op">=</tt> <tt id="link-521" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_complex" class="py-name" href="#" onclick="return doclink('link-521', '_handle_complex', 'link-480');">_handle_complex</a></tt><tt class="py-op">(</tt><tt class="py-string">"node_id"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-string">"provider"</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">has_text</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L847"></a><tt class="py-lineno">847</tt>  <tt class="py-line"> </tt>
<a name="L848"></a><tt class="py-lineno">848</tt>  <tt class="py-line">    <tt id="link-522" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.point
Bio.Phylo.PhyloXMLIO.Writer.point" class="py-name" href="#" onclick="return doclink('link-522', 'point', 'link-362');">point</a></tt> <tt class="py-op">=</tt> <tt id="link-523" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_complex" class="py-name" href="#" onclick="return doclink('link-523', '_handle_complex', 'link-480');">_handle_complex</a></tt><tt class="py-op">(</tt><tt class="py-string">"point"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-string">"geodetic_datum"</tt><tt class="py-op">,</tt> <tt class="py-string">"alt_unit"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L849"></a><tt class="py-lineno">849</tt>  <tt class="py-line">                            <tt class="py-op">(</tt><tt class="py-string">"lat"</tt><tt class="py-op">,</tt> <tt class="py-string">"long"</tt><tt class="py-op">,</tt> <tt class="py-string">"alt"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L850"></a><tt class="py-lineno">850</tt>  <tt class="py-line"> </tt>
<a name="L851"></a><tt class="py-lineno">851</tt>  <tt class="py-line">    <tt id="link-524" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.polygon
Bio.Phylo.PhyloXMLIO.Writer.polygon" class="py-name" href="#" onclick="return doclink('link-524', 'polygon', 'link-364');">polygon</a></tt> <tt class="py-op">=</tt> <tt id="link-525" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_complex" class="py-name" href="#" onclick="return doclink('link-525', '_handle_complex', 'link-480');">_handle_complex</a></tt><tt class="py-op">(</tt><tt class="py-string">"polygon"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">"point"</tt><tt class="py-op">,</tt> <tt class="py-string">"points"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L852"></a><tt class="py-lineno">852</tt>  <tt class="py-line"> </tt>
<a name="L853"></a><tt class="py-lineno">853</tt>  <tt class="py-line">    <tt id="link-526" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-526', 'property', 'link-304');">property</a></tt> <tt class="py-op">=</tt> <tt id="link-527" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_complex" class="py-name" href="#" onclick="return doclink('link-527', '_handle_complex', 'link-480');">_handle_complex</a></tt><tt class="py-op">(</tt><tt class="py-string">"property"</tt><tt class="py-op">,</tt> </tt>
<a name="L854"></a><tt class="py-lineno">854</tt>  <tt class="py-line">                               <tt class="py-op">(</tt><tt class="py-string">"ref"</tt><tt class="py-op">,</tt> <tt class="py-string">"unit"</tt><tt class="py-op">,</tt> <tt class="py-string">"datatype"</tt><tt class="py-op">,</tt> </tt>
<a name="L855"></a><tt class="py-lineno">855</tt>  <tt class="py-line">                                <tt class="py-string">"applies_to"</tt><tt class="py-op">,</tt> <tt class="py-string">"id_ref"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L856"></a><tt class="py-lineno">856</tt>  <tt class="py-line">                               <tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">has_text</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L857"></a><tt class="py-lineno">857</tt>  <tt class="py-line"> </tt>
<a name="L858"></a><tt class="py-lineno">858</tt>  <tt class="py-line">    <tt id="link-528" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.reference()=Bio.Phylo.PhyloXMLIO.Parser-class.html#reference,Method Bio.Phylo.PhyloXMLIO.Writer.reference()=Bio.Phylo.PhyloXMLIO.Writer-class.html#reference,Method Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.reference()=Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer-class.html#reference"><a title="Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference
Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.reference" class="py-name" href="#" onclick="return doclink('link-528', 'reference', 'link-528');">reference</a></tt> <tt class="py-op">=</tt> <tt id="link-529" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_complex" class="py-name" href="#" onclick="return doclink('link-529', '_handle_complex', 'link-480');">_handle_complex</a></tt><tt class="py-op">(</tt><tt class="py-string">"reference"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-string">"doi"</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-string">"desc"</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L859"></a><tt class="py-lineno">859</tt>  <tt class="py-line"> </tt>
<a name="L860"></a><tt class="py-lineno">860</tt>  <tt class="py-line">    <tt id="link-530" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-530', 'sequence', 'link-226');">sequence</a></tt> <tt class="py-op">=</tt> <tt id="link-531" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_complex" class="py-name" href="#" onclick="return doclink('link-531', '_handle_complex', 'link-480');">_handle_complex</a></tt><tt class="py-op">(</tt><tt class="py-string">"sequence"</tt><tt class="py-op">,</tt> </tt>
<a name="L861"></a><tt class="py-lineno">861</tt>  <tt class="py-line">                               <tt class="py-op">(</tt><tt class="py-string">"type"</tt><tt class="py-op">,</tt> <tt class="py-string">"id_ref"</tt><tt class="py-op">,</tt> <tt class="py-string">"id_source"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L862"></a><tt class="py-lineno">862</tt>  <tt class="py-line">                               <tt class="py-op">(</tt><tt class="py-string">"symbol"</tt><tt class="py-op">,</tt> </tt>
<a name="L863"></a><tt class="py-lineno">863</tt>  <tt class="py-line">                                <tt class="py-string">"accession"</tt><tt class="py-op">,</tt> </tt>
<a name="L864"></a><tt class="py-lineno">864</tt>  <tt class="py-line">                                <tt class="py-string">"name"</tt><tt class="py-op">,</tt> </tt>
<a name="L865"></a><tt class="py-lineno">865</tt>  <tt class="py-line">                                <tt class="py-string">"location"</tt><tt class="py-op">,</tt> </tt>
<a name="L866"></a><tt class="py-lineno">866</tt>  <tt class="py-line">                                <tt class="py-string">"mol_seq"</tt><tt class="py-op">,</tt> </tt>
<a name="L867"></a><tt class="py-lineno">867</tt>  <tt class="py-line">                                <tt class="py-string">"uri"</tt><tt class="py-op">,</tt> </tt>
<a name="L868"></a><tt class="py-lineno">868</tt>  <tt class="py-line">                                <tt class="py-op">(</tt><tt class="py-string">"annotation"</tt><tt class="py-op">,</tt> <tt class="py-string">"annotations"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L869"></a><tt class="py-lineno">869</tt>  <tt class="py-line">                                <tt class="py-string">"domain_architecture"</tt><tt class="py-op">,</tt> </tt>
<a name="L870"></a><tt class="py-lineno">870</tt>  <tt class="py-line">                                <tt class="py-op">(</tt><tt class="py-string">"other"</tt><tt class="py-op">,</tt> <tt class="py-string">"other"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L871"></a><tt class="py-lineno">871</tt>  <tt class="py-line">                                <tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L872"></a><tt class="py-lineno">872</tt>  <tt class="py-line"> </tt>
<a name="L873"></a><tt class="py-lineno">873</tt>  <tt class="py-line">    <tt id="link-532" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.sequence_relation()=Bio.Phylo.PhyloXMLIO.Parser-class.html#sequence_relation,Method Bio.Phylo.PhyloXMLIO.Writer.sequence_relation()=Bio.Phylo.PhyloXMLIO.Writer-class.html#sequence_relation"><a title="Bio.Phylo.PhyloXMLIO.Parser.sequence_relation
Bio.Phylo.PhyloXMLIO.Writer.sequence_relation" class="py-name" href="#" onclick="return doclink('link-532', 'sequence_relation', 'link-532');">sequence_relation</a></tt> <tt class="py-op">=</tt> <tt id="link-533" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_complex" class="py-name" href="#" onclick="return doclink('link-533', '_handle_complex', 'link-480');">_handle_complex</a></tt><tt class="py-op">(</tt><tt class="py-string">"sequence_relation"</tt><tt class="py-op">,</tt> </tt>
<a name="L874"></a><tt class="py-lineno">874</tt>  <tt class="py-line">                                        <tt class="py-op">(</tt><tt class="py-string">"id_ref_0"</tt><tt class="py-op">,</tt> <tt class="py-string">"id_ref_1"</tt><tt class="py-op">,</tt> </tt>
<a name="L875"></a><tt class="py-lineno">875</tt>  <tt class="py-line">                                         <tt class="py-string">"distance"</tt><tt class="py-op">,</tt> <tt class="py-string">"type"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L876"></a><tt class="py-lineno">876</tt>  <tt class="py-line">                                        <tt class="py-op">(</tt><tt class="py-string">"confidence"</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L877"></a><tt class="py-lineno">877</tt>  <tt class="py-line"> </tt>
<a name="L878"></a><tt class="py-lineno">878</tt>  <tt class="py-line">    <tt id="link-534" class="py-name"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-534', 'taxonomy', 'link-257');">taxonomy</a></tt> <tt class="py-op">=</tt> <tt id="link-535" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_complex" class="py-name" href="#" onclick="return doclink('link-535', '_handle_complex', 'link-480');">_handle_complex</a></tt><tt class="py-op">(</tt><tt class="py-string">"taxonomy"</tt><tt class="py-op">,</tt> </tt>
<a name="L879"></a><tt class="py-lineno">879</tt>  <tt class="py-line">                               <tt class="py-op">(</tt><tt class="py-string">"id_source"</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L880"></a><tt class="py-lineno">880</tt>  <tt class="py-line">                               <tt class="py-op">(</tt><tt class="py-string">"id"</tt><tt class="py-op">,</tt> </tt>
<a name="L881"></a><tt class="py-lineno">881</tt>  <tt class="py-line">                                <tt class="py-string">"code"</tt><tt class="py-op">,</tt> </tt>
<a name="L882"></a><tt class="py-lineno">882</tt>  <tt class="py-line">                                <tt class="py-string">"scientific_name"</tt><tt class="py-op">,</tt> </tt>
<a name="L883"></a><tt class="py-lineno">883</tt>  <tt class="py-line">                                <tt class="py-string">"authority"</tt><tt class="py-op">,</tt> </tt>
<a name="L884"></a><tt class="py-lineno">884</tt>  <tt class="py-line">                                <tt class="py-op">(</tt><tt class="py-string">"common_name"</tt><tt class="py-op">,</tt> <tt class="py-string">"common_names"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L885"></a><tt class="py-lineno">885</tt>  <tt class="py-line">                                <tt class="py-op">(</tt><tt class="py-string">"synonym"</tt><tt class="py-op">,</tt> <tt class="py-string">"synonyms"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L886"></a><tt class="py-lineno">886</tt>  <tt class="py-line">                                <tt class="py-string">"rank"</tt><tt class="py-op">,</tt> </tt>
<a name="L887"></a><tt class="py-lineno">887</tt>  <tt class="py-line">                                <tt class="py-string">"uri"</tt><tt class="py-op">,</tt> </tt>
<a name="L888"></a><tt class="py-lineno">888</tt>  <tt class="py-line">                                <tt class="py-op">(</tt><tt class="py-string">"other"</tt><tt class="py-op">,</tt> <tt class="py-string">"other"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L889"></a><tt class="py-lineno">889</tt>  <tt class="py-line">                                <tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L890"></a><tt class="py-lineno">890</tt>  <tt class="py-line"> </tt>
<a name="L891"></a><tt class="py-lineno">891</tt>  <tt class="py-line">    <tt id="link-536" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-536', 'uri', 'link-8');">uri</a></tt> <tt class="py-op">=</tt> <tt id="link-537" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_complex" class="py-name" href="#" onclick="return doclink('link-537', '_handle_complex', 'link-480');">_handle_complex</a></tt><tt class="py-op">(</tt><tt class="py-string">"uri"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-string">"desc"</tt><tt class="py-op">,</tt> <tt class="py-string">"type"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">has_text</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L892"></a><tt class="py-lineno">892</tt>  <tt class="py-line"> </tt>
<a name="L893"></a><tt class="py-lineno">893</tt>  <tt class="py-line">    <tt class="py-comment"># Primitive types</tt> </tt>
<a name="L894"></a><tt class="py-lineno">894</tt>  <tt class="py-line"> </tt>
<a name="L895"></a><tt class="py-lineno">895</tt>  <tt class="py-line">    <tt class="py-comment"># Floating point</tt> </tt>
<a name="L896"></a><tt class="py-lineno">896</tt>  <tt class="py-line">    <tt id="link-538" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.alt" class="py-name" href="#" onclick="return doclink('link-538', 'alt', 'link-403');">alt</a></tt> <tt class="py-op">=</tt> <tt id="link-539" class="py-name" targets="Function Bio.Phylo.PhyloXMLIO._handle_simple()=Bio.Phylo.PhyloXMLIO-module.html#_handle_simple"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-539', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"alt"</tt><tt class="py-op">)</tt> </tt>
<a name="L897"></a><tt class="py-lineno">897</tt>  <tt class="py-line">    <tt id="link-540" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-540', 'branch_length', 'link-155');">branch_length</a></tt> <tt class="py-op">=</tt> <tt id="link-541" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-541', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"branch_length"</tt><tt class="py-op">)</tt> </tt>
<a name="L898"></a><tt class="py-lineno">898</tt>  <tt class="py-line">    <tt id="link-542" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.lat()=Bio.Phylo.PhyloXMLIO.Writer-class.html#lat"><a title="Bio.Phylo.PhyloXMLIO.Writer.lat" class="py-name" href="#" onclick="return doclink('link-542', 'lat', 'link-542');">lat</a></tt> <tt class="py-op">=</tt> <tt id="link-543" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-543', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"lat"</tt><tt class="py-op">)</tt> </tt>
<a name="L899"></a><tt class="py-lineno">899</tt>  <tt class="py-line">    <tt id="link-544" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.long()=Bio.Phylo.PhyloXMLIO.Writer-class.html#long"><a title="Bio.Phylo.PhyloXMLIO.Writer.long" class="py-name" href="#" onclick="return doclink('link-544', 'long', 'link-544');">long</a></tt> <tt class="py-op">=</tt> <tt id="link-545" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-545', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"long"</tt><tt class="py-op">)</tt> </tt>
<a name="L900"></a><tt class="py-lineno">900</tt>  <tt class="py-line">    <tt id="link-546" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.maximum" class="py-name" href="#" onclick="return doclink('link-546', 'maximum', 'link-355');">maximum</a></tt> <tt class="py-op">=</tt> <tt id="link-547" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-547', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"maximum"</tt><tt class="py-op">)</tt> </tt>
<a name="L901"></a><tt class="py-lineno">901</tt>  <tt class="py-line">    <tt id="link-548" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.minimum" class="py-name" href="#" onclick="return doclink('link-548', 'minimum', 'link-353');">minimum</a></tt> <tt class="py-op">=</tt> <tt id="link-549" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-549', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"minimum"</tt><tt class="py-op">)</tt> </tt>
<a name="L902"></a><tt class="py-lineno">902</tt>  <tt class="py-line">    <tt id="link-550" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-550', 'value', 'link-287');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-551" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-551', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"value"</tt><tt class="py-op">)</tt> </tt>
<a name="L903"></a><tt class="py-lineno">903</tt>  <tt class="py-line">    <tt id="link-552" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.width
Bio.Phylo.PhyloXMLIO.Writer.width" class="py-name" href="#" onclick="return doclink('link-552', 'width', 'link-204');">width</a></tt> <tt class="py-op">=</tt> <tt id="link-553" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-553', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"width"</tt><tt class="py-op">)</tt> </tt>
<a name="L904"></a><tt class="py-lineno">904</tt>  <tt class="py-line"> </tt>
<a name="L905"></a><tt class="py-lineno">905</tt>  <tt class="py-line">    <tt class="py-comment"># Integers</tt> </tt>
<a name="L906"></a><tt class="py-lineno">906</tt>  <tt class="py-line">    <tt id="link-554" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.blue" class="py-name" href="#" onclick="return doclink('link-554', 'blue', 'link-335');">blue</a></tt> <tt class="py-op">=</tt> <tt id="link-555" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-555', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"blue"</tt><tt class="py-op">)</tt> </tt>
<a name="L907"></a><tt class="py-lineno">907</tt>  <tt class="py-line">    <tt id="link-556" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.duplications" class="py-name" href="#" onclick="return doclink('link-556', 'duplications', 'link-382');">duplications</a></tt> <tt class="py-op">=</tt> <tt id="link-557" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-557', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"duplications"</tt><tt class="py-op">)</tt> </tt>
<a name="L908"></a><tt class="py-lineno">908</tt>  <tt class="py-line">    <tt id="link-558" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.green" class="py-name" href="#" onclick="return doclink('link-558', 'green', 'link-334');">green</a></tt> <tt class="py-op">=</tt> <tt id="link-559" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-559', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"green"</tt><tt class="py-op">)</tt> </tt>
<a name="L909"></a><tt class="py-lineno">909</tt>  <tt class="py-line">    <tt id="link-560" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.losses" class="py-name" href="#" onclick="return doclink('link-560', 'losses', 'link-386');">losses</a></tt> <tt class="py-op">=</tt> <tt id="link-561" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-561', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"losses"</tt><tt class="py-op">)</tt> </tt>
<a name="L910"></a><tt class="py-lineno">910</tt>  <tt class="py-line">    <tt id="link-562" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.red" class="py-name" href="#" onclick="return doclink('link-562', 'red', 'link-333');">red</a></tt> <tt class="py-op">=</tt> <tt id="link-563" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-563', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"red"</tt><tt class="py-op">)</tt> </tt>
<a name="L911"></a><tt class="py-lineno">911</tt>  <tt class="py-line">    <tt id="link-564" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.speciations" class="py-name" href="#" onclick="return doclink('link-564', 'speciations', 'link-384');">speciations</a></tt> <tt class="py-op">=</tt> <tt id="link-565" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-565', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"speciations"</tt><tt class="py-op">)</tt> </tt>
<a name="L912"></a><tt class="py-lineno">912</tt>  <tt class="py-line"> </tt>
<a name="L913"></a><tt class="py-lineno">913</tt>  <tt class="py-line">    <tt class="py-comment"># Strings</tt> </tt>
<a name="L914"></a><tt class="py-lineno">914</tt>  <tt class="py-line">    <tt id="link-566" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.bc" class="py-name" href="#" onclick="return doclink('link-566', 'bc', 'link-489');">bc</a></tt> <tt class="py-op">=</tt> <tt id="link-567" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-567', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"bc"</tt><tt class="py-op">)</tt> </tt>
<a name="L915"></a><tt class="py-lineno">915</tt>  <tt class="py-line">    <tt id="link-568" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.code()=Bio.Phylo.PhyloXMLIO.Writer-class.html#code"><a title="Bio.Phylo.PhyloXMLIO.Writer.code" class="py-name" href="#" onclick="return doclink('link-568', 'code', 'link-568');">code</a></tt> <tt class="py-op">=</tt> <tt id="link-569" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-569', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"code"</tt><tt class="py-op">)</tt> </tt>
<a name="L916"></a><tt class="py-lineno">916</tt>  <tt class="py-line">    <tt id="link-570" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.common_name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#common_name"><a title="Bio.Phylo.PhyloXMLIO.Writer.common_name" class="py-name" href="#" onclick="return doclink('link-570', 'common_name', 'link-570');">common_name</a></tt> <tt class="py-op">=</tt> <tt id="link-571" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-571', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"common_name"</tt><tt class="py-op">)</tt> </tt>
<a name="L917"></a><tt class="py-lineno">917</tt>  <tt class="py-line">    <tt id="link-572" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.desc" class="py-name" href="#" onclick="return doclink('link-572', 'desc', 'link-297');">desc</a></tt> <tt class="py-op">=</tt> <tt id="link-573" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-573', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"desc"</tt><tt class="py-op">)</tt> </tt>
<a name="L918"></a><tt class="py-lineno">918</tt>  <tt class="py-line">    <tt id="link-574" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.description()=Bio.Phylo.PhyloXMLIO.Writer-class.html#description,Method Bio.SearchIO._legacy.NCBIStandalone._DescriptionConsumer.description()=Bio.SearchIO._legacy.NCBIStandalone._DescriptionConsumer-class.html#description,Variable Bio.SearchIO._model.hit.Hit.description=Bio.SearchIO._model.hit.Hit-class.html#description,Variable Bio.SearchIO._model.query.QueryResult.description=Bio.SearchIO._model.query.QueryResult-class.html#description"><a title="Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._legacy.NCBIStandalone._DescriptionConsumer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-574', 'description', 'link-574');">description</a></tt> <tt class="py-op">=</tt> <tt id="link-575" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-575', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"description"</tt><tt class="py-op">)</tt> </tt>
<a name="L919"></a><tt class="py-lineno">919</tt>  <tt class="py-line">    <tt id="link-576" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.location()=Bio.GenBank._FeatureConsumer-class.html#location,Method Bio.GenBank._RecordConsumer.location()=Bio.GenBank._RecordConsumer-class.html#location,Method Bio.Phylo.PhyloXMLIO.Writer.location()=Bio.Phylo.PhyloXMLIO.Writer-class.html#location"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-576', 'location', 'link-576');">location</a></tt> <tt class="py-op">=</tt> <tt id="link-577" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-577', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"location"</tt><tt class="py-op">)</tt> </tt>
<a name="L920"></a><tt class="py-lineno">920</tt>  <tt class="py-line">    <tt id="link-578" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-578', 'name', 'link-208');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-579" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-579', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"name"</tt><tt class="py-op">)</tt> </tt>
<a name="L921"></a><tt class="py-lineno">921</tt>  <tt class="py-line">    <tt id="link-580" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.rank()=Bio.Phylo.PhyloXMLIO.Writer-class.html#rank"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-580', 'rank', 'link-580');">rank</a></tt> <tt class="py-op">=</tt> <tt id="link-581" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-581', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"rank"</tt><tt class="py-op">)</tt> </tt>
<a name="L922"></a><tt class="py-lineno">922</tt>  <tt class="py-line">    <tt id="link-582" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.scientific_name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#scientific_name"><a title="Bio.Phylo.PhyloXMLIO.Writer.scientific_name" class="py-name" href="#" onclick="return doclink('link-582', 'scientific_name', 'link-582');">scientific_name</a></tt> <tt class="py-op">=</tt> <tt id="link-583" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-583', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"scientific_name"</tt><tt class="py-op">)</tt> </tt>
<a name="L923"></a><tt class="py-lineno">923</tt>  <tt class="py-line">    <tt id="link-584" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.symbol()=Bio.Phylo.PhyloXMLIO.Writer-class.html#symbol"><a title="Bio.Phylo.PhyloXMLIO.Writer.symbol" class="py-name" href="#" onclick="return doclink('link-584', 'symbol', 'link-584');">symbol</a></tt> <tt class="py-op">=</tt> <tt id="link-585" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-585', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"symbol"</tt><tt class="py-op">)</tt> </tt>
<a name="L924"></a><tt class="py-lineno">924</tt>  <tt class="py-line">    <tt id="link-586" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.synonym()=Bio.Phylo.PhyloXMLIO.Writer-class.html#synonym"><a title="Bio.Phylo.PhyloXMLIO.Writer.synonym" class="py-name" href="#" onclick="return doclink('link-586', 'synonym', 'link-586');">synonym</a></tt> <tt class="py-op">=</tt> <tt id="link-587" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-587', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"synonym"</tt><tt class="py-op">)</tt> </tt>
<a name="L925"></a><tt class="py-lineno">925</tt>  <tt class="py-line">    <tt id="link-588" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-588', 'type', 'link-310');">type</a></tt> <tt class="py-op">=</tt> <tt id="link-589" class="py-name"><a title="Bio.Phylo.PhyloXMLIO._handle_simple" class="py-name" href="#" onclick="return doclink('link-589', '_handle_simple', 'link-539');">_handle_simple</a></tt><tt class="py-op">(</tt><tt class="py-string">"type"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L926"></a><tt class="py-lineno">926</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Nov  6 13:34:03 2019
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
