<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.SeqIO.InsdcIO</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.SeqIO-module.html">Package&nbsp;SeqIO</a> ::
        Module&nbsp;InsdcIO
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.SeqIO.InsdcIO-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.SeqIO.InsdcIO-module.html">Module Bio.SeqIO.InsdcIO</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">   1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2007-2011 by Peter Cock.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">   2</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L3"></a><tt class="py-lineno">   3</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L4"></a><tt class="py-lineno">   4</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L5"></a><tt class="py-lineno">   5</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package..</tt> </tt>
<a name="L6"></a><tt class="py-lineno">   6</tt>  <tt class="py-line"> </tt>
<a name="L7"></a><tt class="py-lineno">   7</tt>  <tt class="py-line"><tt class="py-docstring">"""Bio.SeqIO support for the "genbank" and "embl" file formats.</tt> </tt>
<a name="L8"></a><tt class="py-lineno">   8</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L9"></a><tt class="py-lineno">   9</tt>  <tt class="py-line"><tt class="py-docstring">You are expected to use this module via the Bio.SeqIO functions.</tt> </tt>
<a name="L10"></a><tt class="py-lineno">  10</tt>  <tt class="py-line"><tt class="py-docstring">Note that internally this module calls Bio.GenBank to do the actual</tt> </tt>
<a name="L11"></a><tt class="py-lineno">  11</tt>  <tt class="py-line"><tt class="py-docstring">parsing of GenBank, EMBL and IMGT files.</tt> </tt>
<a name="L12"></a><tt class="py-lineno">  12</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L13"></a><tt class="py-lineno">  13</tt>  <tt class="py-line"><tt class="py-docstring">See also:</tt> </tt>
<a name="L14"></a><tt class="py-lineno">  14</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L15"></a><tt class="py-lineno">  15</tt>  <tt class="py-line"><tt class="py-docstring">International Nucleotide Sequence Database Collaboration</tt> </tt>
<a name="L16"></a><tt class="py-lineno">  16</tt>  <tt class="py-line"><tt class="py-docstring">http://www.insdc.org/</tt> </tt>
<a name="L17"></a><tt class="py-lineno">  17</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L18"></a><tt class="py-lineno">  18</tt>  <tt class="py-line"><tt class="py-docstring">GenBank</tt> </tt>
<a name="L19"></a><tt class="py-lineno">  19</tt>  <tt class="py-line"><tt class="py-docstring">http://www.ncbi.nlm.nih.gov/Genbank/</tt> </tt>
<a name="L20"></a><tt class="py-lineno">  20</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L21"></a><tt class="py-lineno">  21</tt>  <tt class="py-line"><tt class="py-docstring">EMBL Nucleotide Sequence Database</tt> </tt>
<a name="L22"></a><tt class="py-lineno">  22</tt>  <tt class="py-line"><tt class="py-docstring">http://www.ebi.ac.uk/embl/</tt> </tt>
<a name="L23"></a><tt class="py-lineno">  23</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L24"></a><tt class="py-lineno">  24</tt>  <tt class="py-line"><tt class="py-docstring">DDBJ (DNA Data Bank of Japan)</tt> </tt>
<a name="L25"></a><tt class="py-lineno">  25</tt>  <tt class="py-line"><tt class="py-docstring">http://www.ddbj.nig.ac.jp/</tt> </tt>
<a name="L26"></a><tt class="py-lineno">  26</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L27"></a><tt class="py-lineno">  27</tt>  <tt class="py-line"><tt class="py-docstring">IMGT (use a variant of EMBL format with longer feature indents)</tt> </tt>
<a name="L28"></a><tt class="py-lineno">  28</tt>  <tt class="py-line"><tt class="py-docstring">http://imgt.cines.fr/download/LIGM-DB/userman_doc.html</tt> </tt>
<a name="L29"></a><tt class="py-lineno">  29</tt>  <tt class="py-line"><tt class="py-docstring">http://imgt.cines.fr/download/LIGM-DB/ftable_doc.html</tt> </tt>
<a name="L30"></a><tt class="py-lineno">  30</tt>  <tt class="py-line"><tt class="py-docstring">http://www.ebi.ac.uk/imgt/hla/docs/manual.html</tt> </tt>
<a name="L31"></a><tt class="py-lineno">  31</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L32"></a><tt class="py-lineno">  32</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L33"></a><tt class="py-lineno">  33</tt>  <tt class="py-line"> </tt>
<a name="L34"></a><tt class="py-lineno">  34</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">__future__</tt> <tt class="py-keyword">import</tt> <tt class="py-name">print_function</tt> </tt>
<a name="L35"></a><tt class="py-lineno">  35</tt>  <tt class="py-line"> </tt>
<a name="L36"></a><tt class="py-lineno">  36</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L37"></a><tt class="py-lineno">  37</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-1" class="py-name" targets="Class Bio.BiopythonWarning=Bio.BiopythonWarning-class.html"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-1', 'BiopythonWarning', 'link-1');">BiopythonWarning</a></tt> </tt>
<a name="L38"></a><tt class="py-lineno">  38</tt>  <tt class="py-line"> </tt>
<a name="L39"></a><tt class="py-lineno">  39</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-2" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-2', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-3" class="py-name" targets="Module Bio.Seq=Bio.Seq-module.html,Class Bio.Seq.Seq=Bio.Seq.Seq-class.html"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-3', 'Seq', 'link-3');">Seq</a></tt> <tt class="py-keyword">import</tt> <tt id="link-4" class="py-name" targets="Class Bio.Seq.UnknownSeq=Bio.Seq.UnknownSeq-class.html"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-4', 'UnknownSeq', 'link-4');">UnknownSeq</a></tt> </tt>
<a name="L40"></a><tt class="py-lineno">  40</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-5" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-5', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-6" class="py-name" targets="Package Bio.GenBank=Bio.GenBank-module.html"><a title="Bio.GenBank" class="py-name" href="#" onclick="return doclink('link-6', 'GenBank', 'link-6');">GenBank</a></tt><tt class="py-op">.</tt><tt id="link-7" class="py-name" targets="Module Bio.GenBank.Scanner=Bio.GenBank.Scanner-module.html"><a title="Bio.GenBank.Scanner" class="py-name" href="#" onclick="return doclink('link-7', 'Scanner', 'link-7');">Scanner</a></tt> <tt class="py-keyword">import</tt> <tt id="link-8" class="py-name" targets="Class Bio.GenBank.Scanner.GenBankScanner=Bio.GenBank.Scanner.GenBankScanner-class.html"><a title="Bio.GenBank.Scanner.GenBankScanner" class="py-name" href="#" onclick="return doclink('link-8', 'GenBankScanner', 'link-8');">GenBankScanner</a></tt><tt class="py-op">,</tt> <tt id="link-9" class="py-name" targets="Class Bio.GenBank.Scanner.EmblScanner=Bio.GenBank.Scanner.EmblScanner-class.html"><a title="Bio.GenBank.Scanner.EmblScanner" class="py-name" href="#" onclick="return doclink('link-9', 'EmblScanner', 'link-9');">EmblScanner</a></tt><tt class="py-op">,</tt> <tt id="link-10" class="py-name" targets="Class Bio.GenBank.Scanner._ImgtScanner=Bio.GenBank.Scanner._ImgtScanner-class.html"><a title="Bio.GenBank.Scanner._ImgtScanner" class="py-name" href="#" onclick="return doclink('link-10', '_ImgtScanner', 'link-10');">_ImgtScanner</a></tt> </tt>
<a name="L41"></a><tt class="py-lineno">  41</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-11" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-11', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-12" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-12', 'Alphabet', 'link-12');">Alphabet</a></tt> </tt>
<a name="L42"></a><tt class="py-lineno">  42</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt><tt id="link-13" class="py-name" targets="Module Bio.AlignIO.Interfaces=Bio.AlignIO.Interfaces-module.html,Module Bio.SeqIO.Interfaces=Bio.SeqIO.Interfaces-module.html"><a title="Bio.AlignIO.Interfaces
Bio.SeqIO.Interfaces" class="py-name" href="#" onclick="return doclink('link-13', 'Interfaces', 'link-13');">Interfaces</a></tt> <tt class="py-keyword">import</tt> <tt id="link-14" class="py-name" targets="Class Bio.SeqIO.Interfaces.SequentialSequenceWriter=Bio.SeqIO.Interfaces.SequentialSequenceWriter-class.html"><a title="Bio.SeqIO.Interfaces.SequentialSequenceWriter" class="py-name" href="#" onclick="return doclink('link-14', 'SequentialSequenceWriter', 'link-14');">SequentialSequenceWriter</a></tt> </tt>
<a name="L43"></a><tt class="py-lineno">  43</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-15" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-15', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-16" class="py-name" targets="Module Bio.SeqFeature=Bio.SeqFeature-module.html,Class Bio.SeqFeature.SeqFeature=Bio.SeqFeature.SeqFeature-class.html"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-16', 'SeqFeature', 'link-16');">SeqFeature</a></tt> </tt>
<a name="L44"></a><tt class="py-lineno">  44</tt>  <tt class="py-line"> </tt>
<a name="L45"></a><tt class="py-lineno">  45</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-17" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-17', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-18" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-18', '_py3k', 'link-18');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt id="link-19" class="py-name" targets="Function Bio._py3k._is_int_or_long()=Bio._py3k-module.html#_is_int_or_long"><a title="Bio._py3k._is_int_or_long" class="py-name" href="#" onclick="return doclink('link-19', '_is_int_or_long', 'link-19');">_is_int_or_long</a></tt> </tt>
<a name="L46"></a><tt class="py-lineno">  46</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-20" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-20', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-21" class="py-name"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-21', '_py3k', 'link-18');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">basestring</tt> </tt>
<a name="L47"></a><tt class="py-lineno">  47</tt>  <tt class="py-line"> </tt>
<a name="L48"></a><tt class="py-lineno">  48</tt>  <tt class="py-line"><tt class="py-name">__docformat__</tt> <tt class="py-op">=</tt> <tt class="py-string">"restructuredtext en"</tt> </tt>
<a name="L49"></a><tt class="py-lineno">  49</tt>  <tt class="py-line"> </tt>
<a name="L50"></a><tt class="py-lineno">  50</tt>  <tt class="py-line"><tt class="py-comment"># NOTE</tt> </tt>
<a name="L51"></a><tt class="py-lineno">  51</tt>  <tt class="py-line"><tt class="py-comment"># ====</tt> </tt>
<a name="L52"></a><tt class="py-lineno">  52</tt>  <tt class="py-line"><tt class="py-comment"># The "brains" for parsing GenBank, EMBL and IMGT files (and any</tt> </tt>
<a name="L53"></a><tt class="py-lineno">  53</tt>  <tt class="py-line"><tt class="py-comment"># other flat file variants from the INSDC in future) is in</tt> </tt>
<a name="L54"></a><tt class="py-lineno">  54</tt>  <tt class="py-line"><tt class="py-comment"># Bio.GenBank.Scanner (plus the _FeatureConsumer in Bio.GenBank)</tt> </tt>
<a name="L55"></a><tt class="py-lineno">  55</tt>  <tt class="py-line"><tt class="py-comment"># However, all the writing code is in this file.</tt> </tt>
<a name="L56"></a><tt class="py-lineno">  56</tt>  <tt class="py-line"> </tt>
<a name="L57"></a><tt class="py-lineno">  57</tt>  <tt class="py-line"> </tt>
<a name="GenBankIterator"></a><div id="GenBankIterator-def"><a name="L58"></a><tt class="py-lineno">  58</tt> <a class="py-toggle" href="#" id="GenBankIterator-toggle" onclick="return toggle('GenBankIterator');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO-module.html#GenBankIterator">GenBankIterator</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GenBankIterator-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="GenBankIterator-expanded"><a name="L59"></a><tt class="py-lineno">  59</tt>  <tt class="py-line">    <tt class="py-docstring">"""Breaks up a Genbank file into SeqRecord objects.</tt> </tt>
<a name="L60"></a><tt class="py-lineno">  60</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L61"></a><tt class="py-lineno">  61</tt>  <tt class="py-line"><tt class="py-docstring">    Every section from the LOCUS line to the terminating // becomes</tt> </tt>
<a name="L62"></a><tt class="py-lineno">  62</tt>  <tt class="py-line"><tt class="py-docstring">    a single SeqRecord with associated annotation and features.</tt> </tt>
<a name="L63"></a><tt class="py-lineno">  63</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L64"></a><tt class="py-lineno">  64</tt>  <tt class="py-line"><tt class="py-docstring">    Note that for genomes or chromosomes, there is typically only</tt> </tt>
<a name="L65"></a><tt class="py-lineno">  65</tt>  <tt class="py-line"><tt class="py-docstring">    one record."""</tt> </tt>
<a name="L66"></a><tt class="py-lineno">  66</tt>  <tt class="py-line">    <tt class="py-comment"># This calls a generator function:</tt> </tt>
<a name="L67"></a><tt class="py-lineno">  67</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-22" class="py-name"><a title="Bio.GenBank.Scanner.GenBankScanner" class="py-name" href="#" onclick="return doclink('link-22', 'GenBankScanner', 'link-8');">GenBankScanner</a></tt><tt class="py-op">(</tt><tt class="py-name">debug</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-23" class="py-name" targets="Method Bio.GenBank.Scanner.InsdcScanner.parse_records()=Bio.GenBank.Scanner.InsdcScanner-class.html#parse_records"><a title="Bio.GenBank.Scanner.InsdcScanner.parse_records" class="py-name" href="#" onclick="return doclink('link-23', 'parse_records', 'link-23');">parse_records</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
</div><a name="L68"></a><tt class="py-lineno">  68</tt>  <tt class="py-line"> </tt>
<a name="L69"></a><tt class="py-lineno">  69</tt>  <tt class="py-line"> </tt>
<a name="EmblIterator"></a><div id="EmblIterator-def"><a name="L70"></a><tt class="py-lineno">  70</tt> <a class="py-toggle" href="#" id="EmblIterator-toggle" onclick="return toggle('EmblIterator');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO-module.html#EmblIterator">EmblIterator</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EmblIterator-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="EmblIterator-expanded"><a name="L71"></a><tt class="py-lineno">  71</tt>  <tt class="py-line">    <tt class="py-docstring">"""Breaks up an EMBL file into SeqRecord objects.</tt> </tt>
<a name="L72"></a><tt class="py-lineno">  72</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L73"></a><tt class="py-lineno">  73</tt>  <tt class="py-line"><tt class="py-docstring">    Every section from the LOCUS line to the terminating // becomes</tt> </tt>
<a name="L74"></a><tt class="py-lineno">  74</tt>  <tt class="py-line"><tt class="py-docstring">    a single SeqRecord with associated annotation and features.</tt> </tt>
<a name="L75"></a><tt class="py-lineno">  75</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L76"></a><tt class="py-lineno">  76</tt>  <tt class="py-line"><tt class="py-docstring">    Note that for genomes or chromosomes, there is typically only</tt> </tt>
<a name="L77"></a><tt class="py-lineno">  77</tt>  <tt class="py-line"><tt class="py-docstring">    one record."""</tt> </tt>
<a name="L78"></a><tt class="py-lineno">  78</tt>  <tt class="py-line">    <tt class="py-comment"># This calls a generator function:</tt> </tt>
<a name="L79"></a><tt class="py-lineno">  79</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-24" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner" class="py-name" href="#" onclick="return doclink('link-24', 'EmblScanner', 'link-9');">EmblScanner</a></tt><tt class="py-op">(</tt><tt class="py-name">debug</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-25" class="py-name"><a title="Bio.GenBank.Scanner.InsdcScanner.parse_records" class="py-name" href="#" onclick="return doclink('link-25', 'parse_records', 'link-23');">parse_records</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
</div><a name="L80"></a><tt class="py-lineno">  80</tt>  <tt class="py-line"> </tt>
<a name="L81"></a><tt class="py-lineno">  81</tt>  <tt class="py-line"> </tt>
<a name="ImgtIterator"></a><div id="ImgtIterator-def"><a name="L82"></a><tt class="py-lineno">  82</tt> <a class="py-toggle" href="#" id="ImgtIterator-toggle" onclick="return toggle('ImgtIterator');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO-module.html#ImgtIterator">ImgtIterator</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ImgtIterator-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="ImgtIterator-expanded"><a name="L83"></a><tt class="py-lineno">  83</tt>  <tt class="py-line">    <tt class="py-docstring">"""Breaks up an IMGT file into SeqRecord objects.</tt> </tt>
<a name="L84"></a><tt class="py-lineno">  84</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L85"></a><tt class="py-lineno">  85</tt>  <tt class="py-line"><tt class="py-docstring">    Every section from the LOCUS line to the terminating // becomes</tt> </tt>
<a name="L86"></a><tt class="py-lineno">  86</tt>  <tt class="py-line"><tt class="py-docstring">    a single SeqRecord with associated annotation and features.</tt> </tt>
<a name="L87"></a><tt class="py-lineno">  87</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L88"></a><tt class="py-lineno">  88</tt>  <tt class="py-line"><tt class="py-docstring">    Note that for genomes or chromosomes, there is typically only</tt> </tt>
<a name="L89"></a><tt class="py-lineno">  89</tt>  <tt class="py-line"><tt class="py-docstring">    one record."""</tt> </tt>
<a name="L90"></a><tt class="py-lineno">  90</tt>  <tt class="py-line">    <tt class="py-comment"># This calls a generator function:</tt> </tt>
<a name="L91"></a><tt class="py-lineno">  91</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-26" class="py-name"><a title="Bio.GenBank.Scanner._ImgtScanner" class="py-name" href="#" onclick="return doclink('link-26', '_ImgtScanner', 'link-10');">_ImgtScanner</a></tt><tt class="py-op">(</tt><tt class="py-name">debug</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-27" class="py-name"><a title="Bio.GenBank.Scanner.InsdcScanner.parse_records" class="py-name" href="#" onclick="return doclink('link-27', 'parse_records', 'link-23');">parse_records</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
</div><a name="L92"></a><tt class="py-lineno">  92</tt>  <tt class="py-line"> </tt>
<a name="L93"></a><tt class="py-lineno">  93</tt>  <tt class="py-line"> </tt>
<a name="GenBankCdsFeatureIterator"></a><div id="GenBankCdsFeatureIterator-def"><a name="L94"></a><tt class="py-lineno">  94</tt> <a class="py-toggle" href="#" id="GenBankCdsFeatureIterator-toggle" onclick="return toggle('GenBankCdsFeatureIterator');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO-module.html#GenBankCdsFeatureIterator">GenBankCdsFeatureIterator</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt id="link-28" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-28', 'Alphabet', 'link-12');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-29" class="py-name" targets="Variable Bio.Alphabet.generic_protein=Bio.Alphabet-module.html#generic_protein"><a title="Bio.Alphabet.generic_protein" class="py-name" href="#" onclick="return doclink('link-29', 'generic_protein', 'link-29');">generic_protein</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GenBankCdsFeatureIterator-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="GenBankCdsFeatureIterator-expanded"><a name="L95"></a><tt class="py-lineno">  95</tt>  <tt class="py-line">    <tt class="py-docstring">"""Breaks up a Genbank file into SeqRecord objects for each CDS feature.</tt> </tt>
<a name="L96"></a><tt class="py-lineno">  96</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L97"></a><tt class="py-lineno">  97</tt>  <tt class="py-line"><tt class="py-docstring">    Every section from the LOCUS line to the terminating // can contain</tt> </tt>
<a name="L98"></a><tt class="py-lineno">  98</tt>  <tt class="py-line"><tt class="py-docstring">    many CDS features.  These are returned as with the stated amino acid</tt> </tt>
<a name="L99"></a><tt class="py-lineno">  99</tt>  <tt class="py-line"><tt class="py-docstring">    translation sequence (if given).</tt> </tt>
<a name="L100"></a><tt class="py-lineno"> 100</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L101"></a><tt class="py-lineno"> 101</tt>  <tt class="py-line">    <tt class="py-comment"># This calls a generator function:</tt> </tt>
<a name="L102"></a><tt class="py-lineno"> 102</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-30" class="py-name"><a title="Bio.GenBank.Scanner.GenBankScanner" class="py-name" href="#" onclick="return doclink('link-30', 'GenBankScanner', 'link-8');">GenBankScanner</a></tt><tt class="py-op">(</tt><tt class="py-name">debug</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-31" class="py-name" targets="Method Bio.GenBank.Scanner.InsdcScanner.parse_cds_features()=Bio.GenBank.Scanner.InsdcScanner-class.html#parse_cds_features"><a title="Bio.GenBank.Scanner.InsdcScanner.parse_cds_features" class="py-name" href="#" onclick="return doclink('link-31', 'parse_cds_features', 'link-31');">parse_cds_features</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt id="link-32" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.alphabet=Bio.SearchIO._model.hsp.HSP-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSPFragment.alphabet=Bio.SearchIO._model.hsp.HSPFragment-class.html#alphabet"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-32', 'alphabet', 'link-32');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L103"></a><tt class="py-lineno"> 103</tt>  <tt class="py-line"> </tt>
<a name="L104"></a><tt class="py-lineno"> 104</tt>  <tt class="py-line"> </tt>
<a name="EmblCdsFeatureIterator"></a><div id="EmblCdsFeatureIterator-def"><a name="L105"></a><tt class="py-lineno"> 105</tt> <a class="py-toggle" href="#" id="EmblCdsFeatureIterator-toggle" onclick="return toggle('EmblCdsFeatureIterator');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO-module.html#EmblCdsFeatureIterator">EmblCdsFeatureIterator</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt id="link-33" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-33', 'Alphabet', 'link-12');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-34" class="py-name"><a title="Bio.Alphabet.generic_protein" class="py-name" href="#" onclick="return doclink('link-34', 'generic_protein', 'link-29');">generic_protein</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EmblCdsFeatureIterator-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="EmblCdsFeatureIterator-expanded"><a name="L106"></a><tt class="py-lineno"> 106</tt>  <tt class="py-line">    <tt class="py-docstring">"""Breaks up a EMBL file into SeqRecord objects for each CDS feature.</tt> </tt>
<a name="L107"></a><tt class="py-lineno"> 107</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L108"></a><tt class="py-lineno"> 108</tt>  <tt class="py-line"><tt class="py-docstring">    Every section from the LOCUS line to the terminating // can contain</tt> </tt>
<a name="L109"></a><tt class="py-lineno"> 109</tt>  <tt class="py-line"><tt class="py-docstring">    many CDS features.  These are returned as with the stated amino acid</tt> </tt>
<a name="L110"></a><tt class="py-lineno"> 110</tt>  <tt class="py-line"><tt class="py-docstring">    translation sequence (if given).</tt> </tt>
<a name="L111"></a><tt class="py-lineno"> 111</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L112"></a><tt class="py-lineno"> 112</tt>  <tt class="py-line">    <tt class="py-comment"># This calls a generator function:</tt> </tt>
<a name="L113"></a><tt class="py-lineno"> 113</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-35" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner" class="py-name" href="#" onclick="return doclink('link-35', 'EmblScanner', 'link-9');">EmblScanner</a></tt><tt class="py-op">(</tt><tt class="py-name">debug</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-36" class="py-name"><a title="Bio.GenBank.Scanner.InsdcScanner.parse_cds_features" class="py-name" href="#" onclick="return doclink('link-36', 'parse_cds_features', 'link-31');">parse_cds_features</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt id="link-37" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-37', 'alphabet', 'link-32');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L114"></a><tt class="py-lineno"> 114</tt>  <tt class="py-line"> </tt>
<a name="L115"></a><tt class="py-lineno"> 115</tt>  <tt class="py-line"> </tt>
<a name="_insdc_feature_position_string"></a><div id="_insdc_feature_position_string-def"><a name="L116"></a><tt class="py-lineno"> 116</tt> <a class="py-toggle" href="#" id="_insdc_feature_position_string-toggle" onclick="return toggle('_insdc_feature_position_string');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO-module.html#_insdc_feature_position_string">_insdc_feature_position_string</a><tt class="py-op">(</tt><tt class="py-param">pos</tt><tt class="py-op">,</tt> <tt class="py-param">offset</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_insdc_feature_position_string-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_insdc_feature_position_string-expanded"><a name="L117"></a><tt class="py-lineno"> 117</tt>  <tt class="py-line">    <tt class="py-docstring">"""Build a GenBank/EMBL position string (PRIVATE).</tt> </tt>
<a name="L118"></a><tt class="py-lineno"> 118</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L119"></a><tt class="py-lineno"> 119</tt>  <tt class="py-line"><tt class="py-docstring">    Use offset=1 to add one to convert a start position from python counting.</tt> </tt>
<a name="L120"></a><tt class="py-lineno"> 120</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L121"></a><tt class="py-lineno"> 121</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">pos</tt><tt class="py-op">,</tt> <tt id="link-38" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-38', 'SeqFeature', 'link-16');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-39" class="py-name" targets="Class Bio.SeqFeature.ExactPosition=Bio.SeqFeature.ExactPosition-class.html"><a title="Bio.SeqFeature.ExactPosition" class="py-name" href="#" onclick="return doclink('link-39', 'ExactPosition', 'link-39');">ExactPosition</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L122"></a><tt class="py-lineno"> 122</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">"%i"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">pos</tt><tt class="py-op">.</tt><tt id="link-40" class="py-name" targets="Variable Bio.SeqFeature.AfterPosition.position=Bio.SeqFeature.AfterPosition-class.html#position,Variable Bio.SeqFeature.BeforePosition.position=Bio.SeqFeature.BeforePosition-class.html#position,Variable Bio.SeqFeature.BetweenPosition.position=Bio.SeqFeature.BetweenPosition-class.html#position,Variable Bio.SeqFeature.ExactPosition.position=Bio.SeqFeature.ExactPosition-class.html#position,Variable Bio.SeqFeature.OneOfPosition.position=Bio.SeqFeature.OneOfPosition-class.html#position,Variable Bio.SeqFeature.UnknownPosition.position=Bio.SeqFeature.UnknownPosition-class.html#position,Variable Bio.SeqFeature.WithinPosition.position=Bio.SeqFeature.WithinPosition-class.html#position"><a title="Bio.SeqFeature.AfterPosition.position
Bio.SeqFeature.BeforePosition.position
Bio.SeqFeature.BetweenPosition.position
Bio.SeqFeature.ExactPosition.position
Bio.SeqFeature.OneOfPosition.position
Bio.SeqFeature.UnknownPosition.position
Bio.SeqFeature.WithinPosition.position" class="py-name" href="#" onclick="return doclink('link-40', 'position', 'link-40');">position</a></tt> <tt class="py-op">+</tt> <tt class="py-name">offset</tt><tt class="py-op">)</tt> </tt>
<a name="L123"></a><tt class="py-lineno"> 123</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">pos</tt><tt class="py-op">,</tt> <tt id="link-41" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-41', 'SeqFeature', 'link-16');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-42" class="py-name" targets="Class Bio.SeqFeature.WithinPosition=Bio.SeqFeature.WithinPosition-class.html"><a title="Bio.SeqFeature.WithinPosition" class="py-name" href="#" onclick="return doclink('link-42', 'WithinPosition', 'link-42');">WithinPosition</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L124"></a><tt class="py-lineno"> 124</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">"(%i.%i)"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">pos</tt><tt class="py-op">.</tt><tt id="link-43" class="py-name"><a title="Bio.SeqFeature.AfterPosition.position
Bio.SeqFeature.BeforePosition.position
Bio.SeqFeature.BetweenPosition.position
Bio.SeqFeature.ExactPosition.position
Bio.SeqFeature.OneOfPosition.position
Bio.SeqFeature.UnknownPosition.position
Bio.SeqFeature.WithinPosition.position" class="py-name" href="#" onclick="return doclink('link-43', 'position', 'link-40');">position</a></tt> <tt class="py-op">+</tt> <tt class="py-name">offset</tt><tt class="py-op">,</tt> </tt>
<a name="L125"></a><tt class="py-lineno"> 125</tt>  <tt class="py-line">                            <tt class="py-name">pos</tt><tt class="py-op">.</tt><tt id="link-44" class="py-name"><a title="Bio.SeqFeature.AfterPosition.position
Bio.SeqFeature.BeforePosition.position
Bio.SeqFeature.BetweenPosition.position
Bio.SeqFeature.ExactPosition.position
Bio.SeqFeature.OneOfPosition.position
Bio.SeqFeature.UnknownPosition.position
Bio.SeqFeature.WithinPosition.position" class="py-name" href="#" onclick="return doclink('link-44', 'position', 'link-40');">position</a></tt> <tt class="py-op">+</tt> <tt class="py-name">pos</tt><tt class="py-op">.</tt><tt id="link-45" class="py-name" targets="Variable Bio.SeqFeature.AfterPosition.extension=Bio.SeqFeature.AfterPosition-class.html#extension,Variable Bio.SeqFeature.BeforePosition.extension=Bio.SeqFeature.BeforePosition-class.html#extension,Variable Bio.SeqFeature.BetweenPosition.extension=Bio.SeqFeature.BetweenPosition-class.html#extension,Variable Bio.SeqFeature.ExactPosition.extension=Bio.SeqFeature.ExactPosition-class.html#extension,Variable Bio.SeqFeature.OneOfPosition.extension=Bio.SeqFeature.OneOfPosition-class.html#extension,Variable Bio.SeqFeature.UnknownPosition.extension=Bio.SeqFeature.UnknownPosition-class.html#extension,Variable Bio.SeqFeature.WithinPosition.extension=Bio.SeqFeature.WithinPosition-class.html#extension"><a title="Bio.SeqFeature.AfterPosition.extension
Bio.SeqFeature.BeforePosition.extension
Bio.SeqFeature.BetweenPosition.extension
Bio.SeqFeature.ExactPosition.extension
Bio.SeqFeature.OneOfPosition.extension
Bio.SeqFeature.UnknownPosition.extension
Bio.SeqFeature.WithinPosition.extension" class="py-name" href="#" onclick="return doclink('link-45', 'extension', 'link-45');">extension</a></tt> <tt class="py-op">+</tt> <tt class="py-name">offset</tt><tt class="py-op">)</tt> </tt>
<a name="L126"></a><tt class="py-lineno"> 126</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">pos</tt><tt class="py-op">,</tt> <tt id="link-46" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-46', 'SeqFeature', 'link-16');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-47" class="py-name" targets="Class Bio.SeqFeature.BetweenPosition=Bio.SeqFeature.BetweenPosition-class.html"><a title="Bio.SeqFeature.BetweenPosition" class="py-name" href="#" onclick="return doclink('link-47', 'BetweenPosition', 'link-47');">BetweenPosition</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L127"></a><tt class="py-lineno"> 127</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">"(%i^%i)"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">pos</tt><tt class="py-op">.</tt><tt id="link-48" class="py-name"><a title="Bio.SeqFeature.AfterPosition.position
Bio.SeqFeature.BeforePosition.position
Bio.SeqFeature.BetweenPosition.position
Bio.SeqFeature.ExactPosition.position
Bio.SeqFeature.OneOfPosition.position
Bio.SeqFeature.UnknownPosition.position
Bio.SeqFeature.WithinPosition.position" class="py-name" href="#" onclick="return doclink('link-48', 'position', 'link-40');">position</a></tt> <tt class="py-op">+</tt> <tt class="py-name">offset</tt><tt class="py-op">,</tt> </tt>
<a name="L128"></a><tt class="py-lineno"> 128</tt>  <tt class="py-line">                            <tt class="py-name">pos</tt><tt class="py-op">.</tt><tt id="link-49" class="py-name"><a title="Bio.SeqFeature.AfterPosition.position
Bio.SeqFeature.BeforePosition.position
Bio.SeqFeature.BetweenPosition.position
Bio.SeqFeature.ExactPosition.position
Bio.SeqFeature.OneOfPosition.position
Bio.SeqFeature.UnknownPosition.position
Bio.SeqFeature.WithinPosition.position" class="py-name" href="#" onclick="return doclink('link-49', 'position', 'link-40');">position</a></tt> <tt class="py-op">+</tt> <tt class="py-name">pos</tt><tt class="py-op">.</tt><tt id="link-50" class="py-name"><a title="Bio.SeqFeature.AfterPosition.extension
Bio.SeqFeature.BeforePosition.extension
Bio.SeqFeature.BetweenPosition.extension
Bio.SeqFeature.ExactPosition.extension
Bio.SeqFeature.OneOfPosition.extension
Bio.SeqFeature.UnknownPosition.extension
Bio.SeqFeature.WithinPosition.extension" class="py-name" href="#" onclick="return doclink('link-50', 'extension', 'link-45');">extension</a></tt> <tt class="py-op">+</tt> <tt class="py-name">offset</tt><tt class="py-op">)</tt> </tt>
<a name="L129"></a><tt class="py-lineno"> 129</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">pos</tt><tt class="py-op">,</tt> <tt id="link-51" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-51', 'SeqFeature', 'link-16');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-52" class="py-name" targets="Class Bio.SeqFeature.BeforePosition=Bio.SeqFeature.BeforePosition-class.html"><a title="Bio.SeqFeature.BeforePosition" class="py-name" href="#" onclick="return doclink('link-52', 'BeforePosition', 'link-52');">BeforePosition</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L130"></a><tt class="py-lineno"> 130</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">"&lt;%i"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">pos</tt><tt class="py-op">.</tt><tt id="link-53" class="py-name"><a title="Bio.SeqFeature.AfterPosition.position
Bio.SeqFeature.BeforePosition.position
Bio.SeqFeature.BetweenPosition.position
Bio.SeqFeature.ExactPosition.position
Bio.SeqFeature.OneOfPosition.position
Bio.SeqFeature.UnknownPosition.position
Bio.SeqFeature.WithinPosition.position" class="py-name" href="#" onclick="return doclink('link-53', 'position', 'link-40');">position</a></tt> <tt class="py-op">+</tt> <tt class="py-name">offset</tt><tt class="py-op">)</tt> </tt>
<a name="L131"></a><tt class="py-lineno"> 131</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">pos</tt><tt class="py-op">,</tt> <tt id="link-54" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-54', 'SeqFeature', 'link-16');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-55" class="py-name" targets="Class Bio.SeqFeature.AfterPosition=Bio.SeqFeature.AfterPosition-class.html"><a title="Bio.SeqFeature.AfterPosition" class="py-name" href="#" onclick="return doclink('link-55', 'AfterPosition', 'link-55');">AfterPosition</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L132"></a><tt class="py-lineno"> 132</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">"&gt;%i"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">pos</tt><tt class="py-op">.</tt><tt id="link-56" class="py-name"><a title="Bio.SeqFeature.AfterPosition.position
Bio.SeqFeature.BeforePosition.position
Bio.SeqFeature.BetweenPosition.position
Bio.SeqFeature.ExactPosition.position
Bio.SeqFeature.OneOfPosition.position
Bio.SeqFeature.UnknownPosition.position
Bio.SeqFeature.WithinPosition.position" class="py-name" href="#" onclick="return doclink('link-56', 'position', 'link-40');">position</a></tt> <tt class="py-op">+</tt> <tt class="py-name">offset</tt><tt class="py-op">)</tt> </tt>
<a name="L133"></a><tt class="py-lineno"> 133</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">pos</tt><tt class="py-op">,</tt> <tt id="link-57" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-57', 'SeqFeature', 'link-16');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-58" class="py-name" targets="Class Bio.SeqFeature.OneOfPosition=Bio.SeqFeature.OneOfPosition-class.html"><a title="Bio.SeqFeature.OneOfPosition" class="py-name" href="#" onclick="return doclink('link-58', 'OneOfPosition', 'link-58');">OneOfPosition</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L134"></a><tt class="py-lineno"> 134</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">"one-of(%s)"</tt> \ </tt>
<a name="L135"></a><tt class="py-lineno"> 135</tt>  <tt class="py-line">               <tt class="py-op">%</tt> <tt class="py-string">","</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt id="link-59" class="py-name" targets="Function Bio.SeqIO.InsdcIO._insdc_feature_position_string()=Bio.SeqIO.InsdcIO-module.html#_insdc_feature_position_string"><a title="Bio.SeqIO.InsdcIO._insdc_feature_position_string" class="py-name" href="#" onclick="return doclink('link-59', '_insdc_feature_position_string', 'link-59');">_insdc_feature_position_string</a></tt><tt class="py-op">(</tt><tt class="py-name">p</tt><tt class="py-op">,</tt> <tt class="py-name">offset</tt><tt class="py-op">)</tt> </tt>
<a name="L136"></a><tt class="py-lineno"> 136</tt>  <tt class="py-line">                          <tt class="py-keyword">for</tt> <tt class="py-name">p</tt> <tt class="py-keyword">in</tt> <tt class="py-name">pos</tt><tt class="py-op">.</tt><tt class="py-name">position_choices</tt><tt class="py-op">)</tt> </tt>
<a name="L137"></a><tt class="py-lineno"> 137</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">pos</tt><tt class="py-op">,</tt> <tt id="link-60" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-60', 'SeqFeature', 'link-16');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-61" class="py-name" targets="Class Bio.SeqFeature.AbstractPosition=Bio.SeqFeature.AbstractPosition-class.html"><a title="Bio.SeqFeature.AbstractPosition" class="py-name" href="#" onclick="return doclink('link-61', 'AbstractPosition', 'link-61');">AbstractPosition</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L138"></a><tt class="py-lineno"> 138</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">NotImplementedError</tt><tt class="py-op">(</tt><tt class="py-string">"Please report this as a bug in Biopython."</tt><tt class="py-op">)</tt> </tt>
<a name="L139"></a><tt class="py-lineno"> 139</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L140"></a><tt class="py-lineno"> 140</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Expected a SeqFeature position object."</tt><tt class="py-op">)</tt> </tt>
</div><a name="L141"></a><tt class="py-lineno"> 141</tt>  <tt class="py-line"> </tt>
<a name="L142"></a><tt class="py-lineno"> 142</tt>  <tt class="py-line"> </tt>
<a name="_insdc_location_string_ignoring_strand_and_subfeatures"></a><div id="_insdc_location_string_ignoring_strand_and_subfeatures-def"><a name="L143"></a><tt class="py-lineno"> 143</tt> <a class="py-toggle" href="#" id="_insdc_location_string_ignoring_strand_and_subfeatures-toggle" onclick="return toggle('_insdc_location_string_ignoring_strand_and_subfeatures');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO-module.html#_insdc_location_string_ignoring_strand_and_subfeatures">_insdc_location_string_ignoring_strand_and_subfeatures</a><tt class="py-op">(</tt><tt class="py-param">location</tt><tt class="py-op">,</tt> <tt class="py-param">rec_length</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_insdc_location_string_ignoring_strand_and_subfeatures-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_insdc_location_string_ignoring_strand_and_subfeatures-expanded"><a name="L144"></a><tt class="py-lineno"> 144</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt id="link-62" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.location()=Bio.GenBank._FeatureConsumer-class.html#location,Method Bio.GenBank._RecordConsumer.location()=Bio.GenBank._RecordConsumer-class.html#location,Method Bio.Phylo.PhyloXMLIO.Writer.location()=Bio.Phylo.PhyloXMLIO.Writer-class.html#location"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-62', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-63" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.ref=Bio.SeqFeature.CompoundLocation-class.html#ref,Variable Bio.SeqFeature.SeqFeature.ref=Bio.SeqFeature.SeqFeature-class.html#ref"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-63', 'ref', 'link-63');">ref</a></tt><tt class="py-op">:</tt> </tt>
<a name="L145"></a><tt class="py-lineno"> 145</tt>  <tt class="py-line">        <tt id="link-64" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-64', 'ref', 'link-63');">ref</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"%s:"</tt> <tt class="py-op">%</tt> <tt id="link-65" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-65', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-66" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-66', 'ref', 'link-63');">ref</a></tt> </tt>
<a name="L146"></a><tt class="py-lineno"> 146</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L147"></a><tt class="py-lineno"> 147</tt>  <tt class="py-line">        <tt id="link-67" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-67', 'ref', 'link-63');">ref</a></tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L148"></a><tt class="py-lineno"> 148</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt id="link-68" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-68', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-69" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.ref_db=Bio.SeqFeature.CompoundLocation-class.html#ref_db,Variable Bio.SeqFeature.SeqFeature.ref_db=Bio.SeqFeature.SeqFeature-class.html#ref_db"><a title="Bio.SeqFeature.CompoundLocation.ref_db
Bio.SeqFeature.SeqFeature.ref_db" class="py-name" href="#" onclick="return doclink('link-69', 'ref_db', 'link-69');">ref_db</a></tt> </tt>
<a name="L149"></a><tt class="py-lineno"> 149</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-70" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-70', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-71" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.start=Bio.SeqFeature.CompoundLocation-class.html#start,Variable Bio.SeqFeature.FeatureLocation.start=Bio.SeqFeature.FeatureLocation-class.html#start"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-71', 'start', 'link-71');">start</a></tt><tt class="py-op">,</tt> <tt id="link-72" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-72', 'SeqFeature', 'link-16');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-73" class="py-name"><a title="Bio.SeqFeature.ExactPosition" class="py-name" href="#" onclick="return doclink('link-73', 'ExactPosition', 'link-39');">ExactPosition</a></tt><tt class="py-op">)</tt> \ </tt>
<a name="L150"></a><tt class="py-lineno"> 150</tt>  <tt class="py-line">        <tt class="py-keyword">and</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-74" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-74', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-75" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.end=Bio.SeqFeature.CompoundLocation-class.html#end,Variable Bio.SeqFeature.FeatureLocation.end=Bio.SeqFeature.FeatureLocation-class.html#end"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-75', 'end', 'link-75');">end</a></tt><tt class="py-op">,</tt> <tt id="link-76" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-76', 'SeqFeature', 'link-16');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-77" class="py-name"><a title="Bio.SeqFeature.ExactPosition" class="py-name" href="#" onclick="return doclink('link-77', 'ExactPosition', 'link-39');">ExactPosition</a></tt><tt class="py-op">)</tt> \ </tt>
<a name="L151"></a><tt class="py-lineno"> 151</tt>  <tt class="py-line">            <tt class="py-keyword">and</tt> <tt id="link-78" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-78', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-79" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-79', 'start', 'link-71');">start</a></tt><tt class="py-op">.</tt><tt id="link-80" class="py-name"><a title="Bio.SeqFeature.AfterPosition.position
Bio.SeqFeature.BeforePosition.position
Bio.SeqFeature.BetweenPosition.position
Bio.SeqFeature.ExactPosition.position
Bio.SeqFeature.OneOfPosition.position
Bio.SeqFeature.UnknownPosition.position
Bio.SeqFeature.WithinPosition.position" class="py-name" href="#" onclick="return doclink('link-80', 'position', 'link-40');">position</a></tt> <tt class="py-op">==</tt> <tt id="link-81" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-81', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-82" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-82', 'end', 'link-75');">end</a></tt><tt class="py-op">.</tt><tt id="link-83" class="py-name"><a title="Bio.SeqFeature.AfterPosition.position
Bio.SeqFeature.BeforePosition.position
Bio.SeqFeature.BetweenPosition.position
Bio.SeqFeature.ExactPosition.position
Bio.SeqFeature.OneOfPosition.position
Bio.SeqFeature.UnknownPosition.position
Bio.SeqFeature.WithinPosition.position" class="py-name" href="#" onclick="return doclink('link-83', 'position', 'link-40');">position</a></tt><tt class="py-op">:</tt> </tt>
<a name="L152"></a><tt class="py-lineno"> 152</tt>  <tt class="py-line">        <tt class="py-comment"># Special case, for 12:12 return 12^13</tt> </tt>
<a name="L153"></a><tt class="py-lineno"> 153</tt>  <tt class="py-line">        <tt class="py-comment"># (a zero length slice, meaning the point between two letters)</tt> </tt>
<a name="L154"></a><tt class="py-lineno"> 154</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-84" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-84', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-85" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-85', 'end', 'link-75');">end</a></tt><tt class="py-op">.</tt><tt id="link-86" class="py-name"><a title="Bio.SeqFeature.AfterPosition.position
Bio.SeqFeature.BeforePosition.position
Bio.SeqFeature.BetweenPosition.position
Bio.SeqFeature.ExactPosition.position
Bio.SeqFeature.OneOfPosition.position
Bio.SeqFeature.UnknownPosition.position
Bio.SeqFeature.WithinPosition.position" class="py-name" href="#" onclick="return doclink('link-86', 'position', 'link-40');">position</a></tt> <tt class="py-op">==</tt> <tt class="py-name">rec_length</tt><tt class="py-op">:</tt> </tt>
<a name="L155"></a><tt class="py-lineno"> 155</tt>  <tt class="py-line">            <tt class="py-comment"># Very special case, for a between position at the end of a</tt> </tt>
<a name="L156"></a><tt class="py-lineno"> 156</tt>  <tt class="py-line">            <tt class="py-comment"># sequence (used on some circular genomes, Bug 3098) we have</tt> </tt>
<a name="L157"></a><tt class="py-lineno"> 157</tt>  <tt class="py-line">            <tt class="py-comment"># N:N so return N^1</tt> </tt>
<a name="L158"></a><tt class="py-lineno"> 158</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-string">"%s%i^1"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-87" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-87', 'ref', 'link-63');">ref</a></tt><tt class="py-op">,</tt> <tt class="py-name">rec_length</tt><tt class="py-op">)</tt> </tt>
<a name="L159"></a><tt class="py-lineno"> 159</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L160"></a><tt class="py-lineno"> 160</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-string">"%s%i^%i"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-88" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-88', 'ref', 'link-63');">ref</a></tt><tt class="py-op">,</tt> <tt id="link-89" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-89', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-90" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-90', 'end', 'link-75');">end</a></tt><tt class="py-op">.</tt><tt id="link-91" class="py-name"><a title="Bio.SeqFeature.AfterPosition.position
Bio.SeqFeature.BeforePosition.position
Bio.SeqFeature.BetweenPosition.position
Bio.SeqFeature.ExactPosition.position
Bio.SeqFeature.OneOfPosition.position
Bio.SeqFeature.UnknownPosition.position
Bio.SeqFeature.WithinPosition.position" class="py-name" href="#" onclick="return doclink('link-91', 'position', 'link-40');">position</a></tt><tt class="py-op">,</tt> </tt>
<a name="L161"></a><tt class="py-lineno"> 161</tt>  <tt class="py-line">                                <tt id="link-92" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-92', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-93" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-93', 'end', 'link-75');">end</a></tt><tt class="py-op">.</tt><tt id="link-94" class="py-name"><a title="Bio.SeqFeature.AfterPosition.position
Bio.SeqFeature.BeforePosition.position
Bio.SeqFeature.BetweenPosition.position
Bio.SeqFeature.ExactPosition.position
Bio.SeqFeature.OneOfPosition.position
Bio.SeqFeature.UnknownPosition.position
Bio.SeqFeature.WithinPosition.position" class="py-name" href="#" onclick="return doclink('link-94', 'position', 'link-40');">position</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L162"></a><tt class="py-lineno"> 162</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-95" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-95', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-96" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-96', 'start', 'link-71');">start</a></tt><tt class="py-op">,</tt> <tt id="link-97" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-97', 'SeqFeature', 'link-16');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-98" class="py-name"><a title="Bio.SeqFeature.ExactPosition" class="py-name" href="#" onclick="return doclink('link-98', 'ExactPosition', 'link-39');">ExactPosition</a></tt><tt class="py-op">)</tt> \ </tt>
<a name="L163"></a><tt class="py-lineno"> 163</tt>  <tt class="py-line">        <tt class="py-keyword">and</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-99" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-99', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-100" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-100', 'end', 'link-75');">end</a></tt><tt class="py-op">,</tt> <tt id="link-101" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-101', 'SeqFeature', 'link-16');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-102" class="py-name"><a title="Bio.SeqFeature.ExactPosition" class="py-name" href="#" onclick="return doclink('link-102', 'ExactPosition', 'link-39');">ExactPosition</a></tt><tt class="py-op">)</tt> \ </tt>
<a name="L164"></a><tt class="py-lineno"> 164</tt>  <tt class="py-line">            <tt class="py-keyword">and</tt> <tt id="link-103" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-103', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-104" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-104', 'start', 'link-71');">start</a></tt><tt class="py-op">.</tt><tt id="link-105" class="py-name"><a title="Bio.SeqFeature.AfterPosition.position
Bio.SeqFeature.BeforePosition.position
Bio.SeqFeature.BetweenPosition.position
Bio.SeqFeature.ExactPosition.position
Bio.SeqFeature.OneOfPosition.position
Bio.SeqFeature.UnknownPosition.position
Bio.SeqFeature.WithinPosition.position" class="py-name" href="#" onclick="return doclink('link-105', 'position', 'link-40');">position</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt> <tt class="py-op">==</tt> <tt id="link-106" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-106', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-107" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-107', 'end', 'link-75');">end</a></tt><tt class="py-op">.</tt><tt id="link-108" class="py-name"><a title="Bio.SeqFeature.AfterPosition.position
Bio.SeqFeature.BeforePosition.position
Bio.SeqFeature.BetweenPosition.position
Bio.SeqFeature.ExactPosition.position
Bio.SeqFeature.OneOfPosition.position
Bio.SeqFeature.UnknownPosition.position
Bio.SeqFeature.WithinPosition.position" class="py-name" href="#" onclick="return doclink('link-108', 'position', 'link-40');">position</a></tt><tt class="py-op">:</tt> </tt>
<a name="L165"></a><tt class="py-lineno"> 165</tt>  <tt class="py-line">        <tt class="py-comment"># Special case, for 11:12 return 12 rather than 12..12</tt> </tt>
<a name="L166"></a><tt class="py-lineno"> 166</tt>  <tt class="py-line">        <tt class="py-comment"># (a length one slice, meaning a single letter)</tt> </tt>
<a name="L167"></a><tt class="py-lineno"> 167</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">"%s%i"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-109" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-109', 'ref', 'link-63');">ref</a></tt><tt class="py-op">,</tt> <tt id="link-110" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-110', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-111" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-111', 'end', 'link-75');">end</a></tt><tt class="py-op">.</tt><tt id="link-112" class="py-name"><a title="Bio.SeqFeature.AfterPosition.position
Bio.SeqFeature.BeforePosition.position
Bio.SeqFeature.BetweenPosition.position
Bio.SeqFeature.ExactPosition.position
Bio.SeqFeature.OneOfPosition.position
Bio.SeqFeature.UnknownPosition.position
Bio.SeqFeature.WithinPosition.position" class="py-name" href="#" onclick="return doclink('link-112', 'position', 'link-40');">position</a></tt><tt class="py-op">)</tt> </tt>
<a name="L168"></a><tt class="py-lineno"> 168</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-113" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-113', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-114" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-114', 'start', 'link-71');">start</a></tt><tt class="py-op">,</tt> <tt id="link-115" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-115', 'SeqFeature', 'link-16');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-116" class="py-name" targets="Class Bio.SeqFeature.UnknownPosition=Bio.SeqFeature.UnknownPosition-class.html"><a title="Bio.SeqFeature.UnknownPosition" class="py-name" href="#" onclick="return doclink('link-116', 'UnknownPosition', 'link-116');">UnknownPosition</a></tt><tt class="py-op">)</tt> \ </tt>
<a name="L169"></a><tt class="py-lineno"> 169</tt>  <tt class="py-line">            <tt class="py-keyword">or</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-117" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-117', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-118" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-118', 'end', 'link-75');">end</a></tt><tt class="py-op">,</tt> <tt id="link-119" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-119', 'SeqFeature', 'link-16');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-120" class="py-name"><a title="Bio.SeqFeature.UnknownPosition" class="py-name" href="#" onclick="return doclink('link-120', 'UnknownPosition', 'link-116');">UnknownPosition</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L170"></a><tt class="py-lineno"> 170</tt>  <tt class="py-line">        <tt class="py-comment"># Special case for features from SwissProt/UniProt files</tt> </tt>
<a name="L171"></a><tt class="py-lineno"> 171</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-121" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-121', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-122" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-122', 'start', 'link-71');">start</a></tt><tt class="py-op">,</tt> <tt id="link-123" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-123', 'SeqFeature', 'link-16');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-124" class="py-name"><a title="Bio.SeqFeature.UnknownPosition" class="py-name" href="#" onclick="return doclink('link-124', 'UnknownPosition', 'link-116');">UnknownPosition</a></tt><tt class="py-op">)</tt> \ </tt>
<a name="L172"></a><tt class="py-lineno"> 172</tt>  <tt class="py-line">                <tt class="py-keyword">and</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-125" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-125', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-126" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-126', 'end', 'link-75');">end</a></tt><tt class="py-op">,</tt> <tt id="link-127" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-127', 'SeqFeature', 'link-16');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-128" class="py-name"><a title="Bio.SeqFeature.UnknownPosition" class="py-name" href="#" onclick="return doclink('link-128', 'UnknownPosition', 'link-116');">UnknownPosition</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L173"></a><tt class="py-lineno"> 173</tt>  <tt class="py-line">            <tt class="py-comment"># warnings.warn("Feature with unknown location", BiopythonWarning)</tt> </tt>
<a name="L174"></a><tt class="py-lineno"> 174</tt>  <tt class="py-line">            <tt class="py-comment"># return "?"</tt> </tt>
<a name="L175"></a><tt class="py-lineno"> 175</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Feature with unknown location"</tt><tt class="py-op">)</tt> </tt>
<a name="L176"></a><tt class="py-lineno"> 176</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-129" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-129', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-130" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-130', 'start', 'link-71');">start</a></tt><tt class="py-op">,</tt> <tt id="link-131" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-131', 'SeqFeature', 'link-16');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-132" class="py-name"><a title="Bio.SeqFeature.UnknownPosition" class="py-name" href="#" onclick="return doclink('link-132', 'UnknownPosition', 'link-116');">UnknownPosition</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L177"></a><tt class="py-lineno"> 177</tt>  <tt class="py-line">            <tt class="py-comment"># Treat the unknown start position as a BeforePosition</tt> </tt>
<a name="L178"></a><tt class="py-lineno"> 178</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-string">"%s&lt;%i..%s"</tt> \ </tt>
<a name="L179"></a><tt class="py-lineno"> 179</tt>  <tt class="py-line">                <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-133" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-133', 'ref', 'link-63');">ref</a></tt><tt class="py-op">,</tt> </tt>
<a name="L180"></a><tt class="py-lineno"> 180</tt>  <tt class="py-line">                   <tt id="link-134" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-134', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-135" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.nofuzzy_end=Bio.SeqFeature.CompoundLocation-class.html#nofuzzy_end,Variable Bio.SeqFeature.FeatureLocation.nofuzzy_end=Bio.SeqFeature.FeatureLocation-class.html#nofuzzy_end"><a title="Bio.SeqFeature.CompoundLocation.nofuzzy_end
Bio.SeqFeature.FeatureLocation.nofuzzy_end" class="py-name" href="#" onclick="return doclink('link-135', 'nofuzzy_end', 'link-135');">nofuzzy_end</a></tt><tt class="py-op">,</tt> </tt>
<a name="L181"></a><tt class="py-lineno"> 181</tt>  <tt class="py-line">                   <tt id="link-136" class="py-name"><a title="Bio.SeqIO.InsdcIO._insdc_feature_position_string" class="py-name" href="#" onclick="return doclink('link-136', '_insdc_feature_position_string', 'link-59');">_insdc_feature_position_string</a></tt><tt class="py-op">(</tt><tt id="link-137" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-137', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-138" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-138', 'end', 'link-75');">end</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L182"></a><tt class="py-lineno"> 182</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L183"></a><tt class="py-lineno"> 183</tt>  <tt class="py-line">            <tt class="py-comment"># Treat the unknown end position as an AfterPosition</tt> </tt>
<a name="L184"></a><tt class="py-lineno"> 184</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-string">"%s%s..&gt;%i"</tt> \ </tt>
<a name="L185"></a><tt class="py-lineno"> 185</tt>  <tt class="py-line">                <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-139" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-139', 'ref', 'link-63');">ref</a></tt><tt class="py-op">,</tt> </tt>
<a name="L186"></a><tt class="py-lineno"> 186</tt>  <tt class="py-line">                   <tt id="link-140" class="py-name"><a title="Bio.SeqIO.InsdcIO._insdc_feature_position_string" class="py-name" href="#" onclick="return doclink('link-140', '_insdc_feature_position_string', 'link-59');">_insdc_feature_position_string</a></tt><tt class="py-op">(</tt><tt id="link-141" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-141', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-142" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-142', 'start', 'link-71');">start</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L187"></a><tt class="py-lineno"> 187</tt>  <tt class="py-line">                   <tt id="link-143" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-143', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-144" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.nofuzzy_start=Bio.SeqFeature.CompoundLocation-class.html#nofuzzy_start,Variable Bio.SeqFeature.FeatureLocation.nofuzzy_start=Bio.SeqFeature.FeatureLocation-class.html#nofuzzy_start"><a title="Bio.SeqFeature.CompoundLocation.nofuzzy_start
Bio.SeqFeature.FeatureLocation.nofuzzy_start" class="py-name" href="#" onclick="return doclink('link-144', 'nofuzzy_start', 'link-144');">nofuzzy_start</a></tt><tt class="py-op">)</tt> </tt>
<a name="L188"></a><tt class="py-lineno"> 188</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L189"></a><tt class="py-lineno"> 189</tt>  <tt class="py-line">        <tt class="py-comment"># Typical case, e.g. 12..15 gets mapped to 11:15</tt> </tt>
<a name="L190"></a><tt class="py-lineno"> 190</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-145" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-145', 'ref', 'link-63');">ref</a></tt> \ </tt>
<a name="L191"></a><tt class="py-lineno"> 191</tt>  <tt class="py-line">            <tt class="py-op">+</tt> <tt id="link-146" class="py-name"><a title="Bio.SeqIO.InsdcIO._insdc_feature_position_string" class="py-name" href="#" onclick="return doclink('link-146', '_insdc_feature_position_string', 'link-59');">_insdc_feature_position_string</a></tt><tt class="py-op">(</tt><tt id="link-147" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-147', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-148" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-148', 'start', 'link-71');">start</a></tt><tt class="py-op">,</tt> <tt class="py-op">+</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> \ </tt>
<a name="L192"></a><tt class="py-lineno"> 192</tt>  <tt class="py-line">            <tt class="py-op">+</tt> <tt class="py-string">".."</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L193"></a><tt class="py-lineno"> 193</tt>  <tt class="py-line">            <tt id="link-149" class="py-name"><a title="Bio.SeqIO.InsdcIO._insdc_feature_position_string" class="py-name" href="#" onclick="return doclink('link-149', '_insdc_feature_position_string', 'link-59');">_insdc_feature_position_string</a></tt><tt class="py-op">(</tt><tt id="link-150" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-150', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-151" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-151', 'end', 'link-75');">end</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L194"></a><tt class="py-lineno"> 194</tt>  <tt class="py-line"> </tt>
<a name="L195"></a><tt class="py-lineno"> 195</tt>  <tt class="py-line"> </tt>
<a name="_insdc_location_string"></a><div id="_insdc_location_string-def"><a name="L196"></a><tt class="py-lineno"> 196</tt> <a class="py-toggle" href="#" id="_insdc_location_string-toggle" onclick="return toggle('_insdc_location_string');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO-module.html#_insdc_location_string">_insdc_location_string</a><tt class="py-op">(</tt><tt class="py-param">location</tt><tt class="py-op">,</tt> <tt class="py-param">rec_length</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_insdc_location_string-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_insdc_location_string-expanded"><a name="L197"></a><tt class="py-lineno"> 197</tt>  <tt class="py-line">    <tt class="py-docstring">"""Build a GenBank/EMBL location from a (Compound) FeatureLocation (PRIVATE).</tt> </tt>
<a name="L198"></a><tt class="py-lineno"> 198</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L199"></a><tt class="py-lineno"> 199</tt>  <tt class="py-line"><tt class="py-docstring">    There is a choice of how to show joins on the reverse complement strand,</tt> </tt>
<a name="L200"></a><tt class="py-lineno"> 200</tt>  <tt class="py-line"><tt class="py-docstring">    GenBank used "complement(join(1,10),(20,100))" while EMBL used to use</tt> </tt>
<a name="L201"></a><tt class="py-lineno"> 201</tt>  <tt class="py-line"><tt class="py-docstring">    "join(complement(20,100),complement(1,10))" instead (but appears to have</tt> </tt>
<a name="L202"></a><tt class="py-lineno"> 202</tt>  <tt class="py-line"><tt class="py-docstring">    now adopted the GenBank convention). Notice that the order of the entries</tt> </tt>
<a name="L203"></a><tt class="py-lineno"> 203</tt>  <tt class="py-line"><tt class="py-docstring">    is reversed! This function therefore uses the first form. In this situation</tt> </tt>
<a name="L204"></a><tt class="py-lineno"> 204</tt>  <tt class="py-line"><tt class="py-docstring">    we expect the CompoundFeatureLocation and its parts to all be marked as</tt> </tt>
<a name="L205"></a><tt class="py-lineno"> 205</tt>  <tt class="py-line"><tt class="py-docstring">    strand == -1, and to be in the order 19:100 then 0:10.</tt> </tt>
<a name="L206"></a><tt class="py-lineno"> 206</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L207"></a><tt class="py-lineno"> 207</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L208"></a><tt class="py-lineno"> 208</tt>  <tt class="py-line">        <tt id="link-152" class="py-name" targets="Variable Bio.SeqFeature.FeatureLocation.parts=Bio.SeqFeature.FeatureLocation-class.html#parts"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-152', 'parts', 'link-152');">parts</a></tt> <tt class="py-op">=</tt> <tt id="link-153" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-153', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-154" class="py-name"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-154', 'parts', 'link-152');">parts</a></tt> </tt>
<a name="L209"></a><tt class="py-lineno"> 209</tt>  <tt class="py-line">        <tt class="py-comment"># CompoundFeatureLocation</tt> </tt>
<a name="L210"></a><tt class="py-lineno"> 210</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-155" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-155', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-156" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.strand()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#strand,Variable Bio.SeqFeature.CompoundLocation.strand=Bio.SeqFeature.CompoundLocation-class.html#strand,Variable Bio.SeqFeature.FeatureLocation.strand=Bio.SeqFeature.FeatureLocation-class.html#strand,Variable Bio.SeqFeature.SeqFeature.strand=Bio.SeqFeature.SeqFeature-class.html#strand"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-156', 'strand', 'link-156');">strand</a></tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L211"></a><tt class="py-lineno"> 211</tt>  <tt class="py-line">            <tt class="py-comment"># Special case, put complement outside the join/order/... and reverse order</tt> </tt>
<a name="L212"></a><tt class="py-lineno"> 212</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-string">"complement(%s(%s))"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-157" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-157', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt class="py-name">operator</tt><tt class="py-op">,</tt> </tt>
<a name="L213"></a><tt class="py-lineno"> 213</tt>  <tt class="py-line">                   <tt class="py-string">","</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt id="link-158" class="py-name" targets="Function Bio.SeqIO.InsdcIO._insdc_location_string_ignoring_strand_and_subfeatures()=Bio.SeqIO.InsdcIO-module.html#_insdc_location_string_ignoring_strand_and_subfeatures"><a title="Bio.SeqIO.InsdcIO._insdc_location_string_ignoring_strand_and_subfeatures" class="py-name" href="#" onclick="return doclink('link-158', '_insdc_location_string_ignoring_strand_and_subfeatures', 'link-158');">_insdc_location_string_ignoring_strand_and_subfeatures</a></tt><tt class="py-op">(</tt><tt class="py-name">p</tt><tt class="py-op">,</tt> <tt class="py-name">rec_length</tt><tt class="py-op">)</tt> </tt>
<a name="L214"></a><tt class="py-lineno"> 214</tt>  <tt class="py-line">                            <tt class="py-keyword">for</tt> <tt class="py-name">p</tt> <tt class="py-keyword">in</tt> <tt id="link-159" class="py-name"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-159', 'parts', 'link-152');">parts</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L215"></a><tt class="py-lineno"> 215</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L216"></a><tt class="py-lineno"> 216</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-string">"%s(%s)"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-160" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-160', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt class="py-name">operator</tt><tt class="py-op">,</tt> </tt>
<a name="L217"></a><tt class="py-lineno"> 217</tt>  <tt class="py-line">                   <tt class="py-string">","</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt id="link-161" class="py-name" targets="Function Bio.SeqIO.InsdcIO._insdc_location_string()=Bio.SeqIO.InsdcIO-module.html#_insdc_location_string"><a title="Bio.SeqIO.InsdcIO._insdc_location_string" class="py-name" href="#" onclick="return doclink('link-161', '_insdc_location_string', 'link-161');">_insdc_location_string</a></tt><tt class="py-op">(</tt><tt class="py-name">p</tt><tt class="py-op">,</tt> <tt class="py-name">rec_length</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">p</tt> <tt class="py-keyword">in</tt> <tt id="link-162" class="py-name"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-162', 'parts', 'link-152');">parts</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L218"></a><tt class="py-lineno"> 218</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L219"></a><tt class="py-lineno"> 219</tt>  <tt class="py-line">        <tt class="py-comment"># Simple FeatureLocation</tt> </tt>
<a name="L220"></a><tt class="py-lineno"> 220</tt>  <tt class="py-line">        <tt class="py-name">loc</tt> <tt class="py-op">=</tt> <tt id="link-163" class="py-name"><a title="Bio.SeqIO.InsdcIO._insdc_location_string_ignoring_strand_and_subfeatures" class="py-name" href="#" onclick="return doclink('link-163', '_insdc_location_string_ignoring_strand_and_subfeatures', 'link-158');">_insdc_location_string_ignoring_strand_and_subfeatures</a></tt><tt class="py-op">(</tt><tt id="link-164" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-164', 'location', 'link-62');">location</a></tt><tt class="py-op">,</tt> <tt class="py-name">rec_length</tt><tt class="py-op">)</tt> </tt>
<a name="L221"></a><tt class="py-lineno"> 221</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-165" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-165', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-166" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-166', 'strand', 'link-156');">strand</a></tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L222"></a><tt class="py-lineno"> 222</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-string">"complement(%s)"</tt> <tt class="py-op">%</tt> <tt class="py-name">loc</tt> </tt>
<a name="L223"></a><tt class="py-lineno"> 223</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L224"></a><tt class="py-lineno"> 224</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">loc</tt> </tt>
</div><a name="L225"></a><tt class="py-lineno"> 225</tt>  <tt class="py-line"> </tt>
<a name="L226"></a><tt class="py-lineno"> 226</tt>  <tt class="py-line"> </tt>
<a name="_insdc_feature_location_string"></a><div id="_insdc_feature_location_string-def"><a name="L227"></a><tt class="py-lineno"> 227</tt> <a class="py-toggle" href="#" id="_insdc_feature_location_string-toggle" onclick="return toggle('_insdc_feature_location_string');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO-module.html#_insdc_feature_location_string">_insdc_feature_location_string</a><tt class="py-op">(</tt><tt class="py-param">feature</tt><tt class="py-op">,</tt> <tt class="py-param">rec_length</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_insdc_feature_location_string-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_insdc_feature_location_string-expanded"><a name="L228"></a><tt class="py-lineno"> 228</tt>  <tt class="py-line">    <tt class="py-docstring">"""Build a GenBank/EMBL location string from a SeqFeature (PRIVATE, OBSOLETE).</tt> </tt>
<a name="L229"></a><tt class="py-lineno"> 229</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L230"></a><tt class="py-lineno"> 230</tt>  <tt class="py-line"><tt class="py-docstring">    There is a choice of how to show joins on the reverse complement strand,</tt> </tt>
<a name="L231"></a><tt class="py-lineno"> 231</tt>  <tt class="py-line"><tt class="py-docstring">    GenBank used "complement(join(1,10),(20,100))" while EMBL used to use</tt> </tt>
<a name="L232"></a><tt class="py-lineno"> 232</tt>  <tt class="py-line"><tt class="py-docstring">    "join(complement(20,100),complement(1,10))" instead (but appears to have</tt> </tt>
<a name="L233"></a><tt class="py-lineno"> 233</tt>  <tt class="py-line"><tt class="py-docstring">    now adopted the GenBank convention). Notice that the order of the entries</tt> </tt>
<a name="L234"></a><tt class="py-lineno"> 234</tt>  <tt class="py-line"><tt class="py-docstring">    is reversed! This function therefore uses the first form. In this situation</tt> </tt>
<a name="L235"></a><tt class="py-lineno"> 235</tt>  <tt class="py-line"><tt class="py-docstring">    we expect the parent feature and the two children to all be marked as</tt> </tt>
<a name="L236"></a><tt class="py-lineno"> 236</tt>  <tt class="py-line"><tt class="py-docstring">    strand == -1, and in the order 0:10 then 19:100.</tt> </tt>
<a name="L237"></a><tt class="py-lineno"> 237</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L238"></a><tt class="py-lineno"> 238</tt>  <tt class="py-line"><tt class="py-docstring">    Also need to consider dual-strand examples like these from the Arabidopsis</tt> </tt>
<a name="L239"></a><tt class="py-lineno"> 239</tt>  <tt class="py-line"><tt class="py-docstring">    thaliana chloroplast NC_000932: join(complement(69611..69724),139856..140650)</tt> </tt>
<a name="L240"></a><tt class="py-lineno"> 240</tt>  <tt class="py-line"><tt class="py-docstring">    gene ArthCp047, GeneID:844801 or its CDS (protein NP_051038.1 GI:7525057)</tt> </tt>
<a name="L241"></a><tt class="py-lineno"> 241</tt>  <tt class="py-line"><tt class="py-docstring">    which is further complicated by a splice:</tt> </tt>
<a name="L242"></a><tt class="py-lineno"> 242</tt>  <tt class="py-line"><tt class="py-docstring">    join(complement(69611..69724),139856..140087,140625..140650)</tt> </tt>
<a name="L243"></a><tt class="py-lineno"> 243</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L244"></a><tt class="py-lineno"> 244</tt>  <tt class="py-line"><tt class="py-docstring">    For this mixed strand feature, the parent SeqFeature should have</tt> </tt>
<a name="L245"></a><tt class="py-lineno"> 245</tt>  <tt class="py-line"><tt class="py-docstring">    no strand (either 0 or None) while the child features should have either</tt> </tt>
<a name="L246"></a><tt class="py-lineno"> 246</tt>  <tt class="py-line"><tt class="py-docstring">    strand +1 or -1 as appropriate, and be listed in the order given here.</tt> </tt>
<a name="L247"></a><tt class="py-lineno"> 247</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L248"></a><tt class="py-lineno"> 248</tt>  <tt class="py-line">    <tt class="py-comment"># Using private variable to avoid deprecation warning</tt> </tt>
<a name="L249"></a><tt class="py-lineno"> 249</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">_sub_features</tt><tt class="py-op">:</tt> </tt>
<a name="L250"></a><tt class="py-lineno"> 250</tt>  <tt class="py-line">        <tt class="py-comment"># Non-recursive.</tt> </tt>
<a name="L251"></a><tt class="py-lineno"> 251</tt>  <tt class="py-line">        <tt class="py-comment"># assert feature.location_operator == "", \</tt> </tt>
<a name="L252"></a><tt class="py-lineno"> 252</tt>  <tt class="py-line">        <tt class="py-comment">#       "%s has no subfeatures but location_operator %s" \</tt> </tt>
<a name="L253"></a><tt class="py-lineno"> 253</tt>  <tt class="py-line">        <tt class="py-comment">#       % (repr(feature), feature.location_operator)</tt> </tt>
<a name="L254"></a><tt class="py-lineno"> 254</tt>  <tt class="py-line">        <tt id="link-167" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-167', 'location', 'link-62');">location</a></tt> <tt class="py-op">=</tt> <tt id="link-168" class="py-name"><a title="Bio.SeqIO.InsdcIO._insdc_location_string_ignoring_strand_and_subfeatures" class="py-name" href="#" onclick="return doclink('link-168', '_insdc_location_string_ignoring_strand_and_subfeatures', 'link-158');">_insdc_location_string_ignoring_strand_and_subfeatures</a></tt><tt class="py-op">(</tt> </tt>
<a name="L255"></a><tt class="py-lineno"> 255</tt>  <tt class="py-line">            <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-169" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-169', 'location', 'link-62');">location</a></tt><tt class="py-op">,</tt> <tt class="py-name">rec_length</tt><tt class="py-op">)</tt> </tt>
<a name="L256"></a><tt class="py-lineno"> 256</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-170" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-170', 'strand', 'link-156');">strand</a></tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L257"></a><tt class="py-lineno"> 257</tt>  <tt class="py-line">            <tt id="link-171" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-171', 'location', 'link-62');">location</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"complement(%s)"</tt> <tt class="py-op">%</tt> <tt id="link-172" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-172', 'location', 'link-62');">location</a></tt> </tt>
<a name="L258"></a><tt class="py-lineno"> 258</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-173" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-173', 'location', 'link-62');">location</a></tt> </tt>
<a name="L259"></a><tt class="py-lineno"> 259</tt>  <tt class="py-line">    <tt class="py-comment"># As noted above, treat reverse complement strand features carefully:</tt> </tt>
<a name="L260"></a><tt class="py-lineno"> 260</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-174" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-174', 'strand', 'link-156');">strand</a></tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L261"></a><tt class="py-lineno"> 261</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">f</tt> <tt class="py-keyword">in</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">_sub_features</tt><tt class="py-op">:</tt> </tt>
<a name="L262"></a><tt class="py-lineno"> 262</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">f</tt><tt class="py-op">.</tt><tt id="link-175" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-175', 'strand', 'link-156');">strand</a></tt> <tt class="py-op">!=</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L263"></a><tt class="py-lineno"> 263</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Inconsistent strands: %r for parent, %r for child"</tt> </tt>
<a name="L264"></a><tt class="py-lineno"> 264</tt>  <tt class="py-line">                                 <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-176" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-176', 'strand', 'link-156');">strand</a></tt><tt class="py-op">,</tt> <tt class="py-name">f</tt><tt class="py-op">.</tt><tt id="link-177" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-177', 'strand', 'link-156');">strand</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L265"></a><tt class="py-lineno"> 265</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">"complement(%s(%s))"</tt> \ </tt>
<a name="L266"></a><tt class="py-lineno"> 266</tt>  <tt class="py-line">               <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-178" class="py-name" targets="Variable Bio.SeqFeature.SeqFeature.location_operator=Bio.SeqFeature.SeqFeature-class.html#location_operator"><a title="Bio.SeqFeature.SeqFeature.location_operator" class="py-name" href="#" onclick="return doclink('link-178', 'location_operator', 'link-178');">location_operator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L267"></a><tt class="py-lineno"> 267</tt>  <tt class="py-line">                  <tt class="py-string">","</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt id="link-179" class="py-name"><a title="Bio.SeqIO.InsdcIO._insdc_location_string_ignoring_strand_and_subfeatures" class="py-name" href="#" onclick="return doclink('link-179', '_insdc_location_string_ignoring_strand_and_subfeatures', 'link-158');">_insdc_location_string_ignoring_strand_and_subfeatures</a></tt><tt class="py-op">(</tt><tt class="py-name">f</tt><tt class="py-op">.</tt><tt id="link-180" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-180', 'location', 'link-62');">location</a></tt><tt class="py-op">,</tt> <tt class="py-name">rec_length</tt><tt class="py-op">)</tt> </tt>
<a name="L268"></a><tt class="py-lineno"> 268</tt>  <tt class="py-line">                           <tt class="py-keyword">for</tt> <tt class="py-name">f</tt> <tt class="py-keyword">in</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">_sub_features</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L269"></a><tt class="py-lineno"> 269</tt>  <tt class="py-line">    <tt class="py-comment"># if feature.strand == +1:</tt> </tt>
<a name="L270"></a><tt class="py-lineno"> 270</tt>  <tt class="py-line">    <tt class="py-comment">#    for f in feature.sub_features:</tt> </tt>
<a name="L271"></a><tt class="py-lineno"> 271</tt>  <tt class="py-line">    <tt class="py-comment">#        assert f.strand == +1</tt> </tt>
<a name="L272"></a><tt class="py-lineno"> 272</tt>  <tt class="py-line">    <tt class="py-comment"># This covers typical forward strand features, and also an evil mixed strand:</tt> </tt>
<a name="L273"></a><tt class="py-lineno"> 273</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-181" class="py-name"><a title="Bio.SeqFeature.SeqFeature.location_operator" class="py-name" href="#" onclick="return doclink('link-181', 'location_operator', 'link-178');">location_operator</a></tt> <tt class="py-op">!=</tt> <tt class="py-string">""</tt> </tt>
<a name="L274"></a><tt class="py-lineno"> 274</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-string">"%s(%s)"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-182" class="py-name"><a title="Bio.SeqFeature.SeqFeature.location_operator" class="py-name" href="#" onclick="return doclink('link-182', 'location_operator', 'link-178');">location_operator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L275"></a><tt class="py-lineno"> 275</tt>  <tt class="py-line">                       <tt class="py-string">","</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt id="link-183" class="py-name" targets="Function Bio.SeqIO.InsdcIO._insdc_feature_location_string()=Bio.SeqIO.InsdcIO-module.html#_insdc_feature_location_string"><a title="Bio.SeqIO.InsdcIO._insdc_feature_location_string" class="py-name" href="#" onclick="return doclink('link-183', '_insdc_feature_location_string', 'link-183');">_insdc_feature_location_string</a></tt><tt class="py-op">(</tt><tt class="py-name">f</tt><tt class="py-op">,</tt> <tt class="py-name">rec_length</tt><tt class="py-op">)</tt> </tt>
<a name="L276"></a><tt class="py-lineno"> 276</tt>  <tt class="py-line">                                <tt class="py-keyword">for</tt> <tt class="py-name">f</tt> <tt class="py-keyword">in</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">_sub_features</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L277"></a><tt class="py-lineno"> 277</tt>  <tt class="py-line"> </tt>
<a name="L278"></a><tt class="py-lineno"> 278</tt>  <tt class="py-line"> </tt>
<a name="_InsdcWriter"></a><div id="_InsdcWriter-def"><a name="L279"></a><tt class="py-lineno"> 279</tt> <a class="py-toggle" href="#" id="_InsdcWriter-toggle" onclick="return toggle('_InsdcWriter');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO._InsdcWriter-class.html">_InsdcWriter</a><tt class="py-op">(</tt><tt class="py-base-class">SequentialSequenceWriter</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_InsdcWriter-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_InsdcWriter-expanded"><a name="L280"></a><tt class="py-lineno"> 280</tt>  <tt class="py-line">    <tt class="py-docstring">"""Base class for GenBank and EMBL writers (PRIVATE)."""</tt> </tt>
<a name="L281"></a><tt class="py-lineno"> 281</tt>  <tt class="py-line">    <tt id="link-184" class="py-name" targets="Variable Bio.SeqIO.InsdcIO._InsdcWriter.MAX_WIDTH=Bio.SeqIO.InsdcIO._InsdcWriter-class.html#MAX_WIDTH"><a title="Bio.SeqIO.InsdcIO._InsdcWriter.MAX_WIDTH" class="py-name" href="#" onclick="return doclink('link-184', 'MAX_WIDTH', 'link-184');">MAX_WIDTH</a></tt> <tt class="py-op">=</tt> <tt class="py-number">80</tt> </tt>
<a name="L282"></a><tt class="py-lineno"> 282</tt>  <tt class="py-line">    <tt id="link-185" class="py-name" targets="Variable Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT=Bio.SeqIO.InsdcIO.EmblWriter-class.html#QUALIFIER_INDENT,Variable Bio.SeqIO.InsdcIO.GenBankWriter.QUALIFIER_INDENT=Bio.SeqIO.InsdcIO.GenBankWriter-class.html#QUALIFIER_INDENT,Variable Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT=Bio.SeqIO.InsdcIO.ImgtWriter-class.html#QUALIFIER_INDENT,Variable Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT=Bio.SeqIO.InsdcIO._InsdcWriter-class.html#QUALIFIER_INDENT"><a title="Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO.GenBankWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-185', 'QUALIFIER_INDENT', 'link-185');">QUALIFIER_INDENT</a></tt> <tt class="py-op">=</tt> <tt class="py-number">21</tt> </tt>
<a name="L283"></a><tt class="py-lineno"> 283</tt>  <tt class="py-line">    <tt id="link-186" class="py-name" targets="Variable Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT_STR=Bio.SeqIO.InsdcIO.EmblWriter-class.html#QUALIFIER_INDENT_STR,Variable Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT_STR=Bio.SeqIO.InsdcIO.ImgtWriter-class.html#QUALIFIER_INDENT_STR,Variable Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT_STR=Bio.SeqIO.InsdcIO._InsdcWriter-class.html#QUALIFIER_INDENT_STR"><a title="Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT_STR
Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT_STR
Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT_STR" class="py-name" href="#" onclick="return doclink('link-186', 'QUALIFIER_INDENT_STR', 'link-186');">QUALIFIER_INDENT_STR</a></tt> <tt class="py-op">=</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt id="link-187" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO.GenBankWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-187', 'QUALIFIER_INDENT', 'link-185');">QUALIFIER_INDENT</a></tt> </tt>
<a name="L284"></a><tt class="py-lineno"> 284</tt>  <tt class="py-line">    <tt id="link-188" class="py-name" targets="Variable Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT_TMP=Bio.SeqIO.InsdcIO.EmblWriter-class.html#QUALIFIER_INDENT_TMP,Variable Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT_TMP=Bio.SeqIO.InsdcIO.ImgtWriter-class.html#QUALIFIER_INDENT_TMP,Variable Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT_TMP=Bio.SeqIO.InsdcIO._InsdcWriter-class.html#QUALIFIER_INDENT_TMP"><a title="Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT_TMP
Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT_TMP
Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT_TMP" class="py-name" href="#" onclick="return doclink('link-188', 'QUALIFIER_INDENT_TMP', 'link-188');">QUALIFIER_INDENT_TMP</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"     %s                "</tt>  <tt class="py-comment"># 21 if %s is empty</tt> </tt>
<a name="L285"></a><tt class="py-lineno"> 285</tt>  <tt class="py-line">    <tt id="link-189" class="py-name" targets="Variable Bio.SeqIO.InsdcIO._InsdcWriter.FTQUAL_NO_QUOTE=Bio.SeqIO.InsdcIO._InsdcWriter-class.html#FTQUAL_NO_QUOTE"><a title="Bio.SeqIO.InsdcIO._InsdcWriter.FTQUAL_NO_QUOTE" class="py-name" href="#" onclick="return doclink('link-189', 'FTQUAL_NO_QUOTE', 'link-189');">FTQUAL_NO_QUOTE</a></tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-string">"anticodon"</tt><tt class="py-op">,</tt> <tt class="py-string">"citation"</tt><tt class="py-op">,</tt> <tt class="py-string">"codon_start"</tt><tt class="py-op">,</tt> <tt class="py-string">"compare"</tt><tt class="py-op">,</tt> </tt>
<a name="L286"></a><tt class="py-lineno"> 286</tt>  <tt class="py-line">                       <tt class="py-string">"direction"</tt><tt class="py-op">,</tt> <tt class="py-string">"estimated_length"</tt><tt class="py-op">,</tt> <tt class="py-string">"mod_base"</tt><tt class="py-op">,</tt> <tt class="py-string">"number"</tt><tt class="py-op">,</tt> </tt>
<a name="L287"></a><tt class="py-lineno"> 287</tt>  <tt class="py-line">                       <tt class="py-string">"rpt_type"</tt><tt class="py-op">,</tt> <tt class="py-string">"rpt_unit_range"</tt><tt class="py-op">,</tt> <tt class="py-string">"tag_peptide"</tt><tt class="py-op">,</tt> </tt>
<a name="L288"></a><tt class="py-lineno"> 288</tt>  <tt class="py-line">                       <tt class="py-string">"transl_except"</tt><tt class="py-op">,</tt> <tt class="py-string">"transl_table"</tt><tt class="py-op">)</tt> </tt>
<a name="L289"></a><tt class="py-lineno"> 289</tt>  <tt class="py-line"> </tt>
<a name="_InsdcWriter._write_feature_qualifier"></a><div id="_InsdcWriter._write_feature_qualifier-def"><a name="L290"></a><tt class="py-lineno"> 290</tt> <a class="py-toggle" href="#" id="_InsdcWriter._write_feature_qualifier-toggle" onclick="return toggle('_InsdcWriter._write_feature_qualifier');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO._InsdcWriter-class.html#_write_feature_qualifier">_write_feature_qualifier</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">key</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">quote</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_InsdcWriter._write_feature_qualifier-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_InsdcWriter._write_feature_qualifier-expanded"><a name="L291"></a><tt class="py-lineno"> 291</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-190" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.value()=Bio.Phylo.PhyloXMLIO.Writer-class.html#value"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-190', 'value', 'link-190');">value</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L292"></a><tt class="py-lineno"> 292</tt>  <tt class="py-line">            <tt class="py-comment"># Value-less entry like /pseudo</tt> </tt>
<a name="L293"></a><tt class="py-lineno"> 293</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-191" class="py-name" targets="Function Bio.AlignIO.write()=Bio.AlignIO-module.html#write,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.write()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#write,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.write()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#write,Method Bio.Phylo.CDAOIO.Writer.write()=Bio.Phylo.CDAOIO.Writer-class.html#write,Function Bio.Phylo.CDAOIO.write()=Bio.Phylo.CDAOIO-module.html#write,Method Bio.Phylo.NeXMLIO.Writer.write()=Bio.Phylo.NeXMLIO.Writer-class.html#write,Function Bio.Phylo.NeXMLIO.write()=Bio.Phylo.NeXMLIO-module.html#write,Method Bio.Phylo.NewickIO.Writer.write()=Bio.Phylo.NewickIO.Writer-class.html#write,Function Bio.Phylo.NewickIO.write()=Bio.Phylo.NewickIO-module.html#write,Function Bio.Phylo.NexusIO.write()=Bio.Phylo.NexusIO-module.html#write,Method Bio.Phylo.PhyloXMLIO.Writer.write()=Bio.Phylo.PhyloXMLIO.Writer-class.html#write,Function Bio.Phylo.PhyloXMLIO.write()=Bio.Phylo.PhyloXMLIO-module.html#write,Function Bio.Phylo._io.write()=Bio.Phylo._io-module.html#write,Function Bio.SearchIO.write()=Bio.SearchIO-module.html#write,Function Bio.SeqIO.write()=Bio.SeqIO-module.html#write,Method Bio.bgzf.BgzfWriter.write()=Bio.bgzf.BgzfWriter-class.html#write,Function Bio.motifs.jaspar.write()=Bio.motifs.jaspar-module.html#write,Function Bio.motifs.transfac.write()=Bio.motifs.transfac-module.html#write,Function Bio.motifs.write()=Bio.motifs-module.html#write"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-191', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"%s/%s\n"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-192" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT_STR
Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT_STR
Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT_STR" class="py-name" href="#" onclick="return doclink('link-192', 'QUALIFIER_INDENT_STR', 'link-186');">QUALIFIER_INDENT_STR</a></tt><tt class="py-op">,</tt> <tt class="py-name">key</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L294"></a><tt class="py-lineno"> 294</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L295"></a><tt class="py-lineno"> 295</tt>  <tt class="py-line">        <tt class="py-comment"># Quick hack with no line wrapping, may be useful for testing:</tt> </tt>
<a name="L296"></a><tt class="py-lineno"> 296</tt>  <tt class="py-line">        <tt class="py-comment"># self.handle.write('%s/%s="%s"\n' % (self.QUALIFIER_INDENT_STR, key, value))</tt> </tt>
<a name="L297"></a><tt class="py-lineno"> 297</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">quote</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L298"></a><tt class="py-lineno"> 298</tt>  <tt class="py-line">            <tt class="py-comment"># Try to mimic unwritten rules about when quotes can be left out:</tt> </tt>
<a name="L299"></a><tt class="py-lineno"> 299</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-193" class="py-name"><a title="Bio._py3k._is_int_or_long" class="py-name" href="#" onclick="return doclink('link-193', '_is_int_or_long', 'link-19');">_is_int_or_long</a></tt><tt class="py-op">(</tt><tt id="link-194" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-194', 'value', 'link-190');">value</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-195" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter.FTQUAL_NO_QUOTE" class="py-name" href="#" onclick="return doclink('link-195', 'FTQUAL_NO_QUOTE', 'link-189');">FTQUAL_NO_QUOTE</a></tt><tt class="py-op">:</tt> </tt>
<a name="L300"></a><tt class="py-lineno"> 300</tt>  <tt class="py-line">                <tt class="py-name">quote</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L301"></a><tt class="py-lineno"> 301</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L302"></a><tt class="py-lineno"> 302</tt>  <tt class="py-line">                <tt class="py-name">quote</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L303"></a><tt class="py-lineno"> 303</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">quote</tt><tt class="py-op">:</tt> </tt>
<a name="L304"></a><tt class="py-lineno"> 304</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-string">'%s/%s="%s"'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-196" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT_STR
Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT_STR
Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT_STR" class="py-name" href="#" onclick="return doclink('link-196', 'QUALIFIER_INDENT_STR', 'link-186');">QUALIFIER_INDENT_STR</a></tt><tt class="py-op">,</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt id="link-197" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-197', 'value', 'link-190');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L305"></a><tt class="py-lineno"> 305</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L306"></a><tt class="py-lineno"> 306</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-string">'%s/%s=%s'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-198" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT_STR
Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT_STR
Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT_STR" class="py-name" href="#" onclick="return doclink('link-198', 'QUALIFIER_INDENT_STR', 'link-186');">QUALIFIER_INDENT_STR</a></tt><tt class="py-op">,</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt id="link-199" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-199', 'value', 'link-190');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L307"></a><tt class="py-lineno"> 307</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-200" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter.MAX_WIDTH" class="py-name" href="#" onclick="return doclink('link-200', 'MAX_WIDTH', 'link-184');">MAX_WIDTH</a></tt><tt class="py-op">:</tt> </tt>
<a name="L308"></a><tt class="py-lineno"> 308</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-201" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-201', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L309"></a><tt class="py-lineno"> 309</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L310"></a><tt class="py-lineno"> 310</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-202" class="py-name" targets="Method Bio.Seq.Seq.lstrip()=Bio.Seq.Seq-class.html#lstrip"><a title="Bio.Seq.Seq.lstrip" class="py-name" href="#" onclick="return doclink('link-202', 'lstrip', 'link-202');">lstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L311"></a><tt class="py-lineno"> 311</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-203" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter.MAX_WIDTH" class="py-name" href="#" onclick="return doclink('link-203', 'MAX_WIDTH', 'link-184');">MAX_WIDTH</a></tt><tt class="py-op">:</tt> </tt>
<a name="L312"></a><tt class="py-lineno"> 312</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-204" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-204', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L313"></a><tt class="py-lineno"> 313</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> </tt>
<a name="L314"></a><tt class="py-lineno"> 314</tt>  <tt class="py-line">            <tt class="py-comment"># Insert line break...</tt> </tt>
<a name="L315"></a><tt class="py-lineno"> 315</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-205" class="py-name" targets="Method Bio.Crystal.Chain.index()=Bio.Crystal.Chain-class.html#index,Method Bio.SCOP.Raf.SeqMap.index()=Bio.SCOP.Raf.SeqMap-class.html#index,Method Bio.SearchIO._model.hit.Hit.index()=Bio.SearchIO._model.hit.Hit-class.html#index,Method Bio.SearchIO._model.query.QueryResult.index()=Bio.SearchIO._model.query.QueryResult-class.html#index,Function Bio.SearchIO.index()=Bio.SearchIO-module.html#index,Method Bio.Seq.MutableSeq.index()=Bio.Seq.MutableSeq-class.html#index,Function Bio.SeqIO.index()=Bio.SeqIO-module.html#index"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-205', 'index', 'link-205');">index</a></tt> <tt class="py-keyword">in</tt> <tt id="link-206" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-206', 'range', 'link-206');">range</a></tt><tt class="py-op">(</tt><tt id="link-207" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.min=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#min"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-207', 'min', 'link-207');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-208" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter.MAX_WIDTH" class="py-name" href="#" onclick="return doclink('link-208', 'MAX_WIDTH', 'link-184');">MAX_WIDTH</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L316"></a><tt class="py-lineno"> 316</tt>  <tt class="py-line">                               <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-209" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO.GenBankWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-209', 'QUALIFIER_INDENT', 'link-185');">QUALIFIER_INDENT</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L317"></a><tt class="py-lineno"> 317</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt id="link-210" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-210', 'index', 'link-205');">index</a></tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">" "</tt><tt class="py-op">:</tt> </tt>
<a name="L318"></a><tt class="py-lineno"> 318</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L319"></a><tt class="py-lineno"> 319</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt id="link-211" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-211', 'index', 'link-205');">index</a></tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-string">" "</tt><tt class="py-op">:</tt> </tt>
<a name="L320"></a><tt class="py-lineno"> 320</tt>  <tt class="py-line">                <tt class="py-comment"># No nice place to break...</tt> </tt>
<a name="L321"></a><tt class="py-lineno"> 321</tt>  <tt class="py-line">                <tt id="link-212" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-212', 'index', 'link-205');">index</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-213" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter.MAX_WIDTH" class="py-name" href="#" onclick="return doclink('link-213', 'MAX_WIDTH', 'link-184');">MAX_WIDTH</a></tt> </tt>
<a name="L322"></a><tt class="py-lineno"> 322</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt id="link-214" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-214', 'index', 'link-205');">index</a></tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-215" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter.MAX_WIDTH" class="py-name" href="#" onclick="return doclink('link-215', 'MAX_WIDTH', 'link-184');">MAX_WIDTH</a></tt> </tt>
<a name="L323"></a><tt class="py-lineno"> 323</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-216" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-216', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt id="link-217" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-217', 'index', 'link-205');">index</a></tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L324"></a><tt class="py-lineno"> 324</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-218" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT_STR
Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT_STR
Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT_STR" class="py-name" href="#" onclick="return doclink('link-218', 'QUALIFIER_INDENT_STR', 'link-186');">QUALIFIER_INDENT_STR</a></tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt id="link-219" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-219', 'index', 'link-205');">index</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-220" class="py-name"><a title="Bio.Seq.Seq.lstrip" class="py-name" href="#" onclick="return doclink('link-220', 'lstrip', 'link-202');">lstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L325"></a><tt class="py-lineno"> 325</tt>  <tt class="py-line"> </tt>
<a name="_InsdcWriter._wrap_location"></a><div id="_InsdcWriter._wrap_location-def"><a name="L326"></a><tt class="py-lineno"> 326</tt> <a class="py-toggle" href="#" id="_InsdcWriter._wrap_location-toggle" onclick="return toggle('_InsdcWriter._wrap_location');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO._InsdcWriter-class.html#_wrap_location">_wrap_location</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">location</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_InsdcWriter._wrap_location-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_InsdcWriter._wrap_location-expanded"><a name="L327"></a><tt class="py-lineno"> 327</tt>  <tt class="py-line">        <tt class="py-docstring">"""Split a feature location into lines (break at commas)."""</tt> </tt>
<a name="L328"></a><tt class="py-lineno"> 328</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Rewrite this not to recurse!</tt> </tt>
<a name="L329"></a><tt class="py-lineno"> 329</tt>  <tt class="py-line">        <tt id="link-221" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.length()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#length"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-221', 'length', 'link-221');">length</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-222" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter.MAX_WIDTH" class="py-name" href="#" onclick="return doclink('link-222', 'MAX_WIDTH', 'link-184');">MAX_WIDTH</a></tt> <tt class="py-op">-</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-223" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO.GenBankWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-223', 'QUALIFIER_INDENT', 'link-185');">QUALIFIER_INDENT</a></tt> </tt>
<a name="L330"></a><tt class="py-lineno"> 330</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-224" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-224', 'location', 'link-62');">location</a></tt><tt class="py-op">)</tt> <tt class="py-op">&lt;=</tt> <tt id="link-225" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-225', 'length', 'link-221');">length</a></tt><tt class="py-op">:</tt> </tt>
<a name="L331"></a><tt class="py-lineno"> 331</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-226" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-226', 'location', 'link-62');">location</a></tt> </tt>
<a name="L332"></a><tt class="py-lineno"> 332</tt>  <tt class="py-line">        <tt id="link-227" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-227', 'index', 'link-205');">index</a></tt> <tt class="py-op">=</tt> <tt id="link-228" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-228', 'location', 'link-62');">location</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt id="link-229" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-229', 'length', 'link-221');">length</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-230" class="py-name" targets="Method Bio.Seq.Seq.rfind()=Bio.Seq.Seq-class.html#rfind"><a title="Bio.Seq.Seq.rfind" class="py-name" href="#" onclick="return doclink('link-230', 'rfind', 'link-230');">rfind</a></tt><tt class="py-op">(</tt><tt class="py-string">","</tt><tt class="py-op">)</tt> </tt>
<a name="L333"></a><tt class="py-lineno"> 333</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-231" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-231', 'index', 'link-205');">index</a></tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L334"></a><tt class="py-lineno"> 334</tt>  <tt class="py-line">            <tt class="py-comment"># No good place to split (!)</tt> </tt>
<a name="L335"></a><tt class="py-lineno"> 335</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Couldn't split location:\n%s"</tt> <tt class="py-op">%</tt> <tt id="link-232" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-232', 'location', 'link-62');">location</a></tt><tt class="py-op">,</tt> </tt>
<a name="L336"></a><tt class="py-lineno"> 336</tt>  <tt class="py-line">                          <tt id="link-233" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-233', 'BiopythonWarning', 'link-1');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L337"></a><tt class="py-lineno"> 337</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-234" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-234', 'location', 'link-62');">location</a></tt> </tt>
<a name="L338"></a><tt class="py-lineno"> 338</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-235" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-235', 'location', 'link-62');">location</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt id="link-236" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-236', 'index', 'link-205');">index</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L339"></a><tt class="py-lineno"> 339</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-237" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT_STR
Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT_STR
Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT_STR" class="py-name" href="#" onclick="return doclink('link-237', 'QUALIFIER_INDENT_STR', 'link-186');">QUALIFIER_INDENT_STR</a></tt> <tt class="py-op">+</tt> \ </tt>
<a name="L340"></a><tt class="py-lineno"> 340</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-238" class="py-name" targets="Method Bio.SeqIO.InsdcIO._InsdcWriter._wrap_location()=Bio.SeqIO.InsdcIO._InsdcWriter-class.html#_wrap_location"><a title="Bio.SeqIO.InsdcIO._InsdcWriter._wrap_location" class="py-name" href="#" onclick="return doclink('link-238', '_wrap_location', 'link-238');">_wrap_location</a></tt><tt class="py-op">(</tt><tt id="link-239" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-239', 'location', 'link-62');">location</a></tt><tt class="py-op">[</tt><tt id="link-240" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-240', 'index', 'link-205');">index</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L341"></a><tt class="py-lineno"> 341</tt>  <tt class="py-line"> </tt>
<a name="_InsdcWriter._write_feature"></a><div id="_InsdcWriter._write_feature-def"><a name="L342"></a><tt class="py-lineno"> 342</tt> <a class="py-toggle" href="#" id="_InsdcWriter._write_feature-toggle" onclick="return toggle('_InsdcWriter._write_feature');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO._InsdcWriter-class.html#_write_feature">_write_feature</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">feature</tt><tt class="py-op">,</tt> <tt class="py-param">record_length</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_InsdcWriter._write_feature-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_InsdcWriter._write_feature-expanded"><a name="L343"></a><tt class="py-lineno"> 343</tt>  <tt class="py-line">        <tt class="py-docstring">"""Write a single SeqFeature object to features table."""</tt> </tt>
<a name="L344"></a><tt class="py-lineno"> 344</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-241" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.type()=Bio.Phylo.PhyloXMLIO.Writer-class.html#type"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-241', 'type', 'link-241');">type</a></tt><tt class="py-op">,</tt> <tt class="py-name">feature</tt> </tt>
<a name="L345"></a><tt class="py-lineno"> 345</tt>  <tt class="py-line">        <tt id="link-242" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-242', 'location', 'link-62');">location</a></tt> <tt class="py-op">=</tt> <tt id="link-243" class="py-name"><a title="Bio.SeqIO.InsdcIO._insdc_location_string" class="py-name" href="#" onclick="return doclink('link-243', '_insdc_location_string', 'link-161');">_insdc_location_string</a></tt><tt class="py-op">(</tt><tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-244" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-244', 'location', 'link-62');">location</a></tt><tt class="py-op">,</tt> <tt class="py-name">record_length</tt><tt class="py-op">)</tt> </tt>
<a name="L346"></a><tt class="py-lineno"> 346</tt>  <tt class="py-line">        <tt class="py-name">f_type</tt> <tt class="py-op">=</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-245" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-245', 'type', 'link-241');">type</a></tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">" "</tt><tt class="py-op">,</tt> <tt class="py-string">"_"</tt><tt class="py-op">)</tt> </tt>
<a name="L347"></a><tt class="py-lineno"> 347</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-246" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT_TMP
Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT_TMP
Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT_TMP" class="py-name" href="#" onclick="return doclink('link-246', 'QUALIFIER_INDENT_TMP', 'link-188');">QUALIFIER_INDENT_TMP</a></tt> <tt class="py-op">%</tt> <tt class="py-name">f_type</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-247" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO.GenBankWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-247', 'QUALIFIER_INDENT', 'link-185');">QUALIFIER_INDENT</a></tt><tt class="py-op">]</tt> \ </tt>
<a name="L348"></a><tt class="py-lineno"> 348</tt>  <tt class="py-line">               <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-248" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter._wrap_location" class="py-name" href="#" onclick="return doclink('link-248', '_wrap_location', 'link-238');">_wrap_location</a></tt><tt class="py-op">(</tt><tt id="link-249" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-249', 'location', 'link-62');">location</a></tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt> </tt>
<a name="L349"></a><tt class="py-lineno"> 349</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-250" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-250', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L350"></a><tt class="py-lineno"> 350</tt>  <tt class="py-line">        <tt class="py-comment"># Now the qualifiers...</tt> </tt>
<a name="L351"></a><tt class="py-lineno"> 351</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">qualifiers</tt><tt class="py-op">.</tt><tt id="link-251" class="py-name" targets="Method Bio.Crystal.Crystal.keys()=Bio.Crystal.Crystal-class.html#keys,Method Bio.File._IndexedSeqFileDict.keys()=Bio.File._IndexedSeqFileDict-class.html#keys,Method Bio.File._SQLiteManySeqFilesDict.keys()=Bio.File._SQLiteManySeqFilesDict-class.html#keys,Method Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys()=Bio.PDB.AbstractPropertyMap.AbstractPropertyMap-class.html#keys,Method Bio.Phylo.PhyloXML.Events.keys()=Bio.Phylo.PhyloXML.Events-class.html#keys,Method Bio._py3k._ordereddict.OrderedDict.keys()=Bio._py3k._ordereddict.OrderedDict-class.html#keys,Method BioSQL.BioSeqDatabase.BioSeqDatabase.keys()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#keys,Method BioSQL.BioSeqDatabase.DBServer.keys()=BioSQL.BioSeqDatabase.DBServer-class.html#keys"><a title="Bio.Crystal.Crystal.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
Bio._py3k._ordereddict.OrderedDict.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-251', 'keys', 'link-251');">keys</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L352"></a><tt class="py-lineno"> 352</tt>  <tt class="py-line">            <tt id="link-252" class="py-name" targets="Method Bio.Crystal.Crystal.values()=Bio.Crystal.Crystal-class.html#values,Method Bio.File._IndexedSeqFileDict.values()=Bio.File._IndexedSeqFileDict-class.html#values,Method Bio.Phylo.PhyloXML.Events.values()=Bio.Phylo.PhyloXML.Events-class.html#values,Method Bio._py3k._ordereddict.OrderedDict.values()=Bio._py3k._ordereddict.OrderedDict-class.html#values,Method BioSQL.BioSeqDatabase.BioSeqDatabase.values()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#values,Method BioSQL.BioSeqDatabase.DBServer.values()=BioSQL.BioSeqDatabase.DBServer-class.html#values"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
Bio._py3k._ordereddict.OrderedDict.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-252', 'values', 'link-252');">values</a></tt> <tt class="py-op">=</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">qualifiers</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> </tt>
<a name="L353"></a><tt class="py-lineno"> 353</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-253" class="py-name"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
Bio._py3k._ordereddict.OrderedDict.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-253', 'values', 'link-252');">values</a></tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-254" class="py-name"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
Bio._py3k._ordereddict.OrderedDict.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-254', 'values', 'link-252');">values</a></tt><tt class="py-op">,</tt> <tt class="py-name">tuple</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L354"></a><tt class="py-lineno"> 354</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt id="link-255" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-255', 'value', 'link-190');">value</a></tt> <tt class="py-keyword">in</tt> <tt id="link-256" class="py-name"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
Bio._py3k._ordereddict.OrderedDict.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-256', 'values', 'link-252');">values</a></tt><tt class="py-op">:</tt> </tt>
<a name="L355"></a><tt class="py-lineno"> 355</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-257" class="py-name" targets="Method Bio.SeqIO.InsdcIO._InsdcWriter._write_feature_qualifier()=Bio.SeqIO.InsdcIO._InsdcWriter-class.html#_write_feature_qualifier"><a title="Bio.SeqIO.InsdcIO._InsdcWriter._write_feature_qualifier" class="py-name" href="#" onclick="return doclink('link-257', '_write_feature_qualifier', 'link-257');">_write_feature_qualifier</a></tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt id="link-258" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-258', 'value', 'link-190');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L356"></a><tt class="py-lineno"> 356</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L357"></a><tt class="py-lineno"> 357</tt>  <tt class="py-line">                <tt class="py-comment"># String, int, etc - or None for a /pseudo tpy entry</tt> </tt>
<a name="L358"></a><tt class="py-lineno"> 358</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-259" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter._write_feature_qualifier" class="py-name" href="#" onclick="return doclink('link-259', '_write_feature_qualifier', 'link-257');">_write_feature_qualifier</a></tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt id="link-260" class="py-name"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
Bio._py3k._ordereddict.OrderedDict.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-260', 'values', 'link-252');">values</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L359"></a><tt class="py-lineno"> 359</tt>  <tt class="py-line"> </tt>
<a name="_InsdcWriter._get_annotation_str"></a><div id="_InsdcWriter._get_annotation_str-def"><a name="L360"></a><tt class="py-lineno"> 360</tt> <a class="py-toggle" href="#" id="_InsdcWriter._get_annotation_str-toggle" onclick="return toggle('_InsdcWriter._get_annotation_str');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO._InsdcWriter-class.html#_get_annotation_str">_get_annotation_str</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">,</tt> <tt class="py-param">key</tt><tt class="py-op">,</tt> <tt class="py-param">default</tt><tt class="py-op">=</tt><tt class="py-string">"."</tt><tt class="py-op">,</tt> <tt class="py-param">just_first</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_InsdcWriter._get_annotation_str-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_InsdcWriter._get_annotation_str-expanded"><a name="L361"></a><tt class="py-lineno"> 361</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get an annotation dictionary entry (as a string).</tt> </tt>
<a name="L362"></a><tt class="py-lineno"> 362</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L363"></a><tt class="py-lineno"> 363</tt>  <tt class="py-line"><tt class="py-docstring">        Some entries are lists, in which case if just_first=True the first entry</tt> </tt>
<a name="L364"></a><tt class="py-lineno"> 364</tt>  <tt class="py-line"><tt class="py-docstring">        is returned.  If just_first=False (default) this verifies there is only</tt> </tt>
<a name="L365"></a><tt class="py-lineno"> 365</tt>  <tt class="py-line"><tt class="py-docstring">        one entry before returning it."""</tt> </tt>
<a name="L366"></a><tt class="py-lineno"> 366</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L367"></a><tt class="py-lineno"> 367</tt>  <tt class="py-line">            <tt class="py-name">answer</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-261" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.annotations=BioSQL.BioSeq.DBSeqRecord-class.html#annotations"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-261', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> </tt>
<a name="L368"></a><tt class="py-lineno"> 368</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L369"></a><tt class="py-lineno"> 369</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">default</tt> </tt>
<a name="L370"></a><tt class="py-lineno"> 370</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">answer</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L371"></a><tt class="py-lineno"> 371</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">just_first</tt><tt class="py-op">:</tt> </tt>
<a name="L372"></a><tt class="py-lineno"> 372</tt>  <tt class="py-line">                <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">answer</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt> </tt>
<a name="L373"></a><tt class="py-lineno"> 373</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">answer</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L374"></a><tt class="py-lineno"> 374</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L375"></a><tt class="py-lineno"> 375</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">answer</tt><tt class="py-op">)</tt> </tt>
</div><a name="L376"></a><tt class="py-lineno"> 376</tt>  <tt class="py-line"> </tt>
<a name="_InsdcWriter._split_multi_line"></a><div id="_InsdcWriter._split_multi_line-def"><a name="L377"></a><tt class="py-lineno"> 377</tt> <a class="py-toggle" href="#" id="_InsdcWriter._split_multi_line-toggle" onclick="return toggle('_InsdcWriter._split_multi_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO._InsdcWriter-class.html#_split_multi_line">_split_multi_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">,</tt> <tt class="py-param">max_len</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_InsdcWriter._split_multi_line-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_InsdcWriter._split_multi_line-expanded"><a name="L378"></a><tt class="py-lineno"> 378</tt>  <tt class="py-line">        <tt class="py-docstring">"""Returns a list of strings.</tt> </tt>
<a name="L379"></a><tt class="py-lineno"> 379</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L380"></a><tt class="py-lineno"> 380</tt>  <tt class="py-line"><tt class="py-docstring">        Any single words which are too long get returned as a whole line</tt> </tt>
<a name="L381"></a><tt class="py-lineno"> 381</tt>  <tt class="py-line"><tt class="py-docstring">        (e.g. URLs) without an exception or warning.</tt> </tt>
<a name="L382"></a><tt class="py-lineno"> 382</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L383"></a><tt class="py-lineno"> 383</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Do the line spliting while preserving white space?</tt> </tt>
<a name="L384"></a><tt class="py-lineno"> 384</tt>  <tt class="py-line">        <tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt class="py-name">text</tt><tt class="py-op">.</tt><tt id="link-262" class="py-name" targets="Method Bio.Seq.Seq.strip()=Bio.Seq.Seq-class.html#strip"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-262', 'strip', 'link-262');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L385"></a><tt class="py-lineno"> 385</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">max_len</tt><tt class="py-op">:</tt> </tt>
<a name="L386"></a><tt class="py-lineno"> 386</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">text</tt><tt class="py-op">]</tt> </tt>
<a name="L387"></a><tt class="py-lineno"> 387</tt>  <tt class="py-line"> </tt>
<a name="L388"></a><tt class="py-lineno"> 388</tt>  <tt class="py-line">        <tt class="py-name">words</tt> <tt class="py-op">=</tt> <tt class="py-name">text</tt><tt class="py-op">.</tt><tt id="link-263" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-263', 'split', 'link-263');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L389"></a><tt class="py-lineno"> 389</tt>  <tt class="py-line">        <tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L390"></a><tt class="py-lineno"> 390</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">words</tt> <tt class="py-keyword">and</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt> <tt class="py-op">+</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">words</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">max_len</tt><tt class="py-op">:</tt> </tt>
<a name="L391"></a><tt class="py-lineno"> 391</tt>  <tt class="py-line">            <tt class="py-name">text</tt> <tt class="py-op">+=</tt> <tt class="py-string">" "</tt> <tt class="py-op">+</tt> <tt class="py-name">words</tt><tt class="py-op">.</tt><tt id="link-264" class="py-name" targets="Method Bio.File._IndexedSeqFileDict.pop()=Bio.File._IndexedSeqFileDict-class.html#pop,Method Bio.SearchIO._model.hit.Hit.pop()=Bio.SearchIO._model.hit.Hit-class.html#pop,Method Bio.SearchIO._model.query.QueryResult.pop()=Bio.SearchIO._model.query.QueryResult-class.html#pop,Method Bio.Seq.MutableSeq.pop()=Bio.Seq.MutableSeq-class.html#pop,Method Bio._py3k._ordereddict.OrderedDict.pop()=Bio._py3k._ordereddict.OrderedDict-class.html#pop"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop
Bio._py3k._ordereddict.OrderedDict.pop" class="py-name" href="#" onclick="return doclink('link-264', 'pop', 'link-264');">pop</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L392"></a><tt class="py-lineno"> 392</tt>  <tt class="py-line">            <tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt class="py-name">text</tt><tt class="py-op">.</tt><tt id="link-265" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-265', 'strip', 'link-262');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L393"></a><tt class="py-lineno"> 393</tt>  <tt class="py-line">        <tt class="py-comment"># assert len(text) &lt;= max_len</tt> </tt>
<a name="L394"></a><tt class="py-lineno"> 394</tt>  <tt class="py-line">        <tt class="py-name">answer</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">text</tt><tt class="py-op">]</tt> </tt>
<a name="L395"></a><tt class="py-lineno"> 395</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">words</tt><tt class="py-op">:</tt> </tt>
<a name="L396"></a><tt class="py-lineno"> 396</tt>  <tt class="py-line">            <tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt class="py-name">words</tt><tt class="py-op">.</tt><tt id="link-266" class="py-name"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop
Bio._py3k._ordereddict.OrderedDict.pop" class="py-name" href="#" onclick="return doclink('link-266', 'pop', 'link-264');">pop</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L397"></a><tt class="py-lineno"> 397</tt>  <tt class="py-line">            <tt class="py-keyword">while</tt> <tt class="py-name">words</tt> <tt class="py-keyword">and</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt> <tt class="py-op">+</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">words</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">max_len</tt><tt class="py-op">:</tt> </tt>
<a name="L398"></a><tt class="py-lineno"> 398</tt>  <tt class="py-line">                <tt class="py-name">text</tt> <tt class="py-op">+=</tt> <tt class="py-string">" "</tt> <tt class="py-op">+</tt> <tt class="py-name">words</tt><tt class="py-op">.</tt><tt id="link-267" class="py-name"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop
Bio._py3k._ordereddict.OrderedDict.pop" class="py-name" href="#" onclick="return doclink('link-267', 'pop', 'link-264');">pop</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L399"></a><tt class="py-lineno"> 399</tt>  <tt class="py-line">                <tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt class="py-name">text</tt><tt class="py-op">.</tt><tt id="link-268" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-268', 'strip', 'link-262');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L400"></a><tt class="py-lineno"> 400</tt>  <tt class="py-line">            <tt class="py-comment"># assert len(text) &lt;= max_len</tt> </tt>
<a name="L401"></a><tt class="py-lineno"> 401</tt>  <tt class="py-line">            <tt class="py-name">answer</tt><tt class="py-op">.</tt><tt id="link-269" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-269', 'append', 'link-269');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
<a name="L402"></a><tt class="py-lineno"> 402</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt class="py-name">words</tt> </tt>
<a name="L403"></a><tt class="py-lineno"> 403</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">answer</tt> </tt>
</div><a name="L404"></a><tt class="py-lineno"> 404</tt>  <tt class="py-line"> </tt>
<a name="_InsdcWriter._split_contig"></a><div id="_InsdcWriter._split_contig-def"><a name="L405"></a><tt class="py-lineno"> 405</tt> <a class="py-toggle" href="#" id="_InsdcWriter._split_contig-toggle" onclick="return toggle('_InsdcWriter._split_contig');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO._InsdcWriter-class.html#_split_contig">_split_contig</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">,</tt> <tt class="py-param">max_len</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_InsdcWriter._split_contig-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_InsdcWriter._split_contig-expanded"><a name="L406"></a><tt class="py-lineno"> 406</tt>  <tt class="py-line">        <tt class="py-docstring">"""Returns a list of strings, splits on commas."""</tt> </tt>
<a name="L407"></a><tt class="py-lineno"> 407</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Merge this with _write_multi_line method?</tt> </tt>
<a name="L408"></a><tt class="py-lineno"> 408</tt>  <tt class="py-line">        <tt class="py-comment"># It would need the addition of the comma splitting logic...</tt> </tt>
<a name="L409"></a><tt class="py-lineno"> 409</tt>  <tt class="py-line">        <tt class="py-comment"># are there any other cases where that would be sensible?</tt> </tt>
<a name="L410"></a><tt class="py-lineno"> 410</tt>  <tt class="py-line">        <tt class="py-name">contig</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-270" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-270', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-271" class="py-name" targets="Method Bio.Crystal.Crystal.get()=Bio.Crystal.Crystal-class.html#get,Method Bio.Data.CodonTable.AmbiguousForwardTable.get()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#get,Method Bio.File._IndexedSeqFileDict.get()=Bio.File._IndexedSeqFileDict-class.html#get,Method Bio.File._IndexedSeqFileProxy.get()=Bio.File._IndexedSeqFileProxy-class.html#get,Method Bio.File._SQLiteManySeqFilesDict.get()=Bio.File._SQLiteManySeqFilesDict-class.html#get,Method Bio.Restriction.Restriction.RestrictionBatch.get()=Bio.Restriction.Restriction.RestrictionBatch-class.html#get,Method Bio.SearchIO._index.SearchIndexer.get()=Bio.SearchIO._index.SearchIndexer-class.html#get,Method Bio.SeqIO._index.SeqFileRandomAccess.get()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#get,Method Bio.SeqIO._index.SffRandomAccess.get()=Bio.SeqIO._index.SffRandomAccess-class.html#get,Method Bio.SeqIO._index.SffTrimedRandomAccess.get()=Bio.SeqIO._index.SffTrimedRandomAccess-class.html#get,Method Bio.SeqIO._index.UniprotRandomAccess.get()=Bio.SeqIO._index.UniprotRandomAccess-class.html#get"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-271', 'get', 'link-271');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"contig"</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt> </tt>
<a name="L411"></a><tt class="py-lineno"> 411</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">contig</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">contig</tt><tt class="py-op">,</tt> <tt class="py-name">tuple</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L412"></a><tt class="py-lineno"> 412</tt>  <tt class="py-line">            <tt class="py-name">contig</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">contig</tt><tt class="py-op">)</tt> </tt>
<a name="L413"></a><tt class="py-lineno"> 413</tt>  <tt class="py-line">        <tt class="py-name">contig</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-272" class="py-name" targets="Method Bio.AlignIO.Interfaces.AlignmentWriter.clean()=Bio.AlignIO.Interfaces.AlignmentWriter-class.html#clean,Method Bio.SeqIO.Interfaces.SequenceWriter.clean()=Bio.SeqIO.Interfaces.SequenceWriter-class.html#clean"><a title="Bio.AlignIO.Interfaces.AlignmentWriter.clean
Bio.SeqIO.Interfaces.SequenceWriter.clean" class="py-name" href="#" onclick="return doclink('link-272', 'clean', 'link-272');">clean</a></tt><tt class="py-op">(</tt><tt class="py-name">contig</tt><tt class="py-op">)</tt> </tt>
<a name="L414"></a><tt class="py-lineno"> 414</tt>  <tt class="py-line">        <tt class="py-name">answer</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L415"></a><tt class="py-lineno"> 415</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">contig</tt><tt class="py-op">:</tt> </tt>
<a name="L416"></a><tt class="py-lineno"> 416</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">contig</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-name">max_len</tt><tt class="py-op">:</tt> </tt>
<a name="L417"></a><tt class="py-lineno"> 417</tt>  <tt class="py-line">                <tt class="py-comment"># Split lines at the commas</tt> </tt>
<a name="L418"></a><tt class="py-lineno"> 418</tt>  <tt class="py-line">                <tt class="py-name">pos</tt> <tt class="py-op">=</tt> <tt class="py-name">contig</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">max_len</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-273" class="py-name"><a title="Bio.Seq.Seq.rfind" class="py-name" href="#" onclick="return doclink('link-273', 'rfind', 'link-230');">rfind</a></tt><tt class="py-op">(</tt><tt class="py-string">","</tt><tt class="py-op">)</tt> </tt>
<a name="L419"></a><tt class="py-lineno"> 419</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">pos</tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L420"></a><tt class="py-lineno"> 420</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Could not break up CONTIG"</tt><tt class="py-op">)</tt> </tt>
<a name="L421"></a><tt class="py-lineno"> 421</tt>  <tt class="py-line">                <tt class="py-name">text</tt><tt class="py-op">,</tt> <tt class="py-name">contig</tt> <tt class="py-op">=</tt> <tt class="py-name">contig</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">pos</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">contig</tt><tt class="py-op">[</tt><tt class="py-name">pos</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L422"></a><tt class="py-lineno"> 422</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L423"></a><tt class="py-lineno"> 423</tt>  <tt class="py-line">                <tt class="py-name">text</tt><tt class="py-op">,</tt> <tt class="py-name">contig</tt> <tt class="py-op">=</tt> <tt class="py-name">contig</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt> </tt>
<a name="L424"></a><tt class="py-lineno"> 424</tt>  <tt class="py-line">            <tt class="py-name">answer</tt><tt class="py-op">.</tt><tt id="link-274" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-274', 'append', 'link-269');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
<a name="L425"></a><tt class="py-lineno"> 425</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">answer</tt> </tt>
</div></div><a name="L426"></a><tt class="py-lineno"> 426</tt>  <tt class="py-line"> </tt>
<a name="L427"></a><tt class="py-lineno"> 427</tt>  <tt class="py-line"> </tt>
<a name="GenBankWriter"></a><div id="GenBankWriter-def"><a name="L428"></a><tt class="py-lineno"> 428</tt> <a class="py-toggle" href="#" id="GenBankWriter-toggle" onclick="return toggle('GenBankWriter');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO.GenBankWriter-class.html">GenBankWriter</a><tt class="py-op">(</tt><tt class="py-base-class">_InsdcWriter</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GenBankWriter-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="GenBankWriter-expanded"><a name="L429"></a><tt class="py-lineno"> 429</tt>  <tt class="py-line">    <tt id="link-275" class="py-name" targets="Variable Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH=Bio.GenBank.Scanner.EmblScanner-class.html#HEADER_WIDTH,Variable Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH=Bio.GenBank.Scanner.GenBankScanner-class.html#HEADER_WIDTH,Variable Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH=Bio.GenBank.Scanner.InsdcScanner-class.html#HEADER_WIDTH,Variable Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH=Bio.SeqIO.InsdcIO.EmblWriter-class.html#HEADER_WIDTH,Variable Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH=Bio.SeqIO.InsdcIO.GenBankWriter-class.html#HEADER_WIDTH,Variable Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH=Bio.SeqIO.InsdcIO.ImgtWriter-class.html#HEADER_WIDTH"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-275', 'HEADER_WIDTH', 'link-275');">HEADER_WIDTH</a></tt> <tt class="py-op">=</tt> <tt class="py-number">12</tt> </tt>
<a name="L430"></a><tt class="py-lineno"> 430</tt>  <tt class="py-line">    <tt id="link-276" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO.GenBankWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-276', 'QUALIFIER_INDENT', 'link-185');">QUALIFIER_INDENT</a></tt> <tt class="py-op">=</tt> <tt class="py-number">21</tt> </tt>
<a name="L431"></a><tt class="py-lineno"> 431</tt>  <tt class="py-line"> </tt>
<a name="GenBankWriter._write_single_line"></a><div id="GenBankWriter._write_single_line-def"><a name="L432"></a><tt class="py-lineno"> 432</tt> <a class="py-toggle" href="#" id="GenBankWriter._write_single_line-toggle" onclick="return toggle('GenBankWriter._write_single_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO.GenBankWriter-class.html#_write_single_line">_write_single_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">tag</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GenBankWriter._write_single_line-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="GenBankWriter._write_single_line-expanded"><a name="L433"></a><tt class="py-lineno"> 433</tt>  <tt class="py-line">        <tt class="py-docstring">"""Used in the 'header' of each GenBank record."""</tt> </tt>
<a name="L434"></a><tt class="py-lineno"> 434</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-277" class="py-name" targets="Variable Bio.SeqIO.AbiIO.tag=Bio.SeqIO.AbiIO-module.html#tag"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-277', 'tag', 'link-277');">tag</a></tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-278" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-278', 'HEADER_WIDTH', 'link-275');">HEADER_WIDTH</a></tt> </tt>
<a name="L435"></a><tt class="py-lineno"> 435</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-279" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter.MAX_WIDTH" class="py-name" href="#" onclick="return doclink('link-279', 'MAX_WIDTH', 'link-184');">MAX_WIDTH</a></tt> <tt class="py-op">-</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-280" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-280', 'HEADER_WIDTH', 'link-275');">HEADER_WIDTH</a></tt><tt class="py-op">:</tt> </tt>
<a name="L436"></a><tt class="py-lineno"> 436</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Annotation %r too long for %r line"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">,</tt> <tt id="link-281" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-281', 'tag', 'link-277');">tag</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L437"></a><tt class="py-lineno"> 437</tt>  <tt class="py-line">                          <tt id="link-282" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-282', 'BiopythonWarning', 'link-1');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L438"></a><tt class="py-lineno"> 438</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-283" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-283', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"%s%s\n"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-284" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-284', 'tag', 'link-277');">tag</a></tt><tt class="py-op">.</tt><tt class="py-name">ljust</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-285" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-285', 'HEADER_WIDTH', 'link-275');">HEADER_WIDTH</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L439"></a><tt class="py-lineno"> 439</tt>  <tt class="py-line">                                      <tt class="py-name">text</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"\n"</tt><tt class="py-op">,</tt> <tt class="py-string">" "</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L440"></a><tt class="py-lineno"> 440</tt>  <tt class="py-line"> </tt>
<a name="GenBankWriter._write_multi_line"></a><div id="GenBankWriter._write_multi_line-def"><a name="L441"></a><tt class="py-lineno"> 441</tt> <a class="py-toggle" href="#" id="GenBankWriter._write_multi_line-toggle" onclick="return toggle('GenBankWriter._write_multi_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO.GenBankWriter-class.html#_write_multi_line">_write_multi_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">tag</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GenBankWriter._write_multi_line-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="GenBankWriter._write_multi_line-expanded"><a name="L442"></a><tt class="py-lineno"> 442</tt>  <tt class="py-line">        <tt class="py-docstring">"""Used in the 'header' of each GenBank record."""</tt> </tt>
<a name="L443"></a><tt class="py-lineno"> 443</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Do the line spliting while preserving white space?</tt> </tt>
<a name="L444"></a><tt class="py-lineno"> 444</tt>  <tt class="py-line">        <tt class="py-name">max_len</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-286" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter.MAX_WIDTH" class="py-name" href="#" onclick="return doclink('link-286', 'MAX_WIDTH', 'link-184');">MAX_WIDTH</a></tt> <tt class="py-op">-</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-287" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-287', 'HEADER_WIDTH', 'link-275');">HEADER_WIDTH</a></tt> </tt>
<a name="L445"></a><tt class="py-lineno"> 445</tt>  <tt class="py-line">        <tt class="py-name">lines</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-288" class="py-name" targets="Method Bio.SeqIO.InsdcIO._InsdcWriter._split_multi_line()=Bio.SeqIO.InsdcIO._InsdcWriter-class.html#_split_multi_line"><a title="Bio.SeqIO.InsdcIO._InsdcWriter._split_multi_line" class="py-name" href="#" onclick="return doclink('link-288', '_split_multi_line', 'link-288');">_split_multi_line</a></tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">,</tt> <tt class="py-name">max_len</tt><tt class="py-op">)</tt> </tt>
<a name="L446"></a><tt class="py-lineno"> 446</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-289" class="py-name" targets="Method Bio.SeqIO.InsdcIO.EmblWriter._write_single_line()=Bio.SeqIO.InsdcIO.EmblWriter-class.html#_write_single_line,Method Bio.SeqIO.InsdcIO.GenBankWriter._write_single_line()=Bio.SeqIO.InsdcIO.GenBankWriter-class.html#_write_single_line"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_single_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_single_line" class="py-name" href="#" onclick="return doclink('link-289', '_write_single_line', 'link-289');">_write_single_line</a></tt><tt class="py-op">(</tt><tt id="link-290" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-290', 'tag', 'link-277');">tag</a></tt><tt class="py-op">,</tt> <tt class="py-name">lines</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L447"></a><tt class="py-lineno"> 447</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">lines</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L448"></a><tt class="py-lineno"> 448</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-291" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_single_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_single_line" class="py-name" href="#" onclick="return doclink('link-291', '_write_single_line', 'link-289');">_write_single_line</a></tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
</div><a name="L449"></a><tt class="py-lineno"> 449</tt>  <tt class="py-line"> </tt>
<a name="GenBankWriter._write_multi_entries"></a><div id="GenBankWriter._write_multi_entries-def"><a name="L450"></a><tt class="py-lineno"> 450</tt> <a class="py-toggle" href="#" id="GenBankWriter._write_multi_entries-toggle" onclick="return toggle('GenBankWriter._write_multi_entries');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO.GenBankWriter-class.html#_write_multi_entries">_write_multi_entries</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">tag</tt><tt class="py-op">,</tt> <tt class="py-param">text_list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GenBankWriter._write_multi_entries-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="GenBankWriter._write_multi_entries-expanded"><a name="L451"></a><tt class="py-lineno"> 451</tt>  <tt class="py-line">        <tt class="py-comment"># used for DBLINK and any similar later line types.</tt> </tt>
<a name="L452"></a><tt class="py-lineno"> 452</tt>  <tt class="py-line">        <tt class="py-comment"># If the list of strings is empty, nothing is written.</tt> </tt>
<a name="L453"></a><tt class="py-lineno"> 453</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-292" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-292', 'i', 'link-292');">i</a></tt><tt class="py-op">,</tt> <tt class="py-name">text</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt class="py-name">text_list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L454"></a><tt class="py-lineno"> 454</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-293" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-293', 'i', 'link-292');">i</a></tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L455"></a><tt class="py-lineno"> 455</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-294" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_single_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_single_line" class="py-name" href="#" onclick="return doclink('link-294', '_write_single_line', 'link-289');">_write_single_line</a></tt><tt class="py-op">(</tt><tt id="link-295" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-295', 'tag', 'link-277');">tag</a></tt><tt class="py-op">,</tt> <tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
<a name="L456"></a><tt class="py-lineno"> 456</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L457"></a><tt class="py-lineno"> 457</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-296" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_single_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_single_line" class="py-name" href="#" onclick="return doclink('link-296', '_write_single_line', 'link-289');">_write_single_line</a></tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">,</tt> <tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
</div><a name="L458"></a><tt class="py-lineno"> 458</tt>  <tt class="py-line"> </tt>
<a name="GenBankWriter._get_date"></a><div id="GenBankWriter._get_date-def"><a name="L459"></a><tt class="py-lineno"> 459</tt> <a class="py-toggle" href="#" id="GenBankWriter._get_date-toggle" onclick="return toggle('GenBankWriter._get_date');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO.GenBankWriter-class.html#_get_date">_get_date</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GenBankWriter._get_date-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="GenBankWriter._get_date-expanded"><a name="L460"></a><tt class="py-lineno"> 460</tt>  <tt class="py-line">        <tt class="py-name">default</tt> <tt class="py-op">=</tt> <tt class="py-string">"01-JAN-1980"</tt> </tt>
<a name="L461"></a><tt class="py-lineno"> 461</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L462"></a><tt class="py-lineno"> 462</tt>  <tt class="py-line">            <tt id="link-297" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.date()=Bio.GenBank._FeatureConsumer-class.html#date,Method Bio.GenBank._RecordConsumer.date()=Bio.GenBank._RecordConsumer-class.html#date,Method Bio.Phylo.PhyloXMLIO.Parser.date()=Bio.Phylo.PhyloXMLIO.Parser-class.html#date,Method Bio.Phylo.PhyloXMLIO.Writer.date()=Bio.Phylo.PhyloXMLIO.Writer-class.html#date"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-297', 'date', 'link-297');">date</a></tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-298" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-298', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"date"</tt><tt class="py-op">]</tt> </tt>
<a name="L463"></a><tt class="py-lineno"> 463</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L464"></a><tt class="py-lineno"> 464</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">default</tt> </tt>
<a name="L465"></a><tt class="py-lineno"> 465</tt>  <tt class="py-line">        <tt class="py-comment"># Cope with a list of one string:</tt> </tt>
<a name="L466"></a><tt class="py-lineno"> 466</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-299" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-299', 'date', 'link-297');">date</a></tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-300" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-300', 'date', 'link-297');">date</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L467"></a><tt class="py-lineno"> 467</tt>  <tt class="py-line">            <tt id="link-301" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-301', 'date', 'link-297');">date</a></tt> <tt class="py-op">=</tt> <tt id="link-302" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-302', 'date', 'link-297');">date</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L468"></a><tt class="py-lineno"> 468</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - allow a Python date object</tt> </tt>
<a name="L469"></a><tt class="py-lineno"> 469</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-303" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-303', 'date', 'link-297');">date</a></tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-304" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-304', 'date', 'link-297');">date</a></tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-number">11</tt> \ </tt>
<a name="L470"></a><tt class="py-lineno"> 470</tt>  <tt class="py-line">            <tt class="py-keyword">or</tt> <tt id="link-305" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-305', 'date', 'link-297');">date</a></tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-string">"-"</tt> <tt class="py-keyword">or</tt> <tt id="link-306" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-306', 'date', 'link-297');">date</a></tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-string">"-"</tt> \ </tt>
<a name="L471"></a><tt class="py-lineno"> 471</tt>  <tt class="py-line">            <tt class="py-keyword">or</tt> <tt class="py-keyword">not</tt> <tt id="link-307" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-307', 'date', 'link-297');">date</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">isdigit</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-keyword">not</tt> <tt id="link-308" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-308', 'date', 'link-297');">date</a></tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">isdigit</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> \ </tt>
<a name="L472"></a><tt class="py-lineno"> 472</tt>  <tt class="py-line">            <tt class="py-keyword">or</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-309" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-309', 'date', 'link-297');">date</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">31</tt> \ </tt>
<a name="L473"></a><tt class="py-lineno"> 473</tt>  <tt class="py-line">            <tt class="py-keyword">or</tt> <tt id="link-310" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-310', 'date', 'link-297');">date</a></tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">:</tt><tt class="py-number">6</tt><tt class="py-op">]</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"JAN"</tt><tt class="py-op">,</tt> <tt class="py-string">"FEB"</tt><tt class="py-op">,</tt> <tt class="py-string">"MAR"</tt><tt class="py-op">,</tt> <tt class="py-string">"APR"</tt><tt class="py-op">,</tt> <tt class="py-string">"MAY"</tt><tt class="py-op">,</tt> <tt class="py-string">"JUN"</tt><tt class="py-op">,</tt> </tt>
<a name="L474"></a><tt class="py-lineno"> 474</tt>  <tt class="py-line">                                 <tt class="py-string">"JUL"</tt><tt class="py-op">,</tt> <tt class="py-string">"AUG"</tt><tt class="py-op">,</tt> <tt class="py-string">"SEP"</tt><tt class="py-op">,</tt> <tt class="py-string">"OCT"</tt><tt class="py-op">,</tt> <tt class="py-string">"NOV"</tt><tt class="py-op">,</tt> <tt class="py-string">"DEC"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L475"></a><tt class="py-lineno"> 475</tt>  <tt class="py-line">            <tt class="py-comment"># TODO - Check is a valid date (e.g. not 31 Feb)</tt> </tt>
<a name="L476"></a><tt class="py-lineno"> 476</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">default</tt> </tt>
<a name="L477"></a><tt class="py-lineno"> 477</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-311" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-311', 'date', 'link-297');">date</a></tt> </tt>
</div><a name="L478"></a><tt class="py-lineno"> 478</tt>  <tt class="py-line"> </tt>
<a name="GenBankWriter._get_data_division"></a><div id="GenBankWriter._get_data_division-def"><a name="L479"></a><tt class="py-lineno"> 479</tt> <a class="py-toggle" href="#" id="GenBankWriter._get_data_division-toggle" onclick="return toggle('GenBankWriter._get_data_division');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO.GenBankWriter-class.html#_get_data_division">_get_data_division</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GenBankWriter._get_data_division-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="GenBankWriter._get_data_division-expanded"><a name="L480"></a><tt class="py-lineno"> 480</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L481"></a><tt class="py-lineno"> 481</tt>  <tt class="py-line">            <tt class="py-name">division</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-312" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-312', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"data_file_division"</tt><tt class="py-op">]</tt> </tt>
<a name="L482"></a><tt class="py-lineno"> 482</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L483"></a><tt class="py-lineno"> 483</tt>  <tt class="py-line">            <tt class="py-name">division</tt> <tt class="py-op">=</tt> <tt class="py-string">"UNK"</tt> </tt>
<a name="L484"></a><tt class="py-lineno"> 484</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">division</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"PRI"</tt><tt class="py-op">,</tt> <tt class="py-string">"ROD"</tt><tt class="py-op">,</tt> <tt class="py-string">"MAM"</tt><tt class="py-op">,</tt> <tt class="py-string">"VRT"</tt><tt class="py-op">,</tt> <tt class="py-string">"INV"</tt><tt class="py-op">,</tt> <tt class="py-string">"PLN"</tt><tt class="py-op">,</tt> <tt class="py-string">"BCT"</tt><tt class="py-op">,</tt> </tt>
<a name="L485"></a><tt class="py-lineno"> 485</tt>  <tt class="py-line">                        <tt class="py-string">"VRL"</tt><tt class="py-op">,</tt> <tt class="py-string">"PHG"</tt><tt class="py-op">,</tt> <tt class="py-string">"SYN"</tt><tt class="py-op">,</tt> <tt class="py-string">"UNA"</tt><tt class="py-op">,</tt> <tt class="py-string">"EST"</tt><tt class="py-op">,</tt> <tt class="py-string">"PAT"</tt><tt class="py-op">,</tt> <tt class="py-string">"STS"</tt><tt class="py-op">,</tt> </tt>
<a name="L486"></a><tt class="py-lineno"> 486</tt>  <tt class="py-line">                        <tt class="py-string">"GSS"</tt><tt class="py-op">,</tt> <tt class="py-string">"HTG"</tt><tt class="py-op">,</tt> <tt class="py-string">"HTC"</tt><tt class="py-op">,</tt> <tt class="py-string">"ENV"</tt><tt class="py-op">,</tt> <tt class="py-string">"CON"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L487"></a><tt class="py-lineno"> 487</tt>  <tt class="py-line">            <tt class="py-comment"># Good, already GenBank style</tt> </tt>
<a name="L488"></a><tt class="py-lineno"> 488</tt>  <tt class="py-line">            <tt class="py-comment">#    PRI - primate sequences</tt> </tt>
<a name="L489"></a><tt class="py-lineno"> 489</tt>  <tt class="py-line">            <tt class="py-comment">#    ROD - rodent sequences</tt> </tt>
<a name="L490"></a><tt class="py-lineno"> 490</tt>  <tt class="py-line">            <tt class="py-comment">#    MAM - other mammalian sequences</tt> </tt>
<a name="L491"></a><tt class="py-lineno"> 491</tt>  <tt class="py-line">            <tt class="py-comment">#    VRT - other vertebrate sequences</tt> </tt>
<a name="L492"></a><tt class="py-lineno"> 492</tt>  <tt class="py-line">            <tt class="py-comment">#    INV - invertebrate sequences</tt> </tt>
<a name="L493"></a><tt class="py-lineno"> 493</tt>  <tt class="py-line">            <tt class="py-comment">#    PLN - plant, fungal, and algal sequences</tt> </tt>
<a name="L494"></a><tt class="py-lineno"> 494</tt>  <tt class="py-line">            <tt class="py-comment">#    BCT - bacterial sequences [plus archea]</tt> </tt>
<a name="L495"></a><tt class="py-lineno"> 495</tt>  <tt class="py-line">            <tt class="py-comment">#    VRL - viral sequences</tt> </tt>
<a name="L496"></a><tt class="py-lineno"> 496</tt>  <tt class="py-line">            <tt class="py-comment">#    PHG - bacteriophage sequences</tt> </tt>
<a name="L497"></a><tt class="py-lineno"> 497</tt>  <tt class="py-line">            <tt class="py-comment">#    SYN - synthetic sequences</tt> </tt>
<a name="L498"></a><tt class="py-lineno"> 498</tt>  <tt class="py-line">            <tt class="py-comment">#    UNA - unannotated sequences</tt> </tt>
<a name="L499"></a><tt class="py-lineno"> 499</tt>  <tt class="py-line">            <tt class="py-comment">#    EST - EST sequences (expressed sequence tags)</tt> </tt>
<a name="L500"></a><tt class="py-lineno"> 500</tt>  <tt class="py-line">            <tt class="py-comment">#    PAT - patent sequences</tt> </tt>
<a name="L501"></a><tt class="py-lineno"> 501</tt>  <tt class="py-line">            <tt class="py-comment">#    STS - STS sequences (sequence tagged sites)</tt> </tt>
<a name="L502"></a><tt class="py-lineno"> 502</tt>  <tt class="py-line">            <tt class="py-comment">#    GSS - GSS sequences (genome survey sequences)</tt> </tt>
<a name="L503"></a><tt class="py-lineno"> 503</tt>  <tt class="py-line">            <tt class="py-comment">#    HTG - HTGS sequences (high throughput genomic sequences)</tt> </tt>
<a name="L504"></a><tt class="py-lineno"> 504</tt>  <tt class="py-line">            <tt class="py-comment">#    HTC - HTC sequences (high throughput cDNA sequences)</tt> </tt>
<a name="L505"></a><tt class="py-lineno"> 505</tt>  <tt class="py-line">            <tt class="py-comment">#    ENV - Environmental sampling sequences</tt> </tt>
<a name="L506"></a><tt class="py-lineno"> 506</tt>  <tt class="py-line">            <tt class="py-comment">#    CON - Constructed sequences</tt> </tt>
<a name="L507"></a><tt class="py-lineno"> 507</tt>  <tt class="py-line">            <tt class="py-comment">#</tt> </tt>
<a name="L508"></a><tt class="py-lineno"> 508</tt>  <tt class="py-line">            <tt class="py-comment"># (plus UNK for unknown)</tt> </tt>
<a name="L509"></a><tt class="py-lineno"> 509</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
<a name="L510"></a><tt class="py-lineno"> 510</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L511"></a><tt class="py-lineno"> 511</tt>  <tt class="py-line">            <tt class="py-comment"># See if this is in EMBL style:</tt> </tt>
<a name="L512"></a><tt class="py-lineno"> 512</tt>  <tt class="py-line">            <tt class="py-comment">#    Division                 Code</tt> </tt>
<a name="L513"></a><tt class="py-lineno"> 513</tt>  <tt class="py-line">            <tt class="py-comment">#    -----------------        ----</tt> </tt>
<a name="L514"></a><tt class="py-lineno"> 514</tt>  <tt class="py-line">            <tt class="py-comment">#    Bacteriophage            PHG - common</tt> </tt>
<a name="L515"></a><tt class="py-lineno"> 515</tt>  <tt class="py-line">            <tt class="py-comment">#    Environmental Sample     ENV - common</tt> </tt>
<a name="L516"></a><tt class="py-lineno"> 516</tt>  <tt class="py-line">            <tt class="py-comment">#    Fungal                   FUN - map to PLN (plants + fungal)</tt> </tt>
<a name="L517"></a><tt class="py-lineno"> 517</tt>  <tt class="py-line">            <tt class="py-comment">#    Human                    HUM - map to PRI (primates)</tt> </tt>
<a name="L518"></a><tt class="py-lineno"> 518</tt>  <tt class="py-line">            <tt class="py-comment">#    Invertebrate             INV - common</tt> </tt>
<a name="L519"></a><tt class="py-lineno"> 519</tt>  <tt class="py-line">            <tt class="py-comment">#    Other Mammal             MAM - common</tt> </tt>
<a name="L520"></a><tt class="py-lineno"> 520</tt>  <tt class="py-line">            <tt class="py-comment">#    Other Vertebrate         VRT - common</tt> </tt>
<a name="L521"></a><tt class="py-lineno"> 521</tt>  <tt class="py-line">            <tt class="py-comment">#    Mus musculus             MUS - map to ROD (rodent)</tt> </tt>
<a name="L522"></a><tt class="py-lineno"> 522</tt>  <tt class="py-line">            <tt class="py-comment">#    Plant                    PLN - common</tt> </tt>
<a name="L523"></a><tt class="py-lineno"> 523</tt>  <tt class="py-line">            <tt class="py-comment">#    Prokaryote               PRO - map to BCT (poor name)</tt> </tt>
<a name="L524"></a><tt class="py-lineno"> 524</tt>  <tt class="py-line">            <tt class="py-comment">#    Other Rodent             ROD - common</tt> </tt>
<a name="L525"></a><tt class="py-lineno"> 525</tt>  <tt class="py-line">            <tt class="py-comment">#    Synthetic                SYN - common</tt> </tt>
<a name="L526"></a><tt class="py-lineno"> 526</tt>  <tt class="py-line">            <tt class="py-comment">#    Transgenic               TGN - ??? map to SYN ???</tt> </tt>
<a name="L527"></a><tt class="py-lineno"> 527</tt>  <tt class="py-line">            <tt class="py-comment">#    Unclassified             UNC - map to UNK</tt> </tt>
<a name="L528"></a><tt class="py-lineno"> 528</tt>  <tt class="py-line">            <tt class="py-comment">#    Viral                    VRL - common</tt> </tt>
<a name="L529"></a><tt class="py-lineno"> 529</tt>  <tt class="py-line">            <tt class="py-comment">#</tt> </tt>
<a name="L530"></a><tt class="py-lineno"> 530</tt>  <tt class="py-line">            <tt class="py-comment"># (plus XXX for submiting which we can map to UNK)</tt> </tt>
<a name="L531"></a><tt class="py-lineno"> 531</tt>  <tt class="py-line">            <tt class="py-name">embl_to_gbk</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">"FUN"</tt><tt class="py-op">:</tt> <tt class="py-string">"PLN"</tt><tt class="py-op">,</tt> </tt>
<a name="L532"></a><tt class="py-lineno"> 532</tt>  <tt class="py-line">                           <tt class="py-string">"HUM"</tt><tt class="py-op">:</tt> <tt class="py-string">"PRI"</tt><tt class="py-op">,</tt> </tt>
<a name="L533"></a><tt class="py-lineno"> 533</tt>  <tt class="py-line">                           <tt class="py-string">"MUS"</tt><tt class="py-op">:</tt> <tt class="py-string">"ROD"</tt><tt class="py-op">,</tt> </tt>
<a name="L534"></a><tt class="py-lineno"> 534</tt>  <tt class="py-line">                           <tt class="py-string">"PRO"</tt><tt class="py-op">:</tt> <tt class="py-string">"BCT"</tt><tt class="py-op">,</tt> </tt>
<a name="L535"></a><tt class="py-lineno"> 535</tt>  <tt class="py-line">                           <tt class="py-string">"UNC"</tt><tt class="py-op">:</tt> <tt class="py-string">"UNK"</tt><tt class="py-op">,</tt> </tt>
<a name="L536"></a><tt class="py-lineno"> 536</tt>  <tt class="py-line">                           <tt class="py-string">"XXX"</tt><tt class="py-op">:</tt> <tt class="py-string">"UNK"</tt><tt class="py-op">,</tt> </tt>
<a name="L537"></a><tt class="py-lineno"> 537</tt>  <tt class="py-line">                           <tt class="py-op">}</tt> </tt>
<a name="L538"></a><tt class="py-lineno"> 538</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L539"></a><tt class="py-lineno"> 539</tt>  <tt class="py-line">                <tt class="py-name">division</tt> <tt class="py-op">=</tt> <tt class="py-name">embl_to_gbk</tt><tt class="py-op">[</tt><tt class="py-name">division</tt><tt class="py-op">]</tt> </tt>
<a name="L540"></a><tt class="py-lineno"> 540</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L541"></a><tt class="py-lineno"> 541</tt>  <tt class="py-line">                <tt class="py-name">division</tt> <tt class="py-op">=</tt> <tt class="py-string">"UNK"</tt> </tt>
<a name="L542"></a><tt class="py-lineno"> 542</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">division</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">3</tt> </tt>
<a name="L543"></a><tt class="py-lineno"> 543</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">division</tt> </tt>
</div><a name="L544"></a><tt class="py-lineno"> 544</tt>  <tt class="py-line"> </tt>
<a name="GenBankWriter._write_the_first_line"></a><div id="GenBankWriter._write_the_first_line-def"><a name="L545"></a><tt class="py-lineno"> 545</tt> <a class="py-toggle" href="#" id="GenBankWriter._write_the_first_line-toggle" onclick="return toggle('GenBankWriter._write_the_first_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO.GenBankWriter-class.html#_write_the_first_line">_write_the_first_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GenBankWriter._write_the_first_line-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="GenBankWriter._write_the_first_line-expanded"><a name="L546"></a><tt class="py-lineno"> 546</tt>  <tt class="py-line">        <tt class="py-docstring">"""Write the LOCUS line."""</tt> </tt>
<a name="L547"></a><tt class="py-lineno"> 547</tt>  <tt class="py-line"> </tt>
<a name="L548"></a><tt class="py-lineno"> 548</tt>  <tt class="py-line">        <tt id="link-313" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.locus()=Bio.GenBank._FeatureConsumer-class.html#locus,Method Bio.GenBank._RecordConsumer.locus()=Bio.GenBank._RecordConsumer-class.html#locus"><a title="Bio.GenBank._FeatureConsumer.locus
Bio.GenBank._RecordConsumer.locus" class="py-name" href="#" onclick="return doclink('link-313', 'locus', 'link-313');">locus</a></tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-314" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-314', 'name', 'link-314');">name</a></tt> </tt>
<a name="L549"></a><tt class="py-lineno"> 549</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-315" class="py-name"><a title="Bio.GenBank._FeatureConsumer.locus
Bio.GenBank._RecordConsumer.locus" class="py-name" href="#" onclick="return doclink('link-315', 'locus', 'link-313');">locus</a></tt> <tt class="py-keyword">or</tt> <tt id="link-316" class="py-name"><a title="Bio.GenBank._FeatureConsumer.locus
Bio.GenBank._RecordConsumer.locus" class="py-name" href="#" onclick="return doclink('link-316', 'locus', 'link-313');">locus</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"&lt;unknown name&gt;"</tt><tt class="py-op">:</tt> </tt>
<a name="L550"></a><tt class="py-lineno"> 550</tt>  <tt class="py-line">            <tt id="link-317" class="py-name"><a title="Bio.GenBank._FeatureConsumer.locus
Bio.GenBank._RecordConsumer.locus" class="py-name" href="#" onclick="return doclink('link-317', 'locus', 'link-313');">locus</a></tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-318" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-318', 'id', 'link-318');">id</a></tt> </tt>
<a name="L551"></a><tt class="py-lineno"> 551</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-319" class="py-name"><a title="Bio.GenBank._FeatureConsumer.locus
Bio.GenBank._RecordConsumer.locus" class="py-name" href="#" onclick="return doclink('link-319', 'locus', 'link-313');">locus</a></tt> <tt class="py-keyword">or</tt> <tt id="link-320" class="py-name"><a title="Bio.GenBank._FeatureConsumer.locus
Bio.GenBank._RecordConsumer.locus" class="py-name" href="#" onclick="return doclink('link-320', 'locus', 'link-313');">locus</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"&lt;unknown id&gt;"</tt><tt class="py-op">:</tt> </tt>
<a name="L552"></a><tt class="py-lineno"> 552</tt>  <tt class="py-line">            <tt id="link-321" class="py-name"><a title="Bio.GenBank._FeatureConsumer.locus
Bio.GenBank._RecordConsumer.locus" class="py-name" href="#" onclick="return doclink('link-321', 'locus', 'link-313');">locus</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-322" class="py-name" targets="Method Bio.SeqIO.InsdcIO._InsdcWriter._get_annotation_str()=Bio.SeqIO.InsdcIO._InsdcWriter-class.html#_get_annotation_str"><a title="Bio.SeqIO.InsdcIO._InsdcWriter._get_annotation_str" class="py-name" href="#" onclick="return doclink('link-322', '_get_annotation_str', 'link-322');">_get_annotation_str</a></tt><tt class="py-op">(</tt> </tt>
<a name="L553"></a><tt class="py-lineno"> 553</tt>  <tt class="py-line">                <tt class="py-name">record</tt><tt class="py-op">,</tt> <tt class="py-string">"accession"</tt><tt class="py-op">,</tt> <tt class="py-name">just_first</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L554"></a><tt class="py-lineno"> 554</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-323" class="py-name"><a title="Bio.GenBank._FeatureConsumer.locus
Bio.GenBank._RecordConsumer.locus" class="py-name" href="#" onclick="return doclink('link-323', 'locus', 'link-313');">locus</a></tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">16</tt><tt class="py-op">:</tt> </tt>
<a name="L555"></a><tt class="py-lineno"> 555</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Locus identifier %r is too long"</tt> <tt class="py-op">%</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-324" class="py-name"><a title="Bio.GenBank._FeatureConsumer.locus
Bio.GenBank._RecordConsumer.locus" class="py-name" href="#" onclick="return doclink('link-324', 'locus', 'link-313');">locus</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L556"></a><tt class="py-lineno"> 556</tt>  <tt class="py-line"> </tt>
<a name="L557"></a><tt class="py-lineno"> 557</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">99999999999</tt><tt class="py-op">:</tt> </tt>
<a name="L558"></a><tt class="py-lineno"> 558</tt>  <tt class="py-line">            <tt class="py-comment"># Currently GenBank only officially support up to 350000, but</tt> </tt>
<a name="L559"></a><tt class="py-lineno"> 559</tt>  <tt class="py-line">            <tt class="py-comment"># the length field can take eleven digits</tt> </tt>
<a name="L560"></a><tt class="py-lineno"> 560</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Sequence too long!"</tt><tt class="py-op">)</tt> </tt>
<a name="L561"></a><tt class="py-lineno"> 561</tt>  <tt class="py-line"> </tt>
<a name="L562"></a><tt class="py-lineno"> 562</tt>  <tt class="py-line">        <tt class="py-comment"># Get the base alphabet (underneath any Gapped or StopCodon encoding)</tt> </tt>
<a name="L563"></a><tt class="py-lineno"> 563</tt>  <tt class="py-line">        <tt id="link-325" class="py-name" targets="Variable Bio.Graphics.ColorSpiral.ColorSpiral.a=Bio.Graphics.ColorSpiral.ColorSpiral-class.html#a"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-325', 'a', 'link-325');">a</a></tt> <tt class="py-op">=</tt> <tt id="link-326" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-326', 'Alphabet', 'link-12');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-327" class="py-name" targets="Function Bio.Alphabet._get_base_alphabet()=Bio.Alphabet-module.html#_get_base_alphabet"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-327', '_get_base_alphabet', 'link-327');">_get_base_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-328" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.seq=Bio.SeqRecord.SeqRecord-class.html#seq,Variable BioSQL.BioSeq.DBSeqRecord.seq=BioSQL.BioSeq.DBSeqRecord-class.html#seq"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-328', 'seq', 'link-328');">seq</a></tt><tt class="py-op">.</tt><tt id="link-329" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-329', 'alphabet', 'link-32');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L564"></a><tt class="py-lineno"> 564</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-330" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-330', 'a', 'link-325');">a</a></tt><tt class="py-op">,</tt> <tt id="link-331" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-331', 'Alphabet', 'link-12');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-332" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-332', 'Alphabet', 'link-12');">Alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L565"></a><tt class="py-lineno"> 565</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid alphabet"</tt><tt class="py-op">)</tt> </tt>
<a name="L566"></a><tt class="py-lineno"> 566</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-333" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-333', 'a', 'link-325');">a</a></tt><tt class="py-op">,</tt> <tt id="link-334" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-334', 'Alphabet', 'link-12');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-335" class="py-name" targets="Class Bio.Alphabet.ProteinAlphabet=Bio.Alphabet.ProteinAlphabet-class.html"><a title="Bio.Alphabet.ProteinAlphabet" class="py-name" href="#" onclick="return doclink('link-335', 'ProteinAlphabet', 'link-335');">ProteinAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L567"></a><tt class="py-lineno"> 567</tt>  <tt class="py-line">            <tt class="py-name">units</tt> <tt class="py-op">=</tt> <tt class="py-string">"aa"</tt> </tt>
<a name="L568"></a><tt class="py-lineno"> 568</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-336" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-336', 'a', 'link-325');">a</a></tt><tt class="py-op">,</tt> <tt id="link-337" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-337', 'Alphabet', 'link-12');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-338" class="py-name" targets="Class Bio.Alphabet.NucleotideAlphabet=Bio.Alphabet.NucleotideAlphabet-class.html"><a title="Bio.Alphabet.NucleotideAlphabet" class="py-name" href="#" onclick="return doclink('link-338', 'NucleotideAlphabet', 'link-338');">NucleotideAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L569"></a><tt class="py-lineno"> 569</tt>  <tt class="py-line">            <tt class="py-name">units</tt> <tt class="py-op">=</tt> <tt class="py-string">"bp"</tt> </tt>
<a name="L570"></a><tt class="py-lineno"> 570</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L571"></a><tt class="py-lineno"> 571</tt>  <tt class="py-line">            <tt class="py-comment"># Must be something like NucleotideAlphabet or</tt> </tt>
<a name="L572"></a><tt class="py-lineno"> 572</tt>  <tt class="py-line">            <tt class="py-comment"># just the generic Alphabet (default for fasta files)</tt> </tt>
<a name="L573"></a><tt class="py-lineno"> 573</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Need a Nucleotide or Protein alphabet"</tt><tt class="py-op">)</tt> </tt>
<a name="L574"></a><tt class="py-lineno"> 574</tt>  <tt class="py-line"> </tt>
<a name="L575"></a><tt class="py-lineno"> 575</tt>  <tt class="py-line">        <tt class="py-comment"># Get the molecule type</tt> </tt>
<a name="L576"></a><tt class="py-lineno"> 576</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - record this explicitly in the parser?</tt> </tt>
<a name="L577"></a><tt class="py-lineno"> 577</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-339" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-339', 'a', 'link-325');">a</a></tt><tt class="py-op">,</tt> <tt id="link-340" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-340', 'Alphabet', 'link-12');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-341" class="py-name"><a title="Bio.Alphabet.ProteinAlphabet" class="py-name" href="#" onclick="return doclink('link-341', 'ProteinAlphabet', 'link-335');">ProteinAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L578"></a><tt class="py-lineno"> 578</tt>  <tt class="py-line">            <tt class="py-name">mol_type</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L579"></a><tt class="py-lineno"> 579</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-342" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-342', 'a', 'link-325');">a</a></tt><tt class="py-op">,</tt> <tt id="link-343" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-343', 'Alphabet', 'link-12');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-344" class="py-name" targets="Class Bio.Alphabet.DNAAlphabet=Bio.Alphabet.DNAAlphabet-class.html"><a title="Bio.Alphabet.DNAAlphabet" class="py-name" href="#" onclick="return doclink('link-344', 'DNAAlphabet', 'link-344');">DNAAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L580"></a><tt class="py-lineno"> 580</tt>  <tt class="py-line">            <tt class="py-name">mol_type</tt> <tt class="py-op">=</tt> <tt class="py-string">"DNA"</tt> </tt>
<a name="L581"></a><tt class="py-lineno"> 581</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-345" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-345', 'a', 'link-325');">a</a></tt><tt class="py-op">,</tt> <tt id="link-346" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-346', 'Alphabet', 'link-12');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-347" class="py-name" targets="Class Bio.Alphabet.RNAAlphabet=Bio.Alphabet.RNAAlphabet-class.html"><a title="Bio.Alphabet.RNAAlphabet" class="py-name" href="#" onclick="return doclink('link-347', 'RNAAlphabet', 'link-347');">RNAAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L582"></a><tt class="py-lineno"> 582</tt>  <tt class="py-line">            <tt class="py-name">mol_type</tt> <tt class="py-op">=</tt> <tt class="py-string">"RNA"</tt> </tt>
<a name="L583"></a><tt class="py-lineno"> 583</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L584"></a><tt class="py-lineno"> 584</tt>  <tt class="py-line">            <tt class="py-comment"># Must be something like NucleotideAlphabet or</tt> </tt>
<a name="L585"></a><tt class="py-lineno"> 585</tt>  <tt class="py-line">            <tt class="py-comment"># just the generic Alphabet (default for fasta files)</tt> </tt>
<a name="L586"></a><tt class="py-lineno"> 586</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Need a DNA, RNA or Protein alphabet"</tt><tt class="py-op">)</tt> </tt>
<a name="L587"></a><tt class="py-lineno"> 587</tt>  <tt class="py-line"> </tt>
<a name="L588"></a><tt class="py-lineno"> 588</tt>  <tt class="py-line">        <tt class="py-name">division</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-348" class="py-name" targets="Method Bio.SeqIO.InsdcIO.EmblWriter._get_data_division()=Bio.SeqIO.InsdcIO.EmblWriter-class.html#_get_data_division,Method Bio.SeqIO.InsdcIO.GenBankWriter._get_data_division()=Bio.SeqIO.InsdcIO.GenBankWriter-class.html#_get_data_division"><a title="Bio.SeqIO.InsdcIO.EmblWriter._get_data_division
Bio.SeqIO.InsdcIO.GenBankWriter._get_data_division" class="py-name" href="#" onclick="return doclink('link-348', '_get_data_division', 'link-348');">_get_data_division</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L589"></a><tt class="py-lineno"> 589</tt>  <tt class="py-line"> </tt>
<a name="L590"></a><tt class="py-lineno"> 590</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">units</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt> </tt>
<a name="L591"></a><tt class="py-lineno"> 591</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">division</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">3</tt> </tt>
<a name="L592"></a><tt class="py-lineno"> 592</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - date</tt> </tt>
<a name="L593"></a><tt class="py-lineno"> 593</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - mol_type</tt> </tt>
<a name="L594"></a><tt class="py-lineno"> 594</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-string">"LOCUS       %s %s %s    %s           %s %s\n"</tt> \ </tt>
<a name="L595"></a><tt class="py-lineno"> 595</tt>  <tt class="py-line">            <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-349" class="py-name"><a title="Bio.GenBank._FeatureConsumer.locus
Bio.GenBank._RecordConsumer.locus" class="py-name" href="#" onclick="return doclink('link-349', 'locus', 'link-313');">locus</a></tt><tt class="py-op">.</tt><tt class="py-name">ljust</tt><tt class="py-op">(</tt><tt class="py-number">16</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L596"></a><tt class="py-lineno"> 596</tt>  <tt class="py-line">               <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">rjust</tt><tt class="py-op">(</tt><tt class="py-number">11</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L597"></a><tt class="py-lineno"> 597</tt>  <tt class="py-line">               <tt class="py-name">units</tt><tt class="py-op">,</tt> </tt>
<a name="L598"></a><tt class="py-lineno"> 598</tt>  <tt class="py-line">               <tt class="py-name">mol_type</tt><tt class="py-op">.</tt><tt class="py-name">ljust</tt><tt class="py-op">(</tt><tt class="py-number">6</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L599"></a><tt class="py-lineno"> 599</tt>  <tt class="py-line">               <tt class="py-name">division</tt><tt class="py-op">,</tt> </tt>
<a name="L600"></a><tt class="py-lineno"> 600</tt>  <tt class="py-line">               <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-350" class="py-name" targets="Method Bio.SeqIO.InsdcIO.GenBankWriter._get_date()=Bio.SeqIO.InsdcIO.GenBankWriter-class.html#_get_date"><a title="Bio.SeqIO.InsdcIO.GenBankWriter._get_date" class="py-name" href="#" onclick="return doclink('link-350', '_get_date', 'link-350');">_get_date</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L601"></a><tt class="py-lineno"> 601</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">79</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt>  <tt class="py-comment"># plus one for new line</tt> </tt>
<a name="L602"></a><tt class="py-lineno"> 602</tt>  <tt class="py-line"> </tt>
<a name="L603"></a><tt class="py-lineno"> 603</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">12</tt><tt class="py-op">:</tt><tt class="py-number">28</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-351" class="py-name" targets="Method Bio.Seq.Seq.rstrip()=Bio.Seq.Seq-class.html#rstrip"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-351', 'rstrip', 'link-351');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt id="link-352" class="py-name"><a title="Bio.GenBank._FeatureConsumer.locus
Bio.GenBank._RecordConsumer.locus" class="py-name" href="#" onclick="return doclink('link-352', 'locus', 'link-313');">locus</a></tt><tt class="py-op">,</tt> \ </tt>
<a name="L604"></a><tt class="py-lineno"> 604</tt>  <tt class="py-line">            <tt class="py-string">'LOCUS line does not contain the locus at the expected position:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L605"></a><tt class="py-lineno"> 605</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">28</tt><tt class="py-op">:</tt><tt class="py-number">29</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">" "</tt> </tt>
<a name="L606"></a><tt class="py-lineno"> 606</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">29</tt><tt class="py-op">:</tt><tt class="py-number">40</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-353" class="py-name"><a title="Bio.Seq.Seq.lstrip" class="py-name" href="#" onclick="return doclink('link-353', 'lstrip', 'link-202');">lstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> \ </tt>
<a name="L607"></a><tt class="py-lineno"> 607</tt>  <tt class="py-line">            <tt class="py-string">'LOCUS line does not contain the length at the expected position:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L608"></a><tt class="py-lineno"> 608</tt>  <tt class="py-line"> </tt>
<a name="L609"></a><tt class="py-lineno"> 609</tt>  <tt class="py-line">        <tt class="py-comment"># Tests copied from Bio.GenBank.Scanner</tt> </tt>
<a name="L610"></a><tt class="py-lineno"> 610</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">40</tt><tt class="py-op">:</tt><tt class="py-number">44</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">' bp '</tt><tt class="py-op">,</tt> <tt class="py-string">' aa '</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> \ </tt>
<a name="L611"></a><tt class="py-lineno"> 611</tt>  <tt class="py-line">            <tt class="py-string">'LOCUS line does not contain size units at expected position:\n'</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L612"></a><tt class="py-lineno"> 612</tt>  <tt class="py-line">            <tt class="py-name">line</tt> </tt>
<a name="L613"></a><tt class="py-lineno"> 613</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">44</tt><tt class="py-op">:</tt><tt class="py-number">47</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">'   '</tt><tt class="py-op">,</tt> <tt class="py-string">'ss-'</tt><tt class="py-op">,</tt> <tt class="py-string">'ds-'</tt><tt class="py-op">,</tt> <tt class="py-string">'ms-'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> \ </tt>
<a name="L614"></a><tt class="py-lineno"> 614</tt>  <tt class="py-line">            <tt class="py-string">'LOCUS line does not have valid strand type (Single stranded, ...):\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L615"></a><tt class="py-lineno"> 615</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">47</tt><tt class="py-op">:</tt><tt class="py-number">54</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-354" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-354', 'strip', 'link-262');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">""</tt> \ </tt>
<a name="L616"></a><tt class="py-lineno"> 616</tt>  <tt class="py-line">            <tt class="py-keyword">or</tt> <tt class="py-string">'DNA'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">47</tt><tt class="py-op">:</tt><tt class="py-number">54</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-355" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-355', 'strip', 'link-262');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> \ </tt>
<a name="L617"></a><tt class="py-lineno"> 617</tt>  <tt class="py-line">            <tt class="py-keyword">or</tt> <tt class="py-string">'RNA'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">47</tt><tt class="py-op">:</tt><tt class="py-number">54</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-356" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-356', 'strip', 'link-262');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> \ </tt>
<a name="L618"></a><tt class="py-lineno"> 618</tt>  <tt class="py-line">               <tt class="py-string">'LOCUS line does not contain valid sequence type (DNA, RNA, ...):\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L619"></a><tt class="py-lineno"> 619</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">54</tt><tt class="py-op">:</tt><tt class="py-number">55</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">' '</tt><tt class="py-op">,</tt> \ </tt>
<a name="L620"></a><tt class="py-lineno"> 620</tt>  <tt class="py-line">            <tt class="py-string">'LOCUS line does not contain space at position 55:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L621"></a><tt class="py-lineno"> 621</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">55</tt><tt class="py-op">:</tt><tt class="py-number">63</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-357" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-357', 'strip', 'link-262');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">''</tt><tt class="py-op">,</tt> <tt class="py-string">'linear'</tt><tt class="py-op">,</tt> <tt class="py-string">'circular'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> \ </tt>
<a name="L622"></a><tt class="py-lineno"> 622</tt>  <tt class="py-line">            <tt class="py-string">'LOCUS line does not contain valid entry (linear, circular, ...):\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L623"></a><tt class="py-lineno"> 623</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">63</tt><tt class="py-op">:</tt><tt class="py-number">64</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">' '</tt><tt class="py-op">,</tt> \ </tt>
<a name="L624"></a><tt class="py-lineno"> 624</tt>  <tt class="py-line">            <tt class="py-string">'LOCUS line does not contain space at position 64:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L625"></a><tt class="py-lineno"> 625</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">67</tt><tt class="py-op">:</tt><tt class="py-number">68</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">' '</tt><tt class="py-op">,</tt> \ </tt>
<a name="L626"></a><tt class="py-lineno"> 626</tt>  <tt class="py-line">            <tt class="py-string">'LOCUS line does not contain space at position 68:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L627"></a><tt class="py-lineno"> 627</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">70</tt><tt class="py-op">:</tt><tt class="py-number">71</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'-'</tt><tt class="py-op">,</tt> \ </tt>
<a name="L628"></a><tt class="py-lineno"> 628</tt>  <tt class="py-line">            <tt class="py-string">'LOCUS line does not contain - at position 71 in date:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L629"></a><tt class="py-lineno"> 629</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">74</tt><tt class="py-op">:</tt><tt class="py-number">75</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'-'</tt><tt class="py-op">,</tt> \ </tt>
<a name="L630"></a><tt class="py-lineno"> 630</tt>  <tt class="py-line">            <tt class="py-string">'LOCUS line does not contain - at position 75 in date:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
<a name="L631"></a><tt class="py-lineno"> 631</tt>  <tt class="py-line"> </tt>
<a name="L632"></a><tt class="py-lineno"> 632</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-358" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-358', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
</div><a name="L633"></a><tt class="py-lineno"> 633</tt>  <tt class="py-line"> </tt>
<a name="GenBankWriter._write_references"></a><div id="GenBankWriter._write_references-def"><a name="L634"></a><tt class="py-lineno"> 634</tt> <a class="py-toggle" href="#" id="GenBankWriter._write_references-toggle" onclick="return toggle('GenBankWriter._write_references');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO.GenBankWriter-class.html#_write_references">_write_references</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GenBankWriter._write_references-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="GenBankWriter._write_references-expanded"><a name="L635"></a><tt class="py-lineno"> 635</tt>  <tt class="py-line">        <tt id="link-359" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Pathway.number=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#number"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.number" class="py-name" href="#" onclick="return doclink('link-359', 'number', 'link-359');">number</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L636"></a><tt class="py-lineno"> 636</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-360" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-360', 'ref', 'link-63');">ref</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-361" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-361', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"references"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L637"></a><tt class="py-lineno"> 637</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-362" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-362', 'ref', 'link-63');">ref</a></tt><tt class="py-op">,</tt> <tt id="link-363" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-363', 'SeqFeature', 'link-16');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-364" class="py-name" targets="Class Bio.ExPASy.Prodoc.Reference=Bio.ExPASy.Prodoc.Reference-class.html,Class Bio.GenBank.Record.Reference=Bio.GenBank.Record.Reference-class.html,Class Bio.Phylo.PhyloXML.Reference=Bio.Phylo.PhyloXML.Reference-class.html,Class Bio.SeqFeature.Reference=Bio.SeqFeature.Reference-class.html,Class Bio.SwissProt.Reference=Bio.SwissProt.Reference-class.html"><a title="Bio.ExPASy.Prodoc.Reference
Bio.GenBank.Record.Reference
Bio.Phylo.PhyloXML.Reference
Bio.SeqFeature.Reference
Bio.SwissProt.Reference" class="py-name" href="#" onclick="return doclink('link-364', 'Reference', 'link-364');">Reference</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L638"></a><tt class="py-lineno"> 638</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L639"></a><tt class="py-lineno"> 639</tt>  <tt class="py-line">            <tt id="link-365" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.number" class="py-name" href="#" onclick="return doclink('link-365', 'number', 'link-359');">number</a></tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L640"></a><tt class="py-lineno"> 640</tt>  <tt class="py-line">            <tt id="link-366" class="py-name" targets="Variable BioSQL.BioSeq.DBSeq.data=BioSQL.BioSeq.DBSeq-class.html#data"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-366', 'data', 'link-366');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-367" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.number" class="py-name" href="#" onclick="return doclink('link-367', 'number', 'link-359');">number</a></tt><tt class="py-op">)</tt> </tt>
<a name="L641"></a><tt class="py-lineno"> 641</tt>  <tt class="py-line">            <tt class="py-comment"># TODO - support more complex record reference locations?</tt> </tt>
<a name="L642"></a><tt class="py-lineno"> 642</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-368" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-368', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-369" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-369', 'location', 'link-62');">location</a></tt> <tt class="py-keyword">and</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-370" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-370', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-371" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-371', 'location', 'link-62');">location</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L643"></a><tt class="py-lineno"> 643</tt>  <tt class="py-line">                <tt id="link-372" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-372', 'a', 'link-325');">a</a></tt> <tt class="py-op">=</tt> <tt id="link-373" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-373', 'Alphabet', 'link-12');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-374" class="py-name"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-374', '_get_base_alphabet', 'link-327');">_get_base_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-375" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-375', 'seq', 'link-328');">seq</a></tt><tt class="py-op">.</tt><tt id="link-376" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-376', 'alphabet', 'link-32');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L644"></a><tt class="py-lineno"> 644</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-377" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-377', 'a', 'link-325');">a</a></tt><tt class="py-op">,</tt> <tt id="link-378" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-378', 'Alphabet', 'link-12');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-379" class="py-name"><a title="Bio.Alphabet.ProteinAlphabet" class="py-name" href="#" onclick="return doclink('link-379', 'ProteinAlphabet', 'link-335');">ProteinAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L645"></a><tt class="py-lineno"> 645</tt>  <tt class="py-line">                    <tt class="py-name">units</tt> <tt class="py-op">=</tt> <tt class="py-string">"residues"</tt> </tt>
<a name="L646"></a><tt class="py-lineno"> 646</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L647"></a><tt class="py-lineno"> 647</tt>  <tt class="py-line">                    <tt class="py-name">units</tt> <tt class="py-op">=</tt> <tt class="py-string">"bases"</tt> </tt>
<a name="L648"></a><tt class="py-lineno"> 648</tt>  <tt class="py-line">                <tt id="link-380" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-380', 'data', 'link-366');">data</a></tt> <tt class="py-op">+=</tt> <tt class="py-string">"  (%s %i to %i)"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">units</tt><tt class="py-op">,</tt> </tt>
<a name="L649"></a><tt class="py-lineno"> 649</tt>  <tt class="py-line">                                             <tt id="link-381" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-381', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-382" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-382', 'location', 'link-62');">location</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-383" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.nofuzzy_start
Bio.SeqFeature.FeatureLocation.nofuzzy_start" class="py-name" href="#" onclick="return doclink('link-383', 'nofuzzy_start', 'link-144');">nofuzzy_start</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> </tt>
<a name="L650"></a><tt class="py-lineno"> 650</tt>  <tt class="py-line">                                             <tt id="link-384" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-384', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-385" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-385', 'location', 'link-62');">location</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-386" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.nofuzzy_end
Bio.SeqFeature.FeatureLocation.nofuzzy_end" class="py-name" href="#" onclick="return doclink('link-386', 'nofuzzy_end', 'link-135');">nofuzzy_end</a></tt><tt class="py-op">)</tt> </tt>
<a name="L651"></a><tt class="py-lineno"> 651</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-387" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_single_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_single_line" class="py-name" href="#" onclick="return doclink('link-387', '_write_single_line', 'link-289');">_write_single_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"REFERENCE"</tt><tt class="py-op">,</tt> <tt id="link-388" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-388', 'data', 'link-366');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L652"></a><tt class="py-lineno"> 652</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-389" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-389', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-390" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.authors()=Bio.GenBank._FeatureConsumer-class.html#authors,Method Bio.GenBank._RecordConsumer.authors()=Bio.GenBank._RecordConsumer-class.html#authors"><a title="Bio.GenBank._FeatureConsumer.authors
Bio.GenBank._RecordConsumer.authors" class="py-name" href="#" onclick="return doclink('link-390', 'authors', 'link-390');">authors</a></tt><tt class="py-op">:</tt> </tt>
<a name="L653"></a><tt class="py-lineno"> 653</tt>  <tt class="py-line">                <tt class="py-comment"># We store the AUTHORS data as a single string</tt> </tt>
<a name="L654"></a><tt class="py-lineno"> 654</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-391" class="py-name" targets="Method Bio.SeqIO.InsdcIO.EmblWriter._write_multi_line()=Bio.SeqIO.InsdcIO.EmblWriter-class.html#_write_multi_line,Method Bio.SeqIO.InsdcIO.GenBankWriter._write_multi_line()=Bio.SeqIO.InsdcIO.GenBankWriter-class.html#_write_multi_line"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_multi_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_multi_line" class="py-name" href="#" onclick="return doclink('link-391', '_write_multi_line', 'link-391');">_write_multi_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"  AUTHORS"</tt><tt class="py-op">,</tt> <tt id="link-392" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-392', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-393" class="py-name"><a title="Bio.GenBank._FeatureConsumer.authors
Bio.GenBank._RecordConsumer.authors" class="py-name" href="#" onclick="return doclink('link-393', 'authors', 'link-390');">authors</a></tt><tt class="py-op">)</tt> </tt>
<a name="L655"></a><tt class="py-lineno"> 655</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-394" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-394', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-395" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.consrtm()=Bio.GenBank._FeatureConsumer-class.html#consrtm,Method Bio.GenBank._RecordConsumer.consrtm()=Bio.GenBank._RecordConsumer-class.html#consrtm"><a title="Bio.GenBank._FeatureConsumer.consrtm
Bio.GenBank._RecordConsumer.consrtm" class="py-name" href="#" onclick="return doclink('link-395', 'consrtm', 'link-395');">consrtm</a></tt><tt class="py-op">:</tt> </tt>
<a name="L656"></a><tt class="py-lineno"> 656</tt>  <tt class="py-line">                <tt class="py-comment"># We store the consortium as a single string</tt> </tt>
<a name="L657"></a><tt class="py-lineno"> 657</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-396" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_multi_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_multi_line" class="py-name" href="#" onclick="return doclink('link-396', '_write_multi_line', 'link-391');">_write_multi_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"  CONSRTM"</tt><tt class="py-op">,</tt> <tt id="link-397" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-397', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-398" class="py-name"><a title="Bio.GenBank._FeatureConsumer.consrtm
Bio.GenBank._RecordConsumer.consrtm" class="py-name" href="#" onclick="return doclink('link-398', 'consrtm', 'link-395');">consrtm</a></tt><tt class="py-op">)</tt> </tt>
<a name="L658"></a><tt class="py-lineno"> 658</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-399" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-399', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-400" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.title()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#title,Method Bio.GenBank._FeatureConsumer.title()=Bio.GenBank._FeatureConsumer-class.html#title,Method Bio.GenBank._RecordConsumer.title()=Bio.GenBank._RecordConsumer-class.html#title"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-400', 'title', 'link-400');">title</a></tt><tt class="py-op">:</tt> </tt>
<a name="L659"></a><tt class="py-lineno"> 659</tt>  <tt class="py-line">                <tt class="py-comment"># We store the title as a single string</tt> </tt>
<a name="L660"></a><tt class="py-lineno"> 660</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-401" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_multi_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_multi_line" class="py-name" href="#" onclick="return doclink('link-401', '_write_multi_line', 'link-391');">_write_multi_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"  TITLE"</tt><tt class="py-op">,</tt> <tt id="link-402" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-402', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-403" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-403', 'title', 'link-400');">title</a></tt><tt class="py-op">)</tt> </tt>
<a name="L661"></a><tt class="py-lineno"> 661</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-404" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-404', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-405" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.journal()=Bio.GenBank._FeatureConsumer-class.html#journal,Method Bio.GenBank._RecordConsumer.journal()=Bio.GenBank._RecordConsumer-class.html#journal"><a title="Bio.GenBank._FeatureConsumer.journal
Bio.GenBank._RecordConsumer.journal" class="py-name" href="#" onclick="return doclink('link-405', 'journal', 'link-405');">journal</a></tt><tt class="py-op">:</tt> </tt>
<a name="L662"></a><tt class="py-lineno"> 662</tt>  <tt class="py-line">                <tt class="py-comment"># We store this as a single string - holds the journal name,</tt> </tt>
<a name="L663"></a><tt class="py-lineno"> 663</tt>  <tt class="py-line">                <tt class="py-comment"># volume, year, and page numbers of the citation</tt> </tt>
<a name="L664"></a><tt class="py-lineno"> 664</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-406" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_multi_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_multi_line" class="py-name" href="#" onclick="return doclink('link-406', '_write_multi_line', 'link-391');">_write_multi_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"  JOURNAL"</tt><tt class="py-op">,</tt> <tt id="link-407" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-407', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-408" class="py-name"><a title="Bio.GenBank._FeatureConsumer.journal
Bio.GenBank._RecordConsumer.journal" class="py-name" href="#" onclick="return doclink('link-408', 'journal', 'link-405');">journal</a></tt><tt class="py-op">)</tt> </tt>
<a name="L665"></a><tt class="py-lineno"> 665</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-409" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-409', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-410" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.medline_id()=Bio.GenBank._FeatureConsumer-class.html#medline_id,Method Bio.GenBank._RecordConsumer.medline_id()=Bio.GenBank._RecordConsumer-class.html#medline_id"><a title="Bio.GenBank._FeatureConsumer.medline_id
Bio.GenBank._RecordConsumer.medline_id" class="py-name" href="#" onclick="return doclink('link-410', 'medline_id', 'link-410');">medline_id</a></tt><tt class="py-op">:</tt> </tt>
<a name="L666"></a><tt class="py-lineno"> 666</tt>  <tt class="py-line">                <tt class="py-comment"># This line type is obsolete and was removed from the GenBank</tt> </tt>
<a name="L667"></a><tt class="py-lineno"> 667</tt>  <tt class="py-line">                <tt class="py-comment"># flatfile format in April 2005. Should we write it?</tt> </tt>
<a name="L668"></a><tt class="py-lineno"> 668</tt>  <tt class="py-line">                <tt class="py-comment"># Note this has a two space indent:</tt> </tt>
<a name="L669"></a><tt class="py-lineno"> 669</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-411" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_multi_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_multi_line" class="py-name" href="#" onclick="return doclink('link-411', '_write_multi_line', 'link-391');">_write_multi_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"  MEDLINE"</tt><tt class="py-op">,</tt> <tt id="link-412" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-412', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-413" class="py-name"><a title="Bio.GenBank._FeatureConsumer.medline_id
Bio.GenBank._RecordConsumer.medline_id" class="py-name" href="#" onclick="return doclink('link-413', 'medline_id', 'link-410');">medline_id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L670"></a><tt class="py-lineno"> 670</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-414" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-414', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-415" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.pubmed_id()=Bio.GenBank._FeatureConsumer-class.html#pubmed_id,Method Bio.GenBank._RecordConsumer.pubmed_id()=Bio.GenBank._RecordConsumer-class.html#pubmed_id"><a title="Bio.GenBank._FeatureConsumer.pubmed_id
Bio.GenBank._RecordConsumer.pubmed_id" class="py-name" href="#" onclick="return doclink('link-415', 'pubmed_id', 'link-415');">pubmed_id</a></tt><tt class="py-op">:</tt> </tt>
<a name="L671"></a><tt class="py-lineno"> 671</tt>  <tt class="py-line">                <tt class="py-comment"># Note this has a THREE space indent:</tt> </tt>
<a name="L672"></a><tt class="py-lineno"> 672</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-416" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_multi_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_multi_line" class="py-name" href="#" onclick="return doclink('link-416', '_write_multi_line', 'link-391');">_write_multi_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"   PUBMED"</tt><tt class="py-op">,</tt> <tt id="link-417" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-417', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-418" class="py-name"><a title="Bio.GenBank._FeatureConsumer.pubmed_id
Bio.GenBank._RecordConsumer.pubmed_id" class="py-name" href="#" onclick="return doclink('link-418', 'pubmed_id', 'link-415');">pubmed_id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L673"></a><tt class="py-lineno"> 673</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-419" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-419', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-420" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.comment()=Bio.GenBank._FeatureConsumer-class.html#comment,Method Bio.GenBank._RecordConsumer.comment()=Bio.GenBank._RecordConsumer-class.html#comment"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-420', 'comment', 'link-420');">comment</a></tt><tt class="py-op">:</tt> </tt>
<a name="L674"></a><tt class="py-lineno"> 674</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-421" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_multi_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_multi_line" class="py-name" href="#" onclick="return doclink('link-421', '_write_multi_line', 'link-391');">_write_multi_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"  REMARK"</tt><tt class="py-op">,</tt> <tt id="link-422" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-422', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-423" class="py-name"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-423', 'comment', 'link-420');">comment</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L675"></a><tt class="py-lineno"> 675</tt>  <tt class="py-line"> </tt>
<a name="GenBankWriter._write_comment"></a><div id="GenBankWriter._write_comment-def"><a name="L676"></a><tt class="py-lineno"> 676</tt> <a class="py-toggle" href="#" id="GenBankWriter._write_comment-toggle" onclick="return toggle('GenBankWriter._write_comment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO.GenBankWriter-class.html#_write_comment">_write_comment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GenBankWriter._write_comment-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="GenBankWriter._write_comment-expanded"><a name="L677"></a><tt class="py-lineno"> 677</tt>  <tt class="py-line">        <tt class="py-comment"># This is a bit complicated due to the range of possible</tt> </tt>
<a name="L678"></a><tt class="py-lineno"> 678</tt>  <tt class="py-line">        <tt class="py-comment"># ways people might have done their annotation...</tt> </tt>
<a name="L679"></a><tt class="py-lineno"> 679</tt>  <tt class="py-line">        <tt class="py-comment"># Currently the parser uses a single string with newlines.</tt> </tt>
<a name="L680"></a><tt class="py-lineno"> 680</tt>  <tt class="py-line">        <tt class="py-comment"># A list of lines is also reasonable.</tt> </tt>
<a name="L681"></a><tt class="py-lineno"> 681</tt>  <tt class="py-line">        <tt class="py-comment"># A single (long) string is perhaps the most natural of all.</tt> </tt>
<a name="L682"></a><tt class="py-lineno"> 682</tt>  <tt class="py-line">        <tt class="py-comment"># This means we may need to deal with line wrapping.</tt> </tt>
<a name="L683"></a><tt class="py-lineno"> 683</tt>  <tt class="py-line">        <tt id="link-424" class="py-name"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-424', 'comment', 'link-420');">comment</a></tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-425" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-425', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"comment"</tt><tt class="py-op">]</tt> </tt>
<a name="L684"></a><tt class="py-lineno"> 684</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-426" class="py-name"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-426', 'comment', 'link-420');">comment</a></tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L685"></a><tt class="py-lineno"> 685</tt>  <tt class="py-line">            <tt class="py-name">lines</tt> <tt class="py-op">=</tt> <tt id="link-427" class="py-name"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-427', 'comment', 'link-420');">comment</a></tt><tt class="py-op">.</tt><tt id="link-428" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-428', 'split', 'link-263');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L686"></a><tt class="py-lineno"> 686</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-429" class="py-name"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-429', 'comment', 'link-420');">comment</a></tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-430" class="py-name"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-430', 'comment', 'link-420');">comment</a></tt><tt class="py-op">,</tt> <tt class="py-name">tuple</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L687"></a><tt class="py-lineno"> 687</tt>  <tt class="py-line">            <tt class="py-name">lines</tt> <tt class="py-op">=</tt> <tt id="link-431" class="py-name"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-431', 'comment', 'link-420');">comment</a></tt> </tt>
<a name="L688"></a><tt class="py-lineno"> 688</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L689"></a><tt class="py-lineno"> 689</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Could not understand comment annotation"</tt><tt class="py-op">)</tt> </tt>
<a name="L690"></a><tt class="py-lineno"> 690</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-432" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_multi_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_multi_line" class="py-name" href="#" onclick="return doclink('link-432', '_write_multi_line', 'link-391');">_write_multi_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"COMMENT"</tt><tt class="py-op">,</tt> <tt class="py-name">lines</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L691"></a><tt class="py-lineno"> 691</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">lines</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L692"></a><tt class="py-lineno"> 692</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-433" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_multi_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_multi_line" class="py-name" href="#" onclick="return doclink('link-433', '_write_multi_line', 'link-391');">_write_multi_line</a></tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
</div><a name="L693"></a><tt class="py-lineno"> 693</tt>  <tt class="py-line"> </tt>
<a name="GenBankWriter._write_contig"></a><div id="GenBankWriter._write_contig-def"><a name="L694"></a><tt class="py-lineno"> 694</tt> <a class="py-toggle" href="#" id="GenBankWriter._write_contig-toggle" onclick="return toggle('GenBankWriter._write_contig');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO.GenBankWriter-class.html#_write_contig">_write_contig</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GenBankWriter._write_contig-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="GenBankWriter._write_contig-expanded"><a name="L695"></a><tt class="py-lineno"> 695</tt>  <tt class="py-line">        <tt class="py-name">max_len</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-434" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter.MAX_WIDTH" class="py-name" href="#" onclick="return doclink('link-434', 'MAX_WIDTH', 'link-184');">MAX_WIDTH</a></tt> <tt class="py-op">-</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-435" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-435', 'HEADER_WIDTH', 'link-275');">HEADER_WIDTH</a></tt> </tt>
<a name="L696"></a><tt class="py-lineno"> 696</tt>  <tt class="py-line">        <tt class="py-name">lines</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-436" class="py-name" targets="Method Bio.SeqIO.InsdcIO._InsdcWriter._split_contig()=Bio.SeqIO.InsdcIO._InsdcWriter-class.html#_split_contig"><a title="Bio.SeqIO.InsdcIO._InsdcWriter._split_contig" class="py-name" href="#" onclick="return doclink('link-436', '_split_contig', 'link-436');">_split_contig</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">,</tt> <tt class="py-name">max_len</tt><tt class="py-op">)</tt> </tt>
<a name="L697"></a><tt class="py-lineno"> 697</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-437" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_single_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_single_line" class="py-name" href="#" onclick="return doclink('link-437', '_write_single_line', 'link-289');">_write_single_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"CONTIG"</tt><tt class="py-op">,</tt> <tt class="py-name">lines</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L698"></a><tt class="py-lineno"> 698</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">text</tt> <tt class="py-keyword">in</tt> <tt class="py-name">lines</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L699"></a><tt class="py-lineno"> 699</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-438" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_single_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_single_line" class="py-name" href="#" onclick="return doclink('link-438', '_write_single_line', 'link-289');">_write_single_line</a></tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">,</tt> <tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
</div><a name="L700"></a><tt class="py-lineno"> 700</tt>  <tt class="py-line"> </tt>
<a name="GenBankWriter._write_sequence"></a><div id="GenBankWriter._write_sequence-def"><a name="L701"></a><tt class="py-lineno"> 701</tt> <a class="py-toggle" href="#" id="GenBankWriter._write_sequence-toggle" onclick="return toggle('GenBankWriter._write_sequence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO.GenBankWriter-class.html#_write_sequence">_write_sequence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GenBankWriter._write_sequence-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="GenBankWriter._write_sequence-expanded"><a name="L702"></a><tt class="py-lineno"> 702</tt>  <tt class="py-line">        <tt class="py-comment"># Loosely based on code from Howard Salis</tt> </tt>
<a name="L703"></a><tt class="py-lineno"> 703</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Force lower case?</tt> </tt>
<a name="L704"></a><tt class="py-lineno"> 704</tt>  <tt class="py-line">        <tt class="py-name">LETTERS_PER_LINE</tt> <tt class="py-op">=</tt> <tt class="py-number">60</tt> </tt>
<a name="L705"></a><tt class="py-lineno"> 705</tt>  <tt class="py-line">        <tt class="py-name">SEQUENCE_INDENT</tt> <tt class="py-op">=</tt> <tt class="py-number">9</tt> </tt>
<a name="L706"></a><tt class="py-lineno"> 706</tt>  <tt class="py-line"> </tt>
<a name="L707"></a><tt class="py-lineno"> 707</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-439" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-439', 'seq', 'link-328');">seq</a></tt><tt class="py-op">,</tt> <tt id="link-440" class="py-name"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-440', 'UnknownSeq', 'link-4');">UnknownSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L708"></a><tt class="py-lineno"> 708</tt>  <tt class="py-line">            <tt class="py-comment"># We have already recorded the length, and there is no need</tt> </tt>
<a name="L709"></a><tt class="py-lineno"> 709</tt>  <tt class="py-line">            <tt class="py-comment"># to record a long sequence of NNNNNNN...NNN or whatever.</tt> </tt>
<a name="L710"></a><tt class="py-lineno"> 710</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-string">"contig"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-441" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-441', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">:</tt> </tt>
<a name="L711"></a><tt class="py-lineno"> 711</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-442" class="py-name" targets="Method Bio.SeqIO.InsdcIO.EmblWriter._write_contig()=Bio.SeqIO.InsdcIO.EmblWriter-class.html#_write_contig,Method Bio.SeqIO.InsdcIO.GenBankWriter._write_contig()=Bio.SeqIO.InsdcIO.GenBankWriter-class.html#_write_contig"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_contig
Bio.SeqIO.InsdcIO.GenBankWriter._write_contig" class="py-name" href="#" onclick="return doclink('link-442', '_write_contig', 'link-442');">_write_contig</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L712"></a><tt class="py-lineno"> 712</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L713"></a><tt class="py-lineno"> 713</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-443" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-443', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"ORIGIN\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L714"></a><tt class="py-lineno"> 714</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L715"></a><tt class="py-lineno"> 715</tt>  <tt class="py-line"> </tt>
<a name="L716"></a><tt class="py-lineno"> 716</tt>  <tt class="py-line">        <tt class="py-comment"># Catches sequence being None:</tt> </tt>
<a name="L717"></a><tt class="py-lineno"> 717</tt>  <tt class="py-line">        <tt id="link-444" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-444', 'data', 'link-366');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-445" class="py-name" targets="Method Bio.SeqIO.Interfaces.SequenceWriter._get_seq_string()=Bio.SeqIO.Interfaces.SequenceWriter-class.html#_get_seq_string"><a title="Bio.SeqIO.Interfaces.SequenceWriter._get_seq_string" class="py-name" href="#" onclick="return doclink('link-445', '_get_seq_string', 'link-445');">_get_seq_string</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-446" class="py-name" targets="Method Bio.Seq.Seq.lower()=Bio.Seq.Seq-class.html#lower,Method Bio.Seq.UnknownSeq.lower()=Bio.Seq.UnknownSeq-class.html#lower,Method Bio.SeqRecord.SeqRecord.lower()=Bio.SeqRecord.SeqRecord-class.html#lower"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-446', 'lower', 'link-446');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L718"></a><tt class="py-lineno"> 718</tt>  <tt class="py-line">        <tt class="py-name">seq_len</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-447" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-447', 'data', 'link-366');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L719"></a><tt class="py-lineno"> 719</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-448" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-448', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"ORIGIN\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L720"></a><tt class="py-lineno"> 720</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">line_number</tt> <tt class="py-keyword">in</tt> <tt id="link-449" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-449', 'range', 'link-206');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">seq_len</tt><tt class="py-op">,</tt> <tt class="py-name">LETTERS_PER_LINE</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L721"></a><tt class="py-lineno"> 721</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-450" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-450', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">line_number</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">rjust</tt><tt class="py-op">(</tt><tt class="py-name">SEQUENCE_INDENT</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L722"></a><tt class="py-lineno"> 722</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">words</tt> <tt class="py-keyword">in</tt> <tt id="link-451" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-451', 'range', 'link-206');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">line_number</tt><tt class="py-op">,</tt> </tt>
<a name="L723"></a><tt class="py-lineno"> 723</tt>  <tt class="py-line">                               <tt id="link-452" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-452', 'min', 'link-207');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">line_number</tt> <tt class="py-op">+</tt> <tt class="py-name">LETTERS_PER_LINE</tt><tt class="py-op">,</tt> <tt class="py-name">seq_len</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">10</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L724"></a><tt class="py-lineno"> 724</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-453" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-453', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">" %s"</tt> <tt class="py-op">%</tt> <tt id="link-454" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-454', 'data', 'link-366');">data</a></tt><tt class="py-op">[</tt><tt class="py-name">words</tt><tt class="py-op">:</tt><tt class="py-name">words</tt> <tt class="py-op">+</tt> <tt class="py-number">10</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L725"></a><tt class="py-lineno"> 725</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-455" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-455', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"\n"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L726"></a><tt class="py-lineno"> 726</tt>  <tt class="py-line"> </tt>
<a name="GenBankWriter.write_record"></a><div id="GenBankWriter.write_record-def"><a name="L727"></a><tt class="py-lineno"> 727</tt> <a class="py-toggle" href="#" id="GenBankWriter.write_record-toggle" onclick="return toggle('GenBankWriter.write_record');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO.GenBankWriter-class.html#write_record">write_record</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="GenBankWriter.write_record-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="GenBankWriter.write_record-expanded"><a name="L728"></a><tt class="py-lineno"> 728</tt>  <tt class="py-line">        <tt class="py-docstring">"""Write a single record to the output file."""</tt> </tt>
<a name="L729"></a><tt class="py-lineno"> 729</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt> </tt>
<a name="L730"></a><tt class="py-lineno"> 730</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-456" class="py-name" targets="Method Bio.SeqIO.InsdcIO.GenBankWriter._write_the_first_line()=Bio.SeqIO.InsdcIO.GenBankWriter-class.html#_write_the_first_line"><a title="Bio.SeqIO.InsdcIO.GenBankWriter._write_the_first_line" class="py-name" href="#" onclick="return doclink('link-456', '_write_the_first_line', 'link-456');">_write_the_first_line</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L731"></a><tt class="py-lineno"> 731</tt>  <tt class="py-line"> </tt>
<a name="L732"></a><tt class="py-lineno"> 732</tt>  <tt class="py-line">        <tt id="link-457" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.accession()=Bio.GenBank._FeatureConsumer-class.html#accession,Method Bio.GenBank._RecordConsumer.accession()=Bio.GenBank._RecordConsumer-class.html#accession,Method Bio.Phylo.PhyloXMLIO.Parser.accession()=Bio.Phylo.PhyloXMLIO.Parser-class.html#accession,Method Bio.Phylo.PhyloXMLIO.Writer.accession()=Bio.Phylo.PhyloXMLIO.Writer-class.html#accession"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-457', 'accession', 'link-457');">accession</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-458" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter._get_annotation_str" class="py-name" href="#" onclick="return doclink('link-458', '_get_annotation_str', 'link-322');">_get_annotation_str</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">,</tt> <tt class="py-string">"accession"</tt><tt class="py-op">,</tt> </tt>
<a name="L733"></a><tt class="py-lineno"> 733</tt>  <tt class="py-line">                                             <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-459" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-459', 'id', 'link-318');">id</a></tt><tt class="py-op">.</tt><tt id="link-460" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-460', 'split', 'link-263');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"."</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L734"></a><tt class="py-lineno"> 734</tt>  <tt class="py-line">                                             <tt class="py-name">just_first</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L735"></a><tt class="py-lineno"> 735</tt>  <tt class="py-line">        <tt class="py-name">acc_with_version</tt> <tt class="py-op">=</tt> <tt id="link-461" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-461', 'accession', 'link-457');">accession</a></tt> </tt>
<a name="L736"></a><tt class="py-lineno"> 736</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-462" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-462', 'id', 'link-318');">id</a></tt><tt class="py-op">.</tt><tt id="link-463" class="py-name" targets="Method Bio.Seq.Seq.startswith()=Bio.Seq.Seq-class.html#startswith"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-463', 'startswith', 'link-463');">startswith</a></tt><tt class="py-op">(</tt><tt id="link-464" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-464', 'accession', 'link-457');">accession</a></tt> <tt class="py-op">+</tt> <tt class="py-string">"."</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L737"></a><tt class="py-lineno"> 737</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L738"></a><tt class="py-lineno"> 738</tt>  <tt class="py-line">                <tt class="py-name">acc_with_version</tt> <tt class="py-op">=</tt> <tt class="py-string">"%s.%i"</tt> \ </tt>
<a name="L739"></a><tt class="py-lineno"> 739</tt>  <tt class="py-line">                                   <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-465" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-465', 'accession', 'link-457');">accession</a></tt><tt class="py-op">,</tt> </tt>
<a name="L740"></a><tt class="py-lineno"> 740</tt>  <tt class="py-line">                                      <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-466" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-466', 'id', 'link-318');">id</a></tt><tt class="py-op">.</tt><tt id="link-467" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-467', 'split', 'link-263');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"."</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L741"></a><tt class="py-lineno"> 741</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L742"></a><tt class="py-lineno"> 742</tt>  <tt class="py-line">                <tt class="py-keyword">pass</tt> </tt>
<a name="L743"></a><tt class="py-lineno"> 743</tt>  <tt class="py-line">        <tt id="link-468" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.gi()=Bio.GenBank._FeatureConsumer-class.html#gi,Method Bio.GenBank._RecordConsumer.gi()=Bio.GenBank._RecordConsumer-class.html#gi"><a title="Bio.GenBank._FeatureConsumer.gi
Bio.GenBank._RecordConsumer.gi" class="py-name" href="#" onclick="return doclink('link-468', 'gi', 'link-468');">gi</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-469" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter._get_annotation_str" class="py-name" href="#" onclick="return doclink('link-469', '_get_annotation_str', 'link-322');">_get_annotation_str</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">,</tt> <tt class="py-string">"gi"</tt><tt class="py-op">,</tt> <tt class="py-name">just_first</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L744"></a><tt class="py-lineno"> 744</tt>  <tt class="py-line"> </tt>
<a name="L745"></a><tt class="py-lineno"> 745</tt>  <tt class="py-line">        <tt class="py-name">descr</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-470" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DescriptionConsumer.description()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#description,Method Bio.Phylo.PhyloXMLIO.Writer.description()=Bio.Phylo.PhyloXMLIO.Writer-class.html#description,Variable Bio.SearchIO._model.hit.Hit.description=Bio.SearchIO._model.hit.Hit-class.html#description,Variable Bio.SearchIO._model.query.QueryResult.description=Bio.SearchIO._model.query.QueryResult-class.html#description"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-470', 'description', 'link-470');">description</a></tt> </tt>
<a name="L746"></a><tt class="py-lineno"> 746</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">descr</tt> <tt class="py-op">==</tt> <tt class="py-string">"&lt;unknown description&gt;"</tt><tt class="py-op">:</tt> </tt>
<a name="L747"></a><tt class="py-lineno"> 747</tt>  <tt class="py-line">            <tt class="py-name">descr</tt> <tt class="py-op">=</tt> <tt class="py-string">"."</tt> </tt>
<a name="L748"></a><tt class="py-lineno"> 748</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-471" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_multi_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_multi_line" class="py-name" href="#" onclick="return doclink('link-471', '_write_multi_line', 'link-391');">_write_multi_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"DEFINITION"</tt><tt class="py-op">,</tt> <tt class="py-name">descr</tt><tt class="py-op">)</tt> </tt>
<a name="L749"></a><tt class="py-lineno"> 749</tt>  <tt class="py-line"> </tt>
<a name="L750"></a><tt class="py-lineno"> 750</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-472" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_single_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_single_line" class="py-name" href="#" onclick="return doclink('link-472', '_write_single_line', 'link-289');">_write_single_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"ACCESSION"</tt><tt class="py-op">,</tt> <tt id="link-473" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-473', 'accession', 'link-457');">accession</a></tt><tt class="py-op">)</tt> </tt>
<a name="L751"></a><tt class="py-lineno"> 751</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-474" class="py-name"><a title="Bio.GenBank._FeatureConsumer.gi
Bio.GenBank._RecordConsumer.gi" class="py-name" href="#" onclick="return doclink('link-474', 'gi', 'link-468');">gi</a></tt> <tt class="py-op">!=</tt> <tt class="py-string">"."</tt><tt class="py-op">:</tt> </tt>
<a name="L752"></a><tt class="py-lineno"> 752</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-475" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_single_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_single_line" class="py-name" href="#" onclick="return doclink('link-475', '_write_single_line', 'link-289');">_write_single_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"VERSION"</tt><tt class="py-op">,</tt> <tt class="py-string">"%s  GI:%s"</tt> </tt>
<a name="L753"></a><tt class="py-lineno"> 753</tt>  <tt class="py-line">                                    <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">acc_with_version</tt><tt class="py-op">,</tt> <tt id="link-476" class="py-name"><a title="Bio.GenBank._FeatureConsumer.gi
Bio.GenBank._RecordConsumer.gi" class="py-name" href="#" onclick="return doclink('link-476', 'gi', 'link-468');">gi</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L754"></a><tt class="py-lineno"> 754</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L755"></a><tt class="py-lineno"> 755</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-477" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_single_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_single_line" class="py-name" href="#" onclick="return doclink('link-477', '_write_single_line', 'link-289');">_write_single_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"VERSION"</tt><tt class="py-op">,</tt> <tt class="py-string">"%s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">acc_with_version</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L756"></a><tt class="py-lineno"> 756</tt>  <tt class="py-line"> </tt>
<a name="L757"></a><tt class="py-lineno"> 757</tt>  <tt class="py-line">        <tt class="py-comment"># The NCBI only expect two types of link so far,</tt> </tt>
<a name="L758"></a><tt class="py-lineno"> 758</tt>  <tt class="py-line">        <tt class="py-comment"># e.g. "Project:28471" and "Trace Assembly Archive:123456"</tt> </tt>
<a name="L759"></a><tt class="py-lineno"> 759</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Filter the dbxrefs list to just these?</tt> </tt>
<a name="L760"></a><tt class="py-lineno"> 760</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-478" class="py-name" targets="Method Bio.SeqIO.InsdcIO.GenBankWriter._write_multi_entries()=Bio.SeqIO.InsdcIO.GenBankWriter-class.html#_write_multi_entries"><a title="Bio.SeqIO.InsdcIO.GenBankWriter._write_multi_entries" class="py-name" href="#" onclick="return doclink('link-478', '_write_multi_entries', 'link-478');">_write_multi_entries</a></tt><tt class="py-op">(</tt><tt class="py-string">"DBLINK"</tt><tt class="py-op">,</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-479" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.dbxrefs=BioSQL.BioSeq.DBSeqRecord-class.html#dbxrefs"><a title="BioSQL.BioSeq.DBSeqRecord.dbxrefs" class="py-name" href="#" onclick="return doclink('link-479', 'dbxrefs', 'link-479');">dbxrefs</a></tt><tt class="py-op">)</tt> </tt>
<a name="L761"></a><tt class="py-lineno"> 761</tt>  <tt class="py-line"> </tt>
<a name="L762"></a><tt class="py-lineno"> 762</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L763"></a><tt class="py-lineno"> 763</tt>  <tt class="py-line">            <tt class="py-comment"># List of strings</tt> </tt>
<a name="L764"></a><tt class="py-lineno"> 764</tt>  <tt class="py-line">            <tt class="py-comment"># Keywords should be given separated with semi colons,</tt> </tt>
<a name="L765"></a><tt class="py-lineno"> 765</tt>  <tt class="py-line">            <tt id="link-480" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.keywords()=Bio.GenBank._FeatureConsumer-class.html#keywords,Method Bio.GenBank._RecordConsumer.keywords()=Bio.GenBank._RecordConsumer-class.html#keywords"><a title="Bio.GenBank._FeatureConsumer.keywords
Bio.GenBank._RecordConsumer.keywords" class="py-name" href="#" onclick="return doclink('link-480', 'keywords', 'link-480');">keywords</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"; "</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-481" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-481', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"keywords"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L766"></a><tt class="py-lineno"> 766</tt>  <tt class="py-line">            <tt class="py-comment"># with a trailing period:</tt> </tt>
<a name="L767"></a><tt class="py-lineno"> 767</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-482" class="py-name"><a title="Bio.GenBank._FeatureConsumer.keywords
Bio.GenBank._RecordConsumer.keywords" class="py-name" href="#" onclick="return doclink('link-482', 'keywords', 'link-480');">keywords</a></tt><tt class="py-op">.</tt><tt id="link-483" class="py-name" targets="Method Bio.Seq.Seq.endswith()=Bio.Seq.Seq-class.html#endswith"><a title="Bio.Seq.Seq.endswith" class="py-name" href="#" onclick="return doclink('link-483', 'endswith', 'link-483');">endswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"."</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L768"></a><tt class="py-lineno"> 768</tt>  <tt class="py-line">                <tt id="link-484" class="py-name"><a title="Bio.GenBank._FeatureConsumer.keywords
Bio.GenBank._RecordConsumer.keywords" class="py-name" href="#" onclick="return doclink('link-484', 'keywords', 'link-480');">keywords</a></tt> <tt class="py-op">+=</tt> <tt class="py-string">"."</tt> </tt>
<a name="L769"></a><tt class="py-lineno"> 769</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L770"></a><tt class="py-lineno"> 770</tt>  <tt class="py-line">            <tt class="py-comment"># If no keywords, there should be just a period:</tt> </tt>
<a name="L771"></a><tt class="py-lineno"> 771</tt>  <tt class="py-line">            <tt id="link-485" class="py-name"><a title="Bio.GenBank._FeatureConsumer.keywords
Bio.GenBank._RecordConsumer.keywords" class="py-name" href="#" onclick="return doclink('link-485', 'keywords', 'link-480');">keywords</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"."</tt> </tt>
<a name="L772"></a><tt class="py-lineno"> 772</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-486" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_multi_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_multi_line" class="py-name" href="#" onclick="return doclink('link-486', '_write_multi_line', 'link-391');">_write_multi_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"KEYWORDS"</tt><tt class="py-op">,</tt> <tt id="link-487" class="py-name"><a title="Bio.GenBank._FeatureConsumer.keywords
Bio.GenBank._RecordConsumer.keywords" class="py-name" href="#" onclick="return doclink('link-487', 'keywords', 'link-480');">keywords</a></tt><tt class="py-op">)</tt> </tt>
<a name="L773"></a><tt class="py-lineno"> 773</tt>  <tt class="py-line"> </tt>
<a name="L774"></a><tt class="py-lineno"> 774</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"segment"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-488" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-488', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">:</tt> </tt>
<a name="L775"></a><tt class="py-lineno"> 775</tt>  <tt class="py-line">            <tt class="py-comment"># Deal with SEGMENT line found only in segmented records,</tt> </tt>
<a name="L776"></a><tt class="py-lineno"> 776</tt>  <tt class="py-line">            <tt class="py-comment"># e.g. AH000819</tt> </tt>
<a name="L777"></a><tt class="py-lineno"> 777</tt>  <tt class="py-line">            <tt id="link-489" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.segment()=Bio.GenBank._FeatureConsumer-class.html#segment,Method Bio.GenBank._RecordConsumer.segment()=Bio.GenBank._RecordConsumer-class.html#segment"><a title="Bio.GenBank._FeatureConsumer.segment
Bio.GenBank._RecordConsumer.segment" class="py-name" href="#" onclick="return doclink('link-489', 'segment', 'link-489');">segment</a></tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-490" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-490', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"segment"</tt><tt class="py-op">]</tt> </tt>
<a name="L778"></a><tt class="py-lineno"> 778</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-491" class="py-name"><a title="Bio.GenBank._FeatureConsumer.segment
Bio.GenBank._RecordConsumer.segment" class="py-name" href="#" onclick="return doclink('link-491', 'segment', 'link-489');">segment</a></tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L779"></a><tt class="py-lineno"> 779</tt>  <tt class="py-line">                <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-492" class="py-name"><a title="Bio.GenBank._FeatureConsumer.segment
Bio.GenBank._RecordConsumer.segment" class="py-name" href="#" onclick="return doclink('link-492', 'segment', 'link-489');">segment</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt id="link-493" class="py-name"><a title="Bio.GenBank._FeatureConsumer.segment
Bio.GenBank._RecordConsumer.segment" class="py-name" href="#" onclick="return doclink('link-493', 'segment', 'link-489');">segment</a></tt> </tt>
<a name="L780"></a><tt class="py-lineno"> 780</tt>  <tt class="py-line">                <tt id="link-494" class="py-name"><a title="Bio.GenBank._FeatureConsumer.segment
Bio.GenBank._RecordConsumer.segment" class="py-name" href="#" onclick="return doclink('link-494', 'segment', 'link-489');">segment</a></tt> <tt class="py-op">=</tt> <tt id="link-495" class="py-name"><a title="Bio.GenBank._FeatureConsumer.segment
Bio.GenBank._RecordConsumer.segment" class="py-name" href="#" onclick="return doclink('link-495', 'segment', 'link-489');">segment</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L781"></a><tt class="py-lineno"> 781</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-496" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_single_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_single_line" class="py-name" href="#" onclick="return doclink('link-496', '_write_single_line', 'link-289');">_write_single_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"SEGMENT"</tt><tt class="py-op">,</tt> <tt id="link-497" class="py-name"><a title="Bio.GenBank._FeatureConsumer.segment
Bio.GenBank._RecordConsumer.segment" class="py-name" href="#" onclick="return doclink('link-497', 'segment', 'link-489');">segment</a></tt><tt class="py-op">)</tt> </tt>
<a name="L782"></a><tt class="py-lineno"> 782</tt>  <tt class="py-line"> </tt>
<a name="L783"></a><tt class="py-lineno"> 783</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-498" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_multi_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_multi_line" class="py-name" href="#" onclick="return doclink('link-498', '_write_multi_line', 'link-391');">_write_multi_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"SOURCE"</tt><tt class="py-op">,</tt> </tt>
<a name="L784"></a><tt class="py-lineno"> 784</tt>  <tt class="py-line">                               <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-499" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter._get_annotation_str" class="py-name" href="#" onclick="return doclink('link-499', '_get_annotation_str', 'link-322');">_get_annotation_str</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">,</tt> <tt class="py-string">"source"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L785"></a><tt class="py-lineno"> 785</tt>  <tt class="py-line">        <tt class="py-comment"># The ORGANISM line MUST be a single line, as any continuation is the taxonomy</tt> </tt>
<a name="L786"></a><tt class="py-lineno"> 786</tt>  <tt class="py-line">        <tt class="py-name">org</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-500" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter._get_annotation_str" class="py-name" href="#" onclick="return doclink('link-500', '_get_annotation_str', 'link-322');">_get_annotation_str</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">,</tt> <tt class="py-string">"organism"</tt><tt class="py-op">)</tt> </tt>
<a name="L787"></a><tt class="py-lineno"> 787</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">org</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-501" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter.MAX_WIDTH" class="py-name" href="#" onclick="return doclink('link-501', 'MAX_WIDTH', 'link-184');">MAX_WIDTH</a></tt> <tt class="py-op">-</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-502" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-502', 'HEADER_WIDTH', 'link-275');">HEADER_WIDTH</a></tt><tt class="py-op">:</tt> </tt>
<a name="L788"></a><tt class="py-lineno"> 788</tt>  <tt class="py-line">            <tt class="py-name">org</tt> <tt class="py-op">=</tt> <tt class="py-name">org</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-503" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter.MAX_WIDTH" class="py-name" href="#" onclick="return doclink('link-503', 'MAX_WIDTH', 'link-184');">MAX_WIDTH</a></tt> <tt class="py-op">-</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-504" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-504', 'HEADER_WIDTH', 'link-275');">HEADER_WIDTH</a></tt> <tt class="py-op">-</tt> <tt class="py-number">4</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">"..."</tt> </tt>
<a name="L789"></a><tt class="py-lineno"> 789</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-505" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_single_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_single_line" class="py-name" href="#" onclick="return doclink('link-505', '_write_single_line', 'link-289');">_write_single_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"  ORGANISM"</tt><tt class="py-op">,</tt> <tt class="py-name">org</tt><tt class="py-op">)</tt> </tt>
<a name="L790"></a><tt class="py-lineno"> 790</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L791"></a><tt class="py-lineno"> 791</tt>  <tt class="py-line">            <tt class="py-comment"># List of strings</tt> </tt>
<a name="L792"></a><tt class="py-lineno"> 792</tt>  <tt class="py-line">            <tt class="py-comment"># Taxonomy should be given separated with semi colons,</tt> </tt>
<a name="L793"></a><tt class="py-lineno"> 793</tt>  <tt class="py-line">            <tt id="link-506" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.taxonomy()=Bio.GenBank._FeatureConsumer-class.html#taxonomy,Method Bio.GenBank._RecordConsumer.taxonomy()=Bio.GenBank._RecordConsumer-class.html#taxonomy,Variable Bio.Phylo.PhyloXML.Clade.taxonomy=Bio.Phylo.PhyloXML.Clade-class.html#taxonomy,Method Bio.Phylo.PhyloXMLIO.Writer.taxonomy()=Bio.Phylo.PhyloXMLIO.Writer-class.html#taxonomy"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-506', 'taxonomy', 'link-506');">taxonomy</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"; "</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-507" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-507', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"taxonomy"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L794"></a><tt class="py-lineno"> 794</tt>  <tt class="py-line">            <tt class="py-comment"># with a trailing period:</tt> </tt>
<a name="L795"></a><tt class="py-lineno"> 795</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-508" class="py-name"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-508', 'taxonomy', 'link-506');">taxonomy</a></tt><tt class="py-op">.</tt><tt id="link-509" class="py-name"><a title="Bio.Seq.Seq.endswith" class="py-name" href="#" onclick="return doclink('link-509', 'endswith', 'link-483');">endswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"."</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L796"></a><tt class="py-lineno"> 796</tt>  <tt class="py-line">                <tt id="link-510" class="py-name"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-510', 'taxonomy', 'link-506');">taxonomy</a></tt> <tt class="py-op">+=</tt> <tt class="py-string">"."</tt> </tt>
<a name="L797"></a><tt class="py-lineno"> 797</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L798"></a><tt class="py-lineno"> 798</tt>  <tt class="py-line">            <tt id="link-511" class="py-name"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-511', 'taxonomy', 'link-506');">taxonomy</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"."</tt> </tt>
<a name="L799"></a><tt class="py-lineno"> 799</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-512" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_multi_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_multi_line" class="py-name" href="#" onclick="return doclink('link-512', '_write_multi_line', 'link-391');">_write_multi_line</a></tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">,</tt> <tt id="link-513" class="py-name"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-513', 'taxonomy', 'link-506');">taxonomy</a></tt><tt class="py-op">)</tt> </tt>
<a name="L800"></a><tt class="py-lineno"> 800</tt>  <tt class="py-line"> </tt>
<a name="L801"></a><tt class="py-lineno"> 801</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"references"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-514" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-514', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">:</tt> </tt>
<a name="L802"></a><tt class="py-lineno"> 802</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-515" class="py-name" targets="Method Bio.SeqIO.InsdcIO.EmblWriter._write_references()=Bio.SeqIO.InsdcIO.EmblWriter-class.html#_write_references,Method Bio.SeqIO.InsdcIO.GenBankWriter._write_references()=Bio.SeqIO.InsdcIO.GenBankWriter-class.html#_write_references"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_references
Bio.SeqIO.InsdcIO.GenBankWriter._write_references" class="py-name" href="#" onclick="return doclink('link-515', '_write_references', 'link-515');">_write_references</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L803"></a><tt class="py-lineno"> 803</tt>  <tt class="py-line"> </tt>
<a name="L804"></a><tt class="py-lineno"> 804</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"comment"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-516" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-516', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">:</tt> </tt>
<a name="L805"></a><tt class="py-lineno"> 805</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-517" class="py-name" targets="Method Bio.SeqIO.InsdcIO.EmblWriter._write_comment()=Bio.SeqIO.InsdcIO.EmblWriter-class.html#_write_comment,Method Bio.SeqIO.InsdcIO.GenBankWriter._write_comment()=Bio.SeqIO.InsdcIO.GenBankWriter-class.html#_write_comment"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_comment
Bio.SeqIO.InsdcIO.GenBankWriter._write_comment" class="py-name" href="#" onclick="return doclink('link-517', '_write_comment', 'link-517');">_write_comment</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L806"></a><tt class="py-lineno"> 806</tt>  <tt class="py-line"> </tt>
<a name="L807"></a><tt class="py-lineno"> 807</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-518" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-518', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"FEATURES             Location/Qualifiers\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L808"></a><tt class="py-lineno"> 808</tt>  <tt class="py-line">        <tt class="py-name">rec_length</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L809"></a><tt class="py-lineno"> 809</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">feature</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-519" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.features=BioSQL.BioSeq.DBSeqRecord-class.html#features"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-519', 'features', 'link-519');">features</a></tt><tt class="py-op">:</tt> </tt>
<a name="L810"></a><tt class="py-lineno"> 810</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-520" class="py-name" targets="Method Bio.SeqIO.InsdcIO._InsdcWriter._write_feature()=Bio.SeqIO.InsdcIO._InsdcWriter-class.html#_write_feature"><a title="Bio.SeqIO.InsdcIO._InsdcWriter._write_feature" class="py-name" href="#" onclick="return doclink('link-520', '_write_feature', 'link-520');">_write_feature</a></tt><tt class="py-op">(</tt><tt class="py-name">feature</tt><tt class="py-op">,</tt> <tt class="py-name">rec_length</tt><tt class="py-op">)</tt> </tt>
<a name="L811"></a><tt class="py-lineno"> 811</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-521" class="py-name" targets="Method Bio.SeqIO.InsdcIO.EmblWriter._write_sequence()=Bio.SeqIO.InsdcIO.EmblWriter-class.html#_write_sequence,Method Bio.SeqIO.InsdcIO.GenBankWriter._write_sequence()=Bio.SeqIO.InsdcIO.GenBankWriter-class.html#_write_sequence"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_sequence
Bio.SeqIO.InsdcIO.GenBankWriter._write_sequence" class="py-name" href="#" onclick="return doclink('link-521', '_write_sequence', 'link-521');">_write_sequence</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L812"></a><tt class="py-lineno"> 812</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-522" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-522', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"//\n"</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L813"></a><tt class="py-lineno"> 813</tt>  <tt class="py-line"> </tt>
<a name="L814"></a><tt class="py-lineno"> 814</tt>  <tt class="py-line"> </tt>
<a name="EmblWriter"></a><div id="EmblWriter-def"><a name="L815"></a><tt class="py-lineno"> 815</tt> <a class="py-toggle" href="#" id="EmblWriter-toggle" onclick="return toggle('EmblWriter');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO.EmblWriter-class.html">EmblWriter</a><tt class="py-op">(</tt><tt class="py-base-class">_InsdcWriter</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EmblWriter-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="EmblWriter-expanded"><a name="L816"></a><tt class="py-lineno"> 816</tt>  <tt class="py-line">    <tt id="link-523" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-523', 'HEADER_WIDTH', 'link-275');">HEADER_WIDTH</a></tt> <tt class="py-op">=</tt> <tt class="py-number">5</tt> </tt>
<a name="L817"></a><tt class="py-lineno"> 817</tt>  <tt class="py-line">    <tt id="link-524" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO.GenBankWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-524', 'QUALIFIER_INDENT', 'link-185');">QUALIFIER_INDENT</a></tt> <tt class="py-op">=</tt> <tt class="py-number">21</tt> </tt>
<a name="L818"></a><tt class="py-lineno"> 818</tt>  <tt class="py-line">    <tt id="link-525" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT_STR
Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT_STR
Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT_STR" class="py-name" href="#" onclick="return doclink('link-525', 'QUALIFIER_INDENT_STR', 'link-186');">QUALIFIER_INDENT_STR</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"FT"</tt> <tt class="py-op">+</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt id="link-526" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO.GenBankWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-526', 'QUALIFIER_INDENT', 'link-185');">QUALIFIER_INDENT</a></tt> <tt class="py-op">-</tt> <tt class="py-number">2</tt><tt class="py-op">)</tt> </tt>
<a name="L819"></a><tt class="py-lineno"> 819</tt>  <tt class="py-line">    <tt id="link-527" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT_TMP
Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT_TMP
Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT_TMP" class="py-name" href="#" onclick="return doclink('link-527', 'QUALIFIER_INDENT_TMP', 'link-188');">QUALIFIER_INDENT_TMP</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"FT   %s                "</tt>  <tt class="py-comment"># 21 if %s is empty</tt> </tt>
<a name="L820"></a><tt class="py-lineno"> 820</tt>  <tt class="py-line">    <tt id="link-528" class="py-name" targets="Variable Bio.SeqIO.InsdcIO.EmblWriter.FEATURE_HEADER=Bio.SeqIO.InsdcIO.EmblWriter-class.html#FEATURE_HEADER,Variable Bio.SeqIO.InsdcIO.ImgtWriter.FEATURE_HEADER=Bio.SeqIO.InsdcIO.ImgtWriter-class.html#FEATURE_HEADER"><a title="Bio.SeqIO.InsdcIO.EmblWriter.FEATURE_HEADER
Bio.SeqIO.InsdcIO.ImgtWriter.FEATURE_HEADER" class="py-name" href="#" onclick="return doclink('link-528', 'FEATURE_HEADER', 'link-528');">FEATURE_HEADER</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"FH   Key             Location/Qualifiers\n"</tt> </tt>
<a name="L821"></a><tt class="py-lineno"> 821</tt>  <tt class="py-line"> </tt>
<a name="EmblWriter._write_contig"></a><div id="EmblWriter._write_contig-def"><a name="L822"></a><tt class="py-lineno"> 822</tt> <a class="py-toggle" href="#" id="EmblWriter._write_contig-toggle" onclick="return toggle('EmblWriter._write_contig');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO.EmblWriter-class.html#_write_contig">_write_contig</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EmblWriter._write_contig-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EmblWriter._write_contig-expanded"><a name="L823"></a><tt class="py-lineno"> 823</tt>  <tt class="py-line">        <tt class="py-name">max_len</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-529" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter.MAX_WIDTH" class="py-name" href="#" onclick="return doclink('link-529', 'MAX_WIDTH', 'link-184');">MAX_WIDTH</a></tt> <tt class="py-op">-</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-530" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-530', 'HEADER_WIDTH', 'link-275');">HEADER_WIDTH</a></tt> </tt>
<a name="L824"></a><tt class="py-lineno"> 824</tt>  <tt class="py-line">        <tt class="py-name">lines</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-531" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter._split_contig" class="py-name" href="#" onclick="return doclink('link-531', '_split_contig', 'link-436');">_split_contig</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">,</tt> <tt class="py-name">max_len</tt><tt class="py-op">)</tt> </tt>
<a name="L825"></a><tt class="py-lineno"> 825</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">text</tt> <tt class="py-keyword">in</tt> <tt class="py-name">lines</tt><tt class="py-op">:</tt> </tt>
<a name="L826"></a><tt class="py-lineno"> 826</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-532" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_single_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_single_line" class="py-name" href="#" onclick="return doclink('link-532', '_write_single_line', 'link-289');">_write_single_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"CO"</tt><tt class="py-op">,</tt> <tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
</div><a name="L827"></a><tt class="py-lineno"> 827</tt>  <tt class="py-line"> </tt>
<a name="EmblWriter._write_sequence"></a><div id="EmblWriter._write_sequence-def"><a name="L828"></a><tt class="py-lineno"> 828</tt> <a class="py-toggle" href="#" id="EmblWriter._write_sequence-toggle" onclick="return toggle('EmblWriter._write_sequence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO.EmblWriter-class.html#_write_sequence">_write_sequence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EmblWriter._write_sequence-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EmblWriter._write_sequence-expanded"><a name="L829"></a><tt class="py-lineno"> 829</tt>  <tt class="py-line">        <tt class="py-name">LETTERS_PER_BLOCK</tt> <tt class="py-op">=</tt> <tt class="py-number">10</tt> </tt>
<a name="L830"></a><tt class="py-lineno"> 830</tt>  <tt class="py-line">        <tt class="py-name">BLOCKS_PER_LINE</tt> <tt class="py-op">=</tt> <tt class="py-number">6</tt> </tt>
<a name="L831"></a><tt class="py-lineno"> 831</tt>  <tt class="py-line">        <tt class="py-name">LETTERS_PER_LINE</tt> <tt class="py-op">=</tt> <tt class="py-name">LETTERS_PER_BLOCK</tt> <tt class="py-op">*</tt> <tt class="py-name">BLOCKS_PER_LINE</tt> </tt>
<a name="L832"></a><tt class="py-lineno"> 832</tt>  <tt class="py-line">        <tt class="py-name">POSITION_PADDING</tt> <tt class="py-op">=</tt> <tt class="py-number">10</tt> </tt>
<a name="L833"></a><tt class="py-lineno"> 833</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt>  <tt class="py-comment"># save looking up this multiple times</tt> </tt>
<a name="L834"></a><tt class="py-lineno"> 834</tt>  <tt class="py-line"> </tt>
<a name="L835"></a><tt class="py-lineno"> 835</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-533" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-533', 'seq', 'link-328');">seq</a></tt><tt class="py-op">,</tt> <tt id="link-534" class="py-name"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-534', 'UnknownSeq', 'link-4');">UnknownSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L836"></a><tt class="py-lineno"> 836</tt>  <tt class="py-line">            <tt class="py-comment"># We have already recorded the length, and there is no need</tt> </tt>
<a name="L837"></a><tt class="py-lineno"> 837</tt>  <tt class="py-line">            <tt class="py-comment"># to record a long sequence of NNNNNNN...NNN or whatever.</tt> </tt>
<a name="L838"></a><tt class="py-lineno"> 838</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-string">"contig"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-535" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-535', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">:</tt> </tt>
<a name="L839"></a><tt class="py-lineno"> 839</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-536" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_contig
Bio.SeqIO.InsdcIO.GenBankWriter._write_contig" class="py-name" href="#" onclick="return doclink('link-536', '_write_contig', 'link-442');">_write_contig</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L840"></a><tt class="py-lineno"> 840</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L841"></a><tt class="py-lineno"> 841</tt>  <tt class="py-line">                <tt class="py-comment"># TODO - Can the sequence just be left out as in GenBank files?</tt> </tt>
<a name="L842"></a><tt class="py-lineno"> 842</tt>  <tt class="py-line">                <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-537" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-537', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"SQ   \n"</tt><tt class="py-op">)</tt> </tt>
<a name="L843"></a><tt class="py-lineno"> 843</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L844"></a><tt class="py-lineno"> 844</tt>  <tt class="py-line"> </tt>
<a name="L845"></a><tt class="py-lineno"> 845</tt>  <tt class="py-line">        <tt class="py-comment"># Catches sequence being None</tt> </tt>
<a name="L846"></a><tt class="py-lineno"> 846</tt>  <tt class="py-line">        <tt id="link-538" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-538', 'data', 'link-366');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-539" class="py-name"><a title="Bio.SeqIO.Interfaces.SequenceWriter._get_seq_string" class="py-name" href="#" onclick="return doclink('link-539', '_get_seq_string', 'link-445');">_get_seq_string</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-540" class="py-name"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-540', 'lower', 'link-446');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L847"></a><tt class="py-lineno"> 847</tt>  <tt class="py-line">        <tt class="py-name">seq_len</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-541" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-541', 'data', 'link-366');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L848"></a><tt class="py-lineno"> 848</tt>  <tt class="py-line"> </tt>
<a name="L849"></a><tt class="py-lineno"> 849</tt>  <tt class="py-line">        <tt class="py-comment"># Get the base alphabet (underneath any Gapped or StopCodon encoding)</tt> </tt>
<a name="L850"></a><tt class="py-lineno"> 850</tt>  <tt class="py-line">        <tt id="link-542" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-542', 'a', 'link-325');">a</a></tt> <tt class="py-op">=</tt> <tt id="link-543" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-543', 'Alphabet', 'link-12');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-544" class="py-name"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-544', '_get_base_alphabet', 'link-327');">_get_base_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-545" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-545', 'seq', 'link-328');">seq</a></tt><tt class="py-op">.</tt><tt id="link-546" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-546', 'alphabet', 'link-32');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L851"></a><tt class="py-lineno"> 851</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-547" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-547', 'a', 'link-325');">a</a></tt><tt class="py-op">,</tt> <tt id="link-548" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-548', 'Alphabet', 'link-12');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-549" class="py-name"><a title="Bio.Alphabet.DNAAlphabet" class="py-name" href="#" onclick="return doclink('link-549', 'DNAAlphabet', 'link-344');">DNAAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L852"></a><tt class="py-lineno"> 852</tt>  <tt class="py-line">            <tt class="py-comment"># TODO - What if we have RNA?</tt> </tt>
<a name="L853"></a><tt class="py-lineno"> 853</tt>  <tt class="py-line">            <tt class="py-name">a_count</tt> <tt class="py-op">=</tt> <tt id="link-550" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-550', 'data', 'link-366');">data</a></tt><tt class="py-op">.</tt><tt id="link-551" class="py-name" targets="Method Bio.Crystal.Chain.count()=Bio.Crystal.Chain-class.html#count,Method Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count()=Bio.NeuralNetwork.Gene.Pattern.PatternRepository-class.html#count,Method Bio.Seq.MutableSeq.count()=Bio.Seq.MutableSeq-class.html#count,Method Bio.Seq.Seq.count()=Bio.Seq.Seq-class.html#count,Method Bio.Seq.UnknownSeq.count()=Bio.Seq.UnknownSeq-class.html#count,Method Bio.motifs.Instances.count()=Bio.motifs.Instances-class.html#count"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-551', 'count', 'link-551');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">'A'</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt id="link-552" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-552', 'data', 'link-366');">data</a></tt><tt class="py-op">.</tt><tt id="link-553" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-553', 'count', 'link-551');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">'a'</tt><tt class="py-op">)</tt> </tt>
<a name="L854"></a><tt class="py-lineno"> 854</tt>  <tt class="py-line">            <tt class="py-name">c_count</tt> <tt class="py-op">=</tt> <tt id="link-554" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-554', 'data', 'link-366');">data</a></tt><tt class="py-op">.</tt><tt id="link-555" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-555', 'count', 'link-551');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">'C'</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt id="link-556" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-556', 'data', 'link-366');">data</a></tt><tt class="py-op">.</tt><tt id="link-557" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-557', 'count', 'link-551');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">'c'</tt><tt class="py-op">)</tt> </tt>
<a name="L855"></a><tt class="py-lineno"> 855</tt>  <tt class="py-line">            <tt class="py-name">g_count</tt> <tt class="py-op">=</tt> <tt id="link-558" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-558', 'data', 'link-366');">data</a></tt><tt class="py-op">.</tt><tt id="link-559" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-559', 'count', 'link-551');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">'G'</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt id="link-560" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-560', 'data', 'link-366');">data</a></tt><tt class="py-op">.</tt><tt id="link-561" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-561', 'count', 'link-551');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">'g'</tt><tt class="py-op">)</tt> </tt>
<a name="L856"></a><tt class="py-lineno"> 856</tt>  <tt class="py-line">            <tt class="py-name">t_count</tt> <tt class="py-op">=</tt> <tt id="link-562" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-562', 'data', 'link-366');">data</a></tt><tt class="py-op">.</tt><tt id="link-563" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-563', 'count', 'link-551');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">'T'</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt id="link-564" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-564', 'data', 'link-366');">data</a></tt><tt class="py-op">.</tt><tt id="link-565" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-565', 'count', 'link-551');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">'t'</tt><tt class="py-op">)</tt> </tt>
<a name="L857"></a><tt class="py-lineno"> 857</tt>  <tt class="py-line">            <tt id="link-566" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.other()=Bio.Phylo.PhyloXMLIO.Parser-class.html#other,Method Bio.Phylo.PhyloXMLIO.Writer.other()=Bio.Phylo.PhyloXMLIO.Writer-class.html#other"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-566', 'other', 'link-566');">other</a></tt> <tt class="py-op">=</tt> <tt class="py-name">seq_len</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">a_count</tt> <tt class="py-op">+</tt> <tt class="py-name">c_count</tt> <tt class="py-op">+</tt> <tt class="py-name">g_count</tt> <tt class="py-op">+</tt> <tt class="py-name">t_count</tt><tt class="py-op">)</tt> </tt>
<a name="L858"></a><tt class="py-lineno"> 858</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-567" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-567', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"SQ   Sequence %i BP; %i A; %i C; %i G; %i T; %i other;\n"</tt> </tt>
<a name="L859"></a><tt class="py-lineno"> 859</tt>  <tt class="py-line">                         <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">seq_len</tt><tt class="py-op">,</tt> <tt class="py-name">a_count</tt><tt class="py-op">,</tt> <tt class="py-name">c_count</tt><tt class="py-op">,</tt> <tt class="py-name">g_count</tt><tt class="py-op">,</tt> <tt class="py-name">t_count</tt><tt class="py-op">,</tt> <tt id="link-568" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-568', 'other', 'link-566');">other</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L860"></a><tt class="py-lineno"> 860</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L861"></a><tt class="py-lineno"> 861</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-569" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-569', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"SQ   \n"</tt><tt class="py-op">)</tt> </tt>
<a name="L862"></a><tt class="py-lineno"> 862</tt>  <tt class="py-line"> </tt>
<a name="L863"></a><tt class="py-lineno"> 863</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">line_number</tt> <tt class="py-keyword">in</tt> <tt id="link-570" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-570', 'range', 'link-206');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">seq_len</tt> <tt class="py-op">//</tt> <tt class="py-name">LETTERS_PER_LINE</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L864"></a><tt class="py-lineno"> 864</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-571" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-571', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"    "</tt><tt class="py-op">)</tt>  <tt class="py-comment"># Just four, not five</tt> </tt>
<a name="L865"></a><tt class="py-lineno"> 865</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">block</tt> <tt class="py-keyword">in</tt> <tt id="link-572" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-572', 'range', 'link-206');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">BLOCKS_PER_LINE</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L866"></a><tt class="py-lineno"> 866</tt>  <tt class="py-line">                <tt id="link-573" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-573', 'index', 'link-205');">index</a></tt> <tt class="py-op">=</tt> <tt class="py-name">LETTERS_PER_LINE</tt> <tt class="py-op">*</tt> <tt class="py-name">line_number</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L867"></a><tt class="py-lineno"> 867</tt>  <tt class="py-line">                    <tt class="py-name">LETTERS_PER_BLOCK</tt> <tt class="py-op">*</tt> <tt class="py-name">block</tt> </tt>
<a name="L868"></a><tt class="py-lineno"> 868</tt>  <tt class="py-line">                <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-574" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-574', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">" %s"</tt> <tt class="py-op">%</tt> <tt id="link-575" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-575', 'data', 'link-366');">data</a></tt><tt class="py-op">[</tt><tt id="link-576" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-576', 'index', 'link-205');">index</a></tt><tt class="py-op">:</tt><tt id="link-577" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-577', 'index', 'link-205');">index</a></tt> <tt class="py-op">+</tt> <tt class="py-name">LETTERS_PER_BLOCK</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L869"></a><tt class="py-lineno"> 869</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-578" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-578', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">line_number</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L870"></a><tt class="py-lineno"> 870</tt>  <tt class="py-line">                             <tt class="py-op">*</tt> <tt class="py-name">LETTERS_PER_LINE</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">rjust</tt><tt class="py-op">(</tt><tt class="py-name">POSITION_PADDING</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L871"></a><tt class="py-lineno"> 871</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-579" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-579', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L872"></a><tt class="py-lineno"> 872</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">seq_len</tt> <tt class="py-op">%</tt> <tt class="py-name">LETTERS_PER_LINE</tt><tt class="py-op">:</tt> </tt>
<a name="L873"></a><tt class="py-lineno"> 873</tt>  <tt class="py-line">            <tt class="py-comment"># Final (partial) line</tt> </tt>
<a name="L874"></a><tt class="py-lineno"> 874</tt>  <tt class="py-line">            <tt class="py-name">line_number</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">seq_len</tt> <tt class="py-op">//</tt> <tt class="py-name">LETTERS_PER_LINE</tt><tt class="py-op">)</tt> </tt>
<a name="L875"></a><tt class="py-lineno"> 875</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-580" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-580', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"    "</tt><tt class="py-op">)</tt>  <tt class="py-comment"># Just four, not five</tt> </tt>
<a name="L876"></a><tt class="py-lineno"> 876</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">block</tt> <tt class="py-keyword">in</tt> <tt id="link-581" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-581', 'range', 'link-206');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">BLOCKS_PER_LINE</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L877"></a><tt class="py-lineno"> 877</tt>  <tt class="py-line">                <tt id="link-582" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-582', 'index', 'link-205');">index</a></tt> <tt class="py-op">=</tt> <tt class="py-name">LETTERS_PER_LINE</tt> <tt class="py-op">*</tt> <tt class="py-name">line_number</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L878"></a><tt class="py-lineno"> 878</tt>  <tt class="py-line">                    <tt class="py-name">LETTERS_PER_BLOCK</tt> <tt class="py-op">*</tt> <tt class="py-name">block</tt> </tt>
<a name="L879"></a><tt class="py-lineno"> 879</tt>  <tt class="py-line">                <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-583" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-583', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt> </tt>
<a name="L880"></a><tt class="py-lineno"> 880</tt>  <tt class="py-line">                    <tt class="py-op">(</tt><tt class="py-string">" %s"</tt> <tt class="py-op">%</tt> <tt id="link-584" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-584', 'data', 'link-366');">data</a></tt><tt class="py-op">[</tt><tt id="link-585" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-585', 'index', 'link-205');">index</a></tt><tt class="py-op">:</tt><tt id="link-586" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-586', 'index', 'link-205');">index</a></tt> <tt class="py-op">+</tt> <tt class="py-name">LETTERS_PER_BLOCK</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">ljust</tt><tt class="py-op">(</tt><tt class="py-number">11</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L881"></a><tt class="py-lineno"> 881</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-587" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-587', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">seq_len</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">rjust</tt><tt class="py-op">(</tt><tt class="py-name">POSITION_PADDING</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L882"></a><tt class="py-lineno"> 882</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-588" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-588', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"\n"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L883"></a><tt class="py-lineno"> 883</tt>  <tt class="py-line"> </tt>
<a name="EmblWriter._write_single_line"></a><div id="EmblWriter._write_single_line-def"><a name="L884"></a><tt class="py-lineno"> 884</tt> <a class="py-toggle" href="#" id="EmblWriter._write_single_line-toggle" onclick="return toggle('EmblWriter._write_single_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO.EmblWriter-class.html#_write_single_line">_write_single_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">tag</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EmblWriter._write_single_line-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EmblWriter._write_single_line-expanded"><a name="L885"></a><tt class="py-lineno"> 885</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-589" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-589', 'tag', 'link-277');">tag</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt> </tt>
<a name="L886"></a><tt class="py-lineno"> 886</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-590" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-590', 'tag', 'link-277');">tag</a></tt> <tt class="py-op">+</tt> <tt class="py-string">"   "</tt> <tt class="py-op">+</tt> <tt class="py-name">text</tt> </tt>
<a name="L887"></a><tt class="py-lineno"> 887</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-591" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter.MAX_WIDTH" class="py-name" href="#" onclick="return doclink('link-591', 'MAX_WIDTH', 'link-184');">MAX_WIDTH</a></tt><tt class="py-op">:</tt> </tt>
<a name="L888"></a><tt class="py-lineno"> 888</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Line %r too long"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">,</tt> <tt id="link-592" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-592', 'BiopythonWarning', 'link-1');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L889"></a><tt class="py-lineno"> 889</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-593" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-593', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L890"></a><tt class="py-lineno"> 890</tt>  <tt class="py-line"> </tt>
<a name="EmblWriter._write_multi_line"></a><div id="EmblWriter._write_multi_line-def"><a name="L891"></a><tt class="py-lineno"> 891</tt> <a class="py-toggle" href="#" id="EmblWriter._write_multi_line-toggle" onclick="return toggle('EmblWriter._write_multi_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO.EmblWriter-class.html#_write_multi_line">_write_multi_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">tag</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EmblWriter._write_multi_line-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EmblWriter._write_multi_line-expanded"><a name="L892"></a><tt class="py-lineno"> 892</tt>  <tt class="py-line">        <tt class="py-name">max_len</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-594" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter.MAX_WIDTH" class="py-name" href="#" onclick="return doclink('link-594', 'MAX_WIDTH', 'link-184');">MAX_WIDTH</a></tt> <tt class="py-op">-</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-595" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-595', 'HEADER_WIDTH', 'link-275');">HEADER_WIDTH</a></tt> </tt>
<a name="L893"></a><tt class="py-lineno"> 893</tt>  <tt class="py-line">        <tt class="py-name">lines</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-596" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter._split_multi_line" class="py-name" href="#" onclick="return doclink('link-596', '_split_multi_line', 'link-288');">_split_multi_line</a></tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">,</tt> <tt class="py-name">max_len</tt><tt class="py-op">)</tt> </tt>
<a name="L894"></a><tt class="py-lineno"> 894</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">lines</tt><tt class="py-op">:</tt> </tt>
<a name="L895"></a><tt class="py-lineno"> 895</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-597" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_single_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_single_line" class="py-name" href="#" onclick="return doclink('link-597', '_write_single_line', 'link-289');">_write_single_line</a></tt><tt class="py-op">(</tt><tt id="link-598" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-598', 'tag', 'link-277');">tag</a></tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
</div><a name="L896"></a><tt class="py-lineno"> 896</tt>  <tt class="py-line"> </tt>
<a name="EmblWriter._write_the_first_lines"></a><div id="EmblWriter._write_the_first_lines-def"><a name="L897"></a><tt class="py-lineno"> 897</tt> <a class="py-toggle" href="#" id="EmblWriter._write_the_first_lines-toggle" onclick="return toggle('EmblWriter._write_the_first_lines');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO.EmblWriter-class.html#_write_the_first_lines">_write_the_first_lines</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EmblWriter._write_the_first_lines-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EmblWriter._write_the_first_lines-expanded"><a name="L898"></a><tt class="py-lineno"> 898</tt>  <tt class="py-line">        <tt class="py-docstring">"""Write the ID and AC lines."""</tt> </tt>
<a name="L899"></a><tt class="py-lineno"> 899</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"."</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-599" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-599', 'id', 'link-318');">id</a></tt> <tt class="py-keyword">and</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-600" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-600', 'id', 'link-318');">id</a></tt><tt class="py-op">.</tt><tt id="link-601" class="py-name" targets="Method Bio.Seq.Seq.rsplit()=Bio.Seq.Seq-class.html#rsplit"><a title="Bio.Seq.Seq.rsplit" class="py-name" href="#" onclick="return doclink('link-601', 'rsplit', 'link-601');">rsplit</a></tt><tt class="py-op">(</tt><tt class="py-string">"."</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">isdigit</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L900"></a><tt class="py-lineno"> 900</tt>  <tt class="py-line">            <tt id="link-602" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HeaderConsumer.version()=Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#version,Function Bio.Cluster.cluster.version()=Bio.Cluster.cluster-module.html#version,Method Bio.GenBank._FeatureConsumer.version()=Bio.GenBank._FeatureConsumer-class.html#version,Method Bio.GenBank._RecordConsumer.version()=Bio.GenBank._RecordConsumer-class.html#version,Variable Bio.motifs.jaspar.Motif.version=Bio.motifs.jaspar.Motif-class.html#version"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-602', 'version', 'link-602');">version</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"SV "</tt> <tt class="py-op">+</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-603" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-603', 'id', 'link-318');">id</a></tt><tt class="py-op">.</tt><tt id="link-604" class="py-name"><a title="Bio.Seq.Seq.rsplit" class="py-name" href="#" onclick="return doclink('link-604', 'rsplit', 'link-601');">rsplit</a></tt><tt class="py-op">(</tt><tt class="py-string">"."</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L901"></a><tt class="py-lineno"> 901</tt>  <tt class="py-line">            <tt id="link-605" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-605', 'accession', 'link-457');">accession</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-606" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter._get_annotation_str" class="py-name" href="#" onclick="return doclink('link-606', '_get_annotation_str', 'link-322');">_get_annotation_str</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">,</tt> <tt class="py-string">"accession"</tt><tt class="py-op">,</tt> </tt>
<a name="L902"></a><tt class="py-lineno"> 902</tt>  <tt class="py-line">                                                 <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-607" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-607', 'id', 'link-318');">id</a></tt><tt class="py-op">.</tt><tt id="link-608" class="py-name"><a title="Bio.Seq.Seq.rsplit" class="py-name" href="#" onclick="return doclink('link-608', 'rsplit', 'link-601');">rsplit</a></tt><tt class="py-op">(</tt><tt class="py-string">"."</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L903"></a><tt class="py-lineno"> 903</tt>  <tt class="py-line">                                                 <tt class="py-name">just_first</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L904"></a><tt class="py-lineno"> 904</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L905"></a><tt class="py-lineno"> 905</tt>  <tt class="py-line">            <tt id="link-609" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-609', 'version', 'link-602');">version</a></tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L906"></a><tt class="py-lineno"> 906</tt>  <tt class="py-line">            <tt id="link-610" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-610', 'accession', 'link-457');">accession</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-611" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter._get_annotation_str" class="py-name" href="#" onclick="return doclink('link-611', '_get_annotation_str', 'link-322');">_get_annotation_str</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">,</tt> <tt class="py-string">"accession"</tt><tt class="py-op">,</tt> </tt>
<a name="L907"></a><tt class="py-lineno"> 907</tt>  <tt class="py-line">                                                 <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-612" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-612', 'id', 'link-318');">id</a></tt><tt class="py-op">,</tt> </tt>
<a name="L908"></a><tt class="py-lineno"> 908</tt>  <tt class="py-line">                                                 <tt class="py-name">just_first</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L909"></a><tt class="py-lineno"> 909</tt>  <tt class="py-line"> </tt>
<a name="L910"></a><tt class="py-lineno"> 910</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">";"</tt> <tt class="py-keyword">in</tt> <tt id="link-613" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-613', 'accession', 'link-457');">accession</a></tt><tt class="py-op">:</tt> </tt>
<a name="L911"></a><tt class="py-lineno"> 911</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Cannot have semi-colon in EMBL accession, %s"</tt> </tt>
<a name="L912"></a><tt class="py-lineno"> 912</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-614" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-614', 'accession', 'link-457');">accession</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L913"></a><tt class="py-lineno"> 913</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">" "</tt> <tt class="py-keyword">in</tt> <tt id="link-615" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-615', 'accession', 'link-457');">accession</a></tt><tt class="py-op">:</tt> </tt>
<a name="L914"></a><tt class="py-lineno"> 914</tt>  <tt class="py-line">            <tt class="py-comment"># This is out of practicallity... might it be allowed?</tt> </tt>
<a name="L915"></a><tt class="py-lineno"> 915</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Cannot have spaces in EMBL accession, %s"</tt> </tt>
<a name="L916"></a><tt class="py-lineno"> 916</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-616" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-616', 'accession', 'link-457');">accession</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L917"></a><tt class="py-lineno"> 917</tt>  <tt class="py-line"> </tt>
<a name="L918"></a><tt class="py-lineno"> 918</tt>  <tt class="py-line">        <tt class="py-comment"># Get the molecule type</tt> </tt>
<a name="L919"></a><tt class="py-lineno"> 919</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - record this explicitly in the parser?</tt> </tt>
<a name="L920"></a><tt class="py-lineno"> 920</tt>  <tt class="py-line">        <tt class="py-comment"># Get the base alphabet (underneath any Gapped or StopCodon encoding)</tt> </tt>
<a name="L921"></a><tt class="py-lineno"> 921</tt>  <tt class="py-line">        <tt id="link-617" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-617', 'a', 'link-325');">a</a></tt> <tt class="py-op">=</tt> <tt id="link-618" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-618', 'Alphabet', 'link-12');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-619" class="py-name"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-619', '_get_base_alphabet', 'link-327');">_get_base_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-620" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-620', 'seq', 'link-328');">seq</a></tt><tt class="py-op">.</tt><tt id="link-621" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-621', 'alphabet', 'link-32');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L922"></a><tt class="py-lineno"> 922</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-622" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-622', 'a', 'link-325');">a</a></tt><tt class="py-op">,</tt> <tt id="link-623" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-623', 'Alphabet', 'link-12');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-624" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-624', 'Alphabet', 'link-12');">Alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L923"></a><tt class="py-lineno"> 923</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid alphabet"</tt><tt class="py-op">)</tt> </tt>
<a name="L924"></a><tt class="py-lineno"> 924</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-625" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-625', 'a', 'link-325');">a</a></tt><tt class="py-op">,</tt> <tt id="link-626" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-626', 'Alphabet', 'link-12');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-627" class="py-name"><a title="Bio.Alphabet.DNAAlphabet" class="py-name" href="#" onclick="return doclink('link-627', 'DNAAlphabet', 'link-344');">DNAAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L925"></a><tt class="py-lineno"> 925</tt>  <tt class="py-line">            <tt class="py-name">mol_type</tt> <tt class="py-op">=</tt> <tt class="py-string">"DNA"</tt> </tt>
<a name="L926"></a><tt class="py-lineno"> 926</tt>  <tt class="py-line">            <tt class="py-name">units</tt> <tt class="py-op">=</tt> <tt class="py-string">"BP"</tt> </tt>
<a name="L927"></a><tt class="py-lineno"> 927</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-628" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-628', 'a', 'link-325');">a</a></tt><tt class="py-op">,</tt> <tt id="link-629" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-629', 'Alphabet', 'link-12');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-630" class="py-name"><a title="Bio.Alphabet.RNAAlphabet" class="py-name" href="#" onclick="return doclink('link-630', 'RNAAlphabet', 'link-347');">RNAAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L928"></a><tt class="py-lineno"> 928</tt>  <tt class="py-line">            <tt class="py-name">mol_type</tt> <tt class="py-op">=</tt> <tt class="py-string">"RNA"</tt> </tt>
<a name="L929"></a><tt class="py-lineno"> 929</tt>  <tt class="py-line">            <tt class="py-name">units</tt> <tt class="py-op">=</tt> <tt class="py-string">"BP"</tt> </tt>
<a name="L930"></a><tt class="py-lineno"> 930</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-631" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-631', 'a', 'link-325');">a</a></tt><tt class="py-op">,</tt> <tt id="link-632" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-632', 'Alphabet', 'link-12');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-633" class="py-name"><a title="Bio.Alphabet.ProteinAlphabet" class="py-name" href="#" onclick="return doclink('link-633', 'ProteinAlphabet', 'link-335');">ProteinAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L931"></a><tt class="py-lineno"> 931</tt>  <tt class="py-line">            <tt class="py-name">mol_type</tt> <tt class="py-op">=</tt> <tt class="py-string">"PROTEIN"</tt> </tt>
<a name="L932"></a><tt class="py-lineno"> 932</tt>  <tt class="py-line">            <tt class="py-name">units</tt> <tt class="py-op">=</tt> <tt class="py-string">"AA"</tt> </tt>
<a name="L933"></a><tt class="py-lineno"> 933</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L934"></a><tt class="py-lineno"> 934</tt>  <tt class="py-line">            <tt class="py-comment"># Must be something like NucleotideAlphabet</tt> </tt>
<a name="L935"></a><tt class="py-lineno"> 935</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Need a DNA, RNA or Protein alphabet"</tt><tt class="py-op">)</tt> </tt>
<a name="L936"></a><tt class="py-lineno"> 936</tt>  <tt class="py-line"> </tt>
<a name="L937"></a><tt class="py-lineno"> 937</tt>  <tt class="py-line">        <tt class="py-comment"># Get the taxonomy division</tt> </tt>
<a name="L938"></a><tt class="py-lineno"> 938</tt>  <tt class="py-line">        <tt class="py-name">division</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-634" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._get_data_division
Bio.SeqIO.InsdcIO.GenBankWriter._get_data_division" class="py-name" href="#" onclick="return doclink('link-634', '_get_data_division', 'link-348');">_get_data_division</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L939"></a><tt class="py-lineno"> 939</tt>  <tt class="py-line"> </tt>
<a name="L940"></a><tt class="py-lineno"> 940</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Full ID line</tt> </tt>
<a name="L941"></a><tt class="py-lineno"> 941</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt> </tt>
<a name="L942"></a><tt class="py-lineno"> 942</tt>  <tt class="py-line">        <tt class="py-comment"># ID   &lt;1&gt;; SV &lt;2&gt;; &lt;3&gt;; &lt;4&gt;; &lt;5&gt;; &lt;6&gt;; &lt;7&gt; BP.</tt> </tt>
<a name="L943"></a><tt class="py-lineno"> 943</tt>  <tt class="py-line">        <tt class="py-comment"># 1. Primary accession number</tt> </tt>
<a name="L944"></a><tt class="py-lineno"> 944</tt>  <tt class="py-line">        <tt class="py-comment"># 2. Sequence version number</tt> </tt>
<a name="L945"></a><tt class="py-lineno"> 945</tt>  <tt class="py-line">        <tt class="py-comment"># 3. Topology: 'circular' or 'linear'</tt> </tt>
<a name="L946"></a><tt class="py-lineno"> 946</tt>  <tt class="py-line">        <tt class="py-comment"># 4. Molecule type</tt> </tt>
<a name="L947"></a><tt class="py-lineno"> 947</tt>  <tt class="py-line">        <tt class="py-comment"># 5. Data class</tt> </tt>
<a name="L948"></a><tt class="py-lineno"> 948</tt>  <tt class="py-line">        <tt class="py-comment"># 6. Taxonomic division</tt> </tt>
<a name="L949"></a><tt class="py-lineno"> 949</tt>  <tt class="py-line">        <tt class="py-comment"># 7. Sequence length</tt> </tt>
<a name="L950"></a><tt class="py-lineno"> 950</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-635" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_single_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_single_line" class="py-name" href="#" onclick="return doclink('link-635', '_write_single_line', 'link-289');">_write_single_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"ID"</tt><tt class="py-op">,</tt> <tt class="py-string">"%s; %s; ; %s; ; %s; %i %s."</tt> </tt>
<a name="L951"></a><tt class="py-lineno"> 951</tt>  <tt class="py-line">                                <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-636" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-636', 'accession', 'link-457');">accession</a></tt><tt class="py-op">,</tt> <tt id="link-637" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-637', 'version', 'link-602');">version</a></tt><tt class="py-op">,</tt> <tt class="py-name">mol_type</tt><tt class="py-op">,</tt> </tt>
<a name="L952"></a><tt class="py-lineno"> 952</tt>  <tt class="py-line">                                   <tt class="py-name">division</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">units</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L953"></a><tt class="py-lineno"> 953</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-638" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-638', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"XX\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L954"></a><tt class="py-lineno"> 954</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-639" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_single_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_single_line" class="py-name" href="#" onclick="return doclink('link-639', '_write_single_line', 'link-289');">_write_single_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"AC"</tt><tt class="py-op">,</tt> <tt id="link-640" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-640', 'accession', 'link-457');">accession</a></tt> <tt class="py-op">+</tt> <tt class="py-string">";"</tt><tt class="py-op">)</tt> </tt>
<a name="L955"></a><tt class="py-lineno"> 955</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-641" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-641', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"XX\n"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L956"></a><tt class="py-lineno"> 956</tt>  <tt class="py-line"> </tt>
<a name="EmblWriter._get_data_division"></a><div id="EmblWriter._get_data_division-def"><a name="L957"></a><tt class="py-lineno"> 957</tt> <a class="py-toggle" href="#" id="EmblWriter._get_data_division-toggle" onclick="return toggle('EmblWriter._get_data_division');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO.EmblWriter-class.html#_get_data_division">_get_data_division</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EmblWriter._get_data_division-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EmblWriter._get_data_division-expanded"><a name="L958"></a><tt class="py-lineno"> 958</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L959"></a><tt class="py-lineno"> 959</tt>  <tt class="py-line">            <tt class="py-name">division</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-642" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-642', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"data_file_division"</tt><tt class="py-op">]</tt> </tt>
<a name="L960"></a><tt class="py-lineno"> 960</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L961"></a><tt class="py-lineno"> 961</tt>  <tt class="py-line">            <tt class="py-name">division</tt> <tt class="py-op">=</tt> <tt class="py-string">"UNC"</tt> </tt>
<a name="L962"></a><tt class="py-lineno"> 962</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">division</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"PHG"</tt><tt class="py-op">,</tt> <tt class="py-string">"ENV"</tt><tt class="py-op">,</tt> <tt class="py-string">"FUN"</tt><tt class="py-op">,</tt> <tt class="py-string">"HUM"</tt><tt class="py-op">,</tt> <tt class="py-string">"INV"</tt><tt class="py-op">,</tt> <tt class="py-string">"MAM"</tt><tt class="py-op">,</tt> <tt class="py-string">"VRT"</tt><tt class="py-op">,</tt> </tt>
<a name="L963"></a><tt class="py-lineno"> 963</tt>  <tt class="py-line">                        <tt class="py-string">"MUS"</tt><tt class="py-op">,</tt> <tt class="py-string">"PLN"</tt><tt class="py-op">,</tt> <tt class="py-string">"PRO"</tt><tt class="py-op">,</tt> <tt class="py-string">"ROD"</tt><tt class="py-op">,</tt> <tt class="py-string">"SYN"</tt><tt class="py-op">,</tt> <tt class="py-string">"TGN"</tt><tt class="py-op">,</tt> <tt class="py-string">"UNC"</tt><tt class="py-op">,</tt> </tt>
<a name="L964"></a><tt class="py-lineno"> 964</tt>  <tt class="py-line">                        <tt class="py-string">"VRL"</tt><tt class="py-op">,</tt> <tt class="py-string">"XXX"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L965"></a><tt class="py-lineno"> 965</tt>  <tt class="py-line">            <tt class="py-comment"># Good, already EMBL style</tt> </tt>
<a name="L966"></a><tt class="py-lineno"> 966</tt>  <tt class="py-line">            <tt class="py-comment">#    Division                 Code</tt> </tt>
<a name="L967"></a><tt class="py-lineno"> 967</tt>  <tt class="py-line">            <tt class="py-comment">#    -----------------        ----</tt> </tt>
<a name="L968"></a><tt class="py-lineno"> 968</tt>  <tt class="py-line">            <tt class="py-comment">#    Bacteriophage            PHG</tt> </tt>
<a name="L969"></a><tt class="py-lineno"> 969</tt>  <tt class="py-line">            <tt class="py-comment">#    Environmental Sample     ENV</tt> </tt>
<a name="L970"></a><tt class="py-lineno"> 970</tt>  <tt class="py-line">            <tt class="py-comment">#    Fungal                   FUN</tt> </tt>
<a name="L971"></a><tt class="py-lineno"> 971</tt>  <tt class="py-line">            <tt class="py-comment">#    Human                    HUM</tt> </tt>
<a name="L972"></a><tt class="py-lineno"> 972</tt>  <tt class="py-line">            <tt class="py-comment">#    Invertebrate             INV</tt> </tt>
<a name="L973"></a><tt class="py-lineno"> 973</tt>  <tt class="py-line">            <tt class="py-comment">#    Other Mammal             MAM</tt> </tt>
<a name="L974"></a><tt class="py-lineno"> 974</tt>  <tt class="py-line">            <tt class="py-comment">#    Other Vertebrate         VRT</tt> </tt>
<a name="L975"></a><tt class="py-lineno"> 975</tt>  <tt class="py-line">            <tt class="py-comment">#    Mus musculus             MUS</tt> </tt>
<a name="L976"></a><tt class="py-lineno"> 976</tt>  <tt class="py-line">            <tt class="py-comment">#    Plant                    PLN</tt> </tt>
<a name="L977"></a><tt class="py-lineno"> 977</tt>  <tt class="py-line">            <tt class="py-comment">#    Prokaryote               PRO</tt> </tt>
<a name="L978"></a><tt class="py-lineno"> 978</tt>  <tt class="py-line">            <tt class="py-comment">#    Other Rodent             ROD</tt> </tt>
<a name="L979"></a><tt class="py-lineno"> 979</tt>  <tt class="py-line">            <tt class="py-comment">#    Synthetic                SYN</tt> </tt>
<a name="L980"></a><tt class="py-lineno"> 980</tt>  <tt class="py-line">            <tt class="py-comment">#    Transgenic               TGN</tt> </tt>
<a name="L981"></a><tt class="py-lineno"> 981</tt>  <tt class="py-line">            <tt class="py-comment">#    Unclassified             UNC (i.e. unknown)</tt> </tt>
<a name="L982"></a><tt class="py-lineno"> 982</tt>  <tt class="py-line">            <tt class="py-comment">#    Viral                    VRL</tt> </tt>
<a name="L983"></a><tt class="py-lineno"> 983</tt>  <tt class="py-line">            <tt class="py-comment">#</tt> </tt>
<a name="L984"></a><tt class="py-lineno"> 984</tt>  <tt class="py-line">            <tt class="py-comment"># (plus XXX used for submiting data to EMBL)</tt> </tt>
<a name="L985"></a><tt class="py-lineno"> 985</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
<a name="L986"></a><tt class="py-lineno"> 986</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L987"></a><tt class="py-lineno"> 987</tt>  <tt class="py-line">            <tt class="py-comment"># See if this is in GenBank style &amp; can be converted.</tt> </tt>
<a name="L988"></a><tt class="py-lineno"> 988</tt>  <tt class="py-line">            <tt class="py-comment"># Generally a problem as the GenBank groups are wider</tt> </tt>
<a name="L989"></a><tt class="py-lineno"> 989</tt>  <tt class="py-line">            <tt class="py-comment"># than those of EMBL. Note that GenBank use "BCT" for</tt> </tt>
<a name="L990"></a><tt class="py-lineno"> 990</tt>  <tt class="py-line">            <tt class="py-comment"># both bacteria and acherea thus this maps to EMBL's</tt> </tt>
<a name="L991"></a><tt class="py-lineno"> 991</tt>  <tt class="py-line">            <tt class="py-comment"># "PRO" nicely.</tt> </tt>
<a name="L992"></a><tt class="py-lineno"> 992</tt>  <tt class="py-line">            <tt class="py-name">gbk_to_embl</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">"BCT"</tt><tt class="py-op">:</tt> <tt class="py-string">"PRO"</tt><tt class="py-op">,</tt> </tt>
<a name="L993"></a><tt class="py-lineno"> 993</tt>  <tt class="py-line">                           <tt class="py-string">"UNK"</tt><tt class="py-op">:</tt> <tt class="py-string">"UNC"</tt><tt class="py-op">,</tt> </tt>
<a name="L994"></a><tt class="py-lineno"> 994</tt>  <tt class="py-line">                           <tt class="py-op">}</tt> </tt>
<a name="L995"></a><tt class="py-lineno"> 995</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L996"></a><tt class="py-lineno"> 996</tt>  <tt class="py-line">                <tt class="py-name">division</tt> <tt class="py-op">=</tt> <tt class="py-name">gbk_to_embl</tt><tt class="py-op">[</tt><tt class="py-name">division</tt><tt class="py-op">]</tt> </tt>
<a name="L997"></a><tt class="py-lineno"> 997</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L998"></a><tt class="py-lineno"> 998</tt>  <tt class="py-line">                <tt class="py-name">division</tt> <tt class="py-op">=</tt> <tt class="py-string">"UNC"</tt> </tt>
<a name="L999"></a><tt class="py-lineno"> 999</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">division</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">3</tt> </tt>
<a name="L1000"></a><tt class="py-lineno">1000</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">division</tt> </tt>
</div><a name="L1001"></a><tt class="py-lineno">1001</tt>  <tt class="py-line"> </tt>
<a name="EmblWriter._write_keywords"></a><div id="EmblWriter._write_keywords-def"><a name="L1002"></a><tt class="py-lineno">1002</tt> <a class="py-toggle" href="#" id="EmblWriter._write_keywords-toggle" onclick="return toggle('EmblWriter._write_keywords');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO.EmblWriter-class.html#_write_keywords">_write_keywords</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EmblWriter._write_keywords-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EmblWriter._write_keywords-expanded"><a name="L1003"></a><tt class="py-lineno">1003</tt>  <tt class="py-line">        <tt class="py-comment"># Put the keywords right after DE line.</tt> </tt>
<a name="L1004"></a><tt class="py-lineno">1004</tt>  <tt class="py-line">        <tt class="py-comment"># Each 'keyword' can have multiple words and spaces, but we</tt> </tt>
<a name="L1005"></a><tt class="py-lineno">1005</tt>  <tt class="py-line">        <tt class="py-comment"># must not split any 'keyword' between lines.</tt> </tt>
<a name="L1006"></a><tt class="py-lineno">1006</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Combine short keywords onto one line</tt> </tt>
<a name="L1007"></a><tt class="py-lineno">1007</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">keyword</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-643" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-643', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"keywords"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L1008"></a><tt class="py-lineno">1008</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-644" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_single_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_single_line" class="py-name" href="#" onclick="return doclink('link-644', '_write_single_line', 'link-289');">_write_single_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"KW"</tt><tt class="py-op">,</tt> <tt class="py-name">keyword</tt><tt class="py-op">)</tt> </tt>
<a name="L1009"></a><tt class="py-lineno">1009</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-645" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-645', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"XX\n"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1010"></a><tt class="py-lineno">1010</tt>  <tt class="py-line"> </tt>
<a name="EmblWriter._write_references"></a><div id="EmblWriter._write_references-def"><a name="L1011"></a><tt class="py-lineno">1011</tt> <a class="py-toggle" href="#" id="EmblWriter._write_references-toggle" onclick="return toggle('EmblWriter._write_references');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO.EmblWriter-class.html#_write_references">_write_references</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EmblWriter._write_references-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EmblWriter._write_references-expanded"><a name="L1012"></a><tt class="py-lineno">1012</tt>  <tt class="py-line">        <tt class="py-comment"># The order should be RN, RC, RP, RX, RG, RA, RT, RL</tt> </tt>
<a name="L1013"></a><tt class="py-lineno">1013</tt>  <tt class="py-line">        <tt id="link-646" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.number" class="py-name" href="#" onclick="return doclink('link-646', 'number', 'link-359');">number</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L1014"></a><tt class="py-lineno">1014</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-647" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-647', 'ref', 'link-63');">ref</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-648" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-648', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"references"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L1015"></a><tt class="py-lineno">1015</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-649" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-649', 'ref', 'link-63');">ref</a></tt><tt class="py-op">,</tt> <tt id="link-650" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-650', 'SeqFeature', 'link-16');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-651" class="py-name"><a title="Bio.ExPASy.Prodoc.Reference
Bio.GenBank.Record.Reference
Bio.Phylo.PhyloXML.Reference
Bio.SeqFeature.Reference
Bio.SwissProt.Reference" class="py-name" href="#" onclick="return doclink('link-651', 'Reference', 'link-364');">Reference</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1016"></a><tt class="py-lineno">1016</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L1017"></a><tt class="py-lineno">1017</tt>  <tt class="py-line">            <tt id="link-652" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.number" class="py-name" href="#" onclick="return doclink('link-652', 'number', 'link-359');">number</a></tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L1018"></a><tt class="py-lineno">1018</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-653" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_single_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_single_line" class="py-name" href="#" onclick="return doclink('link-653', '_write_single_line', 'link-289');">_write_single_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"RN"</tt><tt class="py-op">,</tt> <tt class="py-string">"[%i]"</tt> <tt class="py-op">%</tt> <tt id="link-654" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.number" class="py-name" href="#" onclick="return doclink('link-654', 'number', 'link-359');">number</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1019"></a><tt class="py-lineno">1019</tt>  <tt class="py-line">            <tt class="py-comment"># TODO - support for RC line (needed in parser too)</tt> </tt>
<a name="L1020"></a><tt class="py-lineno">1020</tt>  <tt class="py-line">            <tt class="py-comment"># TODO - support more complex record reference locations?</tt> </tt>
<a name="L1021"></a><tt class="py-lineno">1021</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-655" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-655', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-656" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-656', 'location', 'link-62');">location</a></tt> <tt class="py-keyword">and</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-657" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-657', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-658" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-658', 'location', 'link-62');">location</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L1022"></a><tt class="py-lineno">1022</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-659" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_single_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_single_line" class="py-name" href="#" onclick="return doclink('link-659', '_write_single_line', 'link-289');">_write_single_line</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1023"></a><tt class="py-lineno">1023</tt>  <tt class="py-line">                    <tt class="py-string">"RP"</tt><tt class="py-op">,</tt> <tt class="py-string">"%i-%i"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-660" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-660', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-661" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-661', 'location', 'link-62');">location</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-662" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.nofuzzy_start
Bio.SeqFeature.FeatureLocation.nofuzzy_start" class="py-name" href="#" onclick="return doclink('link-662', 'nofuzzy_start', 'link-144');">nofuzzy_start</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> </tt>
<a name="L1024"></a><tt class="py-lineno">1024</tt>  <tt class="py-line">                                     <tt id="link-663" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-663', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-664" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-664', 'location', 'link-62');">location</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-665" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.nofuzzy_end
Bio.SeqFeature.FeatureLocation.nofuzzy_end" class="py-name" href="#" onclick="return doclink('link-665', 'nofuzzy_end', 'link-135');">nofuzzy_end</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1025"></a><tt class="py-lineno">1025</tt>  <tt class="py-line">            <tt class="py-comment"># TODO - record any DOI or AGRICOLA identifier in the reference object?</tt> </tt>
<a name="L1026"></a><tt class="py-lineno">1026</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-666" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-666', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-667" class="py-name"><a title="Bio.GenBank._FeatureConsumer.pubmed_id
Bio.GenBank._RecordConsumer.pubmed_id" class="py-name" href="#" onclick="return doclink('link-667', 'pubmed_id', 'link-415');">pubmed_id</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1027"></a><tt class="py-lineno">1027</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-668" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_single_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_single_line" class="py-name" href="#" onclick="return doclink('link-668', '_write_single_line', 'link-289');">_write_single_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"RX"</tt><tt class="py-op">,</tt> <tt class="py-string">"PUBMED; %s."</tt> <tt class="py-op">%</tt> <tt id="link-669" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-669', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-670" class="py-name"><a title="Bio.GenBank._FeatureConsumer.pubmed_id
Bio.GenBank._RecordConsumer.pubmed_id" class="py-name" href="#" onclick="return doclink('link-670', 'pubmed_id', 'link-415');">pubmed_id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1028"></a><tt class="py-lineno">1028</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-671" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-671', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-672" class="py-name"><a title="Bio.GenBank._FeatureConsumer.consrtm
Bio.GenBank._RecordConsumer.consrtm" class="py-name" href="#" onclick="return doclink('link-672', 'consrtm', 'link-395');">consrtm</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1029"></a><tt class="py-lineno">1029</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-673" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_single_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_single_line" class="py-name" href="#" onclick="return doclink('link-673', '_write_single_line', 'link-289');">_write_single_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"RG"</tt><tt class="py-op">,</tt> <tt class="py-string">"%s"</tt> <tt class="py-op">%</tt> <tt id="link-674" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-674', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-675" class="py-name"><a title="Bio.GenBank._FeatureConsumer.consrtm
Bio.GenBank._RecordConsumer.consrtm" class="py-name" href="#" onclick="return doclink('link-675', 'consrtm', 'link-395');">consrtm</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1030"></a><tt class="py-lineno">1030</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-676" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-676', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-677" class="py-name"><a title="Bio.GenBank._FeatureConsumer.authors
Bio.GenBank._RecordConsumer.authors" class="py-name" href="#" onclick="return doclink('link-677', 'authors', 'link-390');">authors</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1031"></a><tt class="py-lineno">1031</tt>  <tt class="py-line">                <tt class="py-comment"># We store the AUTHORS data as a single string</tt> </tt>
<a name="L1032"></a><tt class="py-lineno">1032</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-678" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_multi_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_multi_line" class="py-name" href="#" onclick="return doclink('link-678', '_write_multi_line', 'link-391');">_write_multi_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"RA"</tt><tt class="py-op">,</tt> <tt id="link-679" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-679', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-680" class="py-name"><a title="Bio.GenBank._FeatureConsumer.authors
Bio.GenBank._RecordConsumer.authors" class="py-name" href="#" onclick="return doclink('link-680', 'authors', 'link-390');">authors</a></tt> <tt class="py-op">+</tt> <tt class="py-string">";"</tt><tt class="py-op">)</tt> </tt>
<a name="L1033"></a><tt class="py-lineno">1033</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-681" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-681', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-682" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-682', 'title', 'link-400');">title</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1034"></a><tt class="py-lineno">1034</tt>  <tt class="py-line">                <tt class="py-comment"># We store the title as a single string</tt> </tt>
<a name="L1035"></a><tt class="py-lineno">1035</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-683" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_multi_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_multi_line" class="py-name" href="#" onclick="return doclink('link-683', '_write_multi_line', 'link-391');">_write_multi_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"RT"</tt><tt class="py-op">,</tt> <tt class="py-string">'"%s";'</tt> <tt class="py-op">%</tt> <tt id="link-684" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-684', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-685" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-685', 'title', 'link-400');">title</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1036"></a><tt class="py-lineno">1036</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-686" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-686', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-687" class="py-name"><a title="Bio.GenBank._FeatureConsumer.journal
Bio.GenBank._RecordConsumer.journal" class="py-name" href="#" onclick="return doclink('link-687', 'journal', 'link-405');">journal</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1037"></a><tt class="py-lineno">1037</tt>  <tt class="py-line">                <tt class="py-comment"># We store this as a single string - holds the journal name,</tt> </tt>
<a name="L1038"></a><tt class="py-lineno">1038</tt>  <tt class="py-line">                <tt class="py-comment"># volume, year, and page numbers of the citation</tt> </tt>
<a name="L1039"></a><tt class="py-lineno">1039</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-688" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_multi_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_multi_line" class="py-name" href="#" onclick="return doclink('link-688', '_write_multi_line', 'link-391');">_write_multi_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"RL"</tt><tt class="py-op">,</tt> <tt id="link-689" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-689', 'ref', 'link-63');">ref</a></tt><tt class="py-op">.</tt><tt id="link-690" class="py-name"><a title="Bio.GenBank._FeatureConsumer.journal
Bio.GenBank._RecordConsumer.journal" class="py-name" href="#" onclick="return doclink('link-690', 'journal', 'link-405');">journal</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1040"></a><tt class="py-lineno">1040</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-691" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-691', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"XX\n"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1041"></a><tt class="py-lineno">1041</tt>  <tt class="py-line"> </tt>
<a name="EmblWriter._write_comment"></a><div id="EmblWriter._write_comment-def"><a name="L1042"></a><tt class="py-lineno">1042</tt> <a class="py-toggle" href="#" id="EmblWriter._write_comment-toggle" onclick="return toggle('EmblWriter._write_comment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO.EmblWriter-class.html#_write_comment">_write_comment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EmblWriter._write_comment-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EmblWriter._write_comment-expanded"><a name="L1043"></a><tt class="py-lineno">1043</tt>  <tt class="py-line">        <tt class="py-comment"># This is a bit complicated due to the range of possible</tt> </tt>
<a name="L1044"></a><tt class="py-lineno">1044</tt>  <tt class="py-line">        <tt class="py-comment"># ways people might have done their annotation...</tt> </tt>
<a name="L1045"></a><tt class="py-lineno">1045</tt>  <tt class="py-line">        <tt class="py-comment"># Currently the parser uses a single string with newlines.</tt> </tt>
<a name="L1046"></a><tt class="py-lineno">1046</tt>  <tt class="py-line">        <tt class="py-comment"># A list of lines is also reasonable.</tt> </tt>
<a name="L1047"></a><tt class="py-lineno">1047</tt>  <tt class="py-line">        <tt class="py-comment"># A single (long) string is perhaps the most natural of all.</tt> </tt>
<a name="L1048"></a><tt class="py-lineno">1048</tt>  <tt class="py-line">        <tt class="py-comment"># This means we may need to deal with line wrapping.</tt> </tt>
<a name="L1049"></a><tt class="py-lineno">1049</tt>  <tt class="py-line">        <tt id="link-692" class="py-name"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-692', 'comment', 'link-420');">comment</a></tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-693" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-693', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"comment"</tt><tt class="py-op">]</tt> </tt>
<a name="L1050"></a><tt class="py-lineno">1050</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-694" class="py-name"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-694', 'comment', 'link-420');">comment</a></tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1051"></a><tt class="py-lineno">1051</tt>  <tt class="py-line">            <tt class="py-name">lines</tt> <tt class="py-op">=</tt> <tt id="link-695" class="py-name"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-695', 'comment', 'link-420');">comment</a></tt><tt class="py-op">.</tt><tt id="link-696" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-696', 'split', 'link-263');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L1052"></a><tt class="py-lineno">1052</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-697" class="py-name"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-697', 'comment', 'link-420');">comment</a></tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-698" class="py-name"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-698', 'comment', 'link-420');">comment</a></tt><tt class="py-op">,</tt> <tt class="py-name">tuple</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1053"></a><tt class="py-lineno">1053</tt>  <tt class="py-line">            <tt class="py-name">lines</tt> <tt class="py-op">=</tt> <tt id="link-699" class="py-name"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-699', 'comment', 'link-420');">comment</a></tt> </tt>
<a name="L1054"></a><tt class="py-lineno">1054</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1055"></a><tt class="py-lineno">1055</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Could not understand comment annotation"</tt><tt class="py-op">)</tt> </tt>
<a name="L1056"></a><tt class="py-lineno">1056</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Merge this with the GenBank comment code?</tt> </tt>
<a name="L1057"></a><tt class="py-lineno">1057</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">lines</tt><tt class="py-op">:</tt> </tt>
<a name="L1058"></a><tt class="py-lineno">1058</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L1059"></a><tt class="py-lineno">1059</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">lines</tt><tt class="py-op">:</tt> </tt>
<a name="L1060"></a><tt class="py-lineno">1060</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-700" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_multi_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_multi_line" class="py-name" href="#" onclick="return doclink('link-700', '_write_multi_line', 'link-391');">_write_multi_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"CC"</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1061"></a><tt class="py-lineno">1061</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-701" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-701', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"XX\n"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1062"></a><tt class="py-lineno">1062</tt>  <tt class="py-line"> </tt>
<a name="EmblWriter.write_record"></a><div id="EmblWriter.write_record-def"><a name="L1063"></a><tt class="py-lineno">1063</tt> <a class="py-toggle" href="#" id="EmblWriter.write_record-toggle" onclick="return toggle('EmblWriter.write_record');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO.EmblWriter-class.html#write_record">write_record</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EmblWriter.write_record-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EmblWriter.write_record-expanded"><a name="L1064"></a><tt class="py-lineno">1064</tt>  <tt class="py-line">        <tt class="py-docstring">"""Write a single record to the output file."""</tt> </tt>
<a name="L1065"></a><tt class="py-lineno">1065</tt>  <tt class="py-line"> </tt>
<a name="L1066"></a><tt class="py-lineno">1066</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt> </tt>
<a name="L1067"></a><tt class="py-lineno">1067</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-702" class="py-name" targets="Method Bio.SeqIO.InsdcIO.EmblWriter._write_the_first_lines()=Bio.SeqIO.InsdcIO.EmblWriter-class.html#_write_the_first_lines"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_the_first_lines" class="py-name" href="#" onclick="return doclink('link-702', '_write_the_first_lines', 'link-702');">_write_the_first_lines</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L1068"></a><tt class="py-lineno">1068</tt>  <tt class="py-line"> </tt>
<a name="L1069"></a><tt class="py-lineno">1069</tt>  <tt class="py-line">        <tt class="py-comment"># PR line (0 or 1 lines only), project identifier</tt> </tt>
<a name="L1070"></a><tt class="py-lineno">1070</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">xref</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-703" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.dbxrefs" class="py-name" href="#" onclick="return doclink('link-703', 'dbxrefs', 'link-479');">dbxrefs</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1071"></a><tt class="py-lineno">1071</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">xref</tt><tt class="py-op">.</tt><tt id="link-704" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-704', 'startswith', 'link-463');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"Project:"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1072"></a><tt class="py-lineno">1072</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-705" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_single_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_single_line" class="py-name" href="#" onclick="return doclink('link-705', '_write_single_line', 'link-289');">_write_single_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"PR"</tt><tt class="py-op">,</tt> <tt class="py-name">xref</tt> <tt class="py-op">+</tt> <tt class="py-string">";"</tt><tt class="py-op">)</tt> </tt>
<a name="L1073"></a><tt class="py-lineno">1073</tt>  <tt class="py-line">                <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-706" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-706', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"XX\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L1074"></a><tt class="py-lineno">1074</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L1075"></a><tt class="py-lineno">1075</tt>  <tt class="py-line"> </tt>
<a name="L1076"></a><tt class="py-lineno">1076</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - DT lines (date)</tt> </tt>
<a name="L1077"></a><tt class="py-lineno">1077</tt>  <tt class="py-line"> </tt>
<a name="L1078"></a><tt class="py-lineno">1078</tt>  <tt class="py-line">        <tt class="py-name">descr</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-707" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-707', 'description', 'link-470');">description</a></tt> </tt>
<a name="L1079"></a><tt class="py-lineno">1079</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">descr</tt> <tt class="py-op">==</tt> <tt class="py-string">"&lt;unknown description&gt;"</tt><tt class="py-op">:</tt> </tt>
<a name="L1080"></a><tt class="py-lineno">1080</tt>  <tt class="py-line">            <tt class="py-name">descr</tt> <tt class="py-op">=</tt> <tt class="py-string">"."</tt> </tt>
<a name="L1081"></a><tt class="py-lineno">1081</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-708" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_multi_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_multi_line" class="py-name" href="#" onclick="return doclink('link-708', '_write_multi_line', 'link-391');">_write_multi_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"DE"</tt><tt class="py-op">,</tt> <tt class="py-name">descr</tt><tt class="py-op">)</tt> </tt>
<a name="L1082"></a><tt class="py-lineno">1082</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-709" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-709', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"XX\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L1083"></a><tt class="py-lineno">1083</tt>  <tt class="py-line"> </tt>
<a name="L1084"></a><tt class="py-lineno">1084</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"keywords"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-710" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-710', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1085"></a><tt class="py-lineno">1085</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-711" class="py-name" targets="Method Bio.SeqIO.InsdcIO.EmblWriter._write_keywords()=Bio.SeqIO.InsdcIO.EmblWriter-class.html#_write_keywords"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_keywords" class="py-name" href="#" onclick="return doclink('link-711', '_write_keywords', 'link-711');">_write_keywords</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L1086"></a><tt class="py-lineno">1086</tt>  <tt class="py-line"> </tt>
<a name="L1087"></a><tt class="py-lineno">1087</tt>  <tt class="py-line">        <tt class="py-comment"># Should this be "source" or "organism"?</tt> </tt>
<a name="L1088"></a><tt class="py-lineno">1088</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-712" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_multi_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_multi_line" class="py-name" href="#" onclick="return doclink('link-712', '_write_multi_line', 'link-391');">_write_multi_line</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1089"></a><tt class="py-lineno">1089</tt>  <tt class="py-line">            <tt class="py-string">"OS"</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-713" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter._get_annotation_str" class="py-name" href="#" onclick="return doclink('link-713', '_get_annotation_str', 'link-322');">_get_annotation_str</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">,</tt> <tt class="py-string">"organism"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1090"></a><tt class="py-lineno">1090</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1091"></a><tt class="py-lineno">1091</tt>  <tt class="py-line">            <tt class="py-comment"># List of strings</tt> </tt>
<a name="L1092"></a><tt class="py-lineno">1092</tt>  <tt class="py-line">            <tt id="link-714" class="py-name"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-714', 'taxonomy', 'link-506');">taxonomy</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"; "</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-715" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-715', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"taxonomy"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">"."</tt> </tt>
<a name="L1093"></a><tt class="py-lineno">1093</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L1094"></a><tt class="py-lineno">1094</tt>  <tt class="py-line">            <tt id="link-716" class="py-name"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-716', 'taxonomy', 'link-506');">taxonomy</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"."</tt> </tt>
<a name="L1095"></a><tt class="py-lineno">1095</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-717" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_multi_line
Bio.SeqIO.InsdcIO.GenBankWriter._write_multi_line" class="py-name" href="#" onclick="return doclink('link-717', '_write_multi_line', 'link-391');">_write_multi_line</a></tt><tt class="py-op">(</tt><tt class="py-string">"OC"</tt><tt class="py-op">,</tt> <tt id="link-718" class="py-name"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-718', 'taxonomy', 'link-506');">taxonomy</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1096"></a><tt class="py-lineno">1096</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-719" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-719', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"XX\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L1097"></a><tt class="py-lineno">1097</tt>  <tt class="py-line"> </tt>
<a name="L1098"></a><tt class="py-lineno">1098</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"references"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-720" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-720', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1099"></a><tt class="py-lineno">1099</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-721" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_references
Bio.SeqIO.InsdcIO.GenBankWriter._write_references" class="py-name" href="#" onclick="return doclink('link-721', '_write_references', 'link-515');">_write_references</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L1100"></a><tt class="py-lineno">1100</tt>  <tt class="py-line"> </tt>
<a name="L1101"></a><tt class="py-lineno">1101</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"comment"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-722" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-722', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1102"></a><tt class="py-lineno">1102</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-723" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_comment
Bio.SeqIO.InsdcIO.GenBankWriter._write_comment" class="py-name" href="#" onclick="return doclink('link-723', '_write_comment', 'link-517');">_write_comment</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L1103"></a><tt class="py-lineno">1103</tt>  <tt class="py-line"> </tt>
<a name="L1104"></a><tt class="py-lineno">1104</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-724" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-724', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-725" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter.FEATURE_HEADER
Bio.SeqIO.InsdcIO.ImgtWriter.FEATURE_HEADER" class="py-name" href="#" onclick="return doclink('link-725', 'FEATURE_HEADER', 'link-528');">FEATURE_HEADER</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1105"></a><tt class="py-lineno">1105</tt>  <tt class="py-line">        <tt class="py-name">rec_length</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L1106"></a><tt class="py-lineno">1106</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">feature</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-726" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-726', 'features', 'link-519');">features</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1107"></a><tt class="py-lineno">1107</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-727" class="py-name"><a title="Bio.SeqIO.InsdcIO._InsdcWriter._write_feature" class="py-name" href="#" onclick="return doclink('link-727', '_write_feature', 'link-520');">_write_feature</a></tt><tt class="py-op">(</tt><tt class="py-name">feature</tt><tt class="py-op">,</tt> <tt class="py-name">rec_length</tt><tt class="py-op">)</tt> </tt>
<a name="L1108"></a><tt class="py-lineno">1108</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-728" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-728', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"XX\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L1109"></a><tt class="py-lineno">1109</tt>  <tt class="py-line"> </tt>
<a name="L1110"></a><tt class="py-lineno">1110</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-729" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter._write_sequence
Bio.SeqIO.InsdcIO.GenBankWriter._write_sequence" class="py-name" href="#" onclick="return doclink('link-729', '_write_sequence', 'link-521');">_write_sequence</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L1111"></a><tt class="py-lineno">1111</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-730" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-730', 'write', 'link-191');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"//\n"</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1112"></a><tt class="py-lineno">1112</tt>  <tt class="py-line"> </tt>
<a name="L1113"></a><tt class="py-lineno">1113</tt>  <tt class="py-line"> </tt>
<a name="ImgtWriter"></a><div id="ImgtWriter-def"><a name="L1114"></a><tt class="py-lineno">1114</tt> <a class="py-toggle" href="#" id="ImgtWriter-toggle" onclick="return toggle('ImgtWriter');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO.ImgtWriter-class.html">ImgtWriter</a><tt class="py-op">(</tt><tt class="py-base-class">EmblWriter</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ImgtWriter-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="ImgtWriter-expanded"><a name="L1115"></a><tt class="py-lineno">1115</tt>  <tt class="py-line">    <tt id="link-731" class="py-name"><a title="Bio.GenBank.Scanner.EmblScanner.HEADER_WIDTH
Bio.GenBank.Scanner.GenBankScanner.HEADER_WIDTH
Bio.GenBank.Scanner.InsdcScanner.HEADER_WIDTH
Bio.SeqIO.InsdcIO.EmblWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.GenBankWriter.HEADER_WIDTH
Bio.SeqIO.InsdcIO.ImgtWriter.HEADER_WIDTH" class="py-name" href="#" onclick="return doclink('link-731', 'HEADER_WIDTH', 'link-275');">HEADER_WIDTH</a></tt> <tt class="py-op">=</tt> <tt class="py-number">5</tt> </tt>
<a name="L1116"></a><tt class="py-lineno">1116</tt>  <tt class="py-line">    <tt id="link-732" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO.GenBankWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-732', 'QUALIFIER_INDENT', 'link-185');">QUALIFIER_INDENT</a></tt> <tt class="py-op">=</tt> <tt class="py-number">25</tt>  <tt class="py-comment"># Not 21 as in EMBL</tt> </tt>
<a name="L1117"></a><tt class="py-lineno">1117</tt>  <tt class="py-line">    <tt id="link-733" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT_STR
Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT_STR
Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT_STR" class="py-name" href="#" onclick="return doclink('link-733', 'QUALIFIER_INDENT_STR', 'link-186');">QUALIFIER_INDENT_STR</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"FT"</tt> <tt class="py-op">+</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt id="link-734" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO.GenBankWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT
Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-734', 'QUALIFIER_INDENT', 'link-185');">QUALIFIER_INDENT</a></tt> <tt class="py-op">-</tt> <tt class="py-number">2</tt><tt class="py-op">)</tt> </tt>
<a name="L1118"></a><tt class="py-lineno">1118</tt>  <tt class="py-line">    <tt id="link-735" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter.QUALIFIER_INDENT_TMP
Bio.SeqIO.InsdcIO.ImgtWriter.QUALIFIER_INDENT_TMP
Bio.SeqIO.InsdcIO._InsdcWriter.QUALIFIER_INDENT_TMP" class="py-name" href="#" onclick="return doclink('link-735', 'QUALIFIER_INDENT_TMP', 'link-188');">QUALIFIER_INDENT_TMP</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"FT   %s                    "</tt>  <tt class="py-comment"># 25 if %s is empty</tt> </tt>
<a name="L1119"></a><tt class="py-lineno">1119</tt>  <tt class="py-line">    <tt id="link-736" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblWriter.FEATURE_HEADER
Bio.SeqIO.InsdcIO.ImgtWriter.FEATURE_HEADER" class="py-name" href="#" onclick="return doclink('link-736', 'FEATURE_HEADER', 'link-528');">FEATURE_HEADER</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"FH   Key                 Location/Qualifiers\n"</tt> </tt>
</div><a name="L1120"></a><tt class="py-lineno">1120</tt>  <tt class="py-line"> </tt>
<a name="L1121"></a><tt class="py-lineno">1121</tt>  <tt class="py-line"><tt class="py-keyword">if</tt> <tt class="py-name">__name__</tt> <tt class="py-op">==</tt> <tt class="py-string">"__main__"</tt><tt class="py-op">:</tt> </tt>
<a name="L1122"></a><tt class="py-lineno">1122</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Quick self test"</tt><tt class="py-op">)</tt> </tt>
<a name="L1123"></a><tt class="py-lineno">1123</tt>  <tt class="py-line">    <tt class="py-keyword">import</tt> <tt class="py-name">os</tt> </tt>
<a name="L1124"></a><tt class="py-lineno">1124</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-737" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-737', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-738" class="py-name"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-738', '_py3k', 'link-18');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">StringIO</tt> </tt>
<a name="L1125"></a><tt class="py-lineno">1125</tt>  <tt class="py-line"> </tt>
<a name="compare_record"></a><div id="compare_record-def"><a name="L1126"></a><tt class="py-lineno">1126</tt> <a class="py-toggle" href="#" id="compare_record-toggle" onclick="return toggle('compare_record');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO-module.html#compare_record">compare_record</a><tt class="py-op">(</tt><tt class="py-param">old</tt><tt class="py-op">,</tt> <tt class="py-param">new</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="compare_record-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="compare_record-expanded"><a name="L1127"></a><tt class="py-lineno">1127</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-739" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-739', 'id', 'link-318');">id</a></tt> <tt class="py-op">!=</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-740" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-740', 'id', 'link-318');">id</a></tt> <tt class="py-keyword">and</tt> <tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-741" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-741', 'name', 'link-314');">name</a></tt> <tt class="py-op">!=</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-742" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-742', 'name', 'link-314');">name</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1128"></a><tt class="py-lineno">1128</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"'%s' or '%s' vs '%s' or '%s' records"</tt> </tt>
<a name="L1129"></a><tt class="py-lineno">1129</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-743" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-743', 'id', 'link-318');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-744" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-744', 'name', 'link-314');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-745" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-745', 'id', 'link-318');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-746" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-746', 'name', 'link-314');">name</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1130"></a><tt class="py-lineno">1130</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-747" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-747', 'seq', 'link-328');">seq</a></tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-748" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-748', 'seq', 'link-328');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1131"></a><tt class="py-lineno">1131</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"%i vs %i"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-749" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-749', 'seq', 'link-328');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-750" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-750', 'seq', 'link-328');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1132"></a><tt class="py-lineno">1132</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-751" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-751', 'seq', 'link-328');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-752" class="py-name" targets="Method Bio.Seq.Seq.upper()=Bio.Seq.Seq-class.html#upper,Method Bio.Seq.UnknownSeq.upper()=Bio.Seq.UnknownSeq-class.html#upper,Method Bio.SeqRecord.SeqRecord.upper()=Bio.SeqRecord.SeqRecord-class.html#upper"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-752', 'upper', 'link-752');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-753" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-753', 'seq', 'link-328');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-754" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-754', 'upper', 'link-752');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1133"></a><tt class="py-lineno">1133</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-755" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-755', 'seq', 'link-328');">seq</a></tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">200</tt><tt class="py-op">:</tt> </tt>
<a name="L1134"></a><tt class="py-lineno">1134</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"'%s' vs '%s'"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-756" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-756', 'seq', 'link-328');">seq</a></tt><tt class="py-op">,</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-757" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-757', 'seq', 'link-328');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1135"></a><tt class="py-lineno">1135</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1136"></a><tt class="py-lineno">1136</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt> </tt>
<a name="L1137"></a><tt class="py-lineno">1137</tt>  <tt class="py-line">                    <tt class="py-string">"'%s...' vs '%s...'"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-758" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-758', 'seq', 'link-328');">seq</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">100</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-759" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-759', 'seq', 'link-328');">seq</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">100</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1138"></a><tt class="py-lineno">1138</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-760" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-760', 'features', 'link-519');">features</a></tt> <tt class="py-keyword">and</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-761" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-761', 'features', 'link-519');">features</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1139"></a><tt class="py-lineno">1139</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">compare_features</tt><tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-762" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-762', 'features', 'link-519');">features</a></tt><tt class="py-op">,</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-763" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-763', 'features', 'link-519');">features</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1140"></a><tt class="py-lineno">1140</tt>  <tt class="py-line">        <tt class="py-comment"># Just insist on at least one word in common:</tt> </tt>
<a name="L1141"></a><tt class="py-lineno">1141</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-764" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-764', 'description', 'link-470');">description</a></tt> <tt class="py-keyword">or</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-765" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-765', 'description', 'link-470');">description</a></tt><tt class="py-op">)</tt> \ </tt>
<a name="L1142"></a><tt class="py-lineno">1142</tt>  <tt class="py-line">                <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt id="link-766" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.set()=Bio.Nexus.Nexus.StepMatrix-class.html#set,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.set()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#set,Method Bio.SVDSuperimposer.SVDSuperimposer.set()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#set"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-766', 'set', 'link-766');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-767" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-767', 'description', 'link-470');">description</a></tt><tt class="py-op">.</tt><tt id="link-768" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-768', 'split', 'link-263');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">intersection</tt><tt class="py-op">(</tt><tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-769" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-769', 'description', 'link-470');">description</a></tt><tt class="py-op">.</tt><tt id="link-770" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-770', 'split', 'link-263');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1143"></a><tt class="py-lineno">1143</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"%s versus %s"</tt> </tt>
<a name="L1144"></a><tt class="py-lineno">1144</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-771" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-771', 'description', 'link-470');">description</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-772" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-772', 'description', 'link-470');">description</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1145"></a><tt class="py-lineno">1145</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - check annotation</tt> </tt>
<a name="L1146"></a><tt class="py-lineno">1146</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"contig"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-773" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-773', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1147"></a><tt class="py-lineno">1147</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-774" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-774', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"contig"</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> \ </tt>
<a name="L1148"></a><tt class="py-lineno">1148</tt>  <tt class="py-line">                <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-775" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-775', 'annotations', 'link-261');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"contig"</tt><tt class="py-op">]</tt> </tt>
<a name="L1149"></a><tt class="py-lineno">1149</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">True</tt> </tt>
</div><a name="L1150"></a><tt class="py-lineno">1150</tt>  <tt class="py-line"> </tt>
<a name="compare_records"></a><div id="compare_records-def"><a name="L1151"></a><tt class="py-lineno">1151</tt> <a class="py-toggle" href="#" id="compare_records-toggle" onclick="return toggle('compare_records');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO-module.html#compare_records">compare_records</a><tt class="py-op">(</tt><tt class="py-param">old_list</tt><tt class="py-op">,</tt> <tt class="py-param">new_list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="compare_records-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="compare_records-expanded"><a name="L1152"></a><tt class="py-lineno">1152</tt>  <tt class="py-line">        <tt class="py-docstring">"""Check two lists of SeqRecords agree, raises a ValueError if mismatch."""</tt> </tt>
<a name="L1153"></a><tt class="py-lineno">1153</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">old_list</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">new_list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1154"></a><tt class="py-lineno">1154</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt> </tt>
<a name="L1155"></a><tt class="py-lineno">1155</tt>  <tt class="py-line">                <tt class="py-string">"%i vs %i records"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">old_list</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">new_list</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1156"></a><tt class="py-lineno">1156</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">old</tt><tt class="py-op">,</tt> <tt class="py-name">new</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">old_list</tt><tt class="py-op">,</tt> <tt class="py-name">new_list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1157"></a><tt class="py-lineno">1157</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">compare_record</tt><tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">,</tt> <tt class="py-name">new</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1158"></a><tt class="py-lineno">1158</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
<a name="L1159"></a><tt class="py-lineno">1159</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">True</tt> </tt>
</div><a name="L1160"></a><tt class="py-lineno">1160</tt>  <tt class="py-line"> </tt>
<a name="compare_feature"></a><div id="compare_feature-def"><a name="L1161"></a><tt class="py-lineno">1161</tt> <a class="py-toggle" href="#" id="compare_feature-toggle" onclick="return toggle('compare_feature');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO-module.html#compare_feature">compare_feature</a><tt class="py-op">(</tt><tt class="py-param">old</tt><tt class="py-op">,</tt> <tt class="py-param">new</tt><tt class="py-op">,</tt> <tt class="py-param">ignore_sub_features</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="compare_feature-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="compare_feature-expanded"><a name="L1162"></a><tt class="py-lineno">1162</tt>  <tt class="py-line">        <tt class="py-docstring">"""Check two SeqFeatures agree."""</tt> </tt>
<a name="L1163"></a><tt class="py-lineno">1163</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-776" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-776', 'type', 'link-241');">type</a></tt> <tt class="py-op">!=</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-777" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-777', 'type', 'link-241');">type</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1164"></a><tt class="py-lineno">1164</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Type %s versus %s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-778" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-778', 'type', 'link-241');">type</a></tt><tt class="py-op">,</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-779" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-779', 'type', 'link-241');">type</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1165"></a><tt class="py-lineno">1165</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-780" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-780', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-781" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.nofuzzy_start
Bio.SeqFeature.FeatureLocation.nofuzzy_start" class="py-name" href="#" onclick="return doclink('link-781', 'nofuzzy_start', 'link-144');">nofuzzy_start</a></tt> <tt class="py-op">!=</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-782" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-782', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-783" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.nofuzzy_start
Bio.SeqFeature.FeatureLocation.nofuzzy_start" class="py-name" href="#" onclick="return doclink('link-783', 'nofuzzy_start', 'link-144');">nofuzzy_start</a></tt> \ </tt>
<a name="L1166"></a><tt class="py-lineno">1166</tt>  <tt class="py-line">                <tt class="py-keyword">or</tt> <tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-784" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-784', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-785" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.nofuzzy_end
Bio.SeqFeature.FeatureLocation.nofuzzy_end" class="py-name" href="#" onclick="return doclink('link-785', 'nofuzzy_end', 'link-135');">nofuzzy_end</a></tt> <tt class="py-op">!=</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-786" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-786', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-787" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.nofuzzy_end
Bio.SeqFeature.FeatureLocation.nofuzzy_end" class="py-name" href="#" onclick="return doclink('link-787', 'nofuzzy_end', 'link-135');">nofuzzy_end</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1167"></a><tt class="py-lineno">1167</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"%s versus %s:\n%s\nvs:\n%s"</tt> </tt>
<a name="L1168"></a><tt class="py-lineno">1168</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-788" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-788', 'location', 'link-62');">location</a></tt><tt class="py-op">,</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-789" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-789', 'location', 'link-62');">location</a></tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">new</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1169"></a><tt class="py-lineno">1169</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-790" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-790', 'strand', 'link-156');">strand</a></tt> <tt class="py-op">!=</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-791" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-791', 'strand', 'link-156');">strand</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1170"></a><tt class="py-lineno">1170</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt> </tt>
<a name="L1171"></a><tt class="py-lineno">1171</tt>  <tt class="py-line">                <tt class="py-string">"Different strand:\n%s\nvs:\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">new</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1172"></a><tt class="py-lineno">1172</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-792" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-792', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-793" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-793', 'start', 'link-71');">start</a></tt> <tt class="py-op">!=</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-794" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-794', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-795" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-795', 'start', 'link-71');">start</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1173"></a><tt class="py-lineno">1173</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Start %s versus %s:\n%s\nvs:\n%s"</tt> </tt>
<a name="L1174"></a><tt class="py-lineno">1174</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-796" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-796', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-797" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-797', 'start', 'link-71');">start</a></tt><tt class="py-op">,</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-798" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-798', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-799" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-799', 'start', 'link-71');">start</a></tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">new</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1175"></a><tt class="py-lineno">1175</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-800" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-800', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-801" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-801', 'end', 'link-75');">end</a></tt> <tt class="py-op">!=</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-802" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-802', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-803" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-803', 'end', 'link-75');">end</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1176"></a><tt class="py-lineno">1176</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"End %s versus %s:\n%s\nvs:\n%s"</tt> </tt>
<a name="L1177"></a><tt class="py-lineno">1177</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-804" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-804', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-805" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-805', 'end', 'link-75');">end</a></tt><tt class="py-op">,</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-806" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-806', 'location', 'link-62');">location</a></tt><tt class="py-op">.</tt><tt id="link-807" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-807', 'end', 'link-75');">end</a></tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">new</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1178"></a><tt class="py-lineno">1178</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">ignore_sub_features</tt><tt class="py-op">:</tt> </tt>
<a name="L1179"></a><tt class="py-lineno">1179</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-808" class="py-name" targets="Variable Bio.SeqFeature.SeqFeature.sub_features=Bio.SeqFeature.SeqFeature-class.html#sub_features"><a title="Bio.SeqFeature.SeqFeature.sub_features" class="py-name" href="#" onclick="return doclink('link-808', 'sub_features', 'link-808');">sub_features</a></tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-809" class="py-name"><a title="Bio.SeqFeature.SeqFeature.sub_features" class="py-name" href="#" onclick="return doclink('link-809', 'sub_features', 'link-808');">sub_features</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1180"></a><tt class="py-lineno">1180</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Different sub features"</tt><tt class="py-op">)</tt> </tt>
<a name="L1181"></a><tt class="py-lineno">1181</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-810" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-810', 'a', 'link-325');">a</a></tt><tt class="py-op">,</tt> <tt id="link-811" class="py-name" targets="Variable Bio.Graphics.ColorSpiral.ColorSpiral.b=Bio.Graphics.ColorSpiral.ColorSpiral-class.html#b"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-811', 'b', 'link-811');">b</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">.</tt><tt id="link-812" class="py-name"><a title="Bio.SeqFeature.SeqFeature.sub_features" class="py-name" href="#" onclick="return doclink('link-812', 'sub_features', 'link-808');">sub_features</a></tt><tt class="py-op">,</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt id="link-813" class="py-name"><a title="Bio.SeqFeature.SeqFeature.sub_features" class="py-name" href="#" onclick="return doclink('link-813', 'sub_features', 'link-808');">sub_features</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1182"></a><tt class="py-lineno">1182</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">compare_feature</tt><tt class="py-op">(</tt><tt id="link-814" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-814', 'a', 'link-325');">a</a></tt><tt class="py-op">,</tt> <tt id="link-815" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-815', 'b', 'link-811');">b</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1183"></a><tt class="py-lineno">1183</tt>  <tt class="py-line">                    <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
<a name="L1184"></a><tt class="py-lineno">1184</tt>  <tt class="py-line">        <tt class="py-comment"># This only checks key shared qualifiers</tt> </tt>
<a name="L1185"></a><tt class="py-lineno">1185</tt>  <tt class="py-line">        <tt class="py-comment"># Would a white list be easier?</tt> </tt>
<a name="L1186"></a><tt class="py-lineno">1186</tt>  <tt class="py-line">        <tt class="py-comment"># for key in ["name", "gene", "translation", "codon_table", "codon_start", "locus_tag"]:</tt> </tt>
<a name="L1187"></a><tt class="py-lineno">1187</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt id="link-816" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-816', 'set', 'link-766');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">.</tt><tt class="py-name">qualifiers</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">intersection</tt><tt class="py-op">(</tt><tt class="py-name">new</tt><tt class="py-op">.</tt><tt class="py-name">qualifiers</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1188"></a><tt class="py-lineno">1188</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"db_xref"</tt><tt class="py-op">,</tt> <tt class="py-string">"protein_id"</tt><tt class="py-op">,</tt> <tt class="py-string">"product"</tt><tt class="py-op">,</tt> <tt class="py-string">"note"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L1189"></a><tt class="py-lineno">1189</tt>  <tt class="py-line">                <tt class="py-comment"># EMBL and GenBank files are use different references/notes/etc</tt> </tt>
<a name="L1190"></a><tt class="py-lineno">1190</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L1191"></a><tt class="py-lineno">1191</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">old</tt><tt class="py-op">.</tt><tt class="py-name">qualifiers</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt class="py-name">qualifiers</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L1192"></a><tt class="py-lineno">1192</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Qualifier mis-match for %s:\n%s\n%s"</tt> </tt>
<a name="L1193"></a><tt class="py-lineno">1193</tt>  <tt class="py-line">                                 <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">old</tt><tt class="py-op">.</tt><tt class="py-name">qualifiers</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">new</tt><tt class="py-op">.</tt><tt class="py-name">qualifiers</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1194"></a><tt class="py-lineno">1194</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">True</tt> </tt>
</div><a name="L1195"></a><tt class="py-lineno">1195</tt>  <tt class="py-line"> </tt>
<a name="compare_features"></a><div id="compare_features-def"><a name="L1196"></a><tt class="py-lineno">1196</tt> <a class="py-toggle" href="#" id="compare_features-toggle" onclick="return toggle('compare_features');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO-module.html#compare_features">compare_features</a><tt class="py-op">(</tt><tt class="py-param">old_list</tt><tt class="py-op">,</tt> <tt class="py-param">new_list</tt><tt class="py-op">,</tt> <tt class="py-param">ignore_sub_features</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="compare_features-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="compare_features-expanded"><a name="L1197"></a><tt class="py-lineno">1197</tt>  <tt class="py-line">        <tt class="py-docstring">"""Check two lists of SeqFeatures agree, raises a ValueError if mismatch."""</tt> </tt>
<a name="L1198"></a><tt class="py-lineno">1198</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">old_list</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">new_list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1199"></a><tt class="py-lineno">1199</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt> </tt>
<a name="L1200"></a><tt class="py-lineno">1200</tt>  <tt class="py-line">                <tt class="py-string">"%i vs %i features"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">old_list</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">new_list</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1201"></a><tt class="py-lineno">1201</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">old</tt><tt class="py-op">,</tt> <tt class="py-name">new</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">old_list</tt><tt class="py-op">,</tt> <tt class="py-name">new_list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1202"></a><tt class="py-lineno">1202</tt>  <tt class="py-line">            <tt class="py-comment"># This assumes they are in the same order</tt> </tt>
<a name="L1203"></a><tt class="py-lineno">1203</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">compare_feature</tt><tt class="py-op">(</tt><tt class="py-name">old</tt><tt class="py-op">,</tt> <tt class="py-name">new</tt><tt class="py-op">,</tt> <tt class="py-name">ignore_sub_features</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1204"></a><tt class="py-lineno">1204</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
<a name="L1205"></a><tt class="py-lineno">1205</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">True</tt> </tt>
</div><a name="L1206"></a><tt class="py-lineno">1206</tt>  <tt class="py-line"> </tt>
<a name="check_genbank_writer"></a><div id="check_genbank_writer-def"><a name="L1207"></a><tt class="py-lineno">1207</tt> <a class="py-toggle" href="#" id="check_genbank_writer-toggle" onclick="return toggle('check_genbank_writer');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO-module.html#check_genbank_writer">check_genbank_writer</a><tt class="py-op">(</tt><tt class="py-param">records</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="check_genbank_writer-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="check_genbank_writer-expanded"><a name="L1208"></a><tt class="py-lineno">1208</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">StringIO</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1209"></a><tt class="py-lineno">1209</tt>  <tt class="py-line">        <tt id="link-817" class="py-name" targets="Class Bio.SeqIO.InsdcIO.GenBankWriter=Bio.SeqIO.InsdcIO.GenBankWriter-class.html"><a title="Bio.SeqIO.InsdcIO.GenBankWriter" class="py-name" href="#" onclick="return doclink('link-817', 'GenBankWriter', 'link-817');">GenBankWriter</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-818" class="py-name" targets="Method Bio.AlignIO.Interfaces.AlignmentWriter.write_file()=Bio.AlignIO.Interfaces.AlignmentWriter-class.html#write_file,Method Bio.AlignIO.Interfaces.SequentialAlignmentWriter.write_file()=Bio.AlignIO.Interfaces.SequentialAlignmentWriter-class.html#write_file,Method Bio.AlignIO.NexusIO.NexusWriter.write_file()=Bio.AlignIO.NexusIO.NexusWriter-class.html#write_file,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.write_file()=Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter-class.html#write_file,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.write_file()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter-class.html#write_file,Method Bio.SearchIO.BlatIO.BlatPslWriter.write_file()=Bio.SearchIO.BlatIO.BlatPslWriter-class.html#write_file,Method Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.write_file()=Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter-class.html#write_file,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.write_file()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter-class.html#write_file,Method Bio.SeqIO.Interfaces.SequenceWriter.write_file()=Bio.SeqIO.Interfaces.SequenceWriter-class.html#write_file,Method Bio.SeqIO.Interfaces.SequentialSequenceWriter.write_file()=Bio.SeqIO.Interfaces.SequentialSequenceWriter-class.html#write_file,Method Bio.SeqIO.SffIO.SffWriter.write_file()=Bio.SeqIO.SffIO.SffWriter-class.html#write_file"><a title="Bio.AlignIO.Interfaces.AlignmentWriter.write_file
Bio.AlignIO.Interfaces.SequentialAlignmentWriter.write_file
Bio.AlignIO.NexusIO.NexusWriter.write_file
Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.write_file
Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.write_file
Bio.SearchIO.BlatIO.BlatPslWriter.write_file
Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.write_file
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.write_file
Bio.SeqIO.Interfaces.SequenceWriter.write_file
Bio.SeqIO.Interfaces.SequentialSequenceWriter.write_file
Bio.SeqIO.SffIO.SffWriter.write_file" class="py-name" href="#" onclick="return doclink('link-818', 'write_file', 'link-818');">write_file</a></tt><tt class="py-op">(</tt><tt class="py-name">records</tt><tt class="py-op">)</tt> </tt>
<a name="L1210"></a><tt class="py-lineno">1210</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-819" class="py-name" targets="Method Bio.File.UndoHandle.seek()=Bio.File.UndoHandle-class.html#seek,Method Bio.SeqIO.SffIO._AddTellHandle.seek()=Bio.SeqIO.SffIO._AddTellHandle-class.html#seek,Method Bio.bgzf.BgzfReader.seek()=Bio.bgzf.BgzfReader-class.html#seek"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-819', 'seek', 'link-819');">seek</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L1211"></a><tt class="py-lineno">1211</tt>  <tt class="py-line"> </tt>
<a name="L1212"></a><tt class="py-lineno">1212</tt>  <tt class="py-line">        <tt class="py-name">records2</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-820" class="py-name" targets="Function Bio.SeqIO.InsdcIO.GenBankIterator()=Bio.SeqIO.InsdcIO-module.html#GenBankIterator"><a title="Bio.SeqIO.InsdcIO.GenBankIterator" class="py-name" href="#" onclick="return doclink('link-820', 'GenBankIterator', 'link-820');">GenBankIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1213"></a><tt class="py-lineno">1213</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">compare_records</tt><tt class="py-op">(</tt><tt class="py-name">records</tt><tt class="py-op">,</tt> <tt class="py-name">records2</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1214"></a><tt class="py-lineno">1214</tt>  <tt class="py-line"> </tt>
<a name="check_embl_writer"></a><div id="check_embl_writer-def"><a name="L1215"></a><tt class="py-lineno">1215</tt> <a class="py-toggle" href="#" id="check_embl_writer-toggle" onclick="return toggle('check_embl_writer');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.InsdcIO-module.html#check_embl_writer">check_embl_writer</a><tt class="py-op">(</tt><tt class="py-param">records</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="check_embl_writer-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="check_embl_writer-expanded"><a name="L1216"></a><tt class="py-lineno">1216</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">StringIO</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1217"></a><tt class="py-lineno">1217</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1218"></a><tt class="py-lineno">1218</tt>  <tt class="py-line">            <tt id="link-821" class="py-name" targets="Class Bio.SeqIO.InsdcIO.EmblWriter=Bio.SeqIO.InsdcIO.EmblWriter-class.html"><a title="Bio.SeqIO.InsdcIO.EmblWriter" class="py-name" href="#" onclick="return doclink('link-821', 'EmblWriter', 'link-821');">EmblWriter</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-822" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentWriter.write_file
Bio.AlignIO.Interfaces.SequentialAlignmentWriter.write_file
Bio.AlignIO.NexusIO.NexusWriter.write_file
Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.write_file
Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.write_file
Bio.SearchIO.BlatIO.BlatPslWriter.write_file
Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.write_file
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.write_file
Bio.SeqIO.Interfaces.SequenceWriter.write_file
Bio.SeqIO.Interfaces.SequentialSequenceWriter.write_file
Bio.SeqIO.SffIO.SffWriter.write_file" class="py-name" href="#" onclick="return doclink('link-822', 'write_file', 'link-818');">write_file</a></tt><tt class="py-op">(</tt><tt class="py-name">records</tt><tt class="py-op">)</tt> </tt>
<a name="L1219"></a><tt class="py-lineno">1219</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt> <tt class="py-keyword">as</tt> <tt class="py-name">err</tt><tt class="py-op">:</tt> </tt>
<a name="L1220"></a><tt class="py-lineno">1220</tt>  <tt class="py-line">            <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">err</tt><tt class="py-op">)</tt> </tt>
<a name="L1221"></a><tt class="py-lineno">1221</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L1222"></a><tt class="py-lineno">1222</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-823" class="py-name"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-823', 'seek', 'link-819');">seek</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L1223"></a><tt class="py-lineno">1223</tt>  <tt class="py-line"> </tt>
<a name="L1224"></a><tt class="py-lineno">1224</tt>  <tt class="py-line">        <tt class="py-name">records2</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-824" class="py-name" targets="Function Bio.SeqIO.InsdcIO.EmblIterator()=Bio.SeqIO.InsdcIO-module.html#EmblIterator"><a title="Bio.SeqIO.InsdcIO.EmblIterator" class="py-name" href="#" onclick="return doclink('link-824', 'EmblIterator', 'link-824');">EmblIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1225"></a><tt class="py-lineno">1225</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">compare_records</tt><tt class="py-op">(</tt><tt class="py-name">records</tt><tt class="py-op">,</tt> <tt class="py-name">records2</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1226"></a><tt class="py-lineno">1226</tt>  <tt class="py-line"> </tt>
<a name="L1227"></a><tt class="py-lineno">1227</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">filename</tt> <tt class="py-keyword">in</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">listdir</tt><tt class="py-op">(</tt><tt class="py-string">"../../Tests/GenBank"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1228"></a><tt class="py-lineno">1228</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">filename</tt><tt class="py-op">.</tt><tt id="link-825" class="py-name"><a title="Bio.Seq.Seq.endswith" class="py-name" href="#" onclick="return doclink('link-825', 'endswith', 'link-483');">endswith</a></tt><tt class="py-op">(</tt><tt class="py-string">".gbk"</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt class="py-name">filename</tt><tt class="py-op">.</tt><tt id="link-826" class="py-name"><a title="Bio.Seq.Seq.endswith" class="py-name" href="#" onclick="return doclink('link-826', 'endswith', 'link-483');">endswith</a></tt><tt class="py-op">(</tt><tt class="py-string">".gb"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1229"></a><tt class="py-lineno">1229</tt>  <tt class="py-line">            <tt class="py-keyword">continue</tt> </tt>
<a name="L1230"></a><tt class="py-lineno">1230</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">)</tt> </tt>
<a name="L1231"></a><tt class="py-lineno">1231</tt>  <tt class="py-line"> </tt>
<a name="L1232"></a><tt class="py-lineno">1232</tt>  <tt class="py-line">        <tt class="py-keyword">with</tt> <tt id="link-827" class="py-name" targets="Function Bio.bgzf.open()=Bio.bgzf-module.html#open"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-827', 'open', 'link-827');">open</a></tt><tt class="py-op">(</tt><tt class="py-string">"../../Tests/GenBank/%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">filename</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1233"></a><tt class="py-lineno">1233</tt>  <tt class="py-line">            <tt class="py-name">records</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-828" class="py-name"><a title="Bio.SeqIO.InsdcIO.GenBankIterator" class="py-name" href="#" onclick="return doclink('link-828', 'GenBankIterator', 'link-820');">GenBankIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1234"></a><tt class="py-lineno">1234</tt>  <tt class="py-line"> </tt>
<a name="L1235"></a><tt class="py-lineno">1235</tt>  <tt class="py-line">        <tt class="py-name">check_genbank_writer</tt><tt class="py-op">(</tt><tt class="py-name">records</tt><tt class="py-op">)</tt> </tt>
<a name="L1236"></a><tt class="py-lineno">1236</tt>  <tt class="py-line">        <tt class="py-name">check_embl_writer</tt><tt class="py-op">(</tt><tt class="py-name">records</tt><tt class="py-op">)</tt> </tt>
<a name="L1237"></a><tt class="py-lineno">1237</tt>  <tt class="py-line"> </tt>
<a name="L1238"></a><tt class="py-lineno">1238</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">filename</tt> <tt class="py-keyword">in</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">listdir</tt><tt class="py-op">(</tt><tt class="py-string">"../../Tests/EMBL"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1239"></a><tt class="py-lineno">1239</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">filename</tt><tt class="py-op">.</tt><tt id="link-829" class="py-name"><a title="Bio.Seq.Seq.endswith" class="py-name" href="#" onclick="return doclink('link-829', 'endswith', 'link-483');">endswith</a></tt><tt class="py-op">(</tt><tt class="py-string">".embl"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1240"></a><tt class="py-lineno">1240</tt>  <tt class="py-line">            <tt class="py-keyword">continue</tt> </tt>
<a name="L1241"></a><tt class="py-lineno">1241</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">)</tt> </tt>
<a name="L1242"></a><tt class="py-lineno">1242</tt>  <tt class="py-line"> </tt>
<a name="L1243"></a><tt class="py-lineno">1243</tt>  <tt class="py-line">        <tt class="py-keyword">with</tt> <tt id="link-830" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-830', 'open', 'link-827');">open</a></tt><tt class="py-op">(</tt><tt class="py-string">"../../Tests/EMBL/%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">filename</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1244"></a><tt class="py-lineno">1244</tt>  <tt class="py-line">            <tt class="py-name">records</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-831" class="py-name"><a title="Bio.SeqIO.InsdcIO.EmblIterator" class="py-name" href="#" onclick="return doclink('link-831', 'EmblIterator', 'link-824');">EmblIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1245"></a><tt class="py-lineno">1245</tt>  <tt class="py-line"> </tt>
<a name="L1246"></a><tt class="py-lineno">1246</tt>  <tt class="py-line">        <tt class="py-name">check_genbank_writer</tt><tt class="py-op">(</tt><tt class="py-name">records</tt><tt class="py-op">)</tt> </tt>
<a name="L1247"></a><tt class="py-lineno">1247</tt>  <tt class="py-line">        <tt class="py-name">check_embl_writer</tt><tt class="py-op">(</tt><tt class="py-name">records</tt><tt class="py-op">)</tt> </tt>
<a name="L1248"></a><tt class="py-lineno">1248</tt>  <tt class="py-line"> </tt>
<a name="L1249"></a><tt class="py-lineno">1249</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-832" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-832', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-833" class="py-name" targets="Package Bio.SeqIO=Bio.SeqIO-module.html"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-833', 'SeqIO', 'link-833');">SeqIO</a></tt> </tt>
<a name="L1250"></a><tt class="py-lineno">1250</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">filename</tt> <tt class="py-keyword">in</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">listdir</tt><tt class="py-op">(</tt><tt class="py-string">"../../Tests/SwissProt"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1251"></a><tt class="py-lineno">1251</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">filename</tt><tt class="py-op">.</tt><tt id="link-834" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-834', 'startswith', 'link-463');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"sp"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1252"></a><tt class="py-lineno">1252</tt>  <tt class="py-line">            <tt class="py-keyword">continue</tt> </tt>
<a name="L1253"></a><tt class="py-lineno">1253</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">)</tt> </tt>
<a name="L1254"></a><tt class="py-lineno">1254</tt>  <tt class="py-line"> </tt>
<a name="L1255"></a><tt class="py-lineno">1255</tt>  <tt class="py-line">        <tt class="py-keyword">with</tt> <tt id="link-835" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-835', 'open', 'link-827');">open</a></tt><tt class="py-op">(</tt><tt class="py-string">"../../Tests/SwissProt/%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">filename</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1256"></a><tt class="py-lineno">1256</tt>  <tt class="py-line">            <tt class="py-name">records</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-836" class="py-name"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-836', 'SeqIO', 'link-833');">SeqIO</a></tt><tt class="py-op">.</tt><tt id="link-837" class="py-name" targets="Function Bio.AlignIO.parse()=Bio.AlignIO-module.html#parse,Method Bio.Blast.NCBIStandalone.BlastErrorParser.parse()=Bio.Blast.NCBIStandalone.BlastErrorParser-class.html#parse,Method Bio.Blast.NCBIStandalone.BlastParser.parse()=Bio.Blast.NCBIStandalone.BlastParser-class.html#parse,Method Bio.Blast.NCBIStandalone.PSIBlastParser.parse()=Bio.Blast.NCBIStandalone.PSIBlastParser-class.html#parse,Function Bio.Blast.NCBIXML.parse()=Bio.Blast.NCBIXML-module.html#parse,Function Bio.Compass.parse()=Bio.Compass-module.html#parse,Function Bio.Emboss.Primer3.parse()=Bio.Emboss.Primer3-module.html#parse,Method Bio.Entrez.Parser.DataHandler.parse()=Bio.Entrez.Parser.DataHandler-class.html#parse,Function Bio.Entrez.parse()=Bio.Entrez-module.html#parse,Function Bio.ExPASy.Enzyme.parse()=Bio.ExPASy.Enzyme-module.html#parse,Function Bio.ExPASy.Prodoc.parse()=Bio.ExPASy.Prodoc-module.html#parse,Function Bio.ExPASy.Prosite.parse()=Bio.ExPASy.Prosite-module.html#parse,Method Bio.GenBank.FeatureParser.parse()=Bio.GenBank.FeatureParser-class.html#parse,Method Bio.GenBank.RecordParser.parse()=Bio.GenBank.RecordParser-class.html#parse,Method Bio.GenBank.Scanner.InsdcScanner.parse()=Bio.GenBank.Scanner.InsdcScanner-class.html#parse,Function Bio.Geo.parse()=Bio.Geo-module.html#parse,Function Bio.KEGG.Compound.parse()=Bio.KEGG.Compound-module.html#parse,Function Bio.KEGG.Enzyme.parse()=Bio.KEGG.Enzyme-module.html#parse,Method Bio.KEGG.KGML.KGML_parser.KGMLParser.parse()=Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html#parse,Function Bio.KEGG.KGML.KGML_parser.parse()=Bio.KEGG.KGML.KGML_parser-module.html#parse,Function Bio.KEGG.Map.parse()=Bio.KEGG.Map-module.html#parse,Function Bio.Medline.parse()=Bio.Medline-module.html#parse,Function Bio.Motif.parse()=Bio.Motif-module.html#parse,Method Bio.ParserSupport.AbstractParser.parse()=Bio.ParserSupport.AbstractParser-class.html#parse,Method Bio.Phylo.CDAOIO.Parser.parse()=Bio.Phylo.CDAOIO.Parser-class.html#parse,Function Bio.Phylo.CDAOIO.parse()=Bio.Phylo.CDAOIO-module.html#parse,Method Bio.Phylo.NeXMLIO.Parser.parse()=Bio.Phylo.NeXMLIO.Parser-class.html#parse,Function Bio.Phylo.NeXMLIO.parse()=Bio.Phylo.NeXMLIO-module.html#parse,Method Bio.Phylo.NewickIO.Parser.parse()=Bio.Phylo.NewickIO.Parser-class.html#parse,Function Bio.Phylo.NewickIO.parse()=Bio.Phylo.NewickIO-module.html#parse,Function Bio.Phylo.NexusIO.parse()=Bio.Phylo.NexusIO-module.html#parse,Method Bio.Phylo.PhyloXMLIO.Parser.parse()=Bio.Phylo.PhyloXMLIO.Parser-class.html#parse,Function Bio.Phylo.PhyloXMLIO.parse()=Bio.Phylo.PhyloXMLIO-module.html#parse,Function Bio.Phylo._io.parse()=Bio.Phylo._io-module.html#parse,Function Bio.SCOP.Cla.parse()=Bio.SCOP.Cla-module.html#parse,Function Bio.SCOP.Des.parse()=Bio.SCOP.Des-module.html#parse,Function Bio.SCOP.Dom.parse()=Bio.SCOP.Dom-module.html#parse,Function Bio.SCOP.Hie.parse()=Bio.SCOP.Hie-module.html#parse,Function Bio.SCOP.Raf.parse()=Bio.SCOP.Raf-module.html#parse,Function Bio.SearchIO.parse()=Bio.SearchIO-module.html#parse,Method Bio.SeqIO.UniprotIO.Parser.parse()=Bio.SeqIO.UniprotIO.Parser-class.html#parse,Function Bio.SeqIO.parse()=Bio.SeqIO-module.html#parse,Function Bio.Sequencing.Ace.parse()=Bio.Sequencing.Ace-module.html#parse,Function Bio.Sequencing.Phd.parse()=Bio.Sequencing.Phd-module.html#parse,Function Bio.SwissProt.KeyWList.parse()=Bio.SwissProt.KeyWList-module.html#parse,Function Bio.SwissProt.parse()=Bio.SwissProt-module.html#parse,Function Bio.UniGene.parse()=Bio.UniGene-module.html#parse,Function Bio.Wise.psw.parse()=Bio.Wise.psw-module.html#parse,Function Bio.motifs.parse()=Bio.motifs-module.html#parse"><a title="Bio.AlignIO.parse
Bio.Blast.NCBIStandalone.BlastErrorParser.parse
Bio.Blast.NCBIStandalone.BlastParser.parse
Bio.Blast.NCBIStandalone.PSIBlastParser.parse
Bio.Blast.NCBIXML.parse
Bio.Compass.parse
Bio.Emboss.Primer3.parse
Bio.Entrez.Parser.DataHandler.parse
Bio.Entrez.parse
Bio.ExPASy.Enzyme.parse
Bio.ExPASy.Prodoc.parse
Bio.ExPASy.Prosite.parse
Bio.GenBank.FeatureParser.parse
Bio.GenBank.RecordParser.parse
Bio.GenBank.Scanner.InsdcScanner.parse
Bio.Geo.parse
Bio.KEGG.Compound.parse
Bio.KEGG.Enzyme.parse
Bio.KEGG.KGML.KGML_parser.KGMLParser.parse
Bio.KEGG.KGML.KGML_parser.parse
Bio.KEGG.Map.parse
Bio.Medline.parse
Bio.Motif.parse
Bio.ParserSupport.AbstractParser.parse
Bio.Phylo.CDAOIO.Parser.parse
Bio.Phylo.CDAOIO.parse
Bio.Phylo.NeXMLIO.Parser.parse
Bio.Phylo.NeXMLIO.parse
Bio.Phylo.NewickIO.Parser.parse
Bio.Phylo.NewickIO.parse
Bio.Phylo.NexusIO.parse
Bio.Phylo.PhyloXMLIO.Parser.parse
Bio.Phylo.PhyloXMLIO.parse
Bio.Phylo._io.parse
Bio.SCOP.Cla.parse
Bio.SCOP.Des.parse
Bio.SCOP.Dom.parse
Bio.SCOP.Hie.parse
Bio.SCOP.Raf.parse
Bio.SearchIO.parse
Bio.SeqIO.UniprotIO.Parser.parse
Bio.SeqIO.parse
Bio.Sequencing.Ace.parse
Bio.Sequencing.Phd.parse
Bio.SwissProt.KeyWList.parse
Bio.SwissProt.parse
Bio.UniGene.parse
Bio.Wise.psw.parse
Bio.motifs.parse" class="py-name" href="#" onclick="return doclink('link-837', 'parse', 'link-837');">parse</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-string">"swiss"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1257"></a><tt class="py-lineno">1257</tt>  <tt class="py-line"> </tt>
<a name="L1258"></a><tt class="py-lineno">1258</tt>  <tt class="py-line">        <tt class="py-name">check_genbank_writer</tt><tt class="py-op">(</tt><tt class="py-name">records</tt><tt class="py-op">)</tt> </tt>
<a name="L1259"></a><tt class="py-lineno">1259</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:04:55 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
