<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.SeqIO.AbiIO</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.SeqIO-module.html">Package&nbsp;SeqIO</a> ::
        Module&nbsp;AbiIO
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.SeqIO.AbiIO-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.SeqIO.AbiIO-module.html">Module Bio.SeqIO.AbiIO</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2011 by Wibowo Arindrarto (w.arindrarto@gmail.com)</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># Revisions copyright 2011-2016 by Peter Cock.</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># This file is part of the Biopython distribution and governed by your</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-comment"># choice of the "Biopython License Agreement" or the "BSD 3-Clause License".</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-comment"># Please see the LICENSE file that should have been included as part of this</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-comment"># package.</tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-docstring">"""Bio.SeqIO parser for the ABI format.</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-docstring">ABI is the format used by Applied Biosystem's sequencing machines to store</tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-docstring">sequencing results.</tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-docstring">For more details on the format specification, visit:</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-docstring">http://www6.appliedbiosystems.com/support/software_community/ABIF_File_Format.pdf</tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">datetime</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">struct</tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt> <tt class="py-keyword">import</tt> <tt class="py-name">basename</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-1" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-1', 'Alphabet', 'link-1');">Alphabet</a></tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-2" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-2', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-3" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-3', 'Alphabet', 'link-1');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-4" class="py-name" targets="Module Bio.Alphabet.IUPAC=Bio.Alphabet.IUPAC-module.html"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-4', 'IUPAC', 'link-4');">IUPAC</a></tt> <tt class="py-keyword">import</tt> <tt id="link-5" class="py-name" targets="Variable Bio.Alphabet.IUPAC.ambiguous_dna=Bio.Alphabet.IUPAC-module.html#ambiguous_dna"><a title="Bio.Alphabet.IUPAC.ambiguous_dna" class="py-name" href="#" onclick="return doclink('link-5', 'ambiguous_dna', 'link-5');">ambiguous_dna</a></tt><tt class="py-op">,</tt> <tt id="link-6" class="py-name" targets="Variable Bio.Alphabet.IUPAC.unambiguous_dna=Bio.Alphabet.IUPAC-module.html#unambiguous_dna"><a title="Bio.Alphabet.IUPAC.unambiguous_dna" class="py-name" href="#" onclick="return doclink('link-6', 'unambiguous_dna', 'link-6');">unambiguous_dna</a></tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-7" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-7', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-8" class="py-name" targets="Module Bio.Seq=Bio.Seq-module.html,Class Bio.Seq.Seq=Bio.Seq.Seq-class.html"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-8', 'Seq', 'link-8');">Seq</a></tt> <tt class="py-keyword">import</tt> <tt id="link-9" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-9', 'Seq', 'link-8');">Seq</a></tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-10" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-10', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-11" class="py-name" targets="Module Bio.SeqRecord=Bio.SeqRecord-module.html,Class Bio.SeqRecord.SeqRecord=Bio.SeqRecord.SeqRecord-class.html"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-11', 'SeqRecord', 'link-11');">SeqRecord</a></tt> <tt class="py-keyword">import</tt> <tt id="link-12" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-12', 'SeqRecord', 'link-11');">SeqRecord</a></tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-13" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-13', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-14" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-14', '_py3k', 'link-14');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt id="link-15" class="py-name" targets="Function Bio._py3k._bytes_to_string()=Bio._py3k-module.html#_bytes_to_string"><a title="Bio._py3k._bytes_to_string" class="py-name" href="#" onclick="return doclink('link-15', '_bytes_to_string', 'link-15');">_bytes_to_string</a></tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-16" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-16', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-17" class="py-name"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-17', '_py3k', 'link-14');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">zip</tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"><tt class="py-comment"># dictionary for determining which tags goes into SeqRecord annotation</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-comment"># each key is tag_name + tag_number</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-comment"># if a tag entry needs to be added, just add its key and its key</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-comment"># for the annotations dictionary as the value</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"><tt class="py-comment"># dictionary for tags that require preprocessing before use in creating</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"><tt class="py-comment"># seqrecords</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"><tt id="link-18" class="py-name" targets="Variable Bio.SeqIO.AbiIO._EXTRACT=Bio.SeqIO.AbiIO-module.html#_EXTRACT"><a title="Bio.SeqIO.AbiIO._EXTRACT" class="py-name" href="#" onclick="return doclink('link-18', '_EXTRACT', 'link-18');">_EXTRACT</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line">    <tt class="py-string">'TUBE1'</tt><tt class="py-op">:</tt> <tt class="py-string">'sample_well'</tt><tt class="py-op">,</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line">    <tt class="py-string">'DySN1'</tt><tt class="py-op">:</tt> <tt class="py-string">'dye'</tt><tt class="py-op">,</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line">    <tt class="py-string">'GTyp1'</tt><tt class="py-op">:</tt> <tt class="py-string">'polymer'</tt><tt class="py-op">,</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line">    <tt class="py-string">'MODL1'</tt><tt class="py-op">:</tt> <tt class="py-string">'machine_model'</tt><tt class="py-op">,</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line"><tt class="py-op">}</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line"> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line"> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line"><tt class="py-comment"># Complete data structure representing 98% of the API. The general section</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line"><tt class="py-comment"># represents the part of the API that's common to ALL instruments, whereas the</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"><tt class="py-comment"># instrument specific sections are labelled as they are in the ABIF spec</tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line"><tt class="py-comment"># Keys don't seem to clash from machine to machine, so when we parse, we look</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line"><tt class="py-comment"># for ANY key, and store that in the raw ABIF data structure attached to the</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line"><tt class="py-comment"># annotations, with the assumption that anyone parsing the data can look up the</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line"><tt class="py-comment"># spec themself</tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line"><tt class="py-comment"># Key definitions are retained in case end users want "nice" labels pre-made</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line"><tt class="py-comment"># for them for all of the available fields.</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"><tt id="link-19" class="py-name" targets="Variable Bio.SeqIO.AbiIO._INSTRUMENT_SPECIFIC_TAGS=Bio.SeqIO.AbiIO-module.html#_INSTRUMENT_SPECIFIC_TAGS"><a title="Bio.SeqIO.AbiIO._INSTRUMENT_SPECIFIC_TAGS" class="py-name" href="#" onclick="return doclink('link-19', '_INSTRUMENT_SPECIFIC_TAGS', 'link-19');">_INSTRUMENT_SPECIFIC_TAGS</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line"> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line"><tt id="link-20" class="py-name"><a title="Bio.SeqIO.AbiIO._INSTRUMENT_SPECIFIC_TAGS" class="py-name" href="#" onclick="return doclink('link-20', '_INSTRUMENT_SPECIFIC_TAGS', 'link-19');">_INSTRUMENT_SPECIFIC_TAGS</a></tt><tt class="py-op">[</tt><tt class="py-string">'general'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line">    <tt class="py-string">'APFN2'</tt><tt class="py-op">:</tt> <tt class="py-string">'Sequencing Analysis parameters file name'</tt><tt class="py-op">,</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line">    <tt class="py-string">'APXV1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Analysis Protocol XML schema version'</tt><tt class="py-op">,</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line">    <tt class="py-string">'APrN1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Analysis Protocol settings name'</tt><tt class="py-op">,</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line">    <tt class="py-string">'APrV1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Analysis Protocol settings version'</tt><tt class="py-op">,</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line">    <tt class="py-string">'APrX1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Analysis Protocol XML string'</tt><tt class="py-op">,</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line">    <tt class="py-string">'CMNT1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Sample Comment'</tt><tt class="py-op">,</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line">    <tt class="py-string">'CTID1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Container Identifier, a.k.a. plate barcode'</tt><tt class="py-op">,</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line">    <tt class="py-string">'CTNM1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Container name, usually identical to CTID, but not necessarily so'</tt><tt class="py-op">,</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line">    <tt class="py-string">'CTTL1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Comment Title'</tt><tt class="py-op">,</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line">    <tt class="py-string">'CpEP1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Capillary type electrophoresis. 1 for a capillary based machine. 0 for a slab gel based machine.'</tt><tt class="py-op">,</tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line">    <tt class="py-string">'DATA1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Channel 1 raw data'</tt><tt class="py-op">,</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line">    <tt class="py-string">'DATA2'</tt><tt class="py-op">:</tt> <tt class="py-string">'Channel 2 raw data'</tt><tt class="py-op">,</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line">    <tt class="py-string">'DATA3'</tt><tt class="py-op">:</tt> <tt class="py-string">'Channel 3 raw data'</tt><tt class="py-op">,</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line">    <tt class="py-string">'DATA4'</tt><tt class="py-op">:</tt> <tt class="py-string">'Channel 4 raw data'</tt><tt class="py-op">,</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">    <tt class="py-string">'DATA5'</tt><tt class="py-op">:</tt> <tt class="py-string">'Short Array holding measured volts/10 (EP voltage) during run'</tt><tt class="py-op">,</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line">    <tt class="py-string">'DATA6'</tt><tt class="py-op">:</tt> <tt class="py-string">'Short Array holding measured milliAmps trace (EP current) during run'</tt><tt class="py-op">,</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line">    <tt class="py-string">'DATA7'</tt><tt class="py-op">:</tt> <tt class="py-string">'Short Array holding measured milliWatts trace (Laser EP Power) during run'</tt><tt class="py-op">,</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line">    <tt class="py-string">'DATA8'</tt><tt class="py-op">:</tt> <tt class="py-string">'Short Array holding measured oven Temperature (polymer temperature) trace during run'</tt><tt class="py-op">,</tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line">    <tt class="py-string">'DATA9'</tt><tt class="py-op">:</tt> <tt class="py-string">'Channel 9 processed data'</tt><tt class="py-op">,</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line">    <tt class="py-string">'DATA10'</tt><tt class="py-op">:</tt> <tt class="py-string">'Channel 10 processed data'</tt><tt class="py-op">,</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line">    <tt class="py-string">'DATA11'</tt><tt class="py-op">:</tt> <tt class="py-string">'Channel 11 processed data'</tt><tt class="py-op">,</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">    <tt class="py-string">'DATA12'</tt><tt class="py-op">:</tt> <tt class="py-string">'Channel 12 processed data'</tt><tt class="py-op">,</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line">    <tt class="py-comment"># Prism 3100/3100-Avant may provide DATA105</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line">    <tt class="py-comment">#          3130/3130-XL may provide DATA105</tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line">    <tt class="py-comment"># 3530/3530-XL may provide DATA105-199, 9-12, 205-299</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line">    <tt class="py-string">'DSam1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Downsampling factor'</tt><tt class="py-op">,</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line">    <tt class="py-string">'DySN1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Dye set name'</tt><tt class="py-op">,</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line">    <tt class="py-string">'Dye#1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Number of dyes'</tt><tt class="py-op">,</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line">    <tt class="py-string">'DyeN1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Dye 1 name'</tt><tt class="py-op">,</tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line">    <tt class="py-string">'DyeN2'</tt><tt class="py-op">:</tt> <tt class="py-string">'Dye 2 name'</tt><tt class="py-op">,</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line">    <tt class="py-string">'DyeN3'</tt><tt class="py-op">:</tt> <tt class="py-string">'Dye 3 name'</tt><tt class="py-op">,</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">    <tt class="py-string">'DyeN4'</tt><tt class="py-op">:</tt> <tt class="py-string">'Dye 4 name'</tt><tt class="py-op">,</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line">    <tt class="py-string">'DyeW1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Dye 1 wavelength'</tt><tt class="py-op">,</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">    <tt class="py-string">'DyeW2'</tt><tt class="py-op">:</tt> <tt class="py-string">'Dye 2 wavelength'</tt><tt class="py-op">,</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line">    <tt class="py-string">'DyeW3'</tt><tt class="py-op">:</tt> <tt class="py-string">'Dye 3 wavelength'</tt><tt class="py-op">,</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line">    <tt class="py-string">'DyeW4'</tt><tt class="py-op">:</tt> <tt class="py-string">'Dye 4 wavelength'</tt><tt class="py-op">,</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line">    <tt class="py-comment"># 'DyeN5-N': 'Dye 5-N Name',</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line">    <tt class="py-comment"># 'DyeW5-N': 'Dye 5-N Wavelength',</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line">    <tt class="py-string">'EPVt1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Electrophoresis voltage setting (volts)'</tt><tt class="py-op">,</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">    <tt class="py-string">'EVNT1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Start Run event'</tt><tt class="py-op">,</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line">    <tt class="py-string">'EVNT2'</tt><tt class="py-op">:</tt> <tt class="py-string">'Stop Run event'</tt><tt class="py-op">,</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line">    <tt class="py-string">'EVNT3'</tt><tt class="py-op">:</tt> <tt class="py-string">'Start Collection event'</tt><tt class="py-op">,</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line">    <tt class="py-string">'EVNT4'</tt><tt class="py-op">:</tt> <tt class="py-string">'Stop Collection event'</tt><tt class="py-op">,</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">    <tt class="py-string">'FWO_1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Base Order. Sequencing Analysis Filter wheel order. Fixed for 3500 at "GATC"'</tt><tt class="py-op">,</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line">    <tt class="py-string">'GTyp1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Gel or polymer Type'</tt><tt class="py-op">,</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">    <tt class="py-string">'InSc1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Injection time (seconds)'</tt><tt class="py-op">,</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">    <tt class="py-string">'InVt1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Injection voltage (volts)'</tt><tt class="py-op">,</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line">    <tt class="py-string">'LANE1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Lane/Capillary'</tt><tt class="py-op">,</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line">    <tt class="py-string">'LIMS1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Sample tracking ID'</tt><tt class="py-op">,</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">    <tt class="py-string">'LNTD1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Length to detector'</tt><tt class="py-op">,</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">    <tt class="py-string">'LsrP1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Laser Power setting (micro Watts)'</tt><tt class="py-op">,</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">    <tt class="py-string">'MCHN1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Instrument name and serial number'</tt><tt class="py-op">,</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">    <tt class="py-string">'MODF1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Data collection module file'</tt><tt class="py-op">,</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">    <tt class="py-string">'MODL1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Model number'</tt><tt class="py-op">,</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">    <tt class="py-string">'NAVG1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Pixels averaged per lane'</tt><tt class="py-op">,</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">    <tt class="py-string">'NLNE1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Number of capillaries'</tt><tt class="py-op">,</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">    <tt class="py-string">'OfSc1'</tt><tt class="py-op">:</tt> <tt class="py-string">'List of scans that are marked off scale in Collection. (optional)'</tt><tt class="py-op">,</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">    <tt class="py-comment"># OvrI and OrvV are listed as "1-N", and "One for each dye (unanalyzed</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line">    <tt class="py-comment"># and/or analyzed data)"</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line">    <tt class="py-string">'OvrI1'</tt><tt class="py-op">:</tt> <tt class="py-string">'List of scan number indexes that have values greater than 32767 but did not '</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">             <tt class="py-string">'saturate the camera. In Genemapper samples, this can have indexes with '</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">             <tt class="py-string">'values greater than 32000. In sequencing samples, this cannot have '</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">             <tt class="py-string">'indexes with values greater than 32000.'</tt><tt class="py-op">,</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">    <tt class="py-string">'OvrI2'</tt><tt class="py-op">:</tt> <tt class="py-string">'List of scan number indexes that have values greater than 32767 but did not '</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">             <tt class="py-string">'saturate the camera. In Genemapper samples, this can have indexes with '</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line">             <tt class="py-string">'values greater than 32000. In sequencing samples, this cannot have '</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line">             <tt class="py-string">'indexes with values greater than 32000.'</tt><tt class="py-op">,</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line">    <tt class="py-string">'OvrI3'</tt><tt class="py-op">:</tt> <tt class="py-string">'List of scan number indexes that have values greater than 32767 but did not '</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line">             <tt class="py-string">'saturate the camera. In Genemapper samples, this can have indexes with '</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line">             <tt class="py-string">'values greater than 32000. In sequencing samples, this cannot have '</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line">             <tt class="py-string">'indexes with values greater than 32000.'</tt><tt class="py-op">,</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line">    <tt class="py-string">'OvrI4'</tt><tt class="py-op">:</tt> <tt class="py-string">'List of scan number indexes that have values greater than 32767 but did not '</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line">             <tt class="py-string">'saturate the camera. In Genemapper samples, this can have indexes with '</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line">             <tt class="py-string">'values greater than 32000. In sequencing samples, this cannot have '</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line">             <tt class="py-string">'indexes with values greater than 32000.'</tt><tt class="py-op">,</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line">    <tt class="py-string">'OvrV1'</tt><tt class="py-op">:</tt> <tt class="py-string">'List of color data values found at the locations listed in the OvrI tag. '</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">             <tt class="py-string">'There must be exactly as many numbers in this array as in the OvrI array.'</tt><tt class="py-op">,</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line">    <tt class="py-string">'OvrV2'</tt><tt class="py-op">:</tt> <tt class="py-string">'List of color data values found at the locations listed in the OvrI tag. '</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line">             <tt class="py-string">'There must be exactly as many numbers in this array as in the OvrI array.'</tt><tt class="py-op">,</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line">    <tt class="py-string">'OvrV3'</tt><tt class="py-op">:</tt> <tt class="py-string">'List of color data values found at the locations listed in the OvrI tag. '</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line">             <tt class="py-string">'There must be exactly as many numbers in this array as in the OvrI array.'</tt><tt class="py-op">,</tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line">    <tt class="py-string">'OvrV4'</tt><tt class="py-op">:</tt> <tt class="py-string">'List of color data values found at the locations listed in the OvrI tag. '</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line">             <tt class="py-string">'There must be exactly as many numbers in this array as in the OvrI array.'</tt><tt class="py-op">,</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line">    <tt class="py-string">'PDMF1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Sequencing Analysis Mobility file name chosen in collection'</tt><tt class="py-op">,</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line">    <tt class="py-string">'RMXV1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Run Module XML schema version'</tt><tt class="py-op">,</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">    <tt class="py-string">'RMdN1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Run Module name (same as MODF)'</tt><tt class="py-op">,</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">    <tt class="py-string">'RMdX1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Run Module XML string'</tt><tt class="py-op">,</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line">    <tt class="py-string">'RPrN1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Run Protocol name'</tt><tt class="py-op">,</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">    <tt class="py-string">'RPrV1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Run Protocol version'</tt><tt class="py-op">,</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line">    <tt class="py-string">'RUND1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Run Started Date'</tt><tt class="py-op">,</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">    <tt class="py-string">'RUND2'</tt><tt class="py-op">:</tt> <tt class="py-string">'Run Stopped Date'</tt><tt class="py-op">,</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">    <tt class="py-string">'RUND3'</tt><tt class="py-op">:</tt> <tt class="py-string">'Data Collection Started Date'</tt><tt class="py-op">,</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">    <tt class="py-string">'RUND4'</tt><tt class="py-op">:</tt> <tt class="py-string">'Data Collection Stopped date'</tt><tt class="py-op">,</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">    <tt class="py-string">'RUNT1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Run Started Time'</tt><tt class="py-op">,</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line">    <tt class="py-string">'RUNT2'</tt><tt class="py-op">:</tt> <tt class="py-string">'Run Stopped Time'</tt><tt class="py-op">,</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">    <tt class="py-string">'RUNT3'</tt><tt class="py-op">:</tt> <tt class="py-string">'Data Collection Started Time'</tt><tt class="py-op">,</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">    <tt class="py-string">'RUNT4'</tt><tt class="py-op">:</tt> <tt class="py-string">'Data Collection Stopped Time'</tt><tt class="py-op">,</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">    <tt class="py-string">'Rate1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Scanning Rate. Milliseconds per frame.'</tt><tt class="py-op">,</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">    <tt class="py-string">'RunN1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Run Name'</tt><tt class="py-op">,</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">    <tt class="py-string">'SCAN1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Number of scans'</tt><tt class="py-op">,</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">    <tt class="py-string">'SMED1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Polymer lot expiration date'</tt><tt class="py-op">,</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">    <tt class="py-string">'SMLt1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Polymer lot number'</tt><tt class="py-op">,</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">    <tt class="py-string">'SMPL1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Sample name'</tt><tt class="py-op">,</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">    <tt class="py-string">'SVER1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Data collection software version'</tt><tt class="py-op">,</tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">    <tt class="py-string">'SVER3'</tt><tt class="py-op">:</tt> <tt class="py-string">'Data collection firmware version'</tt><tt class="py-op">,</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">    <tt class="py-string">'Satd1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Array of longs representing the scan numbers of data points, which are flagged as saturated by data collection (optional)'</tt><tt class="py-op">,</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">    <tt class="py-string">'Scal1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Rescaling divisor for color data'</tt><tt class="py-op">,</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">    <tt class="py-string">'Scan1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Number of scans (legacy - use SCAN)'</tt><tt class="py-op">,</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">    <tt class="py-string">'TUBE1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Well ID'</tt><tt class="py-op">,</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line">    <tt class="py-string">'Tmpr1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Run temperature setting'</tt><tt class="py-op">,</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line">    <tt class="py-string">'User1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Name of user who created the plate (optional)'</tt><tt class="py-op">,</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line"><tt class="py-op">}</tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line"> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line"><tt class="py-comment">#  No instrument specific tags</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line"><tt class="py-comment"># _INSTRUMENT_SPECIFIC_TAGS['abi_prism_3100/3100-Avant'] = {</tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line"><tt class="py-comment"># }</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line"> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line"><tt id="link-21" class="py-name"><a title="Bio.SeqIO.AbiIO._INSTRUMENT_SPECIFIC_TAGS" class="py-name" href="#" onclick="return doclink('link-21', '_INSTRUMENT_SPECIFIC_TAGS', 'link-19');">_INSTRUMENT_SPECIFIC_TAGS</a></tt><tt class="py-op">[</tt><tt class="py-string">'abi_3130/3130xl'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">    <tt class="py-string">'CTOw1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Container owner'</tt><tt class="py-op">,</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">    <tt class="py-string">'HCFG1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Instrument Class'</tt><tt class="py-op">,</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">    <tt class="py-string">'HCFG2'</tt><tt class="py-op">:</tt> <tt class="py-string">'Instrument Family'</tt><tt class="py-op">,</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">    <tt class="py-string">'HCFG3'</tt><tt class="py-op">:</tt> <tt class="py-string">'Official Instrument Name'</tt><tt class="py-op">,</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">    <tt class="py-string">'HCFG4'</tt><tt class="py-op">:</tt> <tt class="py-string">'Instrument Parameters'</tt><tt class="py-op">,</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">    <tt class="py-string">'RMdVa1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Run Module version'</tt><tt class="py-op">,</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line"><tt class="py-op">}</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line"> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line"><tt id="link-22" class="py-name"><a title="Bio.SeqIO.AbiIO._INSTRUMENT_SPECIFIC_TAGS" class="py-name" href="#" onclick="return doclink('link-22', '_INSTRUMENT_SPECIFIC_TAGS', 'link-19');">_INSTRUMENT_SPECIFIC_TAGS</a></tt><tt class="py-op">[</tt><tt class="py-string">'abi_3530/3530xl'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">    <tt class="py-string">'AAct1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Primary Analysis Audit Active indication. True if system auditing was enabled during the last write of this file, '</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">             <tt class="py-string">'false if system auditing was disabled.'</tt><tt class="py-op">,</tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">    <tt class="py-string">'ABED1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Anode buffer expiration date using ISO 8601 format using the patterns YYYY-MM-DDTHH:MM:SS.ss+/-HH:MM. Hundredths of a second are optional.'</tt><tt class="py-op">,</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">    <tt class="py-string">'ABID1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Anode buffer tray first installed date'</tt><tt class="py-op">,</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">    <tt class="py-string">'ABLt1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Anode buffer lot number'</tt><tt class="py-op">,</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">    <tt class="py-string">'ABRn1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Number of runs (injections) processed with the current Anode Buffer (runs allowed - runs remaining)'</tt><tt class="py-op">,</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">    <tt class="py-string">'ABTp1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Anode buffer type'</tt><tt class="py-op">,</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">    <tt class="py-string">'AEPt1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Analysis Ending scan number for basecalling on initial analysis'</tt><tt class="py-op">,</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">    <tt class="py-string">'AEPt2'</tt><tt class="py-op">:</tt> <tt class="py-string">'Analysis Ending scan number for basecalling on last analysis'</tt><tt class="py-op">,</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">    <tt class="py-string">'APCN1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Amplicon name'</tt><tt class="py-op">,</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">    <tt class="py-string">'ARTN1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Analysis Return code. Produced only by 5 Prime basecaller 1.0b3'</tt><tt class="py-op">,</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line">    <tt class="py-string">'ASPF1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Flag to indicate whether adaptive processing worked or not'</tt><tt class="py-op">,</tt> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line">    <tt class="py-string">'ASPt1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Analysis Starting scan number for first analysis'</tt><tt class="py-op">,</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line">    <tt class="py-string">'ASPt2'</tt><tt class="py-op">:</tt> <tt class="py-string">'Analysis Starting scan number for last analysis'</tt><tt class="py-op">,</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line">    <tt class="py-string">'AUDT2'</tt><tt class="py-op">:</tt> <tt class="py-string">'Audit log used across 3500 software (optional)'</tt><tt class="py-op">,</tt> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line">    <tt class="py-string">'AVld1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Assay validation flag (true or false)'</tt><tt class="py-op">,</tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line">    <tt class="py-string">'AmbT1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Record of ambient temperature readings'</tt><tt class="py-op">,</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line">    <tt class="py-string">'AsyC1'</tt><tt class="py-op">:</tt> <tt class="py-string">'The assay contents (xml format)'</tt><tt class="py-op">,</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line">    <tt class="py-string">'AsyN1'</tt><tt class="py-op">:</tt> <tt class="py-string">'The assay name'</tt><tt class="py-op">,</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line">    <tt class="py-string">'AsyV1'</tt><tt class="py-op">:</tt> <tt class="py-string">'The assay version'</tt><tt class="py-op">,</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line">    <tt class="py-string">'B1Pt1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Reference scan number for mobility and spacing curves for first analysis'</tt><tt class="py-op">,</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line">    <tt class="py-string">'B1Pt2'</tt><tt class="py-op">:</tt> <tt class="py-string">'Reference scan number for mobility and spacing curves for last analysis'</tt><tt class="py-op">,</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line">    <tt class="py-string">'BCTS1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Basecaller timestamp. Time of completion of most recent analysis'</tt><tt class="py-op">,</tt> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line">    <tt class="py-string">'BcRn1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Basecalling qc code'</tt><tt class="py-op">,</tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line">    <tt class="py-string">'BcRs1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Basecalling warnings, a concatenated comma separated string'</tt><tt class="py-op">,</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line">    <tt class="py-string">'BcRs2'</tt><tt class="py-op">:</tt> <tt class="py-string">'Basecalling errors, a concatenated comma separated string'</tt><tt class="py-op">,</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line">    <tt class="py-string">'CAED1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Capillary array expiration'</tt><tt class="py-op">,</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line">    <tt class="py-string">'CALt1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Capillary array lot number'</tt><tt class="py-op">,</tt> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">    <tt class="py-string">'CARn1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Number of injections processed (including the one of which this sample was a part) through the capillary array'</tt><tt class="py-op">,</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line">    <tt class="py-string">'CASN1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Capillary array serial number'</tt><tt class="py-op">,</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line">    <tt class="py-string">'CBED1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Cathode buffer expiration date'</tt><tt class="py-op">,</tt> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line">    <tt class="py-string">'CBID1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Cathode buffer tray first installed date'</tt><tt class="py-op">,</tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line">    <tt class="py-string">'CBLt1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Cathode buffer lot number'</tt><tt class="py-op">,</tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">    <tt class="py-string">'CBRn1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Number of runs (injections) processed with the current Cathode Buffer (runs allowed - runs remaining)'</tt><tt class="py-op">,</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line">    <tt class="py-string">'CBTp1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Cathode buffer type'</tt><tt class="py-op">,</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line">    <tt class="py-string">'CLRG1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Start of the clear range (inclusive).'</tt><tt class="py-op">,</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line">    <tt class="py-string">'CLRG2'</tt><tt class="py-op">:</tt> <tt class="py-string">'Clear range length'</tt><tt class="py-op">,</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line">    <tt class="py-string">'CRLn1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Contiguous read length'</tt><tt class="py-op">,</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line">    <tt class="py-string">'CRLn2'</tt><tt class="py-op">:</tt> <tt class="py-string">'One of "Pass", "Fail", or "Check"'</tt><tt class="py-op">,</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line">    <tt class="py-string">'CTOw1'</tt><tt class="py-op">:</tt> <tt class="py-string">'The name entered as the Owner of a plate, in the plate editor'</tt><tt class="py-op">,</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line">    <tt class="py-string">'CkSm1'</tt><tt class="py-op">:</tt> <tt class="py-string">'File checksum'</tt><tt class="py-op">,</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line">    <tt class="py-string">'DCEv1'</tt><tt class="py-op">:</tt> <tt class="py-string">'A list of door-close events, separated by semicolon. Door open events are generally paired with door close events.'</tt><tt class="py-op">,</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">    <tt class="py-string">'DCHT1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Reserved for backward compatibility. The detection cell heater temperature setting from the Run Module. Not used for 3500.'</tt><tt class="py-op">,</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line">    <tt class="py-string">'DOEv1'</tt><tt class="py-op">:</tt> <tt class="py-string">'A list of door-open events, separated by semicolon. Door close events are generally paired with door open events.'</tt><tt class="py-op">,</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line">    <tt class="py-string">'ESig2'</tt><tt class="py-op">:</tt> <tt class="py-string">'Electronic signature record used across 3500 software'</tt><tt class="py-op">,</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">    <tt class="py-string">'FTab1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Feature table. Can be created by Nibbler for Clear Range.'</tt><tt class="py-op">,</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">    <tt class="py-string">'FVoc1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Feature table vocabulary. Can be created by Nibbler for Clear Range.'</tt><tt class="py-op">,</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line">    <tt class="py-string">'Feat1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Features. Can be created by Nibbler for Clear Range.'</tt><tt class="py-op">,</tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line">    <tt class="py-string">'HCFG1'</tt><tt class="py-op">:</tt> <tt class="py-string">'The Instrument Class. All upper case, no spaces. Initial valid value: CE'</tt><tt class="py-op">,</tt> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line">    <tt class="py-string">'HCFG2'</tt><tt class="py-op">:</tt> <tt class="py-string">'The Instrument Family. All upper case, no spaces. Valid values: 31XX or 37XX for UDC, 35XX (for 3500)'</tt><tt class="py-op">,</tt> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line">    <tt class="py-string">'HCFG3'</tt><tt class="py-op">:</tt> <tt class="py-string">'The official instrument name. Mixed case, minus any special formatting. Initial valid values: 3130, 3130xl, 3730, 3730xl, 3500, 3500xl.'</tt><tt class="py-op">,</tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line">    <tt class="py-string">'HCFG4'</tt><tt class="py-op">:</tt> <tt class="py-string">'Instrument parameters. Contains key-value pairs of instrument configuration information, separated by semicolons. '</tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line">             <tt class="py-string">'Four parameters are included initially: UnitID=&lt;UNITD number&gt;, CPUBoard=&lt;board type&gt;, '</tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line">             <tt class="py-string">'ArraySize=&lt;# of capillaries&gt;, SerialNumber=&lt;Instrument Serial#&gt;.'</tt><tt class="py-op">,</tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line">    <tt class="py-string">'InjN1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Injection name'</tt><tt class="py-op">,</tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line">    <tt class="py-string">'LAST1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Parameter settings information'</tt><tt class="py-op">,</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line">    <tt class="py-string">'NOIS1'</tt><tt class="py-op">:</tt> <tt class="py-string">'The estimate of rms baseline noise (S/N ratio) for each dye for a successfully analyzed sample. '</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line">             <tt class="py-string">'Corresponds in order to the raw data in tags DATA 1-4. KB basecaller only.'</tt><tt class="py-op">,</tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line">    <tt class="py-string">'P1AM1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Amplitude of primary peak, which is not necessarily equal to corresponding signal strength at that position'</tt><tt class="py-op">,</tt> </tt>
<a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line">    <tt class="py-string">'P1RL1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Deviation of primary peak position from (PLoc,2), times 100, rounded to integer'</tt><tt class="py-op">,</tt> </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line">    <tt class="py-string">'P1WD1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Full-width Half-max of primary peak, times 100, rounded to integer. '</tt> </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line">             <tt class="py-string">'Corresponding signal intensity is not necessarily equal to one half of primary peak amplitude'</tt><tt class="py-op">,</tt> </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line">    <tt class="py-string">'P2AM1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Amplitude of secondary peak, which is not necessarily equal to corresponding signal strength at that position'</tt><tt class="py-op">,</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line">    <tt class="py-string">'P2BA1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Base of secondary peak'</tt><tt class="py-op">,</tt> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line">    <tt class="py-string">'P2RL1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Deviation of secondary peak position from (PLoc,2), times 100, rounded to integer'</tt><tt class="py-op">,</tt> </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line">    <tt class="py-string">'PBAS1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Array of sequence characters edited by user'</tt><tt class="py-op">,</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line">    <tt class="py-string">'PBAS2'</tt><tt class="py-op">:</tt> <tt class="py-string">'Array of sequence characters as called by Basecaller'</tt><tt class="py-op">,</tt> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line">    <tt class="py-string">'PCON1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Array of quality Values (0-255) as edited by user'</tt><tt class="py-op">,</tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line">    <tt class="py-string">'PCON2'</tt><tt class="py-op">:</tt> <tt class="py-string">'Array of quality values (0-255) as called by Basecaller'</tt><tt class="py-op">,</tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line">    <tt class="py-string">'PDMF2'</tt><tt class="py-op">:</tt> <tt class="py-string">'Mobility file name chosen in most recent analysis (identical to PDMF1)'</tt><tt class="py-op">,</tt> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line">    <tt class="py-string">'PLOC1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Array of peak locations edited by user'</tt><tt class="py-op">,</tt> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line">    <tt class="py-string">'PLOC2'</tt><tt class="py-op">:</tt> <tt class="py-string">'Array of peak locations as called by Basecaller'</tt><tt class="py-op">,</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line">    <tt class="py-string">'PRJT1'</tt><tt class="py-op">:</tt> <tt class="py-string">'SeqScape 2.0 project template name'</tt><tt class="py-op">,</tt> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line">    <tt class="py-string">'PROJ4'</tt><tt class="py-op">:</tt> <tt class="py-string">'SeqScape 2.0 project name'</tt><tt class="py-op">,</tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line">    <tt class="py-string">'PSZE1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Plate size. The number of sample positions in the container. Current allowed values: 96, 384.'</tt><tt class="py-op">,</tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line">    <tt class="py-string">'PTYP1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Plate type. Current allowed values: 96-Well, 384-Well.'</tt><tt class="py-op">,</tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line">    <tt class="py-string">'PuSc1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Median pupscore'</tt><tt class="py-op">,</tt> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line">    <tt class="py-string">'QV201'</tt><tt class="py-op">:</tt> <tt class="py-string">'QV20+ value'</tt><tt class="py-op">,</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line">    <tt class="py-string">'QV202'</tt><tt class="py-op">:</tt> <tt class="py-string">'One of "Pass", "Fail", or "Check"'</tt><tt class="py-op">,</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">    <tt class="py-string">'QcPa1'</tt><tt class="py-op">:</tt> <tt class="py-string">'QC parameters'</tt><tt class="py-op">,</tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">    <tt class="py-string">'QcRn1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Trimming and QC code'</tt><tt class="py-op">,</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line">    <tt class="py-string">'QcRs1'</tt><tt class="py-op">:</tt> <tt class="py-string">'QC warnings, a concatenated comma separated string'</tt><tt class="py-op">,</tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">    <tt class="py-string">'QcRs2'</tt><tt class="py-op">:</tt> <tt class="py-string">'QC errors, a concatenated comma separated string'</tt><tt class="py-op">,</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">    <tt class="py-string">'RGOw1'</tt><tt class="py-op">:</tt> <tt class="py-string">'The name entered as the Owner of a Results Group, in the Results Group Editor. Implemented as the user name from the results group.'</tt><tt class="py-op">,</tt> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line">    <tt class="py-string">'RInj1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Reinjection number. The reinjection number that this sample belongs to. Not present if there was no reinjection.'</tt><tt class="py-op">,</tt> </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line">    <tt class="py-string">'RNmF1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Raman normalization factor'</tt><tt class="py-op">,</tt> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line">    <tt class="py-string">'RevC1'</tt><tt class="py-op">:</tt> <tt class="py-string">'for whether the sequence has been complemented'</tt><tt class="py-op">,</tt> </tt>
<a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line">    <tt class="py-string">'RunN1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Run name (which, for 3500, is different from injection name)'</tt><tt class="py-op">,</tt> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line">    <tt class="py-string">'S/N%1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Signal strength for each dye'</tt><tt class="py-op">,</tt> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line">    <tt class="py-string">'SMID1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Polymer first installed date'</tt><tt class="py-op">,</tt> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line">    <tt class="py-string">'SMRn1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Number of runs (injections) processed with the current polymer (runs allowed - runs remaining)'</tt><tt class="py-op">,</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line">    <tt class="py-string">'SPAC1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Average peak spacing used in last analysis'</tt><tt class="py-op">,</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line">    <tt class="py-string">'SPAC2'</tt><tt class="py-op">:</tt> <tt class="py-string">'Basecaller name - corresponds to name of bcp file.'</tt><tt class="py-op">,</tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line">    <tt class="py-string">'SPAC3'</tt><tt class="py-op">:</tt> <tt class="py-string">'Average peak spacing last calculated by the Basecaller.'</tt><tt class="py-op">,</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line">    <tt class="py-string">'SPEC1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Sequencing Analysis Specimen Name'</tt><tt class="py-op">,</tt> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line">    <tt class="py-string">'SVER2'</tt><tt class="py-op">:</tt> <tt class="py-string">'Basecaller version number'</tt><tt class="py-op">,</tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line">    <tt class="py-string">'SVER4'</tt><tt class="py-op">:</tt> <tt class="py-string">'Sample File Format Version String'</tt><tt class="py-op">,</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">    <tt class="py-string">'ScPa1'</tt><tt class="py-op">:</tt> <tt class="py-string">'The parameter string of size caller'</tt><tt class="py-op">,</tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line">    <tt class="py-string">'ScSt1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Raw data start point. Set to 0 for 3500 data collection.'</tt><tt class="py-op">,</tt> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line">    <tt class="py-string">'SpeN1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Active spectral calibration name'</tt><tt class="py-op">,</tt> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">    <tt class="py-string">'TrPa1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Timming parameters'</tt><tt class="py-op">,</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">    <tt class="py-string">'TrSc1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Trace score.'</tt><tt class="py-op">,</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">    <tt class="py-string">'TrSc2'</tt><tt class="py-op">:</tt> <tt class="py-string">'One of "Pass", "Fail", or "Check"'</tt><tt class="py-op">,</tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">    <tt class="py-string">'phAR1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Trace peak aria ratio'</tt><tt class="py-op">,</tt> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line">    <tt class="py-string">'phCH1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Chemistry type ("term", "prim", "unknown"), based on DYE_1 information'</tt><tt class="py-op">,</tt> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line">    <tt class="py-string">'phDY1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Dye ("big", "d-rhod", "unknown"), based on mob file information'</tt><tt class="py-op">,</tt> </tt>
<a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line">    <tt class="py-string">'phQL1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Maximum Quality Value'</tt><tt class="py-op">,</tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line">    <tt class="py-string">'phTR1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Set Trim region'</tt><tt class="py-op">,</tt> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line">    <tt class="py-string">'phTR2'</tt><tt class="py-op">:</tt> <tt class="py-string">'Trim probability'</tt><tt class="py-op">,</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line"><tt class="py-op">}</tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line"> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line"><tt id="link-23" class="py-name"><a title="Bio.SeqIO.AbiIO._INSTRUMENT_SPECIFIC_TAGS" class="py-name" href="#" onclick="return doclink('link-23', '_INSTRUMENT_SPECIFIC_TAGS', 'link-19');">_INSTRUMENT_SPECIFIC_TAGS</a></tt><tt class="py-op">[</tt><tt class="py-string">'abi_3730/3730xl'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line">    <tt class="py-string">'BufT1'</tt><tt class="py-op">:</tt> <tt class="py-string">'Buffer tray heater temperature (degrees C)'</tt><tt class="py-op">,</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line"><tt class="py-op">}</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line"> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line"><tt class="py-comment"># dictionary for data unpacking format</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line"><tt id="link-24" class="py-name" targets="Variable Bio.SeqIO.AbiIO._BYTEFMT=Bio.SeqIO.AbiIO-module.html#_BYTEFMT"><a title="Bio.SeqIO.AbiIO._BYTEFMT" class="py-name" href="#" onclick="return doclink('link-24', '_BYTEFMT', 'link-24');">_BYTEFMT</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line">    <tt class="py-number">1</tt><tt class="py-op">:</tt> <tt class="py-string">'b'</tt><tt class="py-op">,</tt>     <tt class="py-comment"># byte</tt> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line">    <tt class="py-number">2</tt><tt class="py-op">:</tt> <tt class="py-string">'s'</tt><tt class="py-op">,</tt>     <tt class="py-comment"># char</tt> </tt>
<a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line">    <tt class="py-number">3</tt><tt class="py-op">:</tt> <tt class="py-string">'H'</tt><tt class="py-op">,</tt>     <tt class="py-comment"># word</tt> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">    <tt class="py-number">4</tt><tt class="py-op">:</tt> <tt class="py-string">'h'</tt><tt class="py-op">,</tt>     <tt class="py-comment"># short</tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line">    <tt class="py-number">5</tt><tt class="py-op">:</tt> <tt class="py-string">'i'</tt><tt class="py-op">,</tt>     <tt class="py-comment"># long</tt> </tt>
<a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line">    <tt class="py-number">6</tt><tt class="py-op">:</tt> <tt class="py-string">'2i'</tt><tt class="py-op">,</tt>    <tt class="py-comment"># rational, legacy unsupported</tt> </tt>
<a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line">    <tt class="py-number">7</tt><tt class="py-op">:</tt> <tt class="py-string">'f'</tt><tt class="py-op">,</tt>     <tt class="py-comment"># float</tt> </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line">    <tt class="py-number">8</tt><tt class="py-op">:</tt> <tt class="py-string">'d'</tt><tt class="py-op">,</tt>     <tt class="py-comment"># double</tt> </tt>
<a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line">    <tt class="py-number">10</tt><tt class="py-op">:</tt> <tt class="py-string">'h2B'</tt><tt class="py-op">,</tt>  <tt class="py-comment"># date</tt> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line">    <tt class="py-number">11</tt><tt class="py-op">:</tt> <tt class="py-string">'4B'</tt><tt class="py-op">,</tt>   <tt class="py-comment"># time</tt> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line">    <tt class="py-number">12</tt><tt class="py-op">:</tt> <tt class="py-string">'2i2b'</tt><tt class="py-op">,</tt>  <tt class="py-comment"># thumb</tt> </tt>
<a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line">    <tt class="py-number">13</tt><tt class="py-op">:</tt> <tt class="py-string">'B'</tt><tt class="py-op">,</tt>    <tt class="py-comment"># bool</tt> </tt>
<a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line">    <tt class="py-number">14</tt><tt class="py-op">:</tt> <tt class="py-string">'2h'</tt><tt class="py-op">,</tt>   <tt class="py-comment"># point, legacy unsupported</tt> </tt>
<a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line">    <tt class="py-number">15</tt><tt class="py-op">:</tt> <tt class="py-string">'4h'</tt><tt class="py-op">,</tt>   <tt class="py-comment"># rect, legacy unsupported</tt> </tt>
<a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">    <tt class="py-number">16</tt><tt class="py-op">:</tt> <tt class="py-string">'2i'</tt><tt class="py-op">,</tt>   <tt class="py-comment"># vPoint, legacy unsupported</tt> </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line">    <tt class="py-number">17</tt><tt class="py-op">:</tt> <tt class="py-string">'4i'</tt><tt class="py-op">,</tt>   <tt class="py-comment"># vRect, legacy unsupported</tt> </tt>
<a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line">    <tt class="py-number">18</tt><tt class="py-op">:</tt> <tt class="py-string">'s'</tt><tt class="py-op">,</tt>    <tt class="py-comment"># pString</tt> </tt>
<a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line">    <tt class="py-number">19</tt><tt class="py-op">:</tt> <tt class="py-string">'s'</tt><tt class="py-op">,</tt>    <tt class="py-comment"># cString</tt> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">    <tt class="py-number">20</tt><tt class="py-op">:</tt> <tt class="py-string">'2i'</tt><tt class="py-op">,</tt>   <tt class="py-comment"># tag, legacy unsupported</tt> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line"><tt class="py-op">}</tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line"><tt class="py-comment"># header data structure (exluding 4 byte ABIF marker)</tt> </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line"><tt id="link-25" class="py-name" targets="Variable Bio.SeqIO.AbiIO._HEADFMT=Bio.SeqIO.AbiIO-module.html#_HEADFMT"><a title="Bio.SeqIO.AbiIO._HEADFMT" class="py-name" href="#" onclick="return doclink('link-25', '_HEADFMT', 'link-25');">_HEADFMT</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'&gt;H4sI2H3I'</tt> </tt>
<a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line"><tt class="py-comment"># directory data structure</tt> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line"><tt id="link-26" class="py-name" targets="Variable Bio.SeqIO.AbiIO._DIRFMT=Bio.SeqIO.AbiIO-module.html#_DIRFMT"><a title="Bio.SeqIO.AbiIO._DIRFMT" class="py-name" href="#" onclick="return doclink('link-26', '_DIRFMT', 'link-26');">_DIRFMT</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'&gt;4sI2H4I'</tt> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line"> </tt>
<a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line"><tt id="link-27" class="py-name" targets="Variable Bio.SeqIO.AbiIO.__global_tag_listing=Bio.SeqIO.AbiIO-module.html#__global_tag_listing"><a title="Bio.SeqIO.AbiIO.__global_tag_listing" class="py-name" href="#" onclick="return doclink('link-27', '__global_tag_listing', 'link-27');">__global_tag_listing</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line"><tt class="py-keyword">for</tt> <tt id="link-28" class="py-name" targets="Variable Bio.SeqIO.AbiIO.tag=Bio.SeqIO.AbiIO-module.html#tag"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-28', 'tag', 'link-28');">tag</a></tt> <tt class="py-keyword">in</tt> <tt id="link-29" class="py-name"><a title="Bio.SeqIO.AbiIO._INSTRUMENT_SPECIFIC_TAGS" class="py-name" href="#" onclick="return doclink('link-29', '_INSTRUMENT_SPECIFIC_TAGS', 'link-19');">_INSTRUMENT_SPECIFIC_TAGS</a></tt><tt class="py-op">.</tt><tt id="link-30" class="py-name" targets="Method Bio.Crystal.Crystal.values()=Bio.Crystal.Crystal-class.html#values,Method Bio.File._IndexedSeqFileDict.values()=Bio.File._IndexedSeqFileDict-class.html#values,Method Bio.Phylo.PhyloXML.Events.values()=Bio.Phylo.PhyloXML.Events-class.html#values,Method BioSQL.BioSeqDatabase.BioSeqDatabase.values()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#values,Method BioSQL.BioSeqDatabase.DBServer.values()=BioSQL.BioSeqDatabase.DBServer-class.html#values"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-30', 'values', 'link-30');">values</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line">    <tt id="link-31" class="py-name"><a title="Bio.SeqIO.AbiIO.__global_tag_listing" class="py-name" href="#" onclick="return doclink('link-31', '__global_tag_listing', 'link-27');">__global_tag_listing</a></tt> <tt class="py-op">+=</tt> <tt id="link-32" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-32', 'tag', 'link-28');">tag</a></tt><tt class="py-op">.</tt><tt id="link-33" class="py-name" targets="Method Bio.Crystal.Crystal.keys()=Bio.Crystal.Crystal-class.html#keys,Variable Bio.Entrez.Parser.ListConsumer.keys=Bio.Entrez.Parser.ListConsumer-class.html#keys,Method Bio.File._IndexedSeqFileDict.keys()=Bio.File._IndexedSeqFileDict-class.html#keys,Method Bio.File._SQLiteManySeqFilesDict.keys()=Bio.File._SQLiteManySeqFilesDict-class.html#keys,Method Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys()=Bio.PDB.AbstractPropertyMap.AbstractPropertyMap-class.html#keys,Method Bio.Phylo.PhyloXML.Events.keys()=Bio.Phylo.PhyloXML.Events-class.html#keys,Method BioSQL.BioSeqDatabase.BioSeqDatabase.keys()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#keys,Method BioSQL.BioSeqDatabase.DBServer.keys()=BioSQL.BioSeqDatabase.DBServer-class.html#keys"><a title="Bio.Crystal.Crystal.keys
Bio.Entrez.Parser.ListConsumer.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-33', 'keys', 'link-33');">keys</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line"> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line"> </tt>
<a name="AbiIterator"></a><div id="AbiIterator-def"><a name="L334"></a><tt class="py-lineno">334</tt> <a class="py-toggle" href="#" id="AbiIterator-toggle" onclick="return toggle('AbiIterator');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.AbiIO-module.html#AbiIterator">AbiIterator</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">trim</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbiIterator-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="AbiIterator-expanded"><a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return an iterator for the Abi file format."""</tt> </tt>
<a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line">    <tt class="py-comment"># raise exception is alphabet is not dna</tt> </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt id="link-34" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.alphabet=Bio.SearchIO._model.hsp.HSP-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSPFragment.alphabet=Bio.SearchIO._model.hsp.HSPFragment-class.html#alphabet"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-34', 'alphabet', 'link-34');">alphabet</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-35" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-35', 'Alphabet', 'link-1');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-36" class="py-name" targets="Function Bio.Alphabet._get_base_alphabet()=Bio.Alphabet-module.html#_get_base_alphabet"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-36', '_get_base_alphabet', 'link-36');">_get_base_alphabet</a></tt><tt class="py-op">(</tt><tt id="link-37" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-37', 'alphabet', 'link-34');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line">                      <tt id="link-38" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-38', 'Alphabet', 'link-1');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-39" class="py-name" targets="Class Bio.Alphabet.ProteinAlphabet=Bio.Alphabet.ProteinAlphabet-class.html"><a title="Bio.Alphabet.ProteinAlphabet" class="py-name" href="#" onclick="return doclink('link-39', 'ProteinAlphabet', 'link-39');">ProteinAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt> </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line">                <tt class="py-string">"Invalid alphabet, ABI files do not hold proteins."</tt><tt class="py-op">)</tt> </tt>
<a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-40" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-40', 'Alphabet', 'link-1');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-41" class="py-name"><a title="Bio.Alphabet._get_base_alphabet" class="py-name" href="#" onclick="return doclink('link-41', '_get_base_alphabet', 'link-36');">_get_base_alphabet</a></tt><tt class="py-op">(</tt><tt id="link-42" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-42', 'alphabet', 'link-34');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line">                      <tt id="link-43" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-43', 'Alphabet', 'link-1');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-44" class="py-name" targets="Class Bio.Alphabet.RNAAlphabet=Bio.Alphabet.RNAAlphabet-class.html"><a title="Bio.Alphabet.RNAAlphabet" class="py-name" href="#" onclick="return doclink('link-44', 'RNAAlphabet', 'link-44');">RNAAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid alphabet, ABI files do not hold RNA."</tt><tt class="py-op">)</tt> </tt>
<a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line"> </tt>
<a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line">    <tt class="py-comment"># raise exception if handle mode is not 'rb'</tt> </tt>
<a name="L347"></a><tt class="py-lineno">347</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-string">'mode'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-45" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.set()=Bio.Nexus.Nexus.StepMatrix-class.html#set,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.set()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#set,Method Bio.SVDSuperimposer.SVDSuperimposer.set()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#set"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-45', 'set', 'link-45');">set</a></tt><tt class="py-op">(</tt><tt class="py-string">'rb'</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt id="link-46" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-46', 'set', 'link-45');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt class="py-name">mode</tt><tt class="py-op">.</tt><tt id="link-47" class="py-name" targets="Method Bio.Seq.Seq.lower()=Bio.Seq.Seq-class.html#lower,Method Bio.Seq.UnknownSeq.lower()=Bio.Seq.UnknownSeq-class.html#lower,Method Bio.SeqRecord.SeqRecord.lower()=Bio.SeqRecord.SeqRecord-class.html#lower"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-47', 'lower', 'link-47');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"ABI files has to be opened in 'rb' mode."</tt><tt class="py-op">)</tt> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line"> </tt>
<a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line">    <tt class="py-comment"># check if input file is a valid Abi file</tt> </tt>
<a name="L352"></a><tt class="py-lineno">352</tt>  <tt class="py-line">    <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-48" class="py-name" targets="Method Bio.File.UndoHandle.seek()=Bio.File.UndoHandle-class.html#seek,Method Bio.SeqIO.SffIO._AddTellHandle.seek()=Bio.SeqIO.SffIO._AddTellHandle-class.html#seek,Method Bio.bgzf.BgzfReader.seek()=Bio.bgzf.BgzfReader-class.html#seek"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-48', 'seek', 'link-48');">seek</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line">    <tt class="py-name">marker</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-49" class="py-name" targets="Function Bio.Affy.CelFile.read()=Bio.Affy.CelFile-module.html#read,Function Bio.AlignIO.read()=Bio.AlignIO-module.html#read,Function Bio.Blast.NCBIXML.read()=Bio.Blast.NCBIXML-module.html#read,Function Bio.Cluster.read()=Bio.Cluster-module.html#read,Function Bio.Compass.read()=Bio.Compass-module.html#read,Function Bio.Emboss.Primer3.read()=Bio.Emboss.Primer3-module.html#read,Function Bio.Emboss.PrimerSearch.read()=Bio.Emboss.PrimerSearch-module.html#read,Method Bio.Entrez.Parser.DataHandler.read()=Bio.Entrez.Parser.DataHandler-class.html#read,Function Bio.Entrez.read()=Bio.Entrez-module.html#read,Function Bio.ExPASy.Enzyme.read()=Bio.ExPASy.Enzyme-module.html#read,Function Bio.ExPASy.Prodoc.read()=Bio.ExPASy.Prodoc-module.html#read,Function Bio.ExPASy.Prosite.read()=Bio.ExPASy.Prosite-module.html#read,Function Bio.ExPASy.ScanProsite.read()=Bio.ExPASy.ScanProsite-module.html#read,Function Bio.ExPASy.cellosaurus.read()=Bio.ExPASy.cellosaurus-module.html#read,Method Bio.File.UndoHandle.read()=Bio.File.UndoHandle-class.html#read,Function Bio.KEGG.Enzyme.read()=Bio.KEGG.Enzyme-module.html#read,Function Bio.KEGG.KGML.KGML_parser.read()=Bio.KEGG.KGML.KGML_parser-module.html#read,Function Bio.Medline.read()=Bio.Medline-module.html#read,Method Bio.Nexus.Nexus.Nexus.read()=Bio.Nexus.Nexus.Nexus-class.html#read,Function Bio.Phylo.PAML.baseml.read()=Bio.Phylo.PAML.baseml-module.html#read,Function Bio.Phylo.PAML.codeml.read()=Bio.Phylo.PAML.codeml-module.html#read,Function Bio.Phylo.PAML.yn00.read()=Bio.Phylo.PAML.yn00-module.html#read,Method Bio.Phylo.PhyloXMLIO.Parser.read()=Bio.Phylo.PhyloXMLIO.Parser-class.html#read,Function Bio.Phylo.PhyloXMLIO.read()=Bio.Phylo.PhyloXMLIO-module.html#read,Function Bio.Phylo._io.read()=Bio.Phylo._io-module.html#read,Function Bio.PopGen.GenePop.FileParser.read()=Bio.PopGen.GenePop.FileParser-module.html#read,Function Bio.PopGen.GenePop.LargeFileParser.read()=Bio.PopGen.GenePop.LargeFileParser-module.html#read,Function Bio.PopGen.GenePop.read()=Bio.PopGen.GenePop-module.html#read,Function Bio.SearchIO.read()=Bio.SearchIO-module.html#read,Method Bio.SeqIO.SffIO._AddTellHandle.read()=Bio.SeqIO.SffIO._AddTellHandle-class.html#read,Function Bio.SeqIO.read()=Bio.SeqIO-module.html#read,Function Bio.Sequencing.Ace.read()=Bio.Sequencing.Ace-module.html#read,Function Bio.Sequencing.Phd.read()=Bio.Sequencing.Phd-module.html#read,Function Bio.SwissProt.read()=Bio.SwissProt-module.html#read,Function Bio.UniGene.read()=Bio.UniGene-module.html#read,Method Bio.bgzf.BgzfReader.read()=Bio.bgzf.BgzfReader-class.html#read,Function Bio.motifs.alignace.read()=Bio.motifs.alignace-module.html#read,Function Bio.motifs.jaspar.read()=Bio.motifs.jaspar-module.html#read,Function Bio.motifs.mast.read()=Bio.motifs.mast-module.html#read,Function Bio.motifs.meme.read()=Bio.motifs.meme-module.html#read,Function Bio.motifs.minimal.read()=Bio.motifs.minimal-module.html#read,Function Bio.motifs.read()=Bio.motifs-module.html#read,Function Bio.motifs.transfac.read()=Bio.motifs.transfac-module.html#read,Function Bio.phenotype.read()=Bio.phenotype-module.html#read"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.ExPASy.cellosaurus.read
Bio.File.UndoHandle.read
Bio.KEGG.Enzyme.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.minimal.read
Bio.motifs.read
Bio.motifs.transfac.read
Bio.phenotype.read" class="py-name" href="#" onclick="return doclink('link-49', 'read', 'link-49');">read</a></tt><tt class="py-op">(</tt><tt class="py-number">4</tt><tt class="py-op">)</tt> </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">marker</tt><tt class="py-op">:</tt> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">        <tt class="py-comment"># handle empty file gracefully</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> </tt>
<a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">marker</tt> <tt class="py-op">!=</tt> <tt class="py-string">b"ABIF"</tt><tt class="py-op">:</tt> </tt>
<a name="L358"></a><tt class="py-lineno">358</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">IOError</tt><tt class="py-op">(</tt><tt class="py-string">'File should start ABIF, not %r'</tt> <tt class="py-op">%</tt> <tt class="py-name">marker</tt><tt class="py-op">)</tt> </tt>
<a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line"> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line">    <tt class="py-comment"># dirty hack for handling time information</tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line">    <tt class="py-name">times</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'RUND1'</tt><tt class="py-op">:</tt> <tt class="py-string">''</tt><tt class="py-op">,</tt> <tt class="py-string">'RUND2'</tt><tt class="py-op">:</tt> <tt class="py-string">''</tt><tt class="py-op">,</tt> <tt class="py-string">'RUNT1'</tt><tt class="py-op">:</tt> <tt class="py-string">''</tt><tt class="py-op">,</tt> <tt class="py-string">'RUNT2'</tt><tt class="py-op">:</tt> <tt class="py-string">''</tt><tt class="py-op">,</tt> <tt class="py-op">}</tt> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line"> </tt>
<a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line">    <tt class="py-comment"># initialize annotations</tt> </tt>
<a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line">    <tt class="py-name">annot</tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-name">zip</tt><tt class="py-op">(</tt><tt id="link-50" class="py-name"><a title="Bio.SeqIO.AbiIO._EXTRACT" class="py-name" href="#" onclick="return doclink('link-50', '_EXTRACT', 'link-18');">_EXTRACT</a></tt><tt class="py-op">.</tt><tt id="link-51" class="py-name"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-51', 'values', 'link-30');">values</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-name">None</tt><tt class="py-op">]</tt> <tt class="py-op">*</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-52" class="py-name"><a title="Bio.SeqIO.AbiIO._EXTRACT" class="py-name" href="#" onclick="return doclink('link-52', '_EXTRACT', 'link-18');">_EXTRACT</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line"> </tt>
<a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line">    <tt class="py-comment"># parse header and extract data from directories</tt> </tt>
<a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line">    <tt id="link-53" class="py-name" targets="Variable Bio.Wise.dnal.Statistics.header=Bio.Wise.dnal.Statistics-class.html#header"><a title="Bio.Wise.dnal.Statistics.header" class="py-name" href="#" onclick="return doclink('link-53', 'header', 'link-53');">header</a></tt> <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">unpack</tt><tt class="py-op">(</tt><tt id="link-54" class="py-name"><a title="Bio.SeqIO.AbiIO._HEADFMT" class="py-name" href="#" onclick="return doclink('link-54', '_HEADFMT', 'link-25');">_HEADFMT</a></tt><tt class="py-op">,</tt> </tt>
<a name="L368"></a><tt class="py-lineno">368</tt>  <tt class="py-line">                           <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-55" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.ExPASy.cellosaurus.read
Bio.File.UndoHandle.read
Bio.KEGG.Enzyme.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.minimal.read
Bio.motifs.read
Bio.motifs.transfac.read
Bio.phenotype.read" class="py-name" href="#" onclick="return doclink('link-55', 'read', 'link-49');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt id="link-56" class="py-name"><a title="Bio.SeqIO.AbiIO._HEADFMT" class="py-name" href="#" onclick="return doclink('link-56', '_HEADFMT', 'link-25');">_HEADFMT</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line"> </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line">    <tt class="py-comment"># Set default sample ID value, which we expect to be present in most cases</tt> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line">    <tt class="py-comment"># in the SMPL1 tag, but may be missing.</tt> </tt>
<a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line">    <tt class="py-name">sample_id</tt> <tt class="py-op">=</tt> <tt class="py-string">'&lt;unknown id&gt;'</tt> </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line"> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line">    <tt id="link-57" class="py-name" targets="Method Bio.Nexus.StandardData.StandardData.raw()=Bio.Nexus.StandardData.StandardData-class.html#raw"><a title="Bio.Nexus.StandardData.StandardData.raw" class="py-name" href="#" onclick="return doclink('link-57', 'raw', 'link-57');">raw</a></tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">tag_name</tt><tt class="py-op">,</tt> <tt class="py-name">tag_number</tt><tt class="py-op">,</tt> <tt class="py-name">tag_data</tt> <tt class="py-keyword">in</tt> <tt id="link-58" class="py-name" targets="Function Bio.SeqIO.AbiIO._abi_parse_header()=Bio.SeqIO.AbiIO-module.html#_abi_parse_header"><a title="Bio.SeqIO.AbiIO._abi_parse_header" class="py-name" href="#" onclick="return doclink('link-58', '_abi_parse_header', 'link-58');">_abi_parse_header</a></tt><tt class="py-op">(</tt><tt id="link-59" class="py-name"><a title="Bio.Wise.dnal.Statistics.header" class="py-name" href="#" onclick="return doclink('link-59', 'header', 'link-53');">header</a></tt><tt class="py-op">,</tt> <tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line">        <tt class="py-name">key</tt> <tt class="py-op">=</tt> <tt class="py-name">tag_name</tt> <tt class="py-op">+</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">tag_number</tt><tt class="py-op">)</tt> </tt>
<a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line"> </tt>
<a name="L378"></a><tt class="py-lineno">378</tt>  <tt class="py-line">        <tt id="link-60" class="py-name"><a title="Bio.Nexus.StandardData.StandardData.raw" class="py-name" href="#" onclick="return doclink('link-60', 'raw', 'link-57');">raw</a></tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">tag_data</tt> </tt>
<a name="L379"></a><tt class="py-lineno">379</tt>  <tt class="py-line"> </tt>
<a name="L380"></a><tt class="py-lineno">380</tt>  <tt class="py-line">        <tt class="py-comment"># PBAS2 is base-called sequence, only available in 3530</tt> </tt>
<a name="L381"></a><tt class="py-lineno">381</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-op">==</tt> <tt class="py-string">'PBAS2'</tt><tt class="py-op">:</tt> </tt>
<a name="L382"></a><tt class="py-lineno">382</tt>  <tt class="py-line">            <tt id="link-61" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.seq=Bio.SeqRecord.SeqRecord-class.html#seq,Variable BioSQL.BioSeq.DBSeqRecord.seq=BioSQL.BioSeq.DBSeqRecord-class.html#seq"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-61', 'seq', 'link-61');">seq</a></tt> <tt class="py-op">=</tt> <tt class="py-name">tag_data</tt> </tt>
<a name="L383"></a><tt class="py-lineno">383</tt>  <tt class="py-line">            <tt class="py-name">ambigs</tt> <tt class="py-op">=</tt> <tt class="py-string">'KYWMRS'</tt> </tt>
<a name="L384"></a><tt class="py-lineno">384</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-62" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-62', 'alphabet', 'link-34');">alphabet</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L385"></a><tt class="py-lineno">385</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-63" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-63', 'set', 'link-45');">set</a></tt><tt class="py-op">(</tt><tt id="link-64" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-64', 'seq', 'link-61');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">intersection</tt><tt class="py-op">(</tt><tt class="py-name">ambigs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L386"></a><tt class="py-lineno">386</tt>  <tt class="py-line">                    <tt id="link-65" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-65', 'alphabet', 'link-34');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-66" class="py-name"><a title="Bio.Alphabet.IUPAC.ambiguous_dna" class="py-name" href="#" onclick="return doclink('link-66', 'ambiguous_dna', 'link-5');">ambiguous_dna</a></tt> </tt>
<a name="L387"></a><tt class="py-lineno">387</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L388"></a><tt class="py-lineno">388</tt>  <tt class="py-line">                    <tt id="link-67" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-67', 'alphabet', 'link-34');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-68" class="py-name"><a title="Bio.Alphabet.IUPAC.unambiguous_dna" class="py-name" href="#" onclick="return doclink('link-68', 'unambiguous_dna', 'link-6');">unambiguous_dna</a></tt> </tt>
<a name="L389"></a><tt class="py-lineno">389</tt>  <tt class="py-line">        <tt class="py-comment"># PCON2 is quality values of base-called sequence</tt> </tt>
<a name="L390"></a><tt class="py-lineno">390</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">key</tt> <tt class="py-op">==</tt> <tt class="py-string">'PCON2'</tt><tt class="py-op">:</tt> </tt>
<a name="L391"></a><tt class="py-lineno">391</tt>  <tt class="py-line">            <tt class="py-name">qual</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">ord</tt><tt class="py-op">(</tt><tt class="py-name">val</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">val</tt> <tt class="py-keyword">in</tt> <tt class="py-name">tag_data</tt><tt class="py-op">]</tt> </tt>
<a name="L392"></a><tt class="py-lineno">392</tt>  <tt class="py-line">        <tt class="py-comment"># SMPL1 is sample id entered before sequencing run</tt> </tt>
<a name="L393"></a><tt class="py-lineno">393</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">key</tt> <tt class="py-op">==</tt> <tt class="py-string">'SMPL1'</tt><tt class="py-op">:</tt> </tt>
<a name="L394"></a><tt class="py-lineno">394</tt>  <tt class="py-line">            <tt class="py-name">sample_id</tt> <tt class="py-op">=</tt> <tt class="py-name">tag_data</tt> </tt>
<a name="L395"></a><tt class="py-lineno">395</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">times</tt><tt class="py-op">:</tt> </tt>
<a name="L396"></a><tt class="py-lineno">396</tt>  <tt class="py-line">            <tt class="py-name">times</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">tag_data</tt> </tt>
<a name="L397"></a><tt class="py-lineno">397</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L398"></a><tt class="py-lineno">398</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt id="link-69" class="py-name"><a title="Bio.SeqIO.AbiIO._EXTRACT" class="py-name" href="#" onclick="return doclink('link-69', '_EXTRACT', 'link-18');">_EXTRACT</a></tt><tt class="py-op">:</tt> </tt>
<a name="L399"></a><tt class="py-lineno">399</tt>  <tt class="py-line">                <tt class="py-name">annot</tt><tt class="py-op">[</tt><tt id="link-70" class="py-name"><a title="Bio.SeqIO.AbiIO._EXTRACT" class="py-name" href="#" onclick="return doclink('link-70', '_EXTRACT', 'link-18');">_EXTRACT</a></tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">tag_data</tt> </tt>
<a name="L400"></a><tt class="py-lineno">400</tt>  <tt class="py-line"> </tt>
<a name="L401"></a><tt class="py-lineno">401</tt>  <tt class="py-line">    <tt class="py-comment"># set time annotations</tt> </tt>
<a name="L402"></a><tt class="py-lineno">402</tt>  <tt class="py-line">    <tt class="py-name">annot</tt><tt class="py-op">[</tt><tt class="py-string">'run_start'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">'%s %s'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">times</tt><tt class="py-op">[</tt><tt class="py-string">'RUND1'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">times</tt><tt class="py-op">[</tt><tt class="py-string">'RUNT1'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L403"></a><tt class="py-lineno">403</tt>  <tt class="py-line">    <tt class="py-name">annot</tt><tt class="py-op">[</tt><tt class="py-string">'run_finish'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">'%s %s'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">times</tt><tt class="py-op">[</tt><tt class="py-string">'RUND2'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">times</tt><tt class="py-op">[</tt><tt class="py-string">'RUNT2'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L404"></a><tt class="py-lineno">404</tt>  <tt class="py-line"> </tt>
<a name="L405"></a><tt class="py-lineno">405</tt>  <tt class="py-line">    <tt class="py-comment"># raw data (for advanced end users benefit)</tt> </tt>
<a name="L406"></a><tt class="py-lineno">406</tt>  <tt class="py-line">    <tt class="py-name">annot</tt><tt class="py-op">[</tt><tt class="py-string">'abif_raw'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-71" class="py-name"><a title="Bio.Nexus.StandardData.StandardData.raw" class="py-name" href="#" onclick="return doclink('link-71', 'raw', 'link-57');">raw</a></tt> </tt>
<a name="L407"></a><tt class="py-lineno">407</tt>  <tt class="py-line"> </tt>
<a name="L408"></a><tt class="py-lineno">408</tt>  <tt class="py-line">    <tt class="py-comment"># fsa check</tt> </tt>
<a name="L409"></a><tt class="py-lineno">409</tt>  <tt class="py-line">    <tt class="py-name">is_fsa_file</tt> <tt class="py-op">=</tt> <tt class="py-name">all</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">tn</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt id="link-72" class="py-name"><a title="Bio.Nexus.StandardData.StandardData.raw" class="py-name" href="#" onclick="return doclink('link-72', 'raw', 'link-57');">raw</a></tt> <tt class="py-keyword">for</tt> <tt class="py-name">tn</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'PBAS1'</tt><tt class="py-op">,</tt> <tt class="py-string">'PBAS2'</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L410"></a><tt class="py-lineno">410</tt>  <tt class="py-line"> </tt>
<a name="L411"></a><tt class="py-lineno">411</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">is_fsa_file</tt><tt class="py-op">:</tt> </tt>
<a name="L412"></a><tt class="py-lineno">412</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L413"></a><tt class="py-lineno">413</tt>  <tt class="py-line">            <tt class="py-name">file_name</tt> <tt class="py-op">=</tt> <tt class="py-name">basename</tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-73" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-73', 'name', 'link-73');">name</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'.fsa'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L414"></a><tt class="py-lineno">414</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L415"></a><tt class="py-lineno">415</tt>  <tt class="py-line">            <tt class="py-name">file_name</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L416"></a><tt class="py-lineno">416</tt>  <tt class="py-line">        <tt class="py-name">sample_id</tt> <tt class="py-op">=</tt> <tt id="link-74" class="py-name"><a title="Bio.Nexus.StandardData.StandardData.raw" class="py-name" href="#" onclick="return doclink('link-74', 'raw', 'link-57');">raw</a></tt><tt class="py-op">.</tt><tt id="link-75" class="py-name" targets="Method Bio.Crystal.Crystal.get()=Bio.Crystal.Crystal-class.html#get,Method Bio.Data.CodonTable.AmbiguousForwardTable.get()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#get,Method Bio.File._IndexedSeqFileDict.get()=Bio.File._IndexedSeqFileDict-class.html#get,Method Bio.File._IndexedSeqFileProxy.get()=Bio.File._IndexedSeqFileProxy-class.html#get,Method Bio.File._SQLiteManySeqFilesDict.get()=Bio.File._SQLiteManySeqFilesDict-class.html#get,Method Bio.Restriction.Restriction.RestrictionBatch.get()=Bio.Restriction.Restriction.RestrictionBatch-class.html#get,Method Bio.SearchIO._index.SearchIndexer.get()=Bio.SearchIO._index.SearchIndexer-class.html#get,Method Bio.SeqIO._index.SeqFileRandomAccess.get()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#get,Method Bio.SeqIO._index.SffRandomAccess.get()=Bio.SeqIO._index.SffRandomAccess-class.html#get,Method Bio.SeqIO._index.SffTrimedRandomAccess.get()=Bio.SeqIO._index.SffTrimedRandomAccess-class.html#get,Method Bio.SeqIO._index.UniprotRandomAccess.get()=Bio.SeqIO._index.UniprotRandomAccess-class.html#get"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-75', 'get', 'link-75');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'LIMS1'</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">sample_id</tt> </tt>
<a name="L417"></a><tt class="py-lineno">417</tt>  <tt class="py-line">        <tt id="link-76" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.description()=Bio.Phylo.PhyloXMLIO.Writer-class.html#description,Method Bio.SearchIO._legacy.NCBIStandalone._DescriptionConsumer.description()=Bio.SearchIO._legacy.NCBIStandalone._DescriptionConsumer-class.html#description,Variable Bio.SearchIO._model.hit.Hit.description=Bio.SearchIO._model.hit.Hit-class.html#description,Variable Bio.SearchIO._model.query.QueryResult.description=Bio.SearchIO._model.query.QueryResult-class.html#description"><a title="Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._legacy.NCBIStandalone._DescriptionConsumer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-76', 'description', 'link-76');">description</a></tt> <tt class="py-op">=</tt> <tt id="link-77" class="py-name"><a title="Bio.Nexus.StandardData.StandardData.raw" class="py-name" href="#" onclick="return doclink('link-77', 'raw', 'link-57');">raw</a></tt><tt class="py-op">.</tt><tt id="link-78" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-78', 'get', 'link-75');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'CTID1'</tt><tt class="py-op">,</tt> <tt class="py-string">'&lt;unknown description&gt;'</tt><tt class="py-op">)</tt> </tt>
<a name="L418"></a><tt class="py-lineno">418</tt>  <tt class="py-line">        <tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt id="link-79" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-79', 'SeqRecord', 'link-11');">SeqRecord</a></tt><tt class="py-op">(</tt><tt id="link-80" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-80', 'Seq', 'link-8');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L419"></a><tt class="py-lineno">419</tt>  <tt class="py-line">                           <tt id="link-81" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Variable Bio.PDB.Entity.Entity.id=Bio.PDB.Entity.Entity-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-81', 'id', 'link-81');">id</a></tt><tt class="py-op">=</tt><tt class="py-name">sample_id</tt><tt class="py-op">,</tt> </tt>
<a name="L420"></a><tt class="py-lineno">420</tt>  <tt class="py-line">                           <tt id="link-82" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-82', 'name', 'link-73');">name</a></tt><tt class="py-op">=</tt><tt class="py-name">file_name</tt><tt class="py-op">,</tt> </tt>
<a name="L421"></a><tt class="py-lineno">421</tt>  <tt class="py-line">                           <tt id="link-83" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._legacy.NCBIStandalone._DescriptionConsumer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-83', 'description', 'link-76');">description</a></tt><tt class="py-op">=</tt><tt id="link-84" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._legacy.NCBIStandalone._DescriptionConsumer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-84', 'description', 'link-76');">description</a></tt><tt class="py-op">,</tt> </tt>
<a name="L422"></a><tt class="py-lineno">422</tt>  <tt class="py-line">                           <tt id="link-85" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.annotations=BioSQL.BioSeq.DBSeqRecord-class.html#annotations"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-85', 'annotations', 'link-85');">annotations</a></tt><tt class="py-op">=</tt><tt class="py-name">annot</tt><tt class="py-op">)</tt> </tt>
<a name="L423"></a><tt class="py-lineno">423</tt>  <tt class="py-line"> </tt>
<a name="L424"></a><tt class="py-lineno">424</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L425"></a><tt class="py-lineno">425</tt>  <tt class="py-line">        <tt class="py-comment"># use the file name as SeqRecord.name if available</tt> </tt>
<a name="L426"></a><tt class="py-lineno">426</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L427"></a><tt class="py-lineno">427</tt>  <tt class="py-line">            <tt class="py-name">file_name</tt> <tt class="py-op">=</tt> <tt class="py-name">basename</tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-86" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-86', 'name', 'link-73');">name</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'.ab1'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L428"></a><tt class="py-lineno">428</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L429"></a><tt class="py-lineno">429</tt>  <tt class="py-line">            <tt class="py-name">file_name</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L430"></a><tt class="py-lineno">430</tt>  <tt class="py-line">        <tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt id="link-87" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-87', 'SeqRecord', 'link-11');">SeqRecord</a></tt><tt class="py-op">(</tt><tt id="link-88" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-88', 'Seq', 'link-8');">Seq</a></tt><tt class="py-op">(</tt><tt id="link-89" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-89', 'seq', 'link-61');">seq</a></tt><tt class="py-op">,</tt> <tt id="link-90" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-90', 'alphabet', 'link-34');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L431"></a><tt class="py-lineno">431</tt>  <tt class="py-line">                           <tt id="link-91" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-91', 'id', 'link-81');">id</a></tt><tt class="py-op">=</tt><tt class="py-name">sample_id</tt><tt class="py-op">,</tt> <tt id="link-92" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-92', 'name', 'link-73');">name</a></tt><tt class="py-op">=</tt><tt class="py-name">file_name</tt><tt class="py-op">,</tt> </tt>
<a name="L432"></a><tt class="py-lineno">432</tt>  <tt class="py-line">                           <tt id="link-93" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._legacy.NCBIStandalone._DescriptionConsumer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-93', 'description', 'link-76');">description</a></tt><tt class="py-op">=</tt><tt class="py-string">''</tt><tt class="py-op">,</tt> </tt>
<a name="L433"></a><tt class="py-lineno">433</tt>  <tt class="py-line">                           <tt id="link-94" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-94', 'annotations', 'link-85');">annotations</a></tt><tt class="py-op">=</tt><tt class="py-name">annot</tt><tt class="py-op">,</tt> </tt>
<a name="L434"></a><tt class="py-lineno">434</tt>  <tt class="py-line">                           <tt id="link-95" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.letter_annotations=Bio.SeqRecord.SeqRecord-class.html#letter_annotations"><a title="Bio.SeqRecord.SeqRecord.letter_annotations" class="py-name" href="#" onclick="return doclink('link-95', 'letter_annotations', 'link-95');">letter_annotations</a></tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-string">'phred_quality'</tt><tt class="py-op">:</tt> <tt class="py-name">qual</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L435"></a><tt class="py-lineno">435</tt>  <tt class="py-line"> </tt>
<a name="L436"></a><tt class="py-lineno">436</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">trim</tt> <tt class="py-keyword">or</tt> <tt class="py-name">is_fsa_file</tt><tt class="py-op">:</tt> </tt>
<a name="L437"></a><tt class="py-lineno">437</tt>  <tt class="py-line">        <tt class="py-keyword">yield</tt> <tt class="py-name">record</tt> </tt>
<a name="L438"></a><tt class="py-lineno">438</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L439"></a><tt class="py-lineno">439</tt>  <tt class="py-line">        <tt class="py-keyword">yield</tt> <tt id="link-96" class="py-name" targets="Function Bio.SeqIO.AbiIO._abi_trim()=Bio.SeqIO.AbiIO-module.html#_abi_trim"><a title="Bio.SeqIO.AbiIO._abi_trim" class="py-name" href="#" onclick="return doclink('link-96', '_abi_trim', 'link-96');">_abi_trim</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
</div><a name="L440"></a><tt class="py-lineno">440</tt>  <tt class="py-line"> </tt>
<a name="L441"></a><tt class="py-lineno">441</tt>  <tt class="py-line"> </tt>
<a name="_AbiTrimIterator"></a><div id="_AbiTrimIterator-def"><a name="L442"></a><tt class="py-lineno">442</tt> <a class="py-toggle" href="#" id="_AbiTrimIterator-toggle" onclick="return toggle('_AbiTrimIterator');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.AbiIO-module.html#_AbiTrimIterator">_AbiTrimIterator</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_AbiTrimIterator-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_AbiTrimIterator-expanded"><a name="L443"></a><tt class="py-lineno">443</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return an iterator for the Abi file format that yields trimmed SeqRecord objects (PRIVATE)."""</tt> </tt>
<a name="L444"></a><tt class="py-lineno">444</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-97" class="py-name" targets="Function Bio.SeqIO.AbiIO.AbiIterator()=Bio.SeqIO.AbiIO-module.html#AbiIterator"><a title="Bio.SeqIO.AbiIO.AbiIterator" class="py-name" href="#" onclick="return doclink('link-97', 'AbiIterator', 'link-97');">AbiIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-name">trim</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
</div><a name="L445"></a><tt class="py-lineno">445</tt>  <tt class="py-line"> </tt>
<a name="L446"></a><tt class="py-lineno">446</tt>  <tt class="py-line"> </tt>
<a name="_abi_parse_header"></a><div id="_abi_parse_header-def"><a name="L447"></a><tt class="py-lineno">447</tt> <a class="py-toggle" href="#" id="_abi_parse_header-toggle" onclick="return toggle('_abi_parse_header');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.AbiIO-module.html#_abi_parse_header">_abi_parse_header</a><tt class="py-op">(</tt><tt class="py-param">header</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_abi_parse_header-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_abi_parse_header-expanded"><a name="L448"></a><tt class="py-lineno">448</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return directory contents (PRIVATE)."""</tt> </tt>
<a name="L449"></a><tt class="py-lineno">449</tt>  <tt class="py-line">    <tt class="py-comment"># header structure (after ABIF marker):</tt> </tt>
<a name="L450"></a><tt class="py-lineno">450</tt>  <tt class="py-line">    <tt class="py-comment"># file version, tag name, tag number,</tt> </tt>
<a name="L451"></a><tt class="py-lineno">451</tt>  <tt class="py-line">    <tt class="py-comment"># element type code, element size, number of elements</tt> </tt>
<a name="L452"></a><tt class="py-lineno">452</tt>  <tt class="py-line">    <tt class="py-comment"># data size, data offset, handle (not file handle)</tt> </tt>
<a name="L453"></a><tt class="py-lineno">453</tt>  <tt class="py-line">    <tt class="py-name">head_elem_size</tt> <tt class="py-op">=</tt> <tt id="link-98" class="py-name"><a title="Bio.Wise.dnal.Statistics.header" class="py-name" href="#" onclick="return doclink('link-98', 'header', 'link-53');">header</a></tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt> </tt>
<a name="L454"></a><tt class="py-lineno">454</tt>  <tt class="py-line">    <tt class="py-name">head_elem_num</tt> <tt class="py-op">=</tt> <tt id="link-99" class="py-name"><a title="Bio.Wise.dnal.Statistics.header" class="py-name" href="#" onclick="return doclink('link-99', 'header', 'link-53');">header</a></tt><tt class="py-op">[</tt><tt class="py-number">5</tt><tt class="py-op">]</tt> </tt>
<a name="L455"></a><tt class="py-lineno">455</tt>  <tt class="py-line">    <tt class="py-name">head_offset</tt> <tt class="py-op">=</tt> <tt id="link-100" class="py-name"><a title="Bio.Wise.dnal.Statistics.header" class="py-name" href="#" onclick="return doclink('link-100', 'header', 'link-53');">header</a></tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">]</tt> </tt>
<a name="L456"></a><tt class="py-lineno">456</tt>  <tt class="py-line">    <tt id="link-101" class="py-name" targets="Method Bio.Crystal.Chain.index()=Bio.Crystal.Chain-class.html#index,Method Bio.SCOP.Raf.SeqMap.index()=Bio.SCOP.Raf.SeqMap-class.html#index,Method Bio.SearchIO._model.hit.Hit.index()=Bio.SearchIO._model.hit.Hit-class.html#index,Method Bio.SearchIO._model.query.QueryResult.index()=Bio.SearchIO._model.query.QueryResult-class.html#index,Function Bio.SearchIO.index()=Bio.SearchIO-module.html#index,Method Bio.Seq.MutableSeq.index()=Bio.Seq.MutableSeq-class.html#index,Function Bio.SeqIO.index()=Bio.SeqIO-module.html#index"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-101', 'index', 'link-101');">index</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L457"></a><tt class="py-lineno">457</tt>  <tt class="py-line"> </tt>
<a name="L458"></a><tt class="py-lineno">458</tt>  <tt class="py-line">    <tt class="py-keyword">while</tt> <tt id="link-102" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-102', 'index', 'link-101');">index</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-name">head_elem_num</tt><tt class="py-op">:</tt> </tt>
<a name="L459"></a><tt class="py-lineno">459</tt>  <tt class="py-line">        <tt id="link-103" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.start=Bio.SeqFeature.CompoundLocation-class.html#start,Variable Bio.SeqFeature.FeatureLocation.start=Bio.SeqFeature.FeatureLocation-class.html#start"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-103', 'start', 'link-103');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">head_offset</tt> <tt class="py-op">+</tt> <tt id="link-104" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-104', 'index', 'link-101');">index</a></tt> <tt class="py-op">*</tt> <tt class="py-name">head_elem_size</tt> </tt>
<a name="L460"></a><tt class="py-lineno">460</tt>  <tt class="py-line">        <tt class="py-comment"># add directory offset to tuple</tt> </tt>
<a name="L461"></a><tt class="py-lineno">461</tt>  <tt class="py-line">        <tt class="py-comment"># to handle directories with data size &lt;= 4 bytes</tt> </tt>
<a name="L462"></a><tt class="py-lineno">462</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-105" class="py-name"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-105', 'seek', 'link-48');">seek</a></tt><tt class="py-op">(</tt><tt id="link-106" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-106', 'start', 'link-103');">start</a></tt><tt class="py-op">)</tt> </tt>
<a name="L463"></a><tt class="py-lineno">463</tt>  <tt class="py-line">        <tt class="py-name">dir_entry</tt> <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">unpack</tt><tt class="py-op">(</tt><tt id="link-107" class="py-name"><a title="Bio.SeqIO.AbiIO._DIRFMT" class="py-name" href="#" onclick="return doclink('link-107', '_DIRFMT', 'link-26');">_DIRFMT</a></tt><tt class="py-op">,</tt> </tt>
<a name="L464"></a><tt class="py-lineno">464</tt>  <tt class="py-line">                                  <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-108" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.ExPASy.cellosaurus.read
Bio.File.UndoHandle.read
Bio.KEGG.Enzyme.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.minimal.read
Bio.motifs.read
Bio.motifs.transfac.read
Bio.phenotype.read" class="py-name" href="#" onclick="return doclink('link-108', 'read', 'link-49');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt id="link-109" class="py-name"><a title="Bio.SeqIO.AbiIO._DIRFMT" class="py-name" href="#" onclick="return doclink('link-109', '_DIRFMT', 'link-26');">_DIRFMT</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt id="link-110" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-110', 'start', 'link-103');">start</a></tt><tt class="py-op">,</tt><tt class="py-op">)</tt> </tt>
<a name="L465"></a><tt class="py-lineno">465</tt>  <tt class="py-line">        <tt id="link-111" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-111', 'index', 'link-101');">index</a></tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L466"></a><tt class="py-lineno">466</tt>  <tt class="py-line">        <tt class="py-comment"># only parse desired dirs</tt> </tt>
<a name="L467"></a><tt class="py-lineno">467</tt>  <tt class="py-line">        <tt class="py-name">key</tt> <tt class="py-op">=</tt> <tt id="link-112" class="py-name"><a title="Bio._py3k._bytes_to_string" class="py-name" href="#" onclick="return doclink('link-112', '_bytes_to_string', 'link-15');">_bytes_to_string</a></tt><tt class="py-op">(</tt><tt class="py-name">dir_entry</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L468"></a><tt class="py-lineno">468</tt>  <tt class="py-line">        <tt class="py-name">key</tt> <tt class="py-op">+=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">dir_entry</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L469"></a><tt class="py-lineno">469</tt>  <tt class="py-line"> </tt>
<a name="L470"></a><tt class="py-lineno">470</tt>  <tt class="py-line">        <tt class="py-name">tag_name</tt> <tt class="py-op">=</tt> <tt id="link-113" class="py-name"><a title="Bio._py3k._bytes_to_string" class="py-name" href="#" onclick="return doclink('link-113', '_bytes_to_string', 'link-15');">_bytes_to_string</a></tt><tt class="py-op">(</tt><tt class="py-name">dir_entry</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L471"></a><tt class="py-lineno">471</tt>  <tt class="py-line">        <tt class="py-name">tag_number</tt> <tt class="py-op">=</tt> <tt class="py-name">dir_entry</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L472"></a><tt class="py-lineno">472</tt>  <tt class="py-line">        <tt class="py-name">elem_code</tt> <tt class="py-op">=</tt> <tt class="py-name">dir_entry</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> </tt>
<a name="L473"></a><tt class="py-lineno">473</tt>  <tt class="py-line">        <tt class="py-name">elem_num</tt> <tt class="py-op">=</tt> <tt class="py-name">dir_entry</tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt> </tt>
<a name="L474"></a><tt class="py-lineno">474</tt>  <tt class="py-line">        <tt class="py-name">data_size</tt> <tt class="py-op">=</tt> <tt class="py-name">dir_entry</tt><tt class="py-op">[</tt><tt class="py-number">5</tt><tt class="py-op">]</tt> </tt>
<a name="L475"></a><tt class="py-lineno">475</tt>  <tt class="py-line">        <tt class="py-name">data_offset</tt> <tt class="py-op">=</tt> <tt class="py-name">dir_entry</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">]</tt> </tt>
<a name="L476"></a><tt class="py-lineno">476</tt>  <tt class="py-line">        <tt class="py-name">tag_offset</tt> <tt class="py-op">=</tt> <tt class="py-name">dir_entry</tt><tt class="py-op">[</tt><tt class="py-number">8</tt><tt class="py-op">]</tt> </tt>
<a name="L477"></a><tt class="py-lineno">477</tt>  <tt class="py-line">        <tt class="py-comment"># if data size &lt;= 4 bytes, data is stored inside tag</tt> </tt>
<a name="L478"></a><tt class="py-lineno">478</tt>  <tt class="py-line">        <tt class="py-comment"># so offset needs to be changed</tt> </tt>
<a name="L479"></a><tt class="py-lineno">479</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">data_size</tt> <tt class="py-op">&lt;=</tt> <tt class="py-number">4</tt><tt class="py-op">:</tt> </tt>
<a name="L480"></a><tt class="py-lineno">480</tt>  <tt class="py-line">            <tt class="py-name">data_offset</tt> <tt class="py-op">=</tt> <tt class="py-name">tag_offset</tt> <tt class="py-op">+</tt> <tt class="py-number">20</tt> </tt>
<a name="L481"></a><tt class="py-lineno">481</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-114" class="py-name"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-114', 'seek', 'link-48');">seek</a></tt><tt class="py-op">(</tt><tt class="py-name">data_offset</tt><tt class="py-op">)</tt> </tt>
<a name="L482"></a><tt class="py-lineno">482</tt>  <tt class="py-line">        <tt id="link-115" class="py-name" targets="Variable BioSQL.BioSeq.DBSeq.data=BioSQL.BioSeq.DBSeq-class.html#data"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-115', 'data', 'link-115');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-116" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.ExPASy.cellosaurus.read
Bio.File.UndoHandle.read
Bio.KEGG.Enzyme.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.minimal.read
Bio.motifs.read
Bio.motifs.transfac.read
Bio.phenotype.read" class="py-name" href="#" onclick="return doclink('link-116', 'read', 'link-49');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">data_size</tt><tt class="py-op">)</tt> </tt>
<a name="L483"></a><tt class="py-lineno">483</tt>  <tt class="py-line">        <tt class="py-keyword">yield</tt> <tt class="py-name">tag_name</tt><tt class="py-op">,</tt> <tt class="py-name">tag_number</tt><tt class="py-op">,</tt> \ </tt>
<a name="L484"></a><tt class="py-lineno">484</tt>  <tt class="py-line">            <tt id="link-117" class="py-name" targets="Function Bio.SeqIO.AbiIO._parse_tag_data()=Bio.SeqIO.AbiIO-module.html#_parse_tag_data"><a title="Bio.SeqIO.AbiIO._parse_tag_data" class="py-name" href="#" onclick="return doclink('link-117', '_parse_tag_data', 'link-117');">_parse_tag_data</a></tt><tt class="py-op">(</tt><tt class="py-name">elem_code</tt><tt class="py-op">,</tt> <tt class="py-name">elem_num</tt><tt class="py-op">,</tt> <tt id="link-118" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-118', 'data', 'link-115');">data</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L485"></a><tt class="py-lineno">485</tt>  <tt class="py-line"> </tt>
<a name="L486"></a><tt class="py-lineno">486</tt>  <tt class="py-line"> </tt>
<a name="_abi_trim"></a><div id="_abi_trim-def"><a name="L487"></a><tt class="py-lineno">487</tt> <a class="py-toggle" href="#" id="_abi_trim-toggle" onclick="return toggle('_abi_trim');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.AbiIO-module.html#_abi_trim">_abi_trim</a><tt class="py-op">(</tt><tt class="py-param">seq_record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_abi_trim-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_abi_trim-expanded"><a name="L488"></a><tt class="py-lineno">488</tt>  <tt class="py-line">    <tt class="py-docstring">"""Trims the sequence using Richard Mott's modified trimming algorithm (PRIVATE).</tt> </tt>
<a name="L489"></a><tt class="py-lineno">489</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L490"></a><tt class="py-lineno">490</tt>  <tt class="py-line"><tt class="py-docstring">    Arguments:</tt> </tt>
<a name="L491"></a><tt class="py-lineno">491</tt>  <tt class="py-line"><tt class="py-docstring">        - seq_record - SeqRecord object to be trimmed.</tt> </tt>
<a name="L492"></a><tt class="py-lineno">492</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L493"></a><tt class="py-lineno">493</tt>  <tt class="py-line"><tt class="py-docstring">    Trimmed bases are determined from their segment score, which is a</tt> </tt>
<a name="L494"></a><tt class="py-lineno">494</tt>  <tt class="py-line"><tt class="py-docstring">    cumulative sum of each base's score. Base scores are calculated from</tt> </tt>
<a name="L495"></a><tt class="py-lineno">495</tt>  <tt class="py-line"><tt class="py-docstring">    their quality values.</tt> </tt>
<a name="L496"></a><tt class="py-lineno">496</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L497"></a><tt class="py-lineno">497</tt>  <tt class="py-line"><tt class="py-docstring">    More about the trimming algorithm:</tt> </tt>
<a name="L498"></a><tt class="py-lineno">498</tt>  <tt class="py-line"><tt class="py-docstring">    http://www.phrap.org/phredphrap/phred.html</tt> </tt>
<a name="L499"></a><tt class="py-lineno">499</tt>  <tt class="py-line"><tt class="py-docstring">    http://resources.qiagenbioinformatics.com/manuals/clcgenomicsworkbench/650/Quality_trimming.html</tt> </tt>
<a name="L500"></a><tt class="py-lineno">500</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L501"></a><tt class="py-lineno">501</tt>  <tt class="py-line">    <tt id="link-119" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-119', 'start', 'link-103');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt>   <tt class="py-comment"># flag for starting position of trimmed sequence</tt> </tt>
<a name="L502"></a><tt class="py-lineno">502</tt>  <tt class="py-line">    <tt id="link-120" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.segment()=Bio.GenBank._FeatureConsumer-class.html#segment,Method Bio.GenBank._RecordConsumer.segment()=Bio.GenBank._RecordConsumer-class.html#segment"><a title="Bio.GenBank._FeatureConsumer.segment
Bio.GenBank._RecordConsumer.segment" class="py-name" href="#" onclick="return doclink('link-120', 'segment', 'link-120');">segment</a></tt> <tt class="py-op">=</tt> <tt class="py-number">20</tt>    <tt class="py-comment"># minimum sequence length</tt> </tt>
<a name="L503"></a><tt class="py-lineno">503</tt>  <tt class="py-line">    <tt class="py-name">trim_start</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt>  <tt class="py-comment"># init start index</tt> </tt>
<a name="L504"></a><tt class="py-lineno">504</tt>  <tt class="py-line">    <tt class="py-name">cutoff</tt> <tt class="py-op">=</tt> <tt class="py-number">0.05</tt>   <tt class="py-comment"># default cutoff value for calculating base score</tt> </tt>
<a name="L505"></a><tt class="py-lineno">505</tt>  <tt class="py-line"> </tt>
<a name="L506"></a><tt class="py-lineno">506</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">seq_record</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;=</tt> <tt id="link-121" class="py-name"><a title="Bio.GenBank._FeatureConsumer.segment
Bio.GenBank._RecordConsumer.segment" class="py-name" href="#" onclick="return doclink('link-121', 'segment', 'link-120');">segment</a></tt><tt class="py-op">:</tt> </tt>
<a name="L507"></a><tt class="py-lineno">507</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">seq_record</tt> </tt>
<a name="L508"></a><tt class="py-lineno">508</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L509"></a><tt class="py-lineno">509</tt>  <tt class="py-line">        <tt class="py-comment"># calculate base score</tt> </tt>
<a name="L510"></a><tt class="py-lineno">510</tt>  <tt class="py-line">        <tt class="py-name">score_list</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">cutoff</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-number">10</tt> <tt class="py-op">**</tt> <tt class="py-op">(</tt><tt class="py-name">qual</tt> <tt class="py-op">/</tt> <tt class="py-op">-</tt><tt class="py-number">10.0</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">qual</tt> <tt class="py-keyword">in</tt> </tt>
<a name="L511"></a><tt class="py-lineno">511</tt>  <tt class="py-line">                      <tt class="py-name">seq_record</tt><tt class="py-op">.</tt><tt id="link-122" class="py-name"><a title="Bio.SeqRecord.SeqRecord.letter_annotations" class="py-name" href="#" onclick="return doclink('link-122', 'letter_annotations', 'link-95');">letter_annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'phred_quality'</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L512"></a><tt class="py-lineno">512</tt>  <tt class="py-line"> </tt>
<a name="L513"></a><tt class="py-lineno">513</tt>  <tt class="py-line">        <tt class="py-comment"># calculate cummulative score</tt> </tt>
<a name="L514"></a><tt class="py-lineno">514</tt>  <tt class="py-line">        <tt class="py-comment"># if cummulative value &lt; 0, set it to 0</tt> </tt>
<a name="L515"></a><tt class="py-lineno">515</tt>  <tt class="py-line">        <tt class="py-comment"># first value is set to 0, because of the assumption that</tt> </tt>
<a name="L516"></a><tt class="py-lineno">516</tt>  <tt class="py-line">        <tt class="py-comment"># the first base will always be trimmed out</tt> </tt>
<a name="L517"></a><tt class="py-lineno">517</tt>  <tt class="py-line">        <tt class="py-name">cummul_score</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L518"></a><tt class="py-lineno">518</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-123" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-123', 'i', 'link-123');">i</a></tt> <tt class="py-keyword">in</tt> <tt id="link-124" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-124', 'range', 'link-124');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">score_list</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L519"></a><tt class="py-lineno">519</tt>  <tt class="py-line">            <tt id="link-125" class="py-name" targets="Method Bio.Align.PairwiseAligner.score()=Bio.Align.PairwiseAligner-class.html#score,Method Bio.SearchIO._legacy.NCBIStandalone._HSPConsumer.score()=Bio.SearchIO._legacy.NCBIStandalone._HSPConsumer-class.html#score"><a title="Bio.Align.PairwiseAligner.score
Bio.SearchIO._legacy.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-125', 'score', 'link-125');">score</a></tt> <tt class="py-op">=</tt> <tt class="py-name">cummul_score</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">score_list</tt><tt class="py-op">[</tt><tt id="link-126" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-126', 'i', 'link-123');">i</a></tt><tt class="py-op">]</tt> </tt>
<a name="L520"></a><tt class="py-lineno">520</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-127" class="py-name"><a title="Bio.Align.PairwiseAligner.score
Bio.SearchIO._legacy.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-127', 'score', 'link-125');">score</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L521"></a><tt class="py-lineno">521</tt>  <tt class="py-line">                <tt class="py-name">cummul_score</tt><tt class="py-op">.</tt><tt id="link-128" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-128', 'append', 'link-128');">append</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L522"></a><tt class="py-lineno">522</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L523"></a><tt class="py-lineno">523</tt>  <tt class="py-line">                <tt class="py-name">cummul_score</tt><tt class="py-op">.</tt><tt id="link-129" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-129', 'append', 'link-128');">append</a></tt><tt class="py-op">(</tt><tt id="link-130" class="py-name"><a title="Bio.Align.PairwiseAligner.score
Bio.SearchIO._legacy.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-130', 'score', 'link-125');">score</a></tt><tt class="py-op">)</tt> </tt>
<a name="L524"></a><tt class="py-lineno">524</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-131" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-131', 'start', 'link-103');">start</a></tt><tt class="py-op">:</tt> </tt>
<a name="L525"></a><tt class="py-lineno">525</tt>  <tt class="py-line">                    <tt class="py-comment"># trim_start = value when cummulative score is first &gt; 0</tt> </tt>
<a name="L526"></a><tt class="py-lineno">526</tt>  <tt class="py-line">                    <tt class="py-name">trim_start</tt> <tt class="py-op">=</tt> <tt id="link-132" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-132', 'i', 'link-123');">i</a></tt> </tt>
<a name="L527"></a><tt class="py-lineno">527</tt>  <tt class="py-line">                    <tt id="link-133" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-133', 'start', 'link-103');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L528"></a><tt class="py-lineno">528</tt>  <tt class="py-line"> </tt>
<a name="L529"></a><tt class="py-lineno">529</tt>  <tt class="py-line">        <tt class="py-comment"># trim_finish = index of highest cummulative score,</tt> </tt>
<a name="L530"></a><tt class="py-lineno">530</tt>  <tt class="py-line">        <tt class="py-comment"># marking the end of sequence segment with highest cummulative score</tt> </tt>
<a name="L531"></a><tt class="py-lineno">531</tt>  <tt class="py-line">        <tt class="py-name">trim_finish</tt> <tt class="py-op">=</tt> <tt class="py-name">cummul_score</tt><tt class="py-op">.</tt><tt id="link-134" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-134', 'index', 'link-101');">index</a></tt><tt class="py-op">(</tt><tt id="link-135" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.max=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#max"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-135', 'max', 'link-135');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">cummul_score</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L532"></a><tt class="py-lineno">532</tt>  <tt class="py-line"> </tt>
<a name="L533"></a><tt class="py-lineno">533</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">seq_record</tt><tt class="py-op">[</tt><tt class="py-name">trim_start</tt><tt class="py-op">:</tt><tt class="py-name">trim_finish</tt><tt class="py-op">]</tt> </tt>
</div><a name="L534"></a><tt class="py-lineno">534</tt>  <tt class="py-line"> </tt>
<a name="L535"></a><tt class="py-lineno">535</tt>  <tt class="py-line"> </tt>
<a name="_parse_tag_data"></a><div id="_parse_tag_data-def"><a name="L536"></a><tt class="py-lineno">536</tt> <a class="py-toggle" href="#" id="_parse_tag_data-toggle" onclick="return toggle('_parse_tag_data');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.AbiIO-module.html#_parse_tag_data">_parse_tag_data</a><tt class="py-op">(</tt><tt class="py-param">elem_code</tt><tt class="py-op">,</tt> <tt class="py-param">elem_num</tt><tt class="py-op">,</tt> <tt class="py-param">raw_data</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_parse_tag_data-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_parse_tag_data-expanded"><a name="L537"></a><tt class="py-lineno">537</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return single data value (PRIVATE).</tt> </tt>
<a name="L538"></a><tt class="py-lineno">538</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L539"></a><tt class="py-lineno">539</tt>  <tt class="py-line"><tt class="py-docstring">    Arguments:</tt> </tt>
<a name="L540"></a><tt class="py-lineno">540</tt>  <tt class="py-line"><tt class="py-docstring">     - elem_code - What kind of data</tt> </tt>
<a name="L541"></a><tt class="py-lineno">541</tt>  <tt class="py-line"><tt class="py-docstring">     - elem_num - How many data points</tt> </tt>
<a name="L542"></a><tt class="py-lineno">542</tt>  <tt class="py-line"><tt class="py-docstring">     - raw_data - abi file object from which the tags would be unpacked</tt> </tt>
<a name="L543"></a><tt class="py-lineno">543</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L544"></a><tt class="py-lineno">544</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L545"></a><tt class="py-lineno">545</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">elem_code</tt> <tt class="py-keyword">in</tt> <tt id="link-136" class="py-name"><a title="Bio.SeqIO.AbiIO._BYTEFMT" class="py-name" href="#" onclick="return doclink('link-136', '_BYTEFMT', 'link-24');">_BYTEFMT</a></tt><tt class="py-op">:</tt> </tt>
<a name="L546"></a><tt class="py-lineno">546</tt>  <tt class="py-line">        <tt class="py-comment"># because '&gt;1s' unpack differently from '&gt;s'</tt> </tt>
<a name="L547"></a><tt class="py-lineno">547</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">elem_num</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L548"></a><tt class="py-lineno">548</tt>  <tt class="py-line">            <tt class="py-name">num</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L549"></a><tt class="py-lineno">549</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L550"></a><tt class="py-lineno">550</tt>  <tt class="py-line">            <tt class="py-name">num</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">elem_num</tt><tt class="py-op">)</tt> </tt>
<a name="L551"></a><tt class="py-lineno">551</tt>  <tt class="py-line">        <tt class="py-name">fmt</tt> <tt class="py-op">=</tt> <tt class="py-string">'&gt;'</tt> <tt class="py-op">+</tt> <tt class="py-name">num</tt> <tt class="py-op">+</tt> <tt id="link-137" class="py-name"><a title="Bio.SeqIO.AbiIO._BYTEFMT" class="py-name" href="#" onclick="return doclink('link-137', '_BYTEFMT', 'link-24');">_BYTEFMT</a></tt><tt class="py-op">[</tt><tt class="py-name">elem_code</tt><tt class="py-op">]</tt> </tt>
<a name="L552"></a><tt class="py-lineno">552</tt>  <tt class="py-line"> </tt>
<a name="L553"></a><tt class="py-lineno">553</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">raw_data</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">calcsize</tt><tt class="py-op">(</tt><tt class="py-name">fmt</tt><tt class="py-op">)</tt> </tt>
<a name="L554"></a><tt class="py-lineno">554</tt>  <tt class="py-line">        <tt id="link-138" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-138', 'data', 'link-115');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">struct</tt><tt class="py-op">.</tt><tt class="py-name">unpack</tt><tt class="py-op">(</tt><tt class="py-name">fmt</tt><tt class="py-op">,</tt> <tt class="py-name">raw_data</tt><tt class="py-op">)</tt> </tt>
<a name="L555"></a><tt class="py-lineno">555</tt>  <tt class="py-line"> </tt>
<a name="L556"></a><tt class="py-lineno">556</tt>  <tt class="py-line">        <tt class="py-comment"># no need to use tuple if len(data) == 1</tt> </tt>
<a name="L557"></a><tt class="py-lineno">557</tt>  <tt class="py-line">        <tt class="py-comment"># also if data is date / time</tt> </tt>
<a name="L558"></a><tt class="py-lineno">558</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">elem_code</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-number">10</tt><tt class="py-op">,</tt> <tt class="py-number">11</tt><tt class="py-op">]</tt> <tt class="py-keyword">and</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-139" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-139', 'data', 'link-115');">data</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L559"></a><tt class="py-lineno">559</tt>  <tt class="py-line">            <tt id="link-140" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-140', 'data', 'link-115');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-141" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-141', 'data', 'link-115');">data</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L560"></a><tt class="py-lineno">560</tt>  <tt class="py-line"> </tt>
<a name="L561"></a><tt class="py-lineno">561</tt>  <tt class="py-line">        <tt class="py-comment"># account for different data types</tt> </tt>
<a name="L562"></a><tt class="py-lineno">562</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">elem_code</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L563"></a><tt class="py-lineno">563</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-142" class="py-name"><a title="Bio._py3k._bytes_to_string" class="py-name" href="#" onclick="return doclink('link-142', '_bytes_to_string', 'link-15');">_bytes_to_string</a></tt><tt class="py-op">(</tt><tt id="link-143" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-143', 'data', 'link-115');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L564"></a><tt class="py-lineno">564</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">elem_code</tt> <tt class="py-op">==</tt> <tt class="py-number">10</tt><tt class="py-op">:</tt> </tt>
<a name="L565"></a><tt class="py-lineno">565</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">datetime</tt><tt class="py-op">.</tt><tt id="link-144" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.date()=Bio.GenBank._FeatureConsumer-class.html#date,Method Bio.GenBank._RecordConsumer.date()=Bio.GenBank._RecordConsumer-class.html#date,Method Bio.Phylo.PhyloXMLIO.Parser.date()=Bio.Phylo.PhyloXMLIO.Parser-class.html#date,Method Bio.Phylo.PhyloXMLIO.Writer.date()=Bio.Phylo.PhyloXMLIO.Writer-class.html#date"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-144', 'date', 'link-144');">date</a></tt><tt class="py-op">(</tt><tt class="py-op">*</tt><tt id="link-145" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-145', 'data', 'link-115');">data</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L566"></a><tt class="py-lineno">566</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">elem_code</tt> <tt class="py-op">==</tt> <tt class="py-number">11</tt><tt class="py-op">:</tt> </tt>
<a name="L567"></a><tt class="py-lineno">567</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">datetime</tt><tt class="py-op">.</tt><tt class="py-name">time</tt><tt class="py-op">(</tt><tt class="py-op">*</tt><tt id="link-146" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-146', 'data', 'link-115');">data</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L568"></a><tt class="py-lineno">568</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">elem_code</tt> <tt class="py-op">==</tt> <tt class="py-number">13</tt><tt class="py-op">:</tt> </tt>
<a name="L569"></a><tt class="py-lineno">569</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">bool</tt><tt class="py-op">(</tt><tt id="link-147" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-147', 'data', 'link-115');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L570"></a><tt class="py-lineno">570</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">elem_code</tt> <tt class="py-op">==</tt> <tt class="py-number">18</tt><tt class="py-op">:</tt> </tt>
<a name="L571"></a><tt class="py-lineno">571</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-148" class="py-name"><a title="Bio._py3k._bytes_to_string" class="py-name" href="#" onclick="return doclink('link-148', '_bytes_to_string', 'link-15');">_bytes_to_string</a></tt><tt class="py-op">(</tt><tt id="link-149" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-149', 'data', 'link-115');">data</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L572"></a><tt class="py-lineno">572</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">elem_code</tt> <tt class="py-op">==</tt> <tt class="py-number">19</tt><tt class="py-op">:</tt> </tt>
<a name="L573"></a><tt class="py-lineno">573</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-150" class="py-name"><a title="Bio._py3k._bytes_to_string" class="py-name" href="#" onclick="return doclink('link-150', '_bytes_to_string', 'link-15');">_bytes_to_string</a></tt><tt class="py-op">(</tt><tt id="link-151" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-151', 'data', 'link-115');">data</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L574"></a><tt class="py-lineno">574</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L575"></a><tt class="py-lineno">575</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-152" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-152', 'data', 'link-115');">data</a></tt> </tt>
<a name="L576"></a><tt class="py-lineno">576</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L577"></a><tt class="py-lineno">577</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L578"></a><tt class="py-lineno">578</tt>  <tt class="py-line"> </tt>
<a name="L579"></a><tt class="py-lineno">579</tt>  <tt class="py-line"> </tt>
<a name="L580"></a><tt class="py-lineno">580</tt>  <tt class="py-line"><tt class="py-keyword">if</tt> <tt class="py-name">__name__</tt> <tt class="py-op">==</tt> <tt class="py-string">'__main__'</tt><tt class="py-op">:</tt> </tt>
<a name="L581"></a><tt class="py-lineno">581</tt>  <tt class="py-line">    <tt class="py-keyword">pass</tt> </tt>
<a name="L582"></a><tt class="py-lineno">582</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Tue Dec 18 15:58:28 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
