<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.Seq</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        Module&nbsp;Seq
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.Seq-module.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== MODULE DESCRIPTION ==================== -->
<h1 class="epydoc">Module Seq</h1><p class="nomargin-top"><span class="codelink"><a href="Bio.Seq-pysrc.html">source&nbsp;code</a></span></p>
<p>Provide objects to represent biological sequences with alphabets.</p>
<dl class="rst-docutils">
<dt>See also the <a class="rst-reference external" href="http://biopython.org/wiki/Seq" target="_top">Seq</a> wiki and the chapter in our tutorial:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li><a class="rst-reference external" href="http://biopython.org/DIST/docs/tutorial/Tutorial.html" target="_top">HTML Tutorial</a></li>
<li><a class="rst-reference external" href="http://biopython.org/DIST/docs/tutorial/Tutorial.pdf" target="_top">PDF Tutorial</a></li>
</ul>
</dd>
</dl>

<!-- ==================== CLASSES ==================== -->
<a name="section-Classes"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Classes</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Classes"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.Seq.Seq-class.html" class="summary-name">Seq</a><br />
      Read-only sequence object (essentially a string with an alphabet).
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.Seq.UnknownSeq-class.html" class="summary-name">UnknownSeq</a><br />
      Read-only sequence object of known length but unknown contents.
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.Seq.MutableSeq-class.html" class="summary-name">MutableSeq</a><br />
      An editable sequence object (with an alphabet).
    </td>
  </tr>
</table>
<!-- ==================== FUNCTIONS ==================== -->
<a name="section-Functions"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Functions</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Functions"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq-module.html#_maketrans" class="summary-sig-name" onclick="show_private();">_maketrans</a>(<span class="summary-sig-arg">complement_mapping</span>)</span><br />
      Make a python string translation table (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#_maketrans">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq-module.html#transcribe" class="summary-sig-name">transcribe</a>(<span class="summary-sig-arg">dna</span>)</span><br />
      Transcribe a DNA sequence into RNA.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#transcribe">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq-module.html#back_transcribe" class="summary-sig-name">back_transcribe</a>(<span class="summary-sig-arg">rna</span>)</span><br />
      Return the RNA sequence back-transcribed into DNA.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#back_transcribe">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq-module.html#_translate_str" class="summary-sig-name" onclick="show_private();">_translate_str</a>(<span class="summary-sig-arg">sequence</span>,
        <span class="summary-sig-arg">table</span>,
        <span class="summary-sig-arg">stop_symbol</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">*</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">to_stop</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">cds</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">pos_stop</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">X</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">gap</span>=<span class="summary-sig-default">None</span>)</span><br />
      Translate nucleotide string into a protein string (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#_translate_str">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq-module.html#translate" class="summary-sig-name">translate</a>(<span class="summary-sig-arg">sequence</span>,
        <span class="summary-sig-arg">table</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">Standard</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">stop_symbol</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">*</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">to_stop</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">cds</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">gap</span>=<span class="summary-sig-default">None</span>)</span><br />
      Translate a nucleotide sequence into amino acids.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#translate">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq-module.html#reverse_complement" class="summary-sig-name">reverse_complement</a>(<span class="summary-sig-arg">sequence</span>)</span><br />
      Return the reverse complement sequence of a nucleotide string.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#reverse_complement">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Seq-module.html#complement" class="summary-sig-name">complement</a>(<span class="summary-sig-arg">sequence</span>)</span><br />
      Return the complement sequence of a nucleotide string.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#complement">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_test"></a><span class="summary-sig-name">_test</span>()</span><br />
      Run the Bio.Seq module's doctests (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Seq-pysrc.html#_test">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
</table>
<!-- ==================== VARIABLES ==================== -->
<a name="section-Variables"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Variables</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Variables"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.Seq-module.html#_dna_complement_table" class="summary-name" onclick="show_private();">_dna_complement_table</a> = <code title="'''\x00\x01\x02\x03\x04\x05\x06\x07\x08\t
\x0b\x0c\r\x0e\x0f\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\
\x1d\x1e\x1f !&quot;#$%&amp;\'()*+,-./0123456789:;&lt;=&gt;?@TVGHEFCDIJMLKNOPQYSAUBWX\
RZ[\\]^_`tvghefcdijmlknopqysaubwxrz{|}~\x7f\x80\x81\x82\x83\x84\x85\x8\
6\x87\x88\x89\x8a\x8b\x8c\x8d\x8e\x8f\x90\x91\x92\x93\x94\x95\x96\x97\\
x98\x99\x9a\x9b\x9c\x9d\x9e\x9f\xa0\xa1\xa2\xa3\xa4\xa5\xa6\xa7\xa8\xa\
9\xaa\xab\xac\xad\xae\xaf\xb0\xb1\xb2\xb3\xb4\xb5\xb6\xb7\xb8\xb9\xba\\
xbb\xbc\xbd\xbe\xbf\xc0\xc1\xc2\xc3\xc4\xc5\xc6\xc7\xc8\xc9\xca\xcb\xc\
..."><code class="variable-quote">'</code><code class="variable-string">\x00\x01\x02\x03\x04\x05\x06\x07\x08\</code><code class="variable-ellipsis">...</code></code>
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.Seq-module.html#_rna_complement_table" class="summary-name" onclick="show_private();">_rna_complement_table</a> = <code title="'''\x00\x01\x02\x03\x04\x05\x06\x07\x08\t
\x0b\x0c\r\x0e\x0f\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\
\x1d\x1e\x1f !&quot;#$%&amp;\'()*+,-./0123456789:;&lt;=&gt;?@UVGHEFCDIJMLKNOPQYSTABWX\
RZ[\\]^_`uvghefcdijmlknopqystabwxrz{|}~\x7f\x80\x81\x82\x83\x84\x85\x8\
6\x87\x88\x89\x8a\x8b\x8c\x8d\x8e\x8f\x90\x91\x92\x93\x94\x95\x96\x97\\
x98\x99\x9a\x9b\x9c\x9d\x9e\x9f\xa0\xa1\xa2\xa3\xa4\xa5\xa6\xa7\xa8\xa\
9\xaa\xab\xac\xad\xae\xaf\xb0\xb1\xb2\xb3\xb4\xb5\xb6\xb7\xb8\xb9\xba\\
xbb\xbc\xbd\xbe\xbf\xc0\xc1\xc2\xc3\xc4\xc5\xc6\xc7\xc8\xc9\xca\xcb\xc\
..."><code class="variable-quote">'</code><code class="variable-string">\x00\x01\x02\x03\x04\x05\x06\x07\x08\</code><code class="variable-ellipsis">...</code></code>
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="__package__"></a><span class="summary-name">__package__</span> = <code title="'Bio'"><code class="variable-quote">'</code><code class="variable-string">Bio</code><code class="variable-quote">'</code></code>
    </td>
  </tr>
</table>
<!-- ==================== FUNCTION DETAILS ==================== -->
<a name="section-FunctionDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Function Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-FunctionDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="_maketrans"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_maketrans</span>(<span class="sig-arg">complement_mapping</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#_maketrans">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Make a python string translation table (PRIVATE).</p>
<dl class="rst-docutils">
<dt>Arguments:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>complement_mapping - a dictionary such as ambiguous_dna_complement
and ambiguous_rna_complement from Data.IUPACData.</li>
</ul>
</dd>
</dl>
<p>Returns a translation table (a string of length 256) for use with the
python string's translate method to use in a (reverse) complement.</p>
<p>Compatible with lower case and upper case sequences.</p>
<p>For internal use only.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="transcribe"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">transcribe</span>(<span class="sig-arg">dna</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#transcribe">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Transcribe a DNA sequence into RNA.</p>
<p>If given a string, returns a new string object.</p>
<p>Given a Seq or MutableSeq, returns a new Seq object with an RNA alphabet.</p>
<p>Trying to transcribe a protein or RNA sequence raises an exception.</p>
<p>e.g.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>transcribe(<span class="py-string">&quot;ACTGN&quot;</span>)
<span class="py-output">'ACUGN'</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="back_transcribe"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">back_transcribe</span>(<span class="sig-arg">rna</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#back_transcribe">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Return the RNA sequence back-transcribed into DNA.</p>
<p>If given a string, returns a new string object.</p>
<p>Given a Seq or MutableSeq, returns a new Seq object with an RNA alphabet.</p>
<p>Trying to transcribe a protein or DNA sequence raises an exception.</p>
<p>e.g.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>back_transcribe(<span class="py-string">&quot;ACUGN&quot;</span>)
<span class="py-output">'ACTGN'</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_translate_str"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_translate_str</span>(<span class="sig-arg">sequence</span>,
        <span class="sig-arg">table</span>,
        <span class="sig-arg">stop_symbol</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">*</code><code class="variable-quote">'</code></span>,
        <span class="sig-arg">to_stop</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">cds</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">pos_stop</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">X</code><code class="variable-quote">'</code></span>,
        <span class="sig-arg">gap</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#_translate_str">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Translate nucleotide string into a protein string (PRIVATE).</p>
<dl class="rst-docutils">
<dt>Arguments:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>sequence - a string</li>
<li>table - a CodonTable object (NOT a table name or id number)</li>
<li>stop_symbol - a single character string, what to use for terminators.</li>
<li>to_stop - boolean, should translation terminate at the first
in frame stop codon?  If there is no in-frame stop codon
then translation continues to the end.</li>
<li>pos_stop - a single character string for a possible stop codon
(e.g. TAN or NNN)</li>
<li>cds - Boolean, indicates this is a complete CDS.  If True, this
checks the sequence starts with a valid alternative start
codon (which will be translated as methionine, M), that the
sequence length is a multiple of three, and that there is a
single in frame stop codon at the end (this will be excluded
from the protein sequence, regardless of the to_stop option).
If these tests fail, an exception is raised.</li>
<li>gap - Single character string to denote symbol used for gaps.
Defaults to None.</li>
</ul>
</dd>
</dl>
<p>Returns a string.</p>
<p>e.g.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Data <span class="py-keyword">import</span> CodonTable
<span class="py-prompt">&gt;&gt;&gt; </span>table = CodonTable.ambiguous_dna_by_id[1]
<span class="py-prompt">&gt;&gt;&gt; </span>_translate_str(<span class="py-string">&quot;AAA&quot;</span>, table)
<span class="py-output">'K'</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>_translate_str(<span class="py-string">&quot;TAR&quot;</span>, table)
<span class="py-output">'*'</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>_translate_str(<span class="py-string">&quot;TAN&quot;</span>, table)
<span class="py-output">'X'</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>_translate_str(<span class="py-string">&quot;TAN&quot;</span>, table, pos_stop=<span class="py-string">&quot;@&quot;</span>)
<span class="py-output">'@'</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>_translate_str(<span class="py-string">&quot;TA?&quot;</span>, table)
<span class="py-except">Traceback (most recent call last):</span>
<span class="py-except">   ...</span>
<span class="py-except">Bio.Data.CodonTable.TranslationError: Codon 'TA?' is invalid</span></pre>
<p>In a change to older versions of Biopython, partial codons are now
always regarded as an error (previously only checked if cds=True)
and will trigger a warning (likely to become an exception in a
future release).</p>
<p>If <strong>cds=True</strong>, the start and stop codons are checked, and the start
codon will be translated at methionine. The sequence must be an
while number of codons.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>_translate_str(<span class="py-string">&quot;ATGCCCTAG&quot;</span>, table, cds=True)
<span class="py-output">'MP'</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>_translate_str(<span class="py-string">&quot;AAACCCTAG&quot;</span>, table, cds=True)
<span class="py-except">Traceback (most recent call last):</span>
<span class="py-except">   ...</span>
<span class="py-except">Bio.Data.CodonTable.TranslationError: First codon 'AAA' is not a start codon</span>
<span class="py-except"></span><span class="py-prompt">&gt;&gt;&gt; </span>_translate_str(<span class="py-string">&quot;ATGCCCTAGCCCTAG&quot;</span>, table, cds=True)
<span class="py-except">Traceback (most recent call last):</span>
<span class="py-except">   ...</span>
<span class="py-except">Bio.Data.CodonTable.TranslationError: Extra in frame stop codon found.</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="translate"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">translate</span>(<span class="sig-arg">sequence</span>,
        <span class="sig-arg">table</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">Standard</code><code class="variable-quote">'</code></span>,
        <span class="sig-arg">stop_symbol</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">*</code><code class="variable-quote">'</code></span>,
        <span class="sig-arg">to_stop</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">cds</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">gap</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#translate">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Translate a nucleotide sequence into amino acids.</p>
<p>If given a string, returns a new string object. Given a Seq or
MutableSeq, returns a Seq object with a protein alphabet.</p>
<dl class="rst-docutils">
<dt>Arguments:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>table - Which codon table to use?  This can be either a name
(string), an NCBI identifier (integer), or a CodonTable object
(useful for non-standard genetic codes).  Defaults to the &quot;Standard&quot;
table.</li>
<li>stop_symbol - Single character string, what to use for any
terminators, defaults to the asterisk, &quot;*&quot;.</li>
<li>to_stop - Boolean, defaults to False meaning do a full
translation continuing on past any stop codons
(translated as the specified stop_symbol).  If
True, translation is terminated at the first in
frame stop codon (and the stop_symbol is not
appended to the returned protein sequence).</li>
<li>cds - Boolean, indicates this is a complete CDS.  If True, this
checks the sequence starts with a valid alternative start
codon (which will be translated as methionine, M), that the
sequence length is a multiple of three, and that there is a
single in frame stop codon at the end (this will be excluded
from the protein sequence, regardless of the to_stop option).
If these tests fail, an exception is raised.</li>
<li>gap - Single character string to denote symbol used for gaps.
Defaults to None.</li>
</ul>
</dd>
</dl>
<p>A simple string example using the default (standard) genetic code:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>coding_dna = <span class="py-string">&quot;GTGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG&quot;</span>
<span class="py-prompt">&gt;&gt;&gt; </span>translate(coding_dna)
<span class="py-output">'VAIVMGR*KGAR*'</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>translate(coding_dna, stop_symbol=<span class="py-string">&quot;@&quot;</span>)
<span class="py-output">'VAIVMGR@KGAR@'</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>translate(coding_dna, to_stop=True)
<span class="py-output">'VAIVMGR'</span></pre>
<p>Now using NCBI table 2, where TGA is not a stop codon:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>translate(coding_dna, table=2)
<span class="py-output">'VAIVMGRWKGAR*'</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>translate(coding_dna, table=2, to_stop=True)
<span class="py-output">'VAIVMGRWKGAR'</span></pre>
<p>In fact this example uses an alternative start codon valid under NCBI
table 2, GTG, which means this example is a complete valid CDS which
when translated should really start with methionine (not valine):</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>translate(coding_dna, table=2, cds=True)
<span class="py-output">'MAIVMGRWKGAR'</span></pre>
<p>Note that if the sequence has no in-frame stop codon, then the to_stop
argument has no effect:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>coding_dna2 = <span class="py-string">&quot;GTGGCCATTGTAATGGGCCGC&quot;</span>
<span class="py-prompt">&gt;&gt;&gt; </span>translate(coding_dna2)
<span class="py-output">'VAIVMGR'</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>translate(coding_dna2, to_stop=True)
<span class="py-output">'VAIVMGR'</span></pre>
<p>NOTE - Ambiguous codons like &quot;TAN&quot; or &quot;NNN&quot; could be an amino acid
or a stop codon.  These are translated as &quot;X&quot;.  Any invalid codon
(e.g. &quot;TA?&quot; or &quot;T-A&quot;) will throw a TranslationError.</p>
<p>It will however translate either DNA or RNA.</p>
<p>NOTE - Since version 1.71 Biopython contains codon tables with 'ambiguous
stop codons'. These are stop codons with unambiguous sequence but which
have a context dependent coding as STOP or as amino acid. With these tables
'to_stop' must be False (otherwise a ValueError is raised). The dual
coding codons will always be translated as amino acid, except for
'cds=True', where the last codon will be translated as STOP.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>coding_dna3 = <span class="py-string">&quot;ATGGCACGGAAGTGA&quot;</span>
<span class="py-prompt">&gt;&gt;&gt; </span>translate(coding_dna3)
<span class="py-output">'MARK*'</span></pre>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>translate(coding_dna3, table=27)  <span class="py-comment"># Table 27: TGA -&gt; STOP or W</span>
<span class="py-output">'MARKW'</span></pre>
<p>It will however raise a BiopythonWarning (not shown).</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>translate(coding_dna3, table=27, cds=True)
<span class="py-output">'MARK'</span></pre>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>translate(coding_dna3, table=27, to_stop=True)
<span class="py-except">Traceback (most recent call last):</span>
<span class="py-except">   ...</span>
<span class="py-except">ValueError: You cannot use 'to_stop=True' with this table ...</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="reverse_complement"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">reverse_complement</span>(<span class="sig-arg">sequence</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#reverse_complement">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Return the reverse complement sequence of a nucleotide string.</p>
<p>If given a string, returns a new string object.
Given a Seq or a MutableSeq, returns a new Seq object with the same
alphabet.</p>
<p>Supports unambiguous and ambiguous nucleotide sequences.</p>
<p>e.g.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>reverse_complement(<span class="py-string">&quot;ACTG-NH&quot;</span>)
<span class="py-output">'DN-CAGT'</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="complement"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">complement</span>(<span class="sig-arg">sequence</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Seq-pysrc.html#complement">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Return the complement sequence of a nucleotide string.</p>
<p>If given a string, returns a new string object.
Given a Seq or a MutableSeq, returns a new Seq object with the same
alphabet.</p>
<p>Supports unambiguous and ambiguous nucleotide sequences.</p>
<p>e.g.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>complement(<span class="py-string">&quot;ACTG-NH&quot;</span>)
<span class="py-output">'TGAC-ND'</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== VARIABLES DETAILS ==================== -->
<a name="section-VariablesDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Variables Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-VariablesDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="_dna_complement_table"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">_dna_complement_table</h3>
  
  <dl class="fields">
  </dl>
  <dl class="fields">
    <dt>Value:</dt>
      <dd><table><tr><td><pre class="variable">
<code class="variable-quote">'''</code><code class="variable-string">\x00\x01\x02\x03\x04\x05\x06\x07\x08\t</code>
<code class="variable-string">\x0b\x0c\r\x0e\x0f\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c</code><span class="variable-linewrap"><img src="crarr.png" alt="\" /></span>
<code class="variable-string">\x1d\x1e\x1f !&quot;#$%&amp;\'()*+,-./0123456789:;&lt;=&gt;?@TVGHEFCDIJMLKNOPQYSAUBWX</code><span class="variable-linewrap"><img src="crarr.png" alt="\" /></span>
<code class="variable-string">RZ[\\]^_`tvghefcdijmlknopqysaubwxrz{|}~\x7f\x80\x81\x82\x83\x84\x85\x8</code><span class="variable-linewrap"><img src="crarr.png" alt="\" /></span>
<code class="variable-string">6\x87\x88\x89\x8a\x8b\x8c\x8d\x8e\x8f\x90\x91\x92\x93\x94\x95\x96\x97\</code><span class="variable-linewrap"><img src="crarr.png" alt="\" /></span>
<code class="variable-string">x98\x99\x9a\x9b\x9c\x9d\x9e\x9f\xa0\xa1\xa2\xa3\xa4\xa5\xa6\xa7\xa8\xa</code><span class="variable-linewrap"><img src="crarr.png" alt="\" /></span>
<code class="variable-string">9\xaa\xab\xac\xad\xae\xaf\xb0\xb1\xb2\xb3\xb4\xb5\xb6\xb7\xb8\xb9\xba\</code><span class="variable-linewrap"><img src="crarr.png" alt="\" /></span>
<code class="variable-string">xbb\xbc\xbd\xbe\xbf\xc0\xc1\xc2\xc3\xc4\xc5\xc6\xc7\xc8\xc9\xca\xcb\xc</code><span class="variable-linewrap"><img src="crarr.png" alt="\" /></span>
<code class="variable-ellipsis">...</code>
</pre></td></tr></table>
</dd>
  </dl>
</td></tr></table>
</div>
<a name="_rna_complement_table"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">_rna_complement_table</h3>
  
  <dl class="fields">
  </dl>
  <dl class="fields">
    <dt>Value:</dt>
      <dd><table><tr><td><pre class="variable">
<code class="variable-quote">'''</code><code class="variable-string">\x00\x01\x02\x03\x04\x05\x06\x07\x08\t</code>
<code class="variable-string">\x0b\x0c\r\x0e\x0f\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c</code><span class="variable-linewrap"><img src="crarr.png" alt="\" /></span>
<code class="variable-string">\x1d\x1e\x1f !&quot;#$%&amp;\'()*+,-./0123456789:;&lt;=&gt;?@UVGHEFCDIJMLKNOPQYSTABWX</code><span class="variable-linewrap"><img src="crarr.png" alt="\" /></span>
<code class="variable-string">RZ[\\]^_`uvghefcdijmlknopqystabwxrz{|}~\x7f\x80\x81\x82\x83\x84\x85\x8</code><span class="variable-linewrap"><img src="crarr.png" alt="\" /></span>
<code class="variable-string">6\x87\x88\x89\x8a\x8b\x8c\x8d\x8e\x8f\x90\x91\x92\x93\x94\x95\x96\x97\</code><span class="variable-linewrap"><img src="crarr.png" alt="\" /></span>
<code class="variable-string">x98\x99\x9a\x9b\x9c\x9d\x9e\x9f\xa0\xa1\xa2\xa3\xa4\xa5\xa6\xa7\xa8\xa</code><span class="variable-linewrap"><img src="crarr.png" alt="\" /></span>
<code class="variable-string">9\xaa\xab\xac\xad\xae\xaf\xb0\xb1\xb2\xb3\xb4\xb5\xb6\xb7\xb8\xb9\xba\</code><span class="variable-linewrap"><img src="crarr.png" alt="\" /></span>
<code class="variable-string">xbb\xbc\xbd\xbe\xbf\xc0\xc1\xc2\xc3\xc4\xc5\xc6\xc7\xc8\xc9\xca\xcb\xc</code><span class="variable-linewrap"><img src="crarr.png" alt="\" /></span>
<code class="variable-ellipsis">...</code>
</pre></td></tr></table>
</dd>
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Nov  6 13:32:23 2019
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
