<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.SeqIO.SwissIO</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.SeqIO-module.html">Package&nbsp;SeqIO</a> ::
        Module&nbsp;SwissIO
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.SeqIO.SwissIO-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.SeqIO.SwissIO-module.html">Module Bio.SeqIO.SwissIO</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2006-2013 by Peter Cock.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># Revisions copyright 2008-2009 by Michiel de Hoon.</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># All rights reserved.</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-comment"># This file is part of the Biopython distribution and governed by your</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-comment"># choice of the "Biopython License Agreement" or the "BSD 3-Clause License".</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-comment"># Please see the LICENSE file that should have been included as part of this</tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-comment"># package.</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-docstring">"""Bio.SeqIO support for the "swiss" (aka SwissProt/UniProt) file format.</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-docstring">You are expected to use this module via the Bio.SeqIO functions.</tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-docstring">See also the Bio.SwissProt module which offers more than just accessing</tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-docstring">the sequences as SeqRecord objects.</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-docstring">See also Bio.SeqIO.UniprotIO.py which supports the "uniprot-xml" format.</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">__future__</tt> <tt class="py-keyword">import</tt> <tt class="py-name">print_function</tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-1" class="py-name" targets="Module Bio.Seq=Bio.Seq-module.html,Class Bio.Seq.Seq=Bio.Seq.Seq-class.html"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-1', 'Seq', 'link-1');">Seq</a></tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-2" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-2', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-3" class="py-name" targets="Module Bio.SeqRecord=Bio.SeqRecord-module.html,Class Bio.SeqRecord.SeqRecord=Bio.SeqRecord.SeqRecord-class.html"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-3', 'SeqRecord', 'link-3');">SeqRecord</a></tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-4" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-4', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-5" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-5', 'Alphabet', 'link-5');">Alphabet</a></tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-6" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-6', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-7" class="py-name" targets="Module Bio.SeqFeature=Bio.SeqFeature-module.html,Class Bio.SeqFeature.SeqFeature=Bio.SeqFeature.SeqFeature-class.html"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-7', 'SeqFeature', 'link-7');">SeqFeature</a></tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-8" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-8', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-9" class="py-name" targets="Package Bio.SwissProt=Bio.SwissProt-module.html"><a title="Bio.SwissProt" class="py-name" href="#" onclick="return doclink('link-9', 'SwissProt', 'link-9');">SwissProt</a></tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"> </tt>
<a name="_make_position"></a><div id="_make_position-def"><a name="L27"></a><tt class="py-lineno"> 27</tt> <a class="py-toggle" href="#" id="_make_position-toggle" onclick="return toggle('_make_position');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SwissIO-module.html#_make_position">_make_position</a><tt class="py-op">(</tt><tt class="py-param">location_string</tt><tt class="py-op">,</tt> <tt class="py-param">offset</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_make_position-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_make_position-expanded"><a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line">    <tt class="py-docstring">"""Turn a Swiss location position into a SeqFeature position object (PRIVATE).</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"><tt class="py-docstring">    An offset of -1 is used with a start location to make it pythonic.</tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">location_string</tt> <tt class="py-op">==</tt> <tt class="py-string">"?"</tt><tt class="py-op">:</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-10" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-10', 'SeqFeature', 'link-7');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-11" class="py-name" targets="Class Bio.SeqFeature.UnknownPosition=Bio.SeqFeature.UnknownPosition-class.html"><a title="Bio.SeqFeature.UnknownPosition" class="py-name" href="#" onclick="return doclink('link-11', 'UnknownPosition', 'link-11');">UnknownPosition</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line">    <tt class="py-comment"># Hack so that feature from 0 to 0 becomes 0 to 0, not -1 to 0.</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-12" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-12', 'SeqFeature', 'link-7');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-13" class="py-name" targets="Class Bio.SeqFeature.ExactPosition=Bio.SeqFeature.ExactPosition-class.html"><a title="Bio.SeqFeature.ExactPosition" class="py-name" href="#" onclick="return doclink('link-13', 'ExactPosition', 'link-13');">ExactPosition</a></tt><tt class="py-op">(</tt><tt id="link-14" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.max=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#max"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-14', 'max', 'link-14');">max</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">offset</tt> <tt class="py-op">+</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">location_string</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">location_string</tt><tt class="py-op">.</tt><tt id="link-15" class="py-name" targets="Method Bio.Seq.Seq.startswith()=Bio.Seq.Seq-class.html#startswith"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-15', 'startswith', 'link-15');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"&lt;"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-16" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-16', 'SeqFeature', 'link-7');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-17" class="py-name" targets="Class Bio.SeqFeature.BeforePosition=Bio.SeqFeature.BeforePosition-class.html"><a title="Bio.SeqFeature.BeforePosition" class="py-name" href="#" onclick="return doclink('link-17', 'BeforePosition', 'link-17');">BeforePosition</a></tt><tt class="py-op">(</tt><tt id="link-18" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-18', 'max', 'link-14');">max</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">offset</tt> <tt class="py-op">+</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">location_string</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">location_string</tt><tt class="py-op">.</tt><tt id="link-19" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-19', 'startswith', 'link-15');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"&gt;"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># e.g. "&gt;13"</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-20" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-20', 'SeqFeature', 'link-7');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-21" class="py-name" targets="Class Bio.SeqFeature.AfterPosition=Bio.SeqFeature.AfterPosition-class.html"><a title="Bio.SeqFeature.AfterPosition" class="py-name" href="#" onclick="return doclink('link-21', 'AfterPosition', 'link-21');">AfterPosition</a></tt><tt class="py-op">(</tt><tt id="link-22" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-22', 'max', 'link-14');">max</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">offset</tt> <tt class="py-op">+</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">location_string</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">location_string</tt><tt class="py-op">.</tt><tt id="link-23" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-23', 'startswith', 'link-15');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"?"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># e.g. "?22"</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-24" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-24', 'SeqFeature', 'link-7');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-25" class="py-name" targets="Class Bio.SeqFeature.UncertainPosition=Bio.SeqFeature.UncertainPosition-class.html"><a title="Bio.SeqFeature.UncertainPosition" class="py-name" href="#" onclick="return doclink('link-25', 'UncertainPosition', 'link-25');">UncertainPosition</a></tt><tt class="py-op">(</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line">                <tt id="link-26" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-26', 'max', 'link-14');">max</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">offset</tt> <tt class="py-op">+</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">location_string</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line">            <tt class="py-op">)</tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line">    <tt class="py-keyword">raise</tt> <tt class="py-name">NotImplementedError</tt><tt class="py-op">(</tt><tt class="py-string">"Cannot parse location '%s'"</tt> <tt class="py-op">%</tt> <tt class="py-name">location_string</tt><tt class="py-op">)</tt> </tt>
</div><a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line"> </tt>
<a name="_make_seqfeature"></a><div id="_make_seqfeature-def"><a name="L59"></a><tt class="py-lineno"> 59</tt> <a class="py-toggle" href="#" id="_make_seqfeature-toggle" onclick="return toggle('_make_seqfeature');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SwissIO-module.html#_make_seqfeature">_make_seqfeature</a><tt class="py-op">(</tt><tt class="py-param">name</tt><tt class="py-op">,</tt> <tt class="py-param">from_res</tt><tt class="py-op">,</tt> <tt class="py-param">to_res</tt><tt class="py-op">,</tt> <tt class="py-param">description</tt><tt class="py-op">,</tt> <tt class="py-param">ft_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_make_seqfeature-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_make_seqfeature-expanded"><a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line">    <tt class="py-docstring">"""Construct SeqFeature from feature data from parser (PRIVATE)."""</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line">    <tt class="py-name">loc</tt> <tt class="py-op">=</tt> <tt id="link-27" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-27', 'SeqFeature', 'link-7');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-28" class="py-name" targets="Class Bio.SeqFeature.FeatureLocation=Bio.SeqFeature.FeatureLocation-class.html"><a title="Bio.SeqFeature.FeatureLocation" class="py-name" href="#" onclick="return doclink('link-28', 'FeatureLocation', 'link-28');">FeatureLocation</a></tt><tt class="py-op">(</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line">        <tt id="link-29" class="py-name" targets="Function Bio.SeqIO.SwissIO._make_position()=Bio.SeqIO.SwissIO-module.html#_make_position"><a title="Bio.SeqIO.SwissIO._make_position" class="py-name" href="#" onclick="return doclink('link-29', '_make_position', 'link-29');">_make_position</a></tt><tt class="py-op">(</tt><tt class="py-name">from_res</tt><tt class="py-op">,</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-30" class="py-name"><a title="Bio.SeqIO.SwissIO._make_position" class="py-name" href="#" onclick="return doclink('link-30', '_make_position', 'link-29');">_make_position</a></tt><tt class="py-op">(</tt><tt class="py-name">to_res</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line">    <tt class="py-op">)</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">ft_id</tt><tt class="py-op">:</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line">        <tt class="py-name">ft_id</tt> <tt class="py-op">=</tt> <tt class="py-string">"&lt;unknown id&gt;"</tt>  <tt class="py-comment"># The default in SeqFeature object</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-31" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-31', 'SeqFeature', 'link-7');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-32" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-32', 'SeqFeature', 'link-7');">SeqFeature</a></tt><tt class="py-op">(</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line">        <tt class="py-name">loc</tt><tt class="py-op">,</tt> <tt id="link-33" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.type()=Bio.Phylo.PhyloXMLIO.Writer-class.html#type"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-33', 'type', 'link-33');">type</a></tt><tt class="py-op">=</tt><tt id="link-34" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-34', 'name', 'link-34');">name</a></tt><tt class="py-op">,</tt> <tt id="link-35" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Variable Bio.PDB.Entity.Entity.id=Bio.PDB.Entity.Entity-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-35', 'id', 'link-35');">id</a></tt><tt class="py-op">=</tt><tt class="py-name">ft_id</tt><tt class="py-op">,</tt> <tt class="py-name">qualifiers</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-string">"description"</tt><tt class="py-op">:</tt> <tt id="link-36" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.description()=Bio.Phylo.PhyloXMLIO.Writer-class.html#description,Method Bio.SearchIO._legacy.NCBIStandalone._DescriptionConsumer.description()=Bio.SearchIO._legacy.NCBIStandalone._DescriptionConsumer-class.html#description,Variable Bio.SearchIO._model.hit.Hit.description=Bio.SearchIO._model.hit.Hit-class.html#description,Variable Bio.SearchIO._model.query.QueryResult.description=Bio.SearchIO._model.query.QueryResult-class.html#description"><a title="Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._legacy.NCBIStandalone._DescriptionConsumer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-36', 'description', 'link-36');">description</a></tt><tt class="py-op">}</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line">    <tt class="py-op">)</tt> </tt>
</div><a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line"> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line"> </tt>
<a name="SwissIterator"></a><div id="SwissIterator-def"><a name="L71"></a><tt class="py-lineno"> 71</tt> <a class="py-toggle" href="#" id="SwissIterator-toggle" onclick="return toggle('SwissIterator');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.SwissIO-module.html#SwissIterator">SwissIterator</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SwissIterator-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="SwissIterator-expanded"><a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line">    <tt class="py-docstring">"""Break up a Swiss-Prot/UniProt file into SeqRecord objects.</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line"><tt class="py-docstring">    Every section from the ID line to the terminating // becomes</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line"><tt class="py-docstring">    a single SeqRecord with associated annotation and features.</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line"><tt class="py-docstring">    This parser is for the flat file "swiss" format as used by:</tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line"><tt class="py-docstring">     - Swiss-Prot aka SwissProt</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line"><tt class="py-docstring">     - TrEMBL</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line"><tt class="py-docstring">     - UniProtKB aka UniProt Knowledgebase</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line"><tt class="py-docstring">    For consistency with BioPerl and EMBOSS we call this the "swiss"</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line"><tt class="py-docstring">    format. See also the SeqIO support for "uniprot-xml" format.</tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line"><tt class="py-docstring">    Rather than calling it directly, you are expected to use this</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line"><tt class="py-docstring">    parser via Bio.SeqIO.parse(..., format="swiss") instead.</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line">    <tt class="py-name">swiss_records</tt> <tt class="py-op">=</tt> <tt id="link-37" class="py-name"><a title="Bio.SwissProt" class="py-name" href="#" onclick="return doclink('link-37', 'SwissProt', 'link-9');">SwissProt</a></tt><tt class="py-op">.</tt><tt id="link-38" class="py-name" targets="Function Bio.AlignIO.parse()=Bio.AlignIO-module.html#parse,Function Bio.Blast.NCBIXML.parse()=Bio.Blast.NCBIXML-module.html#parse,Function Bio.Compass.parse()=Bio.Compass-module.html#parse,Function Bio.Emboss.Primer3.parse()=Bio.Emboss.Primer3-module.html#parse,Method Bio.Entrez.Parser.DataHandler.parse()=Bio.Entrez.Parser.DataHandler-class.html#parse,Function Bio.Entrez.parse()=Bio.Entrez-module.html#parse,Function Bio.ExPASy.Enzyme.parse()=Bio.ExPASy.Enzyme-module.html#parse,Function Bio.ExPASy.Prodoc.parse()=Bio.ExPASy.Prodoc-module.html#parse,Function Bio.ExPASy.Prosite.parse()=Bio.ExPASy.Prosite-module.html#parse,Function Bio.ExPASy.cellosaurus.parse()=Bio.ExPASy.cellosaurus-module.html#parse,Method Bio.GenBank.FeatureParser.parse()=Bio.GenBank.FeatureParser-class.html#parse,Method Bio.GenBank.RecordParser.parse()=Bio.GenBank.RecordParser-class.html#parse,Method Bio.GenBank.Scanner.InsdcScanner.parse()=Bio.GenBank.Scanner.InsdcScanner-class.html#parse,Function Bio.Geo.parse()=Bio.Geo-module.html#parse,Function Bio.KEGG.Compound.parse()=Bio.KEGG.Compound-module.html#parse,Function Bio.KEGG.Enzyme.parse()=Bio.KEGG.Enzyme-module.html#parse,Function Bio.KEGG.Gene.parse()=Bio.KEGG.Gene-module.html#parse,Method Bio.KEGG.KGML.KGML_parser.KGMLParser.parse()=Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html#parse,Function Bio.KEGG.KGML.KGML_parser.parse()=Bio.KEGG.KGML.KGML_parser-module.html#parse,Function Bio.KEGG.Map.parse()=Bio.KEGG.Map-module.html#parse,Function Bio.Medline.parse()=Bio.Medline-module.html#parse,Method Bio.Phylo.CDAOIO.Parser.parse()=Bio.Phylo.CDAOIO.Parser-class.html#parse,Function Bio.Phylo.CDAOIO.parse()=Bio.Phylo.CDAOIO-module.html#parse,Method Bio.Phylo.NeXMLIO.Parser.parse()=Bio.Phylo.NeXMLIO.Parser-class.html#parse,Function Bio.Phylo.NeXMLIO.parse()=Bio.Phylo.NeXMLIO-module.html#parse,Method Bio.Phylo.NewickIO.Parser.parse()=Bio.Phylo.NewickIO.Parser-class.html#parse,Function Bio.Phylo.NewickIO.parse()=Bio.Phylo.NewickIO-module.html#parse,Function Bio.Phylo.NexusIO.parse()=Bio.Phylo.NexusIO-module.html#parse,Method Bio.Phylo.PhyloXMLIO.Parser.parse()=Bio.Phylo.PhyloXMLIO.Parser-class.html#parse,Function Bio.Phylo.PhyloXMLIO.parse()=Bio.Phylo.PhyloXMLIO-module.html#parse,Function Bio.Phylo._io.parse()=Bio.Phylo._io-module.html#parse,Function Bio.SCOP.Cla.parse()=Bio.SCOP.Cla-module.html#parse,Function Bio.SCOP.Des.parse()=Bio.SCOP.Des-module.html#parse,Function Bio.SCOP.Dom.parse()=Bio.SCOP.Dom-module.html#parse,Function Bio.SCOP.Hie.parse()=Bio.SCOP.Hie-module.html#parse,Function Bio.SCOP.Raf.parse()=Bio.SCOP.Raf-module.html#parse,Method Bio.SearchIO._legacy.NCBIStandalone.BlastErrorParser.parse()=Bio.SearchIO._legacy.NCBIStandalone.BlastErrorParser-class.html#parse,Method Bio.SearchIO._legacy.NCBIStandalone.BlastParser.parse()=Bio.SearchIO._legacy.NCBIStandalone.BlastParser-class.html#parse,Method Bio.SearchIO._legacy.NCBIStandalone.PSIBlastParser.parse()=Bio.SearchIO._legacy.NCBIStandalone.PSIBlastParser-class.html#parse,Method Bio.SearchIO._legacy.ParserSupport.AbstractParser.parse()=Bio.SearchIO._legacy.ParserSupport.AbstractParser-class.html#parse,Function Bio.SearchIO.parse()=Bio.SearchIO-module.html#parse,Method Bio.SeqIO.UniprotIO.Parser.parse()=Bio.SeqIO.UniprotIO.Parser-class.html#parse,Function Bio.SeqIO.parse()=Bio.SeqIO-module.html#parse,Function Bio.Sequencing.Ace.parse()=Bio.Sequencing.Ace-module.html#parse,Function Bio.Sequencing.Phd.parse()=Bio.Sequencing.Phd-module.html#parse,Function Bio.SwissProt.KeyWList.parse()=Bio.SwissProt.KeyWList-module.html#parse,Function Bio.SwissProt.parse()=Bio.SwissProt-module.html#parse,Function Bio.UniGene.parse()=Bio.UniGene-module.html#parse,Function Bio.Wise.psw.parse()=Bio.Wise.psw-module.html#parse,Function Bio.motifs.parse()=Bio.motifs-module.html#parse,Function Bio.phenotype.parse()=Bio.phenotype-module.html#parse"><a title="Bio.AlignIO.parse
Bio.Blast.NCBIXML.parse
Bio.Compass.parse
Bio.Emboss.Primer3.parse
Bio.Entrez.Parser.DataHandler.parse
Bio.Entrez.parse
Bio.ExPASy.Enzyme.parse
Bio.ExPASy.Prodoc.parse
Bio.ExPASy.Prosite.parse
Bio.ExPASy.cellosaurus.parse
Bio.GenBank.FeatureParser.parse
Bio.GenBank.RecordParser.parse
Bio.GenBank.Scanner.InsdcScanner.parse
Bio.Geo.parse
Bio.KEGG.Compound.parse
Bio.KEGG.Enzyme.parse
Bio.KEGG.Gene.parse
Bio.KEGG.KGML.KGML_parser.KGMLParser.parse
Bio.KEGG.KGML.KGML_parser.parse
Bio.KEGG.Map.parse
Bio.Medline.parse
Bio.Phylo.CDAOIO.Parser.parse
Bio.Phylo.CDAOIO.parse
Bio.Phylo.NeXMLIO.Parser.parse
Bio.Phylo.NeXMLIO.parse
Bio.Phylo.NewickIO.Parser.parse
Bio.Phylo.NewickIO.parse
Bio.Phylo.NexusIO.parse
Bio.Phylo.PhyloXMLIO.Parser.parse
Bio.Phylo.PhyloXMLIO.parse
Bio.Phylo._io.parse
Bio.SCOP.Cla.parse
Bio.SCOP.Des.parse
Bio.SCOP.Dom.parse
Bio.SCOP.Hie.parse
Bio.SCOP.Raf.parse
Bio.SearchIO._legacy.NCBIStandalone.BlastErrorParser.parse
Bio.SearchIO._legacy.NCBIStandalone.BlastParser.parse
Bio.SearchIO._legacy.NCBIStandalone.PSIBlastParser.parse
Bio.SearchIO._legacy.ParserSupport.AbstractParser.parse
Bio.SearchIO.parse
Bio.SeqIO.UniprotIO.Parser.parse
Bio.SeqIO.parse
Bio.Sequencing.Ace.parse
Bio.Sequencing.Phd.parse
Bio.SwissProt.KeyWList.parse
Bio.SwissProt.parse
Bio.UniGene.parse
Bio.Wise.psw.parse
Bio.motifs.parse
Bio.phenotype.parse" class="py-name" href="#" onclick="return doclink('link-38', 'parse', 'link-38');">parse</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">swiss_record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">swiss_records</tt><tt class="py-op">:</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line">        <tt class="py-comment"># Convert the SwissProt record to a SeqRecord</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">        <tt id="link-39" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.seq=Bio.SeqRecord.SeqRecord-class.html#seq,Variable BioSQL.BioSeq.DBSeqRecord.seq=BioSQL.BioSeq.DBSeqRecord-class.html#seq"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-39', 'seq', 'link-39');">seq</a></tt> <tt class="py-op">=</tt> <tt id="link-40" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-40', 'Seq', 'link-1');">Seq</a></tt><tt class="py-op">.</tt><tt id="link-41" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-41', 'Seq', 'link-1');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt id="link-42" class="py-name" targets="Method Bio.FSSP.FSSPAlignDict.sequence()=Bio.FSSP.FSSPAlignDict-class.html#sequence,Method Bio.GenBank._FeatureConsumer.sequence()=Bio.GenBank._FeatureConsumer-class.html#sequence,Method Bio.GenBank._RecordConsumer.sequence()=Bio.GenBank._RecordConsumer-class.html#sequence,Method Bio.Phylo.PhyloXMLIO.Writer.sequence()=Bio.Phylo.PhyloXMLIO.Writer-class.html#sequence"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-42', 'sequence', 'link-42');">sequence</a></tt><tt class="py-op">,</tt> <tt id="link-43" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-43', 'Alphabet', 'link-5');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-44" class="py-name" targets="Variable Bio.Alphabet.generic_protein=Bio.Alphabet-module.html#generic_protein"><a title="Bio.Alphabet.generic_protein" class="py-name" href="#" onclick="return doclink('link-44', 'generic_protein', 'link-44');">generic_protein</a></tt><tt class="py-op">)</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line">        <tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt id="link-45" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-45', 'SeqRecord', 'link-3');">SeqRecord</a></tt><tt class="py-op">.</tt><tt id="link-46" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-46', 'SeqRecord', 'link-3');">SeqRecord</a></tt><tt class="py-op">(</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">            <tt id="link-47" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-47', 'seq', 'link-39');">seq</a></tt><tt class="py-op">,</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line">            <tt id="link-48" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-48', 'id', 'link-35');">id</a></tt><tt class="py-op">=</tt><tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">accessions</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line">            <tt id="link-49" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-49', 'name', 'link-34');">name</a></tt><tt class="py-op">=</tt><tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">entry_name</tt><tt class="py-op">,</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line">            <tt id="link-50" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._legacy.NCBIStandalone._DescriptionConsumer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-50', 'description', 'link-36');">description</a></tt><tt class="py-op">=</tt><tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt id="link-51" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._legacy.NCBIStandalone._DescriptionConsumer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-51', 'description', 'link-36');">description</a></tt><tt class="py-op">,</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line">            <tt id="link-52" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.features=BioSQL.BioSeq.DBSeqRecord-class.html#features"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-52', 'features', 'link-52');">features</a></tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt id="link-53" class="py-name" targets="Function Bio.SeqIO.SwissIO._make_seqfeature()=Bio.SeqIO.SwissIO-module.html#_make_seqfeature"><a title="Bio.SeqIO.SwissIO._make_seqfeature" class="py-name" href="#" onclick="return doclink('link-53', '_make_seqfeature', 'link-53');">_make_seqfeature</a></tt><tt class="py-op">(</tt><tt class="py-op">*</tt><tt class="py-name">f</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">f</tt> <tt class="py-keyword">in</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt id="link-54" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-54', 'features', 'link-52');">features</a></tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line">        <tt class="py-op">)</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">        <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-55" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._legacy.NCBIStandalone._DescriptionConsumer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-55', 'description', 'link-36');">description</a></tt> <tt class="py-op">=</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt id="link-56" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._legacy.NCBIStandalone._DescriptionConsumer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-56', 'description', 'link-36');">description</a></tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">cross_reference</tt> <tt class="py-keyword">in</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">cross_references</tt><tt class="py-op">:</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">cross_reference</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">            <tt id="link-57" class="py-name" targets="Method Bio.SearchIO._legacy.NCBIStandalone._DatabaseReportConsumer.database()=Bio.SearchIO._legacy.NCBIStandalone._DatabaseReportConsumer-class.html#database"><a title="Bio.SearchIO._legacy.NCBIStandalone._DatabaseReportConsumer.database" class="py-name" href="#" onclick="return doclink('link-57', 'database', 'link-57');">database</a></tt><tt class="py-op">,</tt> <tt id="link-58" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.accession()=Bio.GenBank._FeatureConsumer-class.html#accession,Method Bio.GenBank._RecordConsumer.accession()=Bio.GenBank._RecordConsumer-class.html#accession,Method Bio.Phylo.PhyloXMLIO.Parser.accession()=Bio.Phylo.PhyloXMLIO.Parser-class.html#accession,Method Bio.Phylo.PhyloXMLIO.Writer.accession()=Bio.Phylo.PhyloXMLIO.Writer-class.html#accession"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-58', 'accession', 'link-58');">accession</a></tt> <tt class="py-op">=</tt> <tt class="py-name">cross_reference</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line">            <tt class="py-name">dbxref</tt> <tt class="py-op">=</tt> <tt class="py-string">"%s:%s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-59" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._DatabaseReportConsumer.database" class="py-name" href="#" onclick="return doclink('link-59', 'database', 'link-57');">database</a></tt><tt class="py-op">,</tt> <tt id="link-60" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-60', 'accession', 'link-58');">accession</a></tt><tt class="py-op">)</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">dbxref</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-61" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.dbxrefs=BioSQL.BioSeq.DBSeqRecord-class.html#dbxrefs"><a title="BioSQL.BioSeq.DBSeqRecord.dbxrefs" class="py-name" href="#" onclick="return doclink('link-61', 'dbxrefs', 'link-61');">dbxrefs</a></tt><tt class="py-op">:</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">                <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-62" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.dbxrefs" class="py-name" href="#" onclick="return doclink('link-62', 'dbxrefs', 'link-61');">dbxrefs</a></tt><tt class="py-op">.</tt><tt id="link-63" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-63', 'append', 'link-63');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">dbxref</tt><tt class="py-op">)</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line">        <tt id="link-64" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.annotations=BioSQL.BioSeq.DBSeqRecord-class.html#annotations"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-64', 'annotations', 'link-64');">annotations</a></tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-65" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-65', 'annotations', 'link-64');">annotations</a></tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line">        <tt id="link-66" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-66', 'annotations', 'link-64');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"accessions"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">accessions</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">protein_existence</tt><tt class="py-op">:</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">            <tt id="link-67" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-67', 'annotations', 'link-64');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"protein_existence"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">protein_existence</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">created</tt><tt class="py-op">:</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">            <tt id="link-68" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-68', 'annotations', 'link-64');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"date"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">created</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">            <tt id="link-69" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-69', 'annotations', 'link-64');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"sequence_version"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">created</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">sequence_update</tt><tt class="py-op">:</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">            <tt id="link-70" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-70', 'annotations', 'link-64');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"date_last_sequence_update"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">sequence_update</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">            <tt id="link-71" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-71', 'annotations', 'link-64');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"sequence_version"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">sequence_update</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">annotation_update</tt><tt class="py-op">:</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line">            <tt id="link-72" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-72', 'annotations', 'link-64');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"date_last_annotation_update"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">annotation_update</tt><tt class="py-op">[</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line">                <tt class="py-number">0</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">            <tt class="py-op">]</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">            <tt id="link-73" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-73', 'annotations', 'link-64');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"entry_version"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">annotation_update</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">gene_name</tt><tt class="py-op">:</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">            <tt id="link-74" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-74', 'annotations', 'link-64');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"gene_name"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">gene_name</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">        <tt id="link-75" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-75', 'annotations', 'link-64');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"organism"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt id="link-76" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.organism()=Bio.GenBank._FeatureConsumer-class.html#organism,Method Bio.GenBank._RecordConsumer.organism()=Bio.GenBank._RecordConsumer-class.html#organism"><a title="Bio.GenBank._FeatureConsumer.organism
Bio.GenBank._RecordConsumer.organism" class="py-name" href="#" onclick="return doclink('link-76', 'organism', 'link-76');">organism</a></tt><tt class="py-op">.</tt><tt id="link-77" class="py-name" targets="Method Bio.Seq.Seq.rstrip()=Bio.Seq.Seq-class.html#rstrip"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-77', 'rstrip', 'link-77');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-string">"."</tt><tt class="py-op">)</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line">        <tt id="link-78" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-78', 'annotations', 'link-64');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"taxonomy"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">organism_classification</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line">        <tt id="link-79" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-79', 'annotations', 'link-64');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"ncbi_taxid"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">taxonomy_id</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">host_organism</tt><tt class="py-op">:</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line">            <tt id="link-80" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-80', 'annotations', 'link-64');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"organism_host"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">host_organism</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">host_taxonomy_id</tt><tt class="py-op">:</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line">            <tt id="link-81" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-81', 'annotations', 'link-64');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"host_ncbi_taxid"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">host_taxonomy_id</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">comments</tt><tt class="py-op">:</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line">            <tt id="link-82" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-82', 'annotations', 'link-64');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"comment"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">"\n"</tt><tt class="py-op">.</tt><tt id="link-83" class="py-name" targets="Method Bio.Seq.MutableSeq.join()=Bio.Seq.MutableSeq-class.html#join,Method Bio.Seq.Seq.join()=Bio.Seq.Seq-class.html#join,Method Bio.Seq.UnknownSeq.join()=Bio.Seq.UnknownSeq-class.html#join"><a title="Bio.Seq.MutableSeq.join
Bio.Seq.Seq.join
Bio.Seq.UnknownSeq.join" class="py-name" href="#" onclick="return doclink('link-83', 'join', 'link-83');">join</a></tt><tt class="py-op">(</tt><tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">comments</tt><tt class="py-op">)</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">references</tt><tt class="py-op">:</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line">            <tt id="link-84" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-84', 'annotations', 'link-64');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"references"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-85" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.reference()=Bio.Phylo.PhyloXMLIO.Parser-class.html#reference,Method Bio.Phylo.PhyloXMLIO.Writer.reference()=Bio.Phylo.PhyloXMLIO.Writer-class.html#reference,Method Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.reference()=Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer-class.html#reference"><a title="Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference
Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.reference" class="py-name" href="#" onclick="return doclink('link-85', 'reference', 'link-85');">reference</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt class="py-name">references</tt><tt class="py-op">:</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">                <tt class="py-name">feature</tt> <tt class="py-op">=</tt> <tt id="link-86" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-86', 'SeqFeature', 'link-7');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-87" class="py-name" targets="Class Bio.ExPASy.Prodoc.Reference=Bio.ExPASy.Prodoc.Reference-class.html,Class Bio.GenBank.Record.Reference=Bio.GenBank.Record.Reference-class.html,Class Bio.Phylo.PhyloXML.Reference=Bio.Phylo.PhyloXML.Reference-class.html,Class Bio.SeqFeature.Reference=Bio.SeqFeature.Reference-class.html,Class Bio.SwissProt.Reference=Bio.SwissProt.Reference-class.html"><a title="Bio.ExPASy.Prodoc.Reference
Bio.GenBank.Record.Reference
Bio.Phylo.PhyloXML.Reference
Bio.SeqFeature.Reference
Bio.SwissProt.Reference" class="py-name" href="#" onclick="return doclink('link-87', 'Reference', 'link-87');">Reference</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line">                <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-88" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.comment()=Bio.GenBank._FeatureConsumer-class.html#comment,Method Bio.GenBank._RecordConsumer.comment()=Bio.GenBank._RecordConsumer-class.html#comment"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-88', 'comment', 'link-88');">comment</a></tt> <tt class="py-op">=</tt> <tt class="py-string">" "</tt><tt class="py-op">.</tt><tt id="link-89" class="py-name"><a title="Bio.Seq.MutableSeq.join
Bio.Seq.Seq.join
Bio.Seq.UnknownSeq.join" class="py-name" href="#" onclick="return doclink('link-89', 'join', 'link-83');">join</a></tt><tt class="py-op">(</tt><tt class="py-string">"%s=%s;"</tt> <tt class="py-op">%</tt> <tt class="py-name">k_v</tt> <tt class="py-keyword">for</tt> <tt class="py-name">k_v</tt> <tt class="py-keyword">in</tt> <tt id="link-90" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference
Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.reference" class="py-name" href="#" onclick="return doclink('link-90', 'reference', 'link-85');">reference</a></tt><tt class="py-op">.</tt><tt class="py-name">comments</tt><tt class="py-op">)</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt id="link-91" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.value()=Bio.Phylo.PhyloXMLIO.Writer-class.html#value"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-91', 'value', 'link-91');">value</a></tt> <tt class="py-keyword">in</tt> <tt id="link-92" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference
Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.reference" class="py-name" href="#" onclick="return doclink('link-92', 'reference', 'link-85');">reference</a></tt><tt class="py-op">.</tt><tt class="py-name">references</tt><tt class="py-op">:</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-op">==</tt> <tt class="py-string">"PubMed"</tt><tt class="py-op">:</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line">                        <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-93" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.pubmed_id()=Bio.GenBank._FeatureConsumer-class.html#pubmed_id,Method Bio.GenBank._RecordConsumer.pubmed_id()=Bio.GenBank._RecordConsumer-class.html#pubmed_id"><a title="Bio.GenBank._FeatureConsumer.pubmed_id
Bio.GenBank._RecordConsumer.pubmed_id" class="py-name" href="#" onclick="return doclink('link-93', 'pubmed_id', 'link-93');">pubmed_id</a></tt> <tt class="py-op">=</tt> <tt id="link-94" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-94', 'value', 'link-91');">value</a></tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line">                    <tt class="py-keyword">elif</tt> <tt class="py-name">key</tt> <tt class="py-op">==</tt> <tt class="py-string">"MEDLINE"</tt><tt class="py-op">:</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line">                        <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-95" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.medline_id()=Bio.GenBank._FeatureConsumer-class.html#medline_id,Method Bio.GenBank._RecordConsumer.medline_id()=Bio.GenBank._RecordConsumer-class.html#medline_id"><a title="Bio.GenBank._FeatureConsumer.medline_id
Bio.GenBank._RecordConsumer.medline_id" class="py-name" href="#" onclick="return doclink('link-95', 'medline_id', 'link-95');">medline_id</a></tt> <tt class="py-op">=</tt> <tt id="link-96" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-96', 'value', 'link-91');">value</a></tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line">                    <tt class="py-keyword">elif</tt> <tt class="py-name">key</tt> <tt class="py-op">==</tt> <tt class="py-string">"DOI"</tt><tt class="py-op">:</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line">                        <tt class="py-keyword">pass</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">                    <tt class="py-keyword">elif</tt> <tt class="py-name">key</tt> <tt class="py-op">==</tt> <tt class="py-string">"AGRICOLA"</tt><tt class="py-op">:</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">                        <tt class="py-keyword">pass</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">                        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unknown key %s found in references"</tt> <tt class="py-op">%</tt> <tt class="py-name">key</tt><tt class="py-op">)</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line">                <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-97" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.authors()=Bio.GenBank._FeatureConsumer-class.html#authors,Method Bio.GenBank._RecordConsumer.authors()=Bio.GenBank._RecordConsumer-class.html#authors"><a title="Bio.GenBank._FeatureConsumer.authors
Bio.GenBank._RecordConsumer.authors" class="py-name" href="#" onclick="return doclink('link-97', 'authors', 'link-97');">authors</a></tt> <tt class="py-op">=</tt> <tt id="link-98" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference
Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.reference" class="py-name" href="#" onclick="return doclink('link-98', 'reference', 'link-85');">reference</a></tt><tt class="py-op">.</tt><tt id="link-99" class="py-name"><a title="Bio.GenBank._FeatureConsumer.authors
Bio.GenBank._RecordConsumer.authors" class="py-name" href="#" onclick="return doclink('link-99', 'authors', 'link-97');">authors</a></tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">                <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-100" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.title()=Bio.GenBank._FeatureConsumer-class.html#title,Method Bio.GenBank._RecordConsumer.title()=Bio.GenBank._RecordConsumer-class.html#title,Method Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.title()=Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer-class.html#title"><a title="Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title
Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.title" class="py-name" href="#" onclick="return doclink('link-100', 'title', 'link-100');">title</a></tt> <tt class="py-op">=</tt> <tt id="link-101" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference
Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.reference" class="py-name" href="#" onclick="return doclink('link-101', 'reference', 'link-85');">reference</a></tt><tt class="py-op">.</tt><tt id="link-102" class="py-name"><a title="Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title
Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.title" class="py-name" href="#" onclick="return doclink('link-102', 'title', 'link-100');">title</a></tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">                <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-103" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.journal()=Bio.GenBank._FeatureConsumer-class.html#journal,Method Bio.GenBank._RecordConsumer.journal()=Bio.GenBank._RecordConsumer-class.html#journal"><a title="Bio.GenBank._FeatureConsumer.journal
Bio.GenBank._RecordConsumer.journal" class="py-name" href="#" onclick="return doclink('link-103', 'journal', 'link-103');">journal</a></tt> <tt class="py-op">=</tt> <tt id="link-104" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference
Bio.SearchIO._legacy.NCBIStandalone._HeaderConsumer.reference" class="py-name" href="#" onclick="return doclink('link-104', 'reference', 'link-85');">reference</a></tt><tt class="py-op">.</tt><tt id="link-105" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.location()=Bio.GenBank._FeatureConsumer-class.html#location,Method Bio.GenBank._RecordConsumer.location()=Bio.GenBank._RecordConsumer-class.html#location,Method Bio.Phylo.PhyloXMLIO.Writer.location()=Bio.Phylo.PhyloXMLIO.Writer-class.html#location"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-105', 'location', 'link-105');">location</a></tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">                <tt id="link-106" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-106', 'annotations', 'link-64');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"references"</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-107" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-107', 'append', 'link-63');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">feature</tt><tt class="py-op">)</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt id="link-108" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.keywords()=Bio.GenBank._FeatureConsumer-class.html#keywords,Method Bio.GenBank._RecordConsumer.keywords()=Bio.GenBank._RecordConsumer-class.html#keywords"><a title="Bio.GenBank._FeatureConsumer.keywords
Bio.GenBank._RecordConsumer.keywords" class="py-name" href="#" onclick="return doclink('link-108', 'keywords', 'link-108');">keywords</a></tt><tt class="py-op">:</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-109" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-109', 'annotations', 'link-64');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"keywords"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">swiss_record</tt><tt class="py-op">.</tt><tt id="link-110" class="py-name"><a title="Bio.GenBank._FeatureConsumer.keywords
Bio.GenBank._RecordConsumer.keywords" class="py-name" href="#" onclick="return doclink('link-110', 'keywords', 'link-108');">keywords</a></tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">        <tt class="py-keyword">yield</tt> <tt class="py-name">record</tt> </tt>
</div><a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Nov  6 13:33:48 2019
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
