<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.PDB.ResidueDepth'</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.PDB-module.html">Package&nbsp;PDB</a> ::
        Module&nbsp;ResidueDepth'
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.PDB.ResidueDepth%27-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.PDB.ResidueDepth%27-module.html">Module Bio.PDB.ResidueDepth'</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright (C) 2002, Thomas Hamelryck (thamelry@binf.ku.dk)</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># Copyright (C) 2017, Joao Rodrigues (j.p.g.l.m.rodrigues@gmail.com)</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-docstring">"""Calculation of residue depth using command line tool MSMS.</tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-docstring">This module uses Michel Sanner's MSMS program for the surface calculation.</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-docstring">See: http://mgltools.scripps.edu/packages/MSMS</tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-docstring">Residue depth is the average distance of the atoms of a residue from</tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-docstring">the solvent accessible surface.</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-docstring">Residue Depth::</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-docstring">    from Bio.PDB.ResidueDepth import ResidueDepth</tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-docstring">    from Bio.PDB.PDBParser import PDBParser</tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-docstring">    parser = PDBParser()</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-docstring">    structure = parser.get_structure("1a8o", "Tests/PDB/1A8O.pdb")</tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-docstring">    model = structure[0]</tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-docstring">    rd = ResidueDepth(model)</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-docstring">    print(rd['A',(' ', 152, ' ')])</tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-docstring">Direct MSMS interface, typical use::</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"><tt class="py-docstring">    from Bio.PDB.ResidueDepth import get_surface</tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt class="py-docstring">    surface = get_surface(model)</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"><tt class="py-docstring">The surface is a Numeric array with all the surface vertices.</tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"><tt class="py-docstring">Distance to surface::</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-docstring">    from Bio.PDB.ResidueDepth import min_dist</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-docstring">    coord = (1.113, 35.393,  9.268)</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"><tt class="py-docstring">    dist = min_dist(coord, surface)</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"><tt class="py-docstring">where coord is the coord of an atom within the volume bound by</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"><tt class="py-docstring">the surface (ie. atom depth).</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"><tt class="py-docstring">To calculate the residue depth (average atom depth of the atoms</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"><tt class="py-docstring">in a residue)::</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line"><tt class="py-docstring">    from Bio.PDB.ResidueDepth import residue_depth</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line"><tt class="py-docstring">    chain = model['A']</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line"><tt class="py-docstring">    res152 = chain[152]</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line"><tt class="py-docstring">    rd = residue_depth(res152, surface)</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line"> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">__future__</tt> <tt class="py-keyword">import</tt> <tt class="py-name">print_function</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line"> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">os</tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">tempfile</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line"> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">numpy</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line"> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio.PDB=Bio.PDB-module.html"><a title="Bio.PDB" class="py-name" href="#" onclick="return doclink('link-1', 'PDB', 'link-1');">PDB</a></tt> <tt class="py-keyword">import</tt> <tt id="link-2" class="py-name" targets="Class Bio.PDB.PDBParser'.PDBParser=Bio.PDB.PDBParser%27.PDBParser-class.html"><a title="Bio.PDB.PDBParser'.PDBParser" class="py-name" href="#" onclick="return doclink('link-2', 'PDBParser', 'link-2');">PDBParser</a></tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-3" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-3', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-4" class="py-name"><a title="Bio.PDB" class="py-name" href="#" onclick="return doclink('link-4', 'PDB', 'link-1');">PDB</a></tt> <tt class="py-keyword">import</tt> <tt id="link-5" class="py-name" targets="Module Bio.PDB.Selection=Bio.PDB.Selection-module.html"><a title="Bio.PDB.Selection" class="py-name" href="#" onclick="return doclink('link-5', 'Selection', 'link-5');">Selection</a></tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-6" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-6', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-7" class="py-name"><a title="Bio.PDB" class="py-name" href="#" onclick="return doclink('link-7', 'PDB', 'link-1');">PDB</a></tt><tt class="py-op">.</tt><tt id="link-8" class="py-name" targets="Module Bio.PDB.AbstractPropertyMap=Bio.PDB.AbstractPropertyMap-module.html,Class Bio.PDB.AbstractPropertyMap.AbstractPropertyMap=Bio.PDB.AbstractPropertyMap.AbstractPropertyMap-class.html"><a title="Bio.PDB.AbstractPropertyMap
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap" class="py-name" href="#" onclick="return doclink('link-8', 'AbstractPropertyMap', 'link-8');">AbstractPropertyMap</a></tt> <tt class="py-keyword">import</tt> <tt id="link-9" class="py-name"><a title="Bio.PDB.AbstractPropertyMap
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap" class="py-name" href="#" onclick="return doclink('link-9', 'AbstractPropertyMap', 'link-8');">AbstractPropertyMap</a></tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-10" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-10', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-11" class="py-name"><a title="Bio.PDB" class="py-name" href="#" onclick="return doclink('link-11', 'PDB', 'link-1');">PDB</a></tt><tt class="py-op">.</tt><tt id="link-12" class="py-name" targets="Module Bio.PDB.Polypeptide=Bio.PDB.Polypeptide-module.html,Class Bio.PDB.Polypeptide.Polypeptide=Bio.PDB.Polypeptide.Polypeptide-class.html"><a title="Bio.PDB.Polypeptide
Bio.PDB.Polypeptide.Polypeptide" class="py-name" href="#" onclick="return doclink('link-12', 'Polypeptide', 'link-12');">Polypeptide</a></tt> <tt class="py-keyword">import</tt> <tt id="link-13" class="py-name" targets="Function Bio.PDB.Polypeptide.is_aa()=Bio.PDB.Polypeptide-module.html#is_aa"><a title="Bio.PDB.Polypeptide.is_aa" class="py-name" href="#" onclick="return doclink('link-13', 'is_aa', 'link-13');">is_aa</a></tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line"> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-14" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-14', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-15" class="py-name" targets="Class Bio.BiopythonWarning=Bio.BiopythonWarning-class.html"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-15', 'BiopythonWarning', 'link-15');">BiopythonWarning</a></tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-16" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-16', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-17" class="py-name" targets="Class Bio.BiopythonDeprecationWarning=Bio.BiopythonDeprecationWarning-class.html"><a title="Bio.BiopythonDeprecationWarning" class="py-name" href="#" onclick="return doclink('link-17', 'BiopythonDeprecationWarning', 'link-17');">BiopythonDeprecationWarning</a></tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line"> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line"><tt class="py-comment"># PDB_TO_XYZR is a BASH script and will not run on Windows</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line"><tt class="py-comment"># Since it only reads atmtypenumbers to a mapping structure we can replicate</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line"><tt class="py-comment"># that functionality here and avoid this dependency altogether.</tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line"><tt class="py-comment"># Description of PDB_TO_XYZR</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line"><tt class="py-comment"># Maps residue type and atom name pairs into Connolly ".atm" numeric codes</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line"><tt class="py-comment">#  as used in MS and AMS, and into actual radius values</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line"><tt class="py-comment"># In case of missing radius, use 0.01</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line"><tt class="py-comment"># Table 1: Atom Type to radius</tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line"><tt id="link-18" class="py-name" targets="Variable Bio.PDB.ResidueDepth'._atomic_radii=Bio.PDB.ResidueDepth%27-module.html#_atomic_radii"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-18', '_atomic_radii', 'link-18');">_atomic_radii</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line">    <tt class="py-comment">#   atom num dist  Rexplicit Runited-atom</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line">    <tt class="py-number">1</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0.57</tt><tt class="py-op">,</tt> <tt class="py-number">1.40</tt><tt class="py-op">,</tt> <tt class="py-number">1.40</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">    <tt class="py-number">2</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0.66</tt><tt class="py-op">,</tt> <tt class="py-number">1.40</tt><tt class="py-op">,</tt> <tt class="py-number">1.60</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line">    <tt class="py-number">3</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0.57</tt><tt class="py-op">,</tt> <tt class="py-number">1.40</tt><tt class="py-op">,</tt> <tt class="py-number">1.40</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line">    <tt class="py-number">4</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0.70</tt><tt class="py-op">,</tt> <tt class="py-number">1.54</tt><tt class="py-op">,</tt> <tt class="py-number">1.70</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line">    <tt class="py-number">5</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0.70</tt><tt class="py-op">,</tt> <tt class="py-number">1.54</tt><tt class="py-op">,</tt> <tt class="py-number">1.80</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line">    <tt class="py-number">6</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0.70</tt><tt class="py-op">,</tt> <tt class="py-number">1.54</tt><tt class="py-op">,</tt> <tt class="py-number">2.00</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line">    <tt class="py-number">7</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0.77</tt><tt class="py-op">,</tt> <tt class="py-number">1.74</tt><tt class="py-op">,</tt> <tt class="py-number">2.00</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line">    <tt class="py-number">8</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0.77</tt><tt class="py-op">,</tt> <tt class="py-number">1.74</tt><tt class="py-op">,</tt> <tt class="py-number">2.00</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line">    <tt class="py-number">9</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0.77</tt><tt class="py-op">,</tt> <tt class="py-number">1.74</tt><tt class="py-op">,</tt> <tt class="py-number">2.00</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line">    <tt class="py-number">10</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0.67</tt><tt class="py-op">,</tt> <tt class="py-number">1.74</tt><tt class="py-op">,</tt> <tt class="py-number">1.74</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line">    <tt class="py-number">11</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0.70</tt><tt class="py-op">,</tt> <tt class="py-number">1.74</tt><tt class="py-op">,</tt> <tt class="py-number">1.86</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">    <tt class="py-number">12</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">1.04</tt><tt class="py-op">,</tt> <tt class="py-number">1.80</tt><tt class="py-op">,</tt> <tt class="py-number">1.85</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line">    <tt class="py-number">13</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">1.04</tt><tt class="py-op">,</tt> <tt class="py-number">1.80</tt><tt class="py-op">,</tt> <tt class="py-number">1.80</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  <tt class="py-comment"># P, S, and LonePairs</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">    <tt class="py-number">14</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0.70</tt><tt class="py-op">,</tt> <tt class="py-number">1.54</tt><tt class="py-op">,</tt> <tt class="py-number">1.54</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  <tt class="py-comment"># non-protonated nitrogens</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line">    <tt class="py-number">15</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0.37</tt><tt class="py-op">,</tt> <tt class="py-number">1.20</tt><tt class="py-op">,</tt> <tt class="py-number">1.20</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  <tt class="py-comment"># H, D  hydrogen and deuterium</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line">    <tt class="py-number">16</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0.70</tt><tt class="py-op">,</tt> <tt class="py-number">0.00</tt><tt class="py-op">,</tt> <tt class="py-number">1.50</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  <tt class="py-comment"># obsolete entry, purpose unknown</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line">    <tt class="py-number">17</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">3.50</tt><tt class="py-op">,</tt> <tt class="py-number">5.00</tt><tt class="py-op">,</tt> <tt class="py-number">5.00</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  <tt class="py-comment"># pseudoatom - big ball</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line">    <tt class="py-number">18</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">1.74</tt><tt class="py-op">,</tt> <tt class="py-number">1.97</tt><tt class="py-op">,</tt> <tt class="py-number">1.97</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  <tt class="py-comment"># Ca calcium</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line">    <tt class="py-number">19</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">1.25</tt><tt class="py-op">,</tt> <tt class="py-number">1.40</tt><tt class="py-op">,</tt> <tt class="py-number">1.40</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  <tt class="py-comment"># Zn zinc    (traditional radius)</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">    <tt class="py-number">20</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">1.17</tt><tt class="py-op">,</tt> <tt class="py-number">1.40</tt><tt class="py-op">,</tt> <tt class="py-number">1.40</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  <tt class="py-comment"># Cu copper  (traditional radius)</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line">    <tt class="py-number">21</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">1.45</tt><tt class="py-op">,</tt> <tt class="py-number">1.30</tt><tt class="py-op">,</tt> <tt class="py-number">1.30</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  <tt class="py-comment"># Fe heme iron</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line">    <tt class="py-number">22</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">1.41</tt><tt class="py-op">,</tt> <tt class="py-number">1.49</tt><tt class="py-op">,</tt> <tt class="py-number">1.49</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  <tt class="py-comment"># Cd cadmium</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line">    <tt class="py-number">23</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0.01</tt><tt class="py-op">,</tt> <tt class="py-number">0.01</tt><tt class="py-op">,</tt> <tt class="py-number">0.01</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  <tt class="py-comment"># pseudoatom - tiny dot</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">    <tt class="py-number">24</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0.37</tt><tt class="py-op">,</tt> <tt class="py-number">1.20</tt><tt class="py-op">,</tt> <tt class="py-number">0.00</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  <tt class="py-comment"># hydrogen vanishing if united-atoms</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line">    <tt class="py-number">25</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">1.16</tt><tt class="py-op">,</tt> <tt class="py-number">1.24</tt><tt class="py-op">,</tt> <tt class="py-number">1.24</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  <tt class="py-comment"># Fe not in heme</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">    <tt class="py-number">26</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">1.36</tt><tt class="py-op">,</tt> <tt class="py-number">1.60</tt><tt class="py-op">,</tt> <tt class="py-number">1.60</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  <tt class="py-comment"># Mg magnesium</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">    <tt class="py-number">27</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">1.17</tt><tt class="py-op">,</tt> <tt class="py-number">1.24</tt><tt class="py-op">,</tt> <tt class="py-number">1.24</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  <tt class="py-comment"># Mn manganese</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line">    <tt class="py-number">28</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">1.16</tt><tt class="py-op">,</tt> <tt class="py-number">1.25</tt><tt class="py-op">,</tt> <tt class="py-number">1.25</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  <tt class="py-comment"># Co cobalt</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line">    <tt class="py-number">29</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">1.17</tt><tt class="py-op">,</tt> <tt class="py-number">2.15</tt><tt class="py-op">,</tt> <tt class="py-number">2.15</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  <tt class="py-comment"># Se selenium</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">    <tt class="py-number">30</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">3.00</tt><tt class="py-op">,</tt> <tt class="py-number">3.00</tt><tt class="py-op">,</tt> <tt class="py-number">3.00</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  <tt class="py-comment"># obsolete entry, original purpose unknown</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">    <tt class="py-number">31</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">1.15</tt><tt class="py-op">,</tt> <tt class="py-number">1.15</tt><tt class="py-op">,</tt> <tt class="py-number">1.15</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  <tt class="py-comment"># Yb ytterbium +3 ion --- wild guess only</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">    <tt class="py-number">38</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0.95</tt><tt class="py-op">,</tt> <tt class="py-number">1.80</tt><tt class="py-op">,</tt> <tt class="py-number">1.80</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  <tt class="py-comment"># obsolete entry, original purpose unknown</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line"><tt class="py-op">}</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line"> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line"><tt class="py-comment"># Table 2: Resname/Aname to Atom Type</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line"><tt class="py-comment"># MSMS uses an awk/gawk pattern matching strategy that we cannot replicate</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line"><tt class="py-comment"># We will take advantage of our parser to help us in the mapping.</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line"> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line"> </tt>
<a name="_get_atom_radius"></a><div id="_get_atom_radius-def"><a name="L119"></a><tt class="py-lineno">119</tt> <a class="py-toggle" href="#" id="_get_atom_radius-toggle" onclick="return toggle('_get_atom_radius');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.PDB.ResidueDepth'-module.html#_get_atom_radius">_get_atom_radius</a><tt class="py-op">(</tt><tt class="py-param">atom</tt><tt class="py-op">,</tt> <tt class="py-param">rtype</tt><tt class="py-op">=</tt><tt class="py-string">'united'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_get_atom_radius-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_get_atom_radius-expanded"><a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">    <tt class="py-docstring">"""Translate an atom object to an atomic radius defined in MSMS (PRIVATE).</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line"><tt class="py-docstring">    Uses information from the parent residue and the atom object to define</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line"><tt class="py-docstring">    the atom type.</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line"><tt class="py-docstring">    Returns the radius (float) according to the selected type:</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line"><tt class="py-docstring">     - explicit (reads hydrogens)</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line"><tt class="py-docstring">     - united (default)</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">rtype</tt> <tt class="py-op">==</tt> <tt class="py-string">'explicit'</tt><tt class="py-op">:</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line">        <tt class="py-name">typekey</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">rtype</tt> <tt class="py-op">==</tt> <tt class="py-string">'united'</tt><tt class="py-op">:</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line">        <tt class="py-name">typekey</tt> <tt class="py-op">=</tt> <tt class="py-number">2</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Radius type (%r) not understood. "</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">                         <tt class="py-string">"Must be 'explicit' or 'united'"</tt> <tt class="py-op">%</tt> <tt class="py-name">rtype</tt><tt class="py-op">)</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line"> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line">    <tt class="py-name">resname</tt> <tt class="py-op">=</tt> <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt class="py-name">parent</tt><tt class="py-op">.</tt><tt class="py-name">resname</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line">    <tt class="py-name">het_atm</tt> <tt class="py-op">=</tt> <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt class="py-name">parent</tt><tt class="py-op">.</tt><tt id="link-19" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Variable Bio.PDB.Entity.Entity.id=Bio.PDB.Entity.Entity-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-19', 'id', 'link-19');">id</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line"> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line">    <tt class="py-name">at_name</tt> <tt class="py-op">=</tt> <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-20" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-20', 'name', 'link-20');">name</a></tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line">    <tt class="py-name">at_elem</tt> <tt class="py-op">=</tt> <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-21" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.element=Bio.KEGG.KGML.KGML_pathway.Component-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Entry.element=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Graphics.element=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.element=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.element=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Relation.element=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#element,Variable Bio.Phylo._cdao_owl.element=Bio.Phylo._cdao_owl-module.html#element"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-21', 'element', 'link-21');">element</a></tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line"> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line">    <tt class="py-comment"># Hydrogens</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">at_elem</tt> <tt class="py-op">==</tt> <tt class="py-string">'H'</tt> <tt class="py-keyword">or</tt> <tt class="py-name">at_elem</tt> <tt class="py-op">==</tt> <tt class="py-string">'D'</tt><tt class="py-op">:</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-22" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-22', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">15</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line">    <tt class="py-comment"># HETATMs</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">het_atm</tt> <tt class="py-op">==</tt> <tt class="py-string">'W'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_elem</tt> <tt class="py-op">==</tt> <tt class="py-string">'O'</tt><tt class="py-op">:</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-23" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-23', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">het_atm</tt> <tt class="py-op">!=</tt> <tt class="py-string">' '</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_elem</tt> <tt class="py-op">==</tt> <tt class="py-string">'CA'</tt><tt class="py-op">:</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-24" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-24', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">18</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">het_atm</tt> <tt class="py-op">!=</tt> <tt class="py-string">' '</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_elem</tt> <tt class="py-op">==</tt> <tt class="py-string">'CD'</tt><tt class="py-op">:</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-25" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-25', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">22</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'ACE'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'CA'</tt><tt class="py-op">:</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-26" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-26', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">    <tt class="py-comment"># Main chain atoms</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'N'</tt><tt class="py-op">:</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-27" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-27', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'CA'</tt><tt class="py-op">:</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-28" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-28', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'C'</tt><tt class="py-op">:</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-29" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-29', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">10</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'O'</tt><tt class="py-op">:</tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-30" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-30', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'P'</tt><tt class="py-op">:</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-31" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-31', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">13</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">    <tt class="py-comment"># CB atoms</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'CB'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'ALA'</tt><tt class="py-op">:</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-32" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-32', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'CB'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'ILE'</tt><tt class="py-op">,</tt> <tt class="py-string">'THR'</tt><tt class="py-op">,</tt> <tt class="py-string">'VAL'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-33" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-33', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'CB'</tt><tt class="py-op">:</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-34" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-34', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">8</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line">    <tt class="py-comment"># CG atoms</tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'CG'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'ASN'</tt><tt class="py-op">,</tt> <tt class="py-string">'ASP'</tt><tt class="py-op">,</tt> <tt class="py-string">'ASX'</tt><tt class="py-op">,</tt> <tt class="py-string">'HIS'</tt><tt class="py-op">,</tt> <tt class="py-string">'HIP'</tt><tt class="py-op">,</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">                                         <tt class="py-string">'HIE'</tt><tt class="py-op">,</tt> <tt class="py-string">'HID'</tt><tt class="py-op">,</tt> <tt class="py-string">'HISN'</tt><tt class="py-op">,</tt> <tt class="py-string">'HISL'</tt><tt class="py-op">,</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">                                         <tt class="py-string">'LEU'</tt><tt class="py-op">,</tt> <tt class="py-string">'PHE'</tt><tt class="py-op">,</tt> <tt class="py-string">'TRP'</tt><tt class="py-op">,</tt> <tt class="py-string">'TYR'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-35" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-35', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">10</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'CG'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'LEU'</tt><tt class="py-op">:</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-36" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-36', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'CG'</tt><tt class="py-op">:</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-37" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-37', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">8</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">    <tt class="py-comment"># General amino acids in alphabetical order</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'GLN'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_elem</tt> <tt class="py-op">==</tt> <tt class="py-string">'O'</tt><tt class="py-op">:</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-38" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-38', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'ACE'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'CH3'</tt><tt class="py-op">:</tt> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-39" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-39', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'ARG'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'CD'</tt><tt class="py-op">:</tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-40" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-40', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">8</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'ARG'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'NE'</tt><tt class="py-op">,</tt> <tt class="py-string">'RE'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-41" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-41', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'ARG'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'CZ'</tt><tt class="py-op">:</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-42" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-42', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">10</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'ARG'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-43" class="py-name" targets="Method Bio.Seq.Seq.startswith()=Bio.Seq.Seq-class.html#startswith"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-43', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">'NH'</tt><tt class="py-op">,</tt> <tt class="py-string">'RH'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-44" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-44', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">5</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'ASN'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'OD1'</tt><tt class="py-op">:</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-45" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-45', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'ASN'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'ND2'</tt><tt class="py-op">:</tt> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-46" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-46', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">5</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'ASN'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-47" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-47', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'AD'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-48" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-48', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'ASP'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-49" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-49', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">'OD'</tt><tt class="py-op">,</tt> <tt class="py-string">'ED'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-50" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-50', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'ASX'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-51" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-51', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'OD1'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-52" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-52', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'ASX'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'ND2'</tt><tt class="py-op">:</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-53" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-53', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'ASX'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-54" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-54', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">'OD'</tt><tt class="py-op">,</tt> <tt class="py-string">'AD'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-55" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-55', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'CYX'</tt><tt class="py-op">,</tt> <tt class="py-string">'CYM'</tt><tt class="py-op">}</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'SG'</tt><tt class="py-op">:</tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-56" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-56', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">13</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'MET'</tt><tt class="py-op">}</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-57" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-57', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'LP'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-58" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-58', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">13</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'CUH'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'SG'</tt><tt class="py-op">:</tt> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-59" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-59', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">12</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'GLU'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-60" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-60', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">'OE'</tt><tt class="py-op">,</tt> <tt class="py-string">'EE'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-61" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-61', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'GLU'</tt><tt class="py-op">,</tt> <tt class="py-string">'GLN'</tt><tt class="py-op">,</tt> <tt class="py-string">'GLX'</tt><tt class="py-op">}</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'CD'</tt><tt class="py-op">:</tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-62" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-62', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">10</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'GLN'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'OE1'</tt><tt class="py-op">:</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-63" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-63', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'GLN'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'NE2'</tt><tt class="py-op">:</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-64" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-64', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">5</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'GLN'</tt><tt class="py-op">,</tt> <tt class="py-string">'GLX'</tt><tt class="py-op">}</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-65" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-65', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'AE'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-66" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-66', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line">    <tt class="py-comment"># Histdines and friends</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line">    <tt class="py-comment"># There are 4 kinds of HIS rings: HIS (no protons), HID (proton on Delta),</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line">    <tt class="py-comment">#   HIE (proton on epsilon), and HIP (protons on both)</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">    <tt class="py-comment"># Protonated nitrogens are numbered 4, else 14</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line">    <tt class="py-comment"># HIS is treated here as the same as HIE</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line">    <tt class="py-comment">#</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">    <tt class="py-comment"># HISL is a deprotonated HIS (the L means liganded)</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'HIS'</tt><tt class="py-op">,</tt> <tt class="py-string">'HID'</tt><tt class="py-op">,</tt> <tt class="py-string">'HIE'</tt><tt class="py-op">,</tt> <tt class="py-string">'HIP'</tt><tt class="py-op">,</tt> <tt class="py-string">'HISL'</tt><tt class="py-op">}</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'CE1'</tt><tt class="py-op">,</tt> <tt class="py-string">'CD2'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt>  <tt class="py-comment"># noqa: E501</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-67" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-67', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">11</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'HIS'</tt><tt class="py-op">,</tt> <tt class="py-string">'HID'</tt><tt class="py-op">,</tt> <tt class="py-string">'HIE'</tt><tt class="py-op">,</tt> <tt class="py-string">'HISL'</tt><tt class="py-op">}</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'ND1'</tt><tt class="py-op">:</tt> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-68" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-68', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">14</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'HID'</tt><tt class="py-op">,</tt> <tt class="py-string">'HIP'</tt><tt class="py-op">}</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'ND1'</tt><tt class="py-op">,</tt> <tt class="py-string">'RD1'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-69" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-69', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'HIS'</tt><tt class="py-op">,</tt> <tt class="py-string">'HIE'</tt><tt class="py-op">,</tt> <tt class="py-string">'HIP'</tt><tt class="py-op">}</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'NE2'</tt><tt class="py-op">,</tt> <tt class="py-string">'RE2'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-70" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-70', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'HID'</tt><tt class="py-op">,</tt> <tt class="py-string">'HISL'</tt><tt class="py-op">}</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'NE2'</tt><tt class="py-op">,</tt> <tt class="py-string">'RE2'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-71" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-71', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">14</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'HIS'</tt><tt class="py-op">,</tt> <tt class="py-string">'HID'</tt><tt class="py-op">,</tt> <tt class="py-string">'HIP'</tt><tt class="py-op">,</tt> <tt class="py-string">'HISL'</tt><tt class="py-op">}</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-72" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-72', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">'AD'</tt><tt class="py-op">,</tt> <tt class="py-string">'AE'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># noqa: E501</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-73" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-73', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line">    <tt class="py-comment"># More amino acids</tt> </tt>
<a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'ILE'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'CG1'</tt><tt class="py-op">:</tt> </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-74" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-74', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">8</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'ILE'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'CG2'</tt><tt class="py-op">:</tt> </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-75" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-75', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'ILE'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'CD'</tt><tt class="py-op">,</tt> <tt class="py-string">'CD1'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-76" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-76', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'LEU'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-77" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-77', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'CD'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-78" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-78', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'LYS'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'CG'</tt><tt class="py-op">,</tt> <tt class="py-string">'CD'</tt><tt class="py-op">,</tt> <tt class="py-string">'CE'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-79" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-79', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">8</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'LYS'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'NZ'</tt><tt class="py-op">,</tt> <tt class="py-string">'KZ'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-80" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-80', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'MET'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'SD'</tt><tt class="py-op">:</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-81" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-81', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">13</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'MET'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'CE'</tt><tt class="py-op">:</tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-82" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-82', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'PHE'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-83" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-83', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">'CD'</tt><tt class="py-op">,</tt> <tt class="py-string">'CE'</tt><tt class="py-op">,</tt> <tt class="py-string">'CZ'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-84" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-84', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">11</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'PRO'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'CG'</tt><tt class="py-op">,</tt> <tt class="py-string">'CD'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-85" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-85', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">8</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'CSO'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'SE'</tt><tt class="py-op">,</tt> <tt class="py-string">'SEG'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-86" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-86', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'CSO'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-87" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-87', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'OD'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-88" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-88', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'SER'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'OG'</tt><tt class="py-op">:</tt> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-89" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-89', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'THR'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'OG1'</tt><tt class="py-op">:</tt> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-90" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-90', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'THR'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'CG2'</tt><tt class="py-op">:</tt> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-91" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-91', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'TRP'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'CD1'</tt><tt class="py-op">:</tt> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-92" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-92', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">11</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'TRP'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'CD2'</tt><tt class="py-op">,</tt> <tt class="py-string">'CE2'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-93" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-93', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">10</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'TRP'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'NE1'</tt><tt class="py-op">:</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-94" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-94', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'TRP'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'CE3'</tt><tt class="py-op">,</tt> <tt class="py-string">'CZ2'</tt><tt class="py-op">,</tt> <tt class="py-string">'CZ3'</tt><tt class="py-op">,</tt> <tt class="py-string">'CH2'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-95" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-95', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">11</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'TYR'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'CD1'</tt><tt class="py-op">,</tt> <tt class="py-string">'CD2'</tt><tt class="py-op">,</tt> <tt class="py-string">'CE1'</tt><tt class="py-op">,</tt> <tt class="py-string">'CE2'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-96" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-96', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">11</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'TYR'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'CZ'</tt><tt class="py-op">:</tt> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-97" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-97', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">10</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'TYR'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'OH'</tt><tt class="py-op">:</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-98" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-98', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'VAL'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'CG1'</tt><tt class="py-op">,</tt> <tt class="py-string">'CG2'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-99" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-99', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'CD'</tt><tt class="py-op">,</tt> <tt class="py-string">'CD'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-100" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-100', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">8</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line">    <tt class="py-comment"># Co-factors, and other weirdos</tt> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'FS3'</tt><tt class="py-op">,</tt> <tt class="py-string">'FS4'</tt><tt class="py-op">}</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-101" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-101', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'FE'</tt><tt class="py-op">)</tt> \ </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line">            <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-102" class="py-name" targets="Method Bio.Seq.Seq.endswith()=Bio.Seq.Seq-class.html#endswith"><a title="Bio.Seq.Seq.endswith" class="py-name" href="#" onclick="return doclink('link-102', 'endswith', 'link-102');">endswith</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">'1'</tt><tt class="py-op">,</tt> <tt class="py-string">'2'</tt><tt class="py-op">,</tt> <tt class="py-string">'3'</tt><tt class="py-op">,</tt> <tt class="py-string">'4'</tt><tt class="py-op">,</tt> <tt class="py-string">'5'</tt><tt class="py-op">,</tt> <tt class="py-string">'6'</tt><tt class="py-op">,</tt> <tt class="py-string">'7'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-103" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-103', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">21</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'FS3'</tt><tt class="py-op">,</tt> <tt class="py-string">'FS4'</tt><tt class="py-op">}</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-104" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-104', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'S'</tt><tt class="py-op">)</tt> \ </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line">            <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-105" class="py-name"><a title="Bio.Seq.Seq.endswith" class="py-name" href="#" onclick="return doclink('link-105', 'endswith', 'link-102');">endswith</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">'1'</tt><tt class="py-op">,</tt> <tt class="py-string">'2'</tt><tt class="py-op">,</tt> <tt class="py-string">'3'</tt><tt class="py-op">,</tt> <tt class="py-string">'4'</tt><tt class="py-op">,</tt> <tt class="py-string">'5'</tt><tt class="py-op">,</tt> <tt class="py-string">'6'</tt><tt class="py-op">,</tt> <tt class="py-string">'7'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-106" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-106', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">13</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'FS3'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'OXO'</tt><tt class="py-op">:</tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-107" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-107', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'FEO'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'FE1'</tt><tt class="py-op">,</tt> <tt class="py-string">'FE2'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-108" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-108', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">21</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'HEM'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'O1'</tt><tt class="py-op">,</tt> <tt class="py-string">'O2'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-109" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-109', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'HEM'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'FE'</tt><tt class="py-op">:</tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-110" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-110', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">21</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'HEM'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'CHA'</tt><tt class="py-op">,</tt> <tt class="py-string">'CHB'</tt><tt class="py-op">,</tt> <tt class="py-string">'CHC'</tt><tt class="py-op">,</tt> <tt class="py-string">'CHD'</tt><tt class="py-op">,</tt> </tt>
<a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line">                                          <tt class="py-string">'CAB'</tt><tt class="py-op">,</tt> <tt class="py-string">'CAC'</tt><tt class="py-op">,</tt> <tt class="py-string">'CBB'</tt><tt class="py-op">,</tt> <tt class="py-string">'CBC'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-111" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-111', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">11</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'HEM'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'NA'</tt><tt class="py-op">,</tt> <tt class="py-string">'NB'</tt><tt class="py-op">,</tt> <tt class="py-string">'NC'</tt><tt class="py-op">,</tt> <tt class="py-string">'ND'</tt><tt class="py-op">,</tt> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line">                                          <tt class="py-string">'N A'</tt><tt class="py-op">,</tt> <tt class="py-string">'N B'</tt><tt class="py-op">,</tt> <tt class="py-string">'N C'</tt><tt class="py-op">,</tt> <tt class="py-string">'N D'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-112" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-112', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">14</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'HEM'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'C1A'</tt><tt class="py-op">,</tt> <tt class="py-string">'C1B'</tt><tt class="py-op">,</tt> <tt class="py-string">'C1C'</tt><tt class="py-op">,</tt> <tt class="py-string">'C1D'</tt><tt class="py-op">,</tt> </tt>
<a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line">                                          <tt class="py-string">'C2A'</tt><tt class="py-op">,</tt> <tt class="py-string">'C2B'</tt><tt class="py-op">,</tt> <tt class="py-string">'C2C'</tt><tt class="py-op">,</tt> <tt class="py-string">'C2D'</tt><tt class="py-op">,</tt> </tt>
<a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line">                                          <tt class="py-string">'C3A'</tt><tt class="py-op">,</tt> <tt class="py-string">'C3B'</tt><tt class="py-op">,</tt> <tt class="py-string">'C3C'</tt><tt class="py-op">,</tt> <tt class="py-string">'C3D'</tt><tt class="py-op">,</tt> </tt>
<a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">                                          <tt class="py-string">'C4A'</tt><tt class="py-op">,</tt> <tt class="py-string">'C4B'</tt><tt class="py-op">,</tt> <tt class="py-string">'C4C'</tt><tt class="py-op">,</tt> <tt class="py-string">'C4D'</tt><tt class="py-op">,</tt> </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line">                                          <tt class="py-string">'CGA'</tt><tt class="py-op">,</tt> <tt class="py-string">'CGD'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-113" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-113', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">10</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'HEM'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'CMA'</tt><tt class="py-op">,</tt> <tt class="py-string">'CMB'</tt><tt class="py-op">,</tt> <tt class="py-string">'CMC'</tt><tt class="py-op">,</tt> <tt class="py-string">'CMD'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-114" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-114', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'HEM'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'OH2'</tt><tt class="py-op">:</tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-115" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-115', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'AZI'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'N1'</tt><tt class="py-op">,</tt> <tt class="py-string">'N2'</tt><tt class="py-op">,</tt> <tt class="py-string">'N3'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-116" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-116', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">14</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'MPD'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'C1'</tt><tt class="py-op">,</tt> <tt class="py-string">'C5'</tt><tt class="py-op">,</tt> <tt class="py-string">'C6'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-117" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-117', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'MPD'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'C2'</tt><tt class="py-op">:</tt> </tt>
<a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-118" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-118', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">10</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'MPD'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'C3'</tt><tt class="py-op">:</tt> </tt>
<a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-119" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-119', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">8</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'MPD'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'C4'</tt><tt class="py-op">:</tt> </tt>
<a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-120" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-120', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'MPD'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'O7'</tt><tt class="py-op">,</tt> <tt class="py-string">'O8'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-121" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-121', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'SO4'</tt><tt class="py-op">,</tt> <tt class="py-string">'SUL'</tt><tt class="py-op">}</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'S'</tt><tt class="py-op">:</tt> </tt>
<a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-122" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-122', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">13</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'SO4'</tt><tt class="py-op">,</tt> <tt class="py-string">'SUL'</tt><tt class="py-op">,</tt> <tt class="py-string">'PO4'</tt><tt class="py-op">,</tt> <tt class="py-string">'PHO'</tt><tt class="py-op">}</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'O1'</tt><tt class="py-op">,</tt> <tt class="py-string">'O2'</tt><tt class="py-op">,</tt> <tt class="py-string">'O3'</tt><tt class="py-op">,</tt> <tt class="py-string">'O4'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt>  <tt class="py-comment"># noqa: E501</tt> </tt>
<a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-123" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-123', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'PC '</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'O1'</tt><tt class="py-op">,</tt> <tt class="py-string">'O2'</tt><tt class="py-op">,</tt> <tt class="py-string">'O3'</tt><tt class="py-op">,</tt> <tt class="py-string">'O4'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-124" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-124', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'PC '</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'P1'</tt><tt class="py-op">:</tt> </tt>
<a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-125" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-125', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">13</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'PC '</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'C1'</tt><tt class="py-op">,</tt> <tt class="py-string">'C2'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-126" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-126', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">8</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L347"></a><tt class="py-lineno">347</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'PC '</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'C3'</tt><tt class="py-op">,</tt> <tt class="py-string">'C4'</tt><tt class="py-op">,</tt> <tt class="py-string">'C5'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-127" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-127', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'PC '</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'N1'</tt><tt class="py-op">:</tt> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-128" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-128', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">14</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'BIG'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'BAL'</tt><tt class="py-op">:</tt> </tt>
<a name="L352"></a><tt class="py-lineno">352</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-129" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-129', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">17</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'POI'</tt><tt class="py-op">,</tt> <tt class="py-string">'DOT'</tt><tt class="py-op">}</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'POI'</tt><tt class="py-op">,</tt> <tt class="py-string">'DOT'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-130" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-130', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">23</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'FMN'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'N1'</tt><tt class="py-op">,</tt> <tt class="py-string">'N5'</tt><tt class="py-op">,</tt> <tt class="py-string">'N10'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-131" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-131', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'FMN'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'C2'</tt><tt class="py-op">,</tt> <tt class="py-string">'C4'</tt><tt class="py-op">,</tt> <tt class="py-string">'C7'</tt><tt class="py-op">,</tt> <tt class="py-string">'C8'</tt><tt class="py-op">,</tt> <tt class="py-string">'C10'</tt><tt class="py-op">,</tt> </tt>
<a name="L358"></a><tt class="py-lineno">358</tt>  <tt class="py-line">                                          <tt class="py-string">'C4A'</tt><tt class="py-op">,</tt> <tt class="py-string">'C5A'</tt><tt class="py-op">,</tt> <tt class="py-string">'C9A'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-132" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-132', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">10</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'FMN'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'O2'</tt><tt class="py-op">,</tt> <tt class="py-string">'O4'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-133" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-133', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'FMN'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'N3'</tt><tt class="py-op">:</tt> </tt>
<a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-134" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-134', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">14</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'FMN'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'C6'</tt><tt class="py-op">,</tt> <tt class="py-string">'C9'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-135" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-135', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">11</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'FMN'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'C7M'</tt><tt class="py-op">,</tt> <tt class="py-string">'C8M'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-136" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-136', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L368"></a><tt class="py-lineno">368</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'FMN'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-137" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-137', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">'C1'</tt><tt class="py-op">,</tt> <tt class="py-string">'C2'</tt><tt class="py-op">,</tt> <tt class="py-string">'C3'</tt><tt class="py-op">,</tt> <tt class="py-string">'C4'</tt><tt class="py-op">,</tt> <tt class="py-string">'C5'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># noqa: E501</tt> </tt>
<a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-138" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-138', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">8</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'FMN'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-139" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-139', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">'O2'</tt><tt class="py-op">,</tt> <tt class="py-string">'O3'</tt><tt class="py-op">,</tt> <tt class="py-string">'O4'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-140" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-140', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'FMN'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-141" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-141', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'O5'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-142" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-142', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'FMN'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'OP1'</tt><tt class="py-op">,</tt> <tt class="py-string">'OP2'</tt><tt class="py-op">,</tt> <tt class="py-string">'OP3'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-143" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-143', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'ALK'</tt><tt class="py-op">,</tt> <tt class="py-string">'MYR'</tt><tt class="py-op">}</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'OT1'</tt><tt class="py-op">:</tt> </tt>
<a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-144" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-144', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L378"></a><tt class="py-lineno">378</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'ALK'</tt><tt class="py-op">,</tt> <tt class="py-string">'MYR'</tt><tt class="py-op">}</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'C01'</tt><tt class="py-op">:</tt> </tt>
<a name="L379"></a><tt class="py-lineno">379</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-145" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-145', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">10</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L380"></a><tt class="py-lineno">380</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'ALK'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'C16'</tt><tt class="py-op">:</tt> </tt>
<a name="L381"></a><tt class="py-lineno">381</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-146" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-146', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L382"></a><tt class="py-lineno">382</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-op">==</tt> <tt class="py-string">'MYR'</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'C14'</tt><tt class="py-op">:</tt> </tt>
<a name="L383"></a><tt class="py-lineno">383</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-147" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-147', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L384"></a><tt class="py-lineno">384</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'ALK'</tt><tt class="py-op">,</tt> <tt class="py-string">'MYR'</tt><tt class="py-op">}</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-148" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-148', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'C'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L385"></a><tt class="py-lineno">385</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-149" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-149', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">8</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L386"></a><tt class="py-lineno">386</tt>  <tt class="py-line">    <tt class="py-comment"># Metals</tt> </tt>
<a name="L387"></a><tt class="py-lineno">387</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_elem</tt> <tt class="py-op">==</tt> <tt class="py-string">'CU'</tt><tt class="py-op">:</tt> </tt>
<a name="L388"></a><tt class="py-lineno">388</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-150" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-150', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">20</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L389"></a><tt class="py-lineno">389</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_elem</tt> <tt class="py-op">==</tt> <tt class="py-string">'ZN'</tt><tt class="py-op">:</tt> </tt>
<a name="L390"></a><tt class="py-lineno">390</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-151" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-151', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">19</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L391"></a><tt class="py-lineno">391</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_elem</tt> <tt class="py-op">==</tt> <tt class="py-string">'MN'</tt><tt class="py-op">:</tt> </tt>
<a name="L392"></a><tt class="py-lineno">392</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-152" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-152', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">27</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L393"></a><tt class="py-lineno">393</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_elem</tt> <tt class="py-op">==</tt> <tt class="py-string">'FE'</tt><tt class="py-op">:</tt> </tt>
<a name="L394"></a><tt class="py-lineno">394</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-153" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-153', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">25</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L395"></a><tt class="py-lineno">395</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_elem</tt> <tt class="py-op">==</tt> <tt class="py-string">'MG'</tt><tt class="py-op">:</tt> </tt>
<a name="L396"></a><tt class="py-lineno">396</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-154" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-154', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">26</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L397"></a><tt class="py-lineno">397</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_elem</tt> <tt class="py-op">==</tt> <tt class="py-string">'CO'</tt><tt class="py-op">:</tt> </tt>
<a name="L398"></a><tt class="py-lineno">398</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-155" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-155', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">28</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L399"></a><tt class="py-lineno">399</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_elem</tt> <tt class="py-op">==</tt> <tt class="py-string">'SE'</tt><tt class="py-op">:</tt> </tt>
<a name="L400"></a><tt class="py-lineno">400</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-156" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-156', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">29</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L401"></a><tt class="py-lineno">401</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_elem</tt> <tt class="py-op">==</tt> <tt class="py-string">'YB'</tt><tt class="py-op">:</tt> </tt>
<a name="L402"></a><tt class="py-lineno">402</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-157" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-157', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">31</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L403"></a><tt class="py-lineno">403</tt>  <tt class="py-line">    <tt class="py-comment"># Others</tt> </tt>
<a name="L404"></a><tt class="py-lineno">404</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'SEG'</tt><tt class="py-op">:</tt> </tt>
<a name="L405"></a><tt class="py-lineno">405</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-158" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-158', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L406"></a><tt class="py-lineno">406</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_name</tt> <tt class="py-op">==</tt> <tt class="py-string">'OXT'</tt><tt class="py-op">:</tt> </tt>
<a name="L407"></a><tt class="py-lineno">407</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-159" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-159', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L408"></a><tt class="py-lineno">408</tt>  <tt class="py-line">    <tt class="py-comment"># Catch-alls</tt> </tt>
<a name="L409"></a><tt class="py-lineno">409</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-160" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-160', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">'OT'</tt><tt class="py-op">,</tt> <tt class="py-string">'E'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L410"></a><tt class="py-lineno">410</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-161" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-161', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L411"></a><tt class="py-lineno">411</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-162" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-162', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'S'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L412"></a><tt class="py-lineno">412</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-163" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-163', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">13</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L413"></a><tt class="py-lineno">413</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-164" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-164', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'C'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L414"></a><tt class="py-lineno">414</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-165" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-165', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L415"></a><tt class="py-lineno">415</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-166" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-166', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'A'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L416"></a><tt class="py-lineno">416</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-167" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-167', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">11</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L417"></a><tt class="py-lineno">417</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-168" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-168', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'O'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L418"></a><tt class="py-lineno">418</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-169" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-169', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L419"></a><tt class="py-lineno">419</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-170" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-170', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">'N'</tt><tt class="py-op">,</tt> <tt class="py-string">'R'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L420"></a><tt class="py-lineno">420</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-171" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-171', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L421"></a><tt class="py-lineno">421</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-172" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-172', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'K'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L422"></a><tt class="py-lineno">422</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-173" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-173', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L423"></a><tt class="py-lineno">423</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'PA'</tt><tt class="py-op">,</tt> <tt class="py-string">'PB'</tt><tt class="py-op">,</tt> <tt class="py-string">'PC'</tt><tt class="py-op">,</tt> <tt class="py-string">'PD'</tt><tt class="py-op">}</tt><tt class="py-op">:</tt> </tt>
<a name="L424"></a><tt class="py-lineno">424</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-174" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-174', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">13</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L425"></a><tt class="py-lineno">425</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-175" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-175', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'P'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L426"></a><tt class="py-lineno">426</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-176" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-176', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">13</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L427"></a><tt class="py-lineno">427</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'FAD'</tt><tt class="py-op">,</tt> <tt class="py-string">'NAD'</tt><tt class="py-op">,</tt> <tt class="py-string">'AMX'</tt><tt class="py-op">,</tt> <tt class="py-string">'APU'</tt><tt class="py-op">}</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-177" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-177', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'O'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># noqa: E501</tt> </tt>
<a name="L428"></a><tt class="py-lineno">428</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-178" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-178', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L429"></a><tt class="py-lineno">429</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'FAD'</tt><tt class="py-op">,</tt> <tt class="py-string">'NAD'</tt><tt class="py-op">,</tt> <tt class="py-string">'AMX'</tt><tt class="py-op">,</tt> <tt class="py-string">'APU'</tt><tt class="py-op">}</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-179" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-179', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'N'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># noqa: E501</tt> </tt>
<a name="L430"></a><tt class="py-lineno">430</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-180" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-180', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L431"></a><tt class="py-lineno">431</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'FAD'</tt><tt class="py-op">,</tt> <tt class="py-string">'NAD'</tt><tt class="py-op">,</tt> <tt class="py-string">'AMX'</tt><tt class="py-op">,</tt> <tt class="py-string">'APU'</tt><tt class="py-op">}</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-181" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-181', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'C'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># noqa: E501</tt> </tt>
<a name="L432"></a><tt class="py-lineno">432</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-182" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-182', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L433"></a><tt class="py-lineno">433</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'FAD'</tt><tt class="py-op">,</tt> <tt class="py-string">'NAD'</tt><tt class="py-op">,</tt> <tt class="py-string">'AMX'</tt><tt class="py-op">,</tt> <tt class="py-string">'APU'</tt><tt class="py-op">}</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-183" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-183', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'P'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># noqa: E501</tt> </tt>
<a name="L434"></a><tt class="py-lineno">434</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-184" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-184', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">13</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L435"></a><tt class="py-lineno">435</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">resname</tt> <tt class="py-keyword">in</tt> <tt class="py-op">{</tt><tt class="py-string">'FAD'</tt><tt class="py-op">,</tt> <tt class="py-string">'NAD'</tt><tt class="py-op">,</tt> <tt class="py-string">'AMX'</tt><tt class="py-op">,</tt> <tt class="py-string">'APU'</tt><tt class="py-op">}</tt> <tt class="py-keyword">and</tt> <tt class="py-name">at_name</tt><tt class="py-op">.</tt><tt id="link-185" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-185', 'startswith', 'link-43');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'H'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># noqa: E501</tt> </tt>
<a name="L436"></a><tt class="py-lineno">436</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-186" class="py-name"><a title="Bio.PDB.ResidueDepth'._atomic_radii" class="py-name" href="#" onclick="return doclink('link-186', '_atomic_radii', 'link-18');">_atomic_radii</a></tt><tt class="py-op">[</tt><tt class="py-number">15</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">typekey</tt><tt class="py-op">]</tt> </tt>
<a name="L437"></a><tt class="py-lineno">437</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L438"></a><tt class="py-lineno">438</tt>  <tt class="py-line">        <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">'{}:{} not in radii library.'</tt><tt class="py-op">.</tt><tt id="link-187" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.format()=Bio.Align.MultipleSeqAlignment-class.html#format,Method Bio.Phylo.BaseTree.Tree.format()=Bio.Phylo.BaseTree.Tree-class.html#format,Method Bio.Restriction.Restriction.RestrictionBatch.format()=Bio.Restriction.Restriction.RestrictionBatch-class.html#format,Method Bio.SeqRecord.SeqRecord.format()=Bio.SeqRecord.SeqRecord-class.html#format,Method Bio.motifs.Motif.format()=Bio.motifs.Motif-class.html#format"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-187', 'format', 'link-187');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">at_name</tt><tt class="py-op">,</tt> <tt class="py-name">resname</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L439"></a><tt class="py-lineno">439</tt>  <tt class="py-line">                      <tt id="link-188" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-188', 'BiopythonWarning', 'link-15');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L440"></a><tt class="py-lineno">440</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-number">0.01</tt> </tt>
</div><a name="L441"></a><tt class="py-lineno">441</tt>  <tt class="py-line"> </tt>
<a name="L442"></a><tt class="py-lineno">442</tt>  <tt class="py-line"> </tt>
<a name="_read_vertex_array"></a><div id="_read_vertex_array-def"><a name="L443"></a><tt class="py-lineno">443</tt> <a class="py-toggle" href="#" id="_read_vertex_array-toggle" onclick="return toggle('_read_vertex_array');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.PDB.ResidueDepth'-module.html#_read_vertex_array">_read_vertex_array</a><tt class="py-op">(</tt><tt class="py-param">filename</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_read_vertex_array-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_read_vertex_array-expanded"><a name="L444"></a><tt class="py-lineno">444</tt>  <tt class="py-line">    <tt class="py-docstring">"""Read the vertex list into a Numeric array (PRIVATE)."""</tt> </tt>
<a name="L445"></a><tt class="py-lineno">445</tt>  <tt class="py-line">    <tt class="py-keyword">with</tt> <tt id="link-189" class="py-name" targets="Function Bio.bgzf.open()=Bio.bgzf-module.html#open"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-189', 'open', 'link-189');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-string">"r"</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">fp</tt><tt class="py-op">:</tt> </tt>
<a name="L446"></a><tt class="py-lineno">446</tt>  <tt class="py-line">        <tt class="py-name">vertex_list</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L447"></a><tt class="py-lineno">447</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">l</tt> <tt class="py-keyword">in</tt> <tt class="py-name">fp</tt><tt class="py-op">:</tt> </tt>
<a name="L448"></a><tt class="py-lineno">448</tt>  <tt class="py-line">            <tt class="py-name">sl</tt> <tt class="py-op">=</tt> <tt class="py-name">l</tt><tt class="py-op">.</tt><tt id="link-190" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-190', 'split', 'link-190');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L449"></a><tt class="py-lineno">449</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">sl</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-number">9</tt><tt class="py-op">:</tt> </tt>
<a name="L450"></a><tt class="py-lineno">450</tt>  <tt class="py-line">                <tt class="py-comment"># skip header</tt> </tt>
<a name="L451"></a><tt class="py-lineno">451</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L452"></a><tt class="py-lineno">452</tt>  <tt class="py-line">            <tt class="py-name">vl</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">float</tt><tt class="py-op">(</tt><tt id="link-191" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.x=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#x,Variable Bio.Statistics.lowess.x=Bio.Statistics.lowess-module.html#x"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-191', 'x', 'link-191');">x</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-192" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-192', 'x', 'link-191');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">sl</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">:</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L453"></a><tt class="py-lineno">453</tt>  <tt class="py-line">            <tt class="py-name">vertex_list</tt><tt class="py-op">.</tt><tt id="link-193" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-193', 'append', 'link-193');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">vl</tt><tt class="py-op">)</tt> </tt>
<a name="L454"></a><tt class="py-lineno">454</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">numpy</tt><tt class="py-op">.</tt><tt class="py-name">array</tt><tt class="py-op">(</tt><tt class="py-name">vertex_list</tt><tt class="py-op">)</tt> </tt>
</div><a name="L455"></a><tt class="py-lineno">455</tt>  <tt class="py-line"> </tt>
<a name="L456"></a><tt class="py-lineno">456</tt>  <tt class="py-line"> </tt>
<a name="get_surface"></a><div id="get_surface-def"><a name="L457"></a><tt class="py-lineno">457</tt> <a class="py-toggle" href="#" id="get_surface-toggle" onclick="return toggle('get_surface');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.PDB.ResidueDepth'-module.html#get_surface">get_surface</a><tt class="py-op">(</tt><tt class="py-param">model</tt><tt class="py-op">,</tt> <tt class="py-param">PDB_TO_XYZR</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">MSMS</tt><tt class="py-op">=</tt><tt class="py-string">"msms"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="get_surface-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="get_surface-expanded"><a name="L458"></a><tt class="py-lineno">458</tt>  <tt class="py-line">    <tt class="py-docstring">"""Represent molecular surface as a vertex list array.</tt> </tt>
<a name="L459"></a><tt class="py-lineno">459</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L460"></a><tt class="py-lineno">460</tt>  <tt class="py-line"><tt class="py-docstring">    Return a Numpy array that represents the vertex list of the</tt> </tt>
<a name="L461"></a><tt class="py-lineno">461</tt>  <tt class="py-line"><tt class="py-docstring">    molecular surface.</tt> </tt>
<a name="L462"></a><tt class="py-lineno">462</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L463"></a><tt class="py-lineno">463</tt>  <tt class="py-line"><tt class="py-docstring">    Arguments:</tt> </tt>
<a name="L464"></a><tt class="py-lineno">464</tt>  <tt class="py-line"><tt class="py-docstring">     - PDB_TO_XYZR - deprecated, ignore this.</tt> </tt>
<a name="L465"></a><tt class="py-lineno">465</tt>  <tt class="py-line"><tt class="py-docstring">     - MSMS - msms executable (used as argument to os.system)</tt> </tt>
<a name="L466"></a><tt class="py-lineno">466</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L467"></a><tt class="py-lineno">467</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L468"></a><tt class="py-lineno">468</tt>  <tt class="py-line">    <tt class="py-comment"># Issue warning if PDB_TO_XYZR is given</tt> </tt>
<a name="L469"></a><tt class="py-lineno">469</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">PDB_TO_XYZR</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L470"></a><tt class="py-lineno">470</tt>  <tt class="py-line">        <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">"PDB_TO_XYZR argument will be deprecated soon"</tt> </tt>
<a name="L471"></a><tt class="py-lineno">471</tt>  <tt class="py-line">                       <tt class="py-string">" in favor of an internal mapping algorithm."</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L472"></a><tt class="py-lineno">472</tt>  <tt class="py-line">                      <tt id="link-194" class="py-name"><a title="Bio.BiopythonDeprecationWarning" class="py-name" href="#" onclick="return doclink('link-194', 'BiopythonDeprecationWarning', 'link-17');">BiopythonDeprecationWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L473"></a><tt class="py-lineno">473</tt>  <tt class="py-line"> </tt>
<a name="L474"></a><tt class="py-lineno">474</tt>  <tt class="py-line">    <tt class="py-comment"># Replace pdb_to_xyzr</tt> </tt>
<a name="L475"></a><tt class="py-lineno">475</tt>  <tt class="py-line">    <tt class="py-comment"># Make x,y,z,radius file</tt> </tt>
<a name="L476"></a><tt class="py-lineno">476</tt>  <tt class="py-line">    <tt class="py-name">atom_list</tt> <tt class="py-op">=</tt> <tt id="link-195" class="py-name"><a title="Bio.PDB.Selection" class="py-name" href="#" onclick="return doclink('link-195', 'Selection', 'link-5');">Selection</a></tt><tt class="py-op">.</tt><tt id="link-196" class="py-name" targets="Function Bio.PDB.Selection.unfold_entities()=Bio.PDB.Selection-module.html#unfold_entities"><a title="Bio.PDB.Selection.unfold_entities" class="py-name" href="#" onclick="return doclink('link-196', 'unfold_entities', 'link-196');">unfold_entities</a></tt><tt class="py-op">(</tt><tt class="py-name">model</tt><tt class="py-op">,</tt> <tt class="py-string">'A'</tt><tt class="py-op">)</tt> </tt>
<a name="L477"></a><tt class="py-lineno">477</tt>  <tt class="py-line"> </tt>
<a name="L478"></a><tt class="py-lineno">478</tt>  <tt class="py-line">    <tt class="py-name">xyz_tmp</tt> <tt class="py-op">=</tt> <tt class="py-name">tempfile</tt><tt class="py-op">.</tt><tt class="py-name">mktemp</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L479"></a><tt class="py-lineno">479</tt>  <tt class="py-line">    <tt class="py-keyword">with</tt> <tt id="link-197" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-197', 'open', 'link-189');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">xyz_tmp</tt><tt class="py-op">,</tt> <tt class="py-string">'w'</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">pdb_to_xyzr</tt><tt class="py-op">:</tt> </tt>
<a name="L480"></a><tt class="py-lineno">480</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">atom</tt> <tt class="py-keyword">in</tt> <tt class="py-name">atom_list</tt><tt class="py-op">:</tt> </tt>
<a name="L481"></a><tt class="py-lineno">481</tt>  <tt class="py-line">            <tt id="link-198" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-198', 'x', 'link-191');">x</a></tt><tt class="py-op">,</tt> <tt id="link-199" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.y=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#y"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-199', 'y', 'link-199');">y</a></tt><tt class="py-op">,</tt> <tt class="py-name">z</tt> <tt class="py-op">=</tt> <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt class="py-name">coord</tt> </tt>
<a name="L482"></a><tt class="py-lineno">482</tt>  <tt class="py-line">            <tt class="py-name">radius</tt> <tt class="py-op">=</tt> <tt id="link-200" class="py-name" targets="Function Bio.PDB.ResidueDepth'._get_atom_radius()=Bio.PDB.ResidueDepth%27-module.html#_get_atom_radius"><a title="Bio.PDB.ResidueDepth'._get_atom_radius" class="py-name" href="#" onclick="return doclink('link-200', '_get_atom_radius', 'link-200');">_get_atom_radius</a></tt><tt class="py-op">(</tt><tt class="py-name">atom</tt><tt class="py-op">,</tt> <tt class="py-name">rtype</tt><tt class="py-op">=</tt><tt class="py-string">'united'</tt><tt class="py-op">)</tt> </tt>
<a name="L483"></a><tt class="py-lineno">483</tt>  <tt class="py-line">            <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">'{:6.3f}\t{:6.3f}\t{:6.3f}\t{:1.2f}'</tt><tt class="py-op">.</tt><tt id="link-201" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-201', 'format', 'link-187');">format</a></tt><tt class="py-op">(</tt><tt id="link-202" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-202', 'x', 'link-191');">x</a></tt><tt class="py-op">,</tt> <tt id="link-203" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-203', 'y', 'link-199');">y</a></tt><tt class="py-op">,</tt> <tt class="py-name">z</tt><tt class="py-op">,</tt> <tt class="py-name">radius</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L484"></a><tt class="py-lineno">484</tt>  <tt class="py-line">                  <tt class="py-name">file</tt><tt class="py-op">=</tt><tt class="py-name">pdb_to_xyzr</tt><tt class="py-op">)</tt> </tt>
<a name="L485"></a><tt class="py-lineno">485</tt>  <tt class="py-line"> </tt>
<a name="L486"></a><tt class="py-lineno">486</tt>  <tt class="py-line">    <tt class="py-comment"># make surface</tt> </tt>
<a name="L487"></a><tt class="py-lineno">487</tt>  <tt class="py-line">    <tt class="py-name">surface_tmp</tt> <tt class="py-op">=</tt> <tt class="py-name">tempfile</tt><tt class="py-op">.</tt><tt class="py-name">mktemp</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L488"></a><tt class="py-lineno">488</tt>  <tt class="py-line">    <tt class="py-name">MSMS</tt> <tt class="py-op">=</tt> <tt class="py-name">MSMS</tt> <tt class="py-op">+</tt> <tt class="py-string">" -probe_radius 1.5 -if %s -of %s &gt; "</tt> <tt class="py-op">+</tt> <tt class="py-name">tempfile</tt><tt class="py-op">.</tt><tt class="py-name">mktemp</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L489"></a><tt class="py-lineno">489</tt>  <tt class="py-line">    <tt class="py-name">make_surface</tt> <tt class="py-op">=</tt> <tt class="py-name">MSMS</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">xyz_tmp</tt><tt class="py-op">,</tt> <tt class="py-name">surface_tmp</tt><tt class="py-op">)</tt> </tt>
<a name="L490"></a><tt class="py-lineno">490</tt>  <tt class="py-line">    <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">system</tt><tt class="py-op">(</tt><tt class="py-name">make_surface</tt><tt class="py-op">)</tt> </tt>
<a name="L491"></a><tt class="py-lineno">491</tt>  <tt class="py-line">    <tt class="py-name">surface_file</tt> <tt class="py-op">=</tt> <tt class="py-name">surface_tmp</tt> <tt class="py-op">+</tt> <tt class="py-string">".vert"</tt> </tt>
<a name="L492"></a><tt class="py-lineno">492</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">isfile</tt><tt class="py-op">(</tt><tt class="py-name">surface_file</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L493"></a><tt class="py-lineno">493</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">RuntimeError</tt><tt class="py-op">(</tt><tt class="py-string">"Failed to generate surface file using "</tt> </tt>
<a name="L494"></a><tt class="py-lineno">494</tt>  <tt class="py-line">                           <tt class="py-string">"command:\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">make_surface</tt><tt class="py-op">)</tt> </tt>
<a name="L495"></a><tt class="py-lineno">495</tt>  <tt class="py-line"> </tt>
<a name="L496"></a><tt class="py-lineno">496</tt>  <tt class="py-line">    <tt class="py-comment"># read surface vertices from vertex file</tt> </tt>
<a name="L497"></a><tt class="py-lineno">497</tt>  <tt class="py-line">    <tt class="py-name">surface</tt> <tt class="py-op">=</tt> <tt id="link-204" class="py-name" targets="Function Bio.PDB.ResidueDepth'._read_vertex_array()=Bio.PDB.ResidueDepth%27-module.html#_read_vertex_array"><a title="Bio.PDB.ResidueDepth'._read_vertex_array" class="py-name" href="#" onclick="return doclink('link-204', '_read_vertex_array', 'link-204');">_read_vertex_array</a></tt><tt class="py-op">(</tt><tt class="py-name">surface_file</tt><tt class="py-op">)</tt> </tt>
<a name="L498"></a><tt class="py-lineno">498</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">surface</tt> </tt>
</div><a name="L499"></a><tt class="py-lineno">499</tt>  <tt class="py-line"> </tt>
<a name="L500"></a><tt class="py-lineno">500</tt>  <tt class="py-line"> </tt>
<a name="min_dist"></a><div id="min_dist-def"><a name="L501"></a><tt class="py-lineno">501</tt> <a class="py-toggle" href="#" id="min_dist-toggle" onclick="return toggle('min_dist');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.PDB.ResidueDepth'-module.html#min_dist">min_dist</a><tt class="py-op">(</tt><tt class="py-param">coord</tt><tt class="py-op">,</tt> <tt class="py-param">surface</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="min_dist-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="min_dist-expanded"><a name="L502"></a><tt class="py-lineno">502</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return minimum distance between coord and surface."""</tt> </tt>
<a name="L503"></a><tt class="py-lineno">503</tt>  <tt class="py-line">    <tt class="py-name">d</tt> <tt class="py-op">=</tt> <tt class="py-name">surface</tt> <tt class="py-op">-</tt> <tt class="py-name">coord</tt> </tt>
<a name="L504"></a><tt class="py-lineno">504</tt>  <tt class="py-line">    <tt class="py-name">d2</tt> <tt class="py-op">=</tt> <tt class="py-name">numpy</tt><tt class="py-op">.</tt><tt id="link-205" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.sum()=Bio.Nexus.Nexus.StepMatrix-class.html#sum,Method Bio.SubsMat.SeqMat.sum()=Bio.SubsMat.SeqMat-class.html#sum"><a title="Bio.Nexus.Nexus.StepMatrix.sum
Bio.SubsMat.SeqMat.sum" class="py-name" href="#" onclick="return doclink('link-205', 'sum', 'link-205');">sum</a></tt><tt class="py-op">(</tt><tt class="py-name">d</tt> <tt class="py-op">*</tt> <tt class="py-name">d</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L505"></a><tt class="py-lineno">505</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">numpy</tt><tt class="py-op">.</tt><tt class="py-name">sqrt</tt><tt class="py-op">(</tt><tt id="link-206" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.min=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#min"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-206', 'min', 'link-206');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">d2</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L506"></a><tt class="py-lineno">506</tt>  <tt class="py-line"> </tt>
<a name="L507"></a><tt class="py-lineno">507</tt>  <tt class="py-line"> </tt>
<a name="residue_depth"></a><div id="residue_depth-def"><a name="L508"></a><tt class="py-lineno">508</tt> <a class="py-toggle" href="#" id="residue_depth-toggle" onclick="return toggle('residue_depth');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.PDB.ResidueDepth'-module.html#residue_depth">residue_depth</a><tt class="py-op">(</tt><tt class="py-param">residue</tt><tt class="py-op">,</tt> <tt class="py-param">surface</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="residue_depth-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="residue_depth-expanded"><a name="L509"></a><tt class="py-lineno">509</tt>  <tt class="py-line">    <tt class="py-docstring">"""Residue depth as average depth of all its atoms.</tt> </tt>
<a name="L510"></a><tt class="py-lineno">510</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L511"></a><tt class="py-lineno">511</tt>  <tt class="py-line"><tt class="py-docstring">    Return average distance to surface for all atoms in a residue,</tt> </tt>
<a name="L512"></a><tt class="py-lineno">512</tt>  <tt class="py-line"><tt class="py-docstring">    ie. the residue depth.</tt> </tt>
<a name="L513"></a><tt class="py-lineno">513</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L514"></a><tt class="py-lineno">514</tt>  <tt class="py-line">    <tt class="py-name">atom_list</tt> <tt class="py-op">=</tt> <tt class="py-name">residue</tt><tt class="py-op">.</tt><tt id="link-207" class="py-name" targets="Method Bio.PDB.Chain.Chain.get_unpacked_list()=Bio.PDB.Chain.Chain-class.html#get_unpacked_list,Method Bio.PDB.Residue.Residue.get_unpacked_list()=Bio.PDB.Residue.Residue-class.html#get_unpacked_list"><a title="Bio.PDB.Chain.Chain.get_unpacked_list
Bio.PDB.Residue.Residue.get_unpacked_list" class="py-name" href="#" onclick="return doclink('link-207', 'get_unpacked_list', 'link-207');">get_unpacked_list</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L515"></a><tt class="py-lineno">515</tt>  <tt class="py-line">    <tt id="link-208" class="py-name" targets="Method Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length()=Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer-class.html#length"><a title="Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-208', 'length', 'link-208');">length</a></tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">atom_list</tt><tt class="py-op">)</tt> </tt>
<a name="L516"></a><tt class="py-lineno">516</tt>  <tt class="py-line">    <tt class="py-name">d</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L517"></a><tt class="py-lineno">517</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">atom</tt> <tt class="py-keyword">in</tt> <tt class="py-name">atom_list</tt><tt class="py-op">:</tt> </tt>
<a name="L518"></a><tt class="py-lineno">518</tt>  <tt class="py-line">        <tt class="py-name">coord</tt> <tt class="py-op">=</tt> <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-209" class="py-name" targets="Method Bio.PDB.Atom.Atom.get_coord()=Bio.PDB.Atom.Atom-class.html#get_coord"><a title="Bio.PDB.Atom.Atom.get_coord" class="py-name" href="#" onclick="return doclink('link-209', 'get_coord', 'link-209');">get_coord</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L519"></a><tt class="py-lineno">519</tt>  <tt class="py-line">        <tt class="py-name">d</tt> <tt class="py-op">=</tt> <tt class="py-name">d</tt> <tt class="py-op">+</tt> <tt id="link-210" class="py-name" targets="Function Bio.PDB.ResidueDepth'.min_dist()=Bio.PDB.ResidueDepth%27-module.html#min_dist"><a title="Bio.PDB.ResidueDepth'.min_dist" class="py-name" href="#" onclick="return doclink('link-210', 'min_dist', 'link-210');">min_dist</a></tt><tt class="py-op">(</tt><tt class="py-name">coord</tt><tt class="py-op">,</tt> <tt class="py-name">surface</tt><tt class="py-op">)</tt> </tt>
<a name="L520"></a><tt class="py-lineno">520</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">d</tt> <tt class="py-op">/</tt> <tt id="link-211" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-211', 'length', 'link-208');">length</a></tt> </tt>
</div><a name="L521"></a><tt class="py-lineno">521</tt>  <tt class="py-line"> </tt>
<a name="L522"></a><tt class="py-lineno">522</tt>  <tt class="py-line"> </tt>
<a name="ca_depth"></a><div id="ca_depth-def"><a name="L523"></a><tt class="py-lineno">523</tt> <a class="py-toggle" href="#" id="ca_depth-toggle" onclick="return toggle('ca_depth');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.PDB.ResidueDepth'-module.html#ca_depth">ca_depth</a><tt class="py-op">(</tt><tt class="py-param">residue</tt><tt class="py-op">,</tt> <tt class="py-param">surface</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ca_depth-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="ca_depth-expanded"><a name="L524"></a><tt class="py-lineno">524</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return CA depth."""</tt> </tt>
<a name="L525"></a><tt class="py-lineno">525</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">residue</tt><tt class="py-op">.</tt><tt id="link-212" class="py-name" targets="Method Bio.PDB.Chain.Chain.has_id()=Bio.PDB.Chain.Chain-class.html#has_id,Method Bio.PDB.Entity.Entity.has_id()=Bio.PDB.Entity.Entity-class.html#has_id"><a title="Bio.PDB.Chain.Chain.has_id
Bio.PDB.Entity.Entity.has_id" class="py-name" href="#" onclick="return doclink('link-212', 'has_id', 'link-212');">has_id</a></tt><tt class="py-op">(</tt><tt class="py-string">"CA"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L526"></a><tt class="py-lineno">526</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
<a name="L527"></a><tt class="py-lineno">527</tt>  <tt class="py-line">    <tt class="py-name">ca</tt> <tt class="py-op">=</tt> <tt class="py-name">residue</tt><tt class="py-op">[</tt><tt class="py-string">"CA"</tt><tt class="py-op">]</tt> </tt>
<a name="L528"></a><tt class="py-lineno">528</tt>  <tt class="py-line">    <tt class="py-name">coord</tt> <tt class="py-op">=</tt> <tt class="py-name">ca</tt><tt class="py-op">.</tt><tt id="link-213" class="py-name"><a title="Bio.PDB.Atom.Atom.get_coord" class="py-name" href="#" onclick="return doclink('link-213', 'get_coord', 'link-209');">get_coord</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L529"></a><tt class="py-lineno">529</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-214" class="py-name"><a title="Bio.PDB.ResidueDepth'.min_dist" class="py-name" href="#" onclick="return doclink('link-214', 'min_dist', 'link-210');">min_dist</a></tt><tt class="py-op">(</tt><tt class="py-name">coord</tt><tt class="py-op">,</tt> <tt class="py-name">surface</tt><tt class="py-op">)</tt> </tt>
</div><a name="L530"></a><tt class="py-lineno">530</tt>  <tt class="py-line"> </tt>
<a name="L531"></a><tt class="py-lineno">531</tt>  <tt class="py-line"> </tt>
<a name="ResidueDepth"></a><div id="ResidueDepth-def"><a name="L532"></a><tt class="py-lineno">532</tt> <a class="py-toggle" href="#" id="ResidueDepth-toggle" onclick="return toggle('ResidueDepth');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.PDB.ResidueDepth'.ResidueDepth-class.html">ResidueDepth</a><tt class="py-op">(</tt><tt class="py-base-class">AbstractPropertyMap</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ResidueDepth-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="ResidueDepth-expanded"><a name="L533"></a><tt class="py-lineno">533</tt>  <tt class="py-line">    <tt class="py-docstring">"""Calculate residue and CA depth for all residues."""</tt> </tt>
<a name="L534"></a><tt class="py-lineno">534</tt>  <tt class="py-line"> </tt>
<a name="ResidueDepth.__init__"></a><div id="ResidueDepth.__init__-def"><a name="L535"></a><tt class="py-lineno">535</tt> <a class="py-toggle" href="#" id="ResidueDepth.__init__-toggle" onclick="return toggle('ResidueDepth.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.PDB.ResidueDepth'.ResidueDepth-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">model</tt><tt class="py-op">,</tt> <tt class="py-param">pdb_file</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ResidueDepth.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ResidueDepth.__init__-expanded"><a name="L536"></a><tt class="py-lineno">536</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the class."""</tt> </tt>
<a name="L537"></a><tt class="py-lineno">537</tt>  <tt class="py-line">        <tt class="py-comment"># Issue warning if pdb_file is given</tt> </tt>
<a name="L538"></a><tt class="py-lineno">538</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">pdb_file</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L539"></a><tt class="py-lineno">539</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"ResidueDepth no longer requires a pdb file. "</tt> </tt>
<a name="L540"></a><tt class="py-lineno">540</tt>  <tt class="py-line">                          <tt class="py-string">"This argument will be removed in a future release "</tt> </tt>
<a name="L541"></a><tt class="py-lineno">541</tt>  <tt class="py-line">                          <tt class="py-string">"of Biopython."</tt><tt class="py-op">,</tt> <tt id="link-215" class="py-name"><a title="Bio.BiopythonDeprecationWarning" class="py-name" href="#" onclick="return doclink('link-215', 'BiopythonDeprecationWarning', 'link-17');">BiopythonDeprecationWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L542"></a><tt class="py-lineno">542</tt>  <tt class="py-line"> </tt>
<a name="L543"></a><tt class="py-lineno">543</tt>  <tt class="py-line">        <tt class="py-name">depth_dict</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L544"></a><tt class="py-lineno">544</tt>  <tt class="py-line">        <tt class="py-name">depth_list</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L545"></a><tt class="py-lineno">545</tt>  <tt class="py-line">        <tt class="py-name">depth_keys</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L546"></a><tt class="py-lineno">546</tt>  <tt class="py-line">        <tt class="py-comment"># get_residue</tt> </tt>
<a name="L547"></a><tt class="py-lineno">547</tt>  <tt class="py-line">        <tt class="py-name">residue_list</tt> <tt class="py-op">=</tt> <tt id="link-216" class="py-name"><a title="Bio.PDB.Selection" class="py-name" href="#" onclick="return doclink('link-216', 'Selection', 'link-5');">Selection</a></tt><tt class="py-op">.</tt><tt id="link-217" class="py-name"><a title="Bio.PDB.Selection.unfold_entities" class="py-name" href="#" onclick="return doclink('link-217', 'unfold_entities', 'link-196');">unfold_entities</a></tt><tt class="py-op">(</tt><tt class="py-name">model</tt><tt class="py-op">,</tt> <tt class="py-string">'R'</tt><tt class="py-op">)</tt> </tt>
<a name="L548"></a><tt class="py-lineno">548</tt>  <tt class="py-line">        <tt class="py-comment"># make surface from PDB file using MSMS</tt> </tt>
<a name="L549"></a><tt class="py-lineno">549</tt>  <tt class="py-line">        <tt class="py-name">surface</tt> <tt class="py-op">=</tt> <tt id="link-218" class="py-name" targets="Function Bio.PDB.ResidueDepth'.get_surface()=Bio.PDB.ResidueDepth%27-module.html#get_surface"><a title="Bio.PDB.ResidueDepth'.get_surface" class="py-name" href="#" onclick="return doclink('link-218', 'get_surface', 'link-218');">get_surface</a></tt><tt class="py-op">(</tt><tt class="py-name">model</tt><tt class="py-op">)</tt> </tt>
<a name="L550"></a><tt class="py-lineno">550</tt>  <tt class="py-line">        <tt class="py-comment"># calculate rdepth for each residue</tt> </tt>
<a name="L551"></a><tt class="py-lineno">551</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">residue</tt> <tt class="py-keyword">in</tt> <tt class="py-name">residue_list</tt><tt class="py-op">:</tt> </tt>
<a name="L552"></a><tt class="py-lineno">552</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-219" class="py-name"><a title="Bio.PDB.Polypeptide.is_aa" class="py-name" href="#" onclick="return doclink('link-219', 'is_aa', 'link-13');">is_aa</a></tt><tt class="py-op">(</tt><tt class="py-name">residue</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L553"></a><tt class="py-lineno">553</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L554"></a><tt class="py-lineno">554</tt>  <tt class="py-line">            <tt id="link-220" class="py-name" targets="Class Bio.Sequencing.Ace.rd=Bio.Sequencing.Ace.rd-class.html"><a title="Bio.Sequencing.Ace.rd" class="py-name" href="#" onclick="return doclink('link-220', 'rd', 'link-220');">rd</a></tt> <tt class="py-op">=</tt> <tt id="link-221" class="py-name" targets="Function Bio.PDB.ResidueDepth'.residue_depth()=Bio.PDB.ResidueDepth%27-module.html#residue_depth"><a title="Bio.PDB.ResidueDepth'.residue_depth" class="py-name" href="#" onclick="return doclink('link-221', 'residue_depth', 'link-221');">residue_depth</a></tt><tt class="py-op">(</tt><tt class="py-name">residue</tt><tt class="py-op">,</tt> <tt class="py-name">surface</tt><tt class="py-op">)</tt> </tt>
<a name="L555"></a><tt class="py-lineno">555</tt>  <tt class="py-line">            <tt class="py-name">ca_rd</tt> <tt class="py-op">=</tt> <tt id="link-222" class="py-name" targets="Function Bio.PDB.ResidueDepth'.ca_depth()=Bio.PDB.ResidueDepth%27-module.html#ca_depth"><a title="Bio.PDB.ResidueDepth'.ca_depth" class="py-name" href="#" onclick="return doclink('link-222', 'ca_depth', 'link-222');">ca_depth</a></tt><tt class="py-op">(</tt><tt class="py-name">residue</tt><tt class="py-op">,</tt> <tt class="py-name">surface</tt><tt class="py-op">)</tt> </tt>
<a name="L556"></a><tt class="py-lineno">556</tt>  <tt class="py-line">            <tt class="py-comment"># Get the key</tt> </tt>
<a name="L557"></a><tt class="py-lineno">557</tt>  <tt class="py-line">            <tt class="py-name">res_id</tt> <tt class="py-op">=</tt> <tt class="py-name">residue</tt><tt class="py-op">.</tt><tt id="link-223" class="py-name" targets="Method Bio.Nexus.Nodes.Node.get_id()=Bio.Nexus.Nodes.Node-class.html#get_id,Method Bio.PDB.Atom.Atom.get_id()=Bio.PDB.Atom.Atom-class.html#get_id,Method Bio.PDB.Entity.DisorderedEntityWrapper.get_id()=Bio.PDB.Entity.DisorderedEntityWrapper-class.html#get_id,Method Bio.PDB.Entity.Entity.get_id()=Bio.PDB.Entity.Entity-class.html#get_id,Method Bio.PDB.FragmentMapper'.Fragment.get_id()=Bio.PDB.FragmentMapper%27.Fragment-class.html#get_id"><a title="Bio.Nexus.Nodes.Node.get_id
Bio.PDB.Atom.Atom.get_id
Bio.PDB.Entity.DisorderedEntityWrapper.get_id
Bio.PDB.Entity.Entity.get_id
Bio.PDB.FragmentMapper'.Fragment.get_id" class="py-name" href="#" onclick="return doclink('link-223', 'get_id', 'link-223');">get_id</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L558"></a><tt class="py-lineno">558</tt>  <tt class="py-line">            <tt id="link-224" class="py-name" targets="Variable Bio.FSSP.fssp_rec.align.chain_id=Bio.FSSP.fssp_rec.align-class.html#chain_id"><a title="Bio.FSSP.fssp_rec.align.chain_id" class="py-name" href="#" onclick="return doclink('link-224', 'chain_id', 'link-224');">chain_id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">residue</tt><tt class="py-op">.</tt><tt id="link-225" class="py-name" targets="Method Bio.PDB.Atom.Atom.get_parent()=Bio.PDB.Atom.Atom-class.html#get_parent,Method Bio.PDB.Entity.DisorderedEntityWrapper.get_parent()=Bio.PDB.Entity.DisorderedEntityWrapper-class.html#get_parent,Method Bio.PDB.Entity.Entity.get_parent()=Bio.PDB.Entity.Entity-class.html#get_parent"><a title="Bio.PDB.Atom.Atom.get_parent
Bio.PDB.Entity.DisorderedEntityWrapper.get_parent
Bio.PDB.Entity.Entity.get_parent" class="py-name" href="#" onclick="return doclink('link-225', 'get_parent', 'link-225');">get_parent</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-226" class="py-name"><a title="Bio.Nexus.Nodes.Node.get_id
Bio.PDB.Atom.Atom.get_id
Bio.PDB.Entity.DisorderedEntityWrapper.get_id
Bio.PDB.Entity.Entity.get_id
Bio.PDB.FragmentMapper'.Fragment.get_id" class="py-name" href="#" onclick="return doclink('link-226', 'get_id', 'link-223');">get_id</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L559"></a><tt class="py-lineno">559</tt>  <tt class="py-line">            <tt class="py-name">depth_dict</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt id="link-227" class="py-name"><a title="Bio.FSSP.fssp_rec.align.chain_id" class="py-name" href="#" onclick="return doclink('link-227', 'chain_id', 'link-224');">chain_id</a></tt><tt class="py-op">,</tt> <tt class="py-name">res_id</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt id="link-228" class="py-name"><a title="Bio.Sequencing.Ace.rd" class="py-name" href="#" onclick="return doclink('link-228', 'rd', 'link-220');">rd</a></tt><tt class="py-op">,</tt> <tt class="py-name">ca_rd</tt><tt class="py-op">)</tt> </tt>
<a name="L560"></a><tt class="py-lineno">560</tt>  <tt class="py-line">            <tt class="py-name">depth_list</tt><tt class="py-op">.</tt><tt id="link-229" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-229', 'append', 'link-193');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">residue</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt id="link-230" class="py-name"><a title="Bio.Sequencing.Ace.rd" class="py-name" href="#" onclick="return doclink('link-230', 'rd', 'link-220');">rd</a></tt><tt class="py-op">,</tt> <tt class="py-name">ca_rd</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L561"></a><tt class="py-lineno">561</tt>  <tt class="py-line">            <tt class="py-name">depth_keys</tt><tt class="py-op">.</tt><tt id="link-231" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-231', 'append', 'link-193');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt id="link-232" class="py-name"><a title="Bio.FSSP.fssp_rec.align.chain_id" class="py-name" href="#" onclick="return doclink('link-232', 'chain_id', 'link-224');">chain_id</a></tt><tt class="py-op">,</tt> <tt class="py-name">res_id</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L562"></a><tt class="py-lineno">562</tt>  <tt class="py-line">            <tt class="py-comment"># Update xtra information</tt> </tt>
<a name="L563"></a><tt class="py-lineno">563</tt>  <tt class="py-line">            <tt class="py-name">residue</tt><tt class="py-op">.</tt><tt class="py-name">xtra</tt><tt class="py-op">[</tt><tt class="py-string">'EXP_RD'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-233" class="py-name"><a title="Bio.Sequencing.Ace.rd" class="py-name" href="#" onclick="return doclink('link-233', 'rd', 'link-220');">rd</a></tt> </tt>
<a name="L564"></a><tt class="py-lineno">564</tt>  <tt class="py-line">            <tt class="py-name">residue</tt><tt class="py-op">.</tt><tt class="py-name">xtra</tt><tt class="py-op">[</tt><tt class="py-string">'EXP_RD_CA'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">ca_rd</tt> </tt>
<a name="L565"></a><tt class="py-lineno">565</tt>  <tt class="py-line">        <tt id="link-234" class="py-name"><a title="Bio.PDB.AbstractPropertyMap
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap" class="py-name" href="#" onclick="return doclink('link-234', 'AbstractPropertyMap', 'link-8');">AbstractPropertyMap</a></tt><tt class="py-op">.</tt><tt id="link-235" class="py-name" targets="Method Bio.Affy.CelFile.ParserError.__init__()=Bio.Affy.CelFile.ParserError-class.html#__init__,Method Bio.Affy.CelFile.Record.__init__()=Bio.Affy.CelFile.Record-class.html#__init__,Method Bio.Align.AlignInfo.PSSM.__init__()=Bio.Align.AlignInfo.PSSM-class.html#__init__,Method Bio.Align.AlignInfo.SummaryInfo.__init__()=Bio.Align.AlignInfo.SummaryInfo-class.html#__init__,Method Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline.__init__()=Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline-class.html#__init__,Method Bio.Align.Applications._Clustalw.ClustalwCommandline.__init__()=Bio.Align.Applications._Clustalw.ClustalwCommandline-class.html#__init__,Method Bio.Align.Applications._Dialign.DialignCommandline.__init__()=Bio.Align.Applications._Dialign.DialignCommandline-class.html#__init__,Method Bio.Align.Applications._MSAProbs.MSAProbsCommandline.__init__()=Bio.Align.Applications._MSAProbs.MSAProbsCommandline-class.html#__init__,Method Bio.Align.Applications._Mafft.MafftCommandline.__init__()=Bio.Align.Applications._Mafft.MafftCommandline-class.html#__init__,Method Bio.Align.Applications._Muscle.MuscleCommandline.__init__()=Bio.Align.Applications._Muscle.MuscleCommandline-class.html#__init__,Method Bio.Align.Applications._Prank.PrankCommandline.__init__()=Bio.Align.Applications._Prank.PrankCommandline-class.html#__init__,Method Bio.Align.Applications._Probcons.ProbconsCommandline.__init__()=Bio.Align.Applications._Probcons.ProbconsCommandline-class.html#__init__,Method Bio.Align.Applications._TCoffee.TCoffeeCommandline.__init__()=Bio.Align.Applications._TCoffee.TCoffeeCommandline-class.html#__init__,Method Bio.Align.MultipleSeqAlignment.__init__()=Bio.Align.MultipleSeqAlignment-class.html#__init__,Method Bio.Align.PairwiseAlignment.__init__()=Bio.Align.PairwiseAlignment-class.html#__init__,Method Bio.Align.PairwiseAlignments.__init__()=Bio.Align.PairwiseAlignments-class.html#__init__,Method Bio.AlignIO.Interfaces.AlignmentIterator.__init__()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#__init__,Method Bio.AlignIO.Interfaces.AlignmentWriter.__init__()=Bio.AlignIO.Interfaces.AlignmentWriter-class.html#__init__,Method Bio.AlignIO.Interfaces.SequentialAlignmentWriter.__init__()=Bio.AlignIO.Interfaces.SequentialAlignmentWriter-class.html#__init__,Method Bio.AlignIO.MafIO.MafIndex.__init__()=Bio.AlignIO.MafIO.MafIndex-class.html#__init__,Method Bio.AlignIO.MauveIO.MauveWriter.__init__()=Bio.AlignIO.MauveIO.MauveWriter-class.html#__init__,Method Bio.Alphabet.AlphabetEncoder.__init__()=Bio.Alphabet.AlphabetEncoder-class.html#__init__,Method Bio.Alphabet.Gapped.__init__()=Bio.Alphabet.Gapped-class.html#__init__,Method Bio.Alphabet.HasStopCodon.__init__()=Bio.Alphabet.HasStopCodon-class.html#__init__,Method Bio.Application.AbstractCommandline.__init__()=Bio.Application.AbstractCommandline-class.html#__init__,Method Bio.Application.ApplicationError.__init__()=Bio.Application.ApplicationError-class.html#__init__,Method Bio.Application._AbstractParameter.__init__()=Bio.Application._AbstractParameter-class.html#__init__,Method Bio.Application._Argument.__init__()=Bio.Application._Argument-class.html#__init__,Method Bio.Application._Option.__init__()=Bio.Application._Option-class.html#__init__,Method Bio.Application._StaticArgument.__init__()=Bio.Application._StaticArgument-class.html#__init__,Method Bio.Application._Switch.__init__()=Bio.Application._Switch-class.html#__init__,Method Bio.Blast.Applications.NcbiblastformatterCommandline.__init__()=Bio.Blast.Applications.NcbiblastformatterCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastnCommandline.__init__()=Bio.Blast.Applications.NcbiblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastpCommandline.__init__()=Bio.Blast.Applications.NcbiblastpCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastxCommandline.__init__()=Bio.Blast.Applications.NcbiblastxCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbideltablastCommandline.__init__()=Bio.Blast.Applications.NcbideltablastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbimakeblastdbCommandline.__init__()=Bio.Blast.Applications.NcbimakeblastdbCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbipsiblastCommandline.__init__()=Bio.Blast.Applications.NcbipsiblastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbirpsblastCommandline.__init__()=Bio.Blast.Applications.NcbirpsblastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbirpstblastnCommandline.__init__()=Bio.Blast.Applications.NcbirpstblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbitblastnCommandline.__init__()=Bio.Blast.Applications.NcbitblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbitblastxCommandline.__init__()=Bio.Blast.Applications.NcbitblastxCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbibaseblastCommandline.__init__()=Bio.Blast.Applications._NcbibaseblastCommandline-class.html#__init__,Method Bio.Blast.Applications._Ncbiblast2SeqCommandline.__init__()=Bio.Blast.Applications._Ncbiblast2SeqCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbiblastCommandline.__init__()=Bio.Blast.Applications._NcbiblastCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbiblastMain2SeqCommandline.__init__()=Bio.Blast.Applications._NcbiblastMain2SeqCommandline-class.html#__init__,Method Bio.Blast.NCBIXML.BlastParser.__init__()=Bio.Blast.NCBIXML.BlastParser-class.html#__init__,Method Bio.Blast.NCBIXML._XMLparser.__init__()=Bio.Blast.NCBIXML._XMLparser-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableEntry.__init__()=Bio.Blast.ParseBlastTable.BlastTableEntry-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableReader.__init__()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableRec.__init__()=Bio.Blast.ParseBlastTable.BlastTableRec-class.html#__init__,Method Bio.Blast.Record.Alignment.__init__()=Bio.Blast.Record.Alignment-class.html#__init__,Method Bio.Blast.Record.Blast.__init__()=Bio.Blast.Record.Blast-class.html#__init__,Method Bio.Blast.Record.DatabaseReport.__init__()=Bio.Blast.Record.DatabaseReport-class.html#__init__,Method Bio.Blast.Record.Description.__init__()=Bio.Blast.Record.Description-class.html#__init__,Method Bio.Blast.Record.DescriptionExt.__init__()=Bio.Blast.Record.DescriptionExt-class.html#__init__,Method Bio.Blast.Record.DescriptionExtItem.__init__()=Bio.Blast.Record.DescriptionExtItem-class.html#__init__,Method Bio.Blast.Record.HSP.__init__()=Bio.Blast.Record.HSP-class.html#__init__,Method Bio.Blast.Record.Header.__init__()=Bio.Blast.Record.Header-class.html#__init__,Method Bio.Blast.Record.MultipleAlignment.__init__()=Bio.Blast.Record.MultipleAlignment-class.html#__init__,Method Bio.Blast.Record.PSIBlast.__init__()=Bio.Blast.Record.PSIBlast-class.html#__init__,Method Bio.Blast.Record.Parameters.__init__()=Bio.Blast.Record.Parameters-class.html#__init__,Method Bio.Blast.Record.Round.__init__()=Bio.Blast.Record.Round-class.html#__init__,Method Bio.CAPS.CAPSMap.__init__()=Bio.CAPS.CAPSMap-class.html#__init__,Method Bio.CAPS.DifferentialCutsite.__init__()=Bio.CAPS.DifferentialCutsite-class.html#__init__,Method Bio.Cluster.Record.__init__()=Bio.Cluster.Record-class.html#__init__,Method Bio.Compass.Record.__init__()=Bio.Compass.Record-class.html#__init__,Method Bio.Crystal.Chain.__init__()=Bio.Crystal.Chain-class.html#__init__,Method Bio.Crystal.Crystal.__init__()=Bio.Crystal.Crystal-class.html#__init__,Method Bio.Crystal.Hetero.__init__()=Bio.Crystal.Hetero-class.html#__init__,Method Bio.Data.CodonTable.AmbiguousCodonTable.__init__()=Bio.Data.CodonTable.AmbiguousCodonTable-class.html#__init__,Method Bio.Data.CodonTable.AmbiguousForwardTable.__init__()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#__init__,Method Bio.Data.CodonTable.CodonTable.__init__()=Bio.Data.CodonTable.CodonTable-class.html#__init__,Method Bio.Data.CodonTable.NCBICodonTable.__init__()=Bio.Data.CodonTable.NCBICodonTable-class.html#__init__,Method Bio.Emboss.Applications.DiffseqCommandline.__init__()=Bio.Emboss.Applications.DiffseqCommandline-class.html#__init__,Method Bio.Emboss.Applications.EInvertedCommandline.__init__()=Bio.Emboss.Applications.EInvertedCommandline-class.html#__init__,Method Bio.Emboss.Applications.ETandemCommandline.__init__()=Bio.Emboss.Applications.ETandemCommandline-class.html#__init__,Method Bio.Emboss.Applications.Est2GenomeCommandline.__init__()=Bio.Emboss.Applications.Est2GenomeCommandline-class.html#__init__,Method Bio.Emboss.Applications.FConsenseCommandline.__init__()=Bio.Emboss.Applications.FConsenseCommandline-class.html#__init__,Method Bio.Emboss.Applications.FDNADistCommandline.__init__()=Bio.Emboss.Applications.FDNADistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FDNAParsCommandline.__init__()=Bio.Emboss.Applications.FDNAParsCommandline-class.html#__init__,Method Bio.Emboss.Applications.FNeighborCommandline.__init__()=Bio.Emboss.Applications.FNeighborCommandline-class.html#__init__,Method Bio.Emboss.Applications.FProtDistCommandline.__init__()=Bio.Emboss.Applications.FProtDistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FProtParsCommandline.__init__()=Bio.Emboss.Applications.FProtParsCommandline-class.html#__init__,Method Bio.Emboss.Applications.FSeqBootCommandline.__init__()=Bio.Emboss.Applications.FSeqBootCommandline-class.html#__init__,Method Bio.Emboss.Applications.FTreeDistCommandline.__init__()=Bio.Emboss.Applications.FTreeDistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FuzznucCommandline.__init__()=Bio.Emboss.Applications.FuzznucCommandline-class.html#__init__,Method Bio.Emboss.Applications.FuzzproCommandline.__init__()=Bio.Emboss.Applications.FuzzproCommandline-class.html#__init__,Method Bio.Emboss.Applications.IepCommandline.__init__()=Bio.Emboss.Applications.IepCommandline-class.html#__init__,Method Bio.Emboss.Applications.NeedleCommandline.__init__()=Bio.Emboss.Applications.NeedleCommandline-class.html#__init__,Method Bio.Emboss.Applications.NeedleallCommandline.__init__()=Bio.Emboss.Applications.NeedleallCommandline-class.html#__init__,Method Bio.Emboss.Applications.PalindromeCommandline.__init__()=Bio.Emboss.Applications.PalindromeCommandline-class.html#__init__,Method Bio.Emboss.Applications.Primer3Commandline.__init__()=Bio.Emboss.Applications.Primer3Commandline-class.html#__init__,Method Bio.Emboss.Applications.PrimerSearchCommandline.__init__()=Bio.Emboss.Applications.PrimerSearchCommandline-class.html#__init__,Method Bio.Emboss.Applications.SeqmatchallCommandline.__init__()=Bio.Emboss.Applications.SeqmatchallCommandline-class.html#__init__,Method Bio.Emboss.Applications.SeqretCommandline.__init__()=Bio.Emboss.Applications.SeqretCommandline-class.html#__init__,Method Bio.Emboss.Applications.StretcherCommandline.__init__()=Bio.Emboss.Applications.StretcherCommandline-class.html#__init__,Method Bio.Emboss.Applications.TranalignCommandline.__init__()=Bio.Emboss.Applications.TranalignCommandline-class.html#__init__,Method Bio.Emboss.Applications.WaterCommandline.__init__()=Bio.Emboss.Applications.WaterCommandline-class.html#__init__,Method Bio.Emboss.Applications._EmbossCommandLine.__init__()=Bio.Emboss.Applications._EmbossCommandLine-class.html#__init__,Method Bio.Emboss.Applications._EmbossMinimalCommandLine.__init__()=Bio.Emboss.Applications._EmbossMinimalCommandLine-class.html#__init__,Method Bio.Emboss.Primer3.Primers.__init__()=Bio.Emboss.Primer3.Primers-class.html#__init__,Method Bio.Emboss.Primer3.Record.__init__()=Bio.Emboss.Primer3.Record-class.html#__init__,Method Bio.Emboss.PrimerSearch.Amplifier.__init__()=Bio.Emboss.PrimerSearch.Amplifier-class.html#__init__,Method Bio.Emboss.PrimerSearch.InputRecord.__init__()=Bio.Emboss.PrimerSearch.InputRecord-class.html#__init__,Method Bio.Emboss.PrimerSearch.OutputRecord.__init__()=Bio.Emboss.PrimerSearch.OutputRecord-class.html#__init__,Method Bio.Entrez.Parser.CorruptedXMLError.__init__()=Bio.Entrez.Parser.CorruptedXMLError-class.html#__init__,Method Bio.Entrez.Parser.DataHandler.__init__()=Bio.Entrez.Parser.DataHandler-class.html#__init__,Method Bio.Entrez.Parser.DictionaryElement.__init__()=Bio.Entrez.Parser.DictionaryElement-class.html#__init__,Method Bio.Entrez.Parser.ListElement.__init__()=Bio.Entrez.Parser.ListElement-class.html#__init__,Method Bio.Entrez.Parser.NoneElement.__init__()=Bio.Entrez.Parser.NoneElement-class.html#__init__,Method Bio.Entrez.Parser.NotXMLError.__init__()=Bio.Entrez.Parser.NotXMLError-class.html#__init__,Method Bio.Entrez.Parser.ValidationError.__init__()=Bio.Entrez.Parser.ValidationError-class.html#__init__,Method Bio.ExPASy.Enzyme.Record.__init__()=Bio.ExPASy.Enzyme.Record-class.html#__init__,Method Bio.ExPASy.Prodoc.Record.__init__()=Bio.ExPASy.Prodoc.Record-class.html#__init__,Method Bio.ExPASy.Prodoc.Reference.__init__()=Bio.ExPASy.Prodoc.Reference-class.html#__init__,Method Bio.ExPASy.Prosite.Record.__init__()=Bio.ExPASy.Prosite.Record-class.html#__init__,Method Bio.ExPASy.ScanProsite.ContentHandler.__init__()=Bio.ExPASy.ScanProsite.ContentHandler-class.html#__init__,Method Bio.ExPASy.ScanProsite.Parser.__init__()=Bio.ExPASy.ScanProsite.Parser-class.html#__init__,Method Bio.ExPASy.ScanProsite.Record.__init__()=Bio.ExPASy.ScanProsite.Record-class.html#__init__,Method Bio.ExPASy.cellosaurus.Record.__init__()=Bio.ExPASy.cellosaurus.Record-class.html#__init__,Method Bio.FSSP.FSSPAlignDict.__init__()=Bio.FSSP.FSSPAlignDict-class.html#__init__,Method Bio.FSSP.FSSPAlignRec.__init__()=Bio.FSSP.FSSPAlignRec-class.html#__init__,Method Bio.FSSP.FSSPHeader.__init__()=Bio.FSSP.FSSPHeader-class.html#__init__,Method Bio.FSSP.FSSPSumRec.__init__()=Bio.FSSP.FSSPSumRec-class.html#__init__,Method Bio.FSSP.FSSPTools.FSSPMultAlign.__init__()=Bio.FSSP.FSSPTools.FSSPMultAlign-class.html#__init__,Method Bio.FSSP.PosAlign.__init__()=Bio.FSSP.PosAlign-class.html#__init__,Method Bio.FSSP.fssp_rec.fff_rec.__init__()=Bio.FSSP.fssp_rec.fff_rec-class.html#__init__,Method Bio.File.UndoHandle.__init__()=Bio.File.UndoHandle-class.html#__init__,Method Bio.File._IndexedSeqFileDict.__init__()=Bio.File._IndexedSeqFileDict-class.html#__init__,Method Bio.File._SQLiteManySeqFilesDict.__init__()=Bio.File._SQLiteManySeqFilesDict-class.html#__init__,Method Bio.GenBank.FeatureParser.__init__()=Bio.GenBank.FeatureParser-class.html#__init__,Method Bio.GenBank.Iterator.__init__()=Bio.GenBank.Iterator-class.html#__init__,Method Bio.GenBank.Record.Feature.__init__()=Bio.GenBank.Record.Feature-class.html#__init__,Method Bio.GenBank.Record.Qualifier.__init__()=Bio.GenBank.Record.Qualifier-class.html#__init__,Method Bio.GenBank.Record.Record.__init__()=Bio.GenBank.Record.Record-class.html#__init__,Method Bio.GenBank.Record.Reference.__init__()=Bio.GenBank.Record.Reference-class.html#__init__,Method Bio.GenBank.RecordParser.__init__()=Bio.GenBank.RecordParser-class.html#__init__,Method Bio.GenBank.Scanner.InsdcScanner.__init__()=Bio.GenBank.Scanner.InsdcScanner-class.html#__init__,Method Bio.GenBank._BaseGenBankConsumer.__init__()=Bio.GenBank._BaseGenBankConsumer-class.html#__init__,Method Bio.GenBank._FeatureConsumer.__init__()=Bio.GenBank._FeatureConsumer-class.html#__init__,Method Bio.GenBank._RecordConsumer.__init__()=Bio.GenBank._RecordConsumer-class.html#__init__,Method Bio.GenBank.utils.FeatureValueCleaner.__init__()=Bio.GenBank.utils.FeatureValueCleaner-class.html#__init__,Method Bio.Geo.Record.Record.__init__()=Bio.Geo.Record.Record-class.html#__init__,Method Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment.__init__()=Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome.Chromosome.__init__()=Bio.Graphics.BasicChromosome.Chromosome-class.html#__init__,Method Bio.Graphics.BasicChromosome.ChromosomeSegment.__init__()=Bio.Graphics.BasicChromosome.ChromosomeSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome.Organism.__init__()=Bio.Graphics.BasicChromosome.Organism-class.html#__init__,Method Bio.Graphics.BasicChromosome.TelomereSegment.__init__()=Bio.Graphics.BasicChromosome.TelomereSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome._ChromosomeComponent.__init__()=Bio.Graphics.BasicChromosome._ChromosomeComponent-class.html#__init__,Method Bio.Graphics.ColorSpiral.ColorSpiral.__init__()=Bio.Graphics.ColorSpiral.ColorSpiral-class.html#__init__,Method Bio.Graphics.Comparative.ComparativeScatterPlot.__init__()=Bio.Graphics.Comparative.ComparativeScatterPlot-class.html#__init__,Method Bio.Graphics.DisplayRepresentation.ChromosomeCounts.__init__()=Bio.Graphics.DisplayRepresentation.ChromosomeCounts-class.html#__init__,Method Bio.Graphics.Distribution.BarChartDistribution.__init__()=Bio.Graphics.Distribution.BarChartDistribution-class.html#__init__,Method Bio.Graphics.Distribution.DistributionPage.__init__()=Bio.Graphics.Distribution.DistributionPage-class.html#__init__,Method Bio.Graphics.Distribution.LineDistribution.__init__()=Bio.Graphics.Distribution.LineDistribution-class.html#__init__,Method Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.__init__()=Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.__init__()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.__init__()=Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#__init__,Method Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.__init__()=Bio.Graphics.GenomeDiagram._CrossLink.CrossLink-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.__init__()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Feature.Feature.__init__()=Bio.Graphics.GenomeDiagram._Feature.Feature-class.html#__init__,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__init__()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.__init__()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#__init__,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__init__()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#__init__,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.__init__()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Track.Track.__init__()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#__init__,Method Bio.Graphics.KGML_vis.KGMLCanvas.__init__()=Bio.Graphics.KGML_vis.KGMLCanvas-class.html#__init__,Method Bio.HMM.DynamicProgramming.AbstractDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.AbstractDPAlgorithms-class.html#__init__,Method Bio.HMM.DynamicProgramming.LogDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.LogDPAlgorithms-class.html#__init__,Method Bio.HMM.DynamicProgramming.ScaledDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.ScaledDPAlgorithms-class.html#__init__,Method Bio.HMM.MarkovModel.HiddenMarkovModel.__init__()=Bio.HMM.MarkovModel.HiddenMarkovModel-class.html#__init__,Method Bio.HMM.MarkovModel.MarkovModelBuilder.__init__()=Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#__init__,Method Bio.HMM.Trainer.AbstractTrainer.__init__()=Bio.HMM.Trainer.AbstractTrainer-class.html#__init__,Method Bio.HMM.Trainer.BaumWelchTrainer.__init__()=Bio.HMM.Trainer.BaumWelchTrainer-class.html#__init__,Method Bio.HMM.Trainer.KnownStateTrainer.__init__()=Bio.HMM.Trainer.KnownStateTrainer-class.html#__init__,Method Bio.HMM.Trainer.TrainingSequence.__init__()=Bio.HMM.Trainer.TrainingSequence-class.html#__init__,Method Bio.Index._InMemoryIndex.__init__()=Bio.Index._InMemoryIndex-class.html#__init__,Method Bio.Index._ShelveIndex.__init__()=Bio.Index._ShelveIndex-class.html#__init__,Method Bio.KDTree.KDTree'.KDTree.__init__()=Bio.KDTree.KDTree%27.KDTree-class.html#__init__,Method Bio.KEGG.Compound.Record.__init__()=Bio.KEGG.Compound.Record-class.html#__init__,Method Bio.KEGG.Enzyme.Record.__init__()=Bio.KEGG.Enzyme.Record-class.html#__init__,Method Bio.KEGG.Gene.Record.__init__()=Bio.KEGG.Gene.Record-class.html#__init__,Method Bio.KEGG.KGML.KGML_parser.KGMLParser.__init__()=Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Component.__init__()=Bio.KEGG.KGML.KGML_pathway.Component-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Entry.__init__()=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Graphics.__init__()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Pathway.__init__()=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Reaction.__init__()=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Relation.__init__()=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#__init__,Method Bio.LogisticRegression.LogisticRegression.__init__()=Bio.LogisticRegression.LogisticRegression-class.html#__init__,Method Bio.MarkovModel.MarkovModel.__init__()=Bio.MarkovModel.MarkovModel-class.html#__init__,Method Bio.MaxEntropy.MaxEntropy.__init__()=Bio.MaxEntropy.MaxEntropy-class.html#__init__,Method Bio.NMR.xpktools.Peaklist.__init__()=Bio.NMR.xpktools.Peaklist-class.html#__init__,Method Bio.NMR.xpktools.XpkEntry.__init__()=Bio.NMR.xpktools.XpkEntry-class.html#__init__,Method Bio.NaiveBayes.NaiveBayes.__init__()=Bio.NaiveBayes.NaiveBayes-class.html#__init__,Method Bio.Nexus.Nexus.Block.__init__()=Bio.Nexus.Nexus.Block-class.html#__init__,Method Bio.Nexus.Nexus.CharBuffer.__init__()=Bio.Nexus.Nexus.CharBuffer-class.html#__init__,Method Bio.Nexus.Nexus.Commandline.__init__()=Bio.Nexus.Nexus.Commandline-class.html#__init__,Method Bio.Nexus.Nexus.Nexus.__init__()=Bio.Nexus.Nexus.Nexus-class.html#__init__,Method Bio.Nexus.Nexus.StepMatrix.__init__()=Bio.Nexus.Nexus.StepMatrix-class.html#__init__,Method Bio.Nexus.Nodes.Chain.__init__()=Bio.Nexus.Nodes.Chain-class.html#__init__,Method Bio.Nexus.Nodes.Node.__init__()=Bio.Nexus.Nodes.Node-class.html#__init__,Method Bio.Nexus.StandardData.StandardData.__init__()=Bio.Nexus.StandardData.StandardData-class.html#__init__,Method Bio.Nexus.Trees.NodeData.__init__()=Bio.Nexus.Trees.NodeData-class.html#__init__,Method Bio.Nexus.Trees.Tree.__init__()=Bio.Nexus.Trees.Tree-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractPropertyMap-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap-class.html#__init__,Method Bio.PDB.Atom.Atom.__init__()=Bio.PDB.Atom.Atom-class.html#__init__,Method Bio.PDB.Atom.DisorderedAtom.__init__()=Bio.PDB.Atom.DisorderedAtom-class.html#__init__,Method Bio.PDB.Chain.Chain.__init__()=Bio.PDB.Chain.Chain-class.html#__init__,Method Bio.PDB.DSSP'.DSSP.__init__()=Bio.PDB.DSSP%27.DSSP-class.html#__init__,Method Bio.PDB.Dice.ChainSelector.__init__()=Bio.PDB.Dice.ChainSelector-class.html#__init__,Method Bio.PDB.Entity.DisorderedEntityWrapper.__init__()=Bio.PDB.Entity.DisorderedEntityWrapper-class.html#__init__,Method Bio.PDB.Entity.Entity.__init__()=Bio.PDB.Entity.Entity-class.html#__init__,Method Bio.PDB.FragmentMapper'.Fragment.__init__()=Bio.PDB.FragmentMapper%27.Fragment-class.html#__init__,Method Bio.PDB.FragmentMapper'.FragmentMapper.__init__()=Bio.PDB.FragmentMapper%27.FragmentMapper-class.html#__init__,Method Bio.PDB.HSExposure.ExposureCN.__init__()=Bio.PDB.HSExposure.ExposureCN-class.html#__init__,Method Bio.PDB.HSExposure.HSExposureCA.__init__()=Bio.PDB.HSExposure.HSExposureCA-class.html#__init__,Method Bio.PDB.HSExposure.HSExposureCB.__init__()=Bio.PDB.HSExposure.HSExposureCB-class.html#__init__,Method Bio.PDB.HSExposure._AbstractHSExposure.__init__()=Bio.PDB.HSExposure._AbstractHSExposure-class.html#__init__,Method Bio.PDB.MMCIF2Dict.MMCIF2Dict.__init__()=Bio.PDB.MMCIF2Dict.MMCIF2Dict-class.html#__init__,Method Bio.PDB.MMCIFParser'.FastMMCIFParser.__init__()=Bio.PDB.MMCIFParser%27.FastMMCIFParser-class.html#__init__,Method Bio.PDB.MMCIFParser'.MMCIFParser.__init__()=Bio.PDB.MMCIFParser%27.MMCIFParser-class.html#__init__,Method Bio.PDB.Model.Model.__init__()=Bio.PDB.Model.Model-class.html#__init__,Method Bio.PDB.NACCESS.NACCESS.__init__()=Bio.PDB.NACCESS.NACCESS-class.html#__init__,Method Bio.PDB.NACCESS.NACCESS_atomic.__init__()=Bio.PDB.NACCESS.NACCESS_atomic-class.html#__init__,Method Bio.PDB.NeighborSearch'.NeighborSearch.__init__()=Bio.PDB.NeighborSearch%27.NeighborSearch-class.html#__init__,Method Bio.PDB.PDBIO'.PDBIO.__init__()=Bio.PDB.PDBIO%27.PDBIO-class.html#__init__,Method Bio.PDB.PDBList'.PDBList.__init__()=Bio.PDB.PDBList%27.PDBList-class.html#__init__,Method Bio.PDB.PDBParser'.PDBParser.__init__()=Bio.PDB.PDBParser%27.PDBParser-class.html#__init__,Method Bio.PDB.PSEA.PSEA.__init__()=Bio.PDB.PSEA.PSEA-class.html#__init__,Method Bio.PDB.Polypeptide.CaPPBuilder.__init__()=Bio.PDB.Polypeptide.CaPPBuilder-class.html#__init__,Method Bio.PDB.Polypeptide.PPBuilder.__init__()=Bio.PDB.Polypeptide.PPBuilder-class.html#__init__,Method Bio.PDB.Polypeptide._PPBuilder.__init__()=Bio.PDB.Polypeptide._PPBuilder-class.html#__init__,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.__init__()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#__init__,Method Bio.PDB.Residue.DisorderedResidue.__init__()=Bio.PDB.Residue.DisorderedResidue-class.html#__init__,Method Bio.PDB.Residue.Residue.__init__()=Bio.PDB.Residue.Residue-class.html#__init__,Method Bio.PDB.ResidueDepth'.ResidueDepth.__init__()=Bio.PDB.ResidueDepth%27.ResidueDepth-class.html#__init__,Method Bio.PDB.Structure.Structure.__init__()=Bio.PDB.Structure.Structure-class.html#__init__,Method Bio.PDB.StructureAlignment'.StructureAlignment.__init__()=Bio.PDB.StructureAlignment%27.StructureAlignment-class.html#__init__,Method Bio.PDB.StructureBuilder.StructureBuilder.__init__()=Bio.PDB.StructureBuilder.StructureBuilder-class.html#__init__,Method Bio.PDB.Superimposer'.Superimposer.__init__()=Bio.PDB.Superimposer%27.Superimposer-class.html#__init__,Method Bio.PDB.mmcifio.MMCIFIO.__init__()=Bio.PDB.mmcifio.MMCIFIO-class.html#__init__,Method Bio.PDB.mmtf.DefaultParser.StructureDecoder.__init__()=Bio.PDB.mmtf.DefaultParser.StructureDecoder-class.html#__init__,Method Bio.PDB.vectors.Vector.__init__()=Bio.PDB.vectors.Vector-class.html#__init__,Method Bio.Pathway.Network.__init__()=Bio.Pathway.Network-class.html#__init__,Method Bio.Pathway.Reaction.__init__()=Bio.Pathway.Reaction-class.html#__init__,Method Bio.Pathway.Rep.Graph.Graph.__init__()=Bio.Pathway.Rep.Graph.Graph-class.html#__init__,Method Bio.Pathway.Rep.MultiGraph.MultiGraph.__init__()=Bio.Pathway.Rep.MultiGraph.MultiGraph-class.html#__init__,Method Bio.Pathway.System.__init__()=Bio.Pathway.System-class.html#__init__,Method Bio.Phylo.Applications._Fasttree.FastTreeCommandline.__init__()=Bio.Phylo.Applications._Fasttree.FastTreeCommandline-class.html#__init__,Method Bio.Phylo.Applications._Phyml.PhymlCommandline.__init__()=Bio.Phylo.Applications._Phyml.PhymlCommandline-class.html#__init__,Method Bio.Phylo.Applications._Raxml.RaxmlCommandline.__init__()=Bio.Phylo.Applications._Raxml.RaxmlCommandline-class.html#__init__,Method Bio.Phylo.BaseTree.BranchColor.__init__()=Bio.Phylo.BaseTree.BranchColor-class.html#__init__,Method Bio.Phylo.BaseTree.Clade.__init__()=Bio.Phylo.BaseTree.Clade-class.html#__init__,Method Bio.Phylo.BaseTree.Tree.__init__()=Bio.Phylo.BaseTree.Tree-class.html#__init__,Method Bio.Phylo.CDAO.Clade.__init__()=Bio.Phylo.CDAO.Clade-class.html#__init__,Method Bio.Phylo.CDAO.Tree.__init__()=Bio.Phylo.CDAO.Tree-class.html#__init__,Method Bio.Phylo.CDAOIO.Parser.__init__()=Bio.Phylo.CDAOIO.Parser-class.html#__init__,Method Bio.Phylo.CDAOIO.Writer.__init__()=Bio.Phylo.CDAOIO.Writer-class.html#__init__,Method Bio.Phylo.NeXML.Clade.__init__()=Bio.Phylo.NeXML.Clade-class.html#__init__,Method Bio.Phylo.NeXML.Tree.__init__()=Bio.Phylo.NeXML.Tree-class.html#__init__,Method Bio.Phylo.NeXMLIO.Parser.__init__()=Bio.Phylo.NeXMLIO.Parser-class.html#__init__,Method Bio.Phylo.NeXMLIO.Writer.__init__()=Bio.Phylo.NeXMLIO.Writer-class.html#__init__,Method Bio.Phylo.Newick.Clade.__init__()=Bio.Phylo.Newick.Clade-class.html#__init__,Method Bio.Phylo.Newick.Tree.__init__()=Bio.Phylo.Newick.Tree-class.html#__init__,Method Bio.Phylo.NewickIO.Parser.__init__()=Bio.Phylo.NewickIO.Parser-class.html#__init__,Method Bio.Phylo.NewickIO.Writer.__init__()=Bio.Phylo.NewickIO.Writer-class.html#__init__,Method Bio.Phylo.PAML._paml.Paml.__init__()=Bio.Phylo.PAML._paml.Paml-class.html#__init__,Method Bio.Phylo.PAML.baseml.Baseml.__init__()=Bio.Phylo.PAML.baseml.Baseml-class.html#__init__,Method Bio.Phylo.PAML.codeml.Codeml.__init__()=Bio.Phylo.PAML.codeml.Codeml-class.html#__init__,Method Bio.Phylo.PAML.yn00.Yn00.__init__()=Bio.Phylo.PAML.yn00.Yn00-class.html#__init__,Method Bio.Phylo.PhyloXML.Accession.__init__()=Bio.Phylo.PhyloXML.Accession-class.html#__init__,Method Bio.Phylo.PhyloXML.Annotation.__init__()=Bio.Phylo.PhyloXML.Annotation-class.html#__init__,Method Bio.Phylo.PhyloXML.BinaryCharacters.__init__()=Bio.Phylo.PhyloXML.BinaryCharacters-class.html#__init__,Method Bio.Phylo.PhyloXML.BranchColor.__init__()=Bio.Phylo.PhyloXML.BranchColor-class.html#__init__,Method Bio.Phylo.PhyloXML.Clade.__init__()=Bio.Phylo.PhyloXML.Clade-class.html#__init__,Method Bio.Phylo.PhyloXML.CladeRelation.__init__()=Bio.Phylo.PhyloXML.CladeRelation-class.html#__init__,Method Bio.Phylo.PhyloXML.Confidence.__init__()=Bio.Phylo.PhyloXML.Confidence-class.html#__init__,Method Bio.Phylo.PhyloXML.Date.__init__()=Bio.Phylo.PhyloXML.Date-class.html#__init__,Method Bio.Phylo.PhyloXML.Distribution.__init__()=Bio.Phylo.PhyloXML.Distribution-class.html#__init__,Method Bio.Phylo.PhyloXML.DomainArchitecture.__init__()=Bio.Phylo.PhyloXML.DomainArchitecture-class.html#__init__,Method Bio.Phylo.PhyloXML.Events.__init__()=Bio.Phylo.PhyloXML.Events-class.html#__init__,Method Bio.Phylo.PhyloXML.Id.__init__()=Bio.Phylo.PhyloXML.Id-class.html#__init__,Method Bio.Phylo.PhyloXML.MolSeq.__init__()=Bio.Phylo.PhyloXML.MolSeq-class.html#__init__,Method Bio.Phylo.PhyloXML.Other.__init__()=Bio.Phylo.PhyloXML.Other-class.html#__init__,Method Bio.Phylo.PhyloXML.Phylogeny.__init__()=Bio.Phylo.PhyloXML.Phylogeny-class.html#__init__,Method Bio.Phylo.PhyloXML.Phyloxml.__init__()=Bio.Phylo.PhyloXML.Phyloxml-class.html#__init__,Method Bio.Phylo.PhyloXML.Point.__init__()=Bio.Phylo.PhyloXML.Point-class.html#__init__,Method Bio.Phylo.PhyloXML.Polygon.__init__()=Bio.Phylo.PhyloXML.Polygon-class.html#__init__,Method Bio.Phylo.PhyloXML.Property.__init__()=Bio.Phylo.PhyloXML.Property-class.html#__init__,Method Bio.Phylo.PhyloXML.ProteinDomain.__init__()=Bio.Phylo.PhyloXML.ProteinDomain-class.html#__init__,Method Bio.Phylo.PhyloXML.Reference.__init__()=Bio.Phylo.PhyloXML.Reference-class.html#__init__,Method Bio.Phylo.PhyloXML.Sequence.__init__()=Bio.Phylo.PhyloXML.Sequence-class.html#__init__,Method Bio.Phylo.PhyloXML.SequenceRelation.__init__()=Bio.Phylo.PhyloXML.SequenceRelation-class.html#__init__,Method Bio.Phylo.PhyloXML.Taxonomy.__init__()=Bio.Phylo.PhyloXML.Taxonomy-class.html#__init__,Method Bio.Phylo.PhyloXML.Uri.__init__()=Bio.Phylo.PhyloXML.Uri-class.html#__init__,Method Bio.Phylo.PhyloXMLIO.Parser.__init__()=Bio.Phylo.PhyloXMLIO.Parser-class.html#__init__,Method Bio.Phylo.PhyloXMLIO.Writer.__init__()=Bio.Phylo.PhyloXMLIO.Writer-class.html#__init__,Method Bio.Phylo.TreeConstruction.DistanceCalculator.__init__()=Bio.Phylo.TreeConstruction.DistanceCalculator-class.html#__init__,Method Bio.Phylo.TreeConstruction.DistanceMatrix.__init__()=Bio.Phylo.TreeConstruction.DistanceMatrix-class.html#__init__,Method Bio.Phylo.TreeConstruction.DistanceTreeConstructor.__init__()=Bio.Phylo.TreeConstruction.DistanceTreeConstructor-class.html#__init__,Method Bio.Phylo.TreeConstruction.NNITreeSearcher.__init__()=Bio.Phylo.TreeConstruction.NNITreeSearcher-class.html#__init__,Method Bio.Phylo.TreeConstruction.ParsimonyScorer.__init__()=Bio.Phylo.TreeConstruction.ParsimonyScorer-class.html#__init__,Method Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor.__init__()=Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor-class.html#__init__,Method Bio.Phylo.TreeConstruction._Matrix.__init__()=Bio.Phylo.TreeConstruction._Matrix-class.html#__init__,Method Bio.PopGen.GenePop.Controller.GenePopController.__init__()=Bio.PopGen.GenePop.Controller.GenePopController-class.html#__init__,Method Bio.PopGen.GenePop.Controller._FileIterator.__init__()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#__init__,Method Bio.PopGen.GenePop.Controller._GenePopCommandline.__init__()=Bio.PopGen.GenePop.Controller._GenePopCommandline-class.html#__init__,Method Bio.PopGen.GenePop.EasyController.EasyController.__init__()=Bio.PopGen.GenePop.EasyController.EasyController-class.html#__init__,Method Bio.PopGen.GenePop.FileParser.FileRecord.__init__()=Bio.PopGen.GenePop.FileParser.FileRecord-class.html#__init__,Method Bio.PopGen.GenePop.LargeFileParser.Record.__init__()=Bio.PopGen.GenePop.LargeFileParser.Record-class.html#__init__,Method Bio.PopGen.GenePop.Record.__init__()=Bio.PopGen.GenePop.Record-class.html#__init__,Method Bio.Restriction.PrintFormat.PrintFormat.__init__()=Bio.Restriction.PrintFormat.PrintFormat-class.html#__init__,Method Bio.Restriction.Restriction.Analysis.__init__()=Bio.Restriction.Restriction.Analysis-class.html#__init__,Method Bio.Restriction.Restriction.FormattedSeq.__init__()=Bio.Restriction.Restriction.FormattedSeq-class.html#__init__,Method Bio.Restriction.Restriction.RestrictionBatch.__init__()=Bio.Restriction.Restriction.RestrictionBatch-class.html#__init__,Method Bio.Restriction.Restriction.RestrictionType.__init__()=Bio.Restriction.Restriction.RestrictionType-class.html#__init__,Method Bio.SCOP.Astral.__init__()=Bio.SCOP.Astral-class.html#__init__,Method Bio.SCOP.Cla.Index.__init__()=Bio.SCOP.Cla.Index-class.html#__init__,Method Bio.SCOP.Cla.Record.__init__()=Bio.SCOP.Cla.Record-class.html#__init__,Method Bio.SCOP.Des.Record.__init__()=Bio.SCOP.Des.Record-class.html#__init__,Method Bio.SCOP.Dom.Record.__init__()=Bio.SCOP.Dom.Record-class.html#__init__,Method Bio.SCOP.Domain.__init__()=Bio.SCOP.Domain-class.html#__init__,Method Bio.SCOP.Hie.Record.__init__()=Bio.SCOP.Hie.Record-class.html#__init__,Method Bio.SCOP.Node.__init__()=Bio.SCOP.Node-class.html#__init__,Method Bio.SCOP.Raf.Res.__init__()=Bio.SCOP.Raf.Res-class.html#__init__,Method Bio.SCOP.Raf.SeqMap.__init__()=Bio.SCOP.Raf.SeqMap-class.html#__init__,Method Bio.SCOP.Raf.SeqMapIndex.__init__()=Bio.SCOP.Raf.SeqMapIndex-class.html#__init__,Method Bio.SCOP.Residues.Residues.__init__()=Bio.SCOP.Residues.Residues-class.html#__init__,Method Bio.SCOP.Scop.__init__()=Bio.SCOP.Scop-class.html#__init__,Method Bio.SVDSuperimposer.SVDSuperimposer.__init__()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabParser.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_text.BlastTextParser.__init__()=Bio.SearchIO.BlastIO.blast_text.BlastTextParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator.__init__()=Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslIndexer.__init__()=Bio.SearchIO.BlatIO.BlatPslIndexer-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslParser.__init__()=Bio.SearchIO.BlatIO.BlatPslParser-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslWriter.__init__()=Bio.SearchIO.BlatIO.BlatPslWriter-class.html#__init__,Method Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.__init__()=Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html#__init__,Method Bio.SearchIO.FastaIO.FastaM10Indexer.__init__()=Bio.SearchIO.FastaIO.FastaM10Indexer-class.html#__init__,Method Bio.SearchIO.FastaIO.FastaM10Parser.__init__()=Bio.SearchIO.FastaIO.FastaM10Parser-class.html#__init__,Method Bio.SearchIO.HHsuiteIO.hhsuite2_text.Hhsuite2TextParser.__init__()=Bio.SearchIO.HHsuiteIO.hhsuite2_text.Hhsuite2TextParser-class.html#__init__,Method Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer.__init__()=Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser.__init__()=Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.__init__()=Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser.__init__()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.__init__()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser.__init__()=Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#__init__,Method Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser.__init__()=Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser-class.html#__init__,Method Bio.SearchIO._index.SearchIndexer.__init__()=Bio.SearchIO._index.SearchIndexer-class.html#__init__,Method Bio.SearchIO._legacy.NCBIStandalone.BlastErrorParser.__init__()=Bio.SearchIO._legacy.NCBIStandalone.BlastErrorParser-class.html#__init__,Method Bio.SearchIO._legacy.NCBIStandalone.BlastParser.__init__()=Bio.SearchIO._legacy.NCBIStandalone.BlastParser-class.html#__init__,Method Bio.SearchIO._legacy.NCBIStandalone.Iterator.__init__()=Bio.SearchIO._legacy.NCBIStandalone.Iterator-class.html#__init__,Method Bio.SearchIO._legacy.NCBIStandalone.PSIBlastParser.__init__()=Bio.SearchIO._legacy.NCBIStandalone.PSIBlastParser-class.html#__init__,Method Bio.SearchIO._legacy.NCBIStandalone._BlastConsumer.__init__()=Bio.SearchIO._legacy.NCBIStandalone._BlastConsumer-class.html#__init__,Method Bio.SearchIO._legacy.NCBIStandalone._BlastErrorConsumer.__init__()=Bio.SearchIO._legacy.NCBIStandalone._BlastErrorConsumer-class.html#__init__,Method Bio.SearchIO._legacy.NCBIStandalone._PSIBlastConsumer.__init__()=Bio.SearchIO._legacy.NCBIStandalone._PSIBlastConsumer-class.html#__init__,Method Bio.SearchIO._legacy.NCBIStandalone._Scanner.__init__()=Bio.SearchIO._legacy.NCBIStandalone._Scanner-class.html#__init__,Method Bio.SearchIO._legacy.ParserSupport.TaggingConsumer.__init__()=Bio.SearchIO._legacy.ParserSupport.TaggingConsumer-class.html#__init__,Method Bio.SearchIO._model.hit.Hit.__init__()=Bio.SearchIO._model.hit.Hit-class.html#__init__,Method Bio.SearchIO._model.hsp.HSP.__init__()=Bio.SearchIO._model.hsp.HSP-class.html#__init__,Method Bio.SearchIO._model.hsp.HSPFragment.__init__()=Bio.SearchIO._model.hsp.HSPFragment-class.html#__init__,Method Bio.SearchIO._model.query.QueryResult.__init__()=Bio.SearchIO._model.query.QueryResult-class.html#__init__,Method Bio.Seq.MutableSeq.__init__()=Bio.Seq.MutableSeq-class.html#__init__,Method Bio.Seq.Seq.__init__()=Bio.Seq.Seq-class.html#__init__,Method Bio.Seq.UnknownSeq.__init__()=Bio.Seq.UnknownSeq-class.html#__init__,Method Bio.SeqFeature.CompoundLocation.__init__()=Bio.SeqFeature.CompoundLocation-class.html#__init__,Method Bio.SeqFeature.FeatureLocation.__init__()=Bio.SeqFeature.FeatureLocation-class.html#__init__,Method Bio.SeqFeature.PositionGap.__init__()=Bio.SeqFeature.PositionGap-class.html#__init__,Method Bio.SeqFeature.Reference.__init__()=Bio.SeqFeature.Reference-class.html#__init__,Method Bio.SeqFeature.SeqFeature.__init__()=Bio.SeqFeature.SeqFeature-class.html#__init__,Method Bio.SeqIO.FastaIO.FastaTwoLineWriter.__init__()=Bio.SeqIO.FastaIO.FastaTwoLineWriter-class.html#__init__,Method Bio.SeqIO.FastaIO.FastaWriter.__init__()=Bio.SeqIO.FastaIO.FastaWriter-class.html#__init__,Method Bio.SeqIO.Interfaces.SequenceIterator.__init__()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#__init__,Method Bio.SeqIO.Interfaces.SequenceWriter.__init__()=Bio.SeqIO.Interfaces.SequenceWriter-class.html#__init__,Method Bio.SeqIO.Interfaces.SequentialSequenceWriter.__init__()=Bio.SeqIO.Interfaces.SequentialSequenceWriter-class.html#__init__,Method Bio.SeqIO.NibIO.NibWriter.__init__()=Bio.SeqIO.NibIO.NibWriter-class.html#__init__,Method Bio.SeqIO.PhdIO.PhdWriter.__init__()=Bio.SeqIO.PhdIO.PhdWriter-class.html#__init__,Method Bio.SeqIO.PirIO.PirWriter.__init__()=Bio.SeqIO.PirIO.PirWriter-class.html#__init__,Method Bio.SeqIO.QualityIO.QualPhredWriter.__init__()=Bio.SeqIO.QualityIO.QualPhredWriter-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.SeqXmlIterator.__init__()=Bio.SeqIO.SeqXmlIO.SeqXmlIterator-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.SeqXmlWriter.__init__()=Bio.SeqIO.SeqXmlIO.SeqXmlWriter-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.XMLRecordIterator.__init__()=Bio.SeqIO.SeqXmlIO.XMLRecordIterator-class.html#__init__,Method Bio.SeqIO.SffIO.SffWriter.__init__()=Bio.SeqIO.SffIO.SffWriter-class.html#__init__,Method Bio.SeqIO.SffIO._AddTellHandle.__init__()=Bio.SeqIO.SffIO._AddTellHandle-class.html#__init__,Method Bio.SeqIO.UniprotIO.Parser.__init__()=Bio.SeqIO.UniprotIO.Parser-class.html#__init__,Method Bio.SeqIO._index.IntelliGeneticsRandomAccess.__init__()=Bio.SeqIO._index.IntelliGeneticsRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SeqFileRandomAccess.__init__()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SequentialSeqFileRandomAccess.__init__()=Bio.SeqIO._index.SequentialSeqFileRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SffRandomAccess.__init__()=Bio.SeqIO._index.SffRandomAccess-class.html#__init__,Method Bio.SeqRecord.SeqRecord.__init__()=Bio.SeqRecord.SeqRecord-class.html#__init__,Method Bio.SeqRecord._RestrictedDict.__init__()=Bio.SeqRecord._RestrictedDict-class.html#__init__,Method Bio.SeqUtils.CodonUsage.CodonAdaptationIndex.__init__()=Bio.SeqUtils.CodonUsage.CodonAdaptationIndex-class.html#__init__,Method Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.__init__()=Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint-class.html#__init__,Method Bio.SeqUtils.ProtParam.ProteinAnalysis.__init__()=Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#__init__,Method Bio.Sequencing.Ace.ACEFileRecord.__init__()=Bio.Sequencing.Ace.ACEFileRecord-class.html#__init__,Method Bio.Sequencing.Ace.Contig.__init__()=Bio.Sequencing.Ace.Contig-class.html#__init__,Method Bio.Sequencing.Ace.Reads.__init__()=Bio.Sequencing.Ace.Reads-class.html#__init__,Method Bio.Sequencing.Ace.af.__init__()=Bio.Sequencing.Ace.af-class.html#__init__,Method Bio.Sequencing.Ace.bs.__init__()=Bio.Sequencing.Ace.bs-class.html#__init__,Method Bio.Sequencing.Ace.ct.__init__()=Bio.Sequencing.Ace.ct-class.html#__init__,Method Bio.Sequencing.Ace.ds.__init__()=Bio.Sequencing.Ace.ds-class.html#__init__,Method Bio.Sequencing.Ace.qa.__init__()=Bio.Sequencing.Ace.qa-class.html#__init__,Method Bio.Sequencing.Ace.rd.__init__()=Bio.Sequencing.Ace.rd-class.html#__init__,Method Bio.Sequencing.Ace.rt.__init__()=Bio.Sequencing.Ace.rt-class.html#__init__,Method Bio.Sequencing.Ace.wa.__init__()=Bio.Sequencing.Ace.wa-class.html#__init__,Method Bio.Sequencing.Ace.wr.__init__()=Bio.Sequencing.Ace.wr-class.html#__init__,Method Bio.Sequencing.Applications._Novoalign.NovoalignCommandline.__init__()=Bio.Sequencing.Applications._Novoalign.NovoalignCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaAlignCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaAlignCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaBwaswCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaBwaswCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaIndexCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaIndexCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaMemCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaMemCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaSampeCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaSampeCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaSamseCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaSamseCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsVersion0xSortCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsVersion0xSortCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsVersion1xSortCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsVersion1xSortCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline-class.html#__init__,Method Bio.Sequencing.Phd.Record.__init__()=Bio.Sequencing.Phd.Record-class.html#__init__,Method Bio.SubsMat.FreqTable.FreqTable.__init__()=Bio.SubsMat.FreqTable.FreqTable-class.html#__init__,Method Bio.SubsMat.SeqMat.__init__()=Bio.SubsMat.SeqMat-class.html#__init__,Method Bio.SwissProt.KeyWList.Record.__init__()=Bio.SwissProt.KeyWList.Record-class.html#__init__,Method Bio.SwissProt.Record.__init__()=Bio.SwissProt.Record-class.html#__init__,Method Bio.SwissProt.Reference.__init__()=Bio.SwissProt.Reference-class.html#__init__,Method Bio.UniGene.ProtsimLine.__init__()=Bio.UniGene.ProtsimLine-class.html#__init__,Method Bio.UniGene.Record.__init__()=Bio.UniGene.Record-class.html#__init__,Method Bio.UniGene.STSLine.__init__()=Bio.UniGene.STSLine-class.html#__init__,Method Bio.UniGene.SequenceLine.__init__()=Bio.UniGene.SequenceLine-class.html#__init__,Method Bio.Wise.dnal.Statistics.__init__()=Bio.Wise.dnal.Statistics-class.html#__init__,Method Bio.Wise.psw.AlignmentColumn.__init__()=Bio.Wise.psw.AlignmentColumn-class.html#__init__,Method Bio.Wise.psw.ColumnUnit.__init__()=Bio.Wise.psw.ColumnUnit-class.html#__init__,Method Bio.bgzf.BgzfReader.__init__()=Bio.bgzf.BgzfReader-class.html#__init__,Method Bio.bgzf.BgzfWriter.__init__()=Bio.bgzf.BgzfWriter-class.html#__init__,Method Bio.codonalign.codonalignment.CodonAlignment.__init__()=Bio.codonalign.codonalignment.CodonAlignment-class.html#__init__,Method Bio.codonalign.codonseq.CodonSeq.__init__()=Bio.codonalign.codonseq.CodonSeq-class.html#__init__,Method Bio.kNN.kNN.__init__()=Bio.kNN.kNN-class.html#__init__,Method Bio.motifs.Instances.__init__()=Bio.motifs.Instances-class.html#__init__,Method Bio.motifs.Motif.__init__()=Bio.motifs.Motif-class.html#__init__,Method Bio.motifs.alignace.Record.__init__()=Bio.motifs.alignace.Record-class.html#__init__,Method Bio.motifs.applications._xxmotif.XXmotifCommandline.__init__()=Bio.motifs.applications._xxmotif.XXmotifCommandline-class.html#__init__,Method Bio.motifs.jaspar.Motif.__init__()=Bio.motifs.jaspar.Motif-class.html#__init__,Method Bio.motifs.jaspar.Record.__init__()=Bio.motifs.jaspar.Record-class.html#__init__,Method Bio.motifs.jaspar.db.JASPAR5.__init__()=Bio.motifs.jaspar.db.JASPAR5-class.html#__init__,Method Bio.motifs.mast.Record.__init__()=Bio.motifs.mast.Record-class.html#__init__,Method Bio.motifs.matrix.GenericPositionMatrix.__init__()=Bio.motifs.matrix.GenericPositionMatrix-class.html#__init__,Method Bio.motifs.matrix.PositionWeightMatrix.__init__()=Bio.motifs.matrix.PositionWeightMatrix-class.html#__init__,Method Bio.motifs.meme.Instance.__init__()=Bio.motifs.meme.Instance-class.html#__init__,Method Bio.motifs.meme.Motif.__init__()=Bio.motifs.meme.Motif-class.html#__init__,Method Bio.motifs.meme.Record.__init__()=Bio.motifs.meme.Record-class.html#__init__,Method Bio.motifs.minimal.Record.__init__()=Bio.motifs.minimal.Record-class.html#__init__,Method Bio.motifs.thresholds.ScoreDistribution.__init__()=Bio.motifs.thresholds.ScoreDistribution-class.html#__init__,Method Bio.motifs.transfac.Record.__init__()=Bio.motifs.transfac.Record-class.html#__init__,Method Bio.motifs.xms.XMSScanner.__init__()=Bio.motifs.xms.XMSScanner-class.html#__init__,Method Bio.pairwise2.affine_penalty.__init__()=Bio.pairwise2.affine_penalty-class.html#__init__,Method Bio.pairwise2.dictionary_match.__init__()=Bio.pairwise2.dictionary_match-class.html#__init__,Method Bio.pairwise2.identity_match.__init__()=Bio.pairwise2.identity_match-class.html#__init__,Method Bio.phenotype.phen_micro.JsonWriter.__init__()=Bio.phenotype.phen_micro.JsonWriter-class.html#__init__,Method Bio.phenotype.phen_micro.PlateRecord.__init__()=Bio.phenotype.phen_micro.PlateRecord-class.html#__init__,Method Bio.phenotype.phen_micro.WellRecord.__init__()=Bio.phenotype.phen_micro.WellRecord-class.html#__init__,Method BioSQL.BioSeq.DBSeq.__init__()=BioSQL.BioSeq.DBSeq-class.html#__init__,Method BioSQL.BioSeq.DBSeqRecord.__init__()=BioSQL.BioSeq.DBSeqRecord-class.html#__init__,Method BioSQL.BioSeqDatabase.Adaptor.__init__()=BioSQL.BioSeqDatabase.Adaptor-class.html#__init__,Method BioSQL.BioSeqDatabase.BioSeqDatabase.__init__()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#__init__,Method BioSQL.BioSeqDatabase.DBServer.__init__()=BioSQL.BioSeqDatabase.DBServer-class.html#__init__,Method BioSQL.BioSeqDatabase._CursorWrapper.__init__()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#__init__,Method BioSQL.DBUtils.Generic_dbutils.__init__()=BioSQL.DBUtils.Generic_dbutils-class.html#__init__,Method BioSQL.Loader.DatabaseLoader.__init__()=BioSQL.Loader.DatabaseLoader-class.html#__init__,Method BioSQL.Loader.DatabaseRemover.__init__()=BioSQL.Loader.DatabaseRemover-class.html#__init__"><a title="Bio.Affy.CelFile.ParserError.__init__
Bio.Affy.CelFile.Record.__init__
Bio.Align.AlignInfo.PSSM.__init__
Bio.Align.AlignInfo.SummaryInfo.__init__
Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline.__init__
Bio.Align.Applications._Clustalw.ClustalwCommandline.__init__
Bio.Align.Applications._Dialign.DialignCommandline.__init__
Bio.Align.Applications._MSAProbs.MSAProbsCommandline.__init__
Bio.Align.Applications._Mafft.MafftCommandline.__init__
Bio.Align.Applications._Muscle.MuscleCommandline.__init__
Bio.Align.Applications._Prank.PrankCommandline.__init__
Bio.Align.Applications._Probcons.ProbconsCommandline.__init__
Bio.Align.Applications._TCoffee.TCoffeeCommandline.__init__
Bio.Align.MultipleSeqAlignment.__init__
Bio.Align.PairwiseAlignment.__init__
Bio.Align.PairwiseAlignments.__init__
Bio.AlignIO.Interfaces.AlignmentIterator.__init__
Bio.AlignIO.Interfaces.AlignmentWriter.__init__
Bio.AlignIO.Interfaces.SequentialAlignmentWriter.__init__
Bio.AlignIO.MafIO.MafIndex.__init__
Bio.AlignIO.MauveIO.MauveWriter.__init__
Bio.Alphabet.AlphabetEncoder.__init__
Bio.Alphabet.Gapped.__init__
Bio.Alphabet.HasStopCodon.__init__
Bio.Application.AbstractCommandline.__init__
Bio.Application.ApplicationError.__init__
Bio.Application._AbstractParameter.__init__
Bio.Application._Argument.__init__
Bio.Application._Option.__init__
Bio.Application._StaticArgument.__init__
Bio.Application._Switch.__init__
Bio.Blast.Applications.NcbiblastformatterCommandline.__init__
Bio.Blast.Applications.NcbiblastnCommandline.__init__
Bio.Blast.Applications.NcbiblastpCommandline.__init__
Bio.Blast.Applications.NcbiblastxCommandline.__init__
Bio.Blast.Applications.NcbideltablastCommandline.__init__
Bio.Blast.Applications.NcbimakeblastdbCommandline.__init__
Bio.Blast.Applications.NcbipsiblastCommandline.__init__
Bio.Blast.Applications.NcbirpsblastCommandline.__init__
Bio.Blast.Applications.NcbirpstblastnCommandline.__init__
Bio.Blast.Applications.NcbitblastnCommandline.__init__
Bio.Blast.Applications.NcbitblastxCommandline.__init__
Bio.Blast.Applications._NcbibaseblastCommandline.__init__
Bio.Blast.Applications._Ncbiblast2SeqCommandline.__init__
Bio.Blast.Applications._NcbiblastCommandline.__init__
Bio.Blast.Applications._NcbiblastMain2SeqCommandline.__init__
Bio.Blast.NCBIXML.BlastParser.__init__
Bio.Blast.NCBIXML._XMLparser.__init__
Bio.Blast.ParseBlastTable.BlastTableEntry.__init__
Bio.Blast.ParseBlastTable.BlastTableReader.__init__
Bio.Blast.ParseBlastTable.BlastTableRec.__init__
Bio.Blast.Record.Alignment.__init__
Bio.Blast.Record.Blast.__init__
Bio.Blast.Record.DatabaseReport.__init__
Bio.Blast.Record.Description.__init__
Bio.Blast.Record.DescriptionExt.__init__
Bio.Blast.Record.DescriptionExtItem.__init__
Bio.Blast.Record.HSP.__init__
Bio.Blast.Record.Header.__init__
Bio.Blast.Record.MultipleAlignment.__init__
Bio.Blast.Record.PSIBlast.__init__
Bio.Blast.Record.Parameters.__init__
Bio.Blast.Record.Round.__init__
Bio.CAPS.CAPSMap.__init__
Bio.CAPS.DifferentialCutsite.__init__
Bio.Cluster.Record.__init__
Bio.Compass.Record.__init__
Bio.Crystal.Chain.__init__
Bio.Crystal.Crystal.__init__
Bio.Crystal.Hetero.__init__
Bio.Data.CodonTable.AmbiguousCodonTable.__init__
Bio.Data.CodonTable.AmbiguousForwardTable.__init__
Bio.Data.CodonTable.CodonTable.__init__
Bio.Data.CodonTable.NCBICodonTable.__init__
Bio.Emboss.Applications.DiffseqCommandline.__init__
Bio.Emboss.Applications.EInvertedCommandline.__init__
Bio.Emboss.Applications.ETandemCommandline.__init__
Bio.Emboss.Applications.Est2GenomeCommandline.__init__
Bio.Emboss.Applications.FConsenseCommandline.__init__
Bio.Emboss.Applications.FDNADistCommandline.__init__
Bio.Emboss.Applications.FDNAParsCommandline.__init__
Bio.Emboss.Applications.FNeighborCommandline.__init__
Bio.Emboss.Applications.FProtDistCommandline.__init__
Bio.Emboss.Applications.FProtParsCommandline.__init__
Bio.Emboss.Applications.FSeqBootCommandline.__init__
Bio.Emboss.Applications.FTreeDistCommandline.__init__
Bio.Emboss.Applications.FuzznucCommandline.__init__
Bio.Emboss.Applications.FuzzproCommandline.__init__
Bio.Emboss.Applications.IepCommandline.__init__
Bio.Emboss.Applications.NeedleCommandline.__init__
Bio.Emboss.Applications.NeedleallCommandline.__init__
Bio.Emboss.Applications.PalindromeCommandline.__init__
Bio.Emboss.Applications.Primer3Commandline.__init__
Bio.Emboss.Applications.PrimerSearchCommandline.__init__
Bio.Emboss.Applications.SeqmatchallCommandline.__init__
Bio.Emboss.Applications.SeqretCommandline.__init__
Bio.Emboss.Applications.StretcherCommandline.__init__
Bio.Emboss.Applications.TranalignCommandline.__init__
Bio.Emboss.Applications.WaterCommandline.__init__
Bio.Emboss.Applications._EmbossCommandLine.__init__
Bio.Emboss.Applications._EmbossMinimalCommandLine.__init__
Bio.Emboss.Primer3.Primers.__init__
Bio.Emboss.Primer3.Record.__init__
Bio.Emboss.PrimerSearch.Amplifier.__init__
Bio.Emboss.PrimerSearch.InputRecord.__init__
Bio.Emboss.PrimerSearch.OutputRecord.__init__
Bio.Entrez.Parser.CorruptedXMLError.__init__
Bio.Entrez.Parser.DataHandler.__init__
Bio.Entrez.Parser.DictionaryElement.__init__
Bio.Entrez.Parser.ListElement.__init__
Bio.Entrez.Parser.NoneElement.__init__
Bio.Entrez.Parser.NotXMLError.__init__
Bio.Entrez.Parser.ValidationError.__init__
Bio.ExPASy.Enzyme.Record.__init__
Bio.ExPASy.Prodoc.Record.__init__
Bio.ExPASy.Prodoc.Reference.__init__
Bio.ExPASy.Prosite.Record.__init__
Bio.ExPASy.ScanProsite.ContentHandler.__init__
Bio.ExPASy.ScanProsite.Parser.__init__
Bio.ExPASy.ScanProsite.Record.__init__
Bio.ExPASy.cellosaurus.Record.__init__
Bio.FSSP.FSSPAlignDict.__init__
Bio.FSSP.FSSPAlignRec.__init__
Bio.FSSP.FSSPHeader.__init__
Bio.FSSP.FSSPSumRec.__init__
Bio.FSSP.FSSPTools.FSSPMultAlign.__init__
Bio.FSSP.PosAlign.__init__
Bio.FSSP.fssp_rec.fff_rec.__init__
Bio.File.UndoHandle.__init__
Bio.File._IndexedSeqFileDict.__init__
Bio.File._SQLiteManySeqFilesDict.__init__
Bio.GenBank.FeatureParser.__init__
Bio.GenBank.Iterator.__init__
Bio.GenBank.Record.Feature.__init__
Bio.GenBank.Record.Qualifier.__init__
Bio.GenBank.Record.Record.__init__
Bio.GenBank.Record.Reference.__init__
Bio.GenBank.RecordParser.__init__
Bio.GenBank.Scanner.InsdcScanner.__init__
Bio.GenBank._BaseGenBankConsumer.__init__
Bio.GenBank._FeatureConsumer.__init__
Bio.GenBank._RecordConsumer.__init__
Bio.GenBank.utils.FeatureValueCleaner.__init__
Bio.Geo.Record.Record.__init__
Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment.__init__
Bio.Graphics.BasicChromosome.Chromosome.__init__
Bio.Graphics.BasicChromosome.ChromosomeSegment.__init__
Bio.Graphics.BasicChromosome.Organism.__init__
Bio.Graphics.BasicChromosome.TelomereSegment.__init__
Bio.Graphics.BasicChromosome._ChromosomeComponent.__init__
Bio.Graphics.ColorSpiral.ColorSpiral.__init__
Bio.Graphics.Comparative.ComparativeScatterPlot.__init__
Bio.Graphics.DisplayRepresentation.ChromosomeCounts.__init__
Bio.Graphics.Distribution.BarChartDistribution.__init__
Bio.Graphics.Distribution.DistributionPage.__init__
Bio.Graphics.Distribution.LineDistribution.__init__
Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.__init__
Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.__init__
Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.__init__
Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.__init__
Bio.Graphics.GenomeDiagram._Diagram.Diagram.__init__
Bio.Graphics.GenomeDiagram._Feature.Feature.__init__
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__init__
Bio.Graphics.GenomeDiagram._Graph.GraphData.__init__
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__init__
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.__init__
Bio.Graphics.GenomeDiagram._Track.Track.__init__
Bio.Graphics.KGML_vis.KGMLCanvas.__init__
Bio.HMM.DynamicProgramming.AbstractDPAlgorithms.__init__
Bio.HMM.DynamicProgramming.LogDPAlgorithms.__init__
Bio.HMM.DynamicProgramming.ScaledDPAlgorithms.__init__
Bio.HMM.MarkovModel.HiddenMarkovModel.__init__
Bio.HMM.MarkovModel.MarkovModelBuilder.__init__
Bio.HMM.Trainer.AbstractTrainer.__init__
Bio.HMM.Trainer.BaumWelchTrainer.__init__
Bio.HMM.Trainer.KnownStateTrainer.__init__
Bio.HMM.Trainer.TrainingSequence.__init__
Bio.Index._InMemoryIndex.__init__
Bio.Index._ShelveIndex.__init__
Bio.KDTree.KDTree'.KDTree.__init__
Bio.KEGG.Compound.Record.__init__
Bio.KEGG.Enzyme.Record.__init__
Bio.KEGG.Gene.Record.__init__
Bio.KEGG.KGML.KGML_parser.KGMLParser.__init__
Bio.KEGG.KGML.KGML_pathway.Component.__init__
Bio.KEGG.KGML.KGML_pathway.Entry.__init__
Bio.KEGG.KGML.KGML_pathway.Graphics.__init__
Bio.KEGG.KGML.KGML_pathway.Pathway.__init__
Bio.KEGG.KGML.KGML_pathway.Reaction.__init__
Bio.KEGG.KGML.KGML_pathway.Relation.__init__
Bio.LogisticRegression.LogisticRegression.__init__
Bio.MarkovModel.MarkovModel.__init__
Bio.MaxEntropy.MaxEntropy.__init__
Bio.NMR.xpktools.Peaklist.__init__
Bio.NMR.xpktools.XpkEntry.__init__
Bio.NaiveBayes.NaiveBayes.__init__
Bio.Nexus.Nexus.Block.__init__
Bio.Nexus.Nexus.CharBuffer.__init__
Bio.Nexus.Nexus.Commandline.__init__
Bio.Nexus.Nexus.Nexus.__init__
Bio.Nexus.Nexus.StepMatrix.__init__
Bio.Nexus.Nodes.Chain.__init__
Bio.Nexus.Nodes.Node.__init__
Bio.Nexus.StandardData.StandardData.__init__
Bio.Nexus.Trees.NodeData.__init__
Bio.Nexus.Trees.Tree.__init__
Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap.__init__
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__init__
Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap.__init__
Bio.PDB.Atom.Atom.__init__
Bio.PDB.Atom.DisorderedAtom.__init__
Bio.PDB.Chain.Chain.__init__
Bio.PDB.DSSP'.DSSP.__init__
Bio.PDB.Dice.ChainSelector.__init__
Bio.PDB.Entity.DisorderedEntityWrapper.__init__
Bio.PDB.Entity.Entity.__init__
Bio.PDB.FragmentMapper'.Fragment.__init__
Bio.PDB.FragmentMapper'.FragmentMapper.__init__
Bio.PDB.HSExposure.ExposureCN.__init__
Bio.PDB.HSExposure.HSExposureCA.__init__
Bio.PDB.HSExposure.HSExposureCB.__init__
Bio.PDB.HSExposure._AbstractHSExposure.__init__
Bio.PDB.MMCIF2Dict.MMCIF2Dict.__init__
Bio.PDB.MMCIFParser'.FastMMCIFParser.__init__
Bio.PDB.MMCIFParser'.MMCIFParser.__init__
Bio.PDB.Model.Model.__init__
Bio.PDB.NACCESS.NACCESS.__init__
Bio.PDB.NACCESS.NACCESS_atomic.__init__
Bio.PDB.NeighborSearch'.NeighborSearch.__init__
Bio.PDB.PDBIO'.PDBIO.__init__
Bio.PDB.PDBList'.PDBList.__init__
Bio.PDB.PDBParser'.PDBParser.__init__
Bio.PDB.PSEA.PSEA.__init__
Bio.PDB.Polypeptide.CaPPBuilder.__init__
Bio.PDB.Polypeptide.PPBuilder.__init__
Bio.PDB.Polypeptide._PPBuilder.__init__
Bio.PDB.QCPSuperimposer.QCPSuperimposer.__init__
Bio.PDB.Residue.DisorderedResidue.__init__
Bio.PDB.Residue.Residue.__init__
Bio.PDB.ResidueDepth'.ResidueDepth.__init__
Bio.PDB.Structure.Structure.__init__
Bio.PDB.StructureAlignment'.StructureAlignment.__init__
Bio.PDB.StructureBuilder.StructureBuilder.__init__
Bio.PDB.Superimposer'.Superimposer.__init__
Bio.PDB.mmcifio.MMCIFIO.__init__
Bio.PDB.mmtf.DefaultParser.StructureDecoder.__init__
Bio.PDB.vectors.Vector.__init__
Bio.Pathway.Network.__init__
Bio.Pathway.Reaction.__init__
Bio.Pathway.Rep.Graph.Graph.__init__
Bio.Pathway.Rep.MultiGraph.MultiGraph.__init__
Bio.Pathway.System.__init__
Bio.Phylo.Applications._Fasttree.FastTreeCommandline.__init__
Bio.Phylo.Applications._Phyml.PhymlCommandline.__init__
Bio.Phylo.Applications._Raxml.RaxmlCommandline.__init__
Bio.Phylo.BaseTree.BranchColor.__init__
Bio.Phylo.BaseTree.Clade.__init__
Bio.Phylo.BaseTree.Tree.__init__
Bio.Phylo.CDAO.Clade.__init__
Bio.Phylo.CDAO.Tree.__init__
Bio.Phylo.CDAOIO.Parser.__init__
Bio.Phylo.CDAOIO.Writer.__init__
Bio.Phylo.NeXML.Clade.__init__
Bio.Phylo.NeXML.Tree.__init__
Bio.Phylo.NeXMLIO.Parser.__init__
Bio.Phylo.NeXMLIO.Writer.__init__
Bio.Phylo.Newick.Clade.__init__
Bio.Phylo.Newick.Tree.__init__
Bio.Phylo.NewickIO.Parser.__init__
Bio.Phylo.NewickIO.Writer.__init__
Bio.Phylo.PAML._paml.Paml.__init__
Bio.Phylo.PAML.baseml.Baseml.__init__
Bio.Phylo.PAML.codeml.Codeml.__init__
Bio.Phylo.PAML.yn00.Yn00.__init__
Bio.Phylo.PhyloXML.Accession.__init__
Bio.Phylo.PhyloXML.Annotation.__init__
Bio.Phylo.PhyloXML.BinaryCharacters.__init__
Bio.Phylo.PhyloXML.BranchColor.__init__
Bio.Phylo.PhyloXML.Clade.__init__
Bio.Phylo.PhyloXML.CladeRelation.__init__
Bio.Phylo.PhyloXML.Confidence.__init__
Bio.Phylo.PhyloXML.Date.__init__
Bio.Phylo.PhyloXML.Distribution.__init__
Bio.Phylo.PhyloXML.DomainArchitecture.__init__
Bio.Phylo.PhyloXML.Events.__init__
Bio.Phylo.PhyloXML.Id.__init__
Bio.Phylo.PhyloXML.MolSeq.__init__
Bio.Phylo.PhyloXML.Other.__init__
Bio.Phylo.PhyloXML.Phylogeny.__init__
Bio.Phylo.PhyloXML.Phyloxml.__init__
Bio.Phylo.PhyloXML.Point.__init__
Bio.Phylo.PhyloXML.Polygon.__init__
Bio.Phylo.PhyloXML.Property.__init__
Bio.Phylo.PhyloXML.ProteinDomain.__init__
Bio.Phylo.PhyloXML.Reference.__init__
Bio.Phylo.PhyloXML.Sequence.__init__
Bio.Phylo.PhyloXML.SequenceRelation.__init__
Bio.Phylo.PhyloXML.Taxonomy.__init__
Bio.Phylo.PhyloXML.Uri.__init__
Bio.Phylo.PhyloXMLIO.Parser.__init__
Bio.Phylo.PhyloXMLIO.Writer.__init__
Bio.Phylo.TreeConstruction.DistanceCalculator.__init__
Bio.Phylo.TreeConstruction.DistanceMatrix.__init__
Bio.Phylo.TreeConstruction.DistanceTreeConstructor.__init__
Bio.Phylo.TreeConstruction.NNITreeSearcher.__init__
Bio.Phylo.TreeConstruction.ParsimonyScorer.__init__
Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor.__init__
Bio.Phylo.TreeConstruction._Matrix.__init__
Bio.PopGen.GenePop.Controller.GenePopController.__init__
Bio.PopGen.GenePop.Controller._FileIterator.__init__
Bio.PopGen.GenePop.Controller._GenePopCommandline.__init__
Bio.PopGen.GenePop.EasyController.EasyController.__init__
Bio.PopGen.GenePop.FileParser.FileRecord.__init__
Bio.PopGen.GenePop.LargeFileParser.Record.__init__
Bio.PopGen.GenePop.Record.__init__
Bio.Restriction.PrintFormat.PrintFormat.__init__
Bio.Restriction.Restriction.Analysis.__init__
Bio.Restriction.Restriction.FormattedSeq.__init__
Bio.Restriction.Restriction.RestrictionBatch.__init__
Bio.Restriction.Restriction.RestrictionType.__init__
Bio.SCOP.Astral.__init__
Bio.SCOP.Cla.Index.__init__
Bio.SCOP.Cla.Record.__init__
Bio.SCOP.Des.Record.__init__
Bio.SCOP.Dom.Record.__init__
Bio.SCOP.Domain.__init__
Bio.SCOP.Hie.Record.__init__
Bio.SCOP.Node.__init__
Bio.SCOP.Raf.Res.__init__
Bio.SCOP.Raf.SeqMap.__init__
Bio.SCOP.Raf.SeqMapIndex.__init__
Bio.SCOP.Residues.Residues.__init__
Bio.SCOP.Scop.__init__
Bio.SVDSuperimposer.SVDSuperimposer.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabParser.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.__init__
Bio.SearchIO.BlastIO.blast_text.BlastTextParser.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.__init__
Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator.__init__
Bio.SearchIO.BlatIO.BlatPslIndexer.__init__
Bio.SearchIO.BlatIO.BlatPslParser.__init__
Bio.SearchIO.BlatIO.BlatPslWriter.__init__
Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.__init__
Bio.SearchIO.FastaIO.FastaM10Indexer.__init__
Bio.SearchIO.FastaIO.FastaM10Parser.__init__
Bio.SearchIO.HHsuiteIO.hhsuite2_text.Hhsuite2TextParser.__init__
Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer.__init__
Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser.__init__
Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.__init__
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser.__init__
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.__init__
Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser.__init__
Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser.__init__
Bio.SearchIO._index.SearchIndexer.__init__
Bio.SearchIO._legacy.NCBIStandalone.BlastErrorParser.__init__
Bio.SearchIO._legacy.NCBIStandalone.BlastParser.__init__
Bio.SearchIO._legacy.NCBIStandalone.Iterator.__init__
Bio.SearchIO._legacy.NCBIStandalone.PSIBlastParser.__init__
Bio.SearchIO._legacy.NCBIStandalone._BlastConsumer.__init__
Bio.SearchIO._legacy.NCBIStandalone._BlastErrorConsumer.__init__
Bio.SearchIO._legacy.NCBIStandalone._PSIBlastConsumer.__init__
Bio.SearchIO._legacy.NCBIStandalone._Scanner.__init__
Bio.SearchIO._legacy.ParserSupport.TaggingConsumer.__init__
Bio.SearchIO._model.hit.Hit.__init__
Bio.SearchIO._model.hsp.HSP.__init__
Bio.SearchIO._model.hsp.HSPFragment.__init__
Bio.SearchIO._model.query.QueryResult.__init__
Bio.Seq.MutableSeq.__init__
Bio.Seq.Seq.__init__
Bio.Seq.UnknownSeq.__init__
Bio.SeqFeature.CompoundLocation.__init__
Bio.SeqFeature.FeatureLocation.__init__
Bio.SeqFeature.PositionGap.__init__
Bio.SeqFeature.Reference.__init__
Bio.SeqFeature.SeqFeature.__init__
Bio.SeqIO.FastaIO.FastaTwoLineWriter.__init__
Bio.SeqIO.FastaIO.FastaWriter.__init__
Bio.SeqIO.Interfaces.SequenceIterator.__init__
Bio.SeqIO.Interfaces.SequenceWriter.__init__
Bio.SeqIO.Interfaces.SequentialSequenceWriter.__init__
Bio.SeqIO.NibIO.NibWriter.__init__
Bio.SeqIO.PhdIO.PhdWriter.__init__
Bio.SeqIO.PirIO.PirWriter.__init__
Bio.SeqIO.QualityIO.QualPhredWriter.__init__
Bio.SeqIO.SeqXmlIO.SeqXmlIterator.__init__
Bio.SeqIO.SeqXmlIO.SeqXmlWriter.__init__
Bio.SeqIO.SeqXmlIO.XMLRecordIterator.__init__
Bio.SeqIO.SffIO.SffWriter.__init__
Bio.SeqIO.SffIO._AddTellHandle.__init__
Bio.SeqIO.UniprotIO.Parser.__init__
Bio.SeqIO._index.IntelliGeneticsRandomAccess.__init__
Bio.SeqIO._index.SeqFileRandomAccess.__init__
Bio.SeqIO._index.SequentialSeqFileRandomAccess.__init__
Bio.SeqIO._index.SffRandomAccess.__init__
Bio.SeqRecord.SeqRecord.__init__
Bio.SeqRecord._RestrictedDict.__init__
Bio.SeqUtils.CodonUsage.CodonAdaptationIndex.__init__
Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.__init__
Bio.SeqUtils.ProtParam.ProteinAnalysis.__init__
Bio.Sequencing.Ace.ACEFileRecord.__init__
Bio.Sequencing.Ace.Contig.__init__
Bio.Sequencing.Ace.Reads.__init__
Bio.Sequencing.Ace.af.__init__
Bio.Sequencing.Ace.bs.__init__
Bio.Sequencing.Ace.ct.__init__
Bio.Sequencing.Ace.ds.__init__
Bio.Sequencing.Ace.qa.__init__
Bio.Sequencing.Ace.rd.__init__
Bio.Sequencing.Ace.rt.__init__
Bio.Sequencing.Ace.wa.__init__
Bio.Sequencing.Ace.wr.__init__
Bio.Sequencing.Applications._Novoalign.NovoalignCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaAlignCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaBwaswCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaIndexCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaMemCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaSampeCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaSamseCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsVersion0xSortCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsVersion1xSortCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline.__init__
Bio.Sequencing.Phd.Record.__init__
Bio.SubsMat.FreqTable.FreqTable.__init__
Bio.SubsMat.SeqMat.__init__
Bio.SwissProt.KeyWList.Record.__init__
Bio.SwissProt.Record.__init__
Bio.SwissProt.Reference.__init__
Bio.UniGene.ProtsimLine.__init__
Bio.UniGene.Record.__init__
Bio.UniGene.STSLine.__init__
Bio.UniGene.SequenceLine.__init__
Bio.Wise.dnal.Statistics.__init__
Bio.Wise.psw.AlignmentColumn.__init__
Bio.Wise.psw.ColumnUnit.__init__
Bio.bgzf.BgzfReader.__init__
Bio.bgzf.BgzfWriter.__init__
Bio.codonalign.codonalignment.CodonAlignment.__init__
Bio.codonalign.codonseq.CodonSeq.__init__
Bio.kNN.kNN.__init__
Bio.motifs.Instances.__init__
Bio.motifs.Motif.__init__
Bio.motifs.alignace.Record.__init__
Bio.motifs.applications._xxmotif.XXmotifCommandline.__init__
Bio.motifs.jaspar.Motif.__init__
Bio.motifs.jaspar.Record.__init__
Bio.motifs.jaspar.db.JASPAR5.__init__
Bio.motifs.mast.Record.__init__
Bio.motifs.matrix.GenericPositionMatrix.__init__
Bio.motifs.matrix.PositionWeightMatrix.__init__
Bio.motifs.meme.Instance.__init__
Bio.motifs.meme.Motif.__init__
Bio.motifs.meme.Record.__init__
Bio.motifs.minimal.Record.__init__
Bio.motifs.thresholds.ScoreDistribution.__init__
Bio.motifs.transfac.Record.__init__
Bio.motifs.xms.XMSScanner.__init__
Bio.pairwise2.affine_penalty.__init__
Bio.pairwise2.dictionary_match.__init__
Bio.pairwise2.identity_match.__init__
Bio.phenotype.phen_micro.JsonWriter.__init__
Bio.phenotype.phen_micro.PlateRecord.__init__
Bio.phenotype.phen_micro.WellRecord.__init__
BioSQL.BioSeq.DBSeq.__init__
BioSQL.BioSeq.DBSeqRecord.__init__
BioSQL.BioSeqDatabase.Adaptor.__init__
BioSQL.BioSeqDatabase.BioSeqDatabase.__init__
BioSQL.BioSeqDatabase.DBServer.__init__
BioSQL.BioSeqDatabase._CursorWrapper.__init__
BioSQL.DBUtils.Generic_dbutils.__init__
BioSQL.Loader.DatabaseLoader.__init__
BioSQL.Loader.DatabaseRemover.__init__" class="py-name" href="#" onclick="return doclink('link-235', '__init__', 'link-235');">__init__</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-name">depth_dict</tt><tt class="py-op">,</tt> <tt class="py-name">depth_keys</tt><tt class="py-op">,</tt> <tt class="py-name">depth_list</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L566"></a><tt class="py-lineno">566</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Tue Jul 16 17:06:19 2019
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
