<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.AlignIO.MafIO.MafIndex</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.AlignIO-module.html">Package&nbsp;AlignIO</a> ::
        <a href="Bio.AlignIO.MafIO-module.html">Module&nbsp;MafIO</a> ::
        Class&nbsp;MafIndex
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.AlignIO.MafIO.MafIndex-class.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== CLASS DESCRIPTION ==================== -->
<h1 class="epydoc">Class MafIndex</h1><p class="nomargin-top"><span class="codelink"><a href="Bio.AlignIO.MafIO-pysrc.html#MafIndex">source&nbsp;code</a></span></p>
<pre class="base-tree">
object --+
         |
        <strong class="uidshort">MafIndex</strong>
</pre>

<hr />
<p>Index for a MAF file.</p>
<p>The index is a sqlite3 database that is built upon creation of the object
if necessary, and queried when methods <em>search</em> or <em>get_spliced</em> are
used.</p>

<!-- ==================== INSTANCE METHODS ==================== -->
<a name="section-InstanceMethods"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Instance Methods</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-InstanceMethods"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.AlignIO.MafIO.MafIndex-class.html#__init__" class="summary-sig-name">__init__</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">sqlite_file</span>,
        <span class="summary-sig-arg">maf_file</span>,
        <span class="summary-sig-arg">target_seqname</span>)</span><br />
      Indexes or loads the index of a MAF file.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.AlignIO.MafIO-pysrc.html#MafIndex.__init__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__check_existing_db"></a><span class="summary-sig-name">__check_existing_db</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Perform basic sanity checks upon loading an existing index (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.AlignIO.MafIO-pysrc.html#MafIndex.__check_existing_db">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__make_new_index"></a><span class="summary-sig-name">__make_new_index</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Read MAF file and generate SQLite index (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.AlignIO.MafIO-pysrc.html#MafIndex.__make_new_index">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.AlignIO.MafIO.MafIndex-class.html#__maf_indexer" class="summary-sig-name" onclick="show_private();">__maf_indexer</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Return index information for each bundle (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.AlignIO.MafIO-pysrc.html#MafIndex.__maf_indexer">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_get_record"></a><span class="summary-sig-name">_get_record</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">offset</span>)</span><br />
      Retrieve a single MAF record located at the offset provided (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.AlignIO.MafIO-pysrc.html#MafIndex._get_record">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.AlignIO.MafIO.MafIndex-class.html#search" class="summary-sig-name">search</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">starts</span>,
        <span class="summary-sig-arg">ends</span>)</span><br />
      Search index database for MAF records overlapping ranges provided.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.AlignIO.MafIO-pysrc.html#MafIndex.search">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.AlignIO.MafIO.MafIndex-class.html#get_spliced" class="summary-sig-name">get_spliced</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">starts</span>,
        <span class="summary-sig-arg">ends</span>,
        <span class="summary-sig-arg">strand</span>=<span class="summary-sig-default">1</span>)</span><br />
      Return a multiple alignment of the exact sequence range provided.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.AlignIO.MafIO-pysrc.html#MafIndex.get_spliced">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.AlignIO.MafIO.MafIndex-class.html#__repr__" class="summary-sig-name">__repr__</a>(<span class="summary-sig-arg">self</span>)</span><br />
      repr(x)</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.AlignIO.MafIO-pysrc.html#MafIndex.__repr__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__len__"></a><span class="summary-sig-name">__len__</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Return the number of records in the index.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.AlignIO.MafIO-pysrc.html#MafIndex.__len__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__delattr__</code>,
      <code>__format__</code>,
      <code>__getattribute__</code>,
      <code>__hash__</code>,
      <code>__new__</code>,
      <code>__reduce__</code>,
      <code>__reduce_ex__</code>,
      <code>__setattr__</code>,
      <code>__sizeof__</code>,
      <code>__str__</code>,
      <code>__subclasshook__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== STATIC METHODS ==================== -->
<a name="section-StaticMethods"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Static Methods</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-StaticMethods"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.AlignIO.MafIO.MafIndex-class.html#_region2bin" class="summary-sig-name" onclick="show_private();">_region2bin</a>(<span class="summary-sig-arg">start</span>,
        <span class="summary-sig-arg">end</span>)</span><br />
      Find bins that a region may belong to (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.AlignIO.MafIO-pysrc.html#MafIndex._region2bin">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.AlignIO.MafIO.MafIndex-class.html#_ucscbin" class="summary-sig-name" onclick="show_private();">_ucscbin</a>(<span class="summary-sig-arg">start</span>,
        <span class="summary-sig-arg">end</span>)</span><br />
      Return the smallest bin a given region will fit into (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.AlignIO.MafIO-pysrc.html#MafIndex._ucscbin">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
</table>
<!-- ==================== PROPERTIES ==================== -->
<a name="section-Properties"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Properties</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Properties"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__class__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== METHOD DETAILS ==================== -->
<a name="section-MethodDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Method Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-MethodDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="__init__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__init__</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">sqlite_file</span>,
        <span class="sig-arg">maf_file</span>,
        <span class="sig-arg">target_seqname</span>)</span>
    <br /><em class="fname">(Constructor)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.AlignIO.MafIO-pysrc.html#MafIndex.__init__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  Indexes or loads the index of a MAF file.
  <dl class="fields">
    <dt>Overrides:
        object.__init__
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="__maf_indexer"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__maf_indexer</span>(<span class="sig-arg">self</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.AlignIO.MafIO-pysrc.html#MafIndex.__maf_indexer">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Return index information for each bundle (PRIVATE).</p>
<p>Yields index information for each bundle in the form of
(bin, start, end, offset) tuples where start and end are
0-based inclusive coordinates.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_region2bin"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_region2bin</span>(<span class="sig-arg">start</span>,
        <span class="sig-arg">end</span>)</span>
    <br /><em class="fname">Static Method</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.AlignIO.MafIO-pysrc.html#MafIndex._region2bin">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Find bins that a region may belong to (PRIVATE).</p>
<p>Converts a region to a list of bins that it may belong to, including largest
and smallest bins.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_ucscbin"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_ucscbin</span>(<span class="sig-arg">start</span>,
        <span class="sig-arg">end</span>)</span>
    <br /><em class="fname">Static Method</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.AlignIO.MafIO-pysrc.html#MafIndex._ucscbin">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Return the smallest bin a given region will fit into (PRIVATE).</p>
<p>Adapted from <a class="rst-reference external" href="http://genomewiki.ucsc.edu/index.php/Bin_indexing_system" target="_top">http://genomewiki.ucsc.edu/index.php/Bin_indexing_system</a></p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="search"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">search</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">starts</span>,
        <span class="sig-arg">ends</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.AlignIO.MafIO-pysrc.html#MafIndex.search">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Search index database for MAF records overlapping ranges provided.</p>
<p>Returns <em>MultipleSeqAlignment</em> results in order by start, then end, then
internal offset field.</p>
<p><em>starts</em> should be a list of 0-based start coordinates of segments in the reference.
<em>ends</em> should be the list of the corresponding segment ends
(in the half-open UCSC convention:
<a class="rst-reference external" href="http://genome.ucsc.edu/blog/the-ucsc-genome-browser-coordinate-counting-systems/" target="_top">http://genome.ucsc.edu/blog/the-ucsc-genome-browser-coordinate-counting-systems/</a>).</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="get_spliced"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">get_spliced</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">starts</span>,
        <span class="sig-arg">ends</span>,
        <span class="sig-arg">strand</span>=<span class="sig-default">1</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.AlignIO.MafIO-pysrc.html#MafIndex.get_spliced">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Return a multiple alignment of the exact sequence range provided.</p>
<p>Accepts two lists of start and end positions on target_seqname, representing
exons to be spliced in silico.  Returns a <em>MultipleSeqAlignment</em> of the
desired sequences spliced together.</p>
<p><em>starts</em> should be a list of 0-based start coordinates of segments in the reference.
<em>ends</em> should be the list of the corresponding segment ends
(in the half-open UCSC convention:
<a class="rst-reference external" href="http://genome.ucsc.edu/blog/the-ucsc-genome-browser-coordinate-counting-systems/" target="_top">http://genome.ucsc.edu/blog/the-ucsc-genome-browser-coordinate-counting-systems/</a>).</p>
<p>To ask for the alignment portion corresponding to the first 100
nucleotides of the reference sequence, you would use
<tt class="rst-docutils literal"><span class="pre">search([0],</span> [100])</tt></p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="__repr__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__repr__</span>(<span class="sig-arg">self</span>)</span>
    <br /><em class="fname">(Representation operator)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.AlignIO.MafIO-pysrc.html#MafIndex.__repr__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  repr(x)
  <dl class="fields">
    <dt>Overrides:
        object.__repr__
        <dd><em class="note">(inherited documentation)</em></dd>
    </dt>
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Mon Jul 10 15:14:22 2017
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
