<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.Application.AbstractCommandline</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.Application-module.html">Package&nbsp;Application</a> ::
        Class&nbsp;AbstractCommandline
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.Application.AbstractCommandline-class.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== CLASS DESCRIPTION ==================== -->
<h1 class="epydoc">Class AbstractCommandline</h1><p class="nomargin-top"><span class="codelink"><a href="Bio.Application-pysrc.html#AbstractCommandline">source&nbsp;code</a></span></p>
<pre class="base-tree">
object --+
         |
        <strong class="uidshort">AbstractCommandline</strong>
</pre>

<dl><dt>Known Subclasses:</dt>
<dd>
      <ul class="subclass-list">
<li class="private"><a href="Bio.PopGen.GenePop.Controller._GenePopCommandline-class.html" onclick="show_private();">PopGen.GenePop.Controller._GenePopCommandline</a></li><li class="private">, <a href="Bio.PopGen.SimCoal.Controller._FastSimCoalCommandLine-class.html" onclick="show_private();">PopGen.SimCoal.Controller._FastSimCoalCommandLine</a></li><li>, <a href="Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline-class.html">Align.Applications._ClustalOmega.ClustalOmegaCommandline</a></li><li>, <a href="Bio.Align.Applications._Clustalw.ClustalwCommandline-class.html">Align.Applications._Clustalw.ClustalwCommandline</a></li><li>, <a href="Bio.Align.Applications._Dialign.DialignCommandline-class.html">Align.Applications._Dialign.DialignCommandline</a></li><li>, <a href="Bio.Align.Applications._MSAProbs.MSAProbsCommandline-class.html">Align.Applications._MSAProbs.MSAProbsCommandline</a></li><li>, <a href="Bio.Align.Applications._Mafft.MafftCommandline-class.html">Align.Applications._Mafft.MafftCommandline</a></li><li>, <a href="Bio.Align.Applications._Muscle.MuscleCommandline-class.html">Align.Applications._Muscle.MuscleCommandline</a></li><li>, <a href="Bio.Align.Applications._Prank.PrankCommandline-class.html">Align.Applications._Prank.PrankCommandline</a></li><li>, <a href="Bio.Align.Applications._Probcons.ProbconsCommandline-class.html">Align.Applications._Probcons.ProbconsCommandline</a></li><li>, <a href="Bio.Align.Applications._TCoffee.TCoffeeCommandline-class.html">Align.Applications._TCoffee.TCoffeeCommandline</a></li><li>, <a href="Bio.Sequencing.Applications._bwa.BwaAlignCommandline-class.html">Sequencing.Applications._bwa.BwaAlignCommandline</a></li><li>, <a href="Bio.Sequencing.Applications._bwa.BwaBwaswCommandline-class.html">Sequencing.Applications._bwa.BwaBwaswCommandline</a></li><li>, <a href="Bio.Sequencing.Applications._bwa.BwaIndexCommandline-class.html">Sequencing.Applications._bwa.BwaIndexCommandline</a></li><li>, <a href="Bio.Sequencing.Applications._bwa.BwaSampeCommandline-class.html">Sequencing.Applications._bwa.BwaSampeCommandline</a></li><li>, <a href="Bio.Sequencing.Applications._bwa.BwaSamseCommandline-class.html">Sequencing.Applications._bwa.BwaSamseCommandline</a></li><li>, <a href="Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline-class.html">Sequencing.Applications._samtools.SamtoolsCalmdCommandline</a></li><li>, <a href="Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline-class.html">Sequencing.Applications._samtools.SamtoolsCatCommandline</a></li><li>, <a href="Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline-class.html">Sequencing.Applications._samtools.SamtoolsFaidxCommandline</a></li><li>, <a href="Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline-class.html">Sequencing.Applications._samtools.SamtoolsFixmateCommandline</a></li><li>, <a href="Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline-class.html">Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline</a></li><li>, <a href="Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline-class.html">Sequencing.Applications._samtools.SamtoolsIndexCommandline</a></li><li>, <a href="Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline-class.html">Sequencing.Applications._samtools.SamtoolsMergeCommandline</a></li><li>, <a href="Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline-class.html">Sequencing.Applications._samtools.SamtoolsMpileupCommandline</a></li><li>, <a href="Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline-class.html">Sequencing.Applications._samtools.SamtoolsPhaseCommandline</a></li><li>, <a href="Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline-class.html">Sequencing.Applications._samtools.SamtoolsReheaderCommandline</a></li><li>, <a href="Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline-class.html">Sequencing.Applications._samtools.SamtoolsRmdupCommandline</a></li><li>, <a href="Bio.Sequencing.Applications._samtools.SamtoolsSortCommandline-class.html">Sequencing.Applications._samtools.SamtoolsSortCommandline</a></li><li>, <a href="Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline-class.html">Sequencing.Applications._samtools.SamtoolsTargetcutCommandline</a></li><li>, <a href="Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline-class.html">Sequencing.Applications._samtools.SamtoolsViewCommandline</a></li><li>, <a href="Bio.Sequencing.Applications._Novoalign.NovoalignCommandline-class.html">Sequencing.Applications._Novoalign.NovoalignCommandline</a></li><li class="private">, <a href="Bio.Blast.Applications._NcbibaseblastCommandline-class.html" onclick="show_private();">Blast.Applications._NcbibaseblastCommandline</a></li><li>, <a href="Bio.Phylo.Applications._Fasttree.FastTreeCommandline-class.html">Phylo.Applications._Fasttree.FastTreeCommandline</a></li><li>, <a href="Bio.Phylo.Applications._Phyml.PhymlCommandline-class.html">Phylo.Applications._Phyml.PhymlCommandline</a></li><li>, <a href="Bio.Phylo.Applications._Raxml.RaxmlCommandline-class.html">Phylo.Applications._Raxml.RaxmlCommandline</a></li><li>, <a href="Bio.motifs.applications._xxmotif.XXmotifCommandline-class.html">motifs.applications._xxmotif.XXmotifCommandline</a></li><li>, <a href="Bio.motifs.applications._alignace.AlignAceCommandline-class.html">motifs.applications._alignace.AlignAceCommandline</a></li><li>, <a href="Bio.motifs.applications._alignace.CompareAceCommandline-class.html">motifs.applications._alignace.CompareAceCommandline</a></li><li class="private">, <a href="Bio.Emboss.Applications._EmbossMinimalCommandLine-class.html" onclick="show_private();">Emboss.Applications._EmbossMinimalCommandLine</a></li>  </ul>
</dd></dl>

<hr />
<p>Generic interface for constructing command line strings.</p>
<p>This class shouldn't be called directly; it should be subclassed to
provide an implementation for a specific application.</p>
<p>For a usage example we'll show one of the EMBOSS wrappers.  You can set
options when creating the wrapper object using keyword arguments - or
later using their corresponding properties:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Emboss.Applications <span class="py-keyword">import</span> WaterCommandline
<span class="py-prompt">&gt;&gt;&gt; </span>cline = WaterCommandline(gapopen=10, gapextend=0.5)
<span class="py-prompt">&gt;&gt;&gt; </span>cline
<span class="py-output">WaterCommandline(cmd='water', gapopen=10, gapextend=0.5)</span></pre>
<p>You can instead manipulate the parameters via their properties, e.g.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>cline.gapopen
<span class="py-output">10</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>cline.gapopen = 20
<span class="py-prompt">&gt;&gt;&gt; </span>cline
<span class="py-output">WaterCommandline(cmd='water', gapopen=20, gapextend=0.5)</span></pre>
<p>You can clear a parameter you have already added by 'deleting' the
corresponding property:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">del</span> cline.gapopen
<span class="py-prompt">&gt;&gt;&gt; </span>cline.gapopen
<span class="py-prompt">&gt;&gt;&gt; </span>cline
<span class="py-output">WaterCommandline(cmd='water', gapextend=0.5)</span></pre>
<p>Once you have set the parameters you need, you can turn the object into
a string (e.g. to log the command):</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>str(cline)
<span class="py-except">Traceback (most recent call last):</span>
<span class="py-except">...</span>
<span class="py-except">ValueError: You must either set outfile (output filename), or enable filter or stdout (output to stdout).</span></pre>
<p>In this case the wrapper knows certain arguments are required to construct
a valid command line for the tool.  For a complete example,</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Emboss.Applications <span class="py-keyword">import</span> WaterCommandline
<span class="py-prompt">&gt;&gt;&gt; </span>water_cmd = WaterCommandline(gapopen=10, gapextend=0.5)
<span class="py-prompt">&gt;&gt;&gt; </span>water_cmd.asequence = <span class="py-string">&quot;asis:ACCCGGGCGCGGT&quot;</span>
<span class="py-prompt">&gt;&gt;&gt; </span>water_cmd.bsequence = <span class="py-string">&quot;asis:ACCCGAGCGCGGT&quot;</span>
<span class="py-prompt">&gt;&gt;&gt; </span>water_cmd.outfile = <span class="py-string">&quot;temp_water.txt&quot;</span>
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(water_cmd)
<span class="py-output">water -outfile=temp_water.txt -asequence=asis:ACCCGGGCGCGGT -bsequence=asis:ACCCGAGCGCGGT -gapopen=10 -gapextend=0.5</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>water_cmd
<span class="py-output">WaterCommandline(cmd='water', outfile='temp_water.txt', asequence='asis:ACCCGGGCGCGGT', bsequence='asis:ACCCGAGCGCGGT', gapopen=10, gapextend=0.5)</span></pre>
<p>You would typically run the command line via a standard Python operating
system call using the subprocess module for full control. For the simple
case where you just want to run the command and get the output:</p>
<p>stdout, stderr = water_cmd()</p>
<p>Note that by default we assume the underlying tool is installed on the
system $PATH environment variable. This is normal under Linux/Unix, but
may need to be done manually under Windows. Alternatively, you can specify
the full path to the binary as the first argument (cmd):</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Emboss.Applications <span class="py-keyword">import</span> WaterCommandline
<span class="py-prompt">&gt;&gt;&gt; </span>water_cmd = WaterCommandline(<span class="py-string">&quot;C:\Program Files\EMBOSS\water.exe&quot;</span>,
<span class="py-more">... </span>                             gapopen=10, gapextend=0.5,
<span class="py-more">... </span>                             asequence=<span class="py-string">&quot;asis:ACCCGGGCGCGGT&quot;</span>,
<span class="py-more">... </span>                             bsequence=<span class="py-string">&quot;asis:ACCCGAGCGCGGT&quot;</span>,
<span class="py-more">... </span>                             outfile=<span class="py-string">&quot;temp_water.txt&quot;</span>)
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(water_cmd)
<span class="py-output">&quot;C:\Program Files\EMBOSS\water.exe&quot; -outfile=temp_water.txt -asequence=asis:ACCCGGGCGCGGT -bsequence=asis:ACCCGAGCGCGGT -gapopen=10 -gapextend=0.5</span></pre>
<p>Notice that since the path name includes a space it has automatically
been quoted.</p>

<!-- ==================== INSTANCE METHODS ==================== -->
<a name="section-InstanceMethods"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Instance Methods</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-InstanceMethods"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Application.AbstractCommandline-class.html#__init__" class="summary-sig-name">__init__</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">cmd</span>,
        <span class="summary-sig-arg">**kwargs</span>)</span><br />
      Create a new instance of a command line wrapper object.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Application-pysrc.html#AbstractCommandline.__init__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Application.AbstractCommandline-class.html#_validate" class="summary-sig-name" onclick="show_private();">_validate</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Make sure the required parameters have been set (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Application-pysrc.html#AbstractCommandline._validate">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Application.AbstractCommandline-class.html#__str__" class="summary-sig-name">__str__</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Make the commandline string with the currently set options.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Application-pysrc.html#AbstractCommandline.__str__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Application.AbstractCommandline-class.html#__repr__" class="summary-sig-name">__repr__</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Return a representation of the command line object for debugging.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Application-pysrc.html#AbstractCommandline.__repr__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_get_parameter"></a><span class="summary-sig-name">_get_parameter</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">name</span>)</span><br />
      Get a commandline option value.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Application-pysrc.html#AbstractCommandline._get_parameter">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_clear_parameter"></a><span class="summary-sig-name">_clear_parameter</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">name</span>)</span><br />
      Reset or clear a commandline option value.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Application-pysrc.html#AbstractCommandline._clear_parameter">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Application.AbstractCommandline-class.html#set_parameter" class="summary-sig-name">set_parameter</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">name</span>,
        <span class="summary-sig-arg">value</span>=<span class="summary-sig-default">None</span>)</span><br />
      Set a commandline option for a program (OBSOLETE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Application-pysrc.html#AbstractCommandline.set_parameter">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Application.AbstractCommandline-class.html#_check_value" class="summary-sig-name" onclick="show_private();">_check_value</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">value</span>,
        <span class="summary-sig-arg">name</span>,
        <span class="summary-sig-arg">check_function</span>)</span><br />
      Check whether the given value is valid.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Application-pysrc.html#AbstractCommandline._check_value">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Application.AbstractCommandline-class.html#__setattr__" class="summary-sig-name">__setattr__</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">name</span>,
        <span class="summary-sig-arg">value</span>)</span><br />
      Set attribute name to value (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Application-pysrc.html#AbstractCommandline.__setattr__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Application.AbstractCommandline-class.html#__call__" class="summary-sig-name">__call__</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">stdin</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">stdout</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">stderr</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">cwd</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">env</span>=<span class="summary-sig-default">None</span>)</span><br />
      Executes the command, waits for it to finish, and returns output.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Application-pysrc.html#AbstractCommandline.__call__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__delattr__</code>,
      <code>__format__</code>,
      <code>__getattribute__</code>,
      <code>__hash__</code>,
      <code>__new__</code>,
      <code>__reduce__</code>,
      <code>__reduce_ex__</code>,
      <code>__sizeof__</code>,
      <code>__subclasshook__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== CLASS VARIABLES ==================== -->
<a name="section-ClassVariables"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Class Variables</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-ClassVariables"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="parameters"></a><span class="summary-name">parameters</span> = <code title="None">None</code><br />
      hash(x)
    </td>
  </tr>
</table>
<!-- ==================== PROPERTIES ==================== -->
<a name="section-Properties"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Properties</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Properties"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__class__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== METHOD DETAILS ==================== -->
<a name="section-MethodDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Method Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-MethodDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="__init__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__init__</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">cmd</span>,
        <span class="sig-arg">**kwargs</span>)</span>
    <br /><em class="fname">(Constructor)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Application-pysrc.html#AbstractCommandline.__init__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  Create a new instance of a command line wrapper object.
  <dl class="fields">
    <dt>Overrides:
        object.__init__
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="_validate"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_validate</span>(<span class="sig-arg">self</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Application-pysrc.html#AbstractCommandline._validate">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Make sure the required parameters have been set (PRIVATE).</p>
<p>No return value - it either works or raises a ValueError.</p>
<p>This is a separate method (called from __str__) so that subclasses may
override it.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="__str__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__str__</span>(<span class="sig-arg">self</span>)</span>
    <br /><em class="fname">(Informal representation operator)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Application-pysrc.html#AbstractCommandline.__str__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Make the commandline string with the currently set options.</p>
<p>e.g.
&gt;&gt;&gt; from Bio.Emboss.Applications import WaterCommandline
&gt;&gt;&gt; cline = WaterCommandline(gapopen=10, gapextend=0.5)
&gt;&gt;&gt; cline.asequence = &quot;asis:ACCCGGGCGCGGT&quot;
&gt;&gt;&gt; cline.bsequence = &quot;asis:ACCCGAGCGCGGT&quot;
&gt;&gt;&gt; cline.outfile = &quot;temp_water.txt&quot;
&gt;&gt;&gt; print(cline)
water -outfile=temp_water.txt -asequence=asis:ACCCGGGCGCGGT -bsequence=asis:ACCCGAGCGCGGT -gapopen=10 -gapextend=0.5
&gt;&gt;&gt; str(cline)
'water -outfile=temp_water.txt -asequence=asis:ACCCGGGCGCGGT -bsequence=asis:ACCCGAGCGCGGT -gapopen=10 -gapextend=0.5'</p>
  <dl class="fields">
    <dt>Overrides:
        object.__str__
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="__repr__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__repr__</span>(<span class="sig-arg">self</span>)</span>
    <br /><em class="fname">(Representation operator)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Application-pysrc.html#AbstractCommandline.__repr__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Return a representation of the command line object for debugging.</p>
<p>e.g.
&gt;&gt;&gt; from Bio.Emboss.Applications import WaterCommandline
&gt;&gt;&gt; cline = WaterCommandline(gapopen=10, gapextend=0.5)
&gt;&gt;&gt; cline.asequence = &quot;asis:ACCCGGGCGCGGT&quot;
&gt;&gt;&gt; cline.bsequence = &quot;asis:ACCCGAGCGCGGT&quot;
&gt;&gt;&gt; cline.outfile = &quot;temp_water.txt&quot;
&gt;&gt;&gt; print(cline)
water -outfile=temp_water.txt -asequence=asis:ACCCGGGCGCGGT -bsequence=asis:ACCCGAGCGCGGT -gapopen=10 -gapextend=0.5
&gt;&gt;&gt; cline
WaterCommandline(cmd='water', outfile='temp_water.txt', asequence='asis:ACCCGGGCGCGGT', bsequence='asis:ACCCGAGCGCGGT', gapopen=10, gapextend=0.5)</p>
  <dl class="fields">
    <dt>Overrides:
        object.__repr__
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="set_parameter"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">set_parameter</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">name</span>,
        <span class="sig-arg">value</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Application-pysrc.html#AbstractCommandline.set_parameter">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Set a commandline option for a program (OBSOLETE).</p>
<p>Every parameter is available via a property and as a named
keyword when creating the instance. Using either of these is
preferred to this legacy set_parameter method which is now
OBSOLETE, and likely to be DEPRECATED and later REMOVED in
future releases.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_check_value"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_check_value</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">value</span>,
        <span class="sig-arg">name</span>,
        <span class="sig-arg">check_function</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Application-pysrc.html#AbstractCommandline._check_value">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Check whether the given value is valid.</p>
<p>No return value - it either works or raises a ValueError.</p>
<p>This uses the passed function 'check_function', which can either
return a [0, 1] (bad, good) value or raise an error. Either way
this function will raise an error if the value is not valid, or
finish silently otherwise.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="__setattr__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__setattr__</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">name</span>,
        <span class="sig-arg">value</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Application-pysrc.html#AbstractCommandline.__setattr__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Set attribute name to value (PRIVATE).</p>
<p>This code implements a workaround for a user interface issue.
Without this __setattr__ attribute-based assignment of parameters
will silently accept invalid parameters, leading to known instances
of the user assuming that parameters for the application are set,
when they are not.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Emboss.Applications <span class="py-keyword">import</span> WaterCommandline
<span class="py-prompt">&gt;&gt;&gt; </span>cline = WaterCommandline(gapopen=10, gapextend=0.5, stdout=True)
<span class="py-prompt">&gt;&gt;&gt; </span>cline.asequence = <span class="py-string">&quot;a.fasta&quot;</span>
<span class="py-prompt">&gt;&gt;&gt; </span>cline.bsequence = <span class="py-string">&quot;b.fasta&quot;</span>
<span class="py-prompt">&gt;&gt;&gt; </span>cline.csequence = <span class="py-string">&quot;c.fasta&quot;</span>
<span class="py-except">Traceback (most recent call last):</span>
<span class="py-except">...</span>
<span class="py-except">ValueError: Option name csequence was not found.</span>
<span class="py-except"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(cline)
<span class="py-output">water -stdout -asequence=a.fasta -bsequence=b.fasta -gapopen=10 -gapextend=0.5</span></pre>
<p>This workaround uses a whitelist of object attributes, and sets the
object attribute list as normal, for these.  Other attributes are
assumed to be parameters, and passed to the self.set_parameter method
for validation and assignment.</p>
  <dl class="fields">
    <dt>Overrides:
        object.__setattr__
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="__call__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__call__</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">stdin</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">stdout</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">stderr</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">cwd</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">env</span>=<span class="sig-default">None</span>)</span>
    <br /><em class="fname">(Call operator)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Application-pysrc.html#AbstractCommandline.__call__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Executes the command, waits for it to finish, and returns output.</p>
<p>Runs the command line tool and waits for it to finish. If it returns
a non-zero error level, an exception is raised. Otherwise two strings
are returned containing stdout and stderr.</p>
<p>The optional stdin argument should be a string of data which will be
passed to the tool as standard input.</p>
<p>The optional stdout and stderr argument may be filenames (string),
but otherwise are treated as a booleans, and control if the output
should be captured as strings (True, default), or ignored by sending
it to /dev/null to avoid wasting memory (False). If sent to a file
or ignored, then empty string(s) are returned.</p>
<p>The optional cwd argument is a string giving the working directory
to run the command from. See Python's subprocess module documentation
for more details.</p>
<p>The optional env argument is a dictionary setting the environment
variables to be used in the new process. By default the current
process' environment variables are used. See Python's subprocess
module documentation for more details.</p>
<p>Default example usage:</p>
<pre class="rst-literal-block">
from Bio.Emboss.Applications import WaterCommandline
water_cmd = WaterCommandline(gapopen=10, gapextend=0.5,
                             stdout=True, auto=True,
                             asequence=&quot;a.fasta&quot;, bsequence=&quot;b.fasta&quot;)
print(&quot;About to run: %s&quot; % water_cmd)
std_output, err_output = water_cmd()
</pre>
<p>This functionality is similar to subprocess.check_output() added in
Python 2.7. In general if you require more control over running the
command, use subprocess directly.</p>
<p>As of Biopython 1.56, when the program called returns a non-zero error
level, a custom ApplicationError exception is raised. This includes
any stdout and stderr strings captured as attributes of the exception
object, since they may be useful for diagnosing what went wrong.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:01:34 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
