<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.Phylo.PhyloXML.Phylogeny</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.Phylo-module.html">Package&nbsp;Phylo</a> ::
        <a href="Bio.Phylo.PhyloXML-module.html">Module&nbsp;PhyloXML</a> ::
        Class&nbsp;Phylogeny
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.Phylo.PhyloXML.Phylogeny-class.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== CLASS DESCRIPTION ==================== -->
<h1 class="epydoc">Class Phylogeny</h1><p class="nomargin-top"><span class="codelink"><a href="Bio.Phylo.PhyloXML-pysrc.html#Phylogeny">source&nbsp;code</a></span></p>
<pre class="base-tree">
          object --+        
                   |        
<a href="Bio.Phylo.BaseTree.TreeElement-class.html">BaseTree.TreeElement</a> --+    
                       |    
            <a href="Bio.Phylo.PhyloXML.PhyloElement-class.html">PhyloElement</a> --+
                           |
          object --+       |
                   |       |
<a href="Bio.Phylo.BaseTree.TreeElement-class.html">BaseTree.TreeElement</a> --+   |
                       |   |
          object --+   |   |
                   |   |   |
  <a href="Bio.Phylo.BaseTree.TreeMixin-class.html">BaseTree.TreeMixin</a> --+   |
                       |   |
           <a href="Bio.Phylo.BaseTree.Tree-class.html">BaseTree.Tree</a> --+
                           |
                          <strong class="uidshort">Phylogeny</strong>
</pre>

<hr />
A phylogenetic tree.

<!-- ==================== INSTANCE METHODS ==================== -->
<a name="section-InstanceMethods"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Instance Methods</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-InstanceMethods"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Phylo.PhyloXML.Phylogeny-class.html#__init__" class="summary-sig-name">__init__</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">root</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">rooted</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">rerootable</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">branch_length_unit</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">type</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">name</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">id</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">description</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">date</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">confidences</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">clade_relations</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">sequence_relations</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">properties</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">other</span>=<span class="summary-sig-default">None</span>)</span><br />
      x.__init__(...) initializes x; see x.__class__.__doc__ for signature</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Phylo.PhyloXML-pysrc.html#Phylogeny.__init__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Phylo.PhyloXML.Phylogeny-class.html#as_phyloxml" class="summary-sig-name">as_phyloxml</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Return this tree, a PhyloXML-compatible Phylogeny object.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Phylo.PhyloXML-pysrc.html#Phylogeny.as_phyloxml">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="to_phyloxml_container"></a><span class="summary-sig-name">to_phyloxml_container</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">**kwargs</span>)</span><br />
      Create a new Phyloxml object containing just this phylogeny.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Phylo.PhyloXML-pysrc.html#Phylogeny.to_phyloxml_container">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="to_alignment"></a><span class="summary-sig-name">to_alignment</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Construct an alignment from the aligned sequences in this tree.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Phylo.PhyloXML-pysrc.html#Phylogeny.to_alignment">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Phylo.PhyloXML.Phylogeny-class.html#_get_confidence" class="summary-sig-name" onclick="show_private();">_get_confidence</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Equivalent to self.confidences[0] if there is only 1 value.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Phylo.PhyloXML-pysrc.html#Phylogeny._get_confidence">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_set_confidence"></a><span class="summary-sig-name">_set_confidence</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">value</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Phylo.PhyloXML-pysrc.html#Phylogeny._set_confidence">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_del_confidence"></a><span class="summary-sig-name">_del_confidence</span>(<span class="summary-sig-arg">self</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Phylo.PhyloXML-pysrc.html#Phylogeny._del_confidence">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code><a href="Bio.Phylo.BaseTree.Tree-class.html">BaseTree.Tree</a></code></b>:
      <code><a href="Bio.Phylo.BaseTree.Tree-class.html#__format__">__format__</a></code>,
      <code><a href="Bio.Phylo.BaseTree.Tree-class.html#__str__">__str__</a></code>,
      <code><a href="Bio.Phylo.BaseTree.Tree-class.html#format">format</a></code>,
      <code><a href="Bio.Phylo.BaseTree.Tree-class.html#is_terminal">is_terminal</a></code>,
      <code><a href="Bio.Phylo.BaseTree.Tree-class.html#root_at_midpoint">root_at_midpoint</a></code>,
      <code><a href="Bio.Phylo.BaseTree.Tree-class.html#root_with_outgroup">root_with_outgroup</a></code>
      </p>
    <p class="indent-wrapped-lines"><b>Inherited from <code><a href="Bio.Phylo.BaseTree.TreeElement-class.html">BaseTree.TreeElement</a></code></b>:
      <code><a href="Bio.Phylo.BaseTree.TreeElement-class.html#__repr__">__repr__</a></code>
      </p>
    <p class="indent-wrapped-lines"><b>Inherited from <code><a href="Bio.Phylo.BaseTree.TreeMixin-class.html">BaseTree.TreeMixin</a></code></b>:
      <code><a href="Bio.Phylo.BaseTree.TreeMixin-class.html#collapse">collapse</a></code>,
      <code><a href="Bio.Phylo.BaseTree.TreeMixin-class.html#collapse_all">collapse_all</a></code>,
      <code><a href="Bio.Phylo.BaseTree.TreeMixin-class.html#common_ancestor">common_ancestor</a></code>,
      <code><a href="Bio.Phylo.BaseTree.TreeMixin-class.html#count_terminals">count_terminals</a></code>,
      <code><a href="Bio.Phylo.BaseTree.TreeMixin-class.html#depths">depths</a></code>,
      <code><a href="Bio.Phylo.BaseTree.TreeMixin-class.html#distance">distance</a></code>,
      <code><a href="Bio.Phylo.BaseTree.TreeMixin-class.html#find_any">find_any</a></code>,
      <code><a href="Bio.Phylo.BaseTree.TreeMixin-class.html#find_clades">find_clades</a></code>,
      <code><a href="Bio.Phylo.BaseTree.TreeMixin-class.html#find_elements">find_elements</a></code>,
      <code><a href="Bio.Phylo.BaseTree.TreeMixin-class.html#get_nonterminals">get_nonterminals</a></code>,
      <code><a href="Bio.Phylo.BaseTree.TreeMixin-class.html#get_path">get_path</a></code>,
      <code><a href="Bio.Phylo.BaseTree.TreeMixin-class.html#get_terminals">get_terminals</a></code>,
      <code><a href="Bio.Phylo.BaseTree.TreeMixin-class.html#is_bifurcating">is_bifurcating</a></code>,
      <code><a href="Bio.Phylo.BaseTree.TreeMixin-class.html#is_monophyletic">is_monophyletic</a></code>,
      <code><a href="Bio.Phylo.BaseTree.TreeMixin-class.html#is_parent_of">is_parent_of</a></code>,
      <code><a href="Bio.Phylo.BaseTree.TreeMixin-class.html#is_preterminal">is_preterminal</a></code>,
      <code><a href="Bio.Phylo.BaseTree.TreeMixin-class.html#ladderize">ladderize</a></code>,
      <code><a href="Bio.Phylo.BaseTree.TreeMixin-class.html#prune">prune</a></code>,
      <code><a href="Bio.Phylo.BaseTree.TreeMixin-class.html#split">split</a></code>,
      <code><a href="Bio.Phylo.BaseTree.TreeMixin-class.html#total_branch_length">total_branch_length</a></code>,
      <code><a href="Bio.Phylo.BaseTree.TreeMixin-class.html#trace">trace</a></code>
      </p>
    <div class="private">    <p class="indent-wrapped-lines"><b>Inherited from <code><a href="Bio.Phylo.BaseTree.TreeMixin-class.html">BaseTree.TreeMixin</a></code></b> (private):
      <code><a href="Bio.Phylo.BaseTree.TreeMixin-class.html#_filter_search" onclick="show_private();">_filter_search</a></code>
      </p></div>
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__delattr__</code>,
      <code>__getattribute__</code>,
      <code>__hash__</code>,
      <code>__new__</code>,
      <code>__reduce__</code>,
      <code>__reduce_ex__</code>,
      <code>__setattr__</code>,
      <code>__sizeof__</code>,
      <code>__subclasshook__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== CLASS METHODS ==================== -->
<a name="section-ClassMethods"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Class Methods</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-ClassMethods"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Phylo.PhyloXML.Phylogeny-class.html#from_tree" class="summary-sig-name">from_tree</a>(<span class="summary-sig-arg">cls</span>,
        <span class="summary-sig-arg">tree</span>,
        <span class="summary-sig-arg">**kwargs</span>)</span><br />
      Create a new Phylogeny given a Tree (from Newick/Nexus or BaseTree).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Phylo.PhyloXML-pysrc.html#Phylogeny.from_tree">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Phylo.PhyloXML.Phylogeny-class.html#from_clade" class="summary-sig-name">from_clade</a>(<span class="summary-sig-arg">cls</span>,
        <span class="summary-sig-arg">clade</span>,
        <span class="summary-sig-arg">**kwargs</span>)</span><br />
      Create a new Phylogeny given a Newick or BaseTree Clade object.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Phylo.PhyloXML-pysrc.html#Phylogeny.from_clade">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code><a href="Bio.Phylo.BaseTree.Tree-class.html">BaseTree.Tree</a></code></b>:
      <code><a href="Bio.Phylo.BaseTree.Tree-class.html#randomized">randomized</a></code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== PROPERTIES ==================== -->
<a name="section-Properties"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Properties</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Properties"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.Phylo.PhyloXML.Phylogeny-class.html#confidence" class="summary-name">confidence</a><br />
      Equivalent to self.confidences[0] if there is only 1 value.
    </td>
  </tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code><a href="Bio.Phylo.BaseTree.Tree-class.html">BaseTree.Tree</a></code></b>:
      <code><a href="Bio.Phylo.BaseTree.Tree-class.html#clade">clade</a></code>
      </p>
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__class__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== METHOD DETAILS ==================== -->
<a name="section-MethodDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Method Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-MethodDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="__init__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__init__</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">root</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">rooted</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">rerootable</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">branch_length_unit</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">type</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">name</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">id</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">description</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">date</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">confidences</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">clade_relations</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">sequence_relations</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">properties</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">other</span>=<span class="sig-default">None</span>)</span>
    <br /><em class="fname">(Constructor)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Phylo.PhyloXML-pysrc.html#Phylogeny.__init__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <pre class="literalblock">
x.__init__(...) initializes x; see x.__class__.__doc__ for signature

</pre>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>root</code></strong> (Clade) - the root node/clade of this tree</li>
        <li><strong class="pname"><code>rooted</code></strong> (bool) - True if this tree is rooted</li>
        <li><strong class="pname"><code>rerootable</code></strong> (bool) - True if this tree is rerootable</li>
        <li><strong class="pname"><code>branch_length_unit</code></strong> (string) - unit for branch_length values on clades</li>
        <li><strong class="pname"><code>name</code></strong> (string) - identifier for this tree, not required to be unique</li>
        <li><strong class="pname"><code>id</code></strong> (Id) - unique identifier for this tree</li>
        <li><strong class="pname"><code>description</code></strong> (string) - plain-text description</li>
        <li><strong class="pname"><code>date</code></strong> (Date) - date for the root node of this tree</li>
        <li><strong class="pname"><code>confidences</code></strong> (list) - Confidence objects for this tree</li>
        <li><strong class="pname"><code>clade_relations</code></strong> (list) - CladeRelation objects</li>
        <li><strong class="pname"><code>sequence_relations</code></strong> (list) - SequenceRelation objects</li>
        <li><strong class="pname"><code>properties</code></strong> (list) - Property objects</li>
        <li><strong class="pname"><code>other</code></strong> (list) - non-phyloXML elements (type <a href="Bio.Phylo.PhyloXML.Other-class.html" class="link">Other</a>)</li>
    </ul></dd>
    <dt>Overrides:
        object.__init__
        <dd><em class="note">(inherited documentation)</em></dd>
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="from_tree"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">from_tree</span>(<span class="sig-arg">cls</span>,
        <span class="sig-arg">tree</span>,
        <span class="sig-arg">**kwargs</span>)</span>
    <br /><em class="fname">Class Method</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Phylo.PhyloXML-pysrc.html#Phylogeny.from_tree">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Create a new Phylogeny given a Tree (from Newick/Nexus or BaseTree).</p>
<p>Keyword arguments are the usual <a href="Bio.Phylo.PhyloXML.Phylogeny-class.html" class="link">Phylogeny</a> constructor parameters.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="from_clade"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">from_clade</span>(<span class="sig-arg">cls</span>,
        <span class="sig-arg">clade</span>,
        <span class="sig-arg">**kwargs</span>)</span>
    <br /><em class="fname">Class Method</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Phylo.PhyloXML-pysrc.html#Phylogeny.from_clade">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Create a new Phylogeny given a Newick or BaseTree Clade object.</p>
<p>Keyword arguments are the usual <a href="Bio.Phylo.PhyloXML.Clade-class.html" class="link">PhyloXML.Clade</a> constructor parameters.</p>
  <dl class="fields">
    <dt>Overrides:
        <a href="Bio.Phylo.BaseTree.Tree-class.html#from_clade">BaseTree.Tree.from_clade</a>
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="as_phyloxml"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">as_phyloxml</span>(<span class="sig-arg">self</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Phylo.PhyloXML-pysrc.html#Phylogeny.as_phyloxml">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Return this tree, a PhyloXML-compatible Phylogeny object.</p>
<p>Overrides the <a href="Bio.Phylo.BaseTree-module.html" class="link">BaseTree</a> method.</p>
  <dl class="fields">
    <dt>Overrides:
        <a href="Bio.Phylo.BaseTree.Tree-class.html#as_phyloxml">BaseTree.Tree.as_phyloxml</a>
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="_get_confidence"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_get_confidence</span>(<span class="sig-arg">self</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Phylo.PhyloXML-pysrc.html#Phylogeny._get_confidence">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Equivalent to self.confidences[0] if there is only 1 value.</p>
<p>See also: <a href="Bio.Phylo.PhyloXML.Clade-class.html#confidence" class="link">Clade.confidence</a>, <a href="Bio.Phylo.PhyloXML.Clade-class.html#taxonomy" class="link">Clade.taxonomy</a></p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== PROPERTY DETAILS ==================== -->
<a name="section-PropertyDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Property Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-PropertyDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="confidence"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">confidence</h3>
  <p>Equivalent to self.confidences[0] if there is only 1 value.</p>
<p>See also: <a href="Bio.Phylo.PhyloXML.Clade-class.html#confidence" class="link">Clade.confidence</a>, <a href="Bio.Phylo.PhyloXML.Clade-class.html#taxonomy" class="link">Clade.taxonomy</a></p>
  <dl class="fields">
    <dt>Get Method:</dt>
    <dd class="value"><span class="summary-sig"><a href="Bio.Phylo.PhyloXML.Phylogeny-class.html#_get_confidence" class="summary-sig-name" onclick="show_private();">_get_confidence</a>(<span class="summary-sig-arg">self</span>)</span>
        - Equivalent to self.confidences[0] if there is only 1 value.
    </dd>
    <dt>Set Method:</dt>
    <dd class="value"><span class="summary-sig"><a href="Bio.Phylo.PhyloXML.Phylogeny-class.html#_set_confidence" class="summary-sig-name" onclick="show_private();">_set_confidence</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">value</span>)</span>
    </dd>
    <dt>Delete Method:</dt>
    <dd class="value"><span class="summary-sig"><a href="Bio.Phylo.PhyloXML.Phylogeny-class.html#_del_confidence" class="summary-sig-name" onclick="show_private();">_del_confidence</a>(<span class="summary-sig-arg">self</span>)</span>
    </dd>
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:01:57 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
