<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.Phylo.PhyloXML</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.Phylo-module.html">Package&nbsp;Phylo</a> ::
        Module&nbsp;PhyloXML
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.Phylo.PhyloXML-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.Phylo.PhyloXML-module.html">Module Bio.Phylo.PhyloXML</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">   1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright (C) 2009 by Eric Talevich (eric.talevich@gmail.com)</tt> </tt>
<a name="L2"></a><tt class="py-lineno">   2</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L3"></a><tt class="py-lineno">   3</tt>  <tt class="py-line"><tt class="py-comment"># license. Please see the LICENSE file that should have been included</tt> </tt>
<a name="L4"></a><tt class="py-lineno">   4</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L5"></a><tt class="py-lineno">   5</tt>  <tt class="py-line"> </tt>
<a name="L6"></a><tt class="py-lineno">   6</tt>  <tt class="py-line"><tt class="py-docstring">"""Classes corresponding to phyloXML elements.</tt> </tt>
<a name="L7"></a><tt class="py-lineno">   7</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L8"></a><tt class="py-lineno">   8</tt>  <tt class="py-line"><tt class="py-docstring">See Also</tt> </tt>
<a name="L9"></a><tt class="py-lineno">   9</tt>  <tt class="py-line"><tt class="py-docstring">--------</tt> </tt>
<a name="L10"></a><tt class="py-lineno">  10</tt>  <tt class="py-line"><tt class="py-docstring">Official specification:</tt> </tt>
<a name="L11"></a><tt class="py-lineno">  11</tt>  <tt class="py-line"><tt class="py-docstring">   http://phyloxml.org/</tt> </tt>
<a name="L12"></a><tt class="py-lineno">  12</tt>  <tt class="py-line"><tt class="py-docstring">Journal article:</tt> </tt>
<a name="L13"></a><tt class="py-lineno">  13</tt>  <tt class="py-line"><tt class="py-docstring">    Han and Zmasek (2009), https://doi.org/10.1186/1471-2105-10-356</tt> </tt>
<a name="L14"></a><tt class="py-lineno">  14</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L15"></a><tt class="py-lineno">  15</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L16"></a><tt class="py-lineno">  16</tt>  <tt class="py-line"> </tt>
<a name="L17"></a><tt class="py-lineno">  17</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">re</tt> </tt>
<a name="L18"></a><tt class="py-lineno">  18</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">sys</tt> </tt>
<a name="L19"></a><tt class="py-lineno">  19</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L20"></a><tt class="py-lineno">  20</tt>  <tt class="py-line"> </tt>
<a name="L21"></a><tt class="py-lineno">  21</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-1', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">basestring</tt> </tt>
<a name="L22"></a><tt class="py-lineno">  22</tt>  <tt class="py-line"> </tt>
<a name="L23"></a><tt class="py-lineno">  23</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-2" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-2', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-3" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-3', 'Alphabet', 'link-3');">Alphabet</a></tt> </tt>
<a name="L24"></a><tt class="py-lineno">  24</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-4" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-4', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-5" class="py-name" targets="Package Bio.Align=Bio.Align-module.html"><a title="Bio.Align" class="py-name" href="#" onclick="return doclink('link-5', 'Align', 'link-5');">Align</a></tt> <tt class="py-keyword">import</tt> <tt id="link-6" class="py-name" targets="Class Bio.Align.MultipleSeqAlignment=Bio.Align.MultipleSeqAlignment-class.html"><a title="Bio.Align.MultipleSeqAlignment" class="py-name" href="#" onclick="return doclink('link-6', 'MultipleSeqAlignment', 'link-6');">MultipleSeqAlignment</a></tt> </tt>
<a name="L25"></a><tt class="py-lineno">  25</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-7" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-7', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-8" class="py-name" targets="Module Bio.Seq=Bio.Seq-module.html,Class Bio.Seq.Seq=Bio.Seq.Seq-class.html"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-8', 'Seq', 'link-8');">Seq</a></tt> <tt class="py-keyword">import</tt> <tt id="link-9" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-9', 'Seq', 'link-8');">Seq</a></tt> </tt>
<a name="L26"></a><tt class="py-lineno">  26</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-10" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-10', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-11" class="py-name" targets="Module Bio.SeqFeature=Bio.SeqFeature-module.html,Class Bio.SeqFeature.SeqFeature=Bio.SeqFeature.SeqFeature-class.html"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-11', 'SeqFeature', 'link-11');">SeqFeature</a></tt> <tt class="py-keyword">import</tt> <tt id="link-12" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-12', 'SeqFeature', 'link-11');">SeqFeature</a></tt><tt class="py-op">,</tt> <tt id="link-13" class="py-name" targets="Class Bio.SeqFeature.FeatureLocation=Bio.SeqFeature.FeatureLocation-class.html"><a title="Bio.SeqFeature.FeatureLocation" class="py-name" href="#" onclick="return doclink('link-13', 'FeatureLocation', 'link-13');">FeatureLocation</a></tt> </tt>
<a name="L27"></a><tt class="py-lineno">  27</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-14" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-14', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-15" class="py-name" targets="Module Bio.SeqRecord=Bio.SeqRecord-module.html,Class Bio.SeqRecord.SeqRecord=Bio.SeqRecord.SeqRecord-class.html"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-15', 'SeqRecord', 'link-15');">SeqRecord</a></tt> <tt class="py-keyword">import</tt> <tt id="link-16" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-16', 'SeqRecord', 'link-15');">SeqRecord</a></tt> </tt>
<a name="L28"></a><tt class="py-lineno">  28</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-17" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-17', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-18" class="py-name" targets="Class Bio.BiopythonWarning=Bio.BiopythonWarning-class.html"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-18', 'BiopythonWarning', 'link-18');">BiopythonWarning</a></tt> </tt>
<a name="L29"></a><tt class="py-lineno">  29</tt>  <tt class="py-line"> </tt>
<a name="L30"></a><tt class="py-lineno">  30</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-19" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-19', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-20" class="py-name" targets="Package Bio.Phylo=Bio.Phylo-module.html"><a title="Bio.Phylo" class="py-name" href="#" onclick="return doclink('link-20', 'Phylo', 'link-20');">Phylo</a></tt> <tt class="py-keyword">import</tt> <tt id="link-21" class="py-name" targets="Module Bio.Phylo.BaseTree=Bio.Phylo.BaseTree-module.html"><a title="Bio.Phylo.BaseTree" class="py-name" href="#" onclick="return doclink('link-21', 'BaseTree', 'link-21');">BaseTree</a></tt> </tt>
<a name="PhyloXMLWarning"></a><div id="PhyloXMLWarning-def"><a name="L31"></a><tt class="py-lineno">  31</tt>  <tt class="py-line"> </tt>
<a name="L32"></a><tt class="py-lineno">  32</tt>  <tt class="py-line"> </tt>
<a name="L33"></a><tt class="py-lineno">  33</tt> <a class="py-toggle" href="#" id="PhyloXMLWarning-toggle" onclick="return toggle('PhyloXMLWarning');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.PhyloXMLWarning-class.html">PhyloXMLWarning</a><tt class="py-op">(</tt><tt class="py-base-class">BiopythonWarning</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PhyloXMLWarning-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="PhyloXMLWarning-expanded"><a name="L34"></a><tt class="py-lineno">  34</tt>  <tt class="py-line">    <tt class="py-docstring">"""Warning for non-compliance with the phyloXML specification."""</tt> </tt>
<a name="L35"></a><tt class="py-lineno">  35</tt>  <tt class="py-line"> </tt>
<a name="L36"></a><tt class="py-lineno">  36</tt>  <tt class="py-line">    <tt class="py-keyword">pass</tt> </tt>
</div><a name="L37"></a><tt class="py-lineno">  37</tt>  <tt class="py-line"> </tt>
<a name="_check_str"></a><div id="_check_str-def"><a name="L38"></a><tt class="py-lineno">  38</tt>  <tt class="py-line"> </tt>
<a name="L39"></a><tt class="py-lineno">  39</tt> <a class="py-toggle" href="#" id="_check_str-toggle" onclick="return toggle('_check_str');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML-module.html#_check_str">_check_str</a><tt class="py-op">(</tt><tt class="py-param">text</tt><tt class="py-op">,</tt> <tt class="py-param">testfunc</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_check_str-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_check_str-expanded"><a name="L40"></a><tt class="py-lineno">  40</tt>  <tt class="py-line">    <tt class="py-docstring">"""Check a string using testfunc, and warn if there's no match (PRIVATE)."""</tt> </tt>
<a name="L41"></a><tt class="py-lineno">  41</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">text</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt class="py-name">testfunc</tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L42"></a><tt class="py-lineno">  42</tt>  <tt class="py-line">        <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"String %s doesn't match the given regexp"</tt> <tt class="py-op">%</tt> <tt class="py-name">text</tt><tt class="py-op">,</tt> </tt>
<a name="L43"></a><tt class="py-lineno">  43</tt>  <tt class="py-line">                      <tt id="link-22" class="py-name" targets="Class Bio.Phylo.PhyloXML.PhyloXMLWarning=Bio.Phylo.PhyloXML.PhyloXMLWarning-class.html"><a title="Bio.Phylo.PhyloXML.PhyloXMLWarning" class="py-name" href="#" onclick="return doclink('link-22', 'PhyloXMLWarning', 'link-22');">PhyloXMLWarning</a></tt><tt class="py-op">,</tt> <tt class="py-name">stacklevel</tt><tt class="py-op">=</tt><tt class="py-number">2</tt><tt class="py-op">)</tt> </tt>
</div><a name="L44"></a><tt class="py-lineno">  44</tt>  <tt class="py-line"> </tt>
<a name="PhyloElement"></a><div id="PhyloElement-def"><a name="L45"></a><tt class="py-lineno">  45</tt>  <tt class="py-line"> </tt>
<a name="L46"></a><tt class="py-lineno">  46</tt>  <tt class="py-line"><tt class="py-comment"># Core elements</tt> </tt>
<a name="L47"></a><tt class="py-lineno">  47</tt>  <tt class="py-line"> </tt>
<a name="L48"></a><tt class="py-lineno">  48</tt> <a class="py-toggle" href="#" id="PhyloElement-toggle" onclick="return toggle('PhyloElement');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.PhyloElement-class.html">PhyloElement</a><tt class="py-op">(</tt><tt class="py-base-class">BaseTree</tt><tt class="py-op">.</tt><tt class="py-base-class">TreeElement</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PhyloElement-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="PhyloElement-expanded"><a name="L49"></a><tt class="py-lineno">  49</tt>  <tt class="py-line">    <tt class="py-docstring">"""Base class for all PhyloXML objects."""</tt> </tt>
</div><a name="L50"></a><tt class="py-lineno">  50</tt>  <tt class="py-line"> </tt>
<a name="Phyloxml"></a><div id="Phyloxml-def"><a name="L51"></a><tt class="py-lineno">  51</tt>  <tt class="py-line"> </tt>
<a name="L52"></a><tt class="py-lineno">  52</tt> <a class="py-toggle" href="#" id="Phyloxml-toggle" onclick="return toggle('Phyloxml');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Phyloxml-class.html">Phyloxml</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Phyloxml-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Phyloxml-expanded"><a name="L53"></a><tt class="py-lineno">  53</tt>  <tt class="py-line">    <tt class="py-docstring">"""Root node of the PhyloXML document.</tt> </tt>
<a name="L54"></a><tt class="py-lineno">  54</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L55"></a><tt class="py-lineno">  55</tt>  <tt class="py-line"><tt class="py-docstring">    Contains an arbitrary number of Phylogeny elements, possibly followed by</tt> </tt>
<a name="L56"></a><tt class="py-lineno">  56</tt>  <tt class="py-line"><tt class="py-docstring">    elements from other namespaces.</tt> </tt>
<a name="L57"></a><tt class="py-lineno">  57</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L58"></a><tt class="py-lineno">  58</tt>  <tt class="py-line"><tt class="py-docstring">    :Parameters:</tt> </tt>
<a name="L59"></a><tt class="py-lineno">  59</tt>  <tt class="py-line"><tt class="py-docstring">        attributes : dict</tt> </tt>
<a name="L60"></a><tt class="py-lineno">  60</tt>  <tt class="py-line"><tt class="py-docstring">            (XML namespace definitions)</tt> </tt>
<a name="L61"></a><tt class="py-lineno">  61</tt>  <tt class="py-line"><tt class="py-docstring">        phylogenies : list</tt> </tt>
<a name="L62"></a><tt class="py-lineno">  62</tt>  <tt class="py-line"><tt class="py-docstring">            The phylogenetic trees</tt> </tt>
<a name="L63"></a><tt class="py-lineno">  63</tt>  <tt class="py-line"><tt class="py-docstring">        other : list</tt> </tt>
<a name="L64"></a><tt class="py-lineno">  64</tt>  <tt class="py-line"><tt class="py-docstring">            Arbitrary non-phyloXML elements, if any</tt> </tt>
<a name="L65"></a><tt class="py-lineno">  65</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L66"></a><tt class="py-lineno">  66</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L67"></a><tt class="py-lineno">  67</tt>  <tt class="py-line"> </tt>
<a name="Phyloxml.__init__"></a><div id="Phyloxml.__init__-def"><a name="L68"></a><tt class="py-lineno">  68</tt> <a class="py-toggle" href="#" id="Phyloxml.__init__-toggle" onclick="return toggle('Phyloxml.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Phyloxml-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">attributes</tt><tt class="py-op">,</tt> <tt class="py-param">phylogenies</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Phyloxml.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Phyloxml.__init__-expanded"><a name="L69"></a><tt class="py-lineno">  69</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize parameters for PhyloXML object."""</tt> </tt>
<a name="L70"></a><tt class="py-lineno">  70</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">attributes</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L71"></a><tt class="py-lineno">  71</tt>  <tt class="py-line">            <tt class="py-comment"># standard</tt> </tt>
<a name="L72"></a><tt class="py-lineno">  72</tt>  <tt class="py-line">            <tt class="py-string">"xmlns:xsi"</tt><tt class="py-op">:</tt> <tt class="py-string">"http://www.w3.org/2001/XMLSchema-instance"</tt><tt class="py-op">,</tt> </tt>
<a name="L73"></a><tt class="py-lineno">  73</tt>  <tt class="py-line">            <tt class="py-string">"xmlns"</tt><tt class="py-op">:</tt> <tt class="py-string">"http://www.phyloxml.org"</tt><tt class="py-op">,</tt> </tt>
<a name="L74"></a><tt class="py-lineno">  74</tt>  <tt class="py-line">            <tt class="py-string">"xsi:schemaLocation"</tt><tt class="py-op">:</tt> <tt class="py-string">"http://www.phyloxml.org http://www.phyloxml.org/1.10/phyloxml.xsd"</tt><tt class="py-op">,</tt> </tt>
<a name="L75"></a><tt class="py-lineno">  75</tt>  <tt class="py-line">        <tt class="py-op">}</tt> </tt>
<a name="L76"></a><tt class="py-lineno">  76</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">attributes</tt><tt class="py-op">:</tt> </tt>
<a name="L77"></a><tt class="py-lineno">  77</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">attributes</tt><tt class="py-op">.</tt><tt id="link-23" class="py-name" targets="Method Bio.File._IndexedSeqFileDict.update()=Bio.File._IndexedSeqFileDict-class.html#update,Method Bio.Index._InMemoryIndex.update()=Bio.Index._InMemoryIndex-class.html#update,Method Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.update()=Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer-class.html#update,Method Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.update()=Bio.NeuralNetwork.BackPropagation.Layer.InputLayer-class.html#update,Method Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.update()=Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer-class.html#update,Method Bio.SeqRecord._RestrictedDict.update()=Bio.SeqRecord._RestrictedDict-class.html#update"><a title="Bio.File._IndexedSeqFileDict.update
Bio.Index._InMemoryIndex.update
Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.update
Bio.SeqRecord._RestrictedDict.update" class="py-name" href="#" onclick="return doclink('link-23', 'update', 'link-23');">update</a></tt><tt class="py-op">(</tt><tt class="py-name">attributes</tt><tt class="py-op">)</tt> </tt>
<a name="L78"></a><tt class="py-lineno">  78</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">phylogenies</tt> <tt class="py-op">=</tt> <tt class="py-name">phylogenies</tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L79"></a><tt class="py-lineno">  79</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-24" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.other()=Bio.Phylo.PhyloXMLIO.Parser-class.html#other,Method Bio.Phylo.PhyloXMLIO.Writer.other()=Bio.Phylo.PhyloXMLIO.Writer-class.html#other"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-24', 'other', 'link-24');">other</a></tt> <tt class="py-op">=</tt> <tt id="link-25" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-25', 'other', 'link-24');">other</a></tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div><a name="L80"></a><tt class="py-lineno">  80</tt>  <tt class="py-line"> </tt>
<a name="Phyloxml.__getitem__"></a><div id="Phyloxml.__getitem__-def"><a name="L81"></a><tt class="py-lineno">  81</tt> <a class="py-toggle" href="#" id="Phyloxml.__getitem__-toggle" onclick="return toggle('Phyloxml.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Phyloxml-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">index</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Phyloxml.__getitem__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Phyloxml.__getitem__-expanded"><a name="L82"></a><tt class="py-lineno">  82</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get a phylogeny by index or name."""</tt> </tt>
<a name="L83"></a><tt class="py-lineno">  83</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-26" class="py-name" targets="Method Bio.Crystal.Chain.index()=Bio.Crystal.Chain-class.html#index,Method Bio.SCOP.Raf.SeqMap.index()=Bio.SCOP.Raf.SeqMap-class.html#index,Method Bio.SearchIO._model.hit.Hit.index()=Bio.SearchIO._model.hit.Hit-class.html#index,Method Bio.SearchIO._model.query.QueryResult.index()=Bio.SearchIO._model.query.QueryResult-class.html#index,Function Bio.SearchIO.index()=Bio.SearchIO-module.html#index,Method Bio.Seq.MutableSeq.index()=Bio.Seq.MutableSeq-class.html#index,Function Bio.SeqIO.index()=Bio.SeqIO-module.html#index"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-26', 'index', 'link-26');">index</a></tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">int</tt><tt class="py-op">,</tt> <tt class="py-name">slice</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L84"></a><tt class="py-lineno">  84</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">phylogenies</tt><tt class="py-op">[</tt><tt id="link-27" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-27', 'index', 'link-26');">index</a></tt><tt class="py-op">]</tt> </tt>
<a name="L85"></a><tt class="py-lineno">  85</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-28" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-28', 'index', 'link-26');">index</a></tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L86"></a><tt class="py-lineno">  86</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">KeyError</tt><tt class="py-op">(</tt><tt class="py-string">"can't use %s as an index"</tt> <tt class="py-op">%</tt> <tt id="link-29" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.type()=Bio.Phylo.PhyloXMLIO.Writer-class.html#type"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-29', 'type', 'link-29');">type</a></tt><tt class="py-op">(</tt><tt id="link-30" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-30', 'index', 'link-26');">index</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L87"></a><tt class="py-lineno">  87</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">tree</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">phylogenies</tt><tt class="py-op">:</tt> </tt>
<a name="L88"></a><tt class="py-lineno">  88</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">tree</tt><tt class="py-op">.</tt><tt id="link-31" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-31', 'name', 'link-31');">name</a></tt> <tt class="py-op">==</tt> <tt id="link-32" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-32', 'index', 'link-26');">index</a></tt><tt class="py-op">:</tt> </tt>
<a name="L89"></a><tt class="py-lineno">  89</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">tree</tt> </tt>
<a name="L90"></a><tt class="py-lineno">  90</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L91"></a><tt class="py-lineno">  91</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">KeyError</tt><tt class="py-op">(</tt><tt class="py-string">"no phylogeny found with name "</tt> <tt class="py-op">+</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt id="link-33" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-33', 'index', 'link-26');">index</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L92"></a><tt class="py-lineno">  92</tt>  <tt class="py-line"> </tt>
<a name="Phyloxml.__iter__"></a><div id="Phyloxml.__iter__-def"><a name="L93"></a><tt class="py-lineno">  93</tt> <a class="py-toggle" href="#" id="Phyloxml.__iter__-toggle" onclick="return toggle('Phyloxml.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Phyloxml-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Phyloxml.__iter__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Phyloxml.__iter__-expanded"><a name="L94"></a><tt class="py-lineno">  94</tt>  <tt class="py-line">        <tt class="py-docstring">"""Iterate through the phylogenetic trees in this object."""</tt> </tt>
<a name="L95"></a><tt class="py-lineno">  95</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">iter</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">phylogenies</tt><tt class="py-op">)</tt> </tt>
</div><a name="L96"></a><tt class="py-lineno">  96</tt>  <tt class="py-line"> </tt>
<a name="Phyloxml.__len__"></a><div id="Phyloxml.__len__-def"><a name="L97"></a><tt class="py-lineno">  97</tt> <a class="py-toggle" href="#" id="Phyloxml.__len__-toggle" onclick="return toggle('Phyloxml.__len__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Phyloxml-class.html#__len__">__len__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Phyloxml.__len__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Phyloxml.__len__-expanded"><a name="L98"></a><tt class="py-lineno">  98</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the number of phylogenetic trees in this object."""</tt> </tt>
<a name="L99"></a><tt class="py-lineno">  99</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">phylogenies</tt><tt class="py-op">)</tt> </tt>
</div><a name="L100"></a><tt class="py-lineno"> 100</tt>  <tt class="py-line"> </tt>
<a name="Phyloxml.__str__"></a><div id="Phyloxml.__str__-def"><a name="L101"></a><tt class="py-lineno"> 101</tt> <a class="py-toggle" href="#" id="Phyloxml.__str__-toggle" onclick="return toggle('Phyloxml.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Phyloxml-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Phyloxml.__str__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Phyloxml.__str__-expanded"><a name="L102"></a><tt class="py-lineno"> 102</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return name of phylogenies in the object."""</tt> </tt>
<a name="L103"></a><tt class="py-lineno"> 103</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">'%s([%s])'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">,</tt> </tt>
<a name="L104"></a><tt class="py-lineno"> 104</tt>  <tt class="py-line">                             <tt class="py-string">',\n'</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt id="link-34" class="py-name" targets="Method Bio.SearchIO._model.hit.Hit.map()=Bio.SearchIO._model.hit.Hit-class.html#map"><a title="Bio.SearchIO._model.hit.Hit.map" class="py-name" href="#" onclick="return doclink('link-34', 'map', 'link-34');">map</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">phylogenies</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L105"></a><tt class="py-lineno"> 105</tt>  <tt class="py-line"> </tt>
<a name="Other"></a><div id="Other-def"><a name="L106"></a><tt class="py-lineno"> 106</tt>  <tt class="py-line"> </tt>
<a name="L107"></a><tt class="py-lineno"> 107</tt> <a class="py-toggle" href="#" id="Other-toggle" onclick="return toggle('Other');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Other-class.html">Other</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Other-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Other-expanded"><a name="L108"></a><tt class="py-lineno"> 108</tt>  <tt class="py-line">    <tt class="py-docstring">"""Container for non-phyloXML elements in the tree.</tt> </tt>
<a name="L109"></a><tt class="py-lineno"> 109</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L110"></a><tt class="py-lineno"> 110</tt>  <tt class="py-line"><tt class="py-docstring">    Usually, an Other object will have either a 'value' or a non-empty list</tt> </tt>
<a name="L111"></a><tt class="py-lineno"> 111</tt>  <tt class="py-line"><tt class="py-docstring">    of 'children', but not both. This is not enforced here, though.</tt> </tt>
<a name="L112"></a><tt class="py-lineno"> 112</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L113"></a><tt class="py-lineno"> 113</tt>  <tt class="py-line"><tt class="py-docstring">    :Parameters:</tt> </tt>
<a name="L114"></a><tt class="py-lineno"> 114</tt>  <tt class="py-line"><tt class="py-docstring">        tag : string</tt> </tt>
<a name="L115"></a><tt class="py-lineno"> 115</tt>  <tt class="py-line"><tt class="py-docstring">            local tag for the XML node</tt> </tt>
<a name="L116"></a><tt class="py-lineno"> 116</tt>  <tt class="py-line"><tt class="py-docstring">        namespace : string</tt> </tt>
<a name="L117"></a><tt class="py-lineno"> 117</tt>  <tt class="py-line"><tt class="py-docstring">            XML namespace for the node -- should not be the default phyloXML</tt> </tt>
<a name="L118"></a><tt class="py-lineno"> 118</tt>  <tt class="py-line"><tt class="py-docstring">            namespace.</tt> </tt>
<a name="L119"></a><tt class="py-lineno"> 119</tt>  <tt class="py-line"><tt class="py-docstring">        attributes : dict of strings</tt> </tt>
<a name="L120"></a><tt class="py-lineno"> 120</tt>  <tt class="py-line"><tt class="py-docstring">            attributes on the XML node</tt> </tt>
<a name="L121"></a><tt class="py-lineno"> 121</tt>  <tt class="py-line"><tt class="py-docstring">        value : string</tt> </tt>
<a name="L122"></a><tt class="py-lineno"> 122</tt>  <tt class="py-line"><tt class="py-docstring">            text contained directly within this XML node</tt> </tt>
<a name="L123"></a><tt class="py-lineno"> 123</tt>  <tt class="py-line"><tt class="py-docstring">        children : list</tt> </tt>
<a name="L124"></a><tt class="py-lineno"> 124</tt>  <tt class="py-line"><tt class="py-docstring">            child nodes, if any (also `Other` instances)</tt> </tt>
<a name="L125"></a><tt class="py-lineno"> 125</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L126"></a><tt class="py-lineno"> 126</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L127"></a><tt class="py-lineno"> 127</tt>  <tt class="py-line"> </tt>
<a name="Other.__init__"></a><div id="Other.__init__-def"><a name="L128"></a><tt class="py-lineno"> 128</tt> <a class="py-toggle" href="#" id="Other.__init__-toggle" onclick="return toggle('Other.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Other-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">tag</tt><tt class="py-op">,</tt> <tt class="py-param">namespace</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">attributes</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L129"></a><tt class="py-lineno"> 129</tt>  <tt class="py-line">                 <tt class="py-param">children</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Other.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Other.__init__-expanded"><a name="L130"></a><tt class="py-lineno"> 130</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize values for non-phyloXML elements."""</tt> </tt>
<a name="L131"></a><tt class="py-lineno"> 131</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-35" class="py-name" targets="Variable Bio.SeqIO.AbiIO.tag=Bio.SeqIO.AbiIO-module.html#tag"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-35', 'tag', 'link-35');">tag</a></tt> <tt class="py-op">=</tt> <tt id="link-36" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-36', 'tag', 'link-35');">tag</a></tt> </tt>
<a name="L132"></a><tt class="py-lineno"> 132</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">namespace</tt> <tt class="py-op">=</tt> <tt class="py-name">namespace</tt> </tt>
<a name="L133"></a><tt class="py-lineno"> 133</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">attributes</tt> <tt class="py-op">=</tt> <tt class="py-name">attributes</tt> <tt class="py-keyword">or</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L134"></a><tt class="py-lineno"> 134</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-37" class="py-name" targets="Variable Bio.Entrez.Parser.Consumer.value=Bio.Entrez.Parser.Consumer-class.html#value,Variable Bio.Entrez.Parser.DictionaryConsumer.value=Bio.Entrez.Parser.DictionaryConsumer-class.html#value,Variable Bio.Entrez.Parser.ErrorConsumer.value=Bio.Entrez.Parser.ErrorConsumer-class.html#value,Variable Bio.Entrez.Parser.IntegerConsumer.value=Bio.Entrez.Parser.IntegerConsumer-class.html#value,Variable Bio.Entrez.Parser.ListConsumer.value=Bio.Entrez.Parser.ListConsumer-class.html#value,Variable Bio.Entrez.Parser.StringConsumer.value=Bio.Entrez.Parser.StringConsumer-class.html#value,Method Bio.Phylo.PhyloXMLIO.Writer.value()=Bio.Phylo.PhyloXMLIO.Writer-class.html#value"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-37', 'value', 'link-37');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-38" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-38', 'value', 'link-37');">value</a></tt> </tt>
<a name="L135"></a><tt class="py-lineno"> 135</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-39" class="py-name" targets="Method Bio.Pathway.Rep.Graph.Graph.children()=Bio.Pathway.Rep.Graph.Graph-class.html#children,Method Bio.Pathway.Rep.MultiGraph.MultiGraph.children()=Bio.Pathway.Rep.MultiGraph.MultiGraph-class.html#children"><a title="Bio.Pathway.Rep.Graph.Graph.children
Bio.Pathway.Rep.MultiGraph.MultiGraph.children" class="py-name" href="#" onclick="return doclink('link-39', 'children', 'link-39');">children</a></tt> <tt class="py-op">=</tt> <tt id="link-40" class="py-name"><a title="Bio.Pathway.Rep.Graph.Graph.children
Bio.Pathway.Rep.MultiGraph.MultiGraph.children" class="py-name" href="#" onclick="return doclink('link-40', 'children', 'link-39');">children</a></tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div><a name="L136"></a><tt class="py-lineno"> 136</tt>  <tt class="py-line"> </tt>
<a name="Other.__iter__"></a><div id="Other.__iter__-def"><a name="L137"></a><tt class="py-lineno"> 137</tt> <a class="py-toggle" href="#" id="Other.__iter__-toggle" onclick="return toggle('Other.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Other-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Other.__iter__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Other.__iter__-expanded"><a name="L138"></a><tt class="py-lineno"> 138</tt>  <tt class="py-line">        <tt class="py-docstring">"""Iterate through the children of this object (if any)."""</tt> </tt>
<a name="L139"></a><tt class="py-lineno"> 139</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">iter</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-41" class="py-name"><a title="Bio.Pathway.Rep.Graph.Graph.children
Bio.Pathway.Rep.MultiGraph.MultiGraph.children" class="py-name" href="#" onclick="return doclink('link-41', 'children', 'link-39');">children</a></tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L140"></a><tt class="py-lineno"> 140</tt>  <tt class="py-line"> </tt>
<a name="Phylogeny"></a><div id="Phylogeny-def"><a name="L141"></a><tt class="py-lineno"> 141</tt>  <tt class="py-line"> </tt>
<a name="L142"></a><tt class="py-lineno"> 142</tt> <a class="py-toggle" href="#" id="Phylogeny-toggle" onclick="return toggle('Phylogeny');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Phylogeny-class.html">Phylogeny</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">,</tt> <tt class="py-base-class">BaseTree</tt><tt class="py-op">.</tt><tt class="py-base-class">Tree</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Phylogeny-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Phylogeny-expanded"><a name="L143"></a><tt class="py-lineno"> 143</tt>  <tt class="py-line">    <tt class="py-docstring">"""A phylogenetic tree.</tt> </tt>
<a name="L144"></a><tt class="py-lineno"> 144</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L145"></a><tt class="py-lineno"> 145</tt>  <tt class="py-line"><tt class="py-docstring">    :Parameters:</tt> </tt>
<a name="L146"></a><tt class="py-lineno"> 146</tt>  <tt class="py-line"><tt class="py-docstring">        root : Clade</tt> </tt>
<a name="L147"></a><tt class="py-lineno"> 147</tt>  <tt class="py-line"><tt class="py-docstring">            the root node/clade of this tree</tt> </tt>
<a name="L148"></a><tt class="py-lineno"> 148</tt>  <tt class="py-line"><tt class="py-docstring">        rooted : bool</tt> </tt>
<a name="L149"></a><tt class="py-lineno"> 149</tt>  <tt class="py-line"><tt class="py-docstring">            True if this tree is rooted</tt> </tt>
<a name="L150"></a><tt class="py-lineno"> 150</tt>  <tt class="py-line"><tt class="py-docstring">        rerootable : bool</tt> </tt>
<a name="L151"></a><tt class="py-lineno"> 151</tt>  <tt class="py-line"><tt class="py-docstring">            True if this tree is rerootable</tt> </tt>
<a name="L152"></a><tt class="py-lineno"> 152</tt>  <tt class="py-line"><tt class="py-docstring">        branch_length_unit : string</tt> </tt>
<a name="L153"></a><tt class="py-lineno"> 153</tt>  <tt class="py-line"><tt class="py-docstring">            unit for branch_length values on clades</tt> </tt>
<a name="L154"></a><tt class="py-lineno"> 154</tt>  <tt class="py-line"><tt class="py-docstring">        name : string</tt> </tt>
<a name="L155"></a><tt class="py-lineno"> 155</tt>  <tt class="py-line"><tt class="py-docstring">            identifier for this tree, not required to be unique</tt> </tt>
<a name="L156"></a><tt class="py-lineno"> 156</tt>  <tt class="py-line"><tt class="py-docstring">        id : Id</tt> </tt>
<a name="L157"></a><tt class="py-lineno"> 157</tt>  <tt class="py-line"><tt class="py-docstring">            unique identifier for this tree</tt> </tt>
<a name="L158"></a><tt class="py-lineno"> 158</tt>  <tt class="py-line"><tt class="py-docstring">        description : string</tt> </tt>
<a name="L159"></a><tt class="py-lineno"> 159</tt>  <tt class="py-line"><tt class="py-docstring">            plain-text description</tt> </tt>
<a name="L160"></a><tt class="py-lineno"> 160</tt>  <tt class="py-line"><tt class="py-docstring">        date : Date</tt> </tt>
<a name="L161"></a><tt class="py-lineno"> 161</tt>  <tt class="py-line"><tt class="py-docstring">            date for the root node of this tree</tt> </tt>
<a name="L162"></a><tt class="py-lineno"> 162</tt>  <tt class="py-line"><tt class="py-docstring">        confidences : list</tt> </tt>
<a name="L163"></a><tt class="py-lineno"> 163</tt>  <tt class="py-line"><tt class="py-docstring">            Confidence objects for this tree</tt> </tt>
<a name="L164"></a><tt class="py-lineno"> 164</tt>  <tt class="py-line"><tt class="py-docstring">        clade_relations : list</tt> </tt>
<a name="L165"></a><tt class="py-lineno"> 165</tt>  <tt class="py-line"><tt class="py-docstring">            CladeRelation objects</tt> </tt>
<a name="L166"></a><tt class="py-lineno"> 166</tt>  <tt class="py-line"><tt class="py-docstring">        sequence_relations : list</tt> </tt>
<a name="L167"></a><tt class="py-lineno"> 167</tt>  <tt class="py-line"><tt class="py-docstring">            SequenceRelation objects</tt> </tt>
<a name="L168"></a><tt class="py-lineno"> 168</tt>  <tt class="py-line"><tt class="py-docstring">        properties : list</tt> </tt>
<a name="L169"></a><tt class="py-lineno"> 169</tt>  <tt class="py-line"><tt class="py-docstring">            Property objects</tt> </tt>
<a name="L170"></a><tt class="py-lineno"> 170</tt>  <tt class="py-line"><tt class="py-docstring">        other : list</tt> </tt>
<a name="L171"></a><tt class="py-lineno"> 171</tt>  <tt class="py-line"><tt class="py-docstring">            non-phyloXML elements (type `Other`)</tt> </tt>
<a name="L172"></a><tt class="py-lineno"> 172</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L173"></a><tt class="py-lineno"> 173</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L174"></a><tt class="py-lineno"> 174</tt>  <tt class="py-line"> </tt>
<a name="Phylogeny.__init__"></a><div id="Phylogeny.__init__-def"><a name="L175"></a><tt class="py-lineno"> 175</tt> <a class="py-toggle" href="#" id="Phylogeny.__init__-toggle" onclick="return toggle('Phylogeny.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Phylogeny-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">root</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">rooted</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">,</tt> </tt>
<a name="L176"></a><tt class="py-lineno"> 176</tt>  <tt class="py-line">                 <tt class="py-param">rerootable</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">branch_length_unit</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">type</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L177"></a><tt class="py-lineno"> 177</tt>  <tt class="py-line">                 <tt class="py-comment"># Child nodes</tt> </tt>
<a name="L178"></a><tt class="py-lineno"> 178</tt>  <tt class="py-line">                 <tt class="py-param">name</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">id</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">description</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">date</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L179"></a><tt class="py-lineno"> 179</tt>  <tt class="py-line">                 <tt class="py-comment"># Collections</tt> </tt>
<a name="L180"></a><tt class="py-lineno"> 180</tt>  <tt class="py-line">                 <tt class="py-param">confidences</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">clade_relations</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">sequence_relations</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L181"></a><tt class="py-lineno"> 181</tt>  <tt class="py-line">                 <tt class="py-param">properties</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L182"></a><tt class="py-lineno"> 182</tt>  <tt class="py-line">                 <tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Phylogeny.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Phylogeny.__init__-expanded"><a name="L183"></a><tt class="py-lineno"> 183</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize values for phylogenetic tree object."""</tt> </tt>
<a name="L184"></a><tt class="py-lineno"> 184</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">rooted</tt><tt class="py-op">,</tt> <tt class="py-name">bool</tt><tt class="py-op">)</tt> </tt>
<a name="L185"></a><tt class="py-lineno"> 185</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-42" class="py-name" targets="Variable Bio.Phylo.BaseTree.Clade.root=Bio.Phylo.BaseTree.Clade-class.html#root,Variable Bio.Phylo._cdao_owl.root=Bio.Phylo._cdao_owl-module.html#root"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-42', 'root', 'link-42');">root</a></tt> <tt class="py-op">=</tt> <tt id="link-43" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-43', 'root', 'link-42');">root</a></tt> </tt>
<a name="L186"></a><tt class="py-lineno"> 186</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">rooted</tt> <tt class="py-op">=</tt> <tt class="py-name">rooted</tt> </tt>
<a name="L187"></a><tt class="py-lineno"> 187</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">rerootable</tt> <tt class="py-op">=</tt> <tt class="py-name">rerootable</tt> </tt>
<a name="L188"></a><tt class="py-lineno"> 188</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">branch_length_unit</tt> <tt class="py-op">=</tt> <tt class="py-name">branch_length_unit</tt> </tt>
<a name="L189"></a><tt class="py-lineno"> 189</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-44" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-44', 'type', 'link-29');">type</a></tt> <tt class="py-op">=</tt> <tt id="link-45" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-45', 'type', 'link-29');">type</a></tt> </tt>
<a name="L190"></a><tt class="py-lineno"> 190</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-46" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-46', 'name', 'link-31');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-47" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-47', 'name', 'link-31');">name</a></tt> </tt>
<a name="L191"></a><tt class="py-lineno"> 191</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-48" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Variable Bio.PDB.Entity.Entity.id=Bio.PDB.Entity.Entity-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-48', 'id', 'link-48');">id</a></tt> <tt class="py-op">=</tt> <tt id="link-49" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-49', 'id', 'link-48');">id</a></tt> </tt>
<a name="L192"></a><tt class="py-lineno"> 192</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-50" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DescriptionConsumer.description()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#description,Method Bio.Phylo.PhyloXMLIO.Writer.description()=Bio.Phylo.PhyloXMLIO.Writer-class.html#description,Variable Bio.SearchIO._model.hit.Hit.description=Bio.SearchIO._model.hit.Hit-class.html#description,Variable Bio.SearchIO._model.query.QueryResult.description=Bio.SearchIO._model.query.QueryResult-class.html#description"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-50', 'description', 'link-50');">description</a></tt> <tt class="py-op">=</tt> <tt id="link-51" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-51', 'description', 'link-50');">description</a></tt> </tt>
<a name="L193"></a><tt class="py-lineno"> 193</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-52" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.date()=Bio.GenBank._FeatureConsumer-class.html#date,Method Bio.GenBank._RecordConsumer.date()=Bio.GenBank._RecordConsumer-class.html#date,Method Bio.Phylo.PhyloXMLIO.Parser.date()=Bio.Phylo.PhyloXMLIO.Parser-class.html#date,Method Bio.Phylo.PhyloXMLIO.Writer.date()=Bio.Phylo.PhyloXMLIO.Writer-class.html#date"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-52', 'date', 'link-52');">date</a></tt> <tt class="py-op">=</tt> <tt id="link-53" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-53', 'date', 'link-52');">date</a></tt> </tt>
<a name="L194"></a><tt class="py-lineno"> 194</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">confidences</tt> <tt class="py-op">=</tt> <tt class="py-name">confidences</tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L195"></a><tt class="py-lineno"> 195</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">clade_relations</tt> <tt class="py-op">=</tt> <tt class="py-name">clade_relations</tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L196"></a><tt class="py-lineno"> 196</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">sequence_relations</tt> <tt class="py-op">=</tt> <tt class="py-name">sequence_relations</tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L197"></a><tt class="py-lineno"> 197</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">properties</tt> <tt class="py-op">=</tt> <tt class="py-name">properties</tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L198"></a><tt class="py-lineno"> 198</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-54" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-54', 'other', 'link-24');">other</a></tt> <tt class="py-op">=</tt> <tt id="link-55" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-55', 'other', 'link-24');">other</a></tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div><a name="L199"></a><tt class="py-lineno"> 199</tt>  <tt class="py-line"> </tt>
<a name="L200"></a><tt class="py-lineno"> 200</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">classmethod</tt> </tt>
<a name="Phylogeny.from_tree"></a><div id="Phylogeny.from_tree-def"><a name="L201"></a><tt class="py-lineno"> 201</tt> <a class="py-toggle" href="#" id="Phylogeny.from_tree-toggle" onclick="return toggle('Phylogeny.from_tree');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Phylogeny-class.html#from_tree">from_tree</a><tt class="py-op">(</tt><tt class="py-param">cls</tt><tt class="py-op">,</tt> <tt class="py-param">tree</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Phylogeny.from_tree-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Phylogeny.from_tree-expanded"><a name="L202"></a><tt class="py-lineno"> 202</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a new Phylogeny given a Tree (from Newick/Nexus or BaseTree).</tt> </tt>
<a name="L203"></a><tt class="py-lineno"> 203</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L204"></a><tt class="py-lineno"> 204</tt>  <tt class="py-line"><tt class="py-docstring">        Keyword arguments are the usual `Phylogeny` constructor parameters.</tt> </tt>
<a name="L205"></a><tt class="py-lineno"> 205</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L206"></a><tt class="py-lineno"> 206</tt>  <tt class="py-line">        <tt class="py-name">phy</tt> <tt class="py-op">=</tt> <tt class="py-name">cls</tt><tt class="py-op">(</tt> </tt>
<a name="L207"></a><tt class="py-lineno"> 207</tt>  <tt class="py-line">            <tt id="link-56" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-56', 'root', 'link-42');">root</a></tt><tt class="py-op">=</tt><tt id="link-57" class="py-name" targets="Class Bio.Phylo.BaseTree.Clade=Bio.Phylo.BaseTree.Clade-class.html,Class Bio.Phylo.CDAO.Clade=Bio.Phylo.CDAO.Clade-class.html,Class Bio.Phylo.NeXML.Clade=Bio.Phylo.NeXML.Clade-class.html,Class Bio.Phylo.Newick.Clade=Bio.Phylo.Newick.Clade-class.html,Class Bio.Phylo.PhyloXML.Clade=Bio.Phylo.PhyloXML.Clade-class.html"><a title="Bio.Phylo.BaseTree.Clade
Bio.Phylo.CDAO.Clade
Bio.Phylo.NeXML.Clade
Bio.Phylo.Newick.Clade
Bio.Phylo.PhyloXML.Clade" class="py-name" href="#" onclick="return doclink('link-57', 'Clade', 'link-57');">Clade</a></tt><tt class="py-op">.</tt><tt id="link-58" class="py-name" targets="Class Method Bio.Phylo.BaseTree.Tree.from_clade()=Bio.Phylo.BaseTree.Tree-class.html#from_clade,Class Method Bio.Phylo.PhyloXML.Clade.from_clade()=Bio.Phylo.PhyloXML.Clade-class.html#from_clade,Class Method Bio.Phylo.PhyloXML.Phylogeny.from_clade()=Bio.Phylo.PhyloXML.Phylogeny-class.html#from_clade"><a title="Bio.Phylo.BaseTree.Tree.from_clade
Bio.Phylo.PhyloXML.Clade.from_clade
Bio.Phylo.PhyloXML.Phylogeny.from_clade" class="py-name" href="#" onclick="return doclink('link-58', 'from_clade', 'link-58');">from_clade</a></tt><tt class="py-op">(</tt><tt class="py-name">tree</tt><tt class="py-op">.</tt><tt id="link-59" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-59', 'root', 'link-42');">root</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L208"></a><tt class="py-lineno"> 208</tt>  <tt class="py-line">            <tt class="py-name">rooted</tt><tt class="py-op">=</tt><tt class="py-name">tree</tt><tt class="py-op">.</tt><tt class="py-name">rooted</tt><tt class="py-op">,</tt> </tt>
<a name="L209"></a><tt class="py-lineno"> 209</tt>  <tt class="py-line">            <tt id="link-60" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-60', 'name', 'link-31');">name</a></tt><tt class="py-op">=</tt><tt class="py-name">tree</tt><tt class="py-op">.</tt><tt id="link-61" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-61', 'name', 'link-31');">name</a></tt><tt class="py-op">,</tt> </tt>
<a name="L210"></a><tt class="py-lineno"> 210</tt>  <tt class="py-line">            <tt id="link-62" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-62', 'id', 'link-48');">id</a></tt><tt class="py-op">=</tt><tt class="py-op">(</tt><tt class="py-name">tree</tt><tt class="py-op">.</tt><tt id="link-63" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-63', 'id', 'link-48');">id</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt id="link-64" class="py-name" targets="Class Bio.Phylo.PhyloXML.Id=Bio.Phylo.PhyloXML.Id-class.html"><a title="Bio.Phylo.PhyloXML.Id" class="py-name" href="#" onclick="return doclink('link-64', 'Id', 'link-64');">Id</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">tree</tt><tt class="py-op">.</tt><tt id="link-65" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-65', 'id', 'link-48');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
<a name="L211"></a><tt class="py-lineno"> 211</tt>  <tt class="py-line">        <tt class="py-name">phy</tt><tt class="py-op">.</tt><tt class="py-name">__dict__</tt><tt class="py-op">.</tt><tt id="link-66" class="py-name"><a title="Bio.File._IndexedSeqFileDict.update
Bio.Index._InMemoryIndex.update
Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.update
Bio.SeqRecord._RestrictedDict.update" class="py-name" href="#" onclick="return doclink('link-66', 'update', 'link-23');">update</a></tt><tt class="py-op">(</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
<a name="L212"></a><tt class="py-lineno"> 212</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">phy</tt> </tt>
</div><a name="L213"></a><tt class="py-lineno"> 213</tt>  <tt class="py-line"> </tt>
<a name="L214"></a><tt class="py-lineno"> 214</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">classmethod</tt> </tt>
<a name="Phylogeny.from_clade"></a><div id="Phylogeny.from_clade-def"><a name="L215"></a><tt class="py-lineno"> 215</tt> <a class="py-toggle" href="#" id="Phylogeny.from_clade-toggle" onclick="return toggle('Phylogeny.from_clade');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Phylogeny-class.html#from_clade">from_clade</a><tt class="py-op">(</tt><tt class="py-param">cls</tt><tt class="py-op">,</tt> <tt class="py-param">clade</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Phylogeny.from_clade-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Phylogeny.from_clade-expanded"><a name="L216"></a><tt class="py-lineno"> 216</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a new Phylogeny given a Newick or BaseTree Clade object.</tt> </tt>
<a name="L217"></a><tt class="py-lineno"> 217</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L218"></a><tt class="py-lineno"> 218</tt>  <tt class="py-line"><tt class="py-docstring">        Keyword arguments are the usual `PhyloXML.Clade` constructor parameters.</tt> </tt>
<a name="L219"></a><tt class="py-lineno"> 219</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L220"></a><tt class="py-lineno"> 220</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-67" class="py-name"><a title="Bio.Phylo.BaseTree.Clade
Bio.Phylo.CDAO.Clade
Bio.Phylo.NeXML.Clade
Bio.Phylo.Newick.Clade
Bio.Phylo.PhyloXML.Clade" class="py-name" href="#" onclick="return doclink('link-67', 'Clade', 'link-57');">Clade</a></tt><tt class="py-op">.</tt><tt id="link-68" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.from_clade
Bio.Phylo.PhyloXML.Clade.from_clade
Bio.Phylo.PhyloXML.Phylogeny.from_clade" class="py-name" href="#" onclick="return doclink('link-68', 'from_clade', 'link-58');">from_clade</a></tt><tt class="py-op">(</tt><tt id="link-69" class="py-name" targets="Variable Bio.Phylo.BaseTree.Tree.clade=Bio.Phylo.BaseTree.Tree-class.html#clade,Method Bio.Phylo.PhyloXMLIO.Writer.clade()=Bio.Phylo.PhyloXMLIO.Writer-class.html#clade"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-69', 'clade', 'link-69');">clade</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-70" class="py-name" targets="Method Bio.Phylo.PhyloXML.Clade.to_phylogeny()=Bio.Phylo.PhyloXML.Clade-class.html#to_phylogeny"><a title="Bio.Phylo.PhyloXML.Clade.to_phylogeny" class="py-name" href="#" onclick="return doclink('link-70', 'to_phylogeny', 'link-70');">to_phylogeny</a></tt><tt class="py-op">(</tt><tt class="py-op">**</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
</div><a name="L221"></a><tt class="py-lineno"> 221</tt>  <tt class="py-line"> </tt>
<a name="Phylogeny.as_phyloxml"></a><div id="Phylogeny.as_phyloxml-def"><a name="L222"></a><tt class="py-lineno"> 222</tt> <a class="py-toggle" href="#" id="Phylogeny.as_phyloxml-toggle" onclick="return toggle('Phylogeny.as_phyloxml');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Phylogeny-class.html#as_phyloxml">as_phyloxml</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Phylogeny.as_phyloxml-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Phylogeny.as_phyloxml-expanded"><a name="L223"></a><tt class="py-lineno"> 223</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return this tree, a PhyloXML-compatible Phylogeny object.</tt> </tt>
<a name="L224"></a><tt class="py-lineno"> 224</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L225"></a><tt class="py-lineno"> 225</tt>  <tt class="py-line"><tt class="py-docstring">        Overrides the `BaseTree` method.</tt> </tt>
<a name="L226"></a><tt class="py-lineno"> 226</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L227"></a><tt class="py-lineno"> 227</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt> </tt>
</div><a name="L228"></a><tt class="py-lineno"> 228</tt>  <tt class="py-line"> </tt>
<a name="Phylogeny.to_phyloxml_container"></a><div id="Phylogeny.to_phyloxml_container-def"><a name="L229"></a><tt class="py-lineno"> 229</tt> <a class="py-toggle" href="#" id="Phylogeny.to_phyloxml_container-toggle" onclick="return toggle('Phylogeny.to_phyloxml_container');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Phylogeny-class.html#to_phyloxml_container">to_phyloxml_container</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Phylogeny.to_phyloxml_container-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Phylogeny.to_phyloxml_container-expanded"><a name="L230"></a><tt class="py-lineno"> 230</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a new Phyloxml object containing just this phylogeny."""</tt> </tt>
<a name="L231"></a><tt class="py-lineno"> 231</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-71" class="py-name" targets="Class Bio.Phylo.PhyloXML.Phyloxml=Bio.Phylo.PhyloXML.Phyloxml-class.html"><a title="Bio.Phylo.PhyloXML.Phyloxml" class="py-name" href="#" onclick="return doclink('link-71', 'Phyloxml', 'link-71');">Phyloxml</a></tt><tt class="py-op">(</tt><tt class="py-name">kwargs</tt><tt class="py-op">,</tt> <tt class="py-name">phylogenies</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L232"></a><tt class="py-lineno"> 232</tt>  <tt class="py-line"> </tt>
<a name="Phylogeny.to_alignment"></a><div id="Phylogeny.to_alignment-def"><a name="L233"></a><tt class="py-lineno"> 233</tt> <a class="py-toggle" href="#" id="Phylogeny.to_alignment-toggle" onclick="return toggle('Phylogeny.to_alignment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Phylogeny-class.html#to_alignment">to_alignment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Phylogeny.to_alignment-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Phylogeny.to_alignment-expanded"><a name="L234"></a><tt class="py-lineno"> 234</tt>  <tt class="py-line">        <tt class="py-docstring">"""Construct an alignment from the aligned sequences in this tree."""</tt> </tt>
<a name="L235"></a><tt class="py-lineno"> 235</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">is_aligned_seq</tt><tt class="py-op">(</tt><tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L236"></a><tt class="py-lineno"> 236</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt id="link-72" class="py-name" targets="Class Bio.Phylo.PhyloXML.Sequence=Bio.Phylo.PhyloXML.Sequence-class.html"><a title="Bio.Phylo.PhyloXML.Sequence" class="py-name" href="#" onclick="return doclink('link-72', 'Sequence', 'link-72');">Sequence</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-73" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.mol_seq()=Bio.Phylo.PhyloXMLIO.Parser-class.html#mol_seq,Method Bio.Phylo.PhyloXMLIO.Writer.mol_seq()=Bio.Phylo.PhyloXMLIO.Writer-class.html#mol_seq"><a title="Bio.Phylo.PhyloXMLIO.Parser.mol_seq
Bio.Phylo.PhyloXMLIO.Writer.mol_seq" class="py-name" href="#" onclick="return doclink('link-73', 'mol_seq', 'link-73');">mol_seq</a></tt><tt class="py-op">.</tt><tt class="py-name">is_aligned</tt><tt class="py-op">:</tt> </tt>
<a name="L237"></a><tt class="py-lineno"> 237</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">True</tt> </tt>
<a name="L238"></a><tt class="py-lineno"> 238</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
</div><a name="L239"></a><tt class="py-lineno"> 239</tt>  <tt class="py-line">        <tt class="py-name">seqs</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-74" class="py-name" targets="Method Bio.Phylo.BaseTree.TreeMixin._filter_search()=Bio.Phylo.BaseTree.TreeMixin-class.html#_filter_search"><a title="Bio.Phylo.BaseTree.TreeMixin._filter_search" class="py-name" href="#" onclick="return doclink('link-74', '_filter_search', 'link-74');">_filter_search</a></tt><tt class="py-op">(</tt><tt class="py-name">is_aligned_seq</tt><tt class="py-op">,</tt> <tt class="py-string">'preorder'</tt><tt class="py-op">,</tt> <tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L240"></a><tt class="py-lineno"> 240</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L241"></a><tt class="py-lineno"> 241</tt>  <tt class="py-line">            <tt class="py-name">first_seq</tt> <tt class="py-op">=</tt> <tt id="link-75" class="py-name" targets="Method Bio.AlignIO.Interfaces.AlignmentIterator.next()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#next,Method Bio.Blast.NCBIStandalone.Iterator.next()=Bio.Blast.NCBIStandalone.Iterator-class.html#next,Method Bio.Blast.ParseBlastTable.BlastTableReader.next()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#next,Method Bio.File.UndoHandle.next()=Bio.File.UndoHandle-class.html#next,Method Bio.GenBank.Iterator.next()=Bio.GenBank.Iterator-class.html#next,Method Bio.Nexus.Nexus.CharBuffer.next()=Bio.Nexus.Nexus.CharBuffer-class.html#next,Method Bio.Nexus.StandardData.StandardData.next()=Bio.Nexus.StandardData.StandardData-class.html#next,Method Bio.PopGen.GenePop.Controller._FileIterator.next()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#next,Method Bio.SeqIO.Interfaces.SequenceIterator.next()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#next,Method Bio.bgzf.BgzfReader.next()=Bio.bgzf.BgzfReader-class.html#next"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.Nexus.StandardData.StandardData.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-75', 'next', 'link-75');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">seqs</tt><tt class="py-op">)</tt> </tt>
<a name="L242"></a><tt class="py-lineno"> 242</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L243"></a><tt class="py-lineno"> 243</tt>  <tt class="py-line">            <tt class="py-comment"># No aligned sequences were found --&gt; empty MSA</tt> </tt>
<a name="L244"></a><tt class="py-lineno"> 244</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-76" class="py-name"><a title="Bio.Align.MultipleSeqAlignment" class="py-name" href="#" onclick="return doclink('link-76', 'MultipleSeqAlignment', 'link-6');">MultipleSeqAlignment</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L245"></a><tt class="py-lineno"> 245</tt>  <tt class="py-line">        <tt class="py-name">msa</tt> <tt class="py-op">=</tt> <tt id="link-77" class="py-name"><a title="Bio.Align.MultipleSeqAlignment" class="py-name" href="#" onclick="return doclink('link-77', 'MultipleSeqAlignment', 'link-6');">MultipleSeqAlignment</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">first_seq</tt><tt class="py-op">.</tt><tt id="link-78" class="py-name" targets="Method Bio.Phylo.PhyloXML.Sequence.to_seqrecord()=Bio.Phylo.PhyloXML.Sequence-class.html#to_seqrecord"><a title="Bio.Phylo.PhyloXML.Sequence.to_seqrecord" class="py-name" href="#" onclick="return doclink('link-78', 'to_seqrecord', 'link-78');">to_seqrecord</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L246"></a><tt class="py-lineno"> 246</tt>  <tt class="py-line">                                   <tt class="py-name">first_seq</tt><tt class="py-op">.</tt><tt id="link-79" class="py-name" targets="Method Bio.Phylo.PhyloXML.Sequence.get_alphabet()=Bio.Phylo.PhyloXML.Sequence-class.html#get_alphabet"><a title="Bio.Phylo.PhyloXML.Sequence.get_alphabet" class="py-name" href="#" onclick="return doclink('link-79', 'get_alphabet', 'link-79');">get_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L247"></a><tt class="py-lineno"> 247</tt>  <tt class="py-line">        <tt class="py-name">msa</tt><tt class="py-op">.</tt><tt id="link-80" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.extend()=Bio.Align.MultipleSeqAlignment-class.html#extend,Method Bio.SCOP.Raf.SeqMap.extend()=Bio.SCOP.Raf.SeqMap-class.html#extend,Method Bio.Seq.MutableSeq.extend()=Bio.Seq.MutableSeq-class.html#extend"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-80', 'extend', 'link-80');">extend</a></tt><tt class="py-op">(</tt><tt id="link-81" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.seq=Bio.SeqRecord.SeqRecord-class.html#seq,Variable BioSQL.BioSeq.DBSeqRecord.seq=BioSQL.BioSeq.DBSeqRecord-class.html#seq"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-81', 'seq', 'link-81');">seq</a></tt><tt class="py-op">.</tt><tt id="link-82" class="py-name"><a title="Bio.Phylo.PhyloXML.Sequence.to_seqrecord" class="py-name" href="#" onclick="return doclink('link-82', 'to_seqrecord', 'link-78');">to_seqrecord</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-83" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-83', 'seq', 'link-81');">seq</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">seqs</tt><tt class="py-op">)</tt> </tt>
<a name="L248"></a><tt class="py-lineno"> 248</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">msa</tt> </tt>
</div><a name="L249"></a><tt class="py-lineno"> 249</tt>  <tt class="py-line"> </tt>
<a name="L250"></a><tt class="py-lineno"> 250</tt>  <tt class="py-line">    <tt class="py-comment"># Singular property for plural attribute</tt> </tt>
<a name="Phylogeny._get_confidence"></a><div id="Phylogeny._get_confidence-def"><a name="L251"></a><tt class="py-lineno"> 251</tt> <a class="py-toggle" href="#" id="Phylogeny._get_confidence-toggle" onclick="return toggle('Phylogeny._get_confidence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Phylogeny-class.html#_get_confidence">_get_confidence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Phylogeny._get_confidence-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Phylogeny._get_confidence-expanded"><a name="L252"></a><tt class="py-lineno"> 252</tt>  <tt class="py-line">        <tt class="py-docstring">"""Equivalent to self.confidences[0] if there is only 1 value (PRIVATE).</tt> </tt>
<a name="L253"></a><tt class="py-lineno"> 253</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L254"></a><tt class="py-lineno"> 254</tt>  <tt class="py-line"><tt class="py-docstring">        See Also: `Clade.confidence`, `Clade.taxonomy`</tt> </tt>
<a name="L255"></a><tt class="py-lineno"> 255</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L256"></a><tt class="py-lineno"> 256</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L257"></a><tt class="py-lineno"> 257</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">confidences</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L258"></a><tt class="py-lineno"> 258</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
<a name="L259"></a><tt class="py-lineno"> 259</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">confidences</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L260"></a><tt class="py-lineno"> 260</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">(</tt><tt class="py-string">"more than 1 confidence value available; "</tt> </tt>
<a name="L261"></a><tt class="py-lineno"> 261</tt>  <tt class="py-line">                                 <tt class="py-string">"use Phylogeny.confidences"</tt><tt class="py-op">)</tt> </tt>
<a name="L262"></a><tt class="py-lineno"> 262</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">confidences</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
</div><a name="L263"></a><tt class="py-lineno"> 263</tt>  <tt class="py-line"> </tt>
<a name="Phylogeny._set_confidence"></a><div id="Phylogeny._set_confidence-def"><a name="L264"></a><tt class="py-lineno"> 264</tt> <a class="py-toggle" href="#" id="Phylogeny._set_confidence-toggle" onclick="return toggle('Phylogeny._set_confidence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Phylogeny-class.html#_set_confidence">_set_confidence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Phylogeny._set_confidence-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Phylogeny._set_confidence-expanded"><a name="L265"></a><tt class="py-lineno"> 265</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-84" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-84', 'value', 'link-37');">value</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L266"></a><tt class="py-lineno"> 266</tt>  <tt class="py-line">            <tt class="py-comment"># Special case: mirror the behavior of _get_confidence</tt> </tt>
<a name="L267"></a><tt class="py-lineno"> 267</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">confidences</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L268"></a><tt class="py-lineno"> 268</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L269"></a><tt class="py-lineno"> 269</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-85" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-85', 'value', 'link-37');">value</a></tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">float</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L270"></a><tt class="py-lineno"> 270</tt>  <tt class="py-line">            <tt id="link-86" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-86', 'value', 'link-37');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-87" class="py-name" targets="Class Bio.Phylo.PhyloXML.Confidence=Bio.Phylo.PhyloXML.Confidence-class.html"><a title="Bio.Phylo.PhyloXML.Confidence" class="py-name" href="#" onclick="return doclink('link-87', 'Confidence', 'link-87');">Confidence</a></tt><tt class="py-op">(</tt><tt id="link-88" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-88', 'value', 'link-37');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L271"></a><tt class="py-lineno"> 271</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-89" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-89', 'value', 'link-37');">value</a></tt><tt class="py-op">,</tt> <tt id="link-90" class="py-name"><a title="Bio.Phylo.PhyloXML.Confidence" class="py-name" href="#" onclick="return doclink('link-90', 'Confidence', 'link-87');">Confidence</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L272"></a><tt class="py-lineno"> 272</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"value must be a number or Confidence instance"</tt><tt class="py-op">)</tt> </tt>
<a name="L273"></a><tt class="py-lineno"> 273</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">confidences</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L274"></a><tt class="py-lineno"> 274</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">confidences</tt><tt class="py-op">.</tt><tt id="link-91" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-91', 'append', 'link-91');">append</a></tt><tt class="py-op">(</tt><tt id="link-92" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-92', 'value', 'link-37');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L275"></a><tt class="py-lineno"> 275</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">confidences</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L276"></a><tt class="py-lineno"> 276</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">confidences</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-93" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-93', 'value', 'link-37');">value</a></tt> </tt>
<a name="L277"></a><tt class="py-lineno"> 277</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L278"></a><tt class="py-lineno"> 278</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"multiple confidence values already exist; "</tt> </tt>
<a name="L279"></a><tt class="py-lineno"> 279</tt>  <tt class="py-line">                             <tt class="py-string">"use Phylogeny.confidences instead"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L280"></a><tt class="py-lineno"> 280</tt>  <tt class="py-line"> </tt>
<a name="Phylogeny._del_confidence"></a><div id="Phylogeny._del_confidence-def"><a name="L281"></a><tt class="py-lineno"> 281</tt> <a class="py-toggle" href="#" id="Phylogeny._del_confidence-toggle" onclick="return toggle('Phylogeny._del_confidence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Phylogeny-class.html#_del_confidence">_del_confidence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Phylogeny._del_confidence-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Phylogeny._del_confidence-expanded"><a name="L282"></a><tt class="py-lineno"> 282</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">confidences</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div><a name="L283"></a><tt class="py-lineno"> 283</tt>  <tt class="py-line"> </tt>
<a name="L284"></a><tt class="py-lineno"> 284</tt>  <tt class="py-line">    <tt id="link-94" class="py-name" targets="Variable Bio.Phylo.PhyloXML.Clade.confidence=Bio.Phylo.PhyloXML.Clade-class.html#confidence,Variable Bio.Phylo.PhyloXML.Phylogeny.confidence=Bio.Phylo.PhyloXML.Phylogeny-class.html#confidence,Method Bio.Phylo.PhyloXMLIO.Parser.confidence()=Bio.Phylo.PhyloXMLIO.Parser-class.html#confidence,Method Bio.Phylo.PhyloXMLIO.Writer.confidence()=Bio.Phylo.PhyloXMLIO.Writer-class.html#confidence"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-94', 'confidence', 'link-94');">confidence</a></tt> <tt class="py-op">=</tt> <tt id="link-95" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.property()=Bio.Phylo.PhyloXMLIO.Parser-class.html#property,Method Bio.Phylo.PhyloXMLIO.Writer.property()=Bio.Phylo.PhyloXMLIO.Writer-class.html#property"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-95', 'property', 'link-95');">property</a></tt><tt class="py-op">(</tt><tt id="link-96" class="py-name" targets="Method Bio.Phylo.PhyloXML.Clade._get_confidence()=Bio.Phylo.PhyloXML.Clade-class.html#_get_confidence,Method Bio.Phylo.PhyloXML.Phylogeny._get_confidence()=Bio.Phylo.PhyloXML.Phylogeny-class.html#_get_confidence"><a title="Bio.Phylo.PhyloXML.Clade._get_confidence
Bio.Phylo.PhyloXML.Phylogeny._get_confidence" class="py-name" href="#" onclick="return doclink('link-96', '_get_confidence', 'link-96');">_get_confidence</a></tt><tt class="py-op">,</tt> <tt id="link-97" class="py-name" targets="Method Bio.Phylo.PhyloXML.Clade._set_confidence()=Bio.Phylo.PhyloXML.Clade-class.html#_set_confidence,Method Bio.Phylo.PhyloXML.Phylogeny._set_confidence()=Bio.Phylo.PhyloXML.Phylogeny-class.html#_set_confidence"><a title="Bio.Phylo.PhyloXML.Clade._set_confidence
Bio.Phylo.PhyloXML.Phylogeny._set_confidence" class="py-name" href="#" onclick="return doclink('link-97', '_set_confidence', 'link-97');">_set_confidence</a></tt><tt class="py-op">,</tt> <tt id="link-98" class="py-name" targets="Method Bio.Phylo.PhyloXML.Clade._del_confidence()=Bio.Phylo.PhyloXML.Clade-class.html#_del_confidence,Method Bio.Phylo.PhyloXML.Phylogeny._del_confidence()=Bio.Phylo.PhyloXML.Phylogeny-class.html#_del_confidence"><a title="Bio.Phylo.PhyloXML.Clade._del_confidence
Bio.Phylo.PhyloXML.Phylogeny._del_confidence" class="py-name" href="#" onclick="return doclink('link-98', '_del_confidence', 'link-98');">_del_confidence</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L285"></a><tt class="py-lineno"> 285</tt>  <tt class="py-line"> </tt>
<a name="Clade"></a><div id="Clade-def"><a name="L286"></a><tt class="py-lineno"> 286</tt>  <tt class="py-line"> </tt>
<a name="L287"></a><tt class="py-lineno"> 287</tt> <a class="py-toggle" href="#" id="Clade-toggle" onclick="return toggle('Clade');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Clade-class.html">Clade</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">,</tt> <tt class="py-base-class">BaseTree</tt><tt class="py-op">.</tt><tt class="py-base-class">Clade</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Clade-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Clade-expanded"><a name="L288"></a><tt class="py-lineno"> 288</tt>  <tt class="py-line">    <tt class="py-docstring">"""Describes a branch of the current phylogenetic tree.</tt> </tt>
<a name="L289"></a><tt class="py-lineno"> 289</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L290"></a><tt class="py-lineno"> 290</tt>  <tt class="py-line"><tt class="py-docstring">    Used recursively, describes the topology of a phylogenetic tree.</tt> </tt>
<a name="L291"></a><tt class="py-lineno"> 291</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L292"></a><tt class="py-lineno"> 292</tt>  <tt class="py-line"><tt class="py-docstring">    Both ``color`` and ``width`` elements should be interpreted by client code</tt> </tt>
<a name="L293"></a><tt class="py-lineno"> 293</tt>  <tt class="py-line"><tt class="py-docstring">    as applying to the whole clade, including all descendents, unless</tt> </tt>
<a name="L294"></a><tt class="py-lineno"> 294</tt>  <tt class="py-line"><tt class="py-docstring">    overwritten in-sub clades. This module doesn't automatically assign these</tt> </tt>
<a name="L295"></a><tt class="py-lineno"> 295</tt>  <tt class="py-line"><tt class="py-docstring">    attributes to sub-clades to achieve this cascade -- and neither should you.</tt> </tt>
<a name="L296"></a><tt class="py-lineno"> 296</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L297"></a><tt class="py-lineno"> 297</tt>  <tt class="py-line"><tt class="py-docstring">    :Parameters:</tt> </tt>
<a name="L298"></a><tt class="py-lineno"> 298</tt>  <tt class="py-line"><tt class="py-docstring">        branch_length</tt> </tt>
<a name="L299"></a><tt class="py-lineno"> 299</tt>  <tt class="py-line"><tt class="py-docstring">            parent branch length of this clade</tt> </tt>
<a name="L300"></a><tt class="py-lineno"> 300</tt>  <tt class="py-line"><tt class="py-docstring">        id_source</tt> </tt>
<a name="L301"></a><tt class="py-lineno"> 301</tt>  <tt class="py-line"><tt class="py-docstring">            link other elements to a clade (on the xml-level)</tt> </tt>
<a name="L302"></a><tt class="py-lineno"> 302</tt>  <tt class="py-line"><tt class="py-docstring">        name : string</tt> </tt>
<a name="L303"></a><tt class="py-lineno"> 303</tt>  <tt class="py-line"><tt class="py-docstring">            short label for this clade</tt> </tt>
<a name="L304"></a><tt class="py-lineno"> 304</tt>  <tt class="py-line"><tt class="py-docstring">        confidences : list of Confidence objects</tt> </tt>
<a name="L305"></a><tt class="py-lineno"> 305</tt>  <tt class="py-line"><tt class="py-docstring">            used to indicate the support for a clade/parent branch.</tt> </tt>
<a name="L306"></a><tt class="py-lineno"> 306</tt>  <tt class="py-line"><tt class="py-docstring">        width : float</tt> </tt>
<a name="L307"></a><tt class="py-lineno"> 307</tt>  <tt class="py-line"><tt class="py-docstring">            branch width for this clade (including branch from parent)</tt> </tt>
<a name="L308"></a><tt class="py-lineno"> 308</tt>  <tt class="py-line"><tt class="py-docstring">        color : BranchColor</tt> </tt>
<a name="L309"></a><tt class="py-lineno"> 309</tt>  <tt class="py-line"><tt class="py-docstring">            color used for graphical display of this clade</tt> </tt>
<a name="L310"></a><tt class="py-lineno"> 310</tt>  <tt class="py-line"><tt class="py-docstring">        node_id</tt> </tt>
<a name="L311"></a><tt class="py-lineno"> 311</tt>  <tt class="py-line"><tt class="py-docstring">            unique identifier for the root node of this clade</tt> </tt>
<a name="L312"></a><tt class="py-lineno"> 312</tt>  <tt class="py-line"><tt class="py-docstring">        taxonomies : list</tt> </tt>
<a name="L313"></a><tt class="py-lineno"> 313</tt>  <tt class="py-line"><tt class="py-docstring">            Taxonomy objects</tt> </tt>
<a name="L314"></a><tt class="py-lineno"> 314</tt>  <tt class="py-line"><tt class="py-docstring">        sequences : list</tt> </tt>
<a name="L315"></a><tt class="py-lineno"> 315</tt>  <tt class="py-line"><tt class="py-docstring">            Sequence objects</tt> </tt>
<a name="L316"></a><tt class="py-lineno"> 316</tt>  <tt class="py-line"><tt class="py-docstring">        events : Events</tt> </tt>
<a name="L317"></a><tt class="py-lineno"> 317</tt>  <tt class="py-line"><tt class="py-docstring">            describe such events as gene-duplications at the root node/parent</tt> </tt>
<a name="L318"></a><tt class="py-lineno"> 318</tt>  <tt class="py-line"><tt class="py-docstring">            branch of this clade</tt> </tt>
<a name="L319"></a><tt class="py-lineno"> 319</tt>  <tt class="py-line"><tt class="py-docstring">        binary_characters : BinaryCharacters</tt> </tt>
<a name="L320"></a><tt class="py-lineno"> 320</tt>  <tt class="py-line"><tt class="py-docstring">            binary characters</tt> </tt>
<a name="L321"></a><tt class="py-lineno"> 321</tt>  <tt class="py-line"><tt class="py-docstring">        distributions : list of Distribution objects</tt> </tt>
<a name="L322"></a><tt class="py-lineno"> 322</tt>  <tt class="py-line"><tt class="py-docstring">            distribution(s) of this clade</tt> </tt>
<a name="L323"></a><tt class="py-lineno"> 323</tt>  <tt class="py-line"><tt class="py-docstring">        date : Date</tt> </tt>
<a name="L324"></a><tt class="py-lineno"> 324</tt>  <tt class="py-line"><tt class="py-docstring">            a date for the root node of this clade</tt> </tt>
<a name="L325"></a><tt class="py-lineno"> 325</tt>  <tt class="py-line"><tt class="py-docstring">        references : list</tt> </tt>
<a name="L326"></a><tt class="py-lineno"> 326</tt>  <tt class="py-line"><tt class="py-docstring">            Reference objects</tt> </tt>
<a name="L327"></a><tt class="py-lineno"> 327</tt>  <tt class="py-line"><tt class="py-docstring">        properties : list</tt> </tt>
<a name="L328"></a><tt class="py-lineno"> 328</tt>  <tt class="py-line"><tt class="py-docstring">            Property objects</tt> </tt>
<a name="L329"></a><tt class="py-lineno"> 329</tt>  <tt class="py-line"><tt class="py-docstring">        clades : list Clade objects</tt> </tt>
<a name="L330"></a><tt class="py-lineno"> 330</tt>  <tt class="py-line"><tt class="py-docstring">            Sub-clades</tt> </tt>
<a name="L331"></a><tt class="py-lineno"> 331</tt>  <tt class="py-line"><tt class="py-docstring">        other : list of Other objects</tt> </tt>
<a name="L332"></a><tt class="py-lineno"> 332</tt>  <tt class="py-line"><tt class="py-docstring">            non-phyloXML objects</tt> </tt>
<a name="L333"></a><tt class="py-lineno"> 333</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L334"></a><tt class="py-lineno"> 334</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L335"></a><tt class="py-lineno"> 335</tt>  <tt class="py-line"> </tt>
<a name="Clade.__init__"></a><div id="Clade.__init__-def"><a name="L336"></a><tt class="py-lineno"> 336</tt> <a class="py-toggle" href="#" id="Clade.__init__-toggle" onclick="return toggle('Clade.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Clade-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> </tt>
<a name="L337"></a><tt class="py-lineno"> 337</tt>  <tt class="py-line">                 <tt class="py-comment"># Attributes</tt> </tt>
<a name="L338"></a><tt class="py-lineno"> 338</tt>  <tt class="py-line">                 <tt class="py-param">branch_length</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">id_source</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L339"></a><tt class="py-lineno"> 339</tt>  <tt class="py-line">                 <tt class="py-comment"># Child nodes</tt> </tt>
<a name="L340"></a><tt class="py-lineno"> 340</tt>  <tt class="py-line">                 <tt class="py-param">name</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">width</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">color</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">node_id</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">events</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L341"></a><tt class="py-lineno"> 341</tt>  <tt class="py-line">                 <tt class="py-param">binary_characters</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">date</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L342"></a><tt class="py-lineno"> 342</tt>  <tt class="py-line">                 <tt class="py-comment"># Collections</tt> </tt>
<a name="L343"></a><tt class="py-lineno"> 343</tt>  <tt class="py-line">                 <tt class="py-param">confidences</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">taxonomies</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">sequences</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L344"></a><tt class="py-lineno"> 344</tt>  <tt class="py-line">                 <tt class="py-param">distributions</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">references</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">properties</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">clades</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L345"></a><tt class="py-lineno"> 345</tt>  <tt class="py-line">                 <tt class="py-param">other</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L346"></a><tt class="py-lineno"> 346</tt>  <tt class="py-line">                 <tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Clade.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Clade.__init__-expanded"><a name="L347"></a><tt class="py-lineno"> 347</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize value for the Clade object."""</tt> </tt>
<a name="L348"></a><tt class="py-lineno"> 348</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-99" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.branch_length()=Bio.Phylo.PhyloXMLIO.Writer-class.html#branch_length"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-99', 'branch_length', 'link-99');">branch_length</a></tt> <tt class="py-op">=</tt> <tt id="link-100" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-100', 'branch_length', 'link-99');">branch_length</a></tt> </tt>
<a name="L349"></a><tt class="py-lineno"> 349</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">id_source</tt> <tt class="py-op">=</tt> <tt class="py-name">id_source</tt> </tt>
<a name="L350"></a><tt class="py-lineno"> 350</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-101" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-101', 'name', 'link-31');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-102" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-102', 'name', 'link-31');">name</a></tt> </tt>
<a name="L351"></a><tt class="py-lineno"> 351</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-103" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.width=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#width,Method Bio.Phylo.PhyloXMLIO.Writer.width()=Bio.Phylo.PhyloXMLIO.Writer-class.html#width"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.width
Bio.Phylo.PhyloXMLIO.Writer.width" class="py-name" href="#" onclick="return doclink('link-103', 'width', 'link-103');">width</a></tt> <tt class="py-op">=</tt> <tt id="link-104" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.width
Bio.Phylo.PhyloXMLIO.Writer.width" class="py-name" href="#" onclick="return doclink('link-104', 'width', 'link-103');">width</a></tt> </tt>
<a name="L352"></a><tt class="py-lineno"> 352</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-105" class="py-name" targets="Variable Bio.Phylo.BaseTree.Clade.color=Bio.Phylo.BaseTree.Clade-class.html#color,Method Bio.Phylo.PhyloXMLIO.Parser.color()=Bio.Phylo.PhyloXMLIO.Parser-class.html#color,Method Bio.Phylo.PhyloXMLIO.Writer.color()=Bio.Phylo.PhyloXMLIO.Writer-class.html#color"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-105', 'color', 'link-105');">color</a></tt> <tt class="py-op">=</tt> <tt id="link-106" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-106', 'color', 'link-105');">color</a></tt> </tt>
<a name="L353"></a><tt class="py-lineno"> 353</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-107" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.node_id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#node_id"><a title="Bio.Phylo.PhyloXMLIO.Writer.node_id" class="py-name" href="#" onclick="return doclink('link-107', 'node_id', 'link-107');">node_id</a></tt> <tt class="py-op">=</tt> <tt id="link-108" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.node_id" class="py-name" href="#" onclick="return doclink('link-108', 'node_id', 'link-107');">node_id</a></tt> </tt>
<a name="L354"></a><tt class="py-lineno"> 354</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-109" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.events()=Bio.Phylo.PhyloXMLIO.Parser-class.html#events,Method Bio.Phylo.PhyloXMLIO.Writer.events()=Bio.Phylo.PhyloXMLIO.Writer-class.html#events"><a title="Bio.Phylo.PhyloXMLIO.Parser.events
Bio.Phylo.PhyloXMLIO.Writer.events" class="py-name" href="#" onclick="return doclink('link-109', 'events', 'link-109');">events</a></tt> <tt class="py-op">=</tt> <tt id="link-110" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.events
Bio.Phylo.PhyloXMLIO.Writer.events" class="py-name" href="#" onclick="return doclink('link-110', 'events', 'link-109');">events</a></tt> </tt>
<a name="L355"></a><tt class="py-lineno"> 355</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-111" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.binary_characters()=Bio.Phylo.PhyloXMLIO.Parser-class.html#binary_characters,Method Bio.Phylo.PhyloXMLIO.Writer.binary_characters()=Bio.Phylo.PhyloXMLIO.Writer-class.html#binary_characters"><a title="Bio.Phylo.PhyloXMLIO.Parser.binary_characters
Bio.Phylo.PhyloXMLIO.Writer.binary_characters" class="py-name" href="#" onclick="return doclink('link-111', 'binary_characters', 'link-111');">binary_characters</a></tt> <tt class="py-op">=</tt> <tt id="link-112" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.binary_characters
Bio.Phylo.PhyloXMLIO.Writer.binary_characters" class="py-name" href="#" onclick="return doclink('link-112', 'binary_characters', 'link-111');">binary_characters</a></tt> </tt>
<a name="L356"></a><tt class="py-lineno"> 356</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-113" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-113', 'date', 'link-52');">date</a></tt> <tt class="py-op">=</tt> <tt id="link-114" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-114', 'date', 'link-52');">date</a></tt> </tt>
<a name="L357"></a><tt class="py-lineno"> 357</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">confidences</tt> <tt class="py-op">=</tt> <tt class="py-name">confidences</tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L358"></a><tt class="py-lineno"> 358</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">taxonomies</tt> <tt class="py-op">=</tt> <tt class="py-name">taxonomies</tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L359"></a><tt class="py-lineno"> 359</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">sequences</tt> <tt class="py-op">=</tt> <tt class="py-name">sequences</tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L360"></a><tt class="py-lineno"> 360</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">distributions</tt> <tt class="py-op">=</tt> <tt class="py-name">distributions</tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L361"></a><tt class="py-lineno"> 361</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">references</tt> <tt class="py-op">=</tt> <tt class="py-name">references</tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L362"></a><tt class="py-lineno"> 362</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">properties</tt> <tt class="py-op">=</tt> <tt class="py-name">properties</tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L363"></a><tt class="py-lineno"> 363</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt> <tt class="py-op">=</tt> <tt class="py-name">clades</tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L364"></a><tt class="py-lineno"> 364</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-115" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-115', 'other', 'link-24');">other</a></tt> <tt class="py-op">=</tt> <tt id="link-116" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-116', 'other', 'link-24');">other</a></tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div><a name="L365"></a><tt class="py-lineno"> 365</tt>  <tt class="py-line"> </tt>
<a name="L366"></a><tt class="py-lineno"> 366</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">classmethod</tt> </tt>
<a name="Clade.from_clade"></a><div id="Clade.from_clade-def"><a name="L367"></a><tt class="py-lineno"> 367</tt> <a class="py-toggle" href="#" id="Clade.from_clade-toggle" onclick="return toggle('Clade.from_clade');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Clade-class.html#from_clade">from_clade</a><tt class="py-op">(</tt><tt class="py-param">cls</tt><tt class="py-op">,</tt> <tt class="py-param">clade</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Clade.from_clade-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Clade.from_clade-expanded"><a name="L368"></a><tt class="py-lineno"> 368</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a new PhyloXML Clade from a Newick or BaseTree Clade object.</tt> </tt>
<a name="L369"></a><tt class="py-lineno"> 369</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L370"></a><tt class="py-lineno"> 370</tt>  <tt class="py-line"><tt class="py-docstring">        Keyword arguments are the usual PhyloXML Clade constructor parameters.</tt> </tt>
<a name="L371"></a><tt class="py-lineno"> 371</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L372"></a><tt class="py-lineno"> 372</tt>  <tt class="py-line">        <tt id="link-117" class="py-name" targets="Method Bio.Phylo.CDAOIO.Parser.new_clade()=Bio.Phylo.CDAOIO.Parser-class.html#new_clade,Method Bio.Phylo.NewickIO.Parser.new_clade()=Bio.Phylo.NewickIO.Parser-class.html#new_clade"><a title="Bio.Phylo.CDAOIO.Parser.new_clade
Bio.Phylo.NewickIO.Parser.new_clade" class="py-name" href="#" onclick="return doclink('link-117', 'new_clade', 'link-117');">new_clade</a></tt> <tt class="py-op">=</tt> <tt class="py-name">cls</tt><tt class="py-op">(</tt><tt id="link-118" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-118', 'branch_length', 'link-99');">branch_length</a></tt><tt class="py-op">=</tt><tt id="link-119" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-119', 'clade', 'link-69');">clade</a></tt><tt class="py-op">.</tt><tt id="link-120" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-120', 'branch_length', 'link-99');">branch_length</a></tt><tt class="py-op">,</tt> </tt>
<a name="L373"></a><tt class="py-lineno"> 373</tt>  <tt class="py-line">                        <tt id="link-121" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-121', 'name', 'link-31');">name</a></tt><tt class="py-op">=</tt><tt id="link-122" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-122', 'clade', 'link-69');">clade</a></tt><tt class="py-op">.</tt><tt id="link-123" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-123', 'name', 'link-31');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L374"></a><tt class="py-lineno"> 374</tt>  <tt class="py-line">        <tt id="link-124" class="py-name"><a title="Bio.Phylo.CDAOIO.Parser.new_clade
Bio.Phylo.NewickIO.Parser.new_clade" class="py-name" href="#" onclick="return doclink('link-124', 'new_clade', 'link-117');">new_clade</a></tt><tt class="py-op">.</tt><tt class="py-name">clades</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">cls</tt><tt class="py-op">.</tt><tt id="link-125" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.from_clade
Bio.Phylo.PhyloXML.Clade.from_clade
Bio.Phylo.PhyloXML.Phylogeny.from_clade" class="py-name" href="#" onclick="return doclink('link-125', 'from_clade', 'link-58');">from_clade</a></tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">c</tt> <tt class="py-keyword">in</tt> <tt id="link-126" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-126', 'clade', 'link-69');">clade</a></tt><tt class="py-op">]</tt> </tt>
<a name="L375"></a><tt class="py-lineno"> 375</tt>  <tt class="py-line">        <tt id="link-127" class="py-name"><a title="Bio.Phylo.CDAOIO.Parser.new_clade
Bio.Phylo.NewickIO.Parser.new_clade" class="py-name" href="#" onclick="return doclink('link-127', 'new_clade', 'link-117');">new_clade</a></tt><tt class="py-op">.</tt><tt id="link-128" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-128', 'confidence', 'link-94');">confidence</a></tt> <tt class="py-op">=</tt> <tt id="link-129" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-129', 'clade', 'link-69');">clade</a></tt><tt class="py-op">.</tt><tt id="link-130" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-130', 'confidence', 'link-94');">confidence</a></tt> </tt>
<a name="L376"></a><tt class="py-lineno"> 376</tt>  <tt class="py-line">        <tt id="link-131" class="py-name"><a title="Bio.Phylo.CDAOIO.Parser.new_clade
Bio.Phylo.NewickIO.Parser.new_clade" class="py-name" href="#" onclick="return doclink('link-131', 'new_clade', 'link-117');">new_clade</a></tt><tt class="py-op">.</tt><tt id="link-132" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.width
Bio.Phylo.PhyloXMLIO.Writer.width" class="py-name" href="#" onclick="return doclink('link-132', 'width', 'link-103');">width</a></tt> <tt class="py-op">=</tt> <tt id="link-133" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-133', 'clade', 'link-69');">clade</a></tt><tt class="py-op">.</tt><tt id="link-134" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.width
Bio.Phylo.PhyloXMLIO.Writer.width" class="py-name" href="#" onclick="return doclink('link-134', 'width', 'link-103');">width</a></tt> </tt>
<a name="L377"></a><tt class="py-lineno"> 377</tt>  <tt class="py-line">        <tt id="link-135" class="py-name"><a title="Bio.Phylo.CDAOIO.Parser.new_clade
Bio.Phylo.NewickIO.Parser.new_clade" class="py-name" href="#" onclick="return doclink('link-135', 'new_clade', 'link-117');">new_clade</a></tt><tt class="py-op">.</tt><tt id="link-136" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-136', 'color', 'link-105');">color</a></tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt id="link-137" class="py-name" targets="Class Bio.Phylo.BaseTree.BranchColor=Bio.Phylo.BaseTree.BranchColor-class.html,Class Bio.Phylo.PhyloXML.BranchColor=Bio.Phylo.PhyloXML.BranchColor-class.html"><a title="Bio.Phylo.BaseTree.BranchColor
Bio.Phylo.PhyloXML.BranchColor" class="py-name" href="#" onclick="return doclink('link-137', 'BranchColor', 'link-137');">BranchColor</a></tt><tt class="py-op">(</tt> </tt>
<a name="L378"></a><tt class="py-lineno"> 378</tt>  <tt class="py-line">            <tt id="link-138" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-138', 'clade', 'link-69');">clade</a></tt><tt class="py-op">.</tt><tt id="link-139" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-139', 'color', 'link-105');">color</a></tt><tt class="py-op">.</tt><tt id="link-140" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.red()=Bio.Phylo.PhyloXMLIO.Writer-class.html#red"><a title="Bio.Phylo.PhyloXMLIO.Writer.red" class="py-name" href="#" onclick="return doclink('link-140', 'red', 'link-140');">red</a></tt><tt class="py-op">,</tt> <tt id="link-141" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-141', 'clade', 'link-69');">clade</a></tt><tt class="py-op">.</tt><tt id="link-142" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-142', 'color', 'link-105');">color</a></tt><tt class="py-op">.</tt><tt id="link-143" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.green()=Bio.Phylo.PhyloXMLIO.Writer-class.html#green"><a title="Bio.Phylo.PhyloXMLIO.Writer.green" class="py-name" href="#" onclick="return doclink('link-143', 'green', 'link-143');">green</a></tt><tt class="py-op">,</tt> <tt id="link-144" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-144', 'clade', 'link-69');">clade</a></tt><tt class="py-op">.</tt><tt id="link-145" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-145', 'color', 'link-105');">color</a></tt><tt class="py-op">.</tt><tt id="link-146" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.blue()=Bio.Phylo.PhyloXMLIO.Writer-class.html#blue"><a title="Bio.Phylo.PhyloXMLIO.Writer.blue" class="py-name" href="#" onclick="return doclink('link-146', 'blue', 'link-146');">blue</a></tt><tt class="py-op">)</tt> </tt>
<a name="L379"></a><tt class="py-lineno"> 379</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-147" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-147', 'clade', 'link-69');">clade</a></tt><tt class="py-op">.</tt><tt id="link-148" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-148', 'color', 'link-105');">color</a></tt> <tt class="py-keyword">else</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
<a name="L380"></a><tt class="py-lineno"> 380</tt>  <tt class="py-line">        <tt id="link-149" class="py-name"><a title="Bio.Phylo.CDAOIO.Parser.new_clade
Bio.Phylo.NewickIO.Parser.new_clade" class="py-name" href="#" onclick="return doclink('link-149', 'new_clade', 'link-117');">new_clade</a></tt><tt class="py-op">.</tt><tt class="py-name">__dict__</tt><tt class="py-op">.</tt><tt id="link-150" class="py-name"><a title="Bio.File._IndexedSeqFileDict.update
Bio.Index._InMemoryIndex.update
Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.update
Bio.SeqRecord._RestrictedDict.update" class="py-name" href="#" onclick="return doclink('link-150', 'update', 'link-23');">update</a></tt><tt class="py-op">(</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
<a name="L381"></a><tt class="py-lineno"> 381</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-151" class="py-name"><a title="Bio.Phylo.CDAOIO.Parser.new_clade
Bio.Phylo.NewickIO.Parser.new_clade" class="py-name" href="#" onclick="return doclink('link-151', 'new_clade', 'link-117');">new_clade</a></tt> </tt>
</div><a name="L382"></a><tt class="py-lineno"> 382</tt>  <tt class="py-line"> </tt>
<a name="Clade.to_phylogeny"></a><div id="Clade.to_phylogeny-def"><a name="L383"></a><tt class="py-lineno"> 383</tt> <a class="py-toggle" href="#" id="Clade.to_phylogeny-toggle" onclick="return toggle('Clade.to_phylogeny');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Clade-class.html#to_phylogeny">to_phylogeny</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Clade.to_phylogeny-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Clade.to_phylogeny-expanded"><a name="L384"></a><tt class="py-lineno"> 384</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a new phylogeny containing just this clade."""</tt> </tt>
<a name="L385"></a><tt class="py-lineno"> 385</tt>  <tt class="py-line">        <tt class="py-name">phy</tt> <tt class="py-op">=</tt> <tt id="link-152" class="py-name" targets="Class Bio.Phylo.PhyloXML.Phylogeny=Bio.Phylo.PhyloXML.Phylogeny-class.html"><a title="Bio.Phylo.PhyloXML.Phylogeny" class="py-name" href="#" onclick="return doclink('link-152', 'Phylogeny', 'link-152');">Phylogeny</a></tt><tt class="py-op">(</tt><tt id="link-153" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-153', 'root', 'link-42');">root</a></tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt id="link-154" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-154', 'date', 'link-52');">date</a></tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-155" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-155', 'date', 'link-52');">date</a></tt><tt class="py-op">)</tt> </tt>
<a name="L386"></a><tt class="py-lineno"> 386</tt>  <tt class="py-line">        <tt class="py-name">phy</tt><tt class="py-op">.</tt><tt class="py-name">__dict__</tt><tt class="py-op">.</tt><tt id="link-156" class="py-name"><a title="Bio.File._IndexedSeqFileDict.update
Bio.Index._InMemoryIndex.update
Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.update
Bio.SeqRecord._RestrictedDict.update" class="py-name" href="#" onclick="return doclink('link-156', 'update', 'link-23');">update</a></tt><tt class="py-op">(</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
<a name="L387"></a><tt class="py-lineno"> 387</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">phy</tt> </tt>
</div><a name="L388"></a><tt class="py-lineno"> 388</tt>  <tt class="py-line"> </tt>
<a name="L389"></a><tt class="py-lineno"> 389</tt>  <tt class="py-line">    <tt class="py-comment"># Shortcuts for list attributes that are usually only 1 item</tt> </tt>
<a name="L390"></a><tt class="py-lineno"> 390</tt>  <tt class="py-line">    <tt class="py-comment"># NB: Duplicated from Phylogeny class</tt> </tt>
<a name="Clade._get_confidence"></a><div id="Clade._get_confidence-def"><a name="L391"></a><tt class="py-lineno"> 391</tt> <a class="py-toggle" href="#" id="Clade._get_confidence-toggle" onclick="return toggle('Clade._get_confidence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Clade-class.html#_get_confidence">_get_confidence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Clade._get_confidence-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Clade._get_confidence-expanded"><a name="L392"></a><tt class="py-lineno"> 392</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">confidences</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L393"></a><tt class="py-lineno"> 393</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
<a name="L394"></a><tt class="py-lineno"> 394</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">confidences</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L395"></a><tt class="py-lineno"> 395</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">(</tt><tt class="py-string">"more than 1 confidence value available; "</tt> </tt>
<a name="L396"></a><tt class="py-lineno"> 396</tt>  <tt class="py-line">                                 <tt class="py-string">"use Clade.confidences"</tt><tt class="py-op">)</tt> </tt>
<a name="L397"></a><tt class="py-lineno"> 397</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">confidences</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
</div><a name="L398"></a><tt class="py-lineno"> 398</tt>  <tt class="py-line"> </tt>
<a name="Clade._set_confidence"></a><div id="Clade._set_confidence-def"><a name="L399"></a><tt class="py-lineno"> 399</tt> <a class="py-toggle" href="#" id="Clade._set_confidence-toggle" onclick="return toggle('Clade._set_confidence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Clade-class.html#_set_confidence">_set_confidence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Clade._set_confidence-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Clade._set_confidence-expanded"><a name="L400"></a><tt class="py-lineno"> 400</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-157" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-157', 'value', 'link-37');">value</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L401"></a><tt class="py-lineno"> 401</tt>  <tt class="py-line">            <tt class="py-comment"># Special case: mirror the behavior of _get_confidence</tt> </tt>
<a name="L402"></a><tt class="py-lineno"> 402</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">confidences</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L403"></a><tt class="py-lineno"> 403</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L404"></a><tt class="py-lineno"> 404</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-158" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-158', 'value', 'link-37');">value</a></tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">float</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L405"></a><tt class="py-lineno"> 405</tt>  <tt class="py-line">            <tt id="link-159" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-159', 'value', 'link-37');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-160" class="py-name"><a title="Bio.Phylo.PhyloXML.Confidence" class="py-name" href="#" onclick="return doclink('link-160', 'Confidence', 'link-87');">Confidence</a></tt><tt class="py-op">(</tt><tt id="link-161" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-161', 'value', 'link-37');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L406"></a><tt class="py-lineno"> 406</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-162" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-162', 'value', 'link-37');">value</a></tt><tt class="py-op">,</tt> <tt id="link-163" class="py-name"><a title="Bio.Phylo.PhyloXML.Confidence" class="py-name" href="#" onclick="return doclink('link-163', 'Confidence', 'link-87');">Confidence</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L407"></a><tt class="py-lineno"> 407</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"value must be a number or Confidence instance"</tt><tt class="py-op">)</tt> </tt>
<a name="L408"></a><tt class="py-lineno"> 408</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">confidences</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L409"></a><tt class="py-lineno"> 409</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">confidences</tt><tt class="py-op">.</tt><tt id="link-164" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-164', 'append', 'link-91');">append</a></tt><tt class="py-op">(</tt><tt id="link-165" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-165', 'value', 'link-37');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L410"></a><tt class="py-lineno"> 410</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">confidences</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L411"></a><tt class="py-lineno"> 411</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">confidences</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-166" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-166', 'value', 'link-37');">value</a></tt> </tt>
<a name="L412"></a><tt class="py-lineno"> 412</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L413"></a><tt class="py-lineno"> 413</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"multiple confidence values already exist; "</tt> </tt>
<a name="L414"></a><tt class="py-lineno"> 414</tt>  <tt class="py-line">                             <tt class="py-string">"use Phylogeny.confidences instead"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L415"></a><tt class="py-lineno"> 415</tt>  <tt class="py-line"> </tt>
<a name="Clade._del_confidence"></a><div id="Clade._del_confidence-def"><a name="L416"></a><tt class="py-lineno"> 416</tt> <a class="py-toggle" href="#" id="Clade._del_confidence-toggle" onclick="return toggle('Clade._del_confidence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Clade-class.html#_del_confidence">_del_confidence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Clade._del_confidence-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Clade._del_confidence-expanded"><a name="L417"></a><tt class="py-lineno"> 417</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">confidences</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div><a name="L418"></a><tt class="py-lineno"> 418</tt>  <tt class="py-line"> </tt>
<a name="L419"></a><tt class="py-lineno"> 419</tt>  <tt class="py-line">    <tt id="link-167" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-167', 'confidence', 'link-94');">confidence</a></tt> <tt class="py-op">=</tt> <tt id="link-168" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-168', 'property', 'link-95');">property</a></tt><tt class="py-op">(</tt><tt id="link-169" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade._get_confidence
Bio.Phylo.PhyloXML.Phylogeny._get_confidence" class="py-name" href="#" onclick="return doclink('link-169', '_get_confidence', 'link-96');">_get_confidence</a></tt><tt class="py-op">,</tt> <tt id="link-170" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade._set_confidence
Bio.Phylo.PhyloXML.Phylogeny._set_confidence" class="py-name" href="#" onclick="return doclink('link-170', '_set_confidence', 'link-97');">_set_confidence</a></tt><tt class="py-op">,</tt> <tt id="link-171" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade._del_confidence
Bio.Phylo.PhyloXML.Phylogeny._del_confidence" class="py-name" href="#" onclick="return doclink('link-171', '_del_confidence', 'link-98');">_del_confidence</a></tt><tt class="py-op">)</tt> </tt>
<a name="L420"></a><tt class="py-lineno"> 420</tt>  <tt class="py-line"> </tt>
<a name="Clade._get_taxonomy"></a><div id="Clade._get_taxonomy-def"><a name="L421"></a><tt class="py-lineno"> 421</tt> <a class="py-toggle" href="#" id="Clade._get_taxonomy-toggle" onclick="return toggle('Clade._get_taxonomy');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Clade-class.html#_get_taxonomy">_get_taxonomy</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Clade._get_taxonomy-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Clade._get_taxonomy-expanded"><a name="L422"></a><tt class="py-lineno"> 422</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">taxonomies</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L423"></a><tt class="py-lineno"> 423</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
<a name="L424"></a><tt class="py-lineno"> 424</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">taxonomies</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L425"></a><tt class="py-lineno"> 425</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">(</tt><tt class="py-string">"more than 1 taxonomy value available; "</tt> </tt>
<a name="L426"></a><tt class="py-lineno"> 426</tt>  <tt class="py-line">                                 <tt class="py-string">"use Clade.taxonomies"</tt><tt class="py-op">)</tt> </tt>
<a name="L427"></a><tt class="py-lineno"> 427</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">taxonomies</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
</div><a name="L428"></a><tt class="py-lineno"> 428</tt>  <tt class="py-line"> </tt>
<a name="Clade._set_taxonomy"></a><div id="Clade._set_taxonomy-def"><a name="L429"></a><tt class="py-lineno"> 429</tt> <a class="py-toggle" href="#" id="Clade._set_taxonomy-toggle" onclick="return toggle('Clade._set_taxonomy');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Clade-class.html#_set_taxonomy">_set_taxonomy</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Clade._set_taxonomy-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Clade._set_taxonomy-expanded"><a name="L430"></a><tt class="py-lineno"> 430</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-172" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-172', 'value', 'link-37');">value</a></tt><tt class="py-op">,</tt> <tt id="link-173" class="py-name" targets="Class Bio.Phylo.PhyloXML.Taxonomy=Bio.Phylo.PhyloXML.Taxonomy-class.html"><a title="Bio.Phylo.PhyloXML.Taxonomy" class="py-name" href="#" onclick="return doclink('link-173', 'Taxonomy', 'link-173');">Taxonomy</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L431"></a><tt class="py-lineno"> 431</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"assigned value must be a Taxonomy instance"</tt><tt class="py-op">)</tt> </tt>
<a name="L432"></a><tt class="py-lineno"> 432</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">taxonomies</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L433"></a><tt class="py-lineno"> 433</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">taxonomies</tt><tt class="py-op">.</tt><tt id="link-174" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-174', 'append', 'link-91');">append</a></tt><tt class="py-op">(</tt><tt id="link-175" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-175', 'value', 'link-37');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L434"></a><tt class="py-lineno"> 434</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">taxonomies</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L435"></a><tt class="py-lineno"> 435</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">taxonomies</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-176" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-176', 'value', 'link-37');">value</a></tt> </tt>
<a name="L436"></a><tt class="py-lineno"> 436</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L437"></a><tt class="py-lineno"> 437</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"multiple taxonomy values already exist; "</tt> </tt>
<a name="L438"></a><tt class="py-lineno"> 438</tt>  <tt class="py-line">                             <tt class="py-string">"use Phylogeny.taxonomies instead"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L439"></a><tt class="py-lineno"> 439</tt>  <tt class="py-line"> </tt>
<a name="L440"></a><tt class="py-lineno"> 440</tt>  <tt class="py-line">    <tt id="link-177" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.taxonomy()=Bio.GenBank._FeatureConsumer-class.html#taxonomy,Method Bio.GenBank._RecordConsumer.taxonomy()=Bio.GenBank._RecordConsumer-class.html#taxonomy,Variable Bio.Phylo.PhyloXML.Clade.taxonomy=Bio.Phylo.PhyloXML.Clade-class.html#taxonomy,Method Bio.Phylo.PhyloXMLIO.Writer.taxonomy()=Bio.Phylo.PhyloXMLIO.Writer-class.html#taxonomy"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-177', 'taxonomy', 'link-177');">taxonomy</a></tt> <tt class="py-op">=</tt> <tt id="link-178" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-178', 'property', 'link-95');">property</a></tt><tt class="py-op">(</tt><tt id="link-179" class="py-name" targets="Method Bio.Phylo.PhyloXML.Clade._get_taxonomy()=Bio.Phylo.PhyloXML.Clade-class.html#_get_taxonomy"><a title="Bio.Phylo.PhyloXML.Clade._get_taxonomy" class="py-name" href="#" onclick="return doclink('link-179', '_get_taxonomy', 'link-179');">_get_taxonomy</a></tt><tt class="py-op">,</tt> <tt id="link-180" class="py-name" targets="Method Bio.Phylo.PhyloXML.Clade._set_taxonomy()=Bio.Phylo.PhyloXML.Clade-class.html#_set_taxonomy"><a title="Bio.Phylo.PhyloXML.Clade._set_taxonomy" class="py-name" href="#" onclick="return doclink('link-180', '_set_taxonomy', 'link-180');">_set_taxonomy</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L441"></a><tt class="py-lineno"> 441</tt>  <tt class="py-line"> </tt>
<a name="BranchColor"></a><div id="BranchColor-def"><a name="L442"></a><tt class="py-lineno"> 442</tt>  <tt class="py-line"> </tt>
<a name="L443"></a><tt class="py-lineno"> 443</tt>  <tt class="py-line"><tt class="py-comment"># PhyloXML wrapper for a special BaseTree attribute</tt> </tt>
<a name="L444"></a><tt class="py-lineno"> 444</tt>  <tt class="py-line"> </tt>
<a name="L445"></a><tt class="py-lineno"> 445</tt> <a class="py-toggle" href="#" id="BranchColor-toggle" onclick="return toggle('BranchColor');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.BranchColor-class.html">BranchColor</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">,</tt> <tt class="py-base-class">BaseTree</tt><tt class="py-op">.</tt><tt class="py-base-class">BranchColor</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BranchColor-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="BranchColor-expanded"><a name="L446"></a><tt class="py-lineno"> 446</tt>  <tt class="py-line"> </tt>
<a name="BranchColor.__init__"></a><div id="BranchColor.__init__-def"><a name="L447"></a><tt class="py-lineno"> 447</tt> <a class="py-toggle" href="#" id="BranchColor.__init__-toggle" onclick="return toggle('BranchColor.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.BranchColor-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-op">*</tt><tt class="py-param">args</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BranchColor.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="BranchColor.__init__-expanded"><a name="L448"></a><tt class="py-lineno"> 448</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize parameters for the BranchColor object."""</tt> </tt>
<a name="L449"></a><tt class="py-lineno"> 449</tt>  <tt class="py-line">        <tt id="link-181" class="py-name"><a title="Bio.Phylo.BaseTree" class="py-name" href="#" onclick="return doclink('link-181', 'BaseTree', 'link-21');">BaseTree</a></tt><tt class="py-op">.</tt><tt id="link-182" class="py-name"><a title="Bio.Phylo.BaseTree.BranchColor
Bio.Phylo.PhyloXML.BranchColor" class="py-name" href="#" onclick="return doclink('link-182', 'BranchColor', 'link-137');">BranchColor</a></tt><tt class="py-op">.</tt><tt id="link-183" class="py-name" targets="Method Bio.Affy.CelFile.ParserError.__init__()=Bio.Affy.CelFile.ParserError-class.html#__init__,Method Bio.Affy.CelFile.Record.__init__()=Bio.Affy.CelFile.Record-class.html#__init__,Method Bio.Align.AlignInfo.PSSM.__init__()=Bio.Align.AlignInfo.PSSM-class.html#__init__,Method Bio.Align.AlignInfo.SummaryInfo.__init__()=Bio.Align.AlignInfo.SummaryInfo-class.html#__init__,Method Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline.__init__()=Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline-class.html#__init__,Method Bio.Align.Applications._Clustalw.ClustalwCommandline.__init__()=Bio.Align.Applications._Clustalw.ClustalwCommandline-class.html#__init__,Method Bio.Align.Applications._Dialign.DialignCommandline.__init__()=Bio.Align.Applications._Dialign.DialignCommandline-class.html#__init__,Method Bio.Align.Applications._MSAProbs.MSAProbsCommandline.__init__()=Bio.Align.Applications._MSAProbs.MSAProbsCommandline-class.html#__init__,Method Bio.Align.Applications._Mafft.MafftCommandline.__init__()=Bio.Align.Applications._Mafft.MafftCommandline-class.html#__init__,Method Bio.Align.Applications._Muscle.MuscleCommandline.__init__()=Bio.Align.Applications._Muscle.MuscleCommandline-class.html#__init__,Method Bio.Align.Applications._Prank.PrankCommandline.__init__()=Bio.Align.Applications._Prank.PrankCommandline-class.html#__init__,Method Bio.Align.Applications._Probcons.ProbconsCommandline.__init__()=Bio.Align.Applications._Probcons.ProbconsCommandline-class.html#__init__,Method Bio.Align.Applications._TCoffee.TCoffeeCommandline.__init__()=Bio.Align.Applications._TCoffee.TCoffeeCommandline-class.html#__init__,Method Bio.Align.MultipleSeqAlignment.__init__()=Bio.Align.MultipleSeqAlignment-class.html#__init__,Method Bio.Align.PairwiseAlignment.__init__()=Bio.Align.PairwiseAlignment-class.html#__init__,Method Bio.Align.PairwiseAlignments.__init__()=Bio.Align.PairwiseAlignments-class.html#__init__,Method Bio.AlignIO.Interfaces.AlignmentIterator.__init__()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#__init__,Method Bio.AlignIO.Interfaces.AlignmentWriter.__init__()=Bio.AlignIO.Interfaces.AlignmentWriter-class.html#__init__,Method Bio.AlignIO.Interfaces.SequentialAlignmentWriter.__init__()=Bio.AlignIO.Interfaces.SequentialAlignmentWriter-class.html#__init__,Method Bio.AlignIO.MafIO.MafIndex.__init__()=Bio.AlignIO.MafIO.MafIndex-class.html#__init__,Method Bio.AlignIO.MauveIO.MauveWriter.__init__()=Bio.AlignIO.MauveIO.MauveWriter-class.html#__init__,Method Bio.Alphabet.AlphabetEncoder.__init__()=Bio.Alphabet.AlphabetEncoder-class.html#__init__,Method Bio.Alphabet.Gapped.__init__()=Bio.Alphabet.Gapped-class.html#__init__,Method Bio.Alphabet.HasStopCodon.__init__()=Bio.Alphabet.HasStopCodon-class.html#__init__,Method Bio.Application.AbstractCommandline.__init__()=Bio.Application.AbstractCommandline-class.html#__init__,Method Bio.Application.ApplicationError.__init__()=Bio.Application.ApplicationError-class.html#__init__,Method Bio.Application._AbstractParameter.__init__()=Bio.Application._AbstractParameter-class.html#__init__,Method Bio.Application._Argument.__init__()=Bio.Application._Argument-class.html#__init__,Method Bio.Application._Option.__init__()=Bio.Application._Option-class.html#__init__,Method Bio.Application._StaticArgument.__init__()=Bio.Application._StaticArgument-class.html#__init__,Method Bio.Application._Switch.__init__()=Bio.Application._Switch-class.html#__init__,Method Bio.Blast.Applications.NcbiblastformatterCommandline.__init__()=Bio.Blast.Applications.NcbiblastformatterCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastnCommandline.__init__()=Bio.Blast.Applications.NcbiblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastpCommandline.__init__()=Bio.Blast.Applications.NcbiblastpCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastxCommandline.__init__()=Bio.Blast.Applications.NcbiblastxCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbideltablastCommandline.__init__()=Bio.Blast.Applications.NcbideltablastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbipsiblastCommandline.__init__()=Bio.Blast.Applications.NcbipsiblastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbirpsblastCommandline.__init__()=Bio.Blast.Applications.NcbirpsblastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbirpstblastnCommandline.__init__()=Bio.Blast.Applications.NcbirpstblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbitblastnCommandline.__init__()=Bio.Blast.Applications.NcbitblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbitblastxCommandline.__init__()=Bio.Blast.Applications.NcbitblastxCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbibaseblastCommandline.__init__()=Bio.Blast.Applications._NcbibaseblastCommandline-class.html#__init__,Method Bio.Blast.Applications._Ncbiblast2SeqCommandline.__init__()=Bio.Blast.Applications._Ncbiblast2SeqCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbiblastCommandline.__init__()=Bio.Blast.Applications._NcbiblastCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbiblastMain2SeqCommandline.__init__()=Bio.Blast.Applications._NcbiblastMain2SeqCommandline-class.html#__init__,Method Bio.Blast.NCBIStandalone.BlastErrorParser.__init__()=Bio.Blast.NCBIStandalone.BlastErrorParser-class.html#__init__,Method Bio.Blast.NCBIStandalone.BlastParser.__init__()=Bio.Blast.NCBIStandalone.BlastParser-class.html#__init__,Method Bio.Blast.NCBIStandalone.Iterator.__init__()=Bio.Blast.NCBIStandalone.Iterator-class.html#__init__,Method Bio.Blast.NCBIStandalone.PSIBlastParser.__init__()=Bio.Blast.NCBIStandalone.PSIBlastParser-class.html#__init__,Method Bio.Blast.NCBIStandalone._BlastConsumer.__init__()=Bio.Blast.NCBIStandalone._BlastConsumer-class.html#__init__,Method Bio.Blast.NCBIStandalone._BlastErrorConsumer.__init__()=Bio.Blast.NCBIStandalone._BlastErrorConsumer-class.html#__init__,Method Bio.Blast.NCBIStandalone._PSIBlastConsumer.__init__()=Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html#__init__,Method Bio.Blast.NCBIXML.BlastParser.__init__()=Bio.Blast.NCBIXML.BlastParser-class.html#__init__,Method Bio.Blast.NCBIXML._XMLparser.__init__()=Bio.Blast.NCBIXML._XMLparser-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableEntry.__init__()=Bio.Blast.ParseBlastTable.BlastTableEntry-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableReader.__init__()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableRec.__init__()=Bio.Blast.ParseBlastTable.BlastTableRec-class.html#__init__,Method Bio.Blast.Record.Alignment.__init__()=Bio.Blast.Record.Alignment-class.html#__init__,Method Bio.Blast.Record.Blast.__init__()=Bio.Blast.Record.Blast-class.html#__init__,Method Bio.Blast.Record.DatabaseReport.__init__()=Bio.Blast.Record.DatabaseReport-class.html#__init__,Method Bio.Blast.Record.Description.__init__()=Bio.Blast.Record.Description-class.html#__init__,Method Bio.Blast.Record.HSP.__init__()=Bio.Blast.Record.HSP-class.html#__init__,Method Bio.Blast.Record.Header.__init__()=Bio.Blast.Record.Header-class.html#__init__,Method Bio.Blast.Record.MultipleAlignment.__init__()=Bio.Blast.Record.MultipleAlignment-class.html#__init__,Method Bio.Blast.Record.PSIBlast.__init__()=Bio.Blast.Record.PSIBlast-class.html#__init__,Method Bio.Blast.Record.Parameters.__init__()=Bio.Blast.Record.Parameters-class.html#__init__,Method Bio.Blast.Record.Round.__init__()=Bio.Blast.Record.Round-class.html#__init__,Method Bio.CAPS.CAPSMap.__init__()=Bio.CAPS.CAPSMap-class.html#__init__,Method Bio.CAPS.DifferentialCutsite.__init__()=Bio.CAPS.DifferentialCutsite-class.html#__init__,Method Bio.Cluster.Record.__init__()=Bio.Cluster.Record-class.html#__init__,Method Bio.Compass.Record.__init__()=Bio.Compass.Record-class.html#__init__,Method Bio.Crystal.Chain.__init__()=Bio.Crystal.Chain-class.html#__init__,Method Bio.Crystal.Crystal.__init__()=Bio.Crystal.Crystal-class.html#__init__,Method Bio.Crystal.Hetero.__init__()=Bio.Crystal.Hetero-class.html#__init__,Method Bio.Data.CodonTable.AmbiguousCodonTable.__init__()=Bio.Data.CodonTable.AmbiguousCodonTable-class.html#__init__,Method Bio.Data.CodonTable.AmbiguousForwardTable.__init__()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#__init__,Method Bio.Data.CodonTable.CodonTable.__init__()=Bio.Data.CodonTable.CodonTable-class.html#__init__,Method Bio.Data.CodonTable.NCBICodonTable.__init__()=Bio.Data.CodonTable.NCBICodonTable-class.html#__init__,Method Bio.Emboss.Applications.DiffseqCommandline.__init__()=Bio.Emboss.Applications.DiffseqCommandline-class.html#__init__,Method Bio.Emboss.Applications.EInvertedCommandline.__init__()=Bio.Emboss.Applications.EInvertedCommandline-class.html#__init__,Method Bio.Emboss.Applications.ETandemCommandline.__init__()=Bio.Emboss.Applications.ETandemCommandline-class.html#__init__,Method Bio.Emboss.Applications.Est2GenomeCommandline.__init__()=Bio.Emboss.Applications.Est2GenomeCommandline-class.html#__init__,Method Bio.Emboss.Applications.FConsenseCommandline.__init__()=Bio.Emboss.Applications.FConsenseCommandline-class.html#__init__,Method Bio.Emboss.Applications.FDNADistCommandline.__init__()=Bio.Emboss.Applications.FDNADistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FDNAParsCommandline.__init__()=Bio.Emboss.Applications.FDNAParsCommandline-class.html#__init__,Method Bio.Emboss.Applications.FNeighborCommandline.__init__()=Bio.Emboss.Applications.FNeighborCommandline-class.html#__init__,Method Bio.Emboss.Applications.FProtDistCommandline.__init__()=Bio.Emboss.Applications.FProtDistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FProtParsCommandline.__init__()=Bio.Emboss.Applications.FProtParsCommandline-class.html#__init__,Method Bio.Emboss.Applications.FSeqBootCommandline.__init__()=Bio.Emboss.Applications.FSeqBootCommandline-class.html#__init__,Method Bio.Emboss.Applications.FTreeDistCommandline.__init__()=Bio.Emboss.Applications.FTreeDistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FuzznucCommandline.__init__()=Bio.Emboss.Applications.FuzznucCommandline-class.html#__init__,Method Bio.Emboss.Applications.FuzzproCommandline.__init__()=Bio.Emboss.Applications.FuzzproCommandline-class.html#__init__,Method Bio.Emboss.Applications.IepCommandline.__init__()=Bio.Emboss.Applications.IepCommandline-class.html#__init__,Method Bio.Emboss.Applications.NeedleCommandline.__init__()=Bio.Emboss.Applications.NeedleCommandline-class.html#__init__,Method Bio.Emboss.Applications.NeedleallCommandline.__init__()=Bio.Emboss.Applications.NeedleallCommandline-class.html#__init__,Method Bio.Emboss.Applications.PalindromeCommandline.__init__()=Bio.Emboss.Applications.PalindromeCommandline-class.html#__init__,Method Bio.Emboss.Applications.Primer3Commandline.__init__()=Bio.Emboss.Applications.Primer3Commandline-class.html#__init__,Method Bio.Emboss.Applications.PrimerSearchCommandline.__init__()=Bio.Emboss.Applications.PrimerSearchCommandline-class.html#__init__,Method Bio.Emboss.Applications.SeqmatchallCommandline.__init__()=Bio.Emboss.Applications.SeqmatchallCommandline-class.html#__init__,Method Bio.Emboss.Applications.SeqretCommandline.__init__()=Bio.Emboss.Applications.SeqretCommandline-class.html#__init__,Method Bio.Emboss.Applications.StretcherCommandline.__init__()=Bio.Emboss.Applications.StretcherCommandline-class.html#__init__,Method Bio.Emboss.Applications.TranalignCommandline.__init__()=Bio.Emboss.Applications.TranalignCommandline-class.html#__init__,Method Bio.Emboss.Applications.WaterCommandline.__init__()=Bio.Emboss.Applications.WaterCommandline-class.html#__init__,Method Bio.Emboss.Applications._EmbossCommandLine.__init__()=Bio.Emboss.Applications._EmbossCommandLine-class.html#__init__,Method Bio.Emboss.Applications._EmbossMinimalCommandLine.__init__()=Bio.Emboss.Applications._EmbossMinimalCommandLine-class.html#__init__,Method Bio.Emboss.Primer3.Primers.__init__()=Bio.Emboss.Primer3.Primers-class.html#__init__,Method Bio.Emboss.Primer3.Record.__init__()=Bio.Emboss.Primer3.Record-class.html#__init__,Method Bio.Emboss.PrimerSearch.Amplifier.__init__()=Bio.Emboss.PrimerSearch.Amplifier-class.html#__init__,Method Bio.Emboss.PrimerSearch.InputRecord.__init__()=Bio.Emboss.PrimerSearch.InputRecord-class.html#__init__,Method Bio.Emboss.PrimerSearch.OutputRecord.__init__()=Bio.Emboss.PrimerSearch.OutputRecord-class.html#__init__,Method Bio.Entrez.Parser.Consumer.__init__()=Bio.Entrez.Parser.Consumer-class.html#__init__,Method Bio.Entrez.Parser.CorruptedXMLError.__init__()=Bio.Entrez.Parser.CorruptedXMLError-class.html#__init__,Method Bio.Entrez.Parser.DataHandler.__init__()=Bio.Entrez.Parser.DataHandler-class.html#__init__,Method Bio.Entrez.Parser.DictionaryConsumer.__init__()=Bio.Entrez.Parser.DictionaryConsumer-class.html#__init__,Method Bio.Entrez.Parser.ErrorConsumer.__init__()=Bio.Entrez.Parser.ErrorConsumer-class.html#__init__,Method Bio.Entrez.Parser.IntegerConsumer.__init__()=Bio.Entrez.Parser.IntegerConsumer-class.html#__init__,Method Bio.Entrez.Parser.ListConsumer.__init__()=Bio.Entrez.Parser.ListConsumer-class.html#__init__,Method Bio.Entrez.Parser.NotXMLError.__init__()=Bio.Entrez.Parser.NotXMLError-class.html#__init__,Method Bio.Entrez.Parser.StringConsumer.__init__()=Bio.Entrez.Parser.StringConsumer-class.html#__init__,Method Bio.Entrez.Parser.ValidationError.__init__()=Bio.Entrez.Parser.ValidationError-class.html#__init__,Method Bio.ExPASy.Enzyme.Record.__init__()=Bio.ExPASy.Enzyme.Record-class.html#__init__,Method Bio.ExPASy.Prodoc.Record.__init__()=Bio.ExPASy.Prodoc.Record-class.html#__init__,Method Bio.ExPASy.Prodoc.Reference.__init__()=Bio.ExPASy.Prodoc.Reference-class.html#__init__,Method Bio.ExPASy.Prosite.Record.__init__()=Bio.ExPASy.Prosite.Record-class.html#__init__,Method Bio.ExPASy.ScanProsite.ContentHandler.__init__()=Bio.ExPASy.ScanProsite.ContentHandler-class.html#__init__,Method Bio.ExPASy.ScanProsite.Parser.__init__()=Bio.ExPASy.ScanProsite.Parser-class.html#__init__,Method Bio.ExPASy.ScanProsite.Record.__init__()=Bio.ExPASy.ScanProsite.Record-class.html#__init__,Method Bio.ExPASy.cellosaurus.Record.__init__()=Bio.ExPASy.cellosaurus.Record-class.html#__init__,Method Bio.FSSP.FSSPAlignDict.__init__()=Bio.FSSP.FSSPAlignDict-class.html#__init__,Method Bio.FSSP.FSSPAlignRec.__init__()=Bio.FSSP.FSSPAlignRec-class.html#__init__,Method Bio.FSSP.FSSPHeader.__init__()=Bio.FSSP.FSSPHeader-class.html#__init__,Method Bio.FSSP.FSSPSumRec.__init__()=Bio.FSSP.FSSPSumRec-class.html#__init__,Method Bio.FSSP.FSSPTools.FSSPMultAlign.__init__()=Bio.FSSP.FSSPTools.FSSPMultAlign-class.html#__init__,Method Bio.FSSP.PosAlign.__init__()=Bio.FSSP.PosAlign-class.html#__init__,Method Bio.FSSP.fssp_rec.fff_rec.__init__()=Bio.FSSP.fssp_rec.fff_rec-class.html#__init__,Method Bio.File.UndoHandle.__init__()=Bio.File.UndoHandle-class.html#__init__,Method Bio.File._IndexedSeqFileDict.__init__()=Bio.File._IndexedSeqFileDict-class.html#__init__,Method Bio.File._SQLiteManySeqFilesDict.__init__()=Bio.File._SQLiteManySeqFilesDict-class.html#__init__,Method Bio.GA.Crossover.General.SafeFitnessCrossover.__init__()=Bio.GA.Crossover.General.SafeFitnessCrossover-class.html#__init__,Method Bio.GA.Crossover.GeneralPoint.GeneralPointCrossover.__init__()=Bio.GA.Crossover.GeneralPoint.GeneralPointCrossover-class.html#__init__,Method Bio.GA.Crossover.GeneralPoint.InterleaveCrossover.__init__()=Bio.GA.Crossover.GeneralPoint.InterleaveCrossover-class.html#__init__,Method Bio.GA.Crossover.Point.SinglePointCrossover.__init__()=Bio.GA.Crossover.Point.SinglePointCrossover-class.html#__init__,Method Bio.GA.Crossover.TwoPoint.TwoPointCrossover.__init__()=Bio.GA.Crossover.TwoPoint.TwoPointCrossover-class.html#__init__,Method Bio.GA.Crossover.Uniform.UniformCrossover.__init__()=Bio.GA.Crossover.Uniform.UniformCrossover-class.html#__init__,Method Bio.GA.Evolver.GenerationEvolver.__init__()=Bio.GA.Evolver.GenerationEvolver-class.html#__init__,Method Bio.GA.Evolver.SteadyStateEvolver.__init__()=Bio.GA.Evolver.SteadyStateEvolver-class.html#__init__,Method Bio.GA.Mutation.General.SafeFitnessMutation.__init__()=Bio.GA.Mutation.General.SafeFitnessMutation-class.html#__init__,Method Bio.GA.Mutation.Simple.ConversionMutation.__init__()=Bio.GA.Mutation.Simple.ConversionMutation-class.html#__init__,Method Bio.GA.Mutation.Simple.SinglePositionMutation.__init__()=Bio.GA.Mutation.Simple.SinglePositionMutation-class.html#__init__,Method Bio.GA.Organism.Organism.__init__()=Bio.GA.Organism.Organism-class.html#__init__,Method Bio.GA.Repair.Stabilizing.AmbiguousRepair.__init__()=Bio.GA.Repair.Stabilizing.AmbiguousRepair-class.html#__init__,Method Bio.GA.Selection.Abstract.AbstractSelection.__init__()=Bio.GA.Selection.Abstract.AbstractSelection-class.html#__init__,Method Bio.GA.Selection.Diversity.DiversitySelection.__init__()=Bio.GA.Selection.Diversity.DiversitySelection-class.html#__init__,Method Bio.GA.Selection.RouletteWheel.RouletteWheelSelection.__init__()=Bio.GA.Selection.RouletteWheel.RouletteWheelSelection-class.html#__init__,Method Bio.GA.Selection.Tournament.TournamentSelection.__init__()=Bio.GA.Selection.Tournament.TournamentSelection-class.html#__init__,Method Bio.GenBank.FeatureParser.__init__()=Bio.GenBank.FeatureParser-class.html#__init__,Method Bio.GenBank.Iterator.__init__()=Bio.GenBank.Iterator-class.html#__init__,Method Bio.GenBank.Record.Feature.__init__()=Bio.GenBank.Record.Feature-class.html#__init__,Method Bio.GenBank.Record.Qualifier.__init__()=Bio.GenBank.Record.Qualifier-class.html#__init__,Method Bio.GenBank.Record.Record.__init__()=Bio.GenBank.Record.Record-class.html#__init__,Method Bio.GenBank.Record.Reference.__init__()=Bio.GenBank.Record.Reference-class.html#__init__,Method Bio.GenBank.RecordParser.__init__()=Bio.GenBank.RecordParser-class.html#__init__,Method Bio.GenBank.Scanner.InsdcScanner.__init__()=Bio.GenBank.Scanner.InsdcScanner-class.html#__init__,Method Bio.GenBank._BaseGenBankConsumer.__init__()=Bio.GenBank._BaseGenBankConsumer-class.html#__init__,Method Bio.GenBank._FeatureConsumer.__init__()=Bio.GenBank._FeatureConsumer-class.html#__init__,Method Bio.GenBank._RecordConsumer.__init__()=Bio.GenBank._RecordConsumer-class.html#__init__,Method Bio.GenBank.utils.FeatureValueCleaner.__init__()=Bio.GenBank.utils.FeatureValueCleaner-class.html#__init__,Method Bio.Geo.Record.Record.__init__()=Bio.Geo.Record.Record-class.html#__init__,Method Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment.__init__()=Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome.Chromosome.__init__()=Bio.Graphics.BasicChromosome.Chromosome-class.html#__init__,Method Bio.Graphics.BasicChromosome.ChromosomeSegment.__init__()=Bio.Graphics.BasicChromosome.ChromosomeSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome.Organism.__init__()=Bio.Graphics.BasicChromosome.Organism-class.html#__init__,Method Bio.Graphics.BasicChromosome.TelomereSegment.__init__()=Bio.Graphics.BasicChromosome.TelomereSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome._ChromosomeComponent.__init__()=Bio.Graphics.BasicChromosome._ChromosomeComponent-class.html#__init__,Method Bio.Graphics.ColorSpiral.ColorSpiral.__init__()=Bio.Graphics.ColorSpiral.ColorSpiral-class.html#__init__,Method Bio.Graphics.Comparative.ComparativeScatterPlot.__init__()=Bio.Graphics.Comparative.ComparativeScatterPlot-class.html#__init__,Method Bio.Graphics.DisplayRepresentation.ChromosomeCounts.__init__()=Bio.Graphics.DisplayRepresentation.ChromosomeCounts-class.html#__init__,Method Bio.Graphics.Distribution.BarChartDistribution.__init__()=Bio.Graphics.Distribution.BarChartDistribution-class.html#__init__,Method Bio.Graphics.Distribution.DistributionPage.__init__()=Bio.Graphics.Distribution.DistributionPage-class.html#__init__,Method Bio.Graphics.Distribution.LineDistribution.__init__()=Bio.Graphics.Distribution.LineDistribution-class.html#__init__,Method Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.__init__()=Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.__init__()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.__init__()=Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#__init__,Method Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.__init__()=Bio.Graphics.GenomeDiagram._CrossLink.CrossLink-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.__init__()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Feature.Feature.__init__()=Bio.Graphics.GenomeDiagram._Feature.Feature-class.html#__init__,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__init__()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.__init__()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#__init__,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__init__()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#__init__,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.__init__()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Track.Track.__init__()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#__init__,Method Bio.Graphics.KGML_vis.KGMLCanvas.__init__()=Bio.Graphics.KGML_vis.KGMLCanvas-class.html#__init__,Method Bio.HMM.DynamicProgramming.AbstractDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.AbstractDPAlgorithms-class.html#__init__,Method Bio.HMM.DynamicProgramming.LogDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.LogDPAlgorithms-class.html#__init__,Method Bio.HMM.DynamicProgramming.ScaledDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.ScaledDPAlgorithms-class.html#__init__,Method Bio.HMM.MarkovModel.HiddenMarkovModel.__init__()=Bio.HMM.MarkovModel.HiddenMarkovModel-class.html#__init__,Method Bio.HMM.MarkovModel.MarkovModelBuilder.__init__()=Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#__init__,Method Bio.HMM.Trainer.AbstractTrainer.__init__()=Bio.HMM.Trainer.AbstractTrainer-class.html#__init__,Method Bio.HMM.Trainer.BaumWelchTrainer.__init__()=Bio.HMM.Trainer.BaumWelchTrainer-class.html#__init__,Method Bio.HMM.Trainer.KnownStateTrainer.__init__()=Bio.HMM.Trainer.KnownStateTrainer-class.html#__init__,Method Bio.HMM.Trainer.TrainingSequence.__init__()=Bio.HMM.Trainer.TrainingSequence-class.html#__init__,Method Bio.Index._InMemoryIndex.__init__()=Bio.Index._InMemoryIndex-class.html#__init__,Method Bio.Index._ShelveIndex.__init__()=Bio.Index._ShelveIndex-class.html#__init__,Method Bio.KDTree.KDTree'.KDTree.__init__()=Bio.KDTree.KDTree%27.KDTree-class.html#__init__,Method Bio.KEGG.Compound.Record.__init__()=Bio.KEGG.Compound.Record-class.html#__init__,Method Bio.KEGG.Enzyme.Record.__init__()=Bio.KEGG.Enzyme.Record-class.html#__init__,Method Bio.KEGG.Gene.Record.__init__()=Bio.KEGG.Gene.Record-class.html#__init__,Method Bio.KEGG.KGML.KGML_parser.KGMLParser.__init__()=Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Component.__init__()=Bio.KEGG.KGML.KGML_pathway.Component-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Entry.__init__()=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Graphics.__init__()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Pathway.__init__()=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Reaction.__init__()=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Relation.__init__()=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#__init__,Method Bio.LogisticRegression.LogisticRegression.__init__()=Bio.LogisticRegression.LogisticRegression-class.html#__init__,Method Bio.MarkovModel.MarkovModel.__init__()=Bio.MarkovModel.MarkovModel-class.html#__init__,Method Bio.MaxEntropy.MaxEntropy.__init__()=Bio.MaxEntropy.MaxEntropy-class.html#__init__,Method Bio.NMR.xpktools.Peaklist.__init__()=Bio.NMR.xpktools.Peaklist-class.html#__init__,Method Bio.NMR.xpktools.XpkEntry.__init__()=Bio.NMR.xpktools.XpkEntry-class.html#__init__,Method Bio.NaiveBayes.NaiveBayes.__init__()=Bio.NaiveBayes.NaiveBayes-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.AbstractLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.AbstractLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.InputLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Network.BasicNetwork.__init__()=Bio.NeuralNetwork.BackPropagation.Network.BasicNetwork-class.html#__init__,Method Bio.NeuralNetwork.Gene.Motif.MotifCoder.__init__()=Bio.NeuralNetwork.Gene.Motif.MotifCoder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Motif.MotifFinder.__init__()=Bio.NeuralNetwork.Gene.Motif.MotifFinder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.__init__()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#__init__,Method Bio.NeuralNetwork.Gene.Pattern.PatternRepository.__init__()=Bio.NeuralNetwork.Gene.Pattern.PatternRepository-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness.__init__()=Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder.__init__()=Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness.__init__()=Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator.__init__()=Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.Schema.__init__()=Bio.NeuralNetwork.Gene.Schema.Schema-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SchemaCoder.__init__()=Bio.NeuralNetwork.Gene.Schema.SchemaCoder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SchemaFactory.__init__()=Bio.NeuralNetwork.Gene.Schema.SchemaFactory-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SchemaFinder.__init__()=Bio.NeuralNetwork.Gene.Schema.SchemaFinder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SimpleFinisher.__init__()=Bio.NeuralNetwork.Gene.Schema.SimpleFinisher-class.html#__init__,Method Bio.NeuralNetwork.Gene.Signature.SignatureCoder.__init__()=Bio.NeuralNetwork.Gene.Signature.SignatureCoder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Signature.SignatureFinder.__init__()=Bio.NeuralNetwork.Gene.Signature.SignatureFinder-class.html#__init__,Method Bio.NeuralNetwork.StopTraining.ValidationIncreaseStop.__init__()=Bio.NeuralNetwork.StopTraining.ValidationIncreaseStop-class.html#__init__,Method Bio.NeuralNetwork.Training.ExampleManager.__init__()=Bio.NeuralNetwork.Training.ExampleManager-class.html#__init__,Method Bio.NeuralNetwork.Training.TrainingExample.__init__()=Bio.NeuralNetwork.Training.TrainingExample-class.html#__init__,Method Bio.Nexus.Nexus.Block.__init__()=Bio.Nexus.Nexus.Block-class.html#__init__,Method Bio.Nexus.Nexus.CharBuffer.__init__()=Bio.Nexus.Nexus.CharBuffer-class.html#__init__,Method Bio.Nexus.Nexus.Commandline.__init__()=Bio.Nexus.Nexus.Commandline-class.html#__init__,Method Bio.Nexus.Nexus.Nexus.__init__()=Bio.Nexus.Nexus.Nexus-class.html#__init__,Method Bio.Nexus.Nexus.StepMatrix.__init__()=Bio.Nexus.Nexus.StepMatrix-class.html#__init__,Method Bio.Nexus.Nodes.Chain.__init__()=Bio.Nexus.Nodes.Chain-class.html#__init__,Method Bio.Nexus.Nodes.Node.__init__()=Bio.Nexus.Nodes.Node-class.html#__init__,Method Bio.Nexus.StandardData.StandardData.__init__()=Bio.Nexus.StandardData.StandardData-class.html#__init__,Method Bio.Nexus.Trees.NodeData.__init__()=Bio.Nexus.Trees.NodeData-class.html#__init__,Method Bio.Nexus.Trees.Tree.__init__()=Bio.Nexus.Trees.Tree-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractPropertyMap-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap-class.html#__init__,Method Bio.PDB.Atom.Atom.__init__()=Bio.PDB.Atom.Atom-class.html#__init__,Method Bio.PDB.Atom.DisorderedAtom.__init__()=Bio.PDB.Atom.DisorderedAtom-class.html#__init__,Method Bio.PDB.Chain.Chain.__init__()=Bio.PDB.Chain.Chain-class.html#__init__,Method Bio.PDB.DSSP'.DSSP.__init__()=Bio.PDB.DSSP%27.DSSP-class.html#__init__,Method Bio.PDB.Dice.ChainSelector.__init__()=Bio.PDB.Dice.ChainSelector-class.html#__init__,Method Bio.PDB.Entity.DisorderedEntityWrapper.__init__()=Bio.PDB.Entity.DisorderedEntityWrapper-class.html#__init__,Method Bio.PDB.Entity.Entity.__init__()=Bio.PDB.Entity.Entity-class.html#__init__,Method Bio.PDB.FragmentMapper'.Fragment.__init__()=Bio.PDB.FragmentMapper%27.Fragment-class.html#__init__,Method Bio.PDB.FragmentMapper'.FragmentMapper.__init__()=Bio.PDB.FragmentMapper%27.FragmentMapper-class.html#__init__,Method Bio.PDB.HSExposure.ExposureCN.__init__()=Bio.PDB.HSExposure.ExposureCN-class.html#__init__,Method Bio.PDB.HSExposure.HSExposureCA.__init__()=Bio.PDB.HSExposure.HSExposureCA-class.html#__init__,Method Bio.PDB.HSExposure.HSExposureCB.__init__()=Bio.PDB.HSExposure.HSExposureCB-class.html#__init__,Method Bio.PDB.HSExposure._AbstractHSExposure.__init__()=Bio.PDB.HSExposure._AbstractHSExposure-class.html#__init__,Method Bio.PDB.MMCIF2Dict.MMCIF2Dict.__init__()=Bio.PDB.MMCIF2Dict.MMCIF2Dict-class.html#__init__,Method Bio.PDB.MMCIFParser'.FastMMCIFParser.__init__()=Bio.PDB.MMCIFParser%27.FastMMCIFParser-class.html#__init__,Method Bio.PDB.MMCIFParser'.MMCIFParser.__init__()=Bio.PDB.MMCIFParser%27.MMCIFParser-class.html#__init__,Method Bio.PDB.Model.Model.__init__()=Bio.PDB.Model.Model-class.html#__init__,Method Bio.PDB.NACCESS.NACCESS.__init__()=Bio.PDB.NACCESS.NACCESS-class.html#__init__,Method Bio.PDB.NACCESS.NACCESS_atomic.__init__()=Bio.PDB.NACCESS.NACCESS_atomic-class.html#__init__,Method Bio.PDB.NeighborSearch'.NeighborSearch.__init__()=Bio.PDB.NeighborSearch%27.NeighborSearch-class.html#__init__,Method Bio.PDB.PDBIO'.PDBIO.__init__()=Bio.PDB.PDBIO%27.PDBIO-class.html#__init__,Method Bio.PDB.PDBList'.PDBList.__init__()=Bio.PDB.PDBList%27.PDBList-class.html#__init__,Method Bio.PDB.PDBParser'.PDBParser.__init__()=Bio.PDB.PDBParser%27.PDBParser-class.html#__init__,Method Bio.PDB.PSEA.PSEA.__init__()=Bio.PDB.PSEA.PSEA-class.html#__init__,Method Bio.PDB.Polypeptide.CaPPBuilder.__init__()=Bio.PDB.Polypeptide.CaPPBuilder-class.html#__init__,Method Bio.PDB.Polypeptide.PPBuilder.__init__()=Bio.PDB.Polypeptide.PPBuilder-class.html#__init__,Method Bio.PDB.Polypeptide._PPBuilder.__init__()=Bio.PDB.Polypeptide._PPBuilder-class.html#__init__,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.__init__()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#__init__,Method Bio.PDB.Residue.DisorderedResidue.__init__()=Bio.PDB.Residue.DisorderedResidue-class.html#__init__,Method Bio.PDB.Residue.Residue.__init__()=Bio.PDB.Residue.Residue-class.html#__init__,Method Bio.PDB.ResidueDepth'.ResidueDepth.__init__()=Bio.PDB.ResidueDepth%27.ResidueDepth-class.html#__init__,Method Bio.PDB.Structure.Structure.__init__()=Bio.PDB.Structure.Structure-class.html#__init__,Method Bio.PDB.StructureAlignment'.StructureAlignment.__init__()=Bio.PDB.StructureAlignment%27.StructureAlignment-class.html#__init__,Method Bio.PDB.StructureBuilder.StructureBuilder.__init__()=Bio.PDB.StructureBuilder.StructureBuilder-class.html#__init__,Method Bio.PDB.Superimposer'.Superimposer.__init__()=Bio.PDB.Superimposer%27.Superimposer-class.html#__init__,Method Bio.PDB.mmcifio.MMCIFIO.__init__()=Bio.PDB.mmcifio.MMCIFIO-class.html#__init__,Method Bio.PDB.mmtf.DefaultParser.StructureDecoder.__init__()=Bio.PDB.mmtf.DefaultParser.StructureDecoder-class.html#__init__,Method Bio.PDB.vectors.Vector.__init__()=Bio.PDB.vectors.Vector-class.html#__init__,Method Bio.ParserSupport.TaggingConsumer.__init__()=Bio.ParserSupport.TaggingConsumer-class.html#__init__,Method Bio.Pathway.Network.__init__()=Bio.Pathway.Network-class.html#__init__,Method Bio.Pathway.Reaction.__init__()=Bio.Pathway.Reaction-class.html#__init__,Method Bio.Pathway.Rep.Graph.Graph.__init__()=Bio.Pathway.Rep.Graph.Graph-class.html#__init__,Method Bio.Pathway.Rep.MultiGraph.MultiGraph.__init__()=Bio.Pathway.Rep.MultiGraph.MultiGraph-class.html#__init__,Method Bio.Pathway.System.__init__()=Bio.Pathway.System-class.html#__init__,Method Bio.Phylo.Applications._Fasttree.FastTreeCommandline.__init__()=Bio.Phylo.Applications._Fasttree.FastTreeCommandline-class.html#__init__,Method Bio.Phylo.Applications._Phyml.PhymlCommandline.__init__()=Bio.Phylo.Applications._Phyml.PhymlCommandline-class.html#__init__,Method Bio.Phylo.Applications._Raxml.RaxmlCommandline.__init__()=Bio.Phylo.Applications._Raxml.RaxmlCommandline-class.html#__init__,Method Bio.Phylo.BaseTree.BranchColor.__init__()=Bio.Phylo.BaseTree.BranchColor-class.html#__init__,Method Bio.Phylo.BaseTree.Clade.__init__()=Bio.Phylo.BaseTree.Clade-class.html#__init__,Method Bio.Phylo.BaseTree.Tree.__init__()=Bio.Phylo.BaseTree.Tree-class.html#__init__,Method Bio.Phylo.CDAO.Clade.__init__()=Bio.Phylo.CDAO.Clade-class.html#__init__,Method Bio.Phylo.CDAO.Tree.__init__()=Bio.Phylo.CDAO.Tree-class.html#__init__,Method Bio.Phylo.CDAOIO.Parser.__init__()=Bio.Phylo.CDAOIO.Parser-class.html#__init__,Method Bio.Phylo.CDAOIO.Writer.__init__()=Bio.Phylo.CDAOIO.Writer-class.html#__init__,Method Bio.Phylo.NeXML.Clade.__init__()=Bio.Phylo.NeXML.Clade-class.html#__init__,Method Bio.Phylo.NeXML.Tree.__init__()=Bio.Phylo.NeXML.Tree-class.html#__init__,Method Bio.Phylo.NeXMLIO.Parser.__init__()=Bio.Phylo.NeXMLIO.Parser-class.html#__init__,Method Bio.Phylo.NeXMLIO.Writer.__init__()=Bio.Phylo.NeXMLIO.Writer-class.html#__init__,Method Bio.Phylo.Newick.Clade.__init__()=Bio.Phylo.Newick.Clade-class.html#__init__,Method Bio.Phylo.Newick.Tree.__init__()=Bio.Phylo.Newick.Tree-class.html#__init__,Method Bio.Phylo.NewickIO.Parser.__init__()=Bio.Phylo.NewickIO.Parser-class.html#__init__,Method Bio.Phylo.NewickIO.Writer.__init__()=Bio.Phylo.NewickIO.Writer-class.html#__init__,Method Bio.Phylo.PAML._paml.Paml.__init__()=Bio.Phylo.PAML._paml.Paml-class.html#__init__,Method Bio.Phylo.PAML.baseml.Baseml.__init__()=Bio.Phylo.PAML.baseml.Baseml-class.html#__init__,Method Bio.Phylo.PAML.codeml.Codeml.__init__()=Bio.Phylo.PAML.codeml.Codeml-class.html#__init__,Method Bio.Phylo.PAML.yn00.Yn00.__init__()=Bio.Phylo.PAML.yn00.Yn00-class.html#__init__,Method Bio.Phylo.PhyloXML.Accession.__init__()=Bio.Phylo.PhyloXML.Accession-class.html#__init__,Method Bio.Phylo.PhyloXML.Annotation.__init__()=Bio.Phylo.PhyloXML.Annotation-class.html#__init__,Method Bio.Phylo.PhyloXML.BinaryCharacters.__init__()=Bio.Phylo.PhyloXML.BinaryCharacters-class.html#__init__,Method Bio.Phylo.PhyloXML.BranchColor.__init__()=Bio.Phylo.PhyloXML.BranchColor-class.html#__init__,Method Bio.Phylo.PhyloXML.Clade.__init__()=Bio.Phylo.PhyloXML.Clade-class.html#__init__,Method Bio.Phylo.PhyloXML.CladeRelation.__init__()=Bio.Phylo.PhyloXML.CladeRelation-class.html#__init__,Method Bio.Phylo.PhyloXML.Confidence.__init__()=Bio.Phylo.PhyloXML.Confidence-class.html#__init__,Method Bio.Phylo.PhyloXML.Date.__init__()=Bio.Phylo.PhyloXML.Date-class.html#__init__,Method Bio.Phylo.PhyloXML.Distribution.__init__()=Bio.Phylo.PhyloXML.Distribution-class.html#__init__,Method Bio.Phylo.PhyloXML.DomainArchitecture.__init__()=Bio.Phylo.PhyloXML.DomainArchitecture-class.html#__init__,Method Bio.Phylo.PhyloXML.Events.__init__()=Bio.Phylo.PhyloXML.Events-class.html#__init__,Method Bio.Phylo.PhyloXML.Id.__init__()=Bio.Phylo.PhyloXML.Id-class.html#__init__,Method Bio.Phylo.PhyloXML.MolSeq.__init__()=Bio.Phylo.PhyloXML.MolSeq-class.html#__init__,Method Bio.Phylo.PhyloXML.Other.__init__()=Bio.Phylo.PhyloXML.Other-class.html#__init__,Method Bio.Phylo.PhyloXML.Phylogeny.__init__()=Bio.Phylo.PhyloXML.Phylogeny-class.html#__init__,Method Bio.Phylo.PhyloXML.Phyloxml.__init__()=Bio.Phylo.PhyloXML.Phyloxml-class.html#__init__,Method Bio.Phylo.PhyloXML.Point.__init__()=Bio.Phylo.PhyloXML.Point-class.html#__init__,Method Bio.Phylo.PhyloXML.Polygon.__init__()=Bio.Phylo.PhyloXML.Polygon-class.html#__init__,Method Bio.Phylo.PhyloXML.Property.__init__()=Bio.Phylo.PhyloXML.Property-class.html#__init__,Method Bio.Phylo.PhyloXML.ProteinDomain.__init__()=Bio.Phylo.PhyloXML.ProteinDomain-class.html#__init__,Method Bio.Phylo.PhyloXML.Reference.__init__()=Bio.Phylo.PhyloXML.Reference-class.html#__init__,Method Bio.Phylo.PhyloXML.Sequence.__init__()=Bio.Phylo.PhyloXML.Sequence-class.html#__init__,Method Bio.Phylo.PhyloXML.SequenceRelation.__init__()=Bio.Phylo.PhyloXML.SequenceRelation-class.html#__init__,Method Bio.Phylo.PhyloXML.Taxonomy.__init__()=Bio.Phylo.PhyloXML.Taxonomy-class.html#__init__,Method Bio.Phylo.PhyloXML.Uri.__init__()=Bio.Phylo.PhyloXML.Uri-class.html#__init__,Method Bio.Phylo.PhyloXMLIO.Parser.__init__()=Bio.Phylo.PhyloXMLIO.Parser-class.html#__init__,Method Bio.Phylo.PhyloXMLIO.Writer.__init__()=Bio.Phylo.PhyloXMLIO.Writer-class.html#__init__,Method Bio.Phylo.TreeConstruction.DistanceCalculator.__init__()=Bio.Phylo.TreeConstruction.DistanceCalculator-class.html#__init__,Method Bio.Phylo.TreeConstruction.DistanceMatrix.__init__()=Bio.Phylo.TreeConstruction.DistanceMatrix-class.html#__init__,Method Bio.Phylo.TreeConstruction.DistanceTreeConstructor.__init__()=Bio.Phylo.TreeConstruction.DistanceTreeConstructor-class.html#__init__,Method Bio.Phylo.TreeConstruction.NNITreeSearcher.__init__()=Bio.Phylo.TreeConstruction.NNITreeSearcher-class.html#__init__,Method Bio.Phylo.TreeConstruction.ParsimonyScorer.__init__()=Bio.Phylo.TreeConstruction.ParsimonyScorer-class.html#__init__,Method Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor.__init__()=Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor-class.html#__init__,Method Bio.Phylo.TreeConstruction._Matrix.__init__()=Bio.Phylo.TreeConstruction._Matrix-class.html#__init__,Method Bio.PopGen.GenePop.Controller.GenePopController.__init__()=Bio.PopGen.GenePop.Controller.GenePopController-class.html#__init__,Method Bio.PopGen.GenePop.Controller._FileIterator.__init__()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#__init__,Method Bio.PopGen.GenePop.Controller._GenePopCommandline.__init__()=Bio.PopGen.GenePop.Controller._GenePopCommandline-class.html#__init__,Method Bio.PopGen.GenePop.EasyController.EasyController.__init__()=Bio.PopGen.GenePop.EasyController.EasyController-class.html#__init__,Method Bio.PopGen.GenePop.FileParser.FileRecord.__init__()=Bio.PopGen.GenePop.FileParser.FileRecord-class.html#__init__,Method Bio.PopGen.GenePop.LargeFileParser.Record.__init__()=Bio.PopGen.GenePop.LargeFileParser.Record-class.html#__init__,Method Bio.PopGen.GenePop.Record.__init__()=Bio.PopGen.GenePop.Record-class.html#__init__,Method Bio.Restriction.PrintFormat.PrintFormat.__init__()=Bio.Restriction.PrintFormat.PrintFormat-class.html#__init__,Method Bio.Restriction.Restriction.Analysis.__init__()=Bio.Restriction.Restriction.Analysis-class.html#__init__,Method Bio.Restriction.Restriction.FormattedSeq.__init__()=Bio.Restriction.Restriction.FormattedSeq-class.html#__init__,Method Bio.Restriction.Restriction.RestrictionBatch.__init__()=Bio.Restriction.Restriction.RestrictionBatch-class.html#__init__,Method Bio.Restriction.Restriction.RestrictionType.__init__()=Bio.Restriction.Restriction.RestrictionType-class.html#__init__,Method Bio.SCOP.Astral.__init__()=Bio.SCOP.Astral-class.html#__init__,Method Bio.SCOP.Cla.Index.__init__()=Bio.SCOP.Cla.Index-class.html#__init__,Method Bio.SCOP.Cla.Record.__init__()=Bio.SCOP.Cla.Record-class.html#__init__,Method Bio.SCOP.Des.Record.__init__()=Bio.SCOP.Des.Record-class.html#__init__,Method Bio.SCOP.Dom.Record.__init__()=Bio.SCOP.Dom.Record-class.html#__init__,Method Bio.SCOP.Domain.__init__()=Bio.SCOP.Domain-class.html#__init__,Method Bio.SCOP.Hie.Record.__init__()=Bio.SCOP.Hie.Record-class.html#__init__,Method Bio.SCOP.Node.__init__()=Bio.SCOP.Node-class.html#__init__,Method Bio.SCOP.Raf.Res.__init__()=Bio.SCOP.Raf.Res-class.html#__init__,Method Bio.SCOP.Raf.SeqMap.__init__()=Bio.SCOP.Raf.SeqMap-class.html#__init__,Method Bio.SCOP.Raf.SeqMapIndex.__init__()=Bio.SCOP.Raf.SeqMapIndex-class.html#__init__,Method Bio.SCOP.Residues.Residues.__init__()=Bio.SCOP.Residues.Residues-class.html#__init__,Method Bio.SCOP.Scop.__init__()=Bio.SCOP.Scop-class.html#__init__,Method Bio.SVDSuperimposer.SVDSuperimposer.__init__()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabParser.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_text.BlastTextParser.__init__()=Bio.SearchIO.BlastIO.blast_text.BlastTextParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator.__init__()=Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslIndexer.__init__()=Bio.SearchIO.BlatIO.BlatPslIndexer-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslParser.__init__()=Bio.SearchIO.BlatIO.BlatPslParser-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslWriter.__init__()=Bio.SearchIO.BlatIO.BlatPslWriter-class.html#__init__,Method Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.__init__()=Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html#__init__,Method Bio.SearchIO.FastaIO.FastaM10Indexer.__init__()=Bio.SearchIO.FastaIO.FastaM10Indexer-class.html#__init__,Method Bio.SearchIO.FastaIO.FastaM10Parser.__init__()=Bio.SearchIO.FastaIO.FastaM10Parser-class.html#__init__,Method Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer.__init__()=Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser.__init__()=Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.__init__()=Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser.__init__()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.__init__()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser.__init__()=Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#__init__,Method Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser.__init__()=Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser-class.html#__init__,Method Bio.SearchIO._index.SearchIndexer.__init__()=Bio.SearchIO._index.SearchIndexer-class.html#__init__,Method Bio.SearchIO._model.hit.Hit.__init__()=Bio.SearchIO._model.hit.Hit-class.html#__init__,Method Bio.SearchIO._model.hsp.HSP.__init__()=Bio.SearchIO._model.hsp.HSP-class.html#__init__,Method Bio.SearchIO._model.hsp.HSPFragment.__init__()=Bio.SearchIO._model.hsp.HSPFragment-class.html#__init__,Method Bio.SearchIO._model.query.QueryResult.__init__()=Bio.SearchIO._model.query.QueryResult-class.html#__init__,Method Bio.Seq.MutableSeq.__init__()=Bio.Seq.MutableSeq-class.html#__init__,Method Bio.Seq.Seq.__init__()=Bio.Seq.Seq-class.html#__init__,Method Bio.Seq.UnknownSeq.__init__()=Bio.Seq.UnknownSeq-class.html#__init__,Method Bio.SeqFeature.CompoundLocation.__init__()=Bio.SeqFeature.CompoundLocation-class.html#__init__,Method Bio.SeqFeature.FeatureLocation.__init__()=Bio.SeqFeature.FeatureLocation-class.html#__init__,Method Bio.SeqFeature.PositionGap.__init__()=Bio.SeqFeature.PositionGap-class.html#__init__,Method Bio.SeqFeature.Reference.__init__()=Bio.SeqFeature.Reference-class.html#__init__,Method Bio.SeqFeature.SeqFeature.__init__()=Bio.SeqFeature.SeqFeature-class.html#__init__,Method Bio.SeqIO.FastaIO.FastaTwoLineWriter.__init__()=Bio.SeqIO.FastaIO.FastaTwoLineWriter-class.html#__init__,Method Bio.SeqIO.FastaIO.FastaWriter.__init__()=Bio.SeqIO.FastaIO.FastaWriter-class.html#__init__,Method Bio.SeqIO.Interfaces.SequenceIterator.__init__()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#__init__,Method Bio.SeqIO.Interfaces.SequenceWriter.__init__()=Bio.SeqIO.Interfaces.SequenceWriter-class.html#__init__,Method Bio.SeqIO.Interfaces.SequentialSequenceWriter.__init__()=Bio.SeqIO.Interfaces.SequentialSequenceWriter-class.html#__init__,Method Bio.SeqIO.PhdIO.PhdWriter.__init__()=Bio.SeqIO.PhdIO.PhdWriter-class.html#__init__,Method Bio.SeqIO.PirIO.PirWriter.__init__()=Bio.SeqIO.PirIO.PirWriter-class.html#__init__,Method Bio.SeqIO.QualityIO.QualPhredWriter.__init__()=Bio.SeqIO.QualityIO.QualPhredWriter-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.SeqXmlIterator.__init__()=Bio.SeqIO.SeqXmlIO.SeqXmlIterator-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.SeqXmlWriter.__init__()=Bio.SeqIO.SeqXmlIO.SeqXmlWriter-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.XMLRecordIterator.__init__()=Bio.SeqIO.SeqXmlIO.XMLRecordIterator-class.html#__init__,Method Bio.SeqIO.SffIO.SffWriter.__init__()=Bio.SeqIO.SffIO.SffWriter-class.html#__init__,Method Bio.SeqIO.SffIO._AddTellHandle.__init__()=Bio.SeqIO.SffIO._AddTellHandle-class.html#__init__,Method Bio.SeqIO.UniprotIO.Parser.__init__()=Bio.SeqIO.UniprotIO.Parser-class.html#__init__,Method Bio.SeqIO._index.IntelliGeneticsRandomAccess.__init__()=Bio.SeqIO._index.IntelliGeneticsRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SeqFileRandomAccess.__init__()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SequentialSeqFileRandomAccess.__init__()=Bio.SeqIO._index.SequentialSeqFileRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SffRandomAccess.__init__()=Bio.SeqIO._index.SffRandomAccess-class.html#__init__,Method Bio.SeqRecord.SeqRecord.__init__()=Bio.SeqRecord.SeqRecord-class.html#__init__,Method Bio.SeqRecord._RestrictedDict.__init__()=Bio.SeqRecord._RestrictedDict-class.html#__init__,Method Bio.SeqUtils.CodonUsage.CodonAdaptationIndex.__init__()=Bio.SeqUtils.CodonUsage.CodonAdaptationIndex-class.html#__init__,Method Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.__init__()=Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint-class.html#__init__,Method Bio.SeqUtils.ProtParam.ProteinAnalysis.__init__()=Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#__init__,Method Bio.Sequencing.Ace.ACEFileRecord.__init__()=Bio.Sequencing.Ace.ACEFileRecord-class.html#__init__,Method Bio.Sequencing.Ace.Contig.__init__()=Bio.Sequencing.Ace.Contig-class.html#__init__,Method Bio.Sequencing.Ace.Reads.__init__()=Bio.Sequencing.Ace.Reads-class.html#__init__,Method Bio.Sequencing.Ace.af.__init__()=Bio.Sequencing.Ace.af-class.html#__init__,Method Bio.Sequencing.Ace.bs.__init__()=Bio.Sequencing.Ace.bs-class.html#__init__,Method Bio.Sequencing.Ace.ct.__init__()=Bio.Sequencing.Ace.ct-class.html#__init__,Method Bio.Sequencing.Ace.ds.__init__()=Bio.Sequencing.Ace.ds-class.html#__init__,Method Bio.Sequencing.Ace.qa.__init__()=Bio.Sequencing.Ace.qa-class.html#__init__,Method Bio.Sequencing.Ace.rd.__init__()=Bio.Sequencing.Ace.rd-class.html#__init__,Method Bio.Sequencing.Ace.rt.__init__()=Bio.Sequencing.Ace.rt-class.html#__init__,Method Bio.Sequencing.Ace.wa.__init__()=Bio.Sequencing.Ace.wa-class.html#__init__,Method Bio.Sequencing.Ace.wr.__init__()=Bio.Sequencing.Ace.wr-class.html#__init__,Method Bio.Sequencing.Applications._Novoalign.NovoalignCommandline.__init__()=Bio.Sequencing.Applications._Novoalign.NovoalignCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaAlignCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaAlignCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaBwaswCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaBwaswCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaIndexCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaIndexCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaSampeCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaSampeCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaSamseCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaSamseCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsVersion0xSortCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsVersion0xSortCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsVersion1xSortCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsVersion1xSortCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline-class.html#__init__,Method Bio.Sequencing.Phd.Record.__init__()=Bio.Sequencing.Phd.Record-class.html#__init__,Method Bio.SubsMat.FreqTable.FreqTable.__init__()=Bio.SubsMat.FreqTable.FreqTable-class.html#__init__,Method Bio.SubsMat.SeqMat.__init__()=Bio.SubsMat.SeqMat-class.html#__init__,Method Bio.SwissProt.KeyWList.Record.__init__()=Bio.SwissProt.KeyWList.Record-class.html#__init__,Method Bio.SwissProt.Record.__init__()=Bio.SwissProt.Record-class.html#__init__,Method Bio.SwissProt.Reference.__init__()=Bio.SwissProt.Reference-class.html#__init__,Method Bio.UniGene.ProtsimLine.__init__()=Bio.UniGene.ProtsimLine-class.html#__init__,Method Bio.UniGene.Record.__init__()=Bio.UniGene.Record-class.html#__init__,Method Bio.UniGene.STSLine.__init__()=Bio.UniGene.STSLine-class.html#__init__,Method Bio.UniGene.SequenceLine.__init__()=Bio.UniGene.SequenceLine-class.html#__init__,Method Bio.Wise.dnal.Statistics.__init__()=Bio.Wise.dnal.Statistics-class.html#__init__,Method Bio.Wise.psw.AlignmentColumn.__init__()=Bio.Wise.psw.AlignmentColumn-class.html#__init__,Method Bio.Wise.psw.ColumnUnit.__init__()=Bio.Wise.psw.ColumnUnit-class.html#__init__,Method Bio.bgzf.BgzfReader.__init__()=Bio.bgzf.BgzfReader-class.html#__init__,Method Bio.bgzf.BgzfWriter.__init__()=Bio.bgzf.BgzfWriter-class.html#__init__,Method Bio.codonalign.codonalignment.CodonAlignment.__init__()=Bio.codonalign.codonalignment.CodonAlignment-class.html#__init__,Method Bio.codonalign.codonseq.CodonSeq.__init__()=Bio.codonalign.codonseq.CodonSeq-class.html#__init__,Method Bio.kNN.kNN.__init__()=Bio.kNN.kNN-class.html#__init__,Method Bio.motifs.Instances.__init__()=Bio.motifs.Instances-class.html#__init__,Method Bio.motifs.Motif.__init__()=Bio.motifs.Motif-class.html#__init__,Method Bio.motifs.alignace.Record.__init__()=Bio.motifs.alignace.Record-class.html#__init__,Method Bio.motifs.applications._xxmotif.XXmotifCommandline.__init__()=Bio.motifs.applications._xxmotif.XXmotifCommandline-class.html#__init__,Method Bio.motifs.jaspar.Motif.__init__()=Bio.motifs.jaspar.Motif-class.html#__init__,Method Bio.motifs.jaspar.Record.__init__()=Bio.motifs.jaspar.Record-class.html#__init__,Method Bio.motifs.jaspar.db.JASPAR5.__init__()=Bio.motifs.jaspar.db.JASPAR5-class.html#__init__,Method Bio.motifs.mast.Record.__init__()=Bio.motifs.mast.Record-class.html#__init__,Method Bio.motifs.matrix.GenericPositionMatrix.__init__()=Bio.motifs.matrix.GenericPositionMatrix-class.html#__init__,Method Bio.motifs.matrix.PositionWeightMatrix.__init__()=Bio.motifs.matrix.PositionWeightMatrix-class.html#__init__,Method Bio.motifs.meme.Instance.__init__()=Bio.motifs.meme.Instance-class.html#__init__,Method Bio.motifs.meme.Motif.__init__()=Bio.motifs.meme.Motif-class.html#__init__,Method Bio.motifs.meme.Record.__init__()=Bio.motifs.meme.Record-class.html#__init__,Method Bio.motifs.thresholds.ScoreDistribution.__init__()=Bio.motifs.thresholds.ScoreDistribution-class.html#__init__,Method Bio.motifs.transfac.Record.__init__()=Bio.motifs.transfac.Record-class.html#__init__,Method Bio.pairwise2.affine_penalty.__init__()=Bio.pairwise2.affine_penalty-class.html#__init__,Method Bio.pairwise2.dictionary_match.__init__()=Bio.pairwise2.dictionary_match-class.html#__init__,Method Bio.pairwise2.identity_match.__init__()=Bio.pairwise2.identity_match-class.html#__init__,Method Bio.phenotype.phen_micro.JsonWriter.__init__()=Bio.phenotype.phen_micro.JsonWriter-class.html#__init__,Method Bio.phenotype.phen_micro.PlateRecord.__init__()=Bio.phenotype.phen_micro.PlateRecord-class.html#__init__,Method Bio.phenotype.phen_micro.WellRecord.__init__()=Bio.phenotype.phen_micro.WellRecord-class.html#__init__,Method BioSQL.BioSeq.DBSeq.__init__()=BioSQL.BioSeq.DBSeq-class.html#__init__,Method BioSQL.BioSeq.DBSeqRecord.__init__()=BioSQL.BioSeq.DBSeqRecord-class.html#__init__,Method BioSQL.BioSeqDatabase.Adaptor.__init__()=BioSQL.BioSeqDatabase.Adaptor-class.html#__init__,Method BioSQL.BioSeqDatabase.BioSeqDatabase.__init__()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#__init__,Method BioSQL.BioSeqDatabase.DBServer.__init__()=BioSQL.BioSeqDatabase.DBServer-class.html#__init__,Method BioSQL.BioSeqDatabase._CursorWrapper.__init__()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#__init__,Method BioSQL.DBUtils.Generic_dbutils.__init__()=BioSQL.DBUtils.Generic_dbutils-class.html#__init__,Method BioSQL.Loader.DatabaseLoader.__init__()=BioSQL.Loader.DatabaseLoader-class.html#__init__,Method BioSQL.Loader.DatabaseRemover.__init__()=BioSQL.Loader.DatabaseRemover-class.html#__init__"><a title="Bio.Affy.CelFile.ParserError.__init__
Bio.Affy.CelFile.Record.__init__
Bio.Align.AlignInfo.PSSM.__init__
Bio.Align.AlignInfo.SummaryInfo.__init__
Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline.__init__
Bio.Align.Applications._Clustalw.ClustalwCommandline.__init__
Bio.Align.Applications._Dialign.DialignCommandline.__init__
Bio.Align.Applications._MSAProbs.MSAProbsCommandline.__init__
Bio.Align.Applications._Mafft.MafftCommandline.__init__
Bio.Align.Applications._Muscle.MuscleCommandline.__init__
Bio.Align.Applications._Prank.PrankCommandline.__init__
Bio.Align.Applications._Probcons.ProbconsCommandline.__init__
Bio.Align.Applications._TCoffee.TCoffeeCommandline.__init__
Bio.Align.MultipleSeqAlignment.__init__
Bio.Align.PairwiseAlignment.__init__
Bio.Align.PairwiseAlignments.__init__
Bio.AlignIO.Interfaces.AlignmentIterator.__init__
Bio.AlignIO.Interfaces.AlignmentWriter.__init__
Bio.AlignIO.Interfaces.SequentialAlignmentWriter.__init__
Bio.AlignIO.MafIO.MafIndex.__init__
Bio.AlignIO.MauveIO.MauveWriter.__init__
Bio.Alphabet.AlphabetEncoder.__init__
Bio.Alphabet.Gapped.__init__
Bio.Alphabet.HasStopCodon.__init__
Bio.Application.AbstractCommandline.__init__
Bio.Application.ApplicationError.__init__
Bio.Application._AbstractParameter.__init__
Bio.Application._Argument.__init__
Bio.Application._Option.__init__
Bio.Application._StaticArgument.__init__
Bio.Application._Switch.__init__
Bio.Blast.Applications.NcbiblastformatterCommandline.__init__
Bio.Blast.Applications.NcbiblastnCommandline.__init__
Bio.Blast.Applications.NcbiblastpCommandline.__init__
Bio.Blast.Applications.NcbiblastxCommandline.__init__
Bio.Blast.Applications.NcbideltablastCommandline.__init__
Bio.Blast.Applications.NcbipsiblastCommandline.__init__
Bio.Blast.Applications.NcbirpsblastCommandline.__init__
Bio.Blast.Applications.NcbirpstblastnCommandline.__init__
Bio.Blast.Applications.NcbitblastnCommandline.__init__
Bio.Blast.Applications.NcbitblastxCommandline.__init__
Bio.Blast.Applications._NcbibaseblastCommandline.__init__
Bio.Blast.Applications._Ncbiblast2SeqCommandline.__init__
Bio.Blast.Applications._NcbiblastCommandline.__init__
Bio.Blast.Applications._NcbiblastMain2SeqCommandline.__init__
Bio.Blast.NCBIStandalone.BlastErrorParser.__init__
Bio.Blast.NCBIStandalone.BlastParser.__init__
Bio.Blast.NCBIStandalone.Iterator.__init__
Bio.Blast.NCBIStandalone.PSIBlastParser.__init__
Bio.Blast.NCBIStandalone._BlastConsumer.__init__
Bio.Blast.NCBIStandalone._BlastErrorConsumer.__init__
Bio.Blast.NCBIStandalone._PSIBlastConsumer.__init__
Bio.Blast.NCBIXML.BlastParser.__init__
Bio.Blast.NCBIXML._XMLparser.__init__
Bio.Blast.ParseBlastTable.BlastTableEntry.__init__
Bio.Blast.ParseBlastTable.BlastTableReader.__init__
Bio.Blast.ParseBlastTable.BlastTableRec.__init__
Bio.Blast.Record.Alignment.__init__
Bio.Blast.Record.Blast.__init__
Bio.Blast.Record.DatabaseReport.__init__
Bio.Blast.Record.Description.__init__
Bio.Blast.Record.HSP.__init__
Bio.Blast.Record.Header.__init__
Bio.Blast.Record.MultipleAlignment.__init__
Bio.Blast.Record.PSIBlast.__init__
Bio.Blast.Record.Parameters.__init__
Bio.Blast.Record.Round.__init__
Bio.CAPS.CAPSMap.__init__
Bio.CAPS.DifferentialCutsite.__init__
Bio.Cluster.Record.__init__
Bio.Compass.Record.__init__
Bio.Crystal.Chain.__init__
Bio.Crystal.Crystal.__init__
Bio.Crystal.Hetero.__init__
Bio.Data.CodonTable.AmbiguousCodonTable.__init__
Bio.Data.CodonTable.AmbiguousForwardTable.__init__
Bio.Data.CodonTable.CodonTable.__init__
Bio.Data.CodonTable.NCBICodonTable.__init__
Bio.Emboss.Applications.DiffseqCommandline.__init__
Bio.Emboss.Applications.EInvertedCommandline.__init__
Bio.Emboss.Applications.ETandemCommandline.__init__
Bio.Emboss.Applications.Est2GenomeCommandline.__init__
Bio.Emboss.Applications.FConsenseCommandline.__init__
Bio.Emboss.Applications.FDNADistCommandline.__init__
Bio.Emboss.Applications.FDNAParsCommandline.__init__
Bio.Emboss.Applications.FNeighborCommandline.__init__
Bio.Emboss.Applications.FProtDistCommandline.__init__
Bio.Emboss.Applications.FProtParsCommandline.__init__
Bio.Emboss.Applications.FSeqBootCommandline.__init__
Bio.Emboss.Applications.FTreeDistCommandline.__init__
Bio.Emboss.Applications.FuzznucCommandline.__init__
Bio.Emboss.Applications.FuzzproCommandline.__init__
Bio.Emboss.Applications.IepCommandline.__init__
Bio.Emboss.Applications.NeedleCommandline.__init__
Bio.Emboss.Applications.NeedleallCommandline.__init__
Bio.Emboss.Applications.PalindromeCommandline.__init__
Bio.Emboss.Applications.Primer3Commandline.__init__
Bio.Emboss.Applications.PrimerSearchCommandline.__init__
Bio.Emboss.Applications.SeqmatchallCommandline.__init__
Bio.Emboss.Applications.SeqretCommandline.__init__
Bio.Emboss.Applications.StretcherCommandline.__init__
Bio.Emboss.Applications.TranalignCommandline.__init__
Bio.Emboss.Applications.WaterCommandline.__init__
Bio.Emboss.Applications._EmbossCommandLine.__init__
Bio.Emboss.Applications._EmbossMinimalCommandLine.__init__
Bio.Emboss.Primer3.Primers.__init__
Bio.Emboss.Primer3.Record.__init__
Bio.Emboss.PrimerSearch.Amplifier.__init__
Bio.Emboss.PrimerSearch.InputRecord.__init__
Bio.Emboss.PrimerSearch.OutputRecord.__init__
Bio.Entrez.Parser.Consumer.__init__
Bio.Entrez.Parser.CorruptedXMLError.__init__
Bio.Entrez.Parser.DataHandler.__init__
Bio.Entrez.Parser.DictionaryConsumer.__init__
Bio.Entrez.Parser.ErrorConsumer.__init__
Bio.Entrez.Parser.IntegerConsumer.__init__
Bio.Entrez.Parser.ListConsumer.__init__
Bio.Entrez.Parser.NotXMLError.__init__
Bio.Entrez.Parser.StringConsumer.__init__
Bio.Entrez.Parser.ValidationError.__init__
Bio.ExPASy.Enzyme.Record.__init__
Bio.ExPASy.Prodoc.Record.__init__
Bio.ExPASy.Prodoc.Reference.__init__
Bio.ExPASy.Prosite.Record.__init__
Bio.ExPASy.ScanProsite.ContentHandler.__init__
Bio.ExPASy.ScanProsite.Parser.__init__
Bio.ExPASy.ScanProsite.Record.__init__
Bio.ExPASy.cellosaurus.Record.__init__
Bio.FSSP.FSSPAlignDict.__init__
Bio.FSSP.FSSPAlignRec.__init__
Bio.FSSP.FSSPHeader.__init__
Bio.FSSP.FSSPSumRec.__init__
Bio.FSSP.FSSPTools.FSSPMultAlign.__init__
Bio.FSSP.PosAlign.__init__
Bio.FSSP.fssp_rec.fff_rec.__init__
Bio.File.UndoHandle.__init__
Bio.File._IndexedSeqFileDict.__init__
Bio.File._SQLiteManySeqFilesDict.__init__
Bio.GA.Crossover.General.SafeFitnessCrossover.__init__
Bio.GA.Crossover.GeneralPoint.GeneralPointCrossover.__init__
Bio.GA.Crossover.GeneralPoint.InterleaveCrossover.__init__
Bio.GA.Crossover.Point.SinglePointCrossover.__init__
Bio.GA.Crossover.TwoPoint.TwoPointCrossover.__init__
Bio.GA.Crossover.Uniform.UniformCrossover.__init__
Bio.GA.Evolver.GenerationEvolver.__init__
Bio.GA.Evolver.SteadyStateEvolver.__init__
Bio.GA.Mutation.General.SafeFitnessMutation.__init__
Bio.GA.Mutation.Simple.ConversionMutation.__init__
Bio.GA.Mutation.Simple.SinglePositionMutation.__init__
Bio.GA.Organism.Organism.__init__
Bio.GA.Repair.Stabilizing.AmbiguousRepair.__init__
Bio.GA.Selection.Abstract.AbstractSelection.__init__
Bio.GA.Selection.Diversity.DiversitySelection.__init__
Bio.GA.Selection.RouletteWheel.RouletteWheelSelection.__init__
Bio.GA.Selection.Tournament.TournamentSelection.__init__
Bio.GenBank.FeatureParser.__init__
Bio.GenBank.Iterator.__init__
Bio.GenBank.Record.Feature.__init__
Bio.GenBank.Record.Qualifier.__init__
Bio.GenBank.Record.Record.__init__
Bio.GenBank.Record.Reference.__init__
Bio.GenBank.RecordParser.__init__
Bio.GenBank.Scanner.InsdcScanner.__init__
Bio.GenBank._BaseGenBankConsumer.__init__
Bio.GenBank._FeatureConsumer.__init__
Bio.GenBank._RecordConsumer.__init__
Bio.GenBank.utils.FeatureValueCleaner.__init__
Bio.Geo.Record.Record.__init__
Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment.__init__
Bio.Graphics.BasicChromosome.Chromosome.__init__
Bio.Graphics.BasicChromosome.ChromosomeSegment.__init__
Bio.Graphics.BasicChromosome.Organism.__init__
Bio.Graphics.BasicChromosome.TelomereSegment.__init__
Bio.Graphics.BasicChromosome._ChromosomeComponent.__init__
Bio.Graphics.ColorSpiral.ColorSpiral.__init__
Bio.Graphics.Comparative.ComparativeScatterPlot.__init__
Bio.Graphics.DisplayRepresentation.ChromosomeCounts.__init__
Bio.Graphics.Distribution.BarChartDistribution.__init__
Bio.Graphics.Distribution.DistributionPage.__init__
Bio.Graphics.Distribution.LineDistribution.__init__
Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.__init__
Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.__init__
Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.__init__
Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.__init__
Bio.Graphics.GenomeDiagram._Diagram.Diagram.__init__
Bio.Graphics.GenomeDiagram._Feature.Feature.__init__
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__init__
Bio.Graphics.GenomeDiagram._Graph.GraphData.__init__
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__init__
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.__init__
Bio.Graphics.GenomeDiagram._Track.Track.__init__
Bio.Graphics.KGML_vis.KGMLCanvas.__init__
Bio.HMM.DynamicProgramming.AbstractDPAlgorithms.__init__
Bio.HMM.DynamicProgramming.LogDPAlgorithms.__init__
Bio.HMM.DynamicProgramming.ScaledDPAlgorithms.__init__
Bio.HMM.MarkovModel.HiddenMarkovModel.__init__
Bio.HMM.MarkovModel.MarkovModelBuilder.__init__
Bio.HMM.Trainer.AbstractTrainer.__init__
Bio.HMM.Trainer.BaumWelchTrainer.__init__
Bio.HMM.Trainer.KnownStateTrainer.__init__
Bio.HMM.Trainer.TrainingSequence.__init__
Bio.Index._InMemoryIndex.__init__
Bio.Index._ShelveIndex.__init__
Bio.KDTree.KDTree'.KDTree.__init__
Bio.KEGG.Compound.Record.__init__
Bio.KEGG.Enzyme.Record.__init__
Bio.KEGG.Gene.Record.__init__
Bio.KEGG.KGML.KGML_parser.KGMLParser.__init__
Bio.KEGG.KGML.KGML_pathway.Component.__init__
Bio.KEGG.KGML.KGML_pathway.Entry.__init__
Bio.KEGG.KGML.KGML_pathway.Graphics.__init__
Bio.KEGG.KGML.KGML_pathway.Pathway.__init__
Bio.KEGG.KGML.KGML_pathway.Reaction.__init__
Bio.KEGG.KGML.KGML_pathway.Relation.__init__
Bio.LogisticRegression.LogisticRegression.__init__
Bio.MarkovModel.MarkovModel.__init__
Bio.MaxEntropy.MaxEntropy.__init__
Bio.NMR.xpktools.Peaklist.__init__
Bio.NMR.xpktools.XpkEntry.__init__
Bio.NaiveBayes.NaiveBayes.__init__
Bio.NeuralNetwork.BackPropagation.Layer.AbstractLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.__init__
Bio.NeuralNetwork.BackPropagation.Network.BasicNetwork.__init__
Bio.NeuralNetwork.Gene.Motif.MotifCoder.__init__
Bio.NeuralNetwork.Gene.Motif.MotifFinder.__init__
Bio.NeuralNetwork.Gene.Pattern.PatternIO.__init__
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.__init__
Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness.__init__
Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder.__init__
Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness.__init__
Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator.__init__
Bio.NeuralNetwork.Gene.Schema.Schema.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaCoder.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaFactory.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaFinder.__init__
Bio.NeuralNetwork.Gene.Schema.SimpleFinisher.__init__
Bio.NeuralNetwork.Gene.Signature.SignatureCoder.__init__
Bio.NeuralNetwork.Gene.Signature.SignatureFinder.__init__
Bio.NeuralNetwork.StopTraining.ValidationIncreaseStop.__init__
Bio.NeuralNetwork.Training.ExampleManager.__init__
Bio.NeuralNetwork.Training.TrainingExample.__init__
Bio.Nexus.Nexus.Block.__init__
Bio.Nexus.Nexus.CharBuffer.__init__
Bio.Nexus.Nexus.Commandline.__init__
Bio.Nexus.Nexus.Nexus.__init__
Bio.Nexus.Nexus.StepMatrix.__init__
Bio.Nexus.Nodes.Chain.__init__
Bio.Nexus.Nodes.Node.__init__
Bio.Nexus.StandardData.StandardData.__init__
Bio.Nexus.Trees.NodeData.__init__
Bio.Nexus.Trees.Tree.__init__
Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap.__init__
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__init__
Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap.__init__
Bio.PDB.Atom.Atom.__init__
Bio.PDB.Atom.DisorderedAtom.__init__
Bio.PDB.Chain.Chain.__init__
Bio.PDB.DSSP'.DSSP.__init__
Bio.PDB.Dice.ChainSelector.__init__
Bio.PDB.Entity.DisorderedEntityWrapper.__init__
Bio.PDB.Entity.Entity.__init__
Bio.PDB.FragmentMapper'.Fragment.__init__
Bio.PDB.FragmentMapper'.FragmentMapper.__init__
Bio.PDB.HSExposure.ExposureCN.__init__
Bio.PDB.HSExposure.HSExposureCA.__init__
Bio.PDB.HSExposure.HSExposureCB.__init__
Bio.PDB.HSExposure._AbstractHSExposure.__init__
Bio.PDB.MMCIF2Dict.MMCIF2Dict.__init__
Bio.PDB.MMCIFParser'.FastMMCIFParser.__init__
Bio.PDB.MMCIFParser'.MMCIFParser.__init__
Bio.PDB.Model.Model.__init__
Bio.PDB.NACCESS.NACCESS.__init__
Bio.PDB.NACCESS.NACCESS_atomic.__init__
Bio.PDB.NeighborSearch'.NeighborSearch.__init__
Bio.PDB.PDBIO'.PDBIO.__init__
Bio.PDB.PDBList'.PDBList.__init__
Bio.PDB.PDBParser'.PDBParser.__init__
Bio.PDB.PSEA.PSEA.__init__
Bio.PDB.Polypeptide.CaPPBuilder.__init__
Bio.PDB.Polypeptide.PPBuilder.__init__
Bio.PDB.Polypeptide._PPBuilder.__init__
Bio.PDB.QCPSuperimposer.QCPSuperimposer.__init__
Bio.PDB.Residue.DisorderedResidue.__init__
Bio.PDB.Residue.Residue.__init__
Bio.PDB.ResidueDepth'.ResidueDepth.__init__
Bio.PDB.Structure.Structure.__init__
Bio.PDB.StructureAlignment'.StructureAlignment.__init__
Bio.PDB.StructureBuilder.StructureBuilder.__init__
Bio.PDB.Superimposer'.Superimposer.__init__
Bio.PDB.mmcifio.MMCIFIO.__init__
Bio.PDB.mmtf.DefaultParser.StructureDecoder.__init__
Bio.PDB.vectors.Vector.__init__
Bio.ParserSupport.TaggingConsumer.__init__
Bio.Pathway.Network.__init__
Bio.Pathway.Reaction.__init__
Bio.Pathway.Rep.Graph.Graph.__init__
Bio.Pathway.Rep.MultiGraph.MultiGraph.__init__
Bio.Pathway.System.__init__
Bio.Phylo.Applications._Fasttree.FastTreeCommandline.__init__
Bio.Phylo.Applications._Phyml.PhymlCommandline.__init__
Bio.Phylo.Applications._Raxml.RaxmlCommandline.__init__
Bio.Phylo.BaseTree.BranchColor.__init__
Bio.Phylo.BaseTree.Clade.__init__
Bio.Phylo.BaseTree.Tree.__init__
Bio.Phylo.CDAO.Clade.__init__
Bio.Phylo.CDAO.Tree.__init__
Bio.Phylo.CDAOIO.Parser.__init__
Bio.Phylo.CDAOIO.Writer.__init__
Bio.Phylo.NeXML.Clade.__init__
Bio.Phylo.NeXML.Tree.__init__
Bio.Phylo.NeXMLIO.Parser.__init__
Bio.Phylo.NeXMLIO.Writer.__init__
Bio.Phylo.Newick.Clade.__init__
Bio.Phylo.Newick.Tree.__init__
Bio.Phylo.NewickIO.Parser.__init__
Bio.Phylo.NewickIO.Writer.__init__
Bio.Phylo.PAML._paml.Paml.__init__
Bio.Phylo.PAML.baseml.Baseml.__init__
Bio.Phylo.PAML.codeml.Codeml.__init__
Bio.Phylo.PAML.yn00.Yn00.__init__
Bio.Phylo.PhyloXML.Accession.__init__
Bio.Phylo.PhyloXML.Annotation.__init__
Bio.Phylo.PhyloXML.BinaryCharacters.__init__
Bio.Phylo.PhyloXML.BranchColor.__init__
Bio.Phylo.PhyloXML.Clade.__init__
Bio.Phylo.PhyloXML.CladeRelation.__init__
Bio.Phylo.PhyloXML.Confidence.__init__
Bio.Phylo.PhyloXML.Date.__init__
Bio.Phylo.PhyloXML.Distribution.__init__
Bio.Phylo.PhyloXML.DomainArchitecture.__init__
Bio.Phylo.PhyloXML.Events.__init__
Bio.Phylo.PhyloXML.Id.__init__
Bio.Phylo.PhyloXML.MolSeq.__init__
Bio.Phylo.PhyloXML.Other.__init__
Bio.Phylo.PhyloXML.Phylogeny.__init__
Bio.Phylo.PhyloXML.Phyloxml.__init__
Bio.Phylo.PhyloXML.Point.__init__
Bio.Phylo.PhyloXML.Polygon.__init__
Bio.Phylo.PhyloXML.Property.__init__
Bio.Phylo.PhyloXML.ProteinDomain.__init__
Bio.Phylo.PhyloXML.Reference.__init__
Bio.Phylo.PhyloXML.Sequence.__init__
Bio.Phylo.PhyloXML.SequenceRelation.__init__
Bio.Phylo.PhyloXML.Taxonomy.__init__
Bio.Phylo.PhyloXML.Uri.__init__
Bio.Phylo.PhyloXMLIO.Parser.__init__
Bio.Phylo.PhyloXMLIO.Writer.__init__
Bio.Phylo.TreeConstruction.DistanceCalculator.__init__
Bio.Phylo.TreeConstruction.DistanceMatrix.__init__
Bio.Phylo.TreeConstruction.DistanceTreeConstructor.__init__
Bio.Phylo.TreeConstruction.NNITreeSearcher.__init__
Bio.Phylo.TreeConstruction.ParsimonyScorer.__init__
Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor.__init__
Bio.Phylo.TreeConstruction._Matrix.__init__
Bio.PopGen.GenePop.Controller.GenePopController.__init__
Bio.PopGen.GenePop.Controller._FileIterator.__init__
Bio.PopGen.GenePop.Controller._GenePopCommandline.__init__
Bio.PopGen.GenePop.EasyController.EasyController.__init__
Bio.PopGen.GenePop.FileParser.FileRecord.__init__
Bio.PopGen.GenePop.LargeFileParser.Record.__init__
Bio.PopGen.GenePop.Record.__init__
Bio.Restriction.PrintFormat.PrintFormat.__init__
Bio.Restriction.Restriction.Analysis.__init__
Bio.Restriction.Restriction.FormattedSeq.__init__
Bio.Restriction.Restriction.RestrictionBatch.__init__
Bio.Restriction.Restriction.RestrictionType.__init__
Bio.SCOP.Astral.__init__
Bio.SCOP.Cla.Index.__init__
Bio.SCOP.Cla.Record.__init__
Bio.SCOP.Des.Record.__init__
Bio.SCOP.Dom.Record.__init__
Bio.SCOP.Domain.__init__
Bio.SCOP.Hie.Record.__init__
Bio.SCOP.Node.__init__
Bio.SCOP.Raf.Res.__init__
Bio.SCOP.Raf.SeqMap.__init__
Bio.SCOP.Raf.SeqMapIndex.__init__
Bio.SCOP.Residues.Residues.__init__
Bio.SCOP.Scop.__init__
Bio.SVDSuperimposer.SVDSuperimposer.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabParser.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.__init__
Bio.SearchIO.BlastIO.blast_text.BlastTextParser.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.__init__
Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator.__init__
Bio.SearchIO.BlatIO.BlatPslIndexer.__init__
Bio.SearchIO.BlatIO.BlatPslParser.__init__
Bio.SearchIO.BlatIO.BlatPslWriter.__init__
Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.__init__
Bio.SearchIO.FastaIO.FastaM10Indexer.__init__
Bio.SearchIO.FastaIO.FastaM10Parser.__init__
Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer.__init__
Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser.__init__
Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.__init__
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser.__init__
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.__init__
Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser.__init__
Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser.__init__
Bio.SearchIO._index.SearchIndexer.__init__
Bio.SearchIO._model.hit.Hit.__init__
Bio.SearchIO._model.hsp.HSP.__init__
Bio.SearchIO._model.hsp.HSPFragment.__init__
Bio.SearchIO._model.query.QueryResult.__init__
Bio.Seq.MutableSeq.__init__
Bio.Seq.Seq.__init__
Bio.Seq.UnknownSeq.__init__
Bio.SeqFeature.CompoundLocation.__init__
Bio.SeqFeature.FeatureLocation.__init__
Bio.SeqFeature.PositionGap.__init__
Bio.SeqFeature.Reference.__init__
Bio.SeqFeature.SeqFeature.__init__
Bio.SeqIO.FastaIO.FastaTwoLineWriter.__init__
Bio.SeqIO.FastaIO.FastaWriter.__init__
Bio.SeqIO.Interfaces.SequenceIterator.__init__
Bio.SeqIO.Interfaces.SequenceWriter.__init__
Bio.SeqIO.Interfaces.SequentialSequenceWriter.__init__
Bio.SeqIO.PhdIO.PhdWriter.__init__
Bio.SeqIO.PirIO.PirWriter.__init__
Bio.SeqIO.QualityIO.QualPhredWriter.__init__
Bio.SeqIO.SeqXmlIO.SeqXmlIterator.__init__
Bio.SeqIO.SeqXmlIO.SeqXmlWriter.__init__
Bio.SeqIO.SeqXmlIO.XMLRecordIterator.__init__
Bio.SeqIO.SffIO.SffWriter.__init__
Bio.SeqIO.SffIO._AddTellHandle.__init__
Bio.SeqIO.UniprotIO.Parser.__init__
Bio.SeqIO._index.IntelliGeneticsRandomAccess.__init__
Bio.SeqIO._index.SeqFileRandomAccess.__init__
Bio.SeqIO._index.SequentialSeqFileRandomAccess.__init__
Bio.SeqIO._index.SffRandomAccess.__init__
Bio.SeqRecord.SeqRecord.__init__
Bio.SeqRecord._RestrictedDict.__init__
Bio.SeqUtils.CodonUsage.CodonAdaptationIndex.__init__
Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.__init__
Bio.SeqUtils.ProtParam.ProteinAnalysis.__init__
Bio.Sequencing.Ace.ACEFileRecord.__init__
Bio.Sequencing.Ace.Contig.__init__
Bio.Sequencing.Ace.Reads.__init__
Bio.Sequencing.Ace.af.__init__
Bio.Sequencing.Ace.bs.__init__
Bio.Sequencing.Ace.ct.__init__
Bio.Sequencing.Ace.ds.__init__
Bio.Sequencing.Ace.qa.__init__
Bio.Sequencing.Ace.rd.__init__
Bio.Sequencing.Ace.rt.__init__
Bio.Sequencing.Ace.wa.__init__
Bio.Sequencing.Ace.wr.__init__
Bio.Sequencing.Applications._Novoalign.NovoalignCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaAlignCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaBwaswCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaIndexCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaSampeCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaSamseCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsVersion0xSortCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsVersion1xSortCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline.__init__
Bio.Sequencing.Phd.Record.__init__
Bio.SubsMat.FreqTable.FreqTable.__init__
Bio.SubsMat.SeqMat.__init__
Bio.SwissProt.KeyWList.Record.__init__
Bio.SwissProt.Record.__init__
Bio.SwissProt.Reference.__init__
Bio.UniGene.ProtsimLine.__init__
Bio.UniGene.Record.__init__
Bio.UniGene.STSLine.__init__
Bio.UniGene.SequenceLine.__init__
Bio.Wise.dnal.Statistics.__init__
Bio.Wise.psw.AlignmentColumn.__init__
Bio.Wise.psw.ColumnUnit.__init__
Bio.bgzf.BgzfReader.__init__
Bio.bgzf.BgzfWriter.__init__
Bio.codonalign.codonalignment.CodonAlignment.__init__
Bio.codonalign.codonseq.CodonSeq.__init__
Bio.kNN.kNN.__init__
Bio.motifs.Instances.__init__
Bio.motifs.Motif.__init__
Bio.motifs.alignace.Record.__init__
Bio.motifs.applications._xxmotif.XXmotifCommandline.__init__
Bio.motifs.jaspar.Motif.__init__
Bio.motifs.jaspar.Record.__init__
Bio.motifs.jaspar.db.JASPAR5.__init__
Bio.motifs.mast.Record.__init__
Bio.motifs.matrix.GenericPositionMatrix.__init__
Bio.motifs.matrix.PositionWeightMatrix.__init__
Bio.motifs.meme.Instance.__init__
Bio.motifs.meme.Motif.__init__
Bio.motifs.meme.Record.__init__
Bio.motifs.thresholds.ScoreDistribution.__init__
Bio.motifs.transfac.Record.__init__
Bio.pairwise2.affine_penalty.__init__
Bio.pairwise2.dictionary_match.__init__
Bio.pairwise2.identity_match.__init__
Bio.phenotype.phen_micro.JsonWriter.__init__
Bio.phenotype.phen_micro.PlateRecord.__init__
Bio.phenotype.phen_micro.WellRecord.__init__
BioSQL.BioSeq.DBSeq.__init__
BioSQL.BioSeq.DBSeqRecord.__init__
BioSQL.BioSeqDatabase.Adaptor.__init__
BioSQL.BioSeqDatabase.BioSeqDatabase.__init__
BioSQL.BioSeqDatabase.DBServer.__init__
BioSQL.BioSeqDatabase._CursorWrapper.__init__
BioSQL.DBUtils.Generic_dbutils.__init__
BioSQL.Loader.DatabaseLoader.__init__
BioSQL.Loader.DatabaseRemover.__init__" class="py-name" href="#" onclick="return doclink('link-183', '__init__', 'link-183');">__init__</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-op">*</tt><tt class="py-name">args</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L450"></a><tt class="py-lineno"> 450</tt>  <tt class="py-line"> </tt>
<a name="Accession"></a><div id="Accession-def"><a name="L451"></a><tt class="py-lineno"> 451</tt>  <tt class="py-line"> </tt>
<a name="L452"></a><tt class="py-lineno"> 452</tt>  <tt class="py-line"><tt class="py-comment"># PhyloXML-specific complex types</tt> </tt>
<a name="L453"></a><tt class="py-lineno"> 453</tt>  <tt class="py-line"> </tt>
<a name="L454"></a><tt class="py-lineno"> 454</tt> <a class="py-toggle" href="#" id="Accession-toggle" onclick="return toggle('Accession');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Accession-class.html">Accession</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Accession-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Accession-expanded"><a name="L455"></a><tt class="py-lineno"> 455</tt>  <tt class="py-line">    <tt class="py-docstring">"""Captures the local part in a sequence identifier.</tt> </tt>
<a name="L456"></a><tt class="py-lineno"> 456</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L457"></a><tt class="py-lineno"> 457</tt>  <tt class="py-line"><tt class="py-docstring">    Example: In ``UniProtKB:P17304``, the Accession instance attribute ``value``</tt> </tt>
<a name="L458"></a><tt class="py-lineno"> 458</tt>  <tt class="py-line"><tt class="py-docstring">    is 'P17304' and the ``source`` attribute is 'UniProtKB'.</tt> </tt>
<a name="L459"></a><tt class="py-lineno"> 459</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L460"></a><tt class="py-lineno"> 460</tt>  <tt class="py-line"> </tt>
<a name="Accession.__init__"></a><div id="Accession.__init__-def"><a name="L461"></a><tt class="py-lineno"> 461</tt> <a class="py-toggle" href="#" id="Accession.__init__-toggle" onclick="return toggle('Accession.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Accession-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">,</tt> <tt class="py-param">source</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Accession.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Accession.__init__-expanded"><a name="L462"></a><tt class="py-lineno"> 462</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize value for Accession object."""</tt> </tt>
<a name="L463"></a><tt class="py-lineno"> 463</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-184" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-184', 'value', 'link-37');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-185" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-185', 'value', 'link-37');">value</a></tt> </tt>
<a name="L464"></a><tt class="py-lineno"> 464</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-186" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.source()=Bio.GenBank._FeatureConsumer-class.html#source,Method Bio.GenBank._RecordConsumer.source()=Bio.GenBank._RecordConsumer-class.html#source,Method Bio.Pathway.Network.source()=Bio.Pathway.Network-class.html#source"><a title="Bio.GenBank._FeatureConsumer.source
Bio.GenBank._RecordConsumer.source
Bio.Pathway.Network.source" class="py-name" href="#" onclick="return doclink('link-186', 'source', 'link-186');">source</a></tt> <tt class="py-op">=</tt> <tt id="link-187" class="py-name"><a title="Bio.GenBank._FeatureConsumer.source
Bio.GenBank._RecordConsumer.source
Bio.Pathway.Network.source" class="py-name" href="#" onclick="return doclink('link-187', 'source', 'link-186');">source</a></tt> </tt>
</div><a name="L465"></a><tt class="py-lineno"> 465</tt>  <tt class="py-line"> </tt>
<a name="Accession.__str__"></a><div id="Accession.__str__-def"><a name="L466"></a><tt class="py-lineno"> 466</tt> <a class="py-toggle" href="#" id="Accession.__str__-toggle" onclick="return toggle('Accession.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Accession-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Accession.__str__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Accession.__str__-expanded"><a name="L467"></a><tt class="py-lineno"> 467</tt>  <tt class="py-line">        <tt class="py-docstring">"""Show the class name and an identifying attribute."""</tt> </tt>
<a name="L468"></a><tt class="py-lineno"> 468</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">'%s:%s'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-188" class="py-name"><a title="Bio.GenBank._FeatureConsumer.source
Bio.GenBank._RecordConsumer.source
Bio.Pathway.Network.source" class="py-name" href="#" onclick="return doclink('link-188', 'source', 'link-186');">source</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-189" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-189', 'value', 'link-37');">value</a></tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L469"></a><tt class="py-lineno"> 469</tt>  <tt class="py-line"> </tt>
<a name="Annotation"></a><div id="Annotation-def"><a name="L470"></a><tt class="py-lineno"> 470</tt>  <tt class="py-line"> </tt>
<a name="L471"></a><tt class="py-lineno"> 471</tt> <a class="py-toggle" href="#" id="Annotation-toggle" onclick="return toggle('Annotation');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Annotation-class.html">Annotation</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Annotation-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Annotation-expanded"><a name="L472"></a><tt class="py-lineno"> 472</tt>  <tt class="py-line">    <tt class="py-docstring">"""The annotation of a molecular sequence.</tt> </tt>
<a name="L473"></a><tt class="py-lineno"> 473</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L474"></a><tt class="py-lineno"> 474</tt>  <tt class="py-line"><tt class="py-docstring">    It is recommended to annotate by using the optional 'ref' attribute.</tt> </tt>
<a name="L475"></a><tt class="py-lineno"> 475</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L476"></a><tt class="py-lineno"> 476</tt>  <tt class="py-line"><tt class="py-docstring">    :Parameters:</tt> </tt>
<a name="L477"></a><tt class="py-lineno"> 477</tt>  <tt class="py-line"><tt class="py-docstring">        ref : string</tt> </tt>
<a name="L478"></a><tt class="py-lineno"> 478</tt>  <tt class="py-line"><tt class="py-docstring">            reference string, e.g. 'GO:0008270',</tt> </tt>
<a name="L479"></a><tt class="py-lineno"> 479</tt>  <tt class="py-line"><tt class="py-docstring">            'KEGG:Tetrachloroethene degradation', 'EC:1.1.1.1'</tt> </tt>
<a name="L480"></a><tt class="py-lineno"> 480</tt>  <tt class="py-line"><tt class="py-docstring">        source : string</tt> </tt>
<a name="L481"></a><tt class="py-lineno"> 481</tt>  <tt class="py-line"><tt class="py-docstring">            plain-text source for this annotation</tt> </tt>
<a name="L482"></a><tt class="py-lineno"> 482</tt>  <tt class="py-line"><tt class="py-docstring">        evidence : str</tt> </tt>
<a name="L483"></a><tt class="py-lineno"> 483</tt>  <tt class="py-line"><tt class="py-docstring">            describe evidence as free text (e.g. 'experimental')</tt> </tt>
<a name="L484"></a><tt class="py-lineno"> 484</tt>  <tt class="py-line"><tt class="py-docstring">        desc : string</tt> </tt>
<a name="L485"></a><tt class="py-lineno"> 485</tt>  <tt class="py-line"><tt class="py-docstring">            free text description</tt> </tt>
<a name="L486"></a><tt class="py-lineno"> 486</tt>  <tt class="py-line"><tt class="py-docstring">        confidence : Confidence</tt> </tt>
<a name="L487"></a><tt class="py-lineno"> 487</tt>  <tt class="py-line"><tt class="py-docstring">            state the type and value of support (type Confidence)</tt> </tt>
<a name="L488"></a><tt class="py-lineno"> 488</tt>  <tt class="py-line"><tt class="py-docstring">        properties : list</tt> </tt>
<a name="L489"></a><tt class="py-lineno"> 489</tt>  <tt class="py-line"><tt class="py-docstring">            typed and referenced annotations from external resources</tt> </tt>
<a name="L490"></a><tt class="py-lineno"> 490</tt>  <tt class="py-line"><tt class="py-docstring">        uri : Uri</tt> </tt>
<a name="L491"></a><tt class="py-lineno"> 491</tt>  <tt class="py-line"><tt class="py-docstring">            link</tt> </tt>
<a name="L492"></a><tt class="py-lineno"> 492</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L493"></a><tt class="py-lineno"> 493</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L494"></a><tt class="py-lineno"> 494</tt>  <tt class="py-line"> </tt>
<a name="L495"></a><tt class="py-lineno"> 495</tt>  <tt class="py-line">    <tt id="link-190" class="py-name" targets="Variable Bio.Phylo.PhyloXML.Annotation.re_ref=Bio.Phylo.PhyloXML.Annotation-class.html#re_ref,Variable Bio.Phylo.PhyloXML.Property.re_ref=Bio.Phylo.PhyloXML.Property-class.html#re_ref"><a title="Bio.Phylo.PhyloXML.Annotation.re_ref
Bio.Phylo.PhyloXML.Property.re_ref" class="py-name" href="#" onclick="return doclink('link-190', 're_ref', 'link-190');">re_ref</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'[a-zA-Z0-9_]+:[a-zA-Z0-9_\.\-\s]+'</tt><tt class="py-op">)</tt> </tt>
<a name="L496"></a><tt class="py-lineno"> 496</tt>  <tt class="py-line"> </tt>
<a name="Annotation.__init__"></a><div id="Annotation.__init__-def"><a name="L497"></a><tt class="py-lineno"> 497</tt> <a class="py-toggle" href="#" id="Annotation.__init__-toggle" onclick="return toggle('Annotation.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Annotation-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> </tt>
<a name="L498"></a><tt class="py-lineno"> 498</tt>  <tt class="py-line">                 <tt class="py-comment"># Attributes</tt> </tt>
<a name="L499"></a><tt class="py-lineno"> 499</tt>  <tt class="py-line">                 <tt class="py-param">ref</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">source</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">evidence</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">type</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L500"></a><tt class="py-lineno"> 500</tt>  <tt class="py-line">                 <tt class="py-comment"># Child nodes</tt> </tt>
<a name="L501"></a><tt class="py-lineno"> 501</tt>  <tt class="py-line">                 <tt class="py-param">desc</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">confidence</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">uri</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L502"></a><tt class="py-lineno"> 502</tt>  <tt class="py-line">                 <tt class="py-comment"># Collection</tt> </tt>
<a name="L503"></a><tt class="py-lineno"> 503</tt>  <tt class="py-line">                 <tt class="py-param">properties</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Annotation.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Annotation.__init__-expanded"><a name="L504"></a><tt class="py-lineno"> 504</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize value for the Annotation object."""</tt> </tt>
<a name="L505"></a><tt class="py-lineno"> 505</tt>  <tt class="py-line">        <tt id="link-191" class="py-name" targets="Function Bio.Phylo.PhyloXML._check_str()=Bio.Phylo.PhyloXML-module.html#_check_str"><a title="Bio.Phylo.PhyloXML._check_str" class="py-name" href="#" onclick="return doclink('link-191', '_check_str', 'link-191');">_check_str</a></tt><tt class="py-op">(</tt><tt id="link-192" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.ref=Bio.SeqFeature.CompoundLocation-class.html#ref,Variable Bio.SeqFeature.SeqFeature.ref=Bio.SeqFeature.SeqFeature-class.html#ref"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-192', 'ref', 'link-192');">ref</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-193" class="py-name"><a title="Bio.Phylo.PhyloXML.Annotation.re_ref
Bio.Phylo.PhyloXML.Property.re_ref" class="py-name" href="#" onclick="return doclink('link-193', 're_ref', 'link-190');">re_ref</a></tt><tt class="py-op">.</tt><tt id="link-194" class="py-name" targets="Function Bio.triefind.match()=Bio.triefind-module.html#match"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-194', 'match', 'link-194');">match</a></tt><tt class="py-op">)</tt> </tt>
<a name="L506"></a><tt class="py-lineno"> 506</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-195" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-195', 'ref', 'link-192');">ref</a></tt> <tt class="py-op">=</tt> <tt id="link-196" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-196', 'ref', 'link-192');">ref</a></tt> </tt>
<a name="L507"></a><tt class="py-lineno"> 507</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-197" class="py-name"><a title="Bio.GenBank._FeatureConsumer.source
Bio.GenBank._RecordConsumer.source
Bio.Pathway.Network.source" class="py-name" href="#" onclick="return doclink('link-197', 'source', 'link-186');">source</a></tt> <tt class="py-op">=</tt> <tt id="link-198" class="py-name"><a title="Bio.GenBank._FeatureConsumer.source
Bio.GenBank._RecordConsumer.source
Bio.Pathway.Network.source" class="py-name" href="#" onclick="return doclink('link-198', 'source', 'link-186');">source</a></tt> </tt>
<a name="L508"></a><tt class="py-lineno"> 508</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">evidence</tt> <tt class="py-op">=</tt> <tt class="py-name">evidence</tt> </tt>
<a name="L509"></a><tt class="py-lineno"> 509</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-199" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-199', 'type', 'link-29');">type</a></tt> <tt class="py-op">=</tt> <tt id="link-200" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-200', 'type', 'link-29');">type</a></tt> </tt>
<a name="L510"></a><tt class="py-lineno"> 510</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-201" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.desc()=Bio.Phylo.PhyloXMLIO.Writer-class.html#desc"><a title="Bio.Phylo.PhyloXMLIO.Writer.desc" class="py-name" href="#" onclick="return doclink('link-201', 'desc', 'link-201');">desc</a></tt> <tt class="py-op">=</tt> <tt id="link-202" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.desc" class="py-name" href="#" onclick="return doclink('link-202', 'desc', 'link-201');">desc</a></tt> </tt>
<a name="L511"></a><tt class="py-lineno"> 511</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-203" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-203', 'confidence', 'link-94');">confidence</a></tt> <tt class="py-op">=</tt> <tt id="link-204" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-204', 'confidence', 'link-94');">confidence</a></tt> </tt>
<a name="L512"></a><tt class="py-lineno"> 512</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-205" class="py-name" targets="Variable Bio.Phylo.NeXMLIO.uri=Bio.Phylo.NeXMLIO-module.html#uri,Method Bio.Phylo.PhyloXMLIO.Parser.uri()=Bio.Phylo.PhyloXMLIO.Parser-class.html#uri,Method Bio.Phylo.PhyloXMLIO.Writer.uri()=Bio.Phylo.PhyloXMLIO.Writer-class.html#uri,Variable Bio.Phylo.PhyloXMLIO.uri=Bio.Phylo.PhyloXMLIO-module.html#uri"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-205', 'uri', 'link-205');">uri</a></tt> <tt class="py-op">=</tt> <tt id="link-206" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-206', 'uri', 'link-205');">uri</a></tt> </tt>
<a name="L513"></a><tt class="py-lineno"> 513</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">properties</tt> <tt class="py-op">=</tt> <tt class="py-name">properties</tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div></div><a name="L514"></a><tt class="py-lineno"> 514</tt>  <tt class="py-line"> </tt>
<a name="BinaryCharacters"></a><div id="BinaryCharacters-def"><a name="L515"></a><tt class="py-lineno"> 515</tt>  <tt class="py-line"> </tt>
<a name="L516"></a><tt class="py-lineno"> 516</tt> <a class="py-toggle" href="#" id="BinaryCharacters-toggle" onclick="return toggle('BinaryCharacters');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.BinaryCharacters-class.html">BinaryCharacters</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BinaryCharacters-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="BinaryCharacters-expanded"><a name="L517"></a><tt class="py-lineno"> 517</tt>  <tt class="py-line">    <tt class="py-docstring">"""Binary characters at the root of a clade.</tt> </tt>
<a name="L518"></a><tt class="py-lineno"> 518</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L519"></a><tt class="py-lineno"> 519</tt>  <tt class="py-line"><tt class="py-docstring">    The names and/or counts of binary characters present, gained, and lost</tt> </tt>
<a name="L520"></a><tt class="py-lineno"> 520</tt>  <tt class="py-line"><tt class="py-docstring">    at the root of a clade.</tt> </tt>
<a name="L521"></a><tt class="py-lineno"> 521</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L522"></a><tt class="py-lineno"> 522</tt>  <tt class="py-line"> </tt>
<a name="BinaryCharacters.__init__"></a><div id="BinaryCharacters.__init__-def"><a name="L523"></a><tt class="py-lineno"> 523</tt> <a class="py-toggle" href="#" id="BinaryCharacters.__init__-toggle" onclick="return toggle('BinaryCharacters.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.BinaryCharacters-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> </tt>
<a name="L524"></a><tt class="py-lineno"> 524</tt>  <tt class="py-line">                 <tt class="py-comment"># Attributes</tt> </tt>
<a name="L525"></a><tt class="py-lineno"> 525</tt>  <tt class="py-line">                 <tt class="py-param">type</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">gained_count</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">lost_count</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">present_count</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L526"></a><tt class="py-lineno"> 526</tt>  <tt class="py-line">                 <tt class="py-param">absent_count</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L527"></a><tt class="py-lineno"> 527</tt>  <tt class="py-line">                 <tt class="py-comment"># Child nodes (flattened into collections)</tt> </tt>
<a name="L528"></a><tt class="py-lineno"> 528</tt>  <tt class="py-line">                 <tt class="py-param">gained</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">lost</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">present</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">absent</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BinaryCharacters.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="BinaryCharacters.__init__-expanded"><a name="L529"></a><tt class="py-lineno"> 529</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize values for the BinaryCharacters object."""</tt> </tt>
<a name="L530"></a><tt class="py-lineno"> 530</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-207" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-207', 'type', 'link-29');">type</a></tt> <tt class="py-op">=</tt> <tt id="link-208" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-208', 'type', 'link-29');">type</a></tt> </tt>
<a name="L531"></a><tt class="py-lineno"> 531</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">gained_count</tt> <tt class="py-op">=</tt> <tt class="py-name">gained_count</tt> </tt>
<a name="L532"></a><tt class="py-lineno"> 532</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">lost_count</tt> <tt class="py-op">=</tt> <tt class="py-name">lost_count</tt> </tt>
<a name="L533"></a><tt class="py-lineno"> 533</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">present_count</tt> <tt class="py-op">=</tt> <tt class="py-name">present_count</tt> </tt>
<a name="L534"></a><tt class="py-lineno"> 534</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">absent_count</tt> <tt class="py-op">=</tt> <tt class="py-name">absent_count</tt> </tt>
<a name="L535"></a><tt class="py-lineno"> 535</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">gained</tt> <tt class="py-op">=</tt> <tt class="py-name">gained</tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L536"></a><tt class="py-lineno"> 536</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">lost</tt> <tt class="py-op">=</tt> <tt class="py-name">lost</tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L537"></a><tt class="py-lineno"> 537</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">present</tt> <tt class="py-op">=</tt> <tt class="py-name">present</tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L538"></a><tt class="py-lineno"> 538</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">absent</tt> <tt class="py-op">=</tt> <tt class="py-name">absent</tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div></div><a name="L539"></a><tt class="py-lineno"> 539</tt>  <tt class="py-line"> </tt>
<a name="CladeRelation"></a><div id="CladeRelation-def"><a name="L540"></a><tt class="py-lineno"> 540</tt>  <tt class="py-line"> </tt>
<a name="L541"></a><tt class="py-lineno"> 541</tt> <a class="py-toggle" href="#" id="CladeRelation-toggle" onclick="return toggle('CladeRelation');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.CladeRelation-class.html">CladeRelation</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CladeRelation-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="CladeRelation-expanded"><a name="L542"></a><tt class="py-lineno"> 542</tt>  <tt class="py-line">    <tt class="py-docstring">"""Expresses a typed relationship between two clades.</tt> </tt>
<a name="L543"></a><tt class="py-lineno"> 543</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L544"></a><tt class="py-lineno"> 544</tt>  <tt class="py-line"><tt class="py-docstring">    For example, this could be used to describe multiple parents of a clade.</tt> </tt>
<a name="L545"></a><tt class="py-lineno"> 545</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L546"></a><tt class="py-lineno"> 546</tt>  <tt class="py-line"><tt class="py-docstring">    :type id_ref_0: str</tt> </tt>
<a name="L547"></a><tt class="py-lineno"> 547</tt>  <tt class="py-line"><tt class="py-docstring">    :type id_ref_1: str</tt> </tt>
<a name="L548"></a><tt class="py-lineno"> 548</tt>  <tt class="py-line"><tt class="py-docstring">    :type distance: str</tt> </tt>
<a name="L549"></a><tt class="py-lineno"> 549</tt>  <tt class="py-line"><tt class="py-docstring">    :type type: str</tt> </tt>
<a name="L550"></a><tt class="py-lineno"> 550</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L551"></a><tt class="py-lineno"> 551</tt>  <tt class="py-line"><tt class="py-docstring">    :type confidence: Confidence</tt> </tt>
<a name="L552"></a><tt class="py-lineno"> 552</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L553"></a><tt class="py-lineno"> 553</tt>  <tt class="py-line"> </tt>
<a name="CladeRelation.__init__"></a><div id="CladeRelation.__init__-def"><a name="L554"></a><tt class="py-lineno"> 554</tt> <a class="py-toggle" href="#" id="CladeRelation.__init__-toggle" onclick="return toggle('CladeRelation.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.CladeRelation-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">type</tt><tt class="py-op">,</tt> <tt class="py-param">id_ref_0</tt><tt class="py-op">,</tt> <tt class="py-param">id_ref_1</tt><tt class="py-op">,</tt> </tt>
<a name="L555"></a><tt class="py-lineno"> 555</tt>  <tt class="py-line">                 <tt class="py-param">distance</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">confidence</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CladeRelation.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CladeRelation.__init__-expanded"><a name="L556"></a><tt class="py-lineno"> 556</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize values for the CladeRelation object."""</tt> </tt>
<a name="L557"></a><tt class="py-lineno"> 557</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-209" class="py-name" targets="Method Bio.Nexus.Trees.Tree.distance()=Bio.Nexus.Trees.Tree-class.html#distance,Method Bio.Phylo.BaseTree.TreeMixin.distance()=Bio.Phylo.BaseTree.TreeMixin-class.html#distance"><a title="Bio.Nexus.Trees.Tree.distance
Bio.Phylo.BaseTree.TreeMixin.distance" class="py-name" href="#" onclick="return doclink('link-209', 'distance', 'link-209');">distance</a></tt> <tt class="py-op">=</tt> <tt id="link-210" class="py-name"><a title="Bio.Nexus.Trees.Tree.distance
Bio.Phylo.BaseTree.TreeMixin.distance" class="py-name" href="#" onclick="return doclink('link-210', 'distance', 'link-209');">distance</a></tt> </tt>
<a name="L558"></a><tt class="py-lineno"> 558</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-211" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-211', 'type', 'link-29');">type</a></tt> <tt class="py-op">=</tt> <tt id="link-212" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-212', 'type', 'link-29');">type</a></tt> </tt>
<a name="L559"></a><tt class="py-lineno"> 559</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">id_ref_0</tt> <tt class="py-op">=</tt> <tt class="py-name">id_ref_0</tt> </tt>
<a name="L560"></a><tt class="py-lineno"> 560</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">id_ref_1</tt> <tt class="py-op">=</tt> <tt class="py-name">id_ref_1</tt> </tt>
<a name="L561"></a><tt class="py-lineno"> 561</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-213" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-213', 'confidence', 'link-94');">confidence</a></tt> <tt class="py-op">=</tt> <tt id="link-214" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-214', 'confidence', 'link-94');">confidence</a></tt> </tt>
</div></div><a name="L562"></a><tt class="py-lineno"> 562</tt>  <tt class="py-line"> </tt>
<a name="Confidence"></a><div id="Confidence-def"><a name="L563"></a><tt class="py-lineno"> 563</tt>  <tt class="py-line"> </tt>
<a name="L564"></a><tt class="py-lineno"> 564</tt> <a class="py-toggle" href="#" id="Confidence-toggle" onclick="return toggle('Confidence');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html">Confidence</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Confidence-expanded"><a name="L565"></a><tt class="py-lineno"> 565</tt>  <tt class="py-line">    <tt class="py-docstring">"""A general purpose confidence element.</tt> </tt>
<a name="L566"></a><tt class="py-lineno"> 566</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L567"></a><tt class="py-lineno"> 567</tt>  <tt class="py-line"><tt class="py-docstring">    For example, this can be used to express the bootstrap support value of a</tt> </tt>
<a name="L568"></a><tt class="py-lineno"> 568</tt>  <tt class="py-line"><tt class="py-docstring">    clade (in which case the `type` attribute is 'bootstrap').</tt> </tt>
<a name="L569"></a><tt class="py-lineno"> 569</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L570"></a><tt class="py-lineno"> 570</tt>  <tt class="py-line"><tt class="py-docstring">    :Parameters:</tt> </tt>
<a name="L571"></a><tt class="py-lineno"> 571</tt>  <tt class="py-line"><tt class="py-docstring">        value : float</tt> </tt>
<a name="L572"></a><tt class="py-lineno"> 572</tt>  <tt class="py-line"><tt class="py-docstring">            confidence value</tt> </tt>
<a name="L573"></a><tt class="py-lineno"> 573</tt>  <tt class="py-line"><tt class="py-docstring">        type : string</tt> </tt>
<a name="L574"></a><tt class="py-lineno"> 574</tt>  <tt class="py-line"><tt class="py-docstring">            label for the type of confidence, e.g. 'bootstrap'</tt> </tt>
<a name="L575"></a><tt class="py-lineno"> 575</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L576"></a><tt class="py-lineno"> 576</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L577"></a><tt class="py-lineno"> 577</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__init__"></a><div id="Confidence.__init__-def"><a name="L578"></a><tt class="py-lineno"> 578</tt> <a class="py-toggle" href="#" id="Confidence.__init__-toggle" onclick="return toggle('Confidence.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">,</tt> <tt class="py-param">type</tt><tt class="py-op">=</tt><tt class="py-string">'unknown'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__init__-expanded"><a name="L579"></a><tt class="py-lineno"> 579</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize values for the Confidence object."""</tt> </tt>
<a name="L580"></a><tt class="py-lineno"> 580</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-215" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-215', 'value', 'link-37');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-216" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-216', 'value', 'link-37');">value</a></tt> </tt>
<a name="L581"></a><tt class="py-lineno"> 581</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-217" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-217', 'type', 'link-29');">type</a></tt> <tt class="py-op">=</tt> <tt id="link-218" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-218', 'type', 'link-29');">type</a></tt> </tt>
</div><a name="L582"></a><tt class="py-lineno"> 582</tt>  <tt class="py-line"> </tt>
<a name="L583"></a><tt class="py-lineno"> 583</tt>  <tt class="py-line">    <tt class="py-comment"># Comparison operators</tt> </tt>
<a name="L584"></a><tt class="py-lineno"> 584</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__hash__"></a><div id="Confidence.__hash__-def"><a name="L585"></a><tt class="py-lineno"> 585</tt> <a class="py-toggle" href="#" id="Confidence.__hash__-toggle" onclick="return toggle('Confidence.__hash__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__hash__">__hash__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__hash__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__hash__-expanded"><a name="L586"></a><tt class="py-lineno"> 586</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the hash value of the object.</tt> </tt>
<a name="L587"></a><tt class="py-lineno"> 587</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L588"></a><tt class="py-lineno"> 588</tt>  <tt class="py-line"><tt class="py-docstring">        Hash values are integers. They are used to quickly compare dictionary</tt> </tt>
<a name="L589"></a><tt class="py-lineno"> 589</tt>  <tt class="py-line"><tt class="py-docstring">        keys during a dictionary lookup. Numeric values that compare equal have</tt> </tt>
<a name="L590"></a><tt class="py-lineno"> 590</tt>  <tt class="py-line"><tt class="py-docstring">        the same hash value (even if they are of different types, as is the</tt> </tt>
<a name="L591"></a><tt class="py-lineno"> 591</tt>  <tt class="py-line"><tt class="py-docstring">        case for 1 and 1.0).</tt> </tt>
<a name="L592"></a><tt class="py-lineno"> 592</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L593"></a><tt class="py-lineno"> 593</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-219" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-219', 'id', 'link-48');">id</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
</div><a name="L594"></a><tt class="py-lineno"> 594</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__eq__"></a><div id="Confidence.__eq__-def"><a name="L595"></a><tt class="py-lineno"> 595</tt> <a class="py-toggle" href="#" id="Confidence.__eq__-toggle" onclick="return toggle('Confidence.__eq__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__eq__">__eq__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__eq__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__eq__-expanded"><a name="L596"></a><tt class="py-lineno"> 596</tt>  <tt class="py-line">        <tt class="py-docstring">"""Check for equality between Confidence objects."""</tt> </tt>
<a name="L597"></a><tt class="py-lineno"> 597</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-220" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-220', 'other', 'link-24');">other</a></tt><tt class="py-op">,</tt> <tt id="link-221" class="py-name"><a title="Bio.Phylo.PhyloXML.Confidence" class="py-name" href="#" onclick="return doclink('link-221', 'Confidence', 'link-87');">Confidence</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L598"></a><tt class="py-lineno"> 598</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-222" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-222', 'value', 'link-37');">value</a></tt> <tt class="py-op">==</tt> <tt id="link-223" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-223', 'other', 'link-24');">other</a></tt><tt class="py-op">.</tt><tt id="link-224" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-224', 'value', 'link-37');">value</a></tt> </tt>
<a name="L599"></a><tt class="py-lineno"> 599</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-225" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-225', 'value', 'link-37');">value</a></tt> <tt class="py-op">==</tt> <tt id="link-226" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-226', 'other', 'link-24');">other</a></tt> </tt>
</div><a name="L600"></a><tt class="py-lineno"> 600</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__ne__"></a><div id="Confidence.__ne__-def"><a name="L601"></a><tt class="py-lineno"> 601</tt> <a class="py-toggle" href="#" id="Confidence.__ne__-toggle" onclick="return toggle('Confidence.__ne__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__ne__">__ne__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__ne__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__ne__-expanded"><a name="L602"></a><tt class="py-lineno"> 602</tt>  <tt class="py-line">        <tt class="py-docstring">"""Check for inequality between two Confidence objects."""</tt> </tt>
<a name="L603"></a><tt class="py-lineno"> 603</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-227" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-227', 'other', 'link-24');">other</a></tt><tt class="py-op">,</tt> <tt id="link-228" class="py-name"><a title="Bio.Phylo.PhyloXML.Confidence" class="py-name" href="#" onclick="return doclink('link-228', 'Confidence', 'link-87');">Confidence</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L604"></a><tt class="py-lineno"> 604</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-229" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-229', 'value', 'link-37');">value</a></tt> <tt class="py-op">!=</tt> <tt id="link-230" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-230', 'other', 'link-24');">other</a></tt><tt class="py-op">.</tt><tt id="link-231" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-231', 'value', 'link-37');">value</a></tt> </tt>
<a name="L605"></a><tt class="py-lineno"> 605</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-232" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-232', 'value', 'link-37');">value</a></tt> <tt class="py-op">!=</tt> <tt id="link-233" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-233', 'other', 'link-24');">other</a></tt> </tt>
</div><a name="L606"></a><tt class="py-lineno"> 606</tt>  <tt class="py-line"> </tt>
<a name="L607"></a><tt class="py-lineno"> 607</tt>  <tt class="py-line">    <tt class="py-comment"># Ordering -- see functools.total_ordering in Py2.7</tt> </tt>
<a name="L608"></a><tt class="py-lineno"> 608</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__lt__"></a><div id="Confidence.__lt__-def"><a name="L609"></a><tt class="py-lineno"> 609</tt> <a class="py-toggle" href="#" id="Confidence.__lt__-toggle" onclick="return toggle('Confidence.__lt__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__lt__">__lt__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__lt__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__lt__-expanded"><a name="L610"></a><tt class="py-lineno"> 610</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-234" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-234', 'other', 'link-24');">other</a></tt><tt class="py-op">,</tt> <tt id="link-235" class="py-name"><a title="Bio.Phylo.PhyloXML.Confidence" class="py-name" href="#" onclick="return doclink('link-235', 'Confidence', 'link-87');">Confidence</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L611"></a><tt class="py-lineno"> 611</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-236" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-236', 'value', 'link-37');">value</a></tt> <tt class="py-op">&lt;</tt> <tt id="link-237" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-237', 'other', 'link-24');">other</a></tt><tt class="py-op">.</tt><tt id="link-238" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-238', 'value', 'link-37');">value</a></tt> </tt>
<a name="L612"></a><tt class="py-lineno"> 612</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-239" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-239', 'value', 'link-37');">value</a></tt> <tt class="py-op">&lt;</tt> <tt id="link-240" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-240', 'other', 'link-24');">other</a></tt> </tt>
</div><a name="L613"></a><tt class="py-lineno"> 613</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__le__"></a><div id="Confidence.__le__-def"><a name="L614"></a><tt class="py-lineno"> 614</tt> <a class="py-toggle" href="#" id="Confidence.__le__-toggle" onclick="return toggle('Confidence.__le__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__le__">__le__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__le__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__le__-expanded"><a name="L615"></a><tt class="py-lineno"> 615</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt> <tt class="py-op">&lt;</tt> <tt id="link-241" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-241', 'other', 'link-24');">other</a></tt> <tt class="py-keyword">or</tt> <tt class="py-name">self</tt> <tt class="py-op">==</tt> <tt id="link-242" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-242', 'other', 'link-24');">other</a></tt> </tt>
</div><a name="L616"></a><tt class="py-lineno"> 616</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__gt__"></a><div id="Confidence.__gt__-def"><a name="L617"></a><tt class="py-lineno"> 617</tt> <a class="py-toggle" href="#" id="Confidence.__gt__-toggle" onclick="return toggle('Confidence.__gt__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__gt__">__gt__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__gt__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__gt__-expanded"><a name="L618"></a><tt class="py-lineno"> 618</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-keyword">not</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt> <tt class="py-op">&lt;=</tt> <tt id="link-243" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-243', 'other', 'link-24');">other</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L619"></a><tt class="py-lineno"> 619</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__ge__"></a><div id="Confidence.__ge__-def"><a name="L620"></a><tt class="py-lineno"> 620</tt> <a class="py-toggle" href="#" id="Confidence.__ge__-toggle" onclick="return toggle('Confidence.__ge__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__ge__">__ge__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__ge__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__ge__-expanded"><a name="L621"></a><tt class="py-lineno"> 621</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-keyword">not</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-244" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-244', 'value', 'link-37');">value</a></tt> <tt class="py-op">&lt;</tt> <tt id="link-245" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-245', 'other', 'link-24');">other</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L622"></a><tt class="py-lineno"> 622</tt>  <tt class="py-line"> </tt>
<a name="L623"></a><tt class="py-lineno"> 623</tt>  <tt class="py-line">    <tt class="py-comment"># Arithmetic operators, including reverse</tt> </tt>
<a name="L624"></a><tt class="py-lineno"> 624</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__add__"></a><div id="Confidence.__add__-def"><a name="L625"></a><tt class="py-lineno"> 625</tt> <a class="py-toggle" href="#" id="Confidence.__add__-toggle" onclick="return toggle('Confidence.__add__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__add__">__add__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__add__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__add__-expanded"><a name="L626"></a><tt class="py-lineno"> 626</tt>  <tt class="py-line">        <tt class="py-docstring">"""Conduct additions between value of two Confidence objects."""</tt> </tt>
<a name="L627"></a><tt class="py-lineno"> 627</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-246" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-246', 'value', 'link-37');">value</a></tt> <tt class="py-op">+</tt> <tt id="link-247" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-247', 'other', 'link-24');">other</a></tt> </tt>
</div><a name="L628"></a><tt class="py-lineno"> 628</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__radd__"></a><div id="Confidence.__radd__-def"><a name="L629"></a><tt class="py-lineno"> 629</tt> <a class="py-toggle" href="#" id="Confidence.__radd__-toggle" onclick="return toggle('Confidence.__radd__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__radd__">__radd__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__radd__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__radd__-expanded"><a name="L630"></a><tt class="py-lineno"> 630</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-248" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-248', 'other', 'link-24');">other</a></tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-249" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-249', 'value', 'link-37');">value</a></tt> </tt>
</div><a name="L631"></a><tt class="py-lineno"> 631</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__sub__"></a><div id="Confidence.__sub__-def"><a name="L632"></a><tt class="py-lineno"> 632</tt> <a class="py-toggle" href="#" id="Confidence.__sub__-toggle" onclick="return toggle('Confidence.__sub__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__sub__">__sub__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__sub__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__sub__-expanded"><a name="L633"></a><tt class="py-lineno"> 633</tt>  <tt class="py-line">        <tt class="py-docstring">"""Conduct subtraction between value of two Confidence objects."""</tt> </tt>
<a name="L634"></a><tt class="py-lineno"> 634</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-250" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-250', 'value', 'link-37');">value</a></tt> <tt class="py-op">-</tt> <tt id="link-251" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-251', 'other', 'link-24');">other</a></tt> </tt>
</div><a name="L635"></a><tt class="py-lineno"> 635</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__rsub__"></a><div id="Confidence.__rsub__-def"><a name="L636"></a><tt class="py-lineno"> 636</tt> <a class="py-toggle" href="#" id="Confidence.__rsub__-toggle" onclick="return toggle('Confidence.__rsub__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__rsub__">__rsub__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__rsub__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__rsub__-expanded"><a name="L637"></a><tt class="py-lineno"> 637</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-252" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-252', 'other', 'link-24');">other</a></tt> <tt class="py-op">-</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-253" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-253', 'value', 'link-37');">value</a></tt> </tt>
</div><a name="L638"></a><tt class="py-lineno"> 638</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__mul__"></a><div id="Confidence.__mul__-def"><a name="L639"></a><tt class="py-lineno"> 639</tt> <a class="py-toggle" href="#" id="Confidence.__mul__-toggle" onclick="return toggle('Confidence.__mul__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__mul__">__mul__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__mul__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__mul__-expanded"><a name="L640"></a><tt class="py-lineno"> 640</tt>  <tt class="py-line">        <tt class="py-docstring">"""Conduct multiplication between value of two Confidence objects."""</tt> </tt>
<a name="L641"></a><tt class="py-lineno"> 641</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-254" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-254', 'value', 'link-37');">value</a></tt> <tt class="py-op">*</tt> <tt id="link-255" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-255', 'other', 'link-24');">other</a></tt> </tt>
</div><a name="L642"></a><tt class="py-lineno"> 642</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__rmul__"></a><div id="Confidence.__rmul__-def"><a name="L643"></a><tt class="py-lineno"> 643</tt> <a class="py-toggle" href="#" id="Confidence.__rmul__-toggle" onclick="return toggle('Confidence.__rmul__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__rmul__">__rmul__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__rmul__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__rmul__-expanded"><a name="L644"></a><tt class="py-lineno"> 644</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-256" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-256', 'other', 'link-24');">other</a></tt> <tt class="py-op">*</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-257" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-257', 'value', 'link-37');">value</a></tt> </tt>
</div><a name="L645"></a><tt class="py-lineno"> 645</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__div__"></a><div id="Confidence.__div__-def"><a name="L646"></a><tt class="py-lineno"> 646</tt> <a class="py-toggle" href="#" id="Confidence.__div__-toggle" onclick="return toggle('Confidence.__div__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__div__">__div__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__div__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__div__-expanded"><a name="L647"></a><tt class="py-lineno"> 647</tt>  <tt class="py-line">        <tt class="py-docstring">"""Conduct division between value of two Confidence objects."""</tt> </tt>
<a name="L648"></a><tt class="py-lineno"> 648</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-258" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-258', 'value', 'link-37');">value</a></tt><tt class="py-op">.</tt><tt id="link-259" class="py-name" targets="Method Bio.PDB.vectors.Vector.__div__()=Bio.PDB.vectors.Vector-class.html#__div__,Method Bio.Phylo.PhyloXML.Confidence.__div__()=Bio.Phylo.PhyloXML.Confidence-class.html#__div__,Method Bio.Restriction.Restriction.RestrictionBatch.__div__()=Bio.Restriction.Restriction.RestrictionBatch-class.html#__div__,Method Bio.Restriction.Restriction.RestrictionType.__div__()=Bio.Restriction.Restriction.RestrictionType-class.html#__div__"><a title="Bio.PDB.vectors.Vector.__div__
Bio.Phylo.PhyloXML.Confidence.__div__
Bio.Restriction.Restriction.RestrictionBatch.__div__
Bio.Restriction.Restriction.RestrictionType.__div__" class="py-name" href="#" onclick="return doclink('link-259', '__div__', 'link-259');">__div__</a></tt><tt class="py-op">(</tt><tt id="link-260" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-260', 'other', 'link-24');">other</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L649"></a><tt class="py-lineno"> 649</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__rdiv__"></a><div id="Confidence.__rdiv__-def"><a name="L650"></a><tt class="py-lineno"> 650</tt> <a class="py-toggle" href="#" id="Confidence.__rdiv__-toggle" onclick="return toggle('Confidence.__rdiv__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__rdiv__">__rdiv__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__rdiv__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__rdiv__-expanded"><a name="L651"></a><tt class="py-lineno"> 651</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-261" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-261', 'other', 'link-24');">other</a></tt><tt class="py-op">.</tt><tt id="link-262" class="py-name"><a title="Bio.PDB.vectors.Vector.__div__
Bio.Phylo.PhyloXML.Confidence.__div__
Bio.Restriction.Restriction.RestrictionBatch.__div__
Bio.Restriction.Restriction.RestrictionType.__div__" class="py-name" href="#" onclick="return doclink('link-262', '__div__', 'link-259');">__div__</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-263" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-263', 'value', 'link-37');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L652"></a><tt class="py-lineno"> 652</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__truediv__"></a><div id="Confidence.__truediv__-def"><a name="L653"></a><tt class="py-lineno"> 653</tt> <a class="py-toggle" href="#" id="Confidence.__truediv__-toggle" onclick="return toggle('Confidence.__truediv__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__truediv__">__truediv__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__truediv__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__truediv__-expanded"><a name="L654"></a><tt class="py-lineno"> 654</tt>  <tt class="py-line">        <tt class="py-docstring">"""Rational-style division in Py3.0+.</tt> </tt>
<a name="L655"></a><tt class="py-lineno"> 655</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L656"></a><tt class="py-lineno"> 656</tt>  <tt class="py-line"><tt class="py-docstring">        Also active in Py2.5+ with __future__.division import.</tt> </tt>
<a name="L657"></a><tt class="py-lineno"> 657</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L658"></a><tt class="py-lineno"> 658</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-264" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-264', 'value', 'link-37');">value</a></tt> <tt class="py-op">/</tt> <tt id="link-265" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-265', 'other', 'link-24');">other</a></tt> </tt>
</div><a name="L659"></a><tt class="py-lineno"> 659</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__rtruediv__"></a><div id="Confidence.__rtruediv__-def"><a name="L660"></a><tt class="py-lineno"> 660</tt> <a class="py-toggle" href="#" id="Confidence.__rtruediv__-toggle" onclick="return toggle('Confidence.__rtruediv__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__rtruediv__">__rtruediv__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__rtruediv__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__rtruediv__-expanded"><a name="L661"></a><tt class="py-lineno"> 661</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-266" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-266', 'other', 'link-24');">other</a></tt> <tt class="py-op">/</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-267" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-267', 'value', 'link-37');">value</a></tt> </tt>
</div><a name="L662"></a><tt class="py-lineno"> 662</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__floordiv__"></a><div id="Confidence.__floordiv__-def"><a name="L663"></a><tt class="py-lineno"> 663</tt> <a class="py-toggle" href="#" id="Confidence.__floordiv__-toggle" onclick="return toggle('Confidence.__floordiv__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__floordiv__">__floordiv__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__floordiv__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__floordiv__-expanded"><a name="L664"></a><tt class="py-lineno"> 664</tt>  <tt class="py-line">        <tt class="py-docstring">"""C-style and old-style division in Py3.0+.</tt> </tt>
<a name="L665"></a><tt class="py-lineno"> 665</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L666"></a><tt class="py-lineno"> 666</tt>  <tt class="py-line"><tt class="py-docstring">        Also active in Py2.5+ with __future__.division import.</tt> </tt>
<a name="L667"></a><tt class="py-lineno"> 667</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L668"></a><tt class="py-lineno"> 668</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-268" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-268', 'value', 'link-37');">value</a></tt><tt class="py-op">.</tt><tt id="link-269" class="py-name" targets="Method Bio.Phylo.PhyloXML.Confidence.__floordiv__()=Bio.Phylo.PhyloXML.Confidence-class.html#__floordiv__,Method Bio.Restriction.Restriction.RestrictionType.__floordiv__()=Bio.Restriction.Restriction.RestrictionType-class.html#__floordiv__"><a title="Bio.Phylo.PhyloXML.Confidence.__floordiv__
Bio.Restriction.Restriction.RestrictionType.__floordiv__" class="py-name" href="#" onclick="return doclink('link-269', '__floordiv__', 'link-269');">__floordiv__</a></tt><tt class="py-op">(</tt><tt id="link-270" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-270', 'other', 'link-24');">other</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L669"></a><tt class="py-lineno"> 669</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__rfloordiv__"></a><div id="Confidence.__rfloordiv__-def"><a name="L670"></a><tt class="py-lineno"> 670</tt> <a class="py-toggle" href="#" id="Confidence.__rfloordiv__-toggle" onclick="return toggle('Confidence.__rfloordiv__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__rfloordiv__">__rfloordiv__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__rfloordiv__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__rfloordiv__-expanded"><a name="L671"></a><tt class="py-lineno"> 671</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-271" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-271', 'other', 'link-24');">other</a></tt><tt class="py-op">.</tt><tt id="link-272" class="py-name"><a title="Bio.Phylo.PhyloXML.Confidence.__floordiv__
Bio.Restriction.Restriction.RestrictionType.__floordiv__" class="py-name" href="#" onclick="return doclink('link-272', '__floordiv__', 'link-269');">__floordiv__</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-273" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-273', 'value', 'link-37');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L672"></a><tt class="py-lineno"> 672</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__mod__"></a><div id="Confidence.__mod__-def"><a name="L673"></a><tt class="py-lineno"> 673</tt> <a class="py-toggle" href="#" id="Confidence.__mod__-toggle" onclick="return toggle('Confidence.__mod__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__mod__">__mod__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__mod__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__mod__-expanded"><a name="L674"></a><tt class="py-lineno"> 674</tt>  <tt class="py-line">        <tt class="py-docstring">"""Conduct modulus between value of two Confidence objects."""</tt> </tt>
<a name="L675"></a><tt class="py-lineno"> 675</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-274" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-274', 'value', 'link-37');">value</a></tt> <tt class="py-op">%</tt> <tt id="link-275" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-275', 'other', 'link-24');">other</a></tt> </tt>
</div><a name="L676"></a><tt class="py-lineno"> 676</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__rmod__"></a><div id="Confidence.__rmod__-def"><a name="L677"></a><tt class="py-lineno"> 677</tt> <a class="py-toggle" href="#" id="Confidence.__rmod__-toggle" onclick="return toggle('Confidence.__rmod__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__rmod__">__rmod__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__rmod__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__rmod__-expanded"><a name="L678"></a><tt class="py-lineno"> 678</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-276" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-276', 'other', 'link-24');">other</a></tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-277" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-277', 'value', 'link-37');">value</a></tt> </tt>
</div><a name="L679"></a><tt class="py-lineno"> 679</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__divmod__"></a><div id="Confidence.__divmod__-def"><a name="L680"></a><tt class="py-lineno"> 680</tt> <a class="py-toggle" href="#" id="Confidence.__divmod__-toggle" onclick="return toggle('Confidence.__divmod__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__divmod__">__divmod__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__divmod__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__divmod__-expanded"><a name="L681"></a><tt class="py-lineno"> 681</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">divmod</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-278" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-278', 'value', 'link-37');">value</a></tt><tt class="py-op">,</tt> <tt id="link-279" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-279', 'other', 'link-24');">other</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L682"></a><tt class="py-lineno"> 682</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__rdivmod__"></a><div id="Confidence.__rdivmod__-def"><a name="L683"></a><tt class="py-lineno"> 683</tt> <a class="py-toggle" href="#" id="Confidence.__rdivmod__-toggle" onclick="return toggle('Confidence.__rdivmod__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__rdivmod__">__rdivmod__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__rdivmod__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__rdivmod__-expanded"><a name="L684"></a><tt class="py-lineno"> 684</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">divmod</tt><tt class="py-op">(</tt><tt id="link-280" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-280', 'other', 'link-24');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-281" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-281', 'value', 'link-37');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L685"></a><tt class="py-lineno"> 685</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__pow__"></a><div id="Confidence.__pow__-def"><a name="L686"></a><tt class="py-lineno"> 686</tt> <a class="py-toggle" href="#" id="Confidence.__pow__-toggle" onclick="return toggle('Confidence.__pow__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__pow__">__pow__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">,</tt> <tt class="py-param">modulo</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__pow__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__pow__-expanded"><a name="L687"></a><tt class="py-lineno"> 687</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">modulo</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L688"></a><tt class="py-lineno"> 688</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">pow</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-282" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-282', 'value', 'link-37');">value</a></tt><tt class="py-op">,</tt> <tt id="link-283" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-283', 'other', 'link-24');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">modulo</tt><tt class="py-op">)</tt> </tt>
<a name="L689"></a><tt class="py-lineno"> 689</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">pow</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-284" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-284', 'value', 'link-37');">value</a></tt><tt class="py-op">,</tt> <tt id="link-285" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-285', 'other', 'link-24');">other</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L690"></a><tt class="py-lineno"> 690</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__rpow__"></a><div id="Confidence.__rpow__-def"><a name="L691"></a><tt class="py-lineno"> 691</tt> <a class="py-toggle" href="#" id="Confidence.__rpow__-toggle" onclick="return toggle('Confidence.__rpow__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__rpow__">__rpow__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__rpow__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__rpow__-expanded"><a name="L692"></a><tt class="py-lineno"> 692</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">pow</tt><tt class="py-op">(</tt><tt id="link-286" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-286', 'other', 'link-24');">other</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-287" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-287', 'value', 'link-37');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L693"></a><tt class="py-lineno"> 693</tt>  <tt class="py-line"> </tt>
<a name="L694"></a><tt class="py-lineno"> 694</tt>  <tt class="py-line">    <tt class="py-comment"># Unary arithmetic operations: -, +, abs()</tt> </tt>
<a name="L695"></a><tt class="py-lineno"> 695</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__neg__"></a><div id="Confidence.__neg__-def"><a name="L696"></a><tt class="py-lineno"> 696</tt> <a class="py-toggle" href="#" id="Confidence.__neg__-toggle" onclick="return toggle('Confidence.__neg__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__neg__">__neg__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__neg__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__neg__-expanded"><a name="L697"></a><tt class="py-lineno"> 697</tt>  <tt class="py-line">        <tt class="py-docstring">"""Conduct negation of a Confidence object."""</tt> </tt>
<a name="L698"></a><tt class="py-lineno"> 698</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">-</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-288" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-288', 'value', 'link-37');">value</a></tt> </tt>
</div><a name="L699"></a><tt class="py-lineno"> 699</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__pos__"></a><div id="Confidence.__pos__-def"><a name="L700"></a><tt class="py-lineno"> 700</tt> <a class="py-toggle" href="#" id="Confidence.__pos__-toggle" onclick="return toggle('Confidence.__pos__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__pos__">__pos__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__pos__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__pos__-expanded"><a name="L701"></a><tt class="py-lineno"> 701</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-289" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-289', 'value', 'link-37');">value</a></tt> </tt>
</div><a name="L702"></a><tt class="py-lineno"> 702</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__abs__"></a><div id="Confidence.__abs__-def"><a name="L703"></a><tt class="py-lineno"> 703</tt> <a class="py-toggle" href="#" id="Confidence.__abs__-toggle" onclick="return toggle('Confidence.__abs__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__abs__">__abs__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__abs__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__abs__-expanded"><a name="L704"></a><tt class="py-lineno"> 704</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return absolute value of Confidence object."""</tt> </tt>
<a name="L705"></a><tt class="py-lineno"> 705</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-290" class="py-name" targets="Method Bio.FSSP.FSSPAlignDict.abs()=Bio.FSSP.FSSPAlignDict-class.html#abs"><a title="Bio.FSSP.FSSPAlignDict.abs" class="py-name" href="#" onclick="return doclink('link-290', 'abs', 'link-290');">abs</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-291" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-291', 'value', 'link-37');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L706"></a><tt class="py-lineno"> 706</tt>  <tt class="py-line"> </tt>
<a name="L707"></a><tt class="py-lineno"> 707</tt>  <tt class="py-line">    <tt class="py-comment"># Explicit coercion to numeric types: float, int</tt> </tt>
<a name="L708"></a><tt class="py-lineno"> 708</tt>  <tt class="py-line">    <tt class="py-comment"># (and under Python 2 only long)</tt> </tt>
<a name="L709"></a><tt class="py-lineno"> 709</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__float__"></a><div id="Confidence.__float__-def"><a name="L710"></a><tt class="py-lineno"> 710</tt> <a class="py-toggle" href="#" id="Confidence.__float__-toggle" onclick="return toggle('Confidence.__float__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__float__">__float__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__float__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__float__-expanded"><a name="L711"></a><tt class="py-lineno"> 711</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return float value of Confidence object."""</tt> </tt>
<a name="L712"></a><tt class="py-lineno"> 712</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-292" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-292', 'value', 'link-37');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L713"></a><tt class="py-lineno"> 713</tt>  <tt class="py-line"> </tt>
<a name="Confidence.__int__"></a><div id="Confidence.__int__-def"><a name="L714"></a><tt class="py-lineno"> 714</tt> <a class="py-toggle" href="#" id="Confidence.__int__-toggle" onclick="return toggle('Confidence.__int__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__int__">__int__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__int__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Confidence.__int__-expanded"><a name="L715"></a><tt class="py-lineno"> 715</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return integer value of Confidence object."""</tt> </tt>
<a name="L716"></a><tt class="py-lineno"> 716</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-293" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-293', 'value', 'link-37');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L717"></a><tt class="py-lineno"> 717</tt>  <tt class="py-line"> </tt>
<a name="L718"></a><tt class="py-lineno"> 718</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">version_info</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">3</tt><tt class="py-op">:</tt> </tt>
<a name="Confidence.__long__"></a><div id="Confidence.__long__-def"><a name="L719"></a><tt class="py-lineno"> 719</tt> <a class="py-toggle" href="#" id="Confidence.__long__-toggle" onclick="return toggle('Confidence.__long__');">-</a><tt class="py-line">        <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Confidence-class.html#__long__">__long__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Confidence.__long__-collapsed" style="display:none;" pad="++++" indent="++++++++++++++++"></div><div id="Confidence.__long__-expanded"><a name="L720"></a><tt class="py-lineno"> 720</tt>  <tt class="py-line">            <tt class="py-docstring">"""Return long value of Confidence object."""</tt> </tt>
<a name="L721"></a><tt class="py-lineno"> 721</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-294" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.long()=Bio.Phylo.PhyloXMLIO.Writer-class.html#long"><a title="Bio.Phylo.PhyloXMLIO.Writer.long" class="py-name" href="#" onclick="return doclink('link-294', 'long', 'link-294');">long</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-295" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-295', 'value', 'link-37');">value</a></tt><tt class="py-op">)</tt>  <tt class="py-comment"># noqa : F821</tt> </tt>
</div></div><a name="L722"></a><tt class="py-lineno"> 722</tt>  <tt class="py-line"> </tt>
<a name="Date"></a><div id="Date-def"><a name="L723"></a><tt class="py-lineno"> 723</tt>  <tt class="py-line"> </tt>
<a name="L724"></a><tt class="py-lineno"> 724</tt> <a class="py-toggle" href="#" id="Date-toggle" onclick="return toggle('Date');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Date-class.html">Date</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Date-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Date-expanded"><a name="L725"></a><tt class="py-lineno"> 725</tt>  <tt class="py-line">    <tt class="py-docstring">"""A date associated with a clade/node.</tt> </tt>
<a name="L726"></a><tt class="py-lineno"> 726</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L727"></a><tt class="py-lineno"> 727</tt>  <tt class="py-line"><tt class="py-docstring">    Its value can be numerical by using the 'value' element and/or free text</tt> </tt>
<a name="L728"></a><tt class="py-lineno"> 728</tt>  <tt class="py-line"><tt class="py-docstring">    with the 'desc' element' (e.g. 'Silurian'). If a numerical value is used, it</tt> </tt>
<a name="L729"></a><tt class="py-lineno"> 729</tt>  <tt class="py-line"><tt class="py-docstring">    is recommended to employ the 'unit' attribute.</tt> </tt>
<a name="L730"></a><tt class="py-lineno"> 730</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L731"></a><tt class="py-lineno"> 731</tt>  <tt class="py-line"><tt class="py-docstring">    :Parameters:</tt> </tt>
<a name="L732"></a><tt class="py-lineno"> 732</tt>  <tt class="py-line"><tt class="py-docstring">        unit : string</tt> </tt>
<a name="L733"></a><tt class="py-lineno"> 733</tt>  <tt class="py-line"><tt class="py-docstring">            type of numerical value (e.g. 'mya' for 'million years ago')</tt> </tt>
<a name="L734"></a><tt class="py-lineno"> 734</tt>  <tt class="py-line"><tt class="py-docstring">        value : float</tt> </tt>
<a name="L735"></a><tt class="py-lineno"> 735</tt>  <tt class="py-line"><tt class="py-docstring">            the date value</tt> </tt>
<a name="L736"></a><tt class="py-lineno"> 736</tt>  <tt class="py-line"><tt class="py-docstring">        desc : string</tt> </tt>
<a name="L737"></a><tt class="py-lineno"> 737</tt>  <tt class="py-line"><tt class="py-docstring">            plain-text description of the date</tt> </tt>
<a name="L738"></a><tt class="py-lineno"> 738</tt>  <tt class="py-line"><tt class="py-docstring">        minimum : float</tt> </tt>
<a name="L739"></a><tt class="py-lineno"> 739</tt>  <tt class="py-line"><tt class="py-docstring">            lower bound on the date value</tt> </tt>
<a name="L740"></a><tt class="py-lineno"> 740</tt>  <tt class="py-line"><tt class="py-docstring">        maximum : float</tt> </tt>
<a name="L741"></a><tt class="py-lineno"> 741</tt>  <tt class="py-line"><tt class="py-docstring">            upper bound on the date value</tt> </tt>
<a name="L742"></a><tt class="py-lineno"> 742</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L743"></a><tt class="py-lineno"> 743</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L744"></a><tt class="py-lineno"> 744</tt>  <tt class="py-line"> </tt>
<a name="Date.__init__"></a><div id="Date.__init__-def"><a name="L745"></a><tt class="py-lineno"> 745</tt> <a class="py-toggle" href="#" id="Date.__init__-toggle" onclick="return toggle('Date.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Date-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">unit</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">desc</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L746"></a><tt class="py-lineno"> 746</tt>  <tt class="py-line">                 <tt class="py-param">minimum</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">maximum</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Date.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Date.__init__-expanded"><a name="L747"></a><tt class="py-lineno"> 747</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize values of the Date object."""</tt> </tt>
<a name="L748"></a><tt class="py-lineno"> 748</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-296" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-296', 'value', 'link-37');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-297" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-297', 'value', 'link-37');">value</a></tt> </tt>
<a name="L749"></a><tt class="py-lineno"> 749</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">unit</tt> <tt class="py-op">=</tt> <tt class="py-name">unit</tt> </tt>
<a name="L750"></a><tt class="py-lineno"> 750</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-298" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.desc" class="py-name" href="#" onclick="return doclink('link-298', 'desc', 'link-201');">desc</a></tt> <tt class="py-op">=</tt> <tt id="link-299" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.desc" class="py-name" href="#" onclick="return doclink('link-299', 'desc', 'link-201');">desc</a></tt> </tt>
<a name="L751"></a><tt class="py-lineno"> 751</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-300" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.minimum()=Bio.Phylo.PhyloXMLIO.Writer-class.html#minimum"><a title="Bio.Phylo.PhyloXMLIO.Writer.minimum" class="py-name" href="#" onclick="return doclink('link-300', 'minimum', 'link-300');">minimum</a></tt> <tt class="py-op">=</tt> <tt id="link-301" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.minimum" class="py-name" href="#" onclick="return doclink('link-301', 'minimum', 'link-300');">minimum</a></tt> </tt>
<a name="L752"></a><tt class="py-lineno"> 752</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-302" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.maximum()=Bio.Phylo.PhyloXMLIO.Writer-class.html#maximum"><a title="Bio.Phylo.PhyloXMLIO.Writer.maximum" class="py-name" href="#" onclick="return doclink('link-302', 'maximum', 'link-302');">maximum</a></tt> <tt class="py-op">=</tt> <tt id="link-303" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.maximum" class="py-name" href="#" onclick="return doclink('link-303', 'maximum', 'link-302');">maximum</a></tt> </tt>
</div><a name="L753"></a><tt class="py-lineno"> 753</tt>  <tt class="py-line"> </tt>
<a name="Date.__str__"></a><div id="Date.__str__-def"><a name="L754"></a><tt class="py-lineno"> 754</tt> <a class="py-toggle" href="#" id="Date.__str__-toggle" onclick="return toggle('Date.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Date-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Date.__str__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Date.__str__-expanded"><a name="L755"></a><tt class="py-lineno"> 755</tt>  <tt class="py-line">        <tt class="py-docstring">"""Show the class name and the human-readable date."""</tt> </tt>
<a name="L756"></a><tt class="py-lineno"> 756</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">unit</tt> <tt class="py-keyword">and</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-304" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-304', 'value', 'link-37');">value</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L757"></a><tt class="py-lineno"> 757</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-string">'%s %s'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-305" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-305', 'value', 'link-37');">value</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">unit</tt><tt class="py-op">)</tt> </tt>
<a name="L758"></a><tt class="py-lineno"> 758</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-306" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.desc" class="py-name" href="#" onclick="return doclink('link-306', 'desc', 'link-201');">desc</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L759"></a><tt class="py-lineno"> 759</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-307" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.desc" class="py-name" href="#" onclick="return doclink('link-307', 'desc', 'link-201');">desc</a></tt> </tt>
<a name="L760"></a><tt class="py-lineno"> 760</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt> </tt>
</div></div><a name="L761"></a><tt class="py-lineno"> 761</tt>  <tt class="py-line"> </tt>
<a name="Distribution"></a><div id="Distribution-def"><a name="L762"></a><tt class="py-lineno"> 762</tt>  <tt class="py-line"> </tt>
<a name="L763"></a><tt class="py-lineno"> 763</tt> <a class="py-toggle" href="#" id="Distribution-toggle" onclick="return toggle('Distribution');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Distribution-class.html">Distribution</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Distribution-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Distribution-expanded"><a name="L764"></a><tt class="py-lineno"> 764</tt>  <tt class="py-line">    <tt class="py-docstring">"""Geographic distribution of the items of a clade (species, sequences).</tt> </tt>
<a name="L765"></a><tt class="py-lineno"> 765</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L766"></a><tt class="py-lineno"> 766</tt>  <tt class="py-line"><tt class="py-docstring">    Intended for phylogeographic applications.</tt> </tt>
<a name="L767"></a><tt class="py-lineno"> 767</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L768"></a><tt class="py-lineno"> 768</tt>  <tt class="py-line"><tt class="py-docstring">    :Parameters:</tt> </tt>
<a name="L769"></a><tt class="py-lineno"> 769</tt>  <tt class="py-line"><tt class="py-docstring">        desc : string</tt> </tt>
<a name="L770"></a><tt class="py-lineno"> 770</tt>  <tt class="py-line"><tt class="py-docstring">            free-text description of the location</tt> </tt>
<a name="L771"></a><tt class="py-lineno"> 771</tt>  <tt class="py-line"><tt class="py-docstring">        points : list of `Point` objects</tt> </tt>
<a name="L772"></a><tt class="py-lineno"> 772</tt>  <tt class="py-line"><tt class="py-docstring">            coordinates (similar to the 'Point' element in Google's KML format)</tt> </tt>
<a name="L773"></a><tt class="py-lineno"> 773</tt>  <tt class="py-line"><tt class="py-docstring">        polygons : list of `Polygon` objects</tt> </tt>
<a name="L774"></a><tt class="py-lineno"> 774</tt>  <tt class="py-line"><tt class="py-docstring">            coordinate sets defining geographic regions</tt> </tt>
<a name="L775"></a><tt class="py-lineno"> 775</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L776"></a><tt class="py-lineno"> 776</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L777"></a><tt class="py-lineno"> 777</tt>  <tt class="py-line"> </tt>
<a name="Distribution.__init__"></a><div id="Distribution.__init__-def"><a name="L778"></a><tt class="py-lineno"> 778</tt> <a class="py-toggle" href="#" id="Distribution.__init__-toggle" onclick="return toggle('Distribution.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Distribution-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">desc</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">points</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">polygons</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Distribution.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Distribution.__init__-expanded"><a name="L779"></a><tt class="py-lineno"> 779</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize values of Distribution object."""</tt> </tt>
<a name="L780"></a><tt class="py-lineno"> 780</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-308" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.desc" class="py-name" href="#" onclick="return doclink('link-308', 'desc', 'link-201');">desc</a></tt> <tt class="py-op">=</tt> <tt id="link-309" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.desc" class="py-name" href="#" onclick="return doclink('link-309', 'desc', 'link-201');">desc</a></tt> </tt>
<a name="L781"></a><tt class="py-lineno"> 781</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">points</tt> <tt class="py-op">=</tt> <tt class="py-name">points</tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L782"></a><tt class="py-lineno"> 782</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">polygons</tt> <tt class="py-op">=</tt> <tt class="py-name">polygons</tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div></div><a name="L783"></a><tt class="py-lineno"> 783</tt>  <tt class="py-line"> </tt>
<a name="DomainArchitecture"></a><div id="DomainArchitecture-def"><a name="L784"></a><tt class="py-lineno"> 784</tt>  <tt class="py-line"> </tt>
<a name="L785"></a><tt class="py-lineno"> 785</tt> <a class="py-toggle" href="#" id="DomainArchitecture-toggle" onclick="return toggle('DomainArchitecture');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.DomainArchitecture-class.html">DomainArchitecture</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DomainArchitecture-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="DomainArchitecture-expanded"><a name="L786"></a><tt class="py-lineno"> 786</tt>  <tt class="py-line">    <tt class="py-docstring">"""Domain architecture of a protein.</tt> </tt>
<a name="L787"></a><tt class="py-lineno"> 787</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L788"></a><tt class="py-lineno"> 788</tt>  <tt class="py-line"><tt class="py-docstring">    :Parameters:</tt> </tt>
<a name="L789"></a><tt class="py-lineno"> 789</tt>  <tt class="py-line"><tt class="py-docstring">        length : int</tt> </tt>
<a name="L790"></a><tt class="py-lineno"> 790</tt>  <tt class="py-line"><tt class="py-docstring">            total length of the protein sequence</tt> </tt>
<a name="L791"></a><tt class="py-lineno"> 791</tt>  <tt class="py-line"><tt class="py-docstring">        domains : list ProteinDomain objects</tt> </tt>
<a name="L792"></a><tt class="py-lineno"> 792</tt>  <tt class="py-line"><tt class="py-docstring">            the domains within this protein</tt> </tt>
<a name="L793"></a><tt class="py-lineno"> 793</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L794"></a><tt class="py-lineno"> 794</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L795"></a><tt class="py-lineno"> 795</tt>  <tt class="py-line"> </tt>
<a name="DomainArchitecture.__init__"></a><div id="DomainArchitecture.__init__-def"><a name="L796"></a><tt class="py-lineno"> 796</tt> <a class="py-toggle" href="#" id="DomainArchitecture.__init__-toggle" onclick="return toggle('DomainArchitecture.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.DomainArchitecture-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">length</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">domains</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DomainArchitecture.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DomainArchitecture.__init__-expanded"><a name="L797"></a><tt class="py-lineno"> 797</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize values of the DomainArchitecture object."""</tt> </tt>
<a name="L798"></a><tt class="py-lineno"> 798</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-310" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.length()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#length"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-310', 'length', 'link-310');">length</a></tt> <tt class="py-op">=</tt> <tt id="link-311" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-311', 'length', 'link-310');">length</a></tt> </tt>
<a name="L799"></a><tt class="py-lineno"> 799</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">domains</tt> <tt class="py-op">=</tt> <tt class="py-name">domains</tt> </tt>
</div></div><a name="L800"></a><tt class="py-lineno"> 800</tt>  <tt class="py-line"> </tt>
<a name="Events"></a><div id="Events-def"><a name="L801"></a><tt class="py-lineno"> 801</tt>  <tt class="py-line"> </tt>
<a name="L802"></a><tt class="py-lineno"> 802</tt> <a class="py-toggle" href="#" id="Events-toggle" onclick="return toggle('Events');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Events-class.html">Events</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Events-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Events-expanded"><a name="L803"></a><tt class="py-lineno"> 803</tt>  <tt class="py-line">    <tt class="py-docstring">"""Events at the root node of a clade (e.g. one gene duplication).</tt> </tt>
<a name="L804"></a><tt class="py-lineno"> 804</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L805"></a><tt class="py-lineno"> 805</tt>  <tt class="py-line"><tt class="py-docstring">    All attributes are set to None by default, but this object can also be</tt> </tt>
<a name="L806"></a><tt class="py-lineno"> 806</tt>  <tt class="py-line"><tt class="py-docstring">    treated as a dictionary, in which case None values are treated as missing</tt> </tt>
<a name="L807"></a><tt class="py-lineno"> 807</tt>  <tt class="py-line"><tt class="py-docstring">    keys and deleting a key resets that attribute's value back to None.</tt> </tt>
<a name="L808"></a><tt class="py-lineno"> 808</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L809"></a><tt class="py-lineno"> 809</tt>  <tt class="py-line"> </tt>
<a name="L810"></a><tt class="py-lineno"> 810</tt>  <tt class="py-line">    <tt id="link-312" class="py-name" targets="Variable Bio.Phylo.PhyloXML.Events.ok_type=Bio.Phylo.PhyloXML.Events-class.html#ok_type,Variable Bio.Phylo.PhyloXML.SequenceRelation.ok_type=Bio.Phylo.PhyloXML.SequenceRelation-class.html#ok_type"><a title="Bio.Phylo.PhyloXML.Events.ok_type
Bio.Phylo.PhyloXML.SequenceRelation.ok_type" class="py-name" href="#" onclick="return doclink('link-312', 'ok_type', 'link-312');">ok_type</a></tt> <tt class="py-op">=</tt> <tt id="link-313" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.set()=Bio.Nexus.Nexus.StepMatrix-class.html#set,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.set()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#set,Method Bio.SVDSuperimposer.SVDSuperimposer.set()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#set"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-313', 'set', 'link-313');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">'transfer'</tt><tt class="py-op">,</tt> <tt class="py-string">'fusion'</tt><tt class="py-op">,</tt> <tt class="py-string">'speciation_or_duplication'</tt><tt class="py-op">,</tt> <tt class="py-string">'other'</tt><tt class="py-op">,</tt> </tt>
<a name="L811"></a><tt class="py-lineno"> 811</tt>  <tt class="py-line">                   <tt class="py-string">'mixed'</tt><tt class="py-op">,</tt> <tt class="py-string">'unassigned'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L812"></a><tt class="py-lineno"> 812</tt>  <tt class="py-line"> </tt>
<a name="Events.__init__"></a><div id="Events.__init__-def"><a name="L813"></a><tt class="py-lineno"> 813</tt> <a class="py-toggle" href="#" id="Events.__init__-toggle" onclick="return toggle('Events.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Events-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">type</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">duplications</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">speciations</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L814"></a><tt class="py-lineno"> 814</tt>  <tt class="py-line">                 <tt class="py-param">losses</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">confidence</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Events.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Events.__init__-expanded"><a name="L815"></a><tt class="py-lineno"> 815</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize values of the Events object."""</tt> </tt>
<a name="L816"></a><tt class="py-lineno"> 816</tt>  <tt class="py-line">        <tt id="link-314" class="py-name"><a title="Bio.Phylo.PhyloXML._check_str" class="py-name" href="#" onclick="return doclink('link-314', '_check_str', 'link-191');">_check_str</a></tt><tt class="py-op">(</tt><tt id="link-315" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-315', 'type', 'link-29');">type</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-316" class="py-name"><a title="Bio.Phylo.PhyloXML.Events.ok_type
Bio.Phylo.PhyloXML.SequenceRelation.ok_type" class="py-name" href="#" onclick="return doclink('link-316', 'ok_type', 'link-312');">ok_type</a></tt><tt class="py-op">.</tt><tt id="link-317" class="py-name" targets="Method Bio.Crystal.Chain.__contains__()=Bio.Crystal.Chain-class.html#__contains__,Method Bio.Crystal.Crystal.__contains__()=Bio.Crystal.Crystal-class.html#__contains__,Method Bio.Data.CodonTable.AmbiguousForwardTable.__contains__()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#__contains__,Method Bio.File._IndexedSeqFileDict.__contains__()=Bio.File._IndexedSeqFileDict-class.html#__contains__,Method Bio.File._SQLiteManySeqFilesDict.__contains__()=Bio.File._SQLiteManySeqFilesDict-class.html#__contains__,Method Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__contains__()=Bio.PDB.AbstractPropertyMap.AbstractPropertyMap-class.html#__contains__,Method Bio.PDB.Chain.Chain.__contains__()=Bio.PDB.Chain.Chain-class.html#__contains__,Method Bio.PDB.Entity.DisorderedEntityWrapper.__contains__()=Bio.PDB.Entity.DisorderedEntityWrapper-class.html#__contains__,Method Bio.PDB.Entity.Entity.__contains__()=Bio.PDB.Entity.Entity-class.html#__contains__,Method Bio.PDB.FragmentMapper'.FragmentMapper.__contains__()=Bio.PDB.FragmentMapper%27.FragmentMapper-class.html#__contains__,Method Bio.PDB.vectors.Vector.__contains__()=Bio.PDB.vectors.Vector-class.html#__contains__,Method Bio.Phylo.PhyloXML.Events.__contains__()=Bio.Phylo.PhyloXML.Events-class.html#__contains__,Method Bio.Restriction.Restriction.RestrictionBatch.__contains__()=Bio.Restriction.Restriction.RestrictionBatch-class.html#__contains__,Method Bio.SearchIO._model.hit.Hit.__contains__()=Bio.SearchIO._model.hit.Hit-class.html#__contains__,Method Bio.SearchIO._model.hsp.HSP.__contains__()=Bio.SearchIO._model.hsp.HSP-class.html#__contains__,Method Bio.SearchIO._model.query.QueryResult.__contains__()=Bio.SearchIO._model.query.QueryResult-class.html#__contains__,Method Bio.Seq.Seq.__contains__()=Bio.Seq.Seq-class.html#__contains__,Method Bio.SeqFeature.CompoundLocation.__contains__()=Bio.SeqFeature.CompoundLocation-class.html#__contains__,Method Bio.SeqFeature.FeatureLocation.__contains__()=Bio.SeqFeature.FeatureLocation-class.html#__contains__,Method Bio.SeqFeature.SeqFeature.__contains__()=Bio.SeqFeature.SeqFeature-class.html#__contains__,Method Bio.SeqRecord.SeqRecord.__contains__()=Bio.SeqRecord.SeqRecord-class.html#__contains__,Method Bio.phenotype.phen_micro.PlateRecord.__contains__()=Bio.phenotype.phen_micro.PlateRecord-class.html#__contains__,Method BioSQL.BioSeqDatabase.BioSeqDatabase.__contains__()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#__contains__,Method BioSQL.BioSeqDatabase.DBServer.__contains__()=BioSQL.BioSeqDatabase.DBServer-class.html#__contains__"><a title="Bio.Crystal.Chain.__contains__
Bio.Crystal.Crystal.__contains__
Bio.Data.CodonTable.AmbiguousForwardTable.__contains__
Bio.File._IndexedSeqFileDict.__contains__
Bio.File._SQLiteManySeqFilesDict.__contains__
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__contains__
Bio.PDB.Chain.Chain.__contains__
Bio.PDB.Entity.DisorderedEntityWrapper.__contains__
Bio.PDB.Entity.Entity.__contains__
Bio.PDB.FragmentMapper'.FragmentMapper.__contains__
Bio.PDB.vectors.Vector.__contains__
Bio.Phylo.PhyloXML.Events.__contains__
Bio.Restriction.Restriction.RestrictionBatch.__contains__
Bio.SearchIO._model.hit.Hit.__contains__
Bio.SearchIO._model.hsp.HSP.__contains__
Bio.SearchIO._model.query.QueryResult.__contains__
Bio.Seq.Seq.__contains__
Bio.SeqFeature.CompoundLocation.__contains__
Bio.SeqFeature.FeatureLocation.__contains__
Bio.SeqFeature.SeqFeature.__contains__
Bio.SeqRecord.SeqRecord.__contains__
Bio.phenotype.phen_micro.PlateRecord.__contains__
BioSQL.BioSeqDatabase.BioSeqDatabase.__contains__
BioSQL.BioSeqDatabase.DBServer.__contains__" class="py-name" href="#" onclick="return doclink('link-317', '__contains__', 'link-317');">__contains__</a></tt><tt class="py-op">)</tt> </tt>
<a name="L817"></a><tt class="py-lineno"> 817</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-318" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-318', 'type', 'link-29');">type</a></tt> <tt class="py-op">=</tt> <tt id="link-319" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-319', 'type', 'link-29');">type</a></tt> </tt>
<a name="L818"></a><tt class="py-lineno"> 818</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-320" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.duplications()=Bio.Phylo.PhyloXMLIO.Writer-class.html#duplications"><a title="Bio.Phylo.PhyloXMLIO.Writer.duplications" class="py-name" href="#" onclick="return doclink('link-320', 'duplications', 'link-320');">duplications</a></tt> <tt class="py-op">=</tt> <tt id="link-321" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.duplications" class="py-name" href="#" onclick="return doclink('link-321', 'duplications', 'link-320');">duplications</a></tt> </tt>
<a name="L819"></a><tt class="py-lineno"> 819</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-322" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.speciations()=Bio.Phylo.PhyloXMLIO.Writer-class.html#speciations"><a title="Bio.Phylo.PhyloXMLIO.Writer.speciations" class="py-name" href="#" onclick="return doclink('link-322', 'speciations', 'link-322');">speciations</a></tt> <tt class="py-op">=</tt> <tt id="link-323" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.speciations" class="py-name" href="#" onclick="return doclink('link-323', 'speciations', 'link-322');">speciations</a></tt> </tt>
<a name="L820"></a><tt class="py-lineno"> 820</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-324" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.losses()=Bio.Phylo.PhyloXMLIO.Writer-class.html#losses"><a title="Bio.Phylo.PhyloXMLIO.Writer.losses" class="py-name" href="#" onclick="return doclink('link-324', 'losses', 'link-324');">losses</a></tt> <tt class="py-op">=</tt> <tt id="link-325" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.losses" class="py-name" href="#" onclick="return doclink('link-325', 'losses', 'link-324');">losses</a></tt> </tt>
<a name="L821"></a><tt class="py-lineno"> 821</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-326" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-326', 'confidence', 'link-94');">confidence</a></tt> <tt class="py-op">=</tt> <tt id="link-327" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-327', 'confidence', 'link-94');">confidence</a></tt> </tt>
</div><a name="L822"></a><tt class="py-lineno"> 822</tt>  <tt class="py-line"> </tt>
<a name="Events.items"></a><div id="Events.items-def"><a name="L823"></a><tt class="py-lineno"> 823</tt> <a class="py-toggle" href="#" id="Events.items-toggle" onclick="return toggle('Events.items');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Events-class.html#items">items</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Events.items-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Events.items-expanded"><a name="L824"></a><tt class="py-lineno"> 824</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__dict__</tt><tt class="py-op">.</tt><tt id="link-328" class="py-name" targets="Method Bio.Crystal.Crystal.items()=Bio.Crystal.Crystal-class.html#items,Method Bio.File._IndexedSeqFileDict.items()=Bio.File._IndexedSeqFileDict-class.html#items,Method Bio.Phylo.PhyloXML.Events.items()=Bio.Phylo.PhyloXML.Events-class.html#items,Variable Bio.SearchIO._model.query.QueryResult.items=Bio.SearchIO._model.query.QueryResult-class.html#items,Method BioSQL.BioSeqDatabase.BioSeqDatabase.items()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#items,Method BioSQL.BioSeqDatabase.DBServer.items()=BioSQL.BioSeqDatabase.DBServer-class.html#items"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-328', 'items', 'link-328');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt class="py-name">v</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">]</tt> </tt>
</div><a name="L825"></a><tt class="py-lineno"> 825</tt>  <tt class="py-line"> </tt>
<a name="Events.keys"></a><div id="Events.keys-def"><a name="L826"></a><tt class="py-lineno"> 826</tt> <a class="py-toggle" href="#" id="Events.keys-toggle" onclick="return toggle('Events.keys');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Events-class.html#keys">keys</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Events.keys-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Events.keys-expanded"><a name="L827"></a><tt class="py-lineno"> 827</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">k</tt> <tt class="py-keyword">for</tt> <tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__dict__</tt><tt class="py-op">.</tt><tt id="link-329" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-329', 'items', 'link-328');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt class="py-name">v</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">]</tt> </tt>
</div><a name="L828"></a><tt class="py-lineno"> 828</tt>  <tt class="py-line"> </tt>
<a name="Events.values"></a><div id="Events.values-def"><a name="L829"></a><tt class="py-lineno"> 829</tt> <a class="py-toggle" href="#" id="Events.values-toggle" onclick="return toggle('Events.values');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Events-class.html#values">values</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Events.values-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Events.values-expanded"><a name="L830"></a><tt class="py-lineno"> 830</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return values from a key-value pair in an Events dict."""</tt> </tt>
<a name="L831"></a><tt class="py-lineno"> 831</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">v</tt> <tt class="py-keyword">for</tt> <tt class="py-name">v</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__dict__</tt><tt class="py-op">.</tt><tt id="link-330" class="py-name" targets="Method Bio.Crystal.Crystal.values()=Bio.Crystal.Crystal-class.html#values,Method Bio.File._IndexedSeqFileDict.values()=Bio.File._IndexedSeqFileDict-class.html#values,Method Bio.Phylo.PhyloXML.Events.values()=Bio.Phylo.PhyloXML.Events-class.html#values,Method BioSQL.BioSeqDatabase.BioSeqDatabase.values()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#values,Method BioSQL.BioSeqDatabase.DBServer.values()=BioSQL.BioSeqDatabase.DBServer-class.html#values"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-330', 'values', 'link-330');">values</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt class="py-name">v</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">]</tt> </tt>
</div><a name="L832"></a><tt class="py-lineno"> 832</tt>  <tt class="py-line"> </tt>
<a name="Events.__len__"></a><div id="Events.__len__-def"><a name="L833"></a><tt class="py-lineno"> 833</tt> <a class="py-toggle" href="#" id="Events.__len__-toggle" onclick="return toggle('Events.__len__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Events-class.html#__len__">__len__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Events.__len__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Events.__len__-expanded"><a name="L834"></a><tt class="py-lineno"> 834</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Better way to do this?</tt> </tt>
<a name="L835"></a><tt class="py-lineno"> 835</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-331" class="py-name"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-331', 'values', 'link-330');">values</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L836"></a><tt class="py-lineno"> 836</tt>  <tt class="py-line"> </tt>
<a name="Events.__getitem__"></a><div id="Events.__getitem__-def"><a name="L837"></a><tt class="py-lineno"> 837</tt> <a class="py-toggle" href="#" id="Events.__getitem__-toggle" onclick="return toggle('Events.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Events-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">key</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Events.__getitem__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Events.__getitem__-expanded"><a name="L838"></a><tt class="py-lineno"> 838</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-name">key</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L839"></a><tt class="py-lineno"> 839</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">KeyError</tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">)</tt> </tt>
<a name="L840"></a><tt class="py-lineno"> 840</tt>  <tt class="py-line">        <tt class="py-name">val</tt> <tt class="py-op">=</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-name">key</tt><tt class="py-op">)</tt> </tt>
<a name="L841"></a><tt class="py-lineno"> 841</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">val</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L842"></a><tt class="py-lineno"> 842</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">KeyError</tt><tt class="py-op">(</tt><tt class="py-string">"%s has not been set in this object"</tt> <tt class="py-op">%</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L843"></a><tt class="py-lineno"> 843</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">val</tt> </tt>
</div><a name="L844"></a><tt class="py-lineno"> 844</tt>  <tt class="py-line"> </tt>
<a name="Events.__setitem__"></a><div id="Events.__setitem__-def"><a name="L845"></a><tt class="py-lineno"> 845</tt> <a class="py-toggle" href="#" id="Events.__setitem__-toggle" onclick="return toggle('Events.__setitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Events-class.html#__setitem__">__setitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">key</tt><tt class="py-op">,</tt> <tt class="py-param">val</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Events.__setitem__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Events.__setitem__-expanded"><a name="L846"></a><tt class="py-lineno"> 846</tt>  <tt class="py-line">        <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">val</tt><tt class="py-op">)</tt> </tt>
</div><a name="L847"></a><tt class="py-lineno"> 847</tt>  <tt class="py-line"> </tt>
<a name="Events.__delitem__"></a><div id="Events.__delitem__-def"><a name="L848"></a><tt class="py-lineno"> 848</tt> <a class="py-toggle" href="#" id="Events.__delitem__-toggle" onclick="return toggle('Events.__delitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Events-class.html#__delitem__">__delitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">key</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Events.__delitem__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Events.__delitem__-expanded"><a name="L849"></a><tt class="py-lineno"> 849</tt>  <tt class="py-line">        <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
</div><a name="L850"></a><tt class="py-lineno"> 850</tt>  <tt class="py-line"> </tt>
<a name="Events.__iter__"></a><div id="Events.__iter__-def"><a name="L851"></a><tt class="py-lineno"> 851</tt> <a class="py-toggle" href="#" id="Events.__iter__-toggle" onclick="return toggle('Events.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Events-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Events.__iter__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Events.__iter__-expanded"><a name="L852"></a><tt class="py-lineno"> 852</tt>  <tt class="py-line">        <tt class="py-docstring">"""Iterate over the keys present in a Events dict."""</tt> </tt>
<a name="L853"></a><tt class="py-lineno"> 853</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">iter</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-332" class="py-name" targets="Method Bio.Crystal.Crystal.keys()=Bio.Crystal.Crystal-class.html#keys,Variable Bio.Entrez.Parser.ListConsumer.keys=Bio.Entrez.Parser.ListConsumer-class.html#keys,Method Bio.File._IndexedSeqFileDict.keys()=Bio.File._IndexedSeqFileDict-class.html#keys,Method Bio.File._SQLiteManySeqFilesDict.keys()=Bio.File._SQLiteManySeqFilesDict-class.html#keys,Method Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys()=Bio.PDB.AbstractPropertyMap.AbstractPropertyMap-class.html#keys,Method Bio.Phylo.PhyloXML.Events.keys()=Bio.Phylo.PhyloXML.Events-class.html#keys,Method BioSQL.BioSeqDatabase.BioSeqDatabase.keys()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#keys,Method BioSQL.BioSeqDatabase.DBServer.keys()=BioSQL.BioSeqDatabase.DBServer-class.html#keys"><a title="Bio.Crystal.Crystal.keys
Bio.Entrez.Parser.ListConsumer.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-332', 'keys', 'link-332');">keys</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L854"></a><tt class="py-lineno"> 854</tt>  <tt class="py-line"> </tt>
<a name="Events.__contains__"></a><div id="Events.__contains__-def"><a name="L855"></a><tt class="py-lineno"> 855</tt> <a class="py-toggle" href="#" id="Events.__contains__-toggle" onclick="return toggle('Events.__contains__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Events-class.html#__contains__">__contains__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">key</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Events.__contains__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Events.__contains__-expanded"><a name="L856"></a><tt class="py-lineno"> 856</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-name">key</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-name">key</tt><tt class="py-op">)</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L857"></a><tt class="py-lineno"> 857</tt>  <tt class="py-line"> </tt>
<a name="Id"></a><div id="Id-def"><a name="L858"></a><tt class="py-lineno"> 858</tt>  <tt class="py-line"> </tt>
<a name="L859"></a><tt class="py-lineno"> 859</tt> <a class="py-toggle" href="#" id="Id-toggle" onclick="return toggle('Id');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Id-class.html">Id</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Id-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Id-expanded"><a name="L860"></a><tt class="py-lineno"> 860</tt>  <tt class="py-line">    <tt class="py-docstring">"""A general-purpose identifier element.</tt> </tt>
<a name="L861"></a><tt class="py-lineno"> 861</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L862"></a><tt class="py-lineno"> 862</tt>  <tt class="py-line"><tt class="py-docstring">    Allows to indicate the provider (or authority) of an identifier, e.g. NCBI,</tt> </tt>
<a name="L863"></a><tt class="py-lineno"> 863</tt>  <tt class="py-line"><tt class="py-docstring">    along with the value itself.</tt> </tt>
<a name="L864"></a><tt class="py-lineno"> 864</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L865"></a><tt class="py-lineno"> 865</tt>  <tt class="py-line"> </tt>
<a name="Id.__init__"></a><div id="Id.__init__-def"><a name="L866"></a><tt class="py-lineno"> 866</tt> <a class="py-toggle" href="#" id="Id.__init__-toggle" onclick="return toggle('Id.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Id-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">,</tt> <tt class="py-param">provider</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Id.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Id.__init__-expanded"><a name="L867"></a><tt class="py-lineno"> 867</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize vakues for the identifier object."""</tt> </tt>
<a name="L868"></a><tt class="py-lineno"> 868</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-333" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-333', 'value', 'link-37');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-334" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-334', 'value', 'link-37');">value</a></tt> </tt>
<a name="L869"></a><tt class="py-lineno"> 869</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">provider</tt> <tt class="py-op">=</tt> <tt class="py-name">provider</tt> </tt>
</div><a name="L870"></a><tt class="py-lineno"> 870</tt>  <tt class="py-line"> </tt>
<a name="Id.__str__"></a><div id="Id.__str__-def"><a name="L871"></a><tt class="py-lineno"> 871</tt> <a class="py-toggle" href="#" id="Id.__str__-toggle" onclick="return toggle('Id.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Id-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Id.__str__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Id.__str__-expanded"><a name="L872"></a><tt class="py-lineno"> 872</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">provider</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L873"></a><tt class="py-lineno"> 873</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-string">'%s:%s'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">provider</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-335" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-335', 'value', 'link-37');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L874"></a><tt class="py-lineno"> 874</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-336" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-336', 'value', 'link-37');">value</a></tt> </tt>
</div></div><a name="L875"></a><tt class="py-lineno"> 875</tt>  <tt class="py-line"> </tt>
<a name="MolSeq"></a><div id="MolSeq-def"><a name="L876"></a><tt class="py-lineno"> 876</tt>  <tt class="py-line"> </tt>
<a name="L877"></a><tt class="py-lineno"> 877</tt> <a class="py-toggle" href="#" id="MolSeq-toggle" onclick="return toggle('MolSeq');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.MolSeq-class.html">MolSeq</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MolSeq-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="MolSeq-expanded"><a name="L878"></a><tt class="py-lineno"> 878</tt>  <tt class="py-line">    <tt class="py-docstring">"""Store a molecular sequence.</tt> </tt>
<a name="L879"></a><tt class="py-lineno"> 879</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L880"></a><tt class="py-lineno"> 880</tt>  <tt class="py-line"><tt class="py-docstring">    :Parameters:</tt> </tt>
<a name="L881"></a><tt class="py-lineno"> 881</tt>  <tt class="py-line"><tt class="py-docstring">        value : string</tt> </tt>
<a name="L882"></a><tt class="py-lineno"> 882</tt>  <tt class="py-line"><tt class="py-docstring">            the sequence itself</tt> </tt>
<a name="L883"></a><tt class="py-lineno"> 883</tt>  <tt class="py-line"><tt class="py-docstring">        is_aligned : bool</tt> </tt>
<a name="L884"></a><tt class="py-lineno"> 884</tt>  <tt class="py-line"><tt class="py-docstring">            True if this sequence is aligned with the others (usually meaning</tt> </tt>
<a name="L885"></a><tt class="py-lineno"> 885</tt>  <tt class="py-line"><tt class="py-docstring">            all aligned seqs are the same length and gaps may be present)</tt> </tt>
<a name="L886"></a><tt class="py-lineno"> 886</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L887"></a><tt class="py-lineno"> 887</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L888"></a><tt class="py-lineno"> 888</tt>  <tt class="py-line"> </tt>
<a name="L889"></a><tt class="py-lineno"> 889</tt>  <tt class="py-line">    <tt id="link-337" class="py-name" targets="Variable Bio.Phylo.PhyloXML.MolSeq.re_value=Bio.Phylo.PhyloXML.MolSeq-class.html#re_value"><a title="Bio.Phylo.PhyloXML.MolSeq.re_value" class="py-name" href="#" onclick="return doclink('link-337', 're_value', 'link-337');">re_value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'[a-zA-Z\.\-\?\*_]+'</tt><tt class="py-op">)</tt> </tt>
<a name="L890"></a><tt class="py-lineno"> 890</tt>  <tt class="py-line"> </tt>
<a name="MolSeq.__init__"></a><div id="MolSeq.__init__-def"><a name="L891"></a><tt class="py-lineno"> 891</tt> <a class="py-toggle" href="#" id="MolSeq.__init__-toggle" onclick="return toggle('MolSeq.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.MolSeq-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">,</tt> <tt class="py-param">is_aligned</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MolSeq.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MolSeq.__init__-expanded"><a name="L892"></a><tt class="py-lineno"> 892</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize parameters for the MolSeq object."""</tt> </tt>
<a name="L893"></a><tt class="py-lineno"> 893</tt>  <tt class="py-line">        <tt id="link-338" class="py-name"><a title="Bio.Phylo.PhyloXML._check_str" class="py-name" href="#" onclick="return doclink('link-338', '_check_str', 'link-191');">_check_str</a></tt><tt class="py-op">(</tt><tt id="link-339" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-339', 'value', 'link-37');">value</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-340" class="py-name"><a title="Bio.Phylo.PhyloXML.MolSeq.re_value" class="py-name" href="#" onclick="return doclink('link-340', 're_value', 'link-337');">re_value</a></tt><tt class="py-op">.</tt><tt id="link-341" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-341', 'match', 'link-194');">match</a></tt><tt class="py-op">)</tt> </tt>
<a name="L894"></a><tt class="py-lineno"> 894</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-342" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-342', 'value', 'link-37');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-343" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-343', 'value', 'link-37');">value</a></tt> </tt>
<a name="L895"></a><tt class="py-lineno"> 895</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">is_aligned</tt> <tt class="py-op">=</tt> <tt class="py-name">is_aligned</tt> </tt>
</div><a name="L896"></a><tt class="py-lineno"> 896</tt>  <tt class="py-line"> </tt>
<a name="MolSeq.__str__"></a><div id="MolSeq.__str__-def"><a name="L897"></a><tt class="py-lineno"> 897</tt> <a class="py-toggle" href="#" id="MolSeq.__str__-toggle" onclick="return toggle('MolSeq.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.MolSeq-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MolSeq.__str__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="MolSeq.__str__-expanded"><a name="L898"></a><tt class="py-lineno"> 898</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the value of the Molecular Sequence object."""</tt> </tt>
<a name="L899"></a><tt class="py-lineno"> 899</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-344" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-344', 'value', 'link-37');">value</a></tt> </tt>
</div></div><a name="L900"></a><tt class="py-lineno"> 900</tt>  <tt class="py-line"> </tt>
<a name="Point"></a><div id="Point-def"><a name="L901"></a><tt class="py-lineno"> 901</tt>  <tt class="py-line"> </tt>
<a name="L902"></a><tt class="py-lineno"> 902</tt> <a class="py-toggle" href="#" id="Point-toggle" onclick="return toggle('Point');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Point-class.html">Point</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Point-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Point-expanded"><a name="L903"></a><tt class="py-lineno"> 903</tt>  <tt class="py-line">    <tt class="py-docstring">"""Geographic coordinates of a point, with an optional altitude.</tt> </tt>
<a name="L904"></a><tt class="py-lineno"> 904</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L905"></a><tt class="py-lineno"> 905</tt>  <tt class="py-line"><tt class="py-docstring">    Used by element 'Distribution'.</tt> </tt>
<a name="L906"></a><tt class="py-lineno"> 906</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L907"></a><tt class="py-lineno"> 907</tt>  <tt class="py-line"><tt class="py-docstring">    :Parameters:</tt> </tt>
<a name="L908"></a><tt class="py-lineno"> 908</tt>  <tt class="py-line"><tt class="py-docstring">        geodetic_datum : string, required</tt> </tt>
<a name="L909"></a><tt class="py-lineno"> 909</tt>  <tt class="py-line"><tt class="py-docstring">            the geodetic datum (also called 'map datum'). For example, Google's</tt> </tt>
<a name="L910"></a><tt class="py-lineno"> 910</tt>  <tt class="py-line"><tt class="py-docstring">            KML uses 'WGS84'.</tt> </tt>
<a name="L911"></a><tt class="py-lineno"> 911</tt>  <tt class="py-line"><tt class="py-docstring">        lat : numeric</tt> </tt>
<a name="L912"></a><tt class="py-lineno"> 912</tt>  <tt class="py-line"><tt class="py-docstring">            latitude</tt> </tt>
<a name="L913"></a><tt class="py-lineno"> 913</tt>  <tt class="py-line"><tt class="py-docstring">        long : numeric</tt> </tt>
<a name="L914"></a><tt class="py-lineno"> 914</tt>  <tt class="py-line"><tt class="py-docstring">            longitude</tt> </tt>
<a name="L915"></a><tt class="py-lineno"> 915</tt>  <tt class="py-line"><tt class="py-docstring">        alt : numeric</tt> </tt>
<a name="L916"></a><tt class="py-lineno"> 916</tt>  <tt class="py-line"><tt class="py-docstring">            altitude</tt> </tt>
<a name="L917"></a><tt class="py-lineno"> 917</tt>  <tt class="py-line"><tt class="py-docstring">        alt_unit : string</tt> </tt>
<a name="L918"></a><tt class="py-lineno"> 918</tt>  <tt class="py-line"><tt class="py-docstring">            unit for the altitude (e.g. 'meter')</tt> </tt>
<a name="L919"></a><tt class="py-lineno"> 919</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L920"></a><tt class="py-lineno"> 920</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L921"></a><tt class="py-lineno"> 921</tt>  <tt class="py-line"> </tt>
<a name="Point.__init__"></a><div id="Point.__init__-def"><a name="L922"></a><tt class="py-lineno"> 922</tt> <a class="py-toggle" href="#" id="Point.__init__-toggle" onclick="return toggle('Point.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Point-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">geodetic_datum</tt><tt class="py-op">,</tt> <tt class="py-param">lat</tt><tt class="py-op">,</tt> <tt class="py-param">long</tt><tt class="py-op">,</tt> <tt class="py-param">alt</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">alt_unit</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Point.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Point.__init__-expanded"><a name="L923"></a><tt class="py-lineno"> 923</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize value for the Point object."""</tt> </tt>
<a name="L924"></a><tt class="py-lineno"> 924</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">geodetic_datum</tt> <tt class="py-op">=</tt> <tt class="py-name">geodetic_datum</tt> </tt>
<a name="L925"></a><tt class="py-lineno"> 925</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-345" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.lat()=Bio.Phylo.PhyloXMLIO.Writer-class.html#lat"><a title="Bio.Phylo.PhyloXMLIO.Writer.lat" class="py-name" href="#" onclick="return doclink('link-345', 'lat', 'link-345');">lat</a></tt> <tt class="py-op">=</tt> <tt id="link-346" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.lat" class="py-name" href="#" onclick="return doclink('link-346', 'lat', 'link-345');">lat</a></tt> </tt>
<a name="L926"></a><tt class="py-lineno"> 926</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-347" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.long" class="py-name" href="#" onclick="return doclink('link-347', 'long', 'link-294');">long</a></tt> <tt class="py-op">=</tt> <tt id="link-348" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.long" class="py-name" href="#" onclick="return doclink('link-348', 'long', 'link-294');">long</a></tt> </tt>
<a name="L927"></a><tt class="py-lineno"> 927</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-349" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.alt()=Bio.Phylo.PhyloXMLIO.Writer-class.html#alt"><a title="Bio.Phylo.PhyloXMLIO.Writer.alt" class="py-name" href="#" onclick="return doclink('link-349', 'alt', 'link-349');">alt</a></tt> <tt class="py-op">=</tt> <tt id="link-350" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.alt" class="py-name" href="#" onclick="return doclink('link-350', 'alt', 'link-349');">alt</a></tt> </tt>
<a name="L928"></a><tt class="py-lineno"> 928</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alt_unit</tt> <tt class="py-op">=</tt> <tt class="py-name">alt_unit</tt> </tt>
</div></div><a name="L929"></a><tt class="py-lineno"> 929</tt>  <tt class="py-line"> </tt>
<a name="Polygon"></a><div id="Polygon-def"><a name="L930"></a><tt class="py-lineno"> 930</tt>  <tt class="py-line"> </tt>
<a name="L931"></a><tt class="py-lineno"> 931</tt> <a class="py-toggle" href="#" id="Polygon-toggle" onclick="return toggle('Polygon');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Polygon-class.html">Polygon</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Polygon-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Polygon-expanded"><a name="L932"></a><tt class="py-lineno"> 932</tt>  <tt class="py-line">    <tt class="py-docstring">"""A polygon defined by a list of 'Points' (used by element 'Distribution').</tt> </tt>
<a name="L933"></a><tt class="py-lineno"> 933</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L934"></a><tt class="py-lineno"> 934</tt>  <tt class="py-line"><tt class="py-docstring">    :param points: list of 3 or more points representing vertices.</tt> </tt>
<a name="L935"></a><tt class="py-lineno"> 935</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L936"></a><tt class="py-lineno"> 936</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L937"></a><tt class="py-lineno"> 937</tt>  <tt class="py-line"> </tt>
<a name="Polygon.__init__"></a><div id="Polygon.__init__-def"><a name="L938"></a><tt class="py-lineno"> 938</tt> <a class="py-toggle" href="#" id="Polygon.__init__-toggle" onclick="return toggle('Polygon.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Polygon-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">points</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Polygon.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Polygon.__init__-expanded"><a name="L939"></a><tt class="py-lineno"> 939</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize value for the Polygon object."""</tt> </tt>
<a name="L940"></a><tt class="py-lineno"> 940</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">points</tt> <tt class="py-op">=</tt> <tt class="py-name">points</tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div><a name="L941"></a><tt class="py-lineno"> 941</tt>  <tt class="py-line"> </tt>
<a name="Polygon.__str__"></a><div id="Polygon.__str__-def"><a name="L942"></a><tt class="py-lineno"> 942</tt> <a class="py-toggle" href="#" id="Polygon.__str__-toggle" onclick="return toggle('Polygon.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Polygon-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Polygon.__str__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Polygon.__str__-expanded"><a name="L943"></a><tt class="py-lineno"> 943</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">'%s([%s])'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">,</tt> </tt>
<a name="L944"></a><tt class="py-lineno"> 944</tt>  <tt class="py-line">                             <tt class="py-string">',\n'</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt id="link-351" class="py-name"><a title="Bio.SearchIO._model.hit.Hit.map" class="py-name" href="#" onclick="return doclink('link-351', 'map', 'link-34');">map</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">points</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L945"></a><tt class="py-lineno"> 945</tt>  <tt class="py-line"> </tt>
<a name="Property"></a><div id="Property-def"><a name="L946"></a><tt class="py-lineno"> 946</tt>  <tt class="py-line"> </tt>
<a name="L947"></a><tt class="py-lineno"> 947</tt> <a class="py-toggle" href="#" id="Property-toggle" onclick="return toggle('Property');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Property-class.html">Property</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Property-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Property-expanded"><a name="L948"></a><tt class="py-lineno"> 948</tt>  <tt class="py-line">    <tt class="py-docstring">"""A typed and referenced property from an external resources.</tt> </tt>
<a name="L949"></a><tt class="py-lineno"> 949</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L950"></a><tt class="py-lineno"> 950</tt>  <tt class="py-line"><tt class="py-docstring">    Can be attached to `Phylogeny`, `Clade`, and `Annotation` objects.</tt> </tt>
<a name="L951"></a><tt class="py-lineno"> 951</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L952"></a><tt class="py-lineno"> 952</tt>  <tt class="py-line"><tt class="py-docstring">    :Parameters:</tt> </tt>
<a name="L953"></a><tt class="py-lineno"> 953</tt>  <tt class="py-line"><tt class="py-docstring">        value : string</tt> </tt>
<a name="L954"></a><tt class="py-lineno"> 954</tt>  <tt class="py-line"><tt class="py-docstring">            the value of the property</tt> </tt>
<a name="L955"></a><tt class="py-lineno"> 955</tt>  <tt class="py-line"><tt class="py-docstring">        ref : string</tt> </tt>
<a name="L956"></a><tt class="py-lineno"> 956</tt>  <tt class="py-line"><tt class="py-docstring">            reference to an external resource, e.g. "NOAA:depth"</tt> </tt>
<a name="L957"></a><tt class="py-lineno"> 957</tt>  <tt class="py-line"><tt class="py-docstring">        applies_to : string</tt> </tt>
<a name="L958"></a><tt class="py-lineno"> 958</tt>  <tt class="py-line"><tt class="py-docstring">            indicates the item to which a property applies to (e.g.  'node' for</tt> </tt>
<a name="L959"></a><tt class="py-lineno"> 959</tt>  <tt class="py-line"><tt class="py-docstring">            the parent node of a clade, 'parent_branch' for the parent branch of</tt> </tt>
<a name="L960"></a><tt class="py-lineno"> 960</tt>  <tt class="py-line"><tt class="py-docstring">            a clade, or just 'clade').</tt> </tt>
<a name="L961"></a><tt class="py-lineno"> 961</tt>  <tt class="py-line"><tt class="py-docstring">        datatype : string</tt> </tt>
<a name="L962"></a><tt class="py-lineno"> 962</tt>  <tt class="py-line"><tt class="py-docstring">            the type of a property; limited to xsd-datatypes</tt> </tt>
<a name="L963"></a><tt class="py-lineno"> 963</tt>  <tt class="py-line"><tt class="py-docstring">            (e.g. 'xsd:string', 'xsd:boolean', 'xsd:integer', 'xsd:decimal',</tt> </tt>
<a name="L964"></a><tt class="py-lineno"> 964</tt>  <tt class="py-line"><tt class="py-docstring">            'xsd:float', 'xsd:double', 'xsd:date', 'xsd:anyURI').</tt> </tt>
<a name="L965"></a><tt class="py-lineno"> 965</tt>  <tt class="py-line"><tt class="py-docstring">        unit : string (optional)</tt> </tt>
<a name="L966"></a><tt class="py-lineno"> 966</tt>  <tt class="py-line"><tt class="py-docstring">            the unit of the property, e.g. "METRIC:m"</tt> </tt>
<a name="L967"></a><tt class="py-lineno"> 967</tt>  <tt class="py-line"><tt class="py-docstring">        id_ref : Id (optional)</tt> </tt>
<a name="L968"></a><tt class="py-lineno"> 968</tt>  <tt class="py-line"><tt class="py-docstring">            allows to attached a property specifically to one element (on the</tt> </tt>
<a name="L969"></a><tt class="py-lineno"> 969</tt>  <tt class="py-line"><tt class="py-docstring">            xml-level)</tt> </tt>
<a name="L970"></a><tt class="py-lineno"> 970</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L971"></a><tt class="py-lineno"> 971</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L972"></a><tt class="py-lineno"> 972</tt>  <tt class="py-line"> </tt>
<a name="L973"></a><tt class="py-lineno"> 973</tt>  <tt class="py-line">    <tt id="link-352" class="py-name"><a title="Bio.Phylo.PhyloXML.Annotation.re_ref
Bio.Phylo.PhyloXML.Property.re_ref" class="py-name" href="#" onclick="return doclink('link-352', 're_ref', 'link-190');">re_ref</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'[a-zA-Z0-9_]+:[a-zA-Z0-9_\.\-\s]+'</tt><tt class="py-op">)</tt> </tt>
<a name="L974"></a><tt class="py-lineno"> 974</tt>  <tt class="py-line">    <tt id="link-353" class="py-name" targets="Variable Bio.Phylo.PhyloXML.Property.ok_applies_to=Bio.Phylo.PhyloXML.Property-class.html#ok_applies_to"><a title="Bio.Phylo.PhyloXML.Property.ok_applies_to" class="py-name" href="#" onclick="return doclink('link-353', 'ok_applies_to', 'link-353');">ok_applies_to</a></tt> <tt class="py-op">=</tt> <tt id="link-354" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-354', 'set', 'link-313');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">'phylogeny'</tt><tt class="py-op">,</tt> <tt class="py-string">'clade'</tt><tt class="py-op">,</tt> <tt class="py-string">'node'</tt><tt class="py-op">,</tt> <tt class="py-string">'annotation'</tt><tt class="py-op">,</tt> </tt>
<a name="L975"></a><tt class="py-lineno"> 975</tt>  <tt class="py-line">                         <tt class="py-string">'parent_branch'</tt><tt class="py-op">,</tt> <tt class="py-string">'other'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L976"></a><tt class="py-lineno"> 976</tt>  <tt class="py-line">    <tt id="link-355" class="py-name" targets="Variable Bio.Phylo.PhyloXML.Property.ok_datatype=Bio.Phylo.PhyloXML.Property-class.html#ok_datatype"><a title="Bio.Phylo.PhyloXML.Property.ok_datatype" class="py-name" href="#" onclick="return doclink('link-355', 'ok_datatype', 'link-355');">ok_datatype</a></tt> <tt class="py-op">=</tt> <tt id="link-356" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-356', 'set', 'link-313');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">'xsd:string'</tt><tt class="py-op">,</tt> <tt class="py-string">'xsd:boolean'</tt><tt class="py-op">,</tt> <tt class="py-string">'xsd:decimal'</tt><tt class="py-op">,</tt> <tt class="py-string">'xsd:float'</tt><tt class="py-op">,</tt> </tt>
<a name="L977"></a><tt class="py-lineno"> 977</tt>  <tt class="py-line">                       <tt class="py-string">'xsd:double'</tt><tt class="py-op">,</tt> <tt class="py-string">'xsd:duration'</tt><tt class="py-op">,</tt> <tt class="py-string">'xsd:dateTime'</tt><tt class="py-op">,</tt> <tt class="py-string">'xsd:time'</tt><tt class="py-op">,</tt> <tt class="py-string">'xsd:date'</tt><tt class="py-op">,</tt> </tt>
<a name="L978"></a><tt class="py-lineno"> 978</tt>  <tt class="py-line">                       <tt class="py-string">'xsd:gYearMonth'</tt><tt class="py-op">,</tt> <tt class="py-string">'xsd:gYear'</tt><tt class="py-op">,</tt> <tt class="py-string">'xsd:gMonthDay'</tt><tt class="py-op">,</tt> <tt class="py-string">'xsd:gDay'</tt><tt class="py-op">,</tt> </tt>
<a name="L979"></a><tt class="py-lineno"> 979</tt>  <tt class="py-line">                       <tt class="py-string">'xsd:gMonth'</tt><tt class="py-op">,</tt> <tt class="py-string">'xsd:hexBinary'</tt><tt class="py-op">,</tt> <tt class="py-string">'xsd:base64Binary'</tt><tt class="py-op">,</tt> <tt class="py-string">'xsd:anyURI'</tt><tt class="py-op">,</tt> </tt>
<a name="L980"></a><tt class="py-lineno"> 980</tt>  <tt class="py-line">                       <tt class="py-string">'xsd:normalizedString'</tt><tt class="py-op">,</tt> <tt class="py-string">'xsd:token'</tt><tt class="py-op">,</tt> <tt class="py-string">'xsd:integer'</tt><tt class="py-op">,</tt> </tt>
<a name="L981"></a><tt class="py-lineno"> 981</tt>  <tt class="py-line">                       <tt class="py-string">'xsd:nonPositiveInteger'</tt><tt class="py-op">,</tt> <tt class="py-string">'xsd:negativeInteger'</tt><tt class="py-op">,</tt> <tt class="py-string">'xsd:long'</tt><tt class="py-op">,</tt> <tt class="py-string">'xsd:int'</tt><tt class="py-op">,</tt> </tt>
<a name="L982"></a><tt class="py-lineno"> 982</tt>  <tt class="py-line">                       <tt class="py-string">'xsd:short'</tt><tt class="py-op">,</tt> <tt class="py-string">'xsd:byte'</tt><tt class="py-op">,</tt> <tt class="py-string">'xsd:nonNegativeInteger'</tt><tt class="py-op">,</tt> <tt class="py-string">'xsd:unsignedLong'</tt><tt class="py-op">,</tt> </tt>
<a name="L983"></a><tt class="py-lineno"> 983</tt>  <tt class="py-line">                       <tt class="py-string">'xsd:unsignedInt'</tt><tt class="py-op">,</tt> <tt class="py-string">'xsd:unsignedShort'</tt><tt class="py-op">,</tt> <tt class="py-string">'xsd:unsignedByte'</tt><tt class="py-op">,</tt> </tt>
<a name="L984"></a><tt class="py-lineno"> 984</tt>  <tt class="py-line">                       <tt class="py-string">'xsd:positiveInteger'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L985"></a><tt class="py-lineno"> 985</tt>  <tt class="py-line"> </tt>
<a name="Property.__init__"></a><div id="Property.__init__-def"><a name="L986"></a><tt class="py-lineno"> 986</tt> <a class="py-toggle" href="#" id="Property.__init__-toggle" onclick="return toggle('Property.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Property-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">,</tt> <tt class="py-param">ref</tt><tt class="py-op">,</tt> <tt class="py-param">applies_to</tt><tt class="py-op">,</tt> <tt class="py-param">datatype</tt><tt class="py-op">,</tt> </tt>
<a name="L987"></a><tt class="py-lineno"> 987</tt>  <tt class="py-line">                 <tt class="py-param">unit</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">id_ref</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Property.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Property.__init__-expanded"><a name="L988"></a><tt class="py-lineno"> 988</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize value for the Property object."""</tt> </tt>
<a name="L989"></a><tt class="py-lineno"> 989</tt>  <tt class="py-line">        <tt id="link-357" class="py-name"><a title="Bio.Phylo.PhyloXML._check_str" class="py-name" href="#" onclick="return doclink('link-357', '_check_str', 'link-191');">_check_str</a></tt><tt class="py-op">(</tt><tt id="link-358" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-358', 'ref', 'link-192');">ref</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-359" class="py-name"><a title="Bio.Phylo.PhyloXML.Annotation.re_ref
Bio.Phylo.PhyloXML.Property.re_ref" class="py-name" href="#" onclick="return doclink('link-359', 're_ref', 'link-190');">re_ref</a></tt><tt class="py-op">.</tt><tt id="link-360" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-360', 'match', 'link-194');">match</a></tt><tt class="py-op">)</tt> </tt>
<a name="L990"></a><tt class="py-lineno"> 990</tt>  <tt class="py-line">        <tt id="link-361" class="py-name"><a title="Bio.Phylo.PhyloXML._check_str" class="py-name" href="#" onclick="return doclink('link-361', '_check_str', 'link-191');">_check_str</a></tt><tt class="py-op">(</tt><tt class="py-name">applies_to</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-362" class="py-name"><a title="Bio.Phylo.PhyloXML.Property.ok_applies_to" class="py-name" href="#" onclick="return doclink('link-362', 'ok_applies_to', 'link-353');">ok_applies_to</a></tt><tt class="py-op">.</tt><tt id="link-363" class="py-name"><a title="Bio.Crystal.Chain.__contains__
Bio.Crystal.Crystal.__contains__
Bio.Data.CodonTable.AmbiguousForwardTable.__contains__
Bio.File._IndexedSeqFileDict.__contains__
Bio.File._SQLiteManySeqFilesDict.__contains__
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__contains__
Bio.PDB.Chain.Chain.__contains__
Bio.PDB.Entity.DisorderedEntityWrapper.__contains__
Bio.PDB.Entity.Entity.__contains__
Bio.PDB.FragmentMapper'.FragmentMapper.__contains__
Bio.PDB.vectors.Vector.__contains__
Bio.Phylo.PhyloXML.Events.__contains__
Bio.Restriction.Restriction.RestrictionBatch.__contains__
Bio.SearchIO._model.hit.Hit.__contains__
Bio.SearchIO._model.hsp.HSP.__contains__
Bio.SearchIO._model.query.QueryResult.__contains__
Bio.Seq.Seq.__contains__
Bio.SeqFeature.CompoundLocation.__contains__
Bio.SeqFeature.FeatureLocation.__contains__
Bio.SeqFeature.SeqFeature.__contains__
Bio.SeqRecord.SeqRecord.__contains__
Bio.phenotype.phen_micro.PlateRecord.__contains__
BioSQL.BioSeqDatabase.BioSeqDatabase.__contains__
BioSQL.BioSeqDatabase.DBServer.__contains__" class="py-name" href="#" onclick="return doclink('link-363', '__contains__', 'link-317');">__contains__</a></tt><tt class="py-op">)</tt> </tt>
<a name="L991"></a><tt class="py-lineno"> 991</tt>  <tt class="py-line">        <tt id="link-364" class="py-name"><a title="Bio.Phylo.PhyloXML._check_str" class="py-name" href="#" onclick="return doclink('link-364', '_check_str', 'link-191');">_check_str</a></tt><tt class="py-op">(</tt><tt class="py-name">datatype</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-365" class="py-name"><a title="Bio.Phylo.PhyloXML.Property.ok_datatype" class="py-name" href="#" onclick="return doclink('link-365', 'ok_datatype', 'link-355');">ok_datatype</a></tt><tt class="py-op">.</tt><tt id="link-366" class="py-name"><a title="Bio.Crystal.Chain.__contains__
Bio.Crystal.Crystal.__contains__
Bio.Data.CodonTable.AmbiguousForwardTable.__contains__
Bio.File._IndexedSeqFileDict.__contains__
Bio.File._SQLiteManySeqFilesDict.__contains__
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__contains__
Bio.PDB.Chain.Chain.__contains__
Bio.PDB.Entity.DisorderedEntityWrapper.__contains__
Bio.PDB.Entity.Entity.__contains__
Bio.PDB.FragmentMapper'.FragmentMapper.__contains__
Bio.PDB.vectors.Vector.__contains__
Bio.Phylo.PhyloXML.Events.__contains__
Bio.Restriction.Restriction.RestrictionBatch.__contains__
Bio.SearchIO._model.hit.Hit.__contains__
Bio.SearchIO._model.hsp.HSP.__contains__
Bio.SearchIO._model.query.QueryResult.__contains__
Bio.Seq.Seq.__contains__
Bio.SeqFeature.CompoundLocation.__contains__
Bio.SeqFeature.FeatureLocation.__contains__
Bio.SeqFeature.SeqFeature.__contains__
Bio.SeqRecord.SeqRecord.__contains__
Bio.phenotype.phen_micro.PlateRecord.__contains__
BioSQL.BioSeqDatabase.BioSeqDatabase.__contains__
BioSQL.BioSeqDatabase.DBServer.__contains__" class="py-name" href="#" onclick="return doclink('link-366', '__contains__', 'link-317');">__contains__</a></tt><tt class="py-op">)</tt> </tt>
<a name="L992"></a><tt class="py-lineno"> 992</tt>  <tt class="py-line">        <tt id="link-367" class="py-name"><a title="Bio.Phylo.PhyloXML._check_str" class="py-name" href="#" onclick="return doclink('link-367', '_check_str', 'link-191');">_check_str</a></tt><tt class="py-op">(</tt><tt class="py-name">unit</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-368" class="py-name"><a title="Bio.Phylo.PhyloXML.Annotation.re_ref
Bio.Phylo.PhyloXML.Property.re_ref" class="py-name" href="#" onclick="return doclink('link-368', 're_ref', 'link-190');">re_ref</a></tt><tt class="py-op">.</tt><tt id="link-369" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-369', 'match', 'link-194');">match</a></tt><tt class="py-op">)</tt> </tt>
<a name="L993"></a><tt class="py-lineno"> 993</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">unit</tt> <tt class="py-op">=</tt> <tt class="py-name">unit</tt> </tt>
<a name="L994"></a><tt class="py-lineno"> 994</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">id_ref</tt> <tt class="py-op">=</tt> <tt class="py-name">id_ref</tt> </tt>
<a name="L995"></a><tt class="py-lineno"> 995</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-370" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-370', 'value', 'link-37');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-371" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-371', 'value', 'link-37');">value</a></tt> </tt>
<a name="L996"></a><tt class="py-lineno"> 996</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-372" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-372', 'ref', 'link-192');">ref</a></tt> <tt class="py-op">=</tt> <tt id="link-373" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-373', 'ref', 'link-192');">ref</a></tt> </tt>
<a name="L997"></a><tt class="py-lineno"> 997</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">applies_to</tt> <tt class="py-op">=</tt> <tt class="py-name">applies_to</tt> </tt>
<a name="L998"></a><tt class="py-lineno"> 998</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">datatype</tt> <tt class="py-op">=</tt> <tt class="py-name">datatype</tt> </tt>
</div></div><a name="L999"></a><tt class="py-lineno"> 999</tt>  <tt class="py-line"> </tt>
<a name="ProteinDomain"></a><div id="ProteinDomain-def"><a name="L1000"></a><tt class="py-lineno">1000</tt>  <tt class="py-line"> </tt>
<a name="L1001"></a><tt class="py-lineno">1001</tt> <a class="py-toggle" href="#" id="ProteinDomain-toggle" onclick="return toggle('ProteinDomain');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.ProteinDomain-class.html">ProteinDomain</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ProteinDomain-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="ProteinDomain-expanded"><a name="L1002"></a><tt class="py-lineno">1002</tt>  <tt class="py-line">    <tt class="py-docstring">"""Represents an individual domain in a domain architecture.</tt> </tt>
<a name="L1003"></a><tt class="py-lineno">1003</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1004"></a><tt class="py-lineno">1004</tt>  <tt class="py-line"><tt class="py-docstring">    The locations use 0-based indexing, as most Python objects including</tt> </tt>
<a name="L1005"></a><tt class="py-lineno">1005</tt>  <tt class="py-line"><tt class="py-docstring">    SeqFeature do, rather than the usual biological convention starting at 1.</tt> </tt>
<a name="L1006"></a><tt class="py-lineno">1006</tt>  <tt class="py-line"><tt class="py-docstring">    This means the start and end attributes can be used directly as slice</tt> </tt>
<a name="L1007"></a><tt class="py-lineno">1007</tt>  <tt class="py-line"><tt class="py-docstring">    indexes on Seq objects.</tt> </tt>
<a name="L1008"></a><tt class="py-lineno">1008</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1009"></a><tt class="py-lineno">1009</tt>  <tt class="py-line"><tt class="py-docstring">    :Parameters:</tt> </tt>
<a name="L1010"></a><tt class="py-lineno">1010</tt>  <tt class="py-line"><tt class="py-docstring">        start : non-negative integer</tt> </tt>
<a name="L1011"></a><tt class="py-lineno">1011</tt>  <tt class="py-line"><tt class="py-docstring">            start of the domain on the sequence, using 0-based indexing</tt> </tt>
<a name="L1012"></a><tt class="py-lineno">1012</tt>  <tt class="py-line"><tt class="py-docstring">        end : non-negative integer</tt> </tt>
<a name="L1013"></a><tt class="py-lineno">1013</tt>  <tt class="py-line"><tt class="py-docstring">            end of the domain on the sequence</tt> </tt>
<a name="L1014"></a><tt class="py-lineno">1014</tt>  <tt class="py-line"><tt class="py-docstring">        confidence : float</tt> </tt>
<a name="L1015"></a><tt class="py-lineno">1015</tt>  <tt class="py-line"><tt class="py-docstring">            can be used to store e.g. E-values</tt> </tt>
<a name="L1016"></a><tt class="py-lineno">1016</tt>  <tt class="py-line"><tt class="py-docstring">        id : string</tt> </tt>
<a name="L1017"></a><tt class="py-lineno">1017</tt>  <tt class="py-line"><tt class="py-docstring">            unique identifier/name</tt> </tt>
<a name="L1018"></a><tt class="py-lineno">1018</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1019"></a><tt class="py-lineno">1019</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1020"></a><tt class="py-lineno">1020</tt>  <tt class="py-line"> </tt>
<a name="ProteinDomain.__init__"></a><div id="ProteinDomain.__init__-def"><a name="L1021"></a><tt class="py-lineno">1021</tt> <a class="py-toggle" href="#" id="ProteinDomain.__init__-toggle" onclick="return toggle('ProteinDomain.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.ProteinDomain-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">,</tt> <tt class="py-param">confidence</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">id</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ProteinDomain.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="ProteinDomain.__init__-expanded"><a name="L1022"></a><tt class="py-lineno">1022</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize value for a ProteinDomain object."""</tt> </tt>
<a name="L1023"></a><tt class="py-lineno">1023</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-374" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-374', 'value', 'link-37');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-375" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-375', 'value', 'link-37');">value</a></tt> </tt>
<a name="L1024"></a><tt class="py-lineno">1024</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-376" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.start=Bio.SeqFeature.CompoundLocation-class.html#start,Variable Bio.SeqFeature.FeatureLocation.start=Bio.SeqFeature.FeatureLocation-class.html#start"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-376', 'start', 'link-376');">start</a></tt> <tt class="py-op">=</tt> <tt id="link-377" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-377', 'start', 'link-376');">start</a></tt> </tt>
<a name="L1025"></a><tt class="py-lineno">1025</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-378" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.end=Bio.SeqFeature.CompoundLocation-class.html#end,Variable Bio.SeqFeature.FeatureLocation.end=Bio.SeqFeature.FeatureLocation-class.html#end"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-378', 'end', 'link-378');">end</a></tt> <tt class="py-op">=</tt> <tt id="link-379" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-379', 'end', 'link-378');">end</a></tt> </tt>
<a name="L1026"></a><tt class="py-lineno">1026</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-380" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-380', 'confidence', 'link-94');">confidence</a></tt> <tt class="py-op">=</tt> <tt id="link-381" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-381', 'confidence', 'link-94');">confidence</a></tt> </tt>
<a name="L1027"></a><tt class="py-lineno">1027</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-382" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-382', 'id', 'link-48');">id</a></tt> <tt class="py-op">=</tt> <tt id="link-383" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-383', 'id', 'link-48');">id</a></tt> </tt>
</div><a name="L1028"></a><tt class="py-lineno">1028</tt>  <tt class="py-line"> </tt>
<a name="L1029"></a><tt class="py-lineno">1029</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">classmethod</tt> </tt>
<a name="ProteinDomain.from_seqfeature"></a><div id="ProteinDomain.from_seqfeature-def"><a name="L1030"></a><tt class="py-lineno">1030</tt> <a class="py-toggle" href="#" id="ProteinDomain.from_seqfeature-toggle" onclick="return toggle('ProteinDomain.from_seqfeature');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.ProteinDomain-class.html#from_seqfeature">from_seqfeature</a><tt class="py-op">(</tt><tt class="py-param">cls</tt><tt class="py-op">,</tt> <tt class="py-param">feat</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ProteinDomain.from_seqfeature-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="ProteinDomain.from_seqfeature-expanded"><a name="L1031"></a><tt class="py-lineno">1031</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-384" class="py-name" targets="Class Bio.Phylo.PhyloXML.ProteinDomain=Bio.Phylo.PhyloXML.ProteinDomain-class.html"><a title="Bio.Phylo.PhyloXML.ProteinDomain" class="py-name" href="#" onclick="return doclink('link-384', 'ProteinDomain', 'link-384');">ProteinDomain</a></tt><tt class="py-op">(</tt><tt class="py-name">feat</tt><tt class="py-op">.</tt><tt id="link-385" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-385', 'id', 'link-48');">id</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1032"></a><tt class="py-lineno">1032</tt>  <tt class="py-line">                             <tt class="py-name">feat</tt><tt class="py-op">.</tt><tt id="link-386" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.location()=Bio.GenBank._FeatureConsumer-class.html#location,Method Bio.GenBank._RecordConsumer.location()=Bio.GenBank._RecordConsumer-class.html#location,Method Bio.Phylo.PhyloXMLIO.Writer.location()=Bio.Phylo.PhyloXMLIO.Writer-class.html#location"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-386', 'location', 'link-386');">location</a></tt><tt class="py-op">.</tt><tt id="link-387" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.nofuzzy_start=Bio.SeqFeature.CompoundLocation-class.html#nofuzzy_start,Variable Bio.SeqFeature.FeatureLocation.nofuzzy_start=Bio.SeqFeature.FeatureLocation-class.html#nofuzzy_start"><a title="Bio.SeqFeature.CompoundLocation.nofuzzy_start
Bio.SeqFeature.FeatureLocation.nofuzzy_start" class="py-name" href="#" onclick="return doclink('link-387', 'nofuzzy_start', 'link-387');">nofuzzy_start</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1033"></a><tt class="py-lineno">1033</tt>  <tt class="py-line">                             <tt class="py-name">feat</tt><tt class="py-op">.</tt><tt id="link-388" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-388', 'location', 'link-386');">location</a></tt><tt class="py-op">.</tt><tt id="link-389" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.nofuzzy_end=Bio.SeqFeature.CompoundLocation-class.html#nofuzzy_end,Variable Bio.SeqFeature.FeatureLocation.nofuzzy_end=Bio.SeqFeature.FeatureLocation-class.html#nofuzzy_end"><a title="Bio.SeqFeature.CompoundLocation.nofuzzy_end
Bio.SeqFeature.FeatureLocation.nofuzzy_end" class="py-name" href="#" onclick="return doclink('link-389', 'nofuzzy_end', 'link-389');">nofuzzy_end</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1034"></a><tt class="py-lineno">1034</tt>  <tt class="py-line">                             <tt id="link-390" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-390', 'confidence', 'link-94');">confidence</a></tt><tt class="py-op">=</tt><tt class="py-name">feat</tt><tt class="py-op">.</tt><tt class="py-name">qualifiers</tt><tt class="py-op">.</tt><tt id="link-391" class="py-name" targets="Method Bio.Crystal.Crystal.get()=Bio.Crystal.Crystal-class.html#get,Method Bio.Data.CodonTable.AmbiguousForwardTable.get()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#get,Method Bio.File._IndexedSeqFileDict.get()=Bio.File._IndexedSeqFileDict-class.html#get,Method Bio.File._IndexedSeqFileProxy.get()=Bio.File._IndexedSeqFileProxy-class.html#get,Method Bio.File._SQLiteManySeqFilesDict.get()=Bio.File._SQLiteManySeqFilesDict-class.html#get,Method Bio.Restriction.Restriction.RestrictionBatch.get()=Bio.Restriction.Restriction.RestrictionBatch-class.html#get,Method Bio.SearchIO._index.SearchIndexer.get()=Bio.SearchIO._index.SearchIndexer-class.html#get,Method Bio.SeqIO._index.SeqFileRandomAccess.get()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#get,Method Bio.SeqIO._index.SffRandomAccess.get()=Bio.SeqIO._index.SffRandomAccess-class.html#get,Method Bio.SeqIO._index.SffTrimedRandomAccess.get()=Bio.SeqIO._index.SffTrimedRandomAccess-class.html#get,Method Bio.SeqIO._index.UniprotRandomAccess.get()=Bio.SeqIO._index.UniprotRandomAccess-class.html#get"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-391', 'get', 'link-391');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'confidence'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1035"></a><tt class="py-lineno">1035</tt>  <tt class="py-line"> </tt>
<a name="ProteinDomain.to_seqfeature"></a><div id="ProteinDomain.to_seqfeature-def"><a name="L1036"></a><tt class="py-lineno">1036</tt> <a class="py-toggle" href="#" id="ProteinDomain.to_seqfeature-toggle" onclick="return toggle('ProteinDomain.to_seqfeature');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.ProteinDomain-class.html#to_seqfeature">to_seqfeature</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ProteinDomain.to_seqfeature-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="ProteinDomain.to_seqfeature-expanded"><a name="L1037"></a><tt class="py-lineno">1037</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a SeqFeature from the ProteinDomain Object."""</tt> </tt>
<a name="L1038"></a><tt class="py-lineno">1038</tt>  <tt class="py-line">        <tt class="py-name">feat</tt> <tt class="py-op">=</tt> <tt id="link-392" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-392', 'SeqFeature', 'link-11');">SeqFeature</a></tt><tt class="py-op">(</tt><tt id="link-393" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-393', 'location', 'link-386');">location</a></tt><tt class="py-op">=</tt><tt id="link-394" class="py-name"><a title="Bio.SeqFeature.FeatureLocation" class="py-name" href="#" onclick="return doclink('link-394', 'FeatureLocation', 'link-13');">FeatureLocation</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-395" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-395', 'start', 'link-376');">start</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-396" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-396', 'end', 'link-378');">end</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1039"></a><tt class="py-lineno">1039</tt>  <tt class="py-line">                          <tt id="link-397" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-397', 'id', 'link-48');">id</a></tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-398" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-398', 'value', 'link-37');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1040"></a><tt class="py-lineno">1040</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-string">'confidence'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1041"></a><tt class="py-lineno">1041</tt>  <tt class="py-line">            <tt class="py-name">feat</tt><tt class="py-op">.</tt><tt class="py-name">qualifiers</tt><tt class="py-op">[</tt><tt class="py-string">'confidence'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-399" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-399', 'confidence', 'link-94');">confidence</a></tt> </tt>
<a name="L1042"></a><tt class="py-lineno">1042</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">feat</tt> </tt>
</div></div><a name="L1043"></a><tt class="py-lineno">1043</tt>  <tt class="py-line"> </tt>
<a name="Reference"></a><div id="Reference-def"><a name="L1044"></a><tt class="py-lineno">1044</tt>  <tt class="py-line"> </tt>
<a name="L1045"></a><tt class="py-lineno">1045</tt> <a class="py-toggle" href="#" id="Reference-toggle" onclick="return toggle('Reference');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Reference-class.html">Reference</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reference-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Reference-expanded"><a name="L1046"></a><tt class="py-lineno">1046</tt>  <tt class="py-line">    <tt class="py-docstring">"""Literature reference for a clade.</tt> </tt>
<a name="L1047"></a><tt class="py-lineno">1047</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1048"></a><tt class="py-lineno">1048</tt>  <tt class="py-line"><tt class="py-docstring">    NB: Whenever possible, use the ``doi`` attribute instead of the free-text</tt> </tt>
<a name="L1049"></a><tt class="py-lineno">1049</tt>  <tt class="py-line"><tt class="py-docstring">    ``desc`` element.</tt> </tt>
<a name="L1050"></a><tt class="py-lineno">1050</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1051"></a><tt class="py-lineno">1051</tt>  <tt class="py-line"> </tt>
<a name="L1052"></a><tt class="py-lineno">1052</tt>  <tt class="py-line">    <tt id="link-400" class="py-name" targets="Variable Bio.Phylo.PhyloXML.Reference.re_doi=Bio.Phylo.PhyloXML.Reference-class.html#re_doi"><a title="Bio.Phylo.PhyloXML.Reference.re_doi" class="py-name" href="#" onclick="return doclink('link-400', 're_doi', 'link-400');">re_doi</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'[a-zA-Z0-9_\.]+/[a-zA-Z0-9_\.]+'</tt><tt class="py-op">)</tt> </tt>
<a name="L1053"></a><tt class="py-lineno">1053</tt>  <tt class="py-line"> </tt>
<a name="Reference.__init__"></a><div id="Reference.__init__-def"><a name="L1054"></a><tt class="py-lineno">1054</tt> <a class="py-toggle" href="#" id="Reference.__init__-toggle" onclick="return toggle('Reference.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Reference-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">doi</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">desc</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reference.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Reference.__init__-expanded"><a name="L1055"></a><tt class="py-lineno">1055</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize elements of the Reference class object."""</tt> </tt>
<a name="L1056"></a><tt class="py-lineno">1056</tt>  <tt class="py-line">        <tt id="link-401" class="py-name"><a title="Bio.Phylo.PhyloXML._check_str" class="py-name" href="#" onclick="return doclink('link-401', '_check_str', 'link-191');">_check_str</a></tt><tt class="py-op">(</tt><tt class="py-name">doi</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-402" class="py-name"><a title="Bio.Phylo.PhyloXML.Reference.re_doi" class="py-name" href="#" onclick="return doclink('link-402', 're_doi', 'link-400');">re_doi</a></tt><tt class="py-op">.</tt><tt id="link-403" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-403', 'match', 'link-194');">match</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1057"></a><tt class="py-lineno">1057</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">doi</tt> <tt class="py-op">=</tt> <tt class="py-name">doi</tt> </tt>
<a name="L1058"></a><tt class="py-lineno">1058</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-404" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.desc" class="py-name" href="#" onclick="return doclink('link-404', 'desc', 'link-201');">desc</a></tt> <tt class="py-op">=</tt> <tt id="link-405" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.desc" class="py-name" href="#" onclick="return doclink('link-405', 'desc', 'link-201');">desc</a></tt> </tt>
</div></div><a name="L1059"></a><tt class="py-lineno">1059</tt>  <tt class="py-line"> </tt>
<a name="Sequence"></a><div id="Sequence-def"><a name="L1060"></a><tt class="py-lineno">1060</tt>  <tt class="py-line"> </tt>
<a name="L1061"></a><tt class="py-lineno">1061</tt> <a class="py-toggle" href="#" id="Sequence-toggle" onclick="return toggle('Sequence');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Sequence-class.html">Sequence</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Sequence-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Sequence-expanded"><a name="L1062"></a><tt class="py-lineno">1062</tt>  <tt class="py-line">    <tt class="py-docstring">"""A molecular sequence (Protein, DNA, RNA) associated with a node.</tt> </tt>
<a name="L1063"></a><tt class="py-lineno">1063</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1064"></a><tt class="py-lineno">1064</tt>  <tt class="py-line"><tt class="py-docstring">    One intended use for ``id_ref`` is to link a sequence to a taxonomy (via the</tt> </tt>
<a name="L1065"></a><tt class="py-lineno">1065</tt>  <tt class="py-line"><tt class="py-docstring">    taxonomy's ``id_source``) in case of multiple sequences and taxonomies per</tt> </tt>
<a name="L1066"></a><tt class="py-lineno">1066</tt>  <tt class="py-line"><tt class="py-docstring">    node.</tt> </tt>
<a name="L1067"></a><tt class="py-lineno">1067</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1068"></a><tt class="py-lineno">1068</tt>  <tt class="py-line"><tt class="py-docstring">    :Parameters:</tt> </tt>
<a name="L1069"></a><tt class="py-lineno">1069</tt>  <tt class="py-line"><tt class="py-docstring">        type : {'dna', 'rna', 'protein'}</tt> </tt>
<a name="L1070"></a><tt class="py-lineno">1070</tt>  <tt class="py-line"><tt class="py-docstring">            type of molecule this sequence represents</tt> </tt>
<a name="L1071"></a><tt class="py-lineno">1071</tt>  <tt class="py-line"><tt class="py-docstring">        id_ref : string</tt> </tt>
<a name="L1072"></a><tt class="py-lineno">1072</tt>  <tt class="py-line"><tt class="py-docstring">            reference to another resource</tt> </tt>
<a name="L1073"></a><tt class="py-lineno">1073</tt>  <tt class="py-line"><tt class="py-docstring">        id_source : string</tt> </tt>
<a name="L1074"></a><tt class="py-lineno">1074</tt>  <tt class="py-line"><tt class="py-docstring">            source for the reference</tt> </tt>
<a name="L1075"></a><tt class="py-lineno">1075</tt>  <tt class="py-line"><tt class="py-docstring">        symbol : string</tt> </tt>
<a name="L1076"></a><tt class="py-lineno">1076</tt>  <tt class="py-line"><tt class="py-docstring">            short symbol of the sequence, e.g. 'ACTM' (max. 10 chars)</tt> </tt>
<a name="L1077"></a><tt class="py-lineno">1077</tt>  <tt class="py-line"><tt class="py-docstring">        accession : Accession</tt> </tt>
<a name="L1078"></a><tt class="py-lineno">1078</tt>  <tt class="py-line"><tt class="py-docstring">            accession code for this sequence.</tt> </tt>
<a name="L1079"></a><tt class="py-lineno">1079</tt>  <tt class="py-line"><tt class="py-docstring">        name : string</tt> </tt>
<a name="L1080"></a><tt class="py-lineno">1080</tt>  <tt class="py-line"><tt class="py-docstring">            full name of the sequence, e.g. 'muscle Actin'</tt> </tt>
<a name="L1081"></a><tt class="py-lineno">1081</tt>  <tt class="py-line"><tt class="py-docstring">        location</tt> </tt>
<a name="L1082"></a><tt class="py-lineno">1082</tt>  <tt class="py-line"><tt class="py-docstring">            location of a sequence on a genome/chromosome.</tt> </tt>
<a name="L1083"></a><tt class="py-lineno">1083</tt>  <tt class="py-line"><tt class="py-docstring">        mol_seq : MolSeq</tt> </tt>
<a name="L1084"></a><tt class="py-lineno">1084</tt>  <tt class="py-line"><tt class="py-docstring">            the molecular sequence itself</tt> </tt>
<a name="L1085"></a><tt class="py-lineno">1085</tt>  <tt class="py-line"><tt class="py-docstring">        uri : Uri</tt> </tt>
<a name="L1086"></a><tt class="py-lineno">1086</tt>  <tt class="py-line"><tt class="py-docstring">            link</tt> </tt>
<a name="L1087"></a><tt class="py-lineno">1087</tt>  <tt class="py-line"><tt class="py-docstring">        annotations : list of Annotation objects</tt> </tt>
<a name="L1088"></a><tt class="py-lineno">1088</tt>  <tt class="py-line"><tt class="py-docstring">            annotations on this sequence</tt> </tt>
<a name="L1089"></a><tt class="py-lineno">1089</tt>  <tt class="py-line"><tt class="py-docstring">        domain_architecture : DomainArchitecture</tt> </tt>
<a name="L1090"></a><tt class="py-lineno">1090</tt>  <tt class="py-line"><tt class="py-docstring">            protein domains on this sequence</tt> </tt>
<a name="L1091"></a><tt class="py-lineno">1091</tt>  <tt class="py-line"><tt class="py-docstring">        other : list of Other objects</tt> </tt>
<a name="L1092"></a><tt class="py-lineno">1092</tt>  <tt class="py-line"><tt class="py-docstring">            non-phyloXML elements</tt> </tt>
<a name="L1093"></a><tt class="py-lineno">1093</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1094"></a><tt class="py-lineno">1094</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1095"></a><tt class="py-lineno">1095</tt>  <tt class="py-line"> </tt>
<a name="L1096"></a><tt class="py-lineno">1096</tt>  <tt class="py-line">    <tt id="link-406" class="py-name" targets="Variable Bio.Phylo.PhyloXML.Sequence.alphabets=Bio.Phylo.PhyloXML.Sequence-class.html#alphabets"><a title="Bio.Phylo.PhyloXML.Sequence.alphabets" class="py-name" href="#" onclick="return doclink('link-406', 'alphabets', 'link-406');">alphabets</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'dna'</tt><tt class="py-op">:</tt> <tt id="link-407" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-407', 'Alphabet', 'link-3');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-408" class="py-name" targets="Variable Bio.Alphabet.generic_dna=Bio.Alphabet-module.html#generic_dna"><a title="Bio.Alphabet.generic_dna" class="py-name" href="#" onclick="return doclink('link-408', 'generic_dna', 'link-408');">generic_dna</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1097"></a><tt class="py-lineno">1097</tt>  <tt class="py-line">                 <tt class="py-string">'rna'</tt><tt class="py-op">:</tt> <tt id="link-409" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-409', 'Alphabet', 'link-3');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-410" class="py-name" targets="Variable Bio.Alphabet.generic_rna=Bio.Alphabet-module.html#generic_rna"><a title="Bio.Alphabet.generic_rna" class="py-name" href="#" onclick="return doclink('link-410', 'generic_rna', 'link-410');">generic_rna</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1098"></a><tt class="py-lineno">1098</tt>  <tt class="py-line">                 <tt class="py-string">'protein'</tt><tt class="py-op">:</tt> <tt id="link-411" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-411', 'Alphabet', 'link-3');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-412" class="py-name" targets="Variable Bio.Alphabet.generic_protein=Bio.Alphabet-module.html#generic_protein"><a title="Bio.Alphabet.generic_protein" class="py-name" href="#" onclick="return doclink('link-412', 'generic_protein', 'link-412');">generic_protein</a></tt><tt class="py-op">}</tt> </tt>
<a name="L1099"></a><tt class="py-lineno">1099</tt>  <tt class="py-line">    <tt id="link-413" class="py-name" targets="Variable Bio.Phylo.PhyloXML.Sequence.re_symbol=Bio.Phylo.PhyloXML.Sequence-class.html#re_symbol"><a title="Bio.Phylo.PhyloXML.Sequence.re_symbol" class="py-name" href="#" onclick="return doclink('link-413', 're_symbol', 'link-413');">re_symbol</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'\S{1,10}'</tt><tt class="py-op">)</tt> </tt>
<a name="L1100"></a><tt class="py-lineno">1100</tt>  <tt class="py-line"> </tt>
<a name="Sequence.__init__"></a><div id="Sequence.__init__-def"><a name="L1101"></a><tt class="py-lineno">1101</tt> <a class="py-toggle" href="#" id="Sequence.__init__-toggle" onclick="return toggle('Sequence.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Sequence-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> </tt>
<a name="L1102"></a><tt class="py-lineno">1102</tt>  <tt class="py-line">                 <tt class="py-comment"># Attributes</tt> </tt>
<a name="L1103"></a><tt class="py-lineno">1103</tt>  <tt class="py-line">                 <tt class="py-param">type</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">id_ref</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">id_source</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L1104"></a><tt class="py-lineno">1104</tt>  <tt class="py-line">                 <tt class="py-comment"># Child nodes</tt> </tt>
<a name="L1105"></a><tt class="py-lineno">1105</tt>  <tt class="py-line">                 <tt class="py-param">symbol</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">accession</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">location</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L1106"></a><tt class="py-lineno">1106</tt>  <tt class="py-line">                 <tt class="py-param">mol_seq</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">uri</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">domain_architecture</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L1107"></a><tt class="py-lineno">1107</tt>  <tt class="py-line">                 <tt class="py-comment"># Collections</tt> </tt>
<a name="L1108"></a><tt class="py-lineno">1108</tt>  <tt class="py-line">                 <tt class="py-param">annotations</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L1109"></a><tt class="py-lineno">1109</tt>  <tt class="py-line">                 <tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Sequence.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Sequence.__init__-expanded"><a name="L1110"></a><tt class="py-lineno">1110</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize value for a Sequence object."""</tt> </tt>
<a name="L1111"></a><tt class="py-lineno">1111</tt>  <tt class="py-line">        <tt id="link-414" class="py-name"><a title="Bio.Phylo.PhyloXML._check_str" class="py-name" href="#" onclick="return doclink('link-414', '_check_str', 'link-191');">_check_str</a></tt><tt class="py-op">(</tt><tt id="link-415" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-415', 'type', 'link-29');">type</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-416" class="py-name"><a title="Bio.Phylo.PhyloXML.Sequence.alphabets" class="py-name" href="#" onclick="return doclink('link-416', 'alphabets', 'link-406');">alphabets</a></tt><tt class="py-op">.</tt><tt id="link-417" class="py-name"><a title="Bio.Crystal.Chain.__contains__
Bio.Crystal.Crystal.__contains__
Bio.Data.CodonTable.AmbiguousForwardTable.__contains__
Bio.File._IndexedSeqFileDict.__contains__
Bio.File._SQLiteManySeqFilesDict.__contains__
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__contains__
Bio.PDB.Chain.Chain.__contains__
Bio.PDB.Entity.DisorderedEntityWrapper.__contains__
Bio.PDB.Entity.Entity.__contains__
Bio.PDB.FragmentMapper'.FragmentMapper.__contains__
Bio.PDB.vectors.Vector.__contains__
Bio.Phylo.PhyloXML.Events.__contains__
Bio.Restriction.Restriction.RestrictionBatch.__contains__
Bio.SearchIO._model.hit.Hit.__contains__
Bio.SearchIO._model.hsp.HSP.__contains__
Bio.SearchIO._model.query.QueryResult.__contains__
Bio.Seq.Seq.__contains__
Bio.SeqFeature.CompoundLocation.__contains__
Bio.SeqFeature.FeatureLocation.__contains__
Bio.SeqFeature.SeqFeature.__contains__
Bio.SeqRecord.SeqRecord.__contains__
Bio.phenotype.phen_micro.PlateRecord.__contains__
BioSQL.BioSeqDatabase.BioSeqDatabase.__contains__
BioSQL.BioSeqDatabase.DBServer.__contains__" class="py-name" href="#" onclick="return doclink('link-417', '__contains__', 'link-317');">__contains__</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1112"></a><tt class="py-lineno">1112</tt>  <tt class="py-line">        <tt id="link-418" class="py-name"><a title="Bio.Phylo.PhyloXML._check_str" class="py-name" href="#" onclick="return doclink('link-418', '_check_str', 'link-191');">_check_str</a></tt><tt class="py-op">(</tt><tt id="link-419" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.symbol()=Bio.Phylo.PhyloXMLIO.Writer-class.html#symbol"><a title="Bio.Phylo.PhyloXMLIO.Writer.symbol" class="py-name" href="#" onclick="return doclink('link-419', 'symbol', 'link-419');">symbol</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-420" class="py-name"><a title="Bio.Phylo.PhyloXML.Sequence.re_symbol" class="py-name" href="#" onclick="return doclink('link-420', 're_symbol', 'link-413');">re_symbol</a></tt><tt class="py-op">.</tt><tt id="link-421" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-421', 'match', 'link-194');">match</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1113"></a><tt class="py-lineno">1113</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-422" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-422', 'type', 'link-29');">type</a></tt> <tt class="py-op">=</tt> <tt id="link-423" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-423', 'type', 'link-29');">type</a></tt> </tt>
<a name="L1114"></a><tt class="py-lineno">1114</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">id_ref</tt> <tt class="py-op">=</tt> <tt class="py-name">id_ref</tt> </tt>
<a name="L1115"></a><tt class="py-lineno">1115</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">id_source</tt> <tt class="py-op">=</tt> <tt class="py-name">id_source</tt> </tt>
<a name="L1116"></a><tt class="py-lineno">1116</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-424" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.symbol" class="py-name" href="#" onclick="return doclink('link-424', 'symbol', 'link-419');">symbol</a></tt> <tt class="py-op">=</tt> <tt id="link-425" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.symbol" class="py-name" href="#" onclick="return doclink('link-425', 'symbol', 'link-419');">symbol</a></tt> </tt>
<a name="L1117"></a><tt class="py-lineno">1117</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-426" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.accession()=Bio.GenBank._FeatureConsumer-class.html#accession,Method Bio.GenBank._RecordConsumer.accession()=Bio.GenBank._RecordConsumer-class.html#accession,Method Bio.Phylo.PhyloXMLIO.Parser.accession()=Bio.Phylo.PhyloXMLIO.Parser-class.html#accession,Method Bio.Phylo.PhyloXMLIO.Writer.accession()=Bio.Phylo.PhyloXMLIO.Writer-class.html#accession"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-426', 'accession', 'link-426');">accession</a></tt> <tt class="py-op">=</tt> <tt id="link-427" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-427', 'accession', 'link-426');">accession</a></tt> </tt>
<a name="L1118"></a><tt class="py-lineno">1118</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-428" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-428', 'name', 'link-31');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-429" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-429', 'name', 'link-31');">name</a></tt> </tt>
<a name="L1119"></a><tt class="py-lineno">1119</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-430" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-430', 'location', 'link-386');">location</a></tt> <tt class="py-op">=</tt> <tt id="link-431" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-431', 'location', 'link-386');">location</a></tt> </tt>
<a name="L1120"></a><tt class="py-lineno">1120</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-432" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.mol_seq
Bio.Phylo.PhyloXMLIO.Writer.mol_seq" class="py-name" href="#" onclick="return doclink('link-432', 'mol_seq', 'link-73');">mol_seq</a></tt> <tt class="py-op">=</tt> <tt id="link-433" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.mol_seq
Bio.Phylo.PhyloXMLIO.Writer.mol_seq" class="py-name" href="#" onclick="return doclink('link-433', 'mol_seq', 'link-73');">mol_seq</a></tt> </tt>
<a name="L1121"></a><tt class="py-lineno">1121</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-434" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-434', 'uri', 'link-205');">uri</a></tt> <tt class="py-op">=</tt> <tt id="link-435" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-435', 'uri', 'link-205');">uri</a></tt> </tt>
<a name="L1122"></a><tt class="py-lineno">1122</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-436" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.domain_architecture()=Bio.Phylo.PhyloXMLIO.Parser-class.html#domain_architecture,Method Bio.Phylo.PhyloXMLIO.Writer.domain_architecture()=Bio.Phylo.PhyloXMLIO.Writer-class.html#domain_architecture"><a title="Bio.Phylo.PhyloXMLIO.Parser.domain_architecture
Bio.Phylo.PhyloXMLIO.Writer.domain_architecture" class="py-name" href="#" onclick="return doclink('link-436', 'domain_architecture', 'link-436');">domain_architecture</a></tt> <tt class="py-op">=</tt> <tt id="link-437" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.domain_architecture
Bio.Phylo.PhyloXMLIO.Writer.domain_architecture" class="py-name" href="#" onclick="return doclink('link-437', 'domain_architecture', 'link-436');">domain_architecture</a></tt> </tt>
<a name="L1123"></a><tt class="py-lineno">1123</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-438" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.annotations=BioSQL.BioSeq.DBSeqRecord-class.html#annotations"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-438', 'annotations', 'link-438');">annotations</a></tt> <tt class="py-op">=</tt> <tt id="link-439" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-439', 'annotations', 'link-438');">annotations</a></tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1124"></a><tt class="py-lineno">1124</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-440" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-440', 'other', 'link-24');">other</a></tt> <tt class="py-op">=</tt> <tt id="link-441" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-441', 'other', 'link-24');">other</a></tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div><a name="L1125"></a><tt class="py-lineno">1125</tt>  <tt class="py-line"> </tt>
<a name="L1126"></a><tt class="py-lineno">1126</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">classmethod</tt> </tt>
<a name="Sequence.from_seqrecord"></a><div id="Sequence.from_seqrecord-def"><a name="L1127"></a><tt class="py-lineno">1127</tt> <a class="py-toggle" href="#" id="Sequence.from_seqrecord-toggle" onclick="return toggle('Sequence.from_seqrecord');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Sequence-class.html#from_seqrecord">from_seqrecord</a><tt class="py-op">(</tt><tt class="py-param">cls</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">,</tt> <tt class="py-param">is_aligned</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Sequence.from_seqrecord-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Sequence.from_seqrecord-expanded"><a name="L1128"></a><tt class="py-lineno">1128</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a new PhyloXML Sequence from a SeqRecord object."""</tt> </tt>
<a name="L1129"></a><tt class="py-lineno">1129</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">is_aligned</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1130"></a><tt class="py-lineno">1130</tt>  <tt class="py-line">            <tt class="py-name">is_aligned</tt> <tt class="py-op">=</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-442" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-442', 'seq', 'link-81');">seq</a></tt><tt class="py-op">.</tt><tt id="link-443" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.alphabet=Bio.SearchIO._model.hsp.HSP-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSPFragment.alphabet=Bio.SearchIO._model.hsp.HSPFragment-class.html#alphabet"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-443', 'alphabet', 'link-443');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-444" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-444', 'Alphabet', 'link-3');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-445" class="py-name" targets="Class Bio.Alphabet.Gapped=Bio.Alphabet.Gapped-class.html"><a title="Bio.Alphabet.Gapped" class="py-name" href="#" onclick="return doclink('link-445', 'Gapped', 'link-445');">Gapped</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1131"></a><tt class="py-lineno">1131</tt>  <tt class="py-line">        <tt class="py-name">params</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L1132"></a><tt class="py-lineno">1132</tt>  <tt class="py-line">            <tt class="py-string">'accession'</tt><tt class="py-op">:</tt> <tt id="link-446" class="py-name" targets="Class Bio.Phylo.PhyloXML.Accession=Bio.Phylo.PhyloXML.Accession-class.html"><a title="Bio.Phylo.PhyloXML.Accession" class="py-name" href="#" onclick="return doclink('link-446', 'Accession', 'link-446');">Accession</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-447" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-447', 'id', 'link-48');">id</a></tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1133"></a><tt class="py-lineno">1133</tt>  <tt class="py-line">            <tt class="py-string">'symbol'</tt><tt class="py-op">:</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-448" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-448', 'name', 'link-31');">name</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1134"></a><tt class="py-lineno">1134</tt>  <tt class="py-line">            <tt class="py-string">'name'</tt><tt class="py-op">:</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-449" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-449', 'description', 'link-50');">description</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1135"></a><tt class="py-lineno">1135</tt>  <tt class="py-line">            <tt class="py-string">'mol_seq'</tt><tt class="py-op">:</tt> <tt id="link-450" class="py-name" targets="Class Bio.Phylo.PhyloXML.MolSeq=Bio.Phylo.PhyloXML.MolSeq-class.html"><a title="Bio.Phylo.PhyloXML.MolSeq" class="py-name" href="#" onclick="return doclink('link-450', 'MolSeq', 'link-450');">MolSeq</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-451" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-451', 'seq', 'link-81');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">is_aligned</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1136"></a><tt class="py-lineno">1136</tt>  <tt class="py-line">        <tt class="py-op">}</tt> </tt>
<a name="L1137"></a><tt class="py-lineno">1137</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-452" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-452', 'seq', 'link-81');">seq</a></tt><tt class="py-op">.</tt><tt id="link-453" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-453', 'alphabet', 'link-443');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-454" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-454', 'Alphabet', 'link-3');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-455" class="py-name" targets="Class Bio.Alphabet.DNAAlphabet=Bio.Alphabet.DNAAlphabet-class.html"><a title="Bio.Alphabet.DNAAlphabet" class="py-name" href="#" onclick="return doclink('link-455', 'DNAAlphabet', 'link-455');">DNAAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1138"></a><tt class="py-lineno">1138</tt>  <tt class="py-line">            <tt class="py-name">params</tt><tt class="py-op">[</tt><tt class="py-string">'type'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">'dna'</tt> </tt>
<a name="L1139"></a><tt class="py-lineno">1139</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-456" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-456', 'seq', 'link-81');">seq</a></tt><tt class="py-op">.</tt><tt id="link-457" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-457', 'alphabet', 'link-443');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-458" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-458', 'Alphabet', 'link-3');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-459" class="py-name" targets="Class Bio.Alphabet.RNAAlphabet=Bio.Alphabet.RNAAlphabet-class.html"><a title="Bio.Alphabet.RNAAlphabet" class="py-name" href="#" onclick="return doclink('link-459', 'RNAAlphabet', 'link-459');">RNAAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1140"></a><tt class="py-lineno">1140</tt>  <tt class="py-line">            <tt class="py-name">params</tt><tt class="py-op">[</tt><tt class="py-string">'type'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">'rna'</tt> </tt>
<a name="L1141"></a><tt class="py-lineno">1141</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-460" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-460', 'seq', 'link-81');">seq</a></tt><tt class="py-op">.</tt><tt id="link-461" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-461', 'alphabet', 'link-443');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-462" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-462', 'Alphabet', 'link-3');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-463" class="py-name" targets="Class Bio.Alphabet.ProteinAlphabet=Bio.Alphabet.ProteinAlphabet-class.html"><a title="Bio.Alphabet.ProteinAlphabet" class="py-name" href="#" onclick="return doclink('link-463', 'ProteinAlphabet', 'link-463');">ProteinAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1142"></a><tt class="py-lineno">1142</tt>  <tt class="py-line">            <tt class="py-name">params</tt><tt class="py-op">[</tt><tt class="py-string">'type'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">'protein'</tt> </tt>
<a name="L1143"></a><tt class="py-lineno">1143</tt>  <tt class="py-line"> </tt>
<a name="L1144"></a><tt class="py-lineno">1144</tt>  <tt class="py-line">        <tt class="py-comment"># Unpack record.annotations</tt> </tt>
<a name="L1145"></a><tt class="py-lineno">1145</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'id_ref'</tt><tt class="py-op">,</tt> <tt class="py-string">'id_source'</tt><tt class="py-op">,</tt> <tt class="py-string">'location'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1146"></a><tt class="py-lineno">1146</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-464" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-464', 'annotations', 'link-438');">annotations</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1147"></a><tt class="py-lineno">1147</tt>  <tt class="py-line">                <tt class="py-name">params</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-465" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-465', 'annotations', 'link-438');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> </tt>
<a name="L1148"></a><tt class="py-lineno">1148</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-466" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-466', 'annotations', 'link-438');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-467" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-467', 'get', 'link-391');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'uri'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">dict</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1149"></a><tt class="py-lineno">1149</tt>  <tt class="py-line">            <tt class="py-name">params</tt><tt class="py-op">[</tt><tt class="py-string">'uri'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-468" class="py-name" targets="Class Bio.Phylo.PhyloXML.Uri=Bio.Phylo.PhyloXML.Uri-class.html"><a title="Bio.Phylo.PhyloXML.Uri" class="py-name" href="#" onclick="return doclink('link-468', 'Uri', 'link-468');">Uri</a></tt><tt class="py-op">(</tt><tt class="py-op">**</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-469" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-469', 'annotations', 'link-438');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'uri'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1150"></a><tt class="py-lineno">1150</tt>  <tt class="py-line">        <tt class="py-comment"># Build a Sequence.annotation object</tt> </tt>
<a name="L1151"></a><tt class="py-lineno">1151</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-470" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-470', 'annotations', 'link-438');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-471" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-471', 'get', 'link-391');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'annotations'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1152"></a><tt class="py-lineno">1152</tt>  <tt class="py-line">            <tt class="py-name">params</tt><tt class="py-op">[</tt><tt class="py-string">'annotations'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1153"></a><tt class="py-lineno">1153</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">annot</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-472" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-472', 'annotations', 'link-438');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'annotations'</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L1154"></a><tt class="py-lineno">1154</tt>  <tt class="py-line">                <tt class="py-name">ann_args</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L1155"></a><tt class="py-lineno">1155</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'ref'</tt><tt class="py-op">,</tt> <tt class="py-string">'source'</tt><tt class="py-op">,</tt> <tt class="py-string">'evidence'</tt><tt class="py-op">,</tt> <tt class="py-string">'type'</tt><tt class="py-op">,</tt> <tt class="py-string">'desc'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1156"></a><tt class="py-lineno">1156</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">annot</tt><tt class="py-op">:</tt> </tt>
<a name="L1157"></a><tt class="py-lineno">1157</tt>  <tt class="py-line">                        <tt class="py-name">ann_args</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">annot</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> </tt>
<a name="L1158"></a><tt class="py-lineno">1158</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">annot</tt><tt class="py-op">.</tt><tt id="link-473" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-473', 'get', 'link-391');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'confidence'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1159"></a><tt class="py-lineno">1159</tt>  <tt class="py-line">                    <tt class="py-name">ann_args</tt><tt class="py-op">[</tt><tt class="py-string">'confidence'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-474" class="py-name"><a title="Bio.Phylo.PhyloXML.Confidence" class="py-name" href="#" onclick="return doclink('link-474', 'Confidence', 'link-87');">Confidence</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1160"></a><tt class="py-lineno">1160</tt>  <tt class="py-line">                        <tt class="py-op">*</tt><tt class="py-name">annot</tt><tt class="py-op">[</tt><tt class="py-string">'confidence'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1161"></a><tt class="py-lineno">1161</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">annot</tt><tt class="py-op">.</tt><tt id="link-475" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-475', 'get', 'link-391');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'properties'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1162"></a><tt class="py-lineno">1162</tt>  <tt class="py-line">                    <tt class="py-name">ann_args</tt><tt class="py-op">[</tt><tt class="py-string">'properties'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-476" class="py-name" targets="Class Bio.Phylo.PhyloXML.Property=Bio.Phylo.PhyloXML.Property-class.html"><a title="Bio.Phylo.PhyloXML.Property" class="py-name" href="#" onclick="return doclink('link-476', 'Property', 'link-476');">Property</a></tt><tt class="py-op">(</tt><tt class="py-op">**</tt><tt class="py-name">prop</tt><tt class="py-op">)</tt> </tt>
<a name="L1163"></a><tt class="py-lineno">1163</tt>  <tt class="py-line">                                              <tt class="py-keyword">for</tt> <tt class="py-name">prop</tt> <tt class="py-keyword">in</tt> <tt class="py-name">annot</tt><tt class="py-op">[</tt><tt class="py-string">'properties'</tt><tt class="py-op">]</tt> </tt>
<a name="L1164"></a><tt class="py-lineno">1164</tt>  <tt class="py-line">                                              <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">prop</tt><tt class="py-op">,</tt> <tt class="py-name">dict</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L1165"></a><tt class="py-lineno">1165</tt>  <tt class="py-line">                <tt class="py-name">params</tt><tt class="py-op">[</tt><tt class="py-string">'annotations'</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-477" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-477', 'append', 'link-91');">append</a></tt><tt class="py-op">(</tt><tt id="link-478" class="py-name" targets="Class Bio.Phylo.PhyloXML.Annotation=Bio.Phylo.PhyloXML.Annotation-class.html"><a title="Bio.Phylo.PhyloXML.Annotation" class="py-name" href="#" onclick="return doclink('link-478', 'Annotation', 'link-478');">Annotation</a></tt><tt class="py-op">(</tt><tt class="py-op">**</tt><tt class="py-name">ann_args</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1166"></a><tt class="py-lineno">1166</tt>  <tt class="py-line"> </tt>
<a name="L1167"></a><tt class="py-lineno">1167</tt>  <tt class="py-line">        <tt class="py-comment"># Unpack record.features</tt> </tt>
<a name="L1168"></a><tt class="py-lineno">1168</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-479" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.features=BioSQL.BioSeq.DBSeqRecord-class.html#features"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-479', 'features', 'link-479');">features</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1169"></a><tt class="py-lineno">1169</tt>  <tt class="py-line">            <tt class="py-name">params</tt><tt class="py-op">[</tt><tt class="py-string">'domain_architecture'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-480" class="py-name" targets="Class Bio.Phylo.PhyloXML.DomainArchitecture=Bio.Phylo.PhyloXML.DomainArchitecture-class.html"><a title="Bio.Phylo.PhyloXML.DomainArchitecture" class="py-name" href="#" onclick="return doclink('link-480', 'DomainArchitecture', 'link-480');">DomainArchitecture</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1170"></a><tt class="py-lineno">1170</tt>  <tt class="py-line">                <tt id="link-481" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-481', 'length', 'link-310');">length</a></tt><tt class="py-op">=</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-482" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-482', 'seq', 'link-81');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1171"></a><tt class="py-lineno">1171</tt>  <tt class="py-line">                <tt class="py-name">domains</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt id="link-483" class="py-name"><a title="Bio.Phylo.PhyloXML.ProteinDomain" class="py-name" href="#" onclick="return doclink('link-483', 'ProteinDomain', 'link-384');">ProteinDomain</a></tt><tt class="py-op">.</tt><tt id="link-484" class="py-name" targets="Class Method Bio.Phylo.PhyloXML.ProteinDomain.from_seqfeature()=Bio.Phylo.PhyloXML.ProteinDomain-class.html#from_seqfeature"><a title="Bio.Phylo.PhyloXML.ProteinDomain.from_seqfeature" class="py-name" href="#" onclick="return doclink('link-484', 'from_seqfeature', 'link-484');">from_seqfeature</a></tt><tt class="py-op">(</tt><tt class="py-name">feat</tt><tt class="py-op">)</tt> </tt>
<a name="L1172"></a><tt class="py-lineno">1172</tt>  <tt class="py-line">                         <tt class="py-keyword">for</tt> <tt class="py-name">feat</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-485" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-485', 'features', 'link-479');">features</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1173"></a><tt class="py-lineno">1173</tt>  <tt class="py-line"> </tt>
<a name="L1174"></a><tt class="py-lineno">1174</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-486" class="py-name"><a title="Bio.Phylo.PhyloXML.Sequence" class="py-name" href="#" onclick="return doclink('link-486', 'Sequence', 'link-72');">Sequence</a></tt><tt class="py-op">(</tt><tt class="py-op">**</tt><tt class="py-name">params</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1175"></a><tt class="py-lineno">1175</tt>  <tt class="py-line"> </tt>
<a name="Sequence.to_seqrecord"></a><div id="Sequence.to_seqrecord-def"><a name="L1176"></a><tt class="py-lineno">1176</tt> <a class="py-toggle" href="#" id="Sequence.to_seqrecord-toggle" onclick="return toggle('Sequence.to_seqrecord');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Sequence-class.html#to_seqrecord">to_seqrecord</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Sequence.to_seqrecord-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Sequence.to_seqrecord-expanded"><a name="L1177"></a><tt class="py-lineno">1177</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a SeqRecord object from this Sequence instance.</tt> </tt>
<a name="L1178"></a><tt class="py-lineno">1178</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1179"></a><tt class="py-lineno">1179</tt>  <tt class="py-line"><tt class="py-docstring">        The seqrecord.annotations dictionary is packed like so::</tt> </tt>
<a name="L1180"></a><tt class="py-lineno">1180</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1181"></a><tt class="py-lineno">1181</tt>  <tt class="py-line"><tt class="py-docstring">            { # Sequence attributes with no SeqRecord equivalent:</tt> </tt>
<a name="L1182"></a><tt class="py-lineno">1182</tt>  <tt class="py-line"><tt class="py-docstring">              'id_ref': self.id_ref,</tt> </tt>
<a name="L1183"></a><tt class="py-lineno">1183</tt>  <tt class="py-line"><tt class="py-docstring">              'id_source': self.id_source,</tt> </tt>
<a name="L1184"></a><tt class="py-lineno">1184</tt>  <tt class="py-line"><tt class="py-docstring">              'location': self.location,</tt> </tt>
<a name="L1185"></a><tt class="py-lineno">1185</tt>  <tt class="py-line"><tt class="py-docstring">              'uri': { 'value': self.uri.value,</tt> </tt>
<a name="L1186"></a><tt class="py-lineno">1186</tt>  <tt class="py-line"><tt class="py-docstring">                              'desc': self.uri.desc,</tt> </tt>
<a name="L1187"></a><tt class="py-lineno">1187</tt>  <tt class="py-line"><tt class="py-docstring">                              'type': self.uri.type },</tt> </tt>
<a name="L1188"></a><tt class="py-lineno">1188</tt>  <tt class="py-line"><tt class="py-docstring">              # Sequence.annotations attribute (list of Annotations)</tt> </tt>
<a name="L1189"></a><tt class="py-lineno">1189</tt>  <tt class="py-line"><tt class="py-docstring">              'annotations': [{'ref': ann.ref,</tt> </tt>
<a name="L1190"></a><tt class="py-lineno">1190</tt>  <tt class="py-line"><tt class="py-docstring">                               'source': ann.source,</tt> </tt>
<a name="L1191"></a><tt class="py-lineno">1191</tt>  <tt class="py-line"><tt class="py-docstring">                               'evidence': ann.evidence,</tt> </tt>
<a name="L1192"></a><tt class="py-lineno">1192</tt>  <tt class="py-line"><tt class="py-docstring">                               'type': ann.type,</tt> </tt>
<a name="L1193"></a><tt class="py-lineno">1193</tt>  <tt class="py-line"><tt class="py-docstring">                               'confidence': [ann.confidence.value,</tt> </tt>
<a name="L1194"></a><tt class="py-lineno">1194</tt>  <tt class="py-line"><tt class="py-docstring">                                              ann.confidence.type],</tt> </tt>
<a name="L1195"></a><tt class="py-lineno">1195</tt>  <tt class="py-line"><tt class="py-docstring">                               'properties': [{'value': prop.value,</tt> </tt>
<a name="L1196"></a><tt class="py-lineno">1196</tt>  <tt class="py-line"><tt class="py-docstring">                                                'ref': prop.ref,</tt> </tt>
<a name="L1197"></a><tt class="py-lineno">1197</tt>  <tt class="py-line"><tt class="py-docstring">                                                'applies_to': prop.applies_to,</tt> </tt>
<a name="L1198"></a><tt class="py-lineno">1198</tt>  <tt class="py-line"><tt class="py-docstring">                                                'datatype': prop.datatype,</tt> </tt>
<a name="L1199"></a><tt class="py-lineno">1199</tt>  <tt class="py-line"><tt class="py-docstring">                                                'unit': prop.unit,</tt> </tt>
<a name="L1200"></a><tt class="py-lineno">1200</tt>  <tt class="py-line"><tt class="py-docstring">                                                'id_ref': prop.id_ref}</tt> </tt>
<a name="L1201"></a><tt class="py-lineno">1201</tt>  <tt class="py-line"><tt class="py-docstring">                                               for prop in ann.properties],</tt> </tt>
<a name="L1202"></a><tt class="py-lineno">1202</tt>  <tt class="py-line"><tt class="py-docstring">                              } for ann in self.annotations],</tt> </tt>
<a name="L1203"></a><tt class="py-lineno">1203</tt>  <tt class="py-line"><tt class="py-docstring">            }</tt> </tt>
<a name="L1204"></a><tt class="py-lineno">1204</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1205"></a><tt class="py-lineno">1205</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1206"></a><tt class="py-lineno">1206</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">clean_dict</tt><tt class="py-op">(</tt><tt class="py-param">dct</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1207"></a><tt class="py-lineno">1207</tt>  <tt class="py-line">            <tt class="py-docstring">"""Remove None-valued items from a dictionary."""</tt> </tt>
<a name="L1208"></a><tt class="py-lineno">1208</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">val</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">val</tt> <tt class="py-keyword">in</tt> <tt class="py-name">dct</tt><tt class="py-op">.</tt><tt id="link-487" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-487', 'items', 'link-328');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1209"></a><tt class="py-lineno">1209</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">val</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1210"></a><tt class="py-lineno">1210</tt>  <tt class="py-line"> </tt>
<a name="L1211"></a><tt class="py-lineno">1211</tt>  <tt class="py-line">        <tt class="py-name">seqrec</tt> <tt class="py-op">=</tt> <tt id="link-488" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-488', 'SeqRecord', 'link-15');">SeqRecord</a></tt><tt class="py-op">(</tt><tt id="link-489" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-489', 'Seq', 'link-8');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-490" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.mol_seq
Bio.Phylo.PhyloXMLIO.Writer.mol_seq" class="py-name" href="#" onclick="return doclink('link-490', 'mol_seq', 'link-73');">mol_seq</a></tt><tt class="py-op">.</tt><tt id="link-491" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-491', 'value', 'link-37');">value</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-492" class="py-name"><a title="Bio.Phylo.PhyloXML.Sequence.get_alphabet" class="py-name" href="#" onclick="return doclink('link-492', 'get_alphabet', 'link-79');">get_alphabet</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1212"></a><tt class="py-lineno">1212</tt>  <tt class="py-line">                           <tt class="py-op">**</tt><tt class="py-name">clean_dict</tt><tt class="py-op">(</tt><tt class="py-op">{</tt> </tt>
<a name="L1213"></a><tt class="py-lineno">1213</tt>  <tt class="py-line">                               <tt class="py-string">'id'</tt><tt class="py-op">:</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-493" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-493', 'accession', 'link-426');">accession</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1214"></a><tt class="py-lineno">1214</tt>  <tt class="py-line">                               <tt class="py-string">'name'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-494" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.symbol" class="py-name" href="#" onclick="return doclink('link-494', 'symbol', 'link-419');">symbol</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1215"></a><tt class="py-lineno">1215</tt>  <tt class="py-line">                               <tt class="py-string">'description'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-495" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-495', 'name', 'link-31');">name</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1216"></a><tt class="py-lineno">1216</tt>  <tt class="py-line">                               <tt class="py-comment"># 'dbxrefs': None,</tt> </tt>
<a name="L1217"></a><tt class="py-lineno">1217</tt>  <tt class="py-line">                           <tt class="py-op">}</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1218"></a><tt class="py-lineno">1218</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-496" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.domain_architecture
Bio.Phylo.PhyloXMLIO.Writer.domain_architecture" class="py-name" href="#" onclick="return doclink('link-496', 'domain_architecture', 'link-436');">domain_architecture</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1219"></a><tt class="py-lineno">1219</tt>  <tt class="py-line">            <tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-497" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-497', 'features', 'link-479');">features</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">dom</tt><tt class="py-op">.</tt><tt id="link-498" class="py-name" targets="Method Bio.Phylo.PhyloXML.ProteinDomain.to_seqfeature()=Bio.Phylo.PhyloXML.ProteinDomain-class.html#to_seqfeature"><a title="Bio.Phylo.PhyloXML.ProteinDomain.to_seqfeature" class="py-name" href="#" onclick="return doclink('link-498', 'to_seqfeature', 'link-498');">to_seqfeature</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1220"></a><tt class="py-lineno">1220</tt>  <tt class="py-line">                               <tt class="py-keyword">for</tt> <tt class="py-name">dom</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-499" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.domain_architecture
Bio.Phylo.PhyloXMLIO.Writer.domain_architecture" class="py-name" href="#" onclick="return doclink('link-499', 'domain_architecture', 'link-436');">domain_architecture</a></tt><tt class="py-op">.</tt><tt class="py-name">domains</tt><tt class="py-op">]</tt> </tt>
<a name="L1221"></a><tt class="py-lineno">1221</tt>  <tt class="py-line">        <tt class="py-comment"># Sequence attributes with no SeqRecord equivalent</tt> </tt>
<a name="L1222"></a><tt class="py-lineno">1222</tt>  <tt class="py-line">        <tt class="py-name">seqrec</tt><tt class="py-op">.</tt><tt id="link-500" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-500', 'annotations', 'link-438');">annotations</a></tt> <tt class="py-op">=</tt> <tt class="py-name">clean_dict</tt><tt class="py-op">(</tt><tt class="py-op">{</tt> </tt>
<a name="L1223"></a><tt class="py-lineno">1223</tt>  <tt class="py-line">            <tt class="py-string">'id_ref'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">id_ref</tt><tt class="py-op">,</tt> </tt>
<a name="L1224"></a><tt class="py-lineno">1224</tt>  <tt class="py-line">            <tt class="py-string">'id_source'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">id_source</tt><tt class="py-op">,</tt> </tt>
<a name="L1225"></a><tt class="py-lineno">1225</tt>  <tt class="py-line">            <tt class="py-string">'location'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-501" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-501', 'location', 'link-386');">location</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1226"></a><tt class="py-lineno">1226</tt>  <tt class="py-line">            <tt class="py-string">'uri'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-502" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-502', 'uri', 'link-205');">uri</a></tt> <tt class="py-keyword">and</tt> <tt class="py-name">clean_dict</tt><tt class="py-op">(</tt><tt class="py-op">{</tt> </tt>
<a name="L1227"></a><tt class="py-lineno">1227</tt>  <tt class="py-line">                <tt class="py-string">'value'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-503" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-503', 'uri', 'link-205');">uri</a></tt><tt class="py-op">.</tt><tt id="link-504" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-504', 'value', 'link-37');">value</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1228"></a><tt class="py-lineno">1228</tt>  <tt class="py-line">                <tt class="py-string">'desc'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-505" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-505', 'uri', 'link-205');">uri</a></tt><tt class="py-op">.</tt><tt id="link-506" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.desc" class="py-name" href="#" onclick="return doclink('link-506', 'desc', 'link-201');">desc</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1229"></a><tt class="py-lineno">1229</tt>  <tt class="py-line">                <tt class="py-string">'type'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-507" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-507', 'uri', 'link-205');">uri</a></tt><tt class="py-op">.</tt><tt id="link-508" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-508', 'type', 'link-29');">type</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1230"></a><tt class="py-lineno">1230</tt>  <tt class="py-line">            <tt class="py-op">}</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1231"></a><tt class="py-lineno">1231</tt>  <tt class="py-line">            <tt class="py-string">'annotations'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-509" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-509', 'annotations', 'link-438');">annotations</a></tt> <tt class="py-keyword">and</tt> <tt class="py-op">[</tt> </tt>
<a name="L1232"></a><tt class="py-lineno">1232</tt>  <tt class="py-line">                <tt class="py-name">clean_dict</tt><tt class="py-op">(</tt><tt class="py-op">{</tt> </tt>
<a name="L1233"></a><tt class="py-lineno">1233</tt>  <tt class="py-line">                    <tt class="py-string">'ref'</tt><tt class="py-op">:</tt> <tt class="py-name">ann</tt><tt class="py-op">.</tt><tt id="link-510" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-510', 'ref', 'link-192');">ref</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1234"></a><tt class="py-lineno">1234</tt>  <tt class="py-line">                    <tt class="py-string">'source'</tt><tt class="py-op">:</tt> <tt class="py-name">ann</tt><tt class="py-op">.</tt><tt id="link-511" class="py-name"><a title="Bio.GenBank._FeatureConsumer.source
Bio.GenBank._RecordConsumer.source
Bio.Pathway.Network.source" class="py-name" href="#" onclick="return doclink('link-511', 'source', 'link-186');">source</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1235"></a><tt class="py-lineno">1235</tt>  <tt class="py-line">                    <tt class="py-string">'evidence'</tt><tt class="py-op">:</tt> <tt class="py-name">ann</tt><tt class="py-op">.</tt><tt class="py-name">evidence</tt><tt class="py-op">,</tt> </tt>
<a name="L1236"></a><tt class="py-lineno">1236</tt>  <tt class="py-line">                    <tt class="py-string">'type'</tt><tt class="py-op">:</tt> <tt class="py-name">ann</tt><tt class="py-op">.</tt><tt id="link-512" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-512', 'type', 'link-29');">type</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1237"></a><tt class="py-lineno">1237</tt>  <tt class="py-line">                    <tt class="py-string">'confidence'</tt><tt class="py-op">:</tt> <tt class="py-name">ann</tt><tt class="py-op">.</tt><tt id="link-513" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-513', 'confidence', 'link-94');">confidence</a></tt> <tt class="py-keyword">and</tt> <tt class="py-op">[</tt> </tt>
<a name="L1238"></a><tt class="py-lineno">1238</tt>  <tt class="py-line">                        <tt class="py-name">ann</tt><tt class="py-op">.</tt><tt id="link-514" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-514', 'confidence', 'link-94');">confidence</a></tt><tt class="py-op">.</tt><tt id="link-515" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-515', 'value', 'link-37');">value</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1239"></a><tt class="py-lineno">1239</tt>  <tt class="py-line">                        <tt class="py-name">ann</tt><tt class="py-op">.</tt><tt id="link-516" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-516', 'confidence', 'link-94');">confidence</a></tt><tt class="py-op">.</tt><tt id="link-517" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-517', 'type', 'link-29');">type</a></tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L1240"></a><tt class="py-lineno">1240</tt>  <tt class="py-line">                    <tt class="py-string">'properties'</tt><tt class="py-op">:</tt> <tt class="py-op">[</tt><tt class="py-name">clean_dict</tt><tt class="py-op">(</tt><tt class="py-op">{</tt> </tt>
<a name="L1241"></a><tt class="py-lineno">1241</tt>  <tt class="py-line">                        <tt class="py-string">'value'</tt><tt class="py-op">:</tt> <tt class="py-name">prop</tt><tt class="py-op">.</tt><tt id="link-518" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-518', 'value', 'link-37');">value</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1242"></a><tt class="py-lineno">1242</tt>  <tt class="py-line">                        <tt class="py-string">'ref'</tt><tt class="py-op">:</tt> <tt class="py-name">prop</tt><tt class="py-op">.</tt><tt id="link-519" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-519', 'ref', 'link-192');">ref</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1243"></a><tt class="py-lineno">1243</tt>  <tt class="py-line">                        <tt class="py-string">'applies_to'</tt><tt class="py-op">:</tt> <tt class="py-name">prop</tt><tt class="py-op">.</tt><tt class="py-name">applies_to</tt><tt class="py-op">,</tt> </tt>
<a name="L1244"></a><tt class="py-lineno">1244</tt>  <tt class="py-line">                        <tt class="py-string">'datatype'</tt><tt class="py-op">:</tt> <tt class="py-name">prop</tt><tt class="py-op">.</tt><tt class="py-name">datatype</tt><tt class="py-op">,</tt> </tt>
<a name="L1245"></a><tt class="py-lineno">1245</tt>  <tt class="py-line">                        <tt class="py-string">'unit'</tt><tt class="py-op">:</tt> <tt class="py-name">prop</tt><tt class="py-op">.</tt><tt class="py-name">unit</tt><tt class="py-op">,</tt> </tt>
<a name="L1246"></a><tt class="py-lineno">1246</tt>  <tt class="py-line">                        <tt class="py-string">'id_ref'</tt><tt class="py-op">:</tt> <tt class="py-name">prop</tt><tt class="py-op">.</tt><tt class="py-name">id_ref</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1247"></a><tt class="py-lineno">1247</tt>  <tt class="py-line">                        <tt class="py-keyword">for</tt> <tt class="py-name">prop</tt> <tt class="py-keyword">in</tt> <tt class="py-name">ann</tt><tt class="py-op">.</tt><tt class="py-name">properties</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L1248"></a><tt class="py-lineno">1248</tt>  <tt class="py-line">                <tt class="py-op">}</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">ann</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-520" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-520', 'annotations', 'link-438');">annotations</a></tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L1249"></a><tt class="py-lineno">1249</tt>  <tt class="py-line">        <tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1250"></a><tt class="py-lineno">1250</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">seqrec</tt> </tt>
</div><a name="L1251"></a><tt class="py-lineno">1251</tt>  <tt class="py-line"> </tt>
<a name="Sequence.get_alphabet"></a><div id="Sequence.get_alphabet-def"><a name="L1252"></a><tt class="py-lineno">1252</tt> <a class="py-toggle" href="#" id="Sequence.get_alphabet-toggle" onclick="return toggle('Sequence.get_alphabet');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Sequence-class.html#get_alphabet">get_alphabet</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Sequence.get_alphabet-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Sequence.get_alphabet-expanded"><a name="L1253"></a><tt class="py-lineno">1253</tt>  <tt class="py-line">        <tt class="py-name">alph</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-521" class="py-name"><a title="Bio.Phylo.PhyloXML.Sequence.alphabets" class="py-name" href="#" onclick="return doclink('link-521', 'alphabets', 'link-406');">alphabets</a></tt><tt class="py-op">.</tt><tt id="link-522" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-522', 'get', 'link-391');">get</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-523" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-523', 'type', 'link-29');">type</a></tt><tt class="py-op">,</tt> <tt id="link-524" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-524', 'Alphabet', 'link-3');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-525" class="py-name" targets="Variable Bio.Alphabet.generic_alphabet=Bio.Alphabet-module.html#generic_alphabet"><a title="Bio.Alphabet.generic_alphabet" class="py-name" href="#" onclick="return doclink('link-525', 'generic_alphabet', 'link-525');">generic_alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1254"></a><tt class="py-lineno">1254</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-526" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.mol_seq
Bio.Phylo.PhyloXMLIO.Writer.mol_seq" class="py-name" href="#" onclick="return doclink('link-526', 'mol_seq', 'link-73');">mol_seq</a></tt> <tt class="py-keyword">and</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-527" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.mol_seq
Bio.Phylo.PhyloXMLIO.Writer.mol_seq" class="py-name" href="#" onclick="return doclink('link-527', 'mol_seq', 'link-73');">mol_seq</a></tt><tt class="py-op">.</tt><tt class="py-name">is_aligned</tt><tt class="py-op">:</tt> </tt>
<a name="L1255"></a><tt class="py-lineno">1255</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-528" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-528', 'Alphabet', 'link-3');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-529" class="py-name"><a title="Bio.Alphabet.Gapped" class="py-name" href="#" onclick="return doclink('link-529', 'Gapped', 'link-445');">Gapped</a></tt><tt class="py-op">(</tt><tt class="py-name">alph</tt><tt class="py-op">)</tt> </tt>
<a name="L1256"></a><tt class="py-lineno">1256</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">alph</tt> </tt>
</div></div><a name="L1257"></a><tt class="py-lineno">1257</tt>  <tt class="py-line"> </tt>
<a name="SequenceRelation"></a><div id="SequenceRelation-def"><a name="L1258"></a><tt class="py-lineno">1258</tt>  <tt class="py-line"> </tt>
<a name="L1259"></a><tt class="py-lineno">1259</tt> <a class="py-toggle" href="#" id="SequenceRelation-toggle" onclick="return toggle('SequenceRelation');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.SequenceRelation-class.html">SequenceRelation</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceRelation-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="SequenceRelation-expanded"><a name="L1260"></a><tt class="py-lineno">1260</tt>  <tt class="py-line">    <tt class="py-docstring">"""Express a typed relationship between two sequences.</tt> </tt>
<a name="L1261"></a><tt class="py-lineno">1261</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1262"></a><tt class="py-lineno">1262</tt>  <tt class="py-line"><tt class="py-docstring">    For example, this could be used to describe an orthology (in which case</tt> </tt>
<a name="L1263"></a><tt class="py-lineno">1263</tt>  <tt class="py-line"><tt class="py-docstring">    attribute 'type' is 'orthology').</tt> </tt>
<a name="L1264"></a><tt class="py-lineno">1264</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1265"></a><tt class="py-lineno">1265</tt>  <tt class="py-line"><tt class="py-docstring">    :Parameters:</tt> </tt>
<a name="L1266"></a><tt class="py-lineno">1266</tt>  <tt class="py-line"><tt class="py-docstring">        id_ref_0 : Id</tt> </tt>
<a name="L1267"></a><tt class="py-lineno">1267</tt>  <tt class="py-line"><tt class="py-docstring">            first sequence reference identifier</tt> </tt>
<a name="L1268"></a><tt class="py-lineno">1268</tt>  <tt class="py-line"><tt class="py-docstring">        id_ref_1 : Id</tt> </tt>
<a name="L1269"></a><tt class="py-lineno">1269</tt>  <tt class="py-line"><tt class="py-docstring">            second sequence reference identifier</tt> </tt>
<a name="L1270"></a><tt class="py-lineno">1270</tt>  <tt class="py-line"><tt class="py-docstring">        distance : float</tt> </tt>
<a name="L1271"></a><tt class="py-lineno">1271</tt>  <tt class="py-line"><tt class="py-docstring">            distance between the two sequences</tt> </tt>
<a name="L1272"></a><tt class="py-lineno">1272</tt>  <tt class="py-line"><tt class="py-docstring">        type : restricted string</tt> </tt>
<a name="L1273"></a><tt class="py-lineno">1273</tt>  <tt class="py-line"><tt class="py-docstring">            describe the type of relationship</tt> </tt>
<a name="L1274"></a><tt class="py-lineno">1274</tt>  <tt class="py-line"><tt class="py-docstring">        confidence : Confidence</tt> </tt>
<a name="L1275"></a><tt class="py-lineno">1275</tt>  <tt class="py-line"><tt class="py-docstring">            confidence value for this relation</tt> </tt>
<a name="L1276"></a><tt class="py-lineno">1276</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1277"></a><tt class="py-lineno">1277</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1278"></a><tt class="py-lineno">1278</tt>  <tt class="py-line"> </tt>
<a name="L1279"></a><tt class="py-lineno">1279</tt>  <tt class="py-line">    <tt id="link-530" class="py-name"><a title="Bio.Phylo.PhyloXML.Events.ok_type
Bio.Phylo.PhyloXML.SequenceRelation.ok_type" class="py-name" href="#" onclick="return doclink('link-530', 'ok_type', 'link-312');">ok_type</a></tt> <tt class="py-op">=</tt> <tt id="link-531" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-531', 'set', 'link-313');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">'orthology'</tt><tt class="py-op">,</tt> <tt class="py-string">'one_to_one_orthology'</tt><tt class="py-op">,</tt> <tt class="py-string">'super_orthology'</tt><tt class="py-op">,</tt> </tt>
<a name="L1280"></a><tt class="py-lineno">1280</tt>  <tt class="py-line">                   <tt class="py-string">'paralogy'</tt><tt class="py-op">,</tt> <tt class="py-string">'ultra_paralogy'</tt><tt class="py-op">,</tt> <tt class="py-string">'xenology'</tt><tt class="py-op">,</tt> <tt class="py-string">'unknown'</tt><tt class="py-op">,</tt> <tt class="py-string">'other'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1281"></a><tt class="py-lineno">1281</tt>  <tt class="py-line"> </tt>
<a name="SequenceRelation.__init__"></a><div id="SequenceRelation.__init__-def"><a name="L1282"></a><tt class="py-lineno">1282</tt> <a class="py-toggle" href="#" id="SequenceRelation.__init__-toggle" onclick="return toggle('SequenceRelation.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.SequenceRelation-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">type</tt><tt class="py-op">,</tt> <tt class="py-param">id_ref_0</tt><tt class="py-op">,</tt> <tt class="py-param">id_ref_1</tt><tt class="py-op">,</tt> </tt>
<a name="L1283"></a><tt class="py-lineno">1283</tt>  <tt class="py-line">                 <tt class="py-param">distance</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">confidence</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceRelation.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceRelation.__init__-expanded"><a name="L1284"></a><tt class="py-lineno">1284</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the class."""</tt> </tt>
<a name="L1285"></a><tt class="py-lineno">1285</tt>  <tt class="py-line">        <tt id="link-532" class="py-name"><a title="Bio.Phylo.PhyloXML._check_str" class="py-name" href="#" onclick="return doclink('link-532', '_check_str', 'link-191');">_check_str</a></tt><tt class="py-op">(</tt><tt id="link-533" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-533', 'type', 'link-29');">type</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-534" class="py-name"><a title="Bio.Phylo.PhyloXML.Events.ok_type
Bio.Phylo.PhyloXML.SequenceRelation.ok_type" class="py-name" href="#" onclick="return doclink('link-534', 'ok_type', 'link-312');">ok_type</a></tt><tt class="py-op">.</tt><tt id="link-535" class="py-name"><a title="Bio.Crystal.Chain.__contains__
Bio.Crystal.Crystal.__contains__
Bio.Data.CodonTable.AmbiguousForwardTable.__contains__
Bio.File._IndexedSeqFileDict.__contains__
Bio.File._SQLiteManySeqFilesDict.__contains__
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__contains__
Bio.PDB.Chain.Chain.__contains__
Bio.PDB.Entity.DisorderedEntityWrapper.__contains__
Bio.PDB.Entity.Entity.__contains__
Bio.PDB.FragmentMapper'.FragmentMapper.__contains__
Bio.PDB.vectors.Vector.__contains__
Bio.Phylo.PhyloXML.Events.__contains__
Bio.Restriction.Restriction.RestrictionBatch.__contains__
Bio.SearchIO._model.hit.Hit.__contains__
Bio.SearchIO._model.hsp.HSP.__contains__
Bio.SearchIO._model.query.QueryResult.__contains__
Bio.Seq.Seq.__contains__
Bio.SeqFeature.CompoundLocation.__contains__
Bio.SeqFeature.FeatureLocation.__contains__
Bio.SeqFeature.SeqFeature.__contains__
Bio.SeqRecord.SeqRecord.__contains__
Bio.phenotype.phen_micro.PlateRecord.__contains__
BioSQL.BioSeqDatabase.BioSeqDatabase.__contains__
BioSQL.BioSeqDatabase.DBServer.__contains__" class="py-name" href="#" onclick="return doclink('link-535', '__contains__', 'link-317');">__contains__</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1286"></a><tt class="py-lineno">1286</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-536" class="py-name"><a title="Bio.Nexus.Trees.Tree.distance
Bio.Phylo.BaseTree.TreeMixin.distance" class="py-name" href="#" onclick="return doclink('link-536', 'distance', 'link-209');">distance</a></tt> <tt class="py-op">=</tt> <tt id="link-537" class="py-name"><a title="Bio.Nexus.Trees.Tree.distance
Bio.Phylo.BaseTree.TreeMixin.distance" class="py-name" href="#" onclick="return doclink('link-537', 'distance', 'link-209');">distance</a></tt> </tt>
<a name="L1287"></a><tt class="py-lineno">1287</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-538" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-538', 'type', 'link-29');">type</a></tt> <tt class="py-op">=</tt> <tt id="link-539" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-539', 'type', 'link-29');">type</a></tt> </tt>
<a name="L1288"></a><tt class="py-lineno">1288</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">id_ref_0</tt> <tt class="py-op">=</tt> <tt class="py-name">id_ref_0</tt> </tt>
<a name="L1289"></a><tt class="py-lineno">1289</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">id_ref_1</tt> <tt class="py-op">=</tt> <tt class="py-name">id_ref_1</tt> </tt>
<a name="L1290"></a><tt class="py-lineno">1290</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-540" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-540', 'confidence', 'link-94');">confidence</a></tt> <tt class="py-op">=</tt> <tt id="link-541" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-541', 'confidence', 'link-94');">confidence</a></tt> </tt>
</div></div><a name="L1291"></a><tt class="py-lineno">1291</tt>  <tt class="py-line"> </tt>
<a name="Taxonomy"></a><div id="Taxonomy-def"><a name="L1292"></a><tt class="py-lineno">1292</tt>  <tt class="py-line"> </tt>
<a name="L1293"></a><tt class="py-lineno">1293</tt> <a class="py-toggle" href="#" id="Taxonomy-toggle" onclick="return toggle('Taxonomy');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Taxonomy-class.html">Taxonomy</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Taxonomy-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Taxonomy-expanded"><a name="L1294"></a><tt class="py-lineno">1294</tt>  <tt class="py-line">    <tt class="py-docstring">"""Describe taxonomic information for a clade.</tt> </tt>
<a name="L1295"></a><tt class="py-lineno">1295</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1296"></a><tt class="py-lineno">1296</tt>  <tt class="py-line"><tt class="py-docstring">    :Parameters:</tt> </tt>
<a name="L1297"></a><tt class="py-lineno">1297</tt>  <tt class="py-line"><tt class="py-docstring">        id_source : Id</tt> </tt>
<a name="L1298"></a><tt class="py-lineno">1298</tt>  <tt class="py-line"><tt class="py-docstring">            link other elements to a taxonomy (on the XML level)</tt> </tt>
<a name="L1299"></a><tt class="py-lineno">1299</tt>  <tt class="py-line"><tt class="py-docstring">        id : Id</tt> </tt>
<a name="L1300"></a><tt class="py-lineno">1300</tt>  <tt class="py-line"><tt class="py-docstring">            unique identifier of a taxon, e.g. Id('6500',</tt> </tt>
<a name="L1301"></a><tt class="py-lineno">1301</tt>  <tt class="py-line"><tt class="py-docstring">            provider='ncbi_taxonomy') for the California sea hare</tt> </tt>
<a name="L1302"></a><tt class="py-lineno">1302</tt>  <tt class="py-line"><tt class="py-docstring">        code : restricted string</tt> </tt>
<a name="L1303"></a><tt class="py-lineno">1303</tt>  <tt class="py-line"><tt class="py-docstring">            store UniProt/Swiss-Prot style organism codes, e.g. 'APLCA' for the</tt> </tt>
<a name="L1304"></a><tt class="py-lineno">1304</tt>  <tt class="py-line"><tt class="py-docstring">            California sea hare 'Aplysia californica'</tt> </tt>
<a name="L1305"></a><tt class="py-lineno">1305</tt>  <tt class="py-line"><tt class="py-docstring">        scientific_name : string</tt> </tt>
<a name="L1306"></a><tt class="py-lineno">1306</tt>  <tt class="py-line"><tt class="py-docstring">            the standard scientific name for this organism, e.g. 'Aplysia</tt> </tt>
<a name="L1307"></a><tt class="py-lineno">1307</tt>  <tt class="py-line"><tt class="py-docstring">            californica' for the California sea hare</tt> </tt>
<a name="L1308"></a><tt class="py-lineno">1308</tt>  <tt class="py-line"><tt class="py-docstring">        authority : string</tt> </tt>
<a name="L1309"></a><tt class="py-lineno">1309</tt>  <tt class="py-line"><tt class="py-docstring">            keep the authority, such as 'J. G. Cooper, 1863', associated with</tt> </tt>
<a name="L1310"></a><tt class="py-lineno">1310</tt>  <tt class="py-line"><tt class="py-docstring">            the 'scientific_name'</tt> </tt>
<a name="L1311"></a><tt class="py-lineno">1311</tt>  <tt class="py-line"><tt class="py-docstring">        common_names : list of strings</tt> </tt>
<a name="L1312"></a><tt class="py-lineno">1312</tt>  <tt class="py-line"><tt class="py-docstring">            common names for this organism</tt> </tt>
<a name="L1313"></a><tt class="py-lineno">1313</tt>  <tt class="py-line"><tt class="py-docstring">        synonyms : list of strings</tt> </tt>
<a name="L1314"></a><tt class="py-lineno">1314</tt>  <tt class="py-line"><tt class="py-docstring">            synonyms for this taxon?</tt> </tt>
<a name="L1315"></a><tt class="py-lineno">1315</tt>  <tt class="py-line"><tt class="py-docstring">        rank : restricted string</tt> </tt>
<a name="L1316"></a><tt class="py-lineno">1316</tt>  <tt class="py-line"><tt class="py-docstring">            taxonomic rank</tt> </tt>
<a name="L1317"></a><tt class="py-lineno">1317</tt>  <tt class="py-line"><tt class="py-docstring">        uri : Uri</tt> </tt>
<a name="L1318"></a><tt class="py-lineno">1318</tt>  <tt class="py-line"><tt class="py-docstring">            link</tt> </tt>
<a name="L1319"></a><tt class="py-lineno">1319</tt>  <tt class="py-line"><tt class="py-docstring">        other : list of Other objects</tt> </tt>
<a name="L1320"></a><tt class="py-lineno">1320</tt>  <tt class="py-line"><tt class="py-docstring">            non-phyloXML elements</tt> </tt>
<a name="L1321"></a><tt class="py-lineno">1321</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1322"></a><tt class="py-lineno">1322</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1323"></a><tt class="py-lineno">1323</tt>  <tt class="py-line"> </tt>
<a name="L1324"></a><tt class="py-lineno">1324</tt>  <tt class="py-line">    <tt id="link-542" class="py-name" targets="Variable Bio.Phylo.PhyloXML.Taxonomy.re_code=Bio.Phylo.PhyloXML.Taxonomy-class.html#re_code"><a title="Bio.Phylo.PhyloXML.Taxonomy.re_code" class="py-name" href="#" onclick="return doclink('link-542', 're_code', 'link-542');">re_code</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'[a-zA-Z0-9_]{2,10}'</tt><tt class="py-op">)</tt> </tt>
<a name="L1325"></a><tt class="py-lineno">1325</tt>  <tt class="py-line">    <tt id="link-543" class="py-name" targets="Variable Bio.Phylo.PhyloXML.Taxonomy.ok_rank=Bio.Phylo.PhyloXML.Taxonomy-class.html#ok_rank"><a title="Bio.Phylo.PhyloXML.Taxonomy.ok_rank" class="py-name" href="#" onclick="return doclink('link-543', 'ok_rank', 'link-543');">ok_rank</a></tt> <tt class="py-op">=</tt> <tt id="link-544" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-544', 'set', 'link-313');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">'domain'</tt><tt class="py-op">,</tt> <tt class="py-string">'kingdom'</tt><tt class="py-op">,</tt> <tt class="py-string">'subkingdom'</tt><tt class="py-op">,</tt> <tt class="py-string">'branch'</tt><tt class="py-op">,</tt> <tt class="py-string">'infrakingdom'</tt><tt class="py-op">,</tt> </tt>
<a name="L1326"></a><tt class="py-lineno">1326</tt>  <tt class="py-line">                   <tt class="py-string">'superphylum'</tt><tt class="py-op">,</tt> <tt class="py-string">'phylum'</tt><tt class="py-op">,</tt> <tt class="py-string">'subphylum'</tt><tt class="py-op">,</tt> <tt class="py-string">'infraphylum'</tt><tt class="py-op">,</tt> <tt class="py-string">'microphylum'</tt><tt class="py-op">,</tt> </tt>
<a name="L1327"></a><tt class="py-lineno">1327</tt>  <tt class="py-line">                   <tt class="py-string">'superdivision'</tt><tt class="py-op">,</tt> <tt class="py-string">'division'</tt><tt class="py-op">,</tt> <tt class="py-string">'subdivision'</tt><tt class="py-op">,</tt> <tt class="py-string">'infradivision'</tt><tt class="py-op">,</tt> </tt>
<a name="L1328"></a><tt class="py-lineno">1328</tt>  <tt class="py-line">                   <tt class="py-string">'superclass'</tt><tt class="py-op">,</tt> <tt class="py-string">'class'</tt><tt class="py-op">,</tt> <tt class="py-string">'subclass'</tt><tt class="py-op">,</tt> <tt class="py-string">'infraclass'</tt><tt class="py-op">,</tt> <tt class="py-string">'superlegion'</tt><tt class="py-op">,</tt> </tt>
<a name="L1329"></a><tt class="py-lineno">1329</tt>  <tt class="py-line">                   <tt class="py-string">'legion'</tt><tt class="py-op">,</tt> <tt class="py-string">'sublegion'</tt><tt class="py-op">,</tt> <tt class="py-string">'infralegion'</tt><tt class="py-op">,</tt> <tt class="py-string">'supercohort'</tt><tt class="py-op">,</tt> <tt class="py-string">'cohort'</tt><tt class="py-op">,</tt> </tt>
<a name="L1330"></a><tt class="py-lineno">1330</tt>  <tt class="py-line">                   <tt class="py-string">'subcohort'</tt><tt class="py-op">,</tt> <tt class="py-string">'infracohort'</tt><tt class="py-op">,</tt> <tt class="py-string">'superorder'</tt><tt class="py-op">,</tt> <tt class="py-string">'order'</tt><tt class="py-op">,</tt> <tt class="py-string">'suborder'</tt><tt class="py-op">,</tt> </tt>
<a name="L1331"></a><tt class="py-lineno">1331</tt>  <tt class="py-line">                   <tt class="py-string">'superfamily'</tt><tt class="py-op">,</tt> <tt class="py-string">'family'</tt><tt class="py-op">,</tt> <tt class="py-string">'subfamily'</tt><tt class="py-op">,</tt> <tt class="py-string">'supertribe'</tt><tt class="py-op">,</tt> <tt class="py-string">'tribe'</tt><tt class="py-op">,</tt> <tt class="py-string">'subtribe'</tt><tt class="py-op">,</tt> </tt>
<a name="L1332"></a><tt class="py-lineno">1332</tt>  <tt class="py-line">                   <tt class="py-string">'infratribe'</tt><tt class="py-op">,</tt> <tt class="py-string">'genus'</tt><tt class="py-op">,</tt> <tt class="py-string">'subgenus'</tt><tt class="py-op">,</tt> <tt class="py-string">'superspecies'</tt><tt class="py-op">,</tt> <tt class="py-string">'species'</tt><tt class="py-op">,</tt> </tt>
<a name="L1333"></a><tt class="py-lineno">1333</tt>  <tt class="py-line">                   <tt class="py-string">'subspecies'</tt><tt class="py-op">,</tt> <tt class="py-string">'variety'</tt><tt class="py-op">,</tt> <tt class="py-string">'subvariety'</tt><tt class="py-op">,</tt> <tt class="py-string">'form'</tt><tt class="py-op">,</tt> <tt class="py-string">'subform'</tt><tt class="py-op">,</tt> <tt class="py-string">'cultivar'</tt><tt class="py-op">,</tt> </tt>
<a name="L1334"></a><tt class="py-lineno">1334</tt>  <tt class="py-line">                   <tt class="py-string">'unknown'</tt><tt class="py-op">,</tt> <tt class="py-string">'other'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1335"></a><tt class="py-lineno">1335</tt>  <tt class="py-line"> </tt>
<a name="Taxonomy.__init__"></a><div id="Taxonomy.__init__-def"><a name="L1336"></a><tt class="py-lineno">1336</tt> <a class="py-toggle" href="#" id="Taxonomy.__init__-toggle" onclick="return toggle('Taxonomy.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Taxonomy-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> </tt>
<a name="L1337"></a><tt class="py-lineno">1337</tt>  <tt class="py-line">                 <tt class="py-comment"># Attributes</tt> </tt>
<a name="L1338"></a><tt class="py-lineno">1338</tt>  <tt class="py-line">                 <tt class="py-param">id_source</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L1339"></a><tt class="py-lineno">1339</tt>  <tt class="py-line">                 <tt class="py-comment"># Child nodes</tt> </tt>
<a name="L1340"></a><tt class="py-lineno">1340</tt>  <tt class="py-line">                 <tt class="py-param">id</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">code</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">scientific_name</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">authority</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L1341"></a><tt class="py-lineno">1341</tt>  <tt class="py-line">                 <tt class="py-param">rank</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">uri</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L1342"></a><tt class="py-lineno">1342</tt>  <tt class="py-line">                 <tt class="py-comment"># Collections</tt> </tt>
<a name="L1343"></a><tt class="py-lineno">1343</tt>  <tt class="py-line">                 <tt class="py-param">common_names</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">synonyms</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L1344"></a><tt class="py-lineno">1344</tt>  <tt class="py-line">                 <tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Taxonomy.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Taxonomy.__init__-expanded"><a name="L1345"></a><tt class="py-lineno">1345</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the class."""</tt> </tt>
<a name="L1346"></a><tt class="py-lineno">1346</tt>  <tt class="py-line">        <tt id="link-545" class="py-name"><a title="Bio.Phylo.PhyloXML._check_str" class="py-name" href="#" onclick="return doclink('link-545', '_check_str', 'link-191');">_check_str</a></tt><tt class="py-op">(</tt><tt id="link-546" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.code()=Bio.Phylo.PhyloXMLIO.Writer-class.html#code"><a title="Bio.Phylo.PhyloXMLIO.Writer.code" class="py-name" href="#" onclick="return doclink('link-546', 'code', 'link-546');">code</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-547" class="py-name"><a title="Bio.Phylo.PhyloXML.Taxonomy.re_code" class="py-name" href="#" onclick="return doclink('link-547', 're_code', 'link-542');">re_code</a></tt><tt class="py-op">.</tt><tt id="link-548" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-548', 'match', 'link-194');">match</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1347"></a><tt class="py-lineno">1347</tt>  <tt class="py-line">        <tt id="link-549" class="py-name"><a title="Bio.Phylo.PhyloXML._check_str" class="py-name" href="#" onclick="return doclink('link-549', '_check_str', 'link-191');">_check_str</a></tt><tt class="py-op">(</tt><tt id="link-550" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.rank()=Bio.Phylo.PhyloXMLIO.Writer-class.html#rank"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-550', 'rank', 'link-550');">rank</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-551" class="py-name"><a title="Bio.Phylo.PhyloXML.Taxonomy.ok_rank" class="py-name" href="#" onclick="return doclink('link-551', 'ok_rank', 'link-543');">ok_rank</a></tt><tt class="py-op">.</tt><tt id="link-552" class="py-name"><a title="Bio.Crystal.Chain.__contains__
Bio.Crystal.Crystal.__contains__
Bio.Data.CodonTable.AmbiguousForwardTable.__contains__
Bio.File._IndexedSeqFileDict.__contains__
Bio.File._SQLiteManySeqFilesDict.__contains__
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__contains__
Bio.PDB.Chain.Chain.__contains__
Bio.PDB.Entity.DisorderedEntityWrapper.__contains__
Bio.PDB.Entity.Entity.__contains__
Bio.PDB.FragmentMapper'.FragmentMapper.__contains__
Bio.PDB.vectors.Vector.__contains__
Bio.Phylo.PhyloXML.Events.__contains__
Bio.Restriction.Restriction.RestrictionBatch.__contains__
Bio.SearchIO._model.hit.Hit.__contains__
Bio.SearchIO._model.hsp.HSP.__contains__
Bio.SearchIO._model.query.QueryResult.__contains__
Bio.Seq.Seq.__contains__
Bio.SeqFeature.CompoundLocation.__contains__
Bio.SeqFeature.FeatureLocation.__contains__
Bio.SeqFeature.SeqFeature.__contains__
Bio.SeqRecord.SeqRecord.__contains__
Bio.phenotype.phen_micro.PlateRecord.__contains__
BioSQL.BioSeqDatabase.BioSeqDatabase.__contains__
BioSQL.BioSeqDatabase.DBServer.__contains__" class="py-name" href="#" onclick="return doclink('link-552', '__contains__', 'link-317');">__contains__</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1348"></a><tt class="py-lineno">1348</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">id_source</tt> <tt class="py-op">=</tt> <tt class="py-name">id_source</tt> </tt>
<a name="L1349"></a><tt class="py-lineno">1349</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-553" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-553', 'id', 'link-48');">id</a></tt> <tt class="py-op">=</tt> <tt id="link-554" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-554', 'id', 'link-48');">id</a></tt> </tt>
<a name="L1350"></a><tt class="py-lineno">1350</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-555" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.code" class="py-name" href="#" onclick="return doclink('link-555', 'code', 'link-546');">code</a></tt> <tt class="py-op">=</tt> <tt id="link-556" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.code" class="py-name" href="#" onclick="return doclink('link-556', 'code', 'link-546');">code</a></tt> </tt>
<a name="L1351"></a><tt class="py-lineno">1351</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-557" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.scientific_name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#scientific_name"><a title="Bio.Phylo.PhyloXMLIO.Writer.scientific_name" class="py-name" href="#" onclick="return doclink('link-557', 'scientific_name', 'link-557');">scientific_name</a></tt> <tt class="py-op">=</tt> <tt id="link-558" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.scientific_name" class="py-name" href="#" onclick="return doclink('link-558', 'scientific_name', 'link-557');">scientific_name</a></tt> </tt>
<a name="L1352"></a><tt class="py-lineno">1352</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">authority</tt> <tt class="py-op">=</tt> <tt class="py-name">authority</tt> </tt>
<a name="L1353"></a><tt class="py-lineno">1353</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-559" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-559', 'rank', 'link-550');">rank</a></tt> <tt class="py-op">=</tt> <tt id="link-560" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-560', 'rank', 'link-550');">rank</a></tt> </tt>
<a name="L1354"></a><tt class="py-lineno">1354</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-561" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-561', 'uri', 'link-205');">uri</a></tt> <tt class="py-op">=</tt> <tt id="link-562" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-562', 'uri', 'link-205');">uri</a></tt> </tt>
<a name="L1355"></a><tt class="py-lineno">1355</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">common_names</tt> <tt class="py-op">=</tt> <tt class="py-name">common_names</tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1356"></a><tt class="py-lineno">1356</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">synonyms</tt> <tt class="py-op">=</tt> <tt class="py-name">synonyms</tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1357"></a><tt class="py-lineno">1357</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-563" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-563', 'other', 'link-24');">other</a></tt> <tt class="py-op">=</tt> <tt id="link-564" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-564', 'other', 'link-24');">other</a></tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div><a name="L1358"></a><tt class="py-lineno">1358</tt>  <tt class="py-line"> </tt>
<a name="Taxonomy.__str__"></a><div id="Taxonomy.__str__-def"><a name="L1359"></a><tt class="py-lineno">1359</tt> <a class="py-toggle" href="#" id="Taxonomy.__str__-toggle" onclick="return toggle('Taxonomy.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Taxonomy-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Taxonomy.__str__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Taxonomy.__str__-expanded"><a name="L1360"></a><tt class="py-lineno">1360</tt>  <tt class="py-line">        <tt class="py-docstring">"""Show the class name and an identifying attribute."""</tt> </tt>
<a name="L1361"></a><tt class="py-lineno">1361</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-565" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.code" class="py-name" href="#" onclick="return doclink('link-565', 'code', 'link-546');">code</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1362"></a><tt class="py-lineno">1362</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-566" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.code" class="py-name" href="#" onclick="return doclink('link-566', 'code', 'link-546');">code</a></tt> </tt>
<a name="L1363"></a><tt class="py-lineno">1363</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-567" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.scientific_name" class="py-name" href="#" onclick="return doclink('link-567', 'scientific_name', 'link-557');">scientific_name</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1364"></a><tt class="py-lineno">1364</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-568" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.scientific_name" class="py-name" href="#" onclick="return doclink('link-568', 'scientific_name', 'link-557');">scientific_name</a></tt> </tt>
<a name="L1365"></a><tt class="py-lineno">1365</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-569" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-569', 'rank', 'link-550');">rank</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1366"></a><tt class="py-lineno">1366</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-570" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-570', 'rank', 'link-550');">rank</a></tt> </tt>
<a name="L1367"></a><tt class="py-lineno">1367</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-571" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-571', 'id', 'link-48');">id</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1368"></a><tt class="py-lineno">1368</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-572" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-572', 'id', 'link-48');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1369"></a><tt class="py-lineno">1369</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt> </tt>
</div></div><a name="L1370"></a><tt class="py-lineno">1370</tt>  <tt class="py-line"> </tt>
<a name="Uri"></a><div id="Uri-def"><a name="L1371"></a><tt class="py-lineno">1371</tt>  <tt class="py-line"> </tt>
<a name="L1372"></a><tt class="py-lineno">1372</tt> <a class="py-toggle" href="#" id="Uri-toggle" onclick="return toggle('Uri');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Uri-class.html">Uri</a><tt class="py-op">(</tt><tt class="py-base-class">PhyloElement</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Uri-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Uri-expanded"><a name="L1373"></a><tt class="py-lineno">1373</tt>  <tt class="py-line">    <tt class="py-docstring">"""A uniform resource identifier.</tt> </tt>
<a name="L1374"></a><tt class="py-lineno">1374</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1375"></a><tt class="py-lineno">1375</tt>  <tt class="py-line"><tt class="py-docstring">    In general, this is expected to be an URL (for example, to link to an image</tt> </tt>
<a name="L1376"></a><tt class="py-lineno">1376</tt>  <tt class="py-line"><tt class="py-docstring">    on a website, in which case the ``type`` attribute might be 'image' and</tt> </tt>
<a name="L1377"></a><tt class="py-lineno">1377</tt>  <tt class="py-line"><tt class="py-docstring">    ``desc`` might be 'image of a California sea hare').</tt> </tt>
<a name="L1378"></a><tt class="py-lineno">1378</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1379"></a><tt class="py-lineno">1379</tt>  <tt class="py-line"> </tt>
<a name="Uri.__init__"></a><div id="Uri.__init__-def"><a name="L1380"></a><tt class="py-lineno">1380</tt> <a class="py-toggle" href="#" id="Uri.__init__-toggle" onclick="return toggle('Uri.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Uri-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">,</tt> <tt class="py-param">desc</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">type</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Uri.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Uri.__init__-expanded"><a name="L1381"></a><tt class="py-lineno">1381</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the class."""</tt> </tt>
<a name="L1382"></a><tt class="py-lineno">1382</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-573" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-573', 'value', 'link-37');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-574" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-574', 'value', 'link-37');">value</a></tt> </tt>
<a name="L1383"></a><tt class="py-lineno">1383</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-575" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.desc" class="py-name" href="#" onclick="return doclink('link-575', 'desc', 'link-201');">desc</a></tt> <tt class="py-op">=</tt> <tt id="link-576" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.desc" class="py-name" href="#" onclick="return doclink('link-576', 'desc', 'link-201');">desc</a></tt> </tt>
<a name="L1384"></a><tt class="py-lineno">1384</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-577" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-577', 'type', 'link-29');">type</a></tt> <tt class="py-op">=</tt> <tt id="link-578" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-578', 'type', 'link-29');">type</a></tt> </tt>
</div><a name="L1385"></a><tt class="py-lineno">1385</tt>  <tt class="py-line"> </tt>
<a name="Uri.__str__"></a><div id="Uri.__str__-def"><a name="L1386"></a><tt class="py-lineno">1386</tt> <a class="py-toggle" href="#" id="Uri.__str__-toggle" onclick="return toggle('Uri.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.PhyloXML.Uri-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Uri.__str__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Uri.__str__-expanded"><a name="L1387"></a><tt class="py-lineno">1387</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-579" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-579', 'value', 'link-37');">value</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1388"></a><tt class="py-lineno">1388</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-580" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-580', 'value', 'link-37');">value</a></tt> </tt>
<a name="L1389"></a><tt class="py-lineno">1389</tt>  <tt class="py-line">        <tt class="py-string">"""Initialize the class."""</tt> </tt>
<a name="L1390"></a><tt class="py-lineno">1390</tt>  <tt class="py-line">        <tt class="py-string">"""Initialize the class."""</tt> </tt>
<a name="L1391"></a><tt class="py-lineno">1391</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1392"></a><tt class="py-lineno">1392</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Fri Jun 22 16:36:10 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
