<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.KEGG.KGML.KGML_pathway</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.KEGG-module.html">Package&nbsp;KEGG</a> ::
        <a href="Bio.KEGG.KGML-module.html">Package&nbsp;KGML</a> ::
        Module&nbsp;KGML_pathway
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.KEGG.KGML.KGML_pathway-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.KEGG.KGML.KGML_pathway-module.html">Module Bio.KEGG.KGML.KGML_pathway</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2013 by Leighton Pritchard.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-docstring">"""Classes to represent a KGML Pathway Map.</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-docstring">The KGML definition is as of release KGML v0.7.1</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-docstring">(http://www.kegg.jp/kegg/xml/docs/)</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-docstring">Classes:</tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-docstring"> - Pathway - Specifies graph information for the pathway map</tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-docstring"> - Relation - Specifies a relationship between two proteins or KOs,</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-docstring">   or protein and compound. There is an implied direction to the</tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-docstring">   relationship in some cases.</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-docstring"> - Reaction - A specific chemical reaction between a substrate and</tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-docstring">   a product.</tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-docstring"> - Entry - A node in the pathway graph</tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-docstring"> - Graphics - Entry subelement describing its visual representation</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">time</tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">itertools</tt> <tt class="py-keyword">import</tt> <tt class="py-name">chain</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">xml</tt><tt class="py-op">.</tt><tt class="py-name">dom</tt> <tt class="py-keyword">import</tt> <tt class="py-name">minidom</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">xml</tt><tt class="py-op">.</tt><tt class="py-name">etree</tt><tt class="py-op">.</tt><tt class="py-name">ElementTree</tt> <tt class="py-keyword">as</tt> <tt class="py-name">ET</tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-1', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt id="link-2" class="py-name" targets="Function Bio._py3k._is_int_or_long()=Bio._py3k-module.html#_is_int_or_long"><a title="Bio._py3k._is_int_or_long" class="py-name" href="#" onclick="return doclink('link-2', '_is_int_or_long', 'link-2');">_is_int_or_long</a></tt><tt class="py-op">,</tt> <tt class="py-name">_as_string</tt> </tt>
<a name="Pathway"></a><div id="Pathway-def"><a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"><tt class="py-comment"># Pathway</tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt> <a class="py-toggle" href="#" id="Pathway-toggle" onclick="return toggle('Pathway');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html">Pathway</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Pathway-expanded"><a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line">    <tt class="py-docstring">"""Represents a KGML pathway from KEGG.</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-docstring">    Specifies graph information for the pathway map, as described in</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"><tt class="py-docstring">    release KGML v0.7.1 (http://www.kegg.jp/kegg/xml/docs/)</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"><tt class="py-docstring">    Attributes:</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"><tt class="py-docstring">     - name - KEGGID of the pathway map</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"><tt class="py-docstring">     - org - ko/ec/[org prefix]</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"><tt class="py-docstring">     - number - map number (integer)</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"><tt class="py-docstring">     - title - the map title</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line"><tt class="py-docstring">     - image - URL of the image map for the pathway</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line"><tt class="py-docstring">     - link - URL of information about the pathway</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line"><tt class="py-docstring">     - entries - Dictionary of entries in the pathway, keyed by node ID</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line"><tt class="py-docstring">     - reactions - Set of reactions in the pathway</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"><tt class="py-docstring">    The name attribute has a restricted format, so we make it a property and</tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"><tt class="py-docstring">    enforce the formatting.</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line"><tt class="py-docstring">    The Pathway object is the only allowed route for adding/removing</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line"><tt class="py-docstring">    Entry, Reaction, or Relation elements.</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line"><tt class="py-docstring">    Entries are held in a dictionary and keyed by the node ID for the</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line"><tt class="py-docstring">    pathway graph - this allows for ready access via the Reaction/Relation</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line"><tt class="py-docstring">    etc. elements.  Entries must be added before reference by any other</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"><tt class="py-docstring">    element.</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line"><tt class="py-docstring">    Reactions are held in a dictionary, keyed by node ID for the path.</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line"><tt class="py-docstring">    The elements referred to in the reaction must be added before the</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line"><tt class="py-docstring">    reaction itself.</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line"> </tt>
<a name="Pathway.__init__"></a><div id="Pathway.__init__-def"><a name="L65"></a><tt class="py-lineno"> 65</tt> <a class="py-toggle" href="#" id="Pathway.__init__-toggle" onclick="return toggle('Pathway.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Pathway.__init__-expanded"><a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the class."""</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-3" class="py-name" targets="Method Bio.KEGG.Compound.Record._name()=Bio.KEGG.Compound.Record-class.html#_name,Method Bio.KEGG.Enzyme.Record._name()=Bio.KEGG.Enzyme.Record-class.html#_name,Method Bio.KEGG.Gene.Record._name()=Bio.KEGG.Gene.Record-class.html#_name"><a title="Bio.KEGG.Compound.Record._name
Bio.KEGG.Enzyme.Record._name
Bio.KEGG.Gene.Record._name" class="py-name" href="#" onclick="return doclink('link-3', '_name', 'link-3');">_name</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">org</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_number</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-4" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.title()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#title,Method Bio.GenBank._FeatureConsumer.title()=Bio.GenBank._FeatureConsumer-class.html#title,Method Bio.GenBank._RecordConsumer.title()=Bio.GenBank._RecordConsumer-class.html#title"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-4', 'title', 'link-4');">title</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">image</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-5" class="py-name" targets="Method Bio.Nexus.Nodes.Chain.link()=Bio.Nexus.Nodes.Chain-class.html#link"><a title="Bio.Nexus.Nodes.Chain.link" class="py-name" href="#" onclick="return doclink('link-5', 'link', 'link-5');">link</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">entries</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_reactions</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_relations</tt> <tt class="py-op">=</tt> <tt id="link-6" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.set()=Bio.Nexus.Nexus.StepMatrix-class.html#set,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.set()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#set,Method Bio.SVDSuperimposer.SVDSuperimposer.set()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#set"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-6', 'set', 'link-6');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line"> </tt>
<a name="Pathway.get_KGML"></a><div id="Pathway.get_KGML-def"><a name="L77"></a><tt class="py-lineno"> 77</tt> <a class="py-toggle" href="#" id="Pathway.get_KGML-toggle" onclick="return toggle('Pathway.get_KGML');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#get_KGML">get_KGML</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway.get_KGML-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Pathway.get_KGML-expanded"><a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the pathway as a string in prettified KGML format."""</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line">        <tt id="link-7" class="py-name" targets="Variable Bio.Wise.dnal.Statistics.header=Bio.Wise.dnal.Statistics-class.html#header"><a title="Bio.Wise.dnal.Statistics.header" class="py-name" href="#" onclick="return doclink('link-7', 'header', 'link-7');">header</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'\n'</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-string">'&lt;?xml version="1.0"?&gt;'</tt><tt class="py-op">,</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line">                            <tt class="py-string">'&lt;!DOCTYPE pathway SYSTEM '</tt> <tt class="py-op">+</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">                            <tt class="py-string">'"http://www.genome.jp/kegg/xml/'</tt> <tt class="py-op">+</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line">                            <tt class="py-string">'KGML_v0.7.1_.dtd"&gt;'</tt><tt class="py-op">,</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line">                            <tt class="py-string">'&lt;!-- Created by KGML_Pathway.py %s --&gt;'</tt> <tt class="py-op">%</tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line">                            <tt class="py-name">time</tt><tt class="py-op">.</tt><tt class="py-name">asctime</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line">        <tt class="py-name">rough_xml</tt> <tt class="py-op">=</tt> <tt id="link-8" class="py-name"><a title="Bio.Wise.dnal.Statistics.header" class="py-name" href="#" onclick="return doclink('link-8', 'header', 'link-7');">header</a></tt> <tt class="py-op">+</tt> <tt class="py-name">_as_string</tt><tt class="py-op">(</tt><tt class="py-name">ET</tt><tt class="py-op">.</tt><tt id="link-9" class="py-name" targets="Method Bio.Crystal.Crystal.tostring()=Bio.Crystal.Crystal-class.html#tostring,Method Bio.Seq.MutableSeq.tostring()=Bio.Seq.MutableSeq-class.html#tostring,Method Bio.Seq.Seq.tostring()=Bio.Seq.Seq-class.html#tostring,Method BioSQL.BioSeq.DBSeq.tostring()=BioSQL.BioSeq.DBSeq-class.html#tostring"><a title="Bio.Crystal.Crystal.tostring
Bio.Seq.MutableSeq.tostring
Bio.Seq.Seq.tostring
BioSQL.BioSeq.DBSeq.tostring" class="py-name" href="#" onclick="return doclink('link-9', 'tostring', 'link-9');">tostring</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-10" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.element=Bio.KEGG.KGML.KGML_pathway.Component-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Entry.element=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Graphics.element=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.element=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.element=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Relation.element=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#element,Variable Bio.Phylo._cdao_owl.element=Bio.Phylo._cdao_owl-module.html#element"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-10', 'element', 'link-10');">element</a></tt><tt class="py-op">,</tt> <tt class="py-string">'utf-8'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line">        <tt class="py-name">reparsed</tt> <tt class="py-op">=</tt> <tt class="py-name">minidom</tt><tt class="py-op">.</tt><tt class="py-name">parseString</tt><tt class="py-op">(</tt><tt class="py-name">rough_xml</tt><tt class="py-op">)</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">reparsed</tt><tt class="py-op">.</tt><tt class="py-name">toprettyxml</tt><tt class="py-op">(</tt><tt class="py-name">indent</tt><tt class="py-op">=</tt><tt class="py-string">"  "</tt><tt class="py-op">)</tt> </tt>
</div><a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line"> </tt>
<a name="Pathway.add_entry"></a><div id="Pathway.add_entry-def"><a name="L89"></a><tt class="py-lineno"> 89</tt> <a class="py-toggle" href="#" id="Pathway.add_entry-toggle" onclick="return toggle('Pathway.add_entry');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#add_entry">add_entry</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">entry</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway.add_entry-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Pathway.add_entry-expanded"><a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add an Entry element to the pathway."""</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">        <tt class="py-comment"># We insist that the node ID is an integer</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt id="link-11" class="py-name"><a title="Bio._py3k._is_int_or_long" class="py-name" href="#" onclick="return doclink('link-11', '_is_int_or_long', 'link-2');">_is_int_or_long</a></tt><tt class="py-op">(</tt><tt id="link-12" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Reaction.entry=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#entry,Function Bio.TogoWS.entry()=Bio.TogoWS-module.html#entry"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-12', 'entry', 'link-12');">entry</a></tt><tt class="py-op">.</tt><tt id="link-13" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Variable Bio.PDB.Entity.Entity.id=Bio.PDB.Entity.Entity-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-13', 'id', 'link-13');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> \ </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">            <tt class="py-string">"Node ID must be an integer, got %s (%s)"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-14" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.type()=Bio.Phylo.PhyloXMLIO.Writer-class.html#type"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-14', 'type', 'link-14');">type</a></tt><tt class="py-op">(</tt><tt id="link-15" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-15', 'entry', 'link-12');">entry</a></tt><tt class="py-op">.</tt><tt id="link-16" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-16', 'id', 'link-13');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line">                                                         <tt id="link-17" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-17', 'entry', 'link-12');">entry</a></tt><tt class="py-op">.</tt><tt id="link-18" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-18', 'id', 'link-13');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line">        <tt id="link-19" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-19', 'entry', 'link-12');">entry</a></tt><tt class="py-op">.</tt><tt id="link-20" class="py-name" targets="Method Bio.KEGG.Compound.Record._pathway()=Bio.KEGG.Compound.Record-class.html#_pathway,Method Bio.KEGG.Enzyme.Record._pathway()=Bio.KEGG.Enzyme.Record-class.html#_pathway"><a title="Bio.KEGG.Compound.Record._pathway
Bio.KEGG.Enzyme.Record._pathway" class="py-name" href="#" onclick="return doclink('link-20', '_pathway', 'link-20');">_pathway</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt>           <tt class="py-comment"># Let the entry know about the pathway</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">[</tt><tt id="link-21" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-21', 'entry', 'link-12');">entry</a></tt><tt class="py-op">.</tt><tt id="link-22" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-22', 'id', 'link-13');">id</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-23" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-23', 'entry', 'link-12');">entry</a></tt> </tt>
</div><a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line"> </tt>
<a name="Pathway.remove_entry"></a><div id="Pathway.remove_entry-def"><a name="L98"></a><tt class="py-lineno"> 98</tt> <a class="py-toggle" href="#" id="Pathway.remove_entry-toggle" onclick="return toggle('Pathway.remove_entry');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#remove_entry">remove_entry</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">entry</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway.remove_entry-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Pathway.remove_entry-expanded"><a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">        <tt class="py-docstring">"""Remove an Entry element from the pathway."""</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt id="link-24" class="py-name"><a title="Bio._py3k._is_int_or_long" class="py-name" href="#" onclick="return doclink('link-24', '_is_int_or_long', 'link-2');">_is_int_or_long</a></tt><tt class="py-op">(</tt><tt id="link-25" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-25', 'entry', 'link-12');">entry</a></tt><tt class="py-op">.</tt><tt id="link-26" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-26', 'id', 'link-13');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> \ </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line">            <tt class="py-string">"Node ID must be an integer, got %s (%s)"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-27" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-27', 'type', 'link-14');">type</a></tt><tt class="py-op">(</tt><tt id="link-28" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-28', 'entry', 'link-12');">entry</a></tt><tt class="py-op">.</tt><tt id="link-29" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-29', 'id', 'link-13');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line">                                                         <tt id="link-30" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-30', 'entry', 'link-12');">entry</a></tt><tt class="py-op">.</tt><tt id="link-31" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-31', 'id', 'link-13');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">        <tt class="py-comment"># We need to remove the entry from any other elements that may</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line">        <tt class="py-comment"># contain it, which means removing those elements</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">        <tt class="py-comment"># TODO</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">[</tt><tt id="link-32" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-32', 'entry', 'link-12');">entry</a></tt><tt class="py-op">.</tt><tt id="link-33" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-33', 'id', 'link-13');">id</a></tt><tt class="py-op">]</tt> </tt>
</div><a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line"> </tt>
<a name="Pathway.add_reaction"></a><div id="Pathway.add_reaction-def"><a name="L108"></a><tt class="py-lineno">108</tt> <a class="py-toggle" href="#" id="Pathway.add_reaction-toggle" onclick="return toggle('Pathway.add_reaction');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#add_reaction">add_reaction</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">reaction</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway.add_reaction-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Pathway.add_reaction-expanded"><a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add a Reaction element to the pathway."""</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">        <tt class="py-comment"># We insist that the node ID is an integer and corresponds to an entry</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt id="link-34" class="py-name"><a title="Bio._py3k._is_int_or_long" class="py-name" href="#" onclick="return doclink('link-34', '_is_int_or_long', 'link-2');">_is_int_or_long</a></tt><tt class="py-op">(</tt><tt id="link-35" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.reaction=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#reaction"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.reaction" class="py-name" href="#" onclick="return doclink('link-35', 'reaction', 'link-35');">reaction</a></tt><tt class="py-op">.</tt><tt id="link-36" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-36', 'id', 'link-13');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> \ </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">            <tt class="py-string">"Node ID must be an integer, got %s (%s)"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-37" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-37', 'type', 'link-14');">type</a></tt><tt class="py-op">(</tt><tt id="link-38" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.reaction" class="py-name" href="#" onclick="return doclink('link-38', 'reaction', 'link-35');">reaction</a></tt><tt class="py-op">.</tt><tt id="link-39" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-39', 'id', 'link-13');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">                                                         <tt id="link-40" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.reaction" class="py-name" href="#" onclick="return doclink('link-40', 'reaction', 'link-35');">reaction</a></tt><tt class="py-op">.</tt><tt id="link-41" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-41', 'id', 'link-13');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt id="link-42" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.reaction" class="py-name" href="#" onclick="return doclink('link-42', 'reaction', 'link-35');">reaction</a></tt><tt class="py-op">.</tt><tt id="link-43" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-43', 'id', 'link-13');">id</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">,</tt> \ </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">            <tt class="py-string">"Reaction ID %d has no corresponding entry"</tt> <tt class="py-op">%</tt> <tt id="link-44" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.reaction" class="py-name" href="#" onclick="return doclink('link-44', 'reaction', 'link-35');">reaction</a></tt><tt class="py-op">.</tt><tt id="link-45" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-45', 'id', 'link-13');">id</a></tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">        <tt id="link-46" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.reaction" class="py-name" href="#" onclick="return doclink('link-46', 'reaction', 'link-35');">reaction</a></tt><tt class="py-op">.</tt><tt id="link-47" class="py-name"><a title="Bio.KEGG.Compound.Record._pathway
Bio.KEGG.Enzyme.Record._pathway" class="py-name" href="#" onclick="return doclink('link-47', '_pathway', 'link-20');">_pathway</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt>    <tt class="py-comment"># Let the reaction know about the pathway</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_reactions</tt><tt class="py-op">[</tt><tt id="link-48" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.reaction" class="py-name" href="#" onclick="return doclink('link-48', 'reaction', 'link-35');">reaction</a></tt><tt class="py-op">.</tt><tt id="link-49" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-49', 'id', 'link-13');">id</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-50" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.reaction" class="py-name" href="#" onclick="return doclink('link-50', 'reaction', 'link-35');">reaction</a></tt> </tt>
</div><a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line"> </tt>
<a name="Pathway.remove_reaction"></a><div id="Pathway.remove_reaction-def"><a name="L119"></a><tt class="py-lineno">119</tt> <a class="py-toggle" href="#" id="Pathway.remove_reaction-toggle" onclick="return toggle('Pathway.remove_reaction');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#remove_reaction">remove_reaction</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">reaction</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway.remove_reaction-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Pathway.remove_reaction-expanded"><a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">        <tt class="py-docstring">"""Remove a Reaction element from the pathway."""</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt id="link-51" class="py-name"><a title="Bio._py3k._is_int_or_long" class="py-name" href="#" onclick="return doclink('link-51', '_is_int_or_long', 'link-2');">_is_int_or_long</a></tt><tt class="py-op">(</tt><tt id="link-52" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.reaction" class="py-name" href="#" onclick="return doclink('link-52', 'reaction', 'link-35');">reaction</a></tt><tt class="py-op">.</tt><tt id="link-53" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-53', 'id', 'link-13');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> \ </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">            <tt class="py-string">"Node ID must be an integer, got %s (%s)"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-54" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-54', 'type', 'link-14');">type</a></tt><tt class="py-op">(</tt><tt id="link-55" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.reaction" class="py-name" href="#" onclick="return doclink('link-55', 'reaction', 'link-35');">reaction</a></tt><tt class="py-op">.</tt><tt id="link-56" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-56', 'id', 'link-13');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">                                                         <tt id="link-57" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.reaction" class="py-name" href="#" onclick="return doclink('link-57', 'reaction', 'link-35');">reaction</a></tt><tt class="py-op">.</tt><tt id="link-58" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-58', 'id', 'link-13');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">        <tt class="py-comment"># We need to remove the reaction from any other elements that may</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line">        <tt class="py-comment"># contain it, which means removing those elements</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line">        <tt class="py-comment"># TODO</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_reactions</tt><tt class="py-op">[</tt><tt id="link-59" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.reaction" class="py-name" href="#" onclick="return doclink('link-59', 'reaction', 'link-35');">reaction</a></tt><tt class="py-op">.</tt><tt id="link-60" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-60', 'id', 'link-13');">id</a></tt><tt class="py-op">]</tt> </tt>
</div><a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line"> </tt>
<a name="Pathway.add_relation"></a><div id="Pathway.add_relation-def"><a name="L129"></a><tt class="py-lineno">129</tt> <a class="py-toggle" href="#" id="Pathway.add_relation-toggle" onclick="return toggle('Pathway.add_relation');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#add_relation">add_relation</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">relation</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway.add_relation-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Pathway.add_relation-expanded"><a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add a Relation element to the pathway."""</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line">        <tt class="py-name">relation</tt><tt class="py-op">.</tt><tt id="link-61" class="py-name"><a title="Bio.KEGG.Compound.Record._pathway
Bio.KEGG.Enzyme.Record._pathway" class="py-name" href="#" onclick="return doclink('link-61', '_pathway', 'link-20');">_pathway</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt>    <tt class="py-comment"># Let the reaction know about the pathway</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_relations</tt><tt class="py-op">.</tt><tt id="link-62" class="py-name" targets="Method Bio.Graphics.BasicChromosome._ChromosomeComponent.add()=Bio.Graphics.BasicChromosome._ChromosomeComponent-class.html#add,Method Bio.Nexus.Nexus.StepMatrix.add()=Bio.Nexus.Nexus.StepMatrix-class.html#add,Method Bio.Nexus.Nodes.Chain.add()=Bio.Nexus.Nodes.Chain-class.html#add,Method Bio.PDB.Entity.Entity.add()=Bio.PDB.Entity.Entity-class.html#add,Method Bio.PDB.Residue.DisorderedResidue.add()=Bio.PDB.Residue.DisorderedResidue-class.html#add,Method Bio.PDB.Residue.Residue.add()=Bio.PDB.Residue.Residue-class.html#add,Method Bio.Restriction.Restriction.RestrictionBatch.add()=Bio.Restriction.Restriction.RestrictionBatch-class.html#add"><a title="Bio.Graphics.BasicChromosome._ChromosomeComponent.add
Bio.Nexus.Nexus.StepMatrix.add
Bio.Nexus.Nodes.Chain.add
Bio.PDB.Entity.Entity.add
Bio.PDB.Residue.DisorderedResidue.add
Bio.PDB.Residue.Residue.add
Bio.Restriction.Restriction.RestrictionBatch.add" class="py-name" href="#" onclick="return doclink('link-62', 'add', 'link-62');">add</a></tt><tt class="py-op">(</tt><tt class="py-name">relation</tt><tt class="py-op">)</tt> </tt>
</div><a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line"> </tt>
<a name="Pathway.remove_relation"></a><div id="Pathway.remove_relation-def"><a name="L134"></a><tt class="py-lineno">134</tt> <a class="py-toggle" href="#" id="Pathway.remove_relation-toggle" onclick="return toggle('Pathway.remove_relation');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#remove_relation">remove_relation</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">relation</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway.remove_relation-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Pathway.remove_relation-expanded"><a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line">        <tt class="py-docstring">"""Remove a Relation element from the pathway."""</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_relations</tt><tt class="py-op">.</tt><tt id="link-63" class="py-name" targets="Method Bio.Crystal.Chain.remove()=Bio.Crystal.Chain-class.html#remove,Method Bio.Graphics.BasicChromosome._ChromosomeComponent.remove()=Bio.Graphics.BasicChromosome._ChromosomeComponent-class.html#remove,Method Bio.Restriction.Restriction.RestrictionBatch.remove()=Bio.Restriction.Restriction.RestrictionBatch-class.html#remove,Method Bio.Seq.MutableSeq.remove()=Bio.Seq.MutableSeq-class.html#remove,Method BioSQL.Loader.DatabaseRemover.remove()=BioSQL.Loader.DatabaseRemover-class.html#remove"><a title="Bio.Crystal.Chain.remove
Bio.Graphics.BasicChromosome._ChromosomeComponent.remove
Bio.Restriction.Restriction.RestrictionBatch.remove
Bio.Seq.MutableSeq.remove
BioSQL.Loader.DatabaseRemover.remove" class="py-name" href="#" onclick="return doclink('link-63', 'remove', 'link-63');">remove</a></tt><tt class="py-op">(</tt><tt class="py-name">relation</tt><tt class="py-op">)</tt> </tt>
</div><a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line"> </tt>
<a name="Pathway.__str__"></a><div id="Pathway.__str__-def"><a name="L138"></a><tt class="py-lineno">138</tt> <a class="py-toggle" href="#" id="Pathway.__str__-toggle" onclick="return toggle('Pathway.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway.__str__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Pathway.__str__-expanded"><a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a readable summary description string."""</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line">        <tt class="py-name">outstr</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">'Pathway: %s'</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-64" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-64', 'title', 'link-4');">title</a></tt><tt class="py-op">,</tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line">                  <tt class="py-string">'KEGG ID: %s'</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-65" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-65', 'name', 'link-65');">name</a></tt><tt class="py-op">,</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line">                  <tt class="py-string">'Image file: %s'</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">image</tt><tt class="py-op">,</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line">                  <tt class="py-string">'Organism: %s'</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">org</tt><tt class="py-op">,</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line">                  <tt class="py-string">'Entries: %d'</tt> <tt class="py-op">%</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">                  <tt class="py-string">'Entry types:'</tt><tt class="py-op">]</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">t</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">'ortholog'</tt><tt class="py-op">,</tt> <tt class="py-string">'enzyme'</tt><tt class="py-op">,</tt> <tt class="py-string">'reaction'</tt><tt class="py-op">,</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line">                  <tt class="py-string">'gene'</tt><tt class="py-op">,</tt> <tt class="py-string">'group'</tt><tt class="py-op">,</tt> <tt class="py-string">'compound'</tt><tt class="py-op">,</tt> <tt class="py-string">'map'</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">            <tt class="py-name">etype</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">e</tt> <tt class="py-keyword">for</tt> <tt class="py-name">e</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">.</tt><tt id="link-66" class="py-name" targets="Method Bio.Crystal.Crystal.values()=Bio.Crystal.Crystal-class.html#values,Method Bio.File._IndexedSeqFileDict.values()=Bio.File._IndexedSeqFileDict-class.html#values,Method Bio.Phylo.PhyloXML.Events.values()=Bio.Phylo.PhyloXML.Events-class.html#values,Method BioSQL.BioSeqDatabase.BioSeqDatabase.values()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#values,Method BioSQL.BioSeqDatabase.DBServer.values()=BioSQL.BioSeqDatabase.DBServer-class.html#values"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-66', 'values', 'link-66');">values</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt class="py-name">e</tt><tt class="py-op">.</tt><tt id="link-67" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-67', 'type', 'link-14');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-name">t</tt><tt class="py-op">]</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">etype</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">                <tt class="py-name">outstr</tt><tt class="py-op">.</tt><tt id="link-68" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-68', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">'\t%s: %d'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">t</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">etype</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">'\n'</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">outstr</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">'\n'</tt> </tt>
</div><a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line"> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">    <tt class="py-comment"># Assert correct formatting of the pathway name, and other attributes</tt> </tt>
<a name="Pathway._getname"></a><div id="Pathway._getname-def"><a name="L154"></a><tt class="py-lineno">154</tt> <a class="py-toggle" href="#" id="Pathway._getname-toggle" onclick="return toggle('Pathway._getname');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#_getname">_getname</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway._getname-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Pathway._getname-expanded"><a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-69" class="py-name"><a title="Bio.KEGG.Compound.Record._name
Bio.KEGG.Enzyme.Record._name
Bio.KEGG.Gene.Record._name" class="py-name" href="#" onclick="return doclink('link-69', '_name', 'link-3');">_name</a></tt> </tt>
</div><a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line"> </tt>
<a name="Pathway._setname"></a><div id="Pathway._setname-def"><a name="L157"></a><tt class="py-lineno">157</tt> <a class="py-toggle" href="#" id="Pathway._setname-toggle" onclick="return toggle('Pathway._setname');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#_setname">_setname</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway._setname-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Pathway._setname-expanded"><a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt id="link-70" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.value()=Bio.Phylo.PhyloXMLIO.Writer-class.html#value"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-70', 'value', 'link-70');">value</a></tt><tt class="py-op">.</tt><tt id="link-71" class="py-name" targets="Method Bio.Seq.Seq.startswith()=Bio.Seq.Seq-class.html#startswith"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-71', 'startswith', 'link-71');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'path:'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> \ </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">            <tt class="py-string">"Pathway name should begin with 'path:', got %s"</tt> <tt class="py-op">%</tt> <tt id="link-72" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-72', 'value', 'link-70');">value</a></tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-73" class="py-name"><a title="Bio.KEGG.Compound.Record._name
Bio.KEGG.Enzyme.Record._name
Bio.KEGG.Gene.Record._name" class="py-name" href="#" onclick="return doclink('link-73', '_name', 'link-3');">_name</a></tt> <tt class="py-op">=</tt> <tt id="link-74" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-74', 'value', 'link-70');">value</a></tt> </tt>
</div><a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line"> </tt>
<a name="Pathway._delname"></a><div id="Pathway._delname-def"><a name="L162"></a><tt class="py-lineno">162</tt> <a class="py-toggle" href="#" id="Pathway._delname-toggle" onclick="return toggle('Pathway._delname');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#_delname">_delname</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway._delname-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Pathway._delname-expanded"><a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-75" class="py-name"><a title="Bio.KEGG.Compound.Record._name
Bio.KEGG.Enzyme.Record._name
Bio.KEGG.Gene.Record._name" class="py-name" href="#" onclick="return doclink('link-75', '_name', 'link-3');">_name</a></tt> </tt>
</div><a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">    <tt id="link-76" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-76', 'name', 'link-65');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-77" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.property()=Bio.Phylo.PhyloXMLIO.Parser-class.html#property,Method Bio.Phylo.PhyloXMLIO.Writer.property()=Bio.Phylo.PhyloXMLIO.Writer-class.html#property"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-77', 'property', 'link-77');">property</a></tt><tt class="py-op">(</tt><tt id="link-78" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Entry._getname()=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#_getname,Method Bio.KEGG.KGML.KGML_pathway.Pathway._getname()=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#_getname"><a title="Bio.KEGG.KGML.KGML_pathway.Entry._getname
Bio.KEGG.KGML.KGML_pathway.Pathway._getname" class="py-name" href="#" onclick="return doclink('link-78', '_getname', 'link-78');">_getname</a></tt><tt class="py-op">,</tt> <tt id="link-79" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Entry._setname()=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#_setname,Method Bio.KEGG.KGML.KGML_pathway.Pathway._setname()=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#_setname"><a title="Bio.KEGG.KGML.KGML_pathway.Entry._setname
Bio.KEGG.KGML.KGML_pathway.Pathway._setname" class="py-name" href="#" onclick="return doclink('link-79', '_setname', 'link-79');">_setname</a></tt><tt class="py-op">,</tt> <tt id="link-80" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Entry._delname()=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#_delname,Method Bio.KEGG.KGML.KGML_pathway.Pathway._delname()=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#_delname"><a title="Bio.KEGG.KGML.KGML_pathway.Entry._delname
Bio.KEGG.KGML.KGML_pathway.Pathway._delname" class="py-name" href="#" onclick="return doclink('link-80', '_delname', 'link-80');">_delname</a></tt><tt class="py-op">,</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">                    <tt class="py-string">"The KEGGID for the pathway map."</tt><tt class="py-op">)</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line"> </tt>
<a name="Pathway._getnumber"></a><div id="Pathway._getnumber-def"><a name="L167"></a><tt class="py-lineno">167</tt> <a class="py-toggle" href="#" id="Pathway._getnumber-toggle" onclick="return toggle('Pathway._getnumber');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#_getnumber">_getnumber</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway._getnumber-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Pathway._getnumber-expanded"><a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_number</tt> </tt>
</div><a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line"> </tt>
<a name="Pathway._setnumber"></a><div id="Pathway._setnumber-def"><a name="L170"></a><tt class="py-lineno">170</tt> <a class="py-toggle" href="#" id="Pathway._setnumber-toggle" onclick="return toggle('Pathway._setnumber');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#_setnumber">_setnumber</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway._setnumber-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Pathway._setnumber-expanded"><a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_number</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-81" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-81', 'value', 'link-70');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line"> </tt>
<a name="Pathway._delnumber"></a><div id="Pathway._delnumber-def"><a name="L173"></a><tt class="py-lineno">173</tt> <a class="py-toggle" href="#" id="Pathway._delnumber-toggle" onclick="return toggle('Pathway._delnumber');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#_delnumber">_delnumber</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway._delnumber-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Pathway._delnumber-expanded"><a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_number</tt> </tt>
</div><a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line">    <tt id="link-82" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Pathway.number=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#number"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.number" class="py-name" href="#" onclick="return doclink('link-82', 'number', 'link-82');">number</a></tt> <tt class="py-op">=</tt> <tt id="link-83" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-83', 'property', 'link-77');">property</a></tt><tt class="py-op">(</tt><tt id="link-84" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Pathway._getnumber()=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#_getnumber"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway._getnumber" class="py-name" href="#" onclick="return doclink('link-84', '_getnumber', 'link-84');">_getnumber</a></tt><tt class="py-op">,</tt> <tt id="link-85" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Pathway._setnumber()=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#_setnumber"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway._setnumber" class="py-name" href="#" onclick="return doclink('link-85', '_setnumber', 'link-85');">_setnumber</a></tt><tt class="py-op">,</tt> <tt id="link-86" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Pathway._delnumber()=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#_delnumber"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway._delnumber" class="py-name" href="#" onclick="return doclink('link-86', '_delnumber', 'link-86');">_delnumber</a></tt><tt class="py-op">,</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">                      <tt class="py-string">"The KEGG map number."</tt><tt class="py-op">)</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line"> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Pathway.compounds"></a><div id="Pathway.compounds-def"><a name="L179"></a><tt class="py-lineno">179</tt> <a class="py-toggle" href="#" id="Pathway.compounds-toggle" onclick="return toggle('Pathway.compounds');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#compounds">compounds</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway.compounds-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Pathway.compounds-expanded"><a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get a list of entries of type compound."""</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">e</tt> <tt class="py-keyword">for</tt> <tt class="py-name">e</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">.</tt><tt id="link-87" class="py-name"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-87', 'values', 'link-66');">values</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt class="py-name">e</tt><tt class="py-op">.</tt><tt id="link-88" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-88', 'type', 'link-14');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'compound'</tt><tt class="py-op">]</tt> </tt>
</div><a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line"> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Pathway.maps"></a><div id="Pathway.maps-def"><a name="L184"></a><tt class="py-lineno">184</tt> <a class="py-toggle" href="#" id="Pathway.maps-toggle" onclick="return toggle('Pathway.maps');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#maps">maps</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway.maps-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Pathway.maps-expanded"><a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get a list of entries of type map."""</tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">e</tt> <tt class="py-keyword">for</tt> <tt class="py-name">e</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">.</tt><tt id="link-89" class="py-name"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-89', 'values', 'link-66');">values</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt class="py-name">e</tt><tt class="py-op">.</tt><tt id="link-90" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-90', 'type', 'link-14');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'map'</tt><tt class="py-op">]</tt> </tt>
</div><a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line"> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Pathway.orthologs"></a><div id="Pathway.orthologs-def"><a name="L189"></a><tt class="py-lineno">189</tt> <a class="py-toggle" href="#" id="Pathway.orthologs-toggle" onclick="return toggle('Pathway.orthologs');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#orthologs">orthologs</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway.orthologs-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Pathway.orthologs-expanded"><a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get a list of entries of type ortholog."""</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">e</tt> <tt class="py-keyword">for</tt> <tt class="py-name">e</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">.</tt><tt id="link-91" class="py-name"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-91', 'values', 'link-66');">values</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt class="py-name">e</tt><tt class="py-op">.</tt><tt id="link-92" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-92', 'type', 'link-14');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'ortholog'</tt><tt class="py-op">]</tt> </tt>
</div><a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line"> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Pathway.genes"></a><div id="Pathway.genes-def"><a name="L194"></a><tt class="py-lineno">194</tt> <a class="py-toggle" href="#" id="Pathway.genes-toggle" onclick="return toggle('Pathway.genes');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#genes">genes</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway.genes-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Pathway.genes-expanded"><a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get a list of entries of type gene."""</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">e</tt> <tt class="py-keyword">for</tt> <tt class="py-name">e</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">.</tt><tt id="link-93" class="py-name"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-93', 'values', 'link-66');">values</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt class="py-name">e</tt><tt class="py-op">.</tt><tt id="link-94" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-94', 'type', 'link-14');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'gene'</tt><tt class="py-op">]</tt> </tt>
</div><a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line"> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Pathway.reactions"></a><div id="Pathway.reactions-def"><a name="L199"></a><tt class="py-lineno">199</tt> <a class="py-toggle" href="#" id="Pathway.reactions-toggle" onclick="return toggle('Pathway.reactions');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#reactions">reactions</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway.reactions-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Pathway.reactions-expanded"><a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get a list of reactions in the pathway."""</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_reactions</tt><tt class="py-op">.</tt><tt id="link-95" class="py-name"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-95', 'values', 'link-66');">values</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line"> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Pathway.reaction_entries"></a><div id="Pathway.reaction_entries-def"><a name="L204"></a><tt class="py-lineno">204</tt> <a class="py-toggle" href="#" id="Pathway.reaction_entries-toggle" onclick="return toggle('Pathway.reaction_entries');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#reaction_entries">reaction_entries</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway.reaction_entries-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Pathway.reaction_entries-expanded"><a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line">        <tt class="py-docstring">"""List of entries corresponding to each reaction in the pathway."""</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">[</tt><tt id="link-96" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-96', 'i', 'link-96');">i</a></tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt id="link-97" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-97', 'i', 'link-96');">i</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_reactions</tt><tt class="py-op">]</tt> </tt>
</div><a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line"> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Pathway.relations"></a><div id="Pathway.relations-def"><a name="L209"></a><tt class="py-lineno">209</tt> <a class="py-toggle" href="#" id="Pathway.relations-toggle" onclick="return toggle('Pathway.relations');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#relations">relations</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway.relations-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Pathway.relations-expanded"><a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get a list of relations in the pathway."""</tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_relations</tt><tt class="py-op">)</tt> </tt>
</div><a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line"> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Pathway.element"></a><div id="Pathway.element-def"><a name="L214"></a><tt class="py-lineno">214</tt> <a class="py-toggle" href="#" id="Pathway.element-toggle" onclick="return toggle('Pathway.element');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#element">element</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway.element-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Pathway.element-expanded"><a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the Pathway as a valid KGML element."""</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line">        <tt class="py-comment"># The root is this Pathway element</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line">        <tt class="py-name">pathway</tt> <tt class="py-op">=</tt> <tt class="py-name">ET</tt><tt class="py-op">.</tt><tt class="py-name">Element</tt><tt class="py-op">(</tt><tt class="py-string">'pathway'</tt><tt class="py-op">)</tt> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line">        <tt class="py-name">pathway</tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'name'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-98" class="py-name"><a title="Bio.KEGG.Compound.Record._name
Bio.KEGG.Enzyme.Record._name
Bio.KEGG.Gene.Record._name" class="py-name" href="#" onclick="return doclink('link-98', '_name', 'link-3');">_name</a></tt><tt class="py-op">,</tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line">                          <tt class="py-string">'org'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">org</tt><tt class="py-op">,</tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">                          <tt class="py-string">'number'</tt><tt class="py-op">:</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_number</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line">                          <tt class="py-string">'title'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-99" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-99', 'title', 'link-4');">title</a></tt><tt class="py-op">,</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line">                          <tt class="py-string">'image'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">image</tt><tt class="py-op">,</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line">                          <tt class="py-string">'link'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-100" class="py-name"><a title="Bio.Nexus.Nodes.Chain.link" class="py-name" href="#" onclick="return doclink('link-100', 'link', 'link-5');">link</a></tt><tt class="py-op">,</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line">                          <tt class="py-op">}</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line">        <tt class="py-comment"># We add the Entries in node ID order</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">eid</tt><tt class="py-op">,</tt> <tt id="link-101" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-101', 'entry', 'link-12');">entry</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">.</tt><tt id="link-102" class="py-name" targets="Method Bio.Crystal.Crystal.items()=Bio.Crystal.Crystal-class.html#items,Method Bio.File._IndexedSeqFileDict.items()=Bio.File._IndexedSeqFileDict-class.html#items,Method Bio.Phylo.PhyloXML.Events.items()=Bio.Phylo.PhyloXML.Events-class.html#items,Variable Bio.SearchIO._model.query.QueryResult.items=Bio.SearchIO._model.query.QueryResult-class.html#items,Method BioSQL.BioSeqDatabase.BioSeqDatabase.items()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#items,Method BioSQL.BioSeqDatabase.DBServer.items()=BioSQL.BioSeqDatabase.DBServer-class.html#items"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-102', 'items', 'link-102');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line">            <tt class="py-name">pathway</tt><tt class="py-op">.</tt><tt id="link-103" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-103', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt id="link-104" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-104', 'entry', 'link-12');">entry</a></tt><tt class="py-op">.</tt><tt id="link-105" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-105', 'element', 'link-10');">element</a></tt><tt class="py-op">)</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line">        <tt class="py-comment"># Next we add Relations</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">relation</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_relations</tt><tt class="py-op">:</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line">            <tt class="py-name">pathway</tt><tt class="py-op">.</tt><tt id="link-106" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-106', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">relation</tt><tt class="py-op">.</tt><tt id="link-107" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-107', 'element', 'link-10');">element</a></tt><tt class="py-op">)</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">eid</tt><tt class="py-op">,</tt> <tt id="link-108" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.reaction" class="py-name" href="#" onclick="return doclink('link-108', 'reaction', 'link-35');">reaction</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_reactions</tt><tt class="py-op">.</tt><tt id="link-109" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-109', 'items', 'link-102');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">            <tt class="py-name">pathway</tt><tt class="py-op">.</tt><tt id="link-110" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-110', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt id="link-111" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.reaction" class="py-name" href="#" onclick="return doclink('link-111', 'reaction', 'link-35');">reaction</a></tt><tt class="py-op">.</tt><tt id="link-112" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-112', 'element', 'link-10');">element</a></tt><tt class="py-op">)</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">pathway</tt> </tt>
</div><a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line"> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Pathway.bounds"></a><div id="Pathway.bounds-def"><a name="L236"></a><tt class="py-lineno">236</tt> <a class="py-toggle" href="#" id="Pathway.bounds-toggle" onclick="return toggle('Pathway.bounds');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#bounds">bounds</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Pathway.bounds-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Pathway.bounds-expanded"><a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line">        <tt class="py-docstring">"""Coordinate bounds for all Graphics elements in the Pathway.</tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line"><tt class="py-docstring">        Returns the [(xmin, ymin), (xmax, ymax)] coordinates for all</tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line"><tt class="py-docstring">        Graphics elements in the Pathway</tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line">        <tt class="py-name">xlist</tt><tt class="py-op">,</tt> <tt class="py-name">ylist</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-113" class="py-name" targets="Variable Bio.Graphics.ColorSpiral.ColorSpiral.b=Bio.Graphics.ColorSpiral.ColorSpiral-class.html#b"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-113', 'b', 'link-113');">b</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-name">g</tt><tt class="py-op">.</tt><tt id="link-114" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.bounds=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#bounds,Variable Bio.KEGG.KGML.KGML_pathway.Graphics.bounds=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#bounds,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.bounds=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#bounds"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.bounds
Bio.KEGG.KGML.KGML_pathway.Graphics.bounds
Bio.KEGG.KGML.KGML_pathway.Pathway.bounds" class="py-name" href="#" onclick="return doclink('link-114', 'bounds', 'link-114');">bounds</a></tt> <tt class="py-keyword">for</tt> <tt class="py-name">g</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">.</tt><tt id="link-115" class="py-name"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-115', 'values', 'link-66');">values</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line">            <tt class="py-name">xlist</tt><tt class="py-op">.</tt><tt id="link-116" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.extend()=Bio.Align.MultipleSeqAlignment-class.html#extend,Method Bio.SCOP.Raf.SeqMap.extend()=Bio.SCOP.Raf.SeqMap-class.html#extend,Method Bio.Seq.MutableSeq.extend()=Bio.Seq.MutableSeq-class.html#extend"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-116', 'extend', 'link-116');">extend</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt id="link-117" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-117', 'b', 'link-113');">b</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt id="link-118" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-118', 'b', 'link-113');">b</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line">            <tt class="py-name">ylist</tt><tt class="py-op">.</tt><tt id="link-119" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-119', 'extend', 'link-116');">extend</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt id="link-120" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-120', 'b', 'link-113');">b</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt id="link-121" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-121', 'b', 'link-113');">b</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-op">(</tt><tt id="link-122" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.min=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#min"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-122', 'min', 'link-122');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">xlist</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-123" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-123', 'min', 'link-122');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">ylist</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt id="link-124" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.max=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#max"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-124', 'max', 'link-124');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">xlist</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-125" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-125', 'max', 'link-124');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">ylist</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
</div></div><a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line"> </tt>
<a name="Entry"></a><div id="Entry-def"><a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line"> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line"><tt class="py-comment"># Entry</tt> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt> <a class="py-toggle" href="#" id="Entry-toggle" onclick="return toggle('Entry');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Entry-class.html">Entry</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Entry-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Entry-expanded"><a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line">    <tt class="py-docstring">"""Represent an Entry from KGML.</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line"><tt class="py-docstring">    Each Entry element is a node in the pathway graph, as described in</tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line"><tt class="py-docstring">    release KGML v0.7.1 (http://www.kegg.jp/kegg/xml/docs/)</tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line"><tt class="py-docstring">    Attributes:</tt> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line"><tt class="py-docstring">     - id - The ID of the entry in the pathway map (integer)</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line"><tt class="py-docstring">     - names - List of KEGG IDs for the entry</tt> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line"><tt class="py-docstring">     - type - The type of the entry</tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line"><tt class="py-docstring">     - link - URL of information about the entry</tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line"><tt class="py-docstring">     - reaction - List of KEGG IDs of the corresponding reactions</tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line"><tt class="py-docstring">       (integer)</tt> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line"><tt class="py-docstring">     - graphics -    List of Graphics objects describing the Entry's visual</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line"><tt class="py-docstring">       representation</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line"><tt class="py-docstring">     - components - List of component node ID for this Entry ('group')</tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line"><tt class="py-docstring">     - alt - List of alternate names for the Entry</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line"><tt class="py-docstring">    NOTE: The alt attribute represents a subelement of the substrate and</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line"><tt class="py-docstring">    product elements in the KGML file</tt> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line"> </tt>
<a name="Entry.__init__"></a><div id="Entry.__init__-def"><a name="L274"></a><tt class="py-lineno">274</tt> <a class="py-toggle" href="#" id="Entry.__init__-toggle" onclick="return toggle('Entry.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Entry-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Entry.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Entry.__init__-expanded"><a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the class."""</tt> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_id</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_names</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-126" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-126', 'type', 'link-14');">type</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">image</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-127" class="py-name"><a title="Bio.Nexus.Nodes.Chain.link" class="py-name" href="#" onclick="return doclink('link-127', 'link', 'link-5');">link</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">graphics</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">components</tt> <tt class="py-op">=</tt> <tt id="link-128" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-128', 'set', 'link-6');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-129" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.alt()=Bio.Phylo.PhyloXMLIO.Writer-class.html#alt"><a title="Bio.Phylo.PhyloXMLIO.Writer.alt" class="py-name" href="#" onclick="return doclink('link-129', 'alt', 'link-129');">alt</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-130" class="py-name"><a title="Bio.KEGG.Compound.Record._pathway
Bio.KEGG.Enzyme.Record._pathway" class="py-name" href="#" onclick="return doclink('link-130', '_pathway', 'link-20');">_pathway</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_reactions</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div><a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line"> </tt>
<a name="Entry.__str__"></a><div id="Entry.__str__-def"><a name="L287"></a><tt class="py-lineno">287</tt> <a class="py-toggle" href="#" id="Entry.__str__-toggle" onclick="return toggle('Entry.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Entry-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Entry.__str__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Entry.__str__-expanded"><a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return readable descriptive string."""</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">        <tt class="py-name">outstr</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">'Entry node ID: %d'</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-131" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-131', 'id', 'link-13');">id</a></tt><tt class="py-op">,</tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">                  <tt class="py-string">'Names: %s'</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-132" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-132', 'name', 'link-65');">name</a></tt><tt class="py-op">,</tt> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line">                  <tt class="py-string">'Type: %s'</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-133" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-133', 'type', 'link-14');">type</a></tt><tt class="py-op">,</tt> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line">                  <tt class="py-string">'Components: %s'</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">components</tt><tt class="py-op">,</tt> </tt>
<a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line">                  <tt class="py-string">'Reactions: %s'</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-134" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.reaction" class="py-name" href="#" onclick="return doclink('link-134', 'reaction', 'link-35');">reaction</a></tt><tt class="py-op">,</tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line">                  <tt class="py-string">'Graphics elements: %d %s'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">graphics</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line">                                                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">graphics</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">'\n'</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">outstr</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">'\n'</tt> </tt>
</div><a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line"> </tt>
<a name="Entry.add_component"></a><div id="Entry.add_component-def"><a name="L298"></a><tt class="py-lineno">298</tt> <a class="py-toggle" href="#" id="Entry.add_component-toggle" onclick="return toggle('Entry.add_component');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Entry-class.html#add_component">add_component</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">element</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Entry.add_component-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Entry.add_component-expanded"><a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add an element to the entry.</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line"><tt class="py-docstring">        If the Entry is already part of a pathway, make sure</tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line"><tt class="py-docstring">        the component already exists.</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-135" class="py-name"><a title="Bio.KEGG.Compound.Record._pathway
Bio.KEGG.Enzyme.Record._pathway" class="py-name" href="#" onclick="return doclink('link-135', '_pathway', 'link-20');">_pathway</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt id="link-136" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-136', 'element', 'link-10');">element</a></tt><tt class="py-op">.</tt><tt id="link-137" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-137', 'id', 'link-13');">id</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-138" class="py-name"><a title="Bio.KEGG.Compound.Record._pathway
Bio.KEGG.Enzyme.Record._pathway" class="py-name" href="#" onclick="return doclink('link-138', '_pathway', 'link-20');">_pathway</a></tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">,</tt> \ </tt>
<a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line">                <tt class="py-string">"Component %s is not an entry in the pathway"</tt> <tt class="py-op">%</tt> <tt id="link-139" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-139', 'element', 'link-10');">element</a></tt><tt class="py-op">.</tt><tt id="link-140" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-140', 'id', 'link-13');">id</a></tt> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">components</tt><tt class="py-op">.</tt><tt id="link-141" class="py-name"><a title="Bio.Graphics.BasicChromosome._ChromosomeComponent.add
Bio.Nexus.Nexus.StepMatrix.add
Bio.Nexus.Nodes.Chain.add
Bio.PDB.Entity.Entity.add
Bio.PDB.Residue.DisorderedResidue.add
Bio.PDB.Residue.Residue.add
Bio.Restriction.Restriction.RestrictionBatch.add" class="py-name" href="#" onclick="return doclink('link-141', 'add', 'link-62');">add</a></tt><tt class="py-op">(</tt><tt id="link-142" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-142', 'element', 'link-10');">element</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line"> </tt>
<a name="Entry.remove_component"></a><div id="Entry.remove_component-def"><a name="L309"></a><tt class="py-lineno">309</tt> <a class="py-toggle" href="#" id="Entry.remove_component-toggle" onclick="return toggle('Entry.remove_component');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Entry-class.html#remove_component">remove_component</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Entry.remove_component-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Entry.remove_component-expanded"><a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line">        <tt class="py-docstring">"""Remove the entry with the passed ID from the group."""</tt> </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">components</tt><tt class="py-op">.</tt><tt id="link-143" class="py-name"><a title="Bio.Crystal.Chain.remove
Bio.Graphics.BasicChromosome._ChromosomeComponent.remove
Bio.Restriction.Restriction.RestrictionBatch.remove
Bio.Seq.MutableSeq.remove
BioSQL.Loader.DatabaseRemover.remove" class="py-name" href="#" onclick="return doclink('link-143', 'remove', 'link-63');">remove</a></tt><tt class="py-op">(</tt><tt id="link-144" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-144', 'value', 'link-70');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line"> </tt>
<a name="Entry.add_graphics"></a><div id="Entry.add_graphics-def"><a name="L313"></a><tt class="py-lineno">313</tt> <a class="py-toggle" href="#" id="Entry.add_graphics-toggle" onclick="return toggle('Entry.add_graphics');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Entry-class.html#add_graphics">add_graphics</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">entry</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Entry.add_graphics-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Entry.add_graphics-expanded"><a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add the Graphics entry."""</tt> </tt>
<a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">graphics</tt><tt class="py-op">.</tt><tt id="link-145" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-145', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt id="link-146" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-146', 'entry', 'link-12');">entry</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line"> </tt>
<a name="Entry.remove_graphics"></a><div id="Entry.remove_graphics-def"><a name="L317"></a><tt class="py-lineno">317</tt> <a class="py-toggle" href="#" id="Entry.remove_graphics-toggle" onclick="return toggle('Entry.remove_graphics');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Entry-class.html#remove_graphics">remove_graphics</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">entry</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Entry.remove_graphics-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Entry.remove_graphics-expanded"><a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">        <tt class="py-docstring">"""Remove the Graphics entry with the passed ID from the group."""</tt> </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">graphics</tt><tt class="py-op">.</tt><tt id="link-147" class="py-name"><a title="Bio.Crystal.Chain.remove
Bio.Graphics.BasicChromosome._ChromosomeComponent.remove
Bio.Restriction.Restriction.RestrictionBatch.remove
Bio.Seq.MutableSeq.remove
BioSQL.Loader.DatabaseRemover.remove" class="py-name" href="#" onclick="return doclink('link-147', 'remove', 'link-63');">remove</a></tt><tt class="py-op">(</tt><tt id="link-148" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-148', 'entry', 'link-12');">entry</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line"> </tt>
<a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line">    <tt class="py-comment"># Names may be given as a space-separated list of KEGG identifiers</tt> </tt>
<a name="Entry._getname"></a><div id="Entry._getname-def"><a name="L322"></a><tt class="py-lineno">322</tt> <a class="py-toggle" href="#" id="Entry._getname-toggle" onclick="return toggle('Entry._getname');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Entry-class.html#_getname">_getname</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Entry._getname-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Entry._getname-expanded"><a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">' '</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_names</tt><tt class="py-op">)</tt> </tt>
</div><a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line"> </tt>
<a name="Entry._setname"></a><div id="Entry._setname-def"><a name="L325"></a><tt class="py-lineno">325</tt> <a class="py-toggle" href="#" id="Entry._setname-toggle" onclick="return toggle('Entry._setname');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Entry-class.html#_setname">_setname</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Entry._setname-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Entry._setname-expanded"><a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_names</tt> <tt class="py-op">=</tt> <tt id="link-149" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-149', 'value', 'link-70');">value</a></tt><tt class="py-op">.</tt><tt id="link-150" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-150', 'split', 'link-150');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line"> </tt>
<a name="Entry._delname"></a><div id="Entry._delname-def"><a name="L328"></a><tt class="py-lineno">328</tt> <a class="py-toggle" href="#" id="Entry._delname-toggle" onclick="return toggle('Entry._delname');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Entry-class.html#_delname">_delname</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Entry._delname-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Entry._delname-expanded"><a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_names</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div><a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line">    <tt id="link-151" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-151', 'name', 'link-65');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-152" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-152', 'property', 'link-77');">property</a></tt><tt class="py-op">(</tt><tt id="link-153" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry._getname
Bio.KEGG.KGML.KGML_pathway.Pathway._getname" class="py-name" href="#" onclick="return doclink('link-153', '_getname', 'link-78');">_getname</a></tt><tt class="py-op">,</tt> <tt id="link-154" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry._setname
Bio.KEGG.KGML.KGML_pathway.Pathway._setname" class="py-name" href="#" onclick="return doclink('link-154', '_setname', 'link-79');">_setname</a></tt><tt class="py-op">,</tt> <tt id="link-155" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry._delname
Bio.KEGG.KGML.KGML_pathway.Pathway._delname" class="py-name" href="#" onclick="return doclink('link-155', '_delname', 'link-80');">_delname</a></tt><tt class="py-op">,</tt> </tt>
<a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line">                    <tt class="py-string">"List of KEGG identifiers for the Entry."</tt><tt class="py-op">)</tt> </tt>
<a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line"> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line">    <tt class="py-comment"># Reactions may be given as a space-separated list of KEGG identifiers</tt> </tt>
<a name="Entry._getreaction"></a><div id="Entry._getreaction-def"><a name="L334"></a><tt class="py-lineno">334</tt> <a class="py-toggle" href="#" id="Entry._getreaction-toggle" onclick="return toggle('Entry._getreaction');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Entry-class.html#_getreaction">_getreaction</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Entry._getreaction-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Entry._getreaction-expanded"><a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">' '</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_reactions</tt><tt class="py-op">)</tt> </tt>
</div><a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line"> </tt>
<a name="Entry._setreaction"></a><div id="Entry._setreaction-def"><a name="L337"></a><tt class="py-lineno">337</tt> <a class="py-toggle" href="#" id="Entry._setreaction-toggle" onclick="return toggle('Entry._setreaction');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Entry-class.html#_setreaction">_setreaction</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Entry._setreaction-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Entry._setreaction-expanded"><a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_reactions</tt> <tt class="py-op">=</tt> <tt id="link-156" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-156', 'value', 'link-70');">value</a></tt><tt class="py-op">.</tt><tt id="link-157" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-157', 'split', 'link-150');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line"> </tt>
<a name="Entry._delreaction"></a><div id="Entry._delreaction-def"><a name="L340"></a><tt class="py-lineno">340</tt> <a class="py-toggle" href="#" id="Entry._delreaction-toggle" onclick="return toggle('Entry._delreaction');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Entry-class.html#_delreaction">_delreaction</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Entry._delreaction-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Entry._delreaction-expanded"><a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_reactions</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div><a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line">    <tt id="link-158" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.reaction" class="py-name" href="#" onclick="return doclink('link-158', 'reaction', 'link-35');">reaction</a></tt> <tt class="py-op">=</tt> <tt id="link-159" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-159', 'property', 'link-77');">property</a></tt><tt class="py-op">(</tt><tt id="link-160" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Entry._getreaction()=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#_getreaction"><a title="Bio.KEGG.KGML.KGML_pathway.Entry._getreaction" class="py-name" href="#" onclick="return doclink('link-160', '_getreaction', 'link-160');">_getreaction</a></tt><tt class="py-op">,</tt> <tt id="link-161" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Entry._setreaction()=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#_setreaction"><a title="Bio.KEGG.KGML.KGML_pathway.Entry._setreaction" class="py-name" href="#" onclick="return doclink('link-161', '_setreaction', 'link-161');">_setreaction</a></tt><tt class="py-op">,</tt> <tt id="link-162" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Entry._delreaction()=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#_delreaction"><a title="Bio.KEGG.KGML.KGML_pathway.Entry._delreaction" class="py-name" href="#" onclick="return doclink('link-162', '_delreaction', 'link-162');">_delreaction</a></tt><tt class="py-op">,</tt> </tt>
<a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line">                        <tt class="py-string">"List of reaction KEGG IDs for this Entry."</tt><tt class="py-op">)</tt> </tt>
<a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line"> </tt>
<a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line">    <tt class="py-comment"># We make sure that the node ID is an integer</tt> </tt>
<a name="Entry._getid"></a><div id="Entry._getid-def"><a name="L346"></a><tt class="py-lineno">346</tt> <a class="py-toggle" href="#" id="Entry._getid-toggle" onclick="return toggle('Entry._getid');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Entry-class.html#_getid">_getid</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Entry._getid-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Entry._getid-expanded"><a name="L347"></a><tt class="py-lineno">347</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_id</tt> </tt>
</div><a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line"> </tt>
<a name="Entry._setid"></a><div id="Entry._setid-def"><a name="L349"></a><tt class="py-lineno">349</tt> <a class="py-toggle" href="#" id="Entry._setid-toggle" onclick="return toggle('Entry._setid');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Entry-class.html#_setid">_setid</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Entry._setid-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Entry._setid-expanded"><a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_id</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-163" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-163', 'value', 'link-70');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line"> </tt>
<a name="Entry._delid"></a><div id="Entry._delid-def"><a name="L352"></a><tt class="py-lineno">352</tt> <a class="py-toggle" href="#" id="Entry._delid-toggle" onclick="return toggle('Entry._delid');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Entry-class.html#_delid">_delid</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Entry._delid-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Entry._delid-expanded"><a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_id</tt> </tt>
</div><a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line">    <tt id="link-164" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-164', 'id', 'link-13');">id</a></tt> <tt class="py-op">=</tt> <tt id="link-165" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-165', 'property', 'link-77');">property</a></tt><tt class="py-op">(</tt><tt id="link-166" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Component._getid()=Bio.KEGG.KGML.KGML_pathway.Component-class.html#_getid,Method Bio.KEGG.KGML.KGML_pathway.Entry._getid()=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#_getid,Method Bio.KEGG.KGML.KGML_pathway.Reaction._getid()=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#_getid"><a title="Bio.KEGG.KGML.KGML_pathway.Component._getid
Bio.KEGG.KGML.KGML_pathway.Entry._getid
Bio.KEGG.KGML.KGML_pathway.Reaction._getid" class="py-name" href="#" onclick="return doclink('link-166', '_getid', 'link-166');">_getid</a></tt><tt class="py-op">,</tt> <tt id="link-167" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Component._setid()=Bio.KEGG.KGML.KGML_pathway.Component-class.html#_setid,Method Bio.KEGG.KGML.KGML_pathway.Entry._setid()=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#_setid,Method Bio.KEGG.KGML.KGML_pathway.Reaction._setid()=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#_setid"><a title="Bio.KEGG.KGML.KGML_pathway.Component._setid
Bio.KEGG.KGML.KGML_pathway.Entry._setid
Bio.KEGG.KGML.KGML_pathway.Reaction._setid" class="py-name" href="#" onclick="return doclink('link-167', '_setid', 'link-167');">_setid</a></tt><tt class="py-op">,</tt> <tt id="link-168" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Component._delid()=Bio.KEGG.KGML.KGML_pathway.Component-class.html#_delid,Method Bio.KEGG.KGML.KGML_pathway.Entry._delid()=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#_delid,Method Bio.KEGG.KGML.KGML_pathway.Reaction._delid()=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#_delid"><a title="Bio.KEGG.KGML.KGML_pathway.Component._delid
Bio.KEGG.KGML.KGML_pathway.Entry._delid
Bio.KEGG.KGML.KGML_pathway.Reaction._delid" class="py-name" href="#" onclick="return doclink('link-168', '_delid', 'link-168');">_delid</a></tt><tt class="py-op">,</tt> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">                  <tt class="py-string">"The pathway graph node ID for the Entry."</tt><tt class="py-op">)</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line"> </tt>
<a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Entry.element"></a><div id="Entry.element-def"><a name="L358"></a><tt class="py-lineno">358</tt> <a class="py-toggle" href="#" id="Entry.element-toggle" onclick="return toggle('Entry.element');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Entry-class.html#element">element</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Entry.element-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Entry.element-expanded"><a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the Entry as a valid KGML element."""</tt> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line">        <tt class="py-comment"># The root is this Entry element</tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line">        <tt id="link-169" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-169', 'entry', 'link-12');">entry</a></tt> <tt class="py-op">=</tt> <tt class="py-name">ET</tt><tt class="py-op">.</tt><tt class="py-name">Element</tt><tt class="py-op">(</tt><tt class="py-string">'entry'</tt><tt class="py-op">)</tt> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line">        <tt id="link-170" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-170', 'entry', 'link-12');">entry</a></tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'id'</tt><tt class="py-op">:</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_id</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line">                        <tt class="py-string">'name'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-171" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-171', 'name', 'link-65');">name</a></tt><tt class="py-op">,</tt> </tt>
<a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line">                        <tt class="py-string">'link'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-172" class="py-name"><a title="Bio.Nexus.Nodes.Chain.link" class="py-name" href="#" onclick="return doclink('link-172', 'link', 'link-5');">link</a></tt><tt class="py-op">,</tt> </tt>
<a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line">                        <tt class="py-string">'type'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-173" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-173', 'type', 'link-14');">type</a></tt> </tt>
<a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line">                        <tt class="py-op">}</tt> </tt>
<a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_reactions</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L368"></a><tt class="py-lineno">368</tt>  <tt class="py-line">            <tt id="link-174" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-174', 'entry', 'link-12');">entry</a></tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">[</tt><tt class="py-string">'reaction'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-175" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.reaction" class="py-name" href="#" onclick="return doclink('link-175', 'reaction', 'link-35');">reaction</a></tt> </tt>
<a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">graphics</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">g</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">graphics</tt><tt class="py-op">:</tt> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line">                <tt id="link-176" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-176', 'entry', 'link-12');">entry</a></tt><tt class="py-op">.</tt><tt id="link-177" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-177', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">g</tt><tt class="py-op">.</tt><tt id="link-178" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-178', 'element', 'link-10');">element</a></tt><tt class="py-op">)</tt> </tt>
<a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">components</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">c</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">components</tt><tt class="py-op">:</tt> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line">                <tt id="link-179" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-179', 'entry', 'link-12');">entry</a></tt><tt class="py-op">.</tt><tt id="link-180" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-180', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">.</tt><tt id="link-181" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-181', 'element', 'link-10');">element</a></tt><tt class="py-op">)</tt> </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-182" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-182', 'entry', 'link-12');">entry</a></tt> </tt>
</div><a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line"> </tt>
<a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Entry.bounds"></a><div id="Entry.bounds-def"><a name="L378"></a><tt class="py-lineno">378</tt> <a class="py-toggle" href="#" id="Entry.bounds-toggle" onclick="return toggle('Entry.bounds');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Entry-class.html#bounds">bounds</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Entry.bounds-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Entry.bounds-expanded"><a name="L379"></a><tt class="py-lineno">379</tt>  <tt class="py-line">        <tt class="py-docstring">"""Coordinate bounds for all Graphics elements in the Entry.</tt> </tt>
<a name="L380"></a><tt class="py-lineno">380</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L381"></a><tt class="py-lineno">381</tt>  <tt class="py-line"><tt class="py-docstring">        Return the [(xmin, ymin), (xmax, ymax)] co-ordinates for the Entry</tt> </tt>
<a name="L382"></a><tt class="py-lineno">382</tt>  <tt class="py-line"><tt class="py-docstring">        Graphics elements.</tt> </tt>
<a name="L383"></a><tt class="py-lineno">383</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L384"></a><tt class="py-lineno">384</tt>  <tt class="py-line">        <tt class="py-name">xlist</tt><tt class="py-op">,</tt> <tt class="py-name">ylist</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L385"></a><tt class="py-lineno">385</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-183" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-183', 'b', 'link-113');">b</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-name">g</tt><tt class="py-op">.</tt><tt id="link-184" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.bounds
Bio.KEGG.KGML.KGML_pathway.Graphics.bounds
Bio.KEGG.KGML.KGML_pathway.Pathway.bounds" class="py-name" href="#" onclick="return doclink('link-184', 'bounds', 'link-114');">bounds</a></tt> <tt class="py-keyword">for</tt> <tt class="py-name">g</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">graphics</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L386"></a><tt class="py-lineno">386</tt>  <tt class="py-line">            <tt class="py-name">xlist</tt><tt class="py-op">.</tt><tt id="link-185" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-185', 'extend', 'link-116');">extend</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt id="link-186" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-186', 'b', 'link-113');">b</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt id="link-187" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-187', 'b', 'link-113');">b</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L387"></a><tt class="py-lineno">387</tt>  <tt class="py-line">            <tt class="py-name">ylist</tt><tt class="py-op">.</tt><tt id="link-188" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-188', 'extend', 'link-116');">extend</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt id="link-189" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-189', 'b', 'link-113');">b</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt id="link-190" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.b" class="py-name" href="#" onclick="return doclink('link-190', 'b', 'link-113');">b</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L388"></a><tt class="py-lineno">388</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-op">(</tt><tt id="link-191" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-191', 'min', 'link-122');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">xlist</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-192" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-192', 'min', 'link-122');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">ylist</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L389"></a><tt class="py-lineno">389</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt id="link-193" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-193', 'max', 'link-124');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">xlist</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-194" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-194', 'max', 'link-124');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">ylist</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
</div><a name="L390"></a><tt class="py-lineno">390</tt>  <tt class="py-line"> </tt>
<a name="L391"></a><tt class="py-lineno">391</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Entry.is_reactant"></a><div id="Entry.is_reactant-def"><a name="L392"></a><tt class="py-lineno">392</tt> <a class="py-toggle" href="#" id="Entry.is_reactant-toggle" onclick="return toggle('Entry.is_reactant');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Entry-class.html#is_reactant">is_reactant</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Entry.is_reactant-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Entry.is_reactant-expanded"><a name="L393"></a><tt class="py-lineno">393</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return true if this Entry participates in any reaction in its parent pathway."""</tt> </tt>
<a name="L394"></a><tt class="py-lineno">394</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">rxn</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-195" class="py-name"><a title="Bio.KEGG.Compound.Record._pathway
Bio.KEGG.Enzyme.Record._pathway" class="py-name" href="#" onclick="return doclink('link-195', '_pathway', 'link-20');">_pathway</a></tt><tt class="py-op">.</tt><tt id="link-196" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Pathway.reactions=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#reactions,Method Bio.Pathway.System.reactions()=Bio.Pathway.System-class.html#reactions"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.reactions
Bio.Pathway.System.reactions" class="py-name" href="#" onclick="return doclink('link-196', 'reactions', 'link-196');">reactions</a></tt><tt class="py-op">:</tt> </tt>
<a name="L395"></a><tt class="py-lineno">395</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_id</tt> <tt class="py-keyword">in</tt> <tt class="py-name">rxn</tt><tt class="py-op">.</tt><tt id="link-197" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Reaction.reactant_ids=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#reactant_ids"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.reactant_ids" class="py-name" href="#" onclick="return doclink('link-197', 'reactant_ids', 'link-197');">reactant_ids</a></tt><tt class="py-op">:</tt> </tt>
<a name="L396"></a><tt class="py-lineno">396</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">True</tt> </tt>
<a name="L397"></a><tt class="py-lineno">397</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
</div></div><a name="L398"></a><tt class="py-lineno">398</tt>  <tt class="py-line"> </tt>
<a name="Component"></a><div id="Component-def"><a name="L399"></a><tt class="py-lineno">399</tt>  <tt class="py-line"> </tt>
<a name="L400"></a><tt class="py-lineno">400</tt>  <tt class="py-line"><tt class="py-comment"># Component</tt> </tt>
<a name="L401"></a><tt class="py-lineno">401</tt> <a class="py-toggle" href="#" id="Component-toggle" onclick="return toggle('Component');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Component-class.html">Component</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Component-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Component-expanded"><a name="L402"></a><tt class="py-lineno">402</tt>  <tt class="py-line">    <tt class="py-docstring">"""An Entry subelement used to represents a complex node.</tt> </tt>
<a name="L403"></a><tt class="py-lineno">403</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L404"></a><tt class="py-lineno">404</tt>  <tt class="py-line"><tt class="py-docstring">    A subelement of the Entry element, used when the Entry is a complex</tt> </tt>
<a name="L405"></a><tt class="py-lineno">405</tt>  <tt class="py-line"><tt class="py-docstring">    node, as described in release KGML v0.7.1</tt> </tt>
<a name="L406"></a><tt class="py-lineno">406</tt>  <tt class="py-line"><tt class="py-docstring">    (http://www.kegg.jp/kegg/xml/docs/)</tt> </tt>
<a name="L407"></a><tt class="py-lineno">407</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L408"></a><tt class="py-lineno">408</tt>  <tt class="py-line"><tt class="py-docstring">    The Component acts as a collection (with type 'group', and typically</tt> </tt>
<a name="L409"></a><tt class="py-lineno">409</tt>  <tt class="py-line"><tt class="py-docstring">    its own Graphics subelement), having only an ID.</tt> </tt>
<a name="L410"></a><tt class="py-lineno">410</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L411"></a><tt class="py-lineno">411</tt>  <tt class="py-line"> </tt>
<a name="Component.__init__"></a><div id="Component.__init__-def"><a name="L412"></a><tt class="py-lineno">412</tt> <a class="py-toggle" href="#" id="Component.__init__-toggle" onclick="return toggle('Component.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Component-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">parent</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Component.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Component.__init__-expanded"><a name="L413"></a><tt class="py-lineno">413</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the class."""</tt> </tt>
<a name="L414"></a><tt class="py-lineno">414</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_id</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L415"></a><tt class="py-lineno">415</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_parent</tt> <tt class="py-op">=</tt> <tt class="py-name">parent</tt> </tt>
</div><a name="L416"></a><tt class="py-lineno">416</tt>  <tt class="py-line"> </tt>
<a name="L417"></a><tt class="py-lineno">417</tt>  <tt class="py-line">    <tt class="py-comment"># We make sure that the node ID is an integer</tt> </tt>
<a name="Component._getid"></a><div id="Component._getid-def"><a name="L418"></a><tt class="py-lineno">418</tt> <a class="py-toggle" href="#" id="Component._getid-toggle" onclick="return toggle('Component._getid');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Component-class.html#_getid">_getid</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Component._getid-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Component._getid-expanded"><a name="L419"></a><tt class="py-lineno">419</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_id</tt> </tt>
</div><a name="L420"></a><tt class="py-lineno">420</tt>  <tt class="py-line"> </tt>
<a name="Component._setid"></a><div id="Component._setid-def"><a name="L421"></a><tt class="py-lineno">421</tt> <a class="py-toggle" href="#" id="Component._setid-toggle" onclick="return toggle('Component._setid');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Component-class.html#_setid">_setid</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Component._setid-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Component._setid-expanded"><a name="L422"></a><tt class="py-lineno">422</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_id</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-198" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-198', 'value', 'link-70');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L423"></a><tt class="py-lineno">423</tt>  <tt class="py-line"> </tt>
<a name="Component._delid"></a><div id="Component._delid-def"><a name="L424"></a><tt class="py-lineno">424</tt> <a class="py-toggle" href="#" id="Component._delid-toggle" onclick="return toggle('Component._delid');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Component-class.html#_delid">_delid</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Component._delid-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Component._delid-expanded"><a name="L425"></a><tt class="py-lineno">425</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_id</tt> </tt>
</div><a name="L426"></a><tt class="py-lineno">426</tt>  <tt class="py-line"> </tt>
<a name="L427"></a><tt class="py-lineno">427</tt>  <tt class="py-line">    <tt id="link-199" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-199', 'id', 'link-13');">id</a></tt> <tt class="py-op">=</tt> <tt id="link-200" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-200', 'property', 'link-77');">property</a></tt><tt class="py-op">(</tt><tt id="link-201" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component._getid
Bio.KEGG.KGML.KGML_pathway.Entry._getid
Bio.KEGG.KGML.KGML_pathway.Reaction._getid" class="py-name" href="#" onclick="return doclink('link-201', '_getid', 'link-166');">_getid</a></tt><tt class="py-op">,</tt> <tt id="link-202" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component._setid
Bio.KEGG.KGML.KGML_pathway.Entry._setid
Bio.KEGG.KGML.KGML_pathway.Reaction._setid" class="py-name" href="#" onclick="return doclink('link-202', '_setid', 'link-167');">_setid</a></tt><tt class="py-op">,</tt> <tt id="link-203" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component._delid
Bio.KEGG.KGML.KGML_pathway.Entry._delid
Bio.KEGG.KGML.KGML_pathway.Reaction._delid" class="py-name" href="#" onclick="return doclink('link-203', '_delid', 'link-168');">_delid</a></tt><tt class="py-op">,</tt> </tt>
<a name="L428"></a><tt class="py-lineno">428</tt>  <tt class="py-line">                  <tt class="py-string">"The pathway graph node ID for the Entry"</tt><tt class="py-op">)</tt> </tt>
<a name="L429"></a><tt class="py-lineno">429</tt>  <tt class="py-line"> </tt>
<a name="L430"></a><tt class="py-lineno">430</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Component.element"></a><div id="Component.element-def"><a name="L431"></a><tt class="py-lineno">431</tt> <a class="py-toggle" href="#" id="Component.element-toggle" onclick="return toggle('Component.element');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Component-class.html#element">element</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Component.element-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Component.element-expanded"><a name="L432"></a><tt class="py-lineno">432</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the Component as a valid KGML element."""</tt> </tt>
<a name="L433"></a><tt class="py-lineno">433</tt>  <tt class="py-line">        <tt class="py-comment"># The root is this Component element</tt> </tt>
<a name="L434"></a><tt class="py-lineno">434</tt>  <tt class="py-line">        <tt class="py-name">component</tt> <tt class="py-op">=</tt> <tt class="py-name">ET</tt><tt class="py-op">.</tt><tt class="py-name">Element</tt><tt class="py-op">(</tt><tt class="py-string">'component'</tt><tt class="py-op">)</tt> </tt>
<a name="L435"></a><tt class="py-lineno">435</tt>  <tt class="py-line">        <tt class="py-name">component</tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'id'</tt><tt class="py-op">:</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_id</tt><tt class="py-op">)</tt><tt class="py-op">}</tt> </tt>
<a name="L436"></a><tt class="py-lineno">436</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">component</tt> </tt>
</div></div><a name="L437"></a><tt class="py-lineno">437</tt>  <tt class="py-line"> </tt>
<a name="Graphics"></a><div id="Graphics-def"><a name="L438"></a><tt class="py-lineno">438</tt>  <tt class="py-line"> </tt>
<a name="L439"></a><tt class="py-lineno">439</tt>  <tt class="py-line"><tt class="py-comment"># Graphics</tt> </tt>
<a name="L440"></a><tt class="py-lineno">440</tt> <a class="py-toggle" href="#" id="Graphics-toggle" onclick="return toggle('Graphics');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html">Graphics</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Graphics-expanded"><a name="L441"></a><tt class="py-lineno">441</tt>  <tt class="py-line">    <tt class="py-docstring">"""An Entry subelement used to represents the visual representation.</tt> </tt>
<a name="L442"></a><tt class="py-lineno">442</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L443"></a><tt class="py-lineno">443</tt>  <tt class="py-line"><tt class="py-docstring">    A subelement of Entry, specifying its visual representation, as</tt> </tt>
<a name="L444"></a><tt class="py-lineno">444</tt>  <tt class="py-line"><tt class="py-docstring">    described in release KGML v0.7.1 (http://www.kegg.jp/kegg/xml/docs/)</tt> </tt>
<a name="L445"></a><tt class="py-lineno">445</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L446"></a><tt class="py-lineno">446</tt>  <tt class="py-line"><tt class="py-docstring">    Attributes:</tt> </tt>
<a name="L447"></a><tt class="py-lineno">447</tt>  <tt class="py-line"><tt class="py-docstring">     - name         Label for the graphics object</tt> </tt>
<a name="L448"></a><tt class="py-lineno">448</tt>  <tt class="py-line"><tt class="py-docstring">     - x            X-axis position of the object (int)</tt> </tt>
<a name="L449"></a><tt class="py-lineno">449</tt>  <tt class="py-line"><tt class="py-docstring">     - y            Y-axis position of the object (int)</tt> </tt>
<a name="L450"></a><tt class="py-lineno">450</tt>  <tt class="py-line"><tt class="py-docstring">     - coords       polyline co-ordinates, list of (int, int) tuples</tt> </tt>
<a name="L451"></a><tt class="py-lineno">451</tt>  <tt class="py-line"><tt class="py-docstring">     - type         object shape</tt> </tt>
<a name="L452"></a><tt class="py-lineno">452</tt>  <tt class="py-line"><tt class="py-docstring">     - width        object width (int)</tt> </tt>
<a name="L453"></a><tt class="py-lineno">453</tt>  <tt class="py-line"><tt class="py-docstring">     - height       object height (int)</tt> </tt>
<a name="L454"></a><tt class="py-lineno">454</tt>  <tt class="py-line"><tt class="py-docstring">     - fgcolor      object foreground color (hex RGB)</tt> </tt>
<a name="L455"></a><tt class="py-lineno">455</tt>  <tt class="py-line"><tt class="py-docstring">     - bgcolor      object background color (hex RGB)</tt> </tt>
<a name="L456"></a><tt class="py-lineno">456</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L457"></a><tt class="py-lineno">457</tt>  <tt class="py-line"><tt class="py-docstring">    Some attributes are present only for specific graphics types.  For</tt> </tt>
<a name="L458"></a><tt class="py-lineno">458</tt>  <tt class="py-line"><tt class="py-docstring">    example, line types do not (typically) have a width.</tt> </tt>
<a name="L459"></a><tt class="py-lineno">459</tt>  <tt class="py-line"><tt class="py-docstring">    We permit non-DTD attributes and attribute settings, such as</tt> </tt>
<a name="L460"></a><tt class="py-lineno">460</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L461"></a><tt class="py-lineno">461</tt>  <tt class="py-line"><tt class="py-docstring">    dash         List of ints, describing an on/off pattern for dashes</tt> </tt>
<a name="L462"></a><tt class="py-lineno">462</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L463"></a><tt class="py-lineno">463</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L464"></a><tt class="py-lineno">464</tt>  <tt class="py-line"> </tt>
<a name="Graphics.__init__"></a><div id="Graphics.__init__-def"><a name="L465"></a><tt class="py-lineno">465</tt> <a class="py-toggle" href="#" id="Graphics.__init__-toggle" onclick="return toggle('Graphics.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">parent</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics.__init__-expanded"><a name="L466"></a><tt class="py-lineno">466</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the class."""</tt> </tt>
<a name="L467"></a><tt class="py-lineno">467</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-204" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-204', 'name', 'link-65');">name</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L468"></a><tt class="py-lineno">468</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_x</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L469"></a><tt class="py-lineno">469</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_y</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L470"></a><tt class="py-lineno">470</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_coords</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L471"></a><tt class="py-lineno">471</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-205" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-205', 'type', 'link-14');">type</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L472"></a><tt class="py-lineno">472</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_width</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L473"></a><tt class="py-lineno">473</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_height</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L474"></a><tt class="py-lineno">474</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-206" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.fgcolor=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#fgcolor"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.fgcolor" class="py-name" href="#" onclick="return doclink('link-206', 'fgcolor', 'link-206');">fgcolor</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L475"></a><tt class="py-lineno">475</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-207" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.bgcolor=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#bgcolor"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.bgcolor" class="py-name" href="#" onclick="return doclink('link-207', 'bgcolor', 'link-207');">bgcolor</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L476"></a><tt class="py-lineno">476</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_parent</tt> <tt class="py-op">=</tt> <tt class="py-name">parent</tt> </tt>
</div><a name="L477"></a><tt class="py-lineno">477</tt>  <tt class="py-line"> </tt>
<a name="L478"></a><tt class="py-lineno">478</tt>  <tt class="py-line">    <tt class="py-comment"># We make sure that the XY coordinates, width and height are numbers</tt> </tt>
<a name="Graphics._getx"></a><div id="Graphics._getx-def"><a name="L479"></a><tt class="py-lineno">479</tt> <a class="py-toggle" href="#" id="Graphics._getx-toggle" onclick="return toggle('Graphics._getx');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_getx">_getx</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics._getx-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics._getx-expanded"><a name="L480"></a><tt class="py-lineno">480</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_x</tt> </tt>
</div><a name="L481"></a><tt class="py-lineno">481</tt>  <tt class="py-line"> </tt>
<a name="Graphics._setx"></a><div id="Graphics._setx-def"><a name="L482"></a><tt class="py-lineno">482</tt> <a class="py-toggle" href="#" id="Graphics._setx-toggle" onclick="return toggle('Graphics._setx');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_setx">_setx</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics._setx-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics._setx-expanded"><a name="L483"></a><tt class="py-lineno">483</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_x</tt> <tt class="py-op">=</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt id="link-208" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-208', 'value', 'link-70');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L484"></a><tt class="py-lineno">484</tt>  <tt class="py-line"> </tt>
<a name="Graphics._delx"></a><div id="Graphics._delx-def"><a name="L485"></a><tt class="py-lineno">485</tt> <a class="py-toggle" href="#" id="Graphics._delx-toggle" onclick="return toggle('Graphics._delx');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_delx">_delx</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics._delx-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics._delx-expanded"><a name="L486"></a><tt class="py-lineno">486</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_x</tt> </tt>
</div><a name="L487"></a><tt class="py-lineno">487</tt>  <tt class="py-line">    <tt id="link-209" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.x=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#x,Variable Bio.Statistics.lowess.x=Bio.Statistics.lowess-module.html#x"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-209', 'x', 'link-209');">x</a></tt> <tt class="py-op">=</tt> <tt id="link-210" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-210', 'property', 'link-77');">property</a></tt><tt class="py-op">(</tt><tt id="link-211" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Graphics._getx()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_getx"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics._getx" class="py-name" href="#" onclick="return doclink('link-211', '_getx', 'link-211');">_getx</a></tt><tt class="py-op">,</tt> <tt id="link-212" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Graphics._setx()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_setx"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics._setx" class="py-name" href="#" onclick="return doclink('link-212', '_setx', 'link-212');">_setx</a></tt><tt class="py-op">,</tt> <tt id="link-213" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Graphics._delx()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_delx"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics._delx" class="py-name" href="#" onclick="return doclink('link-213', '_delx', 'link-213');">_delx</a></tt><tt class="py-op">,</tt> </tt>
<a name="L488"></a><tt class="py-lineno">488</tt>  <tt class="py-line">                 <tt class="py-string">"The X coordinate for the graphics element."</tt><tt class="py-op">)</tt> </tt>
<a name="L489"></a><tt class="py-lineno">489</tt>  <tt class="py-line"> </tt>
<a name="Graphics._gety"></a><div id="Graphics._gety-def"><a name="L490"></a><tt class="py-lineno">490</tt> <a class="py-toggle" href="#" id="Graphics._gety-toggle" onclick="return toggle('Graphics._gety');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_gety">_gety</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics._gety-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics._gety-expanded"><a name="L491"></a><tt class="py-lineno">491</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_y</tt> </tt>
</div><a name="L492"></a><tt class="py-lineno">492</tt>  <tt class="py-line"> </tt>
<a name="Graphics._sety"></a><div id="Graphics._sety-def"><a name="L493"></a><tt class="py-lineno">493</tt> <a class="py-toggle" href="#" id="Graphics._sety-toggle" onclick="return toggle('Graphics._sety');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_sety">_sety</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics._sety-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics._sety-expanded"><a name="L494"></a><tt class="py-lineno">494</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_y</tt> <tt class="py-op">=</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt id="link-214" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-214', 'value', 'link-70');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L495"></a><tt class="py-lineno">495</tt>  <tt class="py-line"> </tt>
<a name="Graphics._dely"></a><div id="Graphics._dely-def"><a name="L496"></a><tt class="py-lineno">496</tt> <a class="py-toggle" href="#" id="Graphics._dely-toggle" onclick="return toggle('Graphics._dely');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_dely">_dely</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics._dely-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics._dely-expanded"><a name="L497"></a><tt class="py-lineno">497</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_y</tt> </tt>
</div><a name="L498"></a><tt class="py-lineno">498</tt>  <tt class="py-line">    <tt id="link-215" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.y=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#y"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-215', 'y', 'link-215');">y</a></tt> <tt class="py-op">=</tt> <tt id="link-216" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-216', 'property', 'link-77');">property</a></tt><tt class="py-op">(</tt><tt id="link-217" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Graphics._gety()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_gety"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics._gety" class="py-name" href="#" onclick="return doclink('link-217', '_gety', 'link-217');">_gety</a></tt><tt class="py-op">,</tt> <tt id="link-218" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Graphics._sety()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_sety"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics._sety" class="py-name" href="#" onclick="return doclink('link-218', '_sety', 'link-218');">_sety</a></tt><tt class="py-op">,</tt> <tt id="link-219" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Graphics._dely()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_dely"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics._dely" class="py-name" href="#" onclick="return doclink('link-219', '_dely', 'link-219');">_dely</a></tt><tt class="py-op">,</tt> </tt>
<a name="L499"></a><tt class="py-lineno">499</tt>  <tt class="py-line">                 <tt class="py-string">"The Y coordinate for the graphics element."</tt><tt class="py-op">)</tt> </tt>
<a name="L500"></a><tt class="py-lineno">500</tt>  <tt class="py-line"> </tt>
<a name="Graphics._getwidth"></a><div id="Graphics._getwidth-def"><a name="L501"></a><tt class="py-lineno">501</tt> <a class="py-toggle" href="#" id="Graphics._getwidth-toggle" onclick="return toggle('Graphics._getwidth');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_getwidth">_getwidth</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics._getwidth-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics._getwidth-expanded"><a name="L502"></a><tt class="py-lineno">502</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_width</tt> </tt>
</div><a name="L503"></a><tt class="py-lineno">503</tt>  <tt class="py-line"> </tt>
<a name="Graphics._setwidth"></a><div id="Graphics._setwidth-def"><a name="L504"></a><tt class="py-lineno">504</tt> <a class="py-toggle" href="#" id="Graphics._setwidth-toggle" onclick="return toggle('Graphics._setwidth');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_setwidth">_setwidth</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics._setwidth-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics._setwidth-expanded"><a name="L505"></a><tt class="py-lineno">505</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_width</tt> <tt class="py-op">=</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt id="link-220" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-220', 'value', 'link-70');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L506"></a><tt class="py-lineno">506</tt>  <tt class="py-line"> </tt>
<a name="Graphics._delwidth"></a><div id="Graphics._delwidth-def"><a name="L507"></a><tt class="py-lineno">507</tt> <a class="py-toggle" href="#" id="Graphics._delwidth-toggle" onclick="return toggle('Graphics._delwidth');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_delwidth">_delwidth</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics._delwidth-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics._delwidth-expanded"><a name="L508"></a><tt class="py-lineno">508</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_width</tt> </tt>
</div><a name="L509"></a><tt class="py-lineno">509</tt>  <tt class="py-line">    <tt id="link-221" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.width=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#width,Method Bio.Phylo.PhyloXMLIO.Writer.width()=Bio.Phylo.PhyloXMLIO.Writer-class.html#width"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.width
Bio.Phylo.PhyloXMLIO.Writer.width" class="py-name" href="#" onclick="return doclink('link-221', 'width', 'link-221');">width</a></tt> <tt class="py-op">=</tt> <tt id="link-222" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-222', 'property', 'link-77');">property</a></tt><tt class="py-op">(</tt><tt id="link-223" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Graphics._getwidth()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_getwidth"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics._getwidth" class="py-name" href="#" onclick="return doclink('link-223', '_getwidth', 'link-223');">_getwidth</a></tt><tt class="py-op">,</tt> <tt id="link-224" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Graphics._setwidth()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_setwidth"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics._setwidth" class="py-name" href="#" onclick="return doclink('link-224', '_setwidth', 'link-224');">_setwidth</a></tt><tt class="py-op">,</tt> <tt id="link-225" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Graphics._delwidth()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_delwidth"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics._delwidth" class="py-name" href="#" onclick="return doclink('link-225', '_delwidth', 'link-225');">_delwidth</a></tt><tt class="py-op">,</tt> </tt>
<a name="L510"></a><tt class="py-lineno">510</tt>  <tt class="py-line">                     <tt class="py-string">"The width of the graphics element."</tt><tt class="py-op">)</tt> </tt>
<a name="L511"></a><tt class="py-lineno">511</tt>  <tt class="py-line"> </tt>
<a name="Graphics._getheight"></a><div id="Graphics._getheight-def"><a name="L512"></a><tt class="py-lineno">512</tt> <a class="py-toggle" href="#" id="Graphics._getheight-toggle" onclick="return toggle('Graphics._getheight');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_getheight">_getheight</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics._getheight-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics._getheight-expanded"><a name="L513"></a><tt class="py-lineno">513</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_height</tt> </tt>
</div><a name="L514"></a><tt class="py-lineno">514</tt>  <tt class="py-line"> </tt>
<a name="Graphics._setheight"></a><div id="Graphics._setheight-def"><a name="L515"></a><tt class="py-lineno">515</tt> <a class="py-toggle" href="#" id="Graphics._setheight-toggle" onclick="return toggle('Graphics._setheight');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_setheight">_setheight</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics._setheight-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics._setheight-expanded"><a name="L516"></a><tt class="py-lineno">516</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_height</tt> <tt class="py-op">=</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt id="link-226" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-226', 'value', 'link-70');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L517"></a><tt class="py-lineno">517</tt>  <tt class="py-line"> </tt>
<a name="Graphics._delheight"></a><div id="Graphics._delheight-def"><a name="L518"></a><tt class="py-lineno">518</tt> <a class="py-toggle" href="#" id="Graphics._delheight-toggle" onclick="return toggle('Graphics._delheight');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_delheight">_delheight</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics._delheight-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics._delheight-expanded"><a name="L519"></a><tt class="py-lineno">519</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_height</tt> </tt>
</div><a name="L520"></a><tt class="py-lineno">520</tt>  <tt class="py-line">    <tt id="link-227" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.height=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#height"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.height" class="py-name" href="#" onclick="return doclink('link-227', 'height', 'link-227');">height</a></tt> <tt class="py-op">=</tt> <tt id="link-228" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-228', 'property', 'link-77');">property</a></tt><tt class="py-op">(</tt><tt id="link-229" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Graphics._getheight()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_getheight"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics._getheight" class="py-name" href="#" onclick="return doclink('link-229', '_getheight', 'link-229');">_getheight</a></tt><tt class="py-op">,</tt> <tt id="link-230" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Graphics._setheight()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_setheight"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics._setheight" class="py-name" href="#" onclick="return doclink('link-230', '_setheight', 'link-230');">_setheight</a></tt><tt class="py-op">,</tt> <tt id="link-231" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Graphics._delheight()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_delheight"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics._delheight" class="py-name" href="#" onclick="return doclink('link-231', '_delheight', 'link-231');">_delheight</a></tt><tt class="py-op">,</tt> </tt>
<a name="L521"></a><tt class="py-lineno">521</tt>  <tt class="py-line">                      <tt class="py-string">"The height of the graphics element."</tt><tt class="py-op">)</tt> </tt>
<a name="L522"></a><tt class="py-lineno">522</tt>  <tt class="py-line"> </tt>
<a name="L523"></a><tt class="py-lineno">523</tt>  <tt class="py-line">    <tt class="py-comment"># We make sure that the polyline co-ordinates are integers, too</tt> </tt>
<a name="Graphics._getcoords"></a><div id="Graphics._getcoords-def"><a name="L524"></a><tt class="py-lineno">524</tt> <a class="py-toggle" href="#" id="Graphics._getcoords-toggle" onclick="return toggle('Graphics._getcoords');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_getcoords">_getcoords</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics._getcoords-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics._getcoords-expanded"><a name="L525"></a><tt class="py-lineno">525</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_coords</tt> </tt>
</div><a name="L526"></a><tt class="py-lineno">526</tt>  <tt class="py-line"> </tt>
<a name="Graphics._setcoords"></a><div id="Graphics._setcoords-def"><a name="L527"></a><tt class="py-lineno">527</tt> <a class="py-toggle" href="#" id="Graphics._setcoords-toggle" onclick="return toggle('Graphics._setcoords');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_setcoords">_setcoords</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics._setcoords-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics._setcoords-expanded"><a name="L528"></a><tt class="py-lineno">528</tt>  <tt class="py-line">        <tt class="py-name">clist</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">e</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">e</tt> <tt class="py-keyword">in</tt> <tt id="link-232" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-232', 'value', 'link-70');">value</a></tt><tt class="py-op">.</tt><tt id="link-233" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-233', 'split', 'link-150');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">','</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L529"></a><tt class="py-lineno">529</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_coords</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">tuple</tt><tt class="py-op">(</tt><tt class="py-name">clist</tt><tt class="py-op">[</tt><tt id="link-234" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-234', 'i', 'link-96');">i</a></tt><tt class="py-op">:</tt><tt id="link-235" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-235', 'i', 'link-96');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-236" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-236', 'i', 'link-96');">i</a></tt> <tt class="py-keyword">in</tt> <tt id="link-237" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-237', 'range', 'link-237');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">clist</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">2</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
</div><a name="L530"></a><tt class="py-lineno">530</tt>  <tt class="py-line"> </tt>
<a name="Graphics._delcoords"></a><div id="Graphics._delcoords-def"><a name="L531"></a><tt class="py-lineno">531</tt> <a class="py-toggle" href="#" id="Graphics._delcoords-toggle" onclick="return toggle('Graphics._delcoords');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_delcoords">_delcoords</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics._delcoords-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics._delcoords-expanded"><a name="L532"></a><tt class="py-lineno">532</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_coords</tt> </tt>
</div><a name="L533"></a><tt class="py-lineno">533</tt>  <tt class="py-line">    <tt id="link-238" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.coords=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#coords"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.coords" class="py-name" href="#" onclick="return doclink('link-238', 'coords', 'link-238');">coords</a></tt> <tt class="py-op">=</tt> <tt id="link-239" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-239', 'property', 'link-77');">property</a></tt><tt class="py-op">(</tt><tt id="link-240" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Graphics._getcoords()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_getcoords"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics._getcoords" class="py-name" href="#" onclick="return doclink('link-240', '_getcoords', 'link-240');">_getcoords</a></tt><tt class="py-op">,</tt> <tt id="link-241" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Graphics._setcoords()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_setcoords"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics._setcoords" class="py-name" href="#" onclick="return doclink('link-241', '_setcoords', 'link-241');">_setcoords</a></tt><tt class="py-op">,</tt> <tt id="link-242" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Graphics._delcoords()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_delcoords"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics._delcoords" class="py-name" href="#" onclick="return doclink('link-242', '_delcoords', 'link-242');">_delcoords</a></tt><tt class="py-op">,</tt> </tt>
<a name="L534"></a><tt class="py-lineno">534</tt>  <tt class="py-line">                      <tt class="py-string">"Polyline coordinates for the graphics element."</tt><tt class="py-op">)</tt> </tt>
<a name="L535"></a><tt class="py-lineno">535</tt>  <tt class="py-line"> </tt>
<a name="L536"></a><tt class="py-lineno">536</tt>  <tt class="py-line">    <tt class="py-comment"># Set default colors</tt> </tt>
<a name="Graphics._getfgcolor"></a><div id="Graphics._getfgcolor-def"><a name="L537"></a><tt class="py-lineno">537</tt> <a class="py-toggle" href="#" id="Graphics._getfgcolor-toggle" onclick="return toggle('Graphics._getfgcolor');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_getfgcolor">_getfgcolor</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics._getfgcolor-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics._getfgcolor-expanded"><a name="L538"></a><tt class="py-lineno">538</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_fgcolor</tt> </tt>
</div><a name="L539"></a><tt class="py-lineno">539</tt>  <tt class="py-line"> </tt>
<a name="Graphics._setfgcolor"></a><div id="Graphics._setfgcolor-def"><a name="L540"></a><tt class="py-lineno">540</tt> <a class="py-toggle" href="#" id="Graphics._setfgcolor-toggle" onclick="return toggle('Graphics._setfgcolor');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_setfgcolor">_setfgcolor</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics._setfgcolor-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics._setfgcolor-expanded"><a name="L541"></a><tt class="py-lineno">541</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-243" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-243', 'value', 'link-70');">value</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'none'</tt><tt class="py-op">:</tt> </tt>
<a name="L542"></a><tt class="py-lineno">542</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_fgcolor</tt> <tt class="py-op">=</tt> <tt class="py-string">'#000000'</tt>  <tt class="py-comment"># this default defined in KGML spec</tt> </tt>
<a name="L543"></a><tt class="py-lineno">543</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L544"></a><tt class="py-lineno">544</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_fgcolor</tt> <tt class="py-op">=</tt> <tt id="link-244" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-244', 'value', 'link-70');">value</a></tt> </tt>
</div><a name="L545"></a><tt class="py-lineno">545</tt>  <tt class="py-line"> </tt>
<a name="Graphics._delfgcolor"></a><div id="Graphics._delfgcolor-def"><a name="L546"></a><tt class="py-lineno">546</tt> <a class="py-toggle" href="#" id="Graphics._delfgcolor-toggle" onclick="return toggle('Graphics._delfgcolor');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_delfgcolor">_delfgcolor</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics._delfgcolor-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics._delfgcolor-expanded"><a name="L547"></a><tt class="py-lineno">547</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_fgcolor</tt> </tt>
</div><a name="L548"></a><tt class="py-lineno">548</tt>  <tt class="py-line">    <tt id="link-245" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.fgcolor" class="py-name" href="#" onclick="return doclink('link-245', 'fgcolor', 'link-206');">fgcolor</a></tt> <tt class="py-op">=</tt> <tt id="link-246" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-246', 'property', 'link-77');">property</a></tt><tt class="py-op">(</tt><tt id="link-247" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Graphics._getfgcolor()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_getfgcolor"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics._getfgcolor" class="py-name" href="#" onclick="return doclink('link-247', '_getfgcolor', 'link-247');">_getfgcolor</a></tt><tt class="py-op">,</tt> <tt id="link-248" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Graphics._setfgcolor()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_setfgcolor"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics._setfgcolor" class="py-name" href="#" onclick="return doclink('link-248', '_setfgcolor', 'link-248');">_setfgcolor</a></tt><tt class="py-op">,</tt> <tt id="link-249" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Graphics._delfgcolor()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_delfgcolor"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics._delfgcolor" class="py-name" href="#" onclick="return doclink('link-249', '_delfgcolor', 'link-249');">_delfgcolor</a></tt><tt class="py-op">,</tt> </tt>
<a name="L549"></a><tt class="py-lineno">549</tt>  <tt class="py-line">                       <tt class="py-string">"Foreground color."</tt><tt class="py-op">)</tt> </tt>
<a name="L550"></a><tt class="py-lineno">550</tt>  <tt class="py-line"> </tt>
<a name="Graphics._getbgcolor"></a><div id="Graphics._getbgcolor-def"><a name="L551"></a><tt class="py-lineno">551</tt> <a class="py-toggle" href="#" id="Graphics._getbgcolor-toggle" onclick="return toggle('Graphics._getbgcolor');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_getbgcolor">_getbgcolor</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics._getbgcolor-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics._getbgcolor-expanded"><a name="L552"></a><tt class="py-lineno">552</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_bgcolor</tt> </tt>
</div><a name="L553"></a><tt class="py-lineno">553</tt>  <tt class="py-line"> </tt>
<a name="Graphics._setbgcolor"></a><div id="Graphics._setbgcolor-def"><a name="L554"></a><tt class="py-lineno">554</tt> <a class="py-toggle" href="#" id="Graphics._setbgcolor-toggle" onclick="return toggle('Graphics._setbgcolor');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_setbgcolor">_setbgcolor</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics._setbgcolor-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics._setbgcolor-expanded"><a name="L555"></a><tt class="py-lineno">555</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-250" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-250', 'value', 'link-70');">value</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'none'</tt><tt class="py-op">:</tt> </tt>
<a name="L556"></a><tt class="py-lineno">556</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_bgcolor</tt> <tt class="py-op">=</tt> <tt class="py-string">'#000000'</tt>  <tt class="py-comment"># this default defined in KGML spec</tt> </tt>
<a name="L557"></a><tt class="py-lineno">557</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L558"></a><tt class="py-lineno">558</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_bgcolor</tt> <tt class="py-op">=</tt> <tt id="link-251" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-251', 'value', 'link-70');">value</a></tt> </tt>
</div><a name="L559"></a><tt class="py-lineno">559</tt>  <tt class="py-line"> </tt>
<a name="Graphics._delbgcolor"></a><div id="Graphics._delbgcolor-def"><a name="L560"></a><tt class="py-lineno">560</tt> <a class="py-toggle" href="#" id="Graphics._delbgcolor-toggle" onclick="return toggle('Graphics._delbgcolor');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_delbgcolor">_delbgcolor</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics._delbgcolor-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics._delbgcolor-expanded"><a name="L561"></a><tt class="py-lineno">561</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_bgcolor</tt> </tt>
</div><a name="L562"></a><tt class="py-lineno">562</tt>  <tt class="py-line">    <tt id="link-252" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.bgcolor" class="py-name" href="#" onclick="return doclink('link-252', 'bgcolor', 'link-207');">bgcolor</a></tt> <tt class="py-op">=</tt> <tt id="link-253" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-253', 'property', 'link-77');">property</a></tt><tt class="py-op">(</tt><tt id="link-254" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Graphics._getbgcolor()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_getbgcolor"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics._getbgcolor" class="py-name" href="#" onclick="return doclink('link-254', '_getbgcolor', 'link-254');">_getbgcolor</a></tt><tt class="py-op">,</tt> <tt id="link-255" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Graphics._setbgcolor()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_setbgcolor"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics._setbgcolor" class="py-name" href="#" onclick="return doclink('link-255', '_setbgcolor', 'link-255');">_setbgcolor</a></tt><tt class="py-op">,</tt> <tt id="link-256" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Graphics._delbgcolor()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#_delbgcolor"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics._delbgcolor" class="py-name" href="#" onclick="return doclink('link-256', '_delbgcolor', 'link-256');">_delbgcolor</a></tt><tt class="py-op">,</tt> </tt>
<a name="L563"></a><tt class="py-lineno">563</tt>  <tt class="py-line">                       <tt class="py-string">"Background color."</tt><tt class="py-op">)</tt> </tt>
<a name="L564"></a><tt class="py-lineno">564</tt>  <tt class="py-line"> </tt>
<a name="L565"></a><tt class="py-lineno">565</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Graphics.element"></a><div id="Graphics.element-def"><a name="L566"></a><tt class="py-lineno">566</tt> <a class="py-toggle" href="#" id="Graphics.element-toggle" onclick="return toggle('Graphics.element');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#element">element</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics.element-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics.element-expanded"><a name="L567"></a><tt class="py-lineno">567</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the Graphics as a valid KGML element."""</tt> </tt>
<a name="L568"></a><tt class="py-lineno">568</tt>  <tt class="py-line">        <tt class="py-comment"># The root is this Component element</tt> </tt>
<a name="L569"></a><tt class="py-lineno">569</tt>  <tt class="py-line">        <tt class="py-name">graphics</tt> <tt class="py-op">=</tt> <tt class="py-name">ET</tt><tt class="py-op">.</tt><tt class="py-name">Element</tt><tt class="py-op">(</tt><tt class="py-string">'graphics'</tt><tt class="py-op">)</tt> </tt>
<a name="L570"></a><tt class="py-lineno">570</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-257" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.fgcolor" class="py-name" href="#" onclick="return doclink('link-257', 'fgcolor', 'link-206');">fgcolor</a></tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># Assumes that string is hexstring</tt> </tt>
<a name="L571"></a><tt class="py-lineno">571</tt>  <tt class="py-line">            <tt class="py-name">fghex</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-258" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.fgcolor" class="py-name" href="#" onclick="return doclink('link-258', 'fgcolor', 'link-206');">fgcolor</a></tt> </tt>
<a name="L572"></a><tt class="py-lineno">572</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt>  <tt class="py-comment"># Assumes ReportLab Color object</tt> </tt>
<a name="L573"></a><tt class="py-lineno">573</tt>  <tt class="py-line">            <tt class="py-name">fghex</tt> <tt class="py-op">=</tt> <tt class="py-string">'#'</tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-259" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.fgcolor" class="py-name" href="#" onclick="return doclink('link-259', 'fgcolor', 'link-206');">fgcolor</a></tt><tt class="py-op">.</tt><tt class="py-name">hexval</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L574"></a><tt class="py-lineno">574</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-260" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.bgcolor" class="py-name" href="#" onclick="return doclink('link-260', 'bgcolor', 'link-207');">bgcolor</a></tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># Assumes that string is hexstring</tt> </tt>
<a name="L575"></a><tt class="py-lineno">575</tt>  <tt class="py-line">            <tt class="py-name">bghex</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-261" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.bgcolor" class="py-name" href="#" onclick="return doclink('link-261', 'bgcolor', 'link-207');">bgcolor</a></tt> </tt>
<a name="L576"></a><tt class="py-lineno">576</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt>  <tt class="py-comment"># Assumes ReportLab Color object</tt> </tt>
<a name="L577"></a><tt class="py-lineno">577</tt>  <tt class="py-line">            <tt class="py-name">bghex</tt> <tt class="py-op">=</tt> <tt class="py-string">'#'</tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-262" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.bgcolor" class="py-name" href="#" onclick="return doclink('link-262', 'bgcolor', 'link-207');">bgcolor</a></tt><tt class="py-op">.</tt><tt class="py-name">hexval</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L578"></a><tt class="py-lineno">578</tt>  <tt class="py-line">        <tt class="py-name">graphics</tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'name'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-263" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-263', 'name', 'link-65');">name</a></tt><tt class="py-op">,</tt> </tt>
<a name="L579"></a><tt class="py-lineno">579</tt>  <tt class="py-line">                           <tt class="py-string">'type'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-264" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-264', 'type', 'link-14');">type</a></tt><tt class="py-op">,</tt> </tt>
<a name="L580"></a><tt class="py-lineno">580</tt>  <tt class="py-line">                           <tt class="py-string">'fgcolor'</tt><tt class="py-op">:</tt> <tt class="py-name">fghex</tt><tt class="py-op">,</tt> </tt>
<a name="L581"></a><tt class="py-lineno">581</tt>  <tt class="py-line">                           <tt class="py-string">'bgcolor'</tt><tt class="py-op">:</tt> <tt class="py-name">bghex</tt><tt class="py-op">}</tt> </tt>
<a name="L582"></a><tt class="py-lineno">582</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">,</tt> <tt class="py-name">attr</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-string">'x'</tt><tt class="py-op">,</tt> <tt class="py-string">'_x'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-string">'y'</tt><tt class="py-op">,</tt> <tt class="py-string">'_y'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L583"></a><tt class="py-lineno">583</tt>  <tt class="py-line">                          <tt class="py-op">(</tt><tt class="py-string">'width'</tt><tt class="py-op">,</tt> <tt class="py-string">'_width'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-string">'height'</tt><tt class="py-op">,</tt> <tt class="py-string">'_height'</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L584"></a><tt class="py-lineno">584</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-name">attr</tt><tt class="py-op">)</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L585"></a><tt class="py-lineno">585</tt>  <tt class="py-line">                <tt class="py-name">graphics</tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">[</tt><tt class="py-name">n</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-name">attr</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L586"></a><tt class="py-lineno">586</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-265" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-265', 'type', 'link-14');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'line'</tt><tt class="py-op">:</tt>  <tt class="py-comment"># Need to write polycoords</tt> </tt>
<a name="L587"></a><tt class="py-lineno">587</tt>  <tt class="py-line">            <tt class="py-name">graphics</tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">[</tt><tt class="py-string">'coords'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> \ </tt>
<a name="L588"></a><tt class="py-lineno">588</tt>  <tt class="py-line">                <tt class="py-string">','</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">e</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">e</tt> <tt class="py-keyword">in</tt> <tt class="py-name">chain</tt><tt class="py-op">.</tt><tt class="py-name">from_iterable</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-266" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.coords" class="py-name" href="#" onclick="return doclink('link-266', 'coords', 'link-238');">coords</a></tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L589"></a><tt class="py-lineno">589</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">graphics</tt> </tt>
</div><a name="L590"></a><tt class="py-lineno">590</tt>  <tt class="py-line"> </tt>
<a name="L591"></a><tt class="py-lineno">591</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Graphics.bounds"></a><div id="Graphics.bounds-def"><a name="L592"></a><tt class="py-lineno">592</tt> <a class="py-toggle" href="#" id="Graphics.bounds-toggle" onclick="return toggle('Graphics.bounds');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#bounds">bounds</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics.bounds-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics.bounds-expanded"><a name="L593"></a><tt class="py-lineno">593</tt>  <tt class="py-line">        <tt class="py-docstring">"""Coordinate bounds for the Graphics element.</tt> </tt>
<a name="L594"></a><tt class="py-lineno">594</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L595"></a><tt class="py-lineno">595</tt>  <tt class="py-line"><tt class="py-docstring">        Return the bounds of the Graphics object as an [(xmin, ymin),</tt> </tt>
<a name="L596"></a><tt class="py-lineno">596</tt>  <tt class="py-line"><tt class="py-docstring">        (xmax, ymax)] tuple.  Co-ordinates give the centre of the</tt> </tt>
<a name="L597"></a><tt class="py-lineno">597</tt>  <tt class="py-line"><tt class="py-docstring">        circle, rectangle, roundrectangle elements, so we have to</tt> </tt>
<a name="L598"></a><tt class="py-lineno">598</tt>  <tt class="py-line"><tt class="py-docstring">        adjust for the relevant width/height.</tt> </tt>
<a name="L599"></a><tt class="py-lineno">599</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L600"></a><tt class="py-lineno">600</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-267" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-267', 'type', 'link-14');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'line'</tt><tt class="py-op">:</tt> </tt>
<a name="L601"></a><tt class="py-lineno">601</tt>  <tt class="py-line">            <tt class="py-name">xlist</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-268" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-268', 'x', 'link-209');">x</a></tt> <tt class="py-keyword">for</tt> <tt id="link-269" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-269', 'x', 'link-209');">x</a></tt><tt class="py-op">,</tt> <tt id="link-270" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-270', 'y', 'link-215');">y</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-271" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.coords" class="py-name" href="#" onclick="return doclink('link-271', 'coords', 'link-238');">coords</a></tt><tt class="py-op">]</tt> </tt>
<a name="L602"></a><tt class="py-lineno">602</tt>  <tt class="py-line">            <tt class="py-name">ylist</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-272" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-272', 'y', 'link-215');">y</a></tt> <tt class="py-keyword">for</tt> <tt id="link-273" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-273', 'x', 'link-209');">x</a></tt><tt class="py-op">,</tt> <tt id="link-274" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-274', 'y', 'link-215');">y</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-275" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.coords" class="py-name" href="#" onclick="return doclink('link-275', 'coords', 'link-238');">coords</a></tt><tt class="py-op">]</tt> </tt>
<a name="L603"></a><tt class="py-lineno">603</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-op">(</tt><tt id="link-276" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-276', 'min', 'link-122');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">xlist</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-277" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-277', 'min', 'link-122');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">ylist</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L604"></a><tt class="py-lineno">604</tt>  <tt class="py-line">                    <tt class="py-op">(</tt><tt id="link-278" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-278', 'max', 'link-124');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">xlist</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-279" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-279', 'max', 'link-124');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">ylist</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L605"></a><tt class="py-lineno">605</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L606"></a><tt class="py-lineno">606</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-280" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-280', 'x', 'link-209');">x</a></tt> <tt class="py-op">-</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-281" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.width
Bio.Phylo.PhyloXMLIO.Writer.width" class="py-name" href="#" onclick="return doclink('link-281', 'width', 'link-221');">width</a></tt> <tt class="py-op">*</tt> <tt class="py-number">0.5</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-282" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-282', 'y', 'link-215');">y</a></tt> <tt class="py-op">-</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-283" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.height" class="py-name" href="#" onclick="return doclink('link-283', 'height', 'link-227');">height</a></tt> <tt class="py-op">*</tt> <tt class="py-number">0.5</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L607"></a><tt class="py-lineno">607</tt>  <tt class="py-line">                    <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-284" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-284', 'x', 'link-209');">x</a></tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-285" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.width
Bio.Phylo.PhyloXMLIO.Writer.width" class="py-name" href="#" onclick="return doclink('link-285', 'width', 'link-221');">width</a></tt> <tt class="py-op">*</tt> <tt class="py-number">0.5</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-286" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-286', 'y', 'link-215');">y</a></tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-287" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.height" class="py-name" href="#" onclick="return doclink('link-287', 'height', 'link-227');">height</a></tt> <tt class="py-op">*</tt> <tt class="py-number">0.5</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
</div><a name="L608"></a><tt class="py-lineno">608</tt>  <tt class="py-line"> </tt>
<a name="L609"></a><tt class="py-lineno">609</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Graphics.centre"></a><div id="Graphics.centre-def"><a name="L610"></a><tt class="py-lineno">610</tt> <a class="py-toggle" href="#" id="Graphics.centre-toggle" onclick="return toggle('Graphics.centre');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#centre">centre</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Graphics.centre-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Graphics.centre-expanded"><a name="L611"></a><tt class="py-lineno">611</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the centre of the Graphics object as an (x, y) tuple."""</tt> </tt>
<a name="L612"></a><tt class="py-lineno">612</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-number">0.5</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-288" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.bounds
Bio.KEGG.KGML.KGML_pathway.Graphics.bounds
Bio.KEGG.KGML.KGML_pathway.Pathway.bounds" class="py-name" href="#" onclick="return doclink('link-288', 'bounds', 'link-114');">bounds</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-289" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.bounds
Bio.KEGG.KGML.KGML_pathway.Graphics.bounds
Bio.KEGG.KGML.KGML_pathway.Pathway.bounds" class="py-name" href="#" onclick="return doclink('link-289', 'bounds', 'link-114');">bounds</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L613"></a><tt class="py-lineno">613</tt>  <tt class="py-line">                <tt class="py-number">0.5</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-290" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.bounds
Bio.KEGG.KGML.KGML_pathway.Graphics.bounds
Bio.KEGG.KGML.KGML_pathway.Pathway.bounds" class="py-name" href="#" onclick="return doclink('link-290', 'bounds', 'link-114');">bounds</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-291" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.bounds
Bio.KEGG.KGML.KGML_pathway.Graphics.bounds
Bio.KEGG.KGML.KGML_pathway.Pathway.bounds" class="py-name" href="#" onclick="return doclink('link-291', 'bounds', 'link-114');">bounds</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L614"></a><tt class="py-lineno">614</tt>  <tt class="py-line"> </tt>
<a name="Reaction"></a><div id="Reaction-def"><a name="L615"></a><tt class="py-lineno">615</tt>  <tt class="py-line"> </tt>
<a name="L616"></a><tt class="py-lineno">616</tt>  <tt class="py-line"><tt class="py-comment"># Reaction</tt> </tt>
<a name="L617"></a><tt class="py-lineno">617</tt> <a class="py-toggle" href="#" id="Reaction-toggle" onclick="return toggle('Reaction');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Reaction-class.html">Reaction</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reaction-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Reaction-expanded"><a name="L618"></a><tt class="py-lineno">618</tt>  <tt class="py-line">    <tt class="py-docstring">"""A specific chemical reaction with substrates and products.</tt> </tt>
<a name="L619"></a><tt class="py-lineno">619</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L620"></a><tt class="py-lineno">620</tt>  <tt class="py-line"><tt class="py-docstring">    This describes a specific chemical reaction between one or more</tt> </tt>
<a name="L621"></a><tt class="py-lineno">621</tt>  <tt class="py-line"><tt class="py-docstring">    substrates and one or more products.</tt> </tt>
<a name="L622"></a><tt class="py-lineno">622</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L623"></a><tt class="py-lineno">623</tt>  <tt class="py-line"><tt class="py-docstring">    Attributes:</tt> </tt>
<a name="L624"></a><tt class="py-lineno">624</tt>  <tt class="py-line"><tt class="py-docstring">     - id             Pathway graph node ID of the entry</tt> </tt>
<a name="L625"></a><tt class="py-lineno">625</tt>  <tt class="py-line"><tt class="py-docstring">     - names          List of KEGG identifier(s) from the REACTION database</tt> </tt>
<a name="L626"></a><tt class="py-lineno">626</tt>  <tt class="py-line"><tt class="py-docstring">     - type           String: reversible or irreversible</tt> </tt>
<a name="L627"></a><tt class="py-lineno">627</tt>  <tt class="py-line"><tt class="py-docstring">     - substrate      Entry object of the substrate</tt> </tt>
<a name="L628"></a><tt class="py-lineno">628</tt>  <tt class="py-line"><tt class="py-docstring">     - product        Entry object of the product</tt> </tt>
<a name="L629"></a><tt class="py-lineno">629</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L630"></a><tt class="py-lineno">630</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L631"></a><tt class="py-lineno">631</tt>  <tt class="py-line"> </tt>
<a name="Reaction.__init__"></a><div id="Reaction.__init__-def"><a name="L632"></a><tt class="py-lineno">632</tt> <a class="py-toggle" href="#" id="Reaction.__init__-toggle" onclick="return toggle('Reaction.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reaction.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reaction.__init__-expanded"><a name="L633"></a><tt class="py-lineno">633</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the class."""</tt> </tt>
<a name="L634"></a><tt class="py-lineno">634</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_id</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L635"></a><tt class="py-lineno">635</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_names</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L636"></a><tt class="py-lineno">636</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-292" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-292', 'type', 'link-14');">type</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L637"></a><tt class="py-lineno">637</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_substrates</tt> <tt class="py-op">=</tt> <tt id="link-293" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-293', 'set', 'link-6');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L638"></a><tt class="py-lineno">638</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_products</tt> <tt class="py-op">=</tt> <tt id="link-294" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-294', 'set', 'link-6');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L639"></a><tt class="py-lineno">639</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-295" class="py-name"><a title="Bio.KEGG.Compound.Record._pathway
Bio.KEGG.Enzyme.Record._pathway" class="py-name" href="#" onclick="return doclink('link-295', '_pathway', 'link-20');">_pathway</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L640"></a><tt class="py-lineno">640</tt>  <tt class="py-line"> </tt>
<a name="Reaction.__str__"></a><div id="Reaction.__str__-def"><a name="L641"></a><tt class="py-lineno">641</tt> <a class="py-toggle" href="#" id="Reaction.__str__-toggle" onclick="return toggle('Reaction.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reaction.__str__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reaction.__str__-expanded"><a name="L642"></a><tt class="py-lineno">642</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return an informative human-readable string."""</tt> </tt>
<a name="L643"></a><tt class="py-lineno">643</tt>  <tt class="py-line">        <tt class="py-name">outstr</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">'Reaction node ID: %s'</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-296" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-296', 'id', 'link-13');">id</a></tt><tt class="py-op">,</tt> </tt>
<a name="L644"></a><tt class="py-lineno">644</tt>  <tt class="py-line">                  <tt class="py-string">'Reaction KEGG IDs: %s'</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-297" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-297', 'name', 'link-65');">name</a></tt><tt class="py-op">,</tt> </tt>
<a name="L645"></a><tt class="py-lineno">645</tt>  <tt class="py-line">                  <tt class="py-string">'Type: %s'</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-298" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-298', 'type', 'link-14');">type</a></tt><tt class="py-op">,</tt> </tt>
<a name="L646"></a><tt class="py-lineno">646</tt>  <tt class="py-line">                  <tt class="py-string">'Substrates: %s'</tt> <tt class="py-op">%</tt> </tt>
<a name="L647"></a><tt class="py-lineno">647</tt>  <tt class="py-line">                  <tt class="py-string">','</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-299" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-299', 'name', 'link-65');">name</a></tt> <tt class="py-keyword">for</tt> <tt class="py-name">s</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-300" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Reaction.substrates=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#substrates"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.substrates" class="py-name" href="#" onclick="return doclink('link-300', 'substrates', 'link-300');">substrates</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L648"></a><tt class="py-lineno">648</tt>  <tt class="py-line">                  <tt class="py-string">'Products: %s'</tt> <tt class="py-op">%</tt> </tt>
<a name="L649"></a><tt class="py-lineno">649</tt>  <tt class="py-line">                  <tt class="py-string">','</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-301" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-301', 'name', 'link-65');">name</a></tt> <tt class="py-keyword">for</tt> <tt class="py-name">s</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-302" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Reaction.products=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#products"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.products" class="py-name" href="#" onclick="return doclink('link-302', 'products', 'link-302');">products</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L650"></a><tt class="py-lineno">650</tt>  <tt class="py-line">                  <tt class="py-op">]</tt> </tt>
<a name="L651"></a><tt class="py-lineno">651</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">'\n'</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">outstr</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">'\n'</tt> </tt>
</div><a name="L652"></a><tt class="py-lineno">652</tt>  <tt class="py-line"> </tt>
<a name="Reaction.add_substrate"></a><div id="Reaction.add_substrate-def"><a name="L653"></a><tt class="py-lineno">653</tt> <a class="py-toggle" href="#" id="Reaction.add_substrate-toggle" onclick="return toggle('Reaction.add_substrate');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#add_substrate">add_substrate</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">substrate_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reaction.add_substrate-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reaction.add_substrate-expanded"><a name="L654"></a><tt class="py-lineno">654</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add a substrate, identified by its node ID, to the reaction."""</tt> </tt>
<a name="L655"></a><tt class="py-lineno">655</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-303" class="py-name"><a title="Bio.KEGG.Compound.Record._pathway
Bio.KEGG.Enzyme.Record._pathway" class="py-name" href="#" onclick="return doclink('link-303', '_pathway', 'link-20');">_pathway</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L656"></a><tt class="py-lineno">656</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">substrate_id</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-304" class="py-name"><a title="Bio.KEGG.Compound.Record._pathway
Bio.KEGG.Enzyme.Record._pathway" class="py-name" href="#" onclick="return doclink('link-304', '_pathway', 'link-20');">_pathway</a></tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">,</tt> \ </tt>
<a name="L657"></a><tt class="py-lineno">657</tt>  <tt class="py-line">                <tt class="py-string">"Couldn't add substrate, no node ID %d in Pathway"</tt> <tt class="py-op">%</tt> \ </tt>
<a name="L658"></a><tt class="py-lineno">658</tt>  <tt class="py-line">                <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">substrate_id</tt><tt class="py-op">)</tt> </tt>
<a name="L659"></a><tt class="py-lineno">659</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_substrates</tt><tt class="py-op">.</tt><tt id="link-305" class="py-name"><a title="Bio.Graphics.BasicChromosome._ChromosomeComponent.add
Bio.Nexus.Nexus.StepMatrix.add
Bio.Nexus.Nodes.Chain.add
Bio.PDB.Entity.Entity.add
Bio.PDB.Residue.DisorderedResidue.add
Bio.PDB.Residue.Residue.add
Bio.Restriction.Restriction.RestrictionBatch.add" class="py-name" href="#" onclick="return doclink('link-305', 'add', 'link-62');">add</a></tt><tt class="py-op">(</tt><tt class="py-name">substrate_id</tt><tt class="py-op">)</tt> </tt>
</div><a name="L660"></a><tt class="py-lineno">660</tt>  <tt class="py-line"> </tt>
<a name="Reaction.add_product"></a><div id="Reaction.add_product-def"><a name="L661"></a><tt class="py-lineno">661</tt> <a class="py-toggle" href="#" id="Reaction.add_product-toggle" onclick="return toggle('Reaction.add_product');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#add_product">add_product</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">product_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reaction.add_product-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reaction.add_product-expanded"><a name="L662"></a><tt class="py-lineno">662</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add a product, identified by its node ID, to the reaction."""</tt> </tt>
<a name="L663"></a><tt class="py-lineno">663</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-306" class="py-name"><a title="Bio.KEGG.Compound.Record._pathway
Bio.KEGG.Enzyme.Record._pathway" class="py-name" href="#" onclick="return doclink('link-306', '_pathway', 'link-20');">_pathway</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L664"></a><tt class="py-lineno">664</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">product_id</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-307" class="py-name"><a title="Bio.KEGG.Compound.Record._pathway
Bio.KEGG.Enzyme.Record._pathway" class="py-name" href="#" onclick="return doclink('link-307', '_pathway', 'link-20');">_pathway</a></tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">,</tt> \ </tt>
<a name="L665"></a><tt class="py-lineno">665</tt>  <tt class="py-line">                <tt class="py-string">"Couldn't add product, no node ID %d in Pathway"</tt> <tt class="py-op">%</tt> <tt class="py-name">product_id</tt> </tt>
<a name="L666"></a><tt class="py-lineno">666</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_products</tt><tt class="py-op">.</tt><tt id="link-308" class="py-name"><a title="Bio.Graphics.BasicChromosome._ChromosomeComponent.add
Bio.Nexus.Nexus.StepMatrix.add
Bio.Nexus.Nodes.Chain.add
Bio.PDB.Entity.Entity.add
Bio.PDB.Residue.DisorderedResidue.add
Bio.PDB.Residue.Residue.add
Bio.Restriction.Restriction.RestrictionBatch.add" class="py-name" href="#" onclick="return doclink('link-308', 'add', 'link-62');">add</a></tt><tt class="py-op">(</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">product_id</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L667"></a><tt class="py-lineno">667</tt>  <tt class="py-line"> </tt>
<a name="L668"></a><tt class="py-lineno">668</tt>  <tt class="py-line">    <tt class="py-comment"># The node ID is also the node ID of the Entry that corresponds to the</tt> </tt>
<a name="L669"></a><tt class="py-lineno">669</tt>  <tt class="py-line">    <tt class="py-comment"># reaction; we get the corresponding Entry when there is an associated</tt> </tt>
<a name="L670"></a><tt class="py-lineno">670</tt>  <tt class="py-line">    <tt class="py-comment"># Pathway</tt> </tt>
<a name="Reaction._getid"></a><div id="Reaction._getid-def"><a name="L671"></a><tt class="py-lineno">671</tt> <a class="py-toggle" href="#" id="Reaction._getid-toggle" onclick="return toggle('Reaction._getid');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#_getid">_getid</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reaction._getid-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reaction._getid-expanded"><a name="L672"></a><tt class="py-lineno">672</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_id</tt> </tt>
</div><a name="L673"></a><tt class="py-lineno">673</tt>  <tt class="py-line"> </tt>
<a name="Reaction._setid"></a><div id="Reaction._setid-def"><a name="L674"></a><tt class="py-lineno">674</tt> <a class="py-toggle" href="#" id="Reaction._setid-toggle" onclick="return toggle('Reaction._setid');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#_setid">_setid</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reaction._setid-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reaction._setid-expanded"><a name="L675"></a><tt class="py-lineno">675</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_id</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-309" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-309', 'value', 'link-70');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L676"></a><tt class="py-lineno">676</tt>  <tt class="py-line"> </tt>
<a name="Reaction._delid"></a><div id="Reaction._delid-def"><a name="L677"></a><tt class="py-lineno">677</tt> <a class="py-toggle" href="#" id="Reaction._delid-toggle" onclick="return toggle('Reaction._delid');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#_delid">_delid</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reaction._delid-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reaction._delid-expanded"><a name="L678"></a><tt class="py-lineno">678</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_id</tt> </tt>
</div><a name="L679"></a><tt class="py-lineno">679</tt>  <tt class="py-line">    <tt id="link-310" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-310', 'id', 'link-13');">id</a></tt> <tt class="py-op">=</tt> <tt id="link-311" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-311', 'property', 'link-77');">property</a></tt><tt class="py-op">(</tt><tt id="link-312" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component._getid
Bio.KEGG.KGML.KGML_pathway.Entry._getid
Bio.KEGG.KGML.KGML_pathway.Reaction._getid" class="py-name" href="#" onclick="return doclink('link-312', '_getid', 'link-166');">_getid</a></tt><tt class="py-op">,</tt> <tt id="link-313" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component._setid
Bio.KEGG.KGML.KGML_pathway.Entry._setid
Bio.KEGG.KGML.KGML_pathway.Reaction._setid" class="py-name" href="#" onclick="return doclink('link-313', '_setid', 'link-167');">_setid</a></tt><tt class="py-op">,</tt> <tt id="link-314" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component._delid
Bio.KEGG.KGML.KGML_pathway.Entry._delid
Bio.KEGG.KGML.KGML_pathway.Reaction._delid" class="py-name" href="#" onclick="return doclink('link-314', '_delid', 'link-168');">_delid</a></tt><tt class="py-op">,</tt> </tt>
<a name="L680"></a><tt class="py-lineno">680</tt>  <tt class="py-line">                  <tt class="py-string">"Node ID for the reaction."</tt><tt class="py-op">)</tt> </tt>
<a name="L681"></a><tt class="py-lineno">681</tt>  <tt class="py-line"> </tt>
<a name="L682"></a><tt class="py-lineno">682</tt>  <tt class="py-line">    <tt class="py-comment"># Names may show up as a space-separated list of several KEGG identifiers</tt> </tt>
<a name="Reaction._getnames"></a><div id="Reaction._getnames-def"><a name="L683"></a><tt class="py-lineno">683</tt> <a class="py-toggle" href="#" id="Reaction._getnames-toggle" onclick="return toggle('Reaction._getnames');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#_getnames">_getnames</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reaction._getnames-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reaction._getnames-expanded"><a name="L684"></a><tt class="py-lineno">684</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">' '</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_names</tt><tt class="py-op">)</tt> </tt>
</div><a name="L685"></a><tt class="py-lineno">685</tt>  <tt class="py-line"> </tt>
<a name="Reaction._setnames"></a><div id="Reaction._setnames-def"><a name="L686"></a><tt class="py-lineno">686</tt> <a class="py-toggle" href="#" id="Reaction._setnames-toggle" onclick="return toggle('Reaction._setnames');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#_setnames">_setnames</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reaction._setnames-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reaction._setnames-expanded"><a name="L687"></a><tt class="py-lineno">687</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_names</tt><tt class="py-op">.</tt><tt id="link-315" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-315', 'extend', 'link-116');">extend</a></tt><tt class="py-op">(</tt><tt id="link-316" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-316', 'value', 'link-70');">value</a></tt><tt class="py-op">.</tt><tt id="link-317" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-317', 'split', 'link-150');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L688"></a><tt class="py-lineno">688</tt>  <tt class="py-line"> </tt>
<a name="Reaction._delnames"></a><div id="Reaction._delnames-def"><a name="L689"></a><tt class="py-lineno">689</tt> <a class="py-toggle" href="#" id="Reaction._delnames-toggle" onclick="return toggle('Reaction._delnames');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#_delnames">_delnames</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reaction._delnames-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reaction._delnames-expanded"><a name="L690"></a><tt class="py-lineno">690</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">names</tt> </tt>
</div><a name="L691"></a><tt class="py-lineno">691</tt>  <tt class="py-line">    <tt id="link-318" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-318', 'name', 'link-65');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-319" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-319', 'property', 'link-77');">property</a></tt><tt class="py-op">(</tt><tt id="link-320" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Reaction._getnames()=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#_getnames"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction._getnames" class="py-name" href="#" onclick="return doclink('link-320', '_getnames', 'link-320');">_getnames</a></tt><tt class="py-op">,</tt> <tt id="link-321" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Reaction._setnames()=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#_setnames"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction._setnames" class="py-name" href="#" onclick="return doclink('link-321', '_setnames', 'link-321');">_setnames</a></tt><tt class="py-op">,</tt> <tt id="link-322" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Reaction._delnames()=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#_delnames"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction._delnames" class="py-name" href="#" onclick="return doclink('link-322', '_delnames', 'link-322');">_delnames</a></tt><tt class="py-op">,</tt> </tt>
<a name="L692"></a><tt class="py-lineno">692</tt>  <tt class="py-line">                    <tt class="py-string">"List of KEGG identifiers for the reaction."</tt><tt class="py-op">)</tt> </tt>
<a name="L693"></a><tt class="py-lineno">693</tt>  <tt class="py-line"> </tt>
<a name="L694"></a><tt class="py-lineno">694</tt>  <tt class="py-line">    <tt class="py-comment"># products and substrates are read-only properties, returning lists</tt> </tt>
<a name="L695"></a><tt class="py-lineno">695</tt>  <tt class="py-line">    <tt class="py-comment"># of Entry objects</tt> </tt>
<a name="L696"></a><tt class="py-lineno">696</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Reaction.substrates"></a><div id="Reaction.substrates-def"><a name="L697"></a><tt class="py-lineno">697</tt> <a class="py-toggle" href="#" id="Reaction.substrates-toggle" onclick="return toggle('Reaction.substrates');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#substrates">substrates</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reaction.substrates-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reaction.substrates-expanded"><a name="L698"></a><tt class="py-lineno">698</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return list of substrate Entry elements."""</tt> </tt>
<a name="L699"></a><tt class="py-lineno">699</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-323" class="py-name"><a title="Bio.KEGG.Compound.Record._pathway
Bio.KEGG.Enzyme.Record._pathway" class="py-name" href="#" onclick="return doclink('link-323', '_pathway', 'link-20');">_pathway</a></tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">[</tt><tt class="py-name">sid</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt class="py-name">sid</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_substrates</tt><tt class="py-op">]</tt> </tt>
</div><a name="L700"></a><tt class="py-lineno">700</tt>  <tt class="py-line"> </tt>
<a name="L701"></a><tt class="py-lineno">701</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Reaction.products"></a><div id="Reaction.products-def"><a name="L702"></a><tt class="py-lineno">702</tt> <a class="py-toggle" href="#" id="Reaction.products-toggle" onclick="return toggle('Reaction.products');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#products">products</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reaction.products-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reaction.products-expanded"><a name="L703"></a><tt class="py-lineno">703</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return list of product Entry elements."""</tt> </tt>
<a name="L704"></a><tt class="py-lineno">704</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-324" class="py-name"><a title="Bio.KEGG.Compound.Record._pathway
Bio.KEGG.Enzyme.Record._pathway" class="py-name" href="#" onclick="return doclink('link-324', '_pathway', 'link-20');">_pathway</a></tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">[</tt><tt id="link-325" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.pid()=Bio.GenBank._FeatureConsumer-class.html#pid,Method Bio.GenBank._RecordConsumer.pid()=Bio.GenBank._RecordConsumer-class.html#pid"><a title="Bio.GenBank._FeatureConsumer.pid
Bio.GenBank._RecordConsumer.pid" class="py-name" href="#" onclick="return doclink('link-325', 'pid', 'link-325');">pid</a></tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt id="link-326" class="py-name"><a title="Bio.GenBank._FeatureConsumer.pid
Bio.GenBank._RecordConsumer.pid" class="py-name" href="#" onclick="return doclink('link-326', 'pid', 'link-325');">pid</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_products</tt><tt class="py-op">]</tt> </tt>
</div><a name="L705"></a><tt class="py-lineno">705</tt>  <tt class="py-line"> </tt>
<a name="L706"></a><tt class="py-lineno">706</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Reaction.entry"></a><div id="Reaction.entry-def"><a name="L707"></a><tt class="py-lineno">707</tt> <a class="py-toggle" href="#" id="Reaction.entry-toggle" onclick="return toggle('Reaction.entry');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#entry">entry</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reaction.entry-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reaction.entry-expanded"><a name="L708"></a><tt class="py-lineno">708</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the Entry corresponding to this reaction."""</tt> </tt>
<a name="L709"></a><tt class="py-lineno">709</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-327" class="py-name"><a title="Bio.KEGG.Compound.Record._pathway
Bio.KEGG.Enzyme.Record._pathway" class="py-name" href="#" onclick="return doclink('link-327', '_pathway', 'link-20');">_pathway</a></tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_id</tt><tt class="py-op">]</tt> </tt>
</div><a name="L710"></a><tt class="py-lineno">710</tt>  <tt class="py-line"> </tt>
<a name="L711"></a><tt class="py-lineno">711</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Reaction.reactant_ids"></a><div id="Reaction.reactant_ids-def"><a name="L712"></a><tt class="py-lineno">712</tt> <a class="py-toggle" href="#" id="Reaction.reactant_ids-toggle" onclick="return toggle('Reaction.reactant_ids');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#reactant_ids">reactant_ids</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reaction.reactant_ids-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reaction.reactant_ids-expanded"><a name="L713"></a><tt class="py-lineno">713</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a list of substrate and product reactant IDs."""</tt> </tt>
<a name="L714"></a><tt class="py-lineno">714</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_products</tt><tt class="py-op">.</tt><tt class="py-name">union</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_substrates</tt><tt class="py-op">)</tt> </tt>
</div><a name="L715"></a><tt class="py-lineno">715</tt>  <tt class="py-line"> </tt>
<a name="L716"></a><tt class="py-lineno">716</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Reaction.element"></a><div id="Reaction.element-def"><a name="L717"></a><tt class="py-lineno">717</tt> <a class="py-toggle" href="#" id="Reaction.element-toggle" onclick="return toggle('Reaction.element');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#element">element</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reaction.element-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reaction.element-expanded"><a name="L718"></a><tt class="py-lineno">718</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return KGML element describing the Reaction."""</tt> </tt>
<a name="L719"></a><tt class="py-lineno">719</tt>  <tt class="py-line">        <tt class="py-comment"># The root is this Relation element</tt> </tt>
<a name="L720"></a><tt class="py-lineno">720</tt>  <tt class="py-line">        <tt id="link-328" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.reaction" class="py-name" href="#" onclick="return doclink('link-328', 'reaction', 'link-35');">reaction</a></tt> <tt class="py-op">=</tt> <tt class="py-name">ET</tt><tt class="py-op">.</tt><tt class="py-name">Element</tt><tt class="py-op">(</tt><tt class="py-string">'reaction'</tt><tt class="py-op">)</tt> </tt>
<a name="L721"></a><tt class="py-lineno">721</tt>  <tt class="py-line">        <tt id="link-329" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.reaction" class="py-name" href="#" onclick="return doclink('link-329', 'reaction', 'link-35');">reaction</a></tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'id'</tt><tt class="py-op">:</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-330" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-330', 'id', 'link-13');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L722"></a><tt class="py-lineno">722</tt>  <tt class="py-line">                           <tt class="py-string">'name'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-331" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-331', 'name', 'link-65');">name</a></tt><tt class="py-op">,</tt> </tt>
<a name="L723"></a><tt class="py-lineno">723</tt>  <tt class="py-line">                           <tt class="py-string">'type'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-332" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-332', 'type', 'link-14');">type</a></tt><tt class="py-op">}</tt> </tt>
<a name="L724"></a><tt class="py-lineno">724</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">s</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_substrates</tt><tt class="py-op">:</tt> </tt>
<a name="L725"></a><tt class="py-lineno">725</tt>  <tt class="py-line">            <tt class="py-name">substrate</tt> <tt class="py-op">=</tt> <tt class="py-name">ET</tt><tt class="py-op">.</tt><tt class="py-name">Element</tt><tt class="py-op">(</tt><tt class="py-string">'substrate'</tt><tt class="py-op">)</tt> </tt>
<a name="L726"></a><tt class="py-lineno">726</tt>  <tt class="py-line">            <tt class="py-name">substrate</tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">[</tt><tt class="py-string">'id'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt> </tt>
<a name="L727"></a><tt class="py-lineno">727</tt>  <tt class="py-line">            <tt class="py-name">substrate</tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">[</tt><tt class="py-string">'name'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-333" class="py-name"><a title="Bio.KEGG.Compound.Record._pathway
Bio.KEGG.Enzyme.Record._pathway" class="py-name" href="#" onclick="return doclink('link-333', '_pathway', 'link-20');">_pathway</a></tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">[</tt><tt class="py-name">s</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-334" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-334', 'name', 'link-65');">name</a></tt> </tt>
<a name="L728"></a><tt class="py-lineno">728</tt>  <tt class="py-line">            <tt id="link-335" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.reaction" class="py-name" href="#" onclick="return doclink('link-335', 'reaction', 'link-35');">reaction</a></tt><tt class="py-op">.</tt><tt id="link-336" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-336', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">substrate</tt><tt class="py-op">)</tt> </tt>
<a name="L729"></a><tt class="py-lineno">729</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">p</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_products</tt><tt class="py-op">:</tt> </tt>
<a name="L730"></a><tt class="py-lineno">730</tt>  <tt class="py-line">            <tt class="py-name">product</tt> <tt class="py-op">=</tt> <tt class="py-name">ET</tt><tt class="py-op">.</tt><tt class="py-name">Element</tt><tt class="py-op">(</tt><tt class="py-string">'product'</tt><tt class="py-op">)</tt> </tt>
<a name="L731"></a><tt class="py-lineno">731</tt>  <tt class="py-line">            <tt class="py-name">product</tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">[</tt><tt class="py-string">'id'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">p</tt><tt class="py-op">)</tt> </tt>
<a name="L732"></a><tt class="py-lineno">732</tt>  <tt class="py-line">            <tt class="py-name">product</tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">[</tt><tt class="py-string">'name'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-337" class="py-name"><a title="Bio.KEGG.Compound.Record._pathway
Bio.KEGG.Enzyme.Record._pathway" class="py-name" href="#" onclick="return doclink('link-337', '_pathway', 'link-20');">_pathway</a></tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">[</tt><tt class="py-name">p</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-338" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-338', 'name', 'link-65');">name</a></tt> </tt>
<a name="L733"></a><tt class="py-lineno">733</tt>  <tt class="py-line">            <tt id="link-339" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.reaction" class="py-name" href="#" onclick="return doclink('link-339', 'reaction', 'link-35');">reaction</a></tt><tt class="py-op">.</tt><tt id="link-340" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-340', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">product</tt><tt class="py-op">)</tt> </tt>
<a name="L734"></a><tt class="py-lineno">734</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-341" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.reaction" class="py-name" href="#" onclick="return doclink('link-341', 'reaction', 'link-35');">reaction</a></tt> </tt>
</div></div><a name="L735"></a><tt class="py-lineno">735</tt>  <tt class="py-line"> </tt>
<a name="Relation"></a><div id="Relation-def"><a name="L736"></a><tt class="py-lineno">736</tt>  <tt class="py-line"> </tt>
<a name="L737"></a><tt class="py-lineno">737</tt>  <tt class="py-line"><tt class="py-comment"># Relation</tt> </tt>
<a name="L738"></a><tt class="py-lineno">738</tt> <a class="py-toggle" href="#" id="Relation-toggle" onclick="return toggle('Relation');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Relation-class.html">Relation</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Relation-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Relation-expanded"><a name="L739"></a><tt class="py-lineno">739</tt>  <tt class="py-line">    <tt class="py-docstring">"""A relationship between to products, KOs, or protein and compound.</tt> </tt>
<a name="L740"></a><tt class="py-lineno">740</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L741"></a><tt class="py-lineno">741</tt>  <tt class="py-line"><tt class="py-docstring">    This describes a relationship between two products, KOs, or protein</tt> </tt>
<a name="L742"></a><tt class="py-lineno">742</tt>  <tt class="py-line"><tt class="py-docstring">    and compound, as described in release KGML v0.7.1</tt> </tt>
<a name="L743"></a><tt class="py-lineno">743</tt>  <tt class="py-line"><tt class="py-docstring">    (http://www.kegg.jp/kegg/xml/docs/)</tt> </tt>
<a name="L744"></a><tt class="py-lineno">744</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L745"></a><tt class="py-lineno">745</tt>  <tt class="py-line"><tt class="py-docstring">    Attributes:</tt> </tt>
<a name="L746"></a><tt class="py-lineno">746</tt>  <tt class="py-line"><tt class="py-docstring">     - entry1 - The first Entry object node ID defining the</tt> </tt>
<a name="L747"></a><tt class="py-lineno">747</tt>  <tt class="py-line"><tt class="py-docstring">       relation (int)</tt> </tt>
<a name="L748"></a><tt class="py-lineno">748</tt>  <tt class="py-line"><tt class="py-docstring">     - entry2 - The second Entry object node ID defining the</tt> </tt>
<a name="L749"></a><tt class="py-lineno">749</tt>  <tt class="py-line"><tt class="py-docstring">       relation (int)</tt> </tt>
<a name="L750"></a><tt class="py-lineno">750</tt>  <tt class="py-line"><tt class="py-docstring">     - type - The relation type</tt> </tt>
<a name="L751"></a><tt class="py-lineno">751</tt>  <tt class="py-line"><tt class="py-docstring">     - subtypes - List of subtypes for the relation, as a list of</tt> </tt>
<a name="L752"></a><tt class="py-lineno">752</tt>  <tt class="py-line"><tt class="py-docstring">       (name, value) tuples</tt> </tt>
<a name="L753"></a><tt class="py-lineno">753</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L754"></a><tt class="py-lineno">754</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L755"></a><tt class="py-lineno">755</tt>  <tt class="py-line"> </tt>
<a name="Relation.__init__"></a><div id="Relation.__init__-def"><a name="L756"></a><tt class="py-lineno">756</tt> <a class="py-toggle" href="#" id="Relation.__init__-toggle" onclick="return toggle('Relation.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Relation-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Relation.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Relation.__init__-expanded"><a name="L757"></a><tt class="py-lineno">757</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the class."""</tt> </tt>
<a name="L758"></a><tt class="py-lineno">758</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_entry1</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L759"></a><tt class="py-lineno">759</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_entry2</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L760"></a><tt class="py-lineno">760</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-342" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-342', 'type', 'link-14');">type</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L761"></a><tt class="py-lineno">761</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">subtypes</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L762"></a><tt class="py-lineno">762</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-343" class="py-name"><a title="Bio.KEGG.Compound.Record._pathway
Bio.KEGG.Enzyme.Record._pathway" class="py-name" href="#" onclick="return doclink('link-343', '_pathway', 'link-20');">_pathway</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L763"></a><tt class="py-lineno">763</tt>  <tt class="py-line"> </tt>
<a name="Relation.__str__"></a><div id="Relation.__str__-def"><a name="L764"></a><tt class="py-lineno">764</tt> <a class="py-toggle" href="#" id="Relation.__str__-toggle" onclick="return toggle('Relation.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Relation-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Relation.__str__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Relation.__str__-expanded"><a name="L765"></a><tt class="py-lineno">765</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a useful human-readable string."""</tt> </tt>
<a name="L766"></a><tt class="py-lineno">766</tt>  <tt class="py-line">        <tt class="py-name">outstr</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">'Relation (subtypes: %d):'</tt> <tt class="py-op">%</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">subtypes</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L767"></a><tt class="py-lineno">767</tt>  <tt class="py-line">                  <tt class="py-string">'Entry1:'</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-344" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Relation.entry1=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#entry1"><a title="Bio.KEGG.KGML.KGML_pathway.Relation.entry1" class="py-name" href="#" onclick="return doclink('link-344', 'entry1', 'link-344');">entry1</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L768"></a><tt class="py-lineno">768</tt>  <tt class="py-line">                  <tt class="py-string">'Entry2:'</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-345" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Relation.entry2=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#entry2"><a title="Bio.KEGG.KGML.KGML_pathway.Relation.entry2" class="py-name" href="#" onclick="return doclink('link-345', 'entry2', 'link-345');">entry2</a></tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L769"></a><tt class="py-lineno">769</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">s</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">subtypes</tt><tt class="py-op">:</tt> </tt>
<a name="L770"></a><tt class="py-lineno">770</tt>  <tt class="py-line">            <tt class="py-name">outstr</tt><tt class="py-op">.</tt><tt id="link-346" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-346', 'extend', 'link-116');">extend</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-string">'Subtype: %s'</tt> <tt class="py-op">%</tt> <tt class="py-name">s</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L771"></a><tt class="py-lineno">771</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">'\n'</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">outstr</tt><tt class="py-op">)</tt> </tt>
</div><a name="L772"></a><tt class="py-lineno">772</tt>  <tt class="py-line"> </tt>
<a name="L773"></a><tt class="py-lineno">773</tt>  <tt class="py-line">    <tt class="py-comment"># Properties entry1 and entry2</tt> </tt>
<a name="Relation._getentry1"></a><div id="Relation._getentry1-def"><a name="L774"></a><tt class="py-lineno">774</tt> <a class="py-toggle" href="#" id="Relation._getentry1-toggle" onclick="return toggle('Relation._getentry1');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Relation-class.html#_getentry1">_getentry1</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Relation._getentry1-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Relation._getentry1-expanded"><a name="L775"></a><tt class="py-lineno">775</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-347" class="py-name"><a title="Bio.KEGG.Compound.Record._pathway
Bio.KEGG.Enzyme.Record._pathway" class="py-name" href="#" onclick="return doclink('link-347', '_pathway', 'link-20');">_pathway</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L776"></a><tt class="py-lineno">776</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-348" class="py-name"><a title="Bio.KEGG.Compound.Record._pathway
Bio.KEGG.Enzyme.Record._pathway" class="py-name" href="#" onclick="return doclink('link-348', '_pathway', 'link-20');">_pathway</a></tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_entry1</tt><tt class="py-op">]</tt> </tt>
<a name="L777"></a><tt class="py-lineno">777</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_entry1</tt> </tt>
</div><a name="L778"></a><tt class="py-lineno">778</tt>  <tt class="py-line"> </tt>
<a name="Relation._setentry1"></a><div id="Relation._setentry1-def"><a name="L779"></a><tt class="py-lineno">779</tt> <a class="py-toggle" href="#" id="Relation._setentry1-toggle" onclick="return toggle('Relation._setentry1');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Relation-class.html#_setentry1">_setentry1</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Relation._setentry1-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Relation._setentry1-expanded"><a name="L780"></a><tt class="py-lineno">780</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_entry1</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-349" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-349', 'value', 'link-70');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L781"></a><tt class="py-lineno">781</tt>  <tt class="py-line"> </tt>
<a name="Relation._delentry1"></a><div id="Relation._delentry1-def"><a name="L782"></a><tt class="py-lineno">782</tt> <a class="py-toggle" href="#" id="Relation._delentry1-toggle" onclick="return toggle('Relation._delentry1');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Relation-class.html#_delentry1">_delentry1</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Relation._delentry1-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Relation._delentry1-expanded"><a name="L783"></a><tt class="py-lineno">783</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_entry1</tt> </tt>
</div><a name="L784"></a><tt class="py-lineno">784</tt>  <tt class="py-line">    <tt id="link-350" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Relation.entry1" class="py-name" href="#" onclick="return doclink('link-350', 'entry1', 'link-344');">entry1</a></tt> <tt class="py-op">=</tt> <tt id="link-351" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-351', 'property', 'link-77');">property</a></tt><tt class="py-op">(</tt><tt id="link-352" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Relation._getentry1()=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#_getentry1"><a title="Bio.KEGG.KGML.KGML_pathway.Relation._getentry1" class="py-name" href="#" onclick="return doclink('link-352', '_getentry1', 'link-352');">_getentry1</a></tt><tt class="py-op">,</tt> <tt id="link-353" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Relation._setentry1()=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#_setentry1"><a title="Bio.KEGG.KGML.KGML_pathway.Relation._setentry1" class="py-name" href="#" onclick="return doclink('link-353', '_setentry1', 'link-353');">_setentry1</a></tt><tt class="py-op">,</tt> <tt id="link-354" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Relation._delentry1()=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#_delentry1"><a title="Bio.KEGG.KGML.KGML_pathway.Relation._delentry1" class="py-name" href="#" onclick="return doclink('link-354', '_delentry1', 'link-354');">_delentry1</a></tt><tt class="py-op">,</tt> </tt>
<a name="L785"></a><tt class="py-lineno">785</tt>  <tt class="py-line">                      <tt class="py-string">"Entry1 of the relation."</tt><tt class="py-op">)</tt> </tt>
<a name="L786"></a><tt class="py-lineno">786</tt>  <tt class="py-line"> </tt>
<a name="Relation._getentry2"></a><div id="Relation._getentry2-def"><a name="L787"></a><tt class="py-lineno">787</tt> <a class="py-toggle" href="#" id="Relation._getentry2-toggle" onclick="return toggle('Relation._getentry2');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Relation-class.html#_getentry2">_getentry2</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Relation._getentry2-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Relation._getentry2-expanded"><a name="L788"></a><tt class="py-lineno">788</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-355" class="py-name"><a title="Bio.KEGG.Compound.Record._pathway
Bio.KEGG.Enzyme.Record._pathway" class="py-name" href="#" onclick="return doclink('link-355', '_pathway', 'link-20');">_pathway</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L789"></a><tt class="py-lineno">789</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-356" class="py-name"><a title="Bio.KEGG.Compound.Record._pathway
Bio.KEGG.Enzyme.Record._pathway" class="py-name" href="#" onclick="return doclink('link-356', '_pathway', 'link-20');">_pathway</a></tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_entry2</tt><tt class="py-op">]</tt> </tt>
<a name="L790"></a><tt class="py-lineno">790</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_entry2</tt> </tt>
</div><a name="L791"></a><tt class="py-lineno">791</tt>  <tt class="py-line"> </tt>
<a name="Relation._setentry2"></a><div id="Relation._setentry2-def"><a name="L792"></a><tt class="py-lineno">792</tt> <a class="py-toggle" href="#" id="Relation._setentry2-toggle" onclick="return toggle('Relation._setentry2');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Relation-class.html#_setentry2">_setentry2</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Relation._setentry2-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Relation._setentry2-expanded"><a name="L793"></a><tt class="py-lineno">793</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_entry2</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-357" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-357', 'value', 'link-70');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L794"></a><tt class="py-lineno">794</tt>  <tt class="py-line"> </tt>
<a name="Relation._delentry2"></a><div id="Relation._delentry2-def"><a name="L795"></a><tt class="py-lineno">795</tt> <a class="py-toggle" href="#" id="Relation._delentry2-toggle" onclick="return toggle('Relation._delentry2');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Relation-class.html#_delentry2">_delentry2</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Relation._delentry2-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Relation._delentry2-expanded"><a name="L796"></a><tt class="py-lineno">796</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_entry2</tt> </tt>
</div><a name="L797"></a><tt class="py-lineno">797</tt>  <tt class="py-line">    <tt id="link-358" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Relation.entry2" class="py-name" href="#" onclick="return doclink('link-358', 'entry2', 'link-345');">entry2</a></tt> <tt class="py-op">=</tt> <tt id="link-359" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-359', 'property', 'link-77');">property</a></tt><tt class="py-op">(</tt><tt id="link-360" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Relation._getentry2()=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#_getentry2"><a title="Bio.KEGG.KGML.KGML_pathway.Relation._getentry2" class="py-name" href="#" onclick="return doclink('link-360', '_getentry2', 'link-360');">_getentry2</a></tt><tt class="py-op">,</tt> <tt id="link-361" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Relation._setentry2()=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#_setentry2"><a title="Bio.KEGG.KGML.KGML_pathway.Relation._setentry2" class="py-name" href="#" onclick="return doclink('link-361', '_setentry2', 'link-361');">_setentry2</a></tt><tt class="py-op">,</tt> <tt id="link-362" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Relation._delentry2()=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#_delentry2"><a title="Bio.KEGG.KGML.KGML_pathway.Relation._delentry2" class="py-name" href="#" onclick="return doclink('link-362', '_delentry2', 'link-362');">_delentry2</a></tt><tt class="py-op">,</tt> </tt>
<a name="L798"></a><tt class="py-lineno">798</tt>  <tt class="py-line">                      <tt class="py-string">"Entry2 of the relation."</tt><tt class="py-op">)</tt> </tt>
<a name="L799"></a><tt class="py-lineno">799</tt>  <tt class="py-line"> </tt>
<a name="L800"></a><tt class="py-lineno">800</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Relation.element"></a><div id="Relation.element-def"><a name="L801"></a><tt class="py-lineno">801</tt> <a class="py-toggle" href="#" id="Relation.element-toggle" onclick="return toggle('Relation.element');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_pathway.Relation-class.html#element">element</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Relation.element-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Relation.element-expanded"><a name="L802"></a><tt class="py-lineno">802</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return KGML element describing the Relation."""</tt> </tt>
<a name="L803"></a><tt class="py-lineno">803</tt>  <tt class="py-line">        <tt class="py-comment"># The root is this Relation element</tt> </tt>
<a name="L804"></a><tt class="py-lineno">804</tt>  <tt class="py-line">        <tt class="py-name">relation</tt> <tt class="py-op">=</tt> <tt class="py-name">ET</tt><tt class="py-op">.</tt><tt class="py-name">Element</tt><tt class="py-op">(</tt><tt class="py-string">'relation'</tt><tt class="py-op">)</tt> </tt>
<a name="L805"></a><tt class="py-lineno">805</tt>  <tt class="py-line">        <tt class="py-name">relation</tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'entry1'</tt><tt class="py-op">:</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_entry1</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L806"></a><tt class="py-lineno">806</tt>  <tt class="py-line">                           <tt class="py-string">'entry2'</tt><tt class="py-op">:</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_entry2</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L807"></a><tt class="py-lineno">807</tt>  <tt class="py-line">                           <tt class="py-string">'type'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-363" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-363', 'type', 'link-14');">type</a></tt><tt class="py-op">}</tt> </tt>
<a name="L808"></a><tt class="py-lineno">808</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-op">(</tt><tt id="link-364" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-364', 'name', 'link-65');">name</a></tt><tt class="py-op">,</tt> <tt id="link-365" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-365', 'value', 'link-70');">value</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">subtypes</tt><tt class="py-op">:</tt> </tt>
<a name="L809"></a><tt class="py-lineno">809</tt>  <tt class="py-line">            <tt class="py-name">subtype</tt> <tt class="py-op">=</tt> <tt class="py-name">ET</tt><tt class="py-op">.</tt><tt class="py-name">Element</tt><tt class="py-op">(</tt><tt class="py-string">'subtype'</tt><tt class="py-op">)</tt> </tt>
<a name="L810"></a><tt class="py-lineno">810</tt>  <tt class="py-line">            <tt class="py-name">subtype</tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">[</tt><tt id="link-366" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-366', 'name', 'link-65');">name</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-367" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-367', 'value', 'link-70');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L811"></a><tt class="py-lineno">811</tt>  <tt class="py-line">            <tt class="py-name">relation</tt><tt class="py-op">.</tt><tt id="link-368" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-368', 'append', 'link-68');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">subtype</tt><tt class="py-op">)</tt> </tt>
<a name="L812"></a><tt class="py-lineno">812</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">relation</tt> </tt>
</div></div><a name="L813"></a><tt class="py-lineno">813</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Apr  4 10:49:46 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
